<!DOCTYPE html><html><head><title>Help for package Momocs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Momocs}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_ldk'><p>Adds new landmarks on Out and Opn objects</p></a></li>
<li><a href='#apodemus'><p>Data: Outline coordinates of Apodemus (wood mouse) mandibles</p></a></li>
<li><a href='#arrange'><p>Arrange rows by variables</p></a></li>
<li><a href='#as_df'><p>Turn Momocs objects into tydy data_frames</p></a></li>
<li><a href='#at_least'><p>Retain groups with at least n shapes</p></a></li>
<li><a href='#bezier'><p>Calculates Bezier coefficients from a shape</p></a></li>
<li><a href='#bezier_i'><p>Calculates a shape from Bezier coefficients</p></a></li>
<li><a href='#bot'><p>Data: Outline coordinates of beer and whisky bottles.</p></a></li>
<li><a href='#boxplot.OutCoe'><p>Boxplot of morphometric coefficients</p></a></li>
<li><a href='#boxplot.PCA'><p>Boxplot on PCA objects</p></a></li>
<li><a href='#breed'><p>Jitters Coe (and others) objects</p></a></li>
<li><a href='#bridges'><p>Convert between different classes</p></a></li>
<li><a href='#calibrate_deviations'><p>Quantitative calibration, through deviations, for Out and Opn objects</p></a></li>
<li><a href='#calibrate_harmonicpower'><p>Quantitative calibration, through harmonic power, for Out and Opn objects</p></a></li>
<li><a href='#calibrate_r2'><p>Quantitative r2 calibration for Opn objects</p></a></li>
<li><a href='#calibrate_reconstructions'><p>Calibrate using reconstructed shapes</p></a></li>
<li><a href='#chaff'><p>Data: Landmark and semilandmark coordinates on cereal glumes</p></a></li>
<li><a href='#charring'><p>Data: Outline coordinates from an experimental charring on cereal grains</p></a></li>
<li><a href='#chop'><p>Split to several objects based on a factor</p></a></li>
<li><a href='#classification_metrics'><p>Calculate classification metrics on a confusion matrix</p></a></li>
<li><a href='#CLUST'><p>Hierarchical clustering</p></a></li>
<li><a href='#Coe'><p>Coe &quot;super&quot; class</p></a></li>
<li><a href='#coeff_rearrange'><p>Rearrange a matrix of (typically Fourier) coefficients</p></a></li>
<li><a href='#coeff_sel'><p>Helps to select a given number of harmonics from a numerical vector.</p></a></li>
<li><a href='#coeff_split'><p>Converts a numerical description of harmonic coefficients to a named list.</p></a></li>
<li><a href='#col_transp'><p>Transparency helpers and palettes</p></a></li>
<li><a href='#color_palettes'><p>Some color palettes</p></a></li>
<li><a href='#combine'><p>Combine several objects</p></a></li>
<li><a href='#complex'><p>Convert complex to/from cartesian coordinates</p></a></li>
<li><a href='#Coo'><p>Coo &quot;super&quot; class</p></a></li>
<li><a href='#coo_align'><p>Aligns coordinates</p></a></li>
<li><a href='#coo_aligncalliper'><p>Aligns shapes along their 'calliper length'</p></a></li>
<li><a href='#coo_alignminradius'><p>Aligns shapes using their shortest radius</p></a></li>
<li><a href='#coo_alignxax'><p>Aligns shapes along the x-axis</p></a></li>
<li><a href='#coo_angle_edges'><p>Calculates the angle of every edge of a shape</p></a></li>
<li><a href='#coo_angle_tangent'><p>Calculates the tangent angle along the perimeter of a</p>
shape</a></li>
<li><a href='#coo_area'><p>Calculates the area of a shape</p></a></li>
<li><a href='#coo_arrows'><p>Plots (lollipop) differences between two configurations</p></a></li>
<li><a href='#coo_baseline'><p>Register new baselines</p></a></li>
<li><a href='#coo_bookstein'><p>Register Bookstein's coordinates</p></a></li>
<li><a href='#coo_boundingbox'><p>Calculates coordinates of the bounding box</p></a></li>
<li><a href='#coo_calliper'><p>Calculates the calliper length</p></a></li>
<li><a href='#coo_centdist'><p>Returns the distance between everypoints and the centroid</p></a></li>
<li><a href='#coo_center'><p>Centers coordinates</p></a></li>
<li><a href='#coo_centpos'><p>Calculate centroid coordinates</p></a></li>
<li><a href='#coo_centsize'><p>Calculates centroid size</p></a></li>
<li><a href='#coo_check'><p>Checks shapes</p></a></li>
<li><a href='#coo_chull'><p>Calculates the (recursive) convex hull of a shape</p></a></li>
<li><a href='#coo_circularity'><p>Calculates the Haralick's circularity of a shape</p></a></li>
<li><a href='#coo_close'><p>Closes/uncloses shapes</p></a></li>
<li><a href='#coo_convexity'><p>Calculates the convexity of a shape</p></a></li>
<li><a href='#coo_down'><p>coo_down</p>
Retains coordinates with negative y-coordinates</a></li>
<li><a href='#coo_draw'><p>Adds a shape to the current plot</p></a></li>
<li><a href='#coo_draw_rads'><p>Draw radii to the current plot</p></a></li>
<li><a href='#coo_dxy'><p>Calculate abscissa and ordinate on a shape</p></a></li>
<li><a href='#coo_eccentricity'><p>Calculates the eccentricity of a shape</p></a></li>
<li><a href='#coo_elongation'><p>Calculates the elongation of a shape</p></a></li>
<li><a href='#coo_extract'><p>Extract coordinates from a shape</p></a></li>
<li><a href='#coo_flipx'><p>Flips shapes</p></a></li>
<li><a href='#coo_force2close'><p>Forces shapes to close</p></a></li>
<li><a href='#coo_interpolate'><p>Interpolates coordinates</p></a></li>
<li><a href='#coo_intersect_angle'><p>Nearest intersection between a shape and a segment specified with an angle</p></a></li>
<li><a href='#coo_intersect_segment'><p>Nearest intersection between a shape and a segment</p></a></li>
<li><a href='#coo_is_closed'><p>Test if shapes are closed</p></a></li>
<li><a href='#coo_jitter'><p>Jitters shapes</p></a></li>
<li><a href='#coo_ldk'><p>Defines landmarks interactively</p></a></li>
<li><a href='#coo_left'><p>Retains coordinates with negative x-coordinates</p></a></li>
<li><a href='#coo_length'><p>Calculates the length of a shape</p></a></li>
<li><a href='#coo_likely_clockwise'><p>Tests if shapes are (likely) developping clockwise or anticlockwise</p></a></li>
<li><a href='#coo_listpanel'><p>Plots sets of shapes.</p></a></li>
<li><a href='#coo_lolli'><p>Plots (lollipop) differences between two configurations</p></a></li>
<li><a href='#coo_lw'><p>Calculates length and width of a shape</p></a></li>
<li><a href='#coo_nb'><p>Counts coordinates</p></a></li>
<li><a href='#coo_oscillo'><p>Momocs' 'oscilloscope' for Fourier-based approaches</p></a></li>
<li><a href='#coo_perim'><p>Calculates perimeter and variations</p></a></li>
<li><a href='#coo_plot'><p>Plots a single shape</p></a></li>
<li><a href='#coo_range'><p>Calculate coordinates range</p></a></li>
<li><a href='#coo_rectangularity'><p>Calculates the rectangularity of a shape</p></a></li>
<li><a href='#coo_rectilinearity'><p>Calculates the rectilinearity of a shape</p></a></li>
<li><a href='#coo_rev'><p>Reverses coordinates</p></a></li>
<li><a href='#coo_right'><p>Retains coordinates with positive x-coordinates</p></a></li>
<li><a href='#coo_rotate'><p>Rotates coordinates</p></a></li>
<li><a href='#coo_rotatecenter'><p>Rotates shapes with a custom center</p></a></li>
<li><a href='#coo_ruban'><p>Plots differences as (colored) segments aka a ruban</p></a></li>
<li><a href='#coo_sample'><p>Sample coordinates (among points)</p></a></li>
<li><a href='#coo_sample_prop'><p>Sample a proportion of coordinates (among points)</p></a></li>
<li><a href='#coo_samplerr'><p>Samples coordinates (regular radius)</p></a></li>
<li><a href='#coo_scalars'><p>Calculates all scalar descriptors of shape</p></a></li>
<li><a href='#coo_scale'><p>Scales coordinates</p></a></li>
<li><a href='#coo_shearx'><p>Shears shapes</p></a></li>
<li><a href='#coo_slice'><p>Slices shapes between successive coordinates</p></a></li>
<li><a href='#coo_slide'><p>Slides coordinates</p></a></li>
<li><a href='#coo_slidedirection'><p>Slides coordinates in a particular direction</p></a></li>
<li><a href='#coo_slidegap'><p>Slides coordinates using the widest gap</p></a></li>
<li><a href='#coo_smooth'><p>Smoothes coordinates</p></a></li>
<li><a href='#coo_smoothcurve'><p>Smoothes coordinates on curves</p></a></li>
<li><a href='#coo_solidity'><p>Calculates the solidity of a shape</p></a></li>
<li><a href='#coo_tac'><p>Calculates the total absolute curvature of a shape</p></a></li>
<li><a href='#coo_template'><p>'Templates' shapes</p></a></li>
<li><a href='#coo_trans'><p>Translates coordinates</p></a></li>
<li><a href='#coo_trim'><p>Trims both ends coordinates from shape</p></a></li>
<li><a href='#coo_trimbottom'><p>Trims bottom coordinates from shape</p></a></li>
<li><a href='#coo_trimtop'><p>Trims top coordinates from shape</p></a></li>
<li><a href='#coo_truss'><p>Truss measurement</p></a></li>
<li><a href='#coo_untiltx'><p>Removes rotation so that the centroid and a given point are parallel to the x-axis</p></a></li>
<li><a href='#coo_up'><p>Retains coordinates with positive y-coordinates</p></a></li>
<li><a href='#coo_width'><p>Calculates the width of a shape</p></a></li>
<li><a href='#d'><p>A wrapper to calculates euclidean distances between two points</p></a></li>
<li><a href='#def_ldk'><p>Defines new landmarks on Out and Opn objects</p></a></li>
<li><a href='#def_ldk_angle'><p>Add new landmarks based on angular positions</p></a></li>
<li><a href='#def_ldk_tips'><p>Define tips as new landmarks</p></a></li>
<li><a href='#def_links'><p>Defines links between landmarks</p></a></li>
<li><a href='#def_slidings'><p>Defines sliding landmarks matrix</p></a></li>
<li><a href='#dfourier'><p>Discrete cosinus transform</p></a></li>
<li><a href='#dfourier_i'><p>Investe discrete cosinus transform</p></a></li>
<li><a href='#dfourier_shape'><p>Calculates and draws 'dfourier' shapes</p></a></li>
<li><a href='#dissolve'><p>Dissolve Coe objects</p></a></li>
<li><a href='#drawers'><p>grindr drawers for shape plots</p></a></li>
<li><a href='#ed'><p>Calculates euclidean distance between two points.</p></a></li>
<li><a href='#edi'><p>Calculates euclidean intermediate between two points.</p></a></li>
<li><a href='#edm'><p>Calculates euclidean distance every pairs of points in two matrices.</p></a></li>
<li><a href='#edm_nearest'><p>Calculates the shortest euclidean distance found for every point of one</p>
matrix among those of a second.</a></li>
<li><a href='#efourier'><p>Elliptical Fourier transform (and its normalization)</p></a></li>
<li><a href='#efourier_i'><p>Inverse elliptical Fourier transform</p></a></li>
<li><a href='#efourier_shape'><p>Calculates and draw 'efourier' shapes.</p></a></li>
<li><a href='#export'><p>Exports Coe objects and shapes</p></a></li>
<li><a href='#fac_dispatcher'><p>Brew and serve fac from Momocs object</p></a></li>
<li><a href='#fgProcrustes'><p>Full Generalized Procrustes alignment between shapes</p></a></li>
<li><a href='#fgsProcrustes'><p>Full Generalized Procrustes alignment between shapes with sliding landmarks</p></a></li>
<li><a href='#filter'><p>Subset based on conditions</p></a></li>
<li><a href='#flip_PCaxes'><p>Flips PCA axes</p></a></li>
<li><a href='#flower'><p>Data: Measurement of iris flowers</p></a></li>
<li><a href='#fProcrustes'><p>Full Procrustes alignment between two shapes</p></a></li>
<li><a href='#get_chull_area'><p>Calculates convex hull area/volume of PCA scores</p></a></li>
<li><a href='#get_ldk'><p>Retrieves landmarks coordinates</p></a></li>
<li><a href='#get_pairs'><p>Get paired individual on a Coe, PCA or LDA objects</p></a></li>
<li><a href='#get_slidings'><p>Extracts sliding landmarks coordinates</p></a></li>
<li><a href='#harm_pow'><p>Calculates harmonic power given a list from e/t/rfourier</p></a></li>
<li><a href='#hcontrib'><p>Harmonic contribution to shape</p></a></li>
<li><a href='#hearts'><p>Data: Outline coordinates of hand-drawn hearts</p></a></li>
<li><a href='#img_plot'><p>Plots a .jpg image</p></a></li>
<li><a href='#import_Conte'><p>Extract outlines coordinates from an image silhouette</p></a></li>
<li><a href='#import_jpg'><p>Extract outline coordinates from multiple .jpg files</p></a></li>
<li><a href='#import_jpg1'><p>Extract outline coordinates from a single .jpg file</p></a></li>
<li><a href='#import_StereoMorph_curve1'><p>Import files creates by StereoMorph into Momocs</p></a></li>
<li><a href='#import_tps'><p>Import a tps file</p></a></li>
<li><a href='#import_txt'><p>Import coordinates from a .txt file</p></a></li>
<li><a href='#inspect'><p>Graphical inspection of shapes</p></a></li>
<li><a href='#is'><p>Class and component testers</p></a></li>
<li><a href='#is_equallyspacedradii'><p>Tests if coordinates likely have equally spaced radii</p></a></li>
<li><a href='#KMEANS'><p>KMEANS on PCA objects</p></a></li>
<li><a href='#KMEDOIDS'><p>KMEDOIDS</p></a></li>
<li><a href='#layers'><p>grindr layers for multivariate plots</p></a></li>
<li><a href='#layers_morphospace'><p>Morphospace layers</p></a></li>
<li><a href='#LDA'><p>Linear Discriminant Analysis on Coe objects</p></a></li>
<li><a href='#Ldk'><p>Builds an Ldk object</p></a></li>
<li><a href='#ldk_check'><p>Checks 'ldk' shapes</p></a></li>
<li><a href='#ldk_chull'><p>Draws convex hulls around landmark positions</p></a></li>
<li><a href='#ldk_confell'><p>Draws confidence ellipses for landmark positions</p></a></li>
<li><a href='#ldk_contour'><p>Draws kernel density contours around landmark</p></a></li>
<li><a href='#ldk_labels'><p>Add landmarks labels</p></a></li>
<li><a href='#ldk_links'><p>Draws links between landmarks</p></a></li>
<li><a href='#lf_structure'><p>bind_db.Coe &lt;- bind_db.Coo</p>
Extracts structure from filenames</a></li>
<li><a href='#links_all'><p>Creates links (all pairwise combinations) between landmarks</p></a></li>
<li><a href='#links_delaunay'><p>Creates links (Delaunay triangulation) between landmarks</p></a></li>
<li><a href='#MANOVA'><p>Multivariate analysis of (co)variance on Coe objects</p></a></li>
<li><a href='#MANOVA_PW'><p>Pairwise Multivariate analyses of variance</p></a></li>
<li><a href='#MDS'><p>(Metric) multidimensional scaling</p></a></li>
<li><a href='#measure'><p>Measures shape descriptors</p></a></li>
<li><a href='#molars'><p>Data: Outline coordinates of 360 molars</p></a></li>
<li><a href='#Momocs'><p>Momocs</p></a></li>
<li><a href='#morphospace_positions'><p>Calculates nice positions on a plane for drawing shapes</p></a></li>
<li><a href='#mosaic_engine'><p>Plots mosaics of shapes.</p></a></li>
<li><a href='#mosquito'><p>Data: Outline coordinates of mosquito wings.</p></a></li>
<li><a href='#mouse'><p>Data: Outline coordinates of mouse molars</p></a></li>
<li><a href='#MSHAPES'><p>Mean shape calculation for Coo, Coe, etc.</p></a></li>
<li><a href='#mutate'><p>Add new variables</p></a></li>
<li><a href='#NMDS'><p>Non metric multidimensional scaling</p></a></li>
<li><a href='#npoly'><p>Calculate natural polynomial fits on open outlines</p></a></li>
<li><a href='#nsfishes'><p>Data: Outline coordinates of North Sea fishes</p></a></li>
<li><a href='#oak'><p>Data: Configuration of landmarks of oak leaves</p></a></li>
<li><a href='#olea'><p>Data: Outline coordinates of olive seeds open outlines.</p></a></li>
<li><a href='#Opn'><p>Builds an Opn object</p></a></li>
<li><a href='#OpnCoe'><p>Builds an OpnCoe object</p></a></li>
<li><a href='#opoly'><p>Calculate orthogonal polynomial fits on open outlines</p></a></li>
<li><a href='#opoly_i'><p>Calculates shape from a polynomial model</p></a></li>
<li><a href='#Out'><p>Builds an Out object</p></a></li>
<li><a href='#OutCoe'><p>Builds an OutCoe object</p></a></li>
<li><a href='#palettes'><p>Color palettes</p></a></li>
<li><a href='#panel'><p>Family picture of shapes</p></a></li>
<li><a href='#papers'><p>grindr papers for shape plots</p></a></li>
<li><a href='#PCA'><p>Principal component analysis on Coe objects</p></a></li>
<li><a href='#PCcontrib'><p>Shape variation along PC axes</p></a></li>
<li><a href='#perm'><p>Permutes and breed Coe (and others) objects</p></a></li>
<li><a href='#pile'><p>Graphical pile of shapes</p></a></li>
<li><a href='#pix2chc'><p>Convert (x; y) coordinates to chaincoded coordinates</p></a></li>
<li><a href='#plot_CV'><p>Plots a cross-validation table as an heatmap</p></a></li>
<li><a href='#plot_CV2'><p>Plots a cross-correlation table</p></a></li>
<li><a href='#plot_devsegments'><p>Draws colored segments from a matrix of coordinates.</p></a></li>
<li><a href='#plot_LDA'><p>LDA plot using grindr layers</p></a></li>
<li><a href='#plot_MSHAPES'><p>Pairwise comparison of a list of shapes</p></a></li>
<li><a href='#plot_NMDS'><p>NMDS plot unsing grindr layers</p></a></li>
<li><a href='#plot_PCA'><p>PCA plot using grindr layers</p></a></li>
<li><a href='#plot_silhouette'><p>Silhouette plot</p></a></li>
<li><a href='#plot_table'><p>Plots confusion matrix of sample sizes within $fac</p></a></li>
<li><a href='#plot.LDA'><p>Plots Linear Discriminant Analysis</p></a></li>
<li><a href='#plot.PCA'><p>Plots Principal Component Analysis</p></a></li>
<li><a href='#pProcrustes'><p>Partial Procrustes alignment between two shapes</p></a></li>
<li><a href='#Ptolemy'><p>Ptolemaic ellipses and illustration of efourier</p></a></li>
<li><a href='#rearrange_ldk'><p>Rearrange, (select and reorder) landmarks to retain</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#reLDA'><p>&quot;Redo&quot; a LDA on new data</p></a></li>
<li><a href='#rename'><p>Rename columns by name</p></a></li>
<li><a href='#rePCA'><p>&quot;Redo&quot; a PCA on a new Coe</p></a></li>
<li><a href='#rescale'><p>Rescale coordinates from pixels to real length units</p></a></li>
<li><a href='#rfourier'><p>Radii variation Fourier transform (equally spaced radii)</p></a></li>
<li><a href='#rfourier_i'><p>Inverse radii variation Fourier transform</p></a></li>
<li><a href='#rfourier_shape'><p>Calculates and draw 'rfourier' shapes.</p></a></li>
<li><a href='#rm_asym'><p>Removes asymmetric and symmetric variation on OutCoe objects</p></a></li>
<li><a href='#rm_harm'><p>Removes harmonics from Coe objects</p></a></li>
<li><a href='#rm_missing'><p>Remove shapes with missing data in fac</p></a></li>
<li><a href='#rm_uncomplete'><p>Remove shapes with incomplete slices</p></a></li>
<li><a href='#rw_fac'><p>Renames levels on Momocs objects</p></a></li>
<li><a href='#sample_frac'><p>Sample a fraction of shapes</p></a></li>
<li><a href='#sample_n'><p>Sample n shapes</p></a></li>
<li><a href='#scree'><p>How many axes to retain this much of variance or trace ?</p></a></li>
<li><a href='#select'><p>Select columns by name</p></a></li>
<li><a href='#sfourier'><p>Radii variation Fourier transform (equally spaced curvilinear abscissa)</p></a></li>
<li><a href='#sfourier_i'><p>Inverse radii variation Fourier transform</p></a></li>
<li><a href='#sfourier_shape'><p>Calculates and draw 'sfourier' shapes.</p></a></li>
<li><a href='#shapes'><p>Data: Outline coordinates of various shapes</p></a></li>
<li><a href='#slice'><p>Subset based on positions</p></a></li>
<li><a href='#slidings_scheme'><p>Extracts partitions of sliding coordinates</p></a></li>
<li><a href='#stack'><p>Family picture of shapes</p></a></li>
<li><a href='#subsetize'><p>Subsetize various Momocs objects</p></a></li>
<li><a href='#symmetry'><p>Calcuates symmetry indices on OutCoe objects</p></a></li>
<li><a href='#tfourier'><p>Tangent angle Fourier transform</p></a></li>
<li><a href='#tfourier_i'><p>Inverse tangent angle Fourier transform</p></a></li>
<li><a href='#tfourier_shape'><p>Calculates and draws 'tfourier' shapes.</p></a></li>
<li><a href='#tie_jpg_txt'><p>Binds .jpg outlines from .txt landmarks taken on them</p></a></li>
<li><a href='#tps_arr'><p>Deformation 'vector field' using Thin Plate Splines</p></a></li>
<li><a href='#tps_grid'><p>Deformation grids using Thin Plate Splines</p></a></li>
<li><a href='#tps_iso'><p>Deformation isolines using Thin Plate Splines.</p></a></li>
<li><a href='#tps_raw'><p>Vanilla Thin Plate Splines</p></a></li>
<li><a href='#tps2d'><p>Thin Plate Splines for 2D data</p></a></li>
<li><a href='#TraCoe'><p>Traditional morphometrics class</p></a></li>
<li><a href='#trilo'><p>Data: Outline coordinates of cephalic outlines of trilobite</p></a></li>
<li><a href='#verify'><p>Validates Coo objects</p></a></li>
<li><a href='#which_out'><p>Identify outliers</p></a></li>
<li><a href='#wings'><p>Data: Landmarks coordinates of mosquito wings</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Morphometrics using R</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-13</td>
</tr>
<tr>
<td>Description:</td>
<td>The goal of 'Momocs' is to provide a complete, convenient, 
       reproducible and open-source toolkit for 2D morphometrics.
       It includes most common 2D morphometrics approaches on outlines, open outlines, 
       configurations of landmarks, traditional morphometrics, and facilities for data preparation, 
       manipulation and visualization with a consistent grammar throughout.
       It allows reproducible, complex morphometrics analyses and other morphometrics approaches 
       should be easy to plug in, or develop from, on top of this canvas. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/MomX/Momocs/">https://github.com/MomX/Momocs/</a>, <a href="http://momx.github.io/Momocs/">http://momx.github.io/Momocs/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/MomX/Momocs/issues">https://github.com/MomX/Momocs/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.2)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>cluster, dendextend, dplyr, magrittr, geometry, geomorph,
ggplot2, graphics, grDevices, jpeg, MASS, progress,
RColorBrewer, sp, sf, utils, vegan, tibble</td>
</tr>
<tr>
<td>Suggests:</td>
<td>devtools, knitr, rmarkdown, testthat, covr, roxygen2</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-13 10:05:45 UTC; vbonhomme</td>
</tr>
<tr>
<td>Author:</td>
<td>Vincent Bonhomme [aut, cre],
  Julien Claude [aut] (core functions in base R)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Vincent Bonhomme &lt;bonhomme.vincent@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-13 11:13:30 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_ldk'>Adds new landmarks on Out and Opn objects</h2><span id='topic+add_ldk'></span>

<h3>Description</h3>

<p>Helps to add new landmarks on a <code>Coo</code> object on top of existing ones.
The number of landmarks must be specified and rows indices that
correspond to the nearest points clicked on every outlines are
stored in the <code>$ldk</code> slot of the <code>Coo</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_ldk(Coo, nb.ldk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_ldk_+3A_coo">Coo</code></td>
<td>
<p>an Out or Opn object</p>
</td></tr>
<tr><td><code id="add_ldk_+3A_nb.ldk">nb.ldk</code></td>
<td>
<p>the number of landmarks to add on every shape</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that if no landmarks are already defined,
then this function is equivalent to <a href="#topic+def_ldk">def_ldk</a>.
</p>


<h3>Value</h3>

<p>an Out or an Opn object with some landmarks defined
</p>


<h3>See Also</h3>

<p>Other ldk/slidings methods: 
<code><a href="#topic+def_ldk">def_ldk</a>()</code>,
<code><a href="#topic+def_slidings">def_slidings</a>()</code>,
<code><a href="#topic+get_ldk">get_ldk</a>()</code>,
<code><a href="#topic+get_slidings">get_slidings</a>()</code>,
<code><a href="#topic+rearrange_ldk">rearrange_ldk</a>()</code>,
<code><a href="#topic+slidings_scheme">slidings_scheme</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
hearts &lt;- slice(hearts, 1:5) # to make it shorter to try
# click on 3 points, 5 times.
hearts &lt;- def_ldk(hearts, 3)
# Don't forget to save the object returned by def_ldk...
hearts2 &lt;- add_ldk(hearts, 3)
stack(hearts2)
hearts2$ldk

## End(Not run)
</code></pre>

<hr>
<h2 id='apodemus'>Data: Outline coordinates of Apodemus (wood mouse) mandibles</h2><span id='topic+apodemus'></span>

<h3>Description</h3>

<p>Data: Outline coordinates of Apodemus (wood mouse) mandibles
</p>


<h3>Format</h3>

<p>A <a href="#topic+Out">Out</a> object 64 coordinates of 30 wood molar outlines.
</p>


<h3>Source</h3>

<p>Renaud S, Pale JRM, Michaux JR (2003): Adaptive latitudinal trends in
the mandible shape of <em>Apodemus</em> wood mice. <em>Journal of Biogeography</em> 30:1617-1628.
see <code style="white-space: pre;">&#8288;https://onlinelibrary.wiley.com/doi/full/10.1046/j.1365-2699.2003.00932.x&#8288;</code>
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='arrange'>Arrange rows by variables</h2><span id='topic+arrange'></span>

<h3>Description</h3>

<p>Arrange shapes by variables, from the <code>$fac</code>. See examples and <code>?dplyr::arrange</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arrange(.data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="arrange_+3A_.data">.data</code></td>
<td>
<p>a <code>Coo</code>, <code>Coe</code>, <code>PCA</code> object</p>
</td></tr>
<tr><td><code id="arrange_+3A_...">...</code></td>
<td>
<p>logical conditions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>dplyr verbs are maintained.
</p>


<h3>Value</h3>

<p>a Momocs object of the same class.
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>olea
# we create a new column
olea %&gt;% mutate(id=1:length(.)) %$% fac$id
# same but now, shapes are arranged in a desc order, based on id
olea %&gt;% mutate(id=1:length(.)) %&gt;% arrange(desc(id)) %$% fac$id
</code></pre>

<hr>
<h2 id='as_df'>Turn Momocs objects into tydy data_frames</h2><span id='topic+as_df'></span><span id='topic+as_df.Coo'></span><span id='topic+as_df.Coe'></span><span id='topic+as_df.PCA'></span><span id='topic+as_df.LDA'></span>

<h3>Description</h3>

<p>Used in particular for compatibility with the <code>tidyverse</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_df(x, ...)

## S3 method for class 'Coo'
as_df(x, ...)

## S3 method for class 'Coe'
as_df(x, ...)

## S3 method for class 'PCA'
as_df(x, retain, ...)

## S3 method for class 'LDA'
as_df(x, retain, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_df_+3A_x">x</code></td>
<td>
<p>an object, typically a Momocs object</p>
</td></tr>
<tr><td><code id="as_df_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
<tr><td><code id="as_df_+3A_retain">retain</code></td>
<td>
<p>numeric for use with <a href="#topic+scree">scree</a> methods. Defaut to all. If <code style="white-space: pre;">&#8288;&lt;1&#8288;</code>,
enough axes to retain this proportion of variance; if <code style="white-space: pre;">&#8288;&gt;1&#8288;</code>, this number of axes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code><a href="dplyr.html#topic+reexports">dplyr::tibble()</a></code>
</p>


<h3>See Also</h3>

<p>Other bridges functions: 
<code><a href="#topic+bridges">bridges</a></code>,
<code><a href="#topic+complex">complex</a></code>,
<code><a href="#topic+export">export</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># first, some (baby) objects
b &lt;- bot %&gt;% coo_sample(12)
bf &lt;- b %&gt;% efourier(5, norm=TRUE)
# Coo object
b %&gt;% as_df
# Coe object
bf %&gt;% as_df

# PCA object
bf %&gt;% PCA %&gt;% as_df     # all PCs by default
bf %&gt;% PCA %&gt;% as_df(2) # or 2
bf %&gt;% PCA %&gt;% as_df(0.99) # or enough for 99%

# LDA object
bf %&gt;% LDA(~fake) %&gt;% as_df
# same options apply
</code></pre>

<hr>
<h2 id='at_least'>Retain groups with at least n shapes</h2><span id='topic+at_least'></span>

<h3>Description</h3>

<p>Examples are self-speaking.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>at_least(x, fac, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="at_least_+3A_x">x</code></td>
<td>
<p>any Momocs object</p>
</td></tr>
<tr><td><code id="at_least_+3A_fac">fac</code></td>
<td>
<p>the id of name of the $fac column</p>
</td></tr>
<tr><td><code id="at_least_+3A_n">N</code></td>
<td>
<p>minimal number of individuals to retain the group</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>Note</h3>

<p>if N is too ambitious the original object is returned with a message
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>table(trilo$onto)
at_least(trilo, "onto", 9)
at_least(trilo, "onto", 16)
at_least(trilo, "onto", 2000) # too ambitious !
</code></pre>

<hr>
<h2 id='bezier'>Calculates Bezier coefficients from a shape</h2><span id='topic+bezier'></span>

<h3>Description</h3>

<p>Calculates Bezier coefficients from a shape
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bezier(coo, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bezier_+3A_coo">coo</code></td>
<td>
<p>a matrix or a list of (x; y) coordinates</p>
</td></tr>
<tr><td><code id="bezier_+3A_n">n</code></td>
<td>
<p>the degree, by default the number of coordinates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with components:
</p>

<ul>
<li> <p><code>$J</code> matrix of Bezier coefficients
</p>
</li>
<li> <p><code>$B</code> matrix of Bezier vertices.
</p>
</li></ul>



<h3>Note</h3>

<p>Directly borrowed for Claude (2008), and also called <code>bezier</code> there.
Not implemented for open outlines but may be useful for other purposes.
</p>


<h3>References</h3>

<p>Claude, J. (2008) <em>Morphometrics with R</em>, Use R! series,
Springer 316 pp.
</p>


<h3>See Also</h3>

<p>Other bezier functions: 
<code><a href="#topic+bezier_i">bezier_i</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(34)
x &lt;- coo_sample(efourier_shape(), 5)
plot(x, ylim=c(-3, 3), asp=1, type='b', pch=20)
b &lt;- bezier(x)
bi &lt;- bezier_i(b$B)
lines(bi, col='red')
</code></pre>

<hr>
<h2 id='bezier_i'>Calculates a shape from Bezier coefficients</h2><span id='topic+bezier_i'></span>

<h3>Description</h3>

<p>Calculates a shape from Bezier coefficients
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bezier_i(B, nb.pts = 120)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bezier_i_+3A_b">B</code></td>
<td>
<p>a matrix of Bezier vertices, such as those produced by <a href="#topic+bezier">bezier</a></p>
</td></tr>
<tr><td><code id="bezier_i_+3A_nb.pts">nb.pts</code></td>
<td>
<p>the number of points to sample along the curve.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix of (x; y) coordinates
</p>


<h3>Note</h3>

<p>Directly borrowed for Claude (2008), and called <code>beziercurve</code> there.
Not implemented for open outlines but may be useful for other purposes.
</p>


<h3>References</h3>

<p>Claude, J. (2008) <em>Morphometrics with R</em>, Use R! series,
Springer 316 pp.
</p>


<h3>See Also</h3>

<p>Other bezier functions: 
<code><a href="#topic+bezier">bezier</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(34)
x &lt;- coo_sample(efourier_shape(), 5)
plot(x, ylim=c(-3, 3), asp=1, type='b', pch=20)
b &lt;- bezier(x)
bi &lt;- bezier_i(b$B)
lines(bi, col='red')
</code></pre>

<hr>
<h2 id='bot'>Data: Outline coordinates of beer and whisky bottles.</h2><span id='topic+bot'></span>

<h3>Description</h3>

<p>Data: Outline coordinates of beer and whisky bottles.
</p>


<h3>Format</h3>

<p>A <a href="#topic+Out">Out</a> object containing the outlines coordinates and a grouping factor
for 20 beer and 20 whisky bottles
</p>


<h3>Source</h3>

<p>Images have been grabbed on the internet and prepared by the package's
authors. No particular choice has been made on the dimension of the original
images or the brands cited here.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='boxplot.OutCoe'>Boxplot of morphometric coefficients</h2><span id='topic+boxplot.OutCoe'></span><span id='topic+boxplot.Coe'></span>

<h3>Description</h3>

<p>Explores the distribution of coefficient values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'OutCoe'
boxplot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="boxplot.OutCoe_+3A_x">x</code></td>
<td>
<p>the <a href="#topic+Coe">Coe</a> object</p>
</td></tr>
<tr><td><code id="boxplot.OutCoe_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 object
</p>


<h3>See Also</h3>

<p>Other Coe_graphics: 
<code><a href="#topic+hcontrib">hcontrib</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># on OutCoe
bot %&gt;% efourier(9) %&gt;% rm_harm(1) %&gt;% boxplot()

data(olea)
op &lt;- opoly(olea)
boxplot(op)
</code></pre>

<hr>
<h2 id='boxplot.PCA'>Boxplot on PCA objects</h2><span id='topic+boxplot.PCA'></span>

<h3>Description</h3>

<p>Boxplot on PCA objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PCA'
boxplot(x, fac = NULL, nax, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="boxplot.PCA_+3A_x">x</code></td>
<td>
<p><code>PCA</code>, typically obtained with <a href="#topic+PCA">PCA</a></p>
</td></tr>
<tr><td><code id="boxplot.PCA_+3A_fac">fac</code></td>
<td>
<p>factor, or a name or the column id from the $fac slot</p>
</td></tr>
<tr><td><code id="boxplot.PCA_+3A_nax">nax</code></td>
<td>
<p>the range of PC to plot (1 to 99pc total variance by default)</p>
</td></tr>
<tr><td><code id="boxplot.PCA_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot.f &lt;- efourier(bot, 12)
bot.p &lt;- PCA(bot.f)
boxplot(bot.p)
p &lt;- boxplot(bot.p, 1)
#p +  theme_minimal() + scale_fill_grey()
#p + facet_wrap(~PC, scales = "free")
</code></pre>

<hr>
<h2 id='breed'>Jitters Coe (and others) objects</h2><span id='topic+breed'></span><span id='topic+breed.default'></span><span id='topic+breed.Coe'></span>

<h3>Description</h3>

<p>This methods applies column-wise on the <code>coe</code> of any
<a href="#topic+Coe">Coe</a> object but relies on a function that can be used on any matrix. It
simply uses <a href="stats.html#topic+rnorm">rnorm</a> with the mean and sd calculated for every column (or row).
For a <code>Coe</code> object, on every colum, randomly generates coefficients values
centered on the mean of the column, and with a sd equals to it standard deviates
multiplied by <code>rate</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>breed(x, ...)

## Default S3 method:
breed(x, fac, margin = 2, size, rate = 1, ...)

## S3 method for class 'Coe'
breed(x, fac, size, rate = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="breed_+3A_x">x</code></td>
<td>
<p>the object to permute</p>
</td></tr>
<tr><td><code id="breed_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
<tr><td><code id="breed_+3A_fac">fac</code></td>
<td>
<p>a column, a formula or a column id from <code style="white-space: pre;">&#8288;$fac&#8288;</code></p>
</td></tr>
<tr><td><code id="breed_+3A_margin">margin</code></td>
<td>
<p>numeric whether 1 or 2 (rows or columns)</p>
</td></tr>
<tr><td><code id="breed_+3A_size">size</code></td>
<td>
<p>numeric the required size for the final object, same size by default</p>
</td></tr>
<tr><td><code id="breed_+3A_rate">rate</code></td>
<td>
<p>numeric the number of sd for <a href="stats.html#topic+rnorm">rnorm</a>, 1 by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Coe object of same class
</p>


<h3>See Also</h3>

<p>Other farming: 
<code><a href="#topic+perm">perm</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- matrix(1:12, nrow=3)
breed(m, margin=2, size=4)
breed(m, margin=1, size=10)

bot.f &lt;- efourier(bot, 12)
bot.m &lt;- breed(bot.f, size=80)
bot.m %&gt;% PCA %&gt;% plot

# breed fac wise
# bot.f %&gt;%  breed(~type, size=50) %&gt;% PCA %&gt;% plot(~type)
</code></pre>

<hr>
<h2 id='bridges'>Convert between different classes</h2><span id='topic+bridges'></span><span id='topic+l2m'></span><span id='topic+m2l'></span><span id='topic+d2m'></span><span id='topic+m2d'></span><span id='topic+l2a'></span><span id='topic+a2l'></span><span id='topic+a2m'></span><span id='topic+m2a'></span><span id='topic+m2ll'></span>

<h3>Description</h3>

<p>Convert between different classes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>l2m(l)

m2l(m)

d2m(d)

m2d(m)

l2a(l)

a2l(a)

a2m(a)

m2a(m)

m2ll(m, index = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bridges_+3A_l">l</code></td>
<td>
<p><code>list</code> with x and y coordinates as components</p>
</td></tr>
<tr><td><code id="bridges_+3A_m">m</code></td>
<td>
<p><code>matrix</code> of (x; y) coordinates</p>
</td></tr>
<tr><td><code id="bridges_+3A_d">d</code></td>
<td>
<p><code>data.frame</code> with two columns</p>
</td></tr>
<tr><td><code id="bridges_+3A_a">a</code></td>
<td>
<p><code>array</code> of (x; y) coordinates</p>
</td></tr>
<tr><td><code id="bridges_+3A_index">index</code></td>
<td>
<p><code>numeric</code>, the number of coordinates for every slice</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the data in the required class
</p>


<h3>Note</h3>

<p><code>a2m</code>/<code>m2a</code> change, by essence, the dimension of the data.
<code>m2ll</code> is used internally to hanle coo and cur in <code>Ldk</code> objects but may be
useful elsewhere
</p>


<h3>See Also</h3>

<p>Other bridges functions: 
<code><a href="#topic+as_df">as_df</a>()</code>,
<code><a href="#topic+complex">complex</a></code>,
<code><a href="#topic+export">export</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># matrix/list
wings[1] %&gt;% coo_sample(4) %&gt;%
   m2l() %T&gt;% print %&gt;%        # matrix to list
   l2m()                       # and back

# data.frame/matrix
wings[1] %&gt;% coo_sample(4) %&gt;%
   m2d() %T&gt;% print %&gt;%        # matrix to data.frame
   d2m                         # and back

 # list/array
 wings %&gt;% slice(1:2) %$%
 coo %&gt;% l2a %T&gt;% print %&gt;%    # list to array
 a2l                           # and back

 # array/matrix
 wings %&gt;% slice(1:2) %$%
 l2a(coo) %&gt;%                  # and array (from a list)
 a2m %T&gt;% print %&gt;%            # to matrix
 m2a                           # and back

 # m2ll
m2ll(wings[1], c(6, 4, 3, 5)) # grab slices and coordinates
</code></pre>

<hr>
<h2 id='calibrate_deviations'>Quantitative calibration, through deviations, for Out and Opn objects</h2><span id='topic+calibrate_deviations'></span><span id='topic+calibrate_deviations_efourier'></span><span id='topic+calibrate_deviations_tfourier'></span><span id='topic+calibrate_deviations_rfourier'></span><span id='topic+calibrate_deviations_sfourier'></span><span id='topic+calibrate_deviations_npoly'></span><span id='topic+calibrate_deviations_opoly'></span><span id='topic+calibrate_deviations_dfourier'></span>

<h3>Description</h3>

<p>Calculate deviations from original and reconstructed shapes using a
range of harmonic number.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calibrate_deviations()

calibrate_deviations_efourier(
  x,
  id = 1,
  range,
  norm.centsize = TRUE,
  dist.method = edm_nearest,
  interpolate.factor = 1,
  dist.nbpts = 120,
  plot = TRUE
)

calibrate_deviations_tfourier(
  x,
  id = 1,
  range,
  norm.centsize = TRUE,
  dist.method = edm_nearest,
  interpolate.factor = 1,
  dist.nbpts = 120,
  plot = TRUE
)

calibrate_deviations_rfourier(
  x,
  id = 1,
  range,
  norm.centsize = TRUE,
  dist.method = edm_nearest,
  interpolate.factor = 1,
  dist.nbpts = 120,
  plot = TRUE
)

calibrate_deviations_sfourier(
  x,
  id = 1,
  range,
  norm.centsize = TRUE,
  dist.method = edm_nearest,
  interpolate.factor = 1,
  dist.nbpts = 120,
  plot = TRUE
)

calibrate_deviations_npoly(
  x,
  id = 1,
  range,
  norm.centsize = TRUE,
  dist.method = edm_nearest,
  interpolate.factor = 1,
  dist.nbpts = 120,
  plot = TRUE
)

calibrate_deviations_opoly(
  x,
  id = 1,
  range,
  norm.centsize = TRUE,
  dist.method = edm_nearest,
  interpolate.factor = 1,
  dist.nbpts = 120,
  plot = TRUE
)

calibrate_deviations_dfourier(
  x,
  id = 1,
  range,
  norm.centsize = TRUE,
  dist.method = edm_nearest,
  interpolate.factor = 1,
  dist.nbpts = 120,
  plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calibrate_deviations_+3A_x">x</code></td>
<td>
<p>and <code>Out</code> or <code>Opn</code> object on which to calibrate_deviations</p>
</td></tr>
<tr><td><code id="calibrate_deviations_+3A_id">id</code></td>
<td>
<p>the shape on which to perform calibrate_deviations</p>
</td></tr>
<tr><td><code id="calibrate_deviations_+3A_range">range</code></td>
<td>
<p>vector of harmonics (or degree for opoly and npoly on Opn) on which to perform calibrate_deviations.
If not provided, the harmonics corresponding to 0.9, 0.95 and 0.99% of harmonic power
are used.</p>
</td></tr>
<tr><td><code id="calibrate_deviations_+3A_norm.centsize">norm.centsize</code></td>
<td>
<p>logical whether to normalize deviation by the centroid size</p>
</td></tr>
<tr><td><code id="calibrate_deviations_+3A_dist.method">dist.method</code></td>
<td>
<p>a method such as <a href="#topic+edm_nearest">edm_nearest</a> to calculate deviations</p>
</td></tr>
<tr><td><code id="calibrate_deviations_+3A_interpolate.factor">interpolate.factor</code></td>
<td>
<p>a numeric to increase the number of points on the original shape (1 by default)</p>
</td></tr>
<tr><td><code id="calibrate_deviations_+3A_dist.nbpts">dist.nbpts</code></td>
<td>
<p>numeric the number of points to use for deviations calculations</p>
</td></tr>
<tr><td><code id="calibrate_deviations_+3A_plot">plot</code></td>
<td>
<p>logical whether to print the graph (FALSE is you just want the calculations)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that from version 1.1, the calculation changed and fixed a problem. Before,
the 'best' possible shape was calculated using the highest possible number of harmonics.
This worked well for efourier but not for others (eg rfourier, tfourier) as they
are known to be unstable with high number of harmonics. From now on, Momocs uses
the 'real' shape, as it is (so it must be centered) and uses <a href="#topic+coo_interpolate">coo_interpolate</a>
to produce <code>interpolate.factor</code> times more coordinates as the shape
has and using the default <code>dist.method</code>, eg <a href="#topic+edm_nearest">edm_nearest</a>,
the latter finds the euclidean distance, for each point on the reconstructed shape,
the closest point on this interpolated shape. <code>interpolate.factor</code> being set
to 1 by default, no interpolation will be made in you do not ask for it. Note,
that interpolation to decrease artefactual errors may also be done outside
<code>calibrate_deviations</code> and will be probably be removed from it
in further versions.
</p>
<p>Note also that this code is quite old now and would need a good review,
planned for 2018.
</p>
<p>For *poly methods on Opn objects, the deviations are calculated from a degree 12 polynom.
</p>


<h3>Value</h3>

<p>a ggplot object and the full list of intermediate results. See examples.
</p>


<h3>See Also</h3>

<p>Other calibration: 
<code><a href="#topic+calibrate_harmonicpower">calibrate_harmonicpower</a>()</code>,
<code><a href="#topic+calibrate_r2">calibrate_r2</a>()</code>,
<code><a href="#topic+calibrate_reconstructions">calibrate_reconstructions</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b5 &lt;- slice(bot, 1:5) #for the sake of speed
b5 %&gt;% calibrate_deviations_efourier()
b5 %&gt;% calibrate_deviations_rfourier()
b5 %&gt;% calibrate_deviations_tfourier()
b5 %&gt;% calibrate_deviations_sfourier()

o5 &lt;- slice(olea, 1:5) #for the sake of speed
o5 %&gt;% calibrate_deviations_opoly()
o5 %&gt;% calibrate_deviations_npoly()
o5 %&gt;% calibrate_deviations_dfourier()

</code></pre>

<hr>
<h2 id='calibrate_harmonicpower'>Quantitative calibration, through harmonic power, for Out and Opn objects</h2><span id='topic+calibrate_harmonicpower'></span><span id='topic+calibrate_harmonicpower_efourier'></span><span id='topic+calibrate_harmonicpower_rfourier'></span><span id='topic+calibrate_harmonicpower_tfourier'></span><span id='topic+calibrate_harmonicpower_sfourier'></span><span id='topic+calibrate_harmonicpower_dfourier'></span>

<h3>Description</h3>

<p>Estimates the number of harmonics required for the four Fourier methods
implemented in Momocs: elliptical Fourier analysis
(see <a href="#topic+efourier">efourier</a>), radii variation analysis (see <a href="#topic+rfourier">rfourier</a>)
and tangent angle analysis (see <a href="#topic+tfourier">tfourier</a>) and
discrete Fourier transform (see <a href="#topic+dfourier">dfourier</a>).
It returns and can plot cumulated harmonic power whether dropping
the first harmonic or not, and based and the maximum possible number
of harmonics on the <code>Coo</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calibrate_harmonicpower()

calibrate_harmonicpower_efourier(
  x,
  id = 1:length(x),
  nb.h,
  drop = 1,
  thresh = c(90, 95, 99, 99.9),
  plot = TRUE
)

calibrate_harmonicpower_rfourier(
  x,
  id = 1:length(x),
  nb.h,
  drop = 1,
  thresh = c(90, 95, 99, 99.9),
  plot = TRUE
)

calibrate_harmonicpower_tfourier(
  x,
  id = 1:length(x),
  nb.h,
  drop = 1,
  thresh = c(90, 95, 99, 99.9),
  plot = TRUE
)

calibrate_harmonicpower_sfourier(
  x,
  id = 1:length(x),
  nb.h,
  drop = 1,
  thresh = c(90, 95, 99, 99.9),
  plot = TRUE
)

calibrate_harmonicpower_dfourier(
  x,
  id = 1:length(x),
  nb.h,
  drop = 1,
  thresh = c(90, 95, 99, 99.9),
  plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calibrate_harmonicpower_+3A_x">x</code></td>
<td>
<p>a <code>Coo</code> of <code>Opn</code> object</p>
</td></tr>
<tr><td><code id="calibrate_harmonicpower_+3A_id">id</code></td>
<td>
<p>the shapes on which to perform calibrate_harmonicpower. All of them by default</p>
</td></tr>
<tr><td><code id="calibrate_harmonicpower_+3A_nb.h">nb.h</code></td>
<td>
<p>numeric the maximum number of harmonic, on which to base the cumsum</p>
</td></tr>
<tr><td><code id="calibrate_harmonicpower_+3A_drop">drop</code></td>
<td>
<p>numeric the number of harmonics to drop for the cumulative sum</p>
</td></tr>
<tr><td><code id="calibrate_harmonicpower_+3A_thresh">thresh</code></td>
<td>
<p>vector of numeric for drawing horizontal lines, and also used for
<code>minh</code> below</p>
</td></tr>
<tr><td><code id="calibrate_harmonicpower_+3A_plot">plot</code></td>
<td>
<p>logical whether to plot the result or simply return the matrix
Silent message and progress bars (if any) with <code>options("verbose"=FALSE)</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The power of a given harmonic <code class="reqn">n</code> is calculated as follows for
elliptical Fourier analysis and the n-th harmonic:
<code class="reqn">HarmonicPower_n \frac{A^2_n+B^2_n+C^2_n+D^2_n}{2}</code>
and as follows for radii variation and tangent angle:
<code class="reqn">HarmonicPower_n= \frac{A^2_n+B^2_n+C^2_n+D^2_n}{2}</code>
</p>


<h3>Value</h3>

<p>returns a list with component:
</p>

<ul>
<li> <p><code>gg</code> a ggplot object, <code>q</code> the quantile matrix
</p>
</li>
<li> <p><code>minh</code> a quick summary that returns the number of harmonics required to achieve
a certain proportion of the total harmonic power.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other calibration: 
<code><a href="#topic+calibrate_deviations">calibrate_deviations</a>()</code>,
<code><a href="#topic+calibrate_r2">calibrate_r2</a>()</code>,
<code><a href="#topic+calibrate_reconstructions">calibrate_reconstructions</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b5 &lt;- bot %&gt;% slice(1:5)
b5  %&gt;% calibrate_harmonicpower_efourier(nb.h=12)
b5  %&gt;% calibrate_harmonicpower_rfourier(nb.h=12)
b5  %&gt;% calibrate_harmonicpower_tfourier(nb.h=12)
b5  %&gt;% calibrate_harmonicpower_sfourier(nb.h=12)

# on Opn
olea %&gt;% slice(1:5) %&gt;%
    calibrate_harmonicpower_dfourier(nb.h=12)

# let customize the ggplot
library(ggplot2)
cal &lt;- b5  %&gt;% calibrate_harmonicpower_efourier(nb.h=12)
cal$gg + theme_minimal() +
coord_cartesian(xlim=c(3.5, 12.5), ylim=c(90, 100)) +
ggtitle("Harmonic power calibration")

</code></pre>

<hr>
<h2 id='calibrate_r2'>Quantitative r2 calibration for Opn objects</h2><span id='topic+calibrate_r2'></span><span id='topic+calibrate_r2_opoly'></span><span id='topic+calibrate_r2_npoly'></span>

<h3>Description</h3>

<p>Estimates the r2 to calibrate the degree for <a href="#topic+npoly">npoly</a> and <a href="#topic+opoly">opoly</a> methods.
Also returns a plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calibrate_r2()

calibrate_r2_opoly(
  Opn,
  id = 1:length(Opn),
  degree.range = 1:8,
  thresh = c(0.9, 0.95, 0.99, 0.999),
  plot = TRUE,
  ...
)

calibrate_r2_npoly(
  Opn,
  id = 1:length(Opn),
  degree.range = 1:8,
  thresh = c(0.9, 0.95, 0.99, 0.999),
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calibrate_r2_+3A_opn">Opn</code></td>
<td>
<p>an Opn object</p>
</td></tr>
<tr><td><code id="calibrate_r2_+3A_id">id</code></td>
<td>
<p>the ids of shapes on which to calculate r2 (all by default)</p>
</td></tr>
<tr><td><code id="calibrate_r2_+3A_degree.range">degree.range</code></td>
<td>
<p>on which to calculate r2</p>
</td></tr>
<tr><td><code id="calibrate_r2_+3A_thresh">thresh</code></td>
<td>
<p>the threshold to return diagnostic</p>
</td></tr>
<tr><td><code id="calibrate_r2_+3A_plot">plot</code></td>
<td>
<p>logical whether to print the plot</p>
</td></tr>
<tr><td><code id="calibrate_r2_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
</table>


<h3>Details</h3>

<p>May be long, so you can estimate it on a sample either with id here, or one of
<a href="#topic+sample_n">sample_n</a> or <a href="#topic+sample_frac">sample_frac</a>
</p>


<h3>Value</h3>

<p>a ggpot2 object
</p>


<h3>Note</h3>

<p>Silent message and progress bars (if any) with <code>options("verbose"=FALSE)</code>.
</p>


<h3>See Also</h3>

<p>Other calibration: 
<code><a href="#topic+calibrate_deviations">calibrate_deviations</a>()</code>,
<code><a href="#topic+calibrate_harmonicpower">calibrate_harmonicpower</a>()</code>,
<code><a href="#topic+calibrate_reconstructions">calibrate_reconstructions</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>olea %&gt;% slice(1:5) %&gt;% #for the sake of spped
    calibrate_r2_opoly(degree.range=1:5, thresh=c(0.9, 0.99))

olea %&gt;% slice(1:5) %&gt;% #for the sake of spped
    calibrate_r2_npoly(degree.range=1:5, thresh=c(0.9, 0.99))

</code></pre>

<hr>
<h2 id='calibrate_reconstructions'>Calibrate using reconstructed shapes</h2><span id='topic+calibrate_reconstructions'></span><span id='topic+calibrate_reconstructions_efourier'></span><span id='topic+calibrate_reconstructions_rfourier'></span><span id='topic+calibrate_reconstructions_tfourier'></span><span id='topic+calibrate_reconstructions_sfourier'></span><span id='topic+calibrate_reconstructions_npoly'></span><span id='topic+calibrate_reconstructions_opoly'></span><span id='topic+calibrate_reconstructions_dfourier'></span>

<h3>Description</h3>

<p>Calculate and displays reconstructed shapes using a
range of harmonic number. Compare them visually with the maximal fit.
This explicitely demonstrates how robust efourier is compared to tfourier and rfourier.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calibrate_reconstructions_efourier(x, id, range = 1:9)

calibrate_reconstructions_rfourier(x, id, range = 1:9)

calibrate_reconstructions_tfourier(x, id, range = 1:9)

calibrate_reconstructions_sfourier(x, id, range = 1:9)

calibrate_reconstructions_npoly(
  x,
  id,
  range = 2:10,
  baseline1 = c(-1, 0),
  baseline2 = c(1, 0)
)

calibrate_reconstructions_opoly(
  x,
  id,
  range = 2:10,
  baseline1 = c(-1, 0),
  baseline2 = c(1, 0)
)

calibrate_reconstructions_dfourier(
  x,
  id,
  range = 2:10,
  baseline1 = c(-1, 0),
  baseline2 = c(1, 0)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calibrate_reconstructions_+3A_x">x</code></td>
<td>
<p>the <code>Coo</code> object on which to calibrate_reconstructions</p>
</td></tr>
<tr><td><code id="calibrate_reconstructions_+3A_id">id</code></td>
<td>
<p>the shape on which to perform calibrate_reconstructions</p>
</td></tr>
<tr><td><code id="calibrate_reconstructions_+3A_range">range</code></td>
<td>
<p>vector of harmonics on which to perform calibrate_reconstructions</p>
</td></tr>
<tr><td><code id="calibrate_reconstructions_+3A_baseline1">baseline1</code></td>
<td>
<p><code class="reqn">(x; y)</code> coordinates for the first point of the baseline</p>
</td></tr>
<tr><td><code id="calibrate_reconstructions_+3A_baseline2">baseline2</code></td>
<td>
<p><code class="reqn">(x; y)</code> coordinates for the second point of the baseline</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object and the full list of intermediate results. See examples.
</p>


<h3>See Also</h3>

<p>Other calibration: 
<code><a href="#topic+calibrate_deviations">calibrate_deviations</a>()</code>,
<code><a href="#topic+calibrate_harmonicpower">calibrate_harmonicpower</a>()</code>,
<code><a href="#topic+calibrate_r2">calibrate_r2</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
### On Out
shapes %&gt;%
    calibrate_reconstructions_efourier(id=1, range=1:6)

# you may prefer efourier...
shapes %&gt;%
    calibrate_reconstructions_tfourier(id=1, range=1:6)

#' you may prefer efourier...
shapes %&gt;%
    calibrate_reconstructions_rfourier(id=1, range=1:6)

#' you may prefer efourier... # todo
#shapes %&gt;%
#     calibrate_reconstructions_sfourier(id=5, range=1:6)

### On Opn
olea %&gt;%
    calibrate_reconstructions_opoly(id=1)

olea %&gt;%
    calibrate_reconstructions_npoly(id=1)

olea %&gt;%
    calibrate_reconstructions_dfourier(id=1)

</code></pre>

<hr>
<h2 id='chaff'>Data: Landmark and semilandmark coordinates on cereal glumes</h2><span id='topic+chaff'></span>

<h3>Description</h3>

<p>Data: Landmark and semilandmark coordinates on cereal glumes
</p>


<h3>Format</h3>

<p>An <a href="#topic+Ldk">Ldk</a> object with 21 configurations of landmarks and semi-landmarks (4 partitions)
sampled on cereal glumes
</p>


<h3>Source</h3>

<p>Research support was provided by the European Research Council
(Evolutionary Origins of Agriculture (grant no. 269830-EOA)
PI: Glynis Jones, Dept of Archaeology, Sheffield, UK. Data collected by Emily Forster.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='charring'>Data: Outline coordinates from an experimental charring on cereal grains</h2><span id='topic+charring'></span>

<h3>Description</h3>

<p>Data: Outline coordinates from an experimental charring on cereal grains
</p>


<h3>Format</h3>

<p>An <a href="#topic+Out">Out</a> object with 18 grains, 3 views on each, for 2 cereal species, charred at
different temperatures for 6 hours (0C (no charring), 230C and 260C).
</p>


<h3>Source</h3>

<p>Research support was provided by the European Research Council
(Evolutionary Origins of Agriculture (grant no. 269830-EOA)
PI: Glynis Jones, Dept of Archaeology, Sheffield, UK. Data collected by Emily Forster.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='chop'>Split to several objects based on a factor</h2><span id='topic+chop'></span>

<h3>Description</h3>

<p>Rougher slicing that accepts a classifier
ie a column name from the <code>$fac</code> on Momocs classes.
Returns a named (after every level) list that can be lapply-ed and combined. See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chop(.data, fac)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chop_+3A_.data">.data</code></td>
<td>
<p>a <code>Coo</code> or <code>Coe</code> object</p>
</td></tr>
<tr><td><code id="chop_+3A_fac">fac</code></td>
<td>
<p>a column name from the <code>$fac</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named list of <code>Coo</code> or <code>Coe</code> objects
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> olea %&gt;%
      filter(var == "Aglan") %&gt;% # to have a balanced nb of 'view'
      chop(~view) %&gt;%    # split into a list of 2
      npoly %&gt;%          # separately apply npoly
                         # strict equivalent to lapply(npoly)
      combine %&gt;%       # recombine
      PCA %&gt;% plot      # an illustration of the 2 views
      # treated separately
</code></pre>

<hr>
<h2 id='classification_metrics'>Calculate classification metrics on a confusion matrix</h2><span id='topic+classification_metrics'></span>

<h3>Description</h3>

<p>In some cases, the class correctness or the proportion of correctly classified
individuals is not enough, so here are more detailed metrics when working on classification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>classification_metrics(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="classification_metrics_+3A_x">x</code></td>
<td>
<p>a <code>table</code> or an <a href="#topic+LDA">LDA</a> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following components is returned:
</p>

<ol>
<li> <p><code>accuracy</code>  the fraction of instances that are correctly classified
</p>
</li>
<li> <p><code>macro_prf</code> data.frame containing <code>precision</code>
(the fraction of correct predictions for a certain class);
<code>recall</code>, the fraction of instances of a class that were correctly predicted;
<code>f1</code> the harmonic mean (or a weighted average) of precision and recall.
</p>
</li>
<li> <p><code>macro_avg</code>, just the average of the three <code>macro_prf</code> indices
</p>
</li>
<li> <p><code>ova</code> a list of one-vs-all confusion matrices for each class
</p>
</li>
<li> <p><code>ova_sum</code> a single of all ova matrices
</p>
</li>
<li> <p><code>kappa</code> measure of agreement between the predictions and the actual labels
</p>
</li></ol>



<h3>See Also</h3>

<p>The pages below are of great interest to understand these metrics. The code
used is partley derived from the Revolution Analytics blog post (with their authorization). Thanks to them!
</p>

<ol>
<li> <p><a href="https://en.wikipedia.org/wiki/Precision_and_recall">https://en.wikipedia.org/wiki/Precision_and_recall</a>
</p>
</li>
<li> <p><code>https://blog.revolutionanalytics.com/2016/03/com_class_eval_metrics_r.html</code>
</p>
</li></ol>

<p>Other multivariate: 
<code><a href="#topic+CLUST">CLUST</a>()</code>,
<code><a href="#topic+KMEANS">KMEANS</a>()</code>,
<code><a href="#topic+KMEDOIDS">KMEDOIDS</a>()</code>,
<code><a href="#topic+LDA">LDA</a>()</code>,
<code><a href="#topic+MANOVA_PW">MANOVA_PW</a>()</code>,
<code><a href="#topic+MANOVA">MANOVA</a>()</code>,
<code><a href="#topic+MDS">MDS</a>()</code>,
<code><a href="#topic+MSHAPES">MSHAPES</a>()</code>,
<code><a href="#topic+NMDS">NMDS</a>()</code>,
<code><a href="#topic+PCA">PCA</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># some morphometrics on 'hearts'
hearts %&gt;% fgProcrustes(tol=1) %&gt;%
coo_slide(ldk=1) %&gt;% efourier(norm=FALSE) %&gt;% PCA() %&gt;%
# now the LDA and its summary
LDA(~aut) %&gt;% classification_metrics()
</code></pre>

<hr>
<h2 id='CLUST'>Hierarchical clustering</h2><span id='topic+CLUST'></span><span id='topic+CLUST.default'></span><span id='topic+CLUST.Coe'></span>

<h3>Description</h3>

<p>Performs hierarchical clustering through <a href="stats.html#topic+dist">dist</a> and <a href="stats.html#topic+hclust">hclust</a>. So far it is mainly
a wrapper around these two functions, plus plotting using the <code>dendextend</code> package facilities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CLUST(x, ...)

## Default S3 method:
CLUST(x, ...)

## S3 method for class 'Coe'
CLUST(
  x,
  fac,
  type = c("horizontal", "vertical", "fan")[1],
  k,
  dist_method = "euclidean",
  hclust_method = "complete",
  retain = 0.99,
  labels,
  lwd = 1/4,
  cex = 1/2,
  palette = pal_qual,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CLUST_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+Coe">Coe</a> or <a href="#topic+PCA">PCA</a> object</p>
</td></tr>
<tr><td><code id="CLUST_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
<tr><td><code id="CLUST_+3A_fac">fac</code></td>
<td>
<p>factor specification for <a href="#topic+fac_dispatcher">fac_dispatcher</a></p>
</td></tr>
<tr><td><code id="CLUST_+3A_type">type</code></td>
<td>
<p><code>character</code> one of <code>c("horizontal", "vertical", "fan")</code> (default: <code>horizontal</code>)</p>
</td></tr>
<tr><td><code id="CLUST_+3A_k">k</code></td>
<td>
<p><code>numeric</code> if provided and greater than 1, cut the tree into this number of groups</p>
</td></tr>
<tr><td><code id="CLUST_+3A_dist_method">dist_method</code></td>
<td>
<p>to feed <a href="stats.html#topic+dist">dist</a>'s <code>method</code> argument, that is one of
<code>euclidean</code> (default), <code>maximum</code>, <code>manhattan</code>, <code>canberra</code>, <code>binary</code> or <code>minkowski</code>.</p>
</td></tr>
<tr><td><code id="CLUST_+3A_hclust_method">hclust_method</code></td>
<td>
<p>to feed <a href="stats.html#topic+hclust">hclust</a>'s <code>method</code> argument, one of
<code>ward.D</code>, <code>ward.D2</code>, <code>single</code>, <code>complete</code> (default), <code>average</code>, <code>mcquitty</code>, <code>median</code> or <code>centroid</code>.</p>
</td></tr>
<tr><td><code id="CLUST_+3A_retain">retain</code></td>
<td>
<p>number of axis to retain if a <a href="#topic+PCA">PCA</a> object is passed. If a number &lt; 1 is passed, then the number of PCs retained
will be enough to capture this proportion of variance via <a href="#topic+scree_min">scree_min</a></p>
</td></tr>
<tr><td><code id="CLUST_+3A_labels">labels</code></td>
<td>
<p>factor specification for labelling tips and to feed <a href="#topic+fac_dispatcher">fac_dispatcher</a></p>
</td></tr>
<tr><td><code id="CLUST_+3A_lwd">lwd</code></td>
<td>
<p>for branches (default: <code>0.25</code>)</p>
</td></tr>
<tr><td><code id="CLUST_+3A_cex">cex</code></td>
<td>
<p>for labels (default: <code>1</code>)</p>
</td></tr>
<tr><td><code id="CLUST_+3A_palette">palette</code></td>
<td>
<p>one of available <a href="#topic+palettes">palettes</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> plot
</p>


<h3>See Also</h3>

<p>Other multivariate: 
<code><a href="#topic+KMEANS">KMEANS</a>()</code>,
<code><a href="#topic+KMEDOIDS">KMEDOIDS</a>()</code>,
<code><a href="#topic+LDA">LDA</a>()</code>,
<code><a href="#topic+MANOVA_PW">MANOVA_PW</a>()</code>,
<code><a href="#topic+MANOVA">MANOVA</a>()</code>,
<code><a href="#topic+MDS">MDS</a>()</code>,
<code><a href="#topic+MSHAPES">MSHAPES</a>()</code>,
<code><a href="#topic+NMDS">NMDS</a>()</code>,
<code><a href="#topic+PCA">PCA</a>()</code>,
<code><a href="#topic+classification_metrics">classification_metrics</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># On Coe
bf &lt;- bot %&gt;% efourier(6)
CLUST(bf)
# with a factor and vertical
CLUST(bf, ~type, "v")
# with some cutting and different dist/hclust methods
CLUST(bf,
      dist_method="maximum", hclust_method="average",
      labels=~type, k=3, lwd=1, cex=1, palette=pal_manual(c("green", "yellow", "red")))

# On PCA
bf %&gt;% PCA %&gt;% CLUST

</code></pre>

<hr>
<h2 id='Coe'>Coe &quot;super&quot; class</h2><span id='topic+Coe'></span>

<h3>Description</h3>

<p><code>Coe</code> class is the 'parent' or 'super' class of
<code><a href="#topic+OutCoe">OutCoe</a></code>, <code><a href="#topic+OpnCoe">OpnCoe</a></code>, <code>LdkCoe</code> and <code>TraCoe</code> classes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Coe(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Coe_+3A_...">...</code></td>
<td>
<p>anything and, anyway, this function will simply returns a message.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Useful shortcuts are described below. See <code>browseVignettes("Momocs")</code> for
a detail of the design behind Momocs' classes.
</p>
<p><code>Coe</code> class is the 'parent' class of the following 'child' classes </p>

<ul>
<li> <p><code><a href="#topic+OutCoe">OutCoe</a></code> for coefficients from closed <b>out</b>lines morphometrics
</p>
</li>
<li> <p><code><a href="#topic+OpnCoe">OpnCoe</a></code> for coefficients from <b>op</b>e<b>n</b> outlines morphometrics
</p>
</li>
<li> <p><code>LdkCoe</code> for coefficients from configuration of <b>l</b>an<b>d</b>mar<b>k</b>s morphometrics.
</p>
</li></ul>

<p>In other words, <code><a href="#topic+OutCoe">OutCoe</a></code>, <code><a href="#topic+OpnCoe">OpnCoe</a></code> and <code>LdkCoe</code> classes
are all, primarily, <code>Coe</code> objects on which we define generic <em>and</em>
specific methods. See their respective help pages for more help.
</p>
<p>You can access all the methods available for <code>Coe</code> objects
with <code>methods(class=Coe)</code>.
</p>


<h3>Value</h3>

<p>a list of class Coe
</p>


<h3>See Also</h3>

<p>Other classes: 
<code><a href="#topic+Coo">Coo</a>()</code>,
<code><a href="#topic+Ldk">Ldk</a>()</code>,
<code><a href="#topic+OpnCoe">OpnCoe</a>()</code>,
<code><a href="#topic+Opn">Opn</a>()</code>,
<code><a href="#topic+OutCoe">OutCoe</a>()</code>,
<code><a href="#topic+Out">Out</a>()</code>,
<code><a href="#topic+TraCoe">TraCoe</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># to see all methods for Coe objects.
methods(class='Coe')
# to see all methods for OutCoe objects.
methods(class='OutCoe') # same for OpnCoe, LdkCoe, TraCoe

bot.f &lt;- efourier(bot, 12)
bot.f
class(bot.f)
inherits(bot.f, "Coe")

# if you want to work directly on the matrix of coefficients
bot.f$coe

#getters
bot.f[1]
bot.f[1:5]

#setters
bot.f[1] &lt;- 1:48
bot.f[1]

bot.f[1:5] &lt;- matrix(1:48, nrow=5, ncol=48, byrow=TRUE)
bot.f[1:5]

# An illustration of Momocs design. See also browseVignettes("Momocs")
op &lt;- opoly(olea, 5)
op
class(op)
op$coe # same thing

wp &lt;- fgProcrustes(wings, tol=1e-4)
wp
class(wp) # for Ldk methods, LdkCoe objects can also be considered as Coo objects
# so you can apply all Ldk methods available.
wp$coe # Procrustes aligned coordinates
</code></pre>

<hr>
<h2 id='coeff_rearrange'>Rearrange a matrix of (typically Fourier) coefficients</h2><span id='topic+coeff_rearrange'></span>

<h3>Description</h3>

<p>Momocs uses colnamed matrices to store (typically) Fourier coefficients
in <a href="#topic+Coe">Coe</a> objects (typically <a href="#topic+OutCoe">OutCoe</a>). They are arranged as rank-wise:
<code>A1, A2, ..., An, B1, ..., Bn, C1, ..., Cn, D1, ..., Dn</code>. From other softwares they may arrive
as <code>A1, B1, C1, D1, ..., An, Bn, Cn, Dn</code>, this functions helps to go
from one to the other format. In short, this function rearranges column order. See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coeff_rearrange(x, by = c("name", "rank")[1])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coeff_rearrange_+3A_x">x</code></td>
<td>
<p>matrix (with colnames)</p>
</td></tr>
<tr><td><code id="coeff_rearrange_+3A_by">by</code></td>
<td>
<p>character either &quot;name&quot; (<code>A1, A2, ..</code>) or &quot;rank&quot; (<code>A1, B1, ...</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m_name &lt;- m_rank &lt;- matrix(1:32, 2, 16)
# this one is ordered by name
colnames(m_name) &lt;- paste0(rep(letters[1:4], each=4), 1:4)
# this one is ordered by rank
colnames(m_rank) &lt;- paste0(letters[1:4], rep(1:4, each=4))

m_rank
m_rank %&gt;% coeff_rearrange(by="name")
m_rank %&gt;% coeff_rearrange(by="rank") #no change

m_name
m_name %&gt;% coeff_rearrange(by="name") # no change
m_name %&gt;% coeff_rearrange(by="rank")
</code></pre>

<hr>
<h2 id='coeff_sel'>Helps to select a given number of harmonics from a numerical vector.</h2><span id='topic+coeff_sel'></span>

<h3>Description</h3>

<p><code>coeff_sel</code> helps to select a given number of harmonics by returning
their indices when arranged as a numeric vector. For instance, harmonic
coefficients are arranged in the <code>$coe</code> slot of <code><a href="#topic+Coe">Coe</a></code>-objects in
that way: <code class="reqn">A_1, \dots, A_n, B_1, \dots, B_n, C_1, \dots, C_n, D_1,
\dots, D-n</code> after an elliptical Fourier analysis (see <a href="#topic+efourier">efourier</a> and
<a href="#topic+efourier">efourier</a>) while <code class="reqn">C_n and D_n</code> harmonic are absent for radii
variation and tangent angle approaches (see <a href="#topic+rfourier">rfourier</a> and
<a href="#topic+tfourier">tfourier</a> respectively). . This function is used internally but might
be of interest elwewhere.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coeff_sel(retain = 8, drop = 0, nb.h = 32, cph = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coeff_sel_+3A_retain">retain</code></td>
<td>
<p><code>numeric</code>. The number of harmonics to retain.</p>
</td></tr>
<tr><td><code id="coeff_sel_+3A_drop">drop</code></td>
<td>
<p><code>numeric</code>. The number of harmonics to drop</p>
</td></tr>
<tr><td><code id="coeff_sel_+3A_nb.h">nb.h</code></td>
<td>
<p><code>numeric</code>. The maximum harmonic rank.</p>
</td></tr>
<tr><td><code id="coeff_sel_+3A_cph">cph</code></td>
<td>
<p><code>numeric</code>. Must be set to 2 for <code>rfourier</code> and
<code>tfourier</code> were used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>coeff_sel</code> returns indices that can be used to select columns
from an harmonic coefficient matrix. <code>coeff_split</code> returns a named list
of coordinates.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot.f &lt;- efourier(bot, 32)
coe &lt;- bot.f$coe # the raw matrix
coe
# if you want, say the first 8 harmonics but not the first one
retain &lt;- coeff_sel(retain=8, drop=1, nb.h=32, cph=4)
head(coe[, retain])
</code></pre>

<hr>
<h2 id='coeff_split'>Converts a numerical description of harmonic coefficients to a named list.</h2><span id='topic+coeff_split'></span>

<h3>Description</h3>

<p><code>coeff_split</code> returns a named list of coordinates from a vector of
harmonic coefficients. For instance, harmonic coefficients are arranged in
the <code>$coe</code> slot of <code>Coe</code>-objects in that way: <code class="reqn">A_1, \dots,
A_n, B_1, \dots, B_n, C_1, \dots, C_n, D_1, \dots, D-n</code> after an elliptical
Fourier analysis (see <a href="#topic+efourier">efourier</a> and <a href="#topic+efourier">efourier</a>) while <code class="reqn">C_n
and D_n</code> harmonic are absent for radii variation and tangent angle
approaches (see <a href="#topic+rfourier">rfourier</a> and <a href="#topic+tfourier">tfourier</a> respectively). This
function is used internally but might be of interest elwewhere.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coeff_split(cs, nb.h = 8, cph = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coeff_split_+3A_cs">cs</code></td>
<td>
<p>A <code>vector</code> of harmonic coefficients.</p>
</td></tr>
<tr><td><code id="coeff_split_+3A_nb.h">nb.h</code></td>
<td>
<p><code>numeric</code>. The maximum harmonic rank.</p>
</td></tr>
<tr><td><code id="coeff_split_+3A_cph">cph</code></td>
<td>
<p><code>numeric</code>. Must be set to 2 for <code>rfourier</code> and
<code>tfourier</code> were used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a named list of coordinates.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coeff_split(1:128, nb.h=32, cph=4) # efourier
coeff_split(1:64, nb.h=32, cph=2)  # t/r fourier
</code></pre>

<hr>
<h2 id='col_transp'>Transparency helpers and palettes</h2><span id='topic+col_transp'></span><span id='topic+col_alpha'></span>

<h3>Description</h3>

<p>To ease transparency handling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>col_transp(n, col = "#000000", ceiling = 1)

col_alpha(cols, transp = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="col_transp_+3A_n">n</code></td>
<td>
<p>the number of colors to generate</p>
</td></tr>
<tr><td><code id="col_transp_+3A_col">col</code></td>
<td>
<p>a color in hexadecimal format on which to generate levels of transparency</p>
</td></tr>
<tr><td><code id="col_transp_+3A_ceiling">ceiling</code></td>
<td>
<p>the maximal opacity (from 0 to 1)</p>
</td></tr>
<tr><td><code id="col_transp_+3A_cols">cols</code></td>
<td>
<p>on or more colors, provided as hexadecimal values</p>
</td></tr>
<tr><td><code id="col_transp_+3A_transp">transp</code></td>
<td>
<p>numeric between 0 and 1, the value of the transparency to obtain</p>
</td></tr>
</table>


<h3>Value</h3>

<p>colors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- col_transp(10, col='#000000')
x
barplot(1:10, col=x, main='a transparent black is grey')

summer10 &lt;- col_summer(10)
summer10
summer10.transp8 &lt;- col_alpha(summer10, 0.8)
summer10.transp8
summer10.transp2 &lt;- col_alpha(summer10, 0.8)
summer10.transp2
x &lt;- 1:10
barplot(x, col=summer10.transp8)
barplot(x/2, col=summer10.transp2, add=TRUE)
</code></pre>

<hr>
<h2 id='color_palettes'>Some color palettes</h2><span id='topic+color_palettes'></span><span id='topic+col_summer'></span><span id='topic+Palettes'></span><span id='topic+col_summer2'></span><span id='topic+col_spring'></span><span id='topic+col_autumn'></span><span id='topic+col_black'></span><span id='topic+col_solarized'></span><span id='topic+col_gallus'></span><span id='topic+col_qual'></span><span id='topic+col_heat'></span><span id='topic+col_hot'></span><span id='topic+col_cold'></span><span id='topic+col_sari'></span><span id='topic+col_india'></span><span id='topic+col_bw'></span><span id='topic+col_grey'></span>

<h3>Description</h3>

<p>Colors, colors, colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>col_summer(n)

col_summer2(n)

col_spring(n)

col_autumn(n)

col_black(n)

col_solarized(n)

col_gallus(n)

col_qual(n)

col_heat(n)

col_hot(n)

col_cold(n)

col_sari(n)

col_india(n)

col_bw(n)

col_grey(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="color_palettes_+3A_n">n</code></td>
<td>
<p>the number of colors to generate from the color palette</p>
</td></tr>
</table>


<h3>Value</h3>

<p>colors (hexadecimal format)
</p>


<h3>Note</h3>

<p>Among available color palettes, <code>col_solarized</code> is based on Solarized: <a href="https://ethanschoonover.com/solarized/">https://ethanschoonover.com/solarized/</a>;
<code>col_div</code>, <code>col_qual</code>, <code>col_heat</code>, <code>col_cold</code>
and <code>col_gallus</code> are based on on ColorBrewer2: <a href="https://colorbrewer2.org/">https://colorbrewer2.org/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
wheel &lt;- function(palette, n=10){
 op &lt;- par(mar=rep(0, 4)) ; on.exit(par(op))
 pie(rep(1, n), col=palette(n), labels=NA, clockwise=TRUE)}

 # Qualitative
 wheel(col_qual)
 wheel(col_solarized)
 wheel(col_summer)
 wheel(col_summer2)
 wheel(col_spring)
 wheel(col_autumn)

 # Divergent
 wheel(col_gallus)
 wheel(col_india)

 # Sequential
 wheel(col_heat)
 wheel(col_hot)
 wheel(col_cold)
 wheel(col_sari)
 wheel(col_bw)
 wheel(col_grey)

 # Black only for pubs
 wheel(col_black)
</code></pre>

<hr>
<h2 id='combine'>Combine several objects</h2><span id='topic+combine'></span>

<h3>Description</h3>

<p>Combine <code>Coo</code> objects after a slicing, either manual or using <a href="#topic+slice">slice</a> or <a href="#topic+chop">chop</a>. Note that on Coo object,
it combines row-wise (ie, merges shapes as a <code>c</code> would do) ; but on Coe it combines column-wise
(merges coefficients). In the latter case, Coe must have the same number of shapes (not
necessarily the same number of coefficients).
Also the <code>$fac</code> of the first Coe is retrieved.
A separate version may come at some point.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_+3A_...">...</code></td>
<td>
<p>a list of Out(Coe), Opn(Coe), Ldk objects (but of the same class)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>Note</h3>

<p>Note that the order of shapes or their coefficients
is not checked, so anything with the same number of rows will be merged.
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>w &lt;- filter(bot, type=="whisky")
b &lt;- filter(bot, type=="beer")
combine(w, b)
# or, if you have many levels
bot_s &lt;- chop(bot, ~type)
bot_s$whisky
# note that you can apply something (single function or a more
# complex pipe) then combine everyone, since combine also works on lists
# eg:
# bot_s2 &lt;- efourier(bot_s, 10) # equivalent to lapply(bot_s, efourier, 10)
# bot_sf &lt;- combine(bot_s2)

# pipe style
efourier(bot_s, 10) %&gt;% combine()
</code></pre>

<hr>
<h2 id='complex'>Convert complex to/from cartesian coordinates</h2><span id='topic+complex'></span><span id='topic+cpx2coo'></span><span id='topic+coo2cpx'></span>

<h3>Description</h3>

<p>Convert complex to/from cartesian coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cpx2coo(Z)

coo2cpx(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="complex_+3A_z">Z</code></td>
<td>
<p>coordinates expressed in the complex form</p>
</td></tr>
<tr><td><code id="complex_+3A_coo">coo</code></td>
<td>
<p>coordinates expressed in the cartesian form</p>
</td></tr>
</table>


<h3>Value</h3>

<p>coordinates expressed in the cartesian/complex form
</p>


<h3>See Also</h3>

<p>Other bridges functions: 
<code><a href="#topic+as_df">as_df</a>()</code>,
<code><a href="#topic+bridges">bridges</a></code>,
<code><a href="#topic+export">export</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>shapes[4] %&gt;%            # from cartesian
    coo_sample(24) %&gt;%
    coo2cpx() %T&gt;%       # to complex
    cpx2coo()            # and back
</code></pre>

<hr>
<h2 id='Coo'>Coo &quot;super&quot; class</h2><span id='topic+Coo'></span>

<h3>Description</h3>

<p><code>Coo</code> class is the 'parent' or 'super' class of
<code><a href="#topic+Out">Out</a></code>, <code><a href="#topic+Opn">Opn</a></code> and <code><a href="#topic+Ldk">Ldk</a></code> classes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Coo(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Coo_+3A_...">...</code></td>
<td>
<p>anything and, anyway, this function will simply returns a message.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Useful shortcuts are described below. See <code>browseVignettes("Momocs")</code> for
a detail of the design behind Momocs' classes.
</p>
<p><code>Coo</code> class is the 'parent' class of the following 'child' classes </p>

<ul>
<li> <p><code><a href="#topic+Out">Out</a></code> for closed <b>out</b>lines
</p>
</li>
<li> <p><code><a href="#topic+Opn">Opn</a></code> for <b>op</b>e<b>n</b> outlines
</p>
</li>
<li> <p><code><a href="#topic+Ldk">Ldk</a></code> for configuration of <b>l</b>an<b>d</b>mar<b>k</b>s
</p>
</li></ul>

<p>Since all 'child classes' of them handle <code class="reqn">(x; y)</code> coordinates among other generic methods,
but also all have their specificity, this architecture allow to recycle generic methods and
to use specific methods.
</p>
<p>In other words, <code><a href="#topic+Out">Out</a></code>, <code><a href="#topic+Opn">Opn</a></code> and <code><a href="#topic+Ldk">Ldk</a></code> classes
are all, primarily, <code>Coo</code> objects on which we define generic <em>and</em>
specific methods. See their respective help pages for more help.
</p>
<p><code>Coo</code> objects all have the following components:
</p>

<ul>
<li> <p><code style="white-space: pre;">&#8288;$coo&#8288;</code> which is a list of matrices for coordinates
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;$fac&#8288;</code> a <code>data_frame</code> for covariates (if any). You can provide
this data_frame directly, as long as it has as many rows as there are matrices in <code style="white-space: pre;">&#8288;$coo&#8288;</code> (see examples),
or use an helper function such as <a href="#topic+lf_structure">lf_structure</a>.
</p>
</li></ul>

<p>You can access all the methods available for <code>Coo</code> objects
with <code>methods(class=Coo)</code>.
</p>


<h3>Value</h3>

<p>a list of class Coo
</p>


<h3>See Also</h3>

<p>Other classes: 
<code><a href="#topic+Coe">Coe</a>()</code>,
<code><a href="#topic+Ldk">Ldk</a>()</code>,
<code><a href="#topic+OpnCoe">OpnCoe</a>()</code>,
<code><a href="#topic+Opn">Opn</a>()</code>,
<code><a href="#topic+OutCoe">OutCoe</a>()</code>,
<code><a href="#topic+Out">Out</a>()</code>,
<code><a href="#topic+TraCoe">TraCoe</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># to see all methods for Coo objects.
methods(class='Coo')

# to see all methods for Out objects.
methods(class='Out') # same for Opn and Ldk

# Let's take an Out example. But all methods shown here
# work on Ldk (try on 'wings') and on Opn ('olea')
bot

# Primarily a 'Coo' object, but also an 'Out'
class(bot)
inherits(bot, "Coo")
panel(bot)
stack(bot)

# Getters (you can also use it to set data)
bot[1] %&gt;% coo_plot()
bot[1:5] %&gt;% str()

# Setters
bot[1] &lt;- shapes[4]
panel(bot)

bot[1:5] &lt;- shapes[4:8]
panel(bot)

# access the different components
# $coo coordinates
head(bot$coo)
# $fac grouping factors
head(bot$fac)
# or if you know the name of the column of interest
bot$type
# table
table(bot$fac)
# an internal view of an Out object
str(bot)

# subsetting
# see ?filter, ?select, and their 'see also' section for the
# complete list of dplyr-like verbs implemented in Momocs

length(bot) # the number of shapes
names(bot) # access all individual names
bot2 &lt;- bot
names(bot2) &lt;- paste0('newnames', 1:length(bot2)) # define new names

# Add a $fac from scratch
coo &lt;- bot[1:5] # a list of five matrices
length(coo)
sapply(coo, class)

fac &lt;- data.frame(name=letters[1:5], value=c(5:1))
# Then you have to define the subclass using the right builder
# here we have outlines, so we use Out
x &lt;- Out(coo, fac)
x$coo
x$fac
</code></pre>

<hr>
<h2 id='coo_align'>Aligns coordinates</h2><span id='topic+coo_align'></span>

<h3>Description</h3>

<p>Aligns the coordinates along their longer axis using var-cov matrix and eigen values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_align(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_align_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other aligning functions: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_plot(bot[1])
coo_plot(coo_align(bot[1]))

# on a Coo
b &lt;- bot %&gt;% slice(1:5) # for speed sake
stack(coo_align(b))
</code></pre>

<hr>
<h2 id='coo_aligncalliper'>Aligns shapes along their 'calliper length'</h2><span id='topic+coo_aligncalliper'></span>

<h3>Description</h3>

<p>And returns them registered on bookstein coordinates.
See <a href="#topic+coo_bookstein">coo_bookstein</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_aligncalliper(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_aligncalliper_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or any <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other aligning functions: 
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- bot[1]
coo_plot(b)
coo_plot(coo_aligncalliper(b))

b &lt;- bot %&gt;% slice(1:5) # for speed sake
bot.al &lt;- coo_aligncalliper(b)
stack(bot.al)
</code></pre>

<hr>
<h2 id='coo_alignminradius'>Aligns shapes using their shortest radius</h2><span id='topic+coo_alignminradius'></span>

<h3>Description</h3>

<p>And returns them slided with the first coordinate on the east.
May be used as an aligning strategy on shapes with a clear 'invaginate' part.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_alignminradius(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_alignminradius_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other aligning functions: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- bot %&gt;% slice(1:5) # for speed sake
stack(coo_alignminradius(b))
</code></pre>

<hr>
<h2 id='coo_alignxax'>Aligns shapes along the x-axis</h2><span id='topic+coo_alignxax'></span>

<h3>Description</h3>

<p>Align the longest axis of a shape along the x-axis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_alignxax(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_alignxax_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If some shapes are upside-down
(or mirror of each others), try redefining a new starting point (eg with coo_slidedirection) before
the alignment step. This may solve your problem because coo_calliper orders the <code>$arr.ind</code> used by
coo_aligncalliper.
</p>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or any <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other aligning functions: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- bot[1]
coo_plot(b)
coo_plot(coo_alignxax(b))
</code></pre>

<hr>
<h2 id='coo_angle_edges'>Calculates the angle of every edge of a shape</h2><span id='topic+coo_angle_edges'></span><span id='topic+coo_angle_edges.default'></span><span id='topic+coo_angle_edges.Coo'></span>

<h3>Description</h3>

<p>Returns the angle (in radians) of every edge of a shape,
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_angle_edges(coo, method = c("atan2", "acos")[1])

## Default S3 method:
coo_angle_edges(coo, method = c("atan2", "acos")[1])

## S3 method for class 'Coo'
coo_angle_edges(coo, method = c("atan2", "acos")[1])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_angle_edges_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> or a list of (x; y) coordinates or any <code>Coo</code></p>
</td></tr>
<tr><td><code id="coo_angle_edges_+3A_method">method</code></td>
<td>
<p>'atan2' (or 'acos') for a signed (or not) angle.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> the angles in radians for every edge.
</p>


<h3>Note</h3>

<p><code>coo_thetapts</code> is deprecated and will be removed
in future releases.
</p>


<h3>See Also</h3>

<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- coo_sample(bot[1], 64)
coo_angle_edges(b)
</code></pre>

<hr>
<h2 id='coo_angle_tangent'>Calculates the tangent angle along the perimeter of a
shape</h2><span id='topic+coo_angle_tangent'></span><span id='topic+coo_angle_tangent.default'></span><span id='topic+coo_angle_tangent.Coo'></span><span id='topic+coo_tangle'></span>

<h3>Description</h3>

<p>Calculated using complex numbers and returned in radians
minus the first one (modulo 2*pi).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_angle_tangent(coo)

## Default S3 method:
coo_angle_tangent(coo)

## S3 method for class 'Coo'
coo_angle_tangent(coo)

coo_tangle(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_angle_tangent_+3A_coo">coo</code></td>
<td>
<p>a matrix of coordinates or any <code>Coo</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code>, the tangent angle along the perimeter, or a
<code>list</code> of those for <code>Coo</code>
</p>


<h3>See Also</h3>

<p><a href="#topic+tfourier">tfourier</a>
</p>
<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- bot[1]
phi  &lt;- coo_angle_tangent(b)
phi2 &lt;- coo_angle_tangent(coo_smooth(b, 2))
plot(phi, type='l')
plot(phi2, type='l', col='red') # ta is very sensible to noise

# on Coo
bot %&gt;% coo_angle_tangent
</code></pre>

<hr>
<h2 id='coo_area'>Calculates the area of a shape</h2><span id='topic+coo_area'></span>

<h3>Description</h3>

<p>Calculates the area for a (non-crossing) shape.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_area(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_area_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code>, the area.
</p>


<h3>Note</h3>

<p>Using <code>area.poly</code> in gpc package is a good idea, but their licence
impedes Momocs to rely on it. but here is the function to do it, once gpc is loaded:
<code> area.poly(as(coo, 'gpc.poly')) </code>
</p>


<h3>See Also</h3>

<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_area(bot[1])
# for the distribution of the area of the bottles dataset
hist(sapply(bot$coo, coo_area), breaks=10)
</code></pre>

<hr>
<h2 id='coo_arrows'>Plots (lollipop) differences between two configurations</h2><span id='topic+coo_arrows'></span>

<h3>Description</h3>

<p>Draws 'arrows' between two configurations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_arrows(coo1, coo2, length = coo_centsize(coo1)/15, angle = 20, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_arrows_+3A_coo1">coo1</code></td>
<td>
<p>A <code>list</code> or a <code>matrix</code> of coordinates.</p>
</td></tr>
<tr><td><code id="coo_arrows_+3A_coo2">coo2</code></td>
<td>
<p>A <code>list</code> or a <code>matrix</code> of coordinates.</p>
</td></tr>
<tr><td><code id="coo_arrows_+3A_length">length</code></td>
<td>
<p>a length for the arrows.</p>
</td></tr>
<tr><td><code id="coo_arrows_+3A_angle">angle</code></td>
<td>
<p>an angle for the arrows</p>
</td></tr>
<tr><td><code id="coo_arrows_+3A_...">...</code></td>
<td>
<p>optional parameters to fed <a href="graphics.html#topic+arrows">arrows</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>See Also</h3>

<p>Other plotting functions: 
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_arrows(coo_sample(olea[3], 50), coo_sample(olea[6], 50))
title("Hi there !")
</code></pre>

<hr>
<h2 id='coo_baseline'>Register new baselines</h2><span id='topic+coo_baseline'></span>

<h3>Description</h3>

<p>A non-exact baseline registration on <code>t1</code> and <code>t2</code> coordinates,
for the <code>ldk1</code>-th and <code>ldk2</code>-th points.
By default it returns Bookstein's coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_baseline(coo, ldk1, ldk2, t1, t2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_baseline_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_baseline_+3A_ldk1">ldk1</code></td>
<td>
<p><code>numeric</code> the id of the first point of the new baseline</p>
</td></tr>
<tr><td><code id="coo_baseline_+3A_ldk2">ldk2</code></td>
<td>
<p><code>numeric</code> the id of the second point of the new baseline</p>
</td></tr>
<tr><td><code id="coo_baseline_+3A_t1">t1</code></td>
<td>
<p><code>numeric</code> the (x; y) coordinates of the 1st point of the new baseline</p>
</td></tr>
<tr><td><code id="coo_baseline_+3A_t2">t2</code></td>
<td>
<p><code>numeric</code> the (x; y) coordinates of the 2nd point of the new baseline</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other baselining functions: 
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>h &lt;- hearts %&gt;% slice(1:5) # for speed sake
stack(h)
stack(coo_baseline(h, 2, 4, c(-1, 0), c(1, 1)))
</code></pre>

<hr>
<h2 id='coo_bookstein'>Register Bookstein's coordinates</h2><span id='topic+coo_bookstein'></span>

<h3>Description</h3>

<p>Registers a new baseline for the shape, with the <code>ldk1</code>-th
and <code>ldk2</code>-th points being set on <code class="reqn">(x= -0.5; y=0)</code> and <code class="reqn">(x= 0.5; y=0)</code>, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_bookstein(coo, ldk1, ldk2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_bookstein_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_bookstein_+3A_ldk1">ldk1</code></td>
<td>
<p><code>numeric</code> the id of the first point of the new baseline (the first, by default)</p>
</td></tr>
<tr><td><code id="coo_bookstein_+3A_ldk2">ldk2</code></td>
<td>
<p><code>numeric</code> the id of the second point of the new baseline (the last, by default)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <a href="#topic+Out">Out</a>, it tries to do it using <code>$ldk</code> slot. Also the case for <a href="#topic+Opn">Opn</a>, but if
no landmark is defined, it will do it on the first and the last point of the shape.
</p>
<p>For <code>Out</code> and <code>Opn</code> defines the first landmark as the first point of the
new shapes with <a href="#topic+coo_slide">coo_slide</a>.
</p>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other baselining functions: 
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>h &lt;- hearts %&gt;% slice(1:5) # for the sake of speed
stack(h)
stack(coo_bookstein(h, 2, 4))
h &lt;- hearts[1]
coo_plot(h)
coo_plot(coo_bookstein(h, 20, 57), border='red')
</code></pre>

<hr>
<h2 id='coo_boundingbox'>Calculates coordinates of the bounding box</h2><span id='topic+coo_boundingbox'></span>

<h3>Description</h3>

<p>Calculates coordinates of the bounding box
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_boundingbox(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_boundingbox_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>data.frame</code> with coordinates of the bounding box
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>
<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot[1] %&gt;% coo_boundingbox()
bot %&gt;% coo_boundingbox()
</code></pre>

<hr>
<h2 id='coo_calliper'>Calculates the calliper length</h2><span id='topic+coo_calliper'></span>

<h3>Description</h3>

<p>Also called the Feret's diameter, the longest distance between two points of
the shape provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_calliper(coo, arr.ind = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_calliper_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or any <code>Coo</code></p>
</td></tr>
<tr><td><code id="coo_calliper_+3A_arr.ind">arr.ind</code></td>
<td>
<p><code>logical</code>, see below.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code>, the centroid size. If <code>arr.ind=TRUE</code>, a <code>data_frame</code>.
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- bot[1]
coo_calliper(b)
p &lt;- coo_calliper(b, arr.ind=TRUE)
p
p$length
ids &lt;- p$arr_ind[[1]]
coo_plot(b)
segments(b[ids[1], 1], b[ids[1], 2], b[ids[2], 1], b[ids[2], 2], lty=2)

# on a Coo
bot %&gt;%
coo_sample(32) %&gt;% # for speed sake
coo_calliper()

bot %&gt;%
coo_sample(32) %&gt;% # for speed sake
coo_calliper(arr.ind=TRUE)

</code></pre>

<hr>
<h2 id='coo_centdist'>Returns the distance between everypoints and the centroid</h2><span id='topic+coo_centdist'></span>

<h3>Description</h3>

<p>For every point of the shape, returns the (centroid-points) distance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_centdist(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_centdist_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates.
</p>


<h3>See Also</h3>

<p>Other centroid functions: 
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_centsize">coo_centsize</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- coo_sample(bot[1], 64)
d &lt;- coo_centdist(b)
barplot(d, xlab="Points along the outline", ylab="Distance to the centroid (pixels)")
</code></pre>

<hr>
<h2 id='coo_center'>Centers coordinates</h2><span id='topic+coo_center'></span><span id='topic+coo_centre'></span>

<h3>Description</h3>

<p>Returns a shape centered on the origin. The two functions are strictly equivalent.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_center(coo)

coo_centre(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_center_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_plot(bot[1])
# same as
coo_plot(coo_centre(bot[1]))
# this
coo_plot(coo_center(bot[1]))

# on Coo objects
b &lt;- slice(bot, 1:5) # speed sake
stack(slice(b, 1:5))
stack(coo_center(b))
</code></pre>

<hr>
<h2 id='coo_centpos'>Calculate centroid coordinates</h2><span id='topic+coo_centpos'></span>

<h3>Description</h3>

<p>Returns the (x; y) centroid coordinates of a shape.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_centpos(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_centpos_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>(x; y) coordinates of the centroid as a vector or a matrix.
</p>


<h3>See Also</h3>

<p>Other centroid functions: 
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_centsize">coo_centsize</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- bot[1]
coo_plot(b)
xy &lt;- coo_centpos(b)
points(xy[1], xy[2], cex=2, col='blue')
# on a Coo
coo_centpos(bot)
</code></pre>

<hr>
<h2 id='coo_centsize'>Calculates centroid size</h2><span id='topic+coo_centsize'></span>

<h3>Description</h3>

<p>Calculates centroid size
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_centsize(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_centsize_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to integrate size - if meaningful -
to Coo objects. See also <a href="#topic+coo_length">coo_length</a> and <a href="#topic+rescale">rescale</a>.
</p>


<h3>Value</h3>

<p><code>numeric</code>, the centroid size.
</p>


<h3>See Also</h3>

<p>Other centroid functions: 
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_centsize(bot[1])
# on a Coo
coo_centsize(bot)
# add it to $fac
mutate(bot, size=coo_centsize(bot))
</code></pre>

<hr>
<h2 id='coo_check'>Checks shapes</h2><span id='topic+coo_check'></span>

<h3>Description</h3>

<p>A simple utility, used internally, mostly in the coo functions and methods.
Returns a matrix of coordinates, when passed with either a list or a <code>matrix</code> of coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_check(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_check_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#coo_check('Not a shape')
#coo_check(iris)
#coo_check(matrix(1:10, ncol=2))
#coo_check(list(x=1:5, y=6:10))
</code></pre>

<hr>
<h2 id='coo_chull'>Calculates the (recursive) convex hull of a shape</h2><span id='topic+coo_chull'></span><span id='topic+coo_chull.default'></span><span id='topic+coo_chull.Coo'></span><span id='topic+coo_chull_onion'></span><span id='topic+coo_chull_onion.default'></span><span id='topic+coo_chull_onion.Coo'></span>

<h3>Description</h3>

<p><code>coo_chull</code> returns the ids of points that define the convex hull of a shape. A simple wrapper
around <a href="grDevices.html#topic+chull">chull</a>, mainly used in graphical functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_chull(coo)

## Default S3 method:
coo_chull(coo)

## S3 method for class 'Coo'
coo_chull(coo)

coo_chull_onion(coo, close = TRUE)

## Default S3 method:
coo_chull_onion(coo, close = TRUE)

## S3 method for class 'Coo'
coo_chull_onion(coo, close = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_chull_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or any <code>Coo</code>.</p>
</td></tr>
<tr><td><code id="coo_chull_+3A_close">close</code></td>
<td>
<p><code>logical</code> whether to close onion rings (<code>TRUE</code> by default)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>coo_chull_onion</code> recursively find their convex hull,
remove them, until less than 3 points are left.
</p>


<h3>Value</h3>

<p><code>coo_chull</code> returns a <code>matrix</code> of points defining
the convex hull of the shape; a <code>list</code> for <code>Coo</code>.
<code>coo_chull_onion</code> returns a <code>list</code> of successive onions rings,
and a <code>list</code> of <code>list</code>s for <code>Coo</code>.
</p>


<h3>See Also</h3>

<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># coo_chull
h &lt;- coo_sample(hearts[4], 32)
coo_plot(h)
ch &lt;- coo_chull(h)
lines(ch, col='red', lty=2)

bot %&gt;% coo_chull

coo_chull_onion
x &lt;- bot %&gt;% efourier(6) %&gt;% PCA
all_whisky_points &lt;- x %&gt;% as_df() %&gt;% filter(type=="whisky") %&gt;% select(PC1, PC2)
plot(x, ~type, eig=FALSE)
peeling_the_whisky_onion &lt;- all_whisky_points %&gt;% as.matrix %&gt;% coo_chull_onion()
# you may need to par(xpd=NA) to ensure all segments
# even those outside the graphical window are drawn
peeling_the_whisky_onion$coo %&gt;% lapply(coo_draw)
# simulated data
xy &lt;- replicate(2, rnorm(50))
coo_plot(xy, poly=FALSE)
xy %&gt;% coo_chull_onion() %$% coo %&gt;%
lapply(polygon, col="#00000022")
</code></pre>

<hr>
<h2 id='coo_circularity'>Calculates the Haralick's circularity of a shape</h2><span id='topic+coo_circularity'></span><span id='topic+coo_circularity.default'></span><span id='topic+coo_circularity.Coo'></span><span id='topic+coo_circularityharalick'></span><span id='topic+coo_circularityharalick.default'></span><span id='topic+coo_circularityharalick.Coo'></span><span id='topic+coo_circularitynorm'></span><span id='topic+coo_circularitynorm.default'></span><span id='topic+coo_circularitynorm.Coo'></span>

<h3>Description</h3>

<p><code>coo_circularity</code> calculates the 'circularity measure'. Also called 'compactness'
and 'shape factor' sometimes. <code>coo_circularityharalick</code> calculates Haralick's circularity which is less sensible
to digitalization noise than <code>coo_circularity</code>.
<code>coo_circularitynorm</code> calculates 'circularity', also called compactness
and shape factor, but normalized to the unit circle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_circularity(coo)

## Default S3 method:
coo_circularity(coo)

## S3 method for class 'Coo'
coo_circularity(coo)

coo_circularityharalick(coo)

## Default S3 method:
coo_circularityharalick(coo)

## S3 method for class 'Coo'
coo_circularityharalick(coo)

coo_circularitynorm(coo)

## Default S3 method:
coo_circularitynorm(coo)

## S3 method for class 'Coo'
coo_circularitynorm(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_circularity_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or any <code>Coo</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> for single shapes, <code>list</code> for <code>Coo</code> of
the corresponding circularity measurement.
</p>


<h3>Source</h3>

<p>Rosin PL. 2005. Computing global shape measures.
Handbook of Pattern Recognition and Computer Vision. 177-196.
</p>


<h3>See Also</h3>

<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# coo_circularity
bot[1] %&gt;% coo_circularity()
bot %&gt;%
    slice(1:5) %&gt;% # for speed sake only
    coo_circularity

# coo_circularityharalick
bot[1] %&gt;% coo_circularityharalick()
bot %&gt;%
    slice(1:5) %&gt;% # for speed sake only
    coo_circularityharalick

# coo_circularitynorm
bot[1] %&gt;% coo_circularitynorm()
bot %&gt;%
    slice(1:5) %&gt;% # for speed sake only
    coo_circularitynorm
</code></pre>

<hr>
<h2 id='coo_close'>Closes/uncloses shapes</h2><span id='topic+coo_close'></span><span id='topic+coo_unclose'></span>

<h3>Description</h3>

<p>Returns a closed shape from (un)closed shapes. See also <a href="#topic+coo_unclose">coo_unclose</a>.
</p>
<p>Returns a unclosed shape from (un)closed shapes. See also <a href="#topic+coo_close">coo_close</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_close(coo)

coo_unclose(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_close_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>
<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- (matrix(1:10, ncol=2))
x2 &lt;- coo_close(x)
x3 &lt;- coo_unclose(x2)
x
coo_is_closed(x)
x2
coo_is_closed(x2)
x3
coo_is_closed(x3)
x &lt;- (matrix(1:10, ncol=2))
x2 &lt;- coo_close(x)
x3 &lt;- coo_unclose(x2)
x
coo_is_closed(x)
x2
coo_is_closed(x2)
x3
coo_is_closed(x3)
</code></pre>

<hr>
<h2 id='coo_convexity'>Calculates the convexity of a shape</h2><span id='topic+coo_convexity'></span>

<h3>Description</h3>

<p>Calculated using a ratio of the eigen values (inertia axis)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_convexity(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_convexity_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> for a single shape, <code>list</code> for a <code>Coo</code>
</p>


<h3>Source</h3>

<p>Rosin PL. 2005. Computing global shape measures.
Handbook of Pattern Recognition and Computer Vision. 177-196.
</p>


<h3>See Also</h3>

<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_convexity(bot[1])
bot %&gt;%
    slice(1:3) %&gt;% # for speed sake only
    coo_convexity()
</code></pre>

<hr>
<h2 id='coo_down'>coo_down
Retains coordinates with negative y-coordinates</h2><span id='topic+coo_down'></span>

<h3>Description</h3>

<p>Useful when shapes are aligned along the x-axis (e.g. because of a
bilateral symmetry) and when one wants to retain just the lower side.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_down(coo, slidegap = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_down_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_down_+3A_slidegap">slidegap</code></td>
<td>
<p><code>logical</code> whether to apply <a href="#topic+coo_slidegap">coo_slidegap</a> after coo_down</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates or a <a href="#topic+Coo">Coo</a> object (<a href="#topic+Out">Out</a> are returned as <a href="#topic+Opn">Opn</a>)
</p>


<h3>Note</h3>

<p>When shapes are &quot;sliced&quot; along the x-axis, it usually results on open curves and thus to huge/artefactual
gaps between points neighboring this axis. This is usually solved with <a href="#topic+coo_slidegap">coo_slidegap</a>. See examples there.
</p>
<p>Also, when apply a coo_left/right/up/down on an <a href="#topic+Out">Out</a> object, you then obtain an <a href="#topic+Opn">Opn</a> object, which is done
automatically.
</p>


<h3>See Also</h3>

<p>Other opening functions: 
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- coo_alignxax(bot[1])
coo_plot(b)
coo_draw(coo_down(b), border='red')
</code></pre>

<hr>
<h2 id='coo_draw'>Adds a shape to the current plot</h2><span id='topic+coo_draw'></span>

<h3>Description</h3>

<p><code>coo_draw</code> is simply a <a href="#topic+coo_plot">coo_plot</a> with <code>plot.new=FALSE</code>, ie
that adds a shape on the active plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_draw(coo, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_draw_+3A_coo">coo</code></td>
<td>
<p>a <code>list</code> or a <code>matrix</code> of coordinates.</p>
</td></tr>
<tr><td><code id="coo_draw_+3A_...">...</code></td>
<td>
<p>optional parameters for <a href="#topic+coo_plot">coo_plot</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing on the last plot
</p>


<h3>See Also</h3>

<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b1 &lt;- bot[4]
b2 &lt;- bot[5]
coo_plot(b1)
coo_draw(b2, border='red') # all coo_plot arguments will work for coo_draw
</code></pre>

<hr>
<h2 id='coo_draw_rads'>Draw radii to the current plot</h2><span id='topic+coo_draw_rads'></span>

<h3>Description</h3>

<p>Given a shape, all centroid-points radii are drawn using <a href="graphics.html#topic+segments">segments</a>
that can be passed with options
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_draw_rads(coo, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_draw_rads_+3A_coo">coo</code></td>
<td>
<p>a shape</p>
</td></tr>
<tr><td><code id="coo_draw_rads_+3A_...">...</code></td>
<td>
<p>arguments to feed <a href="graphics.html#topic+segments">segments</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing on the last plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>shp &lt;- shapes[4] %&gt;% coo_sample(24) %T&gt;% coo_plot
coo_draw_rads(shp, col=col_summer(24))

</code></pre>

<hr>
<h2 id='coo_dxy'>Calculate abscissa and ordinate on a shape</h2><span id='topic+coo_dxy'></span>

<h3>Description</h3>

<p>A simple wrapper to calculate dxi - dx1 and dyi - dx1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_dxy(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_dxy_+3A_coo">coo</code></td>
<td>
<p>a matrix (or a list) of (x; y) coordinates or any <code>Coo</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>data.frame</code> with two components <code>dx</code> and <code>dy</code> for single shapes
or a <code>list</code> of such <code>data.frame</code>s for <code>Coo</code>
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_dxy(coo_sample(bot[1], 12))

bot %&gt;%
    slice(1:5) %&gt;% coo_sample(12) %&gt;%  # for readability and speed only
    coo_dxy()
</code></pre>

<hr>
<h2 id='coo_eccentricity'>Calculates the eccentricity of a shape</h2><span id='topic+coo_eccentricity'></span><span id='topic+coo_eccentricityeigen'></span><span id='topic+coo_eccentricityeigen.default'></span><span id='topic+coo_eccentricityeigen.Coo'></span><span id='topic+coo_eccentricityboundingbox'></span><span id='topic+coo_eccentricityboundingbox.default'></span><span id='topic+coo_eccentricityboundingbox.Coo'></span>

<h3>Description</h3>

<p><code>coo_eccentricityeigen</code> uses the ratio of
the eigen values (inertia axes of coordinates).
<code>coo_eccentricityboundingbox</code> uses the width/length ratio (see <a href="#topic+coo_lw">coo_lw</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_eccentricityeigen(coo)

## Default S3 method:
coo_eccentricityeigen(coo)

## S3 method for class 'Coo'
coo_eccentricityeigen(coo)

coo_eccentricityboundingbox(coo)

## Default S3 method:
coo_eccentricityboundingbox(coo)

## S3 method for class 'Coo'
coo_eccentricityboundingbox(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_eccentricity_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or any <code>Coo</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> for single shapes, <code>list</code> for <code>Coo</code>.
</p>


<h3>Source</h3>

<p>Rosin PL. 2005. Computing global shape measures.
Handbook of Pattern Recognition and Computer Vision. 177-196.
</p>


<h3>See Also</h3>

<p><a href="#topic+coo_eccentricityboundingbox">coo_eccentricityboundingbox</a>
</p>
<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># coo_eccentricityeigen
bot[1] %&gt;% coo_eccentricityeigen()
bot %&gt;%
    slice(1:3) %&gt;% # for speed sake only
    coo_eccentricityeigen()

# coo_eccentricityboundingbox
bot[1] %&gt;% coo_eccentricityboundingbox()
bot %&gt;%
    slice(1:3) %&gt;% # for speed sake only
    coo_eccentricityboundingbox()
</code></pre>

<hr>
<h2 id='coo_elongation'>Calculates the elongation of a shape</h2><span id='topic+coo_elongation'></span>

<h3>Description</h3>

<p>Calculates the elongation of a shape
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_elongation(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_elongation_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric, the eccentricity of the bounding box
</p>


<h3>Source</h3>

<p>Rosin PL. 2005. Computing global shape measures.
Handbook of Pattern Recognition and Computer Vision. 177-196.
</p>


<h3>See Also</h3>

<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_elongation(bot[1])
# on Coo
# for speed sake
bot %&gt;% slice(1:3) %&gt;% coo_elongation
</code></pre>

<hr>
<h2 id='coo_extract'>Extract coordinates from a shape</h2><span id='topic+coo_extract'></span>

<h3>Description</h3>

<p>Extract ids coordinates from a single shape or a Coo object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_extract(coo, ids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_extract_+3A_coo">coo</code></td>
<td>
<p>either a <code>matrix</code> of (x; y) coordinates or a <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_extract_+3A_ids">ids</code></td>
<td>
<p><code>integer</code>, the ids of points to sample.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It probably only make sense for Coo objects with the same number of coordinates
and them being homologous, typically on Ldk.
</p>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other sampling functions: 
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_extract(bot[1], c(3, 9, 12)) # or :
bot[1] %&gt;% coo_extract(c(3, 9, 12))
</code></pre>

<hr>
<h2 id='coo_flipx'>Flips shapes</h2><span id='topic+coo_flipx'></span><span id='topic+coo_flipy'></span>

<h3>Description</h3>

<p><code>coo_flipx</code> flips shapes about the x-axis; <code>coo_flipy</code> about the y-axis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_flipx(coo)

coo_flipy(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_flipx_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates
</p>


<h3>See Also</h3>

<p>Other transforming functions: 
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cat &lt;- shapes[4]
cat &lt;- coo_center(cat)
coo_plot(cat)
coo_draw(coo_flipx(cat), border="red")
coo_draw(coo_flipy(cat), border="blue")

#' # to flip an entire Coo:
shapes2 &lt;- shapes
shapes$coo &lt;- lapply(shapes2$coo, coo_flipx)
</code></pre>

<hr>
<h2 id='coo_force2close'>Forces shapes to close</h2><span id='topic+coo_force2close'></span>

<h3>Description</h3>

<p>An exotic function that distribute the distance between the first and the last points
of unclosed shapes, so that they become closed. May be useful (?) e.g. for t/rfourier methods
where reconstructed shapes may not be closed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_force2close(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_force2close_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- coo_sample(bot[1], 64)
b &lt;- b[1:40,]
coo_plot(b)
coo_draw(coo_force2close(b), border='red')
</code></pre>

<hr>
<h2 id='coo_interpolate'>Interpolates coordinates</h2><span id='topic+coo_interpolate'></span>

<h3>Description</h3>

<p>Interpolates n coordinates 'among existing points'between' existing points,
along the perimeter of the coordinates provided and keeping the first point
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_interpolate(coo, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_interpolate_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_interpolate_+3A_n">n</code></td>
<td>
<p><code>integer</code>, the number fo points to interpolate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other sampling functions: 
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b5 &lt;- bot %&gt;% slice(1:5) # for speed sake
stack(b5)
stack(coo_scale(b5))
stack(b5)
stack(coo_interpolate(coo_sample(b5, 12), 120))
coo_plot(bot[1])
coo_plot(coo_interpolate(coo_sample(bot[1], 12), 120))
</code></pre>

<hr>
<h2 id='coo_intersect_angle'>Nearest intersection between a shape and a segment specified with an angle</h2><span id='topic+coo_intersect_angle'></span><span id='topic+coo_intersect_direction'></span><span id='topic+coo_intersect_direction.default'></span><span id='topic+coo_intersect_direction.Coo'></span>

<h3>Description</h3>

<p>Take a shape, and segment starting on the centroid and having a particular angle, which point is the nearest
where the segment intersects with the shape?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_intersect_angle(coo, angle = 0)

coo_intersect_direction(coo, direction = c("down", "left", "up", "right")[4])

## Default S3 method:
coo_intersect_direction(coo, direction = c("down", "left", "up", "right")[4])

## S3 method for class 'Coo'
coo_intersect_direction(coo, direction = c("down", "left", "up", "right")[4])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_intersect_angle_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_intersect_angle_+3A_angle">angle</code></td>
<td>
<p><code>numeric</code> an angle in radians (0 by default).</p>
</td></tr>
<tr><td><code id="coo_intersect_angle_+3A_direction">direction</code></td>
<td>
<p><code>character</code> one of <code>"down", "left", "up", "right"</code> (&quot;right&quot; by default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> the id of the nearest point or a <code>list</code> for <code>Coo</code> See examples.
</p>


<h3>Note</h3>

<p>shapes are always centered before this operation. If you need a simple
direction such as <code>(down, left, up, right)ward</code>, then use <a href="#topic+coo_intersect_direction">coo_intersect_direction</a> which
does not need to find an intersection but relies on coordinates and is about 1000.
</p>


<h3>See Also</h3>

<p>Other coo_ intersect: 
<code><a href="#topic+coo_intersect_segment">coo_intersect_segment</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo &lt;- bot[1] %&gt;% coo_center %&gt;% coo_scale
coo_plot(coo)
coo %&gt;% coo_intersect_angle(pi/7) %&gt;%
   coo[., , drop=FALSE] %&gt;% points(col="red")

 # many angles
 coo_plot(coo)
 sapply(seq(0, pi, pi/12),
       function(x) coo %&gt;% coo_intersect_angle(x)) -&gt; ids
 coo[ids, ] %&gt;% points(col="blue")

 coo %&gt;%
 coo_intersect_direction("down") %&gt;%
 coo[.,, drop=FALSE] %&gt;% points(col="orange")

</code></pre>

<hr>
<h2 id='coo_intersect_segment'>Nearest intersection between a shape and a segment</h2><span id='topic+coo_intersect_segment'></span>

<h3>Description</h3>

<p>Take a shape, and an intersecting segment, which point is the nearest
of where the segment intersects with the shape? Most of the time,
centering before makes more sense.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_intersect_segment(coo, seg, center = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_intersect_segment_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_intersect_segment_+3A_seg">seg</code></td>
<td>
<p>a 2x2 <code>matrix</code> defining the starting and ending points;
or a list or a numeric of length 4.</p>
</td></tr>
<tr><td><code id="coo_intersect_segment_+3A_center">center</code></td>
<td>
<p><code>logical</code> whether to center the shape (TRUE by default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> the id of the nearest point, a <code>list</code> for <code>Coo</code>. See examples.
</p>


<h3>See Also</h3>

<p>Other coo_ intersect: 
<code><a href="#topic+coo_intersect_angle">coo_intersect_angle</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo &lt;- bot[1] %&gt;% coo_center %&gt;% coo_scale
seg &lt;- c(0, 0, 2, 2) # passed as a numeric of length(4)
coo_plot(coo)
segments(seg[1], seg[2], seg[3], seg[4])
coo %&gt;% coo_intersect_segment(seg) %T&gt;% print %&gt;%
# prints on the console and draw it
   coo[., , drop=FALSE] %&gt;% points(col="red")

# on Coo
bot %&gt;%
    slice(1:3) %&gt;% # for the sake of speed
    coo_center %&gt;%
    coo_intersect_segment(matrix(c(0, 0, 1000, 1000), ncol=2, byrow=TRUE))
</code></pre>

<hr>
<h2 id='coo_is_closed'>Test if shapes are closed</h2><span id='topic+coo_is_closed'></span><span id='topic+is_open'></span>

<h3>Description</h3>

<p>Returns TRUE/FALSE whether the last coordinate of the shapes is the same
as the first one.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_is_closed(coo)

is_open(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_is_closed_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a single or a vector of <code>logical</code>.
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_is_closed(matrix(1:10, ncol=2))
coo_is_closed(coo_close(matrix(1:10, ncol=2)))
coo_is_closed(bot)
coo_is_closed(coo_close(bot))
</code></pre>

<hr>
<h2 id='coo_jitter'>Jitters shapes</h2><span id='topic+coo_jitter'></span>

<h3>Description</h3>

<p>A simple wrapper around <a href="base.html#topic+jitter">jitter</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_jitter(coo, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_jitter_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_jitter_+3A_...">...</code></td>
<td>
<p>additional parameter for <a href="base.html#topic+jitter">jitter</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates or a Coo object
</p>


<h3>See Also</h3>

<p><a href="#topic+get_pairs">get_pairs</a>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;-bot[1]
coo_plot(b, zoom=0.2)
coo_draw(coo_jitter(b, amount=3), border="red")

# for a Coo example, see \link{get_pairs}
</code></pre>

<hr>
<h2 id='coo_ldk'>Defines landmarks interactively</h2><span id='topic+coo_ldk'></span>

<h3>Description</h3>

<p>Allows to interactively define a <code>nb.ldk</code> number of landarks on a shape.
Used in other facilities to acquire/manipulate data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_ldk(coo, nb.ldk, close = FALSE, points = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_ldk_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> or a list of (x; y) coordinates.</p>
</td></tr>
<tr><td><code id="coo_ldk_+3A_nb.ldk">nb.ldk</code></td>
<td>
<p><code>integer</code>, the number of landmarks to define</p>
</td></tr>
<tr><td><code id="coo_ldk_+3A_close">close</code></td>
<td>
<p><code>logical</code> whether to close (typically for outlines)</p>
</td></tr>
<tr><td><code id="coo_ldk_+3A_points">points</code></td>
<td>
<p><code>logical</code> whether to display points</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> that corresponds to the closest ids,
on the shape, from cliked points.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
b &lt;- bot[1]
coo_ldk(b, 3) # run this, and click 3 times
coo_ldk(bot, 2) # this also works on Out

## End(Not run)
</code></pre>

<hr>
<h2 id='coo_left'>Retains coordinates with negative x-coordinates</h2><span id='topic+coo_left'></span>

<h3>Description</h3>

<p>Useful when shapes are aligned along the y-axis (e.g. because of a
bilateral symmetry) and when one wants to retain just the lower side.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_left(coo, slidegap = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_left_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_left_+3A_slidegap">slidegap</code></td>
<td>
<p><code>logical</code> whether to apply <a href="#topic+coo_slidegap">coo_slidegap</a> after coo_left</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates or a <a href="#topic+Coo">Coo</a> object (<a href="#topic+Out">Out</a> are returned as <a href="#topic+Opn">Opn</a>)
</p>


<h3>Note</h3>

<p>When shapes are &quot;sliced&quot; along the y-axis, it usually results on open curves and thus to huge/artefactual
gaps between points neighboring this axis. This is usually solved with <a href="#topic+coo_slidegap">coo_slidegap</a>. See examples there.
</p>
<p>Also, when apply a coo_left/right/up/down on an <a href="#topic+Out">Out</a> object, you then obtain an <a href="#topic+Opn">Opn</a> object, which is done
automatically.
</p>


<h3>See Also</h3>

<p>Other opening functions: 
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- coo_center(bot[1])
coo_plot(b)
coo_draw(coo_left(b), border='red')
</code></pre>

<hr>
<h2 id='coo_length'>Calculates the length of a shape</h2><span id='topic+coo_length'></span>

<h3>Description</h3>

<p>Nothing more than <code>coo_lw(coo)[1]</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_length(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_length_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or a Coo object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to integrate size - if meaningful -
to Coo objects. See also <a href="#topic+coo_centsize">coo_centsize</a> and <a href="#topic+rescale">rescale</a>.
</p>


<h3>Value</h3>

<p>the length (in pixels) of the shape
</p>


<h3>See Also</h3>

<p><a href="#topic+coo_lw">coo_lw</a>, <a href="#topic+coo_width">coo_width</a>
</p>
<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_length(bot[1])
coo_length(bot)
mutate(bot, size=coo_length(bot))
</code></pre>

<hr>
<h2 id='coo_likely_clockwise'>Tests if shapes are (likely) developping clockwise or anticlockwise</h2><span id='topic+coo_likely_clockwise'></span><span id='topic+coo_likely_clockwise.default'></span><span id='topic+coo_likely_clockwise.Coo'></span><span id='topic+coo_likely_anticlockwise'></span>

<h3>Description</h3>

<p>Tests if shapes are (likely) developping clockwise or anticlockwise
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_likely_clockwise(coo)

## Default S3 method:
coo_likely_clockwise(coo)

## S3 method for class 'Coo'
coo_likely_clockwise(coo)

coo_likely_anticlockwise(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_likely_clockwise_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a single or a vector of <code>logical</code>.
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>shapes[4] %&gt;% coo_sample(64) %&gt;% coo_plot()  #clockwise cat
shapes[4] %&gt;% coo_likely_clockwise()
shapes[4] %&gt;% coo_rev() %&gt;% coo_likely_clockwise()

# on Coo
shapes %&gt;% coo_likely_clockwise %&gt;% `[`(4)
</code></pre>

<hr>
<h2 id='coo_listpanel'>Plots sets of shapes.</h2><span id='topic+coo_listpanel'></span>

<h3>Description</h3>

<p><code>coo_listpanel</code> plots a list of shapes if passed with a list of
coordinates. Mainly used by <a href="#topic+panel.Coo">panel.Coo</a> functions. If used outside the latter,
shapes must be &quot;templated&quot;, see <a href="#topic+coo_template">coo_template</a>. If you want to reorder shapes
according to a factor, use <a href="#topic+arrange">arrange</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_listpanel(
  coo.list,
  dim,
  byrow = TRUE,
  fromtop = TRUE,
  cols,
  borders,
  poly = TRUE,
  points = FALSE,
  points.pch = 3,
  points.cex = 0.2,
  points.col = "#333333",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_listpanel_+3A_coo.list">coo.list</code></td>
<td>
<p>A <code>list</code> of coordinates</p>
</td></tr>
<tr><td><code id="coo_listpanel_+3A_dim">dim</code></td>
<td>
<p>A <code>vector</code> of the form <code>(nb.row, nb.cols)</code> to specify
the panel display. If missing, shapes are arranged in a square.</p>
</td></tr>
<tr><td><code id="coo_listpanel_+3A_byrow">byrow</code></td>
<td>
<p><code>logical</code>. Whether to draw successive shape by row or by col.</p>
</td></tr>
<tr><td><code id="coo_listpanel_+3A_fromtop">fromtop</code></td>
<td>
<p><code>logical</code>. Whether to display shapes from the top of the
plotting region.</p>
</td></tr>
<tr><td><code id="coo_listpanel_+3A_cols">cols</code></td>
<td>
<p>A <code>vector</code> of colors to fill shapes.</p>
</td></tr>
<tr><td><code id="coo_listpanel_+3A_borders">borders</code></td>
<td>
<p>A <code>vector</code> of colors to draw shape borders.</p>
</td></tr>
<tr><td><code id="coo_listpanel_+3A_poly">poly</code></td>
<td>
<p>logical whether to use polygon or lines to draw shapes.
mainly for use for outlines and open outlines.</p>
</td></tr>
<tr><td><code id="coo_listpanel_+3A_points">points</code></td>
<td>
<p>logical if poly is set to FALSE whether to add points</p>
</td></tr>
<tr><td><code id="coo_listpanel_+3A_points.pch">points.pch</code></td>
<td>
<p>if points is TRUE, a pch for these points</p>
</td></tr>
<tr><td><code id="coo_listpanel_+3A_points.cex">points.cex</code></td>
<td>
<p>if points is TRUE, a cex for these points</p>
</td></tr>
<tr><td><code id="coo_listpanel_+3A_points.col">points.col</code></td>
<td>
<p>if points is TRUE, a col  for these points</p>
</td></tr>
<tr><td><code id="coo_listpanel_+3A_...">...</code></td>
<td>
<p>additional arguments to feed generic <code>plot</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns (invisibly) a <code>data.frame</code> with position of shapes that
can be used for other sophisticated plotting design.
</p>


<h3>See Also</h3>

<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_listpanel(bot$coo) # equivalent to panel(bot)
</code></pre>

<hr>
<h2 id='coo_lolli'>Plots (lollipop) differences between two configurations</h2><span id='topic+coo_lolli'></span>

<h3>Description</h3>

<p>Draws 'lollipops' between two configurations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_lolli(coo1, coo2, pch = NA, cex = 0.5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_lolli_+3A_coo1">coo1</code></td>
<td>
<p>A <code>list</code> or a <code>matrix</code> of coordinates.</p>
</td></tr>
<tr><td><code id="coo_lolli_+3A_coo2">coo2</code></td>
<td>
<p>A <code>list</code> or a <code>matrix</code> of coordinates.</p>
</td></tr>
<tr><td><code id="coo_lolli_+3A_pch">pch</code></td>
<td>
<p>a pch for the points (default to NA)</p>
</td></tr>
<tr><td><code id="coo_lolli_+3A_cex">cex</code></td>
<td>
<p>a cex for the points</p>
</td></tr>
<tr><td><code id="coo_lolli_+3A_...">...</code></td>
<td>
<p>optional parameters to fed <a href="graphics.html#topic+points">points</a> and <a href="graphics.html#topic+segments">segments</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing on the last plot
</p>


<h3>See Also</h3>

<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_lolli(coo_sample(olea[3], 50), coo_sample(olea[6], 50))
title("A nice title !")
</code></pre>

<hr>
<h2 id='coo_lw'>Calculates length and width of a shape</h2><span id='topic+coo_lw'></span>

<h3>Description</h3>

<p>Returns the length and width of a shape based on their iniertia axis
i.e. alignment to the x-axis. The length is defined as
the range along the x-axis; the width as the range on the y-axis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_lw(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_lw_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or Coo object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of two <code>numeric</code>: the length and the width.
</p>


<h3>See Also</h3>

<p><a href="#topic+coo_length">coo_length</a>, <a href="#topic+coo_width">coo_width</a>.
</p>
<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_lw(bot[1])
</code></pre>

<hr>
<h2 id='coo_nb'>Counts coordinates</h2><span id='topic+coo_nb'></span>

<h3>Description</h3>

<p>Returns the number of coordinates, for a single shape or a Coo object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_nb(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_nb_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>either a single numeric or a vector of numeric
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># single shape
coo_nb(bot[1])
# Coo object
coo_nb(bot)
</code></pre>

<hr>
<h2 id='coo_oscillo'>Momocs' 'oscilloscope' for Fourier-based approaches</h2><span id='topic+coo_oscillo'></span>

<h3>Description</h3>

<p>Shape analysis deals with curve fitting, whether <code class="reqn">x(t)</code> and <code class="reqn">y(t)</code>
positions along the curvilinear abscissa and/or radius/tangent angle variation.
These functions are mainly intended for (self-)teaching of Fourier-based methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_oscillo(
  coo,
  method = c("efourier", "rfourier", "tfourier", "all")[4],
  shape = TRUE,
  nb.pts = 12
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_oscillo_+3A_coo">coo</code></td>
<td>
<p>A list or a matrix of coordinates.</p>
</td></tr>
<tr><td><code id="coo_oscillo_+3A_method">method</code></td>
<td>
<p>character among <code>c('efourier', 'rfourier', 'tfourier', 'all')</code>.
<code>'all'</code> by default</p>
</td></tr>
<tr><td><code id="coo_oscillo_+3A_shape">shape</code></td>
<td>
<p><code>logical</code> whether to plot the original shape</p>
</td></tr>
<tr><td><code id="coo_oscillo_+3A_nb.pts">nb.pts</code></td>
<td>
<p><code>integer</code>. The number or reference points, sampled
equidistantly along the curvilinear abscissa and added on the oscillo
curves.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the plotted values
</p>


<h3>See Also</h3>

<p>exemplifying functions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_oscillo(shapes[4])
coo_oscillo(shapes[4], 'efourier')
coo_oscillo(shapes[4], 'rfourier')
coo_oscillo(shapes[4], 'tfourier')
#tfourier is prone to high-frequency noise but smoothing can help
coo_oscillo(coo_smooth(shapes[4], 10), 'tfourier')
</code></pre>

<hr>
<h2 id='coo_perim'>Calculates perimeter and variations</h2><span id='topic+coo_perim'></span><span id='topic+coo_perimpts'></span><span id='topic+coo_perimpts.default'></span><span id='topic+coo_perimpts.Coo'></span><span id='topic+coo_perimcum'></span><span id='topic+coo_perimcum.default'></span><span id='topic+coo_perimcum.Coo'></span><span id='topic+coo_perim.default'></span><span id='topic+coo_perim.Coo'></span>

<h3>Description</h3>

<p><code>coo_perim</code> calculates the perimeter;
<code>coo_perimpts</code> calculates the euclidean distance between every points of a shape;
<code>coo_perimcum</code> does the same and calculates and cumulative sum.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_perimpts(coo)

## Default S3 method:
coo_perimpts(coo)

## S3 method for class 'Coo'
coo_perimpts(coo)

coo_perimcum(coo)

## Default S3 method:
coo_perimcum(coo)

## S3 method for class 'Coo'
coo_perimcum(coo)

coo_perim(coo)

## Default S3 method:
coo_perim(coo)

## S3 method for class 'Coo'
coo_perim(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_perim_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of (x; y) coordinates or any <code>Coo</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> the distance between every point or
a <code>list</code> of those.
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># for speed sake
b1 &lt;- coo_sample(bot[1], 12)
b5 &lt;- bot %&gt;% slice(1:5) %&gt;% coo_sample(12)

# coo_perim
coo_perim(b1)
coo_perim(b5)

# coo_perimpts
coo_perimpts(b1)
b5 %&gt;% coo_perimpts()

# coo_perimcum
b1 %&gt;% coo_perimcum()
b5 %&gt;% coo_perimcum()
</code></pre>

<hr>
<h2 id='coo_plot'>Plots a single shape</h2><span id='topic+coo_plot'></span><span id='topic+ldk_plot'></span>

<h3>Description</h3>

<p>A simple wrapper around <a href="graphics.html#topic+plot">plot</a> for plotting shapes. Widely used in Momocs
in other graphical functions, in methods, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_plot(
  coo,
  xlim,
  ylim,
  border = "#333333",
  col = NA,
  lwd = 1,
  lty = 1,
  points = FALSE,
  first.point = TRUE,
  cex.first.point = 0.5,
  centroid = TRUE,
  xy.axis = TRUE,
  pch = 1,
  cex = 0.5,
  main = NA,
  poly = TRUE,
  plot.new = TRUE,
  plot = TRUE,
  zoom = 1,
  ...
)

ldk_plot(coo, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_plot_+3A_coo">coo</code></td>
<td>
<p>A <code>list</code> or a <code>matrix</code> of coordinates.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_xlim">xlim</code></td>
<td>
<p>If <code>coo_plot</code> is called and <code>coo</code> is missing, then a
vector of length 2 specifying the <code>ylim</code> of the ploting area.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_ylim">ylim</code></td>
<td>
<p>If <code>coo_plot</code> is called and <code>coo</code> is missing, then a
vector of length 2 specifying the <code>ylim</code> of the ploting area.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_border">border</code></td>
<td>
<p>A color for the shape border.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_col">col</code></td>
<td>
<p>A color to fill the shape polygon.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_lwd">lwd</code></td>
<td>
<p>The <code>lwd</code> for drawing shapes.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_lty">lty</code></td>
<td>
<p>The <code>lty</code> for drawing shapes.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_points">points</code></td>
<td>
<p><code>logical</code>. Whether to display points. If missing and
number of points is &lt; 100, then points are plotted.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_first.point">first.point</code></td>
<td>
<p><code>logical</code> whether to plot or not the first point.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_cex.first.point">cex.first.point</code></td>
<td>
<p><code>numeric</code> size of this first point</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_centroid">centroid</code></td>
<td>
<p><code>logical</code>. Whether to display centroid.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_xy.axis">xy.axis</code></td>
<td>
<p><code>logical</code>. Whether to draw the xy axis.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_pch">pch</code></td>
<td>
<p>The <code>pch</code> for points.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_cex">cex</code></td>
<td>
<p>The <code>cex</code> for points.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_main">main</code></td>
<td>
<p><code>character</code>. A title for the plot.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_poly">poly</code></td>
<td>
<p>logical whether to use <a href="graphics.html#topic+polygon">polygon</a> and <a href="graphics.html#topic+lines">lines</a> to draw the shape,
or just <a href="graphics.html#topic+points">points</a>. In other words, whether the shape should be considered as a configuration
of landmarks or not (eg a closed outline).</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_plot.new">plot.new</code></td>
<td>
<p><code>logical</code> whether to plot or not a new frame.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_plot">plot</code></td>
<td>
<p>logical whether to plot something or just to create an empty plot.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_zoom">zoom</code></td>
<td>
<p>a numeric to take your distances.</p>
</td></tr>
<tr><td><code id="coo_plot_+3A_...">...</code></td>
<td>
<p>further arguments for use in coo_plot methods. See examples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>See Also</h3>

<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- bot[1]
coo_plot(b)
coo_plot(bot[2], plot.new=FALSE) # equivalent to coo_draw(bot[2])
coo_plot(b, zoom=2)
coo_plot(b, border='blue')
coo_plot(b, first.point=FALSE, centroid=FALSE)
coo_plot(b, points=TRUE, pch=20)
coo_plot(b, xy.axis=FALSE, lwd=2, col='#F2F2F2')
</code></pre>

<hr>
<h2 id='coo_range'>Calculate coordinates range</h2><span id='topic+coo_range'></span><span id='topic+coo_range.default'></span><span id='topic+coo_range.Coo'></span><span id='topic+coo_range_enlarge'></span><span id='topic+coo_range_enlarge.default'></span><span id='topic+coo_range_enlarge.Coo'></span><span id='topic+coo_range_enlarge.list'></span><span id='topic+coo_diffrange'></span><span id='topic+coo_diffrange.default'></span><span id='topic+coo_diffrange.Coo'></span><span id='topic+coo_diffrange.list'></span>

<h3>Description</h3>

<p><code>coo_range</code> simply returns the range,
<code>coo_range_enlarge</code> enlarges it by a <code>k</code> proportion.
<code>coo_diffrange</code> return the amplitude (ie diff after <code>coo_range</code>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_range(coo)

## Default S3 method:
coo_range(coo)

## S3 method for class 'Coo'
coo_range(coo)

coo_range_enlarge(coo, k)

## Default S3 method:
coo_range_enlarge(coo, k = 0)

## S3 method for class 'Coo'
coo_range_enlarge(coo, k = 0)

## S3 method for class 'list'
coo_range_enlarge(coo, k = 0)

coo_diffrange(coo)

## Default S3 method:
coo_diffrange(coo)

## S3 method for class 'Coo'
coo_diffrange(coo)

## S3 method for class 'list'
coo_diffrange(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_range_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_range_+3A_k">k</code></td>
<td>
<p><code>numeric</code> proportion by which to enlarge it</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix of range such as <code style="white-space: pre;">&#8288;(min, max) x (x, y)&#8288;</code>
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot[1] %&gt;% coo_range # single shape
bot    %&gt;% coo_range # Coo object

bot[1] %&gt;% coo_range_enlarge(1/50) # single shape
bot    %&gt;% coo_range_enlarge(1/50) # Coo object
</code></pre>

<hr>
<h2 id='coo_rectangularity'>Calculates the rectangularity of a shape</h2><span id='topic+coo_rectangularity'></span>

<h3>Description</h3>

<p>Calculates the rectangularity of a shape
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_rectangularity(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_rectangularity_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or any <code>Coo</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> for a single shape, <code>list</code> for <code>Coo</code>
</p>


<h3>Source</h3>

<p>Rosin PL. 2005. Computing global shape measures.
Handbook of Pattern Recognition and Computer Vision. 177-196.
</p>


<h3>See Also</h3>

<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_rectangularity(bot[1])

bot %&gt;%
    slice(1:3) %&gt;% # for speed sake only
    coo_rectangularity
</code></pre>

<hr>
<h2 id='coo_rectilinearity'>Calculates the rectilinearity of a shape</h2><span id='topic+coo_rectilinearity'></span>

<h3>Description</h3>

<p>As proposed by Zunic and Rosin (see below). May need some testing/review.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_rectilinearity(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_rectilinearity_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or any <code>Coo</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> for a single shape, <code>list</code> for <code>Coo</code>
</p>


<h3>Note</h3>

<p>due to the laborious nature of the algorithm (in nb.pts^2),
and of its implementation, it may be very long to compute.
</p>


<h3>Source</h3>

<p>Zunic J, Rosin PL. 2003. Rectilinearity measurements for polygons.
IEEE Transactions on Pattern Analysis and Machine Intelligence 25: 1193-1200.
</p>


<h3>See Also</h3>

<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot[1] %&gt;%
    coo_sample(32) %&gt;% # for speed sake only
    coo_rectilinearity

bot %&gt;%
    slice(1:3) %&gt;% coo_sample(32) %&gt;% # for speed sake only
    coo_rectilinearity
</code></pre>

<hr>
<h2 id='coo_rev'>Reverses coordinates</h2><span id='topic+coo_rev'></span>

<h3>Description</h3>

<p>Returns the reverse suite of coordinates, i.e. change shape's orientation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_rev(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_rev_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates or a Coo object
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- coo_sample(bot[1], 4)
b
coo_rev(b)
</code></pre>

<hr>
<h2 id='coo_right'>Retains coordinates with positive x-coordinates</h2><span id='topic+coo_right'></span>

<h3>Description</h3>

<p>Useful when shapes are aligned along the y-axis (e.g. because of a
bilateral symmetry) and when one wants to retain just the upper side.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_right(coo, slidegap = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_right_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_right_+3A_slidegap">slidegap</code></td>
<td>
<p><code>logical</code> whether to apply <a href="#topic+coo_slidegap">coo_slidegap</a> after coo_right</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates or a <a href="#topic+Coo">Coo</a> object (<a href="#topic+Out">Out</a> are returned as <a href="#topic+Opn">Opn</a>)
</p>


<h3>Note</h3>

<p>When shapes are &quot;sliced&quot; along the y-axis, it usually results on open curves and thus to huge/artefactual
gaps between points neighboring this axis. This is usually solved with <a href="#topic+coo_slidegap">coo_slidegap</a>. See examples there.
</p>
<p>Also, when apply a coo_left/right/up/down on an <a href="#topic+Out">Out</a> object, you then obtain an <a href="#topic+Opn">Opn</a> object, which is done
automatically.
</p>


<h3>See Also</h3>

<p>Other opening functions: 
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- coo_center(bot[1])
coo_plot(b)
coo_draw(coo_right(b), border='red')
</code></pre>

<hr>
<h2 id='coo_rotate'>Rotates coordinates</h2><span id='topic+coo_rotate'></span>

<h3>Description</h3>

<p>Rotates the coordinates by a 'theta' angle (in radians) in
the trigonometric direction (anti-clockwise). If not provided,
assumed to be the centroid size. It involves three steps: centering from current position,
dividing coordinates by 'scale', translating to the original position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_rotate(coo, theta = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_rotate_+3A_coo">coo</code></td>
<td>
<p>either a <code>matrix</code> of (x; y) coordinates, or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_rotate_+3A_theta">theta</code></td>
<td>
<p><code>numeric</code>the angle (in radians) to rotate shapes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>
<p>Other rotation functions: 
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_plot(bot[1])
coo_plot(coo_rotate(bot[1], pi/2))

# on Coo
b &lt;- bot %&gt;% slice(1:5) # for speed sake
stack(b)
stack(coo_rotate(b, pi/2))
</code></pre>

<hr>
<h2 id='coo_rotatecenter'>Rotates shapes with a custom center</h2><span id='topic+coo_rotatecenter'></span>

<h3>Description</h3>

<p>rotates a shape of 'theta' angles (in radians) and with a (x; y) 'center'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_rotatecenter(coo, theta, center = c(0, 0))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_rotatecenter_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_rotatecenter_+3A_theta">theta</code></td>
<td>
<p><code>numeric</code> the angle (in radians) to rotate shapes.</p>
</td></tr>
<tr><td><code id="coo_rotatecenter_+3A_center">center</code></td>
<td>
<p><code>numeric</code> the (x; y) position of the center</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other rotation functions: 
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>
<p>Other rotation functions: 
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- bot[1]
coo_plot(b)
coo_draw(coo_rotatecenter(b, -pi/2, c(200, 200)), border='red')
</code></pre>

<hr>
<h2 id='coo_ruban'>Plots differences as (colored) segments aka a ruban</h2><span id='topic+coo_ruban'></span>

<h3>Description</h3>

<p>Useful to display differences between shapes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_ruban(coo, dev, palette = col_heat, normalize = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_ruban_+3A_coo">coo</code></td>
<td>
<p>a shape, typically a mean shape</p>
</td></tr>
<tr><td><code id="coo_ruban_+3A_dev">dev</code></td>
<td>
<p>numeric a vector of distances or anythinh relevant</p>
</td></tr>
<tr><td><code id="coo_ruban_+3A_palette">palette</code></td>
<td>
<p>the color palette to use or any palette</p>
</td></tr>
<tr><td><code id="coo_ruban_+3A_normalize">normalize</code></td>
<td>
<p>logical whether to normalize (TRUE by default) distances</p>
</td></tr>
<tr><td><code id="coo_ruban_+3A_...">...</code></td>
<td>
<p>other parameters to fed segments, eg lwd (see examples)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>See Also</h3>

<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>
<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ms &lt;- MSHAPES(efourier(bot , 10), "type")
b &lt;- ms$shp$beer
w &lt;- ms$shp$whisky
# we obtain the mean shape, then euclidean distances between points
m &lt;- MSHAPES(list(b, w))
d &lt;- edm(b, w)
# First plot
coo_plot(m, plot=FALSE)
coo_draw(b)
coo_draw(w)
coo_ruban(m, d, lwd=5)

#Another example
coo_plot(m, plot=FALSE)
coo_ruban(m, d, palette=col_summer2, lwd=5)

#If you want linewidth rather than color
coo_plot(m, plot=FALSE)
coo_ruban(m, d, palette=col_black)
</code></pre>

<hr>
<h2 id='coo_sample'>Sample coordinates (among points)</h2><span id='topic+coo_sample'></span>

<h3>Description</h3>

<p>Sample n coordinates among existing points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_sample(coo, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_sample_+3A_coo">coo</code></td>
<td>
<p>either a <code>matrix</code> of (x; y) coordinates or an <a href="#topic+Out">Out</a> or an <a href="#topic+Opn">Opn</a> object.</p>
</td></tr>
<tr><td><code id="coo_sample_+3A_n">n</code></td>
<td>
<p><code>integer</code>, the number fo points to sample.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the <a href="#topic+Out">Out</a> an <a href="#topic+Opn">Opn</a>
methods (pointless for <a href="#topic+Ldk">Ldk</a>), in an <code>$ldk</code> component is defined,
it is changed accordingly by multiplying the ids by n over the number of coordinates.
</p>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or an <a href="#topic+Out">Out</a> or an <a href="#topic+Opn">Opn</a> object.
</p>


<h3>See Also</h3>

<p>Other sampling functions: 
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- bot[1]
stack(bot)
stack(coo_sample(bot, 24))
coo_plot(b)
coo_plot(coo_sample(b, 24))
</code></pre>

<hr>
<h2 id='coo_sample_prop'>Sample a proportion of coordinates (among points)</h2><span id='topic+coo_sample_prop'></span>

<h3>Description</h3>

<p>A simple wrapper around <a href="#topic+coo_sample">coo_sample</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_sample_prop(coo, prop = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_sample_prop_+3A_coo">coo</code></td>
<td>
<p>either a <code>matrix</code> of (x; y) coordinates or an <a href="#topic+Out">Out</a> or an <a href="#topic+Opn">Opn</a> object.</p>
</td></tr>
<tr><td><code id="coo_sample_prop_+3A_prop">prop</code></td>
<td>
<p><code>numeric</code>, the proportion of points to sample</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As for <a href="#topic+coo_sample">coo_sample</a> if an <code>$ldk</code> component is defined,
it is changed accordingly by multiplying the ids by n over the number of coordinates.
</p>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or an <a href="#topic+Out">Out</a> or an <a href="#topic+Opn">Opn</a> object.
</p>


<h3>See Also</h3>

<p>Other sampling functions: 
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># single shape
bot[1] %&gt;% coo_nb()
bot[1] %&gt;% coo_sample_prop(0.5) %&gt;% coo_nb()
</code></pre>

<hr>
<h2 id='coo_samplerr'>Samples coordinates (regular radius)</h2><span id='topic+coo_samplerr'></span>

<h3>Description</h3>

<p>Samples n coordinates with a regular angle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_samplerr(coo, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_samplerr_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_samplerr_+3A_n">n</code></td>
<td>
<p><code>integer</code>, the number of points to sample.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By design, this function samples among existing points, so using
<a href="#topic+coo_interpolate">coo_interpolate</a> prior to it may be useful to have
more homogeneous angles. See examples.
</p>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates or a Coo object.
</p>


<h3>See Also</h3>

<p>Other sampling functions: 
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stack(bot)
bot &lt;- coo_center(bot)
stack(coo_samplerr(bot, 12))
coo_plot(bot[1])
coo_plot(rr &lt;- coo_samplerr(bot[1], 12))
cpos &lt;- coo_centpos(bot[1])
segments(cpos[1], cpos[2], rr[, 1], rr[, 2])

# Sometimes, interpolating may be useful:
shp &lt;- hearts[1] %&gt;% coo_center

# given a shp, draw segments from each points on it, to its centroid
draw_rads &lt;- function(shp, ...){
 segments(shp[, 1], shp[, 2], coo_centpos(shp)[1], coo_centpos(shp)[2], ...)
}

# calculate the sd of argument difference in successive points,
# in other words a proxy for the homogeneity of angles
sd_theta_diff &lt;- function(shp)
   shp %&gt;% complex(real=.[, 1], imaginary=.[, 2]) %&gt;%
   Arg %&gt;% `[`(-1) %&gt;% diff %&gt;% sd

# no interpolation: all points are sampled from existing points but the
# angles are not equal
shp %&gt;% coo_plot(points=TRUE, main="no interpolation")
shp %&gt;% coo_samplerr(64) %T&gt;% draw_rads(col="red") %&gt;% sd_theta_diff
# with interpolation: much more homogeneous angles
shp %&gt;% coo_plot(points=TRUE)
shp %&gt;% coo_interpolate(360) %&gt;% coo_samplerr(64) %T&gt;% draw_rads(col="blue") %&gt;% sd_theta_diff
</code></pre>

<hr>
<h2 id='coo_scalars'>Calculates all scalar descriptors of shape</h2><span id='topic+coo_scalars'></span>

<h3>Description</h3>

<p>See examples for the full list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_scalars(coo, rectilinearity = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_scalars_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or any <code>Coo</code></p>
</td></tr>
<tr><td><code id="coo_scalars_+3A_rectilinearity">rectilinearity</code></td>
<td>
<p><code>logical</code> whether to include rectilinearity using <a href="#topic+coo_rectilinearity">coo_rectilinearity</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p><a href="#topic+coo_rectilinearity">coo_rectilinearity</a> being not particularly optimized, it takes around 30 times more
time to include it than to calculate <em>all</em> others and is thus not includedby default.
by default.
</p>


<h3>Value</h3>

<p><code>data_frame</code>
</p>


<h3>See Also</h3>

<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
df &lt;- bot %&gt;% coo_scalars() # pass bot %&gt;% coo_scalars(TRUE) if you want rectilinearity
colnames(df) %&gt;% cat(sep="\n") # all scalars used

# a PCA on all these descriptors
TraCoe(coo_scalars(bot), fac=bot$fac) %&gt;% PCA %&gt;% plot_PCA(~type)

</code></pre>

<hr>
<h2 id='coo_scale'>Scales coordinates</h2><span id='topic+coo_scale'></span><span id='topic+coo_scale.default'></span><span id='topic+coo_scale.Coo'></span><span id='topic+coo_scalex'></span><span id='topic+coo_scalex.default'></span><span id='topic+coo_scalex.Coo'></span><span id='topic+coo_scaley'></span><span id='topic+coo_scaley.default'></span><span id='topic+coo_scaley.Coo'></span>

<h3>Description</h3>

<p><code>coo_scale</code> scales the coordinates by a 'scale' factor. If not provided,
assumed to be the centroid size. It involves three steps: centering from current position,
dividing coordinates by 'scale', pushing back to the original position.
<code>coo_scalex</code> applies a scaling (or shrinking) parallel to the x-axis,
<code>coo_scaley</code> does the same for the y axis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_scale(coo, scale)

## Default S3 method:
coo_scale(coo, scale = coo_centsize(coo))

## S3 method for class 'Coo'
coo_scale(coo, scale)

coo_scalex(coo, scale = 1)

## Default S3 method:
coo_scalex(coo, scale = 1)

## S3 method for class 'Coo'
coo_scalex(coo, scale = 1)

coo_scaley(coo, scale = 1)

## Default S3 method:
coo_scaley(coo, scale = 1)

## S3 method for class 'Coo'
coo_scaley(coo, scale = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_scale_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_scale_+3A_scale">scale</code></td>
<td>
<p>the scaling factor,
by default, the centroid size for <code>coo_scale</code>; 1 for <code>scalex</code> and <code>scaley</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a single shape or a <code>Coo</code> object
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>
<p>Other scaling functions: 
<code><a href="#topic+coo_template">coo_template</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># on a single shape
b &lt;- bot[1] %&gt;% coo_center %&gt;% coo_scale
coo_plot(b, lwd=2)
coo_draw(coo_scalex(b, 1.5), bor="blue")
coo_draw(coo_scaley(b, 0.5), bor="red")

# this also works on Coo objects:
b &lt;- slice(bot, 5) # for speed sake
stack(b)
b %&gt;% coo_center %&gt;% coo_scale %&gt;% stack
b %&gt;% coo_center %&gt;% coo_scaley(0.5) %&gt;% stack
#equivalent to:
#b %&gt;% coo_center %&gt;% coo_scalex(2) %&gt;% stack
</code></pre>

<hr>
<h2 id='coo_shearx'>Shears shapes</h2><span id='topic+coo_shearx'></span><span id='topic+coo_sheary'></span>

<h3>Description</h3>

<p><code>coo_shearx</code> applies a shear mapping on a matrix of (x; y) coordinates (or a list), parallel
to the x-axis (i.e. x' = x + ky; y' = y + kx). <code>coo_sheary</code> does it parallel to the y-axis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_shearx(coo, k)

coo_sheary(coo, k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_shearx_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_shearx_+3A_k">k</code></td>
<td>
<p><code>numeric</code> shear factor</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates.
</p>


<h3>See Also</h3>

<p>Other transforming functions: 
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo &lt;- coo_template(shapes[11])
coo_plot(coo)
coo_draw(coo_shearx(coo, 0.5), border="blue")
coo_draw(coo_sheary(coo, 0.5), border="red")

</code></pre>

<hr>
<h2 id='coo_slice'>Slices shapes between successive coordinates</h2><span id='topic+coo_slice'></span>

<h3>Description</h3>

<p>Takes a shape with n coordinates. When you pass this function with at least
two ids (&lt;= n), the shape will be open on the corresponding coordinates and
slices returned as a list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_slice(coo, ids, ldk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_slice_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_slice_+3A_ids">ids</code></td>
<td>
<p><code>numeric</code> of length &gt;= 2, where to slice the shape(s)</p>
</td></tr>
<tr><td><code id="coo_slice_+3A_ldk">ldk</code></td>
<td>
<p><code>numeric</code> the id of the ldk to use as ids, only on <code>Out</code> and <code>Opn</code>.
If provided, <code>ids</code> will be ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of shapes or a list of <a href="#topic+Opn">Opn</a>
</p>


<h3>See Also</h3>

<p>Have a look to <a href="#topic+coo_slidegap">coo_slidegap</a> if you have problems with gaps
after slicing around landmarks and/or starting points.
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>h &lt;- slice(hearts, 1:5)  # speed purpose only
# single shape, a list of matrices is returned
sh &lt;- coo_slice(h[1], c(12, 24, 36, 48))
coo_plot(sh[[1]])
panel(Opn(sh))
# on a Coo, a list of Opn is returned
# makes no sense if shapes are not normalized first
sh2 &lt;- coo_slice(h, c(12, 24, 36, 48))
panel(sh2[[1]])

# Use coo_slice with `ldk` instead:
# hearts as an example
x &lt;- h %&gt;% fgProcrustes(tol=1)
# 4 landmarks
stack(x)
x$ldk[1:5]

# here we slice
y &lt;- coo_slice(x, ldk=1:4)

# plotting
stack(y[[1]])
stack(y[[2]])

# new ldks from tipping points, new ldks from angle
olea %&gt;% slice(1:5) %&gt;% # for the sake of speed
def_ldk_tips %&gt;%
def_ldk_angle(0.75*pi) %&gt;% def_ldk_angle(0.25*pi) %&gt;%
coo_slice(ldk =1:4) -&gt; oleas
oleas[[1]] %&gt;% stack
oleas[[2]] %&gt;% stack # etc.

# domestic operations
y[[3]] %&gt;% coo_area()
# shape analysis of a slice
y[[1]] %&gt;% coo_bookstein() %&gt;% npoly %&gt;% PCA %&gt;% plot(~aut)

</code></pre>

<hr>
<h2 id='coo_slide'>Slides coordinates</h2><span id='topic+coo_slide'></span>

<h3>Description</h3>

<p>Slides the coordinates so that the id-th point become the first one.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_slide(coo, id, ldk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_slide_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_slide_+3A_id">id</code></td>
<td>
<p><code>numeric</code> the id of the point that will become the new first point. See details below
for the method on Coo objects.</p>
</td></tr>
<tr><td><code id="coo_slide_+3A_ldk">ldk</code></td>
<td>
<p><code>numeric</code> the id of the ldk to use as id, only on <code>Out</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For Coo objects, and in particular for Out and Opn three different ways of coo_sliding
are available:
</p>

<ul>
<li> <p><strong>no ldk passed and a single id is passed</strong>: all id-th points
within the shapes will become the first points. $ldk will be slided accordingly.
</p>
</li>
<li> <p><strong>no ldk passed and a vector of ids matching the length of the Coo</strong>: for every shape,
the id-th point will be used as the id-th point. $ldk will be slided accordingly.
</p>
</li>
<li> <p><strong>a single ldk is passed</strong>: the ldk-th ldk will be used to slide every shape.
If an id is (also) passed, it is ignored with a message.
</p>
</li></ul>

<p>See examples.
</p>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p><a href="#topic+coo_slice">coo_slice</a> and friends.
</p>
<p>Other sliding functions: 
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>h &lt;- hearts %&gt;% slice(1:5) # for speed sake
stack(h)
# set the first landmark as the starting point
stack(coo_slide(h, ldk=1))
# set the 50th point as the starting point (everywhere)
stack(coo_slide(h, id=50))
# set the id-random-th point as the starting point (everywhere)
set.seed(123) # just for the reproducibility
id_random &lt;- sample(x=min(sapply(h$coo, nrow)), size=length(h),
replace=TRUE)
stack(coo_slide(h, id=id_random))
</code></pre>

<hr>
<h2 id='coo_slidedirection'>Slides coordinates in a particular direction</h2><span id='topic+coo_slidedirection'></span>

<h3>Description</h3>

<p>Shapes are centered and then, according to direction, the point northwards, southwards,
eastwards or westwards the centroid, becomes the first point with <a href="#topic+coo_slide">coo_slide</a>.
'right' is possibly the most sensible option (and is by default),
since 0 radians points eastwards, relatively to the origin.
This should be followed by a <a href="#topic+coo_untiltx">coo_untiltx</a> is most cases to remove any rotationnal dephasing/bias.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_slidedirection(
  coo,
  direction = c("down", "left", "up", "right")[4],
  center,
  id
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_slidedirection_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_slidedirection_+3A_direction">direction</code></td>
<td>
<p><code>character</code> one of <code>"down", "left", "up", "right"</code> (&quot;right&quot; by default)</p>
</td></tr>
<tr><td><code id="coo_slidedirection_+3A_center">center</code></td>
<td>
<p><code>logical</code> whether to center or not before sliding</p>
</td></tr>
<tr><td><code id="coo_slidedirection_+3A_id">id</code></td>
<td>
<p><code>numeric</code> whether to return the id of the point or the slided shapes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other sliding functions: 
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- coo_rotate(bot[1], pi/6) # dummy example just to make it obvious
coo_plot(b) # not the first point
coo_plot(coo_slidedirection(b, "up"))
coo_plot(coo_slidedirection(b, "right"))
coo_plot(coo_slidedirection(b, "left"))
coo_plot(coo_slidedirection(b, "down"))

# on Coo objects
b &lt;- bot %&gt;% slice(1:5) # for speed sake
stack(b)
stack(coo_slidedirection(b, "right"))

# This should be followed by a [coo_untiltx] in most (if not all) cases
stack(coo_slidedirection(b, "right") %&gt;% coo_untiltx)

</code></pre>

<hr>
<h2 id='coo_slidegap'>Slides coordinates using the widest gap</h2><span id='topic+coo_slidegap'></span>

<h3>Description</h3>

<p>When slicing a shape using two landmarks, or functions such as <a href="#topic+coo_up">coo_up</a>,
an open curve is obtained and the rank of points make wrong/artefactual results.
If the widest gap is &gt; 5 * median of other gaps, then the couple of coordinates
forming this widest gap is used as starting and ending points. This switch helps
to deal with open curves. Examples are self-speaking.
Use <code>force=TRUE</code> to bypass this check
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_slidegap(coo, force)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_slidegap_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_slidegap_+3A_force">force</code></td>
<td>
<p><code>logical</code> whether to use the widest gap, with no check, as the real gap</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other sliding functions: 
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cat &lt;- coo_center(shapes[4])
coo_plot(cat)

# we only retain the bottom of the cat
cat_down &lt;- coo_down(cat, slidegap=FALSE)

# see? the segment on the x-axis coorespond to the widest gap.
coo_plot(cat_down)

# that's what we meant
coo_plot(coo_slidegap(cat_down))
</code></pre>

<hr>
<h2 id='coo_smooth'>Smoothes coordinates</h2><span id='topic+coo_smooth'></span>

<h3>Description</h3>

<p>Smoothes coordinates using a simple moving average.
May be useful to remove digitization noise, mainly on outlines and open outlines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_smooth(coo, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_smooth_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_smooth_+3A_n">n</code></td>
<td>
<p><code>integer</code> the number of smoothing iterations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other smoothing functions: 
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b5 &lt;- slice(bot, 1:5) # for speed sake
stack(b5)
stack(coo_smooth(b5, 10))
coo_plot(b5[1])
coo_plot(coo_smooth(b5[1], 30))
</code></pre>

<hr>
<h2 id='coo_smoothcurve'>Smoothes coordinates on curves</h2><span id='topic+coo_smoothcurve'></span>

<h3>Description</h3>

<p>Smoothes coordinates using a simple moving average but let the first and last points unchanged.
May be useful to remove digitization noise on curves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_smoothcurve(coo, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_smoothcurve_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_smoothcurve_+3A_n">n</code></td>
<td>
<p><code>integer</code> to specify the number of smoothing iterations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other smoothing functions: 
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>o &lt;- olea[1]
coo_plot(o, border='grey50', points=FALSE)
coo_draw(coo_smooth(o, 24), border='blue', points=FALSE)
coo_draw(coo_smoothcurve(o, 24), border='red', points=FALSE)
</code></pre>

<hr>
<h2 id='coo_solidity'>Calculates the solidity of a shape</h2><span id='topic+coo_solidity'></span>

<h3>Description</h3>

<p>Calculated using the ratio of the shape area and the convex hull area.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_solidity(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_solidity_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or any <code>Coo</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> for a single shape, <code>list</code> for <code>Coo</code>
</p>


<h3>Source</h3>

<p>Rosin PL. 2005. Computing global shape measures.
Handbook of Pattern Recognition and Computer Vision. 177-196.
</p>


<h3>See Also</h3>

<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_solidity(bot[1])

bot %&gt;%
    slice(1:3) %&gt;%  # for speed sake only
    coo_solidity
</code></pre>

<hr>
<h2 id='coo_tac'>Calculates the total absolute curvature of a shape</h2><span id='topic+coo_tac'></span>

<h3>Description</h3>

<p>Calculated using the sum of the absolute value of the second derivative of
the <code>smooth.spline</code> prediction for each defined point.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_tac(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_tac_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or any <code>Coo</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> for a single shape and for <code>Coo</code>
</p>


<h3>Source</h3>

<p>Siobhan Braybrook.
</p>


<h3>See Also</h3>

<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_width">coo_width</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_tac(bot[1])

bot %&gt;%
    slice(1:3) %&gt;%  # for speed sake only
    coo_tac
</code></pre>

<hr>
<h2 id='coo_template'>'Templates' shapes</h2><span id='topic+coo_template'></span><span id='topic+coo_template.default'></span><span id='topic+coo_template.list'></span><span id='topic+coo_template.Coo'></span><span id='topic+coo_template_relatively'></span><span id='topic+coo_template_relatively.list'></span><span id='topic+coo_template_relatively.Coo'></span>

<h3>Description</h3>

<p><code>coo_template</code> returns shape centered on the origin
and inscribed in a <code>size</code>-side square. <code>coo_template_relatively</code>
does the same but the biggest shape (as <code>prod(coo_diffrange)</code>) will
be of <code>size=size</code> and consequently not defined on single shapes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_template(coo, size)

## Default S3 method:
coo_template(coo, size = 1)

## S3 method for class 'list'
coo_template(coo, size = 1)

## S3 method for class 'Coo'
coo_template(coo, size = 1)

coo_template_relatively(coo, size = 1)

## S3 method for class 'list'
coo_template_relatively(coo, size = 1)

## S3 method for class 'Coo'
coo_template_relatively(coo, size = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_template_+3A_coo">coo</code></td>
<td>
<p>A <code>list</code> or a <code>matrix</code> of coordinates.</p>
</td></tr>
<tr><td><code id="coo_template_+3A_size">size</code></td>
<td>
<p><code>numeric</code>. Indicates the length of the side 'inscribing'
the shape.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <a href="#topic+coo_listpanel">coo_listpanel</a> for an illustration of this function. The morphospaces
functions also take profit of this function. May be useful to develop other graphical functions.
</p>


<h3>Value</h3>

<p>Returns a matrix of <code>(x; y)</code>coordinates.
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>
<p>Other scaling functions: 
<code><a href="#topic+coo_scale">coo_scale</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
coo &lt;- bot[1]
coo_plot(coo_template(coo), xlim=c(-1, 1), ylim=c(-1, 1))
rect(-0.5, -0.5, 0.5, 0.5)

s &lt;- 0.01
coo_plot(coo_template(coo, s))
rect(-s/2, -s/2, s/2, s/2)
</code></pre>

<hr>
<h2 id='coo_trans'>Translates coordinates</h2><span id='topic+coo_trans'></span>

<h3>Description</h3>

<p>Translates the coordinates by a 'x' and 'y' value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_trans(coo, x = 0, y = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_trans_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_trans_+3A_x">x</code></td>
<td>
<p><code>numeric</code> translation along the x-axis.</p>
</td></tr>
<tr><td><code id="coo_trans_+3A_y">y</code></td>
<td>
<p><code>numeric</code> translation along the y-axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_plot(bot[1])
coo_plot(coo_trans(bot[1], 50, 100))

# on Coo
b &lt;- bot %&gt;% slice(1:5) # for speed sake
stack(b)
stack(coo_trans(b, 50, 100))
</code></pre>

<hr>
<h2 id='coo_trim'>Trims both ends coordinates from shape</h2><span id='topic+coo_trim'></span>

<h3>Description</h3>

<p>Removes <code>trim</code> coordinates at both ends of a shape, ie
from top and bottom of the shape matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_trim(coo, trim = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_trim_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_trim_+3A_trim">trim</code></td>
<td>
<p><code>numeric</code>, the number of coordinates to trim</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a trimmed shape
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>
<p>Other coo_trimming functions: 
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>olea[1] %&gt;% coo_sample(12) %T&gt;%
   print() %T&gt;% ldk_plot() %&gt;%
   coo_trim(1) %T&gt;% print() %&gt;% points(col="red")
</code></pre>

<hr>
<h2 id='coo_trimbottom'>Trims bottom coordinates from shape</h2><span id='topic+coo_trimbottom'></span>

<h3>Description</h3>

<p>Removes <code>trim</code> coordinates from the bottom of a shape.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_trimbottom(coo, trim = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_trimbottom_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_trimbottom_+3A_trim">trim</code></td>
<td>
<p><code>numeric</code>, the number of coordinates to trim</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a trimmed shape
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>
<p>Other coo_trimming functions: 
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>olea[1] %&gt;% coo_sample(12) %T&gt;%
   print() %T&gt;% ldk_plot() %&gt;%
   coo_trimbottom(4) %T&gt;% print() %&gt;% points(col="red")
</code></pre>

<hr>
<h2 id='coo_trimtop'>Trims top coordinates from shape</h2><span id='topic+coo_trimtop'></span>

<h3>Description</h3>

<p>Removes <code>trim</code> coordinates from the top of a shape.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_trimtop(coo, trim = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_trimtop_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_trimtop_+3A_trim">trim</code></td>
<td>
<p><code>numeric</code>, the number of coordinates to trim</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a trimmed shape
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>
<p>Other coo_trimming functions: 
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>olea[1] %&gt;% coo_sample(12) %T&gt;%
   print() %T&gt;% ldk_plot() %&gt;%
   coo_trimtop(4) %T&gt;% print() %&gt;% points(col="red")
</code></pre>

<hr>
<h2 id='coo_truss'>Truss measurement</h2><span id='topic+coo_truss'></span>

<h3>Description</h3>

<p>A method to calculate on shapes or on <a href="#topic+Coo">Coo</a> truss measurements,
that is all pairwise combinations of euclidean distances
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_truss(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_truss_+3A_x">x</code></td>
<td>
<p>a shape or an Ldk object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a named numeric or matrix
</p>


<h3>Note</h3>

<p>Mainly implemented for historical/didactical reasons.
</p>


<h3>See Also</h3>

<p>Other premodern: 
<code><a href="#topic+measure">measure</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example on a single shape
cat &lt;- coo_sample(shapes[4], 6)
coo_truss(cat)

# example on wings dataset
tx &lt;- coo_truss(wings)

txp &lt;- PCA(tx, scale. = TRUE, center=TRUE, fac=wings$fac)
plot(txp, 1)
</code></pre>

<hr>
<h2 id='coo_untiltx'>Removes rotation so that the centroid and a given point are parallel to the x-axis</h2><span id='topic+coo_untiltx'></span><span id='topic+coo_untilt'></span>

<h3>Description</h3>

<p>Rotationnal biases appear after <a href="#topic+coo_slidedirection">coo_slidedirection</a> (and friends).
Typically useful for outline analysis where phasing matters. See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_untiltx(coo, id, ldk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_untiltx_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_untiltx_+3A_id">id</code></td>
<td>
<p><code>numeric</code> the id of the point that will become the new first point. See details below
for the method on Coo objects.</p>
</td></tr>
<tr><td><code id="coo_untiltx_+3A_ldk">ldk</code></td>
<td>
<p><code>numeric</code> the id of the ldk to use as id, only on <code>Out</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For Coo objects, and in particular for Out and Opn two different ways of coo_sliding
are available:
</p>

<ul>
<li> <p><strong>no ldk passed and an id is passed</strong>: all id-th points
within the shapes will become the first points.
</p>
</li>
<li> <p><strong>a single ldk is passed</strong>: the ldk-th ldk will be used to slide every shape.
If an id is (also) passed, id is ignored with a message.
</p>
</li></ul>



<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates, or a <a href="#topic+Coo">Coo</a> object.
</p>


<h3>See Also</h3>

<p><a href="#topic+coo_slide">coo_slide</a> and friends.
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># on a single shape
bot[1] %&gt;% coo_center %&gt;% coo_align %&gt;%
   coo_sample(12) %&gt;% coo_slidedirection("right") %T&gt;%
   coo_plot() %&gt;% # the first point is not on the x-axis
   coo_untiltx() %&gt;%
   coo_draw(border="red") # this (red) one is

# on an Out
# prepare bot
prebot &lt;- bot %&gt;% coo_center %&gt;% coo_scale %&gt;%
   coo_align %&gt;% coo_slidedirection("right")
prebot %&gt;% stack # some dephasing remains
prebot %&gt;% coo_slidedirection("right") %&gt;% coo_untiltx() %&gt;% stack # much better
# _here_ there is no change but the second, untilted, is correct
prebot %&gt;% efourier(8, norm=FALSE) %&gt;% PCA %&gt;% plot_PCA(~type)
prebot %&gt;% coo_untiltx %&gt;% efourier(8, norm=FALSE) %&gt;% PCA %&gt;% plot_PCA(~type)

# an example using ldks:
# the landmark #2 is on the x-axis
hearts %&gt;%
  slice(1:5) %&gt;% fgProcrustes(tol=1e-3) %&gt;% # for speed sake
  coo_center %&gt;% coo_untiltx(ldk=2) %&gt;% stack
</code></pre>

<hr>
<h2 id='coo_up'>Retains coordinates with positive y-coordinates</h2><span id='topic+coo_up'></span>

<h3>Description</h3>

<p>Useful when shapes are aligned along the x-axis (e.g. because of a
bilateral symmetry) and when one wants to retain just the upper side.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_up(coo, slidegap = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_up_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="coo_up_+3A_slidegap">slidegap</code></td>
<td>
<p><code>logical</code> whether to apply <a href="#topic+coo_slidegap">coo_slidegap</a> after coo_down</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> of (x; y) coordinates or a <a href="#topic+Coo">Coo</a> object (<a href="#topic+Out">Out</a> are returned as <a href="#topic+Opn">Opn</a>)
</p>


<h3>Note</h3>

<p>When shapes are &quot;sliced&quot; along the x-axis, it usually results on open curves and thus to huge/artefactual
gaps between points neighboring this axis. This is usually solved with <a href="#topic+coo_slidegap">coo_slidegap</a>. See examples there.
</p>
<p>Also, when apply a coo_left/right/up/down on an <a href="#topic+Out">Out</a> object, you then obtain an <a href="#topic+Opn">Opn</a> object, which is done
automatically.
</p>


<h3>See Also</h3>

<p>Other opening functions: 
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>
</p>
<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- coo_alignxax(bot[1])
coo_plot(b)
coo_draw(coo_up(b), border='red')
</code></pre>

<hr>
<h2 id='coo_width'>Calculates the width of a shape</h2><span id='topic+coo_width'></span>

<h3>Description</h3>

<p>Nothing more than <code>coo_lw(coo)[2]</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coo_width(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coo_width_+3A_coo">coo</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates or Coo object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the width (in pixels) of the shape
</p>


<h3>See Also</h3>

<p><a href="#topic+coo_lw">coo_lw</a>, <a href="#topic+coo_length">coo_length</a>.
</p>
<p>Other coo_ descriptors: 
<code><a href="#topic+coo_angle_edges">coo_angle_edges</a>()</code>,
<code><a href="#topic+coo_angle_tangent">coo_angle_tangent</a>()</code>,
<code><a href="#topic+coo_area">coo_area</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_chull">coo_chull</a>()</code>,
<code><a href="#topic+coo_circularity">coo_circularity</a>()</code>,
<code><a href="#topic+coo_convexity">coo_convexity</a>()</code>,
<code><a href="#topic+coo_eccentricity">coo_eccentricity</a></code>,
<code><a href="#topic+coo_elongation">coo_elongation</a>()</code>,
<code><a href="#topic+coo_length">coo_length</a>()</code>,
<code><a href="#topic+coo_lw">coo_lw</a>()</code>,
<code><a href="#topic+coo_rectangularity">coo_rectangularity</a>()</code>,
<code><a href="#topic+coo_rectilinearity">coo_rectilinearity</a>()</code>,
<code><a href="#topic+coo_scalars">coo_scalars</a>()</code>,
<code><a href="#topic+coo_solidity">coo_solidity</a>()</code>,
<code><a href="#topic+coo_tac">coo_tac</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_width(bot[1])
</code></pre>

<hr>
<h2 id='d'>A wrapper to calculates euclidean distances between two points</h2><span id='topic+d'></span>

<h3>Description</h3>

<p>The main advantage over <a href="#topic+ed">ed</a> is that it is a method that can
be passed to different objects and used in combination with <a href="#topic+measure">measure</a>.
See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>d(x, id1, id2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="d_+3A_x">x</code></td>
<td>
<p>a Ldk (typically), an Out or a matrix</p>
</td></tr>
<tr><td><code id="d_+3A_id1">id1</code></td>
<td>
<p>id of the 1st row</p>
</td></tr>
<tr><td><code id="d_+3A_id2">id2</code></td>
<td>
<p>id of the 2nd row</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric
</p>


<h3>Note</h3>

<p>On Out objects, we first <a href="#topic+get_ldk">get_ldk</a>.
</p>


<h3>See Also</h3>

<p>if you want all pairwise combinations, see <a href="#topic+coo_truss">coo_truss</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># single shape
d(wings[1], 1, 4)
# Ldk object
d(wings, 1, 4)
# Out object
d(hearts, 2, 4)
</code></pre>

<hr>
<h2 id='def_ldk'>Defines new landmarks on Out and Opn objects</h2><span id='topic+def_ldk'></span>

<h3>Description</h3>

<p>Helps to define landmarks on a <code>Coo</code> object.
The number of landmarks must be specified and rows indices that
correspond to the nearest points clicked on every outlines are
stored in the <code>$ldk</code> slot of the <code>Coo</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_ldk(Coo, nb.ldk, close, points)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_ldk_+3A_coo">Coo</code></td>
<td>
<p>an Out or Opn object</p>
</td></tr>
<tr><td><code id="def_ldk_+3A_nb.ldk">nb.ldk</code></td>
<td>
<p>the number of landmarks to define on every shape</p>
</td></tr>
<tr><td><code id="def_ldk_+3A_close">close</code></td>
<td>
<p><code>logical</code> whether to close (typically for outlines)</p>
</td></tr>
<tr><td><code id="def_ldk_+3A_points">points</code></td>
<td>
<p><code>logical</code> whether to display points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an Out or an Opn object with some landmarks defined
</p>


<h3>See Also</h3>

<p>Other ldk/slidings methods: 
<code><a href="#topic+add_ldk">add_ldk</a>()</code>,
<code><a href="#topic+def_slidings">def_slidings</a>()</code>,
<code><a href="#topic+get_ldk">get_ldk</a>()</code>,
<code><a href="#topic+get_slidings">get_slidings</a>()</code>,
<code><a href="#topic+rearrange_ldk">rearrange_ldk</a>()</code>,
<code><a href="#topic+slidings_scheme">slidings_scheme</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
bot &lt;- bot[1:5] # to make it shorter to try
# click on 3 points, 5 times.
# Don't forget to save the object returned by def_ldk...
bot2 &lt;- def_ldk(bot, 3)
stack(bot2)
bot2$ldk

## End(Not run)
</code></pre>

<hr>
<h2 id='def_ldk_angle'>Add new landmarks based on angular positions</h2><span id='topic+def_ldk_angle'></span><span id='topic+def_ldk_direction'></span><span id='topic+def_ldk_direction.default'></span><span id='topic+def_ldk_direction.Out'></span><span id='topic+def_ldk_direction.Opn'></span>

<h3>Description</h3>

<p>A wrapper on <a href="#topic+coo_intersect_angle">coo_intersect_angle</a> and <a href="#topic+coo_intersect_direction">coo_intersect_direction</a> for
<a href="#topic+Out">Out</a> and <a href="#topic+Opn">Opn</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_ldk_angle(coo, angle)

def_ldk_direction(coo, direction = c("down", "left", "up", "right")[4])

## Default S3 method:
def_ldk_direction(coo, direction = c("down", "left", "up", "right")[4])

## S3 method for class 'Out'
def_ldk_direction(coo, direction = c("down", "left", "up", "right")[4])

## S3 method for class 'Opn'
def_ldk_direction(coo, direction = c("down", "left", "up", "right")[4])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_ldk_angle_+3A_coo">coo</code></td>
<td>
<p>a <code>Out</code> or <code>Opn</code> object</p>
</td></tr>
<tr><td><code id="def_ldk_angle_+3A_angle">angle</code></td>
<td>
<p><code>numeric</code> an angle in radians (0 by default).</p>
</td></tr>
<tr><td><code id="def_ldk_angle_+3A_direction">direction</code></td>
<td>
<p><code>character</code> one of <code>"down", "left", "up", "right"</code> (&quot;right&quot; by default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>Note</h3>

<p>any existing ldk will be preserved.
</p>


<h3>See Also</h3>

<p>Typically used before <a href="#topic+coo_slice">coo_slice</a> and <a href="#topic+coo_slide">coo_slide</a>.
See <a href="#topic+def_ldk_tips">def_ldk_tips</a> as well.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># adds a new landmark towards south east
hearts %&gt;%
   slice(1:5) %&gt;% # for speed purpose only
   def_ldk_angle(-pi/6) %&gt;%
stack()

# on Out and towards NW and NE here
olea %&gt;%
   slice(1:5) %&gt;% #for speed purpose only
   def_ldk_angle(3*pi/4) %&gt;%
   def_ldk_angle(pi/4) %&gt;%
   stack

</code></pre>

<hr>
<h2 id='def_ldk_tips'>Define tips as new landmarks</h2><span id='topic+def_ldk_tips'></span>

<h3>Description</h3>

<p>On <a href="#topic+Opn">Opn</a> objects, this can be used before <a href="#topic+coo_slice">coo_slice</a>. See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_ldk_tips(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_ldk_tips_+3A_coo">coo</code></td>
<td>
<p><code>Opn</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>Note</h3>

<p>any existing ldk will be preserved.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_ldk(olea) # no ldk for olea
olea %&gt;%
slice(1:3) %&gt;% #for the sake of speed
def_ldk_tips %&gt;%
def_ldk_angle(3*pi/4) %&gt;% def_ldk_angle(pi/4) %T&gt;% stack %&gt;%
coo_slice(ldk=1:4) -&gt; oleas
stack(oleas[[1]])
stack(oleas[[2]]) # etc.
</code></pre>

<hr>
<h2 id='def_links'>Defines links between landmarks</h2><span id='topic+def_links'></span>

<h3>Description</h3>

<p>Works on Ldk objects, on 2-cols matrices, 3-dim arrays (<a href="#topic+MSHAPES">MSHAPES</a> turns it into a matrix).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_links(x, nb.ldk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_links_+3A_x">x</code></td>
<td>
<p>Ldk, matric or array</p>
</td></tr>
<tr><td><code id="def_links_+3A_nb.ldk">nb.ldk</code></td>
<td>
<p>numeric the iterative procedure is stopped when the
user click on the top of the graphical window.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>See Also</h3>

<p>Other ldk helpers: 
<code><a href="#topic+ldk_check">ldk_check</a>()</code>,
<code><a href="#topic+links_all">links_all</a>()</code>,
<code><a href="#topic+links_delaunay">links_delaunay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
wm &lt;- MSHAPES(wings)
links &lt;- def_links(wm, 3) # click to define pairs of landmarks
ldk_links(wm, links)

## End(Not run)
</code></pre>

<hr>
<h2 id='def_slidings'>Defines sliding landmarks matrix</h2><span id='topic+def_slidings'></span>

<h3>Description</h3>

<p>Defines sliding landmarks matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>def_slidings(Coo, slidings)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="def_slidings_+3A_coo">Coo</code></td>
<td>
<p>an <a href="#topic+Ldk">Ldk</a> object</p>
</td></tr>
<tr><td><code id="def_slidings_+3A_slidings">slidings</code></td>
<td>
<p>a matrix, a numeric or a list of numeric. See Details</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>$slidings</code> in <a href="#topic+Ldk">Ldk</a> must be a 'valid' matrix: containing
ids of coordinates, none of them being lower than 1 and higher the number of coordinates
in <code>$coo</code>.
</p>
<p><code>slidings</code> matrix contains 3 columns (<code>before</code>, <code>slide</code>, <code>after</code>).
It is inspired by <code>geomorph</code> and should be compatible with it.
</p>
<p>This matrix can be passed directly if the <code>slidings</code> argument is a matrix. Of course,
it is strictly equivalent to <code>Ldk$slidings &lt;- slidings</code>.
</p>
<p><code>slidings</code> can also be passed as &quot;partition(s)&quot;, when sliding landmarks
identified by their ids (which are a row number) are consecutive in the <code>$coo</code>.
</p>
<p>A single partition can be passed either as a numeric (eg <code>4:12</code>), if points
5 to 11 must be considered as sliding landmarks (4 and 12 being fixed); or as a list of numeric.
</p>
<p>See examples below.
</p>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>See Also</h3>

<p>Other ldk/slidings methods: 
<code><a href="#topic+add_ldk">add_ldk</a>()</code>,
<code><a href="#topic+def_ldk">def_ldk</a>()</code>,
<code><a href="#topic+get_ldk">get_ldk</a>()</code>,
<code><a href="#topic+get_slidings">get_slidings</a>()</code>,
<code><a href="#topic+rearrange_ldk">rearrange_ldk</a>()</code>,
<code><a href="#topic+slidings_scheme">slidings_scheme</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#waiting for a sliding dataset...

</code></pre>

<hr>
<h2 id='dfourier'>Discrete cosinus transform</h2><span id='topic+dfourier'></span><span id='topic+dfourier.default'></span><span id='topic+dfourier.Opn'></span><span id='topic+dfourier.list'></span><span id='topic+dfourier.Coo'></span>

<h3>Description</h3>

<p>Calculates discrete cosine transforms, as introduced by Dommergues and colleagues, on
a shape (mainly open outlines).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfourier(coo, nb.h)

## Default S3 method:
dfourier(coo, nb.h)

## S3 method for class 'Opn'
dfourier(coo, nb.h)

## S3 method for class 'list'
dfourier(coo, nb.h)

## S3 method for class 'Coo'
dfourier(coo, nb.h)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dfourier_+3A_coo">coo</code></td>
<td>
<p>a matrix (or a list) of (x; y) coordinates</p>
</td></tr>
<tr><td><code id="dfourier_+3A_nb.h">nb.h</code></td>
<td>
<p>numeric the number of harmonics to calculate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following components:
</p>

<ul>
<li><p> an the A harmonic coefficients
</p>
</li>
<li><p> bn the B harmonic coefficients
</p>
</li>
<li><p> mod the modules of the points
</p>
</li>
<li><p> arg the arguments of the points
</p>
</li></ul>



<h3>Note</h3>

<p>This method has been only poorly tested in Momocs and should be considered as
experimental. Yet improved by a factor 10, this method is still long to execute.
It will be improved in further releases but it should not be so painful right now.
It also explains the progress bar. Shapes should be aligned
before performing the dct transform.
</p>
<p>Silent message and progress bars (if any) with <code>options("verbose"=FALSE)</code>.
</p>


<h3>References</h3>


<ul>
<li><p> Dommergues, C. H., Dommergues, J.-L., &amp; Verrecchia, E. P. (2007).
The Discrete Cosine Transform, a Fourier-related Method for Morphometric Analysis of Open Contours.
<em>Mathematical Geology</em>, 39(8), 749-763. doi:10.1007/s11004-007-9124-6
</p>
</li>
<li><p> Many thanks to Remi Laffont for the translation in R).
</p>
</li></ul>



<h3>See Also</h3>

<p>Other dfourier: 
<code><a href="#topic+dfourier_i">dfourier_i</a>()</code>,
<code><a href="#topic+dfourier_shape">dfourier_shape</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>o &lt;- olea %&gt;% slice(1:5) # for the sake of speed
od &lt;- dfourier(o)
od
op &lt;- PCA(od)
plot(op, 1)

# dfourier and inverse dfourier
o &lt;- olea[1]
o &lt;- coo_bookstein(o)
coo_plot(o)
o.dfourier &lt;- dfourier(o, nb.h=12)
o.dfourier
o.i &lt;- dfourier_i(o.dfourier)
o.i &lt;- coo_bookstein(o.i)
coo_draw(o.i, border='red')

#future calibrate_reconstructions
o &lt;- olea[1]
h.range &lt;- 2:13
coo &lt;- list()
for (i in seq(along=h.range)){
coo[[i]] &lt;- dfourier_i(dfourier(o, nb.h=h.range[i]))}
names(coo) &lt;- paste0('h', h.range)
panel(Opn(coo), borders=col_india(12), names=TRUE)
title('Discrete Cosine Transforms')
</code></pre>

<hr>
<h2 id='dfourier_i'>Investe discrete cosinus transform</h2><span id='topic+dfourier_i'></span>

<h3>Description</h3>

<p>Calculates inverse discrete cosine transforms (see <a href="#topic+dfourier">dfourier</a>), given a list of A and B harmonic coefficients,
typically such as those produced by <a href="#topic+dfourier">dfourier</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfourier_i(df, nb.h, nb.pts = 60)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dfourier_i_+3A_df">df</code></td>
<td>
<p>a list with <code>$A</code> and <code>$B</code> components, containing harmonic coefficients.</p>
</td></tr>
<tr><td><code id="dfourier_i_+3A_nb.h">nb.h</code></td>
<td>
<p>a custom number of harmonics to use</p>
</td></tr>
<tr><td><code id="dfourier_i_+3A_nb.pts">nb.pts</code></td>
<td>
<p>numeric the number of pts for the shape reconstruction</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix of (x; y) coordinates
</p>


<h3>Note</h3>

<p>Only the core functions so far. Will be implemented as an <a href="#topic+Opn">Opn</a> method soon.
</p>


<h3>References</h3>


<ul>
<li><p> Dommergues, C. H., Dommergues, J.-L., &amp; Verrecchia, E. P. (2007).
The Discrete Cosine Transform, a Fourier-related Method for Morphometric Analysis of Open Contours.
<em>Mathematical Geology</em>, 39(8), 749-763. doi:10.1007/s11004-007-9124-6
</p>
</li>
<li><p> Many thanks to Remi Laffont for the translation in R).
</p>
</li></ul>



<h3>See Also</h3>

<p>Other dfourier: 
<code><a href="#topic+dfourier_shape">dfourier_shape</a>()</code>,
<code><a href="#topic+dfourier">dfourier</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># dfourier and inverse dfourier
o &lt;- olea[1]
o &lt;- coo_bookstein(o)
coo_plot(o)
o.dfourier &lt;- dfourier(o, nb.h=12)
o.dfourier
o.i &lt;- dfourier_i(o.dfourier)
o.i &lt;- coo_bookstein(o.i)
coo_draw(o.i, border='red')

o &lt;- olea[1]
h.range &lt;- 2:13
coo &lt;- list()
for (i in seq(along=h.range)){
coo[[i]] &lt;- dfourier_i(dfourier(o, nb.h=h.range[i]))}
names(coo) &lt;- paste0('h', h.range)
panel(Opn(coo), borders=col_india(12), names=TRUE)
title('Discrete Cosine Transforms')
</code></pre>

<hr>
<h2 id='dfourier_shape'>Calculates and draws 'dfourier' shapes</h2><span id='topic+dfourier_shape'></span>

<h3>Description</h3>

<p>Calculates shapes based on 'Discrete cosine transforms' given harmonic coefficients
(see <a href="#topic+dfourier">dfourier</a>) or can generate some random 'dfourier' shapes.
Mainly intended to generate shapes and/or to understand how dfourier works.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfourier_shape(A, B, nb.h, nb.pts = 60, alpha = 2, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dfourier_shape_+3A_a">A</code></td>
<td>
<p>vector of harmonic coefficients</p>
</td></tr>
<tr><td><code id="dfourier_shape_+3A_b">B</code></td>
<td>
<p>vector of harmonic coefficients</p>
</td></tr>
<tr><td><code id="dfourier_shape_+3A_nb.h">nb.h</code></td>
<td>
<p>if <code>A</code> and/or <code>B</code> are not provided,
the number of harmonics to generate</p>
</td></tr>
<tr><td><code id="dfourier_shape_+3A_nb.pts">nb.pts</code></td>
<td>
<p>if <code>A</code> and/or <code>B</code> are not provided,
the number of points to use to reconstruct the shapes</p>
</td></tr>
<tr><td><code id="dfourier_shape_+3A_alpha">alpha</code></td>
<td>
<p>The power coefficient associated with the (usually decreasing)
amplitude of the harmonic coefficients (see <a href="#topic+efourier_shape">efourier_shape</a>)</p>
</td></tr>
<tr><td><code id="dfourier_shape_+3A_plot">plot</code></td>
<td>
<p>logical whether to plot the shape</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of shapes or a plot
</p>


<h3>See Also</h3>

<p>Other dfourier: 
<code><a href="#topic+dfourier_i">dfourier_i</a>()</code>,
<code><a href="#topic+dfourier">dfourier</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># some signatures
panel(coo_align(Opn(replicate(48, dfourier_shape(alpha=0.5, nb.h=6)))))
# some worms
panel(coo_align(Opn(replicate(48, dfourier_shape(alpha=2, nb.h=6)))))
</code></pre>

<hr>
<h2 id='dissolve'>Dissolve Coe objects</h2><span id='topic+dissolve'></span>

<h3>Description</h3>

<p>the opposite of combine, typically used after it. Note that the <code>$fac</code> slot may be wrong since
combine...well combines... this <code>$fac</code>. See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dissolve(x, retain)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dissolve_+3A_x">x</code></td>
<td>
<p>a Coe object</p>
</td></tr>
<tr><td><code id="dissolve_+3A_retain">retain</code></td>
<td>
<p>the partition id to retain. Or their name if the partitions are named
(see x$method) eg after a chop</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bot)
w &lt;- filter(bot, type=="whisky")
b &lt;- filter(bot, type=="beer")
wf &lt;- efourier(w, 10)
bf &lt;- efourier(b, 10)
wbf &lt;- combine(wf, bf)
dissolve(wbf, 1)
dissolve(wbf, 2)

# or using chop (yet combine here makes no sense)
bw &lt;- bot %&gt;% chop(~type) %&gt;% lapply(efourier, 10) %&gt;% combine
bw %&gt;% dissolve(1)
bw %&gt;% dissolve(2)
</code></pre>

<hr>
<h2 id='drawers'>grindr drawers for shape plots</h2><span id='topic+drawers'></span><span id='topic+draw_polygon'></span><span id='topic+draw_outline'></span><span id='topic+draw_outlines'></span><span id='topic+draw_points'></span><span id='topic+draw_landmarks'></span><span id='topic+draw_lines'></span><span id='topic+draw_centroid'></span><span id='topic+draw_curve'></span><span id='topic+draw_curves'></span><span id='topic+draw_firstpoint'></span><span id='topic+draw_axes'></span><span id='topic+draw_ticks'></span><span id='topic+draw_labels'></span><span id='topic+draw_links'></span><span id='topic+draw_title'></span>

<h3>Description</h3>

<p>Useful drawers for building custom
shape plots using the grindr approach. See examples and vignettes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_polygon(
  coo,
  f,
  col = par("fg"),
  fill = NA,
  lwd = 1,
  lty = 1,
  transp = 0,
  pal = pal_qual,
  ...
)

draw_outline(
  coo,
  f,
  col = par("fg"),
  fill = NA,
  lwd = 1,
  lty = 1,
  transp = 0,
  pal = pal_qual,
  ...
)

draw_outlines(
  coo,
  f,
  col = par("fg"),
  fill = NA,
  lwd = 1,
  lty = 1,
  transp = 0,
  pal = pal_qual,
  ...
)

draw_points(
  coo,
  f,
  col = par("fg"),
  cex = 1/2,
  pch = 20,
  transp = 0,
  pal = pal_qual,
  ...
)

draw_landmarks(
  coo,
  f,
  col = par("fg"),
  cex = 1/2,
  pch = 20,
  transp = 0,
  pal = pal_qual,
  ...
)

draw_lines(
  coo,
  f,
  col = par("fg"),
  lwd = 1,
  lty = 1,
  transp = 0,
  pal = pal_qual,
  ...
)

draw_centroid(
  coo,
  f,
  col = par("fg"),
  pch = 3,
  cex = 0.5,
  transp = 0,
  pal = pal_qual,
  ...
)

draw_curve(
  coo,
  f,
  col = par("fg"),
  lwd = 1,
  lty = 1,
  transp = 0,
  pal = pal_qual,
  ...
)

draw_curves(
  coo,
  f,
  col = par("fg"),
  lwd = 1,
  lty = 1,
  transp = 0,
  pal = pal_qual,
  ...
)

draw_firstpoint(
  coo,
  f,
  label = "^",
  col = par("fg"),
  cex = 3/4,
  transp = 0,
  pal = pal_qual,
  ...
)

draw_axes(coo, col = "#999999", lwd = 1/2, ...)

draw_ticks(coo, col = "#333333", cex = 3/4, lwd = 3/4, ...)

draw_labels(coo, labels = 1:nrow(coo), cex = 1/2, d = 1/20, ...)

draw_links(
  coo,
  f,
  links,
  col = "#99999955",
  lwd = 1/2,
  lty = 1,
  transp = 0,
  pal = pal_qual,
  ...
)

draw_title(
  coo,
  main = "",
  sub = "",
  cex = c(1, 3/4),
  font = c(2, 1),
  padding = 1/200,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drawers_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of 2 columns for (x, y) coordinates</p>
</td></tr>
<tr><td><code id="drawers_+3A_f">f</code></td>
<td>
<p>an optionnal factor specification to feed. See examples and vignettes.</p>
</td></tr>
<tr><td><code id="drawers_+3A_col">col</code></td>
<td>
<p>color (hexadecimal) to draw components</p>
</td></tr>
<tr><td><code id="drawers_+3A_fill">fill</code></td>
<td>
<p>color (hexadecimal) to draw components</p>
</td></tr>
<tr><td><code id="drawers_+3A_lwd">lwd</code></td>
<td>
<p>to draw components</p>
</td></tr>
<tr><td><code id="drawers_+3A_lty">lty</code></td>
<td>
<p>to draw components</p>
</td></tr>
<tr><td><code id="drawers_+3A_transp">transp</code></td>
<td>
<p><code>numeric</code> transparency (default:0, min:0, max:1)</p>
</td></tr>
<tr><td><code id="drawers_+3A_pal">pal</code></td>
<td>
<p>a palette to use if no col/border/etc. are provided. See <code style="white-space: pre;">&#8288;[palettes]&#8288;</code></p>
</td></tr>
<tr><td><code id="drawers_+3A_...">...</code></td>
<td>
<p>additional options to feed core functions for each drawer</p>
</td></tr>
<tr><td><code id="drawers_+3A_cex">cex</code></td>
<td>
<p>to draw components ((<code>c(2, 1)</code> by default) for <code>draw_title</code>)</p>
</td></tr>
<tr><td><code id="drawers_+3A_pch">pch</code></td>
<td>
<p>to draw components</p>
</td></tr>
<tr><td><code id="drawers_+3A_label">label</code></td>
<td>
<p>to indicate first point</p>
</td></tr>
<tr><td><code id="drawers_+3A_labels">labels</code></td>
<td>
<p><code>character</code> name of labels to draw (defaut to <code>1:nrow(coo)</code>)</p>
</td></tr>
<tr><td><code id="drawers_+3A_d">d</code></td>
<td>
<p><code>numeric</code> proportion of <code>d(centroid-each_point)</code> to add when centrifugating landmarks</p>
</td></tr>
<tr><td><code id="drawers_+3A_links">links</code></td>
<td>
<p><code>matrix</code> of links to use to draw segments between landmarks. See <code>wings$ldk</code> for an example</p>
</td></tr>
<tr><td><code id="drawers_+3A_main">main</code></td>
<td>
<p><code>character</code> title (empty by default)</p>
</td></tr>
<tr><td><code id="drawers_+3A_sub">sub</code></td>
<td>
<p><code>character</code> subtitle (empty by default)</p>
</td></tr>
<tr><td><code id="drawers_+3A_font">font</code></td>
<td>
<p><code>numeric</code> to feed <a href="graphics.html#topic+text">text</a> (<code>c(2, 1)</code> by default)</p>
</td></tr>
<tr><td><code id="drawers_+3A_padding">padding</code></td>
<td>
<p><code>numeric</code> a fraction of the graphical window (<code>1/200</code> by default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing layer
</p>


<h3>Note</h3>

<p>This approach will (soon) replace <a href="#topic+coo_plot">coo_plot</a> and friends in further versions.
All comments are welcome.
</p>


<h3>See Also</h3>

<p>grindr_layers
</p>
<p>Other grindr: 
<code><a href="#topic+layers_morphospace">layers_morphospace</a></code>,
<code><a href="#topic+layers">layers</a></code>,
<code><a href="#topic+mosaic_engine">mosaic_engine</a>()</code>,
<code><a href="#topic+papers">papers</a></code>,
<code><a href="#topic+pile">pile</a>()</code>,
<code><a href="#topic+plot_LDA">plot_LDA</a>()</code>,
<code><a href="#topic+plot_NMDS">plot_NMDS</a>()</code>,
<code><a href="#topic+plot_PCA">plot_PCA</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot[1] %&gt;% paper_grid() %&gt;% draw_polygon()
olea %&gt;% paper_chess %&gt;% draw_lines(~var)

hearts[240] %&gt;% paper_white() %&gt;% draw_outline() %&gt;%
  coo_sample(24) %&gt;% draw_landmarks %&gt;% draw_labels() %&gt;%
  draw_links(links=replicate(2, sample(1:24, 8)))

bot %&gt;%
    paper_grid() %&gt;%
    draw_outlines() %&gt;%
    draw_title("Alcohol abuse \nis dangerous for health", "Drink responsibly")
</code></pre>

<hr>
<h2 id='ed'>Calculates euclidean distance between two points.</h2><span id='topic+ed'></span>

<h3>Description</h3>

<p><code>ed</code> simply calculates euclidean distance between two points defined by
their (x; y) coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ed(pt1, pt2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ed_+3A_pt1">pt1</code></td>
<td>
<p>(x; y) coordinates of the first point.</p>
</td></tr>
<tr><td><code id="ed_+3A_pt2">pt2</code></td>
<td>
<p>(x; y) coordinates of the second point.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the euclidean distance between the two points.
</p>


<h3>See Also</h3>

<p><a href="#topic+edm">edm</a>, <a href="#topic+edm_nearest">edm_nearest</a>, <a href="stats.html#topic+dist">dist</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ed(c(0,1), c(1,0))
</code></pre>

<hr>
<h2 id='edi'>Calculates euclidean intermediate between two points.</h2><span id='topic+edi'></span>

<h3>Description</h3>

<p><code>edi</code> simply calculates coordinates of a points at the relative
distance <code>r</code> on the <code>pt1-pt2</code> defined by their (x; y) coordinates.
This function is used internally but may be of interest for other analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edi(pt1, pt2, r = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edi_+3A_pt1">pt1</code></td>
<td>
<p><code class="reqn">(x; y)</code> coordinates of the first point.</p>
</td></tr>
<tr><td><code id="edi_+3A_pt2">pt2</code></td>
<td>
<p><code class="reqn">(x; y)</code> coordinates of the second point.</p>
</td></tr>
<tr><td><code id="edi_+3A_r">r</code></td>
<td>
<p>the relative distance from <code>pt1</code> to <code>pt2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns the <code class="reqn">(x; y)</code> interpolated coordinates.
</p>


<h3>See Also</h3>

<p><a href="#topic+ed">ed</a>, <a href="#topic+edm">edm</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>edi(c(0,1), c(1,0), r = 0.5)
</code></pre>

<hr>
<h2 id='edm'>Calculates euclidean distance every pairs of points in two matrices.</h2><span id='topic+edm'></span>

<h3>Description</h3>

<p><code>edm</code> returns the euclidean distances between points <code class="reqn">1 -&gt; n</code> of
two 2-col matrices of the same dimension. This function is used internally
but may be of interest for other analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edm(m1, m2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edm_+3A_m1">m1</code></td>
<td>
<p>The first <code>matrix</code> of coordinates.</p>
</td></tr>
<tr><td><code id="edm_+3A_m2">m2</code></td>
<td>
<p>The second <code>matrix</code> of coordinates.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If one wishes to align two (or more shapes) Procrustes surimposition may
provide a better solution.
</p>


<h3>Value</h3>

<p>Returns a <code>vector</code> of euclidean distances between pairwise
coordinates in the two matrices.
</p>


<h3>See Also</h3>

<p><a href="#topic+ed">ed</a>, <a href="#topic+edm_nearest">edm_nearest</a>, <a href="stats.html#topic+dist">dist</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- matrix(1:10, nc=2)
edm(x, x)
edm(x, x+1)
</code></pre>

<hr>
<h2 id='edm_nearest'>Calculates the shortest euclidean distance found for every point of one
matrix among those of a second.</h2><span id='topic+edm_nearest'></span>

<h3>Description</h3>

<p><code>edm_nearest</code> calculates the shortest euclidean distance found for
every point of one matrix among those of a second. In other words, if
<code>m1, m2</code> have <code>n</code> rows, the result will be the shortest distance
for the first point of <code>m1</code> to any point of <code>m2</code> and so on,
<code>n</code> times. This function is used internally but may be of interest for
other analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edm_nearest(m1, m2, full = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="edm_nearest_+3A_m1">m1</code></td>
<td>
<p>The first <code>list</code> or <code>matrix</code> of coordinates.</p>
</td></tr>
<tr><td><code id="edm_nearest_+3A_m2">m2</code></td>
<td>
<p>The second <code>list</code> or <code>matrix</code> of coordinates.</p>
</td></tr>
<tr><td><code id="edm_nearest_+3A_full">full</code></td>
<td>
<p><code>logical</code>. Whether to returns a condensed version of the
results.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>So far this function is quite time consumming since it performs <code class="reqn"> n
\times n </code> euclidean distance computation.  If one wishes to align two (or
more shapes) Procrustes surimposition may provide a better solution.
</p>


<h3>Value</h3>

<p>If <code>full</code> is <code>TRUE</code>, returns a <code>list</code> with two
components: <code>d</code> which is for every point of <code>m1</code> the shortest
distance found between it and any point in <code>m2</code>, and <code>pos</code> the
(<code>m2</code>) row indices of these points. Otherwise returns <code>d</code> as a
numeric vector of the shortest distances.
</p>


<h3>See Also</h3>

<p><a href="#topic+ed">ed</a>, <a href="#topic+edm">edm</a>, <a href="stats.html#topic+dist">dist</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- matrix(1:10, nc=2)
edm_nearest(x, x+rnorm(10))
edm_nearest(x, x+rnorm(10), full=TRUE)
</code></pre>

<hr>
<h2 id='efourier'>Elliptical Fourier transform (and its normalization)</h2><span id='topic+efourier'></span><span id='topic+efourier.default'></span><span id='topic+efourier.Out'></span><span id='topic+efourier.list'></span><span id='topic+efourier_norm'></span>

<h3>Description</h3>

<p><code>efourier</code> computes Elliptical Fourier Analysis (or Transforms or EFT)
from a matrix (or a list) of (x; y) coordinates. <code>efourier_norm</code> normalizes Fourier coefficients.
Read Details carefully.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>efourier(x, ...)

## Default S3 method:
efourier(x, nb.h, smooth.it = 0, ...)

## S3 method for class 'Out'
efourier(x, nb.h, smooth.it = 0, norm = TRUE, start = FALSE, ...)

## S3 method for class 'list'
efourier(x, ...)

efourier_norm(ef, start = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="efourier_+3A_x">x</code></td>
<td>
<p>A <code>list</code> or a <code>matrix</code> of coordinates or a <code>Out</code> object</p>
</td></tr>
<tr><td><code id="efourier_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
<tr><td><code id="efourier_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to use. If missing, 12 is used on shapes;
99 percent of harmonic power on Out objects, both with messages.</p>
</td></tr>
<tr><td><code id="efourier_+3A_smooth.it">smooth.it</code></td>
<td>
<p><code>integer</code>. The number of smoothing iterations to
perform.</p>
</td></tr>
<tr><td><code id="efourier_+3A_norm">norm</code></td>
<td>
<p>whether to normalize the coefficients using <a href="#topic+efourier_norm">efourier_norm</a></p>
</td></tr>
<tr><td><code id="efourier_+3A_start">start</code></td>
<td>
<p><code>logical</code>. For <code>efourier</code> whether to consider the first point as homologous;
for <code>efourier_norm</code> whether to conserve the position of the first
point of the outline.</p>
</td></tr>
<tr><td><code id="efourier_+3A_ef">ef</code></td>
<td>
<p><code>list</code> with <code>a_n</code>, <code>b_n</code>, <code>c_n</code> and
<code>d_n</code> Fourier coefficients, typically returned by <a href="#topic+efourier">efourier</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the maths behind see the paper in JSS.
</p>
<p>Normalization of coefficients has long been a matter of trouble,
and not only for newcomers. There are two ways of normalizing outlines: the first,
and by far the most used, is to use a &quot;numerical&quot; alignment, directly on the
matrix of coefficients. The coefficients of the first harmonic are consumed
by this process but harmonics of higher rank are normalized in terms of size
and rotation. This is sometimes referred as using the &quot;first ellipse&quot;, as the
harmonics define an ellipse in the plane, and the first one is the mother of all
ellipses, on which all others &quot;roll&quot; along. This approach is really convenient
as it is done easily by most software (if not the only option) and by Momocs too.
It is the default option of <code>efourier</code>.
</p>
<p>But here is the pitfall: if your shapes are prone to bad aligments among all
the first ellipses, this will result in poorly (or even not at all) &quot;homologous&quot; coefficients.
The shapes particularly prone to this are either (at least roughly) circular and/or with a strong
bilateral symmetry. You can try to use <code><a href="#topic+stack">stack</a></code> on the <code><a href="#topic+Coe">Coe</a></code> object
returned by <code>efourier</code>. Also, and perhaps more explicitely, morphospace usually show a mirroring symmetry,
typically visible when calculated in some couple of components (usually the first two).
</p>
<p>If you see these  upside-down (or 180 degrees rotated) shapes on the morphospace,
you should seriously consider aligning your shapes <strong>before</strong> the <a href="#topic+efourier">efourier</a> step,
and performing the latter with <code>norm = FALSE</code>.
</p>
<p>Such a pitfall explains the (quite annoying) message when passing <code>efourier</code> with just the <code>Out</code>.
</p>
<p>You have several options to align your shapes, using control points (or landmarks),
by far the most time consuming (and less reproducible) but possibly the best one too
when alignment is too tricky to automate.
You can also try Procrustes alignment (see <code><a href="#topic+fgProcrustes">fgProcrustes</a></code>) through their calliper
length (see <code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a></code>), etc. You should also make the first
point homologous either with <code><a href="#topic+coo_slide">coo_slide</a></code> or <code><a href="#topic+coo_slidedirection">coo_slidedirection</a></code>
to minimize any subsequent problems.
</p>
<p>I will dedicate (some day) a vignette or a paper to this problem.
</p>


<h3>Value</h3>

<p>For <code>efourier</code>, a list with components: <code>an</code>, <code>bn</code>, <code>cn</code>, <code>dn</code> harmonic coefficients, plus <code>ao</code> and <code>co</code>.
The latter should have been named <code>a0</code> and <code>c0</code> in Claude (2008) but I (intentionnaly) propagated the error.
</p>
<p>For <code>efourier_norm</code>, a list with components: <code>A</code>, <code>B</code>, <code>C</code>, <code>D</code>
for harmonic coefficients, plus <code>size</code>, the magnitude of the semi-major axis of the first
fitting ellipse, <code>theta</code> angle, in radians, between the starting and the semi-major axis
of the first fitting ellipse, <code>psi</code> orientation of the first fitting ellipse, <code>ao</code> and <code>do</code>, same as above,
and <code>lnef</code> that is the concatenation of coefficients.
</p>


<h3>Note</h3>

<p>Directly borrowed for Claude (2008).
</p>
<p>Silent message and progress bars (if any) with <code>options("verbose"=FALSE)</code>.
</p>


<h3>References</h3>

<p>Claude, J. (2008) <em>Morphometrics with R</em>, Use R! series,
Springer 316 pp.
Ferson S, Rohlf FJ, Koehn RK. 1985. Measuring shape variation of
two-dimensional outlines. <em>Systematic Biology</em> <b>34</b>: 59-68.
</p>


<h3>See Also</h3>

<p>Other efourier: 
<code><a href="#topic+efourier_i">efourier_i</a>()</code>,
<code><a href="#topic+efourier_shape">efourier_shape</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># single shape
coo &lt;- bot[1]
coo_plot(coo)
ef &lt;- efourier(coo, 12)
# same but silent
efourier(coo, 12, norm=TRUE)
# inverse EFT
efi &lt;- efourier_i(ef)
coo_draw(efi, border='red', col=NA)

# on Out
bot %&gt;% slice(1:5) %&gt;% efourier
</code></pre>

<hr>
<h2 id='efourier_i'>Inverse elliptical Fourier transform</h2><span id='topic+efourier_i'></span>

<h3>Description</h3>

<p><code>efourier_i</code> uses the inverse elliptical Fourier transformation to
calculate a shape, when given a list with Fourier coefficients, typically
obtained computed with <a href="#topic+efourier">efourier</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>efourier_i(ef, nb.h, nb.pts = 120)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="efourier_i_+3A_ef">ef</code></td>
<td>
<p><code>list</code>. A list containing <code class="reqn">a_n</code>, <code class="reqn">b_n</code>, <code class="reqn">c_n</code> and
<code class="reqn">d_n</code> Fourier coefficients, such as returned by <code>efourier</code>.</p>
</td></tr>
<tr><td><code id="efourier_i_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to use. If not
specified, <code>length(ef$an)</code> is used.</p>
</td></tr>
<tr><td><code id="efourier_i_+3A_nb.pts">nb.pts</code></td>
<td>
<p><code>integer</code>. The number of points to calculate.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <a href="#topic+efourier">efourier</a> for the mathematical background.
</p>


<h3>Value</h3>

<p>A matrix of (x; y) coordinates.
</p>


<h3>Note</h3>

<p>Directly borrowed for Claude (2008), and also called <code>iefourier</code> there.
</p>


<h3>References</h3>

<p>Claude, J. (2008) <em>Morphometrics with R</em>, Use R! series,
Springer 316 pp.
Ferson S, Rohlf FJ, Koehn RK. 1985. Measuring shape variation of
two-dimensional outlines. <em>Systematic Biology</em> <b>34</b>: 59-68.
</p>


<h3>See Also</h3>

<p>Other efourier: 
<code><a href="#topic+efourier_shape">efourier_shape</a>()</code>,
<code><a href="#topic+efourier">efourier</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo &lt;- bot[1]
coo_plot(coo)
ef  &lt;- efourier(coo, 12)
ef
efi &lt;- efourier_i(ef)
coo_draw(efi, border='red', col=NA)
</code></pre>

<hr>
<h2 id='efourier_shape'>Calculates and draw 'efourier' shapes.</h2><span id='topic+efourier_shape'></span>

<h3>Description</h3>

<p><code>efourier_shape</code> calculates a 'Fourier elliptical shape' given Fourier
coefficients (see <code>Details</code>) or can generate some 'efourier' shapes.
Mainly intended to generate shapes and/or to understand how efourier works.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>efourier_shape(an, bn, cn, dn, nb.h, nb.pts = 60, alpha = 2, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="efourier_shape_+3A_an">an</code></td>
<td>
<p><code>numeric</code>. The <code class="reqn">a_n</code> Fourier coefficients on which to
calculate a shape.</p>
</td></tr>
<tr><td><code id="efourier_shape_+3A_bn">bn</code></td>
<td>
<p><code>numeric</code>. The <code class="reqn">b_n</code> Fourier coefficients on which to
calculate a shape.</p>
</td></tr>
<tr><td><code id="efourier_shape_+3A_cn">cn</code></td>
<td>
<p><code>numeric</code>. The <code class="reqn">c_n</code> Fourier coefficients on which to
calculate a shape.</p>
</td></tr>
<tr><td><code id="efourier_shape_+3A_dn">dn</code></td>
<td>
<p><code>numeric</code>. The <code class="reqn">d_n</code> Fourier coefficients on which to
calculate a shape.</p>
</td></tr>
<tr><td><code id="efourier_shape_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to use.</p>
</td></tr>
<tr><td><code id="efourier_shape_+3A_nb.pts">nb.pts</code></td>
<td>
<p><code>integer</code>. The number of points to calculate.</p>
</td></tr>
<tr><td><code id="efourier_shape_+3A_alpha">alpha</code></td>
<td>
<p><code>numeric</code>. The power coefficient associated with the
(usually decreasing) amplitude of the Fourier coefficients (see
<b>Details</b>).</p>
</td></tr>
<tr><td><code id="efourier_shape_+3A_plot">plot</code></td>
<td>
<p><code>logical</code>. Whether to plot or not the shape.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>efourier_shape</code> can be used by specifying <code>nb.h</code> and
<code>alpha</code>. The coefficients are then sampled in an uniform distribution
<code class="reqn">(-\pi ; \pi)</code> and this amplitude is then divided by
<code class="reqn">harmonicrank^alpha</code>. If <code>alpha</code> is lower than 1, consecutive
coefficients will thus increase. See <a href="#topic+efourier">efourier</a> for the mathematical
background.
</p>


<h3>Value</h3>

<p>A list with components:
</p>

<ul>
<li> <p><code>x</code> <code>vector</code> of x-coordinates
</p>
</li>
<li> <p><code>y</code> <code>vector</code> of y-coordinates.
</p>
</li></ul>



<h3>References</h3>

<p>Claude, J. (2008) <em>Morphometrics with R</em>, Use R! series,
Springer 316 pp.
</p>
<p>Ferson S, Rohlf FJ, Koehn RK. 1985. Measuring shape variation of
two-dimensional outlines. <em>Systematic Biology</em> <b>34</b>: 59-68.
</p>


<h3>See Also</h3>

<p>Other efourier: 
<code><a href="#topic+efourier_i">efourier_i</a>()</code>,
<code><a href="#topic+efourier">efourier</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ef &lt;- efourier(bot[1], 24)
efourier_shape(ef$an, ef$bn, ef$cn, ef$dn) # equivalent to efourier_i(ef)
efourier_shape() # is autonomous

panel(Out(a2l(replicate(100,
efourier_shape(nb.h=6, alpha=2.5, plot=FALSE))))) # Bubble family
</code></pre>

<hr>
<h2 id='export'>Exports Coe objects and shapes</h2><span id='topic+export'></span>

<h3>Description</h3>

<p>Writes a <code>.txt</code> or <code>.xls</code> or whatever readable from a single shape,
a  <a href="#topic+Coe">Coe</a>, or a <a href="#topic+PCA">PCA</a> object, along with individual names and <code>$fac</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export(x, file, sep, dec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="export_+3A_x">x</code></td>
<td>
<p>a <code>Coe</code> or <code>PCA</code> object</p>
</td></tr>
<tr><td><code id="export_+3A_file">file</code></td>
<td>
<p>the filenames <code>data.txt</code> by default</p>
</td></tr>
<tr><td><code id="export_+3A_sep">sep</code></td>
<td>
<p>the field separator string to feed <a href="utils.html#topic+write.table">write.table</a>). (default to tab)
tab by default</p>
</td></tr>
<tr><td><code id="export_+3A_dec">dec</code></td>
<td>
<p>the string  to feed <a href="utils.html#topic+write.table">write.table</a>) (default <code>"."</code>)
by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an external file
</p>


<h3>Note</h3>

<p>This is a simple wrapper around <a href="utils.html#topic+write.table">write.table</a>.
</p>
<p>Default parameters will write a <code>.txt</code> file,
readable by foreign programs.
With default parameters, numbers will use dots as decimal points, which is
considered as a character chain in Excel in many countries (locale versions).
This can be solved by using <code>dec=','</code> as in the examples below.
</p>
<p>If you are looking for your file, and did not specified <code>file</code>,
<code>getwd()</code> will help.
</p>
<p>I have to mention that everytime you use this function,
and cowardly run from R to Excel and
do 'statistics' there, an innocent and adorable kitten
is probably murdered somewhere. Use R!
</p>


<h3>See Also</h3>

<p>Other bridges functions: 
<code><a href="#topic+as_df">as_df</a>()</code>,
<code><a href="#topic+bridges">bridges</a></code>,
<code><a href="#topic+complex">complex</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Will write (and remove) files on your working directory!
## Not run: 
bf &lt;- efourier(bot, 6)
# Export Coe (here Fourier coefficients)
export(bf) # data.txt which can be opened by every software including MS Excel

# If you come from a country that uses comma as decimal separator (not recommended, but...)
export(bf, dec=',')
export(bf, file='data.xls', dec=',')

# Export PCA scores
bf %&gt;% PCA %&gt;% export()

# for shapes (matrices)
# export(bot[1], file='bot1.txt')

 # remove these files from your machine
 file.remove("coefficients.txt", "data.xls", "scores.txt")

## End(Not run)
</code></pre>

<hr>
<h2 id='fac_dispatcher'>Brew and serve fac from Momocs object</h2><span id='topic+fac_dispatcher'></span>

<h3>Description</h3>

<p>Ease various specifications for fac specification
when passed to Momocs objects. Intensively used (internally).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fac_dispatcher(x, fac)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fac_dispatcher_+3A_x">x</code></td>
<td>
<p>a Momocs object (any <code>Coo</code>, <code>Coe</code>, <code>PCA</code>, etc.)</p>
</td></tr>
<tr><td><code id="fac_dispatcher_+3A_fac">fac</code></td>
<td>
<p>a specification to extract from <code>fac</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>fac</code> can be:
</p>

<ul>
<li><p> a factor, passed on the fly
</p>
</li>
<li><p> a column id from <code style="white-space: pre;">&#8288;$fac&#8288;</code>
</p>
</li>
<li><p> a column name from <code>fac</code>; if not found, return <code>NULL</code> with a message
</p>
</li>
<li><p> a formula in the form: <code>~column_name</code> (from <code style="white-space: pre;">&#8288;$fac&#8288;</code>, no quotes).
It expresses more in a concise way. Also allows interacting on the fly. See examples.
</p>
</li>
<li><p> a <code>NULL</code> returns a <code>NULL</code>, with a message
</p>
</li></ul>



<h3>Value</h3>

<p>a prepared <code>factor</code> (or a <code>numeric</code>). See examples
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
bot &lt;- mutate(bot, s=rnorm(40), fake=factor(rep(letters[1:4], 10)))

# factor, on the fly
fac_dispatcher(bot, factor(rep(letters[1:4], 10)))

# column id
fac_dispatcher(bot, 1)

# column name
fac_dispatcher(bot, "type")
# same, numeric case
fac_dispatcher(bot, "s")

# formula interface
fac_dispatcher(bot, ~type)

# formula interface + interaction on the fly
fac_dispatcher(bot, ~type+fake)

# when passing NULL or non existing column
fac_dispatcher(42, NULL)
fac_dispatcher(bot, "loser")
</code></pre>

<hr>
<h2 id='fgProcrustes'>Full Generalized Procrustes alignment between shapes</h2><span id='topic+fgProcrustes'></span>

<h3>Description</h3>

<p>Directly borrowed from Claude (2008), called there the <code>fgpa2</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fgProcrustes(x, tol, coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fgProcrustes_+3A_x">x</code></td>
<td>
<p>an array, a list of configurations, or an <a href="#topic+Out">Out</a>, <a href="#topic+Opn">Opn</a> or <a href="#topic+Ldk">Ldk</a> object</p>
</td></tr>
<tr><td><code id="fgProcrustes_+3A_tol">tol</code></td>
<td>
<p>numeric when to stop iterations</p>
</td></tr>
<tr><td><code id="fgProcrustes_+3A_coo">coo</code></td>
<td>
<p>logical, when working on <code>Out</code> or <code>Opn</code>, whether to use <code>$coo</code> rather than <code>$ldk</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>If performed on an <a href="#topic+Out">Out</a> or an <a href="#topic+Opn">Opn</a> object, will try to use the <code>$ldk</code> slot,
if landmarks have been previousy defined, then (with a message) on the <code>$coo</code> slot,
but in that case, all shapes must have the same number of coordinates (<a href="#topic+coo_sample">coo_sample</a> may help).
</p>


<h3>Value</h3>

<p>a list with components:
</p>

<ul>
<li> <p><code>rotated</code> array of superimposed configurations
</p>
</li>
<li> <p><code>iterationnumber</code> number of iterations
</p>
</li>
<li> <p><code>Q</code> convergence criterion
</p>
</li>
<li> <p><code>Qi</code> full list of Q
</p>
</li>
<li> <p><code>Qd</code> difference between successive Q
</p>
</li>
<li> <p><code>interproc.dist</code> minimal sum of squared norms of pairwise differences between
all shapes in the superimposed sample
</p>
</li>
<li> <p><code>mshape</code> mean shape configuration
</p>
</li>
<li> <p><code>cent.size</code> vector of centroid sizes.
</p>
</li></ul>
<p> or an <a href="#topic+Out">Out</a>, <a href="#topic+Opn">Opn</a> or an <a href="#topic+Ldk">Ldk</a> object.
</p>


<h3>Note</h3>

<p>Slightly less optimized than procGPA in the shapes package (~20% on my machine).
Will be optimized when performance will be the last thing to improve!
Silent message and progress bars (if any) with <code>options("verbose"=FALSE)</code>.
</p>


<h3>References</h3>

<p>Claude, J. (2008). Morphometrics with R. Analysis (p. 316). Springer.
</p>


<h3>See Also</h3>

<p>Other procrustes functions: 
<code><a href="#topic+fProcrustes">fProcrustes</a>()</code>,
<code><a href="#topic+fgsProcrustes">fgsProcrustes</a>()</code>,
<code><a href="#topic+pProcrustes">pProcrustes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># on Ldk
w &lt;- wings %&gt;% slice(1:5) # for the sake of speed
stack(w)
fgProcrustes(w, tol=0.1) %&gt;% stack()

# on Out
h &lt;- hearts %&gt;% slice(1:5) # for the sake of speed
stack(h)
fgProcrustes(h) %&gt;%  stack()
</code></pre>

<hr>
<h2 id='fgsProcrustes'>Full Generalized Procrustes alignment between shapes with sliding landmarks</h2><span id='topic+fgsProcrustes'></span>

<h3>Description</h3>

<p>Directly wrapped around <code>geomorph::gpagen</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fgsProcrustes(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fgsProcrustes_+3A_x">x</code></td>
<td>
<p>Ldk object with some <code>$slidings</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list
</p>


<h3>Note</h3>

<p>Landmarks methods are the less tested in Momocs. Keep in mind that some features
are still experimental and that your help is welcome.
</p>


<h3>Source</h3>

<p>See <code>?gpagen</code> in <code>geomorph</code> package
</p>


<h3>See Also</h3>

<p>Other procrustes functions: 
<code><a href="#topic+fProcrustes">fProcrustes</a>()</code>,
<code><a href="#topic+fgProcrustes">fgProcrustes</a>()</code>,
<code><a href="#topic+pProcrustes">pProcrustes</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ch &lt;- chaff %&gt;% slice(1:5) # for the sake of speed
chaffp &lt;- fgsProcrustes(ch)
chaffp
chaffp %&gt;% PCA() %&gt;% plot("taxa")
</code></pre>

<hr>
<h2 id='filter'>Subset based on conditions</h2><span id='topic+filter'></span>

<h3>Description</h3>

<p>Return shapes with matching conditions, from the <code>$fac</code>. See examples and <code>?dplyr::filter</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter(.data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_+3A_.data">.data</code></td>
<td>
<p>a <code>Coo</code>, <code>Coe</code>, <code>PCA</code> object</p>
</td></tr>
<tr><td><code id="filter_+3A_...">...</code></td>
<td>
<p>logical conditions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>dplyr verbs are maintained. You should probbaly not filter on PCA objects.
The latter are calculated using all individuals and filtering may lead to false conclusions.
If you want to highlith some individuals, see examples in <a href="#topic+plot_PCA">plot_PCA</a>.
</p>


<h3>Value</h3>

<p>a Momocs object of the same class.
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>olea
# we retain on dorsal views
filter(olea, view=="VD")
# only dorsal views and Aglan+PicMa varieties
filter(olea, view=="VD", var %in% c("Aglan", "PicMa"))
# we create an id column and retain the 120 first shapes
olea %&gt;% mutate(id=1:length(olea)) %&gt;% filter(id &gt; 120)
</code></pre>

<hr>
<h2 id='flip_PCaxes'>Flips PCA axes</h2><span id='topic+flip_PCaxes'></span>

<h3>Description</h3>

<p>Simply multiply by -1, corresponding scores and rotation vectors for PCA objects.
PC orientation being arbitrary, this may help to have a better display.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flip_PCaxes(x, axs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="flip_PCaxes_+3A_x">x</code></td>
<td>
<p>a PCA object</p>
</td></tr>
<tr><td><code id="flip_PCaxes_+3A_axs">axs</code></td>
<td>
<p>numeric which PC(s) to flip</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>bp &lt;- bot %&gt;% efourier(6) %&gt;% PCA
bp %&gt;% plot
bp %&gt;% flip_PCaxes(1) %&gt;% plot()
</code></pre>

<hr>
<h2 id='flower'>Data: Measurement of iris flowers</h2><span id='topic+flower'></span>

<h3>Description</h3>

<p>Data: Measurement of iris flowers
</p>


<h3>Format</h3>

<p>A <code>TraCoe</code> object with 150 measurements of 4 variables
(petal + sepal) x (length x width) on 3 species of iris. This dataset is
the classical <a href="datasets.html#topic+iris">iris</a> formatted for Momocs.
</p>


<h3>Source</h3>

<p>see <a href="datasets.html#topic+iris">iris</a>
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='fProcrustes'>Full Procrustes alignment between two shapes</h2><span id='topic+fProcrustes'></span>

<h3>Description</h3>

<p>Directly borrowed from Claude (2008), called there the <code>fPsup</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fProcrustes(coo1, coo2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fProcrustes_+3A_coo1">coo1</code></td>
<td>
<p>configuration matrix to be superimposed onto the centered preshape of coo2.</p>
</td></tr>
<tr><td><code id="fProcrustes_+3A_coo2">coo2</code></td>
<td>
<p>reference configuration matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with components:
</p>

<ul>
<li> <p><code>coo1</code> superimposed centered preshape of coo1 onto the centered preshape of coo2
</p>
</li>
<li> <p><code>coo2</code> centered preshape of coo2
</p>
</li>
<li> <p><code>rotation</code> rotation matrix
</p>
</li>
<li> <p><code>scale</code> scale parameter
</p>
</li>
<li> <p><code>DF</code> full Procrustes distance between coo1 and coo2.
</p>
</li></ul>



<h3>References</h3>

<p>Claude, J. (2008). Morphometrics with R. Analysis (p. 316). Springer.
</p>


<h3>See Also</h3>

<p>Other procrustes functions: 
<code><a href="#topic+fgProcrustes">fgProcrustes</a>()</code>,
<code><a href="#topic+fgsProcrustes">fgsProcrustes</a>()</code>,
<code><a href="#topic+pProcrustes">pProcrustes</a>()</code>
</p>

<hr>
<h2 id='get_chull_area'>Calculates convex hull area/volume of PCA scores</h2><span id='topic+get_chull_area'></span><span id='topic+get_chull_volume'></span>

<h3>Description</h3>

<p>May be useful to compare shape diversity. Expressed in PCA units that should
only be compared within the same PCA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_chull_area(x, fac, xax = 1, yax = 2)

get_chull_volume(x, fac, xax = 1, yax = 2, zax = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_chull_area_+3A_x">x</code></td>
<td>
<p>a PCA object</p>
</td></tr>
<tr><td><code id="get_chull_area_+3A_fac">fac</code></td>
<td>
<p>(optionnal) column name or ID from the $fac slot.</p>
</td></tr>
<tr><td><code id="get_chull_area_+3A_xax">xax</code></td>
<td>
<p>the first PC axis to use (1 by default)</p>
</td></tr>
<tr><td><code id="get_chull_area_+3A_yax">yax</code></td>
<td>
<p>the second PC axis (2 by default)</p>
</td></tr>
<tr><td><code id="get_chull_area_+3A_zax">zax</code></td>
<td>
<p>the third PC axis (3 by default only for volume)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>get_chull_area is calculated using <a href="#topic+coo_chull">coo_chull</a> followed by <a href="#topic+coo_area">coo_area</a>;
get_chull_volume is calculated using geometry::convexhulln
</p>


<h3>Value</h3>

<p>If fac is not provided global area/volume is returned; otherwise a named
list for every level of fac
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bp &lt;- PCA(efourier(bot, 12))
get_chull_area(bp)
get_chull_area(bp, 1)

get_chull_volume(bp)
get_chull_volume(bp, 1)
</code></pre>

<hr>
<h2 id='get_ldk'>Retrieves landmarks coordinates</h2><span id='topic+get_ldk'></span>

<h3>Description</h3>

<p>See Details for the different behaviors implemented.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ldk(Coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_ldk_+3A_coo">Coo</code></td>
<td>
<p>an Out, Opn or Ldk object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Different behaviors depending on the class of the object:
</p>

<ul>
<li> <p><a href="#topic+Ldk">Ldk</a>: retrieves landmarks.
</p>
</li>
<li><p> Ldk with slidings defined: retrieves only the fixed landmarks, not the sliding ones.
See also <a href="#topic+get_slidings">get_slidings</a>.
</p>
</li>
<li> <p><a href="#topic+Out">Out</a> landmarks from <code>$ldk</code> and <code>$coo</code>, if any.
</p>
</li>
<li> <p><a href="#topic+Opn">Opn</a>: same as above.
</p>
</li></ul>



<h3>Value</h3>

<p>a list of shapes
</p>


<h3>See Also</h3>

<p>Other ldk/slidings methods: 
<code><a href="#topic+add_ldk">add_ldk</a>()</code>,
<code><a href="#topic+def_ldk">def_ldk</a>()</code>,
<code><a href="#topic+def_slidings">def_slidings</a>()</code>,
<code><a href="#topic+get_slidings">get_slidings</a>()</code>,
<code><a href="#topic+rearrange_ldk">rearrange_ldk</a>()</code>,
<code><a href="#topic+slidings_scheme">slidings_scheme</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Out example
ldk.h &lt;- get_ldk(hearts)
stack(Ldk(ldk.h))

# on Ldk (no slidings)
get_ldk(wings) # equivalent to wings$coo

# on Ldk (slidings)
get_ldk(chaff)
get_ldk(chaff) %&gt;% Ldk %&gt;% fgProcrustes(tol=0.1) %&gt;% stack
</code></pre>

<hr>
<h2 id='get_pairs'>Get paired individual on a Coe, PCA or LDA objects</h2><span id='topic+get_pairs'></span>

<h3>Description</h3>

<p>If you have paired individuals, i.e. before and after a treatment or for repeated measures,
and if you have coded coded it into <code>$fac</code>, this methods allows you to retrieve the corresponding PC/LD scores,
or coefficients for <a href="#topic+Coe">Coe</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pairs(x, fac, range)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pairs_+3A_x">x</code></td>
<td>
<p>any <a href="#topic+Coe">Coe</a>, <a href="#topic+PCA">PCA</a> of <a href="#topic+LDA">LDA</a> object.</p>
</td></tr>
<tr><td><code id="get_pairs_+3A_fac">fac</code></td>
<td>
<p>factor or column name or id corresponding to the pairing factor.</p>
</td></tr>
<tr><td><code id="get_pairs_+3A_range">range</code></td>
<td>
<p>numeric the range of coefficients for <code>Coe</code>, or PC (LD) axes on which to return scores.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with components <code>x1</code> all coefficients/scores corresponding to the
first level of the <code>fac</code> provided; <code>x2</code> same thing for the second level;
<code>fac</code> the corresponding <code>fac</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot2 &lt;- bot1 &lt;- coo_scale(coo_center(coo_sample(bot, 60)))
bot1$fac$session &lt;- factor(rep("session1", 40))
# we simulate an measurement error
bot2 &lt;- coo_jitter(bot1, amount=0.01)
bot2$fac$session &lt;- factor(rep("session2", 40))
botc &lt;- combine(bot1, bot2)
botcf &lt;- efourier(botc, 12)

# we gonna plot the PCA with the two measurement sessions and the two types
botcp &lt;- PCA(botcf)
plot(botcp, "type", col=col_summer(2), pch=rep(c(1, 20), each=40), eigen=FALSE)
bot.pairs &lt;- get_pairs(botcp, fac = "session", range=1:2)
segments(bot.pairs$session1[, 1], bot.pairs$session1[, 2],
       bot.pairs$session2[, 1], bot.pairs$session2[, 2],
       col=col_summer(2)[bot.pairs$fac$type])


</code></pre>

<hr>
<h2 id='get_slidings'>Extracts sliding landmarks coordinates</h2><span id='topic+get_slidings'></span>

<h3>Description</h3>

<p>From an <a href="#topic+Ldk">Ldk</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_slidings(Coo, partition)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_slidings_+3A_coo">Coo</code></td>
<td>
<p>an Ldk object</p>
</td></tr>
<tr><td><code id="get_slidings_+3A_partition">partition</code></td>
<td>
<p>numeric which one(s) to get.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of list(s) of coordinates.
</p>


<h3>See Also</h3>

<p>Other ldk/slidings methods: 
<code><a href="#topic+add_ldk">add_ldk</a>()</code>,
<code><a href="#topic+def_ldk">def_ldk</a>()</code>,
<code><a href="#topic+def_slidings">def_slidings</a>()</code>,
<code><a href="#topic+get_ldk">get_ldk</a>()</code>,
<code><a href="#topic+rearrange_ldk">rearrange_ldk</a>()</code>,
<code><a href="#topic+slidings_scheme">slidings_scheme</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># for each example below a list with partition containign shapes is returned
# extracts the first partition
get_slidings(chaff, 1) %&gt;% names()
# the first and the fourth
get_slidings(chaff, c(1, 4)) %&gt;%  names()
# all of them
get_slidings(chaff) %&gt;%  names
# here we want to see it
get_slidings(chaff, 1)[[1]] %&gt;%  Ldk %&gt;% stack
</code></pre>

<hr>
<h2 id='harm_pow'>Calculates harmonic power given a list from e/t/rfourier</h2><span id='topic+harm_pow'></span>

<h3>Description</h3>

<p>Given a list with <code>an, bn (and eventually cn and dn)</code>, returns the
harmonic power.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>harm_pow(xf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="harm_pow_+3A_xf">xf</code></td>
<td>
<p>A list with an, bn (and cn, dn) components, typically from a
e/r/tfourier passed on coo_</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code>vector</code> of harmonic power
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ef &lt;- efourier(bot[1], 24)
rf &lt;- efourier(bot[1], 24)
harm_pow(ef)
harm_pow(rf)

plot(cumsum(harm_pow(ef)[-1]), type='o',
  main='Cumulated harmonic power without the first harmonic',
  ylab='Cumulated harmonic power', xlab='Harmonic rank')

</code></pre>

<hr>
<h2 id='hcontrib'>Harmonic contribution to shape</h2><span id='topic+hcontrib'></span><span id='topic+hcontrib.OutCoe'></span>

<h3>Description</h3>

<p>Calculates contribution of harmonics to shape. The amplitude of every coefficients
of a given harmonic is multiplied by the coefficients provided and the resulting
shapes are reconstructed and plotted. Naturally, only works on Fourier-based methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hcontrib(Coe, ...)

## S3 method for class 'OutCoe'
hcontrib(
  Coe,
  id,
  harm.r,
  amp.r = c(0, 0.5, 1, 2, 5, 10),
  main = "Harmonic contribution to shape",
  xlab = "Harmonic rank",
  ylab = "Amplification factor",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hcontrib_+3A_coe">Coe</code></td>
<td>
<p>a <code><a href="#topic+Coe">Coe</a></code> object (either <code>OutCoe</code> or (soon) <code>OpnCoe</code>)</p>
</td></tr>
<tr><td><code id="hcontrib_+3A_...">...</code></td>
<td>
<p>additional parameter to pass to <code><a href="#topic+coo_draw">coo_draw</a></code></p>
</td></tr>
<tr><td><code id="hcontrib_+3A_id">id</code></td>
<td>
<p>the id of a particular shape, otherwise working on the meanshape</p>
</td></tr>
<tr><td><code id="hcontrib_+3A_harm.r">harm.r</code></td>
<td>
<p>range of harmonics on which to explore contributions</p>
</td></tr>
<tr><td><code id="hcontrib_+3A_amp.r">amp.r</code></td>
<td>
<p>a vector of numeric for multiplying coefficients</p>
</td></tr>
<tr><td><code id="hcontrib_+3A_main">main</code></td>
<td>
<p>a title for the plot</p>
</td></tr>
<tr><td><code id="hcontrib_+3A_xlab">xlab</code></td>
<td>
<p>a title for the x-axis</p>
</td></tr>
<tr><td><code id="hcontrib_+3A_ylab">ylab</code></td>
<td>
<p>a title for the y-axis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>See Also</h3>

<p>Other Coe_graphics: 
<code><a href="#topic+boxplot.OutCoe">boxplot.OutCoe</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bot)
bot.f &lt;- efourier(bot, 12)
hcontrib(bot.f)
hcontrib(bot.f, harm.r=3:10, amp.r=1:8, col="grey20",
   main="A huge panel")
</code></pre>

<hr>
<h2 id='hearts'>Data: Outline coordinates of hand-drawn hearts</h2><span id='topic+hearts'></span>

<h3>Description</h3>

<p>Data: Outline coordinates of hand-drawn hearts
</p>


<h3>Format</h3>

<p>A <a href="#topic+Out">Out</a> object with the outline coordinates of 240 hand-drawn hearts
by 8 different persons, with 4 landmarks.
</p>


<h3>Source</h3>

<p>We thank the fellows of the Ecology Department of the French Institute
of Pondicherry that drawn the hearts, that then have been smoothed, scaled, centered,
and downsampled to 80 coordinates per outline.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='img_plot'>Plots a .jpg image</h2><span id='topic+img_plot'></span><span id='topic+img_plot0'></span>

<h3>Description</h3>

<p>A very simple image plotter. If provided with a path,
reads the .jpg and plots it. If not provided with an imagematrix, will
ask you to choose interactively a <code>.jpeg</code> image.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>img_plot(img)

img_plot0(img)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="img_plot_+3A_img">img</code></td>
<td>
<p>a matrix of an image, such as those obtained with <a href="jpeg.html#topic+readJPEG">readJPEG</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>img_plot</code> is used in import functions such as <a href="#topic+import_jpg1">import_jpg1</a>;
<code>img_plot0</code> does the same job but preserves the <code>par</code> and plots axes.
</p>


<h3>Value</h3>

<p>a plot
</p>

<hr>
<h2 id='import_Conte'>Extract outlines coordinates from an image silhouette</h2><span id='topic+import_Conte'></span>

<h3>Description</h3>

<p>Provided with an image 'mask' (i.e. black pixels on a white background),
and a point form where to start the algorithm, returns the (x; y) coordinates of its outline.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_Conte(img, x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_Conte_+3A_img">img</code></td>
<td>
<p>a matrix of a binary image mask.</p>
</td></tr>
<tr><td><code id="import_Conte_+3A_x">x</code></td>
<td>
<p>numeric the (x; y) coordinates of a starting point within the shape.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Used internally by <a href="#topic+import_jpg1">import_jpg1</a> but may be useful for other purposes.
</p>


<h3>Value</h3>

<p>a matrix the (x; y) coordinates of the outline points.
</p>


<h3>Note</h3>

<p>Note this function will be deprecated from Momocs
when <code>Momacs</code> and <code>Momit</code>
will be fully operationnal.
</p>
<p>If you have an image with more than a single shape, then
you may want to try <code>imager::highlight</code> function. Momocs may use this at some point.
</p>


<h3>References</h3>


<ul>
<li><p> The original algorithm is due to: Pavlidis, T. (1982). <em>Algorithms
for graphics and image processing</em>. Computer science press.
</p>
</li>
<li><p> is detailed in: Rohlf, F. J. (1990). An overview of image processing and
analysis techniques for morphometrics. In <em>Proceedings of the Michigan Morphometrics Workshop</em>. Special Publication No. 2 (pp. 47-60). University of Michigan Museum of Zoology: Ann Arbor.
</p>
</li>
<li><p> and translated in R by: Claude, J. (2008). <em>Morphometrics with R</em>. (p. 316). Springer.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other import functions: 
<code><a href="#topic+import_StereoMorph_curve1">import_StereoMorph_curve1</a>()</code>,
<code><a href="#topic+import_jpg1">import_jpg1</a>()</code>,
<code><a href="#topic+import_jpg">import_jpg</a>()</code>,
<code><a href="#topic+import_tps">import_tps</a>()</code>,
<code><a href="#topic+import_txt">import_txt</a>()</code>,
<code><a href="#topic+pix2chc">pix2chc</a>()</code>
</p>

<hr>
<h2 id='import_jpg'>Extract outline coordinates from multiple .jpg files</h2><span id='topic+import_jpg'></span>

<h3>Description</h3>

<p>This function is used to import outline coordinates and is built around
<a href="#topic+import_jpg1">import_jpg1</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_jpg(
  jpg.paths = .lf.auto(),
  auto.notcentered = TRUE,
  fun.notcentered = NULL,
  threshold = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_jpg_+3A_jpg.paths">jpg.paths</code></td>
<td>
<p>a vector of paths corresponding to the .jpg files to import. If not
provided (or <code>NULL</code>), switches to the automatic version. See Details below.</p>
</td></tr>
<tr><td><code id="import_jpg_+3A_auto.notcentered">auto.notcentered</code></td>
<td>
<p>logical if TRUE random locations will be used until.
one of them is (assumed) to be within the shape (because of a black pixel);
if FALSE a <a href="graphics.html#topic+locator">locator</a> will be called, and you will have to click on a
point within the shape.</p>
</td></tr>
<tr><td><code id="import_jpg_+3A_fun.notcentered">fun.notcentered</code></td>
<td>
<p>NULL by default. Is your shapes are not centered and if a random pick of
a black pixel is not satisfactory. See <a href="#topic+import_jpg1">import_jpg1</a> help and examples.</p>
</td></tr>
<tr><td><code id="import_jpg_+3A_threshold">threshold</code></td>
<td>
<p>the threshold value use to binarize the images. Above, pixels
are turned to 1, below to 0.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>see <a href="#topic+import_jpg1">import_jpg1</a> for important informations about how the outlines are extracted,
and <a href="#topic+import_Conte">import_Conte</a> for the algorithm itself.
</p>
<p>If <code>jpg.paths</code> is not provided (or <code>NULL</code>), you will have to select any <code>.jpg</code>
file in the folder that contains all your files. All the outlines should be imported then.
</p>


<h3>Value</h3>

<p>a list of matrices of (x; y) coordinates that can be passed to <a href="#topic+Out">Out</a>
</p>


<h3>Note</h3>

<p>Note this function will be deprecated from Momocs
when <code>Momacs</code> and <code>Momit</code>
will be fully operationnal.
</p>
<p>Silent message and progress bars (if any) with <code>options("verbose"=FALSE)</code>.
</p>


<h3>See Also</h3>

<p>Other import functions: 
<code><a href="#topic+import_Conte">import_Conte</a>()</code>,
<code><a href="#topic+import_StereoMorph_curve1">import_StereoMorph_curve1</a>()</code>,
<code><a href="#topic+import_jpg1">import_jpg1</a>()</code>,
<code><a href="#topic+import_tps">import_tps</a>()</code>,
<code><a href="#topic+import_txt">import_txt</a>()</code>,
<code><a href="#topic+pix2chc">pix2chc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

lf &lt;- list.files('/foo/jpegs', full.names=TRUE)
coo &lt;- import_jpg(lf)
Out(coo)

coo &lt;- import_jpg()

</code></pre>

<hr>
<h2 id='import_jpg1'>Extract outline coordinates from a single .jpg file</h2><span id='topic+import_jpg1'></span>

<h3>Description</h3>

<p>Used to import outline coordinates from .jpg files. This function is used for
single images and is wrapped by <a href="#topic+import_jpg">import_jpg</a>. It relies itself on <a href="#topic+import_Conte">import_Conte</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_jpg1(
  jpg.path,
  auto.notcentered = TRUE,
  fun.notcentered = NULL,
  threshold = 0.5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_jpg1_+3A_jpg.path">jpg.path</code></td>
<td>
<p>vector of paths corresponding to the .jpg files to import, such as
those obtained with <a href="base.html#topic+list.files">list.files</a>.</p>
</td></tr>
<tr><td><code id="import_jpg1_+3A_auto.notcentered">auto.notcentered</code></td>
<td>
<p>logical if TRUE random locations will be used until
one of them is (assumed) to be within the shape (because it corresponds to a black pixel) and only if
the middle point is not black;
if FALSE a <a href="graphics.html#topic+locator">locator</a> will be called, and you will have to click on a
point within the shape.</p>
</td></tr>
<tr><td><code id="import_jpg1_+3A_fun.notcentered">fun.notcentered</code></td>
<td>
<p>NULL by default but can accept a function that, when passed with an imagematrix and returns
a numeric of length two that corresponds to a starting point on the imagematrix for the Conte
algorithm. A <code>while</code> instruction wraps it, so the function may be wrong in proposing this
starting position. See the examples below for a quick example.</p>
</td></tr>
<tr><td><code id="import_jpg1_+3A_threshold">threshold</code></td>
<td>
<p>the threshold value use to binarize the images. Above, pixels
are turned to 1, below to 0.</p>
</td></tr>
<tr><td><code id="import_jpg1_+3A_...">...</code></td>
<td>
<p>arguments to be passed to <a href="utils.html#topic+read.table">read.table</a>, eg. 'skip', 'dec', etc.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>jpegs can be provided either as RVB or as 8-bit greylevels or monochrome.
The function binarizes pixels values using the 'threshold' argument. It will try to start to
apply the <a href="#topic+import_Conte">import_Conte</a> algorithm from the center of
the image and 'looking' downwards for the first black/white 'frontier' in
the pixels. This point will be the first of the outlines. The latter may be useful
if you align manually the images and if you want to retain this information
in the consequent morphometric analyses.
</p>
<p>If the point at the center of the
image is not within the shape, i.e. is 'white' you have two choices defined by
the 'auto.notcentered' argument. If it's TRUE, some random starting points
will be tried until on of them is 'black' and within the shape; if FALSE
you will be asked to click on a point within the shape.
</p>
<p>If some pixels on the borders are not white, this functions adds a 2-pixel
border of white pixels; otherwise <a href="#topic+import_Conte">import_Conte</a> would fail and return an error.
</p>
<p>Finally, remember that if the images are not in your working directory,
<a href="base.html#topic+list.files">list.files</a> must be called with the argument <code>full.names=TRUE</code>!
</p>
<p>Note that the use of the <code>fun.notcentered</code> argument will probably leads to serious headaches
and will probably imply the dissection of these functions: <a href="#topic+import_Conte">import_Conte</a>, <a href="#topic+img_plot">img_plot</a> and
<code>import_jpg</code> itself
</p>


<h3>Value</h3>

<p>a matrix of (x; y) coordinates that can be passed to Out
</p>


<h3>Note</h3>

<p>Note this function will be deprecated from Momocs
when <code>Momacs</code> and <code>Momit</code>
will be fully operationnal.
</p>


<h3>See Also</h3>

<p><a href="#topic+import_jpg">import_jpg</a>, <a href="#topic+import_Conte">import_Conte</a>, <a href="#topic+import_txt">import_txt</a>, <a href="#topic+lf_structure">lf_structure</a>.
See also Momocs' vignettes for data import.
</p>
<p>Other import functions: 
<code><a href="#topic+import_Conte">import_Conte</a>()</code>,
<code><a href="#topic+import_StereoMorph_curve1">import_StereoMorph_curve1</a>()</code>,
<code><a href="#topic+import_jpg">import_jpg</a>()</code>,
<code><a href="#topic+import_tps">import_tps</a>()</code>,
<code><a href="#topic+import_txt">import_txt</a>()</code>,
<code><a href="#topic+pix2chc">pix2chc</a>()</code>
</p>

<hr>
<h2 id='import_StereoMorph_curve1'>Import files creates by StereoMorph into Momocs</h2><span id='topic+import_StereoMorph_curve1'></span><span id='topic+import_StereoMorph_curve'></span><span id='topic+import_StereoMorph_ldk1'></span><span id='topic+import_StereoMorph_ldk'></span>

<h3>Description</h3>

<p>Helps to read <code>.txt</code> files created by StereoMorph into (x; y) coordinates
or Momocs objects. Can be applied to 'curves' or 'ldk' text files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_StereoMorph_curve1(path)

import_StereoMorph_curve(path, names)

import_StereoMorph_ldk1(path)

import_StereoMorph_ldk(path, names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_StereoMorph_curve1_+3A_path">path</code></td>
<td>
<p>toward a single file or a folder containing <code>.txt</code> files produced by StereoMorph</p>
</td></tr>
<tr><td><code id="import_StereoMorph_curve1_+3A_names">names</code></td>
<td>
<p>to feed <a href="#topic+lf_structure">lf_structure</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>*1 functions import a single <code>.txt</code> file. Their counterpart (no '1')
work when path indicates the folder, i.e. 'curves' or 'ldk'. They then return a list
of <a href="#topic+Opn">Opn</a> or <a href="#topic+Ldk">Ldk</a> objects, respectively. Please do not hesitate to contact me
should you have a particular case or need something.
</p>


<h3>Value</h3>

<p>a list of class Coo
</p>


<h3>Note</h3>

<p>Note this function will be deprecated from Momocs
when <code>Momacs</code> and <code>Momit</code>
will be fully operationnal.
</p>


<h3>See Also</h3>

<p>Other import functions: 
<code><a href="#topic+import_Conte">import_Conte</a>()</code>,
<code><a href="#topic+import_jpg1">import_jpg1</a>()</code>,
<code><a href="#topic+import_jpg">import_jpg</a>()</code>,
<code><a href="#topic+import_tps">import_tps</a>()</code>,
<code><a href="#topic+import_txt">import_txt</a>()</code>,
<code><a href="#topic+pix2chc">pix2chc</a>()</code>
</p>
<p>Other import functions: 
<code><a href="#topic+import_Conte">import_Conte</a>()</code>,
<code><a href="#topic+import_jpg1">import_jpg1</a>()</code>,
<code><a href="#topic+import_jpg">import_jpg</a>()</code>,
<code><a href="#topic+import_tps">import_tps</a>()</code>,
<code><a href="#topic+import_txt">import_txt</a>()</code>,
<code><a href="#topic+pix2chc">pix2chc</a>()</code>
</p>
<p>Other import functions: 
<code><a href="#topic+import_Conte">import_Conte</a>()</code>,
<code><a href="#topic+import_jpg1">import_jpg1</a>()</code>,
<code><a href="#topic+import_jpg">import_jpg</a>()</code>,
<code><a href="#topic+import_tps">import_tps</a>()</code>,
<code><a href="#topic+import_txt">import_txt</a>()</code>,
<code><a href="#topic+pix2chc">pix2chc</a>()</code>
</p>
<p>Other import functions: 
<code><a href="#topic+import_Conte">import_Conte</a>()</code>,
<code><a href="#topic+import_jpg1">import_jpg1</a>()</code>,
<code><a href="#topic+import_jpg">import_jpg</a>()</code>,
<code><a href="#topic+import_tps">import_tps</a>()</code>,
<code><a href="#topic+import_txt">import_txt</a>()</code>,
<code><a href="#topic+pix2chc">pix2chc</a>()</code>
</p>

<hr>
<h2 id='import_tps'>Import a tps file</h2><span id='topic+import_tps'></span><span id='topic+tps2coo'></span>

<h3>Description</h3>

<p>And returns a list of coordinates, curves, scale
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_tps(tps.path, curves = TRUE)

tps2coo(tps, curves = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_tps_+3A_tps.path">tps.path</code></td>
<td>
<p>lines, typically from <a href="base.html#topic+readLines">readLines</a>, describing a single shape in tps-like format.
You will need to manually build your <code>Coo</code> object from it: eg <code>Out(coo=your_list$coo)</code>.</p>
</td></tr>
<tr><td><code id="import_tps_+3A_curves">curves</code></td>
<td>
<p><code>logical</code> whether to read curves, if any</p>
</td></tr>
<tr><td><code id="import_tps_+3A_tps">tps</code></td>
<td>
<p>lines for a single tps file
<code>tps2coo</code> is used in <a href="#topic+import_tps">import_tps</a> and may be useful for data import. When provided
with lines (eg after <a href="base.html#topic+readLines">readLines</a>) from a tps-like description (with &quot;LM&quot;, &quot;CURVES&quot;, etc.) returns a list of
coordinates, curves, etc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with components:
<code>coo</code> a matrix of coordinates; <code>cur</code> a list of matrices; <code>scale</code> the scale as a numeric.
</p>


<h3>Note</h3>

<p>Note this function will be deprecated from Momocs
when <code>Momacs</code> and <code>Momit</code>
will be fully operationnal.
</p>


<h3>See Also</h3>

<p>Other import functions: 
<code><a href="#topic+import_Conte">import_Conte</a>()</code>,
<code><a href="#topic+import_StereoMorph_curve1">import_StereoMorph_curve1</a>()</code>,
<code><a href="#topic+import_jpg1">import_jpg1</a>()</code>,
<code><a href="#topic+import_jpg">import_jpg</a>()</code>,
<code><a href="#topic+import_txt">import_txt</a>()</code>,
<code><a href="#topic+pix2chc">pix2chc</a>()</code>
</p>
<p>Other import functions: 
<code><a href="#topic+import_Conte">import_Conte</a>()</code>,
<code><a href="#topic+import_StereoMorph_curve1">import_StereoMorph_curve1</a>()</code>,
<code><a href="#topic+import_jpg1">import_jpg1</a>()</code>,
<code><a href="#topic+import_jpg">import_jpg</a>()</code>,
<code><a href="#topic+import_txt">import_txt</a>()</code>,
<code><a href="#topic+pix2chc">pix2chc</a>()</code>
</p>

<hr>
<h2 id='import_txt'>Import coordinates from a .txt file</h2><span id='topic+import_txt'></span>

<h3>Description</h3>

<p>A wrapper around <a href="utils.html#topic+read.table">read.table</a>
that can be used to import outline/landmark coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_txt(txt.paths = .lf.auto(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_txt_+3A_txt.paths">txt.paths</code></td>
<td>
<p>a vector of paths corresponding to the .txt files to import. If not
provided (or <code>NULL</code>), switches to the automatic version, just as in <a href="#topic+import_jpg">import_jpg</a>.
See Details there.</p>
</td></tr>
<tr><td><code id="import_txt_+3A_...">...</code></td>
<td>
<p>arguments to be passed to <a href="utils.html#topic+read.table">read.table</a>, eg. 'skip', 'dec', etc.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Columns are not named in the <code>.txt</code> files. You can tune this using the <code>...</code> argument.
Define the <a href="utils.html#topic+read.table">read.table</a> arguments that allow to import a single file, and then
pass them to this function, ie if your <code>.txt</code> file
has a header (eg ('x', 'y')), do not forget <code>header=TRUE</code>.
</p>


<h3>Value</h3>

<p>a list of matrix(ces) of (x; y) coordinates that can be passed to
<a href="#topic+Out">Out</a>, <a href="#topic+Opn">Opn</a> and <a href="#topic+Ldk">Ldk</a>.
</p>


<h3>Note</h3>

<p>Note this function will be deprecated from Momocs
when <code>Momacs</code> and <code>Momit</code>
will be fully operationnal.
</p>
<p>Silent message and progress bars (if any) with <code>options("verbose"=FALSE)</code>.
</p>


<h3>See Also</h3>

<p>Other import functions: 
<code><a href="#topic+import_Conte">import_Conte</a>()</code>,
<code><a href="#topic+import_StereoMorph_curve1">import_StereoMorph_curve1</a>()</code>,
<code><a href="#topic+import_jpg1">import_jpg1</a>()</code>,
<code><a href="#topic+import_jpg">import_jpg</a>()</code>,
<code><a href="#topic+import_tps">import_tps</a>()</code>,
<code><a href="#topic+pix2chc">pix2chc</a>()</code>
</p>

<hr>
<h2 id='inspect'>Graphical inspection of shapes</h2><span id='topic+inspect'></span>

<h3>Description</h3>

<p>Allows to plot shapes, individually, for <a href="#topic+Coo">Coo</a> (<a href="#topic+Out">Out</a>, <a href="#topic+Opn">Opn</a> or <a href="#topic+Ldk">Ldk</a>) objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inspect(x, id, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="inspect_+3A_x">x</code></td>
<td>
<p>the <a href="#topic+Coo">Coo</a> object</p>
</td></tr>
<tr><td><code id="inspect_+3A_id">id</code></td>
<td>
<p>the id of the shape to plot, if not provided a
random shape is plotted. If passed with <code>'all'</code> all shapes are plotted,
one by one.</p>
</td></tr>
<tr><td><code id="inspect_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <a href="#topic+coo_plot">coo_plot</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>an interactive plot
</p>


<h3>See Also</h3>

<p>Other Coo_graphics: 
<code><a href="#topic+panel">panel</a>()</code>,
<code><a href="#topic+stack">stack</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
inspect(bot, 5)
inspect(bot)
inspect(bot, 5, pch=3, points=TRUE) # an example of '...' use

## End(Not run)
</code></pre>

<hr>
<h2 id='is'>Class and component testers</h2><span id='topic+is'></span><span id='topic+is_Coo'></span><span id='topic+is_PCA'></span><span id='topic+is_LDA'></span><span id='topic+is_Out'></span><span id='topic+is_Opn'></span><span id='topic+is_Ldk'></span><span id='topic+is_Coe'></span><span id='topic+is_OutCoe'></span><span id='topic+is_OpnCoe'></span><span id='topic+is_LdkCoe'></span><span id='topic+is_TraCoe'></span><span id='topic+is_shp'></span><span id='topic+is_fac'></span><span id='topic+is_ldk'></span><span id='topic+is_slidings'></span><span id='topic+is_links'></span>

<h3>Description</h3>

<p>Class testers test if any of the classes of an object <em>is</em> of a given class. For instance
<code>is_PCA</code> on a <a href="#topic+PCA">PCA</a> object (of classes <code>PCA</code> and <code>prcomp</code>) will return <code>TRUE</code>.
Component testers check if <em>there_is</em> a particular component (eg <code style="white-space: pre;">&#8288;$fac&#8288;</code>, etc.) in an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_Coo(x)

is_PCA(x)

is_LDA(x)

is_Out(x)

is_Opn(x)

is_Ldk(x)

is_Coe(x)

is_OutCoe(x)

is_OpnCoe(x)

is_LdkCoe(x)

is_TraCoe(x)

is_shp(x)

is_fac(x)

is_ldk(x)

is_slidings(x)

is_links(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_+3A_x">x</code></td>
<td>
<p>the object to test</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>logical</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_Coo(bot)
is_Out(bot)
is_Ldk(bot)
is_ldk(hearts) # mind the capitals!
</code></pre>

<hr>
<h2 id='is_equallyspacedradii'>Tests if coordinates likely have equally spaced radii</h2><span id='topic+is_equallyspacedradii'></span>

<h3>Description</h3>

<p>Returns TRUE/FALSE whether the sd of angles between all successive
radii is below/above <code>thesh</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_equallyspacedradii(coo, thres)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_equallyspacedradii_+3A_coo">coo</code></td>
<td>
<p><code>matrix</code> of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates or any <a href="#topic+Coo">Coo</a> object.</p>
</td></tr>
<tr><td><code id="is_equallyspacedradii_+3A_thres">thres</code></td>
<td>
<p>numeric a threshold (arbitrarily <code>pi/90</code>, eg 2 degrees, by default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a single or a vector of <code>logical</code>. If <code>NA</code> are returned,
some coordinates are likely identical, at least for x or y.
</p>


<h3>See Also</h3>

<p>Other coo_ utilities: 
<code><a href="#topic+coo_aligncalliper">coo_aligncalliper</a>()</code>,
<code><a href="#topic+coo_alignminradius">coo_alignminradius</a>()</code>,
<code><a href="#topic+coo_alignxax">coo_alignxax</a>()</code>,
<code><a href="#topic+coo_align">coo_align</a>()</code>,
<code><a href="#topic+coo_baseline">coo_baseline</a>()</code>,
<code><a href="#topic+coo_bookstein">coo_bookstein</a>()</code>,
<code><a href="#topic+coo_boundingbox">coo_boundingbox</a>()</code>,
<code><a href="#topic+coo_calliper">coo_calliper</a>()</code>,
<code><a href="#topic+coo_centdist">coo_centdist</a>()</code>,
<code><a href="#topic+coo_center">coo_center</a>()</code>,
<code><a href="#topic+coo_centpos">coo_centpos</a>()</code>,
<code><a href="#topic+coo_close">coo_close</a>()</code>,
<code><a href="#topic+coo_down">coo_down</a>()</code>,
<code><a href="#topic+coo_dxy">coo_dxy</a>()</code>,
<code><a href="#topic+coo_extract">coo_extract</a>()</code>,
<code><a href="#topic+coo_flipx">coo_flipx</a>()</code>,
<code><a href="#topic+coo_force2close">coo_force2close</a>()</code>,
<code><a href="#topic+coo_interpolate">coo_interpolate</a>()</code>,
<code><a href="#topic+coo_is_closed">coo_is_closed</a>()</code>,
<code><a href="#topic+coo_jitter">coo_jitter</a>()</code>,
<code><a href="#topic+coo_left">coo_left</a>()</code>,
<code><a href="#topic+coo_likely_clockwise">coo_likely_clockwise</a>()</code>,
<code><a href="#topic+coo_nb">coo_nb</a>()</code>,
<code><a href="#topic+coo_perim">coo_perim</a>()</code>,
<code><a href="#topic+coo_range">coo_range</a>()</code>,
<code><a href="#topic+coo_rev">coo_rev</a>()</code>,
<code><a href="#topic+coo_right">coo_right</a>()</code>,
<code><a href="#topic+coo_rotatecenter">coo_rotatecenter</a>()</code>,
<code><a href="#topic+coo_rotate">coo_rotate</a>()</code>,
<code><a href="#topic+coo_sample_prop">coo_sample_prop</a>()</code>,
<code><a href="#topic+coo_samplerr">coo_samplerr</a>()</code>,
<code><a href="#topic+coo_sample">coo_sample</a>()</code>,
<code><a href="#topic+coo_scale">coo_scale</a>()</code>,
<code><a href="#topic+coo_shearx">coo_shearx</a>()</code>,
<code><a href="#topic+coo_slice">coo_slice</a>()</code>,
<code><a href="#topic+coo_slidedirection">coo_slidedirection</a>()</code>,
<code><a href="#topic+coo_slidegap">coo_slidegap</a>()</code>,
<code><a href="#topic+coo_slide">coo_slide</a>()</code>,
<code><a href="#topic+coo_smoothcurve">coo_smoothcurve</a>()</code>,
<code><a href="#topic+coo_smooth">coo_smooth</a>()</code>,
<code><a href="#topic+coo_template">coo_template</a>()</code>,
<code><a href="#topic+coo_trans">coo_trans</a>()</code>,
<code><a href="#topic+coo_trimbottom">coo_trimbottom</a>()</code>,
<code><a href="#topic+coo_trimtop">coo_trimtop</a>()</code>,
<code><a href="#topic+coo_trim">coo_trim</a>()</code>,
<code><a href="#topic+coo_untiltx">coo_untiltx</a>()</code>,
<code><a href="#topic+coo_up">coo_up</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot[1] %&gt;% is_equallyspacedradii
bot[1] %&gt;% coo_samplerr(36) %&gt;% is_equallyspacedradii
# higher tolerance but wrong
bot[1] %&gt;% coo_samplerr(36) %&gt;% is_equallyspacedradii(thres=5*2*pi/360)
# coo_interpolate is a better option
bot[1] %&gt;% coo_interpolate(1200) %&gt;% coo_samplerr(36) %&gt;% is_equallyspacedradii
# Coo method
bot %&gt;% coo_interpolate(360) %&gt;% coo_samplerr(36) %&gt;% is_equallyspacedradii
</code></pre>

<hr>
<h2 id='KMEANS'>KMEANS on PCA objects</h2><span id='topic+KMEANS'></span><span id='topic+KMEANS.PCA'></span>

<h3>Description</h3>

<p>A very basic implementation of k-means.
Beware that morphospaces are calculated so far for the 1st and 2nd
component.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KMEANS(x, ...)

## S3 method for class 'PCA'
KMEANS(x, centers, nax = 1:2, pch = 20, cex = 0.5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KMEANS_+3A_x">x</code></td>
<td>
<p>PCA object</p>
</td></tr>
<tr><td><code id="KMEANS_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed to <a href="stats.html#topic+kmeans">kmeans</a></p>
</td></tr>
<tr><td><code id="KMEANS_+3A_centers">centers</code></td>
<td>
<p>numeric number of centers</p>
</td></tr>
<tr><td><code id="KMEANS_+3A_nax">nax</code></td>
<td>
<p>numeric the range of PC components to use (1:2 by default)</p>
</td></tr>
<tr><td><code id="KMEANS_+3A_pch">pch</code></td>
<td>
<p>to draw the points</p>
</td></tr>
<tr><td><code id="KMEANS_+3A_cex">cex</code></td>
<td>
<p>to draw the points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the same thing as <a href="stats.html#topic+kmeans">kmeans</a>
</p>


<h3>See Also</h3>

<p>Other multivariate: 
<code><a href="#topic+CLUST">CLUST</a>()</code>,
<code><a href="#topic+KMEDOIDS">KMEDOIDS</a>()</code>,
<code><a href="#topic+LDA">LDA</a>()</code>,
<code><a href="#topic+MANOVA_PW">MANOVA_PW</a>()</code>,
<code><a href="#topic+MANOVA">MANOVA</a>()</code>,
<code><a href="#topic+MDS">MDS</a>()</code>,
<code><a href="#topic+MSHAPES">MSHAPES</a>()</code>,
<code><a href="#topic+NMDS">NMDS</a>()</code>,
<code><a href="#topic+PCA">PCA</a>()</code>,
<code><a href="#topic+classification_metrics">classification_metrics</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bot)
bp &lt;- PCA(efourier(bot, 10))
KMEANS(bp, 2)
</code></pre>

<hr>
<h2 id='KMEDOIDS'>KMEDOIDS</h2><span id='topic+KMEDOIDS'></span><span id='topic+KMEDOIDS.default'></span><span id='topic+KMEDOIDS.Coe'></span><span id='topic+KMEDOIDS.PCA'></span>

<h3>Description</h3>

<p>A basic implementation of kmedoids on top of <a href="cluster.html#topic+pam">cluster::pam</a>
Beware that morphospaces are calculated so far for the 1st and 2nd
component.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KMEDOIDS(x, k, metric = "euclidean", ...)

## Default S3 method:
KMEDOIDS(x, k, metric = "euclidean", ...)

## S3 method for class 'Coe'
KMEDOIDS(x, k, metric = "euclidean", ...)

## S3 method for class 'PCA'
KMEDOIDS(x, k, metric = "euclidean", retain, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KMEDOIDS_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+Coe">Coe</a> or <a href="#topic+PCA">PCA</a> object</p>
</td></tr>
<tr><td><code id="KMEDOIDS_+3A_k">k</code></td>
<td>
<p>numeric number of centers</p>
</td></tr>
<tr><td><code id="KMEDOIDS_+3A_metric">metric</code></td>
<td>
<p>one of <code>euclidean</code> (default) or <code>manhattan</code>, to feed <a href="cluster.html#topic+pam">cluster::pam</a></p>
</td></tr>
<tr><td><code id="KMEDOIDS_+3A_...">...</code></td>
<td>
<p>additional arguments to feed <a href="cluster.html#topic+pam">cluster::pam</a></p>
</td></tr>
<tr><td><code id="KMEDOIDS_+3A_retain">retain</code></td>
<td>
<p>when passing a <a href="#topic+PCA">PCA</a> how many PCs to retain, or a proportion of total variance, see <a href="#topic+LDA">LDA</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>see <a href="cluster.html#topic+pam">cluster::pam</a>. Other components are returned (<code>fac</code>, etc.)
</p>


<h3>See Also</h3>

<p>Other multivariate: 
<code><a href="#topic+CLUST">CLUST</a>()</code>,
<code><a href="#topic+KMEANS">KMEANS</a>()</code>,
<code><a href="#topic+LDA">LDA</a>()</code>,
<code><a href="#topic+MANOVA_PW">MANOVA_PW</a>()</code>,
<code><a href="#topic+MANOVA">MANOVA</a>()</code>,
<code><a href="#topic+MDS">MDS</a>()</code>,
<code><a href="#topic+MSHAPES">MSHAPES</a>()</code>,
<code><a href="#topic+NMDS">NMDS</a>()</code>,
<code><a href="#topic+PCA">PCA</a>()</code>,
<code><a href="#topic+classification_metrics">classification_metrics</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bot)
bp &lt;- PCA(efourier(bot, 10))
KMEANS(bp, 2)

set.seed(123) # for reproducibility on a dummy matrix
matrix(rnorm(100, 10, 10)) %&gt;%
KMEDOIDS(5)

# On a Coe
bot_f &lt;- bot %&gt;% efourier()

bot_k &lt;- bot_f %&gt;% KMEDOIDS(2)
# confusion matrix
table(bot_k$fac$type, bot_k$clustering)

# on a PCA
bot_k2 &lt;- bot_f %&gt;% PCA() %&gt;% KMEDOIDS(12, retain=0.9)

# confusion matrix
with(bot_k, table(fac$type, clustering))
# silhouette plot
bot_k %&gt;% plot_silhouette()

# average width as a function of k
k_range &lt;- 2:12
widths &lt;- sapply(k_range, function(k) KMEDOIDS(bot_f, k=k)$silinfo$avg.width)
plot(k_range, widths, type="b")
</code></pre>

<hr>
<h2 id='layers'>grindr layers for multivariate plots</h2><span id='topic+layers'></span><span id='topic+layer_frame'></span><span id='topic+layer_axes'></span><span id='topic+layer_ticks'></span><span id='topic+layer_grid'></span><span id='topic+layer_box'></span><span id='topic+layer_fullframe'></span><span id='topic+layer_points'></span><span id='topic+layer_ellipses'></span><span id='topic+layer_ellipsesfilled'></span><span id='topic+layer_ellipsesaxes'></span><span id='topic+layer_chull'></span><span id='topic+layer_chullfilled'></span><span id='topic+layer_stars'></span><span id='topic+layer_delaunay'></span><span id='topic+layer_density'></span><span id='topic+layer_labelpoints'></span><span id='topic+layer_labelgroups'></span><span id='topic+layer_rug'></span><span id='topic+layer_histogram_2'></span><span id='topic+layer_density_2'></span><span id='topic+layer_title'></span><span id='topic+layer_axesnames'></span><span id='topic+layer_eigen'></span><span id='topic+layer_axesvar'></span><span id='topic+layer_legend'></span>

<h3>Description</h3>

<p>Useful layers for building custom
mutivariate plots using the cheapbabi approach. See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>layer_frame(x, center_origin = TRUE, zoom = 0.9)

layer_axes(x, col = "#999999", lwd = 1/2, ...)

layer_ticks(x, col = "#333333", cex = 3/4, lwd = 3/4, ...)

layer_grid(x, col = "#999999", lty = 3, grid = 3, ...)

layer_box(x, border = "#e5e5e5", ...)

layer_fullframe(x, ...)

layer_points(x, pch = 20, cex = 4/log1p(nrow(x$xy)), transp = 0, ...)

layer_ellipses(x, conf = 0.5, lwd = 1, alpha = 0, ...)

layer_ellipsesfilled(x, conf = 0.5, lwd = 1, alpha = 0, ...)

layer_ellipsesaxes(x, conf = 0.5, lwd = 1, alpha = 0, ...)

layer_chull(x, ...)

layer_chullfilled(x, alpha = 0.8, ...)

layer_stars(x, alpha = 0.5, ...)

layer_delaunay(x, ...)

layer_density(
  x,
  levels_density = 20,
  levels_contour = 4,
  alpha = 1/3,
  n = 200,
  density = TRUE,
  contour = TRUE
)

layer_labelpoints(
  x,
  col = par("fg"),
  cex = 2/3,
  font = 1,
  abbreviate = FALSE,
  ...
)

layer_labelgroups(
  x,
  col = par("fg"),
  cex = 3/4,
  font = 2,
  rect = TRUE,
  alpha = 1/4,
  abbreviate = FALSE,
  ...
)

layer_rug(x, size = 1/200, ...)

layer_histogram_2(x, freq = FALSE, breaks, split = FALSE, transp = 0)

layer_density_2(x, bw, split = FALSE, rug = TRUE, transp = 0)

layer_title(x, title = "", cex = 3/4, ...)

layer_axesnames(x, cex = 3/4, name = "Axis", ...)

layer_eigen(x, nb_max = 5, cex = 1/2, ...)

layer_axesvar(x, cex = 3/4, ...)

layer_legend(x, probs = seq(0, 1, 0.25), cex = 3/4, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="layers_+3A_x">x</code></td>
<td>
<p>a list, typically returned by <a href="#topic+plot_PCA">plot_PCA</a></p>
</td></tr>
<tr><td><code id="layers_+3A_center_origin">center_origin</code></td>
<td>
<p><code>logical</code> whether to center the origin (default <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="layers_+3A_zoom">zoom</code></td>
<td>
<p><code>numeric</code> to change the zoom (default <code>0.9</code>)</p>
</td></tr>
<tr><td><code id="layers_+3A_col">col</code></td>
<td>
<p>color (hexadecimal) to use for drawing components</p>
</td></tr>
<tr><td><code id="layers_+3A_lwd">lwd</code></td>
<td>
<p>linewidth for drawing components</p>
</td></tr>
<tr><td><code id="layers_+3A_...">...</code></td>
<td>
<p>additional options to feed core functions for each layer</p>
</td></tr>
<tr><td><code id="layers_+3A_cex">cex</code></td>
<td>
<p>to use for drawing components</p>
</td></tr>
<tr><td><code id="layers_+3A_lty">lty</code></td>
<td>
<p>linetype for drawing components</p>
</td></tr>
<tr><td><code id="layers_+3A_grid">grid</code></td>
<td>
<p><code>numeric</code> number of grid to draw</p>
</td></tr>
<tr><td><code id="layers_+3A_border">border</code></td>
<td>
<p>color (hexadecimal) to use to draw border</p>
</td></tr>
<tr><td><code id="layers_+3A_pch">pch</code></td>
<td>
<p>to use for drawing components</p>
</td></tr>
<tr><td><code id="layers_+3A_transp">transp</code></td>
<td>
<p>transparency to use (min: 0 defaut:0 max:1)</p>
</td></tr>
<tr><td><code id="layers_+3A_conf">conf</code></td>
<td>
<p><code>numeric</code> between 0 and 1 for confidence ellipses</p>
</td></tr>
<tr><td><code id="layers_+3A_alpha">alpha</code></td>
<td>
<p><code>numeric</code> between 0 and 1 for the transparency of components</p>
</td></tr>
<tr><td><code id="layers_+3A_levels_density">levels_density</code></td>
<td>
<p><code>numeric</code> number of levels to use to feed <code>MASS::kde2d</code></p>
</td></tr>
<tr><td><code id="layers_+3A_levels_contour">levels_contour</code></td>
<td>
<p><code>numeric</code> number of levels to use to feed <code>graphics::contour</code></p>
</td></tr>
<tr><td><code id="layers_+3A_n">n</code></td>
<td>
<p><code>numeric</code> number of grid points to feed <code>MASS::kde2d</code></p>
</td></tr>
<tr><td><code id="layers_+3A_density">density</code></td>
<td>
<p><code>logical</code> whether to draw density estimate</p>
</td></tr>
<tr><td><code id="layers_+3A_contour">contour</code></td>
<td>
<p><code>logical</code> whether to draw contour lines</p>
</td></tr>
<tr><td><code id="layers_+3A_font">font</code></td>
<td>
<p>to feed <a href="graphics.html#topic+text">text</a></p>
</td></tr>
<tr><td><code id="layers_+3A_abbreviate">abbreviate</code></td>
<td>
<p><code>logical</code> whether to abbreviate names</p>
</td></tr>
<tr><td><code id="layers_+3A_rect">rect</code></td>
<td>
<p><code>logical</code> whether to draw a rectangle below names</p>
</td></tr>
<tr><td><code id="layers_+3A_size">size</code></td>
<td>
<p><code>numeric</code> as a fraction of graphical window (default: <code>1/200</code>)</p>
</td></tr>
<tr><td><code id="layers_+3A_freq">freq</code></td>
<td>
<p><code>logical</code>to feed<code style="white-space: pre;">&#8288;[hist] (default:&#8288;</code>FALSE')</p>
</td></tr>
<tr><td><code id="layers_+3A_breaks">breaks</code></td>
<td>
<p>to feed <a href="graphics.html#topic+hist">hist</a> (default: calculated on the pooled values)</p>
</td></tr>
<tr><td><code id="layers_+3A_split">split</code></td>
<td>
<p><code>logical</code> whether to split the two distributions into two plots</p>
</td></tr>
<tr><td><code id="layers_+3A_bw">bw</code></td>
<td>
<p>to feed <a href="stats.html#topic+density">density</a> (default: <a href="stats.html#topic+bandwidth">stats::bw.nrd0</a>)</p>
</td></tr>
<tr><td><code id="layers_+3A_rug">rug</code></td>
<td>
<p><code>logical</code> whether to add <a href="graphics.html#topic+rug">rug</a> (default: <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="layers_+3A_title">title</code></td>
<td>
<p>to add to the plot (default <code>""</code>)</p>
</td></tr>
<tr><td><code id="layers_+3A_name">name</code></td>
<td>
<p>to use on axes (default <code>"Axis"</code>)</p>
</td></tr>
<tr><td><code id="layers_+3A_nb_max">nb_max</code></td>
<td>
<p><code>numeric</code> number of eigen values to display (default <code>5</code>)</p>
</td></tr>
<tr><td><code id="layers_+3A_probs">probs</code></td>
<td>
<p><code>numeric</code> sequence to feed <code>stats::quantile</code>
and to indicate where to draw ticks and legend labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing layer
</p>


<h3>See Also</h3>

<p>grindr_drawers
</p>
<p>Other grindr: 
<code><a href="#topic+drawers">drawers</a></code>,
<code><a href="#topic+layers_morphospace">layers_morphospace</a></code>,
<code><a href="#topic+mosaic_engine">mosaic_engine</a>()</code>,
<code><a href="#topic+papers">papers</a></code>,
<code><a href="#topic+pile">pile</a>()</code>,
<code><a href="#topic+plot_LDA">plot_LDA</a>()</code>,
<code><a href="#topic+plot_NMDS">plot_NMDS</a>()</code>,
<code><a href="#topic+plot_PCA">plot_PCA</a>()</code>
</p>

<hr>
<h2 id='layers_morphospace'>Morphospace layers</h2><span id='topic+layers_morphospace'></span><span id='topic+layer_morphospace_PCA'></span><span id='topic+layer_morphospace_LDA'></span>

<h3>Description</h3>

<p>Used internally by <a href="#topic+plot_PCA">plot_PCA</a>, <a href="#topic+plot_LDA">plot_LDA</a>, etc. but may be useful elsewhere.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>layer_morphospace_PCA(
  x,
  position = c("range", "full", "circle", "xy", "range_axes", "full_axes")[1],
  nb = 12,
  nr = 6,
  nc = 5,
  rotate = 0,
  size = 0.9,
  col = "#999999",
  flipx = FALSE,
  flipy = FALSE,
  draw = TRUE
)

layer_morphospace_LDA(
  x,
  position = c("range", "full", "circle", "xy", "range_axes", "full_axes")[1],
  nb = 12,
  nr = 6,
  nc = 5,
  rotate = 0,
  size = 0.9,
  col = "#999999",
  flipx = FALSE,
  flipy = FALSE,
  draw = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="layers_morphospace_+3A_x">x</code></td>
<td>
<p>layered <a href="#topic+PCA">PCA</a> or <a href="#topic+LDA">LDA</a>. Typically, the object returned by <a href="#topic+plot_PCA">plot_PCA</a> and <a href="#topic+plot_LDA">plot_LDA</a></p>
</td></tr>
<tr><td><code id="layers_morphospace_+3A_position">position</code></td>
<td>
<p>one of <code style="white-space: pre;">&#8288;range, full, circle, xy, range_axes, full_axes&#8288;</code> to feed <a href="#topic+morphospace_positions">morphospace_positions</a> (default: <code>range</code>)</p>
</td></tr>
<tr><td><code id="layers_morphospace_+3A_nb">nb</code></td>
<td>
<p><code>numeric</code> total number of shapes when <code>position="circle"</code> (default: <code>12</code>)</p>
</td></tr>
<tr><td><code id="layers_morphospace_+3A_nr">nr</code></td>
<td>
<p><code>numeric</code> number of rows to position shapes (default: <code>6</code>)</p>
</td></tr>
<tr><td><code id="layers_morphospace_+3A_nc">nc</code></td>
<td>
<p><code>numeric</code> number of columns to position shapes (default <code>5</code>)</p>
</td></tr>
<tr><td><code id="layers_morphospace_+3A_rotate">rotate</code></td>
<td>
<p><code>numeric</code> angle (in radians) to rotate shapes when displayed on the morphospace (default: <code>0</code>)</p>
</td></tr>
<tr><td><code id="layers_morphospace_+3A_size">size</code></td>
<td>
<p><code>numeric</code> size to use to feed <a href="#topic+coo_template">coo_template</a> (default: <code>0.9</code>)</p>
</td></tr>
<tr><td><code id="layers_morphospace_+3A_col">col</code></td>
<td>
<p>color to draw shapes (default: <code style="white-space: pre;">&#8288;#999999&#8288;</code>)</p>
</td></tr>
<tr><td><code id="layers_morphospace_+3A_flipx">flipx</code></td>
<td>
<p><code>logical</code> whether to flip shapes against the x-axis (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="layers_morphospace_+3A_flipy">flipy</code></td>
<td>
<p><code>logical</code> whether to flip shapes against the y-axis (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="layers_morphospace_+3A_draw">draw</code></td>
<td>
<p><code>logical</code> whether to draw shapes (default: <code>TRUE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing layer
</p>


<h3>See Also</h3>

<p>Other grindr: 
<code><a href="#topic+drawers">drawers</a></code>,
<code><a href="#topic+layers">layers</a></code>,
<code><a href="#topic+mosaic_engine">mosaic_engine</a>()</code>,
<code><a href="#topic+papers">papers</a></code>,
<code><a href="#topic+pile">pile</a>()</code>,
<code><a href="#topic+plot_LDA">plot_LDA</a>()</code>,
<code><a href="#topic+plot_NMDS">plot_NMDS</a>()</code>,
<code><a href="#topic+plot_PCA">plot_PCA</a>()</code>
</p>
<p>Other grindr: 
<code><a href="#topic+drawers">drawers</a></code>,
<code><a href="#topic+layers">layers</a></code>,
<code><a href="#topic+mosaic_engine">mosaic_engine</a>()</code>,
<code><a href="#topic+papers">papers</a></code>,
<code><a href="#topic+pile">pile</a>()</code>,
<code><a href="#topic+plot_LDA">plot_LDA</a>()</code>,
<code><a href="#topic+plot_NMDS">plot_NMDS</a>()</code>,
<code><a href="#topic+plot_PCA">plot_PCA</a>()</code>
</p>

<hr>
<h2 id='LDA'>Linear Discriminant Analysis on Coe objects</h2><span id='topic+LDA'></span><span id='topic+LDA.default'></span><span id='topic+LDA.PCA'></span>

<h3>Description</h3>

<p>Calculates a LDA on <a href="#topic+Coe">Coe</a> on top of <a href="MASS.html#topic+lda">MASS::lda</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDA(x, fac, retain, ...)

## Default S3 method:
LDA(x, fac, retain, ...)

## S3 method for class 'PCA'
LDA(x, fac, retain = 0.99, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDA_+3A_x">x</code></td>
<td>
<p>a  Coe or a PCA object</p>
</td></tr>
<tr><td><code id="LDA_+3A_fac">fac</code></td>
<td>
<p>the grouping factor (names of one of the $fac column or column id)</p>
</td></tr>
<tr><td><code id="LDA_+3A_retain">retain</code></td>
<td>
<p>the proportion of the total variance to retain (if retain&lt;1) using <a href="#topic+scree">scree</a>, or the number of PC axis (if retain&gt;1).</p>
</td></tr>
<tr><td><code id="LDA_+3A_...">...</code></td>
<td>
<p>additional arguments to feed <a href="MASS.html#topic+lda">lda</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 'LDA' object on which to apply <a href="#topic+plot.LDA">plot.LDA</a>, which is a list with components:
</p>

<ul>
<li> <p><code>x</code> any <a href="#topic+Coe">Coe</a> object (or a matrix)
</p>
</li>
<li> <p><code>fac</code> grouping factor used
</p>
</li>
<li> <p><code>removed</code> ids of columns in the original matrix that have been removed since constant (if any)
</p>
</li>
<li> <p><code>mod</code> the raw lda mod from <a href="MASS.html#topic+lda">lda</a>
</p>
</li>
<li> <p><code>mod.pred</code> the predicted model using x and mod
</p>
</li>
<li> <p><code>CV.fac</code> cross-validated classification
</p>
</li>
<li> <p><code>CV.tab</code> cross-validation tabke
</p>
</li>
<li> <p><code>CV.correct</code> proportion of correctly classified individuals
</p>
</li>
<li> <p><code>CV.ce</code> class error
</p>
</li>
<li> <p><code>LDs</code> unstandardized LD scores see Claude (2008)
</p>
</li>
<li> <p><code>mshape</code> mean values of coefficients in the original matrix
</p>
</li>
<li> <p><code>method</code> inherited from the Coe object (if any)
</p>
</li></ul>



<h3>Note</h3>

<p>For LDA.PCA, retain can be passed as a vector (eg: 1:5, and retain=1, retain=2, ...,
retain=5) will be tried, or as &quot;best&quot; (same as before but retain=1:number_of_pc_axes is used).
</p>
<p>Silent message and progress bars (if any) with <code>options("verbose"=FALSE)</code>.
</p>


<h3>See Also</h3>

<p>Other multivariate: 
<code><a href="#topic+CLUST">CLUST</a>()</code>,
<code><a href="#topic+KMEANS">KMEANS</a>()</code>,
<code><a href="#topic+KMEDOIDS">KMEDOIDS</a>()</code>,
<code><a href="#topic+MANOVA_PW">MANOVA_PW</a>()</code>,
<code><a href="#topic+MANOVA">MANOVA</a>()</code>,
<code><a href="#topic+MDS">MDS</a>()</code>,
<code><a href="#topic+MSHAPES">MSHAPES</a>()</code>,
<code><a href="#topic+NMDS">NMDS</a>()</code>,
<code><a href="#topic+PCA">PCA</a>()</code>,
<code><a href="#topic+classification_metrics">classification_metrics</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot.f &lt;- efourier(bot, 24)
bot.p &lt;- PCA(bot.f)
LDA(bot.p, 'type', retain=0.99) # retains 0.99 of the total variance
LDA(bot.p, 'type', retain=5) # retain 5 axis
bot.l &lt;- LDA(bot.p, 'type', retain=0.99)
plot_LDA(bot.l)
bot.f &lt;- mutate(bot.f, plop=factor(rep(letters[1:4], each=10)))
bot.l &lt;- LDA(PCA(bot.f), 'plop')
plot_LDA(bot.l) # will replace the former soon
</code></pre>

<hr>
<h2 id='Ldk'>Builds an Ldk object</h2><span id='topic+Ldk'></span><span id='topic+LdkCoe'></span>

<h3>Description</h3>

<p>In Momocs, <code>Ldk</code> classes objects are
lists of configurations of <b>l</b>an<b>d</b>mar<b>k</b>s, with optionnal components,
on which generic methods such as plotting methods (e.g. <a href="#topic+stack">stack</a>)
and specific methods (e.g. <a href="#topic+fgProcrustes">fgProcrustes</a>).
<code>Ldk</code> objects are primarily <code><a href="#topic+Coo">Coo</a></code> objects. In a sense, morphometrics methods
on Ldk objects preserves (x, y) coordinates and <code>LdkCoe</code> are also <code>Ldk</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ldk(coo, fac = dplyr::tibble(), links = NULL, slidings = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ldk_+3A_coo">coo</code></td>
<td>
<p>a <code>list</code> of matrices of (x; y) coordinates,
or an array, or an Ldk object or a data.frame (and friends)</p>
</td></tr>
<tr><td><code id="Ldk_+3A_fac">fac</code></td>
<td>
<p>(optionnal) a <code>data.frame</code> of factors and/or numerics
specifying the grouping structure</p>
</td></tr>
<tr><td><code id="Ldk_+3A_links">links</code></td>
<td>
<p>(optionnal) a 2-columns <code>matrix</code> of 'links' between landmarks, mainly for plotting</p>
</td></tr>
<tr><td><code id="Ldk_+3A_slidings">slidings</code></td>
<td>
<p>(optionnal) a 3-columns <code>matrix</code> defining (if any) sliding landmarks</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All the shapes in x must have the same number of landmarks. If you are
trying to make an Ldk object from an Out or an Opn object, try <a href="#topic+coo_sample">coo_sample</a> beforehand
to homogeneize the number of coordinates among shapes.
Please note that Ldk methods are as experimental.
</p>
<p>implementation of <code>$slidings</code> is inspired by <code>geomorph</code>
</p>


<h3>Value</h3>

<p>an <code>Ldk</code> object
</p>


<h3>See Also</h3>

<p>Other classes: 
<code><a href="#topic+Coe">Coe</a>()</code>,
<code><a href="#topic+Coo">Coo</a>()</code>,
<code><a href="#topic+OpnCoe">OpnCoe</a>()</code>,
<code><a href="#topic+Opn">Opn</a>()</code>,
<code><a href="#topic+OutCoe">OutCoe</a>()</code>,
<code><a href="#topic+Out">Out</a>()</code>,
<code><a href="#topic+TraCoe">TraCoe</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Methods on Ldk
methods(class=Ldk)

str(mosquito)
</code></pre>

<hr>
<h2 id='ldk_check'>Checks 'ldk' shapes</h2><span id='topic+ldk_check'></span>

<h3>Description</h3>

<p>A simple utility, used internally, mostly by <a href="#topic+Ldk">Ldk</a> methods,
in some graphical functions, and notably in <a href="#topic+l2a">l2a</a>.
Returns an array of landmarks arranged as <code>(nb.ldk) x (x; y) x (nb.shapes)</code>,
when passed with either a list, a matrix or an array of coordinates.
If a list is provided, checks that the number of landmarks is consistent.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldk_check(ldk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ldk_check_+3A_ldk">ldk</code></td>
<td>
<p>a <code>matrix</code> of (x; y) coordinates, a list, or an array.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an <code>array</code> of (x; y) coordinates.
</p>


<h3>See Also</h3>

<p>Other ldk helpers: 
<code><a href="#topic+def_links">def_links</a>()</code>,
<code><a href="#topic+links_all">links_all</a>()</code>,
<code><a href="#topic+links_delaunay">links_delaunay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#coo_check('Not a shape')
#coo_check(matrix(1:10, ncol=2))
#coo_check(list(x=1:5, y=6:10))
</code></pre>

<hr>
<h2 id='ldk_chull'>Draws convex hulls around landmark positions</h2><span id='topic+ldk_chull'></span>

<h3>Description</h3>

<p>A wrapper that uses <a href="#topic+coo_chull">coo_chull</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldk_chull(ldk, col = "grey40", lty = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ldk_chull_+3A_ldk">ldk</code></td>
<td>
<p>an array (or a list) of landmarks</p>
</td></tr>
<tr><td><code id="ldk_chull_+3A_col">col</code></td>
<td>
<p>a color for drawing the convex hull</p>
</td></tr>
<tr><td><code id="ldk_chull_+3A_lty">lty</code></td>
<td>
<p>an lty for drawing the convex hulls</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing on the last plot
</p>


<h3>See Also</h3>

<p><a href="#topic+coo_chull">coo_chull</a>, <a href="grDevices.html#topic+chull">chull</a>, <a href="#topic+ldk_confell">ldk_confell</a>, <a href="#topic+ldk_contour">ldk_contour</a>
</p>
<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>
<p>Other ldk plotters: 
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_plot(MSHAPES(wings))
ldk_chull(wings$coo)
</code></pre>

<hr>
<h2 id='ldk_confell'>Draws confidence ellipses for landmark positions</h2><span id='topic+ldk_confell'></span>

<h3>Description</h3>

<p>Draws confidence ellipses for landmark positions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldk_confell(
  ldk,
  conf = 0.5,
  col = "grey40",
  ell.lty = 1,
  ax = TRUE,
  ax.lty = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ldk_confell_+3A_ldk">ldk</code></td>
<td>
<p>an array (or a list) of landmarks</p>
</td></tr>
<tr><td><code id="ldk_confell_+3A_conf">conf</code></td>
<td>
<p>the confidence level (normal quantile, 0.5 by default)</p>
</td></tr>
<tr><td><code id="ldk_confell_+3A_col">col</code></td>
<td>
<p>the color for the ellipse</p>
</td></tr>
<tr><td><code id="ldk_confell_+3A_ell.lty">ell.lty</code></td>
<td>
<p>an lty for the ellipse</p>
</td></tr>
<tr><td><code id="ldk_confell_+3A_ax">ax</code></td>
<td>
<p>logical whether to draw ellipses axes</p>
</td></tr>
<tr><td><code id="ldk_confell_+3A_ax.lty">ax.lty</code></td>
<td>
<p>an lty for ellipses axes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing on the last plot
</p>


<h3>See Also</h3>

<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>
<p>Other ldk plotters: 
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_plot(MSHAPES(wings))
ldk_confell(wings$coo)
</code></pre>

<hr>
<h2 id='ldk_contour'>Draws kernel density contours around landmark</h2><span id='topic+ldk_contour'></span>

<h3>Description</h3>

<p>Using <a href="MASS.html#topic+kde2d">kde2d</a> in the MASS package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldk_contour(ldk, nlevels = 5, grid.nb = 50, col = "grey60")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ldk_contour_+3A_ldk">ldk</code></td>
<td>
<p>an array (or a list) of landmarks</p>
</td></tr>
<tr><td><code id="ldk_contour_+3A_nlevels">nlevels</code></td>
<td>
<p>the number of contour lines</p>
</td></tr>
<tr><td><code id="ldk_contour_+3A_grid.nb">grid.nb</code></td>
<td>
<p>the grid.nb</p>
</td></tr>
<tr><td><code id="ldk_contour_+3A_col">col</code></td>
<td>
<p>a color for drawing the contour lines</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing on the last plot
</p>


<h3>See Also</h3>

<p><a href="MASS.html#topic+kde2d">kde2d</a>, <a href="#topic+ldk_confell">ldk_confell</a>, <a href="#topic+ldk_chull">ldk_chull</a>
</p>
<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>
<p>Other ldk plotters: 
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_plot(MSHAPES(wings))
ldk_contour(wings$coo)
</code></pre>

<hr>
<h2 id='ldk_labels'>Add landmarks labels</h2><span id='topic+ldk_labels'></span>

<h3>Description</h3>

<p>Add landmarks labels
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldk_labels(ldk, d = 0.05, cex = 2/3, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ldk_labels_+3A_ldk">ldk</code></td>
<td>
<p>a matrix of (x; y) coordinates: where to plot the labels</p>
</td></tr>
<tr><td><code id="ldk_labels_+3A_d">d</code></td>
<td>
<p>how far from the coordinates, on a (centroid-landmark) segment</p>
</td></tr>
<tr><td><code id="ldk_labels_+3A_cex">cex</code></td>
<td>
<p>the cex for the label</p>
</td></tr>
<tr><td><code id="ldk_labels_+3A_...">...</code></td>
<td>
<p>additional parameters to fed <a href="graphics.html#topic+text">text</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing on the last plot
</p>


<h3>See Also</h3>

<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>
<p>Other ldk plotters: 
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo_plot(wings[1])
ldk_labels(wings[1])
# closer and smaller
coo_plot(wings[1])
ldk_labels(wings[1], d=0.05, cex=0.5)
</code></pre>

<hr>
<h2 id='ldk_links'>Draws links between landmarks</h2><span id='topic+ldk_links'></span>

<h3>Description</h3>

<p>Cosmetics only but useful to visualize shape variation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldk_links(ldk, links, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ldk_links_+3A_ldk">ldk</code></td>
<td>
<p>a matrix of (x; y) coordinates</p>
</td></tr>
<tr><td><code id="ldk_links_+3A_links">links</code></td>
<td>
<p>a matrix of links. On the first column the starting-id,
on the second column the ending-id (id= the number of the coordinate)</p>
</td></tr>
<tr><td><code id="ldk_links_+3A_...">...</code></td>
<td>
<p>additional parameters to fed <a href="graphics.html#topic+segments">segments</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing on the last plot
</p>


<h3>See Also</h3>

<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>
<p>Other ldk plotters: 
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>
</p>

<hr>
<h2 id='lf_structure'>bind_db.Coe &lt;- bind_db.Coo
Extracts structure from filenames</h2><span id='topic+lf_structure'></span>

<h3>Description</h3>

<p>If filenames are consistently named with the same character serating factors,
and with every individual including its belonging levels, e.g.:
</p>

<ul>
<li> <p><code>001_speciesI_siteA_ind1_dorsalview</code>
</p>
</li>
<li> <p><code>002_speciesI_siteA_ind2_lateralview</code> </p>
</li></ul>
<p> etc., this function returns a <a href="base.html#topic+data.frame">data.frame</a>
from it that can be passed to <a href="#topic+Out">Out</a>, <a href="#topic+Opn">Opn</a>, <a href="#topic+Ldk">Ldk</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lf_structure(lf, names = character(), split = "_", trim.extension = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lf_structure_+3A_lf">lf</code></td>
<td>
<p>a list (its names are used, except if it is a list from <a href="#topic+import_tps">import_tps</a>
in this case <code>names(lf$coo)</code> is used) of a list of filenames, as characters, typically such as
those obtained with <a href="base.html#topic+list.files">list.files</a>. Alternatively, a path to a folder
containing the files. Actually, if lf is of length 1 (a single character),
the function assumes it is a path and do a <a href="base.html#topic+list.files">list.files</a> on it.</p>
</td></tr>
<tr><td><code id="lf_structure_+3A_names">names</code></td>
<td>
<p>the names of the groups, as a vector of characters which length corresponds
to the number of groups.</p>
</td></tr>
<tr><td><code id="lf_structure_+3A_split">split</code></td>
<td>
<p>character, the spliting factor used for the file names.</p>
</td></tr>
<tr><td><code id="lf_structure_+3A_trim.extension">trim.extension</code></td>
<td>
<p>logical. Whether to remove the last for characters in
filenames, typically their extension, e.g. '.jpg'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The number of groups must be consistent across filenames.
</p>


<h3>Value</h3>

<p>data.frame with, for every individual, the corresponding level
for every group.
</p>


<h3>Note</h3>

<p>This is, to my view, a good practice to 'store' the grouping structure
in filenames, but it is of course not mandatory.
</p>
<p>Note also that you can: i) do a <a href="#topic+import_jpg">import_jpg</a> and save is a list, say 'foo';
then ii) pass 'names(foo)' to lf_structure. See Momocs' vignette for an illustration.
</p>
<p>Note this function will be deprecated from Momocs
when <code>Momacs</code> and <code>Momit</code>
will be fully operationnal.
</p>


<h3>See Also</h3>

<p><a href="#topic+import_jpg1">import_jpg1</a>, <a href="#topic+import_Conte">import_Conte</a>, <a href="#topic+import_txt">import_txt</a>, <a href="#topic+lf_structure">lf_structure</a>.
See also Momocs' vignettes for data import.
</p>
<p>Other babel functions: 
<code><a href="#topic+tie_jpg_txt">tie_jpg_txt</a>()</code>
</p>

<hr>
<h2 id='links_all'>Creates links (all pairwise combinations) between landmarks</h2><span id='topic+links_all'></span>

<h3>Description</h3>

<p>Creates links (all pairwise combinations) between landmarks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>links_all(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="links_all_+3A_coo">coo</code></td>
<td>
<p>a matrix (or a list) of (x; y) coordinates</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix that can be passed to <a href="#topic+ldk_links">ldk_links</a>, etc. The columns
are the row ids of the original shape.
</p>


<h3>See Also</h3>

<p>Other ldk helpers: 
<code><a href="#topic+def_links">def_links</a>()</code>,
<code><a href="#topic+ldk_check">ldk_check</a>()</code>,
<code><a href="#topic+links_delaunay">links_delaunay</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>w &lt;- wings[1]
coo_plot(w)
links &lt;- links_all(w)
ldk_links(w, links)
</code></pre>

<hr>
<h2 id='links_delaunay'>Creates links (Delaunay triangulation) between landmarks</h2><span id='topic+links_delaunay'></span>

<h3>Description</h3>

<p>Creates links (Delaunay triangulation) between landmarks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>links_delaunay(coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="links_delaunay_+3A_coo">coo</code></td>
<td>
<p>a matrix (or a list) of (x; y) coordinates</p>
</td></tr>
</table>


<h3>Details</h3>

<p>uses <a href="geometry.html#topic+delaunayn">delaunayn</a> in the <code>geometry</code> package.
</p>


<h3>Value</h3>

<p>a matrix that can be passed to <a href="#topic+ldk_links">ldk_links</a>, etc. The columns
are the row ids of the original shape.
</p>


<h3>See Also</h3>

<p>Other ldk helpers: 
<code><a href="#topic+def_links">def_links</a>()</code>,
<code><a href="#topic+ldk_check">ldk_check</a>()</code>,
<code><a href="#topic+links_all">links_all</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>w &lt;- wings[1]
coo_plot(w, poly=FALSE)
links &lt;- links_delaunay(w)
ldk_links(w, links)
</code></pre>

<hr>
<h2 id='MANOVA'>Multivariate analysis of (co)variance on Coe objects</h2><span id='topic+MANOVA'></span><span id='topic+MANOVA.OpnCoe'></span><span id='topic+MANOVA.OutCoe'></span><span id='topic+MANOVA.PCA'></span>

<h3>Description</h3>

<p>Performs multivariate analysis of variance on <a href="#topic+PCA">PCA</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MANOVA(x, fac, test = "Hotelling", retain, drop)

## S3 method for class 'OpnCoe'
MANOVA(x, fac, test = "Hotelling", retain, drop)

## S3 method for class 'OutCoe'
MANOVA(x, fac, test = "Hotelling", retain, drop)

## S3 method for class 'PCA'
MANOVA(x, fac, test = "Hotelling", retain = 0.99, drop)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MANOVA_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+Coe">Coe</a> object</p>
</td></tr>
<tr><td><code id="MANOVA_+3A_fac">fac</code></td>
<td>
<p>a name of a colum in the <code>$fac</code> slot, or its id, or a formula</p>
</td></tr>
<tr><td><code id="MANOVA_+3A_test">test</code></td>
<td>
<p>a test for <a href="stats.html#topic+manova">manova</a> (<code>'Hotelling'</code> by default)</p>
</td></tr>
<tr><td><code id="MANOVA_+3A_retain">retain</code></td>
<td>
<p>how many harmonics (or polynomials) to retain, for PCA
the highest number of PC axis to retain, or the proportion of the variance to capture.</p>
</td></tr>
<tr><td><code id="MANOVA_+3A_drop">drop</code></td>
<td>
<p>how many harmonics (or polynomials) to drop</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Performs a MANOVA/MANCOVA on PC scores. Just a wrapper around <a href="stats.html#topic+manova">manova</a>. See examples for multifactorial manova and
<a href="stats.html#topic+summary.manova">summary.manova</a> for more details and examples.
</p>


<h3>Value</h3>

<p>a list of matrices of (x,y) coordinates.
</p>


<h3>Note</h3>

<p>Needs a review and should be considered as experimental.
Silent message and progress bars (if any) with <code>options("verbose"=FALSE)</code>.
</p>


<h3>See Also</h3>

<p>Other multivariate: 
<code><a href="#topic+CLUST">CLUST</a>()</code>,
<code><a href="#topic+KMEANS">KMEANS</a>()</code>,
<code><a href="#topic+KMEDOIDS">KMEDOIDS</a>()</code>,
<code><a href="#topic+LDA">LDA</a>()</code>,
<code><a href="#topic+MANOVA_PW">MANOVA_PW</a>()</code>,
<code><a href="#topic+MDS">MDS</a>()</code>,
<code><a href="#topic+MSHAPES">MSHAPES</a>()</code>,
<code><a href="#topic+NMDS">NMDS</a>()</code>,
<code><a href="#topic+PCA">PCA</a>()</code>,
<code><a href="#topic+classification_metrics">classification_metrics</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># MANOVA
bot.p &lt;- PCA(efourier(bot, 12))
MANOVA(bot.p, 'type')

op &lt;- PCA(npoly(olea, 5))
MANOVA(op, 'domes')

 m &lt;- manova(op$x[, 1:5] ~  op$fac$domes * op$fac$var)
 summary(m)
 summary.aov(m)

 # MANCOVA example
 # we create a numeric variable, based on centroid size
 bot %&lt;&gt;% mutate(cs=coo_centsize(.))
 # same pipe
 bot %&gt;% efourier %&gt;% PCA %&gt;% MANOVA("cs")

</code></pre>

<hr>
<h2 id='MANOVA_PW'>Pairwise Multivariate analyses of variance</h2><span id='topic+MANOVA_PW'></span><span id='topic+MANOVA_PW.PCA'></span>

<h3>Description</h3>

<p>A wrapper for pairwise <a href="#topic+MANOVA">MANOVA</a>s on <a href="#topic+Coe">Coe</a> objects. Calculates a MANOVA for every
pairwise combination of the factor provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MANOVA_PW(x, ...)

## S3 method for class 'PCA'
MANOVA_PW(x, fac, retain = 0.99, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MANOVA_PW_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+PCA">PCA</a> object</p>
</td></tr>
<tr><td><code id="MANOVA_PW_+3A_...">...</code></td>
<td>
<p>more arguments to feed <a href="#topic+MANOVA">MANOVA</a></p>
</td></tr>
<tr><td><code id="MANOVA_PW_+3A_fac">fac</code></td>
<td>
<p>a name (or its id) of a grouping factor in <code>$fac</code> or a factor or a formula.</p>
</td></tr>
<tr><td><code id="MANOVA_PW_+3A_retain">retain</code></td>
<td>
<p>the number of PC axis to retain (1:retain) or the proportion of variance to capture (0.99 par default).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following components is returned (invisibly because $manovas
may be very long, see examples):
</p>

<ul>
<li><p> manovas a list containing all the raw manovas
</p>
</li>
<li><p> summary
</p>
</li>
<li><p> stars.tab a table with 'significance stars', discutable but largely used:
'<em><strong>' if Pr(&gt;F) &lt; 0.001; '</strong>' of &lt; 0.01; '</em>' if &lt; 0.05; '.' if &lt; 0.10 and '-' if above.
</p>
</li></ul>



<h3>Note</h3>

<p>Needs a review and should be considered as experimental.
If the fac passed has only two levels, there is only pair and it is
equivalent to <a href="#topic+MANOVA">MANOVA</a>. <code>MANOVA_PW.PCA</code> works with the regular <a href="stats.html#topic+manova">manova</a>.
</p>


<h3>See Also</h3>

<p><a href="#topic+MANOVA">MANOVA</a>, <a href="stats.html#topic+manova">manova</a>.
</p>
<p>Other multivariate: 
<code><a href="#topic+CLUST">CLUST</a>()</code>,
<code><a href="#topic+KMEANS">KMEANS</a>()</code>,
<code><a href="#topic+KMEDOIDS">KMEDOIDS</a>()</code>,
<code><a href="#topic+LDA">LDA</a>()</code>,
<code><a href="#topic+MANOVA">MANOVA</a>()</code>,
<code><a href="#topic+MDS">MDS</a>()</code>,
<code><a href="#topic+MSHAPES">MSHAPES</a>()</code>,
<code><a href="#topic+NMDS">NMDS</a>()</code>,
<code><a href="#topic+PCA">PCA</a>()</code>,
<code><a href="#topic+classification_metrics">classification_metrics</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># we create a fake factor with 4 levels
bot$fac$fake &lt;- factor(rep(letters[1:4], each=10))
bot.p &lt;- PCA(efourier(bot, 8))
MANOVA_PW(bot.p, 'fake') # or MANOVA_PW(bot.p, 2)

# an example on open outlines
op &lt;- PCA(npoly(olea))
MANOVA_PW(op, 'domes')
# to get the results
res &lt;- MANOVA_PW(op, 'domes')
res$manovas
res$stars.tab
res$summary
</code></pre>

<hr>
<h2 id='MDS'>(Metric) multidimensional scaling</h2><span id='topic+MDS'></span>

<h3>Description</h3>

<p>A wrapper around <a href="stats.html#topic+cmdscale">stats::cmdscale</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MDS(x, method = "euclidean", k = 2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MDS_+3A_x">x</code></td>
<td>
<p>any <a href="#topic+Coe">Coe</a> object</p>
</td></tr>
<tr><td><code id="MDS_+3A_method">method</code></td>
<td>
<p>a dissiminarity index to feed <code>method</code> in <a href="stats.html#topic+dist">stats::dist</a> (default: <code>euclidean</code>)</p>
</td></tr>
<tr><td><code id="MDS_+3A_k">k</code></td>
<td>
<p><code>numeric</code> number of dimensions to feed <a href="stats.html#topic+cmdscale">stats::cmdscale</a> (default: 2)</p>
</td></tr>
<tr><td><code id="MDS_+3A_...">...</code></td>
<td>
<p>additional parameters to feed <a href="stats.html#topic+cmdscale">stats::cmdscale</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For Details, see <a href="vegan.html#topic+metaMDS">vegan::metaMDS</a>
</p>


<h3>Value</h3>

<p>what is returned by <a href="stats.html#topic+dist">stats::dist</a> plus <code style="white-space: pre;">&#8288;$fac&#8288;</code>. And prepend <code>MDS</code> class to it.
</p>


<h3>See Also</h3>

<p>Other multivariate: 
<code><a href="#topic+CLUST">CLUST</a>()</code>,
<code><a href="#topic+KMEANS">KMEANS</a>()</code>,
<code><a href="#topic+KMEDOIDS">KMEDOIDS</a>()</code>,
<code><a href="#topic+LDA">LDA</a>()</code>,
<code><a href="#topic+MANOVA_PW">MANOVA_PW</a>()</code>,
<code><a href="#topic+MANOVA">MANOVA</a>()</code>,
<code><a href="#topic+MSHAPES">MSHAPES</a>()</code>,
<code><a href="#topic+NMDS">NMDS</a>()</code>,
<code><a href="#topic+PCA">PCA</a>()</code>,
<code><a href="#topic+classification_metrics">classification_metrics</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- bot %&gt;% efourier %&gt;% MDS
x


</code></pre>

<hr>
<h2 id='measure'>Measures shape descriptors</h2><span id='topic+measure'></span>

<h3>Description</h3>

<p>Calculates shape descriptors on Coo and other objects.
Any function that returns a scalar when fed coordinates can be passed
and naturally those of Momocs (pick some there <code>apropos("coo_")</code>). Functions
without arguments (eg <a href="#topic+coo_area">coo_area</a>) have to be passed without brackets but
functions with arguments (eg <a href="#topic+d">d</a>) have to be passed &quot;entirely&quot;. See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>measure(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="measure_+3A_x">x</code></td>
<td>
<p>any <code>Coo</code> object, or a list of shapes, or a shape as a matrix.</p>
</td></tr>
<tr><td><code id="measure_+3A_...">...</code></td>
<td>
<p>a list of functions. See examples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="#topic+TraCoe">TraCoe</a> object, or a raw data.frame
</p>


<h3>See Also</h3>

<p>Other premodern: 
<code><a href="#topic+coo_truss">coo_truss</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bm &lt;- measure(bot, coo_area, coo_perim)
bm
bm$coe

# how to use arguments, eg with the d() function
measure(wings, coo_area, d(1, 3), d(4, 5))

# alternatively, to get a data_frame
measure(bot$coo, coo_area, coo_perim)

# and also, to get a data_frame (one row)
measure(bot[1], coo_area, coo_perim)
</code></pre>

<hr>
<h2 id='molars'>Data: Outline coordinates of 360 molars</h2><span id='topic+molars'></span>

<h3>Description</h3>

<p>Courtesy of Julien Corny and Florent Detroit.
</p>


<h3>Format</h3>

<p>A <a href="#topic+Out">Out</a> object containing 79 equilinearly spaced (x; y)
coordinates for 360 crown outlines, of modern human molars,
along with their type (<code>$type</code>) - 90 first upper molars (UM1), 90 second upper molars (UM2),
90 first lower molars (LM1), 90 second lower molars (LM2) -
and the individual (<code>ind</code>) they come from (the data of the 360 molars are taken
from 180 individuals).
</p>


<h3>Source</h3>

<p>Corny, J., &amp; Detroit, F. (2014). Technical Note: Anatomic
identification of isolated modern human molars: testing Procrustes aligned
outlines as a standardization procedure for elliptic fourier analysis.
<em>American Journal of Physical Anthropology</em>, 153(2), 314-22. doi:10.1002/ajpa.22428
see <code style="white-space: pre;">&#8288;https://onlinelibrary.wiley.com/doi/abs/10.1002/ajpa.22428&#8288;</code>
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='Momocs'>Momocs</h2><span id='topic+Momocs'></span>

<h3>Description</h3>

<p>The goal of Momocs is to provide a complete, convenient,
reproducible and open-source toolkit for 2D morphometrics.
It includes most common 2D morphometrics approaches on outlines,
open outlines, configurations of landmarks, traditional
morphometrics, and facilities for data preparation,
manipulation and visualization with a consistent grammar throughout.
It allows reproducible, complex morphometric analyses
and other morphometrics approaches should be easy to
plug in, or develop from, on top of this canvas.
</p>


<h3>Details</h3>

<p>To cite Momocs in publications: <code>citation("Momocs")</code>.
</p>


<h3>Value</h3>

<p>nothing
</p>


<h3>Cheers</h3>

<p>We are very grateful to (in alphabetical order): Sean Asselin, Laurent Bouby, Matt Bulbert, Simon Crameri, Julia Cooke, April Dinwiddie,
Carl Lipo, Cedric Gaucherel, Catherine Girard, QGouil (GitHub), Christian Steven Hoggard,
Sarah Ivorra, Glynis Jones, Nathalie Keller, Ricardo Kriebel, Remi Laffont, Fabien Lafuma,
Matthias Mace, Stas Malavin, Neus Martinez, Ben Marwick, Sabrina Renaud, Marcelo Reginato,
Evan Saitta, Bill Sellers, David Siddons, Eleanor Stillman,
Theodore Stammer, Tom Stubbs, Norbert Telmon, Jean-Frederic Terral,
Bill Venables, Daniele Ventura, Michael Wallace,
Asher Wishkerman, John Wood for their helpful ideas and bug reports.
</p>


<h3>References</h3>


<ul>
<li><p> Bonhomme V, Picq S, Gaucherel C, Claude J. 2014. Momocs: Outline Analysis Using R.
<em>Journal of Statistical Software</em> <b>56</b>. <a href="https://www.jstatsoft.org/v56/i13">https://www.jstatsoft.org/v56/i13</a>.
</p>
</li>
<li><p> Claude J. 2008. <em>Morphometrics with R</em>. Springer-Verlag, New-York.
</p>
</li></ul>



<h3>See Also</h3>


<ul>
<li> <p><b>Homepage</b>: <a href="https://github.com/MomX/Momocs">https://github.com/MomX/Momocs</a>
</p>
</li>
<li> <p><b>Issues</b>: <a href="https://github.com/MomX/Momocs/issues">https://github.com/MomX/Momocs/issues</a>
</p>
</li>
<li> <p><b>Tutorial</b>: <code>browseVignettes("Momocs")</code> or <a href="http://momx.github.io/Momocs/">http://momx.github.io/Momocs/</a>
</p>
</li>
<li> <p><b>Email</b>: <code>bonhomme.vincent@gmail.com</code> to contribute to dev,
ask for something, propose collaboration, share your data, etc.
</p>
</li></ul>


<hr>
<h2 id='morphospace_positions'>Calculates nice positions on a plane for drawing shapes</h2><span id='topic+morphospace_positions'></span>

<h3>Description</h3>

<p>Calculates nice positions on a plane for drawing shapes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>morphospace_positions(
  xy,
  pos.shp = c("range", "full", "circle", "xy", "range_axes", "full_axes")[1],
  nb.shp = 12,
  nr.shp = 6,
  nc.shp = 5,
  circle.r.shp
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="morphospace_positions_+3A_xy">xy</code></td>
<td>
<p>a matrix of points typically from a PCA or other multivariate method on
which morphospace can be calculated</p>
</td></tr>
<tr><td><code id="morphospace_positions_+3A_pos.shp">pos.shp</code></td>
<td>
<p>how shapes should be positionned: <code>range</code> of xy,
<code>full</code> extent of the plane, <code>circle</code> as a rosewind,
on <code>xy</code> values provided, <code>range_axes</code> on the range of xy
but on the axes, <code>full_axes</code> same thing but on (0.85) range of the axes.
You can also directly pass a matrix (or a data.frame)
with columns named <code>("x", "y")</code>.</p>
</td></tr>
<tr><td><code id="morphospace_positions_+3A_nb.shp">nb.shp</code></td>
<td>
<p>the total number of shapes</p>
</td></tr>
<tr><td><code id="morphospace_positions_+3A_nr.shp">nr.shp</code></td>
<td>
<p>the number of rows to position shapes</p>
</td></tr>
<tr><td><code id="morphospace_positions_+3A_nc.shp">nc.shp</code></td>
<td>
<p>the number of cols to position shapes</p>
</td></tr>
<tr><td><code id="morphospace_positions_+3A_circle.r.shp">circle.r.shp</code></td>
<td>
<p>if circle, its radius</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <a href="#topic+plot.PCA">plot.PCA</a> for self-speaking examples
</p>


<h3>Value</h3>

<p>a data.frame of positions
</p>

<hr>
<h2 id='mosaic_engine'>Plots mosaics of shapes.</h2><span id='topic+mosaic_engine'></span><span id='topic+mosaic'></span><span id='topic+mosaic.Out'></span><span id='topic+mosaic.Opn'></span><span id='topic+mosaic.Ldk'></span>

<h3>Description</h3>

<p>Will soon replace <a href="#topic+panel">panel</a>. See examples and vignettes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mosaic_engine(
  coo_list,
  dim,
  asp = 1,
  byrow = TRUE,
  fromtop = TRUE,
  sample = 60,
  relatively = FALSE,
  template_size = 0.92
)

mosaic(x, ...)

## S3 method for class 'Out'
mosaic(
  x,
  f,
  relatively = FALSE,
  pal = pal_qual,
  sample = 60,
  paper_fun = paper_white,
  draw_fun = draw_outlines,
  legend = TRUE,
  dim = NA,
  asp = 1,
  byrow = TRUE,
  fromtop = TRUE,
  ...
)

## S3 method for class 'Opn'
mosaic(
  x,
  f,
  relatively = FALSE,
  pal = pal_qual,
  sample = 60,
  paper_fun = paper_white,
  draw_fun = draw_curves,
  legend = TRUE,
  dim = NA,
  asp = 1,
  byrow = TRUE,
  fromtop = TRUE,
  ...
)

## S3 method for class 'Ldk'
mosaic(
  x,
  f,
  relatively = FALSE,
  pal = pal_qual,
  sample = 60,
  paper_fun = paper_white,
  draw_fun = draw_landmarks,
  legend = TRUE,
  dim = NA,
  asp = 1,
  byrow = TRUE,
  fromtop = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mosaic_engine_+3A_coo_list">coo_list</code></td>
<td>
<p><code>list</code> of shapes</p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_dim">dim</code></td>
<td>
<p><code>numeric</code> of length 2, the desired dimensions for rows and columns</p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_asp">asp</code></td>
<td>
<p><code>numeric</code> the yx ratio used to calculate <code>dim</code> (1 by default).</p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_byrow">byrow</code></td>
<td>
<p><code>logical</code> whether to order shapes by rows</p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_fromtop">fromtop</code></td>
<td>
<p><code>logical</code> whether to order shapes from top</p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_sample">sample</code></td>
<td>
<p><code>numeric</code> number of points to <a href="#topic+coo_sample">coo_sample</a></p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_relatively">relatively</code></td>
<td>
<p><code>logical</code> if <code>TRUE</code> use <a href="#topic+coo_template_relatively">coo_template_relatively</a>
or, if <code>FALSE</code>(by default) <a href="#topic+coo_template">coo_template</a>.
In other words, whether to preserve size or not.</p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_template_size">template_size</code></td>
<td>
<p><code>numeric</code> to feed <code style="white-space: pre;">&#8288;coo_template(_relatively)&#8288;</code>.
Only useful to add padding around shapes when the default value (0.95) is lowered.</p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_x">x</code></td>
<td>
<p>any <a href="#topic+Coo">Coo</a> object</p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_...">...</code></td>
<td>
<p>additional arguments to feed the main drawer
if the number of shapes is &gt; 1000 (default: 64).
If non-numeric (eg <code>FALSE</code>) do not sample.</p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_f">f</code></td>
<td>
<p>factor specification to feed <a href="#topic+fac_dispatcher">fac_dispatcher</a></p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_pal">pal</code></td>
<td>
<p>one of <a href="#topic+palettes">palettes</a></p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_paper_fun">paper_fun</code></td>
<td>
<p>a <a href="#topic+papers">papers</a> function (default: <code>paper</code>)</p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_draw_fun">draw_fun</code></td>
<td>
<p>one of <a href="#topic+drawers">drawers</a> for <code>pile.list</code></p>
</td></tr>
<tr><td><code id="mosaic_engine_+3A_legend">legend</code></td>
<td>
<p><code>logical</code> whether to draw a legend (will be improved in further versions)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of templated and translated shapes
</p>


<h3>See Also</h3>

<p>Other grindr: 
<code><a href="#topic+drawers">drawers</a></code>,
<code><a href="#topic+layers_morphospace">layers_morphospace</a></code>,
<code><a href="#topic+layers">layers</a></code>,
<code><a href="#topic+papers">papers</a></code>,
<code><a href="#topic+pile">pile</a>()</code>,
<code><a href="#topic+plot_LDA">plot_LDA</a>()</code>,
<code><a href="#topic+plot_NMDS">plot_NMDS</a>()</code>,
<code><a href="#topic+plot_PCA">plot_PCA</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# On Out ---
bot %&gt;% mosaic
bot %&gt;% mosaic(~type)

# As with other grindr functions you can continue the pipe
bot %&gt;% mosaic(~type, asp=0.5) %&gt;% draw_firstpoint

# On Opn ---- same grammar
olea %&gt;% mosaic(~view+var, paper_fun=paper_dots)

 # On Ldk
 mosaic(wings, ~group, pal=pal_qual_Dark2, pch=3)

 # On Out with different sizes
 # would work on other Coo too
shapes2 &lt;- shapes
sizes &lt;- runif(30, 1, 2)
shapes2 %&gt;% mosaic(relatively=FALSE)
shapes2 %&gt;% mosaic(relatively=TRUE) %&gt;% draw_centroid()
</code></pre>

<hr>
<h2 id='mosquito'>Data: Outline coordinates of mosquito wings.</h2><span id='topic+mosquito'></span>

<h3>Description</h3>

<p>Data: Outline coordinates of mosquito wings.
</p>


<h3>Format</h3>

<p>A <a href="#topic+Out">Out</a> object with the 126 mosquito wing outlines outlines
used Rohlf and Archie (1984). Note that the links defined here are quite approximate.
</p>


<h3>Source</h3>

<p>Rohlf F, Archie J. 1984. A comparison of Fourier methods for the
description of wing shape in mosquitoes (Diptera: Culicidae). <em>Systematic Biology</em>: 302-317.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='mouse'>Data: Outline coordinates of mouse molars</h2><span id='topic+mouse'></span>

<h3>Description</h3>

<p>Data: Outline coordinates of mouse molars
</p>


<h3>Format</h3>

<p>A <a href="#topic+Out">Out</a> object 64 coordinates of 30 wood molar outlines.
</p>


<h3>Source</h3>

<p>Renaud S, Dufour AB, Hardouin EA, Ledevin R, Auffray JC (2015):
Once upon multivariate analyses: When they tell several stories about biological evolution.
<em>PLoS One</em> 10:1-18 <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0132801">https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0132801</a>
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='MSHAPES'>Mean shape calculation for Coo, Coe, etc.</h2><span id='topic+MSHAPES'></span>

<h3>Description</h3>

<p>Quite a versatile function that calculates mean (or median, or whatever function)
on list or an array of shapes, an Ldk object. It can also be used on Coe objects.
In that case, the reverse transformation (from coefficients to shapes) is calculated, (within
groups defined with the fac argument if provided) and the Coe object is <em>also</em> returned
(in <code style="white-space: pre;">&#8288;$Coe&#8288;</code>) along with a list of shapes (in <code style="white-space: pre;">&#8288;$shp&#8288;</code>) and can then be passed to <code>plot_MSHAPES</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MSHAPES(x, fac = NULL, FUN = mean, nb.pts = 120, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MSHAPES_+3A_x">x</code></td>
<td>
<p>a list, array, Ldk, LdkCoe, OutCoe or OpnCoe or PCA object</p>
</td></tr>
<tr><td><code id="MSHAPES_+3A_fac">fac</code></td>
<td>
<p>factor specification for <a href="#topic+fac_dispatcher">fac_dispatcher</a></p>
</td></tr>
<tr><td><code id="MSHAPES_+3A_fun">FUN</code></td>
<td>
<p>a function to compute the mean shape (<a href="base.html#topic+mean">mean</a> by default, by <a href="stats.html#topic+median">median</a> can be considered)</p>
</td></tr>
<tr><td><code id="MSHAPES_+3A_nb.pts">nb.pts</code></td>
<td>
<p>numeric the number of points for calculated shapes (only Coe objects)</p>
</td></tr>
<tr><td><code id="MSHAPES_+3A_...">...</code></td>
<td>
<p>useless here.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the averaged shape; on Coe objects, a list with two components: <code>$Coe</code> object of the same class, and
<code>$shp</code> a list of matrices of (x, y) coordinates. On <a href="#topic+PCA">PCA</a> and <a href="#topic+LDA">LDA</a> objects, the FUN (typically mean or median)
of scores on <code>PCs</code> or <code>LDs</code>. This method used on the latter objects may be moved to another function at some point.
</p>


<h3>See Also</h3>

<p>Other multivariate: 
<code><a href="#topic+CLUST">CLUST</a>()</code>,
<code><a href="#topic+KMEANS">KMEANS</a>()</code>,
<code><a href="#topic+KMEDOIDS">KMEDOIDS</a>()</code>,
<code><a href="#topic+LDA">LDA</a>()</code>,
<code><a href="#topic+MANOVA_PW">MANOVA_PW</a>()</code>,
<code><a href="#topic+MANOVA">MANOVA</a>()</code>,
<code><a href="#topic+MDS">MDS</a>()</code>,
<code><a href="#topic+NMDS">NMDS</a>()</code>,
<code><a href="#topic+PCA">PCA</a>()</code>,
<code><a href="#topic+classification_metrics">classification_metrics</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#### on shapes
MSHAPES(wings)
MSHAPES(wings$coo)
MSHAPES(coo_sample(bot, 24)$coo)
stack(wings)
coo_draw(MSHAPES(wings))

bot.f &lt;- efourier(bot, 12)
MSHAPES(bot.f) # the mean (global) shape
ms &lt;- MSHAPES(bot.f, 'type')
ms$Coe
class(ms$Coe)
ms &lt;- ms$shp
coo_plot(ms$beer)
coo_draw(ms$whisky, border='forestgreen')
</code></pre>

<hr>
<h2 id='mutate'>Add new variables</h2><span id='topic+mutate'></span>

<h3>Description</h3>

<p>Add new variables to the <code>$fac</code>. See examples and <code>?dplyr::mutate</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutate(.data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutate_+3A_.data">.data</code></td>
<td>
<p>a <code>Coo</code>, <code>Coe</code>, <code>PCA</code> object</p>
</td></tr>
<tr><td><code id="mutate_+3A_...">...</code></td>
<td>
<p>comma separated list of unquoted expressions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>dplyr verbs are maintained.
</p>


<h3>Value</h3>

<p>a Momocs object of the same class.
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>olea
mutate(olea, id=factor(1:length(olea)))
</code></pre>

<hr>
<h2 id='NMDS'>Non metric multidimensional scaling</h2><span id='topic+NMDS'></span>

<h3>Description</h3>

<p>A wrapper around <a href="vegan.html#topic+metaMDS">vegan::metaMDS</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NMDS(x, distance = "bray", k = 2, try = 20, trymax = 20, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NMDS_+3A_x">x</code></td>
<td>
<p>any <a href="#topic+Coe">Coe</a> object</p>
</td></tr>
<tr><td><code id="NMDS_+3A_distance">distance</code></td>
<td>
<p>a dissiminarity index to feed <a href="vegan.html#topic+vegdist">vegan::vegdist</a> (default: <code>bray</code>)</p>
</td></tr>
<tr><td><code id="NMDS_+3A_k">k</code></td>
<td>
<p><code>numeric</code> number of dimensions to feed <a href="vegan.html#topic+metaMDS">vegan::metaMDS</a> (default: 2)</p>
</td></tr>
<tr><td><code id="NMDS_+3A_try">try</code></td>
<td>
<p><code>numeric</code> minimum number of random starts to feed <a href="vegan.html#topic+metaMDS">vegan::metaMDS</a> (default: 20)</p>
</td></tr>
<tr><td><code id="NMDS_+3A_trymax">trymax</code></td>
<td>
<p><code>numeric</code> minimum number of random starts to feed <a href="vegan.html#topic+metaMDS">vegan::metaMDS</a> (default: 20)</p>
</td></tr>
<tr><td><code id="NMDS_+3A_...">...</code></td>
<td>
<p>additional parameters to feed <a href="vegan.html#topic+metaMDS">vegan::metaMDS</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>For Details, see <a href="vegan.html#topic+metaMDS">vegan::metaMDS</a>
</p>


<h3>Value</h3>

<p>what is returned by <a href="vegan.html#topic+metaMDS">vegan::metaMDS</a> plus <code style="white-space: pre;">&#8288;$fac&#8288;</code>. And prepend <code>NMDS</code> class to it.
</p>


<h3>See Also</h3>

<p>Other multivariate: 
<code><a href="#topic+CLUST">CLUST</a>()</code>,
<code><a href="#topic+KMEANS">KMEANS</a>()</code>,
<code><a href="#topic+KMEDOIDS">KMEDOIDS</a>()</code>,
<code><a href="#topic+LDA">LDA</a>()</code>,
<code><a href="#topic+MANOVA_PW">MANOVA_PW</a>()</code>,
<code><a href="#topic+MANOVA">MANOVA</a>()</code>,
<code><a href="#topic+MDS">MDS</a>()</code>,
<code><a href="#topic+MSHAPES">MSHAPES</a>()</code>,
<code><a href="#topic+PCA">PCA</a>()</code>,
<code><a href="#topic+classification_metrics">classification_metrics</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- bot %&gt;% efourier %&gt;% NMDS

# Shepard diagram # before a Momocs wrapper
# vegan::stressplot(x)

</code></pre>

<hr>
<h2 id='npoly'>Calculate natural polynomial fits on open outlines</h2><span id='topic+npoly'></span><span id='topic+npoly.default'></span><span id='topic+npoly.Opn'></span><span id='topic+npoly.list'></span>

<h3>Description</h3>

<p>Calculates natural polynomial coefficients,
through a linear model fit (see <a href="stats.html#topic+lm">lm</a>), from a matrix of (x; y) coordinates
or an <a href="#topic+Opn">Opn</a> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>npoly(x, ...)

## Default S3 method:
npoly(x, degree, ...)

## S3 method for class 'Opn'
npoly(
  x,
  degree,
  baseline1 = c(-0.5, 0),
  baseline2 = c(0.5, 0),
  nb.pts = 120,
  ...
)

## S3 method for class 'list'
npoly(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="npoly_+3A_x">x</code></td>
<td>
<p>a matrix (or a list) of (x; y) coordinates or an <a href="#topic+Opn">Opn</a> object</p>
</td></tr>
<tr><td><code id="npoly_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
<tr><td><code id="npoly_+3A_degree">degree</code></td>
<td>
<p>polynomial degree for the fit (the Intercept is also returned)</p>
</td></tr>
<tr><td><code id="npoly_+3A_baseline1">baseline1</code></td>
<td>
<p>numeric the <code class="reqn">(x; y)</code> coordinates of the first baseline
by default <code class="reqn">(x= -0.5; y=0)</code></p>
</td></tr>
<tr><td><code id="npoly_+3A_baseline2">baseline2</code></td>
<td>
<p>numeric the <code class="reqn">(x; y)</code> coordinates of the second baseline
by default <code class="reqn">(x= 0.5; y=0)</code></p>
</td></tr>
<tr><td><code id="npoly_+3A_nb.pts">nb.pts</code></td>
<td>
<p>number of points to sample and on which to calculate polynomials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>when applied on a single shape, a list with components:
</p>

<ul>
<li> <p><code>coeff</code> the coefficients (including the intercept)
</p>
</li>
<li> <p><code>ortho</code> whether orthogonal or natural polynomials were fitted
</p>
</li>
<li> <p><code>degree</code> degree of the fit (could be retrieved through <code>coeff</code> though)
</p>
</li>
<li> <p><code>baseline1</code> the first baseline point (so far the first point)
</p>
</li>
<li> <p><code>baseline2</code> the second baseline point (so far the last point)
</p>
</li>
<li> <p><code>r2</code> the r2 from the fit
</p>
</li>
<li> <p><code>mod</code> the raw lm model
</p>
</li></ul>

<p>otherwise, an <a href="#topic+OpnCoe">OpnCoe</a> object.
</p>


<h3>See Also</h3>

<p>Other polynomials: 
<code><a href="#topic+opoly_i">opoly_i</a>()</code>,
<code><a href="#topic+opoly">opoly</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(olea)
o &lt;- olea[1]
op &lt;- opoly(o, degree=4)
op
# shape reconstruction
opi &lt;- opoly_i(op)
coo_plot(o)
coo_draw(opi, border="red")
# R2 for degree 1 to 10
r &lt;- numeric()
for (i in 1:10) { r[i] &lt;- npoly(o, degree=i)$r2 }
plot(2:10, r[2:10], type='b', pch=20, col='red', main='R2 / degree')
</code></pre>

<hr>
<h2 id='nsfishes'>Data: Outline coordinates of North Sea fishes</h2><span id='topic+nsfishes'></span>

<h3>Description</h3>

<p>Data: Outline coordinates of North Sea fishes
</p>


<h3>Format</h3>

<p>A <a href="#topic+Out">Out</a> object containing the outlines coordinates for 218 fishes from
the North Sea along with taxonomical cofactors.
</p>


<h3>Source</h3>

<p>Caillon F, Frelat R, Mollmann C, Bonhomme V (submitted)
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='oak'>Data: Configuration of landmarks of oak leaves</h2><span id='topic+oak'></span>

<h3>Description</h3>

<p>From Viscosi and Cardini (2001).
</p>


<h3>Format</h3>

<p>A <a href="#topic+Ldk">Ldk</a> object containing 11 (x; y) landmarks  from 176 oak leaves wings, from
</p>


<h3>Source</h3>

<p>Viscosi, V., &amp; Cardini, A. (2011). Leaf morphology, taxonomy and
geometric morphometrics: a simplified protocol for beginners.
PloS One, 6(10), e25630. doi:10.1371/journal.pone.0025630
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='olea'>Data: Outline coordinates of olive seeds open outlines.</h2><span id='topic+olea'></span>

<h3>Description</h3>

<p>Data: Outline coordinates of olive seeds open outlines.
</p>


<h3>Format</h3>

<p>An <a href="#topic+Opn">Opn</a> object with the outline coordinates of olive seeds.
</p>


<h3>Source</h3>

<p>We thank Jean-Frederic Terral and Sarah Ivorra (UMR CBAE, Montpellier, France)
from allowing us to share the data.
</p>
<p>You can have a look to the original paper:
Terral J-F, Alonso N, Capdevila RB i, Chatti N, Fabre L, Fiorentino G,
Marinval P, Jorda GP, Pradat B, Rovira N, et al. 2004.
Historical biogeography of olive domestication (<em>Olea europaea</em> L.)
as revealed by geometrical morphometry applied to biological and
archaeological material. <em>Journal of Biogeography</em> <b>31</b>: 63-77.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='Opn'>Builds an Opn object</h2><span id='topic+Opn'></span>

<h3>Description</h3>

<p>In Momocs, <code>Opn</code> classes objects are
lists of <b>op</b>e<b>n</b> outlines, with optionnal components,
on which generic methods such as plotting methods (e.g. <a href="#topic+stack">stack</a>)
and specific methods (e.g. <a href="#topic+npoly">npoly</a> can be applied.
<code><a href="#topic+Opn">Opn</a></code> objects are primarily <code><a href="#topic+Coo">Coo</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Opn(x, fac = dplyr::tibble(), ldk = list())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Opn_+3A_x">x</code></td>
<td>
<p><code>list</code> of matrices of (x; y) coordinates, or an array, or a data.frame (and friends)</p>
</td></tr>
<tr><td><code id="Opn_+3A_fac">fac</code></td>
<td>
<p>(optionnal) a <code>data.frame</code> of factors and/or numerics
specifying the grouping structure</p>
</td></tr>
<tr><td><code id="Opn_+3A_ldk">ldk</code></td>
<td>
<p>(optionnal) <code>list</code> of landmarks as row number indices</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an <code>Opn</code> object
</p>


<h3>See Also</h3>

<p>Other classes: 
<code><a href="#topic+Coe">Coe</a>()</code>,
<code><a href="#topic+Coo">Coo</a>()</code>,
<code><a href="#topic+Ldk">Ldk</a>()</code>,
<code><a href="#topic+OpnCoe">OpnCoe</a>()</code>,
<code><a href="#topic+OutCoe">OutCoe</a>()</code>,
<code><a href="#topic+Out">Out</a>()</code>,
<code><a href="#topic+TraCoe">TraCoe</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Methods on Opn
methods(class=Opn)
# we load some open outlines. See ?olea for credits
olea
panel(olea)
# orthogonal polynomials
op &lt;- opoly(olea, degree=5)
# we print the Coe
op
# Let's do a PCA on it
op.p &lt;- PCA(op)
plot(op.p, 'domes')
plot(op.p, 'var')
# and now an LDA after a PCA
olda &lt;- LDA(PCA(op), 'var')
# for CV table and others
olda
plot_LDA(olda)
</code></pre>

<hr>
<h2 id='OpnCoe'>Builds an OpnCoe object</h2><span id='topic+OpnCoe'></span>

<h3>Description</h3>

<p>In Momocs, <code>OpnCoe</code> classes objects are wrapping around
lists of morphometric coefficients, along with other informations,
on which generic methods such as plotting methods (e.g. <a href="graphics.html#topic+boxplot">boxplot</a>)
and specific methods can be applied.
<code>OpnCoe</code> objects are primarily <code><a href="#topic+Coe">Coe</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OpnCoe(
  coe = matrix(),
  fac = dplyr::tibble(),
  method = character(),
  baseline1 = numeric(),
  baseline2 = numeric(),
  mod = list(),
  r2 = numeric()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OpnCoe_+3A_coe">coe</code></td>
<td>
<p><code>matrix</code> of morphometric coefficients</p>
</td></tr>
<tr><td><code id="OpnCoe_+3A_fac">fac</code></td>
<td>
<p>(optionnal) a <code>data.frame</code> of factors,
specifying the grouping structure</p>
</td></tr>
<tr><td><code id="OpnCoe_+3A_method">method</code></td>
<td>
<p>used to obtain these coefficients</p>
</td></tr>
<tr><td><code id="OpnCoe_+3A_baseline1">baseline1</code></td>
<td>
<p><code class="reqn">(x; y)</code> coordinates of the first baseline point</p>
</td></tr>
<tr><td><code id="OpnCoe_+3A_baseline2">baseline2</code></td>
<td>
<p><code class="reqn">(x; y)</code> coordinates of the second baseline point</p>
</td></tr>
<tr><td><code id="OpnCoe_+3A_mod">mod</code></td>
<td>
<p>an R <a href="stats.html#topic+lm">lm</a> object, used to reconstruct shapes</p>
</td></tr>
<tr><td><code id="OpnCoe_+3A_r2">r2</code></td>
<td>
<p>numeric, the r-squared from every model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an <code>OpnCoe</code> object
</p>


<h3>See Also</h3>

<p>Other classes: 
<code><a href="#topic+Coe">Coe</a>()</code>,
<code><a href="#topic+Coo">Coo</a>()</code>,
<code><a href="#topic+Ldk">Ldk</a>()</code>,
<code><a href="#topic+Opn">Opn</a>()</code>,
<code><a href="#topic+OutCoe">OutCoe</a>()</code>,
<code><a href="#topic+Out">Out</a>()</code>,
<code><a href="#topic+TraCoe">TraCoe</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># all OpnCoe classes
methods(class='OpnCoe')
</code></pre>

<hr>
<h2 id='opoly'>Calculate orthogonal polynomial fits on open outlines</h2><span id='topic+opoly'></span><span id='topic+opoly.default'></span><span id='topic+opoly.Opn'></span><span id='topic+opoly.list'></span>

<h3>Description</h3>

<p>Calculates orthogonal polynomial coefficients,
through a linear model fit (see <a href="stats.html#topic+lm">lm</a>), from a matrix of (x; y) coordinates or
a <a href="#topic+Opn">Opn</a> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opoly(x, ...)

## Default S3 method:
opoly(x, degree, ...)

## S3 method for class 'Opn'
opoly(
  x,
  degree,
  baseline1 = c(-0.5, 0),
  baseline2 = c(0.5, 0),
  nb.pts = 120,
  ...
)

## S3 method for class 'list'
opoly(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="opoly_+3A_x">x</code></td>
<td>
<p>a matrix (or a list) of (x; y) coordinates</p>
</td></tr>
<tr><td><code id="opoly_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
<tr><td><code id="opoly_+3A_degree">degree</code></td>
<td>
<p>polynomial degree for the fit (the Intercept is also returned)</p>
</td></tr>
<tr><td><code id="opoly_+3A_baseline1">baseline1</code></td>
<td>
<p>numeric the <code class="reqn">(x; y)</code> coordinates of the first baseline
by default <code class="reqn">(x= -0.5; y=0)</code></p>
</td></tr>
<tr><td><code id="opoly_+3A_baseline2">baseline2</code></td>
<td>
<p>numeric the <code class="reqn">(x; y)</code> coordinates of the second baseline
by default <code class="reqn">(x= 0.5; y=0)</code></p>
</td></tr>
<tr><td><code id="opoly_+3A_nb.pts">nb.pts</code></td>
<td>
<p>number of points to sample and on which to calculate polynomials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with components when applied on a single shape:
</p>

<ul>
<li> <p><code>coeff</code> the coefficients (including the intercept)
</p>
</li>
<li> <p><code>ortho</code> whether orthogonal or natural polynomials were fitted
</p>
</li>
<li> <p><code>degree</code> degree of the fit (could be retrieved through <code>coeff</code> though)
</p>
</li>
<li> <p><code>baseline1</code> the first baseline point (so far the first point)
</p>
</li>
<li> <p><code>baseline2</code> the second baseline point (so far the last point)
</p>
</li>
<li> <p><code>r2</code> the r2 from the fit
</p>
</li>
<li> <p><code>mod</code> the raw lm model
</p>
</li></ul>

<p>otherwise an <a href="#topic+OpnCoe">OpnCoe</a> object.
</p>


<h3>Note</h3>

<p>Orthogonal polynomials are sometimes called Legendre's polynomials. They are
preferred over natural polynomials since adding a degree do not change lower orders coefficients.
</p>


<h3>See Also</h3>

<p>Other polynomials: 
<code><a href="#topic+npoly">npoly</a>()</code>,
<code><a href="#topic+opoly_i">opoly_i</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(olea)
o &lt;- olea[1]
op &lt;- opoly(o, degree=4)
op
# shape reconstruction
opi &lt;- opoly_i(op)
coo_plot(o)
coo_draw(opi)
lines(opi, col='red')
# R2 for degree 1 to 10
r &lt;- numeric()
for (i in 1:10) { r[i] &lt;- opoly(o, degree=i)$r2 }
plot(2:10, r[2:10], type='b', pch=20, col='red', main='R2 / degree')
</code></pre>

<hr>
<h2 id='opoly_i'>Calculates shape from a polynomial model</h2><span id='topic+opoly_i'></span><span id='topic+npoly_i'></span>

<h3>Description</h3>

<p>Returns a matrix of (x; y) coordinates when passed with a list obtained with
<a href="#topic+opoly">opoly</a> or <a href="#topic+npoly">npoly</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opoly_i(pol, nb.pts = 120, reregister = TRUE)

npoly_i(pol, nb.pts = 120, reregister = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="opoly_i_+3A_pol">pol</code></td>
<td>
<p>a pol list such as created by <a href="#topic+npoly">npoly</a> or <a href="#topic+opoly">opoly</a></p>
</td></tr>
<tr><td><code id="opoly_i_+3A_nb.pts">nb.pts</code></td>
<td>
<p>the number of points to predict. By default (and cannot be higher)
the number of points in the original shape.</p>
</td></tr>
<tr><td><code id="opoly_i_+3A_reregister">reregister</code></td>
<td>
<p>logical whether to reregister the shape with the original baseline.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix of (x; y) coordinates.
</p>


<h3>See Also</h3>

<p>Other polynomials: 
<code><a href="#topic+npoly">npoly</a>()</code>,
<code><a href="#topic+opoly">opoly</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(olea)
o &lt;- olea[5]
coo_plot(o)
for (i in 2:7){
x &lt;- opoly_i(opoly(o, i))
coo_draw(x, border=col_summer(7)[i], points=FALSE)  }
</code></pre>

<hr>
<h2 id='Out'>Builds an Out object</h2><span id='topic+Out'></span>

<h3>Description</h3>

<p>In Momocs, <code>Out</code>-classes objects are lists of closed <b>out</b>lines,
with optional components, and on which generic methods such as plotting methods (e.g. <a href="#topic+stack">stack</a>)
and specific methods (e.g. <a href="#topic+efourier">efourier</a> can be applied.
<code>Out</code> objects are primarily <code><a href="#topic+Coo">Coo</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Out(x, fac = dplyr::tibble(), ldk = list())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Out_+3A_x">x</code></td>
<td>
<p>a <code>list</code> of matrices of <code style="white-space: pre;">&#8288;(x; y)&#8288;</code> coordinates,
or an array or an Out object or an Ldk object, or a data.frame (and friends)</p>
</td></tr>
<tr><td><code id="Out_+3A_fac">fac</code></td>
<td>
<p>(optional) a <code>data.frame</code> of factors and/or numerics
specifying the grouping structure</p>
</td></tr>
<tr><td><code id="Out_+3A_ldk">ldk</code></td>
<td>
<p>(optional) <code>list</code> of landmarks as row number indices</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an <code>Out</code> object
</p>


<h3>See Also</h3>

<p>Other classes: 
<code><a href="#topic+Coe">Coe</a>()</code>,
<code><a href="#topic+Coo">Coo</a>()</code>,
<code><a href="#topic+Ldk">Ldk</a>()</code>,
<code><a href="#topic+OpnCoe">OpnCoe</a>()</code>,
<code><a href="#topic+Opn">Opn</a>()</code>,
<code><a href="#topic+OutCoe">OutCoe</a>()</code>,
<code><a href="#topic+TraCoe">TraCoe</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>methods(class=Out)
</code></pre>

<hr>
<h2 id='OutCoe'>Builds an OutCoe object</h2><span id='topic+OutCoe'></span>

<h3>Description</h3>

<p>In Momocs, <code>OutCoe</code> classes objects are wrapping around
lists of morphometric coefficients, along with other informations,
on which generic methods such as plotting methods (e.g. <a href="graphics.html#topic+boxplot">boxplot</a>)
and specific methods can be applied.
<code>OutCoe</code> objects are primarily <code><a href="#topic+Coe">Coe</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OutCoe(coe = matrix(), fac = dplyr::tibble(), method, norm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OutCoe_+3A_coe">coe</code></td>
<td>
<p><code>matrix</code> of harmonic coefficients</p>
</td></tr>
<tr><td><code id="OutCoe_+3A_fac">fac</code></td>
<td>
<p>(optional) a <code>data.frame</code> of factors,
specifying the grouping structure</p>
</td></tr>
<tr><td><code id="OutCoe_+3A_method">method</code></td>
<td>
<p>used to obtain these coefficients</p>
</td></tr>
<tr><td><code id="OutCoe_+3A_norm">norm</code></td>
<td>
<p>the normalisation used to obtain these coefficients</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These methods can be applied on <code>Out</code> objects:
</p>


<h3>Value</h3>

<p>an <code>OutCoe</code> object
</p>


<h3>See Also</h3>

<p>Other classes: 
<code><a href="#topic+Coe">Coe</a>()</code>,
<code><a href="#topic+Coo">Coo</a>()</code>,
<code><a href="#topic+Ldk">Ldk</a>()</code>,
<code><a href="#topic+OpnCoe">OpnCoe</a>()</code>,
<code><a href="#topic+Opn">Opn</a>()</code>,
<code><a href="#topic+Out">Out</a>()</code>,
<code><a href="#topic+TraCoe">TraCoe</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># all OutCoe methods
methods(class='OutCoe')
</code></pre>

<hr>
<h2 id='palettes'>Color palettes</h2><span id='topic+palettes'></span><span id='topic+pal_alpha'></span><span id='topic+pal'></span><span id='topic+palette'></span><span id='topic+pal_manual'></span><span id='topic+pal_qual_solarized'></span><span id='topic+pal_seq_grey'></span><span id='topic+pal_div_BrBG'></span><span id='topic+pal_div_PiYG'></span><span id='topic+pal_div_PRGn'></span><span id='topic+pal_div_PuOr'></span><span id='topic+pal_div_RdBu'></span><span id='topic+pal_div_RdYlBu'></span><span id='topic+pal_qual_Dark2'></span><span id='topic+pal_qual_Paired'></span><span id='topic+pal_qual_Set2'></span><span id='topic+pal_seq_Blues'></span><span id='topic+pal_seq_BuGn'></span><span id='topic+pal_seq_BuPu'></span><span id='topic+pal_seq_GnBu'></span><span id='topic+pal_seq_Greens'></span><span id='topic+pal_seq_Greys'></span><span id='topic+pal_seq_Oranges'></span><span id='topic+pal_seq_OrRd'></span><span id='topic+pal_seq_PuBu'></span><span id='topic+pal_seq_PuBuGn'></span><span id='topic+pal_seq_PuRd'></span><span id='topic+pal_seq_Purples'></span><span id='topic+pal_seq_RdPu'></span><span id='topic+pal_seq_Reds'></span><span id='topic+pal_seq_YlGn'></span><span id='topic+pal_seq_YlGnBu'></span><span id='topic+pal_seq_YlOrBr'></span><span id='topic+pal_seq_YlOrRd'></span><span id='topic+pal_seq_magma'></span><span id='topic+pal_seq_inferno'></span><span id='topic+pal_seq_plasma'></span><span id='topic+pal_seq_viridis'></span><span id='topic+pal_qual'></span><span id='topic+pal_seq'></span><span id='topic+pal_div'></span>

<h3>Description</h3>

<p>All colorblind friendly RColorBrewer palettes
recreated without the number of colors limitation
and with transparency support thanks to <code>pal_alpha</code> that can be used alone.
Also, all viridis palettes (see the <a href="https://CRAN.R-project.org/package=viridis">package on CRAN</a>),
yet color ramps are borrowed and Momocs does not depend on it.
Also, <code>pal_qual_solarized</code> based on Solarized: <a href="https://ethanschoonover.com/solarized/">https://ethanschoonover.com/solarized/</a>
and <code>pal_seq_grey</code> only shades of grey from <code>grey10</code> to <code>grey90</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pal_alpha(cols, transp = 0)

pal_manual(cols, transp = 0)

pal_qual_solarized(n, transp = 0)

pal_seq_grey(n, transp = 0)

pal_div_BrBG(n, transp = 0)

pal_div_PiYG(n, transp = 0)

pal_div_PRGn(n, transp = 0)

pal_div_PuOr(n, transp = 0)

pal_div_RdBu(n, transp = 0)

pal_div_RdYlBu(n, transp = 0)

pal_qual_Dark2(n, transp = 0)

pal_qual_Paired(n, transp = 0)

pal_qual_Set2(n, transp = 0)

pal_seq_Blues(n, transp = 0)

pal_seq_BuGn(n, transp = 0)

pal_seq_BuPu(n, transp = 0)

pal_seq_GnBu(n, transp = 0)

pal_seq_Greens(n, transp = 0)

pal_seq_Greys(n, transp = 0)

pal_seq_Oranges(n, transp = 0)

pal_seq_OrRd(n, transp = 0)

pal_seq_PuBu(n, transp = 0)

pal_seq_PuBuGn(n, transp = 0)

pal_seq_PuRd(n, transp = 0)

pal_seq_Purples(n, transp = 0)

pal_seq_RdPu(n, transp = 0)

pal_seq_Reds(n, transp = 0)

pal_seq_YlGn(n, transp = 0)

pal_seq_YlGnBu(n, transp = 0)

pal_seq_YlOrBr(n, transp = 0)

pal_seq_YlOrRd(n, transp = 0)

pal_seq_magma(n, transp = 0)

pal_seq_inferno(n, transp = 0)

pal_seq_plasma(n, transp = 0)

pal_seq_viridis(n, transp = 0)

pal_qual(n, transp = 0)

pal_seq(n, transp = 0)

pal_div(n, transp = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="palettes_+3A_cols">cols</code></td>
<td>
<p>color(s) as hexadecimal values</p>
</td></tr>
<tr><td><code id="palettes_+3A_transp">transp</code></td>
<td>
<p><code>numeric</code> between 0 and 1 (0, eg opaque, by default)</p>
</td></tr>
<tr><td><code id="palettes_+3A_n">n</code></td>
<td>
<p><code>numeric</code> number of colors</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Default color palettes are currently:
</p>

<ul>
<li> <p><code>pal_qual=pal_qual_Set2</code>
</p>
</li>
<li> <p><code>pal_seq=pal_seq_viridis</code>
</p>
</li>
<li> <p><code>pal_div=pal_div_RdBu</code>
</p>
</li></ul>



<h3>Value</h3>

<p>a palette function
</p>


<h3>Note</h3>

<p>RColorBrewer palettes are not happy when <code>n</code> is lower than 3 and above
a given number for each palette. If this is the case, these functions will
create a color palette with <a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a> and return colors even so.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pal_div_BrBG(5) %&gt;% barplot(rep(1, 5), col=.)
pal_div_BrBG(5, 0.5) %&gt;% barplot(rep(1, 5), col=.)
</code></pre>

<hr>
<h2 id='panel'>Family picture of shapes</h2><span id='topic+panel'></span><span id='topic+panel.Coo'></span><span id='topic+panel.Out'></span><span id='topic+panel.Opn'></span><span id='topic+panel.Ldk'></span>

<h3>Description</h3>

<p>Plots all the outlines, side by side, from
a <a href="#topic+Coo">Coo</a> (<a href="#topic+Out">Out</a>, <a href="#topic+Opn">Opn</a> or <a href="#topic+Ldk">Ldk</a>) objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>panel(x, ...)

## S3 method for class 'Out'
panel(
  x,
  dim,
  cols,
  borders,
  fac,
  palette = col_summer,
  coo_sample = 120,
  names = NULL,
  cex.names = 0.6,
  points = TRUE,
  points.pch = 3,
  points.cex = 0.2,
  points.col,
  ...
)

## S3 method for class 'Opn'
panel(
  x,
  cols,
  borders,
  fac,
  palette = col_summer,
  coo_sample = 120,
  names = NULL,
  cex.names = 0.6,
  points = TRUE,
  points.pch = 3,
  points.cex = 0.2,
  points.col,
  ...
)

## S3 method for class 'Ldk'
panel(
  x,
  cols,
  borders,
  fac,
  palette = col_summer,
  names = NULL,
  cex.names = 0.6,
  points = TRUE,
  points.pch = 3,
  points.cex = 0.2,
  points.col = "#333333",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="panel_+3A_x">x</code></td>
<td>
<p>The <code>Coo</code> object to plot.</p>
</td></tr>
<tr><td><code id="panel_+3A_...">...</code></td>
<td>
<p>additional arguments to feed generic <code>plot</code></p>
</td></tr>
<tr><td><code id="panel_+3A_dim">dim</code></td>
<td>
<p>for <a href="#topic+coo_listpanel">coo_listpanel</a>: a numeric of length 2
specifying the dimensions of the panel</p>
</td></tr>
<tr><td><code id="panel_+3A_cols">cols</code></td>
<td>
<p>A <code>vector</code> of colors for drawing the outlines.
Either a single value or of length exactly equal to the number of coordinates.</p>
</td></tr>
<tr><td><code id="panel_+3A_borders">borders</code></td>
<td>
<p>A <code>vector</code> of colors for drawing the borders.
Either a single value or of length exactly equals to the number of coordinates.</p>
</td></tr>
<tr><td><code id="panel_+3A_fac">fac</code></td>
<td>
<p>a factor within the $fac slot for colors</p>
</td></tr>
<tr><td><code id="panel_+3A_palette">palette</code></td>
<td>
<p>a color <a href="#topic+palette">palette</a></p>
</td></tr>
<tr><td><code id="panel_+3A_coo_sample">coo_sample</code></td>
<td>
<p>if not NULL the number of point per shape to display (to plot quickly)</p>
</td></tr>
<tr><td><code id="panel_+3A_names">names</code></td>
<td>
<p>whether to plot names or not. If TRUE uses shape names, or something for <a href="#topic+fac_dispatcher">fac_dispatcher</a></p>
</td></tr>
<tr><td><code id="panel_+3A_cex.names">cex.names</code></td>
<td>
<p>a cex for the names</p>
</td></tr>
<tr><td><code id="panel_+3A_points">points</code></td>
<td>
<p><code>logical</code> (for Ldk) whether to draw points</p>
</td></tr>
<tr><td><code id="panel_+3A_points.pch">points.pch</code></td>
<td>
<p>(for Ldk) and a pch for these points</p>
</td></tr>
<tr><td><code id="panel_+3A_points.cex">points.cex</code></td>
<td>
<p>(for Ldk) and a cex for these points</p>
</td></tr>
<tr><td><code id="panel_+3A_points.col">points.col</code></td>
<td>
<p>(for Ldk) and a col  for these points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>Note</h3>

<p>If you want to reorder shapes according to a factor, use <a href="#topic+arrange">arrange</a>.
</p>


<h3>See Also</h3>

<p>Other Coo_graphics: 
<code><a href="#topic+inspect">inspect</a>()</code>,
<code><a href="#topic+stack">stack</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>panel(mosquito, names=TRUE, cex.names=0.5)
panel(olea)
panel(bot, c(4, 10))
# an illustration of the use of fac
panel(bot, fac='type', palette=col_spring, names=TRUE)
</code></pre>

<hr>
<h2 id='papers'>grindr papers for shape plots</h2><span id='topic+papers'></span><span id='topic+paper'></span><span id='topic+paper_white'></span><span id='topic+paper_grid'></span><span id='topic+paper_chess'></span><span id='topic+paper_dots'></span>

<h3>Description</h3>

<p>Papers on which to use <a href="#topic+drawers">drawers</a> for building custom
shape plots using the grindr approach. See examples and vignettes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>paper(coo, ...)

paper_white(coo)

paper_grid(coo, grid = c(10, 5), cols = c("#ffa500", "#e5e5e5"), ...)

paper_chess(coo, n = 50, col = "#E5E5E5")

paper_dots(coo, pch = 20, n = 50, col = "#7F7F7F")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="papers_+3A_coo">coo</code></td>
<td>
<p>a single shape or any <a href="#topic+Coo">Coo</a> object</p>
</td></tr>
<tr><td><code id="papers_+3A_...">...</code></td>
<td>
<p>more arguments to feed the plotting function within each <code>paper</code> function</p>
</td></tr>
<tr><td><code id="papers_+3A_grid">grid</code></td>
<td>
<p><code>numeric</code> of length 2 to (roughly) specify the
number of majors lines, and the number of minor lines within two major ones</p>
</td></tr>
<tr><td><code id="papers_+3A_cols">cols</code></td>
<td>
<p>colors (hexadecimal preferred) to use for grid drawing</p>
</td></tr>
<tr><td><code id="papers_+3A_n">n</code></td>
<td>
<p><code>numeric</code> number of squares for the chessboard</p>
</td></tr>
<tr><td><code id="papers_+3A_col">col</code></td>
<td>
<p>color (hexadecimal) to use for chessboard drawing</p>
</td></tr>
<tr><td><code id="papers_+3A_pch">pch</code></td>
<td>
<p>to use for dots</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing layer
</p>


<h3>Note</h3>

<p>This approach will (soon) replace <a href="#topic+coo_plot">coo_plot</a> and friends in further versions.
All comments are welcome.
</p>


<h3>See Also</h3>

<p>Other grindr: 
<code><a href="#topic+drawers">drawers</a></code>,
<code><a href="#topic+layers_morphospace">layers_morphospace</a></code>,
<code><a href="#topic+layers">layers</a></code>,
<code><a href="#topic+mosaic_engine">mosaic_engine</a>()</code>,
<code><a href="#topic+pile">pile</a>()</code>,
<code><a href="#topic+plot_LDA">plot_LDA</a>()</code>,
<code><a href="#topic+plot_NMDS">plot_NMDS</a>()</code>,
<code><a href="#topic+plot_PCA">plot_PCA</a>()</code>
</p>

<hr>
<h2 id='PCA'>Principal component analysis on Coe objects</h2><span id='topic+PCA'></span><span id='topic+PCA.OutCoe'></span><span id='topic+PCA.OpnCoe'></span><span id='topic+PCA.LdkCoe'></span><span id='topic+PCA.TraCoe'></span><span id='topic+PCA.default'></span><span id='topic+as_PCA'></span>

<h3>Description</h3>

<p>Performs a PCA on <a href="#topic+Coe">Coe</a> objects, using <a href="stats.html#topic+prcomp">prcomp</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCA(x, scale., center, fac)

## S3 method for class 'OutCoe'
PCA(x, scale. = FALSE, center = TRUE, fac)

## S3 method for class 'OpnCoe'
PCA(x, scale. = FALSE, center = TRUE, fac)

## S3 method for class 'LdkCoe'
PCA(x, scale. = FALSE, center = TRUE, fac)

## S3 method for class 'TraCoe'
PCA(x, scale. = TRUE, center = TRUE, fac)

## Default S3 method:
PCA(x, scale. = TRUE, center = TRUE, fac = dplyr::tibble())

as_PCA(x, fac)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PCA_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+Coe">Coe</a> object or an appropriate object (eg <a href="stats.html#topic+prcomp">prcomp</a>) for <code>as_PCA</code></p>
</td></tr>
<tr><td><code id="PCA_+3A_scale.">scale.</code></td>
<td>
<p>logical whether to scale the input data</p>
</td></tr>
<tr><td><code id="PCA_+3A_center">center</code></td>
<td>
<p>logical whether to center the input data</p>
</td></tr>
<tr><td><code id="PCA_+3A_fac">fac</code></td>
<td>
<p>any factor or data.frame to be passed to <code>as_PCA</code> and for use with <a href="#topic+plot.PCA">plot.PCA</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, methods on <a href="#topic+Coe">Coe</a> object do not scale the input data but center them.
There is also a generic method (eg for traditional morphometrics) that centers and scales data.
</p>


<h3>Value</h3>

<p>a 'PCA' object on which to apply <a href="#topic+plot.PCA">plot.PCA</a>, among others. This list has several
components, most of them inherited from the <code>prcomp</code> object:
</p>

<ol>
<li> <p><code>sdev</code> the standard deviations of the principal components
(i.e., the square roots of the eigenvalues of the
covariance/correlation matrix, though the calculation
is actually done with the singular values of the data matrix)
</p>
</li>
<li> <p><code>eig</code> the cumulated proportion of variance along the PC axes
</p>
</li>
<li> <p><code>rotation</code> the matrix of variable loadings (i.e., a matrix whose columns contain the eigenvectors).
The function princomp returns this in the element loadings.
</p>
</li>
<li> <p><code>center</code>, scale the centering and scaling used
</p>
</li>
<li> <p><code>x</code> PCA scores (the value of the rotated data (the centred (and scaled if requested)
data multiplied by the rotation matrix))
</p>
</li>
<li><p> other components are inherited from the <code>Coe</code> object passed to <code>PCA</code>,
eg <code>fac</code>, <code>mshape</code>, <code>method</code>, <code>baseline1</code> and <code>baseline2</code>, etc. They
are documented in the corresponding <code>*Coe</code> file.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other multivariate: 
<code><a href="#topic+CLUST">CLUST</a>()</code>,
<code><a href="#topic+KMEANS">KMEANS</a>()</code>,
<code><a href="#topic+KMEDOIDS">KMEDOIDS</a>()</code>,
<code><a href="#topic+LDA">LDA</a>()</code>,
<code><a href="#topic+MANOVA_PW">MANOVA_PW</a>()</code>,
<code><a href="#topic+MANOVA">MANOVA</a>()</code>,
<code><a href="#topic+MDS">MDS</a>()</code>,
<code><a href="#topic+MSHAPES">MSHAPES</a>()</code>,
<code><a href="#topic+NMDS">NMDS</a>()</code>,
<code><a href="#topic+classification_metrics">classification_metrics</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot.f &lt;- efourier(bot, 12)
bot.p &lt;- PCA(bot.f)
bot.p
plot(bot.p, morpho=FALSE)
plot(bot.p, 'type')

op &lt;- npoly(olea, 5)
op.p &lt;- PCA(op)
op.p
plot(op.p, 1, morpho=TRUE)

wp &lt;- fgProcrustes(wings, tol=1e-4)
wpp &lt;- PCA(wp)
wpp
plot(wpp, 1)

# "foreign prcomp"
head(iris)
iris.p &lt;- prcomp(iris[, 1:4])
iris.p &lt;- as_PCA(iris.p, iris[, 5])
class(iris.p)
plot(iris.p, 1)
</code></pre>

<hr>
<h2 id='PCcontrib'>Shape variation along PC axes</h2><span id='topic+PCcontrib'></span><span id='topic+PCcontrib.PCA'></span>

<h3>Description</h3>

<p>Calculates and plots shape variation along Principal Component axes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCcontrib(PCA, ...)

## S3 method for class 'PCA'
PCcontrib(PCA, nax, sd.r = c(-2, -1, -0.5, 0, 0.5, 1, 2), gap = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PCcontrib_+3A_pca">PCA</code></td>
<td>
<p>a <code>PCA</code> object</p>
</td></tr>
<tr><td><code id="PCcontrib_+3A_...">...</code></td>
<td>
<p>additional parameter to pass to <code><a href="#topic+coo_draw">coo_draw</a></code></p>
</td></tr>
<tr><td><code id="PCcontrib_+3A_nax">nax</code></td>
<td>
<p>the range of PCs to plot (1 to 99pc total variance by default)</p>
</td></tr>
<tr><td><code id="PCcontrib_+3A_sd.r">sd.r</code></td>
<td>
<p>a single or a range of mean +/- sd values (eg: c(-1, 0, 1))</p>
</td></tr>
<tr><td><code id="PCcontrib_+3A_gap">gap</code></td>
<td>
<p>for combined-Coe, an adjustment variable for gap between shapes. (bug)Default
to 1 (whish should never superimpose shapes), reduce it to get a more compact plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>(invisibly) a list with <code>gg</code> the ggplot object and <code>shp</code> the list of shapes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot.p &lt;- PCA(efourier(bot, 12))
PCcontrib(bot.p, nax=1:3)

library(ggplot2)
gg &lt;- PCcontrib(bot.p, nax=1:8, sd.r=c(-5, -3, -2, -1, -0.5, 0, 0.5, 1, 2, 3, 5))
gg$gg + geom_polygon(fill="slategrey", col="black") + ggtitle("A nice title")

</code></pre>

<hr>
<h2 id='perm'>Permutes and breed Coe (and others) objects</h2><span id='topic+perm'></span><span id='topic+perm.default'></span><span id='topic+perm.Coe'></span>

<h3>Description</h3>

<p>This methods applies permutations column-wise on the <code>coe</code> of any
<a href="#topic+Coe">Coe</a> object but relies on a function that can be used on any matrix.
For a Coe object, it uses <a href="base.html#topic+sample">sample</a> on every column (or row) with (or without)
replacement.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perm(x, ...)

## Default S3 method:
perm(x, margin = 2, size, replace = TRUE, ...)

## S3 method for class 'Coe'
perm(x, size, replace = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perm_+3A_x">x</code></td>
<td>
<p>the object to permute</p>
</td></tr>
<tr><td><code id="perm_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
<tr><td><code id="perm_+3A_margin">margin</code></td>
<td>
<p>numeric whether 1 or 2 (rows or columns)</p>
</td></tr>
<tr><td><code id="perm_+3A_size">size</code></td>
<td>
<p>numeric the required size for the final object, same size by default.</p>
</td></tr>
<tr><td><code id="perm_+3A_replace">replace</code></td>
<td>
<p>logical, whether to use <a href="base.html#topic+sample">sample</a> with replacement</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Coe object of same class
</p>


<h3>See Also</h3>

<p>Other farming: 
<code><a href="#topic+breed">breed</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- matrix(1:12, nrow=3)
m
perm(m, margin=2, size=5)
perm(m, margin=1, size=10)

bot.f &lt;- efourier(bot, 12)
bot.m &lt;- perm(bot.f, 80)
bot.m
</code></pre>

<hr>
<h2 id='pile'>Graphical pile of shapes</h2><span id='topic+pile'></span><span id='topic+pile.default'></span><span id='topic+pile.list'></span><span id='topic+pile.array'></span><span id='topic+pile.Out'></span><span id='topic+pile.Opn'></span><span id='topic+pile.Ldk'></span>

<h3>Description</h3>

<p>Pile all shapes in the same graphical window. Useful to check
their normalization in terms of size, position, rotation, first point, etc.
It is, essentially, a shortcut around <code>paper + drawers</code> of the grindr family.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pile(coo, f, sample, subset, pal, paper_fun, draw_fun, transp, ...)

## Default S3 method:
pile(
  coo,
  f,
  sample,
  subset,
  pal = pal_qual,
  paper_fun = paper,
  draw_fun = draw_curves,
  transp = 0,
  ...
)

## S3 method for class 'list'
pile(
  coo,
  f,
  sample = 64,
  subset = 1000,
  pal = pal_qual,
  paper_fun = paper,
  draw_fun = draw_curves,
  transp = 0,
  ...
)

## S3 method for class 'array'
pile(
  coo,
  f,
  sample = 64,
  subset = 1000,
  pal = pal_qual,
  paper_fun = paper,
  draw_fun = draw_landmarks,
  transp = 0,
  ...
)

## S3 method for class 'Out'
pile(
  coo,
  f,
  sample = 64,
  subset = 1000,
  pal = pal_qual,
  paper_fun = paper,
  draw_fun = draw_outlines,
  transp = 0,
  ...
)

## S3 method for class 'Opn'
pile(
  coo,
  f,
  sample = 64,
  subset = 1000,
  pal = pal_qual,
  paper_fun = paper,
  draw_fun = draw_curves,
  transp = 0,
  ...
)

## S3 method for class 'Ldk'
pile(
  coo,
  f,
  sample = 64,
  subset = 1000,
  pal = pal_qual,
  paper_fun = paper,
  draw_fun = draw_landmarks,
  transp = 0,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pile_+3A_coo">coo</code></td>
<td>
<p>a single shape or any  <a href="#topic+Coo">Coo</a> object</p>
</td></tr>
<tr><td><code id="pile_+3A_f">f</code></td>
<td>
<p>factor specification</p>
</td></tr>
<tr><td><code id="pile_+3A_sample">sample</code></td>
<td>
<p><code>numeric</code> number of points to <a href="#topic+coo_sample">coo_sample</a>
if the number of shapes is &gt; 1000 (default: 64).
If non-numeric (eg <code>FALSE</code>) do not sample.</p>
</td></tr>
<tr><td><code id="pile_+3A_subset">subset</code></td>
<td>
<p><code>numeric</code> only draw this number of (randomly chosen) shapes if
the number of shapes is &gt; 1000 (default: 1000) If non-numeric (eg <code>FALSE</code>) do not sample.</p>
</td></tr>
<tr><td><code id="pile_+3A_pal">pal</code></td>
<td>
<p>palette among <a href="#topic+palettes">palettes</a> (default: pal_qual)</p>
</td></tr>
<tr><td><code id="pile_+3A_paper_fun">paper_fun</code></td>
<td>
<p>a <a href="#topic+papers">papers</a> function (default: <code>paper</code>)</p>
</td></tr>
<tr><td><code id="pile_+3A_draw_fun">draw_fun</code></td>
<td>
<p>one of <a href="#topic+drawers">drawers</a> for <code>pile.list</code></p>
</td></tr>
<tr><td><code id="pile_+3A_transp">transp</code></td>
<td>
<p><code>numeric</code> for transparency (default:adjusted, min:0, max=0)</p>
</td></tr>
<tr><td><code id="pile_+3A_...">...</code></td>
<td>
<p>more arguments to feed the core drawer, depending on the object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Large <code>Coo</code> are sampled, both in terms of the number of shapes and of points to drawn.
</p>


<h3>Value</h3>

<p>a plot
</p>


<h3>Note</h3>

<p>A variation of this plot was called <code>stack</code> before <code style="white-space: pre;">&#8288;Momocs 1.2.5&#8288;</code>
</p>


<h3>See Also</h3>

<p>Other grindr: 
<code><a href="#topic+drawers">drawers</a></code>,
<code><a href="#topic+layers_morphospace">layers_morphospace</a></code>,
<code><a href="#topic+layers">layers</a></code>,
<code><a href="#topic+mosaic_engine">mosaic_engine</a>()</code>,
<code><a href="#topic+papers">papers</a></code>,
<code><a href="#topic+plot_LDA">plot_LDA</a>()</code>,
<code><a href="#topic+plot_NMDS">plot_NMDS</a>()</code>,
<code><a href="#topic+plot_PCA">plot_PCA</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># all Coo are supported with sensible defaults
pile(bot)    # outlines
pile(olea, ~var, pal=pal_qual_Dark2, paper_fun=paper_grid)   # curves
pile(wings)  # landmarks

# you can continue the pipe with compatible drawers
pile(bot, trans=0.9) %&gt;% draw_centroid

# if you are not happy with this, build your own !
# eg see Momocs::pile.Out (no quotes)

my_pile &lt;- function(x, col_labels="red", transp=0.5){
    x %&gt;% paper_chess(n=100) %&gt;%
          draw_landmarks(transp=transp) %&gt;%
          draw_labels(col=col_labels)
}
# using it
wings %&gt;% my_pile(transp=3/4)

 # and as gridr functions propagate, you can even continue:
 wings %&gt;% my_pile() %&gt;% draw_centroid(col="blue", cex=5)

 # method on lists
 bot$coo %&gt;% pile

 # it can be tuned when we have a list of landmarks with:
 wings$coo %&gt;% pile(draw_fun=draw_landmarks)

 # or on arrays (turn for draw_landmarks)
 wings$coo %&gt;% l2a %&gt;% #we now have an array
     pile
</code></pre>

<hr>
<h2 id='pix2chc'>Convert (x; y) coordinates to chaincoded coordinates</h2><span id='topic+pix2chc'></span><span id='topic+chc2pix'></span>

<h3>Description</h3>

<p>Useful to convert (x; y) coordinates to chain-coded coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pix2chc(coo)

chc2pix(chc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pix2chc_+3A_coo">coo</code></td>
<td>
<p>(x; y) coordinates passed as a matrix</p>
</td></tr>
<tr><td><code id="pix2chc_+3A_chc">chc</code></td>
<td>
<p>chain coordinates</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix or a numeric
</p>


<h3>Note</h3>

<p>Note this function will be deprecated from Momocs
when <code>Momacs</code> and <code>Momit</code>
will be fully operationnal.
</p>


<h3>References</h3>

<p>Kuhl, F. P., &amp; Giardina, C. R. (1982).
Elliptic Fourier features of a closed contour.
<em>Computer Graphics and Image Processing</em>, 18(3), 236-258.
</p>


<h3>See Also</h3>

<p><a href="#topic+chc2pix">chc2pix</a>
</p>
<p>Other import functions: 
<code><a href="#topic+import_Conte">import_Conte</a>()</code>,
<code><a href="#topic+import_StereoMorph_curve1">import_StereoMorph_curve1</a>()</code>,
<code><a href="#topic+import_jpg1">import_jpg1</a>()</code>,
<code><a href="#topic+import_jpg">import_jpg</a>()</code>,
<code><a href="#topic+import_tps">import_tps</a>()</code>,
<code><a href="#topic+import_txt">import_txt</a>()</code>
</p>
<p>Other import functions: 
<code><a href="#topic+import_Conte">import_Conte</a>()</code>,
<code><a href="#topic+import_StereoMorph_curve1">import_StereoMorph_curve1</a>()</code>,
<code><a href="#topic+import_jpg1">import_jpg1</a>()</code>,
<code><a href="#topic+import_jpg">import_jpg</a>()</code>,
<code><a href="#topic+import_tps">import_tps</a>()</code>,
<code><a href="#topic+import_txt">import_txt</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pix2chc(shapes[1]) %T&gt;% print %&gt;% # from pix to chc
chc2pix()                         # and back
</code></pre>

<hr>
<h2 id='plot_CV'>Plots a cross-validation table as an heatmap</h2><span id='topic+plot_CV'></span><span id='topic+plot_CV.default'></span><span id='topic+plot_CV.LDA'></span>

<h3>Description</h3>

<p>Either with frequencies (or percentages) plus marginal sums,
and values as heatmaps. Used in Momocs for plotting cross-validation tables
but may be used for any table (likely with <code>freq=FALSE</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_CV(
  x,
  freq = FALSE,
  rm0 = FALSE,
  pc = FALSE,
  fill = TRUE,
  labels = TRUE,
  axis.size = 10,
  axis.x.angle = 45,
  cell.size = 2.5,
  signif = 2,
  ...
)

## Default S3 method:
plot_CV(
  x,
  freq = FALSE,
  rm0 = FALSE,
  pc = FALSE,
  fill = TRUE,
  labels = TRUE,
  axis.size = 10,
  axis.x.angle = 45,
  cell.size = 2.5,
  signif = 2,
  ...
)

## S3 method for class 'LDA'
plot_CV(
  x,
  freq = TRUE,
  rm0 = TRUE,
  pc = TRUE,
  fill = TRUE,
  labels = TRUE,
  axis.size = 10,
  axis.x.angle = 45,
  cell.size = 2.5,
  signif = 2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_CV_+3A_x">x</code></td>
<td>
<p>a (cross-validation table) or an LDA object</p>
</td></tr>
<tr><td><code id="plot_CV_+3A_freq">freq</code></td>
<td>
<p>logical whether to display frequencies (within an actual class) or counts</p>
</td></tr>
<tr><td><code id="plot_CV_+3A_rm0">rm0</code></td>
<td>
<p>logical whether to remove zeros</p>
</td></tr>
<tr><td><code id="plot_CV_+3A_pc">pc</code></td>
<td>
<p>logical whether to multiply proportion by 100, ie display percentages</p>
</td></tr>
<tr><td><code id="plot_CV_+3A_fill">fill</code></td>
<td>
<p>logical whether to fill cell according to count/freq</p>
</td></tr>
<tr><td><code id="plot_CV_+3A_labels">labels</code></td>
<td>
<p>logical whether to add text labels on cells</p>
</td></tr>
<tr><td><code id="plot_CV_+3A_axis.size">axis.size</code></td>
<td>
<p>numeric to adjust axis labels</p>
</td></tr>
<tr><td><code id="plot_CV_+3A_axis.x.angle">axis.x.angle</code></td>
<td>
<p>numeric to rotate x-axis labels</p>
</td></tr>
<tr><td><code id="plot_CV_+3A_cell.size">cell.size</code></td>
<td>
<p>numeric to adjust text labels on cells</p>
</td></tr>
<tr><td><code id="plot_CV_+3A_signif">signif</code></td>
<td>
<p>numeric to round frequencies using <a href="base.html#topic+signif">signif</a></p>
</td></tr>
<tr><td><code id="plot_CV_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>See Also</h3>

<p><a href="#topic+LDA">LDA</a>, <a href="#topic+plot.LDA">plot.LDA</a>, and (pretty much the same) <a href="#topic+plot_table">plot_table</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>h &lt;- hearts %&gt;%
     fgProcrustes(0.01) %&gt;% coo_slide(ldk=2) %T&gt;% stack %&gt;%
     efourier(6, norm=FALSE) %&gt;% LDA(~aut)

h %&gt;% plot_CV()
h %&gt;% plot_CV(freq=FALSE, rm0=FALSE, fill=FALSE)
# you can customize the returned gg with some ggplot2 functions
h %&gt;% plot_CV(labels=FALSE, fill=TRUE, axis.size=5) + ggplot2::ggtitle("A confusion matrix")

# or build your own using the prepared data_frame:
df &lt;- h %&gt;% plot_CV() %$% data
df

# you can even use it as a cross-table plotter
bot$fac %&gt;% table %&gt;% plot_CV()

</code></pre>

<hr>
<h2 id='plot_CV2'>Plots a cross-correlation table</h2><span id='topic+plot_CV2'></span><span id='topic+plot_CV2.LDA'></span><span id='topic+plot_CV2.table'></span>

<h3>Description</h3>

<p>Or any contingency/confusion table. A simple graphic representation based on variable
width and/or color for arrows or segments, based on the relative frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_CV2(x, ...)

## S3 method for class 'LDA'
plot_CV2(x, ...)

## S3 method for class 'table'
plot_CV2(
  x,
  links.FUN = arrows,
  col = TRUE,
  col0 = "black",
  col.breaks = 5,
  palette = col_heat,
  lwd = TRUE,
  lwd0 = 5,
  gap.dots = 0.2,
  pch.dots = 20,
  gap.names = 0.25,
  cex.names = 1,
  legend = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_CV2_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+LDA">LDA</a> object, a table or a squared matrix</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_...">...</code></td>
<td>
<p>useless here.</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_links.fun">links.FUN</code></td>
<td>
<p>a function to draw the links: eg <a href="graphics.html#topic+segments">segments</a> (by default), <a href="graphics.html#topic+arrows">arrows</a>, etc.</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_col">col</code></td>
<td>
<p>logical whether to vary the color of the links</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_col0">col0</code></td>
<td>
<p>a color for the default link (when <code>col = FALSE</code>)</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_col.breaks">col.breaks</code></td>
<td>
<p>the number of different colors</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_palette">palette</code></td>
<td>
<p>a color palette, eg <a href="#topic+col_summer">col_summer</a>, <a href="#topic+col_hot">col_hot</a>, etc.</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_lwd">lwd</code></td>
<td>
<p>logical whether to vary the width of the links</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_lwd0">lwd0</code></td>
<td>
<p>a width for the default link (when <code>lwd = FALSE</code>)</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_gap.dots">gap.dots</code></td>
<td>
<p>numeric to set space between the dots and the links</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_pch.dots">pch.dots</code></td>
<td>
<p>a pch for the dots</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_gap.names">gap.names</code></td>
<td>
<p>numeric to set the space between the dots and the group names</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_cex.names">cex.names</code></td>
<td>
<p>a cex for the names</p>
</td></tr>
<tr><td><code id="plot_CV2_+3A_legend">legend</code></td>
<td>
<p>logical whether to add a legend</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 object
</p>


<h3>Note</h3>

<p>When <code>freq=FALSE</code>, the fill colors are not weighted
within actual classes and should not be displayed if classes sizes are not balanced.
</p>


<h3>See Also</h3>

<p><a href="#topic+LDA">LDA</a>, <a href="#topic+plot.LDA">plot.LDA</a>, <a href="#topic+plot_CV">plot_CV</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Below various table that you can try. We will use the last one for the examples.
#pure random
a &lt;- sample(rep(letters[1:4], each=10))
b &lt;- sample(rep(letters[1:4], each=10))
tab &lt;- table(a, b)

# veryhuge + some structure
a &lt;- sample(rep(letters[1:10], each=10))
b &lt;- sample(rep(letters[1:10], each=10))
tab &lt;- table(a, b)
diag(tab) &lt;- round(runif(10, 10, 20))

tab &lt;- matrix(c(8, 3, 1, 0, 0,
                2, 7, 1, 2, 3,
                3, 5, 9, 1, 1,
                1, 1, 2, 7, 1,
                0, 9, 1, 4, 5), 5, 5, byrow=TRUE)
tab &lt;- as.table(tab)

# good prediction
tab &lt;- matrix(c(8, 1, 1, 0, 0,
               1, 7, 1, 0, 0,
                1, 2, 9, 1, 0,
                1, 1, 1, 7, 1,
                0, 0, 0, 1, 8), 5, 5, byrow=TRUE)
tab &lt;- as.table(tab)


plot_CV2(tab)
plot_CV2(tab, arrows) # if you prefer arrows
plot_CV2(tab, lwd=FALSE, lwd0=1, palette=col_india) # if you like india but not lwds
plot_CV2(tab, col=FALSE, col0='pink') # only lwd
plot_CV2(tab, col=FALSE, lwd0=10, cex.names=2) # if you're getting old
plot_CV2(tab, col=FALSE, lwd=FALSE) # pretty but useless
plot_CV2(tab, col.breaks=2) # if you think it's either good or bad
plot_CV2(tab, pch=NA) # if you do not like dots
plot_CV2(tab, gap.dots=0) # if you want to 'fill the gap'
plot_CV2(tab, gap.dots=1) # or not

#trilo examples
trilo.f &lt;- efourier(trilo, 8)
trilo.l &lt;- LDA(PCA(trilo.f), 'onto')
trilo.l
plot_CV2(trilo.l)

# olea example
op &lt;- opoly(olea, 5)
opl &lt;- LDA(PCA(op), 'var')
plot_CV2(opl)
</code></pre>

<hr>
<h2 id='plot_devsegments'>Draws colored segments from a matrix of coordinates.</h2><span id='topic+plot_devsegments'></span>

<h3>Description</h3>

<p>Given a matrix of (x; y) coordinates, draws segments between every points
defined by the row of the matrix and uses a color to display an information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_devsegments(coo, cols, lwd = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_devsegments_+3A_coo">coo</code></td>
<td>
<p>A matrix of coordinates.</p>
</td></tr>
<tr><td><code id="plot_devsegments_+3A_cols">cols</code></td>
<td>
<p>A vector of color of <code>length = nrow(coo)</code>.</p>
</td></tr>
<tr><td><code id="plot_devsegments_+3A_lwd">lwd</code></td>
<td>
<p>The <code>lwd</code> to use for drawing segments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing on the last plot
</p>


<h3>See Also</h3>

<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_table">plot_table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# we load some data
guinness &lt;- coo_sample(bot[9], 100)

# we calculate the diff between 48 harm and one with 6 harm.
out.6    &lt;- efourier_i(efourier(guinness, nb.h=6), nb.pts=120)

# we calculate deviations, you can also try 'edm'
dev &lt;- edm_nearest(out.6, guinness) / coo_centsize(out.6)

# we prepare the color scale
d.cut &lt;- cut(dev, breaks=20, labels=FALSE, include.lowest=TRUE)
cols  &lt;- paste0(col_summer(20)[d.cut], 'CC')

# we draw the results
coo_plot(guinness, main='Guiness fitted with 6 harm.', points=FALSE)
par(xpd=NA)
plot_devsegments(out.6, cols=cols, lwd=4)
coo_draw(out.6, lty=2, points=FALSE, col=NA)
par(xpd=FALSE)
</code></pre>

<hr>
<h2 id='plot_LDA'>LDA plot using grindr layers</h2><span id='topic+plot_LDA'></span>

<h3>Description</h3>

<p>Quickly vizualise <a href="#topic+LDA">LDA</a> objects and build customs plots
using the <a href="#topic+layers">layers</a>. See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_LDA(
  x,
  axes = c(1, 2),
  palette = pal_qual,
  points = TRUE,
  points_transp = 1/4,
  morphospace = FALSE,
  morphospace_position = "range",
  chull = TRUE,
  chullfilled = FALSE,
  labelgroups = FALSE,
  legend = TRUE,
  title = "",
  center_origin = TRUE,
  zoom = 0.9,
  eigen = TRUE,
  box = TRUE,
  iftwo_layer = layer_histogram_2,
  iftwo_split = FALSE,
  axesnames = TRUE,
  axesvar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_LDA_+3A_x">x</code></td>
<td>
<p><a href="#topic+LDA">LDA</a> object</p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_axes">axes</code></td>
<td>
<p><code>numeric</code> of length two to select PCs to use
(<code>c(1, 2)</code> by default)</p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_palette">palette</code></td>
<td>
<p><code>color palette</code> to use <code>col_summer</code> by default</p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_points">points</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_points">layer_points</a></p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_points_transp">points_transp</code></td>
<td>
<p><code>numeric</code> to feed <a href="#topic+layer_points">layer_points</a> (default:0.25)</p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_morphospace">morphospace</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_morphospace_PCA">layer_morphospace_PCA</a></p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_morphospace_position">morphospace_position</code></td>
<td>
<p>to feed <a href="#topic+layer_morphospace_PCA">layer_morphospace_PCA</a> (default: &quot;range&quot;)</p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_chull">chull</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_chull">layer_chull</a></p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_chullfilled">chullfilled</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_chullfilled">layer_chullfilled</a></p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_labelgroups">labelgroups</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_labelgroups">layer_labelgroups</a></p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_legend">legend</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_legend">layer_legend</a></p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_title">title</code></td>
<td>
<p><code>character</code> if specified, fee <a href="#topic+layer_title">layer_title</a> (default to <code>""</code>)</p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_center_origin">center_origin</code></td>
<td>
<p><code>logical</code> whether to center origin</p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_zoom">zoom</code></td>
<td>
<p><code>numeric</code> zoom level for the frame (default: 0.9)</p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_eigen">eigen</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_eigen">layer_eigen</a></p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_box">box</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_box">layer_box</a></p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_iftwo_layer">iftwo_layer</code></td>
<td>
<p>function (no quotes) for drawing LD1 when there are two levels.
So far, one of <a href="#topic+layer_histogram_2">layer_histogram_2</a> (default) or <a href="#topic+layer_density_2">layer_density_2</a></p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_iftwo_split">iftwo_split</code></td>
<td>
<p>to feed <code>split</code> argument in <a href="#topic+layer_histogram_2">layer_histogram_2</a> or <a href="#topic+layer_density_2">layer_density_2</a></p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_axesnames">axesnames</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_axesnames">layer_axesnames</a></p>
</td></tr>
<tr><td><code id="plot_LDA_+3A_axesvar">axesvar</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_axesvar">layer_axesvar</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>Note</h3>

<p>This approach will replace <a href="#topic+plot.LDA">plot.LDA</a>.
This is part of <code>grindr</code> approach that may be packaged at some point. All comments are welcome.
</p>


<h3>See Also</h3>

<p>Other grindr: 
<code><a href="#topic+drawers">drawers</a></code>,
<code><a href="#topic+layers_morphospace">layers_morphospace</a></code>,
<code><a href="#topic+layers">layers</a></code>,
<code><a href="#topic+mosaic_engine">mosaic_engine</a>()</code>,
<code><a href="#topic+papers">papers</a></code>,
<code><a href="#topic+pile">pile</a>()</code>,
<code><a href="#topic+plot_NMDS">plot_NMDS</a>()</code>,
<code><a href="#topic+plot_PCA">plot_PCA</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### First prepare an LDA object

# Some outlines with bot
bl &lt;- bot %&gt;%
      # cheap alignement before efourier
      coo_align() %&gt;% coo_center %&gt;% coo_slidedirection("left") %&gt;%
      # add a fake column
      mutate(fake=sample(letters[1:5], 40, replace=TRUE)) %&gt;%
      # EFT
      efourier(6, norm=FALSE) %&gt;%
      # LDA
      LDA(~fake)

bl %&gt;% plot_LDA %&gt;% layer_morphospace_LDA

# Below inherited from plot_PCA and to adapt here.
#plot_PCA(bp)
#plot_PCA(bp, ~type)
#plot_PCA(bp, ~fake)

# Some curves with olea
#op &lt;- olea %&gt;%
#mutate(s=coo_area(.)) %&gt;%
#filter(var != "Cypre") %&gt;%
#chop(~view) %&gt;% lapply(opoly, 5, nb.pts=90) %&gt;%
#combine %&gt;% PCA
#op$fac$s %&lt;&gt;% as.character() %&gt;% as.numeric()

#op %&gt;% plot_PCA(title="hi there!")

### Now we can play with layers
# and for instance build a custom plot
# it should start with plot_PCA()

#my_plot &lt;- function(x, ...){

#x %&gt;%
#     plot_PCA(...) %&gt;%
#    layer_points %&gt;%
#     layer_ellipsesaxes %&gt;%
#    layer_rug
# }

# and even continue after this function
# op %&gt;% my_plot(~var, axes=c(1, 3)) %&gt;%
#     layer_title("hi there!") %&gt;%
#    layer_stars()

# You get the idea.
</code></pre>

<hr>
<h2 id='plot_MSHAPES'>Pairwise comparison of a list of shapes</h2><span id='topic+plot_MSHAPES'></span>

<h3>Description</h3>

<p>&quot;Confusion matrix&quot; of a list of shapes. See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_MSHAPES(x, draw_fun, size, palette)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_MSHAPES_+3A_x">x</code></td>
<td>
<p>a list of shapes (eg as returned by <a href="#topic+MSHAPES">MSHAPES</a>)</p>
</td></tr>
<tr><td><code id="plot_MSHAPES_+3A_draw_fun">draw_fun</code></td>
<td>
<p>one of <a href="#topic+draw_outline">draw_outline</a>, <a href="#topic+draw_curves">draw_curves</a>, <a href="#topic+draw_landmarks">draw_landmarks</a>.
When the result of <a href="#topic+MSHAPES">MSHAPES</a> is passed, detected based on <code style="white-space: pre;">&#8288;$Coe&#8288;</code>, otherwise default to <code>draw_curves</code>.</p>
</td></tr>
<tr><td><code id="plot_MSHAPES_+3A_size">size</code></td>
<td>
<p>numeric shrinking factor for shapes (and <a href="#topic+coo_template">coo_template</a>;
3/4 by default)</p>
</td></tr>
<tr><td><code id="plot_MSHAPES_+3A_palette">palette</code></td>
<td>
<p>on of <a href="#topic+palettes">palettes</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>Note</h3>

<p>Directly inspired by Chitwood et al. (2016) in <em>New Phytologist</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- bot %&gt;% efourier(6) %&gt;% MSHAPES(~type)

# custom colors
x %&gt;% plot_MSHAPES(palette=pal_manual(c("darkgreen", "orange")))

# also works on list of shapes, eg:
leaves &lt;- shapes %&gt;% slice(grep("leaf", names(shapes))) %$% coo
class(leaves)
leaves %&gt;% plot_MSHAPES()

# or
shapes %&gt;%
# subset and degrade
slice(1:12) %&gt;% coo_sample(60) %$%  # grab the coo
    coo %&gt;%
    plot_MSHAPES()
</code></pre>

<hr>
<h2 id='plot_NMDS'>NMDS plot unsing grindr layers</h2><span id='topic+plot_NMDS'></span><span id='topic+plot_MDS'></span>

<h3>Description</h3>

<p>Quickly vizualise <a href="#topic+MDS">MDS</a> and <a href="#topic+NMDS">NMDS</a> objects and build customs plots
using the <a href="#topic+layers">layers</a>. See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_NMDS(
  x,
  f = NULL,
  axes = c(1, 2),
  points = TRUE,
  points_transp = 1/4,
  chull = TRUE,
  chullfilled = FALSE,
  labelgroups = FALSE,
  legend = TRUE,
  title = "",
  box = TRUE,
  axesnames = TRUE,
  palette = pal_qual
)

plot_MDS(
  x,
  f = NULL,
  axes = c(1, 2),
  points = TRUE,
  points_transp = 1/4,
  chull = TRUE,
  chullfilled = FALSE,
  labelgroups = FALSE,
  legend = TRUE,
  title = "",
  box = TRUE,
  axesnames = TRUE,
  palette = pal_qual
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_NMDS_+3A_x">x</code></td>
<td>
<p>the result of <a href="#topic+MDS">MDS</a> or <a href="#topic+NMDS">NMDS</a></p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_f">f</code></td>
<td>
<p>factor specification to feed <a href="#topic+fac_dispatcher">fac_dispatcher</a></p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_axes">axes</code></td>
<td>
<p><code>numeric</code> of length two to select PCs to use
(<code>c(1, 2)</code> by default)</p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_points">points</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_points">layer_points</a></p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_points_transp">points_transp</code></td>
<td>
<p><code>numeric</code> to feed <a href="#topic+layer_points">layer_points</a> (default:0.25)</p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_chull">chull</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_chull">layer_chull</a></p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_chullfilled">chullfilled</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_chullfilled">layer_chullfilled</a></p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_labelgroups">labelgroups</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_labelgroups">layer_labelgroups</a></p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_legend">legend</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_legend">layer_legend</a></p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_title">title</code></td>
<td>
<p><code>character</code> if specified, fee <a href="#topic+layer_title">layer_title</a> (default to <code>""</code>)</p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_box">box</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_box">layer_box</a></p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_axesnames">axesnames</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_axesnames">layer_axesnames</a></p>
</td></tr>
<tr><td><code id="plot_NMDS_+3A_palette">palette</code></td>
<td>
<p><code>color palette</code> to use <code>col_summer</code> by default</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>See Also</h3>

<p>Other grindr: 
<code><a href="#topic+drawers">drawers</a></code>,
<code><a href="#topic+layers_morphospace">layers_morphospace</a></code>,
<code><a href="#topic+layers">layers</a></code>,
<code><a href="#topic+mosaic_engine">mosaic_engine</a>()</code>,
<code><a href="#topic+papers">papers</a></code>,
<code><a href="#topic+pile">pile</a>()</code>,
<code><a href="#topic+plot_LDA">plot_LDA</a>()</code>,
<code><a href="#topic+plot_PCA">plot_PCA</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### First prepare an NMDS object
x &lt;- bot %&gt;% efourier %&gt;% NMDS

plot_NMDS(x)
plot_NMDS(x, ~type) %&gt;% layer_stars() %&gt;% layer_labelpoints()

### Same on MDS object
x &lt;- bot %&gt;% efourier %&gt;% MDS

plot_MDS(x)
plot_MDS(x, ~type) %&gt;% layer_stars() %&gt;% layer_labelpoints()
</code></pre>

<hr>
<h2 id='plot_PCA'>PCA plot using grindr layers</h2><span id='topic+plot_PCA'></span>

<h3>Description</h3>

<p>Quickly vizualise <a href="#topic+PCA">PCA</a> objects and friends and build customs plots
using the <a href="#topic+layers">layers</a>. See examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_PCA(
  x,
  f = NULL,
  axes = c(1, 2),
  palette = NULL,
  points = TRUE,
  points_transp = 1/4,
  morphospace = TRUE,
  morphospace_position = "range",
  chull = TRUE,
  chullfilled = FALSE,
  labelpoints = FALSE,
  labelgroups = FALSE,
  legend = TRUE,
  title = "",
  center_origin = TRUE,
  zoom = 0.9,
  eigen = TRUE,
  box = TRUE,
  axesnames = TRUE,
  axesvar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_PCA_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+PCA">PCA</a> object</p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_f">f</code></td>
<td>
<p>factor specification to feed <a href="#topic+fac_dispatcher">fac_dispatcher</a></p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_axes">axes</code></td>
<td>
<p><code>numeric</code> of length two to select PCs to use
(<code>c(1, 2)</code> by default)</p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_palette">palette</code></td>
<td>
<p><code>color palette</code> to use <code>col_summer</code> by default</p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_points">points</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_points">layer_points</a></p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_points_transp">points_transp</code></td>
<td>
<p><code>numeric</code> to feed <a href="#topic+layer_points">layer_points</a> (default:0.25)</p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_morphospace">morphospace</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_morphospace_PCA">layer_morphospace_PCA</a></p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_morphospace_position">morphospace_position</code></td>
<td>
<p>to feed <a href="#topic+layer_morphospace_PCA">layer_morphospace_PCA</a> (default: &quot;range&quot;)</p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_chull">chull</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_chull">layer_chull</a></p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_chullfilled">chullfilled</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_chullfilled">layer_chullfilled</a></p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_labelpoints">labelpoints</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_labelpoints">layer_labelpoints</a></p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_labelgroups">labelgroups</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_labelgroups">layer_labelgroups</a></p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_legend">legend</code></td>
<td>
<p><code>logical</code> whether to draw this with <a href="#topic+layer_legend">layer_legend</a></p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_title">title</code></td>
<td>
<p><code>character</code> if specified, fee <a href="#topic+layer_title">layer_title</a> (default to <code>""</code>)</p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_center_origin">center_origin</code></td>
<td>
<p><code>logical</code> whether to center origin</p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_zoom">zoom</code></td>
<td>
<p><code>numeric</code> zoom level for the frame (default: 0.9)</p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_eigen">eigen</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_eigen">layer_eigen</a></p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_box">box</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_box">layer_box</a></p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_axesnames">axesnames</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_axesnames">layer_axesnames</a></p>
</td></tr>
<tr><td><code id="plot_PCA_+3A_axesvar">axesvar</code></td>
<td>
<p><code>logical</code> whether to draw this using <a href="#topic+layer_axesvar">layer_axesvar</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>Note</h3>

<p>This approach will replace <a href="#topic+plot.PCA">plot.PCA</a> (and <code>plot.lda</code> in further versions.
This is part of <code>grindr</code> approach that may be packaged at some point. All comments are welcome.
</p>


<h3>See Also</h3>

<p>Other grindr: 
<code><a href="#topic+drawers">drawers</a></code>,
<code><a href="#topic+layers_morphospace">layers_morphospace</a></code>,
<code><a href="#topic+layers">layers</a></code>,
<code><a href="#topic+mosaic_engine">mosaic_engine</a>()</code>,
<code><a href="#topic+papers">papers</a></code>,
<code><a href="#topic+pile">pile</a>()</code>,
<code><a href="#topic+plot_LDA">plot_LDA</a>()</code>,
<code><a href="#topic+plot_NMDS">plot_NMDS</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### First prepare two PCA objects.

# Some outlines with bot
bp &lt;- bot %&gt;% mutate(fake=sample(letters[1:5], 40, replace=TRUE)) %&gt;%
efourier(6) %&gt;% PCA
plot_PCA(bp)
plot_PCA(bp, ~type)
plot_PCA(bp, ~fake)

# Some curves with olea
op &lt;- olea %&gt;%
mutate(s=coo_area(.)) %&gt;%
filter(var != "Cypre") %&gt;%
chop(~view) %&gt;% opoly(5, nb.pts=90) %&gt;%
combine %&gt;% PCA
op$fac$s %&lt;&gt;% as.character() %&gt;% as.numeric()

op %&gt;% plot_PCA(title="hi there!")

### Now we can play with layers
# and for instance build a custom plot
# it should start with plot_PCA()

my_plot &lt;- function(x, ...){

x %&gt;%
    plot_PCA(...) %&gt;%
    layer_points %&gt;%
    layer_ellipsesaxes %&gt;%
    layer_rug
}

# and even continue after this function
op %&gt;% my_plot(~var, axes=c(1, 3)) %&gt;%
    layer_title("hi there!")

# grindr allows (almost nice) tricks like highlighting:

# bp %&gt;% .layerize_PCA(~fake) %&gt;%
#   layer_frame %&gt;% layer_axes() %&gt;%
#   layer_morphospace_PCA() -&gt; x

# highlight &lt;- function(x, ..., col_F="#CCCCCC", col_T="#FC8D62FF"){
#  args &lt;- list(...)
#  x$colors_groups &lt;- c(col_F, col_T)
#  x$colors_rows &lt;- c(col_F, col_T)[(x$f %in% args)+1]
#  x
#  }
# x %&gt;% highlight("a", "b") %&gt;% layer_points()

# You get the idea.
</code></pre>

<hr>
<h2 id='plot_silhouette'>Silhouette plot</h2><span id='topic+plot_silhouette'></span>

<h3>Description</h3>

<p>Only used, so far, after <a href="#topic+KMEDOIDS">KMEDOIDS</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_silhouette(x, palette = pal_qual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_silhouette_+3A_x">x</code></td>
<td>
<p>object returned by <a href="#topic+KMEDOIDS">KMEDOIDS</a></p>
</td></tr>
<tr><td><code id="plot_silhouette_+3A_palette">palette</code></td>
<td>
<p>one of <a href="#topic+palettes">palettes</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot</code> plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
olea %&gt;% opoly(5) %&gt;%
    KMEDOIDS(4) %&gt;%
    plot_silhouette(pal_qual_solarized)
</code></pre>

<hr>
<h2 id='plot_table'>Plots confusion matrix of sample sizes within $fac</h2><span id='topic+plot_table'></span>

<h3>Description</h3>

<p>An utility that plots a confusion matrix of sample size (or a barplot)
for every object with a $fac. Useful to visually how large are sample sizes,
how (un)balanced are designs, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_table(x, fac1, fac2 = fac1, rm0 = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_table_+3A_x">x</code></td>
<td>
<p>any object with a $fac slot (Coo, Coe, PCA, etc.)</p>
</td></tr>
<tr><td><code id="plot_table_+3A_fac1">fac1</code></td>
<td>
<p>the name or id of the first factor</p>
</td></tr>
<tr><td><code id="plot_table_+3A_fac2">fac2</code></td>
<td>
<p>the name of id of the second factor</p>
</td></tr>
<tr><td><code id="plot_table_+3A_rm0">rm0</code></td>
<td>
<p>logical whether to print zeros</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 object
</p>


<h3>See Also</h3>

<p>Other plotting functions: 
<code><a href="#topic+coo_arrows">coo_arrows</a>()</code>,
<code><a href="#topic+coo_draw">coo_draw</a>()</code>,
<code><a href="#topic+coo_listpanel">coo_listpanel</a>()</code>,
<code><a href="#topic+coo_lolli">coo_lolli</a>()</code>,
<code><a href="#topic+coo_plot">coo_plot</a>()</code>,
<code><a href="#topic+coo_ruban">coo_ruban</a>()</code>,
<code><a href="#topic+ldk_chull">ldk_chull</a>()</code>,
<code><a href="#topic+ldk_confell">ldk_confell</a>()</code>,
<code><a href="#topic+ldk_contour">ldk_contour</a>()</code>,
<code><a href="#topic+ldk_labels">ldk_labels</a>()</code>,
<code><a href="#topic+ldk_links">ldk_links</a>()</code>,
<code><a href="#topic+plot_devsegments">plot_devsegments</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_table(olea, "var")
plot_table(olea, "domes", "var")
gg &lt;- plot_table(olea, "domes", "var", rm0 = TRUE)
gg
library(ggplot2)
gg + coord_equal()
gg + scale_fill_gradient(low="green", high = "red")
gg + coord_flip()
</code></pre>

<hr>
<h2 id='plot.LDA'>Plots Linear Discriminant Analysis</h2><span id='topic+plot.LDA'></span>

<h3>Description</h3>

<p>The Momocs' <code><a href="#topic+LDA">LDA</a></code> plotter with many graphical options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'LDA'
plot(
  x,
  fac = x$fac,
  xax = 1,
  yax = 2,
  points = TRUE,
  col = "#000000",
  pch = 20,
  cex = 0.5,
  palette = col_solarized,
  center.origin = FALSE,
  zoom = 1,
  xlim = NULL,
  ylim = NULL,
  bg = par("bg"),
  grid = TRUE,
  nb.grids = 3,
  morphospace = FALSE,
  pos.shp = c("range", "full", "circle", "xy", "range_axes", "full_axes")[1],
  amp.shp = 1,
  size.shp = 1,
  nb.shp = 12,
  nr.shp = 6,
  nc.shp = 5,
  rotate.shp = 0,
  flipx.shp = FALSE,
  flipy.shp = FALSE,
  pts.shp = 60,
  border.shp = col_alpha("#000000", 0.5),
  lwd.shp = 1,
  col.shp = col_alpha("#000000", 0.95),
  stars = FALSE,
  ellipses = FALSE,
  conf.ellipses = 0.5,
  ellipsesax = TRUE,
  conf.ellipsesax = c(0.5, 0.9),
  lty.ellipsesax = 1,
  lwd.ellipsesax = sqrt(2),
  chull = FALSE,
  chull.lty = 1,
  chull.filled = FALSE,
  chull.filled.alpha = 0.92,
  density = FALSE,
  lev.density = 20,
  contour = FALSE,
  lev.contour = 3,
  n.kde2d = 100,
  delaunay = FALSE,
  loadings = FALSE,
  labelspoints = FALSE,
  col.labelspoints = par("fg"),
  cex.labelspoints = 0.6,
  abbreviate.labelspoints = TRUE,
  labelsgroups = TRUE,
  cex.labelsgroups = 0.8,
  rect.labelsgroups = FALSE,
  abbreviate.labelsgroups = FALSE,
  color.legend = FALSE,
  axisnames = TRUE,
  axisvar = TRUE,
  unit = FALSE,
  eigen = TRUE,
  rug = TRUE,
  title = substitute(x),
  box = TRUE,
  old.par = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.LDA_+3A_x">x</code></td>
<td>
<p>an object of class &quot;LDA&quot;, typically obtained with <a href="#topic+LDA">LDA</a></p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_fac">fac</code></td>
<td>
<p>name or the column id from the $fac slot, or a formula combining colum names
from the $fac slot (cf. examples). A factor or a numeric of the same length
can also be passed on the fly.</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_xax">xax</code></td>
<td>
<p>the first PC axis</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_yax">yax</code></td>
<td>
<p>the second PC axis</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_points">points</code></td>
<td>
<p>logical whether to plot points</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_col">col</code></td>
<td>
<p>a color for the points (either global, for every level of the fac
or for every individual, see examples)</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_pch">pch</code></td>
<td>
<p>a pch for the points (either global, for every level of the fac
or for every individual, see examples)</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_cex">cex</code></td>
<td>
<p>the size of the points</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_palette">palette</code></td>
<td>
<p>a <a href="#topic+palette">palette</a></p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_center.origin">center.origin</code></td>
<td>
<p>logical whether to center the plot onto the origin</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_zoom">zoom</code></td>
<td>
<p>to keep your distances</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_xlim">xlim</code></td>
<td>
<p>numeric of length two ; if provided along with ylim, the x and y lims to use</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_ylim">ylim</code></td>
<td>
<p>numeric of length two ; if provided along with xlim, the x and y lims to use</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_bg">bg</code></td>
<td>
<p>color for the background</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_grid">grid</code></td>
<td>
<p>logical whether to draw a grid</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_nb.grids">nb.grids</code></td>
<td>
<p>and how many of them</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_morphospace">morphospace</code></td>
<td>
<p>logical whether to add the morphological space</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_pos.shp">pos.shp</code></td>
<td>
<p>passed to <a href="#topic+morphospace_positions">morphospace_positions</a>, one of
<code>"range", "full", "circle", "xy", "range_axes", "full_axes"</code>. Or directly
a matrix of positions. See <a href="#topic+morphospace_positions">morphospace_positions</a></p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_amp.shp">amp.shp</code></td>
<td>
<p>amplification factor for shape deformation</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_size.shp">size.shp</code></td>
<td>
<p>the size of the shapes</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_nb.shp">nb.shp</code></td>
<td>
<p>(pos.shp=&quot;circle&quot;) the number of shapes on the compass</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_nr.shp">nr.shp</code></td>
<td>
<p>(pos.shp=&quot;full&quot; or &quot;range) the number of shapes per row</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_nc.shp">nc.shp</code></td>
<td>
<p>(pos.shp=&quot;full&quot; or &quot;range) the number of shapes per column</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_rotate.shp">rotate.shp</code></td>
<td>
<p>angle in radians to rotate shapes (if several methods, a vector of angles)</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_flipx.shp">flipx.shp</code></td>
<td>
<p>same as above, whether to apply coo_flipx</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_flipy.shp">flipy.shp</code></td>
<td>
<p>same as above, whether to apply coo_flipy</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_pts.shp">pts.shp</code></td>
<td>
<p>the number of points fro drawing shapes</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_border.shp">border.shp</code></td>
<td>
<p>the border color of the shapes</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_lwd.shp">lwd.shp</code></td>
<td>
<p>the line width for these shapes</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_col.shp">col.shp</code></td>
<td>
<p>the color of the shapes</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_stars">stars</code></td>
<td>
<p>logical whether to draw &quot;stars&quot;</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_ellipses">ellipses</code></td>
<td>
<p>logical whether to draw confidence ellipses</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_conf.ellipses">conf.ellipses</code></td>
<td>
<p>numeric the quantile for the (bivariate gaussian) confidence ellipses</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_ellipsesax">ellipsesax</code></td>
<td>
<p>logical whether to draw ellipse axes</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_conf.ellipsesax">conf.ellipsesax</code></td>
<td>
<p>one or more numeric, the quantiles for the (bivariate gaussian) ellipses axes</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_lty.ellipsesax">lty.ellipsesax</code></td>
<td>
<p>if yes, the lty with which to draw these axes</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_lwd.ellipsesax">lwd.ellipsesax</code></td>
<td>
<p>if yes, one or more numeric for the line widths</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_chull">chull</code></td>
<td>
<p>logical whether to draw a convex hull</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_chull.lty">chull.lty</code></td>
<td>
<p>if yes, its linetype</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_chull.filled">chull.filled</code></td>
<td>
<p>logical whether to add filled convex hulls</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_chull.filled.alpha">chull.filled.alpha</code></td>
<td>
<p>numeric alpha transparency</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_density">density</code></td>
<td>
<p>whether to add a 2d density kernel estimation (based on <a href="MASS.html#topic+kde2d">kde2d</a>)</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_lev.density">lev.density</code></td>
<td>
<p>if yes, the number of levels to plot (through <a href="Matrix.html#topic+image">image</a>)</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_contour">contour</code></td>
<td>
<p>whether to add contour lines based on 2d density kernel</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_lev.contour">lev.contour</code></td>
<td>
<p>if yes, the (approximate) number of lines to draw</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_n.kde2d">n.kde2d</code></td>
<td>
<p>the number of bins for <a href="MASS.html#topic+kde2d">kde2d</a>, ie the 'smoothness' of density kernel</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_delaunay">delaunay</code></td>
<td>
<p>logical whether to add a delaunay 'mesh' between points</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_loadings">loadings</code></td>
<td>
<p>logical whether to add loadings for every variables</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_labelspoints">labelspoints</code></td>
<td>
<p>if TRUE rownames are used as labels, a colname from $fac can also be passed</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_col.labelspoints">col.labelspoints</code></td>
<td>
<p>a color for these labels, otherwise inherited from fac</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_cex.labelspoints">cex.labelspoints</code></td>
<td>
<p>a cex for these labels</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_abbreviate.labelspoints">abbreviate.labelspoints</code></td>
<td>
<p>logical whether to abbreviate</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_labelsgroups">labelsgroups</code></td>
<td>
<p>logical whether to add labels for groups</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_cex.labelsgroups">cex.labelsgroups</code></td>
<td>
<p>ifyes, a numeric for the size of the labels</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_rect.labelsgroups">rect.labelsgroups</code></td>
<td>
<p>logical whether to add a rectangle behind groups names</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_abbreviate.labelsgroups">abbreviate.labelsgroups</code></td>
<td>
<p>logical, whether to abbreviate group names</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_color.legend">color.legend</code></td>
<td>
<p>logical whether to add a (cheap) color legend for numeric fac</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_axisnames">axisnames</code></td>
<td>
<p>logical whether to add PC names</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_axisvar">axisvar</code></td>
<td>
<p>logical whether to draw the variance they explain</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_unit">unit</code></td>
<td>
<p>logical whether to add plane unit</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_eigen">eigen</code></td>
<td>
<p>logical whether to draw a plot of the eigen values</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_rug">rug</code></td>
<td>
<p>logical whether to add rug to margins</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_title">title</code></td>
<td>
<p>character a name for the plot</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_box">box</code></td>
<td>
<p>whether to draw a box around the plotting region</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_old.par">old.par</code></td>
<td>
<p>whether to restore the old <a href="graphics.html#topic+par">par</a>. Set it to <code>FALSE</code> if you want to reuse the graphical window.</p>
</td></tr>
<tr><td><code id="plot.LDA_+3A_...">...</code></td>
<td>
<p>useless here, just to fit the generic plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Widely inspired by the &quot;layers&quot; philosophy behind graphical functions
of the ade4 R package.
</p>


<h3>Value</h3>

<p>a plot
</p>


<h3>Note</h3>

<p>Morphospaces are deprecated so far. 99% of the code
is shared with <a href="#topic+plot.PCA">plot.PCA</a> waiting for a general rewriting of a multivariate plotter.
See https://github.com/vbonhomme/Momocs/issues/121
</p>


<h3>See Also</h3>

<p><a href="#topic+LDA">LDA</a>, <a href="#topic+plot_CV">plot_CV</a>, <a href="#topic+plot_CV2">plot_CV2</a>, <a href="#topic+plot.PCA">plot.PCA</a>.
</p>

<hr>
<h2 id='plot.PCA'>Plots Principal Component Analysis</h2><span id='topic+plot.PCA'></span>

<h3>Description</h3>

<p>The Momocs' <code><a href="#topic+PCA">PCA</a></code> plotter with morphospaces and many graphical options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PCA'
plot(
  x,
  fac,
  xax = 1,
  yax = 2,
  points = TRUE,
  col = "#000000",
  pch = 20,
  cex = 0.5,
  palette = col_solarized,
  center.origin = FALSE,
  zoom = 1,
  xlim = NULL,
  ylim = NULL,
  bg = par("bg"),
  grid = TRUE,
  nb.grids = 3,
  morphospace = TRUE,
  pos.shp = c("range", "full", "circle", "xy", "range_axes", "full_axes")[1],
  amp.shp = 1,
  size.shp = 1,
  nb.shp = 12,
  nr.shp = 6,
  nc.shp = 5,
  rotate.shp = 0,
  flipx.shp = FALSE,
  flipy.shp = FALSE,
  pts.shp = 60,
  border.shp = col_alpha("#000000", 0.5),
  lwd.shp = 1,
  col.shp = col_alpha("#000000", 0.95),
  stars = FALSE,
  ellipses = FALSE,
  conf.ellipses = 0.5,
  ellipsesax = FALSE,
  conf.ellipsesax = c(0.5, 0.9),
  lty.ellipsesax = 1,
  lwd.ellipsesax = sqrt(2),
  chull = FALSE,
  chull.lty = 1,
  chull.filled = TRUE,
  chull.filled.alpha = 0.92,
  density = FALSE,
  lev.density = 20,
  contour = FALSE,
  lev.contour = 3,
  n.kde2d = 100,
  delaunay = FALSE,
  loadings = FALSE,
  labelspoints = FALSE,
  col.labelspoints = par("fg"),
  cex.labelspoints = 0.6,
  abbreviate.labelspoints = TRUE,
  labelsgroups = TRUE,
  cex.labelsgroups = 0.8,
  rect.labelsgroups = FALSE,
  abbreviate.labelsgroups = FALSE,
  color.legend = FALSE,
  axisnames = TRUE,
  axisvar = TRUE,
  unit = FALSE,
  eigen = TRUE,
  rug = TRUE,
  title = substitute(x),
  box = TRUE,
  old.par = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.PCA_+3A_x">x</code></td>
<td>
<p><code>PCA</code>, typically obtained with <a href="#topic+PCA">PCA</a></p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_fac">fac</code></td>
<td>
<p>name or the column id from the $fac slot, or a formula combining colum names
from the $fac slot (cf. examples). A factor or a numeric of the same length
can also be passed on the fly.</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_xax">xax</code></td>
<td>
<p>the first PC axis</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_yax">yax</code></td>
<td>
<p>the second PC axis</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_points">points</code></td>
<td>
<p>logical whether to plot points</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_col">col</code></td>
<td>
<p>a color for the points (either global, for every level of the fac
or for every individual, see examples)</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_pch">pch</code></td>
<td>
<p>a pch for the points (either global, for every level of the fac
or for every individual, see examples)</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_cex">cex</code></td>
<td>
<p>the size of the points</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_palette">palette</code></td>
<td>
<p>a <a href="#topic+palette">palette</a></p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_center.origin">center.origin</code></td>
<td>
<p>logical whether to center the plot onto the origin</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_zoom">zoom</code></td>
<td>
<p>to keep your distances</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_xlim">xlim</code></td>
<td>
<p>numeric of length two ; if provided along with ylim, the x and y lims to use</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_ylim">ylim</code></td>
<td>
<p>numeric of length two ; if provided along with xlim, the x and y lims to use</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_bg">bg</code></td>
<td>
<p>color for the background</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_grid">grid</code></td>
<td>
<p>logical whether to draw a grid</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_nb.grids">nb.grids</code></td>
<td>
<p>and how many of them</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_morphospace">morphospace</code></td>
<td>
<p>logical whether to add the morphological space</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_pos.shp">pos.shp</code></td>
<td>
<p>passed to <a href="#topic+morphospace_positions">morphospace_positions</a>, one of
<code>"range", "full", "circle", "xy", "range_axes", "full_axes"</code>. Or directly
a matrix of positions. See <a href="#topic+morphospace_positions">morphospace_positions</a></p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_amp.shp">amp.shp</code></td>
<td>
<p>amplification factor for shape deformation</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_size.shp">size.shp</code></td>
<td>
<p>the size of the shapes</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_nb.shp">nb.shp</code></td>
<td>
<p>(pos.shp=&quot;circle&quot;) the number of shapes on the compass</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_nr.shp">nr.shp</code></td>
<td>
<p>(pos.shp=&quot;full&quot; or &quot;range) the number of shapes per row</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_nc.shp">nc.shp</code></td>
<td>
<p>(pos.shp=&quot;full&quot; or &quot;range) the number of shapes per column</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_rotate.shp">rotate.shp</code></td>
<td>
<p>angle in radians to rotate shapes (if several methods, a vector of angles)</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_flipx.shp">flipx.shp</code></td>
<td>
<p>same as above, whether to apply coo_flipx</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_flipy.shp">flipy.shp</code></td>
<td>
<p>same as above, whether to apply coo_flipy</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_pts.shp">pts.shp</code></td>
<td>
<p>the number of points fro drawing shapes</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_border.shp">border.shp</code></td>
<td>
<p>the border color of the shapes</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_lwd.shp">lwd.shp</code></td>
<td>
<p>the line width for these shapes</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_col.shp">col.shp</code></td>
<td>
<p>the color of the shapes</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_stars">stars</code></td>
<td>
<p>logical whether to draw &quot;stars&quot;</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_ellipses">ellipses</code></td>
<td>
<p>logical whether to draw confidence ellipses</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_conf.ellipses">conf.ellipses</code></td>
<td>
<p>numeric the quantile for the (bivariate gaussian) confidence ellipses</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_ellipsesax">ellipsesax</code></td>
<td>
<p>logical whether to draw ellipse axes</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_conf.ellipsesax">conf.ellipsesax</code></td>
<td>
<p>one or more numeric, the quantiles for the (bivariate gaussian) ellipses axes</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_lty.ellipsesax">lty.ellipsesax</code></td>
<td>
<p>if yes, the lty with which to draw these axes</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_lwd.ellipsesax">lwd.ellipsesax</code></td>
<td>
<p>if yes, one or more numeric for the line widths</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_chull">chull</code></td>
<td>
<p>logical whether to draw a convex hull</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_chull.lty">chull.lty</code></td>
<td>
<p>if yes, its linetype</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_chull.filled">chull.filled</code></td>
<td>
<p>logical whether to add filled convex hulls</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_chull.filled.alpha">chull.filled.alpha</code></td>
<td>
<p>numeric alpha transparency</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_density">density</code></td>
<td>
<p>whether to add a 2d density kernel estimation (based on <a href="MASS.html#topic+kde2d">kde2d</a>)</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_lev.density">lev.density</code></td>
<td>
<p>if yes, the number of levels to plot (through <a href="Matrix.html#topic+image">image</a>)</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_contour">contour</code></td>
<td>
<p>whether to add contour lines based on 2d density kernel</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_lev.contour">lev.contour</code></td>
<td>
<p>if yes, the (approximate) number of lines to draw</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_n.kde2d">n.kde2d</code></td>
<td>
<p>the number of bins for <a href="MASS.html#topic+kde2d">kde2d</a>, ie the 'smoothness' of density kernel</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_delaunay">delaunay</code></td>
<td>
<p>logical whether to add a delaunay 'mesh' between points</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_loadings">loadings</code></td>
<td>
<p>logical whether to add loadings for every variables</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_labelspoints">labelspoints</code></td>
<td>
<p>if TRUE rownames are used as labels, a colname from $fac can also be passed</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_col.labelspoints">col.labelspoints</code></td>
<td>
<p>a color for these labels, otherwise inherited from fac</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_cex.labelspoints">cex.labelspoints</code></td>
<td>
<p>a cex for these labels</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_abbreviate.labelspoints">abbreviate.labelspoints</code></td>
<td>
<p>logical whether to abbreviate</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_labelsgroups">labelsgroups</code></td>
<td>
<p>logical whether to add labels for groups</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_cex.labelsgroups">cex.labelsgroups</code></td>
<td>
<p>ifyes, a numeric for the size of the labels</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_rect.labelsgroups">rect.labelsgroups</code></td>
<td>
<p>logical whether to add a rectangle behind groups names</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_abbreviate.labelsgroups">abbreviate.labelsgroups</code></td>
<td>
<p>logical, whether to abbreviate group names</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_color.legend">color.legend</code></td>
<td>
<p>logical whether to add a (cheap) color legend for numeric fac</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_axisnames">axisnames</code></td>
<td>
<p>logical whether to add PC names</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_axisvar">axisvar</code></td>
<td>
<p>logical whether to draw the variance they explain</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_unit">unit</code></td>
<td>
<p>logical whether to add plane unit</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_eigen">eigen</code></td>
<td>
<p>logical whether to draw a plot of the eigen values</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_rug">rug</code></td>
<td>
<p>logical whether to add rug to margins</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_title">title</code></td>
<td>
<p>character a name for the plot</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_box">box</code></td>
<td>
<p>whether to draw a box around the plotting region</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_old.par">old.par</code></td>
<td>
<p>whether to restore the old <a href="graphics.html#topic+par">par</a>. Set it to <code>FALSE</code> if you want to reuse the graphical window.</p>
</td></tr>
<tr><td><code id="plot.PCA_+3A_...">...</code></td>
<td>
<p>useless here, just to fit the generic plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Widely inspired by the &quot;layers&quot; philosophy behind graphical functions
of the ade4 R package.
</p>


<h3>Value</h3>

<p>a plot
</p>


<h3>Note</h3>

<p>NAs is <code>$fac</code> are handled quite experimentally.
More importantly, as of early 2018, I plan I complete rewrite of
<code>plot.PCA</code> and other multivariate plotters.
</p>


<h3>See Also</h3>

<p><a href="#topic+plot.LDA">plot.LDA</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
bot.f &lt;- efourier(bot, 12)
bot.p &lt;- PCA(bot.f)

### Morphospace options
plot(bot.p, pos.shp="full")
plot(bot.p, pos.shp="range")
plot(bot.p, pos.shp="xy")
plot(bot.p, pos.shp="circle")
plot(bot.p, pos.shp="range_axes")
plot(bot.p, pos.shp="full_axes")

plot(bot.p, morpho=FALSE)

### Passing factors to plot.PCA
# 3 equivalent methods
plot(bot.p, "type")
plot(bot.p, 1)
plot(bot.p, ~type)

# let's create a dummy factor of the correct length
# and another added to the $fac with mutate
# and a numeric of the correct length
f &lt;- factor(rep(letters[1:2], 20))
z &lt;- factor(rep(LETTERS[1:2], 20))
bot %&lt;&gt;% mutate(cs=coo_centsize(.), z=z)
bp &lt;- bot %&gt;% efourier %&gt;% PCA
# so bp contains type, cs (numeric) and z; not f
# yet f can be passed on the fly
plot(bp, f)
# numeric fac are allowed
plot(bp, "cs", cex=3, color.legend=TRUE)
# formula allows combinations of factors
plot(bp, ~type+z)

### other morphometric approaches works the same
# open curves
op &lt;- npoly(olea, 5)
op.p &lt;- PCA(op)
op.p
plot(op.p, ~ domes + var, morpho=TRUE) # use of formula

# landmarks
wp &lt;- fgProcrustes(wings, tol=1e-4)
wpp &lt;- PCA(wp)
wpp
plot(wpp, 1)

### Cosmetic options
# window
plot(bp, 1, zoom=2)
plot(bp, zoom=0.5)
plot(bp, center.origin=FALSE, grid=FALSE)

# colors
plot(bp, col="red") # globally
plot(bp, 1, col=c("#00FF00", "#0000FF")) # for every level
# a color vector of the right length
plot(bp, 1, col=rep(c("#00FF00", "#0000FF"), each=20))
# a color vector of the right length, mixign Rcolor names (not a good idea though)
plot(bp, 1, col=rep(c("#00FF00", "forestgreen"), each=20))


# ellipses
plot(bp, 1, conf.ellipsesax=2/3)
plot(bp, 1, ellipsesax=FALSE)
plot(bp, 1, ellipsesax=TRUE, ellipses=TRUE)

# stars
plot(bp, 1, stars=TRUE, ellipsesax=FALSE)

# convex hulls
plot(bp, 1, chull=TRUE)
plot(bp, 1, chull.lty=3)

# filled convex hulls
plot(bp, 1, chull.filled=TRUE)
plot(bp, 1, chull.filled.alpha = 0.8, chull.lty =1) # you can omit chull.filled=TRUE

# density kernel
plot(bp, 1, density=TRUE, contour=TRUE, lev.contour=10)

# delaunay
plot(bp, 1, delaunay=TRUE)

# loadings
flower %&gt;% PCA %&gt;% plot(1, loadings=TRUE)

# point/group labelling
plot(bp, 1, labelspoint=TRUE) # see options for abbreviations
plot(bp, 1, labelsgroup=TRUE) # see options for abbreviations

# clean axes, no rug, no border, random title
plot(bp, axisvar=FALSE, axisnames=FALSE, rug=FALSE, box=FALSE, title="random")

# no eigen
plot(bp, eigen=FALSE) # eigen cause troubles to graphical window
# eigen may causes troubles to the graphical window. you can try old.par = TRUE

</code></pre>

<hr>
<h2 id='pProcrustes'>Partial Procrustes alignment between two shapes</h2><span id='topic+pProcrustes'></span>

<h3>Description</h3>

<p>Directly borrowed from Claude (2008), and called <code>pPsup</code> there.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pProcrustes(coo1, coo2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pProcrustes_+3A_coo1">coo1</code></td>
<td>
<p>Configuration matrix to be superimposed onto the centered preshape of coo2.</p>
</td></tr>
<tr><td><code id="pProcrustes_+3A_coo2">coo2</code></td>
<td>
<p>Reference configuration matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with components
</p>

<ul>
<li> <p><code>coo1</code> superimposed centered preshape of coo1 onto the centered preshape of coo2
</p>
</li>
<li> <p><code>coo2</code> centered preshape of coo2
</p>
</li>
<li> <p><code>rotation</code> rotation matrix
</p>
</li>
<li> <p><code>DP</code> partial Procrustes distance between coo1 and coo2
</p>
</li>
<li> <p><code>rho</code> trigonometric Procrustes distance.
</p>
</li></ul>



<h3>References</h3>

<p>Claude, J. (2008). Morphometrics with R. Analysis (p. 316). Springer.
</p>


<h3>See Also</h3>

<p>Other procrustes functions: 
<code><a href="#topic+fProcrustes">fProcrustes</a>()</code>,
<code><a href="#topic+fgProcrustes">fgProcrustes</a>()</code>,
<code><a href="#topic+fgsProcrustes">fgsProcrustes</a>()</code>
</p>

<hr>
<h2 id='Ptolemy'>Ptolemaic ellipses and illustration of efourier</h2><span id='topic+Ptolemy'></span>

<h3>Description</h3>

<p>Calculate and display Ptolemaic ellipses which illustrates
intuitively the principle behing elliptical Fourier analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ptolemy(
  coo,
  t = seq(0, 2 * pi, length = 7)[-1],
  nb.h = 3,
  nb.pts = 360,
  palette = col_heat,
  zoom = 5/4,
  legend = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ptolemy_+3A_coo">coo</code></td>
<td>
<p>a matrix of (x; y) coordinates</p>
</td></tr>
<tr><td><code id="Ptolemy_+3A_t">t</code></td>
<td>
<p>A <code>vector</code> af angles (in radians) on which to display ellipses</p>
</td></tr>
<tr><td><code id="Ptolemy_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to display</p>
</td></tr>
<tr><td><code id="Ptolemy_+3A_nb.pts">nb.pts</code></td>
<td>
<p><code>integer</code>. The number of points to use to display shapes</p>
</td></tr>
<tr><td><code id="Ptolemy_+3A_palette">palette</code></td>
<td>
<p>a color palette</p>
</td></tr>
<tr><td><code id="Ptolemy_+3A_zoom">zoom</code></td>
<td>
<p>numeric a zoom factor for <a href="#topic+coo_plot">coo_plot</a></p>
</td></tr>
<tr><td><code id="Ptolemy_+3A_legend">legend</code></td>
<td>
<p><code>logical</code>. Whether to plot the legend box</p>
</td></tr>
<tr><td><code id="Ptolemy_+3A_...">...</code></td>
<td>
<p>additional parameters to feed <a href="#topic+coo_plot">coo_plot</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a drawing on the last plot
</p>


<h3>References</h3>

<p>This method has been inspired by the figures found in the followings papers.
Kuhl FP, Giardina CR. 1982. Elliptic Fourier features of a closed contour.
<em>Computer Graphics and Image Processing</em> <b>18</b>: 236-258.
Crampton JS. 1995. Elliptical Fourier shape analysis of fossil bivalves:
some practical considerations. <em>Lethaia</em> <b>28</b>: 179-186.
</p>


<h3>See Also</h3>

<p>An intuitive explanation of elliptic Fourier analysis can be found in
the <b>Details</b> section of the <a href="#topic+efourier">efourier</a> function.
</p>
<p>exemplifying functions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cat &lt;- shapes[4]
Ptolemy(cat, main="An EFT cat")
</code></pre>

<hr>
<h2 id='rearrange_ldk'>Rearrange, (select and reorder) landmarks to retain</h2><span id='topic+rearrange_ldk'></span>

<h3>Description</h3>

<p>Helps reorder and retain landmarks by simply changing the order in which they
are recorded in the <code>Coo</code> objects. Note that for <code>Out</code> and <code>Opn</code>
objects, this rearranges the <code>$ldk</code> component. For <code>Ldk</code>, it rearranges
the <code>$coo</code> directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rearrange_ldk(Coo, new_ldk_ids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rearrange_ldk_+3A_coo">Coo</code></td>
<td>
<p>any appropriate <code>Coo</code> object (typically an <code>Ldk</code>)
with landmarks inside</p>
</td></tr>
<tr><td><code id="rearrange_ldk_+3A_new_ldk_ids">new_ldk_ids</code></td>
<td>
<p>a vector of numeric with the ldk to retain <em>and</em>
in the right order (see below)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>See Also</h3>

<p>Other ldk/slidings methods: 
<code><a href="#topic+add_ldk">add_ldk</a>()</code>,
<code><a href="#topic+def_ldk">def_ldk</a>()</code>,
<code><a href="#topic+def_slidings">def_slidings</a>()</code>,
<code><a href="#topic+get_ldk">get_ldk</a>()</code>,
<code><a href="#topic+get_slidings">get_slidings</a>()</code>,
<code><a href="#topic+slidings_scheme">slidings_scheme</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Out example
hearts %&gt;% slice(1) %T&gt;% stack %$% ldk
hearts %&gt;% rearrange_ldk(c(4, 1)) %&gt;%
       slice(1) %T&gt;%stack %$% ldk

# Ldk example
wings %&gt;% slice(1) %T&gt;% stack %$% coo
wings %&gt;% rearrange_ldk(c(1, 3, 12:15)) %&gt;%
      slice(1) %T&gt;% stack %$% coo
</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic++25+3E+25'></span><span id='topic++25+3C+3E+25'></span><span id='topic++25+24+25'></span><span id='topic++25T+3E+25'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>magrittr</dt><dd><p><code><a href="magrittr.html#topic+exposition">%$%</a></code>, <code><a href="magrittr.html#topic+compound">%&lt;&gt;%</a></code>, <code><a href="magrittr.html#topic+pipe">%&gt;%</a></code>, <code><a href="magrittr.html#topic+tee">%T&gt;%</a></code></p>
</dd>
</dl>

<hr>
<h2 id='reLDA'>&quot;Redo&quot; a LDA on new data</h2><span id='topic+reLDA'></span><span id='topic+reLDA.default'></span><span id='topic+reLDA.PCA'></span><span id='topic+reLDA.Coe'></span>

<h3>Description</h3>

<p>Basically a wrapper around <a href="MASS.html#topic+predict.lda">predict.lda</a> from the package MASS. Uses a LDA model
to classify new data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reLDA(newdata, LDA)

## Default S3 method:
reLDA(newdata, LDA)

## S3 method for class 'PCA'
reLDA(newdata, LDA)

## S3 method for class 'Coe'
reLDA(newdata, LDA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reLDA_+3A_newdata">newdata</code></td>
<td>
<p>to use, a <a href="#topic+PCA">PCA</a> or any <a href="#topic+Coe">Coe</a> object</p>
</td></tr>
<tr><td><code id="reLDA_+3A_lda">LDA</code></td>
<td>
<p>a <a href="#topic+LDA">LDA</a> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with components (from ?predict.lda ).
</p>

<ul>
<li><p> class factor of classification
</p>
</li>
<li><p> posterior posterior probabilities for the classes
</p>
</li>
<li><p> x the scores of test cases
</p>
</li>
<li><p> res data.frame of the results
</p>
</li>
<li><p> CV.tab a confusion matrix of the results
</p>
</li>
<li><p> CV.correct proportion of the diagonal of CV.tab
</p>
</li>
<li><p> newdata the data used to calculate passed to predict.lda
</p>
</li></ul>



<h3>Note</h3>

<p>Uses the same number of PC axis as the LDA object provided. You should probably use <a href="#topic+rePCA">rePCA</a> in
conjunction with reLDA to get 'homologous' scores.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># We select the first 10 individuals in bot,
# for whisky and beer bottles. It will be our referential.
bot1   &lt;- slice(bot, c(1:10, 21:30))
# Same thing for the other 10 individuals.
# It will be our unknown dataset on which we want
# to calculate classes.
bot2   &lt;- slice(bot, c(11:20, 31:40))

# We calculate efourier on these two datasets
bot1.f &lt;- efourier(bot1, 8)
bot2.f &lt;- efourier(bot2, 8)

# Here we obtain our LDA model: first, a PCA, then a LDA
bot1.p &lt;- PCA(bot1.f)
bot1.l &lt;- LDA(bot1.p, "type")

# we redo the same PCA since we worked with scores
bot2.p &lt;- rePCA(bot1.p, bot2.f)

# we finally "predict" with the model obtained before
bot2.l &lt;- reLDA(bot2.p, bot1.l)
bot2.l

</code></pre>

<hr>
<h2 id='rename'>Rename columns by name</h2><span id='topic+rename'></span>

<h3>Description</h3>

<p>Rename variables, from the <code>$fac</code>. See examples and <a href="dplyr.html#topic+rename">dplyr::rename</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rename(.data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rename_+3A_.data">.data</code></td>
<td>
<p>a <code>Coo</code>, <code>Coe</code>, <code>PCA</code> object</p>
</td></tr>
<tr><td><code id="rename_+3A_...">...</code></td>
<td>
<p>comma separated list of unquoted expressions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>dplyr verbs are maintained.
</p>


<h3>Value</h3>

<p>a Momocs object of the same class.
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>olea
rename(olea, variety=var, domesticated=domes) # rename var column

</code></pre>

<hr>
<h2 id='rePCA'>&quot;Redo&quot; a PCA on a new Coe</h2><span id='topic+rePCA'></span>

<h3>Description</h3>

<p>Basically reapply rotation to a new Coe object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rePCA(PCA, Coe)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rePCA_+3A_pca">PCA</code></td>
<td>
<p>a <a href="#topic+PCA">PCA</a> object</p>
</td></tr>
<tr><td><code id="rePCA_+3A_coe">Coe</code></td>
<td>
<p>a <a href="#topic+Coe">Coe</a> object</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Quite experimental. Dimensions of the matrices and methods must match.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>b &lt;- filter(bot, type=="beer")
w &lt;- filter(bot, type=="whisky")

bf &lt;- efourier(b, 8)
bp &lt;- PCA(bf)

wf &lt;- efourier(w, 8)

# and we use the "beer" PCA on the whisky coefficients
wp &lt;- rePCA(bp, wf)

plot(wp)

plot(bp, eig=FALSE)
points(wp$x[, 1:2], col="red", pch=4)

</code></pre>

<hr>
<h2 id='rescale'>Rescale coordinates from pixels to real length units</h2><span id='topic+rescale'></span>

<h3>Description</h3>

<p>Most of the time, (x, y) coordinates are recorded in pixels. If we want to have
them in mm, cm, etc. we need to convert them and to rescale them. This functions
does the job for the two cases: i) either an homogeneous rescaling factor,
e.g. if all pictures were taken using the very same magnification or ii) with various
magnifications. More in the Details section
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rescale(x, scaling_factor, scale_mapping, magnification_col, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rescale_+3A_x">x</code></td>
<td>
<p>any <code>Coo</code> object</p>
</td></tr>
<tr><td><code id="rescale_+3A_scaling_factor">scaling_factor</code></td>
<td>
<p>numeric an homogeneous scaling factor. If all you (x, y) coordinates
have the same scale</p>
</td></tr>
<tr><td><code id="rescale_+3A_scale_mapping">scale_mapping</code></td>
<td>
<p>either a data.frame or a path to read such a data.frame. It MUST contain
three columns in that order: magnification found in <code style="white-space: pre;">&#8288;$fac&#8288;</code>, column <code>"magnification_col"</code>, pixels, real length unit.
Column names do not matter but must be specified, as read.table reads with <code>header=TRUE</code> Every
different magnification level found in <code style="white-space: pre;">&#8288;$fac&#8288;</code>, column <code>"magnification_col"</code> must have its row.</p>
</td></tr>
<tr><td><code id="rescale_+3A_magnification_col">magnification_col</code></td>
<td>
<p>the name or id of the $fac column to look for magnification levels for every image</p>
</td></tr>
<tr><td><code id="rescale_+3A_...">...</code></td>
<td>
<p>additional arguments (besides header=TRUE) to pass to read.table if 'scale_mapping' is a path</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The i) case above is straightforward, if 1cm is 500pix long on all your pictures,
just call <code>rescale(your_Coo, scaling_factor=1/500)</code> and all coordinates will be in cm.
</p>
<p>The ii) second case is more subtle. First you need to code in your <a href="#topic+Coo">Coo</a> object, in the fac
slot, a column named, say &quot;mag&quot;, for magnification. Imagine you have 4 magnifications: 0.5, 1, 2 and 5,
we have to indicate for each magnification, how many pixels stands for how many units in the real world.
</p>
<p>This information is passed as a data.frame, built externally or in R, that must look like this:
</p>
<pre>
mag   pix    cm
0.5   1304   10
1     921    10
2     816    5
5     1020   5
</pre><p>.
</p>
<p>We have to do that because, for optical reasons, the ratio pix/real_unit, is not a linear
function of the magnification.
</p>
<p>All shapes will be centered to apply (the single or the different) scaling_factor.
</p>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>Note</h3>

<p>This function is simple but quite complex to detail. Feel free to contact me should you have any
problem with it. You can just access its code (type <code>rescale</code>) and reply it yourself.
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>

<hr>
<h2 id='rfourier'>Radii variation Fourier transform (equally spaced radii)</h2><span id='topic+rfourier'></span><span id='topic+rfourier.default'></span><span id='topic+rfourier.Out'></span><span id='topic+rfourier.list'></span>

<h3>Description</h3>

<p><code>rfourier</code> computes radii variation Fourier analysis from a matrix or a
list of coordinates where points are equally spaced radii.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rfourier(x, ...)

## Default S3 method:
rfourier(x, nb.h, smooth.it = 0, norm = FALSE, ...)

## S3 method for class 'Out'
rfourier(x, nb.h = 40, smooth.it = 0, norm = TRUE, thres = pi/90, ...)

## S3 method for class 'list'
rfourier(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rfourier_+3A_x">x</code></td>
<td>
<p>A <code>list</code> or <code>matrix</code> of coordinates or an <code>Out</code> object</p>
</td></tr>
<tr><td><code id="rfourier_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
<tr><td><code id="rfourier_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to use. If missing, 12 is used on shapes;
99 percent of harmonic power on Out objects, both with messages.</p>
</td></tr>
<tr><td><code id="rfourier_+3A_smooth.it">smooth.it</code></td>
<td>
<p><code>integer</code>. The number of smoothing iterations to
perform.</p>
</td></tr>
<tr><td><code id="rfourier_+3A_norm">norm</code></td>
<td>
<p><code>logical</code>. Whether to scale the outlines so that the mean
length of the radii used equals 1.</p>
</td></tr>
<tr><td><code id="rfourier_+3A_thres">thres</code></td>
<td>
<p><code>numeric</code> a tolerance to feed <a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>see the JSS paper for the maths behind. The methods for <code>Out</code> objects
tests if coordinates have equally spaced radii using <a href="#topic+is_equallyspacedradii">is_equallyspacedradii</a>. A
message is printed if this is not the case.
</p>


<h3>Value</h3>

<p>A list with following components:
</p>

<ul>
<li> <p><code>an</code> vector of <code class="reqn">a_{1-&gt;n}</code> harmonic coefficients
</p>
</li>
<li> <p><code>bn</code> vector of <code class="reqn">b_{1-&gt;n}</code> harmonic coefficients
</p>
</li>
<li> <p><code>ao</code> ao harmonic coefficient.
</p>
</li>
<li> <p><code>r</code> vector of radii lengths.
</p>
</li></ul>



<h3>Note</h3>

<p>Silent message and progress bars (if any) with <code>options("verbose"=FALSE)</code>.
</p>
<p>Directly borrowed for Claude (2008), and called <code>fourier1</code> there.
</p>


<h3>References</h3>

<p>Claude, J. (2008) <em>Morphometrics with R</em>, Use R! series,
Springer 316 pp.
</p>


<h3>See Also</h3>

<p>Other rfourier: 
<code><a href="#topic+rfourier_i">rfourier_i</a>()</code>,
<code><a href="#topic+rfourier_shape">rfourier_shape</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bot)
coo &lt;- coo_center(bot[1]) # centering is almost mandatory for rfourier family
coo_plot(coo)
rf  &lt;- rfourier(coo, 12)
rf
rfi &lt;- rfourier_i(rf)
coo_draw(rfi, border='red', col=NA)

# Out method
bot %&gt;% rfourier()
</code></pre>

<hr>
<h2 id='rfourier_i'>Inverse radii variation Fourier transform</h2><span id='topic+rfourier_i'></span>

<h3>Description</h3>

<p><code>rfourier_i</code> uses the inverse radii variation (equally spaced radii) transformation to
calculate a shape, when given a list with Fourier coefficients, typically
obtained computed with <a href="#topic+rfourier">rfourier</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rfourier_i(rf, nb.h, nb.pts = 120)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rfourier_i_+3A_rf">rf</code></td>
<td>
<p>A <code>list</code> with <code>ao</code>, <code>an</code> and <code>bn</code> components,
typically as returned by <code>rfourier</code>.</p>
</td></tr>
<tr><td><code id="rfourier_i_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to calculate/use.</p>
</td></tr>
<tr><td><code id="rfourier_i_+3A_nb.pts">nb.pts</code></td>
<td>
<p><code>integer</code>. The number of points to calculate.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the JSS paper for the maths behind.
</p>


<h3>Value</h3>

<p>A list with components: </p>
<table>
<tr><td><code>x</code></td>
<td>
<p><code>vector</code> of
<code>x</code>-coordinates.</p>
</td></tr> <tr><td><code>y</code></td>
<td>
<p><code>vector</code> of <code>y</code>-coordinates.</p>
</td></tr>
<tr><td><code>angle</code></td>
<td>
<p><code>vector</code> of angles used.</p>
</td></tr> <tr><td><code>r</code></td>
<td>
<p><code>vector</code> of radii
calculated.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Directly borrowed for Claude (2008), and called <code>ifourier1</code> there.
</p>


<h3>References</h3>

<p>Claude, J. (2008) <em>Morphometrics with R</em>, Use R! series,
Springer 316 pp.
</p>


<h3>See Also</h3>

<p>Other rfourier: 
<code><a href="#topic+rfourier_shape">rfourier_shape</a>()</code>,
<code><a href="#topic+rfourier">rfourier</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bot)
coo &lt;- coo_center(bot[1]) # centering is almost mandatory for rfourier family
coo_plot(coo)
rf  &lt;- rfourier(coo, 12)
rf
rfi &lt;- rfourier_i(rf)
coo_draw(rfi, border='red', col=NA)

</code></pre>

<hr>
<h2 id='rfourier_shape'>Calculates and draw 'rfourier' shapes.</h2><span id='topic+rfourier_shape'></span>

<h3>Description</h3>

<p><code>rfourier_shape</code> calculates a 'Fourier radii variation shape' given
Fourier coefficients (see <code>Details</code>) or can generate some 'rfourier'
shapes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rfourier_shape(an, bn, nb.h, nb.pts = 80, alpha = 2, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rfourier_shape_+3A_an">an</code></td>
<td>
<p><code>numeric</code>. The <code class="reqn">a_n</code> Fourier coefficients on which to
calculate a shape.</p>
</td></tr>
<tr><td><code id="rfourier_shape_+3A_bn">bn</code></td>
<td>
<p><code>numeric</code>. The <code class="reqn">b_n</code> Fourier coefficients on which to
calculate a shape.</p>
</td></tr>
<tr><td><code id="rfourier_shape_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to use.</p>
</td></tr>
<tr><td><code id="rfourier_shape_+3A_nb.pts">nb.pts</code></td>
<td>
<p><code>integer</code>. The number of points to calculate.</p>
</td></tr>
<tr><td><code id="rfourier_shape_+3A_alpha">alpha</code></td>
<td>
<p><code>numeric</code>. The power coefficient associated with the
(usually decreasing) amplitude of the Fourier coefficients (see
<b>Details</b>).</p>
</td></tr>
<tr><td><code id="rfourier_shape_+3A_plot">plot</code></td>
<td>
<p><code>logical</code>. Whether to plot or not the shape.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>rfourier_shape</code> can be used by specifying <code>nb.h</code> and
<code>alpha</code>. The coefficients are then sampled in an uniform distribution
<code class="reqn">(-\pi ; \pi)</code> and this amplitude is then divided by
<code class="reqn">harmonicrank^alpha</code>. If <code>alpha</code> is lower than 1, consecutive
coefficients will thus increase. See <a href="#topic+rfourier">rfourier</a> for the mathematical
background.
</p>


<h3>Value</h3>

<p>A matrix of (x; y) coordinates.
</p>


<h3>References</h3>

<p>Claude, J. (2008) <em>Morphometrics with R</em>, Use R! series,
Springer 316 pp.
</p>


<h3>See Also</h3>

<p>Other rfourier: 
<code><a href="#topic+rfourier_i">rfourier_i</a>()</code>,
<code><a href="#topic+rfourier">rfourier</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bot)
rf &lt;- rfourier(bot[1], 24)
rfourier_shape(rf$an, rf$bn) # equivalent to rfourier_i(rf)
rfourier_shape() # not very interesting

rfourier_shape(nb.h=12) # better
rfourier_shape(nb.h=6, alpha=0.4, nb.pts=500)

# Butterflies of the vignette' cover
panel(Out(a2l(replicate(100,
rfourier_shape(nb.h=6, alpha=0.4, nb.pts=200, plot=FALSE)))))
</code></pre>

<hr>
<h2 id='rm_asym'>Removes asymmetric and symmetric variation on OutCoe objects</h2><span id='topic+rm_asym'></span><span id='topic+rm_sym'></span><span id='topic+rm_asym.default'></span><span id='topic+rm_asym.OutCoe'></span><span id='topic+rm_sym.default'></span><span id='topic+rm_sym.OutCoe'></span>

<h3>Description</h3>

<p>Only for those obtained with <a href="#topic+efourier">efourier</a>, otherwise a message is returned.
<code>rm_asym</code> sets all B and C coefficients to 0; <code>rm_sym</code> sets
all A and D coefficients to 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rm_asym(OutCoe)

## Default S3 method:
rm_asym(OutCoe)

## S3 method for class 'OutCoe'
rm_asym(OutCoe)

rm_sym(OutCoe)

## Default S3 method:
rm_sym(OutCoe)

## S3 method for class 'OutCoe'
rm_sym(OutCoe)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rm_asym_+3A_outcoe">OutCoe</code></td>
<td>
<p>an OutCoe object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an OutCoe object
</p>


<h3>References</h3>

<p>Below: the first mention, and two applications.
</p>

<p>#' </p>
<ul>
<li><p> Iwata, H., Niikura, S., Matsuura, S., Takano, Y., &amp; Ukai, Y. (1998).
Evaluation of variation of root shape of Japanese radish (Raphanus sativus L.)
based on image analysis using elliptic Fourier descriptors. Euphytica, 102, 143-149.
</p>
</li>
<li><p> Iwata, H., Nesumi, H., Ninomiya, S., Takano, Y., &amp; Ukai, Y. (2002).
The Evaluation of Genotype x Environment Interactions of Citrus Leaf Morphology
Using Image Analysis and Elliptic Fourier Descriptors. Breeding Science, 52(2),
89-94. doi:10.1270/jsbbs.52.89
</p>
</li>
<li><p> Yoshioka, Y., Iwata, H., Ohsawa, R., &amp; Ninomiya, S. (2004).
Analysis of petal shape variation of Primula sieboldii by elliptic fourier descriptors
and principal component analysis. Annals of Botany, 94(5), 657-64. doi:10.1093/aob/mch190
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+symmetry">symmetry</a> and the note there.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>botf &lt;- efourier(bot, 12)
botSym &lt;- rm_asym(botf)
boxplot(botSym)
botSymp &lt;- PCA(botSym)
plot(botSymp)
plot(botSymp, amp.shp=5)

# Asymmetric only
botAsym &lt;- rm_sym(botf)
boxplot(botAsym)
botAsymp &lt;- PCA(botAsym)
plot(botAsymp)
# strange shapes because the original shape was mainly symmetric and would need its
# symmetric (eg its average) for a proper reconstruction. Should only be used like that:
plot(botAsymp, morpho=FALSE)
</code></pre>

<hr>
<h2 id='rm_harm'>Removes harmonics from Coe objects</h2><span id='topic+rm_harm'></span>

<h3>Description</h3>

<p>Useful to drop harmonics on Coe objects. Should only work for
Fourier-based approached since it looks for <code>[A-D][1-drop]</code> pattern.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rm_harm(x, drop = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rm_harm_+3A_x">x</code></td>
<td>
<p>Coe object</p>
</td></tr>
<tr><td><code id="rm_harm_+3A_drop">drop</code></td>
<td>
<p>numeric number of harmonics to drop</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bot)
bf &lt;- efourier(bot)
colnames(rm_harm(bf, 1)$coe)
</code></pre>

<hr>
<h2 id='rm_missing'>Remove shapes with missing data in fac</h2><span id='topic+rm_missing'></span>

<h3>Description</h3>

<p>Any row (or within a given column if <code>by</code> is specified) containing <code>NA</code> in <code style="white-space: pre;">&#8288;$fac&#8288;</code> and the corresponding shapes in <code style="white-space: pre;">&#8288;$coo&#8288;</code>, lines in <code style="white-space: pre;">&#8288;$coe&#8288;</code> or other objects
will also be dropped.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rm_missing(x, by)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rm_missing_+3A_x">x</code></td>
<td>
<p>the object on which to NA</p>
</td></tr>
<tr><td><code id="rm_missing_+3A_by">by</code></td>
<td>
<p>which column of the $fac should objects have complete views</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot$fac$type[3] &lt;- NA
bot$fac$fake[9] &lt;- NA

bot %&gt;% length()
bot %&gt;% rm_missing() %&gt;% length
bot %&gt;% rm_missing("fake") %&gt;% length()
</code></pre>

<hr>
<h2 id='rm_uncomplete'>Remove shapes with incomplete slices</h2><span id='topic+rm_uncomplete'></span>

<h3>Description</h3>

<p>Imagine you take three views of every object you study. Then, you can <a href="#topic+slice">slice</a>,
<a href="#topic+filter">filter</a> or <a href="#topic+chop">chop</a> your entire dataset, do morphometrics on it,
then want to <a href="#topic+combine">combine</a> it. But if you have forgotten one view, or if it
was impossible to obtain, for one or more objects, combine will not work.
This function helps you to remove those ugly ducklings. See examples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rm_uncomplete(x, id, by)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rm_uncomplete_+3A_x">x</code></td>
<td>
<p>the object on which to remove uncomplete &quot;by&quot;</p>
</td></tr>
<tr><td><code id="rm_uncomplete_+3A_id">id</code></td>
<td>
<p>of the objects, within the $fac slot</p>
</td></tr>
<tr><td><code id="rm_uncomplete_+3A_by">by</code></td>
<td>
<p>which column of the $fac should objects have complete views</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># we load olea
data(olea)
# we select the var Aglan since it is the only one complete
ol &lt;- filter(olea, var == "Aglan")
# everything seems fine
table(ol$view, ol$ind)
# indeed
rm_uncomplete(ol, id="ind", by="view")

# we mess the ol object by removing a single shape
ol.pb &lt;- slice(ol, -1)
table(ol.pb$view, ol.pb$ind)
# the counterpart has been removed with a notice
ol.ok &lt;- rm_uncomplete(ol.pb, "ind", "view")
# now you can combine them
table(ol.ok$view, ol.ok$ind)
</code></pre>

<hr>
<h2 id='rw_fac'>Renames levels on Momocs objects</h2><span id='topic+rw_fac'></span>

<h3>Description</h3>

<p>rw_fac stands for 'rewriting rule'. Typically useful to correct typos
at the import, or merge some levels within covariates. Drops levels silently.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rw_fac(x, fac, from, to)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rw_fac_+3A_x">x</code></td>
<td>
<p>any Momocs object</p>
</td></tr>
<tr><td><code id="rw_fac_+3A_fac">fac</code></td>
<td>
<p>the id of the name of the $fac column to look for (<a href="#topic+fac_dispatcher">fac_dispatcher</a> not yet supported)</p>
</td></tr>
<tr><td><code id="rw_fac_+3A_from">from</code></td>
<td>
<p>which level(s) should be renamed; passed as a single or several characters</p>
</td></tr>
<tr><td><code id="rw_fac_+3A_to">to</code></td>
<td>
<p>which name should be used to rename this/these levels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of the same class
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># single renaming
rw_fac(bot, "type", "whisky", "agua_de_fuego")$type # 1 instead of "type" is fine too
# several renaming
bot2 &lt;- mutate(bot, fake=factor(rep(letters[1:4], 10)))
rw_fac(bot2, "fake", c("a", "e"), "ae")$fake
</code></pre>

<hr>
<h2 id='sample_frac'>Sample a fraction of shapes</h2><span id='topic+sample_frac'></span>

<h3>Description</h3>

<p>Sample a fraction of shapes from a Momocs object. See examples and <code>?dplyr::sample_n</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_frac(tbl, size, replace, fac, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_frac_+3A_tbl">tbl</code></td>
<td>
<p>a Momocs object (Coo, Coe)</p>
</td></tr>
<tr><td><code id="sample_frac_+3A_size">size</code></td>
<td>
<p>numeric (0 &lt; numeric &lt;= 1) the fraction of shapes to select</p>
</td></tr>
<tr><td><code id="sample_frac_+3A_replace">replace</code></td>
<td>
<p>logical whether sample should be done with ot without replacement</p>
</td></tr>
<tr><td><code id="sample_frac_+3A_fac">fac</code></td>
<td>
<p>a column name if a <code>$fac</code> is defined; size is then applied within levels of this factor</p>
</td></tr>
<tr><td><code id="sample_frac_+3A_...">...</code></td>
<td>
<p>additional arguments to dplyr::sample_frac and to maintain generic compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>Note</h3>

<p>the resulting fraction is rounded with <a href="base.html#topic+ceiling">ceiling</a>.
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# samples 50% of the bottles no matter their type
sample_frac(bot, 0.5)
# 80% bottles of beer and of whisky
table(sample_frac(bot, 0.8, fac="type")$fac)
# bootstrap the same number of bootles of each type but with replacement
table(names(sample_frac(bot, 1, replace=TRUE)))

</code></pre>

<hr>
<h2 id='sample_n'>Sample n shapes</h2><span id='topic+sample_n'></span>

<h3>Description</h3>

<p>Sample n shapes from a Momocs object. See examples and <code>?dplyr::sample_n</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_n(tbl, size, replace, fac, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_n_+3A_tbl">tbl</code></td>
<td>
<p>a Momocs object (Coo, Coe)</p>
</td></tr>
<tr><td><code id="sample_n_+3A_size">size</code></td>
<td>
<p>numeric how many shapes should we sample</p>
</td></tr>
<tr><td><code id="sample_n_+3A_replace">replace</code></td>
<td>
<p>logical whether sample should be done with ot without replacement</p>
</td></tr>
<tr><td><code id="sample_n_+3A_fac">fac</code></td>
<td>
<p>a column name if a <code>$fac</code> is defined; size is then applied within levels of this factor</p>
</td></tr>
<tr><td><code id="sample_n_+3A_...">...</code></td>
<td>
<p>additional arguments to dplyr::sample_n and to maintain generic compatibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a Momocs object of same class
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# samples 5 bottles no matter their type
sample_n(bot, 5)
# 5 bottles of beer and of whisky
table(sample_n(bot, 5, fac="type")$type)
# many repetitions
table(names(sample_n(bot, 400, replace=TRUE)))

</code></pre>

<hr>
<h2 id='scree'>How many axes to retain this much of variance or trace ?</h2><span id='topic+scree'></span><span id='topic+scree.PCA'></span><span id='topic+scree.LDA'></span><span id='topic+scree_min'></span><span id='topic+scree_plot'></span>

<h3>Description</h3>

<p>A set of functions around PCA/LDA eigen/trace. <code>scree</code> calculates their proportion and cumulated proportion;
<code>scree_min</code> returns the minimal number of axis to use to retain a given proportion; <code>scree_plot</code> displays a screeplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scree(x, nax)

## S3 method for class 'PCA'
scree(x, nax)

## S3 method for class 'LDA'
scree(x, nax)

scree_min(x, prop)

scree_plot(x, nax)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scree_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+PCA">PCA</a> object</p>
</td></tr>
<tr><td><code id="scree_+3A_nax">nax</code></td>
<td>
<p>numeric range of axes to consider.
All by default for <code>scree_min</code>, display until <code>0.99</code> for <code>scree_plot</code></p>
</td></tr>
<tr><td><code id="scree_+3A_prop">prop</code></td>
<td>
<p>numeric how many axes are enough to gather this proportion of variance.
Default to 1, all axes are returned
defaut to 1: all axis are returned</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scree returns a data.frame, scree_min a numeric, scree_plot a ggplot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># On PCA
bp &lt;- PCA(efourier(bot))
scree(bp)
scree_min(bp, 0.99)
scree_min(bp, 1)

scree_plot(bp)
scree_plot(bp, 1:5)

# on LDA, it uses svd
bl &lt;- LDA(PCA(opoly(olea)), "var")
scree(bl)

</code></pre>

<hr>
<h2 id='select'>Select columns by name</h2><span id='topic+select'></span>

<h3>Description</h3>

<p>Select variables by name, from the <code>$fac</code>. Selected variables
can also be renamed on the fly.
See examples and <code>?dplyr::select</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select(.data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_+3A_.data">.data</code></td>
<td>
<p>a <code>Coo</code>, <code>Coe</code>, <code>PCA</code> object</p>
</td></tr>
<tr><td><code id="select_+3A_...">...</code></td>
<td>
<p>comma separated list of unquoted expressions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>dplyr verbs are maintained.
</p>


<h3>Value</h3>

<p>a Momocs object of the same class.
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>olea
select(olea, var, view) # drops domes and ind
select(olea, variety=var, domesticated_status=domes, view)
# combine with filter with magrittr pipes
# only dorsal views, and 'var' and 'domes' columns
filter(olea, view=="VD") %&gt;% select(var, domes)
head(olea$fac)
# select some columns
select(olea, domes, view)
# remove some columns
select(olea, -ind)
# rename on the fly and select some columns
select(olea, foo=domes)

</code></pre>

<hr>
<h2 id='sfourier'>Radii variation Fourier transform (equally spaced curvilinear abscissa)</h2><span id='topic+sfourier'></span><span id='topic+sfourier.default'></span><span id='topic+sfourier.Out'></span><span id='topic+sfourier.list'></span>

<h3>Description</h3>

<p><code>sfourier</code> computes radii variation Fourier analysis from a matrix or a
list of coordinates where points are equally spaced aong the curvilinear abscissa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sfourier(x, nb.h)

## Default S3 method:
sfourier(x, nb.h)

## S3 method for class 'Out'
sfourier(x, nb.h)

## S3 method for class 'list'
sfourier(x, nb.h)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sfourier_+3A_x">x</code></td>
<td>
<p>A <code>list</code> or <code>matrix</code> of coordinates or an <code>Out</code> object</p>
</td></tr>
<tr><td><code id="sfourier_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to use. If missing, 12 is used on shapes;
99 percent of harmonic power on Out objects, both with messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with following components:
</p>

<ul>
<li> <p><code>an</code> vector of <code class="reqn">a_{1-&gt;n}</code> harmonic coefficients
</p>
</li>
<li> <p><code>bn</code> vector of <code class="reqn">b_{1-&gt;n}</code> harmonic coefficients
</p>
</li>
<li> <p><code>ao</code> ao harmonic coefficient
</p>
</li>
<li> <p><code>r</code> vector of radii lengths
</p>
</li></ul>



<h3>Note</h3>

<p>The implementation is still quite experimental (as of Dec. 2016)
</p>


<h3>References</h3>

<p>Renaud S, Michaux JR (2003): Adaptive latitudinal trends in the mandible shape
of <em>Apodemus</em> wood mice. <em>J Biogeogr</em> 30:1617-1628.
</p>


<h3>See Also</h3>

<p>Other sfourier: 
<code><a href="#topic+sfourier_i">sfourier_i</a>()</code>,
<code><a href="#topic+sfourier_shape">sfourier_shape</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>molars[4] %&gt;%
coo_center %&gt;% coo_scale %&gt;% coo_interpolate(1080) %&gt;%
coo_slidedirection("right") %&gt;%
   coo_sample(360) %T&gt;% coo_plot(zoom=2) %&gt;%
   sfourier(16) %&gt;%
   sfourier_i() %&gt;%
   coo_draw(bor="red", points=TRUE)
</code></pre>

<hr>
<h2 id='sfourier_i'>Inverse radii variation Fourier transform</h2><span id='topic+sfourier_i'></span>

<h3>Description</h3>

<p><code>sfourier_i</code> uses the inverse radii variation (equally spaced curvilinear abscissa) transformation to
calculate a shape, when given a list with Fourier coefficients, typically
obtained computed with <a href="#topic+sfourier">sfourier</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sfourier_i(rf, nb.h, nb.pts = 120, dtheta = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sfourier_i_+3A_rf">rf</code></td>
<td>
<p>A <code>list</code> with <code>ao</code>, <code>an</code> and <code>bn</code> components,
typically as returned by <code>sfourier</code>.</p>
</td></tr>
<tr><td><code id="sfourier_i_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to calculate/use.</p>
</td></tr>
<tr><td><code id="sfourier_i_+3A_nb.pts">nb.pts</code></td>
<td>
<p><code>integer</code>. The number of points to calculate.</p>
</td></tr>
<tr><td><code id="sfourier_i_+3A_dtheta">dtheta</code></td>
<td>
<p><code>logical</code>. Whether to use the dtheta correction method.
<code>FALSE</code> by default. When <code>TRUE</code>, tries to correct the angular difference between
reconstructed points; otherwise equal angles are used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with components: </p>
<table>
<tr><td><code>x</code></td>
<td>
<p><code>vector</code> of
<code>x</code>-coordinates.</p>
</td></tr> <tr><td><code>y</code></td>
<td>
<p><code>vector</code> of <code>y</code>-coordinates.</p>
</td></tr>
<tr><td><code>angle</code></td>
<td>
<p><code>vector</code> of angles used.</p>
</td></tr> <tr><td><code>r</code></td>
<td>
<p><code>vector</code> of radii
calculated.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Renaud S, Pale JRM, Michaux JR (2003): Adaptive latitudinal trends in the mandible shape
of <em>Apodemus</em> wood mice. <em>J Biogeogr</em> 30:1617-1628.
</p>


<h3>See Also</h3>

<p>Other sfourier: 
<code><a href="#topic+sfourier_shape">sfourier_shape</a>()</code>,
<code><a href="#topic+sfourier">sfourier</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo &lt;- coo_center(bot[1]) # centering is almost mandatory for sfourier family
coo_plot(coo)
rf  &lt;- sfourier(coo, 12)
rf
rfi &lt;- sfourier_i(rf)
coo_draw(rfi, border='red', col=NA)

</code></pre>

<hr>
<h2 id='sfourier_shape'>Calculates and draw 'sfourier' shapes.</h2><span id='topic+sfourier_shape'></span>

<h3>Description</h3>

<p><code>sfourier_shape</code> calculates a 'Fourier radii variation shape' given
Fourier coefficients (see <code>Details</code>) or can generate some 'sfourier'
shapes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sfourier_shape(an, bn, nb.h, nb.pts = 80, alpha = 2, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sfourier_shape_+3A_an">an</code></td>
<td>
<p><code>numeric</code>. The <code class="reqn">a_n</code> Fourier coefficients on which to
calculate a shape.</p>
</td></tr>
<tr><td><code id="sfourier_shape_+3A_bn">bn</code></td>
<td>
<p><code>numeric</code>. The <code class="reqn">b_n</code> Fourier coefficients on which to
calculate a shape.</p>
</td></tr>
<tr><td><code id="sfourier_shape_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to use.</p>
</td></tr>
<tr><td><code id="sfourier_shape_+3A_nb.pts">nb.pts</code></td>
<td>
<p><code>integer</code>. The number of points to calculate.</p>
</td></tr>
<tr><td><code id="sfourier_shape_+3A_alpha">alpha</code></td>
<td>
<p><code>numeric</code>. The power coefficient associated with the
(usually decreasing) amplitude of the Fourier coefficients (see
<b>Details</b>).</p>
</td></tr>
<tr><td><code id="sfourier_shape_+3A_plot">plot</code></td>
<td>
<p><code>logical</code>. Whether to plot or not the shape.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>sfourier_shape</code> can be used by specifying <code>nb.h</code> and
<code>alpha</code>. The coefficients are then sampled in an uniform distribution
<code class="reqn">(-\pi ; \pi)</code> and this amplitude is then divided by
<code class="reqn">harmonicrank^alpha</code>. If <code>alpha</code> is lower than 1, consecutive
coefficients will thus increase. See <a href="#topic+sfourier">sfourier</a> for the mathematical
background.
</p>


<h3>Value</h3>

<p>A matrix of (x; y) coordinates.
</p>


<h3>References</h3>

<p>Renaud S, Pale JRM, Michaux JR (2003): Adaptive latitudinal trends in the mandible shape
of <em>Apodemus</em> wood mice. <em>J Biogeogr</em> 30:1617-1628.
</p>


<h3>See Also</h3>

<p>Other sfourier: 
<code><a href="#topic+sfourier_i">sfourier_i</a>()</code>,
<code><a href="#topic+sfourier">sfourier</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rf &lt;- sfourier(bot[1], 24)
sfourier_shape(rf$an, rf$bn) # equivalent to sfourier_i(rf)
sfourier_shape() # not very interesting

sfourier_shape(nb.h=12) # better
sfourier_shape(nb.h=6, alpha=0.4, nb.pts=500)

# Butterflies of the vignette' cover
panel(Out(a2l(replicate(100,
sfourier_shape(nb.h=6, alpha=0.4, nb.pts=200, plot=FALSE)))))
</code></pre>

<hr>
<h2 id='shapes'>Data: Outline coordinates of various shapes</h2><span id='topic+shapes'></span>

<h3>Description</h3>

<p>Data: Outline coordinates of various shapes
</p>


<h3>Format</h3>

<p>An <a href="#topic+Out">Out</a> object with the outline coordinates of some various shapes.
</p>


<h3>Source</h3>

<p>Borrowed default shapes from (c) Adobe Photoshop. Do not send me to jail.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+trilo">trilo</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='slice'>Subset based on positions</h2><span id='topic+slice'></span>

<h3>Description</h3>

<p>Select rows by position, based on <code>$fac</code>. See examples and <code>?dplyr::slice</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slice(.data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="slice_+3A_.data">.data</code></td>
<td>
<p>a <code>Coo</code>, <code>Coe</code>, <code>PCA</code> object</p>
</td></tr>
<tr><td><code id="slice_+3A_...">...</code></td>
<td>
<p>logical conditions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>dplyr verbs are maintained.
</p>


<h3>Value</h3>

<p>a Momocs object of the same class.
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+subsetize">subsetize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>olea
slice(olea, 1) # if you only want the coordinates, try bot[1]
slice(olea, 1:20)
slice(olea, 21:30)
</code></pre>

<hr>
<h2 id='slidings_scheme'>Extracts partitions of sliding coordinates</h2><span id='topic+slidings_scheme'></span>

<h3>Description</h3>

<p>Helper function that deduces (likely to be a reminder)
partition scheme from <code>$slidings</code> of <code>Ldk</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slidings_scheme(Coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="slidings_scheme_+3A_coo">Coo</code></td>
<td>
<p>an Ldk object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with two components: <code>n</code> the number of partition; <code>id</code>
their position. Or a NULL if no slidings are defined
</p>


<h3>See Also</h3>

<p>Other ldk/slidings methods: 
<code><a href="#topic+add_ldk">add_ldk</a>()</code>,
<code><a href="#topic+def_ldk">def_ldk</a>()</code>,
<code><a href="#topic+def_slidings">def_slidings</a>()</code>,
<code><a href="#topic+get_ldk">get_ldk</a>()</code>,
<code><a href="#topic+get_slidings">get_slidings</a>()</code>,
<code><a href="#topic+rearrange_ldk">rearrange_ldk</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># no slidings defined a NULL is returned with a message
slidings_scheme(wings)

# slidings defined
slidings_scheme(chaff)

</code></pre>

<hr>
<h2 id='stack'>Family picture of shapes</h2><span id='topic+stack'></span><span id='topic+stack.Coo'></span><span id='topic+stack.Ldk'></span>

<h3>Description</h3>

<p>Plots all the outlines, on the same graph, from a <a href="#topic+Coo">Coo</a> (<a href="#topic+Out">Out</a>, <a href="#topic+Opn">Opn</a> or <a href="#topic+Ldk">Ldk</a>)
object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Coo'
stack(
  x,
  cols,
  borders,
  fac,
  palette = col_summer,
  coo_sample = 120,
  points = FALSE,
  first.point = TRUE,
  centroid = TRUE,
  ldk = TRUE,
  ldk_pch = 3,
  ldk_col = "#FF000055",
  ldk_cex = 0.5,
  ldk_links = FALSE,
  ldk_confell = FALSE,
  ldk_contour = FALSE,
  ldk_chull = FALSE,
  ldk_labels = FALSE,
  xy.axis = TRUE,
  title = substitute(x),
  ...
)

## S3 method for class 'Ldk'
stack(
  x,
  cols,
  borders,
  first.point = TRUE,
  centroid = TRUE,
  ldk = TRUE,
  ldk_pch = 20,
  ldk_col = col_alpha("#000000", 0.5),
  ldk_cex = 0.3,
  meanshape = FALSE,
  meanshape_col = "#FF0000",
  ldk_links = FALSE,
  ldk_confell = FALSE,
  ldk_contour = FALSE,
  ldk_chull = FALSE,
  ldk_labels = FALSE,
  slidings = TRUE,
  slidings_pch = "",
  xy.axis = TRUE,
  title = substitute(x),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stack_+3A_x">x</code></td>
<td>
<p>The <code>Coo</code> object to plot.</p>
</td></tr>
<tr><td><code id="stack_+3A_cols">cols</code></td>
<td>
<p>A <code>vector</code> of colors for drawing the outlines.
Either a single value or of length exactly equals to the number of coordinates.</p>
</td></tr>
<tr><td><code id="stack_+3A_borders">borders</code></td>
<td>
<p>A <code>vector</code> of colors for drawing the borders.
Either a single value or of length exactly equals to the number of coordinates.</p>
</td></tr>
<tr><td><code id="stack_+3A_fac">fac</code></td>
<td>
<p>a factor within the $fac slot for colors</p>
</td></tr>
<tr><td><code id="stack_+3A_palette">palette</code></td>
<td>
<p>a color palette to use when fac is provided</p>
</td></tr>
<tr><td><code id="stack_+3A_coo_sample">coo_sample</code></td>
<td>
<p>if not NULL the number of point per shape to display (to plot quickly)</p>
</td></tr>
<tr><td><code id="stack_+3A_points">points</code></td>
<td>
<p><code>logical</code> whether to draw or not points</p>
</td></tr>
<tr><td><code id="stack_+3A_first.point">first.point</code></td>
<td>
<p><code>logical</code> whether to draw or not the first point</p>
</td></tr>
<tr><td><code id="stack_+3A_centroid">centroid</code></td>
<td>
<p><code>logical</code> whether to draw or not the centroid</p>
</td></tr>
<tr><td><code id="stack_+3A_ldk">ldk</code></td>
<td>
<p><code>logical</code>. Whether to display landmarks (if any).</p>
</td></tr>
<tr><td><code id="stack_+3A_ldk_pch">ldk_pch</code></td>
<td>
<p><code>pch</code> for these landmarks</p>
</td></tr>
<tr><td><code id="stack_+3A_ldk_col">ldk_col</code></td>
<td>
<p>color for these landmarks</p>
</td></tr>
<tr><td><code id="stack_+3A_ldk_cex">ldk_cex</code></td>
<td>
<p><code>cex</code> for these landmarks</p>
</td></tr>
<tr><td><code id="stack_+3A_ldk_links">ldk_links</code></td>
<td>
<p><code>logical</code> whether to draw links (of the mean shape)</p>
</td></tr>
<tr><td><code id="stack_+3A_ldk_confell">ldk_confell</code></td>
<td>
<p><code>logical</code> whether to draw conf ellipses</p>
</td></tr>
<tr><td><code id="stack_+3A_ldk_contour">ldk_contour</code></td>
<td>
<p><code>logical</code> whether to draw contour lines</p>
</td></tr>
<tr><td><code id="stack_+3A_ldk_chull">ldk_chull</code></td>
<td>
<p><code>logical</code> whether to draw convex hull</p>
</td></tr>
<tr><td><code id="stack_+3A_ldk_labels">ldk_labels</code></td>
<td>
<p><code>logical</code> whether to draw landmark labels</p>
</td></tr>
<tr><td><code id="stack_+3A_xy.axis">xy.axis</code></td>
<td>
<p>whether to draw or not the x and y axes</p>
</td></tr>
<tr><td><code id="stack_+3A_title">title</code></td>
<td>
<p>a title for the plot. The name of the <code>Coo</code> by default</p>
</td></tr>
<tr><td><code id="stack_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <a href="#topic+coo_plot">coo_plot</a></p>
</td></tr>
<tr><td><code id="stack_+3A_meanshape">meanshape</code></td>
<td>
<p><code>logical</code> whether to add meanshape related stuff (below)</p>
</td></tr>
<tr><td><code id="stack_+3A_meanshape_col">meanshape_col</code></td>
<td>
<p>a color for everything meanshape</p>
</td></tr>
<tr><td><code id="stack_+3A_slidings">slidings</code></td>
<td>
<p><code>logical</code> whether to draw slidings semi landmarks</p>
</td></tr>
<tr><td><code id="stack_+3A_slidings_pch">slidings_pch</code></td>
<td>
<p><code>pch</code> for semi landmarks</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a plot
</p>


<h3>See Also</h3>

<p>Other Coo_graphics: 
<code><a href="#topic+inspect">inspect</a>()</code>,
<code><a href="#topic+panel">panel</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
stack(bot)
bot.f &lt;- efourier(bot, 12)
stack(bot.f)
stack(mosquito, borders='#1A1A1A22', first.point=FALSE)
stack(hearts)
stack(hearts, ldk=FALSE)
stack(hearts, borders='#1A1A1A22', ldk=TRUE, ldk_col=col_summer(4), ldk_pch=20)
stack(hearts, fac="aut", palette=col_sari)

chaffal &lt;- fgProcrustes(chaff)
stack(chaffal, slidings=FALSE)
stack(chaffal, meanshape=TRUE, meanshape_col="blue")

</code></pre>

<hr>
<h2 id='subsetize'>Subsetize various Momocs objects</h2><span id='topic+subsetize'></span>

<h3>Description</h3>

<p>Subsetize is a wrapper around dplyr's verbs and should NOT be used directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetize(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetize_+3A_x">x</code></td>
<td>
<p>a <code>Coo</code> or a <a href="#topic+Coe">Coe</a> object.</p>
</td></tr>
<tr><td><code id="subsetize_+3A_subset">subset</code></td>
<td>
<p>logical taken from the <code>$fac</code> slot, or indices. See examples.</p>
</td></tr>
<tr><td><code id="subsetize_+3A_...">...</code></td>
<td>
<p>useless here but maintains consistence with the generic subset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a subsetted object of same class
</p>


<h3>See Also</h3>

<p>Other handling functions: 
<code><a href="#topic+arrange">arrange</a>()</code>,
<code><a href="#topic+at_least">at_least</a>()</code>,
<code><a href="#topic+chop">chop</a>()</code>,
<code><a href="#topic+combine">combine</a>()</code>,
<code><a href="#topic+dissolve">dissolve</a>()</code>,
<code><a href="#topic+fac_dispatcher">fac_dispatcher</a>()</code>,
<code><a href="#topic+filter">filter</a>()</code>,
<code><a href="#topic+mutate">mutate</a>()</code>,
<code><a href="#topic+rename">rename</a>()</code>,
<code><a href="#topic+rescale">rescale</a>()</code>,
<code><a href="#topic+rm_harm">rm_harm</a>()</code>,
<code><a href="#topic+rm_missing">rm_missing</a>()</code>,
<code><a href="#topic+rm_uncomplete">rm_uncomplete</a>()</code>,
<code><a href="#topic+rw_fac">rw_fac</a>()</code>,
<code><a href="#topic+sample_frac">sample_frac</a>()</code>,
<code><a href="#topic+sample_n">sample_n</a>()</code>,
<code><a href="#topic+select">select</a>()</code>,
<code><a href="#topic+slice">slice</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Do not use subset directly
</code></pre>

<hr>
<h2 id='symmetry'>Calcuates symmetry indices on OutCoe objects</h2><span id='topic+symmetry'></span>

<h3>Description</h3>

<p>For <a href="#topic+OutCoe">OutCoe</a> objects obtained with <a href="#topic+efourier">efourier</a>, calculates several
indices on the matrix of coefficients: <code>AD</code>, the sum of absolute values of
harmonic coefficients A and D; <code>BC</code> same thing for B and C; <code>amp</code> the
sum of the absolute value of all harmonic coefficients and <code>sym</code> which is the ratio
of <code>AD</code> over <code>amp</code>. See references below for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symmetry(OutCoe)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="symmetry_+3A_outcoe">OutCoe</code></td>
<td>
<p><a href="#topic+efourier">efourier</a> objects</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix with 4 colums described above.
</p>


<h3>Note</h3>

<p>What we call symmetry here is bilateral symmetry.
By comparing coefficients resulting from <a href="#topic+efourier">efourier</a>,
with AD responsible for amplitude of the Fourier functions,
and BC for their phase, it results in the plane and for
fitted/reconstructed shapes that symmetry. As long as your shapes are
aligned along their bilateral symmetry axis, you can use the approach
coined by Iwata et al., and here implemented in Momocs.
</p>


<h3>References</h3>

<p>Below: the first mention, and two applications.
</p>

<p>#' </p>
<ul>
<li><p> Iwata, H., Niikura, S., Matsuura, S., Takano, Y., &amp; Ukai, Y. (1998).
Evaluation of variation of root shape of Japanese radish (Raphanus sativus L.)
based on image analysis using elliptic Fourier descriptors. Euphytica, 102, 143-149.
</p>
</li>
<li><p> Iwata, H., Nesumi, H., Ninomiya, S., Takano, Y., &amp; Ukai, Y. (2002).
The Evaluation of Genotype x Environment Interactions of Citrus Leaf Morphology
Using Image Analysis and Elliptic Fourier Descriptors. Breeding Science, 52(2),
89-94. doi:10.1270/jsbbs.52.89
</p>
</li>
<li><p> Yoshioka, Y., Iwata, H., Ohsawa, R., &amp; Ninomiya, S. (2004).
Analysis of petal shape variation of Primula sieboldii by elliptic fourier descriptors
and principal component analysis. Annals of Botany, 94(5), 657-64. doi:10.1093/aob/mch190
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+rm_asym">rm_asym</a> and <a href="#topic+rm_sym">rm_sym</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bot.f &lt;- efourier(bot, 12)
res &lt;- symmetry(bot.f)
hist(res[, 'sym'])
</code></pre>

<hr>
<h2 id='tfourier'>Tangent angle Fourier transform</h2><span id='topic+tfourier'></span><span id='topic+tfourier.default'></span><span id='topic+tfourier.Out'></span><span id='topic+tfourier.list'></span>

<h3>Description</h3>

<p><code>tfourier</code> computes tangent angle Fourier analysis from a matrix or a
list of coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tfourier(x, ...)

## Default S3 method:
tfourier(x, nb.h, smooth.it = 0, norm = FALSE, ...)

## S3 method for class 'Out'
tfourier(x, nb.h = 40, smooth.it = 0, norm = TRUE, ...)

## S3 method for class 'list'
tfourier(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tfourier_+3A_x">x</code></td>
<td>
<p>A list or matrix of coordinates or an <code>Out</code></p>
</td></tr>
<tr><td><code id="tfourier_+3A_...">...</code></td>
<td>
<p>useless here</p>
</td></tr>
<tr><td><code id="tfourier_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to use. If missing, 12 is used on shapes;
99 percent of harmonic power on Out objects, both with messages.</p>
</td></tr>
<tr><td><code id="tfourier_+3A_smooth.it">smooth.it</code></td>
<td>
<p><code>integer</code>. The number of smoothing iterations to
perform</p>
</td></tr>
<tr><td><code id="tfourier_+3A_norm">norm</code></td>
<td>
<p><code>logical</code>. Whether to scale and register new coordinates so
that the first point used is sent on the origin.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>

<ul>
<li> <p><code>ao</code> ao harmonic coefficient
</p>
</li>
<li> <p><code>an</code> vector of <code class="reqn">a_{1-&gt;n}</code> harmonic coefficients
</p>
</li>
<li> <p><code>bn</code> vector of <code class="reqn">b_{1-&gt;n}</code> harmonic coefficients
</p>
</li>
<li> <p><code>phi</code> vector of variation of the tangent angle
</p>
</li>
<li> <p><code>t</code> vector of distance along the perimeter expressed in radians
</p>
</li>
<li> <p><code>perimeter</code> numeric. The perimeter of the outline
</p>
</li>
<li> <p><code>thetao</code> numeric. The first tangent angle
</p>
</li>
<li> <p><code>x1</code> The x-coordinate of the first point
</p>
</li>
<li> <p><code>y1</code> The y-coordinate of the first point.
</p>
</li></ul>



<h3>Note</h3>

<p>Silent message and progress bars (if any) with <code>options("verbose"=FALSE)</code>.
</p>
<p>Directly borrowed for Claude (2008), and called <code>fourier2</code> there.
</p>


<h3>References</h3>

<p>Zahn CT, Roskies RZ. 1972. Fourier Descriptors for Plane Closed
Curves. <em>IEEE Transactions on Computers</em> <b>C-21</b>: 269-281.
</p>
<p>Claude, J. (2008) <em>Morphometrics with R</em>, Use R! series, Springer 316
pp.
</p>


<h3>See Also</h3>

<p>Other tfourier: 
<code><a href="#topic+tfourier_i">tfourier_i</a>()</code>,
<code><a href="#topic+tfourier_shape">tfourier_shape</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coo &lt;- bot[1]
coo_plot(coo)
tf  &lt;- tfourier(coo, 12)
tf
tfi &lt;- tfourier_i(tf)
coo_draw(tfi, border='red', col=NA) # the outline is not closed...
coo_draw(tfourier_i(tf, force2close=TRUE), border='blue', col=NA) # we force it to close.
</code></pre>

<hr>
<h2 id='tfourier_i'>Inverse tangent angle Fourier transform</h2><span id='topic+tfourier_i'></span>

<h3>Description</h3>

<p><code>tfourier_i</code> uses the inverse tangent angle Fourier transformation to
calculate a shape, when given a list with Fourier coefficients, typically
obtained computed with <a href="#topic+tfourier">tfourier</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tfourier_i(
  tf,
  nb.h,
  nb.pts = 120,
  force2close = FALSE,
  rescale = TRUE,
  perim = 2 * pi,
  thetao = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tfourier_i_+3A_tf">tf</code></td>
<td>
<p>a list with ao, an and bn components, typically as returned by
tfourier</p>
</td></tr>
<tr><td><code id="tfourier_i_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to calculate/use</p>
</td></tr>
<tr><td><code id="tfourier_i_+3A_nb.pts">nb.pts</code></td>
<td>
<p><code>integer</code>. The number of points to calculate</p>
</td></tr>
<tr><td><code id="tfourier_i_+3A_force2close">force2close</code></td>
<td>
<p><code>logical</code>. Whether to force the outlines calculated
to close (see <a href="#topic+coo_force2close">coo_force2close</a>).</p>
</td></tr>
<tr><td><code id="tfourier_i_+3A_rescale">rescale</code></td>
<td>
<p><code>logical</code>. Whether to rescale the points calculated so
that their perimeter equals <code>perim</code>.</p>
</td></tr>
<tr><td><code id="tfourier_i_+3A_perim">perim</code></td>
<td>
<p>The perimeter length to rescale shapes.</p>
</td></tr>
<tr><td><code id="tfourier_i_+3A_thetao">thetao</code></td>
<td>
<p><code>numeric</code>. Radius angle to the reference (in radians)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <a href="#topic+tfourier">tfourier</a> for the mathematical background.
</p>


<h3>Value</h3>

<p>A list with components: </p>
<table>
<tr><td><code>x</code></td>
<td>
<p><code>vector</code> of
<code>x</code>-coordinates.</p>
</td></tr> <tr><td><code>y</code></td>
<td>
<p><code>vector</code> of <code>y</code>-coordinates.</p>
</td></tr>
<tr><td><code>phi</code></td>
<td>
<p><code>vector</code> of interpolated changes on the tangent angle.</p>
</td></tr>
<tr><td><code>angle</code></td>
<td>
<p><code>vector</code> of position on the perimeter (in radians).</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Directly borrowed for Claude (2008), and called <code>ifourier2</code> there.
</p>


<h3>References</h3>

<p>Zahn CT, Roskies RZ. 1972. Fourier Descriptors for Plane Closed
Curves. <em>IEEE Transactions on Computers</em> <b>C-21</b>: 269-281.
</p>
<p>Claude, J. (2008) <em>Morphometrics with R</em>, Use R! series, Springer 316
pp.
</p>


<h3>See Also</h3>

<p>Other tfourier: 
<code><a href="#topic+tfourier_shape">tfourier_shape</a>()</code>,
<code><a href="#topic+tfourier">tfourier</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tfourier(bot[1], 24)
tfourier_shape()
</code></pre>

<hr>
<h2 id='tfourier_shape'>Calculates and draws 'tfourier' shapes.</h2><span id='topic+tfourier_shape'></span>

<h3>Description</h3>

<p><code>tfourier_shape</code> calculates a 'Fourier tangent angle shape' given
Fourier coefficients (see <code>Details</code>) or can generate some 'tfourier'
shapes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tfourier_shape(an, bn, ao = 0, nb.h, nb.pts = 80, alpha = 2, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tfourier_shape_+3A_an">an</code></td>
<td>
<p><code>numeric</code>. The <code class="reqn">a_n</code> Fourier coefficients on which to
calculate a shape.</p>
</td></tr>
<tr><td><code id="tfourier_shape_+3A_bn">bn</code></td>
<td>
<p><code>numeric</code>. The <code class="reqn">b_n</code> Fourier coefficients on which to
calculate a shape.</p>
</td></tr>
<tr><td><code id="tfourier_shape_+3A_ao">ao</code></td>
<td>
<p><code>ao</code> Harmonic coefficient.</p>
</td></tr>
<tr><td><code id="tfourier_shape_+3A_nb.h">nb.h</code></td>
<td>
<p><code>integer</code>. The number of harmonics to use.</p>
</td></tr>
<tr><td><code id="tfourier_shape_+3A_nb.pts">nb.pts</code></td>
<td>
<p><code>integer</code>. The number of points to calculate.</p>
</td></tr>
<tr><td><code id="tfourier_shape_+3A_alpha">alpha</code></td>
<td>
<p><code>numeric</code>. The power coefficient associated with the
(usually decreasing) amplitude of the Fourier coefficients (see
<b>Details</b>).</p>
</td></tr>
<tr><td><code id="tfourier_shape_+3A_plot">plot</code></td>
<td>
<p><code>logical</code>. Whether to plot or not the shape.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of (x; y) coordinates.
</p>


<h3>References</h3>

<p>Claude, J. (2008) <em>Morphometrics with R</em>, Use R! series,
Springer 316 pp.
</p>


<h3>See Also</h3>

<p>Other tfourier: 
<code><a href="#topic+tfourier_i">tfourier_i</a>()</code>,
<code><a href="#topic+tfourier">tfourier</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tf &lt;- tfourier(bot[1], 24)
tfourier_shape(tf$an, tf$bn) # equivalent to rfourier_i(rf)
tfourier_shape()
tfourier_shape(nb.h=6, alpha=0.4, nb.pts=500)
panel(Out(a2l(replicate(100,
coo_force2close(tfourier_shape(nb.h=6, alpha=2, nb.pts=200, plot=FALSE)))))) # biological shapes
</code></pre>

<hr>
<h2 id='tie_jpg_txt'>Binds .jpg outlines from .txt landmarks taken on them</h2><span id='topic+tie_jpg_txt'></span>

<h3>Description</h3>

<p>Given a list of files (lf) that includes matching filenames with .jpg (black masks)
and .txt (landmark positions on them as .txt), returns an Out with $ldk defined.
Typically be useful if you use ImageJ to define landmarks on your outlines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tie_jpg_txt(lf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tie_jpg_txt_+3A_lf">lf</code></td>
<td>
<p>a list of filenames</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an Out object
</p>


<h3>Note</h3>

<p>Not optimized (images are read twice). Please do not hesitate to contact me
should you have a particular case or need something.
</p>


<h3>See Also</h3>

<p>Other babel functions: 
<code><a href="#topic+lf_structure">lf_structure</a>()</code>
</p>

<hr>
<h2 id='tps_arr'>Deformation 'vector field' using Thin Plate Splines</h2><span id='topic+tps_arr'></span>

<h3>Description</h3>

<p><code>tps_arr</code>(ows) calculates deformations between two configurations and
illustrate them using arrows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tps_arr(
  fr,
  to,
  amp = 1,
  grid = TRUE,
  over = 1.2,
  palette = col_summer,
  arr.nb = 200,
  arr.levels = 100,
  arr.len = 0.1,
  arr.ang = 20,
  arr.lwd = 0.75,
  arr.col = "grey50",
  poly = TRUE,
  shp = TRUE,
  shp.col = rep(NA, 2),
  shp.border = col_qual(2),
  shp.lwd = c(2, 2),
  shp.lty = c(1, 1),
  legend = TRUE,
  legend.text,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tps_arr_+3A_fr">fr</code></td>
<td>
<p>the reference <code class="reqn">(x; y)</code> coordinates</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_to">to</code></td>
<td>
<p>the target <code class="reqn">(x; y)</code> coordinates</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_amp">amp</code></td>
<td>
<p>an amplification factor of differences between <code>fr</code> and
<code>to</code></p>
</td></tr>
<tr><td><code id="tps_arr_+3A_grid">grid</code></td>
<td>
<p>whether to calculate and plot changes across the graphical window
<code>TRUE</code> or just within the starting shape (<code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_over">over</code></td>
<td>
<p><code>numeric</code> that indicates how much the thin plate splines
extends over the shapes</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_palette">palette</code></td>
<td>
<p>a color palette such those included in Momocs or produced
with <a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></p>
</td></tr>
<tr><td><code id="tps_arr_+3A_arr.nb">arr.nb</code></td>
<td>
<p><code>numeric</code> The number of arrows to calculate</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_arr.levels">arr.levels</code></td>
<td>
<p><code>numeric</code>. The number of levels for the color of
arrows</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_arr.len">arr.len</code></td>
<td>
<p><code>numeric</code> for the length of arrows</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_arr.ang">arr.ang</code></td>
<td>
<p><code>numeric</code> for the angle for arrows' heads</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_arr.lwd">arr.lwd</code></td>
<td>
<p><code>numeric</code> for the <code>lwd</code> for drawing arrows</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_arr.col">arr.col</code></td>
<td>
<p>if <code>palette</code> is not used the color for arrows</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_poly">poly</code></td>
<td>
<p>whether to draw polygons (for outlines) or points (for landmarks)</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_shp">shp</code></td>
<td>
<p><code>logical</code>. whether to draw shapes</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_shp.col">shp.col</code></td>
<td>
<p>two colors for filling the shapes</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_shp.border">shp.border</code></td>
<td>
<p>two colors for drawing the borders</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_shp.lwd">shp.lwd</code></td>
<td>
<p>two <code>lwd</code> for drawing shapes</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_shp.lty">shp.lty</code></td>
<td>
<p>two <code>lty</code> fro drawing the shapes</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_legend">legend</code></td>
<td>
<p>logical whether to plot a legend</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_legend.text">legend.text</code></td>
<td>
<p>some text for the legend</p>
</td></tr>
<tr><td><code id="tps_arr_+3A_...">...</code></td>
<td>
<p>additional arguments to feed <a href="#topic+coo_draw">coo_draw</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>See Also</h3>

<p>Other thin plate splines: 
<code><a href="#topic+tps2d">tps2d</a>()</code>,
<code><a href="#topic+tps_grid">tps_grid</a>()</code>,
<code><a href="#topic+tps_iso">tps_iso</a>()</code>,
<code><a href="#topic+tps_raw">tps_raw</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>botF &lt;- efourier(bot)
x &lt;- MSHAPES(botF, 'type', nb.pts=80)$shp
fr &lt;- x$beer
to &lt;- x$whisky
tps_arr(fr, to, arr.nb=200, palette=col_sari, amp=3)
tps_arr(fr, to, arr.nb=200, palette=col_sari, amp=3, grid=FALSE)
</code></pre>

<hr>
<h2 id='tps_grid'>Deformation grids using Thin Plate Splines</h2><span id='topic+tps_grid'></span>

<h3>Description</h3>

<p><code>tps_grid</code> calculates and plots deformation grids between two
configurations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tps_grid(
  fr,
  to,
  amp = 1,
  over = 1.2,
  grid.size = 15,
  grid.col = "grey80",
  poly = TRUE,
  shp = TRUE,
  shp.col = rep(NA, 2),
  shp.border = col_qual(2),
  shp.lwd = c(1, 1),
  shp.lty = c(1, 1),
  legend = TRUE,
  legend.text,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tps_grid_+3A_fr">fr</code></td>
<td>
<p>the reference <code class="reqn">(x; y)</code> coordinates</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_to">to</code></td>
<td>
<p>the target <code class="reqn">(x; y)</code> coordinates</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_amp">amp</code></td>
<td>
<p>an amplification factor of differences between <code>fr</code> and
<code>to</code></p>
</td></tr>
<tr><td><code id="tps_grid_+3A_over">over</code></td>
<td>
<p><code>numeric</code> that indicates how much the thin plate splines
extends over the shapes</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_grid.size">grid.size</code></td>
<td>
<p><code>numeric</code> to specify the number of grid cells on the
longer axis on the outlines</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_grid.col">grid.col</code></td>
<td>
<p>color for drawing the grid</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_poly">poly</code></td>
<td>
<p>whether to draw polygons (for outlines) or points (for landmarks)</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_shp">shp</code></td>
<td>
<p><code>logical</code>. Whether to draw shapes</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_shp.col">shp.col</code></td>
<td>
<p>Two colors for filling the shapes</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_shp.border">shp.border</code></td>
<td>
<p>Two colors for drawing the borders</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_shp.lwd">shp.lwd</code></td>
<td>
<p>Two <code>lwd</code> for drawing shapes</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_shp.lty">shp.lty</code></td>
<td>
<p>Two <code>lty</code> fro drawing the shapes</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_legend">legend</code></td>
<td>
<p>logical whether to plot a legend</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_legend.text">legend.text</code></td>
<td>
<p>some text for the legend</p>
</td></tr>
<tr><td><code id="tps_grid_+3A_...">...</code></td>
<td>
<p>additional arguments to feed <a href="#topic+coo_draw">coo_draw</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>See Also</h3>

<p>Other thin plate splines: 
<code><a href="#topic+tps2d">tps2d</a>()</code>,
<code><a href="#topic+tps_arr">tps_arr</a>()</code>,
<code><a href="#topic+tps_iso">tps_iso</a>()</code>,
<code><a href="#topic+tps_raw">tps_raw</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>botF &lt;- efourier(bot)
x &lt;- MSHAPES(botF, 'type', nb.pts=80)$shp
fr &lt;- x$beer
to &lt;- x$whisky
tps_grid(fr, to, amp=3, grid.size=10)
</code></pre>

<hr>
<h2 id='tps_iso'>Deformation isolines using Thin Plate Splines.</h2><span id='topic+tps_iso'></span>

<h3>Description</h3>

<p><code>tps_iso</code> calculates deformations between two configurations and map
them with or without isolines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tps_iso(
  fr,
  to,
  amp = 1,
  grid = FALSE,
  over = 1.2,
  palette = col_spring,
  iso.nb = 1000,
  iso.levels = 12,
  cont = TRUE,
  cont.col = "black",
  poly = TRUE,
  shp = TRUE,
  shp.border = col_qual(2),
  shp.lwd = c(2, 2),
  shp.lty = c(1, 1),
  legend = TRUE,
  legend.text,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tps_iso_+3A_fr">fr</code></td>
<td>
<p>The reference <code class="reqn">(x; y)</code> coordinates</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_to">to</code></td>
<td>
<p>The target <code class="reqn">(x; y)</code> coordinates</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_amp">amp</code></td>
<td>
<p>An amplification factor of differences between <code>fr</code> and
<code>to</code></p>
</td></tr>
<tr><td><code id="tps_iso_+3A_grid">grid</code></td>
<td>
<p>whether to calculate and plot changes across the graphical window
<code>TRUE</code> or just within the starting shape (<code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_over">over</code></td>
<td>
<p>A <code>numeric</code> that indicates how much the thin plate splines
extends over the shapes</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_palette">palette</code></td>
<td>
<p>A color palette such those included in Momocs or produced
with <a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></p>
</td></tr>
<tr><td><code id="tps_iso_+3A_iso.nb">iso.nb</code></td>
<td>
<p>A <code>numeric</code>. The number of points to use for the
calculation of deformation</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_iso.levels">iso.levels</code></td>
<td>
<p><code>numeric</code>. The number of levels for mapping the
deformations</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_cont">cont</code></td>
<td>
<p><code>logical</code>. Whether to draw contour lines</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_cont.col">cont.col</code></td>
<td>
<p>A color for drawing the contour lines</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_poly">poly</code></td>
<td>
<p>whether to draw polygons (for outlines) or points (for landmarks)</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_shp">shp</code></td>
<td>
<p><code>logical</code>. Whether to draw shapes</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_shp.border">shp.border</code></td>
<td>
<p>Two colors for drawing the borders</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_shp.lwd">shp.lwd</code></td>
<td>
<p>Two <code>lwd</code> for drawing shapes</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_shp.lty">shp.lty</code></td>
<td>
<p>Two <code>lty</code> fro drawing the shapes</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_legend">legend</code></td>
<td>
<p>logical whether to plot a legend</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_legend.text">legend.text</code></td>
<td>
<p>some text for the legend</p>
</td></tr>
<tr><td><code id="tps_iso_+3A_...">...</code></td>
<td>
<p>additional arguments to feed <a href="#topic+coo_draw">coo_draw</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>No returned value
</p>


<h3>See Also</h3>

<p>Other thin plate splines: 
<code><a href="#topic+tps2d">tps2d</a>()</code>,
<code><a href="#topic+tps_arr">tps_arr</a>()</code>,
<code><a href="#topic+tps_grid">tps_grid</a>()</code>,
<code><a href="#topic+tps_raw">tps_raw</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>botF &lt;- efourier(bot)
x &lt;- MSHAPES(botF, 'type', nb.pts=80)$shp
fr &lt;- x$beer
to &lt;- x$whisky
tps_iso(fr, to, iso.nb=200, amp=3)
tps_iso(fr, to, iso.nb=200, amp=3, grid=TRUE)
</code></pre>

<hr>
<h2 id='tps_raw'>Vanilla Thin Plate Splines</h2><span id='topic+tps_raw'></span>

<h3>Description</h3>

<p><code>tps_raw</code> calculates deformation grids and
returns position of sampled points on it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tps_raw(fr, to, amp = 1, over = 1.2, grid.size = 15)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tps_raw_+3A_fr">fr</code></td>
<td>
<p>the reference <code class="reqn">(x; y)</code> coordinates</p>
</td></tr>
<tr><td><code id="tps_raw_+3A_to">to</code></td>
<td>
<p>the target <code class="reqn">(x; y)</code> coordinates</p>
</td></tr>
<tr><td><code id="tps_raw_+3A_amp">amp</code></td>
<td>
<p>an amplification factor of differences between <code>fr</code> and
<code>to</code></p>
</td></tr>
<tr><td><code id="tps_raw_+3A_over">over</code></td>
<td>
<p><code>numeric</code> that indicates how much the thin plate splines
extends over the shapes</p>
</td></tr>
<tr><td><code id="tps_raw_+3A_grid.size">grid.size</code></td>
<td>
<p><code>numeric</code> to specify the number of grid cells on the
longer axis on the outlines</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with two components: <code>grid</code> the xy coordinates of sampled
points along the grid; <code>dim</code> the dimension of the grid.
</p>


<h3>See Also</h3>

<p>Other thin plate splines: 
<code><a href="#topic+tps2d">tps2d</a>()</code>,
<code><a href="#topic+tps_arr">tps_arr</a>()</code>,
<code><a href="#topic+tps_grid">tps_grid</a>()</code>,
<code><a href="#topic+tps_iso">tps_iso</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ms &lt;- MSHAPES(efourier(bot, 10), "type")
b &lt;- ms$shp$beer
w &lt;- ms$shp$whisky
g &lt;- tps_raw(b, w)
ldk_plot(g$grid)

# a wavy plot
ldk_plot(g$grid, pch=NA)
cols_ids &lt;- 1:g$dim[1]
for (i in 1:g$dim[2]) lines(g$grid[cols_ids + (i-1)*g$dim[1], ])

</code></pre>

<hr>
<h2 id='tps2d'>Thin Plate Splines for 2D data</h2><span id='topic+tps2d'></span><span id='topic+tps_apply'></span>

<h3>Description</h3>

<p><code>tps2d</code> is the core function for Thin Plate Splines. It is used internally for
all TPS graphical functions.<code>tps_apply</code> is the very same function but with
arguments properly named (I maintain tps2d as it is for historical reasons) when
we want a apply a trasnformation grid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tps2d(grid0, fr, to)

tps_apply(fr, to, new)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tps2d_+3A_grid0">grid0</code></td>
<td>
<p>a matrix of coordinates on which to calculate deformations</p>
</td></tr>
<tr><td><code id="tps2d_+3A_fr">fr</code></td>
<td>
<p>the reference shape</p>
</td></tr>
<tr><td><code id="tps2d_+3A_to">to</code></td>
<td>
<p>the target shape</p>
</td></tr>
<tr><td><code id="tps2d_+3A_new">new</code></td>
<td>
<p>the shape on which to apply the <code>shp1-&gt;shp2</code> calibrated tps trasnformation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a shape.
</p>


<h3>See Also</h3>

<p>Other thin plate splines: 
<code><a href="#topic+tps_arr">tps_arr</a>()</code>,
<code><a href="#topic+tps_grid">tps_grid</a>()</code>,
<code><a href="#topic+tps_iso">tps_iso</a>()</code>,
<code><a href="#topic+tps_raw">tps_raw</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
shapes &lt;- shapes %&gt;%
 coo_scale() %&gt;% coo_center() %&gt;%
 coo_slidedirection("up") %&gt;%
 coo_sample(64)

leaf1 &lt;- shapes[14]
leaf2 &lt;- shapes[15]

# tps grid on the two leafs2
tps_grid(leaf1, leaf2)
# apply the (leaf1 -&gt; leaf2) tps trasnformation  onto leaf1
# (that thus get closer to leaf2)
tps_apply(leaf1, leaf2, leaf1) %&gt;% coo_draw(bor="purple")

</code></pre>

<hr>
<h2 id='TraCoe'>Traditional morphometrics class</h2><span id='topic+TraCoe'></span>

<h3>Description</h3>

<p>Defines the builder for traditional measurement class in Momocs.
Is is intended to ease calculations, data handling and multivariate statistics
just ad the other Momocs' classes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TraCoe(coe = matrix(), fac = dplyr::tibble())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TraCoe_+3A_coe">coe</code></td>
<td>
<p>a matrix of measurements</p>
</td></tr>
<tr><td><code id="TraCoe_+3A_fac">fac</code></td>
<td>
<p>a data.frame for covariates</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of class TraCoe
</p>


<h3>See Also</h3>

<p>Other classes: 
<code><a href="#topic+Coe">Coe</a>()</code>,
<code><a href="#topic+Coo">Coo</a>()</code>,
<code><a href="#topic+Ldk">Ldk</a>()</code>,
<code><a href="#topic+OpnCoe">OpnCoe</a>()</code>,
<code><a href="#topic+Opn">Opn</a>()</code>,
<code><a href="#topic+OutCoe">OutCoe</a>()</code>,
<code><a href="#topic+Out">Out</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># let's (more or less) rebuild the flower dataset
fl &lt;- TraCoe(iris[, 1:4], dplyr::tibble(sp=iris$Species))
fl %&gt;% PCA() %&gt;% plot("sp")
</code></pre>

<hr>
<h2 id='trilo'>Data: Outline coordinates of cephalic outlines of trilobite</h2><span id='topic+trilo'></span>

<h3>Description</h3>

<p>Data: Outline coordinates of cephalic outlines of trilobite
</p>


<h3>Format</h3>

<p>A <a href="#topic+Out">Out</a> object 64 coordinates of 50 cephalic outlines from different
ontogenetic stages of trilobite.
</p>


<h3>Source</h3>

<p>Arranged from: <a href="https://folk.universitetetioslo.no/">https://folk.universitetetioslo.no/</a> (used to be in <code>ohammer</code> website but
seems to be deprecated now).
The original data included 51 outlines and 5 ontogenetic stages,
but one of them has just a single outline thas has been removed.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+wings">wings</a></code>
</p>

<hr>
<h2 id='verify'>Validates Coo objects</h2><span id='topic+verify'></span>

<h3>Description</h3>

<p>No validation for S3 objects, so this method is a (cheap) attempt at checking
<a href="#topic+Coo">Coo</a> objects, <a href="#topic+Out">Out</a>, <a href="#topic+Opn">Opn</a> and <a href="#topic+Ldk">Ldk</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verify(Coo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="verify_+3A_coo">Coo</code></td>
<td>
<p>any Coo object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Implemented before all morphometric methods and handling verbs.
To see what is checked, try eg <code>Momocs:::verify.Coo</code>
</p>


<h3>Value</h3>

<p>a Coo object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>verify(bot)
bot[12] &lt;- NA
# you would not use try, but here we cope with R CMD CHECK standards
plop &lt;- try(verify(bot), silent=TRUE)
class(plop)

verify(hearts)
hearts$ldk[[4]] &lt;- c(1, 2)
# same remark
plop2 &lt;- try(verify(hearts), silent=TRUE)
class(plop2)
</code></pre>

<hr>
<h2 id='which_out'>Identify outliers</h2><span id='topic+which_out'></span>

<h3>Description</h3>

<p>A simple wrapper around <a href="stats.html#topic+dnorm">dnorm</a> that helps identify outliers. In particular,
it may be useful on <a href="#topic+Coe">Coe</a> object (in this case a PCA is first calculated) and also
on <a href="#topic+Ldk">Ldk</a> for detecting possible outliers on freshly digitized/imported datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>which_out(x, conf, nax, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="which_out_+3A_x">x</code></td>
<td>
<p>object, either Coe or a numeric on which to search for outliers</p>
</td></tr>
<tr><td><code id="which_out_+3A_conf">conf</code></td>
<td>
<p>confidence for dnorm (1e-3 by default)</p>
</td></tr>
<tr><td><code id="which_out_+3A_nax">nax</code></td>
<td>
<p>number of axes to retain (only for Coe),
if &lt;1 retain enough axes to retain this proportion of the variance</p>
</td></tr>
<tr><td><code id="which_out_+3A_...">...</code></td>
<td>
<p>additional parameters to be passed to PCA (only for Coe)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of indices
</p>


<h3>Note</h3>

<p>experimental. dnorm parameters used are <code>median(x), sd(x)</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># on a numeric
x &lt;- rnorm(10)
x[4] &lt;- 99
which_out(x)

# on a Coe
bf &lt;- bot %&gt;% efourier(6)
bf$coe[c(1, 6), 1] &lt;- 5
which_out(bf)

# on Ldk
w_no &lt;- w_ok &lt;- wings
w_no$coo[[2]][1, 1] &lt;- 2
w_no$coo[[6]][2, 2] &lt;- 2
which_out(w_ok, conf=1e-12) # with low conf, no outliers
which_out(w_no, conf=1e-12) # as expected

# a way to illustrate, filter outliers
# conf has been chosen deliberately low to show some outliers
x_f &lt;- bot %&gt;% efourier
x_p &lt;- PCA(x_f)
# which are outliers (conf is ridiculously low here)
which_out(x_p$x[, 1], 0.5)
cols &lt;- rep("black", nrow(x_p$x))
outliers &lt;- which_out(x_p$x[, 1], 0.5)
cols[outliers] &lt;- "red"
plot(x_p, col=cols)
# remove them for Coe, rePCA, replot
x_f %&gt;% slice(-outliers) %&gt;% PCA %&gt;% plot

# or directly with which_out.Coe
# which relies on a PCA
outliers &lt;- x_f %&gt;% which_out(0.5, nax=0.95) %&gt;% na.omit()
x_f %&gt;% slice(-outliers) %&gt;% PCA %&gt;% plot
</code></pre>

<hr>
<h2 id='wings'>Data: Landmarks coordinates of mosquito wings</h2><span id='topic+wings'></span>

<h3>Description</h3>

<p>Data: Landmarks coordinates of mosquito wings
</p>


<h3>Format</h3>

<p>A <a href="#topic+Ldk">Ldk</a> object containing 18 (x; y) landmarks  from 127 mosquito wings, from
</p>


<h3>Source</h3>

<p>Rohlf and Slice 1990.
</p>


<h3>See Also</h3>

<p>Other datasets: 
<code><a href="#topic+apodemus">apodemus</a></code>,
<code><a href="#topic+bot">bot</a></code>,
<code><a href="#topic+chaff">chaff</a></code>,
<code><a href="#topic+charring">charring</a></code>,
<code><a href="#topic+flower">flower</a></code>,
<code><a href="#topic+hearts">hearts</a></code>,
<code><a href="#topic+molars">molars</a></code>,
<code><a href="#topic+mosquito">mosquito</a></code>,
<code><a href="#topic+mouse">mouse</a></code>,
<code><a href="#topic+nsfishes">nsfishes</a></code>,
<code><a href="#topic+oak">oak</a></code>,
<code><a href="#topic+olea">olea</a></code>,
<code><a href="#topic+shapes">shapes</a></code>,
<code><a href="#topic+trilo">trilo</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
