<!DOCTYPE html><html lang="en"><head><title>Help for package MPGE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MPGE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#environment_data'><p>An example of data of the environmental variable (e.g., smoking status).</p>
Here, environment_data is a data frame with single column for the environmental
variable. The order of the 500 individuals in the row must be the same as provided
in the phenotype and genotype data. Here, the environmental variable has two
categories which were coded as 1 and 0 (e.g., smokers and non-smokers).
Instead of numeric values, these can also be considered to be factors in
the absence of a defined order in the categories.</a></li>
<li><a href='#genotype_data'><p>An example of genotype data for two genetic variants (SNPs).</p>
Here, genotype\_data is a data.frame with the columns as SNPs (e.g., rs1 and rs2 here).
The rows correspond to the 500 individuals in the same order as in the phenotype data.</a></li>
<li><a href='#MPGE'><p>MPGE: an R package to implement a two-step approach to testing overall effect of</p>
gene-environment interaction for multiple phenotypes.</a></li>
<li><a href='#mv_G_GE'><p>Test for marginal overall genetic association with multivariate phenotype,</p>
and test for overall GxE effect on the multivariate phenotype
in presence of marginal effect due to the genetic variant
and a marginal effect due to the environmental variable.</a></li>
<li><a href='#mv_G_GxE_pvalues'><p>An example of step 1 (marginal genetic association) and step 2 (GxE interaction)</p>
p-values across genetic variants (SNPs).
Here, mv_G_GxE_pvalues is a data.frame with three columns. First column
lists the set of 1000 genetic variants. Second column provides
the vector of p-values obtained from testing the marginal multivariate
genetic association for these SNPs. And the third column provides the
vector of p-values obtained from testing the overall GxE effect
in presence of possible marginal genetic effect and marginal environmental effect.</a></li>
<li><a href='#phenotype_data'><p>An example of phenotype data.</p></a></li>
<li><a href='#SST'><p>Subset multiple hypothesis testing procedure to combine two steps of testing gene-environment interaction</p>
in a two-step procedure.</a></li>
<li><a href='#WHT'><p>Weighted multiple hypothesis testing procedure to combine two steps of testing gene-environment interaction</p>
in a two-step procedure.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Two-Step Approach to Testing Overall Effect of
Gene-Environment Interaction for Multiple Phenotypes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-10-14</td>
</tr>
<tr>
<td>Description:</td>
<td>Interaction between a genetic variant (e.g., a single nucleotide polymorphism) and an environmental variable (e.g., physical activity) can have a shared effect on multiple phenotypes (e.g., blood lipids). We implement a two-step method to test for an overall interaction effect on multiple phenotypes. In first step, the method tests for an overall marginal genetic association between the genetic variant and the multivariate phenotype. The genetic variants which show an evidence of marginal overall genetic effect in the first step are prioritized while testing for an overall gene-environment interaction effect in the second step. Methodology is available from: A Majumdar, KS Burch, S Sankararaman, B Pasaniuc, WJ Gauderman, JS Witte (2020) &lt;<a href="https://doi.org/10.1101%2F2020.07.06.190256">doi:10.1101/2020.07.06.190256</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ArunabhaCodes/MPGE">https://github.com/ArunabhaCodes/MPGE</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ArunabhaCodes/MPGE/issues">https://github.com/ArunabhaCodes/MPGE/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>car, purrr, stats, utils</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-10-14 20:45:02 UTC; arunmacpro13</td>
</tr>
<tr>
<td>Author:</td>
<td>Arunabha Majumdar [aut, cre],
  Tanushree Haldar [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Arunabha Majumdar &lt;statgen.arunabha@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-10-23 15:40:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='environment_data'>An example of data of the environmental variable (e.g., smoking status).
Here, environment_data is a data frame with single column for the environmental
variable. The order of the 500 individuals in the row must be the same as provided
in the phenotype and genotype data. Here, the environmental variable has two
categories which were coded as 1 and 0 (e.g., smokers and non-smokers).
Instead of numeric values, these can also be considered to be factors in
the absence of a defined order in the categories.</h2><span id='topic+environment_data'></span>

<h3>Description</h3>

<p>An example of data of the environmental variable (e.g., smoking status).
Here, environment_data is a data frame with single column for the environmental
variable. The order of the 500 individuals in the row must be the same as provided
in the phenotype and genotype data. Here, the environmental variable has two
categories which were coded as 1 and 0 (e.g., smokers and non-smokers).
Instead of numeric values, these can also be considered to be factors in
the absence of a defined order in the categories.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(environment_data)
</code></pre>


<h3>Format</h3>

<p>A data.frame with single column for the environmental
variable. The order of the 500 individuals in the row must be the same as provided
in the phenotype and genotype data:
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(environment_data)
geno &lt;- environment_data
</code></pre>

<hr>
<h2 id='genotype_data'>An example of genotype data for two genetic variants (SNPs).
Here, genotype\_data is a data.frame with the columns as SNPs (e.g., rs1 and rs2 here).
The rows correspond to the 500 individuals in the same order as in the phenotype data.</h2><span id='topic+genotype_data'></span>

<h3>Description</h3>

<p>An example of genotype data for two genetic variants (SNPs).
Here, genotype\_data is a data.frame with the columns as SNPs (e.g., rs1 and rs2 here).
The rows correspond to the 500 individuals in the same order as in the phenotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(genotype_data)
</code></pre>


<h3>Format</h3>

<p>A data.frame with the columns as SNPs (e.g., rs1 and rs2 here)
and individuals in the rows with the same order as in the phenotype data:
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genotype_data)
geno &lt;- genotype_data
</code></pre>

<hr>
<h2 id='MPGE'>MPGE: an R package to implement a two-step approach to testing overall effect of
gene-environment interaction for multiple phenotypes.</h2><span id='topic+MPGE'></span>

<h3>Description</h3>

<p>Interaction between a genetic variant (e.g., a SNP) and an environmental variable
(e.g., physical activity) can have a shared effect on multiple phenotypes
(e.g., LDL and HDL). MPGE is a two-step method to test for an overall interaction
effect on multiple phenotypes. In first step, the method tests for an overall
marginal genetic association between the genetic variant and the multivariate
phenotype. In the second step, SNPs which show an evidence of marginal
overall genetic effect in the first step are prioritized while testing
for an overall GxE effect. That is, a more liberal threshold of
significance level is considered in the second step while testing for
an overall GxE effect for these promising SNPs compared to the other
SNPs.
</p>


<h3>Details</h3>

<p>The package consists of following functions:
<code><a href="#topic+mv_G_GE">mv_G_GE</a></code>, <code><a href="#topic+WHT">WHT</a></code>; <code><a href="#topic+SST">SST</a></code>.
</p>


<h3>Functions</h3>


<dl>
<dt><code><a href="#topic+mv_G_GE">mv_G_GE</a></code></dt><dd><p>for a batch of genetic variants, this function provides
two different p-values for each genetic variant, one from the test
of marginal overall genetic association with multiple phenotypes
, and the other from the test of overall GxE effect on multivariate
phenotype allowing for a possible marginal effect due to the genetic
variant and a marginal effect due to the environmental variable.</p>
</dd>
<dt><code><a href="#topic+WHT">WHT</a></code></dt><dd><p>this function implements the weighted multiple hypothesis testing
procedure to adjust for multiple testing while combining the two steps of
testing gene-environment interaction in the two-step GxE testing procedure,
given two sets of p-values obtained using the previous function mv_G_GE for
genome-wide genetic variants.</p>
</dd>
<dt><code><a href="#topic+SST">SST</a></code></dt><dd><p>this function implements
the subset multiple hypothesis testing procedure to adjust for multiple
testing while combining the two steps of testing gene-environment
interaction based on the same two sets of p-values described above.</p>
</dd>
</dl>



<h3>References</h3>

<p>A Majumdar, KS Burch, S Sankararaman, B Pasaniuc, WJ Gauderman, JS Witte (2020)
A two-step approach to testing overall effect of gene-environment interaction for multiple phenotypes.
bioRxiv, doi: https://doi.org/10.1101/2020.07.06.190256
</p>

<hr>
<h2 id='mv_G_GE'>Test for marginal overall genetic association with multivariate phenotype,
and test for overall GxE effect on the multivariate phenotype
in presence of marginal effect due to the genetic variant
and a marginal effect due to the environmental variable.</h2><span id='topic+mv_G_GE'></span>

<h3>Description</h3>

<p>Run <code><a href="#topic+mv_G_GE">mv_G_GE</a></code> to obtain two different sets of p-values, one from the
test for marginal overall genetic association with multiple phenotypes (using multivariate linear regression),
and the other
from the test of overall GxE effect on multivariate phenotype allowing for a possible
genetic effect
due to the genetic variant and an effect due to the environmental variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mv_G_GE(pheno, geno, env)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mv_G_GE_+3A_pheno">pheno</code></td>
<td>
<p>A numeric matrix or data.frame with the number of individuals (n) as the number of rows and the
number of phenotypes (k) as the number of columns. It contains the values of k phenotypes
(e.g. LDL and HDL) across the individuals. Each phenotype (e.g. LDL) must be individually
adjusted for relevant covariates (age, sex, principal components of
genetic ancestries, etc) beforehand. Therefore, each column of pheno matrix should be the adjusted
residuals
of the corresponding phenotype. Each final phenotype (column) should be continuous
and normally distributed. No default.</p>
</td></tr>
<tr><td><code id="mv_G_GE_+3A_geno">geno</code></td>
<td>
<p>A numeric matrix/data.frame (for a batch of genetic variants), or a numeric vector (for a single genetic variant).
It contains the genotype values of the genetic variants/variant across the individuals.
For a batch of variants, columns correspond to variants, and rows correspond to n individuals.
For a SNP, three different ways of genotype coding are possible:
additive, dominant and recessive, where additive coding is more common. No default.</p>
</td></tr>
<tr><td><code id="mv_G_GE_+3A_env">env</code></td>
<td>
<p>A vector of length n (number of individuals). It contains the values of the
environmental variable (e.g., frequency of alcohol consumption). It
can also contain factors, e.g., &quot;yes&quot; or &quot;no&quot; smoking status.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a data.frame with three columns. First column is the name of
the SNPs or genetic variants. The main columns are as follows:
</p>
<table role = "presentation">
<tr><td><code>G.P</code></td>
<td>
<p>P value of testing multivariate marginal genetic association
between the genetic variant
and the vector of phenotypes.</p>
</td></tr>
<tr><td><code>GE.P</code></td>
<td>
<p>P value of testing multivariate overall GxE effect
in presence of possible marginal effect due
to the genetic variant and marginal effect
due to the environmental variable.</p>
</td></tr>
</table>


<h3>References</h3>

<p>A Majumdar, KS Burch, S Sankararaman, B Pasaniuc, WJ Gauderman, JS Witte (2020)
A two-step approach to testing overall effect of gene-environment interaction for multiple phenotypes.
bioRxiv, doi: https://doi.org/10.1101/2020.07.06.190256
</p>


<h3>See Also</h3>

<p><code><a href="#topic+WHT">WHT</a></code>, <code><a href="#topic+SST">SST</a></code>
</p>

<hr>
<h2 id='mv_G_GxE_pvalues'>An example of step 1 (marginal genetic association) and step 2 (GxE interaction)
p-values across genetic variants (SNPs).
Here, mv_G_GxE_pvalues is a data.frame with three columns. First column
lists the set of 1000 genetic variants. Second column provides
the vector of p-values obtained from testing the marginal multivariate
genetic association for these SNPs. And the third column provides the
vector of p-values obtained from testing the overall GxE effect
in presence of possible marginal genetic effect and marginal environmental effect.</h2><span id='topic+mv_G_GxE_pvalues'></span>

<h3>Description</h3>

<p>An example of step 1 (marginal genetic association) and step 2 (GxE interaction)
p-values across genetic variants (SNPs).
Here, mv_G_GxE_pvalues is a data.frame with three columns. First column
lists the set of 1000 genetic variants. Second column provides
the vector of p-values obtained from testing the marginal multivariate
genetic association for these SNPs. And the third column provides the
vector of p-values obtained from testing the overall GxE effect
in presence of possible marginal genetic effect and marginal environmental effect.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mv_G_GxE_pvalues)
</code></pre>


<h3>Format</h3>

<p>A data.frame with three columns. First column
lists the set of 1000 genetic variants. Second column provides
the vector of p-values obtained from testing the marginal multivariate
genetic association for these SNPs. And the third column provides the
vector of p-values obtained from testing the overall GxE effect
in presence of possible marginal genetic effect and marginal environmental effect:
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mv_G_GxE_pvalues)
geno &lt;- mv_G_GxE_pvalues
</code></pre>

<hr>
<h2 id='phenotype_data'>An example of phenotype data.</h2><span id='topic+phenotype_data'></span>

<h3>Description</h3>

<p>Here phenotype\_data is a data.frame with three columns for three phenotypes
and the number of rows to be the number of individuals in the sample (500 in this toy data).
Data for each phenotype provided must be adjusted individually for relevant
covariates (e.g., age, sex, genetic ancestry) beforehand,
and should follow a normal distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(phenotype_data)
</code></pre>


<h3>Format</h3>

<p>A numeric matrix or data.frame with three columns for three phenotypes
and 500 rows for the individuals in the sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(phenotype_data)
pheno &lt;- phenotype_data
</code></pre>

<hr>
<h2 id='SST'>Subset multiple hypothesis testing procedure to combine two steps of testing gene-environment interaction
in a two-step procedure.</h2><span id='topic+SST'></span>

<h3>Description</h3>

<p>Run <code><a href="#topic+SST">SST</a></code> to adjust for multiple testing
while combining two steps of the GxE interaction testing procedure. The procedure is applicable for
a multivariate phenotype, as well as a univariate phenotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SST(PVAL, Pg_thr_step1 = 0.005, FWER_step2 = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SST_+3A_pval">PVAL</code></td>
<td>
<p>A data.frame with three columns.
The first column (PVAL$SNP) provides the name of all SNPs or genetic variants tested.
Second column (PVAL$G.P) contains the p-values of the variants obtained from testing
an overall marginal genetic
association between the multivariate phenotype and each genetic variant individually.
And the third column (PVAL$GE.P) contains the p-values obtained from testing overall GxE effect on the
multivariate phenotype in presence of possible marginal effect due to the genetic variant and
a marginal effect
due to the environmental variable. Number of rows in PVAL is the same as the number
of genetic variants, and it has the same structure as in the output of <code><a href="#topic+mv_G_GE">mv_G_GE</a></code>.
No default.</p>
</td></tr>
<tr><td><code id="SST_+3A_pg_thr_step1">Pg_thr_step1</code></td>
<td>
<p>A positive real number between 0 and 1 providing the p-value threshold
to select the set of
promising SNPs in step 1. These selected SNPs will be tested for GxE effect in the second step.
Default is 0.005.</p>
</td></tr>
<tr><td><code id="SST_+3A_fwer_step2">FWER_step2</code></td>
<td>
<p>A positive real number between 0 and 1 specifying the family-wise error rate
to be maintained in the second
step while identifying the genetic variants having a genome-wide significant overall GxE effect
on the multivariate phenotype. Default is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a vector of SNPs identified to have a genome-wide significant overall GxE effect.
</p>


<h3>References</h3>

<p>A Majumdar, KS Burch, S Sankararaman, B Pasaniuc, WJ Gauderman, JS Witte (2020)
A two-step approach to testing overall effect of gene-environment interaction for multiple phenotypes.
bioRxiv, doi: https://doi.org/10.1101/2020.07.06.190256
</p>


<h3>See Also</h3>

<p><code><a href="#topic+WHT">WHT</a></code>, <code><a href="#topic+mv_G_GE">mv_G_GE</a></code>
</p>

<hr>
<h2 id='WHT'>Weighted multiple hypothesis testing procedure to combine two steps of testing gene-environment interaction
in a two-step procedure.</h2><span id='topic+WHT'></span>

<h3>Description</h3>

<p>Run <code><a href="#topic+WHT">WHT</a></code> to adjust for multiple testing
while combining two steps of the GxE interaction testing procedure. The procedure is applicable for
a multivariate phenotype, as well as a univariate phenotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WHT(PVAL, first_bin_size = 5, FWER = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="WHT_+3A_pval">PVAL</code></td>
<td>
<p>A data.frame with three columns.
The first column (PVAL$SNP) provides the name of all SNPs or genetic variants tested.
Second column (PVAL$G.P) contains the p-values of the variants obtained from testing
an overall marginal genetic
association between the multivariate phenotype and each genetic variant individually.
And the third column (PVAL$GE.P) contains the p-values obtained from testing overall GxE effect on the
multivariate phenotype in presence of possible marginal effect due to the genetic variant and
a marginal effect
due to the environmental variable. Number of rows in PVAL is the same as the number
of genetic variants, and it has the same structure as in the output of <code><a href="#topic+mv_G_GE">mv_G_GE</a></code>.
No default.</p>
</td></tr>
<tr><td><code id="WHT_+3A_first_bin_size">first_bin_size</code></td>
<td>
<p>A positive integer providing the number of SNPs in the top bin
while ranking the SNPs or genetic variants according to their
relative importance in the first step, which is evaluated with respect to the strength of
overall marginal genetic association with
the multivariate phenotype. Default is 5.</p>
</td></tr>
<tr><td><code id="WHT_+3A_fwer">FWER</code></td>
<td>
<p>A positive real number between 0 and 1 providing the overall
family wise error rate to be maintained while
identifying the genetic variants having a genome-wide significant overall GxE effect
on the multivariate phenotype.
Default is 0.05.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output produced by the function is a list consisting of:
</p>
<table role = "presentation">
<tr><td><code>GEsnps</code></td>
<td>
<p>Vector of SNPs/genetic variants identified to have a genome-wide significant overall GxE effect.</p>
</td></tr>
<tr><td><code>adjusted.PV</code></td>
<td>
<p>A data.frame providing the adjusted p-values with the corresponding
genetic variants obtained by the weighted multiple hypothesis testing procedure.</p>
</td></tr>
</table>


<h3>References</h3>

<p>A Majumdar, KS Burch, S Sankararaman, B Pasaniuc, WJ Gauderman, JS Witte (2020)
A two-step approach to testing overall effect of gene-environment interaction for multiple phenotypes.
bioRxiv, doi: https://doi.org/10.1101/2020.07.06.190256
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SST">SST</a></code>, <code><a href="#topic+mv_G_GE">mv_G_GE</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
