<!DOCTYPE html><html><head><title>Help for package mpmi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mpmi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cmi'><p> Calculate BCMI between a set of continuous variables</p></a></li>
<li><a href='#cts'><p>A group of simulated continuous variables</p></a></li>
<li><a href='#disc'><p>A group of simulated categorical (discrete) variables</p></a></li>
<li><a href='#dmi'><p> Calculate BCMI for categorical (discrete) data</p></a></li>
<li><a href='#mmi'><p> Calculate mixed-pair BCMI between a set of continuous variables and a set</p>
of discrete variables.</a></li>
<li><a href='#mp'><p>Matrix Plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Mixed-Pair Mutual Information Estimators</td>
</tr>
<tr>
<td>Version:</td>
<td>0.43.2.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-04-22</td>
</tr>
<tr>
<td>Author:</td>
<td>Chris Pardy</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chris Pardy &lt;cooliomcdude@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Uses a kernel smoothing approach to calculate Mutual Information
    for comparisons between all types of variables including continuous vs
    continuous, continuous vs discrete and discrete vs discrete. Uses a
    nonparametric bias correction giving Bias Corrected Mutual Information
    (BCMI). Implemented efficiently in Fortran 95 with OpenMP and suited to
    large genomic datasets.  </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://r-forge.r-project.org/projects/mpmi/">https://r-forge.r-project.org/projects/mpmi/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.6.2), KernSmooth</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-07-05 09:30:40 UTC; ripley</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-07-05 09:37:38 UTC</td>
</tr>
</table>
<hr>
<h2 id='cmi'> Calculate BCMI between a set of continuous variables
</h2><span id='topic+cmi'></span><span id='topic+cminjk'></span><span id='topic+cmi.pw'></span><span id='topic+cminjk.pw'></span>

<h3>Description</h3>

<p>This function calculates MI and BCMI between a set of continuous variables
held as columns in a matrix. It also performs jackknife bias correction and
provides a z-score for the hypothesis of no association. Also included are
the *.pw functions that calculate MI between two vectors only. The *njk
functions do not perform the jackknife and are therefore faster. 

</p>


<h3>Usage</h3>

<pre><code class='language-R'>cmi(cts, level = 3L, na.rm = FALSE, h, ...)
cminjk(cts, level = 3L, na.rm = FALSE, h, ...)
cmi.pw(v1, v2, h, ...)
cminjk.pw(v1, v2, h, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cmi_+3A_cts">cts</code></td>
<td>
<p> The data matrix. Each row is an observation and each column is a
variable of interest. Should be numerical data.
</p>
</td></tr>
<tr><td><code id="cmi_+3A_level">level</code></td>
<td>
<p> The number of levels used for plug-in bandwidth estimation (see
the documentation for the KernSmooth package.)
</p>
</td></tr>
<tr><td><code id="cmi_+3A_na.rm">na.rm</code></td>
<td>
<p> Remove missing values if TRUE. This is required for the
bandwidth calculation.
</p>
</td></tr>
<tr><td><code id="cmi_+3A_h">h</code></td>
<td>
<p> A (double) vector of smoothing bandwidths, one for each variable. If
missing this will be calculated using the dpik() function from the
KernSmooth package.
</p>
</td></tr>
<tr><td><code id="cmi_+3A_...">...</code></td>
<td>
<p> Additional options passed to dpik() if necessary.
</p>
</td></tr> 
<tr><td><code id="cmi_+3A_v1">v1</code></td>
<td>
<p> A vector for the pairwise version</p>
</td></tr>
<tr><td><code id="cmi_+3A_v2">v2</code></td>
<td>
<p> A vector for the pairwise version</p>
</td></tr>
</table>


<h3>Details</h3>

<p> The results of cmi() are in many ways similar to a correlation matrix,
with each row and column index corresponding to a given variable.
cminjk() and cminjk.pw() just returns the MI values without performing the
jackknife. cmi.pw() and cminjk.pw() each only require two bandwidths, one
for each variable. The number of processor cores used can be changed by
setting the environment variable &quot;OMP_NUM_THREADS&quot; <em>before</em> starting R.
</p>


<h3>Value</h3>

<p> Returns a list of 3 matrices each of size ncol(cts) by
ncol(cts)
</p>
<table>
<tr><td><code>mi</code></td>
<td>
<p> The raw MI estimates.</p>
</td></tr>
<tr><td><code>bcmi</code></td>
<td>
<p>Jackknife bias corrected MI estimates (BCMI). These are each MI value
minus the corresponding jackknife estimate of bias.</p>
</td></tr>
<tr><td><code>zvalues</code></td>
<td>
<p>Z-scores for each hypothesis that the corresponding 
BCMI value is zero. These have poor statistical properties but can be useful
as a rough measure of the strength of association.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>##################################################
# The USArrests dataset

# Matrix version
c1 &lt;- cmi(USArrests)
lapply(c1, round, 2)

# Pairwise version
cmi.pw(USArrests[,1], USArrests[,2])

# Without jackknife
c2 &lt;- cminjk(USArrests)
round(c2, 2)
cminjk.pw(USArrests[,1], USArrests[,2])

##################################################
# A look at Anscombe's famous dataset.
par(mfrow = c(2,2))
plot(anscombe$x1, anscombe$y1)
plot(anscombe$x2, anscombe$y2)
plot(anscombe$x3, anscombe$y3)
plot(anscombe$x4, anscombe$y4)

cor(anscombe$x1, anscombe$y1)
cor(anscombe$x2, anscombe$y2)
cor(anscombe$x3, anscombe$y3)
cor(anscombe$x4, anscombe$y4)

cmi.pw(anscombe$x1, anscombe$y1)
cmi.pw(anscombe$x2, anscombe$y2)
cmi.pw(anscombe$x3, anscombe$y3)
# dpik() has some trouble with zero scale estimates on this one:
cmi.pw(anscombe$x4, anscombe$y4, scalest = "stdev")
##################################################

##################################################
# The highly collinear Longley dataset

pairs(longley, main = "longley data")
l1 &lt;- cmi(longley)
lapply(l1, round, 2)

# Here we demonstrate the scale-invariance of MI.
# Note: Scaling can help stabilise estimates when there are
# difficulties with the bandwidth estimation, but is unnecessary
# here.
long2 &lt;- scale(longley)
l2 &lt;- cmi(long2)
lapply(l2, round, 2)

##################################################
# See the vignette for large-scale examples.
</code></pre>

<hr>
<h2 id='cts'>A group of simulated continuous variables</h2><span id='topic+cts'></span><span id='topic+cts'></span>

<h3>Description</h3>

<p>50 observations on each of 100 variables. The data are simulated such that
variables with similar indices are associated with the degree of association 
decaying as variables are further apart (i.e., a correlation or information
matrix with have larger values near the diagonal). Details are given in the vignette.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cts</code></pre>


<h3>Format</h3>

<p>A matrix with 50 rows and 100 columns</p>

<hr>
<h2 id='disc'>A group of simulated categorical (discrete) variables</h2><span id='topic+disc'></span>

<h3>Description</h3>

<p>50 observations on each of 75 categorical variables. These variables are 
designed to be similar to categorical single nucleotide polymorphism (SNP) data
which have 3 categories (A, H and B where H represents a heterozygous mutation). 
There are no associations between any of the variables. The variables are stored 
as characters. See the vignette for details
</p>


<h3>Usage</h3>

<pre><code class='language-R'>disc</code></pre>


<h3>Format</h3>

<p>A matrix of characters with 50 rows and 75 columns</p>

<hr>
<h2 id='dmi'> Calculate BCMI for categorical (discrete) data
</h2><span id='topic+dmi'></span><span id='topic+dminjk'></span><span id='topic+dmi.pw'></span><span id='topic+dminjk.pw'></span>

<h3>Description</h3>

<p> This function calculates MI and BCMI between a set of discrete variables
held as columns in a matrix. It also performs jackknife bias correction and
provides a z-score for the hypothesis of no association. Also included are
the *.pw functions that calculate MI between two vectors only. The *njk
functions do not perform the jackknife and are therefore faster. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dmi(dmat)
dminjk(dmat)
dmi.pw(disc1, disc2)
dminjk.pw(disc1, disc2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dmi_+3A_dmat">dmat</code></td>
<td>
<p> The data matrix. Each row is an observation and each column is a
variable of interest. Should contain categorical data, all types of data
will be coerced via factors to integers.
</p>
</td></tr>
<tr><td><code id="dmi_+3A_disc1">disc1</code></td>
<td>
<p> A vector for the pairwise version</p>
</td></tr>
<tr><td><code id="dmi_+3A_disc2">disc2</code></td>
<td>
<p> A vector for the pairwise version</p>
</td></tr>
</table>


<h3>Details</h3>

<p> The results of dmi() are in many ways similar to a correlation matrix,
with each row and column index corresponding to a given variable.
dminjk() and dminjk.pw() just returns the MI values without performing the
jackknife. The number of processor cores used can be changed by
setting the environment variable &quot;OMP_NUM_THREADS&quot; <em>before</em> starting R.
</p>


<h3>Value</h3>

<p> Returns a list of 3 matrices each of size <code>ncol(dmat)</code> by
<code>ncol(dmat)</code>
</p>
<table>
<tr><td><code>mi</code></td>
<td>
<p> The raw MI estimates.</p>
</td></tr>
<tr><td><code>bcmi</code></td>
<td>
<p> Jackknife bias corrected MI estimates (BCMI). These are each MI value
minus the corresponding jackknife estimate of bias.</p>
</td></tr>
<tr><td><code>zvalues</code></td>
<td>
<p> Z-scores for each hypothesis that the corresponding 
bcmi value is zero. These have poor statistical properties but can be useful
as a rough measure of the strength of association.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(cars)

# Discretise the data first
d &lt;- cut(cars$dist, breaks = 10)
s &lt;- cut(cars$speed, breaks = 10)

# Discrete MI values
dmi.pw(s, d)

# For comparison, analysed as continuous data:
cmi.pw(cars$dist, cars$speed)

# Exploring a group of categorical variables
dat &lt;- mtcars[, c("cyl","vs","am","gear","carb")]
discresults &lt;- dmi(dat)
discresults

# Plot the relative magnitude of the BCMI values
diag(discresults$bcmi) &lt;- NA
mp(discresults$bcmi)
</code></pre>

<hr>
<h2 id='mmi'> Calculate mixed-pair BCMI between a set of continuous variables and a set
of discrete variables.
</h2><span id='topic+mmi'></span><span id='topic+mminjk'></span><span id='topic+mmi.pw'></span><span id='topic+mminjk.pw'></span>

<h3>Description</h3>

<p>This function calculates MI and BCMI between a set of continuous variables
and a set of discrete variables (variables in columns). It also performs
jackknife bias correction and provides a z-score for the hypothesis of no
association. Also included are the *.pw functions that calculate MI between
two vectors only. The *njk functions do not perform the jackknife and are
therefore faster. 

</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmi(cts, disc, level = 3L, na.rm = FALSE, h, ...)
mminjk(cts, disc, level = 3L, na.rm = FALSE, h, ...)
mmi.pw(cts, disc, h, ...) 
mminjk.pw(cts, disc, h, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mmi_+3A_cts">cts</code></td>
<td>
<p> The data matrix. Each row is an observation and each column is a
variable of interest. Should be numerical data. (For the pairwise functions this
should be a vector.)
</p>
</td></tr>
<tr><td><code id="mmi_+3A_disc">disc</code></td>
<td>
<p> Matrix of discrete data, each row is an observation and each
column is a variable. Will be coerced to integers. (For the pairwise functions this
should be a vector.)</p>
</td></tr>
<tr><td><code id="mmi_+3A_level">level</code></td>
<td>
<p> The number of levels used for plug-in bandwidth estimation (see
the documentation for the KernSmooth package.)
</p>
</td></tr>
<tr><td><code id="mmi_+3A_na.rm">na.rm</code></td>
<td>
<p> Remove missing values if TRUE. This is required for the
bandwidth calculation.
</p>
</td></tr>
<tr><td><code id="mmi_+3A_h">h</code></td>
<td>
<p> A (double) vector of smoothing bandwidths, one for each variable. If
missing this will be calculated using the dpik() function from the
KernSmooth package.
</p>
</td></tr>
<tr><td><code id="mmi_+3A_...">...</code></td>
<td>
<p> Additional options passed to dpik() if necessary.
</p>
</td></tr> 
</table>


<h3>Details</h3>

<p>mminjk() and mminjk.pw() return just the MI values without performing the
jackknife. mmi.pw() and mminjk.pw() only require one bandwidth for the
continuous variable. The number of processor cores used can be changed by
setting the environment variable &quot;OMP_NUM_THREADS&quot; <em>before</em> starting R.
</p>


<h3>Value</h3>

<p> Returns a list of 3 matrices each of size ncol(cts) by
ncol(disc). Each row index represents a continuous variable and each column
index a discrete variable. 
</p>
<table>
<tr><td><code>mi</code></td>
<td>
<p>The raw MI estimates.</p>
</td></tr>
<tr><td><code>bcmi</code></td>
<td>
<p>Jackknife bias corrected MI estimates (BCMI). These are each MI value
minus the corresponding jackknife estimate of bias.</p>
</td></tr>
<tr><td><code>zvalues</code></td>
<td>
<p>z-scores for each hypothesis that the corresponding 
bcmi value is zero. These have poor statistical properties but can be useful
as a rough measure of the strength of association.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
##################################################
# A dataset with discrete and continuous variables

cts &lt;- state.x77
disc &lt;- data.frame(state.division,state.region)
summary(cts)
table(disc)
m1 &lt;- mmi(cts, disc)
lapply(m1, round, 2)
# Division gives more information about the continuous variables than region.

# Here is one where both division and region show a strong association:
boxplot(cts[,6] ~ disc[,1])
boxplot(cts[,6] ~ disc[,2])

# In this case the states need to be divided into regions before a clear
# association can be seen:
boxplot(cts[,1] ~ disc[,1])
boxplot(cts[,1] ~ disc[,2])

# Look at associations within the continuous variables:
pairs(cts, col = state.region)
c1 &lt;- cmi(cts)
lapply(c1, round, 2)

##################################################
# A pairwise comparison

# Note that the ANOVA homoskedasticity assumption is not satisfied here.
boxplot(InsectSprays[,1] ~ InsectSprays[,2])
mmi.pw(InsectSprays[,1], InsectSprays[,2])

##################################################
# Another pairwise comparison

boxplot(morley[,3] ~ morley[,1])
m2 &lt;- mmi.pw(morley[,3], morley[,1])
m2

##################################################
# See the vignette for large-scale examples.
</code></pre>

<hr>
<h2 id='mp'>Matrix Plot</h2><span id='topic+mp'></span>

<h3>Description</h3>

<p>Plot a matrix of values in the same order that it is stored 
(the usual mathematical way).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mp(mat, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mp_+3A_mat">mat</code></td>
<td>
<p>A numeric matrix to be plotted</p>
</td></tr>
<tr><td><code id="mp_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass to <code>image()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>mp()</code> function is a simple wrapper to
<code>image()</code> with a few minor changes. The plot is
flipped so that points correspond to their position in the
matrix. Also, the colours are scaled so that red is the
maximum value in the matrix and white is the minimum value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># From the vignette:
data(mpmidata)
ctsresult &lt;- cmi(cts)
mp(ctsresult$bcmi)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
