<!DOCTYPE html><html><head><title>Help for package mr.raps</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mr.raps}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bmi.bmi'><p>&quot;Effect&quot; of Body Mass Index (BMI) on Body Mass Index (BMI)</p></a></li>
<li><a href='#bmi.sbp'><p>Effect of Body Mass Index (BMI) on Systolic Blood Pressure (SBP)</p></a></li>
<li><a href='#mr.raps'><p>Main function</p></a></li>
<li><a href='#mr.raps-package'><p>Two Sample Mendelian Randomization using Robust Adjusted Profile Score</p></a></li>
<li><a href='#rho.huber'><p>Huber loss function and its derivatives</p></a></li>
<li><a href='#rho.tukey'><p>Tukey's beweight loss function and its derivatives</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Two Sample Mendelian Randomization using Robust Adjusted Profile
Score</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-01-29</td>
</tr>
<tr>
<td>Description:</td>
<td>Mendelian randomization is a method of identifying and estimating a confounded causal effect using genetic instrumental variables. This packages implements methods for two-sample Mendelian randomization with summary statistics by using Robust Adjusted Profile Score (RAPS). References: Qingyuan Zhao, Jingshu Wang, Jack Bowden, Dylan S. Small. Statistical inference in two-sample summary-data Mendelian randomization using robust adjusted profile score. &lt;<a href="https://arxiv.org/abs/1801.09652">arXiv:1801.09652</a>&gt;.</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, graphics, nortest</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-01-30 04:47:53 UTC; qyzhao</td>
</tr>
<tr>
<td>Author:</td>
<td>Qingyuan Zhao [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Qingyuan Zhao &lt;qingyzhao@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-01-30 12:48:29 UTC</td>
</tr>
</table>
<hr>
<h2 id='bmi.bmi'>&quot;Effect&quot; of Body Mass Index (BMI) on Body Mass Index (BMI)</h2><span id='topic+bmi.bmi'></span>

<h3>Description</h3>

<p>Summary data obtained by combining three genome-wide association studies:
</p>

<ol>
<li><p>BMI-GIANT: BMI in the Genetic Investigation of ANthropometric Traits (GIANT) consortium (sample size: 339224).
</p>
</li>
<li><p>BMI-UKBB-1: BMI in a half of the United Kingdom BioBank (UKBB) data (sample size: 234070)
</p>
</li>
<li><p>SBP-UKBB-2: BMI in the other half of the UKBB data (sample size: 234070)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>data(bmi.bmi)
</code></pre>


<h3>Format</h3>

<p>A data.frame.</p>


<h3>Details</h3>

<p>The BMI-GIANT dataset is used for SNP selection (column <code>pval.selection</code>). The BMI-UKBB-1 dataset estimates the SNPs' effects on BMI (columns <code>beta.exposure</code> and <code>se.exposure</code>) and the BMI-UKBB-2 dataset provides independent estimates of the same effects (columns <code>beta.outcome</code> and <code>se.outcome</code>).
</p>

<hr>
<h2 id='bmi.sbp'>Effect of Body Mass Index (BMI) on Systolic Blood Pressure (SBP)</h2><span id='topic+bmi.sbp'></span>

<h3>Description</h3>

<p>Summary data obtained by combining three genome-wide association studies:
</p>

<ol>
<li><p>BMI-FEM: BMI in females by the Genetic Investigation of ANthropometric Traits (GIANT) consortium (sample size: 171977).
</p>
</li>
<li><p>BMI-MAL: BMI in males in the same study by the GIANT consortium (sam- ple size: 152893)
</p>
</li>
<li><p>SBP-UKBB: SBP using the United Kingdom BioBank (UKBB) data (sample size: 317754)
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>data(bmi.sbp)
</code></pre>


<h3>Format</h3>

<p>A data.frame.</p>


<h3>Details</h3>

<p>The BMI-FEM dataset is used for SNP selection (column <code>pval.selection</code>). The BMI-MAL dataset estimates the SNPs' effect on BMI and the SBP-UKBB dataset estimates the SNPs' on SBP.
</p>

<hr>
<h2 id='mr.raps'>Main function</h2><span id='topic+mr.raps'></span><span id='topic+mr.raps.all'></span><span id='topic+mr.raps.simple'></span><span id='topic+mr.raps.overdispersed'></span><span id='topic+mr.raps.simple.robust'></span><span id='topic+mr.raps.overdispersed.robust'></span>

<h3>Description</h3>

<p><code>mr.raps</code> is the main function.
</p>
<p><code>mr.raps.all</code>: Quick analysis with all six methods
</p>
<p><code>mr.raps.simple</code>: No overdispersion, l2 loss
</p>
<p><code>mr.raps.overdispersed</code>: Overdispersion, l2 loss
</p>
<p><code>mr.raps.simple.robust</code>: No overdispersion, robust loss
</p>
<p><code>mr.raps.overdispersed.robust</code>: Overdispersed, robust loss
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr.raps(b_exp, b_out, se_exp, se_out, over.dispersion = FALSE,
  loss.function = c("l2", "huber", "tukey"), diagnosis = FALSE,
  se.method = c("sandwich", "bootstrap"), k = switch(loss.function[1], l2 =
  NULL, huber = 1.345, tukey = 4.685), B = 1000, suppress.warning = FALSE)

mr.raps.all(b_exp, b_out, se_exp, se_out)

mr.raps.simple(b_exp, b_out, se_exp, se_out, diagnosis = FALSE)

mr.raps.overdispersed(b_exp, b_out, se_exp, se_out,
  initialization = c("simple", "mode"), suppress.warning = FALSE,
  diagnosis = FALSE, niter = 20, tol = .Machine$double.eps^0.5)

mr.raps.simple.robust(b_exp, b_out, se_exp, se_out, loss.function = c("huber",
  "tukey"), k = switch(loss.function[1], huber = 1.345, tukey = 4.685),
  diagnosis = FALSE)

mr.raps.overdispersed.robust(b_exp, b_out, se_exp, se_out,
  loss.function = c("huber", "tukey"), k = switch(loss.function[1], huber =
  1.345, tukey = 4.685), initialization = c("l2", "mode"),
  suppress.warning = FALSE, diagnosis = FALSE, niter = 20,
  tol = .Machine$double.eps^0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr.raps_+3A_b_exp">b_exp</code></td>
<td>
<p>A vector of SNP effects on the exposure variable, usually obtained from a GWAS.</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_b_out">b_out</code></td>
<td>
<p>A vector of SNP effects on the outcome variable, usually obtained from a GWAS.</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_se_exp">se_exp</code></td>
<td>
<p>A vector of standard errors of <code>b_exp</code>.</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_se_out">se_out</code></td>
<td>
<p>A vector of standard errors of <code>b_out</code>.</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_over.dispersion">over.dispersion</code></td>
<td>
<p>Should the model consider overdispersion (systematic pleiotropy)? Default is FALSE.</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_loss.function">loss.function</code></td>
<td>
<p>Either the squared error loss (<code>l2</code>) or robust loss functions/scores (<code>huber</code> or <code>tukey</code>).</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_diagnosis">diagnosis</code></td>
<td>
<p>Should the function returns diagnostic plots and results? Default is FALSE</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_se.method">se.method</code></td>
<td>
<p>How should the standard error be estimated? Either by sandwich variance formula (default and recommended) or the bootstrap.</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_k">k</code></td>
<td>
<p>Threshold parameter in the Huber and Tukey loss functions.</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_b">B</code></td>
<td>
<p>Number of bootstrap resamples</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_suppress.warning">suppress.warning</code></td>
<td>
<p>Should warning messages be suppressed?</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_initialization">initialization</code></td>
<td>
<p>Method to initialize the robust estimator. &quot;Mode&quot; is not supported currently.</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_niter">niter</code></td>
<td>
<p>Maximum number of interations to solve the estimating equations.</p>
</td></tr>
<tr><td><code id="mr.raps_+3A_tol">tol</code></td>
<td>
<p>Numerical precision.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>

<dl>
<dt>beta.hat</dt><dd><p>Estimated causal effect</p>
</dd>
<dt>beta.se</dt><dd><p>Standard error of <code>beta.hat</code></p>
</dd>
<dt>beta.p.value</dt><dd><p>Two-sided p-value of <code>beta.hat</code></p>
</dd>
<dt>tau2.hat</dt><dd><p>Overdispersion parameter if <code>over.dispersion = TRUE</code></p>
</dd>
<dt>tau2.se</dt><dd><p>Standard error of <code>tau2.hat</code></p>
</dd>
<dt>std.resid</dt><dd><p>Standardized residuals of each SNP, returned if <code>diagnosis = TRUE</code></p>
</dd>
<dt>beta.hat.loo</dt><dd><p>Leave-one-out estimates of <code>beta.hat</code>, returned if <code>diagnosis = TRUE</code></p>
</dd>
<dt>beta.hat.bootstrap</dt><dd><p>Median of the bootstrap estimates, returned if <code>se.method = "bootstrap"</code></p>
</dd>
<dt>beta.se.bootstrap</dt><dd><p>Median absolute deviation of the bootstrap estimates, returned if <code>se.method = "bootstrap"</code></p>
</dd>
</dl>



<h3>Functions</h3>


<ul>
<li> <p><code>mr.raps.all</code>: 
</p>
</li>
<li> <p><code>mr.raps.simple</code>: 
</p>
</li>
<li> <p><code>mr.raps.overdispersed</code>: 
</p>
</li>
<li> <p><code>mr.raps.simple.robust</code>: 
</p>
</li>
<li> <p><code>mr.raps.overdispersed.robust</code>: 
</p>
</li></ul>


<h3>References</h3>

<p>Qingyuan Zhao, Jingshu Wang, Jack Bowden, Dylan S. Small. Statistical inference in two-sample summary-data Mendelian randomization using robust adjusted profile score. <a href="https://arxiv.org/abs/1801.09652">https://arxiv.org/abs/1801.09652</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(bmi.sbp)
attach(bmi.sbp)

## All estimators
mr.raps.all(beta.exposure, beta.outcome, se.exposure, se.outcome)

## Diagnostic plots
res &lt;- mr.raps(beta.exposure, beta.outcome, se.exposure, se.outcome,
diagnosis = TRUE)
res &lt;- mr.raps(beta.exposure, beta.outcome, se.exposure, se.outcome,
TRUE, diagnosis = TRUE)
res &lt;- mr.raps(beta.exposure, beta.outcome, se.exposure, se.outcome,
TRUE, "tukey", diagnosis = TRUE)

detach(bmi.sbp)

data(bmi.bmi)
attach(bmi.bmi)

## Because both the exposure and the outcome are BMI, the true "causal" effect should be 1.

## All estimators
mr.raps.all(beta.exposure, beta.outcome, se.exposure, se.outcome)

detach(bmi.bmi)

</code></pre>

<hr>
<h2 id='mr.raps-package'>Two Sample Mendelian Randomization using Robust Adjusted Profile Score</h2><span id='topic+mr.raps-package'></span>

<h3>Description</h3>

<p>Mendelian randomization is a method of identifying and estimating a confounded causal effect using genetic instrumental variables. This packages implements methods for two sasmple Mendelian randomization with summary statistics by using Robust Adjusted Profile Score (RAPS).
</p>

<hr>
<h2 id='rho.huber'>Huber loss function and its derivatives</h2><span id='topic+rho.huber'></span>

<h3>Description</h3>

<p>Huber loss function and its derivatives
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rho.huber(r, k = 1.345, deriv = 0)
</code></pre>

<hr>
<h2 id='rho.tukey'>Tukey's beweight loss function and its derivatives</h2><span id='topic+rho.tukey'></span>

<h3>Description</h3>

<p>Tukey's beweight loss function and its derivatives
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rho.tukey(r, k = 4.685, deriv = 0)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
