<!DOCTYPE html><html><head><title>Help for package mrbayes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mrbayes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bmi_insulin'><p>Dataset from Richmond et. al 2017 investigating the association of BMI on insulin resistance</p></a></li>
<li><a href='#dodata'><p>Dataset from Do et al., Nat Gen, 2013 containing summary level data on associations of genotypes with lipid traits and the risk of coronary heart diseases</p></a></li>
<li><a href='#mr_egger_rjags'><p>Bayesian implementation of the MR-Egger multivariate model with choice of prior distributions fitted using JAGS.</p></a></li>
<li><a href='#mr_egger_stan'><p>Bayesian inverse variance weighted model with a choice of prior distributions fitted using Stan</p></a></li>
<li><a href='#mr_format'><p>Organises the summary level data for use in the Bayesian MR functions</p></a></li>
<li><a href='#mr_ivw_rjags'><p>Bayesian inverse variance weighted model with a choice of prior distributions fitted using JAGS.</p></a></li>
<li><a href='#mr_ivw_stan'><p>Bayesian inverse variance weighted model with a choice of prior distributions fitted using RStan.</p></a></li>
<li><a href='#mr_radialegger_rjags'><p>Bayesian radial MR-Egger model with a choice of prior distributions fitted using JAGS.</p></a></li>
<li><a href='#mr_radialegger_stan'><p>Bayesian inverse variance weighted model with a choice of prior distributions fitted using RStan.</p></a></li>
<li><a href='#mrbayes-package'><p>mrbayes: Bayesian implementation of the IVW and MR-Egger models for two-sample Mendelian randomization analyses</p></a></li>
<li><a href='#mrinput_mr_format'><p>Convert an object of class MRInput from the MendelianRandomization package to the mrbayes mr_format class</p></a></li>
<li><a href='#mvmr_egger_rjags'><p>Bayesian implementation of the MVMR-Egger model with choice of prior distributions fitted using JAGS.</p></a></li>
<li><a href='#mvmr_egger_stan'><p>Bayesian implementation of the MVMR-Egger model with choice of prior distributions fitted using RStan.</p></a></li>
<li><a href='#mvmr_format'><p>Organises the summary level data for use in the Bayesian MR functions</p></a></li>
<li><a href='#mvmr_ivw_rjags'><p>Bayesian multivariate inverse variance weighted model with a choice of prior distributions fitted using JAGS.</p></a></li>
<li><a href='#mvmr_ivw_stan'><p>Bayesian multivariate inverse variance weighted model with a choice of prior distributions fitted using RStan.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Summary Data Models for Mendelian Randomization Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Bayesian estimation of inverse variance weighted (IVW), Burgess 
    et al. (2013) &lt;<a href="https://doi.org/10.1002%2Fgepi.21758">doi:10.1002/gepi.21758</a>&gt;, and MR-Egger, Bowden et 
    al. (2015) &lt;<a href="https://doi.org/10.1093%2Fije%2Fdyv080">doi:10.1093/ije/dyv080</a>&gt;, summary data models for Mendelian 
    randomization analyses.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, Rcpp (&ge; 0.12.0), rstan (&ge; 2.18.1), rstantools (&ge;
2.0.0), plyr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, rjags (&ge; 4-9), tidyverse, knitr, ggplot2, ggmcmc,
foreign, DescTools, bayesmeta, utils, kableExtra, roxygen2,
bookdown, MendelianRandomization</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/okezie94/mrbayes">https://github.com/okezie94/mrbayes</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/okezie94/mrbayes/issues">https://github.com/okezie94/mrbayes/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>Biarch:</td>
<td>true</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>BH (&ge; 1.66.0), Rcpp (&ge; 0.12.0), RcppEigen (&ge; 0.3.3.3.0),
rstan (&ge; 2.18.1), StanHeaders (&ge; 2.18.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-10-02 12:16:01 UTC; okezieuche-ikonne</td>
</tr>
<tr>
<td>Author:</td>
<td>Okezie Uche-Ikonne
    <a href="https://orcid.org/0000-0002-8461-8099"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Frank Dondelinger <a href="https://orcid.org/0000-0003-1816-6300"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Tom Palmer <a href="https://orcid.org/0000-0003-4655-4511"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Okezie Uche-Ikonne &lt;okezieikonne@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-10-02 14:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bmi_insulin'>Dataset from Richmond et. al 2017 investigating the association of BMI on insulin resistance</h2><span id='topic+bmi_insulin'></span>

<h3>Description</h3>

<p>A two-sample summary level dataset, Richmond et al. (2017) doi: <a href="https://doi.org/10.1101/155739">10.1101/155739</a>, containing 14 single nucleiodtide polymorphisms (SNPs) which have genotype-phenotype associations (BMI) and genotype-outcome associations (insulin) with their respective standard errors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bmi_insulin
</code></pre>


<h3>Format</h3>

<p>A data frame with 14 rows and 44 columns:
</p>

<dl>
<dt>rsid</dt><dd><p>SNP RSID number</p>
</dd>
<dt>beta.exposure</dt><dd><p>The genotype-BMI associations</p>
</dd>
<dt>beta.outcome</dt><dd><p>The genotype-outcome associations</p>
</dd>
<dt>se.exposure</dt><dd><p>The standard errors of the genotype-phenotype associations</p>
</dd>
<dt>se.outcome</dt><dd><p>The standard errors of the genotype-outcome associations</p>
</dd>
</dl>



<h3>Details</h3>

<p>bmi_insulin.
</p>


<h3>References</h3>

<p>Richmond, R. et al., Investigating the role of insulin in increased adiposity: Bi-directional Mendelian randomization study. bioRxiv, 2017, doi: <a href="https://doi.org/10.1101/155739">10.1101/155739</a>.
</p>

<hr>
<h2 id='dodata'>Dataset from Do et al., Nat Gen, 2013 containing summary level data on associations of genotypes with lipid traits and the risk of coronary heart diseases</h2><span id='topic+dodata'></span>

<h3>Description</h3>

<p>A summary-level dataset, from Do et al. (2013) doi: <a href="https://doi.org/10.1038/ng.2795">10.1038/ng.2795</a>,
containing 185 single nucleodtide polymorphisms (SNPs) which have genotype-phenotype associations and standard errors for
LDL-C, HDL-C, Triglycerides, and genotype-outcome associations for coronary heart disease
with their respective standard errors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dodata
</code></pre>


<h3>Format</h3>

<p>A data frame with 185 rows and 21 columns with the following variables:
</p>

<dl>
<dt>rsid</dt><dd><p>RSID number</p>
</dd>
<dt>a1</dt><dd><p>Allele 1</p>
</dd>
<dt>a2</dt><dd><p>Allele 2</p>
</dd>
<dt>chr</dt><dd><p>Chromosome</p>
</dd>
<dt>pos</dt><dd><p>Genomic position</p>
</dd>
<dt>ldlcbeta</dt><dd><p>The genotype-low-density lipoprotein cholestrol associations</p>
</dd>
<dt>hdlcbeta</dt><dd><p>The genotype-high-density lipoprotein cholestrol associations</p>
</dd>
<dt>tgbeta</dt><dd><p>The genotype-triglyceride associations</p>
</dd>
<dt>chdbeta</dt><dd><p>The genotype-coronary heart disease associations, on the log odds ratio scale</p>
</dd>
<dt>ldlcp2</dt><dd><p>P-value for genotype-LDL-C associations</p>
</dd>
<dt>hdlcp2</dt><dd><p>P-value for genotype-HDL-C associations</p>
</dd>
<dt>tgp2</dt><dd><p>P-value for genotype-triglyceride associations</p>
</dd>
<dt>chdp2</dt><dd><p>P-value for genotype-coronary heart disease associations</p>
</dd>
<dt>ldlcz</dt><dd><p>Z-score for genotype-LDL-C associations</p>
</dd>
<dt>ldlcse</dt><dd><p>The standard errors of the genotype-low-density lipoprotein cholestrol associations</p>
</dd>
<dt>hdlcz</dt><dd><p>Z-score for genotype-HDL-C associations</p>
</dd>
<dt>hdlcse</dt><dd><p>The standard errors of the genotype-high-density lipoprotein cholestrol associations</p>
</dd>
<dt>tgz</dt><dd><p>Z-score for genotype-triglyceride associations</p>
</dd>
<dt>tgse</dt><dd><p>The standard errors of the genotype-triglyceride cholestrol associations</p>
</dd>
<dt>chdz</dt><dd><p>Z-score for genotype-coronary heart disease associations</p>
</dd>
<dt>chdse</dt><dd><p>The standard errors of the genotype-coronary heart disease associations</p>
</dd>
</dl>



<h3>Details</h3>

<p>dodata.
</p>


<h3>References</h3>

<p>Do, R. et al., Common variants associated with plasma triglycerides and risk for coronary artery disease. Nature Genetics, 2013, 45, 1345-1352, doi: <a href="https://doi.org/10.1038/ng.2795">10.1038/ng.2795</a>.
</p>

<hr>
<h2 id='mr_egger_rjags'>Bayesian implementation of the MR-Egger multivariate model with choice of prior distributions fitted using JAGS.</h2><span id='topic+mr_egger_rjags'></span>

<h3>Description</h3>

<p>Bayesian implementation of the MR-Egger multivariate model with choice of prior distributions fitted using JAGS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_egger_rjags(
  object,
  prior = "default",
  betaprior = "",
  sigmaprior = "",
  n.chains = 3,
  n.burn = 1000,
  n.iter = 5000,
  seed = NULL,
  rho = 0.5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_egger_rjags_+3A_object">object</code></td>
<td>
<p>A data object of class <code><a href="#topic+mr_format">mr_format</a></code>.</p>
</td></tr>
<tr><td><code id="mr_egger_rjags_+3A_prior">prior</code></td>
<td>
<p>A character string for selecting the prior distributions;
</p>

<ul>
<li> <p><code>"default"</code> selects a non-informative set of priors;
</p>
</li>
<li> <p><code>"weak"</code> selects weakly informative priors;
</p>
</li>
<li> <p><code>"pseudo"</code> selects a pseudo-horseshoe prior on the causal effect;
</p>
</li>
<li> <p><code>"joint"</code> selects a joint prior on the intercept and slope.
</p>
</li></ul>
</td></tr>
<tr><td><code id="mr_egger_rjags_+3A_betaprior">betaprior</code></td>
<td>
<p>A character string in JAGS syntax to allow a user defined prior for the causal effect.</p>
</td></tr>
<tr><td><code id="mr_egger_rjags_+3A_sigmaprior">sigmaprior</code></td>
<td>
<p>A character string in JAGS syntax to allow a user defined prior for the residual standard deviation.</p>
</td></tr>
<tr><td><code id="mr_egger_rjags_+3A_n.chains">n.chains</code></td>
<td>
<p>Numeric indicating the number of chains used in the MCMC estimation, the default is <code>3</code> chains.</p>
</td></tr>
<tr><td><code id="mr_egger_rjags_+3A_n.burn">n.burn</code></td>
<td>
<p>Numeric indicating the burn-in period of the Bayesian MCMC estimation. The default is <code>1000</code> samples.</p>
</td></tr>
<tr><td><code id="mr_egger_rjags_+3A_n.iter">n.iter</code></td>
<td>
<p>Numeric indicating the number of iterations in the Bayesian MCMC estimation. The default is <code>5000</code> iterations.</p>
</td></tr>
<tr><td><code id="mr_egger_rjags_+3A_seed">seed</code></td>
<td>
<p>Numeric indicating the random number seed. The default is the rjags default.</p>
</td></tr>
<tr><td><code id="mr_egger_rjags_+3A_rho">rho</code></td>
<td>
<p>Numeric indicating the correlation coefficient input into the joint prior distribution. The default value is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="mr_egger_rjags_+3A_...">...</code></td>
<td>
<p>Additional arguments passed through to <code><a href="rjags.html#topic+jags.model">rjags::jags.model()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>eggerjags</code> containing the following components:
</p>

<dl>
<dt>AvgPleio</dt><dd><p>The mean of the simulated pleiotropic effect</p>
</dd>
<dt>CausalEffect</dt><dd><p>The mean of the simulated causal effect</p>
</dd>
<dt>StandardError</dt><dd><p>Standard deviation of the simulated causal effect</p>
</dd>
<dt>sigma</dt><dd><p>The value of the residual standard deviation</p>
</dd>
<dt>CredibleInterval</dt><dd><p>The credible interval for the causal effect, which includes the lower (2.5%), median (50%) and upper intervals (97.5%)</p>
</dd>
<dt>samples</dt><dd><p>Output of the Bayesian MCMC samples</p>
</dd>
<dt>Priors</dt><dd><p>The specified priors</p>
</dd>
</dl>



<h3>References</h3>

<p>Bowden et. al., Mendelian randomization with invalid instruments: effect estimation and bias detection through Egger regression. International Journal of Epidemiology 2015. 44(2): p. 512-525. doi: <a href="https://doi.org/10.1093/ije/dyv080">10.1093/ije/dyv080</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("rjags", quietly = TRUE)) {
fit &lt;- mr_egger_rjags(bmi_insulin)
summary(fit)
plot(fit$samples)
# 90% credible interval
fitdf &lt;- do.call(rbind.data.frame, fit$samples)
cri90 &lt;- sapply(fitdf, quantile, probs = c(0.05,0.95))
print(cri90)
}

</code></pre>

<hr>
<h2 id='mr_egger_stan'>Bayesian inverse variance weighted model with a choice of prior distributions fitted using Stan</h2><span id='topic+mr_egger_stan'></span>

<h3>Description</h3>

<p>Bayesian inverse variance weighted model with a choice of prior distributions fitted using Stan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_egger_stan(
  data,
  prior = 1,
  n.chains = 3,
  n.burn = 1000,
  n.iter = 5000,
  seed = 12345,
  rho = 0.5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_egger_stan_+3A_data">data</code></td>
<td>
<p>A data of class <code><a href="#topic+mr_format">mr_format</a></code>.</p>
</td></tr>
<tr><td><code id="mr_egger_stan_+3A_prior">prior</code></td>
<td>
<p>An integer for selecting the prior distributions;
</p>

<ul>
<li> <p><code>1</code> selects a non-informative set of priors;
</p>
</li>
<li> <p><code>2</code> selects weakly informative priors;
</p>
</li>
<li> <p><code>3</code> selects a pseudo-horseshoe prior on the causal effect;
</p>
</li>
<li> <p><code>4</code> selects joint prior of the intercept and causal effect estimate.
</p>
</li></ul>
</td></tr>
<tr><td><code id="mr_egger_stan_+3A_n.chains">n.chains</code></td>
<td>
<p>Numeric indicating the number of chains used in the HMC estimation in rstan, the default is <code>3</code> chains.</p>
</td></tr>
<tr><td><code id="mr_egger_stan_+3A_n.burn">n.burn</code></td>
<td>
<p>Numeric indicating the burn-in period of the Bayesian HMC estimation. The default is <code>1000</code> samples.</p>
</td></tr>
<tr><td><code id="mr_egger_stan_+3A_n.iter">n.iter</code></td>
<td>
<p>Numeric indicating the number of iterations in the Bayesian HMC estimation. The default is <code>5000</code> iterations.</p>
</td></tr>
<tr><td><code id="mr_egger_stan_+3A_seed">seed</code></td>
<td>
<p>Numeric indicating the random number seed. The default is <code>12345</code>.</p>
</td></tr>
<tr><td><code id="mr_egger_stan_+3A_rho">rho</code></td>
<td>
<p>Numeric indicating the correlation coefficient input into the joint prior distribution. The default is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="mr_egger_stan_+3A_...">...</code></td>
<td>
<p>Additional arguments passed through to <code><a href="rstan.html#topic+stanmodel-method-sampling">rstan::sampling()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>stanfit</code>.
</p>


<h3>References</h3>

<p>Bowden J, Davey Smith G, Burgess S. Mendelian randomization with invalid instruments: effect estimation and bias detection through Egger regression. International Journal of Epidemiology, 2015, 44, 2, 512-525. doi: <a href="https://doi.org/10.1093/ije/dyv080">10.1093/ije/dyv080</a>.
</p>
<p>Stan Development Team (2020). &quot;RStan: the R interface to Stan.&quot; R package version 2.19.3, <a href="https://mc-stan.org/">https://mc-stan.org/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("rstan", quietly = TRUE)) {
# Note we recommend setting n.burn and n.iter to larger values
egger_fit &lt;- mr_egger_stan(bmi_insulin, n.burn = 500, n.iter = 1000)
print(egger_fit)
}

</code></pre>

<hr>
<h2 id='mr_format'>Organises the summary level data for use in the Bayesian MR functions</h2><span id='topic+mr_format'></span>

<h3>Description</h3>

<p>Organises the summary level data for use in the Bayesian MR functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_format(rsid, xbeta, ybeta, xse, yse)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_format_+3A_rsid">rsid</code></td>
<td>
<p>A vector of genetic variants used for analysis, if unspecified a vector is automatically generated.</p>
</td></tr>
<tr><td><code id="mr_format_+3A_xbeta">xbeta</code></td>
<td>
<p>A numeric vector of the instrument-phenotype associations.</p>
</td></tr>
<tr><td><code id="mr_format_+3A_ybeta">ybeta</code></td>
<td>
<p>A numeric vector of the instrument-outcome associations.</p>
</td></tr>
<tr><td><code id="mr_format_+3A_xse">xse</code></td>
<td>
<p>The standard errors of the instrument-phenotype associations <code>xbeta</code>.</p>
</td></tr>
<tr><td><code id="mr_format_+3A_yse">yse</code></td>
<td>
<p>The standard errors of the instrument-outcome associations <code>ybeta</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A formatted data frame for analysis of class <code>mr_format</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(bmi_insulin)
dat &lt;- mr_format(rsid = bmi_insulin[,"rsid"],
          xbeta = bmi_insulin[,"beta.exposure"],
          ybeta = bmi_insulin[,"beta.outcome"],
          xse = bmi_insulin[,"se.exposure"],
          yse = bmi_insulin[,"se.outcome"])
class(dat)

</code></pre>

<hr>
<h2 id='mr_ivw_rjags'>Bayesian inverse variance weighted model with a choice of prior distributions fitted using JAGS.</h2><span id='topic+mr_ivw_rjags'></span>

<h3>Description</h3>

<p>Bayesian inverse variance weighted model with a choice of prior distributions fitted using JAGS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_ivw_rjags(
  object,
  prior = "default",
  betaprior = "",
  n.chains = 3,
  n.burn = 1000,
  n.iter = 5000,
  seed = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_ivw_rjags_+3A_object">object</code></td>
<td>
<p>A data object of class <code><a href="#topic+mr_format">mr_format</a></code>.</p>
</td></tr>
<tr><td><code id="mr_ivw_rjags_+3A_prior">prior</code></td>
<td>
<p>A character string for selecting the prior distributions;
</p>

<ul>
<li> <p><code>"default"</code> selects a non-informative set of priors;
</p>
</li>
<li> <p><code>"weak"</code> selects weakly informative priors;
</p>
</li>
<li> <p><code>"pseudo"</code> selects a pseudo-horseshoe prior on the causal effect.
</p>
</li></ul>
</td></tr>
<tr><td><code id="mr_ivw_rjags_+3A_betaprior">betaprior</code></td>
<td>
<p>A character string in JAGS syntax to allow a user defined prior for the causal effect.</p>
</td></tr>
<tr><td><code id="mr_ivw_rjags_+3A_n.chains">n.chains</code></td>
<td>
<p>Numeric indicating the number of chains used in the MCMC estimation, the default is <code>3</code> chains.</p>
</td></tr>
<tr><td><code id="mr_ivw_rjags_+3A_n.burn">n.burn</code></td>
<td>
<p>Numeric indicating the burn-in period of the Bayesian MCMC estimation. The default is <code>1000</code> samples.</p>
</td></tr>
<tr><td><code id="mr_ivw_rjags_+3A_n.iter">n.iter</code></td>
<td>
<p>Numeric indicating the number of iterations in the Bayesian MCMC estimation. The default is <code>5000</code> iterations.</p>
</td></tr>
<tr><td><code id="mr_ivw_rjags_+3A_seed">seed</code></td>
<td>
<p>Numeric indicating the random number seed. The default is the rjags default.</p>
</td></tr>
<tr><td><code id="mr_ivw_rjags_+3A_...">...</code></td>
<td>
<p>Additional arguments passed through to <code><a href="rjags.html#topic+jags.model">rjags::jags.model()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>ivwjags</code> containing the following components:
</p>

<dl>
<dt>CausalEffect</dt><dd><p>The mean of the simulated causal effects</p>
</dd>
<dt>StandardError</dt><dd><p>Standard deviation of the simulated causal effects</p>
</dd>
<dt>CredibleInterval</dt><dd><p>The credible interval for the causal effect, which indicates the lower(2.5%), median (50%) and upper intervals (97.5%)</p>
</dd>
<dt>samples</dt><dd><p>Output of the Bayesian MCMC samples with the different chains</p>
</dd>
<dt>Priors</dt><dd><p>The specified priors</p>
</dd>
</dl>



<h3>References</h3>

<p>Burgess, S., Butterworth, A., Thompson S.G. Mendelian randomization analysis with multiple genetic variants using summarized data. Genetic Epidemiology, 2013, 37, 7, 658-665 doi: <a href="https://doi.org/10.1002/gepi.21758">10.1002/gepi.21758</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("rjags", quietly = TRUE)) {
fit &lt;- mr_ivw_rjags(bmi_insulin)
print(fit)
summary(fit)
plot(fit$samples)
# 90% credible interval
fitdf &lt;- do.call(rbind.data.frame, fit$samples)
cri90 &lt;- quantile(fitdf$Estimate, probs = c(0.05,0.95))
print(cri90)
}

</code></pre>

<hr>
<h2 id='mr_ivw_stan'>Bayesian inverse variance weighted model with a choice of prior distributions fitted using RStan.</h2><span id='topic+mr_ivw_stan'></span>

<h3>Description</h3>

<p>Bayesian inverse variance weighted model with a choice of prior distributions fitted using RStan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_ivw_stan(
  data,
  prior = 1,
  n.chains = 3,
  n.burn = 1000,
  n.iter = 5000,
  seed = 12345,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_ivw_stan_+3A_data">data</code></td>
<td>
<p>A data of class <code><a href="#topic+mr_format">mr_format</a></code>.</p>
</td></tr>
<tr><td><code id="mr_ivw_stan_+3A_prior">prior</code></td>
<td>
<p>An integer for selecting the prior distributions;
</p>

<ul>
<li> <p><code>1</code> selects a non-informative set of priors;
</p>
</li>
<li> <p><code>2</code> selects weakly informative priors;
</p>
</li>
<li> <p><code>3</code> selects a pseudo-horseshoe prior on the causal effect.
</p>
</li></ul>
</td></tr>
<tr><td><code id="mr_ivw_stan_+3A_n.chains">n.chains</code></td>
<td>
<p>Numeric indicating the number of chains used in the HMC estimation in rstan, the default is <code>3</code> chains.</p>
</td></tr>
<tr><td><code id="mr_ivw_stan_+3A_n.burn">n.burn</code></td>
<td>
<p>Numeric indicating the burn-in period of the Bayesian HMC estimation. The default is <code>1000</code> samples.</p>
</td></tr>
<tr><td><code id="mr_ivw_stan_+3A_n.iter">n.iter</code></td>
<td>
<p>Numeric indicating the number of iterations in the Bayesian MCMC estimation. The default is <code>5000</code> iterations.</p>
</td></tr>
<tr><td><code id="mr_ivw_stan_+3A_seed">seed</code></td>
<td>
<p>Numeric indicating the random number seed. The default is <code>12345</code>.</p>
</td></tr>
<tr><td><code id="mr_ivw_stan_+3A_...">...</code></td>
<td>
<p>Additional arguments passed through to <code><a href="rstan.html#topic+stanmodel-method-sampling">rstan::sampling()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>stanfit</code>.
</p>


<h3>References</h3>

<p>Burgess, S., Butterworth, A., Thompson S.G. Mendelian randomization analysis with multiple genetic variants using summarized data. Genetic Epidemiology, 2013, 37, 7, 658-665 doi: <a href="https://doi.org/10.1002/gepi.21758">10.1002/gepi.21758</a>.
</p>
<p>Stan Development Team (2020). &quot;RStan: the R interface to Stan.&quot; R package version 2.19.3, <a href="https://mc-stan.org/">https://mc-stan.org/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("rstan", quietly = TRUE)) {
ivw_fit &lt;- mr_ivw_stan(bmi_insulin)
print(ivw_fit)
rstan::traceplot(ivw_fit)
}
</code></pre>

<hr>
<h2 id='mr_radialegger_rjags'>Bayesian radial MR-Egger model with a choice of prior distributions fitted using JAGS.</h2><span id='topic+mr_radialegger_rjags'></span>

<h3>Description</h3>

<p>Bayesian radial MR-Egger model with a choice of prior distributions fitted using JAGS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_radialegger_rjags(
  object,
  prior = "default",
  betaprior = "",
  sigmaprior = "",
  n.chains = 3,
  n.burn = 1000,
  n.iter = 5000,
  seed = NULL,
  rho = 0.5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_radialegger_rjags_+3A_object">object</code></td>
<td>
<p>A data object of class <code><a href="#topic+mr_format">mr_format</a></code>.</p>
</td></tr>
<tr><td><code id="mr_radialegger_rjags_+3A_prior">prior</code></td>
<td>
<p>A character string for selecting the prior distributions;
</p>

<ul>
<li> <p><code>"default"</code> selects a non-informative set of priors;
</p>
</li>
<li> <p><code>"weak"</code> selects weakly informative priors;
</p>
</li>
<li> <p><code>"pseudo"</code> selects a pseudo-horseshoe prior on the causal effect;
</p>
</li>
<li> <p><code>"joint"</code> selects a joint prior on the intercept and slope.
</p>
</li></ul>
</td></tr>
<tr><td><code id="mr_radialegger_rjags_+3A_betaprior">betaprior</code></td>
<td>
<p>A character string in JAGS syntax to allow a user defined prior for the causal effect.</p>
</td></tr>
<tr><td><code id="mr_radialegger_rjags_+3A_sigmaprior">sigmaprior</code></td>
<td>
<p>A character string in JAGS syntax to allow a user defined prior for the residual standard deviation.</p>
</td></tr>
<tr><td><code id="mr_radialegger_rjags_+3A_n.chains">n.chains</code></td>
<td>
<p>Numeric indicating the number of chains used in the MCMC estimation, the default is <code>3</code> chains.</p>
</td></tr>
<tr><td><code id="mr_radialegger_rjags_+3A_n.burn">n.burn</code></td>
<td>
<p>Numeric indicating the burn-in period of the Bayesian MCMC estimation. The default is <code>1000</code> samples.</p>
</td></tr>
<tr><td><code id="mr_radialegger_rjags_+3A_n.iter">n.iter</code></td>
<td>
<p>Numeric indicating the number of iterations in the Bayesian MCMC estimation. The default is <code>5000</code> iterations.</p>
</td></tr>
<tr><td><code id="mr_radialegger_rjags_+3A_seed">seed</code></td>
<td>
<p>Numeric indicating the random number seed. The default is the rjags default.</p>
</td></tr>
<tr><td><code id="mr_radialegger_rjags_+3A_rho">rho</code></td>
<td>
<p>Numeric indicating the correlation coefficient input into the joint prior distribution. The default is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="mr_radialegger_rjags_+3A_...">...</code></td>
<td>
<p>Additional arguments passed through to <code><a href="rjags.html#topic+jags.model">rjags::jags.model()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>radialeggerjags</code> containing the following components:
</p>

<dl>
<dt>AvgPleio</dt><dd><p>The mean of the simulated pleiotropic effect</p>
</dd>
<dt>CausalEffect</dt><dd><p>The mean of the simulated causal effect</p>
</dd>
<dt>StandardError</dt><dd><p>Standard deviation of the simulated causal effect</p>
</dd>
<dt>sigma</dt><dd><p>The mean of the simaulted residual standard deviation</p>
</dd>
<dt>CredibleInterval</dt><dd><p>The credible interval for the causal effect, which includes the lower (2.5%), median (50%) and upper intervals (97.5%)</p>
</dd>
<dt>samples</dt><dd><p>Output of the Bayesian MCMC samples</p>
</dd>
<dt>Prior</dt><dd><p>The specified priors</p>
</dd>
</dl>



<h3>References</h3>

<p>Bowden, J., et al., Improving the visualization, interpretation and analysis of two-sample summary data Mendelian randomization via the Radial plot and Radial regression. International Journal of Epidemiology, 2018. 47(4): p. 1264-1278. doi: <a href="https://doi.org/10.1093/ije/dyy101">10.1093/ije/dyy101</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("rjags", quietly = TRUE)) {
fit &lt;- mr_radialegger_rjags(bmi_insulin)
summary(fit)
plot(fit$samples)
# 90% credible interval
fitdf &lt;- do.call(rbind.data.frame, fit$samples)
cri90 &lt;- quantile(fitdf$Estimate, probs = c(0.05,0.95))
print(cri90)
}

</code></pre>

<hr>
<h2 id='mr_radialegger_stan'>Bayesian inverse variance weighted model with a choice of prior distributions fitted using RStan.</h2><span id='topic+mr_radialegger_stan'></span>

<h3>Description</h3>

<p>Bayesian inverse variance weighted model with a choice of prior distributions fitted using RStan
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_radialegger_stan(
  data,
  prior = 1,
  n.chains = 3,
  n.burn = 1000,
  n.iter = 5000,
  rho = 0.5,
  seed = 12345,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_radialegger_stan_+3A_data">data</code></td>
<td>
<p>A data of class <code><a href="#topic+mr_format">mr_format</a></code>.</p>
</td></tr>
<tr><td><code id="mr_radialegger_stan_+3A_prior">prior</code></td>
<td>
<p>An integer for selecting the prior distributions;
</p>

<ul>
<li> <p><code>1</code> selects a non-informative set of priors;
</p>
</li>
<li> <p><code>2</code> selects weakly informative priors;
</p>
</li>
<li> <p><code>3</code> selects a pseudo-horseshoe prior on the causal effect;
</p>
</li>
<li> <p><code>4</code> selects joint prior of the intercept and causal effect estimate.
</p>
</li></ul>
</td></tr>
<tr><td><code id="mr_radialegger_stan_+3A_n.chains">n.chains</code></td>
<td>
<p>Numeric indicating the number of chains used in the HMC estimation in rstan, the default is <code>3</code> chains.</p>
</td></tr>
<tr><td><code id="mr_radialegger_stan_+3A_n.burn">n.burn</code></td>
<td>
<p>Numeric indicating the burn-in period of the Bayesian HMC estimation. The default is <code>1000</code> samples.</p>
</td></tr>
<tr><td><code id="mr_radialegger_stan_+3A_n.iter">n.iter</code></td>
<td>
<p>Numeric indicating the number of iterations in the Bayesian HMC estimation. The default is <code>5000</code> iterations.</p>
</td></tr>
<tr><td><code id="mr_radialegger_stan_+3A_rho">rho</code></td>
<td>
<p>Numeric indicating the correlation coefficient input into the joint prior distribution. The default is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="mr_radialegger_stan_+3A_seed">seed</code></td>
<td>
<p>Numeric indicating the random number seed. The default is <code>12345</code>.</p>
</td></tr>
<tr><td><code id="mr_radialegger_stan_+3A_...">...</code></td>
<td>
<p>Additional arguments passed through to <code><a href="rstan.html#topic+stanmodel-method-sampling">rstan::sampling()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>stanfit</code>.
</p>


<h3>References</h3>

<p>Bowden, J., et al., Improving the visualization, interpretation and analysis of two-sample summary data Mendelian randomization via the Radial plot and Radial regression. International Journal of Epidemiology, 2018. 47(4): p. 1264-1278. doi: <a href="https://doi.org/10.1093/ije/dyy101">10.1093/ije/dyy101</a>.
</p>
<p>Stan Development Team (2020). &quot;RStan: the R interface to Stan.&quot; R package version 2.19.3, <a href="https://mc-stan.org/">https://mc-stan.org/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("rstan", quietly = TRUE)) {
# Note we recommend setting n.burn and n.iter to larger values
radegger_fit &lt;- mr_radialegger_stan(bmi_insulin, n.burn = 500, n.iter = 1000)
print(radegger_fit)
}

</code></pre>

<hr>
<h2 id='mrbayes-package'>mrbayes: Bayesian implementation of the IVW and MR-Egger models for two-sample Mendelian randomization analyses</h2><span id='topic+mrbayes-package'></span><span id='topic+mrbayes'></span>

<h3>Description</h3>

<p>Bayesian implementation of the IVW and MR-Egger models and their radial and
multivariate versions for two-sample Mendelian randomization analyses.
</p>


<h3>References</h3>

<p>Stan Development Team (2019). RStan: the R interface to Stan. R package version 2.19.2. <a href="https://mc-stan.org">https://mc-stan.org</a>
</p>

<hr>
<h2 id='mrinput_mr_format'>Convert an object of class MRInput from the MendelianRandomization package to the mrbayes mr_format class</h2><span id='topic+mrinput_mr_format'></span>

<h3>Description</h3>

<p>Creates a data.frame with class <code>mr_format</code> from an object of class <code>MRInput</code> generated by <code><a href="MendelianRandomization.html#topic+mr_input">MendelianRandomization::mr_input</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrinput_mr_format(dat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrinput_mr_format_+3A_dat">dat</code></td>
<td>
<p>Object from <code><a href="MendelianRandomization.html#topic+mr_input">MendelianRandomization::mr_input</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code>mr_format</code>, the mrbayes format
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("MendelianRandomization", quietly = TRUE)) {
dat &lt;- MendelianRandomization::mr_input(bx = bmi_insulin$beta.exposure,
                                        bxse = bmi_insulin$se.exposure,
                                        by = bmi_insulin$beta.outcome,
                                        byse = bmi_insulin$se.outcome,
                                        snps = bmi_insulin$rsid)
dat &lt;- mrinput_mr_format(dat)
head(dat)
class(dat)
}
</code></pre>

<hr>
<h2 id='mvmr_egger_rjags'>Bayesian implementation of the MVMR-Egger model with choice of prior distributions fitted using JAGS.</h2><span id='topic+mvmr_egger_rjags'></span>

<h3>Description</h3>

<p>Bayesian implementation of the MVMR-Egger model with choice of prior distributions fitted using JAGS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mvmr_egger_rjags(
  object,
  prior = "default",
  betaprior = "",
  sigmaprior = "",
  orientate = 1,
  n.chains = 3,
  n.burn = 1000,
  n.iter = 5000,
  seed = NULL,
  rho = 0.5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mvmr_egger_rjags_+3A_object">object</code></td>
<td>
<p>A data object of class <code><a href="#topic+mvmr_format">mvmr_format</a></code>.</p>
</td></tr>
<tr><td><code id="mvmr_egger_rjags_+3A_prior">prior</code></td>
<td>
<p>A character string for selecting the prior distributions;
</p>

<ul>
<li> <p><code>"default"</code> selects a non-informative set of priors;
</p>
</li>
<li> <p><code>"weak"</code> selects weakly informative priors;
</p>
</li>
<li> <p><code>"pseudo"</code> selects a pseudo-horseshoe prior on the causal effect;
</p>
</li></ul>
</td></tr>
<tr><td><code id="mvmr_egger_rjags_+3A_betaprior">betaprior</code></td>
<td>
<p>A character string in JAGS syntax to allow a user defined prior for the causal effect.</p>
</td></tr>
<tr><td><code id="mvmr_egger_rjags_+3A_sigmaprior">sigmaprior</code></td>
<td>
<p>A character string in JAGS syntax to allow a user defined prior for the residual standard deviation.</p>
</td></tr>
<tr><td><code id="mvmr_egger_rjags_+3A_orientate">orientate</code></td>
<td>
<p>Numeric value to indicate the oriented exposure</p>
</td></tr>
<tr><td><code id="mvmr_egger_rjags_+3A_n.chains">n.chains</code></td>
<td>
<p>Numeric indicating the number of chains used in the MCMC estimation, the default is <code>3</code> chains.</p>
</td></tr>
<tr><td><code id="mvmr_egger_rjags_+3A_n.burn">n.burn</code></td>
<td>
<p>Numeric indicating the burn-in period of the Bayesian MCMC estimation. The default is <code>1000</code> samples.</p>
</td></tr>
<tr><td><code id="mvmr_egger_rjags_+3A_n.iter">n.iter</code></td>
<td>
<p>Numeric indicating the number of iterations in the Bayesian MCMC estimation. The default is <code>5000</code> iterations.</p>
</td></tr>
<tr><td><code id="mvmr_egger_rjags_+3A_seed">seed</code></td>
<td>
<p>Numeric indicating the random number seed. The default is the rjags default.</p>
</td></tr>
<tr><td><code id="mvmr_egger_rjags_+3A_rho">rho</code></td>
<td>
<p>Numeric indicating the correlation coefficient input into the joint prior distribution. The default value is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="mvmr_egger_rjags_+3A_...">...</code></td>
<td>
<p>Additional arguments passed through to <code><a href="rjags.html#topic+jags.model">rjags::jags.model()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>mveggerjags</code> containing the following components:
if (requireNamespace(&quot;rjags&quot;, quietly = TRUE)) 
</p>










<h3>References</h3>

<p>Bowden et. al., Mendelian randomization with invalid instruments: effect estimation and bias detection through Egger regression. International Journal of Epidemiology 2015. 44(2): p. 512-525. doi: <a href="https://doi.org/10.1093/ije/dyv080">10.1093/ije/dyv080</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
if (requireNamespace("rjags", quietly = TRUE)) {
dat &lt;- mvmr_format(rsid = dodata$rsid,
         xbeta = cbind(dodata$ldlcbeta,dodata$hdlcbeta,dodata$tgbeta),
         ybeta = dodata$chdbeta,
         xse = cbind(dodata$ldlcse,dodata$hdlcse,dodata$tgse),
         yse = dodata$chdse)

fit &lt;- mvmr_egger_rjags(dat)
summary(fit)
plot(fit$samples)
# 90% credible interval
fitdf &lt;- do.call(rbind.data.frame, fit$samples)
cri90 &lt;- sapply(fitdf, quantile, probs = c(0.05, 0.95))
print(cri90)
}

## End(Not run)
</code></pre>

<hr>
<h2 id='mvmr_egger_stan'>Bayesian implementation of the MVMR-Egger model with choice of prior distributions fitted using RStan.</h2><span id='topic+mvmr_egger_stan'></span>

<h3>Description</h3>

<p>Bayesian implementation of the MVMR-Egger model with choice of prior distributions fitted using RStan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mvmr_egger_stan(
  data,
  prior = 1,
  n.chains = 3,
  n.burn = 1000,
  n.iter = 5000,
  seed = 12345,
  rho = 0.5,
  orientate = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mvmr_egger_stan_+3A_data">data</code></td>
<td>
<p>A data of class <code><a href="#topic+mvmr_format">mvmr_format</a></code>.</p>
</td></tr>
<tr><td><code id="mvmr_egger_stan_+3A_prior">prior</code></td>
<td>
<p>An integer for selecting the prior distributions;
</p>

<ul>
<li> <p><code>1</code> selects a non-informative set of priors;
</p>
</li>
<li> <p><code>2</code> selects weakly informative priors;
</p>
</li>
<li> <p><code>3</code> selects a pseudo-horseshoe prior on the causal effect;
</p>
</li></ul>
</td></tr>
<tr><td><code id="mvmr_egger_stan_+3A_n.chains">n.chains</code></td>
<td>
<p>Numeric indicating the number of chains used in the HMC estimation in rstan, the default is <code>3</code> chains.</p>
</td></tr>
<tr><td><code id="mvmr_egger_stan_+3A_n.burn">n.burn</code></td>
<td>
<p>Numeric indicating the burn-in period of the Bayesian HMC estimation. The default is <code>1000</code> samples.</p>
</td></tr>
<tr><td><code id="mvmr_egger_stan_+3A_n.iter">n.iter</code></td>
<td>
<p>Numeric indicating the number of iterations in the Bayesian HMC estimation. The default is <code>5000</code> iterations.</p>
</td></tr>
<tr><td><code id="mvmr_egger_stan_+3A_seed">seed</code></td>
<td>
<p>Numeric indicating the random number seed. The default is <code>12345</code>.</p>
</td></tr>
<tr><td><code id="mvmr_egger_stan_+3A_rho">rho</code></td>
<td>
<p>Numeric indicating the correlation coefficient input into the joint prior distribution. The default is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="mvmr_egger_stan_+3A_orientate">orientate</code></td>
<td>
<p>Numeric value to indicate the oriented exposure.</p>
</td></tr>
<tr><td><code id="mvmr_egger_stan_+3A_...">...</code></td>
<td>
<p>Additional arguments passed through to <code><a href="rstan.html#topic+stanmodel-method-sampling">rstan::sampling()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>stanfit</code>.
</p>


<h3>References</h3>

<p>Bowden J, Davey Smith G, Burgess S. Mendelian randomization with invalid instruments: effect estimation and bias detection through Egger regression. International Journal of Epidemiology, 2015, 44, 2, 512-525. doi: <a href="https://doi.org/10.1093/ije/dyv080">10.1093/ije/dyv080</a>.
</p>
<p>Stan Development Team (2020). &quot;RStan: the R interface to Stan.&quot; R package version 2.19.3, <a href="https://mc-stan.org/">https://mc-stan.org/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("rstan", quietly = TRUE)) {
# Note we recommend setting n.burn and n.iter to larger values

dat &lt;- mvmr_format(rsid = dodata$rsid,
         xbeta = cbind(dodata$ldlcbeta,dodata$hdlcbeta,dodata$tgbeta),
         ybeta = dodata$chdbeta,
         xse = cbind(dodata$ldlcse,dodata$hdlcse,dodata$tgse),
         yse = dodata$chdse)
mvegger_fit &lt;- mvmr_egger_stan(dat, n.burn = 500, n.iter = 1000)
print(mvegger_fit)
}

</code></pre>

<hr>
<h2 id='mvmr_format'>Organises the summary level data for use in the Bayesian MR functions</h2><span id='topic+mvmr_format'></span>

<h3>Description</h3>

<p>Organises the summary level data for use in the Bayesian MR functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mvmr_format(rsid, xbeta, ybeta, xse, yse)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mvmr_format_+3A_rsid">rsid</code></td>
<td>
<p>A vector of genetic variants used for analysis, if unspecified a vector is automatically generated.</p>
</td></tr>
<tr><td><code id="mvmr_format_+3A_xbeta">xbeta</code></td>
<td>
<p>A matrix of multiple instrument-phenotypes associations.</p>
</td></tr>
<tr><td><code id="mvmr_format_+3A_ybeta">ybeta</code></td>
<td>
<p>A numeric vector of the instrument-outcome associations.</p>
</td></tr>
<tr><td><code id="mvmr_format_+3A_xse">xse</code></td>
<td>
<p>The matrix for corresponding standard errors of the instrument-phenotypes associations <code>xbeta</code>.</p>
</td></tr>
<tr><td><code id="mvmr_format_+3A_yse">yse</code></td>
<td>
<p>The standard errors of the instrument-outcome associations <code>ybeta</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A formatted data frame for analysis of class <code>mvmr_format</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dodata)
dat &lt;- mvmr_format(rsid = dodata$rsid,
          xbeta = cbind(dodata$ldlcbeta,dodata$hdlcbeta,dodata$tgbeta),
          ybeta = dodata$chdbeta,
          xse = cbind(dodata$ldlcse,dodata$hdlcse,dodata$tgse),
          yse = dodata$chdse)
class(dat)

</code></pre>

<hr>
<h2 id='mvmr_ivw_rjags'>Bayesian multivariate inverse variance weighted model with a choice of prior distributions fitted using JAGS.</h2><span id='topic+mvmr_ivw_rjags'></span>

<h3>Description</h3>

<p>Bayesian multivariate inverse variance weighted model with a choice of prior distributions fitted using JAGS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mvmr_ivw_rjags(
  object,
  prior = "default",
  betaprior = "",
  n.chains = 3,
  n.burn = 1000,
  n.iter = 5000,
  seed = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mvmr_ivw_rjags_+3A_object">object</code></td>
<td>
<p>A data object of class <code><a href="#topic+mvmr_format">mvmr_format</a></code>.</p>
</td></tr>
<tr><td><code id="mvmr_ivw_rjags_+3A_prior">prior</code></td>
<td>
<p>A character string for selecting the prior distributions;
</p>

<ul>
<li> <p><code>"default"</code> selects a non-informative set of priors;
</p>
</li>
<li> <p><code>"weak"</code> selects weakly informative priors;
</p>
</li>
<li> <p><code>"pseudo"</code> selects a pseudo-horseshoe prior on the causal effect.
</p>
</li></ul>
</td></tr>
<tr><td><code id="mvmr_ivw_rjags_+3A_betaprior">betaprior</code></td>
<td>
<p>A character string in JAGS syntax to allow a user defined prior for the causal effect.</p>
</td></tr>
<tr><td><code id="mvmr_ivw_rjags_+3A_n.chains">n.chains</code></td>
<td>
<p>Numeric indicating the number of chains used in the MCMC estimation, the default is <code>3</code> chains.</p>
</td></tr>
<tr><td><code id="mvmr_ivw_rjags_+3A_n.burn">n.burn</code></td>
<td>
<p>Numeric indicating the burn-in period of the Bayesian MCMC estimation. The default is <code>1000</code> samples.</p>
</td></tr>
<tr><td><code id="mvmr_ivw_rjags_+3A_n.iter">n.iter</code></td>
<td>
<p>Numeric indicating the number of iterations in the Bayesian MCMC estimation. The default is <code>5000</code> iterations.</p>
</td></tr>
<tr><td><code id="mvmr_ivw_rjags_+3A_seed">seed</code></td>
<td>
<p>Numeric indicating the random number seed. The default is the rjags default.</p>
</td></tr>
<tr><td><code id="mvmr_ivw_rjags_+3A_...">...</code></td>
<td>
<p>Additional arguments passed through to <code><a href="rjags.html#topic+jags.model">rjags::jags.model()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>mvivwjags</code> containing the following components:
</p>

<dl>
<dt>CausalEffect</dt><dd><p>The mean of the simulated causal effects</p>
</dd>
<dt>StandardError</dt><dd><p>Standard deviation of the simulated causal effects</p>
</dd>
<dt>CredibleInterval</dt><dd><p>The credible interval for the causal effect, which indicates the lower(2.5%), median (50%) and upper intervals (97.5%)</p>
</dd>
<dt>samples</dt><dd><p>Output of the Bayesian MCMC samples with the different chains</p>
</dd>
<dt>Priors</dt><dd><p>The specified priors</p>
</dd>
</dl>



<h3>References</h3>

<p>Burgess, S., Butterworth, A., Thompson S.G. Mendelian randomization analysis with multiple genetic variants using summarized data. Genetic Epidemiology, 2013, 37, 7, 658-665 doi: <a href="https://doi.org/10.1002/gepi.21758">10.1002/gepi.21758</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("rjags", quietly = TRUE)) {
dat &lt;- mvmr_format(rsid = dodata$rsid,
         xbeta = cbind(dodata$ldlcbeta,dodata$hdlcbeta,dodata$tgbeta),
         ybeta = dodata$chdbeta,
         xse = cbind(dodata$ldlcse,dodata$hdlcse,dodata$tgse),
         yse = dodata$chdse)

fit &lt;- mvmr_ivw_rjags(dat)
print(fit)
summary(fit)
plot(fit$samples)
# 90% credible interval
fitdf &lt;- do.call(rbind.data.frame, fit$samples)
cri90 &lt;- sapply(fitdf, quantile, probs = c(0.05, 0.95))
print(cri90)
}

</code></pre>

<hr>
<h2 id='mvmr_ivw_stan'>Bayesian multivariate inverse variance weighted model with a choice of prior distributions fitted using RStan.</h2><span id='topic+mvmr_ivw_stan'></span>

<h3>Description</h3>

<p>Bayesian multivariate inverse variance weighted model with a choice of prior distributions fitted using RStan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mvmr_ivw_stan(
  data,
  prior = 1,
  n.chains = 3,
  n.burn = 1000,
  n.iter = 5000,
  seed = 12345,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mvmr_ivw_stan_+3A_data">data</code></td>
<td>
<p>A data of class <code><a href="#topic+mvmr_format">mvmr_format</a></code>.</p>
</td></tr>
<tr><td><code id="mvmr_ivw_stan_+3A_prior">prior</code></td>
<td>
<p>An integer for selecting the prior distributions;
</p>

<ul>
<li> <p><code>1</code> selects a non-informative set of priors;
</p>
</li>
<li> <p><code>2</code> selects weakly informative priors;
</p>
</li>
<li> <p><code>3</code> selects a pseudo-horseshoe prior on the causal effect.
</p>
</li></ul>
</td></tr>
<tr><td><code id="mvmr_ivw_stan_+3A_n.chains">n.chains</code></td>
<td>
<p>Numeric indicating the number of chains used in the HMC estimation in rstan, the default is <code>3</code> chains.</p>
</td></tr>
<tr><td><code id="mvmr_ivw_stan_+3A_n.burn">n.burn</code></td>
<td>
<p>Numeric indicating the burn-in period of the Bayesian HMC estimation. The default is <code>1000</code> samples.</p>
</td></tr>
<tr><td><code id="mvmr_ivw_stan_+3A_n.iter">n.iter</code></td>
<td>
<p>Numeric indicating the number of iterations in the Bayesian MCMC estimation. The default is <code>5000</code> iterations.</p>
</td></tr>
<tr><td><code id="mvmr_ivw_stan_+3A_seed">seed</code></td>
<td>
<p>Numeric indicating the random number seed. The default is <code>12345</code>.</p>
</td></tr>
<tr><td><code id="mvmr_ivw_stan_+3A_...">...</code></td>
<td>
<p>Additional arguments passed through to <code><a href="rstan.html#topic+stanmodel-method-sampling">rstan::sampling()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>stanfit</code>.
</p>


<h3>References</h3>

<p>Burgess, S., Butterworth, A., Thompson S.G. Mendelian randomization analysis with multiple genetic variants using summarized data. Genetic Epidemiology, 2013, 37, 7, 658-665 doi: <a href="https://doi.org/10.1002/gepi.21758">10.1002/gepi.21758</a>.
</p>
<p>Stan Development Team (2020). &quot;RStan: the R interface to Stan.&quot; R package version 2.19.3, <a href="https://mc-stan.org/">https://mc-stan.org/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("rstan", quietly = TRUE)) {
dat &lt;- mvmr_format(rsid = dodata$rsid,
          xbeta = cbind(dodata$ldlcbeta,dodata$hdlcbeta,dodata$tgbeta),
          ybeta = dodata$chdbeta,
          xse = cbind(dodata$ldlcse,dodata$hdlcse,dodata$tgse),
          yse = dodata$chdse)
mvivw_fit &lt;- mvmr_ivw_stan(dat)
print(mvivw_fit)
rstan::traceplot(mvivw_fit)
}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
