<!DOCTYPE html><html><head><title>Help for package mrMLM.GUI</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mrMLM.GUI}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#FASTmrEMMA'><p>To perform GWAS with FASTmrEMMA method</p></a></li>
<li><a href='#FASTmrMLM'><p>To perform GWAS with FASTmrMLM method</p></a></li>
<li><a href='#fmegen'>
<p>Genotype data</p></a></li>
<li><a href='#fmegenraw'>
<p>raw genotype data</p></a></li>
<li><a href='#ISIS'><p>To perform GWAS with ISIS EM-BLASSO method</p></a></li>
<li><a href='#mrgen'>
<p>Genotype data</p></a></li>
<li><a href='#mrgenraw'>
<p>raw genotype data</p></a></li>
<li><a href='#mrMLM.GUI-package'>
<p>Multi-Locus Random-SNP-Effect Mixed Linear Model Tools for Genome-Wide Association Study with Graphical User Interface</p></a></li>
<li><a href='#mrMLMFun'><p>To perform GWAS with mrMLM method</p></a></li>
<li><a href='#mrphe'>
<p>phenotype data</p></a></li>
<li><a href='#multiplication_speed'><p>Matrix multiplication acceleration algorithm.</p></a></li>
<li><a href='#pKWmEB'><p>To perform GWAS with pKWmEB method</p></a></li>
<li><a href='#pLARmEB'><p>To perform GWAS with pLARmEB method</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multi-Locus Random-SNP-Effect Mixed Linear Model Tools for
Genome-Wide Association Study with Graphical User Interface</td>
</tr>
<tr>
<td>Version:</td>
<td>4.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-10-8</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhang Ya-Wen, Li Pei, and Zhang Yuan-Ming </td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Conduct multi-locus genome-wide association study under the framework of multi-locus random-SNP-effect mixed linear model (mrMLM). First, each marker on the genome is scanned. Bonferroni correction is replaced by a less stringent selection criterion for significant test. Then, all the markers that are potentially associated with the trait are included in a multi-locus genetic model, their effects are estimated by empirical Bayes and all the nonzero effects were further identified by likelihood ratio test for true QTL. Wen YJ, Zhang H, Ni YL, Huang B, Zhang J, Feng JY, Wang SB, Dunwell JM, Zhang YM, Wu R (2018) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbw145">doi:10.1093/bib/bbw145</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0),shiny,lars</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge;
0.12.14),methods,foreach,ncvreg,coin,shinyjs,data.table,doParallel,sampling,bigmemory,mrMLM,sbl</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp,RcppEigen</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-10-08 03:57:06 UTC; &lt;e4&gt;&lt;ba&gt;&lt;9a&gt;&lt;e9&gt;&lt;9b&gt;&lt;af&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-10-08 10:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='FASTmrEMMA'>To perform GWAS with FASTmrEMMA method</h2><span id='topic+FASTmrEMMA'></span>

<h3>Description</h3>

<p>FAST multi-locus random-SNP-effect EMMA</p>


<h3>Usage</h3>

<pre><code class='language-R'>FASTmrEMMA(gen,phe,outATCG,genRaw,kk,psmatrix,svpal,svmlod,Genformat,Likelihood,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FASTmrEMMA_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_svpal">svpal</code></td>
<td>
<p>Critical P-value for selecting variable.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_svmlod">svmlod</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_likelihood">Likelihood</code></td>
<td>
<p>restricted maximum likelihood (REML) and maximum likelihood (ML).</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Li Pei, and Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(fmegen)
P1=data(mrphe)
G2=data(fmegenraw)
result=FASTmrEMMA(fmegen,mrphe,outATCG=NULL,fmegenraw,kk=NULL,
psmatrix=NULL,0.005,3,1,Likelihood="REML",CLO=1)
</code></pre>

<hr>
<h2 id='FASTmrMLM'>To perform GWAS with FASTmrMLM method</h2><span id='topic+FASTmrMLM'></span>

<h3>Description</h3>

<p>FAST multi-locus random-SNP-effect Mixed Linear Model</p>


<h3>Usage</h3>

<pre><code class='language-R'>FASTmrMLM(gen,phe,outATCG,genRaw,kk,psmatrix,svpal,svrad,svmlod,Genformat,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FASTmrMLM_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_svpal">svpal</code></td>
<td>
<p>Critical P-value for selecting variable.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_svrad">svrad</code></td>
<td>
<p>Search Radius in search of potentially associated QTN.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_svmlod">svmlod</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Li Pei, and Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(mrgen)
P1=data(mrphe)
G2=data(mrgenraw)
result=FASTmrMLM(mrgen,mrphe,outATCG=NULL,mrgenraw,kk=NULL,psmatrix=NULL,
0.01,20,3,1,CLO=1)
</code></pre>

<hr>
<h2 id='fmegen'>
Genotype data 
</h2><span id='topic+fmegen'></span><span id='topic+chr'></span><span id='topic+posi'></span><span id='topic+gen'></span>

<h3>Description</h3>

<p>Numeric format of genotype dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(fmegen)
</code></pre>


<h3>Details</h3>

<p>Dataset input of Genotype for FASTmrEMMA function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='fmegenraw'>
raw genotype data 
</h2><span id='topic+fmegenraw'></span><span id='topic+fmers+23'></span><span id='topic+fmechrom'></span><span id='topic+fmepos'></span>

<h3>Description</h3>

<p>Numeric format of raw genotype dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(fmegenraw)
</code></pre>


<h3>Details</h3>

<p>Dataset input of raw genotype for FASTmrEMMA function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='ISIS'>To perform GWAS with ISIS EM-BLASSO method</h2><span id='topic+ISIS'></span>

<h3>Description</h3>

<p>Iterative Sure Independence Screening EM-Bayesian LASSO</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISIS(gen,phe,outATCG,genRaw,kk,psmatrix,svpal,svmlod,Genformat,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ISIS_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_svpal">svpal</code></td>
<td>
<p>Critical P-value for selecting variable.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_svmlod">svmlod</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Li Pei, and Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(mrgen)
P1=data(mrphe)
G2=data(mrgenraw)
result=ISIS(mrgen,mrphe,outATCG=NULL,mrgenraw,kk=NULL,psmatrix=NULL,
0.01,3,1,CLO=1)
</code></pre>

<hr>
<h2 id='mrgen'>
Genotype data 
</h2><span id='topic+mrgen'></span><span id='topic+chromosome'></span><span id='topic+position'></span><span id='topic+genotype'></span>

<h3>Description</h3>

<p>Numeric format of genotype dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(mrgen)
</code></pre>


<h3>Details</h3>

<p>Dataset input of Genotype for mrMLM function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='mrgenraw'>
raw genotype data 
</h2><span id='topic+mrgenraw'></span><span id='topic+rs+23'></span><span id='topic+chrom'></span><span id='topic+pos'></span>

<h3>Description</h3>

<p>Numeric format of raw genotype dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(mrgenraw)
</code></pre>


<h3>Details</h3>

<p>Dataset input of raw genotype for mrMLM function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='mrMLM.GUI-package'>
Multi-Locus Random-SNP-Effect Mixed Linear Model Tools for Genome-Wide Association Study with Graphical User Interface
</h2><span id='topic+mrMLM.GUI-package'></span><span id='topic+mrMLM.GUI'></span>

<h3>Description</h3>

<p>Description: Conduct multi-locus genome-wide association study under the framework of multi-locus random-SNP-effect mixed linear model (mrMLM). First, each marker on the genome is scanned. Bonferroni correction is replaced by a less stringent selection criterion for significant test. Then, all the markers that are potentially associated with the trait are included in a multi-locus genetic model, their effects are estimated by empirical Bayes and all the nonzero effects were further identified by likelihood ratio test for ture QTL. 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> mrMLM.GUI</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 4.0.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-10-8</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> shiny,lars</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> methods,foreach,ncvreg,coin,sampling,data.table,doParallel,shinyjs,bigmemory,mrMLM</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL version 2 or newer</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Users can use library(mrMLM.GUI) to start the GUI and use 'mrMLM.GUI()' to restart the programe.
</p>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Li Pei, Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>References</h3>

<p>1. Zhang YM, Mao Y, Xie C, Smith H, Luo L, Xu S*. Genetics 2005,169:2267-2275.
2. Wang SB, Feng JY, Ren WL, Huang B, Zhou L, Wen YJ, et al. Sci Rep 2016,6:19444.
3. Tamba CL, Ni YL, Zhang YM*. PLoS Comput Biol 2017,13(1):e1005357.
4. Zhang J, Feng JY, Ni YL, Wen YJ, Niu Y, Tamba CL, et al. Heredity 2018,118(6):517-524.
5. Ren WL, Wen YJ, Dunwell JM, Zhang YM*. Heredity 2018,120(3): 208-218.
6. Wen YJ, Zhang H, Ni YL, Huang B, Zhang J, Feng JY, et al. Brief Bioinform 2018,19(4):700-712.
7. Tamba CL, Zhang YM. bioRxiv,preprint first posted online Jun. 7, 2018, doi:https://doi.org/10.1101/341784.
8. Zhang YW, Tamba CL, Wen YJ, Li P, Ren WL, Ni YL, et al. Genomics, Proteomics &amp; Bioinformatics, Accept.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:  mrMLM.GUI() 
</code></pre>

<hr>
<h2 id='mrMLMFun'>To perform GWAS with mrMLM method</h2><span id='topic+mrMLMFun'></span>

<h3>Description</h3>

<p>multi-locus random-SNP-effect Mixed Linear Model</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrMLMFun(gen,phe,outATCG,genRaw,kk,psmatrix,svpal,svrad,svmlod,Genformat,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrMLMFun_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_svpal">svpal</code></td>
<td>
<p>Critical P-value for selecting variable</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_svrad">svrad</code></td>
<td>
<p>Search Radius in search of potentially associated QTN.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_svmlod">svmlod</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Li Pei, and Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(mrgen)
P1=data(mrphe)
G2=data(mrgenraw)
result=mrMLMFun(mrgen,mrphe,outATCG=NULL,mrgenraw,kk=NULL,psmatrix=NULL,
0.01,20,3,1,CLO=1)
</code></pre>

<hr>
<h2 id='mrphe'>
phenotype data 
</h2><span id='topic+mrphe'></span><span id='topic+phenotype'></span>

<h3>Description</h3>

<p>phenotype dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(mrphe)
</code></pre>


<h3>Details</h3>

<p>Dataset input of phenotype for mrMLM function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='multiplication_speed'>Matrix multiplication acceleration algorithm.</h2><span id='topic+multiplication_speed'></span>

<h3>Description</h3>

<p>Matrix multiplication acceleration algorithm.</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiplication_speed(A,B)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiplication_speed_+3A_a">A</code></td>
<td>
<p>matrix A.</p>
</td></tr>
<tr><td><code id="multiplication_speed_+3A_b">B</code></td>
<td>
<p>matrix B.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wen Yang-Jun, Wang Shi-Bo, and Zhang Yuan-Ming<br />
Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
A&lt;-matrix(1:10,2,5)
B&lt;-matrix(1:10,5:2)
result&lt;-multiplication_speed(A,B)

## End(Not run)
</code></pre>

<hr>
<h2 id='pKWmEB'>To perform GWAS with pKWmEB method</h2><span id='topic+pKWmEB'></span>

<h3>Description</h3>

<p>Kruskal-Wallis test with empirical Bayes under polygenic background control</p>


<h3>Usage</h3>

<pre><code class='language-R'>pKWmEB(gen,phe,outATCG,genRaw,kk,psmatrix,svpal,svmlod,Genformat,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pKWmEB_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_svpal">svpal</code></td>
<td>
<p>Critical P-value for selecting variable.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_svmlod">svmlod</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Li Pei, and Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(mrgen)
P1=data(mrphe)
G2=data(mrgenraw)
result=pKWmEB(mrgen,mrphe,outATCG=NULL,mrgenraw,kk=NULL,psmatrix=NULL,
0.05,3,1,CLO=1)
</code></pre>

<hr>
<h2 id='pLARmEB'>To perform GWAS with pLARmEB method</h2><span id='topic+pLARmEB'></span>

<h3>Description</h3>

<p>polygene-background-control-based least angle regression plus Empirical Bayes</p>


<h3>Usage</h3>

<pre><code class='language-R'>pLARmEB(gen,phe,outATCG,genRaw,kk,psmatrix,CriLOD,lars1,Genformat,Bootstrap,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pLARmEB_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_lars1">lars1</code></td>
<td>
<p>No. of potentially associated variables selected by LARS.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_bootstrap">Bootstrap</code></td>
<td>
<p>Bootstrap=FALSE indicates the analysis of only real dataset, Bootstrap=TRUE indicates the analysis of both real dataset and four resampling datasets.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Li Pei, and Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(mrgen)
P1=data(mrphe)
G2=data(mrgenraw)
result=pLARmEB(mrgen,mrphe,outATCG=NULL,mrgenraw,kk=NULL,psmatrix=NULL,
3,20,1,Bootstrap=FALSE,CLO=1)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
