<!DOCTYPE html><html><head><title>Help for package mrMLM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mrMLM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#DoData'><p>process raw data</p></a></li>
<li><a href='#FASTmrEMMA'><p>To perform GWAS with FASTmrEMMA method</p></a></li>
<li><a href='#FASTmrMLM'><p>To perform GWAS with FASTmrMLM method</p></a></li>
<li><a href='#Gen'>
<p>Genotype data</p></a></li>
<li><a href='#Genotype '>
<p>Genotype of real data</p></a></li>
<li><a href='#inputData'><p>Input data which have been transformed</p></a></li>
<li><a href='#ISIS'><p>To perform GWAS with ISIS EM-BLASSO method</p></a></li>
<li><a href='#mrMLM'><p>Multi-Locus Random-SNP-Effect Mixed Linear Model Tools for GWAS</p></a></li>
<li><a href='#mrMLMFun'><p>To perform GWAS with mrMLM method</p></a></li>
<li><a href='#MultiManhattan'><p>Drawing multi-locus Manhattan plot</p></a></li>
<li><a href='#multiplication_speed'><p>Matrix multiplication acceleration algorithm.</p></a></li>
<li><a href='#Phe'>
<p>Phenotype dataset</p></a></li>
<li><a href='#Phenotype'>
<p>Phenotype of real data</p></a></li>
<li><a href='#pKWmEB'><p>To perform GWAS with pKWmEB method</p></a></li>
<li><a href='#pLARmEB'><p>To perform GWAS with pLARmEB method</p></a></li>
<li><a href='#ReadData'><p>read raw data</p></a></li>
<li><a href='#ResultFinal'>
<p>Final result used to draw manhattan plot.</p></a></li>
<li><a href='#ResultIntermediate'>
<p>Intermediate result used to draw manhattan plot.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multi-Locus Random-SNP-Effect Mixed Linear Model Tools for GWAS</td>
</tr>
<tr>
<td>Version:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-3-27</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuan-Ming Zhang &lt;soyzhang@mail.hzau.edu.cn&gt;</td>
</tr>
<tr>
<td>Contact:</td>
<td>Yuan-Ming Zhang &lt;soyzhang@mail.hzau.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Conduct multi-locus genome-wide association study under the framework of multi-locus random-SNP-effect mixed linear model (mrMLM). First, each marker on the genome is scanned. Bonferroni correction is replaced by a less stringent selection criterion for significant test. Then, all the markers that are potentially associated with the trait are included in a multi-locus genetic model, their effects are estimated by empirical Bayes, and all the nonzero effects were further identified by likelihood ratio test for significant QTL. The program may run on a desktop or laptop computers. If marker genotypes in association mapping population are almost homozygous, these methods in this software are very effective. If there are many heterozygous marker genotypes, the IIIVmrMLM software is recommended. Wen YJ, Zhang H, Ni YL, Huang B, Zhang J, Feng JY, Wang SB, Dunwell JM, Zhang YM, Wu R (2018, &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbw145">doi:10.1093/bib/bbw145</a>&gt;), and Li M, Zhang YW, Zhang ZC, Xiang Y, Liu MH, Zhou YH, Zuo JF, Zhang HQ, Chen Y, Zhang YM (2022, &lt;<a href="https://doi.org/10.1016%2Fj.molp.2022.02.012">doi:10.1016/j.molp.2022.02.012</a>&gt;).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0),lars</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.14),methods,foreach,ncvreg,coin(&ge;
1.1-0),sampling,data.table,doParallel,sbl,BEDMatrix</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-03-27 08:03:33 UTC; Administrator</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-03-27 08:50:02 UTC</td>
</tr>
<tr>
<td>Author:</td>
<td>Ya-Wen Zhang [aut],
  Jing-Tian Wang [aut],
  Pei Li [aut],
  Yuan-Ming Zhang <a href="https://orcid.org/0000-0003-2317-2190"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
</table>
<hr>
<h2 id='DoData'>process raw data</h2><span id='topic+DoData'></span>

<h3>Description</h3>

<p>process raw data for later use</p>


<h3>Usage</h3>

<pre><code class='language-R'>DoData(genRaw,Genformat,pheRaw1q,kkRaw,psmatrixRaw,covmatrixRaw,trait,
type,PopStrType)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DoData_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype matrix.</p>
</td></tr>
<tr><td><code id="DoData_+3A_genformat">Genformat</code></td>
<td>
<p>genotype format.</p>
</td></tr>
<tr><td><code id="DoData_+3A_pheraw1q">pheRaw1q</code></td>
<td>
<p>raw phenotype matrix.</p>
</td></tr>
<tr><td><code id="DoData_+3A_kkraw">kkRaw</code></td>
<td>
<p>raw kinship matrix.</p>
</td></tr>
<tr><td><code id="DoData_+3A_psmatrixraw">psmatrixRaw</code></td>
<td>
<p>raw population structure matrix.</p>
</td></tr>
<tr><td><code id="DoData_+3A_covmatrixraw">covmatrixRaw</code></td>
<td>
<p>raw covariate matrix.</p>
</td></tr>
<tr><td><code id="DoData_+3A_trait">trait</code></td>
<td>
<p>which trait to analysis.</p>
</td></tr> 
<tr><td><code id="DoData_+3A_type">type</code></td>
<td>
<p>which type to transform.</p>
</td></tr>
<tr><td><code id="DoData_+3A_popstrtype">PopStrType</code></td>
<td>
<p>The type of population structure.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wang Jing-Tian, Li Pei, Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(Gen)
P1=data(Phe)
readraw=ReadData(fileGen=Gen,filePhe=Phe,fileKin=NULL,filePS =NULL,
fileCov=NULL,Genformat=1)
result=DoData(readraw$genRaw,Genformat=1,readraw$pheRaw1q,readraw$kkRaw,
readraw$psmatrixRaw,readraw$covmatrixRaw,trait=1,type=2,PopStrType=NULL)
</code></pre>

<hr>
<h2 id='FASTmrEMMA'>To perform GWAS with FASTmrEMMA method</h2><span id='topic+FASTmrEMMA'></span>

<h3>Description</h3>

<p>FAST multi-locus random-SNP-effect EMMA</p>


<h3>Usage</h3>

<pre><code class='language-R'>FASTmrEMMA(gen,phe,outATCG,genRaw,kk,psmatrix,svpal,svmlod,Genformat,Likelihood,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FASTmrEMMA_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_svpal">svpal</code></td>
<td>
<p>Critical P-value for selecting variable.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_svmlod">svmlod</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_likelihood">Likelihood</code></td>
<td>
<p>restricted maximum likelihood (REML) and maximum likelihood (ML).</p>
</td></tr>
<tr><td><code id="FASTmrEMMA_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wang Jing-Tian, Li Pei, Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(Gen)
P1=data(Phe)
Readraw=ReadData(fileGen=Gen,filePhe=Phe,fileKin=NULL,filePS =NULL,
Genformat=1)
InputData=inputData(readraw=Readraw,Genformat=1,method="FASTmrEMMA",trait=1)
result=FASTmrEMMA(InputData$doFME$gen,InputData$doFME$phe,
InputData$doFME$outATCG,InputData$doFME$genRaw,
InputData$doFME$kk,InputData$doFME$psmatrix,0.005,
svmlod=3,Genformat=1,Likelihood="REML",CLO=1)   
</code></pre>

<hr>
<h2 id='FASTmrMLM'>To perform GWAS with FASTmrMLM method</h2><span id='topic+FASTmrMLM'></span>

<h3>Description</h3>

<p>FAST multi-locus random-SNP-effect Mixed Linear Model</p>


<h3>Usage</h3>

<pre><code class='language-R'>FASTmrMLM(gen,phe,outATCG,genRaw,kk,psmatrix,svpal,svrad,svmlod,Genformat,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FASTmrMLM_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_svpal">svpal</code></td>
<td>
<p>Critical P-value for selecting variable.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_svrad">svrad</code></td>
<td>
<p>Search Radius in search of potentially associated QTN.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_svmlod">svmlod</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="FASTmrMLM_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wang Jing-Tian, Li Pei, Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(Gen)
P1=data(Phe)
Readraw=ReadData(fileGen=Gen,filePhe=Phe,fileKin=NULL,filePS =NULL,
Genformat=1)
InputData=inputData(readraw=Readraw,Genformat=1,method="FASTmrMLM",trait=1)
result=FASTmrMLM(InputData$doMR$gen,InputData$doMR$phe,
InputData$doMR$outATCG,InputData$doMR$genRaw,
InputData$doMR$kk,InputData$doMR$psmatrix,0.01,svrad=20,
svmlod=3,Genformat=1,CLO=1)   
</code></pre>

<hr>
<h2 id='Gen'>
Genotype data 
</h2><span id='topic+Gen'></span><span id='topic+rs+23'></span><span id='topic+chrom'></span><span id='topic+pos'></span>

<h3>Description</h3>

<p>Numeric format of genotype dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(Gen)
</code></pre>


<h3>Details</h3>

<p>Dataset input of Genotype for mrMLM function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='Genotype+20'>
Genotype of real data 
</h2><span id='topic+Genotype'></span><span id='topic+33-16'></span><span id='topic+Nov-38'></span><span id='topic+A4226'></span>

<h3>Description</h3>

<p>Numeric format of genotype dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(Genotype)
</code></pre>


<h3>Details</h3>

<p>Dataset input of Genotype for mrMLM function. 
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='inputData'>Input data which have been transformed</h2><span id='topic+inputData'></span>

<h3>Description</h3>

<p>Input all the dataset which have been transformed</p>


<h3>Usage</h3>

<pre><code class='language-R'>inputData(readraw,Genformat,method,trait,PopStrType)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="inputData_+3A_readraw">readraw</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="inputData_+3A_genformat">Genformat</code></td>
<td>
<p>genotype format.</p>
</td></tr>
<tr><td><code id="inputData_+3A_method">method</code></td>
<td>
<p>which method to analysis.</p>
</td></tr>
<tr><td><code id="inputData_+3A_trait">trait</code></td>
<td>
<p>which trait to analysis.</p>
</td></tr>
<tr><td><code id="inputData_+3A_popstrtype">PopStrType</code></td>
<td>
<p>The type of population structure.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wang Jing-Tian, Li Pei, Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(Gen)
P1=data(Phe)
Readraw=ReadData(fileGen=Gen,filePhe=Phe,fileKin=NULL,filePS =NULL,
fileCov=NULL,Genformat=1)
result=inputData(readraw=Readraw,Genformat=1,method="mrMLM",trait=1,
PopStrType=NULL)
</code></pre>

<hr>
<h2 id='ISIS'>To perform GWAS with ISIS EM-BLASSO method</h2><span id='topic+ISIS'></span>

<h3>Description</h3>

<p>Iterative Sure Independence Screening EM-Bayesian LASSO</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISIS(gen,phe,outATCG,genRaw,kk,psmatrix,svpal,svmlod,Genformat,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ISIS_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_svpal">svpal</code></td>
<td>
<p>Critical P-value for selecting variable.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_svmlod">svmlod</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="ISIS_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Li Pei, Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(Gen)
P1=data(Phe)
Readraw=ReadData(fileGen=Gen,filePhe=Phe,fileKin=NULL,filePS =NULL,
Genformat=1)
InputData=inputData(readraw=Readraw,Genformat=1,method="ISIS EM-BLASSO",
trait=1)
result=ISIS(InputData$doMR$gen,InputData$doMR$phe,InputData$doMR$outATCG,
InputData$doMR$genRaw,InputData$doMR$kk,InputData$doMR$psmatrix,
0.01,svmlod=3,Genformat=1,CLO=1) 
</code></pre>

<hr>
<h2 id='mrMLM'>Multi-Locus Random-SNP-Effect Mixed Linear Model Tools for GWAS 
</h2><span id='topic+mrMLM'></span><span id='topic+mrMLM'></span>

<h3>Description</h3>

<p>Conduct multi-locus genome-wide association study under the framework of multi-locus random-SNP-effect mixed linear model (mrMLM). First, each marker on the genome is scanned. Bonferroni correction is replaced by a less stringent selection criterion for significant test. Then, all the markers that are potentially associated with the trait are included in a multi-locus genetic model, their effects are estimated by empirical Bayes, and all the nonzero effects were further identified by likelihood ratio test for true QTL. The program may run on a desktop or laptop computers. If marker genotypes in association mapping population are almost homozygous, these methods in this software are very effective. If there are many heterozygous marker genotypes, the IIIVmrMLM software is recommended. Wen YJ, Zhang H, Ni YL, Huang B, Zhang J, Feng JY, Wang SB, Dunwell JM, Zhang YM, Wu R (2018, &lt;doi:10.1093/bib/bbw145&gt;), and Li M, Zhang YW, Zhang ZC, Xiang Y, Liu MH, Zhou YH, Zuo JF, Zhang HQ, Chen Y, Zhang YM (2022, &lt;doi:10.1016/j.molp.2022.02.012&gt;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrMLM(fileGen,filePhe,fileKin,filePS,PopStrType,fileCov,Genformat,
method,Likelihood,trait,SearchRadius,CriLOD,SelectVariable,Bootstrap,
DrawPlot,Plotformat,dir,PC,RAM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrMLM_+3A_filegen">fileGen</code></td>
<td>
<p>File path and name in your computer of Genotype,
i.e.,&quot;D:/Users/Genotype_num.csv&quot;.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_filephe">filePhe</code></td>
<td>
<p>File path and name in your computer of Phenotype,
i.e.,&quot;D:/Users/Phenotype.csv&quot;.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_filekin">fileKin</code></td>
<td>
<p>File path and name in your computer of Kinship,
i.e.,&quot;D:/Users/Kinship.csv&quot;.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_fileps">filePS</code></td>
<td>
<p>File path and name in your computer of Population 
Structure,i.e.,&quot;D:/Users/PopStr.csv&quot;.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_popstrtype">PopStrType</code></td>
<td>
<p>The type of population structure,i.e.,Q (Q matrix), 
PCA (principal components), EvolPopStr (evolutionary population 
structure).</p>
</td></tr>                    
<tr><td><code id="mrMLM_+3A_filecov">fileCov</code></td>
<td>
<p>File path and name in your computer of covariate,  
i.e.,&quot;D:/Users/Covariate.csv&quot;.</p>
</td></tr> 
<tr><td><code id="mrMLM_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes, Num (number), Cha   
(character) and Hmp (Hapmap).</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_method">method</code></td>
<td>
<p>Six multi-locus GWAS methods. Users may select one to 
six methods, including mrMLM, FASTmrMLM, FASTmrEMMA, pLARmEB, pKWmEB 
and ISIS EM-BLASSO.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_likelihood">Likelihood</code></td>
<td>
<p>This parameter is only for FASTmrEMMA, including 
REML(restricted maximum likelihood) and  ML(maximum likelihood).</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_trait">trait</code></td>
<td>
<p>Traits analyzed from number 1 to number 2,i.e.,1:2.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_searchradius">SearchRadius</code></td>
<td>
<p>This parameter is only for mrMLM and FASTmrMLM, 
indicating Search Radius in search of potentially associated QTN,the 
default is 20.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr> 
<tr><td><code id="mrMLM_+3A_selectvariable">SelectVariable</code></td>
<td>
<p>This parameter is only for pLARmEB. 
SelectVariable=50 indicates that 50 potentially associated variables 
are selected from each chromosome. Users may change this number in 
real data analysis in order to obtain the best results as final 
results,the default is 50.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_bootstrap">Bootstrap</code></td>
<td>
<p>This parameter is only for pLARmEB, including FASLE 
and TRUE, Bootstrap=FALSE indicates the analysis of only real 
dataset, Bootstrap=TRUE indicates the analysis of both real dataset 
and four resampling datasets,the default is FALSE.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_drawplot">DrawPlot</code></td>
<td>
<p>This parameter is for all the six methods, including 
FALSE and TRUE, DrawPlot=FALSE indicates no figure output, 
DrawPlot=TRUE indicates the output of the Manhattan, QQ figures,the 
default is TRUE.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_plotformat">Plotformat</code></td>
<td>
<p>This parameter is for all the figure files,
including *.jpeg, *.png, *.tiff and *.pdf,the default is &quot;tiff&quot;.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_dir">dir</code></td>
<td>
<p>This parameter is for the save path,i.e.,&quot;D:/Users&quot;</p>
</td></tr> 
<tr><td><code id="mrMLM_+3A_pc">PC</code></td>
<td>
<p>This parameter is used to specify 
whether only small RAM device is available to run the mrMLM program, 
such as desktop or laptop. The default value is PC=FALSE. 
PC=TRUE indicates running the program on low RAM desktop or laptop.</p>
</td></tr>
<tr><td><code id="mrMLM_+3A_ram">RAM</code></td>
<td>
<p>This parameter is the RAM of your desktop or laptop. 
The default value is RAM=4. RAM=4 indicates the RAM of your device is 4G.</p>
</td></tr>
</table>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> mrMLM</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 5.0.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2022-3-27</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> lars</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> methods,foreach,ncvreg,coin,sampling,data.table,doParallel,BEDMatrix
</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL version 2 or newer</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Note</h3>

<p>Once the running of the software mrMLM v5.0.1 is ended, 
the &quot;results&quot; files should appear on the Directory, which was 
set up by users before running the software. The results for 
each trait include &quot;*_intermediate result.csv&quot;, &quot;*_Final result.csv&quot;, 
Manhattan plot, and QQ plot. If only pLARmEB and ISIS EM-BLASSO
methods are selected, there will be no intermediate results and 
figures output. Users can decompress the mrMLM package and find 
the User Manual file (name: Instruction.pdf) in the folder of 
&quot;.../mrMLM/inst&quot;.
</p>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wang Jing-Tian, Li Pei, Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>References</h3>

<p>1. Zhang YM, Mao Y, Xie C, Smith H, Luo L, Xu S. Genetics 2005,169:2267-2275.
2. Wang SB, Feng JY, Ren WL, Huang B, Zhou L, Wen YJ, Zhang J, Dunwell JM, Xu S, Zhang YM. Sci Rep 2016,6:19444.
3. Tamba CL, Ni YL, Zhang YM. PLoS Comput Biol 2017,13(1):e1005357.
4. Zhang J, Feng JY, Ni YL, Wen YJ, Niu Y, Tamba CL, Yue C, Song Q, Zhang YM. Heredity 2018,118(6):517-524.
5. Ren WL, Wen YJ, Dunwell JM, Zhang YM. Heredity 2018,120(3): 208-218.
6. Wen YJ, Zhang H, Ni YL, Huang B, Zhang J, Feng JY, Wang SB, Dunwell JM, Zhang YM, Wu R. Brief Bioinform 2018,19(4): 700-712.
7. Tamba CL, Zhang YM. bioRxiv,preprint first posted online Jun. 7, 2018,
doi:https://doi.org/10.1101/341784.
8. Zhang YW, Tamba CL, Wen YJ, Li P, Ren WL, Ni YL, Gao J, Zhang YM. Genomics, Proteomics &amp; Bioinformatics 2020, 18: 481-487.
9.Li M, Zhang YW, Zhang ZC, Xiang Y, Liu MH, Zhou YH, Zuo JF, Zhang HQ, Chen Y, Zhang YM. 
A compressed variance component mixed model for detecting QTNs, 
and QTN-by-environment and QTN-by-QTN interactions in genome-wide association studies. 
Molecular Plant 2022, online, S1674-2052(22)00060-0. doi: 10.1016/j.molp.2022.02.012.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Ge1=data(Genotype)
Ph1=data(Phenotype)
mrMLM(fileGen=Genotype,filePhe=Phenotype,Genformat="Num",
method=c("FASTmrMLM"),trait=1,CriLOD=3,DrawPlot=FALSE,
dir=tempdir(),PC=FALSE,RAM=4)
</code></pre>

<hr>
<h2 id='mrMLMFun'>To perform GWAS with mrMLM method</h2><span id='topic+mrMLMFun'></span>

<h3>Description</h3>

<p>multi-locus random-SNP-effect Mixed Linear Model</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrMLMFun(gen,phe,outATCG,genRaw,kk,psmatrix,svpal,svrad,svmlod,Genformat,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrMLMFun_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_svpal">svpal</code></td>
<td>
<p>Critical P-value for selecting variable</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_svrad">svrad</code></td>
<td>
<p>Search Radius in search of potentially associated QTN.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_svmlod">svmlod</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="mrMLMFun_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wang Jing-Tian, Li Pei, Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(Gen)
P1=data(Phe)
Readraw=ReadData(fileGen=Gen,filePhe=Phe,fileKin=NULL,filePS =NULL,
Genformat=1)
InputData=inputData(readraw=Readraw,Genformat=1,method="mrMLM",trait=1)
result=mrMLMFun(InputData$doMR$gen,InputData$doMR$phe,InputData$doMR$outATCG,
InputData$doMR$genRaw,InputData$doMR$kk,InputData$doMR$psmatrix,
0.01,svrad=20,svmlod=3,Genformat=1,CLO=1)
</code></pre>

<hr>
<h2 id='MultiManhattan'>Drawing multi-locus Manhattan plot</h2><span id='topic+MultiManhattan'></span>

<h3>Description</h3>

<p>Using the results of the mrMLM software to draw a multi-locus Manhattan plot </p>


<h3>Usage</h3>

<pre><code class='language-R'>MultiManhattan(ResultIntermediate,ResultFinal,mar=c(2.9,2.8,0.7,2.8), 
LabDistance=1.5,ScaleDistance=0.4,LabelSize=0.8,ScaleSize=0.7,
AxisLwd=5,TckLength=-0.03,LogTimes=2,LODTimes=1.2,lodline=3, 
dirplot=getwd(), PlotFormat="tiff", 
width=28000,height=7000,pointsize = 60,res=600,
MarkGene=FALSE,Pos_x=NULL,Pos_y=NULL,GeneName=NULL,
GeneNameColour=NULL,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MultiManhattan_+3A_resultintermediate">ResultIntermediate</code></td>
<td>
<p>Intermediate results obtained by the mrMLM
software,&quot;D:/Users/ResultIntermediate.csv&quot;.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_resultfinal">ResultFinal</code></td>
<td>
<p>Final results obtained by the mrMLM 
software,&quot;D:/Users/ResultFinal.csv&quot;.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_mar">mar</code></td>
<td>
<p>A numerical vector of the form c(bottom, left, top, 
right) which gives the number of lines of margin to be specified on 
the four sides of the plot, and the default is c(2.9, 2.8, 0.7, 
2.8).</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_labdistance">LabDistance</code></td>
<td>
<p>Distance between label and axis; the default is  
1.5.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_scaledistance">ScaleDistance</code></td>
<td>
<p>Distance between scale values and axis; the 
default is 0.4.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_labelsize">LabelSize</code></td>
<td>
<p>Size of all the three labels; the default is 0.8.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_scalesize">ScaleSize</code></td>
<td>
<p>Size of scale values; the default is 0.7.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_axislwd">AxisLwd</code></td>
<td>
<p>The width of axis, a positive number; the default 
is 5.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_tcklength">TckLength</code></td>
<td>
<p>The length of tick marks; the default is -0.03.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_logtimes">LogTimes</code></td>
<td>
<p>Magnification of -log10(P-value); the default is
2.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_lodtimes">LODTimes</code></td>
<td>
<p>Magnification of LOD score; the default is 1.2.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_lodline">lodline</code></td>
<td>
<p>The significant LOD score; the default is 3.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_dirplot">dirplot</code></td>
<td>
<p>Path to save plot; the default is current working
directory</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_plotformat">PlotFormat</code></td>
<td>
<p>Format of the plot.i.e., *.tiff, *.png, *.jpeg, 
*.pdf</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_width">width</code></td>
<td>
<p>Figure width; the default is 28000.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_height">height</code></td>
<td>
<p>Figure height; the default is 7000.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_pointsize">pointsize</code></td>
<td>
<p>Word resolution, with the unit of 1/72 inch, being 
pixels per inch (ppi); the default is 60.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_res">res</code></td>
<td>
<p>Figure resolution, with the unit of pixels per inch 
(ppi); the default is 600.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_markgene">MarkGene</code></td>
<td>
<p>To mark genes in plot or not; if &quot;TRUE&quot; is selected, a  
file, namely &quot;Reference information to mark gene.csv&quot;, that contains 
the x and y axis information of all the significant QTNs will  
generate. The default is &quot;FALSE&quot;, indicating that no candidate or  
known gene names are marked in Manhattan plot.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_pos_x">Pos_x</code></td>
<td>
<p>Numeric vectors of x axis where the text labels   
should be written.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_pos_y">Pos_y</code></td>
<td>
<p>Numeric vectors of y axis where the text labels 
should be written.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_genename">GeneName</code></td>
<td>
<p>A character vector or expression specifying the text 
to be written.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_genenamecolour">GeneNameColour</code></td>
<td>
<p>The colour of gene names.</p>
</td></tr>
<tr><td><code id="MultiManhattan_+3A_...">...</code></td>
<td>
<p>Arguments passed to points, axis, text.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wang Jing-Tian, Li Pei, and Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>inter&lt;-data(ResultIntermediate)
fin&lt;-data(ResultFinal)
MultiManhattan(ResultIntermediate=ResultIntermediate,ResultFinal=ResultFinal,dirplot=tempdir())
</code></pre>

<hr>
<h2 id='multiplication_speed'>Matrix multiplication acceleration algorithm.</h2><span id='topic+multiplication_speed'></span>

<h3>Description</h3>

<p>Matrix multiplication acceleration algorithm.</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiplication_speed(A,B)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiplication_speed_+3A_a">A</code></td>
<td>
<p>matrix A.</p>
</td></tr>
<tr><td><code id="multiplication_speed_+3A_b">B</code></td>
<td>
<p>matrix B.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wen Yang-Jun, Wang Shi-Bo, and Zhang Yuan-Ming<br />
Maintainer: Yuanming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
A&lt;-matrix(1:10,2,5)
B&lt;-matrix(1:10,5:2)
result&lt;-multiplication_speed(A,B)

## End(Not run)
</code></pre>

<hr>
<h2 id='Phe'>
Phenotype dataset    
</h2><span id='topic+Phe'></span><span id='topic++3Cphenotype+3E'></span><span id='topic+rep-1'></span><span id='topic+rep-2'></span><span id='topic+rep-3'></span>

<h3>Description</h3>

<p>Phenotype dataset of multiple traits. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(Phe)
</code></pre>


<h3>Details</h3>

<p>Dataset input of phenotype in mrMLM function.
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='Phenotype'>
Phenotype of real data   
</h2><span id='topic+Phenotype'></span><span id='topic+trait1'></span><span id='topic+trait2'></span><span id='topic+trait3'></span>

<h3>Description</h3>

<p>Phenotype dataset of multiple traits. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  data(Phenotype)
</code></pre>


<h3>Details</h3>

<p>Dataset input of phenotype in mrMLM function.
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='pKWmEB'>To perform GWAS with pKWmEB method</h2><span id='topic+pKWmEB'></span>

<h3>Description</h3>

<p>Kruskal-Wallis test with empirical Bayes under polygenic background control</p>


<h3>Usage</h3>

<pre><code class='language-R'>pKWmEB(gen,phe,outATCG,genRaw,kk,psmatrix,svpal,svmlod,Genformat,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pKWmEB_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_svpal">svpal</code></td>
<td>
<p>Critical P-value for selecting variable.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_svmlod">svmlod</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="pKWmEB_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wang Jing-Tian, Li Pei, Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(Gen)
P1=data(Phe)
Readraw=ReadData(fileGen=Gen,filePhe=Phe,fileKin=NULL,filePS =NULL,
Genformat=1)
InputData=inputData(readraw=Readraw,Genformat=1,method="pKWmEB",trait=1)
result=pKWmEB(InputData$doMR$gen,InputData$doMR$phe,InputData$doMR$outATCG,
InputData$doMR$genRaw,InputData$doMR$kk,InputData$doMR$psmatrix,
0.05,svmlod=3,Genformat=1,CLO=1)  
</code></pre>

<hr>
<h2 id='pLARmEB'>To perform GWAS with pLARmEB method</h2><span id='topic+pLARmEB'></span>

<h3>Description</h3>

<p>polygene-background-control-based least angle regression plus Empirical Bayes</p>


<h3>Usage</h3>

<pre><code class='language-R'>pLARmEB(gen,phe,outATCG,genRaw,kk,psmatrix,CriLOD,lars1,Genformat,Bootstrap,CLO)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pLARmEB_+3A_gen">gen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_phe">phe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_outatcg">outATCG</code></td>
<td>
<p>genotype for code 1.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_genraw">genRaw</code></td>
<td>
<p>raw genotype.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_kk">kk</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_psmatrix">psmatrix</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_crilod">CriLOD</code></td>
<td>
<p>Critical LOD score for significant QTN.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_lars1">lars1</code></td>
<td>
<p>No. of potentially associated variables selected by LARS.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_genformat">Genformat</code></td>
<td>
<p>Format for genotypic codes.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_bootstrap">Bootstrap</code></td>
<td>
<p>Bootstrap=FALSE indicates the analysis of only real dataset, Bootstrap=TRUE indicates the analysis of both real dataset and four resampling datasets.</p>
</td></tr>
<tr><td><code id="pLARmEB_+3A_clo">CLO</code></td>
<td>
<p>number of CPU.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wang Jing-Tian, Li Pei, Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(Gen)
P1=data(Phe)
Readraw=ReadData(fileGen=Gen,filePhe=Phe,fileKin=NULL,filePS =NULL,
Genformat=1)
InputData=inputData(readraw=Readraw,Genformat=1,method="pLARmEB",trait=1)
result=pLARmEB(InputData$doMR$gen,InputData$doMR$phe,InputData$doMR$outATCG,
InputData$doMR$genRaw,InputData$doMR$kk,InputData$doMR$psmatrix,
CriLOD=3,lars1=20,Genformat=1,Bootstrap=FALSE,CLO=1)   
</code></pre>

<hr>
<h2 id='ReadData'>read raw data</h2><span id='topic+ReadData'></span>

<h3>Description</h3>

<p>read raw data which have not been transformed</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadData(fileGen,filePhe,fileKin,filePS,fileCov,Genformat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadData_+3A_filegen">fileGen</code></td>
<td>
<p>genotype matrix.</p>
</td></tr>
<tr><td><code id="ReadData_+3A_filephe">filePhe</code></td>
<td>
<p>phenotype matrix.</p>
</td></tr>
<tr><td><code id="ReadData_+3A_filekin">fileKin</code></td>
<td>
<p>kinship matrix.</p>
</td></tr>
<tr><td><code id="ReadData_+3A_fileps">filePS</code></td>
<td>
<p>population structure matrix.</p>
</td></tr>
<tr><td><code id="ReadData_+3A_filecov">fileCov</code></td>
<td>
<p>Covariate matrix.</p>
</td></tr>
<tr><td><code id="ReadData_+3A_genformat">Genformat</code></td>
<td>
<p>genotype format.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zhang Ya-Wen, Wang Jing-Tian, Li Pei, Zhang Yuan-Ming<br />
Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G1=data(Gen)
P1=data(Phe)
result=ReadData(fileGen=Gen,filePhe=Phe,fileKin=NULL,filePS =NULL,
fileCov=NULL,Genformat=1)
</code></pre>

<hr>
<h2 id='ResultFinal'>
Final result used to draw manhattan plot.
</h2><span id='topic+ResultFinal'></span><span id='topic+Chromosome'></span><span id='topic+QTN.effect'></span><span id='topic+LOD.score'></span>

<h3>Description</h3>

<p>Final result used to draw manhattan plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> data(ResultFinal)
</code></pre>


<h3>Details</h3>

<p>Final result used to draw manhattan plot.
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

<hr>
<h2 id='ResultIntermediate'>
Intermediate result used to draw manhattan plot.
</h2><span id='topic+ResultIntermediate'></span><span id='topic+Trait.ID'></span><span id='topic+Trait.name'></span><span id='topic+RS.'></span>

<h3>Description</h3>

<p>Intermediate result used to draw manhattan plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> data(ResultIntermediate)
</code></pre>


<h3>Details</h3>

<p>Intermediate result used to draw manhattan plot.
</p>


<h3>Author(s)</h3>

<p>Maintainer: Yuan-Ming Zhang&lt;soyzhang@mail.hzau.edu.cn&gt;
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
