<!DOCTYPE html><html lang="en"><head><title>Help for package msaR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {msaR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.check_lengths'><p>check_lengths</p></a></li>
<li><a href='#as.fasta'><p>as.fasta</p></a></li>
<li><a href='#as.sequences'><p>as.sequences</p></a></li>
<li><a href='#msaR'><p>msaR</p></a></li>
<li><a href='#msaROutput'><p>Widget output function for use in Shiny</p></a></li>
<li><a href='#renderMsaR'><p>Widget render function for use in Shiny</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Multiple Sequence Alignment for R Shiny</td>
</tr>
<tr>
<td>Version:</td>
<td>0.6.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Visualizes multiple sequence alignments dynamically within the
    Shiny web application framework.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.2)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="http://www.boost.org/LICENSE_1_0.txt">BSL-1.0</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, htmlwidgets</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Biostrings, knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-22 02:54:25 UTC; zachcp</td>
</tr>
<tr>
<td>Author:</td>
<td>Benedikt Rauscher [aut, cph],
  Zachary Charlop-Powers [cre, aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zachary Charlop-Powers &lt;zach.charlop.powers@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-22 04:30:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='.check_lengths'>check_lengths</h2><span id='topic+.check_lengths'></span>

<h3>Description</h3>

<p>check_lengths
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.check_lengths(seqlist)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id=".check_lengths_+3A_seqlist">seqlist</code></td>
<td>
<p>optimistic loading can lead to issues. All MSAs
should have identical length. if not theres a problem.</p>
</td></tr>
</table>

<hr>
<h2 id='as.fasta'>as.fasta</h2><span id='topic+as.fasta'></span>

<h3>Description</h3>

<p>functionality to convert objects to a fasta string. Currently
this can handle character objects which are interpreted as filenames or
several of the popular means of storing sequence data: <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>, 
<code><a href="Biostrings.html#topic+DNAStringSet">DNAStringSet</a></code>, <code><a href="Biostrings.html#topic+AAStringSet">AAStringSet</a></code>,
<code><a href="Biostrings.html#topic+RNAStringSet">RNAStringSet</a></code>, <code><a href="Biostrings.html#topic+BStringSet">BStringSet</a></code>,
<code><a href="Biostrings.html#topic+DNAMultipleAlignment">DNAMultipleAlignment</a></code>, <code><a href="Biostrings.html#topic+RNAMultipleAlignment">RNAMultipleAlignment</a></code>,
or <code><a href="Biostrings.html#topic+AAMultipleAlignment">AAMultipleAlignment</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.fasta(seqs)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as.fasta_+3A_seqs">seqs</code></td>
<td>
<p>(Required.) the sequence/alignment to be displayed. A character vector,  <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>, <code><a href="Biostrings.html#topic+DNAStringSet">DNAStringSet</a></code>,  <code><a href="Biostrings.html#topic+AAStringSet">AAStringSet</a></code>,
or <code><a href="Biostrings.html#topic+RNAStringSet">RNAStringSet</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string in fasta format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>seqfile &lt;- system.file("sequences","AHBA.aln",package="msaR")
as.fasta(seqfile)
help("as.fasta")

## Not run: 
if (requireNamespace("Biostrings")) {
   seqs &lt;- readDNAStringSet(seqfile)
   as.fasta(seqs)
 }

## End(Not run)
</code></pre>

<hr>
<h2 id='as.sequences'>as.sequences</h2><span id='topic+as.sequences'></span>

<h3>Description</h3>

<p>functionality to convert sequence objects into R lists that can be serialized to 
JS as JSON. Currently, this can handle character objects which are interpreted 
as filenames or several of the popular means of storing sequence data: <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>, 
<code><a href="Biostrings.html#topic+DNAStringSet">DNAStringSet</a></code>, <code><a href="Biostrings.html#topic+AAStringSet">AAStringSet</a></code>,
<code><a href="Biostrings.html#topic+RNAStringSet">RNAStringSet</a></code>, <code><a href="Biostrings.html#topic+BStringSet">BStringSet</a></code>,
<code><a href="Biostrings.html#topic+DNAMultipleAlignment">DNAMultipleAlignment</a></code>, <code><a href="Biostrings.html#topic+RNAMultipleAlignment">RNAMultipleAlignment</a></code>,
or <code><a href="Biostrings.html#topic+AAMultipleAlignment">AAMultipleAlignment</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.sequences(seqs)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as.sequences_+3A_seqs">seqs</code></td>
<td>
<p>(Required.) the sequence/alignment to be displayed. A character vector,  <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>, <code><a href="Biostrings.html#topic+DNAStringSet">DNAStringSet</a></code>,  <code><a href="Biostrings.html#topic+AAStringSet">AAStringSet</a></code>,
or <code><a href="Biostrings.html#topic+RNAStringSet">RNAStringSet</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of named lists where each sublist has name, id, and seq members.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>seqfile &lt;- system.file("sequences","AHBA.aln",package="msaR")
as.sequences(seqfile)
help("as.sequences")

## Not run: 
if (requireNamespace("Biostrings")) {
   seqs &lt;- readDNAStringSet(seqfile)
   as.sequences(seqs)
 }

## End(Not run)

</code></pre>

<hr>
<h2 id='msaR'>msaR</h2><span id='topic+msaR'></span>

<h3>Description</h3>

<p>Dynamic Multiple Sequence Alignments in R and Shiny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msaR(
  msa,
  menu = TRUE,
  width = NULL,
  height = NULL,
  rowheight = 15,
  alignmentHeight = 225,
  overviewbox = TRUE,
  seqlogo = TRUE,
  colorscheme = "nucleotide",
  conservation = FALSE,
  markers = TRUE,
  metacell = FALSE,
  leftheader = TRUE,
  labels = TRUE,
  labelname = TRUE,
  labelid = FALSE,
  labelNameLength = 100,
  overviewboxWidth = "auto",
  overviewboxHeight = "fixed"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msaR_+3A_msa">msa</code></td>
<td>
<p>File or BioString  Object representing a multiple Sequence Alignment.</p>
</td></tr>
<tr><td><code id="msaR_+3A_menu">menu</code></td>
<td>
<p>Optional. Default <code>TRUE</code>. Determines whether to include the interactive menu.</p>
</td></tr>
<tr><td><code id="msaR_+3A_width">width</code></td>
<td>
<p>Optional. Default <code>NULL</code>. The width of the html widget element.</p>
</td></tr>
<tr><td><code id="msaR_+3A_height">height</code></td>
<td>
<p>Optional. Default <code>NULL</code>. The height of the html widget element.</p>
</td></tr>
<tr><td><code id="msaR_+3A_rowheight">rowheight</code></td>
<td>
<p>Optional. Default <code>20</code>. Height of a row in the MSA.</p>
</td></tr>
<tr><td><code id="msaR_+3A_alignmentheight">alignmentHeight</code></td>
<td>
<p>Optional. Default <code>225</code>. Height of the MSA.</p>
</td></tr>
<tr><td><code id="msaR_+3A_overviewbox">overviewbox</code></td>
<td>
<p>optional. Default <code>TRUE</code>. Include the overview box?</p>
</td></tr>
<tr><td><code id="msaR_+3A_seqlogo">seqlogo</code></td>
<td>
<p>optional. Default <code>TRUE</code>. Include the seqlogo?</p>
</td></tr>
<tr><td><code id="msaR_+3A_colorscheme">colorscheme</code></td>
<td>
<p>optional. Default <code>"nucleotide"</code>. The color scheme to use. Can be one of the following: 
&quot;buried&quot;,&quot;cinema&quot;,&quot;clustal&quot;,&quot;clustal2&quot;,&quot;helix&quot;,&quot;hydro&quot;,&quot;lesk&quot;,&quot;mae&quot;,&quot;nucleotide&quot;,&quot;purine&quot;,&quot;strand&quot;,&quot;taylor&quot;,&quot;turn&quot;,&quot;zappo&quot;</p>
</td></tr>
<tr><td><code id="msaR_+3A_conservation">conservation</code></td>
<td>
<p>optional. Default <code>TRUE</code>. Include the conservation widget?</p>
</td></tr>
<tr><td><code id="msaR_+3A_markers">markers</code></td>
<td>
<p>optional. Default <code>TRUE</code>. Include the alignment markers? These are the numbers along the top that</p>
</td></tr>
<tr><td><code id="msaR_+3A_metacell">metacell</code></td>
<td>
<p>optional. Default <code>FALSE</code>. Include the per-sequence metadata.</p>
</td></tr>
<tr><td><code id="msaR_+3A_leftheader">leftheader</code></td>
<td>
<p>optional. Default <code>TRUE</code>. Include the header information.</p>
</td></tr>
<tr><td><code id="msaR_+3A_labels">labels</code></td>
<td>
<p>optional. Default <code>TRUE</code>. Include all of the sequence information msa Labels.</p>
</td></tr>
<tr><td><code id="msaR_+3A_labelname">labelname</code></td>
<td>
<p>optional. Default <code>TRUE</code>. Include sequence name?</p>
</td></tr>
<tr><td><code id="msaR_+3A_labelid">labelid</code></td>
<td>
<p>optional. Default <code>FALSE</code>. Include the labelid?</p>
</td></tr>
<tr><td><code id="msaR_+3A_labelnamelength">labelNameLength</code></td>
<td>
<p>optional. Default <code>100</code>. Width of the Label Names.</p>
</td></tr>
<tr><td><code id="msaR_+3A_overviewboxwidth">overviewboxWidth</code></td>
<td>
<p>optional. Default. <code>"auto"</code>. Can also be &quot;fixed&quot;</p>
</td></tr>
<tr><td><code id="msaR_+3A_overviewboxheight">overviewboxHeight</code></td>
<td>
<p>optional. Default. <code>"fixed"</code>. Can also be an integer value.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>seqfile &lt;- system.file("sequences","AHBA.aln",package="msaR")
msaR(seqfile)
</code></pre>

<hr>
<h2 id='msaROutput'>Widget output function for use in Shiny</h2><span id='topic+msaROutput'></span>

<h3>Description</h3>

<p>Widget output function for use in Shiny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msaROutput(outputId, width = "100%", height = "100%")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msaROutput_+3A_outputid">outputId</code></td>
<td>
<p>output id</p>
</td></tr>
<tr><td><code id="msaROutput_+3A_width">width</code></td>
<td>
<p>width</p>
</td></tr>
<tr><td><code id="msaROutput_+3A_height">height</code></td>
<td>
<p>height</p>
</td></tr>
</table>

<hr>
<h2 id='renderMsaR'>Widget render function for use in Shiny</h2><span id='topic+renderMsaR'></span>

<h3>Description</h3>

<p>Widget render function for use in Shiny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>renderMsaR(expr, env = parent.frame(), quoted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="renderMsaR_+3A_expr">expr</code></td>
<td>
<p>expr</p>
</td></tr>
<tr><td><code id="renderMsaR_+3A_env">env</code></td>
<td>
<p>env</p>
</td></tr>
<tr><td><code id="renderMsaR_+3A_quoted">quoted</code></td>
<td>
<p>quoted</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
