<!DOCTYPE html><html><head><title>Help for package MSCsimtester</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MSCsimtester}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ADtest'><p>Anderson-Darling test comparing sample and theoretical pairwise distance distributions.</p></a></li>
<li><a href='#genetreeSample'><p>Simulated gene tree dataset.</p></a></li>
<li><a href='#MSCsimtester'><p>Validity tests of simulators of the multspecies coalescent model in phylogenomics.</p></a></li>
<li><a href='#pairwiseDist'><p>Compute and plot sample and theoretical pairwise distance densities.</p></a></li>
<li><a href='#plotEdgeOrder'><p>Plot species tree, with edge numbers on edges.</p></a></li>
<li><a href='#plotPops'><p>Plot species tree, with population sizes on edges.</p></a></li>
<li><a href='#print.ADtestOutput'><p>Print function for objects of class <code>ADtestOutput</code>.</p></a></li>
<li><a href='#print.rootedTripleOutput'><p>Print function for objects of class <code>rootedTripleOutput</code>.</p></a></li>
<li><a href='#rootedTriple'><p>Compare expected and sample frequencies of topological rooted triples.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tests of Multispecies Coalescent Gene Tree Simulator Output</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Elizabeth Allman &lt;e.allman@alaska.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Statistical tests for validating multispecies coalescent gene tree simulators, using pairwise distances and rooted triple counts. Background is given by Allman, Banos, and Rhodes (2019) &lt;<a href="https://arxiv.org/abs/1908.01424">arXiv:1908.01424</a>&gt;. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rdpack, kSamples, graphics, stats</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.10),ape(&ge; 5.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-12-14 23:46:24 UTC; eallman</td>
</tr>
<tr>
<td>Author:</td>
<td>Elizabeth Allman [aut, cre, cph],
  Hector Banos [aut, cph],
  John Rhodes [aut, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-12-15 00:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ADtest'>Anderson-Darling test comparing sample and theoretical pairwise distance distributions.</h2><span id='topic+ADtest'></span>

<h3>Description</h3>

<p>Takes as input theoretical pairwise distance densities under the MSC and
empirical pairwise distances from gene trees in a sample, as returned by
the function <code>pairwiseDist</code>.  Uses the package <code>kSamples</code> to perform
either one test on the entire dataset or multiple tests on subsamples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ADtest(distanceDensities, subsampleSize = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ADtest_+3A_distancedensities">distanceDensities</code></td>
<td>
<p>A list containing values needed for performing Anderson-Darling
test(s) on a gene tree sample and species tree, as output by <code>pairwiseDist</code>.
For details, see code for <code>pairwiseDist</code>.</p>
</td></tr>
<tr><td><code id="ADtest_+3A_subsamplesize">subsampleSize</code></td>
<td>
<p>A positive integer to perform multiple tests on subsamples,
or <code>FALSE</code> (default) to perform one test on full sample.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Anderson-Darling test compares the empirical distance distribution for a supplied gene tree
sample to a sample drawn from the theoretical distribution. The output, passed from the <code>kSamples</code> package,
thus says that 2 samples are being compared, to test a null-hypothesis that they come from the same distribution.
See <code>kSamples</code> documentation for function <code>ad.test</code> for more details.
</p>
<p>Repeated runs of this function will give different results, since the sample from the theoretical distribution
will vary. Under the null hypothesis p-values for different runs should be approximately
uniformly distributed.
</p>
<p>Numerical issues may result in poor performance of Anderson-Darling tests when the sample size
is very large, so
an optional parameter <code>subsampleSize</code> can be set to create subsamples of smaller size.
If <code>subsampleSize</code> is a positive integer,
Anderson-Darling tests are performed on each subset, comparing them to
a random sample of the same size from the theoretical distribution. Good fit is indicated by an approximately uniform
distribution of the subsample p-values.
</p>


<h3>Value</h3>

<p>An object of type <code>ADtestOutput</code> including a sample <code>$Sample</code> from the theoretical distance distribution of
the same size as the empirical one, and <code>$ADtest</code> which is of type <code>kSamples</code> and
has all output from the Anderson-Darling test if only
one test was performed, or the number of tests if tests were performed on subsamples.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwiseDist">pairwiseDist</a></code>, <code><a href="kSamples.html#topic+kSamples-package">kSamples-package</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stree=read.tree(text="((((a:10000,b:10000):10000,c:20000):10000,d:30000):10000,e:40000);")
pops=c(15000,25000,10000,1,1,1,1,1,12000)
gts=read.tree(file=system.file("extdata","genetreeSample",package="MSCsimtester"))
distDen=pairwiseDist(stree,pops,gts,"a","b")
ADtest(distDen)
ADtest(distDen,1000) 

</code></pre>

<hr>
<h2 id='genetreeSample'>Simulated gene tree dataset.</h2><span id='topic+genetreeSample'></span>

<h3>Description</h3>

<p>A dataset of 10,000 gene trees on 5 taxa simulated under the MSC on a species tree.
</p>


<h3>Format</h3>

<p>A text file with 10,000 metric Newick gene trees on the taxa a,b,c,d,e
</p>


<h3>Details</h3>

<p>This simulated dataset was produced by SimPhy (Mallo et al. 2016),
using the species tree
</p>
<p>((((a:10000,b:10000):10000,c:20000):10000,d:30000):10000,e:40000);
</p>
<p>with population sizes
</p>
<p>c(15000,25000,10000,1,1,1,1,1,12000)
</p>
<p>and edges ordered by the <code>ape</code> function <code>read.tree</code>.
</p>
<p>File is accessed as  <code>system.file("extdata","genetreeSample",package="MSCsimtester")</code>, for example
using the ape command:
</p>
<p><code>gts=read.tree(file=system.file("extdata","genetreeSample",package="MSCsimtester") )</code>
</p>


<h3>References</h3>

<p>Mallo D, De Oliveira Martins L, Posada D (2016).
&ldquo;SimPhy: Phylogenomic Simulation of Gene, Locus, and Species Trees.&rdquo;
<em>Syst. Biol.</em>, <b>65</b>(2), 334-344.
doi: <a href="https://doi.org/10.1093/sysbio/syv082">10.1093/sysbio/syv082</a>, <a href="http://dx.doi.org/10.1093/sysbio/syv082">http://dx.doi.org/10.1093/sysbio/syv082</a>.
</p>

<hr>
<h2 id='MSCsimtester'>Validity tests of simulators of the multspecies coalescent model in phylogenomics.</h2><span id='topic+MSCsimtester'></span>

<h3>Description</h3>

<p>The package performs comparisons of certain summary statistics for simulated gene tree samples to
theoretical predictions under the multispecies coalescent model.
The primary functions are <code>rootedTriple</code> for comparison of frequencies of topological rooted triples
on gene trees, and <code>pairwiseDist</code> and <code>ADtest</code> for comparison of the distributions of pairwise
distances between taxa on gene trees.
</p>


<h3>Details</h3>

<p>Required input is a collection of gene trees, stored as a <code>multiPhylo</code> object by the <code>ape</code> package,
and a rooted species tree, as a <code>Phylo</code> object,
with edge lengths in generations, together with constant population sizes for each edge.
</p>
<p><code>MSCsimtester</code> builds on the packages <code>ape</code> and <code>kSamples</code>.
</p>
<p>For further examples of use and citation purposes, see (Allman et al. 2019).
</p>


<h3>References</h3>

<p>Allman ES, Ba√±os H, Rhodes JA (2019).
&ldquo;Testing Multispecies Coalescent Simulators Using Summary Statistics.&rdquo;
<em>arXiv:1908.01424</em>.
</p>

<hr>
<h2 id='pairwiseDist'>Compute and plot sample and theoretical pairwise distance densities.</h2><span id='topic+pairwiseDist'></span>

<h3>Description</h3>

<p>Computes theoretical pairwise distance densities under the MSC on a species tree and empirical pairwise distances from
gene trees in a sample. A histogram of empirical values is plotted over the theoretical pdf.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwiseDist(
  stree,
  popSizes,
  gtSample,
  taxon1,
  taxon2,
  numSteps = 1000,
  tailProb = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwiseDist_+3A_stree">stree</code></td>
<td>
<p>An object of class <code>phylo</code> containing a rooted metric species tree. Edge lengths are in
number of generations.</p>
</td></tr>
<tr><td><code id="pairwiseDist_+3A_popsizes">popSizes</code></td>
<td>
<p>A vector containing constant population sizes, one entry for each
edge/population in the species tree, for a haploid population.  Sizes should be doubled for diploids.
If <code>stree</code> has k edges, then <code>popSizes</code> must have k+1 elements, with final entry
the size of the population ancestral to the root.</p>
</td></tr>
<tr><td><code id="pairwiseDist_+3A_gtsample">gtSample</code></td>
<td>
<p>An object of class <code>multiPhylo</code> holding a sample of gene trees from a simulation.
Taxon labels on gene trees must be identical to those on <code>stree</code>.</p>
</td></tr>
<tr><td><code id="pairwiseDist_+3A_taxon1">taxon1</code></td>
<td>
<p>A string specifying one taxon on <code>stree</code>.</p>
</td></tr>
<tr><td><code id="pairwiseDist_+3A_taxon2">taxon2</code></td>
<td>
<p>A string specifying a second taxon on <code>stree</code>, distinct
from <code>taxon1</code>.</p>
</td></tr>
<tr><td><code id="pairwiseDist_+3A_numsteps">numSteps</code></td>
<td>
<p>A positive integer number of values to be computed for
graphing the theoretical pairwise distance density.  Default is <code>numSteps = 1000</code>.
A larger value produces a smoother plot.</p>
</td></tr>
<tr><td><code id="pairwiseDist_+3A_tailprob">tailProb</code></td>
<td>
<p>A cutoff value, between 0 and 1, for the theoretical density, with a default of 0.01.
The theoretical pairwise distance is plotted from (0, xMax), where  xMax
is the larger of the maximum pairwise distance in the gene tree sample and the value cutting off
a tail of area <code>tailProb</code> under the pdf. A message returns the proportion of sample distances in this tail.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of items needed for Anderson-Darling test(s), for use by <code>ADtest</code>,
returned invisibly. See function code for more details.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotEdgeOrder">plotEdgeOrder</a></code>, <code><a href="#topic+plotPops">plotPops</a></code>, <code><a href="#topic+ADtest">ADtest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stree=read.tree(text="((((a:10000,b:10000):10000,c:20000):10000,d:30000):10000,e:40000);")
pops=c(15000,25000,10000,1,1,1,1,1,12000)
gts=read.tree(file=system.file("extdata","genetreeSample",package="MSCsimtester"))
pairwiseDist(stree,pops,gts,"a","b")

</code></pre>

<hr>
<h2 id='plotEdgeOrder'>Plot species tree, with edge numbers on edges.</h2><span id='topic+plotEdgeOrder'></span>

<h3>Description</h3>

<p>Under the MSC, each edge in the species tree must be assigned a population
size.  This function displays the species tree with the edges
numbered, to aid the user in entering constant population sizes as an
appropriately ordered list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotEdgeOrder(stree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotEdgeOrder_+3A_stree">stree</code></td>
<td>
<p>An object of class <code>phylo</code> containing a rooted metric
species tree.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NONE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwiseDist">pairwiseDist</a></code>, <code><a href="#topic+rootedTriple">rootedTriple</a></code>, <code><a href="#topic+plotPops">plotPops</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stree=read.tree(text="(((a:10000,b:10000):10000,c:20000):10000,d:30000);")
plotEdgeOrder(stree)
pops=c(30000,20000,1,1,1,1,10000)
plotPops(stree,pops)

</code></pre>

<hr>
<h2 id='plotPops'>Plot species tree, with population sizes on edges.</h2><span id='topic+plotPops'></span>

<h3>Description</h3>

<p>Plot species tree, with population sizes on edges.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPops(stree, populations)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPops_+3A_stree">stree</code></td>
<td>
<p>An object of class <code>phylo</code> containing a rooted metric species tree.</p>
</td></tr>
<tr><td><code id="plotPops_+3A_populations">populations</code></td>
<td>
<p>A vector containing constant population sizes, one entry for each
edge/population in the species tree, with last entry for
the population ancestral to the root.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NONE
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairwiseDist">pairwiseDist</a></code>, <code><a href="#topic+rootedTriple">rootedTriple</a></code>, <code><a href="#topic+plotEdgeOrder">plotEdgeOrder</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stree=read.tree(text="(((a:10000,b:10000):10000,c:20000):10000,d:30000);")
plotEdgeOrder(stree)
pops=c(30000,20000,1,1,1,1,10000)
plotPops(stree,pops)

</code></pre>

<hr>
<h2 id='print.ADtestOutput'>Print function for objects of class <code>ADtestOutput</code>.</h2><span id='topic+print.ADtestOutput'></span>

<h3>Description</h3>

<p>Print function for objects of class <code>ADtestOutput</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ADtestOutput'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.ADtestOutput_+3A_x">x</code></td>
<td>
<p>An object of class <code>ADtestOutput</code>, as produced by <code>ADtest</code> function.</p>
</td></tr>
<tr><td><code id="print.ADtestOutput_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NONE
</p>

<hr>
<h2 id='print.rootedTripleOutput'>Print function for objects of class <code>rootedTripleOutput</code>.</h2><span id='topic+print.rootedTripleOutput'></span>

<h3>Description</h3>

<p>Print function for objects of class <code>rootedTripleOutput</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rootedTripleOutput'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.rootedTripleOutput_+3A_x">x</code></td>
<td>
<p>An object of class <code>rootedTripleOutput</code>, as produced by <code>rootedTriple</code> function.</p>
</td></tr>
<tr><td><code id="print.rootedTripleOutput_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Print function for objects of class <code>rootedTripleOutput</code>.
</p>


<h3>Value</h3>

<p>NONE
</p>

<hr>
<h2 id='rootedTriple'>Compare expected and sample frequencies of topological rooted triples.</h2><span id='topic+rootedTriple'></span>

<h3>Description</h3>

<p>For a given species tree with population sizes, compares the expected frequencies
of rooted triples to empirical frequencies in a sample of gene trees, using Chi-squared tests with 2 d.f.
The exact and estimated internal branch length (in coalescent units) of the rooted triple
in the species tree are also computed for comparison.  A single test can be performed
on the entire gene tree sample, or multiple tests on subsamples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rootedTriple(
  stree,
  popSizes,
  gtSample,
  taxon1,
  taxon2,
  taxon3,
  subsampleSize = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rootedTriple_+3A_stree">stree</code></td>
<td>
<p>An object of class <code>phylo</code> containing a rooted metric species tree.
Edge lengths are in number of generations.</p>
</td></tr>
<tr><td><code id="rootedTriple_+3A_popsizes">popSizes</code></td>
<td>
<p>An ordered list containing constant population sizes for each species tree edge, for a haploid organism.
Sizes should be doubled for diploids. If <code>stree</code> has k edges, then <code>popSizes</code> must have k+1 elements,
with the final entry for the population ancestral to the root.</p>
</td></tr>
<tr><td><code id="rootedTriple_+3A_gtsample">gtSample</code></td>
<td>
<p>An object of class <code>multiPhylo</code> holding a sample of gene trees from a simulation.
Taxon labels on gene trees must be identical to those on <code>stree</code>.</p>
</td></tr>
<tr><td><code id="rootedTriple_+3A_taxon1">taxon1</code></td>
<td>
<p>A string specifying one taxon on <code>stree</code>.</p>
</td></tr>
<tr><td><code id="rootedTriple_+3A_taxon2">taxon2</code></td>
<td>
<p>A string specifying a second taxon on <code>stree</code>, distinct from <code>taxon1</code>.</p>
</td></tr>
<tr><td><code id="rootedTriple_+3A_taxon3">taxon3</code></td>
<td>
<p>A string specifying a third taxon on <code>stree</code>, distinct from <code>taxon1</code>, <code>taxon2</code>.</p>
</td></tr>
<tr><td><code id="rootedTriple_+3A_subsamplesize">subsampleSize</code></td>
<td>
<p>A positive integer or <code>FALSE</code>, giving size of subsamples of <code>gtSample</code> to analyze.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When <code>subsampleSize</code> is <code>FALSE</code> the Chi-squared test is performed using all
gene trees in <code>gtSample</code>. Results are reported in tabular form in the console.
</p>
<p>When <code>subsampleSize</code> is positive, the  <code>N</code> trees in <code>gtSample</code>
will be partitioned into <code>N/subsampleSize</code> subsamples, with a Chi-squared test
performed for each. Histograms are plotted for (1) the p-values for the Chi-squared tests on
subsamples, and (2) subsample estimates of the internal branch length for the rooted triple on the
species tree, with the true value marked.
</p>
<p>Three distinct taxon names must be supplied, all of which must occur on <code>stree</code>
and in each of the gene trees in the sample.
</p>


<h3>Value</h3>

<p>If <code>subsampleSize</code> is <code>FALSE</code>, returns an object of type <code>rootedTripleOutput</code>
which contains a table <code>$TripletCounts</code> of empirical and expected rooted triple counts,
a p-value <code>$pv</code> from the Chi-squared test, and a column <code>$InternalEdge</code> of estimated and exact internal edge lengths.
If <code>subsampleSize</code> is <code>TRUE</code>, returns <code>NULL</code> but produces several plots.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotEdgeOrder">plotEdgeOrder</a></code>, <code><a href="#topic+plotPops">plotPops</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stree=read.tree(text="((((a:10000,b:10000):10000,c:20000):10000,d:30000):10000,e:40000);")
pops=c(15000,25000,10000,1,1,1,1,1,12000)
gts=read.tree(file=system.file("extdata","genetreeSample",package="MSCsimtester"))
rootedTriple(stree,pops,gts,"a","b","c")
rootedTriple(stree,pops,gts,"a","b","c",1000) 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
