<!DOCTYPE html><html><head><title>Help for package mseapca</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mseapca}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#csv2list'><p>Convert metabolite set / csv to list</p></a></li>
<li><a href='#list2xml'><p>Save compound set as XML file</p></a></li>
<li><a href='#msea_ora'><p>MSEA by over representation analysis</p></a></li>
<li><a href='#msea_sub'><p>MSEA by Subramanian et al.</p></a></li>
<li><a href='#pathbank2list'><p>Generate metabolite set list from PathBank database</p></a></li>
<li><a href='#pathway'><p>Example dataset for fasting and covid19 datasets</p></a></li>
<li><a href='#read_pathway'><p>Read metabolite set file (*.xml)</p></a></li>
<li><a href='#setlabel'><p>Generate binary label matrix of metabolite set</p></a></li>
<li><a href='#ssea_ora'><p>Single sample enrichment analysis by over representation analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Metabolite Set Enrichment Analysis for Loadings</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-5</td>
</tr>
<tr>
<td>Author:</td>
<td>Hiroyuki Yamamoto</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hiroyuki Yamamoto &lt;h.yama2396@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Computing metabolite set enrichment analysis (MSEA) (Yamamoto, H. et al. (2014) &lt;<a href="https://doi.org/10.1186%2F1471-2105-15-51">doi:10.1186/1471-2105-15-51</a>&gt;) and single sample enrichment analysis (SSEA) (Yamamoto, H. (2023) &lt;<a href="https://doi.org/10.51094%2Fjxiv.262">doi:10.51094/jxiv.262</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>XML</td>
</tr>
<tr>
<td>Depends:</td>
<td>loadings</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-04 10:42:29 UTC; hyama</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-04 11:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='csv2list'>Convert metabolite set / csv to list</h2><span id='topic+csv2list'></span>

<h3>Description</h3>

<p>This function converts your own metabolite set (csv file to list).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>csv2list(filepath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="csv2list_+3A_filepath">filepath</code></td>
<td>

<p>file path of metabolite set (csv file)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The first row of csv file are &quot;metabolite set name&quot; and &quot;metabolite IDs&quot; as header.
The first column must be metabolite IDs and second column must be metabolite set name.
</p>


<h3>Value</h3>

<p>list of metabolite set name and metabolite IDs
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
	# ---------------------------
	#  Convert csv file to list
	# ---------------------------
	filepath &lt;- "C:/pathway.csv"	# filepath of csv file
	N &lt;- csv2list(filepath)	# convert csv file to list
  
## End(Not run)
</code></pre>

<hr>
<h2 id='list2xml'>Save compound set as XML file</h2><span id='topic+list2xml'></span>

<h3>Description</h3>

<p>This function save compound set of list format as XML file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list2xml(filepath, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="list2xml_+3A_filepath">filepath</code></td>
<td>

<p>filepath of XML file to save
</p>
</td></tr>
<tr><td><code id="list2xml_+3A_m">M</code></td>
<td>

<p>list fomat of compound set and compound names
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to store a compound set. Saved xml file can be read using the read_pathway function.
</p>


<h3>Value</h3>

<p>filepath of saved XML file
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
	data(pathway)
	M &lt;- pathway$fasting
	xml_file &lt;- "pathway_fasting.xml"
	N &lt;- list2xml(xml_file, M)
	# XML::saveXML(N,filepath)
	
## End(Not run)
</code></pre>

<hr>
<h2 id='msea_ora'>MSEA by over representation analysis</h2><span id='topic+msea_ora'></span>

<h3>Description</h3>

<p>This function performs metabolite set enrichment analysis by over representation analysis (ORA).
Statistical hypothesis test of cross tabulation is performed by one-sided Fisher's exact test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msea_ora(SIG, ALL, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="msea_ora_+3A_sig">SIG</code></td>
<td>

<p>Metabolite names of significant metabolites
</p>
</td></tr>
<tr><td><code id="msea_ora_+3A_all">ALL</code></td>
<td>

<p>Metabolite names of all detected metabolites
</p>
</td></tr>
<tr><td><code id="msea_ora_+3A_m">M</code></td>
<td>

<p>list of metabolite set name and metabolite name
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of p-value and q-value for metabolite set and selected (significant) metabolite IDs for each metabolite set
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Draghici S, Khatri P, Martins RP, Ostermeier GC, Krawetz SA.
Global functional profiling of gene expression.
Genomics. 2003 Feb;81(2):98-104.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example1 : Metabolome data
data(fasting)
data(pathway)

# pca and pca loading
pca &lt;- prcomp(fasting$X, scale=TRUE)
pca &lt;- pca_loading(pca)

# all detected metabolites
metabolites &lt;- colnames(fasting$X)

# statistically significant negatively correlated metabolites in PC1 loading
SIG &lt;- metabolites[pca$loading$R[,1] &lt; 0 &amp; pca$loading$p.value[,1] &lt; 0.05]
ALL &lt;- metabolites #all detected metabolites

# metabolite set list
M &lt;- pathway$fasting

# MSEA by over representation analysis
B &lt;- msea_ora(SIG, ALL, M)
B$`Result of MSEA(ORA)`

## Example2 : Proteome data
data(covid19)
data(pathway)

X &lt;- covid19$X$proteomics
Y &lt;- covid19$Y
D &lt;- covid19$D
tau &lt;- covid19$tau

protein_name &lt;- colnames(X)

# pls-rog and pls-rog loading
plsrog &lt;- pls_rog(X,Y,D)
plsrog &lt;- plsrog_loading(plsrog)

# statistically significant proteins
index_prot &lt;- which(plsrog$loading$R[,1]&gt;0 &amp; plsrog$loading$p.value[,1]&lt;0.05)
sig_prot &lt;- protein_name[index_prot]

# detected proteins
protein_name &lt;- colnames(X)

# protein set list
M &lt;- pathway$covid19$proteomics

# MSEA by over representation analysis
B &lt;- msea_ora(sig_prot, protein_name, M)
B$`Result of MSEA(ORA)`

</code></pre>

<hr>
<h2 id='msea_sub'>MSEA by Subramanian et al.</h2><span id='topic+msea_sub'></span>

<h3>Description</h3>

<p>This function performs metabolite set enrichment analysis implemented in the same fashion as gene set enrichment analysis (Subramanian et al. 2005).
In this function, a permutation procedure is performed for a metabolite set rather than class label.
This procedure corresponds to a &quot;gene set&quot; of permutation type in GSEA-P software (Subramanian et al. 2007).
A leading-edge subset analysis is also undertaken following the standard GSEA procedure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msea_sub(M, D, y, maxiter = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="msea_sub_+3A_m">M</code></td>
<td>

<p>list of metbolite set name and metabolite IDs
</p>
</td></tr>
<tr><td><code id="msea_sub_+3A_d">D</code></td>
<td>

<p>data.frame(metabolite ID, data matix)
</p>
</td></tr>
<tr><td><code id="msea_sub_+3A_y">y</code></td>
<td>

<p>response variable (e.g. PC score)
</p>
</td></tr>
<tr><td><code id="msea_sub_+3A_maxiter">maxiter</code></td>
<td>

<p>maximum number of iterations in random permutation (default=1000)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of normalized enrichment score, p-value and q-value for metabolite set, and the results of leading edge subset
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Subramanian, A., Tamayo, P., Mootha, V. K., Mukherjee, S., Ebert, B. L., Gillette, M. A., Paulovich, A., Pomeroy, S. L., Golub, T. R., Lander, E. S. &amp; Mesirov, J. P. (2005)
Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles. Proc. Natl. Acad. Sci. USA 102, 15545-15550.
</p>
<p>Subramanian, A., Kuehn, H., Gould, J., Tamayo, P., Mesirov, J.P. (2007) GSEA-P: A desktop application for Gene Set Enrichment Analysis. Bioinformatics, doi: 10.1093/bioinformatics/btm369.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(fasting)
data(pathway)

# pca and pca loading
pca &lt;- prcomp(fasting$X, scale=TRUE)
pca &lt;- pca_loading(pca)

# all detected metabolites
metabolites &lt;- colnames(fasting$X)

# statistically significant negatively correlated metabolites in PC1 loading
SIG &lt;- metabolites[pca$loading$R[,1] &lt; 0 &amp; pca$loading$p.value[,1] &lt; 0.05]
ALL &lt;- metabolites #all detected metabolites

# Set response variable
y &lt;- pca$x[,1]

# preparing dataframe
D &lt;- data.frame(ALL,t(fasting$X)) 		# preparing dataframe

# MSEA by Subramanian et al.
M &lt;- pathway$fasting
P &lt;- msea_sub(M,D,y, maxiter = 10) # iteration was set ato 10 for demonstration

</code></pre>

<hr>
<h2 id='pathbank2list'>Generate metabolite set list from PathBank database</h2><span id='topic+pathbank2list'></span>

<h3>Description</h3>

<p>This function generates metabolite set list of PathBank database by referencing the AHPathbankDbs Bioconductor package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathbank2list(tbl_pathbank, subject, id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathbank2list_+3A_tbl_pathbank">tbl_pathbank</code></td>
<td>

<p>tibble from AHPathbankDbs
</p>
</td></tr>
<tr><td><code id="pathbank2list_+3A_subject">subject</code></td>
<td>

<p>Pathway subject (Metabolic, Disease, etc.) in tibble
</p>
</td></tr>
<tr><td><code id="pathbank2list_+3A_id">id</code></td>
<td>

<p>database ID (HMDB ID, Uniprot ID, etc.) used for analysis
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>AHPathbankDbs needs to be installed separately.
</p>


<h3>Value</h3>

<p>list of metabolite or protein set
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## PathBank
library(AnnotationHub)

ah &lt;- AnnotationHub()
qr &lt;- query(ah, c("pathbank", "Homo sapiens"))

#tbl_pathbank &lt;- qr[[1]] # metabolomics
tbl_pathbank &lt;- qr[[2]] # proteomics

ids &lt;- names(tbl_pathbank)[-c(1:4)]
id &lt;- ids[1] # Uniprot ID

subs &lt;- unique(tbl_pathbank$`Pathway Subject`)
subject &lt;- subs[6] # Protein

M &lt;- pathbank2list(tbl_pathbank, subject, id)

## End(Not run)
</code></pre>

<hr>
<h2 id='pathway'>Example dataset for fasting and covid19 datasets</h2><span id='topic+pathway'></span>

<h3>Description</h3>

<p>This data includes a metabolite set list and metabolite name list for fasting, and a metabolite set list for covid19 dataset within the &quot;loadings&quot; package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pathway)
</code></pre>


<h3>Arguments</h3>

<p>The list object pathway contains the following elements:
</p>
<p>fasting : 	metabolite set list for fasting mouse dataset
</p>
<p>data$fasting : metabolite name list for fasting mouse dataset
</p>
<p>covid19$proteomics :   protein set list for covid19 dataset.
</p>


<h3>References</h3>

<p>Yamamoto H., Fujimori T., Sato H., Ishikawa G., Kami K., Ohashi Y. (2014). &quot;Statistical hypothesis testing of factor loading in principal component analysis and its application to metabolite set enrichment analysis&quot;. BMC Bioinformatics, (2014) 15(1):51.
</p>
<p>B. Shen, et al, Proteomic and Metabolomic Characterization of COVID-19 Patient Sera, Cell. 182 (2020) 59-72.e15.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pathway)
</code></pre>

<hr>
<h2 id='read_pathway'>Read metabolite set file (*.xml)</h2><span id='topic+read_pathway'></span>

<h3>Description</h3>

<p>This function generates metabolite set list from metabolite set file (XML).
This is mainly used to be called by other functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_pathway(fullpath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_pathway_+3A_fullpath">fullpath</code></td>
<td>

<p>file path of metabolite set (XML)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of metabolite set name and metabolite IDs.
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
	filename &lt;- "C:/R/pathway.xml"	# load metabolite set file
	M &lt;- read_pathway(filename)		# Convert XML to metabolite set (list)
  
## End(Not run)
</code></pre>

<hr>
<h2 id='setlabel'>Generate binary label matrix of metabolite set</h2><span id='topic+setlabel'></span>

<h3>Description</h3>

<p>This function generates binary label matrix of metabolite names and metabolite sets.
This is mainly used to be called by other functions, and used
to count the number of metabolites in a specific metabolite set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setlabel(M_ID, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setlabel_+3A_m_id">M_ID</code></td>
<td>

<p>detected metabolites
</p>
</td></tr>
<tr><td><code id="setlabel_+3A_m">M</code></td>
<td>

<p>list of metabolite set and metabolite names
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If single peak has multiple metabolite IDs in M_ID, split by &quot;,&quot; or &quot;;&quot;.
</p>


<h3>Value</h3>

<p>binary label matrix of metabolite names in metabolite sets
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(fasting)
data(pathway)

M_ID &lt;- colnames(fasting$X) # detected metabolites
M &lt;- pathway$fasting # metabolite set list

L &lt;- setlabel(M_ID,M)	# binary label matrix
</code></pre>

<hr>
<h2 id='ssea_ora'>Single sample enrichment analysis by over representation analysis</h2><span id='topic+ssea_ora'></span>

<h3>Description</h3>

<p>This function performs single sample enrichment analysis (SSEA) by over representation analysis (ORA). SSEA performs MSEA by ORA between detected and not detected metabolites in each sample.&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ssea_ora(det_list, det_all, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ssea_ora_+3A_det_list">det_list</code></td>
<td>

<p>metabolite names of detected metabolites
</p>
</td></tr>
<tr><td><code id="ssea_ora_+3A_det_all">det_all</code></td>
<td>

<p>metabolite names of all metabolites
</p>
</td></tr>
<tr><td><code id="ssea_ora_+3A_m">M</code></td>
<td>

<p>list of metabolite set and metabolite names
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The threshold for determining whether a metabolite is detected or not is typically set by the signal-to-noise (S/N) ratio. If the S/N ratio is unavailable, one might consider using the signal intensity or peak area for each metabolite as an alternative. In such cases, all values below the threshold can be set to 0.
</p>


<h3>Value</h3>

<p>A matrix where each row represents a sample and each column represents a set of metabolites.
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H., Single sample enrichment analysisfor mass spectrometry-based omics data, Jxiv.(2023)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(fasting)
data(pathway)

det_list &lt;- pathway$data$fasting
M &lt;- pathway$fasting
det_all &lt;- unique(c(colnames(fasting$X), as.character(unlist(M)))) 

# SSEA
Z &lt;- ssea_ora(det_list, det_all, M)

## PCA for SSEA score
pca &lt;- prcomp(Z, scale=TRUE)
pca &lt;- pca_loading(pca)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
