<!DOCTYPE html><html lang="en"><head><title>Help for package msigdbr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {msigdbr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#msigdbr-package'><p>msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format</p></a></li>
<li><a href='#msigdbr'><p>Retrieve the gene sets data frame</p></a></li>
<li><a href='#msigdbr_check_data'><p>Check that the data package is installed</p></a></li>
<li><a href='#msigdbr_collections'><p>List the collections available in the msigdbr package</p></a></li>
<li><a href='#msigdbr_species'><p>List the species available in the msigdbr package</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format</td>
</tr>
<tr>
<td>Version:</td>
<td>10.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides the 'Molecular Signatures Database' (MSigDB) gene
    sets typically used with the 'Gene Set Enrichment Analysis' (GSEA)
    software (Subramanian et al. 2005 &lt;<a href="https://doi.org/10.1073%2Fpnas.0506580102">doi:10.1073/pnas.0506580102</a>&gt;,
    Liberzon et al. 2015 &lt;<a href="https://doi.org/10.1016%2Fj.cels.2015.12.004">doi:10.1016/j.cels.2015.12.004</a>&gt;, Castanza et al.
    2023 &lt;<a href="https://doi.org/10.1038%2Fs41592-023-02014-7">doi:10.1038/s41592-023-02014-7</a>&gt;) as an R data frame. The package
    includes the human genes as listed in MSigDB as well as the
    corresponding symbols and IDs for frequently studied model organisms
    such as mouse, rat, pig, fly, and yeast.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://igordot.github.io/msigdbr/">https://igordot.github.io/msigdbr/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/igordot/msigdbr/issues">https://github.com/igordot/msigdbr/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>assertthat, babelgene (&ge; 22.9), dplyr (&ge; 1.1.1), lifecycle,
methods, rlang, tibble, tidyselect (&ge; 1.2.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, msigdbdf, rmarkdown, roxygen2, testthat</td>
</tr>
<tr>
<td>Additional_repositories:</td>
<td><a href="https://igordot.r-universe.dev">https://igordot.r-universe.dev</a></td>
</tr>
<tr>
<td>Config/Needs/website:</td>
<td>rmarkdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-19 03:09:11 UTC; id460</td>
</tr>
<tr>
<td>Author:</td>
<td>Igor Dolgalev <a href="https://orcid.org/0000-0003-4451-126X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Igor Dolgalev &lt;igor.dolgalev@nyumc.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-19 04:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='msigdbr-package'>msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format</h2><span id='topic+msigdbr-package'></span>

<h3>Description</h3>

<p>Provides the 'Molecular Signatures Database' (MSigDB) gene sets typically used with the 'Gene Set Enrichment Analysis' (GSEA) software (Subramanian et al. 2005 <a href="https://doi.org/10.1073/pnas.0506580102">doi:10.1073/pnas.0506580102</a>, Liberzon et al. 2015 <a href="https://doi.org/10.1016/j.cels.2015.12.004">doi:10.1016/j.cels.2015.12.004</a>, Castanza et al. 2023 <a href="https://doi.org/10.1038/s41592-023-02014-7">doi:10.1038/s41592-023-02014-7</a>) as an R data frame. The package includes the human genes as listed in MSigDB as well as the corresponding symbols and IDs for frequently studied model organisms such as mouse, rat, pig, fly, and yeast.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Igor Dolgalev <a href="mailto:igor.dolgalev@nyumc.org">igor.dolgalev@nyumc.org</a> (<a href="https://orcid.org/0000-0003-4451-126X">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://igordot.github.io/msigdbr/">https://igordot.github.io/msigdbr/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/igordot/msigdbr/issues">https://github.com/igordot/msigdbr/issues</a>
</p>
</li></ul>


<hr>
<h2 id='msigdbr'>Retrieve the gene sets data frame</h2><span id='topic+msigdbr'></span>

<h3>Description</h3>

<p>Retrieve a data frame of gene sets and their member genes.
The original human genes can be converted into their corresponding counterparts in various model organisms, including mouse, rat, pig, zebrafish, fly, and yeast.
The output includes gene symbols along with NCBI and Ensembl IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msigdbr(
  species = "Homo sapiens",
  db_species = "HS",
  collection = NULL,
  subcollection = NULL,
  category = deprecated(),
  subcategory = deprecated()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msigdbr_+3A_species">species</code></td>
<td>
<p>Species name for output genes, such as <code>"Homo sapiens"</code> or <code>"Mus musculus"</code>. Use <code>msigdbr_species()</code> for available options.</p>
</td></tr>
<tr><td><code id="msigdbr_+3A_db_species">db_species</code></td>
<td>
<p>Species abbreviation for the human or mouse databases (<code>"HS"</code> or <code>"MM"</code>).</p>
</td></tr>
<tr><td><code id="msigdbr_+3A_collection">collection</code></td>
<td>
<p>Collection abbreviation, such as <code>"H"</code> or <code>"C1"</code>. Use <code>msigdbr_collections()</code> for the available options.</p>
</td></tr>
<tr><td><code id="msigdbr_+3A_subcollection">subcollection</code></td>
<td>
<p>Sub-collection abbreviation, such as <code>"CGP"</code> or <code>"BP"</code>. Use <code>msigdbr_collections()</code> for the available options.</p>
</td></tr>
<tr><td><code id="msigdbr_+3A_category">category</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> use the <code>collection</code> argument</p>
</td></tr>
<tr><td><code id="msigdbr_+3A_subcategory">subcategory</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> use the <code>subcollection</code> argument</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Historically, the MSigDB resource has been tailored to the analysis of human-specific datasets, with gene sets exclusively aligned to the human genome.
Starting with release 2022.1, MSigDB incorporated a database of mouse-native gene sets and was split into human and mouse divisions (&quot;Hs&quot; and &quot;Mm&quot;).
Each one is provided in the approved gene symbols of its respective species.
The versioning convention of MSigDB is in the format <code>Year.Release.Species</code>.
The genes within each gene set may originate from a species different from the database target species as indicated by the <code>gs_source_species</code> and <code>db_target_species</code> fields.
</p>
<p>Mouse MSigDB includes gene sets curated from mouse-centric datasets and specified in native mouse gene identifiers, eliminating the need for ortholog mapping.
</p>
<p>To access the full dataset, please install the msigdbdf package (not available on CRAN):
</p>
<div class="sourceCode"><pre>install.packages("msigdbdf", repos = "https://igordot.r-universe.dev")
</pre></div>


<h3>Value</h3>

<p>A data frame of gene sets with one gene per row.
</p>


<h3>References</h3>

<p><a href="https://www.gsea-msigdb.org/gsea/msigdb/index.jsp">https://www.gsea-msigdb.org/gsea/msigdb/index.jsp</a>
</p>

<hr>
<h2 id='msigdbr_check_data'>Check that the data package is installed</h2><span id='topic+msigdbr_check_data'></span>

<h3>Description</h3>

<p>Check that the 'msigdbdf' data package is installed.
If not, provide instructions for installation.
A dependency listed in DESCRIPTION Suggests is not guaranteed to be installed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msigdbr_check_data(require_data = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msigdbr_check_data_+3A_require_data">require_data</code></td>
<td>
<p>Stop execution if the data package is not installed.</p>
</td></tr>
</table>

<hr>
<h2 id='msigdbr_collections'>List the collections available in the msigdbr package</h2><span id='topic+msigdbr_collections'></span>

<h3>Description</h3>

<p>List the collections available in the msigdbr package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msigdbr_collections(db_species = "Hs")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msigdbr_collections_+3A_db_species">db_species</code></td>
<td>
<p>Species abbreviation for the human or mouse databases (<code>"Hs"</code> or <code>"Mm"</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of the available collections.
</p>

<hr>
<h2 id='msigdbr_species'>List the species available in the msigdbr package</h2><span id='topic+msigdbr_species'></span>

<h3>Description</h3>

<p>List the species available in the msigdbr package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msigdbr_species()
</code></pre>


<h3>Value</h3>

<p>A data frame of the available species.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
