<!DOCTYPE html><html><head><title>Help for package MTAR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MTAR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#beta.example'><p>Example3: the genetic effect estimates and their standard errors</p></a></li>
<li><a href='#Get_UV_from_beta'><p>Compute the summary statistics given the genetic effect estimates and their standard errors</p></a></li>
<li><a href='#Get_UV_from_data'><p>Compute the summary statistics given the individual-level data</p></a></li>
<li><a href='#Get_UV_from_varU'><p>Compute the summary statistics given the score statistics and their variance.</p></a></li>
<li><a href='#Get_zeta'><p>Calculate Covariances of Z-scores between Traits from Overlapping Samples</p></a></li>
<li><a href='#MTAR'><p>Multiple-Traits Analysis of Rare-Variant Association Test</p></a></li>
<li><a href='#MTAR.example'><p>PNPLA2 Gene</p></a></li>
<li><a href='#rawdata'><p>Example1: individual-level data</p></a></li>
<li><a href='#varU.example'><p>Example2: the summary statistics and their variance</p></a></li>
<li><a href='#zeta.example'><p>Example4: the summary statistics of 737 common and null SNPs</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multi-Trait Analysis of Rare-Variant Association Study</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lan Luo &lt;lanluoruc@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform multi-trait rare-variant association tests using the summary statistics and adjust for possible sample overlap. Package is based on "Multi-Trait Analysis of Rare-Variant Association Summary Statistics using MTAR" by Luo, L., Shen, J., Zhang, H., Chhibber, A. Mehrotra, D.V., Tang, Z., 2019 (submitted).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>CompQuadForm (&ge; 1.4.3), MASS (&ge; 7.3-51.4), Matrix (&ge;
1.2.2), stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-04-22 19:24:54 UTC; lanluo</td>
</tr>
<tr>
<td>Author:</td>
<td>Lan Luo [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-04-22 20:44:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='beta.example'>Example3: the genetic effect estimates and their standard errors</h2><span id='topic+beta.example'></span>

<h3>Description</h3>

<p>An example of calculating U and V for MTAR given the genetic effect estimates and their standard errors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(beta.example)
</code></pre>


<h3>Format</h3>

<p>A list with 3 sublists:
</p>

<dl>
<dt>Beta</dt><dd><p>a list of genetic effect estimates, each sublist is <code class="reqn">m \times K</code> matrix for each study</p>
</dd>
<dt>Beta.se</dt><dd><p>a list of standard errors of genetic effect estimates, each sublist is <code class="reqn">m \times K</code> matrix for each study</p>
</dd>
<dt>R</dt><dd><p>a SNP correlation matrix for the union of SNPs appearing in all the studies</p>
</dd></dl>

<hr>
<h2 id='Get_UV_from_beta'>Compute the summary statistics given the genetic effect estimates and their standard errors</h2><span id='topic+Get_UV_from_beta'></span>

<h3>Description</h3>

<p>This function allows you to calculate the score summary statistics <code class="reqn">U</code> and their covariance matrix <code class="reqn">V</code> for MTAR, given the genetic effect estimates and their standard errors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Get_UV_from_beta(Beta, Beta.se, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Get_UV_from_beta_+3A_beta">Beta</code></td>
<td>
<p>a numeric list, each sublist containing estimation information of genetic effect estimates <code class="reqn">\beta</code> for each study. In each study, a numeric <code class="reqn">m \times K</code> matrix with each row as a SNP and each column as a separate trait. The number of traits and the number of SNPs in each study can be different but their names are required.</p>
</td></tr>
<tr><td><code id="Get_UV_from_beta_+3A_beta.se">Beta.se</code></td>
<td>
<p>a numeric list, each sublist containing the standard error of estimators information for each study. In each study, a numeric <code class="reqn">m \times K</code> matrix with each row as a SNP and each column as a separate trait.</p>
</td></tr>
<tr><td><code id="Get_UV_from_beta_+3A_r">R</code></td>
<td>
<p>a SNP correlation matrix, which should contain the correlation of all the SNPs in these studies.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing summary statistics for each traits, including the score summary statistics <code class="reqn">U</code> and their covariance matrix <code class="reqn">V</code>.
</p>


<h3>Author(s)</h3>

<p>Lan Luo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("beta.example")
attach(beta.example)
obs.stat &lt;- Get_UV_from_beta(Beta = Beta, Beta.se = Beta.se, R = R)
detach(beta.example)
</code></pre>

<hr>
<h2 id='Get_UV_from_data'>Compute the summary statistics given the individual-level data</h2><span id='topic+Get_UV_from_data'></span>

<h3>Description</h3>

<p>This function allows you to calculate the score summary statistics <code class="reqn">U</code> and their covariance matrix <code class="reqn">V</code> for MTAR, given the traits, covariates and genotype data sets. If one trait only takes no more than two values, this function will treat this trait as a binary trait.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Get_UV_from_data(traits, covariates, genotype, covariance = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Get_UV_from_data_+3A_traits">traits</code></td>
<td>
<p>a numeric list, each sublist containing trait information for each study. In each study, a numeric <code class="reqn">n \times K</code> matrix with each row as an independent individual and each column as a separate trait. If subject <code class="reqn">i</code> is without trait <code class="reqn">k</code>, the corresponding value is set as NA. The number of traits in each study can be different but the names of traits are required.</p>
</td></tr>
<tr><td><code id="Get_UV_from_data_+3A_covariates">covariates</code></td>
<td>
<p>a numeric list, each sublist containing covariates information for each study. In each study, a numeric <code class="reqn">n \times D</code> matrix with each row as an independent individual and each column as a covariate.</p>
</td></tr>
<tr><td><code id="Get_UV_from_data_+3A_genotype">genotype</code></td>
<td>
<p>a numeric list, each sublist containing genotype information for each study. In each study, a numeric <code class="reqn">n \times m</code> matrix with each row as an independent individual and each column as a SNP Each genotype should be coded as 0, 1, 2, and NA for AA, Aa, aa, and missing, where A and a represents a major and minor allele, respectively. The number of SNPs in each study can be different but the names of SNPs are required. Also, the number of studies must be the same in genotype, covariates and traits lists. The order of subject ID must be the same among traits, covariates, and genotype within each study.</p>
</td></tr>
<tr><td><code id="Get_UV_from_data_+3A_covariance">covariance</code></td>
<td>
<p>a logical value indicating whether to calculate the covariance matrix of score summary statistics <code class="reqn">U</code>. The default value is TRUE. If covariance is set as FALSE, then only the diagonal values of the covairance matrix are calculated, which is faster. In estimating the zeta matrix correcting for overlap samples, we recommend set covariance as FALSE in calculating the summary statistics for common variants. Since the number of common variants may be large and time-consuming.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing summary statistics for each trait. If covariance is TRUE, the score summary statistics <code class="reqn">U</code> and its covariance matrix <code class="reqn">V</code> are returned. Otherwise, only <code class="reqn">U</code> and the diagonal elements of covariance matrix  are returned.
</p>


<h3>Author(s)</h3>

<p>Lan Luo
</p>


<h3>References</h3>

<p>Hu, Y.J., Berndt, S.I., Gustafsson, S., Ganna, A., MÃ¤gi, R., Wheeler, E., Feitosa, M.F., Justice, A.E., Monda, K.L., Croteau-Chonka, D.C. and Day, F.R., 2013. Meta-analysis of gene-level associations for rare variants based on single-variant statistics. The American Journal of Human Genetics, 93(2), pp.236-248.
</p>
<p>Tang, Z.Z. and Lin, D.Y., 2015. Meta-analysis for discovering rare-variant associations: statistical methods and software programs. The American Journal of Human Genetics, 97(1), pp.35-53.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rawdata)
attach(rawdata)
obs.stat &lt;- Get_UV_from_data(traits = traits.dat,
                             covariates = cov.dat,
                             genotype = geno.dat,
                             covariance = TRUE)
obs.stat
detach(rawdata)
</code></pre>

<hr>
<h2 id='Get_UV_from_varU'>Compute the summary statistics given the score statistics and their variance.</h2><span id='topic+Get_UV_from_varU'></span>

<h3>Description</h3>

<p>This function allows you to calculate the score summary statistics <code class="reqn">U</code> and their covariance matrix <code class="reqn">V</code> for MTAR, given the score summary statistics and their variance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Get_UV_from_varU(U, varU, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Get_UV_from_varU_+3A_u">U</code></td>
<td>
<p>a numeric list, each sublist containing score summary statistics <code class="reqn">U</code> for each study. In each study, a numeric <code class="reqn">m \times K</code> matrix with each row as a SNP and each column as a separate trait. The number of traits and the number of SNPs in each study can be different but their names are required.</p>
</td></tr>
<tr><td><code id="Get_UV_from_varU_+3A_varu">varU</code></td>
<td>
<p>a numeric list, each sublist containing the variance of score summary statistics information for each study. In each study, a numeric <code class="reqn">m \times K</code> matrix with each row as a SNP and each column as a separate trait.</p>
</td></tr>
<tr><td><code id="Get_UV_from_varU_+3A_r">R</code></td>
<td>
<p>a SNP correlation matrix, which should contain the correlation of all the SNPs in these studies.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing summary statistics for each traits, the score summary statistics <code class="reqn">U</code> and their covariance matrix <code class="reqn">V</code>.
</p>


<h3>Author(s)</h3>

<p>Lan Luo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("varU.example")
attach(varU.example)
obs.stat &lt;- Get_UV_from_varU(U = U, varU = varU, R= R)
obs.stat
detach(varU.example)
</code></pre>

<hr>
<h2 id='Get_zeta'>Calculate Covariances of Z-scores between Traits from Overlapping Samples</h2><span id='topic+Get_zeta'></span>

<h3>Description</h3>

<p>This function allows you to estimate the matrix <code class="reqn">\zeta</code> to adjust for the potential sample overlap in the data set. Here we applied LD pruning (<code class="reqn">r^2 &lt; 0.1</code> in 500kb region) on 1000 Genome genotype dataset (hg19) as a list of reference independent SNPs. The SNP ID is chr:pos.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Get_zeta(Zscore, pval_cutoff = 0.05, Indp_common_snp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Get_zeta_+3A_zscore">Zscore</code></td>
<td>
<p>a numeric list, each sublist containing a vector of Z scores of SNPs with minor allele frequency (MAF) larger than 0.05. The chr:pos for each SNP is required.</p>
</td></tr>
<tr><td><code id="Get_zeta_+3A_pval_cutoff">pval_cutoff</code></td>
<td>
<p>a numeric value indicating the cutoff threshold of p-values. The default value is 0.05. Variants with p-value less than or equal to this threshold will be automatically removed.</p>
</td></tr>
<tr><td><code id="Get_zeta_+3A_indp_common_snp">Indp_common_snp</code></td>
<td>
<p>a numeric list of independent common SNPs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code class="reqn">K \times K</code> matrix <code class="reqn">\zeta</code>, where <code class="reqn">K</code> is the number of traits.
</p>


<h3>Author(s)</h3>

<p>Lan Luo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(zeta.example)
attach(zeta.example)
# Downloading independent common SNPs from 1000Genome data set.
githubURL &lt;- "https://github.com/lan/MTAR/blob/master/indp_snps.1KG.rda?raw=true"
utils::download.file(githubURL,"1kgfile")
load("1kgfile")
zeta1 &lt;- Get_zeta(Zscore = Zscore, Indp_common_snp = indp_snps.1KG)
zeta1
detach(zeta.example)

</code></pre>

<hr>
<h2 id='MTAR'>Multiple-Traits Analysis of Rare-Variant Association Test</h2><span id='topic+MTAR'></span>

<h3>Description</h3>

<p>Test for association between a set of rare SNPs and multiple traits with input of summary statistics, possibly from overlap samples. The input number of SNPs in each trait can be different, MTAR function will analyze the union of SNPs that show up in at least one trait, and automatically handle the non-polymorphic SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MTAR(
  U,
  V,
  MAF,
  MAF_UB = 0.05,
  zeta = NULL,
  genetic_cor.trait = NULL,
  rho.SNP = c(0, 0.5, 1),
  rho.trait = c(0, 0.5, 1),
  weight.SNP = c(1, 25)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MTAR_+3A_u">U</code></td>
<td>
<p>a numeric list, each sublist containing summary statistics U for each traits. The SNP IDs must be provided.</p>
</td></tr>
<tr><td><code id="MTAR_+3A_v">V</code></td>
<td>
<p>a numeric list, each sublist containing the corresponding covariance matrix of summary statistics. If your original summary statistics are other format, please use Get_UV_from_data, Get_UV_from_varU or Get_UV_from_beta to generate the summary statistics <code class="reqn">U</code> and <code class="reqn">V</code> for MTAR.</p>
</td></tr>
<tr><td><code id="MTAR_+3A_maf">MAF</code></td>
<td>
<p>a numeric vector containing minor allele frequency for the SNPs show up in at least one trait. The SNP IDs must be provided.</p>
</td></tr>
<tr><td><code id="MTAR_+3A_maf_ub">MAF_UB</code></td>
<td>
<p>a numeric value indicating the cutoff threshold of minor allele frequency for SNPs The default value is 0.05.</p>
</td></tr>
<tr><td><code id="MTAR_+3A_zeta">zeta</code></td>
<td>
<p>a numeric matrix containing the sample correlation of Z-scores over a large number of independent null common SNPs across genome. The default value is NULL, where MTAR assumes there are no overlap samples. However, if there is overlapping in subjects, zeta must be provided. zeta can be estimated using MTAR::Get_zeta.</p>
</td></tr>
<tr><td><code id="MTAR_+3A_genetic_cor.trait">genetic_cor.trait</code></td>
<td>
<p>a numeric matrix containing the genetic correlation among traits. The default value of genetic_cor.trait is NULL, where an exchangeable correlation structure with the correlation coefficient denoted by rho.trait (<code class="reqn">\rho_2</code>) is assumed. In this case, there is no difference between cMTAR and iMTAR.</p>
</td></tr>
<tr><td><code id="MTAR_+3A_rho.snp">rho.SNP</code></td>
<td>
<p>a numeric vector containing all the possible values of <code class="reqn">\rho_1</code>. The default value is c(0, 0.5, 1).</p>
</td></tr>
<tr><td><code id="MTAR_+3A_rho.trait">rho.trait</code></td>
<td>
<p>a numeric vector containing all the possible values of <code class="reqn">\rho_2</code>. The default value is c(0, 0.5, 1).</p>
</td></tr>
<tr><td><code id="MTAR_+3A_weight.snp">weight.SNP</code></td>
<td>
<p>a numeric vector containing the parameters in Beta density function to calculate the weight among SNPs. The default value is c(1, 25).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MTAR assumes that the genetic effect estimates <code class="reqn">\beta</code> has covariance matrix <code class="reqn">B</code>, which is a Kronecker product of two pre-specified matrices: among-variant effect covariance <code class="reqn">B_1</code> and among-trait effect covariance <code class="reqn">B_2</code>. An exchaneable correlation structure with the correlation coefficient denoted by rho.SNPs (<code class="reqn">\rho_1</code>) for <code class="reqn">B_1</code> is assumed. The default MTAR requires the input of genetic correlation matrix genetoc_cor.trait, if missing, then an exchaneable correlation structure for rho.trait (<code class="reqn">\rho_2</code>) is assumed. The default weight of <code class="reqn">B_1</code> is <code class="reqn">dBeta(MAF, 1, 25)</code>, which can be changed freely by users.
</p>


<h3>Value</h3>

<p>a list of p-values of MTAR-O, cMTAR, iMTAR and cctP as well as ancillary information. Here cctP is the Cauchy-combined p-value of SKAT and burden tests with default weight <code class="reqn">dBeta(MAF, 1, 25)</code>.
</p>


<h3>Author(s)</h3>

<p>Lan Luo
</p>
<p>Lan Luo
</p>


<h3>References</h3>

<p>Liu, Y., Chen, S., Li, Z., Morrison, A.C., Boerwinkle, E. and Lin, X., 2019. ACAT: A fast and powerful p value combination method for rare-variant analysis in sequencing studies. The American Journal of Human Genetics, 104(3), pp.410-421.
</p>
<p>Liu, Y. and Xie, J., 2019. Cauchy combination test: a powerful test with analytic p-value calculation under arbitrary dependency structures. Journal of the American Statistical Association, pp.1-18.
</p>
<p>Luo, L., Shen, J., Zhang, H., Chhibber, A., Mehrotra, D. V., Tang, Z., 2019. Multi-trait analysis of rare-variant association summary statistics using MTAR.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MTAR.example)
attach(MTAR.example)
pval &lt;- MTAR(U = U, V = V, MAF = MAF, genetic_cor.trait = genetic_cor.trait,
             zeta = zeta)
pval
detach(MTAR.example)

</code></pre>

<hr>
<h2 id='MTAR.example'>PNPLA2 Gene</h2><span id='topic+MTAR.example'></span>

<h3>Description</h3>

<p>PNPLA2 Gene Inofmation required by MTAR function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(MTAR.example)
</code></pre>


<h3>Format</h3>

<p>A list with 6 sublists:
</p>

<dl>
<dt>annotation</dt><dd><p>some annotation information of gene PNPLA2</p>
</dd>
<dt>U</dt><dd><p>a list containing the summary statistics for each trait</p>
</dd>
<dt>V</dt><dd><p>a list containing the covariance matrix of summary statistics for each trait</p>
</dd>
<dt>MAF</dt><dd><p>the minor allele frequency of all the rare variants in this gene</p>
</dd>
<dt>genetic_cor.trait</dt><dd><p>the genetic correlation among traits</p>
</dd>
<dt>zeta</dt><dd><p>the overlapping estimation matrix, which is approximated by the sample correlation matrix of the scaled summary statistics over a large number of independent null common SNPs</p>
</dd></dl>

<hr>
<h2 id='rawdata'>Example1: individual-level data</h2><span id='topic+rawdata'></span>

<h3>Description</h3>

<p>An example of calculating U and V for MTAR given the individual-level data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(rawdata)
</code></pre>


<h3>Format</h3>

<p>A list with 3 sublists:
</p>

<dl>
<dt>geno.dat</dt><dd><p>a list of genotype data, each sublist is <code class="reqn">n \times m</code> matrix for each study</p>
</dd>
<dt>traits.dat</dt><dd><p>a list of trait data, each sublist is <code class="reqn">n \times K</code> matrix for each study</p>
</dd>
<dt>cov.dat</dt><dd><p>a list of covariates data, each sublist is <code class="reqn">n \times D</code> matrix for each study</p>
</dd></dl>

<p>There are 3 studies, 3 continuous traits and 10 rare variants. Specifically, there are 1500 subjects in study1, but each subject only has one trait measurement. In study2 and study3, the sample size is 500 and each subject has two or three traits measurements.</p>

<hr>
<h2 id='varU.example'>Example2: the summary statistics and their variance</h2><span id='topic+varU.example'></span>

<h3>Description</h3>

<p>An example of calculating U and V for MTAR given the summary statistics and its variance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(varU.example)
</code></pre>


<h3>Format</h3>

<p>A list with 3 sublists:
</p>

<dl>
<dt>U</dt><dd><p>a list of summary statistics, each sublist is <code class="reqn">m \times K</code> matrix for each study</p>
</dd>
<dt>varU</dt><dd><p>a list of variance of summary statistics, each sublist is <code class="reqn">m \times K</code> matrix for each study</p>
</dd>
<dt>R</dt><dd><p>a SNP correlation matrix for the union of SNPs appearing in all the studies</p>
</dd></dl>

<hr>
<h2 id='zeta.example'>Example4: the summary statistics of 737 common and null SNPs</h2><span id='topic+zeta.example'></span>

<h3>Description</h3>

<p>An example of estimating matrix <code class="reqn">\zeta</code> given the summary statistics information of null and common SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(zeta.example)
</code></pre>


<h3>Format</h3>

<p>A list with 1 sublist:
</p>

<dl>
<dt>Zscore</dt><dd><p>a list, each sublist contains the Z-scores of 737 null and common SNPs for each trait</p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
