<!DOCTYPE html><html lang="en"><head><title>Help for package mulea</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mulea}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#filter_ontology'><p>Filter Ontology</p></a></li>
<li><a href='#gsea-class'><p>Gene Set Enrichment Analysis (GSEA)</p></a></li>
<li><a href='#list_to_gmt'><p>Convert a list to ontology (GMT) <code>data.frame</code>.</p></a></li>
<li><a href='#MuleaHypergeometricTest-class'><p>PRIVATE class : An S4 class to represent a Hypergeometric tests in</p>
mulea.</a></li>
<li><a href='#ora-class'><p>An S4 class to represent a set based tests in mulea.</p></a></li>
<li><a href='#plot_barplot'><p>Plot Barplot</p></a></li>
<li><a href='#plot_graph'><p>Plot Graph (Network)</p></a></li>
<li><a href='#plot_heatmap'><p>Plot Heatmap</p></a></li>
<li><a href='#plot_lollipop'><p>Plot Lollipop</p></a></li>
<li><a href='#read_gmt'><p>Read GMT File</p></a></li>
<li><a href='#reshape_results'><p>Reshape Results</p></a></li>
<li><a href='#run_test'><p>Run enrichment analysis procedure</p></a></li>
<li><a href='#SetBasedEnrichmentTest-class'><p>PRIVATE class : An S4 class to represent a Hypergeometric tests in</p>
mulea.</a></li>
<li><a href='#SubramanianTest-class'><p>PRIVATE class : An S4 class to represent a ranked based tests in mulea.</p></a></li>
<li><a href='#write_gmt'><p>Write GMT file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Enrichment Analysis Using Multiple Ontologies and False
Discovery Rate</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Background - Traditional gene set enrichment analyses are 
    typically limited to a few ontologies and do not account for the 
    interdependence of gene sets or terms, resulting in overcorrected p-values. 
    To address these challenges, we introduce mulea, an R package offering 
    comprehensive overrepresentation and functional enrichment analysis. 
    Results - mulea employs a progressive empirical false discovery rate 
    (eFDR) method, specifically designed for interconnected biological data, 
    to accurately identify significant terms within diverse ontologies. mulea 
    expands beyond traditional tools by incorporating a wide range of 
    ontologies, encompassing Gene Ontology, pathways, regulatory elements, 
    genomic locations, and protein domains. This flexibility enables 
    researchers to tailor enrichment analysis to their specific questions, 
    such as identifying enriched transcriptional regulators in gene expression 
    data or overrepresented protein domains in protein sets. To facilitate 
    seamless analysis, mulea provides gene sets (in standardised GMT format) 
    for 27 model organisms, covering 22 ontology types from 16 databases and 
    various identifiers resulting in almost 900 files. Additionally, the 
    muleaData ExperimentData Bioconductor package simplifies access to these 
    pre-defined ontologies. Finally, mulea's architecture allows for easy 
    integration of user-defined ontologies, or GMT files from external 
    sources (e.g., MSigDB or Enrichr), expanding its applicability across 
    diverse research areas. Conclusions - mulea is distributed as a CRAN R 
    package. It offers researchers a powerful and flexible toolkit for 
    functional enrichment analysis, addressing limitations of traditional 
    tools with its progressive eFDR and by supporting a variety of ontologies. 
    Overall, mulea fosters the exploration of diverse biological questions 
    across various model organisms.</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Annotation, DifferentialExpression, GeneExpression,
GeneSetEnrichment, GO, GraphAndNetwork, MultipleComparison,
Pathways, Reactome, Software, Transcription, Visualization</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table (&ge; 1.13.0), dplyr, fgsea (&ge; 1.0.2), ggplot2,
ggraph (&ge; 2.0.3), magrittr (&ge; 2.0.3), methods, parallel (&ge;
4.0.2), plyr (&ge; 1.8.4), Rcpp, readr, rlang, scales, stats,
stringi, tibble, tidygraph, tidyverse</td>
</tr>
<tr>
<td>Suggests:</td>
<td>devtools, knitr, rmarkdown, testthat (&ge; 3.1.4)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ELTEbioinformatics/mulea">https://github.com/ELTEbioinformatics/mulea</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ELTEbioinformatics/mulea/issues">https://github.com/ELTEbioinformatics/mulea/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-19 08:47:51 UTC; tamas</td>
</tr>
<tr>
<td>Author:</td>
<td>Cezary Turek <a href="https://orcid.org/0000-0002-1445-5378"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Marton Olbei <a href="https://orcid.org/0000-0002-4903-6237"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Tamas Stirling <a href="https://orcid.org/0000-0002-8964-6443"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Gergely Fekete <a href="https://orcid.org/0000-0001-9939-4860"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Ervin Tasnadi <a href="https://orcid.org/0000-0002-4713-5397"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Leila Gul [aut],
  Balazs Bohar <a href="https://orcid.org/0000-0002-3033-5448"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Balazs Papp <a href="https://orcid.org/0000-0003-3093-8852"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Wiktor Jurkowski <a href="https://orcid.org/0000-0002-7820-1991"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Eszter Ari <a href="https://orcid.org/0000-0001-7774-1067"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tamas Stirling &lt;stirling.tamas@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-19 10:20:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='filter_ontology'>Filter Ontology</h2><span id='topic+filter_ontology'></span>

<h3>Description</h3>

<p>Filtering ontology to contain entries having number of elements
(genes or proteins) between a given range. The reason for this is
enrichment analysis results can sometimes be skewed by overly specific or
broad entries. Filtering ontologies allows you to customize the size of
ontology entries, ensuring your analysis aligns with your desired scope.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_ontology(gmt, min_nr_of_elements = NULL, max_nr_of_elements = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_ontology_+3A_gmt">gmt</code></td>
<td>
<p>A <code>data.frame</code> which contains the entries
(gene or protein sets), imported from a GMT file with the
<code>read_gmt</code> function.</p>
</td></tr>
<tr><td><code id="filter_ontology_+3A_min_nr_of_elements">min_nr_of_elements</code></td>
<td>
<p>Minimum number of elements. Ontology entries
containing as many or fewer elements (genes or proteins) will be excluded.</p>
</td></tr>
<tr><td><code id="filter_ontology_+3A_max_nr_of_elements">max_nr_of_elements</code></td>
<td>
<p>Maximum number of elements. Ontology entries
containing as many or more elements (genes or proteins) will be excluded.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a <code>data.frame</code>which contains the entries (gene or protein
sets) in a similar format that produced by the <code>read_gmt</code> function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file(
    package="mulea", "extdata", 
    "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
tf_gmt_filtered &lt;- filter_ontology(gmt = tf_gmt,
        min_nr_of_elements = 3,
        max_nr_of_elements = 400)
</code></pre>

<hr>
<h2 id='gsea-class'>Gene Set Enrichment Analysis (GSEA)</h2><span id='topic+gsea-class'></span><span id='topic+gsea'></span><span id='topic+run_test+2Cgsea-method'></span>

<h3>Description</h3>

<p>An S4 class to represent the gsea tests in mulea.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'gsea'
run_test(model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gsea-class_+3A_model">model</code></td>
<td>
<p>Object of S4 class representing the mulea test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>GSEA object. This object represents the result of the
<code>gsea</code> tests.
</p>
<p>run_test method for GSEA object. Returns results of
the enrichment analysis.
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>run_test(gsea)</code>: runs test calculations.
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>gmt</code></dt><dd><p>A <code>data.frame</code> representing the ontology GMT.</p>
</dd>
<dt><code>element_names</code></dt><dd><p>A vector of elements names
(gene or protein names or identifiers) to include in the analysis.</p>
</dd>
<dt><code>element_scores</code></dt><dd><p>A vector of numeric values representing a
score (<em>e.g.</em> <em>p</em>-value, <em>z</em>-score, log fold change) for each
'element_name', in the same number and order as element_name.</p>
</dd>
<dt><code>gsea_power</code></dt><dd><p>A power of weight. Default value is 1.</p>
</dd>
<dt><code>element_score_type</code></dt><dd><p>Defines the GSEA score type.
</p>

<ul>
<li><p> 'pos': Only positive element_scores
</p>
</li>
<li><p> 'neg': Only negative element_scores
</p>
</li>
<li><p> 'std': standard, containing both positive and negative scores
Default value is 'std'.
</p>
</li></ul>
</dd>
<dt><code>number_of_permutations</code></dt><dd><p>The number of permutations used in
<code>gsea</code> test. Default value is 1000.</p>
</dd>
<dt><code>test</code></dt><dd><p>character</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file(
    package="mulea", "extdata",
    "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
tf_gmt_filtered &lt;- filter_ontology(gmt = tf_gmt, min_nr_of_elements = 3, 
    max_nr_of_elements = 400)

# loading the example `data.frame`
scored_gene_tab &lt;- read.delim(file = system.file(package = "mulea", "extdata", 
        "ordered_set.tsv"))

# creating the GSEA model
gsea_model &lt;- gsea(gmt = tf_gmt_filtered,
        # the names of elements to test
        element_names = scored_gene_tab$Gene.symbol,
        # the logFC-s of elements to test
        element_scores = scored_gene_tab$logFC,
        # consider elements having positive logFC values only
        element_score_type = "pos",
        # the number of permutations
        number_of_permutations = 10000)
library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file(package="mulea", "extdata", 
        "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
tf_gmt_filtered &lt;- filter_ontology(gmt = tf_gmt, min_nr_of_elements = 3, 
        max_nr_of_elements = 400)

# loading the example `data.frame`
scored_gene_tab &lt;- read.delim(file = system.file(package = "mulea", "extdata", 
        "ordered_set.tsv"))

# creating the GSEA model
gsea_model &lt;- gsea(gmt = tf_gmt_filtered,
        # the names of elements to test
        element_names = scored_gene_tab$Gene.symbol,
        # the logFC-s of elements to test
        element_scores = scored_gene_tab$logFC,
        # consider elements having positive logFC values only
        element_score_type = "pos",
        # the number of permutations
        number_of_permutations = 10000)

# running the test
gsea_results &lt;- run_test(gsea_model)
</code></pre>

<hr>
<h2 id='list_to_gmt'>Convert a list to ontology (GMT) <code>data.frame</code>.</h2><span id='topic+list_to_gmt'></span>

<h3>Description</h3>

<p>Converts a list of ontology elements (gene sets) to an ontology
(GMT) <code>data.frame</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_to_gmt(gmt_list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="list_to_gmt_+3A_gmt_list">gmt_list</code></td>
<td>
<p>A list with named character vectors. The name will become the
'ontology_id', and the elements in the vector will become the
'list_of_values' in the ontology (GMT) <code>data.frame</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns ontology (GMT) <code>data.frame</code> where the 'ontology_name'
contains random unique strings.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(mulea)

# creating a list of gene sets
ontology_list &lt;- list(gene_set1 = c("gene1", "gene2", "gene3"),
        gene_set2 = c("gene4", "gene5", "gene6"))

# converting the list to a ontology (GMT) object
new_ontology_object &lt;- list_to_gmt(ontology_list)
</code></pre>

<hr>
<h2 id='MuleaHypergeometricTest-class'>PRIVATE class : An S4 class to represent a Hypergeometric tests in
mulea.</h2><span id='topic+MuleaHypergeometricTest-class'></span><span id='topic+MuleaHypergeometricTest'></span><span id='topic+run_test+2CMuleaHypergeometricTest-method'></span>

<h3>Description</h3>

<p>PRIVATE class : An S4 class to represent a Hypergeometric tests in
mulea.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'MuleaHypergeometricTest'
run_test(model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MuleaHypergeometricTest-class_+3A_model">model</code></td>
<td>
<p>Object of s4 class represents mulea Test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>MuleaHypergeometricTest object. Used as private function.
</p>
<p>run_test method for MuleaHypergeometricTest object. Used as private
function.
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>run_test(MuleaHypergeometricTest)</code>: runs test calculations.
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>gmt</code></dt><dd><p>A data.frame representing the GMT model.</p>
</dd>
<dt><code>element_names</code></dt><dd><p>Data to be analysed across the model.</p>
</dd>
<dt><code>background_element_names</code></dt><dd><p>Background data used for the test.</p>
</dd>
</dl>

<hr>
<h2 id='ora-class'>An S4 class to represent a set based tests in mulea.</h2><span id='topic+ora-class'></span><span id='topic+ora'></span>

<h3>Description</h3>

<p>An S4 class to represent a set based tests in mulea.
</p>


<h3>Value</h3>

<p>ora object. This object represents the result of the
overrepresentation test in mulea.
</p>


<h3>Slots</h3>


<dl>
<dt><code>method</code></dt><dd><p>The overrepresentation (ora) method. Possible values:
&quot;Hypergeometric&quot;, &quot;SetBasedEnrichment&quot;.</p>
</dd>
<dt><code>gmt</code></dt><dd><p>A <code>data.frame</code> representing the ontology GMT.</p>
</dd>
<dt><code>element_names</code></dt><dd><p>A vector of elements names
(gene or protein names or identifiers) representing the target set to
analyse. For example
differentially expressed genes.</p>
</dd>
<dt><code>background_element_names</code></dt><dd><p>A vector of elements names
(gene or protein names or identifiers) representing all the
elements involved in the previous analyses For example all
genes that were measured in differential expression analysis.</p>
</dd>
<dt><code>p_value_adjustment_method</code></dt><dd><p>A character string representing the
type of the <em>p</em>-value adjustment method. Possible values:
</p>

<ul>
<li><p> 'eFDR': empirical false discovery rate correction method
</p>
</li>
<li><p> all <code>method</code> options from <code>stats::p.adjust</code> documentation.
</p>
</li></ul>
</dd>
<dt><code>number_of_permutations</code></dt><dd><p>A numeric value representing the number of
permutations used to calculate the eFDR values. Default value is 10000.</p>
</dd>
<dt><code>nthreads</code></dt><dd><p>Number of processor's threads to use in calculations.</p>
</dd>
<dt><code>random_seed</code></dt><dd><p>Optional natural number (1, 2, 3, ...) setting the seed
for the random generator, to make the results reproducible.</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file(
    package="mulea", "extdata", 
    "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
tf_gmt_filtered &lt;- filter_ontology(gmt = tf_gmt, min_nr_of_elements = 3, 
        max_nr_of_elements = 400)

# loading the example data
sign_genes &lt;- readLines(system.file(package = "mulea", "extdata", 
        "target_set.txt"))
background_genes &lt;- readLines(system.file(package="mulea", "extdata", 
        "background_set.txt"))

# creating the ORA model
ora_model &lt;- ora(gmt = tf_gmt_filtered, 
        # the test set variable
        element_names = sign_genes, 
        # the background set variable
        background_element_names = background_genes, 
        # the p-value adjustment method
        p_value_adjustment_method = "eFDR", 
        # the number of permutations
        number_of_permutations = 10000,
        # the number of processor threads to use
        nthreads = 2)
# running the ORA
ora_results &lt;- run_test(ora_model)
</code></pre>

<hr>
<h2 id='plot_barplot'>Plot Barplot</h2><span id='topic+plot_barplot'></span>

<h3>Description</h3>

<p>Plots barplot of p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_barplot(
  reshaped_results,
  ontology_id_colname = "ontology_id",
  selected_rows_to_plot = NULL,
  p_value_type_colname = "eFDR",
  p_value_max_threshold = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_barplot_+3A_reshaped_results">reshaped_results</code></td>
<td>
<p>'data.table' in relaxed form, obtained as the
output of the <code>reshape_results</code> function.
The data source for generating the barplot.</p>
</td></tr>
<tr><td><code id="plot_barplot_+3A_ontology_id_colname">ontology_id_colname</code></td>
<td>
<p>Character, specifies the column name that contains
ontology IDs in the input data.</p>
</td></tr>
<tr><td><code id="plot_barplot_+3A_selected_rows_to_plot">selected_rows_to_plot</code></td>
<td>
<p>A numeric vector specifying which rows of the
reshaped results 'data.frame' should be included in the plot.
Default is 'NULL'.</p>
</td></tr>
<tr><td><code id="plot_barplot_+3A_p_value_type_colname">p_value_type_colname</code></td>
<td>
<p>Character, specifies the column name for
<em>p</em>-values in the input data.
Default is 'eFDR'.</p>
</td></tr>
<tr><td><code id="plot_barplot_+3A_p_value_max_threshold">p_value_max_threshold</code></td>
<td>
<p>Numeric, representing the maximum <em>p</em>-value
threshold for filtering data.
Default is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Create a customized barplot of <em>p</em>-values, facilitating visual exploration
and analysis of statistical significance within ontology categories.
</p>


<h3>Value</h3>

<p>Returns a barplot.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reshape_results">reshape_results</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file(package="mulea", "extdata", 
    "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
tf_gmt_filtered &lt;- filter_ontology(gmt = tf_gmt, min_nr_of_elements = 3, 
    max_nr_of_elements = 400)

# loading the example data
sign_genes &lt;- readLines(system.file(package = "mulea", "extdata", 
    "target_set.txt"))
background_genes &lt;- readLines(system.file(
    package="mulea", "extdata", "background_set.txt"))

# creating the ORA model
ora_model &lt;- ora(gmt = tf_gmt_filtered,
    # the test set variable
    element_names = sign_genes, 
    # the background set variable
    background_element_names = background_genes, 
    # the p-value adjustment method
    p_value_adjustment_method = "eFDR", 
    # the number of permutations
    number_of_permutations = 10000,
    # the number of processor threads to use
    nthreads = 2)
# running the ORA
ora_results &lt;- run_test(ora_model)

# reshaping results for visualisation
ora_reshaped_results &lt;- reshape_results(model = ora_model,
    model_results = ora_results,
    # choosing which column to use for the indication of significance
    p_value_type_colname = "eFDR")

# Plot barplot
plot_barplot(reshaped_results = ora_reshaped_results,
    # the column containing the names we wish to plot
    ontology_id_colname = "ontology_id",
    # upper threshold for the value indicating the significance
    p_value_max_threshold = 0.05,
    # column that indicates the significance values
    p_value_type_colname = "eFDR")
</code></pre>

<hr>
<h2 id='plot_graph'>Plot Graph (Network)</h2><span id='topic+plot_graph'></span>

<h3>Description</h3>

<p>Plots graph representation of enrichment results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_graph(
  reshaped_results,
  ontology_id_colname = "ontology_id",
  ontology_element_colname = "element_id_in_ontology",
  shared_elements_min_threshold = 0,
  p_value_type_colname = "eFDR",
  p_value_max_threshold = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_graph_+3A_reshaped_results">reshaped_results</code></td>
<td>
<p>Character, the input <code>data.table</code> containing the
reshaped results.</p>
</td></tr>
<tr><td><code id="plot_graph_+3A_ontology_id_colname">ontology_id_colname</code></td>
<td>
<p>Character, the name of the column in the reshaped
results that contains ontology identifiers or names. Default value is
'ontology_id'.</p>
</td></tr>
<tr><td><code id="plot_graph_+3A_ontology_element_colname">ontology_element_colname</code></td>
<td>
<p>Character, the name of the column in the
reshaped results that contains element identifiers within the ontology.
Default value is 'element_id_in_ontology'.</p>
</td></tr>
<tr><td><code id="plot_graph_+3A_shared_elements_min_threshold">shared_elements_min_threshold</code></td>
<td>
<p>Numeric, threshold specifying the
minimum number of shared elements required between two ontologies to
consider them connected by an edge on the graph. Default value is 0.</p>
</td></tr>
<tr><td><code id="plot_graph_+3A_p_value_type_colname">p_value_type_colname</code></td>
<td>
<p>Character, the name of the column in the reshaped
results that contains the type of <em>p</em>-values associated with the ontology
elements. Default value is 'eFDR'.</p>
</td></tr>
<tr><td><code id="plot_graph_+3A_p_value_max_threshold">p_value_max_threshold</code></td>
<td>
<p>Numeric, a threshold value for filtering
rows in the reshaped results based on the
<em>p</em>-values. Rows with <em>p</em>-values greater
than this threshold will be filtered out.
Default value is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates a graph (network) visualization of the
enriched ontology entries. On the plot each node represents an ontology
entry below a given <em>p</em>-value threshold, and is coloured based on its
significance level. A connection (edge) is drawn between two nodes if they
share at least one common element (gene) belonging to the target set &ndash; in
the case of ORA results &ndash; or all analysed elements &ndash; in the case of GSEA
results.
</p>


<h3>Value</h3>

<p>Returns a graph plot.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reshape_results">reshape_results</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file(package="mulea", "extdata", 
    "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
tf_gmt_filtered &lt;- filter_ontology(gmt = tf_gmt, min_nr_of_elements = 3, 
    max_nr_of_elements = 400)

# loading the example data
sign_genes &lt;- readLines(system.file(package = "mulea", "extdata", 
    "target_set.txt"))
background_genes &lt;- readLines(system.file(
    package="mulea", "extdata", "background_set.txt"))

# creating the ORA model
ora_model &lt;- ora(gmt = tf_gmt_filtered,
    # the test set variable
    element_names = sign_genes, 
    # the background set variable
    background_element_names = background_genes, 
    # the p-value adjustment method
    p_value_adjustment_method = "eFDR", 
    # the number of permutations
    number_of_permutations = 10000,
    # the number of processor threads to use
    nthreads = 2)
# running the ORA
ora_results &lt;- run_test(ora_model)

# reshaping results for visualisation
ora_reshaped_results &lt;- reshape_results(model = ora_model,
    model_results = ora_results,
    # choosing which column to use for the indication of significance
    p_value_type_colname = "eFDR")

# Plot graph
plot_graph(reshaped_results = ora_reshaped_results,
    # the column containing the names we wish to plot
    ontology_id_colname = "ontology_id",
    # upper threshold for the value indicating the significance
    p_value_max_threshold = 0.05,
    # column that indicates the significance values
    p_value_type_colname = "eFDR")
</code></pre>

<hr>
<h2 id='plot_heatmap'>Plot Heatmap</h2><span id='topic+plot_heatmap'></span>

<h3>Description</h3>

<p>Plots heatmap of enriched terms and obtained p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heatmap(
  reshaped_results,
  ontology_id_colname = "ontology_id",
  ontology_element_colname = "element_id_in_ontology",
  p_value_type_colname = "eFDR",
  p_value_max_threshold = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heatmap_+3A_reshaped_results">reshaped_results</code></td>
<td>
<p>data.table in relaxed form, obtained as the output
of the <code>reshape_results</code> function. The data source for generating the
barplot.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_ontology_id_colname">ontology_id_colname</code></td>
<td>
<p>Character, specifies the column name that contains
ontology IDs in the input data.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_ontology_element_colname">ontology_element_colname</code></td>
<td>
<p>Character, specifying the column name that
contains ontology elements or terms in the input data. Default:
'element_id_in_ontology'.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_p_value_type_colname">p_value_type_colname</code></td>
<td>
<p>Character, specifies the column name for p-values
in the input data. Default is 'eFDR'.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_p_value_max_threshold">p_value_max_threshold</code></td>
<td>
<p>Numeric, representing the maximum p-value
threshold for filtering data. Default is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>plot_heatmap</code> function provides a convenient way to create a ggplot2
heatmap illustrating the significance of enriched terms within ontology
categories based on their associated p-values.
</p>


<h3>Value</h3>

<p>Returns a ggplot2 heatmap.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reshape_results">reshape_results</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file(package="mulea", "extdata", 
    "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
tf_gmt_filtered &lt;- filter_ontology(gmt = tf_gmt, min_nr_of_elements = 3, 
    max_nr_of_elements = 400)

# loading the example data
sign_genes &lt;- readLines(system.file(package = "mulea", "extdata", 
    "target_set.txt"))
background_genes &lt;- readLines(system.file(
    package="mulea", "extdata", "background_set.txt"))

# creating the ORA model
ora_model &lt;- ora(gmt = tf_gmt_filtered,
    # the test set variable
    element_names = sign_genes, 
    # the background set variable
    background_element_names = background_genes, 
    # the p-value adjustment method
    p_value_adjustment_method = "eFDR", 
    # the number of permutations
    number_of_permutations = 10000,
    # the number of processor threads to use
    nthreads = 2)
# running the ORA
ora_results &lt;- run_test(ora_model)

# reshaping results for visualisation
ora_reshaped_results &lt;- reshape_results(
    model = ora_model,
    model_results = ora_results,
    # choosing which column to use for the indication of significance
    p_value_type_colname = "eFDR")

# Plot heatmap                                        
plot_heatmap(reshaped_results = ora_reshaped_results,
    # the column containing the names we wish to plot
    ontology_id_colname = "ontology_id",
    # column that indicates the significance values
    p_value_type_colname = "eFDR")
</code></pre>

<hr>
<h2 id='plot_lollipop'>Plot Lollipop</h2><span id='topic+plot_lollipop'></span>

<h3>Description</h3>

<p>Plots lollipop plot of p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_lollipop(
  reshaped_results,
  ontology_id_colname = "ontology_id",
  selected_rows_to_plot = NULL,
  p_value_type_colname = "eFDR",
  p_value_max_threshold = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_lollipop_+3A_reshaped_results">reshaped_results</code></td>
<td>
<p>data.table in relaxed form, obtained as the output
of the <code>reshape_results</code> function. The data source for generating the
barplot.</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_ontology_id_colname">ontology_id_colname</code></td>
<td>
<p>Character, specifies the column name that contains
ontology IDs in the input data.</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_selected_rows_to_plot">selected_rows_to_plot</code></td>
<td>
<p>A numeric vector specifying which rows of the
reshaped results data frame should be included in the plot. Default is NULL.
frame should be included in the plot?</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_p_value_type_colname">p_value_type_colname</code></td>
<td>
<p>Character, specifies the column name for p-values
in the input data. Default is 'eFDR'.</p>
</td></tr>
<tr><td><code id="plot_lollipop_+3A_p_value_max_threshold">p_value_max_threshold</code></td>
<td>
<p>Numeric, representing the maximum p-value
threshold for filtering data. Default is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Create a customized  lollipop plot of p-values, facilitating visual
exploration and analysis of statistical significance within ontology
categories.
</p>


<h3>Value</h3>

<p>Returns a lollipop plot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reshape_results">reshape_results</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file(package="mulea", "extdata", 
    "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
tf_gmt_filtered &lt;- filter_ontology(gmt = tf_gmt, min_nr_of_elements = 3, 
    max_nr_of_elements = 400)

# loading the example data
sign_genes &lt;- readLines(system.file(package = "mulea", "extdata", 
    "target_set.txt"))
background_genes &lt;- readLines(system.file(
    package="mulea", "extdata", "background_set.txt"))

# creating the ORA model
ora_model &lt;- ora(gmt = tf_gmt_filtered,
    # the test set variable
    element_names = sign_genes, 
    # the background set variable
    background_element_names = background_genes, 
    # the p-value adjustment method
    p_value_adjustment_method = "eFDR", 
    # the number of permutations
    number_of_permutations = 10000,
    # the number of processor threads to use
    nthreads = 2)
# running the ORA
ora_results &lt;- run_test(ora_model)

# reshaping results for visualisation
ora_reshaped_results &lt;- reshape_results(
    model = ora_model,
    model_results = ora_results,
    # choosing which column to use for the indication of significance
    p_value_type_colname = "eFDR")
        
# Plot lollipop
plot_lollipop(reshaped_results = ora_reshaped_results,
    # the column containing the names we wish to plot
    ontology_id_colname = "ontology_id",
    # upper threshold for the value indicating the significance
    p_value_max_threshold = 0.05,
    # column that indicates the significance values
    p_value_type_colname = "eFDR")

</code></pre>

<hr>
<h2 id='read_gmt'>Read GMT File</h2><span id='topic+read_gmt'></span>

<h3>Description</h3>

<p>Reads gene set or ontology data from a Gene Matrix Transposed
(GMT) file and parse into a <code>data.frame</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_gmt(file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_gmt_+3A_file">file</code></td>
<td>
<p>Character, a path to a file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code>data.frame</code> with three columns:
</p>

<ul>
<li><p> 'ontology_id': Ontology identifier that uniquely identifies the element
within the referenced ontology.
</p>
</li>
<li><p> 'ontology_name': Ontology name or description that provides a
user-friendly label or textual description for the 'ontology_id'.
</p>
</li>
<li><p> 'list_of_values': Associated genes or proteins that is a vector of
identifiers of genes or proteins belonging to the 'ontology_id'.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># import example gene set
library(mulea)
tf_gmt &lt;- read_gmt(file = system.file(
    package="mulea", "extdata", 
    "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
</code></pre>

<hr>
<h2 id='reshape_results'>Reshape Results</h2><span id='topic+reshape_results'></span>

<h3>Description</h3>

<p>This function takes a model and model_results data, reshapes
them into a suitable format for plotting, and returns the resulting data
frame, which can be used for further analysis or visualization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reshape_results(
  model = NULL,
  model_results = NULL,
  model_ontology_col_name = "ontology_id",
  ontology_id_colname = "ontology_id",
  p_value_type_colname = "eFDR",
  p_value_max_threshold = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reshape_results_+3A_model">model</code></td>
<td>
<p>a mulea model, created by the
<code>ora</code> or the <code>gsea</code> functions.</p>
</td></tr>
<tr><td><code id="reshape_results_+3A_model_results">model_results</code></td>
<td>
<p>Result <code>data.frame</code>
returned by the <code>run_test</code> function.</p>
</td></tr>
<tr><td><code id="reshape_results_+3A_model_ontology_col_name">model_ontology_col_name</code></td>
<td>
<p>Character, specifies the column name in the
model that contains ontology IDs. It defines which column in the model
should be used for matching ontology IDs. Possible values are 'ontology_id'
and 'ontology_name'. The default value is 'ontology_id'.</p>
</td></tr>
<tr><td><code id="reshape_results_+3A_ontology_id_colname">ontology_id_colname</code></td>
<td>
<p>Character, specifies the column name for ontology
IDs in the model results. It indicates which column in the model results
contains ontology IDs for merging. Possible values are 'ontology_id' and
'ontology_name'. The default value is 'ontology_id'.</p>
</td></tr>
<tr><td><code id="reshape_results_+3A_p_value_type_colname">p_value_type_colname</code></td>
<td>
<p>Character, specifies the column name
for the type or raw or adjusted <em>p</em>-value in the result
<code>data.frame</code> returned by the <code>run_test</code> function.
The default value is 'eFDR'.</p>
</td></tr>
<tr><td><code id="reshape_results_+3A_p_value_max_threshold">p_value_max_threshold</code></td>
<td>
<p>Logical, indicating whether to apply a
<em>p</em>-value threshold when filtering the resulting data. If TRUE,
the function filters the data based on a <em>p</em>-value threshold.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return detailed and relaxed <code>data.table</code> where model and results are
merged for plotting purposes.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_graph">plot_graph</a></code>, <code><a href="#topic+plot_barplot">plot_barplot</a></code>,
<code><a href="#topic+plot_heatmap">plot_heatmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file(package="mulea", "extdata", 
    "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
tf_gmt_filtered &lt;- filter_ontology(gmt = tf_gmt, min_nr_of_elements = 3, 
    max_nr_of_elements = 400)

# loading the example data
sign_genes &lt;- readLines(system.file(
    package = "mulea", "extdata", "target_set.txt"))
background_genes &lt;- readLines(
    system.file(package="mulea", "extdata", "background_set.txt"))

# creating the ORA model
ora_model &lt;- ora(gmt = tf_gmt_filtered,
    # the test set variable
    element_names = sign_genes, 
    # the background set variable
    background_element_names = background_genes, 
    # the p-value adjustment method
    p_value_adjustment_method = "eFDR", 
    # the number of permutations
    number_of_permutations = 10000,
    # the number of processor threads to use
    nthreads = 2)
# running the ORA
ora_results &lt;- run_test(ora_model)

# reshaping results for visualisation
ora_reshaped_results &lt;- reshape_results(model = ora_model,
    model_results = ora_results,
    # choosing which column to use for the indication of significance
    p_value_type_colname = "eFDR")
</code></pre>

<hr>
<h2 id='run_test'>Run enrichment analysis procedure</h2><span id='topic+run_test'></span><span id='topic+run_test+2Cora-method'></span>

<h3>Description</h3>

<p>This is a generic function that chooses an enrichment analysis procedure
based on the model class and runs the analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_test(model)

## S4 method for signature 'ora'
run_test(model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_test_+3A_model">model</code></td>
<td>
<p>Object of S4 class representing the mulea test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function requires the definition of a model. Models currently
implemented in mulea include Gene Set Enrichment Analysis (GSEA) and
Over-Representation Analysis (ORA). These models must be defined through
their specific functions which are provided in this package.
</p>


<h3>Value</h3>

<p>Results in form of <code>data.frame</code>. Structure of <code>data.frame</code> depends on
object processed by this generic method.
In the case of <code>run_test</code> was used with the model generated
by the <code>ora</code> function the returned
<code>data.frame</code> contains the following columns:
</p>

<ol>
<li><p> 'ontology_id': Identifiers of the ontology elements.
</p>
</li>
<li><p> 'ontology_name': Names of the ontology elements.
</p>
</li>
<li><p> 'nr_common_with_tested_elements': Number of common elements between the
ontology element and the vector defined by the element_names parameter
of the <code>ora</code> function.
</p>
</li>
<li><p> 'nr_common_with_background_elements': Number of common elements between
the ontology element and the vector defined by the
background_element_names parameter of the <code>ora</code> function.
</p>
</li>
<li><p> 'p_value': The raw <em>p</em>-value of the overrepresentation analysis.
</p>
</li>
<li><p> The adjusted <em>p</em>-value.
The column named based on the
p_value_adjustment_method parameter of the
<code>ora</code> function, <em>e.g.</em> 'eFDR'
</p>
</li></ol>

<p>In the case of <code>run_test</code> was used with the model
generated by the <code>gsea</code> function the returned
<code>data.frame</code> contains the following columns:
</p>

<ol>
<li><p> 'ontology_id': Identifiers of the ontology elements.
</p>
</li>
<li><p> 'ontology_name': Names of the ontology elements.
</p>
</li>
<li><p> 'nr_common_with_tested_elements': Number of common elements between the
ontology element and the vector defined by the element_names parameter
of the <code>gsea</code> function.
</p>
</li>
<li><p> 'p_value': The raw <em>p</em>-value of the gene set enrichment analysis.
</p>
</li>
<li><p> 'adjusted_p_value': The adjusted <em>p</em>-value.
</p>
</li></ol>

<p>run_test method for ora object. Returns the results of the
overrepresentation analysis.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>run_test(ora)</code>: ora test.
</p>
</li></ul>


<h3>See Also</h3>

<p><code><a href="#topic+gsea">gsea</a></code>, <code><a href="#topic+ora">ora</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file( package="mulea", "extdata", 
    "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
tf_gmt_filtered &lt;- filter_ontology(gmt = tf_gmt, min_nr_of_elements = 3, 
    max_nr_of_elements = 400)

# loading the example data
sign_genes &lt;- readLines(system.file(package = "mulea", "extdata", 
    "target_set.txt"))
background_genes &lt;- readLines(system.file(package="mulea", "extdata", "
    background_set.txt"))

# creating the ORA model
ora_model &lt;- ora(gmt = tf_gmt_filtered, 
    # the test set variable
    element_names = sign_genes, 
    # the background set variable
    background_element_names = background_genes, 
    # the p-value adjustment method
    p_value_adjustment_method = "eFDR", 
    # the number of permutations
    number_of_permutations = 10000,
    # the number of processor threads to use
    nthreads = 2)
# running the ORA
ora_results &lt;- run_test(ora_model)

library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file(
        package="mulea", "extdata", 
        "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))
tf_gmt_filtered &lt;- filter_ontology(gmt = tf_gmt, min_nr_of_elements = 3, 
        max_nr_of_elements = 400)

# loading the example data
sign_genes &lt;- readLines(system.file(package = "mulea", "extdata", 
        "target_set.txt"))
background_genes &lt;- readLines(system.file(package="mulea", "extdata", 
        "background_set.txt"))

# creating the ORA model
ora_model &lt;- ora(gmt = tf_gmt_filtered, 
        # the test set variable
        element_names = sign_genes, 
        # the background set variable
        background_element_names = background_genes, 
        # the p-value adjustment method
        p_value_adjustment_method = "eFDR", 
        # the number of permutations
        number_of_permutations = 10000,
        # the number of processor threads to use
        nthreads = 2)
# running the ORA
ora_results &lt;- run_test(ora_model)
</code></pre>

<hr>
<h2 id='SetBasedEnrichmentTest-class'>PRIVATE class : An S4 class to represent a Hypergeometric tests in
mulea.</h2><span id='topic+SetBasedEnrichmentTest-class'></span><span id='topic+SetBasedEnrichmentTest'></span><span id='topic+run_test+2CSetBasedEnrichmentTest-method'></span>

<h3>Description</h3>

<p>PRIVATE class : An S4 class to represent a Hypergeometric tests in
mulea.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'SetBasedEnrichmentTest'
run_test(model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SetBasedEnrichmentTest-class_+3A_model">model</code></td>
<td>
<p>Object of s4 class represents mulea Test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SetBasedEnrichmentTest object. Used as private function.
</p>
<p>run_test method for SetBasedEnrichmentTest object. Used as private
function.
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>run_test(SetBasedEnrichmentTest)</code>: runs test calculations.
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>gmt</code></dt><dd><p>A data.frame representing GMT's representation of model.</p>
</dd>
<dt><code>element_names</code></dt><dd><p>A data from experiment to analyse across model.</p>
</dd>
<dt><code>pool</code></dt><dd><p>A background data to count test.</p>
</dd>
<dt><code>nthreads</code></dt><dd><p>Number of processor's threads used in calculations.</p>
</dd>
<dt><code>random_seed</code></dt><dd><p>Setup seed for random generator.</p>
</dd>
</dl>

<hr>
<h2 id='SubramanianTest-class'>PRIVATE class : An S4 class to represent a ranked based tests in mulea.</h2><span id='topic+SubramanianTest-class'></span><span id='topic+SubramanianTest'></span><span id='topic+run_test+2CSubramanianTest-method'></span>

<h3>Description</h3>

<p>PRIVATE class : An S4 class to represent a ranked based tests in mulea.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'SubramanianTest'
run_test(model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="SubramanianTest-class_+3A_model">model</code></td>
<td>
<p>Object of s4 class represents mulea Test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>data.frame</code> with presented columns 'ontology_id', 'ontology_name',
'nr_common_with_tested_elements',
'p_value', 'adjusted_p_value'
</p>
<p>run_test method for SubramanianTest object. Used as private
function.
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>run_test(SubramanianTest)</code>: runs test calculations.
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>gmt</code></dt><dd><p>A <code>data.frame</code> representing the ontology GMT.</p>
</dd>
<dt><code>element_names</code></dt><dd><p>A vector of elements names (gene or protein names or
identifiers) to include in the analysis.</p>
</dd>
<dt><code>element_scores</code></dt><dd><p>A vector of numeric values representing a
score (<em>e.g.</em> <em>p</em>-value, <em>z</em>-score, log fold change) for each
'element_name', in the same number and order as element_name.</p>
</dd>
<dt><code>p</code></dt><dd><p>A power of weight.</p>
</dd>
<dt><code>element_score_type</code></dt><dd><p>Defines the GSEA score type.
</p>

<ul>
<li><p> &quot;pos&quot;: Only positive element_scores
</p>
</li>
<li><p> &quot;neg&quot;: only negative element_scores - &quot;neg&quot; and mixed
</p>
</li>
<li><p> &quot;std&quot;: standard &ndash; containing both positive and negative scores
Default value is &quot;std&quot;.
</p>
</li></ul>
</dd>
</dl>

<hr>
<h2 id='write_gmt'>Write GMT file</h2><span id='topic+write_gmt'></span>

<h3>Description</h3>

<p>Writes gene set or ontology <code>data.frame</code> with specific
formatting (columns representing ontology identifiers, descriptions, and
associated lists of values) and writes it to a file in a standardized Gene
Matrix Transposed (GMT) file format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_gmt(gmt, file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_gmt_+3A_gmt">gmt</code></td>
<td>
<p>A <code>data.frame</code> containing the data to be written,
imported from a GMT file with the <code>read_gmt</code> function.</p>
</td></tr>
<tr><td><code id="write_gmt_+3A_file">file</code></td>
<td>
<p>Character, a path to a file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the input as a GMT file at a specific location.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(mulea)

# loading and filtering the example ontology from a GMT file
tf_gmt &lt;- read_gmt(file = system.file(
    package="mulea", "extdata", 
    "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt"))

# writing the filtered ontology to a GMT file

write_gmt(
    gmt = tf_gmt, 
    file = "Transcription_factor_RegulonDB_Escherichia_coli_GeneSymbol.gmt")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
