<!DOCTYPE html><html lang="en"><head><title>Help for package multimedia</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {multimedia}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+5B+2Cmediation_data+2CANY+2CANY+2CANY-method'><p>Subset a mediation dataset</p></a></li>
<li><a href='#ansi_aware_handler'><p>Pretty Printing</p></a></li>
<li><a href='#bind_mediation'><p>Convert <code>mediation_data</code> to a single data.frame</p></a></li>
<li><a href='#bootstrap'><p>Bootstrap Distribution for Estimators</p></a></li>
<li><a href='#brms_model'><p>Bayesian Regression Model across Responses</p></a></li>
<li><a href='#brms_sampler'><p>Sample from a Bayesian Regression Model</p></a></li>
<li><a href='#contrast_predictions'><p>Estimate the Difference between Profiles</p></a></li>
<li><a href='#contrast_samples'><p>Difference between Samples at Contrasting Profiles</p></a></li>
<li><a href='#demo_joy'><p>A Demo Dataset (Random)</p></a></li>
<li><a href='#demo_spline'><p>A Demo Dataset (Spline)</p></a></li>
<li><a href='#direct_effect'><p>Direct Effects from Estimated Model</p></a></li>
<li><a href='#edges'><p>Graphical Structure for Mediation Objects</p></a></li>
<li><a href='#edges+2Cmultimedia-method'><p>Access Mediation Model DAG</p></a></li>
<li><a href='#effect_summary'><p>Average Effects across j</p></a></li>
<li><a href='#estimate'><p>Estimate a Mediation Model</p></a></li>
<li><a href='#estimator'><p>Accessor for Model Estimators</p></a></li>
<li><a href='#estimator+2Cmodel-method'><p>Accessor for Estimators</p></a></li>
<li><a href='#exper_df'><p>Convert a Summarized Experiment to a data.frame</p></a></li>
<li><a href='#fdr_summary'><p>Calibration using Synthetic Nulls</p></a></li>
<li><a href='#glmnet_model'><p>Regularized 'Glmnet' Model across Responses</p></a></li>
<li><a href='#glmnet_sampler'><p>Sample from a 'Glmnet' Package Model</p></a></li>
<li><a href='#indirect_overall'><p>Overall Indirect Effect</p></a></li>
<li><a href='#indirect_pathwise'><p>Indirect Effects via Single Mediation Paths</p></a></li>
<li><a href='#lm_model'><p>Linear Model across Responses</p></a></li>
<li><a href='#lm_sampler'><p>Sample a Linear Model</p></a></li>
<li><a href='#lnm_model'><p>Logistic Normal Multinomial Model</p></a></li>
<li><a href='#lnm_sampler'><p>Sample from the Logistic Normal Multinomial</p></a></li>
<li><a href='#mediation_data'><p><code>mediation_data</code> Constructor</p></a></li>
<li><a href='#mediation_models'><p>Accessor for Outcome Models</p></a></li>
<li><a href='#mediators'><p>Access Class-Specific Mediators</p></a></li>
<li><a href='#mediators+2Cmediation_data-method'><p>Access to @mediators in Mediation Data</p></a></li>
<li><a href='#mediators+2Cmultimedia-method'><p>Names of Mediators in a Multimedia Object</p></a></li>
<li><a href='#mediators+26lt+3B-'><p>Set Mediators</p></a></li>
<li><a href='#mediators+26lt+3B-+2Cmediation_data-method'><p>Set the Mediators in a Mediation Data Object</p></a></li>
<li><a href='#mindfulness'><p>Mindfulness Dataset</p></a></li>
<li><a href='#model-class'><p>Representation of an Outcome or Mediation Model</p></a></li>
<li><a href='#multimedia'><p><code>multimedia</code> Constructor</p></a></li>
<li><a href='#n_mediators'><p>Number of Mediators in a Multimedia Object</p></a></li>
<li><a href='#n_outcomes'><p>Number of Outcomes in a Multimedia Object</p></a></li>
<li><a href='#nrow+2Cmediation_data-method'><p>How many samples in the mediation dataset?</p></a></li>
<li><a href='#null_contrast'><p>Compare Effects from Experimental vs. Null Mediation Data</p></a></li>
<li><a href='#nullify'><p>Nullify Active Edges</p></a></li>
<li><a href='#outcome_model'><p>Access the Outcome Model in a Multimedia Object</p></a></li>
<li><a href='#outcome_models'><p>Accessor for Outcome Models</p></a></li>
<li><a href='#outcomes'><p>Access Outcomes</p></a></li>
<li><a href='#outcomes+2Cmediation_data-method'><p>Outcomes Data in a Mediation Data Object</p></a></li>
<li><a href='#outcomes+2Cmultimedia-method'><p>Names of Outcomes in a Multimedia Object</p></a></li>
<li><a href='#outcomes+26lt+3B-'><p>Set Outcomes</p>
This is an setter method for outcomes in an S4 object, usually of class
mediation_data.</a></li>
<li><a href='#outcomes+26lt+3B-+2Cmediation_data-method'><p>Set the Outcomes in a Mediation Data Object</p></a></li>
<li><a href='#parallelize'><p>Parallelize Estimation across Responses</p></a></li>
<li><a href='#plot_mediators'><p>Visualize Indirect Effects</p></a></li>
<li><a href='#plot_sensitivity'><p>Generic Sensitivity Plot</p></a></li>
<li><a href='#predict_across'><p>Predict a Subset of Responses</p></a></li>
<li><a href='#predict+2Cmultimedia-method'><p>Predictions from a Multimedia Class</p></a></li>
<li><a href='#pretreatments'><p>Access Pretreatments</p></a></li>
<li><a href='#pretreatments+2Cmediation_data-method'><p>Pretreatments in a Mediation Data Object</p></a></li>
<li><a href='#pretreatments+26lt+3B-'><p>Set Pretreatments</p>
This is an setter method for pretreatments in an S4 object, usually of class
mediation_data.</a></li>
<li><a href='#pretreatments+26lt+3B-+2Cmediation_data-method'><p>Set the Pretreatments in a Mediation Data Object</p></a></li>
<li><a href='#retrieve_names'><p>Variables in a Multimedia Object</p></a></li>
<li><a href='#rf_model'><p>Random Forest Model</p></a></li>
<li><a href='#rf_sampler'><p>Sample from a Random Forest Model</p></a></li>
<li><a href='#sample+2Cmultimedia-method'><p>Sample New Mediator/Outcome Data</p></a></li>
<li><a href='#sensitivity'><p>Sensitivity Analysis for Overall Indirect Effect</p></a></li>
<li><a href='#sensitivity_pathwise'><p>Sensitivity Analysis for Pathwise Indirect Effects</p></a></li>
<li><a href='#sensitivity_perturb'><p>Sensitivity to User-Specified Perturbations</p></a></li>
<li><a href='#setup_profile'><p>Define a <code>treatment_profile</code> object</p></a></li>
<li><a href='#spline_fun'><p>Generate Random Spline</p></a></li>
<li><a href='#sub_formula'><p>Helper to Modify Formulas</p></a></li>
<li><a href='#treatment_profile-class'><p>Define a Treatment Profile</p></a></li>
<li><a href='#treatments'><p>Access Treatments</p></a></li>
<li><a href='#treatments+2Cmediation_data-method'><p>Treatments in a Mediation Data Object</p></a></li>
<li><a href='#treatments+2Cmultimedia-method'><p>Names of Treatments in a Multimedia Object</p></a></li>
<li><a href='#treatments+26lt+3B-'><p>Set Treatments</p>
This is an setter method for treatments in an S4 object, usually of class
mediation_data.</a></li>
<li><a href='#treatments+26lt+3B-+2Cmediation_data-method'><p>Set the Treatments in a Mediation Data Object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Multimodal Mediation Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Multimodal mediation analysis is an emerging problem in microbiome data analysis. Multimedia make advanced mediation analysis techniques easy to use, ensuring that all statistical components are transparent and adaptable to specific problem contexts. The package provides a uniform interface to direct and indirect effect estimation, synthetic null hypothesis testing, bootstrap confidence interval construction, and sensitivity analysis. More details are available in Jiang et al. (2024) "multimedia: Multimodal Mediation Analysis of Microbiome Data" &lt;<a href="https://doi.org/10.1101%2F2024.03.27.587024">doi:10.1101/2024.03.27.587024</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/publicdomain/zero/1.0/legalcode">CC0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Coverage, Microbiome, Regression, Sequencing, Software,
StatisticalMethod, StructuralEquationModels</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0), brms, glmnetUtils, ranger, tidyselect</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, S4Vectors, SummarizedExperiment, cli, dplyr, fansi,
formula.tools, ggplot2, glue, methods, miniLNM, patchwork,
phyloseq, progress, purrr, rlang, stats, tidygraph, tidyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>compositions, ggdist, ggraph, ggrepel, knitr, rmarkdown,
testthat (&ge; 3.0.0), tidyverse, vroom</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/krisrs1128/multimedia/issues/">https://github.com/krisrs1128/multimedia/issues/</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://krisrs1128.github.io/multimedia/">https://krisrs1128.github.io/multimedia/</a>,
<a href="https://github.com/krisrs1128/multimedia/">https://github.com/krisrs1128/multimedia/</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-17 19:41:43 UTC; krissankaran</td>
</tr>
<tr>
<td>Author:</td>
<td>Kris Sankaran <a href="https://orcid.org/0000-0002-9415-1971"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Hanying Jiang [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kris Sankaran &lt;ksankaran@wisc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-18 12:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='+5B+2Cmediation_data+2CANY+2CANY+2CANY-method'>Subset a mediation dataset</h2><span id='topic++5B+2Cmediation_data+2CANY+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>We can subset samples by indexing into a mediation dataset. It will subsample
all fields &ndash; pretreatments, treatments, mediators, and outcomes. Note that
there is no way to subset columns in this way, since they would be different
across each source.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mediation_data,ANY,ANY,ANY'
x[i, j, ..., drop = TRUE]
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B5B+2B2Cmediation_data+2B2CANY+2B2CANY+2B2CANY-method_+3A_x">x</code></td>
<td>
<p>An object of class <code>mediation_data</code> whose samples we want to subset.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cmediation_data+2B2CANY+2B2CANY+2B2CANY-method_+3A_i">i</code></td>
<td>
<p>An integer or integer/logical vector specifying the samples to
subset to.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cmediation_data+2B2CANY+2B2CANY+2B2CANY-method_+3A_j">j</code></td>
<td>
<p>A placeholder to argree with R's <code>[</code> function. Never used.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cmediation_data+2B2CANY+2B2CANY+2B2CANY-method_+3A_...">...</code></td>
<td>
<p>A placeholder to agree with R's <code>[</code> function. Never used.</p>
</td></tr>
<tr><td><code id="+2B5B+2B2Cmediation_data+2B2CANY+2B2CANY+2B2CANY-method_+3A_drop">drop</code></td>
<td>
<p>A placeholder to agree with R's <code>[</code> function. Never used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of the input mediation_data object whose @mediators,
@outcomes, @treatments, and @pretreatments rows have all been subsetted
according to <code>i</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
exper[1]
exper[1:10]
</code></pre>

<hr>
<h2 id='ansi_aware_handler'>Pretty Printing</h2><span id='topic+ansi_aware_handler'></span>

<h3>Description</h3>

<p>Helper function for printing ANSI in Rmarkdown output. Use this at the start
of your Rmarkdown files to include colors in the printed object names in the
final compiled output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ansi_aware_handler(x, options)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ansi_aware_handler_+3A_x">x</code></td>
<td>
<p>A character vector potentially including ANSI.</p>
</td></tr>
<tr><td><code id="ansi_aware_handler_+3A_options">options</code></td>
<td>
<p>Unused placeholder argument.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Taken from the post at
</p>
<p>https://blog.djnavarro.net/posts/2021-04-18_pretty-little-clis/
</p>


<h3>Value</h3>

<p>A string with HTML reformatted to ensure colors appear in printed
code blocks in rmarkdown output.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>knitr::knit_hooks$set(output = ansi_aware_handler)
options(crayon.enabled = TRUE)
</code></pre>

<hr>
<h2 id='bind_mediation'>Convert <code>mediation_data</code> to a single data.frame</h2><span id='topic+bind_mediation'></span>

<h3>Description</h3>

<p>It can be helpful to combine all the data in a <code>mediation_data</code> S4 object
into a single data.frame. This concatenates all data sources horizontally,
into an N samples x (all outcomes, mediators, treatments, ...) matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bind_mediation(exper)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bind_mediation_+3A_exper">exper</code></td>
<td>
<p>An object of class <code>mediation_data</code> containing the variables
that we want horizontally concatenated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing all variables from the original
<code>mediation_data</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
exper
bind_mediation(exper)

exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
exper
bind_mediation(exper)
</code></pre>

<hr>
<h2 id='bootstrap'>Bootstrap Distribution for Estimators</h2><span id='topic+bootstrap'></span>

<h3>Description</h3>

<p>Given a mediation model specification, estimators fs, and original dataset
exper, this will re-estimate the mediation model on resampled versions of
exper and apply each estimator in fs to construct bootstrap distributions
associated wtih those estimators.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootstrap(model, exper, fs = NULL, B = 1000, progress = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bootstrap_+3A_model">model</code></td>
<td>
<p>An object of class multimedia with specified mediation and
outcome models that we want to re-estimate across B bootstrap samples.</p>
</td></tr>
<tr><td><code id="bootstrap_+3A_exper">exper</code></td>
<td>
<p>An object of class multimedia_data containing the mediation and
outcome data from which the direct effects are to be estimated.</p>
</td></tr>
<tr><td><code id="bootstrap_+3A_fs">fs</code></td>
<td>
<p>The estimators whose bootstrap samples we are interested in. These
are assumed to be a vector of functions (for example, direct_effect or
indirect_effect), and they will each be applied to each bootstrap resample.</p>
</td></tr>
<tr><td><code id="bootstrap_+3A_b">B</code></td>
<td>
<p>The number of bootstrap samples. Defaults to 1000.</p>
</td></tr>
<tr><td><code id="bootstrap_+3A_progress">progress</code></td>
<td>
<p>A logical indicating whether to show a progress bar.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>stats A list of length B containing the results of the fs applied on
each of the B bootstrap resamples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example with null data. We set B to 5 just to execute quickly -- it's not
# actually a practical choice of B
exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    bootstrap(exper, B = 5)

# example with another dataset
exper &lt;- demo_spline(n_samples = 100, tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
samples &lt;- multimedia(exper, rf_model(num.trees = 1e3)) |&gt;
    bootstrap(exper, B = 5)
ggplot2::ggplot(samples$direct_effect) +
    ggplot2::geom_histogram(
        ggplot2::aes(direct_effect, fill = indirect_setting),
        bins = 15
    ) +
    ggplot2::facet_wrap(~outcome, scales = "free")
</code></pre>

<hr>
<h2 id='brms_model'>Bayesian Regression Model across Responses</h2><span id='topic+brms_model'></span>

<h3>Description</h3>

<p>Apply a Bayesian regression model in parallel across each response $y$ in an
outcome or mediation model. This can be helpful when we want to share
information across related
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brms_model(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="brms_model_+3A_...">...</code></td>
<td>
<p>Keyword parameters passed to brm.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>model An object of class <code>model</code> with estimator, predictor, and
sampler functions associated wtih a Bayesian regression model.
</p>


<h3>See Also</h3>

<p>glmnet_model lnm_model rf_model lm_model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper, brms_model()) # call estimate() on this to fit
</code></pre>

<hr>
<h2 id='brms_sampler'>Sample from a Bayesian Regression Model</h2><span id='topic+brms_sampler'></span>

<h3>Description</h3>

<p>This samples from the posterior predictive for each component in
a multiresponse Bayesian Regression model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brms_sampler(fits, newdata = NULL, indices = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="brms_sampler_+3A_fits">fits</code></td>
<td>
<p>The fitted 'BRMS' model model from which to draw samples.</p>
</td></tr>
<tr><td><code id="brms_sampler_+3A_newdata">newdata</code></td>
<td>
<p>A data.frame containing new inputs from which to sample
responses. If NULL, defaults to the data used to estimate fit.</p>
</td></tr>
<tr><td><code id="brms_sampler_+3A_indices">indices</code></td>
<td>
<p>The coordinates of the response from which we want to sample.</p>
</td></tr>
<tr><td><code id="brms_sampler_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass to <code>posterior_predict</code> in the 'brms'
package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing a single posterior predictive sample at each
of the newdata rows passed into a fitted BRMS model. Each column
corresponds to one outcome variable, each row to the associated row in the
newdata input..
</p>

<hr>
<h2 id='contrast_predictions'>Estimate the Difference between Profiles</h2><span id='topic+contrast_predictions'></span>

<h3>Description</h3>

<p>Given a fitted multimedia model, contrast the mediation and outcome
predictions associated wtih two treatment profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contrast_predictions(model, profile1, profile2, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="contrast_predictions_+3A_model">model</code></td>
<td>
<p>An object of class multimedia containing the estimated mediation
and outcome models whose mediation and outcome predictions we want to
compare.</p>
</td></tr>
<tr><td><code id="contrast_predictions_+3A_profile1">profile1</code></td>
<td>
<p>An object of class <code>treatment_profile</code> containing the first
treatment profile to consider in the difference.</p>
</td></tr>
<tr><td><code id="contrast_predictions_+3A_profile2">profile2</code></td>
<td>
<p>An object of class <code>treatment_profile</code> containing the second
treatment profile to consider in the difference.</p>
</td></tr>
<tr><td><code id="contrast_predictions_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass to <code>predict()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two elements, <code>mediators</code> and <code>outcomes</code>, containing the
differences in the predicted M(T') - M(T) and Y(T', M(T')) - Y(T, M(T))
between the two profiles T and T'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
model &lt;- multimedia(exper) |&gt;
    estimate(exper)
t1 &lt;- data.frame(treatment = factor("Treatment"))
t2 &lt;- data.frame(treatment = factor("Control"))
profile1 &lt;- setup_profile(model, t1, t1)
profile2 &lt;- setup_profile(model, t2, t2)
contrast_predictions(model, profile1, profile2)
</code></pre>

<hr>
<h2 id='contrast_samples'>Difference between Samples at Contrasting Profiles</h2><span id='topic+contrast_samples'></span>

<h3>Description</h3>

<p>Given a fitted multimedia model, contrast sampled mediation and outcome data
associated with two treatment profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contrast_samples(model, profile1, profile2, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="contrast_samples_+3A_model">model</code></td>
<td>
<p>An object of class multimedia containing the estimated mediation
and outcome models whose mediation and outcome predictions we want to
compare.</p>
</td></tr>
<tr><td><code id="contrast_samples_+3A_profile1">profile1</code></td>
<td>
<p>An object of class <code>treatment_profile</code> containing the first
treatment profile to consider in the difference.</p>
</td></tr>
<tr><td><code id="contrast_samples_+3A_profile2">profile2</code></td>
<td>
<p>An object of class <code>treatment_profile</code> containing the second
treatment profile to consider in the difference.</p>
</td></tr>
<tr><td><code id="contrast_samples_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass to <code>sample()</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two elements, <code>mediators</code> and <code>outcomes</code>, containing the
differences in the sampled M(T') - M(T) and Y(T', M(T')) - Y(T, M(T))
between the two profiles T and T'.
</p>
<p>A list with two elements, mediators and outcomes. These contrast the
values of the mediator and outcomes under the two profiles T and T'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
model &lt;- multimedia(exper) |&gt;
    estimate(exper)
t1 &lt;- data.frame(treatment = factor("Treatment"))
t2 &lt;- data.frame(treatment = factor("Control"))
profile1 &lt;- setup_profile(model, t1, t1)
profile2 &lt;- setup_profile(model, t2, t2)
contrast_samples(model, profile1, profile2)

samples &lt;- purrr::map(
    seq_len(100),
    ~ contrast_samples(model, profile1, profile2)
)
hist(sapply(samples, \(x) x[[1]]$ASV1))
hist(sapply(samples, \(x) x[[1]]$ASV2))
</code></pre>

<hr>
<h2 id='demo_joy'>A Demo Dataset (Random)</h2><span id='topic+demo_joy'></span>

<h3>Description</h3>

<p>This is a simple dataset with random data, used simply to illustrate the
design of multimedia. There is no real association between any treatments,
mediators, or outcomes. It always returns a single outcome (PHQ)
SummarizedExperiment, and it randomly assigns samples to Treatment and
Control (see the colData). It stores all the hypothetical mediator data in
the assay slot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_joy(n_samples = 100, n_mediators = 5, n_pretreatment = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="demo_joy_+3A_n_samples">n_samples</code></td>
<td>
<p>How many random samples to generate?</p>
</td></tr>
<tr><td><code id="demo_joy_+3A_n_mediators">n_mediators</code></td>
<td>
<p>How many random mediators to generate?</p>
</td></tr>
<tr><td><code id="demo_joy_+3A_n_pretreatment">n_pretreatment</code></td>
<td>
<p>How many random pretreatment variables?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SE The summarized experiment containing random data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>demo_joy()

demo_joy(n_samples = 2, n_mediators = 20)
</code></pre>

<hr>
<h2 id='demo_spline'>A Demo Dataset (Spline)</h2><span id='topic+demo_spline'></span>

<h3>Description</h3>

<p>This is a simple dataset with nonlinear relationships between the outcome and
mediators. It is used simply to illustrate the design of multimedia. The
mediator-&gt;outcome effect is generated from a random spline function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_spline(n_samples = 5000, tau = c(2, 2))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="demo_spline_+3A_n_samples">n_samples</code></td>
<td>
<p>The number of samples to generate in the toy example</p>
</td></tr>
<tr><td><code id="demo_spline_+3A_tau">tau</code></td>
<td>
<p>The true direct effects associated with the two outcomes. Defaults
to 2, 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>xy A data.frame whose columns include the treatment, mediation, and
outcome variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>demo_spline()
</code></pre>

<hr>
<h2 id='direct_effect'>Direct Effects from Estimated Model</h2><span id='topic+direct_effect'></span>

<h3>Description</h3>

<p>Estimate direct effects associated with a multimedia model. These estimates
are formed using Equation (10) of our paper. Rather than providing this
average, this function returns the estimated difference for each $j$. To
average across all j, this result can be passed to the ' <code>effect_summary</code>
function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>direct_effect(model, exper = NULL, t1 = 1, t2 = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="direct_effect_+3A_model">model</code></td>
<td>
<p>An object of class multimedia containing the estimated mediation
and outcome models whose mediation and outcome predictions we want to
compare.</p>
</td></tr>
<tr><td><code id="direct_effect_+3A_exper">exper</code></td>
<td>
<p>An object of class multimedia_data containing the mediation and
outcome data from which the direct effects are to be estimated.</p>
</td></tr>
<tr><td><code id="direct_effect_+3A_t1">t1</code></td>
<td>
<p>The reference level of the treatment to be used when computing the
direct effect.</p>
</td></tr>
<tr><td><code id="direct_effect_+3A_t2">t2</code></td>
<td>
<p>The alternative level of the treatment to be used when computing
the direct effect.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame summarizing the direct effects associated with different
settings of j in the equation above.
</p>


<h3>See Also</h3>

<p>effect_summary
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example with null data
exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)

direct_effect(fit)
direct_effect(fit, t1 = 2, t2 = 1)
direct_effect(fit, t1 = 2, t2 = 2)

# example with another dataset
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)
direct_effect(fit)
</code></pre>

<hr>
<h2 id='edges'>Graphical Structure for Mediation Objects</h2><span id='topic+edges'></span>

<h3>Description</h3>

<p>We often want to access the DAG for different mediation-related S4 objects.
This generic helps us access these graphical model edges lists. See method
instantiations for specific examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>edges(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="edges_+3A_object">object</code></td>
<td>
<p>An object whose DAG structure we want to access.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the S4 object that is passed. For multimedia
objects, this will return an edgelist as a two column data.frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    edges()
</code></pre>

<hr>
<h2 id='edges+2Cmultimedia-method'>Access Mediation Model DAG</h2><span id='topic+edges+2Cmultimedia-method'></span>

<h3>Description</h3>

<p>This is an accessor to the edges slot in a multimedia object. It is the
internal representation of the variable conditional dependence graph encoded
by the mediation model's DAG.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multimedia'
edges(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="edges+2B2Cmultimedia-method_+3A_object">object</code></td>
<td>
<p>An object of class multimedia.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame whose rows give edges in the mediation analysis DAG.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    edges()
</code></pre>

<hr>
<h2 id='effect_summary'>Average Effects across j</h2><span id='topic+effect_summary'></span>

<h3>Description</h3>

<p>This averages direct or indirect effects across settings j, leading to
the effect estimates given in equation (10) of the preprint.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effect_summary(effects)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="effect_summary_+3A_effects">effects</code></td>
<td>
<p>The output from direct_effect or indirect_effect. A data.frame
containing effect estimates for each variable and indirect/direct setting
along rows.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of the input with all indirect/direct settings averaged
over.
</p>


<h3>See Also</h3>

<p>direct_effect indirect_effect
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example with null data
exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    estimate(exper) |&gt;
    direct_effect() |&gt;
    effect_summary()

# example with another dataset
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
multimedia(exper) |&gt;
    estimate(exper) |&gt;
    direct_effect() |&gt;
    effect_summary()
</code></pre>

<hr>
<h2 id='estimate'>Estimate a Mediation Model</h2><span id='topic+estimate'></span>

<h3>Description</h3>

<p><code>estimate</code> provides a unified interface to estimate all the models that can
be encapsulated within a <code>multimedia</code> class. It simply calls the
multimedia object. The resulting estimates can be used for downstream direct
effect estimation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate(model, exper)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimate_+3A_model">model</code></td>
<td>
<p>An object of class multimedia containing the estimated mediation
and outcome models whose mediation and outcome predictions we want to
compare.</p>
</td></tr>
<tr><td><code id="estimate_+3A_exper">exper</code></td>
<td>
<p>An object of class multimedia_data containing the mediation and
outcome data from which the direct effects are to be estimated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of the input modified in place so that the @estimates slot
has been filled.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    estimate(exper)

# example with another dataset
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
multimedia(exper) |&gt;
    estimate(exper)

# example with another model
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
multimedia(exper, glmnet_model()) |&gt;
    estimate(exper)
</code></pre>

<hr>
<h2 id='estimator'>Accessor for Model Estimators</h2><span id='topic+estimator'></span>

<h3>Description</h3>

<p>This defines a generic estimator function, which can be applied to different
multimedia model objects. It creates a unified interface to estimating
diverse mediation and outcome model families.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimator(object)

estimator(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimator_+3A_object">object</code></td>
<td>
<p>A model object that we want to estimate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A function that can be called with formula and data arguments, like
</p>
<p>A fitted version of the input model class.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- lm_model()
estimator(m)(mpg ~ hp + wt, data = mtcars)
m &lt;- rf_model()
fit &lt;- estimator(m)(mpg ~ hp + wt, data = mtcars)
</code></pre>

<hr>
<h2 id='estimator+2Cmodel-method'>Accessor for Estimators</h2><span id='topic+estimator+2Cmodel-method'></span>

<h3>Description</h3>

<p>Accessor for Estimators
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'model'
estimator(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimator+2B2Cmodel-method_+3A_object">object</code></td>
<td>
<p>An object of class <code>model</code>, whose estimator function we want
access to.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A fitted version of the input model class.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- rf_model()
fit &lt;- estimator(m)(mpg ~ hp + wt, data = mtcars)
</code></pre>

<hr>
<h2 id='exper_df'>Convert a Summarized Experiment to a data.frame</h2><span id='topic+exper_df'></span>

<h3>Description</h3>

<p>This is a small helepr function to concatenate the colData and assays within
SummarizedExperiment objects into a single data.frame. This is not necessary
for the essential multimedia workflow, it is only exported for potential
independent interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exper_df(exper)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="exper_df_+3A_exper">exper</code></td>
<td>
<p>An object of class SummarizedExperiment.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame combining all slots of a multimedia object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>demo_joy() |&gt;
    exper_df()
</code></pre>

<hr>
<h2 id='fdr_summary'>Calibration using Synthetic Nulls</h2><span id='topic+fdr_summary'></span>

<h3>Description</h3>

<p>This function computes a threshold for indirect or direct effect estimates
that controls the false discovery rate according to estimates made using real
and synthetic null data, against the null hypotheses that effects are zero.
It computes the proportion of synthetic null estimates that are among the top
K largest effects (in magnitude) as an estimate of the FDR.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fdr_summary(contrast, effect = "indirect_overall", q_value = 0.15)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fdr_summary_+3A_contrast">contrast</code></td>
<td>
<p>A data.frame summarizing the differences between outcomes
across hypothetical treatments, typically as output by <code>null_contrast</code>.
Each row is one outcome in one hypothetical scenario.</p>
</td></tr>
<tr><td><code id="fdr_summary_+3A_effect">effect</code></td>
<td>
<p>Either &quot;indirect_overall&quot; (the default), &quot;indirect_pathwise&quot;,
or &quot;direct_effect&quot; specifying the type of effect that we want to select.</p>
</td></tr>
<tr><td><code id="fdr_summary_+3A_q_value">q_value</code></td>
<td>
<p>The target for false discovery rate control. The last time the
estimated FDR is above this threshold is smallest magnitude of effect size
that we will consider.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>fdr A data.frame specifying, for each candidate effect, whether it
should be selected.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example with null data - notice synthetic data has larger effect.
exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    estimate(exper) |&gt;
    null_contrast(exper) |&gt;
    fdr_summary("direct_effect")

multimedia(exper) |&gt;
    estimate(exper) |&gt;
    null_contrast(exper, "M-&gt;Y", indirect_overall) |&gt;
    fdr_summary("indirect_overall")

# example with another dataset - synthetic effect is smaller.
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
multimedia(exper) |&gt;
    estimate(exper) |&gt;
    null_contrast(exper) |&gt;
    fdr_summary("direct_effect")

multimedia(exper) |&gt;
    estimate(exper) |&gt;
    null_contrast(exper, "M-&gt;Y", indirect_overall) |&gt;
    fdr_summary("indirect_overall")
</code></pre>

<hr>
<h2 id='glmnet_model'>Regularized 'Glmnet' Model across Responses</h2><span id='topic+glmnet_model'></span>

<h3>Description</h3>

<p>Apply a regularized (generalized) linear model in parallel across each
response $y$ in an outcome or mediation model. This can be helpful when we
have many mediators or pretreatment variables, making the input
high-dimensional.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmnet_model(progress = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="glmnet_model_+3A_progress">progress</code></td>
<td>
<p>A logical indicating whether to show a progress bar during
estimation.</p>
</td></tr>
<tr><td><code id="glmnet_model_+3A_...">...</code></td>
<td>
<p>Keyword parameters passed to package 'glmnet'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>model An object of class <code>model</code> with estimator, predictor, and
sampler functions associated wtih a lienar model.
</p>


<h3>See Also</h3>

<p>model lm_model rf_model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper, glmnet_model(lambda = 1)) |&gt;
    estimate(exper)

multimedia(exper, glmnet_model(lambda = 1), glmnet_model()) |&gt;
    estimate(exper)

# example with another dataset
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
multimedia(exper, glmnet_model(lambda = 0.1)) |&gt;
    estimate(exper)
</code></pre>

<hr>
<h2 id='glmnet_sampler'>Sample from a 'Glmnet' Package Model</h2><span id='topic+glmnet_sampler'></span>

<h3>Description</h3>

<p>This assumes a continuous response, so that the out-of-sample MSE can be used
to estimate the outcome variability sigma.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmnet_sampler(fits, newdata = NULL, indices = NULL, lambda_ix = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="glmnet_sampler_+3A_fits">fits</code></td>
<td>
<p>The fitted 'glmnet' package model model from which to draw
samples.</p>
</td></tr>
<tr><td><code id="glmnet_sampler_+3A_newdata">newdata</code></td>
<td>
<p>A data.frame containing new inputs from which to sample
responses. If NULL, defaults to the data used to estimate fit.</p>
</td></tr>
<tr><td><code id="glmnet_sampler_+3A_indices">indices</code></td>
<td>
<p>The coordinates of the response from which to draw samples.</p>
</td></tr>
<tr><td><code id="glmnet_sampler_+3A_lambda_ix">lambda_ix</code></td>
<td>
<p>A regularization strength parameter used to maintain
consistency with estimation. Not used during sampling.</p>
</td></tr>
<tr><td><code id="glmnet_sampler_+3A_...">...</code></td>
<td>
<p>Additional parameters to pass to predict.glmnet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>y_star A data.frame of samples y associated with the new inputs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- glmnet_model()
fit &lt;- estimator(m)(mpg ~ hp + wt, data = mtcars)
glmnet_sampler(fit, mtcars)

plm &lt;- parallelize(glmnetUtils::glmnet)
fit &lt;- plm(mpg + disp ~ hp + wt, data = mtcars)
glmnet_sampler(fit, mtcars)
</code></pre>

<hr>
<h2 id='indirect_overall'>Overall Indirect Effect</h2><span id='topic+indirect_overall'></span>

<h3>Description</h3>

<p>Direct Effects from Estimated Model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>indirect_overall(model, exper = NULL, t1 = 1, t2 = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="indirect_overall_+3A_model">model</code></td>
<td>
<p>An object of class multimedia containing the estimated mediation
and outcome models whose mediation and outcome predictions we want to
compare.</p>
</td></tr>
<tr><td><code id="indirect_overall_+3A_exper">exper</code></td>
<td>
<p>An object of class multimedia_data containing the mediation and
outcome data from which the direct effects are to be estimated.</p>
</td></tr>
<tr><td><code id="indirect_overall_+3A_t1">t1</code></td>
<td>
<p>The reference level of the treatment to be used when computing the
indirect effect.</p>
</td></tr>
<tr><td><code id="indirect_overall_+3A_t2">t2</code></td>
<td>
<p>The alternative level of the treatment to be used when computing
the indirect effect.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Estimate direct effects associated with a multimedia model. These estimates
are formed using Equation (10) of our preprint.
</p>


<h3>Value</h3>

<p>A data.frame summarizing the overall indirect effects associated with
different settings of j in the equation above.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example with null data
exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)

indirect_overall(fit)

# example with another dataset
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)
indirect_overall(fit)
</code></pre>

<hr>
<h2 id='indirect_pathwise'>Indirect Effects via Single Mediation Paths</h2><span id='topic+indirect_pathwise'></span>

<h3>Description</h3>

<p>Indirect Effects via Single Mediation Paths
</p>


<h3>Usage</h3>

<pre><code class='language-R'>indirect_pathwise(model, exper = NULL, t1 = 1, t2 = 2, progress = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="indirect_pathwise_+3A_model">model</code></td>
<td>
<p>An object of class multimedia containing the estimated mediation
and outcome models whose mediation and outcome predictions we want to
compare.</p>
</td></tr>
<tr><td><code id="indirect_pathwise_+3A_exper">exper</code></td>
<td>
<p>An object of class multimedia_data containing the mediation and
outcome data from which the direct effects are to be estimated.</p>
</td></tr>
<tr><td><code id="indirect_pathwise_+3A_t1">t1</code></td>
<td>
<p>The reference level of the treatment to be used when computing the
(pathwise) indirect effect.</p>
</td></tr>
<tr><td><code id="indirect_pathwise_+3A_t2">t2</code></td>
<td>
<p>The alternative level of the treatment to be used when computing
the (pathwise) indirect effect.</p>
</td></tr>
<tr><td><code id="indirect_pathwise_+3A_progress">progress</code></td>
<td>
<p>A logical indicating whether to show a progress bar.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame summarizing the pathwise (per-mediator) indirect effects
associated with different settings of the direct effect.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example with null data
exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)
indirect_pathwise(fit)

# example with another dataset
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)
indirect_pathwise(fit)
</code></pre>

<hr>
<h2 id='lm_model'>Linear Model across Responses</h2><span id='topic+lm_model'></span>

<h3>Description</h3>

<p>Apply a linear model in parallel across each response $y$ in an outcome or
mediation model. This is often useful for mediator models with few
pretreatment variables, since each input is low-dimensional, even when there
are many responses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lm_model(progress = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lm_model_+3A_progress">progress</code></td>
<td>
<p>A logical indicating whether to show a progress bar during
estimation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>model An object of class <code>model</code> with estimator, predictor, and
sampler functions associated wtih a linear model.
</p>


<h3>See Also</h3>

<p>model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- lm_model()
estimator(m)(mpg ~ hp + wt, data = mtcars)
</code></pre>

<hr>
<h2 id='lm_sampler'>Sample a Linear Model</h2><span id='topic+lm_sampler'></span>

<h3>Description</h3>

<p>Draw samples from a fitted linear model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lm_sampler(fits, newdata = NULL, indices = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lm_sampler_+3A_fits">fits</code></td>
<td>
<p>The fitted linear model model from which to draw samples.</p>
</td></tr>
<tr><td><code id="lm_sampler_+3A_newdata">newdata</code></td>
<td>
<p>A data.frame containing new inputs from which to sample
responses. If NULL, defaults to the data used to estimate fit.</p>
</td></tr>
<tr><td><code id="lm_sampler_+3A_indices">indices</code></td>
<td>
<p>The coordinates of the response from which to draw samples.</p>
</td></tr>
<tr><td><code id="lm_sampler_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to lm.predict</p>
</td></tr>
</table>


<h3>Value</h3>

<p>y_star A data.frame of samples y associated with the new inputs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fit &lt;- lm(mpg ~ hp + wt, data = mtcars)
lm_sampler(list(f = fit))

plm &lt;- parallelize(lm)
fit &lt;- plm(mpg + disp ~ hp + wt, data = mtcars)
lm_sampler(fit)
</code></pre>

<hr>
<h2 id='lnm_model'>Logistic Normal Multinomial Model</h2><span id='topic+lnm_model'></span>

<h3>Description</h3>

<p>Apply a logistic normal multinomial model to jointly model a vector of count
responses $y$ in an outcome or mediation model. This is a common choice for
data where the parameter of interest is the composition across responses
(e.g., microbiome).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lnm_model(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lnm_model_+3A_...">...</code></td>
<td>
<p>Keyword parameters passed to lnm in the 'miniLNM' package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>model An object of class <code>model</code> with estimator, predictor, and
sampler functions associated wtih a linear model.
</p>


<h3>See Also</h3>

<p>model lm_model rf_model glmnet_model brms_model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- lnm_model()
mat &lt;- data.frame(matrix(rpois(250, 10), 25, 10))
colnames(mat) &lt;- paste0("y", seq_len(6))
fit &lt;- estimator(m)(y1 + y2 + y3 + y4 ~ y5 + y6, mat)
</code></pre>

<hr>
<h2 id='lnm_sampler'>Sample from the Logistic Normal Multinomial</h2><span id='topic+lnm_sampler'></span>

<h3>Description</h3>

<p>This samples from the posterior predictive of a fitted logistic-normal
multinomial model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lnm_sampler(fit, newdata = NULL, indices = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lnm_sampler_+3A_fit">fit</code></td>
<td>
<p>The fitted LNM model from which to draw posterior predictive
samples.</p>
</td></tr>
<tr><td><code id="lnm_sampler_+3A_newdata">newdata</code></td>
<td>
<p>A data.frame containing new inputs from which to sample
responses. If NULL, defaults to the data used to estimate fit.</p>
</td></tr>
<tr><td><code id="lnm_sampler_+3A_indices">indices</code></td>
<td>
<p>The coordinates of the response from which to draw samples.</p>
</td></tr>
<tr><td><code id="lnm_sampler_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to sample.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>y_star A data.frame of samples y associated wtih the new inputs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- lnm_model()
mat &lt;- data.frame(matrix(rpois(250, 10), 25, 10))
colnames(mat) &lt;- paste0("y", 1:6)
fit &lt;- estimator(m)(y1 + y2 + y3 + y4 ~ y5 + y6, mat)
lnm_sampler(fit, depth = 10)
lnm_sampler(fit, depth = 100)
</code></pre>

<hr>
<h2 id='mediation_data'><code>mediation_data</code> Constructor</h2><span id='topic+mediation_data'></span>

<h3>Description</h3>

<p>Convert a SummarizedExperiment, phyloseq object, or data.frame into a
<code>mediation_data</code> object. This conversion helps to organize the variables that
lie on different parts of the mediation analysis graph, so that they are not
all kept in a homogeneous experiment or data.frame. It's possible to specify
outcome, mediator, or treatment variables using either string vectors or
tidyselect syntax, e.g., <code>starts_with("mediator_")</code> will match all columns of
the input data starting with the string mediator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediation_data(x, outcomes, treatments, mediators, pretreatments = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mediation_data_+3A_x">x</code></td>
<td>
<p>A SummarizedExperiment, phyloseq object, or data.frame whose columns
contain all the variables needed for the mediation analysis.</p>
</td></tr>
<tr><td><code id="mediation_data_+3A_outcomes">outcomes</code></td>
<td>
<p>A vector or tidyselect specification of the names of all
outcome variables.</p>
</td></tr>
<tr><td><code id="mediation_data_+3A_treatments">treatments</code></td>
<td>
<p>A vector or tidyselect specification of the names of all
treatment variables.</p>
</td></tr>
<tr><td><code id="mediation_data_+3A_mediators">mediators</code></td>
<td>
<p>A vector or tidyselect specification of the names of all
mediators.</p>
</td></tr>
<tr><td><code id="mediation_data_+3A_pretreatments">pretreatments</code></td>
<td>
<p>A vector containing names of all pretreatment variables.
Defaults to NULL, in which case the pretreatments slot will remain empty.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>result An object of class <code>mediation_data</code>, with separate slots for
each of the node types in a mediation analysis diagram.
</p>


<h3>See Also</h3>

<p>mediation_data-class
</p>


<h3>Examples</h3>

<pre><code class='language-R'># multiple outcomes, one mediator
mediation_data(
    demo_spline(), starts_with("outcome"), "treatment", "mediator"
)

# one outcome, multiple mediators
mediation_data(demo_joy(), "PHQ", "treatment", starts_with("ASV"))
</code></pre>

<hr>
<h2 id='mediation_models'>Accessor for Outcome Models</h2><span id='topic+mediation_models'></span>

<h3>Description</h3>

<p>Accessor for Outcome Models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediation_models(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mediation_models_+3A_object">object</code></td>
<td>
<p>An object of class multimedia whose outcome model estimates we
would like to extract.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing all the fitted mediation models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mindfulness)
exper &lt;- mediation_data(
    mindfulness,
    phyloseq::taxa_names(mindfulness),
    "treatment",
    starts_with("mediator"),
    "subject"
)

m &lt;- multimedia(exper)
mediation_models(m)
</code></pre>

<hr>
<h2 id='mediators'>Access Class-Specific Mediators</h2><span id='topic+mediators'></span>

<h3>Description</h3>

<p>This is a shorthand for accessing mediator-related slots in classes exported
by the multimedia package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediators(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mediators_+3A_object">object</code></td>
<td>
<p>An object whose mediators we want to access (either their names
or values).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output will depend on the class of the object that is passed in.
For multimedia objects, this returns a character vector of mediators. For
mediation data objects, this returns the mediator mediator data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
mediators(exper)
</code></pre>

<hr>
<h2 id='mediators+2Cmediation_data-method'>Access to @mediators in Mediation Data</h2><span id='topic+mediators+2Cmediation_data-method'></span>

<h3>Description</h3>

<p>This is an accessor returns the @mediators slot in a mediation_data object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mediation_data'
mediators(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mediators+2B2Cmediation_data-method_+3A_object">object</code></td>
<td>
<p>An object of class mediation_data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>m A data.frame whose rows are samples and columns are values of
mediators across those samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
mediators(exper)
</code></pre>

<hr>
<h2 id='mediators+2Cmultimedia-method'>Names of Mediators in a Multimedia Object</h2><span id='topic+mediators+2Cmultimedia-method'></span>

<h3>Description</h3>

<p>This is a helper that returns the names of the mediator variables in an
object of class multimedia. It parses the graph in the DAG specifying the
mediation analysis, and it returns all variables between treatment and
outcome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multimedia'
mediators(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mediators+2B2Cmultimedia-method_+3A_object">object</code></td>
<td>
<p>An object of class multimedia.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>m A vector of strings containing the names of all the mediators.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    mediators()
</code></pre>

<hr>
<h2 id='mediators+26lt+3B-'>Set Mediators</h2><span id='topic+mediators+3C-'></span>

<h3>Description</h3>

<p>This is an setter method for mediators in an S4 object, usually of class
mediation_data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mediators(object) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mediators+2B26lt+2B3B-_+3A_object">object</code></td>
<td>
<p>An object whose mediators slot to modify.</p>
</td></tr>
<tr><td><code id="mediators+2B26lt+2B3B-_+3A_value">value</code></td>
<td>
<p>The new mediator values to set within object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Modifies the mediators slot of the input object in place.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
mediators(exper) &lt;- data.frame(new_ASV = rnorm(100))
mediators(exper)
</code></pre>

<hr>
<h2 id='mediators+26lt+3B-+2Cmediation_data-method'>Set the Mediators in a Mediation Data Object</h2><span id='topic+mediators+3C-+2Cmediation_data-method'></span>

<h3>Description</h3>

<p>This is an setter method for the mediators slot in a mediation data object.
It lets you supply a new mediators data.frame for the object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'mediation_data'
mediators(object) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mediators+2B26lt+2B3B-+2B2Cmediation_data-method_+3A_object">object</code></td>
<td>
<p>An object of class mediation_data.</p>
</td></tr>
<tr><td><code id="mediators+2B26lt+2B3B-+2B2Cmediation_data-method_+3A_value">value</code></td>
<td>
<p>The new mediators values for the object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of <code>object</code> whose mediators slot has been replaced.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
mediators(exper) &lt;- data.frame(m = 1:10)
exper
</code></pre>

<hr>
<h2 id='mindfulness'>Mindfulness Dataset</h2><span id='topic+mindfulness'></span>

<h3>Description</h3>

<p>Data from a study of the relationship between mindfulness and the microbiome,
stored as a phyloseq object. Measurements are from 50 subjects before and
after a real and active control mindfulness intervention. We are interested
in changes in subject-level metadata, stored in the sample_data slot
</p>


<h3>Format</h3>

<p>An object of class phyloseq.
</p>


<h3>Value</h3>

<p>A phyloseq object with 307 samples and 55 taxa. Samples are described
by 7 variables (potential mediators) in the sample_data slot. There is no
associated phylogenetic tree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mindfulness)
</code></pre>

<hr>
<h2 id='model-class'>Representation of an Outcome or Mediation Model</h2><span id='topic+model-class'></span>

<h3>Description</h3>

<p>To work with many model types simultaneously, multimedia uses a model class
with the necessary mediation model functionality that wraps any specific
implementation. The slots below define the generally required functionality
for any specific implementation.
</p>


<h3>Slots</h3>


<dl>
<dt><code>estimator</code></dt><dd><p>A function that takes a formula, input data frame X, and an
response data.frame $Y$ and returns a model. For example, for the random
forest model, this is created by wrapping <code>parallelize()</code> on the ranger()
function for random forest estimation function using the 'ranger' package.</p>
</dd>
<dt><code>estimates</code></dt><dd><p>A list containing the estimated model.</p>
</dd>
<dt><code>sampler</code></dt><dd><p>A function that supports sampling new responses from the
estimated model.</p>
</dd>
<dt><code>model_type</code></dt><dd><p>A string specifying the type of model associated with the
class. For example, &quot;rf_model()&quot; denotes a random forest model.</p>
</dd>
<dt><code>predictor</code></dt><dd><p>A function that returns fitted predictions given new inputs.
For example, this can be the original predict() method for a multivariate
response model, or it can be a loop over predicts for each feature in the
mediation or outcome model.</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- lm_model()
estimator(m)(mpg ~ hp + wt, data = mtcars)

m &lt;- rf_model()
estimator(m)(mpg ~ hp + wt, data = mtcars)
</code></pre>

<hr>
<h2 id='multimedia'><code>multimedia</code> Constructor</h2><span id='topic+multimedia'></span>

<h3>Description</h3>

<p><code>multimedia</code> objects encapsulate the model and data that underlie a mediation
analysis, together with metadata (like graph structure) that contextualize
the estimation. This function can be used to construct a new multimedia
instances from a <code>mediation_data</code> dataset and pair of estimators.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multimedia(
  mediation_data,
  outcome_estimator = lm_model(),
  mediation_estimator = lm_model()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multimedia_+3A_mediation_data">mediation_data</code></td>
<td>
<p>An object of class <code>mediation_data</code>, with separate
slots for each of the node types in a mediation analysis diagram.</p>
</td></tr>
<tr><td><code id="multimedia_+3A_outcome_estimator">outcome_estimator</code></td>
<td>
<p>An object of class <code>model</code> that will be used to
estimate the outcome model.</p>
</td></tr>
<tr><td><code id="multimedia_+3A_mediation_estimator">mediation_estimator</code></td>
<td>
<p>An object of class <code>model</code> that will be used to
estimate the mediation model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>multimedia</code> encapsulating the full mediation
model and data.
</p>


<h3>See Also</h3>

<p>multimedia-class
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper)

exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
multimedia(exper)

# real data example with a pretreatment variable
data(mindfulness)
exper &lt;- mediation_data(
    mindfulness,
    phyloseq::taxa_names(mindfulness),
    "treatment",
    starts_with("mediator"),
    "subject"
)
multimedia(exper)
</code></pre>

<hr>
<h2 id='n_mediators'>Number of Mediators in a Multimedia Object</h2><span id='topic+n_mediators'></span>

<h3>Description</h3>

<p>Number of Mediators in a Multimedia Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n_mediators(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="n_mediators_+3A_object">object</code></td>
<td>
<p>An object of class multimedia.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer specifying the number of mediators.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    n_mediators()
</code></pre>

<hr>
<h2 id='n_outcomes'>Number of Outcomes in a Multimedia Object</h2><span id='topic+n_outcomes'></span>

<h3>Description</h3>

<p>Number of Outcomes in a Multimedia Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>n_outcomes(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="n_outcomes_+3A_object">object</code></td>
<td>
<p>An object of class multimedia.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer specifying the number of outcomes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    n_outcomes()
</code></pre>

<hr>
<h2 id='nrow+2Cmediation_data-method'>How many samples in the mediation dataset?</h2><span id='topic+nrow+2Cmediation_data-method'></span>

<h3>Description</h3>

<p>How many samples in the mediation dataset?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mediation_data'
nrow(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nrow+2B2Cmediation_data-method_+3A_x">x</code></td>
<td>
<p>The <code>mediation_data</code> object whose number of samples we want to
return.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer giving the number of samples in the mediation object.
</p>

<hr>
<h2 id='null_contrast'>Compare Effects from Experimental vs. Null Mediation Data</h2><span id='topic+null_contrast'></span>

<h3>Description</h3>

<p>One way to calibrate our conclusions from complex workflows is to see how
they would look on data where we know that there is no effect. This function
compares estimators f between real and synthetic null data, where the null
removes a set of edges according to the nullfication argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>null_contrast(model, exper, nullification = "T-&gt;Y", f = direct_effect)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="null_contrast_+3A_model">model</code></td>
<td>
<p>An object of class multimedia with specified mediation and
outcome models that we want to re-estimate across B bootstrap samples.</p>
</td></tr>
<tr><td><code id="null_contrast_+3A_exper">exper</code></td>
<td>
<p>An object of class multimedia_data containing the mediation and
outcome data from which the direct effects are to be estimated.</p>
</td></tr>
<tr><td><code id="null_contrast_+3A_nullification">nullification</code></td>
<td>
<p>A string specifying the types of edges whose effects we
want to remove in the null samples. Valid options are &quot;T-&gt;Y&quot; (the default),
&quot;T-&gt;M&quot;, &quot;M-&gt;Y&quot;, which remove direct effects, treatment to mediator effects,
and mediator to treatment effects, respectively.</p>
</td></tr>
<tr><td><code id="null_contrast_+3A_f">f</code></td>
<td>
<p>The estimator that we want to compare between real and null data.
This is assumed to be a function taking counterfactual samples, for example
<code>direct_effects</code> or <code>indirect_effects</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame containing estimates on the real and synthetic data for
every coordinate in the estimator f. The column <code>source</code> specifies whether
the estimate was calculated using real or synthetic null data.
</p>


<h3>See Also</h3>

<p>null_contrast fdr_summary
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example with null data - notice synthetic data has larger effect.
exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    estimate(exper) |&gt;
    null_contrast(exper)

# example with another dataset - synthetic effect is smaller.
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
multimedia(exper) |&gt;
    estimate(exper) |&gt;
    null_contrast(exper)
</code></pre>

<hr>
<h2 id='nullify'>Nullify Active Edges</h2><span id='topic+nullify'></span>

<h3>Description</h3>

<p>For inference, we often want to work with synthetic negative controls. One
way to define them is to specify submodels of the full mediation analysis
model. This function defines submodels by removing estimated edges according
to a prespecified vector of IDs. For example, setting nulls = &quot;T -&gt; Y&quot; will
remove any direct effect when sampling or obtaining predictions for the full
mediation analysis model <code class="reqn">hat{Y}</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nullify(multimedia, nulls = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nullify_+3A_multimedia">multimedia</code></td>
<td>
<p>A fitted object of class multimedia with estimates along
all paths in the mediation analysis DAG.</p>
</td></tr>
<tr><td><code id="nullify_+3A_nulls">nulls</code></td>
<td>
<p>A string specifying the indices of edges to ignore. &quot;T-&gt;Y&quot;,
&quot;T-&gt;M&quot;, and &quot;M-&gt;Y&quot; will match all edges between treatment to outcome,
treatment to mediator, etc. Otherwise, the vector of indices specifying
which edges to ignore.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>multimedia A version of the input multimedia model with all edges
matching <code>nulls</code> removed. Enables sampling of synthetic null controls.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example with null data
exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)

nullify(fit, "T-&gt;M") |&gt;
    estimate(exper) |&gt;
    indirect_overall()
nullify(fit, "T-&gt;Y") |&gt;
    estimate(exper) |&gt;
    direct_effect()

# example with another dataset
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)

nullify(fit, "T-&gt;M") |&gt;
    estimate(exper) |&gt;
    indirect_overall()
nullify(fit, "T-&gt;Y") |&gt;
    estimate(exper) |&gt;
    direct_effect()
</code></pre>

<hr>
<h2 id='outcome_model'>Access the Outcome Model in a Multimedia Object</h2><span id='topic+outcome_model'></span>

<h3>Description</h3>

<p>This is an accessor to the outcome slot of a multimedia object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>outcome_model(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="outcome_model_+3A_object">object</code></td>
<td>
<p>An object of class multimedia.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL, if not fitted, or the model learned from the training mediation
data object. For models fit in parallel across outcomes (e.g.,
glmnet_model()) a list of separate model objects. For models fitted jointly
across outcomes (e.g., lnm_model()), a single model object of that class.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    outcome_model()
</code></pre>

<hr>
<h2 id='outcome_models'>Accessor for Outcome Models</h2><span id='topic+outcome_models'></span>

<h3>Description</h3>

<p>Accessor for Outcome Models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>outcome_models(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="outcome_models_+3A_object">object</code></td>
<td>
<p>An object of class multimedia whose outcome model estimates we
would like to extract.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing all the fitted outcome models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mindfulness)
exper &lt;- mediation_data(
    mindfulness,
    phyloseq::taxa_names(mindfulness),
    "treatment",
    starts_with("mediator"),
    "subject"
)

m &lt;- multimedia(exper)
outcome_models(m)
</code></pre>

<hr>
<h2 id='outcomes'>Access Outcomes</h2><span id='topic+outcomes'></span>

<h3>Description</h3>

<p>This is an getter method for outcomes in an S4 object, usually of class
mediation_data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>outcomes(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="outcomes_+3A_object">object</code></td>
<td>
<p>An object whose outcomes slot to modify.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the S4 class of the input object. If it is a
multimedia model object, it will return a character vector of the outcome
variable names. Otherwise it returns the outcome data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
outcomes(exper)
</code></pre>

<hr>
<h2 id='outcomes+2Cmediation_data-method'>Outcomes Data in a Mediation Data Object</h2><span id='topic+outcomes+2Cmediation_data-method'></span>

<h3>Description</h3>

<p>This is an accessor function to the @outcomes slot in a mediation data
object. It returns the entire outcomes dataset, in contrast to outcomes()
applied to a multimedia object, which only returns the names of the outcome
variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mediation_data'
outcomes(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="outcomes+2B2Cmediation_data-method_+3A_object">object</code></td>
<td>
<p>An object of class mediation_data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame whose rows are samples and columns different outcomes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
outcomes(exper)
</code></pre>

<hr>
<h2 id='outcomes+2Cmultimedia-method'>Names of Outcomes in a Multimedia Object</h2><span id='topic+outcomes+2Cmultimedia-method'></span>

<h3>Description</h3>

<p>This is a helper that returns the names of the outcome variables in an
object of class multimedia. It parses the graph in the DAG specifying the
mediation analysis, and it returns all variables of node type outcome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multimedia'
outcomes(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="outcomes+2B2Cmultimedia-method_+3A_object">object</code></td>
<td>
<p>An object of class multimedia.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>m A vector of strings containing the names of all the outcomes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    outcomes()

exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
multimedia(exper) |&gt;
    outcomes()
</code></pre>

<hr>
<h2 id='outcomes+26lt+3B-'>Set Outcomes
This is an setter method for outcomes in an S4 object, usually of class
mediation_data.</h2><span id='topic+outcomes+3C-'></span>

<h3>Description</h3>

<p>Set Outcomes
This is an setter method for outcomes in an S4 object, usually of class
mediation_data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>outcomes(object) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="outcomes+2B26lt+2B3B-_+3A_object">object</code></td>
<td>
<p>An object whose outcomes slot to modify.</p>
</td></tr>
<tr><td><code id="outcomes+2B26lt+2B3B-_+3A_value">value</code></td>
<td>
<p>The new outcome values to set within object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Modifies the outcomes slot of the input object in place.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
outcomes(exper) &lt;- data.frame(new_PHQ = rnorm(100))
</code></pre>

<hr>
<h2 id='outcomes+26lt+3B-+2Cmediation_data-method'>Set the Outcomes in a Mediation Data Object</h2><span id='topic+outcomes+3C-+2Cmediation_data-method'></span>

<h3>Description</h3>

<p>This is an setter method for the outcomes slot in a mediation data object. It
lets you supply a new outcomes data.frame for the object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'mediation_data'
outcomes(object) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="outcomes+2B26lt+2B3B-+2B2Cmediation_data-method_+3A_object">object</code></td>
<td>
<p>An object of class mediation_data.</p>
</td></tr>
<tr><td><code id="outcomes+2B26lt+2B3B-+2B2Cmediation_data-method_+3A_value">value</code></td>
<td>
<p>The new outcome values for the object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of <code>object</code> whose outcomes slot has been replaced.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
outcomes(exper) &lt;- data.frame(y = 1:10)
exper
</code></pre>

<hr>
<h2 id='parallelize'>Parallelize Estimation across Responses</h2><span id='topic+parallelize'></span>

<h3>Description</h3>

<p>For many mediation and outcome models, we simply want to apply a univariate
model across all response variable. Parallelize enables this conversion. For
example, applying parallelize to ranger() returns a function that estimates
separate random forest models for each response on the left hand side of a
formula.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parallelize(f, progress = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="parallelize_+3A_f">f</code></td>
<td>
<p>A function for estimating a single response model given a formula
and input dataset. This is the model that we would like to parallelize
across responses.</p>
</td></tr>
<tr><td><code id="parallelize_+3A_progress">progress</code></td>
<td>
<p>A logical indicating whether to show a progress bar.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>f_multi A function that takes a formula and dataset and applies f to
each response on the left hand side of the original formula.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mat &lt;- data.frame(matrix(rnorm(100), 25, 4))
colnames(mat) &lt;- c("y1", "y2", "x1", "x2")
plm &lt;- parallelize(lm)
plm(y1 + y2 ~ x1 + x2, mat)

prf &lt;- parallelize(ranger::ranger)
prf(mpg + hp ~ wt + disp + cyl, data = mtcars)
</code></pre>

<hr>
<h2 id='plot_mediators'>Visualize Indirect Effects</h2><span id='topic+plot_mediators'></span>

<h3>Description</h3>

<p>This is a helper function to visualize the raw data responsible for the
largest indirect effects. It returns a faceted plot of outcome vs. mediator
pairs for those with high pathwise indirect effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_mediators(
  indirect_effects,
  exper,
  n_digit = 3,
  n_panels = NULL,
  treatment = "treatment",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_mediators_+3A_indirect_effects">indirect_effects</code></td>
<td>
<p>A data.frame containing estimated indirect effects
for each variable, under different counterfactual settings for the &quot;direct
treatment.&quot; This is the output of <code>indirect_pathwise</code>.</p>
</td></tr>
<tr><td><code id="plot_mediators_+3A_exper">exper</code></td>
<td>
<p>An object of class <code>mediation_data</code> containing all mediation
analysis data.</p>
</td></tr>
<tr><td><code id="plot_mediators_+3A_n_digit">n_digit</code></td>
<td>
<p>The number of digits of the indirect effects to print next to
each panel. Defaults to 3.</p>
</td></tr>
<tr><td><code id="plot_mediators_+3A_n_panels">n_panels</code></td>
<td>
<p>The number of mediator-outcome pairs to show. Defaults to 12,
or the number of pathways, if there are fewer than 12.</p>
</td></tr>
<tr><td><code id="plot_mediators_+3A_treatment">treatment</code></td>
<td>
<p>What is the name of the treatment variable that we want to
overlay on points? This is necessary when there are several potential
treatment variables. Defaults to &quot;treatment.&quot;</p>
</td></tr>
<tr><td><code id="plot_mediators_+3A_...">...</code></td>
<td>
<p>Further keyword arguments passed to <code>patchwork::wrap_plots</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A patchwork-based arrangement of ggplot2 grobs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># dataset with no true effects
exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
ie &lt;- multimedia(exper) |&gt;
    estimate(exper) |&gt;
    indirect_pathwise() |&gt;
    effect_summary()
plot_mediators(ie, exper)

# another dataset
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
ie &lt;- multimedia(exper, rf_model()) |&gt;
    estimate(exper) |&gt;
    indirect_pathwise() |&gt;
    effect_summary()
plot_mediators(ie, exper)

</code></pre>

<hr>
<h2 id='plot_sensitivity'>Generic Sensitivity Plot</h2><span id='topic+plot_sensitivity'></span>

<h3>Description</h3>

<p>This function draws a curve of indirect effect against the sensitivity
parameter, allowing users to specify the name of x and y-axis variables using
the x_var and y_var inputs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_sensitivity(sensitivity_curve, x_var = "rho", y_var = "indirect_effect")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_sensitivity_+3A_sensitivity_curve">sensitivity_curve</code></td>
<td>
<p>The output of a call to sensitivity or
sensitivity_perturb. A data.frame whose columns are: outcome, {x_var},
{y_var}, and {y_var}_standard_error, where x_var and y_var are defined
in the two arguments below.</p>
</td></tr>
<tr><td><code id="plot_sensitivity_+3A_x_var">x_var</code></td>
<td>
<p>The type of perturbation variable to plot along the x-axis.
Defaults to &quot;rho&quot;, following the sensitivity approach implemented in
sensitivity_subset.</p>
</td></tr>
<tr><td><code id="plot_sensitivity_+3A_y_var">y_var</code></td>
<td>
<p>The type of effect to plot along the y-axis. Defaults to
indirect_effect.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 grob plotting the sensitivity parameter against the effect
specified by <code>y_var</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sensitivity_curve &lt;- read.csv(url("https://go.wisc.edu/j2kvcj"))
plot_sensitivity(sensitivity_curve)
</code></pre>

<hr>
<h2 id='predict_across'>Predict a Subset of Responses</h2><span id='topic+predict_across'></span>

<h3>Description</h3>

<p>Predict across selected responses in a mediation model object. This is a
lower-level version of the predict method that applies to objets of class
mediation_model. Rather than giving predictions across all outcomes, it
supports predictions across a subset specified as a vector. This can be
convenient when we want to analyze how a specific subset of outcomes changes
and do not need to run predictions across all possible.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_across(object, newdata, name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict_across_+3A_object">object</code></td>
<td>
<p>An object of class <code>model</code> containing an estimated model.</p>
</td></tr>
<tr><td><code id="predict_across_+3A_newdata">newdata</code></td>
<td>
<p>A data.frame containing new inputs from which to sample
responses. If NULL, defaults to the data used to estimate fit.</p>
</td></tr>
<tr><td><code id="predict_across_+3A_name">name</code></td>
<td>
<p>A string or index specifying which of the dimensions of a
multiresponse prediction to extract.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of predicted values for the outcome of interest.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)
predict_across(outcome_model(fit), NULL, "outcome_1")

# predict at newdata
newdata &lt;- bind_mediation(exper)
predict_across(
    outcome_model(fit),
    newdata[seq_len(5), ],
    c("outcome_1", "outcome_2")
)
</code></pre>

<hr>
<h2 id='predict+2Cmultimedia-method'>Predictions from a Multimedia Class</h2><span id='topic+predict+2Cmultimedia-method'></span>

<h3>Description</h3>

<p>This generalizes the built-in predict method to the multimedia class.  Given
an estimated multimedia object, this function supports prediction along the
estimated DAG. It first predicts <code style="white-space: pre;">&#8288;hat[M] | T, X&#8288;</code> and then <code style="white-space: pre;">&#8288;hat[Y] | hat[M], T, X&#8288;</code>. Each prediction step will call the prediction method within the
mediation and outcome models that make up the multimedia object on which this
is called. By passing in new treatment, mediator, or pretreatment data, you
can predict forward along the DAG using these as inputs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multimedia'
predict(object, profile = NULL, mediators = NULL, pretreatment = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict+2B2Cmultimedia-method_+3A_object">object</code></td>
<td>
<p>An object of class multimedia containing the estimated
mediation and outcome models whose mediation and outcome predictions we
want to obtain.</p>
</td></tr>
<tr><td><code id="predict+2B2Cmultimedia-method_+3A_profile">profile</code></td>
<td>
<p>An object of class <code>treatment_profile</code> containing the
treatment profile to consider in the difference. Defaults to a profile with
all the unique treatment configurations observed in the original data,
shared across both the mediators and outcomes.</p>
</td></tr>
<tr><td><code id="predict+2B2Cmultimedia-method_+3A_mediators">mediators</code></td>
<td>
<p>By default, we will return outcome predictions using the
predicted mediators from the mediation model. Modify this argument if you
would like to directly control the mediation inputs for the outcome model.
Must be a data.frame whose columns are named to match the
<code>mediators(object)</code>.</p>
</td></tr>
<tr><td><code id="predict+2B2Cmultimedia-method_+3A_pretreatment">pretreatment</code></td>
<td>
<p>By default, we will return mediation and outcome model
predictions using the same pretreatment variables as used when initially
estimating the models (like setting <code>newdata = NULL</code> in usual <code>predict</code>).
To pass in different pretreatment variables, provide a data.frame here
whose columns match the pretreatments as the originally trained mediation
and outcome models.</p>
</td></tr>
<tr><td><code id="predict+2B2Cmultimedia-method_+3A_...">...</code></td>
<td>
<p>A placeholder to agree with <code>predict</code> in base R. Not ever used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two elements:
$mediators: A data.frame containing predicted values for the mediators.
Each row corresponds to one row of the newdata, or one row of the default
treatment profile, if no newdata is given.
</p>
<p>$outcomes: A data.frame containing predicted values for the outcomes, given
either (i) the predicted values of the mediators or (ii) the provided
values of the mediators. Each row corresponds to one row of the newdata, or
one row of the default treatment profile, if no newdata is given.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
fit &lt;- multimedia(exper, glmnet_model()) |&gt;
    estimate(exper)
predict(fit)

# at new treatment configurations
t1 &lt;- data.frame(treatment = factor(rep(c(0, 1), each = 5)))
profile &lt;- setup_profile(fit, t_mediator = t1, t_outcome = t1)
predict(fit, profile)

# at new treatment and mediator configurations
mediators &lt;- data.frame(mediator = rnorm(10, 0, 1))
predict(fit, profile, mediators)
</code></pre>

<hr>
<h2 id='pretreatments'>Access Pretreatments</h2><span id='topic+pretreatments'></span>

<h3>Description</h3>

<p>This is an getter method for pretreatments in an S4 object, usually of class
mediation_data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pretreatments(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pretreatments_+3A_object">object</code></td>
<td>
<p>An object whose pretreatments slot to modify.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the S4 class of the input object. If it is a
multimedia model object, it will return a character vector of the outcome
variable names. Otherwise it returns the outcome data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
pretreatments(exper)
</code></pre>

<hr>
<h2 id='pretreatments+2Cmediation_data-method'>Pretreatments in a Mediation Data Object</h2><span id='topic+pretreatments+2Cmediation_data-method'></span>

<h3>Description</h3>

<p>This is an accessor function to the @pretreatments slot in a mediation data
object. It returns the entire set of observed pretreatments, in contrast to
pretreatments() applied to a multimedia object, which only returns the names
of the pretreatment variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mediation_data'
pretreatments(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pretreatments+2B2Cmediation_data-method_+3A_object">object</code></td>
<td>
<p>An object of class mediation_data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame whose rows are samples and columns different
pretreatments.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
pretreatments(exper)
</code></pre>

<hr>
<h2 id='pretreatments+26lt+3B-'>Set Pretreatments
This is an setter method for pretreatments in an S4 object, usually of class
mediation_data.</h2><span id='topic+pretreatments+3C-'></span>

<h3>Description</h3>

<p>Set Pretreatments
This is an setter method for pretreatments in an S4 object, usually of class
mediation_data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pretreatments(object) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pretreatments+2B26lt+2B3B-_+3A_object">object</code></td>
<td>
<p>An object whose pretreatments slot to modify.</p>
</td></tr>
<tr><td><code id="pretreatments+2B26lt+2B3B-_+3A_value">value</code></td>
<td>
<p>The new pretreatment values to set within object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Modifies the pretreatments slot of the input object in place.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
pretreatments(exper) &lt;- data.frame(new_var = rnorm(100))
</code></pre>

<hr>
<h2 id='pretreatments+26lt+3B-+2Cmediation_data-method'>Set the Pretreatments in a Mediation Data Object</h2><span id='topic+pretreatments+3C-+2Cmediation_data-method'></span>

<h3>Description</h3>

<p>This is an setter method for the pretreatments slot in a mediation data
object. It lets you supply a new pretreatments data.frame for the object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'mediation_data'
pretreatments(object) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pretreatments+2B26lt+2B3B-+2B2Cmediation_data-method_+3A_object">object</code></td>
<td>
<p>An object of class mediation_data.</p>
</td></tr>
<tr><td><code id="pretreatments+2B26lt+2B3B-+2B2Cmediation_data-method_+3A_value">value</code></td>
<td>
<p>The new pretreatment values for the object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of <code>object</code> whose pretreatments slot has been replaced.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
pretreatments(exper) &lt;- data.frame(x = 1:10)
exper
</code></pre>

<hr>
<h2 id='retrieve_names'>Variables in a Multimedia Object</h2><span id='topic+retrieve_names'></span>

<h3>Description</h3>

<p>This returns all the variables modeled within a multimedia object. This can
be helpful for overviewing an experiment and is called by the print methods
in this package. Also supports filtering to specific node types, e.g.,
mediators.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retrieve_names(object, nm)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="retrieve_names_+3A_object">object</code></td>
<td>
<p>An object of class multimedia</p>
</td></tr>
<tr><td><code id="retrieve_names_+3A_nm">nm</code></td>
<td>
<p>A string specifying the node type to filter down to, e.g.,
'treatment' or 'mediator'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing all the names of the variables of the
node type <code>nm</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    retrieve_names("mediator")
</code></pre>

<hr>
<h2 id='rf_model'>Random Forest Model</h2><span id='topic+rf_model'></span>

<h3>Description</h3>

<p>Apply a random forest model in parallel across a vector of responses $y$ in
either an outcome or mediation model. This is a natural choice when the
relationship between inputs and outputs is thought to be nonlinear.
Internally, each of the models across the response are estimated using
the 'ranger' package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rf_model(progress = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rf_model_+3A_progress">progress</code></td>
<td>
<p>A logical indicating whether to show a progress bar during
estimation.</p>
</td></tr>
<tr><td><code id="rf_model_+3A_...">...</code></td>
<td>
<p>Keyword parameters passed to ranger() in the 'ranger' package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>model An object of class <code>model</code> with estimator, predictor, and
sampler functions associated wtih a lienar model.
</p>


<h3>See Also</h3>

<p>model lm_model rf_model glmnet_model brms_model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper, rf_model(num.trees = 10)) |&gt;
    estimate(exper)

# example with another dataset
exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
multimedia(exper, rf_model(num.trees = 20, max.depth = 2)) |&gt;
    estimate(exper)
</code></pre>

<hr>
<h2 id='rf_sampler'>Sample from a Random Forest Model</h2><span id='topic+rf_sampler'></span>

<h3>Description</h3>

<p>This assumes a continuous response, so that the out-of-sample MSE can be used
to estimate the outcome variability <code class="reqn">\sigma</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rf_sampler(fits, newdata = NULL, indices = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rf_sampler_+3A_fits">fits</code></td>
<td>
<p>The fitted RF model from which to draw samples.</p>
</td></tr>
<tr><td><code id="rf_sampler_+3A_newdata">newdata</code></td>
<td>
<p>A data.frame containing new inputs from which to sample
responses. If NULL, defaults to the data used to estimate fit.</p>
</td></tr>
<tr><td><code id="rf_sampler_+3A_indices">indices</code></td>
<td>
<p>The coordinates of the response from which to draw samples.</p>
</td></tr>
<tr><td><code id="rf_sampler_+3A_...">...</code></td>
<td>
<p>Additional parameters passed to rf_model's predict method.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>y_star A data.frame of samples y associated with the new inputs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- rf_model()
fit &lt;- estimator(m)(mpg ~ hp + wt, data = mtcars)
rf_sampler(fit, mtcars)

prf &lt;- parallelize(ranger::ranger)
fit &lt;- prf(mpg + disp ~ hp + wt, data = mtcars)
rf_sampler(fit, mtcars)
</code></pre>

<hr>
<h2 id='sample+2Cmultimedia-method'>Sample New Mediator/Outcome Data</h2><span id='topic+sample+2Cmultimedia-method'></span>

<h3>Description</h3>

<p>This generalizes the built-in sample method to the multimedia class. Given an
estimated multimedia object, this function supports sampling along the
estimated DAG. It first samples <code style="white-space: pre;">&#8288;M* | T, X&#8288;</code> and then <code style="white-space: pre;">&#8288;Y* | M*, T, X&#8288;</code>. Each
sampling step will call the sample method within the mediation and outcome
models that make up the multimedia object on which this is called.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multimedia'
sample(x, size, pretreatment = NULL, profile = NULL, mediators = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sample+2B2Cmultimedia-method_+3A_x">x</code></td>
<td>
<p>An object of class multimedia containing the estimated
mediation and outcome models whose mediation and outcome samples we
want to obtain.</p>
</td></tr>
<tr><td><code id="sample+2B2Cmultimedia-method_+3A_size">size</code></td>
<td>
<p>A placeholder argument to agree with the default <code>sample</code> method
in R base. We always return the number of samples as set in either the
original input <code>x</code> or a new input <code>profile</code>.</p>
</td></tr>
<tr><td><code id="sample+2B2Cmultimedia-method_+3A_pretreatment">pretreatment</code></td>
<td>
<p>By default, we will return mediation and outcome model
predictions using the same pretreatment variables as used when initially
estimating the models (like setting <code>newdata = NULL</code> in usual <code>predict</code>).
To pass in different pretreatment variables, provide a data.frame here
whose columns match the pretreatments as the originally trained mediation
and outcome models.</p>
</td></tr>
<tr><td><code id="sample+2B2Cmultimedia-method_+3A_profile">profile</code></td>
<td>
<p>An object of class <code>treatment_profile</code> containing the
treatment profile to consider in the difference. Defaults to a profile with
all the unique treatment configurations observed in the original data,
shared across both the mediators and outcomes.</p>
</td></tr>
<tr><td><code id="sample+2B2Cmultimedia-method_+3A_mediators">mediators</code></td>
<td>
<p>By default, we will return outcome predictions using the
predicted mediators from the mediation model. Modify this argument if you
would like to directly control the mediation inputs for the outcome model.
Must be a data.frame whose columns are named to match the
<code>mediators(object)</code>.</p>
</td></tr>
<tr><td><code id="sample+2B2Cmultimedia-method_+3A_...">...</code></td>
<td>
<p>Additional options to pass to the @sampler method in the estimated
mediation model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>multimedia</code> with mediator and outcome slots
sampled according to the description above.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)
samples &lt;- sample(fit)
mediators(samples)
outcomes(samples)

# sampling with just different "n" has no effect.
samples &lt;- sample(fit, 100)

# Instead sample at a new treatment configuration
t1 &lt;- data.frame(treatment = factor(rep(c(0, 1), each = 50)))
profile &lt;- setup_profile(fit, t_mediator = t1, t_outcome = t1)
samples &lt;- sample(fit, profile = profile)
mediators(samples)
outcomes(samples)
</code></pre>

<hr>
<h2 id='sensitivity'>Sensitivity Analysis for Overall Indirect Effect</h2><span id='topic+sensitivity'></span>

<h3>Description</h3>

<p>For causal identification, mediation analysis relies on several untestable
assumptions. One important one is that there is no confounding between the
counterfactual mediator and outcome variables. Even though we can never know
whether this exists, we can measure the sensitivity of our conclusions to the
existence/strength of such a confounder. In this function, we approach this
by inducing (unallowable) correlation between the mediator and outcome model
residuals, simulate forward, and see how the estimated overall indirect
effect changes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sensitivity(
  model,
  exper,
  confound_ix = NULL,
  rho_seq = NULL,
  n_bootstrap = 100,
  progress = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensitivity_+3A_model">model</code></td>
<td>
<p>A <code>multimedia</code> object containing the fitted models for
sensitivity analysis. Note that since our approach relies on correlating
simulated residual error, it is only applicable to models of class
<code>lm_model()</code>, <code>glmnet_model()</code> and <code>rf_model()</code>.</p>
</td></tr>
<tr><td><code id="sensitivity_+3A_exper">exper</code></td>
<td>
<p>The original <code>mediation_data</code> class object used to fit <code>model</code>.
These observations will be resampled to support bootstrap confidence
interval construction of the sensitivity curve.</p>
</td></tr>
<tr><td><code id="sensitivity_+3A_confound_ix">confound_ix</code></td>
<td>
<p>A data.frame specifying which mediator/outcome should
be allowed to be correlated. Should have two columns: 'mediator' and
'outcome' specifying which pairs of mediators and outcomes should be
correlated. Defaults to NULL, which creates a data.frame with no rows (and
so enforcing independence between mediators and outcomes)</p>
</td></tr>
<tr><td><code id="sensitivity_+3A_rho_seq">rho_seq</code></td>
<td>
<p>We will evaluate correlations Cor(e', e) between mediation and
outcome model errors ranging along this grid. Defaults to NULL, which
internally sets the sequence to rho = (-0.9, -0.7 ..., 0.7, 0.9).</p>
</td></tr>
<tr><td><code id="sensitivity_+3A_n_bootstrap">n_bootstrap</code></td>
<td>
<p>The number of bootstrap resamples used to build confidence
bands around the sensitivity curves. Defaults to 100.</p>
</td></tr>
<tr><td><code id="sensitivity_+3A_progress">progress</code></td>
<td>
<p>A logical indicating whether to show a progress bar.</p>
</td></tr>
<tr><td><code id="sensitivity_+3A_...">...</code></td>
<td>
<p>Additional keyword arguments passed to <code>indirect_overall</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> giving the outputs of <code>indirect_overall</code> across many
values of the correlation rho.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xy_data &lt;- demo_spline()
exper &lt;- mediation_data(
    xy_data, starts_with("outcome"), "treatment", "mediator"
)
model &lt;- multimedia(
    exper,
    outcome_estimator = glmnet_model(lambda = 1e-2)
) |&gt;
    estimate(exper)
rho_seq &lt;- c(-0.2, 0, 0.2)
confound_ix &lt;- expand.grid(mediator = 1, outcome = 1:2)
sensitivity(model, exper, confound_ix, rho_seq, n_bootstrap = 2)
</code></pre>

<hr>
<h2 id='sensitivity_pathwise'>Sensitivity Analysis for Pathwise Indirect Effects</h2><span id='topic+sensitivity_pathwise'></span>

<h3>Description</h3>

<p>For causal identification, mediation analysis relies on several untestable
assumptions. One important one is that there is no confounding between the
counterfactual mediator and outcome variables. Even though we can never know
whether this exists, we can measure the sensitivity of our conclusions to the
existence/strength of such a confounder. In this function, we approach this
by inducing (unallowable) correlation between the mediator and outcome model
residuals, simulate forward, and see how the estimated pathwise indirect
effects change.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sensitivity_pathwise(
  model,
  exper,
  confound_ix = NULL,
  rho_seq = NULL,
  n_bootstrap = 100,
  progress = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensitivity_pathwise_+3A_model">model</code></td>
<td>
<p>A <code>multimedia</code> object containing the fitted models for
sensitivity analysis. Note that since our approach relies on correlating
simulated residual error, it is only applicable to models of class
<code>lm_model()</code>, <code>glmnet_model()</code> and <code>rf_model()</code>.</p>
</td></tr>
<tr><td><code id="sensitivity_pathwise_+3A_exper">exper</code></td>
<td>
<p>The original <code>mediation_data</code> class object used to fit <code>model</code>.
These observations will be resampled to support bootstrap confidence
interval construction of the sensitivity curve.</p>
</td></tr>
<tr><td><code id="sensitivity_pathwise_+3A_confound_ix">confound_ix</code></td>
<td>
<p>A data.frame specifying which mediator/outcome should
be allowed to be correlated. Should have two columns: 'mediator' and
'outcome' specifying which pairs of mediators and outcomes should be
correlated. Defaults to NULL, which creates a data.frame with no rows (and
so enforcing independence between mediators and outcomes)</p>
</td></tr>
<tr><td><code id="sensitivity_pathwise_+3A_rho_seq">rho_seq</code></td>
<td>
<p>We will evaluate correlations Cor(e', e) between mediation and
outcome model errors ranging along this grid. Defaults to NULL, which
internally sets the sequence to rho = (-0.9, -0.7 ..., 0.7, 0.9).</p>
</td></tr>
<tr><td><code id="sensitivity_pathwise_+3A_n_bootstrap">n_bootstrap</code></td>
<td>
<p>The number of bootstrap resamples used to build confidence
bands around the sensitivity curves. Defaults to 100.</p>
</td></tr>
<tr><td><code id="sensitivity_pathwise_+3A_progress">progress</code></td>
<td>
<p>A logical indicating whether to show a progress bar.</p>
</td></tr>
<tr><td><code id="sensitivity_pathwise_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to indirect_pathwise.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>date.frame</code> giving the outputs of <code>indirect_overall</code> across many
values of the correlation rho.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xy_data &lt;- demo_spline()
exper &lt;- mediation_data(
    xy_data, starts_with("outcome"), "treatment", "mediator"
)
model &lt;- multimedia(
    exper,
    outcome_estimator = glmnet_model(lambda = 1e-2)
) |&gt;
    estimate(exper)
rho_seq &lt;- c(-0.2, 0, 0.2)
subset_indices &lt;- expand.grid(
    mediator = n_mediators(model), outcome = n_outcomes(model)
)
sensitivity_pathwise(model, exper, subset_indices, rho_seq, n_bootstrap = 2)
</code></pre>

<hr>
<h2 id='sensitivity_perturb'>Sensitivity to User-Specified Perturbations</h2><span id='topic+sensitivity_perturb'></span>

<h3>Description</h3>

<p>The more standard sensitivity and sensitivity_pathwise functions support
sensitivity analysis to violations in assumptions restricted to specific
mediator-outcome pairs. For more general violations, this function allows
arbitrary modification of the default, diagonal covariance matrix structure
across both mediators and outcomes. This makes it possible to ask what
happens when mediators are correlated with one another or when more some
pairs of mediator-outcome pairs have much stronger correlation than others.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sensitivity_perturb(
  model,
  exper,
  perturb,
  nu_seq = NULL,
  n_bootstrap = 100,
  progress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sensitivity_perturb_+3A_model">model</code></td>
<td>
<p>A <code>multimedia</code> object containing the fitted models for
sensitivity analysis. Note that since our approach relies on correlating
simulated residual error, it is only applicable to models of class
<code>lm_model()</code>, <code>glmnet_model()</code> and <code>rf_model()</code>.</p>
</td></tr>
<tr><td><code id="sensitivity_perturb_+3A_exper">exper</code></td>
<td>
<p>The original <code>mediation_data</code> class object used to fit <code>model</code>.
These observations will be resampled to support bootstrap confidence
interval construction of the sensitivity curve.</p>
</td></tr>
<tr><td><code id="sensitivity_perturb_+3A_perturb">perturb</code></td>
<td>
<p>A matrix towards which the original mediator-outcome
covariance should be perturbed. Must have dimension (n_mediators +
n_outcomes) x (n_mediators + n_outcomes).</p>
</td></tr>
<tr><td><code id="sensitivity_perturb_+3A_nu_seq">nu_seq</code></td>
<td>
<p>The strength of the perturbation towards the matrix perturb.</p>
</td></tr>
<tr><td><code id="sensitivity_perturb_+3A_n_bootstrap">n_bootstrap</code></td>
<td>
<p>The number of bootstrap resamples used to build confidence
bands around the sensitivity curves. Defaults to 100.</p>
</td></tr>
<tr><td><code id="sensitivity_perturb_+3A_progress">progress</code></td>
<td>
<p>A logical indicating whether to show a progress bar.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Specifically, it defines a new covariance matrix across mediators and
outcomes according to diag(sigma^2_mediator, sigma^2_outcome) + nu * perturb.
The estimates sigma^2 are taken from the residuals in the original mediation
and outcome models, and perturb and nu are provided by the user.
</p>


<h3>Value</h3>

<p>A <code>date.frame</code> giving the outputs of <code>indirect_overall</code> across many
values of the correlation rho.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xy_data &lt;- demo_spline()
exper &lt;- mediation_data(
    xy_data, starts_with("outcome"), "treatment", "mediator"
)
model &lt;- multimedia(
    exper,
    outcome_estimator = glmnet_model(lambda = 1e-2)
) |&gt;
    estimate(exper)
nu_seq &lt;- c(-0.2, 0.2)
perturb &lt;- matrix(
    c(
        0, 3, 0,
        3, 0, 0,
        0, 0, 0
    ),
    nrow = 3, byrow = TRUE
)
sensitivity_perturb(model, exper, perturb, nu_seq, n_bootstrap = 2)
</code></pre>

<hr>
<h2 id='setup_profile'>Define a <code>treatment_profile</code> object</h2><span id='topic+setup_profile'></span>

<h3>Description</h3>

<p>For general mediation analysis, we need to provide counterfactuals for both
the outcome and mediator components of each sample. That is, we need to
understand Y(t, M(t')) where t and t' may not be the same.
<code>treatment_profile</code> classes place some more structural requirements on
treatment profiles, so that later effect estimation can make simplifying
assumptions. This function creates a treatment profile from a collection of
possible mediator and outcome treatments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setup_profile(x, t_mediator = NULL, t_outcome = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setup_profile_+3A_x">x</code></td>
<td>
<p>An object of class <code>multimedia</code> specifying the complete mediation
analysis DAG. The treatment, mediator, and outcome names are necessary to
build a profile of counterfactual treatments across each of these
variables.</p>
</td></tr>
<tr><td><code id="setup_profile_+3A_t_mediator">t_mediator</code></td>
<td>
<p>A data.frame whose columns store treatment names and whose
values are the treatment assignments to each sample (row). Defaults to
NULL, in which case this type of data.frame is constructed from the
treatment assignments in the <code>mediation_data</code>'s <code style="white-space: pre;">&#8288;@treatment&#8288;</code> slot. Each
column must be either a numeric or factor variable.</p>
</td></tr>
<tr><td><code id="setup_profile_+3A_t_outcome">t_outcome</code></td>
<td>
<p>A data.frame analogous to <code>t_mediator</code>, but applying to the
outcome node.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>treatment_profile</code> giving treatment assignments
for both mediation and outcome terms.
</p>


<h3>See Also</h3>

<p>check_profile
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)

t1 &lt;- data.frame(treatment = factor(rep(c(0, 1), each = 5)))
profile &lt;- setup_profile(fit, t_mediator = t1, t_outcome = t1)
profile

t2 &lt;- data.frame(treatment = factor(rep(0, 10)))
profile &lt;- setup_profile(fit, t_mediator = t1, t_outcome = t2)
profile
</code></pre>

<hr>
<h2 id='spline_fun'>Generate Random Spline</h2><span id='topic+spline_fun'></span>

<h3>Description</h3>

<p>This generates random spline functions. It is used in the toy nonlinear
dataset created in <code>demo_spline()</code>. This is not necessary for the essential
multimedia workflow, it is only exported for potential independent interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spline_fun(D = 2, knots = NULL, h_ix = seq_len(10), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="spline_fun_+3A_d">D</code></td>
<td>
<p>The number random spline functions to generate internally.</p>
</td></tr>
<tr><td><code id="spline_fun_+3A_knots">knots</code></td>
<td>
<p>The location of knots to use in the spline functions. Defaults
to -4, -2, 0, 2, 4.</p>
</td></tr>
<tr><td><code id="spline_fun_+3A_h_ix">h_ix</code></td>
<td>
<p>The locations along which to generate the underlying spline
function.</p>
</td></tr>
<tr><td><code id="spline_fun_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass to <code>rnorm</code> during the random noise
generation for each call of the returned function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A function that can be used to sample points along the random spline
functions. Takes argument 'x' giving the input to the spline and returns a
D-dimensional response y giving random samples for each of the D splines..
</p>


<h3>See Also</h3>

<p>demo_spline
</p>

<hr>
<h2 id='sub_formula'>Helper to Modify Formulas</h2><span id='topic+sub_formula'></span>

<h3>Description</h3>

<p>Helper to Modify Formulas
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sub_formula(formula, yj)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sub_formula_+3A_formula">formula</code></td>
<td>
<p>The original formula whose response we want to modify.</p>
</td></tr>
<tr><td><code id="sub_formula_+3A_yj">yj</code></td>
<td>
<p>The desired response term for the formula.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new formula object with the LHS replaced by yj.
</p>

<hr>
<h2 id='treatment_profile-class'>Define a Treatment Profile</h2><span id='topic+treatment_profile-class'></span>

<h3>Description</h3>

<p>This class ensures appropriate structure of the treatment assignments for
mediator and outcome variables. It enforces certain structural requirements
(e.g., that the number of samples is the same under the mediator and outcome
counterfactuals) using the <code>check_profile</code> function.
</p>


<h3>See Also</h3>

<p>setup_profile check_profile
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_spline(tau = c(2, 1)) |&gt;
    mediation_data(starts_with("outcome"), "treatment", "mediator")
fit &lt;- multimedia(exper) |&gt;
    estimate(exper)

# helpers for defining treatment profiles
t1 &lt;- data.frame(treatment = factor(rep(c(0, 1), each = 5)))
profile &lt;- setup_profile(fit, t_mediator = t1, t_outcome = t1)
profile

t2 &lt;- data.frame(treatment = factor(rep(0, 10)))
profile &lt;- setup_profile(fit, t_mediator = t1, t_outcome = t2)
profile
</code></pre>

<hr>
<h2 id='treatments'>Access Treatments</h2><span id='topic+treatments'></span>

<h3>Description</h3>

<p>This is an getter method for treatments in an S4 object, usually of class
mediation_data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>treatments(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="treatments_+3A_object">object</code></td>
<td>
<p>An object whose treatments slot to modify.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output depends on the S4 class of the input object. If it is a
multimedia model object, it will return a character vector of the outcome
variable names. Otherwise it returns the outcome data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
treatments(exper)
</code></pre>

<hr>
<h2 id='treatments+2Cmediation_data-method'>Treatments in a Mediation Data Object</h2><span id='topic+treatments+2Cmediation_data-method'></span>

<h3>Description</h3>

<p>This is an accessor function to the @treatments slot in a mediation data
object. It returns the entire set of observed treatments, in contrast to
treatments() applied to a multimedia object, which only returns the names of
the treatment variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mediation_data'
treatments(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="treatments+2B2Cmediation_data-method_+3A_object">object</code></td>
<td>
<p>An object of class mediation_data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame whose rows are samples and columns different treatments.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
treatments(exper)
</code></pre>

<hr>
<h2 id='treatments+2Cmultimedia-method'>Names of Treatments in a Multimedia Object</h2><span id='topic+treatments+2Cmultimedia-method'></span>

<h3>Description</h3>

<p>This is a helper that returns the names of the treatment variables in an
object of class multimedia. It parses the graph in the DAG specifying the
mediation analysis, and it returns all variables of node type treatment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'multimedia'
treatments(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="treatments+2B2Cmultimedia-method_+3A_object">object</code></td>
<td>
<p>An object of class multimedia.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>m A vector of strings containing the names of all the treatments.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
multimedia(exper) |&gt;
    treatments()
</code></pre>

<hr>
<h2 id='treatments+26lt+3B-'>Set Treatments
This is an setter method for treatments in an S4 object, usually of class
mediation_data.</h2><span id='topic+treatments+3C-'></span>

<h3>Description</h3>

<p>Set Treatments
This is an setter method for treatments in an S4 object, usually of class
mediation_data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>treatments(object) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="treatments+2B26lt+2B3B-_+3A_object">object</code></td>
<td>
<p>An object whose treatments slot to modify.</p>
</td></tr>
<tr><td><code id="treatments+2B26lt+2B3B-_+3A_value">value</code></td>
<td>
<p>The new treatment values to set within object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Modifies the treatments slot of the input object in place.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
treatments(exper) &lt;- data.frame(new_treatment = rnorm(100))
</code></pre>

<hr>
<h2 id='treatments+26lt+3B-+2Cmediation_data-method'>Set the Treatments in a Mediation Data Object</h2><span id='topic+treatments+3C-+2Cmediation_data-method'></span>

<h3>Description</h3>

<p>This is an setter method for the treatments slot in a mediation data object.
It lets you supply a new treatments data.frame for the object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'mediation_data'
treatments(object) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="treatments+2B26lt+2B3B-+2B2Cmediation_data-method_+3A_object">object</code></td>
<td>
<p>An object of class mediation_data.</p>
</td></tr>
<tr><td><code id="treatments+2B26lt+2B3B-+2B2Cmediation_data-method_+3A_value">value</code></td>
<td>
<p>The new treatment values for the object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A version of <code>object</code> whose treatment slot has been replaced.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exper &lt;- demo_joy() |&gt;
    mediation_data("PHQ", "treatment", starts_with("ASV"))
treatments(exper) &lt;- data.frame(t = rep(0, 10))
exper
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
