<!DOCTYPE html><html><head><title>Help for package MultBiplotR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MultBiplotR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AddBinVars2Biplot'>
<p>Add suplementary binary variables to a biplot</p></a></li>
<li><a href='#AddCluster2Biplot'>
<p>Add clusters to a biplot object</p></a></li>
<li><a href='#AddContVars2Biplot'>
<p>Adds supplementary continuous variables to a biplot object</p></a></li>
<li><a href='#AddOrdVars2Biplot'>
<p>Adds supplementary ordinal variables to an existing biplot objects.</p></a></li>
<li><a href='#AddSupVars2Biplot'>
<p>Adds supplementary variables to a biplot object</p></a></li>
<li><a href='#anova.RidgeBinaryLogistic'>
<p>Compares two binary logistic models</p></a></li>
<li><a href='#Bartlett.Tests'>
<p>Bartlett tests</p></a></li>
<li><a href='#BasicDescription'>
<p>Basic descriptive sataistics</p></a></li>
<li><a href='#BinaryDistances'><p>Binary Distances</p></a></li>
<li><a href='#BinaryLogBiplotEM'>
<p>Binary logistic biplot with the EM algorithm.</p></a></li>
<li><a href='#BinaryLogBiplotGD'>
<p>Binary Logistic Biplot with Gradient Descent Estimation</p></a></li>
<li><a href='#BinaryLogBiplotGDRecursive'>
<p>Binary Logistic Biplot with Recursive Gradient Descent Estimation</p></a></li>
<li><a href='#BinaryLogBiplotJoint'>
<p>Binary logistic biplot with a gradient descent algorithm.</p></a></li>
<li><a href='#BinaryLogBiplotMirt'>
<p>Binary logistic biplot with Item Response Theory.</p></a></li>
<li><a href='#BinaryLogisticBiplot'>
<p>Binary Logistic Biplot</p></a></li>
<li><a href='#BinaryPLSFit'>
<p>Binary PLS Regression.</p></a></li>
<li><a href='#BinaryPLSR'>
<p>Partial Least Squares Regression with Binary Data</p></a></li>
<li><a href='#BinaryProximities'><p>Proximity Measures for Binary Data</p></a></li>
<li><a href='#Biplot.BinaryPLSR'>
<p>Biplot for a PLSR model with binary data</p></a></li>
<li><a href='#Biplot.PLSR'>
<p>Partial Least Squares Biplot</p></a></li>
<li><a href='#Biplot.PLSR1BIN'>
<p>Biplot for a PLSR model with a binary response</p></a></li>
<li><a href='#Biplot.PLSRBIN'>
<p>Biplot for a PLSR model with binary responses</p></a></li>
<li><a href='#BiplotFPCA'>
<p>External Biplot for functional data from a functional PCA object.</p></a></li>
<li><a href='#BootstrapDistance'>
<p>Bootstrap on the distance matrices used for Principal Coordinates Analysis (PCoA)</p></a></li>
<li><a href='#BootstrapScalar'>
<p>Bootstrap on the  scalar product matrices used for Principal Coordinates Analysis (PCoA)</p></a></li>
<li><a href='#BootstrapSmacof'>
<p>Bootstrap on the distance matrices used for MDS with Smacof</p></a></li>
<li><a href='#BoxPlotPanel'>
<p>Panel of box plots</p></a></li>
<li><a href='#CA'>
<p>Correspondence Analysis</p></a></li>
<li><a href='#Canonical.Variate.Analysis'>
<p>Biplot representation of a Canonical Variate Analysis or a Manova (Canonical-Biplot or MANOVA-Biplot)</p></a></li>
<li><a href='#CanonicalBiplot'>
<p>Biplot representation of a Canonical Variate Analysis or a Manova (Canonical-Biplot or MANOVA-Biplot)</p></a></li>
<li><a href='#CanonicalDistanceAnalysis'>
<p>MANOVA and Canonical Analysis of Distances</p></a></li>
<li><a href='#CanonicalStatisBiplot'>
<p>CANONICAL STATIS-ACT for multiple tables with common rows and its associated Biplot</p></a></li>
<li><a href='#CategoricalDistances'>
<p>Distances among individuals using nominal variables.</p></a></li>
<li><a href='#CategoricalProximities'>
<p>Proximities among individuals using nominal variables.</p></a></li>
<li><a href='#CCA'>
<p>Canonical Correspondence Analysis</p></a></li>
<li><a href='#CheckBinaryMatrix'><p>Checks if a data matrix is binary</p></a></li>
<li><a href='#CheckBinaryVector'>
<p>Checks if a vector is binary</p></a></li>
<li><a href='#Chemical'>
<p>Chemical data</p></a></li>
<li><a href='#Circle'>
<p>Draws a circle</p></a></li>
<li><a href='#Coinertia'><p>Coinertia Analysis.</p></a></li>
<li><a href='#ColContributionPlot'>
<p>Plots the contributios of a biplot</p></a></li>
<li><a href='#ConcEllipse'>
<p>Concentration ellipse for a se of two-dimensional points</p></a></li>
<li><a href='#ConfidenceInterval'>
<p>Confidence Interval for the mean</p></a></li>
<li><a href='#ConstrainedLogisticBiplot'>
<p>Constrained Binary Logistic Biplot</p></a></li>
<li><a href='#ConstrainedOrdinalLogisticBiplot'>
<p>Constrained Ordinal Logistic Biplot</p></a></li>
<li><a href='#ContinuousDistances'><p>Distances for Continuous Data</p></a></li>
<li><a href='#ContinuousProximities'>
<p>Proximities for Continuous Data</p></a></li>
<li><a href='#Convert2ThreeWay'>
<p>Three way array from a two way matrix</p></a></li>
<li><a href='#Convert3wArray2List'>
<p>Converts a three way array into a list</p></a></li>
<li><a href='#ConvertFactors2Integers'>
<p>Convert a factor to integer numbers</p></a></li>
<li><a href='#ConvertList23wArray'>
<p>Converts a list of matrices into a three way array</p></a></li>
<li><a href='#CorrelationCircle'>
<p>Circle of correlations</p></a></li>
<li><a href='#CrissCross'><p>Alternated Least Squares Biplot</p></a></li>
<li><a href='#CumSum'>
<p>Cummulative sums</p></a></li>
<li><a href='#Dataframe2BinaryMatrix'><p>Converts a Data Frame into a Binary Data Matrix</p></a></li>
<li><a href='#DataFrame2Matrix4Regression'>
<p>Prepares a matrix for regression from a data frame</p></a></li>
<li><a href='#DensityBiplot'>
<p>Adds Non-parametric densities to a biplot. Separated densities are calculated for different clusters</p></a></li>
<li><a href='#Dhats'>
<p>Calculation of Disparities</p></a></li>
<li><a href='#diagonal'>
<p>Diagonal matrix from a vector</p></a></li>
<li><a href='#DimensionLabels'>
<p>Labels for the selected dimensions in a biplot</p></a></li>
<li><a href='#dlines'>
<p>Connects two sets of points by lines</p></a></li>
<li><a href='#Doctors'>
<p>Data set extracted from the Careers of doctorate holders survey carried out by Spanish Statistical Office in 2008.</p></a></li>
<li><a href='#ErrorBarPlotPanel'>
<p>Plots a panel of error bars</p></a></li>
<li><a href='#EuclideanDistance'>
<p>Classical Euclidean Distance (Pythagorean Distance)</p></a></li>
<li><a href='#ExpandTable'>
<p>Expands a compressed table of patterns and frequencies</p></a></li>
<li><a href='#ExternalBinaryLogisticBiplot'>
<p>External Logistic Biplot for binary Data</p></a></li>
<li><a href='#ExtractTable'>
<p>Extracts unique patterns and its frequencies for a discrete data matrix (numeric)</p></a></li>
<li><a href='#FA.Biplot'>
<p>Biplot for Factor Analysis.</p></a></li>
<li><a href='#Fact2Bin'>
<p>Converts a Factor into its indicator matrix</p></a></li>
<li><a href='#Fraction'>
<p>Selection of a fraction of the data</p></a></li>
<li><a href='#Games_Howell'>
<p>Games-Howell post-hoc tests for Welch's one-way analysis</p></a></li>
<li><a href='#GD.Biplot'>
<p>Biplot for continuous data based on gradient descent methods</p></a></li>
<li><a href='#GeneralizedProcrustes'><p>Generalized Procrustes Analysis</p></a></li>
<li><a href='#GetBiplotScales'>
<p>Calculates the scales for the variables on a linear biplot</p></a></li>
<li><a href='#GetCCAScales'>
<p>Calculates scales for plotting the environmental variables in a Canonical Correspondence Analysis</p></a></li>
<li><a href='#ginv'>
<p>G inverse</p></a></li>
<li><a href='#GowerProximities'><p>Gower Dissimilarities for mixed types of data</p></a></li>
<li><a href='#GowerSimilarities'><p>Gower Dissimilarities for mixed types of data</p></a></li>
<li><a href='#Hermquad'>
<p>Gauss-Hermite quadrature</p></a></li>
<li><a href='#HistogramPanel'>
<p>Panel of histograms</p></a></li>
<li><a href='#HJ.Biplot'>
<p>HJ Biplot with added features.</p></a></li>
<li><a href='#InBox'>
<p>Checks if a point is inside a box.</p></a></li>
<li><a href='#InitialTransform'>
<p>Initial transformation of data</p></a></li>
<li><a href='#Integer2Binary'><p>Transforms an Integer Variable into a Binary Variable</p></a></li>
<li><a href='#Kruskal.Wallis.Tests'>
<p>Kruskal Wallis Tests</p></a></li>
<li><a href='#Levene.Tests'>
<p>Levene Tests</p></a></li>
<li><a href='#LogFrequencyBiplot'>
<p>Weighted Biplot for a table of frequencies</p></a></li>
<li><a href='#logit'>
<p>Logit function</p></a></li>
<li><a href='#Matrix2Proximities'>
<p>Matrix to Proximities</p></a></li>
<li><a href='#matrixsqrt'>
<p>Matrix squared root</p></a></li>
<li><a href='#matrixsqrtinv'>
<p>Inverse of the Matrix squared root</p></a></li>
<li><a href='#MDS'>
<p>Multidimensional Scaling</p></a></li>
<li><a href='#MGC'>
<p>Mixture Gaussian Clustering</p></a></li>
<li><a href='#MonotoneRegression'>
<p>Weighted Isotonic Regression (Weighted Monotone Regression)</p></a></li>
<li><a href='#moth'>
<p>Moth data</p></a></li>
<li><a href='#MultBiplotR-package'>
<p>Multivariate Analysis using Biplots</p></a></li>
<li><a href='#Multiquad'>
<p>Multidimensional Gauss-Hermite quadrature</p></a></li>
<li><a href='#MultiTableStatistics'>
<p>Statistics for multiple tables</p></a></li>
<li><a href='#MultiTableTransform'>
<p>Initial Transformation of a multi table object</p></a></li>
<li><a href='#NiceNumber'>
<p>Nice numbers: simple decimal numbers</p></a></li>
<li><a href='#NIPALS.Biplot'>
<p>Biplot using the NIPALS algorithm</p></a></li>
<li><a href='#NIPALSPCA'>
<p>NIPALS algorithm for PCA</p></a></li>
<li><a href='#NominalDistances'>
<p>Distances among individuals with nominal variables</p></a></li>
<li><a href='#NormalityTests'>
<p>Normality tests</p></a></li>
<li><a href='#Numeric2Binary'><p>Converts a numeric variable into a binary one</p></a></li>
<li><a href='#ones'>
<p>Matrix of ones</p></a></li>
<li><a href='#OrdinalLogisticFit'>
<p>Fits an ordinal logistic regression with ridge penalization</p></a></li>
<li><a href='#OrdLogBipEM'>
<p>Alternated EM algorithm for Ordinal Logistic Biplots</p></a></li>
<li><a href='#OrdVarBiplot'>
<p>Plots an ordinal variable on the biplot</p></a></li>
<li><a href='#OrdVarCoordinates'>
<p>Coordinates of an ordinal variable on the biplot.</p></a></li>
<li><a href='#OrthogonalizeScores'>
<p>Orthogonalize a set of Scores calculated by other procedure</p></a></li>
<li><a href='#PCA.Analysis'>
<p>Classical PCA Biplot with added features.</p></a></li>
<li><a href='#PCA.Biplot'>
<p>Classical PCA Biplot with added features.</p></a></li>
<li><a href='#PCA.Bootstrap'>
<p>Principal Components Analysis with bootstrap confidence intervals.</p></a></li>
<li><a href='#plot.Binary.Logistic.Biplot'>
<p>Plots the results of a Binary Logistic  Biplot</p></a></li>
<li><a href='#plot.CA.sol'>
<p>Plot the solution of a Coorespondence Analysis</p></a></li>
<li><a href='#plot.Canonical.Biplot'>
<p>Plots a Canonical Biplot</p></a></li>
<li><a href='#plot.CanonicalDistanceAnalysis'>
<p>Plots a Canonical Distance Analysis</p></a></li>
<li><a href='#plot.CCA.sol'>
<p>Plots the solution of a Canonical Correspondence Analysisis</p></a></li>
<li><a href='#plot.ContinuousBiplot'>
<p>Plots a biplot for continuous data.</p></a></li>
<li><a href='#plot.CVA'>
<p>Plot of a Canonical Variate Analysis</p></a></li>
<li><a href='#plot.ellipse'>
<p>Plot a concentration ellipse.</p></a></li>
<li><a href='#plot.External.Binary.Logistic.Biplot'>
<p>Plots an External Logistic Biplot for binary data</p></a></li>
<li><a href='#plot.fraction'>
<p>Plots a fraction of the data as a cluster</p></a></li>
<li><a href='#plot.MGC'>
<p>Plot the results of Model-Based Gaussian Clustering algorithms</p></a></li>
<li><a href='#plot.Ordinal.Logistic.Biplot'>
<p>Plots an ordinal Logistic Biplot</p></a></li>
<li><a href='#plot.PCA.Analysis'>
<p>Plots a Principal Component Analysis</p></a></li>
<li><a href='#plot.PCA.Bootstrap'>
<p>Plots the Bootstrap information for Principal Components Analysis (PCA)</p></a></li>
<li><a href='#plot.PCoABootstrap'>
<p>Plots an object of class PCoABootstrap</p></a></li>
<li><a href='#plot.Principal.Coordinates'><p>Plots an object of class Principal.Coordinates</p></a></li>
<li><a href='#plot.Procrustes'><p>Plots an object of class &quot;Procrustes&quot;</p></a></li>
<li><a href='#plot.StatisBiplot'>
<p>Plots a Statis Biplot Object</p></a></li>
<li><a href='#plot.TetraDualStatis'>
<p>Plots an object of class &quot;tetraDualStatis&quot;.</p></a></li>
<li><a href='#plot.Unfolding'>
<p>Plots an Unfolding Representation</p></a></li>
<li><a href='#PlotBiplotClusters'>
<p>Plot clusters on a biplot.</p></a></li>
<li><a href='#PlotOrdinalResponses'>
<p>Plot the response functions along the directions of best fit.</p></a></li>
<li><a href='#PLSR'>
<p>Partial Least Squares Regression</p></a></li>
<li><a href='#PLSR1Bin'>
<p>Partial Least Squares Regression with Binary Response</p></a></li>
<li><a href='#PLSRBin'>
<p>Partial Least Squares Regression with several Binary Responses</p></a></li>
<li><a href='#PLSRBinFit'>
<p>PLS binary regression.</p></a></li>
<li><a href='#PLSRfit'>
<p>Partial Least Squares Regression (PLSR)</p></a></li>
<li><a href='#PoliticalFigures'>
<p>Political Figures in the USA</p></a></li>
<li><a href='#PolyOrdinalLogBiplot'>
<p>Factor Analysis Biplot based on polychoric correlations</p></a></li>
<li><a href='#PrettyTicks'>
<p>Calculates loose axis ticks and labels using nice numbers</p></a></li>
<li><a href='#PrincipalCoordinates'><p>Principal Coordinates Analysis</p></a></li>
<li><a href='#print.MGC'>
<p>Prints the results of Model-Based Gaussian Clustering algorithms</p></a></li>
<li><a href='#print.RidgeBinaryLogistic'>
<p>prints an object of class RidgeBinaryLogistic</p></a></li>
<li><a href='#Protein'>
<p>Protein consumption data.</p></a></li>
<li><a href='#RAPD'>
<p>Sugar Cane Data</p></a></li>
<li><a href='#RemoveRowsWithNaNs'>
<p>Remove rows that contains NaNs (missing data)</p></a></li>
<li><a href='#riano'>
<p>Ecological data from Riano (Spain)</p></a></li>
<li><a href='#RidgeBinaryLogistic'>
<p>Ridge Binary Logistic Regression for Binary data</p></a></li>
<li><a href='#RidgeBinaryLogisticFit'>
<p>Fits a binary logistic regression with ridge penalization</p></a></li>
<li><a href='#RidgeMultinomialLogisticFit'>
<p>Multinomial logistic regression with ridge penalization</p></a></li>
<li><a href='#RidgeMultinomialLogisticRegression'>
<p>Ridge Multinomial Logistic Regression</p></a></li>
<li><a href='#RidgeOrdinalLogistic'>
<p>Ordinal logistic regression with ridge penalization</p></a></li>
<li><a href='#scores.CCA.sol'>
<p>Extract the scores of a CCA solution object</p></a></li>
<li><a href='#SeparateVarTypes'>
<p>Separation of different types of variables into a list</p></a></li>
<li><a href='#SimpleProcrustes'><p>Simple Procrustes Analysis</p></a></li>
<li><a href='#SMACOF'>
<p>SMACOF</p></a></li>
<li><a href='#smoking'>
<p>Smoking habits</p></a></li>
<li><a href='#Sparse.NIPALSPCA'>
<p>Sparse version of the NIPALS algorithm for PCA.</p></a></li>
<li><a href='#spiders'>
<p>Hunting Spiders Data</p></a></li>
<li><a href='#SpidersEnv'>
<p>Hunting spiders environmental data.</p></a></li>
<li><a href='#SpidersSp'>
<p>Hunting Spiders Data</p></a></li>
<li><a href='#SSI'>
<p>Sustainability Society Index</p></a></li>
<li><a href='#SSI3w'>
<p>Sustainability Society Index (3w)</p></a></li>
<li><a href='#SSIEcon3w'>
<p>Sustainability Society Index</p></a></li>
<li><a href='#SSIEnvir3w'>
<p>Sustainability Society Index</p></a></li>
<li><a href='#SSIHuman3w'>
<p>Sustainability Society Index</p></a></li>
<li><a href='#StatisBiplot'>
<p>STATIS-ACT for multiple tables with common rows and its associated Biplot</p></a></li>
<li><a href='#summary.Canonical.Biplot'>
<p>Summary of the solution of a Canonical Biplot Analysis</p></a></li>
<li><a href='#summary.CCA.sol'>
<p>Summary of the solution of a CCA</p></a></li>
<li><a href='#summary.ContinuousBiplot'>
<p>Summary of the solution of a Biplot for Continuous Data</p></a></li>
<li><a href='#summary.CVA'>
<p>Summary of a Canonical Variate Analysis</p></a></li>
<li><a href='#summary.MGC'>
<p>Summary of Model-Based Gaussian Clustering results</p></a></li>
<li><a href='#summary.PCA.Analysis'>
<p>Summary of the results of a PCA.</p></a></li>
<li><a href='#summary.PCA.Bootstrap'>
<p>Summary of a PCA.Bootstrap object</p></a></li>
<li><a href='#summary.PLSR'>
<p>Summary of a PLSR object</p></a></li>
<li><a href='#summary.PLSR1Bin'>
<p>Summary of PLSR with a Binary Response</p></a></li>
<li><a href='#summary.Principal.Coordinates'><p>Summary of the results of a Principal Coordinates Analysis</p></a></li>
<li><a href='#summary.RidgeBinaryLogistic'>
<p>Summary of a Binary Logistic Regression with Ridge Penalization</p></a></li>
<li><a href='#summary.TetraDualStatis'>
<p>Summary of the results of TetraDualStatis</p></a></li>
<li><a href='#t3pcovr'>
<p>Tucker 3 Principal Covariates Regression</p></a></li>
<li><a href='#TetraDualStatis'>
<p>Dual STATIS-ACT for binary data based on Tetrachoric Correlations</p></a></li>
<li><a href='#textsmart'><p>Labels of a Scatter</p></a></li>
<li><a href='#Three2TwoWay'>
<p>Converts a multitable list to a two way matrix</p></a></li>
<li><a href='#ThreeWay2FrontalSlices'>
<p>Three to two way data</p></a></li>
<li><a href='#TransformIni'>
<p>Initial transformation of a data matrix</p></a></li>
<li><a href='#Truncated.NIPALSPCA'>
<p>Truncated version of the NIPALS algorithm for PCA.</p></a></li>
<li><a href='#Unfolding'>
<p>Multidimensional Unfolding</p></a></li>
<li><a href='#VarBiplot'>
<p>Draws a variable on a biplot</p></a></li>
<li><a href='#wa'>
<p>Extracts the weighted averages of a CCA solution</p></a></li>
<li><a href='#wcor'>
<p>Weighted correlations</p></a></li>
<li><a href='#weighted.quantile'>
<p>Weighted quantiles</p></a></li>
<li><a href='#WeightedPCoA'><p>Weighted Principal Coordinates Analysis</p></a></li>
<li><a href='#wine'>
<p>Wine data</p></a></li>
<li><a href='#zeros'>
<p>Matrix of zeros as in Matlab</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multivariate Analysis Using Biplots in R</td>
</tr>
<tr>
<td>Version:</td>
<td>23.11.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-16</td>
</tr>
<tr>
<td>Author:</td>
<td>Jose Luis Vicente-Villardon, Laura Vicente-Gonzalez, Elisa Frutos-Bernal</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jose Luis Vicente Villardon &lt;villardon@usal.es&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Several multivariate techniques from a biplot perspective. It is the translation (with many improvements) into R of the previous package developed in 'Matlab'. The package contains some of the main developments of my team during the last 30 years together with some more standard techniques. Package includes: Classical Biplots, HJ-Biplot, Canonical Biplots, MANOVA Biplots, Correspondence Analysis, Canonical Correspondence Analysis, Canonical STATIS-ACT, Logistic Biplots for binary and ordinal data, Multidimensional Unfolding, External Biplots for Principal Coordinates Analysis or Multidimensional Scaling, among many others. References can be found in the help of each procedure.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, scales, geometry, deldir, mirt, GPArotation, Hmisc, car,
dunn.test, gplots, lattice, polycor, dae, xtable, mvtnorm,
psych, ThreeWay, knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Archs:</td>
<td>i386, x64</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-20 11:36:24 UTC; joseluis</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-21 15:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='AddBinVars2Biplot'>
Add suplementary binary variables to a biplot
</h2><span id='topic+AddBinVars2Biplot'></span>

<h3>Description</h3>

<p>Add suplementary binary variables to a biplot of any kind
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AddBinVars2Biplot(bip, Y, IncludeConst = TRUE, penalization = 0.2, 
freq = NULL, tolerance = 1e-05, maxiter = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AddBinVars2Biplot_+3A_bip">bip</code></td>
<td>

<p>A biplot object
</p>
</td></tr>
<tr><td><code id="AddBinVars2Biplot_+3A_y">Y</code></td>
<td>

<p>Matrix of binary variables to add
</p>
</td></tr>
<tr><td><code id="AddBinVars2Biplot_+3A_includeconst">IncludeConst</code></td>
<td>

<p>Should include a constant in the fit
</p>
</td></tr>
<tr><td><code id="AddBinVars2Biplot_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the fit
</p>
</td></tr>
<tr><td><code id="AddBinVars2Biplot_+3A_freq">freq</code></td>
<td>

<p>frequencies for each row of Y. By default is 1.
</p>
</td></tr>
<tr><td><code id="AddBinVars2Biplot_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for the fit
</p>
</td></tr>
<tr><td><code id="AddBinVars2Biplot_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fits binary variables to an existing biplot using penalized logistic regression.
</p>


<h3>Value</h3>

<p>The biplot object with supplementary binary variables added.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardón, J. L., &amp; Hernández-Sánchez, J. C. (2020). External Logistic Biplots for Mixed Types of Data. In Advanced Studies in Classification and Data Science (pp. 169-183). Springer, Singapore.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## No examples yet
</code></pre>

<hr>
<h2 id='AddCluster2Biplot'>
Add clusters to a biplot object
</h2><span id='topic+AddCluster2Biplot'></span>

<h3>Description</h3>

<p>The function add clusters to a biplot object to be represented on the biplot. The clusters can be defined by a nominal variable provided by the user, obtained from the <code>hclust</code> function of the base package or from the <code>kmeans</code> function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AddCluster2Biplot(Bip, NGroups=3, ClusterType="hi", Groups=NULL, 
                  Original=FALSE, ClusterColors=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AddCluster2Biplot_+3A_bip">Bip</code></td>
<td>

<p>A Biplot object obtained from any biplot procedure. It has to be a list containing a field called <code>Bip$RowCoordinates</code> in order to calculate the clusters when necessary. 
</p>
</td></tr>
<tr><td><code id="AddCluster2Biplot_+3A_ngroups">NGroups</code></td>
<td>

<p>Number of groups or clusters. Only necessary when hierarchical or k-means procedures are used.
</p>
</td></tr>
<tr><td><code id="AddCluster2Biplot_+3A_clustertype">ClusterType</code></td>
<td>

<p>The type of cluster to add. There are three possibilities &quot;us&quot; (User Defined), &quot;hi&quot; (hierarchical clusters),  &quot;km&quot; (kmeans clustering) or &quot;gm&quot; (gaussian mixture).
</p>
</td></tr>
<tr><td><code id="AddCluster2Biplot_+3A_groups">Groups</code></td>
<td>

<p>A factor defining the groups provided by the user.
</p>
</td></tr>
<tr><td><code id="AddCluster2Biplot_+3A_original">Original</code></td>
<td>

<p>Should the clusters be calculated using the original data rather than the reduced dimensions?.
</p>
</td></tr>
<tr><td><code id="AddCluster2Biplot_+3A_clustercolors">ClusterColors</code></td>
<td>

<p>Colors for the clusters.
</p>
</td></tr>
<tr><td><code id="AddCluster2Biplot_+3A_...">...</code></td>
<td>

<p>Any other parameter for the <code>hclust</code> and <code>kmeans</code> procedures.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>One of the main shortcomings of cluster analysis is that it is not easy to search for the variables associated to the obtained classification; representing the clusters on the biplot can help to perform that interpretation. If you consider the technique for dimension reduction as a way to separate the signal from the noise, clusters should be constructed using the dimensions retained in the biplot, otherwise the complete original data matrix can be used. The colors used by each cluster should match the color used in the Dendrogram.
User defined clusters can also be plotted, for example, to investigate the relation of the biplot solution to an external nominal variable.
</p>


<h3>Value</h3>

<p>The function returns the biplot object with the information about the clusters added in new fields
</p>
<table>
<tr><td><code>ClusterType</code></td>
<td>
<p>The method of clustering as defined in the argument <code>ClusterType</code>.</p>
</td></tr>
<tr><td><code>Clusters</code></td>
<td>
<p>A factor containing the solution or the user defined clusters</p>
</td></tr>
<tr><td><code>ClusterNames</code></td>
<td>
<p>The names of the clusters</p>
</td></tr>
<tr><td><code>ClusterColors</code></td>
<td>
<p>The colors of the clusters</p>
</td></tr>
<tr><td><code>Dendrogram</code></td>
<td>
<p>The Dendrogram if we have used hirarchical clustering</p>
</td></tr>
<tr><td><code>ClusterObject</code></td>
<td>
<p>The object obtained from <code>hclust</code>, <code>kmeans</code> or <code>MGC</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Demey, J. R., Vicente-Villardon, J. L., Galindo-Villardon, M. P., &amp; Zambrano, A. Y. (2008). Identifying molecular markers associated with classification of genotypes by External Logistic Biplots. Bioinformatics, 24(24), 2832-2838.
</p>
<p>Gallego-Alvarez, I., &amp; Vicente-Villardon, J. L. (2012). Analysis of environmental indicators in international companies by applying the logistic biplot. Ecological Indicators, 23, 250-261.
</p>
<p>Galindo, P. V., Vaz, T. D. N., &amp; Nijkamp, P. (2011). Institutional capacity to dynamically innovate: an application to the Portuguese case. Technological Forecasting and Social Change, 78(1), 3-12.
</p>
<p>Vazquez-de-Aldana, B. R., Garcia-Criado, B., Vicente-Tavera, S., &amp; Zabalgogeazcoa, I. (2013). Fungal Endophyte (Epichloë festucae) Alters the Nutrient Content of Festuca rubra Regardless of Water Availability. PloS one, 8(12), e84539.
</p>


<h3>See Also</h3>

<p>For clusters not provided by the user the function uses the standard procedures in  <code>hclust</code> and <code>kmeans</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Protein)
bip=PCA.Biplot(Protein[,3:11])
plot(bip)
# Add user defined clusters containing the region (North, South, Center)
bip=AddCluster2Biplot(bip, ClusterType="us", Groups=Protein$Region)
plot(bip, mode="a", margin=0.1, PlotClus=TRUE)

# Hierarchical clustering on the biplot coordinates using the Ward method
bip=AddCluster2Biplot(bip, ClusterType="hi", method="ward.D")
op &lt;- par(mfrow=c(1,2))
plot(bip, mode="s", margin=0.1, PlotClus=TRUE)
plot(bip$Dendrogram)
par(op)
# K-means cluster on the biplot coordinates using the Ward method
bip=AddCluster2Biplot(bip, ClusterType="hi", method="ward.D")
op &lt;- par(mfrow=c(1,2))
plot(bip, mode="s", margin=0.1, PlotClus=TRUE)
plot(bip$Dendrogram)
par(op)



</code></pre>

<hr>
<h2 id='AddContVars2Biplot'>
Adds supplementary continuous variables to a biplot object
</h2><span id='topic+AddContVars2Biplot'></span>

<h3>Description</h3>

<p>Adds supplementary continuous variables to a biplot object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AddContVars2Biplot(bip, X, dims = NULL, Scaling = 5, Fit = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AddContVars2Biplot_+3A_bip">bip</code></td>
<td>

<p>A biplot object
</p>
</td></tr>
<tr><td><code id="AddContVars2Biplot_+3A_x">X</code></td>
<td>

<p>Matrix containing the supplementary continuos variables
</p>
</td></tr>
<tr><td><code id="AddContVars2Biplot_+3A_dims">dims</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="AddContVars2Biplot_+3A_scaling">Scaling</code></td>
<td>

<p>Transformation to apply to X
</p>
</td></tr>
<tr><td><code id="AddContVars2Biplot_+3A_fit">Fit</code></td>
<td>

<p>Type of fit. Linear by default.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>More types of fit will be added in the future
</p>


<h3>Value</h3>

<p>A biplot object with the  coordinates for the supplementary variables added.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AddSupVars2Biplot">AddSupVars2Biplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet

</code></pre>

<hr>
<h2 id='AddOrdVars2Biplot'>
Adds supplementary ordinal variables to an existing biplot objects.
</h2><span id='topic+AddOrdVars2Biplot'></span>

<h3>Description</h3>

<p>Adds supplementary ordinal variables to an existing biplot objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AddOrdVars2Biplot(bip, Y, tol = 1e-06, maxiterlogist = 100, 
penalization = 0.2, showiter = TRUE, show = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AddOrdVars2Biplot_+3A_bip">bip</code></td>
<td>

<p>A biplot object.
</p>
</td></tr>
<tr><td><code id="AddOrdVars2Biplot_+3A_y">Y</code></td>
<td>

<p>A matrix of ordinal variables.
</p>
</td></tr>
<tr><td><code id="AddOrdVars2Biplot_+3A_tol">tol</code></td>
<td>

<p>Tolerance.
</p>
</td></tr>
<tr><td><code id="AddOrdVars2Biplot_+3A_maxiterlogist">maxiterlogist</code></td>
<td>

<p>Maximum number of iterations for the logistic fit.
</p>
</td></tr>
<tr><td><code id="AddOrdVars2Biplot_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the logistic fit
</p>
</td></tr>
<tr><td><code id="AddOrdVars2Biplot_+3A_showiter">showiter</code></td>
<td>

<p>Should the itrations be shown on screen
</p>
</td></tr>
<tr><td><code id="AddOrdVars2Biplot_+3A_show">show</code></td>
<td>

<p>Show details.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adds supplementary ordinal variables to an existing biplot objects.
</p>


<h3>Value</h3>

<p>An object with the information of the fits
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardon, J. L., &amp; Hernandez-Sanchez, J. C. (2020). External Logistic Biplots for Mixed Types of Data. In Advanced Studies in Classification and Data Science (pp. 169-183). Springer, Singapore.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># not yet
</code></pre>

<hr>
<h2 id='AddSupVars2Biplot'>
Adds supplementary variables to a biplot object
</h2><span id='topic+AddSupVars2Biplot'></span>

<h3>Description</h3>

<p>Adds supplementary bariables to a biplot object constructed with any of the biplot methods of the package. The new variables are fitted using the coordinates for the rows. Each variable is fitted using the adequate procedure for its type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AddSupVars2Biplot(bip, X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AddSupVars2Biplot_+3A_bip">bip</code></td>
<td>

<p>The biplot object
</p>
</td></tr>
<tr><td><code id="AddSupVars2Biplot_+3A_x">X</code></td>
<td>

<p>A data frame with the supplementary variables.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Binary, nominal or ordinal variables are fitted using logistic biplots. Continuous variables are fitted with linear regression.
</p>


<h3>Value</h3>

<p>A biplot object with the  coordinates for the supplementary variables added.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AddContVars2Biplot">AddContVars2Biplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet

</code></pre>

<hr>
<h2 id='anova.RidgeBinaryLogistic'>
Compares two binary logistic models
</h2><span id='topic+anova.RidgeBinaryLogistic'></span>

<h3>Description</h3>

<p>Anova for comparing two binary logistic models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RidgeBinaryLogistic'
anova(object, object2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anova.RidgeBinaryLogistic_+3A_object">object</code></td>
<td>

<p>The first model
</p>
</td></tr>
<tr><td><code id="anova.RidgeBinaryLogistic_+3A_object2">object2</code></td>
<td>

<p>The second model
</p>
</td></tr>
<tr><td><code id="anova.RidgeBinaryLogistic_+3A_...">...</code></td>
<td>

<p>Any additional arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Anova for comparing two binary logistic models
</p>


<h3>Value</h3>

<p>The comparison of the two models.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='Bartlett.Tests'>
Bartlett tests
</h2><span id='topic+Bartlett.Tests'></span>

<h3>Description</h3>

<p>Bartlett tests foor the columns of a matrix  and a grouping variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Bartlett.Tests(X, groups = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Bartlett.Tests_+3A_x">X</code></td>
<td>

<p>A data frame or a matrix containing several numerical variables
</p>
</td></tr>
<tr><td><code id="Bartlett.Tests_+3A_groups">groups</code></td>
<td>

<p>A factor with the groups
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Bartlett tests foor the columns of a matrix  and a grouping variable
</p>


<h3>Value</h3>

<p>A matrix with the tests for each column
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Bartlett, M. S. (1937). &quot;Properties of sufficiency and statistical tests&quot;. Proceedings of the Royal Statistical Society, Series A 160, 268-282 JSTOR 96803
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(wine)
Bartlett.Tests(wine[,4:8], groups = wine$Origin)
</code></pre>

<hr>
<h2 id='BasicDescription'>
Basic descriptive sataistics
</h2><span id='topic+BasicDescription'></span>

<h3>Description</h3>

<p>Basic descriptive sataistics of several variables by the categories of a factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BasicDescription(X, groups = NULL, SortByGroups = FALSE, na.rm = FALSE, Intervals = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BasicDescription_+3A_x">X</code></td>
<td>

<p>A data frame or a matrix containing several numerical variables
</p>
</td></tr>
<tr><td><code id="BasicDescription_+3A_groups">groups</code></td>
<td>

<p>A factor with the groupings
</p>
</td></tr>
<tr><td><code id="BasicDescription_+3A_sortbygroups">SortByGroups</code></td>
<td>

<p>Sorting by groups
</p>
</td></tr>
<tr><td><code id="BasicDescription_+3A_na.rm">na.rm</code></td>
<td>

<p>a logical value indicating whether NA values should be stripped before the computation proceeds.
</p>
</td></tr>
<tr><td><code id="BasicDescription_+3A_intervals">Intervals</code></td>
<td>

<p>Should the confidence intervals be calculated?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Basic descriptive sataistics of several variables by the categories of a factor.
</p>


<h3>Value</h3>

<p>A list with the description of each variable.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
BasicDescription(wine[,4:8], groups = wine$Origin)
</code></pre>

<hr>
<h2 id='BinaryDistances'>Binary Distances </h2><span id='topic+BinaryDistances'></span>

<h3>Description</h3>

<p>Calculates distances among rows of a binary data matrix or among the rows of two binary matrices. 
The end user will use BinaryProximities rather than this function.
Input must be a matrix with 0 or 1 values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryDistances(x, y = NULL, coefficient= "Simple_Matching", transformation="sqrt(1-S)")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BinaryDistances_+3A_x">x</code></td>
<td>

<p>Main binary data matrix. Distances among rows are calculated if y=NULL.
</p>
</td></tr>
<tr><td><code id="BinaryDistances_+3A_y">y</code></td>
<td>

<p>Second binary data matrix. If not NULL the distances among the rows of x and y are calculated 
</p>
</td></tr>
<tr><td><code id="BinaryDistances_+3A_coefficient">coefficient</code></td>
<td>

<p>Similarity coefficient. Use the name (see details)
</p>
</td></tr>
<tr><td><code id="BinaryDistances_+3A_transformation">transformation</code></td>
<td>

<p>Transformation of the similarities. Use the name (see details)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following coefficients are calculated
</p>
<p>1.- Kulezynski =  a/(b + c)
</p>
<p>2.- Russell_and_Rao =  a/(a + b + c+d)
</p>
<p>3.- Jaccard = a/(a + b + c)
</p>
<p>4.- Simple_Matching = (a + d)/(a + b + c + d)
</p>
<p>5.- Anderberg = a/(a + 2 * (b + c))
</p>
<p>6.- Rogers_and_Tanimoto = (a + d)/(a + 2 * (b + c) + d)
</p>
<p>7.- Sorensen_Dice_and_Czekanowski = a/(a + 0.5 * (b + c))
</p>
<p>8.- Sneath_and_Sokal = (a + d)/(a + 0.5 * (b + c) + d)
</p>
<p>9.- Hamman = (a - (b + c) + d)/(a + b + c + d)
</p>
<p>10.- Kulezynski = 0.5 * ((a/(a + b)) + (a/(a + c)))
</p>
<p>11.- Anderberg2 = 0.25 * (a/(a + b) + a/(a + c) + d/(c + d) + d/(b + d))
</p>
<p>12.- Ochiai = a/sqrt((a + b) * (a + c))
</p>
<p>13.- S13 = (a * d)/sqrt((a + b) * (a + c) * (d + b) * (d + c))
</p>
<p>14.- Pearson_phi = (a * d - b * c)/sqrt((a + b) * (a + c) * (d + b) * (d + c))
</p>
<p>15.- Yule = (a * d - b * c)/(a * d + b * c)
</p>
<p>The following transformations of the similarity3 are calculated
</p>
<p>1.- 'Identity'   dis=sim
</p>
<p>2.- '1-S'   dis=1-sim
</p>
<p>3.- 'sqrt(1-S)'  dis = sqrt(1 - sim)
</p>
<p>4.- '-log(s)'  dis=-1*log(sim)
</p>
<p>5.- '1/S-1' dis=1/sim -1 
</p>
<p>6.- 'sqrt(2(1-S))'  dis== sqrt(2*(1 - sim))
</p>
<p>7.- '1-(S+1)/2' dis=1-(sim+1)/2
</p>
<p>8.- '1-abs(S)'  dis=1-abs(sim)
</p>
<p>9.- '1/(S+1)'  dis=1/(sim)+1
</p>


<h3>Value</h3>

<p>An object of class <code>proximities</code>.This has components:
</p>
<table>
<tr><td><code>comp1</code></td>
<td>
<p>Description of 'comp1'</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Gower, J. C. (2006) Similarity dissimilarity and Distance, measures of. Encyclopedia of Statistical Sciences. 2nd. ed. Volume 12. Wiley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PrincipalCoordinates">PrincipalCoordinates</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)

</code></pre>

<hr>
<h2 id='BinaryLogBiplotEM'>
Binary logistic biplot with the EM algorithm.
</h2><span id='topic+BinaryLogBiplotEM'></span>

<h3>Description</h3>

<p>Binary logistic biplot with the EM algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryLogBiplotEM(x, freq = matrix(1, nrow(x), 1), aini = NULL,
dimens = 2, nnodos = 15, tol = 1e-04, maxiter = 100, penalization = 0.2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BinaryLogBiplotEM_+3A_x">x</code></td>
<td>

<p>A binary data matrix
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotEM_+3A_freq">freq</code></td>
<td>

<p>A vector of frequencies.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotEM_+3A_aini">aini</code></td>
<td>

<p>Initial values for the row coordinates.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotEM_+3A_dimens">dimens</code></td>
<td>

<p>Dimension of the solution.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotEM_+3A_nnodos">nnodos</code></td>
<td>

<p>Number of nodes for the gaussian quadrature
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotEM_+3A_tol">tol</code></td>
<td>

<p>Tolerance
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotEM_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotEM_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the fit (ridge)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Binary logistic biplot with the EM algorithm based on marginal maximum likelihood.
</p>


<h3>Value</h3>

<p>A logistic biplot object.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardón, J. L., Galindo-Villardón, M. P., &amp; Blázquez-Zaballos, A. (2006). Logistic biplots. Multiple correspondence analysis and related methods. London: Chapman &amp; Hall, 503-521.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='BinaryLogBiplotGD'>
Binary Logistic Biplot with Gradient Descent Estimation
</h2><span id='topic+BinaryLogBiplotGD'></span>

<h3>Description</h3>

<p>Binary Logistic Biplot with Gradient Descent Estimation. An external optimization function is used to calculate the parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryLogBiplotGD(X, freq = matrix(1, nrow(X), 1), dim = 2, tolerance =
                   1e-07, penalization = 0.01, num_max_iters = 100,
                   RotVarimax = FALSE, seed = 0, OptimMethod = "CG",
                   Initial = "random", Orthogonalize = FALSE, Algorithm =
                   "Joint", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BinaryLogBiplotGD_+3A_x">X</code></td>
<td>

<p>A binary data matrix
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_freq">freq</code></td>
<td>

<p>Frequencies of each row. When adequate.</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_dim">dim</code></td>
<td>

<p>Dimension of the final solution.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence of the algorithm.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_penalization">penalization</code></td>
<td>

<p>Ridge penalization constant.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_num_max_iters">num_max_iters</code></td>
<td>

<p>Maximum number of iterations of the algorithm.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_rotvarimax">RotVarimax</code></td>
<td>

<p>Should the final solution be rotated.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_seed">seed</code></td>
<td>

<p>Seed for the random numbers. Used for reproductibility.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_optimmethod">OptimMethod</code></td>
<td>

<p>Optimization method used by <code>optim</code>.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_initial">Initial</code></td>
<td>

<p>Initial configuration to start the iterations.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_orthogonalize">Orthogonalize</code></td>
<td>

<p>Should te solution be orthogonalized?.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_algorithm">Algorithm</code></td>
<td>

<p>Algorithm for esimation: Joint or alternated.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGD_+3A_...">...</code></td>
<td>

<p>Aditional parameters used by the optimization function.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fits a binary logistic biplot using gradient descent. The general function <code>optim</code> is used to optimize the loss function. Conjugate gradien is used as a default although other alternatives can be USED.
</p>


<h3>Value</h3>

<p>An object of class &quot;Binary.Logistic.Biplot&quot;.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardon, J. L., Galindo, M. P. and Blazquez, A. (2006) Logistic Biplots. In Multiple Correspondence Análisis And Related Methods. Grenacre, M &amp; Blasius, J, Eds,  Chapman and Hall, Boca Raton.
</p>
<p>Demey, J., Vicente-Villardon, J. L., Galindo, M.P.  AND Zambrano, A. (2008) Identifying Molecular Markers Associated With Classification Of Genotypes Using External Logistic Biplots. Bioinformatics, 24(24): 2832-2838.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
X=Dataframe2BinaryMatrix(spiders)

logbip=BinaryLogBiplotGD(X,penalization=0.1)
plot(logbip, Mode="a")
summary(logbip)
</code></pre>

<hr>
<h2 id='BinaryLogBiplotGDRecursive'>
Binary Logistic Biplot with Recursive Gradient Descent Estimation
</h2><span id='topic+BinaryLogBiplotGDRecursive'></span>

<h3>Description</h3>

<p>Binary Logistic Biplot with Recursive Gradient Descent Estimation. An external optimization function is used to calculate the parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryLogBiplotGDRecursive(X, freq = matrix(1, nrow(X), 1), dim = 2, tolerance = 1e-04, 
                          penalization = 0.2, num_max_iters = 100, 
                          RotVarimax = FALSE, OptimMethod = "CG", 
                          Initial = "random", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BinaryLogBiplotGDRecursive_+3A_x">X</code></td>
<td>

<p>A binary data matrix
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGDRecursive_+3A_freq">freq</code></td>
<td>

<p>Frequencies of each row. When adequate.</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGDRecursive_+3A_dim">dim</code></td>
<td>

<p>Dimension of the final solution.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGDRecursive_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence of the algorithm.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGDRecursive_+3A_penalization">penalization</code></td>
<td>

<p>Ridge penalization constant.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGDRecursive_+3A_num_max_iters">num_max_iters</code></td>
<td>

<p>Maximum number of iterations of the algorithm.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGDRecursive_+3A_rotvarimax">RotVarimax</code></td>
<td>

<p>Should the final solution be rotated.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGDRecursive_+3A_optimmethod">OptimMethod</code></td>
<td>

<p>Optimization method used by <code>optim</code>.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGDRecursive_+3A_initial">Initial</code></td>
<td>

<p>Initial configuration to start the iterations.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotGDRecursive_+3A_...">...</code></td>
<td>

<p>Aditional parameters used by the optimization function.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fits a binary logistic biplot using recursive gradient descent. The general function <code>optim</code> is used to optimize the loss function. Conjugate gradien is used as a default although other alternatives can be USED. It can be considered as a generalization of the NIPALS algorithm for a matrix of binary data.
</p>


<h3>Value</h3>

<p>An object of class &quot;Binary.Logistic.Biplot&quot;.
</p>


<h3>Author(s)</h3>

<p>José Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardon, J. L., Galindo, M. P. and Blazquez, A. (2006) Logistic Biplots. In Multiple Correspondence Análisis And Related Methods. Grenacre, M &amp; Blasius, J, Eds,  Chapman and Hall, Boca Raton.
</p>
<p>Demey, J., Vicente-Villardon, J. L., Galindo, M.P.  AND Zambrano, A. (2008) Identifying Molecular Markers Associated With Classification Of Genotypes Using External Logistic Biplots. Bioinformatics, 24(24): 2832-2838.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(spiders)
X=Dataframe2BinaryMatrix(spiders)
logbip=BinaryLogBiplotGDRecursive(X,penalization=0.1)
plot(logbip, Mode="a")
summary(logbip)
</code></pre>

<hr>
<h2 id='BinaryLogBiplotJoint'>
Binary logistic biplot with a gradient descent algorithm.
</h2><span id='topic+BinaryLogBiplotJoint'></span>

<h3>Description</h3>

<p>Binary logistic biplot with a gradient descent algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryLogBiplotJoint(x, freq = matrix(1, nrow(x), 1), dim = 2, 
ainit = NULL, tolerance = 1e-04, maxiter = 30, penalization = 0.2, 
maxcond = 7, RotVarimax = FALSE, lambda = 0.1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BinaryLogBiplotJoint_+3A_x">x</code></td>
<td>

<p>A binary data matrix
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotJoint_+3A_freq">freq</code></td>
<td>

<p>A vector of frequencies.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotJoint_+3A_dim">dim</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotJoint_+3A_ainit">ainit</code></td>
<td>

<p>Initial values for the row coordinates.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotJoint_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotJoint_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations.
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotJoint_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the fit (ridge)
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotJoint_+3A_maxcond">maxcond</code></td>
<td>

<p>Naximum condition number
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotJoint_+3A_rotvarimax">RotVarimax</code></td>
<td>

<p>Should a Varimax Rotation be used?
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotJoint_+3A_lambda">lambda</code></td>
<td>

<p>Penalization argument
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotJoint_+3A_...">...</code></td>
<td>

<p>Aditional arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Binary logistic biplot with a gradient descent algorithm. Estimates row and column parameters at the same time.
</p>


<h3>Value</h3>

<p>A logistic biplot object.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardón, J. L., Galindo-Villardón, M. P., &amp; Blázquez-Zaballos, A. (2006). Logistic biplots. Multiple correspondence analysis and related methods. London: Chapman &amp; Hall, 503-521.
</p>
<p>Vicente-Villardon, J. L., &amp; Vicente-Gonzalez, L. Redundancy Analysis for Binary Data Based on Logistic Responses in Data Analysis and Rationality in a Complex World. Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># not yet
</code></pre>

<hr>
<h2 id='BinaryLogBiplotMirt'>
Binary logistic biplot with Item Response Theory.
</h2><span id='topic+BinaryLogBiplotMirt'></span>

<h3>Description</h3>

<p>Binary logistic biplot with Item Response Theory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryLogBiplotMirt(x, dimens = 2, tolerance = 1e-04, 
maxiter = 30, penalization = 0.2, Rotation = "varimax", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BinaryLogBiplotMirt_+3A_x">x</code></td>
<td>

<p>The binary Data matrix
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotMirt_+3A_dimens">dimens</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotMirt_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance of the algorithm
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotMirt_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotMirt_+3A_penalization">penalization</code></td>
<td>

<p>Rige Penalization
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotMirt_+3A_rotation">Rotation</code></td>
<td>

<p>Should a rotation be applied?
</p>
</td></tr>
<tr><td><code id="BinaryLogBiplotMirt_+3A_...">...</code></td>
<td>

<p>Aditional argumaents.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Binary logistic biplot with Item Response Theory.
</p>


<h3>Value</h3>

<p>A logistic biplot object.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardón, J. L., Galindo-Villardón, M. P., &amp; Blázquez-Zaballos, A. (2006). Logistic biplots. Multiple correspondence analysis and related methods. London: Chapman &amp; Hall, 503-521.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='BinaryLogisticBiplot'>
Binary Logistic Biplot
</h2><span id='topic+BinaryLogisticBiplot'></span>

<h3>Description</h3>

<p>Fits a binary lo gistic biplot to a binary data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryLogisticBiplot(x, dim = 2, compress = FALSE, init = "mca", 
method = "EM", rotation = "none", tol = 1e-04, 
maxiter = 100, penalization = 0.2, similarity = "Simple_Matching", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BinaryLogisticBiplot_+3A_x">x</code></td>
<td>

<p>The binary data matrix
</p>
</td></tr>
<tr><td><code id="BinaryLogisticBiplot_+3A_dim">dim</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="BinaryLogisticBiplot_+3A_compress">compress</code></td>
<td>

<p>Compress the data before the fitting (not yet implemented)
</p>
</td></tr>
<tr><td><code id="BinaryLogisticBiplot_+3A_init">init</code></td>
<td>

<p>Type of initial configuration. (&quot;random&quot;, &quot;mirt&quot;, &quot;PCoA&quot;, &quot;mca&quot;)
</p>
</td></tr>
<tr><td><code id="BinaryLogisticBiplot_+3A_method">method</code></td>
<td>

<p>Method to fit the logistic biplot (&quot;EM&quot;, &quot;Joint&quot;, &quot;mirt&quot;, &quot;JointGD&quot;, &quot;AlternatedGD&quot;, &quot;External&quot;, &quot;Recursive&quot;)
</p>
</td></tr>
<tr><td><code id="BinaryLogisticBiplot_+3A_rotation">rotation</code></td>
<td>

<p>Rotation of the solution (&quot;none&quot;, &quot;oblimin&quot;, &quot;quartimin&quot;, &quot;oblimax&quot; ,&quot;entropy&quot;,  &quot;quartimax&quot;, &quot;varimax&quot;,  &quot;simplimax&quot; ) see GPARotation
</p>
</td></tr>
<tr><td><code id="BinaryLogisticBiplot_+3A_tol">tol</code></td>
<td>

<p>Tolerance for the algorithm
</p>
</td></tr>
<tr><td><code id="BinaryLogisticBiplot_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations.
</p>
</td></tr>
<tr><td><code id="BinaryLogisticBiplot_+3A_penalization">penalization</code></td>
<td>

<p>Panalization for the different algorithms
</p>
</td></tr>
<tr><td><code id="BinaryLogisticBiplot_+3A_similarity">similarity</code></td>
<td>

<p>Similarity coefficient for the initial configuration or the external model
</p>
</td></tr>
<tr><td><code id="BinaryLogisticBiplot_+3A_...">...</code></td>
<td>

<p>Any other argument for each particular method.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fits a binary lo gistic biplot to a binary data matrix.
</p>
<p>Different Initial configurations can be selected:
</p>
<p>1.- random : Random coordinates for each point.
</p>
<p>2.- mirt: scores of the procedure mirt (Multidimensional Item Response Theory)
</p>
<p>3.- PCoA: Principal Coordinates Analysis
</p>
<p>4.- mca: Multiple Correspondence Analysis
</p>
<p>We can use also different methods for the estimation
</p>
<p>1.- Joint: Joint estimation of the row and column parameters. The Initial alorithm.
</p>
<p>2.- EM: Marginal Maximum Likelihood
</p>
<p>3.- mirt: Similar to the previous but fitted using the package mirt.
</p>
<p>4.- JointGD: Joint estimation of the row and column methods using the gradient descent method.
</p>
<p>5.- AlternatedGD: Alternated estimation of the row and column methods using the gradient descent method.
</p>
<p>6.- External: Logistic fits on the Principal Coordinates Analysis.
</p>
<p>7.- Recursive: Recursive (one axis at a time) estimation of the row and column methods using the gradient descent method. This is similar to the NIPALS algorithm for PCA
</p>


<h3>Value</h3>

 
<p>A Logistic Biplot object.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardon, J. L., Galindo, M. P. and Blazquez, A. (2006) Logistic Biplots. In Multiple Correspondence Análisis And Related Methods. Grenacre, M &amp; Blasius, J, Eds,  Chapman and Hall, Boca Raton.
</p>
<p>Demey, J., Vicente-Villardon, J. L., Galindo, M.P.  AND Zambrano, A. (2008) Identifying Molecular Markers Associated With Classification Of Genotypes Using External Logistic Biplots. Bioinformatics, 24(24): 2832-2838.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BinaryLogBiplotJoint">BinaryLogBiplotJoint</a></code>, <code><a href="#topic+BinaryLogBiplotEM">BinaryLogBiplotEM</a></code>, <code><a href="#topic+BinaryLogBiplotGD">BinaryLogBiplotGD</a></code>, <code><a href="#topic+BinaryLogBiplotMirt">BinaryLogBiplotMirt</a></code>, 
</p>


<h3>Examples</h3>

<pre><code class='language-R'># data(spiders)
# X=Dataframe2BinaryMatrix(spiders)

# logbip=BinaryLogBiplotGD(X,penalization=0.1)
# plot(logbip, Mode="a")
# summary(logbip)

</code></pre>

<hr>
<h2 id='BinaryPLSFit'>
Binary PLS Regression.
</h2><span id='topic+BinaryPLSFit'></span>

<h3>Description</h3>

<p>Fits Binary PLS regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryPLSFit(Y, X, S = 2, tolerance = 5e-06, maxiter = 100, show = FALSE, 
          penalization = 0.1, OptimMethod = "CG", seed = 0) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BinaryPLSFit_+3A_y">Y</code></td>
<td>

<p>The response
</p>
</td></tr>
<tr><td><code id="BinaryPLSFit_+3A_x">X</code></td>
<td>

<p>The matrix of independent variables
</p>
</td></tr>
<tr><td><code id="BinaryPLSFit_+3A_s">S</code></td>
<td>

<p>The Dimension of the solution
</p>
</td></tr>
<tr><td><code id="BinaryPLSFit_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence of the algorithm
</p>
</td></tr>
<tr><td><code id="BinaryPLSFit_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum Number of iterations
</p>
</td></tr>
<tr><td><code id="BinaryPLSFit_+3A_show">show</code></td>
<td>

<p>Show the steps of the algorithm
</p>
</td></tr>
<tr><td><code id="BinaryPLSFit_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the Ridge Logistic Regression
</p>
</td></tr>
<tr><td><code id="BinaryPLSFit_+3A_optimmethod">OptimMethod</code></td>
<td>

<p>Optimization methods from optimr
</p>
</td></tr>
<tr><td><code id="BinaryPLSFit_+3A_seed">seed</code></td>
<td>

<p>Seed. By default is 0.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fits Binary PLS Regression. It is used for a higher level function.
</p>


<h3>Value</h3>

<p>The PLS fit used by the BinaryPLSR function.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Ugarte Fajardo, J., Bayona Andrade, O., Criollo Bonilla, R., Cevallos‐Cevallos, J., Mariduena‐Zavala, M., Ochoa Donoso, D., &amp; Vicente Villardon, J. L. (2020). Early detection of black Sigatoka in banana leaves using hyperspectral images. Applications in plant sciences, 8(8), e11383.
</p>
<p>Vicente-Gonzalez, L., &amp; Vicente-Villardon, J. L. (2022). Partial Least Squares Regression for Binary Responses and Its Associated Biplot Representation. Mathematics, 10(15), 2580.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not yet
</code></pre>

<hr>
<h2 id='BinaryPLSR'>
Partial Least Squares Regression with Binary Data
</h2><span id='topic+BinaryPLSR'></span>

<h3>Description</h3>

<p>Fits Partial Least Squares Regression with Binary Data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryPLSR(Y, X, S = 2, tolerance = 5e-05, maxiter = 100, show = FALSE,
                   penalization = 0.1, OptimMethod = "CG", seed = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BinaryPLSR_+3A_y">Y</code></td>
<td>

<p>The response
</p>
</td></tr>
<tr><td><code id="BinaryPLSR_+3A_x">X</code></td>
<td>

<p>The matrix of independent variables
</p>
</td></tr>
<tr><td><code id="BinaryPLSR_+3A_s">S</code></td>
<td>

<p>The Dimension of the solution
</p>
</td></tr>
<tr><td><code id="BinaryPLSR_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence of the algorithm
</p>
</td></tr>
<tr><td><code id="BinaryPLSR_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum Number of iterations
</p>
</td></tr>
<tr><td><code id="BinaryPLSR_+3A_show">show</code></td>
<td>

<p>Show the steps of the algorithm
</p>
</td></tr>
<tr><td><code id="BinaryPLSR_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the Ridge Logistic Regression
</p>
</td></tr>
<tr><td><code id="BinaryPLSR_+3A_optimmethod">OptimMethod</code></td>
<td>

<p>Optimization methods from optim
</p>
</td></tr>
<tr><td><code id="BinaryPLSR_+3A_seed">seed</code></td>
<td>

<p>Seed. By default is 0.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function fits the PLSR method for the case when there are two sets of binary variables, using logistic rather than linear fits to take into account the nature of responses. We term the method BPLSR (Binary Partial Least Squares Regression). This can be considered as a generalization of the NIPALS algorithm when the data are all binary.
</p>


<h3>Value</h3>

<table>
<tr><td><code>Method</code></td>
<td>
<p>Description of 'comp1'</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>The predictors matrix</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>The responses matrix</p>
</td></tr>
<tr><td><code>ScaledX</code></td>
<td>
<p>The scaled X matrix</p>
</td></tr>
<tr><td><code>tolerance</code></td>
<td>
<p>Tolerance used in the algorithm</p>
</td></tr>
<tr><td><code>maxiter</code></td>
<td>
<p>Maximum number of iterations used</p>
</td></tr>
<tr><td><code>penalization</code></td>
<td>
<p>Ridge penalization</p>
</td></tr>
<tr><td><code>XScores</code></td>
<td>
<p>Scores of the X matrix, used later for the biplot</p>
</td></tr>
<tr><td><code>XLoadings</code></td>
<td>
<p>Loadings of the X matrix</p>
</td></tr>
<tr><td><code>YScores</code></td>
<td>
<p>Scores of the Y matrix</p>
</td></tr>
<tr><td><code>YLoadings</code></td>
<td>
<p>Loadings of the Y matrix</p>
</td></tr>

<tr><td><code>XStructure</code></td>
<td>
<p>Correlations among the X variables and the PLS scores</p>
</td></tr>
<tr><td><code>InterceptsY</code></td>
<td>
<p>Intercepts for the Y loadings</p>
</td></tr>
<tr><td><code>InterceptsX</code></td>
<td>
<p>Intercepts for the Y loadings</p>
</td></tr>
<tr><td><code>LinTerm</code></td>
<td>
<p>Linear terms for each response</p>
</td></tr>
<tr><td><code>Expected</code></td>
<td>
<p>Expected probabilities for the responses</p>
</td></tr>
<tr><td><code>Predictions</code></td>
<td>
<p>Binary predictions of the responses</p>
</td></tr>
<tr><td><code>PercentCorrect</code></td>
<td>
<p>Global percent of correct predictions</p>
</td></tr>
<tr><td><code>PercentCorrectCols</code></td>
<td>
<p>Percent of correct predictions for each column</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>José Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Ugarte Fajardo, J., Bayona Andrade, O., Criollo Bonilla, R., Cevallos‐Cevallos, J., Mariduena‐Zavala, M., Ochoa Donoso, D., &amp; Vicente Villardon, J. L. (2020). Early detection of black Sigatoka in banana leaves using hyperspectral images. Applications in plant sciences, 8(8), e11383.
</p>
<p>Vicente-Gonzalez, L., &amp; Vicente-Villardon, J. L. (2022). Partial Least Squares Regression for Binary Responses and Its Associated Biplot Representation. Mathematics, 10(15), 2580.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
X=as.matrix(wine[,4:21])
Y=cbind(Factor2Binary(wine[,1])[,1], Factor2Binary(wine[,2])[,1])
rownames(Y)=wine[,3]
colnames(Y)=c("Year", "Origin")
pls=PLSRBin(Y,X, penalization=0.1, show=TRUE, S=2)

</code></pre>

<hr>
<h2 id='BinaryProximities'>Proximity Measures for Binary Data</h2><span id='topic+BinaryProximities'></span>

<h3>Description</h3>

<p>Calculation of proxymities among rows or columns of a binary data matrix or a 
data frame that will be converted into a binary data matrix. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryProximities(x, y = NULL, coefficient = "Jaccard", transformation =
                 NULL, transpose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BinaryProximities_+3A_x">x</code></td>
<td>

<p>A data frame or a binary data matrix. Proximities among the rows of <code>x</code> will be calculated
</p>
</td></tr>
<tr><td><code id="BinaryProximities_+3A_y">y</code></td>
<td>

<p>Supplementary data. The proximities amond the rows of <code>x</code> and the rows of <code>y</code>  will be also calculated
</p>
</td></tr>
<tr><td><code id="BinaryProximities_+3A_coefficient">coefficient</code></td>
<td>

<p>Similarity coefficient. Use the number or the name (see details)
</p>
</td></tr>
<tr><td><code id="BinaryProximities_+3A_transformation">transformation</code></td>
<td>

<p>Transformation of the similarities. Use the number or the name (see details)
</p>
</td></tr>
<tr><td><code id="BinaryProximities_+3A_transpose">transpose</code></td>
<td>

<p>Logical. If <code>TRUE</code>, proximities among columns are calculated
</p>
</td></tr>
<tr><td><code id="BinaryProximities_+3A_...">...</code></td>
<td>

<p>Used to provide additional parameters for the conversion of the dataframe into a binary matrix
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A binary data matrix is a matrix with values 0 or 1 coding the absence or presence of several binary characters. When a data frame is provided, every variable in the data frame is converted to a binary variable using the function <code>Dataframe2BinaryMatrix</code>. Factors with two levels are converted directly to binary variables, factors with more than two levels are converted to a matrix with as meny columns as levels and numerical variables are converted to binary variables using a cut point that can be the median, the mean or a value provided by the user.
</p>
<p>The following coefficients are calculated
</p>
<p>1.- Kulezynski =  a/(b + c)
</p>
<p>2.- Russell_and_Rao =  a/(a + b + c+d)
</p>
<p>3.- Jaccard = a/(a + b + c)
</p>
<p>4.- Simple_Matching = (a + d)/(a + b + c + d)
</p>
<p>5.- Anderberg = a/(a + 2 * (b + c))
</p>
<p>6.- Rogers_and_Tanimoto = (a + d)/(a + 2 * (b + c) + d)
</p>
<p>7.- Sorensen_Dice_and_Czekanowski = a/(a + 0.5 * (b + c))
</p>
<p>8.- Sneath_and_Sokal = (a + d)/(a + 0.5 * (b + c) + d)
</p>
<p>9.- Hamman = (a - (b + c) + d)/(a + b + c + d)
</p>
<p>10.- Kulezynski = 0.5 * ((a/(a + b)) + (a/(a + c)))
</p>
<p>11.- Anderberg2 = 0.25 * (a/(a + b) + a/(a + c) + d/(c + d) + d/(b + d))
</p>
<p>12.- Ochiai = a/sqrt((a + b) * (a + c))
</p>
<p>13.- S13 = (a * d)/sqrt((a + b) * (a + c) * (d + b) * (d + c))
</p>
<p>14.- Pearson_phi = (a * d - b * c)/sqrt((a + b) * (a + c) * (d + b) * (d + c))
</p>
<p>15.- Yule = (a * d - b * c)/(a * d + b * c)
</p>
<p>The following transformations of the similarity3 are calculated
</p>
<p>1.- 'Identity'   dis=sim
</p>
<p>2.- '1-S'   dis=1-sim
</p>
<p>3.- 'sqrt(1-S)'  dis = sqrt(1 - sim)
</p>
<p>4.- '-log(s)'  dis=-1*log(sim)
</p>
<p>5.- '1/S-1' dis=1/sim -1 
</p>
<p>6.- 'sqrt(2(1-S))'  dis== sqrt(2*(1 - sim))
</p>
<p>7.- '1-(S+1)/2' dis=1-(sim+1)/2
</p>
<p>8.- '1-abs(S)'  dis=1-abs(sim)
</p>
<p>9.- '1/(S+1)'  dis=1/(sim)+1
</p>
<p>Note that, after transformation the similarities are converted to distances except for &quot;Identity&quot;.
Not all the transformations are suitable for all the coefficients. Use them at your own risk.
The default values are admissible combinations.
</p>


<h3>Value</h3>

<p>An object of class <code>proximities</code>.This has components:
</p>
<table>
<tr><td><code>TypeData</code></td>
<td>
<p>Binary, Continuous or Mixed. Binary in this case.</p>
</td></tr>
<tr><td><code>Coefficient</code></td>
<td>
<p>Coefficient used to calculate the proximities</p>
</td></tr>
<tr><td><code>Transformation</code></td>
<td>
<p>Transformation used to calculate the proximities</p>
</td></tr>
<tr><td><code>Data</code></td>
<td>
<p>Data used to calculate the proximities</p>
</td></tr>
<tr><td><code>SupData</code></td>
<td>
<p>Supplementary Data, if any</p>
</td></tr>
<tr><td><code>Proximities</code></td>
<td>
<p>Proximities among rows of <code>x</code>. May be similarities or dissimilarities depending on the transformation</p>
</td></tr>
<tr><td><code>SupProximities</code></td>
<td>
<p>Proximities among rows of <code>x</code> and <code>y</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Gower, J. C. (2006) Similarity dissimilarity and Distance, measures of. Encyclopedia of Statistical Sciences. 2nd. ed. Volume 12. Wiley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BinaryDistances">BinaryDistances</a></code>, <code><a href="#topic+Dataframe2BinaryMatrix">Dataframe2BinaryMatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
D=BinaryProximities(spiders, coefficient="Jaccard", transformation="sqrt(1-S)")
D2=BinaryProximities(spiders, coefficient=3, transformation=3)
</code></pre>

<hr>
<h2 id='Biplot.BinaryPLSR'>
Biplot for a PLSR model with binary data
</h2><span id='topic+Biplot.BinaryPLSR'></span>

<h3>Description</h3>

<p>Builds a Biplot for a PLSR model with binary data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Biplot.BinaryPLSR(plsr, BinBiplotType=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Biplot.BinaryPLSR_+3A_plsr">plsr</code></td>
<td>

<p>A BinaryPLSR object
</p>
</td></tr>
<tr><td><code id="Biplot.BinaryPLSR_+3A_binbiplottype">BinBiplotType</code></td>
<td>

<p>The type of biplot:
</p>
<p>1:The biplot resulting from the fit,  for the binary data.
</p>
<p>2: The biplot for the coefficients
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Builds a Biplot for a PLSR model with binary data. The result is a biplot for the matrix with the binary predictors (X) adding the binary responses as suplementary variables. There are two possible types, 1 for the biplot directly obtained in the fit (the default) and 2 for the biplot obtaines after refitting the binary variables using Ridge Logistic Regression.
</p>


<h3>Value</h3>

<p>An object of class Binary.Logistic.Biplot 
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Ugarte Fajardo, J., Bayona Andrade, O., Criollo Bonilla, R., Cevallos‐Cevallos, J., Mariduena‐Zavala, M., Ochoa Donoso, D., &amp; Vicente Villardon, J. L. (2020). Early detection of black Sigatoka in banana leaves using hyperspectral images. Applications in plant sciences, 8(8), e11383.
</p>
<p>Vicente-Gonzalez, L., &amp; Vicente-Villardon, J. L. (2022). Partial Least Squares Regression for Binary Responses and Its Associated Biplot Representation. Mathematics, 10(15), 2580.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
X=as.matrix(wine[,4:21])
Y=cbind(Factor2Binary(wine[,1])[,1], Factor2Binary(wine[,2])[,1])
rownames(Y)=wine[,3]
colnames(Y)=c("Year", "Origin")
pls=PLSRBin(Y,X, penalization=0.1, show=TRUE, S=2)
plsbip=Biplot.PLSRBIN(pls, BinBiplotType=1)
plsbip=AddCluster2Biplot(plsbip, ClusterType = "us", 
       Groups = wine$Group)
plot(plsbip, margin=0.05, mode="s", PlotClus = TRUE, 
    ModeSupBinVars = "s", ShowAxis = FALSE, 
    ColorSupBinVars = "blue",     CexInd=0.5, 
    ClustCenters = TRUE, LabelInd = FALSE, ShowBox = TRUE)

</code></pre>

<hr>
<h2 id='Biplot.PLSR'>
Partial Least Squares Biplot
</h2><span id='topic+Biplot.PLSR'></span>

<h3>Description</h3>

<p>Adds a Biplot to a Partial Lest Squares (plsr) object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Biplot.PLSR(plsr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Biplot.PLSR_+3A_plsr">plsr</code></td>
<td>

<p>A plsr object from the PLSR function
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adds a Biplot to a Partial Lest Squares (plsr) object. The biplot is constructed with the matrix of predictors, the dependent variable is projected onto the biplot as a continuous supplementary variable.
</p>


<h3>Value</h3>

<p>An object of class ContinuousBiplot with the dependent variables as supplemntary.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Oyedele, O. F., &amp; Lubbe, S. (2015). The construction of a partial least-squares biplot. Journal of Applied Statistics, 42(11), 2449-2460.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PLSR">PLSR</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X=as.matrix(wine[,4:21])
y=as.numeric(wine[,2])-1
mifit=PLSR(y,X, Validation="None")
mibip=Biplot.PLSR(mifit)
plot(mibip, PlotVars=TRUE, IndLabels = y, ColorInd=y+1)
</code></pre>

<hr>
<h2 id='Biplot.PLSR1BIN'>
Biplot for a PLSR model with a binary response
</h2><span id='topic+Biplot.PLSR1BIN'></span>

<h3>Description</h3>

<p>Biplot for a PLSR model with a binary response
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Biplot.PLSR1BIN(plsr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Biplot.PLSR1BIN_+3A_plsr">plsr</code></td>
<td>

<p>An object of class PLSR1BIN.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Biplot for a PLSR model with a binary response
</p>


<h3>Value</h3>

<p>The biplot for the independent variables with the response as supplementary binary variable.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Ugarte-Fajardo, J., Bayona-Andrade, O., Criollo-Bonilla, R., Cevallos-Cevallos, J., Mariduena-Zavala, M., Ochoa-Donoso, D., &amp; Vicente-Villardon, J. L. (2020). Early detection of black Sigatoka in banana leaves using hyperspectral images. Applications in plant sciences, 8(8), e11383.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PLSR1Bin">PLSR1Bin</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not Yet
</code></pre>

<hr>
<h2 id='Biplot.PLSRBIN'>
Biplot for a PLSR model with binary responses
</h2><span id='topic+Biplot.PLSRBIN'></span>

<h3>Description</h3>

<p>Builds a Biplot for a PLSR model with binary responses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Biplot.PLSRBIN(plsr, BinBiplotType = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Biplot.PLSRBIN_+3A_plsr">plsr</code></td>
<td>

<p>A PLSRBin object
</p>
</td></tr>
<tr><td><code id="Biplot.PLSRBIN_+3A_binbiplottype">BinBiplotType</code></td>
<td>

<p>The type of biplot:
</p>
<p>1:The biplot resulting from the fit,  for the binary responses.
</p>
<p>2: The biplot for the coefficients
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Builds a Biplot for a PLSR model with binary responses. The result is a biplot for the matrix with the predictors (X) adding the binary responses as suplementary variables. There are two possible types, 1 for the biplot directly obtained in the fit ( the default) and 2 for the biplot obtaines after refitting the binary variables using Ridge Logistic Regression.
</p>


<h3>Value</h3>

<p>An object of class ContinuousBiplot  
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Ugarte Fajardo, J., Bayona Andrade, O., Criollo Bonilla, R., Cevallos‐Cevallos, J., Mariduena‐Zavala, M., Ochoa Donoso, D., &amp; Vicente Villardon, J. L. (2020). Early detection of black Sigatoka in banana leaves using hyperspectral images. Applications in plant sciences, 8(8), e11383.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
X=as.matrix(wine[,4:21])
Y=cbind(Factor2Binary(wine[,1])[,1], Factor2Binary(wine[,2])[,1])
rownames(Y)=wine[,3]
colnames(Y)=c("Year", "Origin")
pls=PLSRBin(Y,X, penalization=0.1, show=TRUE, S=2)
plsbip=Biplot.PLSRBIN(pls, BinBiplotType=1)
plsbip=AddCluster2Biplot(plsbip, ClusterType = "us", 
       Groups = wine$Group)
plot(plsbip, margin=0.05, mode="s", PlotClus = TRUE, 
    ModeSupBinVars = "s", ShowAxis = FALSE, 
    ColorSupBinVars = "blue",     CexInd=0.5, 
    ClustCenters = TRUE, LabelInd = FALSE, ShowBox = TRUE)

</code></pre>

<hr>
<h2 id='BiplotFPCA'>
External Biplot for functional data from a functional PCA object.
</h2><span id='topic+BiplotFPCA'></span>

<h3>Description</h3>

<p>The function calculates a biplot from a functional PCA object and the data used tocalculate it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BiplotFPCA(FPCA, X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BiplotFPCA_+3A_fpca">FPCA</code></td>
<td>

<p>Functional PCA object
</p>
</td></tr>
<tr><td><code id="BiplotFPCA_+3A_x">X</code></td>
<td>

<p>Data used to calculate the fuctional PCA
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates a biplot from a functional PCA object and the data used tocalculate it. At this moment the function calculates only an external biplot by regressing X o the funcional components. Furure versions will include the internal biplot.
</p>


<h3>Value</h3>

<p>A Continuous biplot object
</p>


<h3>Author(s)</h3>

<p>José Luis Vicente Villardón
</p>


<h3>Examples</h3>

<pre><code class='language-R'># not yet
</code></pre>

<hr>
<h2 id='BootstrapDistance'>
Bootstrap on the distance matrices used for Principal Coordinates Analysis (PCoA)
</h2><span id='topic+BootstrapDistance'></span>

<h3>Description</h3>

<p>Obtains bootstrap replicates of a distance matrix using ramdom samples or permuatations of the residual matrix from a Principal Coordinates (Components) Analysis. The object is to estimate the sampling variability of absorbed variances, coordinates and qualities of representation in a PCoA. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BootstrapDistance(D, W=diag(nrow(D)), nB=200, dimsol=2, 
                  ProcrustesRot=TRUE, method=c("Sampling", "Permutation"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BootstrapDistance_+3A_d">D</code></td>
<td>

<p>A distance matrix
</p>
</td></tr>
<tr><td><code id="BootstrapDistance_+3A_w">W</code></td>
<td>

<p>A diagonal matrix containing waiths for the rows of D
</p>
</td></tr>
<tr><td><code id="BootstrapDistance_+3A_nb">nB</code></td>
<td>

<p>Number of Bootstrap replications
</p>
</td></tr>
<tr><td><code id="BootstrapDistance_+3A_dimsol">dimsol</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="BootstrapDistance_+3A_procrustesrot">ProcrustesRot</code></td>
<td>

<p>Should each replication be rotated to match the initial solution?
</p>
</td></tr>
<tr><td><code id="BootstrapDistance_+3A_method">method</code></td>
<td>

<p>The replications are obtained &quot;Sampling&quot; or &quot;Permutating&quot; the residuals.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates bootstrap confidence intervals for the inertia, coordinates and qualties of representation of a Principal Coordinates Analysis using a distance matrix as a basis. The funcion uses random sampling or permutations of the residuals to obtain the bootstrap replications. The procedure preserves the length of the points in the multidimensional space perturbating only the angles among the vectors. It is done so to preserve the property of positiveness of the diagonal elements of the scalar product matrices. The procedure may result into a scalar product that does not have an euclidean configuration and then has some negative eigenvalues; to avoid this problem the negative eigenvalues are removed to approximate the perturbated matrix by the closest with the required properties.
</p>
<p>It is well known that the eigenvectors of a matrix are unique except for reflections, that is, if we change the sign of each component of the eigenvector we have the same solution. If that happens, an unwanted increase in the variability due to this artifact may invalidate the results. To avoid this we can calculate the scalar product of each eigenvector of the initial matrix with the corresponding eigenvector of the bootstrap replicate and change the signs of the later if the result is negative.
</p>
<p>Another artifact of the procedure may arise when the dimension of the solution is higher than 1 because the eigenvectors of a replicate may generate the same subspace although are not in the same directions, i. e., the subspace is referred to a different system. That also may produce an unwanted increase of the variability that invalidates the results. To avoid this, every replicate may be rotated to match as much as possible the subspace generated by the eigenvectors of the initial matrix. This is done by Procrustes Analysis, taking the rotated matrix as solution. 
The solution to this problem is also a sulution to the reflection, then only this problem is considered.
</p>


<h3>Value</h3>

<p>Returns an object of class &quot;PCoABootstrap&quot; with the information for each bootstrap replication.
</p>
<table>
<tr><td><code>Eigenvalues</code></td>
<td>
<p>A matrix with dimensions in rows and replicates in columns containing the eigenvalues of each replicate in columns</p>
</td></tr>
<tr><td><code>Inertias</code></td>
<td>
<p>A matrix with dimensions in rows and replicates in columns containing the inertias of each replicate in columns</p>
</td></tr>
<tr><td><code>Coordinates</code></td>
<td>
<p>A list with a component for each object. A component contains the coordinates of an object for each replicate (in columns)</p>
</td></tr>
<tr><td><code>Values-Table</code></td>
<td>
<p>A list with a component for each object. A component contains the qualities of an object for each replicate (in columns)</p>
</td></tr>
<tr><td><code>NReplicates</code></td>
<td>
<p>Number of bootstrap replicates</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose L. Vicente-Villardon
</p>


<h3>References</h3>

<p>Efron, B.; Tibshirani, RJ. (1993). An introduction to the bootstrap. New York: Chapman and Hall. 436p.
</p>
<p>Ringrose, T. J. (1992). Bootstrapping and correspondence analysis in archaeology. Journal of Archaeological Science, 19(6), 615-629.
</p>
<p>MILAN, L., &amp; WHITTAKER, J. (1995). Application of the parametric bootstrap to models that incorporate a singular value decomposition. Applied statistics, 44(1), 31-49.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BootstrapScalar">BootstrapScalar</a></code>, ~~~
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
D=BinaryProximities(spiders, coefficient="Jaccard", transformation="sqrt(1-S)")
DB=BootstrapDistance(D$Proximities)
</code></pre>

<hr>
<h2 id='BootstrapScalar'>
Bootstrap on the  scalar product matrices used for Principal Coordinates Analysis (PCoA)
</h2><span id='topic+BootstrapScalar'></span>

<h3>Description</h3>

<p>Obtains bootstrap replicates of a scalar products matrix using ramdom samples or permuatations of the residual matrix from a Principal Coordinates (Components) Analysis. The object is to estimate the sampling variability of absorbed variances, coordinates and qualities of representation in a PCoA. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BootstrapScalar(B, W=diag(nrow(B)), nB=200, dimsol=2, 
                ProcrustesRot=TRUE, method=c("Sampling", "Permutation"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BootstrapScalar_+3A_b">B</code></td>
<td>

<p>A scalar product matrix
</p>
</td></tr>
<tr><td><code id="BootstrapScalar_+3A_w">W</code></td>
<td>

<p>A diagonal matrix containing waiths for the rows of D
</p>
</td></tr>
<tr><td><code id="BootstrapScalar_+3A_nb">nB</code></td>
<td>

<p>Number of Bootstrap replications
</p>
</td></tr>
<tr><td><code id="BootstrapScalar_+3A_dimsol">dimsol</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="BootstrapScalar_+3A_procrustesrot">ProcrustesRot</code></td>
<td>

<p>Should each replication be rotated to match the initial solution?
</p>
</td></tr>
<tr><td><code id="BootstrapScalar_+3A_method">method</code></td>
<td>

<p>The replications are obtained &quot;Sampling&quot; or &quot;Permutating&quot; the residuals.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates bootstrap confidence intervals for the inertia, coordinates and qualties of representation of a Principal Coordinates Analysis using a distance matrix as a basis. The funcion uses random sampling or permutations of the residuals to obtain the bootstrap replications. The procedure preserves the length of the points in the multidimensional space perturbating only the angles among the vectors. It is done so to preserve the property of positiveness of the diagonal elements of the scalar product matrices. The procedure may result into a scalar product that does not have an euclidean configuration and then has some negative eigenvalues; to avoid this problem the negative eigenvalues are removed to approximate the perturbated matrix by the closest with the required properties.
</p>
<p>It is well known that the eigenvectors of a matrix are unique except for reflections, that is, if we change the sign of each component of the eigenvector we have the same solution. If that happens, an unwanted increase in the variability due to this artifact may invalidate the results. To avoid this we can calculate the scalar product of each eigenvector of the initial matrix with the corresponding eigenvector of the bootstrap replicate and change the signs of the later if the result is negative.
</p>
<p>Another artifact of the procedure may arise when the dimension of the solution is higher than 1 because the eigenvectors of a replicate may generate the same subspace although are not in the same directions, i. e., the subspace is referred to a different system. That also may produce an unwanted increase of the variability that invalidates the results. To avoid this, every replicate may be rotated to match as much as possible the subspace generated by the eigenvectors of the initial matrix. This is done by Procrustes Analysis, taking the rotated matrix as solution. 
The solution to this problem is also a sulution to the reflection, then only this problem is considered.
</p>


<h3>Value</h3>

<p>Returns an object of class &quot;PCoABootstrap&quot; with the information for each bootstrap replication.
</p>
<table>
<tr><td><code>Eigenvalues</code></td>
<td>
<p>A matrix with dimensions in rows and replicates in columns containing the eigenvalues of each replicate in columns</p>
</td></tr>
<tr><td><code>Inertias</code></td>
<td>
<p>A matrix with dimensions in rows and replicates in columns containing the inertias of each replicate in columns</p>
</td></tr>
<tr><td><code>Coordinates</code></td>
<td>
<p>A list with a component for each object. A component contains the coordinates of an object for each replicate (in columns)</p>
</td></tr>
<tr><td><code>Values-Table</code></td>
<td>
<p>A list with a component for each object. A component contains the qualities of an object for each replicate (in columns)</p>
</td></tr>
<tr><td><code>NReplicates</code></td>
<td>
<p>Number of bootstrap replicates</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose L. Vicente-Villardon
</p>


<h3>References</h3>

<p>Efron, B.; Tibshirani, RJ. (1993). An introduction to the bootstrap. New York: Chapman and Hall. 436p.
</p>
<p>Ringrose, T. J. (1992). Bootstrapping and correspondence analysis in archaeology. Journal of Archaeological Science, 19(6), 615-629.
</p>
<p>Milan, L., &amp; Whittaker, J. (1995). Application of the parametric bootstrap to models that incorporate a singular value decomposition. Applied statistics, 44(1), 31-49.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BootstrapScalar">BootstrapScalar</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not yet
</code></pre>

<hr>
<h2 id='BootstrapSmacof'>
Bootstrap on the distance matrices used for MDS with Smacof
</h2><span id='topic+BootstrapSmacof'></span>

<h3>Description</h3>

<p>Obtains bootstrap replicates of a distance matrix using ramdom samples or permuatations of a distance matrix. The object is to estimate the sampling variability of the results of the Smacof algorithm. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BootstrapSmacof(D, W=NULL, Model=c("Identity", "Ratio", "Interval", "Ordinal"), 
                dimsol=2, maxiter=100, maxerror=0.000001, StandardizeDisparities=TRUE,
                ShowIter=TRUE, nB=200, ProcrustesRot=TRUE, 
                method=c("Sampling", "Permutation"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BootstrapSmacof_+3A_d">D</code></td>
<td>

<p>A distance matrix
</p>
</td></tr>
<tr><td><code id="BootstrapSmacof_+3A_w">W</code></td>
<td>

<p>A diagonal matrix containing waiths for the rows of D
</p>
</td></tr>
<tr><td><code id="BootstrapSmacof_+3A_model">Model</code></td>
<td>

<p>Mesurement level of the distances
</p>
</td></tr>
<tr><td><code id="BootstrapSmacof_+3A_dimsol">dimsol</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="BootstrapSmacof_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations for the smacof algorithm
</p>
</td></tr>
<tr><td><code id="BootstrapSmacof_+3A_maxerror">maxerror</code></td>
<td>

<p>Tolerance for the smacof algorithm
</p>
</td></tr>
<tr><td><code id="BootstrapSmacof_+3A_standardizedisparities">StandardizeDisparities</code></td>
<td>

<p>Should the disparities be standardized in the smacof algorithm?
</p>
</td></tr>
<tr><td><code id="BootstrapSmacof_+3A_showiter">ShowIter</code></td>
<td>

<p>Should the information on each ieration be printed on the screen?
</p>
</td></tr>
<tr><td><code id="BootstrapSmacof_+3A_nb">nB</code></td>
<td>

<p>Number of Bootstrap replications
</p>
</td></tr>
<tr><td><code id="BootstrapSmacof_+3A_procrustesrot">ProcrustesRot</code></td>
<td>

<p>Should each replication be rotated to match the initial solution?
</p>
</td></tr>
<tr><td><code id="BootstrapSmacof_+3A_method">method</code></td>
<td>

<p>The replications are obtained &quot;Sampling&quot; or &quot;Permutating&quot; the residuals.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates bootstrap confidence intervals for coordinates and different stress measures using a distance matrix as a basis. The funcion uses random sampling or permutations of the residuals to obtain the bootstrap replications. The procedure preserves the length of the points in the multidimensional space perturbating only the angles among the vectors. It is done so to preserve the property of positiveness of the diagonal elements of the scalar product matrices. The procedure may result into a scalar product that does not have an euclidean configuration and then has some negative eigenvalues; to avoid this problem the negative eigenvalues are removed to approximate the perturbated matrix by the closest with the required properties.
</p>
<p>It is well known that the eigenvectors of a matrix are unique except for reflections, that is, if we change the sign of each component of the eigenvector we have the same solution. If that happens, an unwanted increase in the variability due to this artifact may invalidate the results. To avoid this we can calculate the scalar product of each eigenvector of the initial matrix with the corresponding eigenvector of the bootstrap replicate and change the signs of the later if the result is negative.
</p>
<p>Another artifact of the procedure may arise when the dimension of the solution is higher than 1 because the eigenvectors of a replicate may generate the same subspace although are not in the same directions, i. e., the subspace is referred to a different system. That also may produce an unwanted increase of the variability that invalidates the results. To avoid this, every replicate may be rotated to match as much as possible the subspace generated by the eigenvectors of the initial matrix. This is done by Procrustes Analysis, taking the rotated matrix as solution. 
The solution to this problem is also a sulution to the reflection, then only this problem is considered.
</p>


<h3>Value</h3>

<p>Returns an object of class &quot;PCoABootstrap&quot; with the information for each bootstrap replication.
</p>
<table>
<tr><td><code>Info</code></td>
<td>
<p>Information about the procedure</p>
</td></tr>
<tr><td><code>InitialDistance</code></td>
<td>
<p>Initial distance</p>
</td></tr>
<tr><td><code>RawStress</code></td>
<td>
<p>A vector containing the raw stress for all the bootstrap replicates</p>
</td></tr>
<tr><td><code>stress1</code></td>
<td>
<p>A vector containing the value of the stress1 formula for all the bootstrap replicates</p>
</td></tr>
<tr><td><code>stress2</code></td>
<td>
<p>A vector containing the value of the stress2 formula for all the bootstrap replicates</p>
</td></tr>
<tr><td><code>sstress1</code></td>
<td>
<p>A vector containing the value of the sstress1 formula for all the bootstrap replicates</p>
</td></tr>
<tr><td><code>sstress2</code></td>
<td>
<p>A vector containing the value of the sstress2 formula for all the bootstrap replicates</p>
</td></tr>
<tr><td><code>Coordinates</code></td>
<td>
<p>A list with a component for each object. A component contains the coordinates of an object for all the bootstrap replicates (in columns)</p>
</td></tr>
<tr><td><code>NReplicates</code></td>
<td>
<p>Number of bootstrap replicates</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose L. Vicente-Villardon
</p>


<h3>References</h3>

<p>Efron, B.; Tibshirani, RJ. (1993). An introduction to the bootstrap. New York: Chapman and Hall. 436p.
</p>
<p>Ringrose, T. J. (1992). Bootstrapping and correspondence analysis in archaeology. Journal of Archaeological Science, 19(6), 615-629.
</p>
<p>MILAN, L., &amp; WHITTAKER, J. (1995). Application of the parametric bootstrap to models that incorporate a singular value decomposition. Applied statistics, 44(1), 31-49.
</p>
<p>Jacoby, W. G., &amp; Armstrong, D. A. (2014). Bootstrap Confidence Regions for Multidimensional Scaling Solutions. American Journal of Political Science, 58(1), 264-278.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BootstrapScalar">BootstrapScalar</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
D=BinaryProximities(spiders, coefficient="Jaccard", transformation="sqrt(1-S)")
DB=BootstrapDistance(D$Proximities)
</code></pre>

<hr>
<h2 id='BoxPlotPanel'>
Panel of box plots
</h2><span id='topic+BoxPlotPanel'></span>

<h3>Description</h3>

<p>Panel of box plots for a set of numerical variables and a grouping factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BoxPlotPanel(X, groups = NULL, nrows = NULL, panel = TRUE, 
notch = FALSE, GroupsTogether = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BoxPlotPanel_+3A_x">X</code></td>
<td>

<p>The matrix of continuous variables
</p>
</td></tr>
<tr><td><code id="BoxPlotPanel_+3A_groups">groups</code></td>
<td>

<p>The grouping factor
</p>
</td></tr>
<tr><td><code id="BoxPlotPanel_+3A_nrows">nrows</code></td>
<td>

<p>Number of rows of the panel.
</p>
</td></tr>
<tr><td><code id="BoxPlotPanel_+3A_panel">panel</code></td>
<td>

<p>Should the plots be organized into a panel? (or separated)
</p>
</td></tr>
<tr><td><code id="BoxPlotPanel_+3A_notch">notch</code></td>
<td>

<p>Should notches be used in the box plots?
</p>
</td></tr>
<tr><td><code id="BoxPlotPanel_+3A_groupstogether">GroupsTogether</code></td>
<td>

<p>Should all the groups be together in the same plot?
</p>
</td></tr>
<tr><td><code id="BoxPlotPanel_+3A_...">...</code></td>
<td>

<p>Other graphical arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Panel of box plots for a set of numerical variables and a grouping factor.
</p>


<h3>Value</h3>

<p>The box plot panel
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
BoxPlotPanel(wine[,4:7], groups = wine$Origin, nrows = 2, ylab="")

</code></pre>

<hr>
<h2 id='CA'>
Correspondence Analysis
</h2><span id='topic+CA'></span>

<h3>Description</h3>

<p>Correspondence Analysis for a frequency or abundace data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CA(x, dim = 2, alpha = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CA_+3A_x">x</code></td>
<td>

<p>The frequency or abundance data matrix.
</p>
</td></tr>
<tr><td><code id="CA_+3A_dim">dim</code></td>
<td>

<p>Dimension of the final solution
</p>
</td></tr>
<tr><td><code id="CA_+3A_alpha">alpha</code></td>
<td>

<p>Alpha to determine the kind of biplot to use.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates Correspondence Analysis for a tww-way frequency or abundance table
</p>


<h3>Value</h3>

<p>Correspondence analysis solution
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Benzécri, J. P. (1992). Correspondence analysis handbook. New York: Marcel Dekker.
</p>
<p>Greenacre, M. J. (1984). Theory and applications of correspondence analysis. Academic Press.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SpidersSp)
cabip=CA(SpidersSp)
plot(cabip)
</code></pre>

<hr>
<h2 id='Canonical.Variate.Analysis'>
Biplot representation of a Canonical Variate Analysis or a Manova (Canonical-Biplot or MANOVA-Biplot)
</h2><span id='topic+Canonical.Variate.Analysis'></span>

<h3>Description</h3>

<p>Calculates a canonical biplot with confidence regions for the means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Canonical.Variate.Analysis(X, group, InitialTransform = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Canonical.Variate.Analysis_+3A_x">X</code></td>
<td>

<p>A data matrix
</p>
</td></tr>
<tr><td><code id="Canonical.Variate.Analysis_+3A_group">group</code></td>
<td>

<p>A factor containing the groups
</p>
</td></tr>
<tr><td><code id="Canonical.Variate.Analysis_+3A_initialtransform">InitialTransform</code></td>
<td>

<p>Initial transformation of the data matrix
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Biplot method (Gabriel, 1971; Galindo, 1986; Gower and Hand, 1996) is becoming one of the most popular techniques for analysing multivariate data. Biplot methods are techniques for simultaneous representation of the <code class="reqn">n</code> rows and <code class="reqn">n</code> columns of a data matrix <code class="reqn">\bf{X}</code>, in reduced dimensions, where the rows represent individuals, objects or samples and the columns the variables measured on them. Classical Biplot methods are a graphical representation of a Principal Components Analysis (PCA) that it is used to obtain linear combinations that successively maximize the total variability. 
PCA is not considered an appropriate approach where there is known a priori group structure in the data. The most general methodology for discrimination among groups, using multiple observed variables, is Canonical Variate Analysis (CVA). CVA allows us to derive linear combinations that successively maximize the ratio of &quot;between-groups&quot;&quot; to &quot;pooled within-group&quot; sample variance. Several authors propose a Biplot representation for CVA called Canonical Biplot (CB) (Vicente-Villardon, 1992 and Gower &amp; Hand, 1996) when it is oriented to the discrimination between groups or MANOVA-Biplot Gabriel (1972, 1995) when the aim is to study the variables responsible for the discrimination. The main advantage of the Biplot version of the technique is that it is possible not only to establish the differences between groups but also to characterise the variables responsible for them. The methodology is not yet widely used mainly because it is still not available in the major statistical packages.
Amaro, Vicente-Villardon &amp; Galindo (2004) extend the methodology for two-way designs and propose confidence circles based on univariate  and multivariate tests to perform post-hoc analysis of each variable.
</p>


<h3>Value</h3>

<p>An object of class &quot;Canonical.Biplot&quot;
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Amaro, I. R., Vicente-Villardon, J. L., &amp; Galindo-Villardon, M. P. (2004). Manova Biplot para arreglos de tratamientos con dos factores basado en modelos lineales generales multivariantes. Interciencia, 29(1), 26-32.
</p>
<p>Vicente-Villardón, J. L. (1992). Una alternativa a las técnicas factoriales clásicas basada en una generalización de los métodos Biplot (Doctoral dissertation, Tesis. Universidad de Salamanca. España. 248 pp.[Links]).
</p>
<p>Gabriel KR (1971) The biplot graphic display of matrices with application to principal component analysis. Biometrika 58(3):453-467.
</p>
<p>Gabriel, K. R. (1995). MANOVA biplots for two-way contingency tables. WJ Krzanowski (Ed.), Recent advances in descriptive multivariate analysis, Oxford University Press, Toronto. 227-268.
</p>
<p>Galindo Villardon, M. (1986). Una alternativa de representacion simultanea: HJ-Biplot. Qüestiió. 1986, vol. 10, núm. 1.
</p>
<p>Gower y Hand (1996): Biplots. Chapman &amp; Hall.
</p>
<p>Varas, M. J., Vicente-Tavera, S., Molina, E., &amp; Vicente-Villardon, J. L. (2005). Role of canonical biplot method in the study of building stones: an example from Spanish monumental heritage. Environmetrics, 16(4), 405-419.
</p>
<p>Santana, M. A., Romay, G., Matehus, J., Villardon, J. L., &amp; Demey, J. R. (2009). simple and low-cost strategy for micropropagation of cassava (Manihot esculenta Crantz). African Journal of Biotechnology, 8(16).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X=wine[,4:21]
canbip=CanonicalBiplot(X, group=wine$Group)
plot(canbip, mode="s")
</code></pre>

<hr>
<h2 id='CanonicalBiplot'>
Biplot representation of a Canonical Variate Analysis or a Manova (Canonical-Biplot or MANOVA-Biplot)
</h2><span id='topic+CanonicalBiplot'></span>

<h3>Description</h3>

<p>Calculates a canonical biplot with confidence regions for the means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CanonicalBiplot(X, group, SUP = NULL, InitialTransform = 5, LDA=FALSE, MANOVA = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CanonicalBiplot_+3A_x">X</code></td>
<td>

<p>A data matrix
</p>
</td></tr>
<tr><td><code id="CanonicalBiplot_+3A_group">group</code></td>
<td>

<p>A factor containing the groups
</p>
</td></tr>
<tr><td><code id="CanonicalBiplot_+3A_sup">SUP</code></td>
<td>

<p>Supplementary observations to project on the biplot
</p>
</td></tr>
<tr><td><code id="CanonicalBiplot_+3A_initialtransform">InitialTransform</code></td>
<td>

<p>Initial transformation of the data matrix
</p>
</td></tr>
<tr><td><code id="CanonicalBiplot_+3A_lda">LDA</code></td>
<td>

<p>A logical to indicate if the discriminant analysis should also be included
</p>
</td></tr>
<tr><td><code id="CanonicalBiplot_+3A_manova">MANOVA</code></td>
<td>

<p>A logical to indicate if MANOVA should also be included
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Biplot method (Gabriel, 1971; Galindo, 1986; Gower and Hand, 1996) is becoming one of the most popular techniques for analysing multivariate data. Biplot methods are techniques for simultaneous representation of the <code class="reqn">n</code> rows and <code class="reqn">n</code> columns of a data matrix <code class="reqn">\bf{X}</code>, in reduced dimensions, where the rows represent individuals, objects or samples and the columns the variables measured on them. Classical Biplot methods are a graphical representation of a Principal Components Analysis (PCA) that it is used to obtain linear combinations that successively maximize the total variability. 
PCA is not considered an appropriate approach where there is known a priori group structure in the data. The most general methodology for discrimination among groups, using multiple observed variables, is Canonical Variate Analysis (CVA). CVA allows us to derive linear combinations that successively maximize the ratio of &quot;between-groups&quot;&quot; to &quot;pooled within-group&quot; sample variance. Several authors propose a Biplot representation for CVA called Canonical Biplot (CB) (Vicente-Villardon, 1992 and Gower &amp; Hand, 1996) when it is oriented to the discrimination between groups or MANOVA-Biplot Gabriel (1972, 1995) when the aim is to study the variables responsible for the discrimination. The main advantage of the Biplot version of the technique is that it is possible not only to establish the differences between groups but also to characterise the variables responsible for them. The methodology is not yet widely used mainly because it is still not available in the major statistical packages.
Amaro, Vicente-Villardon &amp; Galindo (2004) extend the methodology for two-way designs and propose confidence circles based on univariate  and multivariate tests to perform post-hoc analysis of each variable.
</p>


<h3>Value</h3>

<p>An object of class &quot;Canonical.Biplot&quot;
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Amaro, I. R., Vicente-Villardon, J. L., &amp; Galindo-Villardon, M. P. (2004). Manova Biplot para arreglos de tratamientos con dos factores basado en modelos lineales generales multivariantes. Interciencia, 29(1), 26-32.
</p>
<p>Vicente-Villardón, J. L. (1992). Una alternativa a las técnicas factoriales clásicas basada en una generalización de los métodos Biplot (Doctoral dissertation, Tesis. Universidad de Salamanca. España. 248 pp.[Links]).
</p>
<p>Gabriel KR (1971) The biplot graphic display of matrices with application to principal component analysis. Biometrika 58(3):453-467.
</p>
<p>Gabriel, K. R. (1995). MANOVA biplots for two-way contingency tables. WJ Krzanowski (Ed.), Recent advances in descriptive multivariate analysis, Oxford University Press, Toronto. 227-268.
</p>
<p>Galindo Villardon, M. (1986). Una alternativa de representacion simultanea: HJ-Biplot. Qüestiió. 1986, vol. 10, núm. 1.
</p>
<p>Gower y Hand (1996): Biplots. Chapman &amp; Hall.
</p>
<p>Varas, M. J., Vicente-Tavera, S., Molina, E., &amp; Vicente-Villardon, J. L. (2005). Role of canonical biplot method in the study of building stones: an example from Spanish monumental heritage. Environmetrics, 16(4), 405-419.
</p>
<p>Santana, M. A., Romay, G., Matehus, J., Villardon, J. L., &amp; Demey, J. R. (2009). simple and low-cost strategy for micropropagation of cassava (Manihot esculenta Crantz). African Journal of Biotechnology, 8(16).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X=wine[,4:21]
canbip=CanonicalBiplot(X, group=wine$Group)
plot(canbip, mode="s")
</code></pre>

<hr>
<h2 id='CanonicalDistanceAnalysis'>
MANOVA and Canonical Analysis of Distances
</h2><span id='topic+CanonicalDistanceAnalysis'></span>

<h3>Description</h3>

<p>Performs a MANOVA and a Canonical Analysis based on of Distance Matrices (usally for continuous data)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CanonicalDistanceAnalysis(Prox, group, dimens = 2, Nsamples = 1000, 
PCoA = "Standard", ProjectInd = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CanonicalDistanceAnalysis_+3A_prox">Prox</code></td>
<td>

<p>A object containing proximities
</p>
</td></tr>
<tr><td><code id="CanonicalDistanceAnalysis_+3A_group">group</code></td>
<td>

<p>A factor with the group structure of the rows
</p>
</td></tr>
<tr><td><code id="CanonicalDistanceAnalysis_+3A_dimens">dimens</code></td>
<td>

<p>The dimension of the solution
</p>
</td></tr>
<tr><td><code id="CanonicalDistanceAnalysis_+3A_nsamples">Nsamples</code></td>
<td>

<p>Number of samples for the permutation test. Number of permutations.
</p>
</td></tr>
<tr><td><code id="CanonicalDistanceAnalysis_+3A_pcoa">PCoA</code></td>
<td>

<p>Type of Principal Coordinates for the Canonical Analysis calculated from the Principal coordinates of the Mean Matrix. &quot;Standard&quot; : Standard Principal Coordinates Analysis, &quot;Weighted&quot;: Weighted Principal Coordinates Analysis, &quot;WPCA&quot;)
</p>
</td></tr>
<tr><td><code id="CanonicalDistanceAnalysis_+3A_projectind">ProjectInd</code></td>
<td>

<p>Should the individual points be Projected onto the representation For the moment only available for Continuous Data.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Performs a MANOVA and a Canonical Analysis based on of Distance Matrices (usally for continuous data). The MANOVA statistics is calculated from a decomposition of the distance matrix based on a factor of a external classification. The significance of the test is calculated using a premutation test. The approach depens only on the distances and then is useful with any kind of data.
</p>
<p>The Canonical Representation is calculated from a Principal Coordinates Analysis od the distance matrix among the means. Thus, it is only possible for continuous data. The PCoA representation can be &quot;Standard&quot; using the means directly, &quot;Weighted&quot; weighting each group with its sample size or using weighted Princiopal Components Analysis of the matrix of means.
</p>
<p>A measure of the quality of representation of the groups is provided. When possible, the measure is also provided for the individual points.
</p>
<p>Soon, a biplot representation will also be developed.
</p>


<h3>Value</h3>

<p>An object of class &quot;CanonicalDistanceAnalysis&quot; with:
</p>
<table>
<tr><td><code>Distances</code></td>
<td>
<p>The Matrix of Distances from which the Analysis has been made</p>
</td></tr>
<tr><td><code>Groups</code></td>
<td>
<p>A factor containing the group struture of the individuals</p>
</td></tr>
<tr><td><code>TSS</code></td>
<td>
<p>Total sum of squares</p>
</td></tr>
<tr><td><code>BSS</code></td>
<td>
<p>Between groups sum of squares</p>
</td></tr>
<tr><td><code>WSS</code></td>
<td>
<p>Within groups sum of squares</p>
</td></tr>
<tr><td><code>Fexp</code></td>
<td>
<p>Experimental pseudo F-value</p>
</td></tr>
<tr><td><code>pvalue</code></td>
<td>
<p>p value based on the permutation test</p>
</td></tr>
<tr><td><code>Nsamples</code></td>
<td>
<p>p value based on the permutation test</p>
</td></tr>
<tr><td><code>ExplainedVariance</code></td>
<td>
<p>Variances explained by the PCoA</p>
</td></tr>
<tr><td><code>MeanCoordinates</code></td>
<td>
<p>Coordinates of the groups for the graphical representation</p>
</td></tr>
<tr><td><code>Qualities</code></td>
<td>
<p>Qualities of the representation of the groups</p>
</td></tr>
<tr><td><code>CummulativeQualities</code></td>
<td>
<p>Cummulative qualities of the representation of the groups</p>
</td></tr>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates of the individuals for the graphical representation</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The MANOVA and the representation of the means can be applied to any Distance althoug the projection of the individuals can be made only for continuous data.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Gower, J. C., &amp; Krzanowski, W. J. (1999). Analysis of distance for structured multivariate data and extensions to multivariate analysis of variance. Journal of the Royal Statistical Society: Series C (Applied Statistics), 48(4), 505-519.
</p>
<p>Krzanowski, W. J. (2004). Biplots for multifactorial analysis of distance. Biometrics, 60(2), 517-524.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
group=iris[,5]
X=as.matrix(iris[1:4])
D=ContinuousProximities(X,  coef = 1)
CDA=CanonicalDistanceAnalysis(D, group, dimens=2)
summary(CDA)
plot(CDA)
</code></pre>

<hr>
<h2 id='CanonicalStatisBiplot'>
CANONICAL STATIS-ACT for multiple tables with common rows and its associated Biplot
</h2><span id='topic+CanonicalStatisBiplot'></span>

<h3>Description</h3>

<p>The procedure performs STATIS-ACT methodology for multiple tables with common rows and its associated biplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CanonicalStatisBiplot(X, Groups, InitTransform = "Standardize columns", dimens = 2,
                 SameVar = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CanonicalStatisBiplot_+3A_x">X</code></td>
<td>

<p>A list containing multiple tables with common rows
</p>
</td></tr>
<tr><td><code id="CanonicalStatisBiplot_+3A_groups">Groups</code></td>
<td>

<p>A factor containing the groups
</p>
</td></tr>
<tr><td><code id="CanonicalStatisBiplot_+3A_inittransform">InitTransform</code></td>
<td>

<p>Initial transformation of the data matrices
</p>
</td></tr>
<tr><td><code id="CanonicalStatisBiplot_+3A_dimens">dimens</code></td>
<td>

<p>Dimension of the final solution
</p>
</td></tr>
<tr><td><code id="CanonicalStatisBiplot_+3A_samevar">SameVar</code></td>
<td>

<p>Are the variables the same for all occasions?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The procedure performs Canonical STATIS-ACT methodology for multiple tables with common rows and its associated biplot. When the variables are the same for all occasions trajectories for the variables can also be plotted.
</p>


<h3>Value</h3>

<p>An object of class StatisBiplot
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Vallejo-Arboleda, A., Vicente-Villardon, J. L., &amp; Galindo-Villardon, M. P. (2007). Canonical STATIS: Biplot analysis of multi-table group structured data based on STATIS-ACT methodology. Computational statistics &amp; data analysis, 51(9), 4193-4205.
</p>
<p>Abdi, H., Williams, L.J., Valentin, D., &amp; Bennani-Dosse, M. (2012). STATIS and DISTATIS: optimum multitable principal component analysis and three way metric multidimensional scaling. WIREs Comput Stat, 4, 124-167.
</p>
<p>Efron, B.,Tibshirani, RJ. (1993). An introduction to the bootstrap. New York: Chapman and Hall. 436p.
</p>
<p>Escoufier, Y. (1976). Operateur associe a un tableau de donnees. Annales de laInsee, 22-23, 165-178.
</p>
<p>Escoufier, Y. (1987). The duality diagram: a means for better practical applications. En P. Legendre &amp; L. Legendre (Eds.), Developments in Numerical Ecology, pp. 139-156, NATO Advanced Institute, Serie G. Berlin: Springer.
</p>
<p>L'Hermier des Plantes, H. (1976). Structuration des Tableaux a Trois Indices de la Statistique. [These de Troisieme Cycle]. University of Montpellier, France.
</p>
<p>Ringrose, T.J. (1992). Bootstrapping and Correspondence Analysis in Archaeology. Journal of Archaeological Science. 19:615-629.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Chemical)
x= Chemical[37:144,5:9]
weeks=as.factor(as.numeric(Chemical$WEEKS[37:144]))
levels(weeks)=c("W2" , "W3", "W4")
X=Convert2ThreeWay(x,weeks, columns=FALSE)
Groups=Chemical$Treatment[1:36]
canstbip=CanonicalStatisBiplot(X, Groups, SameVar = TRUE)
plot(canstbip, mode="s", PlotVars=TRUE, ShowBox=TRUE)
</code></pre>

<hr>
<h2 id='CategoricalDistances'>
Distances among individuals using nominal variables.
</h2><span id='topic+CategoricalDistances'></span>

<h3>Description</h3>

<p>Distances among individuals using nominal variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CategoricalDistances(x, y = NULL, coefficient = "GOW", transformation = "sqrt(1-S)")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CategoricalDistances_+3A_x">x</code></td>
<td>

<p>Matrix of Categorical Data
</p>
</td></tr>
<tr><td><code id="CategoricalDistances_+3A_y">y</code></td>
<td>

<p>A second matrix of categorical data with the same variables as x
</p>
</td></tr>
<tr><td><code id="CategoricalDistances_+3A_coefficient">coefficient</code></td>
<td>

<p>Similarity coefficient to use (see details)
</p>
</td></tr>
<tr><td><code id="CategoricalDistances_+3A_transformation">transformation</code></td>
<td>

<p>Transformation of the similarity into a distance
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates similarities and dissimilarities among a set ob ogjects characterized by a set of nominal variables. The function uses similarities and converts into dissimilarities using a variety of transformations controled by the user.
</p>


<h3>Value</h3>

<p>A matrix with distances among the rows of x and y. If y is NULL the interdistances among the rows of x are calculated.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>dos Santos, T. R., &amp; Zarate, L. E. (2015). Categorical data clustering: What similarity measure to recommend?. Expert Systems with Applications, 42(3), 1247-1260.
</p>
<p>Boriah, S., Chandola, V., &amp; Kumar, V. (2008). Similarity measures for categorical data: A comparative evaluation. red, 30(2), 3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
</code></pre>

<hr>
<h2 id='CategoricalProximities'>
Proximities among individuals using nominal variables.
</h2><span id='topic+CategoricalProximities'></span>

<h3>Description</h3>

<p>Proximities among individuals using nominal variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CategoricalProximities(Data, SUP = NULL, coefficient = "GOW", transformation = 3, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CategoricalProximities_+3A_data">Data</code></td>
<td>

<p>A data frame containing categorical (nominal) variables
</p>
</td></tr>
<tr><td><code id="CategoricalProximities_+3A_sup">SUP</code></td>
<td>

<p>Supplementary data (Used to project supplementary individuals onto the PCoA configuration, for example)
</p>
</td></tr>
<tr><td><code id="CategoricalProximities_+3A_coefficient">coefficient</code></td>
<td>

<p>Similarity coefficient to use (see details)
</p>
</td></tr>
<tr><td><code id="CategoricalProximities_+3A_transformation">transformation</code></td>
<td>

<p>Transformation of the similarity into a distance
</p>
</td></tr>
<tr><td><code id="CategoricalProximities_+3A_...">...</code></td>
<td>

<p>Extra parameters
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates similarities and dissimilarities among a set ob ogjects characterized by a set of nominal variables. The function uses similarities and converts into dissimilarities using a variety of transformations controled by the user.
</p>


<h3>Value</h3>

<p>A list of Values
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>dos Santos, T. R., &amp; Zarate, L. E. (2015). Categorical data clustering: What similarity measure to recommend?. Expert Systems with Applications, 42(3), 1247-1260.
</p>
<p>Boriah, S., Chandola, V., &amp; Kumar, V. (2008). Similarity measures for categorical data: A comparative evaluation. red, 30(2), 3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Doctors)
Dis=CategoricalProximities(Doctors, SUP=NULL, coefficient="GOW" , transformation=3)
pco=PrincipalCoordinates(Dis)
plot(pco, RowCex=0.7, RowColors=as.integer(Doctors[[1]]), RowLabels=as.character(Doctors[[1]]))
</code></pre>

<hr>
<h2 id='CCA'>
Canonical Correspondence Analysis
</h2><span id='topic+CCA'></span>

<h3>Description</h3>

<p>Calculates the solution of a Canonical Correspondence Analysis Biplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CCA(P, Z, alpha = 1, dimens = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CCA_+3A_p">P</code></td>
<td>

<p>Abundance Matrix of sites by species.
</p>
</td></tr>
<tr><td><code id="CCA_+3A_z">Z</code></td>
<td>

<p>Environmental variables measured at the same sites
</p>
</td></tr>
<tr><td><code id="CCA_+3A_alpha">alpha</code></td>
<td>

<p>Alpha for the biplot decomposition [0,1]. With alpha=1 the emphasis is on the sites
and with alpha=0 the emphasis is on the species
</p>
</td></tr>
<tr><td><code id="CCA_+3A_dimens">dimens</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A pair of ecological tables, made of a species abundance matrix and an environmental variables matrix measured at the same sampling sites, is usually analyzed by Canonical Correspondence Analysis (CCA) (Ter BRAAK, 1986).  CCA can be considered as a Correspondence Analysis (CA) in which the ordination axis are constrained to be linear combinations of the environmental variables. Recently the procedure has been extended to other disciplines as Sociology or Psichology and it is potentially useful in many other fields.
</p>


<h3>Value</h3>

<p>A CCA solution object
</p>


<h3>Author(s)</h3>

<p>Jose Luis vicente Villardon
</p>


<h3>References</h3>

<p>Ter Braak, C. J. (1986). Canonical correspondence analysis: a new eigenvector technique for multivariate direct gradient analysis. Ecology, 67(5), 1167-1179.
</p>
<p>Johnson, K. W., &amp; Altman, N. S. (1999). Canonical correspondence analysis as an approximation to Gaussian ordination. Environmetrics, 10(1), 39-52.
</p>
<p>Graffelman, J. (2001). Quality statistics in canonical correspondence analysis. Environmetrics, 12(5), 485-497.
</p>
<p>Graffelman, J., &amp; Tuft, R. (2004). Site scores and conditional biplots in canonical correspondence analysis. Environmetrics, 15(1), 67-80.
</p>
<p>Greenacre, M. (2010). Canonical correspondence analysis in social science research (pp. 279-286). Springer Berlin Heidelberg.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(riano)
Sp=riano[,3:15]
Env=riano[,16:25]
ccabip=CCA(Sp, Env)
plot(ccabip)
</code></pre>

<hr>
<h2 id='CheckBinaryMatrix'>Checks if a data matrix is binary</h2><span id='topic+CheckBinaryMatrix'></span>

<h3>Description</h3>

<p>Checks if a data matrix is binary
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckBinaryMatrix(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckBinaryMatrix_+3A_x">x</code></td>
<td>

<p>Matrix to check.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Checks if all the entries of the matix are either 0 or 1.
</p>


<h3>Value</h3>

<p><code>TRUE</code> if the matrix is binary.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
sp=Dataframe2BinaryMatrix(spiders)
CheckBinaryMatrix(sp)

</code></pre>

<hr>
<h2 id='CheckBinaryVector'>
Checks if a vector is binary
</h2><span id='topic+CheckBinaryVector'></span>

<h3>Description</h3>

<p>Checks if all the entries of a vector are 0 or 1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckBinaryVector(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckBinaryVector_+3A_x">x</code></td>
<td>

<p>he vector to check
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The logical result 
</p>


<h3>Author(s)</h3>

<p>Jose luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x=c(0, 0, 0, 0,  1, 1, 1, 2)
CheckBinaryVector(x)
</code></pre>

<hr>
<h2 id='Chemical'>
Chemical data 
</h2><span id='topic+Chemical'></span>

<h3>Description</h3>

<p>Ecological data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("Chemical")</code></pre>


<h3>Format</h3>

<p>A data frame with 324 observations on the following 16 variables.
</p>

<dl>
<dt><code>Treatment</code></dt><dd><p>a factor with levels <code>F0N0</code> <code>F0N1</code> <code>F0N2</code> <code>F0N3</code> <code>F1N0</code> <code>F1N1</code> <code>F1N2</code> <code>F1N3</code> <code>F2N0</code> <code>F2N1</code> <code>F2N2</code> <code>F2N3</code></p>
</dd>
<dt><code>FISH</code></dt><dd><p>a factor with levels <code>F0</code> <code>F1</code> <code>F2</code></p>
</dd>
<dt><code>NUTRIENTS</code></dt><dd><p>a factor with levels <code>N0</code> <code>N1</code> <code>N2</code> <code>N3</code></p>
</dd>
<dt><code>WEEKS</code></dt><dd><p>a factor with levels <code>W1</code> <code>W2</code> <code>W3</code> <code>W4</code> <code>W5</code> <code>W6</code> <code>W7</code> <code>W8</code> <code>W9</code></p>
</dd>
<dt><code>TEMPERATURE</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>pH</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ALKALINITYmeql</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CO2free</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NNH4mgl</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NNO3mgl</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SRPmglP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TPmgl</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TSSmgl</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CONDUCTIVITYmScm</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TSPmglP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Chlorophyllamgl</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Chemical Data
</p>


<h3>Source</h3>

<p>Department of Ecology. University of Leon. (Spain)
</p>


<h3>References</h3>

<p>To add
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Chemical)
## maybe str(Chemical) ; plot(Chemical) ...
</code></pre>

<hr>
<h2 id='Circle'>
Draws a circle
</h2><span id='topic+Circle'></span>

<h3>Description</h3>

<p>Draws a circle for a given radius at the specified center with the given color
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Circle(radius = 1, origin = c(0, 0), col = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Circle_+3A_radius">radius</code></td>
<td>

<p>radius of the circle
</p>
</td></tr>
<tr><td><code id="Circle_+3A_origin">origin</code></td>
<td>

<p>Centre of the circle
</p>
</td></tr>
<tr><td><code id="Circle_+3A_col">col</code></td>
<td>

<p>Color od the circle
</p>
</td></tr>
<tr><td><code id="Circle_+3A_...">...</code></td>
<td>

<p>Aditional graphical parameters
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Draws a circle for a given radius at the specified center with the given color
</p>


<h3>Value</h3>

<p>No value is returned





</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(0,0)
Circle(1,c(0,0))
</code></pre>

<hr>
<h2 id='Coinertia'>Coinertia Analysis.
</h2><span id='topic+Coinertia'></span>

<h3>Description</h3>

<p>Calculates a Coinertia Analysis for two matrices of continuous data 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Coinertia(X, Y, ScalingX = 5, ScalingY = 5, dimsol = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Coinertia_+3A_x">X</code></td>
<td>

<p>The first matrix in the analysis
</p>
</td></tr>
<tr><td><code id="Coinertia_+3A_y">Y</code></td>
<td>

<p>The second matrix in the analysis
</p>
</td></tr>
<tr><td><code id="Coinertia_+3A_scalingx">ScalingX</code></td>
<td>

<p>Transformation of the X matrix
</p>
</td></tr>
<tr><td><code id="Coinertia_+3A_scalingy">ScalingY</code></td>
<td>

<p>Transformation of the Y matrix
</p>
</td></tr>
<tr><td><code id="Coinertia_+3A_dimsol">dimsol</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Coinertia analysis for two continuous data matrices.
</p>


<h3>Value</h3>

<p>An object of class Coinertia.SOL



</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Doledec, S., &amp; Chessel, D. (1994). Co-inertia analysis: an alternative method for studying species-environment relationships. Freshwater biology, 31(3), 277-294.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
SSI$Year == "a2006"
SSI2D=SSI[SSI$Year == "a2006",3:23]
rownames(SSI2D)=as.character(SSI$Country[SSI$Year == "a2006"])
SSIHuman2D=SSI2D[,1:9]
SSIEnvir2D=SSI2D[,10:16]
SSIEcon2D=SSI2D[,17:21]
Coin=Coinertia(SSIHuman2D, SSIEnvir2D)

</code></pre>

<hr>
<h2 id='ColContributionPlot'>
Plots the contributios of a biplot
</h2><span id='topic+ColContributionPlot'></span>

<h3>Description</h3>

<p>Plots the contributios of a biplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ColContributionPlot(bip, A1 = 1, A2 = 2, Colors = NULL, Labs = NULL, 
MinQuality = 0, CorrelationScale = FALSE, ContributionScale = TRUE, 
AddSigns2Labs = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ColContributionPlot_+3A_bip">bip</code></td>
<td>

<p>An object of class ContinuousBiplot
</p>
</td></tr>
<tr><td><code id="ColContributionPlot_+3A_a1">A1</code></td>
<td>

<p>First dimension to plot
</p>
</td></tr>
<tr><td><code id="ColContributionPlot_+3A_a2">A2</code></td>
<td>

<p>Second dimension to plot
</p>
</td></tr>
<tr><td><code id="ColContributionPlot_+3A_colors">Colors</code></td>
<td>

<p>Colors for the variables
</p>
</td></tr>
<tr><td><code id="ColContributionPlot_+3A_labs">Labs</code></td>
<td>

<p>Labels for the variables
</p>
</td></tr>
<tr><td><code id="ColContributionPlot_+3A_minquality">MinQuality</code></td>
<td>

<p>Min quality to plot
</p>
</td></tr>
<tr><td><code id="ColContributionPlot_+3A_correlationscale">CorrelationScale</code></td>
<td>

<p>Scales for correlation
</p>
</td></tr>
<tr><td><code id="ColContributionPlot_+3A_contributionscale">ContributionScale</code></td>
<td>

<p>Scales for contributions
</p>
</td></tr>
<tr><td><code id="ColContributionPlot_+3A_addsigns2labs">AddSigns2Labs</code></td>
<td>

<p>Add the siggns of the correlations to the labels
</p>
</td></tr>
<tr><td><code id="ColContributionPlot_+3A_...">...</code></td>
<td>

<p>Any other graphical parameter
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the contributions on a plot that sows also the sum for both axes-
</p>


<h3>Value</h3>

<p>The contribution plot
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Simple Biplot with arrows
data(Protein)
bip=PCA.Biplot(Protein[,3:11])

# Plot of the Variable Contributions
ColContributionPlot(bip, cex=1)
</code></pre>

<hr>
<h2 id='ConcEllipse'>
Concentration ellipse for a se of two-dimensional points
</h2><span id='topic+ConcEllipse'></span>

<h3>Description</h3>

<p>The function calculates a non-parametric concentration ellipse for a set of two-dimensional points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConcEllipse(data, confidence=1, npoints=100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ConcEllipse_+3A_data">data</code></td>
<td>

<p>The set of two-dimensional points
</p>
</td></tr>
<tr><td><code id="ConcEllipse_+3A_confidence">confidence</code></td>
<td>

<p>Percentage of points to be included in the ellipse
</p>
</td></tr>
<tr><td><code id="ConcEllipse_+3A_npoints">npoints</code></td>
<td>

<p>Number of points to draw the eelipse contour. The hier the number of points the smouther is the ellipse.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The procedre uses the Mahalanobis distances to determine the points that will be used for the calculations.
</p>


<h3>Value</h3>

<p>A list with the following fields
</p>
<table>
<tr><td><code>data</code></td>
<td>
<p>Data Used for the calculations</p>
</td></tr>
<tr><td><code>confidence</code></td>
<td>
<p>The confidence level used</p>
</td></tr>
<tr><td><code>ellipse</code></td>
<td>
<p>The points on the ellipse contour to be plotted</p>
</td></tr>
<tr><td><code>center</code></td>
<td>
<p>The center of the points</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Meulman, J. J., &amp; Heiser, W. J. (1983). The display of bootstrap solutions in multidimensional scaling. Murray Hill, NJ: Bell Laboratories.
</p>
<p>Linting, M., Meulman, J. J., Groenen, P. J., &amp; Van der Kooij, A. J. (2007). Stability of nonlinear principal components analysis: An empirical study using the balanced bootstrap. Psychological Methods, 12(3), 359.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
dat=as.matrix(iris[1:50,1:2])
plot(iris[,1], iris[,2],col=iris[,5], asp=1)
E=ConcEllipse(dat, 0.95)
plot(E)

</code></pre>

<hr>
<h2 id='ConfidenceInterval'>
Confidence Interval for the mean
</h2><span id='topic+ConfidenceInterval'></span>

<h3>Description</h3>

<p>Calculates Confidence Interval for the mean of a Numerical Variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConfidenceInterval(x, Desv = NULL, df = NULL, Confidence = 0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ConfidenceInterval_+3A_x">x</code></td>
<td>

<p>The numerical variable
</p>
</td></tr>
<tr><td><code id="ConfidenceInterval_+3A_desv">Desv</code></td>
<td>

<p>Standard deviation. If NULL, the sd is calculated from the data
</p>
</td></tr>
<tr><td><code id="ConfidenceInterval_+3A_df">df</code></td>
<td>

<p>Degrees of freedom
</p>
</td></tr>
<tr><td><code id="ConfidenceInterval_+3A_confidence">Confidence</code></td>
<td>

<p>Confidence Level
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates Confidence Interval for the mean of a Numerical Variable. 
</p>


<h3>Value</h3>

<p>The confidence Interval for the mean
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='ConstrainedLogisticBiplot'>
Constrained Binary Logistic Biplot
</h2><span id='topic+ConstrainedLogisticBiplot'></span>

<h3>Description</h3>

<p>Constrained Binary Logistic Biplot or Redundancy Analysis for Binary Data based on logistic responses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConstrainedLogisticBiplot(Y, X, dim = 2, Scaling = 5, tolerance = 1e-05, 
maxiter = 100, penalization = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ConstrainedLogisticBiplot_+3A_y">Y</code></td>
<td>

<p>A binary data matrix
</p>
</td></tr>
<tr><td><code id="ConstrainedLogisticBiplot_+3A_x">X</code></td>
<td>

<p>A matrix of predictors
</p>
</td></tr>
<tr><td><code id="ConstrainedLogisticBiplot_+3A_dim">dim</code></td>
<td>

<p>Dimension of the Solution
</p>
</td></tr>
<tr><td><code id="ConstrainedLogisticBiplot_+3A_scaling">Scaling</code></td>
<td>

<p>Transformation of the columns of the predictor matrix.
</p>
</td></tr>
<tr><td><code id="ConstrainedLogisticBiplot_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for the algorithm
</p>
</td></tr>
<tr><td><code id="ConstrainedLogisticBiplot_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations.
</p>
</td></tr>
<tr><td><code id="ConstrainedLogisticBiplot_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the fit (ridge)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Constrained Binary Logistic Biplot or Redundancy Analysis for Binary Data based on logistic responses.
</p>


<h3>Value</h3>

<p>A logistic Biplot with the reponse and the predictive variables projected onto it.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardon, J. L., &amp; Vicente-Gonzalez, L. Redundancy Analysis for Binary Data Based on Logistic Responses in Data Analysis and Rationality in a Complex World. Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># not yet
</code></pre>

<hr>
<h2 id='ConstrainedOrdinalLogisticBiplot'>
Constrained Ordinal Logistic Biplot
</h2><span id='topic+ConstrainedOrdinalLogisticBiplot'></span>

<h3>Description</h3>

<p>Constrained Ordinal Logistic Biplot or Redundancy Analysis for Ordinal Data based on logistic responses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConstrainedOrdinalLogisticBiplot(Y, X, dim = 2, Scaling = 5, 
tolerance = 1e-05, maxiter = 100, penalization = 0.1, show = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ConstrainedOrdinalLogisticBiplot_+3A_y">Y</code></td>
<td>

<p>A binary data matrix
</p>
</td></tr>
<tr><td><code id="ConstrainedOrdinalLogisticBiplot_+3A_x">X</code></td>
<td>

<p>A matrix of predictors
</p>
</td></tr>
<tr><td><code id="ConstrainedOrdinalLogisticBiplot_+3A_dim">dim</code></td>
<td>

<p>Dimension of the Solution
</p>
</td></tr>
<tr><td><code id="ConstrainedOrdinalLogisticBiplot_+3A_scaling">Scaling</code></td>
<td>

<p>Transformation of the columns of the predictor matrix.
</p>
</td></tr>
<tr><td><code id="ConstrainedOrdinalLogisticBiplot_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for the algorithm
</p>
</td></tr>
<tr><td><code id="ConstrainedOrdinalLogisticBiplot_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations.
</p>
</td></tr>
<tr><td><code id="ConstrainedOrdinalLogisticBiplot_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the fit (ridge)
</p>
</td></tr>
<tr><td><code id="ConstrainedOrdinalLogisticBiplot_+3A_show">show</code></td>
<td>

<p>Show each step ot the fit
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Constrained Ordinal Logistic Biplot or Redundancy Analysis for Ordinal Data based on logistic responses.
</p>


<h3>Value</h3>

<p>An ordinal logistic Biplot with the reponse and the predictive variables projected onto it.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardon, J. L., &amp; Vicente-Gonzalez, L. Redundancy Analysis for Binary Data Based on Logistic Responses in Data Analysis and Rationality in a Complex World. Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># not yet
</code></pre>

<hr>
<h2 id='ContinuousDistances'>Distances for Continuous Data</h2><span id='topic+ContinuousDistances'></span>

<h3>Description</h3>

<p>Calculates distances among rows of a continuous data matrix or among the rows of two continuous matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ContinuousDistances(x, y = NULL, coef = "Pythagorean", r = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ContinuousDistances_+3A_x">x</code></td>
<td>

<p>Main data matrix. Distances among rows are calculated if y=NULL.
</p>
</td></tr>
<tr><td><code id="ContinuousDistances_+3A_y">y</code></td>
<td>

<p>Supplementary data matrix. If not NULL the distances among the rows of x and y are calculated 
</p>
</td></tr>
<tr><td><code id="ContinuousDistances_+3A_coef">coef</code></td>
<td>

<p>Distance coefficient. Use the name or the number(see details)
</p>
</td></tr>
<tr><td><code id="ContinuousDistances_+3A_r">r</code></td>
<td>

<p>Exponent for the Minkowsky
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following coefficients are calculated
</p>
<p>1.- Pythagorean =  sqrt(sum((y[i, ] - x[j, ])^2)/p)
</p>
<p>2.- Taxonomic =  sqrt(sum(((y[i,]-x[j,])^2)/r^2)/p)
</p>
<p>3.- City = sum(abs(y[i,]-x[j,])/r)/p
</p>
<p>4.- Minkowski = (sum((abs(y[i,]-x[j,])/r)^t)/p)^(1/t)
</p>
<p>5.- Divergence = sqrt(sum((y[i,]-x[j,])^2/(y[i,]+x[j,])^2)/p)
</p>
<p>6.- dif_sum = sum(abs(y[i,]-x[j,])/abs(y[i,]+x[j,]))/p
</p>
<p>7.- Camberra = sum(abs(y[i,]-x[j,])/(abs(y[i,])+abs(x[j,])))
</p>
<p>8.- Bray_Curtis = sum(abs(y[i,]-x[j,]))/sum(y[i,]+x[j,])
</p>
<p>9.- Soergel = sum(abs(y[i,]-x[j,]))/sum(apply(rbind(y[i,],x[j,]),2,max))
</p>
<p>10.- Ware_hedges = sum(abs(y[i,]-x[j,]))/sum(apply(rbind(y[i,],x[j,]),2,max))
</p>


<h3>Value</h3>

<p>A list with:
</p>
<table>
<tr><td><code>Data</code></td>
<td>
<p>A matrix with the initial data (x matrix).</p>
</td></tr>
<tr><td><code>SupData</code></td>
<td>
<p>A matrix with the supplementary data (y matrix).</p>
</td></tr>
<tr><td><code>D</code></td>
<td>
<p>The matrix of distances</p>
</td></tr>
<tr><td><code>Coefficient</code></td>
<td>
<p>The coefficient used.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Gower, J. C. (2006) Similarity dissimilarity and Distance, measures of. Encyclopedia of Statistical Sciences. 2nd. ed. Volume 12. Wiley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PrincipalCoordinates">PrincipalCoordinates</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
dis=ContinuousDistances(wine[,4:21])
</code></pre>

<hr>
<h2 id='ContinuousProximities'>
Proximities for Continuous Data
</h2><span id='topic+ContinuousProximities'></span>

<h3>Description</h3>

<p>Calculates proximities among rows of a continuous data matrix or among the rows of two continuous matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ContinuousProximities(x, y = NULL, ysup = FALSE, 
transpose = FALSE, coef = "Pythagorean", r = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ContinuousProximities_+3A_x">x</code></td>
<td>

<p>Main data matrix. Distances among rows are calculated if y=NULL.
</p>
</td></tr>
<tr><td><code id="ContinuousProximities_+3A_y">y</code></td>
<td>

<p>Supplementary data matrix. If not NULL the distances among the rows of x and y are calculated 
</p>
</td></tr>
<tr><td><code id="ContinuousProximities_+3A_ysup">ysup</code></td>
<td>

<p>Supplementary Y data
</p>
</td></tr>
<tr><td><code id="ContinuousProximities_+3A_transpose">transpose</code></td>
<td>

<p>Transpose rows and columns
</p>
</td></tr>
<tr><td><code id="ContinuousProximities_+3A_coef">coef</code></td>
<td>

<p>Distance coefficient. Use the name or the number(see details)
</p>
</td></tr>
<tr><td><code id="ContinuousProximities_+3A_r">r</code></td>
<td>

<p>Exponent for the Minkowsky
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following coefficients are calculated
</p>
<p>1.- Pythagorean =  sqrt(sum((y[i, ] - x[j, ])^2)/p)
</p>
<p>2.- Taxonomic =  sqrt(sum(((y[i,]-x[j,])^2)/r^2)/p)
</p>
<p>3.- City = sum(abs(y[i,]-x[j,])/r)/p
</p>
<p>4.- Minkowski = (sum((abs(y[i,]-x[j,])/r)^t)/p)^(1/t)
</p>
<p>5.- Divergence = sqrt(sum((y[i,]-x[j,])^2/(y[i,]+x[j,])^2)/p)
</p>
<p>6.- dif_sum = sum(abs(y[i,]-x[j,])/abs(y[i,]+x[j,]))/p
</p>
<p>7.- Camberra = sum(abs(y[i,]-x[j,])/(abs(y[i,])+abs(x[j,])))
</p>
<p>8.- Bray_Curtis = sum(abs(y[i,]-x[j,]))/sum(y[i,]+x[j,])
</p>
<p>9.- Soergel = sum(abs(y[i,]-x[j,]))/sum(apply(rbind(y[i,],x[j,]),2,max))
</p>
<p>10.- Ware_hedges = sum(abs(y[i,]-x[j,]))/sum(apply(rbind(y[i,],x[j,]),2,max))
</p>


<h3>Value</h3>

<table>
<tr><td><code>Data</code></td>
<td>
<p>A matrix with the initial data (x matrix).</p>
</td></tr>
<tr><td><code>SupData</code></td>
<td>
<p>A matrix with the supplementary data (y matrix).</p>
</td></tr>
<tr><td><code>D</code></td>
<td>
<p>The matrix of distances</p>
</td></tr>
<tr><td><code>Coefficient</code></td>
<td>
<p>The coefficient used.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Gower, J. C. (2006) Similarity dissimilarity and Distance, measures of. Encyclopedia of Statistical Sciences. 2nd. ed. Volume 12. Wiley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
dis=ContinuousProximities(wine[,4:21])
</code></pre>

<hr>
<h2 id='Convert2ThreeWay'>
Three way array from a two way matrix
</h2><span id='topic+Convert2ThreeWay'></span>

<h3>Description</h3>

<p>Converts a two-dimensional matrix into a list where
each cell is the two dimensional data matrix  
for an occasion or group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Convert2ThreeWay(x, groups, columns = FALSE, RowNames = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Convert2ThreeWay_+3A_x">x</code></td>
<td>

<p>The two dimensional matrix
</p>
</td></tr>
<tr><td><code id="Convert2ThreeWay_+3A_groups">groups</code></td>
<td>

<p>A factor defining the groups
</p>
</td></tr>
<tr><td><code id="Convert2ThreeWay_+3A_columns">columns</code></td>
<td>

<p>Are the grouos defined for columns?
</p>
</td></tr>
<tr><td><code id="Convert2ThreeWay_+3A_rownames">RowNames</code></td>
<td>

<p>Names for the rows of each table.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Converts a two dimensional matrix into a multitable list according to the groups provided by the user. Each field of the list has the name of the corresponding group.
</p>


<h3>Value</h3>

<p>A Multitable list. Ech filed is the data matrix for a group.
</p>
<table>
<tr><td><code>X</code></td>
<td>
<p>The multitable list</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Chemical)
x= Chemical[,5:16]
X=Convert2ThreeWay(x,Chemical$WEEKS, columns=FALSE)
</code></pre>

<hr>
<h2 id='Convert3wArray2List'>
Converts a three way array into a list
</h2><span id='topic+Convert3wArray2List'></span>

<h3>Description</h3>

<p>Converts a three way array into a list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Convert3wArray2List(X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Convert3wArray2List_+3A_x">X</code></td>
<td>

<p>A three way array
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Converts a three way array into a list
</p>


<h3>Value</h3>

<p>A list
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#No examples yet
</code></pre>

<hr>
<h2 id='ConvertFactors2Integers'>
Convert a factor to integer numbers
</h2><span id='topic+ConvertFactors2Integers'></span>

<h3>Description</h3>

<p>Convert a factor to integer numbers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConvertFactors2Integers(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ConvertFactors2Integers_+3A_x">x</code></td>
<td>

<p>A vector with a factor
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Convert a factor to integer numbers
</p>


<h3>Value</h3>

<p>a vector with the converted values
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
</code></pre>

<hr>
<h2 id='ConvertList23wArray'>
Converts a list of matrices into a three way array
</h2><span id='topic+ConvertList23wArray'></span>

<h3>Description</h3>

<p>Converts a list of matrices into a three way array. All the matrices in the list must have the same size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConvertList23wArray(X)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ConvertList23wArray_+3A_x">X</code></td>
<td>

<p>A list with data matrices.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Converts a list of matrices into a three way array. All the matrices in the list must have the same size.
</p>


<h3>Value</h3>

<p>A three-way array
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'># No examples yet
</code></pre>

<hr>
<h2 id='CorrelationCircle'>
Circle of correlations
</h2><span id='topic+CorrelationCircle'></span>

<h3>Description</h3>

<p>Circle of correlations among the manifiest variables and the principal comonents (or dimensions of any biplot).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CorrelationCircle(bip, A1 = 1, A2 = 2, Colors = NULL, Labs = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CorrelationCircle_+3A_bip">bip</code></td>
<td>

<p>an biplot object of any kind.
</p>
</td></tr>
<tr><td><code id="CorrelationCircle_+3A_a1">A1</code></td>
<td>

<p>First dimension for the representation
</p>
</td></tr>
<tr><td><code id="CorrelationCircle_+3A_a2">A2</code></td>
<td>

<p>Second dimension for the representation
</p>
</td></tr>
<tr><td><code id="CorrelationCircle_+3A_colors">Colors</code></td>
<td>

<p>Colors of the variables
</p>
</td></tr>
<tr><td><code id="CorrelationCircle_+3A_labs">Labs</code></td>
<td>

<p>Labels of the variables
</p>
</td></tr>
<tr><td><code id="CorrelationCircle_+3A_...">...</code></td>
<td>

<p>Any other graphical parameters
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Circle of correlations among the manifiest variables and the principal comonents (or dimensions of any biplot).
</p>


<h3>Value</h3>

<p>The plot of the circle of correlations
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bip=PCA.Biplot(wine[,4:21])
CorrelationCircle(bip)
</code></pre>

<hr>
<h2 id='CrissCross'>Alternated Least Squares Biplot
</h2><span id='topic+CrissCross'></span>

<h3>Description</h3>

<p>Alternated Least Squares Biplot with any choice of weigths for each element of the data matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CrissCross(x, w = matrix(1, dim(x)[1], dim(x)[2]), dimens = 2, a0 = NULL, 
b0 = NULL, maxiter = 100, tol = 1e-04, addsvd = TRUE, lambda = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CrissCross_+3A_x">x</code></td>
<td>

<p>Data Matrix to be analysed
</p>
</td></tr>
<tr><td><code id="CrissCross_+3A_w">w</code></td>
<td>

<p>Weights matrix. Must be of the same size as X.
</p>
</td></tr>
<tr><td><code id="CrissCross_+3A_dimens">dimens</code></td>
<td>

<p>Dimension of the solution.
</p>
</td></tr>
<tr><td><code id="CrissCross_+3A_a0">a0</code></td>
<td>

<p>Starting row coordinates. Random coordinates are calculated if the argument is NULL.
</p>
</td></tr>
<tr><td><code id="CrissCross_+3A_b0">b0</code></td>
<td>

<p>Starting column coordinates. Random coordinates are calculated if the argument is NULL.
</p>
</td></tr>
<tr><td><code id="CrissCross_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations
</p>
</td></tr>
<tr><td><code id="CrissCross_+3A_tol">tol</code></td>
<td>

<p>Tolerance for the algorithm to converge.
</p>
</td></tr>
<tr><td><code id="CrissCross_+3A_addsvd">addsvd</code></td>
<td>

<p>Calculate an additional SVD at the end of the algorithm. That meakes the final solution more readable
</p>
</td></tr>
<tr><td><code id="CrissCross_+3A_lambda">lambda</code></td>
<td>

<p>Constant to add to the diagonal of the natrices to be inverted in order to improve stability when the matrices are ill-conditioned.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates Alternated Least Squares Biplot with any choice of weigths for each element of the data matrix. The function is useful when we want a low rank approximation of a data matrix in which each element of the matrix has a different weight, for example, all the weights are 1 except for the missing elements that are 0, or to model the logarithms of a frequency table using the frequencies as weights.
</p>


<h3>Value</h3>

<p>An object of class .Biplot&quot; with the following components:
</p>
<table>
<tr><td><code>n</code></td>
<td>
<p>Number of Rows</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>Number of Columns</p>
</td></tr>
<tr><td><code>dim</code></td>
<td>
<p>Dimension of the Biplot</p>
</td></tr>
<tr><td><code>EigenValues</code></td>
<td>
<p>Eigenvalues</p>
</td></tr>
<tr><td><code>Inertia</code></td>
<td>
<p>Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>CumInertia</code></td>
<td>
<p>Cumulative Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates for the rows</p>
</td></tr>
<tr><td><code>ColCoordinates</code></td>
<td>
<p>Coordinates for the columns</p>
</td></tr>
<tr><td><code>RowContributions</code></td>
<td>
<p>Contributions for the rows</p>
</td></tr>
<tr><td><code>ColContributions</code></td>
<td>
<p>Contributions for the columns</p>
</td></tr>
<tr><td><code>Scale_Factor</code></td>
<td>
<p>Scale factor for the traditional plot with points and arrows. The row coordinates are multiplied and the column coordinates divided by that scale factor. The look of the plot is better without changing the inner product. For the HJ-Biplot the scale factor is 1.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>GABRIEL, K.R. and ZAMIR, S. (1979). Lower rank approximation of matrices by least squares with any choice of weights. Technometrics, 21: 489-498.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+LogFrequencyBiplot">LogFrequencyBiplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Protein)
X=as.matrix(Protein[,3:11])
X = InitialTransform(X, transform=5)$X
bip=CrissCross(X)
</code></pre>

<hr>
<h2 id='CumSum'>
Cummulative sums
</h2><span id='topic+CumSum'></span>

<h3>Description</h3>

<p>Cummulative sums 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CumSum(X, dimens = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CumSum_+3A_x">X</code></td>
<td>

<p>Data Matrix
</p>
</td></tr>
<tr><td><code id="CumSum_+3A_dimens">dimens</code></td>
<td>

<p>Dimension for summing
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Cummulative sums within rows (dimens=1) or columns (dimens=2) of a data matrix
</p>


<h3>Value</h3>

<p>A matrix of the same size as X with cummulative sums within each row or each column
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X=wine[,4:21]
CumSum(X,1)
CumSum(X,2)
</code></pre>

<hr>
<h2 id='Dataframe2BinaryMatrix'>Converts a Data Frame into a Binary Data Matrix</h2><span id='topic+Dataframe2BinaryMatrix'></span>

<h3>Description</h3>

<p>Converts a Data Frame into a Binary Data Matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dataframe2BinaryMatrix(dataf, cuttype = "Median", cut = NULL, BinFact = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Dataframe2BinaryMatrix_+3A_dataf">dataf</code></td>
<td>

<p>data.frame to be converted
</p>
</td></tr>
<tr><td><code id="Dataframe2BinaryMatrix_+3A_cuttype">cuttype</code></td>
<td>

<p>Type of cut point for continuous variables. Must be &quot;Median&quot; or &quot;Mean&quot;.
Does not have any effect for factors
</p>
</td></tr>
<tr><td><code id="Dataframe2BinaryMatrix_+3A_cut">cut</code></td>
<td>

<p>Personalized cut value for continuous variables.
</p>
</td></tr>
<tr><td><code id="Dataframe2BinaryMatrix_+3A_binfact">BinFact</code></td>
<td>

<p>Should I treat a factor with two levels as binary. This means that only a single dummy rather than two is used
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function converts a data frame into a Binary Data Matrix  (A matrix with entries either 0 or 1).
</p>
<p>Factors with two levels are directly transformed into a column of 0/1 entries.
</p>
<p>Factors with more than two levels are converted into a binary submatrix with
as many rows as <code>x</code> and as many columns as levels or categories. (Indicator matrix)
</p>
<p>Integer Variables are treated as factors
</p>
<p>Continuous Variables are converted into binary variables using a cut point that can be
the median, the mean or a value provided by the user.
</p>


<h3>Value</h3>

<p>A Binary Data Matrix.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
Dataframe2BinaryMatrix(spiders)
</code></pre>

<hr>
<h2 id='DataFrame2Matrix4Regression'>
Prepares a matrix for regression from a data frame
</h2><span id='topic+DataFrame2Matrix4Regression'></span>

<h3>Description</h3>

<p>Prepares a matrix for regression from a data frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DataFrame2Matrix4Regression(X, last = TRUE, Intercept = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DataFrame2Matrix4Regression_+3A_x">X</code></td>
<td>

<p>A data frame
</p>
</td></tr>
<tr><td><code id="DataFrame2Matrix4Regression_+3A_last">last</code></td>
<td>

<p>Logical to use the last category of nominal variabless as baseline.
</p>
</td></tr>
<tr><td><code id="DataFrame2Matrix4Regression_+3A_intercept">Intercept</code></td>
<td>

<p>Logical to tell the function if a constant must be present
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Nominal variables are converted to a matrix of dummy variables for regression. 
</p>


<h3>Value</h3>

<p>A matrix ready to use as independent variables in a regression
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Vilardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
</code></pre>

<hr>
<h2 id='DensityBiplot'>
Adds Non-parametric densities to a biplot. Separated densities are calculated for different clusters
</h2><span id='topic+DensityBiplot'></span>

<h3>Description</h3>

<p>Adds Non-parametric densities to a biplot. Separated densities are calculated for different clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DensityBiplot(X, y = NULL, grouplabels = NULL, ncontours = 6, 
groupcolors = NULL, ncolors=20, ColorType=4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DensityBiplot_+3A_x">X</code></td>
<td>

<p>Two dimensional coordinates of the points in a biplot (or any other)
</p>
</td></tr>
<tr><td><code id="DensityBiplot_+3A_y">y</code></td>
<td>

<p>A factor containing clusters or groups for separate densities. 
</p>
</td></tr>
<tr><td><code id="DensityBiplot_+3A_grouplabels">grouplabels</code></td>
<td>

<p>Labels for the groups
</p>
</td></tr>
<tr><td><code id="DensityBiplot_+3A_ncontours">ncontours</code></td>
<td>

<p>Number of contours to represent on the biplot
</p>
</td></tr>
<tr><td><code id="DensityBiplot_+3A_groupcolors">groupcolors</code></td>
<td>

<p>Colors for the groups
</p>
</td></tr>
<tr><td><code id="DensityBiplot_+3A_ncolors">ncolors</code></td>
<td>

<p>Number of colors for the density patterns
</p>
</td></tr>
<tr><td><code id="DensityBiplot_+3A_colortype">ColorType</code></td>
<td>

<p>One of the following: &quot;1&quot; = rainbow, &quot;2&quot; = heat.colors, &quot;3&quot; = terrain.colors, &quot;4&quot; = topo.colors, &quot;5&quot; = cm.colors
</p>
</td></tr>      
</table>


<h3>Details</h3>

<p>Non parametric densities are used to investigate the concentration of row points on different areas of the biplot representation. The densities can be calculated for different groups or clusters in order to investigate if individuals with differnt characteristics are concentrated on particular areas of the biplot. The procedure is particularly useful with a high number of individuals.
</p>


<h3>Value</h3>

<p>No value returned. It has effect on the graph.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Gower, J. C., Lubbe, S. G., &amp; Le Roux, N. J. (2011). Understanding biplots. John Wiley &amp; Sons.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bip=PCA.Biplot(iris[,1:4])
plot(bip, mode="s", CexInd=0.1)


</code></pre>

<hr>
<h2 id='Dhats'>
Calculation of Disparities
</h2><span id='topic+Dhats'></span>

<h3>Description</h3>

<p>Calculation of Disparities for a MDS model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dhats(P, D, W, Model = c("Identity", "Ratio", "Interval", "Ordinal"), Standardize = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Dhats_+3A_p">P</code></td>
<td>

<p>A matrix of proximities (usually dissimilarities)
</p>
</td></tr>
<tr><td><code id="Dhats_+3A_d">D</code></td>
<td>

<p>A matrix of distances obtained from an euclidean configuration
</p>
</td></tr>
<tr><td><code id="Dhats_+3A_w">W</code></td>
<td>

<p>A matrix of weights
</p>
</td></tr>
<tr><td><code id="Dhats_+3A_model">Model</code></td>
<td>

<p>Measurement level of the proximities
</p>
</td></tr>
<tr><td><code id="Dhats_+3A_standardize">Standardize</code></td>
<td>

<p>Should the Disparities be standardized?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculation of disparities using standard  or monotone regression depending on the MDS model.
</p>


<h3>Value</h3>

<p>Returns the proximities.
</p>


<h3>Author(s)</h3>

<p>Jose L. Vicente Villardon
</p>


<h3>References</h3>

<p>Borg, I., &amp; Groenen, P. J. (2005). Modern multidimensional scaling: Theory and applications. Springer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Function is used inside MDS or smacof
</code></pre>

<hr>
<h2 id='diagonal'>
Diagonal matrix from a vector
</h2><span id='topic+diagonal'></span>

<h3>Description</h3>

<p>Creates a diagonal matrix from a vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diagonal(d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diagonal_+3A_d">d</code></td>
<td>

<p>A numerical vector
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A diagonal matrix wirh the values of vector in the diagonal a zeros elsewhere
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>diag(c(1, 2, 3, 4, 5))</code></pre>

<hr>
<h2 id='DimensionLabels'>
Labels for the selected dimensions in a biplot
</h2><span id='topic+DimensionLabels'></span>

<h3>Description</h3>

<p>Creates a character vector with labels for the dimensions of the biplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DimensionLabels(dimens, Root = "Dim")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DimensionLabels_+3A_dimens">dimens</code></td>
<td>

<p>Number of dimensions
</p>
</td></tr>
<tr><td><code id="DimensionLabels_+3A_root">Root</code></td>
<td>

<p>Root of the label
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An auxiliary function to cretae labels for the dimensions. Useful to label the matrices of results
</p>


<h3>Value</h3>

<p>Returns a vector of labels
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>DimensionLabels(dimens=3, Root = "Dim")
</code></pre>

<hr>
<h2 id='dlines'>
Connects two sets of points by lines
</h2><span id='topic+dlines'></span>

<h3>Description</h3>

<p>Connects two sets of points by lines in a rowwise manner. Adapted from Graffelman(2013)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dlines(SetA, SetB, lin = "dotted", color = "black", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dlines_+3A_seta">SetA</code></td>
<td>

<p>First set of points
</p>
</td></tr>
<tr><td><code id="dlines_+3A_setb">SetB</code></td>
<td>

<p>Second set of points
</p>
</td></tr>
<tr><td><code id="dlines_+3A_lin">lin</code></td>
<td>

<p>Line style.
</p>
</td></tr>
<tr><td><code id="dlines_+3A_color">color</code></td>
<td>

<p>Line color
</p>
</td></tr>
<tr><td><code id="dlines_+3A_...">...</code></td>
<td>

<p>Any other graphical parameters 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Connects two sets of points by lines 
</p>


<h3>Value</h3>

<p>NULL
</p>


<h3>Author(s)</h3>

<p>Based on Graffelman (2013)
</p>


<h3>References</h3>

<p>Jan Graffelman (2013). calibrate: Calibration of Scatterplot and Biplot Axes. R package version 1.7.2.
http://CRAN.R-project.org/package=calibrate
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## No examples
</code></pre>

<hr>
<h2 id='Doctors'>
Data set extracted from the Careers of doctorate holders survey carried out by Spanish Statistical Office in 2008.
</h2><span id='topic+Doctors'></span>

<h3>Description</h3>

<p>The sample data, as part of a large survey, corresponds to 100 people who have the PhD degree
and it shows the level of satisfaction of the doctorate holders about some issues.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Doctors)</code></pre>


<h3>Format</h3>

<p>This data frame contains 100 observation for the following 5 ordinal variables,
with four categories each:   (1= &quot;Very Satisfied&quot;, 2= &quot;Somewhat Satisfied&quot;,3=&quot;Somewhat dissatisfied&quot;,  4=&quot;Very dissatisfied&quot;)
</p>

<dl>
<dt>Salary</dt><dd></dd>
<dt>Benefits</dt><dd></dd>
<dt>Job Security</dt><dd></dd>
<dt>Job Location</dt><dd></dd>
<dt>Working conditions</dt><dd></dd>
</dl>



<h3>Source</h3>

<p>Spanish Statistical Institute. Survey of PDH holders, 2006. 
URL: http://www.ine.es.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Doctors)
## maybe str(Doctors) ; plot(Doctors) ...
</code></pre>

<hr>
<h2 id='ErrorBarPlotPanel'>
Plots a panel of error bars
</h2><span id='topic+ErrorBarPlotPanel'></span>

<h3>Description</h3>

<p>Plots a panel of error bars to compare the means of several variables in the levels of a factor using confidence intervals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ErrorBarPlotPanel(X, groups = NULL, nrows = NULL, panel = TRUE, 
GroupsTogether = TRUE, Confidence = 0.95, p.adjust.method = "None", 
UseANOVA = FALSE, Colors = "blue", Title = "Error Bar Plot", 
sort = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ErrorBarPlotPanel_+3A_x">X</code></td>
<td>

<p>A matrix containing several variables
</p>
</td></tr>
<tr><td><code id="ErrorBarPlotPanel_+3A_groups">groups</code></td>
<td>

<p>A factor defining groups of individuals
</p>
</td></tr>
<tr><td><code id="ErrorBarPlotPanel_+3A_nrows">nrows</code></td>
<td>

<p>Number of rows of the panel. The function calculates the number of columns needed.
</p>
</td></tr>
<tr><td><code id="ErrorBarPlotPanel_+3A_panel">panel</code></td>
<td>

<p>The plots are shown on a panel (TRUE) or in separated graphs (FALSE)
</p>
</td></tr>
<tr><td><code id="ErrorBarPlotPanel_+3A_groupstogether">GroupsTogether</code></td>
<td>

<p>The groups appear together on the same plot
</p>
</td></tr>
<tr><td><code id="ErrorBarPlotPanel_+3A_confidence">Confidence</code></td>
<td>

<p>Confidence levels for the error bars (confidence intervals)
</p>
</td></tr>
<tr><td><code id="ErrorBarPlotPanel_+3A_p.adjust.method">p.adjust.method</code></td>
<td>

<p>Method for adjusting the p-value to cope with multiple comparisons.
</p>
</td></tr>
<tr><td><code id="ErrorBarPlotPanel_+3A_useanova">UseANOVA</code></td>
<td>

<p>If TRUE the function uses the residual variance of the ANOVA to calculate the confidence interval. (&quot;None&quot;, &quot;Bonferroni&quot; or &quot;Sidak&quot;)
</p>
</td></tr>
<tr><td><code id="ErrorBarPlotPanel_+3A_colors">Colors</code></td>
<td>

<p>Colors to identyfy the groups
</p>
</td></tr>
<tr><td><code id="ErrorBarPlotPanel_+3A_title">Title</code></td>
<td>

<p>Title of the graph
</p>
</td></tr>
<tr><td><code id="ErrorBarPlotPanel_+3A_sort">sort</code></td>
<td>

<p>Should short the means before plotting
</p>
</td></tr>
<tr><td><code id="ErrorBarPlotPanel_+3A_...">...</code></td>
<td>

<p>Other graphical parameters
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The funtion plots a panel of error bars plots to compare several groups for several variables.
</p>


<h3>Value</h3>

<p>A panel of error bars plots.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ErrorBarPlotPanel(wine[4:9], wine$Group, UseANOVA=TRUE, Title="", sort=FALSE)
</code></pre>

<hr>
<h2 id='EuclideanDistance'>
Classical Euclidean Distance (Pythagorean Distance)
</h2><span id='topic+EuclideanDistance'></span>

<h3>Description</h3>

<p>Calculates the eucliden distances among the rows of an euclidean configurations in any dimensions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EuclideanDistance(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EuclideanDistance_+3A_x">x</code></td>
<td>

<p>A matrix containing the euclidean configuration
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>eucliden distances among the rows of an euclidean configurations in any dimensions</p>


<h3>Value</h3>

<p>Returns the distance matrix
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x=matrix(runif(20),10,2)
D=EuclideanDistance(x)
</code></pre>

<hr>
<h2 id='ExpandTable'>
Expands a compressed table of patterns and frequencies
</h2><span id='topic+ExpandTable'></span>

<h3>Description</h3>

<p>Expands a compressed table of patterns and frequencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExpandTable(table)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExpandTable_+3A_table">table</code></td>
<td>

<p>A compressed table of patterns and frequencies
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To simplify the calculations of some of the algorithms we compress the tables by searching for the distinct patterns and its frequencies. This function recovers the original data. It serves also to assign the corrdinates on the biplot to the original individuals.
</p>


<h3>Value</h3>

<p>A matrix with the original data
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
</code></pre>

<hr>
<h2 id='ExternalBinaryLogisticBiplot'>
External Logistic Biplot for binary Data
</h2><span id='topic+ExternalBinaryLogisticBiplot'></span>

<h3>Description</h3>

<p>Fits an External Logistic Biplot to the results of a Principal Coordinates Analysis obtained from binary data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExternalBinaryLogisticBiplot(Pco, IncludeConst=TRUE,  penalization=0.2, freq=NULL, 
tolerance = 1e-05, maxiter = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExternalBinaryLogisticBiplot_+3A_pco">Pco</code></td>
<td>

<p>An object of class &quot;Principal.Coordinates&quot;
</p>
</td></tr>
<tr><td><code id="ExternalBinaryLogisticBiplot_+3A_includeconst">IncludeConst</code></td>
<td>

<p>Should the logistic fit include the constant term?
</p>
</td></tr>
<tr><td><code id="ExternalBinaryLogisticBiplot_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the ridge regression   
</p>
</td></tr>
<tr><td><code id="ExternalBinaryLogisticBiplot_+3A_freq">freq</code></td>
<td>

<p>frequencies for each observation or pattern (usually 1)
</p>
</td></tr>
<tr><td><code id="ExternalBinaryLogisticBiplot_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence
</p>
</td></tr>
<tr><td><code id="ExternalBinaryLogisticBiplot_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let <code class="reqn">{\bf{X}}</code>  be the matrix of binary data scored as present or absent (1 or 0), in which the rows correspond to n individuals or entries (for example, genotypes) and the columns to p binary characters (for example alleles or bands), let  <code class="reqn">{\bf{S}} = ({s_{ij}})</code> be a matrix containing the similarities among rows, obtained from the binary data matrix , and let <code class="reqn">\Delta  = ({\delta _{ij}})</code>  be the corresponding dissimilarity/distance matrix, taking for example <code class="reqn">{\delta _{ij}} = \sqrt {1 - {s_{ij}}}</code>.
Despite the fact that, in Cluster Analysis and Principal Coordinates Analysis, interpretation of the variables responsible for grouping or ordination is not straightforward, those methods are normally used to classify individual in which binary variables have been measured. 
we use a combination of Principal Coordinates Analysis (PCoA), Cluster Analysis (CA) and External Logistic Regression (ELB), as a better way to interpret the binary variables associated to the classification of genotypes. The combination of three standard techniques with some new ideas about the geometry of the procedures, allows to construct a External Logistic Regression (ELB), that helps the interpretation of the variables responsible for the classification or ordination.
Suppose we have obtained an euclidean configuration <code class="reqn">{\bf{Y}}</code> obtained from the Principal Coordinates (PCoA) of the similarity matrix.
To search for the variables associated to the ordination obtained in PCoA, we can look for the directions in the ordination diagram that better predict the probability of presence of each allele.
More formally, if we defined <code class="reqn">{\pi _{ij}} = E({x_{ij}})= {\textstyle{1 \over {1 + \exp ( - ({b_{j0}} + \sum\limits_{s = 1}^k {{b_{js}}{y_{is}}} ))}}}</code> as the expected probability that the allele j be present at genotype for a genotype with coordinates <code class="reqn">y_{is}</code>  (i=1, ...,n; s=1, ..., k) on the ordination diagram, as
where bjs ( j=1,..., p) are the logistic regression coefficients that correspond to the jth variable (alleles or bands) in the sth dimension. The model is a generalized linear model having the logit as a link function.
where    and , y's and b's define a biplot in logit scale. This is called External Logistic Biplot because the coordinates of the genotypes are calculated in an external procedure (PCoA). Given that the y's are known from PCoA, obtaining the b´s is equivalent to performing a logistic regression using the j-th column of X as a response variable and the columns of y as regressors.
</p>


<h3>Value</h3>

<p>An object of class <code>External.Binary.Logistic.Biplot</code> with the fields of the <code>Principal.Coordinates</code> object with the following fields added.
</p>
<table>
<tr><td><code>ColumnParameters</code></td>
<td>
<p>Parameters resulting from fitting a logistic regression to each column of the original binary data matrix</p>
</td></tr>
<tr><td><code>VarInfo</code></td>
<td>
<p>Information of the fit for each variable</p>
</td></tr>
<tr><td><code>VarInfo$Deviances</code></td>
<td>
<p>A vector with the deviances of each variable calculated as the difference with the null model</p>
</td></tr>
<tr><td><code>VarInfo$Dfs</code></td>
<td>
<p>A vector with degrees of freedom for each variable</p>
</td></tr>
<tr><td><code>VarInfo$pvalues</code></td>
<td>
<p>A vector with the p values each variable</p>
</td></tr>
<tr><td><code>VarInfo$Nagelkerke</code></td>
<td>
<p>A vector with the Nagelkerke pseudo R-squared for each variable</p>
</td></tr>
<tr><td><code>VarInfo$PercentsCorrec</code></td>
<td>
<p>A vector with the percentage of correct classifications for each variable</p>
</td></tr>
<tr><td><code>DevianceTotal</code></td>
<td>
<p>Total Deviance as the difference with the null model</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>p value for the complete representation</p>
</td></tr>
<tr><td><code>TotalPercent</code></td>
<td>
<p>Total percentage of correct classification</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Demey, J., Vicente-Villardon, J. L., Galindo, M.P.  AND Zambrano, A. (2008) Identifying Molecular Markers Associated With Classification Of Genotypes Using External Logistic Biplots. Bioinformatics, 24(24): 2832-2838.
</p>
<p>Vicente-Villardon, J. L., Galindo, M. P. and Blazquez, A. (2006) Logistic Biplots. In Multiple Correspondence Análisis And Related Methods. Grenacre, M &amp; Blasius, J, Eds,  Chapman and Hall, Boca Raton.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
x2=Dataframe2BinaryMatrix(spiders)
colnames(x2)=colnames(spiders)
dist=BinaryProximities(x2)
pco=PrincipalCoordinates(dist)
pcobip=ExternalBinaryLogisticBiplot(pco)
</code></pre>

<hr>
<h2 id='ExtractTable'>
Extracts unique patterns and its frequencies for a discrete data matrix (numeric)
</h2><span id='topic+ExtractTable'></span>

<h3>Description</h3>

<p>Extracts the patterns and the frequencies of a discrete data matrix reducing the size of the data matrix in order to accelerate calculations in some techniques.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExtractTable(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExtractTable_+3A_x">x</code></td>
<td>

<p>A matrix of integers containing information of discrete variables. The input matrix must be numerical for the procedure to work properly.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For any numerical matrix, calculates the different patterns and the frequencies associated to each pattern
The result contains the pattern matrix, a vector with the frequencies, a list with rows sharing the same pattern.
The final pattern matrix has different ordering than the original matrix.
</p>


<h3>Value</h3>

<table>
<tr><td><code>OriginalNames</code></td>
<td>
<p>Names before grouping the equal rows</p>
</td></tr>
<tr><td><code>Patterns</code></td>
<td>
<p>The reduced table with only unique patterns</p>
</td></tr>
<tr><td><code>EqualRows</code></td>
<td>
<p>A list with as many components as unique patterns specifying the original rows with each pattern. That will allow for the reconstruction of the initial matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
spidersbin=Dataframe2BinaryMatrix(spiders)
spiderstable=ExtractTable(spidersbin)
</code></pre>

<hr>
<h2 id='FA.Biplot'>
Biplot for Factor Analysis.
</h2><span id='topic+FA.Biplot'></span>

<h3>Description</h3>

<p>Biplot used as a graphical representation of Factor Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FA.Biplot(X, dimension = 3, Extraction="PC", Rotation="varimax", 
         InitComunal="A1", normalize=FALSE, Scores= "Regression",  
         MethodArgs=NULL, sup.rows = NULL, sup.cols = NULL, ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FA.Biplot_+3A_x">X</code></td>
<td>

<p>Data Matrix
</p>
</td></tr>
<tr><td><code id="FA.Biplot_+3A_dimension">dimension</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="FA.Biplot_+3A_extraction">Extraction</code></td>
<td>

<p>Method for the extraction of the factors. Can be &quot;PC&quot;, &quot;IPF&quot; or &quot;ML&quot; (&quot;Principal Components&quot;, &quot;Iterated Principal Factor&quot; or &quot;Maximum Likelihood&quot;)
</p>
</td></tr>
<tr><td><code id="FA.Biplot_+3A_rotation">Rotation</code></td>
<td>

<p>Method for the rotation of the factors. Can be &quot;PC&quot;, &quot;IPF&quot; or &quot;ML&quot;
</p>
</td></tr>
<tr><td><code id="FA.Biplot_+3A_initcomunal">InitComunal</code></td>
<td>

<p>Initial communalities for the iterated principal factor method. Can be &quot;A1&quot;, &quot;HSC&quot; or &quot;MC&quot; (&quot;All 1&quot;, &quot;Highest Simple Correlation&quot; or &quot;Multiple Correlation&quot;)
</p>
</td></tr>
<tr><td><code id="FA.Biplot_+3A_normalize">normalize</code></td>
<td>

<p>Should the loadings be normalized
</p>
</td></tr>
<tr><td><code id="FA.Biplot_+3A_scores">Scores</code></td>
<td>

<p>Method to calculate the Row Scores. Must be &quot;Regression&quot; or &quot;Bartlett&quot;.
</p>
</td></tr>
<tr><td><code id="FA.Biplot_+3A_methodargs">MethodArgs</code></td>
<td>

<p>Aditional arguments associated to the rotation method.
</p>
</td></tr>
<tr><td><code id="FA.Biplot_+3A_sup.rows">sup.rows</code></td>
<td>

<p>Supplementary or illustrative rows, if any.
</p>
</td></tr>
<tr><td><code id="FA.Biplot_+3A_sup.cols">sup.cols</code></td>
<td>

<p>Supplementary or illustrative rows, if any.
</p>
</td></tr>
<tr><td><code id="FA.Biplot_+3A_...">...</code></td>
<td>

<p>Additional arguments for the rotation procedure.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Biplots represent the rows and columns of a data matrix in reduced dimensions. Usually rows represent  individuals, objects or samples and columns are variables measured on them. The most classical versions can be thought as visualizations associated to Principal Components Analysis (PCA) or Factor Analysis (FA) obtained from a Singular Value Decomposition or a related method. From another point of view, Classical Biplots could be obtained from regressions and calibrations that are essentially an alternated least squares algorithm equivalent to an EM-algorithm when data are normal
This routine Calculates a biplot as a graphical representation of a classical Factor Analysis alowing for different extraction methods and different rotations.
</p>


<h3>Value</h3>

<p>An object of class &quot;ContinuousBiplot&quot; with the following components:
</p>
<table>
<tr><td><code>Title</code></td>
<td>
<p>A general title</p>
</td></tr>
<tr><td><code>Non_Scaled_Data</code></td>
<td>
<p>Original Data Matrix</p>
</td></tr>
<tr><td><code>Means</code></td>
<td>
<p>Means of the original Variables</p>
</td></tr>
<tr><td><code>Medians</code></td>
<td>
<p>Medians of the original Variables</p>
</td></tr>
<tr><td><code>Deviations</code></td>
<td>
<p>Standard Deviations of the original Variables</p>
</td></tr>
<tr><td><code>Minima</code></td>
<td>
<p>Minima of the original Variables</p>
</td></tr>
<tr><td><code>Maxima</code></td>
<td>
<p>Maxima of the original Variables</p>
</td></tr>
<tr><td><code>P25</code></td>
<td>
<p>25 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>P75</code></td>
<td>
<p>75 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>Gmean</code></td>
<td>
<p>Global mean of the complete matrix</p>
</td></tr>
<tr><td><code>Sup.Rows</code></td>
<td>
<p>Supplementary rows (Non Transformed)</p>
</td></tr>
<tr><td><code>Sup.Cols</code></td>
<td>
<p>Supplementary columns (Non Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Data</code></td>
<td>
<p>Transformed Data</p>
</td></tr>
<tr><td><code>Scaled_Sup.Rows</code></td>
<td>
<p>Supplementary rows (Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Sup.Cols</code></td>
<td>
<p>Supplementary columns (Transformed)</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>Number of Rows</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>Number of Columns</p>
</td></tr>
<tr><td><code>nrowsSup</code></td>
<td>
<p>Number of Supplementary Rows</p>
</td></tr>
<tr><td><code>ncolsSup</code></td>
<td>
<p>Number of Supplementary Columns</p>
</td></tr>
<tr><td><code>dim</code></td>
<td>
<p>Dimension of the Biplot</p>
</td></tr>
<tr><td><code>EigenValues</code></td>
<td>
<p>Eigenvalues</p>
</td></tr>
<tr><td><code>Inertia</code></td>
<td>
<p>Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>CumInertia</code></td>
<td>
<p>Cumulative Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>EV</code></td>
<td>
<p>EigenVectors</p>
</td></tr>
<tr><td><code>Structure</code></td>
<td>
<p>Correlations of the Principal Components and the Variables</p>
</td></tr>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColCoordinates</code></td>
<td>
<p>Coordinates for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>RowContributions</code></td>
<td>
<p>Contributions for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColContributions</code></td>
<td>
<p>Contributions for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>Scale_Factor</code></td>
<td>
<p>Scale factor for the traditional plot with points and arrows. The row coordinates are multiplied and the column coordinates divided by that scale factor. The look of the plot is better without changing the inner product. For the HJ-Biplot the scale factor is 1.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Gabriel, K.R.(1971): The biplot graphic display of matrices with applications to principal component analysis. Biometrika,  58, 453-467.
</p>
<p>Gabriel, K. R. AND Zamir, S. (1979). Lower rank approximation of matrices by least squares with any choice of weights. Technometrics, 21(21):489&ndash;498, 1979.
</p>
<p>Gabriel, K.R.(1998): Generalised Bilinear Regression. Biometrika,  85, 3, 689-700.
</p>
<p>Gower y Hand (1996): Biplots. Chapman &amp; Hall.
</p>
<p>Vicente-Villardon, J. L., Galindo, M. P. and Blazquez-Zaballos, A. (2006). Logistic Biplots. Multiple Correspondence Analysis and related methods 491-509.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+InitialTransform">InitialTransform</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Protein)
X=Protein[,3:11]
bip=FA.Biplot(X, Extraction="ML", Rotation="oblimin")
plot(bip, mode="s", margin=0.05, AddArrow=TRUE)

</code></pre>

<hr>
<h2 id='Fact2Bin'>
Converts a Factor into its indicator matrix
</h2><span id='topic+Factor2Binary'></span>

<h3>Description</h3>

<p>Converts a factor into a binary matrix with as many columns as categories of the factor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Factor2Binary(y, Name = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Fact2Bin_+3A_y">y</code></td>
<td>

<p>A factor
</p>
</td></tr>
<tr><td><code id="Fact2Bin_+3A_name">Name</code></td>
<td>

<p>Name to use in the final matrix
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An indicator binary matrix
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>y=factor(c(1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 2, 2, 2, 1, 1, 1))
Factor2Binary(y)
</code></pre>

<hr>
<h2 id='Fraction'>
Selection of a fraction of the data
</h2><span id='topic+Fraction'></span>

<h3>Description</h3>

<p>Selects a percentage of the data eliminating the observations with higher Mahalanobis distances to the center.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Fraction(data, confidence = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Fraction_+3A_data">data</code></td>
<td>

<p>Two dimensional data set
</p>
</td></tr>
<tr><td><code id="Fraction_+3A_confidence">confidence</code></td>
<td>

<p>Percentage to retain. (0-1)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is used to select a fraction of the data to be plotted for example when clusters are used. The function eliminates the extreme values.
</p>


<h3>Value</h3>

<p>An object of class <code>fraction</code> with the following fields
</p>
<table>
<tr><td><code>data</code></td>
<td>
<p>The originaldata </p>
</td></tr>
<tr><td><code>fraction</code></td>
<td>
<p>The selected data</p>
</td></tr>
<tr><td><code>confidence</code></td>
<td>
<p>The percentage selected</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Meulman, J. J., &amp; Heiser, W. J. (1983). The display of bootstrap solutions in multidimensional scaling. Murray Hill, NJ: Bell Laboratories.
</p>
<p>Linting, M., Meulman, J. J., Groenen, P. J., &amp; Van der Kooij, A. J. (2007). Stability of nonlinear principal components analysis: An empirical study using the balanced bootstrap. Psychological Methods, 12(3), 359.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ConcEllipse">ConcEllipse</a></code>, <code><a href="#topic+AddCluster2Biplot">AddCluster2Biplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>a=matrix(runif(50), 25,2)
a2=Fraction(a, 0.7)
</code></pre>

<hr>
<h2 id='Games_Howell'>
Games-Howell post-hoc tests for Welch's one-way analysis
</h2><span id='topic+Games_Howell'></span>

<h3>Description</h3>

<p>This function produces results from Games-Howell post-hoc tests for Welch's one-way analysis of variance (ANOVA) for a matrix of numeric data and a grouping variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Games_Howell(data, group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Games_Howell_+3A_data">data</code></td>
<td>

<p>The matrix of continuous data.
</p>
</td></tr>
<tr><td><code id="Games_Howell_+3A_group">group</code></td>
<td>

<p>The grouping variable
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function produces results from Games-Howell post-hoc tests for Welch's one-way analysis of variance (ANOVA) for a matrix of numeric data and a grouping variable.
</p>


<h3>Value</h3>

<p>The tests for each column of the data matrix
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Ruxton, G. D., &amp; Beauchamp, G. (2008). Time for some a priori thinking about post hoc testing. Behavioral ecology, 19(3), 690-693.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='GD.Biplot'>
Biplot for continuous data based on gradient descent methods
</h2><span id='topic+GD.Biplot'></span>

<h3>Description</h3>

<p>Biplot for continuous data based on gradient descent methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GD.Biplot(X, dimension = 2, Scaling = 5, 
         lambda = 0.01, OptimMethod = "CG", 
         Orthogonalize = FALSE, Algorithm = "Alternated", 
         sup.rows = NULL, sup.cols = NULL,
         grouping = NULL, tolerance = 1e-04, 
         num_max_iters = 300, Initial = "random")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GD.Biplot_+3A_x">X</code></td>
<td>

<p>A data matrix with continuous variables.
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_dimension">dimension</code></td>
<td>

<p>Dimension of the final solution.
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_scaling">Scaling</code></td>
<td>

<p>Transformation of the raw data matrix before the calculation of the biplot.
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_lambda">lambda</code></td>
<td>

<p>Constant for the ridge Penalization
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_optimmethod">OptimMethod</code></td>
<td>

<p>Optimization method passed to the <code>optim</code> function. By default is CG (Conjugate Gradient).
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_orthogonalize">Orthogonalize</code></td>
<td>

<p>Should the solution be ortogonalized.
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_algorithm">Algorithm</code></td>
<td>

<p>Algorithm to calculate the Biplot. (Alternated, Joint, Recursive)
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_sup.rows">sup.rows</code></td>
<td>

<p>Supplementary Rows. (not working now)
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_sup.cols">sup.cols</code></td>
<td>

<p>Supplementary Columns. (not working now)
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_grouping">grouping</code></td>
<td>

<p>Grouping factor for the within groups transformation.
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_num_max_iters">num_max_iters</code></td>
<td>

<p>Maximum number of iterations.
</p>
</td></tr>
<tr><td><code id="GD.Biplot_+3A_initial">Initial</code></td>
<td>

<p>Initial Configuration 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates a bilot using gradient descent methods. The function <code>optim</code> is used to optimize the loss function. By default CG (Conjugate Gradient) method is used althoug other possibilities can be used.
</p>


<h3>Value</h3>

<p>An object of class &quot;ContinuousBiplot&quot; is returned.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Protein")
X=Protein[,3:11]
gdbip=GD.Biplot(X, dimension=2, Algorithm="Joint", 
Orthogonalize=FALSE, lambda=0.3, Initial="random")
plot(gdbip)
summary(gdbip)
</code></pre>

<hr>
<h2 id='GeneralizedProcrustes'>Generalized Procrustes Analysis</h2><span id='topic+GeneralizedProcrustes'></span>

<h3>Description</h3>

<p>Generalized Procrustes Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  GeneralizedProcrustes(x, tolerance = 1e-05, maxiter = 100, Plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GeneralizedProcrustes_+3A_x">x</code></td>
<td>

<p>Three dimensional array with the configurations. The first dimension contains the rows of the configurations,
the second contains the columns and the third the number of configurations. So x[,,i] is the <em>i-th</em>
configuration
</p>
</td></tr>
<tr><td><code id="GeneralizedProcrustes_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for the Procrustes algorithm. 
</p>
</td></tr>
<tr><td><code id="GeneralizedProcrustes_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations
</p>
</td></tr>
<tr><td><code id="GeneralizedProcrustes_+3A_plot">Plot</code></td>
<td>

<p>Should the results be plotted?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Generalized Procrustes Analysis for several configurations contained in a three-dimensional
array.
</p>


<h3>Value</h3>

<p>An object of class <code>GenProcustes</code>.This has components:
</p>
<table>
<tr><td><code>History</code></td>
<td>
<p>History of Iterations</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>Initial configurations in a three dimensional array</p>
</td></tr>
<tr><td><code>RotatedX</code></td>
<td>
<p>Transformed configurations in a three dimensional array</p>
</td></tr>
<tr><td><code>Scale</code></td>
<td>
<p>Scale factors for each configuration</p>
</td></tr>
<tr><td><code>Rotations</code></td>
<td>
<p>Rotation Matrices in a three dimensional array</p>
</td></tr>
<tr><td><code>rss</code></td>
<td>
<p>Residual Sum of Squares</p>
</td></tr>
<tr><td><code>Fit</code></td>
<td>
<p>Goodness of fit as percent of expained variance</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Gower, J.C., (1975). Generalised Procrustes analysis. Psychometrika 40, 33-51.
</p>
<p>Ingwer Borg, I. &amp; Groenen, P. J.F. (2005). Modern Multidimensional Scaling. Theory and Applications. Second Edition. Springer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PrincipalCoordinates">PrincipalCoordinates</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
n=dim(spiders)[1]
p=dim(spiders)[2]
prox=array(0,c(n,2,4))

p1=BinaryProximities(spiders,coefficient=5)
prox[,,1]=PrincipalCoordinates(p1)$RowCoordinates
p2=BinaryProximities(spiders,coefficient=2)
prox[,,2]=PrincipalCoordinates(p2)$RowCoordinates
p3=BinaryProximities(spiders,coefficient=3)
prox[,,3]=PrincipalCoordinates(p3)$RowCoordinates
p4=BinaryProximities(spiders,coefficient=4)
prox[,,4]=PrincipalCoordinates(p4)$RowCoordinates
GeneralizedProcrustes(prox)
</code></pre>

<hr>
<h2 id='GetBiplotScales'>
Calculates the scales for the variables on a linear biplot
</h2><span id='topic+GetBiplotScales'></span>

<h3>Description</h3>

<p>Calculates the scales for the variables on a linear prediction biplot
There are several types of scales and values that can be shown on the graphical representation. See details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetBiplotScales(Biplot, nticks = 3, TypeScale = "Complete", ValuesScale = "Original")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetBiplotScales_+3A_biplot">Biplot</code></td>
<td>

<p>Object of class PCA.Biplot
</p>
</td></tr>
<tr><td><code id="GetBiplotScales_+3A_nticks">nticks</code></td>
<td>

<p>Number of ticks for the biplot axes
</p>
</td></tr>
<tr><td><code id="GetBiplotScales_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scale to use : &quot;Complete&quot;, &quot;StdDev&quot; or &quot;BoxPlot&quot;
</p>
</td></tr>
<tr><td><code id="GetBiplotScales_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>Values to show on the scale: &quot;Original&quot; or &quot;Transformed&quot;
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates the points on the biplot axes where the scales should be placed.
</p>
<p>There are three types of scales when the transformations of the raw data are made by columns:
</p>
<p>&quot;Complete&quot;: Covers the whole range of the variable using the number of ticks specified in &quot;nticks&quot;. A smaller number of points could be shown if some fall outsite the range of the scatter.
</p>
<p>&quot;StdDev&quot;: The mean +/- 1, 2 and 3 times the standard deviation.A smaller number of points could be shown if some fall outsite the range of the scatter.
</p>
<p>&quot;BoxPlot&quot;: Median, 25, 75 percentiles maximum and minimum values are shown. The extremes of the interquartile range are connected with a thicker line. A smaller number of points could be shown if some fall outsite the range of the scatter.
</p>
<p>There are two kinds of values that can be shown on the biplot axis:
</p>
<p>&quot;Original&quot;: The values before transformation. Only makes sense when the transformations are for each column.
</p>
<p>&quot;Transformed&quot;: The values after transformation, for example, after standardization. 
</p>
<p>Although the function is public, the end used will not normally use it.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>Ticks</code></td>
<td>
<p>A list containing the ticks for each variable</p>
</td></tr>
<tr><td><code>Labels</code></td>
<td>
<p>A list containing the labels for each variable</p>
</td></tr>

</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
bip=PCA.Biplot(iris[,1:4])
GetBiplotScales(bip)
</code></pre>

<hr>
<h2 id='GetCCAScales'>
Calculates scales for plotting the environmental variables in a Canonical Correspondence Analysis
</h2><span id='topic+GetCCAScales'></span>

<h3>Description</h3>

<p>Calculates scales for plotting the environmental variables in a Canonical Correspondence Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetCCAScales(CCA, nticks = 7, TypeScale = "Complete", ValuesScale = "Original")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetCCAScales_+3A_cca">CCA</code></td>
<td>

<p>A CCA solution object
</p>
</td></tr>
<tr><td><code id="GetCCAScales_+3A_nticks">nticks</code></td>
<td>

<p>Number of ticks to represent
</p>
</td></tr>
<tr><td><code id="GetCCAScales_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scale to represent
</p>
</td></tr>
<tr><td><code id="GetCCAScales_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>Values to represent (Original or Transformed)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates scales for plotting the environmental variables in a Canonical Correspondence Analysis
</p>


<h3>Value</h3>

<p>Returns the points and the labels for each biplot axis
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Gower, J. C., &amp; Hand, D. J. (1995). Biplots (Vol. 54). CRC Press.
</p>
<p>Gower, J. C., Lubbe, S. G., &amp; Le Roux, N. J. (2011). Understanding biplots. John Wiley &amp; Sons.
</p>
<p>Vicente-Villardón, J. L., Galindo Villardón, M. P., &amp; Blázquez Zaballos, A. (2006). Logistic biplots. Multiple correspondence analysis and related methods. London: Chapman &amp; Hall, 503-521.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># No examples yet
</code></pre>

<hr>
<h2 id='ginv'>
G inverse
</h2><span id='topic+ginv'></span>

<h3>Description</h3>

<p>Calculates the g-inverse of a squared matrix using the eigen decomposition and removing the eigenvalues smaller than a tolerance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ginv(X, tol = sqrt(.Machine$double.eps))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ginv_+3A_x">X</code></td>
<td>

<p>Matrix to calculate the g-inverse
</p>
</td></tr>
<tr><td><code id="ginv_+3A_tol">tol</code></td>
<td>

<p>Tolerance. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is useful to avoid singularities.
</p>


<h3>Value</h3>

<p>Returns the g-inverse
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
x=as.matrix(iris[,1:4])
S= t(x) 
ginv(S)

</code></pre>

<hr>
<h2 id='GowerProximities'>Gower Dissimilarities for mixed types of data</h2><span id='topic+GowerProximities'></span>

<h3>Description</h3>

<p>Gower Dissimilarities for mixed types of data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  GowerProximities(x, y = NULL, Binary = NULL, Classes = NULL,
                   transformation = 3, IntegerAsOrdinal = FALSE, BinCoef
                   = "Simple_Matching", ContCoef = "Gower", NomCoef =
                   "GOW", OrdCoef = "GOW")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GowerProximities_+3A_x">x</code></td>
<td>

<p>Main data. Distances among rows are calculated if y=NULL. Must be a data frame.
</p>
</td></tr>
<tr><td><code id="GowerProximities_+3A_y">y</code></td>
<td>

<p>Suplementary data matrix. If not NULL the distances among the rows of x and y are calculated. Must be a data frame with the same columns as x.
</p>
</td></tr>
<tr><td><code id="GowerProximities_+3A_binary">Binary</code></td>
<td>

<p>A vector containing the binary variables.
</p>
</td></tr>
<tr><td><code id="GowerProximities_+3A_classes">Classes</code></td>
<td>

<p>Vector with column types. If NULL the data frame types are used.
</p>
</td></tr>
<tr><td><code id="GowerProximities_+3A_transformation">transformation</code></td>
<td>

<p>Transformation for the similarities.
</p>
</td></tr>
<tr><td><code id="GowerProximities_+3A_integerasordinal">IntegerAsOrdinal</code></td>
<td>

<p>Should integer variables be used as ordinal?
</p>
</td></tr>
<tr><td><code id="GowerProximities_+3A_bincoef">BinCoef</code></td>
<td>

<p>Coefficient for the binary data
</p>
</td></tr>
<tr><td><code id="GowerProximities_+3A_contcoef">ContCoef</code></td>
<td>

<p>Coefficient for the continuous data
</p>
</td></tr>
<tr><td><code id="GowerProximities_+3A_nomcoef">NomCoef</code></td>
<td>

<p>Coefficient for the nominal data
</p>
</td></tr>
<tr><td><code id="GowerProximities_+3A_ordcoef">OrdCoef</code></td>
<td>

<p>Coefficient for the ordinal data
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The transformation <code>sqrt(1-S)</code> is applied to the similarity.
</p>


<h3>Value</h3>

<p>An object of class <code>proximities</code>.This has components:
</p>
<table>
<tr><td><code>comp1</code></td>
<td>
<p>Description of </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>J. C. Gower. (1971) A General Coefficient of Similarity and Some of its Properties. Biometrics, Vol. 27, No. 4, pp. 857-871.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
</code></pre>

<hr>
<h2 id='GowerSimilarities'>Gower Dissimilarities for mixed types of data</h2><span id='topic+GowerSimilarities'></span>

<h3>Description</h3>

<p>Gower Dissimilarities for mixed types of data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  GowerSimilarities(x, y = NULL, Classes = NULL, transformation =
                   "sqrt(1-S)", BinCoef = "Simple_Matching", ContCoef =
                   "Gower", NomCoef = "GOW", OrdCoef = "GOW")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GowerSimilarities_+3A_x">x</code></td>
<td>

<p>Main data. Distances among rows are calculated if y=NULL. Must be a data frame.
</p>
</td></tr>
<tr><td><code id="GowerSimilarities_+3A_y">y</code></td>
<td>

<p>Suplementary data matrix. If not NULL the distances among the rows of x and y are calculated. Must be a data frame with the same columns as x.
</p>
</td></tr>
<tr><td><code id="GowerSimilarities_+3A_classes">Classes</code></td>
<td>

<p>Vector containing the classes of each variable.
</p>
</td></tr>
<tr><td><code id="GowerSimilarities_+3A_transformation">transformation</code></td>
<td>

<p>Transformation to apply to the similarities.
</p>
</td></tr>
<tr><td><code id="GowerSimilarities_+3A_bincoef">BinCoef</code></td>
<td>

<p>Coefficient for the binary data
</p>
</td></tr>
<tr><td><code id="GowerSimilarities_+3A_contcoef">ContCoef</code></td>
<td>

<p>Coefficient for the continuous data
</p>
</td></tr>
<tr><td><code id="GowerSimilarities_+3A_nomcoef">NomCoef</code></td>
<td>

<p>Coefficient for the nominal data
</p>
</td></tr>
<tr><td><code id="GowerSimilarities_+3A_ordcoef">OrdCoef</code></td>
<td>

<p>Coefficient for the ordinal data
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Gower Dissimilarities for mixed types of data.
The  transformation <code>sqrt(1-S)</code> is applied to the similarity by default.
</p>


<h3>Value</h3>

<p>An object of class <code>proximities</code>.This has components:
</p>
<table>
<tr><td><code>comp1</code></td>
<td>
<p>Description of </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>J. C. Gower. (1971) A General Coefficient of Similarity and Some of its Properties. Biometrics, Vol. 27, No. 4, pp. 857-871.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
</code></pre>

<hr>
<h2 id='Hermquad'>
Gauss-Hermite quadrature
</h2><span id='topic+Hermquad'></span>

<h3>Description</h3>

<p>Find the Gauss-Hermite abscissae and weights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Hermquad(N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Hermquad_+3A_n">N</code></td>
<td>

<p>Number of nodes of the quadrature
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Find the Gauss-Hermite abscissae and weights.
</p>


<h3>Value</h3>

<table>
<tr><td><code>X</code></td>
<td>
<p>A column vector containing the abscissae.</p>
</td></tr>
<tr><td><code>W</code></td>
<td>
<p>A vector containing the corresponding weights.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon (translated from a Matlab function by  Greg von Winckel) 
)
</p>


<h3>References</h3>

<p>Press, W. H., Teukolsky, S. A., Vetterling, W. T., &amp; Flannery, B. P. (1992). Numerical Recipes in C: The Art of Scientific Computing (New York. Cambridge University Press, 636-9.
</p>
<p>http://www.mathworks.com/matlabcentral/fileexchange/8836-hermite-quadrature/content/hermquad.m
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Hermquad(10)
</code></pre>

<hr>
<h2 id='HistogramPanel'>
Panel of histograms
</h2><span id='topic+HistogramPanel'></span>

<h3>Description</h3>

<p>Panel of histograms for a set of numerical variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HistogramPanel(X, nrows = NULL, separated = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HistogramPanel_+3A_x">X</code></td>
<td>

<p>The matrix of continuous variables
</p>
</td></tr>
<tr><td><code id="HistogramPanel_+3A_nrows">nrows</code></td>
<td>

<p>Number of rows of the panel.
</p>
</td></tr>
<tr><td><code id="HistogramPanel_+3A_separated">separated</code></td>
<td>

<p>Should the plots be organized into a panel? (or separated)
</p>
</td></tr>
<tr><td><code id="HistogramPanel_+3A_...">...</code></td>
<td>

<p>Aditional graphical arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Value</h3>

<p>The histogram panel.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
HistogramPanel(wine[,4:7], nrows = 2, xlab="")

</code></pre>

<hr>
<h2 id='HJ.Biplot'>
HJ Biplot with added features.
</h2><span id='topic+HJ.Biplot'></span>

<h3>Description</h3>

<p>HJ Biplot with added features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HJ.Biplot(X, dimension = 3, Scaling = 5, sup.rows = NULL, 
         sup.cols = NULL, grouping = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HJ.Biplot_+3A_x">X</code></td>
<td>

<p>Data Matrix
</p>
</td></tr>
<tr><td><code id="HJ.Biplot_+3A_dimension">dimension</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="HJ.Biplot_+3A_scaling">Scaling</code></td>
<td>

<p>Transformation of the original data. See InitialTransform for available transformations.
</p>
</td></tr>
<tr><td><code id="HJ.Biplot_+3A_sup.rows">sup.rows</code></td>
<td>

<p>Supplementary or illustrative rows, if any.
</p>
</td></tr>
<tr><td><code id="HJ.Biplot_+3A_sup.cols">sup.cols</code></td>
<td>

<p>Supplementary or illustrative rows, if any.
</p>
</td></tr>
<tr><td><code id="HJ.Biplot_+3A_grouping">grouping</code></td>
<td>

<p>factor to stadadize with the within groups variability
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Biplots represent the rows and columns of a data matrix in reduced dimensions. Usually rows represent  individuals, objects or samples and columns are variables measured on them. The most classical versions can be thought as visualizations associated to Principal Components Analysis (PCA) or Factor Analysis (FA) obtained from a Singular Value Decomposition or a related method. From another point of view, Classical Biplots could be obtained from regressions and calibrations that are essentially an alternated least squares algorithm equivalent to an EM-algorithm when data are normal.
</p>


<h3>Value</h3>

<p>An object of class ContinuousBiplot with the following components:
</p>
<table>
<tr><td><code>Title</code></td>
<td>
<p>A general title</p>
</td></tr>
<tr><td><code>Non_Scaled_Data</code></td>
<td>
<p>Original Data Matrix</p>
</td></tr>
<tr><td><code>Means</code></td>
<td>
<p>Means of the original Variables</p>
</td></tr>
<tr><td><code>Medians</code></td>
<td>
<p>Medians of the original Variables</p>
</td></tr>
<tr><td><code>Deviations</code></td>
<td>
<p>Standard Deviations of the original Variables</p>
</td></tr>
<tr><td><code>Minima</code></td>
<td>
<p>Minima of the original Variables</p>
</td></tr>
<tr><td><code>Maxima</code></td>
<td>
<p>Maxima of the original Variables</p>
</td></tr>
<tr><td><code>P25</code></td>
<td>
<p>25 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>P75</code></td>
<td>
<p>75 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>Gmean</code></td>
<td>
<p>Global mean of the complete matrix</p>
</td></tr>
<tr><td><code>Sup.Rows</code></td>
<td>
<p>Supplementary rows (Non Transformed)</p>
</td></tr>
<tr><td><code>Sup.Cols</code></td>
<td>
<p>Supplementary columns (Non Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Data</code></td>
<td>
<p>Transformed Data</p>
</td></tr>
<tr><td><code>Scaled_Sup.Rows</code></td>
<td>
<p>Supplementary rows (Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Sup.Cols</code></td>
<td>
<p>Supplementary columns (Transformed)</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>Number of Rows</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>Number of Columns</p>
</td></tr>
<tr><td><code>nrowsSup</code></td>
<td>
<p>Number of Supplementary Rows</p>
</td></tr>
<tr><td><code>ncolsSup</code></td>
<td>
<p>Number of Supplementary Columns</p>
</td></tr>
<tr><td><code>dim</code></td>
<td>
<p>Dimension of the Biplot</p>
</td></tr>
<tr><td><code>EigenValues</code></td>
<td>
<p>Eigenvalues</p>
</td></tr>
<tr><td><code>Inertia</code></td>
<td>
<p>Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>CumInertia</code></td>
<td>
<p>Cumulative Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>EV</code></td>
<td>
<p>EigenVectors</p>
</td></tr>
<tr><td><code>Structure</code></td>
<td>
<p>Correlations of the Principal Components and the Variables</p>
</td></tr>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColCoordinates</code></td>
<td>
<p>Coordinates for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>RowContributions</code></td>
<td>
<p>Contributions for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColContributions</code></td>
<td>
<p>Contributions for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>Scale_Factor</code></td>
<td>
<p>Scale factor for the traditional plot with points and arrows. The row coordinates are multiplied and the column coordinates divided by that scale factor. The look of the plot is better without changing the inner product. For the HJ-Biplot the scale factor is 1.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Galindo Villardon, M. (1986). Una alternativa de representacion simultanea: HJ-Biplot. Questiio. 1986, vol. 10, núm. 1.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+InitialTransform">InitialTransform</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Simple Biplot with arrows
data(Protein)
bip=HJ.Biplot(Protein[,3:11])
plot(bip)

</code></pre>

<hr>
<h2 id='InBox'>
Checks if a point is inside a box.
</h2><span id='topic+InBox'></span>

<h3>Description</h3>

<p>Checks if a point is inside a box. The point is specified bi its x and y coordinates and the bom with the minimum and maximum values on both coordinate axis: xmin, xmax, ymin, ymax. The vertices of the box are then (xmin, ymin), (xmax, ymin), (xmax, ymax) and (xmin, ymax)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InBox(x, y, xmin, xmax, ymin, ymax)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InBox_+3A_x">x</code></td>
<td>

<p>x coordinate of the point
</p>
</td></tr>
<tr><td><code id="InBox_+3A_y">y</code></td>
<td>

<p>x coordinate of the point
</p>
</td></tr>
<tr><td><code id="InBox_+3A_xmin">xmin</code></td>
<td>

<p>minimum value of X
</p>
</td></tr>
<tr><td><code id="InBox_+3A_xmax">xmax</code></td>
<td>

<p>maximum value of X
</p>
</td></tr>
<tr><td><code id="InBox_+3A_ymin">ymin</code></td>
<td>

<p>minimum value of Y
</p>
</td></tr>
<tr><td><code id="InBox_+3A_ymax">ymax</code></td>
<td>

<p>maximum value of Y
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a logical value : TRUE if the point is inside the box and FALSE otherwise.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>InBox(0, 0, -1, 1, -1, 1) 
</code></pre>

<hr>
<h2 id='InitialTransform'>
Initial transformation of data
</h2><span id='topic+InitialTransform'></span>

<h3>Description</h3>

<p>Initial transformation of data before the construction of a biplot. (or any other technique)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InitialTransform(X, sup.rows = NULL, sup.cols = NULL, 
InitTransform = "None", transform = "Standardize columns", 
grouping = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InitialTransform_+3A_x">X</code></td>
<td>

<p>Original Raw Data Matrix
</p>
</td></tr>
<tr><td><code id="InitialTransform_+3A_sup.rows">sup.rows</code></td>
<td>

<p>Supplementary or illustrative rows. 
</p>
</td></tr>
<tr><td><code id="InitialTransform_+3A_sup.cols">sup.cols</code></td>
<td>

<p>Supplementary or illustrative columns. 
</p>
</td></tr>
<tr><td><code id="InitialTransform_+3A_inittransform">InitTransform</code></td>
<td>

<p>Pevious transformation (to use. See details)none or log.
</p>
</td></tr>
<tr><td><code id="InitialTransform_+3A_transform">transform</code></td>
<td>

<p>Transformation to use. See details.
</p>
</td></tr>
<tr><td><code id="InitialTransform_+3A_grouping">grouping</code></td>
<td>

<p>factor to stadadize with the within groups variability
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Possible Transformations are:
</p>
<p>1.- &quot;Raw Data&quot;: When no transformation is required.
</p>
<p>2.- &quot;Substract the global mean&quot;: Eliminate an eefect common to all the observations
</p>
<p>3.- &quot;Double centering&quot; : Interaction residuals. When all the elements of the table are comparable. Useful for AMMI models.
</p>
<p>4.- &quot;Column centering&quot;: Remove the column means.
</p>
<p>5.- &quot;Standardize columns&quot;:  Remove the column means and divide by its standard deviation.
</p>
<p>6.- &quot;Row centering&quot;: Remove the row means.
</p>
<p>7.- &quot;Standardize rows&quot;: Divide each row by its standard deviation.
</p>
<p>8.- &quot;Divide by the column means and center&quot;: The resulting dispersion is the coefficient of variation.
</p>
<p>9.- &quot;Normalized residuals from independence&quot; for a contingency table.
</p>
<p>The transformation can be provided to the function by using the string beetwen the quotes or just the associated number.
</p>
<p>The supplementary rows and columns are not used to calculate the parameters (means, standard deviations, etc). Some of the transformations are not compatible with supplementary data. 
</p>


<h3>Value</h3>

<p>A list with the following components
</p>
<table>
<tr><td><code>X</code></td>
<td>
<p>Transformed data matrix</p>
</td></tr>
<tr><td><code>sup.rows</code></td>
<td>
<p>Transformed supplementary rows</p>
</td></tr>
<tr><td><code>sup.rows</code></td>
<td>
<p>Transformed supplementary columns</p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>M. J. Baxter (1995) Standardization and Transformation in Principal Component Analysis, with Applications to Archaeometry. 
Journal of the Royal Statistical Society. Series C (Applied Statistics). Vol. 44, No. 4 (1995) , pp. 513-527
</p>
<p>Kroonenberg, P. M. (1983). Three-mode principal component analysis: Theory and applications (Vol. 2). DSWO press. (Chapter 6)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
x=as.matrix(iris[,1:4])
x=InitialTransform(x, transform=4)
x
</code></pre>

<hr>
<h2 id='Integer2Binary'>Transforms an Integer Variable into a Binary Variable</h2><span id='topic+Integer2Binary'></span>

<h3>Description</h3>

<p>Transforms an Integer Variable into a Binary Variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  Integer2Binary(y, name = "My_Factor")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Integer2Binary_+3A_y">y</code></td>
<td>

<p>Vector with the factor
</p>
</td></tr>
<tr><td><code id="Integer2Binary_+3A_name">name</code></td>
<td>

<p>name of the factor
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Transforms an Integer vector into a Binary Indicator Matrix
</p>


<h3>Value</h3>

<p>A Binary Data Matrix
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dat=c(1, 2, 2, 4, 1, 1, 4, 2, 4)
Integer2Binary(dat,"Myfactor")
</code></pre>

<hr>
<h2 id='Kruskal.Wallis.Tests'>
Kruskal Wallis Tests
</h2><span id='topic+Kruskal.Wallis.Tests'></span>

<h3>Description</h3>

<p>Kruskal Wallis Tests for a matrix of continuous variables and a grouping factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Kruskal.Wallis.Tests(X, groups, posthoc = "none", alternative = "two.sided", digits = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Kruskal.Wallis.Tests_+3A_x">X</code></td>
<td>

<p>The matrix of continuous variables
</p>
</td></tr>
<tr><td><code id="Kruskal.Wallis.Tests_+3A_groups">groups</code></td>
<td>

<p>The factor with the groups
</p>
</td></tr>
<tr><td><code id="Kruskal.Wallis.Tests_+3A_posthoc">posthoc</code></td>
<td>

<p>Method used for multipe comparisons in the Dunn test
</p>
</td></tr>
<tr><td><code id="Kruskal.Wallis.Tests_+3A_alternative">alternative</code></td>
<td>

<p>Kind of alternative hypothesis
</p>
</td></tr>
<tr><td><code id="Kruskal.Wallis.Tests_+3A_digits">digits</code></td>
<td>

<p>number of digitd for he output
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Kruskal Wallis Tests for a matrix of continuous variables and a grouping factor, including the Dunn test for multiple comparisons.
</p>


<h3>Value</h3>

<p>the organized output.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
Kruskal.Wallis.Tests(wine[,4:7], wine$Group, posthoc = "bonferroni")
</code></pre>

<hr>
<h2 id='Levene.Tests'>
Levene Tests 
</h2><span id='topic+Levene.Tests'></span>

<h3>Description</h3>

<p>Levene Tests  for a matrix of continuous variables and a grouping factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Levene.Tests(X, groups = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Levene.Tests_+3A_x">X</code></td>
<td>

<p>The matrix of continuous variables
</p>
</td></tr>
<tr><td><code id="Levene.Tests_+3A_groups">groups</code></td>
<td>

<p>The factor with the groups
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Levene Tests  for a matrix of continuous variables and a grouping factor.
</p>


<h3>Value</h3>

<p>The organized output
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
Levene.Tests(wine[,4:7], wine$Group)
</code></pre>

<hr>
<h2 id='LogFrequencyBiplot'>
Weighted Biplot for a table of frequencies 
</h2><span id='topic+LogFrequencyBiplot'></span>

<h3>Description</h3>

<p>Biplot for the logarithms of the frequencies of a contingency table using the frequencies as weights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LogFrequencyBiplot(x, Scaling = 2, logoffset = 1, freqoffset = logoffset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LogFrequencyBiplot_+3A_x">x</code></td>
<td>

<p>The frequency table to be biplotted
</p>
</td></tr>
<tr><td><code id="LogFrequencyBiplot_+3A_scaling">Scaling</code></td>
<td>

<p>Transformation of the matrix after the logarithms
</p>
</td></tr>
<tr><td><code id="LogFrequencyBiplot_+3A_logoffset">logoffset</code></td>
<td>

<p>Constant to add to the frequencies before calculating the logarithms. This is to avoid calculating the logaritm of zero, so, a covenient value for this argument is 1.
</p>
</td></tr>
<tr><td><code id="LogFrequencyBiplot_+3A_freqoffset">freqoffset</code></td>
<td>

<p>Constant to add to the frequencies before calculating the weigths. This is usually the same as the offset used to add to the frequencies but may be different when we do not want the frequencies zero to influence the biplot, i. e., we want zero weigths.
</p>
</td></tr>
<tr><td><code id="LogFrequencyBiplot_+3A_...">...</code></td>
<td>

<p>Any other parameter for the CrissCross procedure.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Biplot for the logarithms of the frequencies of a contingency table using the frequencies as weigths.
</p>


<h3>Value</h3>

<p>An object of class .Biplot&quot; with the following components:
</p>
<table>
<tr><td><code>Title</code></td>
<td>
<p>A general title</p>
</td></tr>
<tr><td><code>Non_Scaled_Data</code></td>
<td>
<p>Original Data Matrix</p>
</td></tr>
<tr><td><code>Means</code></td>
<td>
<p>Means of the original Variables</p>
</td></tr>
<tr><td><code>Medians</code></td>
<td>
<p>Medians of the original Variables</p>
</td></tr>
<tr><td><code>Deviations</code></td>
<td>
<p>Standard Deviations of the original Variables</p>
</td></tr>
<tr><td><code>Minima</code></td>
<td>
<p>Minima of the original Variables</p>
</td></tr>
<tr><td><code>Maxima</code></td>
<td>
<p>Maxima of the original Variables</p>
</td></tr>
<tr><td><code>P25</code></td>
<td>
<p>25 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>P75</code></td>
<td>
<p>75 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>Gmean</code></td>
<td>
<p>Global mean of the complete matrix</p>
</td></tr>
<tr><td><code>Sup.Rows</code></td>
<td>
<p>Supplementary rows (Non Transformed)</p>
</td></tr>
<tr><td><code>Sup.Cols</code></td>
<td>
<p>Supplementary columns (Non Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Data</code></td>
<td>
<p>Transformed Data</p>
</td></tr>
<tr><td><code>Scaled_Sup.Rows</code></td>
<td>
<p>Supplementary rows (Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Sup.Cols</code></td>
<td>
<p>Supplementary columns (Transformed)</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>Number of Rows</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>Number of Columns</p>
</td></tr>
<tr><td><code>nrowsSup</code></td>
<td>
<p>Number of Supplementary Rows</p>
</td></tr>
<tr><td><code>ncolsSup</code></td>
<td>
<p>Number of Supplementary Columns</p>
</td></tr>
<tr><td><code>dim</code></td>
<td>
<p>Dimension of the Biplot</p>
</td></tr>
<tr><td><code>EigenValues</code></td>
<td>
<p>Eigenvalues</p>
</td></tr>
<tr><td><code>Inertia</code></td>
<td>
<p>Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>CumInertia</code></td>
<td>
<p>Cumulative Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>EV</code></td>
<td>
<p>EigenVectors</p>
</td></tr>
<tr><td><code>Structure</code></td>
<td>
<p>Correlations of the Principal Components and the Variables</p>
</td></tr>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColCoordinates</code></td>
<td>
<p>Coordinates for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>RowContributions</code></td>
<td>
<p>Contributions for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColContributions</code></td>
<td>
<p>Contributions for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>Scale_Factor</code></td>
<td>
<p>Scale factor for the traditional plot with points and arrows. The row coordinates are multiplied and the column coordinates divided by that scale factor. The look of the plot is better without changing the inner product. For the HJ-Biplot the scale factor is 1.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Gabriel, K. R., Galindo, M. P. &amp; Vicente-Villardon, J. L. (1995) Use of Biplots to Diagnose Independence Models in Three-Way Contingency Tables. in: M. Greenacre &amp; J. Blasius. eds. Visualization of Categorical Data. Academis Press. London.
</p>
<p>GABRIEL, K.R. and ZAMIR, S. (1979). Lower rank approximation of matrices by least squares with any choice of weights. Technometrics, 21: 489-498.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CrissCross">CrissCross</a></code>, ~~~
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(smoking)
logbip=LogFrequencyBiplot(smoking, Scaling=1, logoffset=0, freqoffset=0)
</code></pre>

<hr>
<h2 id='logit'>
Logit function
</h2><span id='topic+logit'></span>

<h3>Description</h3>

<p>Calculates the logit of a probability
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logit(p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logit_+3A_p">p</code></td>
<td>

<p>A probability
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates the logit of a probability
</p>


<h3>Value</h3>

<p>The lo git of the provided probebility
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardón
</p>

<hr>
<h2 id='Matrix2Proximities'>
Matrix to Proximities
</h2><span id='topic+Matrix2Proximities'></span>

<h3>Description</h3>

<p>Converts a matrix of proximities into a Proximities object as used in Principal Coordinates or MDS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Matrix2Proximities(x, TypeData = "User Provided", 
Type = c("dissimilarity", "similarity", "products"), 
Coefficient = "None", Transformation = "None", Data = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Matrix2Proximities_+3A_x">x</code></td>
<td>

<p>The matrix of proximities (a symmetrical matrix)
</p>
</td></tr>
<tr><td><code id="Matrix2Proximities_+3A_typedata">TypeData</code></td>
<td>

<p>By default is User provided but could be any type.
</p>
</td></tr>
<tr><td><code id="Matrix2Proximities_+3A_type">Type</code></td>
<td>

<p>Type of proximity: dissimilarity, similarity or scalar product. If not provided, the default is dissimilarity
</p>
</td></tr>
<tr><td><code id="Matrix2Proximities_+3A_coefficient">Coefficient</code></td>
<td>

<p>Name of the procedure to calculate the proximities (if any).
</p>
</td></tr>
<tr><td><code id="Matrix2Proximities_+3A_transformation">Transformation</code></td>
<td>

<p>Transformation used to calculate dissimilarities from similarities (if any)
</p>
</td></tr>
<tr><td><code id="Matrix2Proximities_+3A_data">Data</code></td>
<td>

<p>Raw data used to calculate the proximity (if any).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Converts a matrix of proximities into a Proximities object as used in Principal Coordinates or MDS aading some extra information about the procedure used to obtain the proximities. Is mainly used when the proximities matrix has been provided by the user and not calculated from raw data using BinaryProximities, ContinuousDistances or any other function.
</p>


<h3>Value</h3>

<p>An object of class <code>Proximities</code> containing the proximities matrix and some extra information about it.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>

<hr>
<h2 id='matrixsqrt'>
Matrix squared root
</h2><span id='topic+matrixsqrt'></span>

<h3>Description</h3>

<p>Matrix square root of a matrix using the eigendecomposition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrixsqrt(S, tol = sqrt(.Machine$double.eps))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matrixsqrt_+3A_s">S</code></td>
<td>

<p>A squered matrix
</p>
</td></tr>
<tr><td><code id="matrixsqrt_+3A_tol">tol</code></td>
<td>

<p>Tolerance for the igenvalues
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Matrix square root of a matrix using the eigendecomposition and removing the eigenvalues smaller than a tolerance
</p>


<h3>Value</h3>

<p>The matrix square root of the argument
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
x=as.matrix(iris[,1:4])
S= t(x) 
matrixsqrt(S)
</code></pre>

<hr>
<h2 id='matrixsqrtinv'>
Inverse of the Matrix squared root
</h2><span id='topic+matrixsqrtinv'></span>

<h3>Description</h3>

<p>Inverse of the Matrix square root of a matrix using the eigendecomposition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrixsqrtinv(S, tol = sqrt(.Machine$double.eps))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matrixsqrtinv_+3A_s">S</code></td>
<td>

<p>A squered matrix
</p>
</td></tr>
<tr><td><code id="matrixsqrtinv_+3A_tol">tol</code></td>
<td>

<p>Tolerance for the igenvalues
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Inverse of the  Matrix square root of a matrix using the eigendecomposition and removing the eigenvalues smaller than a tolerance
</p>


<h3>Value</h3>

<p>The inverse matrix square root of the argument
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ginv">ginv</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
x=as.matrix(iris[,1:4])
S= t(x) 
matrixsqrtinv(S)
</code></pre>

<hr>
<h2 id='MDS'>
Multidimensional Scaling
</h2><span id='topic+MDS'></span>

<h3>Description</h3>

<p>Multidimensional Scaling using SMACOF algorithm and Bootstraping the coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MDS(Proximities, W = NULL, Model = c("Identity", "Ratio", "Interval", "Ordinal"), 
dimsol = 2, maxiter = 100, maxerror = 1e-06, Bootstrap = FALSE, nB = 200, 
ProcrustesRot = TRUE, BootstrapMethod = c("Sampling", "Permutation"), 
StandardizeDisparities = FALSE, ShowIter = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MDS_+3A_proximities">Proximities</code></td>
<td>

<p>An object of class proximities
</p>
</td></tr>
<tr><td><code id="MDS_+3A_w">W</code></td>
<td>

<p>A matrix of weigths
</p>
</td></tr>
<tr><td><code id="MDS_+3A_model">Model</code></td>
<td>

<p>MDS model. &quot;Identity&quot;, &quot;Ratio&quot;, &quot;Interval&quot; or &quot;Ordinal&quot;.
</p>
</td></tr>
<tr><td><code id="MDS_+3A_dimsol">dimsol</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="MDS_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations of the algorithm
</p>
</td></tr>
<tr><td><code id="MDS_+3A_maxerror">maxerror</code></td>
<td>

<p>Tolerance for convergence of the algorithm
</p>
</td></tr>
<tr><td><code id="MDS_+3A_bootstrap">Bootstrap</code></td>
<td>

<p>Should Bootstraping be performed?
</p>
</td></tr>
<tr><td><code id="MDS_+3A_nb">nB</code></td>
<td>

<p>Number of Bootstrap samples.
</p>
</td></tr>
<tr><td><code id="MDS_+3A_procrustesrot">ProcrustesRot</code></td>
<td>

<p>Should the bootstrap replicates be rotated to match the initial configuration using Procrustes?
</p>
</td></tr>
<tr><td><code id="MDS_+3A_bootstrapmethod">BootstrapMethod</code></td>
<td>

<p>The bootstrap is performed by samplig or permutaing the residuals?
</p>
</td></tr>
<tr><td><code id="MDS_+3A_standardizedisparities">StandardizeDisparities</code></td>
<td>

<p>Should the disparities be standardized
</p>
</td></tr>
<tr><td><code id="MDS_+3A_showiter">ShowIter</code></td>
<td>

<p>Show the iteration proccess
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Multidimensional Scaling using SMACOF algorithm and Bootstraping the coordinates. MDS performs multidimensional scaling of proximity data to find a least- squares representation of the objects in a low-dimensional space.  A majorization algorithm guarantees monotone convergence for optionally transformed, metric and nonmetric data under a variety of models.
</p>


<h3>Value</h3>

<p>An object of class <code>Principal.Coordinates</code> and  <code>MDS</code>. The function adds the information of the MDS to the object of class  <code>proximities</code>. Together with the information about the proximities the object has:
</p>
<table>
<tr><td><code>Analysis</code></td>
<td>
<p>The type of analysis performed, &quot;MDS&quot; in this case</p>
</td></tr>
<tr><td><code>Model</code></td>
<td>
<p>MDS model used</p>
</td></tr>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates for the objects in the MDS procedure</p>
</td></tr>
<tr><td><code>RawStress</code></td>
<td>
<p>Raw Stress values</p>
</td></tr>
<tr><td><code>stress1</code></td>
<td>
<p>stress formula 1</p>
</td></tr>
<tr><td><code>stress2</code></td>
<td>
<p>stress formula 2</p>
</td></tr>
<tr><td><code>sstress1</code></td>
<td>
<p>sstress formula 1</p>
</td></tr>
<tr><td><code>sstress2</code></td>
<td>
<p>sstress formula 2</p>
</td></tr>
<tr><td><code>rsq</code></td>
<td>
<p>Squared correlation between disparities and distances</p>
</td></tr>
<tr><td><code>Spearman</code></td>
<td>
<p>Spearman correlation between disparities and distances</p>
</td></tr>
<tr><td><code>Kendall</code></td>
<td>
<p>Kendall correlation between disparities and distances</p>
</td></tr>
<tr><td><code>BootstrapInfo</code></td>
<td>
<p>The result of the bootstrap calculations</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Commandeur, J. J. F. and Heiser, W. J. (1993). Mathematical derivations in the proximity scaling (PROXSCAL) of symmetric data matrices (Tech. Rep. No. RR- 93-03). Leiden, The Netherlands: Department of Data Theory, Leiden University.
</p>
<p>Kruskal, J. B. (1964). Nonmetric multidimensional scaling: A numerical method. Psychometrika, 29, 28-42.
</p>
<p>De Leeuw, J. &amp; Mair, P. (2009). Multidimensional scaling using majorization: The R package smacof. Journal of Statistical Software, 31(3), 1-30, http://www.jstatsoft.org/v31/i03/
</p>
<p>Borg, I., &amp; Groenen, P. J. F. (2005). Modern Multidimensional Scaling (2nd ed.). Springer.
</p>
<p>Borg, I., Groenen, P. J. F., &amp; Mair, P. (2013). Applied Multidimensional Scaling. Springer.
</p>
<p>Groenen, P. J. F., Heiser, W. J. and Meulman, J. J. (1999). Global optimization in least squares multidimensional scaling by distance smoothing. Journal of Classification, 16, 225-254.
</p>
<p>Groenen, P. J. F., van Os, B. and Meulman, J. J. (2000). Optimal scaling by alternating length-constained nonnegative least squares, with application to distance-based analysis. Psychometrika, 65, 511-524.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BootstrapSmacof">BootstrapSmacof</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
Dis=BinaryProximities(spiders)
MDSSol=MDS(Dis, Bootstrap=FALSE)
plot(MDSSol)

</code></pre>

<hr>
<h2 id='MGC'>
Mixture Gaussian Clustering
</h2><span id='topic+MGC'></span>

<h3>Description</h3>

<p>Model based clustering using mixtures of gaussian distriutions. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MGC(x, NG = 2, init = "km", RemoveOutliers=FALSE, ConfidOutliers=0.995, 
tolerance = 1e-07, maxiter = 100, show=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MGC_+3A_x">x</code></td>
<td>

<p>The data matrix
</p>
</td></tr>
<tr><td><code id="MGC_+3A_ng">NG</code></td>
<td>

<p>Number of groups or clusters to obtain
</p>
</td></tr>
<tr><td><code id="MGC_+3A_init">init</code></td>
<td>

<p>Initial centers can be obtained from k-means (&quot;km&quot;) or at random (&quot;rd&quot;)
</p>
</td></tr>
<tr><td><code id="MGC_+3A_removeoutliers">RemoveOutliers</code></td>
<td>

<p>Should the extreme values be removed to calculate the clusters?
</p>
</td></tr>
<tr><td><code id="MGC_+3A_confidoutliers">ConfidOutliers</code></td>
<td>

<p>Percentage of the points to keep for the calculations when RemoveOutliers is true.
</p>
</td></tr>
<tr><td><code id="MGC_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence
</p>
</td></tr>
<tr><td><code id="MGC_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations
</p>
</td></tr>
<tr><td><code id="MGC_+3A_show">show</code></td>
<td>

<p>Should the likelihood at each iteration be shown?
</p>
</td></tr>
<tr><td><code id="MGC_+3A_...">...</code></td>
<td>

<p>Maximum number of iterationsAny other parameter that can affect k-means if that is the initial configuration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A basic algorithm for clustering with mixtures of gaussians with no restrictions on the covariance matrices
</p>


<h3>Value</h3>

<p>Clusters
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Me falta
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X=as.matrix(iris[,1:4])
mod1=MGC(X,NG=3)
plot(iris[,1:4], col=mod1$Classification)
table(iris[,5],mod1$Classification)

</code></pre>

<hr>
<h2 id='MonotoneRegression'>
Weighted Isotonic Regression (Weighted Monotone Regression)
</h2><span id='topic+MonotoneRegression'></span>

<h3>Description</h3>

<p>Performs weighted isotonic (monotone) regression using the non-negative weights in w. The function is a direct translation of the matlab function lsqisotonic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MonotoneRegression(x, y, w = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MonotoneRegression_+3A_x">x</code></td>
<td>

<p>The independent variable vector
</p>
</td></tr>
<tr><td><code id="MonotoneRegression_+3A_y">y</code></td>
<td>

<p>The dependent variable vector
</p>
</td></tr>
<tr><td><code id="MonotoneRegression_+3A_w">w</code></td>
<td>

<p>A vector of weigths
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>YHAT = MonotoneRegression(X,Y) returns a vector of values that minimize the
sum of squares (Y - YHAT).^2 under the monotonicity constraint that
X(I) &gt; X(J) =&gt; YHAT(I) &gt;= YHAT(J), i.e., the values in YHAT are
monotonically non-decreasing with respect to X (sometimes referred
to as &quot;weak monotonicity&quot;).  LSQISOTONIC uses the &quot;pool adjacent
violators&quot; algorithm.
</p>
<p>If X(I) == X(J), then YHAT(I) may be &lt;, ==, or &gt; YHAT(J) (sometimes
referred to as the &quot;primary approach&quot;).  If ties do occur in X, a plot
of YHAT vs. X may appear to be non-monotonic at those points.  In fact,
the above monotonicity constraint is not violated, and a reordering
within each group of ties, by ascending YHAT, will produce the desired
appearance in the plot.
</p>


<h3>Value</h3>

<p>The fitted values after the monotone regression




</p>


<h3>Note</h3>

<p>The function is a direct translation of the matlab function lsqisotonic.
</p>


<h3>Author(s)</h3>

<p>Jose L. Vicente Villardon (from a matlab functiom)
</p>


<h3>References</h3>

<p>Kruskal, J.B. (1964) &quot;Nonmetric multidimensional scaling: a numerical method&quot;, Psychometrika 29:115-129.
</p>
<p>Cox, R.F. and Cox, M.A.A. (1994) Multidimensional Scaling, Chapman&amp;Hall.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Used inside MDS

</code></pre>

<hr>
<h2 id='moth'>
Moth data
</h2><span id='topic+moth'></span>

<h3>Description</h3>

<p>Moth data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("moth")</code></pre>


<h3>Format</h3>

<p>A data frame with 12 observations on the following 14 variables.
</p>

<dl>
<dt><code>s1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>s14</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Moth data
</p>


<h3>Source</h3>

<p>Withaker
</p>


<h3>References</h3>

<p>Application of the Parametric Bootstrap to Models that Incorporate a Singular
Value Decomposition
Luis Milan; Joe Whittaker
Applied Statistics, Vol. 44, No. 1. (1995), pp. 31-49.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(moth)
## maybe str(moth) ; plot(moth) ...
</code></pre>

<hr>
<h2 id='MultBiplotR-package'>
Multivariate Analysis using Biplots 
</h2><span id='topic+MultBiplotR-package'></span><span id='topic+MultBiplot'></span>

<h3>Description</h3>

<p>Classical PCA biplot with aditional features as non-standard data transformations, scales for the variables, together with many graphical aids as sizes or colors of the points according to their qualities of representation or predictiveness. The package includes also Alternating Least Squares (ALS) or Criss-Cross procedures for the calculation of the reduced rank approximation that can deal with missing data, differencial weights for each element of the data matrix or even ronust versions of the procedure. 
</p>
<p>This is part of a bigger project called MULTBIPLOT that contains many other biplot techniques and is a translation to R of the package MULBIPLOT programmed in MATLAB. A GUI for the package is also in preparation.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> MultBiplot</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.1.00</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2015-01-14</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL(&gt;=2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
Maintainer: Jose Luis Vicente Villardon &lt;villardon@usal.es&gt;
</p>


<h3>References</h3>

<p>Vicente-Villardon, J.L. (2010). MULTBIPLOT: A package for Multivariate Analysis using Biplots. Departamento de Estadistica. Universidad de  Salamanca. (http://biplot.usal.es/ClassicalBiplot/index.html).
</p>
<p>Vicente-Villardon, J. L. (1992). Una alternativa a las técnicas factoriales clasicas basada en una generalización de los metodos Biplot (Doctoral dissertation, Tesis. Universidad de Salamanca. España. 248 pp.[Links]).
</p>
<p>Gabriel KR (1971) The biplot graphic display of matrices with application to principal component analysis. Biometrika 58(3):453-467
</p>
<p>Gabriel KR (1998) Generalised bilinear regresion, J. L. (1998). Use of biplots to diagnose independence models in three-way contingency tables. Visualization of Categorical Data. Academic Press. London.
</p>
<p>Gabriel, K. R. (2002). Le biplot-outil d'exploration de donnes multidimensionnelles. Journal de la Societe francaise de statistique, 143(3-4).
</p>
<p>Gabriel KR, Zamir S (1979) Lower rank approximation of matrices by least squares with any choice of weights. Technometrics 21(4):489-498.
</p>
<p>Gower J, Hand D (1996) Biplots. Monographs on statistics and applied probability. 54. London: Chapman and Hall., 277 pp.
</p>
<p>Galindo Villardon, M. (1986). Una alternativa de representacion simultanea: HJ-Biplot. Qüestiió. 1986, vol. 10, núm. 1.
</p>
<p>Demey J, Vicente-Villardon JL, Galindo MP, Zambrano A (2008) Identifying molecular markers associated with classification of genotypes using external logistic biplots. Bioinformatics 24(24):2832-2838.
</p>
<p>Vicente-Villardon JL, Galindo MP, Blazquez-Zaballos A (2006) Logistic biplots. Multiple Correspondence Analysis and related methods pp 491-509.
</p>
<p>Santos, C., Munoz, S. S., Gutierrez, Y., Hebrero, E., Vicente, J. L., Galindo, P., Rivas, J. C. (1991). Characterization of young red wines by application of HJ biplot analysis to anthocyanin profiles. Journal of Agricultural and food chemistry, 39(6), 1086-1090.
</p>
<p>Rivas-Gonzalo, J. C., Gutierrez, Y., Polanco, A. M., Hebrero, E., Vicente, J. L., Galindo, P., Santos-Buelga, C. (1993). Biplot analysis applied to enological parameters in the geographical classification of young red wines. American journal of enology and viticulture, 44(3), 302-308.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
bip=PCA.Biplot(iris[,1:4])
plot(bip)
</code></pre>

<hr>
<h2 id='Multiquad'>
Multidimensional Gauss-Hermite quadrature
</h2><span id='topic+Multiquad'></span>

<h3>Description</h3>

<p>Multidimensional Gauss-Hermite quadrature
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Multiquad(nnodes, dims)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Multiquad_+3A_nnodes">nnodes</code></td>
<td>

<p>Number of nodes of the quadrature
</p>
</td></tr>
<tr><td><code id="Multiquad_+3A_dims">dims</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Multidimensional Gauss-Hermite quadrature
</p>


<h3>Value</h3>

<p>Multidimensional Gauss-Hermite quadrature
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Jackel, P. (2005). A note on multivariate Gauss-Hermite quadrature. http://www.awdz65.dsl.pipex.com/ANoteOnMultivariateGaussHermiteQuadrature.pdf
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Multiquad(5, 3)
</code></pre>

<hr>
<h2 id='MultiTableStatistics'>
Statistics for multiple tables
</h2><span id='topic+MultiTableStatistics'></span>

<h3>Description</h3>

<p>Statistics for multiple tables
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MultiTableStatistics(X, dual = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MultiTableStatistics_+3A_x">X</code></td>
<td>

<p>A multiple table
</p>
</td></tr>
<tr><td><code id="MultiTableStatistics_+3A_dual">dual</code></td>
<td>

<p>Is the transformation for the dual versions?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Statistics for multiple tables
</p>


<h3>Value</h3>

<p>A list with vectors of statistics for each table
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
</code></pre>

<hr>
<h2 id='MultiTableTransform'>
Initial Transformation of a multi table object
</h2><span id='topic+MultiTableTransform'></span>

<h3>Description</h3>

<p>Initial Transformation of a multi table object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MultiTableTransform(X, InitTransform = "Standardize columns", dual = FALSE,
CommonSD = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MultiTableTransform_+3A_x">X</code></td>
<td>

<p>Multi-table object
</p>
</td></tr>
<tr><td><code id="MultiTableTransform_+3A_inittransform">InitTransform</code></td>
<td>

<p>Initial Transformattion
</p>
</td></tr>
<tr><td><code id="MultiTableTransform_+3A_dual">dual</code></td>
<td>

<p>Is the transformation for the dual versions?
</p>
</td></tr>
<tr><td><code id="MultiTableTransform_+3A_commonsd">CommonSD</code></td>
<td>

<p>Should a common standard deviation be used for all the groups?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Initial Transformation of a multi table object
</p>


<h3>Value</h3>

<p>he table transformed
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>

<hr>
<h2 id='NiceNumber'>
Nice numbers: simple decimal numbers
</h2><span id='topic+NiceNumber'></span>

<h3>Description</h3>

<p>Calculates a close nice number, i. e. a number with simple decimals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NiceNumber(x = 6, round = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NiceNumber_+3A_x">x</code></td>
<td>

<p>A number
</p>
</td></tr>
<tr><td><code id="NiceNumber_+3A_round">round</code></td>
<td>

<p>Should the number be rounded?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates a close nice number, i. e. a number with simple decimals.
</p>


<h3>Value</h3>

<p>A number with simple decimals
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Heckbert, P. S. (1990). Nice numbers for graph labels. In Graphics Gems (pp. 61-63). Academic Press Professional, Inc..
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PrettyTicks">PrettyTicks</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>NiceNumber(0.892345)
</code></pre>

<hr>
<h2 id='NIPALS.Biplot'>
Biplot using the NIPALS algorithm
</h2><span id='topic+NIPALS.Biplot'></span>

<h3>Description</h3>

<p>Biplot using the NIPALS algorithm including a truncated and a sparse version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NIPALS.Biplot(X, alpha = 1, dimension = 3, Scaling = 5, 
Type = "Regular", grouping = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NIPALS.Biplot_+3A_x">X</code></td>
<td>

<p>The data matrix
</p>
</td></tr>
<tr><td><code id="NIPALS.Biplot_+3A_alpha">alpha</code></td>
<td>

<p>A number between 0 and 1. 0 for GH-Biplot, 1 for JK-Biplot and 0.5 for SQRT-Biplot. Use 2 or any other value not in the interval [0,1] for HJ-Biplot.
</p>
</td></tr>
<tr><td><code id="NIPALS.Biplot_+3A_dimension">dimension</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="NIPALS.Biplot_+3A_scaling">Scaling</code></td>
<td>

<p>Transformation of the original data. See InitialTransform for available transformations.
</p>
</td></tr>
<tr><td><code id="NIPALS.Biplot_+3A_type">Type</code></td>
<td>

<p>Type of biplot (Regular, Truncated or Sparse)
</p>
</td></tr>
<tr><td><code id="NIPALS.Biplot_+3A_grouping">grouping</code></td>
<td>

<p>Grouping fartor when the scaling is made with the within groups variability
</p>
</td></tr>
<tr><td><code id="NIPALS.Biplot_+3A_...">...</code></td>
<td>

<p>Aditional arguments for the different types of biplots.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Biplot using the NIPALS algorithm including a truncated and a sparse version.
</p>


<h3>Value</h3>

<p>An object of class ContinuousBiplot with the following components:
</p>
<table>
<tr><td><code>Title</code></td>
<td>
<p>A general title</p>
</td></tr>
<tr><td><code>Type</code></td>
<td>
<p>NIPALS</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>call</p>
</td></tr>
<tr><td><code>Non_Scaled_Data</code></td>
<td>
<p>Original Data Matrix</p>
</td></tr>
<tr><td><code>Means</code></td>
<td>
<p>Means of the original Variables</p>
</td></tr>
<tr><td><code>Medians</code></td>
<td>
<p>Medians of the original Variables</p>
</td></tr>
<tr><td><code>Deviations</code></td>
<td>
<p>Standard Deviations of the original Variables</p>
</td></tr>
<tr><td><code>Minima</code></td>
<td>
<p>Minima of the original Variables</p>
</td></tr>
<tr><td><code>Maxima</code></td>
<td>
<p>Maxima of the original Variables</p>
</td></tr>
<tr><td><code>P25</code></td>
<td>
<p>25 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>P75</code></td>
<td>
<p>75 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>Gmean</code></td>
<td>
<p>Global mean of the complete matrix</p>
</td></tr>
<tr><td><code>Sup.Rows</code></td>
<td>
<p>Supplementary rows (Non Transformed)</p>
</td></tr>
<tr><td><code>Sup.Cols</code></td>
<td>
<p>Supplementary columns (Non Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Data</code></td>
<td>
<p>Transformed Data</p>
</td></tr>
<tr><td><code>Scaled_Sup.Rows</code></td>
<td>
<p>Supplementary rows (Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Sup.Cols</code></td>
<td>
<p>Supplementary columns (Transformed)</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>Number of Rows</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>Number of Columns</p>
</td></tr>
<tr><td><code>nrowsSup</code></td>
<td>
<p>Number of Supplementary Rows</p>
</td></tr>
<tr><td><code>ncolsSup</code></td>
<td>
<p>Number of Supplementary Columns</p>
</td></tr>
<tr><td><code>dim</code></td>
<td>
<p>Dimension of the Biplot</p>
</td></tr>
<tr><td><code>EigenValues</code></td>
<td>
<p>Eigenvalues</p>
</td></tr>
<tr><td><code>Inertia</code></td>
<td>
<p>Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>CumInertia</code></td>
<td>
<p>Cumulative Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>EV</code></td>
<td>
<p>EigenVectors</p>
</td></tr>
<tr><td><code>Structure</code></td>
<td>
<p>Correlations of the Principal Components and the Variables</p>
</td></tr>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColCoordinates</code></td>
<td>
<p>Coordinates for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>RowContributions</code></td>
<td>
<p>Contributions for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColContributions</code></td>
<td>
<p>Contributions for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>Scale_Factor</code></td>
<td>
<p>Scale factor for the traditional plot with points and arrows. The row coordinates are multiplied and the column coordinates divided by that scale factor. The look of the plot is better without changing the inner product. For the HJ-Biplot the scale factor is 1.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Wold, H. (1966). Estimation of principal components and related models by iterative least squares. Multivariate analysis. ACEDEMIC PRESS. 391-420.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bip1=NIPALS.Biplot(wine[,4:21], Type="Sparse", lambda=0.15)
plot(bip1)
</code></pre>

<hr>
<h2 id='NIPALSPCA'>
NIPALS algorithm for PCA
</h2><span id='topic+NIPALSPCA'></span>

<h3>Description</h3>

<p>Classical NIPALS algorithm for PCA and Biplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NIPALSPCA(X, dimens = 2, tol = 1e-06, maxiter = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NIPALSPCA_+3A_x">X</code></td>
<td>

<p>The data matrix.
</p>
</td></tr>
<tr><td><code id="NIPALSPCA_+3A_dimens">dimens</code></td>
<td>

<p>The dimension of the solution
</p>
</td></tr>
<tr><td><code id="NIPALSPCA_+3A_tol">tol</code></td>
<td>

<p>Tolerance of the algorithm.
</p>
</td></tr>
<tr><td><code id="NIPALSPCA_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iteratios.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Classical NIPALS algorithm for the singular value decomposition that allows for the construction of PCA and Biplot. 
</p>


<h3>Value</h3>

<p>The singular value decomposition
</p>
<table>
<tr><td><code>u</code></td>
<td>
<p>The coordinates of the rows (standardized)</p>
</td></tr>
<tr><td><code>d</code></td>
<td>
<p>The singuklar values</p>
</td></tr>
<tr><td><code>v</code></td>
<td>
<p>The coordinates of the columns (standardized)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Wold, H. (1966). Estimation of principal components and related models by iterative least squares. Multivariate analysis. ACEDEMIC PRESS. 391-420.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='NominalDistances'>
Distances among individuals with nominal variables 
</h2><span id='topic+NominalDistances'></span>

<h3>Description</h3>

<p>This function computes several measures of distance (or similarity) among individuals from a nominal data matrix. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NominalDistances(X, method = 1, diag = FALSE, upper = FALSE, similarity = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NominalDistances_+3A_x">X</code></td>
<td>

<p>Matrix or data.frame with the nominal variables.
</p>
</td></tr>
<tr><td><code id="NominalDistances_+3A_method">method</code></td>
<td>

<p>An integer between 1 and 6. See details 
</p>
</td></tr>
<tr><td><code id="NominalDistances_+3A_diag">diag</code></td>
<td>

<p>A logical value indicating whether the diagonal of the distance matrix should be printed.
</p>
</td></tr>
<tr><td><code id="NominalDistances_+3A_upper">upper</code></td>
<td>

<p>a logical value indicating whether the upper triangle of the distance matrix should be printed.
</p>
</td></tr>
<tr><td><code id="NominalDistances_+3A_similarity">similarity</code></td>
<td>

<p>A logical value indicating whether the similarity matrix should be computed.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let be the table of nominal data. All these distances are of type <code class="reqn">d=\sqrt{1-s}</code> with <em>s</em> a similarity coefficient.
</p>

<dl>
<dt>1 = Overlap method</dt><dd><p>The overlap measure simply counts the number of attributes that match in the two data instances.</p>
</dd>
<dt>2 = Eskin</dt><dd><p>Eskin et al. proposed a normalization kernel for record-based network intrusion detection data. The original measure is distance-based and assigns a weight of <code class="reqn">\frac{2}{n_{k}^{2}}</code> for mismatches; when adapted to similarity, this becomes a weight of <code class="reqn">\frac{n_{k}^{2}}{n_{k}^{2}+2}</code>.This measure gives more weight to mismatches that occur on attributes that take many values.</p>
</dd>
<dt>3=IOF (Inverse Occurrence Frequency .)</dt><dd>
<p>This measure assigns lower similarity to mismatches on more frequent values. The IOF measure is related to the concept of inverse document frequency which comes from information retrieval, where it is used to signify the relative number of documents that contain a spe- cific word.</p>
</dd>
<dt>4 = OF (Ocurrence Frequency)</dt><dd><p>This measure gives the opposite weighting of the IOF measure for mismatches, i.e., mismatches on less frequent values are assigned lower similarity and mismatches on more frequent values are assigned higher similarity</p>
</dd>
<dt>5 = Goodall3</dt><dd><p>This measure assigns a high similarity if the matching values are infrequent regardless of the frequencies of the other values.</p>
</dd>
<dt>6 = Lin</dt><dd><p>This measure gives higher weight to matches on frequent values, and lower weight to mismatches on infrequent values.</p>
</dd>
</dl>



<h3>Value</h3>

<p>An object of class distance
</p>


<h3>Author(s)</h3>

<p>Jose L. Vicente-Villardon
</p>


<h3>References</h3>

<p>Boriah, S., Chandola, V. &amp; Kumar,V.(2008). Similarity measures for categorical data: A comparative evaluation. In proceedings of the eight SIAM International Conference on Data Mining, pp 243&ndash;254.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BinaryDistances">BinaryDistances</a></code>,<code><a href="#topic+ContinuousDistances">ContinuousDistances</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(Env)
Distance&lt;-NominalDistances(Env,upper=TRUE,diag=TRUE,similarity=FALSE,method=1)

## End(Not run)
</code></pre>

<hr>
<h2 id='NormalityTests'>
Normality tests
</h2><span id='topic+NormalityTests'></span>

<h3>Description</h3>

<p>Normality tests foor the columns of a matrix  and a grouping variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NormalityTests(X, groups = NULL, plot = FALSE, SortByGroups = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NormalityTests_+3A_x">X</code></td>
<td>

<p>A data frame or a matrix containing several numerical variables
</p>
</td></tr>
<tr><td><code id="NormalityTests_+3A_groups">groups</code></td>
<td>

<p>A factor with the groups
</p>
</td></tr>
<tr><td><code id="NormalityTests_+3A_plot">plot</code></td>
<td>

<p>If TRUE the qqnorm plots are shown
</p>
</td></tr>
<tr><td><code id="NormalityTests_+3A_sortbygroups">SortByGroups</code></td>
<td>

<p>Should the results be sorted by groups?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Normality tests foor the columns of a matrix  and a grouping variable.
</p>


<h3>Value</h3>

<p>The normality tests and the plots
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
NormalityTests(wine[,4:6], groups = wine$Origin, plot=TRUE)
</code></pre>

<hr>
<h2 id='Numeric2Binary'>Converts a numeric variable into a binary one</h2><span id='topic+Numeric2Binary'></span>

<h3>Description</h3>

<p>Converts a numeric variable into a binary one using a cut point
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Numeric2Binary(y, name= "MyVar", cut = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Numeric2Binary_+3A_y">y</code></td>
<td>

<p>Vector containing the numeric values
</p>
</td></tr>
<tr><td><code id="Numeric2Binary_+3A_name">name</code></td>
<td>

<p>Name of the variable
</p>
</td></tr>
<tr><td><code id="Numeric2Binary_+3A_cut">cut</code></td>
<td>

<p>Cut point to cut the values of the variable. If is NULL the median is used.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Converts a numeric variable into a binary one using a cut point. If the cut is NULL the median is used.
</p>


<h3>Value</h3>

<p>A binary Variable
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>See Also</h3>

<p>Dataframe2BinaryMatrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>y=c(1, 1.2, 3.2, 2.4, 1.7, 2.2, 2.7, 3.1)
Numeric2Binary(y)
</code></pre>

<hr>
<h2 id='ones'>
Matrix of ones
</h2><span id='topic+ones'></span>

<h3>Description</h3>

<p>Square matrix of ones
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ones(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ones_+3A_n">n</code></td>
<td>

<p>Order of the matrix
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Square matrix of ones
</p>


<h3>Value</h3>

<p>A matrix of ones of order n.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ones(6)
</code></pre>

<hr>
<h2 id='OrdinalLogisticFit'>
Fits an ordinal logistic regression with ridge penalization
</h2><span id='topic+OrdinalLogisticFit'></span>

<h3>Description</h3>

<p>This function fits a logistic regression between a dependent ordinal variable y
and some independent variables x, and solves the separation problem using ridge penalization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OrdinalLogisticFit(y, x, penalization = 0.1, tol = 1e-04, maxiter = 200, show = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OrdinalLogisticFit_+3A_y">y</code></td>
<td>

<p>Dependent variable.
</p>
</td></tr>
<tr><td><code id="OrdinalLogisticFit_+3A_x">x</code></td>
<td>

<p>A matrix with the independent variables.
</p>
</td></tr>
<tr><td><code id="OrdinalLogisticFit_+3A_penalization">penalization</code></td>
<td>

<p>Penalization used to avoid singularities.
</p>
</td></tr>
<tr><td><code id="OrdinalLogisticFit_+3A_tol">tol</code></td>
<td>

<p>Tolerance for the iterations.
</p>
</td></tr>
<tr><td><code id="OrdinalLogisticFit_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations.
</p>
</td></tr>
<tr><td><code id="OrdinalLogisticFit_+3A_show">show</code></td>
<td>

<p>Should the iteration history be printed?.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The problem of the existence of the estimators in logistic regression can be seen in Albert (1984); a solution for the binary case, based on the Firth's method, Firth (1993) is proposed by Heinze(2002). All the procedures were initially developed to remove the bias but work well to avoid the problem of separation. Here we have chosen a simpler solution based on ridge estimators for logistic regression Cessie(1992).  
</p>
<p>Rather than maximizing <code class="reqn">{L_j}(\left. {\bf{G}} \right|{{\bf{b}}_{j0}},{{\bf{B}}_j})</code> we maximize
</p>
<p style="text-align: center;"><code class="reqn">{{L_j}(\left. {\bf{G}} \right|{{\bf{b}}_{j0}},{{\bf{B}}_j})} - \lambda \left( {\left\| {{{\bf{b}}_{j0}}} \right\| + \left\| {{{\bf{B}}_j}} \right\|} \right)</code>
</p>

<p>Changing the values of <code class="reqn">\lambda</code> we obtain slightly different solutions not affected by the separation problem.
</p>


<h3>Value</h3>

<p>An object of class <code>"pordlogist"</code>.  This has components:
</p>
<table>
<tr><td><code>nobs</code></td>
<td>
<p>Number of observations</p>
</td></tr>
<tr><td><code>J</code></td>
<td>
<p>Maximum value of the dependent variable</p>
</td></tr>
<tr><td><code>nvar</code></td>
<td>
<p>Number of independent variables</p>
</td></tr>
<tr><td><code>fitted.values</code></td>
<td>
<p>Matrix with the fitted probabilities</p>
</td></tr>
<tr><td><code>pred</code></td>
<td>
<p>Predicted values for each item</p>
</td></tr>
<tr><td><code>Covariances</code></td>
<td>
<p>Covariances matrix</p>
</td></tr>
<tr><td><code>clasif</code></td>
<td>
<p>Matrix of classification of the items</p>
</td></tr>
<tr><td><code>PercentClasif</code></td>
<td>
<p>Percent of good classifications</p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p> Estimated coefficients for the ordinal logistic regression</p>
</td></tr>
<tr><td><code>thresholds</code></td>
<td>
<p>Thresholds of the estimated model</p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>Logarithm of the likelihood</p>
</td></tr>
<tr><td><code>penalization</code></td>
<td>
<p>Penalization used to avoid singularities</p>
</td></tr>
<tr><td><code>Deviance</code></td>
<td>
<p>Deviance of the model</p>
</td></tr>
<tr><td><code>DevianceNull</code></td>
<td>
<p>Deviance of the null model</p>
</td></tr>
<tr><td><code>Dif</code></td>
<td>
<p>Diference between the two deviances values calculated</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>Degrees of freedom</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>p-value of the contrast</p>
</td></tr>
<tr><td><code>CoxSnell</code></td>
<td>
<p>Cox-Snell pseudo R squared</p>
</td></tr>
<tr><td><code>Nagelkerke</code></td>
<td>
<p>Nagelkerke pseudo R squared</p>
</td></tr>
<tr><td><code>MacFaden</code></td>
<td>
<p>Nagelkerke pseudo R squared</p>
</td></tr>
<tr><td><code>iter</code></td>
<td>
<p>Number of iterations made</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Albert,A. &amp; Anderson,J.A. (1984),<em>On the existence of maximum likelihood estimates in logistic regression models</em>, Biometrika 71(1), 1&ndash;10.
</p>
<p>Bull, S.B., Mak, C. &amp; Greenwood, C.M. (2002), <em>A modified score function for multinomial logistic regression</em>, Computational Statistics and dada Analysis 39, 57&ndash;74.
</p>
<p>Firth, D.(1993), <em>Bias reduction of maximum likelihood estimates</em>, Biometrika 80(1), 27&ndash;38
</p>
<p>Heinze, G. &amp; Schemper, M. (2002), <em>A solution to the problem of separation in logistic regression</em>, Statistics in Medicine 21, 2109&ndash;2419
</p>
<p>Le Cessie, S. &amp; Van Houwelingen, J. (1992), <em>Ridge estimators in logistic regression</em>, Applied Statistics 41(1), 191&ndash;201.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># No examples yet
</code></pre>

<hr>
<h2 id='OrdLogBipEM'>
Alternated EM algorithm for Ordinal Logistic Biplots
</h2><span id='topic+OrdLogBipEM'></span>

<h3>Description</h3>

<p>This function computes, with an alternated algorithm, the row and column parameters of an Ordinal Logistic Biplot for ordered polytomous data. The row coordinates (E-step) are computed using multidimensional Gauss-Hermite quadratures and Expected <em>a posteriori</em> (EAP) scores and parameters for each variable or items (M-step) using Ridge Ordinal Logistic Regression to solve the separation problem present when the points for different categories of a variable are completely separated on the representation plane and the usual fitting methods do not converge. The separation problem is present in almost avery data set for which the goodness of fit is high. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OrdLogBipEM(Data, freq=NULL, dim = 2, nnodes = 15, 
tol = 0.0001, maxiter = 100, maxiterlogist = 100, 
penalization = 0.2, show = FALSE, initial = 1, alfa = 1, 
Orthogonalize=TRUE, Varimax=TRUE, ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OrdLogBipEM_+3A_data">Data</code></td>
<td>

<p>Data frame with the ordinal data. All the variables must be ordered factors. 
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_freq">freq</code></td>
<td>

<p>Frequencies for compacted tables
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_dim">dim</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_nnodes">nnodes</code></td>
<td>

<p>Number of nodes for the multidimensional Gauss-Hermite quadrature
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_tol">tol</code></td>
<td>

<p>Value to stop the process of iterations.
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations for the biplot procedure.
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_maxiterlogist">maxiterlogist</code></td>
<td>

<p>Maximum number of iterations for the logistic regression step or the Mirt initial configuration.
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_penalization">penalization</code></td>
<td>

<p>Penalization used in the diagonal matrix to avoid singularities.
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_show">show</code></td>
<td>

<p>Boolean parameter to specify if the user wants to see every iteration.
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_initial">initial</code></td>
<td>

<p>Method used to choose the initial ability in the algorithm. Default value is 1.
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_alfa">alfa</code></td>
<td>

<p>Optional parameter to calculate row and column coordinates in Simple correspondence analysis if the initial parameter is equal to 1.
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_orthogonalize">Orthogonalize</code></td>
<td>

<p>Should the final row coordinates be orthogonalized?. The column parameters have to be recalculated.
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_varimax">Varimax</code></td>
<td>

<p>Should the final row coordinates be rotated using the varimax procedure?.
</p>
</td></tr>
<tr><td><code id="OrdLogBipEM_+3A_...">...</code></td>
<td>

<p>Aditional argunments for mirt.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"Ordinal.Logistic.Biplot"</code>.This has components:
</p>
<table>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates for the rows or the individuals</p>
</td></tr>
<tr><td><code>ColumnParameters</code></td>
<td>
<p>List with information about the Ordinal Logistic Models calculated for each variable including: estimated parameters with thresholds,percents of correct classifications,and pseudo-Rsquared</p>
</td></tr>
<tr><td><code>loadings</code></td>
<td>
<p>factor loadings</p>
</td></tr>
<tr><td><code>LogLikelihood</code></td>
<td>
<p>Logarithm of the likelihood</p>
</td></tr>
<tr><td><code>r2</code></td>
<td>
<p>R squared coefficient</p>
</td></tr>
<tr><td><code>Ncats</code></td>
<td>
<p>Number of the categories of each variable</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Bock,R. &amp; Aitkin,M. (1981),<em>Marginal maximum likelihood estimation of item parameters: Aplication of an EM algorithm</em>, Phychometrika 46(4), 443-459.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
    data(Doctors)
    olb = OrdLogBipEM(Doctors,dim = 2, nnodes = 10, initial=4,
    tol = 0.001, maxiter = 100, penalization = 0.1, show=TRUE)
    olb
    summary(olb)
    PlotOrdinalResponses(olb)
    
## End(Not run)
</code></pre>

<hr>
<h2 id='OrdVarBiplot'>
Plots an ordinal variable on the biplot
</h2><span id='topic+OrdVarBiplot'></span>

<h3>Description</h3>

<p>Plots an ordinal variable on the biplot from its fitted parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OrdVarBiplot(bi1, bi2, threshold, xmin = -3, xmax = 3, ymin = -3, 
ymax = 3, label = "Point", mode = "a", CexPoint = 0.8,
PchPoint = 1, Color = "green", tl = 0.03, textpos = 1, CexScale= 0.5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OrdVarBiplot_+3A_bi1">bi1</code></td>
<td>

<p>Slope for the first dimension to plot
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_bi2">bi2</code></td>
<td>

<p>Slope for the second dimension to plot
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_threshold">threshold</code></td>
<td>

<p>Thresholds for each category of the variable</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_xmin">xmin</code></td>
<td>

<p>Minimum value of the X on the plot
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_xmax">xmax</code></td>
<td>

<p>Maximum value of the X on the plot
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_ymin">ymin</code></td>
<td>

<p>Minimum value of the Y on the plot
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_ymax">ymax</code></td>
<td>

<p>Maximum value of the X on the plot
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_label">label</code></td>
<td>

<p>Label of the variable
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_mode">mode</code></td>
<td>

<p>Mode of the plot (as in a regular biplot)
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_cexpoint">CexPoint</code></td>
<td>

<p>Size of the point
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_pchpoint">PchPoint</code></td>
<td>

<p>Mark for the point
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_color">Color</code></td>
<td>

<p>Color
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_tl">tl</code></td>
<td>

<p>Tick Length
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_textpos">textpos</code></td>
<td>

<p>Position of the label
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_cexscale">CexScale</code></td>
<td>

<p>Sizes of the scales
</p>
</td></tr>
<tr><td><code id="OrdVarBiplot_+3A_...">...</code></td>
<td>

<p>Any aditional graphical parameter
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots an ordinal variable on the biplot from its fitted parameters. The plot uses the same parameters as any other biplot.
</p>


<h3>Value</h3>

<p>Returns a graphical representation of the ordinal variable on the current plot
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardon, J. L., &amp; Sanchez, J. C. H. (2014). Logistic Biplots for Ordinal Data with an Application to Job Satisfaction of Doctorate Degree Holders in Spain. arXiv preprint arXiv:1405.0294.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
</code></pre>

<hr>
<h2 id='OrdVarCoordinates'>
Coordinates of an ordinal variable on the biplot.
</h2><span id='topic+OrdVarCoordinates'></span>

<h3>Description</h3>

<p>Coordinates of an ordinal variable on the biplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OrdVarCoordinates(tr, b = c(1, 1), inf = -12, sup = 12, step = 0.01,
                 plotresponse = FALSE, label = "Item", labx = "z", laby
                 = "Probability", catnames = NULL, Legend = TRUE,
                 LegendPos = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OrdVarCoordinates_+3A_tr">tr</code></td>
<td>

<p>A vector containing the thresholds of the model, that is, the constatn for each category of the ordinal variable
</p>
</td></tr>
<tr><td><code id="OrdVarCoordinates_+3A_b">b</code></td>
<td>

<p>Vector containing the common slopes for all categories of the ordinal variable
</p>
</td></tr>
<tr><td><code id="OrdVarCoordinates_+3A_inf">inf</code></td>
<td>

<p>The inferior limit of the values to be sampled on the biplot axis (it depends on the scale of the biplot).
</p>
</td></tr>
<tr><td><code id="OrdVarCoordinates_+3A_sup">sup</code></td>
<td>

<p>The superior limit of the values to be sampled on the biplot axis (it depends on the scale of the biplot).
</p>
</td></tr>
<tr><td><code id="OrdVarCoordinates_+3A_step">step</code></td>
<td>

<p>Increment (step) of the squence
</p>
</td></tr>
<tr><td><code id="OrdVarCoordinates_+3A_plotresponse">plotresponse</code></td>
<td>

<p>Should the item be plotted
</p>
</td></tr>
<tr><td><code id="OrdVarCoordinates_+3A_label">label</code></td>
<td>

<p>Label of the item.
</p>
</td></tr>
<tr><td><code id="OrdVarCoordinates_+3A_labx">labx</code></td>
<td>

<p>Label for the X axis in the summary of the item.
</p>
</td></tr>
<tr><td><code id="OrdVarCoordinates_+3A_laby">laby</code></td>
<td>

<p>Label for the Y axis in the summary of the item.
</p>
</td></tr>
<tr><td><code id="OrdVarCoordinates_+3A_catnames">catnames</code></td>
<td>

<p>Names of the categories.
</p>
</td></tr>
<tr><td><code id="OrdVarCoordinates_+3A_legend">Legend</code></td>
<td>

<p>Should a legend be plotted
</p>
</td></tr>
<tr><td><code id="OrdVarCoordinates_+3A_legendpos">LegendPos</code></td>
<td>

<p>Position of the legend.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates the coordinates of the points that define the separation among the categories of an ordinal variable projected onto an ordinal logistic biplot. 
</p>


<h3>Value</h3>

<p>An object of class OrdVarCoord
</p>
<table>
<tr><td><code>z</code></td>
<td>
<p>Values of the cut points on the scale of the biplot axis (not used)</p>
</td></tr>
<tr><td><code>points</code></td>
<td>
<p>The points for the marks to be represented on the biplot.</p>
</td></tr>
<tr><td><code>labels</code></td>
<td>
<p>The labels for the points</p>
</td></tr>
<tr><td><code>hidden</code></td>
<td>
<p>Are there any hidden categories? (Categories whose probability is never hier than the probabilities of the rest)</p>
</td></tr>
<tr><td><code>cathidden</code></td>
<td>
<p>Number of the hidden cateories</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardon, J. L., &amp; Sanchez, J. C. H. (2014). Logistic Biplots for Ordinal Data with an Application to Job Satisfaction of Doctorate Degree Holders in Spain. arXiv preprint arXiv:1405.0294.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># No examples
</code></pre>

<hr>
<h2 id='OrthogonalizeScores'>
Orthogonalize a set of Scores calculated by other procedure
</h2><span id='topic+OrthogonalizeScores'></span>

<h3>Description</h3>

<p>Orthogonalize a set of Scores calculated by other procedure
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OrthogonalizeScores(scores)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OrthogonalizeScores_+3A_scores">scores</code></td>
<td>

<p>A matrix containing the scores
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Orthogonalize a set of Scores calculated by other procedure proyecting onto the dimensions defined by the eigenvectors of the covariance matrix
</p>


<h3>Value</h3>

<p>The orthogonalised scores.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----

</code></pre>

<hr>
<h2 id='PCA.Analysis'>
Classical PCA Biplot with added features.
</h2><span id='topic+PCA.Analysis'></span>

<h3>Description</h3>

<p>Classical PCA Biplot with added features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCA.Analysis(X, dimension = 3, Scaling = 5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PCA.Analysis_+3A_x">X</code></td>
<td>

<p>Data Matrix
</p>
</td></tr>
<tr><td><code id="PCA.Analysis_+3A_dimension">dimension</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="PCA.Analysis_+3A_scaling">Scaling</code></td>
<td>

<p>Transformation of the original data. See InitialTransform for available transformations.
</p>
</td></tr>
<tr><td><code id="PCA.Analysis_+3A_...">...</code></td>
<td>

<p>Any other useful argument
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Biplots represent the rows and columns of a data matrix in reduced dimensions. Usually rows represent  individuals, objects or samples and columns are variables measured on them. The most classical versions can be thought as visualizations associated to Principal Components Analysis (PCA) or Factor Analysis (FA) obtained from a Singular Value Decomposition or a related method. From another point of view, Classical Biplots could be obtained from regressions and calibrations that are essentially an alternated least squares algorithm equivalent to an EM-algorithm when data are normal.
</p>


<h3>Value</h3>

<p>An object of class ContinuousBiplot with the following components:
</p>
<table>
<tr><td><code>Title</code></td>
<td>
<p>A general title</p>
</td></tr>
<tr><td><code>Non_Scaled_Data</code></td>
<td>
<p>Original Data Matrix</p>
</td></tr>
<tr><td><code>Means</code></td>
<td>
<p>Means of the original Variables</p>
</td></tr>
<tr><td><code>Medians</code></td>
<td>
<p>Medians of the original Variables</p>
</td></tr>
<tr><td><code>Deviations</code></td>
<td>
<p>Standard Deviations of the original Variables</p>
</td></tr>
<tr><td><code>Minima</code></td>
<td>
<p>Minima of the original Variables</p>
</td></tr>
<tr><td><code>Maxima</code></td>
<td>
<p>Maxima of the original Variables</p>
</td></tr>
<tr><td><code>P25</code></td>
<td>
<p>25 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>P75</code></td>
<td>
<p>75 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>Gmean</code></td>
<td>
<p>Global mean of the complete matrix</p>
</td></tr>
<tr><td><code>Sup.Rows</code></td>
<td>
<p>Supplementary rows (Non Transformed)</p>
</td></tr>
<tr><td><code>Sup.Cols</code></td>
<td>
<p>Supplementary columns (Non Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Data</code></td>
<td>
<p>Transformed Data</p>
</td></tr>
<tr><td><code>Scaled_Sup.Rows</code></td>
<td>
<p>Supplementary rows (Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Sup.Cols</code></td>
<td>
<p>Supplementary columns (Transformed)</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>Number of Rows</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>Number of Columns</p>
</td></tr>
<tr><td><code>nrowsSup</code></td>
<td>
<p>Number of Supplementary Rows</p>
</td></tr>
<tr><td><code>ncolsSup</code></td>
<td>
<p>Number of Supplementary Columns</p>
</td></tr>
<tr><td><code>dim</code></td>
<td>
<p>Dimension of the Biplot</p>
</td></tr>
<tr><td><code>EigenValues</code></td>
<td>
<p>Eigenvalues</p>
</td></tr>
<tr><td><code>Inertia</code></td>
<td>
<p>Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>CumInertia</code></td>
<td>
<p>Cumulative Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>EV</code></td>
<td>
<p>EigenVectors</p>
</td></tr>
<tr><td><code>Structure</code></td>
<td>
<p>Correlations of the Principal Components and the Variables</p>
</td></tr>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColCoordinates</code></td>
<td>
<p>Coordinates for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>RowContributions</code></td>
<td>
<p>Contributions for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColContributions</code></td>
<td>
<p>Contributions for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>Scale_Factor</code></td>
<td>
<p>Scale factor for the traditional plot with points and arrows. The row coordinates are multiplied and the column coordinates divided by that scale factor. The look of the plot is better without changing the inner product. For the HJ-Biplot the scale factor is 1.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Gabriel, K.R.(1971): The biplot graphic display of matrices with applications to principal component analysis. Biometrika,  58, 453-467.
</p>
<p>Galindo Villardon, M. (1986). Una alternativa de representacion simultanea: HJ-Biplot. Questiio. 1986, vol. 10, núm. 1.
</p>
<p>Gabriel, K. R. AND Zamir, S. (1979). Lower rank approximation of matrices by least squares with any choice of weights. Technometrics, 21(21):489&ndash;498, 1979.
</p>
<p>Gabriel, K.R.(1998): Generalised Bilinear Regression. Biometrika,  85, 3, 689-700.
</p>
<p>Gower y Hand (1996): Biplots. Chapman &amp; Hall.
</p>
<p>Vicente-Villardon, J. L., Galindo, M. P. and Blazquez-Zaballos, A. (2006). Logistic Biplots. Multiple Correspondence Analysis and related methods 491-509.
</p>
<p>Demey, J., Vicente-Villardon, J. L., Galindo, M. P. and Zambrano, A. (2008). Identifying Molecular Markers Associated With Classification Of Genotypes Using External Logistic Biplots. Bioinformatics 24 2832-2838.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+InitialTransform">InitialTransform</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Simple Biplot with arrows
data(Protein)
bip=PCA.Biplot(Protein[,3:11])
plot(bip)

## Biplot with scales on the variables
plot(bip, mode="s", margin=0.2)

# Structure plot (Correlations)
CorrelationCircle(bip)

# Plot of the Variable Contributions
ColContributionPlot(bip, cex=1)


</code></pre>

<hr>
<h2 id='PCA.Biplot'>
Classical PCA Biplot with added features.
</h2><span id='topic+PCA.Biplot'></span>

<h3>Description</h3>

<p>Classical PCA Biplot with added features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCA.Biplot(X, alpha = 1, dimension = 2, Scaling = 5, sup.rows = NULL, 
          sup.cols = NULL, grouping = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PCA.Biplot_+3A_x">X</code></td>
<td>

<p>Data Matrix
</p>
</td></tr>
<tr><td><code id="PCA.Biplot_+3A_alpha">alpha</code></td>
<td>

<p>A number between 0 and 1. 0 for GH-Biplot, 1 for JK-Biplot and 0.5 for SQRT-Biplot. Use 2 or any other value not in the interval [0,1] for HJ-Biplot.
</p>
</td></tr>
<tr><td><code id="PCA.Biplot_+3A_dimension">dimension</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="PCA.Biplot_+3A_scaling">Scaling</code></td>
<td>

<p>Transformation of the original data. See InitialTransform for available transformations.
</p>
</td></tr>
<tr><td><code id="PCA.Biplot_+3A_sup.rows">sup.rows</code></td>
<td>

<p>Supplementary or illustrative rows, if any.
</p>
</td></tr>
<tr><td><code id="PCA.Biplot_+3A_sup.cols">sup.cols</code></td>
<td>

<p>Supplementary or illustrative rows, if any.
</p>
</td></tr>
<tr><td><code id="PCA.Biplot_+3A_grouping">grouping</code></td>
<td>

<p>A factor to standardize with the variability within groups
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Biplots represent the rows and columns of a data matrix in reduced dimensions. Usually rows represent  individuals, objects or samples and columns are variables measured on them. The most classical versions can be thought as visualizations associated to Principal Components Analysis (PCA) or Factor Analysis (FA) obtained from a Singular Value Decomposition or a related method. From another point of view, Classical Biplots could be obtained from regressions and calibrations that are essentially an alternated least squares algorithm equivalent to an EM-algorithm when data are normal.
</p>


<h3>Value</h3>

<p>An object of class ContinuousBiplot with the following components:
</p>
<table>
<tr><td><code>Title</code></td>
<td>
<p>A general title</p>
</td></tr>
<tr><td><code>Non_Scaled_Data</code></td>
<td>
<p>Original Data Matrix</p>
</td></tr>
<tr><td><code>Means</code></td>
<td>
<p>Means of the original Variables</p>
</td></tr>
<tr><td><code>Medians</code></td>
<td>
<p>Medians of the original Variables</p>
</td></tr>
<tr><td><code>Deviations</code></td>
<td>
<p>Standard Deviations of the original Variables</p>
</td></tr>
<tr><td><code>Minima</code></td>
<td>
<p>Minima of the original Variables</p>
</td></tr>
<tr><td><code>Maxima</code></td>
<td>
<p>Maxima of the original Variables</p>
</td></tr>
<tr><td><code>P25</code></td>
<td>
<p>25 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>P75</code></td>
<td>
<p>75 Percentile of the original Variables</p>
</td></tr>
<tr><td><code>Gmean</code></td>
<td>
<p>Global mean of the complete matrix</p>
</td></tr>
<tr><td><code>Sup.Rows</code></td>
<td>
<p>Supplementary rows (Non Transformed)</p>
</td></tr>
<tr><td><code>Sup.Cols</code></td>
<td>
<p>Supplementary columns (Non Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Data</code></td>
<td>
<p>Transformed Data</p>
</td></tr>
<tr><td><code>Scaled_Sup.Rows</code></td>
<td>
<p>Supplementary rows (Transformed)</p>
</td></tr>
<tr><td><code>Scaled_Sup.Cols</code></td>
<td>
<p>Supplementary columns (Transformed)</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>Number of Rows</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>Number of Columns</p>
</td></tr>
<tr><td><code>nrowsSup</code></td>
<td>
<p>Number of Supplementary Rows</p>
</td></tr>
<tr><td><code>ncolsSup</code></td>
<td>
<p>Number of Supplementary Columns</p>
</td></tr>
<tr><td><code>dim</code></td>
<td>
<p>Dimension of the Biplot</p>
</td></tr>
<tr><td><code>EigenValues</code></td>
<td>
<p>Eigenvalues</p>
</td></tr>
<tr><td><code>Inertia</code></td>
<td>
<p>Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>CumInertia</code></td>
<td>
<p>Cumulative Explained variance (Inertia)</p>
</td></tr>
<tr><td><code>EV</code></td>
<td>
<p>EigenVectors</p>
</td></tr>
<tr><td><code>Structure</code></td>
<td>
<p>Correlations of the Principal Components and the Variables</p>
</td></tr>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColCoordinates</code></td>
<td>
<p>Coordinates for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>RowContributions</code></td>
<td>
<p>Contributions for the rows, including the supplementary</p>
</td></tr>
<tr><td><code>ColContributions</code></td>
<td>
<p>Contributions for the columns, including the supplementary</p>
</td></tr>
<tr><td><code>Scale_Factor</code></td>
<td>
<p>Scale factor for the traditional plot with points and arrows. The row coordinates are multiplied and the column coordinates divided by that scale factor. The look of the plot is better without changing the inner product. For the HJ-Biplot the scale factor is 1.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Gabriel, K.R.(1971): The biplot graphic display of matrices with applications to principal component analysis. Biometrika,  58, 453-467.
</p>
<p>Galindo Villardon, M. (1986). Una alternativa de representacion simultanea: HJ-Biplot. Questiio. 1986, vol. 10, núm. 1.
</p>
<p>Gabriel, K. R. AND Zamir, S. (1979). Lower rank approximation of matrices by least squares with any choice of weights. Technometrics, 21(21):489&ndash;498, 1979.
</p>
<p>Gabriel, K.R.(1998): Generalised Bilinear Regression. Biometrika,  85, 3, 689-700.
</p>
<p>Gower y Hand (1996): Biplots. Chapman &amp; Hall.
</p>
<p>Vicente-Villardon, J. L., Galindo, M. P. and Blazquez-Zaballos, A. (2006). Logistic Biplots. Multiple Correspondence Analysis and related methods 491-509.
</p>
<p>Demey, J., Vicente-Villardon, J. L., Galindo, M. P. and Zambrano, A. (2008). Identifying Molecular Markers Associated With Classification Of Genotypes Using External Logistic Biplots. Bioinformatics 24 2832-2838.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+InitialTransform">InitialTransform</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Simple Biplot with arrows
data(Protein)
bip=PCA.Biplot(Protein[,3:11])
plot(bip)

## Biplot with scales on the variables
plot(bip, mode="s", margin=0.2)

# Structure plot (Correlations)
CorrelationCircle(bip)

# Plot of the Variable Contributions
ColContributionPlot(bip, cex=1)


</code></pre>

<hr>
<h2 id='PCA.Bootstrap'>
Principal Components Analysis with bootstrap confidence intervals.
</h2><span id='topic+PCA.Bootstrap'></span>

<h3>Description</h3>

<p>Calculates a Principal Components Analysis with bootstrap confidence intervals for its parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCA.Bootstrap(X, dimens = 2, Scaling = "Standardize columns", B = 1000, type = "np")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PCA.Bootstrap_+3A_x">X</code></td>
<td>

<p>The original raw data matrix
</p>
</td></tr>
<tr><td><code id="PCA.Bootstrap_+3A_dimens">dimens</code></td>
<td>

<p>Desired dimension of the solution.
</p>
</td></tr>
<tr><td><code id="PCA.Bootstrap_+3A_scaling">Scaling</code></td>
<td>

<p>Transformation that should be applied to the raw data.
</p>
</td></tr>
<tr><td><code id="PCA.Bootstrap_+3A_b">B</code></td>
<td>

<p>Number of Bootstrap samples to draw.
</p>
</td></tr>
<tr><td><code id="PCA.Bootstrap_+3A_type">type</code></td>
<td>

<p>Type of Bootstrap (&quot;np&quot;, &quot;pa&quot;, &quot;spper&quot;, &quot;spres&quot;) 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The types of bootstrap used are:
</p>

<dl>
<dt>&quot;np : &quot;</dt><dd><p>Non Parametric</p>
</dd>
<dt>&quot;pa : &quot;</dt><dd><p>parametric (data is obtained from a Multivariate Normal Distribution)</p>
</dd>
<dt>&quot;spper : &quot;</dt><dd><p>Semi-parametric Residuals are permutated</p>
</dd>
<dt>&quot;spres : &quot;</dt><dd><p>Semi-parametric Residuals are resampled</p>
</dd>
</dl>

<p>For the moment, only the non-parametric bootstrap is implemented.
</p>
<p>The Principal Components (eigenvectors) are obtained using bootstrap samples.
</p>
<p>The Row scotes are obtained projecting the completen data matrix into the bootstrap Principal Components. In this way all the individulas have the same number of replications.
</p>


<h3>Value</h3>

<table>
<tr><td><code>Type</code></td>
<td>
<p>The type of Bootstrap used</p>
</td></tr>
<tr><td><code>InitTransform</code></td>
<td>
<p>Transformation of the raw data</p>
</td></tr>
<tr><td><code>InitData</code></td>
<td>
<p>Initial data provided to the function'</p>
</td></tr>
<tr><td><code>TransformedData</code></td>
<td>
<p>Transformed Data</p>
</td></tr>
<tr><td><code>InitialSVD</code></td>
<td>
<p>Singular value decomposition of the transformed data</p>
</td></tr>
<tr><td><code>InitScores</code></td>
<td>
<p>Row Scores for the initial Data</p>
</td></tr>
<tr><td><code>InitCorr</code></td>
<td>
<p>Correlation among variables and Principal Components for the Initial Data</p>
</td></tr>
<tr><td><code>Samples</code></td>
<td>
<p>Matrix containing the members of the Bootstrap Samples</p>
</td></tr>
<tr><td><code>EigVal</code></td>
<td>
<p>Matrix containing the eigenvalues (columns) for each bootstrap sample (columns)</p>
</td></tr>
<tr><td><code>Inertia</code></td>
<td>
<p>Matrix containing the proportions of accounted variance (columns) for each bootstrap sample (columns)</p>
</td></tr>
<tr><td><code>Us</code></td>
<td>
<p>Three-dimensional array containing the left singular vectors for each bootstrap sample</p>
</td></tr>
<tr><td><code>Vs</code></td>
<td>
<p>Three-dimensional array containing the right singular vectors for each bootstrap sample</p>
</td></tr>
<tr><td><code>As</code></td>
<td>
<p>Projection of the bootstrap sampled matrix onto the bottstrap principal components</p>
</td></tr>
<tr><td><code>Bs</code></td>
<td>
<p>Projection of the bootstrap sampled matrix onto the bottstrap principal coordinates</p>
</td></tr>
<tr><td><code>Scores</code></td>
<td>
<p>Projection of the original matrix onto the bootstrap principal components</p>
</td></tr>
<tr><td><code>Struct</code></td>
<td>
<p>Correlation of the Initial Variabblñes and the PCs for each bootstrap sample</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Daudin, J. J., Duby, C., &amp; Trecourt, P. (1988). Stability of principal component analysis studied by the bootstrap method. Statistics: A journal of theoretical and applied statistics, 19(2), 241-258.
</p>
<p>Chateau, F., &amp; Lebart, L. (1996). Assessing sample variability in the visualization techniques related to principal component analysis: bootstrap and alternative simulation methods. COMPSTAT, Physica-Verlag, 205-210.
</p>
<p>Babamoradi, H., van den Berg, F., &amp; Rinnan, Å. (2013). Bootstrap based confidence limits in principal component analysis—A case study. Chemometrics and Intelligent Laboratory Systems, 120, 97-105.
</p>
<p>Fisher, A., Caffo, B., Schwartz, B., &amp; Zipunnikov, V. (2016). Fast, exact bootstrap principal component analysis for p&gt; 1 million. Journal of the American Statistical Association, 111(514), 846-860.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PCA.Biplot">PCA.Biplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: X=wine[,4:21]
grupo=wine$Group
rownames(X)=paste(1:45, grupo, sep="-")
pcaboot=PCA.Bootstrap(X, dimens=2, Scaling = "Standardize columns", B=1000)
plot(pcaboot, ColorInd=as.numeric(grupo))
summary(pcaboot)

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.Binary.Logistic.Biplot'>
Plots the results of a Binary Logistic  Biplot
</h2><span id='topic+plot.Binary.Logistic.Biplot'></span>

<h3>Description</h3>

<p>Plots the results of a Binary Logistic  Biplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Binary.Logistic.Biplot'
plot(x, F1 = 1, F2 = 2, ShowAxis = FALSE, margin = 0, 
PlotVars = TRUE, PlotInd = TRUE, WhatRows = NULL, WhatCols = NULL, 
LabelRows = TRUE, LabelCols = TRUE, ShowBox = FALSE, RowLabels = NULL, 
ColLabels = NULL, RowColors = NULL, ColColors = NULL, Mode = "s", 
TickLength = 0.01, RowCex = 0.8, ColCex = 0.8, SmartLabels = FALSE, 
MinQualityRows = 0, MinQualityCols = 0, dp = 0, PredPoints = 0, 
SizeQualRows = FALSE, SizeQualCols = FALSE, ColorQualRows = FALSE, 
ColorQualCols = FALSE, PchRows = NULL, PchCols = NULL, PlotClus = FALSE, 
TypeClus = "ch", ClustConf = 1, Significant = TRUE, alpha = 0.05, 
Bonferroni = TRUE, PlotSupVars = TRUE, AbbreviateLabels = FALSE, MainTitle = TRUE, Title =
                    NULL, RemoveXYlabs = FALSE, CenterCex = 1.5,  ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_x">x</code></td>
<td>

<p>An object of class Binary.Logistic.Biplot
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_f1">F1</code></td>
<td>

<p>Dimension for the first axis of the representation. Default = 1
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_f2">F2</code></td>
<td>

<p>Dimension for the second axis of the representation. Default = 2
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Should the axis of the representation be shown?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_margin">margin</code></td>
<td>

<p>Margin of the plot as a percentage. It gets some space for the labels.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_plotvars">PlotVars</code></td>
<td>

<p>Should the variables be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_plotind">PlotInd</code></td>
<td>

<p>Should the individuals be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_whatrows">WhatRows</code></td>
<td>

<p>What Rows should be plotted. A binary vector containing which rows (individuals) should be plotted (1) and which should not (0).
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_whatcols">WhatCols</code></td>
<td>

<p>What Columns should be plotted. A binary vector containing which columns (variables) should be plotted (1) and which should not (0).
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_labelrows">LabelRows</code></td>
<td>

<p>Should the individuals be labeled?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_labelcols">LabelCols</code></td>
<td>

<p>Should the individuals be labeled?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_showbox">ShowBox</code></td>
<td>

<p>Should a box around the points be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_rowlabels">RowLabels</code></td>
<td>

<p>A vector of row labels. If NULL the labels contained in the object will be used.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_collabels">ColLabels</code></td>
<td>

<p>A vector of column labels. If NULL the labels contained in the object will be used.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_rowcolors">RowColors</code></td>
<td>

<p>A vector of alternative row colors. 
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_colcolors">ColColors</code></td>
<td>

<p>A vector of alternative column colors. 
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_mode">Mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;h&quot;, &quot;ah&quot; and &quot;s&quot;.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_ticklength">TickLength</code></td>
<td>

<p>Length of the scale ticks for the biplot variables.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_rowcex">RowCex</code></td>
<td>

<p>Cex (Size) of the rows (marks and labels). Can be a single common size for all the points or a vector with individual sizes.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_colcex">ColCex</code></td>
<td>

<p>Cex (Size) of the columns (marks and labels). Can be a single common size for all the points or a vector with individual sizes.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_smartlabels">SmartLabels</code></td>
<td>

<p>Should the labels be placed in a smart way?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_minqualityrows">MinQualityRows</code></td>
<td>

<p>Minimum quality of the rows to be plotted. (Between 0 and 1)
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_minqualitycols">MinQualityCols</code></td>
<td>

<p>Minimum quality of the columns to be plotted. (Between 0 and 1)
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_dp">dp</code></td>
<td>

<p>A vector of variable indices to project all the individuals onto each variable of the vector.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_predpoints">PredPoints</code></td>
<td>

<p>A vector of row indices to project onto each variable.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_sizequalrows">SizeQualRows</code></td>
<td>

<p>Should the size of the Row points be related to its quality?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_sizequalcols">SizeQualCols</code></td>
<td>

<p>Should the size of the Column points be related to its quality?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_colorqualrows">ColorQualRows</code></td>
<td>

<p>Should the color of the Row points be related to its quality?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_colorqualcols">ColorQualCols</code></td>
<td>

<p>Should the color of the Column points be related to its quality?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_pchrows">PchRows</code></td>
<td>

<p>Marks for the rows (numbers). Can be a single common mark for all the points or a vector with individual marks.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_pchcols">PchCols</code></td>
<td>

<p>Marks for the columns (numbers). Can be a single common mark for all the points or a vector with individual marks.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_plotclus">PlotClus</code></td>
<td>

<p>Should the clusters be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of plot for the clusters. (&quot;ch&quot;- Convex Hull, &quot;el&quot;- Ellipse or &quot;st&quot;- Star)
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_clustconf">ClustConf</code></td>
<td>

<p>Percent of points included in the cluster. only the ClusConf percent of the points nearest to the center will be used to calculate the cluster
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_significant">Significant</code></td>
<td>

<p>Should only the significant variables be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_alpha">alpha</code></td>
<td>

<p>Signification level.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_bonferroni">Bonferroni</code></td>
<td>

<p>Should the Bonferroni correction be used?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_plotsupvars">PlotSupVars</code></td>
<td>

<p>Should the Supplementary variables be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_abbreviatelabels">AbbreviateLabels</code></td>
<td>

<p>Should labels be abbreviated?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_maintitle">MainTitle</code></td>
<td>

<p>Should the mail Title be displayed?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_title">Title</code></td>
<td>

<p>Title to display.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_removexylabs">RemoveXYlabs</code></td>
<td>

<p>Should the axis labs be removed?
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_centercex">CenterCex</code></td>
<td>

<p>Size of the point for 0.5 probability.
</p>
</td></tr>
<tr><td><code id="plot.Binary.Logistic.Biplot_+3A_...">...</code></td>
<td>

<p>Any other graphical parameter.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots a biplot for binary data. The Biplot for binary data is taken as the basis of the plot. If there are a mixture of different types of variables (binary, nominal, abundance, ...) are added to the biplot as supplementary parts.
</p>
<p>There are several modes for plotting the biplot.
&quot;p&quot;.- Points (Rows and Columns are represented by points)
</p>
<p>&quot;a&quot; .- Arrows (The traditional representation with points for rows and arrows for columns)
</p>
<p>&quot;b&quot; .- The arrows for the columns are extended to both extremes of the plot and labeled outside the plot area.
</p>
<p>&quot;h&quot; .- The arrows for the columns are extended to the positive extreme of the plot and labeled outside the plot area.
</p>
<p>&quot;ah&quot; .- Same as arrows but labeled outside the plot area.
</p>
<p>&quot;s&quot; .- The directions (or biplot axes) have a graded scale for prediction of the original values.
</p>


<h3>Value</h3>

<p>The plot of the biplot.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardon, J. L., Galindo, M. P. and Blazquez, A. (2006) Logistic Biplots. In Multiple Correspondence Análisis And Related Methods. Grenacre, M &amp; Blasius, J, Eds,  Chapman and Hall, Boca Raton.
</p>
<p>Demey, J., Vicente-Villardon, J. L., Galindo, M.P.  AND Zambrano, A. (2008) Identifying Molecular Markers Associated With Classification Of Genotypes Using External Logistic Biplots. Bioinformatics, 24(24): 2832-2838.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
X=Dataframe2BinaryMatrix(spiders)

logbip=BinaryLogBiplotGD(X,penalization=0.1)
plot(logbip, Mode="a")
summary(logbip)
</code></pre>

<hr>
<h2 id='plot.CA.sol'>
Plot the solution of a Coorespondence Analysis
</h2><span id='topic+plot.CA.sol'></span>

<h3>Description</h3>

<p>Plots the solution of a Correspondence Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CA.sol'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.CA.sol_+3A_x">x</code></td>
<td>

<p>A CA.sol object
</p>
</td></tr>
<tr><td><code id="plot.CA.sol_+3A_...">...</code></td>
<td>

<p>Any other biplot and graphical parameters
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the solution of a Correspondence Analysis
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Add some references here
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.ContinuousBiplot">plot.ContinuousBiplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(riano)
Sp=riano[,3:15]
cabip=CA(Sp)
plot(cabip)
</code></pre>

<hr>
<h2 id='plot.Canonical.Biplot'>
Plots a Canonical Biplot
</h2><span id='topic+plot.Canonical.Biplot'></span>

<h3>Description</h3>

<p>Plots a Canonical Biplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Canonical.Biplot'
plot(x, A1 = 1, A2 = 2, ScaleGraph = TRUE, PlotGroups =
                    TRUE, PlotVars = TRUE, PlotInd = TRUE, WhatInds =
                    NULL, WhatVars = NULL, WhatGroups = NULL, IndLabels =
                    NULL, VarLabels = NULL, GroupLabels = NULL,
                    AbbreviateLabels = FALSE, LabelInd = TRUE, LabelVars =
                    TRUE, CexGroup = 1, PchGroup = 16, margin = 0.1,
                    AddLegend = FALSE, ShowAxes = FALSE, LabelAxes =
                    FALSE, LabelGroups = TRUE, PlotCircle = TRUE,
                    ConvexHulls = FALSE, TypeCircle = "M", ColorGroups =
                    NULL, ColorVars = NULL, LegendPos = "topright",
                    ColorInd = NULL, voronoi = TRUE, mode = "a", TypeScale
                    = "Complete", ValuesScale = "Original", MinQualityVars
                    = 0, dpg = 0, dpi = 0, dp = 0, PredPoints = 0,
                    PlotAxis = FALSE, CexInd = NULL, CexVar = NULL, PchInd
                    = NULL, PchVar = NULL, ColorVar = NULL, ShowAxis =
                    FALSE, VoronoiColor = "black", ShowBox = FALSE,
                    ShowTitle = TRUE, PlotClus = FALSE, TypeClus = "ch",
                    ClustConf = 1, ClustCenters = FALSE, UseClusterColors
                    = TRUE, CexClustCenters = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Canonical.Biplot_+3A_x">x</code></td>
<td>

<p>An object of class &quot;Canonical.Biplot&quot;
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_a1">A1</code></td>
<td>

<p>Dimension for the first axis. 1 is the default.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_a2">A2</code></td>
<td>

<p>Dimension for the second axis. 2 is the default.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_scalegraph">ScaleGraph</code></td>
<td>

<p>Reescale the coordinates to optimal matching.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_plotgroups">PlotGroups</code></td>
<td>

<p>Shoud the group centers be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_plotvars">PlotVars</code></td>
<td>

<p>Should the variables be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_plotind">PlotInd</code></td>
<td>

<p>Should the individuals be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_whatinds">WhatInds</code></td>
<td>

<p>Logical vector to control what individuals (Rows) are plotted. (Can be also a binary vector)
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_whatvars">WhatVars</code></td>
<td>

<p>Logical vector to control what variables (Columns) are plotted. (Can be also a binary vector)
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_whatgroups">WhatGroups</code></td>
<td>

<p>Logical vector to control what groups are plotted. (Can be also a binary vector)
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_indlabels">IndLabels</code></td>
<td>

<p>A set of labels for the individuals. If NULL the default object labels are used
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_varlabels">VarLabels</code></td>
<td>

<p>A set of labels for the variables. If NULL the default object labels are used
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_grouplabels">GroupLabels</code></td>
<td>

<p>A set of labels for the groups. If NULL the default object labels are used
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_abbreviatelabels">AbbreviateLabels</code></td>
<td>

<p>Should labels be abbreviated?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_labelind">LabelInd</code></td>
<td>

<p>Should the individuals be labeled?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_labelvars">LabelVars</code></td>
<td>

<p>Should the variables be labeled?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_cexgroup">CexGroup</code></td>
<td>

<p>Sizes of the points for the groups
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_pchgroup">PchGroup</code></td>
<td>

<p>Markers for the group
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_margin">margin</code></td>
<td>

<p>margin for the graph
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_addlegend">AddLegend</code></td>
<td>

<p>Should a legend with the groups be added?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_showaxes">ShowAxes</code></td>
<td>

<p>Should outside axes be shown?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_labelaxes">LabelAxes</code></td>
<td>

<p>Should outside axes be labelled?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_labelgroups">LabelGroups</code></td>
<td>

<p>Should the groups be labeled?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_plotcircle">PlotCircle</code></td>
<td>

<p>Should the confidence regions for the groups be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_convexhulls">ConvexHulls</code></td>
<td>

<p>Should the convex hulls containing the individuals for each group be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_typecircle">TypeCircle</code></td>
<td>

<p>Type of confidence region: Univariate (U), Bonferroni(B), Multivariate (M) or Classical (C)
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_colorgroups">ColorGroups</code></td>
<td>

<p>User colors for the groups. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_colorvars">ColorVars</code></td>
<td>

<p>User colors for the variables. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_legendpos">LegendPos</code></td>
<td>

<p>Position of the legend.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_colorind">ColorInd</code></td>
<td>

<p>User colors for the individuals. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_voronoi">voronoi</code></td>
<td>

<p>Should the voronoi diagram with the prediction regións for each group be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_mode">mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;h&quot;, &quot;ah&quot; and &quot;s&quot;.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scale to use : &quot;Complete&quot;, &quot;StdDev&quot; or &quot;BoxPlot&quot;
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>Values to show on the scale: &quot;Original&quot; or &quot;Transformed&quot;
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_minqualityvars">MinQualityVars</code></td>
<td>

<p>Minimum quality of representation for a variable to be plotted
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_dpg">dpg</code></td>
<td>

<p>A set of indices with the variables that will show the projections of the gorups 
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_dpi">dpi</code></td>
<td>

<p>A set of indices with the individuasl that will show the projections on the variables 
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_dp">dp</code></td>
<td>

<p>A set of indices with the variables that will show the projections of the individuals
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_predpoints">PredPoints</code></td>
<td>

<p>A vector with integers. The group centers listed in the vector are projected onto all the variables.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_plotaxis">PlotAxis</code></td>
<td>

<p>Not Used
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_cexind">CexInd</code></td>
<td>

<p>Size of the points for individuals.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_cexvar">CexVar</code></td>
<td>

<p>Size of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_pchind">PchInd</code></td>
<td>

<p>Marhers of the points for individuals.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_pchvar">PchVar</code></td>
<td>

<p>Markers of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_colorvar">ColorVar</code></td>
<td>

<p>Colors of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Should axis scales be shown?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_voronoicolor">VoronoiColor</code></td>
<td>

<p>Color for the Voronoi diagram
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_showbox">ShowBox</code></td>
<td>

<p>Should a box around the poitns be plotted?
</p>
</td></tr> 
<tr><td><code id="plot.Canonical.Biplot_+3A_showtitle">ShowTitle</code></td>
<td>

<p>Should the title be shown?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_plotclus">PlotClus</code></td>
<td>

<p>Should the clusters be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of plot for the clusters. (&quot;ch&quot;- Convex Hull, &quot;el&quot;- Ellipse or &quot;st&quot;- Star)
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_clustconf">ClustConf</code></td>
<td>

<p>Percent of points included in the cluster. only the ClusConf percent of the points nearest to the center will be used to calculate the cluster
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_clustcenters">ClustCenters</code></td>
<td>

<p>Should the cluster centers be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_useclustercolors">UseClusterColors</code></td>
<td>

<p>Should the cluster colors be used in the plot
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_cexclustcenters">CexClustCenters</code></td>
<td>

<p>Size of the cluster centres
</p>
</td></tr>
<tr><td><code id="plot.Canonical.Biplot_+3A_...">...</code></td>
<td>

<p>Any other graphical parameters 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function plots the results of a Canononical Biplot. The coordinates for Groups, Individuals and Variables can be shown or not on the plot, each of the three can also be labeled separately.
The are parameters to control the way each different set of coordinates is plotted and labeled.
</p>
<p>There are several modes for plotting the biplot.
</p>
<p>&quot;p&quot;.- Points (Rows and Columns are represented by points)
</p>
<p>&quot;a&quot; .- Arrows (The traditional representation with points for rows and arrows for columns)
</p>
<p>&quot;b&quot; .- The arrows for the columns are extended to both extremes of the plot and labeled outside the plot area.
</p>
<p>&quot;h&quot; .- The arrows for the columns are extended to the positive extreme of the plot and labeled outside the plot area.
</p>
<p>&quot;ah&quot; .- Same as arrows but labeled outside the plot area.
</p>
<p>&quot;s&quot; .- The directions (or biplot axes) have a graded scale for prediction of the original values.
</p>
<p>The <em>TypeScale</em> argument applies only to the <em>&quot;s&quot;</em> mode. There are three types:
</p>
<p>&quot;Complete&quot; .- An equally spaced scale covering the whole range of the data is calculates.
</p>
<p>&quot;StdDev&quot; .- Mean with one, two and three stadard deviations 
</p>
<p>&quot;BoxPlot&quot; .- Box-Plot like Scale (Median, 25 and 75 percentiles, maximum and minimum values.)
</p>
<p>The <em>ValuesScale</em> argument applies only to the <em>&quot;s&quot;</em> mode and controls if the labels show the  <em>Original</em> ot <em>Transformed</em> values.
</p>
<p>Some of the initial transformations are not compatible with some of the types of biplots and scales. For example, It is not possible to recover by projection the original values when you double centre de data. In that case you have the residuals for interaction and only the transformed values make sense.
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Amaro, I. R., Vicente-Villardon, J. L., &amp; Galindo-Villardon, M. P. (2004). Manova Biplot para arreglos de tratamientos con dos factores basado en modelos lineales generales multivariantes. Interciencia, 29(1), 26-32.
</p>
<p>Varas, M. J., Vicente-Tavera, S., Molina, E., &amp; Vicente-Villardon, J. L. (2005). Role of canonical biplot method in the study of building stones: an example from Spanish monumental heritage. Environmetrics, 16(4), 405-419.
</p>
<p>Santana, M. A., Romay, G., Matehus, J., Villardon, J. L., &amp; Demey, J. R. (2009). simple and low-cost strategy for micropropagation of cassava (Manihot esculenta Crantz). African Journal of Biotechnology, 8(16).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X=wine[,4:21]
canbip=CanonicalBiplot(X, group=wine$Group)
plot(canbip, TypeCircle="U")
</code></pre>

<hr>
<h2 id='plot.CanonicalDistanceAnalysis'>
Plots a Canonical Distance Analysis
</h2><span id='topic+plot.CanonicalDistanceAnalysis'></span>

<h3>Description</h3>

<p>Plots a Canonical Distance Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CanonicalDistanceAnalysis'
plot(x, A1 = 1, A2 = 2, ScaleGraph = TRUE, 
ShowAxis = FALSE, ShowAxes = FALSE, LabelAxes = TRUE, margin = 0.1, 
PlotAxis = FALSE, ShowBox = TRUE, PlotGroups = TRUE, LabelGroups = TRUE, 
CexGroup = 1.5, PchGroup = 16, ColorGroup = NULL, voronoi = TRUE, 
VoronoiColor = "black", PlotInd = TRUE, LabelInd = TRUE, CexInd = 0.8, 
PchInd = 3, ColorInd = NULL, WhatInds = NULL, IndLabels = NULL, 
PlotVars = TRUE, LabelVar = TRUE, CexVar = NULL, PchVar = NULL, 
ColorVar = NULL, WhatVars = NULL, VarLabels = NULL, mode = "a", 
TypeScale = "Complete", ValuesScale = "Original", SmartLabels = TRUE, 
AddLegend = TRUE, LegendPos = "topright", PlotCircle = TRUE, 
ConvexHulls = FALSE, TypeCircle = "M", MinQualityVars = 0, dpg = 0, 
dpi = 0, PredPoints = 0, PlotClus = TRUE, TypeClus = "ch", ClustConf = 1, 
CexClustCenters = 1, ClustCenters = FALSE, UseClusterColors = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_x">x</code></td>
<td>

<p>An object of class &quot;CanonicalDistanceAnalysis&quot;
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_a1">A1</code></td>
<td>

<p>Dimension for the first axis. 1 is the default.
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_a2">A2</code></td>
<td>

<p>Dimension for the second axis. 2 is the default.
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_scalegraph">ScaleGraph</code></td>
<td>

<p>Reescale the coordinates to optimal matching.
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Should the axis be shown?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_showaxes">ShowAxes</code></td>
<td>

<p>Not used
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_labelaxes">LabelAxes</code></td>
<td>

<p>Shoud the axis be labelled?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_margin">margin</code></td>
<td>

<p>Margin of the plot
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_plotaxis">PlotAxis</code></td>
<td>

<p>Should the axis be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_showbox">ShowBox</code></td>
<td>

<p>Show a box around the plot
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_plotgroups">PlotGroups</code></td>
<td>

<p>Should the groups be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_labelgroups">LabelGroups</code></td>
<td>

<p>Should the groups be labelled?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_cexgroup">CexGroup</code></td>
<td>

<p>Sizes for the groups
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_pchgroup">PchGroup</code></td>
<td>

<p>Marks for the groups
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_colorgroup">ColorGroup</code></td>
<td>

<p>Colors for the groups
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_voronoi">voronoi</code></td>
<td>

<p>Should a voronoi diagram separating the groups be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_voronoicolor">VoronoiColor</code></td>
<td>

<p>Color for the voronoi diagram
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_plotind">PlotInd</code></td>
<td>

<p>Should the individuals be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_labelind">LabelInd</code></td>
<td>

<p>Should the individuals be labelled?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_cexind">CexInd</code></td>
<td>

<p>Sizes for the individuals
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_pchind">PchInd</code></td>
<td>

<p>Marks for the individuals
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_colorind">ColorInd</code></td>
<td>

<p>Colors for the individuals
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_whatinds">WhatInds</code></td>
<td>

<p>What indivduals are plotted
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_indlabels">IndLabels</code></td>
<td>

<p>Labels for the individuals
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_plotvars">PlotVars</code></td>
<td>

<p>Should the variables be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_labelvar">LabelVar</code></td>
<td>

<p>Should the variables be labelled?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_cexvar">CexVar</code></td>
<td>

<p>Sizes for the variables
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_pchvar">PchVar</code></td>
<td>

<p>Marks for the variables
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_colorvar">ColorVar</code></td>
<td>

<p>User colors for the variables. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_whatvars">WhatVars</code></td>
<td>

<p>What Variables are plotted
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_varlabels">VarLabels</code></td>
<td>

<p>User labels for the variables
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_mode">mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;h&quot;, &quot;ah&quot; and &quot;s&quot;.
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scale to use : &quot;Complete&quot;, &quot;StdDev&quot; or &quot;BoxPlot&quot;
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>Values to show on the scale: &quot;Original&quot; or &quot;Transformed&quot;
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_smartlabels">SmartLabels</code></td>
<td>

<p>Plot the labels in a smart way
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_addlegend">AddLegend</code></td>
<td>

<p>Should a legend be added?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_legendpos">LegendPos</code></td>
<td>

<p>Position of the legend
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_plotcircle">PlotCircle</code></td>
<td>

<p>Should the confidence regions for the groups be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_convexhulls">ConvexHulls</code></td>
<td>

<p>Should the convex hulls containing the individuals for each group be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_typecircle">TypeCircle</code></td>
<td>

<p>Type of confidence region: Univariate (U), Bonferroni(B), Multivariate (M) or Classical (C)
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_minqualityvars">MinQualityVars</code></td>
<td>

<p>Minimum quality of representation for a variable to be plotted
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_dpg">dpg</code></td>
<td>

<p>A set of indices with the variables that will show the projections of the gorups 
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_dpi">dpi</code></td>
<td>

<p>A set of indices with the individuasl that will show the projections on the variables 
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_predpoints">PredPoints</code></td>
<td>

<p>A vector with integers. The group centers listed in the vector are projected onto all the variables.
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_plotclus">PlotClus</code></td>
<td>

<p>Should the clusters be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of plot for the clusters. (&quot;ch&quot;- Convex Hull, &quot;el&quot;- Ellipse or &quot;st&quot;- Star)
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_clustconf">ClustConf</code></td>
<td>

<p>Percent of points included in the cluster. only the ClusConf percent of the points nearest to the center will be used to calculate the cluster
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_cexclustcenters">CexClustCenters</code></td>
<td>

<p>SIze of the cluster centers.
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_clustcenters">ClustCenters</code></td>
<td>

<p>Should the cluster centers be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_useclustercolors">UseClusterColors</code></td>
<td>

<p>Should the cluster colors be used in the plot
</p>
</td></tr>
<tr><td><code id="plot.CanonicalDistanceAnalysis_+3A_...">...</code></td>
<td>

<p>Any other graphical parameters 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots a Canonical Distance Analysis
</p>


<h3>Value</h3>

<p>The plot of a Canonical Distance Analysis
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Gower, J. C. and Krzanowski, W. J. (1999). Analysis of distance for structured multivariate data and extensions to multivariate analysis of variance. Journal of the Royal Statistical Society: Series C (Applied Statistics), 48(4):505-519.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.Canonical.Biplot">plot.Canonical.Biplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='plot.CCA.sol'>
Plots the solution of a Canonical Correspondence Analysisis
</h2><span id='topic+plot.CCA.sol'></span>

<h3>Description</h3>

<p>Plots the solution of a Canonical Correspondence Analysisis using similar parameters to the continuous biplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CCA.sol'
plot(x, A1 = 1, A2 = 2, ShowAxis = FALSE, margin = 0,
                 PlotSites = TRUE, PlotSpecies = TRUE, PlotEnv = TRUE,
                 LabelSites = TRUE, LabelSpecies = TRUE, LabelEnv =
                 TRUE, TypeSites = "wa", SpeciesQuality = FALSE,
                 MinQualityVars = 0.3, dp = 0, pr = 0, PlotAxis =
                 FALSE, TypeScale = "Complete", ValuesScale =
                 "Original", mode = "a", CexSites = NULL, CexSpecies =
                 NULL, CexVar = NULL, ColorSites = NULL, ColorSpecies =
                 NULL, ColorVar = NULL, PchSites = NULL, PchSpecies =
                 NULL, PchVar = NULL, SizeQualSites = FALSE,
                 SizeQualSpecies = FALSE, SizeQualVars = FALSE,
                 ColorQualSites = FALSE, ColorQualSpecies = FALSE,
                 ColorQualVars = FALSE, SmartLabels = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.CCA.sol_+3A_x">x</code></td>
<td>

<p>The results of a CCA model
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_a1">A1</code></td>
<td>

<p>Dimension for the first axis
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_a2">A2</code></td>
<td>

<p>Dimension for the second axis
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Logical variable to control if the coordinate axes should appear in the plot. The default value is FALSE because for most of the biplots its presence is irrelevant.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_margin">margin</code></td>
<td>

<p>Margin for the labels in some of the biplot modes (percentage of the plot width). Default is 0. Increase the value if the labels are not completely plotted.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_plotsites">PlotSites</code></td>
<td>

<p>Should the sites be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_plotspecies">PlotSpecies</code></td>
<td>

<p>Should the species be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_plotenv">PlotEnv</code></td>
<td>

<p>Should the environmental variables be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_labelsites">LabelSites</code></td>
<td>

<p>Labels for the sites
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_labelspecies">LabelSpecies</code></td>
<td>

<p>Labels for the species
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_labelenv">LabelEnv</code></td>
<td>

<p>Labels for the environmental variables.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_typesites">TypeSites</code></td>
<td>

<p>Type for the sites plot
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_speciesquality">SpeciesQuality</code></td>
<td>

<p>Quality for the species
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_minqualityvars">MinQualityVars</code></td>
<td>

<p>Minimum quality to plot a variable
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_dp">dp</code></td>
<td>

<p>A set of indices with the variables that will show the projections of the individuals.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_pr">pr</code></td>
<td>

<p>A set of indices with the individuals to show the projections on the variables.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_plotaxis">PlotAxis</code></td>
<td>

<p>Should the axis be plotted?
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scale to use : &quot;Complete&quot;, &quot;StdDev&quot; or &quot;BoxPlot&quot;
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>Values to show on the scale: &quot;Original&quot; or &quot;Transformed&quot;
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_mode">mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;h&quot;, &quot;ah&quot; and &quot;s&quot;.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_cexsites">CexSites</code></td>
<td>

<p>Size for the symbols and labels of the sites. Can be a single common size for all the points or a vector with individual sizes.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_cexspecies">CexSpecies</code></td>
<td>

<p>Size for the symbols and labels of the species. Can be a single common size for all the points or a vector with individual sizes.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_cexvar">CexVar</code></td>
<td>

<p>Size for the symbols and labels of the variables. Can be a single common size for all the points or a vector with individual sizes.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_colorsites">ColorSites</code></td>
<td>

<p>Color for the symbols and labels of the sites. Can be a single common color for all the points or a vector with individual colors.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_colorspecies">ColorSpecies</code></td>
<td>

<p>Color for the symbols and labels of the species. Can be a single common color for all the points or a vector with individual colors.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_colorvar">ColorVar</code></td>
<td>

<p>Color for the symbols and labels of the variables. Can be a single common color for all the points or a vector with individual colors.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_pchsites">PchSites</code></td>
<td>

<p>Symbol for the sites points. See <code>help(points)</code> for details.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_pchspecies">PchSpecies</code></td>
<td>

<p>Symbol for the species points. See <code>help(points)</code> for details.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_pchvar">PchVar</code></td>
<td>

<p>Symbol for the variables points. See <code>help(points)</code> for details.
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_sizequalsites">SizeQualSites</code></td>
<td>

<p>Should the size of the site points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_sizequalspecies">SizeQualSpecies</code></td>
<td>

<p>Should the size of the species points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_sizequalvars">SizeQualVars</code></td>
<td>

<p>Should the size of the variables points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_colorqualsites">ColorQualSites</code></td>
<td>

<p>Should the color of the sites points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_colorqualspecies">ColorQualSpecies</code></td>
<td>

<p>Should the color of the species points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_colorqualvars">ColorQualVars</code></td>
<td>

<p>Should the color of the variables points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_smartlabels">SmartLabels</code></td>
<td>

<p>Plot the labels in a smart way
</p>
</td></tr>
<tr><td><code id="plot.CCA.sol_+3A_...">...</code></td>
<td>

<p>Aditional graphical parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plotting procedure is similar to the one used for continuous biplots including the calibration of the environmental variables.
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>CCA
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.ContinuousBiplot">plot.ContinuousBiplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
</code></pre>

<hr>
<h2 id='plot.ContinuousBiplot'>
Plots a biplot for continuous data.
</h2><span id='topic+plot.ContinuousBiplot'></span>

<h3>Description</h3>

<p>Plots a biplot for continuous data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ContinuousBiplot'
plot(x, A1 = 1, A2 = 2, ShowAxis = FALSE, margin = 0,
                    PlotVars = TRUE, PlotInd = TRUE, WhatInds = NULL,
                    WhatVars = NULL, LabelVars = TRUE, LabelInd = TRUE,
                    IndLabels = NULL, VarLabels = NULL, mode = "a", CexInd
                    = NULL, CexVar = NULL, ColorInd = NULL, ColorVar =
                    NULL, LabelPos = 1, SmartLabels = FALSE,
                    AbbreviateLabels = FALSE, MinQualityInds = 0,
                    MinQualityVars = 0, dp = 0, PredPoints = 0, PlotAxis =
                    FALSE, TypeScale = "Complete", ValuesScale =
                    "Original", SizeQualInd = FALSE, SizeQualVars = FALSE,
                    ColorQualInd = FALSE, ColorQualVars = FALSE, PchInd =
                    NULL, PchVar = NULL, PlotClus = FALSE, TypeClus =
                    "ch", ClustConf = 1, ClustLegend = FALSE,
                    ClustLegendPos = "topright", ClustCenters = FALSE,
                    UseClusterColors = TRUE, CexClustCenters = 1,
                    PlotSupVars = TRUE, SupMode = "a", ShowBox = FALSE,
                    nticks = 5, NonSelectedGray = FALSE, PlotUnitCircle =
                    TRUE, PlotContribFA = TRUE, AddArrow = FALSE,
                    ColorSupContVars = "red", ColorSupBinVars = "red",
                    ColorSupOrdVars = "red", ModeSupContVars="a", 
                    ModeSupBinVars="a", ModeSupOrdVars="a", 
                    WhatSupBinVars = NULL, Title = NULL, Xlab = NULL, 
                    Ylab = NULL, add = FALSE, PlotTrajVars = FALSE, 
                    PlotTrajInds = FALSE, LabelTraj = "end", Limits = NULL,
                    PlotSupInds = FALSE, WhatSupInds = NULL,
                    ColorSupInd = "black", CexSupInd = 0.8, PchSupInd =
                   16, LabelSupInd = TRUE, PredSupPoints = 0,  CexScale =
                    0.5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.ContinuousBiplot_+3A_x">x</code></td>
<td>

<p>An object of class &quot;Biplot&quot;
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_a1">A1</code></td>
<td>

<p>Dimension for the first axis. 1 is the default.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_a2">A2</code></td>
<td>

<p>Dimension for the second axis. 2 is the default.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Logical variable to control if the coordinate axes should appear in the plot. The default value is FALSE because for most of the biplots its presence is irrelevant.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_margin">margin</code></td>
<td>

<p>Margin for the labels in some of the biplot modes (percentage of the plot width). Default is 0. Increase the value if the labels are not completely plotted.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_plotvars">PlotVars</code></td>
<td>

<p>Logical to control if the Variables (Columns) are plotted.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_plotind">PlotInd</code></td>
<td>

<p>Logical to control if the Individuals (Rows) are plotted.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_whatinds">WhatInds</code></td>
<td>

<p>Logical vector to control what individuals (Rows) are plotted. (Can be also a binary vector)
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_whatvars">WhatVars</code></td>
<td>

<p>Logical vector to control what variables (Columns) are plotted. (Can be also a binary vector)
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_labelvars">LabelVars</code></td>
<td>

<p>Logical to control if the labels for the Variables are shown
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_labelind">LabelInd</code></td>
<td>

<p>Logical to control if the labels for the individuals are shown
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_indlabels">IndLabels</code></td>
<td>

<p>A set of labels for the individuals. If NULL the default object labels are used
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_varlabels">VarLabels</code></td>
<td>

<p>A set of labels for the variables. If NULL the default object labels are used
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_mode">mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;h&quot;, &quot;ah&quot; and &quot;s&quot;.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_cexind">CexInd</code></td>
<td>

<p>Size for the symbols and labels of the individuals. Can be a single common size for all the points or a vector with individual sizes.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_cexvar">CexVar</code></td>
<td>

<p>Size for the symbols and labels of the variables. Can be a single common size for all the points or a vector with individual sizes.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_colorind">ColorInd</code></td>
<td>

<p>Color for the symbols and labels of the individuals. Can be a single common color for all the points or a vector with individual colors.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_colorvar">ColorVar</code></td>
<td>

<p>Color for the symbols and labels of the variables. Can be a single common color for all the points or a vector with individual colors.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_labelpos">LabelPos</code></td>
<td>

<p>Position of the labels in relation to the point. (Se the graphical parameter <code>pos</code> )
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_smartlabels">SmartLabels</code></td>
<td>

<p>Plot the labels in a smart way
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_abbreviatelabels">AbbreviateLabels</code></td>
<td>

<p>Should labels be abbreviated?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_minqualityinds">MinQualityInds</code></td>
<td>

<p>Minimum quality of representation for an individual to be plotted.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_minqualityvars">MinQualityVars</code></td>
<td>

<p>Minimum quality of representation for a variable to be plotted.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_dp">dp</code></td>
<td>

<p>A set of indices with the variables that will show the projections of the individuals.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_predpoints">PredPoints</code></td>
<td>

<p>A vector with integers. The row points listed in the vector are projected onto all the variables.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_plotaxis">PlotAxis</code></td>
<td>

<p>Not Used
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scale to use : &quot;Complete&quot;, &quot;StdDev&quot; or &quot;BoxPlot&quot;
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>Values to show on the scale: &quot;Original&quot; or &quot;Transformed&quot;
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_sizequalind">SizeQualInd</code></td>
<td>

<p>Should the size of the row points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_sizequalvars">SizeQualVars</code></td>
<td>

<p>Should the size of the column points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_colorqualind">ColorQualInd</code></td>
<td>

<p>Should the color of the row points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_colorqualvars">ColorQualVars</code></td>
<td>

<p>Should the color of the column points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_pchind">PchInd</code></td>
<td>

<p>Symbol for the row points. See <code>help(points)</code> for details.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_pchvar">PchVar</code></td>
<td>

<p>Symbol for the column points. See <code>help(points)</code> for details.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_plotclus">PlotClus</code></td>
<td>

<p>Should the clusters be plotted?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of plot for the clusters. (&quot;ch&quot;- Convex Hull, &quot;el&quot;- Ellipse or &quot;st&quot;- Star)
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_clustconf">ClustConf</code></td>
<td>

<p>Percent of points included in the cluster. only the ClusConf percent of the points nearest to the center will be used to calculate the cluster
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_clustlegend">ClustLegend</code></td>
<td>

<p>Should a legend for the clusters be plotted? Default FALSE
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_clustlegendpos">ClustLegendPos</code></td>
<td>

<p>Position of the legend for the clusters. Default &quot;topright&quot;
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_clustcenters">ClustCenters</code></td>
<td>

<p>Should the cluster centers be plotted
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_useclustercolors">UseClusterColors</code></td>
<td>

<p>Should the cluster colors be used in the plot
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_cexclustcenters">CexClustCenters</code></td>
<td>

<p>Size of the cluster centres
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_plotsupvars">PlotSupVars</code></td>
<td>

<p>Should the supplementary variables be plotted?
</p>
</td></tr>   
<tr><td><code id="plot.ContinuousBiplot_+3A_supmode">SupMode</code></td>
<td>

<p>Mode of the supplementary variables.
</p>
</td></tr> 
<tr><td><code id="plot.ContinuousBiplot_+3A_showbox">ShowBox</code></td>
<td>

<p>Should a box around the poitns be plotted?
</p>
</td></tr> 
<tr><td><code id="plot.ContinuousBiplot_+3A_nticks">nticks</code></td>
<td>

<p>Number of ticks for the representation of the variables
</p>
</td></tr> 
<tr><td><code id="plot.ContinuousBiplot_+3A_nonselectedgray">NonSelectedGray</code></td>
<td>

<p>The nonselected individuals and variables aplotted in light gray colors
</p>
</td></tr> 
<tr><td><code id="plot.ContinuousBiplot_+3A_plotunitcircle">PlotUnitCircle</code></td>
<td>

<p>Plot the unit circle in the biplot for a Factor Analysis in which the lenght of the column arrows is smaller than 1 and is the quality of representation.
</p>
</td></tr> 
<tr><td><code id="plot.ContinuousBiplot_+3A_plotcontribfa">PlotContribFA</code></td>
<td>

<p>Plot circles in the biplot for a Factor Analysis with different values of the quality of representation.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_addarrow">AddArrow</code></td>
<td>

<p>Add an arrow to the representation of other modes of the biplot.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_colorsupcontvars">ColorSupContVars</code></td>
<td>

<p>Colors for the continuous supplementary variables.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_colorsupbinvars">ColorSupBinVars</code></td>
<td>

<p>Colors for the binary supplementary variables.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_colorsupordvars">ColorSupOrdVars</code></td>
<td>

<p>Colors for the ordinal supplementary variables.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_modesupcontvars">ModeSupContVars</code></td>
<td>

<p>Mode for the continuous supplementary variables.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_modesupbinvars">ModeSupBinVars</code></td>
<td>

<p>Mode for the binary supplementary variables.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_modesupordvars">ModeSupOrdVars</code></td>
<td>

<p>Mode for the ordinal supplementary variables.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_whatsupbinvars">WhatSupBinVars</code></td>
<td>

<p>What supplementary binary variables should be plotted?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_title">Title</code></td>
<td>

<p>Title of the plot.
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_xlab">Xlab</code></td>
<td>

<p>Label for the X axis
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_ylab">Ylab</code></td>
<td>

<p>Label for the Y axis
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_add">add</code></td>
<td>

<p>Should the plot be added to an existing plot?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_plottrajvars">PlotTrajVars</code></td>
<td>

<p>Plot trajectories for the variables (when appropriate)?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_plottrajinds">PlotTrajInds</code></td>
<td>

<p>Plot trajectories for the individuals (when appropriate)?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_labeltraj">LabelTraj</code></td>
<td>

<p>Label trajectories for the variables (when appropriate)?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_limits">Limits</code></td>
<td>

<p>Limits of the axis for the plot
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_plotsupinds">PlotSupInds</code></td>
<td>

<p>Should the supplementary individuals be plotted?
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_whatsupinds">WhatSupInds</code></td>
<td>

<p>What supplementary individuals are going to be plotted
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_colorsupind">ColorSupInd</code></td>
<td>

<p>Colors for the supplementary individuals
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_cexsupind">CexSupInd</code></td>
<td>

<p>Sizes for the supplementary individuals
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_pchsupind">PchSupInd</code></td>
<td>

<p>Symbols for the supplementary individuals
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_labelsupind">LabelSupInd</code></td>
<td>

<p>Labels for the supplementary individuals
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_predsuppoints">PredSupPoints</code></td>
<td>

<p>Predictions for the supplementary individuals
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_cexscale">CexScale</code></td>
<td>

<p>Sizes of the scales
</p>
</td></tr>
<tr><td><code id="plot.ContinuousBiplot_+3A_...">...</code></td>
<td>

<p>Any other graphical parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots a biplot for continuous data. The Biplot for continuous data is taken as the basis of the plot. If there are a mixture of different types of variables (binary, nominal, abundance, ...) are added to the biplot as supplementary parts.
</p>
<p>There are several modes for plotting the biplot.
&quot;p&quot;.- Points (Rows and Columns are represented by points)
</p>
<p>&quot;a&quot; .- Arrows (The traditional representation with points for rows and arrows for columns)
</p>
<p>&quot;b&quot; .- The arrows for the columns are extended to both extremes of the plot and labeled outside the plot area.
</p>
<p>&quot;h&quot; .- The arrows for the columns are extended to the positive extreme of the plot and labeled outside the plot area.
</p>
<p>&quot;ah&quot; .- Same as arrows but labeled outside the plot area.
</p>
<p>&quot;s&quot; .- The directions (or biplot axes) have a graded scale for prediction of the original values.
</p>
<p>The <em>TypeScale</em> argument applies only to the <em>&quot;s&quot;</em> mode. There are three types:
</p>
<p>&quot;Complete&quot; .- An equally spaced scale covering the whole range of the data is calculates.
</p>
<p>&quot;StdDev&quot; .- Mean with one, two and three stadard deviations 
</p>
<p>&quot;BoxPlot&quot; .- Box-Plot like Scale (Median, 25 and 75 percentiles, maximum and minimum values.)
</p>
<p>The <em>ValuesScale</em> argument applies only to the <em>&quot;s&quot;</em> mode and controls if the labels show the  <em>Original</em> ot <em>Transformed</em> values.
</p>
<p>Some of the initial transformations are not compatible with some of the types of biplots and scales. For example, It is not possible to recover by projection the original values when you double centre de data. In that case you have the residuals for interaction and only the transformed values make sense.
</p>
<p>It is possible to associate the color and the size of the points with the quality of representation. Bigger points correspond to better representation quality.
</p>


<h3>Value</h3>

<p>No value Returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Gabriel, K. R. (1971). The biplot graphic display of matrices with application to principal component analysis. Biometrika, 58(3), 453-467.
</p>
<p>Galindo Villardon, M. (1986). Una alternativa de representacion simultanea: HJ-Biplot. Questiio. 1986, vol. 10, num. 1.
</p>
<p>Vicente-Villardon, J. L., Galindo Villardon, M. P., &amp; Blazquez Zaballos, A. (2006). Logistic biplots. Multiple correspondence analysis and related methods. London: Chapman &amp; Hall, 503-521.
</p>
<p>Gower, J. C., &amp; Hand, D. J. (1995). Biplots (Vol. 54). CRC Press.
</p>
<p>Gower, J. C., Lubbe, S. G., &amp; Le Roux, N. J. (2011). Understanding biplots. John Wiley &amp; Sons.
</p>
<p>Blasius, J., Eilers, P. H., &amp; Gower, J. (2009). Better biplots. Computational Statistics &amp; Data Analysis, 53(8), 3145-3158.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Protein)
bip=PCA.Biplot(Protein[,3:11])
plot(bip, mode="s", margin=0.2, ShowAxis=FALSE)
</code></pre>

<hr>
<h2 id='plot.CVA'>
Plot of a Canonical Variate Analysis
</h2><span id='topic+plot.CVA'></span>

<h3>Description</h3>

<p>Plot of a Canonical Variate Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CVA'
plot(x, A1 = 1, A2 = 2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.CVA_+3A_x">x</code></td>
<td>

<p>Object of class CVA
</p>
</td></tr>
<tr><td><code id="plot.CVA_+3A_a1">A1</code></td>
<td>

<p>Dimension for the first axis of the representation
</p>
</td></tr>
<tr><td><code id="plot.CVA_+3A_a2">A2</code></td>
<td>

<p>Dimension for the second axis of the representation
</p>
</td></tr>
<tr><td><code id="plot.CVA_+3A_...">...</code></td>
<td>

<p>Additional arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plot of a Canonical Variate Analysis
</p>


<h3>Value</h3>

<p>Te Vanonical variate plot
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>

<hr>
<h2 id='plot.ellipse'>
Plot a concentration ellipse.
</h2><span id='topic+plot.ellipse'></span>

<h3>Description</h3>

<p>Plot a concentration ellipse obtained from <code>ConcEllipse</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ellipse'
plot(x, add=TRUE, labeled= FALSE , 
center=FALSE, centerlabel="Center", initial=FALSE,  ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.ellipse_+3A_x">x</code></td>
<td>

<p>An object with class <code>ellipse</code> obtained from <code>ConcEllipse</code>.
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_add">add</code></td>
<td>

<p>Should the ellipse be added to the current plot?
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_labeled">labeled</code></td>
<td>

<p>Should the ellipse be labelled with the confidence level?
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_center">center</code></td>
<td>

<p>Should the center be plotted?
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_centerlabel">centerlabel</code></td>
<td>

<p>Label for the center.
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_initial">initial</code></td>
<td>

<p>Should the initial data be plotted?
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_...">...</code></td>
<td>

<p>Any other graphical parameter that can affects the plot (as color, etc ...)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots an ellipse containing a specified percentage of the data.
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Meulman, J. J., &amp; Heiser, W. J. (1983). The display of bootstrap solutions in multidimensional scaling. Murray Hill, NJ: Bell Laboratories.
</p>
<p>Linting, M., Meulman, J. J., Groenen, P. J., &amp; Van der Kooij, A. J. (2007). Stability of nonlinear principal components analysis: An empirical study using the balanced bootstrap. Psychological Methods, 12(3), 359.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ConcEllipse">ConcEllipse</a></code>, ~~~
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
dat=as.matrix(iris[1:50,1:2])
plot(iris[,1], iris[,2],col=iris[,5], asp=1)
E=ConcEllipse(dat, 0.95)
plot(E, labeled=TRUE, center=TRUE)
</code></pre>

<hr>
<h2 id='plot.External.Binary.Logistic.Biplot'>
Plots an External Logistic Biplot for binary data
</h2><span id='topic+plot.External.Binary.Logistic.Biplot'></span>

<h3>Description</h3>

<p>Plot of an External Binary Logistic Biplot with many arguments controling different aspects of the representation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'External.Binary.Logistic.Biplot'
plot(x, F1 = 1, F2 = 2, 
                    ShowAxis = FALSE, margin = 0.1,
                    PlotVars = TRUE, PlotInd = TRUE, WhatRows = NULL,
                    WhatCols = NULL, LabelRows = TRUE, LabelCols = TRUE,
                    RowLabels = NULL, ColLabels = NULL, RowColors = NULL,
                    ColColors = NULL, Mode = "s", TickLength = 0.01,
                    RowCex = 0.8, ColCex = 0.8, SmartLabels = FALSE,
                    MinQualityRows = 0, MinQualityCols = 0, dp = 0,
                    PredPoints = 0, SizeQualRows = FALSE, ShowBox = FALSE,
                    SizeQualCols = FALSE, ColorQualRows = FALSE,
                    ColorQualCols = FALSE, PchRows = NULL, PchCols = NULL,
                    PlotClus = FALSE, TypeClus = "ch", ClustConf = 1,
                    Significant = FALSE, alpha = 0.05, Bonferroni = FALSE,
                    PlotSupVars = TRUE, ...)
                    
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_x">x</code></td>
<td>

<p>An object of type <code>External.Binary.Logistic.Biplot</code> 
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_f1">F1</code></td>
<td>

<p>Latent factor to represent at the X axis 
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_f2">F2</code></td>
<td>

<p>Latent factor to represent at the Y axis 
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Should the axis be plotted? 
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_margin">margin</code></td>
<td>

<p>Margin for the labels in some of the biplot modes (percentage of the plot width). Default is 0. Increase the value if the labels are not completely plotted.
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_plotvars">PlotVars</code></td>
<td>

<p>Should Variables be plotted
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_plotind">PlotInd</code></td>
<td>

<p>Should Individuals be plotted 
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_whatrows">WhatRows</code></td>
<td>

<p>A binary vector (0 and 1) that indicates if each individual row should be plotted or not 
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_whatcols">WhatCols</code></td>
<td>

<p>A binary vector (0 and 1) that indicates if each individual column should be plotted or not 
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_labelrows">LabelRows</code></td>
<td>

<p>Should Variables be labelled
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_labelcols">LabelCols</code></td>
<td>

<p>Should Individuals be labelled 
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_rowlabels">RowLabels</code></td>
<td>

<p>A vector of Labels for the rows if you do not want to use the data labels
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_collabels">ColLabels</code></td>
<td>

<p>A vector of Labels for the columns if you do not want to use the data labels
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_rowcolors">RowColors</code></td>
<td>

<p>A vector of colors for the rows 
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_colcolors">ColColors</code></td>
<td>

<p>A vector of colors for the rows 
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_mode">Mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;ah&quot; and &quot;s&quot;. See details.
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_ticklength">TickLength</code></td>
<td>

<p>Lenght of the tick marks. Depends on the scale of the graph.
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_rowcex">RowCex</code></td>
<td>

<p>A scalar or a vector containing the sizes of the poitns ans labels for the rows. Default value is 0.8 if the sizes are not provided.
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_colcex">ColCex</code></td>
<td>

<p>A scalar or a vector containing the sizes of the poitns ans labels for the columns. Default value is 0.8 if the sizes are not provided.
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_smartlabels">SmartLabels</code></td>
<td>

<p>Plot the labels in a smart way
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_minqualityrows">MinQualityRows</code></td>
<td>

<p>Minimum quality of representation for a row or individual to be plotted
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_minqualitycols">MinQualityCols</code></td>
<td>

<p>Minimum quality of representation for a column or variable to be plotted
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_dp">dp</code></td>
<td>

<p>&quot;Drop Points&quot; on the variables, a vector with integers. The row points are projected on the directions of the variables listed in the vector.
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_predpoints">PredPoints</code></td>
<td>

<p>A vector with integers. The row points listed in the vector are projected onto all the variables.
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_sizequalrows">SizeQualRows</code></td>
<td>

<p>Should the size of the row points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_showbox">ShowBox</code></td>
<td>

<p>Should abox around the point be displayed?
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_sizequalcols">SizeQualCols</code></td>
<td>

<p>Should the size of the column points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_colorqualrows">ColorQualRows</code></td>
<td>

<p>Should the color of the row points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_colorqualcols">ColorQualCols</code></td>
<td>

<p>Should the color of the column points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_pchrows">PchRows</code></td>
<td>

<p>Symbol for the row points. See <code>help(points)</code> for details.
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_pchcols">PchCols</code></td>
<td>

<p>Symbol for the column points. See <code>help(points)</code> for details.
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_plotclus">PlotClus</code></td>
<td>

<p>Should the clusters be plotted?
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of plot for the clusters. (&quot;ch&quot;- Convex Hull, &quot;el&quot;- Ellipse or &quot;st&quot;- Star)
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_clustconf">ClustConf</code></td>
<td>

<p>Percent of points included in the cluster. only the ClusConf percent of the points nearest to the center will be used to calculate the cluster
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_significant">Significant</code></td>
<td>

<p>If TRUE, only the significant variables are plotted 
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_alpha">alpha</code></td>
<td>

<p>Significance Level
</p>
</td></tr>
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_bonferroni">Bonferroni</code></td>
<td>

<p>Should the Bonferroni correction be used
</p>
</td></tr>   <tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_plotsupvars">PlotSupVars</code></td>
<td>

<p>Should supplementary variables be plotted
</p>
</td></tr>    
<tr><td><code id="plot.External.Binary.Logistic.Biplot_+3A_...">...</code></td>
<td>

<p>Any other graphical parameter you want to use
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The logistic regression equation predicts the probability that a caracter will be present in an individual. Geometrically the y´s can be represented as point in the reduced dimension space and the b's are the vectors showing the directions that best predict the probability of presence of each allele . For a com-plete explanation of the geometrical properties of the ELB see Vicente-Villardón et al (2006).
The prediction of the probabilities is made in the same way as in a linear Biplot, i. e., the projection of a genotype point on the direction of an variable vector predicts the probability of presence of that variable in the individual. To facilitate the interpretation of the graph, fixed prediction probabilities points are situated on each allele vector. To simplify the graph, in our ap-plication, a vector joining the points for 0.5 and 0.75 are placed; this shows the cut point for prediction of presence and the direction of increasing probabilities. The length of the vector can be interpreted as an inverse measure of the discriminatory power of the alleles or bands, in the sense that shorter vectors correspond to alleles that better differentiate individuals. Two alleles pointing in the same direction are highly correlated, two alleles pointing in opposite directions are negatively correlated, and two alleles forming an angle close to 90º are not correlated. A more complete scale with probabilities from 0.1 to 0.9 can also be plotted with this function.
For each variable, the ordination diagram can be divided into two separate regions predicting presence or absence, the two regions are separated by the line that is perpendicular to the variable vector in the Biplot and cuts the vector in the point predicting 0.5.  The variables associated to the configuration are those that predict the presences adequately.
In a practical situation not all the variables are associated to the ordination. Due to the high number usually studied, it is convenient to situate on the graph only those that are related to the configuration, i. e., those that have an adequate goodness of fit after adjusting the logistic regression.
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Demey, J., Vicente-Villardon, J. L., Galindo, M.P.  AND Zambrano, A. (2008) Identifying Molecular Markers Associated With Classification Of Genotypes Using External Logistic Biplots. Bioinformatics, 24(24): 2832-2838.
</p>
<p>Vicente-Villardon, J. L., Galindo, M. P. and Blazquez, A. (2006) Logistic Biplots. In Multiple Correspondence Analysis And Related Methods. Grenacre, M &amp; Blasius, J, Eds,  Chapman and Hall, Boca Raton.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ExternalBinaryLogisticBiplot">ExternalBinaryLogisticBiplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
dist=BinaryProximities(spiders)
pco=PrincipalCoordinates(dist)
pcobip=ExternalBinaryLogisticBiplot(pco)
plot(pcobip, Mode="s")
pcobip=AddCluster2Biplot(pcobip, NGroups=3, ClusterType="hi")
op &lt;- par(mfrow=c(1,2)) 
plot(pcobip, Mode="s", PlotClus = TRUE)
plot(pcobip$Dendrogram)
par(op)

</code></pre>

<hr>
<h2 id='plot.fraction'>
Plots a fraction of the data as a cluster
</h2><span id='topic+plot.fraction'></span>

<h3>Description</h3>

<p>Plots a convex hull or a star  containing a specified percentage of the data. Used to plot clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'fraction'
plot(x, add = TRUE, center = FALSE, 
centerlabel = "Center", initial = FALSE, type = "ch", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.fraction_+3A_x">x</code></td>
<td>

<p>An object with class <code>fraction</code> obtained from <code>Fraction</code>.
</p>
</td></tr>
<tr><td><code id="plot.fraction_+3A_add">add</code></td>
<td>

<p>Should the fraction be added to the current plot?
</p>
</td></tr>
<tr><td><code id="plot.fraction_+3A_center">center</code></td>
<td>

<p>Should the center be plotted?
</p>
</td></tr>
<tr><td><code id="plot.fraction_+3A_centerlabel">centerlabel</code></td>
<td>

<p>Label for the center.
</p>
</td></tr>
<tr><td><code id="plot.fraction_+3A_initial">initial</code></td>
<td>

<p>Should the initial data be plotted?
</p>
</td></tr>
<tr><td><code id="plot.fraction_+3A_type">type</code></td>
<td>

<p>Type of plot. Can be: &quot;ch&quot;- Convex Hull or &quot;st&quot; - Star (Joining each point with the center)
</p>
</td></tr>
<tr><td><code id="plot.fraction_+3A_...">...</code></td>
<td>

<p>Any other graphical parameter that can affects the plot (as color, etc ...)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots a convex hull or a star  containing a specified percentage of the data.
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Fraction">Fraction</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>a=matrix(runif(50), 25,2)
a2=Fraction(a, 0.7)
plot(a2, add=FALSE, type="ch", initial=TRUE, center=TRUE, col="blue")
plot(a2, add=TRUE, type="st", col="red")
</code></pre>

<hr>
<h2 id='plot.MGC'>
Plot the results of Model-Based Gaussian Clustering algorithms
</h2><span id='topic+plot.MGC'></span>

<h3>Description</h3>

<p>PLots an object of type MGC (Model-based Gaussian Clustering)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MGC'
plot(x, vars = NULL, groups = x$Classification, CexPoints = 0.2, Confidence = 0.95, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.MGC_+3A_x">x</code></td>
<td>

<p>An object of type MGC
</p>
</td></tr>
<tr><td><code id="plot.MGC_+3A_vars">vars</code></td>
<td>

<p>A subset of indices of the variables to be plotted
</p>
</td></tr>
<tr><td><code id="plot.MGC_+3A_groups">groups</code></td>
<td>

<p>A factor containing groups to represent. Usually the clusters obtained from the algorithm.
</p>
</td></tr>
<tr><td><code id="plot.MGC_+3A_cexpoints">CexPoints</code></td>
<td>

<p>Size of the points.
</p>
</td></tr>
<tr><td><code id="plot.MGC_+3A_confidence">Confidence</code></td>
<td>

<p>Confidence of the ellipses
</p>
</td></tr>
<tr><td><code id="plot.MGC_+3A_...">...</code></td>
<td>

<p>Anay additional graphical parameters
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PLots an object of type MGC (Model-based Gaussian Clustering) using a splom plot.
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
</code></pre>

<hr>
<h2 id='plot.Ordinal.Logistic.Biplot'>
Plots an ordinal Logistic Biplot
</h2><span id='topic+plot.Ordinal.Logistic.Biplot'></span>

<h3>Description</h3>

<p>Plots an ordinal Logistic Biplot</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Ordinal.Logistic.Biplot'
plot(x, A1 = 1, A2 = 2, 
ShowAxis = FALSE, margin = 0, PlotVars = TRUE, PlotInd = TRUE, 
LabelVars = TRUE, LabelInd = TRUE, mode = "a", CexInd = NULL, 
CexVar = NULL, ColorInd = NULL, ColorVar = NULL, SmartLabels = TRUE,
MinQualityVars = 0, dp = 0, PredPoints = 0, PlotAxis = FALSE, 
TypeScale = "Complete", ValuesScale = "Original", 
SizeQualInd = FALSE, SizeQualVars = FALSE, ColorQualInd = FALSE, 
ColorQualVars = FALSE, PchInd = NULL, PchVar = NULL, 
PlotClus = FALSE, TypeClus = "ch", ClustConf = 1, 
ClustCenters = FALSE, UseClusterColors = TRUE, ClustLegend = TRUE,
ClustLegendPos = "topright", TextVarPos = 1, PlotSupVars = FALSE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_x">x</code></td>
<td>

<p>Plots and object of type &quot;Ordinal.Logistic.Biplot&quot;
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_a1">A1</code></td>
<td>

<p>First dimension to plot
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_a2">A2</code></td>
<td>

<p>Second dimension to plot
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Should the axis be shown
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_margin">margin</code></td>
<td>

<p>Margin for the graph (in order to have space for the variable levels)
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_plotvars">PlotVars</code></td>
<td>

<p>Should the variables be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_plotind">PlotInd</code></td>
<td>

<p>Should the individuals be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_labelvars">LabelVars</code></td>
<td>

<p>Should the variables be labelled?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_labelind">LabelInd</code></td>
<td>

<p>Should the variables be labelled?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_mode">mode</code></td>
<td>

<p>Mode of the biplot (see the classical biplot)
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_cexind">CexInd</code></td>
<td>

<p>Type of marker used for the individuals
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_cexvar">CexVar</code></td>
<td>

<p>Type of marker used for the variables
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_colorind">ColorInd</code></td>
<td>

<p>Colors used for the individuals
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_colorvar">ColorVar</code></td>
<td>

<p>Colors used for the cariables
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_smartlabels">SmartLabels</code></td>
<td>

<p>Should smart placement for the labels be used?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_minqualityvars">MinQualityVars</code></td>
<td>

<p>Minimum quality of representation for a variable to be displayed
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_dp">dp</code></td>
<td>

<p>Set of variables in which the individuals are projected
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_predpoints">PredPoints</code></td>
<td>

<p>Set of points thet will be projected on all the variables
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_plotaxis">PlotAxis</code></td>
<td>

<p>Should the axis be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_typescale">TypeScale</code></td>
<td>

<p>See continuous biplots
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>See continuous biplots
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_sizequalind">SizeQualInd</code></td>
<td>

<p>Should the size of the labels and points be related to the quality of representation for individuals?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_sizequalvars">SizeQualVars</code></td>
<td>

<p>Should the size of the labels and points be related to the quality of representation for variables?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_colorqualind">ColorQualInd</code></td>
<td>

<p>Should the intensity of the color of the labels and points be related to the quality of representation for individuals?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_colorqualvars">ColorQualVars</code></td>
<td>

<p>Should the intensity of the color of the labels and points be related to the quality of representation for variables?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_pchind">PchInd</code></td>
<td>

<p>Markers for the individuals
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_pchvar">PchVar</code></td>
<td>

<p>Markers for the individuals
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_plotclus">PlotClus</code></td>
<td>

<p>Should the added clusters for the individuals be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of plot for the clusters. The types are &quot;ch&quot;, &quot;el&quot; and &quot;st&quot; for &quot;Convex Hull&quot;, &quot;Ellipse&quot; and &quot;Star&quot; repectively.
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_clustconf">ClustConf</code></td>
<td>

<p>Confidence level for the cluster
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_clustcenters">ClustCenters</code></td>
<td>

<p>Should the centers of the clsters be plotted
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_useclustercolors">UseClusterColors</code></td>
<td>

<p>Should the colors of the clusters be used to plot the individuals.
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_clustlegend">ClustLegend</code></td>
<td>

<p>Should a legend for the clusters be added?
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_clustlegendpos">ClustLegendPos</code></td>
<td>

<p>Position of the legend
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_textvarpos">TextVarPos</code></td>
<td>

<p>Position of the labels for the variables
</p>
</td></tr>  
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_plotsupvars">PlotSupVars</code></td>
<td>

<p>Should the supplementary variables be plotted
</p>
</td></tr>
<tr><td><code id="plot.Ordinal.Logistic.Biplot_+3A_...">...</code></td>
<td>

<p>Any other aditional parameters
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots an ordinal Logistic Biplot
</p>


<h3>Value</h3>

<p>The plot ....
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Vicente-Villardón, J. L., &amp; Sánchez, J. C. H. (2014). Logistic Biplots for Ordinal Data with an Application to Job Satisfaction of Doctorate Degree Holders in Spain. arXiv preprint arXiv:1405.0294.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.ContinuousBiplot">plot.ContinuousBiplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    data(Doctors)
    olb = OrdLogBipEM(Doctors,dim = 2, nnodes = 10, initial=4,  tol = 0.001, 
    maxiter = 100, penalization = 0.1, show=TRUE)
    plot(olb, mode="s", ColorInd="gray", ColorVar=1:5)
</code></pre>

<hr>
<h2 id='plot.PCA.Analysis'>
Plots a Principal Component Analysis
</h2><span id='topic+plot.PCA.Analysis'></span>

<h3>Description</h3>

<p>Plots the results of a Principal Component Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PCA.Analysis'
plot(x, A1 = 1, A2 = 2, CorrelationCircle = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.PCA.Analysis_+3A_x">x</code></td>
<td>

<p>The object with the results of a PCA
</p>
</td></tr>
<tr><td><code id="plot.PCA.Analysis_+3A_a1">A1</code></td>
<td>

<p>Dimension for the first axis of the representation
</p>
</td></tr>
<tr><td><code id="plot.PCA.Analysis_+3A_a2">A2</code></td>
<td>

<p>Dimension for the second axis of the representation
</p>
</td></tr>
<tr><td><code id="plot.PCA.Analysis_+3A_correlationcircle">CorrelationCircle</code></td>
<td>

<p>Should the correlation circle be plotted? If false the scores plot is done.
</p>
</td></tr>
<tr><td><code id="plot.PCA.Analysis_+3A_...">...</code></td>
<td>

<p>Any other arguments of the function plot.ContinuousBiplot
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots theresults of a Principal Component Analysis. The plot can be the correlation circle containing the correlations of the variables with the components or a plot of the scores of the individuals.
</p>


<h3>Value</h3>

<p>The PCA plot.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.ContinuousBiplot">plot.ContinuousBiplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='plot.PCA.Bootstrap'>
Plots the Bootstrap information for Principal Components Analysis (PCA)
</h2><span id='topic+plot.PCA.Bootstrap'></span>

<h3>Description</h3>

<p>Plots an object of class &quot;PCA.Bootstrap&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PCA.Bootstrap'
plot(x, Eigenvalues = TRUE, 
Inertia = FALSE, EigenVectors = TRUE, Structure = TRUE, 
Squared = TRUE, Scores = TRUE, ColorInd = "black", TypeScores = "ch", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.PCA.Bootstrap_+3A_x">x</code></td>
<td>

<p>An object of class &quot;PCA.Bootstrap&quot;
</p>
</td></tr>
<tr><td><code id="plot.PCA.Bootstrap_+3A_eigenvalues">Eigenvalues</code></td>
<td>

<p>Should the information for the eigenvalues be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PCA.Bootstrap_+3A_inertia">Inertia</code></td>
<td>

<p>Should the information for the inertia be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PCA.Bootstrap_+3A_eigenvectors">EigenVectors</code></td>
<td>

<p>Should the information for the eigenvectors be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PCA.Bootstrap_+3A_structure">Structure</code></td>
<td>

<p>Should the information for the correlations (variables-dimensions) be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PCA.Bootstrap_+3A_squared">Squared</code></td>
<td>

<p>Should the information for the correlations (variables-dimensions) be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PCA.Bootstrap_+3A_scores">Scores</code></td>
<td>

<p>Should the row (individual) scores be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PCA.Bootstrap_+3A_colorind">ColorInd</code></td>
<td>

<p>Colors for the rows
</p>
</td></tr>
<tr><td><code id="plot.PCA.Bootstrap_+3A_typescores">TypeScores</code></td>
<td>

<p>Type of plot for the scores
</p>
</td></tr>
<tr><td><code id="plot.PCA.Bootstrap_+3A_...">...</code></td>
<td>

<p>Any other graphical argument
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each parameter, box-plots and confidence intervals are plotted. 
The initial estimator and the bootstrap mean are plotted.
</p>
<p>For the eigenvectors, loadings and contributions, the graph is divided into as many rows as dimensions, each row contains a plot of the hole set of variables.
</p>
<p>The scores are plotted on a two dimensional
</p>


<h3>Value</h3>

<p>No value returned 
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Daudin, J. J., Duby, C., &amp; Trecourt, P. (1988). Stability of principal component analysis studied by the bootstrap method. Statistics: A journal of theoretical and applied statistics, 19(2), 241-258.
</p>
<p>Chateau, F., &amp; Lebart, L. (1996). Assessing sample variability in the visualization techniques related to principal component analysis: bootstrap and alternative simulation methods. COMPSTAT, Physica-Verlag, 205-210.
</p>
<p>Babamoradi, H., van den Berg, F., &amp; Rinnan, Å. (2013). Bootstrap based confidence limits in principal component analysis: A case study. Chemometrics and Intelligent Laboratory Systems, 120, 97-105.
</p>
<p>Fisher, A., Caffo, B., Schwartz, B., &amp; Zipunnikov, V. (2016). Fast, exact bootstrap principal component analysis for p&gt; 1 million. Journal of the American Statistical Association, 111(514), 846-860.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PCA.Bootstrap">PCA.Bootstrap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X=wine[,4:21]
grupo=wine$Group
rownames(X)=paste(1:45, grupo, sep="-")
pcaboot=PCA.Bootstrap(X, dimens=2, Scaling = "Standardize columns", B=1000)
plot(pcaboot, ColorInd=as.numeric(grupo))
summary(pcaboot)
</code></pre>

<hr>
<h2 id='plot.PCoABootstrap'>
Plots an object of class PCoABootstrap
</h2><span id='topic+plot.PCoABootstrap'></span>

<h3>Description</h3>

<p>Plots an object of class PCoABootstrap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PCoABootstrap'
plot(x, F1=1, F2=2, Move2Center=TRUE, 
BootstrapPlot="Ellipse", confidence=0.95, Colors=NULL, ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.PCoABootstrap_+3A_x">x</code></td>
<td>

<p>An object of class &quot;PCoABootstrap&quot;
</p>
</td></tr>
<tr><td><code id="plot.PCoABootstrap_+3A_f1">F1</code></td>
<td>

<p>First dimension to plot
</p>
</td></tr>
<tr><td><code id="plot.PCoABootstrap_+3A_f2">F2</code></td>
<td>

<p>Second dimension to plot
</p>
</td></tr>
<tr><td><code id="plot.PCoABootstrap_+3A_move2center">Move2Center</code></td>
<td>

<p>Translate the ellipse center to the coordinates
</p>
</td></tr>
<tr><td><code id="plot.PCoABootstrap_+3A_bootstrapplot">BootstrapPlot</code></td>
<td>

<p>Type of Bootstrap plot to draw: &quot;Ellipse&quot;, &quot;ConvexHull&quot;, &quot;Star&quot;
</p>
</td></tr>
<tr><td><code id="plot.PCoABootstrap_+3A_confidence">confidence</code></td>
<td>

<p>Confidence level for the bootstrap plot
</p>
</td></tr>
<tr><td><code id="plot.PCoABootstrap_+3A_colors">Colors</code></td>
<td>

<p>Colors of the objects
</p>
</td></tr>
<tr><td><code id="plot.PCoABootstrap_+3A_...">...</code></td>
<td>

<p>Additional parameters for graphical representations
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Draws the bootstrap confidence regions for the coordinates of the points obtained from a Principal Coodinates Analysis
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>J.R. Demey, J.L. Vicente-Villardon, M.P. Galindo, A.Y. Zambrano, Identifying molecular markers associated with classifications of genotypes by external logistic
biplot, Bioinformatics 24 (2008) 2832.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
Dis=BinaryProximities(spiders)
pco=PrincipalCoordinates(Dis, Bootstrap=TRUE, BootstrapType="Products")
plot(pco, Bootstrap=TRUE)

</code></pre>

<hr>
<h2 id='plot.Principal.Coordinates'>Plots an object of class Principal.Coordinates</h2><span id='topic+plot.Principal.Coordinates'></span>

<h3>Description</h3>

<p>Plots an object of class Principal.Coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Principal.Coordinates'
plot(x, A1 = 1, A2 = 2, LabelRows = TRUE, 
WhatRows = NULL, RowCex = 1, RowPch = 16, Title = "", RowLabels = NULL, 
RowColors = NULL, ColColors = NULL, ColLabels = NULL, SizeQualInd = FALSE, 
SmartLabels = TRUE, ColorQualInd = FALSE, ColorQual = "black", PlotSup = TRUE, 
Bootstrap = FALSE, BootstrapPlot = c("Ellipse", "CovexHull", "Star"), 
margin = 0, PlotClus = FALSE, TypeClus = "ch", ClustConf = 1, 
CexClustCenters = 1, LegendClust = TRUE, ClustCenters = FALSE, 
UseClusterColors = TRUE, ShowAxis = FALSE, PlotBinaryMeans = FALSE, 
MinIncidence = 0, ShowBox = FALSE, ColorSupContVars = NULL, 
ColorSupBinVars = NULL, ColorSupOrdVars = NULL, TypeScale = "Complete", 
SupMode = "s", PlotSupVars = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Principal.Coordinates_+3A_x">x</code></td>
<td>

<p>Object of class &quot;Principal.Coordinates&quot;
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_a1">A1</code></td>
<td>

<p>First dimenssion of the plot
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_a2">A2</code></td>
<td>

<p>Second dimenssion of the plot
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_labelrows">LabelRows</code></td>
<td>

<p>Controls if the points are labelled. Usually TRUE.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_whatrows">WhatRows</code></td>
<td>

<p>What Rows to plot. A vector of 0/1 elements. If NULL all rows are plotted
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_rowcex">RowCex</code></td>
<td>

<p>Size of the points. Can be a single number or a vector.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_rowpch">RowPch</code></td>
<td>

<p>Symbols for the points.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_title">Title</code></td>
<td>

<p>Title for the graph
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_rowlabels">RowLabels</code></td>
<td>

<p>Labels for the rows. If NULL row names of the data matrix are used.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_rowcolors">RowColors</code></td>
<td>

<p>Colors for the rows. If NULL row deafault colors are assigned. Can be a single value or avector of colors.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_colcolors">ColColors</code></td>
<td>

<p>Colors for the columns (Variables)
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_collabels">ColLabels</code></td>
<td>

<p>Labels for the columns (Variables)
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_sizequalind">SizeQualInd</code></td>
<td>

<p>Controls if the size of points depends on the quality of representation.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_smartlabels">SmartLabels</code></td>
<td>

<p>Controls the way labels are plotted on the graph. If TRUE labels for points with positive x values
are placed to the right of the point and labels for points with negative values to the left
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_colorqualind">ColorQualInd</code></td>
<td>

<p>Controls if the color of the points depends on the quality of representation.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_colorqual">ColorQual</code></td>
<td>

<p>Darker color for the quality scale.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_plotsup">PlotSup</code></td>
<td>

<p>Controls if the supplementary points are plotted.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_bootstrap">Bootstrap</code></td>
<td>

<p>Controls if the bootstrap points are plotted.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_bootstrapplot">BootstrapPlot</code></td>
<td>

<p>Type of plot of the Bootstrap Information. The types are &quot;Ellipse&quot;, &quot;CovexHull&quot; or &quot;Star&quot;.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_margin">margin</code></td>
<td>

<p>Margin for the graph.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_plotclus">PlotClus</code></td>
<td>

<p>Should the clusters be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of plot for the clusters. (&quot;ch&quot;- Convex Hull, &quot;el&quot;- Ellipse or &quot;st&quot;- Star)
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_clustconf">ClustConf</code></td>
<td>

<p>Percent of points included in the cluster. only the ClusConf percent of the points nearest to the center will be used to calculate the cluster
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_cexclustcenters">CexClustCenters</code></td>
<td>

<p>Size of the cluster centers
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_legendclust">LegendClust</code></td>
<td>

<p>Legends for the clusters
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_clustcenters">ClustCenters</code></td>
<td>

<p>Should the cluster centers be plotted
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_useclustercolors">UseClusterColors</code></td>
<td>

<p>Should the cluster colors be used in the plot
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Logical variable to control if the coordinate axes should appear in the plot. The default value is FALSE because for most of the biplots its presence is irrelevant.
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_plotbinarymeans">PlotBinaryMeans</code></td>
<td>

<p>Plot the mean of the presence points for each variable
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_minincidence">MinIncidence</code></td>
<td>

<p>Minimum incidence to keep a variable
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_showbox">ShowBox</code></td>
<td>

<p>Should a box around the poitns be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_colorsupcontvars">ColorSupContVars</code></td>
<td>

<p>Colors for the supplementary continuous variables
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_colorsupbinvars">ColorSupBinVars</code></td>
<td>

<p>Colors for the supplementary binary variables
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_colorsupordvars">ColorSupOrdVars</code></td>
<td>

<p>Colors for the supplementary ordinal variables
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scales for the plot
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_supmode">SupMode</code></td>
<td>

<p>Mode of the supplementary variables
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_plotsupvars">PlotSupVars</code></td>
<td>

<p>Should the supplementary variables be plotted
</p>
</td></tr>
<tr><td><code id="plot.Principal.Coordinates_+3A_...">...</code></td>
<td>

<p>Additional parameters for graphical representations
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Graphical representation of an Principal coordinates Analysis controlling visual aspects of the plot as colors, symbols or sizes of the points. 
</p>


<h3>Value</h3>

<p>No value is returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>J.R. Demey, J.L. Vicente-Villardon, M.P. Galindo, A.Y. Zambrano, Identifying molecular markers associated with classifications of genotypes by external logistic
biplot, Bioinformatics 24 (2008) 2832.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BinaryProximities">BinaryProximities</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
dist=BinaryProximities(spiders)
pco=PrincipalCoordinates(dist)
plot(pco)
</code></pre>

<hr>
<h2 id='plot.Procrustes'>Plots an object of class &quot;Procrustes&quot;</h2><span id='topic+plot.Procrustes'></span>

<h3>Description</h3>

<p>Plots Simple Procrustes Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Procrustes'
plot(x, F1=1, F2=2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Procrustes_+3A_x">x</code></td>
<td>

<p>Object of class &quot;Procrustes&quot;
</p>
</td></tr>
<tr><td><code id="plot.Procrustes_+3A_f1">F1</code></td>
<td>

<p>First dimenssion of the plot
</p>
</td></tr>
<tr><td><code id="plot.Procrustes_+3A_f2">F2</code></td>
<td>

<p>Second dimenssion of the plot
</p>
</td></tr>
<tr><td><code id="plot.Procrustes_+3A_...">...</code></td>
<td>

<p>Additional parameters for graphical representations
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Graphical representation of an Orthogonal Procrustes Analysis. 
</p>


<h3>Value</h3>

<p>No value is returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BinaryProximities">BinaryProximities</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
dist=BinaryProximities(spiders)
pco=PrincipalCoordinates(dist)
plot(pco)
</code></pre>

<hr>
<h2 id='plot.StatisBiplot'>
Plots a Statis Biplot Object
</h2><span id='topic+plot.StatisBiplot'></span>

<h3>Description</h3>

<p>Plots a Statis Biplot Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'StatisBiplot'
plot(x, A1 = 1, A2 = 2, PlotType = "Biplot", 
PlotRowTraj = FALSE, PlotVarTraj = FALSE, LabelTraj = "Begining", 
VarColorType = "ByVar", VarColors = NULL, VarLabels = NULL, 
RowColors = NULL, TableColors = NULL, RowRandomColors = FALSE, 
TypeTraj = "line", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.StatisBiplot_+3A_x">x</code></td>
<td>

<p>A Statis object
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_a1">A1</code></td>
<td>

<p>First dimension of the plot
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_a2">A2</code></td>
<td>

<p>Second dimension of the plot
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_plottype">PlotType</code></td>
<td>

<p>Type of plot: Interstructure, Correlations, Contributions or Biplot
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_plotrowtraj">PlotRowTraj</code></td>
<td>

<p>Should the row trajectories be plotted?
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_plotvartraj">PlotVarTraj</code></td>
<td>

<p>Should the variables trajectories be plotted?
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_labeltraj">LabelTraj</code></td>
<td>

<p>Where the trajecories should be labelled: Begining or End.
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_varcolortype">VarColorType</code></td>
<td>

<p>The colors for the variables should be set by table (ByTable) or by variable (ByVar)
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_varcolors">VarColors</code></td>
<td>

<p>Colors for the variables.
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_varlabels">VarLabels</code></td>
<td>

<p>Labels for the variables
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_rowcolors">RowColors</code></td>
<td>

<p>Colors for the rows
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_tablecolors">TableColors</code></td>
<td>

<p>Colors for each table
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_rowrandomcolors">RowRandomColors</code></td>
<td>

<p>Use random colors for the variables.
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_typetraj">TypeTraj</code></td>
<td>

<p>Type of trajectory to plot: Lines or stars
</p>
</td></tr>
<tr><td><code id="plot.StatisBiplot_+3A_...">...</code></td>
<td>

<p>Aditional parameters</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots a Statis Biplot Object. The arguments of the general biplot are as in a Continuous Biplot.
</p>


<h3>Value</h3>

<p>A biplot
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Vallejo-Arboleda, A., Vicente-Villardon, J. L., &amp; Galindo-Villardon, M. P. (2007). Canonical STATIS: Biplot analysis of multi-table group structured data based on STATIS-ACT methodology. Computational statistics &amp; data analysis, 51(9), 4193-4205.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.ContinuousBiplot">plot.ContinuousBiplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Chemical)
x= Chemical[,5:16]
X=Convert2ThreeWay(x,Chemical$WEEKS, columns=FALSE)
stbip=StatisBiplot(X)
</code></pre>

<hr>
<h2 id='plot.TetraDualStatis'>
Plots an object of class &quot;tetraDualStatis&quot;.
</h2><span id='topic+plot.TetraDualStatis'></span>

<h3>Description</h3>

<p>Plots an object the results of TetraDualStatis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'TetraDualStatis'
plot(x, A1 = 1, A2 = 2, PlotType = "InterStructure", 
                    PlotRowTraj = FALSE, PlotVarTraj = FALSE, LabelTraj = "Begining", 
                    VarColorType = "Biplot", VarColors = NULL, VarLabels = NULL, 
                    RowColors = NULL, TableColors = NULL, RowRandomColors = FALSE, 
                    TypeTraj = "line", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.TetraDualStatis_+3A_x">x</code></td>
<td>

<p>An object of class TetraDualStatis
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_a1">A1</code></td>
<td>

<p>Dimension for the X-axis
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_a2">A2</code></td>
<td>

<p>Dimension for the Y-axis
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_plottype">PlotType</code></td>
<td>

<p>Type of plot: &quot;Biplot&quot;, &quot;Compromise&quot;, &quot;Correlations&quot;, &quot;Contributions&quot;, &quot;InterStructure&quot;.
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_plotrowtraj">PlotRowTraj</code></td>
<td>

<p>Should the row trajectories be plotted?
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_plotvartraj">PlotVarTraj</code></td>
<td>

<p>Should the variables trajectories be plotted?
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_labeltraj">LabelTraj</code></td>
<td>

<p>Should the trajectories be labelled.
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_varcolortype">VarColorType</code></td>
<td>

<p>One of the following: &quot;Biplot&quot;, &quot;ByTable&quot;, &quot;ByVar&quot;.
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_varcolors">VarColors</code></td>
<td>

<p>User colors for the variables.
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_varlabels">VarLabels</code></td>
<td>

<p>User labels for the variables.
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_rowcolors">RowColors</code></td>
<td>

<p>User colors for the rows.
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_tablecolors">TableColors</code></td>
<td>

<p>User colors for the different tables.
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_rowrandomcolors">RowRandomColors</code></td>
<td>

<p>Should use random colors for the rows?
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_typetraj">TypeTraj</code></td>
<td>

<p>Type of trajectory. Normally a line.
</p>
</td></tr>
<tr><td><code id="plot.TetraDualStatis_+3A_...">...</code></td>
<td>

<p>Additional graphical arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots an object the results of TetraDualStatis.
</p>


<h3>Value</h3>

<p>The plot of the results
</p>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----

</code></pre>

<hr>
<h2 id='plot.Unfolding'>
Plots an Unfolding Representation
</h2><span id='topic+plot.Unfolding'></span>

<h3>Description</h3>

<p>Plots an Unfolding Representation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Unfolding'
plot(x, A1 = 1, A2 = 2, ShowAxis = FALSE,
margin = 0.1, PlotSites = TRUE, PlotSpecies = TRUE, PlotEnv = TRUE,
LabelSites = TRUE, LabelSpecies = TRUE, LabelEnv = TRUE, 
SpeciesQuality = FALSE, MinQualityVars = 0, dp = 0, 
PlotAxis = FALSE, TypeScale = "Complete", ValuesScale = "Original", 
mode = "h", CexSites = NULL, CexSpecies = NULL, CexVar = NULL, 
ColorSites = NULL, ColorSpecies = NULL, ColorVar = NULL, 
PchSites = NULL, PchSpecies = NULL, PchVar = NULL, 
SizeQualSites = FALSE, SizeQualSpecies = FALSE, 
SizeQualVars = FALSE, ColorQualSites = FALSE, 
ColorQualSpecies = FALSE, ColorQualVars = FALSE, SmartLabels = FALSE, 
PlotTol = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Unfolding_+3A_x">x</code></td>
<td>

<p>An object of class Unfolding
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_a1">A1</code></td>
<td>

<p>Axis 1 of the representation.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_a2">A2</code></td>
<td>

<p>Axis 1 of the representation.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Should the axis be shown?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_margin">margin</code></td>
<td>

<p>Margin for the plot (precentage)
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_plotsites">PlotSites</code></td>
<td>

<p>Should the sites be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_plotspecies">PlotSpecies</code></td>
<td>

<p>Should the species be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_plotenv">PlotEnv</code></td>
<td>

<p>Should the environmental variables be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_labelsites">LabelSites</code></td>
<td>

<p>Should the sites be labelled?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_labelspecies">LabelSpecies</code></td>
<td>

<p>Should the species be labelled?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_labelenv">LabelEnv</code></td>
<td>

<p>Should the environmental variables be labelled?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_speciesquality">SpeciesQuality</code></td>
<td>

<p>Min species quality to plot
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_minqualityvars">MinQualityVars</code></td>
<td>

<p>Minimum quality of a var to be plotted.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_dp">dp</code></td>
<td>

<p>A set of indices with the variables that will show the projections of the individuals.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_plotaxis">PlotAxis</code></td>
<td>

<p>Should the axis be plotted?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scale to use : &quot;Complete&quot;, &quot;StdDev&quot; or &quot;BoxPlot&quot;
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>Values to show on the scale: &quot;Original&quot; or &quot;Transformed&quot;
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_mode">mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;h&quot;, &quot;ah&quot; and &quot;s&quot;.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_cexsites">CexSites</code></td>
<td>

<p>Size for the symbols and labels of the sites. Can be a single common size for all the points or a vector with individual sizes.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_cexspecies">CexSpecies</code></td>
<td>

<p>Size for the symbols and labels of the species. Can be a single common size for all the points or a vector with individual sizes.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_cexvar">CexVar</code></td>
<td>

<p>Size for the symbols and labels of the variables. Can be a single common size for all the points or a vector with individual sizes.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_colorsites">ColorSites</code></td>
<td>

<p>Color for the symbols and labels of the sites. Can be a single common color for all the points or a vector with individual colors.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_colorspecies">ColorSpecies</code></td>
<td>

<p>Color for the symbols and labels of the species. Can be a single common color for all the points or a vector with individual colors.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_colorvar">ColorVar</code></td>
<td>

<p>Color for the symbols and labels of the variables. Can be a single common color for all the points or a vector with individual colors.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_pchsites">PchSites</code></td>
<td>

<p>Symbol for the sites points. See <code>help(points)</code> for details.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_pchspecies">PchSpecies</code></td>
<td>

<p>Symbol for the species points. See <code>help(points)</code> for details.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_pchvar">PchVar</code></td>
<td>

<p>Symbol for the variables points. See <code>help(points)</code> for details.
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_sizequalsites">SizeQualSites</code></td>
<td>

<p>Should the size of the site points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_sizequalspecies">SizeQualSpecies</code></td>
<td>

<p>Should the size of the species points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_sizequalvars">SizeQualVars</code></td>
<td>

<p>Should the size of the variables points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_colorqualsites">ColorQualSites</code></td>
<td>

<p>Should the color of the sites points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_colorqualspecies">ColorQualSpecies</code></td>
<td>

<p>Should the color of the species points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_colorqualvars">ColorQualVars</code></td>
<td>

<p>Should the color of the variables points be related to their qualities of representation (predictiveness)?
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_smartlabels">SmartLabels</code></td>
<td>

<p>Plot the labels in a smart way
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_plottol">PlotTol</code></td>
<td>

<p>Should the tolerances be plotted
</p>
</td></tr>
<tr><td><code id="plot.Unfolding_+3A_...">...</code></td>
<td>

<p>Aditional graphical parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots an Unfolding Representation
</p>


<h3>Value</h3>

<p>A plot of the unfolding representation.




</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>de Leeuw, J. (2005). Multidimensional unfolding. Encyclopedia of statistics in behavioral science.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='PlotBiplotClusters'>
Plot clusters on a biplot.
</h2><span id='topic+PlotBiplotClusters'></span>

<h3>Description</h3>

<p>Highlights several groups or clusters on a biplot representation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotBiplotClusters(A, Groups = ones(c(nrow(A), 1)), TypeClus = "st",
                 ClusterColors = NULL, ClusterNames = NULL, centers =
                 TRUE, ClustConf = 1, Legend = FALSE, LegendPos =
                 "topright", CexClustCenters = 1,  ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotBiplotClusters_+3A_a">A</code></td>
<td>

<p>Coordinates of the points in the scattergram
</p>
</td></tr>
<tr><td><code id="PlotBiplotClusters_+3A_groups">Groups</code></td>
<td>

<p>Factor defining the groups to be highlited
</p>
</td></tr>
<tr><td><code id="PlotBiplotClusters_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of representation of the clusters. For the moment just a convex hull but in the future ellipses and stars will be added.
</p>
</td></tr>
<tr><td><code id="PlotBiplotClusters_+3A_clustercolors">ClusterColors</code></td>
<td>

<p>A vector of colors with as many elements as clusters. If <code>NULL</code> the function slects the raibow colors.
</p>
</td></tr>
<tr><td><code id="PlotBiplotClusters_+3A_clusternames">ClusterNames</code></td>
<td>

<p>A vector of names with as many elements as clusters.
</p>
</td></tr>
<tr><td><code id="PlotBiplotClusters_+3A_centers">centers</code></td>
<td>

<p>Logical variable to control if centres of the clusters are plotted
</p>
</td></tr>
<tr><td><code id="PlotBiplotClusters_+3A_clustconf">ClustConf</code></td>
<td>

<p>Percent of points included in the cluster. only the ClusConf percent of the points nearest to the center will be used to calculate the cluster
</p>
</td></tr>
<tr><td><code id="PlotBiplotClusters_+3A_legend">Legend</code></td>
<td>

<p>Should a legend be plotted
</p>
</td></tr>
<tr><td><code id="PlotBiplotClusters_+3A_legendpos">LegendPos</code></td>
<td>

<p>Position of the legend.
</p>
</td></tr>
<tr><td><code id="PlotBiplotClusters_+3A_cexclustcenters">CexClustCenters</code></td>
<td>

<p>Size of the cluster centres.
</p>
</td></tr>
<tr><td><code id="PlotBiplotClusters_+3A_...">...</code></td>
<td>

<p>Any other graphical parameters 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The clusters to plot should be added to the biplot object using the function <code>AddCluster2Biplot</code>.
</p>


<h3>Value</h3>

<p>It takes effects on a plot
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AddCluster2Biplot">AddCluster2Biplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
bip=PCA.Biplot(iris[,1:4])
bip=AddCluster2Biplot(bip, NGroups=3, ClusterType="us", Groups=iris[,5], Original=FALSE)
plot(bip, PlotClus = TRUE)
</code></pre>

<hr>
<h2 id='PlotOrdinalResponses'>
Plot the response functions along the directions of best fit.
</h2><span id='topic+PlotOrdinalResponses'></span>

<h3>Description</h3>

<p>Plot the response functions along the directions of best fit for the selected dimensions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotOrdinalResponses(olb, A1 = 1, A2 = 2, inf = -12, sup = 12, 
Legend = TRUE, WhatVars=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotOrdinalResponses_+3A_olb">olb</code></td>
<td>

<p>An object of class &quot;Ordinal.Logistic.Biplot&quot;</p>
</td></tr>
<tr><td><code id="PlotOrdinalResponses_+3A_a1">A1</code></td>
<td>

<p>First dimension of the plot.
</p>
</td></tr>
<tr><td><code id="PlotOrdinalResponses_+3A_a2">A2</code></td>
<td>

<p>Second dimension of the plot
</p>
</td></tr>
<tr><td><code id="PlotOrdinalResponses_+3A_inf">inf</code></td>
<td>

<p>Lower limit of the representation
</p>
</td></tr>
<tr><td><code id="PlotOrdinalResponses_+3A_sup">sup</code></td>
<td>

<p>Upper limit of the representation
</p>
</td></tr>
<tr><td><code id="PlotOrdinalResponses_+3A_legend">Legend</code></td>
<td>

<p>Should a legend be plotted
</p>
</td></tr>
<tr><td><code id="PlotOrdinalResponses_+3A_whatvars">WhatVars</code></td>
<td>

<p>A vector with the numbers of the variables to be plotted. If NULL all the variables are plotted.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plot the response functions along the directions of best fit for the selected dimensions
</p>


<h3>Value</h3>

<p>A plot describing the behaviour of the variable 
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Doctors)
    olb = OrdLogBipEM(Doctors,dim = 2, nnodes = 10, initial=4,  tol = 0.001, 
    maxiter = 100, penalization = 0.1, show=TRUE)
    PlotOrdinalResponses(olb, WhatVars=c(1,2,3,4))
</code></pre>

<hr>
<h2 id='PLSR'>
Partial Least Squares Regression
</h2><span id='topic+PLSR'></span>

<h3>Description</h3>

<p>Partial Least Squares Regression for numerical variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PLSR(Y, X, S = 2, InitTransform = 5, grouping = NULL, 
centerY = TRUE, scaleY = TRUE, tolerance = 5e-06, 
maxiter = 100, show = FALSE, Validation = NULL, nB = 500)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PLSR_+3A_y">Y</code></td>
<td>

<p>Matrix of Dependent Variables
</p>
</td></tr>
<tr><td><code id="PLSR_+3A_x">X</code></td>
<td>

<p>Matrix of Independent Variables
</p>
</td></tr>
<tr><td><code id="PLSR_+3A_s">S</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="PLSR_+3A_inittransform">InitTransform</code></td>
<td>

<p>Initial transformation of the independent variables.
</p>
</td></tr>
<tr><td><code id="PLSR_+3A_grouping">grouping</code></td>
<td>

<p>Fator when the init transformation is the standardization with the within groups deviation.
</p>
</td></tr>
<tr><td><code id="PLSR_+3A_centery">centerY</code></td>
<td>

<p>Should the dependent variables be centered?
</p>
</td></tr>
<tr><td><code id="PLSR_+3A_scaley">scaleY</code></td>
<td>

<p>Should the dependent variables be standadized?
</p>
</td></tr>
<tr><td><code id="PLSR_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for the algorithm
</p>
</td></tr>
<tr><td><code id="PLSR_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations
</p>
</td></tr>
<tr><td><code id="PLSR_+3A_show">show</code></td>
<td>

<p>Show the progress of the algorithm?
</p>
</td></tr>
<tr><td><code id="PLSR_+3A_validation">Validation</code></td>
<td>

<p>Validation (None, Cross, Bootstrap)
</p>
</td></tr>
<tr><td><code id="PLSR_+3A_nb">nB</code></td>
<td>

<p>number of samples for the bottstrap validation
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Partial Least Squares Regression for numerical variables.
</p>


<h3>Value</h3>

<p>An object of class plsr with fiends
</p>
<table>
<tr><td><code>Method</code></td>
<td>
<p>PLSR</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>The X matrix</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>The Y matrix</p>
</td></tr>
<tr><td><code>centerY</code></td>
<td>
<p>Is the Y matrix centered</p>
</td></tr>
<tr><td><code>scaleY</code></td>
<td>
<p>Is the Y matrix scaled</p>
</td></tr>
<tr><td><code>Initial_Transformation</code></td>
<td>
<p>Initial transformation of the Y matrix</p>
</td></tr>
<tr><td><code>ScaledX</code></td>
<td>
<p>Transformed X matrix</p>
</td></tr>
<tr><td><code>ScaledY</code></td>
<td>
<p>Transformed Y matrix</p>
</td></tr>
<tr><td><code>Intercept</code></td>
<td>
<p>Intercept of the model</p>
</td></tr>
<tr><td><code>XScores</code></td>
<td>
<p>Scores for the individals from the X matrix</p>
</td></tr>
<tr><td><code>XWeights</code></td>
<td>
<p>Weigths for the X set</p>
</td></tr>
<tr><td><code>XLoadings</code></td>
<td>
<p>Loadings for the X set</p>
</td></tr>
<tr><td><code>YScores</code></td>
<td>
<p>Scores for the individals from the Y matrix</p>
</td></tr>
<tr><td><code>YWeights</code></td>
<td>
<p>Weigths for the Y set</p>
</td></tr>
<tr><td><code>YLoadings</code></td>
<td>
<p>Loadings for the Y set</p>
</td></tr>
<tr><td><code>RegParameters</code></td>
<td>
<p>Final Regression Parameters</p>
</td></tr>
<tr><td><code>ExpectedY</code></td>
<td>
<p>Expected values of Y</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>R-squared</p>
</td></tr>
<tr><td><code>XStructure</code></td>
<td>
<p>Relation of the X variables with its structure</p>
</td></tr>
<tr><td><code>YStructure</code></td>
<td>
<p>Relation of the Y variables with its structure</p>
</td></tr>
<tr><td><code>YXStructure</code></td>
<td>
<p>Relation of the Y variables with the X components</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>H. Abdi, Partial least squares regression and projection on latent structure regression (PLS regression), WIREs Comput. Stat. 2 (2010), pp. 97-106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Biplot.PLSR">Biplot.PLSR</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X=as.matrix(wine[,4:21])
y=as.numeric(wine[,2])-1
mifit=PLSR(y,X, Validation="None")
</code></pre>

<hr>
<h2 id='PLSR1Bin'>
Partial Least Squares Regression with Binary Response
</h2><span id='topic+PLSR1Bin'></span>

<h3>Description</h3>

<p>Fits Partial Least Squares Regression with Binary Response
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PLSR1Bin(Y, X, S = 2, InitTransform = 5, grouping = NULL, 
tolerance = 5e-06, maxiter = 100, show = FALSE, penalization = 0, 
cte = TRUE, Algorithm = 1, OptimMethod = "CG")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PLSR1Bin_+3A_y">Y</code></td>
<td>

<p>The response
</p>
</td></tr>
<tr><td><code id="PLSR1Bin_+3A_x">X</code></td>
<td>

<p>The matrix of independent variables
</p>
</td></tr>
<tr><td><code id="PLSR1Bin_+3A_s">S</code></td>
<td>

<p>The Dimension of the solution
</p>
</td></tr>
<tr><td><code id="PLSR1Bin_+3A_inittransform">InitTransform</code></td>
<td>

<p>Initial transform for the X matrix
</p>
</td></tr>
<tr><td><code id="PLSR1Bin_+3A_grouping">grouping</code></td>
<td>

<p>Factor for grouping the observations
</p>
</td></tr>
<tr><td><code id="PLSR1Bin_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence of the algorithm
</p>
</td></tr>
<tr><td><code id="PLSR1Bin_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum Number of iterations
</p>
</td></tr>
<tr><td><code id="PLSR1Bin_+3A_show">show</code></td>
<td>

<p>Show the steps of the algorithm
</p>
</td></tr>
<tr><td><code id="PLSR1Bin_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the Ridge Logistic Regression
</p>
</td></tr>
<tr><td><code id="PLSR1Bin_+3A_cte">cte</code></td>
<td>

<p>Should a constant be included in the model?
</p>
</td></tr>
<tr><td><code id="PLSR1Bin_+3A_algorithm">Algorithm</code></td>
<td>

<p>Algorithm used in the calculations
</p>
</td></tr>
<tr><td><code id="PLSR1Bin_+3A_optimmethod">OptimMethod</code></td>
<td>

<p>Optimization methods from optim
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The procedure uses the algorithm proposed by Bastien et al () to fit a Partial Lest Squares Regression when the response is Binary. The procedure will be later converted into a Biplot to visulize the results.
</p>


<h3>Value</h3>

<p>Still to be finished




</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'># No examples yet
</code></pre>

<hr>
<h2 id='PLSRBin'>
Partial Least Squares Regression with several Binary Responses
</h2><span id='topic+PLSRBin'></span>

<h3>Description</h3>

<p>Fits Partial Least Squares Regression with several Binary Responses
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PLSRBin(Y, X, S = 2, InitTransform = 5, grouping = NULL, 
tolerance = 5e-05, maxiter = 100, show = FALSE, penalization = 0.1, 
cte = TRUE, OptimMethod = "CG", Multiple = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PLSRBin_+3A_y">Y</code></td>
<td>

<p>The response
</p>
</td></tr>
<tr><td><code id="PLSRBin_+3A_x">X</code></td>
<td>

<p>The matrix of independent variables
</p>
</td></tr>
<tr><td><code id="PLSRBin_+3A_s">S</code></td>
<td>

<p>The Dimension of the solution
</p>
</td></tr>
<tr><td><code id="PLSRBin_+3A_inittransform">InitTransform</code></td>
<td>

<p>Initial transform for the X matrix
</p>
</td></tr>
<tr><td><code id="PLSRBin_+3A_grouping">grouping</code></td>
<td>

<p>Grouping variable when the inial transformation is standardization within groups.
</p>
</td></tr>
<tr><td><code id="PLSRBin_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence of the algorithm
</p>
</td></tr>
<tr><td><code id="PLSRBin_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum Number of iterations
</p>
</td></tr>
<tr><td><code id="PLSRBin_+3A_show">show</code></td>
<td>

<p>Show the steps of the algorithm
</p>
</td></tr>
<tr><td><code id="PLSRBin_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the Ridge Logistic Regression
</p>
</td></tr>
<tr><td><code id="PLSRBin_+3A_cte">cte</code></td>
<td>

<p>Should a constant be included in the model?
</p>
</td></tr>
<tr><td><code id="PLSRBin_+3A_optimmethod">OptimMethod</code></td>
<td>

<p>Optimization methods from optim
</p>
</td></tr>
<tr><td><code id="PLSRBin_+3A_multiple">Multiple</code></td>
<td>

<p>The responses are the indicators of a multinomial variable?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function fits the PLSR method for the case when there is a set binary dependent variables, using logistic rather than linear fits to take into account the nature of responses. We term the method PLS-BLR (Partial Least Squares Binary Logistic Regression). This can be considered as a generalization of the NIPALS algorithm when the responses are all binary.
</p>


<h3>Value</h3>

<table>
<tr><td><code>Method</code></td>
<td>
<p>Description of 'comp1'</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>The predictors matrix</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>The responses matrix</p>
</td></tr>
<tr><td><code>Initial_Transformation</code></td>
<td>
<p>Initial Transformation of the X matrix</p>
</td></tr>
<tr><td><code>ScaledX</code></td>
<td>
<p>The scaled X matrix</p>
</td></tr>
<tr><td><code>tolerance</code></td>
<td>
<p>Tolerance used in the algorithm</p>
</td></tr>
<tr><td><code>maxiter</code></td>
<td>
<p>Maximum number of iterations used</p>
</td></tr>
<tr><td><code>penalization</code></td>
<td>
<p>Ridge penalization</p>
</td></tr>
<tr><td><code>IncludeConst</code></td>
<td>
<p>Is the constant included in the model?</p>
</td></tr>
<tr><td><code>XScores</code></td>
<td>
<p>Scores of the X matrix, used later for the biplot</p>
</td></tr>
<tr><td><code>XLoadings</code></td>
<td>
<p>Loadings of the X matrix</p>
</td></tr>
<tr><td><code>YScores</code></td>
<td>
<p>Scores of the Y matrix</p>
</td></tr>
<tr><td><code>YLoadings</code></td>
<td>
<p>Loadings of the Y matrix</p>
</td></tr>
<tr><td><code>Coefficients</code></td>
<td>
<p>Regression coefficients</p>
</td></tr>
<tr><td><code>XStructure</code></td>
<td>
<p>Correlations among the X variables and the PLS scores</p>
</td></tr>
<tr><td><code>Intercepts</code></td>
<td>
<p>Intercepts for the Y loadings</p>
</td></tr>
<tr><td><code>LinTerm</code></td>
<td>
<p>Linear terms for each response</p>
</td></tr>
<tr><td><code>Expected</code></td>
<td>
<p>Expected probabilities for the responses</p>
</td></tr>
<tr><td><code>Predictions</code></td>
<td>
<p>Binary predictions of the responses</p>
</td></tr>
<tr><td><code>PercentCorrect</code></td>
<td>
<p>Global percent of correct predictions</p>
</td></tr>
<tr><td><code>PercentCorrectCols</code></td>
<td>
<p>Percent of correct predictions for each column</p>
</td></tr>
<tr><td><code>Maxima</code></td>
<td>
<p>Column with the maximum probability. Useful when the responses are the indicators of a multinomial variable</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>José Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Ugarte Fajardo, J., Bayona Andrade, O., Criollo Bonilla, R., Cevallos‐Cevallos, J., Mariduena‐Zavala, M., Ochoa Donoso, D., &amp; Vicente Villardon, J. L. (2020). Early detection of black Sigatoka in banana leaves using hyperspectral images. Applications in plant sciences, 8(8), e11383.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
X=as.matrix(wine[,4:21])
Y=cbind(Factor2Binary(wine[,1])[,1], Factor2Binary(wine[,2])[,1])
rownames(Y)=wine[,3]
colnames(Y)=c("Year", "Origin")
pls=PLSRBin(Y,X, penalization=0.1, show=TRUE, S=2)

</code></pre>

<hr>
<h2 id='PLSRBinFit'>
PLS binary regression.
</h2><span id='topic+PLSRBinFit'></span>

<h3>Description</h3>

<p>Fits PLS binary regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PLSRBinFit(Y, X, S = 2, tolerance = 5e-06, maxiter = 100, 
show = FALSE, penalization = 0.1, cte = TRUE, OptimMethod = "CG")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PLSRBinFit_+3A_y">Y</code></td>
<td>

<p>The response
</p>
</td></tr>
<tr><td><code id="PLSRBinFit_+3A_x">X</code></td>
<td>

<p>The matrix of independent variables
</p>
</td></tr>
<tr><td><code id="PLSRBinFit_+3A_s">S</code></td>
<td>

<p>The Dimension of the solution
</p>
</td></tr>
<tr><td><code id="PLSRBinFit_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence of the algorithm
</p>
</td></tr>
<tr><td><code id="PLSRBinFit_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum Number of iterations
</p>
</td></tr>
<tr><td><code id="PLSRBinFit_+3A_show">show</code></td>
<td>

<p>Show the steps of the algorithm
</p>
</td></tr>
<tr><td><code id="PLSRBinFit_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the Ridge Logistic Regression
</p>
</td></tr>
<tr><td><code id="PLSRBinFit_+3A_cte">cte</code></td>
<td>

<p>Should a constant be included in the model?
</p>
</td></tr>
<tr><td><code id="PLSRBinFit_+3A_optimmethod">OptimMethod</code></td>
<td>

<p>Optimization methods from optim
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fits PLS binary regression. It is used for a higher level function.
</p>


<h3>Value</h3>

<p>The PLS fit used by the PLSRBin function.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Ugarte Fajardo, J., Bayona Andrade, O., Criollo Bonilla, R., Cevallos‐Cevallos, J., Mariduena‐Zavala, M., Ochoa Donoso, D., &amp; Vicente Villardon, J. L. (2020). Early detection of black Sigatoka in banana leaves using hyperspectral images. Applications in plant sciences, 8(8), e11383.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not yet
</code></pre>

<hr>
<h2 id='PLSRfit'>
Partial Least Squares Regression (PLSR)
</h2><span id='topic+PLSRfit'></span>

<h3>Description</h3>

<p>Fits a Partial Least Squares Regression (PLSR) to two continuous data matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PLSRfit(Y, X, S = 2, tolerance = 5e-06,
maxiter = 100, show = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PLSRfit_+3A_y">Y</code></td>
<td>

<p>The matrix of dependent variables
</p>
</td></tr>
<tr><td><code id="PLSRfit_+3A_x">X</code></td>
<td>

<p>The Matrix of Independent Variables
</p>
</td></tr>
<tr><td><code id="PLSRfit_+3A_s">S</code></td>
<td>

<p>Dimension of the solution. The default is 2
</p>
</td></tr>
<tr><td><code id="PLSRfit_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for the algorithm.
</p>
</td></tr>
<tr><td><code id="PLSRfit_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations for the algorithm.
</p>
</td></tr>
<tr><td><code id="PLSRfit_+3A_show">show</code></td>
<td>

<p>Logical. Should the calculation process be shown on the screen
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fits a Partial Least Squares Regression (PLSR) to a set of two continuous data matrices
</p>


<h3>Value</h3>

<p>An object of class &quot;PLSR&quot; 
</p>
<table>
<tr><td><code>Method</code></td>
<td>
<p>PLSR1</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>Independent Variables</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Dependent Variables</p>
</td></tr>
<tr><td><code>center</code></td>
<td>
<p>Are data centered?</p>
</td></tr>
<tr><td><code>scale</code></td>
<td>
<p>Are data scaled?</p>
</td></tr>
<tr><td><code>ScaledX</code></td>
<td>
<p>Scaled Independent Variables</p>
</td></tr>
<tr><td><code>ScaledY</code></td>
<td>
<p>Scaled Dependent Variables</p>
</td></tr>
<tr><td><code>XScores</code></td>
<td>
<p>Scores for the Independent Variables</p>
</td></tr>
<tr><td><code>XWeights</code></td>
<td>
<p>Weights for the Independent Variables - coefficients of the linear combination</p>
</td></tr>
<tr><td><code>XLoadings</code></td>
<td>
<p>Factor loadings for the Independent Variables </p>
</td></tr>
<tr><td><code>YScores</code></td>
<td>
<p>Scores for the Dependent Variables</p>
</td></tr>
<tr><td><code>YWeights</code></td>
<td>
<p>Weights for the Dependent Variables - coefficients of the linear combination</p>
</td></tr>
<tr><td><code>YLoadings</code></td>
<td>
<p>Factor loadings for the Dependent Variables </p>
</td></tr>
<tr><td><code>XStructure</code></td>
<td>
<p>Structure Correlations for the Independent Variables </p>
</td></tr>
<tr><td><code>YStructure</code></td>
<td>
<p>Structure Correlations for the Dependent Variables </p>
</td></tr>
<tr><td><code>YXStructure</code></td>
<td>
<p>Structure Correlations two groups</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Wold, S., Sjöström, M., &amp; Eriksson, L. (2001). PLS-regression: a basic tool of chemometrics. Chemometrics and intelligent laboratory systems, 58(2), 109-130.
</p>

<hr>
<h2 id='PoliticalFigures'>
Political Figures in the USA
</h2><span id='topic+PoliticalFigures'></span>

<h3>Description</h3>

<p>Does the American public actively differentiate political stimuli along ideological lines?. Dissimilarities among 13 political figurein the USA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("PoliticalFigures")</code></pre>


<h3>Format</h3>

<p>A data frame with the dissimilarities among 13 political figures in the USA.
</p>

<dl>
<dt><code>G._W._Bush</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>John_Kerry</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>Ralph_Nader</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>Dick_Cheney</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>John_Edwards</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>Laura_Bush</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>Hillary_Clinton</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>Bill_Clinton</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>Colin_Powell</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>John_Ashcroft</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>John_McCain</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>Democ._Party</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
<dt><code>Repub._Party</code></dt><dd><p>a numeric vector with the dissimilarities with the other figures</p>
</dd>
</dl>



<h3>Details</h3>

<p>We have taken information from the 2004 CPS American National Election Study. 
Specifically 711 NES respondents' feeling thermometer ratings of thirteen prominent political
figures from the period of the 2004 election: George W. Bush; John Kerry; Ralph Nader;
Richard Cheney; John Edwards; Laura Bush; Hillary Clinton; Bill Clinton; Colin Powell;
John Ashcroft; John McCain; the Democratic party; and the Republican party. 
With the respondent scores, a dissimilarity among each pair of figures 
</p>


<h3>Source</h3>

<p>Jacoby, W. G., &amp; Armstrong, D. A. (2014). Bootstrap Confidence Regions for Multidimensional Scaling Solutions. American Journal of Political Science, 58(1), 264-278.
</p>


<h3>References</h3>

<p>Jacoby, W. G., &amp; Armstrong, D. A. (2014). Bootstrap Confidence Regions for Multidimensional Scaling Solutions. American Journal of Political Science, 58(1), 264-278.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='PolyOrdinalLogBiplot'>
Factor Analysis Biplot based on polychoric correlations
</h2><span id='topic+PolyOrdinalLogBiplot'></span>

<h3>Description</h3>

<p>Calculates a biplot for ordinal data based on polychoric correlations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PolyOrdinalLogBiplot(X, dimension = 3, method = "principal", 
rotate = "varimax", RescaleCoordinates = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PolyOrdinalLogBiplot_+3A_x">X</code></td>
<td>

<p>A matrix of ordinal data
</p>
</td></tr>
<tr><td><code id="PolyOrdinalLogBiplot_+3A_dimension">dimension</code></td>
<td>

<p>Number of dimensiona to retain
</p>
</td></tr>
<tr><td><code id="PolyOrdinalLogBiplot_+3A_method">method</code></td>
<td>

<p>Principal components (principal) or factor analysis (fa)
</p>
</td></tr>
<tr><td><code id="PolyOrdinalLogBiplot_+3A_rotate">rotate</code></td>
<td>

<p>Rotation for the analysis
</p>
</td></tr>
<tr><td><code id="PolyOrdinalLogBiplot_+3A_rescalecoordinates">RescaleCoordinates</code></td>
<td>

<p>Rescale coordinates as in a continuous data biplot
</p>
</td></tr>
<tr><td><code id="PolyOrdinalLogBiplot_+3A_...">...</code></td>
<td>

<p>Any aditional arguments for the principal and fa functions
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The procedure calculates 
</p>


<h3>Value</h3>

<p>A biplot (Continuous or ordinal)
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>See Also</h3>

<p><code><a href="psych.html#topic+fa">fa</a></code>, <code><a href="psych.html#topic+principal">principal</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not Yet
</code></pre>

<hr>
<h2 id='PrettyTicks'>
Calculates loose axis ticks and labels using nice numbers
</h2><span id='topic+PrettyTicks'></span>

<h3>Description</h3>

<p>Calculates axis ticks and labels using nice numbers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PrettyTicks(min = -3, max = 3, ntick = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PrettyTicks_+3A_min">min</code></td>
<td>

<p>Minimum value on the axis
</p>
</td></tr>
<tr><td><code id="PrettyTicks_+3A_max">max</code></td>
<td>

<p>maximum value on the axis.
</p>
</td></tr>
<tr><td><code id="PrettyTicks_+3A_ntick">ntick</code></td>
<td>

<p>Approximated number of desired ticks
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates axis ticks and labels using nice numbers. The resulting labels are known as loose labels.
</p>


<h3>Value</h3>

<p>A list with the following fields
</p>
<table>
<tr><td><code>ticks</code></td>
<td>
<p>Ticks for the axis</p>
</td></tr>
<tr><td><code>labels</code></td>
<td>
<p>The corresponding labels</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Heckbert, P. S. (1990). Nice numbers for graph labels. In Graphics Gems (pp. 61-63). Academic Press Professional, Inc..
</p>


<h3>See Also</h3>

<p><code><a href="#topic+NiceNumber">NiceNumber</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>PrettyTicks(-4, 4, 5)
</code></pre>

<hr>
<h2 id='PrincipalCoordinates'>Principal Coordinates Analysis</h2><span id='topic+PrincipalCoordinates'></span>

<h3>Description</h3>

<p>Principal coordinates Analysis for a matrix of proximities obtained from binary, categorical, continuous or mixed data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PrincipalCoordinates(Proximities, w = NULL, dimension = 2, 
method = "eigen", tolerance = 1e-04, Bootstrap = FALSE, 
BootstrapType = c("Distances", "Products"), nB = 200, 
ProcrustesRot = TRUE, BootstrapMethod = c("Sampling", "Permutation"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PrincipalCoordinates_+3A_proximities">Proximities</code></td>
<td>

<p>An object of class <code>proximities</code>.
</p>
</td></tr>
<tr><td><code id="PrincipalCoordinates_+3A_w">w</code></td>
<td>

<p>An set of weights.
</p>
</td></tr>
<tr><td><code id="PrincipalCoordinates_+3A_dimension">dimension</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="PrincipalCoordinates_+3A_method">method</code></td>
<td>

<p>Method to calculate the eigenvalues and eigenvectors. The default is the usual
eigen function although the Power Method to calculate only tre first eigenvectors can be used.
</p>
</td></tr>
<tr><td><code id="PrincipalCoordinates_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for the eigenvalues
</p>
</td></tr>
<tr><td><code id="PrincipalCoordinates_+3A_bootstrap">Bootstrap</code></td>
<td>

<p>Should Bootstrap be calculated?
</p>
</td></tr>
<tr><td><code id="PrincipalCoordinates_+3A_bootstraptype">BootstrapType</code></td>
<td>

<p>Bootstrap on the residuals of the &quot;distance&quot; or &quot;scalar products&quot; matrix.
</p>
</td></tr>
<tr><td><code id="PrincipalCoordinates_+3A_nb">nB</code></td>
<td>

<p>Number of Bootstrap replications
</p>
</td></tr>
<tr><td><code id="PrincipalCoordinates_+3A_procrustesrot">ProcrustesRot</code></td>
<td>

<p>Should each replication be rotated to match the initial solution?
</p>
</td></tr>
<tr><td><code id="PrincipalCoordinates_+3A_bootstrapmethod">BootstrapMethod</code></td>
<td>

<p>The replications are obtained &quot;Sampling&quot; or &quot;Permutating&quot; the residuals.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Principal Coordinates Analysis for a proximity matrix previously calculated from a matrix of raw data or directly obsrved proximities.
</p>


<h3>Value</h3>

<p>An object of class <code>Principal.Coordinates</code>. The function adds the information of the Principal Coordinates to the object of class  <code>proximities</code>. Together with the information about the proximities the object has:
</p>
<table>
<tr><td><code>Analysis</code></td>
<td>
<p>The type of analysis performed, &quot;Principal Coordinates&quot; in this case</p>
</td></tr>
<tr><td><code>Eigenvalues</code></td>
<td>
<p>The eigenvalues of the PCoA</p>
</td></tr>
<tr><td><code>Inertia</code></td>
<td>
<p>The Inertia of the PCoA</p>
</td></tr>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates for the objects in the PCoA</p>
</td></tr>
<tr><td><code>RowQualities</code></td>
<td>
<p>Qualities of representation for the objects in the PCoA</p>
</td></tr>
<tr><td><code>RawStress</code></td>
<td>
<p>Raw Stress values</p>
</td></tr>
<tr><td><code>stress1</code></td>
<td>
<p>stress formula 1</p>
</td></tr>
<tr><td><code>stress2</code></td>
<td>
<p>stress formula 2</p>
</td></tr>
<tr><td><code>sstress1</code></td>
<td>
<p>sstress formula 1</p>
</td></tr>
<tr><td><code>sstress2</code></td>
<td>
<p>sstress formula 2</p>
</td></tr>
<tr><td><code>rsq</code></td>
<td>
<p>Squared correlation between disparities and distances</p>
</td></tr>
<tr><td><code>Spearman</code></td>
<td>
<p>Spearman correlation between disparities and distances</p>
</td></tr>
<tr><td><code>Kendall</code></td>
<td>
<p>Kendall correlation between disparities and distances</p>
</td></tr>
<tr><td><code>BootstrapInfo</code></td>
<td>
<p>The result of the bootstrap calculations</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Gower, J. C. (2006) Similarity dissimilarity and Distance, measures of. Encyclopedia of Statistical Sciences. 2nd. ed. Volume 12. Wiley
</p>
<p>Gower, J.C. (1966). Some distance properties of latent root and vector methods used in multivariate analysis. Biometrika 53: 325-338.
</p>
<p>J.R. Demey, J.L. Vicente-Villardon, M.P. Galindo, A.Y. Zambrano, Identifying molecular markers associated with classifications of genotypes by external logistic
biplot, Bioinformatics 24 (2008) 2832.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BinaryProximities">BinaryProximities</a></code>, <code><a href="#topic+BootstrapDistance">BootstrapDistance</a></code>, <code><a href="#topic+BootstrapDistance">BootstrapDistance</a></code>, <code><a href="#topic+BinaryProximities">BinaryProximities</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
Dis=BinaryProximities(spiders)
pco=PrincipalCoordinates(Dis)
Dis=BinaryProximities(spiders)
pco=PrincipalCoordinates(Dis, Bootstrap=TRUE)

</code></pre>

<hr>
<h2 id='print.MGC'>
Prints the results of Model-Based Gaussian Clustering algorithms
</h2><span id='topic+print.MGC'></span>

<h3>Description</h3>

<p>Prints the results of Model-Based Gaussian Clustering algorithms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MGC'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.MGC_+3A_x">x</code></td>
<td>

<p>An object of class &quot;MGC&quot;
</p>
</td></tr>
<tr><td><code id="print.MGC_+3A_...">...</code></td>
<td>

<p>Any aditional parameters
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Prints the results of Model-Based Gaussian Clustering algorithms
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

</code></pre>

<hr>
<h2 id='print.RidgeBinaryLogistic'>
prints an object of class RidgeBinaryLogistic
</h2><span id='topic+print.RidgeBinaryLogistic'></span>

<h3>Description</h3>

<p>prints an object of class RidgeBinaryLogistic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RidgeBinaryLogistic'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.RidgeBinaryLogistic_+3A_x">x</code></td>
<td>

<p>An object of class 
</p>
</td></tr>
<tr><td><code id="print.RidgeBinaryLogistic_+3A_...">...</code></td>
<td>

<p>Aditional Arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Prints an object of class RidgeBinaryLogistic
</p>


<h3>Value</h3>

<p>The main resuls of a binary logistic regression
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='Protein'>
Protein consumption data.
</h2><span id='topic+Protein'></span>

<h3>Description</h3>

<p>Protein consumption in twenty-five European countries for nine food groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Protein)</code></pre>


<h3>Format</h3>

<p>A data frame with 25 observations on the following 11 variables.
</p>

<dl>
<dt><code>Comunist</code></dt><dd><p>a factor with levels <code>No</code> <code>Yes</code></p>
</dd>
<dt><code>Region</code></dt><dd><p>a factor with levels <code>North</code> <code>Center</code> <code>South</code></p>
</dd>
<dt><code>Red_Meat</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>White_Meat</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Eggs</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Milk</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Fish</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cereal</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Starch</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Nuts</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Fruits_Vegetables</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>These data measure protein consumption in twenty-five European countries for nine food groups. It is possible to use multivariate methods to determine whether there are groupings of countries and whether meat consumption is related to that of other foods.
</p>


<h3>Source</h3>

<p>http://lib.stat.cmu.edu/DASL/Datafiles/Protein.html
</p>


<h3>References</h3>

<p>Weber, A. (1973) Agrarpolitik im Spannungsfeld der internationalen Ernaehrungspolitik, Institut fuer Agrarpolitik und marktlehre, Kiel. 
</p>
<p>Gabriel, K.R. (1981) Biplot display of multivariate matrices for inspection of data and diagnosis. In Interpreting Multivariate Data (Ed. V. Barnett), New York: John Wiley &amp; Sons, 147-173. 
</p>
<p>Hand, D.J., et al. (1994) A Handbook of Small Data Sets, London: Chapman &amp; Hall, 297-298.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Protein)
## maybe str(Protein) ; plot(Protein) ...
</code></pre>

<hr>
<h2 id='RAPD'>
Sugar Cane Data
</h2><span id='topic+RAPD'></span>

<h3>Description</h3>

<p>Molecular characteristics of 50 varieties of sugar cane.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(RAPD)</code></pre>


<h3>Format</h3>

<p>A data frame with 50 observations on 168 variables. 1-120: Random aplified polymorphic DNA and 121-168: Microsatellites
</p>


<h3>Details</h3>

<p>Dta are codified as presence or absence of the dominant marker.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RAPD)
## maybe str(RAPD) ; plot(RAPD) ...
</code></pre>

<hr>
<h2 id='RemoveRowsWithNaNs'>
Remove rows that contains NaNs (missing data)
</h2><span id='topic+RemoveRowsWithNaNs'></span>

<h3>Description</h3>

<p>Remove rows that contains NaNs to obtain a matrix wothout missind data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RemoveRowsWithNaNs(x, cols = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RemoveRowsWithNaNs_+3A_x">x</code></td>
<td>

<p>The matrix to be arranged
</p>
</td></tr>
<tr><td><code id="RemoveRowsWithNaNs_+3A_cols">cols</code></td>
<td>

<p>A set of columns to check as a vector of integers
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Remove rows that contains NaNs to obtain a matrix wothout missind data
</p>


<h3>Value</h3>



<table>
<tr><td><code>x</code></td>
<td>
<p>Matrix without missing data</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>

<hr>
<h2 id='riano'>
Ecological data from Riano (Spain)
</h2><span id='topic+riano'></span>

<h3>Description</h3>

<p>Ecological data from Riano (Spain)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("riano")</code></pre>


<h3>Format</h3>

<p>A data frame with 70 observations on the following 25 variables.
</p>

<dl>
<dt><code>Week</code></dt><dd><p>a factor with levels <code>A</code> <code>B</code> <code>C</code> <code>D</code> <code>E</code> <code>F</code> <code>G</code> <code>H</code> <code>I</code> <code>J</code></p>
</dd>
<dt><code>Depth</code></dt><dd><p>a factor with levels <code>0</code> <code>2</code> <code>5</code> <code>10</code> <code>15</code> <code>20</code> <code>Bottom</code></p>
</dd>
<dt><code>Cianof</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Crisof</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Haptof</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Crasp</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Cripto</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Dinof</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Diatom</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Euglen</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Prasin</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Clorof</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Zigofi</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Xantof</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>malgas</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Ta</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X02</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>pH</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COND</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SiO2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>P.PO4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Chla</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Chlb</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Chlc</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>IM</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Ecological data from Riano (Spain). Abundance of several algae taxonomic groups and several environmental variables
</p>


<h3>Source</h3>

<p>Department of Ecology. University of Leon. Spain
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(riano)
## maybe str(riano) ; plot(riano) ...
</code></pre>

<hr>
<h2 id='RidgeBinaryLogistic'>
Ridge Binary Logistic Regression for Binary data
</h2><span id='topic+RidgeBinaryLogistic'></span>

<h3>Description</h3>

<p>This function performs a logistic regression between a dependent binary variable <code>y</code> and some independent
variables <code>x</code>, solving the separation problem in this type of regression using ridge 
penalization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RidgeBinaryLogistic(y, X = NULL, data = NULL, freq = NULL, 
tolerance = 1e-05, maxiter = 100, penalization = 0.2, 
cte = FALSE, ref = "first", bootstrap = FALSE, nmB = 100, 
RidgePlot = FALSE, MinLambda = 0, MaxLambda = 2, StepLambda = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RidgeBinaryLogistic_+3A_y">y</code></td>
<td>

<p>A binary dependent variable or a formula
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_x">X</code></td>
<td>

<p>A set of independent variables when y is not a formula.
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_data">data</code></td>
<td>

<p>data frame for the formula
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_freq">freq</code></td>
<td>

<p>frequencies for each observation (usually 1)
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_penalization">penalization</code></td>
<td>

<p>Ridige penalization: a non negative constant. Penalization used in the diagonal matrix to avoid singularities.
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_cte">cte</code></td>
<td>

<p>Should the model have a constant? 
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_ref">ref</code></td>
<td>

<p>Category of reference
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_bootstrap">bootstrap</code></td>
<td>

<p>Should bootstrap confidence intervals be calculated?
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_nmb">nmB</code></td>
<td>

<p>Number of bootstrap samples.
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_ridgeplot">RidgePlot</code></td>
<td>

<p>Should the ridge plot be plotted?
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_minlambda">MinLambda</code></td>
<td>

<p>Minimum value of lambda for the rigge plot
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_maxlambda">MaxLambda</code></td>
<td>

<p>Maximum value of lambda for the rigge plot
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogistic_+3A_steplambda">StepLambda</code></td>
<td>

<p>Step for increasing the values of lambda
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Logistic Regression is a widely used technique in applied work when a binary, nominal or ordinal response variable is available, due to the fact that classical regression methods are not applicable to this kind of variables. The method is available in most of the statistical packages, commercial or free. Maximum Likelihood together with a numerical method as Newton-Raphson, is used to estimate the parameters of the model.
In logistic regression, when in the space generated by the independent variables there are hyperplanes that separate among the individuals belonging to the different groups defined by the response, maximum likelihood does not converge and the estimations tend to the infinity. That is known in the literature as the separation problem in logistic regression. Even when the separation is not complete, the numerical solution of the maximum likelihood has stability problems.
From a practical point of view, that means the estimated model is not accurate precisely when there should be a perfect, or almost perfect, fit to the data.
</p>
<p>The problem of the existence of the estimators in logistic regression can be seen in Albert (1984), a solution for the binary case, based on the Firth method, Firth (1993) is proposed by Heinze(2002).
The extension to nominal logistic model was made by Bull (2002). All the procedures were initially
developed to remove the bias but work well to avoid the problem of separation. Here we have chosen
a simpler solution based on ridge estimators for logistic regression Cessie(1992).
</p>
<p>Rather than maximizing <code class="reqn">{L_j}(\left. {\bf{G}} \right|{{\bf{b}}_{j0}},{{\bf{B}}_j})</code> we maximize
</p>
<p style="text-align: center;"><code class="reqn">{{L_j}(\left. {\bf{G}} \right|{{\bf{b}}_{j0}},{{\bf{B}}_j})} - \lambda \left( {\left\| {{{\bf{b}}_{j0}}} \right\| + \left\| {{{\bf{B}}_j}} \right\|} \right)</code>
</p>

<p>Changing the values of <code class="reqn">\lambda</code> we obtain slightly different solutions not affected by the separation problem.
</p>


<h3>Value</h3>

<p>An object of class <code>RidgeBinaryLogistic</code> with the following components

</p>
<table>
<tr><td><code>beta</code></td>
<td>
<p>Estimates of the coefficients</p>
</td></tr>
<tr><td><code>fitted</code></td>
<td>
<p>Fitted probabilities</p>
</td></tr>
<tr><td><code>residuals</code></td>
<td>
<p>Residuals of the model</p>
</td></tr>
<tr><td><code>Prediction</code></td>
<td>
<p>Predictions of presences and absences</p>
</td></tr>
<tr><td><code>Covariances</code></td>
<td>
<p>Covariances among the estimates</p>
</td></tr>
<tr><td><code>Deviance</code></td>
<td>
<p>Deviance of the current model</p>
</td></tr>
<tr><td><code>NullDeviance</code></td>
<td>
<p>Deviance of the null model</p>
</td></tr>
<tr><td><code>Dif</code></td>
<td>
<p>Difference between the deviances of the cirrent and null models</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>Degrees of freedom of the difference</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>p-value</p>
</td></tr>
<tr><td><code>CoxSnell</code></td>
<td>
<p>Cox-Snell pseudo R-squared</p>
</td></tr>
<tr><td><code>Nagelkerke</code></td>
<td>
<p>Nagelkerke pseudo R-squared</p>
</td></tr>
<tr><td><code>MacFaden</code></td>
<td>
<p>MacFaden pseudo R-squared</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>Pseudo R-squared using the residuals</p>
</td></tr>
<tr><td><code>Classification</code></td>
<td>
<p>Classification table</p>
</td></tr>
<tr><td><code>PercentCorrect</code></td>
<td>
<p>Percentage of correct classification</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Agresti, A. (1990) An Introduction to Categorical Data Analysis. John Wiley and Sons, Inc.
</p>
<p>Albert, A. and Anderson, J. A.  (1984) On the existence of maximum likelihood estimates in logistic regression models. Biometrika, 71(1): 1-10.  
</p>
<p>Anderson, J. A. (1972), Separate sample logistic discrimination. Biometrika, 59(1): 19-35.
</p>
<p>Anderson, J. A. &amp; Philips P. R. (1981)  Regression, discrimination and measurement models for ordered categorical variables. Appl. Statist, 30: 22-31.
</p>
<p>Bull, S. B., Mk, C. &amp; Greenwood, C. M. (2002) A modified score function for multinomial logistic regression. Computational Statistics and data Analysis, 39: 57-74.
</p>
<p>Cortinhas Abrantes, J. &amp; Aerts, M. (2012) A solution to separation for clustered binary data. Statistical Modelling, 12 (1): 3-27.
</p>
<p>Cox, D. R. (1970), Analysis of Binary Data. Methuen. London.
</p>
<p>Demey, J., Vicente-Villardon, J. L., Galindo, M.P.  AND Zambrano, A. (2008) Identifying Molecular Markers Associated With Classification Of Genotypes Using External Logistic Biplots. Bioinformatics, 24(24): 2832-2838.
</p>
<p>Firth D, (1993) Bias Reduction of Maximum Likelihood Estimates, Biometrika, Vol, 80, No, 1, (Mar,, 1993), pp, 27-38.
</p>
<p>Fox, J. (1984) Linear Statistical Models and Related Methods. Wiley. New York.
</p>
<p>Harrell, F. E. (2012). rms: Regression Modeling Strategies. R package version 3.5-0. http://CRAN.R-project.org/package=rms
</p>
<p>Harrell, F. E. (2001). Regression Modeling Strategies: With Applications to Linear Models, Logistic Regression, and Survival Analysis (Springer Series in Statistics). Springer. New York.
</p>
<p>Heinze G, and Schemper M, (2002) A solution to the problem of separation in logistic regresion. Statist. Med., 21:2409-2419
</p>
<p>Heinze G. and Ploner M. (2004) Fixing the nonconvergence bug in logistic regression with SPLUS and SAS. Computer Methods and Programs in Biomedicine 71 p, 181-187
</p>
<p>Heinze, G. (2006) A comparative investigation of methods for logistic regression with separated or nearly separated data. Statist. Med., 25:4216-4226.
</p>
<p>Heinze, G. and Puhr, R. (2010) Bias-reduced and separation-proof conditional logistic regression with small or sparse data sets. Statist. Med. 29: 770-777.
</p>
<p>Hoerl, A. E. and Kennard, R.W. (1971) Rige Regression: biased estimators for nonorthogonal problems. Technometrics, 21: 55 67.
</p>
<p>Sun, H. and Wang S.  Penalized logistic regression for high-dimensional DNA methylation data with case-control studies. Bioinformatics. 28 (10): 1368-1375.
</p>
<p>Hosmer, D. and  Lemeshow, L. (1989) Applied Logistic Regression. John Wiley and Sons. Inc.   
</p>
<p>Le Cessie, S. and Van Houwelingen, J.C. (1992) Ridge Estimators in Logistic Regression.  Appl. Statist. 41 (1): 191-201.
</p>
<p>Malo, N., Libiger, O. and Schork, N. J. (2008) Accommodating Linkage Disequilibrium in Genetic-Association Analyses via Ridge Regression. Am J Hum Genet. 82(2): 375-385.
</p>
<p>Silvapulle, M. J. (1981) On the existence of maximum likelihood estimates for the binomial response models. J. R. Statist. Soc. B 43: 310-3.
</p>
<p>Vicente-Villardon, J. L., Galindo, M. P. and Blazquez, A. (2006) Logistic Biplots. In Multiple Correspondence Análisis And Related Methods. Grenacre, M &amp; Blasius, J, Eds,  Chapman and Hall, Boca Raton.
</p>
<p>Walter, S. and Duncan, D. (1967) Estimation of the probability of an event as a function of several variables. Biometrika. 54:167-79.
</p>
<p>Wedderburn, R. W. M. (1976) On the existence and uniqueness of the maximum likelihood estimates for certain generalized linear models. Biometrika 63, 27-32.
</p>
<p>Zhu, J. and Hastie, T. (2004) Classification of gene microarrays by penalized logistic regression. Biostatistics. 5(3):427-43.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># not yet
</code></pre>

<hr>
<h2 id='RidgeBinaryLogisticFit'>
Fits a binary logistic regression with ridge penalization
</h2><span id='topic+RidgeBinaryLogisticFit'></span>

<h3>Description</h3>

<p>This function fits a logistic regression between a dependent variable y and 
some independent variables x, and solves the separation problem in this type
of regression using ridge regression and penalization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RidgeBinaryLogisticFit(y, xd, freq, tolerance = 1e-05, maxiter = 100, penalization = 0.2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RidgeBinaryLogisticFit_+3A_y">y</code></td>
<td>

<p>A vector with the values of the dependent variable
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogisticFit_+3A_xd">xd</code></td>
<td>

<p>A matrix with the independent variables
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogisticFit_+3A_freq">freq</code></td>
<td>

<p>Frequencies of each pattern
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogisticFit_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for the iterations.
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogisticFit_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations for convergenc~
</p>
</td></tr>
<tr><td><code id="RidgeBinaryLogisticFit_+3A_penalization">penalization</code></td>
<td>

<p>Penalization used in the diagonal matrix to avoid singularities.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fits a binary logistic regression with ridge penalization
</p>


<h3>Value</h3>

<p>The parameters of the fit
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RidgeBinaryLogistic">RidgeBinaryLogistic</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----

</code></pre>

<hr>
<h2 id='RidgeMultinomialLogisticFit'>
Multinomial logistic regression with ridge penalization
</h2><span id='topic+RidgeMultinomialLogisticFit'></span>

<h3>Description</h3>

<p>This function does a logistic regression between a dependent variable y and 
some independent variables x, and solves the separation problem in this type
of regression using ridge regression and penalization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RidgeMultinomialLogisticFit(y, x, penalization = 0.2, 
tol = 1e-04, maxiter = 200, show = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RidgeMultinomialLogisticFit_+3A_y">y</code></td>
<td>

<p>Dependent variable.
</p>
</td></tr>
<tr><td><code id="RidgeMultinomialLogisticFit_+3A_x">x</code></td>
<td>

<p>A matrix with the independent variables.
</p>
</td></tr>
<tr><td><code id="RidgeMultinomialLogisticFit_+3A_penalization">penalization</code></td>
<td>

<p>Penalization used in the diagonal matrix to avoid singularities.
</p>
</td></tr>
<tr><td><code id="RidgeMultinomialLogisticFit_+3A_tol">tol</code></td>
<td>

<p>Tolerance for the iterations.
</p>
</td></tr>
<tr><td><code id="RidgeMultinomialLogisticFit_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations.
</p>
</td></tr>
<tr><td><code id="RidgeMultinomialLogisticFit_+3A_show">show</code></td>
<td>

<p>Should the iteration history be printed?.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The problem of the existence of the estimators in logistic regression can be seen in Albert (1984), a solution for the binary case, based on the Firth's method, Firth (1993) is proposed by Heinze(2002). The extension to nominal logistic model was made by Bull (2002). All the procedures were initially developed to remove the bias but work well to avoid the problem of separation. Here we have chosen a simpler solution based on ridge estimators for logistic regression Cessie(1992).  
</p>
<p>Rather than maximizing <code class="reqn">{L_j}(\left. {\bf{G}} \right|{{\bf{b}}_{j0}},{{\bf{B}}_j})</code> we maximize
</p>
<p style="text-align: center;"><code class="reqn">{{L_j}(\left. {\bf{G}} \right|{{\bf{b}}_{j0}},{{\bf{B}}_j})} - \lambda \left( {\left\| {{{\bf{b}}_{j0}}} \right\| + \left\| {{{\bf{B}}_j}} \right\|} \right)</code>
</p>

<p>Changing the values of <code class="reqn">\lambda</code> we obtain slightly different solutions not affected by the separation problem.
</p>


<h3>Value</h3>

<p>An object of class <code>"rmlr"</code> with components
</p>
<table>
<tr><td><code>fitted</code></td>
<td>
<p>Matrix with the fitted probabilities</p>
</td></tr>
<tr><td><code>cov</code></td>
<td>
<p>Covariance matrix among the estimates</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Indicator matrix for the dependent variable</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>Estimated coefficients for the multinomial logistic regression</p>
</td></tr>
<tr><td><code>stderr</code></td>
<td>
<p>Standard error of the estimates</p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>Logarithm of the likelihood</p>
</td></tr>
<tr><td><code>Deviance</code></td>
<td>
<p>Deviance of the model</p>
</td></tr>
<tr><td><code>AIC</code></td>
<td>
<p>Akaike information criterion indicator</p>
</td></tr> 
<tr><td><code>BIC</code></td>
<td>
<p>Bayesian information criterion indicator</p>
</td></tr> 
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Albert,A. &amp; Anderson,J.A. (1984),<em>On the existence of maximum likelihood estimates in logistic regression models</em>, Biometrika 71(1), 1&ndash;10.
</p>
<p>Bull, S.B., Mak, C. &amp; Greenwood, C.M. (2002), <em>A modified score function for multinomial logistic regression</em>, Computational Statistics and dada Analysis 39, 57&ndash;74.
</p>
<p>Firth, D.(1993), <em>Bias reduction of maximum likelihood estimates</em>, Biometrika 80(1), 27&ndash;38
</p>
<p>Heinze, G. &amp; Schemper, M. (2002), <em>A solution to the problem of separation in logistic regression</em>, Statistics in Medicine 21, 2109&ndash;2419
</p>
<p>Le Cessie, S. &amp; Van Houwelingen, J. (1992), <em>Ridge estimators in logistic regression</em>, Applied Statistics 41(1), 191&ndash;201.  
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # No examples yet
</code></pre>

<hr>
<h2 id='RidgeMultinomialLogisticRegression'>
Ridge Multinomial Logistic Regression
</h2><span id='topic+RidgeMultinomialLogisticRegression'></span>

<h3>Description</h3>

<p>Function that calculates an object with the fitted multinomial logistic regression for a nominal variable. It compares with the null model, so that we will be able to compare which model fits better the variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RidgeMultinomialLogisticRegression(formula, data, penalization = 0.2,
cte = TRUE, tol = 1e-04, maxiter = 200, showIter = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RidgeMultinomialLogisticRegression_+3A_formula">formula</code></td>
<td>

<p>The usual formula notation (or the dependent variable)
</p>
</td></tr>
<tr><td><code id="RidgeMultinomialLogisticRegression_+3A_data">data</code></td>
<td>

<p>The dataframe used by the formula. (or a matrix with the independent variables).
</p>
</td></tr>
<tr><td><code id="RidgeMultinomialLogisticRegression_+3A_penalization">penalization</code></td>
<td>

<p>Penalization used in the diagonal matrix to avoid singularities.
</p>
</td></tr>
<tr><td><code id="RidgeMultinomialLogisticRegression_+3A_cte">cte</code></td>
<td>

<p>Should the model have a constant? 
</p>
</td></tr>
<tr><td><code id="RidgeMultinomialLogisticRegression_+3A_tol">tol</code></td>
<td>

<p>Value to stop the process of iterations.
</p>
</td></tr>
<tr><td><code id="RidgeMultinomialLogisticRegression_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations.
</p>
</td></tr>
<tr><td><code id="RidgeMultinomialLogisticRegression_+3A_showiter">showIter</code></td>
<td>

<p>Should the iteration history be printed?.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object that has the following components:
</p>
<table>
<tr><td><code>fitted</code></td>
<td>
<p>Matrix with the fitted probabilities</p>
</td></tr>
<tr><td><code>cov</code></td>
<td>
<p>Covariance matrix among the estimates</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Indicator matrix for the dependent variable</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>Estimated coefficients for the multinomial logistic regression</p>
</td></tr>
<tr><td><code>stderr</code></td>
<td>
<p>Standard error of the estimates</p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>Logarithm of the likelihood</p>
</td></tr>
<tr><td><code>Deviance</code></td>
<td>
<p>Deviance of the model</p>
</td></tr>
<tr><td><code>AIC</code></td>
<td>
<p>Akaike information criterion indicator</p>
</td></tr> 
<tr><td><code>BIC</code></td>
<td>
<p>Bayesian information criterion indicator</p>
</td></tr> 
<tr><td><code>NullDeviance</code></td>
<td>
<p>Deviance of the null model</p>
</td></tr>
<tr><td><code>Difference</code></td>
<td>
<p>Difference between the two deviance values</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>Degrees of freedom</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>p-value asociated to the chi-squared estimate</p>
</td></tr>
<tr><td><code>CoxSnell</code></td>
<td>
<p>Cox and Snell pseudo R squared</p>
</td></tr>
<tr><td><code>Nagelkerke</code></td>
<td>
<p>Nagelkerke pseudo R squared</p>
</td></tr>
<tr><td><code>MacFaden</code></td>
<td>
<p>MacFaden pseudo R squared</p>
</td></tr>
<tr><td><code>Table</code></td>
<td>
<p>Cross classification of observed and predicted responses</p>
</td></tr>
<tr><td><code>PercentCorrect</code></td>
<td>
<p>Percentage of correct classifications</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Albert,A. &amp; Anderson,J.A. (1984),<em>On the existence of maximum likelihood estimates in logistic regression models</em>, Biometrika 71(1), 1&ndash;10.
</p>
<p>Bull, S.B., Mak, C. &amp; Greenwood, C.M. (2002), <em>A modified score function for multinomial logistic regression</em>, Computational Statistics and dada Analysis 39, 57&ndash;74.
</p>
<p>Firth, D.(1993), <em>Bias reduction of maximum likelihood estimates</em>, Biometrika 80(1), 27&ndash;38
</p>
<p>Heinze, G. &amp; Schemper, M. (2002), <em>A solution to the problem of separation in logistic regression</em>, Statistics in Medicine 21, 2109&ndash;2419
</p>
<p>Le Cessie, S. &amp; Van Houwelingen, J. (1992), <em>Ridge estimators in logistic regression</em>, Applied Statistics 41(1), 191&ndash;201.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RidgeMultinomialLogisticFit">RidgeMultinomialLogisticFit</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
  data(Protein)
  y=Protein[[2]]
  X=Protein[,c(3,11)]
  rmlr = RidgeMultinomialLogisticRegression(y,X,penalization=0.0)
  summary(rmlr)
  
</code></pre>

<hr>
<h2 id='RidgeOrdinalLogistic'>
Ordinal logistic regression with ridge penalization
</h2><span id='topic+RidgeOrdinalLogistic'></span>

<h3>Description</h3>

<p>This function performs a logistic regression between a dependent ordinal variable y
and some independent variables x, and solves the separation problem using ridge penalization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RidgeOrdinalLogistic(y, x, penalization = 0.1, tol = 1e-04, maxiter = 200, show = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RidgeOrdinalLogistic_+3A_y">y</code></td>
<td>

<p>Dependent variable.
</p>
</td></tr>
<tr><td><code id="RidgeOrdinalLogistic_+3A_x">x</code></td>
<td>

<p>A matrix with the independent variables.
</p>
</td></tr>
<tr><td><code id="RidgeOrdinalLogistic_+3A_penalization">penalization</code></td>
<td>

<p>Penalization used to avoid singularities.
</p>
</td></tr>
<tr><td><code id="RidgeOrdinalLogistic_+3A_tol">tol</code></td>
<td>

<p>Tolerance for the iterations.
</p>
</td></tr>
<tr><td><code id="RidgeOrdinalLogistic_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations.
</p>
</td></tr>
<tr><td><code id="RidgeOrdinalLogistic_+3A_show">show</code></td>
<td>

<p>Should the iteration history be printed?.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The problem of the existence of the estimators in logistic regression can be seen in Albert (1984); a solution for the binary case, based on the Firth's method, Firth (1993) is proposed by Heinze(2002). All the procedures were initially developed to remove the bias but work well to avoid the problem of separation. Here we have chosen a simpler solution based on ridge estimators for logistic regression Cessie(1992).  
</p>
<p>Rather than maximizing <code class="reqn">{L_j}(\left. {\bf{G}} \right|{{\bf{b}}_{j0}},{{\bf{B}}_j})</code> we maximize
</p>
<p style="text-align: center;"><code class="reqn">{{L_j}(\left. {\bf{G}} \right|{{\bf{b}}_{j0}},{{\bf{B}}_j})} - \lambda \left( {\left\| {{{\bf{b}}_{j0}}} \right\| + \left\| {{{\bf{B}}_j}} \right\|} \right)</code>
</p>

<p>Changing the values of <code class="reqn">\lambda</code> we obtain slightly different solutions not affected by the separation problem.
</p>


<h3>Value</h3>

<p>An object of class <code>"pordlogist"</code>.  This has components:
</p>
<table>
<tr><td><code>nobs</code></td>
<td>
<p>Number of observations</p>
</td></tr>
<tr><td><code>J</code></td>
<td>
<p>Maximum value of the dependent variable</p>
</td></tr>
<tr><td><code>nvar</code></td>
<td>
<p>Number of independent variables</p>
</td></tr>
<tr><td><code>fitted.values</code></td>
<td>
<p>Matrix with the fitted probabilities</p>
</td></tr>
<tr><td><code>pred</code></td>
<td>
<p>Predicted values for each item</p>
</td></tr>
<tr><td><code>Covariances</code></td>
<td>
<p>Covariances matrix</p>
</td></tr>
<tr><td><code>clasif</code></td>
<td>
<p>Matrix of classification of the items</p>
</td></tr>
<tr><td><code>PercentClasif</code></td>
<td>
<p>Percent of good classifications</p>
</td></tr>
<tr><td><code>coefficients</code></td>
<td>
<p> Estimated coefficients for the ordinal logistic regression</p>
</td></tr>
<tr><td><code>thresholds</code></td>
<td>
<p>Thresholds of the estimated model</p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>Logarithm of the likelihood</p>
</td></tr>
<tr><td><code>penalization</code></td>
<td>
<p>Penalization used to avoid singularities</p>
</td></tr>
<tr><td><code>Deviance</code></td>
<td>
<p>Deviance of the model</p>
</td></tr>
<tr><td><code>DevianceNull</code></td>
<td>
<p>Deviance of the null model</p>
</td></tr>
<tr><td><code>Dif</code></td>
<td>
<p>Diference between the two deviances values calculated</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>Degrees of freedom</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>p-value of the contrast</p>
</td></tr>
<tr><td><code>CoxSnell</code></td>
<td>
<p>Cox-Snell pseudo R squared</p>
</td></tr>
<tr><td><code>Nagelkerke</code></td>
<td>
<p>Nagelkerke pseudo R squared</p>
</td></tr>
<tr><td><code>MacFaden</code></td>
<td>
<p>Nagelkerke pseudo R squared</p>
</td></tr>
<tr><td><code>iter</code></td>
<td>
<p>Number of iterations made</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Albert,A. &amp; Anderson,J.A. (1984),<em>On the existence of maximum likelihood estimates in logistic regression models</em>, Biometrika 71(1), 1&ndash;10.
</p>
<p>Bull, S.B., Mak, C. &amp; Greenwood, C.M. (2002), <em>A modified score function for multinomial logistic regression</em>, Computational Statistics and dada Analysis 39, 57&ndash;74.
</p>
<p>Firth, D.(1993), <em>Bias reduction of maximum likelihood estimates</em>, Biometrika 80(1), 27&ndash;38
</p>
<p>Heinze, G. &amp; Schemper, M. (2002), <em>A solution to the problem of separation in logistic regression</em>, Statistics in Medicine 21, 2109&ndash;2419
</p>
<p>Le Cessie, S. &amp; Van Houwelingen, J. (1992), <em>Ridge estimators in logistic regression</em>, Applied Statistics 41(1), 191&ndash;201.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Doctors)
olb = OrdLogBipEM(Doctors,dim = 2, nnodos = 10,
            tol = 0.001, maxiter = 100, penalization = 0.2)
model = RidgeOrdinalLogistic(Doctors[, 1], olb$RowCoordinates, tol = 0.001,
        maxiter = 100, penalization = 0.2)
model
</code></pre>

<hr>
<h2 id='scores.CCA.sol'>
Extract the scores of a CCA solution object
</h2><span id='topic+scores.CCA.sol'></span>

<h3>Description</h3>

<p>Extract the scores of a CCA solution object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scores.CCA.sol(CCA.sol)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scores.CCA.sol_+3A_cca.sol">CCA.sol</code></td>
<td>

<p>The results of a CCA model
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Extract the scores of a CCA solution object
</p>


<h3>Value</h3>

<p>The species, sites and environmental variables scores of a CCA solution
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CCA">CCA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----

</code></pre>

<hr>
<h2 id='SeparateVarTypes'>
Separation of different types of variables into a list
</h2><span id='topic+SeparateVarTypes'></span>

<h3>Description</h3>

<p>The procedure creates a list in which each field contains the variables of the same type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SeparateVarTypes(X, TypeVar = NULL, TypeFit = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SeparateVarTypes_+3A_x">X</code></td>
<td>

<p>A data frame
</p>
</td></tr>
<tr><td><code id="SeparateVarTypes_+3A_typevar">TypeVar</code></td>
<td>

<p>A vector of characters defining the type of each variable. If not provided the procedure tries to gess the type of each variable. See details for types
</p>
</td></tr>
<tr><td><code id="SeparateVarTypes_+3A_typefit">TypeFit</code></td>
<td>

<p>A vector of characters defining the type of fit for each variable. If not provided the procedure tries to gess the type of fit for each variable. See details for types
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The procedure creates a list in which each field contains the variables of the same type.
The type of Variable can be specified in a vector TypeVar and the type of fit in a vector TypeFit.
The TypeVar is a vector of characters with as many components as variables with types coded as:
</p>
<p>&quot;c&quot; - Continuous (1)
</p>
<p>&quot;b&quot; - Binary (2)
</p>
<p>&quot;n&quot; - Nominal (3)
</p>
<p>&quot;o&quot; - Ordinal (4)
</p>
<p>&quot;f&quot; - Frequency (5)
</p>
<p>&quot;a&quot; - Abundance (5)
</p>
<p>Numbers rhather than characters can also be used.
Unless specified in TypeVar, numerical variables are &quot;Continuous&quot;, factors are &quot;Nominal&quot;, ordered factors are &quot;Ordinal&quot;. Factors with just two values 
are considered as &quot;Binary&quot;.
&quot;Frequencies&quot; and &quot;abundances&quot; should be specified by the user. If Typevar has length 1, all the variables are supposed to have the same type.
</p>
<p>The typeFit is a vector of characters containing the type of fit used for each variable, coded as:
</p>
<p>&quot;a&quot; - Average (1)
</p>
<p>&quot;wa&quot; - Weighted Average (2)
</p>
<p>&quot;r&quot; - Regression (Linear or logistic depending on the type of variable) (3)
</p>
<p>&quot;g&quot; - Gaussian (Equal tolerances) (4)
</p>
<p>&quot;g1&quot; - Gaussian (Different tolerances)  (5)
</p>
<p>Numbers rhather than characters can also be used. Unless specified numerical variables are fitted with linear regression, 
factors with logistic biplots, frequencies with weighted averages and abundances with gaussian regression. 
</p>


<h3>Value</h3>

<p>A list containing the following fields
</p>
<table>
<tr><td><code>Continuous</code></td>
<td>
<p>A list containing a data frame with the numeric variables and a character vector with the type of fit for each variable</p>
</td></tr>
<tr><td><code>Binary</code></td>
<td>
<p>A list containing a data frame with the binary variables and a character vector with the type of fit for each variable</p>
</td></tr>
<tr><td><code>Nominal</code></td>
<td>
<p>A list containing a data frame with the nominal variables and a character vector with the type of fit for each variable</p>
</td></tr>
<tr><td><code>Ordinal</code></td>
<td>
<p>A list containing a data frame with the ordinal variables and a character vector with the type of fit for each variable</p>
</td></tr>
<tr><td><code>Frequency</code></td>
<td>
<p>A list containing a data frame with the frequency variables and a character vector with the type of fit for each variable</p>
</td></tr>
<tr><td><code>Abundance</code></td>
<td>
<p>A list containing a data frame with the abundance variables and a character vector with the type of fit for each variable</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='SimpleProcrustes'>Simple Procrustes Analysis</h2><span id='topic+SimpleProcrustes'></span>

<h3>Description</h3>

<p>Simple Procrustes Analysis for two matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  SimpleProcrustes(X, Y, centre = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SimpleProcrustes_+3A_x">X</code></td>
<td>

<p>Matrix of the first configuration.
</p>
</td></tr>
<tr><td><code id="SimpleProcrustes_+3A_y">Y</code></td>
<td>

<p>Matrix of the second configuration. 
</p>
</td></tr>
<tr><td><code id="SimpleProcrustes_+3A_centre">centre</code></td>
<td>

<p>Should the matrices be centred before the calculations?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Orthogonal Procrustes Analysis for two configurations X and Y. 
The first configuration X is used as a reference and the second, 
Y, is transformed to match the reference as much as possible.
X = s Y T + 1t +E = Z + E
</p>


<h3>Value</h3>

<p>An object of class <code>Procrustes</code>.This has components:
</p>
<table>
<tr><td><code>X</code></td>
<td>
<p>First Configuration</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Second Configuration</p>
</td></tr>
<tr><td><code>Yrot</code></td>
<td>
<p>Second Configuration after the transformation</p>
</td></tr>
<tr><td><code>T</code></td>
<td>
<p>Rotation Matrix</p>
</td></tr>
<tr><td><code>t</code></td>
<td>
<p>Translation Vector</p>
</td></tr>
<tr><td><code>s</code></td>
<td>
<p>Scale Factor</p>
</td></tr>
<tr><td><code>rsss</code></td>
<td>
<p>Residual Sum of Squares</p>
</td></tr>
<tr><td><code>fit</code></td>
<td>
<p>Goodness of fit as percent of expained variance</p>
</td></tr>
<tr><td><code>correlations</code></td>
<td>
<p>Correlations among the columns of X and Z</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Ingwer Borg, I. &amp; Groenen, P. J.F. (2005). Modern Multidimensional Scaling. Theory and Applications. Second Edition. Springer
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PrincipalCoordinates">PrincipalCoordinates</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
</code></pre>

<hr>
<h2 id='SMACOF'>
SMACOF 
</h2><span id='topic+SMACOF'></span>

<h3>Description</h3>

<p>SMACOF algorithm for symmetric proximity matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SMACOF(P, X = NULL, W = NULL, 
Model = c("Identity", "Ratio", "Interval", "Ordinal"), 
dimsol = 2, maxiter = 100, maxerror = 1e-06, 
StandardizeDisparities = TRUE, ShowIter = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SMACOF_+3A_p">P</code></td>
<td>

<p>A matrix of proximities
</p>
</td></tr>
<tr><td><code id="SMACOF_+3A_x">X</code></td>
<td>

<p>Inial configuration
</p>
</td></tr>
<tr><td><code id="SMACOF_+3A_w">W</code></td>
<td>

<p>A matrix of weights~
</p>
</td></tr>
<tr><td><code id="SMACOF_+3A_model">Model</code></td>
<td>

<p>MDS model.
</p>
</td></tr>
<tr><td><code id="SMACOF_+3A_dimsol">dimsol</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="SMACOF_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations of the algorithm
</p>
</td></tr>
<tr><td><code id="SMACOF_+3A_maxerror">maxerror</code></td>
<td>

<p>Tolerance for convergence of the algorithm
</p>
</td></tr>
<tr><td><code id="SMACOF_+3A_standardizedisparities">StandardizeDisparities</code></td>
<td>

<p>Should the disparities be standardized
</p>
</td></tr>
<tr><td><code id="SMACOF_+3A_showiter">ShowIter</code></td>
<td>

<p>Show the iteration proccess
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>SMACOF performs multidimensional scaling of proximity data to find a least- squares representation of the objects in a low-dimensional space.  A majorization algorithm guarantees monotone convergence for optionally transformed, metric and nonmetric data under a variety of models.
</p>


<h3>Value</h3>

<p>An object of class <code>Principal.Coordinates</code> and  <code>MDS</code>. The function adds the information of the MDS to the object of class  <code>proximities</code>. Together with the information about the proximities the object has:
</p>
<table>
<tr><td><code>Analysis</code></td>
<td>
<p>The type of analysis performed, &quot;MDS&quot; in this case</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>Coordinates for the objects</p>
</td></tr>
<tr><td><code>D</code></td>
<td>
<p>Distances</p>
</td></tr>
<tr><td><code>Dh</code></td>
<td>
<p>Disparities</p>
</td></tr>
<tr><td><code>stress</code></td>
<td>
<p>Raw Stress</p>
</td></tr>
<tr><td><code>stress1</code></td>
<td>
<p>stress formula 1</p>
</td></tr>
<tr><td><code>stress2</code></td>
<td>
<p>stress formula 2</p>
</td></tr>
<tr><td><code>sstress1</code></td>
<td>
<p>sstress formula 1</p>
</td></tr>
<tr><td><code>sstress2</code></td>
<td>
<p>sstress formula 2</p>
</td></tr>
<tr><td><code>rsq</code></td>
<td>
<p>Squared correlation between disparities and distances</p>
</td></tr>
<tr><td><code>rho</code></td>
<td>
<p>Spearman correlation between disparities and distances</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>Kendall correlation between disparities and distances</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Commandeur, J. J. F. and Heiser, W. J. (1993). Mathematical derivations in the proximity scaling (PROXSCAL) of symmetric data matrices (Tech. Rep. No. RR- 93-03). Leiden, The Netherlands: Department of Data Theory, Leiden University.
</p>
<p>Kruskal, J. B. (1964). Nonmetric multidimensional scaling: A numerical method. Psychometrika, 29, 28-42.
</p>
<p>De Leeuw, J. &amp; Mair, P. (2009). Multidimensional scaling using majorization: The R package smacof. Journal of Statistical Software, 31(3), 1-30, http://www.jstatsoft.org/v31/i03/
</p>
<p>Borg, I., &amp; Groenen, P. J. F. (2005). Modern Multidimensional Scaling (2nd ed.). Springer.
</p>
<p>Borg, I., Groenen, P. J. F., &amp; Mair, P. (2013). Applied Multidimensional Scaling. Springer.
</p>
<p>Groenen, P. J. F., Heiser, W. J. and Meulman, J. J. (1999). Global optimization in least squares multidimensional scaling by distance smoothing. Journal of Classification, 16, 225-254.
</p>
<p>Groenen, P. J. F., van Os, B. and Meulman, J. J. (2000). Optimal scaling by alternating length-constained nonnegative least squares, with application to distance-based analysis. Psychometrika, 65, 511-524.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MDS">MDS</a></code>, <code><a href="#topic+PrincipalCoordinates">PrincipalCoordinates</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
Dis=BinaryProximities(spiders)
MDSSol=SMACOF(Dis$Proximities)
</code></pre>

<hr>
<h2 id='smoking'>
Smoking habits
</h2><span id='topic+smoking'></span>

<h3>Description</h3>

<p>Frequency table representing smoking habits of different employees in a company 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(smoking)</code></pre>


<h3>Format</h3>

<p>A data frame with 5 observations on the following 4 variables.
</p>

<dl>
<dt><code>None</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Light</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Medium</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Heavy</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Frequency table representing smoking habits of different employees in a company 
</p>


<h3>Source</h3>

<p>http://orange.biolab.si/docs/latest/reference/rst/Orange.projection.correspondence/
</p>


<h3>References</h3>

<p>Greenacre, Michael (1983). Theory and Applications of Correspondence Analysis. London: Academic Press.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(smoking)
## maybe str(smoking) ; plot(smoking) ...
</code></pre>

<hr>
<h2 id='Sparse.NIPALSPCA'>
Sparse version of the NIPALS algorithm for PCA.
</h2><span id='topic+Sparse.NIPALSPCA'></span>

<h3>Description</h3>

<p>Sparse version of the NIPALS algorithm for PCA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sparse.NIPALSPCA(X, dimens = 2, tol = 1e-06, maxiter = 1000, lambda = 0.02)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Sparse.NIPALSPCA_+3A_x">X</code></td>
<td>

<p>The data matrix.
</p>
</td></tr>
<tr><td><code id="Sparse.NIPALSPCA_+3A_dimens">dimens</code></td>
<td>

<p>The dimension of the solution
</p>
</td></tr>
<tr><td><code id="Sparse.NIPALSPCA_+3A_tol">tol</code></td>
<td>

<p>Tolerance of the algorithm.
</p>
</td></tr>
<tr><td><code id="Sparse.NIPALSPCA_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iteratios.
</p>
</td></tr>
<tr><td><code id="Sparse.NIPALSPCA_+3A_lambda">lambda</code></td>
<td>

<p>Value used for sparsity
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sparse version of the NIPALS algorithm for the singular value decomposition that allows for the construction of PCA and Biplot.
</p>


<h3>Value</h3>

<p>The singular value decomposition
</p>
<table>
<tr><td><code>u</code></td>
<td>
<p>The coordinates of the rows (standardized)</p>
</td></tr>
<tr><td><code>d</code></td>
<td>
<p>The singuklar values</p>
</td></tr>
<tr><td><code>v</code></td>
<td>
<p>The coordinates of the columns (standardized)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Have to be written
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='spiders'>
Hunting Spiders Data 
</h2><span id='topic+spiders'></span>

<h3>Description</h3>

<p>Hunting spiders data transformed into Presence/Abscense.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(spiders)</code></pre>


<h3>Format</h3>

<p>A data frame with 28 observations of presence/absence of 12 hunting spider species
</p>

<dl>
<dt>Alopacce</dt><dd><p>Presence/Absence of the species Alopecosa accentuata</p>
</dd>
<dt>Alopcune</dt><dd><p>Presence/Absence of the species Alopecosa cuneata</p>
</dd>
<dt>Alopfabr</dt><dd><p>Presence/Absence of the species Alopecosa fabrilis</p>
</dd>
<dt>Arctlute</dt><dd><p>Presence/Absence of the species Arctosa lutetiana</p>
</dd>
<dt>Arctperi</dt><dd><p>Presence/Absence of the species Arctosa perita</p>
</dd>
<dt>Auloalbi</dt><dd><p>Presence/Absence of the species Aulonia albimana</p>
</dd>
<dt>Pardlugu</dt><dd><p>Presence/Absence of the species Pardosa lugubris</p>
</dd>
<dt>Pardmont</dt><dd><p>Presence/Absence of the species Pardosa monticola</p>
</dd>
<dt>Pardnigr</dt><dd><p>Presence/Absence of the species Pardosa nigriceps</p>
</dd>
<dt>Pardpull</dt><dd><p>Presence/Absence of the species Pardosa pullata</p>
</dd>
<dt>Trocterr</dt><dd><p>Presence/Absence of the species Trochosa terricola</p>
</dd>
<dt>Zoraspin</dt><dd><p>Presence/Absence of the species Zora spinimana</p>
</dd>
</dl>



<h3>Source</h3>

<p>van der Aart, P. J. M., and Smeenk-Enserink, N. (1975) Correlations between distributions of hunting spiders (Lycos- idae, Ctenidae) and environmental characteristics in a dune area. Netherlands Journal of Zoology 25, 1-45.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
</code></pre>

<hr>
<h2 id='SpidersEnv'>
Hunting spiders environmental data.
</h2><span id='topic+SpidersEnv'></span>

<h3>Description</h3>

<p>Hunting spiders environmental data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("SpidersEnv")</code></pre>


<h3>Format</h3>

<p>A data frame with 28 observations on the following 6 variables.
</p>

<dl>
<dt><code>Watcont</code></dt><dd><p>Water content</p>
</dd>
<dt><code>Barsand</code></dt><dd><p>Bare sand</p>
</dd>
<dt><code>Covmoss</code></dt><dd><p>Cover moss</p>
</dd>
<dt><code>Ligrefl</code></dt><dd><p>Light reflection</p>
</dd>
<dt><code>Falltwi</code></dt><dd><p>Fallen Twings</p>
</dd>
<dt><code>Coverher</code></dt><dd><p>Cover Herbs</p>
</dd>
</dl>



<h3>Details</h3>

<p>Hunting spiders environmental data.
</p>


<h3>Source</h3>

<p>van der Aart, P. J. M., and Smeenk-Enserink, N. (1975) Correlations between distributions of hunting spiders (Lycos- idae, Ctenidae) and environmental characteristics in a dune area. Netherlands Journal of Zoology 25, 1-45.
</p>


<h3>References</h3>

<p>Ter Braak, C. J. (1986). Canonical correspondence analysis: a new eigenvector technique for multivariate direct gradient analysis. Ecology, 67(5), 1167-1179.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SpidersEnv)
## maybe str(SpidersEnv) ; plot(SpidersEnv) ...
</code></pre>

<hr>
<h2 id='SpidersSp'>
Hunting Spiders Data 
</h2><span id='topic+SpidersSp'></span>

<h3>Description</h3>

<p>Hunting spiders abundances data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("SpidersSp")</code></pre>


<h3>Format</h3>

<p>A data frame with 28 observations of abundance of 12 hunting spider species
</p>

<dl>
<dt>Alopacce</dt><dd><p>Abundance of the species Alopecosa accentuata</p>
</dd>
<dt>Alopcune</dt><dd><p>Abundance of the species Alopecosa cuneata</p>
</dd>
<dt>Alopfabr</dt><dd><p>Abundance of the species Alopecosa fabrilis</p>
</dd>
<dt>Arctlute</dt><dd><p>Abundance of the species Arctosa lutetiana</p>
</dd>
<dt>Arctperi</dt><dd><p>Abundance of the species Arctosa perita</p>
</dd>
<dt>Auloalbi</dt><dd><p>Abundance of the species Aulonia albimana</p>
</dd>
<dt>Pardlugu</dt><dd><p>Abundance of the species Pardosa lugubris</p>
</dd>
<dt>Pardmont</dt><dd><p>Abundance of the species Pardosa monticola</p>
</dd>
<dt>Pardnigr</dt><dd><p>Abundance of the species Pardosa nigriceps</p>
</dd>
<dt>Pardpull</dt><dd><p>Abundance of the species Pardosa pullata</p>
</dd>
<dt>Trocterr</dt><dd><p>Abundance of the species Trochosa terricola</p>
</dd>
<dt>Zoraspin</dt><dd><p>Abundance of the species Zora spinimana</p>
</dd>
</dl>



<h3>Source</h3>

<p>van der Aart, P. J. M., and Smeenk-Enserink, N. (1975) Correlations between distributions of hunting spiders (Lycos- idae, Ctenidae) and environmental characteristics in a dune area. Netherlands Journal of Zoology 25, 1-45.
</p>


<h3>References</h3>

<p>Ter Braak, C. J. (1986). Canonical correspondence analysis: a new eigenvector technique for multivariate direct gradient analysis. Ecology, 67(5), 1167-1179.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SpidersSp)
## maybe str(SpidersSp) ; plot(SpidersSp) ...
</code></pre>

<hr>
<h2 id='SSI'>
Sustainability Society Index
</h2><span id='topic+SSI'></span>

<h3>Description</h3>

<p>Sustainability Society Index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("SSI")</code></pre>


<h3>Format</h3>

<p>A data frame with 924 observations on the following 23 variables.
</p>

<dl>
<dt><code>Year</code></dt><dd><p>a factor with levels <code>a2006</code> <code>a2008</code> <code>a2010</code> <code>a2012</code> <code>a2014</code> <code>a2016</code></p>
</dd>
<dt><code>Country</code></dt><dd><p>a factor with levels <code>Albania</code> <code>Algeria</code> 
<code>Angola</code> <code>Argentina</code> <code>Armenia</code> <code>Australia</code> <code>Austria</code>
<code>Azerbaijan</code> <code>Bangladesh</code> <code>Belarus</code> <code>Belgium</code> <code>Benin</code> 
<code>Bhutan</code> <code>Bolivia</code> <code>Bosnia-Herzegovina</code> <code>Botswana</code> <code>Brazil</code>
<code>Bulgaria</code> <code>Burkina_Faso</code> <code>Burundi</code> <code>Cambodia</code> <code>Cameroon</code> 
<code>Canada</code> <code>Central_African_Republic</code> <code>Chad</code> <code>Chile</code> 
<code>China</code> <code>Colombia</code> <code>Congo</code> <code>Congo_Democratic_Rep.</code> 
<code>Costa_Rica</code> <code>Cote_dIvoire</code> <code>Croatia</code> <code>Cuba</code>
<code>Cyprus</code> <code>Czech_Republic</code> <code>Denmark</code> <code>Dominican_Republic</code> 
<code>Ecuador</code> <code>Egypt</code> <code>El_Salvador</code> <code>Estonia</code> <code>Ethiopia</code> 
<code>Finland</code> <code>France</code> <code>Gabon</code> <code>Gambia</code> <code>Georgia</code> 
<code>Germany</code> <code>Ghana</code> <code>Greece</code> <code>Guatemala</code> <code>Guinea</code> 
<code>Guinea-Bissau</code> <code>Guyana</code> <code>Haiti</code> <code>Honduras</code> 
<code>Hungary</code> <code>Iceland</code> <code>India</code> <code>Indonesia</code> <code>Iran</code> 
<code>Iraq</code> <code>Ireland</code> <code>Israel</code> <code>Italy</code> <code>Jamaica</code> 
<code>Japan</code> <code>Jordan</code> <code>Kazakhstan</code> <code>Kenya</code> <code>Korea._North</code> 
<code>Korea._South</code> <code>Kuwait</code> <code>Kyrgyz_Republic</code> <code>Laos</code> 
<code>Latvia</code> <code>Lebanon</code> <code>Lesotho</code> <code>Liberia</code> <code>Libya</code> 
<code>Lithuania</code> <code>Luxembourg</code> <code>Macedonia</code> <code>Madagascar</code> 
<code>Malawi</code> <code>Malaysia</code> <code>Mali</code> <code>Malta</code> <code>Mauritania</code> 
<code>Mauritius</code> <code>Mexico</code> <code>Moldova</code> <code>Mongolia</code> <code>Montenegro</code> 
<code>Morocco</code> <code>Mozambique</code> <code>Myanmar</code> <code>Namibia</code> <code>Nepal</code> 
<code>Netherlands</code> <code>New_Zealand</code> <code>Nicaragua</code> <code>Niger</code> <code>Nigeria</code> 
<code>Norway</code> <code>Oman</code> <code>Pakistan</code> <code>Panama</code> <code>Papua_New_Guinea</code> 
<code>Paraguay</code> <code>Peru</code> <code>Philippines</code> <code>Poland</code> <code>Portugal</code> 
<code>Qatar</code> <code>Romania</code> <code>Russia</code> <code>Rwanda</code> <code>Saudi_Arabia</code> 
<code>Senegal</code> <code>Serbia</code> <code>Sierra_Leone</code> <code>Singapore</code> 
<code>Slovak_Republic</code> <code>Slovenia</code> <code>South_Africa</code> <code>Spain</code> 
<code>Sri_Lanka</code> <code>Sudan</code> <code>Sweden</code> <code>Switzerland</code> <code>Syria</code> 
<code>Taiwan</code> <code>Tajikistan</code> <code>Tanzania</code> <code>Thailand</code> <code>Togo</code> <code>Trinidad_and_Tobago</code> <code>Tunisia</code> <code>Turkey</code> <code>Turkmenistan</code> 
<code>Uganda</code> <code>Ukraine</code> <code>United_Arab_Emirates</code> <code>United_Kingdom</code> <code>United_States</code> <code>Uruguay</code> <code>Uzbekistan</code> <code>Venezuela</code> <code>Vietnam</code> <code>Yemen</code> <code>Zambia</code> <code>Zimbabwe</code></p>
</dd>
<dt><code>Sufficient_Food</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Sufficient_to_Drink</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Safe_Sanitation</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Education_</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Healthy_Life</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Gender_Equality</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Income_Distribution</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Population_Growth</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Good_Governance</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Biodiversity_</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Renewable_Water_Resources</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Consumption</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Energy_Use</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Energy_Savings</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Greenhouse_Gases</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Renewable_Energy</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Organic_Farming</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Genuine_Savings</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GDP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Employment</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Public_Debt</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Sustainability Society Index
</p>


<h3>Source</h3>

<p>https://ssi.wi.th-koeln.de
</p>


<h3>References</h3>

<p>Gallego-Alvarez, I., Galindo-Villardon, M. P., &amp; Rodriguez-Rosa, M. (2015). Analysis of the Sustainable Society Index Worldwide: A Study from the Biplot Perspective. Social Indicators Research, 120(1), 29-65. https://doi.org/10.1007/s11205-014-0579-9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SSI)
## maybe str(SSI) ; plot(SSI) ...
</code></pre>

<hr>
<h2 id='SSI3w'>
Sustainability Society Index (3w)
</h2><span id='topic+SSI3w'></span>

<h3>Description</h3>

<p>Sustainability Society Index, Three way table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("SSI3w")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 6
$ a2006: num [1:154, 1:21] 10 9.3 6.6 10 8.9 10 10 10 8.3 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:21] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
$ a2008: num [1:154, 1:21] 10 9.4 7.1 10 9.3 10 10 10 8.3 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:21] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
$ a2010: num [1:154, 1:21] 10 9.4 7.7 10 9.4 10 10 10 8.3 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:21] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
$ a2012: num [1:154, 1:21] 10 10 8.1 10 9.3 10 10 10 8.3 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:21] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
$ a2014: num [1:154, 1:21] 10 10 8.4 10 9.3 10 10 10 8.3 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:21] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
$ a2016: num [1:154, 1:21] 10 10 8.6 10 9.4 10 10 10 8.4 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:21] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
</p>


<h3>Details</h3>

<p>Sustainability Society Index
</p>


<h3>Source</h3>

<p>https://ssi.wi.th-koeln.de
</p>


<h3>References</h3>

<p>Gallego-Alvarez, I., Galindo-Villardon, M. P., &amp; Rodriguez-Rosa, M. (2015). Analysis of the Sustainable Society Index Worldwide: A Study from the Biplot Perspective. Social Indicators Research, 120(1), 29-65. https://doi.org/10.1007/s11205-014-0579-9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SSI3w)
## maybe str(SSI3w) ; plot(SSI3w) ...
</code></pre>

<hr>
<h2 id='SSIEcon3w'>
Sustainability Society Index
</h2><span id='topic+SSIEcon3w'></span>

<h3>Description</h3>

<p>Sustainability Society Index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("SSIEcon3w")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 6
$ a2006: num [1:154, 1:5] 1.2 1 1 4.6 1 5.4 9.9 1.9 1 1 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:5] &quot;Organic_Farming&quot; &quot;Genuine_Savings&quot; &quot;GDP&quot; &quot;Employment&quot; ...
$ a2008: num [1:154, 1:5] 1 1 1 4.2 1 5.6 9.9 1.9 1 1 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:5] &quot;Organic_Farming&quot; &quot;Genuine_Savings&quot; &quot;GDP&quot; &quot;Employment&quot; ...
$ a2010: num [1:154, 1:5] 1.1 1 1 5.8 1.1 5.6 9.9 2 1 1 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:5] &quot;Organic_Farming&quot; &quot;Genuine_Savings&quot; &quot;GDP&quot; &quot;Employment&quot; ...
$ a2012: num [1:154, 1:5] 1.1 1 1 5.7 1.1 5.7 9.9 2 1 1 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:5] &quot;Organic_Farming&quot; &quot;Genuine_Savings&quot; &quot;GDP&quot; &quot;Employment&quot; ...
$ a2014: num [1:154, 1:5] 1.1 1 1 5.3 1.1 5.7 9.9 2.1 1.2 1 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:5] &quot;Organic_Farming&quot; &quot;Genuine_Savings&quot; &quot;GDP&quot; &quot;Employment&quot; ...
$ a2016: num [1:154, 1:5] 1.1 1 1 4.8 1.1 6.8 9.9 2 1.2 1 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:5] &quot;Organic_Farming&quot; &quot;Genuine_Savings&quot; &quot;GDP&quot; &quot;Employment&quot; ...
</p>


<h3>Details</h3>

<p>Sustainability Society Index
</p>


<h3>Source</h3>

<p>https://ssi.wi.th-koeln.de
</p>


<h3>References</h3>

<p>Gallego-Alvarez, I., Galindo-Villardon, M. P., &amp; Rodriguez-Rosa, M. (2015). Analysis of the Sustainable Society Index Worldwide: A Study from the Biplot Perspective. Social Indicators Research, 120(1), 29-65. https://doi.org/10.1007/s11205-014-0579-9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SSIEcon3w)
## maybe str(SSIEcon3w) ; plot(SSIEcon3w) ...
</code></pre>

<hr>
<h2 id='SSIEnvir3w'>
Sustainability Society Index
</h2><span id='topic+SSIEnvir3w'></span>

<h3>Description</h3>

<p>Sustainability Society Index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("SSIEnvir3w")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 6
$ a2006: num [1:154, 1:7] 4.2 6.5 4 4.9 7.7 5.7 8.1 4.9 2.8 6.3 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:7] &quot;Biodiversity_&quot; &quot;Renewable_Water_Resources&quot; &quot;Consumption&quot; &quot;Energy_Use&quot; ...
$ a2008: num [1:154, 1:7] 4.8 6.5 4 5.1 7.7 5.7 8 5.7 2.8 6 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:7] &quot;Biodiversity_&quot; &quot;Renewable_Water_Resources&quot; &quot;Consumption&quot; &quot;Energy_Use&quot; ...
$ a2010: num [1:154, 1:7] 5.4 6.6 4 5.2 7.7 5.7 8 6.4 2.8 5.8 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:7] &quot;Biodiversity_&quot; &quot;Renewable_Water_Resources&quot; &quot;Consumption&quot; &quot;Energy_Use&quot; ...
$ a2012: num [1:154, 1:7] 5.3 6.6 4 5.3 7.7 6.1 8 6.8 2.8 5.8 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:7] &quot;Biodiversity_&quot; &quot;Renewable_Water_Resources&quot; &quot;Consumption&quot; &quot;Energy_Use&quot; ...
$ a2014: num [1:154, 1:7] 5.6 6.6 4 5.3 7.7 7 7.9 7.3 2.8 6 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:7] &quot;Biodiversity_&quot; &quot;Renewable_Water_Resources&quot; &quot;Consumption&quot; &quot;Energy_Use&quot; ...
$ a2016: num [1:154, 1:7] 5.5 6.6 4.1 5.4 7.8 7.3 7.9 7.3 2.9 5.9 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:7] &quot;Biodiversity_&quot; &quot;Renewable_Water_Resources&quot; &quot;Consumption&quot; &quot;Energy_Use&quot; ...
</p>


<h3>Details</h3>

<p>Sustainability Society Index
</p>


<h3>Source</h3>

<p>https://ssi.wi.th-koeln.de
</p>


<h3>References</h3>

<p>Gallego-Alvarez, I., Galindo-Villardon, M. P., &amp; Rodriguez-Rosa, M. (2015). Analysis of the Sustainable Society Index Worldwide: A Study from the Biplot Perspective. Social Indicators Research, 120(1), 29-65. https://doi.org/10.1007/s11205-014-0579-9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SSIEnvir3w)
## maybe str(SSIEnvir3w) ; plot(SSIEnvir3w) ...
</code></pre>

<hr>
<h2 id='SSIHuman3w'>
Sustainability Society Index
</h2><span id='topic+SSIHuman3w'></span>

<h3>Description</h3>

<p>Sustainability Society Index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("SSIHuman3w")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 6
$ a2006: num [1:154, 1:9] 10 9.3 6.6 10 8.9 10 10 10 8.3 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:9] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
$ a2008: num [1:154, 1:9] 10 9.4 7.1 10 9.3 10 10 10 8.3 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:9] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
$ a2010: num [1:154, 1:9] 10 9.4 7.7 10 9.4 10 10 10 8.3 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:9] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
$ a2012: num [1:154, 1:9] 10 10 8.1 10 9.3 10 10 10 8.3 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:9] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
$ a2014: num [1:154, 1:9] 10 10 8.4 10 9.3 10 10 10 8.3 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:9] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
$ a2016: num [1:154, 1:9] 10 10 8.6 10 9.4 10 10 10 8.4 10 ...
..- attr(*, &quot;dimnames&quot;)=List of 2
.. ..$ : chr [1:154] &quot;Albania&quot; &quot;Algeria&quot; &quot;Angola&quot; &quot;Argentina&quot; ...
.. ..$ : chr [1:9] &quot;Sufficient_Food&quot; &quot;Sufficient_to_Drink&quot; &quot;Safe_Sanitation&quot; &quot;Education_&quot; ...
</p>


<h3>Details</h3>

<p>Sustainability Society Index
</p>


<h3>Source</h3>

<p>https://ssi.wi.th-koeln.de
</p>


<h3>References</h3>

<p>Gallego-Alvarez, I., Galindo-Villardon, M. P., &amp; Rodriguez-Rosa, M. (2015). Analysis of the Sustainable Society Index Worldwide: A Study from the Biplot Perspective. Social Indicators Research, 120(1), 29-65. https://doi.org/10.1007/s11205-014-0579-9
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SSIHuman3w)
## maybe str(SSIHuman3w) ; plot(SSIHuman3w) ...
</code></pre>

<hr>
<h2 id='StatisBiplot'>
STATIS-ACT for multiple tables with common rows and its associated Biplot
</h2><span id='topic+StatisBiplot'></span>

<h3>Description</h3>

<p>The procedure performs STATIS-ACT methodology for multiple tables with common rows and its associated biplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>StatisBiplot(X, InitTransform = "Standardize columns", dimens = 2,
                 SameVar = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="StatisBiplot_+3A_x">X</code></td>
<td>

<p>A list containing multiple tables with common rows.
</p>
</td></tr>
<tr><td><code id="StatisBiplot_+3A_inittransform">InitTransform</code></td>
<td>

<p>Initial transformation of the data matrices
</p>
</td></tr>
<tr><td><code id="StatisBiplot_+3A_dimens">dimens</code></td>
<td>

<p>Dimension of the final solution
</p>
</td></tr>
<tr><td><code id="StatisBiplot_+3A_samevar">SameVar</code></td>
<td>

<p>Are the variables the same for all occasions? If so, Biplot trajectories for each variable will be calculated.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The procedure performs STATIS-ACT methodology for multiple tables with common rows and its associated biplot. When the variables are the same for all occasions trajectories for the variables can also be plotted.
Basic plotting includes the consensus individuals and all the variables. Traditional trajectories for individuals and biplot trajectories for variables (when adequate) are optional.
The original matrix will be provided as a list each cell of the list is the data matrix for one ocassion
the number of rows for each occasion must be the same
</p>


<h3>Value</h3>

<p>An object of class StatisBiplot
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Abdi, H., Williams, L.J., Valentin, D., &amp; Bennani-Dosse, M. (2012). STATIS and DISTATIS: optimum multitable principal component analysis and three way metric multidimensional scaling. WIREs Comput Stat, 4, 124-167.
</p>
<p>Efron, B.,Tibshirani, RJ. (1993). An introduction to the bootstrap. New York: Chapman and Hall. 436p.
</p>
<p>Escoufier, Y. (1976). Operateur associe a un tableau de donnees. Annales de laInsee, 22-23, 165-178.
</p>
<p>Escoufier, Y. (1987). The duality diagram: a means for better practical applications. En P. Legendre &amp; L. Legendre (Eds.), Developments in Numerical Ecology, pp. 139-156, NATO Advanced Institute, Serie G. Berlin: Springer.
</p>
<p>L'Hermier des Plantes, H. (1976). Structuration des Tableaux a Trois Indices de la Statistique. [These de Troisieme Cycle]. University of Montpellier, France.
</p>
<p>Ringrose, T.J. (1992). Bootstrapping and Correspondence Analysis in Archaeology. Journal of Archaeological. Science.19:615-629.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Chemical)
# Extract continous data from the original data frame.
x= Chemical[,5:16]
# Obtaining the three way table as a list
X=Convert2ThreeWay(x,Chemical$WEEKS, columns=FALSE)
# Calculating the Biplot associated to STATIS-ACT
stbip=StatisBiplot(X, SameVar=TRUE)
# Basic plot of the results
plot(stbip)
# Colors By Table
plot(stbip, VarColorType="ByTable")
# Colors By Variable
plot(stbip, VarColorType="ByVar", mode="s",  MinQualityVars = 0.5)

plot(stbip, PlotRowTraj = TRUE, PlotVars=FALSE, RowColors=1:36)
</code></pre>

<hr>
<h2 id='summary.Canonical.Biplot'>
Summary of the solution of a Canonical Biplot Analysis
</h2><span id='topic+summary.Canonical.Biplot'></span>

<h3>Description</h3>

<p>Summary of the solution of a Canonical Biplot Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Canonical.Biplot'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.Canonical.Biplot_+3A_object">object</code></td>
<td>

<p>The result of a Canonical Biplot
</p>
</td></tr>
<tr><td><code id="summary.Canonical.Biplot_+3A_...">...</code></td>
<td>

<p>Aditional arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summary of the results of a Canonical Biplot
</p>


<h3>Value</h3>

<p>The summary
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
</code></pre>

<hr>
<h2 id='summary.CCA.sol'>
Summary of the solution of a CCA 
</h2><span id='topic+summary.CCA.sol'></span>

<h3>Description</h3>

<p>Summary of the solution of a CCA 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CCA.sol'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.CCA.sol_+3A_object">object</code></td>
<td>

<p>An object of class CCA.sol
</p>
</td></tr>
<tr><td><code id="summary.CCA.sol_+3A_...">...</code></td>
<td>

<p>Aditional arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summary of the solution of a CCA 
</p>


<h3>Value</h3>

<p>The main results of a CCA
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CCA">CCA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
</code></pre>

<hr>
<h2 id='summary.ContinuousBiplot'>
Summary of the solution of a Biplot for Continuous Data
</h2><span id='topic+summary.ContinuousBiplot'></span>

<h3>Description</h3>

<p>Summary of the solution of a Biplot for Continuous Data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ContinuousBiplot'
summary(object, latex = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.ContinuousBiplot_+3A_object">object</code></td>
<td>

<p>An object of class &quot;ContinuousBiplot&quot;
</p>
</td></tr>
<tr><td><code id="summary.ContinuousBiplot_+3A_latex">latex</code></td>
<td>

<p>Should the results be in latex tables
</p>
</td></tr>
<tr><td><code id="summary.ContinuousBiplot_+3A_...">...</code></td>
<td>

<p>Any aditional parameters
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summary of the solution of a Biplot for Continuous Data
</p>


<h3>Value</h3>

<p>The summary
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Simple Biplot with arrows
data(Protein)
bip=PCA.Biplot(Protein[,3:11])
summary(bip)
</code></pre>

<hr>
<h2 id='summary.CVA'>
Summary of a Canonical Variate Analysis
</h2><span id='topic+summary.CVA'></span>

<h3>Description</h3>

<p>Summary of a Canonical Variate Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'CVA'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.CVA_+3A_object">object</code></td>
<td>

<p>An object of class CVA
</p>
</td></tr>
<tr><td><code id="summary.CVA_+3A_...">...</code></td>
<td>

<p>Any aditional arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summary of a Canonical Variate Analysis
</p>


<h3>Value</h3>

<p>The summary
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='summary.MGC'>
Summary of Model-Based Gaussian Clustering results
</h2><span id='topic+summary.MGC'></span>

<h3>Description</h3>

<p>Summarizes the results of Model-Based Gaussian Clustering algorithms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MGC'
summary(object, Centers = TRUE, Covariances = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.MGC_+3A_object">object</code></td>
<td>

<p>An object of class &quot;MGC&quot;
</p>
</td></tr>
<tr><td><code id="summary.MGC_+3A_centers">Centers</code></td>
<td>

<p>Should the Centers be shown
</p>
</td></tr>
<tr><td><code id="summary.MGC_+3A_covariances">Covariances</code></td>
<td>

<p>Should the Covariances be shown
</p>
</td></tr>
<tr><td><code id="summary.MGC_+3A_...">...</code></td>
<td>

<p>Any aditional Parameters
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summarizes the results of Model-Based Gaussian Clustering algorithms
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as

</code></pre>

<hr>
<h2 id='summary.PCA.Analysis'>
Summary of the results of a PCA.
</h2><span id='topic+summary.PCA.Analysis'></span>

<h3>Description</h3>

<p>Sumarizes the results of a PCA Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PCA.Analysis'
summary(object, latex = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.PCA.Analysis_+3A_object">object</code></td>
<td>

<p>The object with the results of s PCA Analysis.
</p>
</td></tr>
<tr><td><code id="summary.PCA.Analysis_+3A_latex">latex</code></td>
<td>

<p>Should return latex tables?
</p>
</td></tr>
<tr><td><code id="summary.PCA.Analysis_+3A_...">...</code></td>
<td>

<p>Aditional arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Sumarizes the results of a PCA Analysis, including latex tables for presentation.
</p>


<h3>Value</h3>

<p>A summary of the main results
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='summary.PCA.Bootstrap'>
Summary of a PCA.Bootstrap object
</h2><span id='topic+summary.PCA.Bootstrap'></span>

<h3>Description</h3>

<p>Summary of a PCA.Bootstrap object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PCA.Bootstrap'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.PCA.Bootstrap_+3A_object">object</code></td>
<td>

<p>An object of class PCA.Bootstrap 
</p>
</td></tr>
<tr><td><code id="summary.PCA.Bootstrap_+3A_...">...</code></td>
<td>

<p>Additional arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summary of a PCA.Bootstrap object
</p>


<h3>Value</h3>

<p>The summary
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>

<hr>
<h2 id='summary.PLSR'>
Summary of a PLSR object
</h2><span id='topic+summary.PLSR'></span>

<h3>Description</h3>

<p>Summary of a PLSR object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PLSR'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.PLSR_+3A_object">object</code></td>
<td>

<p>An object of class PLSR
</p>
</td></tr>
<tr><td><code id="summary.PLSR_+3A_...">...</code></td>
<td>

<p>Additional arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summary of a PLSR object
</p>


<h3>Value</h3>

<p>The summary of the object
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>

<hr>
<h2 id='summary.PLSR1Bin'>
Summary of PLSR with a Binary Response
</h2><span id='topic+summary.PLSR1Bin'></span>

<h3>Description</h3>

<p>Summary of PLSR with a single binary Response
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PLSR1Bin'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.PLSR1Bin_+3A_object">object</code></td>
<td>

<p>An object of class PLSR1Bin
</p>
</td></tr>
<tr><td><code id="summary.PLSR1Bin_+3A_...">...</code></td>
<td>

<p>Aditional arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summary of PLSR with a single binary Response
</p>


<h3>Value</h3>

<p>The summary
</p>


<h3>Author(s)</h3>

<p>Jose Luis Viecente Villlardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Not yet
</code></pre>

<hr>
<h2 id='summary.Principal.Coordinates'>Summary of the results of a Principal Coordinates Analysis</h2><span id='topic+summary.Principal.Coordinates'></span>

<h3>Description</h3>

<p>Summary of the results of a Principal Coordinates Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Principal.Coordinates'
summary(object, printdata=FALSE, printproximities=FALSE, 
printcoordinates=FALSE, printqualities=FALSE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.Principal.Coordinates_+3A_object">object</code></td>
<td>

<p>An object of Type <code>Principal.Coordinates</code>
</p>
</td></tr>
<tr><td><code id="summary.Principal.Coordinates_+3A_printdata">printdata</code></td>
<td>

<p>Should original data be printed. Default is FALSE
</p>
</td></tr>
<tr><td><code id="summary.Principal.Coordinates_+3A_printproximities">printproximities</code></td>
<td>

<p>Should proximities be printed. Default is FALSE
</p>
</td></tr>
<tr><td><code id="summary.Principal.Coordinates_+3A_printcoordinates">printcoordinates</code></td>
<td>

<p>Should proximities be printed. Default is FALSE
</p>
</td></tr>
<tr><td><code id="summary.Principal.Coordinates_+3A_printqualities">printqualities</code></td>
<td>

<p>Should qualoties of representation be printed. Default is FALSE
</p>
</td></tr>
<tr><td><code id="summary.Principal.Coordinates_+3A_...">...</code></td>
<td>

<p>Additional parameters to summary.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a method for the generic function summary() for class &quot;Principal.Coordinates&quot;. It can be invoked by calling summary(x) for an object x of the appropriate class.
</p>


<h3>Value</h3>

<p>The summary
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
dist=BinaryProximities(spiders)
pco=PrincipalCoordinates(dist)
summary(pco)
</code></pre>

<hr>
<h2 id='summary.RidgeBinaryLogistic'>
Summary of a Binary Logistic Regression with Ridge Penalization
</h2><span id='topic+summary.RidgeBinaryLogistic'></span>

<h3>Description</h3>

<p>Summarizes the results of a Binary Logistic Regression with Ridge Penalization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'RidgeBinaryLogistic'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.RidgeBinaryLogistic_+3A_object">object</code></td>
<td>

<p>The object with te results of the logistic regression.
</p>
</td></tr>
<tr><td><code id="summary.RidgeBinaryLogistic_+3A_...">...</code></td>
<td>

<p>Any other arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summarizes the results of a Binary Logistic Regression with Ridge Penalization.
</p>


<h3>Value</h3>

<p>The summary
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not Yet
</code></pre>

<hr>
<h2 id='summary.TetraDualStatis'>
Summary of the results of TetraDualStatis
</h2><span id='topic+summary.TetraDualStatis'></span>

<h3>Description</h3>

<p>Summary of the results of TetraDualStatis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'TetraDualStatis'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.TetraDualStatis_+3A_object">object</code></td>
<td>

<p>The result of a Tetra Dual Statis Analysis
</p>
</td></tr>
<tr><td><code id="summary.TetraDualStatis_+3A_...">...</code></td>
<td>

<p>aditional arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summarizes the results of TetradUalStatis
</p>


<h3>Value</h3>

<p>No value returned




</p>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, José Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'># No examples yet
</code></pre>

<hr>
<h2 id='t3pcovr'>
Tucker 3 Principal Covariates Regression
</h2><span id='topic+t3pcovr'></span>

<h3>Description</h3>

<p>Tucker 3 Principal Covariates Regression
</p>


<h3>Usage</h3>

<pre><code class='language-R'>t3pcovr(X, Y, I, J, K, L, r1 = 2, r2 = 2, r3 = 2, 
       conv = 1e-06, OriginalAlfa = 0.5, AlternativeLossF = 1, 
       nRuns = 100, StartSeed = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="t3pcovr_+3A_x">X</code></td>
<td>
<p>A two way data matrix with the predictors.</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_y">Y</code></td>
<td>
<p>A three way data matrix with the responses.</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_i">I</code></td>
<td>
<p>Number of elements of first mode of 3D/2D (the common mode: rows)</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_j">J</code></td>
<td>
<p> number of elements of second mode of 3D (columns 3D) </p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_k">K</code></td>
<td>
<p>number of elements of third mode of 3D (slabs)</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_l">L</code></td>
<td>
<p>number of elements of second mode of 2D (columns 2D)</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_r1">r1</code></td>
<td>
<p>Number of extracted components for the A-mode</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_r2">r2</code></td>
<td>
<p>Number of extracted components for the B-mode</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_r3">r3</code></td>
<td>
<p>Number of extracted components for the C-mode</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_conv">conv</code></td>
<td>
<p>value for convergence (tolerance value)</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_originalalfa">OriginalAlfa</code></td>
<td>
<p>(0-1): importance that degree reduction and prediction have in the analysis</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_alternativelossf">AlternativeLossF</code></td>
<td>
<p>Using the alternative loss function? 0 = no (use original loss function: weighted SSQ;               weighted met alfa) 1 = yes (use weighted loss function with scaled SSQ: scaled by the SSQ in X and y )</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_nruns">nRuns</code></td>
<td>
<p>Number of runs</p>
</td></tr>
<tr><td><code id="t3pcovr_+3A_startseed">StartSeed</code></td>
<td>
<p>Seed for the analysis</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In behavioral research it is very common to have to deal with  several data sets which include information relative to the same set of individuals, in such a way that one data set tries to explain the others. The class of models known as PCovR focuses on the analysis of a three-way data array explained by a two-way data matrix. In this paper the Tucker3-PCovR model is proposed that is a particular case of PCovR class. Tucker3-PCovR model reduces the predictors to a few components and predict the criterion by using these components and, at the same time, the three way data is fitted through the Tucker3 model. Both, the reduction of the predictors and the prediction of the criterion are done simultaneously. An alternating least squares algorithm to estimate the Tucker3-PCovR model is proposed. A biplot representation to facilitate the interpretation of the results is presented. A couple of applications are made to coupled empirical data sets related to the field of psychology. 
</p>


<h3>Value</h3>

<table>
<tr><td><code>A</code></td>
<td>
<p>Component matrix for the A-mode)</p>
</td></tr>
<tr><td><code>B1</code></td>
<td>
<p>Component matrix for the B-mode</p>
</td></tr>
<tr><td><code>C</code></td>
<td>
<p>Component matrix for the C-mode</p>
</td></tr>
<tr><td><code>H</code></td>
<td>
<p>Matrized core array (frontal slices)</p>
</td></tr>
<tr><td><code>B2</code></td>
<td>
<p>Loading matrix of components (components x predictors)</p>
</td></tr>
<tr><td><code>...</code></td>
<td>
<p>Further arguments</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Elisa Frutos Bernal (<a href="mailto:efb@usal.es">efb@usal.es</a>)
</p>


<h3>References</h3>

<p><cite>De Jong, S., &amp; Kiers, H. A. (1992). Principal covariates regression: Part I. Theory. Chemometrics and Intelligent Laboratory Systems , 155-164.</cite>
</p>
<p><cite>Marlies Vervloet, Henk A. Kiers, Wim Van den Noortgate, Eva Ceulemans (2015). PCovR: An R Package for Principal Covariates Regression. Journal of Statistical Software, 65(8), 1-14. URL http://www.jstatsoft.org/v65/i08/.</cite>
</p>
<p><cite>Smilde, A. K., Bro, R., &amp; Geladi, P. (2004). Multi-way analysis with applications in the chemical sciences. Chichester, UK: Wiley.</cite></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Not yet
</code></pre>

<hr>
<h2 id='TetraDualStatis'>
Dual STATIS-ACT for binary data based on Tetrachoric Correlations
</h2><span id='topic+TetraDualStatis'></span>

<h3>Description</h3>

<p>Dual STATIS-ACT for binary data based on Tetrachoric Correlations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TetraDualStatis(X, dimens = 2, SameInd = FALSE, RotVarimax = FALSE, 
               OptimMethod = "L-BFGS-B", penalization = 0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TetraDualStatis_+3A_x">X</code></td>
<td>

<p>A three way binary data matrix
</p>
</td></tr>
<tr><td><code id="TetraDualStatis_+3A_dimens">dimens</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="TetraDualStatis_+3A_sameind">SameInd</code></td>
<td>

<p>Are the individuals the same in all occassions?
</p>
</td></tr>
<tr><td><code id="TetraDualStatis_+3A_rotvarimax">RotVarimax</code></td>
<td>

<p>Should the solution be rotated?
</p>
</td></tr>
<tr><td><code id="TetraDualStatis_+3A_optimmethod">OptimMethod</code></td>
<td>

<p>Optimization method for the gradients
</p>
</td></tr>
<tr><td><code id="TetraDualStatis_+3A_penalization">penalization</code></td>
<td>

<p>Penalization for the ridge solution
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The general aim of STATIS-ACT methods is to extract information common to a set of datasets with the same individuals. They will also be represented as a Euclidean configuration or map of points (or vectors), in the same way as in Principal Component Analysis (PCA) or Principal Coordinate Analysis (PCoA). If the object is to analyze the variables and the correlation structures between them we will use a Factor Analysis (FA). When we have tables in which we measure a set of common variables and we want to obtain a consensus structure of all of them, we will use the named STATIS-Dual.
</p>
<p>The method was initially designed to work with individuals common to all the tables, but in this work, we will focus on the dual version, which works with variables common to all of them.
</p>
<p>When we have several tables of binary dataset, the classical methods for continuous data are not suitable. If the individuals are the same in all tables, we can use a STATIS based on distances, also known as DISTATIS. El procedimiento consiste en calcular una matriz de distancias a partir de para un coeficiente de similaridad para datos binarios. Las distancias se convierten en productos escalares, como en ACoP, y se trabaja a partir de ellos como en el STATIS tradicional.
</p>
<p>When we have common variables, and we are interested in the association between them, we could use a coefficient that, instead of similarity, shows the association between the variables. In this work we propose the use of the tetrachoric correlation matrix for each table and develop the necessary adaptations to the method.
</p>


<h3>Value</h3>

<p>An object with the results




</p>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, José Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='textsmart'>Labels of a Scatter</h2><span id='topic+textsmart'></span>

<h3>Description</h3>

<p>Plots labels of points in a scattergram. labels for points with positive <code>x</code> are placed on the right of 
the points, and labels for points with negative values on the left. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>textsmart(A, Labels, CexPoints, ColorPoints, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="textsmart_+3A_a">A</code></td>
<td>

<p>Coordinates of the points for the scaterrgram
</p>
</td></tr>
<tr><td><code id="textsmart_+3A_labels">Labels</code></td>
<td>

<p>Labels for the points
</p>
</td></tr>
<tr><td><code id="textsmart_+3A_cexpoints">CexPoints</code></td>
<td>

<p>Size of the labels
</p>
</td></tr>
<tr><td><code id="textsmart_+3A_colorpoints">ColorPoints</code></td>
<td>

<p>Colors of the labels
</p>
</td></tr>
<tr><td><code id="textsmart_+3A_...">...</code></td>
<td>

<p>Aditional graphical arguments
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is used to improve the readability of the labels in a scatergram.
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.Principal.Coordinates">plot.Principal.Coordinates</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
dist=BinaryProximities(spiders)
pco=PrincipalCoordinates(dist)
plot(pco, SmartLabels =TRUE)
</code></pre>

<hr>
<h2 id='Three2TwoWay'>
Converts a multitable list to a two way matrix 
</h2><span id='topic+Three2TwoWay'></span>

<h3>Description</h3>

<p>Takes a multitable list of matrices X and converts it to a two way matrix with
the structure required by the Statis programs using a _ to separate variable and occassion
or study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Three2TwoWay(X, whatlines = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Three2TwoWay_+3A_x">X</code></td>
<td>

<p>The multitable list.
</p>
</td></tr>
<tr><td><code id="Three2TwoWay_+3A_whatlines">whatlines</code></td>
<td>

<p>Concatenate the rows (1) or the columns (2)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes a multitable list of matrices X and converts it to a two way matrix with
the structure required by the Statis programs using a _ to separate variable and occassion
or study. When whatlines is 1 the final matrix adds the rows of the three dimensional array, then the columns must
be the same for all studies. When whatlines is 2 the columns are concatenated and then the number
of rows must be the same for all studies.
</p>


<h3>Value</h3>

<p>A two way matrix
</p>
<table>
<tr><td><code>x</code></td>
<td>
<p>A two way matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # No examples yet
</code></pre>

<hr>
<h2 id='ThreeWay2FrontalSlices'>
Three to two way data
</h2><span id='topic+ThreeWay2FrontalSlices'></span>

<h3>Description</h3>

<p>Three to two way data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ThreeWay2FrontalSlices(X, Slice = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ThreeWay2FrontalSlices_+3A_x">X</code></td>
<td>

<p>A three way array.
</p>
</td></tr>
<tr><td><code id="ThreeWay2FrontalSlices_+3A_slice">Slice</code></td>
<td>

<p>The mode for the rows
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Three to two way data. The provided mode is placen on the rows. The columns are the result of intercatively coding the other two modes.
</p>


<h3>Value</h3>

<p>A two way matrix.
</p>


<h3>Author(s)</h3>

<p>José Luis Vicente- Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----

</code></pre>

<hr>
<h2 id='TransformIni'>
Initial transformation of a data matrix
</h2><span id='topic+TransformIni'></span>

<h3>Description</h3>

<p>Initial transformation of data before the construction of a biplot. (or any other technique)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TransformIni(X, InitTransform = "None", transform = "Standardize columns")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TransformIni_+3A_x">X</code></td>
<td>

<p>Original Raw Data Matrix
</p>
</td></tr>
<tr><td><code id="TransformIni_+3A_inittransform">InitTransform</code></td>
<td>

<p>Initial transform of the data (usually logarithm)
</p>
</td></tr>
<tr><td><code id="TransformIni_+3A_transform">transform</code></td>
<td>

<p>Transformation to use. See details.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Possible Transformations are:
</p>
<p>1.- &quot;Raw Data&quot;: When no transformation is required.
</p>
<p>2.- &quot;Substract the global mean&quot;: Eliminate an eefect common to all the observations
</p>
<p>3.- &quot;Double centering&quot; : Interaction residuals. When all the elements of the table are comparable. Useful for AMMI models.
</p>
<p>4.- &quot;Column centering&quot;: Remove the column means.
</p>
<p>5.- &quot;Standardize columns&quot;:  Remove the column means and divide by its standard deviation.
</p>
<p>6.- &quot;Row centering&quot;: Remove the row means.
</p>
<p>7.- &quot;Standardize rows&quot;: Divide each row by its standard deviation.
</p>
<p>8.- &quot;Divide by the column means and center&quot;: The resulting dispersion is the coefficient of variation.
</p>
<p>9.- &quot;Normalized residuals from independence&quot; for a contingency table.
</p>
<p>The transformation can be provided to the function by using the string beetwen the quotes or just the associated number.
</p>
<p>The supplementary rows and columns are not used to calculate the parameters (means, standard deviations, etc). Some of the transformations are not compatible with supplementary data. 
</p>


<h3>Value</h3>

<table>
<tr><td><code>X</code></td>
<td>
<p>Transformed data matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>M. J. Baxter (1995) Standardization and Transformation in Principal Component Analysis, with Applications to Archaeometry. 
Journal of the Royal Statistical Society. Series C (Applied Statistics). Vol. 44, No. 4 (1995) , pp. 513-527
</p>
<p>Kroonenberg, P. M. (1983). Three-mode principal component analysis: Theory and applications (Vol. 2). DSWO press. (Chapter 6)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
x=as.matrix(iris[,1:4])
x=TransformIni(x, transform=4)
x
</code></pre>

<hr>
<h2 id='Truncated.NIPALSPCA'>
Truncated version of the NIPALS algorithm for PCA.
</h2><span id='topic+Truncated.NIPALSPCA'></span>

<h3>Description</h3>

<p>Truncated version of the NIPALS algorithm for PCA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Truncated.NIPALSPCA(X, dimens = 2, tol = 1e-06, maxiter = 1000, lambda = 0.02)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Truncated.NIPALSPCA_+3A_x">X</code></td>
<td>

<p>The data matrix.
</p>
</td></tr>
<tr><td><code id="Truncated.NIPALSPCA_+3A_dimens">dimens</code></td>
<td>

<p>The dimension of the solution
</p>
</td></tr>
<tr><td><code id="Truncated.NIPALSPCA_+3A_tol">tol</code></td>
<td>

<p>Tolerance of the algorithm.
</p>
</td></tr>
<tr><td><code id="Truncated.NIPALSPCA_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iteratios.
</p>
</td></tr>
<tr><td><code id="Truncated.NIPALSPCA_+3A_lambda">lambda</code></td>
<td>

<p>Value used for truncation
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Classical NIPALS algorithm for the singular value decomposition that allows for the construction of PCA and Biplot.
</p>


<h3>Value</h3>

<p>The singular value decomposition
</p>
<table>
<tr><td><code>u</code></td>
<td>
<p>The coordinates of the rows (standardized)</p>
</td></tr>
<tr><td><code>d</code></td>
<td>
<p>The singuklar values</p>
</td></tr>
<tr><td><code>v</code></td>
<td>
<p>The coordinates of the columns (standardized)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Have to be written
</p>


<h3>See Also</h3>

<p><code><a href="#topic+NIPALS.Biplot">NIPALS.Biplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Not yet
</code></pre>

<hr>
<h2 id='Unfolding'>
Multidimensional Unfolding
</h2><span id='topic+Unfolding'></span>

<h3>Description</h3>

<p>Multidimensional Unfolding with some adaptations for vegetation analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Unfolding(A, ENV = NULL, TransAbund = "Gaussian Columns", offset = 0.5, 
weight = "All_1", Constrained = FALSE, 
TransEnv = "Standardize columns", 
InitConfig = "SVD", model = "Ratio", 
condition = "Columns", Algorithm = "SMACOF", 
OptimMethod = "CG", r = 2, maxiter = 100, 
tolerance = 1e-05, lambda = 1, omega = 0, plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Unfolding_+3A_a">A</code></td>
<td>

<p>The original proximities matrix
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_env">ENV</code></td>
<td>

<p>The matrix of environmental variables
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_transabund">TransAbund</code></td>
<td>

<p>Initial transformation of the abundances : &quot;None&quot;, &quot;Gaussian&quot;, &quot;Column Percent&quot;, &quot;Gaussian Columns&quot;, &quot;Inverse Square Root&quot;, &quot;Divide by Column Maximum&quot;)
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_offset">offset</code></td>
<td>

<p>offset is the quantity added to the zeros of the table
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_weight">weight</code></td>
<td>

<p>A matrix of weights for each cell of the table
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_constrained">Constrained</code></td>
<td>

<p>Should fit a constrained analysis
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_transenv">TransEnv</code></td>
<td>

<p>Transformation of the environmental variables
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_initconfig">InitConfig</code></td>
<td>

<p>Init configuration for the algorithm
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_model">model</code></td>
<td>

<p>Type of model to be fitted: &quot;Identity&quot;, &quot;Ratio&quot;, &quot;Interval&quot; or &quot;Ordinal&quot;.
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_condition">condition</code></td>
<td>

<p>&quot;Matrix&quot;, &quot;Columns&quot; to condition to the whole matrix or to each column
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_algorithm">Algorithm</code></td>
<td>

<p>Algorithm to fit the model: &quot;SMACOF&quot;, &quot;GD&quot;, &quot;Genefold&quot;
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_optimmethod">OptimMethod</code></td>
<td>

<p>Optimization method for gradient descent
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_r">r</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations in the algorithm
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerace for the algorithm
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_lambda">lambda</code></td>
<td>

<p>First penalization parameter
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_omega">omega</code></td>
<td>

<p>Second penalization parameter
</p>
</td></tr>
<tr><td><code id="Unfolding_+3A_plot">plot</code></td>
<td>

<p>Should the results be plotted?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ological data</p>


<h3>Value</h3>

<p>An object of class &quot;Unfolding&quot;
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>References</h3>

<p>Ver Articulos
</p>


<h3>Examples</h3>

<pre><code class='language-R'>unf=Unfolding(SpidersSp, ENV=SpidersEnv, model="Ratio", Constrained = FALSE, condition="Matrix")
plot(unf, PlotTol=TRUE, PlotEnv = FALSE)
plot(unf, PlotTol=TRUE, PlotEnv = TRUE)
cbind(unf$QualityVars, unf$Var_Fit)
unf2=Unfolding(SpidersSp, ENV=SpidersEnv, model="Ratio", Constrained = TRUE, condition="Matrix")
plot(unf2, PlotTol=FALSE, PlotEnv = TRUE, mode="s")
cbind(unf2$QualityVars, unf2$Var_Fit)
</code></pre>

<hr>
<h2 id='VarBiplot'>
Draws a variable on a biplot
</h2><span id='topic+VarBiplot'></span>

<h3>Description</h3>

<p>Draws a continuous variable on a biplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VarBiplot(bi1, bi2, b0 = 0, xmin = -3, xmax = 3, ymin = -3, ymax
                 = 3, label = "Point", mode = "a", CexPoint = 0.8,
                 PchPoint = 1, Color = "blue", ticks = c(-3, -2.5, -2,
                 -1.5, -1, -0.5, 0.5, 1, 1.5, 2, 2.5, 3), ticklabels =
                 round(ticks, digits = 2), tl = 0.04, ts = "Complete",
                 Position = "Angle", AddArrow=FALSE, CexScale=0.8, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VarBiplot_+3A_bi1">bi1</code></td>
<td>

<p>First component of the direction vector
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_bi2">bi2</code></td>
<td>

<p>Second component of the direction vector
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_b0">b0</code></td>
<td>

<p>Constant for the regression adjusted biplots
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_xmin">xmin</code></td>
<td>

<p>Minimum value of the x axis
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_xmax">xmax</code></td>
<td>

<p>Maximum value of the x axis
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_ymin">ymin</code></td>
<td>

<p>Minimum value of the y axis
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_ymax">ymax</code></td>
<td>

<p>Maximum value of the y axis
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_label">label</code></td>
<td>

<p>Label of the variable
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_mode">mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;h&quot;, &quot;ah&quot; and &quot;s&quot;.
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_cexpoint">CexPoint</code></td>
<td>

<p>Size for the symbols and labels of the variables
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_pchpoint">PchPoint</code></td>
<td>

<p>Symbols for the variable (when represented as a point)
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_color">Color</code></td>
<td>

<p>Color for the variable
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_ticks">ticks</code></td>
<td>

<p>Ticks when the variable is represented as a graded scale
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_ticklabels">ticklabels</code></td>
<td>

<p>Labels for the ticks when the variable is represented as a graded scale
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_tl">tl</code></td>
<td>

<p>Thick length
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_ts">ts</code></td>
<td>

<p>Size of the mark in the gradedv  scale
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_position">Position</code></td>
<td>

<p>If the Position is &quot;Angle&quot; the label of the variable is placed using the angle of the vector
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_addarrow">AddArrow</code></td>
<td>

<p>Add an arrow to the representation of other modes of the biplot.
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_cexscale">CexScale</code></td>
<td>

<p>Sizes of the scales
</p>
</td></tr>
<tr><td><code id="VarBiplot_+3A_...">...</code></td>
<td>

<p>Any other graphical parameters 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See plot.PCA.Biplot
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.ContinuousBiplot">plot.ContinuousBiplot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Protein)
bip=PCA.Biplot(Protein[,3:11])
plot(bip)
</code></pre>

<hr>
<h2 id='wa'> 
Extracts the weighted averages of a CCA solution
</h2><span id='topic+wa'></span>

<h3>Description</h3>

<p>Extracts the weighted averages of a CCA solution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wa(CCA.sol, transformed = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wa_+3A_cca.sol">CCA.sol</code></td>
<td>

<p>The solution of a CCA
</p>
</td></tr>
<tr><td><code id="wa_+3A_transformed">transformed</code></td>
<td>

<p>Average of the transformed or the original data?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Extracts the weighted averages of a CCA solution
</p>


<h3>Value</h3>

<p>A matrix with the averages
</p>


<h3>Author(s)</h3>

<p>icente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
</code></pre>

<hr>
<h2 id='wcor'>
Weighted correlations
</h2><span id='topic+wcor'></span>

<h3>Description</h3>

<p>Weighted correlations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wcor(d1, d2, w = rep(1, nrow(d1))/nrow(d1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wcor_+3A_d1">d1</code></td>
<td>

<p>First Vector
</p>
</td></tr>
<tr><td><code id="wcor_+3A_d2">d2</code></td>
<td>

<p>Second vector to correlate
</p>
</td></tr>
<tr><td><code id="wcor_+3A_w">w</code></td>
<td>

<p>weights for ecah element of the vectors
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Weighted correlations
</p>


<h3>Value</h3>

<p>Weighted correlation
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>

<hr>
<h2 id='weighted.quantile'>
Weighted quantiles
</h2><span id='topic+weighted.quantile'></span>

<h3>Description</h3>

<p>Weighted quantiles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weighted.quantile(x, w, q = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="weighted.quantile_+3A_x">x</code></td>
<td>

<p>The numerical variable.
</p>
</td></tr>
<tr><td><code id="weighted.quantile_+3A_w">w</code></td>
<td>

<p>Weights
</p>
</td></tr>
<tr><td><code id="weighted.quantile_+3A_q">q</code></td>
<td>

<p>Quantile
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The quantile
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----

</code></pre>

<hr>
<h2 id='WeightedPCoA'>Weighted Principal Coordinates Analysis</h2><span id='topic+WeightedPCoA'></span>

<h3>Description</h3>

<p>Weighted Principal Coordinates Analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  WeightedPCoA(Proximities, 
  weigths = matrix(1,dim(Proximities$Proximities)[1],1), 
  dimension = 2, tolerance=0.0001)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WeightedPCoA_+3A_proximities">Proximities</code></td>
<td>

<p>A matrix containing the proximities among a set of objetcs
</p>
</td></tr>
<tr><td><code id="WeightedPCoA_+3A_weigths">weigths</code></td>
<td>

<p>Weigths
</p>
</td></tr>
<tr><td><code id="WeightedPCoA_+3A_dimension">dimension</code></td>
<td>

<p>Dimension of the solution
</p>
</td></tr>
<tr><td><code id="WeightedPCoA_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for the eigenvalues
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Weighted Principal Coordinates Analysis
</p>


<h3>Value</h3>

<p>data(spiders)
dist=BinaryProximities(spiders)
pco=WeightedPCoA(dist)
An object of class <code>Principal.Coordinates</code>
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Gower, J. C. (2006) Similarity dissimilarity and Distance, measures of. Encyclopedia of Statistical Sciences. 2nd. ed. Volume 12. Wiley
</p>
<p>Gower, J.C. (1966). Some distance properties of latent root and vector methods used in multivariate analysis. Biometrika 53: 325-338.
</p>
<p>J.R. Demey, J.L. Vicente-Villardon, M.P. Galindo, A.Y. Zambrano, Identifying molecular markers associated with classifications of genotypes by external logistic
biplot, Bioinformatics 24 (2008) 2832.
</p>
<p>Cuadras, C. M., Fortiana, J. Metric scaling graphical representation of Categorical Data. Proceedings of Statistics Day, The Center for Multivariate Analysis, Pennsylvania State University, Part 2, pp.1-27, 1995.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BinaryProximities">BinaryProximities</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
dist=BinaryProximities(spiders)
pco=WeightedPCoA(dist)

</code></pre>

<hr>
<h2 id='wine'>
Wine data
</h2><span id='topic+wine'></span>

<h3>Description</h3>

<p>Comparison of young wines of Ribera de Duero and Toro
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("wine")</code></pre>


<h3>Format</h3>

<p>A data frame with 45 observations on the following 21 variables.
</p>

<dl>
<dt><code>Year</code></dt><dd><p>A factor with levels <code>1986</code> <code>1987</code></p>
</dd>
<dt><code>Origin</code></dt><dd><p>A factor with levels <code>Ribera</code> <code>Toro</code></p>
</dd>
<dt><code>Group</code></dt><dd><p>A factor with levels <code>R86</code> <code>R87</code> <code>T86</code> <code>T87</code></p>
</dd>
<dt><code>A</code></dt><dd><p>Alcoholic content (percentage)</p>
</dd>
<dt><code>VA</code></dt><dd><p>volatil acidity - g acetic acid/l</p>
</dd>
<dt><code>TA</code></dt><dd><p>Total tritable acidity - g tartaric acid/l</p>
</dd>
<dt><code>FA</code></dt><dd><p>Fixed acidity -  g tartaric acid/l</p>
</dd>
<dt><code>pH</code></dt><dd><p>ph</p>
</dd>
<dt><code>TPR</code></dt><dd><p>Total phenolics - g gallic acid /l - Folin</p>
</dd>
<dt><code>TPS</code></dt><dd><p>Total phenolics - Somers </p>
</dd>
<dt><code>V</code></dt><dd><p>Substances reactive to vanilin - mg catechin/l</p>
</dd>
<dt><code>PC</code></dt><dd><p>Procyanidins - mg cyanidin/l</p>
</dd>
<dt><code>ACR</code></dt><dd><p>Total Anthocyanins - mg/l - method 1</p>
</dd>
<dt><code>ACS</code></dt><dd><p>Total Anthocyanins - mg/l - methods 2</p>
</dd>
<dt><code>ACC</code></dt><dd><p>Malvidin - malvidin-3-glucoside mg/l</p>
</dd>
<dt><code>CI</code></dt><dd><p>Color density - </p>
</dd>
<dt><code>CI2</code></dt><dd><p>Color density 2</p>
</dd>
<dt><code>H</code></dt><dd><p>Wine Hue Color</p>
</dd>
<dt><code>I</code></dt><dd><p>Degree of Ionization - Percent</p>
</dd>
<dt><code>CA</code></dt><dd><p>Chemical Age</p>
</dd>
<dt><code>VPC</code></dt><dd><p>ratio V/PC</p>
</dd>
</dl>



<h3>Details</h3>

<p>Comparison of young wines of Ribera de Duero and Toro
</p>


<h3>Source</h3>

<p>Rivas-Gonzalo, J. C., Gutierrez, Y., Polanco, A. M., Hebrero, E., Vicente-Villardon, J. L., Galindo, P., &amp; Santos-Buelga, C. (1993). Biplot analysis applied to enological parameters in the geographical classification of young red wines. American journal of enology and viticulture, 44(3), 302-308.
</p>


<h3>References</h3>

<p>Rivas-Gonzalo, J. C., Gutierrez, Y., Polanco, A. M., Hebrero, E., Vicente-Villardon, J. L., Galindo, P., &amp; Santos-Buelga, C. (1993). Biplot analysis applied to enological parameters in the geographical classification of young red wines. American journal of enology and viticulture, 44(3), 302-308.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
## maybe str(wine) ; plot(wine) ...
</code></pre>

<hr>
<h2 id='zeros'>
Matrix of zeros as in Matlab
</h2><span id='topic+zeros'></span>

<h3>Description</h3>

<p>Matrix of zeros
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zeros(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zeros_+3A_n">n</code></td>
<td>

<p>Dimension of the matrix
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of zeros
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>zeros(6)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
