<!DOCTYPE html><html lang="en"><head><title>Help for package multiDimBio</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {multiDimBio}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#multiDimBio-package'><p>Multivariate Analysis and Visualization for Biological Data</p></a></li>
<li><a href='#binomPower'>
<p>Power analysis for estimating the heritability of a binomial trait</p></a></li>
<li><a href='#boxWhisker'><p>A function to create a box and whisker plot by group ID</p></a></li>
<li><a href='#completeData'>
<p>Function to impute missing data.</p></a></li>
<li><a href='#CondA'><p>Treatment condition for animals contained in the data set Nuclei</p></a></li>
<li><a href='#CondB'><p>Stress condition for animals contained in the data set Nuclei</p></a></li>
<li><a href='#Dyad'><p>Housing dyad for animals contained in the data set Nuclei</p></a></li>
<li><a href='#FSelect'><p>A Function to perform step-wise discriminant analysis using F statistics</p></a></li>
<li><a href='#getP'>
<p>An internal function for getting empirical p values</p></a></li>
<li><a href='#Groups'><p>The group ID for animals contained in the data set Nuclei</p></a></li>
<li><a href='#h2Estimate'>
<p>Estimates the heritability of a binomial trait</p></a></li>
<li><a href='#IntPlot'><p>A function to visualize the results of a MANOVA</p></a></li>
<li><a href='#LandscapePlot'>
<p>A function to visualize the Functional Landscape of measured traits</p></a></li>
<li><a href='#ldaPlot'>
<p>A function to visualize the results of a discriminant analysis</p></a></li>
<li><a href='#Loadings'>
<p>A function to visualize trait loadings onto discriminant function and principle component axes</p></a></li>
<li><a href='#makeCompMat'><p>A function to create a pairwise comparison matrix</p></a></li>
<li><a href='#MeanCent'>
<p>A function to scale data to mean 0</p></a></li>
<li><a href='#Nuclei'>
<p>Brain activity in 14 brain regions for 71 individuals</p></a></li>
<li><a href='#partialF'>
<p>A function to compute partial F statistics</p></a></li>
<li><a href='#PercentMax'>
<p>A function to scale data to the percent of the maximum observed</p></a></li>
<li><a href='#PermuteLDA'>
<p>A function to determine whether two groups are in statistically different locations in multivariate space See Collyer and Adams 2007</p></a></li>
<li><a href='#plotBinomPower'><p>A function to plot the results of a binomPower run</p></a></li>
<li><a href='#Power'>
<p>A function to estimate the error rate for FSelect and PermuteLDA.</p></a></li>
<li><a href='#PPCA'>
<p>A function to perform a probabilistic principle component analysis</p></a></li>
<li><a href='#Scores'><p>Principle component scores based on the data in Nuclei</p></a></li>
<li><a href='#simPower'>
<p>An internal function of binomPower, which actually calculates the p value</p></a></li>
<li><a href='#ZTrans'>
<p>A function to convert data into a z-score</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multivariate Analysis and Visualization for Biological Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-04-09</td>
</tr>
<tr>
<td>Author:</td>
<td>Samuel V. Scarpino</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Samuel V. Scarpino &lt;scarpino@utexas.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Code to support a systems biology research program from inception through publication.  The methods focus on dimension reduction approaches to detect patterns in complex, multivariate experimental data and places an emphasis on informative visualizations.  The goal for this project is to create a package that will evolve over time, thereby remaining relevant and reflective of current methods and techniques.  As a result, we encourage suggested additions to the package, both methodological and graphical.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3.0)</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2 (&ge; 3.2.0), lme4 (&ge; 1.1-21), pcaMethods (&ge; 1.76.0),
misc3d (&ge; 0.8-4), MASS (&ge; 7.3-29), RColorBrewer(&ge; 1.1-2),
gridGraphics (&ge; 0.1-5)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-04-09 16:48:33 UTC; scarpino</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-04-09 17:10:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='multiDimBio-package'>Multivariate Analysis and Visualization for Biological Data
</h2><span id='topic+multiDimBio-package'></span><span id='topic+multiDimBio'></span>

<h3>Description</h3>

<p>Code to support a systems biology research program from inception through publication.  The methods focus on dimension reduction approaches to detect patterns in complex, multivariate experimental data and places an emphasis on informative visualizations.  The goal for this project is to create a package that will evolve over time, thereby remaining relevant and reflective of current methods and techniques.  As a result, we encourage suggested additions to the package, both methodological and graphical.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> multiDimBio</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.2.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-04-09</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL 3.0</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The datasets are: Nuclei, Groups, CondA, CondB, Scores, and Dyad
</p>
<p>The main functions are: boxWhisker, completeData, F_select, intPlot, ldaPlot, loadings, meanCent, percentMax, permuteLDA, power, ppca_mdb, zTrans, binomPower, h2Estimate, and plotBinomPower.
</p>
<p>Type ?&lt;object&gt; to learn more about these objects, e.g. ?Nuclei
</p>
<p>Type ?&lt;function&gt; to see examples of the function's use, e.g. ?FSelect
</p>


<h3>Author(s)</h3>

<p>Samuel V Scarpino
Maintainer: Samuel V Scarpino &lt;scarpino@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Collyer M, Adams D. (2007) Analysis of Two - State Multivariate Phenotypic Change in Ecological Studies. Ecology: 88(3) 683 - 692.
</p>
<p>Costanza M, Afifi A. (1979) Comparison of Stopping Rules in Forward Stepwise Discriminant Analysis. Journal of the American Statistical Association: pp. 777 - 78
</p>
<p>Crews D, Gillette R, Scarpino SV, Manikkam M, Savenkova MI, Skinner MK. (2012)
Epigenetic Transgenerational Alterations to Stress Response in Brain Gene Networks and 
Behavior. Proc. Natl. Acad. Sci. USA: 109(23) 9143 - 9148.
</p>
<p>Davies SW, Scarpino SV, Pongwarin T, Scott J, Matz MV. (2015) Estimating Trait Heritability in Highly Fecund Species. G3: Genes| Genomes| Genetics: 5(12) 2639 - 45.
</p>
<p>Habbema J, Hermans J. (1977) Selection of Variables in Discriminant Analysis by F-Statistics and Error Rate. Technometrics: 19(4) 487 - 493.
</p>
<p>Jennrich R. (1977) Stepwise discriminant analysis, volume 3. New York Wiley Sons.
</p>
<p>Roweis S. (1997) EM algorithms for PCA and sensible PCA. Neural Inf. Proc. Syst.: 10
626 - 632.
</p>
<p>Stacklies W, Redestig H, Scholz M, Walther D, Selbig J. (2007) pcaMethods - a Bioconductor package providing PCA methods for incomplete data. Bioinformatics: 23 1164 - 1167.
</p>
<p>Troyanskaya O, Cantor M, Sherlock G, Brown P, Hastie T, Tibshirani R, Botstein D, Altman R. (2001) Missing value estimation methods for DNA microarrays. Bioinformatics: 17(6) 520 - 5252.
</p>


<h3>See Also</h3>

<p><code><a href="pcaMethods.html#topic+pcaMethods">pcaMethods</a></code>
</p>

<hr>
<h2 id='binomPower'>
Power analysis for estimating the heritability of a binomial trait
</h2><span id='topic+binomPower'></span>

<h3>Description</h3>

<p>Performs a power analysis for estimating the heritability of a binomial trait.  This function can take a long time to run if either nsims or nperms is large.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binomPower(ndads, mm, vv, tau2, nperms, nsims, nbins, alpha = 0.05, doPlot=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="binomPower_+3A_ndads">ndads</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of dads.
</p>
</td></tr>
<tr><td><code id="binomPower_+3A_mm">mm</code></td>
<td>

<p>a (non-empty) numeric value indicating the mean number of offspring per dad per bin (normal dist).  mm must be less than vv.
</p>
</td></tr>
<tr><td><code id="binomPower_+3A_vv">vv</code></td>
<td>

<p>a (non-empty) numeric value indicating the variance in offspring per dad per bin (normal dist). vv. must be greater than mm.
</p>
</td></tr>
<tr><td><code id="binomPower_+3A_tau2">tau2</code></td>
<td>

<p>a (non-empty) numeric value indicating the dad effect (narrow-sense heritability ~ tau2/(tau2+(pi/sqrt(3))^2)).
</p>
</td></tr>
<tr><td><code id="binomPower_+3A_nperms">nperms</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of bootstrap permutations to use for caluclating a p value.
</p>
</td></tr>
<tr><td><code id="binomPower_+3A_nsims">nsims</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of simulations to run per parameter combination.
</p>
</td></tr>
<tr><td><code id="binomPower_+3A_nbins">nbins</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of bins, data are pooled before analysis.
</p>
</td></tr>
<tr><td><code id="binomPower_+3A_alpha">alpha</code></td>
<td>

<p>a (non-empty) numeric value indicating the cutoff for significant p values.
</p>
</td></tr>
<tr><td><code id="binomPower_+3A_doplot">doPlot</code></td>
<td>

<p>a (non-empty) logical value indicating whether to plot the results of the power analysis.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list and an optional set of .pdfs (if doPlot==TRUE).  The list contains:
</p>
<table role = "presentation">
<tr><td><code>roc</code></td>
<td>

<p>a data.frame with the summarized results of the power analysis.
</p>
</td></tr>
<tr><td><code>params</code></td>
<td>

<p>a numeric matrix with the paramater values.
</p>
</td></tr>
<tr><td><code>results</code></td>
<td>

<p>a numeric matrix with the full results of the analysis.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>	ndads &lt;- c(9,18)
	mm &lt;- 4.629634
	vv &lt;- 6.31339
	tau2 &lt;- c(0,0.5)
	nperms &lt;- 2
	nsims &lt;- 2
	nbins &lt;- 3
	doPlot &lt;- TRUE
	binomPower(ndads,mm,vv,tau2,nperms,nsims,nbins,doPlot)
</code></pre>

<hr>
<h2 id='boxWhisker'>A function to create a box and whisker plot by group ID
</h2><span id='topic+boxWhisker'></span>

<h3>Description</h3>

<p>A function to create a box and whisker plot by group ID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boxWhisker(data, groups, palette = "Paired")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="boxWhisker_+3A_data">data</code></td>
<td>

<p>a (non-empty) matrix of data values
</p>
</td></tr>
<tr><td><code id="boxWhisker_+3A_groups">groups</code></td>
<td>

<p>a (non-empty) vector of group IDs with length equal to the number of rows in data
</p>
</td></tr>
<tr><td><code id="boxWhisker_+3A_palette">palette</code></td>
<td>

<p>A color palette for plotting.  The default is 'Paired.'  See colorbrewer2.org for alternatives.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a box-whisker plot of the data by group ID.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nuclei)
data(Groups)
boxWhisker(Nuclei, Groups)

#changing the color palette

boxWhisker(data = Nuclei, groups = Groups, palette = 'Set1')
</code></pre>

<hr>
<h2 id='completeData'>
Function to impute missing data.
</h2><span id='topic+completeData'></span>

<h3>Description</h3>

<p>This function imputes missing data using a probabilistic principle component analysis framework and is a wrapper around functions implemented in the pcaMethods package (Stacklies et al. 2007), was proposed by Troyanskaya et al 2001 and is based on methods developed in Roweis 1997.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>completeData(data, n_pcs, cut.trait = 0.5, cut.ind = 0.5, show.test = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="completeData_+3A_data">data</code></td>
<td>

<p>a (non-empty) numeric matrix of data values.
</p>
</td></tr>
<tr><td><code id="completeData_+3A_n_pcs">n_pcs</code></td>
<td>

<p>a (non-empty) numeric value indicating the desired number of principle component axes.
</p>
</td></tr>
<tr><td><code id="completeData_+3A_cut.trait">cut.trait</code></td>
<td>

<p>a number indicating the maximum proportion of missing traits before an individual is removed from data.  A value of 1 will not remove any individuals and 0 will remove them all.
</p>
</td></tr>
<tr><td><code id="completeData_+3A_cut.ind">cut.ind</code></td>
<td>

<p>a number indicating the maximum proportion of individuals missing a trait score before that trait is removed from data.  A value of 1 will not remove any traits and 0 will remove them all.
</p>
</td></tr>
<tr><td><code id="completeData_+3A_show.test">show.test</code></td>
<td>

<p>a logical statement indicating whether a diagnostic plot of the data imputation should be returned.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with two entries. 
</p>
<table role = "presentation">
<tr><td><code>complete_dat</code></td>
<td>

<p>an object of class matrix with missing values imputed using a probabilistic principle component framework.
</p>
</td></tr>
<tr><td><code>plots</code></td>
<td>

<p>a list of plots stored as grid plots.
</p>
</td></tr>
</table>


<h3>References</h3>

<p>Roweis S (1997). EM algorithms for PCA and sensible PCA. Neural Inf. Proc. Syst., 10, 626 - 632.
</p>
<p>Stacklies W, Redestig H, Scholz M, Walther D, Selbig J (2007). pcaMethods - a Bioconductor package providing PCA methods for incomplete data. Bioinformatics, 23, 1164 - 1167.
</p>
<p>Troyanskaya O, Cantor M, Sherlock G, Brown P, Hastie T, Tibshirani R, Botstein D, Altman R (2001). Missing value estimation methods for DNA microarrays. Bioinformatics, 17(6), 520 - 5252.
</p>


<h3>See Also</h3>

<p><code><a href="pcaMethods.html#topic+pcaMethods">pcaMethods</a></code>, <code>pca</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nuclei)
npcs&lt;-floor(ncol(Nuclei)/5)

length(which(is.na(Nuclei))==TRUE)

dat.comp&lt;-completeData(data = Nuclei, n_pcs = npcs)

length(which(is.na(dat.comp))==TRUE)
</code></pre>

<hr>
<h2 id='CondA'>Treatment condition for animals contained in the data set Nuclei
</h2><span id='topic+CondA'></span>

<h3>Description</h3>

<p>Animals measured in the Nuclei data set were either from linneages exposed to the fungicide Vinclozolin (Vinclozolin) or not (Control).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(CondA)
</code></pre>


<h3>Format</h3>

<p>A factor vector indicating which treatment group the individuals in Nuclei belong to.
</p>


<h3>Source</h3>

<p>The data are provided courtesy of David Crews at the University of Texas at Austin.
</p>


<h3>References</h3>

<p>Crews, D, R Gillette, SV Scarpino, M Manikkam, MI Savenkova, MK Skinner. 2012.
Epigenetic Transgenerational Alterations to Stress Response in Brain Gene Networks and 
Behavior. Proc. Natl. Acad. Sci. USA. 109 (23). 9143 - 9148.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(CondA)
</code></pre>

<hr>
<h2 id='CondB'>Stress condition for animals contained in the data set Nuclei
</h2><span id='topic+CondB'></span>

<h3>Description</h3>

<p>Animals measured in the Nuclei data set were either subjected to chronic restraint stress (stress) or not (control).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(CondB)
</code></pre>


<h3>Format</h3>

<p>A factor vector indicating which stress group the individuals in Nuclei belong to.
</p>


<h3>Source</h3>

<p>The data are provided courtesy of David Crews at the University of Texas at Austin.
</p>


<h3>References</h3>

<p>Crews, D, R Gillette, SV Scarpino, M Manikkam, MI Savenkova, MK Skinner. 2012.
Epigenetic Transgenerational Alterations to Stress Response in Brain Gene Networks and 
Behavior. Proc. Natl. Acad. Sci. USA. 109 (23). 9143 - 9148.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(CondB)
</code></pre>

<hr>
<h2 id='Dyad'>Housing dyad for animals contained in the data set Nuclei
</h2><span id='topic+Dyad'></span>

<h3>Description</h3>

<p>Animals measured in the Nuclei data set were housed in dyads with one individual from the Vinclozolin line and one from the control line housed together.  Each dyad was either stressed or not stressed.    
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Dyad)
</code></pre>


<h3>Format</h3>

<p>A factor vector indicating which housing dyad the individuals in Nuclei are in.
</p>


<h3>Source</h3>

<p>The data are provided courtesy of David Crews at the University of Texas at Austin.
</p>


<h3>References</h3>

<p>Crews, D, R Gillette, SV Scarpino, M Manikkam, MI Savenkova, MK Skinner. 2012.
Epigenetic Transgenerational Alterations to Stress Response in Brain Gene Networks and 
Behavior. Proc. Natl. Acad. Sci. USA. 109 (23). 9143 - 9148.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Dyad)
</code></pre>

<hr>
<h2 id='FSelect'>A Function to perform step-wise discriminant analysis using F statistics</h2><span id='topic+FSelect'></span>

<h3>Description</h3>

<p>Select data using a F tests
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FSelect(Data, Group, target, p.adj.method = "holm", Missing.Data = "Complete")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FSelect_+3A_data">Data</code></td>
<td>

<p>A (non-empty), numeric matrix of data values
</p>
</td></tr>
<tr><td><code id="FSelect_+3A_group">Group</code></td>
<td>

<p>A (non-empty), vector indicating group membership.  Length(unique(Group))==2
</p>
</td></tr>
<tr><td><code id="FSelect_+3A_target">target</code></td>
<td>

<p>The number of desired traits. Target cannot be greater than the number of columns in Data
</p>
</td></tr>
<tr><td><code id="FSelect_+3A_p.adj.method">p.adj.method</code></td>
<td>

<p>The method used to control for false discovery.  The default setting is 'holm'
</p>
</td></tr>
<tr><td><code id="FSelect_+3A_missing.data">Missing.Data</code></td>
<td>

<p>The method used to handle missing data.  The default, 'Complete' will use completeData to impute missing data, setting Missing.Data='Remove' will remove all individuals with missing data.  FSelect cannot handle missing data.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>FSelect returns list containing at least the following components:
</p>
<table role = "presentation">
<tr><td><code>Selected</code></td>
<td>

<p>An ordered list indicating which columns were selected.	
</p>
</td></tr>
<tr><td><code>F.Selected</code></td>
<td>

<p>An ordered list containing the F statistics for each column indicated in Selected.  
</p>
</td></tr>
<tr><td><code>PrF</code></td>
<td>

<p>An ordered list containing the p values for each column indicated in Selected.
</p>
</td></tr>
<tr><td><code>PrNotes</code></td>
<td>

<p>A string indicating which method was used to control for multiple comparisons
</p>
</td></tr>
<tr><td><code>model</code></td>
<td>

<p>An lm object with the final model results.	
</p>
</td></tr>
</table>


<h3>References</h3>

<p>Costanza M, Afifi A (1979). Comparison of Stopping Rules in Forward Stepwise Discriminant Analysis. Journal of the American Statistical Association, pp. 777 - 78
</p>
<p>Habbema J, Hermans J (1977). Selection of Variables in Discriminant Analysis by F - Statistics and Error Rate. Technometrics, 19(4), 487 - 493.
</p>
<p>Jennrich R (1977). Stepwise discriminant analysis, volume 3. New York Wiley Sons.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+completeData">completeData</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nuclei)
data(Groups)
npcs&lt;-floor(ncol(Nuclei)/5)

dat.comp &lt;- completeData(data = Nuclei, n_pcs = npcs) 
groups.use &lt;- c(1,2) 
use.dat &lt;- which(Groups==groups.use[1]|Groups==groups.use[2])

dat.use &lt;- Nuclei[use.dat,]
GR.use &lt;- Groups[use.dat]

#not run
#FSelect(DAT.use,GR.use,3)
</code></pre>

<hr>
<h2 id='getP'>
An internal function for getting empirical p values
</h2><span id='topic+getP'></span>

<h3>Description</h3>

<p>Simulates p values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getP(ndads, mm, vv, tau2, nperms, nsims, nbins)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getP_+3A_ndads">ndads</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of dads.
</p>
</td></tr>
<tr><td><code id="getP_+3A_mm">mm</code></td>
<td>

<p>a (non-empty) numeric value indicating the mean number of offspring per dad per bin (normal dist). mm must be less than vv. 
</p>
</td></tr>
<tr><td><code id="getP_+3A_vv">vv</code></td>
<td>

<p>a (non-empty) numeric value indicating the variance in offspring per dad per bin (normal dist). vv must be greater than mm.
</p>
</td></tr>
<tr><td><code id="getP_+3A_tau2">tau2</code></td>
<td>

<p>a (non-empty) numeric value indicating the dad effect (narrow-sense heritability ~ tau2/(tau2+(pi/sqrt(3))^2)).
</p>
</td></tr>
<tr><td><code id="getP_+3A_nperms">nperms</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of bootstrap permutations to use for caluclating a p value.
</p>
</td></tr>
<tr><td><code id="getP_+3A_nsims">nsims</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of simulations to run per parameter combination.
</p>
</td></tr>
<tr><td><code id="getP_+3A_nbins">nbins</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of bins, data are pooled before analysis.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a vector of simulated p values.  The list contains:
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	ndads &lt;- c(9,18)
	mm &lt;- 4.629634
	vv &lt;- 6.31339
	tau2 &lt;- c(0,0.5)
	nperms &lt;- 2
	nsims &lt;- 2
	nbins &lt;- 3
	getP(ndads, mm, vv, tau2, nperms, nsims, nbins)
</code></pre>

<hr>
<h2 id='Groups'>The group ID for animals contained in the data set Nuclei
</h2><span id='topic+Groups'></span>

<h3>Description</h3>

<p>Animals measured in the Nuclei data set belong to one of four groups determined by their linneage (Vinclozolin or Control) and their stress treatment (Stressed or Non-Stressed). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Groups)
</code></pre>


<h3>Format</h3>

<p>A factor vector indicating which group the individuals in Nuclei are in.
</p>


<h3>Source</h3>

<p>The data are provided courtesy of David Crews at the University of Texas at Austin.
</p>


<h3>References</h3>

<p>Crews, D, R Gillette, SV Scarpino, M Manikkam, MI Savenkova, MK Skinner. 2012.
Epigenetic Transgenerational Alterations to Stress Response in Brain Gene Networks and 
Behavior. Proc. Natl. Acad. Sci. USA. 109 (23). 9143 - 9148.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Groups)
</code></pre>

<hr>
<h2 id='h2Estimate'>
Estimates the heritability of a binomial trait
</h2><span id='topic+h2Estimate'></span>

<h3>Description</h3>

<p>Estimates the narrow-sense heritability of a binomial trait and calculates a p value by randomization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>h2Estimate(data,nreps=1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="h2Estimate_+3A_data">data</code></td>
<td>

<p>a (non-empty) numeric matrix with three columns.  The first two should contain the trait data (number of occurances of each outcome type) and the third should contain the group ids.
</p>
</td></tr>
<tr><td><code id="h2Estimate_+3A_nreps">nreps</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of resamples to perform when calculating the emperical p value.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Estimates the narrow-sense heritability of a binomial trait.  This function works by fitting two models, one with and one without a random-effect of sire.  These models are compared by randomizing the sire ids nreps times and re-fitting the model.  For each of the nreps model pairs, a deviance is calculated and a &quot;p value&quot; estimated by comparing that distribution of deviance to the observed.  The heritability is approximatly tau2/(tau2+(pi/sqrt(3))^2), where tau2 is the random-effect variance due to sire.
</p>


<h3>Value</h3>

<p>Returns a list.  The list contains:
</p>
<table role = "presentation">
<tr><td><code>h2</code></td>
<td>

<p>The estimated narrow-sense heritability.  The narrow-sense heritability is approximatly tau2/(tau2+(pi/sqrt(3))^2), where tau2 is the random-effect variance due to sire.
</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>

<p>The probability that the best-fit model includes an extra variance term for sire (random effect of dad).  The value is calculated by comparing the deviances from nreps number of randomized model comparisions. 
</p>
</td></tr>
<tr><td><code>deviance</code></td>
<td>

<p>The deviance between a null model without a random effect of dad and a model with.
</p>
</td></tr>
<tr><td><code>sim</code></td>
<td>

<p>The simulated deviances used in calculating the p value in pval.
</p>
</td></tr>
<tr><td><code>obsMod</code></td>
<td>

<p>The glmer model object resulting from the observed data.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>	#non-zero heritability
	ndads &lt;- 18
	mm &lt;- 4
	vv &lt;-  6
	tau2 &lt;- 2.5
	nbins &lt;- 3
	
	mylogit &lt;- function(x) log(x/{1-x})
	ilogit &lt;- function(x) 1/{1+exp(-x)}
	
	swimprob &lt;- ilogit(rnorm(ndads, 0, sqrt(tau2)))
	mytable &lt;- NULL
	for(i in 1:ndads) {
		bincounts &lt;- pmax(1,rnbinom(nbins, mu = mm, size = mm^2/{vv-mm}))
		swim &lt;- rbinom(3, bincounts,swimprob[i])
		set &lt;- bincounts - swim
		theserows &lt;- data.frame(set=set,swim=swim, Dad = i, Bin = 1:nbins)
		mytable &lt;- rbind(mytable, theserows)
	}
	
	est &lt;- h2Estimate(mytable,nreps=10)
	
	print(est$h2)
	
	#zero heritability
	ndads &lt;- 18
	mm &lt;- 4
	vv &lt;-  6
	tau2 &lt;- 0
	nbins &lt;- 3
	
	mylogit &lt;- function(x) log(x/{1-x})
	ilogit &lt;- function(x) 1/{1+exp(-x)}
	
	swimprob &lt;- ilogit(rnorm(ndads, 0, sqrt(tau2)))
	mytable0 &lt;- NULL
	for(i in 1:ndads) {
		bincounts &lt;- pmax(1,rnbinom(nbins, mu = mm, size = mm^2/{vv-mm}))
		swim &lt;- rbinom(3, bincounts,swimprob[i])
		set &lt;- bincounts - swim
		theserows &lt;- data.frame(set=set,swim=swim, Dad = i, Bin = 1:nbins)
		mytable0 &lt;- rbind(mytable0, theserows)
	}
	
	est0 &lt;- h2Estimate(mytable0,nreps=10)
	
	print(est0$h2)
</code></pre>

<hr>
<h2 id='IntPlot'>A function to visualize the results of a MANOVA
</h2><span id='topic+IntPlot'></span>

<h3>Description</h3>

<p>The function produces an interaction plot to demonstrate the results of a MANOVA using the function interaction.plot.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IntPlot(Scores, Cov.A, Cov.B, pvalues = rep(1, 8), int.pvalues = rep(1, 4))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="IntPlot_+3A_scores">Scores</code></td>
<td>

<p>A (non-empty) numeric matrix of principle component scores or raw data.
</p>
</td></tr>
<tr><td><code id="IntPlot_+3A_cov.a">Cov.A</code></td>
<td>

<p>A (non-empty) bivariate factor vector indicating the factor for each row in Scores
</p>
</td></tr>
<tr><td><code id="IntPlot_+3A_cov.b">Cov.B</code></td>
<td>

<p>A (non-empty) bivariate factor vector indicating the factor for each row in Scores
</p>
</td></tr>
<tr><td><code id="IntPlot_+3A_pvalues">pvalues</code></td>
<td>

<p>An optional vector of p values for each covariate across Scores.  The length of pvalues must equal the number of columns in Scores times 2.
</p>
</td></tr>
<tr><td><code id="IntPlot_+3A_int.pvalues">int.pvalues</code></td>
<td>

<p>An optional vector of p values for each interaction.  The length of int.pvalues must equal the number of columns in Scores.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of plots stored as grid plots.  
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+interaction.plot">interaction.plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Scores)
data(CondA)
data(CondB)

pvals&lt;-c(0.03,0.6,0.05,0.07,0.9,0.2,0.5,0.3)
int.pvals&lt;-c(0.3,0.45,0.5,0.12)

IntPlot(Scores,CondA,CondB,pvalues=pvals, int.pvalues=int.pvals)
</code></pre>

<hr>
<h2 id='LandscapePlot'>
A function to visualize the Functional Landscape of measured traits
</h2><span id='topic+LandscapePlot'></span>

<h3>Description</h3>

<p>This function plots a three-dimensional landscape of measured traits.  The peak heights are relative with respect to the input data. The width of each peak is controlled by the argument sigma and has only an aesthetic purpose. The 3D image is generated using the <code><a href="misc3d.html#topic+drawScene">drawScene</a></code> and <code><a href="misc3d.html#topic+surfaceTriangles">surfaceTriangles</a></code> .   
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LandscapePlot(Data, Groups=NULL, PDF=FALSE,LocPlot=FALSE,control=c(75,1,30))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="LandscapePlot_+3A_data">Data</code></td>
<td>

<p>A (non-empty) numeric matrix with trait values
</p>
</td></tr>
<tr><td><code id="LandscapePlot_+3A_groups">Groups</code></td>
<td>

<p>A (non-empty)factor vector indicating the group membership of each row in Data. If there is only a single group present in Data then Groups=NULL (default).
</p>
</td></tr>
<tr><td><code id="LandscapePlot_+3A_pdf">PDF</code></td>
<td>

<p>Logical controlling whether to output the results as a .pdf or a .jpeg.  The default (PDF=FALSE) will produce a .jpeg.  The file size for .pdf output can be large.
</p>
</td></tr>
<tr><td><code id="LandscapePlot_+3A_locplot">LocPlot</code></td>
<td>

<p>Logical controlling whether to output a .pdf naming the peaks according to the columns they represent.  The defaul is FALSE.
</p>
</td></tr>
<tr><td><code id="LandscapePlot_+3A_control">control</code></td>
<td>

<p>An optional numeric vector setting the control parameters for persp.  control[1] = theta, control[2] = r, control[3] = phi
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of plots stored as grid plots (or.pdf if PDF=TRUE) file for each column in data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nuclei) 
data(Groups)
 
#plotting the first six columns
#not run
#LandscapePlot(Nuclei[,1:6], Groups=Groups)
</code></pre>

<hr>
<h2 id='ldaPlot'>
A function to visualize the results of a discriminant analysis
</h2><span id='topic+ldaPlot'></span>

<h3>Description</h3>

<p>The function takes as input the traits and group IDs and will perform a discriminate function analysis and visualize the results. For the pair-wise comparison of groups we use density histograms with points along the x-axis denoting the actual data, Figure 3 For multi-group comparisons we plot a bivariate scatter for all pairwise combinations of discriminate axes. The color of plotting symbols can be altered using the palette argument and the axes comparisons (with max n = number of groups - 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldaPlot(Data, Groups, palette = "BrBG", axes = c(1, 2, 2, 3, 1, 3))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ldaPlot_+3A_data">Data</code></td>
<td>

<p>A (non-empty), numeric matrix of data values
</p>
</td></tr>
<tr><td><code id="ldaPlot_+3A_groups">Groups</code></td>
<td>

<p>A (non-empty), vector indicating group membership.  Length(unique(Group))==2
</p>
</td></tr>
<tr><td><code id="ldaPlot_+3A_palette">palette</code></td>
<td>

<p>A color palette for plotting.  The default is 'Paired.'  See colorbrewer2.org for alternatives.
</p>
</td></tr>
<tr><td><code id="ldaPlot_+3A_axes">axes</code></td>
<td>

<p>A numeric vector describing which axes to compare.  For example, axes=c(1,2) will on produce a single plot comparing the first and second axis.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of ggplot2 plots.
</p>


<h3>See Also</h3>

<p><code><a href="MASS.html#topic+lda">lda</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nuclei)
data(Groups) 
ldaPlot(Nuclei, Groups, palette='BrBG', axes=c(1,2,2,3,1,3))
</code></pre>

<hr>
<h2 id='Loadings'>
A function to visualize trait loadings onto discriminant function and principle component axes
</h2><span id='topic+Loadings'></span>

<h3>Description</h3>

<p>This function produces barplots representative of the contribution of a particular trait or variable to either a discriminant function or principle component axis. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Loadings(DATA, GROUPS, method = c("PCA", "LDA"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Loadings_+3A_data">DATA</code></td>
<td>

<p>A (non-empty) numeric matrix with trait values
</p>
</td></tr>
<tr><td><code id="Loadings_+3A_groups">GROUPS</code></td>
<td>

<p>A (non-empty)factor vector indicating the group membership of each row in DATA
</p>
</td></tr>
<tr><td><code id="Loadings_+3A_method">method</code></td>
<td>

<p>An optional list indicating whether the results for a principle component analysis, 'PCA', or linear discriminant analysis, 'LDA' should be performed.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Outputs a list with values and plots for each test listed in method.
</p>


<h3>See Also</h3>

<p><code>pca</code>, <code><a href="MASS.html#topic+lda">lda</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nuclei) 
data(Groups) 
Loadings(Nuclei, Groups, method=c("PCA", "LDA"))
</code></pre>

<hr>
<h2 id='makeCompMat'>A function to create a pairwise comparison matrix
</h2><span id='topic+makeCompMat'></span>

<h3>Description</h3>

<p>This function creates a pairwise comparison matrix for n groups.  All possible pairwise combinations are created, with rows in the matrix equal to the desired comparison.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeCompMat(ng)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="makeCompMat_+3A_ng">ng</code></td>
<td>

<p>A single number indicating the total number of unique groups
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with two columns and ng choose 2 rows.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PermuteLDA">PermuteLDA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>makeCompMat(3)

makeCompMat(4)

data(Groups)
NGroups&lt;-length(unique(Groups))

makeCompMat(NGroups)
</code></pre>

<hr>
<h2 id='MeanCent'>
A function to scale data to mean 0
</h2><span id='topic+MeanCent'></span>

<h3>Description</h3>

<p>This function rescales the columns in a data matrix to have mean 0.  The variance is not scaled and missing values are ignored in the calculation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
MeanCent(DATA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MeanCent_+3A_data">DATA</code></td>
<td>

<p>A (non-empty) matrix with data values.  Columns should be different traits and rows unique observations of those traits
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with the same dimensions as DATA.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ZTrans">ZTrans</a></code>, <code><a href="#topic+PercentMax">PercentMax</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nuclei)

colMeans(Nuclei, na.rm=TRUE)

Nuclei.MC&lt;-MeanCent(Nuclei)

colMeans(Nuclei.MC, na.rm=TRUE)
</code></pre>

<hr>
<h2 id='Nuclei'>
Brain activity in 14 brain regions for 71 individuals
</h2><span id='topic+Nuclei'></span>

<h3>Description</h3>

<p>The activity in 14 brain nuclei were measured in rats that were in one of four groups: 
1) Non-stressed, Control 
2) Stressed, Control
3) Non-stressed, Vinclozolin
4) Stressed, Vinclozolin
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Nuclei)
</code></pre>


<h3>Format</h3>

<p>A numeric matrix with 71 individuals as rows and the activity of 14 brain nuclei as columns.  NAs indicate missing data.
</p>


<h3>Details</h3>

<p>Two different cohorts of male rats of the F3 generation of Vinclozolin (Vinclozolin-Lineage) and Vehicle Control (Control-Lineage) Lineages produced at Washington State University are shipped to the University of Texas on the day after weaning. Rats are randomly pair-housed (one Control-Lineage and one Vinclozolin-Lineage animal) and remain in these dyads throughout the duration of the study. Half of the dyads are randomly chosen to receive chronic restraint stress (CRS) treatment for 6 hours daily for 21 consecutive days commencing 1 hr after lights off.  Activity in 14 brain nuclei were measured at the end of the study.
</p>


<h3>Source</h3>

<p>The data are provided courtesy of David Crews at the University of Texas at Austin.
</p>


<h3>References</h3>

<p>Crews, D, R Gillette, SV Scarpino, M Manikkam, MI Savenkova, MK Skinner. 2012.
Epigenetic Transgenerational Alterations to Stress Response in Brain Gene Networks and 
Behavior. Proc. Natl. Acad. Sci. USA. 109 (23). 9143 - 9148.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nuclei)
</code></pre>

<hr>
<h2 id='partialF'>
A function to compute partial F statistics
</h2><span id='topic+partialF'></span>

<h3>Description</h3>

<p>This is an internal function used in FSelect.  It can only be used for two groups.  The partial F statistic is the additional contribution to the model from adding one more trait.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>partialF(m.lda, GROUP, T_pm1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="partialF_+3A_m.lda">m.lda</code></td>
<td>

<p>An object of class 'lda'
</p>
</td></tr>
<tr><td><code id="partialF_+3A_group">GROUP</code></td>
<td>

<p>A factor vector indicating group membership
</p>
</td></tr>
<tr><td><code id="partialF_+3A_t_pm1">T_pm1</code></td>
<td>

<p>The F statistic calculated for a discriminant analysis with only the most informative trait.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a partial F statistic
</p>


<h3>References</h3>

<p>Habbema J, Hermans J (1977). Selection of Variables in Discriminant Analysis by F-Statistics and Error Rate. Technometrics, 19(4), 487 - 493.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FSelect">FSelect</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Internal function used in FSelect

data(Nuclei)
data(Groups)

NPC&lt;-floor(ncol(Nuclei)/5)

DAT.comp&lt;-completeData(Nuclei, n_pcs = NPC) 
Groups.use&lt;-c(1,2) 
use.DAT&lt;-which(Groups==Groups.use[1]|Groups==Groups.use[2])

DAT.use&lt;-Nuclei[use.DAT,]
GR.use&lt;-Groups[use.DAT]

traitA&lt;-2

mlda&lt;-MASS::lda(GR.use~DAT.use[,traitA])

F1&lt;-partialF(mlda,GR.use,0)

traitB&lt;-1

mlda2&lt;-MASS::lda(GR.use~DAT.use[,c(traitA,traitB)])

partialF(mlda2,GR.use,F1)

</code></pre>

<hr>
<h2 id='PercentMax'>
A function to scale data to the percent of the maximum observed
</h2><span id='topic+PercentMax'></span>

<h3>Description</h3>

<p>This function rescales the columns in a data matrix to the percent of the maximum observed value.  The variance is not scaled and missing values are ignored in the calculation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
PercentMax(DATA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PercentMax_+3A_data">DATA</code></td>
<td>

<p>A (non-empty) matrix with data values.  Columns should be different traits and rows unique observations of those traits
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with the same dimensions as DATA.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ZTrans">ZTrans</a></code>, <code><a href="#topic+MeanCent">MeanCent</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nuclei)

colMeans(Nuclei, na.rm=TRUE)

Nuclei.PM&lt;-PercentMax(Nuclei)

colMeans(Nuclei.PM, na.rm=TRUE)
</code></pre>

<hr>
<h2 id='PermuteLDA'>
A function to determine whether two groups are in statistically different locations in multivariate space See Collyer and Adams 2007
</h2><span id='topic+PermuteLDA'></span>

<h3>Description</h3>

<p>The function calculates the multivariate distance between two groups across all traits and determines whether they differ signifcantly using a Monte Carlo randomization test.  The Monte Carlo randomization creates a null distribution by randomizing the residual deviation from the group mean across all individuals.  This method controls for heteroscedasticity and was designed by Collyer and Adams (2007) for use in analyzing data sets that have sparse groups sizes relative to the number of traits.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PermuteLDA(Data, Groups, NPerm, Missing.Data = "Complete")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PermuteLDA_+3A_data">Data</code></td>
<td>

<p>A (non-empty), numeric matrix of data values
</p>
</td></tr>
<tr><td><code id="PermuteLDA_+3A_groups">Groups</code></td>
<td>

<p>A (non-empty), vector indicating group membership.  
</p>
</td></tr>
<tr><td><code id="PermuteLDA_+3A_nperm">NPerm</code></td>
<td>

<p>The number of permutations used to generate the null distribution.  The default is 100.
</p>
</td></tr>
<tr><td><code id="PermuteLDA_+3A_missing.data">Missing.Data</code></td>
<td>

<p>The method used to handle missing data.  The default, 'Complete' will use CompleteData to impute missing data, setting Missing.Data='Remove' will remove all individuals with missing data.  FSelect cannot handle missing data.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Determining the statistical significance of a discriminate function analysis along with performing that analysis on sparse data sets, e.g. many traits observed on comparatively few individuals, is a challenge.	Collyer and Adams (2007) developed a Monte Carlo based algorithm for addressing both of those issues. Briefly, the test uses the underlying Var/Cov structure of the data and randomizes the group membership to calculate a null distribution. This test simultaneously controls for heteroscedasticity, a common problem in sparse data sets and allows the approximation of a p-value for the test. For the original implementation and formulation of the method see Collyer and Adams (2007) or http://www.public.iastate. edu/~dcadams/software.html. Unlike the FSelect implementation, PermuteLDA will work properly with an arbitrary number of groups. The time required to run the algorithm is non-linear in the number of groups.
</p>


<h3>Value</h3>

<p>Returns a data frame with four columns and the number of groups choose 2 rows.  Each row is a pairwise comparison between groups.  The column 'Pr' is the p value to reject the null hypothesis of no difference (a value in 'Pr' &lt; 0.05 would result in rejecting the hypothesis that the two groups are not different.  The column 'Distance' is the multivariate distance between the two groups.  
</p>


<h3>References</h3>

<p>Collyer M, Adams D (2007). Analysis of Two - State Multivariate Phenotypic Change in Ecological Studies. Ecology, 88(3), 683 - 692.
</p>
<p>For an implementation of the original method coded in R see http://www.public.iastate. edu/~dcadams/software.html.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PermuteLDA">PermuteLDA</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(Nuclei)
	data(Groups)
	PermuteLDA(Nuclei,Groups,50)
</code></pre>

<hr>
<h2 id='plotBinomPower'>A function to plot the results of a binomPower run
</h2><span id='topic+plotBinomPower'></span>

<h3>Description</h3>

<p>A function to plot the results of a binomPower run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotBinomPower(datPlotBig,params)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotBinomPower_+3A_datplotbig">datPlotBig</code></td>
<td>

<p>a (non-empty) matrix of data values, with columns trueTau, ndads, trueTau2
</p>
</td></tr>
<tr><td><code id="plotBinomPower_+3A_params">params</code></td>
<td>

<p>a (non-empty) matrix of parameter values, with columns mm and vv. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of two plots of the binomPower analysis results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#not run
</code></pre>

<hr>
<h2 id='Power'>
A function to estimate the error rate for FSelect and PermuteLDA.
</h2><span id='topic+Power'></span>

<h3>Description</h3>

<p>Methods are implemented to compute the statistical power, in terms of the type II error rate, based on anticipated sample and effect sizes for FSelect() and PermuteLDA(). By default the power of both tests are determined by iterating over a range of effect and sample sizes. The default settings were selected to be representative of many behavioral genetic studies; however, users can input alternative sample and effect sizes.   For high values of trials this function can be very slow.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Power(func = "PermuteLDA", N = "DEFAULT.N", effect.size = "DEFAULT.e", trials = 100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Power_+3A_func">func</code></td>
<td>

<p>A character string indicating which function to compute the power for, can be either 'PermuteLDA' or 'FSelect'
</p>
</td></tr>
<tr><td><code id="Power_+3A_n">N</code></td>
<td>

<p>A (non-empty) vector of group sizes.  The lenght of N must be greater than 1 and tha minimum group size for 'FSelect' can not be less than 6.  The size of each group is N/2.
</p>
</td></tr>
<tr><td><code id="Power_+3A_effect.size">effect.size</code></td>
<td>

<p>A (non-empty) vector or single value of effect sizes.
</p>
</td></tr>
<tr><td><code id="Power_+3A_trials">trials</code></td>
<td>

<p>A number indicating the number of trials for each combination of N and effect.size to calculate the power.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The algorithm for the power analysis proceeds as follows:
1. Input sample and effect sizes
2. Set the number of significant effects, e to 0. Note - Total number of traits is fixed at 6
3. Draw random deviates for the given sample size for 6 traits. Note - All traits not significant under this iteration are drawn from a N(0,1) distribution.
4. Perform either FSelect() or PermuteLDA() and record the results. 5. Return to step 3 N times, recording the results each time. Note - N is set using the
trials input
6. If e&lt;5 return to step 2 and set the number of significant effects to e+1
7. Proceed to the next combination of sample and effect size.
8. Output the results for each combination of sample and effect size as a function of the number of significant traits.
</p>


<h3>Value</h3>

<p>Outputs a list with plots and results for each effect size.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PermuteLDA">PermuteLDA</a></code>,<code><a href="#topic+FSelect">FSelect</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#not run
#Power(func = 'FSelect', N=c(6,8), effect.size=0.5, trials = 2)
</code></pre>

<hr>
<h2 id='PPCA'>
A function to perform a probabilistic principle component analysis
</h2><span id='topic+PPCA'></span>

<h3>Description</h3>

<p>Performs a probabilistic principle component analysis using the function 'pca' in the package'pcaMethods'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PPCA(Data, nPCs=4, CENTER=TRUE, SCALE='vector')
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PPCA_+3A_data">Data</code></td>
<td>

<p>A (non-empty), numeric matrix of data values
</p>
</td></tr>
<tr><td><code id="PPCA_+3A_npcs">nPCs</code></td>
<td>

<p>The number of resulting principle component axes.  nPCs must be less than or equal to the number of columns in Data.
</p>
</td></tr>
<tr><td><code id="PPCA_+3A_center">CENTER</code></td>
<td>

<p>A logical statement indicating whether data should be centered to mean 0, TRUE, or not, FALSE.
</p>
</td></tr>
<tr><td><code id="PPCA_+3A_scale">SCALE</code></td>
<td>

<p>A character string indicating which method should be used to scale the variances.  The default setting is 'vector.'  
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In PPCA an Expectation Maximization (EM) algorithm is used to fit a Gaussian latent variable model ( Tippping and Bishop (1999)). A latent variable model seeks to relate an observed vector of data to a lower dimensional vector of latent (or unobserved) variables, an approach similar to a factor analysis. Our implementation is a wrapper around the pcaMethods functions ppca and svdimpute (Stacklies et al. (2007)) and is included mainly for convience. The method used in pca was adapted from Roweis (1997) and a Matlab script developed by Jakob Verbeek.
</p>


<h3>Value</h3>

<p>Returns an object of class 'pcaRes.'  See documentation in the package code<a href="pcaMethods.html#topic+pcaMethods"> pcaMethods</a>
</p>


<h3>References</h3>

<p>Roweis S (1997). EM algorithms for PCA and sensible PCA. Neural Inf. Proc. Syst., 10,
626 - 632.
</p>
<p>Stacklies W, Redestig H, Scholz M, Walther D, Selbig J (2007). pcaMethods - a Bioconductor package providing PCA methods for incomplete data. Bioinformatics, 23, 1164 - 1167.
</p>
<p>Tippping M, Bishop C (1999). Probabilistic Principle Componenet Analysis. Journal of the Royal Statistical Society. Series B (Statistical Methodology), 61(3), 611 - 622.
</p>


<h3>See Also</h3>

<p><code><a href="pcaMethods.html#topic+pcaMethods">pcaMethods</a></code>, <code>pca</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	data(Nuclei)
	PPCA1&lt;-PPCA(Nuclei, nPCs=2, CENTER=TRUE, SCALE='vector')
	Scores1&lt;-PPCA1@scores
	</code></pre>

<hr>
<h2 id='Scores'>Principle component scores based on the data in Nuclei
</h2><span id='topic+Scores'></span>

<h3>Description</h3>

<p>Principle component scores were computed using PPCA for the data set Nuclei.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Scores)
</code></pre>


<h3>Format</h3>

<p>A numeric matrix with 4 columns and the same number of rows as Nuclei.  There are no missing values.
</p>


<h3>Source</h3>

<p>The data are provided courtesy of David Crews at the University of Texas at Austin.
</p>


<h3>References</h3>

<p>Crews, D, R Gillette, SV Scarpino, M Manikkam, MI Savenkova, MK Skinner. 2012.
Epigenetic Transgenerational Alterations to Stress Response in Brain Gene Networks and 
Behavior. Proc. Natl. Acad. Sci. USA. 109 (23). 9143 - 9148.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Scores)

data(Nuclei)

SCORES&lt;-PPCA(Nuclei)@scores
</code></pre>

<hr>
<h2 id='simPower'>
An internal function of binomPower, which actually calculates the p value
</h2><span id='topic+simPower'></span>

<h3>Description</h3>

<p>An internal function of binomPower, which actually calculates the p value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simPower(ndads,mm,vv,tau2,nperms,nbins)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simPower_+3A_ndads">ndads</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of dads.
</p>
</td></tr>
<tr><td><code id="simPower_+3A_mm">mm</code></td>
<td>

<p>a (non-empty) numeric value indicating the mean number of offspring per dad per bin (normal dist). mm must be less than vv. 
</p>
</td></tr>
<tr><td><code id="simPower_+3A_vv">vv</code></td>
<td>

<p>a (non-empty) numeric value indicating the variance in offspring per dad per bin (normal dist). vv must be great than mm. 
</p>
</td></tr>
<tr><td><code id="simPower_+3A_tau2">tau2</code></td>
<td>

<p>a (non-empty) numeric value indicating the dad effect (narrow-sense heritability ~ tau2/(tau2+(pi/sqrt(3))^2)).
</p>
</td></tr>
<tr><td><code id="simPower_+3A_nperms">nperms</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of bootstrap permutations to use for caluclating a p value.
</p>
</td></tr>
<tr><td><code id="simPower_+3A_nbins">nbins</code></td>
<td>

<p>a (non-empty) numeric value indicating the number of bins, data are pooled before analysis.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a p value for a given set of conditions over a specificed number of bootstrap permutations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#not run
</code></pre>

<hr>
<h2 id='ZTrans'>
A function to convert data into a z-score
</h2><span id='topic+ZTrans'></span>

<h3>Description</h3>

<p>This function converts the columns in a data matrix into z-scores.  The score is computed by subracting each observation in a column from the column mean and divding by the column standard deviation.  Each column is converted independently of the others missing values are ignored in the calculation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
ZTrans(DATA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZTrans_+3A_data">DATA</code></td>
<td>

<p>A (non-empty) matrix with data values.  Columns should be different traits and rows unique observations of those traits
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with the same dimensions as DATA.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PercentMax">PercentMax</a></code>, <code><a href="#topic+MeanCent">MeanCent</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Nuclei)

colMeans(Nuclei, na.rm=TRUE)

Nuclei.ZT&lt;-ZTrans(Nuclei)

colMeans(Nuclei.ZT, na.rm=TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
