<!DOCTYPE html><html><head><title>Help for package mutationtypes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mutationtypes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mutationtypes-package'><p>mutationtypes: Validate and Convert Mutational Impacts Using Standard Genomic Dictionaries</p></a></li>
<li><a href='#mutation_types_convert_pave_to_maf'><p>Convert PAVE Mutation Types to MAF</p></a></li>
<li><a href='#mutation_types_convert_so_to_maf'><p>Convert SO Mutation Types to MAF</p></a></li>
<li><a href='#mutation_types_identify'><p>Identify Mutation Dictionary Used</p></a></li>
<li><a href='#mutation_types_maf'><p>Dictionary of MAF terms</p></a></li>
<li><a href='#mutation_types_maf_palette'><p>Palettes: MAF</p></a></li>
<li><a href='#mutation_types_pave'><p>Dictionary of PAVE terms</p></a></li>
<li><a href='#mutation_types_pave_palette'><p>Palettes: PAVE</p></a></li>
<li><a href='#mutation_types_so'><p>Dictionary of So terms</p></a></li>
<li><a href='#mutation_types_so_palette'><p>Palettes: SO</p></a></li>
<li><a href='#select_most_severe_consequence_pave'><p>Select the most severe consequence (PAVE)</p></a></li>
<li><a href='#select_most_severe_consequence_so'><p>Select the most severe consequence (SO)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Validate and Convert Mutational Impacts Using Standard Genomic
Dictionaries</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Check concordance of a vector of mutation impacts with standard dictionaries such as Sequence Ontology (SO) <a href="http://www.sequenceontology.org/">http://www.sequenceontology.org/</a>, Mutation Annotation Format (MAF) <a href="https://docs.gdc.cancer.gov/Encyclopedia/pages/Mutation_Annotation_Format_TCGAv2/">https://docs.gdc.cancer.gov/Encyclopedia/pages/Mutation_Annotation_Format_TCGAv2/</a> or Prediction and Annotation of Variant Effects (PAVE) <a href="https://github.com/hartwigmedical/hmftools/tree/master/pave">https://github.com/hartwigmedical/hmftools/tree/master/pave</a>. 
  It enables conversion between SO/PAVE and MAF terms and selection of the most severe consequence where multiple ampersand (&amp;) delimited impacts are given.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-3">LGPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>assertions, cli, data.table, stats, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/selkamand/mutationtypes">https://github.com/selkamand/mutationtypes</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/selkamand/mutationtypes/issues">https://github.com/selkamand/mutationtypes/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-14 09:17:33 UTC; selkamand</td>
</tr>
<tr>
<td>Author:</td>
<td>Sam El-Kamand <a href="https://orcid.org/0000-0003-2270-8088"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Children's Cancer Institute Australia [cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sam El-Kamand &lt;sam.elkamand@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-14 16:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='mutationtypes-package'>mutationtypes: Validate and Convert Mutational Impacts Using Standard Genomic Dictionaries</h2><span id='topic+mutationtypes'></span><span id='topic+mutationtypes-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Check concordance of a vector of mutation impacts with standard dictionaries such as Sequence Ontology (SO) <a href="http://www.sequenceontology.org/">http://www.sequenceontology.org/</a>, Mutation Annotation Format (MAF) <a href="https://docs.gdc.cancer.gov/Encyclopedia/pages/Mutation_Annotation_Format_TCGAv2/">https://docs.gdc.cancer.gov/Encyclopedia/pages/Mutation_Annotation_Format_TCGAv2/</a> or Prediction and Annotation of Variant Effects (PAVE) <a href="https://github.com/hartwigmedical/hmftools/tree/master/pave">https://github.com/hartwigmedical/hmftools/tree/master/pave</a>. It enables conversion between SO/PAVE and MAF terms and selection of the most severe consequence where multiple ampersand (&amp;) delimited impacts are given.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Sam El-Kamand <a href="mailto:sam.elkamand@gmail.com">sam.elkamand@gmail.com</a> (<a href="https://orcid.org/0000-0003-2270-8088">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Children's Cancer Institute Australia [copyright holder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/selkamand/mutationtypes">https://github.com/selkamand/mutationtypes</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/selkamand/mutationtypes/issues">https://github.com/selkamand/mutationtypes/issues</a>
</p>
</li></ul>


<hr>
<h2 id='mutation_types_convert_pave_to_maf'>Convert PAVE Mutation Types to MAF</h2><span id='topic+mutation_types_convert_pave_to_maf'></span>

<h3>Description</h3>

<p>Convert PAVE Mutation Types to MAF
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation_types_convert_pave_to_maf(
  pave_mutation_types,
  variant_type = NULL,
  split_on_ampersand = TRUE,
  missing_to_silent = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutation_types_convert_pave_to_maf_+3A_pave_mutation_types">pave_mutation_types</code></td>
<td>
<p>a vector of PAVE terms you want to convert to MAF variant classifications (character)</p>
</td></tr>
<tr><td><code id="mutation_types_convert_pave_to_maf_+3A_variant_type">variant_type</code></td>
<td>
<p>a vector describing each mutations type. Valid elements include: &quot;SNP&quot;, &quot;DNP&quot;, &quot;TNP&quot;, &quot;ONP&quot;, &quot;DEL&quot;, &quot;INS&quot;. Used to map frameshift_variant to more specific MAF columns (character)</p>
</td></tr>
<tr><td><code id="mutation_types_convert_pave_to_maf_+3A_split_on_ampersand">split_on_ampersand</code></td>
<td>
<p>should '&amp;' separated PAVE terms be automatically converted to single PAVE terms based on highest severity? (flag)</p>
</td></tr>
<tr><td><code id="mutation_types_convert_pave_to_maf_+3A_missing_to_silent">missing_to_silent</code></td>
<td>
<p>should missing (NA) or empty (&rdquo;) mutation types be converted to 'Silent' mutations?</p>
</td></tr>
<tr><td><code id="mutation_types_convert_pave_to_maf_+3A_verbose">verbose</code></td>
<td>
<p>verbose (flag)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matched MAF variant classification terms (character)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mutation_types_convert_pave_to_maf(
  c('upstream_gene_variant', 'stop_lost', 'splice_acceptor_variant')
)
</code></pre>

<hr>
<h2 id='mutation_types_convert_so_to_maf'>Convert SO Mutation Types to MAF</h2><span id='topic+mutation_types_convert_so_to_maf'></span>

<h3>Description</h3>

<p>Convert SO Mutation Types to MAF
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation_types_convert_so_to_maf(
  so_mutation_types,
  variant_type = NULL,
  inframe = NULL,
  split_on_ampersand = TRUE,
  missing_to_silent = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutation_types_convert_so_to_maf_+3A_so_mutation_types">so_mutation_types</code></td>
<td>
<p>a vector of SO terms you want to convert to MAF variant classifications (character)</p>
</td></tr>
<tr><td><code id="mutation_types_convert_so_to_maf_+3A_variant_type">variant_type</code></td>
<td>
<p>a vector describing each mutations type. Valid elements include: &quot;SNP&quot;, &quot;DNP&quot;, &quot;TNP&quot;, &quot;ONP&quot;, &quot;DEL&quot;, &quot;INS&quot;. Used to map frameshift_variant to more specific MAF columns (character)</p>
</td></tr>
<tr><td><code id="mutation_types_convert_so_to_maf_+3A_inframe">inframe</code></td>
<td>
<p>is the mutation inframe? (logical). Used to map protein_altering_variant to valid MAF columns</p>
</td></tr>
<tr><td><code id="mutation_types_convert_so_to_maf_+3A_split_on_ampersand">split_on_ampersand</code></td>
<td>
<p>should '&amp;' separated SO terms be automatically converted to single SO terms based on highest severity? (flag)</p>
</td></tr>
<tr><td><code id="mutation_types_convert_so_to_maf_+3A_missing_to_silent">missing_to_silent</code></td>
<td>
<p>should missing (NA) or empty (&rdquo;) mutation types be converted to 'Silent' mutations?</p>
</td></tr>
<tr><td><code id="mutation_types_convert_so_to_maf_+3A_verbose">verbose</code></td>
<td>
<p>verbose (flag)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matched MAF variant classification terms (character)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mutation_types_convert_so_to_maf(c('INTRAGENIC', 'INTRAGENIC', 'intergenic_region'))
</code></pre>

<hr>
<h2 id='mutation_types_identify'>Identify Mutation Dictionary Used</h2><span id='topic+mutation_types_identify'></span>

<h3>Description</h3>

<p>Looks at variant consequence terms and guesses what mutation dictionary was used.
SO and PAVE dictionaries overlap, meaning an observed set of terms can perfectly match both ontologies.
If this happens, we assume they are SO terms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation_types_identify(
  mutation_types,
  split_on_ampersand = TRUE,
  verbose = TRUE,
  ignore_missing = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutation_types_identify_+3A_mutation_types">mutation_types</code></td>
<td>
<p>mutation types to test (character)</p>
</td></tr>
<tr><td><code id="mutation_types_identify_+3A_split_on_ampersand">split_on_ampersand</code></td>
<td>
<p>split mutation types in a single string separated by ampersand (&amp;) into 2 distinct mutation type columns (flag)</p>
</td></tr>
<tr><td><code id="mutation_types_identify_+3A_verbose">verbose</code></td>
<td>
<p>verbose (flag)</p>
</td></tr>
<tr><td><code id="mutation_types_identify_+3A_ignore_missing">ignore_missing</code></td>
<td>
<p>should we ignore missing (NA) or empty (&rdquo;) mutation_types when identifying a classification scheme (flag)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>one of c('SO', 'MAF', 'UNKNOWN').
Will return 'UNKNOWN' unless ALL mutation types fit with one of the supported dictionaries
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mutation_types_identify(c('bob', 'billy', 'missense_variant'))
</code></pre>

<hr>
<h2 id='mutation_types_maf'>Dictionary of MAF terms</h2><span id='topic+mutation_types_maf'></span>

<h3>Description</h3>

<p>Dictionary of MAF terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation_types_maf()
</code></pre>


<h3>Value</h3>

<p>valid MAF terms (character)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mutation_types_maf()
</code></pre>

<hr>
<h2 id='mutation_types_maf_palette'>Palettes: MAF</h2><span id='topic+mutation_types_maf_palette'></span>

<h3>Description</h3>

<p>Palettes: MAF
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation_types_maf_palette()
</code></pre>


<h3>Value</h3>

<p>named vector. Names are MAF terms. Values are colors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mutation_types_maf_palette()
</code></pre>

<hr>
<h2 id='mutation_types_pave'>Dictionary of PAVE terms</h2><span id='topic+mutation_types_pave'></span>

<h3>Description</h3>

<p>PAVE is a newer annotation software that supports annotaiton of mainly just a subset of SO terms,
but with a couple of important additions to indicate when a non-obvious consequence can be found thanks to phasing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation_types_pave()
</code></pre>


<h3>Value</h3>

<p>valid PAVE terms (character)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mutation_types_pave()
</code></pre>

<hr>
<h2 id='mutation_types_pave_palette'>Palettes: PAVE</h2><span id='topic+mutation_types_pave_palette'></span>

<h3>Description</h3>

<p>Palettes: PAVE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation_types_pave_palette()
</code></pre>


<h3>Value</h3>

<p>named vector. Names are PAVE terms. Values are colors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mutation_types_pave_palette()
</code></pre>

<hr>
<h2 id='mutation_types_so'>Dictionary of So terms</h2><span id='topic+mutation_types_so'></span>

<h3>Description</h3>

<p>Dictionary of So terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation_types_so()
</code></pre>


<h3>Value</h3>

<p>valid SO terms (character)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mutation_types_so()
</code></pre>

<hr>
<h2 id='mutation_types_so_palette'>Palettes: SO</h2><span id='topic+mutation_types_so_palette'></span>

<h3>Description</h3>

<p>Palettes: SO
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mutation_types_so_palette()
</code></pre>


<h3>Value</h3>

<p>named vector. Names are SO terms. Values are colors
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mutation_types_so_palette()
</code></pre>

<hr>
<h2 id='select_most_severe_consequence_pave'>Select the most severe consequence (PAVE)</h2><span id='topic+select_most_severe_consequence_pave'></span>

<h3>Description</h3>

<p>Take a character vector which may contain multiple PAVE mutation types separated by '&amp;'
And choose only the most severe consequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_most_severe_consequence_pave(
  pave_mutation_types,
  missing_is_valid = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_most_severe_consequence_pave_+3A_pave_mutation_types">pave_mutation_types</code></td>
<td>
<p>a character vector of PAVE terms, where multiple pave_mutation_types per field are &amp; delimited, and you want to choose the most severe consequence .</p>
</td></tr>
<tr><td><code id="select_most_severe_consequence_pave_+3A_missing_is_valid">missing_is_valid</code></td>
<td>
<p>should NA values be considered valid mutation classes or should they throw an error? (flag)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the most severe consequence for each element in pave_mutation_types
</p>


<h3>Examples</h3>

<pre><code class='language-R'>select_most_severe_consequence_pave(
 c(
   "upstream_gene_variant&amp;phased_synonymous&amp;5_prime_UTR_variant",
   "missense_variant&amp;frameshift_variant"
 )
)
#&gt; Result:
#&gt; c("phased_synonymous", "frameshift_variant")
</code></pre>

<hr>
<h2 id='select_most_severe_consequence_so'>Select the most severe consequence (SO)</h2><span id='topic+select_most_severe_consequence_so'></span>

<h3>Description</h3>

<p>Take a character vector which may contain multiple so mutation types separated by '&amp;'
And choose only the most severe consequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_most_severe_consequence_so(so_mutation_types, missing_is_valid = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_most_severe_consequence_so_+3A_so_mutation_types">so_mutation_types</code></td>
<td>
<p>a character vector of SO terms, where multiple so_mutation_types per field are &amp; delimited, and you want to choose the most severe consequence .</p>
</td></tr>
<tr><td><code id="select_most_severe_consequence_so_+3A_missing_is_valid">missing_is_valid</code></td>
<td>
<p>should NA values be considered valid mutation classes or should they throw an error? (flag)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the most severe consequence for each element in so_mutation_types
</p>


<h3>Examples</h3>

<pre><code class='language-R'>select_most_severe_consequence_so(
 c(
   "intergenic_variant&amp;feature_truncation&amp;splice_acceptor_variant",
   "initiator_codon_variant&amp;inframe_insertion"
 )
)
#&gt; Result:
#&gt; c("splice_acceptor_variant", "initiator_codon_variant")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
