<!DOCTYPE html><html><head><title>Help for package mutSignatures</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mutSignatures}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mutSignatures-package'><p>Decipher Mutational Signatures from Somatic Mutational Catalogs.</p></a></li>
<li><a href='#[,mutationCounts,numeric,ANY,ANY-method'><p>Subset a mutationCounts-class object.</p></a></li>
<li><a href='#[,mutationSignatures,numeric,ANY,ANY-method'><p>Subset a mutationSignatures-class object.</p></a></li>
<li><a href='#[,mutSignExposures,numeric,ANY,ANY-method'><p>Subset a mutSignExposures-class object.</p></a></li>
<li><a href='#addWeak'><p>Add Weak Mutation TYpes</p></a></li>
<li><a href='#alexaNMF'><p>Perform Non-negative Matrix Factorization using Brunet's Algotithm.</p></a></li>
<li><a href='#as.data.frame,mutationCounts-method'><p>Convert a mutationCounts object to data.frame.</p></a></li>
<li><a href='#as.data.frame,mutationSignatures-method'><p>Convert a mutationSignatures object to data.frame.</p></a></li>
<li><a href='#as.data.frame,mutSignExposures-method'><p>Convert and/or transpose a mutSignExposures object to data.frame.</p></a></li>
<li><a href='#as.list,mutationSignatures-method'><p>Convert a mutationSignatures object to list.</p></a></li>
<li><a href='#as.list,mutFrameworkParams-method'><p>Convert a mutFrameworkParams object to list.</p></a></li>
<li><a href='#as.matrix,mutationCounts-method'><p>Convert a mutationCounts object to matrix.</p></a></li>
<li><a href='#as.mutation.counts'><p>Method as.mutation.counts.</p></a></li>
<li><a href='#as.mutation.signatures'><p>Method as.mutation.signatures.</p></a></li>
<li><a href='#as.mutsign.exposures'><p>Method as.mutsign.exposures.</p></a></li>
<li><a href='#attachContext'><p>Attach Nucleotide Context.</p></a></li>
<li><a href='#attachMutType'><p>Attach Mutation Types.</p></a></li>
<li><a href='#bootstrapCancerGenomes'><p>Bootstrap a Mutation Count Matrix.</p></a></li>
<li><a href='#cbind2'><p>Combine two mutationSignatures-class objects.</p></a></li>
<li><a href='#chihJenNMF'><p>Perform Non-negative Matrix Factorization using Chih-Jen Lin's Algotithm.</p></a></li>
<li><a href='#coerceObj'><p>Method coerceObj.</p></a></li>
<li><a href='#countMutTypes'><p>Count Mutation Types.</p></a></li>
<li><a href='#custom_cssls'><p>Custom CSSLS.</p></a></li>
<li><a href='#custom_fcnnls'><p>Custom Fast Combinatorial Nonnegative Least-Square.</p></a></li>
<li><a href='#decipherMutationalProcesses'><p>Decipher Mutational Processes Contributing to a Collection of Genomic Mutations.</p></a></li>
<li><a href='#deconvoluteMutCounts'><p>Deconvolute Mutation Counts.</p></a></li>
<li><a href='#evaluateStability'><p>Evaluate Results Stability.</p></a></li>
<li><a href='#extractSignatures'><p>Extract Signatures from Genomic Mutational Catalogs.</p></a></li>
<li><a href='#extractXvarlinkData'><p>Extract Variants from XvarlinkData.</p></a></li>
<li><a href='#filterOutIterations'><p>Remove Iterations that Generated Outlier Results.</p></a></li>
<li><a href='#filterSNV'><p>Filter Single Nucleotide Variants.</p></a></li>
<li><a href='#frequencize'><p>Convert Mutation COunts to PerMille Frequencies.</p></a></li>
<li><a href='#getCosmicSignatures'><p>Obtain COSMIC mutational Signatures.</p></a></li>
<li><a href='#getCounts'><p>Method getCounts.</p></a></li>
<li><a href='#getFwkParam'><p>Method getFwkParam.</p></a></li>
<li><a href='#getMutationTypes'><p>Method getMutationTypes.</p></a></li>
<li><a href='#getSampleIdentifiers'><p>Method getSampleIdentifiers.</p></a></li>
<li><a href='#getSignatureIdentifiers'><p>Method getSignatureIdentifiers.</p></a></li>
<li><a href='#getTestRunArgs'><p>Generate Arguments for Running Examples and Mock Runs.</p></a></li>
<li><a href='#importVCFfiles'><p>Import Mutation data from VCF files.</p></a></li>
<li><a href='#leadZeros'><p>Add Leading Zeros to Numbers.</p></a></li>
<li><a href='#matchSignatures'><p>Match Mutational Signatures.</p></a></li>
<li><a href='#msigPlot'><p>Method msigPlot.</p></a></li>
<li><a href='#mutationCounts-class'><p>Class mutationCounts.</p></a></li>
<li><a href='#mutationSignatures-class'><p>Class mutationSignatures.</p></a></li>
<li><a href='#mutFrameworkParams-class'><p>Class mutFrameworkParams.</p></a></li>
<li><a href='#mutSigData'><p>Input Data and Examples for Running Mutational Signatures Analyses</p></a></li>
<li><a href='#mutSignExposures-class'><p>Class mutSignExposures.</p></a></li>
<li><a href='#plotMutTypeProfile'><p>Plot Mutation Signature Profiles.</p></a></li>
<li><a href='#plotSignExposures'><p>Plot Signature Exposure Profiles.</p></a></li>
<li><a href='#prelimProcessAssess'><p>Run a Preliminary Process Assess analysis.</p></a></li>
<li><a href='#processVCFdata'><p>Process VCF data.</p></a></li>
<li><a href='#removeMismatchMut'><p>Remove Mismatched Mutations.</p></a></li>
<li><a href='#removeWeak'><p>Remove Mutation Types Not Meeting the Threshold.</p></a></li>
<li><a href='#resolveMutSignatures'><p>Resolve Mutation Signatures.</p></a></li>
<li><a href='#revCompl'><p>Compute Reverse Complement sequences.</p></a></li>
<li><a href='#setFwkParam'><p>Method setFwkParam.</p></a></li>
<li><a href='#setMutClusterParams'><p>Set Parameters for Extracting Mutational Signatures.</p></a></li>
<li><a href='#show,mutationCounts-method'><p>Show method of the mutationCounts Class.</p></a></li>
<li><a href='#show,mutationSignatures-method'><p>Show method of the mutationSignatures Class.</p></a></li>
<li><a href='#show,mutFrameworkParams-method'><p>Show method of the mutFrameworkParams Class.</p></a></li>
<li><a href='#show,mutSignExposures-method'><p>Show method of the mutSignExposures Class.</p></a></li>
<li><a href='#silhouetteMLB'><p>Silhouette Analysis.</p></a></li>
<li><a href='#simplifySignatures'><p>Simplify Mutational Signatures.</p></a></li>
<li><a href='#sortByMutations'><p>Sort Data by Mutation Type.</p></a></li>
<li><a href='#table2df'><p>Table Mutation Types by Sample.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Decipher Mutational Signatures from Somatic Mutational Catalogs</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-11-01</td>
</tr>
<tr>
<td>Author:</td>
<td>Damiano Fantini, Vania Vidimar, Joshua J Meeks</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Damiano Fantini &lt;damiano.fantini@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Cancer cells accumulate DNA mutations as result of DNA damage and DNA repair processes. This computational framework is aimed at deciphering DNA mutational signatures operating in cancer. The framework includes modules that support raw data import and processing, mutational signature extraction, and results interpretation and visualization. The framework accepts widely used file formats storing information about DNA variants, such as Variant Call Format files. The framework performs Non-Negative Matrix Factorization to extract mutational signatures explaining the observed set of DNA mutations. Bootstrapping is performed as part of the analysis. The framework supports parallelization and is optimized for use on multi-core systems. The software was described by Fantini D et al (2020) &lt;<a href="https://doi.org/10.1038%2Fs41598-020-75062-0">doi:10.1038/s41598-020-75062-0</a>&gt; and is based on a custom R-based implementation of the original MATLAB WTSI framework by Alexandrov LB et al (2013) &lt;<a href="https://doi.org/10.1016%2Fj.celrep.2012.12.008">doi:10.1016/j.celrep.2012.12.008</a>&gt;.   </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5), foreach</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, stats, cluster, doParallel, ggplot2, pracma, proxy,
methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>dplyr, reshape2, kableExtra, gridExtra, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-11-09 04:22:41 UTC; damiano</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-11-09 07:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='mutSignatures-package'>Decipher Mutational Signatures from Somatic Mutational Catalogs.</h2><span id='topic+mutSignatures-package'></span>

<h3>Description</h3>

<p>Cancer cells accumulate DNA mutations as result of DNA damage and DNA repair pro-cesses. 
mutSignatures is a computational framework that is aimed at deciphering DNA mutational 
signatures oper-ating in cancer. The input is a numeric matrix of DNA mutation counts de-tected 
in a panel of cancer samples. The framework performs Non-negative Matrix Factorization to extract 
mutational signatures explaining the observed set of DNA mutations. The framework relies on 
parallelization and is optimized for use on multi-core systems. This framework was described by Fantini D
et al (2020) <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a> and is built upon a custom R-based 
implementation of the original MATLAB WTSI frame-work by Alexandrov LB et al (2013) 
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>. The mutSignatures framework has been described in 
peer-reviewed publications, including Fantini D et al (2018) <a href="https://www.nature.com/articles/s41388-017-0099-6/">https://www.nature.com/articles/s41388-017-0099-6/</a> and 
Fantini D et al (2019) <a href="https://www.sciencedirect.com/science/article/abs/pii/S1078143918303818/">https://www.sciencedirect.com/science/article/abs/pii/S1078143918303818/</a>. The framework includes three modules 
that support raw data import and pre-processing, mutation counts deconvolution, 
and data visualization.
</p>


<h3>References</h3>

<p>More info, examples and vignettes:
</p>

<ol>
<li><p><strong>GitHub Repo</strong>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a> 
</p>
</li>
<li><p><strong>More info and examples</strong> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a> 
</p>
</li>
<li><p><strong>2020 Sci Rep paper</strong> describing the latest version of mutSignatures: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a> 
</p>
</li>
<li><p><strong>Oncogene paper</strong>: Mutational Signatures operative in bladder cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6/">https://www.nature.com/articles/s41388-017-0099-6/</a> 
</p>
</li></ol>


<hr>
<h2 id='+5B+2CmutationCounts+2Cnumeric+2CANY+2CANY-method'>Subset a mutationCounts-class object.</h2><span id='topic++5B+2CmutationCounts+2Cnumeric+2CANY+2CANY-method'></span><span id='topic++5B+2CmutationCounts+2Cnumeric-method'></span>

<h3>Description</h3>

<p>Subset a mutationCounts-class object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutationCounts,numeric,ANY,ANY'
x[i]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2CmutationCounts+2B2Cnumeric+2B2CANY+2B2CANY-method_+3A_x">x</code></td>
<td>
<p>a mutationCounts-class object to subset</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CmutationCounts+2B2Cnumeric+2B2CANY+2B2CANY-method_+3A_i">i</code></td>
<td>
<p>numeric, indeces of the elements to be extracted</p>
</td></tr>
</table>

<hr>
<h2 id='+5B+2CmutationSignatures+2Cnumeric+2CANY+2CANY-method'>Subset a mutationSignatures-class object.</h2><span id='topic++5B+2CmutationSignatures+2Cnumeric+2CANY+2CANY-method'></span><span id='topic++5B+2CmutationSignatures+2Cnumeric-method'></span>

<h3>Description</h3>

<p>Subset a mutationSignatures-class object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutationSignatures,numeric,ANY,ANY'
x[i]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2CmutationSignatures+2B2Cnumeric+2B2CANY+2B2CANY-method_+3A_x">x</code></td>
<td>
<p>a mutationSignatures-class object to subset</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CmutationSignatures+2B2Cnumeric+2B2CANY+2B2CANY-method_+3A_i">i</code></td>
<td>
<p>numeric, indeces of the elements to be extracted</p>
</td></tr>
</table>

<hr>
<h2 id='+5B+2CmutSignExposures+2Cnumeric+2CANY+2CANY-method'>Subset a mutSignExposures-class object.</h2><span id='topic++5B+2CmutSignExposures+2Cnumeric+2CANY+2CANY-method'></span><span id='topic++5B+2CmutSignExposures+2Cnumeric-method'></span>

<h3>Description</h3>

<p>Subset a mutSignExposures-class object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutSignExposures,numeric,ANY,ANY'
x[i]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B2CmutSignExposures+2B2Cnumeric+2B2CANY+2B2CANY-method_+3A_x">x</code></td>
<td>
<p>a mutSignExposures-class object to subset</p>
</td></tr>
<tr><td><code id="+2B5B+2B2CmutSignExposures+2B2Cnumeric+2B2CANY+2B2CANY-method_+3A_i">i</code></td>
<td>
<p>numeric, indeces of the elements to be extracted</p>
</td></tr>
</table>

<hr>
<h2 id='addWeak'>Add Weak Mutation TYpes</h2><span id='topic+addWeak'></span>

<h3>Description</h3>

<p>Restore Mutation Types that were initially excluded because a low number of total counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addWeak(
  mutationTypesToAddSet,
  processes_I,
  processesStd_I,
  Wall_I,
  genomeErrors_I,
  genomesReconstructed_I
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addWeak_+3A_mutationtypestoaddset">mutationTypesToAddSet</code></td>
<td>
<p>Set of mutations to restore</p>
</td></tr>
<tr><td><code id="addWeak_+3A_processes_i">processes_I</code></td>
<td>
<p>Set of Mutational Processes</p>
</td></tr>
<tr><td><code id="addWeak_+3A_processesstd_i">processesStd_I</code></td>
<td>
<p>Set of standard deviations of all Mutational Processes</p>
</td></tr>
<tr><td><code id="addWeak_+3A_wall_i">Wall_I</code></td>
<td>
<p>Set of all W matrices previously extracted</p>
</td></tr>
<tr><td><code id="addWeak_+3A_genomeerrors_i">genomeErrors_I</code></td>
<td>
<p>Set of all residuals</p>
</td></tr>
<tr><td><code id="addWeak_+3A_genomesreconstructed_i">genomesReconstructed_I</code></td>
<td>
<p>Fitted Values according to the most likely Model</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is one of the core functions included in the original mutSignatures R library, 
and in the WTSI MATLAB framework. This is an internal function.
</p>


<h3>Value</h3>

<p>Output is the final result of the deconvolution process
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>


<hr>
<h2 id='alexaNMF'>Perform Non-negative Matrix Factorization using Brunet's Algotithm.</h2><span id='topic+alexaNMF'></span>

<h3>Description</h3>

<p>Perform Non-negative Matrix Factorization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alexaNMF(v, r, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alexaNMF_+3A_v">v</code></td>
<td>
<p>numeric matrix of Mutation Type Counts</p>
</td></tr>
<tr><td><code id="alexaNMF_+3A_r">r</code></td>
<td>
<p>numeric, number of signatures to extract</p>
</td></tr>
<tr><td><code id="alexaNMF_+3A_params">params</code></td>
<td>
<p>list including all paramaters for running the analysis</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is one of the core functions included in the original mutSignatures R library, 
and in the WTSI MATLAB framework. This is an internal function.
</p>


<h3>Value</h3>

<p>list including all paramaters for running the analysis:
</p>

<ol>
<li> <p><b>W</b> extracted signatures
</p>
</li>
<li> <p><b>H</b> contribution of each signature in all the samples of the input mut count matrix
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- mutSignatures:::getTestRunArgs("alexaNMF")
y &lt;- mutSignatures:::alexaNMF(v = x$v, r = x$r, params = x$params)
y$w[1:5, ]

</code></pre>

<hr>
<h2 id='as.data.frame+2CmutationCounts-method'>Convert a mutationCounts object to data.frame.</h2><span id='topic+as.data.frame+2CmutationCounts-method'></span>

<h3>Description</h3>

<p>Coerce a mutationCounts-class object to data.frame by applying the coerceObj method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutationCounts'
as.data.frame(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.data.frame+2B2CmutationCounts-method_+3A_x">x</code></td>
<td>
<p>a mutationCounts object</p>
</td></tr>
</table>

<hr>
<h2 id='as.data.frame+2CmutationSignatures-method'>Convert a mutationSignatures object to data.frame.</h2><span id='topic+as.data.frame+2CmutationSignatures-method'></span>

<h3>Description</h3>

<p>Coerce a mutationSignatures-class object to data.frame by applying the coerceObj method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutationSignatures'
as.data.frame(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.data.frame+2B2CmutationSignatures-method_+3A_x">x</code></td>
<td>
<p>a mutationSignatures object</p>
</td></tr>
</table>

<hr>
<h2 id='as.data.frame+2CmutSignExposures-method'>Convert and/or transpose a mutSignExposures object to data.frame.</h2><span id='topic+as.data.frame+2CmutSignExposures-method'></span>

<h3>Description</h3>

<p>Coerce a mutSignExposures-class object to data.frame by applying the coerceObj method.
The data.frame can be returned in a transposed or non-transposed format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutSignExposures'
as.data.frame(x, row.names = NULL, optional = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.data.frame+2B2CmutSignExposures-method_+3A_x">x</code></td>
<td>
<p>a mutSignExposures object</p>
</td></tr>
<tr><td><code id="as.data.frame+2B2CmutSignExposures-method_+3A_row.names">row.names</code></td>
<td>
<p>NULL, not used</p>
</td></tr>
<tr><td><code id="as.data.frame+2B2CmutSignExposures-method_+3A_optional">optional</code></td>
<td>
<p>NULL, not used</p>
</td></tr>
<tr><td><code id="as.data.frame+2B2CmutSignExposures-method_+3A_...">...</code></td>
<td>
<p>additional parameters to be passed to coerceObj, such as transpose (logical)</p>
</td></tr>
</table>

<hr>
<h2 id='as.list+2CmutationSignatures-method'>Convert a mutationSignatures object to list.</h2><span id='topic+as.list+2CmutationSignatures-method'></span>

<h3>Description</h3>

<p>Coerce a mutationSignatures-class object to list by applying the coerceObj method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutationSignatures'
as.list(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.list+2B2CmutationSignatures-method_+3A_x">x</code></td>
<td>
<p>a mutationSignatures object</p>
</td></tr>
</table>

<hr>
<h2 id='as.list+2CmutFrameworkParams-method'>Convert a mutFrameworkParams object to list.</h2><span id='topic+as.list+2CmutFrameworkParams-method'></span>

<h3>Description</h3>

<p>Coerce a mutFrameworkParams-class object to list by applying the coerceObj method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutFrameworkParams'
as.list(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.list+2B2CmutFrameworkParams-method_+3A_x">x</code></td>
<td>
<p>a mutFrameworkParams object</p>
</td></tr>
</table>

<hr>
<h2 id='as.matrix+2CmutationCounts-method'>Convert a mutationCounts object to matrix.</h2><span id='topic+as.matrix+2CmutationCounts-method'></span>

<h3>Description</h3>

<p>Coerce a mutationCounts-class object to matrix by applying the coerceObj method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutationCounts'
as.matrix(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.matrix+2B2CmutationCounts-method_+3A_x">x</code></td>
<td>
<p>a mutationCounts object</p>
</td></tr>
</table>

<hr>
<h2 id='as.mutation.counts'>Method as.mutation.counts.</h2><span id='topic+as.mutation.counts'></span><span id='topic+as.mutation.counts+2Cdata.frame-method'></span><span id='topic+as.mutation.counts+2Cdata.frame+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>Cast a data.frame into a mutationCounts-class object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.mutation.counts(x, rownames = NULL, colnames = NULL)

## S4 method for signature 'data.frame'
as.mutation.counts(x, rownames = NULL, colnames = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.mutation.counts_+3A_x">x</code></td>
<td>
<p>an object to extract Signature Identifiers from, i.e. a mutSignExposures-class</p>
</td></tr>
<tr><td><code id="as.mutation.counts_+3A_rownames">rownames</code></td>
<td>
<p>character vector to overwrite data row names. Can be NULL if rownames(x) is not NULL.</p>
</td></tr>
<tr><td><code id="as.mutation.counts_+3A_colnames">colnames</code></td>
<td>
<p>character vector to overwrite data column names. Can be NULL if colnames(x) is not NULL.</p>
</td></tr>
</table>

<hr>
<h2 id='as.mutation.signatures'>Method as.mutation.signatures.</h2><span id='topic+as.mutation.signatures'></span><span id='topic+as.mutation.signatures+2Cdata.frame-method'></span>

<h3>Description</h3>

<p>Cast a data.frame into a mutationCounts-class object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.mutation.signatures(x)

## S4 method for signature 'data.frame'
as.mutation.signatures(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.mutation.signatures_+3A_x">x</code></td>
<td>
<p>a data.frame to be converted to a mutationCounts-class object.</p>
</td></tr>
</table>

<hr>
<h2 id='as.mutsign.exposures'>Method as.mutsign.exposures.</h2><span id='topic+as.mutsign.exposures'></span><span id='topic+as.mutsign.exposures+2Cdata.frame+2Clogical-method'></span>

<h3>Description</h3>

<p>Cast a data.frame into a mutSignExposures-class object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.mutsign.exposures(x, samplesAsCols = TRUE)

## S4 method for signature 'data.frame,logical'
as.mutsign.exposures(x, samplesAsCols = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.mutsign.exposures_+3A_x">x</code></td>
<td>
<p>a data.frame to be converted to a mutSignExposures-class object.</p>
</td></tr>
<tr><td><code id="as.mutsign.exposures_+3A_samplesascols">samplesAsCols</code></td>
<td>
<p>logical, are samples listed as columns in the input data.frame. If FALSE, 
samples are expected to be listed as rows in the input data.frame</p>
</td></tr>
</table>

<hr>
<h2 id='attachContext'>Attach Nucleotide Context.</h2><span id='topic+attachContext'></span>

<h3>Description</h3>

<p>Retrieve the nucleotide context around each DNA variant based on the
genomic coordinates of the variant and a reference BSGenome database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>attachContext(
  mutData,
  BSGenomeDb,
  chr_colName = "chr",
  start_colName = "start_position",
  end_colName = "end_position",
  nucl_contextN = 3,
  context_colName = "context"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="attachContext_+3A_mutdata">mutData</code></td>
<td>
<p>data.frame storing mutation data</p>
</td></tr>
<tr><td><code id="attachContext_+3A_bsgenomedb">BSGenomeDb</code></td>
<td>
<p>a BSGenomeDb-class object, storing info about the genome of interest</p>
</td></tr>
<tr><td><code id="attachContext_+3A_chr_colname">chr_colName</code></td>
<td>
<p>string, name of the column storing seqNames. Defaults to &quot;chr&quot;</p>
</td></tr>
<tr><td><code id="attachContext_+3A_start_colname">start_colName</code></td>
<td>
<p>string, name of the column storing start positions. Defaults to &quot;start_position&quot;</p>
</td></tr>
<tr><td><code id="attachContext_+3A_end_colname">end_colName</code></td>
<td>
<p>string, name of the column storing end positions. Defaults to &quot;end_position&quot;</p>
</td></tr>
<tr><td><code id="attachContext_+3A_nucl_contextn">nucl_contextN</code></td>
<td>
<p>integer, the span of nucleotides to be retrieved around the variant. Defaults to 3</p>
</td></tr>
<tr><td><code id="attachContext_+3A_context_colname">context_colName</code></td>
<td>
<p>string, name of the column that will be storing the 
nucleotide context. Defaults to &quot;context&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a modified data.frame including the nucleotide context in a new column
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>


<hr>
<h2 id='attachMutType'>Attach Mutation Types.</h2><span id='topic+attachMutType'></span>

<h3>Description</h3>

<p>Modify a data.frame carrying information about DNA mutation, and add a new column that 
stores formatted multi-nucleotide types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>attachMutType(
  mutData,
  ref_colName = "reference_allele",
  var_colName = "variant_allele",
  var2_colName = NULL,
  context_colName = "context",
  format = 1,
  mutType_dict = "alexa",
  mutType_colName = "mutType"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="attachMutType_+3A_mutdata">mutData</code></td>
<td>
<p>data.frame including information about DNA mutations</p>
</td></tr>
<tr><td><code id="attachMutType_+3A_ref_colname">ref_colName</code></td>
<td>
<p>string, pointing to the column with information about the sequence of the &quot;reference_allele&quot;</p>
</td></tr>
<tr><td><code id="attachMutType_+3A_var_colname">var_colName</code></td>
<td>
<p>string, pointing to the column with information about the sequence of the &quot;variant_allele&quot;</p>
</td></tr>
<tr><td><code id="attachMutType_+3A_var2_colname">var2_colName</code></td>
<td>
<p>string (optional), pointing to the column with information about the 
sequence of a second &quot;variant_allele&quot;. Can be NULL</p>
</td></tr>
<tr><td><code id="attachMutType_+3A_context_colname">context_colName</code></td>
<td>
<p>string, pointing to the column with information about the nucleotidic &quot;context&quot;</p>
</td></tr>
<tr><td><code id="attachMutType_+3A_format">format</code></td>
<td>
<p>integer, indicates the desired mutation type format: (1) N[R&gt;V]N; (2) NN.R&gt;V; (3) R.V[NRN][NVN]</p>
</td></tr>
<tr><td><code id="attachMutType_+3A_muttype_dict">mutType_dict</code></td>
<td>
<p>string, indicates the dictionary to be used for simplifying reverse-complement identical mutation types.
It is recommended to use the standard dictionary from COSMIC, by selecting the default value, i.e. &quot;alexa&quot;.</p>
</td></tr>
<tr><td><code id="attachMutType_+3A_muttype_colname">mutType_colName</code></td>
<td>
<p>string, column name of the new column added to the data.frame where mutTypes are stored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a data.frame including a new column with mutation Types.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'> 
A &lt;- data.frame(REF = c("A", "T", "G"), 
                VAR = c("G", "C", "C"), 
                CTX = c("TAG", "GTG", "CGA"), 
                stringsAsFactors = FALSE)
mutSignatures::attachMutType(mutData = A, ref_colName = "REF", 
                             var_colName = "VAR", context_colName = "CTX")
      
</code></pre>

<hr>
<h2 id='bootstrapCancerGenomes'>Bootstrap a Mutation Count Matrix.</h2><span id='topic+bootstrapCancerGenomes'></span>

<h3>Description</h3>

<p>Rearrange a Mutation count Matrix using the multivariate normal distribution. 
The function returns a bootstrapped Mutation Count matrix whose dimensions 
are identical to the input matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootstrapCancerGenomes(genomes, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bootstrapCancerGenomes_+3A_genomes">genomes</code></td>
<td>
<p>a numeric matrix of Mutation Counts.
Rows correspond to Mutation Types, columns to different samples.</p>
</td></tr>
<tr><td><code id="bootstrapCancerGenomes_+3A_seed">seed</code></td>
<td>
<p>integer, set a seed to obtain reproducible results. Defaulted to NULL</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is one of the core functions included in the original mutSignatures R library, 
and in the WTSI MATLAB framework. This is an internal function.
</p>


<h3>Value</h3>

<p>a numeric matrix of bootstrapped Mutation Counts.
Rows correspond to Mutation Types, columns to different samples.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- cbind(c(10, 100, 20, 200, 30, 5), 
           c(100, 90, 80, 100, 11, 9))
mutSignatures:::bootstrapCancerGenomes(x)                


</code></pre>

<hr>
<h2 id='cbind2'>Combine two mutationSignatures-class objects.</h2><span id='topic+cbind2'></span><span id='topic+cbind2+2CmutationSignatures+2CmutationSignatures-method'></span>

<h3>Description</h3>

<p>Combine two mutationSignatures-class objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutationSignatures,mutationSignatures'
cbind2(x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cbind2_+3A_x">x</code></td>
<td>
<p>the first mutSignExposures-class object to combine</p>
</td></tr>
<tr><td><code id="cbind2_+3A_y">y</code></td>
<td>
<p>the first mutSignExposures-class object to combine</p>
</td></tr>
</table>


<h3>Details</h3>

<p>a variant of this method accepting more than 2 object to combine together is 
under preparation and be available soon...
</p>

<hr>
<h2 id='chihJenNMF'>Perform Non-negative Matrix Factorization using Chih-Jen Lin's Algotithm.</h2><span id='topic+chihJenNMF'></span>

<h3>Description</h3>

<p>Perform Non-negative Matrix Factorization (alternative approach).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chihJenNMF(v, r, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chihJenNMF_+3A_v">v</code></td>
<td>
<p>numeric matrix of Mutation Type Counts</p>
</td></tr>
<tr><td><code id="chihJenNMF_+3A_r">r</code></td>
<td>
<p>numeric, number of signatures to extract</p>
</td></tr>
<tr><td><code id="chihJenNMF_+3A_params">params</code></td>
<td>
<p>list including all paramaters for running the analysis</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a core internal function.
</p>


<h3>Value</h3>

<p>list including all paramaters for running the analysis:
</p>

<ol>
<li> <p><b>W</b> extracted signatures
</p>
</li>
<li> <p><b>H</b> contribution of each signature in all the samples of the input mut count matrix
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li> <p><b>Chih-Jen Lin original paper</b>: <a href="https://ieeexplore.ieee.org/document/4359171/">https://ieeexplore.ieee.org/document/4359171/</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- mutSignatures:::getTestRunArgs("chihJenNMF")
y &lt;- mutSignatures:::chihJenNMF(v = x$v, r = x$r, params = x$params)
y$w[1:10, ]

</code></pre>

<hr>
<h2 id='coerceObj'>Method coerceObj.</h2><span id='topic+coerceObj'></span><span id='topic+coerceObj+2CmutFrameworkParams+2Ccharacter-method'></span><span id='topic+coerceObj+2CmutationSignatures+2Ccharacter-method'></span><span id='topic+coerceObj+2CmutationCounts+2Ccharacter-method'></span><span id='topic+coerceObj+2CmutSignExposures+2Ccharacter-method'></span><span id='topic+coerceObj+2Cdata.frame+2Ccharacter-method'></span>

<h3>Description</h3>

<p>Cast an object to a different format, by extracting and returning the
most appropriate information. Note that data.frames can be coerced to one
of the classes defined in the mutSignatures package using coerceObj.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coerceObj(x, to, ...)

## S4 method for signature 'mutFrameworkParams,character'
coerceObj(x, to)

## S4 method for signature 'mutationSignatures,character'
coerceObj(x, to)

## S4 method for signature 'mutationCounts,character'
coerceObj(x, to, ...)

## S4 method for signature 'mutSignExposures,character'
coerceObj(x, to, ...)

## S4 method for signature 'data.frame,character'
coerceObj(x, to, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coerceObj_+3A_x">x</code></td>
<td>
<p>an object to coerce to a different format</p>
</td></tr>
<tr><td><code id="coerceObj_+3A_to">to</code></td>
<td>
<p>string, indicates the expected format (such as list or data.frame)</p>
</td></tr>
<tr><td><code id="coerceObj_+3A_...">...</code></td>
<td>
<p>additional parameters passed to the functions used for the coercion</p>
</td></tr>
</table>

<hr>
<h2 id='countMutTypes'>Count Mutation Types.</h2><span id='topic+countMutTypes'></span>

<h3>Description</h3>

<p>Analyze a table (data.frame) including mutation counts. Count and aggregate Count Mutation Types. 
If multiple samples are included in the same table, results are aggregated by samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countMutTypes(mutTable, mutType_colName = "mutType", sample_colName = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="countMutTypes_+3A_muttable">mutTable</code></td>
<td>
<p>data.frame including mutation types and an optional sample ID column</p>
</td></tr>
<tr><td><code id="countMutTypes_+3A_muttype_colname">mutType_colName</code></td>
<td>
<p>string, name of the column storing mutTypes</p>
</td></tr>
<tr><td><code id="countMutTypes_+3A_sample_colname">sample_colName</code></td>
<td>
<p>string, name of the column storing sample identifiers. Can be NULL</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a mutationCounts-class object
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- mutSignatures:::getTestRunArgs("countMutTypes")
x
y &lt;- mutSignatures::countMutTypes(mutTable = x, 
                                  mutType_colName = "mutation", 
                                  sample_colName = "sample")
y


</code></pre>

<hr>
<h2 id='custom_cssls'>Custom CSSLS.</h2><span id='topic+custom_cssls'></span>

<h3>Description</h3>

<p>This function contributes to solving a nonnegative least square linear 
problem using normal equations and the fast combinatorial strategy from Van 
Benthem et al. (2004). This implementation is similar to that included in the
NMF R package by Renaud Gaujoux and Cathal Seoighe, 
and it is tailored to the data used in the mutational signature analysis. 
For more info, see: <a href="https://CRAN.R-project.org/package=NMF">https://CRAN.R-project.org/package=NMF</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>custom_cssls(CtC, CtA, Pset)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="custom_cssls_+3A_ctc">CtC</code></td>
<td>
<p>numeric matrix</p>
</td></tr>
<tr><td><code id="custom_cssls_+3A_cta">CtA</code></td>
<td>
<p>numeric matrix</p>
</td></tr>
<tr><td><code id="custom_cssls_+3A_pset">Pset</code></td>
<td>
<p>nueric matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- mutSignatures:::getTestRunArgs(testN = "custom_cssls")
y &lt;- mutSignatures:::custom_cssls(CtC = x$CtC, CtA = x$CtA, Pset = x$Pset)
y
 
</code></pre>

<hr>
<h2 id='custom_fcnnls'>Custom Fast Combinatorial Nonnegative Least-Square.</h2><span id='topic+custom_fcnnls'></span>

<h3>Description</h3>

<p>This function contributes to solve a least square linear 
problem using the fast combinatorial strategy from Van 
Benthem et al. (2004). This implementation is similar to that included in the
NMF R package by Renaud Gaujoux and Cathal Seoighe, 
and it is tailored to the data used in the mutational signature analysis. 
For more info, see: <a href="https://CRAN.R-project.org/package=NMF">https://CRAN.R-project.org/package=NMF</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>custom_fcnnls(mutCounts, signatures)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="custom_fcnnls_+3A_mutcounts">mutCounts</code></td>
<td>
<p>numeric matrix including mutation counts</p>
</td></tr>
<tr><td><code id="custom_fcnnls_+3A_signatures">signatures</code></td>
<td>
<p>numeric matrix including mutation signatures</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list, including: (K) a numeric matrix of estimated exposures; and (Pset) a Pset numeric matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- mutSignatures:::getTestRunArgs(testN = "custom_fcnnls")
y &lt;- mutSignatures:::custom_fcnnls(mutCounts = x$muts, signatures = x$signs)
y$coef
 
</code></pre>

<hr>
<h2 id='decipherMutationalProcesses'>Decipher Mutational Processes Contributing to a Collection of Genomic Mutations.</h2><span id='topic+decipherMutationalProcesses'></span>

<h3>Description</h3>

<p>Decipher Mutational ProCancer cells accumulate DNA mutations as result of DNA 
damage and DNA repair processes. Thiscomputational framework allows to decipher 
mutational processes from cancer-derived somatic mutational catalogs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decipherMutationalProcesses(input, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="decipherMutationalProcesses_+3A_input">input</code></td>
<td>
<p>a mutationCounts-class object, including a mutation counts data.</p>
</td></tr>
<tr><td><code id="decipherMutationalProcesses_+3A_params">params</code></td>
<td>
<p>a mutFrameworkParams-class object including all the parameters required for 
running the mutational signature analysis.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is one of the core functions included in the original mutSignatures R library, 
and in the WTSI MATLAB framework. This is the main user interface for the mutSignatures analysis.
</p>


<h3>Value</h3>

<p>list including all results of the analysis.
The extracted signatures (processes) are included in the &quot;processes&quot; element of the list.
The relative contribution of each signature in each sample is summarized in the 
&quot;exposures&quot; element.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>library(mutSignatures)
x &lt;- mutSignatures:::getTestRunArgs("decipherMutationalProcesses")
x$muts
y &lt;- mutSignatures::decipherMutationalProcesses(input = x$muts, 
                                                params = x$params)
y$Results$signatures

</code></pre>

<hr>
<h2 id='deconvoluteMutCounts'>Deconvolute Mutation Counts.</h2><span id='topic+deconvoluteMutCounts'></span>

<h3>Description</h3>

<p>Characterize mutational signatures from cancer-derived somatic mutational catalogs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deconvoluteMutCounts(input_mutCounts, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deconvoluteMutCounts_+3A_input_mutcounts">input_mutCounts</code></td>
<td>
<p>numeric matrix of Mutation Type Counts</p>
</td></tr>
<tr><td><code id="deconvoluteMutCounts_+3A_params">params</code></td>
<td>
<p>list including all parameters required for running the analysis</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is one of the core functions included in the original mutSignatures R library, 
and in the WTSI MATLAB framework. This is an internal function.
</p>


<h3>Value</h3>

<p>list including all the results from the deconvolution analysis. 
This function is called within thedecipherMutationalProcesses() function 
after parameters and input data have been validated
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'> 
x &lt;- mutSignatures:::getTestRunArgs("deconvoluteMutCounts")
y &lt;- mutSignatures:::deconvoluteMutCounts(input_mutCounts = as.matrix(x$muts),
                                          params = as.list(x$params))
y$processes[1:10,]


</code></pre>

<hr>
<h2 id='evaluateStability'>Evaluate Results Stability.</h2><span id='topic+evaluateStability'></span>

<h3>Description</h3>

<p>Perform a final Stability check comparing the results from all iterations of the analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evaluateStability(wall, hall, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evaluateStability_+3A_wall">wall</code></td>
<td>
<p>numeric matrix including the w results from all the iterations of the analysis</p>
</td></tr>
<tr><td><code id="evaluateStability_+3A_hall">hall</code></td>
<td>
<p>numeric matrix including the h results from all the iterations of the analysis</p>
</td></tr>
<tr><td><code id="evaluateStability_+3A_params">params</code></td>
<td>
<p>list including all the parameters required for running tha analysis</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function evaluates the results from all iterations by performing 
a silhouette check. A silhouette plot will also be plotted.
This is one of the core functions included in the original mutSignatures R library, 
and in the WTSI MATLAB framework. This is an internal function.
</p>


<h3>Value</h3>

<p>list including all results from the stability checks.
This includes the most likely signatures (cen-troids) and exposures. 
All information for plotting the silhoueette plot will also be returned.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'># Obtain sample data
TMP &lt;- mutSignatures:::getTestRunArgs("evaluateStability")
Y &lt;- mutSignatures:::evaluateStability(wall = TMP$W, 
                                       hall = TMP$H, 
                                       params = TMP$params)


</code></pre>

<hr>
<h2 id='extractSignatures'>Extract Signatures from Genomic Mutational Catalogs.</h2><span id='topic+extractSignatures'></span>

<h3>Description</h3>

<p>Extract mutational signatures after the input Data and the 
input parameters have been checked andvalidated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractSignatures(mutCountMatrix, params, bootStrap = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractSignatures_+3A_mutcountmatrix">mutCountMatrix</code></td>
<td>
<p>numeric matrix of mutation counts</p>
</td></tr>
<tr><td><code id="extractSignatures_+3A_params">params</code></td>
<td>
<p>list including all parameters for performing the analysis</p>
</td></tr>
<tr><td><code id="extractSignatures_+3A_bootstrap">bootStrap</code></td>
<td>
<p>logical, shall bootstrapping be performed</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is one of the core functions included in the original mutSignatures R library, 
and in the WTSI MATLAB framework. This is an internal function.
</p>


<h3>Value</h3>

<p>list including the following elements
</p>

<ol>
<li> <p><b>Wall</b>:  all extracted signatures
</p>
</li>
<li> <p><b>Hall</b>: all extracted exposures
</p>
</li>
<li> <p><b>mutCounts.reconstructed</b>: fitted values
</p>
</li>
<li> <p><b>mutCounts.errors</b>: residuals
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- mutSignatures:::getTestRunArgs("extractSignatures")
y &lt;- mutSignatures:::extractSignatures(mutCountMatrix = as.matrix(x$muts), 
                                       params = as.list(x$params), bootStrap = TRUE)
y$Wk[1:10,]


</code></pre>

<hr>
<h2 id='extractXvarlinkData'>Extract Variants from XvarlinkData.</h2><span id='topic+extractXvarlinkData'></span>

<h3>Description</h3>

<p>Extract Variants from data stored as XvarlinkData.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractXvarlinkData(xvarLink_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extractXvarlinkData_+3A_xvarlink_data">xvarLink_data</code></td>
<td>
<p>character vector, including mutation data embedded in XvarlinkData</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a data.frame including mutations as well as corresponding reference nucleotides.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- mutSignatures:::getTestRunArgs("extractXvarlinkData")
y &lt;- mutSignatures:::extractXvarlinkData(xvarLink_data = x)
y

            
</code></pre>

<hr>
<h2 id='filterOutIterations'>Remove Iterations that Generated Outlier Results.</h2><span id='topic+filterOutIterations'></span>

<h3>Description</h3>

<p>Internal function from the WTSI framework, ported to R. This is a core 
function called from within a deconvoluteMutCounts() call. This
function removes iterations that generated outlier results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterOutIterations(wall, hall, cnt_errors, cnt_reconstructed, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterOutIterations_+3A_wall">wall</code></td>
<td>
<p>numeric matrix combining w results from all iterations</p>
</td></tr>
<tr><td><code id="filterOutIterations_+3A_hall">hall</code></td>
<td>
<p>numeric matrix combining h results from all iterations</p>
</td></tr>
<tr><td><code id="filterOutIterations_+3A_cnt_errors">cnt_errors</code></td>
<td>
<p>numeric matrix combining all residuals from all iterations</p>
</td></tr>
<tr><td><code id="filterOutIterations_+3A_cnt_reconstructed">cnt_reconstructed</code></td>
<td>
<p>numeric matrix combining fitted values from all iterations</p>
</td></tr>
<tr><td><code id="filterOutIterations_+3A_params">params</code></td>
<td>
<p>list including alll parameters for running the analysis</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is one of the core functions included in the original mutSignatures R library, 
and in the WTSI MATLAB framework. This is an internal function.
</p>


<h3>Value</h3>

<p>list including all data required for running the subsequent stability check
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>


<hr>
<h2 id='filterSNV'>Filter Single Nucleotide Variants.</h2><span id='topic+filterSNV'></span>

<h3>Description</h3>

<p>Remove entries corresponding to non-SNV, such as insertions and deletions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterSNV(dataSet, seq_colNames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterSNV_+3A_dataset">dataSet</code></td>
<td>
<p>data.frame including variant information</p>
</td></tr>
<tr><td><code id="filterSNV_+3A_seq_colnames">seq_colNames</code></td>
<td>
<p>character vector with the names of the columns storing variant data</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a filtered data.frame only including SNVs
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- mutSignatures:::getTestRunArgs("filterSNV")
nrow(x)
y &lt;- mutSignatures::filterSNV(dataSet = x, 
                              seq_colNames = c("REF", "ALT"))
nrow(y)

                    
</code></pre>

<hr>
<h2 id='frequencize'>Convert Mutation COunts to PerMille Frequencies.</h2><span id='topic+frequencize'></span>

<h3>Description</h3>

<p>Convert Mutation COunts to frequencies. Typically, a permille frequence is returned.
In other words, the resulting number indicates the expected mutation count if the genome hat a 
total of 1000 mutations. This way, the MutSignatures analysis will be
less biased toward the hyper-mutator samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>frequencize(countMatrix, permille = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="frequencize_+3A_countmatrix">countMatrix</code></td>
<td>
<p>numeric matrix of mutation counts</p>
</td></tr>
<tr><td><code id="frequencize_+3A_permille">permille</code></td>
<td>
<p>ligucal, shall the permille conversion be used instead of the standard frequency</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>list including colSums (mutation burden of each sample) and freqs (matrix of frequencies)
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- cbind(c(7, 100, 90, 1000), c(1, 3, 5, 9))
fA &lt;- mutSignatures::frequencize(A)
fA$freqs
 
</code></pre>

<hr>
<h2 id='getCosmicSignatures'>Obtain COSMIC mutational Signatures.</h2><span id='topic+getCosmicSignatures'></span>

<h3>Description</h3>

<p>Obtain latest mutational Signature definitions from COSMIC. FOr more info, please visit: <a href="https://cancer.sanger.ac.uk/cosmic/">https://cancer.sanger.ac.uk/cosmic/</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCosmicSignatures(forceUseMirror = FALSE, asMutSign = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCosmicSignatures_+3A_forceusemirror">forceUseMirror</code></td>
<td>
<p>logical, shall signatures be downloaded from a mirror. Set to TRUE if the COSMIC
server goes down.</p>
</td></tr>
<tr><td><code id="getCosmicSignatures_+3A_asmutsign">asMutSign</code></td>
<td>
<p>logical, shall data be returned as a mutSignatures-class object. Defaults to TRUE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>an object storing COSMIC mutational signature data
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>


<hr>
<h2 id='getCounts'>Method getCounts.</h2><span id='topic+getCounts'></span><span id='topic+getCounts+2CmutationCounts-method'></span>

<h3>Description</h3>

<p>Retrieve mutation counts from an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCounts(x)

## S4 method for signature 'mutationCounts'
getCounts(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCounts_+3A_x">x</code></td>
<td>
<p>an object to extract Mutation counts from, i.e. a mutationCounts-class object</p>
</td></tr>
</table>

<hr>
<h2 id='getFwkParam'>Method getFwkParam.</h2><span id='topic+getFwkParam'></span><span id='topic+getFwkParam+2CmutFrameworkParams+2Ccharacter-method'></span>

<h3>Description</h3>

<p>Retrieve the list of parameters used for running a Mutation Signature Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getFwkParam(x, label)

## S4 method for signature 'mutFrameworkParams,character'
getFwkParam(x, label)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFwkParam_+3A_x">x</code></td>
<td>
<p>a mutFrameworkParams-class object</p>
</td></tr>
<tr><td><code id="getFwkParam_+3A_label">label</code></td>
<td>
<p>string, corresponding to the parameter name to extract</p>
</td></tr>
</table>

<hr>
<h2 id='getMutationTypes'>Method getMutationTypes.</h2><span id='topic+getMutationTypes'></span><span id='topic+getMutationTypes+2CmutationSignatures-method'></span><span id='topic+getMutationTypes+2CmutationCounts-method'></span>

<h3>Description</h3>

<p>Retrieve the list of mutation types from an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMutationTypes(x)

## S4 method for signature 'mutationSignatures'
getMutationTypes(x)

## S4 method for signature 'mutationCounts'
getMutationTypes(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMutationTypes_+3A_x">x</code></td>
<td>
<p>an object to extract Mutation types from, i.e. a mutationSignatures-class
or a mutationCounts-class object</p>
</td></tr>
</table>

<hr>
<h2 id='getSampleIdentifiers'>Method getSampleIdentifiers.</h2><span id='topic+getSampleIdentifiers'></span><span id='topic+getSampleIdentifiers+2CmutationCounts-method'></span><span id='topic+getSampleIdentifiers+2CmutSignExposures-method'></span>

<h3>Description</h3>

<p>Retrieve the list of sample identifiers from an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSampleIdentifiers(x)

## S4 method for signature 'mutationCounts'
getSampleIdentifiers(x = "mutationCounts")

## S4 method for signature 'mutSignExposures'
getSampleIdentifiers(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSampleIdentifiers_+3A_x">x</code></td>
<td>
<p>an object to extract Mutation types from, i.e. a mutationCounts-class
or a mutSignExposures-class object</p>
</td></tr>
</table>

<hr>
<h2 id='getSignatureIdentifiers'>Method getSignatureIdentifiers.</h2><span id='topic+getSignatureIdentifiers'></span><span id='topic+getSignatureIdentifiers+2CmutSignExposures-method'></span><span id='topic+getSignatureIdentifiers+2CmutationSignatures-method'></span>

<h3>Description</h3>

<p>Retrieve the list of signature identifiers from an object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSignatureIdentifiers(x)

## S4 method for signature 'mutSignExposures'
getSignatureIdentifiers(x)

## S4 method for signature 'mutationSignatures'
getSignatureIdentifiers(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSignatureIdentifiers_+3A_x">x</code></td>
<td>
<p>an object to extract Signature Identifiers from, i.e. a mutSignExposures-class
or a mutationSignatures-class object</p>
</td></tr>
</table>

<hr>
<h2 id='getTestRunArgs'>Generate Arguments for Running Examples and Mock Runs.</h2><span id='topic+getTestRunArgs'></span>

<h3>Description</h3>

<p>This function generates objects that can be used for 
running the examples included in the package documentation files, 
as well as some simple mutSignature analyses. Note that his function is not exported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTestRunArgs(testN = "evaluateStability")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTestRunArgs_+3A_testn">testN</code></td>
<td>
<p>string, name of the function that we want to test</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is an internal function.
</p>


<h3>Value</h3>

<p>an object (typically, a list) including sample data to run
analyses or to test mutSignatures functions.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>


<hr>
<h2 id='importVCFfiles'>Import Mutation data from VCF files.</h2><span id='topic+importVCFfiles'></span>

<h3>Description</h3>

<p>Import Mutation data from VCF files. The columns are expected in the following order:
c(&quot;CHROM&quot;, &quot;POS&quot;, &quot;ID&quot;, &quot;REF&quot;, &quot;ALT&quot;, &quot;QUAL&quot;, &quot;FILTER&quot;, &quot;INFO&quot;, &quot;FORMAT&quot;). Optional columns 
can be present to inform about sample ID or other info.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importVCFfiles(vcfFiles, sampleNames = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importVCFfiles_+3A_vcffiles">vcfFiles</code></td>
<td>
<p>character vector, includes the names of the VCF files to be analyzed</p>
</td></tr>
<tr><td><code id="importVCFfiles_+3A_samplenames">sampleNames</code></td>
<td>
<p>character vector with alternative sample names (otherwise, 
VCF file names will be ised to identify each sample).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a concatenated data.frame with all variants found in the input VCF files. Sample
ID is stored in the &quot;SAMPLEID&quot; column.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>


<hr>
<h2 id='leadZeros'>Add Leading Zeros to Numbers.</h2><span id='topic+leadZeros'></span>

<h3>Description</h3>

<p>Internal function to convert a numeric vector into a character vector, where all elements
have the same number of characters (nchar). This is obtained by pasting a series of leading zeros
(or other character) to each number in the input vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>leadZeros(n, m, char = "0", na.value = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="leadZeros_+3A_n">n</code></td>
<td>
<p>numeric vector whose numbers are to be transformed</p>
</td></tr>
<tr><td><code id="leadZeros_+3A_m">m</code></td>
<td>
<p>maximum number that will be used to define how many leading zeros to attach</p>
</td></tr>
<tr><td><code id="leadZeros_+3A_char">char</code></td>
<td>
<p>string (typically, a single character). This character is used to fill the leading space.
Defaults to 0.</p>
</td></tr>
<tr><td><code id="leadZeros_+3A_na.value">na.value</code></td>
<td>
<p>value used to fill mising values. Defaults to NA</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is one of the core functions included in the original mutSignatures R library, 
and in the WTSI MATLAB framework. This is an internal function.
</p>


<h3>Value</h3>

<p>numeric vector of length equal to length(n), where all numbers are converted 
to character and modified by attaching the required number of leading zeros (characters).
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>n = c(5:12, NA, 9:11)
m = 111
mutSignatures:::leadZeros(n=n, m=m)    

</code></pre>

<hr>
<h2 id='matchSignatures'>Match Mutational Signatures.</h2><span id='topic+matchSignatures'></span>

<h3>Description</h3>

<p>Analyze the similarity between mutational signatures from different analyses/runs.
THis function can be helpful to match de novo extracted signatures with previously
described signatures (such as COSMIC), or to reveal signatures that can be 
identified with alternative NMF algorithms, or that may be due to an algorithm bias.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matchSignatures(
  mutSign,
  reference = NULL,
  method = "cosine",
  threshold = 0.5,
  plot = "TRUE"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matchSignatures_+3A_mutsign">mutSign</code></td>
<td>
<p>a mutationSignatures object</p>
</td></tr>
<tr><td><code id="matchSignatures_+3A_reference">reference</code></td>
<td>
<p>a mutationSignatures object. If NULL, COSMIC signatures will be retrieved</p>
</td></tr>
<tr><td><code id="matchSignatures_+3A_method">method</code></td>
<td>
<p>distance method used to compute similarity (1 - distance)</p>
</td></tr>
<tr><td><code id="matchSignatures_+3A_threshold">threshold</code></td>
<td>
<p>signal (similarity) upper threshold for maxing the signal</p>
</td></tr>
<tr><td><code id="matchSignatures_+3A_plot">plot</code></td>
<td>
<p>logical, shall a heatmap be plotted</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>list, including distance matrix and a heatmap plot
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>


<hr>
<h2 id='msigPlot'>Method msigPlot.</h2><span id='topic+msigPlot'></span><span id='topic+msigPlot+2CmutationSignatures-method'></span><span id='topic+msigPlot+2CmutationCounts-method'></span><span id='topic+msigPlot+2CmutSignExposures-method'></span>

<h3>Description</h3>

<p>Generate standard plots using data from mutsignature-class objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msigPlot(x, ...)

## S4 method for signature 'mutationSignatures'
msigPlot(x, ...)

## S4 method for signature 'mutationCounts'
msigPlot(x, ...)

## S4 method for signature 'mutSignExposures'
msigPlot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="msigPlot_+3A_x">x</code></td>
<td>
<p>a mutSignatures object</p>
</td></tr>
<tr><td><code id="msigPlot_+3A_...">...</code></td>
<td>
<p>additional parameters, including standard graphical parameters, as well as 
a set of class-specific arguments:
* x is a mutationSignatures object: 
- signature, numeric; numeric index of the signature to display
- main, string; title of the plot
* x is a mutationCounts object
- sample, numeric or string, i.e. the identifier or the index of the sample to be plotted
- main, string, title of the plot
* x is a mutSignExposures object
- top, integer, the maximum number of samples to be plotted</p>
</td></tr>
</table>

<hr>
<h2 id='mutationCounts-class'>Class mutationCounts.</h2><span id='topic+mutationCounts-class'></span><span id='topic+initialize+2CmutationCounts-method'></span>

<h3>Description</h3>

<p>Class mutationCounts defines objects storing Mutation COunts data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutationCounts'
initialize(.Object, x, muts, samples)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutationCounts-class_+3A_.object">.Object</code></td>
<td>
<p>the mutationCounts object being built</p>
</td></tr>
<tr><td><code id="mutationCounts-class_+3A_x">x</code></td>
<td>
<p>data.frame including mutation count values for each biological sample</p>
</td></tr>
<tr><td><code id="mutationCounts-class_+3A_muts">muts</code></td>
<td>
<p>data.frame including information about mutation types</p>
</td></tr>
<tr><td><code id="mutationCounts-class_+3A_samples">samples</code></td>
<td>
<p>data.frame including information about sample identifiers (unique names)</p>
</td></tr>
</table>


<h3>Slots</h3>


<dl>
<dt><code>counts</code></dt><dd><p>data.frame including information about mutation counts</p>
</dd>
<dt><code>mutTypes</code></dt><dd><p>data.frame including information about mutation types</p>
</dd>
<dt><code>sampleId</code></dt><dd><p>data.frame including information about sample identifiers</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Damiano Fantini <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>

<hr>
<h2 id='mutationSignatures-class'>Class mutationSignatures.</h2><span id='topic+mutationSignatures-class'></span><span id='topic+initialize+2CmutationSignatures-method'></span>

<h3>Description</h3>

<p>Class mutationSignatures defines objects storing Mutational Signatures data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutationSignatures'
initialize(.Object, x, muts, signNames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutationSignatures-class_+3A_.object">.Object</code></td>
<td>
<p>the mutationSignatures object being built</p>
</td></tr>
<tr><td><code id="mutationSignatures-class_+3A_x">x</code></td>
<td>
<p>data.frame including fequency data of multiple mutation signatures</p>
</td></tr>
<tr><td><code id="mutationSignatures-class_+3A_muts">muts</code></td>
<td>
<p>data.frame including information about mutation types</p>
</td></tr>
<tr><td><code id="mutationSignatures-class_+3A_signnames">signNames</code></td>
<td>
<p>data.frame including information about mutation 
signature names (unique identifiers)</p>
</td></tr>
</table>


<h3>Slots</h3>


<dl>
<dt><code>mutationFreq</code></dt><dd><p>data.frame including information about mutation frequencies</p>
</dd>
<dt><code>mutTypes</code></dt><dd><p>data.frame including information about mutation types</p>
</dd>
<dt><code>signatureId</code></dt><dd><p>data.frame including information about mutation signature Identifiers</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Damiano Fantini <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>

<hr>
<h2 id='mutFrameworkParams-class'>Class mutFrameworkParams.</h2><span id='topic+mutFrameworkParams-class'></span><span id='topic+initialize+2CmutFrameworkParams-method'></span>

<h3>Description</h3>

<p>Class mutFrameworkParams defines objects including the set of 
parameters used for running a Mutational Signature Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutFrameworkParams'
initialize(.Object, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutFrameworkParams-class_+3A_.object">.Object</code></td>
<td>
<p>the mutFrameworkParams object being built</p>
</td></tr>
<tr><td><code id="mutFrameworkParams-class_+3A_params">params</code></td>
<td>
<p>list including values for a set of mutFramework params</p>
</td></tr>
</table>


<h3>Slots</h3>


<dl>
<dt><code>params</code></dt><dd><p>list including the set of parameters used for running a Mutational Signature Analysis</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Damiano Fantini <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>

<hr>
<h2 id='mutSigData'>Input Data and Examples for Running Mutational Signatures Analyses</h2><span id='topic+mutSigData'></span>

<h3>Description</h3>

<p>A series of objects, including collections of DNA mutations from 50 Bladder cancer samples, 
as well as mutational signatures extracted from the same samples. Mutation catalogs were obtained
from a TCGA bladder cancer dataset (data available from the BROAD Institute). 
Original sample IDs were shuffled and then re-encoded. 
Data are available in different formats, and can be used as input for running mutational signature analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("mutSigData")
</code></pre>


<h3>Format</h3>

<p>A list with 6 elements. Each element is a different type of <code>mutSignatures</code> input/data:
</p>

<dl>
<dt>inputA</dt><dd><p>data.frame with 10401 rows and 4 columns. DNA mutation data mimicking a TCGA dataset downloaded using TCGAretriever/cBio</p>
</dd>
<dt>inputB</dt><dd><p>data.frame with 13523 rows and 12 columns. DNA mutation data mimicking a TCGA MAF file</p>
</dd>
<dt>inputC</dt><dd><p>data.frame with 13523 rows and 11 columns. DNA mutation data mimicking a VCF file decorated with a SAMPLEID column</p>
</dd>
<dt>inputC.ctx</dt><dd><p>data.frame with 13523 rows and 11 columns. DNA mutation data mimicking a VCF file decorated with a SAMPLEID column</p>
</dd>
<dt>inputD</dt><dd><p>data.frame with 13487 rows and 56 columns. DNA mutation data mimicking a set of VCF files casted into a 2D matrix (samples as columns)</p>
</dd>
<dt>inputS</dt><dd><p>list including data for silhouette plot generation (used in the vignette)</p>
</dd>
<dt>blcaMUTS</dt><dd><p>data.frame with 96 rows and 50 columns. A table of DNA mutation counts (rows are mutation types; columns are samples) </p>
</dd>
<dt>blcaSIGS</dt><dd><p>data.frame with 96 rows and 8 columns. Set of 8 mutational signatures (rows are mutation types; columns are signatures) </p>
</dd>
<dt>.addON</dt><dd><p>list of add-on functions (executed only upon request, not evaluated; these may require manual installation of external libraries from Bioconductor or GitHUB)</p>
</dd>
</dl>



<h3>Details</h3>

<p>Examples and more information are available in the vignette, as well as at the following URL: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>


<h3>Source</h3>

<p>BLCA data were downloaded from <a href="http://gdac.broadinstitute.org/">http://gdac.broadinstitute.org/</a> and then further processed, modified, and formatted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mutSigData)
print(mutSigData$input.A[1:6,])
</code></pre>

<hr>
<h2 id='mutSignExposures-class'>Class mutSignExposures.</h2><span id='topic+mutSignExposures-class'></span><span id='topic+initialize+2CmutSignExposures-method'></span>

<h3>Description</h3>

<p>Class mutSignExposures defines objects storing information about Exposures of 
biological samples to Mutational Signatures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutSignExposures'
initialize(.Object, x, samples, signNames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutSignExposures-class_+3A_.object">.Object</code></td>
<td>
<p>the mutSignExposures object being built</p>
</td></tr>
<tr><td><code id="mutSignExposures-class_+3A_x">x</code></td>
<td>
<p>data.frame including numeric values of exposures to mutational signatures</p>
</td></tr>
<tr><td><code id="mutSignExposures-class_+3A_samples">samples</code></td>
<td>
<p>data.frame including information about biological sample identifiers (unique names)</p>
</td></tr>
<tr><td><code id="mutSignExposures-class_+3A_signnames">signNames</code></td>
<td>
<p>data.frame including information about mutational signature identifiers</p>
</td></tr>
</table>


<h3>Slots</h3>


<dl>
<dt><code>exposures</code></dt><dd><p>data.frame including information about exposures</p>
</dd>
<dt><code>sampleId</code></dt><dd><p>data.frame including information about sample identifiers</p>
</dd>
<dt><code>signatureId</code></dt><dd><p>data.frame including information about signature identifiers</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Damiano Fantini <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>

<hr>
<h2 id='plotMutTypeProfile'>Plot Mutation Signature Profiles.</h2><span id='topic+plotMutTypeProfile'></span>

<h3>Description</h3>

<p>Build a barplot to visualize the relative abundance of mutation counts in a mutational
signature or biological sample of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMutTypeProfile(
  mutCounts,
  mutLabs,
  freq = TRUE,
  ylim = "auto",
  ylab = "Fraction of Variants",
  xlab = "Sequence Motifs",
  xaxis_cex = 0.475,
  cols = c("#4eb3d3", "#040404", "#b30000", "#bdbdbd", "#41ab5d", "#dd3497"),
  main = "MutType Profile"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotMutTypeProfile_+3A_mutcounts">mutCounts</code></td>
<td>
<p>data.frame including mutation types counts or frequencies, such as a
data.frame of mutation counts from samples, or mutation type frequencies from a mutational signature.</p>
</td></tr>
<tr><td><code id="plotMutTypeProfile_+3A_mutlabs">mutLabs</code></td>
<td>
<p>character vector, labels to be used for the mutation types</p>
</td></tr>
<tr><td><code id="plotMutTypeProfile_+3A_freq">freq</code></td>
<td>
<p>logical, shall frequency be plotted rather than counts. Defaults to TRUE</p>
</td></tr>
<tr><td><code id="plotMutTypeProfile_+3A_ylim">ylim</code></td>
<td>
<p>values used for ylim. Defaults to &quot;auto&quot; (ylim automatically set)</p>
</td></tr>
<tr><td><code id="plotMutTypeProfile_+3A_ylab">ylab</code></td>
<td>
<p>string, used as y-axis title. Defaults to &quot;Fraction of Variants&quot;</p>
</td></tr>
<tr><td><code id="plotMutTypeProfile_+3A_xlab">xlab</code></td>
<td>
<p>string, used as x-axis title. Defaults to &quot;Sequence Motifs&quot;</p>
</td></tr>
<tr><td><code id="plotMutTypeProfile_+3A_xaxis_cex">xaxis_cex</code></td>
<td>
<p>numeric, cex value for the xaxis</p>
</td></tr>
<tr><td><code id="plotMutTypeProfile_+3A_cols">cols</code></td>
<td>
<p>character vector, indicates the colors to be used for the bars. It typically requires 6 colors.</p>
</td></tr>
<tr><td><code id="plotMutTypeProfile_+3A_main">main</code></td>
<td>
<p>string, tutle of the plot. Defaults to &quot;MutType Profile&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>NULL. A plot is printed to the active device.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>


<hr>
<h2 id='plotSignExposures'>Plot Signature Exposure Profiles.</h2><span id='topic+plotSignExposures'></span>

<h3>Description</h3>

<p>Build a barplot to visualize exposures to mutation signatures.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSignExposures(mutCount, top = 50)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSignExposures_+3A_mutcount">mutCount</code></td>
<td>
<p>a data.frame including mutation Counts</p>
</td></tr>
<tr><td><code id="plotSignExposures_+3A_top">top</code></td>
<td>
<p>integer, max number of samples to include in the plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a plot (ggplot2 object)
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li></ol>


<hr>
<h2 id='prelimProcessAssess'>Run a Preliminary Process Assess analysis.</h2><span id='topic+prelimProcessAssess'></span>

<h3>Description</h3>

<p>This function is an attempt to analyze the relationship between error and k. In other words, 
the goal of prelimProcessAssess is to visualize the reduction in the error/residuals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prelimProcessAssess(
  input,
  maxProcess = 6,
  approach = "counts",
  plot = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prelimProcessAssess_+3A_input">input</code></td>
<td>
<p>a mutationCounts-class object</p>
</td></tr>
<tr><td><code id="prelimProcessAssess_+3A_maxprocess">maxProcess</code></td>
<td>
<p>integer, maximum k to test</p>
</td></tr>
<tr><td><code id="prelimProcessAssess_+3A_approach">approach</code></td>
<td>
<p>sting, &quot;counts&quot; or &quot;freq&quot;</p>
</td></tr>
<tr><td><code id="prelimProcessAssess_+3A_plot">plot</code></td>
<td>
<p>logical, shall a plot be printed to the active device</p>
</td></tr>
<tr><td><code id="prelimProcessAssess_+3A_verbose">verbose</code></td>
<td>
<p>logical, info about the ongoing analysis be messaged/printed to console</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a data.frame showing the estimated total error with respect to the range of k values
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>


<hr>
<h2 id='processVCFdata'>Process VCF data.</h2><span id='topic+processVCFdata'></span>

<h3>Description</h3>

<p>Check, annotate, and process variants imported from a list of VCF files, so that it can be used
to run a mutational signature analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processVCFdata(
  vcfData,
  BSGenomeDb,
  chr_colName = "CHROM",
  pos_colName = "POS",
  ref_colName = "REF",
  alt_colName = "ALT",
  sample_colName = NULL,
  nucl_contextN = 3,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processVCFdata_+3A_vcfdata">vcfData</code></td>
<td>
<p>data.frame, includes mutation data from 2 or more samples</p>
</td></tr>
<tr><td><code id="processVCFdata_+3A_bsgenomedb">BSGenomeDb</code></td>
<td>
<p>a BSGenomeDb-class object storing the genomic sequences and coordinates</p>
</td></tr>
<tr><td><code id="processVCFdata_+3A_chr_colname">chr_colName</code></td>
<td>
<p>string, name of the column including the chromosome (seq) name. Defaults to &quot;CHROM&quot;</p>
</td></tr>
<tr><td><code id="processVCFdata_+3A_pos_colname">pos_colName</code></td>
<td>
<p>string, name of the column including the genomic coordinates/position. Defaults to &quot;POS&quot;</p>
</td></tr>
<tr><td><code id="processVCFdata_+3A_ref_colname">ref_colName</code></td>
<td>
<p>string, name of the column including the reference nucleotide. Defaults to &quot;REF&quot;</p>
</td></tr>
<tr><td><code id="processVCFdata_+3A_alt_colname">alt_colName</code></td>
<td>
<p>string, name of the column including the variant nucleotide. Defaults to &quot;ALT&quot;</p>
</td></tr>
<tr><td><code id="processVCFdata_+3A_sample_colname">sample_colName</code></td>
<td>
<p>string, name of the column including the sample ID. Can be NULL</p>
</td></tr>
<tr><td><code id="processVCFdata_+3A_nucl_contextn">nucl_contextN</code></td>
<td>
<p>integer, span (in nucelotides) of the context around the variants. Defaults to 3</p>
</td></tr>
<tr><td><code id="processVCFdata_+3A_verbose">verbose</code></td>
<td>
<p>logical, shall information about the ongoing analysis be printed to console</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a data.frame including processed variants from VCF files
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>


<hr>
<h2 id='removeMismatchMut'>Remove Mismatched Mutations.</h2><span id='topic+removeMismatchMut'></span>

<h3>Description</h3>

<p>Remove mutation types that do not match the expected nucleotidic context.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeMismatchMut(
  mutData,
  refMut_colName = "mutation",
  context_colName = "context",
  refMut_format = "N&gt;N"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeMismatchMut_+3A_mutdata">mutData</code></td>
<td>
<p>data.frame including mutation data, as well as the nucleotide
context around the mutated position</p>
</td></tr>
<tr><td><code id="removeMismatchMut_+3A_refmut_colname">refMut_colName</code></td>
<td>
<p>string, name of the column storing REF and VAR data. Defaults to &quot;N&gt;N&quot;</p>
</td></tr>
<tr><td><code id="removeMismatchMut_+3A_context_colname">context_colName</code></td>
<td>
<p>string, name of the column storing nucleotide context around the variant.</p>
</td></tr>
<tr><td><code id="removeMismatchMut_+3A_refmut_format">refMut_format</code></td>
<td>
<p>string, format of mutation types. Defaults to &quot;N&gt;N&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>filtered data.frame
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- mutSignatures:::getTestRunArgs("removeMismatchMut")
y &lt;- mutSignatures:::removeMismatchMut(x, 
                                       refMut_colName = "REF", 
                                       context_colName = "context", 
                                       refMut_format = "N")
y

</code></pre>

<hr>
<h2 id='removeWeak'>Remove Mutation Types Not Meeting the Threshold.</h2><span id='topic+removeWeak'></span>

<h3>Description</h3>

<p>Remove mutation types that account for a total number of mutations below a defined threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeWeak(input_mutCounts, params)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeWeak_+3A_input_mutcounts">input_mutCounts</code></td>
<td>
<p>numeric matrix of Mutation Counts</p>
</td></tr>
<tr><td><code id="removeWeak_+3A_params">params</code></td>
<td>
<p>object (list) including all parameters required for running the analysis</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is one of the core functions included in the original mutSignatures R library, 
and in the WTSI MATLAB framework. This is an internal function.
</p>


<h3>Value</h3>

<p>List including two elements:
</p>

<ol>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- mutSignatures:::getTestRunArgs("removeWeak")
nrow(x$data)
y &lt;- mutSignatures:::removeWeak(input_mutCounts = x$data, params = x$params)
nrow(y)
  
</code></pre>

<hr>
<h2 id='resolveMutSignatures'>Resolve Mutation Signatures.</h2><span id='topic+resolveMutSignatures'></span>

<h3>Description</h3>

<p>If Mutation signatures are known (such as COSMIC signatures), we can
estimate the contribution of each signature in different samples.
This functions used a matrix of mutation counts and a matrix of mutation
signatures, and estimates Exposures to Mutational Signature of each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resolveMutSignatures(mutCountData, signFreqData, byFreq = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="resolveMutSignatures_+3A_mutcountdata">mutCountData</code></td>
<td>
<p>object storing mutation counts</p>
</td></tr>
<tr><td><code id="resolveMutSignatures_+3A_signfreqdata">signFreqData</code></td>
<td>
<p>object storing mutation signatures</p>
</td></tr>
<tr><td><code id="resolveMutSignatures_+3A_byfreq">byFreq</code></td>
<td>
<p>logical, shall exposures be estimated on per_mille normalized counts</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a list of objects including data about exposures to mutational signatures
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>    x &lt;- mutSignatures:::getTestRunArgs("resolveMutSignatures")
    y &lt;- mutSignatures::resolveMutSignatures(mutCountData = x$muts, signFreqData = x$sigs)
    y


</code></pre>

<hr>
<h2 id='revCompl'>Compute Reverse Complement sequences.</h2><span id='topic+revCompl'></span>

<h3>Description</h3>

<p>Transform a DNA sequence into its reverse-complement sequence.
ALternatively, only the reverse sequence (or only the complement) can be returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>revCompl(DNAseq, rev = TRUE, compl = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="revCompl_+3A_dnaseq">DNAseq</code></td>
<td>
<p>character vector of DNA sequences</p>
</td></tr>
<tr><td><code id="revCompl_+3A_rev">rev</code></td>
<td>
<p>logical, shall the reverse sequence be computed</p>
</td></tr>
<tr><td><code id="revCompl_+3A_compl">compl</code></td>
<td>
<p>logical, shall the complementary sequence be computed</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>a character vector including transformed DNA sequences
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- c("TAACCG", "CTCGA", "CNNA")
mutSignatures::revCompl(A)

</code></pre>

<hr>
<h2 id='setFwkParam'>Method setFwkParam.</h2><span id='topic+setFwkParam'></span><span id='topic+setFwkParam+2CmutFrameworkParams+2Ccharacter-method'></span><span id='topic+setFwkParam+2CmutFrameworkParams+2Ccharacter+2CANY-method'></span>

<h3>Description</h3>

<p>Set or update one of the parameters in a mutFrameworkParams-class object. 
Individual paramaters can be set or updated, by passing the parameter label, and the corresponding
parameter value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setFwkParam(x, label, value)

## S4 method for signature 'mutFrameworkParams,character'
setFwkParam(x, label, value)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setFwkParam_+3A_x">x</code></td>
<td>
<p>an object to extract Signature Identifiers from, i.e. a mutSignExposures-class</p>
</td></tr>
<tr><td><code id="setFwkParam_+3A_label">label</code></td>
<td>
<p>string corresponding to the parameter label to be updated</p>
</td></tr>
<tr><td><code id="setFwkParam_+3A_value">value</code></td>
<td>
<p>new value (string or numeric) of the parameter to be updated</p>
</td></tr>
</table>

<hr>
<h2 id='setMutClusterParams'>Set Parameters for Extracting Mutational Signatures.</h2><span id='topic+setMutClusterParams'></span>

<h3>Description</h3>

<p>Create an object including all parameters required for running the mutSignatures framework.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setMutClusterParams(
  num_processesToExtract = 2,
  num_totIterations = 10,
  num_parallelCores = 1,
  thresh_removeWeakMutTypes = 0.01,
  thresh_removeLastPercent = 0.07,
  distanceFunction = "cosine",
  num_totReplicates = 100,
  eps = 2.2204e-16,
  stopconv = 20000,
  niter = 1e+06,
  guided = TRUE,
  debug = FALSE,
  approach = "freq",
  stopRule = "DF",
  algorithm = "brunet",
  logIterations = "lite",
  seed = 12345
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setMutClusterParams_+3A_num_processestoextract">num_processesToExtract</code></td>
<td>
<p>integer, number of mutational signatures to extract</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_num_totiterations">num_totIterations</code></td>
<td>
<p>integer, total number of iterations (bootstrapping)</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_num_parallelcores">num_parallelCores</code></td>
<td>
<p>integer, number of cores to use for the analysis</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_thresh_removeweakmuttypes">thresh_removeWeakMutTypes</code></td>
<td>
<p>numeric, threshold for filtering out under-represented mutation types</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_thresh_removelastpercent">thresh_removeLastPercent</code></td>
<td>
<p>numeric, threshold for removing outlier iteration results</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_distancefunction">distanceFunction</code></td>
<td>
<p>string, method for calculating distances. Default method is &quot;cosine&quot;</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_num_totreplicates">num_totReplicates</code></td>
<td>
<p>integer, number of replicates while checking stability</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_eps">eps</code></td>
<td>
<p>numeric, close-to-zero positive numeric value for replacing zeros and 
preventing negative values to appear in the matrix during NMF</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_stopconv">stopconv</code></td>
<td>
<p>integer, max number of stable iterations before termination. Defaults to 20000.</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_niter">niter</code></td>
<td>
<p>integer, max number of iterations to run. Defaults to 1000000</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_guided">guided</code></td>
<td>
<p>logical, shall clustering be guided to improve aggregation upon bootstrapping</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_debug">debug</code></td>
<td>
<p>logical, shall the analysis be run in DEBUG mode</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_approach">approach</code></td>
<td>
<p>string, indicating whether to model absolute counts (&quot;counts&quot;) or 
per_mille frequency (&quot;freq&quot;). Defaults to &quot;freq&quot;.</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_stoprule">stopRule</code></td>
<td>
<p>= string, use the sub-optimal termination rule (&quot;AL&quot;) from the WTSI package 
(actually, iterations won't terminate, so niter will most certainly reached) or our 
efficient termination rule (&quot;DF&quot;). Defaults to &quot;DF&quot;. The &quot;AL&quot; option is implemented for 
compatibility reasons, but not recommended.</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_algorithm">algorithm</code></td>
<td>
<p>string, algorithm to be used for NMF. Set to &quot;brunet&quot;, or &quot;alexa&quot; for using the standard algorithm (Brunet's), 
otherwise the alternative &quot;chihjen&quot; algorithm will be used.</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_logiterations">logIterations</code></td>
<td>
<p>string indicating if storing and returining all intermediates, 
or only final results. Defaults to &quot;lite&quot;, i.e. returns full output and limited intermediates.
Alternatively, set to &quot;full&quot;.</p>
</td></tr>
<tr><td><code id="setMutClusterParams_+3A_seed">seed</code></td>
<td>
<p>integer, seed to set for reproducibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object including all parameters for running the analysis
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li><p> WTSI framework: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3588146/</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>library(mutSignatures) 
# defaults params
A &lt;- setMutClusterParams()
A
# A second example, set num_processes
B &lt;- setMutClusterParams(num_processesToExtract = 5)
B

    
</code></pre>

<hr>
<h2 id='show+2CmutationCounts-method'>Show method of the mutationCounts Class.</h2><span id='topic+show+2CmutationCounts-method'></span><span id='topic+print+2CmutationCounts-method'></span>

<h3>Description</h3>

<p>Show method of the mutationCounts Class.
</p>
<p>Print method of the mutationCounts Class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutationCounts'
show(object)

## S4 method for signature 'mutationCounts'
print(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2CmutationCounts-method_+3A_object">object</code></td>
<td>
<p>the mutationCounts object being shown</p>
</td></tr>
<tr><td><code id="show+2B2CmutationCounts-method_+3A_x">x</code></td>
<td>
<p>the mutationCounts object being printed</p>
</td></tr>
</table>

<hr>
<h2 id='show+2CmutationSignatures-method'>Show method of the mutationSignatures Class.</h2><span id='topic+show+2CmutationSignatures-method'></span><span id='topic+print+2CmutationSignatures-method'></span>

<h3>Description</h3>

<p>Show method of the mutationSignatures Class.
</p>
<p>Print method of the mutationSignatures Class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutationSignatures'
show(object)

## S4 method for signature 'mutationSignatures'
print(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2CmutationSignatures-method_+3A_object">object</code></td>
<td>
<p>the mutationSignatures object being shown</p>
</td></tr>
<tr><td><code id="show+2B2CmutationSignatures-method_+3A_x">x</code></td>
<td>
<p>the mutationSignatures object being printed</p>
</td></tr>
</table>

<hr>
<h2 id='show+2CmutFrameworkParams-method'>Show method of the mutFrameworkParams Class.</h2><span id='topic+show+2CmutFrameworkParams-method'></span><span id='topic+print+2CmutFrameworkParams-method'></span>

<h3>Description</h3>

<p>Show method of the mutFrameworkParams Class.
</p>
<p>Print method of the mutFrameworkParams Class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutFrameworkParams'
show(object)

## S4 method for signature 'mutFrameworkParams'
print(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2CmutFrameworkParams-method_+3A_object">object</code></td>
<td>
<p>the mutFrameworkParams object being shown</p>
</td></tr>
<tr><td><code id="show+2B2CmutFrameworkParams-method_+3A_x">x</code></td>
<td>
<p>the mutFrameworkParams object being printed</p>
</td></tr>
</table>

<hr>
<h2 id='show+2CmutSignExposures-method'>Show method of the mutSignExposures Class.</h2><span id='topic+show+2CmutSignExposures-method'></span><span id='topic+print+2CmutSignExposures-method'></span>

<h3>Description</h3>

<p>Show method of the mutSignExposures Class.
</p>
<p>Print method of the mutSignExposures Class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'mutSignExposures'
show(object)

## S4 method for signature 'mutSignExposures'
print(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2CmutSignExposures-method_+3A_object">object</code></td>
<td>
<p>the mutSignExposures object being shown</p>
</td></tr>
<tr><td><code id="show+2B2CmutSignExposures-method_+3A_x">x</code></td>
<td>
<p>the mutSignExposures object being printed</p>
</td></tr>
</table>

<hr>
<h2 id='silhouetteMLB'>Silhouette Analysis.</h2><span id='topic+silhouetteMLB'></span>

<h3>Description</h3>

<p>Analyze the clustering quality and generate a Silhouette Plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>silhouetteMLB(data, fac, method = "cosine", plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="silhouetteMLB_+3A_data">data</code></td>
<td>
<p>numeric matrix</p>
</td></tr>
<tr><td><code id="silhouetteMLB_+3A_fac">fac</code></td>
<td>
<p>clustering factor</p>
</td></tr>
<tr><td><code id="silhouetteMLB_+3A_method">method</code></td>
<td>
<p>method to be used as distance function. Defaults to c(&quot;cosine&quot;)</p>
</td></tr>
<tr><td><code id="silhouetteMLB_+3A_plot">plot</code></td>
<td>
<p>logical, shall a barplot showing the cluster silhouettes be printed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector including the silhouette values of the data poointts in the input matrix
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li>
<li> <p><b>Silhouette analysis in R</b>: <a href="http://www.biotechworld.it/bioinf/2017/01/20/translating-matlabs-silhouette-function-to-r/">http://www.biotechworld.it/bioinf/2017/01/20/translating-matlabs-silhouette-function-to-r/</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>library(mutSignatures)
x &lt;- mutSignatures:::getTestRunArgs("silhouetteMLB")
y &lt;- silhouetteMLB(data = x$data, fac = x$fac)
y

</code></pre>

<hr>
<h2 id='simplifySignatures'>Simplify Mutational Signatures.</h2><span id='topic+simplifySignatures'></span>

<h3>Description</h3>

<p>This function is useufl when working with non-standard muation types, such as
tetra-nnucleotide mutation types or mutation types with long/complex context.
THe goal of this function is to aggregated together mutations that can be simplified 
because of a common mutation core.
For example, mutation types AA[A&gt;T]A, TA[A&gt;T]A, CA[A&gt;T]A, and GA[A&gt;T]A can be simplified to the
core tri-nucleotide mutation A[A&gt;T]A. THis function identifies mergeable mutation types, 
and aggregates the corresponding counts/freqs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simplifySignatures(x, asMutationSignatures = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simplifySignatures_+3A_x">x</code></td>
<td>
<p>a mutationSignatures-class object</p>
</td></tr>
<tr><td><code id="simplifySignatures_+3A_asmutationsignatures">asMutationSignatures</code></td>
<td>
<p>logical, shall the results be returned as a mutationSignatures-class object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>object including simplified mutational signatures data
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- data.frame(Sig1=1:5, Sig2=5:1, Sig3=1:5)
A &lt;- A/apply(A, 2, sum)
rownames(A) &lt;- c("AA[C&gt;A]A", "CA[C&gt;A]A", "TA[C&gt;A]A", "TA[C&gt;G]A", "A[C&gt;G]AT")
A &lt;- mutSignatures::as.mutation.signatures(A)
mutSignatures::simplifySignatures(x = A)


</code></pre>

<hr>
<h2 id='sortByMutations'>Sort Data by Mutation Type.</h2><span id='topic+sortByMutations'></span>

<h3>Description</h3>

<p>Reorder a mutationSignatures, mutationCounts, data.frame, or matrix object by
sorting entries by mutation type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sortByMutations(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sortByMutations_+3A_x">x</code></td>
<td>
<p>an object storing mutation count data</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>an object of the same class as x, with entries sorted according to mutation types.
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>GitHub Repo</b>: <a href="https://github.com/dami82/mutSignatures/">https://github.com/dami82/mutSignatures/</a>
</p>
</li>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- data.frame(S1=1:5, S2=5:1, S3=1:5)
rownames(A) &lt;- c("A[A&gt;T]G", "A[C&gt;G]G", "T[A&gt;T]G", "T[C&gt;G]T", "T[C&gt;G]G")
mutSignatures::sortByMutations(A)

</code></pre>

<hr>
<h2 id='table2df'>Table Mutation Types by Sample.</h2><span id='topic+table2df'></span>

<h3>Description</h3>

<p>Prepare a molten data.frame starting from a mutation count matrix.
Mutation types (rows) are countes for each sample (cols). The results are returned in a
3-column data.frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>table2df(dataMatrix, rowLab = "sample", colLab = "feature", valueLab = "count")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="table2df_+3A_datamatrix">dataMatrix</code></td>
<td>
<p>a numeric matrix including mutation counts</p>
</td></tr>
<tr><td><code id="table2df_+3A_rowlab">rowLab</code></td>
<td>
<p>string, name for the column that will be storing row IDs, typically sample IDs</p>
</td></tr>
<tr><td><code id="table2df_+3A_collab">colLab</code></td>
<td>
<p>string, name for the column that will be storing column IDs, typically sample IDs</p>
</td></tr>
<tr><td><code id="table2df_+3A_valuelab">valueLab</code></td>
<td>
<p>string, name for the column that will be storing mutation count values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is part of the user-interface set of tools included in mutSignatures. This is an exported function.
</p>


<h3>Value</h3>

<p>data.frame storing mutation counts by sample
</p>


<h3>Author(s)</h3>

<p>Damiano Fantini, <a href="mailto:damiano.fantini@gmail.com">damiano.fantini@gmail.com</a>
</p>


<h3>References</h3>

<p>More information and examples about mutational signature analysis can be found here:
</p>

<ol>
<li> <p><b>More info and examples</b> about the mutSignatures R library: <a href="https://www.data-pulse.com/dev_site/mutsignatures/">https://www.data-pulse.com/dev_site/mutsignatures/</a>
</p>
</li>
<li> <p><b>Sci Rep paper</b>, introducing mutS: <a href="https://www.nature.com/articles/s41598-020-75062-0/">https://www.nature.com/articles/s41598-020-75062-0/</a>
</p>
</li>
<li> <p><b>Oncogene paper</b>, Mutational Signatures Operative in Bladder Cancer: <a href="https://www.nature.com/articles/s41388-017-0099-6">https://www.nature.com/articles/s41388-017-0099-6</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>A &lt;- cbind(`A&gt;G`=c(5,10),`A&gt;T`=c(3,20),`A&gt;C`=c(15,0))
rownames(A) = c("Smpl1", "Smpl2")
mutSignatures::table2df(A)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
