<!DOCTYPE html><html><head><title>Help for package MVisAGe</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MVisAGe}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cn.mat'><p>DNA copy number data from 98 head and neck squamous cell carcinoma (HNSC) patients</p></a></li>
<li><a href='#cn.region.heatmap'><p>A Function for Creating a Heatmap of DNA Copy Number Data</p></a></li>
<li><a href='#corr.compute'><p>A Function for Computing a Vector of Pearson Correlation Coefficients</p></a></li>
<li><a href='#corr.list.compute'><p>A Function for Creating a List of Pearson Correlation Coefficients</p></a></li>
<li><a href='#data.prep'><p>A Function for Preparing mRNAseq and Copy Number Data Matrices</p></a></li>
<li><a href='#exp.mat'><p>Gene expression data from 100 head and neck squamous cell carcinoma (HNSC) patients</p></a></li>
<li><a href='#gene.annot'><p>Gene annotation data (hg38)</p></a></li>
<li><a href='#perm.significance'><p>A Function for Computing a Vector of Pearson Correlation Coefficients</p></a></li>
<li><a href='#perm.significance.list.compute'><p>A Function for Creating a List of Pearson Correlation Coefficients</p></a></li>
<li><a href='#sample.annot'><p>Sample annotation data</p></a></li>
<li><a href='#smooth.genome.plot'><p>A Function for Plotting Smoothed Pearson Correlation Coefficients Across Multiple Chromosomes</p></a></li>
<li><a href='#smooth.region.plot'><p>A Function for Plotting Smoothed Pearson Correlation Coefficients Genomewide</p></a></li>
<li><a href='#tcga.cn.convert'><p>A Function for Reformatting TCGA DNA Copy Number Matrices</p></a></li>
<li><a href='#tcga.exp.convert'><p>A Function for Reformatting TCGA mRNA Expression Matrices</p></a></li>
<li><a href='#unsmooth.region.plot'><p>A Function for Plotting Pearson Correlation Coefficients in a Given Genomic Region</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Compute and Visualize Bivariate Associations</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Pearson and Spearman correlation coefficients are commonly used to quantify the strength
	of bivariate associations of genomic variables.  For example, correlations of gene-level 
	DNA copy number and gene expression measurements may be used to assess the impact of 
	DNA copy number changes on gene expression in tumor tissue.  'MVisAGe' enables users to 
	quickly compute and visualize the correlations in order to assess the effect of regional 
	genomic events such as changes in DNA copy number or DNA methylation level.  Please see
	Walter V, Du Y, Danilova L, Hayward MC, Hayes DN, 2018. Cancer Research 
	&lt;<a href="https://doi.org/10.1158%2F0008-5472.CAN-17-3464">doi:10.1158/0008-5472.CAN-17-3464</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-05-10 12:52:37 UTC; vwalter</td>
</tr>
<tr>
<td>Author:</td>
<td>Vonn Walter [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Vonn Walter &lt;vwalter1@pennstatehealth.psu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-05-10 21:43:30 UTC</td>
</tr>
</table>
<hr>
<h2 id='cn.mat'>DNA copy number data from 98 head and neck squamous cell carcinoma (HNSC) patients</h2><span id='topic+cn.mat'></span>

<h3>Description</h3>

<p>Quantitative gene-level DNA copy number measurements for 98 samples from The
Cancer Genome Atlas (TCGA) HNSC cohort.  The all_data_by_genes.txt dataset from
the GISTIC2 output was restricted to the first 100 columns and genes that lie
on chromosomes 11 and 12.  Genes appear in rows; samples appear in columns (other
than the first two columns described below).  Gene symbols are used as row
names and sample identifiers are used as column names (other than the first two columns).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cn.mat
</code></pre>


<h3>Format</h3>

<p>A matrix with 2719 rows and 100 columns
</p>

<dl>
<dt>Locus.ID</dt><dd><p>gene identifier</p>
</dd>
<dt>Cytoband</dt><dd><p>cytoband containing the gene of interest</p>
</dd>
<dt>Remaining Columns</dt><dd><p>quantitative DNA copy number</p>
</dd>
<dt>Column names</dt><dd><p>sample identifiers (other than the first two columns)</p>
</dd>
<dt>Row names</dt><dd><p>gene symbols</p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="https://gdac.broadinstitute.org/">https://gdac.broadinstitute.org/</a>
</p>

<hr>
<h2 id='cn.region.heatmap'>A Function for Creating a Heatmap of DNA Copy Number Data</h2><span id='topic+cn.region.heatmap'></span>

<h3>Description</h3>

<p>This function creates a heatmap of DNA copy number data for a given chromosomal region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cn.region.heatmap(cn.mat, gene.annot, plot.chr, plot.start, plot.stop,
  plot.list, sample.annot = NULL, sample.cluster = F, low.thresh = -2,
  high.thresh = 2, num.cols = 50, collist = c("blue", "white", "red"),
  annot.colors = c("black", "red", "green", "blue", "cyan"),
  plot.sample.annot = F, cytoband.colors = c("gray90", "gray60"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cn.region.heatmap_+3A_cn.mat">cn.mat</code></td>
<td>
<p>A matrix of gene-level DNA copy number data (rows = genes, columns = samples).
DNA methylation data can also be used.  Both row names (gene names) and column names (Sample IDs) must be given.</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_gene.annot">gene.annot</code></td>
<td>
<p>A three-column matrix containing gene position information.  Column 1 = chromosome number written in 
the form 'chr1' (note that chrX and chrY should be written chr23 and chr24), Column 2 = position (in base pairs), Column 3 = cytoband.</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_plot.chr">plot.chr</code></td>
<td>
<p>The chromosome used to define the region of interest.</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_plot.start">plot.start</code></td>
<td>
<p>The genomic position (in base pairs) where the region starts.</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_plot.stop">plot.stop</code></td>
<td>
<p>The genomic position (in base pairs) where the region stops.</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_plot.list">plot.list</code></td>
<td>
<p>A list produced by corr.list.compute().</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_sample.annot">sample.annot</code></td>
<td>
<p>An optional two-column matrix of sample annotation data.  Column 1 = sample IDs, Column 2 = sample annotation 
(e.g. tumor vs. normal).  If NULL, sample annot will be created using the common sample IDs and a single group ('1').  Default = NULL.</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_sample.cluster">sample.cluster</code></td>
<td>
<p>Logical values indicating whether the samples should be clustered.  Default = FALSE.</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_low.thresh">low.thresh</code></td>
<td>
<p>Lower threshold for DNA copy number measurements.  All values less than low.thresh are set equal to low.thresh.  Default = -2.</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_high.thresh">high.thresh</code></td>
<td>
<p>Upper threshold for DNA copy number measurements.  All values greater than high.thresh are set equal to high.thresh.  Default = 2.</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_num.cols">num.cols</code></td>
<td>
<p>Number of distinct colors in the heatmap.  Default = 50.</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_collist">collist</code></td>
<td>
<p>Color scheme for displaying copy number values.  Default = (&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;).</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_annot.colors">annot.colors</code></td>
<td>
<p>Character vector used to define the color scheme for sample annotation.  Default = c(&quot;black&quot;, &quot;red&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;cyan&quot;).</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_plot.sample.annot">plot.sample.annot</code></td>
<td>
<p>Logical value used to specify whether the sample annotation information should be plotted.  Default = FALSE.</p>
</td></tr>
<tr><td><code id="cn.region.heatmap_+3A_cytoband.colors">cytoband.colors</code></td>
<td>
<p>Character vector of length two used to define the color scheme for annotating the cytoband.  Default = c(&quot;gray90&quot;, &quot;gray60&quot;).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 prepped.data = data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

 pd.exp = prepped.data[["exp"]]

 pd.cn = prepped.data[["cn"]]

 pd.ga = prepped.data[["gene.annot"]]

 pd.sa = prepped.data[["sample.annot"]]

 output.list = corr.list.compute(pd.exp, pd.cn, pd.ga, pd.sa)

 cn.region.heatmap(cn.mat = pd.cn, gene.annot = pd.ga, plot.chr = 11,

plot.start = 0e6, plot.stop = 135e6, sample.annot = pd.sa, plot.list = output.list)

</code></pre>

<hr>
<h2 id='corr.compute'>A Function for Computing a Vector of Pearson Correlation Coefficients</h2><span id='topic+corr.compute'></span>

<h3>Description</h3>

<p>This function computes Pearson correlation coefficients on a row-by-row basis for two numerical input matrices of the same size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corr.compute(exp.mat, cn.mat, gene.annot, method = "pearson", digits = 5,
  alternative = "greater")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="corr.compute_+3A_exp.mat">exp.mat</code></td>
<td>
<p>A matrix of gene-level expression data (rows = genes, columns = samples).  Missing values are not permitted.</p>
</td></tr>
<tr><td><code id="corr.compute_+3A_cn.mat">cn.mat</code></td>
<td>
<p>A matrix of gene-level DNA copy number data (rows = genes, columns = samples).  Both genes and samples should
appear in the same order as exp.mat.  Missing values are not permitted.</p>
</td></tr>
<tr><td><code id="corr.compute_+3A_gene.annot">gene.annot</code></td>
<td>
<p>A three-column matrix containing gene position information.  Column 1 = chromosome number written in 
the form 'chr1' (note that chrX and chrY should be written chr23 and chr24), Column 2 = position (in base pairs), Column 3 = cytoband.
Genes should appear in the same order as exp.mat and cn.mat.</p>
</td></tr>
<tr><td><code id="corr.compute_+3A_method">method</code></td>
<td>
<p>A character string (either &quot;pearson&quot; or &quot;spearman&quot;) specifying the method used to calculate the correlation coefficient
(default = &quot;pearson&quot;).</p>
</td></tr>
<tr><td><code id="corr.compute_+3A_digits">digits</code></td>
<td>
<p>Used with signif() to specify the number of significant digits (default = 5).</p>
</td></tr>
<tr><td><code id="corr.compute_+3A_alternative">alternative</code></td>
<td>
<p>A character string (&quot;greater&quot; or &quot;less&quot;) that specifies the direction of the alternative hypothesis, 
either rho &gt; 0 or rho &lt; 0 (default = &quot;greater&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a eight-column matrix.  The first three columns are the same as gene.annot.  The fourth column contains
gene-specific Pearson or Spearman correlation coefficients based on the entries in each row of exp.mat and cn.mat, 
respectively (column name = &quot;R&quot;).  The fifth column contains squared Pearson correlation coefficients (column name = &quot;R^2&quot;).  
The sixth column contains t statistics corresponding to the correlation coefficients (column name = &quot;tStat&quot;).  The
seventh column contains the right-tailed p-value based on the t statistic (column name = &quot;pValue&quot;).  The eighth column
contains Benjamini-Hochberg q-values corresponding to the p-values.  Genes with constant gene expression
or DNA copy number are removed because they have zero variance.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>corr.results = exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 prepped.data = data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

 corr.compute(prepped.data[["exp"]], prepped.data[["cn"]], prepped.data[["gene.annot"]])

</code></pre>

<hr>
<h2 id='corr.list.compute'>A Function for Creating a List of Pearson Correlation Coefficients</h2><span id='topic+corr.list.compute'></span>

<h3>Description</h3>

<p>This function uses the corr.compute() function to compute gene-specific Pearson correlation coefficients in each
group of samples defined in a sample annotation matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corr.list.compute(exp.mat, cn.mat, gene.annot, sample.annot = NULL,
  method = "pearson", digits = 5, alternative = "greater")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="corr.list.compute_+3A_exp.mat">exp.mat</code></td>
<td>
<p>A matrix of gene-level expression data (rows = genes, columns = samples).  Missing values are not permitted.</p>
</td></tr>
<tr><td><code id="corr.list.compute_+3A_cn.mat">cn.mat</code></td>
<td>
<p>A matrix of gene-level DNA copy number data (rows = genes, columns = samples).  Both genes and samples should
appear in the same order as exp.mat.  Missing values are not permitted.</p>
</td></tr>
<tr><td><code id="corr.list.compute_+3A_gene.annot">gene.annot</code></td>
<td>
<p>A three-column matrix containing gene position information.  Column 1 = chromosome number written in 
the form 'chr1' (note that chrX and chrY should be written chr23 and chr24), Column 2 = position (in base pairs), Column 3 = cytoband.
Genes should appear in the same order as exp.mat and cn.mat.</p>
</td></tr>
<tr><td><code id="corr.list.compute_+3A_sample.annot">sample.annot</code></td>
<td>
<p>An optional two-column matrix of sample annotation data.  Column 1 = sample IDs, Column 2 = sample annotation 
(e.g. tumor vs. normal).  If NULL, sample annot will be created using the common sample IDs and a single group ('1').  Default = NULL.</p>
</td></tr>
<tr><td><code id="corr.list.compute_+3A_method">method</code></td>
<td>
<p>A character string (either &quot;pearson&quot; or &quot;spearman&quot;) specifying the method used to calculate the correlation coefficient
(default = &quot;pearson&quot;).</p>
</td></tr>
<tr><td><code id="corr.list.compute_+3A_digits">digits</code></td>
<td>
<p>Used with signif() to specify the number of significant digits (default = 5).</p>
</td></tr>
<tr><td><code id="corr.list.compute_+3A_alternative">alternative</code></td>
<td>
<p>A character string (&quot;greater&quot; or &quot;less&quot;) that specifies the direction of the alternative hypothesis, 
either rho &gt; 0 or rho &lt; 0 (default = &quot;greater&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list whose length is the number of unique groups defined by sample.annot.  Each entry in the list is the
output of corr.compute.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 prepped.data = data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

 pd.exp = prepped.data[["exp"]]

 pd.cn = prepped.data[["cn"]]

 pd.ga = prepped.data[["gene.annot"]]

 pd.sa = prepped.data[["sample.annot"]]

 corr.list.compute(pd.exp, pd.cn, pd.ga, pd.sa)

</code></pre>

<hr>
<h2 id='data.prep'>A Function for Preparing mRNAseq and Copy Number Data Matrices</h2><span id='topic+data.prep'></span>

<h3>Description</h3>

<p>This function prepares mRNAseq and copy number data matrices for use in other mVisAGe functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data.prep(exp.mat, cn.mat, gene.annot, sample.annot = NULL, log.exp = FALSE,
  gene.list = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="data.prep_+3A_exp.mat">exp.mat</code></td>
<td>
<p>A matrix of gene-level expression data (rows = genes, columns = samples).
Both row names (gene names) and column names (sample IDs) must be given.</p>
</td></tr>
<tr><td><code id="data.prep_+3A_cn.mat">cn.mat</code></td>
<td>
<p>A matrix of gene-level DNA copy number data (rows = genes, columns = samples).
DNA methylation data can also be used.  Both row names (gene names) and column names (Sample IDs) must be given.</p>
</td></tr>
<tr><td><code id="data.prep_+3A_gene.annot">gene.annot</code></td>
<td>
<p>A three-column matrix containing gene position information.  Column 1 = chromosome number written in 
the form 'chr1' (note that chrX and chrY should be written chr23 and chr24), Column 2 = position (in base pairs), Column 3 = cytoband.</p>
</td></tr>
<tr><td><code id="data.prep_+3A_sample.annot">sample.annot</code></td>
<td>
<p>An optional two-column matrix of sample annotation data.  Column 1 = sample IDs, Column 2 = categorical sample annotation 
(e.g. tumor vs. normal).  If NULL, sample annot will be created using the common sample IDs and a single group ('1').  Default = NULL.</p>
</td></tr>
<tr><td><code id="data.prep_+3A_log.exp">log.exp</code></td>
<td>
<p>A logical value indicating whether or not the expression values have been log-transformed.  Default = FALSE.</p>
</td></tr>
<tr><td><code id="data.prep_+3A_gene.list">gene.list</code></td>
<td>
<p>Used to restrict the output to a set of genes of interest, e.g. genes identified by GISTIC as having
recurrent copy number alterations.  Default = NULL, and in this case all genes are used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with four components:  cn, exp, gene.annot, and sample.annot.  Each of cn, exp, and gene.annot have been restricted 
to a common set of genes, and these appear in the same order.  Similarly, cn, exp, and sample.annot have been restricted to a common set
of subjects that appear in the same order.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

</code></pre>

<hr>
<h2 id='exp.mat'>Gene expression data from 100 head and neck squamous cell carcinoma (HNSC) patients</h2><span id='topic+exp.mat'></span>

<h3>Description</h3>

<p>RSEM gene expression measurements for 100 samples from The Cancer Genome Atlas (TCGA)
HNSC cohort after restricting to genes that lie in chromosomes 11 and 12.  Genes
appear in rows; samples appear in columns (other than the first two columns described 
below).  Gene symbols are used as row names and sample identifiers are used as 
column names (other than the first two columns).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exp.mat
</code></pre>


<h3>Format</h3>

<p>A matrix with 2161 rows and 100 columns
</p>

<dl>
<dt>Columns</dt><dd><p>RSEM gene expression measurements</p>
</dd>
<dt>Column names</dt><dd><p>sample identifiers</p>
</dd>
<dt>Row names</dt><dd><p>gene symbols</p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="https://gdc.cancer.gov/">https://gdc.cancer.gov/</a>
</p>

<hr>
<h2 id='gene.annot'>Gene annotation data (hg38)</h2><span id='topic+gene.annot'></span>

<h3>Description</h3>

<p>Genomic position and cytoband annotation data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene.annot
</code></pre>


<h3>Format</h3>

<p>A matrix with 26417 rows and 3 columns
</p>

<dl>
<dt>chr</dt><dd><p>chromosome containing the gene of interest</p>
</dd>
<dt>pos</dt><dd><p>genomic position (base pairs) for the gene of interest</p>
</dd>
<dt>cytoband</dt><dd><p>cytoband containing the gene of interest</p>
</dd>
<dt>Row names</dt><dd><p>gene symbols</p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="https://usegalaxy.org/">https://usegalaxy.org/</a>
</p>

<hr>
<h2 id='perm.significance'>A Function for Computing a Vector of Pearson Correlation Coefficients</h2><span id='topic+perm.significance'></span>

<h3>Description</h3>

<p>This function computes Pearson correlation coefficients on a row-by-row basis for two numerical input matrices of the same size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perm.significance(exp.mat, cn.mat, gene.annot, method = "pearson",
  digits = 5, num.perms = 100, random.seed = NULL,
  alternative = "greater")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perm.significance_+3A_exp.mat">exp.mat</code></td>
<td>
<p>A matrix of gene-level expression data (rows = genes, columns = samples).  Missing values are not permitted.</p>
</td></tr>
<tr><td><code id="perm.significance_+3A_cn.mat">cn.mat</code></td>
<td>
<p>A matrix of gene-level DNA copy number data (rows = genes, columns = samples).  Both genes and samples should
appear in the same order as exp.mat.  Missing values are not permitted.</p>
</td></tr>
<tr><td><code id="perm.significance_+3A_gene.annot">gene.annot</code></td>
<td>
<p>A three-column matrix containing gene position information.  Column 1 = chromosome number written in 
the form 'chr1' (note that chrX and chrY should be written chr23 and chr24), Column 2 = position (in base pairs), Column 3 = cytoband.
Genes should appear in the same order as exp.mat and cn.mat.</p>
</td></tr>
<tr><td><code id="perm.significance_+3A_method">method</code></td>
<td>
<p>A character string (either &quot;pearson&quot; or &quot;spearman&quot;) specifying the method used to calculate the correlation coefficient
(default = &quot;pearson&quot;).</p>
</td></tr>
<tr><td><code id="perm.significance_+3A_digits">digits</code></td>
<td>
<p>Used with signif() to specify the number of significant digits (default = 5).</p>
</td></tr>
<tr><td><code id="perm.significance_+3A_num.perms">num.perms</code></td>
<td>
<p>Number of permutations used to assess significance (default = 1e2).</p>
</td></tr>
<tr><td><code id="perm.significance_+3A_random.seed">random.seed</code></td>
<td>
<p>Random seed (default = NULL).</p>
</td></tr>
<tr><td><code id="perm.significance_+3A_alternative">alternative</code></td>
<td>
<p>A character string (&quot;greater&quot; or &quot;less&quot;) that specifies the direction of the alternative hypothesis, 
either rho &gt; 0 or rho &lt; 0 (default = &quot;greater&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a five-column matrix.  The first three columns are the same as gene.annot.  The fourth column contains
gene-specific Pearson or Spearman correlation coefficients based on the entries in each row of exp.mat and cn.mat, 
respectively (column name = &quot;R&quot;).  The fifth column contains squared Pearson correlation coefficients (column name = &quot;R^2&quot;).  
The sixth column contains the permutation-based right-tailed p-value of the correlation coefficient (column name = &quot;perm_pValue&quot;).
The seventh column contains Benjamini-Hochberg q-values corresponding to the p-values.  Genes with constant gene expression
or DNA copy number are removed because they have zero variance.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 prepped.data = data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

 perm.significance(prepped.data[["exp"]], prepped.data[["cn"]], prepped.data[["gene.annot"]])

</code></pre>

<hr>
<h2 id='perm.significance.list.compute'>A Function for Creating a List of Pearson Correlation Coefficients</h2><span id='topic+perm.significance.list.compute'></span>

<h3>Description</h3>

<p>This function uses the corr.compute() function to compute gene-specific Pearson correlation coefficients in each
group of samples defined in a sample annotation matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perm.significance.list.compute(exp.mat, cn.mat, gene.annot,
  sample.annot = NULL, method = "pearson", digits = 5, num.perms = 100,
  random.seed = NULL, alternative = "greater")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perm.significance.list.compute_+3A_exp.mat">exp.mat</code></td>
<td>
<p>A matrix of gene-level expression data (rows = genes, columns = samples).  Missing values are not permitted.</p>
</td></tr>
<tr><td><code id="perm.significance.list.compute_+3A_cn.mat">cn.mat</code></td>
<td>
<p>A matrix of gene-level DNA copy number data (rows = genes, columns = samples).  Both genes and samples should
appear in the same order as exp.mat.  Missing values are not permitted.</p>
</td></tr>
<tr><td><code id="perm.significance.list.compute_+3A_gene.annot">gene.annot</code></td>
<td>
<p>A three-column matrix containing gene position information.  Column 1 = chromosome number written in 
the form 'chr1' (note that chrX and chrY should be written chr23 and chr24), Column 2 = position (in base pairs), Column 3 = cytoband.
Genes should appear in the same order as exp.mat and cn.mat.</p>
</td></tr>
<tr><td><code id="perm.significance.list.compute_+3A_sample.annot">sample.annot</code></td>
<td>
<p>An optional two-column matrix of sample annotation data.  Column 1 = sample IDs, Column 2 = sample annotation 
(e.g. tumor vs. normal).  If NULL, sample annot will be created using the common sample IDs and a single group ('1').  Default = NULL.</p>
</td></tr>
<tr><td><code id="perm.significance.list.compute_+3A_method">method</code></td>
<td>
<p>A character string (either &quot;pearson&quot; or &quot;spearman&quot;) specifying the method used to calculate the correlation coefficient
(default = &quot;pearson&quot;).</p>
</td></tr>
<tr><td><code id="perm.significance.list.compute_+3A_digits">digits</code></td>
<td>
<p>Used with signif() to specify the number of significant digits (default = 5).</p>
</td></tr>
<tr><td><code id="perm.significance.list.compute_+3A_num.perms">num.perms</code></td>
<td>
<p>Number of permutations used to assess significance (default = 1e2).</p>
</td></tr>
<tr><td><code id="perm.significance.list.compute_+3A_random.seed">random.seed</code></td>
<td>
<p>Random seed (default = NULL).</p>
</td></tr>
<tr><td><code id="perm.significance.list.compute_+3A_alternative">alternative</code></td>
<td>
<p>A character string (&quot;greater&quot; or &quot;less&quot;) that specifies the direction of the alternative hypothesis, 
either rho &gt; 0 or rho &lt; 0 (default = &quot;greater&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list whose length is the number of unique groups defined by sample.annot.  Each entry in the list is the
output of perm.significance.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 prepped.data = data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

 pd.exp = prepped.data[["exp"]]

 pd.cn = prepped.data[["cn"]]

 pd.ga = prepped.data[["gene.annot"]]

 pd.sa = prepped.data[["sample.annot"]]

 perm.significance.list.compute(pd.exp, pd.cn, pd.ga, pd.sa)

</code></pre>

<hr>
<h2 id='sample.annot'>Sample annotation data</h2><span id='topic+sample.annot'></span>

<h3>Description</h3>

<p>Human papillomavirus (HPV) infection status for the n = 279 patients with head and neck
squamous cell carcinoma in the The Cancer Genoma Atlas cohort.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample.annot
</code></pre>


<h3>Format</h3>

<p>A matrix with 26417 rows and 3 columns
</p>

<dl>
<dt>Barcode</dt><dd><p>sample identifier</p>
</dd>
<dt>New.HPV.Status</dt><dd><p>HPV infection status</p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="http://www.nature.com/nature/journal/v517/n7536/full/nature14129.html">http://www.nature.com/nature/journal/v517/n7536/full/nature14129.html</a>
</p>

<hr>
<h2 id='smooth.genome.plot'>A Function for Plotting Smoothed Pearson Correlation Coefficients Across Multiple Chromosomes</h2><span id='topic+smooth.genome.plot'></span>

<h3>Description</h3>

<p>This function plots smoothed R or R^2 values produced by corr.list.compute() across multiple chromosomes or genomewide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smooth.genome.plot(plot.list, plot.column = "R^2", annot.colors = c("black",
  "red", "green", "blue", "cyan"), vert.pad = 0.05, ylim.low = NULL,
  ylim.high = NULL, plot.legend = TRUE, legend.loc = "bottomright",
  lty.vec = NULL, lwd.vec = NULL, loess.span = 250, expand.size = 50,
  rect.colors = c("light gray", "gray"), chr.label = TRUE,
  xaxis.label = "Chromosome", yaxis.label = NULL, main.label = NULL,
  axis.cex = 1, label.cex = 1, xaxis.line = 1.5, yaxis.line = 2.5,
  main.line = 0, margin.vec = rep(1, 4))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smooth.genome.plot_+3A_plot.list">plot.list</code></td>
<td>
<p>A list produced by corr.list.compute().</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_plot.column">plot.column</code></td>
<td>
<p>&quot;R&quot; or &quot;R^2&quot; depending on whether Pearson correlation coefficients or squared Pearson
correlation coefficients will be plotted.  Default = &quot;R^2&quot;.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_annot.colors">annot.colors</code></td>
<td>
<p>A vector of colors used for plotting values in different entries of plot.list.  Default = c(&quot;black&quot;, &quot;red&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;cyan&quot;).</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_vert.pad">vert.pad</code></td>
<td>
<p>Amount of vertical white space in the plot.  Default = 0.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_ylim.low">ylim.low</code></td>
<td>
<p>Smallest value on the y-axis (used to control the range of values on the y-axis).  Default = NULL.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_ylim.high">ylim.high</code></td>
<td>
<p>Largest value on the y-axis (used to control the range of values on the y-axis).  Default = NULL.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_plot.legend">plot.legend</code></td>
<td>
<p>Logical value specifying whether a legend should be included.  Default = FALSE.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_legend.loc">legend.loc</code></td>
<td>
<p>Character value specifing the location of the legend.  Default = &quot;topright&quot;.  See See <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_lty.vec">lty.vec</code></td>
<td>
<p>Vector specifying line types for plotting values in different entries of plot.list.  Default = NULL.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_lwd.vec">lwd.vec</code></td>
<td>
<p>Vector specifying line widths for plotting values in different entries of  plot.list.  Default = NULL.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_loess.span">loess.span</code></td>
<td>
<p>A numerical value used to control the level of smoothing.  Smoothing is performed separately for each chromosome,
and loess.span effectively defines the number of genes in each smoothing window.  Default = 250.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_expand.size">expand.size</code></td>
<td>
<p>A numerical value used to control smoothing at the ends of chromosomes.  Both ends of each chromosome are artificially
extended by expand.size genes, smoothing is performed on the expanded chromosome, and then the smoothed values are restricted to the size
of the original chromosome.  Default = 50.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_rect.colors">rect.colors</code></td>
<td>
<p>A character vector of length two that controls the background color for each alternating chromosome.  Default = c(&quot;light gray&quot;, &quot;gray&quot;).</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_chr.label">chr.label</code></td>
<td>
<p>Logical value specifying whether chromosome numbers should appear on the plot.  Default = FALSE.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_xaxis.label">xaxis.label</code></td>
<td>
<p>Text used to label the x-axis of the plot.  Default = &quot;Chromosome&quot;.  See <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_yaxis.label">yaxis.label</code></td>
<td>
<p>Text used to label the y-axis of the plot.  Default = NULL.  See <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_main.label">main.label</code></td>
<td>
<p>Text used to label the plot header.  Default = NULL.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_axis.cex">axis.cex</code></td>
<td>
<p>Numerical value used to specify the font size on the axes.  Default = 1.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_label.cex">label.cex</code></td>
<td>
<p>Numerical value used to specify the font size for the axis labels.  Default = 1.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_xaxis.line">xaxis.line</code></td>
<td>
<p>Numerical value used to specify location of xaxis.label.  Default = 0.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_yaxis.line">yaxis.line</code></td>
<td>
<p>Numerical value used to specify location of yaxis.label.  Default = 0.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_main.line">main.line</code></td>
<td>
<p>Numerical value used to specify location of main.label.  Default = 0.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.genome.plot_+3A_margin.vec">margin.vec</code></td>
<td>
<p>Numerical vector specifying margin sizes.  Default = rep(1, 4).  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Creates a plot of gene-level R or R^2 values produced by corr.list.compute().  Values of R
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 prepped.data = data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

 pd.exp = prepped.data[["exp"]]

 pd.cn = prepped.data[["cn"]]

 pd.ga = prepped.data[["gene.annot"]]

 pd.sa = prepped.data[["sample.annot"]]

 output.list = corr.list.compute(pd.exp, pd.cn, pd.ga, pd.sa)

 smooth.genome.plot(plot.list = output.list, lwd.vec = c(3, 3), lty.vec = c(1, 2))

</code></pre>

<hr>
<h2 id='smooth.region.plot'>A Function for Plotting Smoothed Pearson Correlation Coefficients Genomewide</h2><span id='topic+smooth.region.plot'></span>

<h3>Description</h3>

<p>This function plots smoothed R or R^2 values produced by corr.list.compute() genomewide.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smooth.region.plot(plot.list, plot.chr, plot.start, plot.stop,
  plot.column = "R^2", annot.colors = c("black", "red", "green", "blue",
  "cyan"), vert.pad = 0.05, ylim.low = NULL, ylim.high = NULL,
  plot.legend = TRUE, legend.loc = "topleft", lty.vec = NULL,
  lwd.vec = NULL, loess.span = 50, expand.size = 50,
  xaxis.label = "Position (Mb)", yaxis.label = NULL, main.label = NULL,
  axis.cex = 1, label.cex = 1, xaxis.line = 1.5, yaxis.line = 2.5,
  main.line = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smooth.region.plot_+3A_plot.list">plot.list</code></td>
<td>
<p>A list produced by corr.list.compute().</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_plot.chr">plot.chr</code></td>
<td>
<p>The chromosome for which gene-level R or R^2 values will be plotted.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_plot.start">plot.start</code></td>
<td>
<p>The genomic position (in base pairs) where the plot will start.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_plot.stop">plot.stop</code></td>
<td>
<p>The genomic position (in base pairs) where the plot will stop.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_plot.column">plot.column</code></td>
<td>
<p>&quot;R&quot; or &quot;R^2&quot; depending on whether Pearson correlation coefficients or squared Pearson
correlation coefficients will be plotted.  Default = &quot;R^2&quot;.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_annot.colors">annot.colors</code></td>
<td>
<p>A vector of colors used for plotting values in different entries of plot.list.  Default = c(&quot;black&quot;, &quot;red&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;cyan&quot;).</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_vert.pad">vert.pad</code></td>
<td>
<p>Amount of vertical white space in the plot.  Default = 0.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_ylim.low">ylim.low</code></td>
<td>
<p>Smallest value on the y-axis (used to control the range of values on the y-axis).  Default = NULL.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_ylim.high">ylim.high</code></td>
<td>
<p>Largest value on the y-axis (used to control the range of values on the y-axis).  Default = NULL.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_plot.legend">plot.legend</code></td>
<td>
<p>Logical value specifying whether a legend should be included.  Default = FALSE.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_legend.loc">legend.loc</code></td>
<td>
<p>Character value specifing the location of the legend.  Default = &quot;topright&quot;.  See See <code><a href="graphics.html#topic+legend">legend</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_lty.vec">lty.vec</code></td>
<td>
<p>Vector specifying line types for plotting values in different entries of plot.list.  Default = NULL.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_lwd.vec">lwd.vec</code></td>
<td>
<p>Vector specifying line widths for plotting values in different entries of  plot.list.  Default = NULL.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_loess.span">loess.span</code></td>
<td>
<p>A numerical value used to control the level of smoothing.  Smoothing is performed separately for each chromosome,
and loess.span effectively defines the number of genes in each smoothing window.  Default = 250.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_expand.size">expand.size</code></td>
<td>
<p>A numerical value used to control smoothing at the ends of the region of interest.  Both ends of the region are artificially
extended by expand.size genes, smoothing is performed on the expanded region, and then the smoothed values are restricted to the size
of the original region.  Default = 50.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_xaxis.label">xaxis.label</code></td>
<td>
<p>Text used to label the x-axis of the plot.  Default = &quot;Chromosome&quot;.  See <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_yaxis.label">yaxis.label</code></td>
<td>
<p>Text used to label the y-axis of the plot.  Default = NULL.  See <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_main.label">main.label</code></td>
<td>
<p>Text used to label the plot header.  Default = NULL.  See <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_axis.cex">axis.cex</code></td>
<td>
<p>Numerical value used to specify the font size on the axes.  Default = 1.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_label.cex">label.cex</code></td>
<td>
<p>Numerical value used to specify the font size for the axis labels.  Default = 1.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_xaxis.line">xaxis.line</code></td>
<td>
<p>Numerical value used to specify location of xaxis.label.  Default = 0.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_yaxis.line">yaxis.line</code></td>
<td>
<p>Numerical value used to specify location of yaxis.label.  Default = 0.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
<tr><td><code id="smooth.region.plot_+3A_main.line">main.line</code></td>
<td>
<p>Numerical value used to specify location of main.label.  Default = 0.  See <code><a href="graphics.html#topic+mtext">mtext</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Creates a plot of gene-level R or R^2 values produced by corr.list.compute().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 prepped.data = data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

 pd.exp = prepped.data[["exp"]]

 pd.cn = prepped.data[["cn"]]

 pd.ga = prepped.data[["gene.annot"]]

 pd.sa = prepped.data[["sample.annot"]]

 output.list = corr.list.compute(pd.exp, pd.cn, pd.ga, pd.sa)

 smooth.region.plot(plot.list = output.list, plot.chr = 11, plot.start = 0e6, plot.stop = 135e6)

</code></pre>

<hr>
<h2 id='tcga.cn.convert'>A Function for Reformatting TCGA DNA Copy Number Matrices</h2><span id='topic+tcga.cn.convert'></span>

<h3>Description</h3>

<p>This function reformats DNA copy number matrices obtained from the Broad Institute's Firehose GDAC 
(<a href="https://gdac.broadinstitute.org/">https://gdac.broadinstitute.org/</a>) so they can be used as input for mVisAGe functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tcga.cn.convert(cn.mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tcga.cn.convert_+3A_cn.mat">cn.mat</code></td>
<td>
<p>A matrix of DNA copy number data included in the GISTIC2 output.  Typically all_data_by_genes.txt, or a subset thereof,
including the Locus.ID and Cytoband columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of DNA copy number data (rows = genes, columns = samples) that is suitable for input to 
mVisAGe functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cn.mat = tcga.cn.convert(cn.mat)

</code></pre>

<hr>
<h2 id='tcga.exp.convert'>A Function for Reformatting TCGA mRNA Expression Matrices</h2><span id='topic+tcga.exp.convert'></span>

<h3>Description</h3>

<p>This function reformats mRNA expression matrices obtained from the Broad Institute's Firehose GDAC 
(<a href="https://gdac.broadinstitute.org/">https://gdac.broadinstitute.org/</a>) so they can be used as input for mVisAGe functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tcga.exp.convert(exp.mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tcga.exp.convert_+3A_exp.mat">exp.mat</code></td>
<td>
<p>A matrix of mRNA expression data.  Typically illuminahiseq_rnaseqv2-RSEM_genes_normalized, or a subset thereof,
including the header rows.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of mRNA expression data (rows = genes, columns = samples) that is suitable for input to mVisAGe functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp.mat = tcga.exp.convert(exp.mat)

</code></pre>

<hr>
<h2 id='unsmooth.region.plot'>A Function for Plotting Pearson Correlation Coefficients in a Given Genomic Region</h2><span id='topic+unsmooth.region.plot'></span>

<h3>Description</h3>

<p>This function plots unsmoothed R or R^2 values produced by corr.list.compute() in a specified genomic region.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unsmooth.region.plot(plot.list, plot.chr, plot.start, plot.stop,
  plot.column = "R", plot.points = TRUE, plot.lines = TRUE,
  gene.names = TRUE, annot.colors = c("black", "red", "green", "blue",
  "cyan"), vert.pad = 0.05, num.ticks = 5, ylim.low = NULL,
  ylim.high = NULL, pch.vec = NULL, lty.vec = NULL, lwd.vec = NULL,
  plot.legend = TRUE, legend.loc = "topright")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unsmooth.region.plot_+3A_plot.list">plot.list</code></td>
<td>
<p>A list produced by corr.list.compute().</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_plot.chr">plot.chr</code></td>
<td>
<p>The chromosome for which gene-level R or R^2 values will be plotted.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_plot.start">plot.start</code></td>
<td>
<p>The genomic position (in base pairs) where the plot will start.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_plot.stop">plot.stop</code></td>
<td>
<p>The genomic position (in base pairs) where the plot will stop.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_plot.column">plot.column</code></td>
<td>
<p>&quot;R&quot; or &quot;R^2&quot; depending on whether Pearson correlation coefficients or squared Pearson
correlation coefficients will be plotted.  Default = &quot;R^2&quot;.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_plot.points">plot.points</code></td>
<td>
<p>Logical value specifying whether points should be included in the plot.  Default = TRUE.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_plot.lines">plot.lines</code></td>
<td>
<p>Logical values specifying whether points should be connected by lines.  Default = FALSE.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_gene.names">gene.names</code></td>
<td>
<p>Logical value specifying whether gene names should appear on the plot.  Default = FALSE.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_annot.colors">annot.colors</code></td>
<td>
<p>A vector of colors used for plotting values in different entries of plot.list.  Default = c(&quot;black&quot;, &quot;red&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;cyan&quot;).</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_vert.pad">vert.pad</code></td>
<td>
<p>Amount of vertical white space in the plot.  Default = 0.05.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_num.ticks">num.ticks</code></td>
<td>
<p>Number of ticks on the x-axis.  Default = 5.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_ylim.low">ylim.low</code></td>
<td>
<p>Smallest value on the y-axis (used to control the range of values on the y-axis).  Default = NULL.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_ylim.high">ylim.high</code></td>
<td>
<p>Largest value on the y-axis (used to control the range of values on the y-axis).  Default = NULL.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_pch.vec">pch.vec</code></td>
<td>
<p>Vector specifying point characters for plotting values in different entries of plot.list.  Default = NULL.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_lty.vec">lty.vec</code></td>
<td>
<p>Vector specifying line types for plotting values in different entries of plot.list.  Default = NULL.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_lwd.vec">lwd.vec</code></td>
<td>
<p>Vector specifying line widths for plotting values in different entries of  plot.list.  Default = NULL.  See <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_plot.legend">plot.legend</code></td>
<td>
<p>Logical value specifying whether a legend should be included.  Default = FALSE.</p>
</td></tr>
<tr><td><code id="unsmooth.region.plot_+3A_legend.loc">legend.loc</code></td>
<td>
<p>Character value specifing the location of the legend.  Default = &quot;topright&quot;.  See See <code><a href="graphics.html#topic+legend">legend</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Creates a plot of gene-level R or R^2 values produced by corr.list.compute().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp.mat = tcga.exp.convert(exp.mat)

 cn.mat = tcga.cn.convert(cn.mat)

 prepped.data = data.prep(exp.mat, cn.mat, gene.annot, sample.annot, log.exp = FALSE)

 pd.exp = prepped.data[["exp"]]

 pd.cn = prepped.data[["cn"]]

 pd.ga = prepped.data[["gene.annot"]]

 pd.sa = prepped.data[["sample.annot"]]

 output.list = corr.list.compute(pd.exp, pd.cn, pd.ga, pd.sa)

 unsmooth.region.plot(plot.list = output.list, plot.chr = 11, plot.start = 69e6, plot.stop = 70.5e6)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
