<!DOCTYPE html><html lang="en"><head><title>Help for package mvMAPIT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mvMAPIT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cauchy_combined'><p>Cauchy p combine method on mvmapit return</p></a></li>
<li><a href='#fishers_combined'><p>Fisher's combine method on mvmapit return</p></a></li>
<li><a href='#harmonic_combined'><p>Harmonic mean p combine method on mvmapit return</p></a></li>
<li><a href='#mvmapit'><p>Multivariate MArginal ePIstasis Test (mvMAPIT)</p></a></li>
<li><a href='#mvmapit_data'><p>Multivariate MAPIT analysis and exhaustive search analysis.</p></a></li>
<li><a href='#phillips_data'><p>Multivariate MAPIT analysis of binding affinities in broadly neutralizing antibodies.</p></a></li>
<li><a href='#simulate_traits'><p>Simulate phenotye data</p></a></li>
<li><a href='#simulated_data'><p>Genotype and trait data with epistasis.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multivariate Genome Wide Marginal Epistasis Test</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/lcrawlab/mvMAPIT">https://github.com/lcrawlab/mvMAPIT</a>,
<a href="https://lcrawlab.github.io/mvMAPIT/">https://lcrawlab.github.io/mvMAPIT/</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Epistasis, commonly defined as the interaction between genetic
  loci, is known to play an important role in the phenotypic variation of
  complex traits. As a result, many statistical methods have been developed to
  identify genetic variants that are involved in epistasis, and nearly all of
  these approaches carry out this task by focusing on analyzing one trait at a
  time. Previous studies have shown that jointly modeling multiple phenotypes
  can often dramatically increase statistical power for association mapping. In
  this  package, we present the 'multivariate MArginal ePIstasis Test'
  ('mvMAPIT') – a multi-outcome generalization of a recently proposed epistatic
  detection method which seeks to detect marginal epistasis or the combined
  pairwise interaction effects between a given variant and all other variants.
  By searching for marginal epistatic effects, one can identify genetic variants
  that are involved in epistasis without the need to identify the exact
  partners with which the variants interact – thus, potentially alleviating
  much of the statistical and computational burden associated with conventional
  explicit search based methods. Our proposed 'mvMAPIT' builds upon this
  strategy by taking advantage of correlation structure between traits to
  improve the identification of variants involved in epistasis.
  We formulate 'mvMAPIT' as a  multivariate linear mixed model and develop a
  multi-trait variance component estimation algorithm for efficient parameter
  inference and P-value computation. Together with reasonable model
  approximations, our proposed approach is scalable to moderately sized
  genome-wide association studies.
  Crawford et al. (2017) &lt;<a href="https://doi.org/10.1371%2Fjournal.pgen.1006869">doi:10.1371/journal.pgen.1006869</a>&gt;.
  Stamp et al. (2023) &lt;<a href="https://doi.org/10.1093%2Fg3journal%2Fjkad118">doi:10.1093/g3journal/jkad118</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>checkmate, CompQuadForm, dplyr, foreach, harmonicmeanp,
logging, mvtnorm, Rcpp, stats, tidyr, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>GGally, ggplot2, ggrepel, kableExtra, knitr, markdown,
RcppAlgos, rmarkdown, testthat</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, RcppParallel, RcppProgress, RcppSpdlog,
testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-25 17:10:46 UTC; jds</td>
</tr>
<tr>
<td>Author:</td>
<td>Julian Stamp <a href="https://orcid.org/0000-0003-3014-6249"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut],
  Lorin Crawford <a href="https://orcid.org/0000-0003-0178-8242"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Julian Stamp &lt;julian_stamp@brown.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-26 07:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cauchy_combined'>Cauchy p combine method on mvmapit return</h2><span id='topic+cauchy_combined'></span>

<h3>Description</h3>

<p>This function takes in the p-values tibble that mvmapit returned. It then
computes the combined p-values grouped by variant id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cauchy_combined(pvalues, group_col = "id", p_col = "p")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cauchy_combined_+3A_pvalues">pvalues</code></td>
<td>
<p>Tibble with p-values from mvmapit function call. Grouping is
based on the column named &quot;id&quot;</p>
</td></tr>
<tr><td><code id="cauchy_combined_+3A_group_col">group_col</code></td>
<td>
<p>String that denotes column by which to group and combine
p-values.</p>
</td></tr>
<tr><td><code id="cauchy_combined_+3A_p_col">p_col</code></td>
<td>
<p>String that denotes p-value column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Tibble with the combined p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(837)
p &lt;- 200
n &lt;- 100
d &lt;- 2
X &lt;- matrix(
    runif(p * n),
    ncol = p
)
Y &lt;- matrix(
    runif(d * n),
    ncol = d
)
mapit &lt;- mvmapit(
    t(X),
    t(Y),
    test = "normal", cores = 1, logLevel = "INFO"
)
cauchy &lt;- cauchy_combined(mapit$pvalues)
</code></pre>

<hr>
<h2 id='fishers_combined'>Fisher's combine method on mvmapit return</h2><span id='topic+fishers_combined'></span>

<h3>Description</h3>

<p>This function takes in the p-values tibble that mvmapit returned. It then
computes the combined p-values grouped by variant id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fishers_combined(pvalues, group_col = "id", p_col = "p")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fishers_combined_+3A_pvalues">pvalues</code></td>
<td>
<p>Tibble with p-values from mvmapit function call.</p>
</td></tr>
<tr><td><code id="fishers_combined_+3A_group_col">group_col</code></td>
<td>
<p>String that denotes column by which to group and combine
p-values.</p>
</td></tr>
<tr><td><code id="fishers_combined_+3A_p_col">p_col</code></td>
<td>
<p>String that denotes p-value column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Tibble with the combined p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(837)
p &lt;- 200
n &lt;- 100
d &lt;- 2
X &lt;- matrix(
    runif(p * n),
    ncol = p
)
Y &lt;- matrix(
    runif(d * n),
    ncol = d
)
mapit &lt;- mvmapit(
    t(X),
    t(Y),
    test = "normal", cores = 1, logLevel = "INFO"
)
fisher &lt;- fishers_combined(mapit$pvalues)
</code></pre>

<hr>
<h2 id='harmonic_combined'>Harmonic mean p combine method on mvmapit return</h2><span id='topic+harmonic_combined'></span>

<h3>Description</h3>

<p>This function takes in the p-values tibble that mvmapit returned. It then
computes the combined p-values grouped by variant id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>harmonic_combined(pvalues, group_col = "id", p_col = "p")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="harmonic_combined_+3A_pvalues">pvalues</code></td>
<td>
<p>Tibble with p-values from mvmapit function call. Grouping is
based on the column named &quot;id&quot;</p>
</td></tr>
<tr><td><code id="harmonic_combined_+3A_group_col">group_col</code></td>
<td>
<p>String that denotes column by which to group and combine
p-values.</p>
</td></tr>
<tr><td><code id="harmonic_combined_+3A_p_col">p_col</code></td>
<td>
<p>String that denotes p-value column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Tibble with the combined p-values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(837)
p &lt;- 200
n &lt;- 100
d &lt;- 2
X &lt;- matrix(
    runif(p * n),
    ncol = p
)
Y &lt;- matrix(
    runif(d * n),
    ncol = d
)
mapit &lt;- mvmapit(
    t(X),
    t(Y),
    test = "normal", cores = 1, logLevel = "INFO"
)
harmonic &lt;- harmonic_combined(mapit$pvalues)
</code></pre>

<hr>
<h2 id='mvmapit'>Multivariate MArginal ePIstasis Test (mvMAPIT)</h2><span id='topic+mvmapit'></span>

<h3>Description</h3>

<p>This function runs a multivariate version of the MArginal ePIstasis
Test (mvMAPIT) under the following model variations:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mvmapit(
  X,
  Y,
  Z = NULL,
  C = NULL,
  threshold = 0.05,
  accuracy = 1e-08,
  test = c("normal", "davies", "hybrid"),
  cores = 1,
  variantIndex = NULL,
  logLevel = "WARN",
  logFile = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mvmapit_+3A_x">X</code></td>
<td>
<p>is the p x n genotype matrix where p is the number of variants and n is the number of samples. Must be a matrix and not a data.frame.</p>
</td></tr>
<tr><td><code id="mvmapit_+3A_y">Y</code></td>
<td>
<p>is the d x n matrix of d quantitative or continuous traits for n samples.</p>
</td></tr>
<tr><td><code id="mvmapit_+3A_z">Z</code></td>
<td>
<p>is the matrix q x n matrix of covariates. Must be a matrix and not a data.frame.</p>
</td></tr>
<tr><td><code id="mvmapit_+3A_c">C</code></td>
<td>
<p>is an n x n covariance matrix detailing environmental effects and population structure effects.</p>
</td></tr>
<tr><td><code id="mvmapit_+3A_threshold">threshold</code></td>
<td>
<p>is a parameter detailing the value at which to recalibrate the Z test p values. If nothing is defined by the user, the default value will be 0.05 as recommended by the Crawford et al. (2017).</p>
</td></tr>
<tr><td><code id="mvmapit_+3A_accuracy">accuracy</code></td>
<td>
<p>is a parameter setting the davies function numerical approximation accuracy. This parameter is not needed for the normal test. Smaller p-values than the accuracy will be zero.</p>
</td></tr>
<tr><td><code id="mvmapit_+3A_test">test</code></td>
<td>
<p>is a parameter defining what hypothesis test should be run. Takes on values 'normal', 'davies', and 'hybrid'. The 'hybrid' test runs first the 'normal' test and then the 'davies' test on the significant variants.</p>
</td></tr>
<tr><td><code id="mvmapit_+3A_cores">cores</code></td>
<td>
<p>is a parameter detailing the number of cores to parallelize over. It is important to note that this value only matters when the user has installed OPENMP on their operating system.</p>
</td></tr>
<tr><td><code id="mvmapit_+3A_variantindex">variantIndex</code></td>
<td>
<p>is a vector containing indices of variants to be included in the computation.</p>
</td></tr>
<tr><td><code id="mvmapit_+3A_loglevel">logLevel</code></td>
<td>
<p>is a string parameter defining the log level for the logging package.</p>
</td></tr>
<tr><td><code id="mvmapit_+3A_logfile">logFile</code></td>
<td>
<p>is a string parameter defining the name of the log file for the logging output. Default is stdout.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>(1) Standard Model: y = m+g+e
where m ~ MVN(0,omega^2K), g ~ MVN(0,sigma^2G), e ~ MVN(0,tau^2M).
Recall from Crawford et al. (2017) that m is the combined additive effects
from all other variants, represents the additive effect of the k-th variant
under the polygenic background of all other variants; K is the genetic
relatedness matrix computed using genotypes from all variants other than the
k-th; g is the summation of all pairwise interaction effects between the
k-th variant and all other variants; G represents a relatedness matrix
computed based on pairwise interaction terms between the k-th variant and all
other variants. Here, we also denote D = diag(x_k) to be an n × n diagonal
matrix with the genotype vector x_k as its diagonal elements. It is important
to note that both K and G change with every new marker k that is considered.
Lastly; M is a variant specific projection matrix onto both the null space of
the intercept and the corresponding genotypic vector x_k.
</p>
<p>(2) Standard + Covariate Model: y = Wa+m+g+e
where W is a matrix of covariates with effect sizes a.
</p>
<p>(3) Standard + Common Environment Model: y = m+g+c+e i
where c ~ MVN(0,eta^2C) controls for extra environmental effects and
population structure with covariance matrix C.
</p>
<p>(4) Standard + Covariate + Common Environment Model: y = Wa+m+g+c+e
</p>
<p>This function will consider the following three hypothesis testing strategies
which are featured in Crawford et al. (2017):
(1) The Normal or Z test
(2) Davies Method
(3) Hybrid Method (Z test + Davies Method)
</p>


<h3>Value</h3>

<p>A list of P values and PVEs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(837)
p &lt;- 200
n &lt;- 100
d &lt;- 2
X &lt;- matrix(
    runif(p * n),
    ncol = p
)
Y &lt;- matrix(
    runif(d * n),
    ncol = d
)
mapit &lt;- mvmapit(
    t(X),
    t(Y),
    test = "normal", cores = 1, logLevel = "INFO"
)
</code></pre>

<hr>
<h2 id='mvmapit_data'>Multivariate MAPIT analysis and exhaustive search analysis.</h2><span id='topic+mvmapit_data'></span>

<h3>Description</h3>

<p>This data set contains the return object from the multivariate MAPIT method,
the fisher combined p-values, and the result from an exhaustive search using
regression on the SNPs that were significant in the mvMAPIT analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mvmapit_data
</code></pre>


<h3>Format</h3>

<p>A nested list containing tibble data frames:
</p>

<dl>
<dt>mvmapit</dt><dd><p>mvmapit return object; named list containing tibbles 'pvalues', 'pves', and 'duration'.</p>
</dd>
<dt>fisher</dt><dd><p>Tibble containing fisher combined p-values of the mvmapit data.</p>
</dd>
<dt>exhaustive_search</dt><dd><p>A dataframe containing the p-values of an exhaustive search together with the analysed interaction pair.</p>
</dd>
</dl>



<h3>Source</h3>

<p>data-raw/mvmapit_on_simulated_data.R
</p>

<hr>
<h2 id='phillips_data'>Multivariate MAPIT analysis of binding affinities in broadly neutralizing antibodies.</h2><span id='topic+phillips_data'></span>

<h3>Description</h3>

<p>This data set contains the return object from the multivariate MAPIT method,
applied to two binding affinity traits for two broadly neutralizing antibodies.
It also contains the regression coefficients on the same data as reported by Phillips et al. (2021).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phillips_data
</code></pre>


<h3>Format</h3>

<p>A named list containing tibble data frames:
</p>

<dl>
<dt>fisher</dt><dd><p>Tibble containing among other columns the residue id, p-values, antibody species, trait of the Phillips data. Combined p-value with Fisher's method.</p>
</dd>
<dt>harmonic</dt><dd><p>Tibble containing among other columns the residue id, p-values, antibody species, trait of the Phillips data. Combined p-value with harmonic mean p method.</p>
</dd>
<dt>regression</dt><dd><p>Named list containing two tibbles containing regression coefficients as reported by Phillips et al.</p>
</dd>
</dl>



<h3>Details</h3>

<p>The antibody CR9114 was analyzed with influenza H1 and H3.
The antibody CR6261 was analyzed with influenza H1 and H9.
</p>
<p>In the data, the p-values are computed for the test whether a given residue
position has a marginal epistatic effect on the binding affinities.
</p>
<p>Phillips et al. (2021) Binding affinity landscapes constrain the evolution of broadly neutralizing anti-influenza antibodies. eLife 10:e71393
</p>


<h3>Source</h3>

<p>vignette/study-phillips-bnabs.Rmd
</p>

<hr>
<h2 id='simulate_traits'>Simulate phenotye data</h2><span id='topic+simulate_traits'></span>

<h3>Description</h3>

<p>This function simulates trait data from a genotype matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_traits(
  genotype_matrix,
  n_causal = 1000,
  n_trait_specific = 10,
  n_pleiotropic = 10,
  H2 = 0.6,
  d = 2,
  rho = 0.8,
  marginal_correlation = 0.3,
  epistatic_correlation = 0.3,
  group_ratio_trait = 1,
  group_ratio_pleiotropic = 1,
  maf_threshold = 0.01,
  seed = 67132,
  logLevel = "INFO",
  logFile = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simulate_traits_+3A_genotype_matrix">genotype_matrix</code></td>
<td>
<p>Genotype matrix with samples as rows, and SNPs as columns.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_n_causal">n_causal</code></td>
<td>
<p>Number of SNPs that are causal.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_n_trait_specific">n_trait_specific</code></td>
<td>
<p>Number of causal SNPs with single trait epistatic effects.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_n_pleiotropic">n_pleiotropic</code></td>
<td>
<p>Number of SNPs with epistatic effects on all traits.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_h2">H2</code></td>
<td>
<p>Broad-sense heritability. Can be vector.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_d">d</code></td>
<td>
<p>Number of traits.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_rho">rho</code></td>
<td>
<p>Proportion of heritability explained by additivity.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_marginal_correlation">marginal_correlation</code></td>
<td>
<p>Correlation between the additive effects of the trait.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_epistatic_correlation">epistatic_correlation</code></td>
<td>
<p>Correlation between the epistatic effects of the trait.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_group_ratio_trait">group_ratio_trait</code></td>
<td>
<p>Ratio of sizes of trait specific groups that interact, e.g. a ratio 1:3 would be value 3.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_group_ratio_pleiotropic">group_ratio_pleiotropic</code></td>
<td>
<p>Ratio of sizes of pleiotropic groups that interact, e.g. a ratio 1:3 would be value 3.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_maf_threshold">maf_threshold</code></td>
<td>
<p>is a float parameter defining the threshold for the minor allele frequency not included in causal SNPs.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_seed">seed</code></td>
<td>
<p>Random seed for simulation.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_loglevel">logLevel</code></td>
<td>
<p>is a string parameter defining the log level for the logging package.</p>
</td></tr>
<tr><td><code id="simulate_traits_+3A_logfile">logFile</code></td>
<td>
<p>is a string parameter defining the name of the log file for the logging output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes a genotype matrix and simulates trait data under the following model:
beta_i ~ MN(0, V_i, I), i in { additive, epistatic, residual}
</p>
<p>The effect sizes follow a matrix normal distribution with no correlation between the samples but covariance between the effects for different traits
</p>


<h3>Value</h3>

<p>A list object containing the trait data, the genotype data, as well as the causal SNPs and summary statistics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p &lt;- 200
f &lt;- 10
g &lt;- 4
n &lt;- 100
d &lt;- 3
X &lt;- matrix(
    runif(p * n),
    ncol = p
)
data &lt;- simulate_traits(
    X, n_causal = f, n_trait_specific = g, n_pleiotropic = g, d = d, maf_threshold = 0,
    logLevel = "ERROR"
)
</code></pre>

<hr>
<h2 id='simulated_data'>Genotype and trait data with epistasis.</h2><span id='topic+simulated_data'></span>

<h3>Description</h3>

<p>A simulated dataset that has epistatic interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulated_data
</code></pre>


<h3>Format</h3>

<p>A named list with simulated data and simulation parameters:
</p>

<dl>
<dt>parameters</dt><dd><p>Tibble containing simulation parameters for each trait.</p>
</dd>
<dt>trait</dt><dd><p>Matrix containing simulated data for 2 traits and 500 samples.</p>
</dd>
<dt>genotype</dt><dd><p>Matrix containing simulated genotype with 500 samples and 1000 variables.</p>
</dd>
<dt>additive</dt><dd><p>Tibble containing all variants with additive effects on the traits as well as the effect sizes.</p>
</dd>
<dt>epistatic</dt><dd><p>Tibble containing all variants with epistatic effects on the traits as well as the effect sizes.</p>
</dd>
<dt>interactions</dt><dd><p>Tibble containing all interactions, effect size, and trait they affect.</p>
</dd>
<dt>snps.filtered</dt><dd><p>SNPs that were used in the simulations.</p>
</dd>
</dl>



<h3>Source</h3>

<p>data-raw/simulate_epistasis.R
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
