<!DOCTYPE html><html><head><title>Help for package mycobacrvR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mycobacrvR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#abcpred_nmer'><p>Epitope conservation prediction from ABCpred server information</p></a></li>
<li><a href='#algpred'><p>Filter Allergic/Non Allergic proteins using output of Algpred</p></a></li>
<li><a href='#allermatch'><p>Filter Allergic/Non Allergic proteins using output of Allermatch</p></a></li>
<li><a href='#bcepred_nmer'><p>Epitope conservation prediction from Bcepred server information</p></a></li>
<li><a href='#bimas_nmer'><p>Epitope prediction from Bimas server information</p></a></li>
<li><a href='#filter.firstlayer'><p>Filter first layer data</p></a></li>
<li><a href='#iedb_arb_mhci_nmer'><p>Epitope conservation prediction for MHCI using IEDB-ARB server information</p></a></li>
<li><a href='#iedb_arb_mhcii_nmer'><p>Epitope conservation prediction for MHCII using IEDB-ARB server information</p></a></li>
<li><a href='#iedb_consensus_mhci_nmer'><p>Epitope conservation prediction for MHCI using IEDB-consensus information</p></a></li>
<li><a href='#iedb_consensus_mhcii_nmer'><p>Epitope conservation prediction for MHCII using IEDB-consensus information</p></a></li>
<li><a href='#netmhcNN_nmer'><p>Epitope conservation prediction from NetMHC server 3.0 using artificial neural networks (ANNs)</p></a></li>
<li><a href='#netmhcWT_nmer'><p>Epitope conservation prediction from NetMHC server 3.0 using weight matrices</p></a></li>
<li><a href='#propred_nmer'><p>Epitope conservation prediction from Propred server information</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Integrative Immunoinformatics for Mycobacterial Diseases in R
Platform</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-12</td>
</tr>
<tr>
<td>Author:</td>
<td>Deepika Kulshreshtha, Rupanjali Chaudhuri, Surabhi Seth, S. Ramachandran</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>S. Ramachandran&lt;ramu@igib.in&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The mycobacrvR package contains utilities to provide detailed information for B cell and T cell epitopes for predicted adhesins from various servers such as ABCpred, Bcepred, Bimas, Propred, NetMHC and IEDB. Please refer the URL below to download data files (data_mycobacrvR.zip) used in functions of this package.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://mycobacteriarv.igib.res.in/download.html">https://mycobacteriarv.igib.res.in/download.html</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15), methods</td>
</tr>
<tr>
<td>Collate:</td>
<td>'iedb_consensus_mhci_nmer.R' 'abcpred_nmer.R' 'algpred.R'
'allermatch.R' 'bcepred_nmer.R' 'bimas_nmer.R'
'filter.firstlayer.R' 'iedb_arb_mhcii_nmer.R'
'iedb_consensus_mhcii_nmer.R' 'netmhcNN_nmer.R'
'netmhcWT_nmer.R' 'propred_nmer.R' 'iedb_arb_mhci_nmer.R'
'zzz.R'</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-13 09:32:12 UTC; sramachandran</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-13 14:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='abcpred_nmer'>Epitope conservation prediction from ABCpred server information</h2><span id='topic+abcpred_nmer'></span>

<h3>Description</h3>

<p><code>abcpred_nmer</code> is a method to predict conservation of epitopes among orthologs from ABCPred server data</p>


<h3>Usage</h3>

<pre><code class='language-R'>abcpred_nmer(epi_length)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="abcpred_nmer_+3A_epi_length">epi_length</code></td>
<td>
<p>A character vector of epitope length</p>
</td></tr></table>


<h3>Details</h3>

<p>This function takes epitope length as input into epi_length object. Input data for this function is present in &quot;/data_mycobacrvR/abcpred_out&quot; directory, therefore set this directory as working directory before using this function. ABCPred server provide epitopes by selecting window size of epitope length such as 10, 12, 14, 16, 18, 20. Therefore epi_length should be given as &quot;10mer&quot;.</p>


<h3>Value</h3>

<p>abcpred_nmer() will write two files in the working directory. One file contains the information of epitope and the other file contains metadata for epitope that shows orthologs of H37Rv contating that epitope</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: abcpred_nmer("10mer")
##Conservation of all 10mer epitopes from ABCPred server</code></pre>

<hr>
<h2 id='algpred'>Filter Allergic/Non Allergic proteins using output of Algpred</h2><span id='topic+algpred'></span>

<h3>Description</h3>

<p><code>algpred</code> is a function to filter allergic and non allergic proteins of an organism using output of Algpred</p>


<h3>Usage</h3>

<pre><code class='language-R'>algpred(data, organism, ginumber, prediction)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="algpred_+3A_data">data</code></td>
<td>
<p>A character string specifying filenamme containing tabular output from Algpred</p>
</td></tr>
<tr><td><code id="algpred_+3A_organism">organism</code></td>
<td>
<p>A character vector specifying name of species present in &quot;organism.txt&quot;</p>
</td></tr>
<tr><td><code id="algpred_+3A_ginumber">ginumber</code></td>
<td>
<p>A numeric vector specifying ginumber of protein for a species for which allergic preperty is to be predicted</p>
</td></tr>
<tr><td><code id="algpred_+3A_prediction">prediction</code></td>
<td>
<p>A character vector specifying &quot;Allergen&quot; or &quot;Non Allergen&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes tabular output of Algpred in data as input.Input data (Final_algpred.txt) for this function is present in &quot;/data_mycobacrvR/&quot; directory, therefore set working directory to &quot;~/data_mycobacrvR/&quot; before using this function. algpred() filters data using arguments organism, ginumber and prediction. Defalult value for ginumber is &quot;ALL&quot; and for prediction is &quot;Non Allergen&quot;. Organism can be provided by selecting organisms from &quot;organism.txt&quot; file present in working directory </p>


<h3>Value</h3>

<p>A file of filtered proteins named &quot;filtered_algpred.txt&quot; using the given arguments.</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>See Also</h3>

<p><code><a href="#topic+allermatch">allermatch</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: algpred("Final_algpred.txt",organism="Mycobacterium avium 104")</code></pre>

<hr>
<h2 id='allermatch'>Filter Allergic/Non Allergic proteins using output of Allermatch</h2><span id='topic+allermatch'></span>

<h3>Description</h3>

<p><code>allermatch</code> is a function to filter allergic and non allergic proteins of an organism using output of Allermatch</p>


<h3>Usage</h3>

<pre><code class='language-R'>allermatch(data, organism, ginumber, prediction)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="allermatch_+3A_data">data</code></td>
<td>
<p>A character string specifying filenamme containing tabular output from Allermatch</p>
</td></tr>
<tr><td><code id="allermatch_+3A_organism">organism</code></td>
<td>
<p>A character vector specifying name of species present in organism.txt</p>
</td></tr>
<tr><td><code id="allermatch_+3A_ginumber">ginumber</code></td>
<td>
<p>A numeric vector specifying ginumber of protein for a species for which allergic preperty is to be predicted</p>
</td></tr>
<tr><td><code id="allermatch_+3A_prediction">prediction</code></td>
<td>
<p>A character vector specifying &quot;Allergen&quot; or &quot;Non Allergen&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes tabular output of Allermatch in data as input.Input data (Final_allermatch.txt) for this function is present in &quot;/data_mycobacrvR/&quot; directory, therefore set working directory to &quot;~/data_mycobacrvR/&quot; before using this function. allermatch() filters data using arguments organism, ginumber and prediction. Defalult value for ginumber is &quot;ALL&quot; and for prediction is &quot;Non Allergen&quot;. Organism can be provided by selecting organisms from &quot;organism.txt&quot; file present in working directory </p>


<h3>Value</h3>

<p>A file of filtered proteins named &quot;filtered_allermatch.txt&quot; using the given arguments.</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>See Also</h3>

<p><code><a href="#topic+algpred">algpred</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: allermatch("Final_allermatch.txt",organism="Mycobacterium avium 104")</code></pre>

<hr>
<h2 id='bcepred_nmer'>Epitope conservation prediction from Bcepred server information</h2><span id='topic+bcepred_nmer'></span>

<h3>Description</h3>

<p><code>bcepred_nmer</code> is a method to predict conservation of epitopes among orthologs from Bcepred server data</p>


<h3>Usage</h3>

<pre><code class='language-R'>bcepred_nmer(clas)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bcepred_nmer_+3A_clas">clas</code></td>
<td>
<p>A character vector for server name</p>
</td></tr></table>


<h3>Details</h3>

<p>This function takes server name &quot;bcepred&quot; as input into clas object. Input data for this function is present in &quot;/data_mycobacrvR/bcepred_out&quot; directory, therefore set working directory to &quot;~/data_mycobacrvR/bcepred_out/&quot; before using this function. Bcepred server provides epitopes of different length.</p>


<h3>Value</h3>

<p>bcepred_nmer() will write two files in the working directory. One file contains the information of epitope and the other file contains metadata for epitope showing orthologs of H37Rv</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: bcepred_nmer("bcepred")##Conservation of epitopes from Bcepred server</code></pre>

<hr>
<h2 id='bimas_nmer'>Epitope prediction from Bimas server information</h2><span id='topic+bimas_nmer'></span>

<h3>Description</h3>

<p><code>bimas_nmer</code> is a method to predict conservation of epitopes among orthologs from Bimas server data</p>


<h3>Usage</h3>

<pre><code class='language-R'>bimas_nmer(epi_length)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bimas_nmer_+3A_epi_length">epi_length</code></td>
<td>
<p>A character vector of epitope length</p>
</td></tr></table>


<h3>Details</h3>

<p>This function takes epitope length as input into epi_length object. Input data for this function is present in &quot;/data_mycobacrvR/bimas_out&quot; directory, therefore set this directory as working directory before using this function. Bimas server provide epitopes by selecting window size of epitope length such as 8, 9 and 10. Therefore epi_length should be given as &quot;10mer&quot;.</p>


<h3>Value</h3>

<p>bimas_nmer() will write two files in the working directory. One file contains the information of epitope and the other file contains metadata for epitope showing orthologs of H37Rv</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: bimas_nmer("10mer")
##Conservation of 10mer epitopes from Bimas server</code></pre>

<hr>
<h2 id='filter.firstlayer'>Filter first layer data</h2><span id='topic+filter.firstlayer'></span>

<h3>Description</h3>

<p><code>filter.firstlayer</code> is a function to filter first layer data</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter.firstlayer(data, organism, ginumber, spaanscore, 
subcelllocal, tmhelices, Hrefhits)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter.firstlayer_+3A_data">data</code></td>
<td>
<p>A character string specifying filenamme containing first layer data table</p>
</td></tr>
<tr><td><code id="filter.firstlayer_+3A_organism">organism</code></td>
<td>
<p>A character vector specifying species name present in &quot;organism.txt&quot;</p>
</td></tr>
<tr><td><code id="filter.firstlayer_+3A_ginumber">ginumber</code></td>
<td>
<p>A numeric vector specifying ginumber</p>
</td></tr>
<tr><td><code id="filter.firstlayer_+3A_spaanscore">spaanscore</code></td>
<td>
<p>A numeric vector specifying SPAAN score threshold</p>
</td></tr>
<tr><td><code id="filter.firstlayer_+3A_subcelllocal">subcelllocal</code></td>
<td>
<p>A character vector specifying location of the protein</p>
</td></tr>
<tr><td><code id="filter.firstlayer_+3A_tmhelices">tmhelices</code></td>
<td>
<p>A numeric vectr specifying number of TM helix</p>
</td></tr>
<tr><td><code id="filter.firstlayer_+3A_hrefhits">Hrefhits</code></td>
<td>
<p>A character vector specifying non similarity to Human Reference  Proteins</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function filters the first layer data according to &quot;motif and topology&quot;, &quot;subcellular location&quot; and &quot;homology&quot;. Input data (firstlayer.txt) for this function is present in &quot;/data_mycobacrvR/&quot; directory, therefore set working directory to &quot;~/data_mycobacrvR/&quot; before using this function. Defalult value for ginumber,spaanscore,subcelllocal,tmhelices and Hrefhits is &quot;ALL&quot;, &gt;0.6, &quot;Extracellular OR Cellwall&quot;, &lt;2 and &quot;No Hits found&quot; </p>


<h3>Value</h3>

<p>A file of filtered proteins named &quot;filtered_firstlayer.txt&quot; using the given arguments</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha, Rupanjali Chaudhuri, S. Ramachandran</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: filter.firstlayer("firstlayer.txt",organism="Mycobacterium avium 104")</code></pre>

<hr>
<h2 id='iedb_arb_mhci_nmer'>Epitope conservation prediction for MHCI using IEDB-ARB server information</h2><span id='topic+iedb_arb_mhci_nmer'></span>

<h3>Description</h3>

<p><code>iedb_arb_mhci_nmer</code> is a method to predict conservation of epitopes among orthologs for MHCI using IEDB-ARB server data</p>


<h3>Usage</h3>

<pre><code class='language-R'>iedb_arb_mhci_nmer(clas)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iedb_arb_mhci_nmer_+3A_clas">clas</code></td>
<td>
<p>A character vector for server name</p>
</td></tr></table>


<h3>Details</h3>

<p>This function takes server name &quot;IEDB_arb_mhci&quot; as input into clas object. Input data for this function is present in &quot;/data_mycobacrvR/IEDBarb_mhci_out&quot; directory, therefore set working directory to &quot;~/data_mycobacrvR/IEDBarb_mhci_out/&quot; before using this function. IEDB-ARB server provides epitopes of different length.</p>


<h3>Value</h3>

<p>iedb_arb_mhci_nmer() will write two files in the working directory. One file contains the information of epitope and the other file contains metadata for epitope showing orthologs of H37Rv</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: iedb_arb_mhci_nmer("IEDB_arb_mhci")
##Conservation of all epitopes from IEDB server</code></pre>

<hr>
<h2 id='iedb_arb_mhcii_nmer'>Epitope conservation prediction for MHCII using IEDB-ARB server information</h2><span id='topic+iedb_arb_mhcii_nmer'></span>

<h3>Description</h3>

<p><code>iedb_arb_mhcii_nmer</code> is a method to predict conservation of epitopes among orthologs for MHCII using IEDB-ARB server data</p>


<h3>Usage</h3>

<pre><code class='language-R'>iedb_arb_mhcii_nmer(clas)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iedb_arb_mhcii_nmer_+3A_clas">clas</code></td>
<td>
<p>A character vector for server name</p>
</td></tr></table>


<h3>Details</h3>

<p>This function takes server name &quot;IEDB_arbMHCII&quot; as input into clas object. Input data for this function is present in &quot;/data_mycobacrvR/iedb_arb_mhcii_out&quot; directory, therefore set working directory to &quot;~/data_mycobacrvR/iedb_arb_mhcii_out/&quot; before using this function. IEDB-ARB server provides epitopes of different length.</p>


<h3>Value</h3>

<p>iedb_arb_mhcii_nmer() will write two files in the working directory. One file contains the information of epitope and the other file contains metadata for epitope showing orthologs of H37Rv</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: iedb_arb_mhcii_nmer("IEDB_arb_mhcii")
##Conservation of all epitopes from IEDB server</code></pre>

<hr>
<h2 id='iedb_consensus_mhci_nmer'>Epitope conservation prediction for MHCI using IEDB-consensus information</h2><span id='topic+iedb_consensus_mhci_nmer'></span>

<h3>Description</h3>

<p><code>iedb_consensus_mhci_nmer</code> is a method to predict conservation of epitopes among orthologs for MHCI using IEDB-consensus server data</p>


<h3>Usage</h3>

<pre><code class='language-R'>iedb_consensus_mhci_nmer(epi_length)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iedb_consensus_mhci_nmer_+3A_epi_length">epi_length</code></td>
<td>
<p>A character vector of epitope length</p>
</td></tr></table>


<h3>Details</h3>

<p>This function takes epitope length as input into epi_length object. Input data for this function is present in &quot;/data_mycobacrvR/IEDBconsensus_mhci_out&quot; directory, therefore set working directory to &quot;~/data_mycobacrvR/IEDBconsensus_mhci_out/&quot; before using this function. IEDB server provides epitopes by selecting window size of epitope length such as 8, 9, 10. Therefore epi_length should be given as &quot;10mer&quot;.</p>


<h3>Value</h3>

<p>iedb_consensus_mhci_nmer() will write two files in the working directory. One file contains the information of epitope and the other file contains metadata for epitope showing orthologs of H37Rv</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: iedb_consensus_mhci_nmer("IEDB_consensus_mhci")
##Conservation of epitopes from IEDB server</code></pre>

<hr>
<h2 id='iedb_consensus_mhcii_nmer'>Epitope conservation prediction for MHCII using IEDB-consensus information</h2><span id='topic+iedb_consensus_mhcii_nmer'></span>

<h3>Description</h3>

<p><code>iedb_consensus_mhcii_nmer</code> is a method to predict conservation of epitopes among orthologs for MHCII using IEDB-consensus server data</p>


<h3>Usage</h3>

<pre><code class='language-R'>iedb_consensus_mhcii_nmer(clas)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iedb_consensus_mhcii_nmer_+3A_clas">clas</code></td>
<td>
<p>A character vector for server name</p>
</td></tr></table>


<h3>Details</h3>

<p>This function takes server name &quot;IEDB_consensusMHCII&quot; as input into clas object.Input data for this function is present in &quot;/data_mycobacrvR/iedb_consensus_mhcii_out&quot; directory, therefore set working directory to &quot;~/data_mycobacrvR/iedb_consensus_mhcii_out/&quot; before using this function. IEDB-consensus provides epitopes of different length.</p>


<h3>Value</h3>

<p>iedb_consensus_mhcii_nmer() will write two files in the working directory. One file contains the information of epitope and the other file contains metadata for epitope showing orthologs of H37Rv</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: iedb_consensus_mhcii_nmer("IEDB_consensus_mhcii")
##Conservation of all epitopes from IEDB server</code></pre>

<hr>
<h2 id='netmhcNN_nmer'>Epitope conservation prediction from NetMHC server 3.0 using artificial neural networks (ANNs)</h2><span id='topic+netmhcNN_nmer'></span>

<h3>Description</h3>

<p><code>netmhcNN_nmer</code> is a method to conservation of epitopes among orthologs from NetMHC server 3.0 using artificial neural networks (ANNs)</p>


<h3>Usage</h3>

<pre><code class='language-R'>netmhcNN_nmer(epi_length)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="netmhcNN_nmer_+3A_epi_length">epi_length</code></td>
<td>
<p>A character vector of epitope length</p>
</td></tr></table>


<h3>Details</h3>

<p>This function takes epitope length as input into epi_length object. Input data for this function is present in &quot;/data_mycobacrvR/netmhcNN_out&quot; directory, therefore this directory as working directory before using this function. NetMHC server 3.0 provides epitopes by selecting window size of epitope length such as 8mer, 9mer, 10mer, 11mer. Therefore epi_length should be given as &quot;10mer&quot;.</p>


<h3>Value</h3>

<p>netmhcNN_nmer() will write two files in the working directory. One file contains the information of epitope and the other file contains metadata for epitope showing orthologs of H37Rv</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: netmhcNN_nmer("10mer")
##Conservation of 10mer epitopes from NetMHC server</code></pre>

<hr>
<h2 id='netmhcWT_nmer'>Epitope conservation prediction from NetMHC server 3.0 using weight matrices</h2><span id='topic+netmhcWT_nmer'></span>

<h3>Description</h3>

<p><code>netmhcWT_nmer</code> is a method to conservation of epitopes among orthologs from NetMHC server 3.0 using weight matrices data</p>


<h3>Usage</h3>

<pre><code class='language-R'>netmhcWT_nmer(epi_length)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="netmhcWT_nmer_+3A_epi_length">epi_length</code></td>
<td>
<p>A character vector of epitope length</p>
</td></tr></table>


<h3>Details</h3>

<p>This function takes epitope length as input into epi_length object. Input data for this function is present in &quot;/data_mycobacrvR/netmhcWT_out&quot; directory, therefore set this directory as working directory before using this function. NetMHC server 3.0 provides epitopes by selecting window size of epitope length such as 8mer, 9mer, 10mer, 11mer. Therefore epi_length should be given as &quot;10mer&quot;.</p>


<h3>Value</h3>

<p>netmhcWT_nmer() will write two files in the working directory. One file contains the information of epitope and the other file contains metadata for epitope showing orthologs of H37Rv</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: netmhcWT_nmer("10mer")
##Conservation of all 10mer epitopes from NetMHC server</code></pre>

<hr>
<h2 id='propred_nmer'>Epitope conservation prediction from Propred server information</h2><span id='topic+propred_nmer'></span>

<h3>Description</h3>

<p><code>propred_nmer</code> is a method to conservation of epitopes among orthologs from Propred server data</p>


<h3>Usage</h3>

<pre><code class='language-R'>propred_nmer(epi_length)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="propred_nmer_+3A_epi_length">epi_length</code></td>
<td>
<p>A character vector of epitope length</p>
</td></tr></table>


<h3>Details</h3>

<p>This function takes epitope length as input into epi_length object. Input data for this function is present in &quot;/data_mycobacrvR/propred_out&quot; directory, therefore set this directory as working directory before using this function. Propred server provide epitopes of epitope length 9. Therefore epi_length should be given as &quot;9mer&quot;.</p>


<h3>Value</h3>

<p>propred_nmer() will write two files in the working directory. One file contains the information of epitope and the other file contains metadata for epitope showing orthologs of H37Rv</p>


<h3>Author(s)</h3>

<p>Deepika Kulshreshtha</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: propred_nmer("9mer")
##Conservation of all 9mer epitopes from Propred server</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
