<!DOCTYPE html><html><head><title>Help for package myTAI</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {myTAI}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#age.apply'><p>Age Category Specific apply Function</p></a></li>
<li><a href='#bar.colors'><p>Color palette for barplots</p></a></li>
<li><a href='#bootMatrix'><p>Compute a Permutation Matrix for Test Statistics</p></a></li>
<li><a href='#CollapseReplicates'><p>Combine Replicates in an ExpressionSet</p></a></li>
<li><a href='#CombinatorialSignificance'><p>Compute the Statistical Significance of Each Replicate Combination</p></a></li>
<li><a href='#DiffGenes'><p>Differential Gene Expression Analysis</p></a></li>
<li><a href='#DivergenceExpressionSetExample'><p>An Example DivergenceExpressionSet Data Set</p></a></li>
<li><a href='#EarlyConservationTest'><p>Perform Reductive Early Conservation Test</p></a></li>
<li><a href='#ecScore'><p>Compute the Hourglass Score for the EarlyConservationTest</p></a></li>
<li><a href='#EnrichmentTest'><p>Phylostratum or Divergence Stratum Enrichment of a given Gene Set based on Fisher's Test</p></a></li>
<li><a href='#Expressed'><p>Filter for Expressed Genes</p></a></li>
<li><a href='#FlatLineTest'><p>Perform Flat Line Test</p></a></li>
<li><a href='#geom.mean'><p>Geometric Mean</p></a></li>
<li><a href='#GroupDiffs'><p>Quantify the significant differences between gene expression distributions of age groups</p></a></li>
<li><a href='#harm.mean'><p>Harmonic Mean</p></a></li>
<li><a href='#is.ExpressionSet'><p>Test ExpressionSet Standard</p></a></li>
<li><a href='#MatchMap'><p>Match a Phylostratigraphic Map or Divergence Map with a ExpressionMatrix</p></a></li>
<li><a href='#omitMatrix'><p>Compute TAI or TDI Profiles Omitting a Given Gene</p></a></li>
<li><a href='#PhyloExpressionSetExample'><p>An Example PhyloExpressionSet Data Set</p></a></li>
<li><a href='#PlotBarRE'><p>Plot Mean Relative Expression Levels as Barplot</p></a></li>
<li><a href='#PlotCategoryExpr'><p>Plot the Expression Levels of each Age or Divergence Category as Boxplot, Violinplot, or Dotplot</p></a></li>
<li><a href='#PlotCIRatio'><p>Plot Transcriptome Index using bootstrapping and confidence intervals</p></a></li>
<li><a href='#PlotContribution'><p>Plot Cumuative Transcriptome Index</p></a></li>
<li><a href='#PlotCorrelation'><p>Plot the Correlation Between Phylostrata and Divergence Strata</p></a></li>
<li><a href='#PlotDistribution'><p>Plot the Frequency Distribution of Phylostrata or Divergence Strata</p></a></li>
<li><a href='#PlotEnrichment'><p>Plot the Phylostratum or Divergence Stratum Enrichment of a given Gene Set</p></a></li>
<li><a href='#PlotGeneSet'><p>Plot the Expression Profiles of a Gene Set</p></a></li>
<li><a href='#PlotGroupDiffs'><p>Plot the significant differences between gene expression distributions of PS or DS groups</p></a></li>
<li><a href='#PlotMeans'><p>Plot Mean Expression Profiles</p></a></li>
<li><a href='#PlotMedians'><p>Plot Median Expression Profiles</p></a></li>
<li><a href='#PlotPattern'><p>Plot the Transcriptome Age Index or Transcriptome Divergence Index</p></a></li>
<li><a href='#PlotRE'><p>Plot Relative Expression Levels</p></a></li>
<li><a href='#PlotReplicateQuality'><p>Plot the Quality of Biological Replicates</p></a></li>
<li><a href='#PlotSelectedAgeDistr'><p>Plot the PS or DS distribution of a selected set of genes</p></a></li>
<li><a href='#PlotSignature'><p>Plot evolutionary signatures across transcriptomes</p></a></li>
<li><a href='#PlotVars'><p>Plot Variance of Expression Profiles</p></a></li>
<li><a href='#pMatrix'><p>Compute Partial TAI or TDI Values</p></a></li>
<li><a href='#pStrata'><p>Compute Partial Strata Values</p></a></li>
<li><a href='#pTAI'><p>Compute the Phylostratum Contribution to the Global Transcriptome Age Index</p></a></li>
<li><a href='#pTDI'><p>Compute the Divergence Stratum Contribution to the Global Transcriptome Divergence Index</p></a></li>
<li><a href='#RE'><p>Transform to Relative Expression Levels</p></a></li>
<li><a href='#ReductiveHourglassTest'><p>Perform the Reductive Hourglass Test</p></a></li>
<li><a href='#REMatrix'><p>Compute a Relative Expression Matrix</p></a></li>
<li><a href='#reversehourglassScore'><p>Compute the Reverse Hourglass Score for the Reverse Hourglass Test</p></a></li>
<li><a href='#ReverseHourglassTest'><p>Perform the Reverse Hourglass Test</p></a></li>
<li><a href='#rhScore'><p>Compute the Hourglass Score for the Reductive Hourglass Test</p></a></li>
<li><a href='#SelectGeneSet'><p>Select a Subset of Genes in an ExpressionSet</p></a></li>
<li><a href='#TAI'><p>Compute the Transcriptome Age Index (TAI)</p></a></li>
<li><a href='#taxid'><p>Retrieve taxonomy categories from NCBI Taxonomy</p></a></li>
<li><a href='#taxonomy'><p>Retrieving Taxonomic Information of a Query Organism</p></a></li>
<li><a href='#TDI'><p>Compute the Transcriptome Divergence Index (TDI)</p></a></li>
<li><a href='#tf'><p>Transform Gene Expression Levels</p></a></li>
<li><a href='#TPI'><p>Compute the Transcriptome Polymorphism Index (TPI)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Evolutionary Transcriptomics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.9.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-02-22</td>
</tr>
<tr>
<td>Description:</td>
<td>Investigate the evolution of biological processes by capturing evolutionary signatures in transcriptomes (Drost et al. (2017) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtx835">doi:10.1093/bioinformatics/btx835</a>&gt;). The aim of this tool is to provide a transcriptome analysis environment to quantify the average evolutionary age of genes contributing to a transcriptome of interest (Drost et al. (2016) &lt;<a href="https://doi.org/10.1101%2F051565">doi:10.1101/051565</a>&gt;).</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.0), nortest (&ge; 1.0-2), fitdistrplus (&ge; 1.0-2),
parallel (&ge; 3.1.1), foreach (&ge; 1.4.2), doParallel (&ge; 1.0.8),
dplyr (&ge; 0.3.0), RColorBrewer (&ge; 1.1-2), taxize (&ge; 0.6.0),
methods (&ge; 3.1.1), graphics (&ge; 3.1.1), stats (&ge; 3.1.1),
grDevices (&ge; 3.1.1), utils (&ge; 3.1.1), reshape2 (&ge; 1.4.1),
ggplot2 (&ge; 1.0.1), readr (&ge; 0.2.2), tibble, scales,
gridExtra, edgeR</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr (&ge; 1.6), rmarkdown (&ge; 0.3.3), devtools (&ge; 1.6.1),
testthat (&ge; 0.9.1), mgcv</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, cpp11</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/drostlab/myTAI">https://github.com/drostlab/myTAI</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/drostlab/myTAI/issues">https://github.com/drostlab/myTAI/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++11</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-02-23 22:49:22 UTC; hdrost</td>
</tr>
<tr>
<td>Author:</td>
<td>Hajk-Georg Drost <a href="https://orcid.org/0000-0002-1567-306X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hajk-Georg Drost &lt;hajk-georg.drost@tuebingen.mpg.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-02-24 05:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='age.apply'>Age Category Specific apply Function</h2><span id='topic+age.apply'></span>

<h3>Description</h3>

<p>This function performs the split-apply-combine methodology on Phylostrata or Divergence Strata stored within the input ExpressionSet.
</p>
<p>This function is very useful to perform any phylostratum or divergence-stratum specific analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>age.apply(ExpressionSet, FUN, ..., as.list = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="age.apply_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="age.apply_+3A_fun">FUN</code></td>
<td>
<p>a function to be performed on the corresponding expression matrix of each phylostratum or divergence-stratum.</p>
</td></tr>
<tr><td><code id="age.apply_+3A_...">...</code></td>
<td>
<p>additional arguments of FUN.</p>
</td></tr>
<tr><td><code id="age.apply_+3A_as.list">as.list</code></td>
<td>
<p>a boolean value specifying whether the output format shall be a matrix or a list object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses the <code><a href="base.html#topic+split">split</a></code> function to subset the expression matrix into
phylostratum specific sub-matrices. Internally using <code><a href="base.html#topic+lapply">lapply</a></code>, any function can
be performed to the sub-matrices. The return value of this function is a numeric matrix storing
the return values by <code>FUN</code> for each phylostratum and each developmental stage s. 
Note that the input <code>FUN</code> must be an function that can be applied to a matrix (e.g., <code><a href="base.html#topic+colMeans">colMeans</a></code> or <code><a href="#topic+RE">RE</a></code>). 
In case you use an an anymous function you coud use <code>function(x) apply(x , 2 , var)</code> as an example to compute the variance of each phylostratum and each
developmental stage s.
</p>


<h3>Value</h3>

<p>Either a numeric matrix storing the return values of the applied function for each age class
or a numeric list storing the return values of the applied function for each age class in a list.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+split">split</a></code>, <code><a href="base.html#topic+tapply">tapply</a></code>, <code><a href="base.html#topic+lapply">lapply</a></code>, <code><a href="#topic+RE">RE</a></code>, <code><a href="#topic+REMatrix">REMatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
 # source the example dataset
 data(PhyloExpressionSetExample)
 
# Example 1
# get the relative expression profiles for each phylostratum
age.apply(PhyloExpressionSetExample, RE)

# this is analogous to 
REMatrix(PhyloExpressionSetExample)
# Example 2
# compute the mean expression profiles for each phylostratum
age.apply(PhyloExpressionSetExample, colMeans)

# Example 3
# compute the variance profiles for each phylostratum
age.apply(PhyloExpressionSetExample, function(x) apply(x , 2 , var))

# Example 4
# compute the range for each phylostratum
# Note: in this case, the range() function returns 2 values for each phylostratum
# and each developmental stage, hence one should use the argument 'as.list = TRUE'
# to make sure that the results are returned properly 
age.apply(PhyloExpressionSetExample, function(x) apply(x , 2 , range), as.list = TRUE)

</code></pre>

<hr>
<h2 id='bar.colors'>Color palette for barplots</h2><span id='topic+bar.colors'></span>

<h3>Description</h3>

<p>A nice color palette for barplots with several bars.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bar.colors(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bar.colors_+3A_n">n</code></td>
<td>
<p>the number of colors to be in the palette.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to select colors for bar plots.
</p>


<h3>Value</h3>

<p>a character vector containing different color names that can be used for barplots.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# get 5 different colors for 5 different bars
barplot_colors &lt;- bar.colors(5)
</code></pre>

<hr>
<h2 id='bootMatrix'>Compute a Permutation Matrix for Test Statistics</h2><span id='topic+bootMatrix'></span>

<h3>Description</h3>

<p>This function computes the TAI for a row permutated PhyloExpressionSet or DivergenceExpressionSet.
</p>
<p>One can specify the number of permutations which corresponds to the number of <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> profiles
that are being returned as data matrix. The function then returns a <code>TAI</code> or <code>TDI</code> matrix holding
the <code>TAI</code> or <code>TDI</code> profiles of the permutated PhyloExpressionSets or DivergenceExpressionSets. This procedure
can be used for building test statistics based on the <code>TAI</code> or <code>TDI</code> profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootMatrix(ExpressionSet, permutations = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bootMatrix_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="bootMatrix_+3A_permutations">permutations</code></td>
<td>
<p>a numeric value specifying the number of permutations to be performed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The sampled <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> matrix samples the phylostratum or divergence-stratum vector of
a given PhyloExpressionSet or DivergenceExpressionSet and computes the corresponding <code>TAI</code> or <code>TDI</code> profiles
of the randomly assigned phylostrata or divergence-strata. This sampling is then performed N times, yielding N randomly sampled <code>TAI</code> or <code>TDI</code> profiles.
This random <code>TAI</code> or <code>TDI</code> profile matrix can then be used to perform statistical tests 
(such as the <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>, <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>, or <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code>) based on the significance of <code>TAI</code> or <code>TDI</code> patterns.
</p>


<h3>Value</h3>

<p>a numeric matrix representing N randomly permuted <code>TAI</code> or <code>TDI</code> profiles.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490):  98-101.
</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>


<h3>See Also</h3>

<p><code><a href="#topic+FlatLineTest">FlatLineTest</a></code>, <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>, <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# read standard phylotranscriptomics data
data(PhyloExpressionSetExample)
data(DivergenceExpressionSetExample)

# example PhyloExpressionSet using 100 permutations
randomTAI.Matrix &lt;- bootMatrix(PhyloExpressionSetExample, permutations = 100)

# example DivergenceExpressionSet using 100 permutations
randomTDI.Matrix &lt;- bootMatrix(DivergenceExpressionSetExample, permutations = 100)


</code></pre>

<hr>
<h2 id='CollapseReplicates'>Combine Replicates in an ExpressionSet</h2><span id='topic+CollapseReplicates'></span>

<h3>Description</h3>

<p>This function takes an <code>ExpressionSet</code> object storing either a constant or variable number
of biological or technical replicates per stage and collapses replicate expression levels using a defined <code>FUN</code> (window function).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CollapseReplicates(ExpressionSet, nrep, FUN, stage.names = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CollapseReplicates_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="CollapseReplicates_+3A_nrep">nrep</code></td>
<td>
<p>either a numeric value specifying the constant number of replicates per stage or a numeric vector specifying the variable number of replicates for each stage position.</p>
</td></tr>
<tr><td><code id="CollapseReplicates_+3A_fun">FUN</code></td>
<td>
<p>a window function (e.g., <code><a href="base.html#topic+mean">mean</a></code>, <code><a href="stats.html#topic+median">median</a></code>, <code><a href="base.html#topic+max">max</a></code>, <code><a href="base.html#topic+min">min</a></code>, etc.) specifying how replicate expression levels should be collapsed.</p>
</td></tr>
<tr><td><code id="CollapseReplicates_+3A_stage.names">stage.names</code></td>
<td>
<p>a character vector specifying the new names of collapsed stages.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PhyloExpressionSetExample)

# combine the expression levels of the 2 replicates (const) per stage
# using mean as window function and rename new stages: "S1","S2","S3"
CollapseReplicates(ExpressionSet = PhyloExpressionSetExample[1:5,1:8], 
                   nrep          = 2, 
                   FUN           = mean, 
                   stage.names   = c("S1","S2","S3"))

# combine the expression levels of the 2 replicates (stage one), 2 replicates (stage two),
# and 3 replicates (stage three) using mean as window function 
# and rename new stages: "S1","S2","S3"
CollapseReplicates(ExpressionSet = PhyloExpressionSetExample[1:5,1:9], 
                   nrep          = c(2,2,3), 
                   FUN           = mean, 
                   stage.names   = c("S1","S2","S3"))


</code></pre>

<hr>
<h2 id='CombinatorialSignificance'>Compute the Statistical Significance of Each Replicate Combination</h2><span id='topic+CombinatorialSignificance'></span>

<h3>Description</h3>

<p>In case a PhyloExpressionSet or DivergenceExpressionSet stores replicates for each
developmental stage or experiment, this function allows to compute the p-values quantifying
the statistical significance of the underlying pattern for all combinations of replicates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CombinatorialSignificance(
  ExpressionSet,
  replicates,
  TestStatistic = "FlatLineTest",
  permutations = 1000,
  parallel = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CombinatorialSignificance_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="CombinatorialSignificance_+3A_replicates">replicates</code></td>
<td>
<p>a numeric vector storing the number of replicates within each developmental stage or experiment.
In case replicate stores only one value, then the function assumes that each developmental stage or experiment
stores the same number of replicates.</p>
</td></tr>
<tr><td><code id="CombinatorialSignificance_+3A_teststatistic">TestStatistic</code></td>
<td>
<p>a string defining the type of test statistics to be used to quantify the statistical significance the present phylotranscriptomics pattern.
Default is <code>TestStatistic</code> = <code>"FlatLineTest"</code>.</p>
</td></tr>
<tr><td><code id="CombinatorialSignificance_+3A_permutations">permutations</code></td>
<td>
<p>a numeric value specifying the number of permutations to be performed for the <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>.</p>
</td></tr>
<tr><td><code id="CombinatorialSignificance_+3A_parallel">parallel</code></td>
<td>
<p>a boolean value specifying whether parallel processing (multicore processing) shall be performed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The intention of this analysis is to validate that there exists no sequence of replicates 
(for all possible combination of replicates) that results in a non-significant pattern,
when the initial pattern with combined replicates was shown to be significant.
</p>
<p>A small Example: 
</p>
<p>Assume PhyloExpressionSet stores 3 developmental stages with 3 replicates measured for each stage.
The 9 replicates in total are denoted as: 1.1, 1.2, 1.3, 2.1, 2.2, 2.3, 3.1, 3.2, 3.3. Now the function computes the
statistical significance of each pattern derived by the corresponding combination of replicates, e.g.
</p>

<ul>
<li><p> 1.1, 2.1, 3.1 -&gt; p-value for combination 1
</p>
</li>
<li><p> 1.1, 2.2, 3.1  -&gt; p-value for combination 2
</p>
</li>
<li><p> 1.1, 2.3, 3.1 -&gt; p-value for combination 3
</p>
</li>
<li><p> 1.2, 2.1, 3.1 -&gt; p-value for combination 4
</p>
</li>
<li><p> 1.2, 2.1, 3.1 -&gt; p-value for combination 5
</p>
</li>
<li><p> 1.2, 2.1, 3.1 -&gt; p-value for combination 6
</p>
</li>
<li><p> 1.3, 2.1, 3.1 -&gt; p-value for combination 7
</p>
</li>
<li><p> 1.3, 2.2, 3.1 -&gt; p-value for combination 8
</p>
</li>
<li><p> 1.3, 2.3, 3.1 -&gt; p-value for combination 9
</p>
</li></ul>

<p>This procedure yields 27 p-values for the <code class="reqn">3^3</code> (<code class="reqn">n_stages^n_replicates</code>) replicate combinations.
</p>
<p>Note, that in case you have a large amount of stages/experiments and a large amount of replicates
the computation time will increase by <code class="reqn">n_stages^n_replicates</code>. For 11 stages and 4 replicates, 4^11 = 4194304 p-values have to be computed. 
Each p-value computation itself is based on a permutation test running with 1000 or more permutations. Be aware that this might take some time.
</p>
<p>The p-value vector returned by this function can then be used to plot the p-values to see
whether an critical value <code class="reqn">\alpha</code> is exeeded or not (e.g. <code class="reqn">\alpha = 0.05</code>).
</p>
<p>The function receives a standard PhyloExpressionSet or DivergenceExpressionSet object and a vector storing the number of replicates present in each stage or experiment. Based on these arguments the function computes all possible replicate combinations using the <code><a href="base.html#topic+expand.grid">expand.grid</a></code> function and performs a permutation test (either a <code><a href="#topic+FlatLineTest">FlatLineTest</a></code> for each replicate combination. The <em>permutation</em> parameter of this function specifies the number of permutations that shall be performed for each permutation test. When all p-values are computed, a numeric vector storing the corresponding p-values for each replicate combination is returned.  
</p>
<p>In other words, for each replicate combination present in the PhyloExpressionSet or DivergenceExpressionSet object, the TAI or TDI pattern of the corresponding replicate combination is tested for its statistical significance based on the underlying test statistic.
</p>
<p>This function is also able to perform all computations in parallel using multicore processing. The underlying statistical tests are written in C++ and optimized for fast computations.
</p>


<h3>Value</h3>

<p>a numeric vector storing the p-values returned by the underlying test statistic for all possible replicate combinations.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+expand.grid">expand.grid</a></code>, <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load a standard PhyloExpressionSet
data(PhyloExpressionSetExample)

# we assume that the PhyloExpressionSetExample 
# consists of 3 developmental stages 
# and 2 replicates for stage 1, 3 replicates for stage 2, 
# and 2 replicates for stage 3
# FOR REAL ANALYSES PLEASE USE: permutations = 1000 or 10000
# BUT NOTE THAT THIS TAKES MUCH MORE COMPUTATION TIME
p.vector &lt;- CombinatorialSignificance(ExpressionSet = PhyloExpressionSetExample, 
                                      replicates    = c(2,3,2), 
                                      TestStatistic = "FlatLineTest", 
                                      permutations  = 10, 
                                      parallel      = FALSE)




</code></pre>

<hr>
<h2 id='DiffGenes'>Differential Gene Expression Analysis</h2><span id='topic+DiffGenes'></span>

<h3>Description</h3>

<p>Detect differentially expressed genes (DEGs) in a standard <code>ExpressionSet</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DiffGenes(
  ExpressionSet,
  nrep,
  method = "foldchange",
  lib.size = NULL,
  p.adjust.method = NULL,
  comparison = NULL,
  alpha = NULL,
  filter.method = NULL,
  n = NULL,
  stage.names = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DiffGenes_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="DiffGenes_+3A_nrep">nrep</code></td>
<td>
<p>either a numeric value specifying the constant number of replicates per stage or a numeric vector specifying the variable number of replicates for each stage position.</p>
</td></tr>
<tr><td><code id="DiffGenes_+3A_method">method</code></td>
<td>
<p>method to detect differentially expressed genes.</p>
</td></tr>
<tr><td><code id="DiffGenes_+3A_lib.size">lib.size</code></td>
<td>
<p>the library sizes to equalize library sizes by quantile-to-quantile normalization (see <code><a href="edgeR.html#topic+equalizeLibSizes">equalizeLibSizes</a></code>).</p>
</td></tr>
<tr><td><code id="DiffGenes_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>p value correction method that is passed to <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>.
Available options are:</p>

<ul>
<li> <p><code>p.adjust.method = "BH"</code> (Benjamini-Hochberg correction)
</p>
</li>
<li> <p><code>p.adjust.method = "bonferroni"</code> (Bonferroni correction)
</p>
</li>
<li> <p><code>p.adjust.method = "holm"</code>
</p>
</li>
<li> <p><code>p.adjust.method = "hochberg"</code>
</p>
</li>
<li> <p><code>p.adjust.method = "hommel"</code>
</p>
</li>
<li> <p><code>p.adjust.method = "BY"</code>
</p>
</li>
<li> <p><code>p.adjust.method = "fdr"</code>
</p>
</li></ul>

<p>If <code>p.adjust.method = NULL</code> (Default) then no p-value correction is performed.</p>
</td></tr>
<tr><td><code id="DiffGenes_+3A_comparison">comparison</code></td>
<td>
<p>a character string specifying whether genes having fold-change or p-values
below, above, or below AND above (both) the <code>alpha</code> value should be excluded from the dataset.
In case <code>comparison = "both"</code> is chosen, the <code>cut.off</code> argument must be a two dimensional vector defining the lower <code>alpha</code> value at the first position and the upper <code>alpha</code> value
at the second position.</p>
</td></tr>
<tr><td><code id="DiffGenes_+3A_alpha">alpha</code></td>
<td>
<p>a numeric value specifying the cut-off value above which Genes fulfilling the corresponding fold-change, log-fold-change, or p-value should be retained and returned by <code>DiffGenes</code>.</p>
</td></tr>
<tr><td><code id="DiffGenes_+3A_filter.method">filter.method</code></td>
<td>
<p>a method how to <code>alpha</code> values in multiple stages. Options are <code>"const"</code>, <code>"min-set"</code>, and <code>"n-set"</code>.</p>
</td></tr>
<tr><td><code id="DiffGenes_+3A_n">n</code></td>
<td>
<p>a numeric value for <code>method = "n-set"</code>.</p>
</td></tr>
<tr><td><code id="DiffGenes_+3A_stage.names">stage.names</code></td>
<td>
<p>a character vector specifying the new names of collapsed stages.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All methods to perform dection of differentially expressed genes assume that your input
dataset has been normalized before passing it to <em>DiffGenes</em>. For RNA-Seq data
<em>DiffGenes</em> assumes that the libraries have been normalized to have the same size, i.e.,
to have the same expected column sum under the null hypothesis. If this isn't the case
please run <code><a href="edgeR.html#topic+equalizeLibSizes">equalizeLibSizes</a></code> before calling <em>DiffGenes</em>. 
</p>
<p>Available methods for the detection of differentially expressed genes:
</p>

<ul>
<li> <p><code>method = "foldchange"</code>: ratio of replicate geometric means between developmental stages.
Here, the <em>DiffGenes</em> functions assumes that absolute expression levels are stored in your input <code>ExpresisonSet</code>.
</p>
</li>
<li> <p><code>method = "log-foldchange"</code>: difference of replicate arithmetic means between developmental stages. Here, the <em>DiffGenes</em> functions assumes that <em>log2a</em>
transformed expression levels are stored in your input <code>ExpresisonSet</code>.
</p>
</li>
<li> <p><code>method = "t.test"</code>: Welch t.test between replicate expression levels of two samples.
</p>
</li>
<li> <p><code>method = "wilcox.test"</code>: Wilcoxon Rank Sum Test between replicate expression levels of two samples.
</p>
</li>
<li> <p><code>method = "doubletail"</code>: Computes two-sided p-values by doubling the smaller tail probability (see <code><a href="edgeR.html#topic+exactTestDoubleTail">exactTestDoubleTail</a></code> for details).
</p>
</li>
<li> <p><code>method = "smallp"</code>: Performs the method of small probabilities as proposed by Robinson and Smyth (2008) (see <code><a href="edgeR.html#topic+exactTestBySmallP">exactTestBySmallP</a></code> for details).
</p>
</li>
<li> <p><code>method = "deviance"</code>: Uses the deviance goodness of fit statistics to define the rejection region, and is therefore equivalent to a conditional likelihood ratio test (see <code>edgeR</code> package for details).
</p>
</li></ul>

<p>Exclude non differentially expressed genes from the result dataset:
</p>
<p>When specifying the <code>alpha</code> argument you furthermore, need to specify the <code>filter.method</code> to decide how non differentially expressed genes should be classified in multiple sample comparisons and which genes should be retained in the final dataset returned by <code>DiffGenes</code>. In other words, all genes &lt; <code>alpha</code> based on the following <code>filter.method</code> are removed from the result dataset.
</p>
<p>Following extraction criteria are implemented in this function: 
</p>

<ul>
<li> <p><code>const</code>: all genes that have at least one sample comparison that undercuts or exceeds the <code>alpha</code> value <code>cut.off</code> will be excluded from the <code>ExpressionSet</code>. Hence, for a 7 stage <code>ExpressionSet</code> genes passing the <code>alpha</code> threshold in 6 stages will be retained in the <code>ExpressionSet</code>.
</p>
</li>
<li> <p><code>min-set</code>: genes passing the <code>alpha</code> value in <code>ceiling(n/2)</code> stages will be retained in the <code>ExpressionSet</code>, where <em>n</em> is the number of stages in the <code>ExpressionSet</code>.
</p>
</li>
<li> <p><code>n-set</code>: genes passing the <code>alpha</code> value in <code>n</code> stages will be retained in the <code>ExpressionSet</code>. Here, the argument <code>n</code> needs to be specified.
</p>
</li></ul>



<h3>Note</h3>

<p>In case input <code>ExpressionSet</code> objects store 0 values, internally all expression levels are 
shifted by <code>+1</code> to allow sufficient fold-change and p-value computations. Additionally, a warning
is printed to the console in case expression levels have been automatically shifted.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Expressed">Expressed</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)

# Detection of DEGs using the fold-change measure
DEGs &lt;- DiffGenes(ExpressionSet = PhyloExpressionSetExample[ ,1:8],
                  nrep          = 2,
                  comparison    = "below",
                  method        = "foldchange",
                  stage.names   = c("S1","S2","S3"))


head(DEGs)


# Detection of DEGs using the log-fold-change measure
# when choosing method = "log-foldchange" it is assumed that
# your input expression matrix stores log2 expression levels 
log.DEGs &lt;- DiffGenes(ExpressionSet = tf(PhyloExpressionSetExample[1:5,1:8],log2),
                      nrep          = 2,
                      comparison    = "below",
                      method        = "log-foldchange",
                      stage.names   = c("S1","S2","S3"))


head(log.DEGs)


# Remove fold-change values &lt; 2 from the dataset:

## first have a look at the range of fold-change values of all genes 
apply(DEGs[ , 3:8],2,range)

# now remove genes undercutting the alpha = 2 threshold
# hence, remove genes having p-values &lt;= 0.05 in at
# least one sample comparison
DEGs.alpha &lt;- DiffGenes(ExpressionSet = PhyloExpressionSetExample[1:250 ,1:8],
                        nrep          = 2,
                        method        = "t.test",
                        alpha         = 0.05,
                        comparison    = "above",
                        filter.method = "n-set",
                        n             = 1,
                        stage.names   = c("S1","S2","S3"))

# now again have a look at the range and find
# that fold-change values of 2 are the min value
apply(DEGs.alpha[ , 3:5],2,range)

# now check whether each example has at least one stage with a p-value &lt;= 0.05
head(DEGs.alpha)

</code></pre>

<hr>
<h2 id='DivergenceExpressionSetExample'>An Example DivergenceExpressionSet Data Set</h2><span id='topic+DivergenceExpressionSetExample'></span>

<h3>Description</h3>

<p>A standard DivergenceExpressionSet is a <code><a href="base.html#topic+data.frame">data.frame</a></code> consisting of a standardized sequence of columns
to store the age information for each gene and its corresponding gene expression profile.
</p>
<p>The standard is defined as follows:
</p>
<p>Divergencestratum | GeneID | Expression-level 1 | ... | Expression-level N
</p>


<h3>Details</h3>

<p>This example DivergenceExpressionSet dataset covers 7 developmental stages of Arabidopsis thaliana embryo development.
The initial gene expression dataset was published by Xiang et al., 2011 (see references section) 
and was then used by Quint et al., 2012 (see references section) to assign sequence divergence values (divergence strata) to each gene expression profile.
</p>


<h3>Value</h3>

<p>a standard DivergenceExpressionSet object.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Source</h3>

<p>http://www.plantphysiol.org/content/156/1/346/suppl/DC1
</p>


<h3>References</h3>

<p>Quint M et al. 2012. &quot;A transcriptomic hourglass in plant embryogenesis&quot;. Nature (490): 98-101.
Supplementary Table 2 : http://www.nature.com/nature/journal/v490/n7418/full/nature11394.html
</p>
<p>Xiang D et al. 2011. &quot;Genome-Wide Analysis Reveals Gene Expression and Metabolic Network Dynamics during Embryo Development in Arabidopsis&quot;. Plant Physiology (156): 346-356.
Supplemental Table 1 : http://www.plantphysiol.org/content/156/1/346/suppl/DC1
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PhyloExpressionSetExample">PhyloExpressionSetExample</a></code>
</p>

<hr>
<h2 id='EarlyConservationTest'>Perform Reductive Early Conservation Test</h2><span id='topic+EarlyConservationTest'></span>

<h3>Description</h3>

<p>The <em>Reductive Early Conservation Test</em> aims to statistically evaluate the
existence of a monotonically increasing phylotranscriptomic pattern based on <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> computations.
The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern) 
does not follow an early conservation like pattern. A p-value &lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does
indeed follow an early conservation (low-high-high) shape.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EarlyConservationTest(
  ExpressionSet,
  modules = NULL,
  permutations = 1000,
  lillie.test = FALSE,
  plotHistogram = FALSE,
  runs = 10,
  parallel = FALSE,
  gof.warning = FALSE,
  custom.perm.matrix = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EarlyConservationTest_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="EarlyConservationTest_+3A_modules">modules</code></td>
<td>
<p>a list storing three elements: early, mid, and late. Each element expects a numeric
vector specifying the developmental stages or experiments that correspond to each module. 
For example, <code>module</code> = list(early = 1:2, mid = 3:5, late = 6:7) devides a dataset 
storing seven developmental stages into 3 modules.</p>
</td></tr>
<tr><td><code id="EarlyConservationTest_+3A_permutations">permutations</code></td>
<td>
<p>a numeric value specifying the number of permutations to be performed for the <code>ReductiveHourglassTest</code>.</p>
</td></tr>
<tr><td><code id="EarlyConservationTest_+3A_lillie.test">lillie.test</code></td>
<td>
<p>a boolean value specifying whether the Lilliefors Kolmogorov-Smirnov Test shall be performed to quantify the goodness of fit.</p>
</td></tr>
<tr><td><code id="EarlyConservationTest_+3A_plothistogram">plotHistogram</code></td>
<td>
<p>a boolean value specifying whether a <em>Lillifor's Kolmogorov-Smirnov-Test</em> 
shall be performed to test the goodness of fit of the approximated distribution, as well as additional plots quantifying the significance
of the observed phylotranscriptomic pattern.</p>
</td></tr>
<tr><td><code id="EarlyConservationTest_+3A_runs">runs</code></td>
<td>
<p>specify the number of runs to be performed for goodness of fit computations, in case <code>plotHistogram</code> = <code>TRUE</code>.
In most cases <code>runs</code> = 100 is a reasonable choice. Default is <code>runs</code> = 10 (because it takes less computation time for demonstration purposes).</p>
</td></tr>
<tr><td><code id="EarlyConservationTest_+3A_parallel">parallel</code></td>
<td>
<p>performing <code>runs</code> in parallel (takes all cores of your multicore machine).</p>
</td></tr>
<tr><td><code id="EarlyConservationTest_+3A_gof.warning">gof.warning</code></td>
<td>
<p>a logical value indicating whether non significant goodness of fit results should be printed as warning. Default is <code>gof.warning = FALSE</code>.</p>
</td></tr>
<tr><td><code id="EarlyConservationTest_+3A_custom.perm.matrix">custom.perm.matrix</code></td>
<td>
<p>a custom <code><a href="#topic+bootMatrix">bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic. Default is <code>custom.perm.matrix = NULL</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <em>reductive early conservation test</em> is a permutation test based on the following test statistic. 
</p>
<p>(1) A set of developmental stages is partitioned into three modules - early, mid, and late - based on prior biological knowledge.
</p>
<p>(2) The mean <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> value for each of the three modules T_early, T_mid, and T_late are computed. 
</p>
<p>(3) The two differences D1 = T_mid - T_early and D2 = T_late - T_early are calculated.
</p>
<p>(4) The minimum D_min of D1 and D2 is computed as final test statistic of the reductive hourglass test.
</p>
<p>In order to determine the statistical significance of an observed minimum difference D_min 
the following permutation test was performed. Based on the <code><a href="#topic+bootMatrix">bootMatrix</a></code> D_min 
is calculated from each of the permuted <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> profiles, 
approximated by a Gaussian distribution with method of moments estimated parameters returned by <code><a href="fitdistrplus.html#topic+fitdist">fitdist</a></code>, 
and the corresponding p-value is computed by <code><a href="stats.html#topic+pnorm">pnorm</a></code> given the estimated parameters of the Gaussian distribution. 
The <em>goodness of fit</em> for the random vector <em>D_min</em> is statistically quantified by an Lilliefors (Kolmogorov-Smirnov) test 
for normality.
</p>
<p>In case the parameter <em>plotHistogram = TRUE</em>, a multi-plot is generated showing:
</p>
<p>(1) A Cullen and Frey skewness-kurtosis plot generated by <code><a href="fitdistrplus.html#topic+descdist">descdist</a></code>. 
This plot illustrates which distributions seem plausible to fit the resulting permutation vector D_min. 
In the case of the <em>reductive early conservation test</em> a normal distribution seemed plausible.
</p>
<p>(2) A histogram of D_min combined with the density plot is plotted. D_min is then fitted by a normal distribution. 
The corresponding parameters are estimated by <em>moment matching estimation</em> using the <code><a href="fitdistrplus.html#topic+fitdist">fitdist</a></code> function.
</p>
<p>(3) A plot showing the p-values for N independent runs to verify that a specific p-value is biased by a specific permutation order.
</p>
<p>(4) A barplot showing the number of cases in which the underlying goodness of fit (returned by Lilliefors (Kolmogorov-Smirnov) test 
for normality) has shown to be significant (<code>TRUE</code>) or not significant (<code>FALSE</code>). 
This allows to quantify the permutation bias and their implications on the goodness of fit.
</p>


<h3>Value</h3>

<p>a list object containing the list elements:
</p>
<p><code>p.value</code> : the p-value quantifying the statistical significance (low-high-high pattern) of the given phylotranscriptomics pattern.
</p>
<p><code>std.dev</code> : the standard deviation of the N sampled phylotranscriptomics patterns for each developmental stage S.
</p>
<p><code>lillie.test</code> : a boolean value specifying whether the <em>Lillifors KS-Test</em> returned a p-value &gt; 0.05, 
which indicates that fitting the permuted scores with a normal distribution seems plausible.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>
<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.
</p>
<p>Piasecka B, Lichocki P, Moretti S, et al. (2013) <em>The hourglass and the early conservation models co-existing
patterns of developmental constraints in vertebrates</em>. PLoS Genet. 9(4): e1003476.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ecScore">ecScore</a></code>, <code><a href="#topic+bootMatrix">bootMatrix</a></code>, <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>,<code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>, <code><a href="#topic+ReverseHourglassTest">ReverseHourglassTest</a></code>, <code><a href="#topic+PlotSignature">PlotSignature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)

# perform the early conservation test for a PhyloExpressionSet
# here the prior biological knowledge is that stages 1-2 correspond to module 1 = early,
# stages 3-5 to module 2 = mid (phylotypic module), and stages 6-7 correspond to
# module 3 = late
EarlyConservationTest(PhyloExpressionSetExample,
                       modules = list(early = 1:2, mid = 3:5, late = 6:7), 
                       permutations = 1000)


# use your own permutation matrix based on which p-values (EarlyConservationTest)
# shall be computed
custom_perm_matrix &lt;- bootMatrix(PhyloExpressionSetExample,100)

EarlyConservationTest(PhyloExpressionSetExample,
                       modules = list(early = 1:2, mid = 3:5, late = 6:7), 
                       custom.perm.matrix = custom_perm_matrix)
                       
</code></pre>

<hr>
<h2 id='ecScore'>Compute the Hourglass Score for the EarlyConservationTest</h2><span id='topic+ecScore'></span>

<h3>Description</h3>

<p>This function computes the EarlyConservationTest score for a given <code><a href="#topic+TAI">TAI</a></code>
or <code><a href="#topic+TDI">TDI</a></code> pattern.
</p>
<p>The reductive early conservation test is a permutation test based on the following test statistic. 
</p>
<p>- A set of developmental stages is partitioned into three modules - early, mid, and late - based on prior biological knowledge.
</p>
<p>- The mean <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> value for each of the three modules T_early, T_mid, and T_late are computed. 
</p>
<p>- The two differences D1 = T_mid - T_early and D2 = T_late - T_early are calculated.
</p>
<p>- The minimum D_min of D1 and D2 is computed as final test statistic of the reductive hourglass test.
</p>
<p>This function <em>ecScore</em> computes the <em>D_min</em> value for a given <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code>
stored in the <code>age_vals</code> argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ecScore(age_vals, early, mid, late)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ecScore_+3A_age_vals">age_vals</code></td>
<td>
<p>a numeric vector containing <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> values for each developmental stage s.</p>
</td></tr>
<tr><td><code id="ecScore_+3A_early">early</code></td>
<td>
<p>a numeric vector storing the numeric stage values that correspond to the early phase of development.</p>
</td></tr>
<tr><td><code id="ecScore_+3A_mid">mid</code></td>
<td>
<p>a numeric vector storing the numeric stage values that correspond to the middle phase of development.</p>
</td></tr>
<tr><td><code id="ecScore_+3A_late">late</code></td>
<td>
<p>a numeric vector storing the numeric stage values that correspond to the late phase of development.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric value representing the early conservation score.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code>, <code><a href="#topic+TAI">TAI</a></code>, <code><a href="#topic+TDI">TDI</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # read standard phylotranscriptomics data
 data(PhyloExpressionSetExample)
 data(DivergenceExpressionSetExample)

 # Example PhyloExpressionSet:

 # compute the TAI profile
 TAIs &lt;- TAI(PhyloExpressionSetExample)

 # compute the early conservation score for the TAI profile
 ec_score &lt;- ecScore(age_vals = TAIs,early = 1:2,mid = 3:5,late = 6:7)


 # Example DivergenceExpressionSet:

 # compute the TDI profile
 TDIs &lt;- TDI(DivergenceExpressionSetExample)

 # compute the early conservation score for the TDI profile
 ec_score &lt;- ecScore(age_vals = TDIs,early = 1:2,mid = 3:5,late = 6:7)
 
 # compute ecScore() vector from bootMatrix()
 apply(bootMatrix(PhyloExpressionSetExample,10),1,ecScore,early = 1:2,mid = 3:5,late = 6:7)
 
 
</code></pre>

<hr>
<h2 id='EnrichmentTest'>Phylostratum or Divergence Stratum Enrichment of a given Gene Set based on Fisher's Test</h2><span id='topic+EnrichmentTest'></span>

<h3>Description</h3>

<p>This function computes the significance of enriched (over or underrepresented) Phylostrata or Divergence Strata within an input <code>test.set</code> based on the <code><a href="stats.html#topic+fisher.test">fisher.test</a></code>. Please concult <code><a href="#topic+PlotEnrichment">PlotEnrichment</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EnrichmentTest(
  ExpressionSet,
  test.set,
  use.only.map = FALSE,
  measure = "log-foldchange",
  complete.bg = TRUE,
  epsilon = 1e-05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EnrichmentTest_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="EnrichmentTest_+3A_test.set">test.set</code></td>
<td>
<p>a character vector storing the gene ids for which PS/DS enrichment analyses should be performed.</p>
</td></tr>
<tr><td><code id="EnrichmentTest_+3A_use.only.map">use.only.map</code></td>
<td>
<p>a logical value indicating whether instead of a standard <code>ExpressionSet</code> only a <code>Phylostratigraphic Map</code> or <code>Divergene Map</code> is passed to this function.</p>
</td></tr>
<tr><td><code id="EnrichmentTest_+3A_measure">measure</code></td>
<td>
<p>a character string specifying the measure that should be used to quantify over and under representation of PS/DS. Measures can either be <code>measure = "foldchange"</code> (odds) or <code>measure = "log-foldchange"</code> (log-odds).</p>
</td></tr>
<tr><td><code id="EnrichmentTest_+3A_complete.bg">complete.bg</code></td>
<td>
<p>a logical value indicating whether the entire background set of the input ExpressionSet should be considered when performing Fisher's exact test (<code>complete.bg = TRUE</code>) or whether genes that are stored in test.set should be excluded from the background set before performing Fisher's exact test (<code>complete.bg = FALSE</code>).</p>
</td></tr>
<tr><td><code id="EnrichmentTest_+3A_epsilon">epsilon</code></td>
<td>
<p>a small value to shift values by epsilon to avoid log(0) computations.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotEnrichment">PlotEnrichment</a></code>, <code><a href="stats.html#topic+fisher.test">fisher.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)

set.seed(123)
test_set &lt;- sample(PhyloExpressionSetExample[ , 2],1000)

E.Result &lt;- EnrichmentTest(ExpressionSet = PhyloExpressionSetExample,
                           test.set      = test_set ,
                           measure       = "log-foldchange")
                           
# get the log-fold change table
E.Result$enrichment.matrix

# get P-values for the enrichment significance for each Phylostratum
E.Result$p.values

</code></pre>

<hr>
<h2 id='Expressed'>Filter for Expressed Genes</h2><span id='topic+Expressed'></span>

<h3>Description</h3>

<p>This function takes an ExpressionSet object and removes genes from the gene expression matrix that
have an expression level below, above, or below AND above a defined <code>cut.off</code> value. Hence, this function allows to remove
genes that have been defined as <em>not expressed</em> or <em>outliers</em> and returns an <code>ExpressionSet</code> retaining only expressed genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Expressed(
  ExpressionSet,
  cut.off,
  method = "const",
  comparison = "below",
  n = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Expressed_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="Expressed_+3A_cut.off">cut.off</code></td>
<td>
<p>a numeric value specifying the expression cut-off to define genes as <em>not expressed</em> (<code>comparison = "below"</code>) , <em>outliers</em> (<code>comparison = "above"</code>), or both (<code>comparison = "both"</code>). See <code>comparison</code> for details. In case <code>comparison = "both"</code>, the <code>cut.off</code> argument must be a two dimensional vector defining the lower <code>cut.off</code> value at the first position and the upper <code>cut.off</code> value at the second position.</p>
</td></tr>
<tr><td><code id="Expressed_+3A_method">method</code></td>
<td>
<p>a method defining how to treat gene expression values in multiple stages. The corresponding method that is chosen allows to control the stage-wise fulfillment of the threshold criteria. Options are <code>"const"</code>, <code>"min-set"</code>, and <code>"n-set"</code>.</p>
</td></tr>
<tr><td><code id="Expressed_+3A_comparison">comparison</code></td>
<td>
<p>a character string specifying whether genes having expression levels
below, above, or below AND above (both) the <code>cut.off</code> value should be excluded from the dataset.
In case <code>comparison = "both"</code> is chosen, the <code>cut.off</code> argument must be a two dimensional vector defining the lower <code>cut.off</code> value at the first position and the upper <code>cut.off</code> value
at the second position.</p>
</td></tr>
<tr><td><code id="Expressed_+3A_n">n</code></td>
<td>
<p>a numeric value for <code>method = "n-set"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This filter function allows users to remove genes from the <code>ExpressionSet</code> object that undercut or exceed a certain expression level <code>cut.off</code>.
</p>
<p>Following extraction criteria are implemented in this function: 
</p>

<ul>
<li> <p><code>const</code>: all genes that have at least one stage that undercuts or exceeds the expression <code>cut.off</code> will be excluded from the <code>ExpressionSet</code>. Hence, for a 7 stage <code>ExpressionSet</code> genes passing the expression level <code>cut.off</code> in 6 stages will be retained in the <code>ExpressionSet</code>.
</p>
</li>
<li> <p><code>min-set</code>: genes passing the expression level <code>cut.off</code> in <code>ceiling(n/2)</code> stages will be retained in the <code>ExpressionSet</code>, where <em>n</em> is the number of stages in the <code>ExpressionSet</code>.
</p>
</li>
<li> <p><code>n-set</code>: genes passing the expression level <code>cut.off</code> in <code>n</code> stages will be retained in the <code>ExpressionSet</code>. Here, the argument <code>n</code> needs to be specified.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PhyloExpressionSetExample)

# remove genes that have an expression level below 8000 
# in at least one developmental stage
FilterConst &lt;- Expressed(ExpressionSet = PhyloExpressionSetExample, 
                         cut.off       = 8000, 
                         method        = "const",
                         comparison    = "below")
                              
dim(FilterConst) # check number of retained genes

# remove genes that have an expression level below 8000 
# in at least 3 developmental stages 
# (in this case: ceiling(7/2) = 4 stages fulfilling the cut-off criteria)
FilterMinSet &lt;- Expressed(ExpressionSet = PhyloExpressionSetExample, 
                          cut.off       = 8000, 
                          method        = "min-set",
                          comparison    = "below")
                               
dim(FilterMinSet) # check number of retained genes

# remove genes that have an expression level below 8000 
# in at least 5 developmental stages (in this case: n = 2 stages fulfilling the criteria)
FilterNSet &lt;- Expressed(ExpressionSet = PhyloExpressionSetExample, 
                        cut.off       = 8000, 
                        method        = "n-set",
                        comparison    = "below",
                        n             = 2)
                               
dim(FilterMinSet) # check number of retained genes



# remove expression levels that exceed the cut.off criteria
FilterMinSet &lt;- Expressed(ExpressionSet = PhyloExpressionSetExample, 
                          cut.off       = 12000, 
                          method        = "min-set",
                          comparison    = "above")
                               
dim(FilterMinSet) # check number of retained genes


# remove expression levels that undercut AND exceed the cut.off criteria
FilterMinSet &lt;- Expressed(ExpressionSet = PhyloExpressionSetExample, 
                          cut.off       = c(8000,12000), 
                          method        = "min-set",
                          comparison    = "both")
                               
dim(FilterMinSet) # check number of retained genes


</code></pre>

<hr>
<h2 id='FlatLineTest'>Perform Flat Line Test</h2><span id='topic+FlatLineTest'></span>

<h3>Description</h3>

<p>This function quantifies the statistical significance of an observed phylotranscriptomic pattern. In detail, the <em>Flat Line Test</em> quantifies any significant deviation of an observed phylotranscriptomic pattern from a flat line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FlatLineTest(
  ExpressionSet,
  permutations = 1000,
  plotHistogram = FALSE,
  runs = 10,
  parallel = FALSE,
  custom.perm.matrix = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FlatLineTest_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="FlatLineTest_+3A_permutations">permutations</code></td>
<td>
<p>a numeric value specifying the number of permutations that shall be performed for the <em>FlatLineTest</em>.</p>
</td></tr>
<tr><td><code id="FlatLineTest_+3A_plothistogram">plotHistogram</code></td>
<td>
<p>a logical value indicating whether a detailed statistical analysis concerning the goodness of fit should be performed.</p>
</td></tr>
<tr><td><code id="FlatLineTest_+3A_runs">runs</code></td>
<td>
<p>specify the number of runs to be performed for goodness of fit computations. In most cases runs = 100 is a reasonable choice.</p>
</td></tr>
<tr><td><code id="FlatLineTest_+3A_parallel">parallel</code></td>
<td>
<p>performing <code>runs</code> in parallel (takes all cores of your multicore machine).</p>
</td></tr>
<tr><td><code id="FlatLineTest_+3A_custom.perm.matrix">custom.perm.matrix</code></td>
<td>
<p>a custom <code><a href="#topic+bootMatrix">bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic. Default is <code>custom.perm.matrix = NULL</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally the function performs N phylotranscriptomics pattern computations (<code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code>) based on sampled PhyloExpressionSets or DivergenceExpressionSets (see <code><a href="#topic+bootMatrix">bootMatrix</a></code>). 
The test statistics is being developed as follows:
</p>
<p>The variance <em>V_pattern</em> of the S phylotranscriptomics values defines the test statistic for the <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>. 
The basic assumption is, that the variance of a flat line should be equivalent to zero for a perfect flat line. 
Any deviation from a flat line can be measured with a variance value &gt; 0. 
</p>
<p>To determine the null distribution of <em>V_p</em>, all PS or DS values within each developmental stage s are randomly permuted,
S surrogate phylotranscriptomics values are computed from this permuted dataset, and a surrogate value of <em>V_p</em> is
computed from these S phylotranscriptomics values. This permutation process is repeated N times, yielding a histogram of <em>V_p</em>. 
</p>
<p>After applying a <em>Lilliefors Kolmogorov-Smirnov Test for gamma distribution</em>, <em>V_p</em> is approximated by a <em>gamma distribution</em>. 
The two parameters of the <em>gamma distribution</em> are estimated by the function <code><a href="fitdistrplus.html#topic+fitdist">fitdist</a></code> from the <span class="pkg">fitdistrplus</span> package by <em>moment matching estimation</em>. 
The fitted <em>gamma distribution</em> is considered the null distribution of <em>V_pattern</em>, and the p-value of the observed value of <em>V_p</em> 
is computed from this null distribution.
</p>
<p>In case the parameter <em>plotHistogram = TRUE</em>, a multi-plot is generated showing:
</p>
<p>(1) A Cullen and Frey skewness-kurtosis plot generated by <code><a href="fitdistrplus.html#topic+descdist">descdist</a></code>).
</p>
<p>(2) A histogram of V_p combined with the density plot using the Method of Moments estimated parameters returned by the <code><a href="fitdistrplus.html#topic+fitdist">fitdist</a></code> function using a gamma distribution.
</p>
<p>(3) A plot showing the p-values for N independent runs to verify that a specific p-value is biased by a specific permutation order.
</p>
<p>The <em>goodness of fit</em> for the random vector <em>V_p</em> is quantified statistically by an adapted Lilliefors (Kolmogorov-Smirnov) test for gamma distributions.
</p>


<h3>Value</h3>

<p>a list object containing the list elements:
</p>

<ul>
<li> <p><code>p.value</code> the p-value quantifying the statistical significance (deviation from a flat line) of the given phylotranscriptomics pattern.
</p>
</li>
<li> <p><code>std.dev</code> the standard deviation of the N sampled phylotranscriptomics patterns for each developmental stage S.
</p>
</li></ul>



<h3>Note</h3>

<p>In case there are extreme outlier expression values stored in the dataset (PhyloExpressionSet or DivergenceExpressionSet),
the internal <code><a href="fitdistrplus.html#topic+fitdist">fitdist</a></code> function that is based on the <code><a href="#topic+bootMatrix">bootMatrix</a></code> output might return a warning:
&quot;In densfun(x, parm[1], parm[2], ...) : NaNs were produced&quot; which indicates that permutation results caused by extreme outlier expression values 
that could not be fitted accordingly. This warning will not be printed out when the corresponding outlier values are extracted from the dataset.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>
<p>Quint M et al. (2012). A transcriptomic hourglass in plant embryogenesis. Nature (490): 98-101.
</p>
<p>M. L. Delignette-Muller, R. Pouillot, J.-B. Denis and C. Dutang (2014), fitdistrplus: help to fit of a parametric distribution to non-censored or censored data.
</p>
<p>Cullen AC and Frey HC (1999) Probabilistic techniques in exposure assessment. Plenum Press, USA, pp. 81-159.
</p>
<p>Evans M, Hastings N and Peacock B (2000) Statistical distributions. John Wiley and Sons Inc.
</p>
<p>Sokal RR and Rohlf FJ (1995) Biometry. W.H. Freeman and Company, USA, pp. 111-115.
</p>
<p>Juergen Gross and bug fixes by Uwe Ligges (2012). nortest: Tests for Normality. R package version 1.0-2. 
</p>
<p>http://CRAN.R-project.org/package=nortest
</p>
<p>Dallal, G.E. and Wilkinson, L. (1986): An analytic approximation to the distribution of Lilliefors test for normality. The American Statistician, 40, 294-296.
</p>
<p>Stephens, M.A. (1974): EDF statistics for goodness of fit and some comparisons. Journal of the American Statistical Association, 69, 730-737.
</p>
<p>http://stackoverflow.com/questions/4290081/fitting-data-to-distributions?rq=1
</p>
<p>http://stats.stackexchange.com/questions/45033/can-i-use-kolmogorov-smirnov-test-and-estimate-distribution-parameters
</p>
<p>http://cran.r-project.org/doc/contrib/Ricci-distributions-en.pdf
</p>
<p>http://cran.r-project.org/doc/contrib/Ricci-distributions-en.pdf
</p>


<h3>See Also</h3>

<p><code><a href="#topic+TAI">TAI</a></code>, <code><a href="#topic+TDI">TDI</a></code>, <code><a href="#topic+PlotPattern">PlotPattern</a></code>, <code><a href="#topic+bootMatrix">bootMatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# read standard phylotranscriptomics data
data(PhyloExpressionSetExample)

# example PhyloExpressionSet using 1000 permutations
FlatLineTest(PhyloExpressionSetExample, 
             permutations  = 1000, 
             plotHistogram = FALSE)

# use your own permutation matrix based on which p-values (FlatLineTest)
# shall be computed
custom_perm_matrix &lt;- bootMatrix(PhyloExpressionSetExample,100)

FlatLineTest(PhyloExpressionSetExample,
             custom.perm.matrix = custom_perm_matrix)

</code></pre>

<hr>
<h2 id='geom.mean'>Geometric Mean</h2><span id='topic+geom.mean'></span>

<h3>Description</h3>

<p>This function computes the geometric mean of a numeric input vector <code>x</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom.mean(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geom.mean_+3A_x">x</code></td>
<td>
<p>a numeric vector for which geometric mean computations shall be performed.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- 1:10

geom.mean(x)

</code></pre>

<hr>
<h2 id='GroupDiffs'>Quantify the significant differences between gene expression distributions of age groups</h2><span id='topic+GroupDiffs'></span>

<h3>Description</h3>

<p>This function performs a test to quantify the statistical significance between
the global expression level distributions of groups of PS or DS. It therefore, allows users to investigate
significant groups of PS or DS that significantly differ in their gene expression level distibution
within specific developmental stages or experiments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GroupDiffs(
  ExpressionSet,
  Groups = NULL,
  legendName = NULL,
  stat.test = "wilcox.test",
  gene.set = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GroupDiffs_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="GroupDiffs_+3A_groups">Groups</code></td>
<td>
<p>a list containing the phylostrata or divergence strata that correspond 
to the same phylostratum class or divergence class.
For ex. evolutionary old phylostrata: PS1-3 (Class 1) 
and evolutionary young phylostrata: PS4-12 (Class 2). In this case, 
the list could be assigned as, <code>Groups</code> = list(c(1:3), c(4:12)).</p>
</td></tr>
<tr><td><code id="GroupDiffs_+3A_legendname">legendName</code></td>
<td>
<p>a character string specifying whether &quot;PS&quot; or &quot;DS&quot; are used to compute relative expression profiles.</p>
</td></tr>
<tr><td><code id="GroupDiffs_+3A_stat.test">stat.test</code></td>
<td>
<p>the statistical test to quantify PS or DS group differences.</p>
</td></tr>
<tr><td><code id="GroupDiffs_+3A_gene.set">gene.set</code></td>
<td>
<p>a character vector storing the gene ids for which group specific differences shall be statistically quantified.</p>
</td></tr>
<tr><td><code id="GroupDiffs_+3A_...">...</code></td>
<td>
<p>additional plot parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The purpose of this function is to detect groups of PS or DS that significantly differ in their gene expression
level distributions on a global (transcriptome) level. Since relative expression levels (<code><a href="#topic+PlotRE">PlotRE</a></code>) or
PS or DS specific mean expression levels (<code><a href="#topic+PlotMeans">PlotMeans</a></code>) are biased by highly expressed genes,
this function allows users to objectively test the significant difference of transcriptome expression between
groups of PS or DS in a specific developmental stage or experiment.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotGroupDiffs">PlotGroupDiffs</a></code>, <code><a href="#topic+PlotMeans">PlotMeans</a></code>, <code><a href="#topic+PlotRE">PlotRE</a></code>, <code><a href="#topic+PlotBarRE">PlotBarRE</a></code>, <code><a href="#topic+PlotCategoryExpr">PlotCategoryExpr</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)
# perform a Wilcoxon Rank Sum test to statistically quantify the
# difference between PS-Group 1 expression levels versus PS-Group 2
# expression levels
GroupDiffs(ExpressionSet = PhyloExpressionSetExample,
           Groups       = list(group_1 = 1:3,group_2 = 4:12),
           legendName   = "PS")

# quantify the significant difference of a selected set of genes
set.seed(123)
ExampleGeneSet &lt;- sample(PhyloExpressionSetExample[ , 2],5000)  
             
GroupDiffs(ExpressionSet = PhyloExpressionSetExample,
           Groups       = list(group_1 = 1:3,group_2 = 4:12),
           legendName   = "PS",
           gene.set     = ExampleGeneSet)               


</code></pre>

<hr>
<h2 id='harm.mean'>Harmonic Mean</h2><span id='topic+harm.mean'></span>

<h3>Description</h3>

<p>This function computes the harmonic mean of a numeric input vector <code>x</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>harm.mean(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="harm.mean_+3A_x">x</code></td>
<td>
<p>a numeric vector for which harmonic mean computations shall be performed.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- 1:10

harm.mean(x)

</code></pre>

<hr>
<h2 id='is.ExpressionSet'>Test ExpressionSet Standard</h2><span id='topic+is.ExpressionSet'></span>

<h3>Description</h3>

<p>This function tests whether a given ExpressionSet follows the pre-defined PhyloExpressionSet or DivergenceExpressionSet standard.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.ExpressionSet(ExpressionSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.ExpressionSet_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet that shall be tested for format validity.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# read example PhyloExpressionSet
data(PhyloExpressionSetExample)

is.ExpressionSet(PhyloExpressionSetExample)

</code></pre>

<hr>
<h2 id='MatchMap'>Match a Phylostratigraphic Map or Divergence Map with a ExpressionMatrix</h2><span id='topic+MatchMap'></span>

<h3>Description</h3>

<p>This function matches a <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> only storing unique gene ids with a ExpressionMatrix
also storing only unique gene ids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MatchMap(Map, ExpressionMatrix, remove.duplicates = FALSE, accumulate = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MatchMap_+3A_map">Map</code></td>
<td>
<p>a standard <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> object.</p>
</td></tr>
<tr><td><code id="MatchMap_+3A_expressionmatrix">ExpressionMatrix</code></td>
<td>
<p>a standard ExpressionMatrix object.</p>
</td></tr>
<tr><td><code id="MatchMap_+3A_remove.duplicates">remove.duplicates</code></td>
<td>
<p>a logical value indicating whether duplicate gene ids should be removed from the data set.</p>
</td></tr>
<tr><td><code id="MatchMap_+3A_accumulate">accumulate</code></td>
<td>
<p>an accumulation function such as <code>mean()</code>, <code>median()</code>, or <code>min()</code>
to accumulate multiple expression levels that map to the same unique gene id present in the <code>ExpressionMatrix</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In phylotranscriptomics analyses two major techniques are performed to
obtain standard <em>Phylostratigraphic map</em> or <em>Divergence map</em> objects.
</p>
<p>To obtain a <em>Phylostratigraphic Map</em>, <em>Phylostratigraphy</em> (Domazet-Loso et al., 2007) has to be performed. To obtain a <em>Divergence Map</em>, 
orthologous gene detection, Ka/Ks computations, and decilation (Quint et al., 2012; Drost et al., 2015) have to be performed.
</p>
<p>The resulting standard <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> objects consist of 2 colums storing the phylostratum assignment 
or divergence stratum assignment of a given gene in column one, and the corresponding gene id of that gene on columns two.
</p>
<p>A standard ExpressionMatrix is a common gene expression matrix storing the gene ids or probe ids in the first column, and all experiments/stages/replicates in the following columns.
</p>
<p>The <em>MatchMap</em> function takes both standard datasets: <em>Map</em> and <em>ExpressionMatrix</em> to merge both data sets to obtain a standard PhyloExpressionSet or DivergenceExpressionSet object.
</p>
<p>This procedure is analogous to <code><a href="base.html#topic+merge">merge</a></code>, but is customized to the <em>Phylostratigraphic Map</em>, <em>Divergence Map</em>, and <em>ExpressionMatrix</em> standards to allow a faster and more intuitive usage. 
</p>
<p>In case you work with an ExpressionMatrix that stores multiple expression levels for a unique gene id, you
can specify the <code>accumulation</code> argument to accumulate these multiple expression levels to obtain
one expression level for one unique gene.
</p>


<h3>Value</h3>

<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T, Brajkovic J, Tautz D (2007) A phylostratigraphy approach to uncover the genomic history of major adaptations in metazoan lineages. Trends Genet. 23: 533-9.
</p>
<p>Domazet-Loso T, Tautz D (2010) A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns. Nature 468: 815-8.
</p>
<p>Quint M., Drost H.G., Gabel A., Ullrich K.K., Boenn M., Grosse I. (2012) A transcriptomic hourglass in plant embryogenesis. Nature 490: 98-101.
</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>


<h3>Examples</h3>

<pre><code class='language-R'>        
# load a standard PhyloExpressionSet
data(PhyloExpressionSetExample)
        
# in a standard PhyloExpressionSet, 
# column one and column two denote a standard 
# phylostratigraphic map
PhyloMap &lt;- PhyloExpressionSetExample[ , 1:2]
        
# look at the phylostratigraphic map standard
head(PhyloMap)
        
# in a standard PhyloExpressionSet, column two combined 
# with column 3 - N denote a standard ExpressionMatrix
ExpressionMatrixExample &lt;- PhyloExpressionSetExample[ , c(2,3:9)]
        
# these two data sets shall illustrate an example 
# phylostratigraphic map that is returned
# by a standard phylostratigraphy run, and a expression set 
# that is the result of expression data analysis 
# (background correction, normalization, ...)
        
# now we can use the MatchMap function to merge both data sets
# to obtain a standard PhyloExpressionSet
        
PES &lt;- MatchMap(PhyloMap, ExpressionMatrixExample)
        
# note that the function returns a head() 
# of the matched gene ids to enable
# the user to find potential mis-matches
        
# the entire procedure is analogous to merge() 
# with two data sets sharing the same gene ids 
# as column (primary key)
PES_merge &lt;- merge(PhyloMap, ExpressionMatrixExample)
        
        
        
</code></pre>

<hr>
<h2 id='omitMatrix'>Compute TAI or TDI Profiles Omitting a Given Gene</h2><span id='topic+omitMatrix'></span>

<h3>Description</h3>

<p>For each gene i, exclude the corresponding gene i from the global
PhyloExpressionSet or DivergenceExpressionSet and compute the <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> 
profile for the corresponding global PhyloExpressionSet or DivergenceExpressionSet
with excluded gene i. 
</p>
<p>This procedure results in a TAI or TDI profile Matrix storing the TAI or TDI profile for each omitted gene i.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>omitMatrix(ExpressionSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="omitMatrix_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric matrix storing TAI or TDI profile for each omitted gene i.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# read standard phylotranscriptomics data
data(PhyloExpressionSetExample)
data(DivergenceExpressionSetExample)

# example PhyloExpressionSet
omMatrix_ps &lt;- omitMatrix(PhyloExpressionSetExample)

# example DivergenceExpressionSet
omMatrix_ds &lt;- omitMatrix(DivergenceExpressionSetExample)


</code></pre>

<hr>
<h2 id='PhyloExpressionSetExample'>An Example PhyloExpressionSet Data Set</h2><span id='topic+PhyloExpressionSetExample'></span>

<h3>Description</h3>

<p>A standard PhyloExpressionSet is a <code><a href="base.html#topic+data.frame">data.frame</a></code> consisting of a standardized sequence of columns
to store the age information for each gene and its corresponding gene expression profile.
</p>
<p>The standard is defined as follows:
</p>
<p>Phylostratum | GeneID | Expression-level 1 | ... | Expression-level N
</p>


<h3>Details</h3>

<p>This dataset covers 7 developmental stages of Arabidopsis thaliana embryo development.
The initial gene expression dataset was published by Xiang et al., 2011 (see references section) and was then
used by Quint et al., 2012 (see references section) to assign evolutionary ages to each gene expression profile.
</p>


<h3>Value</h3>

<p>a standard PhyloExpressionSet object.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Source</h3>

<p>http://www.plantphysiol.org/content/156/1/346/suppl/DC1
</p>


<h3>References</h3>

<p>Quint M et al. 2012. &quot;A transcriptomic hourglass in plant embryogenesis&quot;. Nature (490): 98-101.
Supplementary Table 2 : http://www.nature.com/nature/journal/v490/n7418/full/nature11394.html
</p>
<p>Xiang D et al. 2011. &quot;Genome-Wide Analysis Reveals Gene Expression and Metabolic Network Dynamics during Embryo Development in Arabidopsis&quot;. Plant Physiology (156): 346-356.
Supplemental Table 1 : http://www.plantphysiol.org/content/156/1/346/suppl/DC1
</p>


<h3>See Also</h3>

<p><code><a href="#topic+DivergenceExpressionSetExample">DivergenceExpressionSetExample</a></code>
</p>

<hr>
<h2 id='PlotBarRE'>Plot Mean Relative Expression Levels as Barplot</h2><span id='topic+PlotBarRE'></span>

<h3>Description</h3>

<p>This function takes a PhyloExpressionSet or DivergenceExpressionSet object as input and computes
for two or more defined phylostratum (divergence stratum) classes the statistical significance of
the differences of mean relative expression of these two (or more) corresponding phylostratum (divergence stratum) classes.
As test-statistic, the function performs a nonparametric <code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code>
based on relative expression values stored within each defined phylostratum class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotBarRE(
  ExpressionSet,
  Groups = NULL,
  wLength = 0.1,
  ratio = FALSE,
  p.adjust.method = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotBarRE_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotBarRE_+3A_groups">Groups</code></td>
<td>
<p>a list containing the phylostrata or divergence-strata that correspond to the same phylostratum class or divergence class.
For ex. evolutionary old phylostrata: PS1-3 (Class 1) and evolutionary young phylostrata: PS4-12 (Class 2). 
In this case, the <code>Groups</code> list could be assigned as, <code>Groups</code> = list(c(1:3), c(4:12)). 
It is also possible to define more than 2 groups of evolutionary ages.
For ex. <code>Groups</code> = list(c(1:3),c(4:8),c(9:12)) would perform a <code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code> to determine the statistical significance
of the evolutionary classes PS1-3, PS4-6, and PS9-12 based on their corresponding mean relative expression levels.</p>
</td></tr>
<tr><td><code id="PlotBarRE_+3A_wlength">wLength</code></td>
<td>
<p>a numeric value defining the whiskers length above the bars. In case there are numerous different phylostratum classes
a smaller <code>wLength</code> parameter should be used for better visualizations.</p>
</td></tr>
<tr><td><code id="PlotBarRE_+3A_ratio">ratio</code></td>
<td>
<p>a boolean value specifying whether the bars in the barplot represent the 
mean relative expression level of phylostrata belonging to the same phylostratum class.
In case <code>ratio</code> = TRUE, the ratio of the mean relative expression level of 
the two phylostrata classes is plotted as lines within the barplot. This parameter can only be used for 2 class comparisons.</p>
</td></tr>
<tr><td><code id="PlotBarRE_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>correction method to adjust p-values for multiple comparisons (see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code> for possible methods).
E.g., <code>p.adjust.method = "BH"</code> (Benjamini &amp; Hochberg (1995)) or <code>p.adjust.method = "bonferroni"</code> (Bonferroni correction).</p>
</td></tr>
<tr><td><code id="PlotBarRE_+3A_...">...</code></td>
<td>
<p>default graphics parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In case a large number of developmental stages is included in the input <code>ExpressionSet</code>,
p-values returned by <code>PlotBarRE</code> should be adjusted for multiple comparisons which can be done
by specifying the <code>p.adjust.method</code> argument.
</p>


<h3>Value</h3>

<p>A barplot comparing Phylostratum-Classes by its mean relative expression levels.
Significant stages are marked by '*' referring to statistically significant differences:
</p>
<p>(1) '*'   = P-Value &lt;= 0.05 
</p>
<p>(2) '**'  = P-Value &lt;= 0.005  
</p>
<p>(3) '***' = P-Value &lt;= 0.0005
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Quint M et al. 2012. &quot;A transcriptomic hourglass in plant embryogenesis&quot;. Nature (490): 98-101.
</p>
<p>Domazet-Loso T. and Tautz D. 2010. &quot;A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns&quot;. Nature (468): 815-818.
</p>
<p>Myles Hollander and Douglas A. Wolfe (1973), Nonparametric Statistical Methods. New York: John Wiley &amp; Sons. Pages 115-120.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RE">RE</a></code>, <code><a href="#topic+REMatrix">REMatrix</a></code>,<code><a href="#topic+PlotRE">PlotRE</a></code>, <code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# read standard phylotranscriptomics data
data(PhyloExpressionSetExample)
data(DivergenceExpressionSetExample)

# example PhyloExpressionSet
PlotBarRE(ExpressionSet = PhyloExpressionSetExample,
          Groups        = list(c(1:3), c(4:12)))


# example DivergenceExpressionSet
PlotBarRE(ExpressionSet = DivergenceExpressionSetExample,
          Groups        = list(c(1:5), c(6:10)))


# Perform PlotBarRE() with p-value adjustment method Benjamini &amp; Hochberg (1995)
PlotBarRE(ExpressionSet   = PhyloExpressionSetExample,
          Groups          = list(c(1:3), c(4:12)),
          p.adjust.method = "BH")
       
             
# Example: plot ratio
# the ratio curve visualizes the ratio between bar 1 / bar 2
# the z - axis shows the corresponding ratio value of bar 1 / bar 2
PlotBarRE(ExpressionSet = PhyloExpressionSetExample,
          Groups        = list(c(1:3), c(4:12)),
          ratio         = TRUE)

</code></pre>

<hr>
<h2 id='PlotCategoryExpr'>Plot the Expression Levels of each Age or Divergence Category as Boxplot, Violinplot, or Dotplot</h2><span id='topic+PlotCategoryExpr'></span>

<h3>Description</h3>

<p>This function visualizes the expression level distribution of each phylostratum during each time point or experiment
as boxplot, dot plot, or violin plot enabling users to quantify the age (PS) or divergence (DS) category specific contribution to the
corresponding transcriptome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotCategoryExpr(
  ExpressionSet,
  legendName,
  test.stat = TRUE,
  type = "category-centered",
  distr.type = "boxplot",
  log.expr = FALSE,
  gene.set = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotCategoryExpr_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotCategoryExpr_+3A_legendname">legendName</code></td>
<td>
<p>a character string specifying whether &quot;PS&quot; or &quot;DS&quot; are used to compute relative expression profiles.</p>
</td></tr>
<tr><td><code id="PlotCategoryExpr_+3A_test.stat">test.stat</code></td>
<td>
<p>a logical value indicating whether a Benjamini-Hochberg adjusted <code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code> should be applied to determine
significant differences in age or divergence category specific expression.</p>
</td></tr>
<tr><td><code id="PlotCategoryExpr_+3A_type">type</code></td>
<td>
<p>type of age or divergence category comparison. Specifications can be <code>type = "category-centered"</code> or <code>type = "stage-centered"</code>.</p>
</td></tr>
<tr><td><code id="PlotCategoryExpr_+3A_distr.type">distr.type</code></td>
<td>
<p>format of visualizing age or divergence category specific expression distributions. Either <code>distr.type = "boxplot"</code>, <code>distr.type = "dotplot"</code>, or
<code>distr.type = "violin"</code>.</p>
</td></tr>
<tr><td><code id="PlotCategoryExpr_+3A_log.expr">log.expr</code></td>
<td>
<p>a logical value specifying whether or not expression levels should internally be log2-transformed before visualization.</p>
</td></tr>
<tr><td><code id="PlotCategoryExpr_+3A_gene.set">gene.set</code></td>
<td>
<p>a character vector storing the gene ids for which gene expression levels shall be visualized.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This way of visualizing the gene expression distribution of each age (PS) or divergence (DS) category during
all developmental stages or experiments allows users to detect specific age or divergence categories contributing significant
levels of gene expression to the underlying biological process (transcriptome).
</p>
<p>This quantification allows users to conclude that genes originating in specific PS or DS contribute significantly more to the overall transcriptome
than other genes originating from different PS or DS categories. More specialized analyses such as <code><a href="#topic+PlotMeans">PlotMeans</a></code>, <code><a href="#topic+PlotRE">PlotRE</a></code>,
<code><a href="#topic+PlotBarRE">PlotBarRE</a></code>, etc. will then allow to study the exact mean expression patterns of these age or divergence categories.
</p>
<p>The statistical quantification of differences between expression levels of different age or divergence categories
is done by performing a <code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code> with Benjamini &amp; Hochberg p-value adjustment for multiple comparisons.
</p>

<ul>
<li> <p><code>type = "category-centered"</code> Here, the <code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code> quantifies the differences of gene expression between all combinations of age or divergence categories for each stage or experiment separately. Here, a significant p-value quantifies that there is at least one pairwise comparison for which age or divergence categories significantly differ in their gene expression distribution. This type of analysis allows users to detect stages or experiments that show high diviation between age or divergence category contributions to the overall transcriptome or no significant deviations of age or divergence categories, suggesting equal age or divergence category contributions to the overall transcriptome.   
</p>
</li>
<li> <p><code>type = "stage-centered"</code> Here, the <code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code> quantifies the differences of gene expression between all stages or experiments for each age or divergence category separately. Hence, the test quantifies whether or not the gene expression distribution of a single age or divergence category significantly changes throughout development or experiments. This type of analysis allows users to detect specific age or divergence categories that significantly change their expression levels throughout development or experiments.
</p>
</li></ul>

<p>Argument Specifications:
</p>
<p>Argument: type
</p>

<ul>
<li> <p><code>type = "category-centered"</code> This specification allows users to compare the differences between all age or
divergence categories during the same stage or experiment.
</p>
</li>
<li> <p><code>type = "stage-centered"</code> This specification allows users to compare the differences between all age or
divergence categories between stages or experiments.
</p>
</li></ul>

<p>Argument: distr.type
</p>

<ul>
<li> <p><code>distr.type = "boxplot"</code> This specification allows users to visualize the expression distribution of all PS or DS as boxplot.
</p>
</li>
<li> <p><code>distr.type = "violin"</code> This specification allows users to visualize the expression distribution of all PS or DS as violin plot.
</p>
</li>
<li> <p><code>distr.type = "dotplot"</code> This specification allows users to visualize the expression distribution of all PS or DS as dot plot.
</p>
</li></ul>

<p>Finally, users can specify a <code>gene.set</code> (a subset of genes included in the input <code>ExpressioSet</code>)
for which expression levels should be visualized as boxplot, dotplot, or violinplot.
</p>


<h3>Value</h3>

<p>A boxplot, violin plot, or dot plot visualizing the gene expression levels of
different PS or DS categories.
</p>
<p>Furthermore, the statistical test results returned from the <code><a href="stats.html#topic+kruskal.test">kruskal.test</a></code> are printed
to the console.
</p>
<p>(1) '*'   = P-Value &lt;= 0.05 
</p>
<p>(2) '**'  = P-Value &lt;= 0.005  
</p>
<p>(3) '***' = P-Value &lt;= 0.0005  
</p>
<p>(4) 'n.s.' = not significant = P-Value &gt; 0.05
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotMeans">PlotMeans</a></code>, <code><a href="#topic+PlotRE">PlotRE</a></code>, <code><a href="#topic+PlotBarRE">PlotBarRE</a></code>, <code><a href="#topic+age.apply">age.apply</a></code>,
<code><a href="#topic+pTAI">pTAI</a></code>, <code><a href="#topic+pTDI">pTDI</a></code>, <code><a href="#topic+pStrata">pStrata</a></code>, <code><a href="#topic+pMatrix">pMatrix</a></code>, <code><a href="#topic+TAI">TAI</a></code>, <code><a href="#topic+TDI">TDI</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)
data(DivergenceExpressionSetExample)

## Not run: 

# category-centered visualization of PS specific expression level distributions (log-scale)
PlotCategoryExpr(ExpressionSet = PhyloExpressionSetExample,
                     legendName    = "PS",
                     test.stat     = TRUE,
                     type          = "category-centered",
                     distr.type    = "boxplot",
                     log.expr      = TRUE)
                     

# stage-centered visualization of PS specific expression level distributions (log-scale)
PlotCategoryExpr(ExpressionSet = PhyloExpressionSetExample,
                     legendName    = "PS",
                     test.stat     = TRUE,
                     distr.type    = "boxplot",
                     type          = "stage-centered",
                     log.expr      = TRUE)

                     
                                                               
# category-centered visualization of PS specific expression level distributions (log-scale)
# as violoin plot
PlotCategoryExpr(ExpressionSet = PhyloExpressionSetExample,
                     legendName    = "PS",
                     test.stat     = TRUE,
                     distr.type    = "violin",
                     type          = "stage-centered",
                     log.expr      = TRUE)




# analogous for DivergenceExpressionSets
PlotCategoryExpr(ExpressionSet = DivergenceExpressionSetExample,
                     legendName    = "DS",
                     test.stat     = TRUE,
                     type          = "category-centered",
                     distr.type    = "boxplot",
                     log.expr      = TRUE)


# visualize the expression levels of 500 example genes
set.seed(234)
example.gene.set &lt;- PhyloExpressionSetExample[sample(1:25260,500) , 2]

PlotCategoryExpr(ExpressionSet = PhyloExpressionSetExample,
                 legendName    = "PS",
                 test.stat     = TRUE,
                 type          = "category-centered",
                 distr.type    = "boxplot",
                 log.expr      = TRUE,
                 gene.set      = example.gene.set)
                 

## End(Not run)
</code></pre>

<hr>
<h2 id='PlotCIRatio'>Plot Transcriptome Index using bootstrapping and confidence intervals</h2><span id='topic+PlotCIRatio'></span>

<h3>Description</h3>

<p>Function to plot and compare the confidence intervals of Transcriptome Index between transformed and non-transformed expression data 
by using bootstrapping appraoches instead of permutation tests used in <code><a href="#topic+PlotSignature">PlotSignature</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotCIRatio(ExpressionSet, measure, nbootstraps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotCIRatio_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet, DivergenceExpressionSet or PolymorphismsExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotCIRatio_+3A_measure">measure</code></td>
<td>
<p>type of transcriptome index that shall be computed. E.g. measure = &quot;TAI&quot; (Transcriptome Age Index), measure = &quot;TDI&quot; (Transcriptome Divergence Index), measure = &quot;TPI&quot; (Transcriptome Polymorphism Index).</p>
</td></tr>
<tr><td><code id="PlotCIRatio_+3A_nbootstraps">nbootstraps</code></td>
<td>
<p>number of independent bootstraps.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to check potential outliers (e.g. a few exramly highly expressed genes) in transcriptome. 
Since Transcriptome Index is weigthed mean, it could be easily affectd by outliers. So, we have to check potential outliers in the transcriptome data.
Because log or sqrt trandformation can alleviate the effect of outliers, if there are some outliers, we could see the confidence intervals (genetated by bootstrapping) 
from non-trandformed expression data are much higher and more variable than from log or sqrt trandformed expression data.
In order to compare the range of confidence intervals in the same scale, we plotted the ratio of upper to lower confidence interval boundary across development.
</p>


<h3>Author(s)</h3>

<p>Jialin Liu
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotSignature">PlotSignature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("PhyloExpressionSetExample")
PlotCIRatio(PhyloExpressionSetExample,"TAI",5)
</code></pre>

<hr>
<h2 id='PlotContribution'>Plot Cumuative Transcriptome Index</h2><span id='topic+PlotContribution'></span>

<h3>Description</h3>

<p>This function computes the cumulative contribution of each Phylostratum or Divergence Stratum to the global <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotContribution(
  ExpressionSet,
  legendName = NULL,
  xlab = "Ontogeny",
  ylab = "Transcriptome Index",
  main = "",
  y.ticks = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotContribution_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotContribution_+3A_legendname">legendName</code></td>
<td>
<p>a character string specifying whether &quot;PS&quot; or &quot;DS&quot; are used to compute relative expression profiles.</p>
</td></tr>
<tr><td><code id="PlotContribution_+3A_xlab">xlab</code></td>
<td>
<p>label of x-axis.</p>
</td></tr>
<tr><td><code id="PlotContribution_+3A_ylab">ylab</code></td>
<td>
<p>label of y-axis.</p>
</td></tr>
<tr><td><code id="PlotContribution_+3A_main">main</code></td>
<td>
<p>main title.</p>
</td></tr>
<tr><td><code id="PlotContribution_+3A_y.ticks">y.ticks</code></td>
<td>
<p>a numeric value specifying the number of ticks to be drawn on the y-axis.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Introduced by Domazet-Loso and Tautz (2010), this function allows users to visualize the cumulative contribution of each Phylostratum or Divergence Stratum to the global Transcriptome Age Index or Transcriptome Divergence Index profile to quantify how each Phylostratum or Divergence Stratum influences the profile of the global TAI or TDI pattern.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T. and Tautz D. (2010). A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns. Nature (468): 815-818.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pTAI">pTAI</a></code>, <code><a href="#topic+pTDI">pTDI</a></code>, <code><a href="#topic+TAI">TAI</a></code>, <code><a href="#topic+TDI">TDI</a></code>, <code><a href="#topic+PlotSignature">PlotSignature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 data(PhyloExpressionSetExample)
 data(DivergenceExpressionSetExample)
 
 # visualize phylostratum contribution to global TAI
 PlotContribution(PhyloExpressionSetExample, legendName = "PS")
 
 # visualize divergence stratum contribution to global TDI
 PlotContribution(DivergenceExpressionSetExample, legendName = "DS")
 
</code></pre>

<hr>
<h2 id='PlotCorrelation'>Plot the Correlation Between Phylostrata and Divergence Strata</h2><span id='topic+PlotCorrelation'></span>

<h3>Description</h3>

<p>This function plots the correlation coefficient between phylostratum values 
and divergence-stratum values of a given PhyloExpressionSet and DivergenceExpressionSet.
</p>
<p>This function can be used to test whether a given PS distribution and DS distribution are 
linear correlated so that the independence of PS and DS can be assumed for 
subsequent analyses (Quint et al., 2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotCorrelation(
  PhyloExpressionSet,
  DivergenceExpressionSet,
  method = "pearson",
  linearModel = FALSE,
  xlab = "Phylostratum",
  ylab = "Divergencestratum"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotCorrelation_+3A_phyloexpressionset">PhyloExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotCorrelation_+3A_divergenceexpressionset">DivergenceExpressionSet</code></td>
<td>
<p>a standard DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotCorrelation_+3A_method">method</code></td>
<td>
<p>a character string specifying the correlation method to cbe used, e.g. &quot;pearson&quot;, &quot;kendall&quot;, &quot;spearman&quot;.</p>
</td></tr>
<tr><td><code id="PlotCorrelation_+3A_linearmodel">linearModel</code></td>
<td>
<p>a boolean value specifying whether a linear model should be
fitted to the data and furthermore, should be visualized in the corresponding plot.</p>
</td></tr>
<tr><td><code id="PlotCorrelation_+3A_xlab">xlab</code></td>
<td>
<p>label of x-axis.</p>
</td></tr>
<tr><td><code id="PlotCorrelation_+3A_ylab">ylab</code></td>
<td>
<p>label of y-axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a jitter-correlation-plot of PS and DS correlation.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.
Drost HG et al. (2015) <em>Evidence for Active Maintenance of Phylotranscriptomic Hourglass Patterns in Animal and Plant Embryogenesis</em>. Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+cor">cor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># read standard phylotranscriptomics data
data(PhyloExpressionSetExample)
data(DivergenceExpressionSetExample)
 
# plot the PS and DS correlation
PlotCorrelation(PhyloExpressionSetExample, 
                DivergenceExpressionSetExample, 
                method      = "pearson", 
                linearModel = TRUE)



</code></pre>

<hr>
<h2 id='PlotDistribution'>Plot the Frequency Distribution of Phylostrata or Divergence Strata</h2><span id='topic+PlotDistribution'></span>

<h3>Description</h3>

<p>This function plots the frequency distribution of genes within the 
corresponding <em>phylostratigraphic map</em> or <em>divergence map</em> and can be used to fastly visualize the PS or DS distribution of a given phylostratum vector or divergence-stratum vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotDistribution(
  PhyloExpressionSet,
  legendName = "PS",
  as.ratio = FALSE,
  use.only.map = FALSE,
  xlab = NULL,
  ylab = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotDistribution_+3A_phyloexpressionset">PhyloExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotDistribution_+3A_legendname">legendName</code></td>
<td>
<p>a character string specifying whether &quot;PS&quot; or &quot;DS&quot; are visualized.</p>
</td></tr>
<tr><td><code id="PlotDistribution_+3A_as.ratio">as.ratio</code></td>
<td>
<p>a boolean value specifying whether the relative frequencies
instead of absolute frequencies shall be plotted.</p>
</td></tr>
<tr><td><code id="PlotDistribution_+3A_use.only.map">use.only.map</code></td>
<td>
<p>logical value indicating whether or not a Phylostratigraphic Map or Divergence Map should be passed to the <code>ExpressionSet</code> argument instead of a standard <code>ExpressionSet</code> object.</p>
</td></tr>
<tr><td><code id="PlotDistribution_+3A_xlab">xlab</code></td>
<td>
<p>label of the x-axis.</p>
</td></tr>
<tr><td><code id="PlotDistribution_+3A_ylab">ylab</code></td>
<td>
<p>label of the y-axis.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The frequency distribution of all genes or a subset of genes might be of interest for subsequent analyses.
</p>
<p>For Example:
</p>
<p>Filtering genes using gene cluster algorithms can result in different groups (classes) of genes.
For each gene group the phylostratum or divergence-stratum distribution can be visualized using this function
and can be compared between different groups.
</p>
<p>This analysis allows to compare different gene expression profiles (or gene groups in general) based
on their evolutionary origins or evolutionary relationships.
</p>


<h3>Value</h3>

<p>a barplot showing the phylostratum distribution or 
divergence-stratum distribution of a given numeric vector containing PS or DS values.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotSelectedAgeDistr">PlotSelectedAgeDistr</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load PhyloExpressionSet
data(PhyloExpressionSetExample)

# plot the phylostratum distribution of a PhyloExpressionSet
PlotDistribution(PhyloExpressionSetExample)

# plot the relative frequency distribution of a PhyloExpressionSet
PlotDistribution(PhyloExpressionSetExample, as.ratio = TRUE)


# a example for visualizing the PS distribution for a subset of genes
PlotDistribution(PhyloExpressionSetExample[sample(20000,5000) , ], as.ratio = TRUE)

</code></pre>

<hr>
<h2 id='PlotEnrichment'>Plot the Phylostratum or Divergence Stratum Enrichment of a given Gene Set</h2><span id='topic+PlotEnrichment'></span>

<h3>Description</h3>

<p>This function computes and visualizes the significance of enriched (over or underrepresented) Phylostrata or Divergence Strata within an input <code>test.set</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotEnrichment(
  ExpressionSet,
  test.set,
  use.only.map = FALSE,
  measure = "log-foldchange",
  complete.bg = TRUE,
  legendName = "",
  over.col = "steelblue",
  under.col = "midnightblue",
  epsilon = 1e-05,
  cex.legend = 1,
  cex.asterisk = 1,
  plot.bars = TRUE,
  p.adjust.method = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotEnrichment_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object (in case <code>only.map = FALSE</code>).</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_test.set">test.set</code></td>
<td>
<p>a character vector storing the gene ids for which PS/DS enrichment analyses should be performed.</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_use.only.map">use.only.map</code></td>
<td>
<p>a logical value indicating whether instead of a standard <code>ExpressionSet</code> only a <code>Phylostratigraphic Map</code> or <code>Divergene Map</code> is passed to this function.</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_measure">measure</code></td>
<td>
<p>a character string specifying the measure that should be used to quantify over and under representation of PS/DS. Measures can either be <code>measure = "foldchange"</code> (odds) or <code>measure = "log-foldchange"</code> (log-odds).</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_complete.bg">complete.bg</code></td>
<td>
<p>a logical value indicating whether the entire background set
of the input <code>ExpressionSet</code> should be considered when performing Fisher's exact
test (<code>complete.bg = TRUE</code>) or whether genes that are stored in <code>test.set</code>
should be excluded from the background set before performing Fisher's exact test
(<code>complete.bg = FALSE</code>).</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_legendname">legendName</code></td>
<td>
<p>a character string specifying whether &quot;PS&quot; or &quot;DS&quot; are used to compute relative expression profiles.</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_over.col">over.col</code></td>
<td>
<p>color of the overrepresentation bars.</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_under.col">under.col</code></td>
<td>
<p>color of the underrepresentation bars.</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_epsilon">epsilon</code></td>
<td>
<p>a small value to shift values by epsilon to avoid log(0) computations.</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_cex.legend">cex.legend</code></td>
<td>
<p>the <code>cex</code> value for the legend.</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_cex.asterisk">cex.asterisk</code></td>
<td>
<p>the <code>cex</code> value for the asterisk.</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_plot.bars">plot.bars</code></td>
<td>
<p>a logical value specifying whether or not bars should be visualized or whether only <code>p.values</code> and <code>enrichment.matrix</code> should be returned.</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>correction method to adjust p-values for multiple comparisons (see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code> for possible methods). E.g., <code>p.adjust.method = "BH"</code> (Benjamini &amp; Hochberg (1995)) or <code>p.adjust.method = "bonferroni"</code> (Bonferroni correction).</p>
</td></tr>
<tr><td><code id="PlotEnrichment_+3A_...">...</code></td>
<td>
<p>default graphics parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This <em>Phylostratum</em> or <em>Divergence Stratum</em> enrichment analysis is motivated 
by Sestak and Domazet-Loso (2015) who perform <em>Phylostratum</em> or <em>Divergence Stratum</em> enrichment analyses to correlate organ evolution with the origin of organ specific genes. 
</p>
<p>In detail this function takes the <em>Phylostratum</em> or <em>Divergence Stratum</em> distribution of all genes stored in the input <code>ExpressionSet</code> as background set and
the <em>Phylostratum</em> or <em>Divergence Stratum</em> distribution of the <code>test.set</code> and performes a <code><a href="stats.html#topic+fisher.test">fisher.test</a></code> for each <em>Phylostratum</em> or <em>Divergence Stratum</em> to quantify the statistical significance of over- or underrepresentated <em>Phylostrata</em> or <em>Divergence Strata</em> within the set of selected <code>test.set</code> genes. 
</p>
<p>To visualize the odds or log-odds of over or underrepresented genes within the <code>test.set</code> the following procedure is performed:
</p>

<ul>
<li><p> N_ij denotes the number of genes in group j and deriving from PS i, with <em>i = 1, .. , n</em> and where <em>j = 1</em> denotes the background set and <em>j = 2</em> denotes the <code>test.set</code>
</p>
</li>
<li><p> N_i. denotes the total number of genes within PS i
</p>
</li>
<li><p> N_.j denotes the total number of genes within group j
</p>
</li>
<li><p> N_.. is the total number of genes within all groups j and all PS i
</p>
</li>
<li><p> f_ij = N_ij / N_.. and g_ij = f_ij / f_.j denote relative frequencies between groups
</p>
</li>
<li><p> f_i. denotes the between group sum of f_ij
</p>
</li></ul>

<p>The result is the fold-change value (odds) denoted as C = g_i2 / f_i. which is visualized above and below zero. 
</p>
<p>In case a large number of Phylostrata or Divergence Strata is included in the input 
<code>ExpressionSet</code>, p-values returned by <code>PlotEnrichment</code> should be adjusted for
multiple comparisons which can be done by specifying the <code>p.adjust.method</code> argument.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Sestak and Domazet-Loso (2015). Phylostratigraphic Profiles in Zebrafish Uncover Chordate Origins of the Vertebrate Brain. Mol. Biol. Evol. 32(2): 299-312.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+EnrichmentTest">EnrichmentTest</a></code>, <code><a href="stats.html#topic+fisher.test">fisher.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)

set.seed(123)
test_set &lt;- sample(PhyloExpressionSetExample[ , 2],10000)

## Examples with complete.bg = TRUE
## Hence: the entire background set of the input ExpressionSet is considered 
## when performing Fisher's exact test 

# measure: log-foldchange
PlotEnrichment(ExpressionSet = PhyloExpressionSetExample,
               test.set      = test_set , 
               legendName    = "PS", 
               measure       = "log-foldchange")
               
               
# measure: foldchange
PlotEnrichment(ExpressionSet = PhyloExpressionSetExample,
               test.set      = test_set , 
               legendName    = "PS", 
               measure       = "foldchange")
   
               
## Examples with complete.bg = FALSE
## Hence: the test.set genes are excluded from the background set before
## Fisher's exact test is performed
     
                                       
# measure: log-foldchange
PlotEnrichment(ExpressionSet = PhyloExpressionSetExample,
               test.set      = test_set ,
                complete.bg  = FALSE,
               legendName    = "PS", 
               measure       = "log-foldchange")
               
               
# measure: foldchange
PlotEnrichment(ExpressionSet = PhyloExpressionSetExample,
               test.set      = test_set , 
               complete.bg   = FALSE,
               legendName    = "PS", 
               measure       = "foldchange")     
               
</code></pre>

<hr>
<h2 id='PlotGeneSet'>Plot the Expression Profiles of a Gene Set</h2><span id='topic+PlotGeneSet'></span>

<h3>Description</h3>

<p>This function simply visualizes the gene expression profiles of
a defined subset of genes stored in the input <code>ExpressionSet</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotGeneSet(
  ExpressionSet,
  gene.set,
  get.subset = FALSE,
  use.only.map = FALSE,
  colors = NULL,
  plot.legend = TRUE,
  y.ticks = 6,
  digits.ylab = 4,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotGeneSet_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotGeneSet_+3A_gene.set">gene.set</code></td>
<td>
<p>a character vector storing the gene ids for which gene expression profiles shall be visualized.</p>
</td></tr>
<tr><td><code id="PlotGeneSet_+3A_get.subset">get.subset</code></td>
<td>
<p>a logical value indicating whether or not an <code>ExpressionSet</code> subset of the selected <code>gene.set</code> should be retuned.</p>
</td></tr>
<tr><td><code id="PlotGeneSet_+3A_use.only.map">use.only.map</code></td>
<td>
<p>a logical value indicating whether instead of a standard <code>ExpressionSet</code> only a <code>Phylostratigraphic Map</code> or <code>Divergene Map</code> is passed to the function.</p>
</td></tr>
<tr><td><code id="PlotGeneSet_+3A_colors">colors</code></td>
<td>
<p>colors for gene expression profiles. Default: <code>colors = NULL</code>, hence default colours are used.</p>
</td></tr>
<tr><td><code id="PlotGeneSet_+3A_plot.legend">plot.legend</code></td>
<td>
<p>a logical value indicating whether gene ids should be printed as legend next to the plot.</p>
</td></tr>
<tr><td><code id="PlotGeneSet_+3A_y.ticks">y.ticks</code></td>
<td>
<p>a numeric value specifying the number of ticks to be drawn on the y-axis.</p>
</td></tr>
<tr><td><code id="PlotGeneSet_+3A_digits.ylab">digits.ylab</code></td>
<td>
<p>a numeric value specifying the number of digits shown for the expression levels on the y-axis.</p>
</td></tr>
<tr><td><code id="PlotGeneSet_+3A_...">...</code></td>
<td>
<p>additional parameters passed to <code><a href="graphics.html#topic+matplot">matplot</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function simply visualizes or subsets the gene expression levels of a set of genes
that are stored in the input <code>ExpressionSet</code>.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SelectGeneSet">SelectGeneSet</a></code>, <code><a href="#topic+PlotEnrichment">PlotEnrichment</a></code>, <code><a href="#topic+DiffGenes">DiffGenes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PhyloExpressionSetExample)

# the best parameter setting to visualize this plot:
# png("test_png.png",700,400)
PlotGeneSet(ExpressionSet = PhyloExpressionSetExample, 
            gene.set      = PhyloExpressionSetExample[1:5, 2], 
            type          = "l", 
            lty           = 1, 
            lwd           = 4,
            xlab          = "Ontogeny",
            ylab          = "Expression Level")

# dev.off()

# In case you would like to work with the expression levels
# of selected genes you can specify the 'get.subset' argument:

PlotGeneSet(ExpressionSet = PhyloExpressionSetExample, 
            gene.set      = PhyloExpressionSetExample[1:5, 2], 
            get.subset    = TRUE)


# get a gene subset using only a phylostratihraphic map
ExamplePSMap &lt;- PhyloExpressionSetExample[ , 1:2]

PlotGeneSet(ExpressionSet = ExamplePSMap, 
            gene.set      = PhyloExpressionSetExample[1:5, 2], 
            get.subset    = TRUE,
            use.only.map  = TRUE)
            
</code></pre>

<hr>
<h2 id='PlotGroupDiffs'>Plot the significant differences between gene expression distributions of PS or DS groups</h2><span id='topic+PlotGroupDiffs'></span>

<h3>Description</h3>

<p>This function performs a test to quantify the statistical significance between
the global expression level distributions of groups of PS or DS. It therefore, allows users to investigate
significant groups of PS or DS that significantly differ in their gene expression level distibution
within specific developmental stages or experiments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotGroupDiffs(
  ExpressionSet,
  Groups = NULL,
  legendName = NULL,
  stat.test = "wilcox.test",
  col = c("turquoise3", "magenta3"),
  plot.type = NULL,
  gene.set = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotGroupDiffs_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotGroupDiffs_+3A_groups">Groups</code></td>
<td>
<p>a list containing the phylostrata or divergence strata that correspond 
to the same phylostratum class or divergence class.
For ex. evolutionary old phylostrata: PS1-3 (Class 1) 
and evolutionary young phylostrata: PS4-12 (Class 2). In this case, 
the list could be assigned as, <code>Groups</code> = list(c(1:3), c(4:12)).</p>
</td></tr>
<tr><td><code id="PlotGroupDiffs_+3A_legendname">legendName</code></td>
<td>
<p>a character string specifying whether &quot;PS&quot; or &quot;DS&quot; are used to compute relative expression profiles.</p>
</td></tr>
<tr><td><code id="PlotGroupDiffs_+3A_stat.test">stat.test</code></td>
<td>
<p>the statistical test to quantify PS or DS group differences.</p>
</td></tr>
<tr><td><code id="PlotGroupDiffs_+3A_col">col</code></td>
<td>
<p>colors for the two box plots representing the expression level distributions of selected PS/DS groups.</p>
</td></tr>
<tr><td><code id="PlotGroupDiffs_+3A_plot.type">plot.type</code></td>
<td>
<p>the type of plot that shall be drawn to visualized the difference in PS/DS group specific expression .</p>
</td></tr>
<tr><td><code id="PlotGroupDiffs_+3A_gene.set">gene.set</code></td>
<td>
<p>a character vector storing the gene ids for which group specific differences shall be statistically quantified.</p>
</td></tr>
<tr><td><code id="PlotGroupDiffs_+3A_...">...</code></td>
<td>
<p>additional plot parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The purpose of this function is to detect groups of PS or DS that significantly differ in their gene expression
level distributions on a global (transcriptome) level. Since relative expression levels (<code><a href="#topic+PlotRE">PlotRE</a></code>) or
PS or DS specific mean expression levels (<code><a href="#topic+PlotMeans">PlotMeans</a></code>) are biased by highly expressed genes,
this function allows users to objectively test the significant difference of transcriptome expression between
groups of PS or DS in a specific developmental stage or experiment.
</p>
<p>In particular, this function divides (for each developmental stage separately) the gene expression levels into two groups: Group1 = genes deriving from selected PS/DS in group 1 and 
Group2 = genes deriving from selected PS/DS in group 2. Within each stage the expression level distributions between group 1 and group 2 are statistically quantified using a <code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code>.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotMeans">PlotMeans</a></code>, <code><a href="#topic+PlotRE">PlotRE</a></code>, <code><a href="#topic+PlotBarRE">PlotBarRE</a></code>, <code><a href="#topic+PlotCategoryExpr">PlotCategoryExpr</a></code>, <code><a href="#topic+GroupDiffs">GroupDiffs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)

PlotGroupDiffs(ExpressionSet = PhyloExpressionSetExample,
               Groups        = list(group_1 = 1:3,group_2 = 4:12),
               legendName    = "PS",
               type          = "b",
               lwd           = 6,
               xlab          = "Ontogeny")
               
               
# only receive the p-values without the corresponding plot               
PlotGroupDiffs(ExpressionSet = PhyloExpressionSetExample,
               Groups        = list(group_1 = 1:3,group_2 = 4:12),
               legendName    = "PS",
               plot.p.vals   = FALSE,
               type          = "b",
               lwd           = 6,
               xlab          = "Ontogeny")
               
               
# quantify the significant difference of a selected set of genes
# only receive the p-values without the corresponding plot
set.seed(123)
ExampleGeneSet &lt;- sample(PhyloExpressionSetExample[ , 2],5000)
                              
PlotGroupDiffs(ExpressionSet = PhyloExpressionSetExample,
               Groups        = list(group_1 = 1:3,group_2 = 4:12),
               legendName    = "PS",
               plot.p.vals   = FALSE,
               gene.set      = ExampleGeneSet)                 


# plot differences as boxplot for each developmental stage
PlotGroupDiffs(ExpressionSet = tf(PhyloExpressionSetExample,log2),
               Groups        = list(group_1 = 1:3,group_2 = 4:12),
               legendName    = "PS",
               plot.type     = "boxplot")


</code></pre>

<hr>
<h2 id='PlotMeans'>Plot Mean Expression Profiles</h2><span id='topic+PlotMeans'></span>

<h3>Description</h3>

<p>This function computes for each age category the corresponding mean expression profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotMeans(
  ExpressionSet,
  Groups = NULL,
  modules = NULL,
  legendName = "age",
  xlab = "Ontogeny",
  ylab = "Mean Expression Level",
  main = "",
  y.ticks = 10,
  adjust.range = TRUE,
  alpha = 0.008,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotMeans_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotMeans_+3A_groups">Groups</code></td>
<td>
<p>a list containing the age categories for which mean expression levels shall be drawn.
For ex. evolutionary users can compare old phylostrata: PS1-3 (Class 1) and evolutionary young phylostrata: PS4-12 (Class 2). 
In this example, the list could be assigned as, <code>Groups = list(c(1:3), c(4:12))</code>. 
The group options is limited to 2 Groups.</p>
</td></tr>
<tr><td><code id="PlotMeans_+3A_modules">modules</code></td>
<td>
<p>a list storing three elements for specifying the modules: early, mid, and late. 
Each element expects a numeric vector specifying the developmental stages 
or experiments that correspond to each module. For example, 
<code>module</code> = <code>list(early = 1:2, mid = 3:5, late = 6:7)</code> devides a dataset storing seven developmental stages into 3 modules. Default is <code>modules = NULL</code>. 
But if specified, a shaded are will be drawn to illustrate stages corresponding to the mid module.</p>
</td></tr>
<tr><td><code id="PlotMeans_+3A_legendname">legendName</code></td>
<td>
<p>a character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="PlotMeans_+3A_xlab">xlab</code></td>
<td>
<p>label of x-axis.</p>
</td></tr>
<tr><td><code id="PlotMeans_+3A_ylab">ylab</code></td>
<td>
<p>label of y-axis.</p>
</td></tr>
<tr><td><code id="PlotMeans_+3A_main">main</code></td>
<td>
<p>main text.</p>
</td></tr>
<tr><td><code id="PlotMeans_+3A_y.ticks">y.ticks</code></td>
<td>
<p>number of ticks that shall be drawn on the y-axis.</p>
</td></tr>
<tr><td><code id="PlotMeans_+3A_adjust.range">adjust.range</code></td>
<td>
<p>logical indicating whether or not the y-axis scale shall be adjusted to the same range in case two groups are specified. Default is <code>adjust.range = TRUE</code>.</p>
</td></tr>
<tr><td><code id="PlotMeans_+3A_alpha">alpha</code></td>
<td>
<p>transparency of the shaded area (between [0,1]). Default is <code>alpha = 0.1</code>.</p>
</td></tr>
<tr><td><code id="PlotMeans_+3A_...">...</code></td>
<td>
<p>place holder for old version of PlotMeans that was based on base graphics instead of ggplot2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This plot may be useful to compare the absolute mean expression        
levels of each age category across stages.
</p>
<p>In different developmental processes different phylostratum or divergence-stratum
classes might be more expressed than others, hence contributing more to the overall
phylotranscriptomics pattern (<code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code>).
This plot can help to identify the phylostratum or divergence-stratum classes 
that contributes most to the overall transcriptome of the given developmental process.
</p>


<h3>Value</h3>

<p>a plot showing mean expression profiles of each age category.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotBarRE">PlotBarRE</a></code>, <code><a href="#topic+RE">RE</a></code>, <code><a href="#topic+REMatrix">REMatrix</a></code>, <code><a href="#topic+PlotRE">PlotRE</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example using a PhyloExpressionSet
### and DivergenceExpressionSet
# load PhyloExpressionSet
data(PhyloExpressionSetExample)

# load PhyloExpressionSet
data(DivergenceExpressionSetExample)

# plot evolutionary old PS (PS1-3) vs evolutionary young PS (PS4-12)
PlotMeans(PhyloExpressionSetExample,
          Groups = list(c(1:3), c(4:12)), 
          legendName = "PS",
          adjust.range = TRUE)

# if users wish to not adjust the y-axis scale when 
# 2 groups are selected they can specify: adjust.range = FALSE
PlotMeans(PhyloExpressionSetExample,
          Groups = list(c(1:3), c(4:12)), 
          legendName = "PS",
          adjust.range = FALSE)
          
          
# plot conserved DS (DS1-5) vs divergent DS (PS6-10)
# NOTE: DS are always defined in the range 1, 2, ... , 10.
# Hence, make sure that your groups are within this range!
PlotMeans(DivergenceExpressionSetExample,
          Groups = list(c(1:5), c(6:10)), 
          legendName = "DS",
          adjust.range = TRUE)

</code></pre>

<hr>
<h2 id='PlotMedians'>Plot Median Expression Profiles</h2><span id='topic+PlotMedians'></span>

<h3>Description</h3>

<p>This function computes for each age category the corresponding median expression profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotMedians(
  ExpressionSet,
  Groups = NULL,
  legendName = "age",
  xlab = "Ontogeny",
  ylab = "Median Expression Level",
  main = "",
  y.ticks = 10,
  adjust.range = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotMedians_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotMedians_+3A_groups">Groups</code></td>
<td>
<p>a list containing the age categories for which median expression levels shall be drawn.
For ex. evolutionary users can compare old phylostrata: PS1-3 (Class 1) and evolutionary young phylostrata: PS4-12 (Class 2). 
In this example, the list could be assigned as, <code>Groups = list(c(1:3), c(4:12))</code>. 
The group options is limited to 2 Groups.</p>
</td></tr>
<tr><td><code id="PlotMedians_+3A_legendname">legendName</code></td>
<td>
<p>a character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="PlotMedians_+3A_xlab">xlab</code></td>
<td>
<p>label of x-axis.</p>
</td></tr>
<tr><td><code id="PlotMedians_+3A_ylab">ylab</code></td>
<td>
<p>label of y-axis.</p>
</td></tr>
<tr><td><code id="PlotMedians_+3A_main">main</code></td>
<td>
<p>main text.</p>
</td></tr>
<tr><td><code id="PlotMedians_+3A_y.ticks">y.ticks</code></td>
<td>
<p>number of ticks that shall be drawn on the y-axis.</p>
</td></tr>
<tr><td><code id="PlotMedians_+3A_adjust.range">adjust.range</code></td>
<td>
<p>logical indicating whether or not the y-axis scale shall be adjusted to the same range in case two groups are specified. Default is <code>adjust.range = TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This plot may be useful to compare the absolute median expression        
levels of each age category across stages.
</p>
<p>In different developmental processes different phylostratum or divergence-stratum
classes might be more expressed than others, hence contributing more to the overall
phylotranscriptomics pattern (<code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code>).
This plot can help to identify the phylostratum or divergence-stratum classes 
that contributes most to the overall transcriptome of the given developmental process.
</p>


<h3>Value</h3>

<p>a plot showing median expression profiles of each age category.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotBarRE">PlotBarRE</a></code>, <code><a href="#topic+RE">RE</a></code>, <code><a href="#topic+REMatrix">REMatrix</a></code>, <code><a href="#topic+PlotRE">PlotRE</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example using a PhyloExpressionSet
### and DivergenceExpressionSet
# load PhyloExpressionSet
data(PhyloExpressionSetExample)

# load PhyloExpressionSet
data(DivergenceExpressionSetExample)

# plot evolutionary old PS (PS1-3) vs evolutionary young PS (PS4-12)
PlotMedians(PhyloExpressionSetExample,
          Groups = list(c(1:3), c(4:12)), 
          legendName = "PS",
          adjust.range = TRUE)

# if users wish to not adjust the y-axis scale when 
# 2 groups are selected they can specify: adjust.range = FALSE
PlotMedians(PhyloExpressionSetExample,
          Groups = list(c(1:3), c(4:12)), 
          legendName = "PS",
          adjust.range = FALSE)
          
          
# plot conserved DS (DS1-5) vs divergent DS (PS6-10)
# NOTE: DS are always defined in the range 1, 2, ... , 10.
# Hence, make sure that your groups are within this range!
PlotMedians(DivergenceExpressionSetExample,
          Groups = list(c(1:5), c(6:10)), 
          legendName = "DS",
          adjust.range = TRUE)

</code></pre>

<hr>
<h2 id='PlotPattern'>Plot the Transcriptome Age Index or Transcriptome Divergence Index</h2><span id='topic+PlotPattern'></span>

<h3>Description</h3>

<p>Function to plot the TAI or TDI of a given PhyloExpressionSet or DivergenceExpressionSet object.
This function plot the <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> of a given PhyloExpressionSet or DivergenceExpressionSet object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPattern(
  ExpressionSet,
  TestStatistic = "FlatLineTest",
  modules = NULL,
  permutations = 1000,
  lillie.test = FALSE,
  digits.ylab = 4,
  p.value = TRUE,
  shaded.area = FALSE,
  y.ticks = 4,
  custom.perm.matrix = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPattern_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotPattern_+3A_teststatistic">TestStatistic</code></td>
<td>
<p>a string defining the type of test statistics to be used to quantify the statistical significance the present phylotranscriptomics pattern.
Possible values can be: <code>TestStatistic</code> = <code>"FlatLineTest"</code> : Statistical test for the deviation from a flat line.
<code>TestStatistic</code> = <code>"ReductiveHourglassTest"</code> : Statistical test for the existence of a hourglass shape (high-low-high pattern).
<code>TestStatistic</code> = <code>"EarlyConservationTest"</code> : Statistical test for the existence of a earlyconservation pattern (low-high-high pattern).</p>
</td></tr>
<tr><td><code id="PlotPattern_+3A_modules">modules</code></td>
<td>
<p>a list storing three elements for the <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code> or <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code>: early, mid, and late. 
Each element expects a numeric vector specifying the developmental stages 
or experiments that correspond to each module. For example, 
<code>module</code> = <code>list(early = 1:2, mid = 3:5, late = 6:7)</code> devides a dataset storing seven developmental stages into 3 modules.</p>
</td></tr>
<tr><td><code id="PlotPattern_+3A_permutations">permutations</code></td>
<td>
<p>a numeric value specifying the number of permutations to be performed for the <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>, <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code> or <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>.</p>
</td></tr>
<tr><td><code id="PlotPattern_+3A_lillie.test">lillie.test</code></td>
<td>
<p>a boolean value specifying whether the Lilliefors Kolmogorov-Smirnov Test shall be performed.</p>
</td></tr>
<tr><td><code id="PlotPattern_+3A_digits.ylab">digits.ylab</code></td>
<td>
<p>a numeric value specifying the number of digits shown for the TAI or TDI values on the y-axis.</p>
</td></tr>
<tr><td><code id="PlotPattern_+3A_p.value">p.value</code></td>
<td>
<p>a boolean value specifying whether the p-value of the test statistic shall be printed within the plot area.</p>
</td></tr>
<tr><td><code id="PlotPattern_+3A_shaded.area">shaded.area</code></td>
<td>
<p>a boolean value specifying whether a shaded area shall 
be drawn for the developmental stages defined to be the presumptive phylotypic period.</p>
</td></tr>
<tr><td><code id="PlotPattern_+3A_y.ticks">y.ticks</code></td>
<td>
<p>a numeric value specifying the number of ticks to be drawn on the y-axis.</p>
</td></tr>
<tr><td><code id="PlotPattern_+3A_custom.perm.matrix">custom.perm.matrix</code></td>
<td>
<p>a custom <code><a href="#topic+bootMatrix">bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic visualized by <code>PlotPattern</code>. Default is <code>custom.perm.matrix = NULL</code>.</p>
</td></tr>
<tr><td><code id="PlotPattern_+3A_...">...</code></td>
<td>
<p>default plot parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes a permutation test quantifying the statistical significance of the prensent phylotranscriptomics pattern. 
The user can choose between the <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>, <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>, or <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code>. 
The <code><a href="#topic+FlatLineTest">FlatLineTest</a></code> tests for any significant deviation from a flat line. 
Each period or stage that significantly deviates from a flat line, might be governed by stronger selective pressure (in terms of natural selection) compared to other stages or periods of development.
The <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code> specificly tests for the statistical significance of a molecular hourglass pattern (high-low-high pattern) with prior biological knowlegde.
The corresponding p-value that is printed within the plot (by default) specifies the statistical significance of the chosen test statistic.
</p>
<p>The <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code> specificly tests for the statistical significance of a low-high-high pattern (monotonically increasing function)
with prior biological knowlegde.
The corresponding p-value that is printed within the plot (by default) specifies the statistical significance of the chosen test statistic.
</p>
<p>The x-axis denotes the developmental series (time course / experiments / ontogeny) of the input ExpressionSet and the y-axis 
denotes the corresponding mean transcriptome age value (<code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code>) of the given ExpressionSet. 
</p>
<p>Furthermore, the grey lines above and below the actual phylotranscriptomics pattern denotes the standard deviations of <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> 
values that have been estimated from the <code><a href="#topic+bootMatrix">bootMatrix</a></code>.
A low mean transcriptome age value denotes an evolutionary older transcriptome being active during the corresponding periods, 
whereas a high mean transcriptome age value denotes an evolutionary younger transcriptome being active during the corresponding periods.
For mean transcriptome divergence, a low mean transcriptome divergence value denotes a more conserved transcriptome 
being active (between two species), whereas a high mean transcriptome divergence value denotes a more divergent transcriptome 
being active (between two species) - in terms of protein-sequence substitution rates.
</p>
<p>This function is useful to fastly plot the <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> 
profile of a given PhyloExpressionSet or DivergenceExpressionSet object and 
the statistical significance of the corresponding pattern.
Internally the function calls several graphics functions, 
such as <code><a href="base.html#topic+plot">plot</a></code>, <code><a href="graphics.html#topic+axis">axis</a></code>, and <code><a href="graphics.html#topic+legend">legend</a></code>. 
For the ellipsis argument <code><a href="base.html#topic+...">...</a></code> all graphics specific arguments can be defined. 
Internally the function specific arguments for e.g. <code><a href="base.html#topic+plot">plot</a></code>, <code><a href="graphics.html#topic+axis">axis</a></code>, 
and <code><a href="graphics.html#topic+legend">legend</a></code> will be detected and are passed to the corresponding function.
</p>
<p>Hence, when calling the function <code>PlotPattern</code>, one can specify arguments 
for <code><a href="base.html#topic+plot">plot</a></code> and <code><a href="graphics.html#topic+axis">axis</a></code> and <code><a href="graphics.html#topic+legend">legend</a></code> 
as <code><a href="base.html#topic+...">...</a></code>. 
</p>
<p>In case prior biological knowledge is present for a specific period of development,
the <code>shaded.area</code> argument can be set to <code>TRUE</code> and the function will use
the values stored in the <code>mid</code> argument to draw a shaded area within the corresponding period of development.
</p>


<h3>Value</h3>

<p>a plot visualizing the phylotranscriptomic pattern of a given PhyloExpressionSet or DivergenceExpressionSet object.
</p>
<p>The corresponding <em>p-value</em> of the test statistic is named as follows:
</p>
<p><code>p_flt</code> = p-value of the corresponding <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>
</p>
<p><code>p_rht</code> = p-value of the corresponding <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>
</p>
<p><code>p_ect</code> = p-value of the corresponding <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code>
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T and Tautz D. (2010). <em>A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns</em>. Nature (468): 815-818.
</p>
<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.
</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+TAI">TAI</a></code>, <code><a href="#topic+TDI">TDI</a></code>, <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>,
<code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>, <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code>,
<code><a href="#topic+PlotSignature">PlotSignature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load PhyloExpressionSet
data(PhyloExpressionSetExample)

# only visualize the TAI profile without any test statistics...
# this is equavalent to performing: plot(TAI(PhyloExpressionSetExample), type = "l", lwd = 6)
PlotPattern(ExpressionSet = PhyloExpressionSetExample,
            TestStatistic = NULL,
            type          = "l",
            xlab          = "Ontogeny",
            ylab          = "TAI",
            lwd           = 9)

# the simplest example of plotting the TAI profile of a given PhyloExpressionSet:
# In this case (default) the FlatLineTest will be performed to quantify
# the statistical significance of the present TAI pattern and will be drawn as 'p = ... '
# in the plot

PlotPattern(ExpressionSet = PhyloExpressionSetExample, 
            TestStatistic = "FlatLineTest",
            permutations  = 100, 
            type          = "l", 
            xlab          = "Ontogeny", 
            ylab          = "TAI", 
            lwd           = 9)

# an example performing the ReductiveHourglassTest and printing the p-value
# and shaded area of the presumptive phylotypic period into the plot
# Here the 'p = ...' denotes the p-value that is returned by the ReductiveHourglassTest

PlotPattern(
            ExpressionSet = PhyloExpressionSetExample,
            TestStatistic = "ReductiveHourglassTest",
            modules       = list(early = 1:2,mid = 3:5,late = 6:7), 
            permutations  = 100, 
            p.value       = TRUE, 
            shaded.area   = TRUE, 
            xlab          = "Ontogeny", 
            ylab          = "TAI", 
            type          = "l", 
            lwd           = 9)

# testing for early conservation model 
PlotPattern( ExpressionSet = PhyloExpressionSetExample,
             TestStatistic = "EarlyConservationTest",
            modules        = list(early = 1:2,mid = 3:5,late = 6:7), 
            permutations   = 100,
            p.value        = TRUE, 
            shaded.area    = TRUE, 
            xlab           = "Ontogeny", 
            ylab           = "TAI", 
            type           = "l", 
            lwd            = 9)
            

# use your own permutation matrix
custom_perm_matrix &lt;- bootMatrix(PhyloExpressionSetExample,100)

PlotPattern(ExpressionSet      = PhyloExpressionSetExample, 
            TestStatistic      = "FlatLineTest",
            custom.perm.matrix = custom_perm_matrix, 
            type               = "l", 
            xlab               = "Ontogeny", 
            ylab               = "TAI", 
            lwd                = 9)


</code></pre>

<hr>
<h2 id='PlotRE'>Plot Relative Expression Levels</h2><span id='topic+PlotRE'></span>

<h3>Description</h3>

<p>This function computes for each age category the corresponding relative expression profile.
</p>
<p>For each age category the corresponding relative expression profile is being computed as follows:
</p>
<p style="text-align: center;"><code class="reqn">f_js = ( e_js - e_j min ) / ( e_j max - e_j min )</code>
</p>

<p>where <code class="reqn">e_j min</code> and <code class="reqn">e_j max</code> denote the minimum/maximum <code><a href="base.html#topic+mean">mean</a></code> expression level 
of phylostratum j over  developmental stages s. This linear transformation corresponds to 
a shift by <code class="reqn">e_j min</code> and a subsequent shrinkage by <code class="reqn">e_j max - e_j min</code>. 
As a result, the relative expression level <code class="reqn">f_js</code> of developmental stage s 
with minimum <code class="reqn">e_js</code> is 0, the relative expression level <code class="reqn">f_js</code> of the developmental 
stage s with maximum <code class="reqn">e_js</code> is 1, and the relative expression levels <code class="reqn">f_js</code> of 
all other stages s range between 0 and 1, accordingly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotRE(
  ExpressionSet,
  Groups = NULL,
  modules = NULL,
  legendName = "age",
  xlab = "Ontogeny",
  ylab = "Relative Expression Level",
  main = "",
  y.ticks = 10,
  adjust.range = TRUE,
  alpha = 0.008,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotRE_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotRE_+3A_groups">Groups</code></td>
<td>
<p>a list containing the age categories for which mean expression levels shall be drawn.
For ex. evolutionary users can compare old phylostrata: PS1-3 (Class 1) and evolutionary young phylostrata: PS4-12 (Class 2). 
In this example, the list could be assigned as, <code>Groups = list(c(1:3), c(4:12))</code>. 
The group options is limited to 2 Groups.</p>
</td></tr>
<tr><td><code id="PlotRE_+3A_modules">modules</code></td>
<td>
<p>a list storing three elements for specifying the modules: early, mid, and late. 
Each element expects a numeric vector specifying the developmental stages 
or experiments that correspond to each module. For example, 
<code>module</code> = <code>list(early = 1:2, mid = 3:5, late = 6:7)</code> devides a dataset storing seven developmental stages into 3 modules. Default is <code>modules = NULL</code>. 
But if specified, a shaded are will be drawn to illustrate stages corresponding to the mid module.</p>
</td></tr>
<tr><td><code id="PlotRE_+3A_legendname">legendName</code></td>
<td>
<p>a character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="PlotRE_+3A_xlab">xlab</code></td>
<td>
<p>label of x-axis.</p>
</td></tr>
<tr><td><code id="PlotRE_+3A_ylab">ylab</code></td>
<td>
<p>label of y-axis.</p>
</td></tr>
<tr><td><code id="PlotRE_+3A_main">main</code></td>
<td>
<p>main text.</p>
</td></tr>
<tr><td><code id="PlotRE_+3A_y.ticks">y.ticks</code></td>
<td>
<p>number of ticks that shall be drawn on the y-axis.</p>
</td></tr>
<tr><td><code id="PlotRE_+3A_adjust.range">adjust.range</code></td>
<td>
<p>logical indicating whether or not the y-axis scale shall be adjusted to the same range in case two groups are specified. Default is <code>adjust.range = TRUE</code>.</p>
</td></tr>
<tr><td><code id="PlotRE_+3A_alpha">alpha</code></td>
<td>
<p>transparency of the shaded area (between [0,1]). Default is <code>alpha = 0.1</code>.</p>
</td></tr>
<tr><td><code id="PlotRE_+3A_...">...</code></td>
<td>
<p>place holder for old version of PlotRE that was based on base graphics instead of ggplot2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Studying the relative expression profiles of each phylostratum or divergence-stratum enables the detection
of common gene expression patterns shared by several phylostrata or divergence-strata.
</p>
<p>Finding similar relative expression profiles among phylostrata or divergence-strata suggests that 
phylostrata or divergence-strata sharing a similar relative expression profile are regulated by similar
gene regulatory elements. Hence, these common phylostrata or divergence-strata might govern similar processes in the given developmental time course.
</p>


<h3>Value</h3>

<p>a plot showing the relative expression profiles of phylostrata or divergence-strata belonging to the same group.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T and Tautz D. 2010. &quot;A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns&quot;. Nature (468): 815-818.
</p>
<p>Quint M et al. 2012. &quot;A transcriptomic hourglass in plant embryogenesis&quot;. Nature (490): 98-101.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotBarRE">PlotBarRE</a></code>, <code><a href="#topic+RE">RE</a></code>, <code><a href="#topic+REMatrix">REMatrix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# read standard phylotranscriptomics data
data(PhyloExpressionSetExample)
data(DivergenceExpressionSetExample)

# example PhyloExpressionSet
PlotRE(PhyloExpressionSetExample,
       Groups = list(c(1:3), c(4:12)), 
       legendName = "PS")


# or you can choose any combination of groups
PlotRE(PhyloExpressionSetExample,
       Groups = list(c(1,7,9), c(2:6,8,10:12)),
       legendName = "PS")

    
# example DivergenceExpressionSet
PlotRE(DivergenceExpressionSetExample,
       Groups = list(c(1:5), c(6:10)), 
       legendName = "DS")


  
</code></pre>

<hr>
<h2 id='PlotReplicateQuality'>Plot the Quality of Biological Replicates</h2><span id='topic+PlotReplicateQuality'></span>

<h3>Description</h3>

<p>This function performs several quality checks to validate the
biological variation between replicates and stages (experiments).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotReplicateQuality(
  ExpressionSet,
  nrep,
  FUN = function(x) log(stats::var(x)),
  legend.pos = "topleft",
  stage.names = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotReplicateQuality_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotReplicateQuality_+3A_nrep">nrep</code></td>
<td>
<p>either a numeric value specifying the constant number of replicates per stage or a numeric vector specifying the variable number of replicates for each stage position.</p>
</td></tr>
<tr><td><code id="PlotReplicateQuality_+3A_fun">FUN</code></td>
<td>
<p>a function that should be applied to quantify the variablity or quality of replicates.
The default function is the log(var(x)) quantifying the variance between replicates.</p>
</td></tr>
<tr><td><code id="PlotReplicateQuality_+3A_legend.pos">legend.pos</code></td>
<td>
<p>the position of the legend, e.g. 'topleft', or 'topright' (see <code><a href="graphics.html#topic+legend">legend</a></code>).</p>
</td></tr>
<tr><td><code id="PlotReplicateQuality_+3A_stage.names">stage.names</code></td>
<td>
<p>a character vector specifying the stage names.</p>
</td></tr>
<tr><td><code id="PlotReplicateQuality_+3A_...">...</code></td>
<td>
<p>additional graphics parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following quality checks can be performed:
</p>

<ul>
<li><p> Quantification of variability between replicates as density function. 
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PhyloExpressionSetExample)

# visualize log(var(x)) between replicates for each gene and developmental stage 
PlotReplicateQuality(ExpressionSet = PhyloExpressionSetExample[1:5000 , 1:8],
                     nrep          = 2,
                     legend.pos   = "topright",
                     ylim          = c(0,0.2),
                     lwd           = 6)
                     

</code></pre>

<hr>
<h2 id='PlotSelectedAgeDistr'>Plot the PS or DS distribution of a selected set of genes</h2><span id='topic+PlotSelectedAgeDistr'></span>

<h3>Description</h3>

<p>This function visualizes the PS or DS distribution of a selected set of genes
as histogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotSelectedAgeDistr(
  ExpressionSet,
  gene.set,
  legendName = NULL,
  as.ratio = FALSE,
  use.only.map = FALSE,
  col = "turquoise4",
  xlab = NULL,
  ylab = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotSelectedAgeDistr_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotSelectedAgeDistr_+3A_gene.set">gene.set</code></td>
<td>
<p>a character vector storing the gene ids for which gene expression profiles shall be visualized.</p>
</td></tr>
<tr><td><code id="PlotSelectedAgeDistr_+3A_legendname">legendName</code></td>
<td>
<p>a character string specifying whether &quot;PS&quot; or &quot;DS&quot; are are visualized.</p>
</td></tr>
<tr><td><code id="PlotSelectedAgeDistr_+3A_as.ratio">as.ratio</code></td>
<td>
<p>logical value indicating whether or not relative frequencies shall be visualized.</p>
</td></tr>
<tr><td><code id="PlotSelectedAgeDistr_+3A_use.only.map">use.only.map</code></td>
<td>
<p>logical value indicating whether or not a Phylostratigraphic Map or Divergence Map should be passed to the <code>ExpressionSet</code> argument instead of a standard <code>ExpressionSet</code> object.</p>
</td></tr>
<tr><td><code id="PlotSelectedAgeDistr_+3A_col">col</code></td>
<td>
<p>colour of the bars.</p>
</td></tr>
<tr><td><code id="PlotSelectedAgeDistr_+3A_xlab">xlab</code></td>
<td>
<p>label of the x-axis.</p>
</td></tr>
<tr><td><code id="PlotSelectedAgeDistr_+3A_ylab">ylab</code></td>
<td>
<p>label of the y-axis.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotDistribution">PlotDistribution</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PhyloExpressionSetExample)

# generate an example gene set
set.seed(123)
ExGeneSet &lt;- sample(PhyloExpressionSetExample[ , 2], 5000)

# gene count example
PlotSelectedAgeDistr(ExpressionSet = PhyloExpressionSetExample,
                     gene.set      = ExGeneSet,
                     legendName    = "PS",
                     as.ratio      = TRUE)

# relative gene count example
PlotSelectedAgeDistr(ExpressionSet = PhyloExpressionSetExample,
                     gene.set      = ExGeneSet,
                     legendName    = "PS",
                     as.ratio      = FALSE)

</code></pre>

<hr>
<h2 id='PlotSignature'>Plot evolutionary signatures across transcriptomes</h2><span id='topic+PlotSignature'></span>

<h3>Description</h3>

<p>Main function to visualize transcriptome indices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotSignature(
  ExpressionSet,
  measure = "TAI",
  TestStatistic = "FlatLineTest",
  modules = NULL,
  permutations = 1000,
  lillie.test = FALSE,
  p.value = TRUE,
  shaded.area = FALSE,
  custom.perm.matrix = NULL,
  xlab = "Ontogeny",
  ylab = "Transcriptome Index",
  main = "",
  lwd = 4,
  alpha = 0.1,
  y.ticks = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotSignature_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet, DivergenceExpressionSet or PolymorphismsExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotSignature_+3A_measure">measure</code></td>
<td>
<p>type of transcriptome index that shall be computed. E.g. 
</p>

<ul>
<li> <p><code>measure = "TAI"</code> (Transcriptome Age Index)
</p>
</li>
<li> <p><code>measure = "TDI"</code> (Transcriptome Divergence Index)
</p>
</li>
<li> <p><code>measure = "TPI"</code> (Transcriptome Polymorphism Index)
</p>
</li></ul>
</td></tr>
<tr><td><code id="PlotSignature_+3A_teststatistic">TestStatistic</code></td>
<td>
<p>a string defining the type of test statistics to be used to quantify the statistical significance the present phylotranscriptomics pattern.
Possible values can be: 
</p>

<ul>
<li> <p><code>TestStatistic</code> = <code>"FlatLineTest"</code> : Statistical test for the deviation from a flat line
</p>
</li>
<li> <p><code>TestStatistic</code> = <code>"ReductiveHourglassTest"</code> : Statistical test for the existence of a hourglass shape (high-low-high pattern)
</p>
</li>
<li> <p><code>TestStatistic</code> = <code>"EarlyConservationTest"</code> : Statistical test for the existence of a earlyconservation pattern (low-high-high pattern)
</p>
</li>
<li> <p><code>TestStatistic</code> = <code>"ReverseHourglassTest"</code> : Statistical test for the existence of a reverse hourglass pattern (low-high-low pattern)
</p>
</li></ul>
</td></tr>
<tr><td><code id="PlotSignature_+3A_modules">modules</code></td>
<td>
<p>a list storing three elements for the <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>, <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code>, or <code><a href="#topic+ReverseHourglassTest">ReverseHourglassTest</a></code>: early, mid, and late. 
Each element expects a numeric vector specifying the developmental stages 
or experiments that correspond to each module. For example:
</p>

<ul>
<li> <p><code>module</code> = <code>list(early = 1:2, mid = 3:5, late = 6:7)</code> devides a dataset storing seven developmental stages into 3 modules.
</p>
</li></ul>
</td></tr>
<tr><td><code id="PlotSignature_+3A_permutations">permutations</code></td>
<td>
<p>a numeric value specifying the number of permutations to be performed for the <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>, <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code>, <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code> or <code><a href="#topic+ReverseHourglassTest">ReverseHourglassTest</a></code>.</p>
</td></tr>
<tr><td><code id="PlotSignature_+3A_lillie.test">lillie.test</code></td>
<td>
<p>a boolean value specifying whether the Lilliefors Kolmogorov-Smirnov Test shall be performed.</p>
</td></tr>
<tr><td><code id="PlotSignature_+3A_p.value">p.value</code></td>
<td>
<p>a boolean value specifying whether the p-value of the test statistic shall be printed within the plot area.</p>
</td></tr>
<tr><td><code id="PlotSignature_+3A_shaded.area">shaded.area</code></td>
<td>
<p>a boolean value specifying whether a shaded area shall 
be drawn for the developmental stages defined to be the presumptive phylotypic period.</p>
</td></tr>
<tr><td><code id="PlotSignature_+3A_custom.perm.matrix">custom.perm.matrix</code></td>
<td>
<p>a custom <code><a href="#topic+bootMatrix">bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic visualized by <code>PlotSignature</code>. Default is <code>custom.perm.matrix = NULL</code>.</p>
</td></tr>
<tr><td><code id="PlotSignature_+3A_xlab">xlab</code></td>
<td>
<p>label of x-axis.</p>
</td></tr>
<tr><td><code id="PlotSignature_+3A_ylab">ylab</code></td>
<td>
<p>label of y-axis.</p>
</td></tr>
<tr><td><code id="PlotSignature_+3A_main">main</code></td>
<td>
<p>figure title.</p>
</td></tr>
<tr><td><code id="PlotSignature_+3A_lwd">lwd</code></td>
<td>
<p>line width.</p>
</td></tr>
<tr><td><code id="PlotSignature_+3A_alpha">alpha</code></td>
<td>
<p>transparency of the shaded area (between [0,1]). Default is <code>alpha = 0.1</code>.</p>
</td></tr>
<tr><td><code id="PlotSignature_+3A_y.ticks">y.ticks</code></td>
<td>
<p>number of ticks on the y-axis. Default is <code>ticks = 10</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function substitutes the functionality of the <code><a href="#topic+PlotPattern">PlotPattern</a></code> function
and is based on ggplot2 insead of base R graphics.
</p>
<p>The following transcriptome indices can be computed and visualized with this function:
</p>

<ul>
<li><p> Transcriptome Age Index (<code><a href="#topic+TAI">TAI</a></code>)
</p>
</li>
<li><p> Transcriptome Divergence Index (<code><a href="#topic+TDI">TDI</a></code>)
</p>
</li>
<li><p> Transcriptome Polymorphism Index (<code><a href="#topic+TPI">TPI</a></code>)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PhyloExpressionSetExample)

# plot TAI pattern and perform flat line test
PlotSignature(PhyloExpressionSetExample, 
              measure       = "TAI", 
              permutations  = 100,
              TestStatistic = "FlatLineTest",
              ylab = "Transcriptome Age Index")
              
</code></pre>

<hr>
<h2 id='PlotVars'>Plot Variance of Expression Profiles</h2><span id='topic+PlotVars'></span>

<h3>Description</h3>

<p>This function computes for each age category the corresponding variance expression profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotVars(
  ExpressionSet,
  Groups = NULL,
  legendName = "age",
  xlab = "Ontogeny",
  ylab = "Variance(Expression Level)",
  main = "",
  y.ticks = 10,
  adjust.range = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotVars_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="PlotVars_+3A_groups">Groups</code></td>
<td>
<p>a list containing the age categories for which variance expression levels shall be drawn.
For ex. evolutionary users can compare old phylostrata: PS1-3 (Class 1) and evolutionary young phylostrata: PS4-12 (Class 2). 
In this example, the list could be assigned as, <code>Groups = list(c(1:3), c(4:12))</code>. 
The group options is limited to 2 Groups.</p>
</td></tr>
<tr><td><code id="PlotVars_+3A_legendname">legendName</code></td>
<td>
<p>a character string specifying the legend title.</p>
</td></tr>
<tr><td><code id="PlotVars_+3A_xlab">xlab</code></td>
<td>
<p>label of x-axis.</p>
</td></tr>
<tr><td><code id="PlotVars_+3A_ylab">ylab</code></td>
<td>
<p>label of y-axis.</p>
</td></tr>
<tr><td><code id="PlotVars_+3A_main">main</code></td>
<td>
<p>main text.</p>
</td></tr>
<tr><td><code id="PlotVars_+3A_y.ticks">y.ticks</code></td>
<td>
<p>number of ticks that shall be drawn on the y-axis.</p>
</td></tr>
<tr><td><code id="PlotVars_+3A_adjust.range">adjust.range</code></td>
<td>
<p>logical indicating whether or not the y-axis scale shall be adjusted to the same range in case two groups are specified. Default is <code>adjust.range = TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This plot may be useful to compare the absolute variance expression        
levels of each age category across stages.
</p>
<p>In different developmental processes different phylostratum or divergence-stratum
classes might be more expressed than others, hence contributing more to the overall
phylotranscriptomics pattern (<code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code>).
This plot can help to identify the phylostratum or divergence-stratum classes 
that contributes most to the overall transcriptome of the given developmental process.
</p>


<h3>Value</h3>

<p>a plot showing variance expression profiles of each age category.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotBarRE">PlotBarRE</a></code>, <code><a href="#topic+RE">RE</a></code>, <code><a href="#topic+REMatrix">REMatrix</a></code>, <code><a href="#topic+PlotRE">PlotRE</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Example using a PhyloExpressionSet
### and DivergenceExpressionSet
# load PhyloExpressionSet
data(PhyloExpressionSetExample)

# load PhyloExpressionSet
data(DivergenceExpressionSetExample)

# plot evolutionary old PS (PS1-3) vs evolutionary young PS (PS4-12)
PlotVars(PhyloExpressionSetExample,
          Groups = list(c(1:3), c(4:12)), 
          legendName = "PS",
          adjust.range = TRUE)

# if users wish to not adjust the y-axis scale when 
# 2 groups are selected they can specify: adjust.range = FALSE
PlotVars(PhyloExpressionSetExample,
          Groups = list(c(1:3), c(4:12)), 
          legendName = "PS",
          adjust.range = FALSE)
          
          
# plot conserved DS (DS1-5) vs divergent DS (PS6-10)
# NOTE: DS are always defined in the range 1, 2, ... , 10.
# Hence, make sure that your groups are within this range!
PlotVars(DivergenceExpressionSetExample,
          Groups = list(c(1:5), c(6:10)), 
          legendName = "DS",
          adjust.range = TRUE)

</code></pre>

<hr>
<h2 id='pMatrix'>Compute Partial TAI or TDI Values</h2><span id='topic+pMatrix'></span>

<h3>Description</h3>

<p>This function computes the partial <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> 
values for each single gene in a PhyloExpressionSet or DivergenceExpressionSet object.
</p>
<p>In detail, each gene gets a <em>TAI contribution profile</em> or <em>TDI contribution profile</em>. 
</p>
<p style="text-align: center;"><code class="reqn">TAI_is = f_is * ps_i</code>
</p>

<p>or
</p>
<p style="text-align: center;"><code class="reqn">TDI_is = f_is * ps_i</code>
</p>

<p>where TAI_is or TDI_is is the partial TAI or TDI value of gene i, 
<code class="reqn">f_is = e_is / \sum e_is</code> and <code class="reqn">ps_i</code> is the phylostratum or divergence-stratum of gene i.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pMatrix(ExpressionSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pMatrix_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The partial TAI or TDI matrix can be used to perform different cluster analyses
and also gives an overall impression of the contribution of each gene to the global <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> pattern.
</p>


<h3>Value</h3>

<p>a numeric matrix storing the partial TAI or TDI values for each gene in the 
corresponding PhyloExpressionSet or DivergenceExpressionSet.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T and Tautz D. 2010. &quot;A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns&quot;. Nature (468): 815-818.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

# read standard phylotranscriptomics data
data(PhyloExpressionSetExample)
data(DivergenceExpressionSetExample)

# example PhyloExpressionSet
PTM_ps &lt;- pMatrix(PhyloExpressionSetExample)

# example DivergenceExpressionSet
PTM_ds &lt;- pMatrix(DivergenceExpressionSetExample)

# boxplot of the pMatrix
boxplot(pMatrix(PhyloExpressionSetExample),outline = FALSE)

# boxplot of the pMatrix using log2 transformed expression levels
boxplot(pMatrix(tf(PhyloExpressionSetExample,log2)))
 

</code></pre>

<hr>
<h2 id='pStrata'>Compute Partial Strata Values</h2><span id='topic+pStrata'></span>

<h3>Description</h3>

<p>This function computes the partial <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code>
values for all Phylostrata or Divergence Strata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pStrata(ExpressionSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pStrata_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)

# compute partial TAI values for each Phylostratum
partialStrata &lt;- pStrata(PhyloExpressionSetExample)

# show that colSums of pStrata is equavalent to the TAI values
all.equal(colSums(partialStrata),TAI(PhyloExpressionSetExample))

# show that colSums of pStrata is equavalent to colSums of pMatrix(PhyloExpressionSetExample)
all.equal(colSums(partialStrata),colSums(pMatrix(PhyloExpressionSetExample)))


</code></pre>

<hr>
<h2 id='pTAI'>Compute the Phylostratum Contribution to the Global Transcriptome Age Index</h2><span id='topic+pTAI'></span>

<h3>Description</h3>

<p>This function takes a standard <em>ExpressionSet</em> object and computes the partial contribution of the different phylostrata (ps) to the global Transcriptome Age Index profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pTAI(ExpressionSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pTAI_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This way of computing the partial contribution of the different phylostrata (ps) to the global Transcriptome Age Index profile was introduced by Domazet-Loso and Tautz, 2010. This function (<code>pTAI</code>) computes the partial TAI contribution for each phylostratum and each developmental stage and returns a data matrix storing the partial TAI contribution value for each phylostratum and each developmental stage.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T. and Tautz D. (2010). <em>A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns</em>. Nature (468): 815-818.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pTDI">pTDI</a></code>, <code><a href="#topic+TAI">TAI</a></code>, <code><a href="#topic+TDI">TDI</a></code>, <code><a href="#topic+PlotPattern">PlotPattern</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)

# get the partial contribution of phylostrata to the global
# TAI pattern
pTAI(PhyloExpressionSetExample)

</code></pre>

<hr>
<h2 id='pTDI'>Compute the Divergence Stratum Contribution to the Global Transcriptome Divergence Index</h2><span id='topic+pTDI'></span>

<h3>Description</h3>

<p>This function takes a standard <em>ExpressionSet</em> object and computes the partial contribution of the different divergence strata (ds) to the global Transcriptome Divergence Index profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pTDI(ExpressionSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pTDI_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard DivergenceExpressionSet object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function (<code>pTDI</code>) computes the partial TDI contribution for each phylostratum and each developmental stage and returns a data matrix storing the partial TDI contribution value for each divergence and each developmental stage.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T. and Tautz D. (2010). <em>A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns</em>. Nature (468): 815-818.
</p>
<p>Quint M et al. (2012). A transcriptomic hourglass in plant embryogenesis. Nature (490): 98-101.
</p>
<p>Drost HG et al. (2015). Evidence for Active Maintenance of Phylotranscriptomic Hourglass Patterns in Animal and Plant Embryogenesis. Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pTAI">pTAI</a></code>, <code><a href="#topic+TAI">TAI</a></code>, <code><a href="#topic+TDI">TDI</a></code>, <code><a href="#topic+PlotPattern">PlotPattern</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(DivergenceExpressionSetExample)

# get the partial contribution of divergence strata to the global
# TDI pattern
pTAI(DivergenceExpressionSetExample)

</code></pre>

<hr>
<h2 id='RE'>Transform to Relative Expression Levels</h2><span id='topic+RE'></span>

<h3>Description</h3>

<p>This function computes the relative expression profiles of any given gene expression set. 
The relative expression profile is being computed as follows:
</p>
<p style="text-align: center;"><code class="reqn">f_s = ( e_s - e_min ) / ( e_max - e_min )</code>
</p>

<p>where <code class="reqn">e_min</code> and <code class="reqn">e_max</code> denote the minimum/maximum mean expression level
over the  developmental stages s. This linear transformation corresponds to a
shift by <code class="reqn">e_min</code> and a subsequent shrinkage by <code class="reqn">e_max - e_min</code>. 
As a result, the relative expression level <code class="reqn">f_s</code> of developmental stage s
with minimum <code class="reqn">e_s</code> is 0, the relative expression level <code class="reqn">f_s</code> of the
developmental stage s with maximum <code class="reqn">e_s</code> is 1, and the relative expression
levels <code class="reqn">f_s</code> of all other stages s range between 0 and 1, accordingly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RE(ExpressionMatrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RE_+3A_expressionmatrix">ExpressionMatrix</code></td>
<td>
<p>a numeric matrix representing a gene expression matrix for which the relative expression profile shall be computed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector containing the relative expression profile of the correspnding data matrix.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T and Tautz D. (2010). <em>A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns</em>. Nature (468): 815-818.
</p>
<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+REMatrix">REMatrix</a></code>, <code><a href="#topic+PlotRE">PlotRE</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# read standard phylotranscriptomics data
data(PhyloExpressionSetExample)

# relative expression profile of PS1 genes
RE(PhyloExpressionSetExample[ which(PhyloExpressionSetExample[ , 1] == 1), 3:9 ])


</code></pre>

<hr>
<h2 id='ReductiveHourglassTest'>Perform the Reductive Hourglass Test</h2><span id='topic+ReductiveHourglassTest'></span>

<h3>Description</h3>

<p>The <em>Reductive Hourglass Test</em> aims to statistically evaluate the
existence of a phylotranscriptomic hourglass pattern based on <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> computations.
The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern) 
does not follow an hourglass like shape. A p-value &lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does
indeed follow an hourglass (high-low-high) shape.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReductiveHourglassTest(
  ExpressionSet,
  modules = NULL,
  permutations = 1000,
  lillie.test = FALSE,
  plotHistogram = FALSE,
  runs = 10,
  parallel = FALSE,
  gof.warning = FALSE,
  custom.perm.matrix = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReductiveHourglassTest_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="ReductiveHourglassTest_+3A_modules">modules</code></td>
<td>
<p>a list storing three elements: early, mid, and late. Each element expects a numeric
vector specifying the developmental stages or experiments that correspond to each module. 
For example, <code>module</code> = list(early = 1:2, mid = 3:5, late = 6:7) devides a dataset 
storing seven developmental stages into 3 modules.</p>
</td></tr>
<tr><td><code id="ReductiveHourglassTest_+3A_permutations">permutations</code></td>
<td>
<p>a numeric value specifying the number of permutations to be performed for the <code>ReductiveHourglassTest</code>.</p>
</td></tr>
<tr><td><code id="ReductiveHourglassTest_+3A_lillie.test">lillie.test</code></td>
<td>
<p>a boolean value specifying whether the Lilliefors Kolmogorov-Smirnov Test shall be performed to quantify the goodness of fit.</p>
</td></tr>
<tr><td><code id="ReductiveHourglassTest_+3A_plothistogram">plotHistogram</code></td>
<td>
<p>a boolean value specifying whether a <em>Lillifor's Kolmogorov-Smirnov-Test</em> 
shall be performed to test the goodness of fit of the approximated distribution, as well as additional plots quantifying the significance
of the observed phylotranscriptomic pattern.</p>
</td></tr>
<tr><td><code id="ReductiveHourglassTest_+3A_runs">runs</code></td>
<td>
<p>specify the number of runs to be performed for goodness of fit computations, in case <code>plotHistogram</code> = <code>TRUE</code>.
In most cases <code>runs</code> = 100 is a reasonable choice. Default is <code>runs</code> = 10 (because it takes less computation time for demonstration purposes).</p>
</td></tr>
<tr><td><code id="ReductiveHourglassTest_+3A_parallel">parallel</code></td>
<td>
<p>performing <code>runs</code> in parallel (takes all cores of your multicore machine).</p>
</td></tr>
<tr><td><code id="ReductiveHourglassTest_+3A_gof.warning">gof.warning</code></td>
<td>
<p>a logical value indicating whether non significant goodness of fit results should be printed as warning. Default is <code>gof.warning = FALSE</code>.</p>
</td></tr>
<tr><td><code id="ReductiveHourglassTest_+3A_custom.perm.matrix">custom.perm.matrix</code></td>
<td>
<p>a custom <code><a href="#topic+bootMatrix">bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic. Default is <code>custom.perm.matrix = NULL</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The reductive hourglass test is a permutation test based on the following test statistic. 
</p>
<p>(1) A set of developmental stages is partitioned into three modules - early, mid, and late - based on prior biological knowledge.
</p>
<p>(2) The mean <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> value for each of the three modules T_early, T_mid, and T_late are computed. 
</p>
<p>(3) The two differences D1 = T_early - T_mid and D2 = T_late - T_mid are calculated.
</p>
<p>(4) The minimum D_min of D1 and D2 is computed as final test statistic of the reductive hourglass test.
</p>
<p>In order to determine the statistical significance of an observed minimum difference D_min 
the following permutation test was performed. Based on the <code><a href="#topic+bootMatrix">bootMatrix</a></code> D_min 
is calculated from each of the permuted <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> profiles, 
approximated by a Gaussian distribution with method of moments estimated parameters returned by <code><a href="fitdistrplus.html#topic+fitdist">fitdist</a></code>, 
and the corresponding p-value is computed by <code><a href="stats.html#topic+pnorm">pnorm</a></code> given the estimated parameters of the Gaussian distribution. 
The <em>goodness of fit</em> for the random vector <em>D_min</em> is statistically quantified by an Lilliefors (Kolmogorov-Smirnov) test 
for normality.
</p>
<p>In case the parameter <em>plotHistogram = TRUE</em>, a multi-plot is generated showing:
</p>
<p>(1) A Cullen and Frey skewness-kurtosis plot generated by <code><a href="fitdistrplus.html#topic+descdist">descdist</a></code>. 
This plot illustrates which distributions seem plausible to fit the resulting permutation vector D_min. 
In the case of the <em>Reductive Hourglass Test</em> a normal distribution seemed plausible.
</p>
<p>(2) A histogram of D_min combined with the density plot is plotted. D_min is then fitted by a normal distribution. 
The corresponding parameters are estimated by <em>moment matching estimation</em> using the <code><a href="fitdistrplus.html#topic+fitdist">fitdist</a></code> function.
</p>
<p>(3) A plot showing the p-values for N independent runs to verify that a specific p-value is biased by a specific permutation order.
</p>
<p>(4) A barplot showing the number of cases in which the underlying goodness of fit (returned by Lilliefors (Kolmogorov-Smirnov) test 
for normality) has shown to be significant (<code>TRUE</code>) or not significant (<code>FALSE</code>). 
This allows to quantify the permutation bias and their implications on the goodness of fit.
</p>


<h3>Value</h3>

<p>a list object containing the list elements:
</p>
<p><code>p.value</code> : the p-value quantifying the statistical significance (high-low-high pattern) of the given phylotranscriptomics pattern.
</p>
<p><code>std.dev</code> : the standard deviation of the N sampled phylotranscriptomics patterns for each developmental stage S.
</p>
<p><code>lillie.test</code> : a boolean value specifying whether the <em>Lillifors KS-Test</em> returned a p-value &gt; 0.05, 
which indicates that fitting the permuted scores with a normal distribution seems plausible.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>
<p>Quint M et al. (2012). A transcriptomic hourglass in plant embryogenesis. Nature (490): 98-101.
</p>
<p>M. L. Delignette-Muller, R. Pouillot, J.-B. Denis and C. Dutang (2014), fitdistrplus: help to fit
of a parametric distribution to non-censored or censored data.
</p>
<p>Cullen AC and Frey HC (1999) Probabilistic techniques in exposure assessment. Plenum Press, USA, pp. 81-159.
</p>
<p>Evans M, Hastings N and Peacock B (2000) Statistical distributions. John Wiley and Sons Inc.
</p>
<p>Sokal RR and Rohlf FJ (1995) Biometry. W.H. Freeman and Company, USA, pp. 111-115.
</p>
<p>Juergen Gross and bug fixes by Uwe Ligges (2012). nortest: Tests for Normality. R package version
1.0-2. 
</p>
<p>http://CRAN.R-project.org/package=nortest
</p>
<p>Dallal, G.E. and Wilkinson, L. (1986): An analytic approximation to the distribution of Lilliefors' test for normality. The American Statistician, 40, 294-296.
</p>
<p>Stephens, M.A. (1974): EDF statistics for goodness of fit and some comparisons. Journal of the American Statistical Association, 69, 730-737.
</p>
<p>http://stackoverflow.com/questions/4290081/fitting-data-to-distributions?rq=1
</p>
<p>http://stats.stackexchange.com/questions/45033/can-i-use-kolmogorov-smirnov-test-and-estimate-distribution-parameters
</p>
<p>http://cran.r-project.org/doc/contrib/Ricci-distributions-en.pdf
</p>
<p>http://cran.r-project.org/doc/contrib/Ricci-distributions-en.pdf
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rhScore">rhScore</a></code>, <code><a href="#topic+bootMatrix">bootMatrix</a></code>, <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>,
<code><a href="#topic+ReverseHourglassTest">ReverseHourglassTest</a></code>, <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code>, <code><a href="#topic+PlotSignature">PlotSignature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)

# perform the reductive hourglass test for a PhyloExpressionSet
# here the prior biological knowledge is that stages 1-2 correspond to module 1 = early,
# stages 3-5 to module 2 = mid (phylotypic module), and stages 6-7 correspond to
# module 3 = late
ReductiveHourglassTest(PhyloExpressionSetExample,
                       modules = list(early = 1:2, mid = 3:5, late = 6:7), 
                       permutations = 1000)


# use your own permutation matrix based on which p-values (ReductiveHourglassTest)
# shall be computed
custom_perm_matrix &lt;- bootMatrix(PhyloExpressionSetExample,100)

ReductiveHourglassTest(PhyloExpressionSetExample,
                     modules = list(early = 1:2, mid = 3:5, late = 6:7),
                     custom.perm.matrix = custom_perm_matrix)


</code></pre>

<hr>
<h2 id='REMatrix'>Compute a Relative Expression Matrix</h2><span id='topic+REMatrix'></span>

<h3>Description</h3>

<p>This function computes the relative expression profiles of 
all given phylostrata or divergence-strata within a given PhyloExpressionSet or DivergenceExpressionSet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>REMatrix(ExpressionSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="REMatrix_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each phylostratum or divergence-stratum the corresponding relative expression profile is being computed as follows:
</p>
<p style="text-align: center;"><code class="reqn">f_js = ( e_js - e_j min ) / ( e_j max - e_j min )</code>
</p>

<p>where <code class="reqn">e_j min</code> and <code class="reqn">e_j max</code> denote the minimum/maximum mean expression level of 
phylostratum j over the  developmental stages s. This linear transformation corresponds
to a shift by <code class="reqn">e_j min</code> and a subsequent shrinkage by <code class="reqn">e_j max - e_j min</code>. 
As a result, the relative expression level <code class="reqn">f_js</code> of developmental stage s with minimum <code class="reqn">e_js</code> is 0,
the relative expression level <code class="reqn">f_js</code> of the developmental stage s with maximum <code class="reqn">e_js</code> is 1, 
and the relative expression levels <code class="reqn">f_js</code> of all other stages s range between 0 and 1, accordingly.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T and Tautz D. (2010). <em>A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns</em>. Nature (468): 815-818.
</p>
<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+RE">RE</a></code>, <code><a href="#topic+PlotRE">PlotRE</a></code>, <code><a href="#topic+PlotBarRE">PlotBarRE</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# read standard phylotranscriptomics data
data(PhyloExpressionSetExample)
data(DivergenceExpressionSetExample)

# example PhyloExpressionSet
REMatrix(PhyloExpressionSetExample)

# example DivergenceExpressionSet
REMatrix(DivergenceExpressionSetExample)


</code></pre>

<hr>
<h2 id='reversehourglassScore'>Compute the Reverse Hourglass Score for the Reverse Hourglass Test</h2><span id='topic+reversehourglassScore'></span>

<h3>Description</h3>

<p>This function reduces the destruction of an hourglass shaped pattern to a single score value.
</p>
<p>Based on a given <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> pattern the given vector is being
divided into three developmental modules: early, mid, and late. The corrisponding <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code>
values in each developmental module are accumulated using the <em>scoringMethod</em> argument (&quot;max-min&quot; or &quot;mean-mean&quot;).
</p>
<p>In more detail:
</p>
<p>(1) for a given <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> vector <em>tai_profile</em> or <em>tdi_profile</em>, we classify each value of <em>tai_profile</em> or <em>tdi_profile</em> into its corresponding developmental module early, mid, or late.
</p>
<p>(2) accumulate the <em>tai_profile</em> or <em>tdi_profile</em> values in each developmental module using the arithmetic mean (<code><a href="base.html#topic+mean">mean</a></code>) in case scoringMethod = &quot;mean-mean&quot;, or accumulate the <em>tai_profile</em> or <em>tdi_profile</em> values in each developmental module using <code><a href="base.html#topic+max">max</a></code> for the early and late module and <code><a href="base.html#topic+min">min</a></code> for the mid module in case scoringMethod = &quot;max-min&quot;.
</p>
<p>(3) then reduce the three values for each developmental module by computing the difference between: mid - early, and mid - late.
</p>
<p>(4) the two difference values are referred to as a_early and a_late. 
</p>
<p>Each developmental module now has an accumulated representation value which is being reduced to one value using the
<em>method</em> argument (&quot;max&quot;, &quot;min&quot;, or &quot;mean&quot;). 
</p>
<p>Given the two accumulated values for each hourglass module: a_early and a_late,
we reduce the two given values by:
</p>

<ul>
<li> <p><em>&quot;max&quot;</em>: <code class="reqn">S = max{a_early,a_late}</code>
</p>
</li>
<li> <p><em>&quot;min&quot;</em>: <code class="reqn">S = min{a_early,a_late}</code>
</p>
</li>
<li> <p><em>&quot;mean&quot;</em>: <code class="reqn">S = mean{a_early,a_late}</code>
</p>
</li></ul>

<p>All together this results in a global score <em>S</em>.
This global score <em>S</em> is being returned by this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reversehourglassScore(age_vals, early, mid, late, method, scoringMethod)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reversehourglassScore_+3A_age_vals">age_vals</code></td>
<td>
<p>a numeric vector containing <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> values for each developmental stage s.</p>
</td></tr>
<tr><td><code id="reversehourglassScore_+3A_early">early</code></td>
<td>
<p>a numeric vector including the numeric stage values that correspond to the early phase of development.</p>
</td></tr>
<tr><td><code id="reversehourglassScore_+3A_mid">mid</code></td>
<td>
<p>a numeric vector including the numeric stage values that correspond to the middle phase of development.</p>
</td></tr>
<tr><td><code id="reversehourglassScore_+3A_late">late</code></td>
<td>
<p>a numeric vector including the numeric stage values that correspond to the late phase of development.</p>
</td></tr>
<tr><td><code id="reversehourglassScore_+3A_method">method</code></td>
<td>
<p>to determine the two value reduction value, resulting in the global score S: &quot;max&quot;, or &quot;min&quot;, or &quot;mean&quot;.</p>
</td></tr>
<tr><td><code id="reversehourglassScore_+3A_scoringmethod">scoringMethod</code></td>
<td>
<p>method to determine the module accumulation value: &quot;max-min&quot; or &quot;mean-mean&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The gpScore is a heuristic score enabling to  construct a test statistic to determine  
the significance of a present (phylotranscriptomic) hourglass pattern.
</p>


<h3>Value</h3>

<p>a numeric value representing the hourglass destruction score.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Drost et al. (2015), Evidence for active maintenance of phylotranscriptomic hourglass patterns in animal and plant embryogenesis. Mol Bio Evol.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ReverseHourglassTest">ReverseHourglassTest</a></code>, <code><a href="#topic+TAI">TAI</a></code>, <code><a href="#topic+TDI">TDI</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # read standard phylotranscriptomics data
 data(PhyloExpressionSetExample)
 data(DivergenceExpressionSetExample)

 # example PhyloExpressionSet:

 # compute the TAI profile
 TAIs &lt;- TAI(PhyloExpressionSetExample)

 # compute the global hourglass destruction score 
 # for the TAIs profile using reduction method: mean(mean-mean)
 reversehourglass_score &lt;- reversehourglassScore(age_vals = TAIs,early = 1:2,mid = 3:5,late = 6:7,
                     method = "mean",scoringMethod = "mean-mean")


 # example DivergenceExpressionSet:

 # compute the TDI profile
 TDIs &lt;- TDI(DivergenceExpressionSetExample)

 # compute the global hourglass destruction score for the TDIs profile 
 # using reduction method: mean(mean-mean)
 reversehourglass_score &lt;- rhScore(age_vals = TDIs,early = 1:2,mid = 3:5,late = 6:7,
                     method = "mean",scoringMethod = "mean-mean")
 
 
</code></pre>

<hr>
<h2 id='ReverseHourglassTest'>Perform the Reverse Hourglass Test</h2><span id='topic+ReverseHourglassTest'></span>

<h3>Description</h3>

<p>The <em>Reverse Hourglass Test</em> aims to statistically evaluate the
existence of a reverse hourglass pattern based on <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> computations.
The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern)
does follow an hourglass like shape. A p-value &lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does
rather follow a reverse hourglass (low-high-low) shape.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReverseHourglassTest(
  ExpressionSet,
  modules = NULL,
  permutations = 1000,
  lillie.test = FALSE,
  plotHistogram = FALSE,
  runs = 10,
  parallel = FALSE,
  gof.warning = FALSE,
  custom.perm.matrix = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReverseHourglassTest_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="ReverseHourglassTest_+3A_modules">modules</code></td>
<td>
<p>a list storing three elements: early, mid, and late. Each element expects a numeric
vector specifying the developmental stages or experiments that correspond to each module.
For example, <code>module</code> = list(early = 1:2, mid = 3:5, late = 6:7) devides a dataset
storing seven developmental stages into 3 modules.</p>
</td></tr>
<tr><td><code id="ReverseHourglassTest_+3A_permutations">permutations</code></td>
<td>
<p>a numeric value specifying the number of permutations to be performed for the <code>ReverseHourglassTest</code>.</p>
</td></tr>
<tr><td><code id="ReverseHourglassTest_+3A_lillie.test">lillie.test</code></td>
<td>
<p>a boolean value specifying whether the Lilliefors Kolmogorov-Smirnov Test shall be performed to quantify the goodness of fit.</p>
</td></tr>
<tr><td><code id="ReverseHourglassTest_+3A_plothistogram">plotHistogram</code></td>
<td>
<p>a boolean value specifying whether a <em>Lillifor's Kolmogorov-Smirnov-Test</em>
shall be performed to test the goodness of fit of the approximated distribution, as well as additional plots quantifying the significance
of the observed phylotranscriptomic pattern.</p>
</td></tr>
<tr><td><code id="ReverseHourglassTest_+3A_runs">runs</code></td>
<td>
<p>specify the number of runs to be performed for goodness of fit computations, in case <code>plotHistogram</code> = <code>TRUE</code>.
In most cases <code>runs</code> = 100 is a reasonable choice. Default is <code>runs</code> = 10 (because it takes less computation time for demonstration purposes).</p>
</td></tr>
<tr><td><code id="ReverseHourglassTest_+3A_parallel">parallel</code></td>
<td>
<p>performing <code>runs</code> in parallel (takes all cores of your multicore machine).</p>
</td></tr>
<tr><td><code id="ReverseHourglassTest_+3A_gof.warning">gof.warning</code></td>
<td>
<p>a logical value indicating whether non significant goodness of fit results should be printed as warning. Default is <code>gof.warning = FALSE</code>.</p>
</td></tr>
<tr><td><code id="ReverseHourglassTest_+3A_custom.perm.matrix">custom.perm.matrix</code></td>
<td>
<p>a custom <code><a href="#topic+bootMatrix">bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic. Default is <code>custom.perm.matrix = NULL</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The reverse hourglass test is a permutation test based on the following test statistic.
</p>
<p>(1) A set of developmental stages is partitioned into three modules - early, mid, and late - based on prior biological knowledge.
</p>
<p>(2) The mean <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> value for each of the three modules T_early, T_mid, and T_late are computed.
</p>
<p>(3) The two differences <code>D1 = T_mid - T_early</code> and <code>D2 = T_mid - T_late</code> are calculated.
</p>
<p>(4) The minimum <code>D_min</code> of <code>D1</code> and <code>D2</code> is computed as final test statistic of the reductive hourglass test.
</p>
<p>In order to determine the statistical significance of an observed minimum difference D_min
the following permutation test was performed. Based on the <code><a href="#topic+bootMatrix">bootMatrix</a></code> D_min
is calculated from each of the permuted <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> profiles,
approximated by a Gaussian distribution with method of moments estimated parameters returned by <code><a href="fitdistrplus.html#topic+fitdist">fitdist</a></code>,
and the corresponding p-value is computed by <code><a href="stats.html#topic+pnorm">pnorm</a></code> given the estimated parameters of the Gaussian distribution.
The <em>goodness of fit</em> for the random vector <em>D_min</em> is statistically quantified by an Lilliefors (Kolmogorov-Smirnov) test for normality.
</p>
<p>In case the parameter <em>plotHistogram = TRUE</em>, a multi-plot is generated showing:
</p>
<p>(1) A Cullen and Frey skewness-kurtosis plot generated by <code><a href="fitdistrplus.html#topic+descdist">descdist</a></code>.
This plot illustrates which distributions seem plausible to fit the resulting permutation vector D_min.
In the case of the <em>Reverse Hourglass Test</em> a normal distribution seemed plausible.
</p>
<p>(2) A histogram of <code>D_min</code> combined with the density plot is plotted. D_min is then fitted by a normal distribution.
The corresponding parameters are estimated by <em>moment matching estimation</em> using the <code><a href="fitdistrplus.html#topic+fitdist">fitdist</a></code> function.
</p>
<p>(3) A plot showing the p-values for N independent runs to verify that a specific p-value is biased by a specific permutation order.
</p>
<p>(4) A barplot showing the number of cases in which the underlying goodness of fit (returned by Lilliefors (Kolmogorov-Smirnov) test
for normality) has shown to be significant (<code>TRUE</code>) or not significant (<code>FALSE</code>).
This allows to quantify the permutation bias and their implications on the goodness of fit.
</p>


<h3>Value</h3>

<p>a list object containing the list elements:
</p>
<p><code>p.value</code> : the p-value quantifying the statistical significance (low-high-low pattern) of the given phylotranscriptomics pattern.
</p>
<p><code>std.dev</code> : the standard deviation of the N sampled phylotranscriptomics patterns for each developmental stage S.
</p>
<p><code>lillie.test</code> : a boolean value specifying whether the <em>Lillifors KS-Test</em> returned a p-value &gt; 0.05,
which indicates that fitting the permuted scores with a normal distribution seems plausible.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>
<p>Quint M et al. (2012). A transcriptomic hourglass in plant embryogenesis. Nature (490): 98-101.
</p>
<p>M. L. Delignette-Muller, R. Pouillot, J.-B. Denis and C. Dutang (2014), fitdistrplus: help to fit
of a parametric distribution to non-censored or censored data.
</p>
<p>Cullen AC and Frey HC (1999) Probabilistic techniques in exposure assessment. Plenum Press, USA, pp. 81-159.
</p>
<p>Evans M, Hastings N and Peacock B (2000) Statistical distributions. John Wiley and Sons Inc.
</p>
<p>Sokal RR and Rohlf FJ (1995) Biometry. W.H. Freeman and Company, USA, pp. 111-115.
</p>
<p>Juergen Gross and bug fixes by Uwe Ligges (2012). nortest: Tests for Normality. R package version
1.0-2.
</p>
<p>http://CRAN.R-project.org/package=nortest
</p>
<p>Dallal, G.E. and Wilkinson, L. (1986): An analytic approximation to the distribution of Lilliefors' test for normality. The American Statistician, 40, 294-296.
</p>
<p>Stephens, M.A. (1974): EDF statistics for goodness of fit and some comparisons. Journal of the American Statistical Association, 69, 730-737.
</p>
<p>http://stackoverflow.com/questions/4290081/fitting-data-to-distributions?rq=1
</p>
<p>http://stats.stackexchange.com/questions/45033/can-i-use-kolmogorov-smirnov-test-and-estimate-distribution-parameters
</p>
<p>http://cran.r-project.org/doc/contrib/Ricci-distributions-en.pdf
</p>
<p>http://cran.r-project.org/doc/contrib/Ricci-distributions-en.pdf
</p>


<h3>See Also</h3>

<p><code><a href="#topic+reversehourglassScore">reversehourglassScore</a></code>, <code><a href="#topic+bootMatrix">bootMatrix</a></code>, <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>, <code><a href="#topic+EarlyConservationTest">EarlyConservationTest</a></code>, <code><a href="#topic+PlotSignature">PlotSignature</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)

# perform the reductive hourglass test for a PhyloExpressionSet
# here the prior biological knowledge is that stages 1-2 correspond to module 1 = early,
# stages 3-5 to module 2 = mid (phylotypic module), and stages 6-7 correspond to
# module 3 = late
ReverseHourglassTest(PhyloExpressionSetExample,
                       modules = list(early = 1:2, mid = 3:5, late = 6:7),
                       permutations = 1000)


# use your own permutation matrix based on which p-values (ReverseHourglassTest)
# shall be computed
custom_perm_matrix &lt;- bootMatrix(PhyloExpressionSetExample,100)

ReverseHourglassTest(PhyloExpressionSetExample,
                     modules = list(early = 1:2, mid = 3:5, late = 6:7),
                     custom.perm.matrix = custom_perm_matrix)


</code></pre>

<hr>
<h2 id='rhScore'>Compute the Hourglass Score for the Reductive Hourglass Test</h2><span id='topic+rhScore'></span>

<h3>Description</h3>

<p>This function reduces the destruction of an hourglass shaped pattern to a single score value.
</p>
<p>Based on a given <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> pattern the given vector is being
divided into three developmental modules: early, mid, and late. The corrisponding <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code>
values in each developmental module are accumulated using the <em>scoringMethod</em> argument (&quot;max-min&quot; or &quot;mean-mean&quot;).
</p>
<p>In more detail:
</p>
<p>(1) for a given <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> vector <em>tai_profile</em> or <em>tdi_profile</em>, we classify each value of <em>tai_profile</em> or <em>tdi_profile</em> into its corresponding developmental module early, mid, or late.
</p>
<p>(2) accumulate the <em>tai_profile</em> or <em>tdi_profile</em> values in each developmental module using the arithmetic mean (<code><a href="base.html#topic+mean">mean</a></code>) in case scoringMethod = &quot;mean-mean&quot;, or accumulate the <em>tai_profile</em> or <em>tdi_profile</em> values in each developmental module using <code><a href="base.html#topic+max">max</a></code> for the early and late module and <code><a href="base.html#topic+min">min</a></code> for the mid module in case scoringMethod = &quot;max-min&quot;.
</p>
<p>(3) then reduce the three values for each developmental module by computing the difference between: early - mid, and late - mid.
</p>
<p>(4) the two difference values are referred to as a_early and a_late. 
</p>
<p>Each developmental module now has an accumulated representation value which is being reduced to one value using the
<em>method</em> argument (&quot;max&quot;, &quot;min&quot;, or &quot;mean&quot;). 
</p>
<p>Given the two accumulated values for each hourglass module: a_early and a_late,
we reduce the two given values by:
</p>

<ul>
<li> <p><em>&quot;max&quot;</em>: <code class="reqn">S = max{a_early,a_late}</code>
</p>
</li>
<li> <p><em>&quot;min&quot;</em>: <code class="reqn">S = min{a_early,a_late}</code>
</p>
</li>
<li> <p><em>&quot;mean&quot;</em>: <code class="reqn">S = mean{a_early,a_late}</code>
</p>
</li></ul>

<p>All together this results in a global score <em>S</em>.
This global score <em>S</em> is being returned by this function <code><a href="#topic+rhScore">rhScore</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rhScore(age_vals, early, mid, late, method, scoringMethod)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rhScore_+3A_age_vals">age_vals</code></td>
<td>
<p>a numeric vector containing <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code> values for each developmental stage s.</p>
</td></tr>
<tr><td><code id="rhScore_+3A_early">early</code></td>
<td>
<p>a numeric vector including the numeric stage values that correspond to the early phase of development.</p>
</td></tr>
<tr><td><code id="rhScore_+3A_mid">mid</code></td>
<td>
<p>a numeric vector including the numeric stage values that correspond to the middle phase of development.</p>
</td></tr>
<tr><td><code id="rhScore_+3A_late">late</code></td>
<td>
<p>a numeric vector including the numeric stage values that correspond to the late phase of development.</p>
</td></tr>
<tr><td><code id="rhScore_+3A_method">method</code></td>
<td>
<p>to determine the two value reduction value, resulting in the global score S: &quot;max&quot;, or &quot;min&quot;, or &quot;mean&quot;.</p>
</td></tr>
<tr><td><code id="rhScore_+3A_scoringmethod">scoringMethod</code></td>
<td>
<p>method to determine the module accumulation value: &quot;max-min&quot; or &quot;mean-mean&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The gpScore is a heuristic score enabling to  construct a test statistic to determine  
the significance of a present (phylotranscriptomic) hourglass pattern.
</p>


<h3>Value</h3>

<p>a numeric value representing the hourglass destruction score.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>, <code><a href="#topic+TAI">TAI</a></code>, <code><a href="#topic+TDI">TDI</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
 # read standard phylotranscriptomics data
 data(PhyloExpressionSetExample)
 data(DivergenceExpressionSetExample)

 # example PhyloExpressionSet:

 # compute the TAI profile
 TAIs &lt;- TAI(PhyloExpressionSetExample)

 # compute the global hourglass destruction score 
 # for the TAIs profile using reduction method: mean(mean-mean)
 rh_score &lt;- rhScore(age_vals = TAIs,early = 1:2,mid = 3:5,late = 6:7,
                     method = "mean",scoringMethod = "mean-mean")


 # example DivergenceExpressionSet:

 # compute the TDI profile
 TDIs &lt;- TDI(DivergenceExpressionSetExample)

 # compute the global hourglass destruction score for the TDIs profile 
 # using reduction method: mean(mean-mean)
 rh_score &lt;- rhScore(age_vals = TDIs,early = 1:2,mid = 3:5,late = 6:7,
                     method = "mean",scoringMethod = "mean-mean")
 
 
</code></pre>

<hr>
<h2 id='SelectGeneSet'>Select a Subset of Genes in an ExpressionSet</h2><span id='topic+SelectGeneSet'></span>

<h3>Description</h3>

<p>Select a subset of genes stored in the input <code>ExpressionSet</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SelectGeneSet(ExpressionSet, gene.set, use.only.map = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SelectGeneSet_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="SelectGeneSet_+3A_gene.set">gene.set</code></td>
<td>
<p>a character vector storing the gene ids for which gene expression profiles shall be visualized.</p>
</td></tr>
<tr><td><code id="SelectGeneSet_+3A_use.only.map">use.only.map</code></td>
<td>
<p>a logical value indicating whether instead of a standard <code>ExpressionSet</code> only a <code>Phylostratigraphic Map</code> or <code>Divergene Map</code> is passed to the function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function selects a subset of genes specified in <code>gene.set</code> stored in the input <code>ExpressionSet</code> and returns a subset <code>ExpressionSet</code>.
</p>
<p>This function is useful for studying the evolutionary <em>properties</em> of a subset of genes
stored in the <code>ExpressionSet</code>.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotGeneSet">PlotGeneSet</a></code>, <code><a href="#topic+PlotEnrichment">PlotEnrichment</a></code>, <code><a href="#topic+DiffGenes">DiffGenes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PhyloExpressionSetExample)

# receive a subset ExpressionSet for the fist 5 genes stored in
# the PhyloExpressionSetExample
SelectGeneSet(ExpressionSet = PhyloExpressionSetExample,
            gene.set      = PhyloExpressionSetExample[1:5, 2])
            
            
# get a gene subset using only a phylostratihraphic map
ExamplePSMap &lt;- PhyloExpressionSetExample[ , 1:2]

SelectGeneSet(ExpressionSet = ExamplePSMap,
              gene.set      = PhyloExpressionSetExample[1:5, 2],
              use.only.map  = TRUE)          
            
</code></pre>

<hr>
<h2 id='TAI'>Compute the Transcriptome Age Index (TAI)</h2><span id='topic+TAI'></span>

<h3>Description</h3>

<p>This function computes the phylogenetically based transcriptome age index (TAI) introduced by Domazet-Loso &amp; Tautz, 2010.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TAI(PhyloExpressionSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TAI_+3A_phyloexpressionset">PhyloExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The TAI measure represents the weighted arithmetic mean (expression levels as
weights for the phylostratum value) over all evolutionary age categories denoted as <em>phylostra</em>.
</p>
<p style="text-align: center;"><code class="reqn">TAI_s = \sum (e_is * ps_i) / \sum e_is</code>
</p>

<p>where TAI_s denotes the TAI value in developmental stage s, 
e_is denotes the gene expression level of gene i in stage s, 
and ps_i denotes the corresponding phylostratum of gene i, <code class="reqn">i = 1,...,N</code> and N = total number of genes.
</p>
<p>Internally the function calls the C++ function <code>cpp_TAI</code> to speed up TAI computations.
</p>


<h3>Value</h3>

<p>a numeric vector containing the TAI values for all given developmental stages.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Domazet-Loso T. and Tautz D. (2010). <em>A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns</em>. Nature (468): 815-818.
</p>
<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.
</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>


<h3>See Also</h3>

<p><code><a href="#topic+TDI">TDI</a></code>, <code><a href="#topic+PlotPattern">PlotPattern</a></code>, <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>, <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# reading a standard PhyloExpressionSet
data(PhyloExpressionSetExample)

# computing the TAI profile of a given PhyloExpressionSet object
TAIs &lt;- TAI(PhyloExpressionSetExample)



</code></pre>

<hr>
<h2 id='taxid'>Retrieve taxonomy categories from NCBI Taxonomy</h2><span id='topic+taxid'></span>

<h3>Description</h3>

<p>This function retrieves category information 
from NCBI Taxonomy and is able to filter kingdom specific taxids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxid(db.path, download = FALSE, update = FALSE, filter = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxid_+3A_db.path">db.path</code></td>
<td>
<p>path to download and store the NCBI Taxonomy <code>categories.dmp</code> file. Default is the <code>tempdir()</code>
directory.</p>
</td></tr>
<tr><td><code id="taxid_+3A_download">download</code></td>
<td>
<p>a logical value specifying whether or not the <code>categories.dmp</code> shall be downloaded (<code>download = TRUE</code>)
or whether a local version already exists on the users machine (<code>download = TRUE</code> - in this case please specify the <code>db.path</code> argument to target the local <code>categories.dmp</code> file).</p>
</td></tr>
<tr><td><code id="taxid_+3A_update">update</code></td>
<td>
<p>should the local file be updated? Please specify the <code>db.path</code> argument to target the local <code>categories.dmp</code> file.</p>
</td></tr>
<tr><td><code id="taxid_+3A_filter">filter</code></td>
<td>
<p>a character string specifying the kingdom of life for which taxids shall be returned. Options are
<code>"Archea"</code>, <code>"Bacteria"</code>, <code>"Eukaryota"</code>, <code>"Viruses"</code>, <code>"Unclassified"</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# download categories.dmp file to current working directory 
# and filter for 'Archea' taxids
Archea.taxids &lt;- taxid(db.path = getwd(), filter = "Archea", download = TRUE)

# Once the NCBI Taxonomy 'categories.dmp' file is downloaded to your machine ('download = TRUE')
# the 'taxid()' function can be proceed on the local 'categories.dmp' file
# e.g. filter for Virus taxids
Virus.taxids &lt;- taxid(db.path = getwd(), filter = "Viruses")

## End(Not run)

</code></pre>

<hr>
<h2 id='taxonomy'>Retrieving Taxonomic Information of a Query Organism</h2><span id='topic+taxonomy'></span>

<h3>Description</h3>

<p>This function takes the scientific name of a query organism
and returns selected output formats of taxonomic information for the corresponding organism.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonomy(organism, db = "ncbi", output = "classification")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxonomy_+3A_organism">organism</code></td>
<td>
<p>a character string specifying the scientific name of a query organism.</p>
</td></tr>
<tr><td><code id="taxonomy_+3A_db">db</code></td>
<td>
<p>a character string specifying the database to query, e.g. <code>db</code> = <code>"itis"</code> or <code>"ncbi"</code>.</p>
</td></tr>
<tr><td><code id="taxonomy_+3A_output">output</code></td>
<td>
<p>a character string specifying the taxonomic information that shall be returned. 
Implemented are: <code>output</code> = <code>"classification"</code>, <code>"taxid"</code>, or <code>"children"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is based on the powerful package <span class="pkg">taxize</span> and implements
the customized retrieval of taxonomic information for a query organism. 
</p>
<p>The following data bases can be selected to retrieve taxonomic information:
</p>

<ul>
<li> <p><code>db = "itis"</code> : Integrated Taxonomic Information Service
</p>
</li>
<li> <p><code>db = "ncbi"</code> : National Center for Biotechnology Information
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Scott Chamberlain and Eduard Szocs (2013). taxize - taxonomic search and retrieval in R. F1000Research,
2:191. URL: http://f1000research.com/articles/2-191/v2.
</p>
<p>Scott Chamberlain, Eduard Szocs, Carl Boettiger, Karthik Ram, Ignasi Bartomeus, and John Baumgartner
(2014) taxize: Taxonomic information from around the web. R package version 0.3.0.
https://github.com/ropensci/taxize
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# retrieving the taxonomic hierarchy of "Arabidopsis thaliana"
# from NCBI Taxonomy
taxonomy("Arabidopsis thaliana",db = "ncbi")

# the same can be applied to database : "itis"
 taxonomy("Arabidopsis thaliana",db = "itis")

# retrieving the taxonomic hierarchy of "Arabidopsis"
 taxonomy("Arabidopsis",db = "ncbi") # analogous : db = "ncbi" or "itis"

# or just "Arabidopsis"
 taxonomy("Arabidopsis",db = "ncbi")

# retrieving the taxonomy id of the query organism and in the correspondning database
# taxonomy("Arabidopsis thaliana",db = "ncbi", output = "taxid")

# the same can be applied to databases : "ncbi" and "itis"
 taxonomy("Arabidopsis thaliana",db = "ncbi", output = "taxid")
 taxonomy("Arabidopsis thaliana",db = "itis", output = "taxid")


# retrieve children taxa of the query organism stored in the correspondning database
 taxonomy("Arabidopsis",db = "ncbi", output = "children")

# the same can be applied to databases : "ncbi" and "itis"
 taxonomy("Arabidopsis thaliana",db = "ncbi", output = "children")
 taxonomy("Arabidopsis thaliana",db = "itis", output = "children")
 

## End(Not run)
</code></pre>

<hr>
<h2 id='TDI'>Compute the Transcriptome Divergence Index (TDI)</h2><span id='topic+TDI'></span>

<h3>Description</h3>

<p>This function computes the sequence distance based transcriptome divergence index (TDI) introduced by
Quint et al., 2012.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TDI(DivergenceExpressionSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TDI_+3A_divergenceexpressionset">DivergenceExpressionSet</code></td>
<td>
<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The TDI measure represents the weighted arithmetic mean (expression levels as
weights for the divergence-stratum value) over all gene divergence categories denoted as <em>divergence-strata</em>.
</p>
<p style="text-align: center;"><code class="reqn">TDI_s = \sum (e_is * ds_i) / \sum e_is</code>
</p>

<p>where TDI_s denotes the TDI value in developmental stage s, e_is denotes the gene expression level of gene i in stage s, and ds_i denotes the corresponding divergence-stratum of gene i, <code class="reqn">i = 1,...,N</code> and N = total number of genes.
</p>
<p>Internally the function is written in C++ to speed up TDI computations.
</p>


<h3>Value</h3>

<p>a numeric vector containing the TDI values for all given developmental stages.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.
</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>


<h3>See Also</h3>

<p><code><a href="#topic+TAI">TAI</a></code>,  <code><a href="#topic+PlotPattern">PlotPattern</a></code>, <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>, <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# reading a standard DivergenceExpressionSet
data(DivergenceExpressionSetExample)

# computing the TDI profile of a given DivergenceExpressionSet object
TDIs &lt;- TDI(DivergenceExpressionSetExample)


</code></pre>

<hr>
<h2 id='tf'>Transform Gene Expression Levels</h2><span id='topic+tf'></span>

<h3>Description</h3>

<p>This function transforms the gene expression set stored in an input 
PhloExpressionSet or DivergenceExpressionSet object and returns a 
PhloExpressionSet or DivergenceExpressionSet object with transformed expression levels. 
The resulting transformed PhloExpressionSet or DivergenceExpressionSet 
object can then be used for subsequent analyses based on transformed expression levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tf(ExpressionSet, FUN)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tf_+3A_expressionset">ExpressionSet</code></td>
<td>
<p>a standard PhloExpressionSet or DivergenceExpressionSet object.</p>
</td></tr>
<tr><td><code id="tf_+3A_fun">FUN</code></td>
<td>
<p>any valid function that transformes gene expression levels.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Motivated by the dicussion raised by Piasecka et al., 2013, the influence of
gene expression transformation on the global phylotranscriptomics pattern does not seem negligible.
Hence, different transformations can result in qualitatively different <code><a href="#topic+TAI">TAI</a></code> or <code><a href="#topic+TDI">TDI</a></code>
patterns.
</p>
<p>Initially, the <code><a href="#topic+TAI">TAI</a></code> and <code><a href="#topic+TDI">TDI</a></code> formulas were defined for absolute expression levels.
So using the initial <code><a href="#topic+TAI">TAI</a></code> and <code><a href="#topic+TDI">TDI</a></code> formulas with transformed expression levels
might turn out in qualitatively different patterns when compared with non-transformed expression levels,
but might also belong to a different class of models, since different valid expression level transformation functions result in different patterns.
</p>
<p>The purpose of this function is to allow the user to study the qualitative impact of different transformation functions on 
the global <code><a href="#topic+TAI">TAI</a></code> and <code><a href="#topic+TDI">TDI</a></code> pattern, or on any subsequent phylotranscriptomics analysis.
</p>
<p>The examples using the <em>PhyloExpressionSetExample</em> data set show that using common gene expression 
transformation functions: <code><a href="base.html#topic+log2">log2</a></code> (Quackenbush, 2001 and 2002), <code><a href="base.html#topic+sqrt">sqrt</a></code> (Yeung et al., 2001), 
<code><a href="MASS.html#topic+boxcox">boxcox</a></code>, or <em>inverse hyperbolic sine transformation</em>, each transformation results 
in qualitatively different patterns. 
Nevertheless, for each resulting pattern the statistical significance can be tested 
using either the <code><a href="#topic+FlatLineTest">FlatLineTest</a></code> or <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code> (Drost et al., 2014) 
to quantify the significance of interest.
</p>


<h3>Value</h3>

<p>a standard PhloExpressionSet or DivergenceExpressionSet object storing transformed gene expression levels.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Piasecka B, Lichocki P, Moretti S, et al. (2013) The hourglass and the early conservation models&ndash;co-existing patterns of developmental constraints in vertebrates. PLoS Genet. 9(4): e1003476.
</p>
<p>Quint M., Drost H.G., Gabel A., Ullrich K.K., Boenn M., Grosse I. (2012) A transcriptomic hourglass in plant embryogenesis. Nature 490: 98-101.
</p>
<p>Domazet-Loso T., Tautz D. (2010) A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns. Nature 468: 815-8.
</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012
</p>
<p>K.Y. Yeung et al.: Model-based clustering and data transformations for gene expression data. Bioinformatics 2001, 17:977-987
</p>
<p>K.Y. Yeung et al.:  Supplement to Model-based clustering and data transformations for gene expression data - Data Transformations and the Gaussian mixture assumption. Bioinformatics 2001, 17:977-987
</p>
<p>P.A.C. Hoen et al.: Deep sequencing-based expression analysis shows major advances in robustness, resolution and inter-lab portability over five microarray platforms. Nucleic Acids Research 2008, Vol. 36, No. 21
</p>
<p>H.H. Thygesen et al.: Comparing transformation methods for DNA microarray data. BMC Bioinformatics 2004, 5:77
</p>
<p>John Quackenbush: Microarray data normalization and transformation. Nature Genetics 2002, 32:496-501
</p>
<p>John Quackenbush: Computational Analysis of Microarray Data. Nature Reviews 2001, 2:418-427
</p>
<p>R. Nadon and J. Shoemaker: Statistical issues with microarrays: processing and analysis. TRENDS in Genetics 2002, Vol. 18 No. 5:265-271
</p>
<p>B.P. Durbin et al.: A variance-stabilizing transformation for gene-expression microarray data. Bioinformatics 2002, 18:S105-S110
</p>
<p>J. M. Bland et al.: Transforming data. BMJ 1996, 312:770
</p>
<p>John B. Burbidge, Lonnie Magee and A. Leslie Robb (1988) Alternative Transformations to Handle Extreme Values of the Dependent Variable. Journal of the American Statistical Association, 83(401): 123-127.
</p>
<p>G. E. P. Box and D. R. Cox (1964) An Analysis of Transformations. Journal of the Royal Statistical Society. Series B (Methodological), 26(2): 211-252.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+TAI">TAI</a></code>, <code><a href="#topic+TDI">TDI</a></code>, <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>, <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(PhyloExpressionSetExample)

# a simple example is to transform the gene expression levels
# of a given PhyloExpressionSet using a sqrt or log2 transformation

PES.sqrt &lt;- tf(PhyloExpressionSetExample, sqrt)

PES.log2 &lt;- tf(PhyloExpressionSetExample, log2)

# in case a given PhyloExpressionSet already stores gene expression levels
# that are log2 transformed and need to be re-transformed to absolute
# expression levels, to perform subsequent phylotranscriptomics analyses 
# (that are defined for absolute expression levels), one can re-transform
# a PhyloExpressionSet like this:

PES.absolute &lt;- tf(PES.log2 , function(x) 2^x)

# which should be the same as  PhyloExpressionSetExample :
head(PhyloExpressionSetExample)
head(PES.absolute)


# plotting the TAI using log2 transformed expression levels
# and performing the Flat Line Test to obtain the p-value
PlotPattern(ExpressionSet = tf(PhyloExpressionSetExample, log2), 
            type          = "l", 
            lwd           = 5, 
            TestStatistic = "FlatLineTest")

</code></pre>

<hr>
<h2 id='TPI'>Compute the Transcriptome Polymorphism Index (TPI)</h2><span id='topic+TPI'></span>

<h3>Description</h3>

<p>This function computes the Transcriptome Polymorphism Index (TPI) introduced by
Gossmann et al., 2015.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TPI(PolymorphismExpressionSet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TPI_+3A_polymorphismexpressionset">PolymorphismExpressionSet</code></td>
<td>
<p>a standard PolymorphismExpressionSet object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The TPI measure represents the weighted arithmetic mean (expression levels as
weights) for the synonymous vs non-synonymous polymorphism ratios.
</p>
<p style="text-align: center;"><code class="reqn">TPI_s = \sum (e_is * P_N/N / ((P_S + 1) / S)) / \sum e_is</code>
</p>

<p>where TPI_s denotes the TPI value in developmental stage s, e_is denotes the gene expression level of gene i in stage s, n denotes the number of genes, PN and PS denote the numbers of non-synonymous and synonymous polymorphisms, and N and S are the numbers of nonsynonymous and synonymous sites, respectively. 
</p>
<p>Internally the function is written in C++ to speed up TPI computations.
</p>


<h3>Value</h3>

<p>a numeric vector containing the TPI values for all given developmental stages.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>References</h3>

<p>Gossmann et al. (2015). <em>Transcriptomes of Plant Gametophytes Have a Higher Proportion of Rapidly Evolving and Young Genes than Sporophytes</em>. Mol Biol Evol. 33 (7): 1669-1678.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+TAI">TAI</a></code>, <code><a href="#topic+TDI">TDI</a></code>,  <code><a href="#topic+PlotSignature">PlotSignature</a></code>, <code><a href="#topic+PlotPattern">PlotPattern</a></code>, <code><a href="#topic+FlatLineTest">FlatLineTest</a></code>, <code><a href="#topic+ReductiveHourglassTest">ReductiveHourglassTest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# reading a standard PolymorphismExpressionSet
data(PolymorphismExpressionSetExample)

# computing the TPI profile of a given PolymorphismExpressionSet object
TPIs &lt;- TPI(PolymorphismExpressionSet)

## End(Not run)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
