<!DOCTYPE html><html lang="en"><head><title>Help for package NameNeedle</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {NameNeedle}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cellLineNames-data'><p>Cell Line Names</p></a></li>
<li><a href='#needles'>
<p>Needleman-Wunsch simple global alignment algorithm</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.2.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-19</td>
</tr>
<tr>
<td>Title:</td>
<td>Using Needleman-Wunsch to Match Sample Names</td>
</tr>
<tr>
<td>Author:</td>
<td>Kevin R. Coombes</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kevin R. Coombes &lt;krc@silicovore.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The Needleman-Wunsch global alignment algorithm can be
  used to find approximate matches between sample names in different
  data sets. See Wang et al. (2010) &lt;<a href="https://doi.org/10.4137%2FCIN.S5613">doi:10.4137/CIN.S5613</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Biostrings</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (== 2.0)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://oompa.r-forge.r-project.org/">http://oompa.r-forge.r-project.org/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-19 19:19:27 UTC; KRC</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-19 19:42:31 UTC</td>
</tr>
</table>
<hr>
<h2 id='cellLineNames-data'>Cell Line Names</h2><span id='topic+cellLineNames-data'></span><span id='topic+cellLineNames'></span><span id='topic+sf2Names'></span><span id='topic+rppaNames'></span><span id='topic+illuNames'></span><span id='topic+illuType'></span>

<h3>Description</h3>

<p>This dataset contains vectors of cell line names that are used to
demonstrate how to use the <span class="pkg">NameNeedle</span> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cellLineNames)
</code></pre>


<h3>Format</h3>

<p>This dataset contains four objects: three character vectors (<code>sf2Names</code>,
<code>rppaNames</code>, and <code>illuNames</code>) and one factor (<code>illuType</code>).
</p>


<h3>Details</h3>

<p>The three character vectors, <code>sf2Names</code>, <code>rppaNames</code>, and
<code>illuNames</code> contain the names of cell lines used in three different
but related experiments.  The factor, <code>illuType</code>, indicates whether
the cell lines named in the <code>illuNames</code> vector were derived from
lung cancer (with the value &quot;<code>Lung</code>&quot;) or from head and neck cancer
(&quot;<code>HNSCC</code>&quot;). 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cellLineNames)
head(rppaNames)
head(sf2Names)
head(illuNames)
summary(illuType)
</code></pre>

<hr>
<h2 id='needles'>
Needleman-Wunsch simple global alignment algorithm
</h2><span id='topic+needles'></span><span id='topic+needleScores'></span><span id='topic+defaultNeedleParams'></span>

<h3>Description</h3>

<p>The Needleman-Wunsch simple gap algorithm was one of the first methods
introduced for global alignment of biological sequences. The same algorithm
can be used to match cell line names or sample names from two related data
sets; we provide examples in the documentation, using data that accompanies
this package.
</p>
<p>While the <span class="pkg">NameNeedle</span> package can be used for biological sequence
alignment, the <span class="pkg">Biostrings</span> package from Bioconductor contains much more
sophisticated tools for that problem.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>needles(pattern, subject, params=defaultNeedleParams)
needleScores(pattern, subjects, params=defaultNeedleParams)
defaultNeedleParams
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="needles_+3A_pattern">pattern</code></td>
<td>
<p>character string to be matched</p>
</td></tr>
<tr><td><code id="needles_+3A_subject">subject</code></td>
<td>
<p>character string to be matched against</p>
</td></tr>
<tr><td><code id="needles_+3A_subjects">subjects</code></td>
<td>
<p>character vector where matches are sought</p>
</td></tr>
<tr><td><code id="needles_+3A_params">params</code></td>
<td>

<p>list containing four required components. The default values are
specified by the object <code>defaultNeedleParams</code>, which contains
the following values: </p>
<pre>   
    $ MATCH   : num 1
    $ MISMATCH: num -1
    $ GAP     : num -1
    $ GAPCHAR : chr "*"
</pre></td></tr></table>


<h3>Details</h3>

<p>The Needleman-Wunsch global alignment algorithm was one of the first
algorithms used to align <abbr><span class="acronym">DNA</span></abbr>, <abbr><span class="acronym">RNA</span></abbr>, or protein sequences.
The basic algorithm uses dynamic programming to find an optimal alignment
between two sequences, with parameters that specify penalties for
mismatches and gaps and a reward for exact matches. More elaborate
algorithms (not implemented here) make use of matrices with different
penalties depending on different kinds of mismatches. The version
implemented here is based on the Perl implementation in the first
section of Chapter 3 of the book <cite>BLAST</cite>. 
</p>


<h3>Value</h3>

<p>The <code>needles</code> function returns a list with five components:
</p>
<table role = "presentation">
<tr><td><code>score</code></td>
<td>
<p>The raw alignment score.</p>
</td></tr>
<tr><td><code>align1</code></td>
<td>
<p>The final (optimal) alignment for the <code>pattern</code>.</p>
</td></tr>
<tr><td><code>align2</code></td>
<td>
<p>The final (optimal) alignment for the <code>subject</code>.</p>
</td></tr>
<tr><td><code>sm</code></td>
<td>
<p>The score matrix.</p>
</td></tr>
<tr><td><code>dm</code></td>
<td>
<p>The backtrace matrix.</p>
</td></tr>
</table>
<p>The <code>needleScores</code> function returns a numeric vector the same
length as the <code>subjects</code> argument, with each entry equal to the
corresponding raw alignment score.
</p>


<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>,
P. Roebuck <a href="mailto:proebuck@mdanderson.org">proebuck@mdanderson.org</a>
</p>


<h3>References</h3>

<p>Needleman SB, Wunsch CD.<br />
<em>A general method applicable to the search for similarities in the
amino acid sequence of two proteins.</em><br />
J Mol Biol 1970, 48(3):443&ndash;453.
</p>
<p>Korf I, Yandell M, Bedell J.<br />
<em>BLAST.</em><br />
O'Reilly Media, 2003.
</p>
<p>Wang J, Byers LA, Yordy JS, Liu W, Shen L, Baggerly KA, Giri U, Myers
JN, Ang KK, Story MD, Girard L, Minna JD, Heymach JV, Coombes KR.<br />
<em>Blasted cell line names.</em><br />
Cancer Inform. 2010; 9:251&ndash;5.
</p>


<h3>See Also</h3>

<p>The <span class="pkg">Biostrings</span> package from Bioconductor used to contain a function
called <code>needwunQS</code> that provided a simple gap implementation of
Needleman-Wunsch, similar to the one presented here. That function has been
deprecated in favor of a more elaborate interface called
<code><a href="Biostrings.html#topic+pairwiseAlignment">pairwiseAlignment</a></code> that incorporates a variety of
other alignment methods in addition. While <code>pairwiseAlignment</code> is much
more useful for applications to biological sequences, it is serious overkill
for the application we have in mind for matching cell line or other sample
names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cellLineNames)
myParam &lt;- defaultNeedleParams
myParam$MATCH &lt;- 2
myParam$MISMATCH &lt;- -2
needles(sf2Names[2], illuNames[1], myParam)
scores &lt;- needleScores(sf2Names[6], illuNames, myParam)
w &lt;- which(scores == max(scores))
w
sf2Names[6]

needles(sf2Names[6], illuNames[w], myParam)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
