<!DOCTYPE html><html><head><title>Help for package NBBttest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {NBBttest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#NBBttest-package'>
<p>Negative Beta Binormail t-Test Package</p></a></li>
<li><a href='#annotat'>
<p>Annotation of genes within which alternative</p>
splicing occurs</a></li>
<li><a href='#betaparametab'>
<p>Estimation of parameters alpha (<code class="reqn">\alpha</code>)</p>
and beta (<code class="reqn">\beta</code>) of beta distribution</a></li>
<li><a href='#betaparametVP'>
<p>Estimation of parameters V and P in count data of RNA reads</p></a></li>
<li><a href='#betaparametw'>
<p>Estimation of proportion weights</p></a></li>
<li><a href='#betattest'>
<p>Beta t-test</p></a></li>
<li><a href='#DDX39_100'>
<p>DDX39 exon data with 100 exons</p></a></li>
<li><a href='#exondata'>
<p>Exon data for NBBplot</p></a></li>
<li><a href='#gbetattest'>
<p>Beta t-tests within groups</p></a></li>
<li><a href='#gtfa'>
<p>gtf</p></a></li>
<li><a href='#jkttcell'>
<p>Jurkat T-cell transcritomic data</p></a></li>
<li><a href='#mbetattest'>
<p>Performance of multiple beta t-test on count data</p></a></li>
<li><a href='#mtpvadjust'>
<p>Adjust p-values for multiple comparisons</p></a></li>
<li><a href='#myheatmap'>
<p>Heatmap</p></a></li>
<li><a href='#myheatmap2'>
<p>Heatmap2</p></a></li>
<li><a href='#NBBplot'>
<p>Plot differential expression of exons within a</p>
specified gene using result outputed by NBBttest.</a></li>
<li><a href='#normalized'>
<p>Normalization of data</p></a></li>
<li><a href='#oddratio'>
<p>Calculation of zeta (<code class="reqn">\zeta</code>)</p></a></li>
<li><a href='#omega'>
<p>Omega calcularion</p></a></li>
<li><a href='#pathwayHeatmap'>
<p>Heatmap for pathways found by gene ontology analysis</p></a></li>
<li><a href='#pathwy.A.up'>
<p>Pathway or function data</p></a></li>
<li><a href='#pratio'>
<p>Calculation of psi (<code class="reqn">\psi </code>)</p></a></li>
<li><a href='#prime3_PRP8_50'>
<p>3'UTR splicing data of 50 genes detected in the</p>
knockdowned PRP8 cell line</a></li>
<li><a href='#QC'>
<p>Count data quality check</p></a></li>
<li><a href='#result'>
<p>Jurkat T-cell transcritomic data with isoforms</p>
selected by NBBttest</a></li>
<li><a href='#sgRNA'>
<p>sgRNA dataset</p></a></li>
<li><a href='#simSplicing'>
<p>Simulated alternative splicing</p></a></li>
<li><a href='#simulat'>
<p>Simulation</p></a></li>
<li><a href='#skjt'>
<p>Simulated Null Transcriptomic data</p></a></li>
<li><a href='#smbetattest'>
<p>Performance of multiple beta t-test on simulated data</p></a></li>
<li><a href='#subdata'>
<p>Splite data into two subsets</p></a></li>
<li><a href='#upGAm'>
<p>Count data of group A treated breast cancer in mice</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Negative Binomial Beta t-Test</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-08</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BiocGenerics,DEXSeq</td>
</tr>
<tr>
<td>Author:</td>
<td>Yuan-De Tan</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yuan-De Tan &lt;tanyuande@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We constructed 'NBBttest' for identifying genes or RNA isoforms differentially expressed between two conditions on RNA-seq count data. Package 'NBBttest' can  perform data quality check, data normalization, differential analysis, annotation and graphic analysis. In differential analysis, 'NBBttest' can identify differentially expressed genes and differential RNA isoforms in alternative splicing sites and alternative polyadenylation sites, differential sgRNA, and differential CRISPR (clustered regularly interspaced short palindromic repeats) screening genes. In graphic analysis, 'NBBttest' provides two types of heatmaps to visualize differential expression at gene or isoform level using z-score and n-score and creates pathway heatmap. 'NBBttest' can plot differentially expressed exons within a specified gene. In addition, 'NBBttest' provides a tool for annotation of genes and exons. The methods used in 'NBBttest' were new statistical methods developed from Tan and others (2015) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0123658">doi:10.1371/journal.pone.0123658</a>&gt;. The statistical methods are robust and very powerful in identifying genes or RNA isoforms differentially expressed between two conditions in small samples.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0),gplots,gtools,stats, utils,graphics,grDevices</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Sequencing, DifferentialExpression, MultipleComparison,
SAGE, GeneExpression, Transcription,
AlternativeSplicing,Coverage, DifferentialSplicing</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-21 16:03:21 UTC; tany</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-29 05:40:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='NBBttest-package'>
Negative Beta Binormail t-Test Package
</h2><span id='topic+NBBtatest-package'></span><span id='topic+NBBtatest'></span>

<h3>Description</h3>

<p>This package consists of 20 functions, of which 
betaparametab.R, betaparametVP.R, betaparametw.R, 
gbetattest.R,betattest.R, mbetattest.R, normalize.R, 
oddratio.R, pratio.R, omega.R, simulat.R, and 
smbetattest.R are used to estimate beta, alpha, weight, 
t-statistics,rho and omega, p-value,and multiple 
test at gene level or RNA isoform level. NBBplot.R is
used to visualize count data of exons within a specified 
gene in given conditions. QC.R is used to check quality 
of data, and myheatmap.R,myheatmap2.R, and pathwayheatmap.R 
are used to show heatmap of differential expressions of 
DE genes,DE RNA isoform or pathways. Run of mbetattest.R 
would output beta t-test results including geneid or 
isoformid, gene name, the other information, t-value, 
p-value, rho, and omega(W). 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> NBBttest</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2018-01-11</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Yuan-De Tan
</p>
<p>Maintainer: Yuan-De Tan 
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Baggerly KA, Deng L, Morris JS, Aldaz CM (2003) 
Differential expression in SAGE: accounting for 
normal between-library variation. <em>Bioinformatics</em>, 
<b>19</b>: 1477-1483.
<br />
Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, 
and Joel R. Neilson(2015) A Powerful Statistical Approach 
for Large-scale Differential Transcription Analysis.
<em>Plos One</em>,10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betaparametab">betaparametab</a></code>, <code><a href="#topic+betaparametVP">betaparametVP</a></code>, 
<code><a href="#topic+betaparametw">betaparametw</a></code>, <code><a href="#topic+gbetattest">gbetattest</a></code>,
<code><a href="#topic+betattest">betattest</a></code>, <code><a href="#topic+mbetattest">mbetattest</a></code>,
<code><a href="#topic+myheatmap">myheatmap</a></code>, <code><a href="#topic+pathwayHeatmap">pathwayHeatmap</a></code>
<code><a href="#topic+myheatmap2">myheatmap2</a></code>, <code><a href="#topic+normalized">normalized</a></code>, 
<code><a href="#topic+oddratio">oddratio</a></code>, <code><a href="#topic+NBBplot">NBBplot</a></code>,
<code><a href="#topic+omega">omega</a></code>,<code><a href="#topic+pratio">pratio</a></code>, 
<code><a href="#topic+QC">QC</a></code>,<code><a href="#topic+simulat">simulat</a></code>, 
<code><a href="#topic+smbetattest">smbetattest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jkttcell) 	
res&lt;-mbetattest(X=jkttcell[1:100, ], nci=7, na=3, 
nb=3, alpha=0.05, norm="yes", side="both", 
level="isoform",padjust_methods="fdr",C=0)
</code></pre>

<hr>
<h2 id='annotat'>
Annotation of genes within which alternative 
splicing occurs
</h2><span id='topic+annotat'></span>

<h3>Description</h3>

<p>Alternative splicing is detected in any element 
of 3'UTR, 5'UTR, exons and introns within a gene 
using RNA-seq data where RNA reads are mapped to 
a reference genome. As an example for annotation, 
the RNA-seq reads derived from human samples can 
be mapped onto human genome reference (GRCh38) 
using different methods, for example, HTSeq, spladder, 
rMAT, cufflinks, etc. These methods can detect 
alternative splicing sites within genes. However, 
none of these methods does gene annotation for users. 
Our NBBttest offers a R function for annotating genes 
with exons or isoforms.      
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotat(infile, mfile, type="gene", colunmset = "NLL")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotat_+3A_infile">infile</code></td>
<td>

<p>input data file with ENSGid column for annotation
</p>
</td></tr>
<tr><td><code id="annotat_+3A_mfile">mfile</code></td>
<td>

<p>reference genome file with ENSG id column.
</p>
</td></tr>
<tr><td><code id="annotat_+3A_type">type</code></td>
<td>

<p>has three options: &quot;gene&quot;, &quot;isoform&quot; or &quot;&quot;isof&quot; and 
&quot;exon&quot;, see details. The default is &quot;gene&quot;.
</p>
</td></tr>
<tr><td><code id="annotat_+3A_colunmset">colunmset</code></td>
<td>

<p>EGNSid column set. If the RNA count data are made by 
using HTSeq and DEXSeq annotation file, then some of 
genes have many different ENSGids. For example, ENSG00000285476+ENSG00000182230+ENSG00000251623 has 
three ENSG ids ENSG00000285476,ENSG00000182230, and 
ENSG00000251623 that share one gene, so it is splited 
into three columns (2,3,4)in excel. The default is 
2(column 2). 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If type = &quot;gene&quot;, then count data of RNA reads are 
obtained at gene level, annotation would be executed 
at gene level or if type = &quot;isof&quot; or &quot;isoform&quot;, then 
RNA reads were mapped onto elements (for example, 3'UTR,
5'UTR, exon or cassette, intron) within genes and  
annotation would be executed at isoform level or if 
type = &quot;exon&quot;, then RNA count data were obtained by 
mapping RNA reads onto exome by DEXseq and annotation 
would be done at exon level defined by DEXSeq. Note that 
GRCh38 is too big so it was removed from data. User may 
request to get it from yuande/github. 
</p>


<h3>Value</h3>

<p>return original data with an additional column for gene. 
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DDX39_100)
data(gtfa)
DDX39_30&lt;-annotat(infile=DDX39_100[1:30,],mfile=gtfa,type="gene")
</code></pre>

<hr>
<h2 id='betaparametab'>
Estimation of parameters alpha (<code class="reqn">\alpha</code>) 
and beta (<code class="reqn">\beta</code>) of beta distribution
</h2><span id='topic+betaparametab'></span>

<h3>Description</h3>

<p>Parameters alpha(a) and beta (b) in betat distribution 
are estimated by using an iteration algorithm. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betaparametab(xn, w, P, V)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betaparametab_+3A_xn">xn</code></td>
<td>

<p>column vector, a set of library sizes.
</p>
</td></tr>
<tr><td><code id="betaparametab_+3A_w">w</code></td>
<td>

<p>column vector, a set of weights.
</p>
</td></tr>
<tr><td><code id="betaparametab_+3A_p">P</code></td>
<td>

<p>proportion of counts of a gene or an isoform.
</p>
</td></tr>
<tr><td><code id="betaparametab_+3A_v">V</code></td>
<td>

<p>variance of proportions of counts of a gene or an isoform over m 
replicate libraries in a condition.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return parameters a and b.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Baggerly KA, Deng L, Morris JS, Aldaz CM (2003) 
Differential expression in SAGE:  accounting for 
normal between-library variation. 
<em>Bioinformatics</em> <b>19</b>: 1477-1483.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betaparametVP">betaparametVP</a></code>, <code><a href="#topic+betaparametw">betaparametw</a></code>	
</p>


<h3>Examples</h3>

<pre><code class='language-R'>XX&lt;-c(2000,2000,2000)
p&lt;-0.15
V=0.004
w&lt;-c(0.3,0.3,0.3)
betaparametab(xn=XX,w=w,P=p,V=V)
</code></pre>

<hr>
<h2 id='betaparametVP'>
Estimation of parameters V and P in count data of RNA reads
</h2><span id='topic+betaparametVP'></span>

<h3>Description</h3>

<p>This function is used to estimate parameters P and V 
by optimalizing estimates of parameters: alpha and beta. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betaparametVP(X, NX)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betaparametVP_+3A_x">X</code></td>
<td>

<p>count dataset derived from <code>m</code> replicate libraries 
in one condition.
</p>
</td></tr>
<tr><td><code id="betaparametVP_+3A_nx">NX</code></td>
<td>

<p>vector of m library sizes. Library size is sum of 
counts over the whole library. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Count data of <em>RNA</em> sequence reads are assumed 
to follow binomial distribution with parameters (<code>P</code>) 
and (<code>n</code>) or negative binomial distribution with 
parameters (<code>P</code>) and (<code>r</code>) , while <code>P</code> 
is frequency of a gene or an isoform in RNA sequence 
population and assumed to follow beta distribution with 
parameters <code>alpha</code> (<code>a</code>) and <code>beta</code>(<code>b</code>). 
Parameters  <code>P and V</code> are estimated by using optimal 
estimation of parameters a and b. The optimal method is an 
iteration algorithm drived by weighting proportion of gene 
or isoform in each replicate library. This is a large-scale 
method for estimating these parameters. Estimation of 
parameters <code>P and V</code> is core of the multiple beta 
t-test method because <code>P</code> and <code>V</code> will be 
used to calculate t-value. 
</p>


<h3>Value</h3>

<p>return a list:
</p>
<table>
<tr><td><code>P</code></td>
<td>
<p>N proportions estimated.</p>
</td></tr>
<tr><td><code>V</code></td>
<td>
<p>N variances estimated.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>betaparametVP requres functions betaparametab and betaparametw.
</p>


<h3>Author(s)</h3>

<p>Yuan-DE Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Baggerly KA, Deng L, Morris JS, Aldaz CM (2003) 
Differential expression in SAGE: accounting for normal 
between-library variation. <em>Bioinformatics</em>, 
<b>19</b>: 1477-1483.
<br />
Yuan-De Tan, Anita M. Chandler, Arindam Chaudhury, 
and Joel R. Neilson(2015) A Powerful Statistical 
Approach for Large-scale Differential Transcription Analysis.
<em>Plos One</em>,10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betaparametab">betaparametab</a></code>, <code><a href="#topic+betaparametw">betaparametw</a></code>	
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jkttcell) 
X&lt;-jkttcell[1:100,]
na&lt;-3
nb&lt;-3
cn&lt;-length(X[1,])
rn&lt;-length(X[,1])
XC&lt;-X[,1:(cn-na-nb)]
XX&lt;-X[,(cn-na-nb+1):cn]
n&lt;-na+nb
XA&lt;-XX[,1:na]
SA&lt;-apply(XA,2,sum)
PA&lt;-betaparametVP(XA,SA)
</code></pre>

<hr>
<h2 id='betaparametw'>
Estimation of proportion weights
</h2><span id='topic+betaparametw'></span>

<h3>Description</h3>

<p>Function betaparametw is used to calculate weights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betaparametw(xn, a, b)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betaparametw_+3A_xn">xn</code></td>
<td>

<p>a vector of <code>m</code> library sizes. Library size 
is sum of counts over the whole library. 
</p>
</td></tr>
<tr><td><code id="betaparametw_+3A_a">a</code></td>
<td>

<p>parameter alpha(<code class="reqn">\alpha</code>) in beta distribution 
derived from output of function betaparametab.
</p>
</td></tr>
<tr><td><code id="betaparametw_+3A_b">b</code></td>
<td>

<p>parameter beta (<code class="reqn">\beta</code>)in beta distribution 
derived from output of function betaparametab.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>alpha and beta (<code class="reqn">\alpha</code>,<code class="reqn">\beta</code>) are 
used to calculate weight. Then weight is in turn 
used to correct bias of estimation of alpha and 
beta in betaparametab function. 
</p>


<h3>Value</h3>

<p>return weight(W).
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Baggerly KA, Deng L, Morris JS, Aldaz CM (2003) 
Differential expression in SAGE: accounting for 
normal between-library variation. 
<em>Bioinformatics</em>, <b>19</b>: 1477-1483.
<br />
Yuan-De Tan, Anita M. Chandler, Arindam Chaudhury, 
and Joel R. Neilson(2015) A Powerful Statistical 
Approach for Large-scale Differential Transcription Analysis. 
<em>Plos One</em>. 2015 DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betaparametab">betaparametab</a></code> and <code><a href="#topic+betaparametVP">betaparametVP</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>XX&lt;-c(2000,2000,2000)
a&lt;-1.1458
b&lt;-6.4932
betaparametw(xn=XX,a=a,b=b)
</code></pre>

<hr>
<h2 id='betattest'>
Beta t-test
</h2><span id='topic+betattest'></span>

<h3>Description</h3>

<p>Beta t-test and degree of freedom for each gene 
or isoform are calculated in this function. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betattest(X, na, nb, NX=100, level)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betattest_+3A_x">X</code></td>
<td>

<p>count data of RNA sequence reads containing 
N genes (or isoforms).
</p>
</td></tr>
<tr><td><code id="betattest_+3A_na">na</code></td>
<td>
                     
<p>number of replicate libraries in condition A.
</p>
</td></tr>
<tr><td><code id="betattest_+3A_nb">nb</code></td>
<td>

<p>number of replicate libraries in condition B.
</p>
</td></tr>
<tr><td><code id="betattest_+3A_nx">NX</code></td>
<td>

<p>numeric value. It is used at level=&quot;isoform&quot;. 
NX=100 is default but does not used at any level.
</p>
</td></tr>
<tr><td><code id="betattest_+3A_level">level</code></td>
<td>

<p>string, has three options: &quot;isoform&quot; or &quot;sgRNA&quot;
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In beta t-test, </p>
<p style="text-align: center;"><code class="reqn">t=\frac{(P_A-P_B)}{\sqrt(V_A+V_B)}</code>
</p>
 
<p>where <code class="reqn">P_A</code> and <code class="reqn">P_B</code> are proportions of a gene or 
an isoform in conditions A and B, <code class="reqn">V_A</code> and <code class="reqn">V_B</code> 
are variances of this gene or isoform in conditions A and B,  
respectively. They are output of betaparametVP.
</p>


<h3>Value</h3>

<p>return two lists:
</p>
<table>
<tr><td><code>t</code></td>
<td>
<p>t-value list.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>df list. df is degree of freedom.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>In our method, pooled standard error &gt; 0 in any case, 
so the t-statistics alway has definition.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Baggerly KA, Deng L, Morris JS, Aldaz CM (2003) 
Differential expression in SAGE: accounting for normal 
between-library variation. 
<em>Bioinformatics</em>, <b>19</b>: 1477-1483.
<br />
Yuan-De Tan, Anita M. Chandler, Arindam Chaudhury, and 
Joel R. Neilson(2015) A Powerful Statistical Approach for 
Large-scale Differential Transcription Analysis. 
<em>Plos One</em>. 2015 DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pratio">pratio</a></code>, <code><a href="#topic+oddratio">oddratio</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jkttcell) 
X&lt;-jkttcell[1:100,]
na&lt;-3
nb&lt;-3
cn&lt;-ncol(X)
rn&lt;-nrow(X)
XC&lt;-X[,1:(cn-na-nb)]
XX&lt;-X[,(cn-na-nb+1):cn]
betattest&lt;-betattest(X=XX, na=3,nb=3, level="isoform")
</code></pre>

<hr>
<h2 id='DDX39_100'>
DDX39 exon data with 100 exons  
</h2><span id='topic+DDX39_100'></span>

<h3>Description</h3>

<p>DDX39_100 dataset is an example for implementing annotation of genes within there are some exons with gtf.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("DDX39_100")</code></pre>


<h3>Format</h3>

<p>A data frame with 100 observations on the following 14 variables.
</p>

<dl>
<dt><code>X</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>gene_id</code></dt><dd><p>a factor with ENSG levels</p>
</dd>
<dt><code>CTL_1_S3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CTL_2_S7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CTL_3_S11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DDX39_1_S1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DDX39_2_S5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DDX3_S9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>tvalue</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rho</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>df</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>pvalue</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>adjp</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>w</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>DDX39_100 is dataset that was RNA-seq data from human cell breast cells with knockout DDX39 gene. DDX39 RNA-seq reads were mapped to human reference annotated with GRCh38 by HTseq with DEXseq annotation and generated count data of RNA reads of exons within genes. DDX39_100 is output of NBBttest where X is row order column; gene_id is ENSG ID; CTL is control cells with replicates 1, 2,and 3; DDX39 is treatment cells with knockout gene DDX39; tvalue is NBBttest t-statistic; rho is exon-wise variable of NBBttest; df is degree of freedom; pvalue is p-value for t-statistics and w is threshold of rho.    
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DDX39_100)
</code></pre>

<hr>
<h2 id='exondata'>
Exon data for NBBplot
</h2><span id='topic+exondata'></span>

<h3>Description</h3>

<p>This dataset is an example for NBBplot to show 
differential exons within genes between two 
conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("exondata")</code></pre>


<h3>Format</h3>

<p>A data frame with 16 observations on the 
following 22 variables.
</p>

<dl>
<dt><code>ID</code></dt><dd><p>a factor with levels 
<code>ENSMUSG00000015597:E1</code> 
<code>ENSMUSG00000015597:E10</code> 
<code>ENSMUSG00000015597:E2</code> 
<code>ENSMUSG00000015597:E3</code> 
<code>ENSMUSG00000015597:E4</code> 
<code>ENSMUSG00000015597:E5</code> 
<code>ENSMUSG00000015597:E6</code> 
<code>ENSMUSG00000015597:E7</code> 
<code>ENSMUSG00000015597:E8</code> 
<code>ENSMUSG00000015597:E9</code> 
<code>ENSMUSG00000079547:E1</code> 
<code>ENSMUSG00000079547:E2</code> 
<code>ENSMUSG00000079547:E3</code> 
<code>ENSMUSG00000079547:E4</code> 
<code>ENSMUSG00000079547:E5</code> 
<code>ENSMUSG00000079547:E6</code></p>
</dd>
<dt><code>chr</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>element</code></dt><dd><p>a factor with levels <code>exon</code></p>
</dd>
<dt><code>start</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>end</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>strand</code></dt><dd><p>a factor with levels <code>+</code></p>
</dd>
<dt><code>gene_id</code></dt><dd><p>a factor with levels 
<code>ENSMUSG00000015597</code> 
<code>ENSMUSG00000079547</code></p>
</dd>
<dt><code>exon</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>gene</code></dt><dd><p>a factor with levels 
<code>H2-DMb1</code> 
<code>Zfp318</code></p>
</dd>
<dt><code>WT_28</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>WT_43</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A25_f0</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>cKO_30</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A26_cKO</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A6_f0</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>tv</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rho</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>pvalue</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>w</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>order_number</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FDR0.05</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>significance</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>This dataset is an object of NBBttest and annoted 
with a R function. It contains columns start and 
end, gene,isoform, tv and pvalue, data, significance 
that NBBplut uses. WT and A25 are count data from 
wild type cell lines 28,43 and fo and A and cKO are 
also count data from knockout a gene, tv is t-statistic, 
rho is exon-wise variable and w is threshold of rho. 
Column &quot;significance&quot; is 1 if pvalue &lt; FDR0.05, 0, 
otherwise. Gene Zfp318 has 10 exons and H2-DMb1 has 
6 exons.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exondata)
</code></pre>

<hr>
<h2 id='gbetattest'>
Beta t-tests within groups
</h2><span id='topic+gbetattest'></span>

<h3>Description</h3>

<p>Beta t-tests are conducted within groups,genes,or libraries. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gbetattest(xx, W, nci, na, nb, level, padjust_methods,C=1.222, side)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gbetattest_+3A_xx">xx</code></td>
<td>

<p>a datasheet consisting of <code>n</code> columns and <code>m</code>
rows. Columns contain information and count data columns 
<code>n</code> must be 1 or more and <code>m</code> must be over 100.
</p>
</td></tr>
<tr><td><code id="gbetattest_+3A_w">W</code></td>
<td>

<p>numeric value. It is omega estimated from null 
simulation.
</p>
</td></tr>
<tr><td><code id="gbetattest_+3A_nci">nci</code></td>
<td>

<p>int numeric value indicating number of  
information columns.
</p>
</td></tr>
<tr><td><code id="gbetattest_+3A_na">na</code></td>
<td>

<p>int numeric value indicating number of replicates 
in condition a.
</p>
</td></tr>
<tr><td><code id="gbetattest_+3A_nb">nb</code></td>
<td>

<p>int numeric value indicating number of replicates 
in condition b.
</p>
</td></tr>
<tr><td><code id="gbetattest_+3A_level">level</code></td>
<td>

<p>string value. It has 6 options: &quot;isoform&quot;, &quot;sgRNA&quot;, 
&quot;RNA&quot;, &quot;polyA.gene&quot;, &quot;CRISPR.gene&quot; and &quot;splicing.gene&quot;.
</p>
</td></tr>
<tr><td><code id="gbetattest_+3A_padjust_methods">padjust_methods</code></td>
<td>

<p>padjust.methods can choose one of &quot;holm&quot;, &quot;hochberg&quot;, 
&quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;,  &quot;fdr&quot;, TX, and &quot;none&quot; 
where &quot;fdr&quot; 	= &quot;BH&quot;, &quot;TX&quot; is Tan and Xu's method (2015) 
with C=1.222 for adjusting p-value.
</p>
</td></tr>
<tr><td><code id="gbetattest_+3A_c">C</code></td>
<td>

<p>float numeric value for specifying a multiple procedure. 
C=0 tells mbetattest to perform single tests, C=1.222 
tells mbetattest to perform BH correction of pvalues, 
C&gt;1000 tells mbetattest to perform Bonferroni correction 
of pvalues.
</p>
</td></tr>
<tr><td><code id="gbetattest_+3A_side">side</code></td>
<td>

<p>string value for specifying one-tail test or two-tail test: 
side=&quot;up&quot; for left-tail test, side=&quot;down&quot; for right-tail 
test and side=&quot;both&quot; for two-tail tests. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Beta t-test will be conducted within a specified group or at
a specified level. If level=&quot;RNA&quot;, then beta t-tests will be 
conducted within a whole library or the whole data. If level=
&quot;isoform&quot;, then data will be sparated in two parts: 
single-isoform and multi-isoform datasets. Single-isoform RNA
indicates that there is only one RNA isform within a gene, 
while multi-isoform RNAs indicate that there are at least two 
RNA isoformswithin a gene. For single-isoforms, data are as a 
group and beta t-tests will be performed in the group. 
For the multi-isoforms, t-test will be performed within genes. 
If level=&quot;polyA.gene&quot; or &quot;CRISPR.gene&quot;, then t-test will be 
performed at gene level. If level=&quot;splicing.gene&quot;, then t-values 
and p-values will be selected from gene groups with the least 
p-values.
</p>


<h3>Value</h3>

<p>return a list containing dataset, t-values, corrected p-values, 
rhos and w.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Baggerly KA, Deng L, Morris JS, Aldaz CM (2003) 
Differential expression in SAGE: accounting for 
normal between-library variation. 
<br />
Benjamini, Y., and Hochberg, Y. (1995). Controlling the false discovery 
rate: a practical and powerful approach to multiple testing. 
<em>Journal of the Royal Statistical Society Series</em>, <b>B</b>, <b>57</b>5, 289-300. 
<a href="doi:10.1111/j.2517-6161.1995.tb02031.x">doi:10.1111/j.2517-6161.1995.tb02031.x</a>,
<a href="https://www.jstor.org/stable/2346101">https://www.jstor.org/stable/2346101</a>.	
<br />
Benjamini, Y., and Yekutieli, D. (2001). The control of the false 
discovery rate in multiple testing under dependency. 
<em>Annals of Statistics</em>, <b>29</b>, 1165 -1188. 
<a href="doi:10.1214/aos/1013699998">doi:10.1214/aos/1013699998</a>.
<br />
Tan YD, Xu H. A general method for accurate estimation of false discovery 
rates in identification of differentially expressed genes. 
<em>Bioinformatics</em>. 2014 Jul 15;30(14):2018-25. 
<a href="doi:10.1093/bioinformatics/btu124">doi:10.1093/bioinformatics/btu124</a>. Epub 2014 Mar 14. PMID: 24632499.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+betattest">betattest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jkttcell)
colnames(jkttcell)[3]&lt;-"Gene"
res.isfo&lt;-gbetattest(xx=jkttcell[1:100,], W=1, nci=7,
na=3, nb=3, level="isoform", padjust_methods="fdr",C=0,side="both")
	
</code></pre>

<hr>
<h2 id='gtfa'>
gtf
</h2><span id='topic+gtfa'></span>

<h3>Description</h3>

<p>gtf is annotation file for genes and exons
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("gtfa")</code></pre>


<h3>Format</h3>

<p>A data frame with 1461 observations on the following 8 variables.
</p>

<dl>
<dt><code>V1</code></dt><dd><p>a factor with levels</p>
</dd>
<dt><code>V2</code></dt><dd><p>a factor with levels</p>
</dd>
<dt><code>V3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V5</code></dt><dd><p>a factor with levels</p>
</dd>
<dt><code>V6</code></dt><dd><p>a factor with levels </p>
</dd>
<dt><code>V7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V8</code></dt><dd><p>a factor with levels</p>
</dd>
</dl>



<h3>Details</h3>

<p>V1:chr,V2:exon,V3:start position,V4:end position,V5:strand,V6:gene id,V7: exon,V8:gene
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gtfa)
</code></pre>

<hr>
<h2 id='jkttcell'>
Jurkat T-cell transcritomic data
</h2><span id='topic+jkttcell'></span>

<h3>Description</h3>

<p>The data are transcriptomic count data of 
<em>RNA</em> reads generated by next 
generation sequencing from Jurkat T-cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("jkttcell")</code></pre>


<h3>Format</h3>

<p>A data frame with 13409 observations on the 
following 13 variables.
</p>

<dl>
<dt><code>tagid</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>geneid</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>name</code></dt><dd><p>: a string vector</p>
</dd>
<dt><code>chr</code></dt><dd><p>: a string vector</p>
</dd> 
<dt><code>strand</code></dt><dd><p>: a charactor vector</p>
</dd>
<dt><code>pos</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>anno</code></dt><dd><p>: a string vector</p>
</dd>
<dt><code>Jurk.NS.A</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.NS.B</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.NS.C</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.48h.A</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.48h.B</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.48h.C</code></dt><dd><p>: a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>The data are count data generated by next generation 
sequencing from Jurkat T-cells. The T-cells were 
treated by resting and stimulating with <em>CD3/CD28</em> 
for 48 hours. The data have 7 columns for the information 
of <em>poly(A)</em> sites: tagid, geneid, gene name, 
chromosome, strand, <em>poly(A)</em> site position, 
<em>poly(A)</em> site annotation and 6 columns 
for count data: <code>Jurk.NS.A</code>,	<code>Jurk.NS.B</code>,	
<code>Jurk.NS.C</code>,	<code>Jurk.48h.A</code>,	<code>Jurk.48h.B</code>,	
<code>Jurk.48h.C</code> where NS means normal state or no 
sitmulation and 48h means 48 hours after <em>CD3/CD28</em> 
stimulatuin of T-cells. 13409 <em>RNA</em> isoforms 
were detected to have alternative 
<em>poly(A)</em> sites.
</p>


<h3>Value</h3>

<p>ID, information, count data of RNA reads</p>


<h3>Source</h3>

<p>Real transcriptomic count data
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, 
and Joel R. Neilson(2015) A Powerful Statistical 
Approach for Large-scale Differential Transcription 
Analysis. <em>Plos One</em>. 
DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jkttcell)
</code></pre>

<hr>
<h2 id='mbetattest'>
Performance of multiple beta t-test on count data
</h2><span id='topic+mbetattest'></span>

<h3>Description</h3>

<p>This function is used to peform multiple beta t-test 
method on real count data. The result lists &quot;geneid&quot; or 
&quot;isoformid&quot;, gene name, the other information, t-value, 
p-value,  rho, and w. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mbetattest(X, nci, na, nb, alpha=0.05, norm="no", 
side="both", level="sgRNA",padjust_methods,C=1.222)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mbetattest_+3A_x">X</code></td>
<td>

<p>count data of RNA sequence reads with na replicates 
in condition A and nb replicates in condition B.
</p>
</td></tr>
<tr><td><code id="mbetattest_+3A_nci">nci</code></td>
<td>

<p>nonnegative int value: number of columns for data information, 
such as geneID, isoformID, gene name etc.	
</p>
</td></tr>
<tr><td><code id="mbetattest_+3A_na">na</code></td>
<td>

<p>nonnegative int value: number of replicate libraries in condition A.
</p>
</td></tr>
<tr><td><code id="mbetattest_+3A_nb">nb</code></td>
<td>

<p>int numeric value: number of replicate libraries in condition B.
</p>
</td></tr>
<tr><td><code id="mbetattest_+3A_alpha">alpha</code></td>
<td>

<p>float numeric value, a probabilistic threshold. The value must 
be in [0,1]. User can set alpha=0.05 or 0.01 or the other values. 
Defalt value is 0.05
</p>
</td></tr>
<tr><td><code id="mbetattest_+3A_norm">norm</code></td>
<td>

<p>logistic value:&quot;yes&quot; or &quot;no&quot;. If norm=&quot;yes&quot;, the count 
data will be normalized and mbetattest will work on the 
normalized data, if norm=&quot;no&quot;, then mbetattest will 
work on the unnormalized data.
</p>
</td></tr>
<tr><td><code id="mbetattest_+3A_side">side</code></td>
<td>

<p>string for specifying tail(s) of t-distribution. 
If side=&quot;up&quot;, then p-value is given with t-test in the 
left tail. If side=&quot;down&quot;, p-value is given with 
t-test in right tail. If side =&quot;both&quot;, p-value is given 
with t-test in both sides.
</p>
</td></tr>
<tr><td><code id="mbetattest_+3A_level">level</code></td>
<td>

<p>string for specifying which level mbetattest work on. 
In the current version, level has 6 options: &quot;isoform&quot;, 
&quot;sgRNA&quot;, &quot;RNA&quot;, &quot;splicing.gene&quot;,&quot;polyA.gene&quot;, and 
&quot;CRISPR.gene&quot;. 
</p>
</td></tr>
<tr><td><code id="mbetattest_+3A_padjust_methods">padjust_methods</code></td>
<td>

<p>string for specifying a method for a multiple procedure. 
padjust_methods can choose one of &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, 
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;,  &quot;fdr&quot;, TX, and &quot;none&quot; where &quot;fdr&quot; 	
= &quot;BH&quot;, &quot;TX&quot; is Tan and Xu's method (2015) with C=1.222 
for adjusting p-value.
</p>
</td></tr>
<tr><td><code id="mbetattest_+3A_c">C</code></td>
<td>

<p>float numeric value for specifying a multiple procedure. 
C=0 tells mbetattest to perform single tests, C=1.222 
tells mbetattest to perform BH correction of pvalues, 
C&gt;1000 tells mbetattest to perform Bonferroni correction 
of pvalues. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>see MBttest2-manual. 
</p>


<h3>Value</h3>

<p>return a data and result list: data columns, t-values, rho. 
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Baggerly KA, Deng L, Morris JS, Aldaz CM (2003) 
Differential expression in SAGE:  accounting for 
normal between-library variation. 
<em>Bioinformatics</em> <b>19</b>: 1477-1483.	
<br />
Benjamini, Y., and Hochberg, Y. (1995). Controlling the false discovery 
rate: a practical and powerful approach to multiple testing. 
<em>Journal of the Royal Statistical Society Series</em>, <b>B</b>, <b>57</b>5, 289-300. 
<a href="doi:10.1111/j.2517-6161.1995.tb02031.x">doi:10.1111/j.2517-6161.1995.tb02031.x</a>,
<a href="https://www.jstor.org/stable/2346101">https://www.jstor.org/stable/2346101</a>.	
<br />
Benjamini, Y., and Yekutieli, D. (2001). The control of the false 
discovery rate in multiple testing under dependency. 
<em>Annals of Statistics</em>, <b>29</b>, 1165 -1188. 
<a href="doi:10.1214/aos/1013699998">doi:10.1214/aos/1013699998</a>.
<br />
Tan YD, Xu H. A general method for accurate estimation of false discovery 
rates in identification of differentially expressed genes. 
<em>Bioinformatics</em>. 2014 Jul 15;30(14):2018-25. 
<a href="doi:10.1093/bioinformatics/btu124">doi:10.1093/bioinformatics/btu124</a>. Epub 2014 Mar 14. PMID: 24632499.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+smbetattest">smbetattest</a></code>, <code><a href="#topic+mtpvadjust">mtpvadjust</a></code>, 
<code><a href="#topic+normalized">normalized</a></code>,<code><a href="#topic+omega">omega</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jkttcell) 
res&lt;-mbetattest(X=jkttcell[1:70, ], nci=7, na=3, 
nb=3, alpha=0.05, norm="yes", side="both", 
level="isoform",padjust_methods="fdr",C=0)
</code></pre>

<hr>
<h2 id='mtpvadjust'>
Adjust p-values for multiple comparisons
</h2><span id='topic+mtpvadjust'></span>

<h3>Description</h3>

<p>Given a set of p-values and chosen a C-value, 
returns a set of adjusted p-values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mtpvadjust(pv, C)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mtpvadjust_+3A_pv">pv</code></td>
<td>

<p>numeric vector of p-values (possibly with NAs). 
Any other R object is coerced by as.numeric.
</p>
</td></tr>
<tr><td><code id="mtpvadjust_+3A_c">C</code></td>
<td>

<p>real numeric value for specifying a multiple procedure. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>C=0 indicates that p-values are not adjusted,  C=1.22 
indicates that p-values are adjusted with Benjamini and 
Hochberg (1995) (&quot;BH&quot;). The adjusted p-values are 
called &quot;fdr&quot;. When <code class="reqn">C\ge 1000</code>,p-values are adjusted 
with the Bonferroni method. <code class="reqn">C&lt;1.22</code> indicates that 
p-values are adjusted by a relaxed BH method while 
C&gt;1.22, p-values are adjusted by a more strict BH method.
</p>


<h3>Value</h3>

<p>A numeric vector of corrected p-values (of the same 
length as p, with names copied from p)
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<br />
<a href="mailto:tanyuande@gmaail.com">tanyuande@gmaail.com</a>
</p>


<h3>References</h3>

<p>Benjamini, Y., and Hochberg, Y. (1995). Controlling the 
false discovery rate: a practical and powerful approach to 
multiple testing. Journal of the Royal Statistical Society 
Series B, 57, 289 - 300. 
<br />
Yuan-De Tan, Hongyan Xu; A general method for 
accurate estimation of false discovery rates in identification 
of differentially expressed genes, Bioinformatics, 
Volume 30, Issue 14, 15 July 2014, Pages 2018 - 2025.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+p.adjust">p.adjust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
x &lt;- rnorm(50, mean = c(rep(0, 25), rep(3, 25)))
p &lt;- 2*pnorm(sort(-abs(x)))
p.1.22&lt;-mtpvadjust(pv=p, C=1.22)
</code></pre>

<hr>
<h2 id='myheatmap'>
Heatmap 
</h2><span id='topic+myheatmap'></span>

<h3>Description</h3>

<p>This function is used to display heatmap of differential 
expressions of genes or isoforms  detected by NBBttest 
in the real count data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>myheatmap(dat, IDcol, nci, r, r1, r2, 
colrs="greenred", rwcex=2.0,clcex=2.0,x=10, tree="both",
method="euclidean", ky=1.5, rowBarColor=NULL,  
colBarColor=NULL,  labrow="yes", labcol="yes", rsort="yes",
adjrow=c(0.3, 0.0 ), adjcol = c(1, 1) , rwangle=30,
clangle=30,  maptitle)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="myheatmap_+3A_dat">dat</code></td>
<td>

<p>is NBBttest object that is outputted by mbetattest, 
includes information columns, data columns, t-value 
column, pvalue column,selection column.
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_idcol">IDcol</code></td>
<td>

<p>nonnegative int value, indicating which column is 
ID column where IDs will be shown in heatmap rows,
is required.
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_nci">nci</code></td>
<td>

<p>nonnegative int value,indicating column number of 
informaction (gene or isoform annotation) columns 
in the data, is required. 	
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_r">r</code></td>
<td>

<p>nonnegative int value, indicating column number of count 
data including empty columns if there are multiple 
datasets with the same row names.
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_r1">r1</code></td>
<td>

<p>nonnegative int value, indicating replicate number 
in condition 1.
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_r2">r2</code></td>
<td>

<p>nonnegative int value, indicating replicate number 
in cocndition 2.
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_colrs">colrs</code></td>
<td>

<p>heatmap colors. User has 8 options: &quot;redgreen&quot;, 
&quot;greenred&quot;, &quot;redblue&quot;, &quot;bluered&quot;, &quot;cm.colors&quot;,
&quot;terrain.colors&quot;,&quot;topo.colors&quot;, and&quot;heat.colors&quot;. 
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_rwcex">rwcex</code></td>
<td>

<p>numeric argument: nonnegative value, used as 
cex.axis in for the row axis labeling.Default=1.8.
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_clcex">clcex</code></td>
<td>

<p>numeric argument: nonnegative value, used as 
cex.axis for the column axis labeling. Default=1.8.	
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_x">x</code></td>
<td>

<p>numeric argument: nonnegative value, used as 
argument of cm.colors(x), terrain.colors(x) and 
topo.colors(x), the default value is 10.	
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_tree">tree</code></td>
<td>

<p>tree(s) drawn on row or colum or both. User has four 
options: &quot;both&quot; for drawing trees on both row and 
column,&quot;row&quot; for drawing tree only on row,&quot;column&quot; 
for drawing tree only on column, and &quot;none&quot; for no 
tree specified on rows and columns. Default is &quot;both&quot;.
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_method">method</code></td>
<td>

<p>method to be chosen to calculate distance between 
columns or rows. It has four options: &quot;euclidean&quot;, 
&quot;pearson&quot;,&quot;spearman&quot; and &quot;kendall&quot;. The latter three 
are d=1-cc where cc is correlation coefficients. 
Default is &quot;euclidean&quot;. 	
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_ky">ky</code></td>
<td>

<p>numeric argument: nonnegative value is used to 
determine key size. The defalut =1.5	
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_rowbarcolor">rowBarColor</code></td>
<td>
<p>(optional) character vector for 
RowSideColors and colRow. Length of rowBarColor 
equals to gene or isoform number or row number.  
rowBarColor contains the color names of classes or 
typpes or groups of row names and show row color 
side bars and color row names. If row names, for 
example, genes in row are not classified, then we 
suggest user use its default value: rowBarColor 
is NULL.
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_colbarcolor">colBarColor</code></td>
<td>

<p>(optional) character vector for colSideColors and 
colCol. Length of colBarColor equals to sample number.  
colBarColor contains the color names of classes or 
typpes or groups of  coloumn names and show column 
color side bars and color column names. If column 
names, for example, samples in column are just two 
types, then we suggest user use its default value: 
colBarColor=NULL.	
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_labrow">labrow</code></td>
<td>

<p>string, logistic value. Rows on heatmap are labeled 
with genes or targets if labrow=&quot;yes&quot;, otherwise, the 
rows are not labeled. 
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_labcol">labcol</code></td>
<td>

<p>string, logistical value. columns on heatmap are labeled 
with samples or treatments if labcol=&quot;yes&quot;, otherwise, 
the columns are not labeled. 
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_rsort">rsort</code></td>
<td>

<p>logistical value. If choose tree=&quot;both&quot; or &quot;row&quot;, then 
rsort does not work. However, if tree =&quot;none&quot; or &quot;columnn&quot;, 
then rsort =&quot;yes&quot; will force rows to be sorted in descent 
way. The default is &quot;yes&quot;	
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_adjrow">adjrow</code></td>
<td>

<p>two numeric values. The first value is used to adjust left 
or right position (x-axis) of labels, and  the second value 
is used to adjust up or down position (y-axis) of labels. 
The fadault values are c(0.4, 1). 	
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_adjcol">adjcol</code></td>
<td>

<p>two numeric values. The first value is used to adjust left 
or right position (x-axis) of labels, and  the second value 
is used to adjust up or down position (y-axis) of labels. 
The fadault values are c(1, 1 ). 	
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_rwangle">rwangle</code></td>
<td>

<p>angle of xlab under heatmap. The Default value is 30. 
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_clangle">clangle</code></td>
<td>

<p>angle of ylab. The default value is 30
</p>
</td></tr>
<tr><td><code id="myheatmap_+3A_maptitle">maptitle</code></td>
<td>

<p>string argument for giving heatmap title, default is 
set to be empty string ' '.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses sigificance to choose data and then 
to normalize the selected data by using z-scale. This 
function has multiple options to select map color, 
distance, cluster and x- and y-lab angles.  
If r&gt;(r1+r2), then data dat are multiple datasets. 
</p>


<h3>Value</h3>

<p>no return value but create a heatmap.
</p>


<h3>Note</h3>

<p>requres gplots and grDevices.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="gplots.html#topic+heatmap.2">heatmap.2</a></code>,
<code><a href="#topic+myheatmap2">myheatmap2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
data(result)
colclass=c("1","1","1","2","2","2")
oldpar &lt;- par(no.readonly =TRUE)
on.exit(par(oldpar))
par(mar=c(7.5,5.5,3.5,1.2))
par(oma=c(3,1,1,3))
myheatmap (dat=result, IDcol=1, nci=7, r=6, r1=3,
r2=3, colrs="terrain.colors", rowBarColor=NULL,  
colBarColor=colclass, labrow="no", labcol="yes", 
rsort="yes", adjrow=c(0.3, 0.0 ), 
adjcol = c(1, 1) , maptitle="My heatmap")

</code></pre>

<hr>
<h2 id='myheatmap2'>
Heatmap2
</h2><span id='topic+myheatmap2'></span>

<h3>Description</h3>

<p>This function is used to display heatmap of 
differential expressions of genes or isoforms  
detected by NBBttest in the real count data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>myheatmap2(dat, IDcol=1, nci=NULL, r, colrs=
"greenred", rwcex=2.8, clcex=2.8, x=10, tree=
"both", method="euclidean", ky=1.5, rowBarColor
=NULL,  colBarColor=NULL, labrow="yes", 
labcol="yes", rsort="yes", adjrow=c(0.2, 0.0 ), 
adjcol = c(1, 1) , rwangle=30, clangle=30, 
maptitle="",keyvalue)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="myheatmap2_+3A_dat">dat</code></td>
<td>

<p>object of NBBttest, including information 
columns, data columns, tvalue column, pvalue and 
selection column.
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_nci">nci</code></td>
<td>

<p>int value, number of columns for information 
of genes or isoforms, like &quot;geneid&quot;, &quot;&quot;tagetid&quot;&quot;, 
&quot;straind&quot;, annotation etc.	
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_idcol">IDcol</code></td>
<td>

<p>int value, indicating which column is ID column 
where IDs will be shown in heatmap rows.
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_r">r</code></td>
<td>

<p>int value, indicating column number of count data 
including empty columns if there are multiple datasets 
with the same row names.
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_colrs">colrs</code></td>
<td>

<p>heatmap colors. User has 8 options: &quot;redgreen&quot;, 
&quot;greenred&quot;, &quot;redblue&quot;, &quot;bluered&quot;, &quot;cm.colors&quot;,
&quot;terrain.colors&quot;,&quot;topo.colors&quot;, and&quot;heat.colors&quot;. 
Default colrs is &quot;redgreen&quot;.   
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_rwcex">rwcex</code></td>
<td>

<p>numeric argument: nonnegative number, used as cex.axis 
for the row axis labeling. Default value is 1.8.
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_clcex">clcex</code></td>
<td>

<p>numeric argument: nonnegative number, used as cex.axis 
for the column axis labeling. Default value is 1.8.
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_x">x</code></td>
<td>

<p>numeric argument: nonnegative number, used as argument 
of cm.colors(x), terrain.colors(x) and topo.colors(x), 
Default value is 10.	
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_tree">tree</code></td>
<td>

<p>tree(s) drawn on row or column or both. User has four 
options: &quot;both&quot; for drawing trees in both row and 
column,&quot;row&quot; for drawing tree only in row,&quot;column&quot; 
for drawing tree in only column, and &quot;none&quot; for no 
tree specified on rows and columns. If tree =both, 
then colomns and rows are sorted by trees. If tree = 
&quot;row&quot;, the colmns are not sorted.  If tree =&quot;column&quot;, 
then rows are not sorted. Default is &quot;both&quot;.
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_method">method</code></td>
<td>

<p>method to be chosen to calculate distance between 
columns or rows. It has four options: &quot;euclidean&quot;, 
&quot;pearson&quot;,&quot;spearman&quot; and &quot;kendall&quot;. The latter three 
are d=1-cc where  cc is correlation coefficients. 
Default is &quot;euclidean&quot;. 	
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_ky">ky</code></td>
<td>

<p>numeric argument: nonnegative number, is used to 
determine key size. The defalut =1.5.	
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_rowbarcolor">rowBarColor</code></td>
<td>

<p>(optional) character vector for RowSideColors and
colRow. Length of rowBarColor equals to gene or 
isoform number or row number.  rowBarColor contains 
the color names of classes or typpes or groups of 
row names and show row color side bars and color 
row names. If row names, for example, genes in row 
are not classified, then we suggest user use its 
default value: rowBarColor=NULL.
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_colbarcolor">colBarColor</code></td>
<td>

<p>(optional) character vector for colSideColors and 
colCol. Length of colBarColor equals to sample number.  
colBarColor contains the color names of classes or 
typpes or groups of  coloumn names and show column 
color side bars and color column names. If column 
names, for example, samples in column are just two 
types, then we suggest user use its default value: 
colBarColor=NULL.	
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_labrow">labrow</code></td>
<td>

<p>a string, logistic value. Rows on heatmap are labeled 
with genes or targets if labrow=&quot;yes&quot;, otherwise, 
the rows are not labeled.
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_labcol">labcol</code></td>
<td>

<p>a string, logistical value. Columns on heatmap are labeled 
with samples or treatments if labcol=&quot;yes&quot;, otherwise, the 
columns are not labeled. 	
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_adjrow">adjrow</code></td>
<td>

<p>two numeric values. The first value used to adjust left or 
right position (x-axis) of labels and  the second value is 
used to adjust up or down position (y-axis) of labels. The 
fadault values are c(0.3, 0.0). 	
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_adjcol">adjcol</code></td>
<td>

<p>two numeric values. The first value used to adjust left or 
right position (x-axis) of labels, and  the second value is 
used to adjust up or down position (y-axis) of labels. 
The fadault values are c(1, 1 ). 	
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_rsort">rsort</code></td>
<td>

<p>logistical value. If choose tree=&quot;both&quot; or &quot;row&quot;, then 
rsort does not work. However, if tree =&quot;none&quot; or &quot;columnn&quot;, 
then rsort =&quot;yes&quot; will force rows to be sorted in descent 
way. The default is &quot;yes&quot;	
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_rwangle">rwangle</code></td>
<td>

<p>heatmap object: angle of xlab. Thed default value is 30. 
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_clangle">clangle</code></td>
<td>

<p>heatmap object: angle of ylab. The default value is 30. 
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_maptitle">maptitle</code></td>
<td>

<p>string argument for giving heatmap title. The default 
value is set to be empty string ' '.
</p>
</td></tr>
<tr><td><code id="myheatmap2_+3A_keyvalue">keyvalue</code></td>
<td>

<p>sting argument for name of key x-axis.	
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses selection to choose genes or isoforms 
in the data and then to normalize the selected data by 
using n-scale. Different from z-score, n-score does not 
follow standard normal distribution with mean = 0 and 
variance =1 for all rows but it has the same largest 
count in all rows and shows multiple colors for numeric 
difference between two conditions. This function has 
multiple options to select map color, distance, cluster 
and x-lab and y-lab angles. This function can be able to 
display multiple datasets in two ways: if multiple 
datasets have the same row names or features, the these 
datasets are put onto the different columns separated 
with empty column named with dataset names.If multiple 
datasets have the same column names of the datasets, 
then put them on different rows saparated with empty 
rows named with dataset names or whatever names user 
specifies.  
</p>


<h3>Value</h3>

<p>not return value but create a heatmap.
</p>


<h3>Note</h3>

<p>requires gplots and grDevices. If the data for heatmap 
are multiple datasets, then tree=&quot;none&quot; and sort=&quot;no&quot;, 
otherwise, myheatmap2 will get error. So before 
performing myheatmap2, user should sort the data in excel. 
</p>


<h3>Author(s)</h3>

<p><a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a> 
</p>


<h3>See Also</h3>

<p><code><a href="gplots.html#topic+heatmap.2">heatmap.2</a></code>, <code><a href="grDevices.html#topic+grDevices">grDevices</a></code>, 
and <code><a href="#topic+myheatmap">myheatmap</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(result)
colclass=c("1","1","1","2","2","2")
oldpar &lt;- par(no.readonly =TRUE)
on.exit(par(oldpar))
par(mar=c(7.5,5.5,3.5,1.2))
par(oma=c(3,1,1,3))
myheatmap2(dat=result, IDcol=1, nci=7, r=6, 
colrs="greenred", rwcex=1.8, clcex=1.8, x=10, 
tree="both", method="euclidean", ky=1.5, 
rowBarColor=NULL,  colBarColor=colclass, 
labrow="no", labcol="yes", adjrow=c(0.2, 0.0 ), 
adjcol = c(1, 1) , rwangle=0, clangle=30,
maptitle="My heatmap2",keyvalue="count")

</code></pre>

<hr>
<h2 id='NBBplot'>
Plot differential expression of exons within a 
specified gene using result outputed by NBBttest.
</h2><span id='topic+NBBplot'></span>

<h3>Description</h3>

<p>After performing NBBttest, NBBplot can be used to 
show differential expression of exons within a 
specified gene in na and nb replicates between
conditions A and B.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NBBplot(res, gene, nci, na, nb, C1, C2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NBBplot_+3A_res">res</code></td>
<td>

<p>object of NBBttest containing information of genes 
including gene name,strand, chromosome,exons, introns,
data, and t-value, p-value, significance/selection etc.
</p>
</td></tr>
<tr><td><code id="NBBplot_+3A_gene">gene</code></td>
<td>

<p>gene name or symbol specified by user.
</p>
</td></tr>
<tr><td><code id="NBBplot_+3A_nci">nci</code></td>
<td>

<p>number of columns for gene information.
</p>
</td></tr>
<tr><td><code id="NBBplot_+3A_na">na</code></td>
<td>

<p>replicate number in condition A
</p>
</td></tr>
<tr><td><code id="NBBplot_+3A_nb">nb</code></td>
<td>

<p>replicate number in condition B
</p>
</td></tr>
<tr><td><code id="NBBplot_+3A_c1">C1</code></td>
<td>

<p>name for condition 1(A)
</p>
</td></tr>
<tr><td><code id="NBBplot_+3A_c2">C2</code></td>
<td>

<p>name for condition 2(B)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>NBBplot consists of two parts: top is expression 
value of each exon in each replicate in two 
conditions marked in red and blue and bottom is 
boxes for exon and solid lines for introns. 
Differential expression of an exon is marked in red.   
</p>


<h3>Value</h3>

<p>output NBBplot figure for given gene.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="DEXSeq.html#topic+plotDEXSeq">plotDEXSeq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exondata)

NBBplot(res=exondata, gene="H2-DMb1", nci=9, na=3, nb=3, C1="WT", C2="KO")

</code></pre>

<hr>
<h2 id='normalized'>
Normalization of data
</h2><span id='topic+normalized'></span>

<h3>Description</h3>

<p>Function normalize makes all libraries in dataset 
have the same library size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalized(dat, nci, m=0, lg2="no")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalized_+3A_dat">dat</code></td>
<td>

<p>count data of RNA reads.
</p>
</td></tr>
<tr><td><code id="normalized_+3A_nci">nci</code></td>
<td>

<p>number of columns for the information of genes or 
isoforms in dataset.
</p>
</td></tr>
<tr><td><code id="normalized_+3A_m">m</code></td>
<td>

<p>numeric value for choosing genes or isoforms. If user 
wants to discard genes or isoforms with mean &lt; 5, 
then m = 5. The default value is 0.
</p>
</td></tr>
<tr><td><code id="normalized_+3A_lg2">lg2</code></td>
<td>

<p>logistic value. lg2=&quot;yes&quot;indicates that data are 
transformed in logarithm of 2.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Due to difference in RNA abstraction between libraries 
or cell samples or tissues, PCR amounts of RNA libraries 
would have difference that is not due to biological 
effects. To correctly compare differential expressisons 
of genes between conditions or samples, one must should 
give the same RNA abstraction in all given samples. 
This is impossible. To address this problem, only one 
way is to normalize these count data across all given 
samples so that all experimental samples (libraries) 
have the same total counts. 
</p>


<h3>Value</h3>

<p>output a standard datasheet.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, and 
Joel R. Neilson(2015) A Powerful Statistical Approach 
for Large-scale Differential Transcription Analysis. 
<em>Plos One</em>, 10.1371/journal.pone.0123658.  
<br />
Anders S, Huber W (2010) Differential expression 
analysis for sequence count data. <em>Genome Biol</em>, 
11: R106.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jkttcell)	
njkttcell&lt;-normalized(dat=jkttcell[1:50,],nci=7)
</code></pre>

<hr>
<h2 id='oddratio'>
Calculation of zeta (<code class="reqn">\zeta</code>)
</h2><span id='topic+oddratio'></span>

<h3>Description</h3>

<p><code class="reqn">\zeta</code>  is used to measure homogeneity intensity of two subdatasets.  If  <code class="reqn">\zeta</code>  is larger than 1, these two subdatasets have good homogeneity; otherwise, <code class="reqn">\zeta &lt;1</code> indicates that two subdatasets have poor homogeneity (big noise).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oddratio(XX, na, nb)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oddratio_+3A_xx">XX</code></td>
<td>

<p>nonnegative count data of RNA reads generated by next generation sequencing.
</p>
</td></tr>
<tr><td><code id="oddratio_+3A_na">na</code></td>
<td>

<p>number of replicate libraries in condition A.
</p>
</td></tr>
<tr><td><code id="oddratio_+3A_nb">nb</code></td>
<td>

<p>number of replicate libraries in condition B.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Zeta (<code class="reqn">\zeta</code>) is defined as 
</p>
<p style="text-align: center;"><code class="reqn">\zeta=ln\left(1+\frac{\bar{X} \times \sigma^2 +1}{\bar{X}_A\times \sigma^2_A+\bar{X}_B\times \sigma^2_B+1}\right) </code>
</p>

<p>where
<code class="reqn">\zeta</code> is different from <code class="reqn">\psi</code>. If two subdatasets have big a 
gap and good homogeneity,then seta value has much larger than 1.
</p>


<h3>Value</h3>

<p>zeta vector.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, and Joel R. Neilson(2015)  A Powerful Statistical Approach for Large-scale Differential Transcription Analysis.  <em>Plos One</em>. 2015 DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pratio">pratio</a></code>, <code><a href="#topic+mbetattest">mbetattest</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>XX&lt;-matrix(NA,2,8)
XX[1,]&lt;-c(112,122, 108,127,302, 314, 322, 328)
XX[2,]&lt;-c(511, 230, 754, 335,771, 842, 1014,798)
oddratio(XX=XX,na=4,nb=4)
</code></pre>

<hr>
<h2 id='omega'>
Omega calcularion
</h2><span id='topic+omega'></span>

<h3>Description</h3>

<p>Omega function is a function that 
is used to estimate omega using simulate  
null data from negative bionomial distribution. 
is a null rho that is used as 
a threshold for real rho. Simulation is 
dependent on the original data. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>omega(XX, nci, r1, r2, sn, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="omega_+3A_xx">XX</code></td>
<td>

<p>the real dataset.
</p>
</td></tr>
<tr><td><code id="omega_+3A_nci">nci</code></td>
<td>

<p>number of columns for information of data, like gene id, 
isoform id, gene name, etc.
</p>
</td></tr>
<tr><td><code id="omega_+3A_r1">r1</code></td>
<td>

<p>size of sample 1 or number of replicates in condition 1.
</p>
</td></tr>
<tr><td><code id="omega_+3A_r2">r2</code></td>
<td>

<p>size of sample 2 or number of replicates in condition 2.
</p>
</td></tr>
<tr><td><code id="omega_+3A_sn">sn</code></td>
<td>

<p>number of simulations specified.
</p>
</td></tr>
<tr><td><code id="omega_+3A_alpha">alpha</code></td>
<td>

<p>significance level of test. Default value is 0.05.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is to use  null data to calculate 
omega value with rho  = 1. 
</p>


<h3>Value</h3>

<p>return a numeric value.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, and 
Joel R. Neilson(2015)  A Powerful Statistical Approach 
for Large-scale Differential Transcription Analysis. 
<em>Plos One</em>. 2015 DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pratio">pratio</a></code> and <code><a href="#topic+oddratio">oddratio</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jkttcell)	

w&lt;-omega(XX=jkttcell[1:100,],nci=7,r1=3,r2=3,sn=2,alpha=0.05)	
</code></pre>

<hr>
<h2 id='pathwayHeatmap'>
Heatmap for pathways found by gene ontology analysis
</h2><span id='topic+pathwayHeatmap'></span>

<h3>Description</h3>

<p>This function is used to show differential expressions 
of pathways or functions between conditions. These 
pathways or functions were detected by function 
annotation or gene ontology methods such as David function 
analysis tools or Ingenuity pathway analysis. Pathway score 
or pathway value is a weighted expression value across 
genes in a pathway or a function. The weigths of genes are 
given by p-values of enrichment or hit in function analysis. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathwayHeatmap(dat, pathway, nci, r1, r2, colclass, rowclass, 
colrs, maptitle)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathwayHeatmap_+3A_dat">dat</code></td>
<td>

<p>count dataset that contains a column for gene name 
and count data columns in two conditions.
</p>
</td></tr>
<tr><td><code id="pathwayHeatmap_+3A_pathway">pathway</code></td>
<td>

<p>a list that lists a function colomn and at least two 
gene columns. Row is function name.
</p>
</td></tr>
<tr><td><code id="pathwayHeatmap_+3A_nci">nci</code></td>
<td>

<p>number of columns for gene information such as gene name, 
strand, chromosome, etc in dataset dat.
</p>
</td></tr>
<tr><td><code id="pathwayHeatmap_+3A_r1">r1</code></td>
<td>

<p>number of replicates in condition 1 in dataset dat.
</p>
</td></tr>
<tr><td><code id="pathwayHeatmap_+3A_r2">r2</code></td>
<td>

<p>number of replicates in condition 2 in dataset dat.
</p>
</td></tr>
<tr><td><code id="pathwayHeatmap_+3A_colclass">colclass</code></td>
<td>

<p>colonm class, equivalent to replicates in two conditions, 
such as c(1,1,1,1,2,2,2,2),meaning that condition A has 4 
replicates and condition B has 4 replicates. Default =NULL.
</p>
</td></tr>
<tr><td><code id="pathwayHeatmap_+3A_rowclass">rowclass</code></td>
<td>

<p>row class, pathway class, or pathway module. Default = NULL.
</p>
</td></tr>
<tr><td><code id="pathwayHeatmap_+3A_colrs">colrs</code></td>
<td>

<p>heatmap colors. User has 8 options: &quot;redgreen&quot;, 
&quot;greenred&quot;, &quot;redblue&quot;, &quot;bluered&quot;, &quot;cm.colors&quot;,
&quot;terrain.colors&quot;,&quot;topo.colors&quot;, and&quot;heat.colors&quot;.  
</p>
</td></tr>
<tr><td><code id="pathwayHeatmap_+3A_maptitle">maptitle</code></td>
<td>

<p>heatmap title, default = &quot;&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses sigificance to choose gene function/
pathwaydata and then to normalize the selected data by 
using z-scale. This function has multiple options to 
select map color, distance, cluster and x- and y-lab 
angles. 
</p>


<h3>Value</h3>

<p>return a graph for heatmap.	
</p>


<h3>Note</h3>

<p>requres gplots and grDevices.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="gplots.html#topic+heatmap.2">heatmap.2</a></code>,<code><a href="#topic+myheatmap">myheatmap</a></code>,
<code><a href="#topic+myheatmap2">myheatmap2</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(upGAm)
data(pathwy.A.up)
pathwayup&lt;-pathwy.A.up
colclass=c("1","1","1","1","2","2","2","2","2","2")
oldpar &lt;- par(no.readonly =TRUE)
on.exit(par(oldpar))
par(mar=c(7.5,5.5,3.5,3))
par(oma=c(3,1,1,10))
pathwayHeatmap(dat=upGAm,pathway=pathwayup,nci=1,
r1=4,r2=6,colclass=colclass,rowclass=NULL,
colrs="greenred",maptitle="pathway up-expression 
in Group A")
</code></pre>

<hr>
<h2 id='pathwy.A.up'>
Pathway or function data
</h2><span id='topic+pathwy.A.up'></span>

<h3>Description</h3>

<p>pathway data are functions or pathways of up-regulated 
genes given by gene annotation tool such as DAVID 
gene function annotion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("pathwy.A.up")</code></pre>


<h3>Format</h3>

<p>A data frame with 39 observations on the following 159 
variables.
</p>
<p> 39 observations, 159 variables             

</p>


<h3>Details</h3>

<p>The 39 observations are 39 functions/pathways 
listed each being a set up-regulated genes found 
by NBBttest in knockdowned DDX39 cell line. Log10 
of pvalue of gene in differential expression 
detection is used to weight conpoment of this 
gene playing role in this function. Then heatmap 
of pathways or functions are made using myheatmap. 
V1-V159 are gene sets of function or pathways   
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pathwy.A.up)
</code></pre>

<hr>
<h2 id='pratio'>
Calculation of psi (<code class="reqn">\psi </code>)
</h2><span id='topic+pratio'></span>

<h3>Description</h3>

<p><code class="reqn">\psi </code> is also called polar ratio. It is used 
to measure overlap of two subdatasets. If <code class="reqn">\psi&gt;1</code>, 
these two subdatasets have a gap, while <code class="reqn">\psi &lt;1</code> 
indicates that two subdatasets overlap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pratio(xx, na, nb)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pratio_+3A_xx">xx</code></td>
<td>

<p>count data of RNA reads generated by next generation 
sequencing.
</p>
</td></tr>
<tr><td><code id="pratio_+3A_na">na</code></td>
<td>

<p>number of replicate libraries in condition A.
</p>
</td></tr>
<tr><td><code id="pratio_+3A_nb">nb</code></td>
<td>

<p>number of replicate libraries in condition B.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Psi (<code class="reqn">\psi </code>) is defined as
</p>
<p style="text-align: center;"><code class="reqn">\psi = max\left(\frac{min(X_A)}{max(X_B)},\frac{min(X_B)}
{max(X_A)}\right)</code>
</p>



<h3>Value</h3>

<table>
<tr><td><code>pratio</code></td>
<td>
<p>pratio list</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, and 
Joel R. Neilson(2015) A Powerful Statistical Approach 
for Large-scale Differential Transcription Analysis. 
<em>Plos One</em>. 2015 DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+omega">omega</a></code>, and <code><a href="#topic+oddratio">oddratio</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>XX&lt;-matrix(NA,2,8)
XX[1,]&lt;-c(112,122, 108,127,302, 314, 322, 328)
XX[2,]&lt;-c(511, 230, 754, 335,771, 842, 1014,798)
pratio(xx=XX,na=4,nb=4)
</code></pre>

<hr>
<h2 id='prime3_PRP8_50'>
3'UTR splicing data of 50 genes detected in the 
knockdowned PRP8 cell line
</h2><span id='topic+prime3_PRP8_50'></span>

<h3>Description</h3>

<p>Splicing events occurring in 3'UTR of 50 genes 
were detected in the knockdowned PRP8 cell line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("prime3_PRP8_50")</code></pre>


<h3>Format</h3>

<p>A data frame with 50 observations on the following 24 variables.
</p>

<dl>
<dt><code>chr</code></dt><dd><p> a factor with levels</p>
</dd>
<dt><code>strand</code></dt><dd><p>a factor with levels</p>
</dd>
<dt><code>isoform</code></dt><dd><p>a factor with levels</p>
</dd>
<dt><code>Gene</code></dt><dd><p>a factor with levels</p>
</dd>
<dt><code>exon_const_start</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>exon_const_end</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>exon_alt1_start</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>exon_alt1_end</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>exon_alt2_start</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>exon_alt2_end</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Siha_Ctl_I_S3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Siha_Ctl_II_S7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Siha_Ctl_III_S11</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PRP8_I_S2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PRP8_II_S6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PRP8_III_S10</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>tvalue</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rho</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>pvalue</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>adjp</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>w</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X.1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>X.2</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>This dataset is an example for annotation of genes within 
which splicing events occurring in 3'UTR were detected by 
spladder-NBBttest in knockdowned PRP8 cell line. Gene 
information columns are assigned to &quot;chr&quot;,&quot;Gene&quot;, &quot;isoform&quot;, 
&quot;exon_const_start&quot; and &quot;exon_const_end&quot; that are start 
and end positions of constituted exons and &quot;exon_alt1_start&quot; 
and &quot;exon_alt1_end&quot; that are start and end positions of 
alternative exons. Count data of RNA reads are assigned 
to three replicate control columns (Siha_Ctl_I_S3,II-S7,
III_S11) and three replicate knokdowned PRP8 columns 
(PRP8_I_S2,II_S6 and III_S10). The tvalue is t-statistic, 
rho is gene-wise variable, pvalue is p-value for t-statistic, 
w is <code class="reqn">\omega</code>,a threshold for rho, FDR (X.1)is false 
discovery rate with <code class="reqn">\alpha</code>= 0.05, selection (X.2)= 1 
if pvalue &lt; FDR, 0, otherwise.    
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(prime3_PRP8_50)
</code></pre>

<hr>
<h2 id='QC'>
Count data quality check </h2><span id='topic+QC'></span>

<h3>Description</h3>

<p>QC is used to check quality of count data in 
two ways: scatter plot of two replicated samples 
and correlation heatmap of all samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QC(dat, nci, S1="NULL", S2="NULL", method=
"plot", colrs="greenred", rwcex=1.8, clcex=1.8, 
x=10, tree="none", log="none", col="blue", 
pch=19, labsize=1.5, axis=1.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QC_+3A_dat">dat</code></td>
<td>

<p>count matric dataset.
</p>
</td></tr>
<tr><td><code id="QC_+3A_nci">nci</code></td>
<td>

<p>number of columns containing data information suach 
as gene id, library id, target id, gene name, 
strand etc.
</p>
</td></tr>
<tr><td><code id="QC_+3A_s1">S1</code></td>
<td>

<p>numeric int value, indicating which column in data 
matrix is specified in x-axis. S1 &gt; nci. 
</p>
</td></tr>
<tr><td><code id="QC_+3A_s2">S2</code></td>
<td>

<p>numeric int value, indicating which column in data 
matrix is specified in y-axis. S2&gt; nci. 
</p>
</td></tr>
<tr><td><code id="QC_+3A_method">method</code></td>
<td>

<p>string. Here two methods are given for choice: 
&quot;plot&quot; and &quot;heatmap&quot;. Default is &quot;plot&quot;. 
</p>
</td></tr>
<tr><td><code id="QC_+3A_colrs">colrs</code></td>
<td>

<p>string. 8 color sets are given for choice in this 
function: &quot;redgreen&quot;, &quot;heat.colors&quot;, &quot;redblue&quot;, 
&quot;greenred&quot;, &quot;bluered&quot;, &quot;cm.colors&quot;, &quot;terrain.colors&quot;, 
&quot;topo.colors&quot;. The default color set is &quot;redgreen&quot;.
</p>
</td></tr>
<tr><td><code id="QC_+3A_rwcex">rwcex</code></td>
<td>

<p>positive numbers, used as cex.axis for the row axis 
labeling. The default value is 1.8. 
</p>
</td></tr>
<tr><td><code id="QC_+3A_clcex">clcex</code></td>
<td>

<p>positive numbers, used as cex.axis for the column 
axis labeling. The default value is 1.8. 
</p>
</td></tr>
<tr><td><code id="QC_+3A_x">x</code></td>
<td>

<p>numeric argument: positive number, used as argument 
of cm.colors(x), terrain.colors(x) and topo.colors(x), 
the default value is 10.	
</p>
</td></tr>
<tr><td><code id="QC_+3A_tree">tree</code></td>
<td>

<p>tree(s) drwan on row or column or both. User has 
four options: &quot;both&quot; for drawing trees on both row 
and column,&quot;row&quot; for drawing treeonly on row,&quot;column&quot; 
for drawing tree on only column, and &quot;none&quot; for no 
tree specified. Default is &quot;none&quot;.
</p>
</td></tr>
<tr><td><code id="QC_+3A_log">log</code></td>
<td>

<p>string. Two options are given for choice: &quot;none&quot; and 
&quot;log&quot;. log=&quot;log&quot; indicates that data value would be 
transfomed with log2. Default is &quot;none&quot;. 
</p>
</td></tr>
<tr><td><code id="QC_+3A_col">col</code></td>
<td>

<p>string, used to specify scatter plot dot color. 
</p>
</td></tr>
<tr><td><code id="QC_+3A_pch">pch</code></td>
<td>

<p>numeric value, used to specify dot type.
</p>
</td></tr>
<tr><td><code id="QC_+3A_labsize">labsize</code></td>
<td>

<p>numeric value for size of xlabel and ylabel	
</p>
</td></tr>
<tr><td><code id="QC_+3A_axis">axis</code></td>
<td>

<p>numeric value for axis scale.	
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>S1, S2 and nci must be given numeric int values for 
plot and heatmap. However, when method is chosen to 
be &quot;heatmap&quot;, then S1 and S2  are not specified.  
Columns of information should be left of matrix and 
count data should be after columns of information. 
</p>


<h3>Value</h3>

<p>not return values but create scatter plot or heatmap plot.
</p>


<h3>Note</h3>

<p>requires gplots and grDevices.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a> 
</p>


<h3>See Also</h3>

<p><code><a href="gplots.html#topic+heatmap.2">heatmap.2</a></code> and <code><a href="grDevices.html#topic+grDevices">grDevices</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jkttcell)
QC(dat=jkttcell, nci=7, S1=8, S2=9, method = "plot", 
log = "log", col = "blue", pch = 19)
QC(dat=jkttcell, nci=7, S1=8, S2=9, method = "plot", 
log = "log", col = "blue", pch = 19)
QC(dat=jkttcell, nci=7, method = "heatmap", log = "log")
</code></pre>

<hr>
<h2 id='result'>
Jurkat T-cell transcritomic data with isoforms 
selected by NBBttest
</h2><span id='topic+result'></span>

<h3>Description</h3>

<p>A data  consist of 7 information columns and 6 
data (numeric) columns and 1953 RNA isoforms 
selected by NBBttest are used to make heatmaps.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("result")</code></pre>


<h3>Format</h3>

<p>A data frame with 1953 observations on the 
following 14 variables.
</p>

<dl>
<dt><code>tagid</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>geneid</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>name</code></dt><dd><p>: a string vector</p>
</dd>
<dt><code>chr</code></dt><dd><p>: a string vector, a set of 24 chrosomomes</p>
</dd>
<dt><code>strand</code></dt><dd><p>: a factor with levels <code>-</code> <code>+</code></p>
</dd>
<dt><code>pos</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>anno</code></dt><dd><p>: string vector, poly(A) site types</p>
</dd>
<dt><code>Jurk.NS.A</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.NS.B</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.NS.C</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.48h.A</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.48h.B</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.48h.C</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>tvalue</code></dt><dd><p>: a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>The original data are count data generated by next 
generation sequencing from Jurkat T-cells. This 
dataset is a short dataset containing 1953 isoforms 
that show differential expression between rest 
status and stimulation status. The T-cells were 
treated by resting and stimulating with <em>CD3/CD28</em> 
for 48 hours. The data have 7 columns for the information 
of <em>poly(A)</em> site: &quot;tagid&quot;, &quot;geneid&quot;, &quot;gene name&quot;, 
&quot;chromosome&quot;, &quot;strand&quot;,<em>poly(A)</em> site position, 
<em>poly(A)</em> site annotation and 6 columns for 
data: Jurk.NS.A, Jurk.NS.B, Jurk.NS.C,	Jurk.48h.A,	
Jurk.48h.B,	Jurk.48h.C. where NS means Normal state 
or no stimulation and 48h means 48 hours after 
<em>CD3/CD28</em> stimulatuin of T-cells. 13409 
<em>RNA</em> isoforms were detected to have alternative 
<em>poly(A)</em> sites.
</p>


<h3>Value</h3>

<p>datasheet containd	ID, information, count data of RNA 
reads.
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, and 
Joel R. Neilson(2015) A Powerful Statistical Approach 
for Large-scale Differential Transcription Analysis.
<em>Plos One</em>. DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(result)
</code></pre>

<hr>
<h2 id='sgRNA'>
sgRNA dataset
</h2><span id='topic+sgRNA'></span>

<h3>Description</h3>

<p>This dataset was created by simulating single 
guide RNAs to edit genes. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("sgRNA")</code></pre>


<h3>Format</h3>

<p>A data frame with 1000 observations on the 
following 11 variables.
</p>

<dl>
<dt><code>gene</code></dt><dd><p>a factor with levels </p>
</dd>
<dt><code>sgRNA</code></dt><dd><p>a factor with levels</p>
</dd>
<dt><code>class</code></dt><dd><p>a factor with levels</p>
</dd>
<dt><code>L1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>L2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>L3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>L4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H4</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>The dataset is CRISPR screening data. Each gene was 
edited by 10 sgRNAs that contains a targeting sequence 
(crRNA sequence) and a Cas9 nuclease-recruiting sequence 
(tracrRNA). L1-L4 samples were lowly targeted by sgRNAs 
and H1-H4 were highly targeted by sgRNAs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sgRNA)
</code></pre>

<hr>
<h2 id='simSplicing'>
Simulated alternative splicing
</h2><span id='topic+simSplicing'></span>

<h3>Description</h3>

<p>This alternative splicing count data were created 
by simulating dorsal and ventral RNA-sequence 
count data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("simSplicing")</code></pre>


<h3>Format</h3>

<p>A data frame with 5000 observations on the following 
9 variables.
</p>

<dl>
<dt><code>Isoform</code></dt><dd><p>a factor with levels</p>
</dd>
<dt><code>geneid</code></dt><dd><p>a factor with levels</p>
</dd>
<dt><code>label</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>D1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>D2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>D3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>V3</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>D1-D3 are samples from mouse dorsal tissue and V1-V3 
are samples from mouse ventral tissue. This dataset 
was simulated with negative binomial ditribution with 
10 percent of isoforms of being differentially spliced  
and differential effect of 500U where U is uniform variable.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(simSplicing)
</code></pre>

<hr>
<h2 id='simulat'>
Simulation
</h2><span id='topic+simulat'></span>

<h3>Description</h3>

<p>This function uses negative binomial (NB) pseudorandom 
generator to create count datasets of RNA isoform 
reads based on real data. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulat(yy, nci, r1, r2, p, q, A)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulat_+3A_yy">yy</code></td>
<td>

<p>real count data
</p>
</td></tr>
<tr><td><code id="simulat_+3A_nci">nci</code></td>
<td>

<p>nonnegative int value: column number of information related 
to genes or isoforms.
</p>
</td></tr>
<tr><td><code id="simulat_+3A_r1">r1</code></td>
<td>

<p>numeric argument: number of replicate libraries in condition 1.
</p>
</td></tr>
<tr><td><code id="simulat_+3A_r2">r2</code></td>
<td>

<p>numeric argument: number of replicate libraries in condition 2.
</p>
</td></tr>
<tr><td><code id="simulat_+3A_p">p</code></td>
<td>

<p>numeric argument: proportion of genes or isoforms 
differentially expressed. The value is in range of 0~1. 
Default is 0.
</p>
</td></tr>
<tr><td><code id="simulat_+3A_q">q</code></td>
<td>

<p>numeric argument: proportion of genes or isoforms artificially 
noised. The value is in range of 0~1. Default is 0.
</p>
</td></tr>
<tr><td><code id="simulat_+3A_a">A</code></td>
<td>

<p>numeric argument: conditional effect value. The value is 
larger than or equal to 0. Default is 0.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Null count data are created by using R negative binomial 
pseudorandom generator rnbinom with <code class="reqn">\mu</code> and size. 
Parameters <code class="reqn">\mu</code> and size are given by mean and  
variance drawn from real read counts of a gene set or 
an isoform set in a condition. Condition (or treatment) 
effect on differential transcription of isoforms is 
linearly and randomly assigned to genes or isoforms. 
The conditional effect = AU where U is uniform variable 
and A is input constant. P percent of genes or isoforms 
is set to be differentially expressed or differentially 
spliced. Q percent of genes or isoforms has technical 
noise.  If P = 0, then simulation is null simulation, 
the data are null data or baseline data.
</p>


<h3>Value</h3>

<p>Return count data.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, and 
Joel R. Neilson(2015) A Powerful Statistical Approach 
for Large-scale Differential Transcription Analysis.
<em>Plos One</em>, 10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+NegBinomial">NegBinomial</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jkttcell)
jknull&lt;-simulat(yy=jkttcell[1:60,],nci=7,r1=3,
r2=3,p=0,q=0.2,A=0)
</code></pre>

<hr>
<h2 id='skjt'>
Simulated Null Transcriptomic data
</h2><span id='topic+skjt'></span>

<h3>Description</h3>

<p>The dataset skjt generated by using R negative binomial 
pseudorandom generator rnbinom is used as an example 
for calculating omega.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("skjt")</code></pre>


<h3>Format</h3>

<p>A data frame with 13409 observations on the following 
14 variables.
</p>

<dl>
<dt><code>geneid</code></dt><dd><p>: a string vector </p>
</dd>
<dt><code>tagid</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>geneid.1</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>name</code></dt><dd><p>: a string vector</p>
</dd>
<dt><code>chr</code></dt><dd><p>: a string vector</p>
</dd>
<dt><code>strand</code></dt><dd><p>: a character vector</p>
</dd>
<dt><code>pos</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>anno</code></dt><dd><p>: a string vector </p>
</dd>
<dt><code>Jurk.NS.A</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.NS.B</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.NS.C</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.48h.A</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.48h.B</code></dt><dd><p>: a numeric vector</p>
</dd>
<dt><code>Jurk.48h.C</code></dt><dd><p>: a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>The dataset skjt was generated by using R negative 
binomial pseudorandom generator rnbinom with <code class="reqn">\mu</code> 
and size. Parameters <code class="reqn">\mu</code> and size are given 
by mean and  variance drawn from real  Jurkat T cell 
transcriptomic count data . Condition (or treatment) 
effect on differential transcription of isoforms was 
set to zero. The data have 13409 genes and 7 information 
columns: geneid	 tagid	name <code>chr</code>,<code>strand</code>,
<code>pos</code>,<code>anno</code>, and 6 data columns.
</p>


<h3>Value</h3>

<p>a datasheet containd ID, information, count data of 
RNA reads.
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, and 
Joel R. Neilson(2015) A Powerful Statistical Approach 
for Large-scale Differential Transcription Analysis. 
<em>Plos One</em>. DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(skjt)
</code></pre>

<hr>
<h2 id='smbetattest'>
Performance of multiple beta t-test on simulated data
</h2><span id='topic+smbetattest'></span>

<h3>Description</h3>

<p>This function is to peform beta t-test with <code class="reqn">\rho</code>
=1 and <code class="reqn">\omega</code>=1 on simulated data. The result 
lists differentially expressed genes or isoforms and 
their <code class="reqn">\rho</code> values. The <code class="reqn">\rho</code> values are used 
to calculate <code class="reqn">\omega</code> value for performance of beta 
t-tests on the real data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smbetattest(X, na, nb, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smbetattest_+3A_x">X</code></td>
<td>

<p>simulated count data with N genes or isoforms.
</p>
</td></tr>
<tr><td><code id="smbetattest_+3A_na">na</code></td>
<td>

<p>number of replicate libraries in condition A.
</p>
</td></tr>
<tr><td><code id="smbetattest_+3A_nb">nb</code></td>
<td>

<p>number of replicate libraries in condition B.
</p>
</td></tr>
<tr><td><code id="smbetattest_+3A_alpha">alpha</code></td>
<td>

<p>statistical probabilistic threshold, default is 0.05.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Before performing NBBttest on real data, user needs 
<code class="reqn">\omega</code> value for the threshold of <code class="reqn">\rho</code>. 
To determine <code class="reqn">\omega</code> value, user is requred to 
generate a set of null data having the same gene or 
isoform number and the same numbers of replicate 
libraries in two conditions and then performs beta 
t-test on the null datasets by setting <code class="reqn">\rho</code> =1 
and <code class="reqn">\omega</code> =1.  In current package, NBBttest 
can automatically perform the simulation of null data,
multiple beta t-test to estimate <code class="reqn">\omega</code>. 
</p>


<h3>Value</h3>

<p>Return a set of null <code class="reqn">\rho</code> values.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>References</h3>

<p>Yuan-De Tan Anita M. Chandler, Arindam Chaudhury, and 
Joel R. Neilson(2015) A Powerful Statistical Approach 
for Large-scale Differential Transcription Analysis. 
<em>Plos One</em>. DOI: 10.1371/journal.pone.0123658. 
</p>


<h3>See Also</h3>

<p>See Also as <code><a href="#topic+mbetattest">mbetattest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(skjt) 
nrho&lt;-smbetattest(X=skjt[1:60,],na=3,nb=3,alpha=0.05)

</code></pre>

<hr>
<h2 id='subdata'>
Splite data into two subsets
</h2><span id='topic+subdata'></span>

<h3>Description</h3>

<p>Data are splited into two subsets: gene 
single-isoform  data and gene multi-isoform data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subdata(xx, sg)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subdata_+3A_xx">xx</code></td>
<td>

<p>real data containing single-isoforms and 
multi-isoforms of genes in rows.
</p>
</td></tr>
<tr><td><code id="subdata_+3A_sg">sg</code></td>
<td>

<p>int value.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the RNA count data, some genes have only 
one isoform, some genes have multiple isoforms.
so data are divided by subdata.R into two datasets: 
single-isoform data and multi-isoform data.
</p>


<h3>Value</h3>

<p>return dataset by setting sg=1 or sg=2.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan
<a href="mailto:tanyuande@gmail.com">tanyuande@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(jkttcell)	
colnames(jkttcell)[3]&lt;-"Gene"
jk.mg&lt;-subdata(xx=jkttcell, sg=2)
</code></pre>

<hr>
<h2 id='upGAm'>
Count data of group A treated breast cancer in mice
</h2><span id='topic+upGAm'></span>

<h3>Description</h3>

<p>Data of RNA-seq reads were obtained from drug 
treated breast cancer in mice and mapped by using 
STAR onto mm10 to create count data. Dataset 
upGAm is output of NBBttest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("upGAm")</code></pre>


<h3>Format</h3>

                                                                                              
<p>A data frame with 263 observations on the following 
19 variables.
</p>

<dl>
<dt><code>gene</code></dt><dd><p>a factor with 263 DE genes</p>
</dd>
<dt><code>A.2_S35S</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A.4_S36S</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A.42_S45S</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A.39_S41S</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A.9_S44R</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A.12_S40R</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A.18_S37R</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A.29_S39R</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A.31_S38R</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>A.38_S34R</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>tvalue</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rho</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>pvalue</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>adjp</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>w</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>order</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FDR</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>significance</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>This dataset is a demo dataset having 263 DE genes 
identifiedby NBBttest. The dataset was created from 
types of two tomur cells: sensitive to drug(S) and 
resistant to drug (R). 4 sensitive cells and 6 
resistant cells are available for differential 
analysis. Tvalue is t-statistic, <code class="reqn">\rho</code> is 
gene-wise variable, pvalue is p-value for t-test, 
<code class="reqn">\omega</code> is threshold for <code class="reqn">\rho</code>, FDR is 
false discovery rate, significance = 1 if pvalue &lt; FDR, 
0, otherwise. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(upGAm)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
