<!DOCTYPE html><html><head><title>Help for package NBPSeq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {NBPSeq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#NBPSeq-package'><p>Negative Binomial Regression Models for Statistical Analysis of RNA-Sequencing Data</p></a></li>
<li><a href='#arab'><p>Arabidopsis RNA-Seq Data Set</p></a></li>
<li><a href='#estimate.disp'><p>Fit a parametric disperison model to thinned counts</p></a></li>
<li><a href='#estimate.dispersion'><p>Estimate Negative Binomial Dispersion</p></a></li>
<li><a href='#estimate.norm.factors'><p>Estiamte Normalization Factors</p></a></li>
<li><a href='#exact.nb.test'><p>Exact Negative Binomial Test for Differential Gene Expression</p></a></li>
<li><a href='#irls.nb.1'><p>Estimate the regression coefficients in an NB GLM model</p></a></li>
<li><a href='#nb.glm.test'><p>Fit Negative Binomial Regression Model and Test for a Regression Coefficient</p></a></li>
<li><a href='#nbp.test'><p>NBP Test for Differential Gene Expression from RNA-Seq</p>
Counts</a></li>
<li><a href='#plot.nb.data'><p>Boxplot and scatterplot matrix of relative frequencies (after normalization)</p></a></li>
<li><a href='#plot.nb.dispersion'><p>Plot the estimated dispersion as a function of the</p>
preliminarily estimated mean relative frequencies</a></li>
<li><a href='#plot.nbp'><p>Diagnostic Plots for an NBP Object</p></a></li>
<li><a href='#prepare.nb.data'><p>Prepare the NB Data Structure for RNA-Seq Read Counts</p></a></li>
<li><a href='#prepare.nbp'><p>Prepare the Data Structure for Exact NB test for Two-Group Comparison</p></a></li>
<li><a href='#print.nb.data'><p>Print summary of the nb counts</p></a></li>
<li><a href='#print.nb.dispersion'><p>Print the estimated dispersion model</p></a></li>
<li><a href='#print.nb.test'><p>Print output from <code>test.coefficient</code></p></a></li>
<li><a href='#print.nbp'><p>Print summary of an NBP Object</p></a></li>
<li><a href='#test.coefficient'><p>Large-sample Test for a Regression Coefficient in an</p>
Negative Binomial Regression Model</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-06-09</td>
</tr>
<tr>
<td>Title:</td>
<td>Negative Binomial Models for RNA-Sequencing Data</td>
</tr>
<tr>
<td>Author:</td>
<td>Yanming Di &lt;diy@stat.oregonstate.edu&gt;, Daniel W Schafer
    &lt;schafer@stat.oregonstate.edu&gt;, with contributions from Jason S Cumbie
    &lt;cumbiej@onid.orst.edu&gt; and Jeff H Chang &lt;changj@cgrb.oregonstate.edu&gt;.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yanming Di &lt;diy@stat.oregonstate.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Negative Binomial (NB) models for two-group comparisons and
    regression inferences from RNA-Sequencing Data.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.00)</td>
</tr>
<tr>
<td>Imports:</td>
<td>splines, qvalue</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-09 09:15:23 UTC; panos</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-09 11:02:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='NBPSeq-package'>Negative Binomial Regression Models for Statistical Analysis of RNA-Sequencing Data</h2><span id='topic+NBPSeq'></span><span id='topic+NBPSeq-package'></span>

<h3>Description</h3>

<p>Negative binomial (NB) two-group and regression models for
RNA-Sequencing data analysis.
</p>


<h3>Details</h3>

<p>See the examples of <code><a href="#topic+test.coefficient">test.coefficient</a></code> and
<code><a href="#topic+exact.nb.test">exact.nb.test</a></code> for typical workflows of using
this package.
</p>

<hr>
<h2 id='arab'>Arabidopsis RNA-Seq Data Set</h2><span id='topic+arab'></span>

<h3>Description</h3>

<p>An RNA-Seq dataset from a pilot study of the defense
response of Arabidopsis to infection by bacteria. We
performed RNA-Seq experiments on three independent
biological samples from each of the two treatment groups.
The matrix contains the frequencies of RNA-Seq reads mapped
to genes in a reference database. Rows correspond to genes
and columns correspond to independent biological samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(arab)
</code></pre>


<h3>Format</h3>

<p>A 26222 by 6 matrix of RNA-Seq read frequencies.</p>


<h3>Details</h3>

<p>We challenged leaves of Arabidopsis with the
defense-eliciting <em><code class="reqn">\Delta</code>hrcC</em> mutant of
<em>Pseudomonas syringae</em> pathovar <em>tomato</em> DC3000.
We also infiltrated leaves of Arabidopsis with 10mM MgCl2
as a mock inoculation. RNA was isolated 7 hours after
inoculation, enriched for mRNA and prepared for RNA-Seq. We
sequenced one replicate per channel on the Illumina Genome
Analyzer (http://www.illumina.com).  The length of the
RNA-Seq reads can vary in length depending on user
preference and the sequencing instrument. The dataset used
here are derived from a 36-cycle sequencing reaction, that
we trimmed to 25mers. We used an in-house computational
pipeline to process, align, and assign RNA-Seq reads to
genes according to a reference database we developed for
Arabidopsis.
</p>


<h3>Author(s)</h3>

<p>Jason S Cumbie <a href="mailto:cumbiej@onid.orst.edu">cumbiej@onid.orst.edu</a> and Jeff H
Chang <a href="mailto:changj@cgrb.oregonstate.edu">changj@cgrb.oregonstate.edu</a>.
</p>


<h3>References</h3>

<p>Di Y, Schafer DW, Cumbie JS, and Chang JH (2011): &quot;The NBP
Negative Binomial Model for Assessing Differential Gene
Expression from RNA-Seq&quot;, Statistical Applications in
Genetics and Molecular Biology, 10 (1).
</p>

<hr>
<h2 id='estimate.disp'>Fit a parametric disperison model to thinned counts</h2><span id='topic+estimate.disp'></span>

<h3>Description</h3>

<p>Fit a parametric dispersion model to RNA-Seq counts data
prepared by <code><a href="#topic+prepare.nbp">prepare.nbp</a></code>. The model parameters
are estimated from the pseudo counts: thinned/down-sampled
counts that have the same effective library size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate.disp(obj, model = "NBQ", print.level = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate.disp_+3A_obj">obj</code></td>
<td>
<p>output from <code><a href="#topic+prepare.nbp">prepare.nbp</a></code>.</p>
</td></tr>
<tr><td><code id="estimate.disp_+3A_model">model</code></td>
<td>
<p>a string, one of &quot;NBQ&quot; (default), &quot;NBP&quot; or
&quot;NB2&quot;.</p>
</td></tr>
<tr><td><code id="estimate.disp_+3A_print.level">print.level</code></td>
<td>
<p>a number, controls the amount of
messages printed: 0 for suppressing all messages, 1 for
basic progress messages, larger values for more detailed
messages.</p>
</td></tr>
<tr><td><code id="estimate.disp_+3A_...">...</code></td>
<td>
<p>additional parameters controlling the
estimation of the parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each individual gene <code class="reqn">i</code>, a negative binomial (NB)
distribution uses a dispersion parameter <code class="reqn">\phi_i</code> to
capture the extra-Poisson variation between biological
replicates: the NB model imposes a mean-variance
relationship <code class="reqn">\sigma_i^2 = \mu_i + \phi_i \mu_i^2</code>.  In
many RNA-Seq data sets, the dispersion parameter
<code class="reqn">\phi_i</code> tends to vary with the mean <code class="reqn">\mu_i</code>. We
proposed to capture the dispersion-mean dependence using
parametric models.
</p>
<p>With this function, <code>estimate.disp</code>, users can choose
from three parametric models: NB2, NBP and NBQ (default).
</p>
<p>Under the NB2 model, the dispersion parameter is a constant
and does not vary with the mean expression levels.
</p>
<p>Under the NBP model, the log dispersion is modeled as a
linear function of preliminarily estimated log mean
relative frequencies (<code>pi.pre</code>):
</p>
<p>log(phi) = par[1] + par[2] * log(pi.pre/pi.offset),
</p>
<p>Under the NBQ model, the log dispersion is modeled as a
quadratic function of preliminarily estimated log mean
relative frequencies (<code>pi.pre</code>):
</p>
<p>log(phi) = par[1] + par[2] * log(pi.pre/pi.offset) + par[3]
* (log(pi.pre/pi.offset))^2;
</p>
<p>The NBQ model is more flexible than the NBP and NB2 models,
and is the current default option.
</p>
<p>In the NBP and NBQ models, <code>pi.offset</code> is fixed to be
1e-4, so par[1] corresponds to the dispersion level when
the relative mean frequency is 100 reads per million (RPM).
</p>
<p>The dispersion parameters are estimated from the pseudo
counts (counts adjusted to have the same effective library
sizes).  The parameters are estimated by maximizing the log
conditional likelihood of the model parameters given the
row sums. The log conditional likelihood is computed for
each gene in each treatment group and then summed over
genes and treatment groups.
</p>


<h3>Value</h3>

<p>The list <code>obj</code> from the input with some added
components summarizing the fitted dispersion model.  Users
can print and plot the output to see brief summaries of the
fitted dispersion model. The output is otherwise not
intended for use by end users directly.
</p>


<h3>Note</h3>

<p>Users should call <code><a href="#topic+prepare.nbp">prepare.nbp</a></code> before calling
this function. The function <code><a href="#topic+prepare.nbp">prepare.nbp</a></code> will
normalize the counts and adjust the counts so that the
effective library sizes are approximately the same
(computing the conditional likelihood requires the library
sizes to be the same).
</p>


<h3>References</h3>

<p>Di Y, Schafer DW, Cumbie JS, and Chang JH (2011): &quot;The NBP
Negative Binomial Model for Assessing Differential Gene
Expression from RNA-Seq&quot;, Statistical Applications in
Genetics and Molecular Biology, 10 (1).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nbp.test">nbp.test</a></code>, <code><a href="#topic+exact.nb.test">exact.nb.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## See the example for nb.exact.test
</code></pre>

<hr>
<h2 id='estimate.dispersion'>Estimate Negative Binomial Dispersion</h2><span id='topic+estimate.dispersion'></span>

<h3>Description</h3>

<p>Estimate NB dispersion by modeling it as a parametric
function of preliminarily estimated log mean relative
frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate.dispersion(nb.data, x, model = "NBQ", method = "MAPL", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate.dispersion_+3A_nb.data">nb.data</code></td>
<td>
<p>output from
<code><a href="#topic+prepare.nb.data">prepare.nb.data</a></code>.</p>
</td></tr>
<tr><td><code id="estimate.dispersion_+3A_x">x</code></td>
<td>
<p>a design matrix specifying the mean structure of
each row.</p>
</td></tr>
<tr><td><code id="estimate.dispersion_+3A_model">model</code></td>
<td>
<p>the name of the dispersion model, one of
&quot;NB2&quot;, &quot;NBP&quot;, &quot;NBQ&quot; (default), &quot;NBS&quot; or &quot;step&quot;.</p>
</td></tr>
<tr><td><code id="estimate.dispersion_+3A_method">method</code></td>
<td>
<p>a character string specifying the method
for estimating the dispersion model, one of &quot;ML&quot; or
&quot;MAPL&quot; (default).</p>
</td></tr>
<tr><td><code id="estimate.dispersion_+3A_...">...</code></td>
<td>
<p>(for future use).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We use a negative binomial (NB) distribution to model the
read frequency of gene <code class="reqn">i</code> in sample <code class="reqn">j</code>.  A
negative binomial (NB) distribution uses a dispersion
parameter <code class="reqn">\phi_{ij}</code> to model the extra-Poisson
variation between biological replicates. Under the NB
model, the mean-variance relationship of a single read
count satisfies <code class="reqn">\sigma_{ij}^2 = \mu_{ij} + \phi_{ij}
\mu_{ij}^2</code>.  Due to the typically small sample sizes of
RNA-Seq experiments, estimating the NB dispersion
<code class="reqn">\phi_{ij}</code> for each gene <code class="reqn">i</code> separately is not
reliable.  One can pool information across genes and
biological samples by modeling <code class="reqn">\phi_{ij}</code> as a
function of the mean frequencies and library sizes.
</p>
<p>Under the NB2 model, the dispersion is a constant across
all genes and samples.
</p>
<p>Under the NBP model, the log dispersion is modeled as a
linear function of the preliminary estimates of the log
mean relative frequencies (<code>pi.pre</code>):
</p>
<p>log(phi) = par[1] + par[2] * log(pi.pre/pi.offset),
</p>
<p>where <code>pi.offset</code> is 1e-4.
</p>
<p>Under the NBQ model, the dispersion is modeled as a
quadratic function of the preliminary estimates of the log
mean relative frequencies (pi.pre):
</p>
<p>log(phi) = par[1] + par[2] * z + par[3] * z^2,
</p>
<p>where z = log(pi.pre/pi.offset). By default, pi.offset is
the median of pi.pre[subset,].
</p>
<p>Under this NBS model, the dispersion is modeled as a smooth
function (a natural cubic spline function) of the
preliminary estimates of the log mean relative frequencies
(pi.pre).
</p>
<p>Under the &quot;step&quot; model, the dispersion is modeled as a step
(piecewise constant) function.
</p>


<h3>Value</h3>

<p>a list with following components:
</p>
<table>
<tr><td><code>estimates</code></td>
<td>
<p>dispersion estimates for each read count,
a matrix of the same dimensions as the <code>counts</code> matrix
in <code>nb.data</code>.</p>
</td></tr> <tr><td><code>likelihood</code></td>
<td>
<p>the likelihood of the
fitted model.</p>
</td></tr> <tr><td><code>model</code></td>
<td>
<p>details of the estimate
dispersion model, NOT intended for use by end users. The
name and contents of this component are subject to change
in future versions.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Currently, it is unclear whether a dispersion-modeling
approach will outperform a more basic approach where
regression model is fitted to each gene separately without
considering the dispersion-mean dependence. Clarifying the
power-robustness of the dispersion-modeling approach is an
ongoing research topic.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## See the example for test.coefficient.
</code></pre>

<hr>
<h2 id='estimate.norm.factors'>Estiamte Normalization Factors</h2><span id='topic+estimate.norm.factors'></span>

<h3>Description</h3>

<p><code>estimate.norm.factors</code> estiamtes normalization
factors to account for apparent reduction or increase in
relative frequencies of non-differentially expressing genes
as a result of compensating the increased or decreased
relative frequencies of truly differentially expressing
genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate.norm.factors(counts, lib.sizes = colSums(counts),
  method = "AH2010")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate.norm.factors_+3A_counts">counts</code></td>
<td>
<p>a matrix of RNA-Seq read counts with rows
corresponding to gene features and columns corresponding
to independent biological samples.</p>
</td></tr>
<tr><td><code id="estimate.norm.factors_+3A_lib.sizes">lib.sizes</code></td>
<td>
<p>a vector of observed library sizes,
usually and by default estimated by column totals.</p>
</td></tr>
<tr><td><code id="estimate.norm.factors_+3A_method">method</code></td>
<td>
<p>a character string specifying the method
for normalization, currenlty, can be NULL or &quot;AH2010&quot;. If
method=NULL, the normalization factors will have values
of 1 (i.e., no normalization is applied); if
method=&quot;AH2010&quot; (default), the normalization method
proposed by Anders and Huber (2010) will be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We take gene expression to be indicated by relative
frequency of RNA-Seq reads mapped to a gene, relative to
library sizes (column sums of the count matrix). Since the
relative frequencies sum to 1 in each library (one column
of the count matrix), the increased relative frequencies of
truly over expressed genes in each column must be
accompanied by decreased relative frequencies of other
genes, even when those others do not truly differentially
express. If not accounted for, this may give a false
impression of biological relevance (see, e.g., Robinson and
Oshlack (2010), for some examples.)  A simple fix is to
compute the relative frequencies relative to effective
library sizes&mdash;library sizes multiplied by normalization
factors.
</p>


<h3>Value</h3>

<p>a vector of normalization factors.
</p>


<h3>References</h3>

<p>Anders, S. and W. Huber (2010): &quot;Differential expression
analysis for sequence count data,&quot; Genome Biol., 11, R106.
</p>
<p>Robinson, M. D. and A. Oshlack (2010): &quot;A scaling
normalization method for differential expression analysis
of RNA-seq data,&quot; Genome Biol., 11, R25.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load Arabidopsis data
data(arab)

## Estimate normalization factors using the method of Anders and Huber (2010)
norm.factors = estimate.norm.factors(arab);
print(norm.factors);
</code></pre>

<hr>
<h2 id='exact.nb.test'>Exact Negative Binomial Test for Differential Gene Expression</h2><span id='topic+exact.nb.test'></span>

<h3>Description</h3>

<p><code>exact.nb.test</code> performs the Robinson and Smyth exact
negative binomial (NB) test for differential gene
expression on each gene and summarizes the results using
p-values and q-values (FDR).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exact.nb.test(obj, grp1, grp2, print.level = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exact.nb.test_+3A_obj">obj</code></td>
<td>
<p>output from <code><a href="#topic+estimate.disp">estimate.disp</a></code>.</p>
</td></tr>
<tr><td><code id="exact.nb.test_+3A_grp1">grp1</code></td>
<td>
<p>Identifier of group 1. A number, character or
string (should match at least one of the obj$grp.ids).</p>
</td></tr>
<tr><td><code id="exact.nb.test_+3A_grp2">grp2</code></td>
<td>
<p>Identifier of group 2. A number, character or
string (should match at least one of the obj$grp.ids).</p>
</td></tr>
<tr><td><code id="exact.nb.test_+3A_print.level">print.level</code></td>
<td>
<p>a number.  Controls the amount of
messages printed: 0 for suppressing all messages, 1 for
basic progress messages, larger values for more detailed
messages.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The negative binomial (NB) distribution offers a more
realistic model for RNA-Seq count variability and still
permits an exact (non-asymptotic) test for comparing
expression levels in two groups.
</p>
<p>For each gene, let <code class="reqn">S_1</code>, <code class="reqn">S_2</code> be the sums of gene
counts from all biological replicates in each group. The
exact NB test is based on the conditional distribution of
<code class="reqn">S_1|S_1+S_2</code>: a value of <code class="reqn">S_1</code> that is too big or
too small, relative to the sum <code class="reqn">S_1+S_2</code>, indicates
evidence for differential gene expression.  When the
effective library sizes are the same in all replicates and
the dispersion parameters are known, we can determine the
probability functions of <code class="reqn">S_1</code>, <code class="reqn">S_2</code> explicitly.
The exact p-value is computed as the total conditional
probability of all possible values of <code class="reqn">(S_1, S_2)</code> that
have the same sum as but are more extreme than the observed
values of <code class="reqn">(S_1, S_2)</code>.
</p>
<p>Note that we assume that the NB dispersion parameters for
the two groups are the same and library sizes (column
totals of the count matrix) are the same.
</p>


<h3>Value</h3>

<p>the list <code>obj</code> from the input with the following added
components: </p>
<table>
<tr><td><code>grp1</code></td>
<td>
<p>same as input.</p>
</td></tr> <tr><td><code>grp2</code></td>
<td>
<p>same as
input.</p>
</td></tr> <tr><td><code>pooled.pie</code></td>
<td>
<p>estimated pooled mean of relative
count frequencies in the two groups being compared.</p>
</td></tr>
<tr><td><code>expression.levels</code></td>
<td>
<p>a matrix of estimated gene
expression levels as indicated by mean relative read
frequencies. It has three columns <code>grp1</code>, <code>grp2</code>,
<code>pooled</code> corresponding to the two treatment groups and
the pooled mean.</p>
</td></tr> <tr><td><code>log.fc</code></td>
<td>
<p>base 2 log fold change in
mean relative frequency between two groups.</p>
</td></tr>
<tr><td><code>p.values</code></td>
<td>
<p>p-values of the exact NB test applied to
each gene (row).</p>
</td></tr> <tr><td><code>q.values</code></td>
<td>
<p>q-values (estimated FDR)
corresponding to the p-values.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Before calling <code><a href="#topic+exact.nb.test">exact.nb.test</a></code>, the user should
call <code><a href="#topic+estimate.norm.factors">estimate.norm.factors</a></code> to estimate
normalization factors, call <code><a href="#topic+prepare.nbp">prepare.nbp</a></code> to
adjust library sizes, and call <code><a href="#topic+estimate.disp">estimate.disp</a></code>
to fit a dispersion model. The exact NB test will be
performed using <code>pseudo.counts</code> in the list
<code>obj</code>, which are normalized and adjusted to have the
same effective library sizes (column sums of the count
matrix, multiplied by normalization factors).
</p>
<p>Users not interested in fine tuning the underlying
statistical model should use <code><a href="#topic+nbp.test">nbp.test</a></code>
instead. The all-in-one function <code><a href="#topic+nbp.test">nbp.test</a></code>
uses sensible approaches to normalize the counts, estimate
the NBP model parameters and test for differential gene
expression.
</p>
<p>A test will be performed on a row (a gene) only when the
total row count is nonzero, otherwise an NA value will be
assigned to the corresponding p-value and q-value.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nbp.test">nbp.test</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load Arabidopsis data
data(arab);

## Specify treatment groups
## grp.ids = c(1, 1, 1, 2, 2, 2); # Numbers or strings are both OK
grp.ids = rep(c("mock", "hrcc"), each=3);

## Estimate normalization factors
norm.factors = estimate.norm.factors(arab);
print(norm.factors);

## Prepare an NBP object, adjust the library sizes by thinning the
## counts. For demonstration purpose, only use the first 100 rows of
## the arab data.
set.seed(999);
obj = prepare.nbp(arab[1:100,], grp.ids, lib.size=colSums(arab), norm.factors=norm.factors);
print(obj);

## Fit a dispersion model (NBQ by default)
obj = estimate.disp(obj);
plot(obj);

## Perform exact NB test
## grp1 = 1;
## grp2 = 2;
grp1 = "mock";
grp2 = "hrcc";

obj = exact.nb.test(obj, grp1, grp2);

## Print and plot results
print(obj);
par(mfrow=c(3,2));
plot(obj);
</code></pre>

<hr>
<h2 id='irls.nb.1'>Estimate the regression coefficients in an NB GLM model</h2><span id='topic+irls.nb.1'></span>

<h3>Description</h3>

<p>Estimate the regression coefficients in an NB GLM model
with known dispersion parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>irls.nb.1(y, s, x, phi, beta0 = rep(NA, p), mustart = NULL, maxit = 50,
  tol.mu = 0.001/length(y), print.level = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="irls.nb.1_+3A_y">y</code></td>
<td>
<p>an n vector of counts</p>
</td></tr>
<tr><td><code id="irls.nb.1_+3A_s">s</code></td>
<td>
<p>a scalar or an n vector of effective library
sizes</p>
</td></tr>
<tr><td><code id="irls.nb.1_+3A_x">x</code></td>
<td>
<p>an n by p design matrix</p>
</td></tr>
<tr><td><code id="irls.nb.1_+3A_phi">phi</code></td>
<td>
<p>a scalar or an n-vector of dispersion
parameters</p>
</td></tr>
<tr><td><code id="irls.nb.1_+3A_mustart">mustart</code></td>
<td>
<p>starting values for the vector of means</p>
</td></tr>
<tr><td><code id="irls.nb.1_+3A_beta0">beta0</code></td>
<td>
<p>a vector specifying known and unknown
components of the regression coefficients: non-NA
components are hypothesized values of beta, NA components
are free components</p>
</td></tr>
<tr><td><code id="irls.nb.1_+3A_maxit">maxit</code></td>
<td>
<p>maximum number of iterations</p>
</td></tr>
<tr><td><code id="irls.nb.1_+3A_tol.mu">tol.mu</code></td>
<td>
<p>a number, convergence criteria</p>
</td></tr>
<tr><td><code id="irls.nb.1_+3A_print.level">print.level</code></td>
<td>
<p>a number, print level</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function estimates the regression coefficients using
iterative reweighted least squares (IRLS) algorithm, which
is equivalent to Fisher scoring. The implementation is
based on <code>glm.fit</code>.
</p>
<p>Users can choose to fix some regression coefficients by
specifying <code>beta0</code>. (This is useful when fitting a
model under a null hypothesis.)
</p>


<h3>Value</h3>

<p>a list of the following components: </p>
<table>
<tr><td><code>beta</code></td>
<td>
<p>a p-vector
of estimated regression coefficients</p>
</td></tr> <tr><td><code>mu</code></td>
<td>
<p>an n-vector
of estimated mean values</p>
</td></tr> <tr><td><code>conv</code></td>
<td>
<p>logical. Was the IRLS
algorithm judged to have converged?</p>
</td></tr> <tr><td><code>zero</code></td>
<td>
<p>logical.
Was any of the fitted mean close to 0?</p>
</td></tr>
</table>

<hr>
<h2 id='nb.glm.test'>Fit Negative Binomial Regression Model and Test for a Regression Coefficient</h2><span id='topic+nb.glm.test'></span>

<h3>Description</h3>

<p>For each row of the input data matrix, <code>nb.glm.test</code>
fits an NB log-linear regression model and performs
large-sample tests for a one-dimensional regression
coefficient.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nb.glm.test(counts, x, beta0, lib.sizes = colSums(counts),
  normalization.method = "AH2010", dispersion.model = "NBQ",
  tests = c("HOA", "LR", "Wald"), alternative = "two.sided",
  subset = 1:dim(counts)[1])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nb.glm.test_+3A_counts">counts</code></td>
<td>
<p>an m by n matrix of RNA-Seq read counts
with rows corresponding to gene features and columns
corresponding to independent biological samples.</p>
</td></tr>
<tr><td><code id="nb.glm.test_+3A_x">x</code></td>
<td>
<p>an n by p design matrix specifying the treatment
structure.</p>
</td></tr>
<tr><td><code id="nb.glm.test_+3A_beta0">beta0</code></td>
<td>
<p>a p-vector specifying the null hypothesis.
Non-NA components specify the parameters to test and
their null values.</p>
</td></tr>
<tr><td><code id="nb.glm.test_+3A_lib.sizes">lib.sizes</code></td>
<td>
<p>a p-vector of observed library sizes,
usually (and by default) estimated by column totals.</p>
</td></tr>
<tr><td><code id="nb.glm.test_+3A_normalization.method">normalization.method</code></td>
<td>
<p>a character string specifying
the method for estimating the normalization factors, can
be <code>NULL</code> or <code>"AH2010"</code>. If <code>method=NULL</code>,
the normalization factors will have values of 1 (i.e., no
normalization is applied); if <code>method="AH2010"</code>, the
normalization method proposed by Anders and Huber (2010)
will be used.</p>
</td></tr>
<tr><td><code id="nb.glm.test_+3A_dispersion.model">dispersion.model</code></td>
<td>
<p>a character string specifying the
dispersion model, and can be one of &quot;NB2&quot;, &quot;NBP&quot;, &quot;NBQ&quot;
(default), &quot;NBS&quot; or &quot;step&quot;.</p>
</td></tr>
<tr><td><code id="nb.glm.test_+3A_tests">tests</code></td>
<td>
<p>a character string vector specifying the
tests to be performed, can be any subset of <code>"HOA"</code>
(higher-order asymptotic test), <code>"LR"</code> (likelihood
ratio test), and <code>"Wald"</code> (Wald test).</p>
</td></tr>
<tr><td><code id="nb.glm.test_+3A_alternative">alternative</code></td>
<td>
<p>a character string specifying the
alternative hypothesis, must be one of <code>"two.sided"</code>
(default), <code>"greater"</code> or <code>"less"</code>.</p>
</td></tr>
<tr><td><code id="nb.glm.test_+3A_subset">subset</code></td>
<td>
<p>specify a subset of rows to perform the
test on</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>nbp.glm.test</code> provides a simple, one-stop interface
to performing a series of core tasks in regression analysis
of RNA-Seq data: it calls
<code><a href="#topic+estimate.norm.factors">estimate.norm.factors</a></code> to estimate
normalization factors; it calls
<code><a href="#topic+prepare.nb.data">prepare.nb.data</a></code> to create an NB data
structure; it calls <code><a href="#topic+estimate.dispersion">estimate.dispersion</a></code> to
estimate the NB dispersion; and it calls
<code><a href="#topic+test.coefficient">test.coefficient</a></code> to test the regression
coefficient.
</p>
<p>To keep the interface simple, <code>nbp.glm.test</code> provides
limited options for fine tuning models/parameters in each
individual step. For more control over individual steps,
advanced users can call
<code><a href="#topic+estimate.norm.factors">estimate.norm.factors</a></code>,
<code><a href="#topic+prepare.nb.data">prepare.nb.data</a></code>,
<code><a href="#topic+estimate.dispersion">estimate.dispersion</a></code>, and
<code><a href="#topic+test.coefficient">test.coefficient</a></code> directly, or even substitute
one or more of them with their own versions.
</p>


<h3>Value</h3>

<p>A list containing the following components: </p>
<table>
<tr><td><code>data</code></td>
<td>
<p>a
list containing the input data matrix with additional
summary quantities, output from
<code><a href="#topic+prepare.nb.data">prepare.nb.data</a></code>.</p>
</td></tr>
<tr><td><code>dispersion</code></td>
<td>
<p>dispersion estimates and models, output
from <code><a href="#topic+estimate.dispersion">estimate.dispersion</a></code>.</p>
</td></tr> <tr><td><code>test</code></td>
<td>
<p>test
results, output from <code><a href="#topic+test.coefficient">test.coefficient</a></code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Load Arabidopsis data
data(arab);

## Specify treatment structure
grp.ids = as.factor(c(1, 1, 1, 2, 2, 2));
x = model.matrix(~grp.ids);

## Specify the null hypothesis
## The null hypothesis is beta[1]=0 (beta[1] is the log fold change).
beta0 = c(NA, 0);

## Fit NB regression model and perform large sample tests.
## The step can take long if the number of genes is large
fit = nb.glm.test(arab, x, beta0, subset=1:50);

## The result contains the data, the dispersion estimates and the test results
print(str(fit));

## Show HOA test results for top ten genes
subset = order(fit$test.results$HOA$p.values)[1:10];
cbind(fit$data$counts[subset,], fit$test.results$HOA[subset,]);

## Show LR test results
subset = order(fit$test.results$LR$p.values)[1:10];
cbind(fit$data$counts[subset,], fit$test.results$LR[subset,]);

</code></pre>

<hr>
<h2 id='nbp.test'>NBP Test for Differential Gene Expression from RNA-Seq
Counts</h2><span id='topic+nbp.test'></span>

<h3>Description</h3>

<p><code>nbp.test</code> fits an NBP model to the RNA-Seq counts and
performs Robinson and Smyth's exact NB test on each gene to
assess differential gene expression between two groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nbp.test(counts, grp.ids, grp1, grp2, norm.factors = rep(1, dim(counts)[2]),
  model.disp = "NBQ", lib.sizes = colSums(counts), print.level = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nbp.test_+3A_counts">counts</code></td>
<td>
<p>an <code class="reqn">n</code> by <code class="reqn">r</code> matrix of RNA-Seq
read counts with rows corresponding to genes (exons, gene
isoforms, etc) and columns corresponding to libraries
(independent biological samples).</p>
</td></tr>
<tr><td><code id="nbp.test_+3A_grp.ids">grp.ids</code></td>
<td>
<p>an <code class="reqn">r</code> vector of treatment group
identifiers (e.g. integers).</p>
</td></tr>
<tr><td><code id="nbp.test_+3A_grp1">grp1</code></td>
<td>
<p>group 1 id</p>
</td></tr>
<tr><td><code id="nbp.test_+3A_grp2">grp2</code></td>
<td>
<p>group 2 id</p>
</td></tr>
<tr><td><code id="nbp.test_+3A_norm.factors">norm.factors</code></td>
<td>
<p>an <code class="reqn">r</code> vector of normalization
factors.</p>
</td></tr>
<tr><td><code id="nbp.test_+3A_model.disp">model.disp</code></td>
<td>
<p>a string, one of &quot;NB2&quot;, &quot;NBP&quot; or &quot;NBQ&quot;
(default).</p>
</td></tr>
<tr><td><code id="nbp.test_+3A_lib.sizes">lib.sizes</code></td>
<td>
<p>(unnormalized) library sizes</p>
</td></tr>
<tr><td><code id="nbp.test_+3A_print.level">print.level</code></td>
<td>
<p>a number, controls the amount of
messages printed: 0 for suppressing all messages, 1
(default) for basic progress messages, and 2 to 5 for
increasingly more detailed messages.</p>
</td></tr>
<tr><td><code id="nbp.test_+3A_...">...</code></td>
<td>
<p>optional parameters to be passed to
<code><a href="#topic+estimate.disp">estimate.disp</a></code>, the function that estimates
the dispersion parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>nbp.test</code> calls <code><a href="#topic+prepare.nbp">prepare.nbp</a></code> to create
the NBP data structure, perform optional normalization and
adjust library sizes, calls <code><a href="#topic+estimate.disp">estimate.disp</a></code> to
estimate the NBP dispersion parameters and
<code><a href="#topic+exact.nb.test">exact.nb.test</a></code> to perform the exact NB test
for differential gene expression on each gene. The results
are summarized using p-values and q-values (FDR).
</p>


<h4>Overview</h4>

<p> For assessing evidence for
differential gene expression from RNA-Seq read counts, it
is critical to adequately model the count variability
between independent biological replicates.  Negative
binomial (NB) distribution offers a more realistic model
for RNA-Seq count variability than Poisson distribution and
still permits an exact (non-asymptotic) test for comparing
two groups.
</p>
<p>For each individual gene, an NB distribution uses a
dispersion parameter <code class="reqn">\phi_i</code> to model the
extra-Poisson variation between biological replicates.
Across all genes, parameter <code class="reqn">\phi_i</code> tends to vary with
the mean <code class="reqn">\mu_i</code>. We capture the dispersion-mean
dependence using a parametric model: NB2, NBP and NBQ. (See
<code><a href="#topic+estimate.disp">estimate.disp</a></code> for more details.)</p>



<h4>Count Normalization</h4>

<p> We take gene expression
to be indicated by relative frequency of RNA-Seq reads
mapped to a gene, relative to library sizes (column sums of
the count matrix). Since the relative frequencies sum to 1
in each library (one column of the count matrix), the
increased relative frequencies of truly over expressed
genes in each column must be accompanied by decreased
relative frequencies of other genes, even when those others
do not truly differentially express. Robinson and Oshlack
(2010) presented examples where this problem is noticeable.
</p>
<p>A simple fix is to compute the relative frequencies
relative to effective library sizes&mdash;library sizes
multiplied by normalization factors.  By default,
<code>nbp.test</code> assumes the normalization factors are 1
(i.e. no normalization is needed). Users can specify
normalization factors through the argument
<code>norm.factors</code>.  Many authors (Robinson and Oshlack
(2010), Anders and Huber (2010)) propose to estimate the
normalization factors based on the assumption that most
genes are NOT differentially expressed. </p>



<h4>Library Size Adjustment</h4>

<p> The exact test
requires that the effective library sizes (column sums of
the count matrix multiplied by normalization factors) are
approximately equal. By default, <code>nbp.test</code> will thin
(downsample) the counts to make the effective library sizes
equal. Thinning may lose statistical efficiency, but is
unlikely to introduce bias.</p>



<h3>Value</h3>

<p>a list with the following components: </p>
<table>
<tr><td><code>counts</code></td>
<td>
<p>an
<code class="reqn">n</code> by <code class="reqn">r</code> matrix of counts, same as input.</p>
</td></tr>
<tr><td><code>lib.sizes</code></td>
<td>
<p>an <code class="reqn">r</code> vector, column sums of the
count matrix.</p>
</td></tr> <tr><td><code>grp.ids</code></td>
<td>
<p>an <code class="reqn">r</code> vector,
identifiers of treatment groups, same as input.</p>
</td></tr>
<tr><td><code>grp1</code>, <code>grp2</code></td>
<td>
<p>identifiers of the two groups to be
compared, same as input.</p>
</td></tr> <tr><td><code>eff.lib.sizes</code></td>
<td>
<p>an <code class="reqn">r</code>
vector, effective library sizes, lib.sizes multiplied by
the normalization factors.</p>
</td></tr> <tr><td><code>pseudo.counts</code></td>
<td>
<p>count
matrix after thinning, same dimension as counts</p>
</td></tr>
<tr><td><code>pseduo.lib.sizes</code></td>
<td>
<p>an <code class="reqn">r</code> vector, effective
library sizes of pseudo counts, i.e., column sums of the
pseudo count matrix multiplied by the normalization.</p>
</td></tr>
<tr><td><code>phi</code>, <code>alpha</code></td>
<td>
<p>two numbers, parameters of the dispersion
model.</p>
</td></tr> <tr><td><code>pie</code></td>
<td>
<p>a matrix, same dimension as
<code>counts</code>, estimated mean relative frequencies of
RNA-Seq reads mapped to each gene.</p>
</td></tr> <tr><td><code>pooled.pie</code></td>
<td>
<p>a
matrix, same dimenions as <code>counts</code>, estimated pooled
mean of relative frequencies in the two groups being
compared.</p>
</td></tr> <tr><td><code>expression.levels</code></td>
<td>
<p>a <code class="reqn">n</code> by 3 matrix,
estimated gene expression levels as indicated by mean
relative frequencies of RNA-Seq reads. It has three columns
<code>grp1</code>, <code>grp2</code>, <code>pooled</code> corresponding to
the two treatment groups and the pooled mean.</p>
</td></tr>
<tr><td><code>log.fc</code></td>
<td>
<p>an <code class="reqn">n</code>-vector, base 2 log fold change in
mean relative frequency between two groups.</p>
</td></tr>
<tr><td><code>p.values</code></td>
<td>
<p>an <code class="reqn">n</code>-vector, p-values of the exact NB
test applied to each gene (row).</p>
</td></tr> <tr><td><code>q.values</code></td>
<td>
<p>an
<code class="reqn">n</code>-vector, q-values (estimated FDR) corresponding to
the p-values.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Due to thinning (random downsampling of counts), two
identical calls to <code>nbp.test</code> may yield slightly
different results. A random number seed can be used to make
the results reproducible. The regression analysis method
implemented in <code><a href="#topic+nb.glm.test">nb.glm.test</a></code> does not require
thinning and can also be used to compare expression in two
groups.
</p>
<p>Advanced users can call
<code><a href="#topic+estimate.norm.factors">estimate.norm.factors</a></code>,
<code><a href="#topic+prepare.nbp">prepare.nbp</a></code>, <code><a href="#topic+estimate.disp">estimate.disp</a></code>,
<code><a href="#topic+exact.nb.test">exact.nb.test</a></code> directly to have more control
over modeling and testing.
</p>


<h3>References</h3>

<p>Di, Y, D. W. Schafer, J. S. Cumbie, and J. H. Chang (2011):
&quot;The NBP Negative Binomial Model for Assessing Differential
Gene Expression from RNA-Seq&quot;, Statistical Applications in
Genetics and Molecular Biology, 10 (1).
</p>
<p>Robinson, M. D. and G. K. Smyth (2007): &quot;Moderated
statistical tests for assessing differences in tag
abundance,&quot; Bioinformatics, 23, 2881-2887.
</p>
<p>Robinson, M. D. and G. K. Smyth (2008): &quot;Small-sample
estimation of negative binomial dispersion, with
applications to SAGE data,&quot; Biostatistics, 9, 321-332.
</p>
<p>Anders, S. and W. Huber (2010): &quot;Differential expression
analysis for sequence count data,&quot; Genome Biol., 11, R106.
</p>
<p>Robinson, M. D. and A. Oshlack (2010): &quot;A scaling
normalization method for differential expression analysis
of RNA-seq data,&quot; Genome Biol., 11, R25.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+prepare.nbp">prepare.nbp</a></code>, <code><a href="#topic+estimate.disp">estimate.disp</a></code>,
<code><a href="#topic+exact.nb.test">exact.nb.test</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load Arabidopsis data
data(arab);

## Specify treatment groups and ids of the two groups to be compared
grp.ids = c(1, 1, 1, 2, 2, 2);
grp1 = 1;
grp2 = 2;

## Estimate normalization factors
norm.factors = estimate.norm.factors(arab);

## Set a random number seed to make results reproducible
set.seed(999);

## Fit the NBP model and perform exact NB test for differential gene expression.
## For demonstration purpose, we will use the first 100 rows of the arab data.
res = nbp.test(arab[1:100,], grp.ids, grp1, grp2,
  lib.sizes = colSums(arab), norm.factors = norm.factors, print.level=3);

## The argument lib.sizes is needed since we only use a subset of
## rows. If all rows are used, the following will be adequate:
##
## res = nbp.test(arab, grp.ids, grp1, grp2, norm.factors = norm.factors);

## Show top ten most differentially expressed genes
subset = order(res$p.values)[1:10];
print(res, subset);

## Count the number of differentially expressed genes (e.g. qvalue &lt; 0.05)
alpha = 0.05;
sig.res = res$q.values &lt; alpha;
table(sig.res);

## Show boxplots, MA-plot, mean-variance plot and mean-dispersion plot
par(mfrow=c(3,2));
plot(res);
</code></pre>

<hr>
<h2 id='plot.nb.data'>Boxplot and scatterplot matrix of relative frequencies (after normalization)</h2><span id='topic+plot.nb.data'></span>

<h3>Description</h3>

<p>Boxplot and scatterplot matrix of relative frequencies
(after normalization)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'nb.data'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.nb.data_+3A_x">x</code></td>
<td>
<p>output from <code><a href="#topic+prepare.nb.data">prepare.nb.data</a></code></p>
</td></tr>
<tr><td><code id="plot.nb.data_+3A_...">...</code></td>
<td>
<p>currently not used</p>
</td></tr>
</table>

<hr>
<h2 id='plot.nb.dispersion'>Plot the estimated dispersion as a function of the
preliminarily estimated mean relative frequencies</h2><span id='topic+plot.nb.dispersion'></span>

<h3>Description</h3>

<p>Plot the estimated dispersion as a function of the
preliminarily estimated mean relative frequencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'nb.dispersion'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.nb.dispersion_+3A_x">x</code></td>
<td>
<p>output from <code><a href="#topic+estimate.dispersion">estimate.dispersion</a></code></p>
</td></tr>
<tr><td><code id="plot.nb.dispersion_+3A_...">...</code></td>
<td>
<p>additional parameters, currently unused</p>
</td></tr>
</table>

<hr>
<h2 id='plot.nbp'>Diagnostic Plots for an NBP Object</h2><span id='topic+plot.nbp'></span>

<h3>Description</h3>

<p>For output from <code><a href="#topic+nbp.test">nbp.test</a></code>, produce a boxplot,
an MA plot, mean-variance plots (one for each group being
compared), and mean-dispersion plots (one for each group
being compared). On the mean-variance and the
mean-dispersion plots, overlay curves corresponding to the
estimated NBP model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'nbp'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.nbp_+3A_x">x</code></td>
<td>
<p>output from <code><a href="#topic+nbp.test">nbp.test</a></code>.</p>
</td></tr>
<tr><td><code id="plot.nbp_+3A_...">...</code></td>
<td>
<p>for future use</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+nbp.test">nbp.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## See the example for nbp.test
</code></pre>

<hr>
<h2 id='prepare.nb.data'>Prepare the NB Data Structure for RNA-Seq Read Counts</h2><span id='topic+prepare.nb.data'></span>

<h3>Description</h3>

<p>Create a data structure to hold the RNA-Seq read counts and
other relevant information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare.nb.data(counts, lib.sizes = colSums(counts), norm.factors = rep(1,
  dim(counts)[2]), tags = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare.nb.data_+3A_counts">counts</code></td>
<td>
<p>an mxn matrix of RNA-Seq read counts with
rows corresponding to gene features and columns
corresponding to independent biological samples.</p>
</td></tr>
<tr><td><code id="prepare.nb.data_+3A_lib.sizes">lib.sizes</code></td>
<td>
<p>an n-vector of observed library sizes.
By default, library sizes are estimated to the column
totals of the matrix <code>counts</code>.</p>
</td></tr>
<tr><td><code id="prepare.nb.data_+3A_norm.factors">norm.factors</code></td>
<td>
<p>an n-vector of normalization factors.
By default, have values 1 (no normalization is applied).</p>
</td></tr>
<tr><td><code id="prepare.nb.data_+3A_tags">tags</code></td>
<td>
<p>a matrix of tags associated with genes, one
row for each gene (having the same number of rows as
<code>counts</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<table>
<tr><td><code>counts</code></td>
<td>
<p>the count matrix, same as input.</p>
</td></tr>
<tr><td><code>lib.sizes</code></td>
<td>
<p>observed library sizes, same as input.</p>
</td></tr>
<tr><td><code>norm.factors</code></td>
<td>
<p>normalization factors, same as input.</p>
</td></tr>
<tr><td><code>eff.lib.sizes</code></td>
<td>
<p>effective library sizes
(<code>lib.sizes</code> x <code>norm.factors</code>).</p>
</td></tr>
<tr><td><code>rel.frequencies</code></td>
<td>
<p>relative frequencies (counts divided
by the effective library sizes).</p>
</td></tr> <tr><td><code>tags</code></td>
<td>
<p>a matrix of
gene tags, same as input.</p>
</td></tr>
</table>

<hr>
<h2 id='prepare.nbp'>Prepare the Data Structure for Exact NB test for Two-Group Comparison</h2><span id='topic+prepare.nbp'></span>

<h3>Description</h3>

<p>Create the NBP data structure, (optionally) normalize the
counts, and thin the counts to make the effective library
sizes equal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare.nbp(counts, grp.ids, lib.sizes = colSums(counts),
  norm.factors = NULL, thinning = TRUE, print.level = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare.nbp_+3A_counts">counts</code></td>
<td>
<p>an <code class="reqn">n</code> by <code class="reqn">r</code> matrix of RNA-Seq
read counts with rows corresponding to genes (exons, gene
isoforms, etc) and columns corresponding to libraries
(independent biological samples).</p>
</td></tr>
<tr><td><code id="prepare.nbp_+3A_grp.ids">grp.ids</code></td>
<td>
<p>an <code class="reqn">r</code> vector of treatment group
identifiers (can be a vector of integers, chars or
strings).</p>
</td></tr>
<tr><td><code id="prepare.nbp_+3A_lib.sizes">lib.sizes</code></td>
<td>
<p>library sizes, an <code class="reqn">r</code> vector of
numbers. By default, library sizes are estimated by
column sums.</p>
</td></tr>
<tr><td><code id="prepare.nbp_+3A_norm.factors">norm.factors</code></td>
<td>
<p>normalization factors, an <code class="reqn">r</code>
vector of numbers. If <code>NULL</code> (default), no
normalization will be applied.</p>
</td></tr>
<tr><td><code id="prepare.nbp_+3A_thinning">thinning</code></td>
<td>
<p>a boolean variable (i.e., logical). If
<code>TRUE</code> (default), the counts will be randomly down
sampled to make effective library sizes approximately
equal.</p>
</td></tr>
<tr><td><code id="prepare.nbp_+3A_print.level">print.level</code></td>
<td>
<p>a number, controls the amount of
messages printed: 0 for suppressing all messages, 1
(default) for basic progress messages, and 2 to 5 for
increasingly more detailed messages.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Normalization
</p>
<p>We take gene expression to be indicated by relative
frequency of RNA-Seq reads mapped to a gene, relative to
library sizes (column sums of the count matrix). Since the
relative frequencies sum to 1 in each library (one column
of the count matrix), the increased relative frequencies of
truly over expressed genes in each column must be
accompanied by decreased relative frequencies of other
genes, even when those others do not truly differently
express. Robinson and Oshlack (2010) presented examples
where this problem is noticeable.
</p>
<p>A simple fix is to compute the relative frequencies
relative to effective library sizes&mdash;library sizes
multiplied by normalization factors. Many authors (Robinson
and Oshlack (2010), Anders and Huber (2010)) propose to
estimate the normalization factors based on the assumption
that most genes are NOT differentially expressed.
</p>
<p>By default, <code>prepare.nbp</code> does not estimate the
normalization factors, but can incorporate user specified
normalization factors through the argument
<code>norm.factors</code>.
</p>
<p>Library Size Adjustment
</p>
<p>The exact test requires that the effective library sizes
(column sums of the count matrix multiplied by
normalization factors) are approximately equal. By default,
<code>prepare.nbp</code> will thin (downsample) the counts to
make the effective library sizes equal. Thinning may lose
statistical efficiency, but is unlikely to introduce bias.
</p>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<table>
<tr><td><code>counts</code></td>
<td>
<p>the count matrix, same as input.</p>
</td></tr>
<tr><td><code>lib.sizes</code></td>
<td>
<p>column sums of the count matrix.</p>
</td></tr>
<tr><td><code>grp.ids</code></td>
<td>
<p>a vector of identifiers of treatment groups,
same as input.</p>
</td></tr> <tr><td><code>eff.lib.sizes</code></td>
<td>
<p>effective library
sizes, lib.sizes multiplied by the normalization factors.</p>
</td></tr>
<tr><td><code>pseudo.counts</code></td>
<td>
<p>count matrix after thinning.</p>
</td></tr>
<tr><td><code>pseduo.lib.sizes</code></td>
<td>
<p>effective library sizes of pseudo
counts, i.e., column sums of the pseudo count matrix
multiplied by the normalization.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Due to thinning (random downsampling of counts), two
identical calls to <code>prepare.nbp</code> may yield slightly
different results. A random number seed can be used to make
the results reproducible.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+nbp.test">nbp.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## See the example for exact.nb.test
</code></pre>

<hr>
<h2 id='print.nb.data'>Print summary of the nb counts</h2><span id='topic+print.nb.data'></span>

<h3>Description</h3>

<p>Print summary of the nb counts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'nb.data'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.nb.data_+3A_x">x</code></td>
<td>
<p>output from <code><a href="#topic+prepare.nb.data">prepare.nb.data</a></code></p>
</td></tr>
<tr><td><code id="print.nb.data_+3A_...">...</code></td>
<td>
<p>additional parameters, currently not used</p>
</td></tr>
</table>

<hr>
<h2 id='print.nb.dispersion'>Print the estimated dispersion model</h2><span id='topic+print.nb.dispersion'></span>

<h3>Description</h3>

<p>Print the estimated dispersion model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'nb.dispersion'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.nb.dispersion_+3A_x">x</code></td>
<td>
<p>output from from
<code><a href="#topic+estimate.dispersion">estimate.dispersion</a></code></p>
</td></tr>
<tr><td><code id="print.nb.dispersion_+3A_...">...</code></td>
<td>
<p>additional parameters, currently unused</p>
</td></tr>
</table>

<hr>
<h2 id='print.nb.test'>Print output from <code><a href="#topic+test.coefficient">test.coefficient</a></code></h2><span id='topic+print.nb.test'></span>

<h3>Description</h3>

<p>We simply print out the structure of <code>x</code>. (Currenlty
the method is equivalent to <code>print(str(x))</code>.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'nb.test'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.nb.test_+3A_x">x</code></td>
<td>
<p>output from <code><a href="#topic+test.coefficient">test.coefficient</a></code></p>
</td></tr>
<tr><td><code id="print.nb.test_+3A_...">...</code></td>
<td>
<p>currenty not used</p>
</td></tr>
</table>

<hr>
<h2 id='print.nbp'>Print summary of an NBP Object</h2><span id='topic+print.nbp'></span>

<h3>Description</h3>

<p>Print contents of an NBP object, output from
<code><a href="#topic+prepare.nbp">prepare.nbp</a></code>, <code><a href="#topic+estimate.disp">estimate.disp</a></code>, or
<code><a href="#topic+nbp.test">nbp.test</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'nbp'
print(x, subset = 1:10, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.nbp_+3A_x">x</code></td>
<td>
<p>Output from <code><a href="#topic+prepare.nbp">prepare.nbp</a></code>,
<code><a href="#topic+estimate.disp">estimate.disp</a></code>, or <code><a href="#topic+nbp.test">nbp.test</a></code>.</p>
</td></tr>
<tr><td><code id="print.nbp_+3A_subset">subset</code></td>
<td>
<p>indices of rows of the count matrix to be
printed.</p>
</td></tr>
<tr><td><code id="print.nbp_+3A_...">...</code></td>
<td>
<p>other parameters (for future use).</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+nbp.test">nbp.test</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## See the example for nbp.test
</code></pre>

<hr>
<h2 id='test.coefficient'>Large-sample Test for a Regression Coefficient in an
Negative Binomial Regression Model</h2><span id='topic+test.coefficient'></span>

<h3>Description</h3>

<p><code>test.coefficient</code> performs large-sample tests
(higher-order asymptotic test, likelihood ratio test,
and/or Wald test) for testing regression coefficients in an
NB regression model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test.coefficient(nb, dispersion, x, beta0, tests = c("HOA", "LR", "Wald"),
  alternative = "two.sided", subset = 1:m, print.level = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="test.coefficient_+3A_nb">nb</code></td>
<td>
<p>an NB data object, output from
<code><a href="#topic+prepare.nb.data">prepare.nb.data</a></code>.</p>
</td></tr>
<tr><td><code id="test.coefficient_+3A_dispersion">dispersion</code></td>
<td>
<p>dispersion estimates, output from
<code><a href="#topic+estimate.disp">estimate.disp</a></code>.</p>
</td></tr>
<tr><td><code id="test.coefficient_+3A_x">x</code></td>
<td>
<p>an <code class="reqn">n</code> by <code class="reqn">p</code> design matrix describing
the treatment structure</p>
</td></tr>
<tr><td><code id="test.coefficient_+3A_beta0">beta0</code></td>
<td>
<p>a <code class="reqn">p</code>-vector specifying the null
hypothesis. Non-NA components specify the parameters to
test and their null values. (Currently, only
one-dimensional test is implemented, so only one non-NA
component is allowed).</p>
</td></tr>
<tr><td><code id="test.coefficient_+3A_tests">tests</code></td>
<td>
<p>a character string vector specifying the
tests to be performed, can be any subset of <code>"HOA"</code>
(higher-order asymptotic test), <code>"LR"</code> (likelihood
ratio test), and <code>"Wald"</code> (Wald test).</p>
</td></tr>
<tr><td><code id="test.coefficient_+3A_alternative">alternative</code></td>
<td>
<p>a character string specifying the
alternative hypothesis, must be one of <code>"two.sided"</code>
(default), <code>"greater"</code> or <code>"less"</code>.</p>
</td></tr>
<tr><td><code id="test.coefficient_+3A_subset">subset</code></td>
<td>
<p>an index vector specifying on which rows
should the tests be performed</p>
</td></tr>
<tr><td><code id="test.coefficient_+3A_print.level">print.level</code></td>
<td>
<p>a number controlling the amount of
messages printed: 0 for suppressing all messages, 1
(default) for basic progress messages, and 2 to 5 for
increasingly more detailed message.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>test.coefficient</code> performs large-sample tests for a
one-dimensional (<code class="reqn">q=1</code>) component <code class="reqn">\psi</code> of the
<code class="reqn">p</code>-dimensional regression coefficient <code class="reqn">\beta</code>. The
hypothesized value <code class="reqn">\psi_0</code> of <code class="reqn">\psi</code> is specified
by the non-NA component of the vector <code>beta0</code> in the
input.
</p>
<p>The likelihood ratio statistic, </p>
<p style="text-align: center;"><code class="reqn"> \lambda = 2
(l(\hat\beta) - l(\tilde\beta)),</code>
</p>
<p> converges in distribution
to a chi-square distribution with <code class="reqn">1</code> degree of
freedom.  The signed square root of the likelihood ratio
statistic <code class="reqn">\lambda</code>, also called the directed deviance,
</p>
<p style="text-align: center;"><code class="reqn">r = sign (\hat\psi - \psi_0) \sqrt \lambda</code>
</p>
<p> converges
to a standard normal distribution.
</p>
<p>For testing a one-dimensional parameter of interest,
Barndorff-Nielsen (1986, 1991) showed that a modified
directed </p>
<p style="text-align: center;"><code class="reqn"> r^* = r - \frac{1}{r} \log(z)</code>
</p>
<p> is, in wide
generality, asymptotically standard normally distributed to
a higher order of accuracy than the directed deviance
<code class="reqn">r</code> itself, where <code class="reqn">z</code> is an adjustment term. Tests
based on high-order asymptotic adjustment to the likelihood
ratio statistic, such as <code class="reqn">r^*</code> or its approximation,
are referred to as higher-order asymptotic (HOA) tests.
They generally have better accuracy than corresponding
unadjusted likelihood ratio tests, especially in situations
where the sample size is small and/or when the number of
nuisance parameters (<code class="reqn">p-q</code>) is large. The
implementation here is based on Skovgaard (2001). See Di et
al. 2013 for more details.
</p>


<h3>Value</h3>

<p>a list containing the following components:
</p>
<table>
<tr><td><code>beta.hat</code></td>
<td>
<p>an <code class="reqn">m</code> by <code class="reqn">p</code> matrix of regression
coefficient under the full model</p>
</td></tr> <tr><td><code>mu.hat</code></td>
<td>
<p>an <code class="reqn">m</code>
by <code class="reqn">n</code> matrix of fitted mean frequencies under the full
model</p>
</td></tr> <tr><td><code>beta.tilde</code></td>
<td>
<p>an <code class="reqn">m</code> by <code class="reqn">p</code> matrix of
regression coefficient under the null model</p>
</td></tr>
<tr><td><code>mu.tilde</code></td>
<td>
<p>an <code class="reqn">m</code> by <code class="reqn">n</code> matrix of fitted mean
frequencies under the null model.</p>
</td></tr> <tr><td><code>HOA</code>, <code>LR</code>, <code>Wald</code></td>
<td>
<p>each is a list of two <code class="reqn">m</code>-vectors,
<code>p.values</code> and <code>q.values</code>, giving p-values and
q-values of the corresponding tests when that test is
included in <code>tests</code>.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Barndorff-Nielsen, O. (1986): &quot;Infereni on full or partial
parameters based on the standardized signed log likelihood
ratio,&quot; Biometrika, 73, 307-322
</p>
<p>Barndorff-Nielsen, O. (1991): &quot;Modified signed log
likelihood ratio,&quot; Biometrika, 78, 557-563.
</p>
<p>Skovgaard, I. (2001): &quot;Likelihood asymptotics,&quot;
Scandinavian Journal of Statistics, 28, 3-32.
</p>
<p>Di Y, Schafer DW, Emerson SC, Chang JH (2013): &quot;Higher
order asymptotics for negative binomial regression
inferences from RNA-sequencing data&quot;. Stat Appl Genet Mol
Biol, 12(1), 49-70.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load Arabidopsis data
data(arab);

## Estimate normalization factors (we want to use the entire data set)
norm.factors = estimate.norm.factors(arab);

## Prepare the data
## For demonstration purpose, only the first 50 rows are used
nb.data = prepare.nb.data(arab[1:50,], lib.sizes = colSums(arab), norm.factors = norm.factors);

## For real analysis, we will use the entire data set, and can omit lib.sizes parameter)
## nb.data = prepare.nb.data(arab, norm.factors = norm.factors);

print(nb.data);
plot(nb.data);

## Specify the model matrix (experimental design)
grp.ids = as.factor(c(1, 1, 1, 2, 2, 2));
x = model.matrix(~grp.ids);

## Estimate dispersion model
dispersion = estimate.dispersion(nb.data, x);

print(dispersion);
plot(dispersion);

## Specify the null hypothesis
## The null hypothesis is beta[2]=0 (beta[2] is the log fold change).
beta0 = c(NA, 0);

## Test regression coefficient
res = test.coefficient(nb.data, dispersion, x, beta0);

## The result contains the data, the dispersion estimates and the test results
print(str(res));

## Show HOA test results for top ten most differentially expressed genes
top = order(res$HOA$p.values)[1:10];
print(cbind(nb.data$counts[top,], res$HOA[top,]));

## Plot log fold change versus the fitted mean of sample 1 (analagous to an MA-plot).
plot(res$mu.tilde[,1], res$beta.hat[,2]/log(2), log="x",
     xlab="Fitted mean of sample 1 under the null",
     ylab="Log (base 2) fold change");

## Highlight top DE genes
points(res$mu.tilde[top,1], res$beta.hat[top,2]/log(2), col="magenta");



</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
