<!DOCTYPE html><html lang="en"><head><title>Help for package neat</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {neat}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#neat-package'><p>neat</p></a></li>
<li><a href='#neat'><p>Performs neat for lists of gene sets</p></a></li>
<li><a href='#networkmatrix'>
<p>Creates a network matrix for <code>neat</code></p></a></li>
<li><a href='#plot.neat'>
<p>Plot method of <code>neat</code></p></a></li>
<li><a href='#print.neat'>
<p>Print method of <code>neat</code></p></a></li>
<li><a href='#summary.neat'>
<p>Summary method of <code>neat</code></p></a></li>
<li><a href='#yeast'>
<p>List collecting various yeast data (see 'description')</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Efficient Network Enrichment Analysis Test</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Includes functions and examples to compute NEAT, the Network
    Enrichment Analysis Test described in Signorelli et al.  (2016,
    &lt;<a href="https://doi.org/10.1186%2Fs12859-016-1203-6">doi:10.1186/s12859-016-1203-6</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://mirkosignorelli.github.io/r">https://mirkosignorelli.github.io/r</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>igraph</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, Matrix, rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-31 13:50:27 UTC; ms</td>
</tr>
<tr>
<td>Author:</td>
<td>Mirko Signorelli <a href="https://orcid.org/0000-0002-8102-3356"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  Veronica Vinciotti [ctb],
  Ernst Wit [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mirko Signorelli &lt;msignorelli.rpackages@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-01 09:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='neat-package'>neat</h2><span id='topic+neat-package'></span>

<h3>Description</h3>

<p>Includes functions and examples to compute NEAT, the Network Enrichment Analysis Test described in Signorelli et al. (2016).</p>


<h3>Author(s)</h3>

<p>Mirko Signorelli</p>


<h3>References</h3>

<p>Signorelli, M., Vinciotti, V., Wit, E. C. (2016). NEAT: an efficient network enrichment analysis test. BMC Bioinformatics, 17:352. Url: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+neat">neat</a></code></p>

<hr>
<h2 id='neat'>Performs neat for lists of gene sets</h2><span id='topic+neat'></span><span id='topic+pvalue'></span><span id='topic+neatc'></span>

<h3>Description</h3>

<p>Compute NEAT (Signorelli et al., 2016), a test for network enrichment analysis between/from a first list of sets ('A sets') and/to a second list of sets ('B sets').</p>


<h3>Usage</h3>

<pre><code class='language-R'>neat(alist, blist = NULL, network, nettype, nodes, 
alpha = NULL, mtc.type = 'fdr', anames = NULL, bnames = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="neat_+3A_alist">alist</code></td>
<td>
<p>List of A sets. Each element within the list is a vector of genes and represents a gene set</p>
</td></tr>
<tr><td><code id="neat_+3A_blist">blist</code></td>
<td>
<p>List of B sets. Each element within the list is a vector of genes and represents a gene set. If <code>nettype == "undirected"</code>, this argument is optional: if provided, every set of <code>blist</code> is compared with every set of <code>alist</code>; if <code>NULL</code>, the function compares sets in <code>alist</code> between themselves</p>
</td></tr>
<tr><td><code id="neat_+3A_network">network</code></td>
<td>
<p>One of the following objects: an adjacency matrix of class &quot;matrix&quot; (see 'Example 1') or a sparse adjacency matrix of class &quot;dgCMatrix&quot;; an <code>igraph</code> object (see 'Example 2'); a two-column matrix where every row represents and edge (for directed networks, parent nodes must be in the first column, and child nodes in the second)</p>
</td></tr>
<tr><td><code id="neat_+3A_nettype">nettype</code></td>
<td>
<p>Either <code>'directed'</code> or <code>'undirected'</code></p>
</td></tr>
<tr><td><code id="neat_+3A_nodes">nodes</code></td>
<td>
<p>Vector containing the (ordered) names of all nodes in the network</p>
</td></tr>
<tr><td><code id="neat_+3A_alpha">alpha</code></td>
<td>
<p>Significance level of the test (optional). If specified, a column with the conclusion of the test is added to the output</p>
</td></tr>
<tr><td><code id="neat_+3A_mtc.type">mtc.type</code></td>
<td>
<p>Type of multiple testing correction (NB: added from package version 1.2.0). Use <code>'fdr'</code> or <code>'BH'</code> for the Benjamini-Hockberg method, and <code>'none'</code> if no multiple testing correction is required. To know the shortcuts for other multiple testing correction methods, see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code></p>
</td></tr>
<tr><td><code id="neat_+3A_anames">anames</code></td>
<td>
<p>Vector of names for the elements of <code>alist</code> (optional: it has to be provided only if the elements of <code>alist</code> are not named)</p>
</td></tr>
<tr><td><code id="neat_+3A_bnames">bnames</code></td>
<td>
<p>Vector of names for the elements of <code>blist</code> (optional: it has to be provided only if the elements of <code>blist</code> are not named)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the following columns:
</p>
<table role = "presentation">
<tr><td><code>A</code></td>
<td>
<p>A set</p>
</td></tr>
<tr><td><code>B</code></td>
<td>
<p>B set</p>
</td></tr>
<tr><td><code>nab</code></td>
<td>
<p>observed number of links from A to B</p>
</td></tr>
<tr><td><code>expected_nab</code></td>
<td>
<p>expected number of links from A to B (in absence of enrichment)</p>
</td></tr>
<tr><td><code>pvalue</code></td>
<td>
<p>p-value of the test</p>
</td></tr>
<tr><td><code>adjusted.p</code></td>
<td>
<p>p-value adjusted to account for multiple testing</p>
</td></tr>
<tr><td><code>conclusion</code></td>
<td>
<p>conclusion of the test (only if <code>alpha</code> is specified): no enrichment, overenrichment or underenrichment</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mirko Signorelli</p>


<h3>References</h3>

<p>Signorelli, M., Vinciotti, V., Wit, E. C. (2016). NEAT: an efficient network enrichment analysis test. BMC Bioinformatics, 17:352. Url: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+networkmatrix">networkmatrix</a></code>, <code><a href="#topic+plot.neat">plot.neat</a></code>,  <code><a href="#topic+print.neat">print.neat</a></code>, <code><a href="#topic+summary.neat">summary.neat</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1: network given as adjacency matrix:
A = matrix(0, nrow=7, ncol=7)
A[1,c(2,3)]=1; A[2,c(5,7)]=1;A[3,c(1,4)]=1;A[4,c(2,5,7)]=1;A[6,c(2,5)]=1;A[7,4]=1
labels = letters[1:7]
set1 = c('a','e')
set2 = c('c','g')
set3 = c('d','f')
alist = list('set 1' = set1, 'set 2' = set2)
blist = list('set 3' = set3)

# test without multiple testing correction
test1 = neat(alist = alist, blist = blist, network=A, 
             nettype='directed', nodes=labels, 
             alpha=0.05, mtc.type = 'none')
print(test1)

# test with FDR multiple testing correction (default)
test1 = neat(alist = alist, blist = blist, network=A, 
             nettype='directed', nodes=labels, 
             alpha=0.05, mtc.type = 'fdr')
print(test1)

# Example 2: network given as igraph object:
library(igraph)
network = erdos.renyi.game(15, 1/3)
set1 = 1:4
set2 = c(2,5,13)
set3 = c(3,9,14)
set4 = c(8,15,20)
alist = list('set 1' = set1, 'set 2' = set2)
blist = list('set 3' = set3, 'set 4' = set4)

test2 = neat(alist, blist, network = network, 
             nettype='undirected', nodes=seq(1,15), 
             alpha=NULL)
print(test2)

# Example 3: network given as list of links:
networklist = matrix(nrow=13, ncol=2)
networklist[,1]=c('a','a','b','b','c','d','d','d','f','f','f','h','h')
networklist[,2]=c('d','e','e','g','d','b','e','g','a','b','e','c','g')

labels = letters[1:8]
set1 = c('a','b','e')
set2 = c('c','g')
set3 = c('d','f')
set4 = c('a','b','f')
alist = list('set 1' = set1, 'set 2' = set2)
blist = list('set 3' = set3, 'set4' = set4)

test3 = neat(alist, blist, network = networklist, 
             nettype = 'undirected', nodes=labels, 
             alpha=0.05, mtc.type = 'none')
print(test3)

alist = list('set 1' = set1, 'set 2' = set2, 'set 3' = set3)
test4 = neat(alist, network = networklist, 
             nettype = 'undirected', nodes=labels, 
             alpha=0.05, mtc.type = 'none')
print(test4)

# Example 4: ESR data
## Not run: 
data(yeast)
esr = list('ESR 1' = yeast$esr1, 'ESR 2' = yeast$esr2)
test = neat(alist = esr, blist = yeast$goslimproc, network = yeast$yeastnet,
            nettype = 'undirected', nodes = yeast$ynetgenes, alpha = 0.01)
# Replace with "blist = yeast$kegg" to use kegg pathways

m = dim(test)[1]
test1 = test[1:(m/2),]
table(test1$conclusion)
plot(test1)
o1=test1[test1$conclusion=='Overenrichment',]
print(o1, nrows='ALL') #display overenrichments

test2 = test[(m/2+1):m,]
table(test2$conclusion)
plot(test2)
o2=test2[test2$conclusion=='Overenrichment',]
print(o2, nrows='ALL') #display overenrichments

## End(Not run)</code></pre>

<hr>
<h2 id='networkmatrix'>
Creates a network matrix for <code>neat</code>
</h2><span id='topic+networkmatrix'></span>

<h3>Description</h3>

<p>Internal function, creates a two-column network matrix that can be further processed by <code><a href="#topic+neat">neat</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>networkmatrix(network, nodes, nettype)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="networkmatrix_+3A_network">network</code></td>
<td>

<p>One of the following objects: an adjacency matrix (class <code>"matrix"</code>), a sparse adjacency matrix (class <code>"dgCMatrix"</code>) or an igraph graph (class <code>"igraph"</code>)
</p>
</td></tr>
<tr><td><code id="networkmatrix_+3A_nodes">nodes</code></td>
<td>

<p>Vector containing the (ordered) names of all nodes in the network
</p>
</td></tr>
<tr><td><code id="networkmatrix_+3A_nettype">nettype</code></td>
<td>

<p>Either <code>'directed'</code> or <code>'undirected'</code>
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is an internal function, that is called within <code><a href="#topic+neat">neat</a></code> to convert different types of network objects (see argument 'network' above) into a standard two-column network matrix, that can then be processed by <code>neat</code>. 
</p>


<h3>Value</h3>

<p>A two-column matrix, where every row represents and edge. For directed networks, parent nodes must be in the first column, and child nodes in the second.
</p>


<h3>Author(s)</h3>

<p>Mirko Signorelli</p>


<h3>References</h3>

<p>Signorelli, M., Vinciotti, V., Wit, E. C. (2016). NEAT: an efficient network enrichment analysis test. BMC Bioinformatics, 17:352. Url: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+neat">neat</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># First case: adjacency matrix
n&lt;-50
adjacency &lt;- matrix(sample(0:1, n^2, replace=TRUE, prob=c(0.9,0.1)), ncol=n)
diag(adjacency) &lt;- 0
lab = paste(rep('gene'),1:n)
head(networkmatrix(adjacency, lab, 'directed'))

# Second case: sparse adjacency matrix
library(Matrix)
sparse_adjacency&lt;-Matrix(adjacency,sparse=TRUE)
head(networkmatrix(sparse_adjacency, lab, 'directed'))

# Third case: igraph object
library(igraph)
igraph_graph = erdos.renyi.game(15, 1/3)
lab = paste(rep('gene'),1:15)
head(networkmatrix(igraph_graph, lab, 'directed'))
</code></pre>

<hr>
<h2 id='plot.neat'>
Plot method of <code>neat</code>
</h2><span id='topic+plot.neat'></span>

<h3>Description</h3>

<p><code>plot</code> method for class &quot;<code>neat</code>&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'neat'
plot(x, nbreaks = 10, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.neat_+3A_x">x</code></td>
<td>
<p>An object of class &quot;<code>neat</code>&quot;</p>
</td></tr>
<tr><td><code id="plot.neat_+3A_nbreaks">nbreaks</code></td>
<td>
<p>Number of breaks to be used in the histogram (default is 10)</p>
</td></tr>
<tr><td><code id="plot.neat_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An histogram showing the distribution of p-values and a p-p plot comparing the distribution of p-values to the uniform distribution.</p>


<h3>Author(s)</h3>

<p>Mirko Signorelli</p>


<h3>References</h3>

<p>Signorelli, M., Vinciotti, V., Wit, E. C. (2016). NEAT: an efficient network enrichment analysis test. BMC Bioinformatics, 17:352. Url: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+neat">neat</a></code>, <code><a href="#topic+print.neat">print.neat</a></code>, <code><a href="#topic+summary.neat">summary.neat</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(yeast)
esr2 = list('ESR 2' = yeast$esr2)

test = neat(alist = esr2, blist = yeast$goslimproc, network = yeast$yeastnet, 
            nettype='undirected', nodes = yeast$ynetgenes, alpha = 0.01)

plot(test)

## End(Not run)</code></pre>

<hr>
<h2 id='print.neat'>
Print method of <code>neat</code>
</h2><span id='topic+print.neat'></span>

<h3>Description</h3>

<p><code>print</code> method for class &quot;<code>neat</code>&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'neat'
print(x, nrows=10, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.neat_+3A_x">x</code></td>
<td>
<p>An object of class &quot;<code>neat</code>&quot;</p>
</td></tr>
<tr><td><code id="print.neat_+3A_nrows">nrows</code></td>
<td>
<p>Maximum number of results to print (default is 10). It can be either an integer number or &quot;ALL&quot;</p>
</td></tr>
<tr><td><code id="print.neat_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe showing the first <code>nrows</code> tests contained in a <code>neat</code> object.</p>


<h3>Author(s)</h3>

<p>Mirko Signorelli</p>


<h3>References</h3>

<p>Signorelli, M., Vinciotti, V., Wit, E. C. (2016). NEAT: an efficient network enrichment analysis test. BMC Bioinformatics, 17:352. Url: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+neat">neat</a></code>, <code><a href="#topic+plot.neat">plot.neat</a></code>, <code><a href="#topic+summary.neat">summary.neat</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>A = matrix(0, nrow=7, ncol=7)
A[1,c(2,3)]=1; A[2,c(5,7)]=1;A[3,c(1,4)]=1;A[4,c(2,5,7)]=1;A[6,c(2,5)]=1;A[7,4]=1

labels = letters[1:7]
set1 = c('a','e')
set2 = c('c','g')
set3 = c('d','f')
alist = list('set 1' = set1, 'set 2' = set2)
blist = list('set 3' = set3)

test = neat(alist, blist, network=A, nettype='directed', nodes=labels, alpha=0.05)
print(test)

</code></pre>

<hr>
<h2 id='summary.neat'>
Summary method of <code>neat</code>
</h2><span id='topic+summary.neat'></span>

<h3>Description</h3>

<p><code>summary</code> method for class &quot;<code>neat</code>&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'neat'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.neat_+3A_object">object</code></td>
<td>
<p>An object of class &quot;<code>neat</code>&quot;</p>
</td></tr>
<tr><td><code id="summary.neat_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The <code>summary.neat</code> function returns the following values:
</p>

<ul>
<li><p> the number of tests computed;
</p>
</li>
<li><p> the number of enrichments at 1% and 5% level;
</p>
</li>
<li><p> the p-value of the Kolmogorov-Smirnov test to check if the distribution of p-values is uniform.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Mirko Signorelli</p>


<h3>References</h3>

<p>Signorelli, M., Vinciotti, V., Wit, E. C. (2016). NEAT: an efficient network enrichment analysis test. BMC Bioinformatics, 17:352. Url: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+neat">neat</a></code>, <code><a href="#topic+plot.neat">plot.neat</a></code>, <code><a href="#topic+summary.neat">summary.neat</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(yeast)
esr = list('ESR 1' = yeast$esr1, 'ESR 2' = yeast$esr2)
test = neat(alist = esr, blist = yeast$goslimproc, network = yeast$yeastnet,
            nettype = 'undirected', nodes = yeast$ynetgenes, alpha = 0.01)

test1 = test[1:99,]
summary(test1)

test2 = test[100:198,]
summary(test2)

## End(Not run)</code></pre>

<hr>
<h2 id='yeast'>
List collecting various yeast data (see 'description')
</h2><span id='topic+yeast'></span>

<h3>Description</h3>

<p><code>yeast</code> is a list that contains:
</p>
<p><code>yeastnet</code>: network matrix representing Yeastnet-v3 (Kim et al., 2013)
</p>
<p><code>ynetgenes</code>: vector with the names of the genes appearing in <code>yeastnet</code>
</p>
<p><code>esr1</code>: vector containing the first of the two gene sets that constitute the &quot;Environmental Stress Response&quot; (ESR) reported by Gasch et al. (2012)
</p>
<p><code>esr2</code>: vector containing the second gene set of the ESR
</p>
<p><code>goslimproc</code>: list containing the gene sets of the GOslim process ontology (Ashburner et al., 2000) for the buddying yeast Saccaromyces Cerevisiae (groups 'biological process' and 'other' are not included)
</p>
<p><code>kegg</code>: list containing the KEGG pathways (Kanehisa and Goto, 2002) for the buddying yeast Saccaromyces Cerevisiae
</p>


<h3>Format</h3>

<p><code>yeast</code>: list
</p>


<h3>Source</h3>

<p>Ashburner, M., Ball, C. A., Blake, J. A., Botstein, D., Butler, H., Cherry, J. M., Davis, A. P., Dolinski, K., Dwight, S. S., Eppig, J. T., et al. (2000). Gene ontology: tool for the unification of biology. Nat. Genet., 25(1), 25-29.
</p>
<p>Gasch, A. P., Spellman, P. T., Kao, C. M., Carmel-Harel, O., Eisen, M. B., Storz, G., Botstein, D., and Brown, P. O. (2000). Genomic expression programs in the response of yeast cells to environmental changes. Mol. Biol. Cell, 11(12), 4241-4257.
</p>
<p>Kanehisa, M., and Goto, S. (2002). KEGG: Kyoto Encyclopedia of Genes and Genomes. Nucleic Acids Res., 28(1), 27-30.
</p>
<p>Kim, H., Shin, J., Kim, E., Kim, H., Hwang, S., Shim, J. E., and Lee, I. (2013). Yeastnet v3: a public database of data-specific and integrated functional gene networks for saccharomyces cerevisiae. Nucleic Acids Res., 42 (D1), D731-6.
</p>
<p>Signorelli, M., Vinciotti, V., Wit, E. C. (2016). NEAT: an efficient network enrichment analysis test. BMC Bioinformatics, 17:352. Url: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6.
</p>


<h3>References</h3>

<p>Ashburner, M., Ball, C. A., Blake, J. A., Botstein, D., Butler, H., Cherry, J. M., Davis, A. P., Dolinski, K., Dwight, S. S., Eppig, J. T., et al. (2000). Gene ontology: tool for the unification of biology. Nat. Genet., 25(1), 25-29.
</p>
<p>Gasch, A. P., Spellman, P. T., Kao, C. M., Carmel-Harel, O., Eisen, M. B., Storz, G., Botstein, D., and Brown, P. O. (2000). Genomic expression programs in the response of yeast cells to environmental changes. Mol. Biol. Cell, 11(12), 4241-4257.
</p>
<p>Kanehisa, M., and Goto, S. (2002). KEGG: Kyoto Encyclopedia of Genes and Genomes. Nucleic Acids Res., 28(1), 27-30.
</p>
<p>Kim, H., Shin, J., Kim, E., Kim, H., Hwang, S., Shim, J. E., and Lee, I. (2013). Yeastnet v3: a public database of data-specific and integrated functional gene networks for saccharomyces cerevisiae. Nucleic Acids Res., 42 (D1), D731-6.
</p>
<p>Signorelli, M., Vinciotti, V., Wit, E. C. (2016). NEAT: an efficient network enrichment analysis test. BMC Bioinformatics, 17:352. Url: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-1203-6.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+neat">neat</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(yeast)
esr = list('ESR 1' = yeast$esr1, 'ESR 2' = yeast$esr2)
test = neat(alist = esr, blist = yeast$goslimproc, network = yeast$yeastnet,
            nettype = 'undirected', nodes = yeast$ynetgenes, alpha = 0.01)
# Replace with "blist = yeast$kegg" to use kegg pathways

m = dim(test)[1]
test1 = test[1:(m/2),]
o1=test1[test1$conclusion=='Overenrichment',]
# list of overenrichments for the first ESR set:
print(o1, nrows='ALL') 

test2 = test[(m/2+1):m,]
o2=test2[test2$conclusion=='Overenrichment',]
# list of overenrichments for the second ESR set:
print(o2, nrows='ALL')

# the same can be done using KEGG pathways:
keggtest = neat(alist = esr, blist = yeast$kegg, network = yeast$yeastnet,
            nettype = 'undirected', nodes = yeast$ynetgenes, alpha = 0.01)

## End(Not run)</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
