<!DOCTYPE html><html lang="en"><head><title>Help for package nebula</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {nebula}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#nebula-package'>
<p>Negative Binomial Mixed Models Using Large-Sample Approximation for Differential Expression Analysis of ScRNA-Seq Data</p></a></li>
<li><a href='#group_cell'><p>Group cells according to subject IDs</p></a></li>
<li><a href='#nbresidual'><p>Extract Pearson residuals from the results of NEBULA</p></a></li>
<li><a href='#nebula'><p>Association analysis of a multi-subject single-cell data set using a fast negative binomial mixed model</p></a></li>
<li><a href='#sample_data'><p>An example data set for testing nebula</p></a></li>
<li><a href='#sample_seurat'>
<p>An example data set for testing scToNeb</p></a></li>
<li><a href='#scToNeb'><p>Retrieve data from Seurat or SingleCellExperiment object to prepare for use in nebula</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Negative Binomial Mixed Models Using Large-Sample Approximation
for Differential Expression Analysis of ScRNA-Seq Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.5.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-15</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Liang He &lt;hyx520101@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A fast negative binomial mixed model for conducting association analysis of multi-subject single-cell data. It can be used for identifying marker genes, differential expression and co-expression analyses. The model includes subject-level random effects to account for the hierarchical structure in multi-subject single-cell data. See He et al. (2021) &lt;<a href="https://doi.org/10.1038%2Fs42003-021-02146-6">doi:10.1038/s42003-021-02146-6</a>&gt;.  </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.7), nloptr, stats, Matrix, methods, Rfast, trust,
parallelly (&ge; 1.34.0), doFuture (&ge; 0.12.2), future (&ge;
1.32.0), foreach (&ge; 1.5.2), doRNG (&ge; 1.8.6), Seurat,
SingleCellExperiment</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/lhe17/nebula">https://github.com/lhe17/nebula</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/lhe17/nebula/issues">https://github.com/lhe17/nebula/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, utils, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-15 22:27:18 UTC; helia</td>
</tr>
<tr>
<td>Author:</td>
<td>Liang He [aut, cre],
  Raghav Sharma [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-15 23:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='nebula-package'>
Negative Binomial Mixed Models Using Large-Sample Approximation for Differential Expression Analysis of ScRNA-Seq Data
</h2><span id='topic+nebula-package'></span>

<h3>Description</h3>

<p>A fast negative binomial mixed model for conducting association analysis of multi-subject single-cell data. It can be used for identifying marker genes, differential expression and co-expression analyses. The model includes subject-level random effects to account for the hierarchical structure in multi-subject single-cell data. See He et al. (2021) &lt;doi:10.1038/s42003-021-02146-6&gt;.  
</p>


<h3>Details</h3>

<p>nebula is an R package for performing association analysis using a fast negative binomial mixed model for multi-subject single-cell data.
</p>


<h3>Author(s)</h3>

<p>NA
</p>
<p>Maintainer: Liang He &lt;hyx520101@gmail.com&gt;
</p>


<h3>References</h3>

<p>He, L., Davila-Velderrain, J., Sumida, T. S., Hafler, D. A., Kellis, M., &amp; Kulminski, A. M. (2021). NEBULA is a fast negative binomial mixed model for differential or co-expression analysis of large-scale multi-subject single-cell data. Communications biology, 4(1), 1-17.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(nebula)
data(sample_data)
pred = model.matrix(~X1+X2+cc,data=sample_data$pred)
re = nebula(count=sample_data$count,id=sample_data$sid,pred=pred)
</code></pre>

<hr>
<h2 id='group_cell'>Group cells according to subject IDs</h2><span id='topic+group_cell'></span>

<h3>Description</h3>

<p>Group cells according to subject IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group_cell(count, id, pred = NULL, offset = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="group_cell_+3A_count">count</code></td>
<td>
<p>A raw count matrix of the single-cell data. The rows are the genes, and the columns are the cells. The matrix can be a matrix object or a sparse dgCMatrix object.</p>
</td></tr>
<tr><td><code id="group_cell_+3A_id">id</code></td>
<td>
<p>A vector of subject IDs. The length should be the same as the number of columns of the count matrix.</p>
</td></tr>
<tr><td><code id="group_cell_+3A_pred">pred</code></td>
<td>
<p>A design matrix of the predictors. The rows are the cells and the columns are the predictors. If not specified, an intercept column will be generated by default.</p>
</td></tr>
<tr><td><code id="group_cell_+3A_offset">offset</code></td>
<td>
<p>A vector of the scaling factor. The values must be strictly positive. If not specified, a vector of all ones will be generated by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>count: A reordered count matrix. If the cells are already grouped, the function returns NULL.
</p>
<p>id: A reordered subject ID vector.
</p>
<p>pred: A reordered design matrix of the predictors.
</p>
<p>offset: A reordered vector of the scaling factor.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(nebula)
data(sample_data)
pred = model.matrix(~X1+X2+cc,data=sample_data$pred)
df_order = group_cell(count=sample_data$count,id=sample_data$sid,pred=pred)

</code></pre>

<hr>
<h2 id='nbresidual'>Extract Pearson residuals from the results of NEBULA</h2><span id='topic+nbresidual'></span>

<h3>Description</h3>

<p>Extract Pearson residuals from the results of NEBULA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nbresidual(nebula, count, id, pred = NULL, offset = NULL, conditional = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nbresidual_+3A_nebula">nebula</code></td>
<td>
<p>An object of the result obtained from running the function nebula.</p>
</td></tr>
<tr><td><code id="nbresidual_+3A_count">count</code></td>
<td>
<p>A raw count matrix of the single-cell data. The rows are the genes, and the columns are the cells. The matrix can be a matrix object or a sparse dgCMatrix object.</p>
</td></tr>
<tr><td><code id="nbresidual_+3A_id">id</code></td>
<td>
<p>A vector of subject IDs. The length should be the same as the number of columns of the count matrix.</p>
</td></tr>
<tr><td><code id="nbresidual_+3A_pred">pred</code></td>
<td>
<p>A design matrix of the predictors. The rows are the cells and the columns are the predictors. If not specified, an intercept column will be generated by default.</p>
</td></tr>
<tr><td><code id="nbresidual_+3A_offset">offset</code></td>
<td>
<p>A vector of the scaling factor. The values must be strictly positive. If not specified, a vector of all ones will be generated by default.</p>
</td></tr>
<tr><td><code id="nbresidual_+3A_conditional">conditional</code></td>
<td>
<p>A logical value. By default (FALSE), the function returns marginal Pearson residuals. If TRUE, the function will return conditional Pearson residuals.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>residuals: A matrix of Pearson residuals. The number of columns is the number of cells in the count matrix. The rows correspond to gene IDs reported in the result from nebula.
</p>
<p>gene: Gene names corresponding to the row names of the count matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(nebula)
data(sample_data)
pred = model.matrix(~X1+X2+cc,data=sample_data$pred)
re = nebula(count=sample_data$count,id=sample_data$sid,pred=pred)
resid = nbresidual(re,count=sample_data$count,id=sample_data$sid,pred=pred)

</code></pre>

<hr>
<h2 id='nebula'>Association analysis of a multi-subject single-cell data set using a fast negative binomial mixed model</h2><span id='topic+nebula'></span>

<h3>Description</h3>

<p>Association analysis of a multi-subject single-cell data set using a fast negative binomial mixed model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nebula(
  count,
  id,
  pred = NULL,
  offset = NULL,
  min = c(1e-04, 1e-04),
  max = c(10, 1000),
  model = "NBGMM",
  method = "LN",
  cutoff_cell = 20,
  kappa = 800,
  opt = "lbfgs",
  verbose = TRUE,
  cpc = 0.005,
  mincp = 5,
  covariance = FALSE,
  output_re = FALSE,
  reml = 0,
  ncore = 2,
  fmaxsize = Inf
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nebula_+3A_count">count</code></td>
<td>
<p>A raw count matrix of the single-cell data. The rows are the genes, and the columns are the cells. The matrix can be a matrix object or a sparse dgCMatrix object.</p>
</td></tr>
<tr><td><code id="nebula_+3A_id">id</code></td>
<td>
<p>A vector of subject IDs. The length should be the same as the number of columns of the count matrix.</p>
</td></tr>
<tr><td><code id="nebula_+3A_pred">pred</code></td>
<td>
<p>A design matrix of the predictors. The rows are the cells and the columns are the predictors. If not specified, an intercept column will be generated by default.</p>
</td></tr>
<tr><td><code id="nebula_+3A_offset">offset</code></td>
<td>
<p>A vector of the scaling factor. The values must be strictly positive. If not specified, a vector of all ones will be generated by default.</p>
</td></tr>
<tr><td><code id="nebula_+3A_min">min</code></td>
<td>
<p>Minimum values for the overdispersions parameters <code class="reqn">\sigma^2</code> and <code class="reqn">\phi</code>. Must be positive. The default is c(1e-4,1e-4).</p>
</td></tr>
<tr><td><code id="nebula_+3A_max">max</code></td>
<td>
<p>Maximum values for the overdispersions parameters <code class="reqn">\sigma^2</code> and <code class="reqn">\phi</code>. Must be positive. The default is c(10,1000).</p>
</td></tr>
<tr><td><code id="nebula_+3A_model">model</code></td>
<td>
<p>'NBGMM', 'PMM' or 'NBLMM'. 'NBGMM' is for fitting a negative binomial gamma mixed model. 'PMM' is for fitting a Poisson gamma mixed model. 'NGLMM' is for fitting a negative binomial lognormal mixed model (the same model as that in the lme4 package). The default is 'NBGMM'.</p>
</td></tr>
<tr><td><code id="nebula_+3A_method">method</code></td>
<td>
<p>'LN' or 'HL'. 'LN' is to use NEBULA-LN and 'HL' is to use NEBULA-HL. The default is 'LN'.</p>
</td></tr>
<tr><td><code id="nebula_+3A_cutoff_cell">cutoff_cell</code></td>
<td>
<p>The data will be refit using NEBULA-HL to estimate both overdispersions if the product of the cells per subject and the estimated cell-level overdispersion parameter <code class="reqn">\phi</code> is smaller than cutoff_cell. The default is 20.</p>
</td></tr>
<tr><td><code id="nebula_+3A_kappa">kappa</code></td>
<td>
<p>Please see the vignettes for more details. The default is 800.</p>
</td></tr>
<tr><td><code id="nebula_+3A_opt">opt</code></td>
<td>
<p>'lbfgs' or 'trust'. Specifying the optimization algorithm used in NEBULA-LN. The default is 'lbfgs'. If it is 'trust', a trust region algorithm based on the Hessian matrix will be used for optimization.</p>
</td></tr>
<tr><td><code id="nebula_+3A_verbose">verbose</code></td>
<td>
<p>An optional logical scalar indicating whether to print additional messages. Default is FALSE.</p>
</td></tr>
<tr><td><code id="nebula_+3A_cpc">cpc</code></td>
<td>
<p>A non-negative threshold for filtering low-expression genes. Genes with counts per cell smaller than the specified value will not be analyzed.</p>
</td></tr>
<tr><td><code id="nebula_+3A_mincp">mincp</code></td>
<td>
<p>A positive integer threshold for filtering low-expression genes. A gene will not be analyzed if its number of cells that have a non-zero count is smaller than the specified value .</p>
</td></tr>
<tr><td><code id="nebula_+3A_covariance">covariance</code></td>
<td>
<p>If TRUE, nebula will output the covariance matrix for the estimated log(FC), which can be used for testing contrasts.</p>
</td></tr>
<tr><td><code id="nebula_+3A_output_re">output_re</code></td>
<td>
<p>If TRUE, nebula will output the subject-level random effects. Only effective for model='NBGMM' or 'NBLMM'.</p>
</td></tr>
<tr><td><code id="nebula_+3A_reml">reml</code></td>
<td>
<p>Either 0 (default) or 1. If it is one, REML will be used to estimate the overdispersions.</p>
</td></tr>
<tr><td><code id="nebula_+3A_ncore">ncore</code></td>
<td>
<p>The number of cores used for parallel computing.</p>
</td></tr>
<tr><td><code id="nebula_+3A_fmaxsize">fmaxsize</code></td>
<td>
<p>The maximum allowed total size (in bytes) of global variables (future.globals.maxSize) when using parallel computing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>summary: The estimated coefficient, standard error and p-value for each predictor.
</p>
<p>overdispersion: The estimated cell-level and subject-level overdispersions <code class="reqn">\sigma^2</code> and <code class="reqn">\phi^{-1}</code>.
</p>
<p>convergence: More information about the convergence of the algorithm for each gene. A value of -20 or lower indicates a potential failure of the convergence. A value of one indicates that the convergence is reached due to a sufficiently small improvement of the function value. A value of -10 indicates that the convergence is reached because the gradients are close to zero (i.e., the critical point) and no improvement of the function value can be found.
</p>
<p>algorithm: The algorithm used for analyzing the gene. More information can be found in the vignettes.
</p>
<p>covariance: The covariance matrix for the estimated log(FC).
</p>
<p>random_effect: The subject-level random effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(nebula)
data(sample_data)
pred = model.matrix(~X1+X2+cc,data=sample_data$pred)
re = nebula(count=sample_data$count,id=sample_data$sid,pred=pred)

</code></pre>

<hr>
<h2 id='sample_data'>An example data set for testing nebula</h2><span id='topic+sample_data'></span>

<h3>Description</h3>

<p>A dataset containing a count matrix, subject IDs, a data frame of predictors and scaling factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_data
</code></pre>


<h3>Format</h3>

<p>A list of four objects:
</p>

<dl>
<dt>count</dt><dd><p>A raw count matrix</p>
</dd>
<dt>sid</dt><dd><p>A vector of subject IDs</p>
</dd>
<dt>pred</dt><dd><p>A data frame of three predictors</p>
</dd>
<dt>offset</dt><dd><p>A vector of scaling factors</p>
</dd>
</dl>


<hr>
<h2 id='sample_seurat'>
An example data set for testing scToNeb
</h2><span id='topic+sample_seurat'></span>

<h3>Description</h3>

<p>A Seurat object containing a subset (1000 genes and 1000 cells) of the eight-pancreas scRNA-seq datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("sample_seurat")</code></pre>


<h3>Format</h3>

<p>A Seurat object
</p>


<h3>Source</h3>

<p>https://github.com/satijalab/seurat-data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sample_seurat)
</code></pre>

<hr>
<h2 id='scToNeb'>Retrieve data from Seurat or SingleCellExperiment object to prepare for use in nebula</h2><span id='topic+scToNeb'></span>

<h3>Description</h3>

<p>Retrieve data from Seurat or SingleCellExperiment object to prepare for use in nebula
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scToNeb(
  obj,
  assay = NULL,
  id = NULL,
  pred = NULL,
  offset = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scToNeb_+3A_obj">obj</code></td>
<td>
<p><code>Seurat</code> or <code>SingleCellExperiment</code> object to create data set for Nebula.</p>
</td></tr>
<tr><td><code id="scToNeb_+3A_assay">assay</code></td>
<td>
<p>Assay to retrieve counts from the corresponding <code>Seurat</code> count matrix.</p>
</td></tr>
<tr><td><code id="scToNeb_+3A_id">id</code></td>
<td>
<p>Sample ID to use metadata object i.e. <code>obj$id</code>.</p>
</td></tr>
<tr><td><code id="scToNeb_+3A_pred">pred</code></td>
<td>
<p>Character vector of predictors from metadata in <code>Seurat</code> or <code>SingleCellExperiment</code> objects.</p>
</td></tr>
<tr><td><code id="scToNeb_+3A_offset">offset</code></td>
<td>
<p>Metadata column corresponding to per-cell scaling factor e.g. TMM.</p>
</td></tr>
<tr><td><code id="scToNeb_+3A_verbose">verbose</code></td>
<td>
<p>Indicating whether to print additional messages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data_neb: A list usable for nebula.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(Seurat)
library(nebula)

data("sample_seurat")
re &lt;- scToNeb(obj = sample_seurat, assay = "RNA", id = "replicate", pred = c("celltype", "tech"))

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
