<!DOCTYPE html><html><head><title>Help for package NEpiC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {NEpiC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#c_dms'>
<p>Combined signals dense modules searching algorithm.</p></a></li>
<li><a href='#c_score'>
<p>Computing the mean and variance combined signal scores</p></a></li>
<li><a href='#candidate_genes'>
<p>Chosing candidate genes.</p></a></li>
<li><a href='#permutation'>
<p>Validating identified modules using permutations</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Network Assisted Algorithm for Epigenetic Studies Using Mean and
Variance Combined Signals</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-03-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Peifeng Ruan</td>
</tr>
<tr>
<td>Depends:</td>
<td>igraph, PairedData</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Peifeng Ruan &lt;pruan12@fudan.edu.cn&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Package for a Network assisted algorithm for Epigenetic studies using mean and variance Combined signals: NEpiC. NEpiC combines both signals in mean and variance differences in methylation level between case and control groups searching for differentially methylated sub-networks (modules) using the protein-protein interaction network.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-03-25 19:10:43 UTC; win7</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-03-26 15:13:10</td>
</tr>
</table>
<hr>
<h2 id='c_dms'>
Combined signals dense modules searching algorithm.
</h2><span id='topic+c_dms'></span>

<h3>Description</h3>

<p>c_dms uses the combined signals in both mean and variance differences in methylation measures between case and control groups to search for differentially methylated sub-networks (modules) using a protein-protein interaction (PPI) network. It returns ordered dense modules. </p>


<h3>Usage</h3>

<pre><code class='language-R'>c_dms(network, gene2weight, d = 1, r = 0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_dms_+3A_network">network</code></td>
<td>

<p>A dataframe describing the edges of s PPI network. Two columns are needed in this dataset: interactor A and interactor B.
</p>
</td></tr>
<tr><td><code id="c_dms_+3A_gene2weight">gene2weight</code></td>
<td>

<p>A dataframe describing combined gene-level scores. The first column of this dataframe, named 'gene', contains the gene names, which should use the same annotation as those used in the network. The second column 'weight' contains gene-level combined scores.
</p>
</td></tr>
<tr><td><code id="c_dms_+3A_d">d</code></td>
<td>

<p>An integer determining the order of neighbour genes to be searched in NEpiC algorithm. We suggest to set d=1 for NEpiC.
</p>
</td></tr>
<tr><td><code id="c_dms_+3A_r">r</code></td>
<td>

<p>A number determining the threshold for including a new gene into the module.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing several statistics for modules, including genes in modules, module scores before and after normalization. 
</p>


<h3>References</h3>

<p>Peifeng Ruan, Jing Shen, Regina M. Santella, Shuigeng Zhou, Shuang Wang.(2015). NEpiC: a Network-assisted algorithm for Epigenetic studies using mean and variance Combined signals.<br />
Peilin Jia, Siyuan Zheng, Jirong Rong, Wei Zheng, Zhongming Zhao. (2011) Bioformatics. dmGWAS: dense module searching for genome-wide association studies in protein-protein interaction networks.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#res=c_dms(network, gene2weight, d = 1, r = 0.1) 
</code></pre>

<hr>
<h2 id='c_score'>
Computing the mean and variance combined signal scores
</h2><span id='topic+c_score'></span>

<h3>Description</h3>

<p>A function computing gene-level scores combining both signals in mean and variance differences. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_score(mean_sig, var_sig, lambda = "default")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_score_+3A_mean_sig">mean_sig</code></td>
<td>

<p>A dataframe describing mean signals for every CpG sites. Three columns are needed in this dataset, which are 'cpg', 'gene' and 'mean_p'. 'cpg' column has names of CpG sites. 'gene' column has has gene annotations for CpG sites. 'mean_p' column has p values of mean differences for CpG sites.
</p>
</td></tr>
<tr><td><code id="c_score_+3A_var_sig">var_sig</code></td>
<td>

<p>A dataframe describing variance signals for every CpG sites. Three columns are needed in this dataset, which are 'cpg', 'gene' and 'var_p'. 'cpg' column has names of CpG sites. 'gene' column has has gene annotations for CpG sites. 'var_p' column has p values of variance differences for CpG sites.
</p>
</td></tr>
<tr><td><code id="c_score_+3A_lambda">lambda</code></td>
<td>

<p>A parameter to balance mean and variance scores. NEpiC computes it as default if it is not otherwise specified by user. If lambda = 1, NEpiC will search dense modules using only mean signals. If lambda = 0, NEpiC will search dense modules using only variance signals. Users can also specify any float between 0 and 1 for lambda if they have any additional prior biology information.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>c_score returns a dataframe describing the combined gene-level scores, which can be used directly in c_dms.
</p>


<h3>References</h3>

<p>NEpiC: a Network-assisted algorithm for Epigenetic studies using mean and variance Combined signals. Peifeng Ruan, Jing Shen, Regina M. Santella, Shuigeng Zhou, Shuang Wang.(2015)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#c_score(mean_sig, var_sig, lambda = "default")
</code></pre>

<hr>
<h2 id='candidate_genes'>
Chosing candidate genes.
</h2><span id='topic+candidate_genes'></span>

<h3>Description</h3>

<p>Chosing genes in top ranked modules as candidate genes for outcomes. Further prioritized candidate genes can be chosen based on percentages selected by top ranked modules selected if prioritizing = T.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>candidate_genes(res, pct = 0.01, prioritizing)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="candidate_genes_+3A_res">res</code></td>
<td>

<p>The result for c_dms.
</p>
</td></tr>
<tr><td><code id="candidate_genes_+3A_pct">pct</code></td>
<td>

<p>A float which determines the number of candidate genes. For example, if pct is 0.01 as default setting, the genes in top 1 percent modules will be chosen as the candidate genes.
</p>
</td></tr>
<tr><td><code id="candidate_genes_+3A_prioritizing">prioritizing</code></td>
<td>

<p>To further prioritize the candidate genes or not.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list for candidate genes or prioritized candidate genes.
</p>


<h3>References</h3>

<p>NEpiC: a Network-assisted algorithm for Epigenetic studies using mean and variance Combined signals Peifeng Ruan, Jing Shen, Regina M. Santella, Shuigeng Zhou, Shuang Wang.(2015)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># candidate_genes(res, pct = 0.01, prioritizing = T)
</code></pre>

<hr>
<h2 id='permutation'>
Validating identified modules using permutations  
</h2><span id='topic+permutation'></span>

<h3>Description</h3>

<p>A function shuffling the tumor/adjacent normal labels within a tumor/adjacent normal pair and repeating the permutation procedures to calculate the permutated module scores for the identified top 1 percentage modules.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>permutation(normal_data, tumor_data, annotation, res, n = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="permutation_+3A_normal_data">normal_data</code></td>
<td>

<p>The Illumina 450k methylation array results for normal samples. The rows represent the methylation sites and the columns represent different samples.
</p>
</td></tr>
<tr><td><code id="permutation_+3A_tumor_data">tumor_data</code></td>
<td>

<p>The Illumina 450k methylation array results for adjacent tumor paire samples. The rows represent the methylation sites and the columns represent different samples.
</p>
</td></tr>
<tr><td><code id="permutation_+3A_annotation">annotation</code></td>
<td>

<p>The annotation file for Illumina 450k methylation array data. The first column is the methylation sites' names, whose column name is &quot;cpg&quot; and the second column is the methylation sites' gene annotations, whose column name is &quot;gene&quot;
</p>
</td></tr>
<tr><td><code id="permutation_+3A_res">res</code></td>
<td>

<p>The result of the c_dms, which contains the information of identified modules.
</p>
</td></tr>
<tr><td><code id="permutation_+3A_n">n</code></td>
<td>

<p>The numbers of permutations.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataset contains identified modules' names, identified modules' score, identified modules' permutated scores and the last column contains the permutation p values for identified modules.
</p>


<h3>References</h3>

<p>Peifeng Ruan, Jing Shen, Regina M. Santella, Shuigeng Zhou, Shuang Wang.(2015). NEpiC: a Network-assisted algorithm for Epigenetic studies using mean and variance Combined signals.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#perm_result = function (normal_data, tumor_data, annotation, res, n = 100) 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
