<!DOCTYPE html><html><head><title>Help for package net4pg</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {net4pg}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cc_composition'><p>Get peptides and peptide-to-protein mappings for each connected component</p></a></li>
<li><a href='#cc_stats'><p>Provide statistics on the CCs size</p></a></li>
<li><a href='#get_adj_matrix'><p>Generate adjacency matrix</p></a></li>
<li><a href='#get_cc'><p>Generate graph and calculate its connected components</p></a></li>
<li><a href='#peptide_stats'><p>Calculate percentage of shared vs specific peptides</p></a></li>
<li><a href='#plot_cc'><p>Plot peptide-to-protein mapping graph</p></a></li>
<li><a href='#read_inc_matrix'><p>Read incidence matrix of proteomic identifications</p></a></li>
<li><a href='#reduce_inc_matrix'><p>Reduce size of incidence matrix for downstream analyses</p></a></li>
<li><a href='#transcriptome_filter'><p>Perform transcriptome-informed post-hoc filtering</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Handle Ambiguity of Protein Identifications from Shotgun
Proteomics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Laura Fancello &lt;laura.fancello@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>In shotgun proteomics, shared peptides (i.e., peptides that might 
    originate from different proteins sharing homology, from different 
    proteoforms due to alternative mRNA splicing, post-translational 
    modifications, proteolytic cleavages, and/or allelic variants) represent a
    major source of ambiguity in protein identifications. The 'net4pg' package 
    allows to assess and handle ambiguity of protein identifications. It 
    implements methods for two main applications. First, it allows to represent
    and quantify ambiguity of protein identifications by means of graph 
    connected components (CCs). In graph theory, CCs are defined as the largest
    subgraphs in which any two vertices are connected to each other by a path 
    and not connected to any other of the vertices in the supergraph. Here, 
    proteins sharing one or more peptides are thus gathered in the same CC 
    (multi-protein CC), while unambiguous protein identifications constitute CCs 
    with a single protein vertex (single-protein CCs). Therefore, the proportion
    of single-protein CCs and the size of multi-protein CCs can be used to
    measure the level of ambiguity of protein identifications. The package
    implements a strategy to efficiently calculate graph connected
    components on large datasets and allows to visually inspect them.
    Secondly, the 'net4pg' package allows to exploit the increasing
    availability of matched transcriptomic and proteomic datasets to
    reduce ambiguity of protein identifications. More precisely, it
    implement a transcriptome-based filtering strategy fundamentally
    consisting in the removal of those proteins whose corresponding
    transcript is not expressed in the sample-matched transcriptome. The
    underlying assumption is that, according to the central dogma of
    biology, there can be no proteins without the corresponding
    transcript. Most importantly, the package allows to visually inspect
    the effect of the filtering on protein identifications and quantify
    ambiguity before and after filtering by means of graph connected
    components. As such, it constitutes a reproducible and transparent
    method to exploit transcriptome information to enhance protein
    identifications. All methods implemented in the 'net4pg' package are fully 
    described in Fancello and Burger (2022) &lt;<a href="https://doi.org/10.1186%2Fs13059-022-02701-2">doi:10.1186/s13059-022-02701-2</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/laurafancello/net4pg">https://github.com/laurafancello/net4pg</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/laurafancello/net4pg/issues">https://github.com/laurafancello/net4pg/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, graph, magrittr, Matrix, methods, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BiocStyle, ggplot2, igraph, knitr, rmarkdown, roxygen2,
testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-09-07 07:30:16 UTC; laura</td>
</tr>
<tr>
<td>Author:</td>
<td>Laura Fancello <a href="https://orcid.org/0000-0003-4708-4080"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Thomas Burger <a href="https://orcid.org/0000-0003-3539-3564"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-09-07 09:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='cc_composition'>Get peptides and peptide-to-protein mappings for each connected component</h2><span id='topic+cc_composition'></span>

<h3>Description</h3>

<p>Get peptides and peptide-to-protein mappings for each connected component.
For each connected component, first extract its protein members; then,
extract all specific and shared peptides mapping on those protein; finally,
extract the subset of incidence matrix representing peptide-to-protein
mappings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cc_composition(cc.proteins, incM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cc_composition_+3A_cc.proteins">cc.proteins</code></td>
<td>
<p>a <code>list</code> of <code>vectors</code> (one for each connected
component), each enumerating the proteins members of a connected component.</p>
</td></tr>
<tr><td><code id="cc_composition_+3A_incm">incM</code></td>
<td>
<p>a <code>logical</code> <code>matrix</code> containing the incidence matrix
with its column and row names (respectively, protein and peptide identifiers)
names and 0 or 1 values indicating whether or not the peptide maps on the
corresponding protein.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>list</code> of two elements: i. a <code>list</code> of <code>vectors</code>
(one for each connected component) enumerating peptides mapping on protein
members of each connected component; ii. a <code>list</code> of <code>matrices</code> or
<code>vectors</code> (one for each connected component) representing
peptide-to-protein mappings for each connected component; matrices are used
if multiple peptides identify protein members of that connected component,
vectors if only a single peptide.
</p>


<h3>Author(s)</h3>

<p>Laura Fancello
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read the tab-delimited file containing the proteome incidence matrix
incM_filename &lt;- system.file("extdata"
                             , "incM_example"
                             , package = "net4pg"
                             , mustWork = TRUE)
rownames_filename &lt;- system.file("extdata"
                                  , "peptideIDs_incM_example"
                                  , package = "net4pg"
                                  , mustWork = TRUE)
colnames_filename &lt;- system.file("extdata"
                                 , "proteinIDs_incM_example"
                                 , package = "net4pg"
                                 , mustWork = TRUE)
incM &lt;- read_inc_matrix(incM_filename = incM_filename
                 , colnames_filename = colnames_filename
                 , rownames_filename = rownames_filename)
# Only retain proteins with at least one shared peptide and all peptides
# mapping on such proteins.
incM_reduced &lt;- reduce_inc_matrix(incM)
# Generate adjacency matrix describing protein-to-protein mappings
adjM &lt;- get_adj_matrix(incM_reduced)
# Generate graph of protein-to-protein connections and calculate its
# connected component
multProteinCC &lt;- get_cc(adjM)
# For each connected component, extract peptides mapping on its protein
# members and the subset of the incidence matrix describing peptide-to-protein
# mapping
cc.peptides.incM &lt;- cc_composition(cc.proteins = multProteinCC$cc
                                  , incM = incM)

</code></pre>

<hr>
<h2 id='cc_stats'>Provide statistics on the CCs size</h2><span id='topic+cc_stats'></span>

<h3>Description</h3>

<p>Provides the CC size distribution, which is the number of CCs including a
given number of protein members, and the proportion of single- vs
multi-protein CCs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cc_stats(incM, cc.proteins, reducedIncM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cc_stats_+3A_incm">incM</code></td>
<td>
<p>a <code>logical</code> <code>matrix</code> containing the incidence matrix
with its column and row names (respectively, protein and peptide identifiers)
names and 0 or 1 values indicating whether or not the peptide maps on the
corresponding protein.</p>
</td></tr>
<tr><td><code id="cc_stats_+3A_cc.proteins">cc.proteins</code></td>
<td>
<p>a <code>list</code> of <code>vectors</code> (one for each connected
component), each enumerating the proteins members of a connected component.</p>
</td></tr>
<tr><td><code id="cc_stats_+3A_reducedincm">reducedIncM</code></td>
<td>
<p>a <code>logical</code> value indicating if the input matrix is
the original complete incidence matrix or a reduced version of it which only
contains proteins with at least one shared peptide and all peptides mapping
on such proteins (generated by the <code><a href="#topic+reduce_inc_matrix">reduce_inc_matrix</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>list</code> of four outputs: i. a <code>vector</code> containing all
those proteins which belong to single-protein connected component (each
protein is a single-protein connected component); ii. an <code>integer</code>
representing the number of single-protein CCs; iii. an <code>integer</code>
representing the number of multi-protein CCs; iv. a <code>data.frame</code>
describing size distribution of connected components (number of connected
components per number of protein members)
</p>


<h3>Author(s)</h3>

<p>Laura Fancello
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read the tab-delimited file containing he proteome incidence matrix
incM_filename &lt;- system.file("extdata"
                             , "incM_example"
                             , package = "net4pg"
                             , mustWork = TRUE)
rownames_filename &lt;- system.file("extdata"
                                  , "peptideIDs_incM_example"
                                  , package = "net4pg"
                                  , mustWork = TRUE)
colnames_filename &lt;- system.file("extdata"
                                 , "proteinIDs_incM_example"
                                 , package = "net4pg"
                                 , mustWork = TRUE)
incM &lt;- read_inc_matrix(incM_filename = incM_filename
                 , colnames_filename = colnames_filename
                 , rownames_filename = rownames_filename)
# Only retain proteins with at least one shared peptide and all peptides
# mapping on such proteins.
incM_reduced &lt;- reduce_inc_matrix(incM)
# Generate adjacency matrix describing protein-to-protein mappings
adjM &lt;- get_adj_matrix(incM_reduced)
# Generate graph of protein-to-protein connections and calculate its
# connected components
multProteinCC &lt;- get_cc(adjM)
# CCs size and percentage of single-vs multi-protein CCs
CCstatsOut &lt;- cc_stats(incM = incM_reduced
                      , cc.proteins = multProteinCC$ccs
                      , reducedIncM = TRUE)

</code></pre>

<hr>
<h2 id='get_adj_matrix'>Generate adjacency matrix</h2><span id='topic+get_adj_matrix'></span>

<h3>Description</h3>

<p>Generate an adjacency matrix representing protein-to-protein connections,
based on shared peptides. It is generated by cross product of the incidence
matrix of peptide-to-protein mappings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_adj_matrix(incM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_adj_matrix_+3A_incm">incM</code></td>
<td>
<p>a <code>logical</code> <code>matrix</code> containing the incidence matrix
with its column and row names (respectively, protein and peptide identifiers)
names and 0 or 1 values indicating whether or not the peptide maps on the
corresponding protein.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>numeric</code> <code>matrix</code> containing the adjacency matrix,
with value &gt;0 or 0 indicating whether or not two proteins are identified by
shared peptide(s)
</p>


<h3>Author(s)</h3>

<p>Laura Fancello
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read the tab-delimited file containing he proteome incidence matrix
incM_filename &lt;- system.file( "extdata"
                             , "incM_example"
                             , package = "net4pg"
                             , mustWork = TRUE)
rownames_filename &lt;- system.file( "extdata"
                                  , "peptideIDs_incM_example"
                                  , package = "net4pg"
                                  , mustWork = TRUE)
colnames_filename &lt;- system.file( "extdata"
                                 , "proteinIDs_incM_example"
                                 , package = "net4pg"
                                 , mustWork = TRUE)
incM &lt;- read_inc_matrix(incM_filename=incM_filename
                 , colnames_filename=colnames_filename
                 , rownames_filename=rownames_filename)
# Only retain proteins with at least one shared peptide and all peptides
# mapping on such proteins.
incM_reduced &lt;- reduce_inc_matrix(incM)
# Generate adjacency matrix describing protein-to-protein mappings
adjM &lt;- get_adj_matrix(incM_reduced)

</code></pre>

<hr>
<h2 id='get_cc'>Generate graph and calculate its connected components</h2><span id='topic+get_cc'></span>

<h3>Description</h3>

<p>Build a graph of protein-to-protein connections from adjacency matrix and
calculate its connected components.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cc(adjM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cc_+3A_adjm">adjM</code></td>
<td>
<p>a <code>numerical</code> <code>matrix</code> containing the adjacency matrix,
with value &gt;0 or 0 indicating whether or not two proteins are identified by
shared peptide(s)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>list</code> of two elements: i. a <code>graph</code> representing
protein-to-protein connections encoded by the adjacency matrix; ii. a
<code>list</code> of <code>vectors</code> (one for each connected component) enumerating
protein members of each connected component.
</p>


<h3>Author(s)</h3>

<p>Laura Fancello
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read the tab-delimited file containing he proteome incidence matrix
incM_filename &lt;- system.file( "extdata"
                             , "incM_example"
                             , package = "net4pg"
                             , mustWork = TRUE)
rownames_filename &lt;- system.file( "extdata"
                                  , "peptideIDs_incM_example"
                                  , package = "net4pg"
                                  , mustWork = TRUE)
colnames_filename &lt;- system.file( "extdata"
                                 , "proteinIDs_incM_example"
                                 , package = "net4pg"
                                 , mustWork = TRUE)
incM &lt;- read_inc_matrix(incM_filename=incM_filename
                 , colnames_filename=colnames_filename
                 , rownames_filename=rownames_filename)
# Only retain proteins with at least one shared peptide and all peptides
# mapping on such proteins.
incM_reduced &lt;- reduce_inc_matrix(incM)
# Generate adjacency matrix describing protein-to-protein mappings
adjM &lt;- get_adj_matrix(incM_reduced)
# Generate graph of protein-to-protein connections and calculate its
# connected components
multProteinCC &lt;- get_cc(adjM)

</code></pre>

<hr>
<h2 id='peptide_stats'>Calculate percentage of shared vs specific peptides</h2><span id='topic+peptide_stats'></span>

<h3>Description</h3>

<p>Read in input the incidence matrix of peptide-to-protein mappings generated
from valid proteomic identifications
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peptide_stats(incM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peptide_stats_+3A_incm">incM</code></td>
<td>
<p>a <code>logical</code> <code>matrix</code> containing the incidence matrix
with its column and row names (respectively, protein and peptide identifiers)
and 0 or 1 values indicating whether or not the peptide maps on the
corresponding protein.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>list</code> of three elements: i. number of shared peptides;
ii. number of specific peptides; iii. percentage of specific peptides
</p>


<h3>Author(s)</h3>

<p>Laura Fancello
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read the tab-delimited file containing he proteome incidence matrix
incM_filename &lt;- system.file( "extdata"
                             , "incM_example"
                             , package = "net4pg"
                             , mustWork = TRUE)
rownames_filename &lt;- system.file( "extdata"
                                  , "peptideIDs_incM_example"
                                  , package = "net4pg"
                                  , mustWork = TRUE)
colnames_filename &lt;- system.file( "extdata"
                                 , "proteinIDs_incM_example"
                                 , package = "net4pg"
                                 , mustWork = TRUE)
incM &lt;- read_inc_matrix(incM_filename = incM_filename
                 , colnames_filename = colnames_filename
                 , rownames_filename = rownames_filename)
# Calculate percentage of shared vs specific peptides
peptideStatsOut &lt;- peptide_stats(incM = incM)

</code></pre>

<hr>
<h2 id='plot_cc'>Plot peptide-to-protein mapping graph</h2><span id='topic+plot_cc'></span>

<h3>Description</h3>

<p>Plot a bipartite subgraph representing the connected component to which the
user-selected protein belongs. Peptide-to-protein mappings of that protein
and of all other proteins belonging to the same CC are represented. The
function takes in input a single Ensembl protein identifier (i.e. ENSPXXX
for human, ENSMUSPXXX for mouse), it identifies the connected component it
belongs to and plots all peptide-to-protein mappings of that connected
component.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_cc(prot, cc.proteins, cc.subincM, tagProt, tagContam, incM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_cc_+3A_prot">prot</code></td>
<td>
<p>a <code>character</code> <code>vector</code> containing a single Ensembl
identifier (i.e. ENSPXXXXXXXXXXX for human, ENSMUSPXXXXXXXXXXX for mouse) of
the protein of interest.</p>
</td></tr>
<tr><td><code id="plot_cc_+3A_cc.proteins">cc.proteins</code></td>
<td>
<p>a <code>list</code> of <code>vectors</code> (one for each connected
component) containing protein members of each connected component.</p>
</td></tr>
<tr><td><code id="plot_cc_+3A_cc.subincm">cc.subincM</code></td>
<td>
<p>a <code>list</code> of <code>matrices</code> or <code>vectors</code> (one for
each connected component) representing the incidence matrix of
peptide-to-protein mappings for each connected component; matrices are used
if multiple peptides identify protein members of that connected component,
vectors if only a single peptide.</p>
</td></tr>
<tr><td><code id="plot_cc_+3A_tagprot">tagProt</code></td>
<td>
<p>a <code>character</code> <code>vector</code> reporting the prefix of
protein identifiers (for non contaminant proteins)</p>
</td></tr>
<tr><td><code id="plot_cc_+3A_tagcontam">tagContam</code></td>
<td>
<p>a <code>character</code> <code>vector</code> reporting the tag which
identifies contaminant proteins</p>
</td></tr>
<tr><td><code id="plot_cc_+3A_incm">incM</code></td>
<td>
<p>a <code>logical</code> <code>matrix</code> containing the incidence matrix
with its column and row names (respectively, protein and peptide identifiers)
names and 0 or 1 values indicating whether or not the peptide maps on the
corresponding protein.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of four elements: i. CC identifier; ii. protein members of the
CC; iii. peptide members of the CC; iv. bipartite subgraph representing
peptide-to-protein mapping of that CC (if multi-protein CC)
</p>


<h3>Author(s)</h3>

<p>Laura Fancello
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(igraph)
# Read the tab-delimited file containing he proteome incidence matrix
incM_filename &lt;- system.file( "extdata"
                             , "incM_example"
                             , package = "net4pg"
                             , mustWork = TRUE)
rownames_filename &lt;- system.file( "extdata"
                                  , "peptideIDs_incM_example"
                                  , package = "net4pg"
                                  , mustWork = TRUE)
colnames_filename &lt;- system.file( "extdata"
                                 , "proteinIDs_incM_example"
                                 , package = "net4pg"
                                 , mustWork = TRUE)
incM &lt;- read_inc_matrix(incM_filename = incM_filename
                 , colnames_filename = colnames_filename
                 , rownames_filename = rownames_filename)
# Only retain proteins with at least one shared peptide and all peptides
# mapping on such proteins.
incM_reduced &lt;- reduce_inc_matrix(incM)
# Generate adjacency matrix describing protein-to-protein mappings
adjM &lt;- get_adj_matrix(incM_reduced)
# Generate graph of protein-to-protein connections and calculate its
# connected components
multProteinCC &lt;- get_cc(adjM)
# For each connected component, extract peptides mapping on its protein
# members and the subset of the incidence matrix describing
# peptide-to-protein mappings
cc.peptides.incM &lt;- cc_composition(cc.proteins = multProteinCC$cc
                                   , incM = incM)
# Plot bipartite graph representing peptide-to-protein mappings for the
# connected component of the protein of interest (in this toy example protein
# "ENSP261"; note that identifiers are not authentic but made up for the
# example)
subgraphCC &lt;- plot_cc(prot="ENSP261"
        , cc.proteins=multProteinCC$ccs
        , cc.subincM=cc.peptides.incM$cc.subincM
        , tagProt = "ENSP"
        , tagContam="Contam"
        , incM=incM)
plot.igraph(subgraphCC$g
      , layout = layout_as_bipartite
      , edge.width = 1
      , edge.arrow.width = 0.3
      , vertex.size = 10
      , edge.arrow.size = 0.5
      , vertex.size2 = 3
      , vertex.label.cex = 0.8
      , asp = 0.35
      , margin = -0.1) +
title(paste0("Protein ENSP261 in CC #", subgraphCC$cc_id), line = -1)

</code></pre>

<hr>
<h2 id='read_inc_matrix'>Read incidence matrix of proteomic identifications</h2><span id='topic+read_inc_matrix'></span>

<h3>Description</h3>

<p>Read in input the incidence matrix of peptide-to-protein mappings generated
from valid proteomic identifications.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_inc_matrix(incM_filename, colnames_filename, rownames_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_inc_matrix_+3A_incm_filename">incM_filename</code></td>
<td>
<p>the name of the tab-delimited file containing incidence
matrix values; the input incidence matrix must contain along the columns
protein identifiers and along the rows peptides; each cell must contain a 1
or 0 value indicating whether or not the peptide maps on the corresponding
protein.</p>
</td></tr>
<tr><td><code id="read_inc_matrix_+3A_colnames_filename">colnames_filename</code></td>
<td>
<p>name of the file containing incidence matrix column
names, which are protein identifiers. The file must contain one identifier
per line. Protein identifiers must be in Ensembl format (i.e.,
ENSPXXXXXXXXXXX for human).</p>
</td></tr>
<tr><td><code id="read_inc_matrix_+3A_rownames_filename">rownames_filename</code></td>
<td>
<p>name of the file containing incidence matrix row
names, which are peptide identifiers. The file must contain one identifier
per line. Peptide identifiers can be in any format (i.e. numeric identifiers,
amino acid sequences, ...)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>logical</code> <code>matrix</code> containing the incidence matrix with
its column and row names (respectively, protein and peptide identifiers)
and 0 or 1 values indicating whether or not the peptide maps on the
corresponding protein.
</p>


<h3>Author(s)</h3>

<p>Laura Fancello
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read the tab-delimited file containing he proteome incidence matrix
incM_filename &lt;- system.file( "extdata"
                             , "incM_example"
                             , package = "net4pg"
                             , mustWork = TRUE)
rownames_filename &lt;- system.file( "extdata"
                                  , "peptideIDs_incM_example"
                                  , package = "net4pg"
                                  , mustWork = TRUE)
colnames_filename &lt;- system.file( "extdata"
                                 , "proteinIDs_incM_example"
                                 , package = "net4pg"
                                 , mustWork = TRUE)
incM &lt;- read_inc_matrix(incM_filename = incM_filename
                 , colnames_filename = colnames_filename
                 , rownames_filename = rownames_filename)

</code></pre>

<hr>
<h2 id='reduce_inc_matrix'>Reduce size of incidence matrix for downstream analyses</h2><span id='topic+reduce_inc_matrix'></span>

<h3>Description</h3>

<p>Reduce the size of the incidence matrix describing peptide-to-protein
mappings to ease downstream analyses. The original incidence matrix is
reduced to only contain proteins with at least one shared peptide and all
peptides mapping on such proteins. This means that only proteins ambiguously
identified are retained, which is the most interesting ones when studying
ambiguity of protein identifications.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reduce_inc_matrix(incM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reduce_inc_matrix_+3A_incm">incM</code></td>
<td>
<p>a <code>logical</code> <code>matrix</code> containing the incidence matrix
with its column and row names (respectively, protein and peptide identifiers)
and 0 or 1 values indicating whether or not the peptide maps on the
corresponding protein.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>logical</code> <code>matrix</code> containing a smaller incidence matrix
(with column and row names respectively reporting protein and peptide
identifiers) and 0 or 1 values indicating whether or not the peptide maps on
the corresponding protein. Only proteins with at least one shared peptide and
all peptides mapping on such protein are reported in such reduced incidence
matrix.
</p>


<h3>Author(s)</h3>

<p>Laura Fancello
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read the tab-delimited file containing he proteome incidence matrix
incM_filename &lt;- system.file("extdata"
                             , "incM_example"
                             , package = "net4pg"
                             , mustWork = TRUE)
rownames_filename &lt;- system.file("extdata"
                                  , "peptideIDs_incM_example"
                                  , package = "net4pg"
                                  , mustWork = TRUE)
colnames_filename &lt;- system.file("extdata"
                                 , "proteinIDs_incM_example"
                                 , package = "net4pg"
                                 , mustWork = TRUE)
incM &lt;- read_inc_matrix(incM_filename = incM_filename
                 , colnames_filename = colnames_filename
                 , rownames_filename = rownames_filename)
# Only retain proteins with at least one shared peptide and all peptides
# mapping on such proteins.
incM_reduced &lt;- reduce_inc_matrix(incM)

</code></pre>

<hr>
<h2 id='transcriptome_filter'>Perform transcriptome-informed post-hoc filtering</h2><span id='topic+transcriptome_filter'></span>

<h3>Description</h3>

<p>Implement the transcriptome-informed post-hoc filtering strategy. This
strategy aims to reduce the ambiguity of protein identifications by
exploiting sample-matched transcriptome information, when available. First,
it takes in input the set of transcripts expressed in the sample-matched
transcriptome (reported using the transcript identifier in Ensembl format,
i.e., ENSTXXXX for human) and removes from proteomic identifications:
i. all proteins with no expressed transcripts and peptides exclusively
mapping on removed proteins (&quot;all&quot;); or
ii. only those exclusively identified
by shared peptides and peptides exclusively mapping on removed proteins
(&quot;sharedOnly&quot;); or
iii. only those exclusively identified by shared peptides,
whose peptides are shared with at least one protein with expressed
transcript, so they are not to be removed (&quot;sharedNoRemove&quot;)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transcriptome_filter(
  incM,
  exprTranscriptsFile,
  proteinToTranscriptFile,
  tagContam,
  remove
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transcriptome_filter_+3A_incm">incM</code></td>
<td>
<p>a <code>logical</code> <code>matrix</code> containing the incidence matrix
with its column and row names (respectively, protein and peptide identifiers)
and 0 or 1 values indicating whether or not the peptide maps on the
corresponding protein.</p>
</td></tr>
<tr><td><code id="transcriptome_filter_+3A_exprtranscriptsfile">exprTranscriptsFile</code></td>
<td>
<p>the name of the file containing the set of
transcripts expressed in the sample-matched transcriptome (one per line).
Transcript identifiers must be in the Ensembl format (i.e., ENSTXXXXXXXXXXX
for human)</p>
</td></tr>
<tr><td><code id="transcriptome_filter_+3A_proteintotranscriptfile">proteinToTranscriptFile</code></td>
<td>
<p>the name of a tab-delimited file with protein
identifiers in the first column and the corresponding transcript identifiers
in the second column. Protein and transcript identifiers must be in the
Ensembl format (i.e. ENSPXXXXXXXXXXX and ENSTXXXXXXXXXXX for human)</p>
</td></tr>
<tr><td><code id="transcriptome_filter_+3A_tagcontam">tagContam</code></td>
<td>
<p>a <code>character</code> <code>vector</code> reporting the tag which
identifies contaminant protein</p>
</td></tr>
<tr><td><code id="transcriptome_filter_+3A_remove">remove</code></td>
<td>
<p><code>character</code> <code>vector</code> indicating whether to remove:
i. all proteins with no expressed transcripts and peptides exclusively
mapping on removed proteins (&quot;all&quot;); ii. only those exclusively identified
by shared peptides and peptides exclusively mapping on removed proteins
(&quot;sharedOnly&quot;); iii. only those exclusively identified by shared peptides,
whose peptides are shared with at least one protein with expressed
transcript, so they are not to be removed (&quot;sharedNoRemove&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>matrix</code> representing a filtered incidence matrix of
peptide-to-protein mapping obtained by transcriptome-informed filtering.
</p>


<h3>Author(s)</h3>

<p>Laura Fancello
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read the tab-delimited file containing the proteome incidence matrix
incM_filename &lt;- system.file("extdata"
                             , "incM_example"
                             , package = "net4pg"
                             , mustWork = TRUE)
rownames_filename &lt;- system.file("extdata"
                                  , "peptideIDs_incM_example"
                                  , package = "net4pg"
                                  , mustWork = TRUE)
colnames_filename &lt;- system.file("extdata"
                                 , "proteinIDs_incM_example"
                                 , package = "net4pg"
                                 , mustWork = TRUE)
incM &lt;- read_inc_matrix(incM_filename = incM_filename
                 , colnames_filename = colnames_filename
                 , rownames_filename = rownames_filename)
# Perform transcriptome-informed post-hoc filtering
exprTranscriptsFile &lt;- system.file("extdata"
                                   , "expressed_transcripts.txt"
                                   , package = "net4pg"
                                   , mustWork = TRUE)
protein2transcriptFile &lt;- system.file("extdata"
                                        , "protein_to_transcript"
                                        , package = "net4pg"
                                        , mustWork = TRUE)
incM_filtered &lt;- transcriptome_filter(incM
                         , exprTranscriptsFile = exprTranscriptsFile
                         , proteinToTranscriptFile = protein2transcriptFile
                         , tagContam = "Contam"
                         , remove = "all")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
