<!DOCTYPE html><html><head><title>Help for package netmhc2pan</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {netmhc2pan}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#capitalize_first_char'><p>Convert the first character to upper case.</p></a></li>
<li><a href='#check_alleles'><p>Check the allele names</p></a></li>
<li><a href='#check_can_create_file'><p>Check that a file can be created at a certain path.</p></a></li>
<li><a href='#check_fasta_file_sequence_lengths'><p>Check the lengths of the sequences in a FASTA file</p></a></li>
<li><a href='#check_netmhc2pan_bin_url'><p>Check the URL of the NetMHCIIpan binary tarball</p></a></li>
<li><a href='#check_netmhc2pan_data_url'><p>Check the URL of the NetMHCIIpan binary tarball</p></a></li>
<li><a href='#check_netmhc2pan_installation'><p>Checks the installation of NetMHCIIpan.</p></a></li>
<li><a href='#create_temp_fasta_filename'><p>Create a <code>.fasta</code> file</p></a></li>
<li><a href='#create_temp_xls_filename'><p>Create a <code>.xls</code> file</p></a></li>
<li><a href='#default_params_doc'><p>This function does nothing. It is intended to inherit is parameters'</p>
documentation.</a></li>
<li><a href='#download_netmhc2pan_bin'><p>Download the NetMHCIIpan binary</p></a></li>
<li><a href='#get_default_netmhc2pan_bin_path'><p>Get the full path to the default NetMHC2pan binary</p></a></li>
<li><a href='#get_default_netmhc2pan_bin_tarfile_path'><p>Get the default path for the NetMHCIIpan binary tarball file</p></a></li>
<li><a href='#get_default_netmhc2pan_folder'><p>Get the path to the folder where this package installs</p>
NetMHCIIpan by default</a></li>
<li><a href='#get_default_netmhc2pan_subfolder'><p>Get the full path to the NetMHC2pan sub-folder</p></a></li>
<li><a href='#get_default_netmhc2pan_version'><p>Get the default NetMHC2pan version used</p></a></li>
<li><a href='#get_netmhc2pan_alleles'><p>Get a list of the alleles supported by NetMHCIIpan</p></a></li>
<li><a href='#get_netmhc2pan_archive_filename'><p>Get the filename of the NetMHC2pan archive file</p></a></li>
<li><a href='#get_netmhc2pan_bin_url'><p>Get the NetMHCIIpan binary download URL.</p></a></li>
<li><a href='#get_netmhc2pan_data_url'><p>Get the NetMHCIIpan data tarball URL.</p></a></li>
<li><a href='#get_netmhc2pan_url'><p>Deprecated, use get_netmhc2pan_bin_url instead</p></a></li>
<li><a href='#install_netmhc2pan'><p>Install NetMHCIIpan</p></a></li>
<li><a href='#install_netmhc2pan_bin'><p>Install the NetMHCIIpan binary to a local folder</p></a></li>
<li><a href='#install_netmhc2pan_bin_from_file'><p>Install the NetMHCIIpan binary to a local folder</p></a></li>
<li><a href='#install_netmhc2pan_data'><p>Install the NetMHCIIpan data to a local folder</p></a></li>
<li><a href='#install_netmhc2pan_data_from_file'><p>Install the NetMHCIIpan data from file</p></a></li>
<li><a href='#install_netmhc2pan_from_files'><p>Install NetMHCIIpan from files.</p></a></li>
<li><a href='#is_netmhc2pan_bin_installed'><p>Measure if NetMHCIIpan binary is installed locally</p></a></li>
<li><a href='#is_netmhc2pan_data_installed'><p>Measure if NetMHCIIpan data folder is installed locally</p></a></li>
<li><a href='#is_netmhc2pan_haplotype'><p>Is the haplotype valid?</p></a></li>
<li><a href='#is_netmhc2pan_installed'><p>Measure if NetMHCIIpan is installed locally</p></a></li>
<li><a href='#is_netmhc2pan_set_up'><p>Measure if NetMHCIIpan is set up</p></a></li>
<li><a href='#is_on_appveyor'><p>Determines if the environment is AppVeyor</p></a></li>
<li><a href='#is_on_ci'><p>Determines if the environment is a continuous integration service</p></a></li>
<li><a href='#is_on_github_actions'><p>Determines if the environment is GitHub Actions</p></a></li>
<li><a href='#is_on_travis'><p>Determines if the environment is Travis CI</p></a></li>
<li><a href='#is_tcsh_installed'><p>See if tcsh is installed</p></a></li>
<li><a href='#is_url_valid'><p>Is the download URL valid?</p></a></li>
<li><a href='#netmhc2pan'><p>netmhc2pan: predict peptide binding strengths using NetMHC2pan</p></a></li>
<li><a href='#netmhc2pan_report'><p>Show a <code>netmhc2pan</code> report using message,</p>
to be used when reporting bugs</a></li>
<li><a href='#netmhc2pan_self_test'><p>Test NetMHCIIpan by doing a minimal run.</p></a></li>
<li><a href='#predict_ic50'><p>Predict the IC50 for peptides.</p></a></li>
<li><a href='#predict_ic50s'><p>Predict the IC50s from a sequence</p></a></li>
<li><a href='#read_netmhc2pan_xls_output'><p>Read MHC2pan output</p></a></li>
<li><a href='#run_netmhc2pan'><p>Run NetMHCIIpan</p></a></li>
<li><a href='#set_up_netmhc2pan'><p>Install the NetMHCIIpan binary to a local folder</p></a></li>
<li><a href='#to_netmhc2pan_name'><p>Convert a formal haplotype name to its NetMHC2pan notation</p></a></li>
<li><a href='#uninstall_netmhc2pan'><p>Uninstall NetMHCIIpan</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Interface to 'NetMHCIIpan'</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Richèl J.C. Bilderbeek &lt;richel@richelbilderbeek.nl&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The field of immunology benefits from software that can
    predict which peptide sequences trigger an immune response.
    'NetMHCIIpan' is a such a tool: it predicts the
    binding strength of a short peptide to a Major Histocompatibility
    Complex class II (MHC-II) molecule.
    'NetMHCIIpan' can be used from a web server at 
    <a href="https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/">https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/</a>
    or from the command-line, using a local installation. This package
    allows to call 'NetMHCIIpan' from R.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>devtools, dplyr, rappdirs, readr, seqinr, stringr, testit,
tibble</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat, spelling</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/richelbilderbeek/netmhc2pan/">https://github.com/richelbilderbeek/netmhc2pan/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/richelbilderbeek/netmhc2pan/issues">https://github.com/richelbilderbeek/netmhc2pan/issues</a></td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>NetMHC2pan
(https://services.healthtech.dtu.dk/service.php?NetMHCIIpan-3.2)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-08 16:05:13 UTC; richel</td>
</tr>
<tr>
<td>Author:</td>
<td>Richèl J.C. Bilderbeek
    <a href="https://orcid.org/0000-0003-1107-7049"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Lue Ping Zhao [ctb],
  Byungkuk Min [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-08 23:10:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='capitalize_first_char'>Convert the first character to upper case.</h2><span id='topic+capitalize_first_char'></span>

<h3>Description</h3>

<p>Convert the first character to upper case. If the first
character is upper case, nothing will happen.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>capitalize_first_char(s)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="capitalize_first_char_+3A_s">s</code></td>
<td>
<p>a string</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the string, with the first letter in uppercase
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>capitalize_first_char("hello")
</code></pre>

<hr>
<h2 id='check_alleles'>Check the allele names</h2><span id='topic+check_alleles'></span>

<h3>Description</h3>

<p>Check if the alleles have a valid NetMHC2pan name.
Will <a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_alleles(
  alleles,
  netmhc2pan_folder_name = get_default_netmhc2pan_folder()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_alleles_+3A_alleles">alleles</code></td>
<td>
<p>one or more alleles, e.g. <code>DRB1_0101</code>.
See <a href="#topic+get_netmhc2pan_alleles">get_netmhc2pan_alleles</a> for a full list</p>
</td></tr>
<tr><td><code id="check_alleles_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+to_netmhc2pan_name">to_netmhc2pan_name</a> to convert
a formal name to an NetMHC2pan name.
Use <a href="#topic+get_netmhc2pan_alleles">get_netmhc2pan_alleles</a> for a list
of all supported alleles (in NetMHC2pan naming format).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_netmhc2pan_installed()) {
  check_alleles("DRB1_0101")
  check_alleles(c("DRB1_0102", "DRB1_0103"))
}
</code></pre>

<hr>
<h2 id='check_can_create_file'>Check that a file can be created at a certain path.</h2><span id='topic+check_can_create_file'></span>

<h3>Description</h3>

<p>Will <a href="base.html#topic+stop">stop</a> if not. Will <a href="base.html#topic+stop">stop</a> if the file already exists.
Does so by creating an empty file at the path,
and then deleting it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_can_create_file(filename, overwrite = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_can_create_file_+3A_filename">filename</code></td>
<td>
<p>file that may or may not be created</p>
</td></tr>
<tr><td><code id="check_can_create_file_+3A_overwrite">overwrite</code></td>
<td>
<p>if TRUE, if <code>filename</code> already exists, it
will be deleted by this function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='check_fasta_file_sequence_lengths'>Check the lengths of the sequences in a FASTA file</h2><span id='topic+check_fasta_file_sequence_lengths'></span>

<h3>Description</h3>

<p>Check if the lengths of the sequences in a FASTA file
are at least equal to the desired peptide lengths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_fasta_file_sequence_lengths(fasta_filename, peptide_length)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_fasta_file_sequence_lengths_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>the name of a FASTA file with protein sequences</p>
</td></tr>
<tr><td><code id="check_fasta_file_sequence_lengths_+3A_peptide_length">peptide_length</code></td>
<td>
<p>length of a peptide</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'># FASTA file in which all proteons have a length of at keast 13
fasta_filename &lt;- system.file(
  "extdata", "example.fasta", package = "netmhc2pan"
)
check_fasta_file_sequence_lengths(
  fasta_filename = fasta_filename,
  peptide_length = 13
)
</code></pre>

<hr>
<h2 id='check_netmhc2pan_bin_url'>Check the URL of the NetMHCIIpan binary tarball</h2><span id='topic+check_netmhc2pan_bin_url'></span>

<h3>Description</h3>

<p>Check the URL of the NetMHCIIpan binary tarball is valid,
will stop if not.
This URL link expires after 4 hours.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_netmhc2pan_bin_url(
  netmhc2pan_bin_url = get_netmhc2pan_bin_url(),
  verbose = FALSE,
  netmhc2pan_archive_filename = get_netmhc2pan_archive_filename(),
  temp_local_file = tempfile(pattern = "check_netmhc2pan_bin_url_")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_netmhc2pan_bin_url_+3A_netmhc2pan_bin_url">netmhc2pan_bin_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan binary tarball file from,
similar to, for example,
<code>https://www.cbs.dtu.dk/download/12345678-1234-1234-1234-123456789ABC</code>.
Use <a href="#topic+get_netmhc2pan_bin_url">get_netmhc2pan_bin_url</a> to get the default URL.</p>
</td></tr>
<tr><td><code id="check_netmhc2pan_bin_url_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
<tr><td><code id="check_netmhc2pan_bin_url_+3A_netmhc2pan_archive_filename">netmhc2pan_archive_filename</code></td>
<td>
<p>the NetMHC2pan archive
filename, for example <code>netMHCIIpan-3.2.Linux.tar.gz</code>,
which is used in installation of NetMHC2pan.
Use <a href="#topic+get_netmhc2pan_archive_filename">get_netmhc2pan_archive_filename</a> to get this filename.</p>
</td></tr>
<tr><td><code id="check_netmhc2pan_bin_url_+3A_temp_local_file">temp_local_file</code></td>
<td>
<p>path to the temporary file to store the URL to.
This file will be deleted afterwards.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>

<hr>
<h2 id='check_netmhc2pan_data_url'>Check the URL of the NetMHCIIpan binary tarball</h2><span id='topic+check_netmhc2pan_data_url'></span>

<h3>Description</h3>

<p>Check the URL of the NetMHCIIpan binary tarball is valid,
will stop if not.
This URL link expires after 4 hours.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_netmhc2pan_data_url(
  netmhc2pan_data_url = get_netmhc2pan_data_url(),
  verbose = FALSE,
  temp_local_file = tempfile(pattern = "check_netmhc2pan_data_url_")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_netmhc2pan_data_url_+3A_netmhc2pan_data_url">netmhc2pan_data_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan data tarball file from,
similar to, for example,
<a href="https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz">https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz</a>.
Use <a href="#topic+get_netmhc2pan_data_url">get_netmhc2pan_data_url</a> to get the default URL.</p>
</td></tr>
<tr><td><code id="check_netmhc2pan_data_url_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
<tr><td><code id="check_netmhc2pan_data_url_+3A_temp_local_file">temp_local_file</code></td>
<td>
<p>path to the temporary file to store the URL to.
This file will be deleted afterwards.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>

<hr>
<h2 id='check_netmhc2pan_installation'>Checks the installation of NetMHCIIpan.</h2><span id='topic+check_netmhc2pan_installation'></span>

<h3>Description</h3>

<p>Checks the installation of NetMHCIIpan.
Throws a helpful error message if incomplete,
else does nothing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_netmhc2pan_installation(
  netmhc2pan_folder_name = get_default_netmhc2pan_folder()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_netmhc2pan_installation_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Will stop if NetMHC2pan is not installed
try(check_netmhc2pan_installation())
</code></pre>

<hr>
<h2 id='create_temp_fasta_filename'>Create a <code>.fasta</code> file</h2><span id='topic+create_temp_fasta_filename'></span>

<h3>Description</h3>

<p>Create a <code>.fasta</code> file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_temp_fasta_filename()
</code></pre>


<h3>Value</h3>

<p>a path to a non-existing file,
for example, <code>/home/myusername/.cache/temp_582046426735.fasta</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>create_temp_fasta_filename()
</code></pre>

<hr>
<h2 id='create_temp_xls_filename'>Create a <code>.xls</code> file</h2><span id='topic+create_temp_xls_filename'></span>

<h3>Description</h3>

<p>Create a <code>.xls</code> file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_temp_xls_filename()
</code></pre>


<h3>Value</h3>

<p>a path to a non-existing file,
for example, <code>/home/myusername/.cache/temp_582047dac733.xls</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>create_temp_xls_filename()
</code></pre>

<hr>
<h2 id='default_params_doc'>This function does nothing. It is intended to inherit is parameters'
documentation.</h2><span id='topic+default_params_doc'></span>

<h3>Description</h3>

<p>This function does nothing. It is intended to inherit is parameters'
documentation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>default_params_doc(
  alleles,
  do_filter,
  fasta_filename,
  folder_name,
  mhc_haplotype,
  netmhc2pan_archive_filename,
  netmhc2pan_bin_tarfile_path,
  netmhc2pan_bin_url,
  netmhc2pan_data_tarfile_path,
  netmhc2pan_data_url,
  netmhc2pan_folder_name,
  netmhc2pan_subfolder,
  netmhc2pan_version,
  os,
  peptide_length,
  peptides,
  protein_sequence,
  temp_fasta_filename,
  temp_local_file,
  temp_xls_filename,
  verbose,
  xls_filename
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="default_params_doc_+3A_alleles">alleles</code></td>
<td>
<p>one or more alleles, e.g. <code>DRB1_0101</code>.
See <a href="#topic+get_netmhc2pan_alleles">get_netmhc2pan_alleles</a> for a full list</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_do_filter">do_filter</code></td>
<td>
<p>set to TRUE if the results of NetMHCIIpan must be filtered</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>the name of a FASTA file with protein sequences</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_folder_name">folder_name</code></td>
<td>
<p>the folder to install NetMHCIIpan, which is
<code>"/home/[usename]/.local/share"</code> by default</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mhc_haplotype">mhc_haplotype</code></td>
<td>
<p>one MHC haplotype, e.g. <code>DRB1_0101</code>.
See <a href="#topic+get_netmhc2pan_alleles">get_netmhc2pan_alleles</a> for a full list</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_netmhc2pan_archive_filename">netmhc2pan_archive_filename</code></td>
<td>
<p>the NetMHC2pan archive
filename, for example <code>netMHCIIpan-3.2.Linux.tar.gz</code>,
which is used in installation of NetMHC2pan.
Use <a href="#topic+get_netmhc2pan_archive_filename">get_netmhc2pan_archive_filename</a> to get this filename.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_netmhc2pan_bin_tarfile_path">netmhc2pan_bin_tarfile_path</code></td>
<td>
<p>path of the
NetMHCIIpan binary tarball file</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_netmhc2pan_bin_url">netmhc2pan_bin_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan binary tarball file from,
similar to, for example,
<code>https://www.cbs.dtu.dk/download/12345678-1234-1234-1234-123456789ABC</code>.
Use <a href="#topic+get_netmhc2pan_bin_url">get_netmhc2pan_bin_url</a> to get the default URL.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_netmhc2pan_data_tarfile_path">netmhc2pan_data_tarfile_path</code></td>
<td>
<p>path of the
NetMHCIIpan data tarball file</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_netmhc2pan_data_url">netmhc2pan_data_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan data tarball file from,
similar to, for example,
<a href="https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz">https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz</a>.
Use <a href="#topic+get_netmhc2pan_data_url">get_netmhc2pan_data_url</a> to get the default URL.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_netmhc2pan_subfolder">netmhc2pan_subfolder</code></td>
<td>
<p>the subfolder (to be) used by NetMHCIIpan.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_netmhc2pan_version">netmhc2pan_version</code></td>
<td>
<p>the NetMHCIIpan version,
for example <code>3.2</code>.
Use <a href="#topic+get_default_netmhc2pan_version">get_default_netmhc2pan_version</a> to get the default
NetMHCIIpan version</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_os">os</code></td>
<td>
<p>the operating system as obtained by <code>rappdirs::app_dir()$os</code>.
<code>netmhc2pan</code> supports Linux ('<code>unix</code>') only</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_peptide_length">peptide_length</code></td>
<td>
<p>length of a peptide</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_peptides">peptides</code></td>
<td>
<p>one or more peptide sequences</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_protein_sequence">protein_sequence</code></td>
<td>
<p>a protein sequence, for example <code>FAMILYVW</code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_temp_fasta_filename">temp_fasta_filename</code></td>
<td>
<p>name for a temporary FASTA file,
which will be deleted automatically</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_temp_local_file">temp_local_file</code></td>
<td>
<p>path to the temporary file to store the URL to.
This file will be deleted afterwards.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_temp_xls_filename">temp_xls_filename</code></td>
<td>
<p>name for a temporary <code>xls</code> file,
which will be deleted automatically</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_xls_filename">xls_filename</code></td>
<td>
<p>name of an <code>xls</code> file</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This is an internal function, so it should be marked with
<code>@noRd</code>. This is not done, as this will disallow all
functions to find the documentation parameters
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='download_netmhc2pan_bin'>Download the NetMHCIIpan binary</h2><span id='topic+download_netmhc2pan_bin'></span>

<h3>Description</h3>

<p>Download the NetMHCIIpan binary tarball file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download_netmhc2pan_bin(
  netmhc2pan_bin_url = get_netmhc2pan_bin_url(),
  netmhc2pan_bin_tarfile_path = get_default_netmhc2pan_bin_tarfile_path(),
  verbose = FALSE,
  netmhc2pan_archive_filename = get_netmhc2pan_archive_filename(),
  temp_local_file = tempfile(pattern = "netmhc2pan_download_netmhc2pan_")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download_netmhc2pan_bin_+3A_netmhc2pan_bin_url">netmhc2pan_bin_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan binary tarball file from,
similar to, for example,
<code>https://www.cbs.dtu.dk/download/12345678-1234-1234-1234-123456789ABC</code>.
Use <a href="#topic+get_netmhc2pan_bin_url">get_netmhc2pan_bin_url</a> to get the default URL.</p>
</td></tr>
<tr><td><code id="download_netmhc2pan_bin_+3A_netmhc2pan_bin_tarfile_path">netmhc2pan_bin_tarfile_path</code></td>
<td>
<p>path of the
NetMHCIIpan binary tarball file</p>
</td></tr>
<tr><td><code id="download_netmhc2pan_bin_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
<tr><td><code id="download_netmhc2pan_bin_+3A_netmhc2pan_archive_filename">netmhc2pan_archive_filename</code></td>
<td>
<p>the NetMHC2pan archive
filename, for example <code>netMHCIIpan-3.2.Linux.tar.gz</code>,
which is used in installation of NetMHC2pan.
Use <a href="#topic+get_netmhc2pan_archive_filename">get_netmhc2pan_archive_filename</a> to get this filename.</p>
</td></tr>
<tr><td><code id="download_netmhc2pan_bin_+3A_temp_local_file">temp_local_file</code></td>
<td>
<p>path to the temporary file to store the URL to.
This file will be deleted afterwards.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='get_default_netmhc2pan_bin_path'>Get the full path to the default NetMHC2pan binary</h2><span id='topic+get_default_netmhc2pan_bin_path'></span>

<h3>Description</h3>

<p>Get the full path to the default NetMHC2pan binary
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_default_netmhc2pan_bin_path(
  netmhc2pan_subfolder = get_default_netmhc2pan_subfolder()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_default_netmhc2pan_bin_path_+3A_netmhc2pan_subfolder">netmhc2pan_subfolder</code></td>
<td>
<p>the subfolder (to be) used by NetMHCIIpan.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the full path to the default NetMHC2pan binary
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_default_netmhc2pan_bin_path()
</code></pre>

<hr>
<h2 id='get_default_netmhc2pan_bin_tarfile_path'>Get the default path for the NetMHCIIpan binary tarball file</h2><span id='topic+get_default_netmhc2pan_bin_tarfile_path'></span>

<h3>Description</h3>

<p>Get the default path for the NetMHCIIpan binary tarball file.
This is the location where it will be saved to after downloading.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_default_netmhc2pan_bin_tarfile_path(
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  netmhc2pan_archive_filename = get_netmhc2pan_archive_filename()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_default_netmhc2pan_bin_tarfile_path_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="get_default_netmhc2pan_bin_tarfile_path_+3A_netmhc2pan_archive_filename">netmhc2pan_archive_filename</code></td>
<td>
<p>the NetMHC2pan archive
filename, for example <code>netMHCIIpan-3.2.Linux.tar.gz</code>,
which is used in installation of NetMHC2pan.
Use <a href="#topic+get_netmhc2pan_archive_filename">get_netmhc2pan_archive_filename</a> to get this filename.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the default path for the NetMHCIIpan binary tarball file
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_default_netmhc2pan_bin_tarfile_path()
</code></pre>

<hr>
<h2 id='get_default_netmhc2pan_folder'>Get the path to the folder where this package installs
NetMHCIIpan by default</h2><span id='topic+get_default_netmhc2pan_folder'></span>

<h3>Description</h3>

<p>Get the path to the folder where this package installs
NetMHCIIpan by default
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_default_netmhc2pan_folder()
</code></pre>


<h3>Value</h3>

<p>the path to the folder where this package installs
NetMHCIIpan by default
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_default_netmhc2pan_folder()
</code></pre>

<hr>
<h2 id='get_default_netmhc2pan_subfolder'>Get the full path to the NetMHC2pan sub-folder</h2><span id='topic+get_default_netmhc2pan_subfolder'></span>

<h3>Description</h3>

<p>Get the full path to the NetMHC2pan sub-folder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_default_netmhc2pan_subfolder(
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  netmhc2pan_version = get_default_netmhc2pan_version()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_default_netmhc2pan_subfolder_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="get_default_netmhc2pan_subfolder_+3A_netmhc2pan_version">netmhc2pan_version</code></td>
<td>
<p>the NetMHCIIpan version,
for example <code>3.2</code>.
Use <a href="#topic+get_default_netmhc2pan_version">get_default_netmhc2pan_version</a> to get the default
NetMHCIIpan version</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the full path to the NetMHC2pan sub-folder
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_default_netmhc2pan_subfolder()
</code></pre>

<hr>
<h2 id='get_default_netmhc2pan_version'>Get the default NetMHC2pan version used</h2><span id='topic+get_default_netmhc2pan_version'></span>

<h3>Description</h3>

<p>Get the default NetMHC2pan version used
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_default_netmhc2pan_version()
</code></pre>


<h3>Value</h3>

<p>the default NetMHC2pan version used
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_default_netmhc2pan_version()
</code></pre>

<hr>
<h2 id='get_netmhc2pan_alleles'>Get a list of the alleles supported by NetMHCIIpan</h2><span id='topic+get_netmhc2pan_alleles'></span>

<h3>Description</h3>

<p>Get a list of the alleles supported by NetMHCIIpan
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_netmhc2pan_alleles(
  netmhc2pan_folder_name = get_default_netmhc2pan_folder()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_netmhc2pan_alleles_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector with the NetMHCIIpan alleles
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_netmhc2pan_installed()) {
  get_netmhc2pan_alleles()[1:5]
}
</code></pre>

<hr>
<h2 id='get_netmhc2pan_archive_filename'>Get the filename of the NetMHC2pan archive file</h2><span id='topic+get_netmhc2pan_archive_filename'></span>

<h3>Description</h3>

<p>Get the filename of the NetMHC2pan archive file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_netmhc2pan_archive_filename(
  netmhc2pan_version = get_default_netmhc2pan_version()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_netmhc2pan_archive_filename_+3A_netmhc2pan_version">netmhc2pan_version</code></td>
<td>
<p>the NetMHCIIpan version,
for example <code>3.2</code>.
Use <a href="#topic+get_default_netmhc2pan_version">get_default_netmhc2pan_version</a> to get the default
NetMHCIIpan version</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the filename of the NetMHC2pan archive file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_netmhc2pan_archive_filename()
</code></pre>

<hr>
<h2 id='get_netmhc2pan_bin_url'>Get the NetMHCIIpan binary download URL.</h2><span id='topic+get_netmhc2pan_bin_url'></span>

<h3>Description</h3>

<p>Get the URL for the NetMHCIIpan binary, as is emailed.
These expire after 4 hours.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_netmhc2pan_bin_url()
</code></pre>


<h3>Value</h3>

<p>a download URL
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+is_url_valid">is_url_valid</a> to determine if the download URL is still valid.
Use <a href="#topic+check_netmhc2pan_bin_url">check_netmhc2pan_bin_url</a> to get a helpful error message if this
URL is invalid.
</p>

<hr>
<h2 id='get_netmhc2pan_data_url'>Get the NetMHCIIpan data tarball URL.</h2><span id='topic+get_netmhc2pan_data_url'></span>

<h3>Description</h3>

<p>Get the NetMHCIIpan data tarball URL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_netmhc2pan_data_url()
</code></pre>


<h3>Value</h3>

<p>a download URL
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+is_url_valid">is_url_valid</a> to determine if the download URL is still valid.
Use <a href="#topic+check_netmhc2pan_data_url">check_netmhc2pan_data_url</a> to get a helpful error message if this
URL is invalid.
</p>

<hr>
<h2 id='get_netmhc2pan_url'>Deprecated, use <a href="#topic+get_netmhc2pan_bin_url">get_netmhc2pan_bin_url</a> instead</h2><span id='topic+get_netmhc2pan_url'></span>

<h3>Description</h3>

<p>Deprecated, use <a href="#topic+get_netmhc2pan_bin_url">get_netmhc2pan_bin_url</a> instead
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_netmhc2pan_url()
</code></pre>


<h3>Value</h3>

<p>a download URL
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='install_netmhc2pan'>Install NetMHCIIpan</h2><span id='topic+install_netmhc2pan'></span>

<h3>Description</h3>

<p>Install NetMHCIIpan to a local folder, by downloading the binary and
data files. Use <a href="#topic+install_netmhc2pan_from_files">install_netmhc2pan_from_files</a> to install
NetMHCIIpan from files that are already downloaded.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_netmhc2pan(
  netmhc2pan_bin_url = get_netmhc2pan_bin_url(),
  netmhc2pan_data_url = get_netmhc2pan_data_url(),
  verbose = FALSE,
  netmhc2pan_archive_filename = get_netmhc2pan_archive_filename(),
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  temp_local_file = tempfile(pattern = "netmhc2pan_install_netmhc2pan_")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="install_netmhc2pan_+3A_netmhc2pan_bin_url">netmhc2pan_bin_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan binary tarball file from,
similar to, for example,
<code>https://www.cbs.dtu.dk/download/12345678-1234-1234-1234-123456789ABC</code>.
Use <a href="#topic+get_netmhc2pan_bin_url">get_netmhc2pan_bin_url</a> to get the default URL.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_+3A_netmhc2pan_data_url">netmhc2pan_data_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan data tarball file from,
similar to, for example,
<a href="https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz">https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz</a>.
Use <a href="#topic+get_netmhc2pan_data_url">get_netmhc2pan_data_url</a> to get the default URL.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_+3A_netmhc2pan_archive_filename">netmhc2pan_archive_filename</code></td>
<td>
<p>the NetMHC2pan archive
filename, for example <code>netMHCIIpan-3.2.Linux.tar.gz</code>,
which is used in installation of NetMHC2pan.
Use <a href="#topic+get_netmhc2pan_archive_filename">get_netmhc2pan_archive_filename</a> to get this filename.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_+3A_temp_local_file">temp_local_file</code></td>
<td>
<p>path to the temporary file to store the URL to.
This file will be deleted afterwards.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These are three steps:
</p>

<ul>
<li><p> Install the NetMHCIIpan binary, using <a href="#topic+install_netmhc2pan_bin">install_netmhc2pan_bin</a>
</p>
</li>
<li><p> Install the NetMHCIIpan data, using <a href="#topic+install_netmhc2pan_data">install_netmhc2pan_data</a>
</p>
</li>
<li><p> Set up NetMHCIIpan, using <a href="#topic+set_up_netmhc2pan">set_up_netmhc2pan</a>
</p>
</li></ul>



<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='install_netmhc2pan_bin'>Install the NetMHCIIpan binary to a local folder</h2><span id='topic+install_netmhc2pan_bin'></span>

<h3>Description</h3>

<p>Install the NetMHCIIpan binary to a local folder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_netmhc2pan_bin(
  netmhc2pan_bin_url = get_netmhc2pan_bin_url(),
  verbose = FALSE,
  netmhc2pan_archive_filename = get_netmhc2pan_archive_filename(),
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  temp_local_file = tempfile(pattern = "netmhc2pan_install_netmhc2pan_bin_")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="install_netmhc2pan_bin_+3A_netmhc2pan_bin_url">netmhc2pan_bin_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan binary tarball file from,
similar to, for example,
<code>https://www.cbs.dtu.dk/download/12345678-1234-1234-1234-123456789ABC</code>.
Use <a href="#topic+get_netmhc2pan_bin_url">get_netmhc2pan_bin_url</a> to get the default URL.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_bin_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_bin_+3A_netmhc2pan_archive_filename">netmhc2pan_archive_filename</code></td>
<td>
<p>the NetMHC2pan archive
filename, for example <code>netMHCIIpan-3.2.Linux.tar.gz</code>,
which is used in installation of NetMHC2pan.
Use <a href="#topic+get_netmhc2pan_archive_filename">get_netmhc2pan_archive_filename</a> to get this filename.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_bin_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_bin_+3A_temp_local_file">temp_local_file</code></td>
<td>
<p>path to the temporary file to store the URL to.
This file will be deleted afterwards.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='install_netmhc2pan_bin_from_file'>Install the NetMHCIIpan binary to a local folder</h2><span id='topic+install_netmhc2pan_bin_from_file'></span>

<h3>Description</h3>

<p>Install the NetMHCIIpan binary to a local folder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_netmhc2pan_bin_from_file(
  netmhc2pan_bin_tarfile_path,
  verbose = FALSE,
  netmhc2pan_archive_filename = get_netmhc2pan_archive_filename(),
  netmhc2pan_folder_name = get_default_netmhc2pan_folder()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="install_netmhc2pan_bin_from_file_+3A_netmhc2pan_bin_tarfile_path">netmhc2pan_bin_tarfile_path</code></td>
<td>
<p>path of the
NetMHCIIpan binary tarball file</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_bin_from_file_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_bin_from_file_+3A_netmhc2pan_archive_filename">netmhc2pan_archive_filename</code></td>
<td>
<p>the NetMHC2pan archive
filename, for example <code>netMHCIIpan-3.2.Linux.tar.gz</code>,
which is used in installation of NetMHC2pan.
Use <a href="#topic+get_netmhc2pan_archive_filename">get_netmhc2pan_archive_filename</a> to get this filename.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_bin_from_file_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='install_netmhc2pan_data'>Install the NetMHCIIpan data to a local folder</h2><span id='topic+install_netmhc2pan_data'></span>

<h3>Description</h3>

<p>Install the NetMHCIIpan data to a local folder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_netmhc2pan_data(
  netmhc2pan_data_url = get_netmhc2pan_data_url(),
  netmhc2pan_folder_name = rappdirs::user_data_dir(),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="install_netmhc2pan_data_+3A_netmhc2pan_data_url">netmhc2pan_data_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan data tarball file from,
similar to, for example,
<a href="https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz">https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz</a>.
Use <a href="#topic+get_netmhc2pan_data_url">get_netmhc2pan_data_url</a> to get the default URL.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_data_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_data_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This data can be downloaded from
<a href="https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz">https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz</a>,
without filling in a contact form.
</p>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='install_netmhc2pan_data_from_file'>Install the NetMHCIIpan data from file</h2><span id='topic+install_netmhc2pan_data_from_file'></span>

<h3>Description</h3>

<p>Install the NetMHCIIpan data from the NetMHCIIpan data tarball file
to the default NetMHCIIpan folder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_netmhc2pan_data_from_file(
  netmhc2pan_data_tarfile_path,
  netmhc2pan_folder_name = rappdirs::user_data_dir(),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="install_netmhc2pan_data_from_file_+3A_netmhc2pan_data_tarfile_path">netmhc2pan_data_tarfile_path</code></td>
<td>
<p>path of the
NetMHCIIpan data tarball file</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_data_from_file_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_data_from_file_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The NetMHCIIpan data tarball file can be in any path.
</p>
<p>The data tarball file can be downloaded from
<a href="https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz">https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/data.Linux.tar.gz</a>,
without filling in a contact form.
</p>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='install_netmhc2pan_from_files'>Install NetMHCIIpan from files.</h2><span id='topic+install_netmhc2pan_from_files'></span>

<h3>Description</h3>

<p>Install NetMHCIIpan to a local folder from (already downloaded)
binary and date tarball files.
Use <a href="#topic+install_netmhc2pan">install_netmhc2pan</a> to install
NetMHCIIpan by downloading these files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_netmhc2pan_from_files(
  netmhc2pan_bin_tarfile_path,
  netmhc2pan_data_tarfile_path,
  verbose = FALSE,
  netmhc2pan_archive_filename = get_netmhc2pan_archive_filename(),
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  temp_local_file = tempfile(pattern = "netmhc2pan_install_netmhc2pan_")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="install_netmhc2pan_from_files_+3A_netmhc2pan_bin_tarfile_path">netmhc2pan_bin_tarfile_path</code></td>
<td>
<p>path of the
NetMHCIIpan binary tarball file</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_from_files_+3A_netmhc2pan_data_tarfile_path">netmhc2pan_data_tarfile_path</code></td>
<td>
<p>path of the
NetMHCIIpan data tarball file</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_from_files_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_from_files_+3A_netmhc2pan_archive_filename">netmhc2pan_archive_filename</code></td>
<td>
<p>the NetMHC2pan archive
filename, for example <code>netMHCIIpan-3.2.Linux.tar.gz</code>,
which is used in installation of NetMHC2pan.
Use <a href="#topic+get_netmhc2pan_archive_filename">get_netmhc2pan_archive_filename</a> to get this filename.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_from_files_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="install_netmhc2pan_from_files_+3A_temp_local_file">temp_local_file</code></td>
<td>
<p>path to the temporary file to store the URL to.
This file will be deleted afterwards.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These are three steps:
</p>

<ul>
<li><p> Install the NetMHCIIpan binary,
using <a href="#topic+install_netmhc2pan_bin_from_file">install_netmhc2pan_bin_from_file</a>
</p>
</li>
<li><p> Install the NetMHCIIpan data,
using <a href="#topic+install_netmhc2pan_data_from_file">install_netmhc2pan_data_from_file</a>
</p>
</li>
<li><p> Set up NetMHCIIpan, using <a href="#topic+set_up_netmhc2pan">set_up_netmhc2pan</a>
</p>
</li></ul>



<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_netmhc2pan_bin_installed'>Measure if NetMHCIIpan binary is installed locally</h2><span id='topic+is_netmhc2pan_bin_installed'></span>

<h3>Description</h3>

<p>Measure if NetMHCIIpan binary is installed locally
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_netmhc2pan_bin_installed(
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_netmhc2pan_bin_installed_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="is_netmhc2pan_bin_installed_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE is NetMHCIIpan binary is installed locally,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_netmhc2pan_data_installed'>Measure if NetMHCIIpan data folder is installed locally</h2><span id='topic+is_netmhc2pan_data_installed'></span>

<h3>Description</h3>

<p>Measure if NetMHCIIpan data folder is installed locally
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_netmhc2pan_data_installed(
  netmhc2pan_folder_name = get_default_netmhc2pan_folder()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_netmhc2pan_data_installed_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE is NetMHCIIpan data folder is installed locally,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_netmhc2pan_haplotype'>Is the haplotype valid?</h2><span id='topic+is_netmhc2pan_haplotype'></span>

<h3>Description</h3>

<p>Determines if a haplotype has the same notation
as a NetMHC2pan haplotype
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_netmhc2pan_haplotype(mhc_haplotype)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_netmhc2pan_haplotype_+3A_mhc_haplotype">mhc_haplotype</code></td>
<td>
<p>one MHC haplotype, e.g. <code>DRB1_0101</code>.
See <a href="#topic+get_netmhc2pan_alleles">get_netmhc2pan_alleles</a> for a full list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the haplotype is a valid NetMHCIIpan haplotype
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_netmhc2pan_installed()) {
  is_netmhc2pan_haplotype("DRB1_0311")
  is_netmhc2pan_haplotype("nonsense")
}
</code></pre>

<hr>
<h2 id='is_netmhc2pan_installed'>Measure if NetMHCIIpan is installed locally</h2><span id='topic+is_netmhc2pan_installed'></span>

<h3>Description</h3>

<p>Measure if NetMHCIIpan is installed locally
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_netmhc2pan_installed(
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_netmhc2pan_installed_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="is_netmhc2pan_installed_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE is NetMHCIIpan is installed locally,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_netmhc2pan_set_up'>Measure if NetMHCIIpan is set up</h2><span id='topic+is_netmhc2pan_set_up'></span>

<h3>Description</h3>

<p>Measure if NetMHCIIpan is set up
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_netmhc2pan_set_up(netmhc2pan_folder_name = get_default_netmhc2pan_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_netmhc2pan_set_up_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE is NetMHCIIpan is set up locally,
FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_on_appveyor'>Determines if the environment is AppVeyor</h2><span id='topic+is_on_appveyor'></span>

<h3>Description</h3>

<p>Determines if the environment is AppVeyor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_appveyor()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on AppVeyor, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  if (is_on_appveyor()) {
    message("Running on AppVeyor")
  }
</code></pre>

<hr>
<h2 id='is_on_ci'>Determines if the environment is a continuous integration service</h2><span id='topic+is_on_ci'></span>

<h3>Description</h3>

<p>Determines if the environment is a continuous integration service
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_ci()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on AppVeyor or Travis CI, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Note</h3>

<p>It is possible to fake being on continuous integration service,
in this case GitHub Actions, using:
</p>
<p>&ldquo;'r
Sys.setenv(GITHUB_ACTIONS = &quot;I fake being on GitHub Actions&quot;)
is_on_ci() # Will be true
&ldquo;'
</p>
<p>To undo this, do
</p>
<p>&ldquo;'r
Sys.setenv(GITHUB_ACTIONS = &quot;&quot;)
is_on_ci() # Will be false
&ldquo;'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_on_ci()
</code></pre>

<hr>
<h2 id='is_on_github_actions'>Determines if the environment is GitHub Actions</h2><span id='topic+is_on_github_actions'></span>

<h3>Description</h3>

<p>Determines if the environment is GitHub Actions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_github_actions()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on GitHub Actions, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Note</h3>

<p>It is possible to fake being on GitHub Actions, using:
</p>
<p>&ldquo;'r
Sys.setenv(GITHUB_ACTIONS = &quot;I fake being on GitHub Actions&quot;)
is_on_github_actions() # Will be true
&ldquo;'
</p>
<p>To undo this, do
</p>
<p>&ldquo;'r
Sys.setenv(GITHUB_ACTIONS = &quot;&quot;)
is_on_github_actions() # Will be false
&ldquo;'
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_on_github_actions()) {
  message("Running on GitHub Actions")
}
</code></pre>

<hr>
<h2 id='is_on_travis'>Determines if the environment is Travis CI</h2><span id='topic+is_on_travis'></span>

<h3>Description</h3>

<p>Determines if the environment is Travis CI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_travis()
</code></pre>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if run on Travis CI, <a href="base.html#topic+FALSE">FALSE</a> otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  if (is_on_ci()) {
    message("Running on Travis CI")
  }
</code></pre>

<hr>
<h2 id='is_tcsh_installed'>See if tcsh is installed</h2><span id='topic+is_tcsh_installed'></span>

<h3>Description</h3>

<p>See if tcsh is installed
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_tcsh_installed()
</code></pre>


<h3>Value</h3>

<p>TRUE is tcsh is installed,
FALSE otherwise
</p>


<h3>Note</h3>

<p>To install <code>tcsh</code> under Linux, do <code>sudo apt install tcsh</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_url_valid'>Is the download URL valid?</h2><span id='topic+is_url_valid'></span>

<h3>Description</h3>

<p>The download link expires after 4 hours.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_url_valid(
  netmhc2pan_bin_url = get_netmhc2pan_bin_url(),
  verbose = FALSE,
  netmhc2pan_archive_filename = get_netmhc2pan_archive_filename(),
  temp_local_file = tempfile(pattern = "netmhc2pan_is_url_valid_")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_url_valid_+3A_netmhc2pan_bin_url">netmhc2pan_bin_url</code></td>
<td>
<p>URL to download the
NetMHCIIpan binary tarball file from,
similar to, for example,
<code>https://www.cbs.dtu.dk/download/12345678-1234-1234-1234-123456789ABC</code>.
Use <a href="#topic+get_netmhc2pan_bin_url">get_netmhc2pan_bin_url</a> to get the default URL.</p>
</td></tr>
<tr><td><code id="is_url_valid_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
<tr><td><code id="is_url_valid_+3A_netmhc2pan_archive_filename">netmhc2pan_archive_filename</code></td>
<td>
<p>the NetMHC2pan archive
filename, for example <code>netMHCIIpan-3.2.Linux.tar.gz</code>,
which is used in installation of NetMHC2pan.
Use <a href="#topic+get_netmhc2pan_archive_filename">get_netmhc2pan_archive_filename</a> to get this filename.</p>
</td></tr>
<tr><td><code id="is_url_valid_+3A_temp_local_file">temp_local_file</code></td>
<td>
<p>path to the temporary file to store the URL to.
This file will be deleted afterwards.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the download URL valid, FALSE otherwise
</p>

<hr>
<h2 id='netmhc2pan'>netmhc2pan: predict peptide binding strengths using NetMHC2pan</h2><span id='topic+netmhc2pan'></span>

<h3>Description</h3>

<p>'NetMHCIIpan' is a tool to predict the
binding strength of a short peptide to an MHC-II complex.
'NetMHCIIpan' can be used from a web server at
'https://services.healthtech.dtu.dk/services/NetMHCIIpan-3.2/'
or from the command-line, using a local installation. This package
allows to call 'NetMHCIIpan' from R.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_netmhc2pan_installed()) {

  predict_ic50(
    peptides = c("AIAACAMLLV", "ALVCYIVMPV"),
    mhc_haplotype = "DRB1_0416"
  )
}
</code></pre>

<hr>
<h2 id='netmhc2pan_report'>Show a <code>netmhc2pan</code> report using <a href="base.html#topic+message">message</a>,
to be used when reporting bugs</h2><span id='topic+netmhc2pan_report'></span>

<h3>Description</h3>

<p>Show a <code>netmhc2pan</code> report using <a href="base.html#topic+message">message</a>,
to be used when reporting bugs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>netmhc2pan_report(netmhc2pan_folder_name = get_default_netmhc2pan_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="netmhc2pan_report_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing, it is called for its side effects
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>netmhc2pan_report()
</code></pre>

<hr>
<h2 id='netmhc2pan_self_test'>Test NetMHCIIpan by doing a minimal run.</h2><span id='topic+netmhc2pan_self_test'></span>

<h3>Description</h3>

<p>Test NetMHCIIpan by doing a minimal run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>netmhc2pan_self_test(
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="netmhc2pan_self_test_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="netmhc2pan_self_test_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. If the self-test fails, an error will be raised
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='predict_ic50'>Predict the IC50 for peptides.</h2><span id='topic+predict_ic50'></span>

<h3>Description</h3>

<p>Predict the half maximal inhibitory concentration (aka IC50) (in nM)
for one or more peptides, where the peptides are used as-is,
instead of split into smaller peptides.
Each peptide must be 15 amino acids at
most (use <a href="#topic+predict_ic50s">predict_ic50s</a> to predict the IC50s for longer peptides)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_ic50(
  peptides,
  mhc_haplotype,
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  temp_fasta_filename = netmhc2pan::create_temp_fasta_filename(),
  temp_xls_filename = netmhc2pan::create_temp_xls_filename()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_ic50_+3A_peptides">peptides</code></td>
<td>
<p>one or more peptide sequences</p>
</td></tr>
<tr><td><code id="predict_ic50_+3A_mhc_haplotype">mhc_haplotype</code></td>
<td>
<p>one MHC haplotype, e.g. <code>DRB1_0101</code>.
See <a href="#topic+get_netmhc2pan_alleles">get_netmhc2pan_alleles</a> for a full list</p>
</td></tr>
<tr><td><code id="predict_ic50_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="predict_ic50_+3A_temp_fasta_filename">temp_fasta_filename</code></td>
<td>
<p>name for a temporary FASTA file,
which will be deleted automatically</p>
</td></tr>
<tr><td><code id="predict_ic50_+3A_temp_xls_filename">temp_xls_filename</code></td>
<td>
<p>name for a temporary <code>xls</code> file,
which will be deleted automatically</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a> with two columns:
(1) <code>peptide</code>, which holds the peptide sequence, and
(2) <code>ic50</code>, which holds the predicted IC50
</p>


<h3>Note</h3>

<p>this function uses a temporary file, because NetMHC2pan reads its
input from file. This temporary file is deleted after this function
passed successfully.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_netmhc2pan_installed()) {

  predict_ic50(
    peptides = c("AIAACAMLLV", "ALVCYIVMPV"),
    mhc_haplotype = "DRB1_0416"
  )
}
</code></pre>

<hr>
<h2 id='predict_ic50s'>Predict the IC50s from a sequence</h2><span id='topic+predict_ic50s'></span>

<h3>Description</h3>

<p>Predict the IC50s from a sequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_ic50s(
  protein_sequence,
  peptide_length,
  mhc_haplotype,
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  temp_fasta_filename = netmhc2pan::create_temp_fasta_filename(),
  temp_xls_filename = netmhc2pan::create_temp_xls_filename()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_ic50s_+3A_protein_sequence">protein_sequence</code></td>
<td>
<p>a protein sequence, for example <code>FAMILYVW</code></p>
</td></tr>
<tr><td><code id="predict_ic50s_+3A_peptide_length">peptide_length</code></td>
<td>
<p>length of a peptide</p>
</td></tr>
<tr><td><code id="predict_ic50s_+3A_mhc_haplotype">mhc_haplotype</code></td>
<td>
<p>one MHC haplotype, e.g. <code>DRB1_0101</code>.
See <a href="#topic+get_netmhc2pan_alleles">get_netmhc2pan_alleles</a> for a full list</p>
</td></tr>
<tr><td><code id="predict_ic50s_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="predict_ic50s_+3A_temp_fasta_filename">temp_fasta_filename</code></td>
<td>
<p>name for a temporary FASTA file,
which will be deleted automatically</p>
</td></tr>
<tr><td><code id="predict_ic50s_+3A_temp_xls_filename">temp_xls_filename</code></td>
<td>
<p>name for a temporary <code>xls</code> file,
which will be deleted automatically</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="tibble.html#topic+tibble">tibble</a> with columns:<br />
</p>

<ul>
<li><p> peptide the peptide fragment, each of length <code>peptide_length</code>
</p>
</li>
<li><p> ic50 the predicted IC50 (in nM)
</p>
</li></ul>

<p>The number of rows equals <code>protein_sequence - peptide_length + 1</code>.
</p>

<hr>
<h2 id='read_netmhc2pan_xls_output'>Read MHC2pan output</h2><span id='topic+read_netmhc2pan_xls_output'></span>

<h3>Description</h3>

<p>Read the output produced by MHC2pan,
which is an <code>XLS</code> file by default.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_netmhc2pan_xls_output(xls_filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_netmhc2pan_xls_output_+3A_xls_filename">xls_filename</code></td>
<td>
<p>name of an <code>xls</code> file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a table
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xls_filename &lt;- system.file("extdata", "example.xls", package = "netmhc2pan")
read_netmhc2pan_xls_output(xls_filename)
</code></pre>

<hr>
<h2 id='run_netmhc2pan'>Run NetMHCIIpan</h2><span id='topic+run_netmhc2pan'></span>

<h3>Description</h3>

<p>Run NetMHCIIpan
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_netmhc2pan(
  fasta_filename,
  alleles = "DRB1_0101",
  peptide_length = 15,
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  temp_xls_filename = netmhc2pan::create_temp_xls_filename()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_netmhc2pan_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>the name of a FASTA file with protein sequences</p>
</td></tr>
<tr><td><code id="run_netmhc2pan_+3A_alleles">alleles</code></td>
<td>
<p>one or more alleles, e.g. <code>DRB1_0101</code>.
See <a href="#topic+get_netmhc2pan_alleles">get_netmhc2pan_alleles</a> for a full list</p>
</td></tr>
<tr><td><code id="run_netmhc2pan_+3A_peptide_length">peptide_length</code></td>
<td>
<p>length of a peptide</p>
</td></tr>
<tr><td><code id="run_netmhc2pan_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="run_netmhc2pan_+3A_temp_xls_filename">temp_xls_filename</code></td>
<td>
<p>name for a temporary <code>xls</code> file,
which will be deleted automatically</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with the NetMHCIIpan results
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_netmhc2pan_installed()) {

  fasta_filename &lt;- system.file(
    "extdata", "example.fasta", package = "netmhc2pan"
  )
  run_netmhc2pan(fasta_filename)

  # Two alleles
  alleles &lt;- c("DRB1_0101", "DRB1_0102")
  # Run NetMHCpan with these two alleles
  run_netmhc2pan(fasta_filename, alleles = alleles)
}
</code></pre>

<hr>
<h2 id='set_up_netmhc2pan'>Install the NetMHCIIpan binary to a local folder</h2><span id='topic+set_up_netmhc2pan'></span>

<h3>Description</h3>

<p>Install the NetMHCIIpan binary to a local folder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_up_netmhc2pan(
  netmhc2pan_folder_name = get_default_netmhc2pan_folder(),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_up_netmhc2pan_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
<tr><td><code id="set_up_netmhc2pan_+3A_verbose">verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='to_netmhc2pan_name'>Convert a formal haplotype name to its NetMHC2pan notation</h2><span id='topic+to_netmhc2pan_name'></span>

<h3>Description</h3>

<p>Convert a formal MHC-II haplotype name to the notation used by NetMHC2pan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>to_netmhc2pan_name(mhc_haplotype)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="to_netmhc2pan_name_+3A_mhc_haplotype">mhc_haplotype</code></td>
<td>
<p>one MHC haplotype, e.g. <code>DRB1_0101</code>.
See <a href="#topic+get_netmhc2pan_alleles">get_netmhc2pan_alleles</a> for a full list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the haplotype name in NetMHC2pan notation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>to_netmhc2pan_name("HLA-DRB1*0101")
to_netmhc2pan_name("HLA-DQA1*0501/DQB1*0201")
</code></pre>

<hr>
<h2 id='uninstall_netmhc2pan'>Uninstall NetMHCIIpan</h2><span id='topic+uninstall_netmhc2pan'></span>

<h3>Description</h3>

<p>Uninstall NetMHCIIpan
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uninstall_netmhc2pan(netmhc2pan_folder_name = get_default_netmhc2pan_folder())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="uninstall_netmhc2pan_+3A_netmhc2pan_folder_name">netmhc2pan_folder_name</code></td>
<td>
<p>the folder (to be) used by NetMHCIIpan.
From this location, a subfolder for NetMHCIIpan is created.
Use <a href="#topic+get_default_netmhc2pan_folder">get_default_netmhc2pan_folder</a> to see the location
of the default NetMHCIIpan folder.
Use <a href="#topic+get_default_netmhc2pan_subfolder">get_default_netmhc2pan_subfolder</a> to see the location
of the default NetMHCIIpan subfolder.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
