<!DOCTYPE html><html><head><title>Help for package NetWeaver</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {NetWeaver}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ensemble_rank'><p>Ensemble Rank</p></a></li>
<li><a href='#getrankp'><p>getrankp</p></a></li>
<li><a href='#is.overlap'><p>Region overlap test</p></a></li>
<li><a href='#Modules'><p>Example Module Dataset</p></a></li>
<li><a href='#NetWeaver'><p>Graphic Presentation of Complex Genomic and Network Data Analysis</p></a></li>
<li><a href='#rc.get.coordinates'><p>Get Coordinates</p></a></li>
<li><a href='#rc.get.params'><p>Get Parameters</p></a></li>
<li><a href='#rc.initialize'><p>Initialize Circos Plot Parameters</p></a></li>
<li><a href='#rc.plot.area'><p>Setup Plot Area</p></a></li>
<li><a href='#rc.plot.barchart'><p>Bar Plots</p></a></li>
<li><a href='#rc.plot.grColLegend'><p>Plot Gradient Color Legend</p></a></li>
<li><a href='#rc.plot.heatmap'><p>Plot Heat-map</p></a></li>
<li><a href='#rc.plot.histogram'><p>Plot Histogram, Barchart or Heat-map</p></a></li>
<li><a href='#rc.plot.ideogram'><p>Plot Ideogram</p></a></li>
<li><a href='#rc.plot.line'><p>Plot Line</p></a></li>
<li><a href='#rc.plot.link'><p>Plot Link</p></a></li>
<li><a href='#rc.plot.mHistogram'><p>Plot Histogram or Barplot Across Chromosomes</p></a></li>
<li><a href='#rc.plot.point'><p>Plot Point</p></a></li>
<li><a href='#rc.plot.ribbon'><p>Plot Ribbon</p></a></li>
<li><a href='#rc.plot.text'><p>Plot Text</p></a></li>
<li><a href='#rc.plot.track'><p>Plot Track</p></a></li>
<li><a href='#rc.plot.track.id'><p>Plot Track Id</p></a></li>
<li><a href='#rc.reset.params'><p>Reset Circos Plot Parameters</p></a></li>
<li><a href='#rc.track.pos'><p>Get Track Positions</p></a></li>
<li><a href='#ucsc.hg19.cytoband'><p>Human chromosome cytoband</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Graphic Presentation of Complex Genomic and Network Data
Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-02-26</td>
</tr>
<tr>
<td>Author:</td>
<td>Minghui Wang, Bin Zhang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Minghui Wang &lt;m.h.wang@live.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements various simple function utilities and flexible pipelines to generate circular images for visualizing complex genomic and network data analysis features.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mw201608/NetWeaver/">https://github.com/mw201608/NetWeaver/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mw201608/NetWeaver/issues">https://github.com/mw201608/NetWeaver/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-02-26 18:18:35 UTC; Minghui</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-02-26 18:40:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='ensemble_rank'>Ensemble Rank</h2><span id='topic+ensemble_rank'></span>

<h3>Description</h3>

<p>Compute composite rank score using evidence from multiple features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  ensemble_rank(x, method=c('ProductOfRank','MeanOfLog','MeanOfLogLog','Mean'),
  small=1.0e-320, standardize=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ensemble_rank_+3A_x">x</code></td>
<td>
<p>a matrix of discriminant values (e.g., P values) measuring the strength of the association between objects (eg modules) (in rows) and variables/features (in columns).</p>
</td></tr>
<tr><td><code id="ensemble_rank_+3A_method">method</code></td>
<td>
<p>a character string specifyign the ranking metric. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="ensemble_rank_+3A_small">small</code></td>
<td>
<p>offset the small p values before taking log transformation.</p>
</td></tr>
<tr><td><code id="ensemble_rank_+3A_standardize">standardize</code></td>
<td>
<p>whether to rescale the final ranking score by dividing the maximum value.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Three ranking metrics are currently implemented:
</p>
<p>ProductOfRank is the one used in Zhang et al. (Cell 2013, 153: 707-720) with a slight change to standardize the scores to be between 0 and 1; 
</p>
<p>MeanOfLog computes the mean of -log(p value) which penalizes insignificant p values;
</p>
<p>MeanOfLogLog computes the mean of log(-log(p value)) which penalizes insignificant p values and shrinks the difference in the ranges of p values.
</p>
<p>Means computes the row mean of absolute values of <code>x</code>.
</p>


<h3>Value</h3>

<p>A vector of ranking scores. The larger the score, the higher the rank.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Rank US cities by the significance test of excessive arrests compared to average
#compute the p value of one-tailed z-test
x=apply(USArrests,2,function(x) pnorm((x-mean(x))/sd(x),lower.tail=FALSE))
#compute ranking score using three different metrics
score1=ensemble_rank(x,method='ProductOfRank')
score2=ensemble_rank(x,method='MeanOfLog')
score3=ensemble_rank(x,method='MeanOfLogLog')
#plot the top 5 worst cities by each metric
par(mfrow=c(1,3))
barplot(sort(score1,decreasing=TRUE)[1:5],las=2)
barplot(sort(score2,decreasing=TRUE)[1:5],las=2)
barplot(sort(score3,decreasing=TRUE)[1:5],las=2)
</code></pre>

<hr>
<h2 id='getrankp'>getrankp</h2><span id='topic+getrankp'></span>

<h3>Description</h3>

<p>Compute probability (fraction) of values in a large population more extreme than input series.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  getrankp(x, y, truncated.size=0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getrankp_+3A_x">x</code></td>
<td>
<p>A vector of input series.</p>
</td></tr>
<tr><td><code id="getrankp_+3A_y">y</code></td>
<td>
<p>A vector of population values.</p>
</td></tr>
<tr><td><code id="getrankp_+3A_truncated.size">truncated.size</code></td>
<td>
<p>See <code>Details</code>. Only use when you know this.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to compute permutation-based false discovery rate (two-tailed). In such case, <code>x</code> contains test statistics from observed data while <code>y</code> contains test statistics from permutation. <code>x</code> could be a truncated set, with small values discarded to save memory.
</p>


<h3>Value</h3>

<p>A vector of the same size as <code>x</code>.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x=abs(rnorm(100))
y=abs(rnorm(10000))
getrankp(x,y)[1:4]
</code></pre>

<hr>
<h2 id='is.overlap'>Region overlap test</h2><span id='topic+is.overlap'></span>

<h3>Description</h3>

<p>Test if two regions are overlapping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  is.overlap(x,y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.overlap_+3A_x">x</code></td>
<td>
<p>a vector of two numeric values specifying the start and end positions of the first region.</p>
</td></tr>
<tr><td><code id="is.overlap_+3A_y">y</code></td>
<td>
<p>a vector of two numeric values specifying the start and end positions of the second region</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logic test output.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is.overlap(c(10,100),c(90,120))
is.overlap(c(10,100),c(110,120))
</code></pre>

<hr>
<h2 id='Modules'>Example Module Dataset</h2><span id='topic+Modules'></span>

<h3>Description</h3>

<p>This example dataset and sample code illustrate the use of <code>NetWeaver</code> to visualize the complex features of gene coexpression network modules.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Modules)</code></pre>


<h3>Details</h3>

<p>This dataset contains a data.frame called <code>Modules</code>. Each row is a module, with the module id in the first column. The second column is ranking score. The next 4 columns are coefficients of module-trait correlations. The rest columns are P values of enrichment for various gene signatures. The sample R code shows a pipeline for visualizing the module feature data using a circos style plot.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;, Bin Zhang &lt;bin.zhang@mssm.edu&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.initialize">rc.initialize</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#see a tutorial for how to plot this dataset
vignette("netweaver")

## End(Not run)
</code></pre>

<hr>
<h2 id='NetWeaver'>Graphic Presentation of Complex Genomic and Network Data Analysis</h2><span id='topic+NetWeaver'></span><span id='topic+NetWeaver-package'></span>

<h3>Description</h3>

<p><code>NetWeaver</code> is motivated towards developing a simple and flexible pipeline for visualizing the complex features of enrichment and correlation of gene coexpression network modules. While circos style 2D track plot is one natural choice for such practice, existing packages are designed primarily for handling genome structure and intervals. They are either too complicated to use, requiring certain level of knowledge of scripting, or limited in applications to only genomic structure data. To address these issues, particularly extend beyond applications in genomic structure data, <code>NetWeaver</code> offers a lightweight implementation of circular track plot, providing simple and flexible R function utilities and pipelines to generate circular images for visualizing different types of structure/relationship data.<br /><br />
The original version of this package was developed for Figure 7 of Wang et al (2016) <em>Genome Medicine</em> 8:104, which illustrates more than 20 properties for 50 coexpression network modules with a circular track plot. Please try to cite the paper when you use results from this software in a publication:<br /><br />
Wang M, Roussos P, McKenzie A, Zhou X, Kajiwara Y, Brennand K, DeLuca GC, Crary JF, Casaccia P, Buxbaum J et al. 2016. Integrative Network Analysis of Nineteen Brain Regions Identifies Molecular Signatures and Networks Underlying Selective Regional Vulnerability to Alzheimer's Disease. <em>Genome Medicine</em> 8: 104.
</p>


<h3>Details</h3>

<p>There are two sample pipelines available:
</p>

<ol>
<li>
<p>Analysis of a real dataset of gene coexpression network modules can be reached through <code>vignette("netweaver")</code>.<br />
</p>
</li>
<li>
<p>Analysis of a hypothetical data is shown in sample code of <code><a href="#topic+rc.initialize">rc.initialize</a></code>.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;, Bin Zhang &lt;bin.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Wang M, Roussos P, McKenzie A, Zhou X, Kajiwara Y, Brennand K, DeLuca GC, Crary JF, Casaccia P, Buxbaum J et al. 2016. Integrative Network Analysis of Nineteen Brain Regions Identifies Molecular Signatures and Networks Underlying Selective Regional Vulnerability to Alzheimer's Disease. <em>Genome Medicine</em> 8: 104.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.initialize">rc.initialize</a></code>, <code><a href="#topic+Modules">Modules</a></code></p>

<hr>
<h2 id='rc.get.coordinates'>Get Coordinates</h2><span id='topic+rc.get.coordinates'></span><span id='topic+rc.get.trackCoordinates'></span>

<h3>Description</h3>

<p>Retrieve x and y coordinates in a track.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.get.trackCoordinates(track.id, Start, End, Chr=NULL, degree=NULL, trackThickness=NULL)
rc.get.coordinates(track.id, Pos, Chr=NULL, degree=NULL, innerSide=TRUE, bottomSide=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.get.coordinates_+3A_track.id">track.id</code></td>
<td>
<p>number of track from outermost.</p>
</td></tr>
<tr><td><code id="rc.get.coordinates_+3A_start">Start</code>, <code id="rc.get.coordinates_+3A_end">End</code></td>
<td>
<p>start/end position on a chromosome or since first chromosome. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.get.coordinates_+3A_pos">Pos</code></td>
<td>
<p>a vector of positions on a chromosome or since first chromosome. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.get.coordinates_+3A_chr">Chr</code></td>
<td>
<p>a chromosome id. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.get.coordinates_+3A_degree">degree</code></td>
<td>
<p>the angle of the arc rotation, overwrites <code>Start</code>, <code>End</code>, <code>Pos</code> and <code>Chr</code>.</p>
</td></tr>
<tr><td><code id="rc.get.coordinates_+3A_trackthickness">trackThickness</code></td>
<td>
<p>thickness (height) of a track.</p>
</td></tr>
<tr><td><code id="rc.get.coordinates_+3A_innerside">innerSide</code></td>
<td>
<p>whether to compute coordinates on the inner side of a track. Outer side coordinates will be returned if FALSE. Used for <code>circular</code> layout.</p>
</td></tr>
<tr><td><code id="rc.get.coordinates_+3A_bottomside">bottomSide</code></td>
<td>
<p>whether to compute coordinates on the bottom side of a track. Upper side coordinates will be returned if FALSE. Used for <code>landscape</code> layout.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>Chr</code> is <code>NULL</code>, <code>Start</code>, <code>End</code> and <code>Pos</code> will be considered as cumulative positions since first chromosome.
</p>


<h3>Value</h3>

<p>A list of x and y coordinates.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.initialize">rc.initialize</a></code>, <code><a href="#topic+rc.reset.params">rc.reset.params</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.get.params'>Get Parameters</h2><span id='topic+rc.get.params'></span>

<h3>Description</h3>

<p>Retrieve parameters of Circos plot after initialization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.get.params()</code></pre>


<h3>Value</h3>

<p>A list.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.initialize">rc.initialize</a></code>, <code><a href="#topic+rc.reset.params">rc.reset.params</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.initialize'>Initialize Circos Plot Parameters</h2><span id='topic+rc.initialize'></span>

<h3>Description</h3>

<p>Setup default parameter for Circos plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.initialize(cyto.info, num.tracks=NULL, chr.order=NULL, 
	stepUnit=10^7, Layout=c('circular','landscape'), params=list())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.initialize_+3A_cyto.info">cyto.info</code></td>
<td>
<p>data.frame, see <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.initialize_+3A_num.tracks">num.tracks</code></td>
<td>
<p>integer, number of tracks.</p>
</td></tr>
<tr><td><code id="rc.initialize_+3A_chr.order">chr.order</code></td>
<td>
<p>character vector of chromosome ids specifying the ordering of chromosomes.</p>
</td></tr>
<tr><td><code id="rc.initialize_+3A_stepunit">stepUnit</code></td>
<td>
<p>integer, smoothing factor for faster plotting.</p>
</td></tr>
<tr><td><code id="rc.initialize_+3A_layout">Layout</code></td>
<td>
<p>layout for plotting.</p>
</td></tr>
<tr><td><code id="rc.initialize_+3A_params">params</code></td>
<td>
<p>a list of named items. See <code>Details</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>cyto.info</code> is a data.frame of chromosomal position ordered cytobands, with columns: <code>Chr</code>, <code>Start</code>, <code>End</code>, <code>Stain</code>, and any additional information (like band color),
where <code>Chr</code> is chromosome name, <code>Start</code> and <code>End</code> are the start and end positions on the chromosome, and <code>Stain</code> is the cyto stain.
The stain is normally one of the &quot;gneg&quot;, &quot;acen&quot;, &quot;stalk&quot;, &quot;gvar&quot;, &quot;gpos&quot;, &quot;gpos100&quot;, &quot;gpos75&quot;, &quot;gpos66&quot;, &quot;gpos50&quot;, &quot;gpos33&quot;, and &quot;gpos25&quot;, which will be plotted by color
&quot;white&quot;, &quot;red&quot;, &quot;steelblue&quot;, &quot;lightgrey&quot;, &quot;black&quot;, &quot;black&quot;, &quot;gray40&quot;, &quot;gray50&quot;, &quot;gray60&quot;, &quot;gray70&quot; and &quot;gray80&quot;, accordingly. Customized colors for the cyto bands can be specified in
an additional column named &quot;<code>BandColor</code>&quot;.<br /><br />
Additional plot parameters can be specified through argument <code>params</code>, including:
</p>

<ul>
<li> <p><code>color.line</code>, color for lines and links, default &quot;black&quot;.
</p>
</li>
<li> <p><code>chr.padding</code>, padding between chromsomes is a fraction of the total chromosome sizes, default 0.1.
</p>
</li>
<li> <p><code>track.padding</code>, paddings between tracks is a fraction of the track height, default 0.1
</p>
</li>
<li> <p><code>track.height</code>, track height, default 0.15.
</p>
</li>
<li> <p><code>radius</code> radius of the circos, default 1.
</p>
</li>
<li> <p><code>sector.degree</code>, a value between 0 and <code>2*Pi</code> (the default) specifying the circular sector size of the circos plot.
</p>
</li></ul>

<p>After initialization, the parameter settings can be retrieved by <code><a href="#topic+rc.get.params">rc.get.params</a></code>.
</p>
<p>Noted that while <code>cyto.info</code> requires input to be in a form of chromosome cytobands, the input is not limited to genomic features. As illustrated in example data <code><a href="#topic+Modules">Modules</a></code>, complex features of gene coexpression network modules can also be plotted with the current circos visualization technique.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.get.params">rc.get.params</a></code>, <code><a href="#topic+rc.reset.params">rc.reset.params</a></code>, <code><a href="#topic+rc.plot.ideogram">rc.plot.ideogram</a></code>, <code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code>, <code><a href="#topic+rc.plot.mHistogram">rc.plot.mHistogram</a></code>, <code><a href="#topic+rc.plot.barchart">rc.plot.barchart</a></code>, <code><a href="#topic+rc.plot.link">rc.plot.link</a></code>, <code><a href="#topic+rc.plot.ribbon">rc.plot.ribbon</a></code>, <code><a href="#topic+Modules">Modules</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NetWeaver)
options(stringsAsFactors=FALSE)
## set fake Cyto data
Cyto=data.frame(Chr=paste0('C',1:20), Start=1, End=100, BandColor=colors()[2:21])

## initialize circos
#firstly set number of tracks, should be larger than the actual number of tracks
#used for plotting in order to leave sufficient sapce in the middle
num.tracks=9 
rc.initialize(Cyto, num.tracks=num.tracks, params=list(chr.padding=0.1,sector.degree=2*pi))

##retrieve parameters
params=rc.get.params()
#make plot area
rc.plot.area(size=0.95)

##plot ideogram on track 1 and 2 (start from the outside to inside)
track.num=1:2
rc.plot.ideogram(track.num, plot.band=TRUE, plot.chromosome.id=TRUE)

##plot histogram/barchart that span multiple chromosomes
MultHistData=data.frame(Chr1=paste0('C',seq(1,18,3)), Start1=50, 
	Chr2=paste0('C',seq(3,20,3)), End2=20, Col=1:6)
track.num=3
rc.plot.mHistogram(MultHistData, track.id=track.num, data.col=5,
	color.col=5, fixed.height=FALSE)

##plot histogram that occupies two tracks 4 and 5
HistData=data.frame(Chr=paste0('C',1:20), Start=1, End=50, Data=runif(20))
params$color.hist &lt;- 'black'
rc.reset.params(params)
track.num=5
rc.plot.histogram(HistData, track.num, data.col=4, fixed.height=FALSE, 
	track.border=NA, custom.track.height=params$track.height*2)

##plot heatmap on track 6
HeatData=data.frame(Chr=paste0('C',1:20), Start=1,End=100, Data=1:20)
colfuncHeat=function(n) rev(heat.colors(n))
track.num=track.num+1
rc.plot.histogram(HeatData, track.num, data.col=4, color.gradient=colfuncHeat(50),
	fixed.height=TRUE)

##plot stacked barchart on track 7
BarData=data.frame(Chr=paste0('C',1:20), Start=1, 
	End=seq(10,86,length.out=20), Data=matrix(runif(20*4),nrow=20))
track.num=track.num+1
rc.plot.barchart(BarData, track.num, data.col=4:7)

##plot links in the middle
LinkData=data.frame(Chr1=sample(Cyto$Chr,40,replace=TRUE), Pos1=20, 
	Chr2=sample(Cyto$Chr,40,replace=TRUE),Pos2=20, Data=runif(20))
LinkData=LinkData[LinkData$Chr1 != LinkData$Chr2,]
params$color.line='blue'
rc.reset.params(params)
track.num=track.num+1
rc.plot.link(LinkData, track.num, data.col=4, arrow.length=0.1)
ribbonData=data.frame(Chr1=c('C1','C3'), Start1=c(10,10), End1=c(40,40),
	Chr2=c('C17','C10'), Start2=20, End2=60, Col=c('red','brown'))
rc.plot.ribbon(ribbonData, track.num, color.col='Col', twist=TRUE)
#label track id
rc.plot.track.id(2:7, col=2)
#add text label
rc.plot.text(data.frame(Chr='C3',Pos=50,Label='GeneX'), 
	track.id=3,srt=45,cex=0.8,col='blue')
#add line mark
rc.plot.line(data.frame(Chr='C19',Pos=seq(10,90,by=15),Col='red'), 
	track.id=3, color.col=3,arrow.length=0.2)
</code></pre>

<hr>
<h2 id='rc.plot.area'>Setup Plot Area</h2><span id='topic+rc.plot.area'></span>

<h3>Description</h3>

<p>Create new graphics frame for circle plot .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.area(size=1, oma=rep(0,4), mar=rep(0,4))</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.area_+3A_size">size</code></td>
<td>
<p>a value between 0 to 1, specifying the effective size of the circle plot area in the current window. The smaller the <code>size</code>, the larger the blank area around the circle plot.</p>
</td></tr>
<tr><td><code id="rc.plot.area_+3A_oma">oma</code></td>
<td>
<p>A vector of the form <code>c(bottom, left, top, right)</code> giving the size of the outer margins in lines of text.</p>
</td></tr>
<tr><td><code id="rc.plot.area_+3A_mar">mar</code></td>
<td>
<p>A numerical vector of the form <code>c(bottom, left, top, right)</code> which gives the number of lines of margin to be specified on the four sides of the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>After setting up, the extremes of the user coordinates of the plotting region can be found out by <code>par('usr')</code>.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.initialize">rc.initialize</a></code>, <code><a href="#topic+rc.reset.params">rc.reset.params</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.barchart'>Bar Plots</h2><span id='topic+rc.plot.barchart'></span>

<h3>Description</h3>

<p>Add a track of barcharts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.barchart(Data, track.id, data.col, color.col=NULL, bar.color=NULL,
	track.color=NA, track.border=NULL, polygon.border=NULL,
	custom.track.height=NULL, ratio=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.barchart_+3A_data">Data</code></td>
<td>
<p>data.frame, data to be plotted. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.barchart_+3A_track.id">track.id</code></td>
<td>
<p>integer, in which track to plot.</p>
</td></tr>
<tr><td><code id="rc.plot.barchart_+3A_data.col">data.col</code></td>
<td>
<p>integer, specifying the column(s) that contain the data for plotting. A stacked barchart is created if there are multiple columns of data.</p>
</td></tr>
<tr><td><code id="rc.plot.barchart_+3A_color.col">color.col</code></td>
<td>
<p>integer, specifying the column that contains the color. Only used when there is only one data column.</p>
</td></tr>
<tr><td><code id="rc.plot.barchart_+3A_bar.color">bar.color</code></td>
<td>
<p>a vector of colors for the bars or bar components. By default, <code>grey</code> is used if there is one data column, and a gamma-corrected grey palette if there are multiple columns of data.</p>
</td></tr>
<tr><td><code id="rc.plot.barchart_+3A_track.color">track.color</code></td>
<td>
<p>NA or color code for the track background.</p>
</td></tr>
<tr><td><code id="rc.plot.barchart_+3A_track.border">track.border</code></td>
<td>
<p>the color to draw the track border. Use <code>NA</code> to omit borders.</p>
</td></tr>
<tr><td><code id="rc.plot.barchart_+3A_polygon.border">polygon.border</code></td>
<td>
<p>the color to draw the polygon border. See explanation for <code><a href="graphics.html#topic+polygon">polygon</a></code>.</p>
</td></tr>
<tr><td><code id="rc.plot.barchart_+3A_custom.track.height">custom.track.height</code></td>
<td>
<p>NULL or numeric, specifying customized track height to overwrite the default. See <code><a href="#topic+rc.get.params">rc.get.params</a></code> for default value.</p>
</td></tr>
<tr><td><code id="rc.plot.barchart_+3A_ratio">ratio</code></td>
<td>
<p>logical, specifying whether the data values in each row should be converted to ratios. Used only for stacked barchart.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Data</code> must have at least three columns. The first three columns must be named as <code>Chr</code>, <code>Start</code> and <code>End</code>, specifying the chromosomes and positions for the start and end points of the links. Additional columns can be used to specify data and color.
</p>
<p><code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code> can also be used for plotting barcharts except stacked barchart.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.link">rc.plot.link</a></code>, <code><a href="#topic+rc.plot.mHistogram">rc.plot.mHistogram</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.grColLegend'>Plot Gradient Color Legend</h2><span id='topic+rc.plot.grColLegend'></span>

<h3>Description</h3>

<p>Plot gradient color legend at given position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.grColLegend(x, y, cols, at=NULL, legend=at,
	title='Color', width=0.1, height=0.3, gap=0,
	direction=c('vertical', 'horizontal'), cex.text=1, cex.title=1)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.grColLegend_+3A_x">x</code>, <code id="rc.plot.grColLegend_+3A_y">y</code></td>
<td>
<p>numeric values of coordinates where the legend is plotted. See <code>details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.grColLegend_+3A_cols">cols</code></td>
<td>
<p>a vector of colors.</p>
</td></tr>
<tr><td><code id="rc.plot.grColLegend_+3A_at">at</code></td>
<td>
<p>integers; index <code>legend</code> text in the color vector <code>cols</code>. See <code>details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.grColLegend_+3A_legend">legend</code></td>
<td>
<p>character strings of legend text with the same length of vector <code>at</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.grColLegend_+3A_title">title</code></td>
<td>
<p>character string of the legend title.</p>
</td></tr>
<tr><td><code id="rc.plot.grColLegend_+3A_width">width</code></td>
<td>
<p>numeric, width of the color legend bar.</p>
</td></tr>
<tr><td><code id="rc.plot.grColLegend_+3A_height">height</code></td>
<td>
<p>numeric, height of the color legend  bar.</p>
</td></tr>
<tr><td><code id="rc.plot.grColLegend_+3A_gap">gap</code></td>
<td>
<p>numeric, size of the gap between two color blocks.</p>
</td></tr>
<tr><td><code id="rc.plot.grColLegend_+3A_direction">direction</code></td>
<td>
<p>plot the color legend bar in either vertical or horizontal direction.</p>
</td></tr>
<tr><td><code id="rc.plot.grColLegend_+3A_cex.text">cex.text</code></td>
<td>
<p>numeric, magnitude of the legend text.</p>
</td></tr>
<tr><td><code id="rc.plot.grColLegend_+3A_cex.title">cex.title</code></td>
<td>
<p>numeric, magnitude of the legend title.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>x</code> and <code>y</code> set the coordinates of bottom left starting point. The extremes of the user coordinates of the plotting region can be found out by <code>par('usr')</code>.
</p>
<p>By default <code>NULL</code>, parameter <code>at</code> will be set as a vector with values 1, <code>ceiling(length(cols)/2)</code>, and <code>length(cols)</code>.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.link">rc.plot.link</a></code>, <code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.heatmap'>Plot Heat-map</h2><span id='topic+rc.plot.heatmap'></span>

<h3>Description</h3>

<p>Add multiple tracks of heat-map on a circos plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.heatmap(Data, track.id, color.gradient=NULL,
	track.color=NA, track.border=NULL, polygon.border=NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.heatmap_+3A_data">Data</code></td>
<td>
<p>matrix, data to be plotted. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.heatmap_+3A_track.id">track.id</code></td>
<td>
<p>integer, starting track id.</p>
</td></tr>
<tr><td><code id="rc.plot.heatmap_+3A_color.gradient">color.gradient</code></td>
<td>
<p>a vector of gradient colors. See <code>details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.heatmap_+3A_track.color">track.color</code></td>
<td>
<p>NA or color code for the track background.</p>
</td></tr>
<tr><td><code id="rc.plot.heatmap_+3A_track.border">track.border</code></td>
<td>
<p>the color to draw the track border. Use <code>NA</code> to omit borders.</p>
</td></tr>
<tr><td><code id="rc.plot.heatmap_+3A_polygon.border">polygon.border</code></td>
<td>
<p>the color to draw the polygon border. See explanation for <code><a href="graphics.html#topic+polygon">polygon</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function employs <code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code> to plot heat-map. Every row of <code>Data</code> will be plotted as heat-map in one track. The column names of <code>Data</code> must be the chromosome ids.
</p>
<p>If <code>color.gradient</code> is not <code>NULL</code>, the data will be scaled to positive integers in the range of 1~length(color.gradient) to index the colors in vector <code>color.gradient</code>. If <code>color.gradient</code> is<code>NULL</code>, the input <code>Data</code> must be a matrix of colors or anything that can be converted to colors.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.histogram'>Plot Histogram, Barchart or Heat-map</h2><span id='topic+rc.plot.histogram'></span>

<h3>Description</h3>

<p>Add a track of histogram, barchart, or heat-map on a circos plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.histogram(Data, track.id, data.col=NULL, color.col=NULL,
 color.gradient=NULL, fixed.height=FALSE, track.color=NA, track.border=NULL,
 polygon.border=NULL, custom.track.height=NULL, max.value=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.histogram_+3A_data">Data</code></td>
<td>
<p>data.frame, data to be plotted. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.histogram_+3A_track.id">track.id</code></td>
<td>
<p>integer, in which track to plot.</p>
</td></tr>
<tr><td><code id="rc.plot.histogram_+3A_data.col">data.col</code></td>
<td>
<p>integer, specifying the column that contains the data for plotting.</p>
</td></tr>
<tr><td><code id="rc.plot.histogram_+3A_color.col">color.col</code></td>
<td>
<p>integer, specifying the column that contains the color.</p>
</td></tr>
<tr><td><code id="rc.plot.histogram_+3A_color.gradient">color.gradient</code></td>
<td>
<p>a vector of gradient colors. See <code>details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.histogram_+3A_fixed.height">fixed.height</code></td>
<td>
<p>logical, whether to fix the histogram height which results in heat-map. <code>TRUE</code> if <code>data.col</code> is <code>bar.color</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.histogram_+3A_track.color">track.color</code></td>
<td>
<p>NA or color code for the track background.</p>
</td></tr>
<tr><td><code id="rc.plot.histogram_+3A_track.border">track.border</code></td>
<td>
<p>the color to draw the track border. Use <code>NA</code> to omit borders.</p>
</td></tr>
<tr><td><code id="rc.plot.histogram_+3A_polygon.border">polygon.border</code></td>
<td>
<p>the color to draw the polygon border. See explanation for <code><a href="graphics.html#topic+polygon">polygon</a></code>.</p>
</td></tr>
<tr><td><code id="rc.plot.histogram_+3A_custom.track.height">custom.track.height</code></td>
<td>
<p>NULL or numeric, specifying customized track height to overwrite the default. See <code><a href="#topic+rc.get.params">rc.get.params</a></code> for default value.</p>
</td></tr>
<tr><td><code id="rc.plot.histogram_+3A_max.value">max.value</code></td>
<td>
<p>NULL or numeric, specifying the maximum data value for normalization. Default NULL, get max from data column.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Data</code> must have at least three columns. The first three columns must be named as <code>Chr</code>, <code>Start</code> and <code>End</code>, specifying the chromosomes and positions for the start and end points of the links. Additional columns can be used to specify data and color.
</p>
<p>If <code>color.gradient</code> is not <code>NULL</code>, <code>color.col</code> is ignored and the data will be scaled to positive integers in the range of 1~length(color.gradient) to index the colors in vector <code>color.gradient</code>.
</p>
<p>For plotting stacked barcharts, use function <code><a href="#topic+rc.plot.barchart">rc.plot.barchart</a></code> instead.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.link">rc.plot.link</a></code>, <code><a href="#topic+rc.plot.mHistogram">rc.plot.mHistogram</a></code>, <code><a href="#topic+rc.plot.barchart">rc.plot.barchart</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.ideogram'>Plot Ideogram</h2><span id='topic+rc.plot.ideogram'></span>

<h3>Description</h3>

<p>Add chromosome ideogram on circos plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.ideogram(track.ids, plot.band=TRUE, plot.chromosome.id=TRUE, 
	chrom.alias=NULL, color.chromosome.id=NULL, cex.text=1, track.border=NULL,
	polygon.border=NULL, las=0, custom.track.height=NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.ideogram_+3A_track.ids">track.ids</code></td>
<td>
<p>vector, two integers specifying the two tacks for plotting chromosome name and cyto band respectively.</p>
</td></tr>
<tr><td><code id="rc.plot.ideogram_+3A_plot.band">plot.band</code></td>
<td>
<p>logic, whether to plot ideogram cyto band.</p>
</td></tr>
<tr><td><code id="rc.plot.ideogram_+3A_plot.chromosome.id">plot.chromosome.id</code></td>
<td>
<p>logic, whether to plot chromosome id.</p>
</td></tr>
<tr><td><code id="rc.plot.ideogram_+3A_chrom.alias">chrom.alias</code></td>
<td>
<p>NULL or a chromosome named vector of alias.</p>
</td></tr>
<tr><td><code id="rc.plot.ideogram_+3A_color.chromosome.id">color.chromosome.id</code></td>
<td>
<p>NULL or a chromosome named vector of colors.</p>
</td></tr>
<tr><td><code id="rc.plot.ideogram_+3A_cex.text">cex.text</code></td>
<td>
<p>numeric, scale of text.</p>
</td></tr>
<tr><td><code id="rc.plot.ideogram_+3A_track.border">track.border</code></td>
<td>
<p>the color to draw the track border. Use NA to omit borders.</p>
</td></tr>
<tr><td><code id="rc.plot.ideogram_+3A_polygon.border">polygon.border</code></td>
<td>
<p>the color to draw the polygon border. See explanation for <code><a href="graphics.html#topic+polygon">polygon</a></code>.</p>
</td></tr>
<tr><td><code id="rc.plot.ideogram_+3A_las">las</code></td>
<td>
<p>numeric in 0,1,2,3; the style of chromosome labels. 0, always parallel to the track [default]; 1, always horizontal; 2, always perpendicular to the track; 3, always vertical.</p>
</td></tr>
<tr><td><code id="rc.plot.ideogram_+3A_custom.track.height">custom.track.height</code></td>
<td>
<p>NULL or numeric, specifying customized track height when plotting cyto band.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.line'>Plot Line</h2><span id='topic+rc.plot.line'></span>

<h3>Description</h3>

<p>Plot lines at designated position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.line(Data, track.id, color.col=NULL, custom.length=NULL, arrow.length=0,
	arrow.angle=30, arrow.code=2, ...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.line_+3A_data">Data</code></td>
<td>
<p>a data.frame. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.line_+3A_track.id">track.id</code></td>
<td>
<p>a vector of integers, specifying the tracks for plotting line.</p>
</td></tr>
<tr><td><code id="rc.plot.line_+3A_color.col">color.col</code></td>
<td>
<p>integer, specifying the column that contains the color.</p>
</td></tr>
<tr><td><code id="rc.plot.line_+3A_custom.length">custom.length</code></td>
<td>
<p><code>NULL</code> or numeric, specifying line length.</p>
</td></tr>
<tr><td><code id="rc.plot.line_+3A_arrow.length">arrow.length</code></td>
<td>
<p>length of the edges of the arrow head (in inches). See <code><a href="graphics.html#topic+arrows">arrows</a></code></p>
</td></tr>
<tr><td><code id="rc.plot.line_+3A_arrow.angle">arrow.angle</code></td>
<td>
<p>angle from the shaft of the arrow to the edge of the arrow head. See <code><a href="graphics.html#topic+arrows">arrows</a></code></p>
</td></tr>
<tr><td><code id="rc.plot.line_+3A_arrow.code">arrow.code</code></td>
<td>
<p>integer code, determining kind of arrows to be drawn. See <code><a href="graphics.html#topic+arrows">arrows</a></code></p>
</td></tr>
<tr><td><code id="rc.plot.line_+3A_...">...</code></td>
<td>
<p>further graphical parameters (from <code><a href="graphics.html#topic+par">par</a></code>), such as lty and lwd.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Data</code> is a data.frame, with at least two columns named 'Chr' and 'Pos', specifying the chromosomal positoins for each line in every row. An additional column can be used to specify the line color. Default line length is determined by track height.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code>, <code><a href="#topic+rc.plot.track">rc.plot.track</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.link'>Plot Link</h2><span id='topic+rc.plot.link'></span>

<h3>Description</h3>

<p>Add a track of links on circos plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.link(Data, track.id, data.col=NULL, color.col = NULL,
 max.lwd=1, sort.links=TRUE, arrow.length=0, arrow.angle=30, ...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.link_+3A_data">Data</code></td>
<td>
<p>data.frame of at least four columns. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.link_+3A_track.id">track.id</code></td>
<td>
<p>integer, the track number for plotting.</p>
</td></tr>
<tr><td><code id="rc.plot.link_+3A_data.col">data.col</code></td>
<td>
<p>integer or character string, specifies the column of data.</p>
</td></tr>
<tr><td><code id="rc.plot.link_+3A_color.col">color.col</code></td>
<td>
<p>integer or character string, specifies the column of colors.</p>
</td></tr>
<tr><td><code id="rc.plot.link_+3A_max.lwd">max.lwd</code></td>
<td>
<p>integer, maximum line width.</p>
</td></tr>
<tr><td><code id="rc.plot.link_+3A_sort.links">sort.links</code></td>
<td>
<p>logical, whether to re-assign link start and end positions by sorting them. See <code>details</code></p>
</td></tr>
<tr><td><code id="rc.plot.link_+3A_arrow.length">arrow.length</code></td>
<td>
<p>length of the edges of the arrow head (in inches). See <code><a href="graphics.html#topic+arrows">arrows</a></code></p>
</td></tr>
<tr><td><code id="rc.plot.link_+3A_arrow.angle">arrow.angle</code></td>
<td>
<p>angle from the shaft of the arrow to the edge of the arrow head. See <code><a href="graphics.html#topic+arrows">arrows</a></code></p>
</td></tr>
<tr><td><code id="rc.plot.link_+3A_...">...</code></td>
<td>
<p>further graphical parameters (from <code><a href="graphics.html#topic+par">par</a></code>), such as lty and lwd.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Data</code> must have at least four columns: <code>Chr1</code>, <code>Pos1</code>, <code>Chr2</code> and <code>Pos2</code>, specifying the chromosomes and positions for the start and end points of the links. Additional columns may be used to specify the data, color and max line width.
</p>
<p>If <code>data.col</code> is not NULL, the line width will be proportional to the values in column <code>data.col</code> with the maximum line width determined by <code>max.lwd</code>.
</p>
<p>If <code>sort.links</code> is true, the links originate from the same chromosome will be sorted and equally spaced to minimize crossing.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code>, <code><a href="#topic+rc.plot.ribbon">rc.plot.ribbon</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.mHistogram'>Plot Histogram or Barplot Across Chromosomes</h2><span id='topic+rc.plot.mHistogram'></span>

<h3>Description</h3>

<p>Plot histogram or barplot that overlaps with multiple chromosomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.mHistogram(Data, track.id, data.col, color.col=NULL, color.gradient=NULL,
 fixed.height=FALSE, track.color=NA, track.border=NULL, polygon.border=NULL,
 custom.track.height=NULL, max.value=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.mHistogram_+3A_data">Data</code></td>
<td>
<p>data.frame of at least four columns. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.mHistogram_+3A_track.id">track.id</code></td>
<td>
<p>integer, the track number for plotting.</p>
</td></tr>
<tr><td><code id="rc.plot.mHistogram_+3A_data.col">data.col</code></td>
<td>
<p>integer or character string, specifies the column of data.</p>
</td></tr>
<tr><td><code id="rc.plot.mHistogram_+3A_color.col">color.col</code></td>
<td>
<p>integer or character string, specifies the column of colors.</p>
</td></tr>
<tr><td><code id="rc.plot.mHistogram_+3A_color.gradient">color.gradient</code></td>
<td>
<p>a vector of gradient colors.</p>
</td></tr>
<tr><td><code id="rc.plot.mHistogram_+3A_fixed.height">fixed.height</code></td>
<td>
<p>logical, whether to fix the histogram height, resulting in heat-map if <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.mHistogram_+3A_track.color">track.color</code></td>
<td>
<p>NA or color code for the track background.</p>
</td></tr>
<tr><td><code id="rc.plot.mHistogram_+3A_track.border">track.border</code></td>
<td>
<p>the color to draw the track border. Use <code>NA</code> to omit borders.</p>
</td></tr>
<tr><td><code id="rc.plot.mHistogram_+3A_polygon.border">polygon.border</code></td>
<td>
<p>the color to draw the polygon border. See explanation for <code><a href="graphics.html#topic+polygon">polygon</a></code>.</p>
</td></tr>
<tr><td><code id="rc.plot.mHistogram_+3A_custom.track.height">custom.track.height</code></td>
<td>
<p>NULL or numeric, specifying customized track height to overwrite the default. See <code><a href="#topic+rc.get.params">rc.get.params</a></code> for default value.</p>
</td></tr>
<tr><td><code id="rc.plot.mHistogram_+3A_max.value">max.value</code></td>
<td>
<p>NULL or numeric, specifying the maximum data value for normalization. Default NULL, get max from data column <code>data.col</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Data</code> must have at least four columns: <code>Chr1</code>, <code>Start1</code>, <code>Chr2</code> and <code>End2</code>, specifying the chromosomes and positions for the start and end points. Additional columns may be used to specify the data, color, etc.<br /><br />
If <code>color.gradient</code> is not <code>NULL</code>, it will supersede <code>color.col</code>.<br /><br />
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.link">rc.plot.link</a></code>, <code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.point'>Plot Point</h2><span id='topic+rc.plot.point'></span>

<h3>Description</h3>

<p>Plot point(s) at designated position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.point(Data, track.id, color.col=NULL, custom.track.height=NULL, ...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.point_+3A_data">Data</code></td>
<td>
<p>a data.frame. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.point_+3A_track.id">track.id</code></td>
<td>
<p>a vector of integers, specifying the tracks for plotting point(s).</p>
</td></tr>
<tr><td><code id="rc.plot.point_+3A_color.col">color.col</code></td>
<td>
<p>column id in <code>Data</code> specifying the colors.</p>
</td></tr>
<tr><td><code id="rc.plot.point_+3A_custom.track.height">custom.track.height</code></td>
<td>
<p>NULL or numeric, specifying customized track height.</p>
</td></tr>
<tr><td><code id="rc.plot.point_+3A_...">...</code></td>
<td>
<p>further graphical parameters (from <code><a href="graphics.html#topic+par">par</a></code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Data</code> is a data.frame with at least two columns named 'Chr', 'Pos', and 'Height', specifying the chrosomal positoin and the height within the track of the points. &quot;<code>Height</code>&quot; will be sacled to have a maximum of 1.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code>, <code><a href="#topic+rc.plot.track">rc.plot.track</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.ribbon'>Plot Ribbon</h2><span id='topic+rc.plot.ribbon'></span>

<h3>Description</h3>

<p>Add a track of ribbons on circos plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.ribbon(Data, track.id, color.col = NULL, twist=FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.ribbon_+3A_data">Data</code></td>
<td>
<p>data.frame of at least six columns. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.ribbon_+3A_track.id">track.id</code></td>
<td>
<p>integer, the track number for plotting.</p>
</td></tr>
<tr><td><code id="rc.plot.ribbon_+3A_color.col">color.col</code></td>
<td>
<p>integer or character string, specifies the column of colors.</p>
</td></tr>
<tr><td><code id="rc.plot.ribbon_+3A_twist">twist</code></td>
<td>
<p>logic, whehter to twist the ribbon.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Data</code> must have at least six columns: <code>Chr1</code>, <code>Start1</code>, <code>End1</code>, <code>Chr2</code>, <code>Start2</code> and <code>End2</code>, specifying the chromosomes and positions for the start and end points of the ribbons. Additional columns may be used to specify the color.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code>, <code><a href="#topic+rc.plot.link">rc.plot.link</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.text'>Plot Text</h2><span id='topic+rc.plot.text'></span>

<h3>Description</h3>

<p>Plot text labels at designated position.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.text(Data, track.id, col='black', custom.track.height=NULL, ...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.text_+3A_data">Data</code></td>
<td>
<p>a data.frame. See <code>Details</code>.</p>
</td></tr>
<tr><td><code id="rc.plot.text_+3A_track.id">track.id</code></td>
<td>
<p>a vector of integers, specifying the tracks for plotting text.</p>
</td></tr>
<tr><td><code id="rc.plot.text_+3A_col">col</code></td>
<td>
<p>a vector of color for the text.</p>
</td></tr>
<tr><td><code id="rc.plot.text_+3A_custom.track.height">custom.track.height</code></td>
<td>
<p>NULL or numeric, specifying customized track height.</p>
</td></tr>
<tr><td><code id="rc.plot.text_+3A_...">...</code></td>
<td>
<p>further graphical parameters (from <code><a href="graphics.html#topic+par">par</a></code>), such as srt and family.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Data</code> is a data.frame with at least three columns named 'Chr', 'Pos', and 'Label', specifying the chrosomal positoin and text labels.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code>, <code><a href="#topic+rc.plot.track">rc.plot.track</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.track'>Plot Track</h2><span id='topic+rc.plot.track'></span>

<h3>Description</h3>

<p>Add a track to a circos plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.track(track.id, border='black', col=NA, custom.track.height=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.track_+3A_track.id">track.id</code></td>
<td>
<p>integer, track id.</p>
</td></tr>
<tr><td><code id="rc.plot.track_+3A_border">border</code></td>
<td>
<p>the color to draw the track border. Use NA to omit borders.</p>
</td></tr>
<tr><td><code id="rc.plot.track_+3A_col">col</code></td>
<td>
<p>NA or color code for the track background.</p>
</td></tr>
<tr><td><code id="rc.plot.track_+3A_custom.track.height">custom.track.height</code></td>
<td>
<p>NULL or numeric, specifying customized track height. See <code>details</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>custom.track.height</code> sets customized track height, which will be useful for track(s) requiring bigger/smaller height than default.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code>, <code><a href="#topic+rc.plot.mHistogram">rc.plot.mHistogram</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.plot.track.id'>Plot Track Id</h2><span id='topic+rc.plot.track.id'></span>

<h3>Description</h3>

<p>Plot labels in designated tracks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.plot.track.id(track.id, labels=NULL, degree=0,
	col='black', custom.track.height=NULL, ...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.plot.track.id_+3A_track.id">track.id</code></td>
<td>
<p>a vector of integers, specifying the tracks for plotting id.</p>
</td></tr>
<tr><td><code id="rc.plot.track.id_+3A_labels">labels</code></td>
<td>
<p>NULL or a vector of character string, specifying the text to be written.</p>
</td></tr>
<tr><td><code id="rc.plot.track.id_+3A_degree">degree</code></td>
<td>
<p>the angle of the arc rotation.</p>
</td></tr>
<tr><td><code id="rc.plot.track.id_+3A_col">col</code></td>
<td>
<p>color for the text.</p>
</td></tr>
<tr><td><code id="rc.plot.track.id_+3A_custom.track.height">custom.track.height</code></td>
<td>
<p>NULL or numeric, specifying customized track height.</p>
</td></tr>
<tr><td><code id="rc.plot.track.id_+3A_...">...</code></td>
<td>
<p>further graphical parameters (from par), such as srt and family.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>labels</code> is NULL, values of <code>track.id</code> will be used as text labels.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.plot.histogram">rc.plot.histogram</a></code>, <code><a href="#topic+rc.plot.track">rc.plot.track</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.reset.params'>Reset Circos Plot Parameters</h2><span id='topic+rc.reset.params'></span>

<h3>Description</h3>

<p>Reset parameters for Circos plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.reset.params(params)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.reset.params_+3A_params">params</code></td>
<td>
<p>list.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.initialize">rc.initialize</a></code>, <code><a href="#topic+rc.get.params">rc.get.params</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='rc.track.pos'>Get Track Positions</h2><span id='topic+rc.track.pos'></span>

<h3>Description</h3>

<p>Calculate inner and outer radius of a track.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rc.track.pos(track.id)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rc.track.pos_+3A_track.id">track.id</code></td>
<td>
<p>integer, number of the track from the outermost.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with two radius values: out.pos and in.pos.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.initialize">rc.initialize</a></code>, <code><a href="#topic+rc.reset.params">rc.reset.params</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#This is not to be run alone. Please see tutorial vignette("netweaver") for usage.
</code></pre>

<hr>
<h2 id='ucsc.hg19.cytoband'>Human chromosome cytoband</h2><span id='topic+ucsc.hg19.cytoband'></span><span id='topic+ucsc.hg38.cytoband'></span>

<h3>Description</h3>

<p>Human chromosome cytoband information from UCSC  (http://genome.ucsc.edu/).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ucsc.hg19.cytoband)
data(ucsc.hg38.cytoband)
</code></pre>


<h3>Details</h3>

<p>Cytoband is useful for drawing the chromosome ideograms for the genome. Two human cytoband datasets are provided in this package, corresponding to two genome versions hg19 and hg38. Each cytoband dataset is a five-column tab-delimited data.frame. Each row of the data describes the position of a cytogenetic band.
</p>


<h3>Author(s)</h3>

<p>Minghui Wang &lt;m.h.wang@live.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rc.initialize">rc.initialize</a></code></p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
