<!DOCTYPE html><html><head><title>Help for package neuroimaGene</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {neuroimaGene}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#anno'><p>NIDP annotation data</p></a></li>
<li><a href='#check_db'><p>Check neuroimaGene database downloaded</p></a></li>
<li><a href='#fs_anno'><p>NIDP freesurfer annotation data</p></a></li>
<li><a href='#listNIDPs'><p>List NIDPs</p></a></li>
<li><a href='#neuro_vis'><p>2D visualization plot of a neuroimaGene object</p></a></li>
<li><a href='#neuroimaGene'><p>Main neuroimaGene query</p></a></li>
<li><a href='#ng_vignette'><p>NeuroimaGene object for vignette illustration</p></a></li>
<li><a href='#plot_gnNIDP'><p>Gene by NIDP summary plot of neuroimaGene object</p></a></li>
<li><a href='#plot_gns'><p>Gene summary plot of neuroimaGene object</p></a></li>
<li><a href='#plot_nidps'><p>NIDP summary plot of NeuroimaGene object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Transcriptomic Atlas of Neuroimaging Derived Phenotypes</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xavier Bledsoe &lt;xavier.bledsoe@vanderbilt.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains functions to query and visualize the Neuroimaging features 
  associated with genetically regulated gene expression (GReX). The primary utility,
  neuroimaGene(), relies on a list of user-defined genes and returns a table of
  neuroimaging features (NIDPs) associated with each gene. This resource 
  is designed to assist in the interpretation of genome-wide and transcriptome-wide
  association studies that evaluate brain related traits. Bledsoe (2024) 
  &lt;<a href="https://doi.org/10.1016%2Fj.ajhg.2024.06.002">doi:10.1016/j.ajhg.2024.06.002</a>&gt;. In addition there are several visualization 
  functions that generate summary plots and 2-dimensional visualizations of regional 
  brain measures. Mowinckel (2020).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, ggplot2, DBI, stringr, ggseg, RSQLite</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-02 16:43:17 UTC; xbledsoe</td>
</tr>
<tr>
<td>Author:</td>
<td>Xavier Bledsoe <a href="https://orcid.org/0000-0002-6091-9372"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Eric Gamazon <a href="https://orcid.org/0000-0003-4204-8734"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-03 15:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='anno'>NIDP annotation data</h2><span id='topic+anno'></span>

<h3>Description</h3>

<p>annotation data for all NIDPs taken from the UKbiobank
</p>


<h3>Format</h3>

<p>A data.table with 10 columns and 3935 rows:
</p>

<dl>
<dt>gwas_phenotype</dt><dd><p>character: UKB-derived neuroimaging derived phenotype</p>
</dd>
<dt>modality</dt><dd><p>character: MRI neuroimaging modality</p>
</dd>
<dt>atlas</dt><dd><p>character: neuroimaging cortical atlas</p>
</dd>
<dt>side</dt><dd><p>character: right or left hemisphere or midline/whole brain</p>
</dd>
<dt>primary</dt><dd><p>character: primary cortical region</p>
</dd>
<dt>secondary</dt><dd><p>character: secondary cortical region</p>
</dd>
<dt>region</dt><dd><p>character: named region of the brain</p>
</dd>
<dt>measurement</dt><dd><p>character: morphology measurement</p>
</dd>
<dt>fMRI_node_1</dt><dd><p>character: fMRI node 1</p>
</dd>
<dt>fMRI_node_2</dt><dd><p>character: fMRI node 2</p>
</dd>
</dl>



<h3>Value</h3>

<p>This script has no return. This is a documentation file for the
annotation dataset for all neuroimaging derived phenotypes.
</p>


<h3>Source</h3>

<p>Bledsoe, X. (2024) A transcriptomic atlas of the human brain reveals genetically determined aspects of neuropsychiatric health
</p>

<hr>
<h2 id='check_db'>Check neuroimaGene database downloaded</h2><span id='topic+check_db'></span>

<h3>Description</h3>

<p>Check if the NeuroimaGene database exists in the proper location prior to running the query and prompt user to download if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_db(timeout = 900)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_db_+3A_timeout">timeout</code></td>
<td>
<p>time to spend downloading the NeuroimaGene database in seconds (default = 900)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>no return value, called to give information on status of neuroimaGene
database and prompt user the user to download if resource file is missing.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_db(timeout = 600)
</code></pre>

<hr>
<h2 id='fs_anno'>NIDP freesurfer annotation data</h2><span id='topic+fs_anno'></span>

<h3>Description</h3>

<p>Freesurfer names for select cortical and subcortical NIDPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fs_anno)
</code></pre>


<h3>Format</h3>

<p>A data.table with 7 columns and 890 rows:
</p>

<dl>
<dt>gwas_phenotype</dt><dd><p>character: UKB-derived neuroimaging derived phenotype</p>
</dd>
<dt>atl</dt><dd><p>character: neuroimaging cortical atlas</p>
</dd>
<dt>hemisphere</dt><dd><p>character: right or left hemisphere or midline/whole brain</p>
</dd>
<dt>secondary</dt><dd><p>character: secondary cortical region</p>
</dd>
<dt>fs_name</dt><dd><p>character: freesurfer name</p>
</dd>
<dt>label</dt><dd><p>character: label name for the region</p>
</dd>
<dt>atlas</dt><dd><p>character: freesurfer name for neuroimaging cortical atlas</p>
</dd>
</dl>



<h3>Value</h3>

<p>This script has no return. This is a documentation file for the
annotation dataset for all neuroimaging derived phenotypes in the fsbrain package.
</p>


<h3>Source</h3>

<p>Bledsoe, X. (2024) A transcriptomic atlas of the human brain reveals genetically determined aspects of neuropsychiatric health
</p>

<hr>
<h2 id='listNIDPs'>List NIDPs</h2><span id='topic+listNIDPs'></span>

<h3>Description</h3>

<p>Supplies a list of all NIDPs by name for any given modality or atlas
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listNIDPs(modality = NA, atlas = NA, filename = NA, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listNIDPs_+3A_modality">modality</code></td>
<td>
<p>Neuroimaging modality. Defaults to NA; see README for additional options</p>
</td></tr>
<tr><td><code id="listNIDPs_+3A_atlas">atlas</code></td>
<td>
<p>Neuroimaging parcellation atlas for NIDP query. Defaults to NA; see README for additional options</p>
</td></tr>
<tr><td><code id="listNIDPs_+3A_filename">filename</code></td>
<td>
<p>optional filename for writing data to a table</p>
</td></tr>
<tr><td><code id="listNIDPs_+3A_verbose">verbose</code></td>
<td>
<p>print runtime messages to R console. Default to FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of NIDP names satisfying the required criteria
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dk_names &lt;- listNIDPs(modality = 'T1', atlas = 'Desikan')


</code></pre>

<hr>
<h2 id='neuro_vis'>2D visualization plot of a neuroimaGene object</h2><span id='topic+neuro_vis'></span>

<h3>Description</h3>

<p>Generates a 2D visualization plot of the neuroimaGene object. Neuroimaging
regions are defined by the atlas parameter and colored according to the
magnitude and direction of the aggregate effect from each gene in the
NeuroimaGene object. Colors can be defined by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>neuro_vis(
  ng_obj,
  atlas = "Desikan",
  lowcol = "darkred",
  midcol = "white",
  highcol = "blue4",
  title = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="neuro_vis_+3A_ng_obj">ng_obj</code></td>
<td>
<p>NeuroimaGene object produced by neuroimaGene() function</p>
</td></tr>
<tr><td><code id="neuro_vis_+3A_atlas">atlas</code></td>
<td>
<p>desired atlas for visualization. Desikan (default) Subcortex, DKT, Destrieux.</p>
</td></tr>
<tr><td><code id="neuro_vis_+3A_lowcol">lowcol</code></td>
<td>
<p>color for low end of Zscore spectrum. Default is dark red</p>
</td></tr>
<tr><td><code id="neuro_vis_+3A_midcol">midcol</code></td>
<td>
<p>color for middle of Zscore spectrum. Default is white</p>
</td></tr>
<tr><td><code id="neuro_vis_+3A_highcol">highcol</code></td>
<td>
<p>color for top end of Zscore spectrum. Default is blue4</p>
</td></tr>
<tr><td><code id="neuro_vis_+3A_title">title</code></td>
<td>
<p>optional title tag for the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>class: ggplot object depicting 2D visualization of the NIDPs from the neuroimaGene object portrayed on the brain and shaded by mean effect size.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene_list &lt;- c('TRIM35', 'PROSER3', 'EXOSC6', 'PICK1', 'UPK1A', 'ESPNL', 'ZIC4')
ng &lt;- neuroimaGene(gene_list, atlas = NA, mtc = 'BH', vignette = TRUE)
neuro_vis(ng, atlas = 'DKT')


</code></pre>

<hr>
<h2 id='neuroimaGene'>Main neuroimaGene query</h2><span id='topic+neuroimaGene'></span>

<h3>Description</h3>

<p>Using a user supplied vector of genes (Ensembl ID's or HUGO names), this function queries the NeuroimaGene resource for all statistically significant GReX-NIDP associations and returns all findings as a data.table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>neuroimaGene(
  gene_list,
  modality = "T1",
  atlas = "Desikan",
  mtc = "BH",
  nidps = NA,
  filename = NA,
  verbose = FALSE,
  vignette = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="neuroimaGene_+3A_gene_list">gene_list</code></td>
<td>
<p>List of genes to work as inputs. There are no defaults.</p>
</td></tr>
<tr><td><code id="neuroimaGene_+3A_modality">modality</code></td>
<td>
<p>Neuroimaging modality for NIDP query. Defaults to T1 structural NIDPs.</p>
</td></tr>
<tr><td><code id="neuroimaGene_+3A_atlas">atlas</code></td>
<td>
<p>Neuroimaging parcellation atlas for NIDP query. Defaults to Desikan structural atlas.</p>
</td></tr>
<tr><td><code id="neuroimaGene_+3A_mtc">mtc</code></td>
<td>
<p>Statistical multiple testing correction for NIDP query. Defaults to BH for the Benjamini Hochberg False Discovery Rate.</p>
</td></tr>
<tr><td><code id="neuroimaGene_+3A_nidps">nidps</code></td>
<td>
<p>optional user defined vector of target NIDPs to query. These can be obtained from the listNIDPnames() function.</p>
</td></tr>
<tr><td><code id="neuroimaGene_+3A_filename">filename</code></td>
<td>
<p>optional user defined filename for writing neuroimaGene output to a file.</p>
</td></tr>
<tr><td><code id="neuroimaGene_+3A_verbose">verbose</code></td>
<td>
<p>print runtime messages to R console. Default to FALSE</p>
</td></tr>
<tr><td><code id="neuroimaGene_+3A_vignette">vignette</code></td>
<td>
<p>use for building vignette on installation. Default to FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a neuroimaGene object: data table with all significant associations between the user provided genes and the UKB NIDPs satisfying the multiple testing correction and atlas/modality/name filters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene_list &lt;- c('TRIM35', 'PROSER3', 'EXOSC6', 'PICK1', 'UPK1A', 'ESPNL', 'ZIC4')
ng &lt;- neuroimaGene(gene_list, atlas = NA, mtc = 'BH', vignette = TRUE)

</code></pre>

<hr>
<h2 id='ng_vignette'>NeuroimaGene object for vignette illustration</h2><span id='topic+ng_vignette'></span>

<h3>Description</h3>

<p>NeuroimaGene data table containing select associations used in the package vignette.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ng_vignette)
</code></pre>


<h3>Format</h3>

<p>A data.table with 6 columns and 3824 rows:
</p>

<dl>
<dt>gene</dt><dd><p>character: ENSEMBL Gene ID</p>
</dd>
<dt>gene_name</dt><dd><p>character: HUGO gene name</p>
</dd>
<dt>gwas_phenotype</dt><dd><p>character: neuroimaging derived phenotype</p>
</dd>
<dt>training_model</dt><dd><p>character: JTI derived tissue gene expression model</p>
</dd>
<dt>zscore</dt><dd><p>numeric: normalized effect size of GReX on NIDP morphology</p>
</dd>
<dt>mod_BHpval</dt><dd><p>character: Benjamini Hochberg corrected pvalue corrected by modality </p>
</dd>
</dl>



<h3>Value</h3>

<p>This script has no return. This is a documentation file for the
neuroimaGene data subset required to build the vignette.
</p>


<h3>Source</h3>

<p>Bledsoe, X. (2024) A transcriptomic atlas of the human brain reveals genetically determined aspects of neuropsychiatric health
</p>

<hr>
<h2 id='plot_gnNIDP'>Gene by NIDP summary plot of neuroimaGene object</h2><span id='topic+plot_gnNIDP'></span>

<h3>Description</h3>

<p>Generate overview plot of the neuroimagene object according to gene/NIDP pair
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gnNIDP(ng_obj, maxNidps = 20, maxGns = 15, title = NA, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_gnNIDP_+3A_ng_obj">ng_obj</code></td>
<td>
<p>NeuroimaGene object</p>
</td></tr>
<tr><td><code id="plot_gnNIDP_+3A_maxnidps">maxNidps</code></td>
<td>
<p>maximum number of NIDPs to visualize. default=20</p>
</td></tr>
<tr><td><code id="plot_gnNIDP_+3A_maxgns">maxGns</code></td>
<td>
<p>maximum number of genes to visualize. default=15</p>
</td></tr>
<tr><td><code id="plot_gnNIDP_+3A_title">title</code></td>
<td>
<p>optional title tag for the plot</p>
</td></tr>
<tr><td><code id="plot_gnNIDP_+3A_verbose">verbose</code></td>
<td>
<p>print runtime messages to R console. Default to FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot class heatmap showing tissue models per NIDP/Gene pair
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene_list &lt;- c('TRIM35', 'PROSER3', 'EXOSC6', 'PICK1', 'UPK1A', 'ESPNL', 'ZIC4')
ng &lt;- neuroimaGene(gene_list, atlas = NA, mtc = 'BH', vignette = TRUE)
plot_gnNIDP(ng)

</code></pre>

<hr>
<h2 id='plot_gns'>Gene summary plot of neuroimaGene object</h2><span id='topic+plot_gns'></span>

<h3>Description</h3>

<p>Generates an overview plot of the neuroimaGene object according to each gene input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_gns(ng_obj, maxGns = 15, title = NA, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_gns_+3A_ng_obj">ng_obj</code></td>
<td>
<p>NeuroimaGene Object</p>
</td></tr>
<tr><td><code id="plot_gns_+3A_maxgns">maxGns</code></td>
<td>
<p>maximum number of genes to visualize. default=15</p>
</td></tr>
<tr><td><code id="plot_gns_+3A_title">title</code></td>
<td>
<p>optional title tag for the plot</p>
</td></tr>
<tr><td><code id="plot_gns_+3A_verbose">verbose</code></td>
<td>
<p>print runtime messages to R console. Default to FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot class plot detailing NIDPs per gene, colored by brain measure type
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene_list &lt;- c('TRIM35', 'PROSER3', 'EXOSC6', 'PICK1', 'UPK1A', 'ESPNL', 'ZIC4')
ng &lt;- neuroimaGene(gene_list, atlas = NA, mtc = 'BH', vignette = TRUE)
plot_gns(ng)

</code></pre>

<hr>
<h2 id='plot_nidps'>NIDP summary plot of NeuroimaGene object</h2><span id='topic+plot_nidps'></span>

<h3>Description</h3>

<p>Generate overview plot of the neuroimagene object according to nidps
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_nidps(ng_obj, maxNidps = 30, title = NA, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_nidps_+3A_ng_obj">ng_obj</code></td>
<td>
<p>NeuroimaGene Object</p>
</td></tr>
<tr><td><code id="plot_nidps_+3A_maxnidps">maxNidps</code></td>
<td>
<p>maximum number of NIDPs to visualize. default=30</p>
</td></tr>
<tr><td><code id="plot_nidps_+3A_title">title</code></td>
<td>
<p>optional title tag for the plot</p>
</td></tr>
<tr><td><code id="plot_nidps_+3A_verbose">verbose</code></td>
<td>
<p>print runtime messages to R console. Default to FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot class object detailing mean effect size magnitude per NIDP, colored by brain region
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene_list &lt;- c('TRIM35', 'PROSER3', 'EXOSC6', 'PICK1', 'UPK1A', 'ESPNL', 'ZIC4')
ng &lt;- neuroimaGene(gene_list, atlas = NA, mtc = 'BH', vignette = TRUE)
plot_nidps(ng)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
