<!DOCTYPE html><html><head><title>Help for package neutralitytestr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {neutralitytestr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#lsq_plot'><p>Plot cumulative distribution</p>
<code>lsq_plot</code> Plots the cumulative distribution of the data as well as the best fit linear model line.</a></li>
<li><a href='#neutralitytest'><p>Testing for neutrality on cancer sequencing data</p></a></li>
<li><a href='#neutralitytestr'><p><code>neutralitytestr</code> package</p></a></li>
<li><a href='#normalized_plot'><p>Plot normalized cumulative distribution</p>
<code>normalized_plot</code> Plots the (normalized) cumulative distribution of the data as well as the theoretical expectation from
a neutral evolutionary model.</a></li>
<li><a href='#plot_all'><p>Plot all plots in the package and make composite figure.</p>
<code>plot_all</code> Plots histogram, linear model best fit plot
and normalized plot and plot and makes composite figure.</a></li>
<li><a href='#vaf_histogram'><p>Plot VAF histogram</p>
<code>vaf_histogram</code> Plots a histogram of the variant allele frequencies.</a></li>
<li><a href='#VAFneutral'><p>Synthetic sequencing data generated from a evolutionary based cancer simulation.</p></a></li>
<li><a href='#VAFselection'><p>Synthetic sequencing data generated from a evolutionary based cancer simulation.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Test for a Neutral Evolutionary Model in Cancer Sequencing Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Package takes frequencies of mutations as reported by high throughput sequencing data from cancer and fits a theoretical neutral model of tumour evolution. Package outputs summary statistics and contains code for plotting the data and model fits. See Williams et al 2016 &lt;<a href="https://doi.org/10.1038%2Fng.3489">doi:10.1038/ng.3489</a>&gt; and Williams et al 2017 &lt;<a href="https://doi.org/10.1101%2F096305">doi:10.1101/096305</a>&gt; for further details of the method.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, ggplot2, scales, pracma, ggpmisc, cowplot</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/marcjwilliams1/neutralitytestr">https://github.com/marcjwilliams1/neutralitytestr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/marcjwilliams1/neutralitytestr/issues">https://github.com/marcjwilliams1/neutralitytestr/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-02-16 17:31:07 UTC; william1</td>
</tr>
<tr>
<td>Author:</td>
<td>Marc Williams [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Marc Williams &lt;marcjwilliams1@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-02-16 18:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='lsq_plot'>Plot cumulative distribution
<code>lsq_plot</code> Plots the cumulative distribution of the data as well as the best fit linear model line.</h2><span id='topic+lsq_plot'></span>

<h3>Description</h3>

<p>Plot cumulative distribution
<code>lsq_plot</code> Plots the cumulative distribution of the data as well as the best fit linear model line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lsq_plot(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lsq_plot_+3A_object">object</code></td>
<td>
<p>neutrality test object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lsq_plot(neutralitytest(VAFselection, fmin = 0.1, fmax = 0.25))
</code></pre>

<hr>
<h2 id='neutralitytest'>Testing for neutrality on cancer sequencing data</h2><span id='topic+neutralitytest'></span>

<h3>Description</h3>

<p><code>neutralitytest</code> returns a neutralitytest object which contains the result of
various test statistics to test for neutrality as described in Williams et al. Nature Genetics 2018.
WARNING: This package has been superseded by MOBSTER, see Caravagna et al. Nature Genetics 2020.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>neutralitytest(
  VAF,
  fmin = 0.1,
  fmax = 0.25,
  read_depth = NULL,
  rho = 0,
  cellularity = 1,
  ploidy = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="neutralitytest_+3A_vaf">VAF</code></td>
<td>
<p>Vector of variant allele frequencies (VAFs) from a deep sequencing experiment,
numbers should be between 0 and 1</p>
</td></tr>
<tr><td><code id="neutralitytest_+3A_fmin">fmin</code></td>
<td>
<p>Minimum VAF of integration range, default is 0.1</p>
</td></tr>
<tr><td><code id="neutralitytest_+3A_fmax">fmax</code></td>
<td>
<p>Maximum VAF of integration range, default is 0.25</p>
</td></tr>
<tr><td><code id="neutralitytest_+3A_read_depth">read_depth</code></td>
<td>
<p>Read depth of sample, if this is specified it will be used to calculate an approptiate integration range.
default is NULL in which case the default or inputted fmin and fmax will be used.</p>
</td></tr>
<tr><td><code id="neutralitytest_+3A_rho">rho</code></td>
<td>
<p>Overdispersion of sample if known, default is 0.0. Will be used to calculate integration range if read_depth != NULL</p>
</td></tr>
<tr><td><code id="neutralitytest_+3A_cellularity">cellularity</code></td>
<td>
<p>Cellularity of sample, default is 1.0. Will be used to calculate integration range if read_depth != NULL</p>
</td></tr>
<tr><td><code id="neutralitytest_+3A_ploidy">ploidy</code></td>
<td>
<p>Ploidy of the genome, default is 2. Ideally mutations should be filtered for this ploidy before running the test.
Will be used to calculate integration range if read_depth != NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>neutralitytest object which contains test statistics which tests
if the sequencing data is consistent a neutral evolutionary model.
Test statistics are area between theoretical and empirical curves, kolmogorov distance, mean distance and R^2 statistics
from linear model fit. Also returns an estimate of the mutation rate per tumour tumour doubling, the raw VAFs and
cumulative distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>neutralitytest(runif(100))
neutralitytest(VAFselection, fmin = 0.1, fmax = 0.25)
neutralitytest(VAFneutral, read_depth = 100.0, cellularity = 0.8)
</code></pre>

<hr>
<h2 id='neutralitytestr'><code>neutralitytestr</code> package</h2><span id='topic+neutralitytestr'></span>

<h3>Description</h3>

<p>Package to test a neutral evolutionary model on deep sequencing data.
</p>


<h3>Details</h3>

<p>See the README on
<a href="https://github.com/marcjwilliams1/neutralitytestr#readme">GitHub</a>
</p>

<hr>
<h2 id='normalized_plot'>Plot normalized cumulative distribution
<code>normalized_plot</code> Plots the (normalized) cumulative distribution of the data as well as the theoretical expectation from
a neutral evolutionary model.</h2><span id='topic+normalized_plot'></span>

<h3>Description</h3>

<p>Plot normalized cumulative distribution
<code>normalized_plot</code> Plots the (normalized) cumulative distribution of the data as well as the theoretical expectation from
a neutral evolutionary model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalized_plot(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalized_plot_+3A_object">object</code></td>
<td>
<p>neutrality test object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>normalized_plot(neutralitytest(VAFselection, fmin = 0.1, fmax = 0.25))
</code></pre>

<hr>
<h2 id='plot_all'>Plot all plots in the package and make composite figure.
<code>plot_all</code> Plots histogram, linear model best fit plot
and normalized plot and plot and makes composite figure.</h2><span id='topic+plot_all'></span>

<h3>Description</h3>

<p>Plot all plots in the package and make composite figure.
<code>plot_all</code> Plots histogram, linear model best fit plot
and normalized plot and plot and makes composite figure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_all(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_all_+3A_object">object</code></td>
<td>
<p>neutrality test object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_all(neutralitytest(VAFselection, fmin = 0.1, fmax = 0.25))
</code></pre>

<hr>
<h2 id='vaf_histogram'>Plot VAF histogram
<code>vaf_histogram</code> Plots a histogram of the variant allele frequencies.</h2><span id='topic+vaf_histogram'></span>

<h3>Description</h3>

<p>Plot VAF histogram
<code>vaf_histogram</code> Plots a histogram of the variant allele frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vaf_histogram(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vaf_histogram_+3A_object">object</code></td>
<td>
<p>neutrality test object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vaf_histogram(neutralitytest(VAFselection, fmin = 0.1, fmax = 0.25))
</code></pre>

<hr>
<h2 id='VAFneutral'>Synthetic sequencing data generated from a evolutionary based cancer simulation.</h2><span id='topic+VAFneutral'></span>

<h3>Description</h3>

<p>This data is generated from a neutral evolutionary model where all subclonal mutations are neutral.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VAFneutral
</code></pre>


<h3>Format</h3>

<p>A vector with variant allele frequencies (VAFs) ranging from 0 to 1
</p>


<h3>Source</h3>

<p>Generated using cancer sequencing simulation <a href="https://github.com/marcjwilliams1/CancerSeqSim.jl">https://github.com/marcjwilliams1/CancerSeqSim.jl</a>
</p>

<hr>
<h2 id='VAFselection'>Synthetic sequencing data generated from a evolutionary based cancer simulation.</h2><span id='topic+VAFselection'></span>

<h3>Description</h3>

<p>This data is generated from an evolutionary model where there is on subclonal population
and all other mutations are neutral passengers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VAFselection
</code></pre>


<h3>Format</h3>

<p>A vector with variant allele frequencies (VAFs) ranging from 0 to 1
</p>


<h3>Source</h3>

<p>Generated using cancer sequencing simulation <a href="https://github.com/marcjwilliams1/CancerSeqSim.jl">https://github.com/marcjwilliams1/CancerSeqSim.jl</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
