<!DOCTYPE html><html><head><title>Help for package NewmanOmics</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {NewmanOmics}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bankStat'><p>Newman Banked Statistic</p></a></li>
<li><a href='#fitMix3'><p>Compute FDR from Three-Component Beta Mixture</p></a></li>
<li><a href='#MixOf3Beta-class'><p>Class &quot;MixOf3Beta&quot;</p></a></li>
<li><a href='#newman-data'><p>Datasets to Illustrate the Newman Tests</p></a></li>
<li><a href='#NewmanPaired-class'><p>Class &quot;NewmanPaired&quot;</p></a></li>
<li><a href='#pairedStat'><p>Paired Newman Statistic</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Title:</td>
<td>Extending the Newman Studentized Range Statistic to
Transcriptomics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.10</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-05-09</td>
</tr>
<tr>
<td>Author:</td>
<td>Zachary Abrams, Greg Gershkowitz, Anoushka Joglekar, 
  Chao Liu, Kevin R. Coombes</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kevin R. Coombes &lt;krc@silicovore.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Extends the classical Newman studentized range statistic
  in various ways that can be applied to genome-scale transcriptomic
  or other expression data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (== 2.0)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, stats, graphics, grDevices, oompaBase</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://oompa.r-forge.r-project.org/">http://oompa.r-forge.r-project.org/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-05-09 21:00:56 UTC; KRC</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-05-18 13:20:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='bankStat'>Newman Banked Statistic</h2><span id='topic+bankStat'></span><span id='topic+createBank'></span>

<h3>Description</h3>

<p>The Newman Banked Statistic is used to compare an individual sample to a cohort of similar samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bankStat(bankObj, testSet, bankMatrix)
createBank(bankMatrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bankStat_+3A_bankobj">bankObj</code></td>
<td>
<p>Compressed representation of the cohort being compared to.</p>
</td></tr>
<tr><td><code id="bankStat_+3A_testset">testSet</code></td>
<td>
<p>Matrix containing data from one or more individual
samples to be compared to the bank.</p>
</td></tr>
<tr><td><code id="bankStat_+3A_bankmatrix">bankMatrix</code></td>
<td>
<p>Data for the bank of &quot;normal&quot; or &quot;untreated&quot;
or &quot;baseline&quot; control samples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two matrices: the <code>nu.statistics</code> and the <code>p.values</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GSE6631)
HN &lt;- as.matrix(log2(1 + GSE6631))
bankMatrix &lt;- HN[,seq(1, ncol(HN), 2)] # odd columns are normal
testSet &lt;- HN[, seq(2, 6, 2)]   # evn columns are tumor
bs &lt;- bankStat(testSet = testSet, bankMatrix = bankMatrix)
summary(bs$nu.statistics)
summary(bs$p.values)
</code></pre>

<hr>
<h2 id='fitMix3'>Compute FDR from Three-Component Beta Mixture</h2><span id='topic+fitMix3'></span><span id='topic+computeFDR'></span><span id='topic+computeCutoff'></span>

<h3>Description</h3>

<p>Provides functions to fit a beta-mixture model to a set of p-values
that has peaks at both zero and one, and to estimate false discovery
rates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitMix3(datavec, forever=100, epsilon=0.001, relative = 0.001, print.level=0)
computeFDR(object, alpha)
computeCutoff(object, fdr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitMix3_+3A_datavec">datavec</code></td>
<td>
<p>A numeric vector containing p-values.</p>
</td></tr>
<tr><td><code id="fitMix3_+3A_forever">forever</code></td>
<td>
<p>An integer; maximum number of iterations while fitting
the mixture model.</p>
</td></tr>
<tr><td><code id="fitMix3_+3A_epsilon">epsilon</code></td>
<td>
<p>A real number; change in the log likelihood that should
be used to terminate the model-fitting loop.</p>
</td></tr>
<tr><td><code id="fitMix3_+3A_relative">relative</code></td>
<td>
<p>A real number; change in the relative log likelihood
that should be used to terminate the model-fitting loop.</p>
</td></tr>
<tr><td><code id="fitMix3_+3A_print.level">print.level</code></td>
<td>
<p>An integer; how much detail should <code><a href="stats.html#topic+nlm">nlm</a></code>
print while fitting the model.</p>
</td></tr>
<tr><td><code id="fitMix3_+3A_object">object</code></td>
<td>
<p>An object of the <code><a href="#topic+MixOf3Beta">MixOf3Beta</a></code> class.</p>
</td></tr>
<tr><td><code id="fitMix3_+3A_alpha">alpha</code></td>
<td>
<p>A real number between 0 and 1; the cutoff on the nominal
p-value where the FDR should be computed.</p>
</td></tr>
<tr><td><code id="fitMix3_+3A_fdr">fdr</code></td>
<td>
<p>A real number beteen 0 and 1; the targeted FDR value.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We have observed empirically that the set of p-values obtained when
computing the Newman paired test statistic often has peaks both at zero
(representing genes of interest) and at one (representing &quot;boring&quot;
genes that change much less than expected). We attribute the latter
phenomenon to the fact that we use locally smoothed instead of
gene-by-gene estimates of the standard deviation; genes whose SD is
increased by the smoothing process contribute to the boring peak near
one.
</p>
<p>To estimate p-values in this context, we fit a three-component beta
mixture model, combining (1) a right-peaked distribution Beta(L,1),
(2) a left-peaked dfistribution Beta(1,M), and (3) a uniform
distribution. Specfically, we look for models of the form
</p>
<p style="text-align: center;"><code class="reqn">alpha*Beta(L,1) + beta*Beta(1, M) + gamma*Beta(1,1)</code>
</p>
<p>.
</p>
<p>Model-fitting uses an expectation-maximization (EM) algorithm. In
addition to the parameters <code>mle=c(L,M)</code> and <code>psi=c(alpha,
  beta, gamma)</code>, we introduce a matrix <code>Z</code> of latent variables that
indicate which distribution each point is likely to arise
form. <code>Z</code> has three columns (one for each mixture component) and
one row for each p-value; the entries in each row are nonegative and
sum to one. The M-step of the algorithm uses the <code><a href="stats.html#topic+nlm">nlm</a></code>
optimization function to compute the maximum-likelihood <code>mle</code>
values given <code>psi</code> and <code>Z</code>. The E-step first updates
<code>psi</code> from the <code>Z</code>-matrix, and then updates the values of
<code>Z</code> based on the current <code>mle</code>. 
</p>
<p>We are able to use the mixture distribution to compute the
relationship between a cutoff on the nominal p-values and the false
discovery rate (FDR). 
</p>


<h3>Value</h3>

<p>The model-fitting function, <code>fitMix3</code>, returns an object of the
<code><a href="#topic+MixOf3Beta">MixOf3Beta</a></code> class.
</p>
<p>The <code>computeFDR</code> function returns a real number in [0,1], the
false discovery rate assiociated with the nominal cutoff.
</p>
<p>The <code>computeCutoff</code> function returns a real number in [0,1], the
cutoff required to achieve the desired FDR.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(98765)
ds &lt;- c(rbeta(3000, 20, 1),
        rbeta(1000, 1, 40),
        runif(6000))
fit &lt;- fitMix3(ds)
computeFDR(fit, 0.01)
computeCutoff(fit, 0.01)
computeFDR(fit, 0.0016438)
computeCutoff(fit, 0.05)
computeFDR(fit, 0.00702114)
</code></pre>

<hr>
<h2 id='MixOf3Beta-class'>Class &quot;MixOf3Beta&quot;</h2><span id='topic+MixOf3Beta'></span><span id='topic+MixOf3Beta-class'></span><span id='topic+plot+2CMixOf3Beta+2Cmissing-method'></span><span id='topic+hist+2CMixOf3Beta-method'></span><span id='topic+image+2CMixOf3Beta-method'></span>

<h3>Description</h3>

<p>Represents the results of fitting a beta-mixture model to a set of
p-values that has peaks at both zero and one.
</p>


<h3>Details</h3>

<p>Given a set of p-values (or any data on the interval [0,1]) that has
peaks at both ends of the interval, we fit a three-componet mixture
model. One component is uniform, and represents the expected
distribution under the null hypothesis that nothing interesting is
happening anywhere. The second component has the distribution
Beta(1,M); this has a peak at zero and represents the features of
interest. The final component has the distribution Beta(L,1). In the
context of the Newman paired statistic, this represents genes or
features whose variabilirt is smaller than the locally smoothed
estimate of the standard deviation; we can think of these as
&quot;extra boring&quot;.
</p>


<h3>Creating Objects</h3>

<p>In practice, users will use the <code><a href="#topic+fitMix3">fitMix3</a></code> function to
construct an object of the <code>MixOf3Beta</code> class. Hand
construction is strongly discouraged.
</p>


<h3>Slots</h3>


<dl>
<dt><code>input</code>:</dt><dd><p>A numeric vector containing the input p-values.</p>
</dd>
<dt><code>mle</code>:</dt><dd><p> A numeric vactor of length 2 containing the bea
parameters <code>L</code> and M (in that order).</p>
</dd>
<dt><code>psi</code>:</dt><dd><p>A numeric vector of length three containing the
mixing parameters, in the order (right-peak component, left-peak
component, and uniform-component).</p>
</dd>
<dt><code>Z</code>:</dt><dd><p>A matrix of size N (number of features) by 3. This
contains the latent indicator matrix. Each row corresponds to a
gene or feature, and the entries show the proabbiltiy that the
feature arose from the right, left, or uniform comnponent.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>plot(x, y, ...)</dt><dd><p>Plot the decompositon of the data into thre pieces.</p>
</dd>
<dt>hist(x, lcol = &quot;red&quot;, breaks=101, ...)</dt><dd><p>Plot a histogram of
the p-values along with the fitted model of the distribution.</p>
</dd>
<dt>image(x)</dt><dd><p>Plot a (sorted) image of the latent variable Z-matrix.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>
</p>


<h3>References</h3>

<p>Abrams ZB, Joglekar A, Gershkowitz GR, Sinicropi-yao S, Asiaee A,
Carbone DP, Coombes KR. Personalized Transcriptomics: Selecting Drugs
Based on Gene Expression Profiles. Preprint.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairedStat">pairedStat</a></code>, <code><a href="#topic+NewmanPaired-class">NewmanPaired-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(98765)
ds &lt;- c(rbeta(3000, 20, 1),
        rbeta(1000, 1, 40),
        runif(6000))
fit &lt;- fitMix3(ds)
image(fit, col=topo.colors(64))
hist(fit, col="skyblue", lcol="blue")
plot(fit)
</code></pre>

<hr>
<h2 id='newman-data'>Datasets to Illustrate the Newman Tests</h2><span id='topic+newman-data'></span><span id='topic+LungPair'></span><span id='topic+GSE6631'></span>

<h3>Description</h3>

<p>These data sets contain paired normal and tumor samples usewd to
ilustrate te Newman pairted test and the Newman bank test. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LungPair)
data(GSE6631)
</code></pre>


<h3>Format</h3>

<p><code>LungPair</code> is a data matrix containing normalized second
generation sequencing data from The Cancer Genome Atlas (TCGA), with
20,531 row (genes) and 2 columns (samples). The first column contains
data for the normal sample and the second column contains data for the
tumor sample from the patient with barcode TCGA.38.4625.
</p>
<p><code>GSE6631</code> is a data matrix containing normalized Affymetrix
microarray data from paired head-and-neck cancer samples in the Gene
Expression Omnibus set GSE6631. The matrix contains 200 rows (a rando
subset of genes) and 44 columns (samples). The odd numbered colukmns
are derived from normal mucosa; te even numbered columns are derived
from paired tumor samples from the same patient.
</p>


<h3>Source</h3>

<p>The full squamous cell lung cancer (LUSC) data from TCGA was
downloaded from   <a href="http://firebrowse.org/">http://firebrowse.org/</a>, and the data for this
pair were separated and saved as a binary R data file. The
head-and-neck cancer data were downloaded from the Gene Expression
Omnibus at 
<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6631">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6631</a>. A
subset of 2000 genes was randomly selected before saving the binary R
data file.
</p>


<h3>References</h3>

<p>Kuriakose MA, Chen WT, He ZM, Sikora AG et al. Selection and
validation of differentially expressed genes in head and neck
cancer. Cell Mol Life Sci 2004 Jun;61(11):1372-83. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LungPair)
data(GSE6631)
</code></pre>

<hr>
<h2 id='NewmanPaired-class'>Class &quot;NewmanPaired&quot;</h2><span id='topic+NewmanPaired'></span><span id='topic+NewmanPaired-class'></span><span id='topic+plot+2CNewmanPaired+2Cmissing-method'></span><span id='topic+hist+2CNewmanPaired-method'></span><span id='topic+dim+2CNewmanPaired-method'></span><span id='topic++5B+2CNewmanPaired-method'></span>

<h3>Description</h3>

<p>Represents the reusults of computing the Newman Paired test statistic
on one or more paired samples.
</p>


<h3>Creating Objects</h3>

<p>In practice, users will use the <code><a href="#topic+pairedStat">pairedStat</a></code> function to
construct an object of the <code>NewmanPaired</code> class. Hand
construction is strongly discouraged.
</p>


<h3>Slots</h3>


<dl>
<dt><code>pairedMean</code>:</dt><dd><p>A matrix of size N (number of features) by S (number
of sample pairs). The mean expression of each feature in each paired
sample. Also called &quot;A&quot; in the M-versus-A plots of the microarray era.</p>
</dd>
<dt><code>difference</code>:</dt><dd><p>A matrix of size N (number of features) by S (number
of sample pairs). The difference (perturbed - base) in expression of
each feature in each paired sample. Also called &quot;M&quot; in the M-versus-A
plots of the microarray era.</p>
</dd>
<dt><code>smoothSD</code>:</dt><dd><p>A matrix of size N (number of features) by S (number
of sample pairs). The results of fitting a loess smooth to the
relationship between the <code>PairedMean</code> and the observed estimate
of standard deviation (i.e., <code>abs(difference)/sqrt(2)</code>).</p>
</dd>
<dt><code>nuStatistics</code>:</dt><dd><p>A matrix of size N (number of features) by S (number
of sample pairs). The Newman paired statistics, nu.</p>
</dd>
<dt><code>pValues</code>:</dt><dd><p>A matrix of size N (number of features) by S (number
of sample pairs). Empirical p-values for the Newman statistics.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>x[i,j]</dt><dd><p>Select a subset of features or sample pairs.</p>
</dd>
<dt>dim(x)</dt><dd><p>The dimension, N by S, of the object.</p>
</dd>
<dt>plot(x, y, which = NULL, ask = NULL, high = 0.99, low = 0.01,
...)</dt><dd><p>Plot the results of the analysis of one sample pair.</p>
</dd>
<dt>hist(x, breaks=101, xlab=&quot;P-value&quot;, ...)</dt><dd><p>Plot a histogram of the p-values for one sample-pair.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kevin R. Coombes <a href="mailto:krc@silicovore.com">krc@silicovore.com</a>
</p>


<h3>References</h3>

<p>Abrams ZB, Joglekar A, Gershkowitz GR, Sinicropi-yao S, Asiaee A,
Carbone DP, Coombes KR. Personalized Transcriptomics: Selecting Drugs
Based on Gene Expression Profiles. Preprint.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pairedStat">pairedStat</a></code>, <code><a href="#topic+bankStat">bankStat</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>showClass("NewmanPaired")
</code></pre>

<hr>
<h2 id='pairedStat'>Paired Newman Statistic</h2><span id='topic+pairedStat'></span>

<h3>Description</h3>

<p>The Paired Newman Statistic is used for one-to-one comparison of
paired individual samples. Commonly used to find differential expression
between tumor-normal pairs or before-after treatment pairs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairedStat(baseData, perturbedData = NULL, pairing = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairedStat_+3A_basedata">baseData</code></td>
<td>
<p>Either a list or a matrix. May contain data for just the
base condition (for example, normal samples or samples before
treatment) or for both the base condition and the perturbed condition
(for example, tumor samples or samples after treatment). See details.</p>
</td></tr>
<tr><td><code id="pairedStat_+3A_perturbeddata">perturbedData</code></td>
<td>
<p>An optional matrix containing data for the &quot;perturbed&quot;
samples. May be NULL if the <code>baseData</code> argument is a list or a
matrix containing all the data.</p>
</td></tr>
<tr><td><code id="pairedStat_+3A_pairing">pairing</code></td>
<td>
<p>An optional vector indicating the pairing between base
and perturbed samples. Entries must be integers. Positive integers
indicate perturbed samples and negative integers with the same
absolute value indicate the paired base samples. See details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the simplest case, we have gene expression data on one &quot;base&quot;
sample and one &quot;perturbed&quot; sample, and the goal is to identify genes
whose expression changes between the two states. Our primary
assumption is that the standard deviation (SD) of gene expression
varies as a smooth function of the mean; fitting such a curve allows
us to detect individual genes whose difference is large compared to
the smoothed SD.
</p>
<p>Note that this assumption is most useful on the log-transformed
scale  (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4120293/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4120293/</a>).
If your data is on a raw scale, then we recommend transforming
it before computing the Newman paired statistic. 
</p>
<p>The input arguments to the <code>pairedStats</code> function are moderately
complicated in order to allow users to choose a convenient method for
supplying data when they have multiple paired samples. The first
posssibility is to have all the base samples in one matrix and all the
perturbed samples in a second matrix. In this case, we assume (without
checking) that the columns in the two matrices correspond to the
paired samples, and that the genes-rows are in the same order.
</p>
<p>The second possibility is to put the data for both the base samples
and the perturbed samples in the same matrix. In this case, the user
must supply a <code>pairing</code> vector to explain how the samples should
be matched. If the column order is (&quot;base1&quot;, &quot;perturbed1&quot;, &quot;base2&quot;,
&quot;perturbed2&quot;, ...), then the pairiing vector should be written as
<code>c(-1, 1, -2, 2, -3, 3, ...)</code>.
</p>
<p>The third possibility is to provide the paired samples in a list,
each of whose entries is a matrix with two columns,with the first
column being the base state and the second column being the
perturbed state.
</p>
<p>This flexibility means that there are three equivalent ways to input
the data even if you have only one base sample (with data in the
one-column matrix B) and one perturbed sample (with data in the
one-column matrix P). If we let <code>BP &lt;- cbind(B, P) </code>, then we can
choose (1) <code>pairedStats(B, P)</code>, or (2)
<code>pairedStats(list(BP))</code>, or (3)  <code>pairedStats(BP,
  pairing = c(-1,1))</code>. 
</p>


<h3>Value</h3>

<p>A list containing two marices: the <code>nu.statistics</code> and the <code>p.values</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GSE6631)
Normal &lt;- GSE6631[, c(1,3)]
Tumor &lt;- GSE6631[, c(2,4)]

### input two separate matrices
ps1 &lt;- pairedStat(Normal, Tumor)
summary(ps1@nu.statistics)
summary(ps1@p.values)

### input one combined matrix and a pairing vector
ps2 &lt;- pairedStat(GSE6631, pairing=c(-1, 1, -2, 2))
summary(ps2@nu.statistics)
summary(ps2@p.values)

### input a list of matrix-pairs
ps3 &lt;- pairedStat(list(One = GSE6631[, 1:2],
                       Two = GSE6631[, 3:4]))
summary(ps3@nu.statistics)
summary(ps3@p.values)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
