<!DOCTYPE html><html><head><title>Help for package NIPTeR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {NIPTeR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_samples_controlgroup'><p>Add a sample to an existing control group</p></a></li>
<li><a href='#as_control_group'><p>Convert list of nipt samples to nipt control group</p></a></li>
<li><a href='#bin_bam_sample'><p>Load and bin BAM file</p></a></li>
<li><a href='#calculate_ncv_score'><p>Use an NCV template to calculate a NCV score for sample of interest</p></a></li>
<li><a href='#calculate_z_score'><p>Calculate 'standard' Z-score</p></a></li>
<li><a href='#chi_correct'><p>Performs chi-square based variation reduction</p></a></li>
<li><a href='#chrfractions'><p>Calculate chromosomal fraction</p></a></li>
<li><a href='#diagnose_control_group'><p>Diagnose control group</p></a></li>
<li><a href='#gc_correct'><p>Perform a GC bias correction on nipt sample</p></a></li>
<li><a href='#getcontrolchromosomes'><p>Get control chromosomes names</p></a></li>
<li><a href='#getfractionscontrolgroup'><p>Get all chromosomal fractions of a control group</p></a></li>
<li><a href='#getreadscontrolgroup'><p>Get reads per chromosome per control group sample</p></a></li>
<li><a href='#match_control_group'><p>Best matching control group by least sum of squares</p></a></li>
<li><a href='#perform_regression'><p>Regression based Z score</p></a></li>
<li><a href='#prepare_ncv'><p>Prepare NCV calculation</p></a></li>
<li><a href='#remove_duplicates_controlgroup'><p>Remove duplicate samples from control group</p></a></li>
<li><a href='#remove_sample_controlgroup'><p>Remove a sample by samplename from control group</p></a></li>
<li><a href='#retrieve_fractions_of_interest'><p>Retrieve the chromosomal fractions of a chromosome of interest</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Title:</td>
<td>Fast and Accurate Trisomy Prediction in Non-Invasive Prenatal
Testing</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-03-09</td>
</tr>
<tr>
<td>Author:</td>
<td>Dirk de Weerd, Lennart Johansson</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lennart Johansson &lt;l.johansson@umcg.nl&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Fast and Accurate Trisomy Prediction in Non-Invasive Prenatal
    Testing.</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, Rsamtools, sets, S4Vectors</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0),</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-2.1">LGPL-2.1</a> | <a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a> [expanded from: GNU Lesser General Public License]</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, pander</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-03-09 14:02:38 UTC; ljohansson</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-03-09 18:55:22</td>
</tr>
</table>
<hr>
<h2 id='add_samples_controlgroup'>Add a sample to an existing control group</h2><span id='topic+add_samples_controlgroup'></span>

<h3>Description</h3>

<p>This functions adds NIPTSample objects to an existing control group and returns a new 
NIPTControlGroup object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_samples_controlgroup(nipt_control_group, samples_to_add)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_samples_controlgroup_+3A_nipt_control_group">nipt_control_group</code></td>
<td>
<p>The NIPTControlGroup to add the samples to</p>
</td></tr>
<tr><td><code id="add_samples_controlgroup_+3A_samples_to_add">samples_to_add</code></td>
<td>
<p>A list with sample(s) to add. This always needs to be a list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NIPTControlGroup object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
##First bin the new sample
new_binned_sample &lt;- bin_bam_sample(bam_filepath = "/path/to/file.bam", 
                                    separate_strands = T)

##Then add the sample to the control group
new_control_group &lt;- add_samples_controlgroup(nipt_control_group = my_control_group, 
                                              samples_to_add = new_binned_sample)

## End(Not run)

</code></pre>

<hr>
<h2 id='as_control_group'>Convert list of nipt samples to nipt control group</h2><span id='topic+as_control_group'></span>

<h3>Description</h3>

<p>Convert list of nipt samples to nipt control group
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_control_group(nipt_samples, control_group_type = generic_control_group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_control_group_+3A_nipt_samples">nipt_samples</code></td>
<td>
<p>List of nipt_sample objects to be combined to a control
group</p>
</td></tr>
<tr><td><code id="as_control_group_+3A_control_group_type">control_group_type</code></td>
<td>
<p>Control group type, either 'generic control group' 
or 'fitted to sample'. Leave this argument blank</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns an S3 object of class nipt_control_group. It is a list 
with 3 items:
</p>

<ul>
<li><p> List <strong>Samples</strong> nipt_sample objects in the control group
</p>
</li>
<li><p> Character <strong>Correction_status</strong> Correction_status(es) in the 
control group
</p>
</li>
<li><p> Character <strong>Samplenames</strong> The sample names of samples present 
in the control group
</p>
</li></ul>

<p>Read count strategy should be uniform in all samples in a control group object;
meaning samples where forward and reverse reads are counted separately 
cannot be in the same control group object as samples where forward and reverse
reads are counted together.
</p>
<p>A control group object with duplicate samples or samples with different 
correction statusses is possible but not recommended and will generate a 
warning message.
</p>


<h3>Value</h3>

<p>NIPTControlGroup object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
##Retrieve filenames
bam_filepaths &lt;- list.files(path = "/Path/to/bamfiles/", pattern = ".bam", full.names = T)
##Load files and convert to control group
control_group  &lt;- as_control_group(nipt_samples = lapply(X = bam_filepaths, bin_bam_sample, 
                                                         do_sort = F, separate_strands = FALSE))
##Save control group for later
saveRDS(object = control_group, file = "/Path/to/directory/control_group.rds")

## End(Not run)
</code></pre>

<hr>
<h2 id='bin_bam_sample'>Load and bin BAM file</h2><span id='topic+bin_bam_sample'></span><span id='topic+nipt_sample'></span>

<h3>Description</h3>

<p>Load a BAM file and count reads in bins of size 50.000 base pairs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bin_bam_sample(bam_filepath, do_sort = FALSE, separate_strands = FALSE,
  custom_name = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bin_bam_sample_+3A_bam_filepath">bam_filepath</code></td>
<td>
<p>Character The location and filename on the file system where the bam file is stored</p>
</td></tr>
<tr><td><code id="bin_bam_sample_+3A_do_sort">do_sort</code></td>
<td>
<p>Boolean Sort the bam file? If the bam is unsorted set to true, 
but the use of pre-sorted bam files is recommended.</p>
</td></tr>
<tr><td><code id="bin_bam_sample_+3A_separate_strands">separate_strands</code></td>
<td>
<p>Boolean If set to true, reads from forward and reverse strands are counted and stored separately. 
This option should be used if you are planning on using regression, since this doubles the number of 
predictors (F+R) and distributes predictive power more equally over prediction sets since F and R 
strand from the same chromosome cannot be both in one prediction set.</p>
</td></tr>
<tr><td><code id="bin_bam_sample_+3A_custom_name">custom_name</code></td>
<td>
<p>String The name of sample. Default samplename is the filename of the bam 
file without the .bam suffix and filepath prefix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns an object of class NIPTSample, the main 'currency' of this package.
It is a list with 5 items:
</p>

<ul>
<li><p> List <strong>autosomal_chromosome_reads</strong> Autosomal reads are stored in
a matrix where the columns are the bins and rows (22) represent the autosomal
chromosomes. The length of this list is either 1 or 2, depending if the 
forward and reverse reads are counted separately.
</p>
</li>
<li><p> Character <strong>correction_status_autosomal_chromosomes</strong> The correction
status of the autosomal reads. The status can either be <em>Uncorrected</em> or
<em>GC Corrected</em> and/or <em>Chi Corrected</em>
</p>
</li>
<li><p> List <strong>sex_chromosome_reads</strong> Sex chromosome reads are stored in a 
similar matrix(es) as the autosomal chromosome reads, now with 2 (X and Y) rows.
</p>
</li>
<li><p> Character <strong>correction_status_autosomal_chromosomes</strong> The status can 
either be <em>Uncorrected</em> or <em>GC Corrected</em> and/or <em>Chi Corrected</em>.
</p>
</li>
<li><p> Character <strong>sample_name</strong> Sample name 
</p>
</li></ul>



<h3>Value</h3>

<p>Object NIPTSample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
##To process a single sample
binned_sample &lt;- bin_bam_sample(bam_filepath = "/path/to/file.bam", 
                                separate_strands = T)

##To create a control group out of a set of bam files
bam_filepaths &lt;- list.files(path = "/Path/to/bamfiles/", 
                            pattern = ".bam", full.names = T)

control_group  &lt;- as_control_group(nipt_samples = lapply(X = bam_filepaths, 
                                   bin_bam_sample, do_sort = F, 
                                   separate_strands = T))

## End(Not run)

</code></pre>

<hr>
<h2 id='calculate_ncv_score'>Use an NCV template to calculate a NCV score for sample of interest</h2><span id='topic+calculate_ncv_score'></span>

<h3>Description</h3>

<p>Use an NCV template to calculate a NCV score for sample of interest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_ncv_score(nipt_sample, ncv_template)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_ncv_score_+3A_nipt_sample">nipt_sample</code></td>
<td>
<p>nipt_sample object of interest</p>
</td></tr>
<tr><td><code id="calculate_ncv_score_+3A_ncv_template">ncv_template</code></td>
<td>
<p>ncv_template object, result from <code><a href="#topic+prepare_ncv">prepare_ncv</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+prepare_ncv">prepare_ncv</a></code>
</p>


<h3>Value</h3>

<p>ncv_result object
</p>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/21519036">Sehnert et al.</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
##Use NCVTemplate to get NCV scores for the sample of interest
ncv_score_13 &lt;- calculate_ncv_score(nipt_sample = sample_of_interest, 
                                    ncv_template = new_ncv_template_13)

## End(Not run)

</code></pre>

<hr>
<h2 id='calculate_z_score'>Calculate 'standard' Z-score</h2><span id='topic+calculate_z_score'></span>

<h3>Description</h3>

<p>Calculate 'standard' Z-score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_z_score(nipt_sample, nipt_control_group, chromo_focus)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_z_score_+3A_nipt_sample">nipt_sample</code></td>
<td>
<p>The NIPTSample object that is the focus of the analysis</p>
</td></tr>
<tr><td><code id="calculate_z_score_+3A_nipt_control_group">nipt_control_group</code></td>
<td>
<p>The NIPTControlGroup object used in the analysis</p>
</td></tr>
<tr><td><code id="calculate_z_score_+3A_chromo_focus">chromo_focus</code></td>
<td>
<p>The chromosome of interest. Most commonly chromosome 13, 18 or 21. 
However, every autosomal chromosome can be predicted</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the Z-score approach, introduced by Chiu et al in 2008, 
the chromosomal fraction of interest of a sample is compared 
to the chromosomal fractions of interest of the reference samples, 
the 'NIPTControlGroup' object.
The output of the function is an object of class 'ZscoreResult'. It is a named list containing seven fields:
</p>

<ul>
<li><p> numeric <strong>sample_Zscore</strong> The Z score for the sample of interest for the sample of interest
</p>
</li>
<li><p> named num <strong>control_group_statistics</strong> Named num of length 3, the first field being the 
mean (name mean), the second field is the standard deviation (name SD) 
and the third field is the P value of the Shapiro-Wilk test (name Shapiro_P_value)
</p>
</li>
<li><p> matrix <strong>control_group_Zscores</strong> containing the Z scores of the chromosome of interest for 
all used control samples
</p>
</li>
<li><p> integer <strong>focus_chromosome</strong> The chromosome of interest. Most commonly chromosome 13, 18 or 21.
However, every autosomal chromosome can be predicted
</p>
</li>
<li><p> string <strong>control_group_sample_names</strong> The sample names of all control group samples used in the
analysis
</p>
</li>
<li><p> string <strong>correction_status</strong> The correction status of the control group
</p>
</li>
<li><p> string <strong>sample_name</strong> The sample_name of the sample of interest
</p>
</li></ul>



<h3>Value</h3>

<p>ZscoreResult object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
z_score_result_13 &lt;- calculate_z_score(nipt_sample = sample_of_interest, 
                                       nipt_control_group = control_group, 
                                       chromo_focus = 13)

## End(Not run)

</code></pre>

<hr>
<h2 id='chi_correct'>Performs chi-square based variation reduction</h2><span id='topic+chi_correct'></span>

<h3>Description</h3>

<p>Performs chi-square based variation reduction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chi_correct(nipt_sample, nipt_control_group, chi_cutoff = 3.5,
  include_XY = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chi_correct_+3A_nipt_sample">nipt_sample</code></td>
<td>
<p>The NIPTSample object that is the focus of the analysis</p>
</td></tr>
<tr><td><code id="chi_correct_+3A_nipt_control_group">nipt_control_group</code></td>
<td>
<p>The NIPTControlGroup object used in the analysis</p>
</td></tr>
<tr><td><code id="chi_correct_+3A_chi_cutoff">chi_cutoff</code></td>
<td>
<p>The Z-score cutoff. If a bin has a Z-score above this threshold,
it will be corrected</p>
</td></tr>
<tr><td><code id="chi_correct_+3A_include_xy">include_XY</code></td>
<td>
<p>Also apply correction to X and Y chromosomes?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The chi-squared based variation reduction identifies overdispersed bins within 
the control group and corrects these bins in both the sample of interest and the control group. 
The function takes in a 'NIPTSample' and a 'NIPTControlGroup' object, both to be corrected. 
For every corresponding bin in the control group a chi-squared score is calculated and this 
total score is converted to a normal distribution. Corresponding bins with a normalized score 
above _chi_cutoff_ (default 3.5) are corrected by dividing the number of reads by the total 
chi-squared score divided by degrees of freedom
</p>


<h3>Value</h3>

<p>Named list of length 2. The corrected nipt_sample is in index 1 and the 
corrected control group in index 2
to extract the corrected sample use <code>$sample</code> or <code>[[1]]</code>. 
To extract the control group from the list use
<code>$control_group</code> or <code>[[2]]</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
##Apply chi-squared based variation reduction method
chi_corrected_data &lt;- chicorrect(nipt_sample = gc_LOESS_corrected_sample, 
                                 nipt_control_group = subset_loess_corrected_control_group)
##Extract sample and control group
loess_chi_corrected_sample &lt;- chi_corrected_data$sample
subset_loess_chi_corrected_control_group &lt;- chi_corrected_data$control_group

## End(Not run)

</code></pre>

<hr>
<h2 id='chrfractions'>Calculate chromosomal fraction</h2><span id='topic+chrfractions'></span>

<h3>Description</h3>

<p>Calculate chromosomal fraction
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chrfractions(nipt_sample)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chrfractions_+3A_nipt_sample">nipt_sample</code></td>
<td>
<p>NIPTSample to retrieve chromosomal fraction for</p>
</td></tr>
</table>

<hr>
<h2 id='diagnose_control_group'>Diagnose control group</h2><span id='topic+diagnose_control_group'></span>

<h3>Description</h3>

<p>Compute a regular Z-score for every chromosome of every sample in a NIPTControlGroup object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diagnose_control_group(nipt_control_group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diagnose_control_group_+3A_nipt_control_group">nipt_control_group</code></td>
<td>
<p>The NIPTControlGroup object to diagnose</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes a regular Z-score for every chromosome of
every sample in a NIPTControlGroup object. It returns a named list with 
diagnostics information.
</p>
<p>The function returns a named list with 3 fields:
</p>

<ul>
<li> <p><strong>Z_scores</strong> A matrix containing Z-scores for every sample and every chromosome
</p>
</li>
<li> <p><strong>abberant_scores</strong> Dataframe with samplename and chromosome of Z-scores outside -3  3 range 
</p>
</li>
<li> <p><strong>control_group_statistics</strong> Matrix with mean, standard deviation and P value of Shapiro-Wilk test
</p>
</li></ul>



<h3>Value</h3>

<p>named list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
diagnose_control_group(nipt_control_group = control_group)

## End(Not run)

</code></pre>

<hr>
<h2 id='gc_correct'>Perform a GC bias correction on nipt sample</h2><span id='topic+gc_correct'></span>

<h3>Description</h3>

<p>LOESS based GC bias correction algorithm described by Chen et al (2011)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gc_correct(nipt_object, method = "LOESS", include_XY = F, span = 0.75,
  ref_genome = "hg37")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gc_correct_+3A_nipt_object">nipt_object</code></td>
<td>
<p>The object that will be corrected. This can either be a 
'NIPTSample' or a 'NIPTControlGroup' object</p>
</td></tr>
<tr><td><code id="gc_correct_+3A_method">method</code></td>
<td>
<p>To select the LOESS based method use  <em>&quot;LOESS&quot;</em>,
to select the bin weights based method use  <em>&quot;bin&quot;</em>.</p>
</td></tr>
<tr><td><code id="gc_correct_+3A_include_xy">include_XY</code></td>
<td>
<p>Also apply correction to X and Y chromosomes?</p>
</td></tr>
<tr><td><code id="gc_correct_+3A_span">span</code></td>
<td>
<p>The span for the LOESS fit. Only applicable when LOESS method is used.</p>
</td></tr>
<tr><td><code id="gc_correct_+3A_ref_genome">ref_genome</code></td>
<td>
<p>The reference genome used. Either <em>&quot;hg37&quot;</em> or <em>&quot;hg38&quot;</em>
default = <em>&quot;hg37&quot;</em></p>
</td></tr>
</table>


<h3>Details</h3>

<p>GC content bias is the correlation between the number of reads mapped to a specific genomic 
region and the GC content of this region. In NIPTeR, two GC bias correction algorithms 
have been implemented, the LOESS based method introduced by Chen et al. (2011) and the bin 
weight based method described by Fan and Quake (2010).
</p>


<h3>Value</h3>

<p>Depending on the input object either a NIPTSample or a NIPTControlGroup object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
##Correct NIPTSample object using LOESS method
loess_corrected_sample &lt;- gc_correct(nipt_object = sample_of_interest, method = "LOESS",
                                     include_XY = F, span = 0.75)
##Correct NIPTControlGroup object using bin method
gc_bin_corrected_control_group &lt;- gc_correct(nipt_object = control_group, method = "bin", 
                                             include_XY = T)

## End(Not run)

</code></pre>

<hr>
<h2 id='getcontrolchromosomes'>Get control chromosomes names</h2><span id='topic+getcontrolchromosomes'></span>

<h3>Description</h3>

<p>Get control chromosomes names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getcontrolchromosomes(nipt_sample, control_chromosomes = control_chromosomes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getcontrolchromosomes_+3A_nipt_sample">nipt_sample</code></td>
<td>
<p>A sample to check wether combined or separated strands 
are used</p>
</td></tr>
<tr><td><code id="getcontrolchromosomes_+3A_control_chromosomes">control_chromosomes</code></td>
<td>
<p>Vector with control chromosomes</p>
</td></tr>
</table>

<hr>
<h2 id='getfractionscontrolgroup'>Get all chromosomal fractions of a control group</h2><span id='topic+getfractionscontrolgroup'></span>

<h3>Description</h3>

<p>Get all chromosomal fractions of a control group
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getfractionscontrolgroup(nipt_control_group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getfractionscontrolgroup_+3A_nipt_control_group">nipt_control_group</code></td>
<td>
<p>The NIPTControlGroup to retrieve the 
chromosomal fraction of every autosome for</p>
</td></tr>
</table>

<hr>
<h2 id='getreadscontrolgroup'>Get reads per chromosome per control group sample</h2><span id='topic+getreadscontrolgroup'></span>

<h3>Description</h3>

<p>Get reads per chromosome per control group sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getreadscontrolgroup(nipt_control_group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getreadscontrolgroup_+3A_nipt_control_group">nipt_control_group</code></td>
<td>
<p>The control group to retrieve reads for</p>
</td></tr>
</table>

<hr>
<h2 id='match_control_group'>Best matching control group by least sum of squares</h2><span id='topic+match_control_group'></span>

<h3>Description</h3>

<p>The matchcontrolgroup function determines how well an NIPTSample fits within the NIPTControlGroup
</p>


<h3>Usage</h3>

<pre><code class='language-R'>match_control_group(nipt_sample, nipt_control_group, mode, n_of_samples,
  include_chromosomes = NULL, exclude_chromosomes = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="match_control_group_+3A_nipt_sample">nipt_sample</code></td>
<td>
<p>The NIPTSample object that is the focus of the analysis</p>
</td></tr>
<tr><td><code id="match_control_group_+3A_nipt_control_group">nipt_control_group</code></td>
<td>
<p>The NIPTControlGroup object used in the analysis</p>
</td></tr>
<tr><td><code id="match_control_group_+3A_mode">mode</code></td>
<td>
<p>The function mode. This can either be <em>&quot;subset&quot;</em> or <em>&quot;report&quot;</em>.
Mode <em>&quot;subset&quot;</em> means the return value will be a new 'NIPTControlGroup' object containing  <em>n</em>
samples. When mode <em>&quot;report&quot;</em> is used the output is a matrix containing the sum of squares score 
of the differences between the chromosomal fractions of the sample and the control for every control 
sample, sorted in increasing score.</p>
</td></tr>
<tr><td><code id="match_control_group_+3A_n_of_samples">n_of_samples</code></td>
<td>
<p>The length of the resulting NIPTControlGroup. Only applicable if mode <em>&quot;subset&quot;</em> is used.</p>
</td></tr>
<tr><td><code id="match_control_group_+3A_include_chromosomes">include_chromosomes</code></td>
<td>
<p>integer. Include potential trisomic chromosomes into the comparison? Default = NULL,
meaning chromosomes 13, 18 and 21 are not included</p>
</td></tr>
<tr><td><code id="match_control_group_+3A_exclude_chromosomes">exclude_chromosomes</code></td>
<td>
<p>integer.Exclude other autosomal chromosomes besides chromosomes 
13, 18 and 21? Default = NULL</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 'matchcontrolgroup' function determines how well an NIPTSample fits within the NIPTControlGroup
and, if needed, makes a subset 'NIPTControlGroup' of length <em>n</em>.
</p>


<h3>Value</h3>

<p>The output for mode <em>subset</em> is a new 'NIPTControlGroup' composed of _n_ samples. 
The output for mode <em>report</em> is a matrix with a single column containing the sum of squares
in ascending order.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
##Mode report
scores_control_group &lt;- matchcontrolgroup(nipt_sample = sample_of_interest, 
                                          nipt_control_group = control_group, 
                                          mode = "report", include_chromosomes = c(13,18))

##Mode subset
subset_control_group &lt;- matchcontrolgroup(nipt_sample = sample_of_interest, 
                                          nipt_control_group = control_group, 
                                          mode = "subset", n_of_samples = 50)

## End(Not run) 


</code></pre>

<hr>
<h2 id='perform_regression'>Regression based Z score</h2><span id='topic+perform_regression'></span>

<h3>Description</h3>

<p>Make multiple models using linear regression and calculate Z-score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perform_regression(nipt_sample, nipt_control_group, chromo_focus,
  n_models = 4, n_predictors = 4, exclude_chromosomes = NULL,
  include_chromosomes = NULL, use_test_train_set = T,
  size_of_train_set = 0.6, overdispersion_rate = 1.15,
  force_practical_cv = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perform_regression_+3A_nipt_sample">nipt_sample</code></td>
<td>
<p>The NIPTSample object that is the focus of the analysis</p>
</td></tr>
<tr><td><code id="perform_regression_+3A_nipt_control_group">nipt_control_group</code></td>
<td>
<p>The NIPTControlGroup object used in the analysis</p>
</td></tr>
<tr><td><code id="perform_regression_+3A_chromo_focus">chromo_focus</code></td>
<td>
<p>The chromosome of interest. Most commonly chromosome 13, 18 or 21.
However, every autosomal chromosome can be predicted</p>
</td></tr>
<tr><td><code id="perform_regression_+3A_n_models">n_models</code></td>
<td>
<p>Integer Number of linear models to be made. Default setting is 4 models</p>
</td></tr>
<tr><td><code id="perform_regression_+3A_n_predictors">n_predictors</code></td>
<td>
<p>Integer The number of predictors each model contains. Default is 4</p>
</td></tr>
<tr><td><code id="perform_regression_+3A_exclude_chromosomes">exclude_chromosomes</code></td>
<td>
<p>integer. Exclude which autosomal chromosomes as potential predictors? 
Default potential trisomic chromosomes 13, 18 and 21 are exluded.</p>
</td></tr>
<tr><td><code id="perform_regression_+3A_include_chromosomes">include_chromosomes</code></td>
<td>
<p>integer. Include potential trisomic chromosomes? 
Options are: chromosomes 13, 18 and 21</p>
</td></tr>
<tr><td><code id="perform_regression_+3A_use_test_train_set">use_test_train_set</code></td>
<td>
<p>Use a test and train set to build the models? Default is TRUE</p>
</td></tr>
<tr><td><code id="perform_regression_+3A_size_of_train_set">size_of_train_set</code></td>
<td>
<p>The size of the train set expressed in a decimal. 
Default is 0.6 (60 of the control samples)</p>
</td></tr>
<tr><td><code id="perform_regression_+3A_overdispersion_rate">overdispersion_rate</code></td>
<td>
<p>The standard error of the mean is multiplied by this factor</p>
</td></tr>
<tr><td><code id="perform_regression_+3A_force_practical_cv">force_practical_cv</code></td>
<td>
<p>Boolean, Ignore the theoretical CV and always use the practical CV?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The regression based Z-score builds <em>n</em> models with <em>m</em> predictors using stepwise regression 
with forward selection. The models are used to predict the chromosomal fraction of interest, for the 
sample and for the control group. The observed fractions are then divided by the expected fraction, 
and Z-scores are calculated over these ratios. The Z-score is calculated by subtracting one from the 
ratio of the sample and dividing this result by the coefficient of variation. 
The coefficient of variation (CV) can either be the Practical or Theoretical CV. 
The Theoretical CV is the standard error multiplied by the overdispersion. 
Theoretically, the CV cannot be lower than the standard error of the mean. 
If it is case the CV is lower than Theoretical CV, then the Theoretical CV is used.  
</p>
<p>The output of this function is an object of type RegressionResult, a named list containing: 
</p>

<ul>
<li> <p><strong>prediction_statistics</strong> A dataframe with 7 rows and a column for every model. The rows are: 
</p>

<ul>
<li> <p><strong>Z_score_sample</strong> The regression based Z score for the model
</p>
</li>
<li> <p><strong>CV</strong> The coefficient of varation for the model
</p>
</li>
<li> <p><strong>cv_types</strong> The CV type used to calculate the regression based Z score for the model. Either 
<em>Practical_CV</em> or <em>Theoretical_CV</em>
</p>
</li>
<li> <p><strong>P_value_shapiro</strong> The P value of the Shaipro-Wilk test for normality of the control group
regression based Z scores for the model
</p>
</li>
<li> <p><strong>Predictor_chromosomes</strong> The predictor chromosomes used in the model
</p>
</li>
<li> <p><strong>Mean_test_set</strong> The mean of the test set. Note that for calculating the regression based
Z scores the mean is replaced by one. The mean, however, can be seen as a quality metric for the model 
</p>
</li>
<li> <p><strong>CV_train_set</strong> The CV of the train set. The difference between this CV and the CV of 
the test can be used as a measure to quantify overfit
</p>
</li></ul>

</li>
<li> <p><strong>control_group_Zscores</strong>  A matrix containing the regression based Z-scores for the control sample
</p>
</li>
<li> <p><strong>focus_chromosome</strong> he chromosome of interest. Most commonly chromosome 13, 18 or 21. 
However, every autosomal chromosome can be predicted
</p>
</li>
<li> <p><strong>correction_status</strong> The correction status of the control group autosomes
</p>
</li>
<li> <p><strong>control_group_sample_names</strong> The sample names of the test set group
</p>
</li>
<li> <p><strong>models</strong> List of the summary.lm output for every model
</p>
</li>
<li> <p><strong>potential_predictors</strong> The total pool of chromosomes where the predictors are selected from
</p>
</li>
<li> <p><strong>all_control_group_Z_scores</strong> Z-scores for every sample using theoretical and practical VCs
</p>
</li>
<li> <p><strong>additional_statistics</strong> Statistics for both the practical and theoretical CVs for every prediction set
</p>
</li></ul>



<h3>Value</h3>

<p>RegressionResult object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
regression_score_21 &lt;- perform_regression(nipt_sample = sample_of_interest, 
                       nipt_control_group = control_group, chromo_focus = 21)

## End(Not run)

</code></pre>

<hr>
<h2 id='prepare_ncv'>Prepare NCV calculation</h2><span id='topic+prepare_ncv'></span>

<h3>Description</h3>

<p>Determine the best NCV chromosomes,  calculate NCV scores 
and asses normal distribution control group using Shapiro-Wilk test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_ncv(nipt_control_group, chr_focus, max_elements,
  exclude_chromosomes = NULL, include_chromosomes = NULL,
  use_test_train_set = T, size_of_train_set = 0.6)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prepare_ncv_+3A_nipt_control_group">nipt_control_group</code></td>
<td>
<p>The NIPTControlGroup object used in the analysis</p>
</td></tr>
<tr><td><code id="prepare_ncv_+3A_chr_focus">chr_focus</code></td>
<td>
<p>Integer.The chromosome of interest. Most commonly chromosome 13, 18 or 21. 
However, every autosomal chromosome can be predicted</p>
</td></tr>
<tr><td><code id="prepare_ncv_+3A_max_elements">max_elements</code></td>
<td>
<p>Integer, The maximum number of denominator chromosomes.</p>
</td></tr>
<tr><td><code id="prepare_ncv_+3A_exclude_chromosomes">exclude_chromosomes</code></td>
<td>
<p>Integer. Exclude which autosomal chromosomes as potential predictors? 
Default potential trisomic chromosomes 13, 18 and 21 are exluded.</p>
</td></tr>
<tr><td><code id="prepare_ncv_+3A_include_chromosomes">include_chromosomes</code></td>
<td>
<p>Integer. Which potential trisomic chromosomes
(13,18 and 21) to include?</p>
</td></tr>
<tr><td><code id="prepare_ncv_+3A_use_test_train_set">use_test_train_set</code></td>
<td>
<p>Boolean. Use a test and train set?</p>
</td></tr>
<tr><td><code id="prepare_ncv_+3A_size_of_train_set">size_of_train_set</code></td>
<td>
<p>Double The size of the train set expressed in a decimal. 
Default is 0.6 (60% of the control group samples)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>chromosomes to calculate the chromosomal fractions. The 'best' subset is the set which 
yields the lowest coefficient of variation for the chromosomal fractions of the chromosome 
of interest in the control group. Because a brute force approach is used to determine the 
best subset, which can be computationally intensive,this method is divided into two functions, 
prepare_ncv and calculate_ncv. prepare_ncv returns a template object (NCVTemplate) for 
a given chromosome of interest and the control group used. This template can be used for 
any number of analyses. If the control group or chromosome of interest changes, 
a new template must be made.
</p>
<p>The ncv_template object is a list containing:
</p>

<ul>
<li><p> Character <strong>denominators</strong> The set of denominator chromosomes
</p>
</li>
<li><p> Character <strong>focus_chromosome</strong>The chromosome of interest used for this 'NCVTemplate' object
</p>
</li>
<li><p> Character <strong>nipt_sample_names</strong> The sample names of the test set samples
</p>
</li>
<li><p> Character <strong>correction_status</strong> The correction status(es) of the
control group samples
</p>
</li>
<li><p> Data.frame <strong>control_group_Z_scores</strong> The NCV scores for the test set samples
</p>
</li>
<li><p> Character <strong>potential_denominators</strong> The total pool of denominators the best denominators
are selected from
</p>
</li>
<li><p> Numeric <strong>control_group_statistics</strong> Named num of length 3, the first field being the mean 
(name mean), the second field is the standard deviation (name SD) and the third field is the P value 
of the Shapiro-Wilk test (name Shapiro_P_value)
</p>
</li></ul>

<p>If a Test and Train set is used the ncv_template object also includes:
</p>

<ul>
<li><p> Character <strong>sample_names_train_set</strong> The sample name where the model
is trained on
</p>
</li>
<li><p> Numeric <strong>train_set_statistics</strong> Mean, SD and Shapiro-Wilk test
P value of the Z scores of the train set
</p>
</li>
<li><p> Data.frame <strong>train_set_Zscores</strong> The Z scores of the train set
</p>
</li></ul>



<h3>Value</h3>

<p>ncv template object
</p>


<h3>References</h3>

<p><a href="http://www.ncbi.nlm.nih.gov/pubmed/21519036">Sehnert et al.</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
##Create NCVTemplates for chromosome 13 with max 9 denominators and default settings, so:
##All autosomals chromosomes are potential predictors, 
##except the potential trisomic chromosomes 13, 18 and 21
new_ncv_template_13 &lt;- prepare_ncv(nipt_control_group = control_group, 
                                   chr_focus = 13, max_elements = 9)

## End(Not run)

</code></pre>

<hr>
<h2 id='remove_duplicates_controlgroup'>Remove duplicate samples from control group</h2><span id='topic+remove_duplicates_controlgroup'></span>

<h3>Description</h3>

<p>Removes all duplicate samples in control group by samplename.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_duplicates_controlgroup(nipt_control_group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_duplicates_controlgroup_+3A_nipt_control_group">nipt_control_group</code></td>
<td>
<p>NIPTControlGroup object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions removes duplicate samples from the control group based on name. 
It returns a new NIPTControlGroup object.
</p>


<h3>Value</h3>

<p>NIPTControlGroup object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
new_control_group &lt;- remove_duplicates_controlgroup(nipt_control_group = old_control_group)

## End(Not run)

</code></pre>

<hr>
<h2 id='remove_sample_controlgroup'>Remove a sample by samplename from control group</h2><span id='topic+remove_sample_controlgroup'></span>

<h3>Description</h3>

<p>Remove a sample by samplename from control group
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_sample_controlgroup(samplename, nipt_control_group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_sample_controlgroup_+3A_samplename">samplename</code></td>
<td>
<p>Regular expression string. All matching samplenames 
are removed from the control group</p>
</td></tr>
<tr><td><code id="remove_sample_controlgroup_+3A_nipt_control_group">nipt_control_group</code></td>
<td>
<p>NIPTControlGroup object to remove samples from</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function removes a sample from the 'NIPTControlGroup' object by name.
Note that this function uses a regular expression, and if more sample_names 
satisfy the regular expression, they will also be removed. It returns a new 
NIPTControlGroup object.
</p>


<h3>Value</h3>

<p>NIPTControlGroup object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
new_control_group &lt;- remove_sample_controlgroup(samplename = unwanted_sample, 
                                                nipt_control_group = old_control_group)

## End(Not run)

</code></pre>

<hr>
<h2 id='retrieve_fractions_of_interest'>Retrieve the chromosomal fractions of a chromosome of interest</h2><span id='topic+retrieve_fractions_of_interest'></span>

<h3>Description</h3>

<p>Retrieve the chromosomal fractions of a chromosome of interest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retrieve_fractions_of_interest(nipt_sample, chromo_focus, chromosomal_fracs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="retrieve_fractions_of_interest_+3A_nipt_sample">nipt_sample</code></td>
<td>
<p>NIPTSample to check wether the strands are combined or 
separated</p>
</td></tr>
<tr><td><code id="retrieve_fractions_of_interest_+3A_chromo_focus">chromo_focus</code></td>
<td>
<p>The chromosome of interest</p>
</td></tr>
<tr><td><code id="retrieve_fractions_of_interest_+3A_chromosomal_fracs">chromosomal_fracs</code></td>
<td>
<p>The chromosomal fractions to extract the 
chromosome of interest from</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
