<!DOCTYPE html><html><head><title>Help for package NlsyLinks</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {NlsyLinks}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#NlsyLinks-package'><p>Utilities and kinship information for Behavior Genetics and Developmental research using the NLSY.</p></a></li>
<li><a href='#Ace'><p>Estimates the heritability of additive traits using a single variable.</p></a></li>
<li><a href='#AceEstimate-class'><p>Class AceEstimate</p></a></li>
<li><a href='#AceLavaanGroup'><p>A simple multiple-group ACE model with the <span class="pkg">lavaan</span> package.</p></a></li>
<li><a href='#CleanSemAceDataset'><p>Produces a cleaned dataset that works well with when using SEM to estimate a univariate ACE model.</p></a></li>
<li><a href='#ColumnUtilities'><p>A collection of functions that helps data management data.frames, particularly those derived from NLSY Extracts.</p></a></li>
<li><a href='#CreateAceEstimate'><p>Instantiate an AceEstimate object.</p></a></li>
<li><a href='#CreatePairLinks'><p>Creates a pairs linking file.</p></a></li>
<li><a href='#CreateSpatialNeighbours'><p>Distances between related family members, formatted for spatial analysis.</p></a></li>
<li><a href='#CreateSubjectTag'><p>Creates a SubjectTag.  This value uniquely identifies subjects, when both generations are included in the same dataset.</p></a></li>
<li><a href='#ExtraOutcomes79'><p>Extra outcome variables in the NLSY79</p></a></li>
<li><a href='#GetDetails-methods'><p>GetDetails-methods</p></a></li>
<li><a href='#Links79Pair'><p>Kinship linking file for pairs of relatives in the NLSY79 and NLSY79 Children and Young Adults</p></a></li>
<li><a href='#Links79PairExpanded'><p>Kinship linking file for pairs of relatives.  It builds upon the Links79Pair dataset.</p></a></li>
<li><a href='#Links97Pair'><p>Kinship linking file for pairs of relatives in the NLSY97</p></a></li>
<li><a href='#Links97PairExpanded'><p>Kinship linking file for pairs of relatives In the NLSY97.  It builds upon the Links97Pair dataset.</p></a></li>
<li><a href='#ReadCsvNlsy79'><p>Read a CSV file downloaded from the NLS Investigator</p></a></li>
<li><a href='#RGroupSummary'><p>Calculates summary statistics for each <em>R</em>elatedness Group in the sample.</p></a></li>
<li><a href='#SubjectDetails79'><p>Dataset containing further details of the Gen1 and Gen2 subjects.</p></a></li>
<li><a href='#Survey79'><p>Dataset containing survey details for each subject, for each year</p></a></li>
<li><a href='#ValidateOutcomeDataset'><p>Validates the schema of datasets containing outcome variables.</p></a></li>
<li><a href='#ValidatePairLinks'><p>Validates the schema of a links for pairs of relatives</p></a></li>
<li><a href='#ValidatePairLinksAreSymmetric'><p>Verifies that the pair relationships are symmetric.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Utilities and Kinship Information for Research with the NLSY</td>
</tr>
<tr>
<td>Version:</td>
<td>2.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Utilities and kinship information for behavior genetics and
    developmental research using the National Longitudinal Survey of Youth
    (NLSY; <a href="https://www.nlsinfo.org/">https://www.nlsinfo.org/</a>).</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://nlsy-links.github.io/NlsyLinks/">https://nlsy-links.github.io/NlsyLinks/</a>,
<a href="https://github.com/nlsy-links/NlsyLinks">https://github.com/nlsy-links/NlsyLinks</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/nlsy-links/NlsyLinks/issues">https://github.com/nlsy-links/NlsyLinks/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), stats</td>
</tr>
<tr>
<td>Imports:</td>
<td>lavaan, methods</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, devtools, dplyr, spelling, ggplot2, knitr, rmarkdown,
scales, testit, testthat (&ge; 2.0.0), xtable</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-21 20:32:22 UTC; smaso</td>
</tr>
<tr>
<td>Author:</td>
<td>William Howard Beasley
    <a href="https://orcid.org/0000-0002-5613-5006"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Joseph Lee Rodgers
    <a href="https://orcid.org/0000-0002-8615-6713"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  David Bard [ctb],
  Michael D. Hunter <a href="https://orcid.org/0000-0002-3651-6709"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Patrick O'Keefe <a href="https://orcid.org/0000-0002-5343-0130"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Kelly Meredith Williams [ctb],
  S. Mason Garrison <a href="https://orcid.org/0000-0002-4804-6003"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>S. Mason Garrison &lt;garrissm@wfu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-22 07:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='NlsyLinks-package'>Utilities and kinship information for Behavior Genetics and Developmental research using the NLSY.</h2><span id='topic+NlsyLinks-package'></span><span id='topic+NlsyLinks'></span>

<h3>Description</h3>

<p>Utilities and kinship information for Behavior Genetics and Developmental research using the NLSY.
Researchers and grad students interested using the <a href="https://www.nlsinfo.org/">NLSY</a> for Behavior Genetics and family research, please start with our 2016 article,
'<a href="https://link.springer.com/article/10.1007/s10519-016-9785-3"><em>The NLSY Kinship Links: Using the NLSY79 and NLSY-Children Data to Conduct Genetically-Informed and Family-Oriented Research</em></a>.
</p>


<h3>Note</h3>

<p>This package considers both Gen1 and Gen2 subjects.  &quot;Gen1&quot; refers to
subjects in the original NLSY79 sample
(https://www.nlsinfo.org/content/cohorts/nlsy79).  &quot;Gen2&quot; subjects are the
biological children of the Gen1 females -ie, those in the NLSY79 Children
and Young Adults sample (https://www.nlsinfo.org/content/cohorts/nlsy79-children).
</p>
<p>The release version is available through <a href="https://cran.r-project.org/package=NlsyLinks">CRAN</a> by
running <code>install.packages('NlsyLinks')</code>.
The most recent development version is available through <a href="https://github.com/nlsy-links/NlsyLinks">GitHub</a> by
running
<code>devtools::install_github</code> <code>(repo = 'nlsy-links/NlsyLinks')</code>
(make sure <a href="https://cran.r-project.org/package=devtools">devtools</a> is already installed).
If you're having trouble with the package, please install the development version.  If this doesn't solve
your problem, please create a <a href="https://github.com/nlsy-links/NlsyLinks/issues">new issue</a>, or email Will.
</p>


<h3>Author(s)</h3>

<p><a href="http://scholar.google.com/citations?user=ffsJTC0AAAAJ">William Howard Beasley</a> (Howard Live Oak LLC, Norman)
</p>
<p><a href="https://www.vanderbilt.edu/psychological_sciences/bio/joe-rodgers">Joseph Lee Rodgers</a> (Vanderbilt University, Nashville)
</p>
<p><a href="https://medicine.ouhsc.edu/Academic-Departments/Pediatrics/Sections/Developmental-Behavioral-Pediatrics/Faculty/david-e-bard-phd">David Bard</a> (University of Oklahoma Health Sciences Center, OKC)
</p>
<p><a href="https://scholar.google.com/citations?user=TbTNHKwAAAAJ&amp;hl=en">Michael D. Hunter</a> (Pennsylvania State University)
</p>
<p>Patrick O'Keefe (Oregon Health and Science University)
</p>
<p>Kelly Meredith Williams (Oklahoma City University, OKC)
</p>
<p><a href="https://scholar.google.com/citations?user=5to21boAAAAJ">S. Mason Garrison</a> (Wake Forest University)
</p>
<p>Maintainer: S. Mason Garrison <a href="mailto:garrissm@wfu.edu">garrissm@wfu.edu</a>
</p>


<h3>References</h3>

<p>This package's development was largely supported by the NIH
Grant 1R01HD65865, <a href="https://taggs.hhs.gov/Detail/AwardDetail?arg_awardNum=R01HD065865&amp;arg_ProgOfficeCode=50">&quot;NLSY Kinship Links: Reliable and Valid Sibling Identification&quot;</a>
(PI: Joe Rodgers).  A more complete list of research articles
using NLSY Kinship Links is maintained on our <a href="https://nlsy-links.github.io/NlsyLinks/articles/publications.html">package's website</a>.
</p>
<p>Rodgers, Joseph Lee, &amp; Kohler, Hans-Peter (2005).
<a href="https://pubmed.ncbi.nlm.nih.gov/15685433/">Reformulating and simplifying the DF analysis model</a>.
<em>Behavior Genetics, 35</em> (2), 211-217.
</p>
<p>Rodgers, J.L., Bard, D., Johnson, A., D'Onofrio, B., &amp; Miller, W.B. (2008).
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2712575/">The Cross-Generational Mother-Daughter-Aunt-Niece Design: Establishing Validity of the MDAN Design with NLSY Fertility Variables.</a>.
<em>Behavior Genetics, 38</em>, 567-578.
</p>
<p>D'Onofrio, B.M., Van Hulle, C.A., Waldman, I.D., Rodgers, J.L., Rathouz,
P.J., &amp; Lahey, B.B. (2007). <a href="https://pubmed.ncbi.nlm.nih.gov/17984398/">Causal inferences regarding prenatal alcohol exposure and childhood externalizing problems.</a>.
<em>Archives of General Psychiatry, 64</em>, 1296-1304.
</p>
<p>Rodgers, J.L. &amp; Doughty, D. (2000).  <a href="https://link.springer.com/chapter/10.1007/978-1-4615-4467-8_6">Genetic and environmental influences on fertility expectations and outcomes using NLSY kinship data.</a>
In J.L. Rodgers, D. Rowe, &amp; W.B. Miller (Eds.)
<em>Genetic influences on fertility and sexuality.</em> Boston: Kluwer Academic Press.
</p>
<p>Cleveland, H.H., Wiebe, R.P., van den Oord, E.J.C.G., &amp; Rowe, D.C. (2000).
<a href="https://pubmed.ncbi.nlm.nih.gov/10953940/">Behavior problems among children from different family structures: The influence of genetic self-selection.</a>
<em>Child Development, 71</em>, 733-751.
</p>
<p>Rodgers, J.L., Rowe, D.C., &amp; Buster, M. (1999). <a href="https://pubmed.ncbi.nlm.nih.gov/10081235/">Nature, nurture, and first sexual intercourse in the USA: Fitting behavioural genetic models to NLSY kinship data</a>.
<em>Journal of Biosocial Sciences, 31</em>.
</p>
<p>Rodgers, J.L., Rowe, D.C., &amp; Li, C. (1994). <a href="https://psycnet.apa.org:443/journals/dev/30/3/374/">Beyond nature versus nurture: DF analysis of nonshared influences on problem behaviors.</a>
<em>Developmental Psychology, 30</em>, 374-384.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.
summary(Links79Pair) # Summarize the five variables.
hist(Links79Pair$R) # Display a histogram of the Relatedness values.
table(Links79Pair$R) # Create a table of the Relatedness values for the whole sample.

## Not run: 
# Install/update NlsyLinks with the release version from CRAN.
install.packages("NlsyLinks")

# Install/update NlsyLinks with the development version from GitHub
# install.packages('devtools') #Uncomment if 'devtools' isn't installed already.
devtools::install_github("nlsy-links/NlsyLinks")

## End(Not run)
</code></pre>

<hr>
<h2 id='Ace'>Estimates the heritability of additive traits using a single variable.</h2><span id='topic+Ace'></span><span id='topic+AceUnivariate'></span><span id='topic+DeFriesFulkerMethod1'></span><span id='topic+DeFriesFulkerMethod3'></span>

<h3>Description</h3>

<p>An ACE model is the foundation of most behavior genetic research.  It
estimates the additive heritability (with <em>a</em>), common environment
(with <em>c</em>) and unshared heritability/environment (with <em>e</em>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AceUnivariate(
  method         = c("DeFriesFulkerMethod1","DeFriesFulkerMethod3"),
  dataSet,
  oName_S1,
  oName_S2,
  rName          = "R",
  manifestScale  = "Continuous"
)

DeFriesFulkerMethod1(dataSet, oName_S1, oName_S2, rName="R")

DeFriesFulkerMethod3(dataSet, oName_S1, oName_S2, rName="R")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ace_+3A_method">method</code></td>
<td>
<p>The specific estimation technique.</p>
</td></tr>
<tr><td><code id="Ace_+3A_dataset">dataSet</code></td>
<td>
<p>The <a href="base.html#topic+data.frame">base::data.frame</a> that contains the two outcome
variables and the relatedness coefficient (corresponding to <code>oName_S1</code>,
<code>oName_S2</code>, and <code>rName</code>)</p>
</td></tr>
<tr><td><code id="Ace_+3A_oname_s1">oName_S1</code></td>
<td>
<p>The name of the outcome variable corresponding to the first
subject in the pair. This should be a <code>character</code> value.</p>
</td></tr>
<tr><td><code id="Ace_+3A_oname_s2">oName_S2</code></td>
<td>
<p>The name of the outcome variable corresponding to the second
subject in the pair. This should be a <code>character</code> value.</p>
</td></tr>
<tr><td><code id="Ace_+3A_rname">rName</code></td>
<td>
<p>The name of the relatedness coefficient for the pair (this is
typically abbreviated as <code>R</code>). This should be a <code>character</code> value.</p>
</td></tr>
<tr><td><code id="Ace_+3A_manifestscale">manifestScale</code></td>
<td>
<p>Currently, only <em>continuous</em> manifest/outcome variables are supported.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code><a href="#topic+AceUnivariate">AceUnivariate()</a></code> function is a wrapper that calls
<code><a href="#topic+DeFriesFulkerMethod1">DeFriesFulkerMethod1()</a></code> or <code><a href="#topic+DeFriesFulkerMethod3">DeFriesFulkerMethod3()</a></code>.  Future
versions will incorporate methods that use latent variable models.
</p>


<h3>Value</h3>

<p>Currently, a list is returned with the arguments <code>ASquared</code>, <code>CSquared</code>, <code>ESquared</code>, and <code>RowCount</code>.
In the future, this may be changed to an <code>S4</code> class.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>References</h3>

<p>Rodgers, Joseph Lee, &amp; Kohler, Hans-Peter (2005).
<a href="https://pubmed.ncbi.nlm.nih.gov/15685433/">Reformulating and simplifying the DF analysis model.</a>
<em>Behavior Genetics, 35</em> (2), 211-217.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.
dsOutcomes &lt;- ExtraOutcomes79
dsOutcomes$SubjectTag &lt;- CreateSubjectTag(
  subjectID    = dsOutcomes$SubjectID,
  generation   = dsOutcomes$Generation
)
dsLinks &lt;- Links79Pair
dsLinks &lt;- dsLinks[dsLinks$RelationshipPath == "Gen2Siblings", ] # Only Gen2 Sibs (ie, NLSY79C)
dsDF &lt;- CreatePairLinksDoubleEntered(
  outcomeDataset     = dsOutcomes,
  linksPairDataset   = dsLinks,
  outcomeNames       = c("MathStandardized", "HeightZGenderAge", "WeightZGenderAge")
)

estimatedAdultHeight &lt;- DeFriesFulkerMethod3(
  dataSet    = dsDF,
  oName_S1   = "HeightZGenderAge_S1",
  oName_S2   = "HeightZGenderAge_S2"
)
estimatedAdultHeight # ASquared and CSquared should be 0.60 and 0.10 for this rough analysis.

estimatedMath &lt;- DeFriesFulkerMethod3(
  dataSet    = dsDF,
  oName_S1   = "MathStandardized_S1",
  oName_S2   = "MathStandardized_S2"
)
estimatedMath # ASquared and CSquared should be 0.85 and 0.045.

class(GetDetails(estimatedMath))
summary(GetDetails(estimatedMath))
</code></pre>

<hr>
<h2 id='AceEstimate-class'>Class AceEstimate</h2><span id='topic+AceEstimate-class'></span><span id='topic+GetDetails+2CAceEstimate-method'></span>

<h3>Description</h3>

<p>A class containing information about a single univariate ACE model.
</p>
<p>A generic function for extracting the <code>Details</code> slot of an AceEstimation object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'AceEstimate'
GetDetails(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AceEstimate-class_+3A_object">object</code></td>
<td>
<p>ACE object</p>
</td></tr>
</table>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the
form:
<code>new("AceEstimate", aSquared, cSquared, eSquared, caseCount, unity, withinBounds, details, ...)</code>
</p>


<h3>Note</h3>

<p>The contents of the <code>Details</code> list depends on the underlying
estimation routine.  For example, when the ACE model is estimated with a DF
analysis, the output is an <code><a href="stats.html#topic+lm">stats::lm()</a></code> object, because the <code><a href="stats.html#topic+lm">stats::lm()</a></code> function
was used (ie, the basic general linear model).  Alternatively, if the
user specified the <code><a href="lavaan.html#topic+lavaan">lavaan::lavaan()</a></code> package should estimate that ACE model,
the output is a <code><a href="lavaan.html#topic+lavaan">lavaan::lavaan()</a></code> object.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.

showClass("AceEstimate")
est &lt;- CreateAceEstimate(.5, .2, .3, 40)
est
print(est)

</code></pre>

<hr>
<h2 id='AceLavaanGroup'>A simple multiple-group ACE model with the <span class="pkg">lavaan</span> package.</h2><span id='topic+AceLavaanGroup'></span>

<h3>Description</h3>

<p>This function uses the <span class="pkg">lavaan</span> package to estimate a univariate ACE model, using multiple groups.
Each group has a unique value of <code>R</code> (i.e., the <em>R</em>elatedness coefficient).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AceLavaanGroup(
  dsClean,
  estimateA = TRUE,
  estimateC = TRUE,
  printOutput = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AceLavaanGroup_+3A_dsclean">dsClean</code></td>
<td>
<p>The <a href="base.html#topic+data.frame">base::data.frame</a> containing complete cases for the <code>R</code> groups to be included in the estimation.</p>
</td></tr>
<tr><td><code id="AceLavaanGroup_+3A_estimatea">estimateA</code></td>
<td>
<p>Should the <em>A</em> variance component be estimated?  A^2 represents the proportion of variability due to a shared genetic influence.</p>
</td></tr>
<tr><td><code id="AceLavaanGroup_+3A_estimatec">estimateC</code></td>
<td>
<p>Should the <em>C</em> variance component be estimated?  C^2 represents the proportion of variability due to a shared environmental influence.</p>
</td></tr>
<tr><td><code id="AceLavaanGroup_+3A_printoutput">printOutput</code></td>
<td>
<p>Indicates if the estimated parameters and fit statistics are printed to the console.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The variance component for <em>E</em> is always estimated, while the <em>A</em> and <em>C</em> estimates can be fixed to zero (when <code>estimateA</code> and/or <em>estimateC</em> are set to <code>FALSE</code>).
</p>


<h3>Value</h3>

<p>An AceEstimate object.
</p>


<h3>Note</h3>

<p>Currently, the variables in <code>dsClean</code> must be named <code>O1</code>, <code>O2</code> and <code>R</code>; the letter 'O' stands for <em>O</em>utcome.  This may not be as restrictive as it initially seems, because <code>dsClean</code> is intended to be produced by <code><a href="#topic+CleanSemAceDataset">CleanSemAceDataset()</a></code>.  If this is too restrictive for your uses, we'd like to here about it (<em>please email wibeasley at hotmail period com</em>).
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>References</h3>

<p>The <span class="pkg">lavaan</span> package is developed by Yves Rosseel at Ghent University.
Three good starting points are the package website (http://lavaan.ugent.be/), the package documentation (https://cran.r-project.org/package=lavaan)
and the JSS paper.
</p>
<p>Rosseel, Yves (2012), <a href="https://www.jstatsoft.org/v48/i02/">lavaan: An R Package for Structural Equation Modeling</a>. <em>Journal of Statistical Software, 48</em>, (2), 1-36.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CleanSemAceDataset">CleanSemAceDataset()</a></code>.  Further ACE model details are discussed in our package's <a href="https://cran.r-project.org/package=NlsyLinks">vignettes</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.
dsLinks &lt;- Links79PairExpanded # Start with the built-in data.frame in NlsyLinks
dsLinks &lt;- dsLinks[dsLinks$RelationshipPath == "Gen2Siblings", ] # Use only Gen2 Siblings (NLSY79-C)

oName_S1 &lt;- "MathStandardized_S1" # Stands for Outcome1
oName_S2 &lt;- "MathStandardized_S2" # Stands for Outcome2

dsGroupSummary &lt;- RGroupSummary(dsLinks, oName_S1, oName_S2)
dsClean &lt;- CleanSemAceDataset(dsDirty = dsLinks, dsGroupSummary, oName_S1, oName_S2)

ace &lt;- AceLavaanGroup(dsClean)
ace

# Should produce:
# [1] "Results of ACE estimation: [show]"
#     ASquared     CSquared     ESquared    CaseCount
#    0.6681874    0.1181227    0.2136900 8390.0000000

library(lavaan) # Load the package to access methods of the lavaan class.
GetDetails(ace)

# Exmaine fit stats like Chi-Squared, RMSEA, CFI, etc.
fitMeasures(GetDetails(ace)) # The function 'fitMeasures' is defined in the lavaan package.

# Examine low-level details like each group's individual parameter estimates and standard errors.
summary(GetDetails(ace))

# Extract low-level details. This may be useful when programming simulations.
inspect(GetDetails(ace), what = "converged") # The lavaan package defines 'inspect'.
inspect(GetDetails(ace), what = "coef")
</code></pre>

<hr>
<h2 id='CleanSemAceDataset'>Produces a cleaned dataset that works well with when using SEM to estimate a univariate ACE model.</h2><span id='topic+CleanSemAceDataset'></span>

<h3>Description</h3>

<p>This function takes a 'GroupSummary' <a href="base.html#topic+data.frame">base::data.frame</a> (which is created by the <code><a href="#topic+RGroupSummary">RGroupSummary()</a></code> function) and returns a <a href="base.html#topic+data.frame">base::data.frame</a> that is used by the <code><a href="#topic+Ace">Ace()</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CleanSemAceDataset(dsDirty, dsGroupSummary, oName_S1, oName_S2, rName = "R")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CleanSemAceDataset_+3A_dsdirty">dsDirty</code></td>
<td>
<p>This is the <a href="base.html#topic+data.frame">base::data.frame</a> to be cleaned.</p>
</td></tr>
<tr><td><code id="CleanSemAceDataset_+3A_dsgroupsummary">dsGroupSummary</code></td>
<td>
<p>The <a href="base.html#topic+data.frame">base::data.frame</a> containing information about which groups should be included in the analyses.  It should be created by the <code><a href="#topic+RGroupSummary">RGroupSummary()</a></code> function.</p>
</td></tr>
<tr><td><code id="CleanSemAceDataset_+3A_oname_s1">oName_S1</code></td>
<td>
<p>The name of the manifest variable (in <code>dsDirty</code>) for the first subject in each pair.</p>
</td></tr>
<tr><td><code id="CleanSemAceDataset_+3A_oname_s2">oName_S2</code></td>
<td>
<p>The name of the manifest variable (in <code>dsDirty</code>) for the second subject in each pair.</p>
</td></tr>
<tr><td><code id="CleanSemAceDataset_+3A_rname">rName</code></td>
<td>
<p>The name of the variable (in <code>dsDirty</code>) indicating the pair's relatedness coefficient.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function takes <code>dsDirty</code> and produces a new <a href="base.html#topic+data.frame">base::data.frame</a> with the following features:
</p>

<ol>
<li><p> Only three existing columns are retained: <code>O1</code>, <code>O2</code>, and <code>R</code>.  They are assigned these names.
</p>
</li>
<li><p> A new column called <code>GroupID</code> is created to reflect their group membership (which is based on the <code>R</code> value).  These values are sequential integers, starting at 1.  The group with the weakest <code>R</code> is 1.  The group with the strongest <code>R</code> has the largest <code>GroupID</code> (this is typically the MZ twins).
</p>
</li>
<li><p> Any row is excluded if it has a missing data point for <code>O1</code>, <code>O2</code>, or <code>R</code>.
</p>
</li>
<li><p> The <a href="base.html#topic+data.frame">base::data.frame</a> is sorted by the <code>R</code> value.  This helps program against the multiple-group SEM API sometimes.
</p>
</li></ol>



<h3>Value</h3>

<p>A <a href="base.html#topic+data.frame">base::data.frame</a> with one row per subject pair.  The <a href="base.html#topic+data.frame">base::data.frame</a> contains the following variables (which can NOT be changed by the user through optional parameters):
</p>

<ul>
<li> <p><strong>R</strong> The pair's <code>R</code> value.
</p>
</li>
<li> <p><strong>O1</strong> The outcome variable for the first subject in each pair.
</p>
</li>
<li> <p><strong>O2</strong> The outcome variable for the second subject in each pair.
</p>
</li>
<li> <p><strong>GroupID</strong> Indicates the pair's group membership.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.
dsLinks &lt;- Links79PairExpanded # Start with the built-in data.frame in NlsyLinks
dsLinks &lt;- dsLinks[dsLinks$RelationshipPath == "Gen2Siblings", ] # Use only NLSY79-C siblings

oName_S1 &lt;- "MathStandardized_S1" # Stands for Outcome1
oName_S2 &lt;- "MathStandardized_S2" # Stands for Outcome2
dsGroupSummary &lt;- RGroupSummary(dsLinks, oName_S1, oName_S2)

dsClean &lt;- CleanSemAceDataset(dsDirty = dsLinks, dsGroupSummary, oName_S1, oName_S2, rName = "R")
summary(dsClean)

dsClean$AbsDifference &lt;- abs(dsClean$O1 - dsClean$O2)
plot(jitter(dsClean$R), dsClean$AbsDifference, col = "gray70")

</code></pre>

<hr>
<h2 id='ColumnUtilities'>A collection of functions that helps data management data.frames, particularly those derived from NLSY Extracts.</h2><span id='topic+ColumnUtilities'></span><span id='topic+VerifyColumnExists'></span><span id='topic+RenameColumn'></span><span id='topic+RenameNlsyColumn'></span>

<h3>Description</h3>

<p>A collection of functions that helps data management <a href="base.html#topic+data.frame">base::data.frame</a>s, particularly those derived from NLSY Extracts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VerifyColumnExists( dataFrame, columnName )

RenameColumn( dataFrame, oldColumnName, newColumnName )

RenameNlsyColumn( dataFrame, nlsyRNumber, newColumnName )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ColumnUtilities_+3A_dataframe">dataFrame</code></td>
<td>
<p>The <a href="base.html#topic+data.frame">base::data.frame</a> whose columns are to be verified or renamed.</p>
</td></tr>
<tr><td><code id="ColumnUtilities_+3A_columnname">columnName</code></td>
<td>
<p>The name of the column to verify is present in the <a href="base.html#topic+data.frame">base::data.frame</a>.</p>
</td></tr>
<tr><td><code id="ColumnUtilities_+3A_nlsyrnumber">nlsyRNumber</code></td>
<td>
<p>The name of the column to change.</p>
</td></tr>
<tr><td><code id="ColumnUtilities_+3A_oldcolumnname">oldColumnName</code></td>
<td>
<p>The name of the column to change.</p>
</td></tr>
<tr><td><code id="ColumnUtilities_+3A_newcolumnname">newColumnName</code></td>
<td>
<p>The desired name of the column.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The RNumber assigned by the NLS has a pattern.  In the Nlsy79 Gen1 dataset, the names start with a 'R' or 'T' and are followed by seven digits (eg, R0000100).  In the Nlsy79 Gen2 dataset, the names start with 'C' or 'Y' and are followed by seven digits (eg, C0007030, Y1994600).
</p>
<p>In the NLS Investigator, a decimal is present in the RNumber (eg, R00001.00).  When the Investigator saves the dataset as a CSV, the decimal is removed (eg, R0000100).
</p>


<h3>Value</h3>

<p><em>IMPORTANT</em> The <code><a href="#topic+RenameColumn">RenameColumn()</a></code> and <code><a href="#topic+RenameNlsyColumn">RenameNlsyColumn()</a></code> functions do not use side-effects to rename the <a href="base.html#topic+data.frame">base::data.frame</a>.  Instead, it returns a new <a href="base.html#topic+data.frame">base::data.frame</a>.  In the example below, notice the assignment to <code>ds</code>: <code>ds &lt;- RenameNlsyColumn(...)</code>.
</p>
<p>The <code><a href="#topic+VerifyColumnExists">VerifyColumnExists()</a></code> function check that exactly one column exists with the specified <code>columnName</code>.  If so, the index of the column is returned.  If not, an exception is thrown.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>

<hr>
<h2 id='CreateAceEstimate'>Instantiate an <a href="#topic+AceEstimate-class">AceEstimate</a> object.</h2><span id='topic+CreateAceEstimate'></span>

<h3>Description</h3>

<p>Creates an instance of the <code>S4</code> class, <a href="#topic+AceEstimate-class">AceEstimate</a> instantiating arguments set the parameter values estimated by the ACE model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateAceEstimate(
  aSquared,
  cSquared,
  eSquared,
  caseCount,
  details = list(),
  unityTolerance = 1e-11
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateAceEstimate_+3A_asquared">aSquared</code></td>
<td>
<p>The proportion of variability due to a shared genetic influence (typically represented as a^2, or sometimes h^2).</p>
</td></tr>
<tr><td><code id="CreateAceEstimate_+3A_csquared">cSquared</code></td>
<td>
<p>The proportion of variability due to shared common environmental influence.</p>
</td></tr>
<tr><td><code id="CreateAceEstimate_+3A_esquared">eSquared</code></td>
<td>
<p>The proportion of variability due to unexplained/residual/error influence.</p>
</td></tr>
<tr><td><code id="CreateAceEstimate_+3A_casecount">caseCount</code></td>
<td>
<p>The number of cases used to estimate the model.</p>
</td></tr>
<tr><td><code id="CreateAceEstimate_+3A_details">details</code></td>
<td>
<p>A <code>list</code> that contains the modeling output and details.</p>
</td></tr>
<tr><td><code id="CreateAceEstimate_+3A_unitytolerance">unityTolerance</code></td>
<td>
<p>Specifies how close the the sum of the ACE components should be to one, to be considered properly scaled to one.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The contents of the <code>details</code> list depends on the underlying estimation routine.  For example, when the ACE model is estimated with a DF analysis, the output is a <a href="stats.html#topic+lm">stats::lm</a> object, because the <a href="stats.html#topic+lm">stats::lm</a> function was used (ie, the basic general linear model).  Alternatively, if the user specified the <code>lavaan</code> package should estimate that ACE model, the output is a <a href="lavaan.html#topic+lavaan">lavaan::lavaan</a> object.
</p>


<h3>Value</h3>

<p>An S4 object of <a href="#topic+AceEstimate-class">AceEstimate</a>.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>

<hr>
<h2 id='CreatePairLinks'>Creates a pairs linking file.</h2><span id='topic+CreatePairLinks'></span><span id='topic+CreatePairLinksDoubleEntered'></span><span id='topic+CreatePairLinksSingleEntered'></span><span id='topic+CreatePairLinksDoubleEnteredWithNoOutcomes'></span>

<h3>Description</h3>

<p>Creates a linking file for BG designs using this file structure (e.g., DF analysis, other ACE modeling).
DF analysis requires a double-entered file that contains the <code>R</code>
value for the pair, and their two outcome variable values.
</p>
<p><code><a href="#topic+CreatePairLinksDoubleEnteredWithNoOutcomes">CreatePairLinksDoubleEnteredWithNoOutcomes()</a></code> is intended to be a
primarily a helper function for <code><a href="#topic+CreateSpatialNeighbours">CreateSpatialNeighbours()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreatePairLinksDoubleEntered(
  outcomeDataset,
  linksPairDataset,
  outcomeNames,
  linksNames             = c("ExtendedID", "R", "RelationshipPath"),
  validateOutcomeDataset = TRUE,
  subject1Qualifier      = "_S1",
  subject2Qualifier      = "_S2"
)

CreatePairLinksSingleEntered(
  outcomeDataset,
  linksPairDataset,
  outcomeNames,
  linksNames             = c("ExtendedID", "R", "RelationshipPath"),
  validateOutcomeDataset = TRUE,
  subject1Qualifier      = "_S1",
  subject2Qualifier      = "_S2"
)

CreatePairLinksDoubleEnteredWithNoOutcomes(
  linksPairDataset,
  linksNames = c("ExtendedID", "R", "RelationshipPath")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreatePairLinks_+3A_outcomedataset">outcomeDataset</code></td>
<td>
<p>A data frame containing the outcome variable(s)</p>
</td></tr>
<tr><td><code id="CreatePairLinks_+3A_linkspairdataset">linksPairDataset</code></td>
<td>
<p>A data frame containing the <code>SubjectTag</code>s of
each subject in the pair and their <code>R</code> coefficient.</p>
</td></tr>
<tr><td><code id="CreatePairLinks_+3A_outcomenames">outcomeNames</code></td>
<td>
<p>The column names of the outcome variable(s)</p>
</td></tr>
<tr><td><code id="CreatePairLinks_+3A_linksnames">linksNames</code></td>
<td>
<p>The column names desired to be present in the newly
created data frame.  <code>SubjectTag_S1</code> and <code>SubjectTag_S2</code> are included
automatically.</p>
</td></tr>
<tr><td><code id="CreatePairLinks_+3A_validateoutcomedataset">validateOutcomeDataset</code></td>
<td>
<p>Indicates if characteristics of the outcomeDataset should be validated.</p>
</td></tr>
<tr><td><code id="CreatePairLinks_+3A_subject1qualifier">subject1Qualifier</code></td>
<td>
<p>Indicates how the outcome variable for the pair's first subject is distinguished from the other subject.  The default is <code style="white-space: pre;">&#8288;_S1&#8288;</code>.</p>
</td></tr>
<tr><td><code id="CreatePairLinks_+3A_subject2qualifier">subject2Qualifier</code></td>
<td>
<p>Indicates how the outcome variable for the pair's second subject is distinguished from the other subject.  The default is <code style="white-space: pre;">&#8288;_S2&#8288;</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>References</h3>

<p>For more information about a DF analysis, see Rodgers, Joseph Lee, &amp; Kohler, Hans-Peter (2005).
<a href="https://pubmed.ncbi.nlm.nih.gov/15685433/">Reformulating and simplifying the DF analysis model.</a>
<em>Behavior Genetics, 35</em> (2), 211-217.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dsSingleLinks &lt;- data.frame(
  ExtendedID = c(1, 1, 1, 2),
  SubjectTag_S1 = c(101, 101, 102, 201),
  SubjectTag_S2 = c(102, 103, 103, 202),
  R = c(.5, .25, .25, .5),
  RelationshipPath = rep("Gen2Siblings", 4)
)
dsSingleOutcomes &lt;- data.frame(
  SubjectTag = c(101, 102, 103, 201, 202),
  DV1        = c(11, 12, 13, 41, 42),
  DV2        = c(21, 22, 23, 51, 52)
)
dsDouble &lt;- CreatePairLinksDoubleEntered(
  outcomeDataset         = dsSingleOutcomes,
  linksPairDataset       = dsSingleLinks,
  outcomeNames           = c("DV1", "DV2"),
  validateOutcomeDataset = TRUE
)
dsDouble # Show the 8 rows in the double-entered pair links
summary(dsDouble) # Summarize the variables

ValidatePairLinksAreSymmetric(dsDouble) # Should return TRUE.
</code></pre>

<hr>
<h2 id='CreateSpatialNeighbours'>Distances between related family members, formatted for spatial analysis.</h2><span id='topic+CreateSpatialNeighbours'></span>

<h3>Description</h3>

<p>This helper function formats the LinksPair datasets so it can be used in some types of spatial analyses. The <span class="pkg">spdep</span> (Spatial Dependence) uses a sparse matrix (actually a <a href="base.html#topic+data.frame">base::data.frame</a>) to represent neighbours.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateSpatialNeighbours(linksPairsDoubleEntered)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateSpatialNeighbours_+3A_linkspairsdoubleentered">linksPairsDoubleEntered</code></td>
<td>
<p>A <a href="base.html#topic+data.frame">base::data.frame</a> containing the links, preferably created by a function like <code><a href="#topic+CreatePairLinksDoubleEntered">CreatePairLinksDoubleEntered()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There is one row per unique pair of subjects, <em>respecting order</em>.  This has twice as many rows as <a href="#topic+Links79Pair">Links79Pair</a> and <a href="#topic+Links79PairExpanded">Links79PairExpanded</a> (which have one row per unique pair of subjects, <em>irrespective of order</em>).
</p>
<p><code><a href="#topic+CreateSpatialNeighbours">CreateSpatialNeighbours()</a></code> accepts any paired relationships in a <a href="base.html#topic+data.frame">base::data.frame</a>, as long as it contains the columns <code>SubjectTag_S1</code>, <code>SubjectTag_S2</code>, and <code>R</code>.  See <a href="#topic+Links79Pair">Links79Pair</a> for more details about these columns.
</p>


<h3>Value</h3>

<p>An S3 <code>spdep::spatial.neighbours</code> object to work with functions in the <span class="pkg">spdep</span> package.
</p>
<p><code>SubjectTag_S1</code> is renamed '<code>from</code>'.
</p>
<p><code>SubjectTag_S2</code> is renamed '<code>to</code>'.
</p>
<p><code>R</code> is renamed '<code>weight</code>'.
</p>
<p>The attribute <code>region.id</code> specifies each unique SubjectTag.
</p>
<p>The attribute <code>n</code> specifies the number of unique subjects.
</p>


<h3>Note</h3>

<p>Notice the British variant of 'neighbo<em>u</em>rs' is used, to be consistent with the <code>spdep::spatial.neighbour</code> class.
</p>


<h3>Author(s)</h3>

<p>Will Beasley and  David Bard
</p>


<h3>References</h3>

<p>Bard, D.E., Beasley, W.H., Meredith, K., &amp; Rodgers, J.L. (2012). <a href="https://link.springer.com/article/10.1007/s10519-012-9566-6"><em>Biometric Analysis of Complex NLSY Pedigrees: Introducing a Conditional Autoregressive Biometric (CARB) Mixed Model</em></a>. Behavior Genetics Association 42nd Annual Meeting. <a href="https://r-forge.r-project.org/forum/forum.php?thread_id=4761&amp;forum_id=4266&amp;group_id=1330">Slides</a>.
</p>
<p>Bivand, R., Pebesma, E., &amp; Gomez-Rubio, V. (2013). <a href="https://link.springer.com/book/10.1007/978-1-4614-7618-4"><em>Applied Spatial Data Analysis with R.</em></a> New York: Springer. (Especially Chapter 9.)
</p>
<p>Banerjee, S., Carlin, B.P., &amp; Gelfand, A.E. (2004). <a href="http://books.google.com/books/about/Hierarchical_Modeling_and_Analysis_for_S.html?id=YqpZKTp-Wh0C"><em>Hierarchical Modeling and Analysis for Spatial Data</em></a>. Boca Raton: CRC Press.
</p>
<p>Lawson, A.B (2013). <a href="http://books.google.com/books?id=g7RJEZb1umwC"><em>Bayesian Disease Mapping: Hierarchical Modeling in Spatial Epidemiology, Second Edition</em></a>. Boca Raton: CRC Press.
</p>
<p>The <span class="pkg">spdep</span> package documentation: <a href="https://cran.r-project.org/package=spdep">spdep: Spatial dependence: weighting schemes, statistics and models</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dsLinksAll &lt;- Links79Pair
dsLinksGen1Housemates &lt;- dsLinksAll[dsLinksAll$RelationshipPath == "Gen1Housemates", ]
dsLinksGen2Siblings &lt;- dsLinksAll[dsLinksAll$RelationshipPath == "Gen2Siblings", ]

spGen1 &lt;- CreateSpatialNeighbours(dsLinksGen1Housemates)
spGen2 &lt;- CreateSpatialNeighbours(dsLinksGen2Siblings)

head(spGen2)
# Returns:
#   from  to weight
# 3  201 202   0.50
# 6  301 302   0.50
# 7  301 303   0.50
# 9  302 303   0.50
# 24  401 403   0.25
# 28  801 802   0.50

table(spGen2$weight)
# Returns:
# 0.25 0.375   0.5  0.75     1
# 3442   610  6997    12    27
</code></pre>

<hr>
<h2 id='CreateSubjectTag'>Creates a SubjectTag.  This value uniquely identifies subjects, when both generations are included in the same dataset.</h2><span id='topic+CreateSubjectTag'></span>

<h3>Description</h3>

<p>A <code>SubjectTag</code> uniquely identify subjects.  For Gen2 subjects, the SubjectTag is identical to their CID (ie, C00001.00 -the SubjectID assigned in the NLSY79-Children files).  However for Gen1 subjects, the SubjectTag is their CaseID (ie, R00001.00), with &quot;00&quot; appended.  This manipulation is necessary to identify subjects uniquely in inter-generational datasets.  A Gen1 subject with an ID of 43 becomes 4300.  The SubjectTags of her four children remain 4301, 4302, 4303, and 4304.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateSubjectTag(subjectID, generation)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateSubjectTag_+3A_subjectid">subjectID</code></td>
<td>
<p>The ID assigned by the NLSY.  For Gen1 subjects, this will be their CaseID (ie, R00001.00).  For Gen2 subjects, this will be their CID (ie, C00001.00).</p>
</td></tr>
<tr><td><code id="CreateSubjectTag_+3A_generation">generation</code></td>
<td>
<p>The generation of the subject.  Values are either 1 or 2, representing Gen1 and Gen2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a fuller explanation of <code>SubjectTag</code> in context, see the <a href="#topic+Links79Pair">Links79Pair</a> dataset documentation.
</p>


<h3>Value</h3>

<p>A integer value under normal circumstances. An error is thrown if the vectors <code>subjectID</code> and <code>generation</code> are different lengths. If either input vector has NA values, the respective output element(s) will be NA too.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>See Also</h3>

<p><a href="#topic+Links79Pair">Links79Pair</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.

# Typically these two vectors will come from a data frame.
subjectIDs &lt;- c(71:82, 10001:10012)
generation &lt;- c(rep(1, 12), rep(2, 12))

CreateSubjectTag(subjectIDs, generation)
# Returns 7100, ..., 8200, 10001, ..., 10012

# Use the ExtraOutcomes79 dataset, with numeric variables 'SubjectID' and 'Generation'.
ExtraOutcomes79$SubjectTag &lt;- CreateSubjectTag(
  subjectID = ExtraOutcomes79$SubjectID,
  generation = ExtraOutcomes79$Generation
)
</code></pre>

<hr>
<h2 id='ExtraOutcomes79'>Extra outcome variables in the NLSY79</h2><span id='topic+ExtraOutcomes79'></span>

<h3>Description</h3>

<p>This dataset is provided primarily to facilitate documentation examples.
</p>


<h3>Format</h3>

<p>A data frame with 11,495 observations on the following 6 variables. There is one row per subject.
</p>

<ul>
<li> <p><strong>SubjectTag</strong> The ID value assigned by NLS to the first subject.  For Gen1 Subjects, this is their &quot;CaseID&quot; (ie, R00001.00).  For Gen2 subjects, this is their &quot;CID&quot; (ie, C00001.00).
</p>
</li>
<li> <p><strong>SubjectID</strong> The ID value assigned by NLS to the first subject.  For Gen1 Subjects, this is their &quot;CaseID&quot; (ie, R00001.00).  For Gen2 subjects, this is their &quot;CID&quot; (ie, C00001.00).
</p>
</li>
<li> <p><strong>Generation</strong> The generation of the subject.  Values are either 1 or 2, representing Gen1 and Gen2.  Note that this variable is not a  <code>factor</code> (in contrast with data frames like
<a href="#topic+Links79Pair">Links79Pair</a>).  This dataset is supposed to mimic the dataset provided by the researcher, which typically will not have been converted to a <code>factor</code>.
</p>
</li>
<li> <p><strong>HeightZGenderAge</strong> The subject's height, standardized by gender and age (see Details).
</p>
</li>
<li> <p><strong>WeightZGenderAge</strong> The subject's weight, standardized by gender and age (see Details).
</p>
</li>
<li> <p><strong>AfqtRescaled2006Gaussified</strong> Armed Forces Qualification Test Score (Gen1 only; see Details).
</p>
</li>
<li> <p><strong>Afi</strong> Self-reported age of first intercourse (Gen1 only; see Details).
</p>
</li>
<li> <p><strong>Afm</strong> Self-reported age of first menstruation (Gen1 only; see Details).
</p>
</li>
<li> <p><strong>MathStandardized</strong> Standardized PIAT Math scores (Gen2 only; see Details).
</p>
</li></ul>



<h3>Details</h3>

<p>The <code>SubjectTag</code> variable uniquely identify subjects.  For Gen2
subjects, the SubjectTag is identical to their CID (ie, C00001.00 -the
SubjectID assigned in the NLSY79-Children files).  However for Gen1
subjects, the SubjectTag is their CaseID (ie, R00001.00), with &quot;00&quot;
appended.  This manipulation is necessary to identify subjects uniquely in
inter-generational datasets.  A Gen1 subject with an ID of 43 has a
<code>SubjectTag</code> of 4300.  The SubjectTags of her four children remain
4301, 4302, 4303, and 4304.
</p>
<p>For Gen2, an NLSY79 variable of <code>MathStandardized</code> is C05801.00.
</p>
<p><code>Afi</code> and <code>Afm</code>, values were simplified
(to one value per subject) by Kelly Williams in Sept 2010.
</p>
<p>The variables for height and weight were manipulated in R files available in a
<a href="https://github.com/nlsy-links/nlsy-links-determination-2017/tree/master/data-public/derived">repository</a> available to the public.
Find the appropriate subfolder, and view the HTML report for more details.
</p>
<p><strong>Download CSV</strong>
If you're using the NlsyLinks package in R, the dataset is automatically available.
However to use the kinship information in a different environment, download the
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-79/extra-outcomes-79.csv?raw=true">csv</a>,
which is readable by all statistical software.
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-79/links-metadata-2017-79.yml">links-metadata-2017-79.yml</a>
documents the dataset version information.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>Source</h3>

<p>Gen1 information comes from the Summer 2013 release of the
<a href="https://www.nlsinfo.org/content/cohorts/nlsy79">NLSY79 sample</a>.  Gen2 information comes from the Summer 2013 release of the
<a href="https://www.nlsinfo.org/content/cohorts/nlsy79-children">NLSY79 Children and Young Adults sample</a>.  Data were extracted with the NLS Investigator
(https://www.nlsinfo.org/investigator/).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.
gen2Outcomes &lt;- subset(ExtraOutcomes79, Generation == 2) # Create a dataset of only Gen2 subjects.

# plot(ExtraOutcomes79) #Uncomment to see a large scatterplot matrix.
summary(ExtraOutcomes79)

oldPar &lt;- par(mfrow = c(3, 2))
hist(ExtraOutcomes79$Generation)
hist(ExtraOutcomes79$MathStandardized)
hist(ExtraOutcomes79$HeightZGenderAge)
hist(ExtraOutcomes79$WeightZGenderAge)
hist(ExtraOutcomes79$Afi)
hist(ExtraOutcomes79$Afm)
par(oldPar)

</code></pre>

<hr>
<h2 id='GetDetails-methods'>GetDetails-methods</h2><span id='topic+GetDetails-methods'></span><span id='topic+GetDetails'></span><span id='topic+AceEstimate-method'></span>

<h3>Description</h3>

<p>GetDetails-methods
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetDetails(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetDetails-methods_+3A_object">object</code></td>
<td>
<p>ACE object</p>
</td></tr>
</table>

<hr>
<h2 id='Links79Pair'>Kinship linking file for pairs of relatives in the NLSY79 and NLSY79 Children and Young Adults</h2><span id='topic+Links79Pair'></span>

<h3>Description</h3>

<p>This dataset specifies the relatedness coefficient (ie, '<code>R</code>') between
subjects in the same extended family.  Each row represents a unique
relationship pair.
</p>
<p>NOTE: Two variable names changed in November 2013. <code>Subject1Tag</code> and <code>Subject2Tag</code> became <code>SubjectTag_S1</code> and <code>SubjectTag_S2</code>.
</p>


<h3>Format</h3>

<p>A data frame with 42,773 observations on the following 5 variables.
There is one row per unique pair of subjects, irrespective of order.
</p>

<ul>
<li> <p><strong>ExtendedID</strong> Identity of the extended family of the pair; it corresponds to the HHID in the NLSY79.  See References below.
</p>
</li>
<li> <p><strong>SubjectTag_S1</strong> Identity of the pair's first subject.  See Details below.
</p>
</li>
<li> <p><strong>SubjectTag_S2</strong> Identity of the pair's second subject.  See Details below.
</p>
</li>
<li> <p><strong>R</strong> The pair's Relatedness coefficient.  See Details below.
</p>
</li>
<li> <p><strong>RelationshipPath</strong> Specifies the relationship category of the pair.  This variable is a factor, with levels <code>Gen1Housemates</code>=1, <code>Gen2Siblings</code>=2, <code>Gen2Cousins</code>=3, <code>ParentChild</code>=4, <code>AuntNiece</code>=5.
</p>
</li></ul>



<h3>Details</h3>

<p>The dataset contains Gen1 and Gen2 subjects.  &quot;Gen1&quot; refers to subjects in
the original NLSY79 sample (https://www.nlsinfo.org/content/cohorts/nlsy79).
&quot;Gen2&quot; subjects are the biological children of the Gen1 females -ie, those
in the NLSY79 Children and Young Adults sample
(https://www.nlsinfo.org/content/cohorts/nlsy79-children).
</p>
<p>Subjects will be in the same extended family if either:
</p>

<ol>
<li><p> they are Gen1 housemates,
</p>
</li>
<li><p> they are Gen2 siblings,
</p>
</li>
<li><p> they are Gen2 cousins (ie, they have mothers who are Gen1 sisters in the NLSY79,
</p>
</li>
<li><p> they are mother and child (in Gen1 and Gen2, respectively), or
</p>
</li>
<li><p> they are aunt|uncle and niece|nephew (in Gen1 and Gen2, respectively).
</p>
</li></ol>

<p>The variables <code>SubjectTag_S1</code> and <code>SubjectTag_S2</code> uniquely identify
subjects.  For Gen2 subjects, the SubjectTag is identical to their CID (ie,
C00001.00 -the SubjectID assigned in the NLSY79-Children files).  However
for Gen1 subjects, the SubjectTag is their CaseID (ie, R00001.00), with
&quot;00&quot; appended.  This manipulation is necessary to identify subjects
uniquely in inter-generational datasets.  A Gen1 subject with an ID of 43
has a <code>SubjectTag</code> of 4300.  The SubjectTags of her four children
remain 4301, 4302, 4303, and 4304.
</p>
<p>Level 5 of <code>RelationshipPath</code> (ie, AuntNiece) is gender neutral.  The
relationship could be either Aunt-Niece, Aunt-Nephew, Uncle-Niece, or
Uncle-Nephew.  If there's a widely-accepted gender-neutral term, please
tell me.
</p>
<p>An extended family with <code class="reqn">k</code> subjects will have
<code class="reqn">k</code>(<code class="reqn">k</code>-1)/2 rows.  Typically, Subject1 is older while Subject2 is
younger.
</p>
<p>MZ twins have <em>R</em>=1.  DZ twins and full-siblings have <em>R</em>=.5.
Half-siblings have <em>R</em>=.25. Typical first cousins have <em>R</em>=.125.
Unrelated subjects have <em>R</em>=0 (this occasionally happens for
<code>Gen1Housemates</code>, but never for the other paths).
Other <em>R</em> coefficients are possible.
</p>
<p>There are several other uncommon possibilities, such as half-cousins (<em>R</em>=.0625) and
ambiguous aunt-nieces (<em>R</em>=.125, which is an average of 1/4 and 0/4).
The variable coding for genetic relatedness,<code>R</code>, in <code><a href="#topic+Links79Pair">Links79Pair</a></code> contains
only the common values of <em>R</em> whose groups are likely to have stable estimates.
However the variable <code>RFull</code> in <code><a href="#topic+Links79PairExpanded">Links79PairExpanded</a></code> contains all <em>R</em> values.
We strongly recommend using <code>R</code> in this <a href="base.html#topic+data.frame">base::data.frame</a>.  Move to
<code>RFull</code> (or some combination) only if you have a good reason, and are willing
to carefully monitor a variety of validity checks.  Some of these
excluded groups are too small to be estimated reliably.
</p>
<p>Furthermore, some of these groups have members who are more strongly genetically related than their
items would indicate. For instance, there are 41 Gen1 pairs who explicitly claim they are not biologically related
(<em>ie</em>, <code>RExplicit</code>=0), yet their correlation for Adult Height is <em>r</em>=0.24.  This is
much higher than would be expected for two people sampled randomly; it is nearly identical to
the <em>r</em>=0.26 we observed among the 268 Gen1 half-sibling pairs who claim they share exactly 1
biological parent.
</p>
<p>The <code>LinksPair79</code> dataset contains columns necessary for a
basic BG analysis.  The <code>Links79PairExpanded</code> dataset contains
further information that might be useful in more complicated BG analyses.
</p>
<p>A tutorial that produces a similar dataset is
http://www.nlsinfo.org/childya/nlsdocs/tutorials/linking_mothers_and_children/linking_mothers_and_children_tutorial.html.
It provides examples in SAS, SPSS, and STATA.
</p>
<p><code>RelationshipPath</code> variable.  Code written using this dataset should
NOT assume it contains only Gen2 sibling pairs.  See below for an example of
filtering the relationship category in the in <code>Links79Pair</code>
documentation.
</p>
<p>The specific steps to determine the <em>R</em> coefficient will be described
in an upcoming publication.  The following information may influence the
decisions of an applied researcher.
</p>
<p>A distinction is made between <code>Explicit</code> and <code>Implicit</code> information.
Explicit information comes from survey items that directly address the
subject's relationships.  For instance in 2006, surveys asked if the
sibling pair share the same biological father (eg, Y19940.00 and
T00020.00).  Implicit information comes from items where the subject
typically isn't aware that their responses may be used to determine genetic
relatedness.  For instance, if two siblings have biological fathers with
the same month of death (eg, R37722.00 and R37723.00), it may be reasonable
to assume they share the same biological father.
</p>
<p><code>Interpolation</code> is our lingo when other siblings are used to leverage
insight into the current pair.  For example, assume Subject 101, 102, and
103 have the same mother.  Further assume 101 and 102 report they share a
biological father, and that 101 and 103 share one too.  Finally, assume
that we don't have information about the relationship between 102 and 103.
If we are comfortable with our level of uncertainty of these
determinations, then we can interpolate/infer that 102 and 103 are
full-siblings as well.
</p>
<p>The math and height scores are duplicated from
<a href="#topic+ExtraOutcomes79">ExtraOutcomes79</a>, but are included here to make some examples
more concise and accessible.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>Source</h3>

<p>Gen1 information comes from the Summer 2013 release of the
<a href="https://www.nlsinfo.org/content/cohorts/nlsy79">NLSY79 sample</a>.  Gen2 information
comes from the Summer 2013 release of the
<a href="https://www.nlsinfo.org/content/cohorts/nlsy79-children">NLSY79 Children and Young Adults</a>.
Data were extracted with the NLS Investigator
(https://www.nlsinfo.org/investigator/).
</p>
<p>The internal version for the links is <code>Links2011V84</code>.
</p>


<h3>References</h3>

<p>The NLSY79 variable HHID (ie, R00001.49) is the source for the
<code>ExtendedID</code> variable.  This is discussed at
http://www.nlsinfo.org/nlsy79/docs/79html/79text/hhcomp.htm.
</p>
<p>For more information on <em>R</em> (<em>ie</em>, the Relatedness coefficient), please see
Rodgers, Joseph Lee, &amp; Kohler, Hans-Peter (2005).
<a href="https://pubmed.ncbi.nlm.nih.gov/15685433/">Reformulating and simplifying the DF analysis model.</a>
<em>Behavior Genetics, 35</em> (2), 211-217.
</p>


<h3>See Also</h3>

<p>The <code>LinksPair79</code> dataset contains columns necessary for a
basic BG analysis.  The <a href="#topic+Links79PairExpanded">Links79PairExpanded</a> dataset contains
further information that might be useful in more complicated BG analyses.
</p>
<p>A tutorial that produces a similar dataset is
http://www.nlsinfo.org/childya/nlsdocs/tutorials/linking_mothers_and_children/linking_mothers_and_children_tutorial.html.
It provides examples in SAS, SPSS, and STATA.
</p>
<p>The current dataset (ie, <code>Links79Pair</code>) can be saved as a CSV file
(comma-separated file) and imported into in other programs and languages.
In the R console, type the following two lines of code:
</p>
<p><code>library(NlsyLinks)</code>
</p>
<p><code>write.csv(Links79Pair, "C:/BGDirectory/Links79Pair.csv")</code>
</p>
<p>where <code>"C:/BGDirectory/"</code> is replaced by your preferred directory.
Remember to use forward slashes instead of backslashes; for instance, the
path <code>"C:\BGDirectory\Links79Pair.csv"</code> can be misinterpreted.
</p>
<p><strong>Download CSV</strong>
If you're using the NlsyLinks package in R, the dataset is automatically available.
To use it in a different environment,
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-79/links-2017-79.csv?raw=true">download the csv</a>,
which is readable by all statistical software.
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-79/links-metadata-2017-79.yml">links-metadata-2017-79.yml</a>
documents the dataset version information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.
summary(Links79Pair) # Summarize the five variables.
hist(Links79Pair$R) # Display a histogram of the Relatedness coefficients.
table(Links79Pair$R) # Create a table of the Relatedness coefficients for the whole sample.

# Create a dataset of only Gen2 sibs, and display the distribution of R.
gen2Siblings &lt;- subset(Links79Pair, RelationshipPath == "Gen2Siblings")
table(gen2Siblings$R) # Create a table of the Relatedness coefficients for the Gen2 sibs.

</code></pre>

<hr>
<h2 id='Links79PairExpanded'>Kinship linking file for pairs of relatives.  It builds upon the <a href="#topic+Links79Pair">Links79Pair</a> dataset.</h2><span id='topic+Links79PairExpanded'></span>

<h3>Description</h3>

<p>Please first read the documentation for <a href="#topic+Links79Pair">Links79Pair</a>.  That
dataset contains the same pairs/rows, but only a subset of the
variables/columns.
</p>
<p>NOTE: In Nov 2013, the variable naming scheme changed in order to be more consistent across variables.  For variables
that are measured separately for both subjects (eg, Gender), the subjects' variable name will have an <code style="white-space: pre;">&#8288;_S1&#8288;</code> or <code style="white-space: pre;">&#8288;_S2&#8288;</code>
appended to it.  For instance, the variables <code>LastSurvey_S1</code> and <code>LastSurvey_S2</code> correspond to the last surveys completed
by the pair's first and second subject, respectively.  Similarly, the functions <code><a href="#topic+CreatePairLinksDoubleEntered">CreatePairLinksDoubleEntered()</a></code> and
<code><a href="#topic+CreatePairLinksSingleEntered">CreatePairLinksSingleEntered()</a></code> now by default append <code style="white-space: pre;">&#8288;_S1&#8288;</code> and <code style="white-space: pre;">&#8288;_S2&#8288;</code>, instead of <code style="white-space: pre;">&#8288;_1&#8288;</code> and <code style="white-space: pre;">&#8288;_2&#8288;</code>.  However this can be
modified using the 'subject1Qualifier' and 'subject2Qualifier' parameters.
</p>


<h3>Format</h3>

<p>A data frame with 11,075 observations on the following 22 variables.
There is one row per unique pair of subjects, irrespective of order.
</p>

<ul>
<li> <p><strong>ExtendedID</strong> see the variable of the same name in <a href="#topic+Links79Pair">Links79Pair</a>
</p>
</li>
<li> <p><strong>SubjectTag_S1</strong> see the variable of the same name in <a href="#topic+Links79Pair">Links79Pair</a>
</p>
</li>
<li> <p><strong>SubjectTag_S2</strong> see the variable of the same name in <a href="#topic+Links79Pair">Links79Pair</a>
</p>
</li>
<li> <p><strong>R</strong> see the variable of the same name in <a href="#topic+Links79Pair">Links79Pair</a>
</p>
</li>
<li> <p><strong>RFull</strong> This is a superset of <code>R</code>.  This includes all the <em>R</em> values we estimated, while <code>R</code> (i.e., the variable above) excludes values like <em>R</em>=0 for <code>Gen1Housemates</code>, and the associated relationships based on this <em>R</em> value (i.e., <code>Gen2Cousin</code>s and <code>AuntNiece</code>s).
</p>
</li>
<li> <p><strong>RelationshipPath</strong> see the variable of the same name in <a href="#topic+Links79Pair">Links79Pair</a>
</p>
</li>
<li> <p><strong>EverSharedHouse</strong> Indicate if the pair likely live in the same house.  This is <code>TRUE</code> for <code>Gen1Housemates</code>, <code>Gen2Siblings</code>, and <code>ParentChild</code>. This is <code>FALSE</code> for <code>AuntNiece</code> and <code>Gen2Cousins</code>
</p>
</li>
<li> <p><strong>IsMz</strong> Indicates if the pair is from the same zygote (ie, they are identical twins/triplets). This variable is a factor, with levels <code>No</code>=0, <code>Yes</code>=1, <code>DoNotKnow</code>=255.
</p>
</li>
<li> <p><strong>LastSurvey_S1</strong> The year of Subject1's most recently completed survey. This may be different that the survey's administration date.
</p>
</li>
<li> <p><strong>LastSurvey_S2</strong> The year of Subject2's most recently completed survey. This may be different that the survey's administration date.
</p>
</li>
<li> <p><strong>RImplicitPass1</strong> The pair's <em>R</em> coefficient, using only implicit information.  Interpolation was NOT used.
</p>
</li>
<li> <p><strong>RImplicit</strong> The pair's <em>R</em> coefficient, using only implicit information.  Interpolation was used.
</p>
</li>
<li> <p><strong>RImplicit2004</strong> The pair's <em>R</em> coefficient released in our previous projects (<strong>need reference</strong>).  This variable is provided primarily for previous users wishing to replicate previous analyses.
</p>
</li>
<li> <p><strong>RExplicitPass1</strong> The pair's <em>R</em> coefficient, using only explicit information.  Interpolation was NOT used.
</p>
</li>
<li> <p><strong>RExplicit</strong> The pair's <em>R</em> coefficient, using only explicit information.  Interpolation was used.
</p>
</li>
<li> <p><strong>RExplicitOlderSibVersion</strong> The pair's <em>R</em> coefficient, according to the explicit item responses of the older sibling.
</p>
</li>
<li> <p><strong>RExplicitYoungerSibVersion</strong> The pair's <em>R</em> coefficient, according to the explicit item responses of the younger sibling.
</p>
</li>
<li> <p><strong>RPass1</strong> The pair's estimated <em>R</em> coefficient, using both implicit and explicit information.  Interpolation was NOT used.  The variable <code>R</code> is identically constructed, but it did use interpolation.
</p>
</li>
<li> <p><strong>Generation_S1</strong> The generation of the first subject.  Values for Gen1 and Gen2 are <code>1</code> and <code>2</code>, respectively.
</p>
</li>
<li> <p><strong>Generation_S2</strong> The generation of the second subject.  Values for Gen1 and Gen2 are <code>1</code> and <code>2</code>, respectively.
</p>
</li>
<li> <p><strong>SubjectID_S1</strong> The ID value assigned by NLS to the first subject.  For Gen1 Subjects, this is their &quot;CaseID&quot; (ie, R00001.00).  For Gen2 subjects, this is their &quot;CID&quot; (ie, C00001.00).
</p>
</li>
<li> <p><strong>SubjectID_S2</strong> The ID value assigned by NLS to the second subject.
</p>
</li>
<li> <p><strong>MathStandardized_S1</strong> The PIAT-Math score for Subject1.  See <a href="#topic+ExtraOutcomes79">ExtraOutcomes79</a> for more information about its source.
</p>
</li>
<li> <p><strong>MathStandardized_S2</strong> The PIAT-Math score for Subject2.
</p>
</li>
<li> <p><strong>HeightZGenderAge_S1</strong> The early adult height for Subject1.  See <a href="#topic+ExtraOutcomes79">ExtraOutcomes79</a> for more information about its source.
</p>
</li>
<li> <p><strong>HeightZGenderAge_S2</strong> The early adult height for Subject2.
</p>
</li></ul>



<h3>Details</h3>

<p>Specifies the relatedness coefficient (ie, '<em>R</em>') between subjects in the
same extended family.  Each row represents a unique relationship pair.  An
extended family with <code class="reqn">k</code> subjects will have <code class="reqn">k</code>(<code class="reqn">k</code>-1)/2 rows.
Typically, Subject1 is older while Subject2 is younger.
</p>
<p><code>RelationshipPath</code> variable.  Code written using this dataset should
NOT assume it contains only Gen2 sibling pairs.  See an example of
filtering the relationship category in the in <a href="#topic+Links79Pair">Links79Pair</a>
documentation.
</p>
<p>The specific steps to determine the <em>R</em> coefficient will be described
in an upcoming publication.  The following information may influence the
decisions of an applied researcher.
</p>
<p>A distinction is made between 'Explicit' and 'Implicit' information.
Explicit information comes from survey items that directly address the
subject's relationships.  For instance in 2006, surveys asked if the
sibling pair share the same biological father (eg, Y19940.00 and
T00020.00).  Implicit information comes from items where the subject
typically isn't aware that their responses may be used to determine genetic
relatedness.  For instance, if two siblings have biological fathers with
the same month of death (eg, R37722.00 and R37723.00), it may be reasonable
to assume they share the same biological father.
</p>
<p>'Interpolation' is our lingo when other siblings are used to leverage
insight into the current pair.  For example, assume Subject 101, 102, and
103 have the same mother.  Further assume 101 and 102 report they share a
biological father, and that 101 and 103 share one too.  Finally, assume
that we don't have information about the relationship between 102 and 103.
If we are comfortable with our level of uncertainty of these
determinations, then we can interpolate/infer that 102 and 103 are
full-siblings as well.
</p>
<p>The math and height scores are duplicated from
<a href="#topic+ExtraOutcomes79">ExtraOutcomes79</a>, but are included here to make some examples
more concise and accessible.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>Source</h3>

<p>See <a href="#topic+Links79Pair">Links79Pair</a>.
</p>


<h3>See Also</h3>

<p><strong>Download CSV</strong>
If you're using the NlsyLinks package in R, the dataset is automatically available.
To use it in a different environment,
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-79/links-2017-79.csv?raw=true">download the csv</a>,
which is readable by all statistical software.
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-79/links-metadata-2017-79.yml">links-metadata-2017-79.yml</a>
documents the dataset version information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.
# olderR   &lt;- Links79PairExpanded$RExplicitOlderSibVersion   # Declare a concise variable name.
# youngerR &lt;- Links79PairExpanded$RExplicitYoungerSibVersion # Declare a concise variable name.

# plot(jitter(olderR), jitter(youngerR))  # Scatterplot the siblings' responses.
# table( youngerR, olderR)  # Table of the relationship between the siblings' responses.
# ftable(youngerR, olderR, dnn=c("Younger's Version", "Older's Version")) # A formatted table.

# write.csv(
#   Links79PairExpanded,
#   file      ='~/NlsyLinksStaging/Links79PairExpanded.csv',
#   row.names = FALSE
# )

</code></pre>

<hr>
<h2 id='Links97Pair'>Kinship linking file for pairs of relatives in the NLSY97</h2><span id='topic+Links97Pair'></span>

<h3>Description</h3>

<p>This dataset specifies the relatedness coefficient (ie, '<code>R</code>') between
subjects in the same extended family.  Each row represents a unique
relationship pair.
</p>
<p>NOTE: Two variable names changed in November 2013. <code>Subject1Tag</code> and <code>Subject2Tag</code> became <code>SubjectTag_S1</code> and <code>SubjectTag_S2</code>.
</p>


<h3>Format</h3>

<p>A data frame with 2,519 observations on the following 5 variables.
There is one row per unique pair of subjects, irrespective of order.
</p>

<ul>
<li> <p><strong>ExtendedID</strong> Identity of the extended family of the pair; it corresponds to the HHID in the NLSY97.  See References below.
</p>
</li>
<li> <p><strong>SubjectTag_S1</strong> Identity of the pair's first subject.  See Details below.
</p>
</li>
<li> <p><strong>SubjectTag_S2</strong> Identity of the pair's second subject.  See Details below.
</p>
</li>
<li> <p><strong>R</strong> The pair's Relatedness coefficient.  See Details below.
</p>
</li>
<li> <p><strong>RelationshipPath</strong> Specifies the relationship category of the pair.  This variable is a factor, with level <code>Housemates</code>=1.
</p>
</li></ul>



<h3>Details</h3>

<p>The variable <code>ExtendedID</code> corresponds to the NLSY97 variable <code style="white-space: pre;">&#8288;[SIDCODE]&#8288;</code>
(<em>e.g.</em>, <a href="https://www.nlsinfo.org/investigator/pages/search.jsp#R11930.00">R11930.00</a>),
which uniquely identifies a <em>household</em> that may contain multiple NLSY97 subjects.
</p>
<p>The variables <code>SubjectTag_S1</code> and <code>SubjectTag_S2</code> uniquely identify
subjects.  It corresponds to the NLSY97 variable <code style="white-space: pre;">&#8288;[PUBID]&#8288;</code>,
(<em>e.g.</em>, <a href="https://www.nlsinfo.org/investigator/pages/search.jsp#R00001.00">R00001.00</a>).
</p>
<p>The <code>RelationshipPath</code> variable is not useful with this dataset,
but is included to be consistent with the <a href="#topic+Links97Pair">Links97Pair</a> dataset.
</p>
<p>An extended family with <code class="reqn">k</code> subjects will have
<code class="reqn">k</code>(<code class="reqn">k</code>-1)/2 rows.  Typically, Subject1 is older while Subject2 is
younger.
</p>
<p>MZ twins have <em>R</em>=1.  DZ twins and full-siblings have <em>R</em>=.5.
Half-siblings have <em>R</em>=.25. Typical first cousins have <em>R</em>=.125.
Unrelated subjects have <em>R</em>=0 (this occasionally happens for
<code>Housemates</code>, but never for the other paths).
Other <em>R</em> coefficients are possible.
</p>
<p>There are several other uncommon possibilities, such as half-cousins (<em>R</em>=.0625) and
ambiguous aunt-nieces (<em>R</em>=.125, which is an average of 1/4 and 0/4).
The variable coding for genetic relatedness,<code>R</code>, in <code><a href="#topic+Links97Pair">Links97Pair</a></code> contains
only the common values of <em>R</em> whose groups are likely to have stable estimates.
However the variable <code>RFull</code> in <code><a href="#topic+Links97PairExpanded">Links97PairExpanded</a></code> contains all <em>R</em> values.
We strongly recommend using <code>R</code> in this <a href="base.html#topic+data.frame">base::data.frame</a>.  Move to
<code>RFull</code> (or some combination) only if you have a good reason, and are willing
to carefully monitor a variety of validity checks.  Some of these
excluded groups are too small to be estimated reliably.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>Source</h3>

<p>Information comes from the Summer 2018 release of the
<a href="https://www.nlsinfo.org/content/cohorts/nlsy97">NLSY97 sample</a>.
Data were extracted with the NLS Investigator
(https://www.nlsinfo.org/investigator/).
</p>


<h3>References</h3>

<p>For more information on <em>R</em> (<em>ie</em>, the Relatedness coefficient), please see
Rodgers, Joseph Lee, &amp; Kohler, Hans-Peter (2005).
<a href="https://pubmed.ncbi.nlm.nih.gov/15685433/">Reformulating and simplifying the DF analysis model.</a>
<em>Behavior Genetics, 35</em> (2), 211-217.
</p>


<h3>See Also</h3>

<p>The <code>LinksPair97</code> dataset contains columns necessary for a
basic BG analysis.  The <a href="#topic+Links97PairExpanded">Links97PairExpanded</a> dataset contains
further information that might be useful in more complicated BG analyses.
</p>
<p>A tutorial that produces a similar dataset is
http://www.nlsinfo.org/childya/nlsdocs/tutorials/linking_mothers_and_children/linking_mothers_and_children_tutorial.html.
It provides examples in SAS, SPSS, and STATA.
</p>
<p>The current dataset (ie, <code>Links97Pair</code>) can be saved as a CSV file
(comma-separated file) and imported into in other programs and languages.
In the R console, type the following two lines of code:
</p>
<p><code>library(NlsyLinks)</code>
<code>write.csv(Links97Pair, "C:/BGDirectory/Links97Pair.csv")</code>
</p>
<p>where <code>"C:/BGDirectory/"</code> is replaced by your preferred directory.
Remember to use forward slashes instead of backslashes; for instance, the
path <code>"C:\BGDirectory\Links97Pair.csv"</code> can be misinterpreted.
</p>
<p><strong>Download CSV</strong>
If you're using the NlsyLinks package in R, the dataset is automatically available.
To use it in a different environment,
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-97/links-2017-97.csv?raw=true">download the csv</a>,
which is readable by all statistical software.
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-97/links-metadata-2017-97.yml">links-metadata-2017-97.yml</a>
documents the dataset version information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.
summary(Links97Pair) # Summarize the five variables.
hist(Links97Pair$R) # Display a histogram of the Relatedness coefficients.
table(Links97Pair$R) # Create a table of the Relatedness coefficients for the whole sample.

# Create a dataset of only monozygotic sibs.
mz_sibs &lt;- subset(Links97Pair, R &gt; .9)
summary(mz_sibs) # Create a table MZ sibs.

</code></pre>

<hr>
<h2 id='Links97PairExpanded'>Kinship linking file for pairs of relatives In the NLSY97.  It builds upon the <a href="#topic+Links97Pair">Links97Pair</a> dataset.</h2><span id='topic+Links97PairExpanded'></span>

<h3>Description</h3>

<p>Please first read the documentation for <a href="#topic+Links97Pair">Links97Pair</a>.  That
dataset contains the same pairs/rows, but only a subset of the
variables/columns.
</p>
<p>For variables
that are measured separately for both subjects (eg, Gender), the subjects' variable name will have an <code style="white-space: pre;">&#8288;_S1&#8288;</code> or <code style="white-space: pre;">&#8288;_S2&#8288;</code>
appended to it.  For instance, the variables <code>LastSurvey_S1</code> and <code>LastSurvey_S2</code> correspond to the last surveys completed
by the pair's first and second subject, respectively.  Similarly, the functions <code><a href="#topic+CreatePairLinksDoubleEntered">CreatePairLinksDoubleEntered()</a></code> and
<code><a href="#topic+CreatePairLinksSingleEntered">CreatePairLinksSingleEntered()</a></code> by default append <code style="white-space: pre;">&#8288;_S1&#8288;</code> and <code style="white-space: pre;">&#8288;_S2&#8288;</code>.  However this can be
modified using the 'subject1Qualifier' and 'subject2Qualifier' parameters.
</p>


<h3>Format</h3>

<p>A data frame with 11,075 observations on the following 22 variables.
There is one row per unique pair of subjects, irrespective of order.
</p>

<ul>
<li> <p><strong>ExtendedID</strong> see the variable of the same name in <a href="#topic+Links97Pair">Links97Pair</a>
</p>
</li>
<li> <p><strong>SubjectTag_S1</strong> see the variable of the same name in <a href="#topic+Links97Pair">Links97Pair</a>
</p>
</li>
<li> <p><strong>SubjectTag_S2</strong> see the variable of the same name in <a href="#topic+Links97Pair">Links97Pair</a>
</p>
</li>
<li> <p><strong>R</strong> see the variable of the same name in <a href="#topic+Links97Pair">Links97Pair</a>
</p>
</li>
<li> <p><strong>RFull</strong> This is a superset of <code>R</code>.  This includes all the <em>R</em> values we estimated, while <code>R</code> (i.e., the variable above) excludes values like <em>R</em>=0.
</p>
</li>
<li> <p><strong>RelationshipPath</strong> see the variable of the same name in <a href="#topic+Links97Pair">Links97Pair</a>
</p>
</li>
<li> <p><strong>EverSharedHouse</strong> Indicate if the pair likely live in the same house.  This is <code>TRUE</code> for all pairs in this NLSY97 dataset.
</p>
</li>
<li> <p><strong>IsMz</strong> Indicates if the pair is from the same zygote (ie, they are identical twins/triplets). This variable is a factor, with levels <code>No</code>=0, <code>Yes</code>=1, <code>DoNotKnow</code>=255.
</p>
</li>
<li> <p><strong>LastSurvey_S1</strong> The year of Subject1's most recently completed survey. This may be different that the survey's administration date.
</p>
</li>
<li> <p><strong>LastSurvey_S2</strong> The year of Subject2's most recently completed survey. This may be different that the survey's administration date.
</p>
</li>
<li> <p><strong>RPass1</strong> The pair's estimated <em>R</em> coefficient, using both implicit and explicit information.  Interpolation was NOT used.  The variable <code>R</code> is identically constructed, but it did use interpolation.
</p>
</li>
<li> <p><strong>SubjectID_S1</strong> The ID value assigned by NLS to the first subject.  For Gen1 Subjects, this is their &quot;CaseID&quot; (ie, R00001.00).  For Gen2 subjects, this is their &quot;CID&quot; (ie, C00001.00).
</p>
</li>
<li> <p><strong>SubjectID_S2</strong> The ID value assigned by NLS to the second subject.
</p>
</li></ul>



<h3>Details</h3>

<p>Specifies the relatedness coefficient (ie, '<em>R</em>') between subjects in the
same extended family.  Each row represents a unique relationship pair.  An
extended family with <code class="reqn">k</code> subjects will have <code class="reqn">k</code>(<code class="reqn">k</code>-1)/2 rows.
Typically, Subject1 is older while Subject2 is younger.
</p>
<p>The specific steps to determine the <em>R</em> coefficient will be described
in an upcoming publication.  The following information may influence the
decisions of an applied researcher.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>Source</h3>

<p>See <a href="#topic+Links97Pair">Links97Pair</a>.
</p>


<h3>See Also</h3>

<p><strong>Download CSV</strong>
If you're using the NlsyLinks package in R, the dataset automatically is available.
To use it in a different environment,
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-97/links-2017-97.csv?raw=true">download the csv</a>,
which is readable by all statistical software.
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-97/links-metadata-2017-97.yml">links-metadata-2017-97.yml</a>
documents the dataset version information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.
hist(Links97PairExpanded$R) # Declare a concise variable name.

# write.csv(
#   Links97PairExpanded,
#   file      ='~/NlsyLinksStaging/Links97PairExpanded.csv',
#   row.names = FALSE
# )

</code></pre>

<hr>
<h2 id='ReadCsvNlsy79'>Read a CSV file downloaded from the NLS Investigator</h2><span id='topic+ReadCsvNlsy79'></span><span id='topic+ReadCsvNlsy79Gen1'></span><span id='topic+ReadCsvNlsy79Gen2'></span>

<h3>Description</h3>

<p>The function accepts a (file path to) CSV file and creates a <a href="base.html#topic+data.frame">base::data.frame</a>.  The <a href="base.html#topic+data.frame">base::data.frame</a> is modified and augmented with columns to assist later routines.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadCsvNlsy79Gen1(filePath, dsExtract = utils::read.csv(filePath))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadCsvNlsy79_+3A_filepath">filePath</code></td>
<td>
<p>A path to the CSV file. Remember to use double back-slashes in Windows, or forward-slashes in Windows or Linux.</p>
</td></tr>
<tr><td><code id="ReadCsvNlsy79_+3A_dsextract">dsExtract</code></td>
<td>
<p>A 'data.frame' (containing the extract) can be passed instead of the file path if the data has already been read into R's memory.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function does seven things.
</p>

<ol>
<li><p> Reads the CSV into a <a href="base.html#topic+data.frame">base::data.frame</a>.
</p>
</li>
<li><p> Checks that the NLSY variables <code>C00001.00</code> and <code>C00002.00</code> exist in the <a href="base.html#topic+data.frame">base::data.frame</a>.
</p>
</li>
<li><p> The NLSY variable <code>C00001.00</code> is renamed <code>SubjectID</code>.
</p>
</li>
<li><p> A variable named <code>Generation</code> is given a value of 2 for all subjects.
</p>
</li>
<li><p> The <code>SubjectTag</code> variable is created.
</p>
</li>
<li><p> The NLSY variable <code>C00002.00</code> is multiplied by 100 and renamed <code>SubjectTagOfMother</code>.
</p>
</li>
<li><p> The NLSY variable <code>R00001.49</code> (ie, their Mother's <code>HHID</code> is attached to each Gen2 record).
</p>
</li></ol>



<h3>Value</h3>

<p>A <a href="base.html#topic+data.frame">base::data.frame</a> to facilitate biometric analysis.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
filePathGen2 &lt;- "~/Nlsy/Datasets/gen2-birth.csv"
ds &lt;- ReadCsvNlsy79Gen2(filePath = filePathGen2)

## End(Not run)
</code></pre>

<hr>
<h2 id='RGroupSummary'>Calculates summary statistics for each <em>R</em>elatedness Group in the sample.</h2><span id='topic+RGroupSummary'></span>

<h3>Description</h3>

<p>Before and after running ACE Models, it is important to examine the characteristics of the different groups.  When the ACE is estimated with an SEM using multiple groups, it is even even more important.  Groups may contain too few subjects to have a well-behaved covariance matrix.
</p>
<p>If a group's covariance matrix is not Positive Definite (or it's misbehaving in some other way), it's typically recommended to exclude that group from the SEM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RGroupSummary(
  ds,
  oName_S1,
  oName_S2,
  rName = "R",
  determinantThreshold = 1e-05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RGroupSummary_+3A_ds">ds</code></td>
<td>
<p>The <a href="base.html#topic+data.frame">base::data.frame</a> containing the following variables:</p>
</td></tr>
<tr><td><code id="RGroupSummary_+3A_oname_s1">oName_S1</code></td>
<td>
<p>The name of the outcome variable corresponding to the first subject in the pair.</p>
</td></tr>
<tr><td><code id="RGroupSummary_+3A_oname_s2">oName_S2</code></td>
<td>
<p>The name of the outcome variable corresponding to the first subject in the pair.</p>
</td></tr>
<tr><td><code id="RGroupSummary_+3A_rname">rName</code></td>
<td>
<p>The name of the variable specifying the pair's <code>Relatedness</code> coefficient.</p>
</td></tr>
<tr><td><code id="RGroupSummary_+3A_determinantthreshold">determinantThreshold</code></td>
<td>
<p>The minimum value the covariance matrix's determinant (for the group) should exceed to be considered Positive Definite.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function isn't specific to an ACE model and groups defined by <code>R</code>.  It could be applied to any multiple-group SEM with two manifest/outcome variables.  In the future, we may generalize it beyond two manifest variables.
</p>
<p>To get summary stats for the entire sample, create a dummy indicator variable that assigns everyone to the same group.  See the second example below.
</p>
<p>The default determinantThreshold value is nonzero, in order to forgive slight numerical inaccuracies caused by fixed-precision arithmetic.
</p>


<h3>Value</h3>

<p>A <a href="base.html#topic+data.frame">base::data.frame</a> with one row per group.  The <a href="base.html#topic+data.frame">base::data.frame</a> contains the following variables:
</p>

<ul>
<li> <p><strong>R</strong> The group's <code>R</code> value.  Note the name of this variable can be changed by the user, by specifying a non-default value to the <code>rName</code> argument.
</p>
</li>
<li> <p><strong>Included</strong> Indicates if the group should be included in a multiple-group SEM.
</p>
</li>
<li> <p><strong>PairCount</strong> The number of pairs in the group with <em>complete</em> data for <code>R</code> and the two outcome/manifest variables.
</p>
</li>
<li> <p><strong>O1Mean</strong> The mean (of the outcome variable) among the group's first members, excluding the missing values.
</p>
</li>
<li> <p><strong>O2Mean</strong> The mean (of the outcome variable) among the group's second members, excluding the missing values.
</p>
</li>
<li> <p><strong>O1Variance</strong> The variance (of the outcome variable) among the group's first members.
</p>
</li>
<li> <p><strong>O2Variance</strong> The variance (of the outcome variable) among the group's second members.
</p>
</li>
<li> <p><strong>O1O2Covariance</strong> The covariance (of the outcome variable) across the group's first and second members.
</p>
</li>
<li> <p><strong>Correlation</strong> The correlation (of the outcome variable) across the group's first and second members.
</p>
</li>
<li> <p><strong>Determinant</strong> The determinant of the group's covariance matrix.
</p>
</li>
<li> <p><strong>PosDefinite</strong> Indicates if the group's covariance matrix is positive definite.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Will Beasley and David Bard
</p>


<h3>References</h3>

<p>Please see <a href="http://ibgwww.colorado.edu/workshop2006/cdrom/HTML/book2004a.pdf">Neale &amp; Maes</a> for more information about SEM with multiple groups.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.
dsLinks &lt;- Links79PairExpanded # Load the dataset from the NlsyLinks package.
dsLinks &lt;- dsLinks[dsLinks$RelationshipPath == "Gen2Siblings", ]
oName_S1 &lt;- "MathStandardized_S1" # Stands for Outcome1
oName_S2 &lt;- "MathStandardized_S2" # Stands for Outcome2
dsGroupSummary &lt;- RGroupSummary(dsLinks, oName_S1, oName_S2)
dsGroupSummary

# Should return:
#      R Included PairCount   O1Mean   O2Mean O1Variance O2Variance O1O2Covariance Correlation
# 1 0.250     TRUE      2718  94.6439  95.5990    169.650    207.842        41.0783    0.218761
# 2 0.375     TRUE       139  92.6043  93.1655    172.531    187.081        40.4790    0.225311
# 3 0.500     TRUE      5511  99.8940 100.1789    230.504    232.971       107.3707    0.463336
# 4 0.750    FALSE         2 108.5000 106.0000    220.500     18.000        63.0000    1.000000
# 5 1.000     TRUE        22  98.6364  95.5455    319.195    343.117       277.5887    0.838789
#  Determinant PosDefinite
# 1     33573.0        TRUE
# 2     30638.7        TRUE
# 3     42172.2        TRUE
# 4         0.0       FALSE
# 5     32465.6        TRUE

# To get summary stats for the whole sample, create one large inclusive group.
dsLinks$Dummy &lt;- 1
(dsSampleSummary &lt;- RGroupSummary(dsLinks, oName_S1, oName_S2, rName = "Dummy"))

# Should return:
#  Dummy Included PairCount   O1Mean   O2Mean O1Variance O2Variance O1O2Covariance
# 1     1     TRUE      8392 98.07162 98.56864    216.466   229.2988       90.90266
#  Correlation Determinant PosDefinite
# 1   0.4080195     41372.1        TRUE
###
### ReadCsvNlsy79
###
## Not run: 
filePathGen2 &lt;- "~/Nlsy/Datasets/gen2-birth.csv"
ds &lt;- ReadCsvNlsy79Gen2(filePath = filePathGen2)

## End(Not run)

</code></pre>

<hr>
<h2 id='SubjectDetails79'>Dataset containing further details of the Gen1 and Gen2 subjects.</h2><span id='topic+SubjectDetails79'></span>

<h3>Description</h3>

<p>These variables are useful to many types of analyses (not just behavior genetics), and are provided to save users time.
</p>


<h3>Format</h3>

<p>A data frame with 24,181 observations on the following 12
variables.
</p>

<ul>
<li> <p><strong>SubjectTag</strong> see the variable of the same name in <a href="#topic+Links79Pair">Links79Pair</a>
</p>
</li>
<li> <p><strong>ExtendedID</strong> see the variable of the same name in <a href="#topic+Links79Pair">Links79Pair</a>
</p>
</li>
<li> <p><strong>Generation</strong> Indicates if the subject is in generation <code>1</code> or <code>2</code>.
</p>
</li>
<li> <p><strong>Gender</strong> Indicates if the subject is <code>Male</code> or <code>Female</code>.
</p>
</li>
<li> <p><strong>RaceCohort</strong> Indicates if the race cohort is <code>Hispanic</code>, <code>Black</code> or <code>Nbnh</code> (<em>ie</em>, Non-black, non-hispanic).  This comes from the Gen1 variable <code>R02147.00</code> and Gen2 variable <code>C00053.00</code>.
</p>
</li>
<li> <p><strong>SiblingCountInNls</strong> The number of the subject's siblings, including himself/herself (a singleton has a value of one).  This considers only the siblings in the NLSY.  For Gen1, this can exclude anyone outside the age range.  For Gen2, this excludes anyone who doesn't share the same mother.
</p>
</li>
<li> <p><strong>BirthOrderInNls</strong> Indicates the subject's birth order among the NLSY siblings.
</p>
</li>
<li> <p><strong>SimilarAgeCount</strong> The number of children who were born within roughly 30 days of the subject's birthday, including the subject (for instance, even an only child will have a value of 1).  For Gen2 subjects, this should reflect how many children the Gen1 mother gave birth to at the same time (1: singleton; 2: twins, 3: triplets).  For Gen1 subjects, this is less certain, because the individual might have been living with a similarly-aged housemate, born to a different mother.
</p>
</li>
<li> <p><strong>HasMzPossibly</strong> Indicates if the subject <em>might</em> be a member of an MZ twin/triplet. This will be true if there is a sibling with a DOB within a month, and they are the same gender.
</p>
</li>
<li> <p><strong>IsMz</strong> Indicates if the subject has been identified as a member of an MZ twin/triplet.
</p>
</li>
<li> <p><strong>KidCountBio</strong> The number of biological children known to the NLSY (but not necessarily interviewed by the NLSY.
</p>
</li>
<li> <p><strong>KidCountInNls</strong> The number of children who belong to the NLSY. This is nonnull for only Gen1 subjects.
</p>
</li>
<li> <p><strong>Mob</strong> The subject's month of birth.  The exact day is not available to the public.  By default, we set their birthday to the 15th day of the month.
</p>
</li>
<li> <p><strong>LastSurveyYearCompleted</strong> The year of the most recently completed survey.
</p>
</li>
<li> <p><strong>AgeAtLastSurvey</strong> The subject's age at the most recently completed survey.
</p>
</li>
<li> <p><strong>IsDead</strong> ##This variable is not available yet## Indicates if the subject was alive for the last attempted survey.
</p>
</li>
<li> <p><strong>DeathDate</strong> ##This variable is not available yet## The subject's month of death.  The exact day is not available to the public. By default, we set their birthday to the 15th day of the month.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>Source</h3>

<p>Gen1 information comes from the Summer 2013 release of the <a href="https://www.nlsinfo.org/content/cohorts/nlsy79">NLSY79 sample</a>.  Gen2 information
comes from the Summer 2013 release of the
<a href="https://www.nlsinfo.org/content/cohorts/nlsy79-children">NLSY79 Children and Young Adults sample</a>.  Data were extracted with the NLS Investigator
(https://www.nlsinfo.org/investigator/).
</p>


<h3>See Also</h3>

<p><strong>Download CSV</strong>
If you're using the NlsyLinks package in R, the dataset is automatically available.
To use it in a different environment,
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-79/subject-details.csv?raw=true">download the csv</a>,
which is readable by all statistical software.
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-79/links-metadata-2017-79.yml">links-metadata-2017-79.yml</a>
documents the dataset version information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.

summary(SubjectDetails79)

oldPar &lt;- par(mfrow = c(3, 2), mar = c(2, 2, 1, .5), tcl = 0, mgp = c(1, 0, 0))
hist(
  SubjectDetails79$SiblingCountInNls,
  main = "",
  breaks = seq(from = 0, to = max(SubjectDetails79$SiblingCountInNls, na.rm = TRUE), by = 1)
)
hist(
  SubjectDetails79$BirthOrderInNls,
  main   = "",
  breaks = seq(from = 0, to = max(SubjectDetails79$BirthOrderInNls, na.rm = TRUE), by = 1)
)
hist(
  SubjectDetails79$SimilarAgeCount,
  main   = "",
  breaks = seq(from = 0, to = max(SubjectDetails79$SimilarAgeCount, na.rm = TRUE), by = 1)
)
hist(
  SubjectDetails79$KidCountBio,
  main   = "",
  breaks = seq(from = 0, to = max(SubjectDetails79$KidCountBio, na.rm = TRUE), by = 1)
)
hist(
  SubjectDetails79$KidCountInNls,
  main   = "",
  breaks = seq(from = 0, to = max(SubjectDetails79$KidCountInNls, na.rm = TRUE), by = 1)
)
# hist(SubjectDetails79$Mob, main="",
#     breaks=seq.Date(
#       from=min(SubjectDetails79$Mob, na.rm=TRUE),
#       to=max(SubjectDetails79$Mob, na.rm=TRUE),
#       by="year")
# )
par(oldPar)

</code></pre>

<hr>
<h2 id='Survey79'>Dataset containing survey details for each subject, for each year</h2><span id='topic+Survey79'></span>

<h3>Description</h3>

<p>Each row represents a survey that a subject completed (or didn't complete).
It can be very helpful when restructuring the NLS investigator extracts into a
longitudinal dataset that's aligned by age (instead of by survey wave).
The Age variables can help to align other response variables across subjects.
While the 'SurveySource' indicates where to look for their responses.
</p>
<p>These variables are useful to many types of analyses (not just behavior genetics), and are
provided to save users time.
</p>


<h3>Format</h3>

<p>A data frame with 580,752 observations on the following 6 variables.
</p>

<ul>
<li> <p><strong>SubjectTag</strong> see the variable of the same name in <a href="#topic+Links79Pair">Links79Pair</a>
</p>
</li>
<li> <p><strong>SurveySource</strong> The location of that subject's survey responses that year.  Values are <code>NoInterview</code>, <code>Gen1</code>, <code>Gen2C</code> or <code>Gen2YA</code>.
</p>
</li>
<li> <p><strong>SurveyYear</strong> The year/wave of the survey.
</p>
</li>
<li> <p><strong>Survey79</strong> The exact date of the administered survey.
</p>
</li>
<li> <p><strong>Age</strong> The subject's age at the time of the survey, rounded to 1 decimal.
See Details below.
</p>
</li></ul>



<h3>Details</h3>

<p><code>Age</code> uses a variable called <code>AgeCalculateYears</code> (according to a their own
response, or their mother's response) if it's available, and uses
<code>AgeSelfReportYears</code> (according to a their own response, or their mother's
response)
if not.  The values usually agree, but not always.
These two separated variables are available for download,
but withheld from the R package
(otherwise we would exceed CRAN's maximum size).
</p>
<p>The exact <em>date</em> of birth isn't public (only the subject's <em>month</em> of birth).
To balance the downward bias of two weeks, their birthday is set to the
15th day of the month to produce <code>AgeCalculateYears</code>.
</p>
<p>In the Gen2 Child dataset, self-reported age is
stated by month (eg, the child is 38 months old); a constant of 0.5 months has been added to balance the downward bias.  In the Gen2 YA and
Gen1 datasets, self-reported age is stated by year (eg, the subject is 52 years old); a constant of 0.5 years has been added.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>
<p><strong>Download CSV</strong>
If you're using the NlsyLinks package in R, the dataset is automatically available.
To use it in a different environment,
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-79/survey-79.csv?raw=true">download the csv</a>,
which is readable by all statistical software.
<a href="https://github.com/nlsy-links/NlsyLinks/blob/master/outside-data/nlsy-79/links-metadata-2017-79.yml">links-metadata-2017-79.yml</a>
documents the dataset version information.
</p>


<h3>Source</h3>

<p>Gen1 information comes from the Summer 2013 release of the <a href="https://www.nlsinfo.org/content/cohorts/nlsy79">NLSY79 sample</a>.  Gen2 information
comes from the January 2015 release of the
<a href="https://www.nlsinfo.org/content/cohorts/nlsy79-children">NLSY79 Children and Young Adults sample</a>.  Data were extracted with the NLS Investigator
(https://www.nlsinfo.org/investigator/).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.

summary(Survey79)
table(Survey79$SurveyYear, Survey79$SurveySource)
table(is.na(Survey79$AgeSelfReportYears), is.na(Survey79$AgeCalculateYears))

if (require(ggplot2) &amp; require(dplyr)) {
  dsSourceYear &lt;- Survey79 %&gt;%
    dplyr::count(SurveyYear, SurveySource) %&gt;%
    dplyr::filter(SurveySource != "NoInterview")

  Survey79 %&gt;%
    dplyr::filter(SurveySource != "NoInterview") %&gt;%
    dplyr::group_by(SurveySource, SurveyYear) %&gt;%
    dplyr::summarize(
      age_min = min(Age, na.rm = TRUE),
      age_max = max(Age, na.rm = TRUE)
    ) %&gt;%
    dplyr::ungroup() %&gt;%
    ggplot(aes(x = SurveyYear, ymin = age_min, ymax = age_max, color = SurveySource)) +
    geom_errorbar() +
    scale_color_brewer(palette = "Dark2") +
    theme_minimal() +
    theme(legend.position = c(0, 1), legend.justification = c(0, 1))
}

</code></pre>

<hr>
<h2 id='ValidateOutcomeDataset'>Validates the schema of datasets containing outcome variables.</h2><span id='topic+ValidateOutcomeDataset'></span>

<h3>Description</h3>

<p>The <span class="pkg">NlsyLinks</span> handles a lot of the plumbing code needed to transform extracted NLSY datasets into a format that statistical routines can interpret.  In some cases, a dataset of measured variables is needed, with one row per subject.  This function validates the measured/outcome dataset, to ensure it posses an interpretable schema.  For a specific list of the requirements, see <code>Details</code> below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ValidateOutcomeDataset(dsOutcome, outcomeNames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ValidateOutcomeDataset_+3A_dsoutcome">dsOutcome</code></td>
<td>
<p>A <a href="base.html#topic+data.frame">base::data.frame</a> with the measured variables</p>
</td></tr>
<tr><td><code id="ValidateOutcomeDataset_+3A_outcomenames">outcomeNames</code></td>
<td>
<p>The column names of the measure variables that eventually will be used by a statistical procedure.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>dsOutcome</code> parameter must:
</p>

<ol>
<li><p> Have a non-missing value.
</p>
</li>
<li><p> Contain at least one row.
</p>
</li>
<li><p> Contain a column called 'SubjectTag' (case sensitive).
</p>
</li>
<li><p> Have the SubjectTag column containing only positive numbers.
</p>
</li>
<li><p> Have the SubjectTag column where all values are unique (ie, two rows/subjects cannot have the same value).
</p>
</li></ol>

<p>The <code>outcomeNames</code> parameter must:
</p>

<ol>
<li><p> Have a non-missing value
</p>
</li>
<li><p> Contain only column names that are present in the <code>dsOutcome</code> data frame.
</p>
</li></ol>



<h3>Value</h3>

<p>Returns <code>TRUE</code> if the validation passes.
Returns an error (and associated descriptive message) if it false.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(NlsyLinks) # Load the package into the current R session.
ds &lt;- ExtraOutcomes79
outcomeNames &lt;- c("MathStandardized", "WeightZGenderAge")
ValidateOutcomeDataset(dsOutcome = ds, outcomeNames = outcomeNames) # Returns TRUE.
outcomeNamesBad &lt;- c("MathMisspelled", "WeightZGenderAge")
# ValidateOutcomeDataset(dsOutcome=ds, outcomeNames=outcomeNamesBad) #Throws error.

</code></pre>

<hr>
<h2 id='ValidatePairLinks'>Validates the schema of a links for pairs of relatives</h2><span id='topic+ValidatePairLinks'></span>

<h3>Description</h3>

<p>A helper function that verifies the linking dataset contains (A) the
essential columns exist, and (B) at least one row.  It is called by <code>CreatePairLinks</code>.
</p>
<p>Typical use of <span class="pkg">NlsyLinks</span> will not require this function, since a
valid paired links are supplied for each supported sample (ie, <a href="#topic+Links79Pair">Links79Pair</a>).
</p>
<p>The <span class="pkg">NlsyLinks</span> uses several types of linking schemas.  This function
validates the type where each relative subject has their own row.
</p>
<p>The following four columns must be present: (1) <code>Subect1Tag</code>, (2)
<code>Subect2Tag</code>, (3) <code>R</code>, and (4) <code>MultipleBirth</code>.  They must
have a <code>numeric</code> mode/datatype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ValidatePairLinks(linksPair)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ValidatePairLinks_+3A_linkspair">linksPair</code></td>
<td>
<p>The <a href="base.html#topic+data.frame">base::data.frame</a> to validate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if the validation passes. Returns an error (and associated descriptive message) if it false.
</p>


<h3>See Also</h3>

<p><a href="#topic+Links79Pair">Links79Pair</a>, <a href="#topic+Links79PairExpanded">Links79PairExpanded</a>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dsSingleLinks &lt;- data.frame(
  ExtendedID        = c(1, 1, 1, 2),
  SubjectTag_S1     = c(101, 101, 102, 201),
  SubjectTag_S2     = c(102, 103, 103, 202),
  R                 = c(.5, .25, .25, .5),
  RelationshipPath  = rep("Gen2Siblings", 4)
)
dsSingleOutcomes &lt;- data.frame(
  SubjectTag = c(101, 102, 103, 201, 202),
  DV1        = c(11, 12, 13, 41, 42),
  DV2        = c(21, 22, 23, 51, 52)
)
dsDouble &lt;- CreatePairLinksDoubleEntered(
  outcomeDataset         = dsSingleOutcomes,
  linksPairDataset       = dsSingleLinks,
  outcomeNames           = c("DV1", "DV2"),
  validateOutcomeDataset = TRUE
)
dsDouble # Show the 8 rows in the double-entered pair links
summary(dsDouble) # Summarize the variables

ValidatePairLinksAreSymmetric(dsDouble) # Should return TRUE.
</code></pre>

<hr>
<h2 id='ValidatePairLinksAreSymmetric'>Verifies that the pair relationships are symmetric.</h2><span id='topic+ValidatePairLinksAreSymmetric'></span>

<h3>Description</h3>

<p>For certain analyses, the pairs links (which can be considered a type of
sparse matrix) need to be symmetric. For instance, if there is a row for
Subjects 201 and 202 with <code>R</code>=0.5, there should be a second row for
Subjects 202 and 201 with <code>R</code>=0.5.
</p>
<p>This validation function is useful to some types of DF methods and some
spatially-inspired methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ValidatePairLinksAreSymmetric(linksPair)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ValidatePairLinksAreSymmetric_+3A_linkspair">linksPair</code></td>
<td>
<p>The <a href="base.html#topic+data.frame">base::data.frame</a> object that should be symmetric</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if symmetric.  Throw an error with <code><a href="base.html#topic+stop">base::stop()</a></code> if asymmetric.
</p>


<h3>Author(s)</h3>

<p>Will Beasley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CreatePairLinksDoubleEntered">CreatePairLinksDoubleEntered()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dsSingleLinks &lt;- data.frame(
  ExtendedID       = c(1, 1, 1, 2),
  SubjectTag_S1    = c(101, 101, 102, 201),
  SubjectTag_S2    = c(102, 103, 103, 202),
  R                = c(.5, .25, .25, .5),
  RelationshipPath = rep("Gen2Siblings", 4)
)
dsSingleOutcomes &lt;- data.frame(
  SubjectTag = c(101, 102, 103, 201, 202),
  DV1        = c(11, 12, 13, 41, 42),
  DV2        = c(21, 22, 23, 51, 52)
)
dsDouble &lt;- CreatePairLinksDoubleEntered(
  outcomeDataset         = dsSingleOutcomes,
  linksPairDataset       = dsSingleLinks,
  outcomeNames           = c("DV1", "DV2"),
  validateOutcomeDataset = TRUE
)
dsDouble # Show the 8 rows in the double-entered pair links
summary(dsDouble) # Summarize the variables

ValidatePairLinksAreSymmetric(dsDouble) # Should return TRUE.
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
