<!DOCTYPE html><html><head><title>Help for package nodeSub</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {nodeSub}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#nodeSub-package'>
<p>Package providing functions to simulate sequences under different DNA evolution models</p></a></li>
<li><a href='#calc_expected_hidden_nodes'><p>Calculate the number of expected hidden nodes in a phylogenetic tree</p></a></li>
<li><a href='#calc_fraction'><p>Calculate the expected fraction of substitutions at the nodes,</p>
relative to the fraction at the branches</a></li>
<li><a href='#calc_required_node_time'><p>Calculate the required node time to obtain a desired fraction of</p>
substitutions at the node</a></li>
<li><a href='#calc_sum_stats'><p>calculate summary statistics of a phylogenetic tree,</p>
compared with a reference tree. The following statistics are calculated:
the beta statistic, gamma statistic, crown age, mean branch length,
number of tips, the nLTT statistic and the laplacian difference, given by
RPANDA's JSDtree. Because JSDtree can sometimes cause issues, some additional
checks are performed to ensure that is possible to run this function.</a></li>
<li><a href='#count_hidden'><p>Function to calculate the number of hidden speciation events, e.g. speciation</p>
events that have lead to an extinct species. Thus, these hidden speciation
events can only be detected in complete trees (as opposed to reconstructed
trees).</a></li>
<li><a href='#create_balanced_tree'><p>create a balanced tree out of branching times</p></a></li>
<li><a href='#create_equal_alignment'><p>function create an alignment with identical information content</p></a></li>
<li><a href='#create_equal_alignment_explicit'><p>function create an alignment with identical information content, using the</p>
explicit method to simulate substitutions</a></li>
<li><a href='#create_unbalanced_tree'><p>create an unbalanced tree out of branching times</p></a></li>
<li><a href='#estimate_marginal_models'><p>estimate the marginal likelihood of the relaxed and strict clock model for</p>
a provided alignment</a></li>
<li><a href='#get_p_matrix'><p>calculate p matrix</p></a></li>
<li><a href='#infer_phylogeny'><p>infer the time calibrated phylogeny associated with the provided alignment.</p>
This function uses the R package babette to infer the phylogeny using BEAST2.</a></li>
<li><a href='#reduce_tree'><p>Function to remove speciation events occuring after an extinction event.</p>
Extinct species are pruned randomly, such that only a single extinct species
per branching event (if any extinct species) remains.</a></li>
<li><a href='#sim_linked'><p>simulate a sequence assuming conditional substitutions on the node.</p></a></li>
<li><a href='#sim_normal'><p>Simulate sequences for a given evolutionary tree, using a standard model</p>
of sequence evolution along the branches. Code for this function was heavily
inspired by the function <code>simSeq</code> from the phangorn package.</a></li>
<li><a href='#sim_normal_explicit'><p>simulate a sequence assuming substitutions are only accumulated along the</p>
branches, using the explicit simulation method (e.g. reverse substitutions
are modeled explicitly)</a></li>
<li><a href='#sim_unlinked'><p>Simulate a sequence assuming node substitutions are not shared amongst</p>
offspring, given two substitution matrices: one for substitutions occuring
on the nodes, and one for substitutions occuring along the branches.</a></li>
<li><a href='#sim_unlinked_explicit'><p>Simulate a sequence assuming node substitutions are not shared amongst</p>
offspring, using the explicit simulation method (e.g. reverse substitutions
are modeled explicitly)</a></li>
<li><a href='#slow_matrix'><p>this function calculates the p matrix within R</p>
this is slower than the C++ implementation in <code>get_p_matrix</code>
but provides a way to debug and verify</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulate DNA Alignments Using Node Substitutions</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.8</td>
</tr>
<tr>
<td>Author:</td>
<td>Thijs Janzen</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thijs Janzen &lt;thijsjanzen@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulate DNA sequences for the node substitution model.
    In the node substitution model, substitutions accumulate additionally 
    during a speciation event, providing a potential mechanistic explanation for 
    substitution rate variation. This package provides tools to simulate
    such a process, simulate a reference process with only substitutions along
    the branches, and provides tools to infer phylogenies from alignments. More
    information can be found in Janzen (2021) &lt;<a href="https://doi.org/10.1093%2Fsysbio%2Fsyab085">doi:10.1093/sysbio/syab085</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/thijsjanzen/nodeSub">https://github.com/thijsjanzen/nodeSub</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/thijsjanzen/nodeSub">https://github.com/thijsjanzen/nodeSub</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Depends:</td>
<td>Rcpp, ape</td>
</tr>
<tr>
<td>Imports:</td>
<td>phangorn, tibble, DDD, Rmpfr, pbapply, phylobase, geiger,
beautier, beastier, tracerer, rappdirs, testit, stringr,
lifecycle</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, TreeSim, dplyr, knitr, rmarkdown, ggplot2,
magrittr, tidyr, nLTT, RPANDA</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-14 13:45:14 UTC; P251362</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-14 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='nodeSub-package'>
Package providing functions to simulate sequences under different DNA evolution models
</h2><span id='topic+nodeSub-package'></span><span id='topic+nodeSub'></span>

<h3>Description</h3>

<p>Simulate DNA sequences for the node substitution model.
In the node substitution model, substitutions accumulate additionally
during a speciation event, providing a potential mechanistic explanation for
substitution rate variation. This package provides tools to simulate
such a process, simulate a reference process with only substitutions along
the branches, and provides tools to infer phylogenies from alignments. More
information can be found in Janzen (2021) &lt;doi:10.1093/sysbio/syab085&gt;.
</p>
<p>Version History: <br />
Version 1.2.7   -  Removed beta calculation due to apTreeshape removal from
CRAN <br />
Version 1.2.3   -  Removed summary statistic tests for CRAN <br />
Version 1.2.2   -  Changed codedov links in README <br />
Version 1.2.1   -  Expanded depenency on RPANDA <br />
Version 1.2     -  Release on CRAN <br />
</p>


<h3>Author(s)</h3>

<p>Thijs Janzen
Maintainer: Thijs Janzen &lt;t.janzen@rug.nl&gt;
</p>


<h3>References</h3>

<p>Thijs Janzen, Folmer Bokma, Rampal S Etienne, Nucleotide Substitutions during Speciation may Explain Substitution Rate Variation, Systematic Biology, 2021; syab085
</p>

<hr>
<h2 id='calc_expected_hidden_nodes'>Calculate the number of expected hidden nodes in a phylogenetic tree</h2><span id='topic+calc_expected_hidden_nodes'></span>

<h3>Description</h3>

<p>Calculate the number of expected hidden nodes
using equation 1 in Manceau et al. 2020
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_expected_hidden_nodes(phy, lambda = NULL, mu = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_expected_hidden_nodes_+3A_phy">phy</code></td>
<td>
<p>phylogenetic tree</p>
</td></tr>
<tr><td><code id="calc_expected_hidden_nodes_+3A_lambda">lambda</code></td>
<td>
<p>birth rate</p>
</td></tr>
<tr><td><code id="calc_expected_hidden_nodes_+3A_mu">mu</code></td>
<td>
<p>death rate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>expected number of hidden nodes
</p>


<h3>References</h3>

<p>Manceau, M., Marin, J., Morlon, H., &amp; Lambert, A. (2020).
Model-based inference of punctuated molecular evolution.
Molecular Biology and Evolution, 37(11), 3308-3323.
</p>

<hr>
<h2 id='calc_fraction'>Calculate the expected fraction of substitutions at the nodes,
relative to the fraction at the branches</h2><span id='topic+calc_fraction'></span>

<h3>Description</h3>

<p>calculates the relative contribution of substitutions at
the nodes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_fraction(phy = NULL, node_time = 0, model = "unlinked")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_fraction_+3A_phy">phy</code></td>
<td>
<p>phylogenetic tree (optional)</p>
</td></tr>
<tr><td><code id="calc_fraction_+3A_node_time">node_time</code></td>
<td>
<p>time spent at the node</p>
</td></tr>
<tr><td><code id="calc_fraction_+3A_model">model</code></td>
<td>
<p>node substitution model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>expected fraction
</p>

<hr>
<h2 id='calc_required_node_time'>Calculate the required node time to obtain a desired fraction of
substitutions at the node</h2><span id='topic+calc_required_node_time'></span>

<h3>Description</h3>

<p>calculates the required node time to obtain a desired fraction
of substitutions at the node
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_required_node_time(phy = NULL, s = 0.5, model = "unlinked")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_required_node_time_+3A_phy">phy</code></td>
<td>
<p>phylogenetic tree</p>
</td></tr>
<tr><td><code id="calc_required_node_time_+3A_s">s</code></td>
<td>
<p>desired fraction</p>
</td></tr>
<tr><td><code id="calc_required_node_time_+3A_model">model</code></td>
<td>
<p>node substitution model, either &quot;linked&quot; or &quot;unlinked&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>expected fraction
</p>

<hr>
<h2 id='calc_sum_stats'>calculate summary statistics of a phylogenetic tree,
compared with a reference tree. The following statistics are calculated:
the beta statistic, gamma statistic, crown age, mean branch length,
number of tips, the nLTT statistic and the laplacian difference, given by
RPANDA's JSDtree. Because JSDtree can sometimes cause issues, some additional
checks are performed to ensure that is possible to run this function.</h2><span id='topic+calc_sum_stats'></span>

<h3>Description</h3>

<p>calculate summary statistics of a phylogenetic tree,
compared with a reference tree. The following statistics are calculated:
the beta statistic, gamma statistic, crown age, mean branch length,
number of tips, the nLTT statistic and the laplacian difference, given by
RPANDA's JSDtree. Because JSDtree can sometimes cause issues, some additional
checks are performed to ensure that is possible to run this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_sum_stats(trees, true_tree, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_sum_stats_+3A_trees">trees</code></td>
<td>
<p>a phyloList object containing multiple trees</p>
</td></tr>
<tr><td><code id="calc_sum_stats_+3A_true_tree">true_tree</code></td>
<td>
<p>a phylo object containing the reference tree, preferably
without extinct lineages. If extinct lineages are found,
these are dropped.</p>
</td></tr>
<tr><td><code id="calc_sum_stats_+3A_verbose">verbose</code></td>
<td>
<p>verbose output if true (e.g. progressbars)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with two tibbles
1) containing the summary statistics of all trees and
2) containing the difference with the true tree
</p>

<hr>
<h2 id='count_hidden'>Function to calculate the number of hidden speciation events, e.g. speciation
events that have lead to an extinct species. Thus, these hidden speciation
events can only be detected in complete trees (as opposed to reconstructed
trees).</h2><span id='topic+count_hidden'></span>

<h3>Description</h3>

<p>Function to calculate the number of hidden speciation events, e.g. speciation
events that have lead to an extinct species. Thus, these hidden speciation
events can only be detected in complete trees (as opposed to reconstructed
trees).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_hidden(tree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_hidden_+3A_tree">tree</code></td>
<td>
<p>phylo object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>number of hidden speciation events
</p>

<hr>
<h2 id='create_balanced_tree'>create a balanced tree out of branching times</h2><span id='topic+create_balanced_tree'></span>

<h3>Description</h3>

<p>create a balanced tree out of branching times
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_balanced_tree(brts)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_balanced_tree_+3A_brts">brts</code></td>
<td>
<p>vector of branching times</p>
</td></tr>
</table>


<h3>Value</h3>

<p>phylo phylo object
</p>

<hr>
<h2 id='create_equal_alignment'>function create an alignment with identical information content</h2><span id='topic+create_equal_alignment'></span>

<h3>Description</h3>

<p>function create an alignment with identical information content
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_equal_alignment(
  input_tree,
  sub_rate,
  alignment_result,
  sim_function = NULL,
  verbose = FALSE,
  node_time = NULL,
  input_alignment_type = "nodesub"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_equal_alignment_+3A_input_tree">input_tree</code></td>
<td>
<p>phylogeny for which to generate alignment</p>
</td></tr>
<tr><td><code id="create_equal_alignment_+3A_sub_rate">sub_rate</code></td>
<td>
<p>substitution rate used in the original phylogeny</p>
</td></tr>
<tr><td><code id="create_equal_alignment_+3A_alignment_result">alignment_result</code></td>
<td>
<p>result of sim_normal, sim_linked or sim_unlinked</p>
</td></tr>
<tr><td><code id="create_equal_alignment_+3A_sim_function">sim_function</code></td>
<td>
<p>function that accepts a tree, sequence length,
rootsequence and substitution rate (in that order). Default is sim_normal</p>
</td></tr>
<tr><td><code id="create_equal_alignment_+3A_verbose">verbose</code></td>
<td>
<p>provide intermediate output</p>
</td></tr>
<tr><td><code id="create_equal_alignment_+3A_node_time">node_time</code></td>
<td>
<p>node time</p>
</td></tr>
<tr><td><code id="create_equal_alignment_+3A_input_alignment_type">input_alignment_type</code></td>
<td>
<p>was the input alignment simulated with a node
substitution model or a normal substitution model? Used to calculate the
twin mutation rate. Options are &quot;nodesub&quot; and &quot;normal&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with four properties: 1) alignment: the alignment itself,
2) adjusted rate: the substitution rate used to obtain identical information
content 3) total_accumulated_substitutions: the total number of
substitutions accumulated. 4) total_node_substitutions: total number of
substitutions accumulated on the nodes 5) total_branch_substitutions: total
number of substitutions accumulated on the branches.
</p>

<hr>
<h2 id='create_equal_alignment_explicit'>function create an alignment with identical information content, using the
explicit method to simulate substitutions</h2><span id='topic+create_equal_alignment_explicit'></span>

<h3>Description</h3>

<p>function create an alignment with identical information content, using the
explicit method to simulate substitutions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_equal_alignment_explicit(
  input_tree,
  sub_rate,
  alignment_result,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_equal_alignment_explicit_+3A_input_tree">input_tree</code></td>
<td>
<p>phylogeny for which to generate alignment</p>
</td></tr>
<tr><td><code id="create_equal_alignment_explicit_+3A_sub_rate">sub_rate</code></td>
<td>
<p>substitution rate used in the original phylogeny</p>
</td></tr>
<tr><td><code id="create_equal_alignment_explicit_+3A_alignment_result">alignment_result</code></td>
<td>
<p>result of sim_normal, sim_linked or sim_unlinked</p>
</td></tr>
<tr><td><code id="create_equal_alignment_explicit_+3A_verbose">verbose</code></td>
<td>
<p>provide intermediate output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with four properties: 1) alignment: the alignment itself,
2) adjusted rate: the substitution rate used to obtain identical information
content 3) total_accumulated_substitutions: the total number of
substitutions accumulated. 4) total_node_substitutions: total number of
substitutions accumulated on the nodes 5) total_branch_substitutions: total
number of substitutions accumulated on the branches.
</p>

<hr>
<h2 id='create_unbalanced_tree'>create an unbalanced tree out of branching times</h2><span id='topic+create_unbalanced_tree'></span>

<h3>Description</h3>

<p>create an unbalanced tree out of branching times
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_unbalanced_tree(brts)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_unbalanced_tree_+3A_brts">brts</code></td>
<td>
<p>vector of branching times</p>
</td></tr>
</table>


<h3>Value</h3>

<p>phylo phylo object
</p>

<hr>
<h2 id='estimate_marginal_models'>estimate the marginal likelihood of the relaxed and strict clock model for
a provided alignment</h2><span id='topic+estimate_marginal_models'></span>

<h3>Description</h3>

<p>estimate_marginal_models estimates the marginal likelihood of
both the strict and the relaxed clock model, given the JC69 substitution
model, using the NS package in BEAST, made available via the babette R
package. The NS package performs nested sampling, and uses an MCMC approach
to estimate the marginal likelihood. Sampling is performed until convergence
of the MCMC chain. Unfortunately, currently the babette package is unavailable
on CRAN, requiring installation through GitHub to enjoy the full
functionality of this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_marginal_models(
  fasta_filename,
  use_yule_prior = FALSE,
  rng_seed = 42,
  sub_rate = 1,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_marginal_models_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>file name of fasta file holding alignment for which the
marginal likelihood is to be estimated</p>
</td></tr>
<tr><td><code id="estimate_marginal_models_+3A_use_yule_prior">use_yule_prior</code></td>
<td>
<p>by default, a birth-death prior is used as tree prior,
but if use_yule_prior is set to TRUE, a pure-birth prior will be used.</p>
</td></tr>
<tr><td><code id="estimate_marginal_models_+3A_rng_seed">rng_seed</code></td>
<td>
<p>seed of pseudo-random number generator</p>
</td></tr>
<tr><td><code id="estimate_marginal_models_+3A_sub_rate">sub_rate</code></td>
<td>
<p>substitution rate</p>
</td></tr>
<tr><td><code id="estimate_marginal_models_+3A_verbose">verbose</code></td>
<td>
<p>boolean indicating if verbose intermediate output is to be
generated</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with marginal likelihoods and relative weights per clock
model.
</p>

<hr>
<h2 id='get_p_matrix'>calculate p matrix</h2><span id='topic+get_p_matrix'></span>

<h3>Description</h3>

<p>calculates the p matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_p_matrix(branch_length, eig = phangorn::edQt(), rate = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_p_matrix_+3A_branch_length">branch_length</code></td>
<td>
<p>branch length</p>
</td></tr>
<tr><td><code id="get_p_matrix_+3A_eig">eig</code></td>
<td>
<p>eigen object</p>
</td></tr>
<tr><td><code id="get_p_matrix_+3A_rate">rate</code></td>
<td>
<p>rate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p matrix
</p>

<hr>
<h2 id='infer_phylogeny'>infer the time calibrated phylogeny associated with the provided alignment.
This function uses the R package babette to infer the phylogeny using BEAST2.</h2><span id='topic+infer_phylogeny'></span>

<h3>Description</h3>

<p>infer the time calibrated phylogeny associated with the provided alignment.
This function uses the R package babette to infer the phylogeny using BEAST2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>infer_phylogeny(
  alignment,
  treatment_name,
  tree_prior = beautier::create_bd_tree_prior(),
  clock_prior = beautier::create_strict_clock_model(),
  mcmc_seed = NULL,
  chain_length = 1e+07,
  sample_interval = 5000,
  burnin = 0.1,
  working_dir = NULL,
  sub_rate = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="infer_phylogeny_+3A_alignment">alignment</code></td>
<td>
<p>Phydat object containing the focal alignment</p>
</td></tr>
<tr><td><code id="infer_phylogeny_+3A_treatment_name">treatment_name</code></td>
<td>
<p>string to be appended to BEAST files</p>
</td></tr>
<tr><td><code id="infer_phylogeny_+3A_tree_prior">tree_prior</code></td>
<td>
<p>tree prior used, default = birth-death prior</p>
</td></tr>
<tr><td><code id="infer_phylogeny_+3A_clock_prior">clock_prior</code></td>
<td>
<p>clock prior used, default = strict clock</p>
</td></tr>
<tr><td><code id="infer_phylogeny_+3A_mcmc_seed">mcmc_seed</code></td>
<td>
<p>seed of the mcmc chain, default is the system time</p>
</td></tr>
<tr><td><code id="infer_phylogeny_+3A_chain_length">chain_length</code></td>
<td>
<p>length of the mcmc chain, default is 1e7.</p>
</td></tr>
<tr><td><code id="infer_phylogeny_+3A_sample_interval">sample_interval</code></td>
<td>
<p>interval of sampling, default is 5000</p>
</td></tr>
<tr><td><code id="infer_phylogeny_+3A_burnin">burnin</code></td>
<td>
<p>burnin of posterior distribution</p>
</td></tr>
<tr><td><code id="infer_phylogeny_+3A_working_dir">working_dir</code></td>
<td>
<p>beast2 working dir</p>
</td></tr>
<tr><td><code id="infer_phylogeny_+3A_sub_rate">sub_rate</code></td>
<td>
<p>substitution rate used to generate the original
alignment (if available), default is 1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with all trees, and the consensus tree
</p>

<hr>
<h2 id='reduce_tree'>Function to remove speciation events occuring after an extinction event.
Extinct species are pruned randomly, such that only a single extinct species
per branching event (if any extinct species) remains.</h2><span id='topic+reduce_tree'></span>

<h3>Description</h3>

<p>Function to remove speciation events occuring after an extinction event.
Extinct species are pruned randomly, such that only a single extinct species
per branching event (if any extinct species) remains.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reduce_tree(tree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reduce_tree_+3A_tree">tree</code></td>
<td>
<p>phylo object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pruned tree
</p>

<hr>
<h2 id='sim_linked'>simulate a sequence assuming conditional substitutions on the node.</h2><span id='topic+sim_linked'></span>

<h3>Description</h3>

<p>simulate a sequence assuming conditional substitutions on the node.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_linked(
  phy,
  Q = rep(1, 6),
  rate = 0.1,
  node_mut_rate_double = 1e-09,
  l = 1000,
  bf = rep(0.25, 4),
  rootseq = NULL,
  node_time = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_linked_+3A_phy">phy</code></td>
<td>
<p>tree for which to simulate sequences</p>
</td></tr>
<tr><td><code id="sim_linked_+3A_q">Q</code></td>
<td>
<p>substitution matrix along the branches, default = JC</p>
</td></tr>
<tr><td><code id="sim_linked_+3A_rate">rate</code></td>
<td>
<p>mutation rate , default = 1</p>
</td></tr>
<tr><td><code id="sim_linked_+3A_node_mut_rate_double">node_mut_rate_double</code></td>
<td>
<p>mutation rate on the node, default = 1e-9</p>
</td></tr>
<tr><td><code id="sim_linked_+3A_l">l</code></td>
<td>
<p>number of base pairs to simulate</p>
</td></tr>
<tr><td><code id="sim_linked_+3A_bf">bf</code></td>
<td>
<p>base frequencies, default = c(0.25, 0.25, 0.25, 0.25)</p>
</td></tr>
<tr><td><code id="sim_linked_+3A_rootseq">rootseq</code></td>
<td>
<p>sequence at the root, simulated by default</p>
</td></tr>
<tr><td><code id="sim_linked_+3A_node_time">node_time</code></td>
<td>
<p>time spent at the node</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with four items </p>

<ol>
<li><p>alignment Phydat object with the resulting alignment
</p>
</li>
<li><p>rootseq the rootsequence used
</p>
</li>
<li><p>total_branch_substitutions total number of substitutions accumulated
on the branches
</p>
</li>
<li><p>total_node_substitutions total number of substitutions accumulated at
the nodes</p>
</li></ol>


<hr>
<h2 id='sim_normal'>Simulate sequences for a given evolutionary tree, using a standard model
of sequence evolution along the branches. Code for this function was heavily
inspired by the function <code>simSeq</code> from the phangorn package.</h2><span id='topic+sim_normal'></span>

<h3>Description</h3>

<p>Simulate sequences for a given evolutionary tree, using a standard model
of sequence evolution along the branches. Code for this function was heavily
inspired by the function <code>simSeq</code> from the phangorn package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_normal(x, l = 1000, Q = NULL, bf = NULL, rootseq = NULL, rate = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_normal_+3A_x">x</code></td>
<td>
<p>a phylogenetic tree <code>tree</code>, i.e. an object of class
<code>phylo</code></p>
</td></tr>
<tr><td><code id="sim_normal_+3A_l">l</code></td>
<td>
<p>length of the sequence to simulate.</p>
</td></tr>
<tr><td><code id="sim_normal_+3A_q">Q</code></td>
<td>
<p>the rate matrix.</p>
</td></tr>
<tr><td><code id="sim_normal_+3A_bf">bf</code></td>
<td>
<p>base frequencies.</p>
</td></tr>
<tr><td><code id="sim_normal_+3A_rootseq">rootseq</code></td>
<td>
<p>a vector of length l containing the root sequence, other root
sequence is randomly generated.</p>
</td></tr>
<tr><td><code id="sim_normal_+3A_rate">rate</code></td>
<td>
<p>mutation rate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with four items </p>

<ol>
<li><p>alignment Phydat object with the resulting alignment
</p>
</li>
<li><p>rootseq the rootsequence used
</p>
</li>
<li><p>total_branch_substitutions total number of substitutions accumulated
on the branches
</p>
</li>
<li><p>total_node_substitutions total number of substitutions accumulated at
the nodes</p>
</li></ol>



<h3>Author(s)</h3>

<p>Klaus Schliep <a href="mailto:klaus.schliep@gmail.com">klaus.schliep@gmail.com</a>
</p>

<hr>
<h2 id='sim_normal_explicit'>simulate a sequence assuming substitutions are only accumulated along the
branches, using the explicit simulation method (e.g. reverse substitutions
are modeled explicitly)</h2><span id='topic+sim_normal_explicit'></span>

<h3>Description</h3>

<p>simulate a sequence assuming substitutions are only accumulated along the
branches, using the explicit simulation method (e.g. reverse substitutions
are modeled explicitly)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_normal_explicit(x, l = 1000, Q = NULL, bf = NULL, rootseq = NULL, rate = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_normal_explicit_+3A_x">x</code></td>
<td>
<p>a phylogenetic tree <code>tree</code>, i.e. an object of class
<code>phylo</code> or and object of class <code>pml</code>.</p>
</td></tr>
<tr><td><code id="sim_normal_explicit_+3A_l">l</code></td>
<td>
<p>length of the sequence to simulate.</p>
</td></tr>
<tr><td><code id="sim_normal_explicit_+3A_q">Q</code></td>
<td>
<p>the rate matrix.</p>
</td></tr>
<tr><td><code id="sim_normal_explicit_+3A_bf">bf</code></td>
<td>
<p>base frequencies.</p>
</td></tr>
<tr><td><code id="sim_normal_explicit_+3A_rootseq">rootseq</code></td>
<td>
<p>a vector of length l containing the root sequence, other root
sequence is randomly generated.</p>
</td></tr>
<tr><td><code id="sim_normal_explicit_+3A_rate">rate</code></td>
<td>
<p>mutation rate or scaler for the edge length, a numerical value
greater than zero.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with four items </p>

<ol>
<li><p>alignment Phydat object with the resulting alignment
</p>
</li>
<li><p>rootseq the rootsequence used
</p>
</li>
<li><p>total_branch_substitutions total number of substitutions accumulated
on the branches
</p>
</li>
<li><p>total_node_substitutions total number of substitutions accumulated at
the nodes</p>
</li></ol>


<hr>
<h2 id='sim_unlinked'>Simulate a sequence assuming node substitutions are not shared amongst
offspring, given two substitution matrices: one for substitutions occuring
on the nodes, and one for substitutions occuring along the branches.</h2><span id='topic+sim_unlinked'></span>

<h3>Description</h3>

<p>Simulate a sequence assuming node substitutions are not shared amongst
offspring, given two substitution matrices: one for substitutions occuring
on the nodes, and one for substitutions occuring along the branches.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_unlinked(
  phy,
  Q1 = rep(1, 6),
  Q2 = rep(1, 6),
  rate1 = 0.1,
  rate2 = 0.1,
  l = 1000,
  bf = rep(0.25, 4),
  rootseq = NULL,
  node_time = 0.001
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_unlinked_+3A_phy">phy</code></td>
<td>
<p>tree for which to simulate sequences</p>
</td></tr>
<tr><td><code id="sim_unlinked_+3A_q1">Q1</code></td>
<td>
<p>substitution matrix along the branches, default = JC</p>
</td></tr>
<tr><td><code id="sim_unlinked_+3A_q2">Q2</code></td>
<td>
<p>substitution matrix on the nodes, default = JC</p>
</td></tr>
<tr><td><code id="sim_unlinked_+3A_rate1">rate1</code></td>
<td>
<p>mutation rate along the branch, default = 0.1</p>
</td></tr>
<tr><td><code id="sim_unlinked_+3A_rate2">rate2</code></td>
<td>
<p>mutation rate on the node, default = 0.1</p>
</td></tr>
<tr><td><code id="sim_unlinked_+3A_l">l</code></td>
<td>
<p>number of base pairs to simulate</p>
</td></tr>
<tr><td><code id="sim_unlinked_+3A_bf">bf</code></td>
<td>
<p>base frequencies, default = c(0.25, 0.25, 0.25, 0.25)</p>
</td></tr>
<tr><td><code id="sim_unlinked_+3A_rootseq">rootseq</code></td>
<td>
<p>sequence at the root, simulated by default</p>
</td></tr>
<tr><td><code id="sim_unlinked_+3A_node_time">node_time</code></td>
<td>
<p>amount of time spent at the nodes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with four items </p>

<ol>
<li><p>alignment Phydat object with the resulting alignment
</p>
</li>
<li><p>rootseq the rootsequence used
</p>
</li>
<li><p>total_branch_substitutions total number of substitutions accumulated
on the branches
</p>
</li>
<li><p>total_node_substitutions total number of substitutions accumulated at
the nodes</p>
</li></ol>


<hr>
<h2 id='sim_unlinked_explicit'>Simulate a sequence assuming node substitutions are not shared amongst
offspring, using the explicit simulation method (e.g. reverse substitutions
are modeled explicitly)</h2><span id='topic+sim_unlinked_explicit'></span>

<h3>Description</h3>

<p>Simulate a sequence assuming node substitutions are not shared amongst
offspring, using the explicit simulation method (e.g. reverse substitutions
are modeled explicitly)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_unlinked_explicit(
  phy,
  Q1 = rep(1, 6),
  Q2 = rep(1, 6),
  rate1 = 0.1,
  rate2 = 0.1,
  l = 1000,
  bf = rep(0.25, 4),
  rootseq = NULL,
  node_time = 0.001
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_unlinked_explicit_+3A_phy">phy</code></td>
<td>
<p>phylogenetic tree for which to simulate sequences</p>
</td></tr>
<tr><td><code id="sim_unlinked_explicit_+3A_q1">Q1</code></td>
<td>
<p>substitution matrix along the branches, default = JC</p>
</td></tr>
<tr><td><code id="sim_unlinked_explicit_+3A_q2">Q2</code></td>
<td>
<p>substitution matrix on the nodes, default = JC</p>
</td></tr>
<tr><td><code id="sim_unlinked_explicit_+3A_rate1">rate1</code></td>
<td>
<p>mutation rate along the branch, default = 0.1</p>
</td></tr>
<tr><td><code id="sim_unlinked_explicit_+3A_rate2">rate2</code></td>
<td>
<p>mutation rate on the node, default = 0.1</p>
</td></tr>
<tr><td><code id="sim_unlinked_explicit_+3A_l">l</code></td>
<td>
<p>number of base pairs to simulate</p>
</td></tr>
<tr><td><code id="sim_unlinked_explicit_+3A_bf">bf</code></td>
<td>
<p>base frequencies, default = c(0.25, 0.25, 0.25, 0.25)</p>
</td></tr>
<tr><td><code id="sim_unlinked_explicit_+3A_rootseq">rootseq</code></td>
<td>
<p>sequence at the root, simulated by default</p>
</td></tr>
<tr><td><code id="sim_unlinked_explicit_+3A_node_time">node_time</code></td>
<td>
<p>amount of time spent at the nodes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with four items </p>

<ol>
<li><p>alignment Phydat object with the resulting alignment
</p>
</li>
<li><p>rootseq the rootsequence used
</p>
</li>
<li><p>total_branch_substitutions total number of substitutions accumulated
on the branches
</p>
</li>
<li><p>total_node_substitutions total number of substitutions accumulated at
the nodes</p>
</li></ol>


<hr>
<h2 id='slow_matrix'>this function calculates the p matrix within R
this is slower than the C++ implementation in <code>get_p_matrix</code>
but provides a way to debug and verify</h2><span id='topic+slow_matrix'></span>

<h3>Description</h3>

<p>this function calculates the p matrix within R
this is slower than the C++ implementation in <code>get_p_matrix</code>
but provides a way to debug and verify
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slow_matrix(eig, branch_length, rate)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="slow_matrix_+3A_eig">eig</code></td>
<td>
<p>eigen object</p>
</td></tr>
<tr><td><code id="slow_matrix_+3A_branch_length">branch_length</code></td>
<td>
<p>branch length</p>
</td></tr>
<tr><td><code id="slow_matrix_+3A_rate">rate</code></td>
<td>
<p>substitution rate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p matrix
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
