<!DOCTYPE html><html><head><title>Help for package nodiv</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {nodiv}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#nodiv-package'>
<p>nodiv - Node-based analysis of species distributions</p></a></li>
<li><a href='#add_shape'>
<p>Add a shapefile to a <code>nodiv</code> data object</p></a></li>
<li><a href='#basal_node'>
<p>Identify nodes in a phylogeny</p></a></li>
<li><a href='#coquettes'>
<p>Distribution of coquette hummingbirds in Northern South America</p></a></li>
<li><a href='#distrib_data'>
<p>Data objects for node-based analysis</p></a></li>
<li><a href='#gridData'>
<p>Apply a grid (2D bins) to nodiv objects of type 'points'</p></a></li>
<li><a href='#Node_analysis'>
<p>Calculate GND and SOS scores for a phylogeny and community data set</p></a></li>
<li><a href='#Node_size'>
<p>Attributes of individual clades</p></a></li>
<li><a href='#Nodesig'>
<p>Compare the distributions of a pair of sister clades</p></a></li>
<li><a href='#Nspecies'>
<p>Basic attributes of <code>distrib_data</code> objects</p></a></li>
<li><a href='#occurrences'>
<p>Access functions to the community matrix of <code>distrib_data</code> objects</p></a></li>
<li><a href='#phyplot'><p>Plot Phylogenies</p></a></li>
<li><a href='#plot_nodes_phylo'>
<p>Plot a phylogeny with colored node labels</p></a></li>
<li><a href='#plot_points'>
<p>Plot a variable on spatial points or a spatial grid using a color scale.</p></a></li>
<li><a href='#plot_richness'>
<p>Plot a richness map for an object of type distrib_data, nodiv_data or nodiv_result</p></a></li>
<li><a href='#sitestat'>
<p>Site statistics of <code>nodiv</code> data objects</p></a></li>
<li><a href='#SOS'>
<p>Gives the SOS values for a given node.</p></a></li>
<li><a href='#species_stat'>
<p>Species statistics of <code>nodiv</code> data objects</p></a></li>
<li><a href='#subsample'>
<p>Subsampling data objects to certain species or sites</p></a></li>
<li><a href='#two_color_map'>
<p>Plot the relative amounts of two variables using a purple-green color scheme</p></a></li>
<li><a href='#update_object'>
<p>Update <code>distrib_data</code> objects to the <code>nodiv</code> version &gt;= 1.1 format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Compares the Distribution of Sister Clades Through a Phylogeny</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-21</td>
</tr>
<tr>
<td>Author:</td>
<td>Michael Krabbe Borregaard</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Krabbe Borregaard &lt;mkborregaard@sund.ku.dk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of the nodiv algorithm, see Borregaard, M.K., Rahbek, C., Fjeldsaa, J., Parra, J.L., Whittaker, R.J. &amp; Graham, C.H. 2014. Node-based analysis of species distributions. Methods in Ecology and Evolution 5(11): 1225-1235. &lt;<a href="https://doi.org/10.1111%2F2041-210X.12283">doi:10.1111/2041-210X.12283</a>&gt;. Package for phylogenetic analysis of species distributions. The main function goes through each node in the phylogeny, compares the distributions of the two descendant nodes, and compares the result to a null model. This highlights nodes where major distributional divergence have occurred. The distributional divergence for these nodes is mapped.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>picante, raster, ape, sp, vegan, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>RColorBrewer, parallel, testthat, colorspace</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mkborregaard/nodiv">https://github.com/mkborregaard/nodiv</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-21 07:50:39 UTC; cvg147</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-21 08:12:36 UTC</td>
</tr>
</table>
<hr>
<h2 id='nodiv-package'>
nodiv - Node-based analysis of species distributions
</h2><span id='topic+nodiv-package'></span><span id='topic+nodiv'></span>

<h3>Description</h3>

<p>The package implements Borregaard et al. (2014) method for identifying nodes in a phylogeny associated with divergent distributions. The main algorithm goes through each node in the phylogeny and relates node overlap to a null model. The package also provides functions for preparing the data sets, for exploratory plots and further analysis.
</p>


<h3>Details</h3>

<p>The workflow starts by creating a <code>nodiv_data</code> data object, which makes sure that the phylogeny, the species distributions and the spatial coordinates of sites are matched correctly. The function takes a number of data types. There are several functions for exploratory data analysis, including <code>plot</code>, <code>summary</code>, <code>richness</code>, <code>Node_occupancy</code>, <code>Node_size</code> etc. The core of the package are the functions <code>Nodesig</code>, which compares the distributions of the two clades descending from a node, and <code>Node_analysis</code>, which applies this function to all nodes in the phylogeny and summarizes the results as a <code>nodiv_result</code> object. There is a set of functions for interpretation of the results, including <code>plot</code>, <code>plotSOS</code>, and <code>summary</code>. The package also provides basic functions for plotting and manipulating data sets that combine spatial distributions with phylogenies, e.g. <code>subsample</code>, <code>plot_grid</code> and <code>plot_points</code>.  
Note that the package may still be unstable - if you experience problems, try cloning the github repository instead: <code>library(devtools); install_github("mkborregaard/nodiv")</code>, or feel free to contact the maintainer.
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>
<p>Maintainer: Michael Krabbe Borregaard &lt;mkborregaard@sund.ku.dk&gt;
</p>


<h3>References</h3>

<p>Borregaard, M.K., Rahbek, C., Fjeldsaa, J., Parra, J.L., Whittaker, R.J. &amp; Graham, C.H. 2014. Node-based analysis of species distributions. Methods in Ecology and Evolution 5(11): 1225-1235. &lt;DOI:10.1111/2041-210X.12283&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(coquettes)
res &lt;- Node_analysis(coquettes, 50, "rdtable")
opar &lt;- par(mfrow = c(1,2))
plot(res)
plotSOS(res, 28, shapefill = terrain.colors(64))
summary(res)
par(opar)
</code></pre>

<hr>
<h2 id='add_shape'>
Add a shapefile to a <code>nodiv</code> data object 
</h2><span id='topic+add_shape'></span>

<h3>Description</h3>

<p>Takes a shapefile, e.g. as points or polygons, and adds it to the data object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_shape(distrib_data, shape)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_shape_+3A_distrib_data">distrib_data</code></td>
<td>

<p>An object of type <code>distrib_data</code>, <code>nodiv_data</code> or <code>nodiv_result</code>
</p>
</td></tr>
<tr><td><code id="add_shape_+3A_shape">shape</code></td>
<td>

<p>A GIS shapefile, from e.g. packages <code>maptools</code> or <code>rgdal</code>
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the object with the new shapefile
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>


<h3>See Also</h3>

<p><code><a href="#topic+distrib_data">distrib_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # remove the shapefile from coquettes
  data(coquettes)
  sh &lt;- coquettes$shape
  coquettes$shape &lt;- NULL
  # and add it back in
  coquettes &lt;- add_shape(coquettes, sh)
</code></pre>

<hr>
<h2 id='basal_node'>
Identify nodes in a phylogeny
</h2><span id='topic+basal_node'></span><span id='topic+nodenumbers'></span><span id='topic+nodes'></span><span id='topic+Parent'></span><span id='topic+Descendants'></span><span id='topic+Sister'></span><span id='topic+MostRecentAncestor'></span>

<h3>Description</h3>

<p>Returns the internal (<code>ape</code>) node numbers corresponding to certain nodes in the phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basal_node(tree)
nodenumbers(tree)
nodes(tree, all = FALSE)
Descendants(node, tree)
Parent(node, tree)
Sister(node, tree)
MostRecentAncestor(tips, tree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="basal_node_+3A_tree">tree</code></td>
<td>

<p>An object of type <code>phylo</code>, or alternatively an object of type <code>nodiv_data</code>
</p>
</td></tr>
<tr><td><code id="basal_node_+3A_node">node</code></td>
<td>

<p>The internal node number of a focal node in the phylogeny
</p>
</td></tr>
<tr><td><code id="basal_node_+3A_all">all</code></td>
<td>

<p>Give all node labels in order of occurrence. If FALSE, only non-empty node labels will be shown in alphabetical order
</p>
</td></tr>
<tr><td><code id="basal_node_+3A_tips">tips</code></td>
<td>

<p>The species to find the most recent common ancestor for
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>Parent</code> gives the code immediately basal to the focal node, <code>Descendants</code> gives the nodes descending from the focal node, and <code>Sister</code> gives the sister node(s). <code>MostRecentAncestor</code> gives the most recent common node for the given tips.
</p>


<h3>Value</h3>

<p>The internal (<code>ape</code>) node number of the node
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>

<hr>
<h2 id='coquettes'>
Distribution of coquette hummingbirds in Northern South America
</h2><span id='topic+coquettes'></span>

<h3>Description</h3>

<p>Phylogeny and distributions for the coquette clade of hummingbirds in Ecuador, Peru and Colombia
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(coquettes)</code></pre>


<h3>Format</h3>

<p>a <code>nodiv_data</code> object with the phylogeny and distributions of all species
</p>


<h3>Source</h3>

<p>Graham, C.H., Parra, J.L., Tinoco, B.A., Stiles, F.G. &amp; McGuire, J.A. (2012). Untangling the influence of ecological and evolutionary factors on trait variation across hummingbird assemblages. Ecology, 93, S99-S111.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(coquettes)
plot(coquettes, shapefill = terrain.colors(64), pch = 21)
</code></pre>

<hr>
<h2 id='distrib_data'>
Data objects for node-based analysis
</h2><span id='topic+distrib_data'></span><span id='topic+nodiv_data'></span><span id='topic+plot.distrib_data'></span><span id='topic+plot.nodiv_data'></span><span id='topic+print.distrib_data'></span><span id='topic+print.nodiv_data'></span><span id='topic+summary.distrib_data'></span><span id='topic+identify.distrib_data'></span><span id='topic+summary.nodiv_data'></span>

<h3>Description</h3>

<p>Functions to collect and display data with matching species distributions and phylogenies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distrib_data(commatrix, coords = NULL, proj4string_in = CRS(as.character(NA)), 
  type = c("auto", "grid", "points"), shape = NULL)
nodiv_data(phylo, commatrix, coords, proj4string_in = CRS(as.character(NA)), 
  type = c("auto", "grid", "points"), shape = NULL)
## S3 method for class 'distrib_data'
summary(object, ...)
## S3 method for class 'nodiv_data'
summary(object, ...)
## S3 method for class 'distrib_data'
plot(x, ...)
## S3 method for class 'nodiv_data'
plot(x,  ...)
## S3 method for class 'distrib_data'
identify(x,  ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distrib_data_+3A_phylo">phylo</code></td>
<td>

<p>A phylogeny in ape's phylo format, with tip.labels that correspond to the species names in commatrix
</p>
</td></tr>
<tr><td><code id="distrib_data_+3A_commatrix">commatrix</code></td>
<td>

<p>A community matrix or data.frame of species distributions, with species as columns, sites as rows, and 0/1 denoting the absence or presence of a species in a given site. Alternatively, the functions accept a <code>data.frame</code> in the phylocom format, i.e. with three columns containing plot id, abundance (ignored) and species id. The <code>nodiv_data</code> function also accepts an object of type <code>distrib_data</code>, in which case <code>coords</code> is ignored. The <code>nodiv_data</code> function matches the species in the matrix to the tips of <code>phylo</code>.
</p>
</td></tr>
<tr><td><code id="distrib_data_+3A_coords">coords</code></td>
<td>

<p>The coordinates of all sites in the data. The input can be of a variety of different formats. Can be a <code>data.frame</code> or <code>matrix</code> with two columns. If more columns are present, the function will search for typical coordinate names such as X/Y or Lon/Lat. Alternatively, an object inheriting from <code>SpatialPoints</code> from package <code>sp</code> is accepted. The coordinates are transformed to <code>SpatialPoints</code> using the proj4string argument. Site names can be given as a third column of the data.frame, as rownames for the matrix or omitted. If site names are given, and <code>commatrix</code> has row.names, these are matched. 
</p>
</td></tr>
<tr><td><code id="distrib_data_+3A_proj4string_in">proj4string_in</code></td>
<td>

<p>Projection string as defined in package <code>sp</code> 
</p>
</td></tr>
<tr><td><code id="distrib_data_+3A_type">type</code></td>
<td>

<p>A descriptor of whether the sites are point (community) sites, or the centres of regular grid cells. If omitted or set to <code>auto</code>, the function will guess the type from <code>coords</code>.
</p>
</td></tr>
<tr><td><code id="distrib_data_+3A_shape">shape</code></td>
<td>

<p>An optional shapefile to overlay plots upon.
</p>
</td></tr>
<tr><td><code id="distrib_data_+3A_object">object</code></td>
<td>

<p>An object of type distrib_data or nodiv_data
</p>
</td></tr>
<tr><td><code id="distrib_data_+3A_x">x</code></td>
<td>

<p>An object of type distrib_data or nodiv_data
</p>
</td></tr>
<tr><td><code id="distrib_data_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to the plotting function
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When making an object of type 'grid', the coordinates will be stored as <code>SpatialPixels</code> from package <code>sp</code>. This may result in a warning if there are rows or columns in the grid without data. The function removes these. This may affect plotting, but can otherwise be ignored.
</p>


<h3>Value</h3>

<p>The <code>distrib_data</code> function produces an object of type <code>distrib_data</code>, which has the following components:
</p>
<table>
<tr><td><code>coords</code></td>
<td>
<p>A <code>SpatialPointsDataFrame</code> or <code>SpatialPixelsDataFrame</code> (from <code>sp</code>) with spatial coordinates, and one column named <code>sites</code> containing the site names</p>
</td></tr>
<tr><td><code>comm</code></td>
<td>
<p>The community matrix</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>A character vector describing the type of data</p>
</td></tr>
<tr><td><code>species</code></td>
<td>
<p>A list of species names</p>
</td></tr>
<tr><td><code>shape</code></td>
<td>
<p>An optional slot detailing a shapefile object for plotting the data</p>
</td></tr>
</table>
<p>The <code>nodiv_data</code> function produces an object of type <code>nodiv_data</code>, which contains the additional component:
</p>
<table>
<tr><td><code>phylo</code></td>
<td>
<p>A phylogeny of type <code>phylo</code> from package <code>ape</code></p>
</td></tr> 
</table>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Node_analysis">Node_analysis</a></code>
</p>

<hr>
<h2 id='gridData'>
Apply a grid (2D bins) to nodiv objects of type 'points'
</h2><span id='topic+gridData'></span>

<h3>Description</h3>

<p>Takes an object of type <code>distrib_data</code> or <code>nodiv_data</code>, and puts the point distributions on a grid. Can also be used for resampling grid data at a coarser scale. If the comm matrix has abundances (any one value larger than 1), they are summed across points in the grid cell.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gridData(dist_data, cellsize_x = 1, cellsize_y = cellsize_x, xll_corner, yll_corner)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gridData_+3A_dist_data">dist_data</code></td>
<td>

<p>An object of type <code>distrib_data</code> or <code>nodiv_data</code> 
</p>
</td></tr>
<tr><td><code id="gridData_+3A_cellsize_x">cellsize_x</code></td>
<td>

<p>The size of the bins along the X (longitudinal) axis, in units of the spatial coordinates of the original points
</p>
</td></tr>
<tr><td><code id="gridData_+3A_cellsize_y">cellsize_y</code></td>
<td>

<p>The size of the bins along the Y (latitudinal) axis, in units of the spatial coordinates of the original points
</p>
</td></tr>
<tr><td><code id="gridData_+3A_xll_corner">xll_corner</code></td>
<td>

<p>The x coordinate of the lower left corner of the lower left grid cell of the intended grid. The corner must be left of all points in the data set
</p>
</td></tr>
<tr><td><code id="gridData_+3A_yll_corner">yll_corner</code></td>
<td>

<p>The y coordinate of the lower left corner of the lower left grid cell of the intended grid. The corner must be below all points in the data set
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The return value is the original <code>dist_data</code> object with gridded distribution data
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>


<h3>See Also</h3>

<p><code><a href="#topic+distrib_data">distrib_data</a></code>, ~~~
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(coquettes)
new &lt;- gridData(coquettes, 2)
plot_richness(new)
## End(Not run)
</code></pre>

<hr>
<h2 id='Node_analysis'>
Calculate GND and SOS scores for a phylogeny and community data set
</h2><span id='topic+Node_analysis'></span><span id='topic+nodiv_result'></span><span id='topic+print.nodiv_result'></span><span id='topic+summary.nodiv_result'></span><span id='topic+print.summary_nodiv_result'></span><span id='topic+plot.nodiv_result'></span>

<h3>Description</h3>

<p>Goes through each node in the phylogeny, and compares the distribution overlap of sister clades to that expected from a null mode that considers node size, species richness and range size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Node_analysis(nodiv_data, repeats = 100, method = c("rdtable", "quasiswap"), 
                cores = 1, log_parallel_progress = FALSE)
## S3 method for class 'nodiv_result'
summary(object, ...)
## S3 method for class 'nodiv_result'
print(x, printlen = 4, ...)
## S3 method for class 'nodiv_result'
plot(x, label = nodenumbers(x), zlim = 0:1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Node_analysis_+3A_nodiv_data">nodiv_data</code></td>
<td>

<p>An object created by the <code>nodiv_data</code> function
</p>
</td></tr>
<tr><td><code id="Node_analysis_+3A_repeats">repeats</code></td>
<td>

<p>Number of null communities to create at each node
</p>
</td></tr>
<tr><td><code id="Node_analysis_+3A_method">method</code></td>
<td>

<p>The null model algorithm used to simulate random communities
</p>
</td></tr>
<tr><td><code id="Node_analysis_+3A_cores">cores</code></td>
<td>

<p>The number of cores to use in the computation. Note that multi-core processing is yet unimplemented.
</p>
</td></tr>
<tr><td><code id="Node_analysis_+3A_log_parallel_progress">log_parallel_progress</code></td>
<td>

<p>If this is TRUE, the results from each node will be saved as an '.rda' file in the working directory. Useful for large parallel computations.
</p>
</td></tr>
<tr><td><code id="Node_analysis_+3A_object">object</code></td>
<td>

<p>The <code>nodiv_result</code> object
</p>
</td></tr>
<tr><td><code id="Node_analysis_+3A_x">x</code></td>
<td>

<p>The <code>nodiv_result</code> object
</p>
</td></tr>
<tr><td><code id="Node_analysis_+3A_printlen">printlen</code></td>
<td>

<p>The number of species to print to screen
</p>
</td></tr>
<tr><td><code id="Node_analysis_+3A_label">label</code></td>
<td>

<p>The text to use for labelling nodes in the plot
</p>
</td></tr>
<tr><td><code id="Node_analysis_+3A_zlim">zlim</code></td>
<td>

<p>The values of GND to use for the extremes of the color scale on the plot
</p>
</td></tr>
<tr><td><code id="Node_analysis_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed to the function
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The &quot;rdtable&quot; algorithm is efficient, but may underestimate the deviation between nodes. It should primarily be used for exploratory data analysis. Note that the more robust &quot;quasiswap&quot; algorithm may take days to run on larger data sets.
</p>
<p>The <code>plot</code> function for <code>nodiv_result</code> plots the phylogenetic tree with colored symbols at the nodes. The color and size the nodes are proportional with the GND value of the node.
</p>


<h3>Value</h3>

<p>The result is an object of type <code>nodiv_result</code>, which has the following components:
</p>
<table>
<tr><td><code>GND</code></td>
<td>
<p>A numeric vector between 0 and 1 describing the distributional divergence associated with a node</p>
</td></tr>
<tr><td><code>SOS</code></td>
<td>
<p>A numeric matrix given the SOS of values, which expresses the difference between empirical and expected species richness of each clade at each site</p>
</td></tr>
</table>
<p>It also includes all the elements of the <code>nodiv_data</code> object:
</p>
<table>
<tr><td><code>coords</code></td>
<td>
<p>A <code>SpatialPointsDataFrame</code> or <code>SpatialPixelsDataFrame</code> (from <code>sp</code>) with spatial coordinates, and one column named <code>sites</code> containing the site names</p>
</td></tr>
<tr><td><code>comm</code></td>
<td>
<p>The community matrix</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>A character vector describing the type of data</p>
</td></tr>
<tr><td><code>species</code></td>
<td>
<p>A list of species names</p>
</td></tr>
<tr><td><code>shape</code></td>
<td>
<p>An optional slot detailing a shapefile object for plotting the data</p>
</td></tr>
<tr><td><code>phylo</code></td>
<td>
<p>A phylogeny of type <code>phylo</code> from package <code>ape</code></p>
</td></tr> 
</table>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>


<h3>References</h3>

<p>Borregaard, M.K., Rahbek, C., Fjeldsaa, J., Parra, J.L., Whittaker, R.J. &amp; Graham, C.H. 2014. Node-based analysis of species distributions. Methods in Ecology and Evolution 5(11): 1225-1235. DOI: 10.1111/2041-210X.12283
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotSOS">plotSOS</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(coquettes)
res &lt;- Node_analysis(coquettes, 50, "rdtable")
opar &lt;- par(mfrow = c(1,2))
plot(res)
plotSOS(res, 28, shapefill = terrain.colors(64))
summary(res)
par(opar)
</code></pre>

<hr>
<h2 id='Node_size'>
Attributes of individual clades
</h2><span id='topic+Node_size'></span><span id='topic+Node_sites'></span><span id='topic+Node_species'></span><span id='topic+Node_occupancy'></span>

<h3>Description</h3>

<p>Functions that summarize the species that descend from a node, and their distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Node_size(nodiv_data, node = NULL)
Node_sites(nodiv_data, node, names = TRUE)
Node_species(nodiv_data, node, names = TRUE)
Node_occupancy(nodiv_data, node = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Node_size_+3A_node">node</code></td>
<td>

<p>A character or numeric variable specifying one or more nodes in the phylogeny
</p>
</td></tr>
<tr><td><code id="Node_size_+3A_nodiv_data">nodiv_data</code></td>
<td>

<p>An object of type <code>nodiv_data</code> or <code>nodiv_result</code>. <code>Node_species</code> also accepts objects of type <code>phylo</code>
</p>
</td></tr>
<tr><td><code id="Node_size_+3A_names">names</code></td>
<td>

<p>If <code>TRUE</code>, return a character vector of species names; otherwise return the species index numbers in the community matrix 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If no <code>node</code> is specified in <code>Node_size</code> or <code>Node_occupancy</code>, the function will calculate the value for all nodes in the phylogeny
</p>


<h3>Value</h3>

<p>For <code>Node_size</code> and <code>Node_occupancy</code> the number of species or number of occupied sites of the nodes; or a vector giving this value for all nodes. For <code>Node_species</code> or <code>Node_sites</code> a character vector of names of species descending from a node, or the sites where they occur.
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>

<hr>
<h2 id='Nodesig'>
Compare the distributions of a pair of sister clades
</h2><span id='topic+Nodesig'></span>

<h3>Description</h3>

<p>Compares the overlap of species richness of a pair of sister clades to the expectation from a null model. By default, the function compares the two descendants from the basal node of the phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nodesig(nodiv_data, Node_sp = NULL, repeats = 100, method = c("rdtable", "quasiswap"),
  show = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nodesig_+3A_nodiv_data">nodiv_data</code></td>
<td>

<p>An object of type <code>nodiv_data</code>
</p>
</td></tr>
<tr><td><code id="Nodesig_+3A_node_sp">Node_sp</code></td>
<td>

<p>A character or numeric vector indicating which species that belong to one of the clades. If NULL, the species of a descendant from the basal node of nodiv_data is chosen randomly 
</p>
</td></tr>
<tr><td><code id="Nodesig_+3A_repeats">repeats</code></td>
<td>

<p>The number of repeats to use for the null model
</p>
</td></tr>
<tr><td><code id="Nodesig_+3A_method">method</code></td>
<td>

<p>The null model algorithm used in the calculations
</p>
</td></tr>
<tr><td><code id="Nodesig_+3A_show">show</code></td>
<td>

<p>Should a progress bar be printed during calculations?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The \&quot;rdtable\&quot; algorithm is efficient, but may underestimate the deviation between nodes. It should primarily be used for exploratory data analysis. Note that the more robust \&quot;quasiswap\&quot; algorithm may take hours to run on data sets with a few hundred species.
</p>


<h3>Value</h3>

<p>The return value is a list with 5 elements:
</p>
<table>
<tr><td><code>SR</code></td>
<td>
<p>A numeric vector of the standardized difference between the empirical and simulated species richness for one node in each site</p>
</td></tr>
<tr><td><code>rval</code></td>
<td>
<p>The rank of the empirical species richness of one node in the distribution of simulated values</p>
</td></tr>
<tr><td><code>nodeemp</code></td>
<td>
<p>The numerical species richness in each site</p>
</td></tr>
<tr><td><code>nodemeans</code></td>
<td>
<p>The mean simulated species richness of one node in each site</p>
</td></tr>
<tr><td><code>nodesds</code></td>
<td>
<p>The standard deviation of the simulated species richness of one node in each site</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>


<h3>References</h3>

<p>Borregaard, M.K., Rahbek, C., Fjeldsaa, J., Parra, J.L., Whittaker, R.J. &amp; Graham, C.H. 2014. Node-based analysis of species distributions. Methods in Ecology and Evolution 5(11): 1225-1235. DOI: 10.1111/2041-210X.12283
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Node_analysis">Node_analysis</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(coquettes)
res &lt;- Nodesig(coquettes, repeats = 50)
hist(res$rval, xlim = 0:1)
</code></pre>

<hr>
<h2 id='Nspecies'>
Basic attributes of <code>distrib_data</code> objects
</h2><span id='topic+Nspecies'></span><span id='topic+Nsites'></span><span id='topic+richness'></span><span id='topic+occupancy'></span><span id='topic+sites'></span><span id='topic+species'></span><span id='topic+coords'></span>

<h3>Description</h3>

<p>Calculates the species richness and site occupancy of a <code>distrib_data</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nspecies(distrib_data)
Nsites(distrib_data)
richness(distrib_data, sites = NULL)
occupancy(distrib_data, species = NULL)
sites(distrib_data)
species(distrib_data)
coords(distrib_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nspecies_+3A_distrib_data">distrib_data</code></td>
<td>

<p>An object of type <code>distrib_data</code>, <code>nodiv_data</code> or <code>nodiv_result</code>
</p>
</td></tr>
<tr><td><code id="Nspecies_+3A_sites">sites</code></td>
<td>

<p>The site for which to return the richness
</p>
</td></tr>
<tr><td><code id="Nspecies_+3A_species">species</code></td>
<td>

<p>The species for which to return the occupancy
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>Nspecies</code> gives the number of species in the dataset, and <code>Nsites</code> the number of sites. 
<code>richness</code> gives a numeric vector with the species richness of all sites.
<code>occupancy</code> gives the number of sites occupied by all species.
<code>sites</code> gives a list of site names in the object. 
<code>species</code> gives a list of species names in the object. 
<code>nodes</code> gives a node labels or node numbers 
<code>coords</code> returns the spatial coordinates of points or grid cell centers 
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>

<hr>
<h2 id='occurrences'>
Access functions to the community matrix of <code>distrib_data</code> objects
</h2><span id='topic+assemblage'></span><span id='topic+occurrences'></span>

<h3>Description</h3>

<p>Gives the species in a site, or the sites occupied by a species, in a <code>distrib_data</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>occurrences(distrib_data, species, value = c("index", "names", "logical", "raw"))
assemblage(distrib_data, site, value = c("index", "names", "logical", "raw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="occurrences_+3A_distrib_data">distrib_data</code></td>
<td>

<p>An object of type <code>distrib_data</code>, <code>nodiv_data</code> or <code>nodiv_result</code>
</p>
</td></tr>
<tr><td><code id="occurrences_+3A_species">species</code></td>
<td>

<p>The number or name of a species in the data set
</p>
</td></tr>
<tr><td><code id="occurrences_+3A_site">site</code></td>
<td>

<p>The number or name of a site in the data set
</p>
</td></tr>
<tr><td><code id="occurrences_+3A_value">value</code></td>
<td>

<p>The type of data to return, see below.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>type = "01"</code> gives a vector of 0s and 1s from the community matrix
<code>type = "logical"</code> gives a logical vector identifying species or sites
<code>type = "index"</code> gives the index number of species or sites in the community matrix
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>

<hr>
<h2 id='phyplot'>Plot Phylogenies</h2><span id='topic+phyplot'></span>

<h3>Description</h3>

<p>This is just the ape::plot.phylo method, which is imported and exported directly by the nodiv package. These functions plot phylogenetic trees on the current graphical
device.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  phyplot(x, type = "phylogram", use.edge.length = TRUE,
    node.pos = NULL, show.tip.label = TRUE, show.node.label = FALSE,
    edge.color = "black", edge.width = 1, edge.lty = 1, font = 3,
    cex = par("cex"), adj = NULL, srt = 0, no.margin = FALSE,
    root.edge = FALSE, label.offset = 0, underscore = FALSE,
    x.lim = NULL, y.lim = NULL, direction = "rightwards",
    lab4ut = NULL, tip.color = "black", plot = TRUE,
    rotate.tree = 0, open.angle = 0, node.depth = 1, align.tip.label = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyplot_+3A_x">x</code></td>
<td>
<p>an object of class <code>"phylo"</code> </p>
</td></tr>
<tr><td><code id="phyplot_+3A_type">type</code></td>
<td>
<p>a character string specifying the type of phylogeny to be
drawn; it must be one of &quot;phylogram&quot; (the default), &quot;cladogram&quot;,
&quot;fan&quot;, &quot;unrooted&quot;, &quot;radial&quot; or any unambiguous abbreviation of
these.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_use.edge.length">use.edge.length</code></td>
<td>
<p>a logical indicating whether to use the edge
lengths of the phylogeny to draw the branches (the default) or not
(if <code>FALSE</code>). This option has no effect if the object of class
<code>"phylo"</code> has no &lsquo;edge.length&rsquo; element.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_node.pos">node.pos</code></td>
<td>
<p>a numeric taking the value 1 or 2 which specifies the
vertical position of the nodes with respect to their descendants. If
<code>NULL</code> (the default), then the value is determined in relation
to &lsquo;type&rsquo; and &lsquo;use.edge.length&rsquo; (see details).</p>
</td></tr>
<tr><td><code id="phyplot_+3A_show.tip.label">show.tip.label</code></td>
<td>
<p>a logical indicating whether to show the tip
labels on the phylogeny (defaults to <code>TRUE</code>, i.e. the labels
are shown).</p>
</td></tr>
<tr><td><code id="phyplot_+3A_show.node.label">show.node.label</code></td>
<td>
<p>a logical indicating whether to show the node
labels on the phylogeny (defaults to <code>FALSE</code>, i.e. the labels
are not shown).</p>
</td></tr>
<tr><td><code id="phyplot_+3A_edge.color">edge.color</code></td>
<td>
<p>a vector of mode character giving the colours used
to draw the branches of the plotted phylogeny. These are taken to be
in the same order than the component <code>edge</code> of <code>phy</code>. If
fewer colours are given than the length of <code>edge</code>, then the
colours are recycled.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_edge.width">edge.width</code></td>
<td>
<p>a numeric vector giving the width of the branches of
the plotted phylogeny. These are taken to be in the same order than
the component <code>edge</code> of <code>phy</code>. If fewer widths are given
than the length of <code>edge</code>, then these are recycled.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_edge.lty">edge.lty</code></td>
<td>
<p>same than the previous argument but for line types;
1: plain, 2: dashed, 3: dotted, 4: dotdash, 5: longdash, 6: twodash.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_font">font</code></td>
<td>
<p>an integer specifying the type of font for the labels: 1
(plain text), 2 (bold), 3 (italic, the default), or 4 (bold
italic).</p>
</td></tr>
<tr><td><code id="phyplot_+3A_cex">cex</code></td>
<td>
<p>a numeric value giving the factor scaling of the tip and
node labels (Character EXpansion). The default is to take the
current value from the graphical parameters.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_adj">adj</code></td>
<td>
<p>a numeric specifying the justification of the text strings
of the labels: 0 (left-justification), 0.5 (centering), or 1
(right-justification). This option has no effect if <code>type =
      "unrooted"</code>. If <code>NULL</code> (the default) the value is set with
respect of <code>direction</code> (see details).</p>
</td></tr>
<tr><td><code id="phyplot_+3A_srt">srt</code></td>
<td>
<p>a numeric giving how much the labels are rotated in degrees
(negative values are allowed resulting in clock-like rotation); the
value has an effect respectively to the value of
<code>direction</code> (see Examples). This option has no effect if
<code>type = "unrooted"</code>.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_no.margin">no.margin</code></td>
<td>
<p>a logical. If <code>TRUE</code>, the margins are set to
zero and the plot uses all the space of the device (note that this
was the behaviour of <code>plot.phylo</code> up to version 0.2-1 of &lsquo;ape&rsquo;
with no way to modify it by the user, at least easily).</p>
</td></tr>
<tr><td><code id="phyplot_+3A_root.edge">root.edge</code></td>
<td>
<p>a logical indicating whether to draw the root edge
(defaults to FALSE); this has no effect if &lsquo;use.edge.length = FALSE&rsquo;
or if &lsquo;type = &quot;unrooted&quot;&rsquo;.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_label.offset">label.offset</code></td>
<td>
<p>a numeric giving the space between the nodes and
the tips of the phylogeny and their corresponding labels. This
option has no effect if <code>type = "unrooted"</code>.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_underscore">underscore</code></td>
<td>
<p>a logical specifying whether the underscores in tip
labels should be written as spaces (the default) or left as are (if
<code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="phyplot_+3A_x.lim">x.lim</code></td>
<td>
<p>a numeric vector of length one or two giving the limit(s)
of the x-axis. If <code>NULL</code>, this is computed with respect to
various parameters such as the string lengths of the labels and the
branch lengths. If a single value is given, this is taken as the
upper limit.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_y.lim">y.lim</code></td>
<td>
<p>same than above for the y-axis.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_direction">direction</code></td>
<td>
<p>a character string specifying the direction of the
tree. Four values are possible: &quot;rightwards&quot; (the default),
&quot;leftwards&quot;, &quot;upwards&quot;, and &quot;downwards&quot;.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_lab4ut">lab4ut</code></td>
<td>
<p>(= labels for unrooted trees) a character string
specifying the display of tip labels for unrooted trees (can be
abbreviated): either <code>"horizontal"</code> where all labels are
horizontal (the default if <code>type = "u"</code>), or <code>"axial"</code>
where the labels are displayed in the axis of the corresponding
terminal branches. This option has an effect if <code>type = "u"</code>,
<code>"f"</code>, or <code>"r"</code>.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_tip.color">tip.color</code></td>
<td>
<p>the colours used for the tip labels, eventually
recycled (see examples).</p>
</td></tr>
<tr><td><code id="phyplot_+3A_plot">plot</code></td>
<td>
<p>a logical controlling whether to draw the tree. If
<code>FALSE</code>, the graphical device is set as if the tree was
plotted, and the coordinates are saved as well.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_rotate.tree">rotate.tree</code></td>
<td>
<p>for &quot;fan&quot;, &quot;unrooted&quot;, or &quot;radial&quot; trees: the
rotation of the whole tree in degrees (negative values are
accepted).</p>
</td></tr>
<tr><td><code id="phyplot_+3A_open.angle">open.angle</code></td>
<td>
<p>if <code>type = "f"</code> or <code>"r"</code>, the angle in
degrees left blank. Use a non-zero value if you want to call
<code><a href="ape.html#topic+axisPhylo">axisPhylo</a></code> after the tree is plotted.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_node.depth">node.depth</code></td>
<td>
<p>an integer value (1 or 2) used if branch lengths are
not used to plot the tree; 1: the node depths are proportional to
the number of tips descending from each node (the default and was the
only possibility previously), 2: they are evenly spaced.</p>
</td></tr>
<tr><td><code id="phyplot_+3A_align.tip.label">align.tip.label</code></td>
<td>
<p>a logical value or an integer. If TRUE, the tips 
are aligned and dotted lines are drawn between the tips of the tree and
the labels. If an integer, the tips are aligned and this gives the type
of the lines (lty).</p>
</td></tr>
<tr><td><code id="phyplot_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code>plot</code> or to
<code>plot.phylo</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The font format of the labels of the nodes and the tips is the same.
</p>
<p>If <code>no.margin = TRUE</code>, the margins are set to zero and are not
restored after plotting the tree, so that the user can access the
coordinates system of the plot.
</p>
<p>The option &lsquo;node.pos&rsquo; allows the user to alter the vertical position
(i.e., ordinates) of the nodes. If <code>node.pos = 1</code>, then the
ordinate of a node is the mean of the ordinates of its direct
descendants (nodes and/or tips). If <code>node.pos = 2</code>, then the
ordinate of a node is the mean of the ordinates of all the tips of
which it is the ancestor. If <code>node.pos = NULL</code> (the default),
then its value is determined with respect to other options: if
<code>type = "phylogram"</code> then &lsquo;node.pos = 1&rsquo;; if <code>type =
    "cladogram"</code> and <code>use.edge.length = FALSE</code> then &lsquo;node.pos = 2&rsquo;;
if <code>type = "cladogram"</code> and <code>use.edge.length = TRUE</code> then
&lsquo;node.pos = 1&rsquo;. Remember that in this last situation, the branch
lengths make sense when projected on the x-axis.
</p>
<p>If <code>adj</code> is not specified, then the value is determined with
respect to <code>direction</code>: if <code>direction = "leftwards"</code> then
<code>adj = 1</code> (0 otherwise).
</p>
<p>If the arguments <code>x.lim</code> and <code>y.lim</code> are not specified by the
user, they are determined roughly by the function. This may not always
give a nice result: the user may check these values with the
(invisibly) returned list (see &ldquo;Value:&rdquo;).
</p>
<p>If you resize manually the graphical device (windows or X11) you may
need to replot the tree.
</p>


<h3>Value</h3>

<p><code>plot.phylo</code> returns invisibly a list with the following
components which values are those used for the current plot:
</p>
<table>
<tr><td><code>type</code></td>
<td>
</td></tr>
<tr><td><code>use.edge.length</code></td>
<td>
</td></tr>
<tr><td><code>node.pos</code></td>
<td>
</td></tr>
<tr><td><code>node.depth</code></td>
<td>
</td></tr>
<tr><td><code>show.tip.label</code></td>
<td>
</td></tr>
<tr><td><code>show.node.label</code></td>
<td>
</td></tr>
<tr><td><code>font</code></td>
<td>
</td></tr>
<tr><td><code>cex</code></td>
<td>
</td></tr>
<tr><td><code>adj</code></td>
<td>
</td></tr>
<tr><td><code>srt</code></td>
<td>
</td></tr>
<tr><td><code>no.margin</code></td>
<td>
</td></tr>
<tr><td><code>label.offset</code></td>
<td>
</td></tr>
<tr><td><code>x.lim</code></td>
<td>
</td></tr>
<tr><td><code>y.lim</code></td>
<td>
</td></tr>
<tr><td><code>direction</code></td>
<td>
</td></tr>
<tr><td><code>tip.color</code></td>
<td>
</td></tr>
<tr><td><code>Ntip</code></td>
<td>
</td></tr>
<tr><td><code>Nnode</code></td>
<td>
</td></tr>
<tr><td><code>root.time</code></td>
<td>
</td></tr>
</table>


<h3>Note</h3>

<p>The argument <code>asp</code> cannot be passed with <code>...</code>.
</p>


<h3>Author(s)</h3>

<p>Emmanuel Paradis</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>
</p>

<hr>
<h2 id='plot_nodes_phylo'>
Plot a phylogeny with colored node labels
</h2><span id='topic+plot_nodes_phylo'></span>

<h3>Description</h3>

<p>A wrapper for the plot.phylo function defined in package <code>ape</code>, which is useful for plotting the values at each node
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_nodes_phylo(variable, tree, label = variable, main = deparse(substitute(variable)),
  zlim = NULL, col, show.legend = TRUE, sig.cutoff, nodes, 
  roundoff = TRUE, show.tip.label = NULL, cex = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_nodes_phylo_+3A_variable">variable</code></td>
<td>

<p>The variable to be plotted
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_label">label</code></td>
<td>

<p>The text to write in each node label
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_tree">tree</code></td>
<td>

<p>The phylogeny to plot, in the <code>ape phylo</code> format
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_main">main</code></td>
<td>

<p>The title text
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_zlim">zlim</code></td>
<td>

<p>The values to use for the most extreme colors of the color scale
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_col">col</code></td>
<td>

<p>The color palette used to create the color scale
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_show.legend">show.legend</code></td>
<td>

<p>Should a legend for the color scale be plotted?
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_sig.cutoff">sig.cutoff</code></td>
<td>

<p>The minimum value for nodes to be plotted on the tree, useful for trees with many nodes
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_nodes">nodes</code></td>
<td>

<p>The nodes to be plotted on the tree
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_roundoff">roundoff</code></td>
<td>

<p>Whether to round values when writing them as nodelabel text
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_show.tip.label">show.tip.label</code></td>
<td>

<p>Show tip labels on the tree
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_cex">cex</code></td>
<td>

<p>Controls the size of the node symbols
</p>
</td></tr>
<tr><td><code id="plot_nodes_phylo_+3A_...">...</code></td>
<td>

<p>Further arguments to pass to plot.phylo
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(coquettes)
Clade_occupancy &lt;- Node_occupancy(coquettes)
plot_nodes_phylo(Clade_occupancy, coquettes$phylo, cex = 0.7)
</code></pre>

<hr>
<h2 id='plot_points'>
Plot a variable on spatial points or a spatial grid using a color scale.
</h2><span id='topic+plot_points'></span><span id='topic+plot_grid'></span><span id='topic+plot_species'></span>

<h3>Description</h3>

<p>Plots a variable either as colored points or as a colored raster. <code>plot_sitestat</code> is a wrapper function that dispatches the other functions according to the <code>type</code> of <code>distrib_data</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_points(x, coords, col, shape = NULL, 
  shapefill = "grey",  zlim= NULL,  zoom_to_points = FALSE, 
  pch = 16, bg = par("bg"), legend = TRUE, ...)
plot_grid(x, coords, col, shape = NULL, 
  shapefill = "grey",  shapeborder = NA, zlim = NULL, 
  zoom_to_points = FALSE, legend = TRUE, gridcol, gridlwd, 
  gridsites, overlay_shape = FALSE, 
  colscale = c("equal_interval","quantiles"), legendlabels = NULL, ...)
plot_species(distrib_data, species, col = c("darkgreen", "red"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_points_+3A_x">x</code></td>
<td>

<p>A numerical variable to be plotted using a color scale
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_coords">coords</code></td>
<td>

<p>A matrix or <code>data.frame</code> with two columns, or an object inheriting from <code>SpatialPoints</code>, of the same length as <code>x</code>
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_col">col</code></td>
<td>

<p>The color palette to use for the color scale. Can be given as color names, a vector of hex colors from a call to a color palette (e.g., <code>terrain.colors(64)</code>), 
or as the name of one of nodiv's internal palettes. Possibillities are <code>"parula", "jet", "blackbody", "HMblueyellow", "HMrainbow", 
"HMlinear_optimal", "HMoptimal_scale", "cube1", "cubeyf1"</code>. Alternatively <code>col</code> can be one of <code>"auto", "ramp", "monochrome", "divergent" or "individual"</code>, allowing <code>nodiv</code> to choose a palette of the given type
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_shape">shape</code></td>
<td>

<p>A shape file giving a map to use as a background for the plot
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_shapefill">shapefill</code></td>
<td>

<p>If shape is a polygon, the color to use to fill the polygons. If shape is a line shapefile, the color to use for plotting lines
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_shapeborder">shapeborder</code></td>
<td>

<p>If shape is a polygon, the color to use for plotting the line border
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_zlim">zlim</code></td>
<td>

<p>The values to use for the extremes of the color scale
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_zoom_to_points">zoom_to_points</code></td>
<td>

<p>Given a shapefile, the function will set the outer limits of the plot to the bounding box of the shapefile. If <code>zoom_to_points</code> is <code>TRUE</code>, the outer limits will be set by the bounding box of the data points instead.
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_pch">pch</code></td>
<td>

<p>The point character to use for plotting points
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_bg">bg</code></td>
<td>

<p>The central color of points when using a pch value between 21 and 25
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_distrib_data">distrib_data</code></td>
<td>

<p>An object of type <code>distrib_data</code>, <code>nodiv_data</code> or <code>nodiv_result</code> to use for plotting x.
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_legend">legend</code></td>
<td>

<p>Whether to plot a color legend
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_species">species</code></td>
<td>

<p>The number or name of the species to be plotted
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_gridcol">gridcol</code></td>
<td>

<p>If specified, plots the sampling grid cell borders in this color
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_gridlwd">gridlwd</code></td>
<td>

<p>If specified, set the line width of overlaid grid cell borders
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_gridsites">gridsites</code></td>
<td>

<p>If specified, which grid cells to draw cell borders around
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_overlay_shape">overlay_shape</code></td>
<td>

<p>Whether to overlay the shape file border on top of the points - only works if shape is a vector shapefile
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_colscale">colscale</code></td>
<td>

<p>Specifies whether to define the colors on an equal interval or a quantile-based color scale
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_legendlabels">legendlabels</code></td>
<td>

<p>Specifies custom labels for the colorbar legend
</p>
</td></tr>
<tr><td><code id="plot_points_+3A_...">...</code></td>
<td>

<p>Additional arguments to pass to plot
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that if <code>shape</code> is a <code>raster</code> object, the points may shift relative to the background when resizing the image. This seems to be due to a bug in the <code>raster</code> package. 
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard 
</p>

<hr>
<h2 id='plot_richness'>
Plot a richness map for an object of type distrib_data, nodiv_data or nodiv_result
</h2><span id='topic+plot_richness'></span><span id='topic+plot_node'></span>

<h3>Description</h3>

<p>Create a raster or point plot showing the species richness of an object. Useful for comparing the species richness of sister clades
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_richness(distrib_data, ...)
plot_node(nodiv_data, node = basal_node(nodiv_data), sites = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_richness_+3A_distrib_data">distrib_data</code></td>
<td>

<p>An object of types <code>distrib_data</code>, <code>nodiv_data</code> or <code>nodiv_result</code>
</p>
</td></tr>
<tr><td><code id="plot_richness_+3A_nodiv_data">nodiv_data</code></td>
<td>

<p>An object of types <code>nodiv_data</code> or <code>nodiv_result</code>
</p>
</td></tr>
<tr><td><code id="plot_richness_+3A_node">node</code></td>
<td>

<p>The name or internal number of the node to be plotted
</p>
</td></tr>
<tr><td><code id="plot_richness_+3A_sites">sites</code></td>
<td>

<p>If <code>sites = "all"</code> the plotting region retains all sites (including sites unoccupied by that node)
</p>
</td></tr>
<tr><td><code id="plot_richness_+3A_...">...</code></td>
<td>

<p>Further arguments to pass to the plotting function
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot will be done as a spatial grid or as colored points, depending on the <code>type</code> of the <code>nodiv_result</code> object. <code>plot_richness</code> plots the richness of the entire dataset, whereas <code>plot_node</code> plots the species richness of a given node.
</p>


<h3>Value</h3>

<p>If <code>distrib_data</code> has <code>type = raster</code>, a raster object of the SOS values is returned. 
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>

<hr>
<h2 id='sitestat'>
Site statistics of <code>nodiv</code> data objects 
</h2><span id='topic+add_sitestat'></span><span id='topic+sitestat'></span><span id='topic+plot_sitestat'></span>

<h3>Description</h3>

<p>Adds, plots or reads site statistics from objects in the <code>nodiv</code> package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_sitestat(distrib_data, site_stat, site = NULL)
plot_sitestat(distrib_data, x, shape = NULL, type = c("auto",
                 "points", "grid"), ...)
sitestat(distrib_data, statname = NULL, site = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sitestat_+3A_distrib_data">distrib_data</code></td>
<td>

<p>An object of type <code>distrib_data</code>, <code>nodiv_data</code> or <code>nodiv_result</code>
</p>
</td></tr>
<tr><td><code id="sitestat_+3A_site_stat">site_stat</code></td>
<td>

<p>A site statistic, such as species richness or mean temperature. This can be a vector or a data.frame. If 
<code>site</code> is not specified, the function tries to match <code>site_stat</code> to the site names of the <code>distrib_data</code> 
object. The function looks for vector names, rownames, or variables in the data.frame that match the site names. 
If sitenames are absent and the number of sitestats match, the sitestat is assumed to be in the correct order. 
</p>
</td></tr>
<tr><td><code id="sitestat_+3A_site">site</code></td>
<td>

<p>A character or numeric variable specifying the sites in which to enter or read sitestat
</p>
</td></tr>
<tr><td><code id="sitestat_+3A_statname">statname</code></td>
<td>

<p>The name of a sitestatistic in the <code>distrib_data</code> object. If statname is not given, a list of
names of sitestatistics is given
</p>
</td></tr>
<tr><td><code id="sitestat_+3A_x">x</code></td>
<td>

<p>A variable to plot on sites, can either be a numeric variable, or the name of a 
sitestatistic in the <code>distrib_data</code> object
</p>
</td></tr>
<tr><td><code id="sitestat_+3A_shape">shape</code></td>
<td>

<p>A shape file giving a map to use as a background for the plot
</p>
</td></tr>
<tr><td><code id="sitestat_+3A_type">type</code></td>
<td>
<p>A character vector describing the type of data</p>
</td></tr>
<tr><td><code id="sitestat_+3A_...">...</code></td>
<td>

<p>Further arguments to the plotting function
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>add_sitestat</code> returns the object with the new sitestat inside. 
<code>sitestat</code> returns the named sitestatistic.
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>


<h3>See Also</h3>

<p><code><a href="#topic+distrib_data">distrib_data</a></code>
</p>

<hr>
<h2 id='SOS'>
Gives the SOS values for a given node.
</h2><span id='topic+SOS'></span><span id='topic+plotSOS'></span><span id='topic+GND'></span>

<h3>Description</h3>

<p><code>SOS</code> gives the SOS value for the node, used in the interpretation of node-based analysis of species distributions after running <code>Node_analysis</code>.  <code>plotSOS</code> plots the values on a map. <code>GND</code> gives the GND values.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SOS(nodiv_result, node)
plotSOS(nodiv_result, node, zlim, ...)
GND(nodiv_result, node = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SOS_+3A_nodiv_result">nodiv_result</code></td>
<td>

<p>The result of running <code>Node_analysis</code>
</p>
</td></tr>
<tr><td><code id="SOS_+3A_node">node</code></td>
<td>

<p>The node label or node number specifying the node to plot. Optional for <code>GND</code>
</p>
</td></tr>
<tr><td><code id="SOS_+3A_zlim">zlim</code></td>
<td>

<p>The values to use for the extremes of the color palette
</p>
</td></tr>
<tr><td><code id="SOS_+3A_...">...</code></td>
<td>

<p>Further arguments to pass to the plotting function
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The first descendant clade (i.e. the first returned by a call to <code>Descendants</code>) is associated with positive SOS values, the second descendant clade is associated with negative values. The plot will be done as a spatial grid or as colored points, depending on the <code>type</code> of the <code>nodiv_result</code> object. If the <code>nodiv_result</code> object has a <code>shape</code> object of type <code>raster</code>, this will be plotted normally in the background - but resizing the window will cause the plots/pixels and the underlying raster object to lose alignment, due to a bug in the <code>raster</code> package.
</p>


<h3>Value</h3>

<p><code>SOS</code> returns a numerical vector with the SOS values. If <code>nodiv_result</code> has <code>type raster</code>, a raster object of the SOS values is returned by <code>plotSOS</code>. 
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>


<h3>References</h3>

<p>Borregaard, M.K., Rahbek, C., Fjeldsaa, J., Parra, J.L., Whittaker, R.J. &amp; Graham, C.H. 2014. Node-based analysis of species distributions. Methods in Ecology and Evolution 5(11): 1225-1235. DOI: 10.1111/2041-210X.12283
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Node_analysis">Node_analysis</a></code>
</p>

<hr>
<h2 id='species_stat'>
Species statistics of <code>nodiv</code> data objects 
</h2><span id='topic+add_species_stat'></span><span id='topic+species_stat'></span>

<h3>Description</h3>

<p>Adds, or reads species statistics from objects in the <code>nodiv</code> package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_species_stat(distrib_data, species_stat, specs = NULL)
species_stat(distrib_data, statname = NULL, specs = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="species_stat_+3A_distrib_data">distrib_data</code></td>
<td>

<p>An object of type <code>distrib_data</code>, <code>nodiv_data</code> or <code>nodiv_result</code>
</p>
</td></tr>
<tr><td><code id="species_stat_+3A_species_stat">species_stat</code></td>
<td>

<p>A species statistic, such as body size or cold tolerance. This can be a vector or a data.frame. If 
<code>specs</code> is not specified, the function tries to match <code>species_stat</code> to the site names of the 
<code>distrib_data</code> object. The function looks for vector names, rownames, or variables in the data.frame that match the species names. If species names are absent and the number of species_stats match, the species_stat is assumed to be in the correct order. 
</p>
</td></tr>
<tr><td><code id="species_stat_+3A_specs">specs</code></td>
<td>

<p>A character or numeric variable specifying the species of interest, or for which species_stat is specified
</p>
</td></tr>
<tr><td><code id="species_stat_+3A_statname">statname</code></td>
<td>

<p>The name of a species statistic in the <code>distrib_data</code> object. If statname is not given, a list of
names of species statistics is given
</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>add_species_stat</code> returns the object with the new species_stat inside. 
<code>species_stat</code> returns the named sitestatistic.
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>


<h3>See Also</h3>

<p><code><a href="#topic+distrib_data">distrib_data</a></code>
</p>

<hr>
<h2 id='subsample'>
Subsampling data objects to certain species or sites 
</h2><span id='topic+subsample'></span><span id='topic+subsample.distrib_data'></span><span id='topic+subsample.nodiv_data'></span><span id='topic+subsample.nodiv_result'></span>

<h3>Description</h3>

<p>Takes an object of type <code>distrib_data</code> or <code>nodiv_data</code>, and subsamples it to contain certain species or sites, while keeping the integrity of the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsample(x, ...)
## S3 method for class 'distrib_data'
subsample(x, sites = NULL, species = NULL, ...)
## S3 method for class 'nodiv_data'
subsample(x, sites = NULL, species = NULL, node = NULL, ...)
## S3 method for class 'nodiv_result'
subsample(x, node = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsample_+3A_x">x</code></td>
<td>

<p>An object of type <code>distrib_data</code> or <code>nodiv_data</code> 
</p>
</td></tr>
<tr><td><code id="subsample_+3A_sites">sites</code></td>
<td>

<p>A numeric or character vector identifying the sites to keep in the new object. If specified, the function will subsample the data object to only include the defined sites, and the species that exist there. If sites == \&quot;all\&quot;, all sites will be kept when subsampling for species, even if no species now exist in the sites. 
</p>
</td></tr>
<tr><td><code id="subsample_+3A_species">species</code></td>
<td>

<p>A numeric or character vector identifying the species to keep in the new object. If specified, the function will subsample the data object to only include the defined species, and the sites where they exist. 
</p>
</td></tr>
<tr><td><code id="subsample_+3A_node">node</code></td>
<td>

<p>A numeric or character vector identifying a node in the phylogeny. Only species descending from this node will be kept in the new object, and the sites where they exist.
</p>
</td></tr>
<tr><td><code id="subsample_+3A_...">...</code></td>
<td>

<p>Further arguments to the subsample function
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The return value is a new object of the same type as <code>x</code>
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>


<h3>See Also</h3>

<p><code><a href="#topic+distrib_data">distrib_data</a></code>, ~~~
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:
data(coquettes)
new &lt;- subsample(coquettes, sites = which(richness(coquettes) &gt; 3))
plot(new, show.tip = FALSE)
## End(Not run)
</code></pre>

<hr>
<h2 id='two_color_map'>
Plot the relative amounts of two variables using a purple-green color scheme
</h2><span id='topic+two_color_map'></span>

<h3>Description</h3>

<p>Plots site statistics from objects in the <code>nodiv</code> package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>two_color_map(distrib_data, vec1, vec2, res = 10, showlegend = T, 
  legend_coords = c(0.2, 0.26, 0.36, 0.44), type = c("auto", "grid", "points"), 
  colour = c("green_purple", "green_red_purple_cyan"), ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="two_color_map_+3A_distrib_data">distrib_data</code></td>
<td>

<p>An object of type <code>distrib_data</code>, <code>nodiv_data</code> or <code>nodiv_result</code>, or, alternatively, an object of spatial points from package <code>sp</code>, or a matrix of xy values in two columns
</p>
</td></tr>
<tr><td><code id="two_color_map_+3A_vec1">vec1</code></td>
<td>

<p>A site statistic to be plotted, such as species richness or mean temperature. This can be a vector of values or the name of a site statistic in distrib_data. 
</p>
</td></tr>
<tr><td><code id="two_color_map_+3A_vec2">vec2</code></td>
<td>

<p>The oter site statistic to be plotted.
</p>
</td></tr>
<tr><td><code id="two_color_map_+3A_res">res</code></td>
<td>

<p>The resolution of colors on the color legend
</p>
</td></tr>
<tr><td><code id="two_color_map_+3A_showlegend">showlegend</code></td>
<td>

<p>A boolean specifying whether to show the legend
</p>
</td></tr>
<tr><td><code id="two_color_map_+3A_legend_coords">legend_coords</code></td>
<td>

<p>A vector x1, x2, y1, y2 specifying the location and extent of the color legend
</p>
</td></tr>
<tr><td><code id="two_color_map_+3A_type">type</code></td>
<td>

<p>To specify whether to plot as spatial points or a grid
</p>
</td></tr>
<tr><td><code id="two_color_map_+3A_colour">colour</code></td>
<td>

<p>What colour scheme to use for plotting
</p>
</td></tr>
<tr><td><code id="two_color_map_+3A_...">...</code></td>
<td>

<p>Further arguements to the plotting function
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame giving codes and indices of the colors shown in the plot
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>


<h3>See Also</h3>

<p><code><a href="#topic+distrib_data">distrib_data</a></code>
<code><a href="#topic+plot_points">plot_points</a></code>
</p>

<hr>
<h2 id='update_object'>
Update <code>distrib_data</code> objects to the <code>nodiv</code> version &gt;= 1.1 format
</h2><span id='topic+update_object'></span>

<h3>Description</h3>

<p>Creates the <code>species_stats</code> element
</p>


<h3>Usage</h3>

<pre><code class='language-R'>update_object(distrib_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="update_object_+3A_distrib_data">distrib_data</code></td>
<td>

<p>An object of type <code>distrib_data</code>, <code>nodiv_data</code> or <code>nodiv_result</code>
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated version of the object
</p>


<h3>Author(s)</h3>

<p>Michael Krabbe Borregaard
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
