<!DOCTYPE html><html><head><title>Help for package noia</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {noia}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='# NOIA package'>
<p>Implementation of the Natural and Orthogonal InterAction (NOIA) model</p></a></li>
<li><a href='#Effects names'><p>Names of Genetic Effects</p></a></li>
<li><a href='#Genetic effects'><p> Genetic Effects</p></a></li>
<li><a href='#Genetic regression'><p> Linear and Multilinear Genetic Regressions</p></a></li>
<li><a href='#Genotype names'><p> Names of Genotypes</p></a></li>
<li><a href='#Genotype-to-Phenotype map'><p> Genotype-to-Phenotype Mapping</p></a></li>
<li><a href='#GP map analysis'><p> Noia analysis of genotype-to-phenotype (GP) maps in ideal populations</p></a></li>
<li><a href='#Marginal locus calculation'><p> Estimation of parameters for specific allele frequencies</p></a></li>
<li><a href='#plot.noia'><p> Graphical display of genetic regressions and genotype-phenotype</p>
maps</a></li>
<li><a href='#print.noia'><p> Printing Genetic Regressions and GP map analyses</p></a></li>
<li><a href='#Simulate population'><p> Simulates a Population from a Genotype-Phenotype Map</p></a></li>
<li><a href='#Variance decomposition'><p> Decomposition of Genetic Variance</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Implementation of the Natural and Orthogonal InterAction (NOIA)
Model</td>
</tr>
<tr>
<td>Version:</td>
<td>0.97.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-03-07</td>
</tr>
<tr>
<td>Author:</td>
<td>Arnaud Le Rouzic (2007-2015),  Arne B. Gjuvsland (2010), Olivier Ariste (2010)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Arnaud Le Rouzic &lt;arnaud.le-rouzic@universite-paris-saclay.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>parallel, numDeriv</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/lerouzic/noia">https://github.com/lerouzic/noia</a></td>
</tr>
<tr>
<td>Description:</td>
<td>The NOIA model, as described extensively in Alvarez-Castro &amp; Carlborg (2007), is a framework facilitating the estimation of genetic effects and genotype-to-phenotype maps. This package provides the basic tools to perform linear and multilinear regressions from real populations (provided the phenotype and the genotype of every individuals), estimating the genetic effects from different reference points, the genotypic values, and the decomposition of genetic variances in a multi-locus, 2 alleles system. This package is presented in Le Rouzic &amp; Alvarez-Castro (2008). </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-07 15:53:52 UTC; lerouzic</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-08 08:10:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='+20NOIA+20package'>
Implementation of the Natural and Orthogonal InterAction (NOIA) model
</h2><span id='topic+noia-package'></span><span id='topic+noia'></span>

<h3>Description</h3>

<p>The NOIA model, as described extensively in Alvarez-Castro &amp; 
Carlborg (2007), is a framework facilitating the 
estimation of geneticEffects and genotype-to-phenotype maps. This package
provides the basic tools to perform linear and multilinear regressions from
real populations, analyse pure genotype-to-phenotype (GP) maps in ideal populations, estimating the genetic effects from different reference
points, the genotypic values, and the decomposition of genetic variances in
a multi-locus, 2 alleles system. This package is extensively described in 
Le Rouzic &amp; Alvarez-Castro (2008).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> noia</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.94.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2010-04-20</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong>Regression data set</strong>: The user must provide (i) The vector of phenotypes 
of all individuals measured in 
the population, and (ii) The matrix of the genotypes. There are two input 
formats for the genotype, see <code><a href="#topic+linearRegression">linearRegression</a></code>.
</p>
<p><strong>Regression functions</strong>: <code><a href="#topic+linearRegression">linearRegression</a></code> and 
<code><a href="#topic+multilinearRegression">multilinearRegression</a></code>.
</p>
<p><strong>GP map data set</strong>: The user must provide (i) The <code class="reqn">3^L</code> (where <code class="reqn">L</code> is the number of loci) vector of genotypic values 
(<strong>G</strong> in Alvarez-Castro &amp; Carlborg (2007))
(ii) Allele or genotype frequencies in the reference population.
</p>
<p><strong>GP map analysis function</strong>: <code><a href="#topic+linearGPmapanalysis">linearGPmapanalysis</a></code>.
</p>
<p><strong>Change of reference</strong>: <code><a href="#topic+geneticEffects">geneticEffects</a></code>.
</p>
<p><strong>Genotype-to-phenotype map</strong>: <code><a href="#topic+GPmap">GPmap</a></code>.
</p>
<p><strong>Decomposition of genetic variance</strong>: <code><a href="#topic+varianceDecomposition">varianceDecomposition</a></code>.
</p>


<h3>Author(s)</h3>

<p>Arnaud Le Rouzic, Arne B. Gjuvsland
</p>
<p>Maintainer: Arnaud Le Rouzic
</p>


<h3>References</h3>

<p>Alvarez-Castro JM, Carlborg O. (2007). A unified model for functional 
and statistical epistasis and its application in quantitative trait 
loci analysis. Genetics 176(2):1151-1167.
</p>
<p>Alvarez-Castro JM, Le Rouzic A, Carlborg O. (2008). How to perform meaningful
estimates of genetic effects. PLoS Genetics 4(5):e1000062.
</p>
<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics 4. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123456789)

map &lt;- c(0.25, -0.75, -0.75, -0.75, 2.25, 2.25, -0.75, 2.25, 2.25)
names(map) &lt;- genNames(2)
pop &lt;- simulatePop(map, N=500, sigmaE=0.2, type="F2")

# Regressions

linear &lt;- linearRegression(phen=pop$phen, gen=pop[2:3])

multilinear &lt;- multilinearRegression(phen=pop$phen, gen=cbind(pop$Loc1, 
	pop$Loc2))

# Linear effects, associated variances and stderr
linear

# Multilinear effects
multilinear

# Genotype-to-phenotype map analysis
linearGP &lt;- linearGPmapanalysis(map, reference="F2")

# Linear effects in ideal F2 population
linearGP

# Change of reference: geneticEffects in the "11" genotype (parental 1)
geneticEffects(linear, ref.genotype="P1")

# Variance decomposition
varianceDecomposition(linear)
varianceDecomposition(linearGP)

# GP maps
maps &lt;- cbind(map, GPmap(linear)[,1], GPmap(multilinear)[,1])
colnames(maps) &lt;- c("Actual", "Linear", "Multilinear")
maps
</code></pre>

<hr>
<h2 id='Effects+20names'>Names of Genetic Effects</h2><span id='topic+effectsNamesGeneral'></span><span id='topic+effectsNamesMultilinear'></span>

<h3>Description</h3>

<p>Provides and manipulates labels of genetic effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effectsNamesGeneral(nloc = 2, max.level=NULL, max.dom=NULL) 
effectsNamesMultilinear(nloc=2, max.level=2, max.dom=2)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Effects+2B20names_+3A_nloc">nloc</code></td>
<td>
<p> Number of loci. </p>
</td></tr>
<tr><td><code id="Effects+2B20names_+3A_max.level">max.level</code></td>
<td>
<p> Maximum order of interactions. </p>
</td></tr>
<tr><td><code id="Effects+2B20names_+3A_max.dom">max.dom</code></td>
<td>
<p> Maximum order for dominance. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The codes for genetic effects are stored into a vector of length 4, 
<code>effectsNames</code>. The first element of the vector is the code for 
the absence of effect (default: <code>"."</code>). The three other elements are
respectively additive effects (default: <code>"a"</code>) dominance effects 
(default: <code>"d"</code>), and multilinear epistatic effects (default: 
<code>"e"</code>). 
</p>
<p>The names of genetic effects contains as many characters as the number of
loci in the system. The additive effect of the first locus in a 3-locus
system will be <code>"a.."</code>, and the &quot;Dominance by Dominance&quot; between loci 2 
and 4 in a 5-locus system will be <code>".d.d."</code>. Directionality of epistasis
between two (or more) loci is indicated by as many <code>"e"</code> as necessary 
(e.g. <code>".ee."</code> for the interaction between loci 2 and 3 in a 4-locus
case). 
</p>
<p><code>effectsNamesGeneral</code> and <code>effectsNamesMultilinear</code> provide 
a list of the names of the genetic effects, in the correct order to be
processed in the NOIA framework (Alvarez-Castro and Carlborg 2007). 
</p>


<h3>References</h3>

<p>Alvarez-Castro JM, Carlborg O. (2007). A unified model for functional 
and statistical epistasis and its application in quantitative trait 
loci analysis. Genetics 176(2):1151-1167.
</p>
<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics, 4. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+geneticEffects">geneticEffects</a></code>,   <code><a href="#topic+genNames">genNames</a></code>,   
<code><a href="#topic+linearRegression">linearRegression</a></code>,   <code><a href="#topic+multilinearRegression">multilinearRegression</a></code>. </p>


<h3>Examples</h3>

<pre><code class='language-R'>effectsNamesGeneral(3)
  </code></pre>

<hr>
<h2 id='Genetic+20effects'> Genetic Effects </h2><span id='topic+geneticEffects'></span><span id='topic+effectsNames'></span>

<h3>Description</h3>

<p><code>geneticEffects</code> displays the genetic effects (and their standard 
errors) from the result of <code><a href="#topic+linearRegression">linearRegression</a></code>. If a new
reference point is provided, a &quot;change of reference&quot; operation is performed
(Alvarez-Castro and Carlborg 2007). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneticEffects(obj, reference="P1", ref.genotype = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Genetic+2B20effects_+3A_obj">obj</code></td>
<td>
<p> An object of class <code>"noia.linear"</code> provided by 
<code><a href="#topic+linearRegression">linearRegression</a></code>. </p>
</td></tr>
<tr><td><code id="Genetic+2B20effects_+3A_reference">reference</code></td>
<td>
<p> The new reference point. Can
be <code>"F2"</code>, <code>"F1"</code>, <code>"Finf"</code>, <code>"P1"</code>, <code>"P2"</code> (see
<code><a href="#topic+linearRegression">linearRegression</a></code> for details. </p>
</td></tr>
<tr><td><code id="Genetic+2B20effects_+3A_ref.genotype">ref.genotype</code></td>
<td>
<p> The same as <code>reference</code>, provided for compatibility
with older versions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Variance decomposition and change of reference operation are not possible
from the result of a multilinear regression.
</p>


<h3>Author(s)</h3>

<p>Arnaud Le Rouzic </p>


<h3>References</h3>

 
<p>Alvarez-Castro JM, Carlborg O. (2007). A unified model for functional 
and statistical epistasis and its application in quantitative trait 
loci analysis. Genetics 176(2):1151-1167.
</p>
<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics, 4. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+linearRegression">linearRegression</a></code>,   <code><a href="#topic+multilinearRegression">multilinearRegression</a></code>. </p>


<h3>Examples</h3>

<pre><code class='language-R'>map &lt;- c(0.25, -0.75, -0.75, -0.75, 2.25, 2.25, -0.75, 2.25, 2.25)
pop &lt;- simulatePop(map, N=500, sigmaE=0.2, type="F2")

# Regressions

linear &lt;- linearRegression(phen=pop$phen, gen=cbind(pop$Loc1, pop$Loc2))

geneticEffects(linear, "P1")
</code></pre>

<hr>
<h2 id='Genetic+20regression'> Linear and Multilinear Genetic Regressions </h2><span id='topic+linearRegression'></span><span id='topic+multilinearRegression'></span>

<h3>Description</h3>

<p>The regression aims at estimating genetic effects from a population in which the genotypes and phenotypes are known. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linearRegression(phen, gen=NULL, genZ=NULL, 
    reference="noia", max.level=NULL, max.dom=NULL, fast=FALSE)
multilinearRegression(phen, gen=NULL, genZ=NULL, 
    reference="noia", max.level=NULL, max.dom=NULL, fast=FALSE, 
    e.unique=FALSE, start.algo = "linear", start.values=NULL, 
    robust=FALSE, bilinear.steps=1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Genetic+2B20regression_+3A_phen">phen</code></td>
<td>
<p> The vector of individual phenotypes measured in the population. </p>
</td></tr>
<tr><td><code id="Genetic+2B20regression_+3A_gen">gen</code></td>
<td>
<p> The matrix of individual genotypes in the population, one column per locus. See <code><a href="#topic+genNames">genNames</a></code> for the genotype encoding. Not necessary if <code>genZ</code> is provided. </p>
</td></tr>
<tr><td><code id="Genetic+2B20regression_+3A_genz">genZ</code></td>
<td>
<p> The matrix of individual genotypic probabilities in the population, 3 columns per locus, corresponding of the probability of each of the 3 genotypes (the sum must be 1). Not necessary if <code>gen</code> is provided. </p>
</td></tr>
<tr><td><code id="Genetic+2B20regression_+3A_reference">reference</code></td>
<td>
<p> The reference point from which the regression is performed. By default, the <code>"noia"</code> reference point is used, since it provides a fairly good orthogonality. Other possibilities are <code>"G2A"</code>, <code>"F2"</code>, <code>"F1"</code>, <code>"Finf"</code>, <code>"UWR"</code>, <code>"P1"</code> and <code>"P2"</code>. </p>
</td></tr>
<tr><td><code id="Genetic+2B20regression_+3A_max.level">max.level</code></td>
<td>
<p> Maximum level of interactions. </p>
</td></tr>
<tr><td><code id="Genetic+2B20regression_+3A_max.dom">max.dom</code></td>
<td>
<p> Maximum level for dominance effects. Does not have any effect if &gt;= <code>max.level</code>. In the multilinear regression, the maximum level for dominance effects cannot be &gt; 1. </p>
</td></tr>
<tr><td><code id="Genetic+2B20regression_+3A_fast">fast</code></td>
<td>
<p> This &quot;fast&quot; algorithm should be used when (i) the number of loci is high (&gt; 8) and (ii) there are uncertainties in the dataset (missing values or Haley-Knott regression). This algorithm computes the regression matrix directly function, i.e. without computing <code>Z</code> nor <code>S</code> matrices.</p>
</td></tr>  
<tr><td><code id="Genetic+2B20regression_+3A_e.unique">e.unique</code></td>
<td>
<p> Whether the multilinear term is the same for all pairs. </p>
</td></tr>
<tr><td><code id="Genetic+2B20regression_+3A_start.algo">start.algo</code></td>
<td>
<p> Algorithm used to compute the starting values. Can be <code>"linear"</code>, <code>"multilinear"</code>, <code>"subset"</code> or <code>"bilinear"</code>. Ignored if <code>start.values</code> are provided.</p>
</td></tr>
<tr><td><code id="Genetic+2B20regression_+3A_start.values">start.values</code></td>
<td>
<p> Vector of starting values. </p>
</td></tr>
<tr><td><code id="Genetic+2B20regression_+3A_robust">robust</code></td>
<td>
<p> Tries sequentially all starting values algorithms.</p>
</td></tr>
<tr><td><code id="Genetic+2B20regression_+3A_bilinear.steps">bilinear.steps</code></td>
<td>
<p> Number of steps. Ignored if <code>start.algo</code> is not <code>"bilinear"</code>. If <code>NULL</code>, the bilinear algorithm is run until (almost) convergence.</p>
</td></tr>
<tr><td><code id="Genetic+2B20regression_+3A_...">...</code></td>
<td>
<p> Extra parameters to the non-linear regression function <code><a href="stats.html#topic+nls">nls</a></code>, including <code><a href="stats.html#topic+nls.control">nls.control</a></code>. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a <code>gen</code> data set is provided, it will be turned into a <code>genZ</code>. Missing data (unknown genotypes) are considered as loci for which genotypic probabilities are identical to the genotypic frequencies in the population. 
</p>
<p>The algebraic framework is described extensively in Alvarez-Castro &amp; Carlborg 2007. The default reference point (<code>"noia"</code>) provides an orthogonal decomposition of genetic effects in the 1-locus case, whatever the genotypic frequencies. It remains a good approximation of orthogonality in the multi-locus case if linkage disequilibrium is small. Other optional reference points are those of the <code>"G2A"</code> model (Zeng et al. 2005), and the unweighted regression model <code>"UWR"</code> (Cheverud &amp; Routman, 1995). Several key populations can be taken as reference as well: <code>"F2"</code>, <code>"F1"</code>, <code>"Finf"</code> (F infinity), and the two &quot;parental&quot; homozygous populations <code>"P1"</code> and <code>"P2"</code>. 
</p>
<p>The multilinear model for genetic interactions is an alternative way to model epistatic interactions between at least two loci (see Hansen &amp; Wagner 2001). The computation of multilinear estimates requires a non-linear regression step that relies on the <code><a href="stats.html#topic+nls">nls</a></code> function. Providing good starting values for the non-linear regression is a key to ensure convergence, and different algorithms are provided, that can be specified by the <code>"start.algo"</code> option. <code>"linear"</code> performs a linear regression and approximates the genetic effects from it, while <code>"multilinear"</code> performs a simpler multilinear regression (without dominance) to initialize the genetic effects. <code>"subset"</code> estimate all genetic effects from a random subset (50%) of the population, and <code>"bilinear"</code> estimate alternatively marginal and epistatic effects. 
</p>


<h3>Value</h3>

<p><code>linearRegression</code> and <code>multilinearRegression</code> return an object of class <code>"noia.linear"</code> or <code>"noia.multilinear"</code>, both having their own <code><a href="base.html#topic+print">print</a></code> methods: <code><a href="#topic+print.noia.linear">print.noia.linear</a></code> and <code><a href="#topic+print.noia.multilinear">print.noia.multilinear</a></code>. 
</p>


<h3>Author(s)</h3>

<p> Arnaud Le Rouzic </p>


<h3>References</h3>

 
<p>Alvarez-Castro JM, Carlborg O. (2007). A unified model for functional 
and statistical epistasis and its application in quantitative trait 
loci analysis. Genetics 176(2):1151-1167.
</p>
<p>Alvarez-Castro JM, Le Rouzic A, Carlborg O. (2008). How to perform meaningful
estimates of genetic effects. PLoS Genetics 4(5):e1000062.
</p>
<p>Cheverud JM, Routman, EJ. (1995). Epistasis and its contribution to 
genetic variance components. Genetics 139:1455-1461.
</p>
<p>Hansen TF, Wagner G. (2001) Modeling genetic architecture: A multilinear 
theory of gene interactions. Theoretical Population Biology 59:61-86.
</p>
<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics 4. 
</p>
<p>Zeng ZB, Wang T, Zou W. (2005). Modelling quantitative trait loci and 
interpretation of models. Genetics 169: 1711-1725. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+geneticEffects">geneticEffects</a></code>, <code><a href="#topic+GPmap">GPmap</a></code>, <code><a href="#topic+varianceDecomposition">varianceDecomposition</a></code>. </p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123456789)

map &lt;- c(0.25, -0.75, -0.75, -0.75, 2.25, 2.25, -0.75, 2.25, 2.25)
pop &lt;- simulatePop(map, N=500, sigmaE=0.2, type="F2")

# Regressions

linear &lt;- linearRegression(phen=pop$phen, gen=cbind(pop$Loc1, pop$Loc2))

multilinear &lt;- multilinearRegression(phen=pop$phen, 
    gen=cbind(pop$Loc1, pop$Loc2))

# Linear effects, associated variances and stderr
linear

# Multilinear effects
multilinear
</code></pre>

<hr>
<h2 id='Genotype+20names'> Names of Genotypes </h2><span id='topic+genNames'></span><span id='topic+genotypesNames'></span>

<h3>Description</h3>

<p><code>genNames</code> provides the names of all possible genotypes in the 
order required by the NOIA model (Alvarez-Castro and Carlborg 2007). The codes for the genotypes are
stored in the vector <code>genotypesNames</code>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genNames(nloc = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Genotype+2B20names_+3A_nloc">nloc</code></td>
<td>
<p> Number of loci</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The names of the genotypes are stored in the vector 
<code>genotypesNames</code>. By default, they are <code>"1"</code>, <code>"2"</code>, and 
<code>"3"</code>, the heterozygotes being <code>"2"</code>. The genotypes at different loci
are then put together, such as <code>"123"</code> for 3 loci. 
</p>


<h3>Author(s)</h3>

<p> Arnaud Le Rouzic </p>


<h3>References</h3>

 
<p>Alvarez-Castro JM, Carlborg O. (2007). A unified model for functional 
and statistical epistasis and its application in quantitative trait 
loci analysis. Genetics 176(2):1151-1167.
</p>
<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics, 4. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    genNames(3)
</code></pre>

<hr>
<h2 id='Genotype-to-Phenotype+20map'> Genotype-to-Phenotype Mapping </h2><span id='topic+GPmap'></span>

<h3>Description</h3>

<p>The Genotype-to-Phenotype map is a vector providing the estimate of 
the genotypic value for any multi-locus genotype. The estimates may
be computed from <code><a href="#topic+linearRegression">linearRegression</a></code> or 
<code><a href="#topic+multilinearRegression">multilinearRegression</a></code>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GPmap(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Genotype-to-Phenotype+2B20map_+3A_obj">obj</code></td>
<td>
<p> An object of class <code>"noia.linear"</code> or 
<code>"noia.multilinear"</code>. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with two columns: the first one is the estimate of genotypic
effects, the second one the standard error of this estimate.
</p>


<h3>Author(s)</h3>

<p> Arnaud Le Rouzic </p>


<h3>References</h3>

<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics, 4. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+linearRegression">linearRegression</a></code>, <code><a href="#topic+multilinearRegression">multilinearRegression</a></code>, 
<code><a href="#topic+genNames">genNames</a></code>. </p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123456789)

map &lt;- c(0.25, -0.75, -0.75, -0.75, 2.25, 2.25, -0.75, 2.25, 2.25)
pop &lt;- simulatePop(map, N=500, sigmaE=0.2, type="F2")

# Regression
linear &lt;- linearRegression(phen=pop$phen, gen=cbind(pop$Loc1, pop$Loc2))

# GP map
GPmap(linear)
</code></pre>

<hr>
<h2 id='GP+20map+20analysis'> Noia analysis of genotype-to-phenotype (GP) maps in ideal populations </h2><span id='topic+linearGPmapanalysis'></span>

<h3>Description</h3>

<p> Functions for doing a NOIA analysis of a GP map for <code class="reqn">L</code> loci in a population where the loci are in complete linkage equilibrium. </p>


<h3>Usage</h3>

<pre><code class='language-R'>linearGPmapanalysis(gmap, reference="F2", freqmat=NULL, 
                    max.level=NULL , S_full=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GP+2B20map+2B20analysis_+3A_gmap">gmap</code></td>
<td>
<p> Vector of length <code class="reqn">3^L</code> with genotypic values for all possible genotypes in the order defined by <code><a href="#topic+genNames">genNames</a></code>. </p>
</td></tr>
<tr><td><code id="GP+2B20map+2B20analysis_+3A_reference">reference</code></td>
<td>
<p> The reference population in which the analysis is done. By default, the <code>"F2"</code> population is used. Other possibilities are <code>"noia"</code>, <code>"G2A"</code>, <code>"UWR"</code>. </p>
</td></tr>
<tr><td><code id="GP+2B20map+2B20analysis_+3A_freqmat">freqmat</code></td>
<td>
<p> For <code>reference="G2A"</code>: A vector of length <code class="reqn">L</code> containing allele frequencies such that 
<code>freqmat[i]=frequency(allele 1)</code> for locus <code>i</code>.
</p>
<p>For <code>reference="noia"</code>: A <code class="reqn">(L\times3)</code> matrix of genotype frequencies such that 
<code>freqmat[i,]=[frequency(1) frequency(2) frequency(3)]</code> for locus <code>i</code>. </p>
</td></tr>
<tr><td><code id="GP+2B20map+2B20analysis_+3A_max.level">max.level</code></td>
<td>
<p> Maximum level of interactions. </p>
</td></tr>
<tr><td><code id="GP+2B20map+2B20analysis_+3A_s_full">S_full</code></td>
<td>
<p> Boolean argument indicating whether to keep full <code>S</code> matrix <code class="reqn">(3^L\times3^L)</code> in memory or alternatively to keep 
<code class="reqn">L</code> single locus <code>S</code> matrices <code class="reqn">(3\times3)</code> and compute single row and columns of the full matrix. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The algebraic framework is described extensively in Alvarez-Castro &amp; Carlborg 2007. When analysing GP maps in ideal populations
we can work directly with the <code>S</code> matrix and do not have to consider the <code>X</code> and <code>Z</code> matrices used in <code><a href="#topic+linearRegression">linearRegression</a></code>.
When it comes to the <code>S_full</code> argument keeping the multilocus <code>S</code> matrix in memory is generally fastest for computing all <code class="reqn">3^L</code> 
genetic effects. However it does not allow for computing only a subset of the effects and also runs out of memory for <code class="reqn">L&gt;8</code> on a typical desktop machine.
For S_full=NULL in <code>linearGPmapanalysis</code> a full <code>S</code> matrix is used if <code class="reqn">L&lt;=8</code> and max.level=NULL, while <code class="reqn">L</code> single locus <code>S</code> matrices are used otherwise.  
</p>


<h3>Value</h3>

<p><code>linearGPmapanalysis</code> returns an object of class <code>"noia.linear.gpmap"</code> , with its own <code><a href="base.html#topic+print">print</a></code> method: <code><a href="#topic+print.noia.linear.gpmap">print.noia.linear.gpmap</a></code>. 
</p>


<h3>Author(s)</h3>

<p> Arne B. Gjuvsland </p>


<h3>References</h3>

 
<p>Alvarez-Castro JM, Carlborg O. (2007). A unified model for functional 
and statistical epistasis and its application in quantitative trait 
loci analysis. Genetics 176(2):1151-1167.
</p>
<p>Cheverud JM, Routman, EJ. (1995). Epistasis and its contribution to 
genetic variance components. Genetics 139:1455-1461.
</p>
<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics 4. 
</p>
<p>Zeng ZB, Wang T, Zou W. (2005). Modelling quantitative trait loci and 
interpretation of models. Genetics 169: 1711-1725. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+varianceDecomposition">varianceDecomposition</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>map &lt;- c(0.25, -0.75, -0.75, -0.75, 2.25, 2.25, -0.75, 2.25, 2.25)

# Genotype-to-phenotype map analysis
linearGP &lt;- linearGPmapanalysis(map, reference="F2")

# Linear effects in ideal F2 population
linearGP

</code></pre>

<hr>
<h2 id='Marginal+20locus+20calculation'> Estimation of parameters for specific allele frequencies </h2><span id='topic+marginallocus'></span><span id='topic+plot.noia.marloc'></span><span id='topic+image.noia.marloc'></span>

<h3>Description</h3>

 
<p>This function computes some parameters of interest 
(mean phenotype, genetic variance, additive variance, and evolutionary change in additive variance)
for a combination of allele frequencies, based on a genotype-phenotype map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>marginallocus(gmap, freq=NULL, what="mean", definition=11, mc.cores=1, ...)
## S3 method for class 'noia.marloc'
plot(x, xlab=NULL, ylim=NULL, ylab=attr(x, "what"), ...)
## S3 method for class 'noia.marloc'
image(x, xlab=NULL, ylab=NULL, zlim=NULL, 
      main=attr(x, "what"), col.max="red", col.min="blue", col.zero="white", 
      n.cols=1000, zeropart=0.01, contour.levels=10, contour.options=list(), ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_gmap">gmap</code></td>
<td>
<p> Either an object of class <code>noia.gpmap</code>, or a vector of phenotypic values in the order defined as in <code><a href="#topic+genotypesNames">genotypesNames</a></code> </p>
</td></tr></table>
<p>.
</p>
<table>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_freq">freq</code></td>
<td>
<p> A vector indicating the loci that should be analysed. See Details. </p>
</td></tr>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_what">what</code></td>
<td>
<p> A character string among &quot;mean&quot;, &quot;varA&quot;, &quot;varG&quot;, or &quot;dvarA.dt&quot;. </p>
</td></tr>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_definition">definition</code></td>
<td>
<p> The number of allele frequencies to try for each locus. </p>
</td></tr>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_mc.cores">mc.cores</code></td>
<td>
<p> If more than 1, the calculation is run on <code>mc.cores</code> cores via the library <code><a href="lattice.html#topic+parallel">parallel</a></code>. </p>
</td></tr>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_x">x</code></td>
<td>
<p> An object of class <code>noia.marloc</code> obtained after running <code>marginallocus</code>. </p>
</td></tr>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_col.max">col.max</code>, <code id="Marginal+2B20locus+2B20calculation_+3A_col.min">col.min</code>, <code id="Marginal+2B20locus+2B20calculation_+3A_col.zero">col.zero</code></td>
<td>
<p> Colors standing for the maximal, minimal, and nil values, respectively. Setting <code>col.zero</code> to <code>NULL</code> generates a color gradient between <code>col.min</code> and <code>col.max</code>. </p>
</td></tr>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_n.cols">n.cols</code></td>
<td>
<p> Number of colors in the gradient. </p>
</td></tr>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_zeropart">zeropart</code></td>
<td>
<p> Width (relative to the full amplitude) of the region around zero which will be colored as <code>col.zero</code>. </p>
</td></tr>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_contour.levels">contour.levels</code></td>
<td>
<p> Number of contour lines. Setting this to 0 leads to no contour lines. </p>
</td></tr>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_contour.options">contour.options</code></td>
<td>
<p> List of additional options to the <code><a href="graphics.html#topic+contour">contour</a></code> function. </p>
</td></tr>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_xlab">xlab</code>, <code id="Marginal+2B20locus+2B20calculation_+3A_ylab">ylab</code>, <code id="Marginal+2B20locus+2B20calculation_+3A_ylim">ylim</code>, <code id="Marginal+2B20locus+2B20calculation_+3A_zlim">zlim</code>, <code id="Marginal+2B20locus+2B20calculation_+3A_main">main</code></td>
<td>
<p> Classical parameters passed to <code><a href="graphics.html#topic+plot">plot</a></code> and <code><a href="Matrix.html#topic+image">image</a></code>. </p>
</td></tr>
<tr><td><code id="Marginal+2B20locus+2B20calculation_+3A_...">...</code></td>
<td>
<p> Additional parameters to internal functions. </p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>marginallocus</code> computes a population parameter for a series of allele frequencies. The loci under investigation are provided through the <code>freq</code> vector, which need to have as many elements as loci in the system. Values of the <code>freq</code> vector indicate fixed allele frequencies, while <code>NA</code> indicate loci under investigation. For instance, <code>freq=c(NA, 1, NA, 0.5)</code>, will investigate the effect of varying loci 1 and 3, while keeping loci 2 and 4 at constant allele frequencies. The population is assumed to be at Hardy-Weinberg frequencies. If <code>freq</code> is not provided, all loci will be investigated.
</p>


<h3>Value</h3>

<p><code>marginallocus</code> returns an array with as many dimensions as loci under investigation. This array is an object of class <code>"noia.marloc"</code> which can be graphically illustrated through the provided <code>plot</code> (for 1-dimensional data) and <code>image</code> (for 2-dimensional data). Arrays of higher dimensionality cannot be represented graphically.
</p>


<h3>Author(s)</h3>

<p> Arnaud Le Rouzic </p>


<h3>See Also</h3>

 <p><code><a href="#topic+linearGPmapanalysis">linearGPmapanalysis</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
map &lt;- c(0.25, -0.75, -0.75, -0.75, 2.25, 2.25, -0.75, 2.25, 2.25)

mrg2D &lt;- marginallocus(map)
mrg1D &lt;- marginallocus(map, freq=c(NA, 0)) # the second locus is fixed for allele 1

image(mrg2D)
plot(mrg1D)
</code></pre>

<hr>
<h2 id='plot.noia'> Graphical display of genetic regressions and genotype-phenotype 
maps </h2><span id='topic+plot.noia.linear'></span><span id='topic+barplot.noia.gpmap'></span>

<h3>Description</h3>

<p> These functions allow a graphic representation of the 
result of genetic regressions from <code><a href="#topic+linearRegression">linearRegression</a></code>
and <code><a href="#topic+GPmap">GPmap</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'noia.linear'
plot(x, loc = 1:x$nloc, effect=TRUE, epistasis = TRUE, 
ylim=range(GPmap(x)[,1]) + c(-1,1)*max(GPmap(x)[,2]), ...)
## S3 method for class 'noia.gpmap'
barplot(height, GPcol = c("indianred", "palegreen", "royalblue"), 
arrowscol = "purple", stderr = TRUE , main=NA, ylab=NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.noia_+3A_x">x</code></td>
<td>
<p> An object of class <code>"noia.linear"</code> for the 
<code><a href="graphics.html#topic+plot">plot</a></code> function, or of class <code>"noia.gpmap"</code> for the
<code><a href="graphics.html#topic+barplot">barplot</a></code> function. </p>
</td></tr>
<tr><td><code id="plot.noia_+3A_loc">loc</code></td>
<td>
<p> The vector loci to plot (by default, all of them are 
displayed).</p>
</td></tr>
<tr><td><code id="plot.noia_+3A_effect">effect</code></td>
<td>
<p> Whether genetic effects have to be plotted for each
locus.</p>
</td></tr>
<tr><td><code id="plot.noia_+3A_epistasis">epistasis</code></td>
<td>
<p> Whether pairwise effects have to be plotted. </p>
</td></tr>
<tr><td><code id="plot.noia_+3A_height">height</code></td>
<td>
<p> An object of class <code>"noia.gpmap"</code>. </p>
</td></tr>
<tr><td><code id="plot.noia_+3A_gpcol">GPcol</code></td>
<td>
<p> Colors for each of the three genotypes. </p>
</td></tr>
<tr><td><code id="plot.noia_+3A_arrowscol">arrowscol</code></td>
<td>
<p>Color of the error bars.</p>
</td></tr>
<tr><td><code id="plot.noia_+3A_stderr">stderr</code></td>
<td>
<p> If <code>TRUE</code>, error bars stand for starndard errors.
Otherwise, error bars are 95% condidence intervals.</p>
</td></tr>
<tr><td><code id="plot.noia_+3A_main">main</code></td>
<td>
<p> The same as in <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>   
<tr><td><code id="plot.noia_+3A_ylab">ylab</code></td>
<td>
<p> The same as in <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr> 
<tr><td><code id="plot.noia_+3A_ylim">ylim</code></td>
<td>
<p> The same as in <code><a href="graphics.html#topic+plot">plot</a></code>.</p>
</td></tr>  
<tr><td><code id="plot.noia_+3A_...">...</code></td>
<td>
<p> Additional options for the <code><a href="graphics.html#topic+plot">plot</a></code> and 
<code><a href="graphics.html#topic+barplot">barplot</a></code> routines. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Olivier Ariste, Arnaud Le Rouzic </p>

<hr>
<h2 id='print.noia'> Printing Genetic Regressions and GP map analyses </h2><span id='topic+print.noia.common'></span><span id='topic+print.noia.linear'></span><span id='topic+print.noia.multilinear'></span><span id='topic+print.noia.linear.gpmap'></span>

<h3>Description</h3>

<p>Display the output of functions <code><a href="#topic+linearRegression">linearRegression</a></code>,
<code><a href="#topic+multilinearRegression">multilinearRegression</a></code> and <code><a href="#topic+linearGPmapanalysis">linearGPmapanalysis</a></code> 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'noia.linear'
print(x, ...)
## S3 method for class 'noia.multilinear'
print(x, ...)
## S3 method for class 'noia.common'
print(x, ...)
## S3 method for class 'noia.linear.gpmap'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.noia_+3A_x">x</code></td>
<td>
<p> An object of class <code>"noia.linear"</code>, class <code>"noia.linear.gpmap"</code> or class <code>"noia.multilinear"</code>. </p>
</td></tr>
<tr><td><code id="print.noia_+3A_...">...</code></td>
<td>
<p> No effect for the moment. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>print</code> method being actually very similar for the linear and 
multilinear regressions, both call the common method <code>print.noia.common</code>. 
</p>


<h3>Author(s)</h3>

<p> Arnaud Le Rouzic, Arne B. Gjuvsland </p>


<h3>References</h3>

 
<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics, 4. 
</p>

<hr>
<h2 id='Simulate+20population'> Simulates a Population from a Genotype-Phenotype Map </h2><span id='topic+simulatePop'></span>

<h3>Description</h3>

<p>The <code>simulatePop</code> function takes a Genotype-to-Phenotype map (i.e. a vector
defining the genotypic value of all possible genotypes) and 
returns a data frame containing the simulated population. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulatePop(gmap, N = 100, sigmaE = 1, type = "F2", freqmat=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Simulate+2B20population_+3A_gmap">gmap</code></td>
<td>
<p> The Genotype-to-phenotype map: a vector of size <code class="reqn">3^L</code>, where L is
the number of loci. The vector should be named with the code of each genotype
(see <code><a href="#topic+genNames">genNames</a></code>.</p>
</td></tr>
<tr><td><code id="Simulate+2B20population_+3A_n">N</code></td>
<td>
<p> Number of individuals. </p>
</td></tr>
<tr><td><code id="Simulate+2B20population_+3A_sigmae">sigmaE</code></td>
<td>
<p> Standard deviation of the environmental noise (normally
distributed). </p>
</td></tr>
<tr><td><code id="Simulate+2B20population_+3A_type">type</code></td>
<td>
<p> Type of population. <code>"F2"</code>, <code>"Finf"</code>, <code>"F1"</code>, <code>"UWR"</code>, <code>"G2A"</code>, and <code>"noia"</code> are possible. </p>
</td></tr>
<tr><td><code id="Simulate+2B20population_+3A_freqmat">freqmat</code></td>
<td>
<p> For <code>type="G2A"</code>: A vector of length <code>nloc</code> containing allele frequencies such that 
<code>freqmat[i]=frequency(allele 1)</code> for locus <code>i</code>.
</p>
<p>For <code>type="noia"</code>: A <code>(nloc x 3)</code> matrix of genotype frequencies such that 
<code>freqmat[i,]=[frequency(1) frequency(2) frequency(3)]</code> for locus <code>i</code>. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The type of population refers to the expected allelic and genotypic frequences:
</p>

<ul>
<li><p>&quot;F1&quot;First generation of an intercross between two parental populations fixed for alleles A and B respectively; expected genotypic frequencies are: AA: 0, AB: 1, BB: 0.
</p>
</li>
<li><p>&quot;F2&quot;Second generation of an intercross between two parental populations fixed for alleles A and B respectively; expected genotypic frequencies are AA: 0.25, AB: 0.5, BB: 0.25.
</p>
</li>
<li><p>&quot;Finf&quot;Theoretical population from an infinite number of generations after an intercross between two parental populations fixed for alleles A and B respectively; expected genotypic frequencies are AA: 0.5, AB: 0, BB: 0.5.
</p>
</li>
<li><p>&quot;UWR&quot;Theoretical population corresponding to ideal (but experimentally unrealistic) equal genotypic frequencies; expected genotypic frequencies are AA: 0.333, AB: 0.333, BB: 0.333. In such a population, the &quot;UnWeighted Regression model&quot; (UWR) by Cheverud and Routman 1995 provides orthogonal estimates.
</p>
</li>
<li><p>&quot;G2A&quot;Population at Hardy-Weinberg frequencies; expected genotypic frequencies are: AA: p*p, AB: 2p(1-p), BB: (1-p)(1-p), the frequency of allele A (p) at locus i being provided by the i-th element of vector <code>freqmat</code>. &quot;G2A&quot; is the name of the statistical model by Zeng et al. (2005) in which genetic effects estimated from such a population are orthogonal.
</p>
</li>
<li><p>&quot;noia&quot;Population in which genotypic frequencies are arbitrary; expected genotypic frequencies are: AA: pAA, AB: pAB, BB: pBB, frequences pAA, pAB, and pBB at locus i being provided by the i-th line of matrix <code>freqmat</code>. &quot;noia&quot; is the name of the statistical model by Alvarez-Castro and Carlborg (2007) in which genetic effects estimated from such a population are orthogonal. In all populations, loci are considered as independent and are at linkage equilibrium.         
</p>
</li></ul>



<h3>Value</h3>

<p>Returns a data frame, in which the first column (<code>$phen</code>) contains the 
phenotypes, and the following ones (<code>$Loc1</code>, <code>$loc2</code>, etc) the 
genotypes of all individuals.
</p>


<h3>Author(s)</h3>

<p> Arnaud Le Rouzic, Arne B. Gjuvsland </p>


<h3>References</h3>

    
<p>Alvarez-Castro JM, Carlborg O. (2007). A unified model for functional 
and statistical epistasis and its application in quantitative trait 
loci analysis. Genetics 176(2):1151-1167.
</p>
<p>Cheverud JM, Routman, EJ. (1995). Epistasis and its contribution to 
genetic variance components. Genetics 139:1455-1461.
</p>
<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics, 4. 
</p>
<p>Zeng ZB, Wang T, Zou W. (2005). Modelling quantitative trait loci and 
interpretation of models. Genetics 169: 1711-1725. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+GPmap">GPmap</a></code>,   <code><a href="#topic+genNames">genNames</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123456789)

map &lt;- c(0.25, -0.75, -0.75, -0.75, 2.25, 2.25, -0.75, 2.25, 2.25)
pop &lt;- simulatePop(map, N=500, sigmaE=0.2, type="F2")
str(pop)

## Create a "noia" population with genotype frequencies 1/3,1/3,1/3 for locus 1 
## and 0.2,0.6,0.2 for locus 2
pop = simulatePop(map, N=1000, sigma=1, type='noia', 
  freqmat=matrix(c(1/3,1/3,1/3,0.2,0.6,0.2),nrow=2, byrow=TRUE))
</code></pre>

<hr>
<h2 id='Variance+20decomposition'> Decomposition of Genetic Variance </h2><span id='topic+varianceDecomposition'></span><span id='topic+print.noia.vardec'></span>

<h3>Description</h3>

<p>Variance decomposition in a classical operation in quantitative genetics
(e.g. Fisher 1918, Lynch and Walsh 1998).
The genetic variance, i.e. the part of phenotypic variance that can be
identify as due to genetic factors, can be decomposed into several
orthogonal components (generally, the part due to additive factors Var(A), 
to dominance factors Var(D), and to genetic interactions Var(I)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varianceDecomposition(obj)
## S3 method for class 'noia.vardec'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Variance+2B20decomposition_+3A_obj">obj</code></td>
<td>
<p> An object of class <code>"noia.linear"</code>, the output of 
<code><a href="#topic+linearRegression">linearRegression</a></code> or of class <code>"noia.linear.gpmap"</code>, the output of 
<code><a href="#topic+linearGPmapanalysis">linearGPmapanalysis</a></code>. </p>
</td></tr>
<tr><td><code id="Variance+2B20decomposition_+3A_x">x</code></td>
<td>
<p> An object of class <code>"noia.vardec"</code>, the output of 
<code><a href="#topic+varianceDecomposition">varianceDecomposition</a></code>. </p>
</td></tr>
<tr><td><code id="Variance+2B20decomposition_+3A_...">...</code></td>
<td>
<p> No effect for the moment. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The details of the variance decomposition are provided for all levels of
interaction: Var(A) and Var(D) for marginal effects, Var(AA), Var(AD) and Var(DD) 
for 2nd order interactions, etc.
</p>


<h3>Value</h3>

<p><code>varianceDecomposition</code> returns a list of vectors. Each element of the list corresponds
to an order of interactions, and the vectors detail the variance 
decomposition within each level. 
<code>print.noia.vardec</code> prints the previous list in a nice way, and computed the percentage of
genetic variance explained by each variance component. 
</p>


<h3>Author(s)</h3>

<p> Arnaud Le Rouzic </p>


<h3>References</h3>

 
<p>Alvarez-Castro JM, Carlborg O. (2007). A unified model for functional 
and statistical epistasis and its application in quantitative trait 
loci analysis. Genetics 176(2):1151-1167.
</p>
<p>Fisher RA. (1918). The correlation between relatives on the supposition of 
Mendelian inheritance. Thans. Roy. Soc. Edinburgh 52:339-433. 
</p>
<p>Le Rouzic A, Alvarez-Castro JM. (2008). Estimation of genetic effects and
genotype-phenotype maps. Evolutionary Bioinformatics, 4. 
</p>
<p>Lynch M, Walsh B (1998) Genetics and Analysis of Quantitative Traits. 
Sunderland, MA; Sinauer Associates. 
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+linearRegression">linearRegression</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>
map &lt;- c(0.25, -0.75, -0.75, -0.75, 2.25, 2.25, -0.75, 2.25, 2.25)
pop &lt;- simulatePop(map, N=500, sigmaE=0.2, type="F2")

# Regression

linear &lt;- linearRegression(phen=pop$phen, gen=cbind(pop$Loc1, pop$Loc2))

# Variance decomposition
varianceDecomposition(linear)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
