<!DOCTYPE html><html><head><title>Help for package noise</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {noise}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#computeExtrinsicNoise'>
<p>Compute estimates of the extrinsic noise.</p></a></li>
<li><a href='#computeExtrinsicNoiseKnownCor'>
<p>Compute estimates of the extrinsic noise using a known correlation in the min-MSE estimator.</p></a></li>
<li><a href='#computeIntrinsicNoise'>
<p>Compute estimates of the intrinsic noise.</p></a></li>
<li><a href='#computeNoiseForSubset'>
<p>Select a random subset of data for noise estimation.</p></a></li>
<li><a href='#elowitz_data'>
<p>Expression of reporter genes in the D22 and M22 E. coli cells from Elowitz et al (2002).</p></a></li>
<li><a href='#simulateSC'><p>Simulate single-cell expression levels of two reporters</p></a></li>
<li><a href='#yang_nl10'>
<p>Expression of reporter genes in the NL010 E. coli cells from Yang et al (2014).</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimation of Intrinsic and Extrinsic Noise from Single-Cell
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-29</td>
</tr>
<tr>
<td>Author:</td>
<td>Audrey Qiuyan Fu and Lior Pachter</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Audrey Q. Fu &lt;audreyqyfu@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions to calculate estimates of intrinsic and extrinsic noise from the two-reporter single-cell experiment, as in Elowitz, M. B., A. J. Levine, E. D. Siggia, and P. S. Swain (2002) Stochastic gene expression in a single cell. Science, 297, 1183-1186.  Functions implement multiple estimators developed for unbiasedness or min Mean Squared Error (MSE) in Fu, A. Q. and Pachter, L. (2016). Estimating intrinsic and extrinsic noise from single-cell gene expression measurements. Statistical Applications in Genetics and Molecular Biology, 15(6), 447-471.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>preprocessCore</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-29 14:11:50 UTC; audreyq.fu</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-29 14:40:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='computeExtrinsicNoise'>
Compute estimates of the extrinsic noise.
</h2><span id='topic+computeExtrinsicNoise'></span>

<h3>Description</h3>

<p>This function computes several estimates of the extrinsic noise (unscaled by the mean).  The estimators, described in Fu and Pachter (2016), include the original estimators developed in Elowitz et al. (2002), the unbiased estimator, an min-MSE estimator, and an asymptotic estimator for large sample sizes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computeExtrinsicNoise(reporter1, reporter2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="computeExtrinsicNoise_+3A_reporter1">reporter1</code></td>
<td>

<p>A vector of continuous values.  
</p>
</td></tr>
<tr><td><code id="computeExtrinsicNoise_+3A_reporter2">reporter2</code></td>
<td>

<p>A vector of continuous values.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Four (unscaled) estimates of extrinsic noise: the original estimators developed in Elowitz et al. (2002), the unbiased estimator, an min-MSE estimator, and an asymptotic estimator for large sample sizes.
</p>


<h3>Author(s)</h3>

<p>Audrey Q. Fu
</p>


<h3>References</h3>

<p>Fu, A. Q. and Pachter, L. (2016). Estimating intrinsic and extrinsic noise from single-cell gene expression measurements.  arXiv:1601.03334.
Elowitz, M. B., A. J. Levine, E. D. Siggia, and P. S. Swain (2002) Stochastic gene expression in a single cell. Science, 297, 1183-1186.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+computeIntrinsicNoise">computeIntrinsicNoise</a></code>, <code><a href="#topic+simulateSC">simulateSC</a></code>.  See estimates for data <code><a href="#topic+elowitz_data">elowitz_data</a></code> and <code><a href="#topic+yang_nl10">yang_nl10</a></code>.
</p>

<hr>
<h2 id='computeExtrinsicNoiseKnownCor'>
Compute estimates of the extrinsic noise using a known correlation in the min-MSE estimator.
</h2><span id='topic+computeExtrinsicNoiseKnownCor'></span>

<h3>Description</h3>

<p>This function is similar to <code><a href="#topic+computeExtrinsicNoise">computeExtrinsicNoise</a></code>, and computes several estimates of the extrinsic noise (unscaled by the mean).  The estimators, described in Fu and Pachter (2016), include the original estimators developed in Elowitz et al. (2002), the unbiased estimator, an min-MSE estimator, and an asymptotic estimator for large sample sizes.  The only difference between this function calculates the min-MSE estimate using a given correlation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computeExtrinsicNoiseKnownCor(reporter1, reporter2, true.cor)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="computeExtrinsicNoiseKnownCor_+3A_reporter1">reporter1</code></td>
<td>

<p>A vector of continuous values.  
</p>
</td></tr>
<tr><td><code id="computeExtrinsicNoiseKnownCor_+3A_reporter2">reporter2</code></td>
<td>

<p>A vector of continuous values.
</p>
</td></tr>
<tr><td><code id="computeExtrinsicNoiseKnownCor_+3A_true.cor">true.cor</code></td>
<td>

<p>A scalar.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Four (unscaled) estimates of extrinsic noise: the original estimators developed in Elowitz et al. (2002), the unbiased estimator, an min-MSE estimator (using the given correlation), and an asymptotic estimator for large sample sizes.
</p>


<h3>Author(s)</h3>

<p>Audrey Q. Fu
</p>


<h3>References</h3>

<p>Fu, A. Q. and Pachter, L. (2016). Estimating intrinsic and extrinsic noise from single-cell gene expression measurements.  arXiv:1601.03334.
Elowitz, M. B., A. J. Levine, E. D. Siggia, and P. S. Swain (2002) Stochastic gene expression in a single cell. Science, 297, 1183-1186.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+computeExtrinsicNoise">computeExtrinsicNoise</a></code>, <code><a href="#topic+simulateSC">simulateSC</a></code>.
</p>

<hr>
<h2 id='computeIntrinsicNoise'>
Compute estimates of the intrinsic noise.
</h2><span id='topic+computeIntrinsicNoise'></span>

<h3>Description</h3>

<p>This function computes several estimates of the intrinsic noise (unscaled by the mean).  The estimators, described in Fu and Pachter (2016), include the original estimators developed in Elowitz et al. (2002), unbiased estimators with and without assuming equal mean of the two reporters, min-MSE estimators with and without assuming equal mean, and asymptotic estimators for large sample sizes with and without assuming equal mean.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computeIntrinsicNoise(reporter1, reporter2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="computeIntrinsicNoise_+3A_reporter1">reporter1</code></td>
<td>

<p>A vector of continuous values.  
</p>
</td></tr>
<tr><td><code id="computeIntrinsicNoise_+3A_reporter2">reporter2</code></td>
<td>

<p>A vector of continuous values.  
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Six (unscaled) estimates of intrinsic noise: the original estimators developed in Elowitz et al. (2002), unbiased estimators with and without assuming equal mean of the two reporters, min-MSE estimators with and without assuming equal mean, and asymptotic estimators for large sample sizes with and without assuming equal mean.
</p>


<h3>Author(s)</h3>

<p>Audrey Q. Fu
</p>


<h3>References</h3>

<p>Fu, A. Q. and Pachter, L. (2016). Estimating intrinsic and extrinsic noise from single-cell gene expression measurements.  arXiv:1601.03334.
Elowitz, M. B., A. J. Levine, E. D. Siggia, and P. S. Swain (2002) Stochastic gene expression in a single cell. Science, 297, 1183-1186.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+computeExtrinsicNoise">computeExtrinsicNoise</a></code>, <code><a href="#topic+simulateSC">simulateSC</a></code>.  See estimates for data <code><a href="#topic+elowitz_data">elowitz_data</a></code> and <code><a href="#topic+yang_nl10">yang_nl10</a></code>.
</p>

<hr>
<h2 id='computeNoiseForSubset'>
Select a random subset of data for noise estimation.
</h2><span id='topic+computeNoiseForSubset'></span>

<h3>Description</h3>

<p>This function randomly selects a subset of cells (rows) from the data set, computes multiple estimates of intrinsic and extrinsic noise, as well as their mean and standard deviation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computeNoiseForSubset(data, sample.size, n.iter)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="computeNoiseForSubset_+3A_data">data</code></td>
<td>

<p>A numeric matrix of two columns.  Each row is a cell, and each column expression of a reporter gene.
</p>
</td></tr>
<tr><td><code id="computeNoiseForSubset_+3A_sample.size">sample.size</code></td>
<td>

<p>An integer that specifies the number of cells in the subset.
</p>
</td></tr>
<tr><td><code id="computeNoiseForSubset_+3A_n.iter">n.iter</code></td>
<td>

<p>An integer that specifies the number of iterations (for calcuation of mean and standard deviation).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that consists of the following components:
</p>
<table>
<tr><td><code>intrinsic</code></td>
<td>
<p>A numeric matrix of esimated intrinsic noise. 7 rows and n.iter columns.</p>
</td></tr>
<tr><td><code>extrinsic</code></td>
<td>
<p>A numeric matrix of esimated extrinsic noise. 4 rows and n.iter columns.</p>
</td></tr>
<tr><td><code>intrinsic.mean</code></td>
<td>
<p>A numeric vector of length 7 that contains the mean estimates of intrinsic noise.</p>
</td></tr>
<tr><td><code>intrinsic.sd</code></td>
<td>
<p>A numeric vector of length 7 that contains the standard deviation of the estimates of intrinsic noise.</p>
</td></tr>
<tr><td><code>extrinsic.mean</code></td>
<td>
<p>A numeric vector of length 7 that contains the mean estimates of extrinsic noise.</p>
</td></tr>
<tr><td><code>extrinsic.sd</code></td>
<td>
<p>A numeric vector of length 7 that contains the standard deviation of the estimates of extrinsic noise.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Audrey Q. Fu
</p>


<h3>References</h3>

<p>Fu, A. Q. and Pachter, L. (2016). Estimating intrinsic and extrinsic noise from single-cell gene expression measurements.  arXiv:1601.03334.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+computeIntrinsicNoise">computeIntrinsicNoise</a></code>, <code><a href="#topic+computeExtrinsicNoise">computeExtrinsicNoise</a></code>, <code><a href="#topic+elowitz_data">elowitz_data</a></code>, <code><a href="#topic+yang_nl10">yang_nl10</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(yang_nl10)

# quantile normalization on log2 transformed data
# install bioconductor package for quantile normalization
# source('http://bioconductor.org/biocLite.R')
# biocLite('preprocessCore')
library(preprocessCore)

# ignore a few values that are negative
yang_nl10.pos &lt;- yang_nl10[-which (yang_nl10[,1]&lt;0),]
yang_nl10.pos.log2.quant &lt;- normalize.quantiles (as.matrix (log2 (yang_nl10.pos[,c(1,3)])))

# subset the data and compute noise estimates
yang.50 &lt;- computeNoiseForSubset (yang_nl10.pos.log2.quant, sample.size=50, n.iter=1000)
summary (yang.50)
</code></pre>

<hr>
<h2 id='elowitz_data'>
Expression of reporter genes in the D22 and M22 E. coli cells from Elowitz et al (2002).
</h2><span id='topic+elowitz_data'></span>

<h3>Description</h3>

<p>Expression of reporter genes CFP and YFP in over 200 E. coli cells of two strains: D22 and M22.  These values are displayed in a scatterplot in Elowitz et al (2002) Fig 3a.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("elowitz_data")</code></pre>


<h3>Format</h3>

<p>The format is:
List of 2
$ D22:'data.frame':	284 obs. of  2 variables:
..$ CFP: num [1:284] 3080 3082 2893 3053 2891 ...
..$ YFP: num [1:284] 2309 2394 2145 2340 2245 ...
$ M22:'data.frame':	250 obs. of  2 variables:
..$ CFP: num [1:250] 2438 2316 2521 2646 2830 ...
..$ YFP: num [1:250] 1409 1391 1511 1460 1638 ...
</p>


<h3>References</h3>

<p>Elowitz, M. B., A. J. Levine, E. D. Siggia, and P. S. Swain (2002) Stochastic gene expression in a single cell. Science, 297, 1183-1186.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(elowitz_data)

# Normalize data such that they are 
# comparable to Fig 3a in Elowitz et al. (2002).
# Normalized data have mean 1.
D22.cfp.norm &lt;- (elowitz_data$D22[,1]-mean (elowitz_data$D22[,1]))/sd(elowitz_data$D22[,1])/8+1
D22.yfp.norm &lt;- (elowitz_data$D22[,2]-mean (elowitz_data$D22[,2]))/sd(elowitz_data$D22[,2])/8+1

M22.cfp.norm &lt;- (elowitz_data$M22[,1]-mean (elowitz_data$M22[,1]))/sd(elowitz_data$M22[,1])/12+1
M22.yfp.norm &lt;- (elowitz_data$M22[,2]-mean (elowitz_data$M22[,2]))/sd(elowitz_data$M22[,2])/12+1

# Compute noise estimates.
# Since the mean is 1, estimates with and without 
# the scaling are the same.
unlist (computeIntrinsicNoise (D22.cfp.norm, D22.yfp.norm))
unlist (computeExtrinsicNoise (D22.cfp.norm, D22.yfp.norm))

unlist (computeIntrinsicNoise (M22.cfp.norm, M22.yfp.norm))
unlist (computeExtrinsicNoise (M22.cfp.norm, M22.yfp.norm))

</code></pre>

<hr>
<h2 id='simulateSC'>Simulate single-cell expression levels of two reporters
</h2><span id='topic+simulateSC'></span>

<h3>Description</h3>

<p>This function simulates expression levels of two reporters across 
single cells, mimicking the two-reporter assay.  The hierarchical model described in Fu 
and Pachter (2016) is used for simulation.  We further make the simplifying assumption 
that intrinsic noise is the same across cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateSC(n = 1000, intrinsic = 0.7, extrinsic = 0.8, mean = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulateSC_+3A_n">n</code></td>
<td>
<p>Number of single cells (sample size).
</p>
</td></tr>
<tr><td><code id="simulateSC_+3A_intrinsic">intrinsic</code></td>
<td>
<p>Scalar.  The (unscaled) intrinsic noise (or within-cell variability), denoted 
by <code class="reqn">\sigma^2</code> (equation 8) in Fu and Pachter (2016).
</p>
</td></tr>
<tr><td><code id="simulateSC_+3A_extrinsic">extrinsic</code></td>
<td>
<p>Scalar.  The (unscaled) extrinsic noise (or between-cell variability), denoted 
by <code class="reqn">\sigma^2_\mu</code> (equation 9) in Fu and Pachter (2016).
</p>
</td></tr>
<tr><td><code id="simulateSC_+3A_mean">mean</code></td>
<td>
<p>Scalar.  The overall mean of expression level, denoted by <code class="reqn">\mu</code> (equation 6) in Fu and Pachter (2016).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of two columns and <code class="reqn">n</code> rows.  Each column contains the expression levels of a reporter.  Each row is a single cell.
</p>


<h3>Author(s)</h3>

<p>Audrey Q. Fu
</p>


<h3>References</h3>

<p>Fu, A. Q. and Pachter, L. (2016). Estimating intrinsic and extrinsic noise from single-cell gene expression measurements.  arXiv:1601.03334.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+computeIntrinsicNoise">computeIntrinsicNoise</a></code>, <code><a href="#topic+computeExtrinsicNoise">computeExtrinsicNoise</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulation #1

# simulate 500 data sets
n.simu &lt;- 500

# true intrinsic and extrinsic noise
int.true &lt;- 0.7
ext.true &lt;- 0.8

# create matrices to hold estimated intrinsic and extrinsic noise
# using different estimators
int.simu.mtx &lt;- matrix (0, nrow=n.simu, ncol=8)
ext.simu.mtx &lt;- matrix (0, nrow=n.simu, ncol=4)

for (i in 1:n.simu) {
    n &lt;- 1000
    simu &lt;- simulateSC (n=n, intrinsic=int.true, extrinsic=ext.true, mean=1)
    
    int.simu.mtx[i,] &lt;- c(unlist (computeIntrinsicNoise (simu[,1], simu[,2])), 
        cor (simu[,1], simu[,2]))
    ext.simu.mtx[i,] &lt;- unlist (computeExtrinsicNoise (simu[,1], simu[,2]))
    
}

# add column names to simulation estimates
colnames (int.simu.mtx) &lt;- c("ELSS", "unbiasedGeneral", "unbiasedEqualMean", 
    "minMSEGeneral", "minMSEEqualMean", "asymptoticGeneral", 
    "asymptoticEqualMean", "cor")
colnames (ext.simu.mtx) &lt;- c("ELSS", "unbiased", "minMSE", "asymptotic")


# simulation #2

# simulate 500 data sets
n.simu &lt;- 500

# true intrinsic and extrinsic noise
int.true &lt;- 0.7
ext.true &lt;- 0.8

# use true correlation for the min-MSE estimates of extrinsic noise
true.cor &lt;- ext.true / (ext.true + int.true)

# create matrices to hold estimated intrinsic and extrinsic noise
# using different estimators
int.simu.mtx &lt;- matrix (0, nrow=n.simu, ncol=8)
ext.simu.mtx &lt;- matrix (0, nrow=n.simu, ncol=4)
ext.simu.mtx.2 &lt;- matrix (0, nrow=n.simu, ncol=4)

for (i in 1:n.simu) {
    n &lt;- 50
    simu &lt;- simulateSC (n=n, intrinsic=int.true, extrinsic=ext.true, mean=1)
    
    int.simu.mtx[i,] &lt;- c(unlist (computeIntrinsicNoise (simu[,1], simu[,2])), 
      cor (simu[,1], simu[,2]))
    ext.simu.mtx[i,] &lt;- unlist (computeExtrinsicNoise (simu[,1], simu[,2]))
    ext.simu.mtx.2[i,] &lt;- c(unlist (computeExtrinsicNoiseKnownCor (simu[,1], 
      simu[,2], true.cor)))
}

# add column names to simulation estimates
colnames (int.simu.mtx) &lt;- c("ELSS", "unbiasedGeneral", "unbiasedEqualMean", 
    "minMSEGeneral", "minMSEEqualMean", "asymptoticGeneral", 
    "asymptoticEqualMean", "cor")
colnames (ext.simu.mtx) &lt;- c("ELSS", "unbiased", "minMSE", "asymptotic")
colnames (ext.simu.mtx.2) &lt;- c("ELSS", "unbiased", "minMSE", "asymptotic")

# compute the MSE of estimates
computeMSE &lt;- function (a, t) {return (mean((a-t)^2))}
apply (int.simu.mtx[,1:7], 2, computeMSE, t=int.true)
apply (ext.simu.mtx, 2, computeMSE, t=ext.true)
apply (ext.simu.mtx.2, 2, computeMSE, t=ext.true)

</code></pre>

<hr>
<h2 id='yang_nl10'>
Expression of reporter genes in the NL010 E. coli cells from Yang et al (2014).
</h2><span id='topic+yang_nl10'></span>

<h3>Description</h3>

<p>Expression of reporter genes CFP and mCherry in over 40,000 E. coli cells.  A subset of these values are displayed in a scatterplot in Yang et al (2014) Fig 3a rightmost panel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("yang_nl10")</code></pre>


<h3>Format</h3>

<p>A data frame with 40683 observations on the following 3 variables.
</p>

<dl>
<dt><code>CFP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Venus</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>mCherry</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>References</h3>

<p>Yang, S., S. Kim, Y. R. Lim, C. Kim, H. J. An, J.-H. Kim, J. Sung, and N. K. Lee (2014) Contribution of RNA polymerase concentration variation to protein expression noise. Nature Communications, 5, 4761.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(yang_nl10)

# compute the noise estimates

# no normalization
# unscaled by mean
unlist (computeIntrinsicNoise (yang_nl10[,1], yang_nl10[,3]))
unlist (computeExtrinsicNoise (yang_nl10[,1], yang_nl10[,3]))

# scaled by mean
unlist (computeIntrinsicNoise (yang_nl10[,1], yang_nl10[,3])) / 
    mean (yang_nl10[,1]) / mean(yang_nl10[,3])
unlist (computeExtrinsicNoise (yang_nl10[,1], yang_nl10[,3])) / 
    mean (yang_nl10[,1]) / mean(yang_nl10[,3])

# quantile normalization on log2 transformed data
# install bioconductor package for quantile normalization
# source('http://bioconductor.org/biocLite.R')
# biocLite('preprocessCore')
library(preprocessCore)

# ignore a few values that are negative
yang_nl10.pos &lt;- yang_nl10[-which (yang_nl10[,1]&lt;0),]
yang_nl10.pos.log2.quant &lt;- normalize.quantiles (as.matrix (log2 (yang_nl10.pos[,c(1,3)])))

# unscaled by mean
unlist (computeIntrinsicNoise (yang_nl10.pos.log2.quant[,1], yang_nl10.pos.log2.quant[,2]))
unlist (computeExtrinsicNoise (yang_nl10.pos.log2.quant[,1], yang_nl10.pos.log2.quant[,2]))

# scaled by mean
unlist (computeIntrinsicNoise (yang_nl10.pos.log2.quant[,1], yang_nl10.pos.log2.quant[,2])) / 
    mean (yang_nl10.pos.log2.quant[,1]) / mean(yang_nl10.pos.log2.quant[,2])
unlist (computeExtrinsicNoise (yang_nl10.pos.log2.quant[,1], yang_nl10.pos.log2.quant[,2])) / 
    mean (yang_nl10.pos.log2.quant[,1]) / mean(yang_nl10.pos.log2.quant[,2])

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
