<!DOCTYPE html><html><head><title>Help for package Nonpareil</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Nonpareil}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+,Nonpareil.Set,ANY-method'><p>Alias of <code>Nonpareil.add.curve</code>.</p></a></li>
<li><a href='#$,Nonpareil.Curve-method'><p>Get attribute.</p></a></li>
<li><a href='#$,Nonpareil.Set-method'><p>Get attribute.</p></a></li>
<li><a href='#$&lt;-,Nonpareil.Curve-method'><p>Set attribute.</p></a></li>
<li><a href='#$&lt;-,Nonpareil.Set-method'><p>Set attribute.</p></a></li>
<li><a href='#Nonpareil'><p>Nonpareil: Metagenome Coverage Estimation and Projections for 'Nonpareil'.</p></a></li>
<li><a href='#Nonpareil.add.curve'><p>Adds a <code>Nonpareil.Curve</code> to a <code>Nonpareil.Set</code>.</p></a></li>
<li><a href='#Nonpareil.antif'><p>Complement function of <code>Nonpareil.f</code>.</p></a></li>
<li><a href='#Nonpareil.col'><p>Returns the color of the curve.</p></a></li>
<li><a href='#Nonpareil.coverage_factor'><p>Factor to transform redundancy into coverage (internal function).</p></a></li>
<li><a href='#Nonpareil.curve'><p>Generates a Nonpareil curve from an .npo file</p></a></li>
<li><a href='#Nonpareil.Curve-class'><p>A single Nonpareil curve. This object can be produced by</p>
<code>Nonpareil.curve</code> and supports S4 methods <code>plot</code>, <code>summary</code>,
<code>print</code>, and <code>predict</code>. For additional details, see help for
<code>summary.Nonpareil.Curve</code>.</a></li>
<li><a href='#Nonpareil.curve.batch'><p>Alias of <code>Nonpareil.set</code>.</p></a></li>
<li><a href='#Nonpareil.f'><p>Function of the projected model.</p></a></li>
<li><a href='#Nonpareil.fit_model'><p>Fit the sigmoidal model to the rarefied coverage.</p></a></li>
<li><a href='#Nonpareil.legend'><p>Generates a legend for Nonpareil plots.</p></a></li>
<li><a href='#Nonpareil.read_data'><p>Read the data tables and extract direct estimates.</p></a></li>
<li><a href='#Nonpareil.read_metadata'><p>Read the metadata headers.</p></a></li>
<li><a href='#Nonpareil.set'><p>Generates a collection of Nonpareil curves (a <code>Nonpareil.Set</code> object)</p>
and (optionally) plots all of them in a single canvas.</a></li>
<li><a href='#Nonpareil.Set-class'><p>Collection of <code>Nonpareil.Curve</code> objects. This object can be produced by</p>
<code>Nonpareil.curve.batch</code> and supports S4 methods <code>plot</code>,
<code>summary</code>, and <code>print</code>.</a></li>
<li><a href='#plot.Nonpareil.Curve'><p>Plot a <code>Nonpareil.Curve</code> object.</p></a></li>
<li><a href='#plot.Nonpareil.Set'><p>Plot a <code>Nonpareil.Set</code> object.</p></a></li>
<li><a href='#predict.Nonpareil.Curve'><p>Predict the coverage for a given sequencing effort.</p></a></li>
<li><a href='#print.Nonpareil.Curve'><p>Prints and returns invisibly a summary of the <code>Nonpareil.Curve</code> results.</p></a></li>
<li><a href='#print.Nonpareil.Set'><p>Prints and returns invisibly a summary of the <code>Nonpareil.Set</code> results.</p></a></li>
<li><a href='#summary.Nonpareil.Curve'><p>Returns a summary of the <code>Nonpareil.Curve</code> results.</p></a></li>
<li><a href='#summary.Nonpareil.Set'><p>Returns a summary of the <code>Nonpareil.Set</code> results.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Metagenome Coverage Estimation and Projections for 'Nonpareil'</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>3.4.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Luis M. Rodriguez-R [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Luis M. Rodriguez-R &lt;lmrodriguezr@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Plot, process, and analyze NPO files produced by
    'Nonpareil' <a href="http://enve-omics.ce.gatech.edu/nonpareil/">http://enve-omics.ce.gatech.edu/nonpareil/</a>.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://enve-omics.ce.gatech.edu/nonpareil/">http://enve-omics.ce.gatech.edu/nonpareil/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.9)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-01-28 17:45:57 UTC; miguel</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-01-28 18:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+2B+2CNonpareil.Set+2CANY-method'>Alias of <code>Nonpareil.add.curve</code>.</h2><span id='topic++2B+2CNonpareil.Set+2CANY-method'></span>

<h3>Description</h3>

<p>Alias of <code>Nonpareil.add.curve</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Nonpareil.Set,ANY'
e1 + e2
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B2B+2B2CNonpareil.Set+2B2CANY-method_+3A_e1">e1</code></td>
<td>
<p><code>Nonpareil.Set</code> object (<code>nps</code>).</p>
</td></tr>
<tr><td><code id="+2B2B+2B2CNonpareil.Set+2B2CANY-method_+3A_e2">e2</code></td>
<td>
<p><code>Nonpareil.Curve</code> object (<code>np</code>).</p>
</td></tr>
</table>

<hr>
<h2 id='+24+2CNonpareil.Curve-method'>Get attribute.</h2><span id='topic++24+2CNonpareil.Curve-method'></span>

<h3>Description</h3>

<p>Get attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Nonpareil.Curve'
x$name
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B24+2B2CNonpareil.Curve-method_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Curve</code> object.</p>
</td></tr>
<tr><td><code id="+2B24+2B2CNonpareil.Curve-method_+3A_name">name</code></td>
<td>
<p>Attribute.</p>
</td></tr>
</table>

<hr>
<h2 id='+24+2CNonpareil.Set-method'>Get attribute.</h2><span id='topic++24+2CNonpareil.Set-method'></span>

<h3>Description</h3>

<p>Get attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Nonpareil.Set'
x$name
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B24+2B2CNonpareil.Set-method_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Set</code> object.</p>
</td></tr>
<tr><td><code id="+2B24+2B2CNonpareil.Set-method_+3A_name">name</code></td>
<td>
<p>Attribute.</p>
</td></tr>
</table>

<hr>
<h2 id='+24+26lt+3B-+2CNonpareil.Curve-method'>Set attribute.</h2><span id='topic++24+3C-+2CNonpareil.Curve-method'></span>

<h3>Description</h3>

<p>Set attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'Nonpareil.Curve'
x$name &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B24+2B26lt+2B3B-+2B2CNonpareil.Curve-method_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Curve</code> object.</p>
</td></tr>
<tr><td><code id="+2B24+2B26lt+2B3B-+2B2CNonpareil.Curve-method_+3A_name">name</code></td>
<td>
<p>Attribute.</p>
</td></tr>
<tr><td><code id="+2B24+2B26lt+2B3B-+2B2CNonpareil.Curve-method_+3A_value">value</code></td>
<td>
<p>New value.</p>
</td></tr>
</table>

<hr>
<h2 id='+24+26lt+3B-+2CNonpareil.Set-method'>Set attribute.</h2><span id='topic++24+3C-+2CNonpareil.Set-method'></span>

<h3>Description</h3>

<p>Set attribute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'Nonpareil.Set'
x$name &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B24+2B26lt+2B3B-+2B2CNonpareil.Set-method_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Set</code> object.</p>
</td></tr>
<tr><td><code id="+2B24+2B26lt+2B3B-+2B2CNonpareil.Set-method_+3A_name">name</code></td>
<td>
<p>Attribute.</p>
</td></tr>
<tr><td><code id="+2B24+2B26lt+2B3B-+2B2CNonpareil.Set-method_+3A_value">value</code></td>
<td>
<p>New value.</p>
</td></tr>
</table>

<hr>
<h2 id='Nonpareil'>Nonpareil: Metagenome Coverage Estimation and Projections for 'Nonpareil'.</h2><span id='topic+Nonpareil'></span><span id='topic+Nonpareil-package'></span>

<h3>Description</h3>

<p>Plot, process, and analyze NPO files produced by 'Nonpareil'
<a href="http://enve-omics.ce.gatech.edu/nonpareil/">http://enve-omics.ce.gatech.edu/nonpareil/</a>.
</p>


<h3>Citation</h3>

<p>If you use Nonpareil, please cite:
Rodriguez-R et al. 2018. Nonpareil 3: Fast estimation of metagenomic coverage
and sequence diversity.
mSystems 3(3): e00039-18. DOI: 10.1128/mSystems.00039-18.
</p>
<p>Rodriguez-R &amp; Konstantinidis. 2014. Nonpareil: a redundancy-based approach to
assess the level of coverage in metagenomic datasets.
Bioinformatics 30 (5): 629-635. DOI: 10.1093/bioinformatics/btt584.
</p>
<p>For an extended discussion on coverage in metagenomic data, see also:
</p>
<p>Rodriguez-R &amp; Konstantinidis. 2014. Estimating coverage in metagenomic data
sets and why it matters.
The ISME Journal 8: 2349â€“2351. DOI: 10.1038/ismej.2014.76.
</p>

<hr>
<h2 id='Nonpareil.add.curve'>Adds a <code>Nonpareil.Curve</code> to a <code>Nonpareil.Set</code>.</h2><span id='topic+Nonpareil.add.curve'></span>

<h3>Description</h3>

<p>Adds a <code>Nonpareil.Curve</code> to a <code>Nonpareil.Set</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.add.curve(nps, np)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.add.curve_+3A_nps">nps</code></td>
<td>
<p><code>Nonpareil.Set</code> object.</p>
</td></tr>
<tr><td><code id="Nonpareil.add.curve_+3A_np">np</code></td>
<td>
<p><code>Nonpareil.Curve</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the <code>Nonpareil.Set</code> including the newly added
<code>Nonpareil.Curve</code>.
</p>

<hr>
<h2 id='Nonpareil.antif'>Complement function of <code>Nonpareil.f</code>.</h2><span id='topic+Nonpareil.antif'></span>

<h3>Description</h3>

<p>Complement function of <code>Nonpareil.f</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.antif(y, a, b)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.antif_+3A_y">y</code></td>
<td>
<p>Values of abundance-weighted average coverage.</p>
</td></tr>
<tr><td><code id="Nonpareil.antif_+3A_a">a</code></td>
<td>
<p>Parameter alpha of the gamma CDF.</p>
</td></tr>
<tr><td><code id="Nonpareil.antif_+3A_b">b</code></td>
<td>
<p>Parameter beta of the gamma CDF.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated sequencing effort.
</p>

<hr>
<h2 id='Nonpareil.col'>Returns the color of the curve.</h2><span id='topic+Nonpareil.col'></span>

<h3>Description</h3>

<p>Returns the color of the curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.col(x, alpha = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.col_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Curve</code> or <code>Nonpareil.Set</code> object.</p>
</td></tr>
<tr><td><code id="Nonpareil.col_+3A_alpha">alpha</code></td>
<td>
<p>Alpha level of the color from 0 to 1.</p>
</td></tr>
</table>

<hr>
<h2 id='Nonpareil.coverage_factor'>Factor to transform redundancy into coverage (internal function).</h2><span id='topic+Nonpareil.coverage_factor'></span>

<h3>Description</h3>

<p>Factor to transform redundancy into coverage (internal function).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.coverage_factor(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.coverage_factor_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Curve</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric scalar.
</p>

<hr>
<h2 id='Nonpareil.curve'>Generates a Nonpareil curve from an .npo file</h2><span id='topic+Nonpareil.curve'></span>

<h3>Description</h3>

<p>Generates a Nonpareil curve from an .npo file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.curve(file, plot = TRUE, label = NA, col = NA,
  enforce.consistency = TRUE, star = 95, correction.factor = TRUE,
  weights.exp = NA, skip.model = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.curve_+3A_file">file</code></td>
<td>
<p>Path to the .npo file, containing the read redundancy.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve_+3A_plot">plot</code></td>
<td>
<p>Determines if the plot should be produced. If FALSE, it computes the
coverage and the model wihtout plotting.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve_+3A_label">label</code></td>
<td>
<p>Name of the dataset. If NA, it is determined by the file name.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve_+3A_col">col</code></td>
<td>
<p>Color of the curve.
If NA, a random color is assigned (even if <code>plot = FALSE</code>).</p>
</td></tr>
<tr><td><code id="Nonpareil.curve_+3A_enforce.consistency">enforce.consistency</code></td>
<td>
<p>If TRUE, it fails verbosely on insufficient data, otherwise it warns
about the inconsistencies and attempts the estimations.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve_+3A_star">star</code></td>
<td>
<p>Objective coverage in percentage; i.e., coverage value considered
near-complete.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve_+3A_correction.factor">correction.factor</code></td>
<td>
<p>Should the overlap-dependent (or kmer-length-dependent) correction
factor be applied? If FALSE, redundancy is assumed to equal coverage.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve_+3A_weights.exp">weights.exp</code></td>
<td>
<p>Vector of values to be tested (in order) as exponent of the weights
distribution. If the model fails to converge, sometimes manual
modifications in this parameter may help. By default (NA), five
different values are tested in the following order: For linear
sampling, -1.1, -1.2, -0.9, -1.3, -1. For logarithmic sampling (-d
option in Nonpareil), 0, 1, -1, 1.3, -1.1, 1.5, -1.5.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve_+3A_skip.model">skip.model</code></td>
<td>
<p>If set, skips the model estimation altogether.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve_+3A_...">...</code></td>
<td>
<p>Any additional parameters passed to <code>plot.Nonpareil.Curve</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns invisibly a <code>Nonpareil.Curve</code> object
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate a Nonpareil plot
file &lt;- system.file("extdata", "LakeLanier.npo", package="Nonpareil")
np &lt;- Nonpareil.curve(file)

# Show the estimated values
print(np)

# Predict coverage for 20Gbp
predict(np, 20e9)

# Obtain the Nd diversity index
np$diversity
</code></pre>

<hr>
<h2 id='Nonpareil.Curve-class'>A single Nonpareil curve. This object can be produced by
<code>Nonpareil.curve</code> and supports S4 methods <code>plot</code>, <code>summary</code>,
<code>print</code>, and <code>predict</code>. For additional details, see help for
<code>summary.Nonpareil.Curve</code>.</h2><span id='topic+Nonpareil.Curve-class'></span>

<h3>Description</h3>

<p>A single Nonpareil curve. This object can be produced by
<code>Nonpareil.curve</code> and supports S4 methods <code>plot</code>, <code>summary</code>,
<code>print</code>, and <code>predict</code>. For additional details, see help for
<code>summary.Nonpareil.Curve</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>file</code></dt><dd><p>Input .npo file.</p>
</dd>
<dt><code>label</code></dt><dd><p>Name of the dataset.</p>
</dd>
<dt><code>col</code></dt><dd><p>Color of the dataset.</p>
</dd>
<dt><code>L</code></dt><dd><p>Read length.</p>
</dd>
<dt><code>AL</code></dt><dd><p>Adjusted read length (same as L for alignment).</p>
</dd>
<dt><code>R</code></dt><dd><p>Number of reads.</p>
</dd>
<dt><code>LR</code></dt><dd><p>Effective sequencing effort used.</p>
</dd>
<dt><code>overlap</code></dt><dd><p>Minimum read overlap.</p>
</dd>
<dt><code>ksize</code></dt><dd><p>K-mer size (for kmer kernel only).</p>
</dd>
<dt><code>log.sample</code></dt><dd><p>Multiplier of the log-sampling (or zero if linear).</p>
</dd>
<dt><code>kernel</code></dt><dd><p>Read-comparison kernel.</p>
</dd>
<dt><code>version</code></dt><dd><p>Nonpareil version used.</p>
</dd>
<dt><code>x.obs</code></dt><dd><p>Rarefied sequencing effort.</p>
</dd>
<dt><code>x.adj</code></dt><dd><p>Adjusted rarefied sequencing effort.</p>
</dd>
<dt><code>y.red</code></dt><dd><p>Rarefied redundancy (observed).</p>
</dd>
<dt><code>y.cov</code></dt><dd><p>Rarefied coverage (corrected).</p>
</dd>
<dt><code>y.sd</code></dt><dd><p>Standard deviation of rarefied coverage.</p>
</dd>
<dt><code>y.p25</code></dt><dd><p>Percentile 25 (1st quartile) of rarefied coverage.</p>
</dd>
<dt><code>y.p50</code></dt><dd><p>Percentile 50 (median) of rarefied coverage.</p>
</dd>
<dt><code>y.p75</code></dt><dd><p>Percentile 75 (3rd quartile) of rarefied coverage.</p>
</dd>
<dt><code>kappa</code></dt><dd><p>Dataset redundancy.</p>
</dd>
<dt><code>C</code></dt><dd><p>Dataset coverage.</p>
</dd>
<dt><code>consistent</code></dt><dd><p>Is the data sufficient for accurate estimation?</p>
</dd>
<dt><code>star</code></dt><dd><p>Coverage considered 'nearly complete'.</p>
</dd>
<dt><code>has.model</code></dt><dd><p>Was the model successfully estimated?</p>
</dd>
<dt><code>warning</code></dt><dd><p>Warnings generated on consistency or model fitting.</p>
</dd>
<dt><code>LRstar</code></dt><dd><p>Projected seq. effort for nearly complete coverage.</p>
</dd>
<dt><code>modelR</code></dt><dd><p>Pearson's R for the estimated model.</p>
</dd>
<dt><code>diversity</code></dt><dd><p>Dataset Nd index of sequence diversity.</p>
</dd>
<dt><code>model</code></dt><dd><p>Fitted sigmoidal model.</p>
</dd>
<dt><code>call</code></dt><dd><p>Call producing this object.</p>
</dd>
</dl>

<hr>
<h2 id='Nonpareil.curve.batch'>Alias of <code>Nonpareil.set</code>.</h2><span id='topic+Nonpareil.curve.batch'></span>

<h3>Description</h3>

<p>Alias of <code>Nonpareil.set</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.curve.batch(files, col = NA, labels = NA, plot = TRUE,
  plot.opts = list(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.curve.batch_+3A_files">files</code></td>
<td>
<p>Vector with the paths to the .npo files.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve.batch_+3A_col">col</code></td>
<td>
<p>Color of the curves (vector). If not passed, values are randomly
assigned. Values are recycled.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve.batch_+3A_labels">labels</code></td>
<td>
<p>Labels of the curves (vector). If not passed, values are determined by
the filename. Values are recycled.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve.batch_+3A_plot">plot</code></td>
<td>
<p>If TRUE, it generates the Nonpareil curve plots.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve.batch_+3A_plot.opts">plot.opts</code></td>
<td>
<p>Any parameters accepted by <code>plot.Nonpareil.Set</code> as a list.</p>
</td></tr>
<tr><td><code id="Nonpareil.curve.batch_+3A_...">...</code></td>
<td>
<p>Any additional parameters accepted by <code>Nonpareil.curve</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='Nonpareil.f'>Function of the projected model.</h2><span id='topic+Nonpareil.f'></span>

<h3>Description</h3>

<p>Function of the projected model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.f(x, a, b)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.f_+3A_x">x</code></td>
<td>
<p>Values of sequencing effort (in bp).</p>
</td></tr>
<tr><td><code id="Nonpareil.f_+3A_a">a</code></td>
<td>
<p>Parameter alpha of the Gamma CDF.</p>
</td></tr>
<tr><td><code id="Nonpareil.f_+3A_b">b</code></td>
<td>
<p>Parameter beta of the Gamma CDF.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Predicted values of abundance-weighted average coverage.
</p>

<hr>
<h2 id='Nonpareil.fit_model'>Fit the sigmoidal model to the rarefied coverage.</h2><span id='topic+Nonpareil.fit_model'></span>

<h3>Description</h3>

<p>Fit the sigmoidal model to the rarefied coverage.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.fit_model(np, weights.exp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.fit_model_+3A_np">np</code></td>
<td>
<p><code>Nonpareil.Curve</code> object.</p>
</td></tr>
<tr><td><code id="Nonpareil.fit_model_+3A_weights.exp">weights.exp</code></td>
<td>
<p>Numeric; see <code>Nonpareil.curve</code> for details.</p>
</td></tr>
</table>

<hr>
<h2 id='Nonpareil.legend'>Generates a legend for Nonpareil plots.</h2><span id='topic+Nonpareil.legend'></span>

<h3>Description</h3>

<p>Generates a legend for Nonpareil plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.legend(np, x, y = 0.3, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.legend_+3A_np">np</code></td>
<td>
<p>A <code>Nonpareil.Set</code> or <code>Nonpareil.Curve</code> object, or a list of
<code>Nonpareil.Curve</code> objects.</p>
</td></tr>
<tr><td><code id="Nonpareil.legend_+3A_x">x</code></td>
<td>
<p>X coordinate, or any character string accepted by legend (e.g.,
'bottomright').</p>
</td></tr>
<tr><td><code id="Nonpareil.legend_+3A_y">y</code></td>
<td>
<p>Y coordinate.</p>
</td></tr>
<tr><td><code id="Nonpareil.legend_+3A_...">...</code></td>
<td>
<p>Any other parameters supported by legend().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns invisibly a list, same as <code>legend</code>.
</p>

<hr>
<h2 id='Nonpareil.read_data'>Read the data tables and extract direct estimates.</h2><span id='topic+Nonpareil.read_data'></span>

<h3>Description</h3>

<p>Read the data tables and extract direct estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.read_data(x, correction.factor)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.read_data_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Curve</code> object.</p>
</td></tr>
<tr><td><code id="Nonpareil.read_data_+3A_correction.factor">correction.factor</code></td>
<td>
<p>Logical; see <code>Nonpareil.curve</code> for details.</p>
</td></tr>
</table>

<hr>
<h2 id='Nonpareil.read_metadata'>Read the metadata headers.</h2><span id='topic+Nonpareil.read_metadata'></span>

<h3>Description</h3>

<p>Read the metadata headers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.read_metadata(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.read_metadata_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Curve</code> object.</p>
</td></tr>
</table>

<hr>
<h2 id='Nonpareil.set'>Generates a collection of Nonpareil curves (a <code>Nonpareil.Set</code> object)
and (optionally) plots all of them in a single canvas.</h2><span id='topic+Nonpareil.set'></span>

<h3>Description</h3>

<p>Generates a collection of Nonpareil curves (a <code>Nonpareil.Set</code> object)
and (optionally) plots all of them in a single canvas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Nonpareil.set(files, col = NA, labels = NA, plot = TRUE,
  plot.opts = list(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Nonpareil.set_+3A_files">files</code></td>
<td>
<p>Vector with the paths to the .npo files.</p>
</td></tr>
<tr><td><code id="Nonpareil.set_+3A_col">col</code></td>
<td>
<p>Color of the curves (vector). If not passed, values are randomly
assigned. Values are recycled.</p>
</td></tr>
<tr><td><code id="Nonpareil.set_+3A_labels">labels</code></td>
<td>
<p>Labels of the curves (vector). If not passed, values are determined by
the filename. Values are recycled.</p>
</td></tr>
<tr><td><code id="Nonpareil.set_+3A_plot">plot</code></td>
<td>
<p>If TRUE, it generates the Nonpareil curve plots.</p>
</td></tr>
<tr><td><code id="Nonpareil.set_+3A_plot.opts">plot.opts</code></td>
<td>
<p>Any parameters accepted by <code>plot.Nonpareil.Set</code> as a list.</p>
</td></tr>
<tr><td><code id="Nonpareil.set_+3A_...">...</code></td>
<td>
<p>Any additional parameters accepted by <code>Nonpareil.curve</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns invisibly a <code>Nonpareil.Set</code> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate a Nonpareil plot with multiple curves
files &lt;- system.file("extdata",
      c("HumanGut.npo","LakeLanier.npo","IowaSoil.npo"),
      package="Nonpareil")
col &lt;- c("orange","darkcyan","firebrick4")
nps &lt;- Nonpareil.set(files, col=col,
      plot.opts=list(plot.observed=FALSE, model.lwd=2))

# Show the estimated values
print(nps)

# Show current coverage (as %)
summary(nps)[,"C"]*100

# Extract Nd diversity index
summary(nps)[,"diversity"]

# Extract sequencing effort for nearly complete coverage (in Gbp)
summary(nps)[,"LRstar"]/1e9

# Predict coverage for a sequencing effort of 10Gbp
sapply(nps$np.curves, predict, 10e9)
</code></pre>

<hr>
<h2 id='Nonpareil.Set-class'>Collection of <code>Nonpareil.Curve</code> objects. This object can be produced by
<code>Nonpareil.curve.batch</code> and supports S4 methods <code>plot</code>,
<code>summary</code>, and <code>print</code>.</h2><span id='topic+Nonpareil.Set-class'></span>

<h3>Description</h3>

<p>Collection of <code>Nonpareil.Curve</code> objects. This object can be produced by
<code>Nonpareil.curve.batch</code> and supports S4 methods <code>plot</code>,
<code>summary</code>, and <code>print</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>np.curves</code></dt><dd><p>List of <code>Nonpareil.Curve</code> objects.</p>
</dd>
<dt><code>call</code></dt><dd><p>Call producing this object.</p>
</dd>
</dl>

<hr>
<h2 id='plot.Nonpareil.Curve'>Plot a <code>Nonpareil.Curve</code> object.</h2><span id='topic+plot.Nonpareil.Curve'></span>

<h3>Description</h3>

<p>Plot a <code>Nonpareil.Curve</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Nonpareil.Curve'
plot(x, col = NA, add = FALSE, new = !add,
  plot.observed = TRUE, plot.model = TRUE, plot.dispersion = FALSE,
  plot.diversity = TRUE, xlim = c(1000, 1e+13), ylim = c(1e-06, 1),
  main = paste("Nonpareil Curve for", x$label),
  xlab = "Sequencing effort (bp)", ylab = "Estimated Average Coverage",
  curve.lwd = 2, curve.alpha = 0.4, model.lwd = 1, model.alpha = 1,
  log = "x", arrow.length = 0.05, arrow.head = arrow.length, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Nonpareil.Curve_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Curve</code> object to plot.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_col">col</code></td>
<td>
<p>Color of the curve. If passed, it overrides the colors set in the
<code>Nonpareil.Curve</code> object.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_add">add</code></td>
<td>
<p>If TRUE, it attempts to use a previous (active) canvas to plot the
curve.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_new">new</code></td>
<td>
<p>Inverse of 'add'.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_plot.observed">plot.observed</code></td>
<td>
<p>Indicates if the observed (rarefied) coverage is to be plotted.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_plot.model">plot.model</code></td>
<td>
<p>Indicates if the fitted model is to be plotted.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_plot.dispersion">plot.dispersion</code></td>
<td>
<p>Indicates if (and how) dispersion of the replicates should be
plotted. Supported values are:
</p>

<ul>
<li> <p><code>FALSE</code>: no dispersion is plotted (default),
</p>
</li>
<li> <p><code>'sd'</code>: one standard deviation around the mean,
</p>
</li>
<li> <p><code>'ci95'</code>: 95
</p>
</li>
<li> <p><code>'ci90'</code>: 90
</p>
</li>
<li> <p><code>'ci50'</code>: 50
</p>
</li>
<li> <p><code>'iq'</code>: Inter-quartile range.
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_plot.diversity">plot.diversity</code></td>
<td>
<p>If TRUE, the diversity estimate is plotted as a small arrow below the
Nonpareil curve.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_xlim">xlim</code></td>
<td>
<p>Limits of the sequencing effort (X-axis).</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_ylim">ylim</code></td>
<td>
<p>Limits of the coverage (Y-axis).</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_main">main</code></td>
<td>
<p>Title of the plot.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_xlab">xlab</code></td>
<td>
<p>Label of the X-axis.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_ylab">ylab</code></td>
<td>
<p>Label of the Y-axis.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_curve.lwd">curve.lwd</code></td>
<td>
<p>Line width of the rarefied coverage.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_curve.alpha">curve.alpha</code></td>
<td>
<p>Alpha value (from 0 to 1) of the rarefied coverage.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_model.lwd">model.lwd</code></td>
<td>
<p>Line width of the model.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_model.alpha">model.alpha</code></td>
<td>
<p>Alpha value (from 0 to 1) of the model.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_log">log</code></td>
<td>
<p>Axis to plot in logarithmic scale. Supported values are:
</p>

<ul>
<li> <p><code>'x'</code>: sequencing effort (default),
</p>
</li>
<li> <p><code>'y'</code>: coverage,
</p>
</li>
<li> <p><code>'xy'</code>: both logarithmic, or
</p>
</li>
<li> <p><code>''</code>: both linear.
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_arrow.length">arrow.length</code></td>
<td>
<p>If <code>plot.diversity = TRUE</code>, it determines the length of the
arrow to display the divsersity (as a fraction of the ylim range).</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_arrow.head">arrow.head</code></td>
<td>
<p>If <code>plot.diversity = TRUE</code>, it determines the length of the
arrow head to display the diversity index (in inches).</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Curve_+3A_...">...</code></td>
<td>
<p>Additional graphical parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Retuns invisibly a <code>Nonpareil.Curve</code> object (same as <code>x</code> input).
For additional details see help for <code>summary.Nonpareil.Curve</code>.
</p>

<hr>
<h2 id='plot.Nonpareil.Set'>Plot a <code>Nonpareil.Set</code> object.</h2><span id='topic+plot.Nonpareil.Set'></span>

<h3>Description</h3>

<p>Plot a <code>Nonpareil.Set</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Nonpareil.Set'
plot(x, col = NA, labels = NA,
  main = "Nonpareil Curves", legend.opts = list(), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.Nonpareil.Set_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Set</code> object to plot.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Set_+3A_col">col</code></td>
<td>
<p>Color of the curves (vector).
If passed, it overrides the colors set in the <code>Nonpareil.Curve</code>
objects. Values are recycled.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Set_+3A_labels">labels</code></td>
<td>
<p>Labels of the curves (vector). If passed, it overrides the labels set
in the <code>Nonpareil.Curve</code> objects. Values are recycled.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Set_+3A_main">main</code></td>
<td>
<p>Title of the plot.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Set_+3A_legend.opts">legend.opts</code></td>
<td>
<p>Any additional parameters passed to <code>Nonpareil.legend</code>.
If FALSE, the legend is not displayed.</p>
</td></tr>
<tr><td><code id="plot.Nonpareil.Set_+3A_...">...</code></td>
<td>
<p>Any additional parameters passed to <code>plot.Nonpareil.Curve</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns invisibly a <code>Nonpareil.Set</code> object (same as <code>x</code> input).
</p>

<hr>
<h2 id='predict.Nonpareil.Curve'>Predict the coverage for a given sequencing effort.</h2><span id='topic+predict.Nonpareil.Curve'></span>

<h3>Description</h3>

<p>Predict the coverage for a given sequencing effort.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Nonpareil.Curve'
predict(object, lr = object$LR, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.Nonpareil.Curve_+3A_object">object</code></td>
<td>
<p><code>Nonpareil.Curve</code> object.</p>
</td></tr>
<tr><td><code id="predict.Nonpareil.Curve_+3A_lr">lr</code></td>
<td>
<p>Sequencing effort for the prediction (in bp).</p>
</td></tr>
<tr><td><code id="predict.Nonpareil.Curve_+3A_...">...</code></td>
<td>
<p>Additional parameters ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the expected coverage at the given sequencing effort.
</p>

<hr>
<h2 id='print.Nonpareil.Curve'>Prints and returns invisibly a summary of the <code>Nonpareil.Curve</code> results.</h2><span id='topic+print.Nonpareil.Curve'></span>

<h3>Description</h3>

<p>Prints and returns invisibly a summary of the <code>Nonpareil.Curve</code> results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Nonpareil.Curve'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.Nonpareil.Curve_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Set</code> object.</p>
</td></tr>
<tr><td><code id="print.Nonpareil.Curve_+3A_...">...</code></td>
<td>
<p>Additional parameters ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the summary invisibly. See help for
<code>summary.Nonpareil.Curve</code> for additional information.
</p>

<hr>
<h2 id='print.Nonpareil.Set'>Prints and returns invisibly a summary of the <code>Nonpareil.Set</code> results.</h2><span id='topic+print.Nonpareil.Set'></span>

<h3>Description</h3>

<p>Prints and returns invisibly a summary of the <code>Nonpareil.Set</code> results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Nonpareil.Set'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.Nonpareil.Set_+3A_x">x</code></td>
<td>
<p><code>Nonpareil.Set</code> object.</p>
</td></tr>
<tr><td><code id="print.Nonpareil.Set_+3A_...">...</code></td>
<td>
<p>Additional parameters ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the summary invisibly. See help for
<code>summary.Nonpareil.Curve</code> and <code>summary.Nonpareil.Set</code> for
additional information.
</p>

<hr>
<h2 id='summary.Nonpareil.Curve'>Returns a summary of the <code>Nonpareil.Curve</code> results.</h2><span id='topic+summary.Nonpareil.Curve'></span>

<h3>Description</h3>

<p>Returns a summary of the <code>Nonpareil.Curve</code> results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Nonpareil.Curve'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.Nonpareil.Curve_+3A_object">object</code></td>
<td>
<p><code>Nonpareil.Curve</code> object.</p>
</td></tr>
<tr><td><code id="summary.Nonpareil.Curve_+3A_...">...</code></td>
<td>
<p>Additional parameters ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with the following values for the dataset:
</p>

<ul>
<li><p> kappa: &quot;Redundancy&quot; value of the entire dataset.
</p>
</li>
<li><p> C: Average coverage of the entire dataset.
</p>
</li>
<li><p> LRstar: Estimated sequencing effort required to reach the objective
average coverage (star, 95
</p>
</li>
<li><p> LR: Actual sequencing effort of the dataset.
</p>
</li>
<li><p> modelR: Pearson's R coefficient betweeen the rarefied data and the
projected model.
</p>
</li>
<li><p> diversity: Nonpareil sequence-diversity index (Nd). This value's
units are the natural logarithm of the units of sequencing effort
(log-bp), and indicates the inflection point of the fitted model for
the Nonpareil curve. If the fit doesn't converge, or the model is not
estimated, the value is zero (0).
</p>
</li></ul>


<hr>
<h2 id='summary.Nonpareil.Set'>Returns a summary of the <code>Nonpareil.Set</code> results.</h2><span id='topic+summary.Nonpareil.Set'></span>

<h3>Description</h3>

<p>Returns a summary of the <code>Nonpareil.Set</code> results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'Nonpareil.Set'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.Nonpareil.Set_+3A_object">object</code></td>
<td>
<p><code>Nonpareil.Set</code> object.</p>
</td></tr>
<tr><td><code id="summary.Nonpareil.Set_+3A_...">...</code></td>
<td>
<p>Additional parameters ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix with different values for each dataset. For additional
details on the values returned, see help for
<code>summary.Nonpareil.Curve</code>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
