<!DOCTYPE html><html lang="en"><head><title>Help for package nplplot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {nplplot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bedplot'><p>Creation of BED and BedGraph custom tracks</p></a></li>
<li><a href='#genomeplot'><p>Creation of Genome Graph files</p></a></li>
<li><a href='#grid-internal'><p>Internal nplplot.multi variables</p></a></li>
<li><a href='#lods1'><p>LOD score table for chromosome 1</p></a></li>
<li><a href='#lods2'><p>LOD score table for chromosome 2</p></a></li>
<li><a href='#nplplot'><p>Plotting statistics along a chromosome</p></a></li>
<li><a href='#nplplot.multi'><p>Plotting linkage or association statistics for multiple results files</p></a></li>
<li><a href='#nplplot.old'><p>LOD score plotting (old version of nplplot)</p></a></li>
<li><a href='#prepareplot'><p>Prepare input data files for bedplot and genomeplot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>4.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-12-20</td>
</tr>
<tr>
<td>Title:</td>
<td>Plotting Linkage and Association Results</td>
</tr>
<tr>
<td>Author:</td>
<td>Robert V Baron &lt;rvb5@pitt.edu&gt;, Nandita Mukhopadhyay,
        Xinyu Tang, Daniel E. Weeks &lt;weeks@pitt.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel E. Weeks &lt;weeks@pitt.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides routines for plotting
      linkage and association results along a chromosome,
      with marker names displayed along the top border.
      There are also routines for generating BED and BedGraph
      custom tracks for viewing in the UCSC genome browser.
      The data reformatting program Mega2 uses this
      package to plot output from a variety of
      programs.  </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://watson.hgen.pitt.edu/register/">https://watson.hgen.pitt.edu/register/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-20 23:03:33 UTC; dweeks</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-20 23:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='bedplot'>Creation of BED and BedGraph custom tracks</h2><span id='topic+bedplot'></span>

<h3>Description</h3>

<p>Generates matched sets of files for linkage or association
statistics along a chromosome for viewing in the UCSC genome browser
from an input file containing a table of marker names, physical
positions and one or more statistical scores.</p>


<h3>Usage</h3>

<pre><code class='language-R'> bedplot(bed.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bedplot_+3A_bed.data">bed.data</code></td>
<td>
<p>File containing a table of marker names, physical position and
scores. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>bed.data example:
</p>
<pre>
Marker   Position   TRAIT_ALL   
M1       144255     0.670
-        144305     0.640
M3       144355     0.590
-        144378     0.600
M2       144400     0.610
</pre>
<p>Bedplot creates two types of files: a BED.* file containing a custom
BED annotation track and a BedGraph.* file custom BedGraph annotation
track. These files have the same suffix as the input bed.data file. When
there are multiple scores in <code>bed.data</code> file, a matched pair of
BedGraph track BED track files is created for each score, labelled with
the score names, as well as the chromosome numbers,
e.g. <code>BedGraph.score1.*</code> and <code>BED.score1.*</code>,
<code>BedGraph.score2.*</code> and <code>BED.score2.*</code>, etc.  
</p>


<h3>Value</h3>

<p>TRUE or FALSE depending on whether runs successfully. </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: bedplot("bed.data.05")
</code></pre>

<hr>
<h2 id='genomeplot'>Creation of Genome Graph files</h2><span id='topic+genomeplot'></span>

<h3>Description</h3>

<p>The <code>genomeplot</code> function generates two formatted files, one
containing &ldquo;chromosome base&rdquo; formatted genome data and the other
containing marker-specific results with dbSNP SNP IDs for displaying  
genome-wide data sets in the UCSC genome browser. </p>


<h3>Usage</h3>

<pre><code class='language-R'>genomeplot(gg.data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genomeplot_+3A_gg.data">gg.data</code></td>
<td>
<p>a file containing chromosome, marker, physical position and scores. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>gg.data example:
</p>
<pre>
Chromosome   Marker   Position   TRAIT_ALL   
         5   M1       0.000      0.670
         5   -        2.500      0.640
         5   M3       5.000      0.590
         5   -        6.500      0.600
         5   M2       8.000      0.610                   
         8   M4       0.000      0.670
         8   -        2.500      0.640
         8   M6       5.000      0.590
         8   -        6.500      0.600
         8   M5       8.000      0.610 
</pre>
<p>Two files are created, &ldquo;GG.positons.all&rdquo; for the
&ldquo;chromosome base&rdquo; format, and &ldquo;GG.markers.all&rdquo; for the
marker-names based format. When there are multiple scores in
<code>gg.data</code> file, this results in matched pairs of files, one for
each score, labelled with the score names, e.g. GG.positions.score1.all,
and GG.markers.score1.all, GG.positions.score2.all and
GG.markers.score2.all, and so on.
</p>


<h3>Value</h3>

<p>TRUE or FALSE depending on whether runs successfully. </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: genomeplot("GG.data.all")
</code></pre>

<hr>
<h2 id='grid-internal'>Internal nplplot.multi variables</h2><span id='topic+yline'></span><span id='topic+ymin'></span><span id='topic+ymax'></span><span id='topic+yfix'></span><span id='topic+title'></span><span id='topic+ylabel'></span><span id='topic+draw.lgnd'></span><span id='topic+cex.legend'></span><span id='topic+cex.axis'></span><span id='topic+tcl'></span><span id='topic+bw'></span><span id='topic+na.rm'></span><span id='topic+lgndtxt'></span><span id='topic+ptypes'></span><span id='topic+my.colors'></span><span id='topic+ltypes'></span>

<h3>Description</h3>

<p>Internal nplplot.multi
</p>


<h3>Details</h3>

<p>These are not to be called by the user. These are added to the
namespace when nplplot.multi() reads the lods1header.R and
lods2header.R files.
</p>

<hr>
<h2 id='lods1'>LOD score table for chromosome 1</h2><span id='topic+lods1'></span>

<h3>Description</h3>

<p>This is a data frame with the first two columns containing marker names
and positions, followed by three columns of LOD scores.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lods1)</code></pre>


<h3>Format</h3>

<p>There are 100 markers in the table.
</p>

<hr>
<h2 id='lods2'>LOD score table for chromosome 2</h2><span id='topic+lods2'></span>

<h3>Description</h3>

<p>This is a data frame with the first two columns containing marker names
and positions, followed by three columns of LOD scores.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lods2)</code></pre>


<h3>Format</h3>

<p>There are 87 markers in the table. 
</p>

<hr>
<h2 id='nplplot'>Plotting statistics along a chromosome</h2><span id='topic+nplplot'></span>

<h3>Description</h3>

<p>Plots linkage or association statistics along a chromosome,
contained within a data frame or a file. Marker names are displayed
along the top border.</p>


<h3>Usage</h3>

<pre><code class='language-R'>nplplot(plotdata=NULL, filename=NULL, yline=2.0, ymin=0, ymax=3.0,
         header=TRUE, yfix=FALSE, title=NULL, draw.lgnd=TRUE,
         xlabl="", ylabl="", lgndx=NULL, lgndy=NULL, lgndtxt=NULL,
         cex.legend = 0.7, cex.axis=0.7, tcl=1,
         bw=TRUE, my.colors=NULL, ltypes=NULL, ptypes=NULL,
         na.rm=TRUE, plot.width=0.0, ...) </code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nplplot_+3A_plotdata">plotdata</code></td>
<td>
<p> A data frame containing marker names in the
first column, marker map positions in the second column, and
statistical scores in column 3 onwards. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_filename">filename</code></td>
<td>
<p> A table format file containing the plot
data as described above. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_header">header</code></td>
<td>
<p> TRUE or FALSE depending on whether the plotdata
or file has a header line. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_yline">yline</code></td>
<td>
<p> Y-value for displaying a horizontal cut-off
line. If 'yfix' is set to TRUE and Y-line falls outside of <code>[ymin,
  ymax]</code>, then the cut-off line is omitted. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_ymin">ymin</code>, <code id="nplplot_+3A_ymax">ymax</code></td>
<td>
<p> Y-axis minimum and maximum values. If non-NULL
values are provided, and <code>yfix</code> is set to TRUE, then the plot
area will be cropped to these values. If <code>yfix</code> is set to
FALSE, then <code>ymin</code> and <code>ymax</code> values are ignored. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_yfix">yfix</code></td>
<td>
<p> TRUE or FALSE to denote whether plot area should be
cropped to the <code>ymin, ymax</code> values. This has no effect if
<code>ymin, ymax</code> values are NULL. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_title">title</code></td>
<td>
<p> Used as the subtitle of the plot.</p>
</td></tr>
<tr><td><code id="nplplot_+3A_xlabl">xlabl</code></td>
<td>
<p> X-axis label. May interfere with the display of
the subtitle provided as the <code>title</code> argument. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_ylabl">ylabl</code></td>
<td>
<p> Y-axis label.</p>
</td></tr>
<tr><td><code id="nplplot_+3A_draw.lgnd">draw.lgnd</code></td>
<td>
<p> TRUE or FALSE denoting whether a plot legend
should be displayed. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_lgndx">lgndx</code></td>
<td>
<p> X coordinate for the legend box, passed to the
<code><a href="graphics.html#topic+legend">legend</a></code> command. Ignored if <code>draw.legend</code>
is set to FALSE. If set to NULL with <code>draw.legend</code> set to TRUE,
the X-coordinate is automatically calculated. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_lgndy">lgndy</code></td>
<td>
<p> Y coordinate for the legend box, passed to the
<code><a href="graphics.html#topic+legend">legend</a></code> command. Ignored if <code>draw.legend</code>
is set to FALSE. If set to NULL with <code>draw.legend</code> set to TRUE,
the Y-coordinate is automatically calculated. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_lgndtxt">lgndtxt</code></td>
<td>
<p> Vector of strings to use in the legend. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_cex.legend">cex.legend</code></td>
<td>
<p> Character scaling for legend, passed as the
<code>cex</code> argument to the  <code><a href="graphics.html#topic+legend">legend</a></code> command. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_cex.axis">cex.axis</code></td>
<td>
<p> Character scaling for the axis, passed to the
<code><a href="graphics.html#topic+axis">axis</a></code> command for drawing the top border. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_tcl">tcl</code></td>
<td>
<p> Length of ticks for the top border, passed to the
<code><a href="graphics.html#topic+axis">axis</a></code> command. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_bw">bw</code></td>
<td>
<p> TRUE or FALSE depending on whether plots should be
drawn in color. If set to FALSE, then the colors defined by
<code>my.colors</code> are used. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_my.colors">my.colors</code></td>
<td>
<p> Vector of color specifications as described
in the <code><a href="graphics.html#topic+par">par</a></code> command. Ignored if <code>bw</code> above
is set to FALSE. If <code>bw</code> is to TRUE and my.colors is set to
NULL, the <code><a href="grDevices.html#topic+rainbow">rainbow</a></code> palette will be used. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_ltypes">ltypes</code></td>
<td>
<p> Vector of line types for the plots. Each
non-zero line type is passed on to a <code>lines</code> command. Use 0 or
'none' if a line is to be skipped. If NULL, no lines will be
drawn. For line types see the <code><a href="graphics.html#topic+par">par</a></code> command. If set to
&quot;default&quot;, line-types 1 through the number of plots is used. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_ptypes">ptypes</code></td>
<td>
<p> Vector of characters giving the point
types, to be passed onto the <code>points</code> command. Use 'none' if no
points are to be drawn for a score column. If NULL, no points will
be displayed. If both the line-type and point-type specification for
a results column is set to 'none', that column will not be plotted. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_na.rm">na.rm</code></td>
<td>
<p> TRUE or FALSE depending on whether points with
Y-coordinates set to NAs should be skipped. Setting <code>na.rm</code> to
TRUE eliminates discontinuities in the plots. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_plot.width">plot.width</code></td>
<td>
<p> A number giving the width of the plot in
inches. This is used to decide whether some marker names should be
dynamically hidden, if they are too close to each other along the top
border. If set to 0, the default page-size is used to set the width. </p>
</td></tr>
<tr><td><code id="nplplot_+3A_...">...</code></td>
<td>
<p> Further graphical parameters to be passed onto the
'plot', 'lines' and 'points' commands. </p>
</td></tr>
</table>


<h3>Details</h3>

<p> The nplplot function draws multiple curves within a single
plot by automatically calling 'plot', 'lines', and 'points' multiple
times, thus making it easy for the user to plot many columns of results
using a single plot command. It is intended for the display of linkage
and association analysis results such as LOD scores and P-values. It
allows the marker names to be displayed along the top border of the
plot, as well as a significance threshold line.
</p>
<p>The input plot data has to be in a specific tabular format with
each column separated by white-space :
</p>
<p>Here is an example:
</p>
<pre>
Marker   Position        score1  score2 score3      
d1s228   0.00            0.546   0.345  0.142                    
d1s429   1.00            0.346   0.335  0.252       
d1s347   2.00            0.446   0.245  0.342                    
</pre>
<p>This example file contains a header, therefore the <code>header</code> argument should
be set to TRUE.
</p>
<p>Lines 2-4 contain scores at various marker positions. <b>Missing
scores can be denoted with either &quot;.&quot; or &quot;NA&quot;</b>. The position column
cannot have missing data. There can be any number of score columns
within a file and will be plotted as separate curves within the same
plot.  Each file is plotted as a separate plot.
</p>


<h3>Value</h3>

<p>TRUE or FALSE depending on whether the plot data was successfully
plotted.</p>


<h3>See Also</h3>

<p><code><a href="#topic+nplplot.multi">nplplot.multi</a>, <a href="#topic+nplplot.old">nplplot.old</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'># plot with legend
par(omi=c(0.05, 0.05, 0.5, 0.05))
data(lods1, package="nplplot")
nplplot(plotdata=lods1, draw.lgnd=TRUE)

# plot without legend
data(lods2, package="nplplot")
nplplot(plotdata=lods2, draw.lgnd=FALSE)

# plotting from a data file
datadir &lt;- paste(system.file("data", package="nplplot"), .Platform$file.sep, sep="")
nplplot(filename=paste(datadir, "lods2.txt.gz", sep=""))

</code></pre>

<hr>
<h2 id='nplplot.multi'>Plotting linkage or association statistics for multiple results files</h2><span id='topic+nplplot.multi'></span>

<h3>Description</h3>

<p>Wrapper function for the 'nplplot' function. Creates mutiple plots
from a list of plot files, with custom graphical parameters set by
header files. </p>


<h3>Usage</h3>

<pre><code class='language-R'>nplplot.multi(filenames, plotdata = NULL, col=2, row=2, mode="l",
                output="screen", headerfiles=NULL, lgnd="page",
                customtracks=FALSE, mega2mapfile=NULL,
                pagewidth=NULL, pageheight=NULL, topmargin=0.25,
                ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nplplot.multi_+3A_filenames">filenames</code></td>
<td>
<p> Vector of strings giving file names
containing tables of linkage analysis results. See
<code><a href="#topic+nplplot">nplplot</a></code> for a description of the file format. </p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_plotdata">plotdata</code></td>
<td>
<p> List of dataframes by chromosome
containing tables of linkage analysis results. See
<code><a href="#topic+nplplot">nplplot</a></code> for a description of the format. </p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_col">col</code></td>
<td>
<p> Integer indicating number of columns of plots to be drawn
on a page.</p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_row">row</code></td>
<td>
<p> Integer signifying number of rows of plots will be drawn
on a page.</p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_mode">mode</code></td>
<td>
<p> 'p' or 'l' to denote 'portrait' or 'landscape' mode.</p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_output">output</code></td>
<td>
<p> String giving file name to save plots in. If
set to 'screen', plots will be displayed and not saved. The file
format is determined by the filename extension: '.pdf' for PDF, or
'.ps' for postscript. If no extension is provided, or is not
recognized, a PDF file will be produced with '.pdf' appended to the
file name.</p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_headerfiles">headerfiles</code></td>
<td>
<p> Files containing R language commands to set
various plot parameters, which are passed onto the
<code><a href="#topic+nplplot">nplplot</a></code> command. The recommended use is to have one
headerfile per plot file. For a list of parameters, consult the
<code><a href="#topic+nplplot">nplplot</a></code> documentation. If the number of headerfiles is
fewer than plot files, the last header file will be reused as many
times as needed. If more headerfiles are provided than necessary, the
last ones will be ignored.</p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_lgnd">lgnd</code></td>
<td>
<p> TRUE, FALSE, 'page' or a list consisting of plot
numbers. If a single value is given, TRUE causes legends to be drawn
inside every plot, FALSE omits legends altogether, and 'page' causes
a legend to be drawn inside the first plot on every page. If a list
of numbers is provided, only plots corresponding to these numbers
will have legends. </p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_customtracks">customtracks</code></td>
<td>
<p> TRUE or FALSE. If set to TRUE, data files are
created to draw custom tracks within the UCSC genome browser in BED
format, as well as a combined data file to add a genome-wide track
over all chromosomes present in the data. If set to TRUE, a
<code>mega2mapfile</code> also needs to be supplied (see below). </p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_mega2mapfile">mega2mapfile</code></td>
<td>
<p> Mega2 annotated format map file containing
physical positions for all the markers present in the
nplplot input data files. Rather than a file name, the name of a 
<em>data.frame</em> containing what would have been read from the file,
may be given. </p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_pagewidth">pagewidth</code></td>
<td>
<p> A number denoting width of the plot page in
inches. If set to NULL, a width of 7.0 is used for the plot
area. Assumes that a margin of 0.5 will be available around the plot
area for axis annotations. </p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_pageheight">pageheight</code></td>
<td>
<p> A number denoting height of the plot page in
inches. If set to NULL, a height of 10.0 is used for the plot area. 
Assumes that a margin of 0.5 will be available around the plot
area for axis annotations. </p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_topmargin">topmargin</code></td>
<td>
<p> A number denoting the width of the outside top
margin of each plot. Since this contains marker names, it may need to
be increased to accommodate long names.</p>
</td></tr>
<tr><td><code id="nplplot.multi_+3A_...">...</code></td>
<td>
<p> Further graphical parameters to be passed onto the
'plot', 'lines' and 'points' commands within <code>nplplot</code>. </p>
</td></tr>
</table>


<h3>Details</h3>

<p> This function is designed for use within the Mega2 software to
generate graphical output for some of the target analysis options,
namely Merlin, SimWalk2 and Allegro. It calls nplplot repeatedly to
create plots corresponding to each input file. The input arguments
control characteristics of all plots together, whereas the header
files allow customization within each plot. Thus, it is expected that
there should be as many header files as there are plot data files.
</p>
<p>This function can also be used to create custom tracks within the UCSC
genome browser, as well as a genome-wide plot. To use this feature, make
sure that the names of the nplplot input data files each have a
&quot;Mega2-style&quot; chromosome extension (01 through 09 for chromosomes 1
thorugh 9, otherwise the chromosome number, or X for the human
X-chromosome, 23).
</p>
<p>To make this function more useful to other <strong>R</strong> programs, you may
directly supply a <em>data.frame</em> for the <code>mapfile</code> argument
and a list of <em>data.frames</em> for the <code>plotdata</code> argument
and NULL for the <code>filename</code> argument.  (The name of each list
element is the corresponding chromosome.)
</p>


<h3>Value</h3>

<p>TRUE or FALSE depending on whether all plot commands were successful.</p>


<h3>See Also</h3>

<p><code><a href="#topic+nplplot">nplplot</a>, <a href="#topic+nplplot.old">nplplot.old</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>datadir &lt;- paste(system.file("data", package="nplplot"),
                 .Platform$file.sep, sep="")
f1 &lt;- paste(datadir, "lods1.txt.gz", sep="")
f2 &lt;- paste(datadir, "lods2.txt.gz", sep="")
h1 &lt;- system.file("extdata","lods1header.R",package="nplplot")
h2 &lt;- system.file("extdata","lods2header.R",package="nplplot")
nplplot.multi(c(f1, f2), col=1, row=2, output="screen", 
	     headerfiles=c(h1, h2), topmargin=0.5)
</code></pre>

<hr>
<h2 id='nplplot.old'>LOD score plotting (old version of nplplot)</h2><span id='topic+nplplot.old'></span>

<h3>Description</h3>

<p>Plots score curves contained within one or more
specified results files.</p>


<h3>Usage</h3>

<pre><code class='language-R'>nplplot.old(files, col=2, row=2, mode="p", output="screen", 
          yline=2.0, ymin=NULL, ymax=NULL, yfix=FALSE, batch=FALSE, 
          headerfiles=NULL, titles=NULL, xlabl="", ylabl="", 
          lgnd="page", lgndx=NULL, lgndy=NULL, bw=TRUE, na.rm=TRUE)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nplplot.old_+3A_files">files</code></td>
<td>
<p> List of files names (strings). Each file produces a
separate plot.</p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_col">col</code></td>
<td>
<p> For multiple plots on a single page of pdf or postscript
output, this item defines the number of columns of plots, and should
be an integer greater than or equal to 1. Default is set to 2.</p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_row">row</code></td>
<td>
<p> For multiple plots on a page of pdf or postscript output,
this defines the number of rows of plots, (value should be 1 or
greater). Default value is set to 2. </p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_mode">mode</code></td>
<td>
<p> Orientation for pdf or postscript output,
&quot;p&quot; for portrait
&quot;l&quot; for landscape. </p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_output">output</code></td>
<td>
<p> File name for saving plots; &quot;screen&quot;, the default
causes the plots to  be displayed on the screen. To produce a pdf
file use the extension .pdf. To produce a postscript file, use the
.ps file name extension. If no extension is given a pdf file is
produced. </p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_yline">yline</code></td>
<td>
<p> Y-value for displaying a horizontal cut-off line. </p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_ymin">ymin</code>, <code id="nplplot.old_+3A_ymax">ymax</code></td>
<td>
<p> Y-axis minimum and maximum values with
default values NULL. If non-NULL values are provided, and
<code>yfix</code> is set to TRUE, then the plot area will be cropped to
these values.</p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_yfix">yfix</code></td>
<td>
<p> Set to TRUE or FALSE depending on whether ymin and ymax
should be enforced across all plots irrespective of whether the plot
data lie within these bounds. Ignored if ymin or ymax are set to
NULL.</p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_batch">batch</code></td>
<td>
<p> TRUE or FALSE, to determine whether the display
screen should be closed. If nplplot is called within R, this should 
be set to FALSE.</p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_headerfiles">headerfiles</code></td>
<td>
<p> List of file names, one for each data file
specified above. Each header-file contains a string with column
names corresponding to the columns in the data file. These column
names are used in the plot legend. 
If set to NULL (the default), nplplot uses the first item in 
each column of a data file as plot legend.
If a headerfile is provided, then nplplot will attempt to read in
the first line of the datafile as data, so the user should be
careful not to put in a headerline as well as a headerfile.</p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_titles">titles</code></td>
<td>
<p> Array of strings denoting titles for each plot.
If there are not enough titles, the last string is recycled for the
remaining plots. Default is an empty string. </p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_xlabl">xlabl</code></td>
<td>
<p> Array of strings, to use as the x-axis label on 
each plot.</p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_ylabl">ylabl</code></td>
<td>
<p> Array of strings to use as the y-axis label on 
each plot. </p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_lgnd">lgnd</code></td>
<td>
<p> TRUE, FALSE, &quot;page&quot; or a list of plot numbers denoting whether
the legend should be drawn in all plots, none, first plot on a page,
or specific plot numbers. Default &quot;page&quot;. </p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_lgndx">lgndx</code></td>
<td>
<p> NULL or a real value if a specific x-coordinate should
be used to position the legend. Default NULL. </p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_lgndy">lgndy</code></td>
<td>
<p> NULL or a real value if a specific y-coordinate should
be used to position the legend. Default NULL. </p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_bw">bw</code></td>
<td>
<p> TRUE or FALSE depending on whether plots should be
drawn in color. A list of six colors are defined within nplplot,
which are successively used to draw each curve, and reused as
necessary. The order in which these colors are used is:
magenta, lightblue, grey, navyblue, lightcyan and pink.
The 7th color, reserved for black and white plots is black. </p>
</td></tr>
<tr><td><code id="nplplot.old_+3A_na.rm">na.rm</code></td>
<td>
<p> TRUE or FALSE depending on whether NAs should be removed
prior to plotting the data. Including NAs will produce broken plots,
when lines are drawn. This may be desirable in some cases, if
missing data needs to be reported. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Usually these results would be 
LOD scores, p-values, or log10(p-values). This is targetted
towards p-values or LOD scores obtained at various marker positions
from statistical analysis of genetic data.
A results file has to be in a specific tabular format with
each column separated by white-space :
</p>
<p>A) First line = header line                                      
</p>
<p>B) Next set of lines = any number of data lines
</p>
<p>C) Final two lines = line type &amp; point type definition.
</p>
<p>Here is an example:
</p>
<pre>
marker   location        score1  score2 score3      
d1s228   0.00            0.546   0.345  0.142                    
d1s429   1.00            0.346   0.335  0.252       
d1s347   2.00            0.446   0.245  0.342                    
ltype    -99.99              1       2      3       
ptype    -99.99             15      16     17                  
</pre>
<p>In this example, line 1 column headers for the score columns
may be used as labels within the legend, as described in the 
usage of the &quot;headerfile&quot; argument. <b>The first two headers 
are ignored.</b>
</p>
<p>Lines 2-4 contain scores at various marker positions. <b>Missing
scores can be denoted with either &quot;.&quot; or &quot;NA&quot;</b>. The position column
cannot have missing data. There can be any number of score columns
within a file and will be plotted as separate curves within the same
plot.  Each file is plotted as a separate plot.
</p>
<p>The last two lines give line types and point types for each curve. A
zero line or point type will not plot lines or points for that score
column respectively. For allowable ptype values, consult the R
documentation for &quot;points&quot;. For line types, consult the documentation
on &quot;par&quot;.
</p>
<p>The names in the first column are used as axis labels on the top of
the plot border. Setting a name in the <b>marker</b> column to &quot;-&quot;
will result in no label at that position.
</p>


<h3>Value</h3>

<p>TRUE or FALSE depending on whether the input files were read 
in successfully.</p>


<h3>See Also</h3>

<p><code><a href="#topic+nplplot">nplplot</a>, <a href="#topic+nplplot.multi">nplplot.multi</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: nplplot.old("lod.1", output="lod.1.ps", batch=T, headerfiles="hdr.1")
## Not run: nplplot.old(c("lod.1", "lod.2"), col=1, row=2, headerfiles=c("hdr.1","hdr.2"))
</code></pre>

<hr>
<h2 id='prepareplot'>Prepare input data files for bedplot and genomeplot</h2><span id='topic+prepareplot'></span>

<h3>Description</h3>

<p>The prepareplot function prepares input data files for
<a href="#topic+bedplot">bedplot</a> and <a href="#topic+genomeplot">genomeplot</a> functions from nplplot-formatted
score files and a Mega2 annotated format map file with physical positions. </p>


<h3>Usage</h3>

<pre><code class='language-R'> prepareplot(prefix, chrlist=c(1:23,25), mapfile, output="both")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prepareplot_+3A_prefix">prefix</code></td>
<td>
<p>Prefix of the names of R table files, e.g. &ldquo;RMERLINDATA&rdquo; for R table files 
&ldquo;RMERLINDATA.01&rdquo;, &ldquo;RMERLINDATA.02&rdquo;, etc. Using
<code>chrlist</code> below, it automatically finds R table files with the
specified prefix and chromosome-specific extensions to convert.
</p>
<p>Alternatively, <code>prefix</code> may be a <em>list</em> of
<em>data.frames</em> named by the chromosomes supplied in <code>chrlist</code>.
</p>
</td></tr>
<tr><td><code id="prepareplot_+3A_chrlist">chrlist</code></td>
<td>
<p>List of chromosome numbers to create plots for, default 1 through
23. Chromosomes 23 and 25 produces files for the X chromosome X, 25
denoting pseudo-autosomal markers on chromosome X. </p>
</td></tr>
<tr><td><code id="prepareplot_+3A_mapfile">mapfile</code></td>
<td>
<p>Mega2 annnotated format map file, containing
marker names and and exactly one set of physical
positions. <code>mapfile</code> may instead be a <em>data.frame</em>
containing the same information as the map file, viz. the marker
names and physical positions.
</p>
</td></tr>
<tr><td><code id="prepareplot_+3A_output">output</code></td>
<td>
<p>Which plotting function to generate data for,
&ldquo;both&rdquo; for both bedplot and genomeplot functions,
&ldquo;bed&rdquo; for generating input files for bedplot function,
&ldquo;GG&rdquo; for generating input file for genomeplot
function. output is set to default &ldquo;both&rdquo;. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>mapfile example:
</p>
<pre>
Chromosome   Map.h.a   Name   Map.h.m   Map.h.f   Build52.p
5            0.0       M1     0.0       0.0       144255
5            5.0       M3     2.0       7.0       144355
5            8.0       M2     4.0       12.0      144400
8            0.0       M4     0.0       0.0       144255
8            5.0       M6     2.0       7.0       144355
8            8.0       M5     4.0       12.0      144400
</pre>
<p>The names of R table files should be linkage or association analysis
score files in nplplot-format with Mega2-style file names, i.e., having
a common specified prefix and 01-09, 11- 24, X, or XY as suffixes. The
list of suffixes are determined by the chromosome list. If this list
includes 23 or X, R table files with either the &ldquo;23&rdquo;
suffix or &ldquo;X&rdquo; suffix are accepted. If both files exist, the one
with the &ldquo;X&rdquo; suffix is read in and the user warned. If the XY
chromosome is chosen, R table files can have either &ldquo;24&rdquo; or
&ldquo;XY&rdquo; as a suffix, with &ldquo;XY&rdquo; suffixed file having
precedence.
</p>
<p>The prepareplot function generates chromosome-specific formatted score
files &ldquo;bed.data.#&rdquo; for use by <a href="#topic+bedplot">bedplot</a> with the same
suffix as the R table file. If X chromosome is chosen, the output file
is named &ldquo;bed.data.23&rdquo;. If XY chromosome is chosen, those records
on XY chromosome are included in &ldquo;bed.data.23&rdquo; file. The output
file &ldquo;bed.data.#&rdquo; contains marker names and physical
positions followed by one or more score columns. The header is taken
from the input score file(s).
</p>
<p>Prepareplot generates a combined file over all chromosomes
&ldquo;GG.data.all&rdquo; for <a href="#topic+genomeplot">genomeplot</a>. For pseudo-autosomal markers
denoted by chromosome XY or 24, these scores are assigned the X
chromosome. The output file &ldquo;GG.data.all&rdquo; contains four or more
columns with headings. The first, second and third columns contain
chromosomes, marker names and physical positions respectively, followed
by one or more score columns with score names as headers.
</p>


<h3>Value</h3>

<p>TRUE or FALSE depending on whether runs successfully. </p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: prepareplot("RMERLINDATA", c(5,8), "map.all", "GG")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
