<!DOCTYPE html><html><head><title>Help for package numbat</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {numbat}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#acen_hg19'><p>centromere regions (hg19)</p></a></li>
<li><a href='#acen_hg38'><p>centromere regions (hg38)</p></a></li>
<li><a href='#aggregate_counts'><p>Utility function to make reference gene expression profiles</p></a></li>
<li><a href='#analyze_bulk'><p>Call CNVs in a pseudobulk profile using the Numbat joint HMM</p></a></li>
<li><a href='#annot_consensus'><p>Annotate a consensus segments on a pseudobulk dataframe</p></a></li>
<li><a href='#annot_haplo_segs'><p>Annotate haplotype segments after HMM decoding</p></a></li>
<li><a href='#annot_ref'><p>example reference cell annotation</p></a></li>
<li><a href='#annot_segs'><p>Annotate copy number segments after HMM decoding</p></a></li>
<li><a href='#annot_theta_mle'><p>Annotate the theta parameter for each segment</p></a></li>
<li><a href='#annot_theta_roll'><p>Annotate rolling estimate of imbalance level theta</p></a></li>
<li><a href='#annotate_genes'><p>Annotate genes on allele dataframe</p></a></li>
<li><a href='#approx_phi_post'><p>Laplace approximation of the posterior of expression fold change phi</p></a></li>
<li><a href='#approx_theta_post'><p>Laplace approximation of the posterior of allelic imbalance theta</p></a></li>
<li><a href='#binary_entropy'><p>calculate entropy for a binary variable</p></a></li>
<li><a href='#bulk_example'><p>example pseudobulk dataframe</p></a></li>
<li><a href='#calc_allele_lik'><p>Calculate allele likelihoods</p></a></li>
<li><a href='#calc_allele_LLR'><p>Calculate LLR for an allele HMM</p></a></li>
<li><a href='#calc_cluster_dist'><p>Calculate expression distance matrix between cell populatoins</p></a></li>
<li><a href='#calc_exp_LLR'><p>Calculate LLR for an expression HMM</p></a></li>
<li><a href='#calc_phi_mle_lnpois'><p>Calculate the MLE of expression fold change phi</p></a></li>
<li><a href='#check_allele_df'><p>Check the format of a allele dataframe</p></a></li>
<li><a href='#check_contam'><p>check inter-individual contamination</p></a></li>
<li><a href='#check_exp_noise'><p>check noise level</p></a></li>
<li><a href='#check_exp_ref'><p>check the format of lambdas_ref</p></a></li>
<li><a href='#check_matrix'><p>Check the format of a count matrix</p></a></li>
<li><a href='#check_segs_fix'><p>check the format of a given consensus segment dataframe</p></a></li>
<li><a href='#check_segs_loh'><p>Check the format of a given clonal LOH segment dataframe</p></a></li>
<li><a href='#choose_ref_cor'><p>choose beest reference for each cell based on correlation</p></a></li>
<li><a href='#chrom_sizes_hg19'><p>chromosome sizes (hg19)</p></a></li>
<li><a href='#chrom_sizes_hg38'><p>chromosome sizes (hg38)</p></a></li>
<li><a href='#classify_alleles'><p>classify alleles using viterbi and forward-backward</p></a></li>
<li><a href='#cnv_heatmap'><p>Plot CNV heatmap</p></a></li>
<li><a href='#combine_bulk'><p>Combine allele and expression pseudobulks</p></a></li>
<li><a href='#compute_posterior'><p>Do bayesian averaging to get posteriors</p></a></li>
<li><a href='#contract_nodes'><p>Merge adjacent set of nodes</p></a></li>
<li><a href='#count_mat_example'><p>example gene expression count matrix</p></a></li>
<li><a href='#count_mat_ref'><p>example reference count matrix</p></a></li>
<li><a href='#detect_clonal_loh'><p>Call clonal LOH using SNP density.</p>
Rcommended for cell lines or tumor samples with no normal cells.</a></li>
<li><a href='#df_allele_example'><p>example allele count dataframe</p></a></li>
<li><a href='#exp_hclust'><p>Run smoothed expression-based hclust</p></a></li>
<li><a href='#expand_states'><p>expand multi-allelic CNVs into separate entries in the single-cell posterior dataframe</p></a></li>
<li><a href='#fill_neu_segs'><p>Fill neutral regions into consensus segments</p></a></li>
<li><a href='#filter_genes'><p>filter for mutually expressed genes</p></a></li>
<li><a href='#find_common_diploid'><p>Find the common diploid region in a group of pseudobulks</p></a></li>
<li><a href='#fit_bbinom'><p>fit a Beta-Binomial model by maximum likelihood</p></a></li>
<li><a href='#fit_gamma'><p>fit gamma maximum likelihood</p></a></li>
<li><a href='#fit_lnpois'><p>fit a PLN model by maximum likelihood</p></a></li>
<li><a href='#fit_ref_sse'><p>Fit a reference profile from multiple references using constrained least square</p></a></li>
<li><a href='#fit_snp_rate'><p>negative binomial model</p></a></li>
<li><a href='#gaps_hg19'><p>genome gap regions (hg19)</p></a></li>
<li><a href='#gaps_hg38'><p>genome gap regions (hg38)</p></a></li>
<li><a href='#generate_postfix'><p>Generate alphabetical postfixes</p></a></li>
<li><a href='#genotype'><p>Genotyping main function</p></a></li>
<li><a href='#get_allele_bulk'><p>Aggregate into pseudobulk alelle profile</p></a></li>
<li><a href='#get_allele_hmm'><p>Get an allele HMM</p></a></li>
<li><a href='#get_allele_post'><p>get CNV allele posteriors</p></a></li>
<li><a href='#get_bulk'><p>Aggregate single-cell data into combined bulk expression and allele profile</p></a></li>
<li><a href='#get_clone_post'><p>Map cells to the phylogeny (or genotypes) based on CNV posteriors</p></a></li>
<li><a href='#get_exp_bulk'><p>Aggregate into bulk expression profile</p></a></li>
<li><a href='#get_exp_likelihoods'><p>get the single cell expression likelihoods</p></a></li>
<li><a href='#get_exp_post'><p>compute single-cell expression posteriors</p></a></li>
<li><a href='#get_exp_sc'><p>get the single cell expression dataframe</p></a></li>
<li><a href='#get_gtree'><p>Get a tidygraph tree with simplified mutational history.</p></a></li>
<li><a href='#get_haplotype_post'><p>Get phased haplotypes</p></a></li>
<li><a href='#get_inter_cm'><p>Helper function to get inter-SNP distance</p></a></li>
<li><a href='#get_internal_nodes'><p>Helper function to get the internal nodes of a dendrogram and the leafs in each subtree</p></a></li>
<li><a href='#get_joint_post'><p>get joint posteriors</p></a></li>
<li><a href='#get_lambdas_bar'><p>Get average reference expressio profile based on single-cell ref choices</p></a></li>
<li><a href='#get_move_cost'><p>Get the cost of a mutation reassignment</p></a></li>
<li><a href='#get_move_opt'><p>Get the least costly mutation reassignment</p></a></li>
<li><a href='#get_nodes_celltree'><p>Get the internal nodes of a dendrogram and the leafs in each subtree</p></a></li>
<li><a href='#get_ordered_tips'><p>Get ordered tips from a tree</p></a></li>
<li><a href='#get_segs_consensus'><p>Extract consensus CNV segments</p></a></li>
<li><a href='#get_segs_neu'><p>get neutral segments from multiple pseudobulks</p></a></li>
<li><a href='#get_snps'><p>process VCFs into SNP dataframe</p></a></li>
<li><a href='#get_tree_post'><p>Find maximum lilkelihood assignment of mutations on a tree</p></a></li>
<li><a href='#gexp_roll_example'><p>example smoothed gene expression dataframe</p></a></li>
<li><a href='#gtf_hg19'><p>gene model (hg19)</p></a></li>
<li><a href='#gtf_hg38'><p>gene model (hg38)</p></a></li>
<li><a href='#gtf_mm10'><p>gene model (mm10)</p></a></li>
<li><a href='#hc_example'><p>example hclust tree</p></a></li>
<li><a href='#joint_post_example'><p>example joint single-cell cnv posterior dataframe</p></a></li>
<li><a href='#label_edges'><p>Annotate the direct upstream or downstream mutations on the edges</p></a></li>
<li><a href='#label_genotype'><p>Label the genotypes on a mutation graph</p></a></li>
<li><a href='#log_mem'><p>Log memory usage</p></a></li>
<li><a href='#log_message'><p>Log a message</p></a></li>
<li><a href='#make_group_bulks'><p>Make a group of pseudobulks</p></a></li>
<li><a href='#mark_tumor_lineage'><p>Mark the tumor lineage of a phylogeny</p></a></li>
<li><a href='#Modes'><p>Get the modes of a vector</p></a></li>
<li><a href='#mut_graph_example'><p>example mutation graph</p></a></li>
<li><a href='#Numbat'><p>Numbat R6 class</p></a></li>
<li><a href='#phi_hat_roll'><p>Rolling estimate of expression fold change phi</p></a></li>
<li><a href='#phi_hat_seg'><p>Estimate of expression fold change phi in a segment</p></a></li>
<li><a href='#phylogeny_example'><p>example single-cell phylogeny</p></a></li>
<li><a href='#plot_bulks'><p>Plot a group of pseudobulk HMM profiles</p></a></li>
<li><a href='#plot_consensus'><p>Plot consensus CNVs</p></a></li>
<li><a href='#plot_exp_roll'><p>Plot single-cell smoothed expression magnitude heatmap</p></a></li>
<li><a href='#plot_mut_history'><p>Plot mutational history</p></a></li>
<li><a href='#plot_phylo_heatmap'><p>Plot single-cell CNV calls along with the clonal phylogeny</p></a></li>
<li><a href='#plot_psbulk'><p>Plot a pseudobulk HMM profile</p></a></li>
<li><a href='#plot_sc_tree'><p>Plot single-cell smoothed expression magnitude heatmap</p></a></li>
<li><a href='#pnorm.range.log'><p>Get the total probability from a region of a normal pdf</p></a></li>
<li><a href='#pre_likelihood_hmm'><p>HMM object for unit tests</p></a></li>
<li><a href='#preprocess_allele'><p>Preprocess allele data</p></a></li>
<li><a href='#ref_hca'><p>reference expression magnitudes from HCA</p></a></li>
<li><a href='#ref_hca_counts'><p>reference expression counts from HCA</p></a></li>
<li><a href='#relevel_chrom'><p>Relevel chromosome column</p></a></li>
<li><a href='#resolve_cnvs'><p>Get unique CNVs from set of segments</p></a></li>
<li><a href='#retest_bulks'><p>retest consensus segments on pseudobulks</p></a></li>
<li><a href='#retest_cnv'><p>retest CNVs in a pseudobulk</p></a></li>
<li><a href='#return_missing_columns'><p>Check the format of a given file</p></a></li>
<li><a href='#run_group_hmms'><p>Run multiple HMMs</p></a></li>
<li><a href='#run_numbat'><p>Run workflow to decompose tumor subclones</p></a></li>
<li><a href='#segs_example'><p>example CNV segments dataframe</p></a></li>
<li><a href='#simes_p'><p>Calculate simes' p</p></a></li>
<li><a href='#simplify_history'><p>Simplify the mutational history based on likelihood evidence</p></a></li>
<li><a href='#smooth_expression'><p>filtering, normalization and capping</p></a></li>
<li><a href='#smooth_segs'><p>Smooth the segments after HMM decoding</p></a></li>
<li><a href='#switch_prob_cm'><p>predict phase switch probablity as a function of genetic distance</p></a></li>
<li><a href='#t_test_pval'><p>T-test wrapper, handles error for insufficient observations</p></a></li>
<li><a href='#test_multi_allelic'><p>test for multi-allelic CNVs</p></a></li>
<li><a href='#theta_hat_roll'><p>Rolling estimate of imbalance level theta</p></a></li>
<li><a href='#theta_hat_seg'><p>Estimate of imbalance level theta in a segment</p></a></li>
<li><a href='#transfer_links'><p>Annotate the direct upstream or downstream node on the edges</p></a></li>
<li><a href='#upgma'><p>UPGMA and WPGMA clustering</p></a></li>
<li><a href='#vcf_meta'><p>example VCF header</p></a></li>
<li><a href='#viterbi_loh'><p>Viterbi for clonal LOH detection</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Haplotype-Aware CNV Analysis from scRNA-Seq</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/kharchenkolab/numbat/">https://github.com/kharchenkolab/numbat/</a>,
<a href="https://kharchenkolab.github.io/numbat/">https://kharchenkolab.github.io/numbat/</a></td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A computational method that infers copy number variations (CNVs) in cancer scRNA-seq data and reconstructs the tumor phylogeny. 'numbat' integrates signals from gene expression, allelic ratio, and population haplotype structures to accurately infer allele-specific CNVs in single cells and reconstruct their lineage relationship. 'numbat' can be used to: 1. detect allele-specific copy number variations from single-cells; 2. differentiate tumor versus normal cells in the tumor microenvironment; 3. infer the clonal architecture and evolutionary history of profiled tumors. 'numbat' does not require tumor/normal-paired DNA or genotype data, but operates solely on the donor scRNA-data data (for example, 10x Cell Ranger output). Additional examples and documentations are available at <a href="https://kharchenkolab.github.io/numbat/">https://kharchenkolab.github.io/numbat/</a>. For details on the method please see Gao et al. Nature Biotechnology (2022) &lt;<a href="https://doi.org/10.1038%2Fs41587-022-01468-y">doi:10.1038/s41587-022-01468-y</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0), Matrix</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, caTools, data.table, dendextend, dplyr, GenomicRanges,
ggplot2, ggraph, ggtree, glue, hahmmr, igraph, IRanges, logger,
magrittr, methods, optparse, parallel, parallelDist, patchwork,
pryr, purrr, Rcpp, RhpcBLASctl, R.utils, scales, scistreer (&ge;
1.1.0), stats4, stringr, tibble, tidygraph, tidyr, vcfR, zoo</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggrastr, ggrepel, knitr, matrixStats, testthat (&ge; 3.0.0),</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, roptim</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>Author:</td>
<td>Teng Gao [cre, aut],
  Ruslan Soldatov [aut],
  Hirak Sarkar [aut],
  Evan Biederstedt [aut],
  Peter Kharchenko [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Teng Gao &lt;tgaoteng@gmail.com&gt;</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-23 03:16:40 UTC; tenggao</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-23 08:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='acen_hg19'>centromere regions (hg19)</h2><span id='topic+acen_hg19'></span>

<h3>Description</h3>

<p>centromere regions (hg19)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>acen_hg19
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 22 rows and 3 columns.
</p>

<hr>
<h2 id='acen_hg38'>centromere regions (hg38)</h2><span id='topic+acen_hg38'></span>

<h3>Description</h3>

<p>centromere regions (hg38)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>acen_hg38
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 22 rows and 3 columns.
</p>

<hr>
<h2 id='aggregate_counts'>Utility function to make reference gene expression profiles</h2><span id='topic+aggregate_counts'></span>

<h3>Description</h3>

<p>Utility function to make reference gene expression profiles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aggregate_counts(count_mat, annot, normalized = TRUE, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aggregate_counts_+3A_count_mat">count_mat</code></td>
<td>
<p>matrix/dgCMatrix Gene expression counts</p>
</td></tr>
<tr><td><code id="aggregate_counts_+3A_annot">annot</code></td>
<td>
<p>dataframe Cell annotation with columns &quot;cell&quot; and &quot;group&quot;</p>
</td></tr>
<tr><td><code id="aggregate_counts_+3A_normalized">normalized</code></td>
<td>
<p>logical Whether to return normalized expression values</p>
</td></tr>
<tr><td><code id="aggregate_counts_+3A_verbose">verbose</code></td>
<td>
<p>logical Verbosity</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix Reference gene expression levels
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ref_custom = aggregate_counts(count_mat_ref, annot_ref, verbose = FALSE)
</code></pre>

<hr>
<h2 id='analyze_bulk'>Call CNVs in a pseudobulk profile using the Numbat joint HMM</h2><span id='topic+analyze_bulk'></span>

<h3>Description</h3>

<p>Call CNVs in a pseudobulk profile using the Numbat joint HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyze_bulk(
  bulk,
  t = 1e-05,
  gamma = 20,
  theta_min = 0.08,
  logphi_min = 0.25,
  nu = 1,
  min_genes = 10,
  exp_only = FALSE,
  allele_only = FALSE,
  bal_cnv = TRUE,
  retest = TRUE,
  find_diploid = TRUE,
  diploid_chroms = NULL,
  classify_allele = FALSE,
  run_hmm = TRUE,
  prior = NULL,
  exclude_neu = TRUE,
  phasing = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="analyze_bulk_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pesudobulk profile</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_t">t</code></td>
<td>
<p>numeric Transition probability</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_gamma">gamma</code></td>
<td>
<p>numeric Dispersion parameter for the Beta-Binomial allele model</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_theta_min">theta_min</code></td>
<td>
<p>numeric Minimum imbalance threshold</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_logphi_min">logphi_min</code></td>
<td>
<p>numeric Minimum log expression deviation threshold</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_nu">nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_min_genes">min_genes</code></td>
<td>
<p>integer Minimum number of genes to call an event</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_exp_only">exp_only</code></td>
<td>
<p>logical Whether to run expression-only HMM</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_allele_only">allele_only</code></td>
<td>
<p>logical Whether to run allele-only HMM</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_bal_cnv">bal_cnv</code></td>
<td>
<p>logical Whether to call balanced amplifications/deletions</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_retest">retest</code></td>
<td>
<p>logical Whether to retest CNVs after Viterbi decoding</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_find_diploid">find_diploid</code></td>
<td>
<p>logical Whether to run diploid region identification routine</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_diploid_chroms">diploid_chroms</code></td>
<td>
<p>character vector User-given chromosomes that are known to be in diploid state</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_classify_allele">classify_allele</code></td>
<td>
<p>logical Whether to only classify allele (internal use only)</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_run_hmm">run_hmm</code></td>
<td>
<p>logical Whether to run HMM (internal use only)</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_prior">prior</code></td>
<td>
<p>numeric vector Prior probabilities of states (internal use only)</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_exclude_neu">exclude_neu</code></td>
<td>
<p>logical Whether to exclude neutral segments from retesting (internal use only)</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_phasing">phasing</code></td>
<td>
<p>logical Whether to use phasing information (internal use only)</p>
</td></tr>
<tr><td><code id="analyze_bulk_+3A_verbose">verbose</code></td>
<td>
<p>logical Verbosity</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a pseudobulk profile dataframe with called CNV information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bulk_analyzed = analyze_bulk(bulk_example, t = 1e-5, find_diploid = FALSE, retest = FALSE)
</code></pre>

<hr>
<h2 id='annot_consensus'>Annotate a consensus segments on a pseudobulk dataframe</h2><span id='topic+annot_consensus'></span>

<h3>Description</h3>

<p>Annotate a consensus segments on a pseudobulk dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annot_consensus(bulk, segs_consensus, join_mode = "inner")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annot_consensus_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
<tr><td><code id="annot_consensus_+3A_segs_consensus">segs_consensus</code></td>
<td>
<p>datatframe Consensus segment dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk profile
</p>

<hr>
<h2 id='annot_haplo_segs'>Annotate haplotype segments after HMM decoding</h2><span id='topic+annot_haplo_segs'></span>

<h3>Description</h3>

<p>Annotate haplotype segments after HMM decoding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annot_haplo_segs(bulk)
</code></pre>

<hr>
<h2 id='annot_ref'>example reference cell annotation</h2><span id='topic+annot_ref'></span>

<h3>Description</h3>

<p>example reference cell annotation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annot_ref
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 50 rows and 2 columns.
</p>

<hr>
<h2 id='annot_segs'>Annotate copy number segments after HMM decoding</h2><span id='topic+annot_segs'></span>

<h3>Description</h3>

<p>Annotate copy number segments after HMM decoding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annot_segs(bulk, var = "cnv_state")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annot_segs_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a pseudobulk dataframe
</p>

<hr>
<h2 id='annot_theta_mle'>Annotate the theta parameter for each segment</h2><span id='topic+annot_theta_mle'></span>

<h3>Description</h3>

<p>Annotate the theta parameter for each segment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annot_theta_mle(bulk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annot_theta_mle_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk profile
</p>

<hr>
<h2 id='annot_theta_roll'>Annotate rolling estimate of imbalance level theta</h2><span id='topic+annot_theta_roll'></span>

<h3>Description</h3>

<p>Annotate rolling estimate of imbalance level theta
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annot_theta_roll(bulk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annot_theta_roll_+3A_bulk">bulk</code></td>
<td>
<p>a pseudobulk dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a pseudobulk dataframe
</p>

<hr>
<h2 id='annotate_genes'>Annotate genes on allele dataframe</h2><span id='topic+annotate_genes'></span>

<h3>Description</h3>

<p>Annotate genes on allele dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotate_genes(df, gtf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotate_genes_+3A_df">df</code></td>
<td>
<p>dataframe Allele count dataframe</p>
</td></tr>
<tr><td><code id="annotate_genes_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Gene gtf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Allele dataframe with gene column
</p>

<hr>
<h2 id='approx_phi_post'>Laplace approximation of the posterior of expression fold change phi</h2><span id='topic+approx_phi_post'></span>

<h3>Description</h3>

<p>Laplace approximation of the posterior of expression fold change phi
</p>


<h3>Usage</h3>

<pre><code class='language-R'>approx_phi_post(
  Y_obs,
  lambda_ref,
  d,
  alpha = NULL,
  beta = NULL,
  mu = NULL,
  sig = NULL,
  lower = 0.2,
  upper = 10,
  start = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="approx_phi_post_+3A_y_obs">Y_obs</code></td>
<td>
<p>numeric vector Gene expression counts</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_lambda_ref">lambda_ref</code></td>
<td>
<p>numeric vector Reference expression levels</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_d">d</code></td>
<td>
<p>numeric Total library size</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_alpha">alpha</code></td>
<td>
<p>numeric Shape parameter of the gamma distribution</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_beta">beta</code></td>
<td>
<p>numeric Rate parameter of the gamma distribution</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_mu">mu</code></td>
<td>
<p>numeric Mean of the normal distribution</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_sig">sig</code></td>
<td>
<p>numeric Standard deviation of the normal distribution</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_lower">lower</code></td>
<td>
<p>numeric Lower bound of phi</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_upper">upper</code></td>
<td>
<p>numeric Upper bound of phi</p>
</td></tr>
<tr><td><code id="approx_phi_post_+3A_start">start</code></td>
<td>
<p>numeric Starting value of phi</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric MLE of phi and its standard deviation
</p>

<hr>
<h2 id='approx_theta_post'>Laplace approximation of the posterior of allelic imbalance theta</h2><span id='topic+approx_theta_post'></span>

<h3>Description</h3>

<p>Laplace approximation of the posterior of allelic imbalance theta
</p>


<h3>Usage</h3>

<pre><code class='language-R'>approx_theta_post(
  pAD,
  DP,
  p_s,
  lower = 0.001,
  upper = 0.499,
  start = 0.25,
  gamma = 20
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="approx_theta_post_+3A_pad">pAD</code></td>
<td>
<p>numeric vector Variant allele depth</p>
</td></tr>
<tr><td><code id="approx_theta_post_+3A_dp">DP</code></td>
<td>
<p>numeric vector Total allele depth</p>
</td></tr>
<tr><td><code id="approx_theta_post_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Variant allele frequency</p>
</td></tr>
<tr><td><code id="approx_theta_post_+3A_lower">lower</code></td>
<td>
<p>numeric Lower bound of theta</p>
</td></tr>
<tr><td><code id="approx_theta_post_+3A_upper">upper</code></td>
<td>
<p>numeric Upper bound of theta</p>
</td></tr>
<tr><td><code id="approx_theta_post_+3A_start">start</code></td>
<td>
<p>numeric Starting value of theta</p>
</td></tr>
<tr><td><code id="approx_theta_post_+3A_gamma">gamma</code></td>
<td>
<p>numeric Gamma parameter of the beta-binomial distribution</p>
</td></tr>
</table>

<hr>
<h2 id='binary_entropy'>calculate entropy for a binary variable</h2><span id='topic+binary_entropy'></span>

<h3>Description</h3>

<p>calculate entropy for a binary variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binary_entropy(p)
</code></pre>

<hr>
<h2 id='bulk_example'>example pseudobulk dataframe</h2><span id='topic+bulk_example'></span>

<h3>Description</h3>

<p>example pseudobulk dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bulk_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_df</code> (inherits from <code>tbl</code>, <code>data.frame</code>) with 3935 rows and 83 columns.
</p>

<hr>
<h2 id='calc_allele_lik'>Calculate allele likelihoods</h2><span id='topic+calc_allele_lik'></span>

<h3>Description</h3>

<p>Calculate allele likelihoods
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_allele_lik(pAD, DP, p_s, theta, gamma = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_allele_lik_+3A_pad">pAD</code></td>
<td>
<p>integer vector Paternal allele counts</p>
</td></tr>
<tr><td><code id="calc_allele_lik_+3A_dp">DP</code></td>
<td>
<p>integer vector Total alelle counts</p>
</td></tr>
<tr><td><code id="calc_allele_lik_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="calc_allele_lik_+3A_theta">theta</code></td>
<td>
<p>numeric Haplotype imbalance</p>
</td></tr>
<tr><td><code id="calc_allele_lik_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion in the allele-specific expression</p>
</td></tr>
</table>

<hr>
<h2 id='calc_allele_LLR'>Calculate LLR for an allele HMM</h2><span id='topic+calc_allele_LLR'></span>

<h3>Description</h3>

<p>Calculate LLR for an allele HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_allele_LLR(pAD, DP, p_s, theta_mle, theta_0 = 0, gamma = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_allele_LLR_+3A_pad">pAD</code></td>
<td>
<p>numeric vector Phased allele depth</p>
</td></tr>
<tr><td><code id="calc_allele_LLR_+3A_dp">DP</code></td>
<td>
<p>numeric vector Total allele depth</p>
</td></tr>
<tr><td><code id="calc_allele_LLR_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="calc_allele_LLR_+3A_theta_mle">theta_mle</code></td>
<td>
<p>numeric MLE of imbalance level theta (alternative hypothesis)</p>
</td></tr>
<tr><td><code id="calc_allele_LLR_+3A_theta_0">theta_0</code></td>
<td>
<p>numeric Imbalance level in the null hypothesis</p>
</td></tr>
<tr><td><code id="calc_allele_LLR_+3A_gamma">gamma</code></td>
<td>
<p>numeric Dispersion parameter for the Beta-Binomial allele model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric Log-likelihood ratio
</p>

<hr>
<h2 id='calc_cluster_dist'>Calculate expression distance matrix between cell populatoins</h2><span id='topic+calc_cluster_dist'></span>

<h3>Description</h3>

<p>Calculate expression distance matrix between cell populatoins
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_cluster_dist(count_mat, cell_annot)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_cluster_dist_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix Gene expression counts</p>
</td></tr>
<tr><td><code id="calc_cluster_dist_+3A_cell_annot">cell_annot</code></td>
<td>
<p>dataframe specifying the cell ID and cluster memberships</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a distance matrix
</p>

<hr>
<h2 id='calc_exp_LLR'>Calculate LLR for an expression HMM</h2><span id='topic+calc_exp_LLR'></span>

<h3>Description</h3>

<p>Calculate LLR for an expression HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_exp_LLR(
  Y_obs,
  lambda_ref,
  d,
  phi_mle,
  mu = NULL,
  sig = NULL,
  alpha = NULL,
  beta = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_exp_LLR_+3A_y_obs">Y_obs</code></td>
<td>
<p>numeric vector Gene expression counts</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_lambda_ref">lambda_ref</code></td>
<td>
<p>numeric vector Reference expression levels</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_d">d</code></td>
<td>
<p>numeric vector Total library size</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_phi_mle">phi_mle</code></td>
<td>
<p>numeric MLE of expression fold change phi (alternative hypothesis)</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_mu">mu</code></td>
<td>
<p>numeric Mean parameter for the PLN expression model</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_sig">sig</code></td>
<td>
<p>numeric Dispersion parameter for the PLN expression model</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_alpha">alpha</code></td>
<td>
<p>numeric Hyperparameter for the gamma poisson model (not used)</p>
</td></tr>
<tr><td><code id="calc_exp_LLR_+3A_beta">beta</code></td>
<td>
<p>numeric Hyperparameter for the gamma poisson model (not used)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric Log-likelihood ratio
</p>

<hr>
<h2 id='calc_phi_mle_lnpois'>Calculate the MLE of expression fold change phi</h2><span id='topic+calc_phi_mle_lnpois'></span>

<h3>Description</h3>

<p>Calculate the MLE of expression fold change phi
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_phi_mle_lnpois(Y_obs, lambda_ref, d, mu, sig, lower = 0.1, upper = 10)
</code></pre>

<hr>
<h2 id='check_allele_df'>Check the format of a allele dataframe</h2><span id='topic+check_allele_df'></span>

<h3>Description</h3>

<p>Check the format of a allele dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_allele_df(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_allele_df_+3A_df">df</code></td>
<td>
<p>dataframe Allele dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Allele dataframe
</p>

<hr>
<h2 id='check_contam'>check inter-individual contamination</h2><span id='topic+check_contam'></span>

<h3>Description</h3>

<p>check inter-individual contamination
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_contam(bulk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_contam_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
</table>

<hr>
<h2 id='check_exp_noise'>check noise level</h2><span id='topic+check_exp_noise'></span>

<h3>Description</h3>

<p>check noise level
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_exp_noise(bulk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_exp_noise_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
</table>

<hr>
<h2 id='check_exp_ref'>check the format of lambdas_ref</h2><span id='topic+check_exp_ref'></span>

<h3>Description</h3>

<p>check the format of lambdas_ref
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_exp_ref(lambdas_ref)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_exp_ref_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Expression reference profile</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix Expression reference profile
</p>

<hr>
<h2 id='check_matrix'>Check the format of a count matrix</h2><span id='topic+check_matrix'></span>

<h3>Description</h3>

<p>Check the format of a count matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_matrix(count_mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_matrix_+3A_count_mat">count_mat</code></td>
<td>
<p>matrix Count matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix Count matrix
</p>

<hr>
<h2 id='check_segs_fix'>check the format of a given consensus segment dataframe</h2><span id='topic+check_segs_fix'></span>

<h3>Description</h3>

<p>check the format of a given consensus segment dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_segs_fix(segs_consensus_fix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_segs_fix_+3A_segs_consensus_fix">segs_consensus_fix</code></td>
<td>
<p>dataframe Consensus segment dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Consensus segment dataframe
</p>

<hr>
<h2 id='check_segs_loh'>Check the format of a given clonal LOH segment dataframe</h2><span id='topic+check_segs_loh'></span>

<h3>Description</h3>

<p>Check the format of a given clonal LOH segment dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_segs_loh(segs_loh)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_segs_loh_+3A_segs_loh">segs_loh</code></td>
<td>
<p>dataframe Clonal LOH segment dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Clonal LOH segment dataframe
</p>

<hr>
<h2 id='choose_ref_cor'>choose beest reference for each cell based on correlation</h2><span id='topic+choose_ref_cor'></span>

<h3>Description</h3>

<p>choose beest reference for each cell based on correlation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>choose_ref_cor(count_mat, lambdas_ref, gtf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="choose_ref_cor_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix Gene expression counts</p>
</td></tr>
<tr><td><code id="choose_ref_cor_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Reference expression profiles</p>
</td></tr>
<tr><td><code id="choose_ref_cor_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>named vector Best references for each cell
</p>

<hr>
<h2 id='chrom_sizes_hg19'>chromosome sizes (hg19)</h2><span id='topic+chrom_sizes_hg19'></span>

<h3>Description</h3>

<p>chromosome sizes (hg19)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chrom_sizes_hg19
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 22 rows and 2 columns.
</p>

<hr>
<h2 id='chrom_sizes_hg38'>chromosome sizes (hg38)</h2><span id='topic+chrom_sizes_hg38'></span>

<h3>Description</h3>

<p>chromosome sizes (hg38)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chrom_sizes_hg38
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 22 rows and 2 columns.
</p>

<hr>
<h2 id='classify_alleles'>classify alleles using viterbi and forward-backward</h2><span id='topic+classify_alleles'></span>

<h3>Description</h3>

<p>classify alleles using viterbi and forward-backward
</p>


<h3>Usage</h3>

<pre><code class='language-R'>classify_alleles(bulk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="classify_alleles_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pesudobulk profile</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pesudobulk profile
</p>

<hr>
<h2 id='cnv_heatmap'>Plot CNV heatmap</h2><span id='topic+cnv_heatmap'></span>

<h3>Description</h3>

<p>Plot CNV heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cnv_heatmap(
  segs,
  var = "group",
  label_group = TRUE,
  legend = TRUE,
  exclude_gap = TRUE,
  genome = "hg38"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cnv_heatmap_+3A_segs">segs</code></td>
<td>
<p>dataframe Segments to plot. Need columns &quot;seg_start&quot;, &quot;seg_end&quot;, &quot;cnv_state&quot;</p>
</td></tr>
<tr><td><code id="cnv_heatmap_+3A_var">var</code></td>
<td>
<p>character Column to facet by</p>
</td></tr>
<tr><td><code id="cnv_heatmap_+3A_label_group">label_group</code></td>
<td>
<p>logical Label the groups</p>
</td></tr>
<tr><td><code id="cnv_heatmap_+3A_legend">legend</code></td>
<td>
<p>logical Display the legend</p>
</td></tr>
<tr><td><code id="cnv_heatmap_+3A_exclude_gap">exclude_gap</code></td>
<td>
<p>logical Whether to mark gap regions</p>
</td></tr>
<tr><td><code id="cnv_heatmap_+3A_genome">genome</code></td>
<td>
<p>character Genome build, either 'hg38' or 'hg19'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot Heatmap of CNVs along the genome
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = cnv_heatmap(segs_example)
</code></pre>

<hr>
<h2 id='combine_bulk'>Combine allele and expression pseudobulks</h2><span id='topic+combine_bulk'></span>

<h3>Description</h3>

<p>Combine allele and expression pseudobulks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_bulk(allele_bulk, exp_bulk)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_bulk_+3A_allele_bulk">allele_bulk</code></td>
<td>
<p>dataframe Bulk allele profile</p>
</td></tr>
<tr><td><code id="combine_bulk_+3A_exp_bulk">exp_bulk</code></td>
<td>
<p>dataframe Bulk expression profile</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk allele and expression profile
</p>

<hr>
<h2 id='compute_posterior'>Do bayesian averaging to get posteriors</h2><span id='topic+compute_posterior'></span>

<h3>Description</h3>

<p>Do bayesian averaging to get posteriors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_posterior(PL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_posterior_+3A_pl">PL</code></td>
<td>
<p>dataframe Likelihoods and priors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Posteriors
</p>

<hr>
<h2 id='contract_nodes'>Merge adjacent set of nodes</h2><span id='topic+contract_nodes'></span>

<h3>Description</h3>

<p>Merge adjacent set of nodes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contract_nodes(G, vset, node_tar = NULL, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="contract_nodes_+3A_g">G</code></td>
<td>
<p>igraph Mutation graph</p>
</td></tr>
<tr><td><code id="contract_nodes_+3A_vset">vset</code></td>
<td>
<p>vector Set of adjacent vertices to merge</p>
</td></tr>
</table>


<h3>Value</h3>

<p>igraph Mutation graph
</p>

<hr>
<h2 id='count_mat_example'>example gene expression count matrix</h2><span id='topic+count_mat_example'></span>

<h3>Description</h3>

<p>example gene expression count matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_mat_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>dgCMatrix</code> with 1024 rows and 173 columns.
</p>

<hr>
<h2 id='count_mat_ref'>example reference count matrix</h2><span id='topic+count_mat_ref'></span>

<h3>Description</h3>

<p>example reference count matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_mat_ref
</code></pre>


<h3>Format</h3>

<p>An object of class <code>dgCMatrix</code> with 1000 rows and 50 columns.
</p>

<hr>
<h2 id='detect_clonal_loh'>Call clonal LOH using SNP density. 
Rcommended for cell lines or tumor samples with no normal cells.</h2><span id='topic+detect_clonal_loh'></span>

<h3>Description</h3>

<p>Call clonal LOH using SNP density. 
Rcommended for cell lines or tumor samples with no normal cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>detect_clonal_loh(bulk, t = 1e-05, snp_rate_loh = 5, min_depth = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="detect_clonal_loh_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
<tr><td><code id="detect_clonal_loh_+3A_t">t</code></td>
<td>
<p>numeric Transition probability</p>
</td></tr>
<tr><td><code id="detect_clonal_loh_+3A_snp_rate_loh">snp_rate_loh</code></td>
<td>
<p>numeric The assumed SNP density in clonal LOH regions</p>
</td></tr>
<tr><td><code id="detect_clonal_loh_+3A_min_depth">min_depth</code></td>
<td>
<p>integer Minimum coverage to filter SNPs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe LOH segments
</p>


<h3>Examples</h3>

<pre><code class='language-R'>segs_loh = detect_clonal_loh(bulk_example)
</code></pre>

<hr>
<h2 id='df_allele_example'>example allele count dataframe</h2><span id='topic+df_allele_example'></span>

<h3>Description</h3>

<p>example allele count dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>df_allele_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 41167 rows and 11 columns.
</p>

<hr>
<h2 id='exp_hclust'>Run smoothed expression-based hclust</h2><span id='topic+exp_hclust'></span>

<h3>Description</h3>

<p>Run smoothed expression-based hclust
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exp_hclust(
  count_mat,
  lambdas_ref,
  gtf,
  sc_refs = NULL,
  window = 101,
  ncores = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exp_hclust_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix Gene counts</p>
</td></tr>
<tr><td><code id="exp_hclust_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Reference expression profiles</p>
</td></tr>
<tr><td><code id="exp_hclust_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript GTF</p>
</td></tr>
<tr><td><code id="exp_hclust_+3A_sc_refs">sc_refs</code></td>
<td>
<p>named list Reference choices for single cells</p>
</td></tr>
<tr><td><code id="exp_hclust_+3A_window">window</code></td>
<td>
<p>integer Sliding window size</p>
</td></tr>
<tr><td><code id="exp_hclust_+3A_ncores">ncores</code></td>
<td>
<p>integer Number of cores</p>
</td></tr>
<tr><td><code id="exp_hclust_+3A_verbose">verbose</code></td>
<td>
<p>logical Verbosity</p>
</td></tr>
</table>

<hr>
<h2 id='expand_states'>expand multi-allelic CNVs into separate entries in the single-cell posterior dataframe</h2><span id='topic+expand_states'></span>

<h3>Description</h3>

<p>expand multi-allelic CNVs into separate entries in the single-cell posterior dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expand_states(sc_post, segs_consensus)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expand_states_+3A_sc_post">sc_post</code></td>
<td>
<p>dataframe Single-cell posteriors</p>
</td></tr>
<tr><td><code id="expand_states_+3A_segs_consensus">segs_consensus</code></td>
<td>
<p>dataframe Consensus segments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Single-cell posteriors with multi-allelic CNVs split into different entries
</p>

<hr>
<h2 id='fill_neu_segs'>Fill neutral regions into consensus segments</h2><span id='topic+fill_neu_segs'></span>

<h3>Description</h3>

<p>Fill neutral regions into consensus segments
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fill_neu_segs(segs_consensus, segs_neu)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fill_neu_segs_+3A_segs_consensus">segs_consensus</code></td>
<td>
<p>dataframe CNV segments from multiple samples</p>
</td></tr>
<tr><td><code id="fill_neu_segs_+3A_segs_neu">segs_neu</code></td>
<td>
<p>dataframe Neutral segments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Collections of neutral and aberrant segments with no gaps
</p>

<hr>
<h2 id='filter_genes'>filter for mutually expressed genes</h2><span id='topic+filter_genes'></span>

<h3>Description</h3>

<p>filter for mutually expressed genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_genes(count_mat, lambdas_ref, gtf, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_genes_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix Gene expression counts</p>
</td></tr>
<tr><td><code id="filter_genes_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>named numeric vector A reference expression profile</p>
</td></tr>
<tr><td><code id="filter_genes_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector Genes that are kept after filtering
</p>

<hr>
<h2 id='find_common_diploid'>Find the common diploid region in a group of pseudobulks</h2><span id='topic+find_common_diploid'></span>

<h3>Description</h3>

<p>Find the common diploid region in a group of pseudobulks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_common_diploid(
  bulks,
  grouping = "clique",
  gamma = 20,
  theta_min = 0.08,
  t = 1e-05,
  fc_min = 2^0.25,
  alpha = 1e-04,
  min_genes = 10,
  ncores = 1,
  debug = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_common_diploid_+3A_bulks">bulks</code></td>
<td>
<p>dataframe Pseudobulk profiles (differentiated by &quot;sample&quot; column)</p>
</td></tr>
<tr><td><code id="find_common_diploid_+3A_grouping">grouping</code></td>
<td>
<p>logical Whether to use cliques or components in the graph to find dipoid cluster</p>
</td></tr>
<tr><td><code id="find_common_diploid_+3A_gamma">gamma</code></td>
<td>
<p>numeric Dispersion parameter for the Beta-Binomial allele model</p>
</td></tr>
<tr><td><code id="find_common_diploid_+3A_theta_min">theta_min</code></td>
<td>
<p>numeric Minimum imbalance threshold</p>
</td></tr>
<tr><td><code id="find_common_diploid_+3A_t">t</code></td>
<td>
<p>numeric Transition probability</p>
</td></tr>
<tr><td><code id="find_common_diploid_+3A_fc_min">fc_min</code></td>
<td>
<p>numeric Minimum fold change to call quadruploid cluster</p>
</td></tr>
<tr><td><code id="find_common_diploid_+3A_alpha">alpha</code></td>
<td>
<p>numeric FDR cut-off for q values to determine edges</p>
</td></tr>
<tr><td><code id="find_common_diploid_+3A_ncores">ncores</code></td>
<td>
<p>integer Number of cores to use</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list Ploidy information
</p>

<hr>
<h2 id='fit_bbinom'>fit a Beta-Binomial model by maximum likelihood</h2><span id='topic+fit_bbinom'></span>

<h3>Description</h3>

<p>fit a Beta-Binomial model by maximum likelihood
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_bbinom(AD, DP)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_bbinom_+3A_ad">AD</code></td>
<td>
<p>numeric vector Variant allele depth</p>
</td></tr>
<tr><td><code id="fit_bbinom_+3A_dp">DP</code></td>
<td>
<p>numeric vector Total allele depth</p>
</td></tr>
</table>


<h3>Value</h3>

<p>MLE of alpha and beta
</p>

<hr>
<h2 id='fit_gamma'>fit gamma maximum likelihood</h2><span id='topic+fit_gamma'></span>

<h3>Description</h3>

<p>fit gamma maximum likelihood
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_gamma(AD, DP, start = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_gamma_+3A_ad">AD</code></td>
<td>
<p>numeric vector Variant allele depth</p>
</td></tr>
<tr><td><code id="fit_gamma_+3A_dp">DP</code></td>
<td>
<p>numeric vector Total allele depth</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a fit
</p>

<hr>
<h2 id='fit_lnpois'>fit a PLN model by maximum likelihood</h2><span id='topic+fit_lnpois'></span>

<h3>Description</h3>

<p>fit a PLN model by maximum likelihood
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_lnpois(Y_obs, lambda_ref, d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_lnpois_+3A_y_obs">Y_obs</code></td>
<td>
<p>numeric vector Gene expression counts</p>
</td></tr>
<tr><td><code id="fit_lnpois_+3A_lambda_ref">lambda_ref</code></td>
<td>
<p>numeric vector Reference expression levels</p>
</td></tr>
<tr><td><code id="fit_lnpois_+3A_d">d</code></td>
<td>
<p>numeric Total library size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric MLE of mu and sig
</p>

<hr>
<h2 id='fit_ref_sse'>Fit a reference profile from multiple references using constrained least square</h2><span id='topic+fit_ref_sse'></span>

<h3>Description</h3>

<p>Fit a reference profile from multiple references using constrained least square
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_ref_sse(Y_obs, lambdas_ref, gtf, min_lambda = 2e-06, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_ref_sse_+3A_y_obs">Y_obs</code></td>
<td>
<p>vector</p>
</td></tr>
<tr><td><code id="fit_ref_sse_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>named vector</p>
</td></tr>
<tr><td><code id="fit_ref_sse_+3A_gtf">gtf</code></td>
<td>
<p>dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>fitted expression profile
</p>

<hr>
<h2 id='fit_snp_rate'>negative binomial model</h2><span id='topic+fit_snp_rate'></span>

<h3>Description</h3>

<p>negative binomial model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_snp_rate(gene_snps, gene_length)
</code></pre>

<hr>
<h2 id='gaps_hg19'>genome gap regions (hg19)</h2><span id='topic+gaps_hg19'></span>

<h3>Description</h3>

<p>genome gap regions (hg19)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gaps_hg19
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 28 rows and 3 columns.
</p>

<hr>
<h2 id='gaps_hg38'>genome gap regions (hg38)</h2><span id='topic+gaps_hg38'></span>

<h3>Description</h3>

<p>genome gap regions (hg38)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gaps_hg38
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 30 rows and 3 columns.
</p>

<hr>
<h2 id='generate_postfix'>Generate alphabetical postfixes</h2><span id='topic+generate_postfix'></span>

<h3>Description</h3>

<p>Generate alphabetical postfixes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_postfix(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_postfix_+3A_n">n</code></td>
<td>
<p>vector of integers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of alphabetical postfixes
</p>

<hr>
<h2 id='genotype'>Genotyping main function</h2><span id='topic+genotype'></span>

<h3>Description</h3>

<p>Genotyping main function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotype(label, samples, vcfs, outdir, het_only = FALSE, chr_prefix = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genotype_+3A_label">label</code></td>
<td>
<p>character Individual/sample label</p>
</td></tr>
<tr><td><code id="genotype_+3A_samples">samples</code></td>
<td>
<p>vector Sample names</p>
</td></tr>
<tr><td><code id="genotype_+3A_vcfs">vcfs</code></td>
<td>
<p>list of vcfR VCFs from cellsnp-lite pileup</p>
</td></tr>
<tr><td><code id="genotype_+3A_outdir">outdir</code></td>
<td>
<p>character Output directory</p>
</td></tr>
<tr><td><code id="genotype_+3A_het_only">het_only</code></td>
<td>
<p>logical Whether to only use heterozygous SNPs</p>
</td></tr>
<tr><td><code id="genotype_+3A_chr_prefix">chr_prefix</code></td>
<td>
<p>logical Whether to add chr prefix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>integer Status code
</p>

<hr>
<h2 id='get_allele_bulk'>Aggregate into pseudobulk alelle profile</h2><span id='topic+get_allele_bulk'></span>

<h3>Description</h3>

<p>Aggregate into pseudobulk alelle profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_allele_bulk(df_allele, nu = 1, min_depth = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_allele_bulk_+3A_df_allele">df_allele</code></td>
<td>
<p>dataframe Single-cell allele counts</p>
</td></tr>
<tr><td><code id="get_allele_bulk_+3A_nu">nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td></tr>
<tr><td><code id="get_allele_bulk_+3A_min_depth">min_depth</code></td>
<td>
<p>integer Minimum coverage to filter SNPs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk allele profile
</p>

<hr>
<h2 id='get_allele_hmm'>Get an allele HMM</h2><span id='topic+get_allele_hmm'></span>

<h3>Description</h3>

<p>Get an allele HMM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_allele_hmm(pAD, DP, p_s, theta, gamma = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_allele_hmm_+3A_pad">pAD</code></td>
<td>
<p>integer vector Paternal allele counts</p>
</td></tr>
<tr><td><code id="get_allele_hmm_+3A_dp">DP</code></td>
<td>
<p>integer vector Total alelle counts</p>
</td></tr>
<tr><td><code id="get_allele_hmm_+3A_p_s">p_s</code></td>
<td>
<p>numeric vector Phase switch probabilities</p>
</td></tr>
<tr><td><code id="get_allele_hmm_+3A_theta">theta</code></td>
<td>
<p>numeric Haplotype imbalance</p>
</td></tr>
<tr><td><code id="get_allele_hmm_+3A_gamma">gamma</code></td>
<td>
<p>numeric Overdispersion in the allele-specific expression</p>
</td></tr>
</table>


<h3>Value</h3>

<p>HMM object
</p>

<hr>
<h2 id='get_allele_post'>get CNV allele posteriors</h2><span id='topic+get_allele_post'></span>

<h3>Description</h3>

<p>get CNV allele posteriors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_allele_post(df_allele, haplotypes, segs_consensus)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_allele_post_+3A_df_allele">df_allele</code></td>
<td>
<p>dataframe Allele counts</p>
</td></tr>
<tr><td><code id="get_allele_post_+3A_haplotypes">haplotypes</code></td>
<td>
<p>dataframe Haplotype classification</p>
</td></tr>
<tr><td><code id="get_allele_post_+3A_segs_consensus">segs_consensus</code></td>
<td>
<p>dataframe Consensus CNV segments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Allele posteriors
</p>

<hr>
<h2 id='get_bulk'>Aggregate single-cell data into combined bulk expression and allele profile</h2><span id='topic+get_bulk'></span>

<h3>Description</h3>

<p>Aggregate single-cell data into combined bulk expression and allele profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_bulk(
  count_mat,
  lambdas_ref,
  df_allele,
  gtf,
  subset = NULL,
  min_depth = 0,
  nu = 1,
  segs_loh = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_bulk_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix Gene expression counts</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Reference expression profiles</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_df_allele">df_allele</code></td>
<td>
<p>dataframe Single-cell allele counts</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_subset">subset</code></td>
<td>
<p>vector Subset of cells to aggregate</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_min_depth">min_depth</code></td>
<td>
<p>integer Minimum coverage to filter SNPs</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_nu">nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_segs_loh">segs_loh</code></td>
<td>
<p>dataframe Segments with clonal LOH to be excluded</p>
</td></tr>
<tr><td><code id="get_bulk_+3A_verbose">verbose</code></td>
<td>
<p>logical Verbosity</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk gene expression and allele profile
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bulk_example = get_bulk(
    count_mat = count_mat_example,
    lambdas_ref = ref_hca,
    df_allele = df_allele_example,
    gtf = gtf_hg38)
</code></pre>

<hr>
<h2 id='get_clone_post'>Map cells to the phylogeny (or genotypes) based on CNV posteriors</h2><span id='topic+get_clone_post'></span>

<h3>Description</h3>

<p>Map cells to the phylogeny (or genotypes) based on CNV posteriors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clone_post(gtree, exp_post, allele_post)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_clone_post_+3A_gtree">gtree</code></td>
<td>
<p>tbl_graph A cell lineage tree</p>
</td></tr>
<tr><td><code id="get_clone_post_+3A_exp_post">exp_post</code></td>
<td>
<p>dataframe Expression posteriors</p>
</td></tr>
<tr><td><code id="get_clone_post_+3A_allele_post">allele_post</code></td>
<td>
<p>dataframe Allele posteriors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Clone posteriors
</p>

<hr>
<h2 id='get_exp_bulk'>Aggregate into bulk expression profile</h2><span id='topic+get_exp_bulk'></span>

<h3>Description</h3>

<p>Aggregate into bulk expression profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_exp_bulk(count_mat, lambdas_ref, gtf, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_exp_bulk_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix Gene expression counts</p>
</td></tr>
<tr><td><code id="get_exp_bulk_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Reference expression profiles</p>
</td></tr>
<tr><td><code id="get_exp_bulk_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk gene expression profile
</p>

<hr>
<h2 id='get_exp_likelihoods'>get the single cell expression likelihoods</h2><span id='topic+get_exp_likelihoods'></span>

<h3>Description</h3>

<p>get the single cell expression likelihoods
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_exp_likelihoods(
  exp_counts,
  diploid_chroms = NULL,
  use_loh = FALSE,
  depth_obs = NULL,
  mu = NULL,
  sigma = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_exp_likelihoods_+3A_exp_counts">exp_counts</code></td>
<td>
<p>dataframe Single-cell expression counts (CHROM, seg, cnv_state, gene, Y_obs, lambda_ref)</p>
</td></tr>
<tr><td><code id="get_exp_likelihoods_+3A_diploid_chroms">diploid_chroms</code></td>
<td>
<p>character vector Known diploid chromosomes</p>
</td></tr>
<tr><td><code id="get_exp_likelihoods_+3A_use_loh">use_loh</code></td>
<td>
<p>logical Whether to include CNLOH regions in baseline</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Single-cell CNV likelihood scores
</p>

<hr>
<h2 id='get_exp_post'>compute single-cell expression posteriors</h2><span id='topic+get_exp_post'></span>

<h3>Description</h3>

<p>compute single-cell expression posteriors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_exp_post(
  segs_consensus,
  count_mat,
  gtf,
  lambdas_ref,
  sc_refs = NULL,
  diploid_chroms = NULL,
  use_loh = NULL,
  segs_loh = NULL,
  ncores = 30,
  verbose = TRUE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_exp_post_+3A_segs_consensus">segs_consensus</code></td>
<td>
<p>dataframe Consensus segments</p>
</td></tr>
<tr><td><code id="get_exp_post_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix gene expression count matrix</p>
</td></tr>
<tr><td><code id="get_exp_post_+3A_gtf">gtf</code></td>
<td>
<p>dataframe transcript gtf</p>
</td></tr>
<tr><td><code id="get_exp_post_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Reference expression profiles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Expression posteriors
</p>

<hr>
<h2 id='get_exp_sc'>get the single cell expression dataframe</h2><span id='topic+get_exp_sc'></span>

<h3>Description</h3>

<p>get the single cell expression dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_exp_sc(segs_consensus, count_mat, gtf, segs_loh = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_exp_sc_+3A_segs_consensus">segs_consensus</code></td>
<td>
<p>dataframe Consensus segments</p>
</td></tr>
<tr><td><code id="get_exp_sc_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix gene expression count matrix</p>
</td></tr>
<tr><td><code id="get_exp_sc_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe single cell expression counts annotated with segments
</p>

<hr>
<h2 id='get_gtree'>Get a tidygraph tree with simplified mutational history.</h2><span id='topic+get_gtree'></span>

<h3>Description</h3>

<p>Specify either max_cost or n_cut. 
max_cost works similarly as h and n_cut works similarly as k in stats::cutree.
The top-level normal diploid clone is always included.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gtree(tree, P, n_cut = 0, max_cost = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gtree_+3A_tree">tree</code></td>
<td>
<p>phylo Single-cell phylogenetic tree</p>
</td></tr>
<tr><td><code id="get_gtree_+3A_p">P</code></td>
<td>
<p>matrix Genotype probability matrix</p>
</td></tr>
<tr><td><code id="get_gtree_+3A_n_cut">n_cut</code></td>
<td>
<p>integer Number of cuts on the phylogeny to define subclones</p>
</td></tr>
<tr><td><code id="get_gtree_+3A_max_cost">max_cost</code></td>
<td>
<p>numeric Likelihood threshold to collapse internal branches</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tbl_graph Phylogeny annotated with branch lengths and mutation events
</p>

<hr>
<h2 id='get_haplotype_post'>Get phased haplotypes</h2><span id='topic+get_haplotype_post'></span>

<h3>Description</h3>

<p>Get phased haplotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_haplotype_post(bulks, segs_consensus, naive = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_haplotype_post_+3A_bulks">bulks</code></td>
<td>
<p>dataframe Subtree pseudobulk profiles</p>
</td></tr>
<tr><td><code id="get_haplotype_post_+3A_segs_consensus">segs_consensus</code></td>
<td>
<p>dataframe Consensus CNV segments</p>
</td></tr>
<tr><td><code id="get_haplotype_post_+3A_naive">naive</code></td>
<td>
<p>logical Whether to use naive haplotype classification</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Posterior haplotypes
</p>

<hr>
<h2 id='get_inter_cm'>Helper function to get inter-SNP distance</h2><span id='topic+get_inter_cm'></span>

<h3>Description</h3>

<p>Helper function to get inter-SNP distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_inter_cm(d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_inter_cm_+3A_d">d</code></td>
<td>
<p>numeric vector Genetic positions in centimorgan (cM)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector Inter-SNP genetic distances
</p>

<hr>
<h2 id='get_internal_nodes'>Helper function to get the internal nodes of a dendrogram and the leafs in each subtree</h2><span id='topic+get_internal_nodes'></span>

<h3>Description</h3>

<p>Helper function to get the internal nodes of a dendrogram and the leafs in each subtree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_internal_nodes(den, node, labels)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_internal_nodes_+3A_den">den</code></td>
<td>
<p>dendrogram</p>
</td></tr>
<tr><td><code id="get_internal_nodes_+3A_node">node</code></td>
<td>
<p>character Node name</p>
</td></tr>
<tr><td><code id="get_internal_nodes_+3A_labels">labels</code></td>
<td>
<p>character vector Leaf labels</p>
</td></tr>
</table>

<hr>
<h2 id='get_joint_post'>get joint posteriors</h2><span id='topic+get_joint_post'></span>

<h3>Description</h3>

<p>get joint posteriors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_joint_post(exp_post, allele_post, segs_consensus)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_joint_post_+3A_exp_post">exp_post</code></td>
<td>
<p>dataframe Expression single-cell CNV posteriors</p>
</td></tr>
<tr><td><code id="get_joint_post_+3A_allele_post">allele_post</code></td>
<td>
<p>dataframe Allele single-cell CNV posteriors</p>
</td></tr>
<tr><td><code id="get_joint_post_+3A_segs_consensus">segs_consensus</code></td>
<td>
<p>dataframe Consensus CNV segments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Joint single-cell CNV posteriors
</p>

<hr>
<h2 id='get_lambdas_bar'>Get average reference expressio profile based on single-cell ref choices</h2><span id='topic+get_lambdas_bar'></span>

<h3>Description</h3>

<p>Get average reference expressio profile based on single-cell ref choices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_lambdas_bar(lambdas_ref, sc_refs, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_lambdas_bar_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Reference expression profiles</p>
</td></tr>
<tr><td><code id="get_lambdas_bar_+3A_sc_refs">sc_refs</code></td>
<td>
<p>vector Single-cell reference choices</p>
</td></tr>
<tr><td><code id="get_lambdas_bar_+3A_verbose">verbose</code></td>
<td>
<p>logical Print messages</p>
</td></tr>
</table>

<hr>
<h2 id='get_move_cost'>Get the cost of a mutation reassignment</h2><span id='topic+get_move_cost'></span>

<h3>Description</h3>

<p>Get the cost of a mutation reassignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_move_cost(muts, node_ori, node_tar, l_matrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_move_cost_+3A_muts">muts</code></td>
<td>
<p>character Mutations dlimited by comma</p>
</td></tr>
<tr><td><code id="get_move_cost_+3A_node_ori">node_ori</code></td>
<td>
<p>character Name of the &quot;from&quot; node</p>
</td></tr>
<tr><td><code id="get_move_cost_+3A_node_tar">node_tar</code></td>
<td>
<p>character Name of the &quot;to&quot; node</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric Likelihood cost of the mutation reassignment
</p>

<hr>
<h2 id='get_move_opt'>Get the least costly mutation reassignment</h2><span id='topic+get_move_opt'></span>

<h3>Description</h3>

<p>Get the least costly mutation reassignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_move_opt(G, l_matrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_move_opt_+3A_g">G</code></td>
<td>
<p>igraph Mutation graph</p>
</td></tr>
<tr><td><code id="get_move_opt_+3A_l_matrix">l_matrix</code></td>
<td>
<p>matrix Likelihood matrix of mutation placements</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric Lieklihood cost of performing the mutation move
</p>

<hr>
<h2 id='get_nodes_celltree'>Get the internal nodes of a dendrogram and the leafs in each subtree</h2><span id='topic+get_nodes_celltree'></span>

<h3>Description</h3>

<p>Get the internal nodes of a dendrogram and the leafs in each subtree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_nodes_celltree(hc, clusters)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_nodes_celltree_+3A_hc">hc</code></td>
<td>
<p>hclust Clustering results</p>
</td></tr>
<tr><td><code id="get_nodes_celltree_+3A_clusters">clusters</code></td>
<td>
<p>named vector Cutree output specifying the terminal clusters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list Interal node subtrees with leaf memberships
</p>

<hr>
<h2 id='get_ordered_tips'>Get ordered tips from a tree</h2><span id='topic+get_ordered_tips'></span>

<h3>Description</h3>

<p>Get ordered tips from a tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ordered_tips(tree)
</code></pre>

<hr>
<h2 id='get_segs_consensus'>Extract consensus CNV segments</h2><span id='topic+get_segs_consensus'></span>

<h3>Description</h3>

<p>Extract consensus CNV segments
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_segs_consensus(bulks, min_LLR = 5, min_overlap = 0.45, retest = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_segs_consensus_+3A_bulks">bulks</code></td>
<td>
<p>dataframe Pseudobulks</p>
</td></tr>
<tr><td><code id="get_segs_consensus_+3A_min_llr">min_LLR</code></td>
<td>
<p>numeric LLR threshold to filter CNVs</p>
</td></tr>
<tr><td><code id="get_segs_consensus_+3A_min_overlap">min_overlap</code></td>
<td>
<p>numeric Minimum overlap fraction to determine count two events as as overlapping</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Consensus segments
</p>

<hr>
<h2 id='get_segs_neu'>get neutral segments from multiple pseudobulks</h2><span id='topic+get_segs_neu'></span>

<h3>Description</h3>

<p>get neutral segments from multiple pseudobulks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_segs_neu(bulks)
</code></pre>

<hr>
<h2 id='get_snps'>process VCFs into SNP dataframe</h2><span id='topic+get_snps'></span>

<h3>Description</h3>

<p>process VCFs into SNP dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_snps(vcf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_snps_+3A_vcf">vcf</code></td>
<td>
<p>vcfR object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe SNP information
</p>

<hr>
<h2 id='get_tree_post'>Find maximum lilkelihood assignment of mutations on a tree</h2><span id='topic+get_tree_post'></span>

<h3>Description</h3>

<p>Find maximum lilkelihood assignment of mutations on a tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_tree_post(tree, P)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_tree_post_+3A_tree">tree</code></td>
<td>
<p>phylo Single-cell phylogenetic tree</p>
</td></tr>
<tr><td><code id="get_tree_post_+3A_p">P</code></td>
<td>
<p>matrix Genotype probability matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list Mutation
</p>

<hr>
<h2 id='gexp_roll_example'>example smoothed gene expression dataframe</h2><span id='topic+gexp_roll_example'></span>

<h3>Description</h3>

<p>example smoothed gene expression dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gexp_roll_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 10 rows and 2000 columns.
</p>

<hr>
<h2 id='gtf_hg19'>gene model (hg19)</h2><span id='topic+gtf_hg19'></span>

<h3>Description</h3>

<p>gene model (hg19)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtf_hg19
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 26841 rows and 5 columns.
</p>

<hr>
<h2 id='gtf_hg38'>gene model (hg38)</h2><span id='topic+gtf_hg38'></span>

<h3>Description</h3>

<p>gene model (hg38)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtf_hg38
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 26807 rows and 5 columns.
</p>

<hr>
<h2 id='gtf_mm10'>gene model (mm10)</h2><span id='topic+gtf_mm10'></span>

<h3>Description</h3>

<p>gene model (mm10)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtf_mm10
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 30336 rows and 5 columns.
</p>

<hr>
<h2 id='hc_example'>example hclust tree</h2><span id='topic+hc_example'></span>

<h3>Description</h3>

<p>example hclust tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hc_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>hclust</code> of length 7.
</p>

<hr>
<h2 id='joint_post_example'>example joint single-cell cnv posterior dataframe</h2><span id='topic+joint_post_example'></span>

<h3>Description</h3>

<p>example joint single-cell cnv posterior dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>joint_post_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 3806 rows and 71 columns.
</p>

<hr>
<h2 id='label_edges'>Annotate the direct upstream or downstream mutations on the edges</h2><span id='topic+label_edges'></span>

<h3>Description</h3>

<p>Annotate the direct upstream or downstream mutations on the edges
</p>


<h3>Usage</h3>

<pre><code class='language-R'>label_edges(G)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="label_edges_+3A_g">G</code></td>
<td>
<p>igraph Mutation graph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>igraph Mutation graph
</p>

<hr>
<h2 id='label_genotype'>Label the genotypes on a mutation graph</h2><span id='topic+label_genotype'></span>

<h3>Description</h3>

<p>Label the genotypes on a mutation graph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>label_genotype(G)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="label_genotype_+3A_g">G</code></td>
<td>
<p>igraph Mutation graph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>igraph Mutation graph
</p>

<hr>
<h2 id='log_mem'>Log memory usage</h2><span id='topic+log_mem'></span>

<h3>Description</h3>

<p>Log memory usage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>log_mem()
</code></pre>

<hr>
<h2 id='log_message'>Log a message</h2><span id='topic+log_message'></span>

<h3>Description</h3>

<p>Log a message
</p>


<h3>Usage</h3>

<pre><code class='language-R'>log_message(msg, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="log_message_+3A_msg">msg</code></td>
<td>
<p>string Message to log</p>
</td></tr>
<tr><td><code id="log_message_+3A_verbose">verbose</code></td>
<td>
<p>boolean Whether to print message to console</p>
</td></tr>
</table>

<hr>
<h2 id='make_group_bulks'>Make a group of pseudobulks</h2><span id='topic+make_group_bulks'></span>

<h3>Description</h3>

<p>Make a group of pseudobulks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_group_bulks(
  groups,
  count_mat,
  df_allele,
  lambdas_ref,
  gtf,
  min_depth = 0,
  nu = 1,
  segs_loh = NULL,
  ncores = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_group_bulks_+3A_groups">groups</code></td>
<td>
<p>list Contains fields named &quot;sample&quot;, &quot;cells&quot;, &quot;size&quot;, &quot;members&quot;</p>
</td></tr>
<tr><td><code id="make_group_bulks_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix Gene counts</p>
</td></tr>
<tr><td><code id="make_group_bulks_+3A_df_allele">df_allele</code></td>
<td>
<p>dataframe Alelle counts</p>
</td></tr>
<tr><td><code id="make_group_bulks_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Reference expression profiles</p>
</td></tr>
<tr><td><code id="make_group_bulks_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript GTF</p>
</td></tr>
<tr><td><code id="make_group_bulks_+3A_min_depth">min_depth</code></td>
<td>
<p>integer Minimum allele depth to include</p>
</td></tr>
<tr><td><code id="make_group_bulks_+3A_segs_loh">segs_loh</code></td>
<td>
<p>dataframe Segments with clonal LOH to be excluded</p>
</td></tr>
<tr><td><code id="make_group_bulks_+3A_ncores">ncores</code></td>
<td>
<p>integer Number of cores</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk profiles
</p>

<hr>
<h2 id='mark_tumor_lineage'>Mark the tumor lineage of a phylogeny</h2><span id='topic+mark_tumor_lineage'></span>

<h3>Description</h3>

<p>Mark the tumor lineage of a phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mark_tumor_lineage(gtree)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mark_tumor_lineage_+3A_gtree">gtree</code></td>
<td>
<p>tbl_graph Single-cell phylogeny</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tbl_graph Phylogeny annotated with tumor versus normal compartment
</p>

<hr>
<h2 id='Modes'>Get the modes of a vector</h2><span id='topic+Modes'></span>

<h3>Description</h3>

<p>Get the modes of a vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Modes(x)
</code></pre>

<hr>
<h2 id='mut_graph_example'>example mutation graph</h2><span id='topic+mut_graph_example'></span>

<h3>Description</h3>

<p>example mutation graph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mut_graph_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>igraph</code> of length 5.
</p>

<hr>
<h2 id='Numbat'>Numbat R6 class</h2><span id='topic+Numbat'></span>

<h3>Description</h3>

<p>Used to allow users to plot results
</p>


<h3>Value</h3>

<p>a new 'Numbat' object
</p>


<h3>Public fields</h3>

<div class="r6-fields">

<dl>
<dt><code>label</code></dt><dd><p>character Sample name</p>
</dd>
<dt><code>gtf</code></dt><dd><p>dataframe Transcript annotation</p>
</dd>
<dt><code>joint_post</code></dt><dd><p>dataframe Joint posterior</p>
</dd>
<dt><code>exp_post</code></dt><dd><p>dataframe Expression posterior</p>
</dd>
<dt><code>allele_post</code></dt><dd><p>dataframe Allele posetrior</p>
</dd>
<dt><code>bulk_subtrees</code></dt><dd><p>dataframe Bulk profiles of lineage subtrees</p>
</dd>
<dt><code>bulk_clones</code></dt><dd><p>dataframe Bulk profiles of clones</p>
</dd>
<dt><code>segs_consensus</code></dt><dd><p>dataframe Consensus segments</p>
</dd>
<dt><code>tree_post</code></dt><dd><p>list Tree posterior</p>
</dd>
<dt><code>mut_graph</code></dt><dd><p>igraph Mutation history graph</p>
</dd>
<dt><code>gtree</code></dt><dd><p>tbl_graph Single-cell phylogeny</p>
</dd>
<dt><code>clone_post</code></dt><dd><p>dataframe Clone posteriors</p>
</dd>
<dt><code>gexp_roll_wide</code></dt><dd><p>matrix Smoothed expression of single cells</p>
</dd>
<dt><code>P</code></dt><dd><p>matrix Genotype probability matrix</p>
</dd>
<dt><code>treeML</code></dt><dd><p>matrix Maximum likelihood tree as phylo object</p>
</dd>
<dt><code>hc</code></dt><dd><p>hclust Initial hierarchical clustering</p>
</dd>
</dl>

</div>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-Numbat-new"><code>Numbat$new()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_phylo_heatmap"><code>Numbat$plot_phylo_heatmap()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_exp_roll"><code>Numbat$plot_exp_roll()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_mut_history"><code>Numbat$plot_mut_history()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_sc_tree"><code>Numbat$plot_sc_tree()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_consensus"><code>Numbat$plot_consensus()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-plot_clone_profile"><code>Numbat$plot_clone_profile()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-cutree"><code>Numbat$cutree()</code></a>
</p>
</li>
<li> <p><a href="#method-Numbat-clone"><code>Numbat$clone()</code></a>
</p>
</li></ul>


<hr>
<a id="method-Numbat-new"></a>



<h4>Method <code>new()</code></h4>

<p>initialize Numbat class
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$new(out_dir, i = 2, gtf = gtf_hg38, verbose = TRUE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>out_dir</code></dt><dd><p>character string Output directory</p>
</dd>
<dt><code>i</code></dt><dd><p>integer Get results from which iteration (default=2)</p>
</dd>
<dt><code>gtf</code></dt><dd><p>dataframe Transcript gtf (default=gtf_hg38)</p>
</dd>
<dt><code>verbose</code></dt><dd><p>logical Whether to output verbose results (default=TRUE)</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>a new 'Numbat' object
</p>


<hr>
<a id="method-Numbat-plot_phylo_heatmap"></a>



<h4>Method <code>plot_phylo_heatmap()</code></h4>

<p>Plot the single-cell CNV calls in a heatmap and the corresponding phylogeny
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_phylo_heatmap(...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>...</code></dt><dd><p>additional parameters passed to plot_phylo_heatmap()</p>
</dd>
</dl>

</div>


<hr>
<a id="method-Numbat-plot_exp_roll"></a>



<h4>Method <code>plot_exp_roll()</code></h4>

<p>Plot window-smoothed expression profiles
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_exp_roll(k = 3, n_sample = 300, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>k</code></dt><dd><p>integer Number of clusters</p>
</dd>
<dt><code>n_sample</code></dt><dd><p>integer Number of cells to subsample</p>
</dd>
<dt><code>...</code></dt><dd><p>additional parameters passed to plot_exp_roll()</p>
</dd>
</dl>

</div>


<hr>
<a id="method-Numbat-plot_mut_history"></a>



<h4>Method <code>plot_mut_history()</code></h4>

<p>Plot the mutation history of the tumor
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_mut_history(...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>...</code></dt><dd><p>additional parameters passed to plot_mut_history()</p>
</dd>
</dl>

</div>


<hr>
<a id="method-Numbat-plot_sc_tree"></a>



<h4>Method <code>plot_sc_tree()</code></h4>

<p>Plot the single cell phylogeny
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_sc_tree(...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>...</code></dt><dd><p>additional parameters passed to plot_sc_tree()</p>
</dd>
</dl>

</div>


<hr>
<a id="method-Numbat-plot_consensus"></a>



<h4>Method <code>plot_consensus()</code></h4>

<p>Plot consensus segments
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_consensus(...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>...</code></dt><dd><p>additional parameters passed to plot_sc_tree()</p>
</dd>
</dl>

</div>


<hr>
<a id="method-Numbat-plot_clone_profile"></a>



<h4>Method <code>plot_clone_profile()</code></h4>

<p>Plot clone cnv profiles
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$plot_clone_profile(...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>...</code></dt><dd><p>additional parameters passed to plot_clone_profile()</p>
</dd>
</dl>

</div>


<hr>
<a id="method-Numbat-cutree"></a>



<h4>Method <code>cutree()</code></h4>

<p>Re-define subclones on the phylogeny.
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$cutree(max_cost = 0, n_cut = 0)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>max_cost</code></dt><dd><p>numeric Likelihood threshold to collapse internal branches</p>
</dd>
<dt><code>n_cut</code></dt><dd><p>integer Number of cuts on the phylogeny to define subclones</p>
</dd>
</dl>

</div>


<hr>
<a id="method-Numbat-clone"></a>



<h4>Method <code>clone()</code></h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>Numbat$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt><dd><p>Whether to make a deep clone.</p>
</dd>
</dl>

</div>



<hr>
<h2 id='phi_hat_roll'>Rolling estimate of expression fold change phi</h2><span id='topic+phi_hat_roll'></span>

<h3>Description</h3>

<p>Rolling estimate of expression fold change phi
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phi_hat_roll(Y_obs, lambda_ref, d_obs, mu, sig, h)
</code></pre>

<hr>
<h2 id='phi_hat_seg'>Estimate of expression fold change phi in a segment</h2><span id='topic+phi_hat_seg'></span>

<h3>Description</h3>

<p>Estimate of expression fold change phi in a segment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phi_hat_seg(Y_obs, lambda_ref, d, mu, sig)
</code></pre>

<hr>
<h2 id='phylogeny_example'>example single-cell phylogeny</h2><span id='topic+phylogeny_example'></span>

<h3>Description</h3>

<p>example single-cell phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylogeny_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>tbl_graph</code> (inherits from <code>igraph</code>) of length 345.
</p>

<hr>
<h2 id='plot_bulks'>Plot a group of pseudobulk HMM profiles</h2><span id='topic+plot_bulks'></span>

<h3>Description</h3>

<p>Plot a group of pseudobulk HMM profiles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_bulks(bulks, ..., ncol = 1, title = TRUE, title_size = 8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_bulks_+3A_bulks">bulks</code></td>
<td>
<p>dataframe Pseudobulk profiles annotated with &quot;sample&quot; column</p>
</td></tr>
<tr><td><code id="plot_bulks_+3A_...">...</code></td>
<td>
<p>additional parameters passed to plot_psbulk()</p>
</td></tr>
<tr><td><code id="plot_bulks_+3A_ncol">ncol</code></td>
<td>
<p>integer Number of columns</p>
</td></tr>
<tr><td><code id="plot_bulks_+3A_title">title</code></td>
<td>
<p>logical Whether to add titles to individual plots</p>
</td></tr>
<tr><td><code id="plot_bulks_+3A_title_size">title_size</code></td>
<td>
<p>numeric Size of titles</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = plot_bulks(bulk_example)
</code></pre>

<hr>
<h2 id='plot_consensus'>Plot consensus CNVs</h2><span id='topic+plot_consensus'></span>

<h3>Description</h3>

<p>Plot consensus CNVs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_consensus(segs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_consensus_+3A_segs">segs</code></td>
<td>
<p>dataframe Consensus segments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = plot_consensus(segs_example)
</code></pre>

<hr>
<h2 id='plot_exp_roll'>Plot single-cell smoothed expression magnitude heatmap</h2><span id='topic+plot_exp_roll'></span>

<h3>Description</h3>

<p>Plot single-cell smoothed expression magnitude heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_exp_roll(
  gexp_roll_wide,
  hc,
  k,
  gtf,
  lim = 0.8,
  n_sample = 300,
  reverse = TRUE,
  plot_tree = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_exp_roll_+3A_gexp_roll_wide">gexp_roll_wide</code></td>
<td>
<p>matrix Cell x gene smoothed expression magnitudes</p>
</td></tr>
<tr><td><code id="plot_exp_roll_+3A_hc">hc</code></td>
<td>
<p>hclust Hierarchical clustring result</p>
</td></tr>
<tr><td><code id="plot_exp_roll_+3A_k">k</code></td>
<td>
<p>integer Number of clusters</p>
</td></tr>
<tr><td><code id="plot_exp_roll_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript GTF</p>
</td></tr>
<tr><td><code id="plot_exp_roll_+3A_lim">lim</code></td>
<td>
<p>numeric Limit for expression magnitudes</p>
</td></tr>
<tr><td><code id="plot_exp_roll_+3A_n_sample">n_sample</code></td>
<td>
<p>integer Number of cells to subsample</p>
</td></tr>
<tr><td><code id="plot_exp_roll_+3A_reverse">reverse</code></td>
<td>
<p>logical Whether to reverse the cell order</p>
</td></tr>
<tr><td><code id="plot_exp_roll_+3A_plot_tree">plot_tree</code></td>
<td>
<p>logical Whether to plot the dendrogram</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot A single-cell heatmap of window-smoothed expression CNV signals
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = plot_exp_roll(gexp_roll_example, gtf = gtf_hg38, hc = hc_example, k = 3)
</code></pre>

<hr>
<h2 id='plot_mut_history'>Plot mutational history</h2><span id='topic+plot_mut_history'></span>

<h3>Description</h3>

<p>Plot mutational history
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_mut_history(
  G,
  clone_post = NULL,
  edge_label_size = 4,
  node_label_size = 6,
  node_size = 10,
  arrow_size = 2,
  show_clone_size = TRUE,
  show_distance = TRUE,
  legend = TRUE,
  edge_label = TRUE,
  node_label = TRUE,
  horizontal = TRUE,
  pal = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_mut_history_+3A_g">G</code></td>
<td>
<p>igraph Mutation history graph</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_clone_post">clone_post</code></td>
<td>
<p>dataframe Clone assignment posteriors</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_edge_label_size">edge_label_size</code></td>
<td>
<p>numeric Size of edge label</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_node_label_size">node_label_size</code></td>
<td>
<p>numeric Size of node label</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_node_size">node_size</code></td>
<td>
<p>numeric Size of nodes</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_arrow_size">arrow_size</code></td>
<td>
<p>numeric Size of arrows</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_show_clone_size">show_clone_size</code></td>
<td>
<p>logical Whether to show clone size</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_show_distance">show_distance</code></td>
<td>
<p>logical Whether to show evolutionary distance between clones</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_legend">legend</code></td>
<td>
<p>logical Whether to show legend</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_edge_label">edge_label</code></td>
<td>
<p>logical Whether to label edges</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_node_label">node_label</code></td>
<td>
<p>logical Whether to label nodes</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_horizontal">horizontal</code></td>
<td>
<p>logical Whether to use horizontal layout</p>
</td></tr>
<tr><td><code id="plot_mut_history_+3A_pal">pal</code></td>
<td>
<p>named vector Node colors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = plot_mut_history(mut_graph_example)
</code></pre>

<hr>
<h2 id='plot_phylo_heatmap'>Plot single-cell CNV calls along with the clonal phylogeny</h2><span id='topic+plot_phylo_heatmap'></span>

<h3>Description</h3>

<p>Plot single-cell CNV calls along with the clonal phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_phylo_heatmap(
  gtree,
  joint_post,
  segs_consensus,
  clone_post = NULL,
  p_min = 0.9,
  annot = NULL,
  pal_annot = NULL,
  annot_title = "Annotation",
  annot_scale = NULL,
  clone_dict = NULL,
  clone_bar = TRUE,
  clone_stack = TRUE,
  pal_clone = NULL,
  clone_title = "Genotype",
  clone_legend = TRUE,
  line_width = 0.1,
  tree_height = 1,
  branch_width = 0.2,
  tip_length = 0.2,
  annot_bar_width = 0.25,
  clone_bar_width = 0.25,
  bar_label_size = 7,
  tvn_line = TRUE,
  clone_line = FALSE,
  exclude_gap = FALSE,
  root_edge = TRUE,
  raster = FALSE,
  show_phylo = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_phylo_heatmap_+3A_gtree">gtree</code></td>
<td>
<p>tbl_graph The single-cell phylogeny</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_joint_post">joint_post</code></td>
<td>
<p>dataframe Joint single cell CNV posteriors</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_segs_consensus">segs_consensus</code></td>
<td>
<p>datatframe Consensus segment dataframe</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_clone_post">clone_post</code></td>
<td>
<p>dataframe Clone assignment posteriors</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_p_min">p_min</code></td>
<td>
<p>numeric Probability threshold to display CNV calls</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_annot">annot</code></td>
<td>
<p>dataframe Cell annotations, dataframe with 'cell' and additional annotation columns</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_pal_annot">pal_annot</code></td>
<td>
<p>named vector Colors for cell annotations</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_annot_title">annot_title</code></td>
<td>
<p>character Legend title for the annotation bar</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_annot_scale">annot_scale</code></td>
<td>
<p>ggplot scale Color scale for the annotation bar</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_clone_dict">clone_dict</code></td>
<td>
<p>named vector Clone annotations, mapping from cell name to clones</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_clone_bar">clone_bar</code></td>
<td>
<p>logical Whether to display clone bar plot</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_clone_stack">clone_stack</code></td>
<td>
<p>character Whether to plot clone assignment probabilities as stacked bar</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_pal_clone">pal_clone</code></td>
<td>
<p>named vector Clone colors</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_clone_title">clone_title</code></td>
<td>
<p>character Legend title for the clone bar</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_clone_legend">clone_legend</code></td>
<td>
<p>logical Whether to display the clone legend</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_line_width">line_width</code></td>
<td>
<p>numeric Line width for CNV heatmap</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_tree_height">tree_height</code></td>
<td>
<p>numeric Relative height of the phylogeny plot</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_branch_width">branch_width</code></td>
<td>
<p>numeric Line width in the phylogeny</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_tip_length">tip_length</code></td>
<td>
<p>numeric Length of tips in the phylogeny</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_annot_bar_width">annot_bar_width</code></td>
<td>
<p>numeric Width of annotation bar</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_clone_bar_width">clone_bar_width</code></td>
<td>
<p>numeric Width of clone genotype bar</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_bar_label_size">bar_label_size</code></td>
<td>
<p>numeric Size of sidebar text labels</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_tvn_line">tvn_line</code></td>
<td>
<p>logical Whether to draw line separating tumor and normal cells</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_clone_line">clone_line</code></td>
<td>
<p>logical Whether to display borders for clones in the heatmap</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_exclude_gap">exclude_gap</code></td>
<td>
<p>logical Whether to mark gap regions</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_root_edge">root_edge</code></td>
<td>
<p>logical Whether to plot root edge</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_raster">raster</code></td>
<td>
<p>logical Whether to raster images</p>
</td></tr>
<tr><td><code id="plot_phylo_heatmap_+3A_show_phylo">show_phylo</code></td>
<td>
<p>logical Whether to display phylogeny on y axis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot panel
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = plot_phylo_heatmap(
    gtree = phylogeny_example,
    joint_post = joint_post_example,
    segs_consensus = segs_example)
</code></pre>

<hr>
<h2 id='plot_psbulk'>Plot a pseudobulk HMM profile</h2><span id='topic+plot_psbulk'></span>

<h3>Description</h3>

<p>Plot a pseudobulk HMM profile
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_psbulk(
  bulk,
  use_pos = TRUE,
  allele_only = FALSE,
  min_LLR = 5,
  min_depth = 8,
  exp_limit = 2,
  phi_mle = TRUE,
  theta_roll = FALSE,
  dot_size = 0.8,
  dot_alpha = 0.5,
  legend = TRUE,
  exclude_gap = TRUE,
  genome = "hg38",
  text_size = 10,
  raster = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_psbulk_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_use_pos">use_pos</code></td>
<td>
<p>logical Use marker position instead of index as x coordinate</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_allele_only">allele_only</code></td>
<td>
<p>logical Only plot alleles</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_min_llr">min_LLR</code></td>
<td>
<p>numeric LLR threshold for event filtering</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_min_depth">min_depth</code></td>
<td>
<p>numeric Minimum coverage depth for a SNP to be plotted</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_exp_limit">exp_limit</code></td>
<td>
<p>numeric Expression logFC axis limit</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_phi_mle">phi_mle</code></td>
<td>
<p>logical Whether to plot estimates of segmental expression fold change</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_theta_roll">theta_roll</code></td>
<td>
<p>logical Whether to plot rolling estimates of allele imbalance</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_dot_size">dot_size</code></td>
<td>
<p>numeric Size of marker dots</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_dot_alpha">dot_alpha</code></td>
<td>
<p>numeric Transparency of the marker dots</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_legend">legend</code></td>
<td>
<p>logical Whether to show legend</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_exclude_gap">exclude_gap</code></td>
<td>
<p>logical Whether to mark gap regions and centromeres</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_genome">genome</code></td>
<td>
<p>character Genome build, either 'hg38' or 'hg19'</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_text_size">text_size</code></td>
<td>
<p>numeric Size of text in the plot</p>
</td></tr>
<tr><td><code id="plot_psbulk_+3A_raster">raster</code></td>
<td>
<p>logical Whether to raster images</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot Plot of pseudobulk HMM profile
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = plot_psbulk(bulk_example)
</code></pre>

<hr>
<h2 id='plot_sc_tree'>Plot single-cell smoothed expression magnitude heatmap</h2><span id='topic+plot_sc_tree'></span>

<h3>Description</h3>

<p>Plot single-cell smoothed expression magnitude heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_sc_tree(
  gtree,
  label_mut = TRUE,
  label_size = 3,
  dot_size = 2,
  branch_width = 0.5,
  tip = TRUE,
  tip_length = 0.5,
  pal_clone = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_sc_tree_+3A_gtree">gtree</code></td>
<td>
<p>tbl_graph The single-cell phylogeny</p>
</td></tr>
<tr><td><code id="plot_sc_tree_+3A_label_mut">label_mut</code></td>
<td>
<p>logical Whether to label mutations</p>
</td></tr>
<tr><td><code id="plot_sc_tree_+3A_label_size">label_size</code></td>
<td>
<p>numeric Size of mutation labels</p>
</td></tr>
<tr><td><code id="plot_sc_tree_+3A_dot_size">dot_size</code></td>
<td>
<p>numeric Size of mutation nodes</p>
</td></tr>
<tr><td><code id="plot_sc_tree_+3A_branch_width">branch_width</code></td>
<td>
<p>numeric Width of branches in tree</p>
</td></tr>
<tr><td><code id="plot_sc_tree_+3A_tip">tip</code></td>
<td>
<p>logical Whether to plot tip point</p>
</td></tr>
<tr><td><code id="plot_sc_tree_+3A_tip_length">tip_length</code></td>
<td>
<p>numeric Length of the tips</p>
</td></tr>
<tr><td><code id="plot_sc_tree_+3A_pal_clone">pal_clone</code></td>
<td>
<p>named vector Clone colors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot A single-cell phylogeny with mutation history labeled
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = plot_sc_tree(phylogeny_example)
</code></pre>

<hr>
<h2 id='pnorm.range.log'>Get the total probability from a region of a normal pdf</h2><span id='topic+pnorm.range.log'></span>

<h3>Description</h3>

<p>Get the total probability from a region of a normal pdf
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pnorm.range.log(lower, upper, mu, sd)
</code></pre>

<hr>
<h2 id='pre_likelihood_hmm'>HMM object for unit tests</h2><span id='topic+pre_likelihood_hmm'></span>

<h3>Description</h3>

<p>HMM object for unit tests
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pre_likelihood_hmm
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 10.
</p>

<hr>
<h2 id='preprocess_allele'>Preprocess allele data</h2><span id='topic+preprocess_allele'></span>

<h3>Description</h3>

<p>Preprocess allele data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preprocess_allele(sample, vcf_pu, vcf_phased, AD, DP, barcodes, gtf, gmap)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preprocess_allele_+3A_sample">sample</code></td>
<td>
<p>character Sample label</p>
</td></tr>
<tr><td><code id="preprocess_allele_+3A_vcf_pu">vcf_pu</code></td>
<td>
<p>dataframe Pileup VCF from cell-snp-lite</p>
</td></tr>
<tr><td><code id="preprocess_allele_+3A_vcf_phased">vcf_phased</code></td>
<td>
<p>dataframe Phased VCF from eagle2</p>
</td></tr>
<tr><td><code id="preprocess_allele_+3A_ad">AD</code></td>
<td>
<p>dgTMatrix Alt allele depth matrix from pileup</p>
</td></tr>
<tr><td><code id="preprocess_allele_+3A_dp">DP</code></td>
<td>
<p>dgTMatrix Total allele depth matrix from pileup</p>
</td></tr>
<tr><td><code id="preprocess_allele_+3A_barcodes">barcodes</code></td>
<td>
<p>vector List of barcodes from pileup</p>
</td></tr>
<tr><td><code id="preprocess_allele_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript GTF</p>
</td></tr>
<tr><td><code id="preprocess_allele_+3A_gmap">gmap</code></td>
<td>
<p>dataframe Genetic map</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Allele counts by cell
</p>

<hr>
<h2 id='ref_hca'>reference expression magnitudes from HCA</h2><span id='topic+ref_hca'></span>

<h3>Description</h3>

<p>reference expression magnitudes from HCA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ref_hca
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 24756 rows and 12 columns.
</p>

<hr>
<h2 id='ref_hca_counts'>reference expression counts from HCA</h2><span id='topic+ref_hca_counts'></span>

<h3>Description</h3>

<p>reference expression counts from HCA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ref_hca_counts
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 24857 rows and 12 columns.
</p>

<hr>
<h2 id='relevel_chrom'>Relevel chromosome column</h2><span id='topic+relevel_chrom'></span>

<h3>Description</h3>

<p>Relevel chromosome column
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relevel_chrom(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relevel_chrom_+3A_df">df</code></td>
<td>
<p>dataframe Dataframe with chromosome column</p>
</td></tr>
</table>

<hr>
<h2 id='resolve_cnvs'>Get unique CNVs from set of segments</h2><span id='topic+resolve_cnvs'></span>

<h3>Description</h3>

<p>Get unique CNVs from set of segments
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resolve_cnvs(segs_all, min_overlap = 0.5, debug = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="resolve_cnvs_+3A_segs_all">segs_all</code></td>
<td>
<p>dataframe CNV segments from multiple samples</p>
</td></tr>
<tr><td><code id="resolve_cnvs_+3A_min_overlap">min_overlap</code></td>
<td>
<p>numeric scalar Minimum overlap fraction to determine count two events as as overlapping</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Consensus CNV segments
</p>

<hr>
<h2 id='retest_bulks'>retest consensus segments on pseudobulks</h2><span id='topic+retest_bulks'></span>

<h3>Description</h3>

<p>retest consensus segments on pseudobulks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retest_bulks(
  bulks,
  segs_consensus = NULL,
  t = 1e-05,
  min_genes = 10,
  gamma = 20,
  nu = 1,
  use_loh = FALSE,
  diploid_chroms = NULL,
  ncores = 1,
  exclude_neu = TRUE,
  min_LLR = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="retest_bulks_+3A_bulks">bulks</code></td>
<td>
<p>dataframe Pseudobulk profiles</p>
</td></tr>
<tr><td><code id="retest_bulks_+3A_segs_consensus">segs_consensus</code></td>
<td>
<p>dataframe Consensus segments</p>
</td></tr>
<tr><td><code id="retest_bulks_+3A_use_loh">use_loh</code></td>
<td>
<p>logical Whether to use loh in the baseline</p>
</td></tr>
<tr><td><code id="retest_bulks_+3A_diploid_chroms">diploid_chroms</code></td>
<td>
<p>vector User-provided diploid chromosomes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Retested pseudobulks
</p>

<hr>
<h2 id='retest_cnv'>retest CNVs in a pseudobulk</h2><span id='topic+retest_cnv'></span>

<h3>Description</h3>

<p>retest CNVs in a pseudobulk
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retest_cnv(
  bulk,
  theta_min = 0.08,
  logphi_min = 0.25,
  gamma = 20,
  allele_only = FALSE,
  exclude_neu = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="retest_cnv_+3A_bulk">bulk</code></td>
<td>
<p>pesudobulk dataframe</p>
</td></tr>
<tr><td><code id="retest_cnv_+3A_gamma">gamma</code></td>
<td>
<p>numeric Dispersion parameter for the Beta-Binomial allele model</p>
</td></tr>
<tr><td><code id="retest_cnv_+3A_allele_only">allele_only</code></td>
<td>
<p>whether to retest only using allele data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of segments with CNV posterior information
</p>

<hr>
<h2 id='return_missing_columns'>Check the format of a given file</h2><span id='topic+return_missing_columns'></span>

<h3>Description</h3>

<p>Check the format of a given file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>return_missing_columns(file, expected_colnames = NULL)
</code></pre>

<hr>
<h2 id='run_group_hmms'>Run multiple HMMs</h2><span id='topic+run_group_hmms'></span>

<h3>Description</h3>

<p>Run multiple HMMs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_group_hmms(
  bulks,
  t = 1e-04,
  gamma = 20,
  alpha = 1e-04,
  min_genes = 10,
  nu = 1,
  common_diploid = TRUE,
  diploid_chroms = NULL,
  allele_only = FALSE,
  retest = TRUE,
  run_hmm = TRUE,
  exclude_neu = TRUE,
  ncores = 1,
  verbose = FALSE,
  debug = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_group_hmms_+3A_bulks">bulks</code></td>
<td>
<p>dataframe Pseudobulk profiles</p>
</td></tr>
<tr><td><code id="run_group_hmms_+3A_t">t</code></td>
<td>
<p>numeric Transition probability</p>
</td></tr>
<tr><td><code id="run_group_hmms_+3A_gamma">gamma</code></td>
<td>
<p>numeric Dispersion parameter for the Beta-Binomial allele model</p>
</td></tr>
<tr><td><code id="run_group_hmms_+3A_alpha">alpha</code></td>
<td>
<p>numeric P value cut-off to determine segment clusters in find_diploid</p>
</td></tr>
<tr><td><code id="run_group_hmms_+3A_common_diploid">common_diploid</code></td>
<td>
<p>logical Whether to find common diploid regions between pseudobulks</p>
</td></tr>
<tr><td><code id="run_group_hmms_+3A_diploid_chroms">diploid_chroms</code></td>
<td>
<p>character vector Known diploid chromosomes to use as baseline</p>
</td></tr>
<tr><td><code id="run_group_hmms_+3A_allele_only">allele_only</code></td>
<td>
<p>logical Whether only use allele data to run HMM</p>
</td></tr>
<tr><td><code id="run_group_hmms_+3A_retest">retest</code></td>
<td>
<p>logcial Whether to retest CNVs</p>
</td></tr>
<tr><td><code id="run_group_hmms_+3A_run_hmm">run_hmm</code></td>
<td>
<p>logical Whether to run HMM segments or just retest</p>
</td></tr>
<tr><td><code id="run_group_hmms_+3A_ncores">ncores</code></td>
<td>
<p>integer Number of cores</p>
</td></tr>
</table>

<hr>
<h2 id='run_numbat'>Run workflow to decompose tumor subclones</h2><span id='topic+run_numbat'></span>

<h3>Description</h3>

<p>Run workflow to decompose tumor subclones
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_numbat(
  count_mat,
  lambdas_ref,
  df_allele,
  genome = "hg38",
  out_dir = tempdir(),
  max_iter = 2,
  max_nni = 100,
  t = 1e-05,
  gamma = 20,
  min_LLR = 5,
  alpha = 1e-04,
  eps = 1e-05,
  max_entropy = 0.5,
  init_k = 3,
  min_cells = 50,
  tau = 0.3,
  nu = 1,
  max_cost = ncol(count_mat) * tau,
  n_cut = 0,
  min_depth = 0,
  common_diploid = TRUE,
  min_overlap = 0.45,
  ncores = 1,
  ncores_nni = ncores,
  random_init = FALSE,
  segs_loh = NULL,
  call_clonal_loh = FALSE,
  verbose = TRUE,
  diploid_chroms = NULL,
  segs_consensus_fix = NULL,
  use_loh = NULL,
  min_genes = 10,
  skip_nj = FALSE,
  multi_allelic = TRUE,
  p_multi = 1 - alpha,
  plot = TRUE,
  check_convergence = FALSE,
  exclude_neu = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_numbat_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix Raw count matrices where rownames are genes and column names are cells</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Either a named vector with gene names as names and normalized expression as values, or a matrix where rownames are genes and columns are pseudobulk names</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_df_allele">df_allele</code></td>
<td>
<p>dataframe Allele counts per cell, produced by preprocess_allele</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_genome">genome</code></td>
<td>
<p>character Genome version (hg38, hg19, or mm10)</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_out_dir">out_dir</code></td>
<td>
<p>string Output directory</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_max_iter">max_iter</code></td>
<td>
<p>integer Maximum number of iterations to run the phyologeny optimization</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_max_nni">max_nni</code></td>
<td>
<p>integer Maximum number of iterations to run NNI in the ML phylogeny inference</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_t">t</code></td>
<td>
<p>numeric Transition probability</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_gamma">gamma</code></td>
<td>
<p>numeric Dispersion parameter for the Beta-Binomial allele model</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_min_llr">min_LLR</code></td>
<td>
<p>numeric Minimum LLR to filter CNVs</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_alpha">alpha</code></td>
<td>
<p>numeric P value cutoff for diploid finding</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_eps">eps</code></td>
<td>
<p>numeric Convergence threshold for ML tree search</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_max_entropy">max_entropy</code></td>
<td>
<p>numeric Entropy threshold to filter CNVs</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_init_k">init_k</code></td>
<td>
<p>integer Number of clusters in the initial clustering</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_min_cells">min_cells</code></td>
<td>
<p>integer Minimum number of cells to run HMM on</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_tau">tau</code></td>
<td>
<p>numeric Factor to determine max_cost as a function of the number of cells (0-1)</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_nu">nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_max_cost">max_cost</code></td>
<td>
<p>numeric Likelihood threshold to collapse internal branches</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_n_cut">n_cut</code></td>
<td>
<p>integer Number of cuts on the phylogeny to define subclones</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_min_depth">min_depth</code></td>
<td>
<p>integer Minimum allele depth</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_common_diploid">common_diploid</code></td>
<td>
<p>logical Whether to find common diploid regions in a group of peusdobulks</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_min_overlap">min_overlap</code></td>
<td>
<p>numeric Minimum CNV overlap threshold</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_ncores">ncores</code></td>
<td>
<p>integer Number of threads to use</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_ncores_nni">ncores_nni</code></td>
<td>
<p>integer Number of threads to use for NNI</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_random_init">random_init</code></td>
<td>
<p>logical Whether to initiate phylogney using a random tree (internal use only)</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_segs_loh">segs_loh</code></td>
<td>
<p>dataframe Segments of clonal LOH to be excluded</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_call_clonal_loh">call_clonal_loh</code></td>
<td>
<p>logical Whether to call segments with clonal LOH</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_verbose">verbose</code></td>
<td>
<p>logical Verbosity</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_diploid_chroms">diploid_chroms</code></td>
<td>
<p>vector Known diploid chromosomes</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_segs_consensus_fix">segs_consensus_fix</code></td>
<td>
<p>dataframe Pre-determined segmentation of consensus CNVs</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_use_loh">use_loh</code></td>
<td>
<p>logical Whether to include LOH regions in the expression baseline</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_min_genes">min_genes</code></td>
<td>
<p>integer Minimum number of genes to call a segment</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_skip_nj">skip_nj</code></td>
<td>
<p>logical Whether to skip NJ tree construction and only use UPGMA</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_multi_allelic">multi_allelic</code></td>
<td>
<p>logical Whether to call multi-allelic CNVs</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_p_multi">p_multi</code></td>
<td>
<p>numeric P value cutoff for calling multi-allelic CNVs</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_plot">plot</code></td>
<td>
<p>logical Whether to plot results</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_check_convergence">check_convergence</code></td>
<td>
<p>logical Whether to terminate iterations based on consensus CNV convergence</p>
</td></tr>
<tr><td><code id="run_numbat_+3A_exclude_neu">exclude_neu</code></td>
<td>
<p>logical Whether to exclude neutral segments from CNV retesting (internal use only)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a status code
</p>

<hr>
<h2 id='segs_example'>example CNV segments dataframe</h2><span id='topic+segs_example'></span>

<h3>Description</h3>

<p>example CNV segments dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>segs_example
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 27 rows and 30 columns.
</p>

<hr>
<h2 id='simes_p'>Calculate simes' p</h2><span id='topic+simes_p'></span>

<h3>Description</h3>

<p>Calculate simes' p
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simes_p(p.vals, n_dim)
</code></pre>

<hr>
<h2 id='simplify_history'>Simplify the mutational history based on likelihood evidence</h2><span id='topic+simplify_history'></span>

<h3>Description</h3>

<p>Simplify the mutational history based on likelihood evidence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simplify_history(G, l_matrix, max_cost = 150, n_cut = 0, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simplify_history_+3A_g">G</code></td>
<td>
<p>igraph Mutation graph</p>
</td></tr>
<tr><td><code id="simplify_history_+3A_l_matrix">l_matrix</code></td>
<td>
<p>matrix Mutation placement likelihood matrix (node by mutation)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>igraph Mutation graph
</p>

<hr>
<h2 id='smooth_expression'>filtering, normalization and capping</h2><span id='topic+smooth_expression'></span>

<h3>Description</h3>

<p>filtering, normalization and capping
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smooth_expression(count_mat, lambdas_ref, gtf, window = 101, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smooth_expression_+3A_count_mat">count_mat</code></td>
<td>
<p>dgCMatrix Gene expression counts</p>
</td></tr>
<tr><td><code id="smooth_expression_+3A_lambdas_ref">lambdas_ref</code></td>
<td>
<p>matrix Reference expression profiles</p>
</td></tr>
<tr><td><code id="smooth_expression_+3A_gtf">gtf</code></td>
<td>
<p>dataframe Transcript gtf</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Log(x+1) transformed normalized expression values for single cells
</p>

<hr>
<h2 id='smooth_segs'>Smooth the segments after HMM decoding</h2><span id='topic+smooth_segs'></span>

<h3>Description</h3>

<p>Smooth the segments after HMM decoding
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smooth_segs(bulk, min_genes = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smooth_segs_+3A_bulk">bulk</code></td>
<td>
<p>dataframe Pseudobulk profile</p>
</td></tr>
<tr><td><code id="smooth_segs_+3A_min_genes">min_genes</code></td>
<td>
<p>integer Minimum number of genes to call a segment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Pseudobulk profile
</p>

<hr>
<h2 id='switch_prob_cm'>predict phase switch probablity as a function of genetic distance</h2><span id='topic+switch_prob_cm'></span>

<h3>Description</h3>

<p>predict phase switch probablity as a function of genetic distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>switch_prob_cm(d, nu = 1, min_p = 1e-10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="switch_prob_cm_+3A_d">d</code></td>
<td>
<p>numeric vector Genetic distance in cM</p>
</td></tr>
<tr><td><code id="switch_prob_cm_+3A_nu">nu</code></td>
<td>
<p>numeric Phase switch rate</p>
</td></tr>
<tr><td><code id="switch_prob_cm_+3A_min_p">min_p</code></td>
<td>
<p>numeric Minimum phase switch probability</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector Phase switch probability
</p>

<hr>
<h2 id='t_test_pval'>T-test wrapper, handles error for insufficient observations</h2><span id='topic+t_test_pval'></span>

<h3>Description</h3>

<p>T-test wrapper, handles error for insufficient observations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>t_test_pval(x, y)
</code></pre>

<hr>
<h2 id='test_multi_allelic'>test for multi-allelic CNVs</h2><span id='topic+test_multi_allelic'></span>

<h3>Description</h3>

<p>test for multi-allelic CNVs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_multi_allelic(bulks, segs_consensus, min_LLR = 5, p_min = 0.999)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="test_multi_allelic_+3A_bulks">bulks</code></td>
<td>
<p>dataframe Pseudobulk profiles</p>
</td></tr>
<tr><td><code id="test_multi_allelic_+3A_segs_consensus">segs_consensus</code></td>
<td>
<p>dataframe Consensus segments</p>
</td></tr>
<tr><td><code id="test_multi_allelic_+3A_min_llr">min_LLR</code></td>
<td>
<p>numeric CNV LLR threshold to filter events</p>
</td></tr>
<tr><td><code id="test_multi_allelic_+3A_p_min">p_min</code></td>
<td>
<p>numeric Probability threshold to call multi-allelic events</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe Consensus segments annotated with multi-allelic events
</p>

<hr>
<h2 id='theta_hat_roll'>Rolling estimate of imbalance level theta</h2><span id='topic+theta_hat_roll'></span>

<h3>Description</h3>

<p>Rolling estimate of imbalance level theta
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theta_hat_roll(major_count, minor_count, h)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="theta_hat_roll_+3A_major_count">major_count</code></td>
<td>
<p>vector of major allele count</p>
</td></tr>
<tr><td><code id="theta_hat_roll_+3A_minor_count">minor_count</code></td>
<td>
<p>vector of minor allele count</p>
</td></tr>
<tr><td><code id="theta_hat_roll_+3A_h">h</code></td>
<td>
<p>window size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>rolling estimate of theta
</p>

<hr>
<h2 id='theta_hat_seg'>Estimate of imbalance level theta in a segment</h2><span id='topic+theta_hat_seg'></span>

<h3>Description</h3>

<p>Estimate of imbalance level theta in a segment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theta_hat_seg(major_count, minor_count)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="theta_hat_seg_+3A_major_count">major_count</code></td>
<td>
<p>vector of major allele count</p>
</td></tr>
<tr><td><code id="theta_hat_seg_+3A_minor_count">minor_count</code></td>
<td>
<p>vector of minor allele count</p>
</td></tr>
</table>


<h3>Value</h3>

<p>estimate of theta
</p>

<hr>
<h2 id='transfer_links'>Annotate the direct upstream or downstream node on the edges</h2><span id='topic+transfer_links'></span>

<h3>Description</h3>

<p>Annotate the direct upstream or downstream node on the edges
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transfer_links(G)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transfer_links_+3A_g">G</code></td>
<td>
<p>igraph Mutation graph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>igraph Mutation graph
</p>

<hr>
<h2 id='upgma'>UPGMA and WPGMA clustering</h2><span id='topic+upgma'></span>

<h3>Description</h3>

<p>UPGMA and WPGMA clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>upgma(D, method = "average", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="upgma_+3A_d">D</code></td>
<td>
<p>A distance matrix.</p>
</td></tr>
<tr><td><code id="upgma_+3A_method">method</code></td>
<td>
<p>The agglomeration method to be used. This should be (an
unambiguous abbreviation of) one of &quot;ward&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;average&quot;,
&quot;mcquitty&quot;, &quot;median&quot; or &quot;centroid&quot;. The default is &quot;average&quot;.</p>
</td></tr>
<tr><td><code id="upgma_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>

<hr>
<h2 id='vcf_meta'>example VCF header</h2><span id='topic+vcf_meta'></span>

<h3>Description</h3>

<p>example VCF header
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf_meta
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 65.
</p>

<hr>
<h2 id='viterbi_loh'>Viterbi for clonal LOH detection</h2><span id='topic+viterbi_loh'></span>

<h3>Description</h3>

<p>Viterbi for clonal LOH detection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>viterbi_loh(hmm, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="viterbi_loh_+3A_hmm">hmm</code></td>
<td>
<p>HMM object; expect variables x (SNP count), snp_sig (snp rate standard deviation), 
pm (snp density for ref and loh states), pn (gene lengths),
d (total expression depth), y (expression count), lambda_star (reference expression rate), 
mu (global expression mean), sig (global expression standard deviation),
Pi (transition prob matrix), delta (prior for each state),
phi (expression fold change for each state)</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
