<!DOCTYPE html><html><head><title>Help for package occumb</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {occumb}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#eval_util_L'><p>Expected utility for local species diversity assessments.</p></a></li>
<li><a href='#eval_util_R'><p>Expected utility for regional species diversity assessments.</p></a></li>
<li><a href='#fish'><p>Fish eDNA metabarcoding dataset</p></a></li>
<li><a href='#fish_raw'><p>Fish eDNA metabarcoding dataset</p></a></li>
<li><a href='#get_posterior'><p>Extract posterior samples or summary of parameters from a model-fit object.</p></a></li>
<li><a href='#gof'><p>Goodness-of-fit assessment of the fitted model.</p></a></li>
<li><a href='#list_cond_L'><p>Conditions for local assessment under certain budget and cost values.</p></a></li>
<li><a href='#list_cond_R'><p>Conditions for regional assessment under certain budget and cost values.</p></a></li>
<li><a href='#occumb'><p>Model-fitting function.</p></a></li>
<li><a href='#occumbData'><p>Constructor for occumbData data class.</p></a></li>
<li><a href='#plot+2CoccumbFit-method'><p>Plot method for occumbFit class.</p></a></li>
<li><a href='#plot+2CoccumbGof-method'><p>Plot method for occumbGof class.</p></a></li>
<li><a href='#predict+2CoccumbFit-method'><p>Predict method for occumbFit class.</p></a></li>
<li><a href='#summary+2CoccumbData-method'><p>Summary method for occumbData class.</p></a></li>
<li><a href='#summary+2CoccumbFit-method'><p>Summary method for occumbFit class.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Site Occupancy Modeling for Environmental DNA Metabarcoding</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Fits multispecies site occupancy models to environmental DNA
    metabarcoding data collected using spatially-replicated survey design.
    Model fitting results can be used to evaluate and compare the effectiveness
    of species detection to find an efficient survey design.
    Reference: Fukaya et al. (2022) &lt;<a href="https://doi.org/10.1111%2F2041-210X.13732">doi:10.1111/2041-210X.13732</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, spelling, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, methods, graphics, checkmate, crayon, knitr</td>
</tr>
<tr>
<td>Collate:</td>
<td>'classes.R' 'fish.R' 'fish_raw.R' 'get_posterior.R' 'gof.R'
'occumbData.R' 'occumb.R' 'plot.R' 'predict.R' 'show.R'
'summary.R' 'utility.R' 'utils.R'</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://fukayak.github.io/occumb/">https://fukayak.github.io/occumb/</a>,
<a href="https://github.com/fukayak/occumb">https://github.com/fukayak/occumb</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>jagsUI, R (&ge; 2.10)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>JAGS (http://mcmc-jags.sourceforge.net)</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/fukayak/occumb/issues">https://github.com/fukayak/occumb/issues</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-26 00:58:36 UTC; kfukaya</td>
</tr>
<tr>
<td>Author:</td>
<td>Keiichi Fukaya [aut, cre],
  Ken Kellner [cph] (summary method for occumbFit class)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Keiichi Fukaya &lt;fukayak99@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-26 06:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='eval_util_L'>Expected utility for local species diversity assessments.</h2><span id='topic+eval_util_L'></span>

<h3>Description</h3>

<p><code>eval_util_L()</code> evaluates the expected utility of a local
species diversity assessment by using Monte Carlo integration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eval_util_L(
  settings,
  fit = NULL,
  z = NULL,
  theta = NULL,
  phi = NULL,
  N_rep = 1,
  cores = 1L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eval_util_L_+3A_settings">settings</code></td>
<td>
<p>A data frame that specifies a set of conditions under which
utility is evaluated. It must include columns named <code>K</code> and <code>N</code>, which
specify the number of replicates per site and the sequencing depth per
replicate, respectively.
<code>K</code> and <code>N</code> must be numeric vectors greater than 0. When <code>K</code> contains a
decimal value, it is discarded and treated as an integer.
Additional columns are ignored, but may be included.</p>
</td></tr>
<tr><td><code id="eval_util_L_+3A_fit">fit</code></td>
<td>
<p>An <code>occumbFit</code> object.</p>
</td></tr>
<tr><td><code id="eval_util_L_+3A_z">z</code></td>
<td>
<p>Sample values of site occupancy status of species stored in an array
with sample <code class="reqn">\times</code> species <code class="reqn">\times</code> site dimensions.</p>
</td></tr>
<tr><td><code id="eval_util_L_+3A_theta">theta</code></td>
<td>
<p>Sample values of sequence capture probabilities of species
stored in a matrix with sample <code class="reqn">\times</code> species dimensions or an array
with sample <code class="reqn">\times</code> species <code class="reqn">\times</code> site dimensions.</p>
</td></tr>
<tr><td><code id="eval_util_L_+3A_phi">phi</code></td>
<td>
<p>Sample values of sequence relative dominance of species stored in
a matrix with sample <code class="reqn">\times</code> species dimensions or an array with
sample <code class="reqn">\times</code> species <code class="reqn">\times</code> site dimensions.</p>
</td></tr>
<tr><td><code id="eval_util_L_+3A_n_rep">N_rep</code></td>
<td>
<p>Controls the sample size for the Monte Carlo integration.
The integral is evaluated using <code>N_sample * N_rep</code> random samples,
where <code>N_sample</code> is the maximum size of the MCMC sample in the <code>fit</code>
argument and the parameter sample in the <code>z</code>, <code>theta</code>, and <code>phi</code> arguments.</p>
</td></tr>
<tr><td><code id="eval_util_L_+3A_cores">cores</code></td>
<td>
<p>The number of cores to use for parallelization.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The utility of local species diversity assessment for a given set of sites
can be defined as the expected number of detected species per site
(Fukaya et al. 2022). <code>eval_util_L()</code> evaluates this utility for arbitrary
sets of sites that can potentially have different values for site occupancy
status of species, <code class="reqn">z</code>, sequence capture probabilities of species,
<code class="reqn">\theta</code>, and sequence relative dominance of species,
<code class="reqn">\phi</code>, for the combination of <code>K</code> and <code>N</code> values specified in the
<code>conditions</code> argument.
Such evaluations can be used to balance <code>K</code> and <code>N</code> to maximize the utility
under a constant budget (possible combinations of <code>K</code> and <code>N</code> under a
specified budget and cost values are easily obtained using <code>list_cond_L()</code>;
see the example below).
It is also possible to examine how the utility varies with different <code>K</code>
and <code>N</code> values without setting a budget level, which may be useful for determining
a satisfactory level of <code>K</code> and <code>N</code> from a purely technical point of view.
</p>
<p>The expected utility is defined as the expected value of the conditional
utility in the form:
</p>
<p style="text-align: center;"><code class="reqn">U(K, N \mid \boldsymbol{r}, \boldsymbol{u}) = \frac{1}{J}\sum_{j = 1}^{J}\sum_{i = 1}^{I}\left\{1 - \prod_{k = 1}^{K}\left(1 - \frac{u_{ijk}r_{ijk}}{\sum_{m = 1}^{I}u_{mjk}r_{mjk}} \right)^N \right\}</code>
</p>

<p>where <code class="reqn">u_{ijk}</code> is a latent indicator variable representing
the inclusion of the sequence of species <code class="reqn">i</code> in replicate <code class="reqn">k</code>
at site <code class="reqn">j</code>, and <code class="reqn">r_{ijk}</code> is a latent variable that
is proportional to the relative frequency of the sequence of species
<code class="reqn">i</code>, conditional on its presence in replicate <code class="reqn">k</code> at site
<code class="reqn">j</code> (Fukaya et al. 2022).
Expectations are taken with respect to the posterior (or possibly prior)
predictive distributions of <code class="reqn">\boldsymbol{r} = \{r_{ijk}\}</code> and
<code class="reqn">\boldsymbol{u} = \{u_{ijk}\}</code>, which are evaluated numerically using
Monte Carlo integration. The predictive distributions of
<code class="reqn">\boldsymbol{r}</code> and <code class="reqn">\boldsymbol{u}</code> depend on the model
parameters <code class="reqn">z</code>, <code class="reqn">\theta</code>, and <code class="reqn">\phi</code> values.
Their posterior (or prior) distribution is specified by supplying an
<code>occumbFit</code> object containing their posterior samples via the <code>fit</code> argument,
or by supplying a matrix or array of posterior (or prior) samples of
parameter values via the <code>z</code>, <code>theta</code>, and <code>phi</code> arguments. Higher
approximation accuracy can be obtained by increasing the value of <code>N_rep</code>.
</p>
<p>The <code>eval_util_L()</code> function can be executed by supplying the <code>fit</code> argument
without specifying the <code>z</code>, <code>theta</code>, and <code>phi</code> arguments, by supplying the
three <code>z</code>, <code>theta</code>, and <code>phi</code> arguments without the <code>fit</code> argument, or by
supplying the <code>fit</code> argument and any or all of the <code>z</code>, <code>theta</code>, and <code>phi</code>
arguments. If <code>z</code>, <code>theta</code>, or <code>phi</code> arguments are specified in addition
to the <code>fit</code>, the parameter values given in these arguments are used
preferentially to evaluate the expected utility. If the sample sizes differ among
parameters, parameters with smaller sample sizes are resampled with
replacements to align the sample sizes across parameters.
</p>
<p>The expected utility is evaluated assuming homogeneity of replicates, in the
sense that <code class="reqn">\theta</code> and <code class="reqn">\phi</code>, the model parameters
associated with the species detection process, are constant across
replicates within a site. For this reason, <code>eval_util_L()</code> does not accept
replicate-specific <code class="reqn">\theta</code> and <code class="reqn">\phi</code>. If the
<code>occumbFit</code> object supplied in the <code>fit</code> argument has a replicate-specific
parameter, the parameter samples to be used in the utility evaluation must be
provided explicitly via the <code>theta</code> or <code>phi</code> arguments.
</p>
<p>The Monte Carlo integration is executed in parallel on multiple CPU cores, where
the <code>cores</code> argument controls the degree of parallelization.
</p>


<h3>Value</h3>

<p>A data frame with a column named <code>Utility</code> in which the estimates of the
expected utility are stored. This is obtained by adding the <code>Utility</code> column
to the data frame provided in the <code>settings</code> argument.
</p>


<h3>References</h3>

<p>K. Fukaya, N. I. Kondo, S. S. Matsuzaki and T. Kadoya (2022)
Multispecies site occupancy modelling and study design for spatially
replicated environmental DNA metabarcoding. <em>Methods in Ecology
and Evolution</em> <strong>13</strong>:183&ndash;193.
<a href="https://doi.org/10.1111/2041-210X.13732">doi:10.1111/2041-210X.13732</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(1)

# Generate a random dataset (20 species * 2 sites * 2 reps)
I &lt;- 20 # Number of species
J &lt;- 2  # Number of sites
K &lt;- 2  # Number of replicates
data &lt;- occumbData(
    y = array(sample.int(I * J * K), dim = c(I, J, K)))

# Fitting a null model
fit &lt;- occumb(data = data)

## Estimate expected utility
# Arbitrary K and N values
(util1 &lt;- eval_util_L(expand.grid(K = 1:3, N = c(1E3, 1E4, 1E5)),
                      fit))

# K and N values under specified budget and cost
(util2 &lt;- eval_util_L(list_cond_L(budget = 1E5,
                                  lambda1 = 0.01,
                                  lambda2 = 5000,
                                  fit),
                      fit))

# K values restricted
(util3 &lt;- eval_util_L(list_cond_L(budget = 1E5,
                                  lambda1 = 0.01,
                                  lambda2 = 5000,
                                  fit,
                                  K = 1:5),
                      fit))

# theta and phi values supplied
(util4 &lt;- eval_util_L(list_cond_L(budget = 1E5,
                                  lambda1 = 0.01,
                                  lambda2 = 5000,
                                  fit,
                                  K = 1:5),
                      fit,
                      theta = array(0.5, dim = c(4000, I, J)),
                      phi = array(1, dim = c(4000, I, J))))

# z, theta, and phi values, but no fit object supplied
(util5 &lt;- eval_util_L(list_cond_L(budget = 1E5,
                                  lambda1 = 0.01,
                                  lambda2 = 5000,
                                  fit,
                                  K = 1:5),
                      fit = NULL,
                      z = array(1, dim = c(4000, I, J)),
                      theta = array(0.5, dim = c(4000, I, J)),
                      phi = array(1, dim = c(4000, I, J))))

</code></pre>

<hr>
<h2 id='eval_util_R'>Expected utility for regional species diversity assessments.</h2><span id='topic+eval_util_R'></span>

<h3>Description</h3>

<p><code>eval_util_R()</code> evaluates the expected utility of a regional
species diversity assessment using Monte Carlo integration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eval_util_R(
  settings,
  fit = NULL,
  psi = NULL,
  theta = NULL,
  phi = NULL,
  N_rep = 1,
  cores = 1L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eval_util_R_+3A_settings">settings</code></td>
<td>
<p>A data frame that specifies a set of conditions under which
utility is evaluated. It must include columns named <code>J</code>, <code>K</code>, and <code>N</code>,
which specify the number of sites, number of replicates per site, and
sequencing depth per replicate, respectively.
<code>J</code>, <code>K</code>, and <code>N</code> must be numeric vectors greater than 0. When <code>J</code> and <code>K</code>
contain decimal values, they are discarded and treated as integers.
Additional columns are ignored, but may be included.</p>
</td></tr>
<tr><td><code id="eval_util_R_+3A_fit">fit</code></td>
<td>
<p>An <code>occumbFit</code> object.</p>
</td></tr>
<tr><td><code id="eval_util_R_+3A_psi">psi</code></td>
<td>
<p>Sample values of the site occupancy probabilities of species
stored in a matrix with sample <code class="reqn">\times</code> species dimensions or an array
with sample <code class="reqn">\times</code> species <code class="reqn">\times</code> site dimensions.</p>
</td></tr>
<tr><td><code id="eval_util_R_+3A_theta">theta</code></td>
<td>
<p>Sample values of sequence capture probabilities of species
stored in a matrix with sample <code class="reqn">\times</code> species dimensions or an array
with sample <code class="reqn">\times</code> species <code class="reqn">\times</code> site dimensions.</p>
</td></tr>
<tr><td><code id="eval_util_R_+3A_phi">phi</code></td>
<td>
<p>Sample values of sequence relative dominance of species stored in
a matrix with sample <code class="reqn">\times</code> species dimensions or an array with
sample <code class="reqn">\times</code> species <code class="reqn">\times</code> site dimensions.</p>
</td></tr>
<tr><td><code id="eval_util_R_+3A_n_rep">N_rep</code></td>
<td>
<p>Controls the sample size for the Monte Carlo integration.
The integral is evaluated using a total of <code>N_sample * N_rep</code> random samples,
where <code>N_sample</code> is the maximum size of the MCMC sample in the <code>fit</code>
argument and the parameter sample in the <code>psi</code>, <code>theta</code>, and <code>phi</code> arguments.</p>
</td></tr>
<tr><td><code id="eval_util_R_+3A_cores">cores</code></td>
<td>
<p>The number of cores to use for parallelization.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The utility of a regional species diversity assessment can be defined as
the number of species expected to be detected in the region of interest
(Fukaya et al. 2022). <code>eval_util_R()</code> evaluates this utility for the region
modeled in the <code>occumbFit</code> object for the combination of <code>J</code>, <code>K</code>, and <code>N</code>
values specified in the <code>conditions</code> argument.
Such evaluations can be used to balance <code>J</code>, <code>K</code>, and <code>N</code> to maximize the
utility under a constant budget (possible combinations of <code>J</code>, <code>K</code>, and <code>N</code>
under a specified budget and cost values are easily obtained using
<code>list_cond_R()</code>; see the example below).
It is also possible to examine how the utility varies with different <code>J</code>,
<code>K</code>, and <code>N</code> values without setting a budget level, which may be useful in determining
the satisfactory levels of <code>J</code>, <code>K</code>, and <code>N</code> from a purely technical point of
view.
</p>
<p>The expected utility is defined as the expected value of the conditional
utility in the form:
</p>
<p style="text-align: center;"><code class="reqn">U(J, K, N \mid \boldsymbol{r}, \boldsymbol{u}) = \sum_{i = 1}^{I}\left\{1 - \prod_{j = 1}^{J}\prod_{k = 1}^{K}\left(1 - \frac{u_{ijk}r_{ijk}}{\sum_{m = 1}^{I}u_{mjk}r_{mjk}} \right)^N \right\}</code>
</p>

<p>where <code class="reqn">u_{ijk}</code> is a latent indicator variable representing
the inclusion of the sequence of species <code class="reqn">i</code> in replicate <code class="reqn">k</code>
at site <code class="reqn">j</code>, and <code class="reqn">r_{ijk}</code> is a latent variable that
is proportional to the relative frequency of the sequence of species
<code class="reqn">i</code>, conditional on its presence in replicate <code class="reqn">k</code> at site
<code class="reqn">j</code> (Fukaya et al. 2022).
Expectations are taken with respect to the posterior (or possibly prior)
predictive distributions of <code class="reqn">\boldsymbol{r} = \{r_{ijk}\}</code> and
<code class="reqn">\boldsymbol{u} = \{u_{ijk}\}</code>, which are evaluated numerically using
Monte Carlo integration. The predictive distributions of
<code class="reqn">\boldsymbol{r}</code> and <code class="reqn">\boldsymbol{u}</code> depend on the model
parameters <code class="reqn">\psi</code>, <code class="reqn">\theta</code>, and <code class="reqn">\phi</code> values.
Their posterior (or prior) distribution is specified by supplying an
<code>occumbFit</code> object containing their posterior samples via the <code>fit</code> argument,
or by supplying a matrix or array of posterior (or prior) samples of
parameter values via the <code>psi</code>, <code>theta</code>, and <code>phi</code> arguments. Higher
approximation accuracy can be obtained by increasing the value of <code>N_rep</code>.
</p>
<p>The <code>eval_util_R()</code> function can be executed by supplying the <code>fit</code> argument
without specifying the <code>psi</code>, <code>theta</code>, and <code>phi</code> arguments, by supplying the
three <code>psi</code>, <code>theta</code>, and <code>phi</code> arguments without the <code>fit</code> argument, or by
supplying the <code>fit</code> argument and any or all of the <code>psi</code>, <code>theta</code>, and <code>phi</code>
arguments. If the <code>psi</code>, <code>theta</code>, or <code>phi</code> arguments are specified in addition
to the <code>fit</code>, the parameter values given in these arguments are preferentially
used to evaluate the expected utility. If the sample sizes differed among
parameters, parameters with smaller sample sizes are resampled with
replacements to align the sample sizes across parameters.
</p>
<p>The expected utility is evaluated assuming homogeneity of replicates, in the
sense that <code class="reqn">\theta</code> and <code class="reqn">\phi</code>, the model parameters
associated with the species detection process, are constant across
replicates within a site. For this reason, <code>eval_util_R()</code> does not accept
replicate-specific <code class="reqn">\theta</code> and <code class="reqn">\phi</code>. If the
<code>occumbFit</code> object supplied in the <code>fit</code> argument has a replicate-specific
parameter, the parameter samples to be used in the utility evaluation must be
provided explicitly via the <code>theta</code> or <code>phi</code> arguments.
</p>
<p>If the parameters are modeled as a function of site covariates in the <code>fit</code>
object, or if the <code>psi</code>, <code>theta</code>, and/or <code>phi</code> arguments have site dimensions,
the expected utility is evaluated to account for the site heterogeneity of
the parameters. To incorporate site heterogeneity, the
parameter values for each <code>J</code> site are determined by selecting site-specific
parameter values in the <code>fit</code>, or those supplied in <code>psi</code>, <code>theta</code>, and <code>phi</code>
via random sampling with replacement. Thus, expected utility is
evaluated by assuming a set of supplied parameter values as a statistical
population of site-specific parameters.
</p>
<p>The Monte Carlo integration is executed in parallel on multiple CPU cores, where
the <code>cores</code> argument controls the degree of parallelization.
</p>


<h3>Value</h3>

<p>A data frame with a column named <code>Utility</code> in which the estimates of
the expected utility are stored. This is obtained by adding the <code>Utility</code> column
to the data frame provided in the <code>settings</code> argument.
</p>


<h3>References</h3>

<p>K. Fukaya, N. I. Kondo, S. S. Matsuzaki and T. Kadoya (2022)
Multispecies site occupancy modelling and study design for spatially
replicated environmental DNA metabarcoding. <em>Methods in Ecology
and Evolution</em> <strong>13</strong>:183&ndash;193.
<a href="https://doi.org/10.1111/2041-210X.13732">doi:10.1111/2041-210X.13732</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(1)

# Generate a random dataset (20 species * 2 sites * 2 reps)
I &lt;- 20 # Number of species
J &lt;- 2  # Number of sites
K &lt;- 2  # Number of replicates
data &lt;- occumbData(
    y = array(sample.int(I * J * K), dim = c(I, J, K)))

# Fitting a null model
fit &lt;- occumb(data = data)

## Estimate expected utility
# Arbitrary J, K, and N values
(util1 &lt;- eval_util_R(expand.grid(J = 1:3, K = 1:3, N = c(1E3, 1E4, 1E5)),
                      fit))

# J, K, and N values under specified budget and cost
(util2 &lt;- eval_util_R(list_cond_R(budget = 50000,
                                  lambda1 = 0.01,
                                  lambda2 = 5000,
                                  lambda3 = 5000),
                      fit))

# K values restricted
(util3 &lt;- eval_util_R(list_cond_R(budget = 50000,
                                  lambda1 = 0.01,
                                  lambda2 = 5000,
                                  lambda3 = 5000,
                                  K = 1:5),
                      fit))

# J and K values restricted
(util4 &lt;- eval_util_R(list_cond_R(budget = 50000,
                                  lambda1 = 0.01,
                                  lambda2 = 5000,
                                  lambda3 = 5000,
                                  J = 1:3, K = 1:5),
                      fit))

# theta and phi values supplied
(util5 &lt;- eval_util_R(list_cond_R(budget = 50000,
                                  lambda1 = 0.01,
                                  lambda2 = 5000,
                                  lambda3 = 5000,
                                  J = 1:3, K = 1:5),
                      fit,
                      theta = array(0.5, dim = c(4000, I, J)),
                      phi = array(1, dim = c(4000, I, J))))

# psi, theta, and phi values, but no fit object supplied
(util6 &lt;- eval_util_R(list_cond_R(budget = 50000,
                                  lambda1 = 0.01,
                                  lambda2 = 5000,
                                  lambda3 = 5000,
                                  J = 1:3, K = 1:5),
                      fit = NULL,
                      psi = array(0.9, dim = c(4000, I, J)),
                      theta = array(0.9, dim = c(4000, I, J)),
                      phi = array(1, dim = c(4000, I, J))))

</code></pre>

<hr>
<h2 id='fish'>Fish eDNA metabarcoding dataset</h2><span id='topic+fish'></span>

<h3>Description</h3>

<p>A dataset of fish eDNA metabarcoding collected in the Kasumigaura watershed, Japan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fish
</code></pre>


<h3>Format</h3>

<p>An occumbData class object containing the sequence read count <code>y</code>,
a species covariate <code>mismatch</code>, and a site covariate <code>riverbank</code>.
<code>mismatch</code> represents the total number of mismatched bases in the
priming region of the forward and reverse primers for each species.
<code>riverbank</code> indicates whether the riverbank at each site lacks aquatic
and riparian vegetation.
Sequence reads were obtained from three replicates (collected from the
center of the river and near the left and right riverbanks)
from 50 sites across the watershed, of which read counts from six samples were
missing.
The resulting sequence counts of 50 freshwater fish taxa were recorded.
</p>


<h3>Source</h3>

<p>K. Fukaya, N. I. Kondo, S. S. Matsuzaki, T. Kadoya (2021) Data from: Multispecies site occupancy modeling and study design for spatially replicated environmental DNA metabarcoding. Dryad Digital Repository. <a href="https://doi.org/10.5061/dryad.3bk3j9kkm">doi:10.5061/dryad.3bk3j9kkm</a>
</p>

<hr>
<h2 id='fish_raw'>Fish eDNA metabarcoding dataset</h2><span id='topic+fish_raw'></span>

<h3>Description</h3>

<p>A dataset of fish eDNA metabarcoding collected in the Kasumigaura watershed, Japan.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fish_raw
</code></pre>


<h3>Format</h3>

<p>A list containing an array of sequence read count, <code>y</code>,
a vector of the total number of mismatched bases in the priming region of
the forward and reverse primers for each species, <code>mismatch</code>,
and a factor indicating whether the riverbank at each site lacked aquatic
and riparian vegetation, <code>riverbank</code>.
Sequence reads were obtained from three replicates (collected from the
center of the river and near the left and right riverbanks)
from 50 sites across the watershed, of which read counts from six samples were
missing.
The resulting sequence counts of 50 freshwater fish taxa detected were
recorded.
</p>


<h3>Source</h3>

<p>K. Fukaya, N. I. Kondo, S. S. Matsuzaki, T. Kadoya (2021) Data from: Multispecies site occupancy modeling and study design for spatially replicated environmental DNA metabarcoding. Dryad Digital Repository. <a href="https://doi.org/10.5061/dryad.3bk3j9kkm">doi:10.5061/dryad.3bk3j9kkm</a>
</p>

<hr>
<h2 id='get_posterior'>Extract posterior samples or summary of parameters from a model-fit object.</h2><span id='topic+get_posterior'></span><span id='topic+get_post_samples'></span><span id='topic+get_post_summary'></span>

<h3>Description</h3>

<p><code>get_post_samples()</code> extracts posterior samples of the specified
parameters from a model-fit object.
</p>
<p><code>get_post_summary()</code> extracts posterior summary of the specified
parameters from a model-fit object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_post_samples(
  fit,
  parameter = c("z", "pi", "phi", "theta", "psi", "alpha", "beta", "gamma",
    "alpha_shared", "beta_shared", "gamma_shared", "Mu", "sigma", "rho")
)

get_post_summary(
  fit,
  parameter = c("z", "pi", "phi", "theta", "psi", "alpha", "beta", "gamma",
    "alpha_shared", "beta_shared", "gamma_shared", "Mu", "sigma", "rho")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_posterior_+3A_fit">fit</code></td>
<td>
<p>An <code>occumbFit</code> object.</p>
</td></tr>
<tr><td><code id="get_posterior_+3A_parameter">parameter</code></td>
<td>
<p>A string of parameter name. See Details for possible choices
and corresponding parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The functions return posterior samples or a summary of one of the
following parameters in the model, stored in the model-fit object
<code>fit</code>:
</p>

<dl>
<dt><code>z</code></dt><dd><p>Site occupancy status of species.</p>
</dd>
<dt><code>pi</code></dt><dd><p>Multinomial probabilities of species sequence read counts.</p>
</dd>
<dt><code>phi</code></dt><dd><p>Sequence relative dominance of species.</p>
</dd>
<dt><code>theta</code></dt><dd><p>Sequence capture probabilities of species.</p>
</dd>
<dt><code>psi</code></dt><dd><p>Site occupancy probabilities of species.</p>
</dd>
<dt><code>alpha</code></dt><dd><p>Species-specific effects on sequence relative dominance
(<code>phi</code>).</p>
</dd>
<dt><code>beta</code></dt><dd><p>Species-specific effects on sequence capture
probabilities (<code>theta</code>).</p>
</dd>
<dt><code>gamma</code></dt><dd><p>Species-specific effects on site occupancy
probabilities (<code>psi</code>).</p>
</dd>
<dt><code>alpha_shared</code></dt><dd><p>Effects on sequence relative dominance
(<code>phi</code>) common across species.</p>
</dd>
<dt><code>beta_shared</code></dt><dd><p>Effects on sequence capture probabilities
(<code>theta</code>) that are common across species.</p>
</dd>
<dt><code>gamma_shared</code></dt><dd><p>Effects on site occupancy probabilities
(<code>psi</code>) that are common across species.</p>
</dd>
<dt><code>Mu</code></dt><dd><p>Community-level averages of species-specific effects
(<code>alpha</code>, <code>beta</code>, <code>gamma</code>).</p>
</dd>
<dt><code>sigma</code></dt><dd><p>Standard deviations of species-specific effects
(<code>alpha</code>, <code>beta</code>, <code>gamma</code>).</p>
</dd>
<dt><code>rho</code></dt><dd><p>Correlation coefficients of the species-specific effects
(<code>alpha</code>, <code>beta</code>, <code>gamma</code>).</p>
</dd>
</dl>

<p>See <a href="https://fukayak.github.io/occumb/articles/model_specification.html">the package vignette</a>
for details of these parameters.
</p>
<p>The parameter may have dimensions corresponding to species, sites,
replicates, and effects (covariates) and the <code>dimension</code> and <code>label</code>
attributes are added to the output object to inform these dimensions.
If the sequence read count data <code>y</code> have species, site, or replicate
names appended as the <code>dimnames</code> attribute (see Details in
<code><a href="#topic+occumbData">occumbData</a>()</code>), they are copied into the <code>label</code>
attribute of the returned object.
</p>


<h3>Value</h3>

<p><code>get_post_samples()</code> returns a vector, matrix, or array of posterior
samples for a selected parameter.
</p>
<p><code>get_post_summary()</code> returns a table (matrix) of the posterior summary
of the selected parameters. The elements of the posterior summary are the
same as those obtained with the <code><a href="jagsUI.html#topic+jags">jags</a>()</code> function in the
<code>jagsUI</code> package: they include the mean, standard deviation, percentiles
of posterior samples; the <code>Rhat</code> statistic; the effective sample size,
<code>n.eff</code>; <code>overlap0</code>, which checks if 0 falls in the parameter's 95%
credible interval; and the proportion of the posterior with the same sign
as the mean, <code>f</code>.
</p>
<p>The <code>dimension</code> and <code>label</code> attributes of the output object
provide information regarding the dimensions of the parameter.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate the smallest random dataset (2 species * 2 sites * 2 reps)
I &lt;- 2 # Number of species
J &lt;- 2 # Number of sites
K &lt;- 2 # Number of replicates
y_named &lt;- array(sample.int(I * J * K), dim = c(I, J, K))
dimnames(y_named) &lt;- list(c("species 1", "species 2"),
                          c("site 1", "site 2"), NULL)
data_named &lt;- occumbData(y = y_named)

# Fitting a null model
fit &lt;- occumb(data = data_named, n.iter = 10100)

# Extract posterior samples
(post_sample_z &lt;- get_post_samples(fit, "z"))
# Look dimensions of the parameter
attributes(post_sample_z)

# Extract posterior summary
(post_summary_z &lt;- get_post_summary(fit, "z"))
# Look dimensions of the parameter
attributes(post_summary_z)

</code></pre>

<hr>
<h2 id='gof'>Goodness-of-fit assessment of the fitted model.</h2><span id='topic+gof'></span>

<h3>Description</h3>

<p><code>gof()</code> calculates omnibus discrepancy measures and
their Bayesian <code class="reqn">p</code>-values for the fitted model using the posterior
predictive check approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gof(fit, stats = c("Freeman_Tukey", "deviance"), cores = 1L, plot = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gof_+3A_fit">fit</code></td>
<td>
<p>An <code>occumbFit</code> object.</p>
</td></tr>
<tr><td><code id="gof_+3A_stats">stats</code></td>
<td>
<p>The discrepancy statistics to be applied.</p>
</td></tr>
<tr><td><code id="gof_+3A_cores">cores</code></td>
<td>
<p>The number of cores to use for parallelization.</p>
</td></tr>
<tr><td><code id="gof_+3A_plot">plot</code></td>
<td>
<p>Logical, determine if draw scatter plots of the fit statistics.</p>
</td></tr>
<tr><td><code id="gof_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the default <code>plot</code> method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A discrepancy statistic for the fitted model is obtained using a
posterior predictive checking procedure.
The following statistics are currently available:
</p>

<dl>
<dt>Freeman-Tukey statistics (default)</dt><dd><p><code class="reqn">T_{\textrm{FT}} = \sum_{i,j,k}\left(\sqrt{y_{ijk}} - \sqrt{E(y_{ijk} \mid \pi_{ijk})}\right)^2</code></p>
</dd>
<dt>Deviance statistics</dt><dd><p><code class="reqn">T_{\textrm{deviance}} = -2 \sum_{j,k} \log \textrm{Multinomial}(\boldsymbol{y}_{jk} \mid \boldsymbol{\pi}_{jk})</code></p>
</dd>
</dl>

<p>where <code class="reqn">i</code>, <code class="reqn">j</code>, and <code class="reqn">k</code> are the subscripts of species, site, and
replicate, respectively,
<code class="reqn">y_{ijk}</code> is sequence read count data,
<code class="reqn">\pi_{ijk}</code> is multinomial cell probabilities of sequence read
counts,
<code class="reqn">E(y_{ijk} \mid \pi_{ijk})</code> is expected
value of the sequence read counts conditional on their cell probabilities,
and <code class="reqn">\log \textrm{Multinomial}(\boldsymbol{y}_{jk} \mid \boldsymbol{\pi}_{jk})</code>
is the multinomial log-likelihood of the sequence read counts in replicate
<code class="reqn">k</code> of site <code class="reqn">j</code> conditional on their cell probabilities.
</p>
<p>The Bayesian <code class="reqn">p</code>-value is estimated as the probability that the value of
the discrepancy statistics of the replicated dataset is more extreme than that
of the observed dataset.
An extreme Bayesian <code class="reqn">p</code>-value may indicate inadequate model fit.
See Gelman et al. (2014), Kéry and Royle (2016), and
Conn et al. (2018) for further details on the procedures used for posterior
predictive checking.
</p>
<p>Computations can be run in parallel on multiple CPU cores where the <code>cores</code>
argument controls the degree of parallelization.
</p>


<h3>Value</h3>

<p>A list with the following named elements:
</p>

<dl>
<dt><code>stats</code></dt><dd><p>The discrepancy statistics applied.</p>
</dd>
<dt><code>p_value</code></dt><dd><p>Bayesian <code class="reqn">p</code>-value.</p>
</dd>
<dt><code>stats_obs</code></dt><dd><p>Discrepancy statistics for observed data.</p>
</dd>
<dt><code>stats_rep</code></dt><dd><p>Discrepancy statistics for repeated data.</p>
</dd>
</dl>



<h3>References</h3>

<p>P. B. Conn, D. S. Johnson, P. J. Williams, S. R. Melin and M. B. Hooten.
(2018) A guide to Bayesian model checking for ecologists.
<em>Ecological Monographs</em> <strong>88</strong>:526&ndash;542.
<a href="https://doi.org/10.1002/ecm.1314">doi:10.1002/ecm.1314</a>
</p>
<p>A. Gelman, J. B. Carlin, H. S. Stern D. B. Dunson, A. Vehtari and
D. B. Rubin (2013) <em>Bayesian Data Analysis</em>. 3rd edition.
Chapman and Hall/CRC. <a href="http://www.stat.columbia.edu/~gelman/book/">http://www.stat.columbia.edu/~gelman/book/</a>
</p>
<p>M. Kéry and J. A. Royle (2016) <em>Applied Hierarchical Modeling in
Ecology &mdash; Analysis of Distribution, Abundance and Species Richness in R
and BUGS. Volume 1: Prelude and Static Models</em>. Academic Press.
<a href="https://www.mbr-pwrc.usgs.gov/pubanalysis/keryroylebook/">https://www.mbr-pwrc.usgs.gov/pubanalysis/keryroylebook/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate the smallest random dataset (2 species * 2 sites * 2 reps)
I &lt;- 2 # Number of species
J &lt;- 2 # Number of sites
K &lt;- 2 # Number of replicates
data &lt;- occumbData(
    y = array(sample.int(I * J * K), dim = c(I, J, K)),
    spec_cov = list(cov1 = rnorm(I)),
    site_cov = list(cov2 = rnorm(J),
                    cov3 = factor(1:J)),
    repl_cov = list(cov4 = matrix(rnorm(J * K), J, K)))

# Fitting a null model
fit &lt;- occumb(data = data)

# Goodness-of-fit assessment
gof_result &lt;- gof(fit)
gof_result

</code></pre>

<hr>
<h2 id='list_cond_L'>Conditions for local assessment under certain budget and cost values.</h2><span id='topic+list_cond_L'></span>

<h3>Description</h3>

<p><code>list_cond_L()</code> constructs a list of possible local species
diversity assessment conditions under the specified budget and cost values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_cond_L(budget, lambda1, lambda2, fit, K = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="list_cond_L_+3A_budget">budget</code></td>
<td>
<p>A numeric specifying budget amount. The currency unit
is arbitrary but must be consistent with that of <code>lambda1</code> and <code>lambda2</code>.</p>
</td></tr>
<tr><td><code id="list_cond_L_+3A_lambda1">lambda1</code></td>
<td>
<p>A numeric specifying the cost per sequence read for
high-throughput sequencing. The currency unit is arbitrary but must be
consistent with that of <code>budget</code> and <code>lambda2</code>.</p>
</td></tr>
<tr><td><code id="list_cond_L_+3A_lambda2">lambda2</code></td>
<td>
<p>A numeric specifying the cost per replicate for library
preparation. The currency unit is arbitrary but must be consistent with that
of <code>budget</code> and <code>lambda1</code>.</p>
</td></tr>
<tr><td><code id="list_cond_L_+3A_fit">fit</code></td>
<td>
<p>An <code>occumbFit</code> object.</p>
</td></tr>
<tr><td><code id="list_cond_L_+3A_k">K</code></td>
<td>
<p>An optional vector for manually specifying the number of replicates.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can generate a data frame object to be given to the
<code>settings</code> argument of <code>eval_util_L()</code>; see Examples of <code>eval_util_L()</code>.
By default, it outputs a list of all feasible combinations of values for the
number of replicates per site <code>K</code> and the sequencing depth per replicate
<code>N</code> based on the given budget, cost values, and number of sites
(identified by reference to the <code>fit</code> object). The resulting <code>N</code> can be a
non-integer because it is calculated simply by assuming that the maximum
value can be obtained. To obtain a list for only a subset of the
possible <code>K</code> values under a given budget and cost value, the <code>K</code>
argument is used to provide a vector of the desired <code>K</code> values.
</p>


<h3>Value</h3>

<p>A data frame containing columns named <code>budget</code>, <code>lambda1</code>, <code>lambda2</code>,
<code>K</code>, and <code>N</code>.
</p>

<hr>
<h2 id='list_cond_R'>Conditions for regional assessment under certain budget and cost values.</h2><span id='topic+list_cond_R'></span>

<h3>Description</h3>

<p><code>list_cond_R()</code> constructs a list of possible regional species
diversity assessment conditions under the specified budget and cost values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_cond_R(budget, lambda1, lambda2, lambda3, J = NULL, K = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="list_cond_R_+3A_budget">budget</code></td>
<td>
<p>A numeric specifying budget amount. The currency unit is
arbitrary but must be consistent with that of <code>lambda1</code>, <code>lambda2</code>, and <code>lambda3</code>.</p>
</td></tr>
<tr><td><code id="list_cond_R_+3A_lambda1">lambda1</code></td>
<td>
<p>A numeric specifying the cost per sequence read for
high-throughput sequencing. The currency unit is arbitrary but must be
consistent with that of <code>budget</code>, <code>lambda2</code>, and <code>lambda3</code>.</p>
</td></tr>
<tr><td><code id="list_cond_R_+3A_lambda2">lambda2</code></td>
<td>
<p>A numeric specifying the cost per replicate for library
preparation. The currency unit is arbitrary but must be consistent with that
of <code>budget</code>, <code>lambda1</code>, and <code>lambda3</code>.</p>
</td></tr>
<tr><td><code id="list_cond_R_+3A_lambda3">lambda3</code></td>
<td>
<p>A numeric specifying the visiting cost per site. The currency
unit is arbitrary but must be consistent with that of <code>budget</code>, <code>lambda1</code>,
and <code>lambda2</code>.</p>
</td></tr>
<tr><td><code id="list_cond_R_+3A_j">J</code></td>
<td>
<p>An optional vector for manually specifying the number of sites</p>
</td></tr>
<tr><td><code id="list_cond_R_+3A_k">K</code></td>
<td>
<p>An optional vector used to specify the number of replicates manually.
For computational convenience, the <code>K</code> values must be in ascending order.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can generate a data frame object to be given to the
<code>settings</code> argument of <code>eval_util_R()</code>; see Examples of <code>eval_util_R()</code>.
By default, it outputs a list of all feasible combinations of values for the
number of sites <code>J</code>, number of replicates per site <code>K</code>, and sequencing
depth per replicate <code>N</code> based on the given budget and cost values.
The resulting <code>N</code> can be a non-integer because it is calculated simply by
assuming that the maximum value can be obtained.
If one wants to obtain a list for only a subset of the possible values of <code>J</code>
and <code>K</code> under a given budget and cost value, use the <code>J</code> and/or <code>K</code>
arguments (in fact, it is recommended that a relatively small number of <code>K</code>
values be specified using the <code>K</code> argument because the list of all
conditions achievable under moderate budget and cost values can be large, and
it is rarely practical to have a vast number of replicates per site). If a
given combination of <code>J</code> and <code>K</code> values is not feasible under the specified
budget and cost values, the combination will be ignored and excluded from
the output.
</p>


<h3>Value</h3>

<p>A data frame containing columns named <code>budget</code>, <code>lambda1</code>, <code>lambda2</code>,
<code>lambda3</code>, <code>J</code>, <code>K</code>, and <code>N</code>.
</p>

<hr>
<h2 id='occumb'>Model-fitting function.</h2><span id='topic+occumb'></span>

<h3>Description</h3>

<p><code>occumb()</code> fits the multispecies site-occupancy model for eDNA
metabarcoding (Fukaya et al. 2022) and returns a model-fit object containing
posterior samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>occumb(
  formula_phi = ~1,
  formula_theta = ~1,
  formula_psi = ~1,
  formula_phi_shared = ~1,
  formula_theta_shared = ~1,
  formula_psi_shared = ~1,
  prior_prec = 1e-04,
  prior_ulim = 10000,
  data,
  n.chains = 4,
  n.adapt = NULL,
  n.burnin = 10000,
  n.thin = 10,
  n.iter = 20000,
  parallel = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="occumb_+3A_formula_phi">formula_phi</code></td>
<td>
<p>A right-hand side formula describing species-specific
effects of sequence relative dominance (<code class="reqn">\phi</code>).</p>
</td></tr>
<tr><td><code id="occumb_+3A_formula_theta">formula_theta</code></td>
<td>
<p>A right-hand side formula describing species-specific
effects of sequence capture probability (<code class="reqn">\theta</code>).</p>
</td></tr>
<tr><td><code id="occumb_+3A_formula_psi">formula_psi</code></td>
<td>
<p>A right-hand side formula describing species-specific
effects of occupancy probability (<code class="reqn">\psi</code>).</p>
</td></tr>
<tr><td><code id="occumb_+3A_formula_phi_shared">formula_phi_shared</code></td>
<td>
<p>A right-hand side formula describing effects of
sequence relative dominance (<code class="reqn">\phi</code>) that are common across
species. The intercept term is ignored (see Details).</p>
</td></tr>
<tr><td><code id="occumb_+3A_formula_theta_shared">formula_theta_shared</code></td>
<td>
<p>A right-hand side formula describing effects of
sequence capture probability (<code class="reqn">\theta</code>) that are common across
species.
The intercept term is ignored (see Details).</p>
</td></tr>
<tr><td><code id="occumb_+3A_formula_psi_shared">formula_psi_shared</code></td>
<td>
<p>A right-hand side formula describing effects of
occupancy probability (<code class="reqn">\psi</code>) that are common across species.
The intercept term is ignored (see Details).</p>
</td></tr>
<tr><td><code id="occumb_+3A_prior_prec">prior_prec</code></td>
<td>
<p>Precision of normal prior distribution for the
community-level average of species-specific parameters and effects
common across species.</p>
</td></tr>
<tr><td><code id="occumb_+3A_prior_ulim">prior_ulim</code></td>
<td>
<p>Upper limit of uniform prior distribution for the
standard deviation of species-specific parameters.</p>
</td></tr>
<tr><td><code id="occumb_+3A_data">data</code></td>
<td>
<p>A dataset supplied as an <code><a href="#topic+occumbData">occumbData</a></code> class object.</p>
</td></tr>
<tr><td><code id="occumb_+3A_n.chains">n.chains</code></td>
<td>
<p>Number of Markov chains to run.</p>
</td></tr>
<tr><td><code id="occumb_+3A_n.adapt">n.adapt</code></td>
<td>
<p>Number of iterations to run in the JAGS adaptive phase.</p>
</td></tr>
<tr><td><code id="occumb_+3A_n.burnin">n.burnin</code></td>
<td>
<p>Number of iterations at the beginning of the chain to discard.</p>
</td></tr>
<tr><td><code id="occumb_+3A_n.thin">n.thin</code></td>
<td>
<p>Thinning rate. Must be a positive integer.</p>
</td></tr>
<tr><td><code id="occumb_+3A_n.iter">n.iter</code></td>
<td>
<p>Total number of iterations per chain (including burn-in).</p>
</td></tr>
<tr><td><code id="occumb_+3A_parallel">parallel</code></td>
<td>
<p>If TRUE, run MCMC chains in parallel on multiple CPU cores.</p>
</td></tr>
<tr><td><code id="occumb_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="jagsUI.html#topic+jags">jags</a>()</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>occumb()</code> allows the fitting of a range of multispecies site
occupancy models, including covariates at different levels of the data
generation process.
The most general form of the model can be written as follows (the notation
follows that of the original article; see References).
</p>
<p>Sequence read counts:
</p>
<p style="text-align: center;"><code class="reqn">(y_{1jk}, ..., y_{Ijk}) \sim \textrm{Multinomial}((\pi_{1jk}, ...,  \pi_{Ijk}), N_{jk}),</code>
</p>

<p style="text-align: center;"><code class="reqn">\pi_{ijk} = \frac{u_{ijk}r_{ijk}}{\sum_m u_{mjk}r_{mjk}},</code>
</p>

<p>Relative frequency of species sequences:
</p>
<p style="text-align: center;"><code class="reqn">r_{ijk} \sim \textrm{Gamma}(\phi_{ijk}, 1),</code>
</p>

<p>Capture of species sequences:
</p>
<p style="text-align: center;"><code class="reqn">u_{ijk} \sim \textrm{Bernoulli}(z_{ij}\theta_{ijk}),</code>
</p>

<p>Site occupancy of species:
</p>
<p style="text-align: center;"><code class="reqn">z_{ij} \sim \textrm{Bernoulli}(\psi_{ij}),</code>
</p>

<p>where the variations of <code class="reqn">\phi</code>, <code class="reqn">\theta</code>, and
<code class="reqn">\psi</code> are modeled by specifying model formulas in
<code>formula_phi</code>, <code>formula_theta</code>, <code>formula_psi</code>,
<code>formula_phi_shared</code>, <code>formula_theta_shared</code>, and
<code>formula_psi_shared.</code>
</p>
<p>Each parameter may have species-specific effects and effects that are common
across species, where the former is specified by <code>formula_phi</code>,
<code>formula_theta</code>, and <code>formula_psi</code>, whereas
<code>formula_phi_shared</code>, <code>formula_theta_shared</code>, and
<code>formula_psi_shared</code> specify the latter.
As species-specific intercepts are specified by default, the intercept
terms in <code>formula_phi_shared</code>, <code>formula_theta_shared</code>, and
<code>formula_psi_shared</code> are always ignored.
Covariate terms must be found in the names of the list elements stored
in the <code>spec_cov</code>, <code>site_cov</code>, or <code>repl_cov</code> slots in the dataset
object provided with the <code>data</code> argument.
Covariates are modeled using the log link function for <code class="reqn">\phi</code>
and logit link function for <code class="reqn">\theta</code> and <code class="reqn">\psi.</code>
</p>
<p>The two arguments, <code>prior_prec</code> and <code>prior_ulim</code>, control the
prior distribution of parameters. For the community-level average of
species-specific effects and effects common across species, a normal
prior distribution with a mean of 0 and precision (i.e., the inverse of the
variance) <code>prior_prec</code> is specified. For the standard deviation of
species-specific effects, a uniform prior distribution with a lower limit of
zero and an upper limit of <code>prior_ulim</code> is specified. For the correlation
coefficient of species-specific effects, a uniform prior distribution in the
range of <code class="reqn">-</code>1 to 1 is specified by default.
</p>
<p>See <a href="https://fukayak.github.io/occumb/articles/model_specification.html">the package vignette</a>
for details on the model specifications in <code>occumb()</code>.
</p>
<p>The <code>data</code> argument requires a dataset object to be generated using
<code>ocumbData()</code>; see the document of <code><a href="#topic+occumbData">occumbData</a>()</code>.
</p>
<p>The model is fit using the <code><a href="jagsUI.html#topic+jags">jags</a>()</code> function of the
<a href="https://cran.r-project.org/package=jagsUI">jagsUI</a> package, where
Markov chain Monte Carlo methods are used to
obtain posterior samples of the parameters and latent variables.
Arguments <code>n.chains</code>, <code>n.adapt</code>, <code>n.burnin</code>, <code>n.thin</code>,
<code>n.iter</code>, and <code>parallel</code> are passed on to arguments of the
same name in the <code><a href="jagsUI.html#topic+jags">jags</a>()</code> function.
See the document of <a href="https://cran.r-project.org/package=jagsUI">jagsUI</a>'s
<code><a href="jagsUI.html#topic+jags">jags</a>()</code> function for details.
</p>


<h3>Value</h3>

<p>An S4 object of the <code>occumbFit</code> class containing the results of
the model fitting and the supplied dataset.
</p>


<h3>References</h3>

<p>K. Fukaya, N. I. Kondo, S. S. Matsuzaki and T. Kadoya (2022)
Multispecies site occupancy modelling and study design for spatially
replicated environmental DNA metabarcoding. <em>Methods in Ecology
and Evolution</em> <strong>13</strong>:183&ndash;193.
<a href="https://doi.org/10.1111/2041-210X.13732">doi:10.1111/2041-210X.13732</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate the smallest random dataset (2 species * 2 sites * 2 reps)
I &lt;- 2 # Number of species
J &lt;- 2 # Number of sites
K &lt;- 2 # Number of replicates
data &lt;- occumbData(
    y = array(sample.int(I * J * K), dim = c(I, J, K)),
    spec_cov = list(cov1 = rnorm(I)),
    site_cov = list(cov2 = rnorm(J),
                    cov3 = factor(1:J)),
    repl_cov = list(cov4 = matrix(rnorm(J * K), J, K)))


# Fitting a null model (includes only species-specific intercepts)
res0 &lt;- occumb(data = data)

# Add species-specific effects of site covariates in occupancy probabilities
res1 &lt;- occumb(formula_psi = ~ cov2, data = data)        # Continuous covariate
res2 &lt;- occumb(formula_psi = ~ cov3, data = data)        # Categorical covariate
res3 &lt;- occumb(formula_psi = ~ cov2 * cov3, data = data) # Interaction

# Add species covariate in the three parameters
# Note that species covariates are modeled as common effects
res4 &lt;- occumb(formula_phi_shared = ~ cov1, data = data)   # phi
res5 &lt;- occumb(formula_theta_shared = ~ cov1, data = data) # theta
res6 &lt;- occumb(formula_psi_shared = ~ cov1, data = data)   # psi

# Add replicate covariates
# Note that replicate covariates can only be specified for theta and phi
res7 &lt;- occumb(formula_phi = ~ cov4, data = data)   # phi
res8 &lt;- occumb(formula_theta = ~ cov4, data = data) # theta

# Specify the prior distribution and MCMC settings explicitly
res9 &lt;- occumb(data = data, prior_prec = 1E-2, prior_ulim = 1E2,
               n.chains = 1, n.burnin = 1000, n.thin = 1, n.iter = 2000)
res10 &lt;- occumb(data = data, parallel = TRUE, n.cores = 2) # Run MCMC in parallel

</code></pre>

<hr>
<h2 id='occumbData'>Constructor for occumbData data class.</h2><span id='topic+occumbData'></span>

<h3>Description</h3>

<p><code>occumbData()</code> creates a data list compatible with the model fitting
function <code><a href="#topic+occumb">occumb</a>()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>occumbData(y, spec_cov = NULL, site_cov = NULL, repl_cov = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="occumbData_+3A_y">y</code></td>
<td>
<p>A 3-D array of sequence read counts (<code>integer</code> values) that may
have a <code>dimnames</code> attribute.
The dimensions are ordered by species, site, and replicate.
The data for missing replicates are represented by zero vectors.
<code>NA</code>s are not allowed.</p>
</td></tr>
<tr><td><code id="occumbData_+3A_spec_cov">spec_cov</code></td>
<td>
<p>A named list of species covariates.
Each covariate can be a vector of continuous (<code>numeric</code>
or <code>integer</code>) or discrete (<code>logical</code>,
<code>factor</code>, or <code>character</code>) variables whose length
is <code>dim(y)[1]</code> (i.e., the number of species).
<code>NA</code>s are not allowed.</p>
</td></tr>
<tr><td><code id="occumbData_+3A_site_cov">site_cov</code></td>
<td>
<p>A named list of site covariates.
Each covariate can be a vector of continuous (<code>numeric</code>
or <code>integer</code>) or discrete (<code>logical</code>,
<code>factor</code>, or <code>character</code>) variables whose length
is <code>dim(y)[1]</code> (i.e., the number of sites).
<code>NA</code>s are not allowed.</p>
</td></tr>
<tr><td><code id="occumbData_+3A_repl_cov">repl_cov</code></td>
<td>
<p>A named list of replicate covariates.
Each covariate can be a matrix of continuous (<code>numeric</code>
or <code>integer</code>) or discrete (<code>logical</code> or
<code>character</code>) variables with dimensions equal to
<code>dim(y)[2:3]</code> (i.e., number of sites <code class="reqn">\times</code>
number of replicates).
<code>NA</code>s are not allowed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The element (i.e., covariate) names for <code>spec_cov</code>, <code>site_cov</code>, and
<code>repl_cov</code> must all be unique.
If <code>y</code> has a <code>dimnames</code> attribute, it is retained in the resulting
<code>occumbData</code> object, and can be referenced in subsequent analyses.
</p>


<h3>Value</h3>

<p>An S4 object of the <code>occumbData</code> class.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Generate the smallest random dataset (2 species * 2 sites * 2 reps)
I &lt;- 2 # Number of species
J &lt;- 2 # Number of sites
K &lt;- 2 # Number of replicates
data &lt;- occumbData(
    y = array(sample.int(I * J * K), dim = c(I, J, K)),
    spec_cov = list(cov1 = rnorm(I)),
    site_cov = list(cov2 = rnorm(J), cov3 = factor(1:J)),
    repl_cov = list(cov4 = matrix(rnorm(J * K), J, K))
)

# A case for named y (with species and site names)
y_named &lt;- array(sample.int(I * J * K), dim = c(I, J, K))
dimnames(y_named) &lt;- list(c("common species", "uncommon species"),
                          c("good site", "bad site"), NULL)
data_named &lt;- occumbData(
    y = y_named,
    spec_cov = list(cov1 = rnorm(I)),
    site_cov = list(cov2 = rnorm(J), cov3 = factor(1:J)),
    repl_cov = list(cov4 = matrix(rnorm(J * K), J, K))
)

# A real data example
data(fish_raw)
fish &lt;- occumbData(
    y = fish_raw$y,
    spec_cov = list(mismatch = fish_raw$mismatch),
    site_cov = list(riverbank = fish_raw$riverbank)
)

# Get an overview of the datasets
summary(data)
summary(data_named)
summary(fish)
</code></pre>

<hr>
<h2 id='plot+2CoccumbFit-method'>Plot method for occumbFit class.</h2><span id='topic+plot+2CoccumbFit-method'></span>

<h3>Description</h3>

<p>Applies <a href="https://cran.r-project.org/package=jagsUI">jagsUI</a>'s
plot method to an <code>occumbFit</code> object to draw trace plots
and density plots of MCMC samples of model parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'occumbFit'
plot(x, y = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2CoccumbFit-method_+3A_x">x</code></td>
<td>
<p>An <code>occumbFit</code> object.</p>
</td></tr>
<tr><td><code id="plot+2B2CoccumbFit-method_+3A_y">y</code></td>
<td>
<p><code>NULL</code></p>
</td></tr>
<tr><td><code id="plot+2B2CoccumbFit-method_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the plot method for
<a href="https://cran.r-project.org/package=jagsUI">jagsUI</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>NULL</code> invisibly.
</p>

<hr>
<h2 id='plot+2CoccumbGof-method'>Plot method for occumbGof class.</h2><span id='topic+plot+2CoccumbGof-method'></span>

<h3>Description</h3>

<p>Draws a scatter plot of fit statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'occumbGof'
plot(x, y = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2CoccumbGof-method_+3A_x">x</code></td>
<td>
<p>An <code>occumbGof</code> object.</p>
</td></tr>
<tr><td><code id="plot+2B2CoccumbGof-method_+3A_y">y</code></td>
<td>
<p><code>NULL</code></p>
</td></tr>
<tr><td><code id="plot+2B2CoccumbGof-method_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the default <code>plot</code> method.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>NULL</code> invisibly.
</p>

<hr>
<h2 id='predict+2CoccumbFit-method'>Predict method for occumbFit class.</h2><span id='topic+predict+2CoccumbFit-method'></span>

<h3>Description</h3>

<p>Obtain predictions of parameters related to species occupancy
and detection from an <code>occumbFit</code> model object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'occumbFit'
predict(
  object,
  newdata = NULL,
  parameter = c("phi", "theta", "psi"),
  scale = c("response", "link"),
  type = c("quantiles", "mean", "samples")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict+2B2CoccumbFit-method_+3A_object">object</code></td>
<td>
<p>An <code>occumbFit</code> object.</p>
</td></tr>
<tr><td><code id="predict+2B2CoccumbFit-method_+3A_newdata">newdata</code></td>
<td>
<p>An optional <code>occumbData</code> object with covariates to be
used for prediction. If omitted, the fitted covariates are used.</p>
</td></tr>
<tr><td><code id="predict+2B2CoccumbFit-method_+3A_parameter">parameter</code></td>
<td>
<p>The parameter to be predicted.</p>
</td></tr>
<tr><td><code id="predict+2B2CoccumbFit-method_+3A_scale">scale</code></td>
<td>
<p>The scale on which the prediction is made.
<code>type = "response"</code> returns the prediction on the original scale of the
parameter. <code>type = "link"</code> returns the prediction on the link scale of
the parameter.</p>
</td></tr>
<tr><td><code id="predict+2B2CoccumbFit-method_+3A_type">type</code></td>
<td>
<p>The type of prediction.
<code>type = "quantiles"</code> returns 50% quantile as the posterior median of
the prediction in addition to 2.5 and 97.5% quantiles as the lower and
upper limits of the 95% credible interval of the prediction.
<code>type = "mean"</code> returns the posterior mean of the prediction.
<code>type = "samples"</code> returns the posterior samples of the prediction.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Applying <code>predict()</code> to an <code>occumbFit</code> object generates predictions
for the specified parameter (<code>phi</code>, <code>theta</code>, or <code>psi</code>) based
on the estimated effects and the given covariates. It is important to
recognize that the predictions are specific to the individual species being
modeled since they depend on the estimated species-specific effects (i.e.,
<code>alpha</code>, <code>beta</code>, and <code>gamma</code>; see
<a href="https://fukayak.github.io/occumb/articles/model_specification.html">the package vignette</a> for details).
When providing <code>newdata</code>, it must thus be assumed that the set of
species contained in <code>newdata</code> is the same as that of the data being
fitted.
</p>


<h3>Value</h3>

<p>Predictions are obtained as a matrix or array that can have dimensions
corresponding to statistics (or samples), species, sites, and replicates.
The <code>dimension</code> and <code>label</code> attributes are added to the output
object to inform these dimensions.
If the sequence read count data <code>y</code> has species, site, or replicate
names appended as the <code>dimnames</code> attribute (see Details in
<code><a href="#topic+occumbData">occumbData</a>()</code>), they will be copied into the <code>label</code>
attribute of the returned object.
</p>

<hr>
<h2 id='summary+2CoccumbData-method'>Summary method for occumbData class.</h2><span id='topic+summary+2CoccumbData-method'></span>

<h3>Description</h3>

<p>Summarizes dataset stored in an <code>occumbData</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'occumbData'
summary(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2CoccumbData-method_+3A_object">object</code></td>
<td>
<p>An <code>occumbData</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>NULL</code> invisibly.
</p>

<hr>
<h2 id='summary+2CoccumbFit-method'>Summary method for occumbFit class.</h2><span id='topic+summary+2CoccumbFit-method'></span>

<h3>Description</h3>

<p>Summarizes model fitting result stored in an <code>occumbFit</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'occumbFit'
summary(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2CoccumbFit-method_+3A_object">object</code></td>
<td>
<p>An <code>occumbFit</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>NULL</code> invisibly.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
