<!DOCTYPE html><html><head><title>Help for package oce</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {oce}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#oce'><p>A Package for Oceanographic Analysis</p></a></li>
<li><a href='#[[,adp-method'><p>Extract Something From an adp Object</p></a></li>
<li><a href='#[[,adv-method'><p>Extract Something from an adv Object</p></a></li>
<li><a href='#[[,amsr-method'><p>Extract Something From an amsr Object</p></a></li>
<li><a href='#[[,argo-method'><p>Extract Something From an argo Object</p></a></li>
<li><a href='#[[,bremen-method'><p>Extract Something From a bremen Object</p></a></li>
<li><a href='#[[,cm-method'><p>Extract Something From a cm Object</p></a></li>
<li><a href='#[[,coastline-method'><p>Extract Something From a coastline Object</p></a></li>
<li><a href='#[[,ctd-method'><p>Extract Something From a ctd Object</p></a></li>
<li><a href='#[[,echosounder-method'><p>Extract Something From an echosounder Object</p></a></li>
<li><a href='#[[,g1sst-method'><p>Extract Something From a g1sst Object</p></a></li>
<li><a href='#[[,gps-method'><p>Extract Something From a gps Object</p></a></li>
<li><a href='#[[,ladp-method'><p>Extract Something From an ladp Object</p></a></li>
<li><a href='#[[,landsat-method'><p>Extract Something From a landsat Object</p></a></li>
<li><a href='#[[,lisst-method'><p>Extract Something From a lisst Object</p></a></li>
<li><a href='#[[,lobo-method'><p>Extract Something From a lobo Object</p></a></li>
<li><a href='#[[,met-method'><p>Extract Something From a met Object</p></a></li>
<li><a href='#[[,oce-method'><p>Extract Something From an oce Object</p></a></li>
<li><a href='#[[,odf-method'><p>Extract Something From an odf Object</p></a></li>
<li><a href='#[[,rsk-method'><p>Extract Something From a rsk Object</p></a></li>
<li><a href='#[[,sealevel-method'><p>Extract Something From a sealevel Object</p></a></li>
<li><a href='#[[,section-method'><p>Extract Something From a section Object</p></a></li>
<li><a href='#[[,tidem-method'><p>Extract Something From a tidem Object</p></a></li>
<li><a href='#[[,topo-method'><p>Extract Something From a topo Object</p></a></li>
<li><a href='#[[,windrose-method'><p>Extract Something From a windrose Object</p></a></li>
<li><a href='#[[,xbt-method'><p>Extract Something From an xbt Object</p></a></li>
<li><a href='#[[&lt;-,adp-method'><p>Replace Parts of an adp Object</p></a></li>
<li><a href='#[[&lt;-,adv-method'><p>Replace Parts of an adv Object</p></a></li>
<li><a href='#[[&lt;-,amsr-method'><p>Replace Parts of an amsr Object</p></a></li>
<li><a href='#[[&lt;-,argo-method'><p>Replace Parts of an argo Object</p></a></li>
<li><a href='#[[&lt;-,bremen-method'><p>Replace Parts of a bremen Object</p></a></li>
<li><a href='#[[&lt;-,cm-method'><p>Replace Parts of a cm Object</p></a></li>
<li><a href='#[[&lt;-,coastline-method'><p>Replace Parts of a coastline Object</p></a></li>
<li><a href='#[[&lt;-,ctd-method'><p>Replace Parts of a ctd Object</p></a></li>
<li><a href='#[[&lt;-,echosounder-method'><p>Replace Parts of an echosounder Object</p></a></li>
<li><a href='#[[&lt;-,g1sst-method'><p>Replace Parts of a g1sst Object</p></a></li>
<li><a href='#[[&lt;-,gps-method'><p>Replace Parts of a gps Object</p></a></li>
<li><a href='#[[&lt;-,ladp-method'><p>Replace Parts of an ladp Object</p></a></li>
<li><a href='#[[&lt;-,landsat-method'><p>Replace Parts of a landsat Object</p></a></li>
<li><a href='#[[&lt;-,lisst-method'><p>Replace Parts of a lisst Object</p></a></li>
<li><a href='#[[&lt;-,lobo-method'><p>Replace Parts of a lobo Object</p></a></li>
<li><a href='#[[&lt;-,met-method'><p>Replace Parts of a met Object</p></a></li>
<li><a href='#[[&lt;-,oce-method'><p>Replace Parts of an oce Object</p></a></li>
<li><a href='#[[&lt;-,odf-method'><p>Replace Parts of an odf Object</p></a></li>
<li><a href='#[[&lt;-,rsk-method'><p>Replace Parts of an rsk Object</p></a></li>
<li><a href='#[[&lt;-,sealevel-method'><p>Replace Parts of a sealevel Object</p></a></li>
<li><a href='#[[&lt;-,section-method'><p>Replace Parts of a section Object</p></a></li>
<li><a href='#[[&lt;-,tidem-method'><p>Replace Parts of a tidem Object</p></a></li>
<li><a href='#[[&lt;-,topo-method'><p>Replace Parts of a topo Object</p></a></li>
<li><a href='#[[&lt;-,windrose-method'><p>Replace Parts of a windrose Object</p></a></li>
<li><a href='#[[&lt;-,xbt-method'><p>Replace Parts of an xbt Object</p></a></li>
<li><a href='#abbreviateTimeLabels'><p>Abbreviate a Vector of Times by Removing Commonalities</p></a></li>
<li><a href='#ad2cpCodeToName'><p>Map AD2CP ID Code to oce Name</p></a></li>
<li><a href='#ad2cpHeaderValue'><p>Infer an Item From a Nortek AD2CP File Header</p></a></li>
<li><a href='#addSpine'><p>Add a Spine to a section Object</p></a></li>
<li><a href='#adp'><p>Sample adp Data</p></a></li>
<li><a href='#adp_rdi.000'><p>Sample adp File in RDI Format</p></a></li>
<li><a href='#adp-class'><p>Class to Store Acoustic-Doppler Profiler Data</p></a></li>
<li><a href='#adpAd2cpFileTrim'><p>Trim an AD2CP File</p></a></li>
<li><a href='#adpConvertRawToNumeric'><p>Convert Raw to Numeric Values in an adp Object</p></a></li>
<li><a href='#adpEnsembleAverage'><p>Ensemble Average an adp Object in Time</p></a></li>
<li><a href='#adpFlagPastBoundary'><p>Flag adp Data Past Water Column Boundary</p></a></li>
<li><a href='#adpRdiFileTrim'><p>Trim an RDI adp File</p></a></li>
<li><a href='#adv'><p>Sample adv Data</p></a></li>
<li><a href='#adv-class'><p>Class to Store Acoustic-Doppler Velocimeter Data</p></a></li>
<li><a href='#advSontekAdrFileTrim'><p>Trim a Sontek ADR adv File</p></a></li>
<li><a href='#airRho'><p>Air Density</p></a></li>
<li><a href='#amsr'><p>Sample amsr Data (Near Nova Scotia)</p></a></li>
<li><a href='#amsr-class'><p>Class to Store AMSR-2 Satellite Data</p></a></li>
<li><a href='#angle2hms'><p>Convert Astronomical Angle in Degrees to Hours, Minutes and Seconds</p></a></li>
<li><a href='#angleRemap'><p>Convert Angle From 0:360 to -180:180 Convention</p></a></li>
<li><a href='#applyMagneticDeclination'><p>Alter an Object to Account for Magnetic Declination (Generic)</p></a></li>
<li><a href='#applyMagneticDeclination,adp-method'><p>Alter an adp Object to Account for Magnetic Declination</p></a></li>
<li><a href='#applyMagneticDeclination,adv-method'><p>Alter an adv Object to Account for Magnetic Declination</p></a></li>
<li><a href='#applyMagneticDeclination,cm-method'><p>Alter a cm Object to Account for Magnetic Declination</p></a></li>
<li><a href='#applyMagneticDeclination,oce-method'><p>Alter an Object to Account for Magnetic Declination</p></a></li>
<li><a href='#approx3d'><p>Trilinear Interpolation in a 3D Array</p></a></li>
<li><a href='#argo'><p>Sample argo Data</p></a></li>
<li><a href='#argo-class'><p>Class to Store Argo Profiler Data</p></a></li>
<li><a href='#argoGrid'><p>Grid Argo Float Data</p></a></li>
<li><a href='#argoJuldToTime'><p>Convert Argo Julian Day to R Time</p></a></li>
<li><a href='#argoNames2oceNames'><p>Convert Argo Data Name to Oce Name</p></a></li>
<li><a href='#argShow'><p>Show a Function Argument</p></a></li>
<li><a href='#as.adp'><p>Create an adp Object</p></a></li>
<li><a href='#as.argo'><p>Coerce Data Into an argo Object</p></a></li>
<li><a href='#as.cm'><p>Coerce Data Into a cm Object</p></a></li>
<li><a href='#as.coastline'><p>Coerce Data Into a coastline Object</p></a></li>
<li><a href='#as.ctd'><p>Coerce Data Into a ctd Object</p></a></li>
<li><a href='#as.echosounder'><p>Coerce Data Into an echosounder Object</p></a></li>
<li><a href='#as.gps'><p>Coerce Data Into a gps Object</p></a></li>
<li><a href='#as.ladp'><p>Coerce Data Into an ladp object</p></a></li>
<li><a href='#as.lisst'><p>Coerce Data Into a lisst Object</p></a></li>
<li><a href='#as.lobo'><p>Coerce Data Into a lobo Object</p></a></li>
<li><a href='#as.met'><p>Coerce Data Into a met Object</p></a></li>
<li><a href='#as.oce'><p>Coerce Something Into an oce Object</p></a></li>
<li><a href='#as.rsk'><p>Coerce Data Into a rsk Object</p></a></li>
<li><a href='#as.sealevel'><p>Coerce Data Into a sealevel Object</p></a></li>
<li><a href='#as.section'><p>Create a Section</p></a></li>
<li><a href='#as.tidem'><p>Create tidem Object From Fitted Harmonic Data</p></a></li>
<li><a href='#as.topo'><p>Coerce Data Into a topo Object</p></a></li>
<li><a href='#as.unit'><p>Convert a String to a Unit</p></a></li>
<li><a href='#as.windrose'><p>Create a windrose Object</p></a></li>
<li><a href='#as.xbt'><p>Create an xbt Object</p></a></li>
<li><a href='#bcdToInteger'><p>Convert a BCD Value to an Integer Value</p></a></li>
<li><a href='#beamName'><p>Get Names of Acoustic-Doppler Beams</p></a></li>
<li><a href='#beamToXyz'><p>Change the Coordinate System in an adv or adp Object</p></a></li>
<li><a href='#beamToXyzAdp'><p>Convert adp Object From Beam to XYZ Coordinates</p></a></li>
<li><a href='#beamToXyzAdpAD2CP'><p>Convert From Beam to XYZ Coordinates (AD2CP adp Data)</p></a></li>
<li><a href='#beamToXyzAdv'><p>Convert adv Object from Beam Coordinates to XYZ Coordinates</p></a></li>
<li><a href='#beamUnspreadAdp'><p>Adjust adp Object to Account for Spherical Spreading</p></a></li>
<li><a href='#bilinearInterp'><p>Bilinear Interpolation Within a Grid</p></a></li>
<li><a href='#binApply1D'><p>Apply a Function to Vector Data</p></a></li>
<li><a href='#binApply2D'><p>Apply a Function to Matrix Data</p></a></li>
<li><a href='#binAverage'><p>Bin-average a Vector y, Based on x Values</p></a></li>
<li><a href='#binCount1D'><p>Bin-count Vector Data</p></a></li>
<li><a href='#binCount2D'><p>Bin-count Matrix Data</p></a></li>
<li><a href='#binmapAdp'><p>Bin-map an adp Object</p></a></li>
<li><a href='#binMean1D'><p>Bin-average f=f(x)</p></a></li>
<li><a href='#binMean2D'><p>Bin-average f=f(x,y)</p></a></li>
<li><a href='#bound125'><p>Calculate a Bound, Rounded up to Mantissa 1, 2, or 5</p></a></li>
<li><a href='#bremen-class'><p>Class to Store Bremen-formatted Data</p></a></li>
<li><a href='#byteToBinary'><p>Format Bytes as Binary (Defunct)</p></a></li>
<li><a href='#cm'><p>Sample cm Data</p></a></li>
<li><a href='#cm-class'><p>Class to Store Current Meter Data</p></a></li>
<li><a href='#cnvName2oceName'><p>Infer Variable Name, Units and Scale From a Seabird Header</p></a></li>
<li><a href='#coastline-class'><p>Class to Store Coastline Data</p></a></li>
<li><a href='#coastlineBest'><p>Find the Name of the Best Coastline Object</p></a></li>
<li><a href='#coastlineCut'><p>Cut a Coastline Object at Specified Longitude</p></a></li>
<li><a href='#coastlineWorld'><p>Sample coastline Data (Global, at 1:110M scale)</p></a></li>
<li><a href='#colormap'><p>Calculate a Color Map</p></a></li>
<li><a href='#colormapGMT'><p>Create a GMT-type (CPT) Colormap</p></a></li>
<li><a href='#composite'><p>Create a Composite Object by Averaging Across Good Data</p></a></li>
<li><a href='#composite,amsr-method'><p>Create a Composite of amsr Satellite Data</p></a></li>
<li><a href='#composite,list-method'><p>Composite by Averaging Across Data</p></a></li>
<li><a href='#computableWaterProperties'><p>Determine Available Derived Water Properties</p></a></li>
<li><a href='#concatenate'><p>Concatenate oce Objects (Generic)</p></a></li>
<li><a href='#concatenate,adp-method'><p>Concatenate adp Objects</p></a></li>
<li><a href='#concatenate,list-method'><p>Concatenate a List of oce Objects</p></a></li>
<li><a href='#concatenate,oce-method'><p>Concatenate oce Objects (oce-Specific)</p></a></li>
<li><a href='#coriolis'><p>Coriolis Parameter on the Earth</p></a></li>
<li><a href='#ctd'><p>Sample ctd Data</p></a></li>
<li><a href='#ctd_aml.csv.gz'><p>Sample ctd File in aml Format</p></a></li>
<li><a href='#CTD_BCD2014666_008_1_DN.ODF.gz'><p>Sample ctd File in .odf Format</p></a></li>
<li><a href='#ctd-class'><p>Class to Store CTD (or general hydrographic) Data</p></a></li>
<li><a href='#ctd.cnv.gz'><p>Sample ctd File in .cnv Format</p></a></li>
<li><a href='#ctdDecimate'><p>Decimate a ctd Profile</p></a></li>
<li><a href='#ctdFindProfiles'><p>Find Profiles Within a Tow-Yow ctd Record</p></a></li>
<li><a href='#ctdFindProfilesRBR'><p>Find Profiles Within a ctd Object Read From a RBR File</p></a></li>
<li><a href='#ctdRaw'><p>Sample ctd Data, Not Trimmed of Extraneous Data</p></a></li>
<li><a href='#ctdRepair'><p>Repair a Malformed ctd Object</p></a></li>
<li><a href='#ctdTrim'><p>Trim Beginning and Ending of a CTD cast</p></a></li>
<li><a href='#ctimeToSeconds'><p>Interpret a Character String as a Time Interval</p></a></li>
<li><a href='#curl'><p>Curl of 2D Vector Field</p></a></li>
<li><a href='#d200321-001.ctd.gz'><p>Sample ctd File in .ctd Format</p></a></li>
<li><a href='#d201211_0011.cnv.gz'><p>Sample ctd File in .cnv Format</p></a></li>
<li><a href='#dataLabel'><p>Associate Data Names With Units</p></a></li>
<li><a href='#decimate'><p>Smooth and Decimate, or Subsample, an oce Object</p></a></li>
<li><a href='#decodeHeaderNortek'><p>Decode a Nortek Header</p></a></li>
<li><a href='#decodeTime'><p>Oce Version of as.POSIXct</p></a></li>
<li><a href='#defaultFlags'><p>Suggest a Default Flag Vector for Bad or Suspicious Data</p></a></li>
<li><a href='#despike'><p>Remove Spikes From a Time Series</p></a></li>
<li><a href='#detrend'><p>Detrend a Set of Observations</p></a></li>
<li><a href='#download.amsr'><p>Download and Cache an amsr File</p></a></li>
<li><a href='#download.coastline'><p>Download a coastline File</p></a></li>
<li><a href='#download.met'><p>Download and Cache a met File</p></a></li>
<li><a href='#download.topo'><p>Download and Cache a topo File</p></a></li>
<li><a href='#drawDirectionField'><p>Draw a Direction Field</p></a></li>
<li><a href='#drawIsopycnals'><p>Add Isopycnal Curves to a TS Plot</p></a></li>
<li><a href='#drawPalette'><p>Draw a Palette, Leaving Margins Suitable for an Accompanying Plot</p></a></li>
<li><a href='#echosounder'><p>Sample echosounder Data</p></a></li>
<li><a href='#echosounder-class'><p>Class to Store Echosounder Data</p></a></li>
<li><a href='#eclipticalToEquatorial'><p>Convert Ecliptical Coordinate to Equatorial Coordinate</p></a></li>
<li><a href='#enuToOther'><p>Rotate Acoustic-Doppler Data to a New Coordinate System</p></a></li>
<li><a href='#enuToOtherAdp'><p>Convert adp Object from ENU Coordinate to Rotated Coordinate</p></a></li>
<li><a href='#enuToOtherAdv'><p>Convert ENU to Other Coordinate</p></a></li>
<li><a href='#equatorialToLocalHorizontal'><p>Convert Equatorial Coordinate to Local Horizontal Coordinate</p></a></li>
<li><a href='#errorbars'><p>Draw Error Bars on an Existing xy Diagram</p></a></li>
<li><a href='#fillGap'><p>Fill a Gap in an oce Object</p></a></li>
<li><a href='#findBottom'><p>Find the Ocean Bottom in an Echosounder Object</p></a></li>
<li><a href='#firstFinite'><p>Get First Finite Value in a Vector or Array.</p></a></li>
<li><a href='#formatCI'><p>Format a Confidence Interval</p></a></li>
<li><a href='#formatPosition'><p>Format Geographical Position in Degrees and Minutes</p></a></li>
<li><a href='#fullFilename'><p>Full Name of File, Including Path</p></a></li>
<li><a href='#g1sst-class'><p>Class to Store G1SST Satellite/Model Data</p></a></li>
<li><a href='#gappyIndex'><p>Create a Possibly Gappy Indexing Vector</p></a></li>
<li><a href='#geodDist'><p>Compute Geodesic Distance on Surface of Earth</p></a></li>
<li><a href='#geodGc'><p>Great-circle Segments Between Points on Earth</p></a></li>
<li><a href='#geodXy'><p>Convert From Geographical to Geodesic Coordinates</p></a></li>
<li><a href='#geodXyInverse'><p>Inverse Geodesic Calculation</p></a></li>
<li><a href='#GMTOffsetFromTz'><p>Determine Time Offset From Timezone</p></a></li>
<li><a href='#gps-class'><p>Class to Store GPS Data</p></a></li>
<li><a href='#grad'><p>Calculate Matrix Gradient</p></a></li>
<li><a href='#gravity'><p>Acceleration Due to Earth Gravity</p></a></li>
<li><a href='#handleFlags'><p>Handle Flags in oce Objects (Generic)</p></a></li>
<li><a href='#handleFlags,adp-method'><p>Handle Flags in adp Objects</p></a></li>
<li><a href='#handleFlags,argo-method'><p>Handle Flags in argo Objects</p></a></li>
<li><a href='#handleFlags,ctd-method'><p>Handle Flags in ctd Objects</p></a></li>
<li><a href='#handleFlags,oce-method'><p>Handle Flags in oce Objects</p></a></li>
<li><a href='#handleFlags,section-method'><p>Handle flags in section Objects</p></a></li>
<li><a href='#handleFlags,vector-method'><p>Signal Erroneous Application to non-oce Objects</p></a></li>
<li><a href='#handleFlagsInternal'><p>Low-Level Function for Handling Data-Quality Flags</p></a></li>
<li><a href='#head.oce'><p>Extract The Start of an Oce Object</p></a></li>
<li><a href='#imagep'><p>Plot an Image with a Color Palette</p></a></li>
<li><a href='#initialize,ctd-method'><p>Initialize Storage for a ctd Object</p></a></li>
<li><a href='#initializeFlags'><p>Create and Initialize oce Flags</p></a></li>
<li><a href='#initializeFlags,adp-method'><p>Create and Initialize adp Flags</p></a></li>
<li><a href='#initializeFlags,oce-method'><p>Create and Initialize oce Flags</p></a></li>
<li><a href='#initializeFlagScheme'><p>Establish a Data-Quality Scheme for a oce Object</p></a></li>
<li><a href='#initializeFlagScheme,ctd-method'><p>Establish a Data-Quality Scheme for a ctd Object</p></a></li>
<li><a href='#initializeFlagScheme,oce-method'><p>Establish a Data-Quality Scheme for a oce Object</p></a></li>
<li><a href='#initializeFlagScheme,section-method'><p>Establish a Data-Quality Scheme for a section Object</p></a></li>
<li><a href='#initializeFlagSchemeInternal'><p>Establish a Data-Quality Scheme for a oce Object</p></a></li>
<li><a href='#initializeFlagsInternal'><p>Create and Initialize oce Flags</p></a></li>
<li><a href='#integerToAscii'><p>Infer ASCII Code From an Integer Value</p></a></li>
<li><a href='#integrateTrapezoid'><p>Trapezoidal Integration</p></a></li>
<li><a href='#interpBarnes'><p>Grid Data Using the Barnes Algorithm</p></a></li>
<li><a href='#is.ad2cp'><p>Test Whether Item is a ad2cp-Type adp Object</p></a></li>
<li><a href='#julianCenturyAnomaly'><p>Convert Julian-Day-Number to Julian Century</p></a></li>
<li><a href='#julianDay'><p>Convert a Time to a Julian Day</p></a></li>
<li><a href='#labelWithUnit'><p>Create Label With Unit</p></a></li>
<li><a href='#ladp-class'><p>Class to Store Lowered-adp Data</p></a></li>
<li><a href='#landsat'><p>Sample landsat Data</p></a></li>
<li><a href='#landsat-class'><p>Class to Store Landsat Satellite Data</p></a></li>
<li><a href='#landsatAdd'><p>Add a Band to a landsat Object</p></a></li>
<li><a href='#landsatTrim'><p>Trim a landsat Image to a Geographical Region</p></a></li>
<li><a href='#latFormat'><p>Format a Latitude</p></a></li>
<li><a href='#latlonFormat'><p>Format a Latitude-Longitude Pair</p></a></li>
<li><a href='#lisst'><p>Sample lisst Data</p></a></li>
<li><a href='#lisst-class'><p>Class to Store LISST Data</p></a></li>
<li><a href='#lobo'><p>Sample lobo Data</p></a></li>
<li><a href='#lobo-class'><p>Class to Store LOBO Data</p></a></li>
<li><a href='#locationForGsw'><p>Alter Longitude and Latitude for gsw Computations</p></a></li>
<li><a href='#lon360'><p>Change Longitude From -180:180 to 0:360 Convention</p></a></li>
<li><a href='#lonFormat'><p>Format a Longitude</p></a></li>
<li><a href='#longitudeTighten'><p>Try to Reduce Section Longitude Range</p></a></li>
<li><a href='#lonlat2map'><p>Convert Longitude and Latitude to X and Y</p></a></li>
<li><a href='#lonlat2utm'><p>Convert Longitude and Latitude to UTM</p></a></li>
<li><a href='#lookWithin'><p>Look Within the First Element of a List for Replacement Values</p></a></li>
<li><a href='#lowpass'><p>Lowpass Digital Filtering</p></a></li>
<li><a href='#magneticField'><p>Earth Magnetic Declination, Inclination, and Intensity</p></a></li>
<li><a href='#makeFilter'><p>Make a Digital Filter</p></a></li>
<li><a href='#map2lonlat'><p>Convert X and Y to Longitude and Latitude</p></a></li>
<li><a href='#mapArrows'><p>Add Arrows to a Map</p></a></li>
<li><a href='#mapAxis'><p>Add Axis Labels to an Existing Map</p></a></li>
<li><a href='#mapContour'><p>Add Contours on a Existing map</p></a></li>
<li><a href='#mapCoordinateSystem'><p>Draw a Coordinate System</p></a></li>
<li><a href='#mapDirectionField'><p>Add a Direction Field to an Existing Map</p></a></li>
<li><a href='#mapGrid'><p>Add a Longitude and Latitude Grid to an Existing Map</p></a></li>
<li><a href='#mapImage'><p>Add an Image to a Map</p></a></li>
<li><a href='#mapLines'><p>Add Lines to a Map</p></a></li>
<li><a href='#mapLocator'><p>Locate Points on a Map</p></a></li>
<li><a href='#mapLongitudeLatitudeXY'><p>Convert From Longitude and Latitude to X and Y</p></a></li>
<li><a href='#mapPlot'><p>Draw a Map</p></a></li>
<li><a href='#mapPoints'><p>Add Points to a Map</p></a></li>
<li><a href='#mapPolygon'><p>Add a Polygon to a Map</p></a></li>
<li><a href='#mapScalebar'><p>Add a Scalebar to a Map</p></a></li>
<li><a href='#mapText'><p>Add Text to a Map</p></a></li>
<li><a href='#mapTissot'><p>Add Tissot Indicatrices to a Map</p></a></li>
<li><a href='#matchBytes'><p>Locate Byte Sequences in a Raw Vector</p></a></li>
<li><a href='#matrixShiftLongitude'><p>Rearrange Areal Matrix so Greenwich is Near the Centre</p></a></li>
<li><a href='#matrixSmooth'><p>Smooth a Matrix</p></a></li>
<li><a href='#met'><p>Sample met Data</p></a></li>
<li><a href='#met-class'><p>Class to Store Meteorological Data</p></a></li>
<li><a href='#metNames2oceNames'><p>Convert met Data Name to oce Name</p></a></li>
<li><a href='#moonAngle'><p>Lunar Angle as Function of Space and Time</p></a></li>
<li><a href='#numberAsHMS'><p>Convert a Numeric Time to Hour, Minute, and Second</p></a></li>
<li><a href='#numberAsPOSIXct'><p>Convert a Numeric Time to a POSIXct Time</p></a></li>
<li><a href='#oce-class'><p>Base Class for oce Objects</p></a></li>
<li><a href='#oce-deprecated'><p>Deprecated and Defunct Elements of the oce Package</p></a></li>
<li><a href='#oce.as.raw'><p>Version of as.raw() That Clips Data</p></a></li>
<li><a href='#oce.axis.POSIXct'><p>Oce Version of axis.POSIXct</p></a></li>
<li><a href='#oce.contour'><p>Oce Variant of contour</p></a></li>
<li><a href='#oce.grid'><p>Add a Grid to an Existing Oce Plot</p></a></li>
<li><a href='#oce.plot.ts'><p>Oce Variant of plot.ts</p></a></li>
<li><a href='#oce.write.table'><p>Write the Data Portion of Object to a File</p></a></li>
<li><a href='#oceApprox'><p>Interpolate 1D Data with UNESCO or Reiniger-Ross Algorithm</p></a></li>
<li><a href='#oceAxis'><p>Draw an Axis, Possibly with Decade-style Logarithmic Scaling</p></a></li>
<li><a href='#ocecolors'><p>Data That Define Some Color Palettes</p></a></li>
<li><a href='#oceColors9B'><p>Create Colors in a Red-Yellow-Blue Color Scheme</p></a></li>
<li><a href='#oceColorsCDOM'><p>Create Colors Suitable for CDOM Fields</p></a></li>
<li><a href='#oceColorsChlorophyll'><p>Create Colors Suitable for chlorophyll Fields</p></a></li>
<li><a href='#oceColorsClosure'><p>Create Color Functions</p></a></li>
<li><a href='#oceColorsDensity'><p>Create Colors Suitable for density Fields</p></a></li>
<li><a href='#oceColorsFreesurface'><p>Create Colors Suitable for freesurface Fields</p></a></li>
<li><a href='#oceColorsGebco'><p>Create Colors in a GEBCO-like Scheme</p></a></li>
<li><a href='#oceColorsJet'><p>Create Colors Similar to the Matlab Jet Scheme</p></a></li>
<li><a href='#oceColorsOxygen'><p>Create Colors Suitable for oxygen Fields</p></a></li>
<li><a href='#oceColorsPalette'><p>Create a Vector of Colors</p></a></li>
<li><a href='#oceColorsPAR'><p>Create Colors Suitable for PAR Fields</p></a></li>
<li><a href='#oceColorsPhase'><p>Create Colors Suitable for phase Fields</p></a></li>
<li><a href='#oceColorsSalinity'><p>Create Colors Suitable for salinity Fields</p></a></li>
<li><a href='#oceColorsTemperature'><p>Create Colors Suitable for temperature Fields</p></a></li>
<li><a href='#oceColorsTurbidity'><p>Create Colors Suitable for turbidity Fields</p></a></li>
<li><a href='#oceColorsTurbo'><p>Create Colors Similar to the Google Turbo Scheme</p></a></li>
<li><a href='#oceColorsTwo'><p>Create Two-Color Palette</p></a></li>
<li><a href='#oceColorsVelocity'><p>Create Colors Suitable for velocity Fields</p></a></li>
<li><a href='#oceColorsViridis'><p>Create Colors Similar to the Matlab Viridis Scheme</p></a></li>
<li><a href='#oceColorsVorticity'><p>Create Colors Suitable for vorticity Fields</p></a></li>
<li><a href='#oceConvolve'><p>Convolve Two Time Series</p></a></li>
<li><a href='#oceCRS'><p>Coordinate Reference System Strings for Some Oceans</p></a></li>
<li><a href='#oceDebug'><p>Print a Debugging Message</p></a></li>
<li><a href='#oceDeleteData'><p>Delete Something From the data Slot of an oce Object</p></a></li>
<li><a href='#oceDeleteMetadata'><p>Delete Something in an oce metadata Slot</p></a></li>
<li><a href='#oceEdit'><p>Edit an Oce Object</p></a></li>
<li><a href='#oceFileTrim'><p>Trim an oce File</p></a></li>
<li><a href='#oceFilter'><p>Filter a Time Series</p></a></li>
<li><a href='#oceGetData'><p>Extract Something From the data Slot of an oce Object</p></a></li>
<li><a href='#oceGetMetadata'><p>Extract Something From the metadata Slot of an oce Object</p></a></li>
<li><a href='#oceMagic'><p>Find the Type of an Oceanographic Data File</p></a></li>
<li><a href='#oceNames2whpNames'><p>Translate Oce Data Names to WHP Data Names</p></a></li>
<li><a href='#ocePmatch'><p>Partial Matching of Strings or Numbers</p></a></li>
<li><a href='#oceProject'><p>Wrapper to sf::sf_project()</p></a></li>
<li><a href='#oceRenameData'><p>Rename Something in the data slot of an oce Object</p></a></li>
<li><a href='#oceRenameMetadata'><p>Rename Something in the metadata Slot of an oce Object</p></a></li>
<li><a href='#oceSetData'><p>Set Something in the data Slot of an oce Object</p></a></li>
<li><a href='#oceSetMetadata'><p>Set Something in the metadata Slot of an oce Object</p></a></li>
<li><a href='#oceSmooth'><p>Smooth an oce Object</p></a></li>
<li><a href='#oceSpectrum'><p>Normalize a Spectrum</p></a></li>
<li><a href='#oceUnits2whpUnits'><p>Translate oce Unit to WHP Unit</p></a></li>
<li><a href='#odf-class'><p>Class to Store ODF Data</p></a></li>
<li><a href='#ODF2oce'><p>Create ODF Object From Output of read_ODF in ODF package</p></a></li>
<li><a href='#ODFListFromHeader'><p>Create a List of odf Header Metadata</p></a></li>
<li><a href='#ODFNames2oceNames'><p>Translate ODF CODE Strings to oce Variable Names</p></a></li>
<li><a href='#parseLatLon'><p>Parse a Latitude or Longitude String</p></a></li>
<li><a href='#plot,adp-method'><p>Plot an adp Object</p></a></li>
<li><a href='#plot,adv-method'><p>Plot an adv Object</p></a></li>
<li><a href='#plot,amsr-method'><p>Plot an amsr Object</p></a></li>
<li><a href='#plot,argo-method'><p>Plot an argo Object</p></a></li>
<li><a href='#plot,bremen-method'><p>Plot a bremen Object</p></a></li>
<li><a href='#plot,cm-method'><p>Plot a cm Object</p></a></li>
<li><a href='#plot,coastline-method'><p>Plot a coastline Object</p></a></li>
<li><a href='#plot,ctd-method'><p>Plot a ctd Object</p></a></li>
<li><a href='#plot,echosounder-method'><p>Plot an echosounder Object</p></a></li>
<li><a href='#plot,gps-method'><p>Plot a gps Object</p></a></li>
<li><a href='#plot,ladp-method'><p>Plot an ladp Object</p></a></li>
<li><a href='#plot,landsat-method'><p>Plot a landsat Object</p></a></li>
<li><a href='#plot,lisst-method'><p>Plot a lisst Object</p></a></li>
<li><a href='#plot,lobo-method'><p>Plot a lobo object</p></a></li>
<li><a href='#plot,met-method'><p>Plot a met Object</p></a></li>
<li><a href='#plot,oce-method'><p>Plot an oce Object</p></a></li>
<li><a href='#plot,odf-method'><p>Plot an odf Object</p></a></li>
<li><a href='#plot,rsk-method'><p>Plot a rsk Object</p></a></li>
<li><a href='#plot,satellite-method'><p>Plot a satellite Object</p></a></li>
<li><a href='#plot,sealevel-method'><p>Plot a sealevel Object</p></a></li>
<li><a href='#plot,section-method'><p>Plot a section Object</p></a></li>
<li><a href='#plot,tidem-method'><p>Plot a tidem Object</p></a></li>
<li><a href='#plot,topo-method'><p>Plot a topo Object</p></a></li>
<li><a href='#plot,windrose-method'><p>Plot a windrose Object</p></a></li>
<li><a href='#plot,xbt-method'><p>Plot an xbt Object</p></a></li>
<li><a href='#plotInset'><p>Plot an Inset Diagram</p></a></li>
<li><a href='#plotPolar'><p>Draw a Polar Plot</p></a></li>
<li><a href='#plotProfile'><p>Plot a ctd Profile</p></a></li>
<li><a href='#plotScan'><p>Plot a ctd Object  in a Low-Level Fashion</p></a></li>
<li><a href='#plotSticks'><p>Draw a Stick Plot</p></a></li>
<li><a href='#plotTaylor'><p>Plot a Model-data Comparison Diagram</p></a></li>
<li><a href='#plotTS'><p>Plot Temperature-Salinity Diagram</p></a></li>
<li><a href='#predict.tidem'><p>Predict a Tidal Signal</p></a></li>
<li><a href='#preferAdjusted'><p>Set Preference for Adjusted Values</p></a></li>
<li><a href='#presentTime'><p>Get the Present Time, in a Stated Timezone</p></a></li>
<li><a href='#prettyPosition'><p>Pretty Longitude/Latitude in Degree-Minute-Second Format</p></a></li>
<li><a href='#processingLog&lt;-'><p>Add an Item to a Processing Log</p></a></li>
<li><a href='#processingLogAppend'><p>Append an Item to a Processing Log</p></a></li>
<li><a href='#processingLogItem'><p>Create an Item That can be Inserted into a Processing Log</p></a></li>
<li><a href='#processingLogShow'><p>Show the Processing Log of an oce Object</p></a></li>
<li><a href='#pwelch'><p>Welch Periodogram</p></a></li>
<li><a href='#rangeExtended'><p>Calculate Range, Extended a Little, as is Done for Axes</p></a></li>
<li><a href='#rangeLimit'><p>Substitute NA for Data Outside a Range</p></a></li>
<li><a href='#read.adp'><p>Read an adp File</p></a></li>
<li><a href='#read.adp.ad2cp'><p>Read an adp File in Nortek AD2CP Format</p></a></li>
<li><a href='#read.adp.nortek'><p>Read an adp File in Nortek Format</p></a></li>
<li><a href='#read.adp.rdi'><p>Read an adp File in Teledyne/RDI Format</p></a></li>
<li><a href='#read.adp.sontek'><p>Read an adp File in Sontek Format</p></a></li>
<li><a href='#read.adp.sontek.serial'><p>Read an adp File in Serial Sontek Format</p></a></li>
<li><a href='#read.adv'><p>Read an adv File</p></a></li>
<li><a href='#read.adv.nortek'><p>Read an adv File</p></a></li>
<li><a href='#read.adv.sontek.adr'><p>Read an adv File</p></a></li>
<li><a href='#read.adv.sontek.serial'><p>Read an adv File</p></a></li>
<li><a href='#read.adv.sontek.text'><p>Read an adv File</p></a></li>
<li><a href='#read.amsr'><p>Read an amsr File</p></a></li>
<li><a href='#read.aquadopp'><p>Read an adp File in Nortek Aquadopp Format</p></a></li>
<li><a href='#read.aquadoppHR'><p>Read Nortek Aquadopp-HR File</p></a></li>
<li><a href='#read.aquadoppProfiler'><p>Read an adp File in Nortek Aquadopp Format</p></a></li>
<li><a href='#read.argo'><p>Read an Argo Data File</p></a></li>
<li><a href='#read.argo.copernicus'><p>Read an argo File in Copernicus Format</p></a></li>
<li><a href='#read.bremen'><p>Read a bremen File</p></a></li>
<li><a href='#read.cm'><p>Read a cm File</p></a></li>
<li><a href='#read.coastline'><p>Read a coastline File</p></a></li>
<li><a href='#read.coastline.openstreetmap'><p>Read a coastline File in Openstreetmap Format</p></a></li>
<li><a href='#read.coastline.shapefile'><p>Read a coastline File in Shapefile Format</p></a></li>
<li><a href='#read.ctd'><p>Read a ctd File in General Format</p></a></li>
<li><a href='#read.ctd.aml'><p>Read a ctd File in AML Format</p></a></li>
<li><a href='#read.ctd.itp'><p>Read a ctd File in ITP Format</p></a></li>
<li><a href='#read.ctd.odf'><p>Read a ctd File in odf Format</p></a></li>
<li><a href='#read.ctd.odv'><p>Read a &quot;ctd&quot; File in ODV Format</p></a></li>
<li><a href='#read.ctd.saiv'><p>Read a ctd File in SAIV Format</p></a></li>
<li><a href='#read.ctd.sbe'><p>Read a ctd File in Seabird Format</p></a></li>
<li><a href='#read.ctd.ssda'><p>Read a ctd File in SSDA Format</p></a></li>
<li><a href='#read.ctd.woce'><p>Read a ctd File in WOCE-Exchange Format</p></a></li>
<li><a href='#read.ctd.woce.other'><p>Read a ctd File in WOCE-Exchange EXPOCODE Format</p></a></li>
<li><a href='#read.echosounder'><p>Read an echosounder File</p></a></li>
<li><a href='#read.g1sst'><p>Read a g1sst File</p></a></li>
<li><a href='#read.gps'><p>Read a gps File</p></a></li>
<li><a href='#read.index'><p>Read a NOAA Ocean Index File</p></a></li>
<li><a href='#read.landsat'><p>Read a landsat File Directory</p></a></li>
<li><a href='#read.lisst'><p>Read a lisst File</p></a></li>
<li><a href='#read.lobo'><p>Read a lobo File</p></a></li>
<li><a href='#read.met'><p>Read a met File</p></a></li>
<li><a href='#read.netcdf'><p>Read a NetCDF File</p></a></li>
<li><a href='#read.oce'><p>Read an Oceanographic Data File</p></a></li>
<li><a href='#read.odf'><p>Read an odf File</p></a></li>
<li><a href='#read.rsk'><p>Read a rsk File</p></a></li>
<li><a href='#read.sealevel'><p>Read a sealevel File</p></a></li>
<li><a href='#read.section'><p>Read a section File</p></a></li>
<li><a href='#read.topo'><p>Read a topo File</p></a></li>
<li><a href='#read.woa'><p>Read a World Ocean Atlas NetCDF File</p></a></li>
<li><a href='#read.xbt'><p>Read an xbt file</p></a></li>
<li><a href='#read.xbt.edf'><p>Read an xbt File in Sippican Format</p></a></li>
<li><a href='#read.xbt.noaa1'><p>Read an xbt File in NOAA Format</p></a></li>
<li><a href='#rescale'><p>Rescale Values to lie in a Given Range</p></a></li>
<li><a href='#resizableLabel'><p>Variable Names in Adjustable Sizes</p></a></li>
<li><a href='#retime'><p>Adjust The Time Within an oce Object</p></a></li>
<li><a href='#rotateAboutZ'><p>Rotate Velocity Components Within an oce Object</p></a></li>
<li><a href='#rsk'><p>Sample rsk Data</p></a></li>
<li><a href='#rsk-class'><p>Class to Store Rsk Data</p></a></li>
<li><a href='#rsk2ctd'><p>Create a ctd Object from an rsk Object</p></a></li>
<li><a href='#rskPatm'><p>Estimate Atmospheric Pressure in an rsk Object</p></a></li>
<li><a href='#rskToc'><p>Decode Table-of-Contents From an rsk File</p></a></li>
<li><a href='#runlm'><p>Calculate Running Linear Models</p></a></li>
<li><a href='#satellite-class'><p>Class to Store Satellite Data</p></a></li>
<li><a href='#sealevel'><p>Sample sealevel Data (Halifax Harbour)</p></a></li>
<li><a href='#sealevel-class'><p>Class to Store Sealevel Data</p></a></li>
<li><a href='#sealevelTuktoyaktuk'><p>Sample sealevel Data (Tuktoyaktuk)</p></a></li>
<li><a href='#secondsToCtime'><p>Express Time Interval as Colon-Separated String</p></a></li>
<li><a href='#section'><p>Sample section Data</p></a></li>
<li><a href='#section-class'><p>Class to Store Hydrographic Section Data</p></a></li>
<li><a href='#sectionAddStation'><p>Add a ctd Profile to a section Object</p></a></li>
<li><a href='#sectionGrid'><p>Grid a Section in Pressure Space</p></a></li>
<li><a href='#sectionSmooth'><p>Smooth a Section</p></a></li>
<li><a href='#sectionSort'><p>Sort a Section</p></a></li>
<li><a href='#setFlags'><p>Set Data-Quality Flags within a oce Object</p></a></li>
<li><a href='#setFlags,adp-method'><p>Set Data-Quality Flags within a adp Object</p></a></li>
<li><a href='#setFlags,ctd-method'><p>Set Data-Quality Flags within a ctd Object</p></a></li>
<li><a href='#setFlags,oce-method'><p>Set Data-Quality Flags within a oce Object</p></a></li>
<li><a href='#shiftLongitude'><p>Shift Longitude to Range -180 to 180</p></a></li>
<li><a href='#showMetadataItem'><p>Show an Item in the metadata Slot of an oce Object</p></a></li>
<li><a href='#siderealTime'><p>Convert From POSIXt Time to Sidereal Time</p></a></li>
<li><a href='#snakeToCamel'><p>Convert From Snake-Case to Camel-Case Notation</p></a></li>
<li><a href='#standardDepths'><p>Standard Oceanographic Depths</p></a></li>
<li><a href='#standardizeLongitude'><p>Put Longitude in the Range From -180 to 180</p></a></li>
<li><a href='#subset,adp-method'><p>Subset an adp Object</p></a></li>
<li><a href='#subset,adv-method'><p>Subset an adv Object</p></a></li>
<li><a href='#subset,amsr-method'><p>Subset an amsr Object</p></a></li>
<li><a href='#subset,argo-method'><p>Subset an argo Object</p></a></li>
<li><a href='#subset,cm-method'><p>Subset a cm Object</p></a></li>
<li><a href='#subset,coastline-method'><p>Subset a coastline Object</p></a></li>
<li><a href='#subset,ctd-method'><p>Subset a ctd Object</p></a></li>
<li><a href='#subset,echosounder-method'><p>Subset an echosounder Object</p></a></li>
<li><a href='#subset,lobo-method'><p>Subset a lobo Object</p></a></li>
<li><a href='#subset,met-method'><p>Subset a met Object</p></a></li>
<li><a href='#subset,oce-method'><p>Subset an oce Object</p></a></li>
<li><a href='#subset,odf-method'><p>Subset an odf Object</p></a></li>
<li><a href='#subset,rsk-method'><p>Subset a rsk Object</p></a></li>
<li><a href='#subset,sealevel-method'><p>Subset a sealevel Object</p></a></li>
<li><a href='#subset,section-method'><p>Subset a section Object</p></a></li>
<li><a href='#subset,topo-method'><p>Subset a topo Object</p></a></li>
<li><a href='#subset,xbt-method'><p>Subset an xbt Object</p></a></li>
<li><a href='#subtractBottomVelocity'><p>Subtract Bottom Velocity From an adp Object</p></a></li>
<li><a href='#summary,adp-method'><p>Summarize an adp Object</p></a></li>
<li><a href='#summary,adv-method'><p>Summarize an adv Object</p></a></li>
<li><a href='#summary,amsr-method'><p>Summarize an amsr Object</p></a></li>
<li><a href='#summary,argo-method'><p>Summarize an argo Object</p></a></li>
<li><a href='#summary,bremen-method'><p>Summarize a bremen Object</p></a></li>
<li><a href='#summary,cm-method'><p>Summarize a cm Object</p></a></li>
<li><a href='#summary,coastline-method'><p>Summarize a coastline Object</p></a></li>
<li><a href='#summary,ctd-method'><p>Summarize a ctd Object</p></a></li>
<li><a href='#summary,echosounder-method'><p>Summarize an echosounder Object</p></a></li>
<li><a href='#summary,gps-method'><p>Summarize a gps Object</p></a></li>
<li><a href='#summary,ladp-method'><p>Summarize an ladp Object</p></a></li>
<li><a href='#summary,landsat-method'><p>Summarize a landsat Object</p></a></li>
<li><a href='#summary,lisst-method'><p>Summarize a lisst Object</p></a></li>
<li><a href='#summary,lobo-method'><p>Summarize a lobo Object</p></a></li>
<li><a href='#summary,met-method'><p>Summarize a met Object</p></a></li>
<li><a href='#summary,oce-method'><p>Summarize an oce Object</p></a></li>
<li><a href='#summary,odf-method'><p>Summarize an odf Object</p></a></li>
<li><a href='#summary,rsk-method'><p>Summarize a rsk Object</p></a></li>
<li><a href='#summary,satellite-method'><p>Summarize a satellite Object</p></a></li>
<li><a href='#summary,sealevel-method'><p>Summarize a sealevel Object</p></a></li>
<li><a href='#summary,section-method'><p>Summarize a section Object</p></a></li>
<li><a href='#summary,tidem-method'><p>Summarize a tidem Object</p></a></li>
<li><a href='#summary,topo-method'><p>Summarize a topo Object</p></a></li>
<li><a href='#summary,windrose-method'><p>Summarize a windrose Object</p></a></li>
<li><a href='#summary,xbt-method'><p>Summarize an xbt Object</p></a></li>
<li><a href='#sunAngle'><p>Solar Angle as Function of Space and Time</p></a></li>
<li><a href='#sunDeclinationRightAscension'><p>Sun Declination and Right Ascension</p></a></li>
<li><a href='#swAbsoluteSalinity'><p>Seawater Absolute Salinity (GSW Formulation)</p></a></li>
<li><a href='#swAlpha'><p>Seawater Thermal Expansion Coefficient</p></a></li>
<li><a href='#swAlphaOverBeta'><p>Ratio of Seawater Thermal Expansion Coefficient to Haline Contraction Coefficient</p></a></li>
<li><a href='#swBeta'><p>Seawater Haline Contraction Coefficient</p></a></li>
<li><a href='#swConservativeTemperature'><p>Seawater Conservative Temperature (GSW Formulation)</p></a></li>
<li><a href='#swCSTp'><p>Electrical Conductivity Ratio From Salinity, Temperature and Pressure</p></a></li>
<li><a href='#swDepth'><p>Water Depth</p></a></li>
<li><a href='#swDynamicHeight'><p>Dynamic Height of a Seawater Profile</p></a></li>
<li><a href='#swLapseRate'><p>Seawater Lapse Rate</p></a></li>
<li><a href='#swN2'><p>Squared Buoyancy Frequency for Seawater</p></a></li>
<li><a href='#swPressure'><p>Water Pressure</p></a></li>
<li><a href='#swRho'><p>Seawater Density</p></a></li>
<li><a href='#swRrho'><p>Density Ratio</p></a></li>
<li><a href='#swSCTp'><p>Practical Salinity From Electrical Conductivity, Temperature and Pressure</p></a></li>
<li><a href='#swSigma'><p>Seawater Density Anomaly</p></a></li>
<li><a href='#swSigma0'><p>Seawater Potential Density Anomaly Referenced to Surface Pressure</p></a></li>
<li><a href='#swSigma1'><p>Seawater Potential Density Anomaly Referenced to 1000db Pressure</p></a></li>
<li><a href='#swSigma2'><p>Seawater Potential Density Anomaly Referenced to 2000db Pressure</p></a></li>
<li><a href='#swSigma3'><p>Seawater Potential Density Anomaly Referenced to 3000db Pressure</p></a></li>
<li><a href='#swSigma4'><p>Seawater Potential Density Anomaly Referenced to 4000db Pressure</p></a></li>
<li><a href='#swSigmaT'><p>Seawater Quasi-Potential Density Anomaly</p></a></li>
<li><a href='#swSigmaTheta'><p>Seawater Potential Density Anomaly</p></a></li>
<li><a href='#swSoundAbsorption'><p>Seawater Sound Absorption</p></a></li>
<li><a href='#swSoundSpeed'><p>Seawater Sound Speed</p></a></li>
<li><a href='#swSpecificHeat'><p>Seawater Specific Heat</p></a></li>
<li><a href='#swSpice'><p>Seawater Spiciness</p></a></li>
<li><a href='#swSR'><p>Seawater Reference Salinity (GSW Formulation)</p></a></li>
<li><a href='#swSstar'><p>Seawater Preformed Salinity (GSW Formulation)</p></a></li>
<li><a href='#swSTrho'><p>Seawater Salinity From Temperature and Density</p></a></li>
<li><a href='#swTFreeze'><p>Seawater Freezing Temperature</p></a></li>
<li><a href='#swThermalConductivity'><p>Seawater Thermal Conductivity</p></a></li>
<li><a href='#swTheta'><p>Seawater Potential Temperature (UNESCO Version)</p></a></li>
<li><a href='#swTSrho'><p>Seawater Temperature from Salinity and Density</p></a></li>
<li><a href='#swViscosity'><p>Seawater Viscosity</p></a></li>
<li><a href='#swZ'><p>Vertical Coordinate</p></a></li>
<li><a href='#T68fromT90'><p>Convert From ITS-90 to IPTS-68 Temperature</p></a></li>
<li><a href='#T90fromT48'><p>Convert From ITS-48 to ITS-90 Temperature</p></a></li>
<li><a href='#T90fromT68'><p>Convert From IPTS-68 to ITS-90 Temperature</p></a></li>
<li><a href='#tail.oce'><p>Extract the End of an Oce Object</p></a></li>
<li><a href='#threenum'><p>Calculate Minimum, Mean, and Maximum Values</p></a></li>
<li><a href='#tidalCurrent'><p>Tidal Current Dataset</p></a></li>
<li><a href='#tidedata'><p>Tidal Constituent Information</p></a></li>
<li><a href='#tidem'><p>Fit a Tidal Model to a Timeseries</p></a></li>
<li><a href='#tidem-class'><p>Class to Store Tidal Models</p></a></li>
<li><a href='#tidemAstron'><p>Astronomical Calculations for tidem</p></a></li>
<li><a href='#tidemConstituentNameFix'><p>Change Tidal Constituent Name from T-TIDE to Foreman Convention</p></a></li>
<li><a href='#tidemVuf'><p>Nodal Modulation Calculations for Tidal Analyses</p></a></li>
<li><a href='#timeToArgoJuld'><p>Convert Time to Argo Julian Day (juld)</p></a></li>
<li><a href='#titleCase'><p>Capitalize First Letter of Each of a Vector of Words</p></a></li>
<li><a href='#toEnu'><p>Rotate Acoustic-Doppler Data to the ENU Coordinate System</p></a></li>
<li><a href='#toEnuAdp'><p>Convert an adp Object to ENU Coordinates</p></a></li>
<li><a href='#toEnuAdv'><p>Convert an adv Object to ENU Coordinates</p></a></li>
<li><a href='#topo-class'><p>Class to Store Topographic Data</p></a></li>
<li><a href='#topoInterpolate'><p>Interpolate Within a topo Object</p></a></li>
<li><a href='#topoWorld'><p>Global Topographic Data (at Half-degree Resolution)</p></a></li>
<li><a href='#unabbreviateYear'><p>Determine Year From Various Abbreviations</p></a></li>
<li><a href='#undriftTime'><p>Correct for Drift in an Instrument Clock</p></a></li>
<li><a href='#unduplicateNames'><p>Rename Duplicated Character Strings</p></a></li>
<li><a href='#ungrid'><p>Extract (x, y, z) From (x, y, grid)</p></a></li>
<li><a href='#unitFromString'><p>Decode Units From Strings</p></a></li>
<li><a href='#unitFromStringRsk'><p>Infer rsk Units From a Vector of Strings</p></a></li>
<li><a href='#unwrapAngle'><p>Unwrap an Angle That Suffers Modulo-360 Problems</p></a></li>
<li><a href='#useHeading'><p>Replace the Heading for One Instrument With That of Another</p></a></li>
<li><a href='#usrLonLat'><p>Calculate Geographic Coordinates of Plot Box</p></a></li>
<li><a href='#utm2lonlat'><p>Convert UTM to Longitude and Latitude</p></a></li>
<li><a href='#vectorShow'><p>Show Some Values From a List, Vector or Matrix</p></a></li>
<li><a href='#velocityStatistics'><p>Report Statistics of adp or adv Velocities</p></a></li>
<li><a href='#webtide'><p>Get a Tidal Prediction From a WebTide Database</p></a></li>
<li><a href='#wind'><p>Sample Wind Data</p></a></li>
<li><a href='#window.oce'><p>Window an oce Object by Time or Distance</p></a></li>
<li><a href='#windrose-class'><p>Class to Store windrose Data</p></a></li>
<li><a href='#woceNames2oceNames'><p>Translate WOCE Data Names to Oce Data Names</p></a></li>
<li><a href='#woceUnit2oceUnit'><p>Translate WOCE Units to oce Units</p></a></li>
<li><a href='#write.ctd'><p>Save a ctd Object in a CSV File</p></a></li>
<li><a href='#xbt'><p>Sample xbt Data</p></a></li>
<li><a href='#xbt-class'><p>Class to Store XBT (Expendable Bathythermograph) Data</p></a></li>
<li><a href='#xbt.edf'><p>Sample xbt File in .edf Format</p></a></li>
<li><a href='#xyzToEnu'><p>Convert Acoustic-Doppler Data From XYZ to ENU Coordinates</p></a></li>
<li><a href='#xyzToEnuAdp'><p>Convert adp Object From XYZ to ENU Coordinates</p></a></li>
<li><a href='#xyzToEnuAdpAD2CP'><p>Convert adp Object of AD2CP type From XYZ to ENU Coordinates</p></a></li>
<li><a href='#xyzToEnuAdv'><p>Convert an adv Object From XYZ to ENU Coordinates</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Analysis of Oceanographic Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.8-2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dan Kelley &lt;Dan.Kelley@Dal.Ca&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15), gsw, methods, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>automap, DBI, foreign, interp, knitr, lubridate, ncdf4,
ocedata, rmarkdown, RSQLite, R.utils, sf, terra, testthat (&ge;
3.0.0), tiff, XML</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/dankelley/oce/issues">https://github.com/dankelley/oce/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Supports the analysis of Oceanographic data, including 'ADCP'
    measurements, measurements made with 'argo' floats, 'CTD' measurements,
    sectional data, sea-level time series, coastline and topographic data, etc.
    Provides specialized functions for calculating seawater properties such as
    potential temperature in either the 'UNESCO' or 'TEOS-10' equation of state.
    Produces graphical displays that conform to the conventions of the
    Oceanographic literature. This package is discussed extensively by
    Kelley (2018) "Oceanographic Analysis with R" &lt;<a href="https://doi.org/10.1007%2F978-1-4939-8844-0">doi:10.1007/978-1-4939-8844-0</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://dankelley.github.io/oce/">https://dankelley.github.io/oce/</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>BuildVignettes:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-01 19:20:04 UTC; kelley</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Author:</td>
<td>Dan Kelley <a href="https://orcid.org/0000-0001-7808-5911"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Clark Richards <a href="https://orcid.org/0000-0002-7833-206X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Chantelle Layton <a href="https://orcid.org/0000-0002-3199-5763"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb] (curl() coauthor),
  British Geological Survey [ctb, cph] (magnetic-field subroutine)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-02 00:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='oce'>A Package for Oceanographic Analysis</h2><span id='topic+oce'></span><span id='topic+oce-package'></span>

<h3>Description</h3>

<p>The oce package provides functions for working with
Oceanographic data, for calculations that are specific
to Oceanography, and for producing graphics that
match the conventions of the field.
</p>


<h3>Details</h3>

<p>Over a dozen specialized data types are handled by oce,
with generic plots and summaries for each, along with
the specialized functions needed for typical Oceanographic
analysis.
</p>
<p>See <a href="#topic+oce-class">oce</a> for a summary of the class structure
and links to documentation for the many subclasses of
oce objects, each aligned with a class of instrument or
or type of dataset.  For a more task-oriented approach,
see the several vignettes that are provided with oce,
and a book
(Kelley, Dan E. Oceanographic Analysis with R. New York: Springer-Verlag, 2018.
https://link.springer.com/book/10.1007/978-1-4939-8844-0) written
by one of the oce co-authors.
</p>


<h3>Specialized Functions</h3>

<p>A key function is <code><a href="#topic+read.oce">read.oce()</a></code>, which will attempt
to read Oceanographic data in raw format. This uses
<code><a href="#topic+oceMagic">oceMagic()</a></code> to try to detect the file type,
based on the file name and contents. If this detection
is not possible, users will need to go beyond <code><a href="#topic+read.oce">read.oce()</a></code>,
using a more specialized function, e.g. <code><a href="#topic+read.ctd">read.ctd()</a></code> for CTD files,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> for Teledyne-Seabird files, etc.
</p>


<h3>Generic Methods</h3>

<p>A list of the generic methods in oce is provided by
<a href="utils.html#topic+methods">methods</a><code>(class="oce")</code>; a few that are used frequently
are as follows.
</p>

<ul>
<li> <p><code>[[</code> Finds the value of an item in the object's
<code>metadata</code> or <code>data</code> slot. If the item does
not exist, but can be calculated from the other items,
then the calculated value is returned. As an example of the
latter, consider the built-in <code>ctd</code> dataset, which does
not contain potential temperature, &quot;<code>theta</code>&quot;. Using
<code>ctd[["theta"]]</code> therefore causes <code><a href="#topic+swTheta">swTheta()</a></code>
to be called, to calculate <code>theta</code>.
See <a href="#topic++5B+5B+2Coce-method">[[,oce-method</a> or type <code>?"[[,oce-method"</code>
to learn more about general functioning, or a specialized
method like <a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a> for CTD data, etc.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;[[&lt;-&#8288;</code> Alters the named item in the object's <code>metadata</code> or
<code>data</code> slot.  If the item does not exist, it is created.
See <a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a> or type <code>?"[[&lt;-,oce-method"</code>
to learn more about the general methodology, or a specialized
method like <a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a> for CTD data, etc.
</p>
</li>
<li> <p><code>summary()</code> Displays some information about the object named as an
argument, including a few elements from its <code>metadata</code> slot
and some statistics of the contents of its <code>data</code> slot.
See <a href="#topic+summary+2Coce-method">summary,oce-method</a> or type <code>?"summary,oce-method"</code>
to learn more about general functioning, or a specialized
method like <a href="#topic+summary+2Cctd-method">summary,ctd-method</a> for CTD data, etc.
</p>
</li>
<li> <p><code>subset()</code> Takes a subset of an oce object.
See <a href="#topic+subset+2Coce-method">subset,oce-method</a> or type <code>?"subset,oce-method"</code>
to learn more about general functioning, or a specialized
method like <a href="#topic+subset+2Cctd-method">subset,ctd-method</a> for CTD data, etc.
</p>
</li></ul>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Dan Kelley <a href="mailto:Dan.Kelley@Dal.Ca">Dan.Kelley@Dal.Ca</a> (<a href="https://orcid.org/0000-0001-7808-5911">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Clark Richards <a href="mailto:clark.richards@gmail.com">clark.richards@gmail.com</a> (<a href="https://orcid.org/0000-0002-7833-206X">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Chantelle Layton <a href="mailto:chantelle.layton@dal.ca">chantelle.layton@dal.ca</a> (<a href="https://orcid.org/0000-0002-3199-5763">ORCID</a>) (curl() coauthor) [contributor]
</p>
</li>
<li><p> British Geological Survey (magnetic-field subroutine) [contributor, copyright holder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://dankelley.github.io/oce/">https://dankelley.github.io/oce/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/dankelley/oce/issues">https://github.com/dankelley/oce/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+5B+5B+2Cadp-method'>Extract Something From an adp Object</h2><span id='topic++5B+5B+2Cadp-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adp'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cadp-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cadp-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cadp-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cadp-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>

<p>Note that the entries within <a href="#topic+adp-class">adp</a> objects vary greatly, from
instrument to instrument, and so are only sketched here, and in the output
from <code style="white-space: pre;">&#8288;[["?"]]&#8288;</code>.
</p>

<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are <em>not</em> authoritative, because
information provided by different instruments is so varied.
</p>
</li>
<li><p> If <code>i</code> is <code>"u1"</code> then the return value is <code>v[,1]</code>. The
same holds for 2, etc., depending on the number of beams in
the instrument.
</p>
</li>
<li><p> If <code>i</code> is <code>"a1"</code> then signal amplitude is returned, and similarly
for other digits. The results can be in <code><a href="base.html#topic+raw">raw()</a></code> or numeric form,
as shown in the examples.
</p>
</li>
<li><p> If <code>i</code> is <code>"q1"</code> then signal quality is returned, and similarly
for other digits.  As with amplitude, the result can be in <code><a href="base.html#topic+raw">raw()</a></code>
or numeric form.
</p>
</li>
<li><p> If <code>i</code> is <code>"coordinate"</code>, then the coordinate system is
retrieved.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(adp)
# Tests for beam 1, distance bin 1, first 5 observation times
adp[["v"]][1:5, 1, 1]
adp[["a"]][1:5, 1, 1]
adp[["a", "numeric"]][1:5, 1, 1]
as.numeric(adp[["a"]][1:5, 1, 1]) # same as above

</code></pre>

<hr>
<h2 id='+5B+5B+2Cadv-method'>Extract Something from an adv Object</h2><span id='topic++5B+5B+2Cadv-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adv'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cadv-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cadv-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cadv-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cadv-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively, while <code>dataDerived</code> and
<code>metadataDerived</code> hold the names of related
things that can be derived from the object's contents.
</p>
</li>
<li><p> If <code>i</code> is <code>"u1"</code> then the return value is <code>v[,1]</code>,
and similarly for <code>"u2"</code> and <code>"u3"</code>.
</p>
</li>
<li><p> If <code>i</code> is <code>"a1"</code> then signal amplitude is returned, and similarly
for <code>"a2"</code> and <code>"a3"</code>. The results can be in <code><a href="base.html#topic+raw">raw()</a></code> or numeric form,
as illustrated in the &ldquo;Examples&rdquo;.
</p>
</li>
<li><p> If <code>i</code> is <code>"q1"</code> then signal quality is returned, and similarly
for <code>"q2"</code> and <code>"q3"</code>.  As with amplitude, the result can be in <code><a href="base.html#topic+raw">raw()</a></code>
or numeric form.
</p>
</li>
<li><p> If <code>i</code> is <code>"heading"</code>, <code>"pitch"</code> or <code>"roll"</code>, then
these values are extracted from the
&quot;slow&quot; form in the object (e.g. in <code>headingSlow</code> within
the <code>data</code> slot). In that case, accessing by full name, e.g.
<code>x[["headingSlow"]]</code> retrieves the item as expected, but
<code>x[["heading"]]</code> interpolates to the faster timescale, using
<code><a href="stats.html#topic+approx">approx</a></code><code style="white-space: pre;">&#8288;(timeSlow,headingSlow,time)&#8288;</code>.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(adv)
head(adv[["q"]]) # in raw form
head(adv[["q", "numeric"]]) # in numeric form

</code></pre>

<hr>
<h2 id='+5B+5B+2Camsr-method'>Extract Something From an amsr Object</h2><span id='topic++5B+5B+2Camsr-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'amsr'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Camsr-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+amsr-class">amsr</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Camsr-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Camsr-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Camsr-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Value</h3>

<p><code>[[</code> returns numeric matrix data.
</p>


<h3>Details of the Specialized Method</h3>

<p>The <code style="white-space: pre;">&#8288;[[[&#8288;</code> method handles both old-format and new-format <a href="#topic+amsr-class">amsr</a>
objects. Old-format objects are read by <code><a href="#topic+read.amsr">read.amsr()</a></code>
from from gzipped files holding data in raw format, from which
<code>[[</code> computes numeric results with linear
relationships provided at at <code style="white-space: pre;">&#8288;http://www.remss.com/missions/amsre&#8288;</code>.
By contrast, new-format objects are read
from NetCDF files that hold the data as 4-byte
numeric values that are read directly, without applying a
scaling transformation.  The other
difference is that old-format objects contain day and night values,
e.g. <code>SSTDay</code> and <code>SSTNight</code>, whereas new-format objects contain
single values that combine these, e.g. <code>SST</code>.
</p>
<p>If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are things that
<code>[[</code> can compute and then return.
</p>
<p>Data within the <code>data</code> slot may be found directly (for
both new-format and old-format objects) or indirectly (only
for old-style objects).  For example, <code>SST</code> works by direct
lookup for new-format objects, but it is computed using
<code>SSTNight</code> and <code>SSTDay</code> for old-format objects.  Use e.g.
<code>a[["?"]]</code> for any given object, to see what can be retrieved.
</p>


<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to amsr data: 
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method</a></code>,
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+read.amsr">read.amsr</a>()</code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+summary+2Camsr-method">summary,amsr-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Histogram of SST values (for an old-format dataset)
library(oce)
data(amsr)
hist(amsr[["SST"]])

</code></pre>

<hr>
<h2 id='+5B+5B+2Cargo-method'>Extract Something From an argo Object</h2><span id='topic++5B+5B+2Cargo-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'argo'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cargo-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+argo-class">argo</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cargo-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cargo-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cargo-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>

<p>Note that <a href="#topic+argo-class">argo</a> data may contain both unadjusted data and adjusted
data.  By default, this extraction function refers to the former, but a
preference for the latter may be set with <code><a href="#topic+preferAdjusted">preferAdjusted()</a></code>, the
documentation of which explains (fairly complex) details.
</p>
<p>The results from <code>argo[[i]]</code> or <code>argo[[i,j]]</code> depend on the
nature of <code>i</code> and (if provided) <code>j</code>. The details are as follows.
</p>

<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items hold the names of things
that can be inferred from the object's contents, e.g.
<code>"SA"</code> is named in <code>dataDerived</code>, indicating that
<code>argo[["SA"]]</code> is permitted (to compute Absolute Salinity).
</p>
</li>
<li><p> If <code>i</code> is <code>"profile"</code> and <code>j</code> is an integer vector,
then an argo object is returned, as specified by <code>j</code>. For example,
<code>argo[["profile", 2:5]]</code> is equivalent to
<code>subset(argo, profile %in% 2:5)</code>.
</p>
</li>
<li><p> If <code>i</code> is <code>"CT"</code>, then
Conservative Temperature is returned, as computed with
<code><a href="gsw.html#topic+gsw_CT_from_t">gsw::gsw_CT_from_t</a></code><code style="white-space: pre;">&#8288;(SA,t,p)&#8288;</code>, where
first <code>SA</code> is computed as explained
in the next item, <code>t</code> is in-situ temperature,
and <code>p</code> is pressure.
</p>
</li>
<li><p> If <code>i</code> is <code>"N2"</code>, then the square of buoyancy is returned,
as computed with <code><a href="#topic+swN2">swN2()</a></code>.
</p>
</li>
<li><p> If <code>i</code> is <code>"SA"</code>, then
Absolute Salinity is returned, as computed with
<code><a href="gsw.html#topic+gsw_SA_from_SP">gsw::gsw_SA_from_SP()</a></code>.
</p>
</li>
<li><p> If <code>i</code> is <code>"sigmaTheta"</code>, then
potential density anomaly (referenced to zero
pressure) is computed, with <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code>, where the
equation of state is taken to be
<code><a href="base.html#topic+getOption">getOption</a></code><code style="white-space: pre;">&#8288;("oceEOS",default="gsw")&#8288;</code>.
</p>
</li>
<li><p> If <code>i</code> is <code>"sigma0"</code>, <code>"sigma1"</code>, <code>"sigma2"</code>, <code>"sigma3"</code> or <code>"sigma4"</code>,
then the associated function in the <a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a> package.
For example, <code>"sigma0"</code> uses <code><a href="gsw.html#topic+gsw_sigma0">gsw::gsw_sigma0()</a></code>, which returns
potential density anomaly referenced to 0 dbar,
according to the gsw equation of state.
</p>
</li>
<li><p> If <code>i</code> is <code>"theta"</code>, then
potential temperature (referenced to zero
pressure) is computed, with <code><a href="#topic+swTheta">swTheta()</a></code>, where the
equation of state is taken to be
<code><a href="base.html#topic+getOption">getOption</a></code><code style="white-space: pre;">&#8288;("oceEOS",default="gsw")&#8288;</code>.
</p>
</li>
<li><p> If <code>i</code> is <code>"depth"</code>, then
a matrix of depths is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"id"</code> or <code>"ID"</code>, then the <code>id</code> element within
the <code>metadata</code> slot is returned.
</p>
</li>
<li><p> If <code>i</code> is in the <code>data</code> slot of <code>x</code>,
then it is returned, otherwise if it is in the <code>metadata</code> slot,
then that is returned, otherwise <code>NULL</code> is returned.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(argo)
# 1. show that dataset has 223 profiles, each with 56 levels
dim(argo[["temperature"]])

# 2. show importance of focussing on data flagged 'good'
fivenum(argo[["salinity"]], na.rm = TRUE)
fivenum(argo[["salinity"]][argo[["salinityFlag"]] == 1], na.rm = TRUE)

</code></pre>

<hr>
<h2 id='+5B+5B+2Cbremen-method'>Extract Something From a bremen Object</h2><span id='topic++5B+5B+2Cbremen-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'bremen'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cbremen-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+bremen-class">bremen</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cbremen-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cbremen-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cbremen-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined by <a href="#topic+bremen-class">bremen</a> objects.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to bremen data: 
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+read.bremen">read.bremen</a>()</code>,
<code><a href="#topic+summary+2Cbremen-method">summary,bremen-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Ccm-method'>Extract Something From a cm Object</h2><span id='topic++5B+5B+2Ccm-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'cm'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Ccm-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+cm-class">cm</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ccm-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ccm-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ccm-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined by <a href="#topic+cm-class">cm</a> objects.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to cm data: 
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+as.cm">as.cm</a>()</code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+read.cm">read.cm</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+summary+2Ccm-method">summary,cm-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Ccoastline-method'>Extract Something From a coastline Object</h2><span id='topic++5B+5B+2Ccoastline-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'coastline'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Ccoastline-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+coastline-class">coastline</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ccoastline-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ccoastline-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ccoastline-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined for <a href="#topic+coastline-class">coastline</a> objects.
</p>
</li>
<li><p> In many cases, the focus will be on the coastline trace
in longitude-latitude space, so <code>x[["longitude"]]</code>
and <code>x[["latitude"]]</code> are commonly used.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Cctd-method'>Extract Something From a ctd Object</h2><span id='topic++5B+5B+2Cctd-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ctd'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cctd-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cctd-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cctd-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cctd-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>

<p>Some uses of <code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code> involve direct retrieval of
items within the <code>data</code> slot of the <code>ctd</code> object,
while other uses involve calculations based on items in that
<code>data</code> slot. For example, all <code>ctd</code> objects
should hold an item named <code>temperature</code> in the <code>data</code>
slot, so for example <code>x[["temperature"]]</code> will retrieve that
item. By contrast, <code>x[["sigmaTheta"]]</code> is taken to be a
request to compute <code class="reqn">\sigma_\theta</code>, and so
it yields a call to <a href="#topic+swTheta">swTheta</a><code>(x)</code> even if
the <code>data</code> slot of <code>x</code> might happen to contain an item
named <code>theta</code>. This can be confusing at first, but it tends
to lead to fewer surprises in everyday work, for otherwise the
user would be forced to check the contents of any <code>ctd</code>
object under analysis, to determine whether that item will be looked
up or computed. Nothing is lost in this scheme, since the data
within the object are always accessible with <code><a href="#topic+oceGetData">oceGetData()</a></code>.
</p>
<p>It should be noted that the accessor is set up to retrieve quantities
in conventional units. For example, <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> is
used on a <code>.cnv</code> file that stores pressure in psi, it will
be stored in the same unit within the <code>ctd</code> object, but
<code>x[["pressure"]]</code> will return a value that has been converted
to decibars.  (To get pressure in PSI, use <code>x[["pressurePSI"]]</code>.)
Similarly, temperature is
returned in the ITS-90 scale, with a conversion having been performed with
<code><a href="#topic+T90fromT68">T90fromT68()</a></code>, if the object holds temperature in
IPTS-68.  Again, temperature on the IPTS-68
scale is returned with <code>x@data$temperature</code>.
</p>
<p>This preference for computed over stored quantities is accomplished
by first checking for computed quantities, and then falling
back to the general <code>[[</code> method if no match is found.
</p>
<p>Some quantities are optionally computed. For example, some data files
(e.g. the one upon which the <code><a href="#topic+section">section()</a></code> dataset is based)
store <code>nitrite</code> along with the sum of nitrite and nitrate, the
latter with name <code>NO2+NO3</code>. In this case, e.g. <code>x[["nitrate"]]</code>
will detect the setup, and subtract nitrite from the sum to yield
nitrate.
</p>
<p>The list given below provides notes on some quantities that are
available using e.g. <code>ctd[[i]]</code>.
</p>

<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items hold the names of things
that can be inferred from the object's contents, e.g.
<code>"SA"</code> is named in <code>dataDerived</code>, indicating that
<code>argo[["SA"]]</code> is permitted (to compute Absolute Salinity).
</p>
</li>
<li><p> If <code>i</code> is <code>"conductivity"</code> without a second argument (e.g. <code>a[["conductivity"]]</code>)
then the return value is the seawater electrical conductivity (if available
or computable). However, if a second argument is given,
and it is string specifying a unit, then conversion is made to that unit. The
permitted units are: either <code>""</code> or <code>"ratio"</code> (for ratio),
<code>"uS/cm"</code>, <code>"mS/cm"</code> and <code>"S/m"</code>. The calculations are based on
the definition of conductivity ratio as the ratio between measured conductivity
and the standard value 4.2914 S/m.
</p>
</li>
<li><p> If <code>i</code> is <code>"CT"</code> or <code>"Conservative Temperature"</code> then Conservative
Temperature, computed with <code><a href="gsw.html#topic+gsw_CT_from_t">gsw::gsw_CT_from_t()</a></code>, is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"density"</code> then seawater density, computed with <a href="#topic+swRho">swRho</a><code>(x)</code>,
is returned. (Note that it may be better to call that function directly, to gain
control of the choice of equation of state, etc.)
</p>
</li>
<li><p> If <code>i</code> is <code>"depth"</code> then the depth in metres below the surface, computed
with <a href="#topic+swDepth">swDepth</a><code>(x)</code>, is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"N2"</code> then the square of Brunt-Vaisala frequency, computed  with
<a href="#topic+swN2">swN2</a><code>(x)</code>, is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"potential temperature"</code> or <code>"theta"</code>, then potential temperature in the
UNESCO formulation, computed with <a href="#topic+swTheta">swTheta</a><code>(x)</code>, is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"Rrho"</code> then density ratio, computed with <a href="#topic+swRrho">swRrho</a><code>(x)</code>, is
returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"SA"</code> or <code>"Absolute Salinity"</code> then Absolute Salinity,
computed with <code><a href="gsw.html#topic+gsw_SA_from_SP">gsw::gsw_SA_from_SP()</a></code>, is returned.
The calculation involves location as well as measured water properties.
If the object <code>x</code> does not containing information on the location,
then 30N and 60W is used for the calculation, and a warning is generated.
</p>
</li>
<li><p> If <code>i</code> is <code>"sigmaTheta"</code> then a form of potential density anomaly, computed with
<a href="#topic+swSigmaTheta">swSigmaTheta</a><code>(x)</code>, is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"sigma0"</code> then potential density anomaly
referenced to a sea pressure of 0dbar (the surface), computed with <a href="#topic+swSigma0">swSigma0</a><code>(x)</code>,
is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"sigma2"</code> then potential density anomaly
referenced to a sea pressure of 1000dbar, computed with <a href="#topic+swSigma1">swSigma1</a><code>(x)</code>,
is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"sigma2"</code> then potential density anomaly
referenced to a sea pressure of 2000dbar, computed with <a href="#topic+swSigma2">swSigma2</a><code>(x)</code>,
is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"sigma3"</code> then potential density anomaly
referenced to a sea pressure of 3000dbar, computed with <a href="#topic+swSigma3">swSigma3</a><code>(x)</code>,
is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"sigma4"</code> then potential density anomaly
referenced to a sea pressure of 4000dbar, computed with <a href="#topic+swSigma4">swSigma4</a><code>(x)</code>,
is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"SP"</code> then salinity on the Practical Salinity Scale, which is
<code>salinity</code> in the <code>data</code> slot, is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"spice"</code> or <code>"spiciness0"</code> then a variable that is in some sense
orthogonal to density, calculated with <a href="#topic+swSpice">swSpice</a><code>(x)</code>, is returned.
Note that this is defined differently for <code>eos="unesco"</code> and
<code>eos="gsw"</code>.
</p>
</li>
<li><p> If <code>i</code> is <code>"SR"</code> then Reference Salinity, computed with
<code><a href="gsw.html#topic+gsw_SR_from_SP">gsw::gsw_SR_from_SP()</a></code>, is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"Sstar"</code> then Preformed Salinity, computed with
<code><a href="gsw.html#topic+gsw_SR_from_SP">gsw::gsw_SR_from_SP()</a></code>, is returned.
See <code>SA</code> for a note on longitude and latitude.
</p>
</li>
<li><p> If <code>i</code> is <code>"time"</code> then either vector of times or a single
time, is returned, if available. A vector is returned if <code>time</code>
is present in the <code>data</code> slot, or if a time can be
inferred from other entries in the <code>data</code> slot (some of which,
such as the common <code>timeS</code>, also employ
<code>startTime</code> within the <code>metadata</code> slot). A single
value is returned if the dataset only has information on the start
time (which is stored as <code>startTime</code> within the <code>metadata</code>
slot. If it is impossible to determine the sampling time, then
<code>NULL</code> is returned. These time variants occur, in the
present version of oce, only for data read by <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code>,
the documentation of which explains how times are computed.
</p>
</li>
<li><p> If <code>i</code> is <code>"z"</code> then vertical coordinate in metres
above the surface, computed with <a href="#topic+swZ">swZ</a><code>(x)</code>, is returned.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ctd)
head(ctd[["temperature"]])

</code></pre>

<hr>
<h2 id='+5B+5B+2Cechosounder-method'>Extract Something From an echosounder Object</h2><span id='topic++5B+5B+2Cechosounder-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'echosounder'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cechosounder-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+echosounder-class">echosounder</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cechosounder-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cechosounder-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cechosounder-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>metadataDerived</code> item
is NULL, while the <code>dataDerived</code> item holds
<code>"Sv"</code> and <code>"TS"</code> (see next).
</p>
</li>
<li><p> If <code>i</code> is <code>"Sv"</code>, then the following is returned.
</p>
<pre>
20*log10(a) -
  (x@metadata$sourceLevel+x@metadata$receiverSensitivity+x@metadata$transmitPower) +
  20*log10(r) +
  2*absorption*r -
  x@metadata$correction +
  10*log10(soundSpeed*x@metadata$pulseDuration/1e6*psi/2)
</pre>
</li>
<li><p> If <code>i</code> is <code>"TS"</code>, then the following is returned.
</p>
<pre>
20*log10(a) -
  (x@metadata$sourceLevel+x@metadata$receiverSensitivity+x@metadata$transmitPower) +
  40*log10(r) +
  2*absorption*r +
  x@metadata$correction
</pre>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to echosounder data: 
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic+as.echosounder">as.echosounder</a>()</code>,
<code><a href="#topic+echosounder-class">echosounder-class</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+findBottom">findBottom</a>()</code>,
<code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method</a></code>,
<code><a href="#topic+read.echosounder">read.echosounder</a>()</code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+summary+2Cechosounder-method">summary,echosounder-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Cg1sst-method'>Extract Something From a g1sst Object</h2><span id='topic++5B+5B+2Cg1sst-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'g1sst'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cg1sst-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+g1sst-class">g1sst</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cg1sst-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cg1sst-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cg1sst-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined by <a href="#topic+g1sst-class">g1sst</a> objects.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to g1sst data: 
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic+g1sst-class">g1sst-class</a></code>,
<code><a href="#topic+read.g1sst">read.g1sst</a>()</code>
</p>

<hr>
<h2 id='+5B+5B+2Cgps-method'>Extract Something From a gps Object</h2><span id='topic++5B+5B+2Cgps-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'gps'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cgps-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+gps-class">gps</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cgps-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cgps-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cgps-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined by <code>gps</code> objects.
</p>
</li>
<li><p> If <code>i</code> is <code>"longitude"</code> or <code>"latitude"</code>, then the corresponding
vector is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"filename"</code> then a filename is returned, if
known (i.e. if the object was created with <code><a href="#topic+read.gps">read.gps()</a></code> or
with <code><a href="#topic+as.gps">as.gps()</a></code> with the <code>filename</code> argument specified).
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to gps data: 
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic+as.gps">as.gps</a>()</code>,
<code><a href="#topic+gps-class">gps-class</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+read.gps">read.gps</a>()</code>,
<code><a href="#topic+summary+2Cgps-method">summary,gps-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Cladp-method'>Extract Something From an ladp Object</h2><span id='topic++5B+5B+2Cladp-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ladp'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cladp-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+ladp-class">ladp</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cladp-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cladp-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cladp-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>

<p>If <code>i</code> is <code>"?"</code>, then the return value is a list containing four items, each
of which is a character vector holding the names of things that can be
accessed with <code>[[</code>. The <code>data</code> and <code>metadata</code> items hold the names of entries
in the object's data and metadata slots, respectively. The <code>metadataDerived</code>
item is NULL, and the <code>dataDerived</code> item holds the following synonyms: <code>"p"</code>
for <code>"pressure"</code>, <code>"t"</code> for <code>"temperature"</code> and <code>"S"</code> for <code>"salinity"</code>.
</p>


<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to ladp data: 
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic+as.ladp">as.ladp</a>()</code>,
<code><a href="#topic+ladp-class">ladp-class</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+summary+2Cladp-method">summary,ladp-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Clandsat-method'>Extract Something From a landsat Object</h2><span id='topic++5B+5B+2Clandsat-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'landsat'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Clandsat-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+landsat-class">landsat</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Clandsat-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Clandsat-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Clandsat-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>

<p>If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>data</code> entries are difficult
to deal with directly, and so users are advised to
use <code>dataDerived</code> instead.
</p>
<p><em>Accessing band data.</em>  The data may be accessed with e.g.
<code>landsat[["panchromatic"]]</code>, for the panchromatic band.  If a new
&ldquo;band&rdquo; is added with <code><a href="#topic+landsatAdd">landsatAdd()</a></code>, it may be referred by
name.  In all cases, a second argument can be provided, to govern
decimation.  If this is missing, all the relevant data are returned.  If
this is present and equal to <code>TRUE</code>, then the data will be
automatically decimated (subsampled) to give approximately 800 elements in
the longest side of the matrix.  If this is present and numerical, then its
value governs decimation.  For example,
<code>landsat[["panchromatic",TRUE]]</code> will auto-decimate, typically
reducing the grid width and height from 16000 to about 800.  Similarly,
<code>landsat[["panchromatic",10]]</code> will reduce width and height to about
1600.  On machines with limited RAM (e.g. under about 6GB), decimation is a
good idea in almost all processing steps.  It also makes sense for
plotting, and in fact is done through the 'decimate' argument of
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method()</a></code>.
</p>
<p><em>Accessing derived data.</em>  One may retrieve several derived quantities
that are calculated from data stored in the object:
<code>landsat[["longitude"]]</code> and <code>landsat[["latitude"]]</code> give pixel
locations.  Accessing <code>landsat[["temperature"]]</code> creates an estimate
of ground temperature as follows (see reference 4).  First, the &ldquo;count value&rdquo; in
band 10, denoted <code class="reqn">b_{10}</code> say, is scaled with coefficients stored
in the image metadata using
<code class="reqn">\lambda_L=b_{10}M_L+A_L</code> where <code class="reqn">M_L</code>
and <code class="reqn">A_L</code> are values stored in the metadata (e.g.  the first in
<code>landsat@metadata$header$radiance_mult_band_10</code>) Then the result is
used, again with coefficients in the metadata, to compute Celcius
temperature <code class="reqn">T=K_2/ln(\epsilon
    K_1/\lambda_L+1)-273.15</code>.
The value of the emissivity <code class="reqn">\epsilon</code> is set to unity by
<code><a href="#topic+read.landsat">read.landsat()</a></code>, although it can be changed easily later, by
assigning a new value to 'landsat@metadata$emissivity'. The default
emissivity value set by <code><a href="#topic+read.landsat">read.landsat()</a></code> is from reference 11, and is
within the oceanic range suggested by reference 5. Adjustment is as simple as
altering 'landsat@metadata$emissivity'. This value can be a single
number meant to apply for the whole image, or a matrix with dimensions
matching those of band 10.  The matrix case is probably more useful for
images of land, where one might wish to account for the different
emissivities of soil and vegetation, etc.; for example, Table 4 of
reference 9 lists 0.9668 for soil and 0.9863 for vegetation,
while Table 5 of reference 10 lists 0.971 and 0.987 for the same quantities.
</p>
<p><em>Accessing metadata.</em> Anything in the metadata can be accessed by
name, e.g. <code>landsat[["time"]]</code>.  Note that some items are simply
copied over from the source data file and are not altered by e.g.
decimation.  An exception is the lat-lon box, which is altered by
<code><a href="#topic+landsatTrim">landsatTrim()</a></code>.
</p>


<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to landsat data: 
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic+landsat-class">landsat-class</a></code>,
<code><a href="#topic+landsatAdd">landsatAdd</a>()</code>,
<code><a href="#topic+landsatTrim">landsatTrim</a>()</code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+read.landsat">read.landsat</a>()</code>,
<code><a href="#topic+summary+2Clandsat-method">summary,landsat-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Clisst-method'>Extract Something From a lisst Object</h2><span id='topic++5B+5B+2Clisst-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lisst'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Clisst-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+lisst-class">lisst</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Clisst-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Clisst-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Clisst-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined by <a href="#topic+lisst-class">lisst</a> objects.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to lisst data: 
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic+as.lisst">as.lisst</a>()</code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+read.lisst">read.lisst</a>()</code>,
<code><a href="#topic+summary+2Clisst-method">summary,lisst-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Clobo-method'>Extract Something From a lobo Object</h2><span id='topic++5B+5B+2Clobo-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lobo'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Clobo-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+lobo-class">lobo</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Clobo-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Clobo-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Clobo-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined by <code>cm</code> objects.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to lobo data: 
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic+as.lobo">as.lobo</a>()</code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+read.lobo">read.lobo</a>()</code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+summary+2Clobo-method">summary,lobo-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Cmet-method'>Extract Something From a met Object</h2><span id='topic++5B+5B+2Cmet-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'met'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cmet-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+met-class">met</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cmet-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cmet-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cmet-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined by <a href="#topic+met-class">met</a> objects.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to met data: 
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic+as.met">as.met</a>()</code>,
<code><a href="#topic+download.met">download.met</a>()</code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+read.met">read.met</a>()</code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+summary+2Cmet-method">summary,met-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Coce-method'>Extract Something From an oce Object</h2><span id='topic++5B+5B+2Coce-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'oce'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Coce-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Coce-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Coce-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Coce-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Many <code>oce</code> object classes have specialized versions
of <code>[[</code> that handle the details in specialized way.
</p>
<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Codf-method'>Extract Something From an odf Object</h2><span id='topic++5B+5B+2Codf-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'odf'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Codf-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+odf-class">odf</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Codf-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Codf-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Codf-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>

<p>If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined by <a href="#topic+odf-class">odf</a> objects.
</p>


<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to odf data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ODF2oce">ODF2oce</a>()</code>,
<code><a href="#topic+ODFListFromHeader">ODFListFromHeader</a>()</code>,
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.odf">read.odf</a>()</code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+summary+2Codf-method">summary,odf-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Crsk-method'>Extract Something From a rsk Object</h2><span id='topic++5B+5B+2Crsk-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'rsk'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Crsk-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+rsk-class">rsk</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Crsk-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Crsk-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Crsk-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined by <a href="#topic+rsk-class">rsk</a> objects.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic+as.rsk">as.rsk</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+read.rsk">read.rsk</a>()</code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+rskPatm">rskPatm</a>()</code>,
<code><a href="#topic+rskToc">rskToc</a>()</code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Csealevel-method'>Extract Something From a sealevel Object</h2><span id='topic++5B+5B+2Csealevel-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'sealevel'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Csealevel-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+sealevel-class">sealevel</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Csealevel-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Csealevel-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Csealevel-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined by <a href="#topic+sealevel-class">sealevel</a> objects.
</p>
</li>
<li><p> In many cases, the focus will be on variations of
sealevel elevation over time, so it is common to use e.g.
<code>x[["time"]]</code> and <code>x[["elevation"]]</code> to retrieve vectors
of these quantities. Another common task is to retrieve
the location of the observations, using e.g. <code>x[["longitude"]]</code>
and <code>x[["latitude"]]</code>.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to sealevel data: 
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic+as.sealevel">as.sealevel</a>()</code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+read.sealevel">read.sealevel</a>()</code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+summary+2Csealevel-method">summary,sealevel-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Csection-method'>Extract Something From a section Object</h2><span id='topic++5B+5B+2Csection-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'section'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Csection-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+section-class">section</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Csection-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Csection-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Csection-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>

<p>There are several possibilities, depending on the nature of <code>i</code>.
</p>

<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list containing four items,
each of which is a character vector holding the names of things that can be
accessed with <code>[[</code>. This list is compiled by examining all the stations in
the object, and reporting an entry if it is found in any one of them. The
<code>data</code> and <code>metadata</code> items hold the names of entries in the object's data
and metadata slots, respectively. The <code>dataDerived</code> and <code>metadataDerived</code>
items hold data-like and metadata-like things that can be derived from these.
</p>
</li>
<li><p> If <code>i</code> is <code>"station"</code>, then <code>[[</code> will return a <code><a href="base.html#topic+list">list()</a></code> of <a href="#topic+ctd-class">ctd</a>
objects holding the station data. If <code>j</code> is also given, it specifies a
station (or set of stations) to be returned. if <code>j</code> contains just a single
value, then that station is returned, but otherwise a list is returned. If
<code>j</code> is an integer, then the stations are specified by index, but if it is
character, then stations are specified by the names stored within their
metadata. (Missing stations yield <code>NULL</code> in the return value.)
</p>
</li>
<li><p> If <code>i</code> is <code>"station ID"</code>, then the IDs of the stations in the
section are returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"dynamic height"</code>, then an estimate of dynamic
height is returned, as calculated with <code><a href="#topic+swDynamicHeight">swDynamicHeight</a></code><code>(x)</code>.
</p>
</li>
<li><p> If <code>i</code> is <code>"distance"</code>, then the distance along the section is
returned, using <code><a href="#topic+geodDist">geodDist()</a></code>.
</p>
</li>
<li><p> If <code>i</code> is <code>"depth"</code>, then a vector containing the depths
of the stations is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"z"</code>, then a vector containing the z
coordinates is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"theta"</code> or <code>"potential temperature"</code>, then
the potential temperatures of all the stations are returned in one
vector.  Similarly, <code>"spice"</code> returns the property known
as spice, using <code><a href="#topic+swSpice">swSpice()</a></code>.
</p>
</li>
<li><p> If <code>i</code> is a string ending with <code>"Flag"</code>, then the characters
prior to that ending are taken to be the name of a variable contained
within the stations in the section. If this flag is available in
the first station of the section, then the flag values are looked
up for every station.
</p>
</li></ul>

<p>If <code>j</code> is <code>"byStation"</code>, then a list is returned, with
one (unnamed) item per station.
</p>
<p>If <code>j</code> is <code>"grid:distance-pressure"</code> or <code>"grid:time-pressure"</code>, then a gridded
representation of <code>i</code> is returned, as a list with elements:
<code>distance</code> (in km) or <code>time</code> (in POSIXct); <code>pressure</code> (in dbar) and
<code>field</code> (in whatever unit is used for <code>i</code>). See the
examples in the documentation for <code><a href="#topic+plot+2Csection-method">plot,section-method()</a></code>.
</p>


<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to section data: 
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(section)
length(section[["latitude"]])
length(section[["latitude", "byStation"]])
# Vector of all salinities, for all stations
Sv &lt;- section[["salinity"]]
# List of salinities, grouped by station
Sl &lt;- section[["salinity", "byStation"]]
# First station salinities
Sl[[1]]

</code></pre>

<hr>
<h2 id='+5B+5B+2Ctidem-method'>Extract Something From a tidem Object</h2><span id='topic++5B+5B+2Ctidem-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'tidem'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Ctidem-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+tidem-class">tidem</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ctidem-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ctidem-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ctidem-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. Note that <code>metadataDerived</code>
holds only <code>""</code>, because no derived metadata values
are defined for <code>tidem</code> objects.
</p>
</li>
<li><p> If <code>i</code> is <code>"frequency"</code> or <code>"freq"</code>, then a vector of
constituent frequencies is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"amplitude"</code> then a vector of constituent amplitudes
is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"phase"</code> then a vector of constituent phases
is returned.
</p>
</li>
<li><p> If <code>i</code> is <code>"constituents"</code> then a data frame holding constituent
name, frequency, amplitude and phase is returned.
</p>
</li>
<li><p> If <code>i</code> is a vector of constituent names, then the return
value is as for <code>"constituents"</code>, except that only the named
those constituents are returned.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to tides: 
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>

<hr>
<h2 id='+5B+5B+2Ctopo-method'>Extract Something From a topo Object</h2><span id='topic++5B+5B+2Ctopo-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'topo'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Ctopo-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+topo-class">topo</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ctopo-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ctopo-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Ctopo-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are available for <code>topo</code> objects.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to topo data: 
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic+as.topo">as.topo</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+read.topo">read.topo</a>()</code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+summary+2Ctopo-method">summary,topo-method</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+topoInterpolate">topoInterpolate</a>()</code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(topoWorld)
dim(topoWorld[["z"]])

</code></pre>

<hr>
<h2 id='+5B+5B+2Cwindrose-method'>Extract Something From a windrose Object</h2><span id='topic++5B+5B+2Cwindrose-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'windrose'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cwindrose-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+windrose-class">windrose</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cwindrose-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cwindrose-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cwindrose-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>metadataDerived</code> and
<code>dataDerived</code> items are both NULL.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to windrose data: 
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic+as.windrose">as.windrose</a>()</code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+summary+2Cwindrose-method">summary,windrose-method</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>
</p>

<hr>
<h2 id='+5B+5B+2Cxbt-method'>Extract Something From an xbt Object</h2><span id='topic++5B+5B+2Cxbt-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'xbt'
x[[i, j, ...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B2Cxbt-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+xbt-class">xbt</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cxbt-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cxbt-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B2Cxbt-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the Specialized Method</h3>


<ul>
<li><p> If <code>i</code> is <code>"?"</code>, then the return value is a list
containing four items, each of which is a character vector
holding the names of things that can be accessed with <code>[[</code>.
The <code>data</code> and <code>metadata</code> items hold the names of
entries in the object's data and metadata
slots, respectively. The <code>dataDerived</code>
and <code>metadataDerived</code> items are each NULL, because
no derived values are defined by <code>cm</code> objects.
</p>
</li></ul>



<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>
</p>
<p>Other things related to xbt data: 
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>,
<code><a href="#topic+as.xbt">as.xbt</a>()</code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1</a>()</code>,
<code><a href="#topic+read.xbt">read.xbt</a>()</code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>,
<code><a href="#topic+summary+2Cxbt-method">summary,xbt-method</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cadp-method'>Replace Parts of an adp Object</h2><span id='topic++5B+5B+3C-+2Cadp-method'></span>

<h3>Description</h3>

<p>In addition to the usual insertion of elements by name, note
that e.g. <code>pitch</code> gets stored into <code>pitchSlow</code>.
</p>
<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'adp'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cadp-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cadp-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cadp-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cadp-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cadp-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cadv-method'>Replace Parts of an adv Object</h2><span id='topic++5B+5B+3C-+2Cadv-method'></span>

<h3>Description</h3>

<p>Generally, the <code>[[</code> method lets users extract information from <code>oce</code>
objects, without having to know the details of the internal storage.  For
many <code>oce</code> sub-classes, <code>[[</code> can also return quantities that are computed
from the object's contents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'adv'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cadv-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cadv-method_+3A_i">i</code></td>
<td>
<p>character value indicating the name of an item to extract.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cadv-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cadv-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cadv-method_+3A_value">value</code></td>
<td>
<p>The value to be inserted into <code>x</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the <code>adv</code> object holds slow variables (i.e. if <code>timeSlow</code> is
in the <code>data</code> slot), then assigning to .e.g. <code>heading</code> will not
actually assign to a variable of that name, but instead assigns to
<code>headingSlow</code>. To catch misapplication of this rule, an error
message will be issued if the assigned value is not of the same length
as <code>timeSlow</code>.
</p>
<p>A two-step process is used to try to find the requested information. First, a
class-specific function is used (see &ldquo;Details of the Specialized
Method&rdquo;).  If this yields nothing, then a general method is used (see
&ldquo;Details of the General Method&rdquo;).  If both methods fail, then <code>[[</code>
returns NULL.
</p>
<p>Some understanding of the subclass is required to know what can be retrieved
with <code>[[</code>. When dealing with an unfamiliar subclass, it can be useful to
first use <code>x[["?"]]</code> to get a listing of the retrievable items. See
&ldquo;Details of the Specialized Method&rdquo; for more information.
</p>


<h3>Details of the General Method</h3>

<p>Note: the text of this section is identical for all <code>oce</code> subclasses, and so
some of what you read here may not be relevant to the class being described
in this help page.
</p>
<p>If the specialized method produces no matches, the following generalized
method is applied. As with the specialized method, the procedure hinges first
on the values of <code>i</code> and, optionally, <code>j</code>. The work proceeds in steps, by
testing a sequence of possible conditions in sequence.
</p>

<ol>
<li><p> A check is made as to whether <code>i</code> names one of the standard <code>oce</code> slots.
If so, <code>[[</code> returns the slot contents of that slot.  Thus, <code>x[["metadata"]]</code>
will retrieve the <code>metadata</code> slot, while <code>x[["data"]]</code> and
<code>x[["processingLog"]]</code> return those slots.
</p>
</li>
<li><p> If <code>i</code> is a string ending in the <code>"Unit"</code>, then the characters preceding
that string are taken to be the name of an item in the data object, and a
list containing the unit is returned (or <code>NULL</code> if there is no such unit).
This list consists of an item named <code>unit</code>, which is an <code><a href="base.html#topic+expression">expression()</a></code>, and
an item named <code>scale</code>, which is a string describing the measurement scale.
If the string ends in <code>" unit"</code>, e.g.  <code>x[["temperature unit"]]</code> (note the
space), then just the expression is returned, and if it ends in <code>" scale"</code>,
then just the scale is returned.
</p>
</li>
<li><p> If <code>i</code> is a string ending in <code>"Flag"</code>, then the corresponding data-quality
flag is returned (or <code>NULL</code> if there is no such flag).
</p>
</li>
<li><p> If the object holds hydrographic information (pressure, salinity,
temperature, longitude and latitude) then another set of possibilities
arises.  If <code>i</code> is <code>"sigmaTheta"</code>, then the value of <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> is
called with <code>x</code> as the sole argument, and the results are returned.
Similarly, <code><a href="#topic+swSigma0">swSigma0()</a></code> is used if <code>i="sigma0"</code>, and <code><a href="#topic+swSpice">swSpice()</a></code> is used if
<code>i="spice"</code>. Of course, these actions only make sense for objects that
contain the relevant items within their <code>data</code> slot.
</p>
</li>
<li><p> After these possibilities are eliminated, the action depends on whether
<code>j</code> has been provided.  If <code>j</code> is not provided, or is the string <code>""</code>, then
<code>i</code> is sought in the <code>metadata</code> slot, and then in the <code>data</code> slot, returning
whichever is found first.  In other words, if <code>j</code> is not provided, the
<code>metadata</code> slot takes preference over the <code>data</code> slot. However, if <code>j</code> is
provided, then it must be either the string <code>"metadata"</code> or <code>"data"</code>, and it
directs where to look.
</p>
</li>
<li><p> If none of the above-listed conditions holds, then <code>NULL</code> is returned.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that extract parts of oce objects: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+2Coce-method">[[,oce-method</a></code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Camsr-method'>Replace Parts of an amsr Object</h2><span id='topic++5B+5B+3C-+2Camsr-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'amsr'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Camsr-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+amsr-class">amsr</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Camsr-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Camsr-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Camsr-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Camsr-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to amsr data: 
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method</a></code>,
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+read.amsr">read.amsr</a>()</code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+summary+2Camsr-method">summary,amsr-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cargo-method'>Replace Parts of an argo Object</h2><span id='topic++5B+5B+3C-+2Cargo-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'argo'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cargo-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+argo-class">argo</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cargo-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cargo-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cargo-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cargo-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cbremen-method'>Replace Parts of a bremen Object</h2><span id='topic++5B+5B+3C-+2Cbremen-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'bremen'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cbremen-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+bremen-class">bremen</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cbremen-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cbremen-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cbremen-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cbremen-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to bremen data: 
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+read.bremen">read.bremen</a>()</code>,
<code><a href="#topic+summary+2Cbremen-method">summary,bremen-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Ccm-method'>Replace Parts of a cm Object</h2><span id='topic++5B+5B+3C-+2Ccm-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'cm'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ccm-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+cm-class">cm</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ccm-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ccm-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ccm-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ccm-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to cm data: 
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+as.cm">as.cm</a>()</code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+read.cm">read.cm</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+summary+2Ccm-method">summary,cm-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Ccoastline-method'>Replace Parts of a coastline Object</h2><span id='topic++5B+5B+3C-+2Ccoastline-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'coastline'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ccoastline-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+coastline-class">coastline</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ccoastline-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ccoastline-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ccoastline-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ccoastline-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>
<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cctd-method'>Replace Parts of a ctd Object</h2><span id='topic++5B+5B+3C-+2Cctd-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'ctd'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cctd-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cctd-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cctd-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cctd-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cctd-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ctd)
summary(ctd)
# Move the CTD profile a nautical mile north.
ctd[["latitude"]] &lt;- 1 / 60 + ctd[["latitude"]] # acts in metadata
# Increase the salinity by 0.01.
ctd[["salinity"]] &lt;- 0.01 + ctd[["salinity"]] # acts in data
summary(ctd)

</code></pre>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cechosounder-method'>Replace Parts of an echosounder Object</h2><span id='topic++5B+5B+3C-+2Cechosounder-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'echosounder'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cechosounder-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+echosounder-class">echosounder</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cechosounder-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cechosounder-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cechosounder-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cechosounder-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to echosounder data: 
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic+as.echosounder">as.echosounder</a>()</code>,
<code><a href="#topic+echosounder-class">echosounder-class</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+findBottom">findBottom</a>()</code>,
<code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method</a></code>,
<code><a href="#topic+read.echosounder">read.echosounder</a>()</code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+summary+2Cechosounder-method">summary,echosounder-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cg1sst-method'>Replace Parts of a g1sst Object</h2><span id='topic++5B+5B+3C-+2Cg1sst-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'g1sst'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cg1sst-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+g1sst-class">g1sst</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cg1sst-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cg1sst-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cg1sst-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cg1sst-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to g1sst data: 
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic+g1sst-class">g1sst-class</a></code>,
<code><a href="#topic+read.g1sst">read.g1sst</a>()</code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cgps-method'>Replace Parts of a gps Object</h2><span id='topic++5B+5B+3C-+2Cgps-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'gps'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cgps-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+gps-class">gps</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cgps-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cgps-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cgps-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cgps-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to gps data: 
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic+as.gps">as.gps</a>()</code>,
<code><a href="#topic+gps-class">gps-class</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+read.gps">read.gps</a>()</code>,
<code><a href="#topic+summary+2Cgps-method">summary,gps-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cladp-method'>Replace Parts of an ladp Object</h2><span id='topic++5B+5B+3C-+2Cladp-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'ladp'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cladp-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+ladp-class">ladp</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cladp-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cladp-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cladp-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cladp-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to ladp data: 
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic+as.ladp">as.ladp</a>()</code>,
<code><a href="#topic+ladp-class">ladp-class</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+summary+2Cladp-method">summary,ladp-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Clandsat-method'>Replace Parts of a landsat Object</h2><span id='topic++5B+5B+3C-+2Clandsat-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'landsat'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clandsat-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+landsat-class">landsat</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clandsat-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clandsat-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clandsat-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clandsat-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to landsat data: 
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic+landsat-class">landsat-class</a></code>,
<code><a href="#topic+landsatAdd">landsatAdd</a>()</code>,
<code><a href="#topic+landsatTrim">landsatTrim</a>()</code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+read.landsat">read.landsat</a>()</code>,
<code><a href="#topic+summary+2Clandsat-method">summary,landsat-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Clisst-method'>Replace Parts of a lisst Object</h2><span id='topic++5B+5B+3C-+2Clisst-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'lisst'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clisst-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+lisst-class">lisst</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clisst-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clisst-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clisst-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clisst-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to lisst data: 
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic+as.lisst">as.lisst</a>()</code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+read.lisst">read.lisst</a>()</code>,
<code><a href="#topic+summary+2Clisst-method">summary,lisst-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Clobo-method'>Replace Parts of a lobo Object</h2><span id='topic++5B+5B+3C-+2Clobo-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'lobo'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clobo-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+lobo-class">lobo</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clobo-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clobo-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clobo-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Clobo-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to lobo data: 
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic+as.lobo">as.lobo</a>()</code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+read.lobo">read.lobo</a>()</code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+summary+2Clobo-method">summary,lobo-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cmet-method'>Replace Parts of a met Object</h2><span id='topic++5B+5B+3C-+2Cmet-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'met'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cmet-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+met-class">met</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cmet-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cmet-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cmet-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cmet-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to met data: 
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic+as.met">as.met</a>()</code>,
<code><a href="#topic+download.met">download.met</a>()</code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+read.met">read.met</a>()</code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+summary+2Cmet-method">summary,met-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Coce-method'>Replace Parts of an oce Object</h2><span id='topic++5B+5B+3C-+2Coce-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'oce'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Coce-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Coce-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Coce-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Coce-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Coce-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Codf-method'>Replace Parts of an odf Object</h2><span id='topic++5B+5B+3C-+2Codf-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'odf'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Codf-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+odf-class">odf</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Codf-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Codf-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Codf-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Codf-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to odf data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ODF2oce">ODF2oce</a>()</code>,
<code><a href="#topic+ODFListFromHeader">ODFListFromHeader</a>()</code>,
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.odf">read.odf</a>()</code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+summary+2Codf-method">summary,odf-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Crsk-method'>Replace Parts of an rsk Object</h2><span id='topic++5B+5B+3C-+2Crsk-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'rsk'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Crsk-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+rsk-class">rsk</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Crsk-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Crsk-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Crsk-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Crsk-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic+as.rsk">as.rsk</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+read.rsk">read.rsk</a>()</code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+rskPatm">rskPatm</a>()</code>,
<code><a href="#topic+rskToc">rskToc</a>()</code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Csealevel-method'>Replace Parts of a sealevel Object</h2><span id='topic++5B+5B+3C-+2Csealevel-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'sealevel'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Csealevel-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+sealevel-class">sealevel</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Csealevel-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Csealevel-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Csealevel-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Csealevel-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to sealevel data: 
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic+as.sealevel">as.sealevel</a>()</code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+read.sealevel">read.sealevel</a>()</code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+summary+2Csealevel-method">summary,sealevel-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Csection-method'>Replace Parts of a section Object</h2><span id='topic++5B+5B+3C-+2Csection-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'section'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Csection-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+section-class">section</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Csection-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Csection-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Csection-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Csection-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Change section ID from a03 to A03
data(section)
section[["sectionId"]]
section[["sectionId"]] &lt;- toupper(section[["sectionId"]])
section[["sectionId"]]
# 2. Add a millidegree to temperatures at station 10
section[["station", 10]][["temperature"]] &lt;-
    1e-3 + section[["station", 10]][["temperature"]]

</code></pre>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Ctidem-method'>Replace Parts of a tidem Object</h2><span id='topic++5B+5B+3C-+2Ctidem-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'tidem'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ctidem-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+tidem-class">tidem</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ctidem-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ctidem-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ctidem-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ctidem-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Ctopo-method'>Replace Parts of a topo Object</h2><span id='topic++5B+5B+3C-+2Ctopo-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'topo'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ctopo-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+topo-class">topo</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ctopo-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ctopo-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ctopo-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Ctopo-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other things related to topo data: 
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic+as.topo">as.topo</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+read.topo">read.topo</a>()</code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+summary+2Ctopo-method">summary,topo-method</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+topoInterpolate">topoInterpolate</a>()</code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>
</p>
<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cwindrose-method'>Replace Parts of a windrose Object</h2><span id='topic++5B+5B+3C-+2Cwindrose-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'windrose'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cwindrose-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+windrose-class">windrose</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cwindrose-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cwindrose-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cwindrose-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cwindrose-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>
</p>
<p>Other things related to windrose data: 
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic+as.windrose">as.windrose</a>()</code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+summary+2Cwindrose-method">summary,windrose-method</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>
</p>

<hr>
<h2 id='+5B+5B+26lt+3B-+2Cxbt-method'>Replace Parts of an xbt Object</h2><span id='topic++5B+5B+3C-+2Cxbt-method'></span>

<h3>Description</h3>

<p>The <code>[[&lt;-</code> method works for all <a href="#topic+oce-class">oce</a> objects.
The purpose, as with the related extraction method, <code>[[</code>,
is to insulate users from the internal details of <a href="#topic+oce-class">oce</a>
objects, by looking for items within the various storage
slots of the object. Items not actually stored can also be
replaced, including units and data-quality
flags.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'xbt'
x[[i, j, ...]] &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cxbt-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+xbt-class">xbt</a> object.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cxbt-method_+3A_i">i</code></td>
<td>
<p>character value naming the item to replace.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cxbt-method_+3A_j">j</code></td>
<td>
<p>optional additional information on the <code>i</code> item.</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cxbt-method_+3A_...">...</code></td>
<td>
<p>optional additional information (ignored).</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B+2B26lt+2B3B-+2B2Cxbt-method_+3A_value">value</code></td>
<td>
<p>The value to be placed into <code>x</code>, somewhere.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As with <code>[[</code> method, the procedure works in steps.
</p>
<p>First, the <code>metadata</code> slot of <code>x</code> is checked to
see whether it contains something named with <code>i</code>.
If so, then the named item is replaced with <code>value</code>.
</p>
<p>Otherwise, if the string value of <code>i</code> ends in <code>Unit</code>, then the
characters preceding that are taken as the name of a variable, and
the <code>metadata</code> slot of <code>x</code> is updated to store that unit, e.g.
</p>
<div class="sourceCode"><pre>x[["temperatureUnits"]] &lt;- list(unit=expression(degree*F),scale="")
</pre></div>
<p>Similarly, if <code>i</code> ends in <code>Flag</code>, then quality-control
flags are set up as defined by <code>result</code>, e.g.
</p>
<div class="sourceCode"><pre>o[["temperatureFlags"]] &lt;- c(2,4,2,2)
</pre></div>
<p>Otherwise, <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for a partial-string match with
the names of the items that are in the <code>data</code> slot of <code>x</code>.
The first item found (if any) is then updated to hold the value <code>result</code>.
</p>
<p>If none of these conditions is met, a warning is issued.
</p>


<h3>See Also</h3>

<p>Other functions that replace parts of oce objects: 
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Coce-method">[[&lt;-,oce-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>
</p>
<p>Other things related to xbt data: 
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic+as.xbt">as.xbt</a>()</code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1</a>()</code>,
<code><a href="#topic+read.xbt">read.xbt</a>()</code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>,
<code><a href="#topic+summary+2Cxbt-method">summary,xbt-method</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>

<hr>
<h2 id='abbreviateTimeLabels'>Abbreviate a Vector of Times by Removing Commonalities</h2><span id='topic+abbreviateTimeLabels'></span>

<h3>Description</h3>

<p>Abbreviate a vector of times by removing commonalities (e.g. year)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>abbreviateTimeLabels(t, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="abbreviateTimeLabels_+3A_t">t</code></td>
<td>
<p>vector of times.</p>
</td></tr>
<tr><td><code id="abbreviateTimeLabels_+3A_...">...</code></td>
<td>
<p>optional arguments passed to the <code><a href="base.html#topic+format">format()</a></code>, e.g.
<code>format</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, with help from Clark Richards
</p>


<h3>See Also</h3>

<p>This is used by various functions that draw time labels on axes,
e.g.  <code><a href="#topic+plot+2Cadp-method">plot,adp-method()</a></code>.
</p>

<hr>
<h2 id='ad2cpCodeToName'>Map AD2CP ID Code to oce Name</h2><span id='topic+ad2cpCodeToName'></span>

<h3>Description</h3>

<p>As explained in Nortek (2022, section 6.1, page 80), AD2CP files use a
hexadecimal (in R, &quot;raw&quot;) code to indicate the nature of each data chunk, and
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp()</a></code> uses the present function as it analyses AD2CP files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ad2cpCodeToName(code = NULL, prefix = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ad2cpCodeToName_+3A_code">code</code></td>
<td>
<p>a <a href="base.html#topic+raw">raw</a> (or corresponding integer) vector indicating the IDs of
interest, or NULL to get a summary of possible values.</p>
</td></tr>
<tr><td><code id="ad2cpCodeToName_+3A_prefix">prefix</code></td>
<td>
<p>logical value indicating whether to show the raw value as a
prefix (e.g. <code>"0x1c=echosounder"</code> as opposed to <code>"echosounder"</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The mapping from code (hex or decimal) to oce name is as follows.</p>

<table>
<tr>
 <td style="text-align: right;">
   code (raw) </td><td style="text-align: right;"> code (integer) </td><td style="text-align: right;"> oce name </td>
</tr>
<tr>
 <td style="text-align: right;">
   ---------- </td><td style="text-align: right;"> -------------- </td><td style="text-align: right;"> ----------------- </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x15</code> </td><td style="text-align: right;"> 21 </td><td style="text-align: right;"> <code>burst</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x16</code> </td><td style="text-align: right;"> 22 </td><td style="text-align: right;"> <code>average</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x17</code> </td><td style="text-align: right;"> 23 </td><td style="text-align: right;"> <code>bottomTrack</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x18</code> </td><td style="text-align: right;"> 24 </td><td style="text-align: right;"> <code>interleavedBurst</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1a</code> </td><td style="text-align: right;"> 26 </td><td style="text-align: right;"> <code>burstAltimeterRaw</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1b</code> </td><td style="text-align: right;"> 27 </td><td style="text-align: right;"> <code>DVLBottomTrack</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1c</code> </td><td style="text-align: right;"> 28 </td><td style="text-align: right;"> <code>echosounder</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1d</code> </td><td style="text-align: right;"> 29 </td><td style="text-align: right;"> <code>DVLWaterTrack</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1e</code> </td><td style="text-align: right;"> 30 </td><td style="text-align: right;"> <code>altimeter</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1f</code> </td><td style="text-align: right;"> 31 </td><td style="text-align: right;"> <code>averageAltimeter</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x23</code> </td><td style="text-align: right;"> 35 </td><td style="text-align: right;"> <code>echosounderRaw</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0xa0</code> </td><td style="text-align: right;"> 160 </td><td style="text-align: right;"> <code>text</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>An indication of the mapping.  If <code>code</code> is NULL, this is a data
frame.  Otherwise, it is a character vector with the relevant mappings,
with the raw form of the code linked with the name, as in the example.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Nortek AS. &ldquo;Signature Integration 55|250|500|1000kHz.&rdquo; Nortek AS,
March 31, 2022.
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to ad2cp data: 
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stopifnot(ad2cpCodeToName(0x15) == "0x15=burst")

</code></pre>

<hr>
<h2 id='ad2cpHeaderValue'>Infer an Item From a Nortek AD2CP File Header</h2><span id='topic+ad2cpHeaderValue'></span>

<h3>Description</h3>

<p>Infer an Item From a Nortek AD2CP File Header
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ad2cpHeaderValue(x, key, item, numeric = TRUE, default)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ad2cpHeaderValue_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object that holds AD2CP data.</p>
</td></tr>
<tr><td><code id="ad2cpHeaderValue_+3A_key">key</code></td>
<td>
<p>Character value that identifies a particular line in the file
header.</p>
</td></tr>
<tr><td><code id="ad2cpHeaderValue_+3A_item">item</code></td>
<td>
<p>Character value indicating the name of the item sought.</p>
</td></tr>
<tr><td><code id="ad2cpHeaderValue_+3A_numeric">numeric</code></td>
<td>
<p>Logical value indicating whether to convert the return value
from a string to a numerical value.</p>
</td></tr>
<tr><td><code id="ad2cpHeaderValue_+3A_default">default</code></td>
<td>
<p>Optional value to be used if the item is not found in the
header, or if the header is <code>NULL</code> (as in the case of a split-up file
that lacks the initial header information)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>String or number interpreted from the <code>x[["text"]]</code>, or <code>NULL</code>,
if the desired item is not found there, or if <code>x</code> is not of the required
class and variety.
</p>


<h3>Sample of Usage</h3>

<pre>
if (file.exists("a.ad2cp")) {
    d &lt;- read.oce("a.ad2cp")
    # The examples start with the line in x[["text"]][[1]]; note that in the second
    # example, it would be insuficient to use a key of "BEAMCFGLIST", because that will
    # yield 4 lines, and the function is not designed to handle that.

    # ID,STR=\"Signature1000\",SN=123456
    type &lt;- ad2cpHeaderValue(d, "ID", "STR", numeric=FALSE)
    serialNumber &lt;- ad2cpHeaderValue(d, "ID", "SN")

    # BEAMCFGLIST,BEAM=1,THETA=25.00,PHI=0.00,FREQ=1000,BW=25,BRD=1,HWBEAM=1,ZNOM=60.00
    beam1Angle &lt;- ad2cpHeaderValue(d, "BEAMCFGLIST,BEAM=1", "THETA")
    frequency &lt;- ad2cpHeaderValue(d, "BEAMCFGLIST,BEAM=1", "FREQ", default=NA)
}
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to ad2cp data: 
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>
</p>

<hr>
<h2 id='addSpine'>Add a Spine to a section Object</h2><span id='topic+addSpine'></span>

<h3>Description</h3>

<p>The purpose of this is to permit plotting with <code>xtype="spine"</code>, so that
the section plot will display the distance of stations projected
onto the spine.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addSpine(section, spine, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addSpine_+3A_section">section</code></td>
<td>
<p>a <a href="#topic+section-class">section</a> object.</p>
</td></tr>
<tr><td><code id="addSpine_+3A_spine">spine</code></td>
<td>
<p>either a list or a data frame, containing numeric items named
<code>longitude</code> and <code>latitude</code>, defining a path along the spine.</p>
</td></tr>
<tr><td><code id="addSpine_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+section-class">section</a> object with a spine added.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(section)
eastern &lt;- subset(section, longitude &lt; (-65))
spine &lt;- list(
    longitude = c(-74.5, -69.2, -55),
    latitude = c(38.6, 36.25, 36.25)
)
easternWithSpine &lt;- addSpine(eastern, spine)
# plot(easternWithSpine, which="map")
# plot(easternWithSpine, xtype="distance", which="temperature")
# plot(easternWithSpine, xtype="spine", which="temperature")

</code></pre>

<hr>
<h2 id='adp'>Sample adp Data</h2><span id='topic+adp'></span>

<h3>Description</h3>

<p>This is degraded subsample of measurements that were made with an
upward-pointing, moored, ADP manufactured by Teledyne-RDI, as part of the St Lawrence
Internal Wave Experiment (SLEIWEX).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(adp)
</code></pre>


<h3>Source</h3>

<p>This file came from the SLEIWEX-2008 experiment.
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(adp)

# Velocity components.  (Note: we should probably trim some bins at top.)
plot(adp)

# Note that tides have moved the mooring.
plot(adp, which = 15:18)


</code></pre>

<hr>
<h2 id='adp_rdi.000'>Sample adp File in RDI Format</h2><span id='topic+adp_rdi.000'></span>

<h3>Description</h3>

<p>Sample adp File in RDI Format
</p>


<h3>See Also</h3>

<p>Other raw datasets: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>read.oce(system.file("extdata", "adp_rdi.000", package="oce"))

</code></pre>

<hr>
<h2 id='adp-class'>Class to Store Acoustic-Doppler Profiler Data</h2><span id='topic+adp-class'></span>

<h3>Description</h3>

<p>This class stores data from acoustic Doppler profilers. Some manufacturers
call these ADCPs, while others call them ADPs; here the shorter form is
used by analogy to ADVs.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>adp</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
The key items stored in this slot include <code>time</code>, <code>distance</code>, and <code>v</code>, along with angles <code>heading</code>, <code>pitch</code> and <code>roll</code>.</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>adp</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
Examples that are of common interest include <code>oceCoordinate</code>, <code>orientation</code>, <code>frequency</code>, and <code>beamAngle</code>.</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>adp</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+adp-class">adp</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+adp-class">adp</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Reading/creating <code>adp</code> objects</h3>

<p>The <code>metadata</code> slot contains various
items relating to the dataset, including source file name, sampling rate,
velocity resolution, velocity maximum value, and so on.  Some of these are
particular to particular instrument types, and prudent researchers will take
a moment to examine the whole contents of the metadata, either in summary
form (with <code>str(adp[["metadata"]])</code>) or in detail (with
<code>adp[["metadata"]]</code>).  Perhaps the most useful general properties are
<code>adp[["bin1Distance"]]</code> (the distance, in metres, from the sensor to
the bottom of the first bin), <code>adp[["cellSize"]]</code> (the cell height, in
metres, in the vertical direction, <em>not</em> along the beam), and
<code>adp[["beamAngle"]]</code> (the angle, in degrees, between beams and an
imaginary centre line that bisects all beam pairs).
</p>
<p>The diagram provided below indicates the coordinate-axis and beam-numbering
conventions for three- and four-beam ADP devices, viewed as though the
reader were looking towards the beams being emitted from the transducers.
</p>
<p><img src="../help/figures/adp_beams.png" width="400" alt="Figure: adp\_beams.png" />
</p>
<p>The bin geometry of a four-beam profiler is illustrated below, for
<code>adp[["beamAngle"]]</code> equal to 20 degrees, <code>adp[["bin1Distance"]]</code>
equal to 2m, and <code>adp[["cellSize"]]</code> equal to 1m.   In the diagram, the
viewer is in the plane containing two beams that are not shown, so the two
visible beams are separated by 40 degrees.  Circles indicate the centres of
the range-gated bins within the beams.  The lines enclosing those circles
indicate the coverage of beams that spread plus and minus 2.5 degrees from
their centreline.
</p>
<p><img src="../help/figures/adpgeometry2.png" width="400" alt="Figure: adpgeometry2.png" />
</p>
<p>Note that <code>adp[["oceCoordinate"]]</code> stores the present coordinate system
of the object, and it has possible values <code>"beam"</code>, <code>"xyz"</code>, <code>"sfm"</code> or
<code>"enu"</code>.  (This should not be confused with
<code>adp[["originalCoordinate"]]</code>, which stores the coordinate system used
in the original data file.)
</p>
<p>The <code>data</code> slot holds some standardized items, and
many that vary from instrument to instrument.  One standard item is
<code>adp[["v"]]</code>, a three-dimensional numeric array of velocities in
m/s.  In this matrix, the first index indicates time, the second bin
number, and the third beam number. The meaning of beams number depends on
whether the object is in beam coordinates, frame coordinates, or earth
coordinates.  For example, if in earth coordinates, then beam 1 is
the eastward component of velocity.
Thus, for example,
</p>
<pre>
library(oce)
data(adp)
t &lt;- adp[["time"]]
d &lt;- adp[["distance"]]
eastward &lt;- adp[["v"]][,,1]
imagep(t, d, eastward, missingColor="gray")
</pre>
<p>plots an image of the eastward component of velocity as a function of time (the x axis)
and distance from sensor (y axis), since the <code>adp</code> dataset is
in earth coordinates. Note the semidurnal tidal signal, and the pattern of missing
data at the ocean surface (gray blotches at the top).
</p>
<p>Corresponding to the velocity array are two arrays of type raw, and
identical dimension, accessed by <code>adp[["a"]]</code> and <code>adp[["q"]]</code>,
holding measures of signal strength and data quality (referred
to as &quot;correlation&quot; in some documentation),
respectively.  (The exact meanings of these depend on the particular type
of instrument, and it is assumed that users will be familiar enough with
instruments to know both the meanings and their practical consequences in
terms of data-quality assessment, etc.)
</p>
<p>In addition to the arrays, there are time-based vectors.  The vector
<code>adp[["time"]]</code> (of length equal to the first index of
<code>adp[["v"]]</code>, etc.) holds times of observation.  Depending on type of
instrument and its configuration, there may also be corresponding vectors
for sound speed (<code>adp[["soundSpeed"]]</code>), pressure
(<code>adp[["pressure"]]</code>), temperature (<code>adp[["temperature"]]</code>),
heading (<code>adp[["heading"]]</code>) pitch (<code>adp[["pitch"]]</code>), and roll
(<code>adp[["roll"]]</code>), depending on the setup of the instrument.
</p>
<p>The precise meanings of the data items depend on the instrument type.  All
instruments have <code>v</code> (for velocity), <code>q</code> (for a measure of data
quality) and <code>a</code> (for a measure of backscatter amplitude, also called
echo intensity).
Teledyne-RDI profilers have an additional item <code>g</code> (for
percent-good).
</p>
<p>VmDas-equipped Teledyne-RDI profilers additional navigation data, with
details listed in the table below; note that the RDI documentation (reference 2) and
the RDI gui use inconsistent names for most items.
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Oce name</strong></td><td style="text-align: left;"> <strong>RDI doc name</strong></td><td style="text-align: left;"> <strong>RDI GUI name</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>avgSpeed</code></td><td style="text-align: left;"> Avg Speed</td><td style="text-align: left;"> Speed/Avg/Mag</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>avgMagnitudeVelocityEast</code></td><td style="text-align: left;"> Avg Mag Vel East</td><td style="text-align: left;"> ?</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>avgMagnitudeVelocityNorth</code></td><td style="text-align: left;"> Avg Mag Vel North</td><td style="text-align: left;"> ?</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>avgTrackMagnetic</code></td><td style="text-align: left;"> Avg Track Magnetic</td><td style="text-align: left;"> Speed/Avg/Dir (?)</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>avgTrackTrue</code></td><td style="text-align: left;"> Avg Track True</td><td style="text-align: left;"> Speed/Avg/Dir (?)</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>avgTrueVelocityEast</code></td><td style="text-align: left;"> Avg True Vel East</td><td style="text-align: left;"> ?</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>avgTrueVelocityNorth</code></td><td style="text-align: left;"> Avg True Vel North</td><td style="text-align: left;"> ?</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>directionMadeGood</code></td><td style="text-align: left;"> Direction Made Good</td><td style="text-align: left;"> Speed/Made Good/Dir</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>firstLatitude</code></td><td style="text-align: left;"> First latitude</td><td style="text-align: left;"> Start Lat</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>firstLongitude</code></td><td style="text-align: left;"> First longitude</td><td style="text-align: left;"> Start Lon</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>firstTime</code></td><td style="text-align: left;"> UTC Time of last fix</td><td style="text-align: left;"> End Time</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lastLatitude</code></td><td style="text-align: left;"> Last latitude</td><td style="text-align: left;"> End Lat</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lastLongitude</code></td><td style="text-align: left;"> Last longitude</td><td style="text-align: left;"> End Lon</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lastTime</code></td><td style="text-align: left;"> UTC Time of last fix</td><td style="text-align: left;"> End Time</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>numberOfHeadingSamplesAveraged</code></td><td style="text-align: left;"> Number heading samples averaged</td><td style="text-align: left;"> ?</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>numberOfMagneticTrackSamplesAveraged</code></td><td style="text-align: left;"> Number of magnetic track samples averaged</td><td style="text-align: left;"> ? </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>numberOfPitchRollSamplesAvg</code></td><td style="text-align: left;"> Number of magnetic track samples averaged</td><td style="text-align: left;"> ? </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>numberOfSpeedSamplesAveraged</code></td><td style="text-align: left;"> Number of speed samples averaged</td><td style="text-align: left;"> ? </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>numberOfTrueTrackSamplesAvg</code></td><td style="text-align: left;"> Number of true track samples averaged</td><td style="text-align: left;"> ? </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>primaryFlags</code></td><td style="text-align: left;"> Primary Flags</td><td style="text-align: left;"> ?</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>shipHeading</code></td><td style="text-align: left;"> Heading</td><td style="text-align: left;"> ?</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>shipPitch</code></td><td style="text-align: left;"> Pitch</td><td style="text-align: left;"> ?</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>shipRoll</code></td><td style="text-align: left;"> Roll</td><td style="text-align: left;"> ?</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>speedMadeGood</code></td><td style="text-align: left;"> Speed Made Good</td><td style="text-align: left;"> Speed/Made Good/Mag</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>speedMadeGoodEast</code></td><td style="text-align: left;"> Speed MG East</td><td style="text-align: left;"> ?</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>speedMadeGoodNorth</code></td><td style="text-align: left;"> Speed MG North</td><td style="text-align: left;"> ?</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>For Teledyne-RDI profilers, there are four three-dimensional arrays
holding beamwise data.  In these, the first index indicates time, the
second bin number, and the third beam number (or coordinate number, for
data in <code>xyz</code>, <code>sfm</code>, <code>enu</code> or <code>other</code> coordinate systems).  In
the list below, the quoted phrases are quantities as defined in Figure 9
of reference 1.
</p>

<ul>
<li> <p><code>v</code> is <code>velocity</code> in m/s, inferred from two-byte signed
integer values (multiplied by the scale factor that is stored in
<code>velocityScale</code> in the metadata).
</p>
</li>
<li> <p><code>q</code> is &quot;correlation magnitude&quot; a one-byte quantity stored
as type <code>raw</code> in the object. The values may range from 0 to 255.
</p>
</li>
<li> <p><code>a</code> is &quot;backscatter amplitude&quot;, also known as &quot;echo
intensity&quot; a one-byte quantity stored as type <code>raw</code> in the object.
The values may range from 0 to 255.
</p>
</li>
<li> <p><code>g</code> is &quot;percent good&quot; a one-byte quantity stored as <code>raw</code>
in the object.  The values may range from 0 to 100.
</p>
</li></ul>

<p>Finally, there is a vector <code>adp[["distance"]]</code> that indicates the bin
distances from the sensor, measured in metres along an imaginary centre
line bisecting beam pairs.  The length of this vector equals
<code>dim(adp[["v"]])[2]</code>.
</p>


<h3>Teledyne-RDI Sentinel V ADCPs</h3>

<p>As of 2016-09-27 there is
provisional support for the TRDI &quot;SentinelV&quot; ADCPs, which are 5
beam ADCPs with a vertical centre beam. Relevant vertical beam
fields are called <code>adp[["vv"]]</code>, <code>adp[["va"]]</code>,
<code>adp[["vq"]]</code>, and <code>adp[["vg"]]</code> in analogy with the
standard 4-beam fields.
</p>


<h3>Accessing and altering information within <a href="#topic+adp-class">adp</a> objects</h3>

<p><em>Extracting values</em> Matrix data may be accessed as illustrated
above, e.g.  or an adp object named <code>adv</code>, the data are provided by
<code>adp[["v"]]</code>, <code>adp[["a"]]</code>, and <code>adp[["q"]]</code>.  As a
convenience, the last two of these can be accessed as numeric (as opposed to
raw) values by e.g.  <code>adp[["a", "numeric"]]</code>.  The vectors are accessed
in a similar way, e.g. <code>adp[["heading"]]</code>, etc.  Quantities in the
<code>metadata</code> slot are also available by name, e.g.
<code>adp[["velocityResolution"]]</code>, etc.
</p>
<p><em>Assigning values.</em> This follows the standard form, e.g. to increase
all velocity data by 1 cm/s, use <code>adp[["v"]] &lt;- 0.01 + adp[["v"]]</code>.
</p>
<p><em>Overview of contents</em> The <code>show</code> method (e.g.
<code>show(d)</code>) displays information about an ADP object named <code>d</code>.
</p>


<h3>Dealing with suspect data</h3>

<p>There are many possibilities for confusion
with <code>adp</code> devices, owing partly to the flexibility that manufacturers
provide in the setup.  Prudent users will undertake many tests before trusting
the details of the data.  Are mean currents in the expected direction, and of
the expected magnitude, based on other observations or physical constraints?
Is the phasing of currents as expected?  If the signals are suspect, could an
incorrect scale account for it?  Could the transformation matrix be incorrect?
Might the data have exceeded the maximum value, and then &ldquo;wrapped around&rdquo; to
smaller values?  Time spent on building confidence in data quality is seldom
time wasted.
</p>


<h3>References</h3>


<ol>
<li><p> Teledyne-RDI, 2007.
<em>WorkHorse commands and output data format.</em>
P/N 957-6156-00 (November 2007).
</p>
</li>
<li><p> Teledyne-RDI, 2012. <em>VmDas User's Guide, Ver. 1.46.5</em>.
</p>
</li></ol>



<h3>See Also</h3>

<p>A file containing ADP data is usually recognized by Oce, and so
<code><a href="#topic+read.oce">read.oce()</a></code> will usually read the data.  If not, one may use the
general ADP function <code><a href="#topic+read.adp">read.adp()</a></code> or specialized variants
<code><a href="#topic+read.adp.rdi">read.adp.rdi()</a></code>, <code><a href="#topic+read.adp.nortek">read.adp.nortek()</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp()</a></code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek()</a></code> or <code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial()</a></code>.
</p>
<p>ADP data may be plotted with <code><a href="#topic+plot+2Cadp-method">plot,adp-method()</a></code>, which is a
generic function so it may be called simply as <code>plot</code>.
</p>
<p>Statistical summaries of ADP data are provided by the generic function
<code>summary</code>, while briefer overviews are provided with <code>show</code>.
</p>
<p>Conversion from beam to xyz coordinates may be done with
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp()</a></code>, and from xyz to enu (east north up) may be done
with <code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp()</a></code>.  <code><a href="#topic+toEnuAdp">toEnuAdp()</a></code> may be used to
transfer either beam or xyz to enu.  Enu may be converted to other coordinates
(e.g. aligned with a coastline) with <code><a href="#topic+enuToOtherAdp">enuToOtherAdp()</a></code>.
</p>
<p>Other classes provided by oce: 
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='adpAd2cpFileTrim'>Trim an AD2CP File</h2><span id='topic+adpAd2cpFileTrim'></span>

<h3>Description</h3>

<p>Create an AD2CP file by copying the first <code>n</code> data chunks (regions starting
with <code>0xa5</code>, etc) of another such file. This can be useful in supplying
small sample files for bug reports.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adpAd2cpFileTrim(infile, n = 100L, outfile, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adpAd2cpFileTrim_+3A_infile">infile</code></td>
<td>
<p>name of an AD2CP file.</p>
</td></tr>
<tr><td><code id="adpAd2cpFileTrim_+3A_n">n</code></td>
<td>
<p>integer indicating the number of data chunks to keep. The default is
to keep 100 chunks, a common choice for sample files.</p>
</td></tr>
<tr><td><code id="adpAd2cpFileTrim_+3A_outfile">outfile</code></td>
<td>
<p>optional name of the new AD2CP file to be created. If this is not
supplied, a default is used, by adding <code style="white-space: pre;">&#8288;_trimmed&#8288;</code> to the base filename, e.g.
if <code>infile</code> is <code>"a.ad2cp"</code> then <code>outfile</code> will be <code>a_trimmed.ad2cp</code>.</p>
</td></tr>
<tr><td><code id="adpAd2cpFileTrim_+3A_debug">debug</code></td>
<td>
<p>an integer value indicating the level of debugging. If
this is 1L, then a brief indication is given of the processing steps. If it
is &gt; 1L, then information is given about each data chunk, which can yield
very extensive output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>adpAd2cpFileTrim()</code> returns the name of the output file, <code>outfile</code>, as
provided or constructed.
</p>


<h3>Sample of Usage</h3>

<pre>
# Can only be run by the developer, since it uses a private file.
f  &lt;- "~/Dropbox/oce_secret_data/ad2cp/byg_trimmed.ad2cp"
if (file.exists(f))
    adpAd2cpFileTrim(f, 100L) # this file is already trimmed to 200 chunks
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to ad2cp data: 
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>
</p>
<p>Other functions that trim data files: 
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+oceFileTrim">oceFileTrim</a>()</code>
</p>

<hr>
<h2 id='adpConvertRawToNumeric'>Convert Raw to Numeric Values in an adp Object</h2><span id='topic+adpConvertRawToNumeric'></span>

<h3>Description</h3>

<p>Convert variables in an <a href="#topic+adp-class">adp</a> object from raw to numeric format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adpConvertRawToNumeric(
  object = NULL,
  variables = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adpConvertRawToNumeric_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="adpConvertRawToNumeric_+3A_variables">variables</code></td>
<td>
<p>variables stored in an <a href="#topic+adp-class">adp</a> object that has
the same dimensional as <code>v</code> and is stored in a raw format.</p>
</td></tr>
<tr><td><code id="adpConvertRawToNumeric_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>adpConvertRawToNumeric</code> returns an <a href="#topic+adp-class">adp</a> object whose specified
variables have been converted from raw to numerical format.
</p>


<h3>Author(s)</h3>

<p>Jaimie Harbin and Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adp)
adp[["a"]][, , 1][, 1]
ADP &lt;- adpConvertRawToNumeric(adp)
ADP[["a"]][, , 1][, 1]
</code></pre>

<hr>
<h2 id='adpEnsembleAverage'>Ensemble Average an adp Object in Time</h2><span id='topic+adpEnsembleAverage'></span>

<h3>Description</h3>

<p>Ensemble averaging of <code>adp</code> objects is often necessary to
reduce the uncertainty in velocity estimates from single
pings. Many types of ADPs can be configured to perform the
ensemble averaging during the data collection, due to memory
limitations for long deployments. In cases where the instrument is
not memory limited, it may be desirable to perform the ensemble
averaging during post-processing, thereby reducing the overall
size of the data set and decreasing the uncertainty of the
velocity estimates (by averaging out Doppler noise).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adpEnsembleAverage(x, n = 5, leftover = FALSE, na.rm = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adpEnsembleAverage_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="adpEnsembleAverage_+3A_n">n</code></td>
<td>
<p>number of pings to average together.</p>
</td></tr>
<tr><td><code id="adpEnsembleAverage_+3A_leftover">leftover</code></td>
<td>
<p>a logical value indicating how to proceed in cases
where <code>n</code> does not divide evenly into the number of ensembles
in <code>x</code>. If <code>leftover</code> is <code>FALSE</code> (the default) then any extra
ensembles at the end of <code>x</code> are ignored. Otherwise, they are used
to create a final ensemble in the returned value.</p>
</td></tr>
<tr><td><code id="adpEnsembleAverage_+3A_na.rm">na.rm</code></td>
<td>
<p>a logical value indicating whether NA values should be stripped
before the computation proceeds</p>
</td></tr>
<tr><td><code id="adpEnsembleAverage_+3A_...">...</code></td>
<td>
<p>extra arguments to be passed to the <code>mean()</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <a href="#topic+adp-class">adp</a> object with ensembles averaged as specified. E.g. for
an <code>adp</code> object with 100 pings and <code>n=5</code> the number of rows of the data arrays
will be reduced by a factor of 5.
</p>


<h3>Author(s)</h3>

<p>Clark Richards and Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adp)
adpAvg &lt;- adpEnsembleAverage(adp, n = 2)
plot(adpAvg)

</code></pre>

<hr>
<h2 id='adpFlagPastBoundary'>Flag adp Data Past Water Column Boundary</h2><span id='topic+adpFlagPastBoundary'></span>

<h3>Description</h3>

<p>Flag variables with the same dimension of <code>v</code> in an <a href="#topic+adp-class">adp</a> object that
are beyond the water column boundary while retaining existing flags.
Currently, this operation can only be performed on <a href="#topic+adp-class">adp</a> objects that
contain bottom ranges. Commonly, <code><a href="#topic+handleFlags">handleFlags()</a></code> would then be used to remove
such data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adpFlagPastBoundary(
  x = NULL,
  fields = NULL,
  df = 20,
  trim = 0.15,
  good = 1,
  bad = 4,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adpFlagPastBoundary_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object containing bottom ranges.</p>
</td></tr>
<tr><td><code id="adpFlagPastBoundary_+3A_fields">fields</code></td>
<td>
<p>a variable contained within <code>x</code> indicating which field to flag.
If NULL (the default) then <code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary()</a></code> applies itself to all flag
fields that have the same dimensionality as <code>v</code> in the <code>data</code> slot.</p>
</td></tr>
<tr><td><code id="adpFlagPastBoundary_+3A_df">df</code></td>
<td>
<p>the degrees of freedom to use during the smoothing spline operation.</p>
</td></tr>
<tr><td><code id="adpFlagPastBoundary_+3A_trim">trim</code></td>
<td>
<p>a scale factor for boundary trimming (see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="adpFlagPastBoundary_+3A_good">good</code></td>
<td>
<p>number stored in flags to indicate good data.</p>
</td></tr>
<tr><td><code id="adpFlagPastBoundary_+3A_bad">bad</code></td>
<td>
<p>number stored in flags to indicate bad data.</p>
</td></tr>
<tr><td><code id="adpFlagPastBoundary_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the object's <code>oceCoordinate</code> is <code>"beam"</code>, this works by using
<code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code> on the time-dependent bottom ranges, beam-by-beam. If
<code>oceCoordinate</code> is <code>"enu"</code>, <code>"xyz"</code>, or <code>"other"</code>, a <code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code> is
used on a time-dependent bottom range averaged across all the beams. The <code>df</code>
value of the present function is passed to <code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code>, as a way to
control smoothness.  Once this is done, data within distance of <code class="reqn">1-trim</code>
multiplied by the bottom range are flagged as being bad.  The default value
of <code>trim</code> is 0.15, which is close to the value (0.134) of
<code class="reqn">1-cos(angle*pi/180)</code>, with angle=30 as the beam angle in degrees.
</p>


<h3>Value</h3>

<p><code>adpFlagPastBoundary</code> returns an <a href="#topic+adp-class">adp</a> object with flags
adjusted in the specified fields if data are beyond the water column boundary.
</p>


<h3>Author(s)</h3>

<p>Jaimie Harbin, Clark Richards, and Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='adpRdiFileTrim'>Trim an RDI adp File</h2><span id='topic+adpRdiFileTrim'></span>

<h3>Description</h3>

<p>Create an RDI adp file by copying the first <code>n</code> data chunks (starting with
byte 0x7f 0x7f) of another such file. This can be useful in supplying small
sample files for bug reports.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adpRdiFileTrim(infile, n = 100L, outfile, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adpRdiFileTrim_+3A_infile">infile</code></td>
<td>
<p>name of an RDI file.</p>
</td></tr>
<tr><td><code id="adpRdiFileTrim_+3A_n">n</code></td>
<td>
<p>integer indicating the number of data chunks to keep. The default is
to keep 100 chunks, a common choice for sample files.</p>
</td></tr>
<tr><td><code id="adpRdiFileTrim_+3A_outfile">outfile</code></td>
<td>
<p>optional name of the new RDI file to be created. If this is not
supplied, a default is used, by adding <code style="white-space: pre;">&#8288;_trimmed&#8288;</code> to the base filename, e.g.
if <code>infile</code> is <code>"a.000"</code> then <code>outfile</code> will be <code>a_trimmed.000</code>.</p>
</td></tr>
<tr><td><code id="adpRdiFileTrim_+3A_debug">debug</code></td>
<td>
<p>an integer value indicating the level of debugging. If
this is 1L, then a brief indication is given of the processing steps. If it
is &gt; 1L, then information is given about each data chunk, which can yield
very extensive output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>adpRdiFileTrim()</code> returns the name of the output file, <code>outfile</code>, as
provided or constructed.
</p>


<h3>Sample of Usage</h3>

<pre>
# Can only be run by the developer, since it uses a private file.
file  &lt;- "~/data/archive/sleiwex/2008/moorings/m09/adp/rdi_2615/raw/adp_rdi_2615.000"
if (file.exists(file)) {
    adpRdiFileTrim(file, 9L, "test.000")
}
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that trim data files: 
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+oceFileTrim">oceFileTrim</a>()</code>
</p>

<hr>
<h2 id='adv'>Sample adv Data</h2><span id='topic+adv'></span>

<h3>Description</h3>

<p>This <a href="#topic+adv-class">adv</a> object is a sampling of measurements made with a
Nortek Vector acoustic Doppler velocimeter deployed as part of the St Lawrence
Internal Wave Experiment (SLEIWEX).  Various identifying features have been
redacted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(adv)
</code></pre>


<h3>Source</h3>

<p>This file came from the SLEIWEX-2008 experiment.
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(adv)

# Velocity time-series
plot(adv)

# Spectrum of upward component of velocity, with ``turbulent'' reference line
s &lt;- spectrum(adv[["v"]][, 3], plot = FALSE)
plot(log10(s$freq), log10(s$spec), type = "l")
for (a in seq(-20, 20, by = 1)) {
    abline(a = a, b = -5 / 3, col = "gray", lty = "dotted")
}


</code></pre>

<hr>
<h2 id='adv-class'>Class to Store Acoustic-Doppler Velocimeter Data</h2><span id='topic+adv-class'></span>

<h3>Description</h3>

<p>This class holds data from acoustic-Doppler velocimeters.
</p>


<h3>Details</h3>

<p>A file containing ADV data is usually recognized by Oce, and so
<code><a href="#topic+read.oce">read.oce()</a></code> will usually read the data.  If not, one may use the
general ADV function <code><a href="#topic+read.adv">read.adv()</a></code> or specialized variants
<code><a href="#topic+read.adv.nortek">read.adv.nortek()</a></code>, <code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr()</a></code> or
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text()</a></code>.
</p>
<p>ADV data may be plotted with <code><a href="#topic+plot+2Cadv-method">plot,adv-method()</a></code> function, which is a
generic function so it may be called simply as <code>plot(x)</code>, where
<code>x</code> is an <a href="#topic+adv-class">adv</a> object.
</p>
<p>Statistical summaries of ADV data are provided by the generic function
<code><a href="#topic+summary+2Cadv-method">summary,adv-method()</a></code>.
</p>
<p>Conversion from beam to xyz coordinates may be done with
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv()</a></code>, and from xyz to enu (east north up) may be done
with <code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv()</a></code>.  <code><a href="#topic+toEnuAdv">toEnuAdv()</a></code> may be used to
transfer either beam or xyz to enu.  Enu may be converted to other
coordinates (e.g. aligned with a coastline) with
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv()</a></code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>adv</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
The key items stored in this slot include <code>time</code> and <code>v</code>.</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>adv</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
Examples that are of common interest include <code>frequency</code>, <code>oceCordinate</code>, and <code>frequency</code>.</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>adv</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+adv-class">adv</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+adv-class">adv</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(adv)
adv[["v"]] &lt;- 0.001 + adv[["v"]] # add 1mm/s to all velocity components

</code></pre>

<hr>
<h2 id='advSontekAdrFileTrim'>Trim a Sontek ADR adv File</h2><span id='topic+advSontekAdrFileTrim'></span>

<h3>Description</h3>

<p>Create a Sontek ADR adv (acoustic Doppler velocimeter) file by copying the
header plus the first <code>n</code> data chunks (recognized by the three-byte sequence
<code>0xA5</code>, <code>0x11</code>, &lsquo;0x3c&rsquo;) into a new file. This can be useful in supplying
small sample files for bug reports.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>advSontekAdrFileTrim(infile, n = 100, outfile, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="advSontekAdrFileTrim_+3A_infile">infile</code></td>
<td>
<p>name of a Sontek ADR adp file.</p>
</td></tr>
<tr><td><code id="advSontekAdrFileTrim_+3A_n">n</code></td>
<td>
<p>integer indicating the number of data chunks to keep. The default is
to keep 100 chunks, a common choice for sample files.</p>
</td></tr>
<tr><td><code id="advSontekAdrFileTrim_+3A_outfile">outfile</code></td>
<td>
<p>optional name of the new Sontek ADR adp file to be created. If this is not
supplied, a default is used, by adding <code style="white-space: pre;">&#8288;_trimmed&#8288;</code> to the base filename, e.g.
if <code>infile</code> is <code>"x.adr"</code> then <code>outfile</code> will be <code>x_trimmed.adr</code>.</p>
</td></tr>
<tr><td><code id="advSontekAdrFileTrim_+3A_debug">debug</code></td>
<td>
<p>an integer value indicating the level of debugging. If
this is 1L, then a brief indication is given of the processing steps. If it
is &gt; 1L, then information is given about each data chunk, which can yield
very extensive output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>advSontekAdrFileTrim()</code> returns the name of the output file, <code>outfile</code>, as
provided or constructed.
</p>


<h3>See Also</h3>

<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that trim data files: 
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+oceFileTrim">oceFileTrim</a>()</code>
</p>

<hr>
<h2 id='airRho'>Air Density</h2><span id='topic+airRho'></span>

<h3>Description</h3>

<p>Compute <code class="reqn">\rho</code>, the <em>in-situ</em> density of dry air.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>airRho(temperature, pressure, humidity)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="airRho_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature, in <code class="reqn">^\circ</code>C.</p>
</td></tr>
<tr><td><code id="airRho_+3A_pressure">pressure</code></td>
<td>
<p>numeric value for pressure in Pa (<em>not</em> the kPa used in public
weather forecasts).</p>
</td></tr>
<tr><td><code id="airRho_+3A_humidity">humidity</code></td>
<td>
<p>ignored at present</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This will eventually be a proper equation of state, but for now it just
uses a dry-air formula posted on wikipedia (i.e. not trustworthy).
</p>


<h3>Value</h3>

<p><em>In-situ</em> dry-air density, in kg/m<code class="reqn">^3</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li> <p><code style="white-space: pre;">&#8288;https://en.wikipedia.org/wiki/Density_of_air&#8288;</code>
</p>
</li>
<li><p> National Oceanographic and Atmospheric Agency, 1976.  U.S.
Standard Atmosphere, 1976.  NOAA-S/T 76-1562.  (A PDF of this document may
be available at
<code style="white-space: pre;">&#8288;http://ntrs.nasa.gov/archive/nasa/casi.ntrs.nasa.gov/19770009539_1977009539.pdf&#8288;</code>
or
<code style="white-space: pre;">&#8288;http://www.dtic.mil/cgi-bin/GetTRDoc?Location=U2&amp;doc=GetTRDoc.pdf&amp;AD=ADA035728&#8288;</code>
although neither link has proven to be reliable.)
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>degC &lt;- seq(0, 30, length.out = 100)
p &lt;- seq(98, 102, length.out = 100) * 1e3
contour(x = degC, y = p, z = outer(degC, p, airRho), labcex = 1)
</code></pre>

<hr>
<h2 id='amsr'>Sample amsr Data (Near Nova Scotia)</h2><span id='topic+amsr'></span>

<h3>Description</h3>

<p>This is a three-day composite satellite image for
July 27, 2023, trimmed to show waters south and east of
Nova Scotia, using code provide in the &ldquo;Details&rdquo;
section.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(amsr)
</code></pre>


<h3>Details</h3>

<p>The following code was used to create this dataset.
</p>
<pre>
library(oce)
amsr &lt;- read.amsr(download.amsr(2023, 7, 27, destdir="~/data/amsr"))
amsr &lt;- subset(amsr, -71 &lt; longitude &amp; longitude &lt; -60, debug=2)
amsr &lt;- subset(amsr,  36 &lt; latitude  &amp;  latitude &lt;  45, debug=2)
</pre>


<h3>See Also</h3>

<p>Other satellite datasets provided with oce: 
<code><a href="#topic+landsat">landsat</a></code>
</p>
<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to amsr data: 
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method</a></code>,
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+read.amsr">read.amsr</a>()</code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+summary+2Camsr-method">summary,amsr-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(coastlineWorld)
data(amsr)
plot(amsr, "SST")
lines(coastlineWorld[["longitude"]], coastlineWorld[["latitude"]])

</code></pre>

<hr>
<h2 id='amsr-class'>Class to Store AMSR-2 Satellite Data</h2><span id='topic+amsr-class'></span>

<h3>Description</h3>

<p>This class stores data from the AMSR-2 satellite.
</p>


<h3>Details</h3>

<p>The Advanced Microwave Scanning Radiometer (AMSR-2) is in current operation on
the Japan Aerospace Exploration Agency (JAXA) GCOM-W1 space craft, launched in
May 2012. Data are processed by Remote Sensing Systems. The satellite
completes an ascending and descending pass during local daytime and nighttime
hours respectively. Each daily file contains 7 daytime and 7 nighttime
maps of variables named as follows within the <code>data</code>
slot of amsr objects: <code>timeDay</code>,
<code>SSTDay</code>, <code>LFwindDay</code> (wind at 10m sensed in
the 10.7GHz band), <code>MFwindDay</code> (wind at 10m sensed at 18.7GHz),
<code>vaporDay</code>, <code>cloudDay</code>, and <code>rainDay</code>, along with
similarly-named items that end in <code>Night</code>.
See reference 1 for additional information on the instrument, how
to cite the data source in a paper, etc.
</p>
<p>The bands are stored in <code><a href="base.html#topic+raw">raw()</a></code> form, to save storage. The accessor
function <code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code> can provide these values in <code>raw</code>
form or in physical units; <code><a href="#topic+plot+2Camsr-method">plot,amsr-method()</a></code>, and
<code><a href="#topic+summary+2Camsr-method">summary,amsr-method()</a></code> work with physical units.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>amsr</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>amsr</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
Examples that are of common interest include  <code>longitude</code> and <code>latitude</code>, which define the grid.</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>amsr</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+amsr-class">amsr</a> objects (see <code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+amsr-class">amsr</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Chantelle Layton
</p>


<h3>References</h3>


<ol>
<li><p> Information on the satellite, how to cite the data, etc. is
provided at <code style="white-space: pre;">&#8288;http://www.remss.com/missions/amsr/&#8288;</code>.
</p>
</li>
<li><p> A simple interface for viewing and downloading data is at
<code style="white-space: pre;">&#8288;http://images.remss.com/amsr/amsr2_data_daily.html&#8288;</code>.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other classes holding satellite data: 
<code><a href="#topic+g1sst-class">g1sst-class</a></code>,
<code><a href="#topic+landsat-class">landsat-class</a></code>,
<code><a href="#topic+satellite-class">satellite-class</a></code>
</p>
<p>Other things related to amsr data: 
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method</a></code>,
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+read.amsr">read.amsr</a>()</code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+summary+2Camsr-method">summary,amsr-method</a></code>
</p>

<hr>
<h2 id='angle2hms'>Convert Astronomical Angle in Degrees to Hours, Minutes and Seconds</h2><span id='topic+angle2hms'></span>

<h3>Description</h3>

<p>The purpose of <a href="#topic+angle2hms">angle2hms</a> is to facilitate comparison of
<code>rightAscension</code> angles computed by <code><a href="#topic+sunAngle">sunAngle()</a></code> and <code><a href="#topic+moonAngle">moonAngle()</a></code>
with angles reported in astronomical sources and software, which
often employ an hour-minute-second notation. In that notation,
decimal hour is computed as 24/360 times the angle in degrees,
and from that decimal hour are compute integer hour and minute
values, plus a decimal second value. It is common in the
astronomical literature to use strings to represent the results,
e.g. with <code class="reqn">11^h40^m48^s.10</code> for the
value used in the &ldquo;Examples&rdquo;; see Chapter 1 of
Meeus (1991) for more on angle calculation and representation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>angle2hms(angle)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="angle2hms_+3A_angle">angle</code></td>
<td>
<p>numerical value giving an angle in degrees</p>
</td></tr>
</table>


<h3>Value</h3>

<p><a href="#topic+angle2hms">angle2hms</a> returns a list containing
values <code>time</code> (a numerical value for decimal hour, between 0 and 24),
<code>hour</code>, <code>minute</code>, and <code>second</code> (the last of which may have a fractional
part), and <code>string</code>, a character value indicates the time in hour-minute-second
notation, with the second part to two decimal places and intervening <code>h</code>,
<code>m</code> and <code>s</code> characters between the units.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ul>
<li><p> Meeus, Jean. Astronomical Algorithms. Second Edition.
Richmond, Virginia, USA: Willmann-Bell, 1991.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other things related to astronomy: 
<code><a href="#topic+eclipticalToEquatorial">eclipticalToEquatorial</a>()</code>,
<code><a href="#topic+equatorialToLocalHorizontal">equatorialToLocalHorizontal</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+moonAngle">moonAngle</a>()</code>,
<code><a href="#topic+siderealTime">siderealTime</a>()</code>,
<code><a href="#topic+sunAngle">sunAngle</a>()</code>,
<code><a href="#topic+sunDeclinationRightAscension">sunDeclinationRightAscension</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A randomly-chosen example on page 99 of Meeus (1991).
angle2hms(177.74208) # string component 11h50m58s.10

</code></pre>

<hr>
<h2 id='angleRemap'>Convert Angle From 0:360 to -180:180 Convention</h2><span id='topic+angleRemap'></span>

<h3>Description</h3>

<p>This is mostly used for instrument heading angles, in cases where the
instrument is aligned nearly northward, so that small variations in heading
(e.g. due to mooring motion) can yield values that swing from small angles
to large angles, because of the modulo-360 cut point.
The method is to use the cosine and sine of the angle in order to find &quot;x&quot;
and &quot;y&quot; values on a unit circle, and then to use <code><a href="base.html#topic+atan2">atan2()</a></code> to
infer the angles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>angleRemap(theta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="angleRemap_+3A_theta">theta</code></td>
<td>
<p>an angle (in degrees) that is in the range from 0 to 360
degrees</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of angles, in the range -180 to 180.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
# fake some heading data that lie near due-north (0 degrees)
n &lt;- 20
heading &lt;- 360 + rnorm(n, sd = 10)
heading &lt;- ifelse(heading &gt; 360, heading - 360, heading)
x &lt;- 1:n
plot(x, heading, ylim = c(-10, 360), type = "l", col = "lightgray", lwd = 10)
lines(x, angleRemap(heading))
</code></pre>

<hr>
<h2 id='applyMagneticDeclination'>Alter an Object to Account for Magnetic Declination (Generic)</h2><span id='topic+applyMagneticDeclination'></span>

<h3>Description</h3>

<p>Current-measuring instruments that infer flow direction using magnetic
compasses require a correction for magnetic declination, in order to infer
currents with x and y oriented eastward and northward, respectively.
<code><a href="#topic+applyMagneticDeclination">applyMagneticDeclination()</a></code> is a generic function that handles this task by
altering velocity components (and heading values, if they exist).  It works
for objects of the <a href="#topic+cm-class">cm</a>, <a href="#topic+adp-class">adp</a> and <a href="#topic+adv-class">adv</a> and <a href="#topic+cm-class">cm</a>
classes by calling <code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method()</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method()</a></code>, or
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method()</a></code>, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>applyMagneticDeclination(object = "oce", declination = "ANY", debug = "ANY")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="applyMagneticDeclination_+3A_object">object</code></td>
<td>
<p>an object of <a href="#topic+cm-class">cm</a>, <a href="#topic+adp-class">adp</a>, or <a href="#topic+adv-class">adv</a> class.</p>
</td></tr>
<tr><td><code id="applyMagneticDeclination_+3A_declination">declination</code></td>
<td>
<p>numeric value holding magnetic declination in degrees,
positive for clockwise from north.</p>
</td></tr>
<tr><td><code id="applyMagneticDeclination_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned value is a copy of <code>object</code> that has been modified in 4 ways.
(1) the horizontal components of velocity are rotated clockwise by
<code>declination</code> degrees.  (2) If the object holds heading values, then
<code>declination</code> is added to them. (3) The <code>north</code> item in the <code>metadata</code> slot
is set to <code>"geographic"</code>, and a warning is issued if this was also the value
in <code>object</code>.  (4) The <code>declination</code> item in the <code>metadata</code> slot is set to
the value supplied to this function.
</p>


<h3>Value</h3>

<p>an object of the same class as <code>object</code>, modified as described
in &ldquo;Details&rdquo;.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, aided, for the <a href="#topic+adp-class">adp</a> and <a href="#topic+adv-class">adv</a> variants,
by Clark Richards and Jaimie Harbin.
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+magneticField">magneticField()</a></code> to determine the declination,
inclination and intensity at a given spot on the world, at a given time.
</p>
<p>Other things related to magnetism: 
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Coce-method">applyMagneticDeclination,oce-method</a></code>,
<code><a href="#topic+magneticField">magneticField</a>()</code>
</p>

<hr>
<h2 id='applyMagneticDeclination+2Cadp-method'>Alter an adp Object to Account for Magnetic Declination</h2><span id='topic+applyMagneticDeclination+2Cadp-method'></span>

<h3>Description</h3>

<p>Acoustic-Doppler profiling instruments that infer direction using magnetic
compasses to determine current direction need to have a correction applied
for magnetic declination, if the goal is to infer currents with x and y
oriented eastward and northward, respectively.  This is what the present
function does (see &ldquo;Details&rdquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adp'
applyMagneticDeclination(
  object = "oce",
  declination = 0,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="applyMagneticDeclination+2B2Cadp-method_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="applyMagneticDeclination+2B2Cadp-method_+3A_declination">declination</code></td>
<td>
<p>numeric value holding magnetic declination in degrees,
positive for clockwise from north.</p>
</td></tr>
<tr><td><code id="applyMagneticDeclination+2B2Cadp-method_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned value is a copy of <code>object</code> that has been modified in 4 ways.
(1) the horizontal components of velocity are rotated clockwise by
<code>declination</code> degrees.  (2) If the object holds heading values, then
<code>declination</code> is added to them. (3) The <code>north</code> item in the <code>metadata</code> slot
is set to <code>"geographic"</code>, and a warning is issued if this was also the value
in <code>object</code>.  (4) The <code>declination</code> item in the <code>metadata</code> slot is set to
the value supplied to this function.
</p>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object, modified as outlined  in &ldquo;Description&rdquo;.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, aided by Clark Richards and Jaimie Harbin.
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+magneticField">magneticField()</a></code> to determine the declination,
inclination and intensity at a given spot on the world, at a given time.
</p>
<p>Other things related to magnetism: 
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Coce-method">applyMagneticDeclination,oce-method</a></code>,
<code><a href="#topic+applyMagneticDeclination">applyMagneticDeclination</a>()</code>,
<code><a href="#topic+magneticField">magneticField</a>()</code>
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Transform beam coordinate to xyx, then to enu with respect to
# magnetic north, and then to geographic north.
library(oce)
file &lt;- system.file("extdata", "adp_rdi.000", package = "oce")
lon &lt;- -69.73433
lat &lt;- 47.88126
beam &lt;- read.oce(file, from = 1, to = 4, longitude = lon, latitude = lat)
dec &lt;- magneticField(lon, lat, beam[["time"]][1])$declination
xyz &lt;- beamToXyzAdp(beam)
# Here, we tell xyzToEnuAdp() not to set a declination,
# so enuMag has metadata$north equal to "magnetic".  We could
# also skip the use of applyMagneticDeclination() by supplying
# the known declination to xyzToEnuAdp().
enuMag &lt;- xyzToEnuAdp(xyz, declination = NULL)
enuGeo &lt;- applyMagneticDeclination(enuMag, declination = dec)

</code></pre>

<hr>
<h2 id='applyMagneticDeclination+2Cadv-method'>Alter an adv Object to Account for Magnetic Declination</h2><span id='topic+applyMagneticDeclination+2Cadv-method'></span>

<h3>Description</h3>

<p>Acoustic-Doppler velocimetry instruments that infer direction using magnetic
compasses need to have a correction applied for magnetic declination, if the
goal is to infer currents with x and y oriented eastward and northward,
respectively.  This is what the present function does (see &ldquo;Details&rdquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adv'
applyMagneticDeclination(
  object = "oce",
  declination = 0,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="applyMagneticDeclination+2B2Cadv-method_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="applyMagneticDeclination+2B2Cadv-method_+3A_declination">declination</code></td>
<td>
<p>numeric value holding magnetic declination in degrees,
positive for clockwise from north.</p>
</td></tr>
<tr><td><code id="applyMagneticDeclination+2B2Cadv-method_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned value is a copy of <code>object</code> that has been modified in 4 ways.
(1) the horizontal components of velocity are rotated clockwise by
<code>declination</code> degrees.  (2) If the object holds heading values, then
<code>declination</code> is added to them. (3) The <code>north</code> item in the <code>metadata</code> slot
is set to <code>"geographic"</code>, and a warning is issued if this was also the value
in <code>object</code>.  (4) The <code>declination</code> item in the <code>metadata</code> slot is set to
the value supplied to this function.
</p>


<h3>Value</h3>

<p>A <a href="#topic+adv-class">adv</a> object, adjusted as outlined in &ldquo;Details&rdquo;.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, aided by Clark Richards and Jaimie Harbin.
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+magneticField">magneticField()</a></code> to determine the declination,
inclination and intensity at a given spot on the world, at a given time.
</p>
<p>Other things related to magnetism: 
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Coce-method">applyMagneticDeclination,oce-method</a></code>,
<code><a href="#topic+applyMagneticDeclination">applyMagneticDeclination</a>()</code>,
<code><a href="#topic+magneticField">magneticField</a>()</code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='applyMagneticDeclination+2Ccm-method'>Alter a cm Object to Account for Magnetic Declination</h2><span id='topic+applyMagneticDeclination+2Ccm-method'></span>

<h3>Description</h3>

<p>Current-meter (<code>cm</code>) instruments determine directions from onboard compasses,
so interpreting velocity components in geographical coordinates requires that
magnetic declination be taken into account.  This is what the present
function does (see &ldquo;Details&rdquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'cm'
applyMagneticDeclination(
  object = "oce",
  declination = 0,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="applyMagneticDeclination+2B2Ccm-method_+3A_object">object</code></td>
<td>
<p>a <a href="#topic+cm-class">cm</a> object.</p>
</td></tr>
<tr><td><code id="applyMagneticDeclination+2B2Ccm-method_+3A_declination">declination</code></td>
<td>
<p>numeric value holding magnetic declination in degrees,
positive for clockwise from north.</p>
</td></tr>
<tr><td><code id="applyMagneticDeclination+2B2Ccm-method_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned value is a copy of <code>object</code> that has been modified in 4 ways.
(1) the horizontal components of velocity are rotated clockwise by
<code>declination</code> degrees.  (2) If the object holds heading values, then
<code>declination</code> is added to them. (3) The <code>north</code> item in the <code>metadata</code> slot
is set to <code>"geographic"</code>, and a warning is issued if this was also the value
in <code>object</code>.  (4) The <code>declination</code> item in the <code>metadata</code> slot is set to
the value supplied to this function.
</p>


<h3>Value</h3>

<p>A <a href="#topic+cm-class">cm</a> object, adjusted as outlined in &ldquo;Details&rdquo;.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+magneticField">magneticField()</a></code> to determine the declination,
inclination and intensity at a given spot on the world, at a given time.
</p>
<p>Other things related to magnetism: 
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Coce-method">applyMagneticDeclination,oce-method</a></code>,
<code><a href="#topic+applyMagneticDeclination">applyMagneticDeclination</a>()</code>,
<code><a href="#topic+magneticField">magneticField</a>()</code>
</p>
<p>Other things related to cm data: 
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic+as.cm">as.cm</a>()</code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+read.cm">read.cm</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+summary+2Ccm-method">summary,cm-method</a></code>
</p>

<hr>
<h2 id='applyMagneticDeclination+2Coce-method'>Alter an Object to Account for Magnetic Declination</h2><span id='topic+applyMagneticDeclination+2Coce-method'></span>

<h3>Description</h3>

<p>Current-measuring instruments that infer flow direction using magnetic
compasses require a correction for magnetic declination, in order to infer
currents with x and y oriented eastward and northward, respectively.
<code><a href="#topic+applyMagneticDeclination">applyMagneticDeclination()</a></code> is a generic function that handles this task by
altering velocity components (and heading values, if they exist).  It works
for objects of the <a href="#topic+cm-class">cm</a>, <a href="#topic+adp-class">adp</a> and <a href="#topic+adv-class">adv</a> and <a href="#topic+cm-class">cm</a>
classes by calling <code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method()</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method()</a></code>, or
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method()</a></code>, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'oce'
applyMagneticDeclination(
  object = "oce",
  declination = 0,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="applyMagneticDeclination+2B2Coce-method_+3A_object">object</code></td>
<td>
<p>an object of <a href="#topic+cm-class">cm</a>, <a href="#topic+adp-class">adp</a>, or <a href="#topic+adv-class">adv</a> class.</p>
</td></tr>
<tr><td><code id="applyMagneticDeclination+2B2Coce-method_+3A_declination">declination</code></td>
<td>
<p>numeric value holding magnetic declination in degrees,
positive for clockwise from north.</p>
</td></tr>
<tr><td><code id="applyMagneticDeclination+2B2Coce-method_+3A_debug">debug</code></td>
<td>
<p>a debugging flag, set to a positive value to get debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned value is a copy of <code>object</code> that has been modified in 4 ways.
(1) the horizontal components of velocity are rotated clockwise by
<code>declination</code> degrees.  (2) If the object holds heading values, then
<code>declination</code> is added to them. (3) The <code>north</code> item in the <code>metadata</code> slot
is set to <code>"geographic"</code>, and a warning is issued if this was also the value
in <code>object</code>.  (4) The <code>declination</code> item in the <code>metadata</code> slot is set to
the value supplied to this function.
</p>


<h3>Value</h3>

<p>an object of the same class as <code>object</code>, modified as outlined in
&ldquo;Details&rdquo;.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, aided, for the <a href="#topic+adp-class">adp</a> and <a href="#topic+adv-class">adv</a> variants,
by Clark Richards and Jaimie Harbin.
</p>


<h3>See Also</h3>

<p>Use <code><a href="#topic+magneticField">magneticField()</a></code> to determine the declination,
inclination and intensity at a given spot on the world, at a given time.
</p>
<p>Other things related to magnetism: 
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination">applyMagneticDeclination</a>()</code>,
<code><a href="#topic+magneticField">magneticField</a>()</code>
</p>

<hr>
<h2 id='approx3d'>Trilinear Interpolation in a 3D Array</h2><span id='topic+approx3d'></span>

<h3>Description</h3>

<p>Interpolate within a 3D array, using the trilinear approximation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>approx3d(x, y, z, f, xout, yout, zout)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="approx3d_+3A_x">x</code></td>
<td>
<p>vector of x values for grid (must be equi-spaced)</p>
</td></tr>
<tr><td><code id="approx3d_+3A_y">y</code></td>
<td>
<p>vector of y values for grid (must be equi-spaced)</p>
</td></tr>
<tr><td><code id="approx3d_+3A_z">z</code></td>
<td>
<p>vector of z values for grid (must be equi-spaced)</p>
</td></tr>
<tr><td><code id="approx3d_+3A_f">f</code></td>
<td>
<p>matrix of rank 3, with the gridded values mapping to the <code>x</code>
values (first index of <code>f</code>), etc.</p>
</td></tr>
<tr><td><code id="approx3d_+3A_xout">xout</code></td>
<td>
<p>vector of x values for output.</p>
</td></tr>
<tr><td><code id="approx3d_+3A_yout">yout</code></td>
<td>
<p>vector of y values for output (length must match that of
<code>xout</code>).</p>
</td></tr>
<tr><td><code id="approx3d_+3A_zout">zout</code></td>
<td>
<p>vector of z values for output (length must match that of
<code>xout</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Trilinear interpolation is used to interpolate within the <code>f</code> array,
for those (<code>xout</code>, <code>yout</code> and <code>zout</code>) triplets that are
inside the region specified by <code>x</code>, <code>y</code> and <code>z</code>.  Triplets
that lie outside the range of <code>x</code>, <code>y</code> or <code>z</code> result in
<code>NA</code> values.
</p>


<h3>Value</h3>

<p>A vector of interpolated values (or <code>NA</code> values), with length
matching that of <code>xout</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>Examples</h3>

<pre><code class='language-R'># set up a grid
library(oce)
n &lt;- 5
x &lt;- seq(0, 1, length.out = n)
y &lt;- seq(0, 1, length.out = n)
z &lt;- seq(0, 1, length.out = n)
f &lt;- array(1:n^3, dim = c(length(x), length(y), length(z)))
# interpolate along a diagonal line
m &lt;- 100
xout &lt;- seq(0, 1, length.out = m)
yout &lt;- seq(0, 1, length.out = m)
zout &lt;- seq(0, 1, length.out = m)
approx &lt;- approx3d(x, y, z, f, xout, yout, zout)
# graph the results
plot(xout, approx, type = "l")
points(xout[1], f[1, 1, 1])
points(xout[m], f[n, n, n])
</code></pre>

<hr>
<h2 id='argo'>Sample argo Data</h2><span id='topic+argo'></span>

<h3>Description</h3>

<p>This holds data from ARGO 6900388 in the North Atlantic.
</p>


<h3>Details</h3>

<p>Below is the official citation (note that this DOI has web links for
downloads):
</p>
<p>Argo (2017). Argo float data and metadata from Global Data Assembly Centre
(Argo GDAC) - Snapshot of Argo GDAC of July, 8st 2017. SEANOE.
DOI:<code>10.17882/42182#50865</code>
</p>


<h3>Source</h3>

<p>The netcdf file used by <code><a href="#topic+read.argo">read.argo()</a></code> to create this <a href="#topic+argo-class">argo</a>
object was downloaded using FTP to
<code>ftp.ifremer.fr/ifremer/argo/dac/bodc/6900388/6900388_prof.nc</code>
on 2020 June 24.
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(argo)
summary(argo)
data(coastlineWorld)
plot(argo, which = "trajectory")

</code></pre>

<hr>
<h2 id='argo-class'>Class to Store Argo Profiler Data</h2><span id='topic+argo-class'></span>

<h3>Description</h3>

<p>This class stores data from Argo floats.
</p>


<h3>Details</h3>

<p>An <code>argo</code> object may be read with <code><a href="#topic+read.argo">read.argo()</a></code> or
created with <code><a href="#topic+as.argo">as.argo()</a></code>.  Argo data can be gridded to constant
pressures with <code><a href="#topic+argoGrid">argoGrid()</a></code> or subsetted with
<code><a href="#topic+subset+2Cargo-method">subset,argo-method()</a></code>.  Plots can be made with
<code><a href="#topic+plot+2Cargo-method">plot,argo-method()</a></code>, while <code><a href="#topic+summary+2Cargo-method">summary,argo-method()</a></code>
produces statistical summaries and <code>show</code> produces overviews.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>argo</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
The key items stored in this slot include  equal-length vectors <code>time</code>, <code>longitude</code>, <code>latitude</code> and equal-dimension matrices <code>pressure</code>, <code>salinity</code>, and <code>temperature</code>.</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>argo</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
Examples that are of common interest include <code>id</code>, a vector of ID codes for the profiles, and <code>dataMode</code>, a vector of strings indicating whether the profile is in archived mode (<code>"A"</code>), realtime mode (<code>"R"</code>), or delayed mode (<code>"D"</code>).</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>argo</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+argo-class">argo</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+argo-class">argo</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>

<hr>
<h2 id='argoGrid'>Grid Argo Float Data</h2><span id='topic+argoGrid'></span>

<h3>Description</h3>

<p>Grid an Argo float, by interpolating to fixed pressure levels.
The gridding is done with <code><a href="stats.html#topic+approx">approx()</a></code>.  If there is
sufficient user demand, other methods may be added, by analogy to
<code><a href="#topic+sectionGrid">sectionGrid()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>argoGrid(argo, p, debug = getOption("oceDebug"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="argoGrid_+3A_argo">argo</code></td>
<td>
<p>A <code>argo</code> object to be gridded.</p>
</td></tr>
<tr><td><code id="argoGrid_+3A_p">p</code></td>
<td>
<p>Optional indication of the pressure levels to which interpolation
should be done.  If this is not supplied, the pressure levels will be
calculated based on the existing values, using medians. If <code>p="levitus"</code>,
then pressures will be set to be those of the Levitus atlas, given by
<code><a href="#topic+standardDepths">standardDepths()</a></code>, trimmed to the maximum pressure in <code>argo</code>.
If <code>p</code> is a single numerical value, it is taken as the number of
subdivisions to use in a call to <code><a href="base.html#topic+seq">seq()</a></code> that has range from 0 to the
maximum pressure in <code>argo</code>.  Finally, if a vector numerical values is
provided, then it is used as is.</p>
</td></tr>
<tr><td><code id="argoGrid_+3A_debug">debug</code></td>
<td>
<p>A flag that turns on debugging.  Higher values provide deeper
debugging.</p>
</td></tr>
<tr><td><code id="argoGrid_+3A_...">...</code></td>
<td>
<p>Optional arguments to <code><a href="stats.html#topic+approx">approx()</a></code>, which is used to do the
gridding.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>x an <a href="#topic+argo-class">argo</a> object.
</p>


<h3>A note about flags</h3>

<p>Data-quality flags contained within the original object are ignored by this
function, and the returned value contains no such flags.  This is because such
flags represent an assessment of the original data, not of quantities derived
from those data.  This function produces a warning to this effect. The
recommended practice is to use <code><a href="#topic+handleFlags">handleFlags()</a></code> or some other means to
deal with flags before calling the present function.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>See Also</h3>

<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(argo)
g &lt;- argoGrid(argo, p = seq(0, 100, 1))
par(mfrow = c(2, 1))
t &lt;- g[["time"]]
z &lt;- -g[["pressure"]][, 1]
# Set zlim because of spurious temperatures.
imagep(t, z, t(g[["temperature"]]), ylim = c(-100, 0), zlim = c(0, 20))
imagep(t, z, t(g[["salinity"]]), ylim = c(-100, 0))

</code></pre>

<hr>
<h2 id='argoJuldToTime'>Convert Argo Julian Day to R Time</h2><span id='topic+argoJuldToTime'></span>

<h3>Description</h3>

<p>Convert Argo Julian Day to R Time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>argoJuldToTime(jday)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="argoJuldToTime_+3A_jday">jday</code></td>
<td>
<p>A numerical value indicating the julian day in the Argo convention,
with day=0 at 1950-01-01.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jaimie Harbin and Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>argoJuldToTime(25749)

</code></pre>

<hr>
<h2 id='argoNames2oceNames'>Convert Argo Data Name to Oce Name</h2><span id='topic+argoNames2oceNames'></span>

<h3>Description</h3>

<p>This function is used internally by <code><a href="#topic+read.argo">read.argo()</a></code> to convert Argo-convention
data names to oce-convention names. Users should not call this directly, since
its return value may be changed at any moment (e.g. to include units as well
as names).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>argoNames2oceNames(names, ignore.case = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="argoNames2oceNames_+3A_names">names</code></td>
<td>
<p>vector of character strings containing names in the Argo convention.</p>
</td></tr>
<tr><td><code id="argoNames2oceNames_+3A_ignore.case">ignore.case</code></td>
<td>
<p>a logical value passed to <code><a href="base.html#topic+gsub">gsub()</a></code>, indicating whether to
ignore the case of input strings. The default is set to <code>TRUE</code> because some data
files use lower-case names, despite the fact that the Argo documentation specifies
upper-case.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Initially, Feb 2016, the inference of names was initially done
by an inspection of some data files, based on reference 1. Later, in June
2023, broader inspection of more files and documents yielded about ten
additions, and a single correction: <code>VRSpH</code> was renamed
<code>phSensorVoltageDifference</code>, to match related names that had been added.
</p>
<p>It should be noted that the data files examined contain some names that are not
documented in reference 1, and others that are listed only in its changelog,
with no actual definitions being given. For example, the files had six distinct
variable names that seem to relate to phase in the oxygen sensor, but
these are not translated by the present function because these
variable names are not defined in reference 1, or not defined uniquely
in reference 2.
</p>
<p>The names are converted with
<code><a href="base.html#topic+gsub">gsub()</a></code>, using the <code>ignore.case</code> argument of the present
function.
The procedure
is to first handle the items listed in the following table, with string
searches anchored to the start of the string. After that,
the qualifiers
<code style="white-space: pre;">&#8288;_ADJUSTED&#8288;</code>, <code style="white-space: pre;">&#8288;_ERROR&#8288;</code> and <code style="white-space: pre;">&#8288;_QC&#8288;</code>,
are translated to <code>Adjusted</code>, <code>Error</code>, and <code>QC</code>, respectively.
</p>
<p>An incomplete list of name translations is as follows, where <code>~</code>
represents digit sequences in some instances and letters
in others.  Note that until June 2023, <code>pHSensorVoltageDifference</code>
was called <code>VRSpH</code>.
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Argo name</strong> </td><td style="text-align: left;"> <strong>oce name</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>BBP</code> </td><td style="text-align: left;"> <code>bbp</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>BETA_BACKSCATTERING</code> </td><td style="text-align: left;"> <code>betaBackscattering</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>BPHASE_OXY</code> </td><td style="text-align: left;"> <code>bphaseOxygen</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>C~PHASE_DOXY</code> </td><td style="text-align: left;"> <code>C~phaseOxygen</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CDOM</code> </td><td style="text-align: left;"> <code>CDOM</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CNDC</code> </td><td style="text-align: left;"> <code>conductivity</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CHLA</code> </td><td style="text-align: left;"> <code>chlorophyllA</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CP</code> </td><td style="text-align: left;"> <code>beamAttenuation</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CYCLE_NUMBER</code> </td><td style="text-align: left;"> <code>cycleNumber</code> (both this and <code>cycle</code> are handled by the [[ operator)</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DATA_CENTRE</code> </td><td style="text-align: left;"> <code>dataCentre</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DATA_MODE</code> </td><td style="text-align: left;"> <code>dataMode</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DATA_STATE_INDICATOR</code> </td><td style="text-align: left;"> <code>dataStateIndicator</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DC_REFERENCE</code> </td><td style="text-align: left;"> <code>DCReference</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DIRECTION</code> </td><td style="text-align: left;"> <code>direction</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DOWN_IRRADIANCE</code> </td><td style="text-align: left;"> <code>downwellingIrradiance</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DOWNWELLING_PAR</code> </td><td style="text-align: left;"> <code>downwellingPAR</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>FIRMWARE_VERSION</code> </td><td style="text-align: left;"> <code>firmwareVersion</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>FIT_ERROR_NITRATE</code> </td><td style="text-align: left;"> <code>fitErrorNitrate</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>FLUORESCENCE_CDOM</code> </td><td style="text-align: left;"> <code>fluorescenceCDOM</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>FLUORESCENCE_CHLA</code> </td><td style="text-align: left;"> <code>fluorescenceChlorophyllA</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>IB_PH</code> </td><td style="text-align: left;"> <code>pHBaseCurrent</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>IK_PH</code> </td><td style="text-align: left;"> <code>pHCounterCurrent</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>INST_REFERENCE</code> </td><td style="text-align: left;"> <code>instReference</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>JULD</code> </td><td style="text-align: left;"> <code>juld</code> (and used to compute <code>time</code>)</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>JULD_QC_LOCATION</code> </td><td style="text-align: left;"> <code>juldQCLocation</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>LATITUDE</code> </td><td style="text-align: left;"> <code>latitude</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>LONGITUDE</code> </td><td style="text-align: left;"> <code>longitude</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>MOLAR_DOXY</code> </td><td style="text-align: left;"> <code>oxygenUncompensated</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>MTIME</code> </td><td style="text-align: left;"> <code>mtime</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>NB_SAMPLE_CTD</code> </td><td style="text-align: left;"> <code>nbSampleCtd</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PH_IN_SITU_FREE</code> </td><td style="text-align: left;"> <code>pHFree</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PH_IN_SITU_TOTAL</code> </td><td style="text-align: left;"> <code>pH</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PI_NAME</code> </td><td style="text-align: left;"> <code>PIName</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PLATFORM_NUMBER</code> </td><td style="text-align: left;"> <code>id</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>POSITION_ACCURACY</code> </td><td style="text-align: left;"> <code>positionAccuracy</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>POSITIONING_SYSTEM</code> </td><td style="text-align: left;"> <code>positioningSystem</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PROFILE</code> </td><td style="text-align: left;"> <code>profile</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PROJECT_NAME</code> </td><td style="text-align: left;"> <code>projectName</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>RAW_DOWNWELLING_IRRADIANCE</code> </td><td style="text-align: left;"> <code>rawDownwellingIrradiance</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>RAW_DOWNWELLING_PAR</code> </td><td style="text-align: left;"> <code>rawDownwellingPAR</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>RAW_UPWELLING_RADIANCE</code> </td><td style="text-align: left;"> <code>rawUpwellingRadiance</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>STATION_PARAMETERS</code> </td><td style="text-align: left;"> <code>stationParameters</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TEMP</code> </td><td style="text-align: left;"> <code>temperature</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TEMP_CPU_CHLA</code> </td><td style="text-align: left;"> <code>temperatureCPUChlorophyllA</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TEMP_DOXY</code> </td><td style="text-align: left;"> <code>temperatureOxygen</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TEMP_NITRATE</code> </td><td style="text-align: left;"> <code>temperatureNitrate</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TEMP_PH</code> </td><td style="text-align: left;"> <code>temperaturePH</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TEMP_SPECTROPHOTOMETER_NITRATE</code> </td><td style="text-align: left;"> <code>temperatureSpectrophotometerNitrate</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TILT</code> </td><td style="text-align: left;"> <code>tilt</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TPHASE_DOXY</code> </td><td style="text-align: left;"> <code>tphaseOxygen</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TURBIDITY</code> </td><td style="text-align: left;"> <code>turbidity</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>UP_RADIANCE</code> </td><td style="text-align: left;"> <code>upwellingRadiance</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>UV_INTENSITY</code> </td><td style="text-align: left;"> <code>UVIntensity</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>UV_INTENSITY_DARK_NITRATE</code> </td><td style="text-align: left;"> <code>UVIntensityDarkNitrate</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>UV_INTENSITY_NITRATE</code> </td><td style="text-align: left;"> <code>UVIntensityNitrate</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>VRS_PH</code> </td><td style="text-align: left;"> <code>pHSensorVoltageDifference</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>WMO_INST_TYPE</code> </td><td style="text-align: left;"> <code>WMOInstType</code></td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>A character vector of the same length as <code>names</code>, but with
replacements having been made for all known quantities.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, with help from Anna Victor
</p>


<h3>References</h3>


<ol>
<li><p> Argo User's Manual Version 3.3, Nov 89th, 2019, available at
<code style="white-space: pre;">&#8288;https://archimer.ifremer.fr/doc/00187/29825/&#8288;</code> online.
</p>
</li>
<li><p> Argo list of parameters in an excel spreadsheet, available at
<code style="white-space: pre;">&#8288;http://www.argodatamgt.org/content/download/27444/187206/file/argo-parameters-list-core-and-b.xlsx&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>

<hr>
<h2 id='argShow'>Show a Function Argument</h2><span id='topic+argShow'></span>

<h3>Description</h3>

<p>Show a Function Argument
</p>


<h3>Usage</h3>

<pre><code class='language-R'>argShow(x, nshow = 4, last = FALSE, sep = "=")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="argShow_+3A_x">x</code></td>
<td>
<p>the argument</p>
</td></tr>
<tr><td><code id="argShow_+3A_nshow">nshow</code></td>
<td>
<p>number of values to show at first (if length(x)&gt; 1)</p>
</td></tr>
<tr><td><code id="argShow_+3A_last">last</code></td>
<td>
<p>indicates whether this is the final argument to the function</p>
</td></tr>
<tr><td><code id="argShow_+3A_sep">sep</code></td>
<td>
<p>the separator between name and value</p>
</td></tr>
</table>

<hr>
<h2 id='as.adp'>Create an adp Object</h2><span id='topic+as.adp'></span>

<h3>Description</h3>

<p>Create an adp Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.adp(
  time,
  distance,
  v,
  a = NULL,
  q = NULL,
  orientation = "upward",
  coordinate = "enu"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.adp_+3A_time">time</code></td>
<td>
<p>of observations in POSIXct format</p>
</td></tr>
<tr><td><code id="as.adp_+3A_distance">distance</code></td>
<td>
<p>to centre of bins</p>
</td></tr>
<tr><td><code id="as.adp_+3A_v">v</code></td>
<td>
<p>array of velocities, with first index for time, second for bin number, and third for beam number</p>
</td></tr>
<tr><td><code id="as.adp_+3A_a">a</code></td>
<td>
<p>amplitude, a <code><a href="base.html#topic+raw">raw()</a></code> array with dimensions matching <code>u</code></p>
</td></tr>
<tr><td><code id="as.adp_+3A_q">q</code></td>
<td>
<p>quality, a <code><a href="base.html#topic+raw">raw()</a></code> array with dimensions matching <code>u</code></p>
</td></tr>
<tr><td><code id="as.adp_+3A_orientation">orientation</code></td>
<td>
<p>a string indicating sensor orientation, e.g. <code>"upward"</code> and <code>"downward"</code></p>
</td></tr>
<tr><td><code id="as.adp_+3A_coordinate">coordinate</code></td>
<td>
<p>a string indicating the coordinate system, <code>"enu"</code>, <code>"beam"</code>, <code>"xy"</code>, or <code>"other"</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Construct an <a href="#topic+adp-class">adp</a> object.  Only a basic
subset of the typical <code>data</code> slot is represented in the arguments
to this function, on the assumption that typical usage in reading data
is to set up a nearly-blank <a href="#topic+adp-class">adp</a> object, the <code>data</code>
slot of which is then inserted.  However, in some testing situations it
can be useful to set up artificial <code>adp</code> objects, so the other
arguments may be useful.
</p>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(adp)
t &lt;- adp[["time"]]
d &lt;- adp[["distance"]]
v &lt;- adp[["v"]]
a &lt;- as.adp(time = t, distance = d, v = v)

plot(a)


</code></pre>

<hr>
<h2 id='as.argo'>Coerce Data Into an argo Object</h2><span id='topic+as.argo'></span>

<h3>Description</h3>

<p>Coerce a dataset into an argo dataset. This is not the right way to
read official argo datasets, which are provided in NetCDF format and may
be read with <code><a href="#topic+read.argo">read.argo()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.argo(
  time,
  longitude,
  latitude,
  salinity,
  temperature,
  pressure,
  units = NULL,
  id,
  filename = "",
  missingValue
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.argo_+3A_time">time</code></td>
<td>
<p>a vector of POSIXct times.</p>
</td></tr>
<tr><td><code id="as.argo_+3A_longitude">longitude</code></td>
<td>
<p>a vector of longitudes.</p>
</td></tr>
<tr><td><code id="as.argo_+3A_latitude">latitude</code></td>
<td>
<p>a vector of latitudes.</p>
</td></tr>
<tr><td><code id="as.argo_+3A_salinity">salinity</code></td>
<td>
<p>a vector of salinities.</p>
</td></tr>
<tr><td><code id="as.argo_+3A_temperature">temperature</code></td>
<td>
<p>a vector of temperatures.</p>
</td></tr>
<tr><td><code id="as.argo_+3A_pressure">pressure</code></td>
<td>
<p>a vector of pressures.</p>
</td></tr>
<tr><td><code id="as.argo_+3A_units">units</code></td>
<td>
<p>an optional list containing units. If <code>NULL</code>, the default,
then <code>"degree east"</code> is used for <code>longitude</code>,
<code>"degree north"</code> for <code>latitude</code>,
<code>""</code> for <code>salinity</code>,
<code>"ITS-90"</code> for <code>temperature</code>, and
<code>"dbar"</code> for <code>pressure</code>.</p>
</td></tr>
<tr><td><code id="as.argo_+3A_id">id</code></td>
<td>
<p>an identifier for the argo float, typically a number, but stored within
the object in a character form. (For example, the dataset retrieved with <code>data(argo)</code>
has an <code>id</code> of <code>"6900388"</code>.)</p>
</td></tr>
<tr><td><code id="as.argo_+3A_filename">filename</code></td>
<td>
<p>a source filename, which defaults to an empty string.</p>
</td></tr>
<tr><td><code id="as.argo_+3A_missingvalue">missingValue</code></td>
<td>
<p>an optional missing value, indicating data values that should be
taken as <code>NA</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+argo-class">argo</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for the <a href="#topic+argo-class">argo</a> class explains the structure of argo
objects, and also outlines the other functions dealing with them.
</p>
<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>

<hr>
<h2 id='as.cm'>Coerce Data Into a cm Object</h2><span id='topic+as.cm'></span>

<h3>Description</h3>

<p>Coerce Data Into a cm Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.cm(
  time,
  u = NULL,
  v = NULL,
  pressure = NULL,
  conductivity = NULL,
  temperature = NULL,
  salinity = NULL,
  longitude = NA,
  latitude = NA,
  filename = "",
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.cm_+3A_time">time</code></td>
<td>
<p>A vector of times of observation, or an <code>oce</code> object from which time
and two velocity components can be inferred, e.g. an <a href="#topic+adv-class">adv</a> object, or
an <a href="#topic+adp-class">adp</a> object that has only one distance bin.  If <code>time</code> is an <code>oce</code> object,
then all of the following arguments are ignored.</p>
</td></tr>
<tr><td><code id="as.cm_+3A_u">u</code>, <code id="as.cm_+3A_v">v</code></td>
<td>
<p>optional numerical vectors containing the x and y components of velocity (m/s).</p>
</td></tr>
<tr><td><code id="as.cm_+3A_pressure">pressure</code>, <code id="as.cm_+3A_conductivity">conductivity</code>, <code id="as.cm_+3A_salinity">salinity</code>, <code id="as.cm_+3A_temperature">temperature</code></td>
<td>
<p>optional numerical vectors
containing pressure (dbar), electrical conductivity, practical salinity,
and in-situ temperature (degree C).</p>
</td></tr>
<tr><td><code id="as.cm_+3A_longitude">longitude</code>, <code id="as.cm_+3A_latitude">latitude</code></td>
<td>
<p>optional position specified in degrees East and North.</p>
</td></tr>
<tr><td><code id="as.cm_+3A_filename">filename</code></td>
<td>
<p>optional source file name.</p>
</td></tr>
<tr><td><code id="as.cm_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other things related to cm data: 
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+read.cm">read.cm</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+summary+2Ccm-method">summary,cm-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Example 1: creation from scratch
t &lt;- Sys.time() + 0:50
u &lt;- sin(2 * pi * 0:50 / 5) + rnorm(51)
v &lt;- cos(2 * pi * 0:50 / 5) + rnorm(51)
p &lt;- 100 + rnorm(51)
summary(as.cm(t, u, v, p))

# Example 2: creation from an adv object
data(adv)
summary(as.cm(adv))

</code></pre>

<hr>
<h2 id='as.coastline'>Coerce Data Into a coastline Object</h2><span id='topic+as.coastline'></span>

<h3>Description</h3>

<p>Coerces a sequence of longitudes and latitudes into a coastline dataset.
This may be used when <code><a href="#topic+read.coastline">read.coastline()</a></code> cannot read a file, or
when the data have been manipulated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.coastline(longitude, latitude, fillable = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.coastline_+3A_longitude">longitude</code></td>
<td>
<p>the longitude in decimal degrees, positive east of
Greenwich, or a data frame with columns named <code>latitude</code> and
<code>longitude</code>, in which case these values are extracted from the data
frame and the second argument is ignored.</p>
</td></tr>
<tr><td><code id="as.coastline_+3A_latitude">latitude</code></td>
<td>
<p>the latitude in decimal degrees, positive north of the
Equator.</p>
</td></tr>
<tr><td><code id="as.coastline_+3A_fillable">fillable</code></td>
<td>
<p>boolean indicating whether the coastline can be drawn as a
filled polygon.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="#topic+coastline-class">coastline</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>

<hr>
<h2 id='as.ctd'>Coerce Data Into a ctd Object</h2><span id='topic+as.ctd'></span>

<h3>Description</h3>

<p>Assemble data into a <a href="#topic+ctd-class">ctd</a> object.  This function is complicated
(spanning approximately 500 lines of code) because it tries to handle many
special cases, and tries to make sensible defaults for unspecified
parameters.  If odd results are found, users might find it helpful to call
this function with the first argument being a simple vector of Practical
Salinity values, in which case the processing of the other arguments is
relatively straightforward.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.ctd(
  salinity,
  temperature = NULL,
  pressure = NULL,
  conductivity = NULL,
  scan = NULL,
  time = NULL,
  units = NULL,
  flags = NULL,
  missingValue = NULL,
  type = "",
  serialNumber = NULL,
  ship = NULL,
  cruise = NULL,
  station = NULL,
  startTime = NULL,
  longitude = NULL,
  latitude = NULL,
  deploymentType = "unknown",
  pressureAtmospheric = 0,
  sampleInterval = NULL,
  profile = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.ctd_+3A_salinity">salinity</code></td>
<td>
<p>may be (1) a numeric vector holding Practical Salinity,
(2) a list or data frame holding <code>salinity</code> and other
hydrographic variables or (3) an <code>oce-class</code> object that holds
hydrographic information. If <code>salinity</code> is not provided,
then <code>conductivity</code> must be provided, so that <code><a href="#topic+swSCTp">swSCTp()</a></code>
can be used to compute salinity.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_temperature">temperature</code></td>
<td>
<p>a numeric vector containing <em>in-situ</em> temperature in
<code class="reqn">^\circ</code>C on the ITS-90 scale; see &ldquo;Temperature units&rdquo; in the
documentation for <code><a href="#topic+swRho">swRho()</a></code>.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_pressure">pressure</code></td>
<td>
<p>a numeric vector containing sea pressure values, in decibars.
Typically, this vector has the same length as <code>salinity</code> and <code>temperature</code>,
but it also possible to supply just one value, which will be repeated
to get the right length. Note that <code>as.ctd()</code> stores the
sum of <code>pressure</code> and <code>pressureAtmospheric</code> in the returned object,
although the default value for <code>pressureAtmospheric</code> is zero, so
in the default case, <code>pressure</code> is stored directly.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_conductivity">conductivity</code></td>
<td>
<p>an optional numeric vector containing electrical
conductivity ratio through the water column. To convert from raw conductivity
in milliSeimens per centimeter divide by 42.914 to get conductivity ratio
(see Culkin and Smith, 1980).</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_scan">scan</code></td>
<td>
<p>optional numeric vector holding scan number.  If not provided,
this is set to <a href="base.html#topic+seq_along">seq_along</a><code>(salinity)</code>.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_time">time</code></td>
<td>
<p>optional vector of times of observation.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_units">units</code></td>
<td>
<p>an optional list containing units.  If not supplied,
defaults are set for <code>pressure</code>, <code>temperature</code>, <code>salinity</code>,
and <code>conductivity</code>. Since these are simply guesses, users
are advised strongly to supply <code>units</code>. See &ldquo;Examples&rdquo;.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_flags">flags</code></td>
<td>
<p>if supplied, this is a <a href="base.html#topic+list">list</a> containing data-quality
flags. The elements of this list must have names that match the data
provided to the object.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_missingvalue">missingValue</code></td>
<td>
<p>optional missing value, indicating data that should be
taken as <code>NA</code>. Set to <code>NULL</code> to turn off this feature.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_type">type</code></td>
<td>
<p>optional type of CTD, e.g. &quot;SBE&quot;</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_serialnumber">serialNumber</code></td>
<td>
<p>optional serial number of instrument</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_ship">ship</code></td>
<td>
<p>optional string containing the ship from which the observations were made.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_cruise">cruise</code></td>
<td>
<p>optional string containing a cruise identifier.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_station">station</code></td>
<td>
<p>optional string containing a station identifier.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_starttime">startTime</code></td>
<td>
<p>optional indication of the start time for the profile,
which is used in some several plotting functions.  This is best given as a
<a href="base.html#topic+POSIXt">POSIXt</a> time, but it may also be a character string
that can be converted to a time with <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>,
using <code>UTC</code> as the timezone.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_longitude">longitude</code></td>
<td>
<p>optional numerical value containing longitude in decimal
degrees, positive in the eastern hemisphere. If this is a single number,
then it is stored in the <code>metadata</code> slot of the returned value; if it
is a vector of numbers, then they are stored in the <code>data</code> slot.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_latitude">latitude</code></td>
<td>
<p>optional numerical value containing the latitude in decimal
degrees, positive in the northern hemisphere. See the note on length, for
the <code>longitude</code> argument.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_deploymenttype">deploymentType</code></td>
<td>
<p>character string indicating the type of deployment. Use
<code>"unknown"</code> if this is not known, <code>"profile"</code> for a profile (in
which the data were acquired during a downcast, while the device was lowered
into the water column, perhaps also including an upcast; <code>"moored"</code> if
the device is installed on a fixed mooring, <code>"thermosalinograph"</code> (or
<code>"tsg"</code>) if the device is mounted on a moving vessel, to record
near-surface properties, or <code>"towyo"</code> if the device is repeatedly
lowered and raised.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_pressureatmospheric">pressureAtmospheric</code></td>
<td>
<p>A numerical value (a constant or a vector),
that is subtracted from pressure before storing it in the return value.
(This altered pressure is also used in calculating <code>salinity</code>, if
that is to be computed from <code>conductivity</code>, etc., using
<code><a href="#topic+swSCTp">swSCTp()</a></code>; see <code>salinity</code> above.)</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_sampleinterval">sampleInterval</code></td>
<td>
<p>optional numerical value indicating the time between
samples in the profile.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_profile">profile</code></td>
<td>
<p>optional positive integer specifying the number of the profile
to extract from an object that has data in matrices, such as for some
<code>argo</code> objects. Currently the <code>profile</code> argument is only utilized for
<a href="#topic+argo-class">argo</a> objects.</p>
</td></tr>
<tr><td><code id="as.ctd_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following sections provide an some notes on how <code>as.ctd()</code> handles
certain object types given as the first parameter.
</p>
<p><strong>Converting argo objects</strong>
</p>
<p>If the <code>salinity</code> argument is an object of <a href="#topic+argo-class">argo</a>, then that
object is dismantled and reassembled as a <a href="#topic+ctd-class">ctd</a> object in ways that
are mostly straightforward, although the handling of time depends
on the information in the original netcdf data file that was used
by <code><a href="#topic+read.argo">read.argo()</a></code> to create the <a href="#topic+argo-class">argo</a> object.
</p>
<p>All Argo data files contain an item called <code>juld</code> from which the profile
time can be computed, and some also contain an additional item named <code>MTIME</code>,
from which the times of individual measurements can also be computed.  Both
cases are handled by <code>as.ctd()</code>, using a scheme outlined in
Note 4 of the Details section of the <code><a href="#topic+read.argo">read.argo()</a></code> documentation.
</p>
<p><strong>Converting rsk objects</strong>
</p>
<p>If the <code>salinity</code> argument is an object of <a href="#topic+rsk-class">rsk</a>,
then <code>as.ctd</code> passes it,
<code>pressureAtmospheric</code>,
<code>longitude</code>,
<code>latitude</code>
<code>ship</code>,
<code>cruise</code>,
<code>station</code> and
<code>deploymentType</code>
to <code><a href="#topic+rsk2ctd">rsk2ctd()</a></code>, which builds the ctd object that is
returned by <code>as.ctd</code>. The other arguments to <code>as.ctd</code>
are ignored in this instance, because <code>rsk</code> objects already
contain their information. If required, any data or metadata
element can be added to the value returned by <code>as.ctd</code>
using <code><a href="#topic+oceSetData">oceSetData()</a></code> or <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
respectively.
</p>
<p>The returned <a href="#topic+rsk-class">rsk</a> object contains pressure in a form that
may need to be adjusted, because <code>rsk</code> objects may contain
either absolute pressure or sea pressure. This adjustment is handled
automatically by <code>as.ctd</code>, by examination of the metadata item
named <code>pressureType</code> (described in the documentation for
<code><a href="#topic+read.rsk">read.rsk()</a></code>).  Once the sea pressure is determined,
adjustments may be made with the <code>pressureAtmospheric</code> argument,
although in that case it is better considered a pressure adjustment
than the atmospheric pressure.
</p>
<p><a href="#topic+rsk-class">rsk</a> objects may store sea pressure or absolute pressure (the
sum of sea pressure and atmospheric pressure), depending on how the object was
created with <code><a href="#topic+as.rsk">as.rsk()</a></code> or <code><a href="#topic+read.rsk">read.rsk()</a></code>.  However,
<a href="#topic+ctd-class">ctd</a> objects store sea pressure, which is needed for
plotting, calculating density, etc. This poses no difficulties, however,
because <code>as.ctd</code> automatically converts absolute pressure to sea pressure,
if the metadata in the <a href="#topic+rsk-class">rsk</a> object indicates that this is
appropriate. Further alteration of the pressure can be accomplished with the
<code>pressureAtmospheric</code> argument, as noted above.
</p>


<h3>Value</h3>

<p>A <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Culkin, F., and Norman D. Smith, 1980. Determination of the
concentration of potassium chloride solution having the same electrical
conductivity, at 15 C and infinite frequency, as standard seawater of salinity
35.0000 ppt (Chlorinity 19.37394 ppt). <em>IEEE Journal of Oceanic
Engineering</em>, volume <strong>5</strong>, pages 22-23.
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# 1. fake data, with default units
pressure &lt;- 1:50
temperature &lt;- 10 - tanh((pressure - 20) / 5) + 0.02 * rnorm(50)
salinity &lt;- 34 + 0.5 * tanh((pressure - 20) / 5) + 0.01 * rnorm(50)
ctd &lt;- as.ctd(salinity, temperature, pressure)
# Add a new column
fluo &lt;- 5 * exp(-pressure / 20)
ctd &lt;- oceSetData(ctd,
    name = "fluorescence", value = fluo,
    unit = list(unit = expression(mg / m^3), scale = "")
)
summary(ctd)

# 2. fake data, with supplied units (which are the defaults, actually)
ctd &lt;- as.ctd(salinity, temperature, pressure,
    units = list(
        salinity = list(unit = expression(), scale = "PSS-78"),
        temperature = list(unit = expression(degree * C), scale = "ITS-90"),
        pressure = list(unit = expression(dbar), scale = "")
    )
)

</code></pre>

<hr>
<h2 id='as.echosounder'>Coerce Data Into an echosounder Object</h2><span id='topic+as.echosounder'></span>

<h3>Description</h3>

<p>Coerces a dataset into a echosounder dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.echosounder(
  time,
  depth,
  a,
  src = "",
  sourceLevel = 220,
  receiverSensitivity = -55.4,
  transmitPower = 0,
  pulseDuration = 400,
  beamwidthX = 6.5,
  beamwidthY = 6.5,
  frequency = 41800,
  correction = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.echosounder_+3A_time">time</code></td>
<td>
<p>times of pings.</p>
</td></tr>
<tr><td><code id="as.echosounder_+3A_depth">depth</code></td>
<td>
<p>depths of samples within pings.</p>
</td></tr>
<tr><td><code id="as.echosounder_+3A_a">a</code></td>
<td>
<p>matrix of echo amplitudes, as would be stored with
<code><a href="#topic+read.echosounder">read.echosounder()</a></code>.</p>
</td></tr>
<tr><td><code id="as.echosounder_+3A_src">src</code></td>
<td>
<p>optional string indicating data source.</p>
</td></tr>
<tr><td><code id="as.echosounder_+3A_sourcelevel">sourceLevel</code></td>
<td>
<p>source level, in dB (uPa at 1m), denoted <code>sl</code> in
reference 1 p15, where it is in units 0.1dB (uPa at 1m).</p>
</td></tr>
<tr><td><code id="as.echosounder_+3A_receiversensitivity">receiverSensitivity</code></td>
<td>
<p>receiver sensitivity of the main element, in
dB(counts/uPa), denoted <code>rs</code> in reference 1 p15, where it is in units of
0.1dB(counts/uPa)</p>
</td></tr>
<tr><td><code id="as.echosounder_+3A_transmitpower">transmitPower</code></td>
<td>
<p>transmit power reduction factor, in dB, denoted
<code>tpow</code> in reference 1 p10, where it is in units 0.1 dB.</p>
</td></tr>
<tr><td><code id="as.echosounder_+3A_pulseduration">pulseDuration</code></td>
<td>
<p>duration of transmitted pulse in us</p>
</td></tr>
<tr><td><code id="as.echosounder_+3A_beamwidthx">beamwidthX</code></td>
<td>
<p>x-axis -3dB one-way beamwidth in deg, denoted <code>bwx</code>
in reference 1 p16, where the unit is 0.2 deg</p>
</td></tr>
<tr><td><code id="as.echosounder_+3A_beamwidthy">beamwidthY</code></td>
<td>
<p>y-axis -3dB one-way beamwidth in deg, denoted <code>bwx</code>
in reference 1 p16, where the unit is 0.2 deg</p>
</td></tr>
<tr><td><code id="as.echosounder_+3A_frequency">frequency</code></td>
<td>
<p>transducer frequency in Hz, denoted <code>fq</code> in reference 1 p16</p>
</td></tr>
<tr><td><code id="as.echosounder_+3A_correction">correction</code></td>
<td>
<p>user-defined calibration correction in dB, denoted
<code>corr</code> in reference 1 p14, where the unit is 0.01dB.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates an echosounder file.  The defaults for e.g.  <code>transmitPower</code>
are taken from the <code>echosounder</code> dataset, and they are unlikely to make
sense generally.  The first three parameters must be supplied, and the
dimension of <code>a</code> must align with the lengths of <code>time</code> and <code>depths</code>. The
other parameters have defaults that are unlikely to be correct for
arbitrary application, but they are not essential for basic plots,
etc.
</p>
<p>Those who use the <a href="https://CRAN.R-project.org/package=readHAC"><span class="pkg">readHAC</span></a> package to read echosounder
data should note that it stores the <code>a</code> matrix in a flipped and
transposed format. See that package's demo code for a function
named <code>flip()</code> that transforms the matrix as required by
<code><a href="#topic+as.echosounder">as.echosounder()</a></code>. Indeed, users working with HAC
data ought to study the whole of the <a href="https://CRAN.R-project.org/package=readHAC"><span class="pkg">readHAC</span></a>
documentation, to learn what data are stored, so that
<code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code> and <code><a href="#topic+oceSetData">oceSetData()</a></code> can be used as needed
to flesh out the contents returned by <code><a href="#topic+as.echosounder">as.echosounder()</a></code>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+echosounder-class">echosounder</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to echosounder data: 
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic+echosounder-class">echosounder-class</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+findBottom">findBottom</a>()</code>,
<code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method</a></code>,
<code><a href="#topic+read.echosounder">read.echosounder</a>()</code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+summary+2Cechosounder-method">summary,echosounder-method</a></code>
</p>

<hr>
<h2 id='as.gps'>Coerce Data Into a gps Object</h2><span id='topic+as.gps'></span>

<h3>Description</h3>

<p>Coerces a sequence of longitudes and latitudes into a GPS dataset.
This may be used when <code><a href="#topic+read.gps">read.gps()</a></code> cannot read a file, or when the
data have been manipulated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.gps(longitude, latitude, filename = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.gps_+3A_longitude">longitude</code></td>
<td>
<p>the longitude in decimal degrees, positive east of
Greenwich, or a data frame with columns named <code>latitude</code> and
<code>longitude</code>, in which case these values are extracted from the data
frame and the second argument is ignored.</p>
</td></tr>
<tr><td><code id="as.gps_+3A_latitude">latitude</code></td>
<td>
<p>the latitude in decimal degrees, positive north of the
Equator.</p>
</td></tr>
<tr><td><code id="as.gps_+3A_filename">filename</code></td>
<td>
<p>name of file containing data (if applicable).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+gps-class">gps</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to gps data: 
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic+gps-class">gps-class</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+read.gps">read.gps</a>()</code>,
<code><a href="#topic+summary+2Cgps-method">summary,gps-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Location of the Tower Tank at Dalhousie University
towerTank &lt;- as.gps(-63.59428, 44.63572)

</code></pre>

<hr>
<h2 id='as.ladp'>Coerce Data Into an ladp object</h2><span id='topic+as.ladp'></span>

<h3>Description</h3>

<p>This function assembles vectors of pressure and velocity, possibly also with
shears, salinity, temperature, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.ladp(
  longitude,
  latitude,
  station,
  time,
  pressure,
  u,
  v,
  uz,
  vz,
  salinity,
  temperature,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.ladp_+3A_longitude">longitude</code></td>
<td>
<p>longitude in degrees east, or an <code>oce</code> object that
contains the data otherwise given by <code>longitude</code> and the
other arguments.</p>
</td></tr>
<tr><td><code id="as.ladp_+3A_latitude">latitude</code></td>
<td>
<p>latitude in degrees east (use negative in southern hemisphere).</p>
</td></tr>
<tr><td><code id="as.ladp_+3A_station">station</code></td>
<td>
<p>number or string indicating station ID.</p>
</td></tr>
<tr><td><code id="as.ladp_+3A_time">time</code></td>
<td>
<p>time at the start of the profile, constructed by e.g. <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>.</p>
</td></tr>
<tr><td><code id="as.ladp_+3A_pressure">pressure</code></td>
<td>
<p>pressure in decibars, through the water column.</p>
</td></tr>
<tr><td><code id="as.ladp_+3A_u">u</code></td>
<td>
<p>eastward velocity (m/s).</p>
</td></tr>
<tr><td><code id="as.ladp_+3A_v">v</code></td>
<td>
<p>northward velocity (m/s).</p>
</td></tr>
<tr><td><code id="as.ladp_+3A_uz">uz</code></td>
<td>
<p>vertical derivative of eastward velocity (1/s).</p>
</td></tr>
<tr><td><code id="as.ladp_+3A_vz">vz</code></td>
<td>
<p>vertical derivative of northward velocity (1/s).</p>
</td></tr>
<tr><td><code id="as.ladp_+3A_salinity">salinity</code></td>
<td>
<p>salinity through the water column, in practical salinity units.</p>
</td></tr>
<tr><td><code id="as.ladp_+3A_temperature">temperature</code></td>
<td>
<p>temperature through the water column.</p>
</td></tr>
<tr><td><code id="as.ladp_+3A_...">...</code></td>
<td>
<p>optional additional data columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+ladp-class">ladp</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ladp data: 
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic+ladp-class">ladp-class</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+summary+2Cladp-method">summary,ladp-method</a></code>
</p>

<hr>
<h2 id='as.lisst'>Coerce Data Into a lisst Object</h2><span id='topic+as.lisst'></span>

<h3>Description</h3>

<p>If <code>data</code> contains fewer than 42 columns, an error is reported.  If it
contains more than 42 columns, only the first 42 are used.  This is used by
<code><a href="#topic+read.lisst">read.lisst()</a></code>, the documentation on which explains the meanings
of the columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.lisst(
  data,
  filename = "",
  year = 0,
  tz = "UTC",
  longitude = NA,
  latitude = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.lisst_+3A_data">data</code></td>
<td>
<p>A table (or matrix) containing 42 columns, as in a LISST data
file.</p>
</td></tr>
<tr><td><code id="as.lisst_+3A_filename">filename</code></td>
<td>
<p>Name of file containing the data.</p>
</td></tr>
<tr><td><code id="as.lisst_+3A_year">year</code></td>
<td>
<p>Year in which the first observation was made.  This is necessary
because LISST timestamps do not indicate the year of observation.  The
default value is odd enough to remind users to include this argument.</p>
</td></tr>
<tr><td><code id="as.lisst_+3A_tz">tz</code></td>
<td>
<p>Timezone of observations.  This is necessary because LISST
timestamps do not indicate the timezone.</p>
</td></tr>
<tr><td><code id="as.lisst_+3A_longitude">longitude</code></td>
<td>
<p>Longitude of observation.</p>
</td></tr>
<tr><td><code id="as.lisst_+3A_latitude">latitude</code></td>
<td>
<p>Latitude of observation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+lisst-class">lisst</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to lisst data: 
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+read.lisst">read.lisst</a>()</code>,
<code><a href="#topic+summary+2Clisst-method">summary,lisst-method</a></code>
</p>

<hr>
<h2 id='as.lobo'>Coerce Data Into a lobo Object</h2><span id='topic+as.lobo'></span>

<h3>Description</h3>

<p>Coerce a dataset into a lobo dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.lobo(
  time,
  u,
  v,
  salinity,
  temperature,
  pressure,
  nitrate,
  fluorescence,
  filename = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.lobo_+3A_time">time</code></td>
<td>
<p>vector of times of observation</p>
</td></tr>
<tr><td><code id="as.lobo_+3A_u">u</code></td>
<td>
<p>vector of x velocity component observations</p>
</td></tr>
<tr><td><code id="as.lobo_+3A_v">v</code></td>
<td>
<p>vector of y velocity component observations</p>
</td></tr>
<tr><td><code id="as.lobo_+3A_salinity">salinity</code></td>
<td>
<p>vector of salinity observations</p>
</td></tr>
<tr><td><code id="as.lobo_+3A_temperature">temperature</code></td>
<td>
<p>vector of temperature observations</p>
</td></tr>
<tr><td><code id="as.lobo_+3A_pressure">pressure</code></td>
<td>
<p>vector of pressure observations</p>
</td></tr>
<tr><td><code id="as.lobo_+3A_nitrate">nitrate</code></td>
<td>
<p>vector of nitrate observations</p>
</td></tr>
<tr><td><code id="as.lobo_+3A_fluorescence">fluorescence</code></td>
<td>
<p>vector of fluorescence observations</p>
</td></tr>
<tr><td><code id="as.lobo_+3A_filename">filename</code></td>
<td>
<p>source filename</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+lobo-class">lobo</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to lobo data: 
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+read.lobo">read.lobo</a>()</code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+summary+2Clobo-method">summary,lobo-method</a></code>
</p>

<hr>
<h2 id='as.met'>Coerce Data Into a met Object</h2><span id='topic+as.met'></span>

<h3>Description</h3>

<p>Coerces a dataset into a met dataset. This fills in only a few of the typical
data fields, so the returned object is much
sparser than the output from <code><a href="#topic+read.met">read.met()</a></code>. Also, almost no
metadata fields are filled in, so the resultant object does not store
station location, units of the data, data-quality flags, etc. Anyone working
with data from Environment Canada (reference 2) is advised to use <code><a href="#topic+read.met">read.met()</a></code>
instead of the present function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.met(time, temperature, pressure, u, v, filename = "(constructed from data)")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.met_+3A_time">time</code></td>
<td>
<p>Either a vector of observation times (or character strings that can be
coerced into times) or the output from <code>canadaHCD::hcd_hourly</code> (see reference 1).</p>
</td></tr>
<tr><td><code id="as.met_+3A_temperature">temperature</code></td>
<td>
<p>vector of temperatures.</p>
</td></tr>
<tr><td><code id="as.met_+3A_pressure">pressure</code></td>
<td>
<p>vector of pressures.</p>
</td></tr>
<tr><td><code id="as.met_+3A_u">u</code></td>
<td>
<p>vector of eastward wind speed in m/s.</p>
</td></tr>
<tr><td><code id="as.met_+3A_v">v</code></td>
<td>
<p>vector of northward wind speed in m/s.</p>
</td></tr>
<tr><td><code id="as.met_+3A_filename">filename</code></td>
<td>
<p>optional string indicating data source</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+met-class">met</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> The <code>canadaHCD</code> package is in development by Gavin Simpson; see
<code style="white-space: pre;">&#8288;https://github.com/gavinsimpson/canadaHCD&#8288;</code> for instructions on how
to download and install from GitHub.
</p>
</li>
<li><p> Environment Canada website for Historical Climate Data
<code style="white-space: pre;">&#8288;https://climate.weather.gc.ca/index_e.html&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to met data: 
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic+download.met">download.met</a>()</code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+read.met">read.met</a>()</code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+summary+2Cmet-method">summary,met-method</a></code>
</p>

<hr>
<h2 id='as.oce'>Coerce Something Into an oce Object</h2><span id='topic+as.oce'></span>

<h3>Description</h3>

<p>Coerce Something Into an oce Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.oce(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.oce_+3A_x">x</code></td>
<td>
<p>an item containing data. This may be data frame, list, or an oce object.</p>
</td></tr>
<tr><td><code id="as.oce_+3A_...">...</code></td>
<td>
<p>optional extra arguments, passed to conversion functions
<code><a href="#topic+as.coastline">as.coastline()</a></code> or <code><a href="#topic+ODF2oce">ODF2oce()</a></code>, if these are used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is limited and not intended for common use.
In most circumstances, users should employ a function such
as <code><a href="#topic+as.ctd">as.ctd()</a></code> to construct specialized oce sub-classes.
</p>
<p><code>as.oce</code> creates an oce object from data contained within its
first argument, which may be a list, a data frame, or an object
of <a href="#topic+oce-class">oce</a>.  (In the last case, <code>x</code> is
simply returned, without modification.)
</p>
<p>If <code>x</code> is a list containing items named <code>longitude</code> and
<code>latitude</code>, then <code><a href="#topic+as.coastline">as.coastline()</a></code> is called (with
the specified ... value) to create a coastline object.
</p>
<p>If <code>x</code> is a list created by <code>read_odf()</code> from the (as
yet unreleased) ODF package developed by the Bedford Institute of
Oceanography, then <code><a href="#topic+ODF2oce">ODF2oce()</a></code> is called (with
no arguments other than the first) to calculate a return value.
If the sub-class inference made by <code><a href="#topic+ODF2oce">ODF2oce()</a></code> is
incorrect, users should call that function directly, specifying
a value for its <code>coerce</code> argument.
</p>
<p>If <code>x</code> has not been created by <code>read_odf()</code>, then the names
of the items it contains are examined, and used to try to infer
the proper return value.  There
are only a few cases (although more may be added if there is
sufficient user demand). The cases are as follows.
</p>

<ul>
<li><p> If <code>x</code> contains items named <code>temperature</code>,
<code>pressure</code> and either <code>salinity</code> or <code>conductivity</code>,
then an object of type <a href="#topic+ctd-class">ctd</a> will be returned.
</p>
</li>
<li><p> If <code>x</code> contains columns named <code>longitude</code> and <code>latitude</code>,
but no other columns, then an object of class <a href="#topic+coastline-class">coastline</a>
is returned.
</p>
</li></ul>



<h3>Value</h3>

<p>An <a href="#topic+oce-class">oce</a> object.
</p>

<hr>
<h2 id='as.rsk'>Coerce Data Into a rsk Object</h2><span id='topic+as.rsk'></span>

<h3>Description</h3>

<p>Create a rsk object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.rsk(
  time,
  columns,
  filename = "",
  instrumentType = "rbr",
  serialNumber = "",
  model = "",
  sampleInterval = NA,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.rsk_+3A_time">time</code></td>
<td>
<p>a vector of times for the data.</p>
</td></tr>
<tr><td><code id="as.rsk_+3A_columns">columns</code></td>
<td>
<p>a list or data frame containing the measurements at the indicated
times; see &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="as.rsk_+3A_filename">filename</code></td>
<td>
<p>optional name of file containing the data.</p>
</td></tr>
<tr><td><code id="as.rsk_+3A_instrumenttype">instrumentType</code></td>
<td>
<p>type of instrument.</p>
</td></tr>
<tr><td><code id="as.rsk_+3A_serialnumber">serialNumber</code></td>
<td>
<p>serial number for instrument.</p>
</td></tr>
<tr><td><code id="as.rsk_+3A_model">model</code></td>
<td>
<p>instrument model type, e.g. <code>"RBRduo"</code>.</p>
</td></tr>
<tr><td><code id="as.rsk_+3A_sampleinterval">sampleInterval</code></td>
<td>
<p>sampling interval. If given as <code>NA</code>, then this is
estimated as the median difference in times.</p>
</td></tr>
<tr><td><code id="as.rsk_+3A_debug">debug</code></td>
<td>
<p>a flag that can be set to <code>TRUE</code> to turn on debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The contents of <code>columns</code> are be copied into the <code>data</code> slot
of the returned object directly, so it is critical that the names and units
correspond to those expected by other code dealing with
<a href="#topic+rsk-class">rsk</a> objects. If there is a conductivity, it must be called
<code>conductivity</code>, and it must be in units of mS/cm. If there is a
temperature, it must be called <code>temperature</code>, and it must be an in-situ
value recorded in ITS-90 units.  And if there is a pressure, it must be
<em>absolute</em> pressure (sea pressure plus atmospheric pressure) and it must
be named <code>pressure</code>. No checks are made within <code>as.rsk</code> on any of
these rules, but if they are broken, you may expect problems with any further
processing.
</p>


<h3>Value</h3>

<p>An <a href="#topic+rsk-class">rsk</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+read.rsk">read.rsk</a>()</code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+rskPatm">rskPatm</a>()</code>,
<code><a href="#topic+rskToc">rskToc</a>()</code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method</a></code>
</p>

<hr>
<h2 id='as.sealevel'>Coerce Data Into a sealevel Object</h2><span id='topic+as.sealevel'></span>

<h3>Description</h3>

<p>Coerces a dataset (minimally, a sequence of times and heights) into a
sealevel dataset.
The arguments are based on the standard data format, as were described in a
file formerly available at reference 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.sealevel(
  elevation,
  time,
  header = NULL,
  stationNumber = NA,
  stationVersion = NA,
  stationName = NULL,
  region = NULL,
  year = NA,
  longitude = NA,
  latitude = NA,
  GMTOffset = NA,
  decimationMethod = NA,
  referenceOffset = NA,
  referenceCode = NA,
  deltat
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.sealevel_+3A_elevation">elevation</code></td>
<td>
<p>a list of sea-level heights in metres, in an hourly
sequence.</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_time">time</code></td>
<td>
<p>optional list of times, in POSIXct format.  If missing, the list
will be constructed assuming hourly samples, starting at 0000-01-01
00:00:00.</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_header">header</code></td>
<td>
<p>a character string as read from first line of a standard data
file.</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_stationnumber">stationNumber</code></td>
<td>
<p>three-character string giving station number.</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_stationversion">stationVersion</code></td>
<td>
<p>single character for version of station.</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_stationname">stationName</code></td>
<td>
<p>the name of station (at most 18 characters).</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_region">region</code></td>
<td>
<p>the name of the region or country of station (at most 19
characters).</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_year">year</code></td>
<td>
<p>the year of observation.</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_longitude">longitude</code></td>
<td>
<p>the longitude in decimal degrees, positive east of
Greenwich.</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_latitude">latitude</code></td>
<td>
<p>the latitude in decimal degrees, positive north of the
equator.</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_gmtoffset">GMTOffset</code></td>
<td>
<p>offset from GMT, in hours.</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_decimationmethod">decimationMethod</code></td>
<td>
<p>a coded value, with 1 meaning filtered, 2 meaning a
simple average of all samples, 3 meaning spot readings, and 4 meaning some
other method.</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_referenceoffset">referenceOffset</code></td>
<td>
<p>?</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_referencecode">referenceCode</code></td>
<td>
<p>?</p>
</td></tr>
<tr><td><code id="as.sealevel_+3A_deltat">deltat</code></td>
<td>
<p>optional interval between samples, in hours (as for the
<code><a href="stats.html#topic+ts">ts()</a></code> timeseries function). If this is not provided, and <code>t</code>
can be understood as a time, then the difference between the first two times
is used.  If this is not provided, and <code>t</code> cannot be understood as a
time, then 1 hour is assumed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+sealevel-class">sealevel</a> object (for details, see <code><a href="#topic+read.sealevel">read.sealevel()</a></code>).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p><code style="white-space: pre;">&#8288;http://ilikai.soest.hawaii.edu/rqds/hourly.fmt&#8288;</code> (this link
worked for years but failed at least temporarily on December 4, 2016).
</p>


<h3>See Also</h3>

<p>The documentation for the <a href="#topic+sealevel-class">sealevel</a> class explains the
structure of sealevel objects, and also outlines the other functions dealing
with them.
</p>
<p>Other things related to sealevel data: 
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+read.sealevel">read.sealevel</a>()</code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+summary+2Csealevel-method">summary,sealevel-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Construct a year of M2 tide, starting at the default time
# 0000-01-01T00:00:00.
h &lt;- seq(0, 24 * 365)
elevation &lt;- 2.0 * sin(2 * pi * h / 12.4172)
sl &lt;- as.sealevel(elevation)
summary(sl)

# As above, but start at the Y2K time.
time &lt;- as.POSIXct("2000-01-01") + h * 3600
sl &lt;- as.sealevel(elevation, time)
summary(sl)
</code></pre>

<hr>
<h2 id='as.section'>Create a Section</h2><span id='topic+as.section'></span>

<h3>Description</h3>

<p>Create a section based on columnar data, or a set of <a href="#topic+oce-class">oce</a>
objects that can be coerced to a section. There are three cases.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.section(
  salinity,
  temperature,
  pressure,
  longitude,
  latitude,
  station,
  sectionId = "",
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.section_+3A_salinity">salinity</code></td>
<td>
<p>This may be a numerical vector, in which case it is interpreted
as the salinity, and the other arguments are used for the other components of
<a href="#topic+ctd-class">ctd</a> objects. Alternatively, it may be one of a variety of
other objects from which the CTD objects can be inferred, in which case the
other arguments are ignored; see &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="as.section_+3A_temperature">temperature</code></td>
<td>
<p>Temperature, in a vector holding values for all stations.</p>
</td></tr>
<tr><td><code id="as.section_+3A_pressure">pressure</code></td>
<td>
<p>Pressure, in a vector holding values for all stations.</p>
</td></tr>
<tr><td><code id="as.section_+3A_longitude">longitude</code></td>
<td>
<p>Longitude, in a vector holding values for all stations.</p>
</td></tr>
<tr><td><code id="as.section_+3A_latitude">latitude</code></td>
<td>
<p>Latitude, in a vector holding values for all stations.</p>
</td></tr>
<tr><td><code id="as.section_+3A_station">station</code></td>
<td>
<p>Station identifiers, in a vector holding values for all stations.</p>
</td></tr>
<tr><td><code id="as.section_+3A_sectionid">sectionId</code></td>
<td>
<p>Section identifier.</p>
</td></tr>
<tr><td><code id="as.section_+3A_debug">debug</code></td>
<td>
<p>an integer value that controls whether <code>as.section()</code> prints information
during its work.  The function works quietly if this is 0 and prints out some
information if it is positive.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Case 1. If the first argument is a numerical vector, then it is taken to be the
salinity, and <code><a href="base.html#topic+factor">factor()</a></code> is applied to <code>station</code> to break the
data up into chunks that are assembled into <a href="#topic+ctd-class">ctd</a> objects with
<code><a href="#topic+as.ctd">as.ctd()</a></code> and combined to make a <a href="#topic+section-class">section</a> object
to be returned. This mode of operation is provided as a convenience for datasets
that are already partly processed; if original CTD data are available, the next
mode is preferred, because it permits the storage of much more data and metadata
in the CTD object.
</p>
<p>Case 2. If the first argument is a list containing oce objects, then those
objects are taken as profiles of something.  A requirement for this
to work is that every element of the list contains both <code>longitude</code>
and <code>latitude</code> in either the <code>metadata</code> or <code>data</code> slot (in
the latter case, the mean value is recorded in the section object)
and that every element also contains <code>pressure</code> in its <code>data</code> slot.
</p>
<p>Case 3. If the first argument is a <a href="#topic+argo-class">argo</a> object, then
the profiles it contains are turned into <a href="#topic+ctd-class">ctd</a> objects,
and these are assembled into a section to be returned.
</p>


<h3>Value</h3>

<p>An object of <a href="#topic+section-class">section</a>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
# vector of names of CTD objects
fake &lt;- ctd
fake[["temperature"]] &lt;- ctd[["temperature"]] + 0.5
fake[["salinity"]] &lt;- ctd[["salinity"]] + 0.1
fake[["longitude"]] &lt;- ctd[["longitude"]] + 0.01
fake[["station"]] &lt;- "fake"
sec1 &lt;- as.section(c("ctd", "fake"))
summary(sec1)
# vector of CTD objects
ctds &lt;- vector("list", 2)
ctds[[1]] &lt;- ctd
ctds[[2]] &lt;- fake
sec2 &lt;- as.section(ctds)
summary(sec2)
# argo data (a subset)
data(argo)
sec3 &lt;- as.section(subset(argo, profile &lt; 5))
summary(sec3)

</code></pre>

<hr>
<h2 id='as.tidem'>Create tidem Object From Fitted Harmonic Data</h2><span id='topic+as.tidem'></span>

<h3>Description</h3>

<p>This function takes a set of tidal constituent amplitudes
and phases, and constructs a return value of similar form
to that returned by <code><a href="#topic+tidem">tidem()</a></code>.  Its purpose is to enable
predictions based on published constituent amplitudes
and phases.  Since <code><a href="#topic+as.tidem">as.tidem()</a></code> does not account for a
reference height, it is the user's responsible to account
for this after a prediction is made using <code><a href="#topic+predict.tidem">predict.tidem()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.tidem(
  tRef,
  latitude,
  name,
  amplitude,
  phase,
  frequency,
  speed,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.tidem_+3A_tref">tRef</code></td>
<td>
<p>a POSIXt value indicating the mean time of the
observations used to develop the harmonic model. This is rounded
to the nearest hour in <code><a href="#topic+as.tidem">as.tidem()</a></code>, to match the behaviour
of <code><a href="#topic+tidem">tidem()</a></code>.</p>
</td></tr>
<tr><td><code id="as.tidem_+3A_latitude">latitude</code></td>
<td>
<p>numerical value indicating the latitude of the
observations that were used to create the harmonic model. This
is needed for nodal-correction procedures carried out
by <code><a href="#topic+tidemVuf">tidemVuf()</a></code>.</p>
</td></tr>
<tr><td><code id="as.tidem_+3A_name">name</code></td>
<td>
<p>character vector holding names of constituents.</p>
</td></tr>
<tr><td><code id="as.tidem_+3A_amplitude">amplitude</code>, <code id="as.tidem_+3A_phase">phase</code></td>
<td>
<p>numeric vectors of constituent amplitudes
and phases. These must be of the same length as <code>name</code>.</p>
</td></tr>
<tr><td><code id="as.tidem_+3A_frequency">frequency</code>, <code id="as.tidem_+3A_speed">speed</code></td>
<td>
<p>optional numeric vectors giving the frequencies
of the constituents (in cycles per hour) or the analogous speeds
(in degrees per hour).  Only one of these may be given, and a conversion
is done from the latter to the former, if required.  If the frequencies
are thus specified, then these are used instead of the frequencies that
oce normally used, as defined in <code>data(tideconst)</code>. A warning will
be issued if the absolute value of the relative frequency mismatch for any
given component exceeds 1e-6, and this will occur for any NOAA tables
containing the SA component, for which this relative mismatch
is approximately 4e-5 (see reference 5).</p>
</td></tr>
<tr><td><code id="as.tidem_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All the constituent names used by <code><a href="#topic+tidem">tidem()</a></code> are permitted here,
<em>except</em> for <code>"Z0"</code> (see &lsquo;Description&rsquo; regarding reference
height).
To get a list of constituent names, please consult Foreman (1978),
or type the following in an R console:
</p>
<pre>
data(tidedata)
data.frame(name=tidedata$const$name, freq=tidedata$const$freq)
</pre>
<p>In addition to the above, <code><a href="#topic+as.tidem">as.tidem()</a></code> can handle NOAA names
of constituents.  For the most part, these match oce names, but
there are 4 exceptions: NOAA names
<code>"LAM2"</code>, <code>"M1"</code>, <code>"RHO"</code>, and <code>"2MK3"</code> are converted to oce names
<code>"LDA2"</code>, <code>"NO1"</code>, <code>"RHO1"</code>, and <code>"MO3"</code>. The name mapping was
inferred by matching frequencies; for these constituents, the
fractional mismatch in frequencies was under 4e-8;
see Reference 5 for more details.
A message is printed if these name conversions are required
in the particular use of <code><a href="#topic+as.tidem">as.tidem()</a></code>.
</p>
<p>Apart from the standard oce names and this set of NOAA synonyms,
any other constituent name is reported in a warning message.
</p>


<h3>Value</h3>

<p>An object of <a href="#topic+tidem-class">tidem</a>, with only minimal
contents.
</p>


<h3>Known issues</h3>

<p>There are two known differences between <code><a href="#topic+tidem">tidem()</a></code> and the Matlab
<code>T_TIDE</code> package, as listed in references 3 and 4.
</p>


<h3>References</h3>


<ol>
<li><p> Foreman, M. G. G., 1978. Manual for Tidal Currents Analysis and Prediction.
Pacific Marine Science Report. British Columbia, Canada: Institute of Ocean
Sciences, Patricia Bay.
</p>
</li>
<li><p> Wikipedia, &quot;Theory of Tides.&quot; https://en.wikipedia.org/wiki/Theory_of_tides
Downloaded Aug 17, 2019.
</p>
</li>
<li><p> Github issue 1653 &quot;tidem() and t_tide do not produce identical results&quot;
(https://github.com/dankelley/oce/issues/1653)
</p>
</li>
<li><p> Github issue 1654 &quot;predict(tidem()) uses all constituents, unlike T_TIDE&quot;
(https://github.com/dankelley/oce/issues/1654)
</p>
</li>
<li><p> Github issue 2143 &quot;mismatch in oce/NOAA frequency of SA tidal constituent&quot;
(https://github.com/dankelley/oce/issues/2143)
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Example 1: show agreement with tidem()
data(sealevelTuktoyaktuk)
# 'm0' is model fitted by tidem()
m0 &lt;- tidem(sealevelTuktoyaktuk)
p0 &lt;- predict(m0, sealevelTuktoyaktuk[["time"]])
m1 &lt;- as.tidem(
    mean(sealevelTuktoyaktuk[["time"]]), sealevelTuktoyaktuk[["latitude"]],
    m0[["name"]], m0[["amplitude"]], m0[["phase"]]
)
# Test agreement with tidem() result, by comparing predicted sealevels.
p1 &lt;- predict(m1, sealevelTuktoyaktuk[["time"]])
stopifnot(max(abs(p1 - p0), na.rm = TRUE) &lt; 1e-10)

# Example 2: See the effect of dropping weak constituents
m0[["name"]][which(m0[["amplitude"]] &gt; 0.05)]
h &lt;- "
name  amplitude      phase
  Z0 1.98061875   0.000000
  MM 0.21213065 263.344739
 MSF 0.15605629 133.795004
  O1 0.07641438  74.233130
  K1 0.13473817  81.093134
 OO1 0.05309911 235.749693
  N2 0.08377108  44.521462
  M2 0.49041340  77.703594
  S2 0.22023705 137.475767"
coef &lt;- read.table(text = h, header = TRUE)
m2 &lt;- as.tidem(
    mean(sealevelTuktoyaktuk[["time"]]),
    sealevelTuktoyaktuk[["latitude"]],
    coef$name, coef$amplitude, coef$phase
)
p2 &lt;- predict(m2, sealevelTuktoyaktuk[["time"]])
par(mfrow = c(3, 1))
oce.plot.ts(sealevelTuktoyaktuk[["time"]], p0)
ylim &lt;- par("usr")[3:4] # to match scales in other panels
oce.plot.ts(sealevelTuktoyaktuk[["time"]], p1, ylim = ylim)
oce.plot.ts(sealevelTuktoyaktuk[["time"]], p2, ylim = ylim)

</code></pre>

<hr>
<h2 id='as.topo'>Coerce Data Into a topo Object</h2><span id='topic+as.topo'></span>

<h3>Description</h3>

<p>Coerce Data Into a topo Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.topo(longitude, latitude, z, filename = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.topo_+3A_longitude">longitude</code></td>
<td>
<p>Either a vector of longitudes (in degrees east, and bounded by
-180 and 180), or a <code>bathy</code> object created by <code>getNOAA.bathy()</code> from
the <code>marmap</code> package; in the second case, all other arguments are ignored.</p>
</td></tr>
<tr><td><code id="as.topo_+3A_latitude">latitude</code></td>
<td>
<p>A vector of latitudes.</p>
</td></tr>
<tr><td><code id="as.topo_+3A_z">z</code></td>
<td>
<p>A matrix of heights (positive over land).</p>
</td></tr>
<tr><td><code id="as.topo_+3A_filename">filename</code></td>
<td>
<p>Name of data (used when called by <code><a href="#topic+read.topo">read.topo()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+topo-class">topo</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to topo data: 
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+read.topo">read.topo</a>()</code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+summary+2Ctopo-method">summary,topo-method</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+topoInterpolate">topoInterpolate</a>()</code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>
</p>

<hr>
<h2 id='as.unit'>Convert a String to a Unit</h2><span id='topic+as.unit'></span>

<h3>Description</h3>

<p>This converts strings to unit objects.  Only a few strings are recognized,
because most oce functions have specialized unit vocabularies and so
have little need of this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.unit(u, default = list(unit = expression(), scale = ""))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.unit_+3A_u">u</code></td>
<td>
<p>A character string indicating a unit. Case is ignored, so that e.g.
<code>"dbar"</code> and <code>"DBAR"</code> yield equal results.  The following are recognized:
c(<code>"m-1"</code>, <code>"dbar"</code>, <code>"decibar"</code>, <code>"degree"</code>, <code>"degree_Celcius"</code>,
<code>"degree_north"</code>, <code>"degree_east"</code>, <code>"ipts-68"</code>,
<code>"its-90"</code>, <code>"m/s^1"</code>, <code>"m/s^2"</code>, <code>"pss-78"</code>,
<code>"umol/kg"</code>, <code>"micromole/kg"</code>)</p>
</td></tr>
<tr><td><code id="as.unit_+3A_default">default</code></td>
<td>
<p>A default to be used for the return value, if <code>u</code>
is not a recognized string.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with elements <code>unit</code>, an <code><a href="base.html#topic+expression">expression()</a></code>,
and <code>scale</code>, a string.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>as.unit("DBAR")
as.unit("IPTS-68")
as.unit("ITS-90")
as.unit("PSS-78")
as.unit("UMOL/KG")

</code></pre>

<hr>
<h2 id='as.windrose'>Create a windrose Object</h2><span id='topic+as.windrose'></span>

<h3>Description</h3>

<p>Create a wind-rose object, typically for plotting with
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.windrose(x, y, dtheta = 15, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.windrose_+3A_x">x</code></td>
<td>
<p>The x component of wind speed (or stress) <em>or</em> an object of class
<code>met</code> (see <a href="#topic+met-class">met</a>), in which case the <code>u</code> and
<code>v</code> components of that object are used for the components of wind speed,
and <code>y</code> here is ignored.</p>
</td></tr>
<tr><td><code id="as.windrose_+3A_y">y</code></td>
<td>
<p>The y component of wind speed (or stress).</p>
</td></tr>
<tr><td><code id="as.windrose_+3A_dtheta">dtheta</code></td>
<td>
<p>The angle increment (in degrees) within which to classify the data.</p>
</td></tr>
<tr><td><code id="as.windrose_+3A_debug">debug</code></td>
<td>
<p>A flag that turns on debugging.  Set to 1 to get a moderate amount
of debugging information, or to 2 to get more.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+windrose-class">windrose</a> object, with <code>data</code> slot containing
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Item</strong>   </td><td style="text-align: left;"> <strong>Meaning</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n</code>       </td><td style="text-align: left;"> the number of <code>x</code> values</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>x.mean</code>  </td><td style="text-align: left;"> the mean of the <code>x</code> values</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>y.mean</code>  </td><td style="text-align: left;"> the mean of the <code>y</code> values</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>theta</code>   </td><td style="text-align: left;"> the central angle (in degrees) for the class</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>count</code>   </td><td style="text-align: left;"> the number of observations in this class</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>mean</code>    </td><td style="text-align: left;"> the mean of the observations in this class</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>fivenum</code> </td><td style="text-align: left;"> the <code><a href="stats.html#topic+fivenum">fivenum()</a></code> vector for
observations in this class (the min, the lower hinge, the
median, the upper hinge, and the max)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Dan Kelley, with considerable help from Alex Deckmyn.
</p>


<h3>See Also</h3>

<p>Other things related to windrose data: 
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+summary+2Cwindrose-method">summary,windrose-method</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
set.seed(1234)
theta &lt;- seq(0, 360, 0.25)
x &lt;- 1 + cos(pi / 180 * theta) + rnorm(theta)
y &lt;- sin(pi / 180 * theta) + rnorm(theta)
wr &lt;- as.windrose(x, y)
summary(wr)

</code></pre>

<hr>
<h2 id='as.xbt'>Create an xbt Object</h2><span id='topic+as.xbt'></span>

<h3>Description</h3>

<p>Create an xbt Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.xbt(
  z,
  temperature,
  longitude = NA,
  latitude = NA,
  filename = "",
  sequenceNumber = NA,
  serialNumber = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.xbt_+3A_z">z</code></td>
<td>
<p>numeric vector giving vertical coordinates of measurements. This is the negative of
depth, i.e. <code>z</code> is 0 at the air-sea interface, and negative within the water column.</p>
</td></tr>
<tr><td><code id="as.xbt_+3A_temperature">temperature</code></td>
<td>
<p>numeric vector giving in-situ temperatures at the <code>z</code> values.</p>
</td></tr>
<tr><td><code id="as.xbt_+3A_longitude">longitude</code>, <code id="as.xbt_+3A_latitude">latitude</code></td>
<td>
<p>location in degE and degN.</p>
</td></tr>
<tr><td><code id="as.xbt_+3A_filename">filename</code></td>
<td>
<p>character value naming source file.</p>
</td></tr>
<tr><td><code id="as.xbt_+3A_sequencenumber">sequenceNumber</code></td>
<td>
<p>numerical value of the sequence number of the XBT drop.</p>
</td></tr>
<tr><td><code id="as.xbt_+3A_serialnumber">serialNumber</code></td>
<td>
<p>character value holding the serial number of the XBT.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+xbt-class">xbt</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to xbt data: 
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1</a>()</code>,
<code><a href="#topic+read.xbt">read.xbt</a>()</code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>,
<code><a href="#topic+summary+2Cxbt-method">summary,xbt-method</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>

<hr>
<h2 id='bcdToInteger'>Convert a BCD Value to an Integer Value</h2><span id='topic+bcdToInteger'></span>

<h3>Description</h3>

<p>Convert a BCD Value to an Integer Value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bcdToInteger(x, endian = c("little", "big"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bcdToInteger_+3A_x">x</code></td>
<td>
<p>a raw value, or vector of raw values, coded in binary-coded
decimal.</p>
</td></tr>
<tr><td><code id="bcdToInteger_+3A_endian">endian</code></td>
<td>
<p>character string indicating the endian-ness (&quot;big&quot; or
&quot;little&quot;).  The PC/intel convention is to use &quot;little&quot;, and so most data
files are in that format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer, or list of integers.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
twenty.five &lt;- bcdToInteger(as.raw(0x25))
thirty.seven &lt;- as.integer(as.raw(0x25))
</code></pre>

<hr>
<h2 id='beamName'>Get Names of Acoustic-Doppler Beams</h2><span id='topic+beamName'></span>

<h3>Description</h3>

<p>Get Names of Acoustic-Doppler Beams
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beamName(x, which)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="beamName_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="beamName_+3A_which">which</code></td>
<td>
<p>an integer indicating beam number.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string containing a reasonable name for the beam, of the
form <code>"beam 1"</code>, etc., for beam coordinates, <code>"east"</code>, etc. for
enu coordinates, <code>"u"</code>, etc. for <code>"xyz"</code>, or <code>"u'"</code>, etc.,
for <code>"other"</code> coordinates.  The coordinate system is determined
with <code>x[["coordinate"]]</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>This is used by <code><a href="#topic+read.oce">read.oce()</a></code>.
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='beamToXyz'>Change the Coordinate System in an adv or adp Object</h2><span id='topic+beamToXyz'></span>

<h3>Description</h3>

<p>Convert velocity data from an acoustic-Doppler velocimeter or
acoustic-Doppler profiler from one coordinate system to another.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beamToXyz(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="beamToXyz_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> or <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="beamToXyz_+3A_...">...</code></td>
<td>
<p>extra arguments that are passed on to <code><a href="#topic+beamToXyzAdp">beamToXyzAdp()</a></code>
or <code><a href="#topic+beamToXyzAdv">beamToXyzAdv()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same class as <code>x</code>, but with velocities
in xyz coordinates instead of beam coordinates.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='beamToXyzAdp'>Convert adp Object From Beam to XYZ Coordinates</h2><span id='topic+beamToXyzAdp'></span>

<h3>Description</h3>

<p>Convert ADP velocity components from a beam-based coordinate system to a
xyz-based coordinate system. The action depends on the type of object.
Objects creating by reading RDI Teledyne, Sontek, and some Nortek
instruments are handled directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beamToXyzAdp(x, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="beamToXyzAdp_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="beamToXyzAdp_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a 3-beam Nortek <code>aquadopp</code> object, the beams are transformed into
velocities using the matrix stored in the header.
</p>
<p>For 4-beam objects (and for the slanted 4 beams of 5-beam
objects), the along-beam velocity components <code class="reqn">B_1</code>
<code class="reqn">B_2</code>, <code class="reqn">B_3</code>, and <code class="reqn">B_4</code>
are converted to Cartesian velocity components <code class="reqn">u</code>
<code class="reqn">v</code> and <code class="reqn">w</code>
using formulae from section 5.5 of <em>RD Instruments</em> (1998), viz. the
along-beam velocity components <code class="reqn">B_1</code>, <code class="reqn">B_2</code>, <code class="reqn">B_3</code>,
and <code class="reqn">B_4</code> are used to calculate velocity components in a cartesian
system referenced to the instrument using the following formulae:
<code class="reqn">u=ca(B_1-B_2)</code>, <code class="reqn">v=ca(B_4-B_3)</code>,
<code class="reqn">w=-b(B_1+B_2+B_3+B_4)</code>. In addition to these,
an estimate of the
error in velocity is computed as
<code class="reqn">e=d(B_1+B_2-B_3-B_4)</code>.
The geometrical factors in these formulae are:
<code>c</code> is +1 for convex beam geometry or -1 for concave beam geometry,
<code class="reqn">a=1/(2\sin\theta)</code>
where <code class="reqn">\theta</code> is the angle the beams make to the axial direction
(which is available as <code>x[["beamAngle"]]</code>),
<code class="reqn">b=1/(4\cos\theta)</code>, and
<code class="reqn">d=a/\sqrt{2}</code>.
</p>


<h3>Value</h3>

<p>An object with the first 3 velocity indices having been altered to
represent velocity components in xyz (or instrument) coordinates.  (For
<code>rdi</code> data, the values at the 4th velocity index are changed to
represent the &quot;error&quot; velocity.)
To indicate the change, the value of <code>x[["oceCoordinate"]]</code> is
changed from <code>beam</code> to <code>xyz</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Teledyne RD Instruments. &ldquo;ADCP Coordinate Transformation: Formulas and
Calculations,&rdquo; January 2010. P/N 951-6079-00.
</p>
</li>
<li><p> WHOI/USGS-provided Matlab code for beam-enu transformation
<code style="white-space: pre;">&#8288;http://woodshole.er.usgs.gov/pubs/of2005-1429/MFILES/AQDPTOOLS/beam2enu.m&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>See <code><a href="#topic+read.adp">read.adp()</a></code> for other functions that relate to
objects of class <code>"adp"</code>.
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='beamToXyzAdpAD2CP'>Convert From Beam to XYZ Coordinates (AD2CP adp Data)</h2><span id='topic+beamToXyzAdpAD2CP'></span>

<h3>Description</h3>

<p>This looks at all the items in the <code>data</code> slot of <code>x</code>, to
see if they contain an array named <code>v</code> that holds velocity.
If that velocity has 4 components, and if <code>oceCoordinate</code> for
the item is <code>"beam"</code>, then
along-beam velocity components <code class="reqn">B_1</code>
<code class="reqn">B_2</code>, <code class="reqn">B_3</code>, and <code class="reqn">B_4</code>
are converted to instrument-oriented Cartesian velocity components <code class="reqn">u</code>
<code class="reqn">v</code> and <code class="reqn">w</code>
using the convex-geometry formulae from section 5.5 of reference 1,
viz.
<code class="reqn">u=ca(B_1-B_2)</code>, <code class="reqn">v=ca(B_4-B_3)</code>,
<code class="reqn">w=-b(B_1+B_2+B_3+B_4)</code>. In addition to these,
an estimate of the
error in velocity is computed as
<code class="reqn">e=d(B_1+B_2-B_3-B_4)</code>.
The geometrical factors in these formulae are:
<code class="reqn">a=1/(2\sin\theta)</code>
where <code class="reqn">\theta</code> is the angle the beams make to the axial direction
(which is available as <code>x[["beamAngle"]]</code>),
<code class="reqn">b=1/(4\cos\theta)</code>, and
<code class="reqn">d=a/\sqrt{2}</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beamToXyzAdpAD2CP(x, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="beamToXyzAdpAD2CP_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="beamToXyzAdpAD2CP_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>References</h3>


<ol>
<li><p> Teledyne RD Instruments.
&ldquo;ADCP Coordinate Transformation: Formulas and Calculations,&rdquo;
January 2010. P/N 951-6079-00.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='beamToXyzAdv'>Convert adv Object from Beam Coordinates to XYZ Coordinates</h2><span id='topic+beamToXyzAdv'></span>

<h3>Description</h3>

<p>Convert ADV velocity components from a beam-based coordinate system to a
xyz-based coordinate system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beamToXyzAdv(x, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="beamToXyzAdv_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="beamToXyzAdv_+3A_debug">debug</code></td>
<td>
<p>a flag that, if non-zero, turns on debugging.  Higher values
yield more extensive debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The coordinate transformation is done using the transformation matrix
contained in <code>transformation.matrix</code> in the
<code>metadata</code> slot, which is normally
inferred from the header in the binary file.  If there is no such matrix
(e.g. if the data were streamed through a data logger that did not capture
the header), <code>beamToXyzAdv</code> the user will need to store one in
<code>x</code>, e.g. by doing something like the following:
</p>
<div class="sourceCode"><pre>x[["transformation.matrix"]] &lt;- rbind(c(11100, -5771, -5321),
                                      c( #' 291, 9716, -10002),
                                      c( 1409, 1409, 1409)) / 4096
</pre></div>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li> <p><code style="white-space: pre;">&#8288;https://nortek.zendesk.com/hc/en-us/articles/360029820971-How-is-a-Coordinate-transformation-done-&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>See <code><a href="#topic+read.adv">read.adv()</a></code> for notes on functions relating to
<code>"adv"</code> objects.
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='beamUnspreadAdp'>Adjust adp Object to Account for Spherical Spreading</h2><span id='topic+beamUnspreadAdp'></span>

<h3>Description</h3>

<p>Compensate ADP signal strength for spherical spreading.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beamUnspreadAdp(
  x,
  count2db = c(0.45, 0.45, 0.45, 0.45),
  asMatrix = FALSE,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="beamUnspreadAdp_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="beamUnspreadAdp_+3A_count2db">count2db</code></td>
<td>
<p>a set of coefficients, one per beam, to convert from beam
echo intensity to decibels.</p>
</td></tr>
<tr><td><code id="beamUnspreadAdp_+3A_asmatrix">asMatrix</code></td>
<td>
<p>a boolean that indicates whether to return a numeric matrix,
as opposed to returning an updated object (in which the matrix is cast to a
raw value).</p>
</td></tr>
<tr><td><code id="beamUnspreadAdp_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>First, beam echo intensity is converted from counts to decibels, by
multiplying by <code>count2db</code>.  Then, the signal decrease owing to
spherical spreading is compensated for by adding the term
<code class="reqn">20\log10(r)</code>, where <code class="reqn">r</code> is the distance from the
sensor head to the water from which scattering is occurring.  <code class="reqn">r</code> is
given by <code>x[["distance"]]</code>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>The coefficient to convert to decibels is a personal
communication.  The logarithmic term is explained in textbooks on acoustics,
optics, etc.
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adp)
plot(adp, which = 5) # beam 1 echo intensity
adp.att &lt;- beamUnspreadAdp(adp)
plot(adp.att, which = 5) # beam 1 echo intensity
# Profiles
par(mar = c(4, 4, 1, 1))
a &lt;- adp[["a", "numeric"]] # second arg yields matrix return value
distance &lt;- adp[["distance"]]
plot(apply(a, 2, mean), distance, type = "l", xlim = c(0, 256))
lines(apply(a, 2, median), distance, type = "l", col = "red")
legend("topright", lwd = 1, col = c("black", "red"), legend = c("original", "attenuated"))
# Image
plot(adp.att, which = "amplitude", col = oce.colorsViridis(100))

</code></pre>

<hr>
<h2 id='bilinearInterp'>Bilinear Interpolation Within a Grid</h2><span id='topic+bilinearInterp'></span>

<h3>Description</h3>

<p>This is used by <code><a href="#topic+topoInterpolate">topoInterpolate</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bilinearInterp(x, y, gx, gy, g)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bilinearInterp_+3A_x">x</code></td>
<td>
<p>vector of x values at which to interpolate</p>
</td></tr>
<tr><td><code id="bilinearInterp_+3A_y">y</code></td>
<td>
<p>vector of y values at which to interpolate</p>
</td></tr>
<tr><td><code id="bilinearInterp_+3A_gx">gx</code></td>
<td>
<p>vector of x values for the grid</p>
</td></tr>
<tr><td><code id="bilinearInterp_+3A_gy">gy</code></td>
<td>
<p>vector of y values for the grid</p>
</td></tr>
<tr><td><code id="bilinearInterp_+3A_g">g</code></td>
<td>
<p>matrix of the grid values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of interpolated values
</p>

<hr>
<h2 id='binApply1D'>Apply a Function to Vector Data</h2><span id='topic+binApply1D'></span>

<h3>Description</h3>

<p>The function <code>FUN</code> is applied to <code>f</code> in bins specified by
<code>xbreaks</code>.  The division of data into bins is done with <code><a href="base.html#topic+cut">cut()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binApply1D(x, f, xbreaks, FUN, include.lowest = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binApply1D_+3A_x">x</code></td>
<td>
<p>a vector of numerical values.</p>
</td></tr>
<tr><td><code id="binApply1D_+3A_f">f</code></td>
<td>
<p>a vector of data to which <code>FUN</code> will be applied.</p>
</td></tr>
<tr><td><code id="binApply1D_+3A_xbreaks">xbreaks</code></td>
<td>
<p>optional vector holding values of x at the boundaries between bins.
If this is not given, it is computed by calling <code><a href="base.html#topic+pretty">pretty()</a></code> with n=20 segments.</p>
</td></tr>
<tr><td><code id="binApply1D_+3A_fun">FUN</code></td>
<td>
<p>function that is applied to the <code>f</code> values
in each x bin.  This must take a single numeric vector
as input, and return a single numeric value.</p>
</td></tr>
<tr><td><code id="binApply1D_+3A_include.lowest">include.lowest</code></td>
<td>
<p>logical value indicating whether to include
<code>x</code> values that equal <code>xbreaks[1]</code>.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="binApply1D_+3A_...">...</code></td>
<td>
<p>optional arguments to pass to <code>FUN</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, the sub-intervals defined by the <code>xbreaks</code> argument are open
on the left and closed on the right, to match the behaviour
of <code><a href="base.html#topic+cut">cut()</a></code>.  An open interval does not include points on
the boundary, and so any <code>x</code> values that exactly match
the first <code>breaks</code> value will not be counted.  To include
such points in the calculation, set <code>include.lowest</code> to TRUE.
</p>


<h3>Value</h3>

<p>A list with the following elements: <code>xbreaks</code> as
used, <code>xmids</code> (the mid-points between those breaks) and
<code>result</code> (the result of applying <code>FUN</code> to the <code>f</code> values
in the designated bins).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other bin-related functions: 
<code><a href="#topic+binApply2D">binApply2D</a>()</code>,
<code><a href="#topic+binAverage">binAverage</a>()</code>,
<code><a href="#topic+binCount1D">binCount1D</a>()</code>,
<code><a href="#topic+binCount2D">binCount2D</a>()</code>,
<code><a href="#topic+binMean1D">binMean1D</a>()</code>,
<code><a href="#topic+binMean2D">binMean2D</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# salinity profile (black) with 1-dbar bin means (red)
data(ctd)
plotProfile(ctd, "salinity")
p &lt;- ctd[["pressure"]]
S &lt;- ctd[["salinity"]]
pbreaks &lt;- seq(0, max(p), 1)
binned &lt;- binApply1D(p, S, pbreaks, mean)
lines(binned$result, binned$xmids, lwd = 2, col = rgb(1, 0, 0, 0.9))

</code></pre>

<hr>
<h2 id='binApply2D'>Apply a Function to Matrix Data</h2><span id='topic+binApply2D'></span>

<h3>Description</h3>

<p>The function <code>FUN</code> is applied to <code>f</code> in bins specified by
<code>xbreaks</code> and <code>ybreaks</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binApply2D(x, y, f, xbreaks, ybreaks, FUN, include.lowest = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binApply2D_+3A_x">x</code></td>
<td>
<p>a vector of numerical values.</p>
</td></tr>
<tr><td><code id="binApply2D_+3A_y">y</code></td>
<td>
<p>a vector of numerical values.</p>
</td></tr>
<tr><td><code id="binApply2D_+3A_f">f</code></td>
<td>
<p>a vector of data to which <code>FUN</code> will be applied.</p>
</td></tr>
<tr><td><code id="binApply2D_+3A_xbreaks">xbreaks</code></td>
<td>
<p>values of <code>x</code> at the boundaries between the
bins; calculated using <code><a href="base.html#topic+pretty">pretty()</a></code> if not supplied.</p>
</td></tr>
<tr><td><code id="binApply2D_+3A_ybreaks">ybreaks</code></td>
<td>
<p>as <code>xbreaks</code>, but for <code>y</code>.</p>
</td></tr>
<tr><td><code id="binApply2D_+3A_fun">FUN</code></td>
<td>
<p>function that is applied to the <code>f</code> values
in each (x,y) bin.  This must take two numeric vectors
as input, and return a single numeric value.</p>
</td></tr>
<tr><td><code id="binApply2D_+3A_include.lowest">include.lowest</code></td>
<td>
<p>logical value indicating whether to include
<code>x</code> values that equal <code>xbreaks[1]</code> and <code>y</code> values that equal
<code>ybreaks[1]</code>.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="binApply2D_+3A_...">...</code></td>
<td>
<p>optional arguments to pass to <code>FUN</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The division into bins is done with <code><a href="base.html#topic+cut">cut()</a></code>, to which <code>include.lowest</code>
is passed. By default, the <code>x</code> bins are open at the left and closed
on the right, and the <code>y</code> bins are open at the bottom and closed
at the top.  However, if <code>include.lowest</code> is TRUE, then those
boundary points are included in the calculation.
</p>


<h3>Value</h3>

<p>A list with the following elements:
<code>xbreaks</code> and <code>ybreaks</code> as used, mid-points
<code>xmids</code> and <code>ymids</code>, and <code>result</code>, a matrix containing the
result of applying <code>FUN()</code> to the <code>f</code> values
in the designated bins.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other bin-related functions: 
<code><a href="#topic+binApply1D">binApply1D</a>()</code>,
<code><a href="#topic+binAverage">binAverage</a>()</code>,
<code><a href="#topic+binCount1D">binCount1D</a>()</code>,
<code><a href="#topic+binCount2D">binCount2D</a>()</code>,
<code><a href="#topic+binMean1D">binMean1D</a>()</code>,
<code><a href="#topic+binMean2D">binMean2D</a>()</code>
</p>

<hr>
<h2 id='binAverage'>Bin-average a Vector y, Based on x Values</h2><span id='topic+binAverage'></span>

<h3>Description</h3>

<p><code><a href="#topic+binAverage">binAverage()</a></code> works by calling <code><a href="#topic+binMean1D">binMean1D()</a></code>, after computing
the <code>xbreaks</code> parameter of the latter function as <code>seq(xmin,xmax,xinc)</code>.
Note that the return value of <code><a href="#topic+binAverage">binAverage()</a></code> uses only the <code>xmids</code> and
<code>result</code> entries of the <code><a href="#topic+binMean1D">binMean1D()</a></code> result.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binAverage(x, y, xmin, xmax, xinc, include.lowest = FALSE, na.rm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binAverage_+3A_x">x</code></td>
<td>
<p>a vector of numerical values.</p>
</td></tr>
<tr><td><code id="binAverage_+3A_y">y</code></td>
<td>
<p>a vector of numerical values.</p>
</td></tr>
<tr><td><code id="binAverage_+3A_xmin">xmin</code></td>
<td>
<p>x value at the lower limit of first bin; the minimum <code>x</code>
will be used if this is not provided.</p>
</td></tr>
<tr><td><code id="binAverage_+3A_xmax">xmax</code></td>
<td>
<p>x value at the upper limit of last bin; the maximum <code>x</code>
will be used if this is not provided.</p>
</td></tr>
<tr><td><code id="binAverage_+3A_xinc">xinc</code></td>
<td>
<p>width of bins, in terms of x value; 1/10th of <code>xmax-xmin</code>
will be used if this is not provided.</p>
</td></tr>
<tr><td><code id="binAverage_+3A_include.lowest">include.lowest</code></td>
<td>
<p>logical value indicating whether to include
<code>y</code> values for which the corresponding <code>x</code> is equal to <code>xmin</code>.
See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="binAverage_+3A_na.rm">na.rm</code></td>
<td>
<p>logical value indicating whether to remove NA values before
doing the computation of the average. This is passed to <code><a href="base.html#topic+mean">mean()</a></code>, which
does the work of the present function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, the sub-intervals defined by <code>xmin</code>, <code>xinc</code> and <code>xmax</code>
arguments are open on the left and closed on the right, to match
the behaviour of <code><a href="base.html#topic+cut">cut()</a></code>.  An open interval does not include
points on the boundary, and so any <code>x</code> values that exactly match
the first <code>breaks</code> value will not be counted.  To include
such points in the calculation, set <code>include.lowest</code> to TRUE.
</p>


<h3>Value</h3>

<p>A list with two elements: <code>x</code>, the mid-points of the bins, and
<code>y</code>, the average <code>y</code> value in the bins.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other bin-related functions: 
<code><a href="#topic+binApply1D">binApply1D</a>()</code>,
<code><a href="#topic+binApply2D">binApply2D</a>()</code>,
<code><a href="#topic+binCount1D">binCount1D</a>()</code>,
<code><a href="#topic+binCount2D">binCount2D</a>()</code>,
<code><a href="#topic+binMean1D">binMean1D</a>()</code>,
<code><a href="#topic+binMean2D">binMean2D</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# A. fake linear data
x &lt;- seq(0, 100, 1)
y &lt;- 1 + 2 * x
plot(x, y, pch = 1)
ba &lt;- binAverage(x, y)
points(ba$x, ba$y, pch = 3, col = "red", cex = 3)

# B. fake quadratic data
y &lt;- 1 + x^2
plot(x, y, pch = 1)
ba &lt;- binAverage(x, y)
points(ba$x, ba$y, pch = 3, col = "red", cex = 3)

# C. natural data
data(co2)
plot(co2)
avg &lt;- binAverage(time(co2), co2, 1950, 2000, 2)
points(avg$x, avg$y, col = "red")

</code></pre>

<hr>
<h2 id='binCount1D'>Bin-count Vector Data</h2><span id='topic+binCount1D'></span>

<h3>Description</h3>

<p>Count the number of elements of a given vector that fall within
successive pairs of values within a second vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binCount1D(x, xbreaks, include.lowest = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binCount1D_+3A_x">x</code></td>
<td>
<p>vector of numerical values.</p>
</td></tr>
<tr><td><code id="binCount1D_+3A_xbreaks">xbreaks</code></td>
<td>
<p>Vector of values of x at the boundaries between bins, calculated using
<code><a href="base.html#topic+pretty">pretty()</a></code> if not supplied.</p>
</td></tr>
<tr><td><code id="binCount1D_+3A_include.lowest">include.lowest</code></td>
<td>
<p>logical value indicating whether to include
<code>x</code> values that equal <code>xbreaks[1]</code>.  See &ldquo;Details&rdquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, the sub-intervals defined by the <code>xbreaks</code> argument are open
on the left and closed on the right, to match the behaviour
of <code><a href="base.html#topic+cut">cut()</a></code>.  An open interval does not include points on
the boundary, and so any <code>x</code> values that exactly match
the first <code>breaks</code> value will not be counted.  To count such
points, set <code>include.lowest</code> to TRUE.
</p>
<p>To contextualize <code>binCount1D()</code> in terms of base R functions,
note that
</p>
<div class="sourceCode"><pre>binCount1D(1:20, seq(0, 20, 2))$number
</pre></div>
<p>matches
</p>
<div class="sourceCode"><pre>unname(table(cut(1:20, seq(0, 20, 2))))
</pre></div>


<h3>Value</h3>

<p>A list with the following elements: the breaks (<code>xbreaks</code>,
midpoints (<code>xmids</code>) between those breaks, and
the count (<code>number</code>) of <code>x</code> values between successive breaks.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other bin-related functions: 
<code><a href="#topic+binApply1D">binApply1D</a>()</code>,
<code><a href="#topic+binApply2D">binApply2D</a>()</code>,
<code><a href="#topic+binAverage">binAverage</a>()</code>,
<code><a href="#topic+binCount2D">binCount2D</a>()</code>,
<code><a href="#topic+binMean1D">binMean1D</a>()</code>,
<code><a href="#topic+binMean2D">binMean2D</a>()</code>
</p>

<hr>
<h2 id='binCount2D'>Bin-count Matrix Data</h2><span id='topic+binCount2D'></span>

<h3>Description</h3>

<p>Count the number of elements of a given matrix z=z(x,y) that fall within
successive pairs of breaks in x and y.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binCount2D(x, y, xbreaks, ybreaks, flatten = FALSE, include.lowest = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binCount2D_+3A_x">x</code>, <code id="binCount2D_+3A_y">y</code></td>
<td>
<p>vectors of numerical values.</p>
</td></tr>
<tr><td><code id="binCount2D_+3A_xbreaks">xbreaks</code>, <code id="binCount2D_+3A_ybreaks">ybreaks</code></td>
<td>
<p>vector of values of <code>x</code> and <code>y</code>
at the boundaries between the 2D bins, calculated using
<code><a href="base.html#topic+pretty">pretty()</a></code> on each of <code>x</code> and <code>y</code>, if not supplied.</p>
</td></tr>
<tr><td><code id="binCount2D_+3A_flatten">flatten</code></td>
<td>
<p>A logical value indicating whether
the return value also contains equilength
vectors <code>x</code>, <code>y</code>, <code>z</code> and <code>n</code>, a flattened
representation of <code>xmids</code>, <code>ymids</code>, <code>result</code> and
<code>number</code>.</p>
</td></tr>
<tr><td><code id="binCount2D_+3A_include.lowest">include.lowest</code></td>
<td>
<p>logical value indicating whether to include
points where <code>x</code> equals <code>xbreaks[1]</code> or <code>y</code> equals <code>ybreaks[1]</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, the sub-intervals defined by <code>xbreaks</code> and <code>ybreaks</code>
are open on the left/bottom and closed on the right/top, to match
the behaviour of <code><a href="base.html#topic+cut">cut()</a></code>.  An open interval does not include
points on the boundary, and so any <code>x</code> and <code>y</code> values that equal
<code>xbreaks[1]</code> or <code>ybreaks[1]</code> will not be counted.  To include
such points in the calculation, set <code>include.lowest</code> to TRUE.
</p>


<h3>Value</h3>

<p>A list with the following elements: the breaks (<code>xbreaks</code>
and <code>ybreaks</code>), the midpoints (<code>xmids</code> and <code>ymids</code>)
between those breaks, and
the count (<code>number</code>) of <code>f</code> values in the boxes defined
between successive breaks.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other bin-related functions: 
<code><a href="#topic+binApply1D">binApply1D</a>()</code>,
<code><a href="#topic+binApply2D">binApply2D</a>()</code>,
<code><a href="#topic+binAverage">binAverage</a>()</code>,
<code><a href="#topic+binCount1D">binCount1D</a>()</code>,
<code><a href="#topic+binMean1D">binMean1D</a>()</code>,
<code><a href="#topic+binMean2D">binMean2D</a>()</code>
</p>

<hr>
<h2 id='binmapAdp'>Bin-map an adp Object</h2><span id='topic+binmapAdp'></span>

<h3>Description</h3>

<p>Bin-map an ADP object, by interpolating velocities, backscatter amplitudes,
etc., to uniform depth bins, thus compensating for the pitch and roll of the
instrument.  This only makes sense for ADP objects that are in beam
coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binmapAdp(x, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binmapAdp_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="binmapAdp_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
</p>


<h3>Bugs</h3>

<p>This only works for 4-beam RDI ADP objects.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
file &lt;- "~/data/archive/sleiwex/2008/moorings/m09/adp/rdi_2615/raw/adp_rdi_2615.000"
beam &lt;- read.oce(file,
    from=as.POSIXct("2008-06-26", tz="UTC"),
    to=as.POSIXct("2008-06-26 00:10:00", tz="UTC"),
    longitude=-69.73433, latitude=47.88126)
beam2 &lt;- binmapAdp(beam)
plot(enuToOther(toEnu(beam), heading=-31.5))
plot(enuToOther(toEnu(beam2), heading=-31.5))
plot(beam, which=5:8) # backscatter amplitude
plot(beam2, which=5:8)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>

<p>The method was devised by Clark Richards for use in his PhD work
at Department of Oceanography at Dalhousie University.
</p>


<h3>See Also</h3>

<p>See <a href="#topic+adp-class">adp</a> for a discussion of <code>adp</code> objects
and notes on the many functions dealing with them.
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='binMean1D'>Bin-average f=f(x)</h2><span id='topic+binMean1D'></span>

<h3>Description</h3>

<p>Average the values of a vector <code>f</code> in bins defined on another
vector <code>x</code>. The values are broken up into bins using <code><a href="base.html#topic+cut">cut()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binMean1D(x, f, xbreaks, include.lowest = FALSE, na.rm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binMean1D_+3A_x">x</code></td>
<td>
<p>vector of numerical values that will be categorized into
bins via the <code>xbreaks</code> parameter.</p>
</td></tr>
<tr><td><code id="binMean1D_+3A_f">f</code></td>
<td>
<p>vector of numerical values that are associated with the <code>x</code> values.</p>
</td></tr>
<tr><td><code id="binMean1D_+3A_xbreaks">xbreaks</code></td>
<td>
<p>vector of values of <code>x</code> at the boundaries between bins, calculated using
<code><a href="base.html#topic+pretty">pretty()</a></code> if not supplied.</p>
</td></tr>
<tr><td><code id="binMean1D_+3A_include.lowest">include.lowest</code></td>
<td>
<p>logical value indicating whether to include
<code>x</code> values that equal <code>xbreaks[1]</code>.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="binMean1D_+3A_na.rm">na.rm</code></td>
<td>
<p>logical value indicating whether to remove NA values before
doing the computation of the average. This is passed to <code><a href="base.html#topic+mean">mean()</a></code>, which
does the work of the present function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, the sub-intervals defined by the <code>xbreaks</code> argument are open
on the left and closed on the right, to match the behaviour
of <code><a href="base.html#topic+cut">cut()</a></code>.  An open interval does not include points on
the boundary, and so any <code>x</code> values that exactly match
the first <code>breaks</code> value will not be counted.  To include
such points in the calculation, set <code>include.lowest</code> to TRUE.
</p>


<h3>Value</h3>

<p>A list with the following elements: the breaks (<code>xbreaks</code>,
midpoints (<code>xmids</code>) between those breaks,
the count (<code>number</code>) of <code>x</code> values between successive breaks,
and the resultant average (<code>result</code>) of <code>f</code>, classified by the
<code>x</code> breaks.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other bin-related functions: 
<code><a href="#topic+binApply1D">binApply1D</a>()</code>,
<code><a href="#topic+binApply2D">binApply2D</a>()</code>,
<code><a href="#topic+binAverage">binAverage</a>()</code>,
<code><a href="#topic+binCount1D">binCount1D</a>()</code>,
<code><a href="#topic+binCount2D">binCount2D</a>()</code>,
<code><a href="#topic+binMean2D">binMean2D</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Plot raw temperature profile as circles, with lines indicating
# the result of averaging in 1-metre depth intervals.
library(oce)
data(ctd)
z &lt;- ctd[["z"]]
T &lt;- ctd[["temperature"]]
plot(T, z, cex = 0.3)
TT &lt;- binMean1D(z, T, seq(-100, 0, 1))
lines(TT$result, TT$xmids, col = rgb(1, 0, 0, 0.9), lwd = 2)

</code></pre>

<hr>
<h2 id='binMean2D'>Bin-average f=f(x,y)</h2><span id='topic+binMean2D'></span>

<h3>Description</h3>

<p>Average the values of a vector <code>f(x,y)</code> in bins defined on
vectors <code>x</code> and <code>y</code>. A common example might be averaging
spatial data into location bins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binMean2D(
  x,
  y,
  f,
  xbreaks,
  ybreaks,
  flatten = FALSE,
  fill = FALSE,
  fillgap = -1,
  include.lowest = FALSE,
  na.rm = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="binMean2D_+3A_x">x</code></td>
<td>
<p>vector of numerical values.</p>
</td></tr>
<tr><td><code id="binMean2D_+3A_y">y</code></td>
<td>
<p>vector of numerical values.</p>
</td></tr>
<tr><td><code id="binMean2D_+3A_f">f</code></td>
<td>
<p>Matrix of numerical values, a matrix f=f(x,y).</p>
</td></tr>
<tr><td><code id="binMean2D_+3A_xbreaks">xbreaks</code></td>
<td>
<p>Vector of values of <code>x</code> at the boundaries between bins, calculated using
<code><a href="base.html#topic+pretty">pretty</a></code><code>(x)</code> if not supplied.</p>
</td></tr>
<tr><td><code id="binMean2D_+3A_ybreaks">ybreaks</code></td>
<td>
<p>Vector of values of <code>y</code> at the boundaries between bins, calculated using
<code><a href="base.html#topic+pretty">pretty</a></code><code>(y)</code> if not supplied.</p>
</td></tr>
<tr><td><code id="binMean2D_+3A_flatten">flatten</code></td>
<td>
<p>A logical value indicating whether
the return value also contains equilength
vectors <code>x</code>, <code>y</code>, <code>z</code> and <code>n</code>, a flattened
representation of <code>xmids</code>, <code>ymids</code>, <code>result</code> and
<code>number</code>.</p>
</td></tr>
<tr><td><code id="binMean2D_+3A_fill">fill</code></td>
<td>
<p>Logical value indicating whether to fill <code>NA</code>-value gaps in
the matrix. Gaps will be filled as the average of linear interpolations
across rows and columns. See <code>fillgap</code>, which works together with this.</p>
</td></tr>
<tr><td><code id="binMean2D_+3A_fillgap">fillgap</code></td>
<td>
<p>Integer controlling the size of gap that can be filled
across. If this is negative (as in the default), gaps will be filled
regardless of their size. If it is positive, then gaps exceeding this
number of indices will not be filled.</p>
</td></tr>
<tr><td><code id="binMean2D_+3A_include.lowest">include.lowest</code></td>
<td>
<p>logical value indicating whether to include
<code>y</code> values for which the corresponding <code>x</code> is equal to <code>xmin</code>.
See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="binMean2D_+3A_na.rm">na.rm</code></td>
<td>
<p>logical value indicating whether to remove NA values before
doing the computation of the average. This is passed to <code><a href="base.html#topic+mean">mean()</a></code>, which
does the work of the present function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>By default, i.e. with <code>flatten</code> being FALSE, <code><a href="#topic+binMean2D">binMean2D()</a></code> returns a
list with the following elements: <code>xmids</code>, a vector holding the x-bin midpoints;
<code>ymids</code>, a vector holding the y-bin midpoints; <code>number</code>, a matrix holding the
number the points in each bin; and <code>result</code>, a matrix holding the mean
value in each bin. If <code>flatten</code> is TRUE, the <code>number</code> and <code>result</code>
matrices are renamed as <code>n</code> and <code>f</code> and transformed to vectors, while
the bin midpoints are renamed as <code>x</code> and <code>y</code> and extended to match the length
of <code>n</code> and <code>f</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other bin-related functions: 
<code><a href="#topic+binApply1D">binApply1D</a>()</code>,
<code><a href="#topic+binApply2D">binApply2D</a>()</code>,
<code><a href="#topic+binAverage">binAverage</a>()</code>,
<code><a href="#topic+binCount1D">binCount1D</a>()</code>,
<code><a href="#topic+binCount2D">binCount2D</a>()</code>,
<code><a href="#topic+binMean1D">binMean1D</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
x &lt;- runif(500, 0, 0.5)
y &lt;- runif(500, 0, 0.5)
f &lt;- x^2 + y^2
xb &lt;- seq(0, 0.5, 0.1)
yb &lt;- seq(0, 0.5, 0.1)
m &lt;- binMean2D(x, y, f, xb, yb)
cm &lt;- colormap(f, col = oceColorsTurbo)
opar &lt;- par(no.readonly = TRUE)
drawPalette(colormap = cm)
plot(x, y, col = cm$zcol, pch = 20, cex = 1.4)
contour(m$xmids, m$ymids, m$result, add = TRUE, labcex = 1.4)
par(opar)

</code></pre>

<hr>
<h2 id='bound125'>Calculate a Bound, Rounded up to Mantissa 1, 2, or 5</h2><span id='topic+bound125'></span>

<h3>Description</h3>

<p>Calculate a Bound, Rounded up to Mantissa 1, 2, or 5
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bound125(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bound125_+3A_x">x</code></td>
<td>
<p>a single positive number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>for positive x, a value exceeding x that has mantissa 1, 2, or 5; otherwise, x
</p>

<hr>
<h2 id='bremen-class'>Class to Store Bremen-formatted Data</h2><span id='topic+bremen-class'></span>

<h3>Description</h3>

<p>This class is for data stored in a format used at Bremen.
It is somewhat similar to the <a href="#topic+odf-class">odf</a>, in the sense
that it does not apply just to a particular instrument.
Although some functions are provided for dealing with these
data (see &ldquo;Details&rdquo;), the most common action is to read the
data with <code><a href="#topic+read.bremen">read.bremen()</a></code>, and then to coerce the object to
another storage class (e.g. using <code><a href="#topic+as.ctd">as.ctd()</a></code> for CTD-style
data) so that specialized functions can be used thereafter.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>bremen</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>bremen</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>bremen</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+bremen-class">bremen</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+bremen-class">bremen</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to bremen data: 
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+read.bremen">read.bremen</a>()</code>,
<code><a href="#topic+summary+2Cbremen-method">summary,bremen-method</a></code>
</p>

<hr>
<h2 id='byteToBinary'>Format Bytes as Binary (Defunct)</h2><span id='topic+byteToBinary'></span>

<h3>Description</h3>

<p><strong>WARNING:</strong> The <code>endian</code> argument will soon be removed
from this function; see <a href="#topic+oce-defunct">oce-defunct</a>.
This is because the actions for <code>endian="little"</code> made
no sense in practical work. The default value for <code>endian</code>
was changed to <code>"big"</code> on 2017 May 6.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>byteToBinary(x, endian = "big")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="byteToBinary_+3A_x">x</code></td>
<td>
<p>an integer to be interpreted as a byte.</p>
</td></tr>
<tr><td><code id="byteToBinary_+3A_endian">endian</code></td>
<td>
<p>character string indicating the endian-ness (&quot;big&quot; or
&quot;little&quot;). <strong>WARNING:</strong> This argument will be removed soon.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string representing the bit strings for the elements of
<code>x</code>, in order of significance for the <code>endian="big"</code> case.
(The nibbles, or 4-bit sequences, are interchanged in the now-deprecated
<code>"little"</code> case.)
See &ldquo;Examples&rdquo; for how this relates to the output from
<a href="base.html#topic+rawToBits">rawToBits</a>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Note comparison with rawToBits():
a &lt;- as.raw(0x0a)
byteToBinary(a, "big") # "00001010"
as.integer(rev(rawToBits(a))) # 0 0 0 0 1 0 1 0
</code></pre>

<hr>
<h2 id='cm'>Sample cm Data</h2><span id='topic+cm'></span>

<h3>Description</h3>

<p>The result of using <code><a href="#topic+read.cm">read.cm()</a></code> on a current meter file holding measurements made with an
Interocean S4 device.  See <code><a href="#topic+read.cm">read.cm()</a></code> for some general cautionary notes on reading such
files. Note that the salinities in this sample dataset are known to be incorrect, perhaps
owing to a lack of calibration of an old instrument that had not been used in a long time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(cm)
</code></pre>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to cm data: 
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+as.cm">as.cm</a>()</code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+read.cm">read.cm</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+summary+2Ccm-method">summary,cm-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(cm)
summary(cm)
plot(cm)

</code></pre>

<hr>
<h2 id='cm-class'>Class to Store Current Meter Data</h2><span id='topic+cm-class'></span>

<h3>Description</h3>

<p>This class stores current meter data, e.g. from an Interocean/S4 device
or an Aanderaa/RCM device.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>cm</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
The key items stored in this slot are <code>time</code>, <code>u</code> and <code>v</code>.</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>cm</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>cm</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+cm-class">cm</a> objects (see <code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+cm-class">cm</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to cm data: 
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+as.cm">as.cm</a>()</code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+read.cm">read.cm</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+summary+2Ccm-method">summary,cm-method</a></code>
</p>
<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>

<hr>
<h2 id='cnvName2oceName'>Infer Variable Name, Units and Scale From a Seabird Header</h2><span id='topic+cnvName2oceName'></span>

<h3>Description</h3>

<p>This function is used by <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> to infer data names
and units from the coding used by Teledyne/Seabird (SBE) <code>.cnv</code>
files.  Lacking access to documentation on the SBE format,
the present function is based on inspection of a suite of CNV files
available to the <code>oce</code> developers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cnvName2oceName(h, columns = NULL, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cnvName2oceName_+3A_h">h</code></td>
<td>
<p>The header line.</p>
</td></tr>
<tr><td><code id="cnvName2oceName_+3A_columns">columns</code></td>
<td>
<p>Optional list containing name correspondences, as described for
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code>.</p>
</td></tr>
<tr><td><code id="cnvName2oceName_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A few sample header lines that have been encountered are:
</p>
<div class="sourceCode"><pre># name 4 = t068: temperature, IPTS-68 [deg C]
# name 3 = t090C: Temperature [ITS-90, deg C]
# name 4 = t190C: Temperature, 2 [ITS-90, deg C]
</pre></div>
<p>Examination of several CNV files suggests that it is best to
try to infer the name from the characters between the &quot;<code>=</code>&quot;
and &quot;<code>:</code>&quot; characters, because the material after the colon
seems to vary more between sample files.
</p>
<p>The table given below indicates the translation patterns used. These are
taken from reference 1. The <code>.cnv</code> convention for multiple sensors is to
include optional extra digits in the name, and these are indicated
with <code>~</code> or <code style="white-space: pre;">&#8288;~~&#8288;</code> in the table; their decoding is done with <code><a href="base.html#topic+grep">grep()</a></code>.
</p>
<p>It is important to note that this table is by no means complete, since there
are a great many SBE names listed in their document (reference 1), plus names
not listed there but present in data files
supplied by prominent archiving agencies. If an SBE name is not recognized,
then the oce name is set to that SBE name. This can cause problems in
some other processing steps (e.g. if <code><a href="#topic+swRho">swRho()</a></code> or a similar
function is called with an <code>oce</code> object as first argument), and so
users are well-advised to rename the items as appropriate. The first
step in doing this is to pass the object to <code>summary()</code>, to discover
the SBE names in question. Then consult the SBE documentation to find
an appropriate name for the data, and either manipulate the names in the object
data slot directly or use
<code><a href="#topic+oceRenameData">oceRenameData()</a></code> to rename the elements. Finally, please publish
an 'issue' on the oce Github site <code style="white-space: pre;">&#8288;https://github.com/dankelley/oce/issues&#8288;</code>
so that the developers can add the data type in question. (To save
development time, there is no plan to add all possible data types without
a reasonable and specific expression user interest. Oxygen alone has over forty
variants.)
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Key</strong>       </td><td style="text-align: left;"> <strong>Result</strong>                 </td><td style="text-align: left;"> <strong>Unit;scale</strong>      </td><td style="text-align: left;"> <strong>Notes</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>accM</code>        </td><td style="text-align: left;"> <code>acceleration</code>             </td><td style="text-align: left;"> m/s^2               </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>altM</code>        </td><td style="text-align: left;"> <code>altimeter</code>                </td><td style="text-align: left;"> m                   </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>alt</code>         </td><td style="text-align: left;"> <code>altimeter</code>                </td><td style="text-align: left;"> m                   </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;bat~&#8288;</code>        </td><td style="text-align: left;"> <code>beamAttenuation</code>          </td><td style="text-align: left;"> 1/m                 </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>C2-C1mS/cm</code>  </td><td style="text-align: left;"> <code>conductivityDifference</code>   </td><td style="text-align: left;"> mS/cm               </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>C2-C1S/m</code>    </td><td style="text-align: left;"> <code>conductivityDifference</code>   </td><td style="text-align: left;"> S/m                 </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>C2-C1uS/cm</code>  </td><td style="text-align: left;"> <code>conductivityDifference</code>   </td><td style="text-align: left;"> uS/cm               </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>cond~mS/cm</code>  </td><td style="text-align: left;"> <code>conductivity</code>             </td><td style="text-align: left;"> mS/cm               </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>cond~S/m</code>    </td><td style="text-align: left;"> <code>conductivity</code>             </td><td style="text-align: left;"> S/m                 </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>cond~uS/cm</code>  </td><td style="text-align: left;"> <code>conductivity</code>             </td><td style="text-align: left;"> uS/cm               </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;CStarAt~&#8288;</code>    </td><td style="text-align: left;"> <code>beamAttenuation</code>          </td><td style="text-align: left;"> 1/m                 </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;CStarTr~&#8288;</code>    </td><td style="text-align: left;"> <code>beamTransmission</code>         </td><td style="text-align: left;"> percent             </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>c~mS/cm</code>     </td><td style="text-align: left;"> <code>conductivity</code>             </td><td style="text-align: left;"> mS/cm               </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>c~S/m</code>       </td><td style="text-align: left;"> <code>conductivity</code>             </td><td style="text-align: left;"> S/m                 </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>c~uS/cm</code>     </td><td style="text-align: left;"> <code>conductivity</code>             </td><td style="text-align: left;"> uS/cm               </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;density~~&#8288;</code>   </td><td style="text-align: left;"> <code>density</code>                  </td><td style="text-align: left;"> kg/m^3              </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>depFM</code>       </td><td style="text-align: left;"> <code>depth</code>                    </td><td style="text-align: left;"> m                   </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>depF</code>        </td><td style="text-align: left;"> <code>depth</code>                    </td><td style="text-align: left;"> m                   </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>depSM</code>       </td><td style="text-align: left;"> <code>depth</code>                    </td><td style="text-align: left;"> m                   </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>depS</code>        </td><td style="text-align: left;"> <code>depth</code>                    </td><td style="text-align: left;"> m                   </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>dz/dtM</code>      </td><td style="text-align: left;"> <code>descentRate</code>              </td><td style="text-align: left;"> m/s                 </td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>flCM</code>        </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> ug/l; Chelsea Mini Chl Con</td><td style="text-align: left;"></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;flCUVA~&#8288;</code>     </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> ug/l; Chelsea UV Aquatracka</td><td style="text-align: left;"></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;flC~&#8288;</code>        </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> ug/l; Chelsea Aqua 3</td><td style="text-align: left;">    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;flEC-AFL~&#8288;</code>   </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> mg/m^3; WET Labs ECO-AFL/FLtab</td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;flScufa~&#8288;</code>    </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> -; Turner SCUFA (RFU)</td><td style="text-align: left;"></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>flSPR</code>       </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> -; Seapoint, Rhodamine</td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>flSPuv</code>      </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> -; Seapoint, UV      </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>flSP</code>        </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> -; Seapoint          </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>flS</code>         </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> -; Seatech           </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>flT</code>         </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> -; Turner 10-005 flT </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;f~&#8288;</code>          </td><td style="text-align: left;"> <code>frequency</code>                </td><td style="text-align: left;"> Hz                   </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;f~~&#8288;</code>         </td><td style="text-align: left;"> <code>frequency</code>                </td><td style="text-align: left;"> Hz                   </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>gpa</code>         </td><td style="text-align: left;"> <code>geopotentialAnomaly</code>      </td><td style="text-align: left;"> -; J/kg              </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>latitude</code>    </td><td style="text-align: left;"> <code>latitude</code>                 </td><td style="text-align: left;"> degN                 </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>longitude</code>   </td><td style="text-align: left;"> <code>longitude</code>                </td><td style="text-align: left;"> degE                 </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n2satMg/L</code>   </td><td style="text-align: left;"> <code>nitrogenSaturation</code>       </td><td style="text-align: left;"> mg/l                 </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n2satML/L</code>   </td><td style="text-align: left;"> <code>nitrogenSaturation</code>       </td><td style="text-align: left;"> ml/l                 </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n2satumol/kg</code></td><td style="text-align: left;"> <code>nitrogenSaturation</code>       </td><td style="text-align: left;"> umol/kg              </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nbin</code>        </td><td style="text-align: left;"> <code>nbin</code>                     </td><td style="text-align: left;">                      </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;obsscufa~&#8288;</code>   </td><td style="text-align: left;"> <code>backscatter</code>              </td><td style="text-align: left;"> NTU; Turner SCUFA    </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>opoxMg/L</code>    </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> mg/l; Optode, Aanderaa</td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>opoxML/L</code>    </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> ml/l; Optode, Aanderaa</td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>opoxMm/L</code>    </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> umol/l; Optode, Aanderaa</td><td style="text-align: left;"></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>opoxPS</code>      </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> percent; Optode, Aanderaa</td><td style="text-align: left;"></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>oxsatMg/L</code>   </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> mg/l; Weiss          </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>oxsatML/L</code>   </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> ml/l; Weiss          </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>oxsatMm/Kg</code>  </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> umol/kg; Weiss       </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>oxsolMg/L</code>   </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> mg/l; Garcia-Gordon  </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>oxsolML/L</code>   </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> ml/l; Garcia-Gordon  </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>oxsolMm/Kg</code>  </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> umol/kg; Garcia-Gordon</td><td style="text-align: left;">  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>par/log</code>     </td><td style="text-align: left;"> <code>PAR</code>                      </td><td style="text-align: left;"> log; Satlantic       </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;par~&#8288;</code>        </td><td style="text-align: left;"> <code>PAR</code>                      </td><td style="text-align: left;"> -; Biospherical/Licor</td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ph</code>          </td><td style="text-align: left;"> <code>pH</code>                       </td><td style="text-align: left;"> -                    </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;potemp~68C&#8288;</code>  </td><td style="text-align: left;"> <code>thetaM</code>                   </td><td style="text-align: left;"> degC; IPTS-68        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;potemp~90C&#8288;</code>  </td><td style="text-align: left;"> <code>thetaM</code>                   </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pr50M</code>       </td><td style="text-align: left;"> <code>pressure</code>                 </td><td style="text-align: left;"> dbar; SBE50          </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prDE</code>        </td><td style="text-align: left;"> <code>pressure</code>                 </td><td style="text-align: left;"> psi; digiquartz      </td><td style="text-align: left;"> 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prdE</code>        </td><td style="text-align: left;"> <code>pressure</code>                 </td><td style="text-align: left;"> psi; strain gauge    </td><td style="text-align: left;"> 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prDM</code>        </td><td style="text-align: left;"> <code>pressure</code>                 </td><td style="text-align: left;"> dbar; digiquartz     </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prdM</code>        </td><td style="text-align: left;"> <code>pressure</code>                 </td><td style="text-align: left;"> dbar; strain gauge   </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prM</code>         </td><td style="text-align: left;"> <code>pressure</code>                 </td><td style="text-align: left;"> dbar                 </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prSM</code>        </td><td style="text-align: left;"> <code>pressure</code>                 </td><td style="text-align: left;"> dbar                 </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prSM</code>        </td><td style="text-align: left;"> <code>pressure</code>                 </td><td style="text-align: left;"> dbar; strain gauge   </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pr</code>          </td><td style="text-align: left;"> <code>pressure</code>                 </td><td style="text-align: left;"> dbar                 </td><td style="text-align: left;"> 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ptempC</code>      </td><td style="text-align: left;"> <code>pressureTemperature</code>      </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;"> 3 </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pumps</code>       </td><td style="text-align: left;"> <code>pumpStatus</code>               </td><td style="text-align: left;">                      </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;rhodflTC~&#8288;</code>   </td><td style="text-align: left;"> <code>Rhodamine</code>                </td><td style="text-align: left;"> ppb; Turner Cyclops  </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;sal~~&#8288;</code>       </td><td style="text-align: left;"> <code>salinity</code>                 </td><td style="text-align: left;"> -, PSS-78            </td><td style="text-align: left;"> 4 </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sbeox~ML/L</code>  </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> ml/l; SBE43          </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sbeox~Mm/Kg</code> </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> umol/kg; SBE43       </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sbeox~Mm/L</code>  </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> umol/l; SBE43        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sbeox~PS</code>    </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> percent; SBE43       </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sbeox~V</code>     </td><td style="text-align: left;"> <code>oxygenRaw</code>                </td><td style="text-align: left;"> V; SBE43             </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sbox~dV/dT</code>  </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> dov/dt; SBE43        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sbox~ML/L</code>   </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> ml/l; SBE43          </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sbox~Mm/Kg</code>  </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> umol/kg; SBE43       </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sbox~Mm/L</code>   </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> umol/l; SBE43        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sbox~PS</code>     </td><td style="text-align: left;"> <code>oxygen</code>                   </td><td style="text-align: left;"> percent; SBE43       </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sbox~V</code>      </td><td style="text-align: left;"> <code>oxygenRaw</code>                </td><td style="text-align: left;"> V; SBE43             </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>scan</code>        </td><td style="text-align: left;"> <code>scan</code>                     </td><td style="text-align: left;"> -                    </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;seaTurbMtr~&#8288;</code> </td><td style="text-align: left;"> <code>turbidity</code>                </td><td style="text-align: left;"> FTU; Seapoint        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>secS-priS</code>   </td><td style="text-align: left;"> <code>salinityDifference</code>       </td><td style="text-align: left;"> -, PSS-78            </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sigma-</code>é </td><td style="text-align: left;"> <code>sigmaTheta</code>     </td><td style="text-align: left;"> kg/m^3               </td><td style="text-align: left;"> 5 </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sigma-t</code>     </td><td style="text-align: left;"> <code>sigmaT</code>                   </td><td style="text-align: left;"> kg/m^3               </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sigma-theta</code> </td><td style="text-align: left;"> <code>sigmaTheta</code>               </td><td style="text-align: left;"> kg/m^3               </td><td style="text-align: left;"> 5 </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>spar</code>        </td><td style="text-align: left;"> <code>spar</code>                     </td><td style="text-align: left;"> -                    </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>specc</code>       </td><td style="text-align: left;"> <code>conductivity</code>             </td><td style="text-align: left;"> uS/cm                </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sva</code>         </td><td style="text-align: left;"> <code>specificVolumeAnomaly</code>    </td><td style="text-align: left;"> 1e-8 m^3/kg;         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;svCM~&#8288;</code>       </td><td style="text-align: left;"> <code>soundSpeed</code>               </td><td style="text-align: left;"> m/s; Chen-Millero    </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>t090Cm</code>      </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>t190C</code>       </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;T2~68C&#8288;</code>      </td><td style="text-align: left;"> <code>temperatureDifference</code>    </td><td style="text-align: left;"> degC; IPTS-68        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;T2~90C&#8288;</code>      </td><td style="text-align: left;"> <code>temperatureDifference</code>    </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;t3868C~&#8288;</code>     </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; IPTS-68        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;t3890C~&#8288;</code>     </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;t38~38C&#8288;</code>     </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; IPTS-68        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;t38~90C&#8288;</code>     </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>t4968C</code>      </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; IPTS-68        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>t4990C</code>      </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>timeH</code>       </td><td style="text-align: left;"> <code>timeH</code>                    </td><td style="text-align: left;"> hour; elapsed        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>timeJV2</code>     </td><td style="text-align: left;"> <code>timeJV2</code>                  </td><td style="text-align: left;"> julian day           </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>timeJ</code>       </td><td style="text-align: left;"> <code>timeJ</code>                    </td><td style="text-align: left;"> julian day           </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>timeK</code>       </td><td style="text-align: left;"> <code>timeK</code>                    </td><td style="text-align: left;"> s; since Jan 1, 2000 </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>timeM</code>       </td><td style="text-align: left;"> <code>timeM</code>                    </td><td style="text-align: left;"> minute; elapsed      </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>timeN</code>       </td><td style="text-align: left;"> <code>timeN</code>                    </td><td style="text-align: left;"> s; NMEA since Jan 1, 1970</td><td style="text-align: left;"></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>timeQ</code>       </td><td style="text-align: left;"> <code>timeQ</code>                    </td><td style="text-align: left;"> s; NMEA since Jan 1, 2000</td><td style="text-align: left;"></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>timeS</code>       </td><td style="text-align: left;"> <code>timeS</code>                    </td><td style="text-align: left;"> s; elapsed           </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tnc268C</code>     </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; IPTS-68        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tnc290C</code>     </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tnc68C</code>      </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; IPTS-68        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tnc90C</code>      </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tsa</code>         </td><td style="text-align: left;"> <code>thermostericAnomaly</code>      </td><td style="text-align: left;"> 1e-8 m^3/kg          </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>turbflTCdiff</code></td><td style="text-align: left;"> <code>turbidityDifference</code>      </td><td style="text-align: left;"> NTU; Turner Cyclops  </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;turbflTC~&#8288;</code>   </td><td style="text-align: left;"> <code>turbidity</code>                </td><td style="text-align: left;"> NTU; Turner Cyclops  </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>turbWETbbdiff</code></td><td style="text-align: left;"> <code>turbidityDifference</code>     </td><td style="text-align: left;"> 1/(m\*sr); WET Labs ECO</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;turbWETbb~&#8288;</code>  </td><td style="text-align: left;"> <code>turbidity</code>                </td><td style="text-align: left;"> 1/(m\*sr); WET Labs ECO</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>turbWETntudiff</code></td><td style="text-align: left;"> <code>turbidityDifference</code>    </td><td style="text-align: left;"> NTU; WET Labs ECO    </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;turbWETntu~&#8288;</code> </td><td style="text-align: left;"> <code>turbidity</code>                </td><td style="text-align: left;"> NTU; WET Labs ECO    </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tv268C</code>      </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; IPTS-68        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tv290C</code>      </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;t~68C&#8288;</code>       </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; IPTS-68        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>t~68</code>        </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; IPTS-68        </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;t~90C&#8288;</code>       </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>t~90</code>        </td><td style="text-align: left;"> <code>temperature</code>              </td><td style="text-align: left;"> degC; ITS-90         </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;upoly~&#8288;</code>      </td><td style="text-align: left;"> <code>upoly</code>                    </td><td style="text-align: left;"> -                    </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;user~&#8288;</code>       </td><td style="text-align: left;"> <code>user</code>                     </td><td style="text-align: left;"> -                    </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;v~~&#8288;</code>         </td><td style="text-align: left;"> <code>voltage</code>                  </td><td style="text-align: left;"> V                    </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>wetBAttn</code>    </td><td style="text-align: left;"> <code>beamAttenuation</code>          </td><td style="text-align: left;"> 1/m; WET Labs AC3    </td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>wetBTrans</code>   </td><td style="text-align: left;"> <code>beamTransmission</code>         </td><td style="text-align: left;"> percent; WET Labs AC3</td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>wetCDOMdiff</code> </td><td style="text-align: left;"> <code>fluorescenceDifference</code>   </td><td style="text-align: left;"> mg/m^3; WET Labs CDOM</td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;wetCDOM~&#8288;</code>    </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> mg/m^3; WET Labs CDOM</td><td style="text-align: left;">   </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>wetChAbs</code>    </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> 1/m; WET Labs AC3 absorption</td><td style="text-align: left;"></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>wetStardiff</code> </td><td style="text-align: left;"> <code>fluorescenceDifference</code>   </td><td style="text-align: left;"> mg/m^3; WET Labs WETstar</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;wetStar~&#8288;</code>    </td><td style="text-align: left;"> <code>fluorescence</code>             </td><td style="text-align: left;"> mg/m^3; WET Labs WETstar</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>xmiss</code>       </td><td style="text-align: left;"> <code>beamTransmission</code>         </td><td style="text-align: left;"> percent; Chelsea/Seatech</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;xmiss~&#8288;</code>      </td><td style="text-align: left;"> <code>beamTransmission</code>         </td><td style="text-align: left;"> percent; Chelsea/Seatech</td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Notes:
</p>

<ol>
<li><p> 'pr' is in a Dalhousie-generated data file but seems not to be in reference 1.
</p>
</li>
<li><p> This is an odd unit, and so if <code style="white-space: pre;">&#8288;sw*&#8288;</code> functions are called on an object
containing this, a conversion will be made before performing the computation. Be
on the lookout for errors, since this is a rare situation.
</p>
</li>
<li><p> Assume ITS-90 temperature scale, since sample <code>.cnv</code> file headers do not specify it.
</p>
</li>
<li><p> Some files have PSU for this. Should we handle that? And are there other S scales to consider?
</p>
</li>
<li><p> The 'theta' symbol (here shown accented e) may appear in different ways with different
encoding configurations, set up within R or in the operating system.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> A SBE data processing manual was once at
<code style="white-space: pre;">&#8288;http://www.seabird.com/document/sbe-data-processing-manual&#8288;</code>,
but as of summer 2018, this no longer seems to be provided by SeaBird.
A web search will turn up copies of the manual that have been put
online by various research groups and data-archiving agencies.
As of 2018-07-05, the latest version was named
<code>SBEDataProcessing_7.26.4.pdf</code> and had release date 12/08/2017,
and this was the reference version used in coding <code>oce</code>.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that interpret variable names and units from headers: 
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+unitFromStringRsk">unitFromStringRsk</a>()</code>,
<code><a href="#topic+unitFromString">unitFromString</a>()</code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>
</p>

<hr>
<h2 id='coastline-class'>Class to Store Coastline Data</h2><span id='topic+coastline-class'></span>

<h3>Description</h3>

<p>This class stores coastline data.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>coastline</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
The key items stored in this slot are <code>longitude</code> and <code>latitude</code>.</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>coastline</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>coastline</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+coastline-class">coastline</a> objects (see <code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+coastline-class">coastline</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>

<hr>
<h2 id='coastlineBest'>Find the Name of the Best Coastline Object</h2><span id='topic+coastlineBest'></span>

<h3>Description</h3>

<p>Find the name of the most appropriate coastline for a given locale
Checks <code>coastlineWorld</code>, <code>coastlineWorldFine</code> and
<code>coastlineWorldCoarse</code>, in that order, to find the one most appropriate
for the locale.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coastlineBest(lonRange, latRange, span, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coastlineBest_+3A_lonrange">lonRange</code></td>
<td>
<p>range of longitude for locale</p>
</td></tr>
<tr><td><code id="coastlineBest_+3A_latrange">latRange</code></td>
<td>
<p>range of latitude for locale</p>
</td></tr>
<tr><td><code id="coastlineBest_+3A_span">span</code></td>
<td>
<p>span of domain in km (if provided, previous two arguments are
ignored).</p>
</td></tr>
<tr><td><code id="coastlineBest_+3A_debug">debug</code></td>
<td>
<p>set to a positive value to get debugging information during
processing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The name of a coastline that can be loaded with <code>data()</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>

<hr>
<h2 id='coastlineCut'>Cut a Coastline Object at Specified Longitude</h2><span id='topic+coastlineCut'></span>

<h3>Description</h3>

<p>This can be helpful in preventing <code><a href="#topic+mapPlot">mapPlot()</a></code> from producing ugly
horizontal lines in world maps. These lines occur when a coastline segment
is intersected by longitude lon_0+180.  Since the coastline files in the oce
and ocedata packages are already &quot;cut&quot; at longitudes of -180 and 180, the present
function is not needed for default maps, which have <code>+lon_0=0</code>. However,
may help with other values of <code>lon_0</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coastlineCut(coastline, lon_0 = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coastlineCut_+3A_coastline">coastline</code></td>
<td>
<p>a <a href="#topic+coastline-class">coastline</a> object.</p>
</td></tr>
<tr><td><code id="coastlineCut_+3A_lon_0">lon_0</code></td>
<td>
<p>longitude as would be given in a <code style="white-space: pre;">&#8288;+lon_0=&#8288;</code> item in a
call to <code><a href="sf.html#topic+sf_project">sf::sf_project()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a new coastline object
</p>


<h3>Caution</h3>

<p>This function is provisional. Its behaviour, name and very existence
may change.  Part of the development plan is to see if there is common
ground between this and the <code>clipPolys</code> function in the
<a href="https://CRAN.R-project.org/package=PBSmapping"><span class="pkg">PBSmapping</span></a> package.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(coastlineWorld)
mapPlot(coastlineCut(coastlineWorld, lon_0 = 100),
    projection = "+proj=moll +lon_0=100", col = "gray"
)


</code></pre>

<hr>
<h2 id='coastlineWorld'>Sample coastline Data (Global, at 1:110M scale)</h2><span id='topic+coastlineWorld'></span>

<h3>Description</h3>

<p>This is a coarse resolution coastline at scale 1:110M, with 10,696 points,
suitable for world-scale plots plotted at a small size, e.g. inset diagrams.
Finer resolution coastline files are provided in the
<a href="https://CRAN.R-project.org/package=ocedata"><span class="pkg">ocedata</span></a> package.
</p>


<h3>Installing your own datasets</h3>

<p>Follow the procedure along the lines
described in &ldquo;Details&rdquo;, where of course your source file will differ.
Also, you should change the name of the coastline object from
<code>coastlineWorld</code>, to avoid conflicts with the built-in dataset. Save
the <code>.rda</code> file to some directory of your choosing, e.g. perhaps
<code style="white-space: pre;">&#8288;/data/coastlines&#8288;</code> or <code style="white-space: pre;">&#8288;~/data/coastlines&#8288;</code> on a unix-type machine.
Then, whenever you need the file, use <code><a href="base.html#topic+load">load()</a></code> to load it.  Most
users find it convenient to do the loading in an <code><a href="base.html#topic+Rprofile">Rprofile()</a></code>
startup file.
</p>


<h3>Source</h3>

<p>Downloaded from <code style="white-space: pre;">&#8288;https://www.naturalearthdata.com&#8288;</code>, in
<code>ne_110m_admin_0_countries.shp</code> in July 2015, with an
update on December 16, 2017.
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>

<hr>
<h2 id='colormap'>Calculate a Color Map</h2><span id='topic+colormap'></span>

<h3>Description</h3>

<p>Create a mapping between numeric values and colors, for use in palettes and plots.
The return value can be used in various ways, including colorizing points
on scattergraphs, controlling images created by <code><a href="Matrix.html#topic+image">image()</a></code> or <code><a href="#topic+imagep">imagep()</a></code>,
drawing palettes with <code><a href="#topic+drawPalette">drawPalette()</a></code>, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colormap(
  z = NULL,
  zlim,
  zclip = FALSE,
  breaks,
  col = oceColorsViridis,
  name,
  x0,
  x1,
  col0,
  col1,
  blend = 0,
  missingColor,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colormap_+3A_z">z</code></td>
<td>
<p>an optional vector or other set of numerical values to be examined.
If <code>z</code> is given, the return value will contain an item named
<code>zcol</code> that will be a vector of the same length as <code>z</code>, containing
a color for each point.  If <code>z</code> is not given, <code>zcol</code> will contain
just one item, the color <code>"black"</code>.</p>
</td></tr>
<tr><td><code id="colormap_+3A_zlim">zlim</code></td>
<td>
<p>optional vector containing two numbers that specify the <code>z</code>
limits for the color scale. This can only be provided in cases A
and B, as defined in &ldquo;Details&rdquo;.  For case A, if <code>zlim</code> is not
provided, then it is inferred by using <code><a href="#topic+rangeExtended">rangeExtended()</a></code>
on <code>breaks</code>, if that is provided, or from <code>z</code> otherwise.  Also,
in case A, it is an error to provide both <code>zlim</code> and <code>breaks</code>,
unless the latter is of length 1, meaning the number of
subdivisions to use within the range set by <code>zlim</code>.
In case B, <code>zlim</code> is inferred from using <code><a href="#topic+rangeExtended">rangeExtended()</a></code> on <code>c(x0,x1)</code>.
In case C, providing <code>zlim</code> yields an error message, because it makes no
sense in the context of a named, predefined color scheme.</p>
</td></tr>
<tr><td><code id="colormap_+3A_zclip">zclip</code></td>
<td>
<p>logical, with <code>TRUE</code> indicating that z values outside the
range of <code>zlim</code> or <code>breaks</code> should be painted with
<code>missingColor</code> and <code>FALSE</code> indicating that these values should be
painted with the nearest in-range color.</p>
</td></tr>
<tr><td><code id="colormap_+3A_breaks">breaks</code></td>
<td>
<p>an optional indication of break points between color levels
(see <code><a href="Matrix.html#topic+image">image()</a></code>).  If this is provided, the arguments <code>name</code>
through <code>blend</code> are all ignored (see &ldquo;Details&rdquo;).  If it is
provided, then it may either be a vector of break points, or a single number
indicating the desired number of break points to be computed with
<code><a href="base.html#topic+pretty">pretty</a></code><code style="white-space: pre;">&#8288;(z,breaks)&#8288;</code>.  In either case of non-missing
<code>breaks</code>, the resultant break points must number 1 plus the number of
colors (see <code>col</code>).</p>
</td></tr>
<tr><td><code id="colormap_+3A_col">col</code></td>
<td>
<p>either a vector of colors or a function taking a numerical value
as its single argument and returning a vector of colors.  Prior to 2021-02-08,
the default for <code>col</code> was <code>oceColorsJet</code>, but it was switched to
<code>oceColorsViridis</code> on that date.  The value of
<code>col</code> is ignored if <code>name</code> is provided, or if <code>x0</code> through
<code>col1</code> are provided.</p>
</td></tr>
<tr><td><code id="colormap_+3A_name">name</code></td>
<td>
<p>an optional string naming a built-in colormap (one of
<code>"gmt_relief"</code>, <code>"gmt_ocean"</code>, <code>"gmt_globe"</code> or
<code>"gmt_gebco"</code>) or the name of a file or URL that contains a color map
specification in GMT format.  If <code>name</code> is given, then it is
passed to <code><a href="#topic+colormapGMT">colormapGMT()</a></code>, which creates the colormap.
Note that the colormap thus created has a fixed
relationship between value and color, and <code>zlim</code>,
only other argument that is examined is <code>z</code> (which may be used
so that <code>zcol</code> will be defined in the return value), and warnings
are issued if some irrelevant arguments are provided.</p>
</td></tr>
<tr><td><code id="colormap_+3A_x0">x0</code>, <code id="colormap_+3A_x1">x1</code>, <code id="colormap_+3A_col0">col0</code>, <code id="colormap_+3A_col1">col1</code></td>
<td>
<p>Vectors that specify a color map.  They must all be
the same length, with <code>x0</code> and <code>x1</code> being numerical values, and
<code>col0</code> and <code>col1</code> being colors.  The colors may be strings (e.g.
<code>"red"</code>) or colors as defined by <code><a href="grDevices.html#topic+rgb">rgb()</a></code> or <code><a href="grDevices.html#topic+hsv">hsv()</a></code>.</p>
</td></tr>
<tr><td><code id="colormap_+3A_blend">blend</code></td>
<td>
<p>a number indicating how to blend colors within each band.
This is ignored except when <code>x0</code> through <code>col1</code> are supplied.  A
value of 0 means to use <code>col0[i]</code> through the interval <code>x0[i]</code> to
<code>x1[i]</code>.  A value of 1 means to use <code>col1[i]</code> in that interval.  A
value between 0 and 1 means to blend between the two colors according to
the stated fraction.  Values exceeding 1 are an error at present, but there
is a plan to use this to indicate sub-intervals, so a smooth palette can be
created from a few colors.</p>
</td></tr>
<tr><td><code id="colormap_+3A_missingcolor">missingColor</code></td>
<td>
<p>color to use for missing values. This cannot be provided
if <code>name</code> is also provided (case C), because named schemes have pre-defined
colors.  For other cases, <code>missingColor</code> defaults to <code>"gray"</code>, if it
is not provided as an argument.</p>
</td></tr>
<tr><td><code id="colormap_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>colormap</code> can be used in a variety of ways, including the following.
</p>

<ul>
<li> <p><strong>Case A.</strong> Supply some combination of arguments that
is sufficient to define a mapping of value to color, without
providing <code>x0</code>, <code>col0</code>, <code>x1</code> or <code>col1</code> (see case B for these),
or providing <code>name</code> (see Case C). There are several ways to
do this.  One approach is to supply <code>z</code> but no
other argument, in which case <code>zlim</code>, and <code>breaks</code> will be determined
from <code>z</code>, and the default <code>col</code> will be used.  Another approach is
to specify <code>breaks</code> and <code>col</code> together, in the same way as they
might be specified for the base R function <code><a href="Matrix.html#topic+image">image()</a></code>.  It is
also possible to supply only <code>zlim</code>, in which case <code>breaks</code> is
inferred from that value.
The figure below explains the
('breaks', 'col') method of specifying a color mapping.  Note
that there must be one more break than color.  This is the method used by
e.g. [image()].
<img src="../help/figures/colormap_fig_1.png" alt="colormap_fig_1.png" />
</p>
</li>
<li> <p><strong>Case B.</strong> Supply <code>x0</code>, <code>col0</code>, <code>x1</code>, and <code>col1</code>, but <em>not</em>
<code>zlim</code>, <code>breaks</code>, <code>col</code> or <code>name</code>.
The <code>x0</code>, <code>col0</code>, <code>x1</code> and <code>col1</code> values specify a
value-color mapping that is similar to that used
for GMT color maps.  The method works by using <code><a href="base.html#topic+seq">seq()</a></code> to
interpolate between the elements of the <code>x0</code> vector.  The same is done
for <code>x1</code>.  Similarly, <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette()</a></code> is used to
interpolate between the colors in the <code>col0</code> vector, and the same is
done for <code>col1</code>.  The figure above explains the ('x0',
'x1', 'col0', 'col1') method of specifying a color mapping.
Note that the each of the items has the same length. The case of
'blend=0', which has color 'col0[i]' between 'x0[i]' and
'x1[i]', is illustrated below.
<img src="../help/figures/colormap_fig_2.png" alt="colormap_fig_2.png" />
</p>
</li>
<li> <p><strong>Case C.</strong> Supply <code>name</code> and possibly also <code>z</code>, but <em>not</em>
<code>zlim</code>, <code>breaks</code>, <code>col</code>, <code>x0</code>, <code>col0</code>, <code>x1</code> or <code>col1</code>.
The <code>name</code> may be the name of a pre-defined color palette
(<code>"gmt_relief"</code>, <code>"gmt_ocean"</code>, <code>"gmt_globe"</code> or
<code>"gmt_gebco"</code>), or it may be the name of a file (or URL pointing to a file)
that contains a color map in the GMT format (see &ldquo;References&rdquo;). If
<code>z</code> is supplied along with <code>name</code>, then <code>zcol</code> will be set up in the
return value, e.g. for use in colorizing points.  Another method
for finding colors for data points is to use the <code>colfunction()</code>
function in the return value.
</p>
</li></ul>



<h3>Value</h3>

<p>a list containing the following (not necessarily in this order)
</p>

<ul>
<li> <p><code>zcol</code>, a vector of colors for <code>z</code>, if <code>z</code> was
provided, otherwise <code>"black"</code>
</p>
</li>
<li> <p><code>zlim</code>, a two-element vector suitable as the argument of the same
name supplied to <code><a href="Matrix.html#topic+image">image()</a></code> or <code><a href="#topic+imagep">imagep()</a></code>
</p>
</li>
<li> <p><code>breaks</code> and <code>col</code>, vectors of breakpoints and colors,
suitable as the same-named arguments to <code><a href="Matrix.html#topic+image">image()</a></code> or
<code><a href="#topic+imagep">imagep()</a></code>
</p>
</li>
<li> <p><code>zclip</code> the provided value of <code>zclip</code>.
</p>
</li>
<li> <p><code>x0</code> and <code>x1</code>, numerical vectors of the sides of color
intervals, and <code>col0</code> and <code>col1</code>, vectors of corresponding
colors.  The meaning is the same as on input.  The purpose of returning
these four vectors is to permit users to alter color mapping, as in example
3 in &ldquo;Examples&rdquo;.
</p>
</li>
<li> <p><code>missingColor</code>, a color that could be used to specify missing
values, e.g. as the same-named argument to <code><a href="#topic+imagep">imagep()</a></code>.
</p>
</li>
<li> <p><code>colfunction</code>, a univariate function that returns a vector
of colors, given a vector of <code>z</code> values; see Example 6.
</p>
</li></ul>



<h3>Sample of Usage</h3>

<pre>
# Example 2. topographic image with a standard color scheme
par(mfrow=c(1,1))
data(topoWorld)
cm &lt;- colormap(name="gmt_globe")
imagep(topoWorld, breaks=cm$breaks, col=cm$col)

# Example 3. topographic image with modified colors,
# black for depths below 4km.
cm &lt;- colormap(name="gmt_globe")
deep &lt;- cm$x0 &lt; -4000
cm$col0[deep] &lt;- "black"
cm$col1[deep] &lt;- "black"
cm &lt;- colormap(x0=cm$x0, x1=cm$x1, col0=cm$col0, col1=cm$col1)
imagep(topoWorld, breaks=cm$breaks, col=cm$col)

# Example 4. image of world topography with water colorized
# smoothly from violet at 8km depth to blue
# at 4km depth, then blending in 0.5km increments
# to white at the coast, with tan for land.
cm &lt;- colormap(x0=c(-8000, -4000,   0,  100),
     x1=c(-4000,     0, 100, 5000),
     col0=c("violet","blue","white","tan"),
     col1=c("blue","white","tan","yellow"))
lon &lt;- topoWorld[["longitude"]]
lat &lt;- topoWorld[["latitude"]]
z &lt;- topoWorld[["z"]]
imagep(lon, lat, z, breaks=cm$breaks, col=cm$col)
contour(lon, lat, z, levels=0, add=TRUE)

# Example 5. visualize GMT style color map
cm &lt;- colormap(name="gmt_globe", debug=4)
plot(seq_along(cm$x0), cm$x0, pch=21, bg=cm$col0)
grid()
points(seq_along(cm$x1), cm$x1, pch=21, bg=cm$col1)

# Example 6. colfunction
cm &lt;- colormap(c(0, 1))
x &lt;- 1:10
y &lt;- (x - 5.5)^2
z &lt;- seq(0, 1, length.out=length(x))
drawPalette(colormap=cm)
plot(x, y, pch=21, bg=cm$colfunction(z), cex=3)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Example 1. color scheme for points on xy plot
x &lt;- seq(0, 1, length.out = 40)
y &lt;- sin(2 * pi * x)
par(mar = c(3, 3, 1, 1))
mar &lt;- par("mar") # prevent margin creep by drawPalette()
# First, default breaks
c &lt;- colormap(y)
drawPalette(c$zlim, col = c$col, breaks = c$breaks)
plot(x, y, bg = c$zcol, pch = 21, cex = 1)
grid()
par(mar = mar)
# Second, 100 breaks, yielding a smoother palette
c &lt;- colormap(y, breaks = 100)
drawPalette(c$zlim, col = c$col, breaks = c$breaks)
plot(x, y, bg = c$zcol, pch = 21, cex = 1)
grid()
par(mar = mar)

</code></pre>

<hr>
<h2 id='colormapGMT'>Create a GMT-type (CPT) Colormap</h2><span id='topic+colormapGMT'></span>

<h3>Description</h3>

<p><code>colormapGMT</code> creates colormaps in the Generic Mapping Tools (GMT)
scheme (see References 1 to 4).  A few such schemes are built-in, and may be referred to
by name (<code>"gmt_gebco"</code>, <code>"gmt_globe"</code>, <code>"gmt_ocean"</code>, or <code>"gmt_relief"</code>)
while others are handled by reading local files that are in GMT
format, or URLs providing such files (see Reference 3).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colormapGMT(name, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colormapGMT_+3A_name">name</code></td>
<td>
<p>character value specifying the GMT scheme, or a source for such
a scheme. Four pre-defined schemes are available, accessed by setting <code>name</code> to
<code>"gmt_gebco"</code>, <code>"gmt_globe"</code>, <code>"gmt_ocean"</code>, or <code>"gmt_relief"</code>. If <code>name</code> is
not one of these values, then it is taken to be the name of a local file
in GMT format or, if no such file is found, a URL holding such a file.</p>
</td></tr>
<tr><td><code id="colormapGMT_+3A_debug">debug</code></td>
<td>
<p>integer that, if positive, indicates to print some debugging output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The GMT files understood by <a href="#topic+colormapGMT">colormapGMT</a> are what GMT calls
&quot;Regular CPT files&quot; (see reference 4).  This is a text format that
can be read and (with care) edited in a text editor.  There
are three categories of lines within this file.  (1) Any
line starting with the <code>"#"</code> character is a comment, and is ignored
by <a href="#topic+colormapGMT">colormapGMT</a>. (2) Lines with 8 numbers specify colour bands.
The first number is a z value, and the three numbers after that
are red, green and blue values in the range from 0 to 255. This
set of 4 numbers is followed on the same line with similar values.
Think of this sequence as describing a band of colours between
two z values. (3) Lines starting with a character, followed
by three numbers, specify particular codings.  The character <code>"B"</code>
specifies background colour, while <code>"F"</code> specifies foreground
colour, and <code>"N"</code> specifies the colour to be used for missing
data (the letter stands for not-a-number).  Only <code>"N"</code> is used
by <a href="#topic+colormapGMT">colormapGMT</a>, and it takes on the role that the <code>missingColor</code>
argument would otherwise have.  (This is why <code>missingColor</code> is not
permitted if <code>name</code> is given.)
</p>


<h3>Value</h3>

<p><code>colormap</code> returns a list, in the same format as the return value for <code><a href="#topic+colormap">colormap()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> General overview of GMT system
<code style="white-space: pre;">&#8288;https://www.generic-mapping-tools.org&#8288;</code>.
</p>
</li>
<li><p> Information on GMT color schemes
<code style="white-space: pre;">&#8288;https://docs.generic-mapping-tools.org/dev/cookbook/cpts.html&#8288;</code>
</p>
</li>
<li><p> Source of GMT specification files
<code style="white-space: pre;">&#8288;https://beamreach.org/maps/gmt/share/cpt/&#8288;</code>
</p>
</li>
<li><p> CPT (color palette table) format
<code style="white-space: pre;">&#8288;https://www.soest.hawaii.edu/gmt/gmt/html/GMT_Docs.html#x1-820004.15&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>

<hr>
<h2 id='composite'>Create a Composite Object by Averaging Across Good Data</h2><span id='topic+composite'></span>

<h3>Description</h3>

<p>Items within the <code>data</code> slots of the
objects that are supplied as arguments are averaged in a way that
makes sense for the object class, i.e. taking into account
the particular bad-data codes of that particular class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>composite(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="composite_+3A_object">object</code></td>
<td>
<p>either a <a href="base.html#topic+list">list</a> of <a href="#topic+oce-class">oce</a> objects, in
which case this is the only argument, or a single <a href="#topic+oce-class">oce</a> object,
in which case at least one other argument (an object of the same size)
must be supplied.</p>
</td></tr>
<tr><td><code id="composite_+3A_...">...</code></td>
<td>
<p>Ignored, if <code>object</code> is a list. Otherwise, one or more
<a href="#topic+oce-class">oce</a> objects of the same sub-class as the first argument.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other functions that create composite objects: 
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method</a></code>,
<code><a href="#topic+composite+2Clist-method">composite,list-method</a></code>
</p>

<hr>
<h2 id='composite+2Camsr-method'>Create a Composite of amsr Satellite Data</h2><span id='topic+composite+2Camsr-method'></span>

<h3>Description</h3>

<p>Form averages for each item in the <code>data</code> slot of the supplied objects,
taking into account the bad-data codes.
</p>
<p>Items within the <code>data</code> slots of the
objects that are supplied as arguments are averaged in a way that
makes sense for the object class, i.e. taking into account
the particular bad-data codes of that particular class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'amsr'
composite(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="composite+2B2Camsr-method_+3A_object">object</code></td>
<td>
<p>An <a href="#topic+amsr-class">amsr</a> object.</p>
</td></tr>
<tr><td><code id="composite+2B2Camsr-method_+3A_...">...</code></td>
<td>
<p>Other amsr objects.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If none of the objects has good
data at any particular pixel (i.e. particular latitude and longitude),
the resultant will have the bad-data code of the last item in the argument
list.
The metadata in the result are taken directly from the metadata of the
final argument, except that the filename is set to a comma-separated list
of the component filenames.
</p>


<h3>See Also</h3>

<p>Other things related to amsr data: 
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+read.amsr">read.amsr</a>()</code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+summary+2Camsr-method">summary,amsr-method</a></code>
</p>
<p>Other functions that create composite objects: 
<code><a href="#topic+composite+2Clist-method">composite,list-method</a></code>,
<code><a href="#topic+composite">composite</a>()</code>
</p>

<hr>
<h2 id='composite+2Clist-method'>Composite by Averaging Across Data</h2><span id='topic+composite+2Clist-method'></span>

<h3>Description</h3>

<p>This is done
by calling a specialized version of the function defined in the given
class. In the present
version, the objects must inherit from <a href="#topic+amsr-class">amsr</a>, so the
action is to call
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method()</a></code>.
</p>
<p>Items within the <code>data</code> slots of the
objects that are supplied as arguments are averaged in a way that
makes sense for the object class, i.e. taking into account
the particular bad-data codes of that particular class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'list'
composite(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="composite+2B2Clist-method_+3A_object">object</code></td>
<td>
<p>a <a href="base.html#topic+list">list</a> of <a href="#topic+oce-class">oce</a> objects.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other functions that create composite objects: 
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method</a></code>,
<code><a href="#topic+composite">composite</a>()</code>
</p>

<hr>
<h2 id='computableWaterProperties'>Determine Available Derived Water Properties</h2><span id='topic+computableWaterProperties'></span>

<h3>Description</h3>

<p>This determines what things can be derived from the supplied
variables.  For example, if <code>salinity</code>, <code>temperature</code>,
and <code>pressure</code> are supplied, then potential temperature, sound
speed, and several other things can be derived.  If, in addition,
<code>longitude</code> and <code>latitude</code> are supplied, then Absolute Salinity,
Conservative Temperature, and some other things can be derived.
Similarly, <code>nitrate</code> can be computed from <code>NO2+NO3</code> together
with <code>nitrate</code>, and <code>nitrite</code> can be computed from <code>NO2+NO3</code>
together with <code>nitrate</code>.
See the &ldquo;Examples&rdquo; for a full listing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>computableWaterProperties(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="computableWaterProperties_+3A_x">x</code></td>
<td>
<p>a specification of the names of known variables.  This
may be (a) an <a href="#topic+oce-class">oce</a> object, in which case the names are
determined by calling <code><a href="base.html#topic+names">names()</a></code> on the <code>data</code> slot of <code>x</code>, or
(b) a vector of character values indicating the names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code><a href="#topic+computableWaterProperties">computableWaterProperties()</a></code> returns a sorted
character vector holding the names of computable
water properties, or NULL, if there are no computable values.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Example 1
data(ctd)
computableWaterProperties(ctd)
# Example 2: nothing an be computed from just salinity
computableWaterProperties("salinity")
# Example 3: quite a lot can be computed from this trio of values
computableWaterProperties(c("salinity", "temperature", "pressure"))
# Example 4: now we can get TEOS-10 values as well
computableWaterProperties(c(
    "salinity", "temperature", "pressure",
    "longitude", "latitude"
))

</code></pre>

<hr>
<h2 id='concatenate'>Concatenate oce Objects (Generic)</h2><span id='topic+concatenate'></span>

<h3>Description</h3>

<p>Concatenate oce Objects (Generic)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>concatenate(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concatenate_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="concatenate_+3A_...">...</code></td>
<td>
<p>optional additional <a href="#topic+oce-class">oce</a> objects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class corresponding to that of <code>object</code>.
</p>


<h3>See Also</h3>

<p>Other functions that concatenate oce objects: 
<code><a href="#topic+concatenate+2Cadp-method">concatenate,adp-method</a></code>,
<code><a href="#topic+concatenate+2Clist-method">concatenate,list-method</a></code>,
<code><a href="#topic+concatenate+2Coce-method">concatenate,oce-method</a></code>
</p>

<hr>
<h2 id='concatenate+2Cadp-method'>Concatenate adp Objects</h2><span id='topic+concatenate+2Cadp-method'></span>

<h3>Description</h3>

<p>This function concatenates adp objects. It is intended
for objects holding data sampled through time, and it works by pasting
together data linearly if they are vectors, by row if they are matrices,
and by second index if they are arrays. It has been tested for the following
classes:
<a href="#topic+adp-class">adp</a>,
<a href="#topic+adv-class">adv</a>,
<a href="#topic+ctd-class">ctd</a>,
and
<a href="#topic+met-class">met</a>. It may do useful things for other classes, and
so users are encouraged to try, and to report problems to the developers.
It is unlikely that the function will do anything even remotely useful
for image and topographic data, to name just two cases that do not fit
the sampled-over-time category.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adp'
concatenate(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concatenate+2B2Cadp-method_+3A_object">object</code></td>
<td>
<p>An object of <a href="#topic+adp-class">adp</a>, or a list containing such
objects (in which case the remaining arguments are ignored).</p>
</td></tr>
<tr><td><code id="concatenate+2B2Cadp-method_+3A_...">...</code></td>
<td>
<p>Optional additional objects of <a href="#topic+adp-class">adp</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of <a href="#topic+adp-class">adp</a>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that concatenate oce objects: 
<code><a href="#topic+concatenate+2Clist-method">concatenate,list-method</a></code>,
<code><a href="#topic+concatenate+2Coce-method">concatenate,oce-method</a></code>,
<code><a href="#topic+concatenate">concatenate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## 1. Split, then recombine, a ctd object.
data(ctd)
ctd1 &lt;- subset(ctd, scan &lt;= median(ctd[["scan"]]))
ctd2 &lt;- subset(ctd, scan &gt; median(ctd[["scan"]]))
CTD &lt;- concatenate(ctd1, ctd2)

## 2. Split, then recombine, an adp object.
data(adp)
midtime &lt;- median(adp[["time"]])
adp1 &lt;- subset(adp, time &lt;= midtime)
adp2 &lt;- subset(adp, time &gt; midtime)
ADP &lt;- concatenate(adp1, adp2)

## Not run: 
## 3. Download two met files and combine them.
met1 &lt;- read.met(download.met(id=6358, year=2003, month=8))
met2 &lt;- read.met(download.met(id=6358, year=2003, month=9))
MET &lt;- concatenate(met1, met2)

## End(Not run)

</code></pre>

<hr>
<h2 id='concatenate+2Clist-method'>Concatenate a List of oce Objects</h2><span id='topic+concatenate+2Clist-method'></span>

<h3>Description</h3>

<p>Concatenate a List of oce Objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'list'
concatenate(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concatenate+2B2Clist-method_+3A_object">object</code></td>
<td>
<p>a <a href="base.html#topic+list">list</a> of <a href="#topic+oce-class">oce</a> objects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class corresponding to that in <code>object</code>.
</p>


<h3>See Also</h3>

<p>Other functions that concatenate oce objects: 
<code><a href="#topic+concatenate+2Cadp-method">concatenate,adp-method</a></code>,
<code><a href="#topic+concatenate+2Coce-method">concatenate,oce-method</a></code>,
<code><a href="#topic+concatenate">concatenate</a>()</code>
</p>

<hr>
<h2 id='concatenate+2Coce-method'>Concatenate oce Objects (oce-Specific)</h2><span id='topic+concatenate+2Coce-method'></span>

<h3>Description</h3>

<p>This function concatenates oce objects. It is intended
for objects holding data sampled through time, and it works by pasting
together data linearly if they are vectors, by row if they are matrices,
and by second index if they are arrays. It has been tested for the following
classes:
<a href="#topic+adp-class">adp</a>,
<a href="#topic+adv-class">adv</a>,
<a href="#topic+ctd-class">ctd</a>,
and
<a href="#topic+met-class">met</a>. It may do useful things for other classes, and
so users are encouraged to try, and to report problems to the developers.
It is unlikely that the function will do anything even remotely useful
for image and topographic data, to name just two cases that do not fit
the sampled-over-time category.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'oce'
concatenate(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="concatenate+2B2Coce-method_+3A_object">object</code></td>
<td>
<p>An object of <a href="#topic+oce-class">oce</a>, or a list containing such
objects (in which case the remaining arguments are ignored).</p>
</td></tr>
<tr><td><code id="concatenate+2B2Coce-method_+3A_...">...</code></td>
<td>
<p>Optional additional objects of <a href="#topic+oce-class">oce</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of <a href="#topic+oce-class">oce</a>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that concatenate oce objects: 
<code><a href="#topic+concatenate+2Cadp-method">concatenate,adp-method</a></code>,
<code><a href="#topic+concatenate+2Clist-method">concatenate,list-method</a></code>,
<code><a href="#topic+concatenate">concatenate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## 1. Split, then recombine, a ctd object.
data(ctd)
ctd1 &lt;- subset(ctd, scan &lt;= median(ctd[["scan"]]))
ctd2 &lt;- subset(ctd, scan &gt; median(ctd[["scan"]]))
CTD &lt;- concatenate(ctd1, ctd2)

## 2. Split, then recombine, an adp object.
data(adp)
midtime &lt;- median(adp[["time"]])
adp1 &lt;- subset(adp, time &lt;= midtime)
adp2 &lt;- subset(adp, time &gt; midtime)
ADP &lt;- concatenate(adp1, adp2)

## Not run: 
## 3. Download two met files and combine them.
met1 &lt;- read.met(download.met(id=6358, year=2003, month=8))
met2 &lt;- read.met(download.met(id=6358, year=2003, month=9))
MET &lt;- concatenate(met1, met2)

## End(Not run)

</code></pre>

<hr>
<h2 id='coriolis'>Coriolis Parameter on the Earth</h2><span id='topic+coriolis'></span>

<h3>Description</h3>

<p>Compute <code class="reqn">f</code>, the Coriolis parameter as a function of latitude
(see reference 1),
assuming earth siderial angular rotation rate
<code class="reqn">omega</code>=7292115e-11 rad/s. See reference 1 for general notes, and
see reference 2 for comments on temporal variations
of <code class="reqn">omega</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coriolis(latitude, degrees = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coriolis_+3A_latitude">latitude</code></td>
<td>
<p>Vector of latitudes in <code class="reqn">^\circ</code>N or radians north of the equator.</p>
</td></tr>
<tr><td><code id="coriolis_+3A_degrees">degrees</code></td>
<td>
<p>Flag indicating whether degrees are used for latitude; if set
to <code>FALSE</code>, radians are used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Coriolis parameter, in radian/s.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Gill, A.E., 1982. <em>Atmosphere-ocean Dynamics</em>, Academic
Press, New York, 662 pp.
</p>
</li>
<li><p> Groten, E., 2004: Fundamental Parameters and Current, 2004. Best
Estimates of the Parameters of Common Relevance to Astronomy, Geodesy,
and Geodynamics. Journal of Geodesy, 77:724-797.
(downloaded from
<code style="white-space: pre;">&#8288;http://www.iag-aig.org/attach/e354a3264d1e420ea0a9920fe762f2a0/51-groten.pdf&#8288;</code>
March 11, 2017).
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>C &lt;- coriolis(45) # 1e-4
</code></pre>

<hr>
<h2 id='ctd'>Sample ctd Data</h2><span id='topic+ctd'></span>

<h3>Description</h3>

<p>This is a CTD profile measured in Halifax Harbour in 2003, based
on <code><a href="#topic+ctdRaw">ctdRaw()</a></code>, but trimmed to just the downcast with
<code><a href="#topic+ctdTrim">ctdTrim()</a></code>, using indices inferred by inspection of the
results from <code><a href="#topic+plotScan">plotScan()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ctd)
</code></pre>


<h3>Details</h3>

<p>This station was sampled by students enrolled in the Dan Kelley's Physical
Oceanography class at Dalhousie University.  The data were acquired near the
centre of the Bedford Basin of the Halifax Harbour, during an October 2003
field trip of Dalhousie University's Oceanography 4120/5120 class.  Note
that the <code>startTime</code> in the <code>metadata</code> slot was altered from 1903 to 2003,
using <code><a href="#topic+oceEdit">oceEdit()</a></code>. The change was done because the original time was clearly
incorrect, perhaps owing to the use of software that was designed to work in
the twentieth century only.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
data(ctd)
plot(ctd)
</pre>


<h3>See Also</h3>

<p>The full profile (not trimmed to the downcast) is available as
<code>data(ctdRaw)</code>.
</p>
<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>

<hr>
<h2 id='ctd_aml.csv.gz'>Sample ctd File in aml Format</h2><span id='topic+ctd_aml.csv.gz'></span>

<h3>Description</h3>

<p>This file may be read with <code><a href="#topic+read.ctd.aml">read.ctd.aml()</a></code>.  It is based
on a file donated by Ashley Stanek, which was shortened to
just 50 points for inclusion in oce, and which had some
identifying information (serial number, IP address, and WEP
code) zeroed-out.
</p>


<h3>See Also</h3>

<p>Other raw datasets: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ctd &lt;- read.ctd.aml(system.file("extdata", "ctd_aml.csv.gz", package="oce"))
summary(ctd)
plot(ctd)

</code></pre>

<hr>
<h2 id='CTD_BCD2014666_008_1_DN.ODF.gz'>Sample ctd File in .odf Format</h2><span id='topic+CTD_BCD2014666_008_1_DN.ODF.gz'></span>

<h3>Description</h3>

<p>The location is approximately 30km southeast of Halifax Harbour,
at &quot;Station 2&quot; of the Halifax Line on the Scotian Shelf.
</p>


<h3>See Also</h3>

<p>Other raw datasets: 
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other things related to odf data: 
<code><a href="#topic+ODF2oce">ODF2oce</a>()</code>,
<code><a href="#topic+ODFListFromHeader">ODFListFromHeader</a>()</code>,
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.odf">read.odf</a>()</code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+summary+2Codf-method">summary,odf-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ctd &lt;- read.ctd(system.file("extdata", "CTD_BCD2014666_008_1_DN.ODF.gz", package="oce"))
plot(ctd)

</code></pre>

<hr>
<h2 id='ctd-class'>Class to Store CTD (or general hydrographic) Data</h2><span id='topic+ctd-class'></span>

<h3>Description</h3>

<p>This class stores hydrographic data such as measured with a CTD (conductivity,
temperature, depth) instrument, or with other systems that produce
similar data.  Data repositories may store conductivity, temperature
and depth, as in the instrument name, but it is also common to store
salinity, temperature and pressure instead (or in addition). For this
reason, <code>ctd</code> objects are required to hold <code>salinity</code>,
<code>temperature</code> and <code>pressure</code> in their <code>data</code> slot,
with other data being optional. Formulae are available for converting
between variants of these data triplets, e.g. <code><a href="#topic+swSCTp">swSCTp()</a></code>
can calculate <code>salinity</code> given <code>conductivity</code>, <code>temperature</code>
and <code>pressure</code>, and these are used by the main functions that
create <code>ctd</code> objects. For example, if <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code>
is used to read a Seabird file that contains only conductivity, temperature
and pressure, then that function will automatically append a data
item to hold salinity.  Since <code><a href="#topic+as.ctd">as.ctd()</a></code> does the same with
salinity, the result this is that all <code>ctd</code> objects hold <code>salinity</code>,
<code>temperature</code> and <code>pressure</code>, which are henceforth called
the three basic quantities.
</p>


<h3>Details</h3>

<p>Different units and scales are permitted for the three basic quantities, and
most <code>oce</code> functions check those units and scales before
doing calculations (e.g. of seawater density), because those calculations
demand certain units and scales. The way this is handled is that the
accessor function <code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>] returns values in standardized
form. For example, a <code>ctd</code> object might hold temperature defined on the
IPTS-68 scale, but e.g. <code>ctd[["temperature"]]</code> returns a value on the ITS-90
scale. (The conversion is done with <code><a href="#topic+T90fromT68">T90fromT68()</a></code>.)  Similarly,
pressure may be stored in either dbars or PSI, but e.g. <code>ctd[["pressure"]]</code>
returns a value in dbars, after dividing by 0.689476 if the value is
stored in PSI. Luckily, there is (as of early 2016) only one salinity scale in
common use in data files, namely PSS-78.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>ctd</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
The key items stored in this slot are: <code>salinity</code>, <code>temperature</code>, and <code>pressure</code>, although in many instances there are quite a few additional items.</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>ctd</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
An example of the former might be the location at which a <code>ctd</code> measurement was made, stored in <code>longitude</code> and <code>latitude</code>, and of the latter might be <code>filename</code>, the name of the data source.</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>ctd</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+ctd-class">ctd</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+ctd-class">ctd</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Reading/creating <code>ctd</code> objects</h3>

<p>A file containing CTD profile data may be read with
<code><a href="#topic+read.ctd">read.ctd()</a></code>, and a CTD object can also be created with
<code><a href="#topic+as.ctd">as.ctd()</a></code>.  See <code><a href="#topic+read.ctd">read.ctd()</a></code> for references on data
formats used in CTD files. Data can also be assembled into
<code>ctd</code> objects with <code><a href="#topic+as.ctd">as.ctd()</a></code>.
</p>
<p>Statistical summaries are provided by <code><a href="#topic+summary+2Cctd-method">summary,ctd-method()</a></code>, while
<code><a href="methods.html#topic+show">show()</a></code> displays an overview.
</p>
<p>CTD objects may be plotted with <code><a href="#topic+plot+2Cctd-method">plot,ctd-method()</a></code>, which does much of its
work by calling <code><a href="#topic+plotProfile">plotProfile()</a></code> or <code><a href="#topic+plotTS">plotTS()</a></code>, both of
which can also be called by the user, to get fine control over the plots.
</p>
<p>A CTD profile can be isolated from a larger record with <code><a href="#topic+ctdTrim">ctdTrim()</a></code>,
a task made easier when <code><a href="#topic+plotScan">plotScan()</a></code> is used to examine the results.
Towyow data can be split up into sets of profiles (ascending or descending)
with <code><a href="#topic+ctdFindProfiles">ctdFindProfiles()</a></code>.  CTD data may be smoothed and/or cast onto
specified pressure levels with <code><a href="#topic+ctdDecimate">ctdDecimate()</a></code>.
</p>
<p>As with all oce objects, low-level manipulation may be done with
<code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>. Additionally,
many of the contents of CTD objects may be altered with the
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code> scheme,
and sufficiently skilled users may even manipulate the contents directly.
</p>


<h3>Data sources</h3>

<p>Archived CTD (and other) data may be found on servers such as
</p>

<ol>
<li> <p><code style="white-space: pre;">&#8288;https://cchdo.ucsd.edu/&#8288;</code>
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# 1. Create a ctd object with fake data.
a &lt;- as.ctd(salinity = 35 + 1:3 / 10, temperature = 10 - 1:3 / 10, pressure = 1:3)
summary(a)

# 2. Fix a typo in a station latitude (fake! it's actually okay)
data(ctd)
ctd &lt;- oceSetMetadata(
    ctd, "latitude", ctd[["latitude"]] - 0.001,
    "fix latitude typo in log book"
)

</code></pre>

<hr>
<h2 id='ctd.cnv.gz'>Sample ctd File in .cnv Format</h2><span id='topic+ctd.cnv.gz'></span>

<h3>Description</h3>

<p>Sample ctd File in .cnv Format
</p>


<h3>See Also</h3>

<p>Other raw datasets: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>read.oce(system.file("extdata", "ctd.cnv.gz", package="oce"))

</code></pre>

<hr>
<h2 id='ctdDecimate'>Decimate a ctd Profile</h2><span id='topic+ctdDecimate'></span>

<h3>Description</h3>

<p>Interpolate a CTD profile to specified pressure values.  This is used
by <code><a href="#topic+sectionGrid">sectionGrid()</a></code>, but is also useful for dealing with individual
CTD/bottle profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctdDecimate(
  x,
  p = 1,
  method = "boxcar",
  rule = 1,
  e = 1.5,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctdDecimate_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="ctdDecimate_+3A_p">p</code></td>
<td>
<p>pressure increment, or vector of pressures.  In the first case,
pressures from 0dbar to the rounded maximum pressure are used, incrementing by
<code>p</code> dbars.  If a vector of pressures is given, interpolation is done to
these pressures.</p>
</td></tr>
<tr><td><code id="ctdDecimate_+3A_method">method</code></td>
<td>
<p>the method to be used for calculating decimated values.  This may
be a string specifying the method, or a function. In the string case, the
possibilities are as follows.
</p>

<ul>
<li> <p><code>"boxcar"</code> (based on a local average)
</p>
</li>
<li> <p><code>"approx"</code> (based on linear
interpolation between neighboring points, using <code><a href="stats.html#topic+approx">approx()</a></code>
with the <code>rule</code> argument specified here)
</p>
</li>
<li> <p><code>"approxML"</code> as <code>"approx"</code>,
except that a mixed layer is assumed to apply above the top data value; this
is done by setting the <code>yleft</code> argument to <code><a href="stats.html#topic+approx">approx()</a></code>, and
by calling that function with <code style="white-space: pre;">&#8288;rule=c(2, 1))&#8288;</code>
</p>
</li>
<li> <p><code>"lm"</code> (based on local
regression, with <code>e</code> setting the size of the local region);
</p>
</li>
<li> <p><code>"rr"</code> for the Reiniger and Ross method, carried out with <code><a href="#topic+oce.approx">oce.approx()</a></code>;
</p>
</li>
<li> <p><code>"unesco"</code> (for the UNESCO method, carried out with <code><a href="#topic+oce.approx">oce.approx()</a></code>.
</p>
</li></ul>

<p>On the other hand, if <code>method</code> is a function, then it must take
two arguments, named <code>data</code> and <code>parameters</code>. The first is set to <code>x@data</code> by
<code>ctdTrim()</code>. The second is passed directly to the user's function (see
Example 2). The return value from the function must be a logical vector of
the same length as the <code>pressure</code> data, with TRUE values meaning to keep the
corresponding entries of the <code>data</code> slot.</p>
</td></tr>
<tr><td><code id="ctdDecimate_+3A_rule">rule</code></td>
<td>
<p>an integer that is passed to <code><a href="stats.html#topic+approx">approx()</a></code>, in the
case where <code>method</code> is <code>"approx"</code>. Note that the default value
for <code>rule</code> is 1, which will inhibit extrapolation beyond the observed
pressure range. This is a change from the behaviour previous to May 8, 2017,
when a <code>rule</code> of 2 was used (without stating so as an argument).</p>
</td></tr>
<tr><td><code id="ctdDecimate_+3A_e">e</code></td>
<td>
<p>is an expansion coefficient used to calculate the local neighbourhoods
for the <code>"boxcar"</code> and <code>"lm"</code> methods.  If <code>e=1</code>, then the
neighbourhood for the i-th pressure extends from the (<code>i-1</code>)-th pressure to
the (<code>i+1</code>)-th pressure.  At the endpoints it is assumed that the outside
bin is of the same pressure range as the first inside bin.  For other values of
<code>e</code>, the neighbourhood is expanded linearly in each direction.  If the
<code>"lm"</code> method produces warnings about &quot;prediction from a rank-deficient
fit&quot;, a larger value of <code>"e"</code> should be used.</p>
</td></tr>
<tr><td><code id="ctdDecimate_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>"approx"</code> and <code>"approxML"</code> methods may be best for bottle data,
in which the usual task is
to interpolate from a coarse sampling grid to a finer one. The distinction
is that <code>"approxML"</code> assumes a mixed-layer above the top sample value. For CTD data, the
<code>"boxcar"</code> method may be the preferred choice, because the task is normally
to sub-sample, and some degree of smoothing is usually desired.  (The
<code>"lm"</code> method can be quite slow, and its results may be quite similar to those of the
boxcar method.)
</p>
<p>For widely-spaced data, a sort of numerical cabeling effect can result when
density is computed based on interpolated salinity and temperature.
See reference 2 for a discussion of this issue and possible solutions.
</p>


<h3>Value</h3>

<p>A <a href="#topic+ctd-class">ctd</a> object, with pressures that are as set by
the <code>"p"</code> parameter and all other properties modified appropriately.
</p>


<h3>A note about flags</h3>

<p>Data-quality flags contained within the original object are ignored by this
function, and the returned value contains no such flags.  This is because such
flags represent an assessment of the original data, not of quantities derived
from those data.  This function produces a warning to this effect. The
recommended practice is to use <code><a href="#topic+handleFlags">handleFlags()</a></code> or some other means to
deal with flags before calling the present function.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> R.F. Reiniger and C.K. Ross, 1968.  A method of interpolation with
application to oceanographic data.  <em>Deep Sea Research</em>, <strong>15</strong>,
185-193.
</p>
</li>
<li><p> Oguma, Sachiko, Toru Suzuki, Yutaka Nagata, Hidetoshi Watanabe, Hatsuyo Yamaguchi,
and Kimio Hanawa. &ldquo;Interpolation Scheme for Standard Depth Data Applicable for Areas
with a Complex Hydrographical Structure.&rdquo; Journal of Atmospheric and Oceanic Technology
21, no. 4 (April 1, 2004): 704-15.
</p>
</li></ol>



<h3>See Also</h3>

<p>The documentation for <a href="#topic+ctd-class">ctd</a> explains the structure of
CTD objects, and also outlines the other functions dealing with them.
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
plotProfile(ctd, "salinity", ylim = c(10, 0))
p &lt;- seq(0, 45, 1)
ctd2 &lt;- ctdDecimate(ctd, p = p)
lines(ctd2[["salinity"]], ctd2[["pressure"]], col = "blue")
p &lt;- seq(0, 45, 1)
ctd3 &lt;- ctdDecimate(ctd, p = p, method = function(x, y, xout) {
    predict(smooth.spline(x, y, df = 30), xout)$y
})
lines(ctd3[["salinity"]], ctd3[["pressure"]], col = "red")

</code></pre>

<hr>
<h2 id='ctdFindProfiles'>Find Profiles Within a Tow-Yow ctd Record</h2><span id='topic+ctdFindProfiles'></span>

<h3>Description</h3>

<p>Examine the pressure record looking for extended periods of either ascent or descent, and return
either indices to these events or a vector of CTD records containing the events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctdFindProfiles(
  x,
  cutoff = 0.5,
  minLength = 10,
  minHeight,
  smoother = smooth.spline,
  direction = c("descending", "ascending"),
  breaks,
  arr.ind = FALSE,
  distinct,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctdFindProfiles_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="ctdFindProfiles_+3A_cutoff">cutoff</code></td>
<td>
<p>criterion on pressure difference; see &ldquo;Details&rdquo;. If not
provided, this defaults to 0.5.</p>
</td></tr>
<tr><td><code id="ctdFindProfiles_+3A_minlength">minLength</code></td>
<td>
<p>lower limit on number of points in candidate profiles. If
not
provided, this defaults to 10.</p>
</td></tr>
<tr><td><code id="ctdFindProfiles_+3A_minheight">minHeight</code></td>
<td>
<p>lower limit on height of candidate profiles. If not
provided, this defaults to 0.1 times the pressure span.</p>
</td></tr>
<tr><td><code id="ctdFindProfiles_+3A_smoother">smoother</code></td>
<td>
<p>The smoothing function to use for identifying down/up
casts. The default is <code>smooth.spline</code>, which performs well for
a small number of cycles; see &ldquo;Examples&rdquo; for a method that is
better for a long tow-yo. The return value from <code>smoother</code> must
be either a list containing an element named <code>y</code> or something
that can be coerced to a vector with <code><a href="base.html#topic+as.vector">as.vector()</a></code>. To
turn smoothing off, so that cycles in pressure are determined by
simple first difference, set <code>smoother</code> to <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="ctdFindProfiles_+3A_direction">direction</code></td>
<td>
<p>String indicating the travel direction to be selected.</p>
</td></tr>
<tr><td><code id="ctdFindProfiles_+3A_breaks">breaks</code></td>
<td>
<p>optional integer vector indicating the indices of last
datum in each profile stored within <code>x</code>. Thus, the first profile
in the return value will contain the <code>x</code> data from indices 1
to <code>breaks[1]</code>.  If <code>breaks</code> is given, then all
other arguments except <code>x</code> are ignored. Using <code>breaks</code>
is handy in cases where other schemes fail, or when the author
has independent knowledge of how the profiles are strung together
in <code>x</code>.</p>
</td></tr>
<tr><td><code id="ctdFindProfiles_+3A_arr.ind">arr.ind</code></td>
<td>
<p>logical value indicating whether the array indices should be
returned; the alternative is to return a vector of ctd objects.</p>
</td></tr>
<tr><td><code id="ctdFindProfiles_+3A_distinct">distinct</code></td>
<td>
<p>An optional string indicating how to identify profiles
by unique values. Use <code>"location"</code>
to find profiles by a change in longitude and latitude, or use the name of any
of item in the <code>data</code> slot in <code>x</code>. In these cases, all the
other arguments except <code>x</code> are ignored. However, if <code>distinct</code>
is not supplied, the other arguments are handled as described above.</p>
</td></tr>
<tr><td><code id="ctdFindProfiles_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
<tr><td><code id="ctdFindProfiles_+3A_...">...</code></td>
<td>
<p>Optional extra arguments that are passed to the smoothing function, <code>smoother</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The method works by examining the pressure record.  First, this is smoothed using
<code>smoother()</code> (see &ldquo;Arguments&rdquo;), and then the result is first-differenced
using <code><a href="base.html#topic+diff">diff()</a></code>.  Median values of the positive and
negative first-difference values are then multiplied by <code>cutoff</code>.  This establishes criteria
for any given point to be in an ascending profile, a descending profile, or a non-profile.
Contiguous regions are then found, and those that have fewer than <code>minLength</code> points are
discarded.  Then, those that have pressure ranges less than <code>minHeight</code> are discarded.
</p>
<p>Caution: this method is not well-suited to all datasets. For example, the default
value of <code>smoother</code> is <code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code>, and this works well for just a few
profiles, but poorly for a tow-yo with a long sequence of profiles; in the latter case,
it can be preferable to use simpler smoothers (see &ldquo;Examples&rdquo;). Also, depending
on the sampling protocol, it is often necessary to pass the resultant profiles through
<code><a href="#topic+ctdTrim">ctdTrim()</a></code>, to remove artifacts such as an equilibration phase, etc.
Generally, one is well-advised to use the present function for a quick look at the data,
relying on e.g. <code><a href="#topic+plotScan">plotScan()</a></code> to identify profiles visually, for a final product.
</p>


<h3>Value</h3>

<p>If <code>arr.ind=TRUE</code>, a data frame with columns <code>start</code> and <code>end</code>, the indices
of the downcasts.  Otherwise, a vector of <code>ctd</code> objects. In this second case,
the station names are set to a form like <code>"10/3"</code>, for the third profile within an
original ctd object with station name <code>"10"</code>, or to <code>"3"</code>, if the original
ctd object had no station name defined.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
# These examples cannot be tested, because they are based on
# data objects that are not provided with oce.

# Example 1. Find profiles within a towyo file, as can result
# if the CTD is cycled within the water column as the ship
# moves.
profiles &lt;- ctdFindProfiles(towyo)

# Example 2. Use a moving average to smooth pressure, instead of the
# default smooth.spline() method. This might avoid a tendency of
# the default scheme to miss some profiles in a long towyo.
movingAverage &lt;- function(x, n = 11, ...)
{
    f &lt;- rep(1/n, n)
    stats::filter(x, f, ...)
}
casts &lt;- ctdFindProfiles(towyo, smoother=movingAverage)

# Example 3: glider data read into a ctd object. Chop
# into profiles by looking for pressure jumps exceeding
# 10 dbar.
breaks &lt;- which(diff(gliderAsCtd[["pressure"]]) &gt; 10)
profiles &lt;- ctdFindProfiles(gliderAsCtd, breaks=breaks)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>See Also</h3>

<p>The documentation for <a href="#topic+ctd-class">ctd</a> explains the structure
of CTD objects, and also outlines the other functions dealing with them.
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>

<hr>
<h2 id='ctdFindProfilesRBR'>Find Profiles Within a ctd Object Read From a RBR File</h2><span id='topic+ctdFindProfilesRBR'></span>

<h3>Description</h3>

<p>This uses information about profiles that is contained within the <code>metadata</code>
slot of the first argument, <code>x</code>, having been inserted there by <code><a href="#topic+read.rsk">read.rsk()</a></code>.
If <code>x</code> was created by reading an <code>.rsk</code> file with <code><a href="#topic+read.rsk">read.rsk()</a></code>,
and if that file contained geographical information (that is, if it had a
data table named <code>geodata</code>) then the <em>first</em> longitude and latitude from each
profile is stored in the <code>metadata</code> slot of the returned value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctdFindProfilesRBR(
  x,
  direction = "descending",
  arr.ind = FALSE,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctdFindProfilesRBR_+3A_x">x</code></td>
<td>
<p>either an <a href="#topic+rsk-class">rsk</a> or a <a href="#topic+ctd-class">ctd</a> object; in the former case,
it is converted to a <a href="#topic+ctd-class">ctd</a> object with <code><a href="#topic+as.ctd">as.ctd()</a></code>.</p>
</td></tr>
<tr><td><code id="ctdFindProfilesRBR_+3A_direction">direction</code></td>
<td>
<p>character value, either <code>"descending"</code> or <code>"ascending"</code>,
indicating the sampling direction to be selected.  The default, <code>"descending"</code>,
is the commonly preferred choice.</p>
</td></tr>
<tr><td><code id="ctdFindProfilesRBR_+3A_arr.ind">arr.ind</code></td>
<td>
<p>logical value indicating whether the array indices should be
returned; the alternative is to return a vector of ctd objects.</p>
</td></tr>
<tr><td><code id="ctdFindProfilesRBR_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic+as.rsk">as.rsk</a>()</code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+read.rsk">read.rsk</a>()</code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+rskPatm">rskPatm</a>()</code>,
<code><a href="#topic+rskToc">rskToc</a>()</code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method</a></code>
</p>

<hr>
<h2 id='ctdRaw'>Sample ctd Data, Not Trimmed of Extraneous Data</h2><span id='topic+ctdRaw'></span>

<h3>Description</h3>

<p>This is sample CTD profile provided for testing.  It includes not just the
(useful) portion of the dataset during which the instrument was being lowered,
but also data from the upcast and from time spent near the surface.  Spikes are
also clearly evident in the pressure record.  With such real-world wrinkles,
this dataset provides a good example of data that need trimming with
<code><a href="#topic+ctdTrim">ctdTrim()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ctdRaw)
</code></pre>


<h3>Details</h3>

<p>This station was sampled by students enrolled in the Dan Kelley's Physical
Oceanography class at Dalhousie University.  The data were acquired near the
centre of the Bedford Basin of the Halifax Harbour, during an October 2003
field trip of Dalhousie University's Oceanography 4120/5120 class. (Note that
the <code>startTime</code> in the <code>metadata</code> slot was altered from 1903 to 2003, using
<code><a href="#topic+oceEdit">oceEdit()</a></code>. The change was done because the original time was clearly
incorrect, perhaps owing to the use of software that was designed to work in
the twentieth only.)
</p>


<h3>See Also</h3>

<p>A similar dataset (trimmed to the downcast) is available as
<code>data(ctd)</code>.
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>

<hr>
<h2 id='ctdRepair'>Repair a Malformed ctd Object</h2><span id='topic+ctdRepair'></span>

<h3>Description</h3>

<p>Make a <a href="#topic+ctd-class">ctd</a> object adhere more closely with the expected form, e.g. by
moving certain things from the <code>data</code> slot to the <code>metadata</code> slot, where
other oce functions may assume they will be located.
This can be handy for objects that were set up
incorrectly, perhaps by inappropriate user insertions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctdRepair(x, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctdRepair_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="ctdRepair_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The possible changes fall into the following categories.
</p>

<ol>
<li><p> If unit-length values for <code>latitude</code>, <code>longitude</code>, <code>time</code>, or <code>station</code>
exist in the <code>data</code> slot, move them to the <code>metadata</code> slot.  However,
leave them in <code>data</code>  if their length exceeds 1, because this can
arise with towyo data.
</p>
</li>
<li><p> If the <code>metadata</code> or <code>data</code> slot contains items named
<code>time</code>, <code>recoveryTime</code>, <code>startTime</code>, or <code>systemUploadTime</code>,
and if these are not in POSIXt format, then use <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code> with
<code>tz="UTC"</code> to convert them to POSIXt format. If that conversion fails,
owing to an unrecognizable format, then the original value
is retained, unaltered.
</p>
</li></ol>



<h3>Value</h3>

<p>A <a href="#topic+ctd-class">ctd</a> object that is based on <code>x</code>, but possibly
with some elements changed as described in the &ldquo;Details&rdquo;
section.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
# Insert location information into 'data' slot, although it belongs in 'metadata'.
ctd@data$latitude &lt;- ctd@metadata$latitude # Done by experts only!
ctd@data$longitude &lt;- ctd@metadata$longitude # Done by experts only!
repaired &lt;- ctdRepair(ctd)

</code></pre>

<hr>
<h2 id='ctdTrim'>Trim Beginning and Ending of a CTD cast</h2><span id='topic+ctdTrim'></span>

<h3>Description</h3>

<p>Often in CTD profiling, the goal is to isolate only the downcast, discarding measurements made in
the air, in an equilibration phase in which the device is held below the water surface, and then the
upcast phase that follows the downcast.  This is handled reasonably well by <code>ctdTrim</code> with
<code>method="downcast"</code>, although it is almost always best to use <code><a href="#topic+plotScan">plotScan()</a></code> to
investigate the data, and then use the <code>method="index"</code> or <code>method="scan"</code> method based on
visual inspection of the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctdTrim(
  x,
  method,
  removeDepthInversions = FALSE,
  parameters = NULL,
  indices = FALSE,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctdTrim_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="ctdTrim_+3A_method">method</code></td>
<td>
<p>A string (or a vector of two strings) specifying the trimming method, or a function to
be used to determine data indices to keep.  If <code>method</code> is not provided, <code>"downcast"</code> is
assumed. See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="ctdTrim_+3A_removedepthinversions">removeDepthInversions</code></td>
<td>
<p>Logical value indicating whether to remove any levels at which depth is
less than, or equal to, a depth above.  (This is needed if the object is to be assembled into a
section, unless <code><a href="#topic+ctdDecimate">ctdDecimate()</a></code> will be used, which will remove the inversions.</p>
</td></tr>
<tr><td><code id="ctdTrim_+3A_parameters">parameters</code></td>
<td>
<p>A list whose elements depend on the method; see &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="ctdTrim_+3A_indices">indices</code></td>
<td>
<p>Logical value indicating what to return. If <code>indices=FALSE</code> (the default),
then the return value is a subsetted <a href="#topic+ctd-class">ctd</a> object. If <code>indices=TRUE</code>,
then the return value is a logical vector that could be used to subset the data
with <code><a href="#topic+subset+2Cctd-method">subset,ctd-method()</a></code> or to set data-quality flags.</p>
</td></tr>
<tr><td><code id="ctdTrim_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>ctdTrim</code> begins by examining the pressure differences between subsequent samples. If
these are all of the same value, then the input <code>ctd</code> object is returned, unaltered.
This handles the case of pressure-binned data. However, if the pressure difference
varies, a variety of approaches are taken to trimming the dataset.
</p>

<ul>
<li><p> If <code>method[1]</code> is <code>"downcast"</code> then an attempt is made to keep only data for
which the CTD is descending.  This is done in stages, with variants based on <code>method[2]</code>, if
supplied.
</p>

<ol>
<li><p> The pressure data are despiked with a smooth() filter with method &quot;3R&quot;.
This removes wild spikes that arise from poor instrument connections, etc.
</p>
</li>
<li> <p><em>Step 2.</em> If no <code>parameters</code> are given, then any data with negative pressures
are deleted.  If there is a parameter named <code>pmin</code>, then that pressure (in decibars)
is used instead as the lower limit. This is a commonly-used setup, e.g.
<code>ctdTrim(ctd, parameters=list(pmin=1))</code> removes the top decibar (roughly 1m) from
the data.  Specifying <code>pmin</code> is a simple way to remove near-surface
data, such as a shallow equilibration phase, and if specified will cause <code>ctdTrim</code>
to skip step 4 below.
</p>
</li>
<li><p> The maximum pressure is determined, and data acquired subsequent to
that point are deleted.  This removes the upcast and any subsequent data.
</p>
</li>
<li><p> If the <code>pmin</code> parameter is not specified, an attempt is made to remove an initial
equilibrium phase by a regression of pressure on scan number.  There are three
variants to this, depending on the value of the second <code>method</code> element.
If <code>method</code> is <code>"A"</code> (or not given), the procedure is to
call <code><a href="stats.html#topic+nls">nls()</a></code> to fit a piecewise linear model of pressure as a function of scan,
in which pressure is constant for scan less than a critical value, and then
linearly varying for with scan. This is  meant to handle the common situation
in which the CTD is held at roughly constant depth (typically
a metre or so) to equilibrate, before it is lowered through the water column.
If <code>method</code> is <code>"B"</code>, the procedure is similar, except that the pressure
in the surface region is taken to be zero (this does not make
much sense, but it might help in some cases). Note that, prior to early 2016,
method <code>"B"</code> was called method <code>"C"</code>; the old <code>"B"</code> method was judged useless
and so it was removed.
</p>
</li></ol>

</li>
<li><p> If <code>method="upcast"</code>, a sort of reverse of <code>"downcast"</code> is used. This
was added in late April 2017 and has not been well tested yet.
</p>
</li>
<li><p> If <code>method="sbe"</code>, a method similar to that described
in the SBE Data Processing manual is used to remove the &quot;soak&quot;
period at the beginning of a cast (see Section 6 under subsection
&quot;Loop Edit&quot;). The method is based on the soak procedure whereby
the instrument sits at a fixed depth for a period of time, after
which it is raised toward the surface before beginning the actual
downcast. This enables equilibration of the sensors while still
permitting reasonably good near-surface data. Parameters for the
method can be passed using the <code>parameters</code> argument, which
include <code>minSoak</code> (the minimum depth for the soak) and
<code>maxSoak</code> the maximum depth of the soak. The method finds
the minimum pressure prior to the <code>maxSoak</code> value being
passed, each of which occurring after the scan in which the
<code>minSoak</code> value was reached. For the method to work, the
pre-cast pressure minimum must be less than the <code>minSoak</code>
value. The default values of <code>minSoak</code> and <code>maxSoak</code>
are 1 and 20 dbar, respectively.
</p>
</li>
<li><p> If <code>method="index"</code> or <code>"scan"</code>, then each column of data is subsetted according to the
value of <code>parameters</code>. If the latter is a logical vector of length matching data column
length, then it is used directly for subsetting. If <code>parameters</code> is a numerical vector with
two elements, then the index or scan values that lie between <code>parameters[1]</code>
and <code>parameters[2]</code> (inclusive) are used for subsetting.  The
two-element method is probably the most useful, with the values being determined by visual
inspection of the results of <code><a href="#topic+plotScan">plotScan()</a></code>. While this may take a minute or two, the
analyst should bear in mind that a deep-water CTD profile might take 6 hours, corresponding to
ship-time costs exceeding a week of salary.
</p>
</li>
<li><p> If <code>method="range"</code> then data are selected based on the value of the column named
<code>parameters$item</code>.  This may be by range or by critical value.  By range: select values
between <code>parameters$from</code> (the lower limit) and <code>parameters$to</code> (the upper limit) By
critical value: select if the named column exceeds the value.  For example, <code>ctd2 &lt;-
    ctdTrim(ctd, "range", parameters=list(item="scan", from=5))</code> starts at scan number 5 and
continues to the end, while
<code>ctdTrim(ctd,"range", parameters=list(item="scan", from=5, to=100))</code> also starts at scan 5,
but extends only to scan 100.
</p>
</li>
<li><p> If <code>method</code> is a function, then it must return a vector of <code><a href="base.html#topic+logical">logical()</a></code>
values, computed based on two arguments: <code>data</code> (a
<code><a href="base.html#topic+list">list()</a></code>), and <code>parameters</code> as supplied to <code>ctdTrim</code>.  Both
<code>inferWaterDepth</code> and <code>removeInversions</code> are ignored in the function case. See
&ldquo;Examples&rdquo;.
</p>
</li></ul>



<h3>Value</h3>

<p>Either a <a href="#topic+ctd-class">ctd</a> object or a logical vector of length matching
the data. In the first case, which is the default, the elements of the <code>data</code>
slot will have been trimmed, along with some elements of the <code>metadata</code> slot
(e.g. <code>metadata4flags</code> and, if present and of length matching
<code>data$pressure</code>, both <code>metadata$longitude</code> and <code>metadata$latitude</code>).  The
second case, achieved by setting <code>indices=FALSE</code>, may be helpful for advanced
users who wish to do things like construct data flags to be inserted into the
object.
</p>


<h3>Historical Note</h3>

<p>The subsetting of <code>longitude</code> and <code>latitude</code> in the <code>metadata</code> slot was
introduced on 2022-12-13, for use with <a href="#topic+ctd-class">ctd</a> objects created using
<code><a href="#topic+as.ctd">as.ctd()</a></code> on <a href="#topic+rsk-class">rsk</a> objects created by using <code><a href="#topic+read.rsk">read.rsk()</a></code> on Ruskin
files that hold data from RBR CTD instruments linked with phone/tablet
devices equipped with GPS sensors.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
data(ctdRaw)
# Example 1: focus on downcast
plot(ctdTrim(ctdRaw))
# Example 2: user-supplied function.
trimByIndex&lt;-function(data, parameters) {
    parameters[1] &lt; data$scan &amp; data$scan &lt; parameters[2]
}
trimmed &lt;- ctdTrim(ctdRaw, trimByIndex, parameters=c(130, 380))
plot(trimmed)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>

<p>The Seabird CTD instrument is described at
<code style="white-space: pre;">&#8288;http://www.seabird.com/products/spec_sheets/19plusdata.htm&#8288;</code>.
</p>
<p>Seasoft V2: SBE Data Processing, SeaBird Scientific, 05/26/2016
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>

<hr>
<h2 id='ctimeToSeconds'>Interpret a Character String as a Time Interval</h2><span id='topic+ctimeToSeconds'></span>

<h3>Description</h3>

<p>Infer a time interval from a character string in
the form MM:SS or HH:MM:SS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctimeToSeconds(ctime)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctimeToSeconds_+3A_ctime">ctime</code></td>
<td>
<p>a character string (see &ldquo;Details&rdquo;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value, the number of seconds represented by the string.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+secondsToCtime">secondsToCtime()</a></code>, the inverse of this.
</p>
<p>Other things related to time: 
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+numberAsHMS">numberAsHMS</a>()</code>,
<code><a href="#topic+numberAsPOSIXct">numberAsPOSIXct</a>()</code>,
<code><a href="#topic+secondsToCtime">secondsToCtime</a>()</code>,
<code><a href="#topic+unabbreviateYear">unabbreviateYear</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
cat("10      = ", ctimeToSeconds("10"), "s\n", sep = "")
cat("01:04   = ", ctimeToSeconds("01:04"), "s\n", sep = "")
cat("1:00:00 = ", ctimeToSeconds("1:00:00"), "s\n", sep = "")
</code></pre>

<hr>
<h2 id='curl'>Curl of 2D Vector Field</h2><span id='topic+curl'></span>

<h3>Description</h3>

<p>Calculate the z component of the curl of an x-y vector field.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>curl(u, v, x, y, geographical = FALSE, method = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="curl_+3A_u">u</code></td>
<td>
<p>matrix containing the 'x' component of a vector field</p>
</td></tr>
<tr><td><code id="curl_+3A_v">v</code></td>
<td>
<p>matrix containing the 'y' component of a vector field</p>
</td></tr>
<tr><td><code id="curl_+3A_x">x</code></td>
<td>
<p>the x values for the matrices, a vector of length equal to the
number of rows in <code>u</code> and <code>v</code>.</p>
</td></tr>
<tr><td><code id="curl_+3A_y">y</code></td>
<td>
<p>the y values for the matrices, a vector of length equal to the
number of cols in <code>u</code> and <code>v</code>.</p>
</td></tr>
<tr><td><code id="curl_+3A_geographical">geographical</code></td>
<td>
<p>logical value indicating whether <code>x</code> and <code>y</code>
are longitude and latitude, in which case spherical trigonometry is used.</p>
</td></tr>
<tr><td><code id="curl_+3A_method">method</code></td>
<td>
<p>A number indicating the method to be used to calculate the
first-difference approximations to the derivatives.  See &ldquo;Details&rdquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The computed component of the curl is defined by <code class="reqn">\partial </code><code class="reqn"> v/\partial x - \partial u/\partial y</code> and the
estimate is made using first-difference approximations to the derivatives.
Two methods are provided, selected by the value of <code>method</code>.
</p>

<ul>
<li><p> For <code>method=1</code>, a centred-difference, 5-point stencil is used in
the interior of the domain.  For example, <code class="reqn">\partial v/\partial x</code>
is given by the ratio of <code class="reqn">v_{i+1,j}-v_{i-1,j}</code> to the
x extent of the grid cell at index <code class="reqn">j</code>. (The cell extents depend on
the value of <code>geographical</code>.)  Then, the edges are filled in with
nearest-neighbour values. Finally, the corners are filled in with the
adjacent value along a diagonal.  If <code>geographical=TRUE</code>, then <code>x</code>
and <code>y</code> are taken to be longitude and latitude in degrees, and the
earth shape is approximated as a sphere with radius 6371km.  The resultant
<code>x</code> and <code>y</code> are identical to the provided values, and the
resultant <code>curl</code> is a matrix with dimension identical to that of
<code>u</code>.
</p>
</li>
<li><p> For <code>method=2</code>, each interior cell in the grid is considered
individually, with derivatives calculated at the cell center. For example,
<code class="reqn">\partial v/\partial x</code> is given by the ratio of
<code class="reqn">0.5*(v_{i+1,j}+v_{i+1,j+1}) -
0.5*(v_{i,j}+v_{i,j+1})</code>
to the average of the x extent of the grid cell at indices <code class="reqn">j</code> and
<code class="reqn">j+1</code>. (The cell extents depend on the value of
<code>geographical</code>.)  The returned <code>x</code> and <code>y</code> values are the
mid-points of the supplied values. Thus, the returned <code>x</code> and <code>y</code>
are shorter than the supplied values by 1 item, and the returned <code>curl</code>
matrix dimensions are similarly reduced compared with the dimensions of
<code>u</code> and <code>v</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A list containing vectors <code>x</code> and <code>y</code>, along with matrix
<code>curl</code>.  See &ldquo;Details&rdquo; for the lengths and dimensions, for
various values of <code>method</code>.
</p>


<h3>Development status.</h3>

<p>This function is under active development as
of December 2014 and is unlikely to be stabilized until February 2015.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Chantelle Layton
</p>


<h3>See Also</h3>

<p>Other things relating to vector calculus: 
<code><a href="#topic+grad">grad</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# 1. Shear flow with uniform curl.
x &lt;- 1:4
y &lt;- 1:10
u &lt;- outer(x, y, function(x, y) y / 2)
v &lt;- outer(x, y, function(x, y) -x / 2)
C &lt;- curl(u, v, x, y, FALSE)

# 2. Rankine vortex: constant curl inside circle, zero outside
rankine &lt;- function(x, y) {
    r &lt;- sqrt(x^2 + y^2)
    theta &lt;- atan2(y, x)
    speed &lt;- ifelse(r &lt; 1, 0.5 * r, 0.5 / r)
    list(u = -speed * sin(theta), v = speed * cos(theta))
}
x &lt;- seq(-2, 2, length.out = 100)
y &lt;- seq(-2, 2, length.out = 50)
u &lt;- outer(x, y, function(x, y) rankine(x, y)$u)
v &lt;- outer(x, y, function(x, y) rankine(x, y)$v)
C &lt;- curl(u, v, x, y, FALSE)
# plot results
par(mfrow = c(2, 2))
imagep(x, y, u, zlab = "u", asp = 1)
imagep(x, y, v, zlab = "v", asp = 1)
imagep(x, y, C$curl, zlab = "curl", asp = 1)
hist(C$curl, breaks = 100)
</code></pre>

<hr>
<h2 id='d200321-001.ctd.gz'>Sample ctd File in .ctd Format</h2><span id='topic+d200321-001.ctd.gz'></span>

<h3>Description</h3>

<p>Sample ctd File in .ctd Format
</p>


<h3>See Also</h3>

<p>Other raw datasets: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>read.oce(system.file("extdata", "d200321-001.ctd.gz", package="oce"))

</code></pre>

<hr>
<h2 id='d201211_0011.cnv.gz'>Sample ctd File in .cnv Format</h2><span id='topic+d201211_0011.cnv.gz'></span>

<h3>Description</h3>

<p>Sample ctd File in .cnv Format
</p>


<h3>See Also</h3>

<p>Other raw datasets: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>read.oce(system.file("extdata", "d201211_0011.cnv.gz", package="oce"))

</code></pre>

<hr>
<h2 id='dataLabel'>Associate Data Names With Units</h2><span id='topic+dataLabel'></span>

<h3>Description</h3>

<p>Note that the whole object is not being given as an argument;
possibly this will reduce copying and thus storage impact.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dataLabel(names, units)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dataLabel_+3A_names">names</code></td>
<td>
<p>the names of data within an object</p>
</td></tr>
<tr><td><code id="dataLabel_+3A_units">units</code></td>
<td>
<p>the units from metadata</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of strings, with blank entries for data with unknown units
</p>

<hr>
<h2 id='decimate'>Smooth and Decimate, or Subsample, an oce Object</h2><span id='topic+decimate'></span>

<h3>Description</h3>

<p>Later on, other methods will be added, and <code><a href="#topic+ctdDecimate">ctdDecimate()</a></code> will be
retired in favour of this, a more general, function.  The filtering is done
with the <code><a href="stats.html#topic+filter">filter()</a></code> function of the stats package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decimate(x, by = 10, to, filter, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="decimate_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="decimate_+3A_by">by</code></td>
<td>
<p>an indication of the subsampling.  If this is a single number,
then it indicates the spacing between elements of <code>x</code> that are
selected.  If it is two numbers (a condition only applicable if <code>x</code> is
an <code>echosounder</code> object, at present), then the first number indicates
the time spacing and the second indicates the depth spacing.</p>
</td></tr>
<tr><td><code id="decimate_+3A_to">to</code></td>
<td>
<p>Indices at which to subsample.  If given, this over-rides
<code>by</code>.</p>
</td></tr>
<tr><td><code id="decimate_+3A_filter">filter</code></td>
<td>
<p>optional list of numbers representing a digital filter to be
applied to each variable in the <code>data</code> slot of <code>x</code>, before
decimation is done. If not supplied, then the decimation is done strictly by
sub-sampling.</p>
</td></tr>
<tr><td><code id="decimate_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+oce-class">oce</a> object that has been subsampled appropriately.
</p>


<h3>Bugs</h3>

<p>Only a preliminary version of this function is provided in
the present package.  It only works for objects of class <code>echosounder</code>,
for which the decimation is done after applying a running median filter and
then a boxcar filter, each of length equal to the corresponding component of
<code>by</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Filter coefficients may be calculated using
<code><a href="#topic+makeFilter">makeFilter()</a></code>.  (Note that <code><a href="#topic+ctdDecimate">ctdDecimate()</a></code> will be
retired when the present function gains equivalent functionality.)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adp)
plot(adp)
adpDec &lt;- decimate(adp, by = 2, filter = c(1 / 4, 1 / 2, 1 / 4))
plot(adpDec)
</code></pre>

<hr>
<h2 id='decodeHeaderNortek'>Decode a Nortek Header</h2><span id='topic+decodeHeaderNortek'></span>

<h3>Description</h3>

<p>Decode data in a Nortek ADV or ADP header.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decodeHeaderNortek(
  buf,
  type = c("aquadoppHR", "aquadoppProfiler", "aquadopp", "aquadoppPlusMagnetometer",
    "vector"),
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="decodeHeaderNortek_+3A_buf">buf</code></td>
<td>
<p>a &ldquo;raw&rdquo; buffer containing the header</p>
</td></tr>
<tr><td><code id="decodeHeaderNortek_+3A_type">type</code></td>
<td>
<p>type of device</p>
</td></tr>
<tr><td><code id="decodeHeaderNortek_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="decodeHeaderNortek_+3A_...">...</code></td>
<td>
<p>additional arguments, passed to called routines.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Decodes the header in a binary-format Nortek ADV/ADP file.  This function is
designed to be used by <code><a href="#topic+read.adp">read.adp()</a></code> and <code><a href="#topic+read.adv">read.adv()</a></code>,
but can be used directly as well.  The code is based on information in the
Nortek System Integrator Guide (2008) and on postings on the Nortek
&ldquo;knowledge center&rdquo; discussion board.  One might assume that the latter is
less authoritative than the former.  For example, the inference of cell size
follows advice found at
https://www.nortekusa.com/en/knowledge-center/forum/hr-profilers/736804717,
which contains a typo in an early posting that is
corrected later on.
</p>


<h3>Value</h3>

<p>A list containing elements <code>hardware</code>, <code>head</code>, <code>user</code>
and <code>offset</code>.  The easiest way to find the contents of these is to run
this function with <code>debug=3</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>


<ol>
<li><p> Information on Nortek profilers (including the System
Integrator Guide, which explains the data format byte-by-byte) is available
at <code style="white-space: pre;">&#8288;https://www.nortekusa.com/usa?set_language=usa&#8288;</code> after login.
</p>
</li>
<li><p> The Nortek Knowledge Center
https://www.nortekusa.com/en/knowledge-center
may be of help if problems arise in dealing with data from Nortek instruments.
</p>
</li>
<li><p> Nortek,
&quot;Classic Integrators Guide: Aquadopp | Aquadopp DW | Aquadopp Profiler | HQ Aquadopp Profiler | Vector | AWAC.&quot;
Nortek AS, 2022.
</p>
</li></ol>



<h3>See Also</h3>

<p>Most users should employ the functions <code><a href="#topic+read.adp">read.adp()</a></code> and
<code><a href="#topic+read.adv">read.adv()</a></code> instead of this one.
</p>

<hr>
<h2 id='decodeTime'>Oce Version of as.POSIXct</h2><span id='topic+decodeTime'></span>

<h3>Description</h3>

<p>Each format in <code>timeFormats</code> is used in turn as the <code>format</code>
argument to <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>, and the first that produces a
non-<code>NA</code> result is used.  If <code>timeFormats</code> is missing, the
following formats are tried, in the stated order:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decodeTime(time, timeFormats, tz = "UTC")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="decodeTime_+3A_time">time</code></td>
<td>
<p>Character string with an indication of the time.</p>
</td></tr>
<tr><td><code id="decodeTime_+3A_timeformats">timeFormats</code></td>
<td>
<p>Optional vector of time formats to use, as for <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>.</p>
</td></tr>
<tr><td><code id="decodeTime_+3A_tz">tz</code></td>
<td>
<p>Time zone.</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li> <p><code>"\%b \%d \%Y \%H:\%M:\%S"</code> (e.g. <code>"Jul 1 2013 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%b \%d \%Y"</code> (e.g. <code>"Jul 1 2013"</code>)
</p>
</li>
<li> <p><code>"\%B \%d \%Y \%H:\%M:\%S"</code> (e.g. <code>"July 1 2013 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%B \%d \%Y"</code> (e.g. <code>"July 1 2013"</code>)
</p>
</li>
<li> <p><code>"\%d \%b \%Y \%H:\%M:\%S"</code> (e.g. <code>"1 Jul 2013 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%d \%b \%Y"</code> (e.g. <code>"1 Jul 2013"</code>)
</p>
</li>
<li> <p><code>"\%d \%B \%Y \%H:\%M:\%S"</code> (e.g. <code>"1 July 2013 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%d \%B \%Y"</code> (e.g. <code>"1 July 2013"</code>)
</p>
</li>
<li> <p><code>"\%Y-\%m-\%d \%H:\%M:\%S"</code> (e.g.  <code>"2013-07-01 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%Y-\%m-\%d"</code> (e.g. <code>"2013-07-01"</code>)
</p>
</li>
<li> <p><code>"\%Y-\%b-\%d \%H:\%M:\%S"</code> (e.g.  <code>"2013-July-01 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%Y-\%b-\%d"</code> (e.g.  <code>"2013-Jul-01"</code>)
</p>
</li>
<li> <p><code>"\%Y-\%B-\%d \%H:\%M:\%S"</code> (e.g. <code>"2013-July-01 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%Y-\%B-\%d"</code> (e.g. <code>"2013-July-01"</code>)
</p>
</li>
<li> <p><code>"\%d-\%b-\%Y \%H:\%M:\%S"</code> (e.g.  <code>"01-Jul-2013 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%d-\%b-\%Y"</code> (e.g. <code>"01-Jul-2013"</code>)
</p>
</li>
<li> <p><code>"\%d-\%B-\%Y \%H:\%M:\%S"</code> (e.g. <code>"01-July-2013 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%d-\%B-\%Y"</code> (e.g. <code>"01-July-2013"</code>)
</p>
</li>
<li> <p><code>"\%Y/\%b/\%d \%H:\%M:\%S"</code> (e.g. <code>"2013/Jul/01 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%Y/\%b/\%d"</code> (e.g. <code>"2013/Jul/01"</code>)
</p>
</li>
<li> <p><code>"\%Y/\%B/\%d \%H:\%M:\%S"</code> (e.g. <code>"2013/July/01 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%Y/\%B/\%d"</code> (e.g. <code>"2013/July/01"</code>)
</p>
</li>
<li> <p><code>"\%Y/\%m/\%d \%H:\%M:\%S"</code> (e.g. <code>"2013/07/01 01:02:03"</code>)
</p>
</li>
<li> <p><code>"\%Y/\%m/\%d"</code> (e.g. <code>"2013/07/01"</code>)
</p>
</li></ul>



<h3>Value</h3>

<p>A time as returned by <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>decodeTime("July 1 2013 01:02:03")
decodeTime("Jul 1 2013 01:02:03")
decodeTime("1 July 2013 01:02:03")
decodeTime("1 Jul 2013 01:02:03")
decodeTime("2013-07-01 01:02:03")
decodeTime("2013/07/01 01:02:03")
decodeTime("2013/07/01")

</code></pre>

<hr>
<h2 id='defaultFlags'>Suggest a Default Flag Vector for Bad or Suspicious Data</h2><span id='topic+defaultFlags'></span>

<h3>Description</h3>

<p><code>defaultFlags</code> tries to suggest a reasonable default <code>flag</code> scheme
for use by <code><a href="#topic+handleFlags">handleFlags()</a></code>. It does this by looking for an item
named <code>flagScheme</code> in the <code>metadata</code> slot of <code>object</code>.
If <code>flagScheme</code> is found, and if the scheme is recognized, then a numeric
vector is returned that indicates bad or questionable data. If
<code>flagScheme$default</code> exists, then that scheme is returned. However,
if that does not exist, and if <code>flagScheme$name</code> is recognized,
then a pre-defined (very conservative) scheme is used,
as listed below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>defaultFlags(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="defaultFlags_+3A_object">object</code></td>
<td>
<p>An oce object</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p> for <code>argo</code>, the default is
<code>c(0,3,4,6,7,9)</code>, meaning to act upon <code>not_assessed</code> (0), <code>probably_bad</code> (3),
<code>bad</code> (4), <code>not_used_6</code> (6), <code>not_used_7</code> (7) and <code>missing</code> (9).  See Section
3.2.2 of Carval et al. (2019).
</p>
</li>
<li><p> for <code>BODC</code>, the default is
<code>c(0,2,3,4,5,6,7,8,9)</code>, i.e. all flags except <code>good</code>.
</p>
</li>
<li><p> for <code>DFO</code>, the default is
<code>c(0,2,3,4,5,8,9)</code>, i.e. all flags except <code>appears_correct</code>.
</p>
</li>
<li><p> for <code style="white-space: pre;">&#8288;WHP bottle&#8288;</code>, the default is
<code>c(1,3,4,5,6,7,8,9)</code>, i.e. all flags except <code>no_problems_noted</code>.
</p>
</li>
<li><p> for <code style="white-space: pre;">&#8288;WHP ctd&#8288;</code>, the default is
<code>c(1,3,4,5,6,7,9)</code>, i.e. all flags except <code>acceptable</code>.
</p>
</li></ul>



<h3>Value</h3>

<p>A vector of one or more flag values, or <code>NULL</code> if <code>object</code>
<code>metadata</code> slot lacks a <code>flagScheme</code> as set by <code><a href="#topic+initializeFlagScheme">initializeFlagScheme()</a></code>,
or if it has a scheme that is not in the list provide in &ldquo;Description&rdquo;.
</p>


<h3>References</h3>


<ul>
<li><p> Carval, Thierry, Bob Keeley, Yasushi Takatsuki, Takashi Yoshida, Stephen Loch Loch,
Claudia Schmid, and Roger Goldsmith. Argo User's Manual V3.3. Ifremer, 2019.
<a href="https://doi.org/10.13155/29825">doi:10.13155/29825</a>
</p>
</li></ul>



<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>

<hr>
<h2 id='despike'>Remove Spikes From a Time Series</h2><span id='topic+despike'></span>

<h3>Description</h3>

<p>The method identifies spikes with respect to a &quot;reference&quot; time-series, and
replaces these spikes with the reference value, or with <code>NA</code> according
to the value of <code>action</code>; see &ldquo;Details&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>despike(
  x,
  reference = c("median", "smooth", "trim"),
  n = 4,
  k = 7,
  min = NA,
  max = NA,
  replace = c("reference", "NA"),
  skip
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="despike_+3A_x">x</code></td>
<td>
<p>a vector of (time-series) values, a list of vectors, a data frame,
or an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="despike_+3A_reference">reference</code></td>
<td>
<p>indication of the type of reference time series to be used
in the detection of spikes; see &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="despike_+3A_n">n</code></td>
<td>
<p>an indication of the limit to differences between <code>x</code> and the
reference time series, used for <code>reference="median"</code> or
<code>reference="smooth"</code>; see &ldquo;Details.&rdquo;</p>
</td></tr>
<tr><td><code id="despike_+3A_k">k</code></td>
<td>
<p>length of running median used with <code>reference="median"</code>, and
ignored for other values of <code>reference</code>.</p>
</td></tr>
<tr><td><code id="despike_+3A_min">min</code></td>
<td>
<p>minimum non-spike value of <code>x</code>, used with
<code>reference="trim"</code>.</p>
</td></tr>
<tr><td><code id="despike_+3A_max">max</code></td>
<td>
<p>maximum non-spike value of <code>x</code>, used with
<code>reference="trim"</code>.</p>
</td></tr>
<tr><td><code id="despike_+3A_replace">replace</code></td>
<td>
<p>an indication of what to do with spike values, with
<code>"reference"</code> indicating to replace them with the reference time
series, and <code>"NA"</code> indicating to replace them with <code>NA</code>.</p>
</td></tr>
<tr><td><code id="despike_+3A_skip">skip</code></td>
<td>
<p>optional vector naming columns to be skipped. This is ignored if
<code>x</code> is a simple vector. Any items named in <code>skip</code> will be passed
through to the return value without modification.  In some cases,
<code>despike</code> will set up reasonable defaults for <code>skip</code>, e.g. for a
<code>ctd</code> object, <code>skip</code> will be set to <code>c("time", "scan",
"pressure")</code> if it is not supplied as an argument.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Three modes of operation are permitted, depending on the value of
<code>reference</code>.
</p>

<ol>
<li><p> For <code>reference="median"</code>, the first step is to linearly interpolate
across any gaps (spots where <code>x==NA</code>), using <code><a href="stats.html#topic+approx">approx()</a></code> with
<code>rule=2</code>. The second step is to pass this through
<code><a href="stats.html#topic+runmed">runmed()</a></code> to get a running median spanning <code>k</code>
elements. The result of these two steps is the &quot;reference&quot; time-series.
Then, the standard deviation of the difference between <code>x</code>
and the reference is calculated.  Any <code>x</code> values that differ from
the reference by more than <code>n</code> times this standard deviation are considered
to be spikes.  If <code>replace="reference"</code>, the spike values are
replaced with the reference, and the resultant time series is
returned.  If <code>replace="NA"</code>, the spikes are replaced with <code>NA</code>,
and that result is returned.
</p>
</li>
<li><p> For <code>reference="smooth"</code>, the processing is the same as for
<code>"median"</code>, except that <code><a href="stats.html#topic+smooth">smooth()</a></code> is used to calculate the
reference time series.
</p>
</li>
<li><p> For <code>reference="trim"</code>, the reference time series is constructed by
linear interpolation across any regions in which <code>x&lt;min</code> or
<code>x&gt;max</code>.  (Again, this is done with <code><a href="stats.html#topic+approx">approx()</a></code> with
<code>rule=2</code>.) In this case, the value of <code>n</code> is ignored, and the
return value is the same as <code>x</code>, except that spikes are replaced
with the reference series (if <code>replace="reference"</code> or with
<code>NA</code>, if <code>replace="NA"</code>.
</p>
</li></ol>



<h3>Value</h3>

<p>A new vector in which spikes are replaced as described above.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n &lt;- 50
x &lt;- 1:n
y &lt;- rnorm(n = n)
y[n / 2] &lt;- 10 # 10 standard deviations
plot(x, y, type = "l")
lines(x, despike(y), col = "red")
lines(x, despike(y, reference = "smooth"), col = "darkgreen")
lines(x, despike(y, reference = "trim", min = -3, max = 3), col = "blue")
legend("topright",
    lwd = 1, col = c("black", "red", "darkgreen", "blue"),
    legend = c("raw", "median", "smooth", "trim")
)

# add a spike to a CTD object
data(ctd)
plot(ctd)
T &lt;- ctd[["temperature"]]
T[10] &lt;- T[10] + 10
ctd[["temperature"]] &lt;- T
CTD &lt;- despike(ctd)
plot(CTD)
</code></pre>

<hr>
<h2 id='detrend'>Detrend a Set of Observations</h2><span id='topic+detrend'></span>

<h3>Description</h3>

<p>Detrends <code>y</code> by subtracting a linear trend in <code>x</code>, to create
a vector that is zero for its first and last finite value.
If the second parameter (<code>y</code>) is missing, then <code>x</code> is
taken to be <code>y</code>, and a new <code>x</code> is constructed with
<code><a href="base.html#topic+seq_along">seq_along()</a></code>.  Any <code>NA</code> values are left as-is.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>detrend(x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="detrend_+3A_x">x</code></td>
<td>
<p>a vector of numerical values.  If <code>y</code> is not given, then
<code>x</code> is taken for <code>y</code>.</p>
</td></tr>
<tr><td><code id="detrend_+3A_y">y</code></td>
<td>
<p>an optional vector</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A common application is to bring the end points of a time series
down to zero, prior to applying a digital filter. (See examples.)
</p>


<h3>Value</h3>

<p>A list containing <code>Y</code>, the detrended version of <code>y</code>, and
the intercept <code>a</code> and slope <code>b</code> of the linear function of <code>x</code>
that is subtracted from <code>y</code> to yield <code>Y</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- seq(0, 0.9 * pi, length.out = 50)
y &lt;- sin(x)
y[1] &lt;- NA
y[10] &lt;- NA
plot(x, y, ylim = c(0, 1))
d &lt;- detrend(x, y)
points(x, d$Y, pch = 20)
abline(d$a, d$b, col = "blue")
abline(h = 0)
points(x, d$Y + d$a + d$b * x, col = "blue", pch = "+")
</code></pre>

<hr>
<h2 id='download.amsr'>Download and Cache an amsr File</h2><span id='topic+download.amsr'></span>

<h3>Description</h3>

<p>If the file is already present in <code>destdir</code>, then it is not
downloaded again. The default <code>destdir</code> is the present directory,
but it probably makes more sense to use something like <code>"~/data/amsr"</code>
to make it easy for scripts in other directories to use the cached data.
The file is downloaded with <code><a href="utils.html#topic+download.file">download.file()</a></code>.  Please read the
&lsquo;History&rsquo; section for important details on how <code><a href="#topic+download.amsr">download.amsr()</a></code>
and also <code><a href="#topic+read.amsr">read.amsr()</a></code> have had be altered over the years, to deal
with changes in the directory structure and file format on the
server from which files are downloaded.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download.amsr(
  year = NULL,
  month,
  day,
  destdir = ".",
  server = "https://data.remss.com/amsr2/ocean/L3/v08.2",
  type = "3day",
  debug = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download.amsr_+3A_year">year</code>, <code id="download.amsr_+3A_month">month</code>, <code id="download.amsr_+3A_day">day</code></td>
<td>
<p>a specification of the desired observation time.
There are 3 choices for this specification.  (a) If <code>year</code> is an object
created by <code><a href="base.html#topic+as.Date">as.Date()</a></code>, then that specifies the time, and so <code>month</code>
and <code>day</code> are ignored.  This scheme can be convenient for creating a
sequence of images, starting at a particular date, because adding 1
to an object of class <code>Date</code> increases the time by 1 day, saving
the user from having to know how many days are in any given month.
(b) If <code>year</code> is an integer, then it is taken to be the year, and
the user must also specify <code>month</code> and <code>day</code>, also integers. (c)
If <code>year</code> is NULL (which is the default), then the focus is set to
the most recent date, but this depends on the value of
<code>type</code> (see next).  If <code>type</code> is <code>"3day"</code>, <code>"daily"</code> or <code>"weekly"</code>,
or just the first two of them if <code>type</code> is <code>"monthly"</code>.  If these
things are provided, then they just match exactly the values in the
sought-after file on the remote server.  If <code>year</code> is NULL, then
<code><a href="#topic+download.amsr">download.amsr()</a></code> constructs a URL that ought to be the most recent
available file: 3 days prior to the present date (if <code>type</code> is
<code>"3day"</code> or <code>"daily"</code>), the Saturday two weeks prior to the
present date (if <code>type</code> is <code>"weekly"</code>), or two months in the
past (if <code>type</code> is <code>"monthly"</code>).</p>
</td></tr>
<tr><td><code id="download.amsr_+3A_destdir">destdir</code></td>
<td>
<p>A string naming the directory in which to cache the downloaded file.
The default is to store in the present directory, but many users find it more
helpful to use something like <code>"~/data/amsr"</code> for this, to collect all
downloaded amsr files in one place.</p>
</td></tr>
<tr><td><code id="download.amsr_+3A_server">server</code></td>
<td>
<p>A string naming the server from which data
are to be acquired. See &ldquo;History&rdquo;.</p>
</td></tr>
<tr><td><code id="download.amsr_+3A_type">type</code></td>
<td>
<p>character value indicating where to get the data.  This may be
<code>"3day"</code> (the default), for a composite covering 3 days of observation, which
removes most viewing-path and cloud blanks, <code>"daily"</code> for a daily reading,
<code>"weekly"</code> for a composite covering a week, or <code>"monthly"</code> for a composite
covering a month.  In the <code>"daily"</code> case, the data arrays are 3D, with the
third dimension representing ascending and descending traces, but in all the
other cases, the arrays are 2D.</p>
</td></tr>
<tr><td><code id="download.amsr_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>download.amsr</code> returns a character value holding the full pathname
of the downloaded file.
</p>


<h3>History</h3>

<p>Until 25 March 2017, the default server was
<code>"ftp.ssmi.com/amsr2/bmaps_v07.2"</code>, but this was changed when the author
discovered that this FTP site had been changed to require users to create
accounts to register for downloads.  The default was changed to
<code>"http://data.remss.com/amsr2/bmaps_v07.2"</code> on the named date.
This site was found by a web search, but it seems to provide proper data.
It is assumed that users will do some checking on the best source.
</p>
<p>On 23 January 2018, it was noticed that the server-url naming convention
had changed, e.g.
<code style="white-space: pre;">&#8288;http://data.remss.com/amsr2/bmaps_v07.2/y2017/m01/f34_20170114v7.2.gz&#8288;</code>
becoming
<code style="white-space: pre;">&#8288;http://data.remss.com/amsr2/bmaps_v08/y2017/m01/f34_20170114v8.gz&#8288;</code>
</p>
<p>On 26 July 2023, it was noticed that the server-url naming convention
had changed again, requiring not only the alteration of the default
<code>server</code> value but also the addition of a new parameter named <code>type</code>.
Worse yet &ndash; much worse &ndash; the file format is now changed from a gzipped
format to a NetCDF format, and this will require a complete rewriting
of <code><a href="#topic+read.amsr">read.amsr()</a></code>.
</p>


<h3>Sample of Usage</h3>

<pre>
# The download may take up to about a minute.
f &lt;- download.amsr(2023, 7, 27, destdir="~/data/amsr")
d &lt;- read.amsr(f)
plot(d)
mtext(d[["filename"]], side=3, line=0, adj=0)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that download files: 
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+download.met">download.met</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to amsr data: 
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+read.amsr">read.amsr</a>()</code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+summary+2Camsr-method">summary,amsr-method</a></code>
</p>

<hr>
<h2 id='download.coastline'>Download a coastline File</h2><span id='topic+download.coastline'></span>

<h3>Description</h3>

<p>Constructs a query to the NaturalEarth server (see reference 1) to download coastline
data (or lake data, river data, etc) in any of three resolutions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download.coastline(
  resolution,
  item = "coastline",
  destdir = ".",
  destfile,
  server = "naturalearth",
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download.coastline_+3A_resolution">resolution</code></td>
<td>
<p>A character value specifying the desired resolution. The permitted
choices are <code>"10m"</code> (for 1:10M resolution, the most detailed),
<code>"50m"</code> (for 1:50M resolution)
and <code>"110m"</code> (for 1:110M resolution). If <code>resolution</code> is not supplied,
<code>"50m"</code> will be used.</p>
</td></tr>
<tr><td><code id="download.coastline_+3A_item">item</code></td>
<td>
<p>A character value indicating the quantity to be downloaded.
This is normally one of <code>"coastline"</code>, <code>"land"</code>, <code>"ocean"</code>,
<code>"rivers_lakes_centerlines"</code>, or <code>"lakes"</code>, but the NaturalEarth
server has other types, and advanced users can discover their names by inspecting
the URLs of links on the NaturalEarth site, and use them for <code>item</code>.
If <code>item</code> is not supplied, it defaults to <code>"coastline"</code>.</p>
</td></tr>
<tr><td><code id="download.coastline_+3A_destdir">destdir</code></td>
<td>
<p>Optional string indicating the directory in which to store
downloaded files. If not supplied, <code>"."</code> is used, i.e. the data file
is stored in the present working directory.</p>
</td></tr>
<tr><td><code id="download.coastline_+3A_destfile">destfile</code></td>
<td>
<p>Optional string indicating the name of the file.
If not supplied, the file name is constructed from the other
parameters of the function call, so subsequent calls with the same
parameters will yield the same result, thus providing the key
to the caching scheme.</p>
</td></tr>
<tr><td><code id="download.coastline_+3A_server">server</code></td>
<td>
<p>A character value specifying the server that is to supply
the data. At the moment, the only permitted value is <code>"naturalearth"</code>,
which is the default if <code>server</code> is not supplied.</p>
</td></tr>
<tr><td><code id="download.coastline_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character value indicating the filename of the result; if
there is a problem of any kind, the result will be the empty
string.
</p>


<h3>Non-Executable Examples</h3>

<pre>
library(oce)
# User must create directory ~/data/coastline first.
# As of September 2016, the downloaded file, named
# "ne_50m_coastline.zip", occupies 443K bytes.
filename &lt;- download.coastline(destdir="~/data/coastline")
coastline &lt;- read.coastline(filename)
plot(coastline)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> The NaturalEarth server is at <code style="white-space: pre;">&#8288;https://www.naturalearthdata.com&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>The work is done with <code><a href="utils.html#topic+download.file">utils::download.file()</a></code>.
</p>
<p>Other functions that download files: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+download.met">download.met</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>
</p>
<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>

<hr>
<h2 id='download.met'>Download and Cache a met File</h2><span id='topic+download.met'></span>

<h3>Description</h3>

<p><code><a href="#topic+download.met">download.met()</a></code> attempts to download data from Environment Canada's
historical-data website, and to cache the files locally. Lacking a published
API, this function must rely on reverse-engineering of queries handled by
that web server.  For that reason, it is brittle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download.met(
  id,
  year,
  month,
  deltat,
  type = "xml",
  destdir = ".",
  destfile,
  force = FALSE,
  quiet = FALSE,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download.met_+3A_id">id</code></td>
<td>
<p>A number giving the &quot;Station ID&quot; of the station of interest. If not
provided, <code>id</code> defaults to 6358, for Halifax International Airport. See
&ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="download.met_+3A_year">year</code></td>
<td>
<p>A number giving the year of interest. Ignored unless <code>deltat</code>
is <code>"hour"</code>. If <code>year</code> is not given, it defaults to the present year.</p>
</td></tr>
<tr><td><code id="download.met_+3A_month">month</code></td>
<td>
<p>A number giving the month of interest. Ignored unless <code>deltat</code>
is <code>"hour"</code>. If <code>month</code> is not given, it defaults to the present
month.</p>
</td></tr>
<tr><td><code id="download.met_+3A_deltat">deltat</code></td>
<td>
<p>Optional character string indicating the time step of the
desired dataset. This may be <code>"hour"</code> or <code>"month"</code>.
If <code>deltat</code> is not given, it defaults to <code>"hour"</code>.</p>
</td></tr>
<tr><td><code id="download.met_+3A_type">type</code></td>
<td>
<p>String indicating which type of file to download, either
<code>"xml"</code> (the default) for an XML file or <code>"csv"</code> for a CSV file.</p>
</td></tr>
<tr><td><code id="download.met_+3A_destdir">destdir</code></td>
<td>
<p>Optional string indicating the directory in which to store
downloaded files. If not supplied, <code>"."</code> is used, i.e. the data file
is stored in the present working directory.</p>
</td></tr>
<tr><td><code id="download.met_+3A_destfile">destfile</code></td>
<td>
<p>Optional string indicating the name of the file.
If not supplied, the file name is constructed from the other
parameters of the function call, so subsequent calls with the same
parameters will yield the same result, thus providing the key
to the caching scheme.</p>
</td></tr>
<tr><td><code id="download.met_+3A_force">force</code></td>
<td>
<p>Logical value indicating whether to force a download, even
if the file already exists locally.</p>
</td></tr>
<tr><td><code id="download.met_+3A_quiet">quiet</code></td>
<td>
<p>Logical value passed to <code><a href="utils.html#topic+download.file">download.file()</a></code>; a <code>TRUE</code> value
silences output.</p>
</td></tr>
<tr><td><code id="download.met_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If this function fails, users might try using Gavin Simpson's <code>canadaHCD</code>
package (reference 2). This package maintains a copy of the Environment
Canada listing of stations, and its <code>find_station()</code> function provides an easy
way to determine Station IDs.  After that, its <code>hcd_hourly</code> function (and
related functions) make it easy to read data. These data can then be
converted to the <code>met</code> class with <code><a href="#topic+as.met">as.met()</a></code>, although doing so leaves many
important metadata blank.
</p>


<h3>Value</h3>

<p>String indicating the full pathname to the downloaded file.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
# Download data for Halifax International Airport, in September
# of 2003. This dataset is used for data(met) provided with oce.
# Note that requests for data after 2012 month 10 yield all
# missing values, for reasons unknown to the author.
metFile &lt;- download.met(6358, 2003, 9, destdir=".")
met &lt;- read.met(metFile)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Environment Canada website for Historical Climate Data
<code style="white-space: pre;">&#8288;https://climate.weather.gc.ca/index_e.html&#8288;</code>
</p>
</li>
<li><p> Gavin Simpson's <code>canadaHCD</code> package on GitHub
<code style="white-space: pre;">&#8288;https://github.com/gavinsimpson/canadaHCD&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>The work is done with <code><a href="utils.html#topic+download.file">utils::download.file()</a></code>.
</p>
<p>Other functions that download files: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>
</p>
<p>Other things related to met data: 
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic+as.met">as.met</a>()</code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+read.met">read.met</a>()</code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+summary+2Cmet-method">summary,met-method</a></code>
</p>

<hr>
<h2 id='download.topo'>Download and Cache a topo File</h2><span id='topic+download.topo'></span>

<h3>Description</h3>

<p>Topographic data are downloaded from a data server that holds the
ETOPO1 dataset (Amante, C. and B.W. Eakins, 2009), and saved as a
netCDF file whose name specifies the data request, if a file of
that name is not already present on the local file system.  The
return value is the name of the data file, and its typical use is
as the filename for a call to <code><a href="#topic+read.topo">read.topo()</a></code>.  Given the rules on
file naming, subsequent calls to <code>download.topo</code> with identical
parameters will simply return the name of the cached file, assuming
the user has not deleted it in the meantime.  Note that
<code>download.topo</code> uses the <code>"terra"</code> and <code>"ncdf4"</code> packages,
so an error is reported if they are not available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download.topo(
  west,
  east,
  south,
  north,
  resolution = 4,
  destdir = ".",
  destfile,
  format,
  server = "https://gis.ngdc.noaa.gov",
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download.topo_+3A_west">west</code>, <code id="download.topo_+3A_east">east</code></td>
<td>
<p>numeric values for the limits of the data-selection box, in degrees.
These are converted to the -180 to 180 degree notation, if needed.
Then, <code>west</code> is rounded down to the nearest 1/100th degree, and <code>east</code>
is rounded up to the the nearest 1/100th degree. The results of these
operations are used in constructing the query for the NOAA data server.</p>
</td></tr>
<tr><td><code id="download.topo_+3A_south">south</code>, <code id="download.topo_+3A_north">north</code></td>
<td>
<p>latitude limits, treated in a way that
corresponds to the longitude limits.</p>
</td></tr>
<tr><td><code id="download.topo_+3A_resolution">resolution</code></td>
<td>
<p>numeric value of grid spacing, in geographical minutes.
The default value is 4 minutes, corresponding to 4 nautical miles (approx. 7.4km)
in the north-south direction, and less in the east-west direction.</p>
</td></tr>
<tr><td><code id="download.topo_+3A_destdir">destdir</code></td>
<td>
<p>Optional string indicating the directory in which to store
downloaded files. If not supplied, <code>"."</code> is used, i.e. the data file
is stored in the present working directory.</p>
</td></tr>
<tr><td><code id="download.topo_+3A_destfile">destfile</code></td>
<td>
<p>Optional string indicating the name of the file.
If not supplied, the file name is constructed from the other
parameters of the function call, so subsequent calls with the same
parameters will yield the same result, thus providing the key
to the caching scheme.</p>
</td></tr>
<tr><td><code id="download.topo_+3A_format">format</code></td>
<td>
<p>Deprecated, and ignored, as of June 2020.</p>
</td></tr>
<tr><td><code id="download.topo_+3A_server">server</code></td>
<td>
<p>character value specifying the base from which a
download URL will be constructed.  It is unlikely that any value
other than the default will work, unless it is a similarly-constructed
mirrored site.</p>
</td></tr>
<tr><td><code id="download.topo_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The specified longitude and latitude limits are rounded to 2 digits
(corresponding to a footprint of approximately 1km), and these are
used in the server request. If the resultant request would generate
under 1 row or column in the result, <code>download.topo</code> generates an
error message and stops.
</p>


<h3>Value</h3>

<p>String indicating the full pathname to the downloaded file.
</p>


<h3>Historical note relating to NOAA server changes</h3>

<p>2022 November 13: updated to new NOAA database, with 1/4-minute resolution (a
marked improvement over the previous 1-minute resolution).  The revision was
framed along similar changes to <code>marmap::getNOAAbathy()</code> made earlier today.
Thanks to Clark Richards for pointing this out!
</p>
<p>2020 May 31: updated for a change in the NOAA query structure, taking
hints from <code>marmap::getNOAAbathy()</code>.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
topoFile &lt;- download.topo(west=-66, east=-60, south=43, north=47,
    resolution=1, destdir="~/data/topo")
topo &lt;- read.topo(topoFile)
imagep(topo, zlim=c(-400, 400), col=oceColorsTwo, drawTriangles=TRUE)
if (requireNamespace("ocedata", quietly=TRUE)) {
    data(coastlineWorldFine, package="ocedata")
    lines(coastlineWorldFine[["longitude"]], coastlineWorldFine[["latitude"]])
}
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ul>
<li><p> Amante, C. and B.W. Eakins, 2009. ETOPO1 1 Arc-Minute Global Relief
Model: Procedures, Data Sources and Analysis. NOAA Technical Memorandum
NESDIS NGDC-24. National Geophysical Data Center, NOAA. doi:10.7289/V5C8276M
</p>
</li></ul>



<h3>See Also</h3>

<p>Other functions that download files: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+download.met">download.met</a>()</code>
</p>
<p>Other things related to topo data: 
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic+as.topo">as.topo</a>()</code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+read.topo">read.topo</a>()</code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+summary+2Ctopo-method">summary,topo-method</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+topoInterpolate">topoInterpolate</a>()</code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>
</p>

<hr>
<h2 id='drawDirectionField'>Draw a Direction Field</h2><span id='topic+drawDirectionField'></span>

<h3>Description</h3>

<p>The direction field is indicated variously, depending on the value of
<code>type</code>:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drawDirectionField(
  x,
  y,
  u,
  v,
  scalex,
  scaley,
  skip,
  length = 0.05,
  add = FALSE,
  type = 1,
  col = par("fg"),
  pch = 1,
  cex = par("cex"),
  lwd = par("lwd"),
  lty = par("lty"),
  xlab = "",
  ylab = "",
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drawDirectionField_+3A_x">x</code>, <code id="drawDirectionField_+3A_y">y</code></td>
<td>
<p>coordinates at which velocities are specified. The
length of <code>x</code> and <code>y</code> depends on the form of <code>u</code>
and <code>v</code> (vectors or matrices).</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_u">u</code>, <code id="drawDirectionField_+3A_v">v</code></td>
<td>
<p>velocity components in the x and y directions. Can be
either vectors with the same length as <code style="white-space: pre;">&#8288;x, y&#8288;</code>, or
matrices, of dimension <code>length(x)</code> by <code>length(y)</code>.</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_scalex">scalex</code>, <code id="drawDirectionField_+3A_scaley">scaley</code></td>
<td>
<p>scale to be used for the velocity arrows.
Exactly one of these must be specified.  Arrows that have
<code>u^2+v^2=1</code> will have length <code>scalex</code> along the x
axis, or <code>scaley</code> along the y axis, according to which
argument is given.</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_skip">skip</code></td>
<td>
<p>either an integer, or a two-element vector indicating
the number of points to skip when plotting arrows (for the
matrix <code style="white-space: pre;">&#8288;u, v&#8288;</code> case). If a single value, the same
<code>skip</code> is applied to both the <code>x</code> and <code>y</code>
directions. If a two-element vector, specifies different values
for the <code>x</code> and <code>y</code> directions.</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_length">length</code></td>
<td>
<p>indication of <em>width</em> of arrowheads. The
somewhat confusing name of this argument is a consequence of
the fact that it is passed to <code><a href="graphics.html#topic+arrows">arrows()</a></code> for drawing
arrows.  Note that the present default is smaller than the
default used by <code><a href="graphics.html#topic+arrows">arrows()</a></code>.</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_add">add</code></td>
<td>
<p>if <code>TRUE</code>, the arrows are added to an existing
plot; otherwise, a new plot is started by calling
<code><a href="graphics.html#topic+plot">plot()</a></code> with <code>x</code>, <code>y</code> and <code>type="n"</code>.
In other words, the plot will be very basic. In most cases, the
user will probably want to draw a diagram first, and <code>add</code>
the direction field later.</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_type">type</code></td>
<td>
<p>indication of the style of arrow-like indication of the
direction.</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_col">col</code></td>
<td>
<p>color of line segments or arrows; see <code><a href="graphics.html#topic+par">par()</a></code> for meaning</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_pch">pch</code>, <code id="drawDirectionField_+3A_cex">cex</code></td>
<td>
<p>plot character and expansion factor, used for
<code>type=1</code>; see <code><a href="graphics.html#topic+par">par()</a></code> for meanings</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_lwd">lwd</code>, <code id="drawDirectionField_+3A_lty">lty</code></td>
<td>
<p>line width and type, used for <code>type=2</code>; see <code><a href="graphics.html#topic+par">par()</a></code> for meaning</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_xlab">xlab</code>, <code id="drawDirectionField_+3A_ylab">ylab</code></td>
<td>
<p><code>x</code> and <code>y</code> axis labels</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_debug">debug</code></td>
<td>
<p>debugging value; set to a positive integer to get
debugging information.</p>
</td></tr>
<tr><td><code id="drawDirectionField_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to plotting functions
(e.g. axis labels, etc).</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p> For <code>type=1</code>, each indicator is drawn with a symbol, according to the
value of <code>pch</code> (either supplied globally, or as an element of the
<code>...</code> list) and of size <code>cex</code>, and color <code>col</code>.   Then, a
line segment is drawn for each, and for this <code>lwd</code> and <code>col</code> may
be set globally or in the <code>...</code> list.
</p>
</li>
<li><p> For <code>type=2</code>, the points are not drawn, but arrows are drawn instead
of the line segments.  Again, <code>lwd</code> and <code>col</code> control the type of
the line.
</p>
</li></ul>



<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
plot(c(-1.5, 1.5), c(-1.5, 1.5), xlab = "", ylab = "", type = "n")
drawDirectionField(x = rep(0, 2), y = rep(0, 2),
    u = c(1, 1), v = c(1, -1), scalex = 0.5, add = TRUE)
plot(c(-1.5, 1.5), c(-1.5, 1.5), xlab = "", ylab = "", type = "n")
drawDirectionField(x = rep(0, 2), y = rep(0, 2),
    u = c(1, 1), v = c(1, -1), scalex = 0.5, add = TRUE, type = 2)

# 2D example
x &lt;- seq(-2, 2, 0.1)
y &lt;- x
xx &lt;- expand.grid(x, y)[, 1]
yy &lt;- expand.grid(x, y)[, 2]
z &lt;- matrix(xx * exp(-xx^2 - yy^2), nrow = length(x))
gz &lt;- grad(z, x, y)
drawDirectionField(x, y, gz$gx, gz$gy, scalex = 0.5, type = 2, len = 0.02)
oceContour(x, y, z, add = TRUE)

</code></pre>

<hr>
<h2 id='drawIsopycnals'>Add Isopycnal Curves to a TS Plot</h2><span id='topic+drawIsopycnals'></span>

<h3>Description</h3>

<p>Adds isopycnal lines to an existing temperature-salinity plot.  This is
called by <code><a href="#topic+plotTS">plotTS()</a></code>, and may be called by the user also, e.g. if
an image plot is used to show TS data density.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drawIsopycnals(
  nlevels = 6,
  levels,
  rotate = TRUE,
  rho1000 = FALSE,
  digits = 2,
  eos = getOption("oceEOS", default = "gsw"),
  longitude = NULL,
  latitude = NULL,
  trimIsopycnals = TRUE,
  gridIsopycnals = c(50, 50),
  cex = 0.75 * par("cex"),
  col = "darkgray",
  lwd = par("lwd"),
  lty = par("lty"),
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drawIsopycnals_+3A_nlevels">nlevels</code></td>
<td>
<p>suggested number of density levels (i.e. isopycnal curves);
ignored if <code>levels</code> is supplied.  If this is set to 0, no isopycnal
are drawn (see also <code>levels</code>, next).</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_levels">levels</code></td>
<td>
<p>optional density levels to draw.  If this is <code>NULL</code>, then
no isopycnals are drawn.</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_rotate">rotate</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to write all density labels
horizontally.</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_rho1000">rho1000</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to write isopycnal labels as e.g.
1024 instead of 24.</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_digits">digits</code></td>
<td>
<p>minimum number of decimal digits to use in label (supplied to
<code><a href="base.html#topic+round">round()</a></code>). If the density range is very small, <code><a href="#topic+drawIsopycnals">drawIsopycnals()</a></code>
will increase value of <code>digits</code>, to try to make labels be distinct.</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_eos">eos</code></td>
<td>
<p>equation of state to be used, either <code>"unesco"</code> or
<code>"gsw"</code>.  If it is <code>"gsw"</code> then <code>latitude</code> and <code>longitude</code> must
be supplied, since these are needed to computer density in that
formulation.</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_longitude">longitude</code>, <code id="drawIsopycnals_+3A_latitude">latitude</code></td>
<td>
<p>numerical values giving the location
to be used in density calculations, if <code>eos</code> is <code>"gsw"</code>.</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_trimisopycnals">trimIsopycnals</code></td>
<td>
<p>logical value (<code>TRUE</code> by default) that
indicates whether to trim isopycnal curves (if drawn)
to the region of temperature-salinity space for which density
computations are considered to be valid in the context of the
chosen <code>eos</code>; see the &ldquo;Details&rdquo; of the documentation
for <code><a href="#topic+plotTS">plotTS()</a></code>.</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_gridisopycnals">gridIsopycnals</code></td>
<td>
<p>a parameter that controls how the isopycnals
are computed.  This may be NULL, or an integer vector of length 2.
<em>Case 1:</em> if <code>gridIsopycnals</code> is NULL, then
the isopycnals are drawn by tracing density
isopleths in salinity-temperature space. This method was
used as the default prior to version 1.7-11, but it was
found to yield staircase-like isopycnal curves for highly
zoomed-in plots (e.g. with millidegree temperature ranges).
<em>Case 2 (the new default):</em> If <code>gridIsopycnals</code> is a two-element integer
vector, then a grid of density is constructed, with <code>gridIsopycnals[1]</code>
salinity levels and <code>gridIsopycnals[2]</code> temperature levels, and
then <code><a href="grDevices.html#topic+contourLines">contourLines()</a></code> is used to trace the isopycnals. The default
value of <code>gridIsopycnals</code> yields a grid of millimeter-scale spacing
for a typical plot.</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_cex">cex</code></td>
<td>
<p>size for labels.</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_col">col</code></td>
<td>
<p>color for lines and labels.</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_lwd">lwd</code></td>
<td>
<p>line width for isopycnal curves</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_lty">lty</code></td>
<td>
<p>line type for isopycnal curves</p>
</td></tr>
<tr><td><code id="drawIsopycnals_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default method of drawing isopycnals was changed in February of 2023,
so that even plots that are zoomed in to have millidegree temperature ranges
will have smooth curves.  See the discussion of <code>gridIsopycnals</code> for
details.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ul>
<li><p> Fofonoff, N. P., and R. C. Millard.
&quot;Algorithms for Computation of Fundamental Properties of Seawater.&quot;
UNESCO Technical Papers in Marine Research. SCOR working group on Evaluation of CTD data;
UNESCO/ICES/SCOR/IAPSO Joint Panel on Oceanographic Tables and Standards, 1983.
<code style="white-space: pre;">&#8288;https://unesdoc.unesco.org/ark:/48223/pf0000059832&#8288;</code>.
</p>
</li>
<li><p> McDougall, Trevor J., David R. Jackett, Daniel G. Wright, and Rainer Feistel.
&quot;Accurate and Computationally Efficient Algorithms for Potential Temperature and Density of Seawater.&quot;
Journal of Atmospheric and Oceanic Technology 20, no. 5 (May 1, 2003): 730-41.
<code>https://journals.ametsoc.org/jtech/article/20/5/730/2543/Accurate-and-Computationally-Efficient-Algorithms</code>.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+plotTS">plotTS()</a></code>, which calls this.
</p>

<hr>
<h2 id='drawPalette'>Draw a Palette, Leaving Margins Suitable for an Accompanying Plot</h2><span id='topic+drawPalette'></span>

<h3>Description</h3>

<p>In the normal use, <code><a href="#topic+drawPalette">drawPalette()</a></code> draws an image palette near the
right-hand side of the plotting device, and then adjusts the global margin
settings in such a way as to cause the next plot to appear (with much larger
width) to the left of the palette. The function can also be used, if
<code>zlim</code> is not provided, to adjust the margin without drawing anything;
this is useful in lining up the x axes of a stack of plots, some some of
which will have palettes and others not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drawPalette(
  zlim,
  zlab = "",
  breaks,
  col,
  colormap,
  mai,
  cex = par("cex"),
  pos = 4,
  las = 0,
  labels = NULL,
  at = NULL,
  levels,
  drawContours = FALSE,
  plot = TRUE,
  fullpage = FALSE,
  drawTriangles = FALSE,
  axisPalette,
  tformat,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drawPalette_+3A_zlim">zlim</code></td>
<td>
<p>two-element vector containing the lower and upper limits of z.
This may also be a vector of any length exceeding 1, in which case its range
is used.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_zlab">zlab</code></td>
<td>
<p>label for the palette scale.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_breaks">breaks</code></td>
<td>
<p>optional numeric vector of the z values for breaks in the color scheme.
If <code>colormap</code> is provided, it takes precedence over <code>breaks</code> and <code>col</code>.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_col">col</code></td>
<td>
<p>optional argument, either a vector of colors corresponding to the breaks, of length
1 less than the number of breaks, or a function specifying colors.
If <code>col</code> is not provided, and if <code>colormap</code> is also not provided, then <code>col</code>
defaults to <code><a href="#topic+oceColorsViridis">oceColorsViridis()</a></code>.
If <code>colormap</code> is provided, it takes precedence over <code>breaks</code> and <code>col</code>.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_colormap">colormap</code></td>
<td>
<p>an optional color map as created by <code><a href="#topic+colormap">colormap()</a></code>.
If <code>colormap</code> is provided, it takes precedence over <code>breaks</code> and <code>col</code>.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_mai">mai</code></td>
<td>
<p>margins for palette, as defined in the usual way; see
<code><a href="graphics.html#topic+par">par()</a></code>.  If not given, reasonable values are inferred from the
existence of a non-blank <code>zlab</code>.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_cex">cex</code></td>
<td>
<p>numeric character expansion value for text labels</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_pos">pos</code></td>
<td>
<p>an integer indicating the location of the palette within the
plotting area, 1 for near the bottom, 2 for near the left-hand side, 3 for
near the top side, and 4 (the default) for near the right-hand side.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_las">las</code></td>
<td>
<p>optional argument, passed to <code><a href="graphics.html#topic+axis">axis()</a></code>, to control the orientation
of numbers along the axis. As explained in the help for <code><a href="graphics.html#topic+par">par()</a></code>, the
meaning of <code>las</code> is as follows: <code>las=0</code> (the default) means to put labels
parallel to the axis, <code>las=1</code> means horizontal (regardless of
axis orientation), <code>las=2</code> means perpendicular to the axis,
and <code>las=3</code> means to vertical (regardless of axis orientation).  Note that
the automatic computation of margin spacing parameter <code>mai</code>
assumes that <code>las=0</code>, and so for other cases, the user may need to
specify the <code>mai</code> argument directly.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_labels">labels</code></td>
<td>
<p>optional vector of labels for ticks on palette axis (must
correspond with <code>at</code>)</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_at">at</code></td>
<td>
<p>optional vector of positions for the <code>label</code>s</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_levels">levels</code></td>
<td>
<p>optional contour levels, in preference to <code>breaks</code>
values, to be added to the image if <code>drawContours</code> is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_drawcontours">drawContours</code></td>
<td>
<p>logical value indicating whether to draw contours on the
palette, at the color breaks.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_plot">plot</code></td>
<td>
<p>logical value indicating whether to plot the palette, the
default, or whether to just alter the margins to make space for where the
palette would have gone.  The latter case may be useful in lining up plots,
as in example 1 of &ldquo;Examples&rdquo;.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_fullpage">fullpage</code></td>
<td>
<p>logical value indicating whether to draw the palette filling
the whole plot width (apart from <code>mai</code>, of course).  This can be
helpful if the palette panel is to be created with <code><a href="graphics.html#topic+layout">layout()</a></code>, as
illustrated in the &ldquo;Examples&rdquo;.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_drawtriangles">drawTriangles</code></td>
<td>
<p>logical value indicating whether to draw triangles on
the top and bottom of the palette.  If a single value is provided, it applies
to both ends of the palette.  If a pair is provided, the first refers to the
lower range of the palette, and the second to the upper range.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_axispalette">axisPalette</code></td>
<td>
<p>optional replacement function for <code>axis()</code>, e.g.
for exponential notation on large or small values.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_tformat">tformat</code></td>
<td>
<p>optional format for axis labels, if the variable is a time
type (ignored otherwise).</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="drawPalette_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot positioning is done entirely with margins, not with
<code>par(mfrow)</code> or other R schemes for multi-panel plots.  This means that
the user is free to use those schemes without worrying about nesting or
conflicts.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>Use with multi-panel plots</h3>

<p>An important consequence of the margin
adjustment is that multi-panel plots require that the initial margin be
stored prior to the first call to <code><a href="#topic+drawPalette">drawPalette()</a></code>, and reset after each
palette-plot pair.  This method is illustrated in &ldquo;Examples&rdquo;.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, with help from Clark Richards
</p>


<h3>See Also</h3>

<p>This is used by <code><a href="#topic+imagep">imagep()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
par(mgp = getOption("oceMgp"))

# 1. A three-panel plot
par(mfrow = c(3, 1), mar = c(3, 3, 1, 1))
omar &lt;- par("mar") # save initial margin

# 1a. top panel: simple case with Viridis scheme
drawPalette(zlim = c(0, 1), col = oce.colorsViridis(10))
plot(1:10, 1:10, col = oce.colorsViridis(10)[1:10], pch = 20, cex = 3, xlab = "x", ylab = "y")
par(mar = omar) # reset margin

# 1b. middle panel: colormap
cm &lt;- colormap(name = "gmt_globe")
drawPalette(colormap = cm)
icol &lt;- seq_along(cm$col)
plot(icol, cm$breaks[icol],
    pch = 20, cex = 2, col = cm$col,
    xlab = "Palette index", ylab = "Palette breaks"
)
par(mar = omar) # reset margin

# 1c. bottom panel: space for palette (to line up graphs)
drawPalette(plot = FALSE)
plot(1:10, 1:10, col = oce.colorsViridis(10)[1:10], pch = 20, cex = 3, xlab = "x", ylab = "y")
par(mar = omar) # reset margin

# 2. Use layout to mimic the action of imagep(), with the width
# of the palette region being 14 percent of figure width.
d &lt;- 0.14
layout(matrix(1:2, nrow = 1), widths = c(1 - d, d))
image(volcano, col = oce.colorsViridis(100), zlim = c(90, 200))
contour(volcano, add = TRUE)
drawPalette(c(90, 200), fullpage = TRUE, col = oce.colorsViridis)
</code></pre>

<hr>
<h2 id='echosounder'>Sample echosounder Data</h2><span id='topic+echosounder'></span>

<h3>Description</h3>

<p>This is degraded subsample of measurements that were made with a Biosonics
scientific echosounder, as part of the St Lawrence Internal Wave Experiment
(SLEIWEX).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Source</h3>

<p>This file came from the SLEIWEX-2008 experiment, and was decimated
using <code><a href="#topic+decimate">decimate()</a></code> with <code>by=c()</code>.
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to echosounder data: 
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic+as.echosounder">as.echosounder</a>()</code>,
<code><a href="#topic+echosounder-class">echosounder-class</a></code>,
<code><a href="#topic+findBottom">findBottom</a>()</code>,
<code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method</a></code>,
<code><a href="#topic+read.echosounder">read.echosounder</a>()</code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+summary+2Cechosounder-method">summary,echosounder-method</a></code>
</p>

<hr>
<h2 id='echosounder-class'>Class to Store Echosounder Data</h2><span id='topic+echosounder-class'></span>

<h3>Description</h3>

<p>This class stores echosounder data. Echosounder objects may be
read with <code><a href="#topic+read.echosounder">read.echosounder()</a></code>,
summarized with <code><a href="#topic+summary+2Cechosounder-method">summary,echosounder-method()</a></code>,
and plotted with <code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method()</a></code>.
The <code><a href="#topic+findBottom">findBottom()</a></code>
function infers the ocean bottom from tracing the strongest reflector from
ping to ping.
</p>


<h3>Details</h3>


<ul>
<li><p> An infrequently updated record of the instrument position, in
<code>timeSlow</code>, <code>longitudeSlow</code> and <code>latitudeSlow</code>.  These are
used in plotting maps with <code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method()</a></code>.
</p>
</li>
<li><p> An interpolated record of the instrument position, in <code>time</code>,
<code>longitude</code>, and <code>latitude</code>.  Linear interpolation is used to
infer the longitude and latitude from the variables listed above.
</p>
</li>
<li> <p><code>depth</code>, vector of depths of echo samples (measured positive
downwards in the water column).  This is calculated from the inter-sample
time interval and the sound speed provided as the <code>soundSpeed</code> argument
to <code><a href="#topic+read.echosounder">read.echosounder()</a></code>, so altering the value of the latter will
alter the echosounder plots provided by <code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method()</a></code>.
</p>
</li>
<li><p> The echosounder signal amplitude <code>a</code>, a matrix whose number of
rows matches the length of <code>time</code>, etc., and number of columns equal to
the length of <code>depth</code>.  Thus, for example, <code>a[100,]</code> represents
the depth-dependent amplitude at the time of the 100th ping.
</p>
</li>
<li><p> A matrix named <code>b</code> exists for dual-beam and split-beam cases.
For dual-beam data, this is the wide-beam data, whereas <code>a</code> is the
narrow-beam data.  For split-beam data, this is the x-angle data.
</p>
</li>
<li><p> A matrix named <code>c</code> exists for split-beam data, containing the
y-angle data.
</p>
</li>
<li><p> In addition to these matrices, ad-hoc calculated matrices named
<code>Sv</code> and <code>TS</code> may be accessed as explained in the next section.
</p>
</li></ul>



<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>echosounder</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>echosounder</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>echosounder</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+echosounder-class">echosounder</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+echosounder-class">echosounder</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to echosounder data: 
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic+as.echosounder">as.echosounder</a>()</code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+findBottom">findBottom</a>()</code>,
<code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method</a></code>,
<code><a href="#topic+read.echosounder">read.echosounder</a>()</code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+summary+2Cechosounder-method">summary,echosounder-method</a></code>
</p>

<hr>
<h2 id='eclipticalToEquatorial'>Convert Ecliptical Coordinate to Equatorial Coordinate</h2><span id='topic+eclipticalToEquatorial'></span>

<h3>Description</h3>

<p>Convert from ecliptical to equatorial coordinates, using
equations 8.3 and 8.4 of reference 1, or, equivalently,
equations 12.3 and 12.4 of reference 2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eclipticalToEquatorial(lambda, beta, epsilon)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eclipticalToEquatorial_+3A_lambda">lambda</code></td>
<td>
<p>longitude, in degrees, or a data frame containing
<code>lambda</code>, <code>beta</code>, and <code>epsilon</code>, in which case the next to
arguments are ignored</p>
</td></tr>
<tr><td><code id="eclipticalToEquatorial_+3A_beta">beta</code></td>
<td>
<p>geocentric latitude, in degrees</p>
</td></tr>
<tr><td><code id="eclipticalToEquatorial_+3A_epsilon">epsilon</code></td>
<td>
<p>obliquity of the ecliptic, in degrees</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The code is based on reference 1; see <code><a href="#topic+moonAngle">moonAngle()</a></code> for comments on
the differences in formulae found in reference 2.  Indeed,
reference 2 is only cited
here in case readers want to check the ideas of the formulae; DK
has found that reference 2 is available to him via his university library
inter-library loan system, whereas he owns a copy of reference 1.
</p>


<h3>Value</h3>

<p>A data frame containing columns <code>rightAscension</code> and
<code>declination</code> both in degrees.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, based on formulae in references 1 and 2.
</p>


<h3>References</h3>


<ul>
<li><p> Meeus, Jean. Astronomical Formulas for Calculators. Second Edition.
Richmond, Virginia, USA: Willmann-Bell, 1982.
</p>
</li>
<li><p> Meeus, Jean. Astronomical Algorithms. Second Edition.
Richmond, Virginia, USA: Willmann-Bell, 1991.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other things related to astronomy: 
<code><a href="#topic+angle2hms">angle2hms</a>()</code>,
<code><a href="#topic+equatorialToLocalHorizontal">equatorialToLocalHorizontal</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+moonAngle">moonAngle</a>()</code>,
<code><a href="#topic+siderealTime">siderealTime</a>()</code>,
<code><a href="#topic+sunAngle">sunAngle</a>()</code>,
<code><a href="#topic+sunDeclinationRightAscension">sunDeclinationRightAscension</a>()</code>
</p>

<hr>
<h2 id='enuToOther'>Rotate Acoustic-Doppler Data to a New Coordinate System</h2><span id='topic+enuToOther'></span>

<h3>Description</h3>

<p>Rotate Acoustic-Doppler Data to a New Coordinate System
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enuToOther(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="enuToOther_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> or <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="enuToOther_+3A_...">...</code></td>
<td>
<p>extra arguments that are passed on to <code><a href="#topic+enuToOtherAdp">enuToOtherAdp()</a></code>
or <code><a href="#topic+enuToOtherAdv">enuToOtherAdv()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same class as <code>x</code>, but with velocities
in the rotated coordinate system
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='enuToOtherAdp'>Convert adp Object from ENU Coordinate to Rotated Coordinate</h2><span id='topic+enuToOtherAdp'></span>

<h3>Description</h3>

<p>Convert ADP velocity components from an enu-based coordinate system to
another system, perhaps to align axes with the coastline.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enuToOtherAdp(x, heading = 0, pitch = 0, roll = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="enuToOtherAdp_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="enuToOtherAdp_+3A_heading">heading</code></td>
<td>
<p>number or vector of numbers, giving the angle, in degrees, to
be added to the heading.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="enuToOtherAdp_+3A_pitch">pitch</code></td>
<td>
<p>as <code>heading</code> but for pitch.</p>
</td></tr>
<tr><td><code id="enuToOtherAdp_+3A_roll">roll</code></td>
<td>
<p>as <code>heading</code> but for roll.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The supplied angles specify rotations to be made around the axes for which
heading, pitch, and roll are defined.  For example, an eastward current will
point southeast if <code>heading=45</code> is used.
</p>
<p>The returned value has heading, pitch, and roll matching those of <code>x</code>,
so these angles retain their meaning as the instrument orientation.
</p>
<p>NOTE: this function works similarly to <code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp()</a></code>, except
that in the present function, it makes no difference whether the instrument
points up or down, etc.
</p>


<h3>Value</h3>

<p>An object with <code>data$v[,1:3,]</code> altered appropriately, and
<code>metadata$oce.coordinate</code> changed from <code>enu</code> to <code>other</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Teledyne RD Instruments. &ldquo;ADCP Coordinate Transformation: Formulas and
Calculations,&rdquo; January 2010. P/N 951-6079-00.
</p>
</li></ol>



<h3>See Also</h3>

<p>See <code><a href="#topic+read.adp">read.adp()</a></code> for other functions that relate to
objects of class <code>"adp"</code>.
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(adp)
o &lt;- enuToOtherAdp(adp, heading = -31.5)
plot(o, which = 1:3)

</code></pre>

<hr>
<h2 id='enuToOtherAdv'>Convert ENU to Other Coordinate</h2><span id='topic+enuToOtherAdv'></span>

<h3>Description</h3>

<p>Convert ADV velocity components from an enu-based coordinate system to
another system, perhaps to align axes with the coastline.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enuToOtherAdv(
  x,
  heading = 0,
  pitch = 0,
  roll = 0,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="enuToOtherAdv_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="enuToOtherAdv_+3A_heading">heading</code></td>
<td>
<p>number or vector of numbers, giving the angle, in degrees, to
be added to the heading. If this has length less than the number of velocity
sampling times, then it will be extended using <code><a href="base.html#topic+rep">rep()</a></code>.</p>
</td></tr>
<tr><td><code id="enuToOtherAdv_+3A_pitch">pitch</code></td>
<td>
<p>as <code>heading</code> but for pitch.</p>
</td></tr>
<tr><td><code id="enuToOtherAdv_+3A_roll">roll</code></td>
<td>
<p>as <code>heading</code> but for roll.</p>
</td></tr>
<tr><td><code id="enuToOtherAdv_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The supplied angles specify rotations to be made around the axes for which
heading, pitch, and roll are defined.  For example, an eastward current will
point southeast if <code>heading=45</code> is used.
</p>
<p>The returned value has heading, pitch, and roll matching those of <code>x</code>,
so these angles retain their meaning as the instrument orientation.
</p>
<p>NOTE: this function works similarly to <code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv()</a></code>, except
that in the present function, it makes no difference whether the instrument
points up or down, etc.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='equatorialToLocalHorizontal'>Convert Equatorial Coordinate to Local Horizontal Coordinate</h2><span id='topic+equatorialToLocalHorizontal'></span>

<h3>Description</h3>

<p>Convert from equatorial coordinates to local horizontal coordinates, i.e.
azimuth and altitude.
The method is taken from equations 8.5 and 8.6 of reference 1, or, equivalently,
from equations 12.5 and 12.6 of reference 2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>equatorialToLocalHorizontal(
  rightAscension,
  declination,
  t,
  longitude,
  latitude
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="equatorialToLocalHorizontal_+3A_rightascension">rightAscension</code></td>
<td>
<p>right ascension, e.g. calculated with
<code><a href="#topic+eclipticalToEquatorial">eclipticalToEquatorial()</a></code>.</p>
</td></tr>
<tr><td><code id="equatorialToLocalHorizontal_+3A_declination">declination</code></td>
<td>
<p>declination, e.g. calculated with
<code><a href="#topic+eclipticalToEquatorial">eclipticalToEquatorial()</a></code>.</p>
</td></tr>
<tr><td><code id="equatorialToLocalHorizontal_+3A_t">t</code></td>
<td>
<p>time of observation.</p>
</td></tr>
<tr><td><code id="equatorialToLocalHorizontal_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation, positive in eastern hemisphere.</p>
</td></tr>
<tr><td><code id="equatorialToLocalHorizontal_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation, positive in northern hemisphere.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing columns <code>altitude</code> (angle above
horizon, in degrees) and <code>azimuth</code> (angle anticlockwise from south, in
degrees).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, based on formulae in references 1 and 2.
</p>


<h3>References</h3>


<ul>
<li><p> Meeus, Jean. Astronomical Formulas for Calculators. Second Edition.
Richmond, Virginia, USA: Willmann-Bell, 1982.
</p>
</li>
<li><p> Meeus, Jean. Astronomical Algorithms. Second Edition.
Richmond, Virginia, USA: Willmann-Bell, 1991.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other things related to astronomy: 
<code><a href="#topic+angle2hms">angle2hms</a>()</code>,
<code><a href="#topic+eclipticalToEquatorial">eclipticalToEquatorial</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+moonAngle">moonAngle</a>()</code>,
<code><a href="#topic+siderealTime">siderealTime</a>()</code>,
<code><a href="#topic+sunAngle">sunAngle</a>()</code>,
<code><a href="#topic+sunDeclinationRightAscension">sunDeclinationRightAscension</a>()</code>
</p>

<hr>
<h2 id='errorbars'>Draw Error Bars on an Existing xy Diagram</h2><span id='topic+errorbars'></span>

<h3>Description</h3>

<p>Draw Error Bars on an Existing xy Diagram
</p>


<h3>Usage</h3>

<pre><code class='language-R'>errorbars(x, y, xe, ye, percent = FALSE, style = 0, length = 0.025, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="errorbars_+3A_x">x</code>, <code id="errorbars_+3A_y">y</code></td>
<td>
<p>coordinates of points on the existing plot.</p>
</td></tr>
<tr><td><code id="errorbars_+3A_xe">xe</code>, <code id="errorbars_+3A_ye">ye</code></td>
<td>
<p>errors on x and y coordinates of points on the existing plot,
each either a single number or a vector of length identical to that of
the corresponding coordinate.</p>
</td></tr>
<tr><td><code id="errorbars_+3A_percent">percent</code></td>
<td>
<p>boolean flag indicating whether <code>xe</code> and <code>ye</code> are
in terms of percent of the corresponding <code>x</code> and <code>y</code> values.</p>
</td></tr>
<tr><td><code id="errorbars_+3A_style">style</code></td>
<td>
<p>indication of the style of error bar.  Using <code>style=0</code>
yields simple line segments (drawn with <code><a href="graphics.html#topic+segments">segments()</a></code>) and
<code>style=1</code> yields line segments with short perpendicular endcaps.</p>
</td></tr>
<tr><td><code id="errorbars_+3A_length">length</code></td>
<td>
<p>length of endcaps, for <code>style=1</code> only; it is passed to
<code><a href="graphics.html#topic+arrows">arrows()</a></code>, which is used to draw that style of error bars.</p>
</td></tr>
<tr><td><code id="errorbars_+3A_...">...</code></td>
<td>
<p>graphical parameters passed to the code that produces the error
bars, e.g. to <code><a href="graphics.html#topic+segments">segments()</a></code> for <code>style=0</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
S &lt;- ctd[["salinity"]]
T &lt;- ctd[["temperature"]]
plot(S, T)
errorbars(S, T, 0.05, 0.5)
</code></pre>

<hr>
<h2 id='fillGap'>Fill a Gap in an oce Object</h2><span id='topic+fillGap'></span>

<h3>Description</h3>

<p>Sequences of <code>NA</code> values, are filled by linear interpolation between
the non-<code>NA</code> values that bound the gap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fillGap(x, method = c("linear"), rule = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fillGap_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="fillGap_+3A_method">method</code></td>
<td>
<p>to use; see &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="fillGap_+3A_rule">rule</code></td>
<td>
<p>integer controlling behaviour at start and end of <code>x</code>.  If
<code>rule=1</code>, <code>NA</code> values at the ends are left in the return value.
If <code>rule=2</code>, they are replaced with the nearest non-NA point.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>oce</code> object, with gaps removed.
</p>


<h3>Bugs</h3>


<ol>
<li><p> Eventually, this will be expanded to work
with any <code>oce</code> object.  But, for now, it only works for vectors that
can be coerced to numeric.
</p>
</li>
<li><p> If the first or last point is <code>NA</code>, then <code>x</code> is returned unaltered.
</p>
</li>
<li><p> Only method <code>linear</code> is permitted now.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Integers
x &lt;- c(1:2, NA, NA, 5:6)
y &lt;- fillGap(x)
print(data.frame(x, y))
# Floats
x &lt;- x + 0.1
y &lt;- fillGap(x)
print(data.frame(x, y))
</code></pre>

<hr>
<h2 id='findBottom'>Find the Ocean Bottom in an Echosounder Object</h2><span id='topic+findBottom'></span>

<h3>Description</h3>

<p>Finds the depth in a Biosonics echosounder file, by finding the strongest
reflector and smoothing its trace.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findBottom(x, ignore = 5, clean = despike)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findBottom_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+echosounder-class">echosounder</a> object.</p>
</td></tr>
<tr><td><code id="findBottom_+3A_ignore">ignore</code></td>
<td>
<p>number of metres of data to ignore, near the surface.</p>
</td></tr>
<tr><td><code id="findBottom_+3A_clean">clean</code></td>
<td>
<p>a function to clean the inferred depth of spikes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with elements: the <code>time</code> of a ping, the <code>depth</code> of
the inferred depth in metres, and the <code>index</code> of the inferred bottom
location, referenced to the object's <code>depth</code> vector.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for <a href="#topic+echosounder-class">echosounder</a> explains the
structure of <code>echosounder</code> objects, and also outlines the other
functions dealing with them.
</p>
<p>Other things related to echosounder data: 
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic+as.echosounder">as.echosounder</a>()</code>,
<code><a href="#topic+echosounder-class">echosounder-class</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method</a></code>,
<code><a href="#topic+read.echosounder">read.echosounder</a>()</code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+summary+2Cechosounder-method">summary,echosounder-method</a></code>
</p>

<hr>
<h2 id='firstFinite'>Get First Finite Value in a Vector or Array.</h2><span id='topic+firstFinite'></span>

<h3>Description</h3>

<p>If x is a vector, this is straightforward.  If x is
anything else, it is first converted to a vector with
<code><a href="base.html#topic+as.vector">as.vector()</a></code>, so the first value will be with
respect to storage by columns, for a matrix, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>firstFinite(v)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="firstFinite_+3A_v">v</code></td>
<td>
<p>A numerical vector or array.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The first finite value, or NULL if there are no
finite values.
</p>

<hr>
<h2 id='formatCI'>Format a Confidence Interval</h2><span id='topic+formatCI'></span>

<h3>Description</h3>

<p>This formats a confidence interval in either the +/- notation or
the parenthetic notation.  For example, if a quantity has mean 1
with uncertainty 0.05, which means a CI of 0.95 to 1.05,
the <code>"+-"</code> style yields <code>"1+/-0.05"</code>, and the <code>"parentheses"</code>
style yields &lsquo;&quot;&quot;&rsquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>formatCI(
  ci,
  style = c("+/-", "parentheses"),
  model,
  digits = 2,
  debug = getOption("oceDebug", 0)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="formatCI_+3A_ci">ci</code></td>
<td>
<p>optional vector of length 2 or 3.</p>
</td></tr>
<tr><td><code id="formatCI_+3A_style">style</code></td>
<td>
<p>string indicating notation to be used.</p>
</td></tr>
<tr><td><code id="formatCI_+3A_model">model</code></td>
<td>
<p>optional regression model, e.g. returned by <code><a href="stats.html#topic+lm">lm()</a></code> or
<code><a href="stats.html#topic+nls">nls()</a></code>.</p>
</td></tr>
<tr><td><code id="formatCI_+3A_digits">digits</code></td>
<td>
<p>optional number of digits to use.  This is ignored
if <code>style</code> is <code>"parentheses"</code>.</p>
</td></tr>
<tr><td><code id="formatCI_+3A_debug">debug</code></td>
<td>
<p>integer value indicating debugging level. If 0, then
<code><a href="#topic+formatCI">formatCI()</a></code> works silently.  If greater than 0, then some debugging
messages are printed during processing.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If a <code>model</code> is given, then <code>ci</code> is ignored, and a confidence
interval is calculated using <code><a href="stats.html#topic+confint">confint()</a></code> with <code>level</code> set to
0.6914619.  This <code>level</code> corresponds to a range of plus or minus one
standard deviation, for the t distribution and a large number of degrees of
freedom (since <code>qt(0.6914619, 100000)</code> is 0.5).
</p>
<p>If <code>model</code> is missing, <code>ci</code> must be provided.  If it contains 3
elements, then first and third elements are taken as the range of the
confidence interval (which by convention should use the <code>level</code> stated
in the previous paragraph), and the second element is taken as the central
value.  Alternatively, if <code>ci</code> has 2 elements, they are taken to be
bounds of the confidence interval and their mean is taken to be the central
value.
</p>
<p>In the <code style="white-space: pre;">&#8288;+/-&#8288;</code> notation, e.g. <code class="reqn">a \pm b</code> indicates that the true
value lies between <code class="reqn">a-b</code> and <code class="reqn">a+b</code> with a high degree of
certainty.  Mills et al. (1993, section 4.1 on page 83) suggest that
<code class="reqn">b</code> should be set equal to 2 times the standard uncertainty or
standard deviation.  JCGM (2008, section 7.2.2 on pages 25 and 26), however,
suggest that <code class="reqn">b</code> should be set to the standard uncertainty, while
also recommending that the <code class="reqn">\pm</code> notation (and presumably
the parentheses notation also) be avoided altogether, in favour of
writing sentences that explains uncertainties in clear terms.
</p>
<p>The <code>parentheses</code> notation is often called the compact notation.  In
it, the digits in parentheses indicate the uncertainty in the corresponding
digits to their left, e.g. 12.34(3) means that the last digit (4) has an
uncertainty of 3.  However, as with the <code class="reqn">\pm</code> notation, different
authorities offer different advice on defining this uncertainty; Mills et
al. (1993) provide an example in which the
parenthetic value is half the <code class="reqn">\pm</code> value, whereas
JCM (2008) suggest using the same values.
</p>
<p>The JCM(2008) convention is used by <code><a href="#topic+formatCI">formatCI()</a></code> for the parentheses
notation, as illustrated in Examples 1 and 2.  Note, however, that
if the confidence range exceeds the value, then
a request for <code>parentheses</code> format reverts to <code style="white-space: pre;">&#8288;+/-&#8288;</code> format.
</p>


<h3>Value</h3>

<p>If <code>ci</code> is given, the result is a character string with the
estimate and its uncertainty, in plus/minus or parenthetic notation.  If
<code>model</code> is given, the result is a 1-column matrix holding character
strings, with row names corresponding to the parameters of the model.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> JCGM, 2008. <em>Evaluation of measurement data - Guide to the
expression of uncertainty in measurement (JCGM 100:2008)</em>, published by the
Joint Committee for Guides in Metrology, available (as of November 2023)
at https://www.bipm.org/documents/20126/2071204/JCGM_100_2008_E.pdf.
See section 7.2.2 on Page 25, for a summary of notation, including
an illustration of the use of equal values for both the <code style="white-space: pre;">&#8288;+-&#8288;</code> and the
parentheses notations.
</p>
</li>
<li><p> Mills, I., T. Cvitas, K. Homann, N. Kallay, and K. Kuchitsu, 1993.
<em>Quantities, Units and Symbols in Physical Chemistry</em>, published
Blackwell Science for the International Union of Pure and Applied Chemistry.
(See section 4.1, page 83, for a summary of notation, which shows that a
value to the right of a <code style="white-space: pre;">&#8288;+-&#8288;</code> sign is to be halved if put in
in parentheses, which is not done in the present function, because
of a choice to follow the recommendation of reference 1.
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1: mean=1, uncertainty=0.05, in +/- notation.
formatCI(c(0.95, 1.05)) # "1+/-0.05"

# Example 2: save mean and uncertainty, but in parentheses notation.
formatCI(c(0.95, 1.05), style = "parentheses") # "1.00(5)"

# example 3: using t.test to find a CI.
a &lt;- rnorm(100, mean = 10, sd = 1)
CI &lt;- t.test(a)$conf.int
formatCI(CI)
formatCI(CI, style = "parentheses")

# example 4: specifying a model
x &lt;- seq(0, 10, 0.1)
y &lt;- 2 + 3 * x + rnorm(x, sd = 0.1)
m &lt;- lm(y ~ x)
formatCI(model = m)
formatCI(model = m, style = "parentheses")

</code></pre>

<hr>
<h2 id='formatPosition'>Format Geographical Position in Degrees and Minutes</h2><span id='topic+formatPosition'></span>

<h3>Description</h3>

<p>Format geographical positions to degrees, minutes, and hemispheres
</p>


<h3>Usage</h3>

<pre><code class='language-R'>formatPosition(
  latlon,
  isLat = TRUE,
  type = c("list", "string", "expression"),
  showHemi = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="formatPosition_+3A_latlon">latlon</code></td>
<td>
<p>a vector of latitudes or longitudes</p>
</td></tr>
<tr><td><code id="formatPosition_+3A_islat">isLat</code></td>
<td>
<p>a boolean that indicates whether the quantity is latitude or
longitude</p>
</td></tr>
<tr><td><code id="formatPosition_+3A_type">type</code></td>
<td>
<p>a string indicating the type of return value (see below)</p>
</td></tr>
<tr><td><code id="formatPosition_+3A_showhemi">showHemi</code></td>
<td>
<p>a boolean that indicates whether to indicate the hemisphere</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing <code>degrees</code>, <code>minutes</code>, <code>seconds</code>,
and <code>hemispheres</code>, or a vector of strings or (broken) a vector of
expressions.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions related to maps: 
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
formatPosition(10 + 1:10 / 60 + 2.8 / 3600)
formatPosition(10 + 1:10 / 60 + 2.8 / 3600, type = "string")

</code></pre>

<hr>
<h2 id='fullFilename'>Full Name of File, Including Path</h2><span id='topic+fullFilename'></span>

<h3>Description</h3>

<p>Determines the full name of a file, including the path.  Used by many
<code>read.X</code> routines, where <code>X</code> is the name of a class of object.
This is a wrapper around <code><a href="base.html#topic+normalizePath">normalizePath()</a></code>, with warnings turned
off so that messages are not printed for files that are
not found (e.g. URLs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fullFilename(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fullFilename_+3A_filename">filename</code></td>
<td>
<p>name of file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Full file name
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>

<hr>
<h2 id='g1sst-class'>Class to Store G1SST Satellite/Model Data</h2><span id='topic+g1sst-class'></span>

<h3>Description</h3>

<p>This class stores G1SST model-satellite products.
</p>


<h3>Details</h3>

<p>G1SST is an acronym for global 1-km sea surface temperature, a product
that combines satellite data with the model output. It is provided by
the JPO ROMS (Regional Ocean Modelling System) modelling group.
See the JPL website (reference 1) to learn more about the data, and see
the <code><a href="#topic+read.g1sst">read.g1sst()</a></code> documentation for an example
of downloading and plotting.
</p>
<p>It is important not to regard G1SST data in the same category as,
say, <a href="#topic+amsr-class">amsr</a> data, because the two products
differ greatly with respect to cloud cover. The satellite used by
<a href="#topic+amsr-class">amsr</a> has the ability to sense water temperature
even if there is cloud cover, whereas <code>g1sst</code> fills in cloud
gaps with model simulations.  It can be helpful to consult
reference 1 for a given time, clicking and then unclicking the radio button
that turns off the model-based filling of cloud gaps.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>g1sst</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>g1sst</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>g1sst</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+g1sst-class">g1sst</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+g1sst-class">g1sst</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> JPO OurOcean Portal <code style="white-space: pre;">&#8288;https://ourocean.jpl.nasa.gov/SST/&#8288;</code>
(link worked in 2016 but was seen to fail 2017 Feb 2).
</p>
</li></ol>



<h3>See Also</h3>

<p>Other classes holding satellite data: 
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+landsat-class">landsat-class</a></code>,
<code><a href="#topic+satellite-class">satellite-class</a></code>
</p>
<p>Other things related to g1sst data: 
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic+read.g1sst">read.g1sst</a>()</code>
</p>

<hr>
<h2 id='gappyIndex'>Create a Possibly Gappy Indexing Vector</h2><span id='topic+gappyIndex'></span>

<h3>Description</h3>

<p>This is used internally to construct indexing arrays, mainly for adv and adp
functions, in which <code><a href="base.html#topic+readBin">readBin()</a></code> is used to access isolated regions within a
<a href="base.html#topic+raw">raw</a> vector. The work is done in C++, for speed. Since this function is
designed for use within oce, it does not offer many safeguards on the
parameters, beyond detecting an overlapping situation that would occur if
<code>length</code> exceeded the space between <code>starts</code> elements.  Also, users ought
to be aware that the behaviour of <code>gappyIndex()</code> might change at any time;
simply stated, it is not intended for direct use except by the package
developers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gappyIndex(starts, offset = 0L, length = 4L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gappyIndex_+3A_starts">starts</code></td>
<td>
<p>integer vector of one or more values.</p>
</td></tr>
<tr><td><code id="gappyIndex_+3A_offset">offset</code></td>
<td>
<p>integer value indicating the value to be added
to each of the <code>starts</code> value, as the beginning of the sequence.</p>
</td></tr>
<tr><td><code id="gappyIndex_+3A_length">length</code></td>
<td>
<p>integer value indicating the number of
elements of that sequence.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For example, suppose data elements in a buffer named <code>buf</code> start at bytes
1000 and 2000, and that the goal is to skip the first 4 bytes of each of
these sequences, and then to read the next 2 bytes as an unsigned 16-bit
integer. This could be accomplished as follows.
</p>
<div class="sourceCode"><pre>library(oce)
buf &lt;- readBin("filename", "raw", n=5000, size=1)
i &lt;- gappyIndex(c(1000, 2000, 3000), 4, 2)
# i is 1004,1005, 2004,2005, 3004,3005
values &lt;- readBin(buf[i], "integer", size=2, n=3, endian="little")
</pre></div>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>

<hr>
<h2 id='geodDist'>Compute Geodesic Distance on Surface of Earth</h2><span id='topic+geodDist'></span>

<h3>Description</h3>

<p>This calculates geodesic distance, in km, between points on the earth, i.e.
distance measured along the (presumed ellipsoidal) surface. The method
involves the solution of the geodetic inverse problem, using Vincenty's
(1975) modification of Rainsford's method with Helmert's elliptical terms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geodDist(
  longitude1,
  latitude1 = NULL,
  longitude2 = NULL,
  latitude2 = NULL,
  alongPath = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geodDist_+3A_longitude1">longitude1</code></td>
<td>
<p>longitude or a vector of longitudes, <em>or</em> a
<code>section</code> object, from which longitude and latitude are extracted and
used instead of the next three arguments</p>
</td></tr>
<tr><td><code id="geodDist_+3A_latitude1">latitude1</code></td>
<td>
<p>latitude or vector of latitudes (ignored if
<code>longitude1</code> is a <code>section</code> object)</p>
</td></tr>
<tr><td><code id="geodDist_+3A_longitude2">longitude2</code></td>
<td>
<p>optional longitude or vector of longitudes (ignored if
<code>alongPath=TRUE</code>)</p>
</td></tr>
<tr><td><code id="geodDist_+3A_latitude2">latitude2</code></td>
<td>
<p>optional latitude or vector of latitudes (ignored if
<code>alongPath=TRUE</code>)</p>
</td></tr>
<tr><td><code id="geodDist_+3A_alongpath">alongPath</code></td>
<td>
<p>boolean indicating whether to compute distance along the
path, as opposed to distance from the reference point.  If
<code>alongPath=TRUE</code>, any values provided for <code>latitude2</code> and
<code>longitude2</code> will be ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function may be used in several different ways.
</p>
<p>Case 1: <code>longitude1</code> is a <code>section</code> object. The values of
<code>latitude1</code>, <code>longitude2</code>, and <code>latitude2</code> arguments are
ignored, and the behaviour depends on the value of the <code>alongPath</code>
argument.  If <code>alongPath=FALSE</code>, the return value contains the geodetic
distances of each station from the first one.  If <code>alongPath=TRUE</code>, the
return value is the geodetic distance along the path connecting the
stations, in the order in which they are stored in the section.
</p>
<p>Case 2: <code>longitude1</code> is a vector.  If <code>longitude2</code> and
<code>latitude2</code> are not given, then the return value is a vector containing
the distances of each point from the first one, <em>or</em> the distance
along the path connecting the points, according to the value of
<code>alongPath</code>.  On the other hand, if both <code>longitude2</code> and
<code>latitude2</code> are specified, then the return result depends on the length
of these arguments.  If they are each of length 1, then they are taken as a
reference point, from which the distances to <code>longitude1</code> and
<code>latitude1</code> are calculated (ignoring the value of <code>alongPath</code>).
However, if they are of the same length as <code>longitude1</code> and
<code>latitude1</code>, then the return value is the distance between
corresponding (<code>longitude1</code>,<code>latitude1</code>) and
(<code>longitude2</code>,<code>latitude2</code>) values.
</p>


<h3>Value</h3>

<p>Vector of distances in kilometres.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley based this on R code sent to him by Darren Gillis, who in
2003 had modified Fortran code that, according to comments in the source,
had been written in 1974 by L. Pfeifer and J. G. Gergen.
</p>


<h3>References</h3>

<p>Vincenty, T. &quot;Direct and Inverse Solutions of Geodesics on the
Ellipsoid with Application of Nested Equations.&quot; Survey Review 23, no. 176
(April 1, 1975): 88-93. https://doi.org/10.1179/sre.1975.23.176.88.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geodXy">geodXy()</a></code>
</p>
<p>Other functions relating to geodesy: 
<code><a href="#topic+geodGc">geodGc</a>()</code>,
<code><a href="#topic+geodXyInverse">geodXyInverse</a>()</code>,
<code><a href="#topic+geodXy">geodXy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
km &lt;- geodDist(100, 45, 100, 46)
data(section)
geodDist(section)
geodDist(section, alongPath = TRUE)

</code></pre>

<hr>
<h2 id='geodGc'>Great-circle Segments Between Points on Earth</h2><span id='topic+geodGc'></span>

<h3>Description</h3>

<p>Each pair in the <code>longitude</code> and <code>latitude</code> vectors is considered
in turn.  For long vectors, this may be slow.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geodGc(longitude, latitude, dmax)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geodGc_+3A_longitude">longitude</code></td>
<td>
<p>vector of longitudes, in degrees east</p>
</td></tr>
<tr><td><code id="geodGc_+3A_latitude">latitude</code></td>
<td>
<p>vector of latitudes, in degrees north</p>
</td></tr>
<tr><td><code id="geodGc_+3A_dmax">dmax</code></td>
<td>
<p>maximum angular separation to tolerate between sub-segments, in
degrees.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame of <code>longitude</code> and <code>latitude</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, based on code from Clark Richards, in turn based on
formulae provided by Ed Williams (see reference 1)].
</p>


<h3>References</h3>


<ol>
<li> <p><code style="white-space: pre;">&#8288;http://williams.best.vwh.net/avform.htm#Intermediate&#8288;</code>
(link worked for years but failed 2017-01-16).
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions relating to geodesy: 
<code><a href="#topic+geodDist">geodDist</a>()</code>,
<code><a href="#topic+geodXyInverse">geodXyInverse</a>()</code>,
<code><a href="#topic+geodXy">geodXy</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(coastlineWorld)
mapPlot(coastlineWorld,
    type = "l",
    longitudelim = c(-80, 10), latitudelim = c(35, 80),
    projection = "+proj=merc"
)
# Great circle from New York to Paris (Lindberg's flight)
l &lt;- geodGc(c(-73.94, 2.35), c(40.67, 48.86), 1)
mapLines(l$longitude, l$latitude, col = "red", lwd = 2)


</code></pre>

<hr>
<h2 id='geodXy'>Convert From Geographical to Geodesic Coordinates</h2><span id='topic+geodXy'></span>

<h3>Description</h3>

<p>The method, which may be useful in determining coordinate systems for a
mooring array or a ship transects, calculates (x,y) from distance calculations
along geodesic curves.  See &ldquo;Caution&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geodXy(
  longitude,
  latitude,
  longitudeRef,
  latitudeRef,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geodXy_+3A_longitude">longitude</code>, <code id="geodXy_+3A_latitude">latitude</code></td>
<td>
<p>vector of longitude and latitude</p>
</td></tr>
<tr><td><code id="geodXy_+3A_longituderef">longitudeRef</code>, <code id="geodXy_+3A_latituderef">latitudeRef</code></td>
<td>
<p>numeric reference location. Poor results
will be returned if these values are not close to the locations described
by <code>longitude</code> and <code>latitude</code>. A sensible approach might be
to set <code>longitudeRef</code> to <code>longitude[1]</code>, etc.</p>
</td></tr>
<tr><td><code id="geodXy_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The calculation is as follows.
Consider the <code>i</code>-th point in the <code>longitude</code> and <code>latitude</code>
vectors.  To calculate <code>x[i]</code>, <code><a href="#topic+geodDist">geodDist()</a></code> is
used is to find the distance
<em>along a geodesic curve</em> connecting (<code>longitude[i]</code>, <code>latitude[i]</code>) with
(<code>longitudeRef</code>, <code>latitude[i]</code>). The resultant distance
is multiplied by -1 if <code>longitude[i]-longitudeRef</code> is negative,
and the result is assigned to <code>x[i]</code>.
A similar procedure is used for <code>y[i]</code>.
</p>


<h3>Value</h3>

<p><code>geodXy</code> returns a data frame of <code>x</code> and <code>y</code>,
geodesic distance components, measured in metres.
</p>


<h3>Caution</h3>

<p>This scheme is without known precedent in the literature, and
users should read the documentation carefully before deciding to use it.
</p>


<h3>Change history</h3>

<p>On 2015-11-02, the names of the arguments were changed from <code>lon</code>, etc., to
<code>longitude</code>, etc., to be in keeping with other oce functions.
</p>
<p>On 2017-04-05, four changes were made.
</p>

<ol>
<li><p> Default values of <code>longitudeRef</code> and <code>latitudeRef</code> were removed,
since the old defaults were inappropriate to most work.
</p>
</li>
<li><p> The argument called <code>rotate</code> was eliminated, because it only made
sense if the mean resultant x and y were zero.
</p>
</li>
<li><p> The example was made more useful.
</p>
</li>
<li><p> Pointers were made to <code><a href="#topic+lonlat2utm">lonlat2utm()</a></code>, which may be more useful.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+geodDist">geodDist()</a></code>
</p>
<p>Other functions relating to geodesy: 
<code><a href="#topic+geodDist">geodDist</a>()</code>,
<code><a href="#topic+geodGc">geodGc</a>()</code>,
<code><a href="#topic+geodXyInverse">geodXyInverse</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Develop a transect-based axis system for final data(section) stations
library(oce)
data(section)
lon &lt;- tail(section[["longitude", "byStation"]], 26)
lat &lt;- tail(section[["latitude", "byStation"]], 26)
lonR &lt;- tail(lon, 1)
latR &lt;- tail(lat, 1)
data(coastlineWorld)
mapPlot(coastlineWorld,
    projection = "+proj=merc",
    longitudelim = c(-75, -65), latitudelim = c(35, 43), col = "gray"
)
mapPoints(lon, lat)
XY &lt;- geodXy(lon, lat, mean(lon), mean(lat))
angle &lt;- 180 / pi * atan(coef(lm(y ~ x, data = XY))[2])
mapCoordinateSystem(lonR, latR, 500, angle, col = 2)
# Compare UTM calculation
UTM &lt;- lonlat2utm(lon, lat, zone = 18) # we need to set the zone for this task!
angleUTM &lt;- 180 / pi * atan(coef(lm(northing ~ easting, data = UTM))[2])
mapCoordinateSystem(lonR, latR, 500, angleUTM, col = 3)
legend("topright",
    lwd = 1, col = 2:3, bg = "white", title = "Axis Rotation Angle",
    legend = c(
        sprintf("geod: %.1f deg", angle),
        sprintf("utm:  %.1f deg", angleUTM)
    )
)

</code></pre>

<hr>
<h2 id='geodXyInverse'>Inverse Geodesic Calculation</h2><span id='topic+geodXyInverse'></span>

<h3>Description</h3>

<p>The calculation is done by finding a minimum value of a cost
function that is the vector difference between (<code>x</code>,<code>y</code>)
and the corresponding values returned by <code><a href="#topic+geodXy">geodXy()</a></code>.
See &ldquo;Caution&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geodXyInverse(x, y, longitudeRef, latitudeRef, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geodXyInverse_+3A_x">x</code></td>
<td>
<p>value of x in metres, as given by <code><a href="#topic+geodXy">geodXy()</a></code></p>
</td></tr>
<tr><td><code id="geodXyInverse_+3A_y">y</code></td>
<td>
<p>value of y in metres, as given by <code><a href="#topic+geodXy">geodXy()</a></code></p>
</td></tr>
<tr><td><code id="geodXyInverse_+3A_longituderef">longitudeRef</code></td>
<td>
<p>reference longitude, as supplied to <code><a href="#topic+geodXy">geodXy()</a></code></p>
</td></tr>
<tr><td><code id="geodXyInverse_+3A_latituderef">latitudeRef</code></td>
<td>
<p>reference latitude, as supplied to <code><a href="#topic+geodXy">geodXy()</a></code></p>
</td></tr>
<tr><td><code id="geodXyInverse_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The minimum is calculated in C for speed, using the <code>nmmin</code> function
that is the underpinning for the Nelder-Meade version of the R function
<code><a href="stats.html#topic+optim">optim()</a></code>. If you find odd results, try setting <code>debug=1</code>
and rerunning, to see whether this optimizer is having difficulty
finding a minimum of the mismatch function.
</p>


<h3>Value</h3>

<p>a data frame containing <code>longitude</code> and <code>latitude</code>
</p>


<h3>Caution</h3>

<p>This scheme is without known precedent in the literature, and
users should read the documentation carefully before deciding to use it.
</p>


<h3>See Also</h3>

<p>Other functions relating to geodesy: 
<code><a href="#topic+geodDist">geodDist</a>()</code>,
<code><a href="#topic+geodGc">geodGc</a>()</code>,
<code><a href="#topic+geodXy">geodXy</a>()</code>
</p>

<hr>
<h2 id='GMTOffsetFromTz'>Determine Time Offset From Timezone</h2><span id='topic+GMTOffsetFromTz'></span>

<h3>Description</h3>

<p>The data are from
<code style="white-space: pre;">&#8288;https://www.timeanddate.com/time/zones/&#8288;</code> and were
hand-edited to develop this code, so there may be errors.  Also, note that
some of these contradict; if you examine the code, you'll see some
commented-out portions that represent solving conflicting definitions by
choosing the more common timezone abbreviation over a the less common one.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GMTOffsetFromTz(tz)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GMTOffsetFromTz_+3A_tz">tz</code></td>
<td>
<p>a timezone, e.g. <code>UTC</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Number of hours in offset, e.g. <code>AST</code> yields 4.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
cat("Atlantic Standard Time is ", GMTOffsetFromTz("AST"), "hours after UTC")
</code></pre>

<hr>
<h2 id='gps-class'>Class to Store GPS Data</h2><span id='topic+gps-class'></span>

<h3>Description</h3>

<p>This class stores GPS data. These objects may be read with
<code><a href="#topic+read.gps">read.gps()</a></code> or assembled with <code><a href="#topic+as.gps">as.gps()</a></code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>gps</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>gps</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>gps</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+gps-class">gps</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+gps-class">gps</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to gps data: 
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic+as.gps">as.gps</a>()</code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+read.gps">read.gps</a>()</code>,
<code><a href="#topic+summary+2Cgps-method">summary,gps-method</a></code>
</p>

<hr>
<h2 id='grad'>Calculate Matrix Gradient</h2><span id='topic+grad'></span>

<h3>Description</h3>

<p>In the interior of the matrix, centred second-order differences are used to
infer the components of the grad.  Along the edges, first-order differences
are used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grad(
  h,
  x = seq(0, 1, length.out = nrow(h)),
  y = seq(0, 1, length.out = ncol(h))
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grad_+3A_h">h</code></td>
<td>
<p>a matrix of values</p>
</td></tr>
<tr><td><code id="grad_+3A_x">x</code></td>
<td>
<p>vector of coordinates along matrix columns (defaults to integers)</p>
</td></tr>
<tr><td><code id="grad_+3A_y">y</code></td>
<td>
<p>vector of coordinates along matrix rows (defaults to integers)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing <code class="reqn">|\nabla h|</code> as <code>g</code>,
<code class="reqn">\partial h/\partial x</code> as <code>gx</code>,
and <code class="reqn">\partial h/\partial y</code> as <code>gy</code>,
each of which is a matrix of the same dimension as <code>h</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, based on advice of Clark Richards, and mimicking a matlab function.
</p>


<h3>See Also</h3>

<p>Other things relating to vector calculus: 
<code><a href="#topic+curl">curl</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Built-in volcano dataset
g &lt;- grad(volcano)
par(mfrow = c(2, 2), mar = c(3, 3, 1, 1), mgp = c(2, 0.7, 0))
imagep(volcano, zlab = "h")
imagep(g$g, zlab = "|grad(h)|")
zlim &lt;- c(-1, 1) * max(g$g)
imagep(g$gx, zlab = "dh/dx", zlim = zlim)
imagep(g$gy, zlab = "dh/dy", zlim = zlim)

# 2. Geostrophic flow around an eddy
library(oce)
dx &lt;- 5e3
dy &lt;- 10e3
x &lt;- seq(-200e3, 200e3, dx)
y &lt;- seq(-200e3, 200e3, dy)
R &lt;- 100e3
h &lt;- outer(x, y, function(x, y) 500 * exp(-(x^2 + y^2) / R^2))
grad &lt;- grad(h, x, y)
par(mfrow = c(2, 2), mar = c(3, 3, 1, 1), mgp = c(2, 0.7, 0))
contour(x, y, h, asp = 1, main = expression(h))
f &lt;- 1e-4
gprime &lt;- 9.8 * 1 / 1024
u &lt;- -(gprime / f) * grad$gy
v &lt;- (gprime / f) * grad$gx
contour(x, y, u, asp = 1, main = expression(u))
contour(x, y, v, asp = 1, main = expression(v))
contour(x, y, sqrt(u^2 + v^2), asp = 1, main = expression(speed))

</code></pre>

<hr>
<h2 id='gravity'>Acceleration Due to Earth Gravity</h2><span id='topic+gravity'></span>

<h3>Description</h3>

<p>Compute <code class="reqn">g</code>, the acceleration due to gravity, as a function of
latitude.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gravity(latitude = 45, degrees = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gravity_+3A_latitude">latitude</code></td>
<td>
<p>Latitude in <code class="reqn">^\circ</code>N or radians north of the
equator.</p>
</td></tr>
<tr><td><code id="gravity_+3A_degrees">degrees</code></td>
<td>
<p>Flag indicating whether degrees are used for latitude; if set
to <code>FALSE</code>, radians are used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Value not verified yet, except roughly.
</p>


<h3>Value</h3>

<p>Acceleration due to gravity, in <code class="reqn">m^2/s</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Gill, A.E., 1982. <em>Atmosphere-ocean Dynamics</em>, Academic
Press, New York, 662 pp.
</p>
<p><strong>Caution:</strong> Fofonoff and Millard (1983 UNESCO) use a different formula.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g &lt;- gravity(45) # 9.8
</code></pre>

<hr>
<h2 id='handleFlags'>Handle Flags in oce Objects (Generic)</h2><span id='topic+handleFlags'></span>

<h3>Description</h3>

<p>Data-quality flags are stored in the <code>metadata</code>
slot of <a href="#topic+oce-class">oce</a> objects in a
<a href="base.html#topic+list">list</a> named <code>flags</code>.
The present function (a generic that has specialized versions
for various data classes) provides a way to
manipulate the contents of the <code>data</code> slot, based on
such data-quality flags. For example, a common operation is to replace
erroneous data with <code>NA</code>.
</p>
<p>If the <code>flags</code> within <code>object</code>'s <code>metadata</code> slot is empty,
then <code>object</code> is returned, unaltered.
Otherwise, <code>handleFlags</code> examines <code>object@metadata$flags</code>
in the context of the <code>flags</code> argument, and then
carries out actions that are specified by the <code>actions</code> argument.
By default, this sets the returned <code>data</code> entries to <code>NA</code>,
wherever the corresponding <code>metadata$flag</code> values
signal unreliable data. To maintain a hint as to why
<code>data</code> were changed, <code>metadata$flags</code> in the
returned value is a direct copy of the corresponding
entry in <code>object</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>handleFlags(
  object = "oce",
  flags = NULL,
  actions = NULL,
  where = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="handleFlags_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="handleFlags_+3A_flags">flags</code></td>
<td>
<p>A <a href="base.html#topic+list">list</a> specifying flag values upon which
actions will be taken. This can take two forms.
</p>

<ul>
<li><p> In the first form, the
list has named elements each containing a vector of integers. For example,
salinities flagged with values of 1 or 3:9 would be specified
by <code>flags=list(salinity=c(1,3:9))</code>. Several data items can be specified,
e.g. <code>flags=list(salinity=c(1,3:9), temperature=c(1,3:9))</code> indicates
that the actions are to take place for both salinity and temperature.
</p>
</li>
<li><p> In the second form, <code>flags</code> is a list holding a single <em>unnamed</em> vector, and
this means to apply the actions to <em>all</em> the data entries.  For example,
<code>flags=list(c(1,3:9))</code> means to apply not just to salinity and temperature,
but to everything within the <code>data</code> slot.
</p>
</li></ul>

<p>If <code>flags</code> is not provided, then <code><a href="#topic+defaultFlags">defaultFlags()</a></code> is called, to try to
determine a reasonable default.</p>
</td></tr>
<tr><td><code id="handleFlags_+3A_actions">actions</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that contains items with
names that match those in the <code>flags</code> argument.  If <code>actions</code>
is not supplied, the default will be to set all values identified by
<code>flags</code> to <code>NA</code>; this can also be specified by
specifying <code>actions=list("NA")</code>. It is also possible to specify
functions that calculate replacement values. These are provided
with <code>object</code> as the single argument, and must return a
replacement for the data item in question.
See &ldquo;Details&rdquo; for the default that is used if <code>actions</code> is not supplied.</p>
</td></tr>
<tr><td><code id="handleFlags_+3A_where">where</code></td>
<td>
<p>an optional character value that permits the function to work with
objects that store flags in e.g. <code>object@metadata$flags$where</code>
instead of in <code>object@metadata$flags</code>, and data within
<code>object@data$where</code> instead of within <code>object@data</code>. The
default value of <code>NULL</code> means to look withing <code>object@metadata</code>
itself, and this is the default within <code>oce</code>.  (The purpose of <code>where</code>
is to make <code>oce</code> extensible by other packages, which may choose to store
data two levels deep in the <code>data</code> slot.)</p>
</td></tr>
<tr><td><code id="handleFlags_+3A_debug">debug</code></td>
<td>
<p>An optional integer specifying the degree of debugging, with
value 0 meaning to skip debugging and 1 or higher meaning to print some
information about the arguments and the data. It is usually a good idea to set
this to 1 for initial work with a dataset, to see which flags are being
handled for each data item. If not supplied, this defaults to the value of
<code><a href="base.html#topic+getOption">getOption</a></code><code>("oceDebug")</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each specialized variant of this function has its own defaults
for <code>flags</code> and <code>actions</code>.
</p>


<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>

<hr>
<h2 id='handleFlags+2Cadp-method'>Handle Flags in adp Objects</h2><span id='topic+handleFlags+2Cadp-method'></span>

<h3>Description</h3>

<p>Data-quality flags are stored in the <code>metadata</code>
slot of <a href="#topic+oce-class">oce</a> objects in a
<a href="base.html#topic+list">list</a> named <code>flags</code>.
The present function (a generic that has specialized versions
for various data classes) provides a way to
manipulate the contents of the <code>data</code> slot, based on
such data-quality flags. For example, a common operation is to replace
erroneous data with <code>NA</code>.
</p>
<p>If the <code>flags</code> within <code>object</code>'s <code>metadata</code> slot is empty,
then <code>object</code> is returned, unaltered.
Otherwise, <code>handleFlags</code> examines <code>object@metadata$flags</code>
in the context of the <code>flags</code> argument, and then
carries out actions that are specified by the <code>actions</code> argument.
By default, this sets the returned <code>data</code> entries to <code>NA</code>,
wherever the corresponding <code>metadata$flag</code> values
signal unreliable data. To maintain a hint as to why
<code>data</code> were changed, <code>metadata$flags</code> in the
returned value is a direct copy of the corresponding
entry in <code>object</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adp'
handleFlags(
  object = "oce",
  flags = NULL,
  actions = NULL,
  where = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="handleFlags+2B2Cadp-method_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cadp-method_+3A_flags">flags</code></td>
<td>
<p>A <a href="base.html#topic+list">list</a> specifying flag values upon which
actions will be taken. This can take two forms.
</p>

<ul>
<li><p> In the first form, the
list has named elements each containing a vector of integers. For example,
salinities flagged with values of 1 or 3:9 would be specified
by <code>flags=list(salinity=c(1,3:9))</code>. Several data items can be specified,
e.g. <code>flags=list(salinity=c(1,3:9), temperature=c(1,3:9))</code> indicates
that the actions are to take place for both salinity and temperature.
</p>
</li>
<li><p> In the second form, <code>flags</code> is a list holding a single <em>unnamed</em> vector, and
this means to apply the actions to <em>all</em> the data entries.  For example,
<code>flags=list(c(1,3:9))</code> means to apply not just to salinity and temperature,
but to everything within the <code>data</code> slot.
</p>
</li></ul>

<p>If <code>flags</code> is not provided, then <code><a href="#topic+defaultFlags">defaultFlags()</a></code> is called, to try to
determine a reasonable default.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cadp-method_+3A_actions">actions</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that contains items with
names that match those in the <code>flags</code> argument.  If <code>actions</code>
is not supplied, the default will be to set all values identified by
<code>flags</code> to <code>NA</code>; this can also be specified by
specifying <code>actions=list("NA")</code>. It is also possible to specify
functions that calculate replacement values. These are provided
with <code>object</code> as the single argument, and must return a
replacement for the data item in question.
See &ldquo;Details&rdquo; for the default that is used if <code>actions</code> is not supplied.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cadp-method_+3A_where">where</code></td>
<td>
<p>an optional character value that permits the function to work with
objects that store flags in e.g. <code>object@metadata$flags$where</code>
instead of in <code>object@metadata$flags</code>, and data within
<code>object@data$where</code> instead of within <code>object@data</code>. The
default value of <code>NULL</code> means to look withing <code>object@metadata</code>
itself, and this is the default within <code>oce</code>.  (The purpose of <code>where</code>
is to make <code>oce</code> extensible by other packages, which may choose to store
data two levels deep in the <code>data</code> slot.)</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cadp-method_+3A_debug">debug</code></td>
<td>
<p>An optional integer specifying the degree of debugging, with
value 0 meaning to skip debugging and 1 or higher meaning to print some
information about the arguments and the data. It is usually a good idea to set
this to 1 for initial work with a dataset, to see which flags are being
handled for each data item. If not supplied, this defaults to the value of
<code><a href="base.html#topic+getOption">getOption</a></code><code>("oceDebug")</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>flags</code> and <code>actions</code> are not provided, the
default is to consider a flag value of 1 to indicate bad data,
and 0 to indicate good data. Note that it only makes sense to use
velocity (<code>v</code>) flags, because other flags are, at least
for some instruments, stored as <code>raw</code> quantities, and such
quantities may not be set to <code>NA</code>.
</p>


<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Flag low "goodness" or high "error beam" values.
library(oce)
data(adp)
# Same as Example 2 of ?'setFlags,adp-method'
v &lt;- adp[["v"]]
i2 &lt;- array(FALSE, dim = dim(v))
g &lt;- adp[["g", "numeric"]]
# Set thresholds on percent "goodness" and error "velocity".
G &lt;- 25
V4 &lt;- 0.45
for (k in 1:3) {
    i2[, , k] &lt;- ((g[, , k] + g[, , 4]) &lt; G) | (v[, , 4] &gt; V4)
}
adpQC &lt;- initializeFlags(adp, "v", 2)
adpQC &lt;- setFlags(adpQC, "v", i2, 3)
adpClean &lt;- handleFlags(adpQC, flags = list(3), actions = list("NA"))
# Demonstrate (subtle) change graphically.
par(mfcol = c(2, 1))
plot(adp, which = "u1", drawTimeRange = FALSE)
plot(adpClean, which = "u1", drawTimeRange = FALSE)
t0 &lt;- 1214510000 # from locator()
arrows(t0, 20, t0, 35, length = 0.1, lwd = 3, col = "magenta")
mtext("Slight change above arrow", col = "magenta", font = 2)

</code></pre>

<hr>
<h2 id='handleFlags+2Cargo-method'>Handle Flags in argo Objects</h2><span id='topic+handleFlags+2Cargo-method'></span><span id='topic+handleFlags.argo'></span>

<h3>Description</h3>

<p>Data-quality flags are stored in the <code>metadata</code>
slot of <a href="#topic+oce-class">oce</a> objects in a
<a href="base.html#topic+list">list</a> named <code>flags</code>.
The present function (a generic that has specialized versions
for various data classes) provides a way to
manipulate the contents of the <code>data</code> slot, based on
such data-quality flags. For example, a common operation is to replace
erroneous data with <code>NA</code>.
</p>
<p>If the <code>flags</code> within <code>object</code>'s <code>metadata</code> slot is empty,
then <code>object</code> is returned, unaltered.
Otherwise, <code>handleFlags</code> examines <code>object@metadata$flags</code>
in the context of the <code>flags</code> argument, and then
carries out actions that are specified by the <code>actions</code> argument.
By default, this sets the returned <code>data</code> entries to <code>NA</code>,
wherever the corresponding <code>metadata$flag</code> values
signal unreliable data. To maintain a hint as to why
<code>data</code> were changed, <code>metadata$flags</code> in the
returned value is a direct copy of the corresponding
entry in <code>object</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'argo'
handleFlags(
  object = "oce",
  flags = NULL,
  actions = NULL,
  where = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="handleFlags+2B2Cargo-method_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+argo-class">argo</a> object.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cargo-method_+3A_flags">flags</code></td>
<td>
<p>A <a href="base.html#topic+list">list</a> specifying flag values upon which
actions will be taken. This can take two forms.
</p>

<ul>
<li><p> In the first form, the
list has named elements each containing a vector of integers. For example,
salinities flagged with values of 1 or 3:9 would be specified
by <code>flags=list(salinity=c(1,3:9))</code>. Several data items can be specified,
e.g. <code>flags=list(salinity=c(1,3:9), temperature=c(1,3:9))</code> indicates
that the actions are to take place for both salinity and temperature.
</p>
</li>
<li><p> In the second form, <code>flags</code> is a list holding a single <em>unnamed</em> vector, and
this means to apply the actions to <em>all</em> the data entries.  For example,
<code>flags=list(c(1,3:9))</code> means to apply not just to salinity and temperature,
but to everything within the <code>data</code> slot.
</p>
</li></ul>

<p>If <code>flags</code> is not provided, then <code><a href="#topic+defaultFlags">defaultFlags()</a></code> is called, to try to
determine a reasonable default.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cargo-method_+3A_actions">actions</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that contains items with
names that match those in the <code>flags</code> argument.  If <code>actions</code>
is not supplied, the default will be to set all values identified by
<code>flags</code> to <code>NA</code>; this can also be specified by
specifying <code>actions=list("NA")</code>. It is also possible to specify
functions that calculate replacement values. These are provided
with <code>object</code> as the single argument, and must return a
replacement for the data item in question.
See &ldquo;Details&rdquo; for the default that is used if <code>actions</code> is not supplied.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cargo-method_+3A_where">where</code></td>
<td>
<p>an optional character value that permits the function to work with
objects that store flags in e.g. <code>object@metadata$flags$where</code>
instead of in <code>object@metadata$flags</code>, and data within
<code>object@data$where</code> instead of within <code>object@data</code>. The
default value of <code>NULL</code> means to look withing <code>object@metadata</code>
itself, and this is the default within <code>oce</code>.  (The purpose of <code>where</code>
is to make <code>oce</code> extensible by other packages, which may choose to store
data two levels deep in the <code>data</code> slot.)</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cargo-method_+3A_debug">debug</code></td>
<td>
<p>An optional integer specifying the degree of debugging, with
value 0 meaning to skip debugging and 1 or higher meaning to print some
information about the arguments and the data. It is usually a good idea to set
this to 1 for initial work with a dataset, to see which flags are being
handled for each data item. If not supplied, this defaults to the value of
<code><a href="base.html#topic+getOption">getOption</a></code><code>("oceDebug")</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Wong, Annie, Robert Keeley, Thierry Carval, and Argo Data Management Team.
&quot;Argo Quality Control Manual for CTD and Trajectory Data,&quot; January 1, 2020.
<code style="white-space: pre;">&#8288;https://archimer.ifremer.fr/doc/00228/33951/&#8288;</code>.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>
<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(argo)
argoNew &lt;- handleFlags(argo)
# Demonstrate replacement, looking at the second profile
f &lt;- argo[["salinityFlag"]][, 2]
df &lt;- data.frame(flag = f, orig = argo[["salinity"]][, 2], new = argoNew[["salinity"]][, 2])
df[11:15, ] # notice line 13

</code></pre>

<hr>
<h2 id='handleFlags+2Cctd-method'>Handle Flags in ctd Objects</h2><span id='topic+handleFlags+2Cctd-method'></span>

<h3>Description</h3>

<p>Data-quality flags are stored in the <code>metadata</code>
slot of <a href="#topic+oce-class">oce</a> objects in a
<a href="base.html#topic+list">list</a> named <code>flags</code>.
The present function (a generic that has specialized versions
for various data classes) provides a way to
manipulate the contents of the <code>data</code> slot, based on
such data-quality flags. For example, a common operation is to replace
erroneous data with <code>NA</code>.
</p>
<p>If the <code>flags</code> within <code>object</code>'s <code>metadata</code> slot is empty,
then <code>object</code> is returned, unaltered.
Otherwise, <code>handleFlags</code> examines <code>object@metadata$flags</code>
in the context of the <code>flags</code> argument, and then
carries out actions that are specified by the <code>actions</code> argument.
By default, this sets the returned <code>data</code> entries to <code>NA</code>,
wherever the corresponding <code>metadata$flag</code> values
signal unreliable data. To maintain a hint as to why
<code>data</code> were changed, <code>metadata$flags</code> in the
returned value is a direct copy of the corresponding
entry in <code>object</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ctd'
handleFlags(
  object = "oce",
  flags = NULL,
  actions = NULL,
  where = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="handleFlags+2B2Cctd-method_+3A_object">object</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cctd-method_+3A_flags">flags</code></td>
<td>
<p>A <a href="base.html#topic+list">list</a> specifying flag values upon which
actions will be taken. This can take two forms.
</p>

<ul>
<li><p> In the first form, the
list has named elements each containing a vector of integers. For example,
salinities flagged with values of 1 or 3:9 would be specified
by <code>flags=list(salinity=c(1,3:9))</code>. Several data items can be specified,
e.g. <code>flags=list(salinity=c(1,3:9), temperature=c(1,3:9))</code> indicates
that the actions are to take place for both salinity and temperature.
</p>
</li>
<li><p> In the second form, <code>flags</code> is a list holding a single <em>unnamed</em> vector, and
this means to apply the actions to <em>all</em> the data entries.  For example,
<code>flags=list(c(1,3:9))</code> means to apply not just to salinity and temperature,
but to everything within the <code>data</code> slot.
</p>
</li></ul>

<p>If <code>flags</code> is not provided, then <code><a href="#topic+defaultFlags">defaultFlags()</a></code> is called, to try to
determine a reasonable default.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cctd-method_+3A_actions">actions</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that contains items with
names that match those in the <code>flags</code> argument.  If <code>actions</code>
is not supplied, the default will be to set all values identified by
<code>flags</code> to <code>NA</code>; this can also be specified by
specifying <code>actions=list("NA")</code>. It is also possible to specify
functions that calculate replacement values. These are provided
with <code>object</code> as the single argument, and must return a
replacement for the data item in question.
See &ldquo;Details&rdquo; for the default that is used if <code>actions</code> is not supplied.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cctd-method_+3A_where">where</code></td>
<td>
<p>an optional character value that permits the function to work with
objects that store flags in e.g. <code>object@metadata$flags$where</code>
instead of in <code>object@metadata$flags</code>, and data within
<code>object@data$where</code> instead of within <code>object@data</code>. The
default value of <code>NULL</code> means to look withing <code>object@metadata</code>
itself, and this is the default within <code>oce</code>.  (The purpose of <code>where</code>
is to make <code>oce</code> extensible by other packages, which may choose to store
data two levels deep in the <code>data</code> slot.)</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cctd-method_+3A_debug">debug</code></td>
<td>
<p>An optional integer specifying the degree of debugging, with
value 0 meaning to skip debugging and 1 or higher meaning to print some
information about the arguments and the data. It is usually a good idea to set
this to 1 for initial work with a dataset, to see which flags are being
handled for each data item. If not supplied, this defaults to the value of
<code><a href="base.html#topic+getOption">getOption</a></code><code>("oceDebug")</code>.</p>
</td></tr>
</table>


<h3>References</h3>

<p>The following link used to work, but failed as of December 2020.
</p>

<ol>
<li> <p><code style="white-space: pre;">&#8288;https://www.nodc.noaa.gov/woce/woce_v3/wocedata_1/whp/exchange/exchange_format_desc.htm&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(section)
stn &lt;- section[["station", 100]]
# 1. Default: anything not flagged as 2 is set to NA, to focus
# solely on 'good', in the World Hydrographic Program scheme.
STN1 &lt;- handleFlags(stn, flags = list(c(1, 3:9)))
data.frame(old = stn[["salinity"]], new = STN1[["salinity"]], salinityFlag = stn[["salinityFlag"]])

# 2. Use bottle salinity, if it is good and ctd is bad
replace &lt;- 2 == stn[["salinityBottleFlag"]] &amp; 2 != stn[["salinityFlag"]]
S &lt;- ifelse(replace, stn[["salinityBottle"]], stn[["salinity"]])
STN2 &lt;- oceSetData(stn, "salinity", S)

# 3. Use smoothed TS relationship to nudge questionable data.
f &lt;- function(x) {
    S &lt;- x[["salinity"]]
    T &lt;- x[["temperature"]]
    df &lt;- 0.5 * length(S) # smooths a bit
    sp &lt;- smooth.spline(T, S, df = df)
    0.5 * (S + predict(sp, T)$y)
}
par(mfrow = c(1, 2))
STN3 &lt;- handleFlags(stn, flags = list(salinity = c(1, 3:9)), action = list(salinity = f))
plotProfile(stn, "salinity", mar = c(3, 3, 3, 1))
p &lt;- stn[["pressure"]]
par(mar = c(3, 3, 3, 1))
plot(STN3[["salinity"]] - stn[["salinity"]], p, ylim = rev(range(p)))

# 4. Single-variable flags (vector specification)
data(section)
# Multiple-flag scheme: one per data item
A &lt;- section[["station", 100]]
deep &lt;- A[["pressure"]] &gt; 1500
flag &lt;- ifelse(deep, 7, 2)
for (flagName in names(A[["flags"]])) {
    A[[paste(flagName, "Flag", sep = "")]] &lt;- flag
}
Af &lt;- handleFlags(A)
stopifnot(all.equal(is.na(Af[["salinity"]]), deep))

# 5. Single-variable flags (list specification)
B &lt;- section[["station", 100]]
B[["flags"]] &lt;- list(flag)
Bf &lt;- handleFlags(B)
stopifnot(all.equal(is.na(Bf[["salinity"]]), deep))

</code></pre>

<hr>
<h2 id='handleFlags+2Coce-method'>Handle Flags in oce Objects</h2><span id='topic+handleFlags+2Coce-method'></span>

<h3>Description</h3>

<p>Data-quality flags are stored in the <code>metadata</code>
slot of <a href="#topic+oce-class">oce</a> objects in a
<a href="base.html#topic+list">list</a> named <code>flags</code>.
The present function (a generic that has specialized versions
for various data classes) provides a way to
manipulate the contents of the <code>data</code> slot, based on
such data-quality flags. For example, a common operation is to replace
erroneous data with <code>NA</code>.
</p>
<p>If the <code>flags</code> within <code>object</code>'s <code>metadata</code> slot is empty,
then <code>object</code> is returned, unaltered.
Otherwise, <code>handleFlags</code> examines <code>object@metadata$flags</code>
in the context of the <code>flags</code> argument, and then
carries out actions that are specified by the <code>actions</code> argument.
By default, this sets the returned <code>data</code> entries to <code>NA</code>,
wherever the corresponding <code>metadata$flag</code> values
signal unreliable data. To maintain a hint as to why
<code>data</code> were changed, <code>metadata$flags</code> in the
returned value is a direct copy of the corresponding
entry in <code>object</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'oce'
handleFlags(
  object = "oce",
  flags = NULL,
  actions = NULL,
  where = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="handleFlags+2B2Coce-method_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Coce-method_+3A_flags">flags</code></td>
<td>
<p>A <a href="base.html#topic+list">list</a> specifying flag values upon which
actions will be taken. This can take two forms.
</p>

<ul>
<li><p> In the first form, the
list has named elements each containing a vector of integers. For example,
salinities flagged with values of 1 or 3:9 would be specified
by <code>flags=list(salinity=c(1,3:9))</code>. Several data items can be specified,
e.g. <code>flags=list(salinity=c(1,3:9), temperature=c(1,3:9))</code> indicates
that the actions are to take place for both salinity and temperature.
</p>
</li>
<li><p> In the second form, <code>flags</code> is a list holding a single <em>unnamed</em> vector, and
this means to apply the actions to <em>all</em> the data entries.  For example,
<code>flags=list(c(1,3:9))</code> means to apply not just to salinity and temperature,
but to everything within the <code>data</code> slot.
</p>
</li></ul>

<p>If <code>flags</code> is not provided, then <code><a href="#topic+defaultFlags">defaultFlags()</a></code> is called, to try to
determine a reasonable default.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Coce-method_+3A_actions">actions</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that contains items with
names that match those in the <code>flags</code> argument.  If <code>actions</code>
is not supplied, the default will be to set all values identified by
<code>flags</code> to <code>NA</code>; this can also be specified by
specifying <code>actions=list("NA")</code>. It is also possible to specify
functions that calculate replacement values. These are provided
with <code>object</code> as the single argument, and must return a
replacement for the data item in question.
See &ldquo;Details&rdquo; for the default that is used if <code>actions</code> is not supplied.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Coce-method_+3A_where">where</code></td>
<td>
<p>an optional character value that permits the function to work with
objects that store flags in e.g. <code>object@metadata$flags$where</code>
instead of in <code>object@metadata$flags</code>, and data within
<code>object@data$where</code> instead of within <code>object@data</code>. The
default value of <code>NULL</code> means to look withing <code>object@metadata</code>
itself, and this is the default within <code>oce</code>.  (The purpose of <code>where</code>
is to make <code>oce</code> extensible by other packages, which may choose to store
data two levels deep in the <code>data</code> slot.)</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Coce-method_+3A_debug">debug</code></td>
<td>
<p>An optional integer specifying the degree of debugging, with
value 0 meaning to skip debugging and 1 or higher meaning to print some
information about the arguments and the data. It is usually a good idea to set
this to 1 for initial work with a dataset, to see which flags are being
handled for each data item. If not supplied, this defaults to the value of
<code><a href="base.html#topic+getOption">getOption</a></code><code>("oceDebug")</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Base-level handling of flags.
</p>


<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>

<hr>
<h2 id='handleFlags+2Csection-method'>Handle flags in section Objects</h2><span id='topic+handleFlags+2Csection-method'></span><span id='topic+handleFlags.section'></span>

<h3>Description</h3>

<p>Data-quality flags are stored in the <code>metadata</code>
slot of <a href="#topic+oce-class">oce</a> objects in a
<a href="base.html#topic+list">list</a> named <code>flags</code>.
The present function (a generic that has specialized versions
for various data classes) provides a way to
manipulate the contents of the <code>data</code> slot, based on
such data-quality flags. For example, a common operation is to replace
erroneous data with <code>NA</code>.
</p>
<p>If the <code>flags</code> within <code>object</code>'s <code>metadata</code> slot is empty,
then <code>object</code> is returned, unaltered.
Otherwise, <code>handleFlags</code> examines <code>object@metadata$flags</code>
in the context of the <code>flags</code> argument, and then
carries out actions that are specified by the <code>actions</code> argument.
By default, this sets the returned <code>data</code> entries to <code>NA</code>,
wherever the corresponding <code>metadata$flag</code> values
signal unreliable data. To maintain a hint as to why
<code>data</code> were changed, <code>metadata$flags</code> in the
returned value is a direct copy of the corresponding
entry in <code>object</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'section'
handleFlags(
  object = "oce",
  flags = NULL,
  actions = NULL,
  where = where,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="handleFlags+2B2Csection-method_+3A_object">object</code></td>
<td>
<p>A <a href="#topic+section-class">section</a> object.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Csection-method_+3A_flags">flags</code></td>
<td>
<p>A <a href="base.html#topic+list">list</a> specifying flag values upon which
actions will be taken. This can take two forms.
</p>

<ul>
<li><p> In the first form, the
list has named elements each containing a vector of integers. For example,
salinities flagged with values of 1 or 3:9 would be specified
by <code>flags=list(salinity=c(1,3:9))</code>. Several data items can be specified,
e.g. <code>flags=list(salinity=c(1,3:9), temperature=c(1,3:9))</code> indicates
that the actions are to take place for both salinity and temperature.
</p>
</li>
<li><p> In the second form, <code>flags</code> is a list holding a single <em>unnamed</em> vector, and
this means to apply the actions to <em>all</em> the data entries.  For example,
<code>flags=list(c(1,3:9))</code> means to apply not just to salinity and temperature,
but to everything within the <code>data</code> slot.
</p>
</li></ul>

<p>If <code>flags</code> is not provided, then <code><a href="#topic+defaultFlags">defaultFlags()</a></code> is called, to try to
determine a reasonable default.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Csection-method_+3A_actions">actions</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that contains items with
names that match those in the <code>flags</code> argument.  If <code>actions</code>
is not supplied, the default will be to set all values identified by
<code>flags</code> to <code>NA</code>; this can also be specified by
specifying <code>actions=list("NA")</code>. It is also possible to specify
functions that calculate replacement values. These are provided
with <code>object</code> as the single argument, and must return a
replacement for the data item in question.
See &ldquo;Details&rdquo; for the default that is used if <code>actions</code> is not supplied.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Csection-method_+3A_where">where</code></td>
<td>
<p>an optional character value that permits the function to work with
objects that store flags in e.g. <code>object@metadata$flags$where</code>
instead of in <code>object@metadata$flags</code>, and data within
<code>object@data$where</code> instead of within <code>object@data</code>. The
default value of <code>NULL</code> means to look withing <code>object@metadata</code>
itself, and this is the default within <code>oce</code>.  (The purpose of <code>where</code>
is to make <code>oce</code> extensible by other packages, which may choose to store
data two levels deep in the <code>data</code> slot.)</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Csection-method_+3A_debug">debug</code></td>
<td>
<p>An optional integer specifying the degree of debugging, with
value 0 meaning to skip debugging and 1 or higher meaning to print some
information about the arguments and the data. It is usually a good idea to set
this to 1 for initial work with a dataset, to see which flags are being
handled for each data item. If not supplied, this defaults to the value of
<code><a href="base.html#topic+getOption">getOption</a></code><code>("oceDebug")</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default for <code>flags</code> is based on
calling <code><a href="#topic+defaultFlags">defaultFlags()</a></code> based on the
<code>metadata</code> in the first station in the section. If the
other stations have different flag schemes (which seems highly
unlikely for archived data), this will not work well, and indeed
the only way to proceed would be to use <code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method()</a></code>
on the stations, individually.
</p>


<h3>References</h3>

<p>The following link used to work, but was found to fail in December 2020.
</p>

<ol>
<li> <p><code style="white-space: pre;">&#8288;https://www.nodc.noaa.gov/woce/woce_v3/wocedata_1/whp/exchange/exchange_format_desc.htm&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>
<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(section)
section2 &lt;- handleFlags(section, flags = c(1, 3:9))
par(mfrow = c(2, 1))
plotTS(section)
plotTS(section2)

</code></pre>

<hr>
<h2 id='handleFlags+2Cvector-method'>Signal Erroneous Application to non-oce Objects</h2><span id='topic+handleFlags+2Cvector-method'></span>

<h3>Description</h3>

<p>Signal Erroneous Application to non-oce Objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'vector'
handleFlags(
  object = "oce",
  flags = list(),
  actions = list(),
  where = list(),
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="handleFlags+2B2Cvector-method_+3A_object">object</code></td>
<td>
<p>A vector, which cannot be the case for <code>oce</code> objects.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cvector-method_+3A_flags">flags</code></td>
<td>
<p>Ignored.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cvector-method_+3A_actions">actions</code></td>
<td>
<p>Ignored.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cvector-method_+3A_where">where</code></td>
<td>
<p>Ignored.</p>
</td></tr>
<tr><td><code id="handleFlags+2B2Cvector-method_+3A_debug">debug</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>

<hr>
<h2 id='handleFlagsInternal'>Low-Level Function for Handling Data-Quality Flags</h2><span id='topic+handleFlagsInternal'></span>

<h3>Description</h3>

<p>This function is designed for internal use within the <code>oce</code> package.  Its
purpose is to carry out low-level processing relating to data-quality flags,
as a support for higher-level functions such <a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a> for
<code>ctd</code> objects, <a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a> for <code>adp</code> objects,
etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>handleFlagsInternal(object, flags, actions, where, debug = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="handleFlagsInternal_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="handleFlagsInternal_+3A_flags">flags</code></td>
<td>
<p>a named <a href="base.html#topic+list">list</a> of numeric values.</p>
</td></tr>
<tr><td><code id="handleFlagsInternal_+3A_actions">actions</code></td>
<td>
<p>A character vector indicating actions to be carried out for the corresponding
<code>flags</code> values. This will be lengthened with <code><a href="base.html#topic+rep">rep()</a></code> if necessary, to be
of the same length as <code>flags</code>. A common value for <code>actions</code> is <code>"NA"</code>, which
means that data values that are flagged are replaced by <code>NA</code> in the returned result.</p>
</td></tr>
<tr><td><code id="handleFlagsInternal_+3A_where">where</code></td>
<td>
<p>An optional string that permits the function to work with
objects that store flags in e.g. <code>object@metadata$flags$where</code>
instead of in <code>object@metadata$flags</code>, and data within
<code>object@data$where</code> instead of within <code>object@data</code>. The
appropriate value for <code>where</code> within the oce package is
the default, <code>NULL</code>, which means that this extra subdirectory
is not being used.</p>
</td></tr>
<tr><td><code id="handleFlagsInternal_+3A_debug">debug</code></td>
<td>
<p>An integer indicating the degree of debugging requested, with value <code>0</code>
meaning to act silently, and value <code>1</code> meaning to print some information about the
steps in processing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A copy of <code>object</code>, possibly with modifications to its
<code>data</code> slot, if <code>object</code> contains flag values that have
actions that alter the data.
</p>

<hr>
<h2 id='head.oce'>Extract The Start of an Oce Object</h2><span id='topic+head.oce'></span>

<h3>Description</h3>

<p>Extract The Start of an Oce Object
</p>
<p>This function handles the following object classes directly:
<a href="#topic+adp-class">adp</a>,
<a href="#topic+adv-class">adv</a>,
<a href="#topic+argo-class">argo</a> (selection by profile),
<a href="#topic+coastline-class">coastline</a>,
<a href="#topic+ctd-class">ctd</a>,
<a href="#topic+echosounder-class">echosounder</a> (selection by ping),
<a href="#topic+section-class">section</a> (selection by station)
and
<a href="#topic+topo-class">topo</a> (selection by longitude and latitude).
It does not handle
<a href="#topic+amsr-class">amsr</a> or
<a href="#topic+landsat-class">landsat</a>
yet, instead issuing a warning and returning <code>x</code> in those cases.
For all other classes, it calls <code><a href="utils.html#topic+head">head()</a></code>
with <code>n</code> as provided, for each item in the <code>data</code>
slot, issuing a warning if that item is not a vector; the
author is quite aware that this may not work well for all classes.
The plan is to handle all appropriate classes by July 2018.
Please contact the author if there is a class you need handled
before that date.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'oce'
head(x, n = 6L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="head.oce_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="head.oce_+3A_n">n</code></td>
<td>
<p>Number of elements to extract, as for <code><a href="utils.html#topic+head">head()</a></code>.</p>
</td></tr>
<tr><td><code id="head.oce_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tail.oce">tail.oce()</a></code>, which yields the end of an <code>oce</code> object.
</p>

<hr>
<h2 id='imagep'>Plot an Image with a Color Palette</h2><span id='topic+imagep'></span>

<h3>Description</h3>

<p>Plot an image with a color palette, in a way that does not conflict with
<code><a href="graphics.html#topic+par">par</a></code><code>("mfrow")</code> or <code><a href="graphics.html#topic+layout">layout()</a></code>.  To plot just a palette,
e.g. to get an x-y plot with points colored according to a palette, use
<code><a href="#topic+drawPalette">drawPalette()</a></code> and then draw the main diagram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imagep(
  x,
  y,
  z,
  xlim,
  ylim,
  zlim,
  zclip = FALSE,
  flipy = FALSE,
  xlab = "",
  ylab = "",
  zlab = "",
  zlabPosition = c("top", "side"),
  las.palette = 0,
  decimate = TRUE,
  quiet = FALSE,
  breaks,
  col,
  colormap,
  labels = NULL,
  at = NULL,
  drawContours = FALSE,
  drawPalette = TRUE,
  drawTriangles = FALSE,
  tformat,
  drawTimeRange = getOption("oceDrawTimeRange"),
  filledContour = FALSE,
  missingColor = NULL,
  useRaster,
  mgp = getOption("oceMgp"),
  mar,
  mai.palette,
  xaxs = "i",
  yaxs = "i",
  asp = NA,
  cex = par("cex"),
  cex.axis = cex,
  cex.lab = cex,
  cex.main = cex,
  axes = TRUE,
  main = "",
  axisPalette,
  add = FALSE,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="imagep_+3A_x">x</code>, <code id="imagep_+3A_y">y</code></td>
<td>
<p>These have different meanings in different modes of operation.
</p>
<p><em>Mode 1.</em> One
mode has them meaning the locations of coordinates along which values
matrix <code>z</code> are defined. In this case, both <code>x</code> and <code>y</code>
must be supplied and, within each, the values must be finite and
distinct; if values are out of order, they (and <code>z</code>) will be
transformed to put them in order.
ordered in a matching way).
</p>
<p><em>Mode 2.</em>
If <code>z</code> is provided but not <code>x</code> and <code>y</code>, then the latter
are constructed to
indicate the indices of the matrix, in contrast
to the range of 0 to 1, as is the case for <code><a href="Matrix.html#topic+image">image()</a></code>.
</p>
<p><em>Mode 3.</em> If
<code>x</code> is a list, its components <code>x$x</code> and <code>x$y</code> are used
for <code>x</code> and <code>y</code>, respectively. If the list has component
<code>z</code> this is used for <code>z</code>. (NOTE: these arguments are meant to
mimic those of <code><a href="Matrix.html#topic+image">image()</a></code>, which explains the same description
here.)  <em>Mode 4.</em> There are also some special cases, e.g. if <code>x</code> is a
topographic object such as can be created with <code><a href="#topic+read.topo">read.topo()</a></code>
or <code><a href="#topic+as.topo">as.topo()</a></code>, then longitude and latitude are used for
axes, and topographic height is drawn.</p>
</td></tr>
<tr><td><code id="imagep_+3A_z">z</code></td>
<td>
<p>A matrix containing the values to be plotted (NAs are allowed). Note
that x can be used instead of z for convenience. (NOTE: these arguments
are meant to mimic those of <code><a href="Matrix.html#topic+image">image()</a></code>, which explains the same
description here.)</p>
</td></tr>
<tr><td><code id="imagep_+3A_xlim">xlim</code>, <code id="imagep_+3A_ylim">ylim</code></td>
<td>
<p>Limits on x and y axes.</p>
</td></tr>
<tr><td><code id="imagep_+3A_zlim">zlim</code></td>
<td>
<p>If missing, the z scale is determined by the range of the data.
If provided, <code>zlim</code> may take several forms. First, it may be a pair
of numbers that specify the limits for the color scale.  Second,
it could be the string <code>"histogram"</code>, to yield a flattened
histogram (i.e. to increase contrast). Third, it could be the
string <code>"symmetric"</code>, to yield limits that are symmetric
about zero, which can be helpful in drawing velocity fields,
for which a zero value has a particular meaning (in which case,
a good color scheme might be <code>col=oceColorsTwo</code>).</p>
</td></tr>
<tr><td><code id="imagep_+3A_zclip">zclip</code></td>
<td>
<p>Logical, indicating whether to clip the colors to those
corresponding to <code>zlim</code>. This only works if <code>zlim</code> is
provided. Clipped regions will be colored with <code>missingColor</code>.
Thus, clipping an image is somewhat analogous to clipping in an xy
plot, with clipped data being ignored, which in an image means to be be
colored with <code>missingColor</code>.</p>
</td></tr>
<tr><td><code id="imagep_+3A_flipy">flipy</code></td>
<td>
<p>Logical, with <code>TRUE</code> indicating that the graph
should have the y axis reversed, i.e. with smaller values at
the bottom of the page. (<em>Historical note:</em> until 2019 March 26,
the meaning of <code>flipy</code> was different; it meant to reverse the
range of the y axis, so that if <code>ylim</code> were given as a reversed
range, then setting <code>flipy=TRUE</code> would reverse the flip, yielding
a conventional axis with smaller values at the bottom.)</p>
</td></tr>
<tr><td><code id="imagep_+3A_xlab">xlab</code>, <code id="imagep_+3A_ylab">ylab</code>, <code id="imagep_+3A_zlab">zlab</code></td>
<td>
<p>Names for x axis, y axis, and the image values.</p>
</td></tr>
<tr><td><code id="imagep_+3A_zlabposition">zlabPosition</code></td>
<td>
<p>String indicating where to put the label for the z axis,
either at the top-right of the main image, or on the side, in the axis
for the palette.</p>
</td></tr>
<tr><td><code id="imagep_+3A_las.palette">las.palette</code></td>
<td>
<p>Parameter controlling the orientation of labels on the
image palette, passed as the <code>las</code> argument to <code><a href="#topic+drawPalette">drawPalette()</a></code>.  See the
documentation for <code><a href="#topic+drawPalette">drawPalette()</a></code> for details.</p>
</td></tr>
<tr><td><code id="imagep_+3A_decimate">decimate</code></td>
<td>
<p>Controls whether the image will be decimated before plotting,
in three possible cases.
</p>

<ol>
<li><p> If <code>decimate=FALSE</code> then every grid cell in the matrix will
be represented by a pixel in the image.
</p>
</li>
<li><p> If <code>decimate=TRUE</code> (the default), then decimation will be done
in the horizontal or vertical direction (or both) if the length of the
corresponding edge of the <code>z</code> matrix exceeds 800. (This also creates
a warning message.) The decimation
factor is computed as the integer just below the ratio of <code>z</code> dimension
to 400. Thus, no decimation is done if the dimension is less than 800,
but if the dimension s between 800 and 1199, only every second grid
point is mapped to a pixel in the image.
</p>
</li>
<li><p> If <code>decimate</code> is an integer, then that <code>z</code> is subsampled
at <code>seq.int(1L, dim(z)[1], by=decimate)</code> (as is <code>x</code>), and
the same is done for the <code>y</code> direction.
</p>
</li>
<li><p> If <code>decimate</code> is a vector of two integers, the first is used for
the first index of <code>z</code>, and the second is used for the second
index.
</p>
</li></ol>
</td></tr>
<tr><td><code id="imagep_+3A_quiet">quiet</code></td>
<td>
<p>logical value indicating whether to silence warnings that
might occur if the image is being decimated.</p>
</td></tr>
<tr><td><code id="imagep_+3A_breaks">breaks</code></td>
<td>
<p>The z values for breaks in the color scheme.  If this is of
length 1, the value indicates the desired number of breaks, which is
supplied to <code><a href="base.html#topic+pretty">pretty()</a></code>, in determining clean break points.
If <code>colormap</code> is provided, it takes precedence over <code>breaks</code> and <code>col</code>.</p>
</td></tr>
<tr><td><code id="imagep_+3A_col">col</code></td>
<td>
<p>Either a vector of colors corresponding to the breaks, of length
1 plus the number of breaks, or a function specifying colors.
If <code>col</code> is not provided, and if <code>colormap</code> is also not provided, then <code>col</code>
defaults to <code><a href="#topic+oceColorsViridis">oceColorsViridis()</a></code>.
If <code>colormap</code> is provided, it takes precedence over <code>breaks</code> and <code>col</code>.</p>
</td></tr>
<tr><td><code id="imagep_+3A_colormap">colormap</code></td>
<td>
<p>A color map as created by <code><a href="#topic+colormap">colormap()</a></code>.  If
provided, then <code>colormap$breaks</code> and <code>colormap$col</code> take
precedence over the present arguments <code>breaks</code> and <code>col</code>.
(All of the other contents of <code>colormap</code> are ignored, though.)
If <code>colormap</code> is provided, it takes precedence over <code>breaks</code> and <code>col</code>.</p>
</td></tr>
<tr><td><code id="imagep_+3A_labels">labels</code></td>
<td>
<p>Optional vector of labels for ticks on palette axis (must
correspond with <code>at</code>).</p>
</td></tr>
<tr><td><code id="imagep_+3A_at">at</code></td>
<td>
<p>Optional vector of positions for the <code>label</code>s.</p>
</td></tr>
<tr><td><code id="imagep_+3A_drawcontours">drawContours</code></td>
<td>
<p>Logical value indicating whether to draw contours on the
image, and palette, at the color breaks.  Images with a great deal of
high-wavenumber variation look poor with contours.</p>
</td></tr>
<tr><td><code id="imagep_+3A_drawpalette">drawPalette</code></td>
<td>
<p>Indication of the type of palette to draw, if any.  If
<code>drawPalette=TRUE</code>, a palette is drawn at the right-hand side of the
main image.  If <code>drawPalette=FALSE</code>, no palette is drawn, and the
right-hand side of the plot has a thin margin.  If
<code>drawPalette="space"</code>, then no palette is drawn, but space is put in
the right-hand margin to occupy the region in which the palette would
have been drawn.  This last form is useful for producing stacked plots
with uniform left and right margins, but with palettes on only some of
the images.</p>
</td></tr>
<tr><td><code id="imagep_+3A_drawtriangles">drawTriangles</code></td>
<td>
<p>Logical value indicating whether to draw
triangles on the top and bottom of the palette.  This is passed to
<code><a href="#topic+drawPalette">drawPalette()</a></code>.</p>
</td></tr>
<tr><td><code id="imagep_+3A_tformat">tformat</code></td>
<td>
<p>Optional argument passed to <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code>, for
plot types that call that function.  (See <code><a href="base.html#topic+strptime">strptime()</a></code> for the
format used.)</p>
</td></tr>
<tr><td><code id="imagep_+3A_drawtimerange">drawTimeRange</code></td>
<td>
<p>Logical, only used if the <code>x</code> axis is a
time.  If <code>TRUE</code>, then an indication of the time range of the
data (not the axis) is indicated at the top-left margin of the
graph.  This is useful because the labels on time axes only indicate
hours if the range is less than a day, etc.</p>
</td></tr>
<tr><td><code id="imagep_+3A_filledcontour">filledContour</code></td>
<td>
<p>Boolean value indicating whether to use filled
contours to plot the image.</p>
</td></tr>
<tr><td><code id="imagep_+3A_missingcolor">missingColor</code></td>
<td>
<p>A color to be used to indicate missing data, or
<code>NULL</code> for transparent (to see this, try setting
<code>par("bg")&lt;-"red"</code>).</p>
</td></tr>
<tr><td><code id="imagep_+3A_useraster">useRaster</code></td>
<td>
<p>A logical value passed to <code><a href="Matrix.html#topic+image">image()</a></code>, in cases
where <code>filledContour</code> is <code>FALSE</code>. Setting <code>useRaster=TRUE</code>
can alleviate some anti-aliasing effects on some plot devices;
see the documentation for <code><a href="Matrix.html#topic+image">image()</a></code>.</p>
</td></tr>
<tr><td><code id="imagep_+3A_mgp">mgp</code></td>
<td>
<p>A 3-element numerical vector to use for <code>par(mgp)</code>, and
also for <code>par(mar)</code>, computed from this.  The default is
tighter than the R default, in order to use more space for the
data and less for the axes.</p>
</td></tr>
<tr><td><code id="imagep_+3A_mar">mar</code></td>
<td>
<p>A 4-element Value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.  If not
given, a reasonable value is calculated based on whether <code>xlab</code> and
<code>ylab</code> are empty strings.</p>
</td></tr>
<tr><td><code id="imagep_+3A_mai.palette">mai.palette</code></td>
<td>
<p>Palette margin corrections (in inches), added to the
<code>mai</code> value used for the palette.  Use with care.</p>
</td></tr>
<tr><td><code id="imagep_+3A_xaxs">xaxs</code></td>
<td>
<p>Character indicating whether image should extend to edge
of x axis (with value <code>"i"</code>) or not; see
<code><a href="graphics.html#topic+par">par</a></code><code>("xaxs")</code>.</p>
</td></tr>
<tr><td><code id="imagep_+3A_yaxs">yaxs</code></td>
<td>
<p>As <code>xaxs</code> but for y axis.</p>
</td></tr>
<tr><td><code id="imagep_+3A_asp">asp</code></td>
<td>
<p>Aspect ratio of the plot, as for <code><a href="graphics.html#topic+plot.default">plot.default()</a></code>. If
<code>x</code> inherits from <a href="#topic+topo-class">topo</a> and <code>asp=NA</code> (the
default) then <code>asp</code> is redefined to be the reciprocal of the
mean latitude in <code>x</code>, as a way to reduce geographical distortion.
Otherwise, if <code>asp</code> is not <code>NA</code>, then it is used directly.</p>
</td></tr>
<tr><td><code id="imagep_+3A_cex">cex</code></td>
<td>
<p>numeric character expansion factor, used for <code>cex.axis</code>, <code>cex.lab</code> and <code>cex.main</code>,
if those values are not supplied.</p>
</td></tr>
<tr><td><code id="imagep_+3A_cex.axis">cex.axis</code>, <code id="imagep_+3A_cex.lab">cex.lab</code>, <code id="imagep_+3A_cex.main">cex.main</code></td>
<td>
<p>numeric character expansion factors for axis numbers,
axis names and plot titles; see <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
<tr><td><code id="imagep_+3A_axes">axes</code></td>
<td>
<p>Logical, set <code>TRUE</code> to get axes on the main image.</p>
</td></tr>
<tr><td><code id="imagep_+3A_main">main</code></td>
<td>
<p>Title for plot.</p>
</td></tr>
<tr><td><code id="imagep_+3A_axispalette">axisPalette</code></td>
<td>
<p>Optional replacement function for <code><a href="graphics.html#topic+axis">axis()</a></code>, passed to
<code><a href="#topic+drawPalette">drawPalette()</a></code>.</p>
</td></tr>
<tr><td><code id="imagep_+3A_add">add</code></td>
<td>
<p>Logical value indicating whether to add to an existing plot.
The default value, <code>FALSE</code> indicates that a new plot is to be created.
However, if <code>add</code> is <code>TRUE</code>, the idea is to add an image (but not
its palette or its axes) to an existing plot. Clearly, then, arguments
such <code>xlim</code> are to be ignored. Indeed, if <code>add=TRUE</code>, the only
arguments examined are <code>x</code> (which must be a vector; the mode of providing
a matrix or <code>oce</code> object does not work), <code>y</code>, <code>z</code>,
<code>decimate</code>, plus either <code>colormap</code> or
both <code>breaks</code> and <code>col</code>.</p>
</td></tr>
<tr><td><code id="imagep_+3A_debug">debug</code></td>
<td>
<p>A flag that turns on debugging.  Set to 1 to get a
moderate amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="imagep_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, creates an image with a color palette to the right.  The effect is similar to
<code><a href="graphics.html#topic+filled.contour">filled.contour()</a></code> except that with <code>imagep</code> it is possible to
set the <code><a href="graphics.html#topic+layout">layout()</a></code> outside the function, which enables the creation
of plots with many image-palette panels.  Note that the contour lines may not
coincide with the color transitions, in the case of coarse images.
</p>
<p>Note that this does not use <code><a href="graphics.html#topic+layout">layout()</a></code> or any of the other screen
splitting methods.  It simply manipulates margins, and draws two plots
together.  This lets users employ their favourite layout schemes.
</p>
<p>NOTE: <code>imagep</code> is an analogue of <code><a href="Matrix.html#topic+image">image()</a></code>, and from that
it borrows a the convention that the number of rows in the matrix corresponds to
to <code>x</code> axis, not the <code>y</code> axis.  (Actually, <code><a href="Matrix.html#topic+image">image()</a></code> permits
the length of <code>x</code> to match either <code>nrow(z)</code> or <code>1+nrow(z)</code>, but
here only the first is permitted.)
</p>


<h3>Value</h3>

<p>A list is silently returned, containing <code>xat</code> and <code>yat</code>,
values that can be used by <code><a href="#topic+oce.grid">oce.grid()</a></code> to add a grid to the
plot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>See Also</h3>

<p>This uses <code><a href="#topic+drawPalette">drawPalette()</a></code>, and is used by <code><a href="#topic+plot+2Cadp-method">plot,adp-method()</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method()</a></code>, and other image-generating functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# 1. simplest use
imagep(volcano)

# 2. something oceanographic (internal-wave speed)
h &lt;- seq(0, 50, length.out = 100)
drho &lt;- seq(1, 3, length.out = 200)
speed &lt;- outer(h, drho, function(drho, h) sqrt(9.8 * drho * h / 1024))
imagep(h, drho, speed,
    xlab = "Equivalent depth [m]",
    ylab = expression(paste(Delta * rho, " [kg/m^3]")),
    zlab = "Internal-wave speed [m/s]"
)

# 3. fancy labelling on atan() function
x &lt;- seq(0, 1, 0.01)
y &lt;- seq(0, 1, 0.01)
angle &lt;- outer(x, y, function(x, y) atan2(y, x))
imagep(x, y, angle,
    filledContour = TRUE, breaks = c(0, pi / 4, pi / 2),
    col = c("lightgray", "darkgray"),
    at = c(0, pi / 4, pi / 2),
    labels = c(0, expression(pi / 4), expression(pi / 2))
)

# 5. y-axis flipping
par(mfrow = c(2, 2))
data(adp)
d &lt;- adp[["distance"]]
t &lt;- adp[["time"]]
u &lt;- adp[["v"]][, , 1]
imagep(t, d, u, drawTimeRange = FALSE)
mtext("normal")
imagep(t, d, u, flipy = TRUE, drawTimeRange = FALSE)
mtext("flipy")
imagep(t, d, u, ylim = rev(range(d)), drawTimeRange = FALSE)
mtext("ylim")
imagep(t, d, u, ylim = rev(range(d)), flipy = TRUE, drawTimeRange = FALSE)
mtext("flipy and ylim")
par(mfrow = c(1, 1))

# 6. a colormap case
data(topoWorld)
cm &lt;- colormap(name = "gmt_globe")
imagep(topoWorld, colormap = cm)

</code></pre>

<hr>
<h2 id='initialize+2Cctd-method'>Initialize Storage for a ctd Object</h2><span id='topic+initialize+2Cctd-method'></span><span id='topic+initialize.ctd'></span>

<h3>Description</h3>

<p>This function creates <a href="#topic+ctd-class">ctd</a> objects. It is mainly
used by <code>oce</code> functions such as <code><a href="#topic+read.ctd">read.ctd()</a></code> and <code><a href="#topic+as.ctd">as.ctd()</a></code>,
and it is not intended for novice users, so it may change at any time, without
following the usual rules for transitioning to deprecated and defunct status
(see <a href="#topic+oce-deprecated">oce-deprecated</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ctd'
initialize(
  .Object,
  pressure,
  salinity,
  temperature,
  conductivity,
  units,
  pressureType,
  deploymentType,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initialize+2B2Cctd-method_+3A_.object">.Object</code></td>
<td>
<p>the string <code>"ctd"</code></p>
</td></tr>
<tr><td><code id="initialize+2B2Cctd-method_+3A_pressure">pressure</code></td>
<td>
<p>optional numerical vector of pressures.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cctd-method_+3A_salinity">salinity</code></td>
<td>
<p>optional numerical vector of salinities.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cctd-method_+3A_temperature">temperature</code></td>
<td>
<p>optional numerical vector of temperatures.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cctd-method_+3A_conductivity">conductivity</code></td>
<td>
<p>optional numerical vector of conductivities.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cctd-method_+3A_units">units</code></td>
<td>
<p>optional list indicating units for the quantities specified
in the previous arguments. If this
is not supplied, a default is set up, based on which of the
<code>pressure</code> to <code>conductivity</code> arguments were specified.
If all of those 4 arguments were specified, then <code>units</code> is set
up as if the call included the following:
<code>units=list(temperature=list(unit=expression(degree*C), scale="ITS-90"),
     salinity=list(unit=expression(), scale="PSS-78"),
     conductivity=list(unit=expression(), scale=""),
     pressure=list(unit=expression(dbar), scale=""),
     depth=list(unit=expression(m), scale=""))</code>. This list is trimmed
of any of the 4 items that were not specified in the previous
arguments. Note that if <code>units</code> is specified, then it is just
copied into the <code>metadata</code> slot of the returned object, so the user
must be careful to set up values that will make sense to other <code>oce</code>
functions.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cctd-method_+3A_pressuretype">pressureType</code></td>
<td>
<p>optional character string indicating the type of pressure;
if not supplied, this defaults to <code>"sea"</code>, which indicates the excess of
pressure over the atmospheric value, in dbar.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cctd-method_+3A_deploymenttype">deploymentType</code></td>
<td>
<p>optional character string indicating the type of deployment, which may
be <code>"unknown"</code>, <code>"profile"</code>, <code>"towyo"</code>, or <code>"thermosalinograph"</code>.
If this is not set, the value defaults to <code>"unknown"</code>.</p>
</td></tr>
<tr><td><code id="initialize+2B2Cctd-method_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To save storage, this function has arguments only for quantities that are often present in data
files all cases. For example, not
all data files will have oxygen, so that's not present here.
Extra data may be added after the object is created, using
<code><a href="#topic+oceSetData">oceSetData()</a></code>.
Similarly, <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code> may be used to add metadata (station ID, etc),
while bearing in mind that other functions look for such information
in very particular places (e.g. the station ID is a string named <code>station</code>
within the <code>metadata</code> slot). See <a href="#topic+ctd-class">ctd</a> for more information
on elements stored in <code>ctd</code> objects.
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# 1. empty
new("ctd")

# 2. fake data with no location information, so can only
#    plot with the UNESCO equation of state.
#    NOTE: always name arguments, in case the default order gets changed
ctd &lt;- new("ctd", salinity = 35 + 1:3 / 10, temperature = 10 - 1:3 / 10, pressure = 1:3)
summary(ctd)
plot(ctd, eos = "unesco")

# 3. as 2, but insert location and plot with GSW equation of state.
ctd &lt;- oceSetMetadata(ctd, "latitude", 44)
ctd &lt;- oceSetMetadata(ctd, "longitude", -63)
plot(ctd, eos = "gsw")

</code></pre>

<hr>
<h2 id='initializeFlags'>Create and Initialize oce Flags</h2><span id='topic+initializeFlags'></span>

<h3>Description</h3>

<p>This function creates an item for a named variable within
the <code>flags</code> entry in the object's <code>metadata</code> slot.
The purpose is both to document a flag scheme
and to make it so that <code><a href="#topic+initializeFlags">initializeFlags()</a></code>
and <code><a href="#topic+setFlags">setFlags()</a></code> can specify flags by
name, in addition to number. A generic function, it is
specialized for some classes via interpretation of the
<code>scheme</code> argument (see &ldquo;Details&rdquo;, for those
object classes that have such specializations).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initializeFlags(object, name = NULL, value = NULL, debug = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeFlags_+3A_object">object</code></td>
<td>
<p>An <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="initializeFlags_+3A_name">name</code></td>
<td>
<p>Character value indicating the name of a variable within the
<code>data</code> slot of <code>object</code>.</p>
</td></tr>
<tr><td><code id="initializeFlags_+3A_value">value</code></td>
<td>
<p>Numerical or character value to be stored in the newly-created
entry within <code>flags</code>. (A character value will only work if
<code><a href="#topic+initializeFlags">initializeFlags()</a></code> has been used first on <code>object</code>.)</p>
</td></tr>
<tr><td><code id="initializeFlags_+3A_debug">debug</code></td>
<td>
<p>Integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>object</code> already contains a <code>flags</code> entry with
the indicated name, then it is returned unaltered, and a warning
is issued.
</p>


<h3>Value</h3>

<p>An object with the <code>flags</code> item within the
<code>metadata</code> slot set up as indicated.
</p>


<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>

<hr>
<h2 id='initializeFlags+2Cadp-method'>Create and Initialize adp Flags</h2><span id='topic+initializeFlags+2Cadp-method'></span>

<h3>Description</h3>

<p>This function creates an item for a named variable within
the <code>flags</code> entry in the object's <code>metadata</code> slot.
The purpose is both to document a flag scheme
and to make it so that <code><a href="#topic+initializeFlags">initializeFlags()</a></code>
and <code><a href="#topic+setFlags">setFlags()</a></code> can specify flags by
name, in addition to number. A generic function, it is
specialized for some classes via interpretation of the
<code>scheme</code> argument (see &ldquo;Details&rdquo;, for those
object classes that have such specializations).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adp'
initializeFlags(
  object,
  name = NULL,
  value = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeFlags+2B2Cadp-method_+3A_object">object</code></td>
<td>
<p>An <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="initializeFlags+2B2Cadp-method_+3A_name">name</code></td>
<td>
<p>Character value indicating the name of a variable within the
<code>data</code> slot of <code>object</code>.</p>
</td></tr>
<tr><td><code id="initializeFlags+2B2Cadp-method_+3A_value">value</code></td>
<td>
<p>Numerical or character value to be stored in the newly-created
entry within <code>flags</code>. (A character value will only work if
<code><a href="#topic+initializeFlags">initializeFlags()</a></code> has been used first on <code>object</code>.)</p>
</td></tr>
<tr><td><code id="initializeFlags+2B2Cadp-method_+3A_debug">debug</code></td>
<td>
<p>Integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>object</code> already contains a <code>flags</code> entry with
the indicated name, then it is returned unaltered, and a warning
is issued.
</p>


<h3>Value</h3>

<p>An object with the <code>flags</code> item within the
<code>metadata</code> slot set up as indicated.
</p>


<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>

<hr>
<h2 id='initializeFlags+2Coce-method'>Create and Initialize oce Flags</h2><span id='topic+initializeFlags+2Coce-method'></span>

<h3>Description</h3>

<p>This function creates an item for a named variable within
the <code>flags</code> entry in the object's <code>metadata</code> slot.
The purpose is both to document a flag scheme
and to make it so that <code><a href="#topic+initializeFlags">initializeFlags()</a></code>
and <code><a href="#topic+setFlags">setFlags()</a></code> can specify flags by
name, in addition to number. A generic function, it is
specialized for some classes via interpretation of the
<code>scheme</code> argument (see &ldquo;Details&rdquo;, for those
object classes that have such specializations).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'oce'
initializeFlags(
  object,
  name = NULL,
  value = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeFlags+2B2Coce-method_+3A_object">object</code></td>
<td>
<p>An <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="initializeFlags+2B2Coce-method_+3A_name">name</code></td>
<td>
<p>Character value indicating the name of a variable within the
<code>data</code> slot of <code>object</code>.</p>
</td></tr>
<tr><td><code id="initializeFlags+2B2Coce-method_+3A_value">value</code></td>
<td>
<p>Numerical or character value to be stored in the newly-created
entry within <code>flags</code>. (A character value will only work if
<code><a href="#topic+initializeFlags">initializeFlags()</a></code> has been used first on <code>object</code>.)</p>
</td></tr>
<tr><td><code id="initializeFlags+2B2Coce-method_+3A_debug">debug</code></td>
<td>
<p>Integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>object</code> already contains a <code>flags</code> entry with
the indicated name, then it is returned unaltered, and a warning
is issued.
</p>


<h3>Value</h3>

<p>An object with the <code>flags</code> item within the
<code>metadata</code> slot set up as indicated.
</p>


<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>

<hr>
<h2 id='initializeFlagScheme'>Establish a Data-Quality Scheme for a oce Object</h2><span id='topic+initializeFlagScheme'></span>

<h3>Description</h3>

<p>This function adds an item named <code>flagScheme</code>
to the <code>metadata</code> slot of an object inheriting from
<a href="#topic+oce-class">oce</a>. This is a list containing two
items: <code>name</code> and <code>mapping</code>, as provided in the
function arguments.
The purpose is both to document a flag scheme
and to make it so that <code><a href="#topic+initializeFlags">initializeFlags()</a></code>,
<code><a href="#topic+setFlags">setFlags()</a></code> and <code><a href="#topic+handleFlags">handleFlags()</a></code>
can specify flags by
name, as opposed to number. This is a generic function,
that may be specialized to the class of <code>object</code>
(see &ldquo;Details&rdquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initializeFlagScheme(
  object,
  name = NULL,
  mapping = NULL,
  default = NULL,
  update = NULL,
  debug = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeFlagScheme_+3A_object">object</code></td>
<td>
<p>An oce object.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme_+3A_name">name</code></td>
<td>
<p>a character value naming the scheme. If this refers
to a pre-defined scheme, then <code>mapping</code> must not be provided,
because doing so would contradict the pre-defined scheme, defeating
its purpose of providing concreteness and clarity.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme_+3A_mapping">mapping</code></td>
<td>
<p>a list of named items describing the mapping from
flag meaning to flag numerical value, e.g <code>list(good=1, bad=2)</code>
might be used for a hypothetical class.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme_+3A_default">default</code></td>
<td>
<p>an integer vector of flag values that are not considered
to be good. If this is not provided, but if <code>name</code> is <code>"argo"</code>,
<code>"BODC"</code>,
<code>"DFO"</code>,
<code>"WHP bottle"</code>, or
<code>"WHP CTD"</code>, then a conservative value will be set automatically,
equal to the list of flag values that designate bad or questionable data.
For example, for <code>name="WHP CTD"</code>, the setting will be
<code>c(1,3,4,5,6,7,9)</code>, leaving only value <code>2</code>, which corresponds
with &quot;acceptable&quot; in the notation used for that flag scheme.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme_+3A_update">update</code></td>
<td>
<p>a logical value indicating whether the scheme provided is
to update an existing scheme.  The default value, <code>FALSE</code>, prevents such
an attempt to alter an existing flag scheme, if one is already embedded
in <code>object</code>.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme_+3A_debug">debug</code></td>
<td>
<p>an integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following pre-defined schemes are available (note that the
names are simplified from the phrases used in defining
documentation):
</p>

<ul>
<li> <p><code>name="argo"</code> defaults <code>mapping</code> to
OLD (prior to June 10, 2020)
</p>
</li></ul>

<div class="sourceCode"><pre>list(not_assessed=0, passed_all_tests=1, probably_good=2,
     probably_bad=3, bad=4, averaged=7,
     interpolated=8, missing=9)
</pre></div>
<p>NEW (after June 10, 2020)
</p>
<div class="sourceCode"><pre>list(not_assessed=0, passed_all_tests=1, probably_good=2,
     probably_bad=3, bad=4, changed=5, not_used_6=6, not_used_7=7,
     estimated=8, missing=9)
</pre></div>
<p>See reference 1 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="BODC"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_quality_control=0, good=1, probably_good=2,
     probably_bad=3, bad=4, changed=5,
     below_detection=6, in_excess=7, interpolated=8,
     missing=9)
</pre></div>
<p>See reference 2 for a deeper explanation of the meanings of these codes,
and note that codes <code>A</code> and <code>Q</code> are not provided in
oce.
</p>

<ul>
<li> <p><code>name="DFO"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_quality_control=0, appears_correct=1, appears_inconsistent=2,
     doubtful=3, erroneous=4, changed=5,
     qc_by_originator=8, missing=9)
</pre></div>
<p>See reference 3 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="WHP bottle"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_information=1, no_problems_noted=2, leaking=3,
     did_not_trip=4, not_reported=5, discrepency=6,
     unknown_problem=7, did_not_trip=8, no_sample=9)
</pre></div>
<p>See reference 4 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="WHP CTD"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(not_calibrated=1, acceptable=2, questionable=3,
    bad=4, not_reported=5, interpolated=6,
    despiked=7, missing=9)
</pre></div>
<p>See reference 4 for a deeper explanation of the meanings of these codes.
</p>


<h3>Value</h3>

<p>An object with the <code>metadata</code> slot containing <code>flagScheme</code>.
</p>


<h3>References</h3>


<ol>
<li><p> The codes for <code>"argo"</code> are derived from information in Table 4.1
of Wong, Annie, Robert Keeley, Thierry Carval, and Argo Data Management Team
(8 January 2020), &quot;Argo Quality Control Manual for CTD and Trajectory Data, Version 3.3,&quot;
available at <code style="white-space: pre;">&#8288;https://archimer.ifremer.fr/doc/00228/33951/&#8288;</code>
as of June 2020.
</p>
</li>
<li><p> The codes for <code>"BODC"</code> are defined at
http://seadatanet.maris2.nl/v_bodc_vocab_v2/browse.asp?order=conceptid&amp;formname=search&amp;screen=0&amp;lib=l20
</p>
</li>
<li><p> The codes for <code>"DFO"</code> are defined at
http://www.dfo-mpo.gc.ca/science/data-donnees/code/list/014-eng.html
</p>
</li>
<li><p> The codes for <code>"WHP CTD"</code> and <code>"WHP bottle"</code> are defined at
https://www.nodc.noaa.gov/woce/woce_v3/wocedata_1/whp/exchange/exchange_format_desc.htm
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>
<p>Other things related to oce data: 
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>
</p>

<hr>
<h2 id='initializeFlagScheme+2Cctd-method'>Establish a Data-Quality Scheme for a ctd Object</h2><span id='topic+initializeFlagScheme+2Cctd-method'></span>

<h3>Description</h3>

<p>This function adds an item named <code>flagScheme</code>
to the <code>metadata</code> slot of an object inheriting from
<a href="#topic+ctd-class">ctd</a>. This is a list containing two
items: <code>name</code> and <code>mapping</code>, as provided in the
function arguments.
The purpose is both to document a flag scheme
and to make it so that <code><a href="#topic+initializeFlags">initializeFlags()</a></code>,
<code><a href="#topic+setFlags">setFlags()</a></code> and <code><a href="#topic+handleFlags">handleFlags()</a></code>
can specify flags by
name, as opposed to number. This is a generic function,
that may be specialized to the class of <code>object</code>
(see &ldquo;Details&rdquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ctd'
initializeFlagScheme(
  object,
  name = NULL,
  mapping = NULL,
  default = NULL,
  update = NULL,
  debug = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeFlagScheme+2B2Cctd-method_+3A_object">object</code></td>
<td>
<p>An oce object.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Cctd-method_+3A_name">name</code></td>
<td>
<p>a character value naming the scheme. If this refers
to a pre-defined scheme, then <code>mapping</code> must not be provided,
because doing so would contradict the pre-defined scheme, defeating
its purpose of providing concreteness and clarity.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Cctd-method_+3A_mapping">mapping</code></td>
<td>
<p>a list of named items describing the mapping from
flag meaning to flag numerical value, e.g <code>list(good=1, bad=2)</code>
might be used for a hypothetical class.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Cctd-method_+3A_default">default</code></td>
<td>
<p>an integer vector of flag values that are not considered
to be good. If this is not provided, but if <code>name</code> is <code>"argo"</code>,
<code>"BODC"</code>,
<code>"DFO"</code>,
<code>"WHP bottle"</code>, or
<code>"WHP CTD"</code>, then a conservative value will be set automatically,
equal to the list of flag values that designate bad or questionable data.
For example, for <code>name="WHP CTD"</code>, the setting will be
<code>c(1,3,4,5,6,7,9)</code>, leaving only value <code>2</code>, which corresponds
with &quot;acceptable&quot; in the notation used for that flag scheme.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Cctd-method_+3A_update">update</code></td>
<td>
<p>a logical value indicating whether the scheme provided is
to update an existing scheme.  The default value, <code>FALSE</code>, prevents such
an attempt to alter an existing flag scheme, if one is already embedded
in <code>object</code>.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Cctd-method_+3A_debug">debug</code></td>
<td>
<p>an integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following pre-defined schemes are available (note that the
names are simplified from the phrases used in defining
documentation):
</p>

<ul>
<li> <p><code>name="argo"</code> defaults <code>mapping</code> to
OLD (prior to June 10, 2020)
</p>
</li></ul>

<div class="sourceCode"><pre>list(not_assessed=0, passed_all_tests=1, probably_good=2,
     probably_bad=3, bad=4, averaged=7,
     interpolated=8, missing=9)
</pre></div>
<p>NEW (after June 10, 2020)
</p>
<div class="sourceCode"><pre>list(not_assessed=0, passed_all_tests=1, probably_good=2,
     probably_bad=3, bad=4, changed=5, not_used_6=6, not_used_7=7,
     estimated=8, missing=9)
</pre></div>
<p>See reference 1 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="BODC"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_quality_control=0, good=1, probably_good=2,
     probably_bad=3, bad=4, changed=5,
     below_detection=6, in_excess=7, interpolated=8,
     missing=9)
</pre></div>
<p>See reference 2 for a deeper explanation of the meanings of these codes,
and note that codes <code>A</code> and <code>Q</code> are not provided in
oce.
</p>

<ul>
<li> <p><code>name="DFO"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_quality_control=0, appears_correct=1, appears_inconsistent=2,
     doubtful=3, erroneous=4, changed=5,
     qc_by_originator=8, missing=9)
</pre></div>
<p>See reference 3 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="WHP bottle"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_information=1, no_problems_noted=2, leaking=3,
     did_not_trip=4, not_reported=5, discrepency=6,
     unknown_problem=7, did_not_trip=8, no_sample=9)
</pre></div>
<p>See reference 4 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="WHP CTD"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(not_calibrated=1, acceptable=2, questionable=3,
    bad=4, not_reported=5, interpolated=6,
    despiked=7, missing=9)
</pre></div>
<p>See reference 4 for a deeper explanation of the meanings of these codes.
</p>


<h3>Value</h3>

<p>An object with the <code>metadata</code> slot containing <code>flagScheme</code>.
</p>


<h3>References</h3>


<ol>
<li><p> The codes for <code>"argo"</code> are derived from information in Table 4.1
of Wong, Annie, Robert Keeley, Thierry Carval, and Argo Data Management Team
(8 January 2020), &quot;Argo Quality Control Manual for CTD and Trajectory Data, Version 3.3,&quot;
available at <code style="white-space: pre;">&#8288;https://archimer.ifremer.fr/doc/00228/33951/&#8288;</code>
as of June 2020.
</p>
</li>
<li><p> The codes for <code>"BODC"</code> are defined at
http://seadatanet.maris2.nl/v_bodc_vocab_v2/browse.asp?order=conceptid&amp;formname=search&amp;screen=0&amp;lib=l20
</p>
</li>
<li><p> The codes for <code>"DFO"</code> are defined at
http://www.dfo-mpo.gc.ca/science/data-donnees/code/list/014-eng.html
</p>
</li>
<li><p> The codes for <code>"WHP CTD"</code> and <code>"WHP bottle"</code> are defined at
https://www.nodc.noaa.gov/woce/woce_v3/wocedata_1/whp/exchange/exchange_format_desc.htm
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>

<hr>
<h2 id='initializeFlagScheme+2Coce-method'>Establish a Data-Quality Scheme for a oce Object</h2><span id='topic+initializeFlagScheme+2Coce-method'></span>

<h3>Description</h3>

<p>This function adds an item named <code>flagScheme</code>
to the <code>metadata</code> slot of an object inheriting from
<a href="#topic+oce-class">oce</a>. This is a list containing two
items: <code>name</code> and <code>mapping</code>, as provided in the
function arguments.
The purpose is both to document a flag scheme
and to make it so that <code><a href="#topic+initializeFlags">initializeFlags()</a></code>,
<code><a href="#topic+setFlags">setFlags()</a></code> and <code><a href="#topic+handleFlags">handleFlags()</a></code>
can specify flags by
name, as opposed to number. This is a generic function,
that may be specialized to the class of <code>object</code>
(see &ldquo;Details&rdquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'oce'
initializeFlagScheme(
  object,
  name = NULL,
  mapping = NULL,
  default = NULL,
  update = NULL,
  debug = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeFlagScheme+2B2Coce-method_+3A_object">object</code></td>
<td>
<p>An oce object.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Coce-method_+3A_name">name</code></td>
<td>
<p>a character value naming the scheme. If this refers
to a pre-defined scheme, then <code>mapping</code> must not be provided,
because doing so would contradict the pre-defined scheme, defeating
its purpose of providing concreteness and clarity.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Coce-method_+3A_mapping">mapping</code></td>
<td>
<p>a list of named items describing the mapping from
flag meaning to flag numerical value, e.g <code>list(good=1, bad=2)</code>
might be used for a hypothetical class.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Coce-method_+3A_default">default</code></td>
<td>
<p>an integer vector of flag values that are not considered
to be good. If this is not provided, but if <code>name</code> is <code>"argo"</code>,
<code>"BODC"</code>,
<code>"DFO"</code>,
<code>"WHP bottle"</code>, or
<code>"WHP CTD"</code>, then a conservative value will be set automatically,
equal to the list of flag values that designate bad or questionable data.
For example, for <code>name="WHP CTD"</code>, the setting will be
<code>c(1,3,4,5,6,7,9)</code>, leaving only value <code>2</code>, which corresponds
with &quot;acceptable&quot; in the notation used for that flag scheme.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Coce-method_+3A_update">update</code></td>
<td>
<p>a logical value indicating whether the scheme provided is
to update an existing scheme.  The default value, <code>FALSE</code>, prevents such
an attempt to alter an existing flag scheme, if one is already embedded
in <code>object</code>.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Coce-method_+3A_debug">debug</code></td>
<td>
<p>an integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following pre-defined schemes are available (note that the
names are simplified from the phrases used in defining
documentation):
</p>

<ul>
<li> <p><code>name="argo"</code> defaults <code>mapping</code> to
OLD (prior to June 10, 2020)
</p>
</li></ul>

<div class="sourceCode"><pre>list(not_assessed=0, passed_all_tests=1, probably_good=2,
     probably_bad=3, bad=4, averaged=7,
     interpolated=8, missing=9)
</pre></div>
<p>NEW (after June 10, 2020)
</p>
<div class="sourceCode"><pre>list(not_assessed=0, passed_all_tests=1, probably_good=2,
     probably_bad=3, bad=4, changed=5, not_used_6=6, not_used_7=7,
     estimated=8, missing=9)
</pre></div>
<p>See reference 1 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="BODC"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_quality_control=0, good=1, probably_good=2,
     probably_bad=3, bad=4, changed=5,
     below_detection=6, in_excess=7, interpolated=8,
     missing=9)
</pre></div>
<p>See reference 2 for a deeper explanation of the meanings of these codes,
and note that codes <code>A</code> and <code>Q</code> are not provided in
oce.
</p>

<ul>
<li> <p><code>name="DFO"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_quality_control=0, appears_correct=1, appears_inconsistent=2,
     doubtful=3, erroneous=4, changed=5,
     qc_by_originator=8, missing=9)
</pre></div>
<p>See reference 3 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="WHP bottle"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_information=1, no_problems_noted=2, leaking=3,
     did_not_trip=4, not_reported=5, discrepency=6,
     unknown_problem=7, did_not_trip=8, no_sample=9)
</pre></div>
<p>See reference 4 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="WHP CTD"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(not_calibrated=1, acceptable=2, questionable=3,
    bad=4, not_reported=5, interpolated=6,
    despiked=7, missing=9)
</pre></div>
<p>See reference 4 for a deeper explanation of the meanings of these codes.
</p>


<h3>Value</h3>

<p>An object with the <code>metadata</code> slot containing <code>flagScheme</code>.
</p>


<h3>References</h3>


<ol>
<li><p> The codes for <code>"argo"</code> are derived from information in Table 4.1
of Wong, Annie, Robert Keeley, Thierry Carval, and Argo Data Management Team
(8 January 2020), &quot;Argo Quality Control Manual for CTD and Trajectory Data, Version 3.3,&quot;
available at <code style="white-space: pre;">&#8288;https://archimer.ifremer.fr/doc/00228/33951/&#8288;</code>
as of June 2020.
</p>
</li>
<li><p> The codes for <code>"BODC"</code> are defined at
http://seadatanet.maris2.nl/v_bodc_vocab_v2/browse.asp?order=conceptid&amp;formname=search&amp;screen=0&amp;lib=l20
</p>
</li>
<li><p> The codes for <code>"DFO"</code> are defined at
http://www.dfo-mpo.gc.ca/science/data-donnees/code/list/014-eng.html
</p>
</li>
<li><p> The codes for <code>"WHP CTD"</code> and <code>"WHP bottle"</code> are defined at
https://www.nodc.noaa.gov/woce/woce_v3/wocedata_1/whp/exchange/exchange_format_desc.htm
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>
<p>Other things related to oce data: 
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>
</p>

<hr>
<h2 id='initializeFlagScheme+2Csection-method'>Establish a Data-Quality Scheme for a section Object</h2><span id='topic+initializeFlagScheme+2Csection-method'></span>

<h3>Description</h3>

<p>This function adds an item named <code>flagScheme</code>
to the <code>metadata</code> slot of an object inheriting from
<a href="#topic+section-class">section</a>. This is a list containing two
items: <code>name</code> and <code>mapping</code>, as provided in the
function arguments.
The purpose is both to document a flag scheme
and to make it so that <code><a href="#topic+initializeFlags">initializeFlags()</a></code>,
<code><a href="#topic+setFlags">setFlags()</a></code> and <code><a href="#topic+handleFlags">handleFlags()</a></code>
can specify flags by
name, as opposed to number. This is a generic function,
that may be specialized to the class of <code>object</code>
(see &ldquo;Details&rdquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'section'
initializeFlagScheme(
  object,
  name = NULL,
  mapping = NULL,
  default = NULL,
  update = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeFlagScheme+2B2Csection-method_+3A_object">object</code></td>
<td>
<p>An oce object.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Csection-method_+3A_name">name</code></td>
<td>
<p>a character value naming the scheme. If this refers
to a pre-defined scheme, then <code>mapping</code> must not be provided,
because doing so would contradict the pre-defined scheme, defeating
its purpose of providing concreteness and clarity.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Csection-method_+3A_mapping">mapping</code></td>
<td>
<p>a list of named items describing the mapping from
flag meaning to flag numerical value, e.g <code>list(good=1, bad=2)</code>
might be used for a hypothetical class.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Csection-method_+3A_default">default</code></td>
<td>
<p>an integer vector of flag values that are not considered
to be good. If this is not provided, but if <code>name</code> is <code>"argo"</code>,
<code>"BODC"</code>,
<code>"DFO"</code>,
<code>"WHP bottle"</code>, or
<code>"WHP CTD"</code>, then a conservative value will be set automatically,
equal to the list of flag values that designate bad or questionable data.
For example, for <code>name="WHP CTD"</code>, the setting will be
<code>c(1,3,4,5,6,7,9)</code>, leaving only value <code>2</code>, which corresponds
with &quot;acceptable&quot; in the notation used for that flag scheme.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Csection-method_+3A_update">update</code></td>
<td>
<p>a logical value indicating whether the scheme provided is
to update an existing scheme.  The default value, <code>FALSE</code>, prevents such
an attempt to alter an existing flag scheme, if one is already embedded
in <code>object</code>.</p>
</td></tr>
<tr><td><code id="initializeFlagScheme+2B2Csection-method_+3A_debug">debug</code></td>
<td>
<p>an integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following pre-defined schemes are available (note that the
names are simplified from the phrases used in defining
documentation):
</p>

<ul>
<li> <p><code>name="argo"</code> defaults <code>mapping</code> to
OLD (prior to June 10, 2020)
</p>
</li></ul>

<div class="sourceCode"><pre>list(not_assessed=0, passed_all_tests=1, probably_good=2,
     probably_bad=3, bad=4, averaged=7,
     interpolated=8, missing=9)
</pre></div>
<p>NEW (after June 10, 2020)
</p>
<div class="sourceCode"><pre>list(not_assessed=0, passed_all_tests=1, probably_good=2,
     probably_bad=3, bad=4, changed=5, not_used_6=6, not_used_7=7,
     estimated=8, missing=9)
</pre></div>
<p>See reference 1 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="BODC"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_quality_control=0, good=1, probably_good=2,
     probably_bad=3, bad=4, changed=5,
     below_detection=6, in_excess=7, interpolated=8,
     missing=9)
</pre></div>
<p>See reference 2 for a deeper explanation of the meanings of these codes,
and note that codes <code>A</code> and <code>Q</code> are not provided in
oce.
</p>

<ul>
<li> <p><code>name="DFO"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_quality_control=0, appears_correct=1, appears_inconsistent=2,
     doubtful=3, erroneous=4, changed=5,
     qc_by_originator=8, missing=9)
</pre></div>
<p>See reference 3 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="WHP bottle"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_information=1, no_problems_noted=2, leaking=3,
     did_not_trip=4, not_reported=5, discrepency=6,
     unknown_problem=7, did_not_trip=8, no_sample=9)
</pre></div>
<p>See reference 4 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="WHP CTD"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(not_calibrated=1, acceptable=2, questionable=3,
    bad=4, not_reported=5, interpolated=6,
    despiked=7, missing=9)
</pre></div>
<p>See reference 4 for a deeper explanation of the meanings of these codes.
</p>


<h3>Value</h3>

<p>An object with the <code>metadata</code> slot containing <code>flagScheme</code>.
</p>


<h3>Sample of Usage</h3>

<pre>
data(section)
section &lt;- read.section("a03_hy1.csv", sectionId="a03", institute="SIO",
   ship="R/V Professor Multanovskiy", scientist="Vladimir Tereschenov")
sectionWithFlags &lt;- initializeFlagScheme(section, "WHP bottle")
station1 &lt;- sectionWithFlags[["station", 1]]
str(station1[["flagScheme"]])
</pre>


<h3>References</h3>


<ol>
<li><p> The codes for <code>"argo"</code> are derived from information in Table 4.1
of Wong, Annie, Robert Keeley, Thierry Carval, and Argo Data Management Team
(8 January 2020), &quot;Argo Quality Control Manual for CTD and Trajectory Data, Version 3.3,&quot;
available at <code style="white-space: pre;">&#8288;https://archimer.ifremer.fr/doc/00228/33951/&#8288;</code>
as of June 2020.
</p>
</li>
<li><p> The codes for <code>"BODC"</code> are defined at
http://seadatanet.maris2.nl/v_bodc_vocab_v2/browse.asp?order=conceptid&amp;formname=search&amp;screen=0&amp;lib=l20
</p>
</li>
<li><p> The codes for <code>"DFO"</code> are defined at
http://www.dfo-mpo.gc.ca/science/data-donnees/code/list/014-eng.html
</p>
</li>
<li><p> The codes for <code>"WHP CTD"</code> and <code>"WHP bottle"</code> are defined at
https://www.nodc.noaa.gov/woce/woce_v3/wocedata_1/whp/exchange/exchange_format_desc.htm
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>
<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>

<hr>
<h2 id='initializeFlagSchemeInternal'>Establish a Data-Quality Scheme for a oce Object</h2><span id='topic+initializeFlagSchemeInternal'></span>

<h3>Description</h3>

<p>This function adds an item named <code>flagScheme</code>
to the <code>metadata</code> slot of an object inheriting from
<a href="#topic+oce-class">oce</a>. This is a list containing two
items: <code>name</code> and <code>mapping</code>, as provided in the
function arguments.
The purpose is both to document a flag scheme
and to make it so that <code><a href="#topic+initializeFlags">initializeFlags()</a></code>,
<code><a href="#topic+setFlags">setFlags()</a></code> and <code><a href="#topic+handleFlags">handleFlags()</a></code>
can specify flags by
name, as opposed to number. This is a generic function,
that may be specialized to the class of <code>object</code>
(see &ldquo;Details&rdquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initializeFlagSchemeInternal(
  object,
  name = NULL,
  mapping = NULL,
  default = NULL,
  update = NULL,
  debug = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeFlagSchemeInternal_+3A_object">object</code></td>
<td>
<p>An oce object.</p>
</td></tr>
<tr><td><code id="initializeFlagSchemeInternal_+3A_name">name</code></td>
<td>
<p>a character value naming the scheme. If this refers
to a pre-defined scheme, then <code>mapping</code> must not be provided,
because doing so would contradict the pre-defined scheme, defeating
its purpose of providing concreteness and clarity.</p>
</td></tr>
<tr><td><code id="initializeFlagSchemeInternal_+3A_mapping">mapping</code></td>
<td>
<p>a list of named items describing the mapping from
flag meaning to flag numerical value, e.g <code>list(good=1, bad=2)</code>
might be used for a hypothetical class.</p>
</td></tr>
<tr><td><code id="initializeFlagSchemeInternal_+3A_default">default</code></td>
<td>
<p>an integer vector of flag values that are not considered
to be good. If this is not provided, but if <code>name</code> is <code>"argo"</code>,
<code>"BODC"</code>,
<code>"DFO"</code>,
<code>"WHP bottle"</code>, or
<code>"WHP CTD"</code>, then a conservative value will be set automatically,
equal to the list of flag values that designate bad or questionable data.
For example, for <code>name="WHP CTD"</code>, the setting will be
<code>c(1,3,4,5,6,7,9)</code>, leaving only value <code>2</code>, which corresponds
with &quot;acceptable&quot; in the notation used for that flag scheme.</p>
</td></tr>
<tr><td><code id="initializeFlagSchemeInternal_+3A_update">update</code></td>
<td>
<p>a logical value indicating whether the scheme provided is
to update an existing scheme.  The default value, <code>FALSE</code>, prevents such
an attempt to alter an existing flag scheme, if one is already embedded
in <code>object</code>.</p>
</td></tr>
<tr><td><code id="initializeFlagSchemeInternal_+3A_debug">debug</code></td>
<td>
<p>an integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following pre-defined schemes are available (note that the
names are simplified from the phrases used in defining
documentation):
</p>

<ul>
<li> <p><code>name="argo"</code> defaults <code>mapping</code> to
OLD (prior to June 10, 2020)
</p>
</li></ul>

<div class="sourceCode"><pre>list(not_assessed=0, passed_all_tests=1, probably_good=2,
     probably_bad=3, bad=4, averaged=7,
     interpolated=8, missing=9)
</pre></div>
<p>NEW (after June 10, 2020)
</p>
<div class="sourceCode"><pre>list(not_assessed=0, passed_all_tests=1, probably_good=2,
     probably_bad=3, bad=4, changed=5, not_used_6=6, not_used_7=7,
     estimated=8, missing=9)
</pre></div>
<p>See reference 1 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="BODC"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_quality_control=0, good=1, probably_good=2,
     probably_bad=3, bad=4, changed=5,
     below_detection=6, in_excess=7, interpolated=8,
     missing=9)
</pre></div>
<p>See reference 2 for a deeper explanation of the meanings of these codes,
and note that codes <code>A</code> and <code>Q</code> are not provided in
oce.
</p>

<ul>
<li> <p><code>name="DFO"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_quality_control=0, appears_correct=1, appears_inconsistent=2,
     doubtful=3, erroneous=4, changed=5,
     qc_by_originator=8, missing=9)
</pre></div>
<p>See reference 3 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="WHP bottle"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(no_information=1, no_problems_noted=2, leaking=3,
     did_not_trip=4, not_reported=5, discrepency=6,
     unknown_problem=7, did_not_trip=8, no_sample=9)
</pre></div>
<p>See reference 4 for a deeper explanation of the meanings of these codes.
</p>

<ul>
<li> <p><code>name="WHP CTD"</code> defaults <code>mapping</code> to
</p>
</li></ul>

<div class="sourceCode"><pre>list(not_calibrated=1, acceptable=2, questionable=3,
    bad=4, not_reported=5, interpolated=6,
    despiked=7, missing=9)
</pre></div>
<p>See reference 4 for a deeper explanation of the meanings of these codes.
</p>


<h3>Value</h3>

<p>An object with the <code>metadata</code> slot containing <code>flagScheme</code>.
</p>


<h3>References</h3>


<ol>
<li><p> The codes for <code>"argo"</code> are derived from information in Table 4.1
of Wong, Annie, Robert Keeley, Thierry Carval, and Argo Data Management Team
(8 January 2020), &quot;Argo Quality Control Manual for CTD and Trajectory Data, Version 3.3,&quot;
available at <code style="white-space: pre;">&#8288;https://archimer.ifremer.fr/doc/00228/33951/&#8288;</code>
as of June 2020.
</p>
</li>
<li><p> The codes for <code>"BODC"</code> are defined at
http://seadatanet.maris2.nl/v_bodc_vocab_v2/browse.asp?order=conceptid&amp;formname=search&amp;screen=0&amp;lib=l20
</p>
</li>
<li><p> The codes for <code>"DFO"</code> are defined at
http://www.dfo-mpo.gc.ca/science/data-donnees/code/list/014-eng.html
</p>
</li>
<li><p> The codes for <code>"WHP CTD"</code> and <code>"WHP bottle"</code> are defined at
https://www.nodc.noaa.gov/woce/woce_v3/wocedata_1/whp/exchange/exchange_format_desc.htm
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>
<p>Other things related to oce data: 
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>
</p>

<hr>
<h2 id='initializeFlagsInternal'>Create and Initialize oce Flags</h2><span id='topic+initializeFlagsInternal'></span>

<h3>Description</h3>

<p>This function creates an item for a named variable within
the <code>flags</code> entry in the object's <code>metadata</code> slot.
The purpose is both to document a flag scheme
and to make it so that <code><a href="#topic+initializeFlags">initializeFlags()</a></code>
and <code><a href="#topic+setFlags">setFlags()</a></code> can specify flags by
name, in addition to number. A generic function, it is
specialized for some classes via interpretation of the
<code>scheme</code> argument (see &ldquo;Details&rdquo;, for those
object classes that have such specializations).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initializeFlagsInternal(
  object,
  name = NULL,
  value = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="initializeFlagsInternal_+3A_object">object</code></td>
<td>
<p>An <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="initializeFlagsInternal_+3A_name">name</code></td>
<td>
<p>Character value indicating the name of a variable within the
<code>data</code> slot of <code>object</code>.</p>
</td></tr>
<tr><td><code id="initializeFlagsInternal_+3A_value">value</code></td>
<td>
<p>Numerical or character value to be stored in the newly-created
entry within <code>flags</code>. (A character value will only work if
<code><a href="#topic+initializeFlags">initializeFlags()</a></code> has been used first on <code>object</code>.)</p>
</td></tr>
<tr><td><code id="initializeFlagsInternal_+3A_debug">debug</code></td>
<td>
<p>Integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>object</code> already contains a <code>flags</code> entry with
the indicated name, then it is returned unaltered, and a warning
is issued.
</p>


<h3>Value</h3>

<p>An object with the <code>flags</code> item within the
<code>metadata</code> slot set up as indicated.
</p>


<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>

<hr>
<h2 id='integerToAscii'>Infer ASCII Code From an Integer Value</h2><span id='topic+integerToAscii'></span>

<h3>Description</h3>

<p>Infer ASCII Code From an Integer Value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>integerToAscii(i)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="integerToAscii_+3A_i">i</code></td>
<td>
<p>an integer, or integer vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character, or character vector.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
A &lt;- integerToAscii(65)
cat("A=", A, "\n")
</code></pre>

<hr>
<h2 id='integrateTrapezoid'>Trapezoidal Integration</h2><span id='topic+integrateTrapezoid'></span>

<h3>Description</h3>

<p>Estimate the integral of one-dimensional function using the trapezoidal
rule.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>integrateTrapezoid(x, y, type = c("A", "dA", "cA"), xmin, xmax)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="integrateTrapezoid_+3A_x">x</code>, <code id="integrateTrapezoid_+3A_y">y</code></td>
<td>
<p>vectors of x and y values. In the normal case, these
vectors are both supplied, and of equal length. There are also two
special cases. First, if <code>y</code> is missing, then
<code>x</code> is taken to be <code>y</code>, and a new <code>x</code> is constructed
as <code><a href="base.html#topic+seq_along">seq_along</a></code><code style="white-space: pre;">&#8288;(y)1. Second, if &#8288;</code>length(x)<code style="white-space: pre;">&#8288;is 1 and&#8288;</code>length(y)<code style="white-space: pre;">&#8288;exceeds 1, then&#8288;</code>x<code style="white-space: pre;">&#8288;is replaced by&#8288;</code>x*<code>[</code>seq_along<code style="white-space: pre;">&#8288;]&#8288;</code>(y)'.</p>
</td></tr>
<tr><td><code id="integrateTrapezoid_+3A_type">type</code></td>
<td>
<p>Flag indicating the desired return value (see &ldquo;Value&rdquo;).</p>
</td></tr>
<tr><td><code id="integrateTrapezoid_+3A_xmin">xmin</code>, <code id="integrateTrapezoid_+3A_xmax">xmax</code></td>
<td>
<p>Optional numbers indicating the range of the integration.
These values may be used to restrict the range of integration, or to
extend it; in either case, <code><a href="stats.html#topic+approx">approx()</a></code> with <code>rule=2</code>
is used to create new x and y vectors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>type="A"</code> (the default), a single value is returned,
containing the estimate of the integral of <code>y=y(x)</code>.  If
<code>type="dA"</code>, a numeric vector of the same length as <code>x</code>, of which
the first element is zero, the second element is the integral between
<code>x[1]</code> and <code>x[2]</code>, etc.  If <code>type="cA"</code>, the result is the
cumulative sum (as in <code><a href="base.html#topic+cumsum">cumsum()</a></code>) of the values that would be
returned for <code>type="dA"</code>.  See &ldquo;Examples&rdquo;.
</p>


<h3>Bugs</h3>

<p>There is no handling of <code>NA</code> values.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- seq(0, 1, length.out = 10) # try larger length.out to see if area approaches 2
y &lt;- 2 * x + 3 * x^2
A &lt;- integrateTrapezoid(x, y)
dA &lt;- integrateTrapezoid(x, y, "dA")
cA &lt;- integrateTrapezoid(x, y, "cA")
print(A)
print(sum(dA))
print(tail(cA, 1))
print(integrateTrapezoid(diff(x[1:2]), y))
print(integrateTrapezoid(y))
</code></pre>

<hr>
<h2 id='interpBarnes'>Grid Data Using the Barnes Algorithm</h2><span id='topic+interpBarnes'></span>

<h3>Description</h3>

<p>The algorithm follows that described by Koch et al. (1983), except
that <code>interpBarnes</code> adds (1) the ability to
blank out the grid where data are
sparse, using the <code>trim</code> argument, and (2) the ability to
pre-grid, with the <code>pregrid</code> argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>interpBarnes(
  x,
  y,
  z,
  w,
  xg,
  yg,
  xgl,
  ygl,
  xr,
  yr,
  gamma = 0.5,
  iterations = 2,
  trim = 0,
  pregrid = FALSE,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="interpBarnes_+3A_x">x</code>, <code id="interpBarnes_+3A_y">y</code></td>
<td>
<p>a vector of x and y locations.</p>
</td></tr>
<tr><td><code id="interpBarnes_+3A_z">z</code></td>
<td>
<p>a vector of z values, one at each (x,y) location.</p>
</td></tr>
<tr><td><code id="interpBarnes_+3A_w">w</code></td>
<td>
<p>a optional vector of weights at the (x,y) location.  If not
supplied, then a weight of 1 is used for each point, which means equal
weighting.  Higher weights give data points more influence. If <code>pregrid</code>
is <code>TRUE</code>, then any supplied value of <code>w</code> is ignored, and instead
each of the pregriddd points is given equal weight.</p>
</td></tr>
<tr><td><code id="interpBarnes_+3A_xg">xg</code>, <code id="interpBarnes_+3A_yg">yg</code></td>
<td>
<p>optional vectors defining the x and y grids.  If not supplied,
these values are inferred from the data, using e.g. <code>pretty(x, n=50)</code>.</p>
</td></tr>
<tr><td><code id="interpBarnes_+3A_xgl">xgl</code>, <code id="interpBarnes_+3A_ygl">ygl</code></td>
<td>
<p>optional lengths of the x and y grids, to be constructed with
<code><a href="base.html#topic+seq">seq()</a></code> spanning the data range.  These values <code>xgl</code> are only
examined if <code>xg</code> and <code>yg</code> are not supplied.</p>
</td></tr>
<tr><td><code id="interpBarnes_+3A_xr">xr</code>, <code id="interpBarnes_+3A_yr">yr</code></td>
<td>
<p>optional values defining the x and y radii of the weighting ellipse.
If not supplied, these are calculated as the span of x
and y over the square root of the number of data.</p>
</td></tr>
<tr><td><code id="interpBarnes_+3A_gamma">gamma</code></td>
<td>
<p>grid-focussing parameter.  At each successive iteration, <code>xr</code> and
<code>yr</code> are reduced by a factor of <code>sqrt(gamma)</code>.</p>
</td></tr>
<tr><td><code id="interpBarnes_+3A_iterations">iterations</code></td>
<td>
<p>number of iterations.  Set this to 1 to perform just
one iteration, using the radii as described at <code style="white-space: pre;">&#8288;xr,yr&#8288;</code> above.</p>
</td></tr>
<tr><td><code id="interpBarnes_+3A_trim">trim</code></td>
<td>
<p>a number between 0 and 1, indicating the quantile of data weight
to be used as a criterion for blanking out the gridded value (using
<code>NA</code>).  If 0, the whole <code>zg</code> grid is returned.  If &gt;0, any spots
on the grid where the data weight is less than the <code>trim</code>-th
<code><a href="stats.html#topic+quantile">quantile()</a></code> are set to <code>NA</code>.  See examples.</p>
</td></tr>
<tr><td><code id="interpBarnes_+3A_pregrid">pregrid</code></td>
<td>
<p>an indication of whether to pre-grid the data. If
<code>FALSE</code>, this is not done, i.e. conventional Barnes interpolation is
performed.  Otherwise, then the data are first averaged within grid cells
using <code><a href="#topic+binMean2D">binMean2D()</a></code>.  If <code>pregrid</code> is <code>TRUE</code> or
<code>4</code>, then this averaging is done within a grid that is 4 times finer
than the grid that will be used for the Barnes interpolation. Otherwise,
<code>pregrid</code> may be a single integer indicating the grid refinement (4
being the result if <code>TRUE</code> had been supplied), or a vector of two
integers, for the grid refinement in x and y. The purpose of using
<code>pregrid</code> is to speed processing on large datasets, and to remove
spatial bias (e.g. with a single station that is repeated frequently in an
otherwise seldom-sampled region).  A form of pregridding is done in the
World Ocean Atlas, for example.</p>
</td></tr>
<tr><td><code id="interpBarnes_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 0 for no debugging
information, to 1 for more, etc; the value is reduced by 1 for each
descendent function call.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing: <code>xg</code>, a vector holding the x-grid);
<code>yg</code>, a vector holding the y-grid; <code>zg</code>, a matrix holding the
gridded values; <code>wg</code>, a matrix holding the weights used in the
interpolation at its final iteration; and <code>zd</code>, a vector of the same
length as <code>x</code>, which holds the interpolated values at the data points.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>S. E.  Koch and M.  DesJardins and P. J. Kocin, 1983.  &ldquo;An
interactive Barnes objective map analysis scheme for use with satellite and
conventional data,&rdquo; <em>J.  Climate Appl.  Met.</em>, vol 22, p. 1487-1503.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+wind">wind()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# 1. contouring example, with wind-speed data from Koch et al. (1983)
data(wind)
u &lt;- interpBarnes(wind$x, wind$y, wind$z)
contour(u$xg, u$yg, u$zg, labcex = 1)
text(wind$x, wind$y, wind$z, cex = 0.7, col = "blue")
title("Numbers are the data")

# 2. As 1, but blank out spots where data are sparse
u &lt;- interpBarnes(wind$x, wind$y, wind$z, trim = 0.1)
contour(u$xg, u$yg, u$zg, level = seq(0, 30, 1))
points(wind$x, wind$y, cex = 1.5, pch = 20, col = "blue")

# 3. As 1, but interpolate back to points, and display the percent mismatch
u &lt;- interpBarnes(wind$x, wind$y, wind$z)
contour(u$xg, u$yg, u$zg, labcex = 1)
mismatch &lt;- 100 * (wind$z - u$zd) / wind$z
text(wind$x, wind$y, round(mismatch), col = "blue")
title("Numbers are percent mismatch between grid and data")

# 4. As 3, but contour the mismatch
mismatchGrid &lt;- interpBarnes(wind$x, wind$y, mismatch)
contour(mismatchGrid$xg, mismatchGrid$yg, mismatchGrid$zg, labcex = 1)

# 5. One-dimensional example, smoothing a salinity profile
data(ctd)
p &lt;- ctd[["pressure"]]
y &lt;- rep(1, length(p)) # fake y data, with arbitrary value
S &lt;- ctd[["salinity"]]
pg &lt;- pretty(p, n = 100)
g &lt;- interpBarnes(p, y, S, xg = pg, xr = 1)
plot(S, p, cex = 0.5, col = "blue", ylim = rev(range(p)))
lines(g$zg, g$xg, col = "red")
</code></pre>

<hr>
<h2 id='is.ad2cp'>Test Whether Item is a ad2cp-Type adp Object</h2><span id='topic+is.ad2cp'></span>

<h3>Description</h3>

<p>Test Whether Item is a ad2cp-Type adp Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.ad2cp(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.ad2cp_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce">oce</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical value indicating whether <code>x</code> is an <a href="#topic+adp-class">adp</a> object,
with <code>fileType</code> in its <code>metadata</code> slot equal to <code>"AD2CP"</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to ad2cp data: 
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>
</p>

<hr>
<h2 id='julianCenturyAnomaly'>Convert Julian-Day-Number to Julian Century</h2><span id='topic+julianCenturyAnomaly'></span>

<h3>Description</h3>

<p>Convert a Julian-Day number to a time in julian centuries since noon on
January 1, 1900.
The method follows reference 1 equation 15.1.  The example reproduces the
example provided by reference 1 example 15.a, with fractional error 3e-8.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>julianCenturyAnomaly(jd)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="julianCenturyAnomaly_+3A_jd">jd</code></td>
<td>
<p>a julian day number, e.g. as given by <code><a href="#topic+julianDay">julianDay()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Julian century since noon on January 1, 1900.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ul>
<li><p> Meeus, Jean. Astronomical Formulas for Calculators. Second Edition.
Richmond, Virginia, USA: Willmann-Bell, 1982.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other things related to astronomy: 
<code><a href="#topic+angle2hms">angle2hms</a>()</code>,
<code><a href="#topic+eclipticalToEquatorial">eclipticalToEquatorial</a>()</code>,
<code><a href="#topic+equatorialToLocalHorizontal">equatorialToLocalHorizontal</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+moonAngle">moonAngle</a>()</code>,
<code><a href="#topic+siderealTime">siderealTime</a>()</code>,
<code><a href="#topic+sunAngle">sunAngle</a>()</code>,
<code><a href="#topic+sunDeclinationRightAscension">sunDeclinationRightAscension</a>()</code>
</p>
<p>Other things related to time: 
<code><a href="#topic+ctimeToSeconds">ctimeToSeconds</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+numberAsHMS">numberAsHMS</a>()</code>,
<code><a href="#topic+numberAsPOSIXct">numberAsPOSIXct</a>()</code>,
<code><a href="#topic+secondsToCtime">secondsToCtime</a>()</code>,
<code><a href="#topic+unabbreviateYear">unabbreviateYear</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
t &lt;- ISOdatetime(1978, 11, 13, 4, 35, 0, tz = "UTC")
jca &lt;- julianCenturyAnomaly(julianDay(t))
cat(format(t), "is Julian Century anomaly", format(jca, digits = 8), "\n")

</code></pre>

<hr>
<h2 id='julianDay'>Convert a Time to a Julian Day</h2><span id='topic+julianDay'></span>

<h3>Description</h3>

<p>Convert a POSIXt time (given as either the <code>t</code> argument
or as the <code>year</code>, <code>month</code>, and other arguments) to a Julian day,
using the method provided in
Chapter 3 of Meeus (1982).  It should be noted that
Meeus and other astronomical treatments use fractional days, whereas the
present code follows the R convention of specifying days in whole numbers,
with hours, minutes, and seconds also provided as necessary.  Conversion is
simple, as illustrated in the example for 1977 April 26.4, for which Meeus
calculates julian day 2443259.9.  Note that the R documentation for
<code><a href="base.html#topic+julian">julian()</a></code> suggests another formula, but the point of the present
function is to match the other Meeus formulae, so that suggestion is ignored
here.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>julianDay(
  t,
  year = NA,
  month = NA,
  day = NA,
  hour = NA,
  min = NA,
  sec = NA,
  tz = "UTC"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="julianDay_+3A_t">t</code></td>
<td>
<p>a time, in POSIXt format, e.g. as created by
<code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>, <code><a href="base.html#topic+as.POSIXlt">as.POSIXlt()</a></code>, or
<code><a href="#topic+numberAsPOSIXct">numberAsPOSIXct()</a></code>, or a character string that can be
converted to a time using <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>.  If <code>t</code> is provided,
the other arguments are ignored.</p>
</td></tr>
<tr><td><code id="julianDay_+3A_year">year</code></td>
<td>
<p>year, to be provided along with <code>month</code>, etc., if <code>t</code>
is not provided.</p>
</td></tr>
<tr><td><code id="julianDay_+3A_month">month</code></td>
<td>
<p>numerical value for the month, with January being 1.
(This is required if <code>t</code> is not provided.)</p>
</td></tr>
<tr><td><code id="julianDay_+3A_day">day</code></td>
<td>
<p>numerical value for day in month, starting at 1.
(This is required if <code>t</code> is not provided.)</p>
</td></tr>
<tr><td><code id="julianDay_+3A_hour">hour</code></td>
<td>
<p>numerical value for hour of day, in range 0 to 24.
(This is required if <code>t</code> is not provided.)</p>
</td></tr>
<tr><td><code id="julianDay_+3A_min">min</code></td>
<td>
<p>numerical value of the minute of the hour.
(This is required if <code>t</code> is not provided.)</p>
</td></tr>
<tr><td><code id="julianDay_+3A_sec">sec</code></td>
<td>
<p>numerical value for the second of the minute.
(This is required if <code>t</code> is not provided.)</p>
</td></tr>
<tr><td><code id="julianDay_+3A_tz">tz</code></td>
<td>
<p>timezone</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Julian-Day number, in astronomical convention as explained in
Meeus.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ul>
<li><p> Meeus, Jean. Astronomical Formulas for Calculators. Second Edition.
Richmond, Virginia, USA: Willmann-Bell, 1982.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other things related to astronomy: 
<code><a href="#topic+angle2hms">angle2hms</a>()</code>,
<code><a href="#topic+eclipticalToEquatorial">eclipticalToEquatorial</a>()</code>,
<code><a href="#topic+equatorialToLocalHorizontal">equatorialToLocalHorizontal</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+moonAngle">moonAngle</a>()</code>,
<code><a href="#topic+siderealTime">siderealTime</a>()</code>,
<code><a href="#topic+sunAngle">sunAngle</a>()</code>,
<code><a href="#topic+sunDeclinationRightAscension">sunDeclinationRightAscension</a>()</code>
</p>
<p>Other things related to time: 
<code><a href="#topic+ctimeToSeconds">ctimeToSeconds</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+numberAsHMS">numberAsHMS</a>()</code>,
<code><a href="#topic+numberAsPOSIXct">numberAsPOSIXct</a>()</code>,
<code><a href="#topic+secondsToCtime">secondsToCtime</a>()</code>,
<code><a href="#topic+unabbreviateYear">unabbreviateYear</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# example from Meeus
t &lt;- ISOdatetime(1977, 4, 26, hour = 0, min = 0, sec = 0, tz = "UTC") + 0.4 * 86400
stopifnot(all.equal(julianDay(t), 2443259.9))

</code></pre>

<hr>
<h2 id='labelWithUnit'>Create Label With Unit</h2><span id='topic+labelWithUnit'></span>

<h3>Description</h3>

<p><code>labelWithUnit</code> creates a label with a unit, for graphical
display, e.g. by <code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>.
The unit is enclosed in square brackets, although setting
<code>options(oceUnitBracket="(")</code> will cause parentheses to be
used, instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>labelWithUnit(name, unit = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="labelWithUnit_+3A_name">name</code></td>
<td>
<p>character value naming a quantity.</p>
</td></tr>
<tr><td><code id="labelWithUnit_+3A_unit">unit</code></td>
<td>
<p>a list containing items <code>unit</code> and (optionally) <code>scale</code>, only the
first of which, an <code><a href="base.html#topic+expression">expression()</a></code>, is used.  If <code>unit</code> is not provided,
then a default will be used (see &ldquo;Details&rdquo;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>name</code> is in a standard list, then alterations are made as appropriate,
e.g. <code>"SA"</code> or <code>"Absolute Salinity"</code> yields an S with subscript A; <code>"CT"</code> or
<code>"Conservative Temperature"</code> yields an upper-case Theta, <code>sigmaTheta</code>
yields a sigma with subscript theta, <code>sigma0</code> yields
sigma with subscript 0 (with similar for 1 through 4), <code>"N2"</code> yields &quot;N&quot; with
superscript 2, and <code>"pressure"</code> yields &quot;p&quot;.
These basic hydrographic quantities have default units that will
be used if <code>unit</code> is not supplied (see &ldquo;Examples&rdquo;).
</p>
<p>In addition to the above, several chemical names are recognized,
but no unit is guessed for them, because the oceanographic
community lacks agreed-upon standards.
</p>
<p>If <code>name</code> is not recognized, then it is simply repeated in the
return value.
</p>


<h3>Value</h3>

<p><code>labelWithUnit</code> returns a language object, created with <code><a href="base.html#topic+bquote">bquote()</a></code>,
that that may supplied as a text string to <code><a href="graphics.html#topic+legend">legend()</a></code>, <code><a href="graphics.html#topic+mtext">mtext()</a></code>, <code><a href="graphics.html#topic+text">text()</a></code>,
etc.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that create labels: 
<code><a href="#topic+resizableLabel">resizableLabel</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# 1. temperature has a predefined unit, but this can be overruled
labelWithUnit("temperature")
labelWithUnit(
    "temperature",
    list(unit = expression(m / s), scale = "erroneous")
)
# 2. phosphate lacks a predefined unit
labelWithUnit("phosphate")
data(section)
labelWithUnit(
    "phosphate",
    section[["station", 1]][["phosphateUnit"]]
)

</code></pre>

<hr>
<h2 id='ladp-class'>Class to Store Lowered-adp Data</h2><span id='topic+ladp-class'></span>

<h3>Description</h3>

<p>This class stores data measured with a lowered ADP (also
known as ADCP) device.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>ladp</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>ladp</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>ladp</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+ladp-class">ladp</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+ladp-class">ladp</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ladp data: 
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic+as.ladp">as.ladp</a>()</code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+summary+2Cladp-method">summary,ladp-method</a></code>
</p>

<hr>
<h2 id='landsat'>Sample landsat Data</h2><span id='topic+landsat'></span>

<h3>Description</h3>

<p>This is a subset of the Landsat-8 image designated LC80080292014065LGN00, an
image from March 2014 that covers Nova Scotia and portions of the Bay of
Fundy and the Scotian Shelf. The image is decimated to reduce the memory
requirements of this package, yielding a spatial resolution of about 2km.
</p>


<h3>Details</h3>

<p>The original data were downloaded from the USGS earthexplorer website, although
other sites can also be used to uncover it by name.  The original
data were decimated by a factor of 100 in longitude and latitude,
to reduce the file size from 1G to 100K.
</p>


<h3>See Also</h3>

<p>Other satellite datasets provided with oce: 
<code><a href="#topic+amsr">amsr</a></code>
</p>
<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to landsat data: 
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic+landsat-class">landsat-class</a></code>,
<code><a href="#topic+landsatAdd">landsatAdd</a>()</code>,
<code><a href="#topic+landsatTrim">landsatTrim</a>()</code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+read.landsat">read.landsat</a>()</code>,
<code><a href="#topic+summary+2Clandsat-method">summary,landsat-method</a></code>
</p>

<hr>
<h2 id='landsat-class'>Class to Store Landsat Satellite Data</h2><span id='topic+landsat-class'></span>

<h3>Description</h3>

<p>This class holds landsat data. Such are available at several
websites (e.g. reference 1).
Although the various functions may work for other satellites, the
discussion here focusses on Landsat 8 and Landsat 7.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>landsat</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>landsat</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>landsat</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+landsat-class">landsat</a> objects (see <code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+landsat-class">landsat</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Data storage</h3>

<p>The data are stored with 16-bit resolution.  Oce
breaks these 16 bits up into most-significant and least-significant bytes.
For example, the aerosol band of a Landsat object named <code>x</code> are
contained within <code>x@data$aerosol$msb</code> and <code>x@data$aerosol$lsb</code>,
each of which is a matrix of raw values.  The results may be combined as e.g.
</p>
<pre>
256L*as.integer(x@data[[i]]$msb) + as.integer(x@data[[i]]$lsb)
</pre>
<p>and this is what is returned by executing <code>x[["aerosol"]]</code>.
</p>
<p>Landsat data files typically occupy approximately a
gigabyte of storage.  That means that corresponding Oce objects are about
the same size, and this can pose significant problems on computers with
less than 8GB of memory.  It is sensible to specify bands of interest when
reading data with <code><a href="#topic+read.landsat">read.landsat()</a></code>, and also to use
<code><a href="#topic+landsatTrim">landsatTrim()</a></code> to isolate geographical regions that need
processing.
</p>
<p>Experts may need to get direct access to the data, and this is easy because
all Landsat objects (regardless of satellite) use a similar storage form.
Band information is stored in byte form, to conserve space.  Two bytes are
used for each pixel in Landsat-8 objects, with just one for other objects.
For example, if a Landsat-8 object named <code>L</code> contains the <code>tirs1</code>
band, the most- and least-significant bytes will be stored in matrices
<code>L@data$tirs1$msb</code> and <code>L@data$tirs1$lsb</code>.  A similar Landsat-7
object would have the same items, but <code>msb</code> would be just the value
<code>0x00</code>.
</p>
<p>Derived bands, which may be added to a landsat object with
<code><a href="#topic+landsatAdd">landsatAdd()</a></code>, are not stored in byte matrices.  Instead they
are stored in numerical matrices, which means that they use 4X more storage
space for Landsat-8 images, and 8X more storage space for other satellites.
A computer needs at least 8GB of RAM to work with such data.
</p>


<h3>Landsat 8</h3>

<p>The Landsat 8 satellite has 11 frequency bands, listed below (see reference 2]).
</p>
<pre>
.------------------------------------------------------------------------------.
| Band | Band                      | Band         | Wavelength    | Resolution |
| No.  | Contents                  | Name         | (micrometers) |   (meters) |
|------+---------------------------+--------------+---------------+------------|
|    1 | Coastal aerosol           | aerosol      |  0.43 -  0.45 |         30 |
|    2 | Blue                      | blue         |  0.45 -  0.51 |         30 |
|    3 | Green                     | green        |  0.53 -  0.59 |         30 |
|    4 | Red                       | red          |  0.64 -  0.67 |         30 |
|    5 | Near Infrared (NIR)       | nir          |  0.85 -  0.88 |         30 |
|    6 | SWIR 1                    | swir1        |  1.57 -  1.65 |         30 |
|    7 | SWIR 2                    | swir2        |  2.11 -  2.29 |         30 |
|    8 | Panchromatic              | panchromatic |  0.50 -  0.68 |         15 |
|    9 | Cirrus                    | cirrus       |  1.36 -  1.38 |         30 |
|   10 | Thermal Infrared (TIRS) 1 | tirs1        | 10.60 - 11.19 |        100 |
|   11 | Thermal Infrared (TIRS) 2 | tirs2        | 11.50 - 12.51 |        100 |
.------------------------------------------------------------------------------.
</pre>
<p>In addition to the above, setting <code>band="terralook"</code> may be used as
an abbreviation for <code>band=c("red", "green", "nir")</code>.
</p>
<p>Band 8 is panchromatic, and has the highest resolution.  For convenience of
programming, <code><a href="#topic+read.landsat">read.landsat()</a></code> subsamples the <code>tirs1</code> and
<code>tirs2</code> bands to the 30m resolution of the other bands.  See Reference
3 for information about the evolution of Landsat 8 calibration
coefficients, which as of summer 2014 are still subject to change.
</p>


<h3>Landsat 7</h3>

<p>Band information is as follows (from reference 8).  The names are not official, but
are set up to roughly correspond with Landsat-8 names, according to wavelength.
An exception is the Landsat-7 bands named <code>tirs1</code> and <code>tirs2</code>, which
are at two different gain settings, with identical wavelength span for
each, which roughly matches the range of the Landsat-8 bands <code>tirs1</code>
and <code>tirs2</code> combined.  This may seem confusing, but it lets code like
<code>plot(im, band="tirs1")</code> to work with both Landsat-8 and Landsat-7.
</p>
<pre>
.------------------------------------------------------------------------------.
| Band | Band                      | Band         | Wavelength    | Resolution |
| No.  | Contents                  | Name         | (micrometers) |   (meters) |
|------+---------------------------+--------------+---------------+------------|
|    1 | Blue                      | blue         |  0.45 -  0.52 |         30 |
|    2 | Green                     | green        |  0.52 -  0.60 |         30 |
|    3 | Red                       | red          |  0.63 -  0.69 |         30 |
|    4 | Near IR                   | nir          |  0.77 -  0.90 |         30 |
|    5 | SWIR                      | swir1        |  1.55 -  1.75 |         30 |
|    6 | Thermal IR                | tirs1        | 10.4  - 12.50 |         30 |
|    7 | Thermal IR                | tirs2        | 10.4  - 12.50 |         30 |
|    8 | SWIR                      | swir2        |  2.09 -  2.35 |         30 |
|    9 | Panchromatic              | panchromatic |  0.52 -  0.90 |         15 |
.------------------------------------------------------------------------------.
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>


<ol>
<li><p> See the USGS &quot;glovis&quot; web site.
</p>
</li>
<li><p> see landsat.gsfc.nasa.gov/?page_id=5377
</p>
</li>
<li><p> see landsat.usgs.gov/calibration_notices.php
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;https://dankelley.github.io/r/2014/07/01/landsat.html&#8288;</code>
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;https://scienceofdoom.com/2010/12/27/emissivity-of-the-ocean/&#8288;</code>
</p>
</li>
<li><p> see landsat.usgs.gov/Landsat8_Using_Product.php
</p>
</li>
<li><p> see landsathandbook.gsfc.nasa.gov/pdfs/Landsat7_Handbook.pdf
</p>
</li>
<li><p> see landsat.usgs.gov/band_designations_landsat_satellites.php
</p>
</li>
<li><p> Yu, X. X. Guo and Z. Wu., 2014. Land Surface Temperature Retrieval from
Landsat 8 TIRS-Comparison between Radiative Transfer Equation-Based Method,
Split Window Algorithm and Single Channel Method, <em>Remote Sensing</em>, 6,
9829-9652.  <code style="white-space: pre;">&#8288;https://www.mdpi.com/2072-4292/6/10/9829&#8288;</code>
</p>
</li>
<li><p> Rajeshwari, A., and N. D. Mani, 2014.  Estimation of land surface
temperature of Dindigul district using Landsat 8 data. <em>International
Journal of Research in Engineering and Technology</em>, 3(5), 122-126.
<code style="white-space: pre;">&#8288;http://www.academia.edu/7655089/ESTIMATION_OF_LAND_SURFACE_TEMPERATURE_OF_DINDIGUL_DISTRICT_USING_LANDSAT_8_DATA&#8288;</code>
</p>
</li>
<li><p> Konda, M. Imasato N., Nishi, K., and T. Toda, 1994.  Measurement of the Sea
Surface Emissivity.  <em>Journal of Oceanography</em>, 50, 17:30.
<a href="https://doi.org/10.1007/BF02233853">doi:10.1007/BF02233853</a>
</p>
</li></ol>



<h3>See Also</h3>

<p>Data from AMSR satellites are handled with <a href="#topic+amsr-class">amsr</a>.
</p>
<p>A file containing Landsat data may be read with <code><a href="#topic+read.landsat">read.landsat()</a></code> or
<code><a href="#topic+read.oce">read.oce()</a></code>, and one such file is provided by the <a href="https://CRAN.R-project.org/package=ocedata"><span class="pkg">ocedata</span></a>
package as a dataset named <code>landsat</code>.
</p>
<p>Plots may be made with <code><a href="#topic+plot+2Clandsat-method">plot,landsat-method()</a></code>.  Since plotting can be quite
slow, decimation is available both in the plotting function and as the separate
function <code><a href="#topic+decimate">decimate()</a></code>.  Images may be subsetted with
<code><a href="#topic+landsatTrim">landsatTrim()</a></code>.
</p>
<p>Other classes holding satellite data: 
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+g1sst-class">g1sst-class</a></code>,
<code><a href="#topic+satellite-class">satellite-class</a></code>
</p>
<p>Other things related to landsat data: 
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic+landsatAdd">landsatAdd</a>()</code>,
<code><a href="#topic+landsatTrim">landsatTrim</a>()</code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+read.landsat">read.landsat</a>()</code>,
<code><a href="#topic+summary+2Clandsat-method">summary,landsat-method</a></code>
</p>

<hr>
<h2 id='landsatAdd'>Add a Band to a landsat Object</h2><span id='topic+landsatAdd'></span>

<h3>Description</h3>

<p>Add a band to a <a href="#topic+landsat-class">landsat</a> object. Note that
it will be stored in numeric form, not raw form, and therefore
it will require much more storage than data read with
<code><a href="#topic+read.landsat">read.landsat()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>landsatAdd(x, data, name, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="landsatAdd_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+landsat-class">landsat</a> object.</p>
</td></tr>
<tr><td><code id="landsatAdd_+3A_data">data</code></td>
<td>
<p>A matrix of data, with dimensions matching that of entries already in <code>x</code>.</p>
</td></tr>
<tr><td><code id="landsatAdd_+3A_name">name</code></td>
<td>
<p>The name to be used for the data, i.e. the data can later be
accessed with <code>d[[name]]</code> where <code>d</code> is the name of the return value
from the present function.</p>
</td></tr>
<tr><td><code id="landsatAdd_+3A_debug">debug</code></td>
<td>
<p>A flag that turns on debugging.  Set to 1 to get a moderate amount of debugging
information, or a higher value for more debugging.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+landsat-class">landsat</a> object, with a new data band.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for the <a href="#topic+landsat-class">landsat</a> class explains the
structure of landsat objects, and also outlines the other functions dealing
with them.
</p>
<p>Other things related to landsat data: 
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic+landsat-class">landsat-class</a></code>,
<code><a href="#topic+landsatTrim">landsatTrim</a>()</code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+read.landsat">read.landsat</a>()</code>,
<code><a href="#topic+summary+2Clandsat-method">summary,landsat-method</a></code>
</p>

<hr>
<h2 id='landsatTrim'>Trim a landsat Image to a Geographical Region</h2><span id='topic+landsatTrim'></span>

<h3>Description</h3>

<p>Trim a landsat image to a latitude-longitude box. This is only an approximate
operation, because landsat images are provided in x-y coordinates, not
longitude-latitude coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>landsatTrim(x, ll, ur, box, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="landsatTrim_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+landsat-class">landsat</a> object.</p>
</td></tr>
<tr><td><code id="landsatTrim_+3A_ll">ll</code></td>
<td>
<p>A list containing <code>longitude</code> and <code>latitude</code>, for the
lower-left corner of the portion of the image to retain, or a vector
with first element longitude and second element latitude. If provided,
then <code>ur</code> must also be provided, but <code>box</code> cannot.</p>
</td></tr>
<tr><td><code id="landsatTrim_+3A_ur">ur</code></td>
<td>
<p>A list containing <code>longitude</code> and <code>latitude</code>, for the
upper-right corner of the portion of the image to retain, or a vector
with first element longitude and second element latitude. If provided,
then <code>ll</code> must also be provided, but <code>box</code> cannot.</p>
</td></tr>
<tr><td><code id="landsatTrim_+3A_box">box</code></td>
<td>
<p>A list containing <code>x</code> and <code>y</code> (each of length 2),
corresponding to the values for <code>ll</code> and <code>ur</code>, such as would
be produced by a call to <code>locator(2)</code>. If provided, neither
<code>ll</code> nor <code>ur</code> may be provided.</p>
</td></tr>
<tr><td><code id="landsatTrim_+3A_debug">debug</code></td>
<td>
<p>A flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or a higher value for more debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As of June 25, 2015, the matrices storing the image data are trimmed to indices
determined by linear interpolation based on the location of the <code>ll</code> and
<code>ur</code> corners within the lon-lat corners specified in the image data. (A
previous version trimmed in UTM space, and in fact this may be done in future
also, if a problem in lonlat/utm conversion is resolved.) An error results if
there is no intersection between the trimming box and the image box.
</p>


<h3>Value</h3>

<p>A <a href="#topic+landsat-class">landsat</a> object, with data having
been trimmed as specified.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>See Also</h3>

<p>The documentation for the <a href="#topic+landsat-class">landsat</a> class explains the
structure of landsat objects, and also outlines the other functions dealing
with them.
</p>
<p>Other things related to landsat data: 
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic+landsat-class">landsat-class</a></code>,
<code><a href="#topic+landsatAdd">landsatAdd</a>()</code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+read.landsat">read.landsat</a>()</code>,
<code><a href="#topic+summary+2Clandsat-method">summary,landsat-method</a></code>
</p>

<hr>
<h2 id='latFormat'>Format a Latitude</h2><span id='topic+latFormat'></span>

<h3>Description</h3>

<p>Format a latitude, using &quot;S&quot; for negative latitude.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>latFormat(lat, digits = max(6, getOption("digits") - 1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="latFormat_+3A_lat">lat</code></td>
<td>
<p>latitude in <code class="reqn">^\circ</code>N north of the equator.</p>
</td></tr>
<tr><td><code id="latFormat_+3A_digits">digits</code></td>
<td>
<p>the number of significant digits to use when printing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lonFormat">lonFormat()</a></code> and <code><a href="#topic+latlonFormat">latlonFormat()</a></code>.
</p>

<hr>
<h2 id='latlonFormat'>Format a Latitude-Longitude Pair</h2><span id='topic+latlonFormat'></span>

<h3>Description</h3>

<p>Format a latitude-longitude pair, using &quot;S&quot; for negative latitudes, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>latlonFormat(lat, lon, digits = max(6, getOption("digits") - 1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="latlonFormat_+3A_lat">lat</code></td>
<td>
<p>latitude in <code class="reqn">^\circ</code>N north of the equator.</p>
</td></tr>
<tr><td><code id="latlonFormat_+3A_lon">lon</code></td>
<td>
<p>longitude in <code class="reqn">^\circ</code>N east of Greenwich.</p>
</td></tr>
<tr><td><code id="latlonFormat_+3A_digits">digits</code></td>
<td>
<p>the number of significant digits to use when printing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+latFormat">latFormat()</a></code> and <code><a href="#topic+lonFormat">lonFormat()</a></code>.
</p>

<hr>
<h2 id='lisst'>Sample lisst Data</h2><span id='topic+lisst'></span>

<h3>Description</h3>

<p>LISST (Laser in-situ scattering and transmissometry) dataset, constructed
artificially.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lisst)
</code></pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Source</h3>

<p>This was constructed artificially using <code><a href="#topic+as.lisst">as.lisst()</a></code>,
to approximately match values that might be measured in the field.
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>

<hr>
<h2 id='lisst-class'>Class to Store LISST Data</h2><span id='topic+lisst-class'></span>

<h3>Description</h3>

<p>This class stores LISST (Laser in-situ scattering and transmissometry) data.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>lisst</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>lisst</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>lisst</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+lisst-class">lisst</a> objects (see <code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+lisst-class">lisst</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Information about LISST instruments is provided at
the manufacturer's website, <code style="white-space: pre;">&#8288;https://www.sequoiasci.com&#8288;</code>.
</p>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to lisst data: 
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic+as.lisst">as.lisst</a>()</code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+read.lisst">read.lisst</a>()</code>,
<code><a href="#topic+summary+2Clisst-method">summary,lisst-method</a></code>
</p>

<hr>
<h2 id='lobo'>Sample lobo Data</h2><span id='topic+lobo'></span>

<h3>Description</h3>

<p>This is sample lobo dataset obtained in the Northwest Arm of Halifax by
Satlantic.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Source</h3>

<p>The data were downloaded from a web interface at Satlantic LOBO web
server and then read with <code><a href="#topic+read.lobo">read.lobo()</a></code>.
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to lobo data: 
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic+as.lobo">as.lobo</a>()</code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+read.lobo">read.lobo</a>()</code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+summary+2Clobo-method">summary,lobo-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(lobo)
summary(lobo)
plot(lobo)

</code></pre>

<hr>
<h2 id='lobo-class'>Class to Store LOBO Data</h2><span id='topic+lobo-class'></span>

<h3>Description</h3>

<p>This class stores LOBO data.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>lobo</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>lobo</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>lobo</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+lobo-class">lobo</a> objects (see <code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+lobo-class">lobo</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to lobo data: 
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic+as.lobo">as.lobo</a>()</code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+read.lobo">read.lobo</a>()</code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+summary+2Clobo-method">summary,lobo-method</a></code>
</p>

<hr>
<h2 id='locationForGsw'>Alter Longitude and Latitude for gsw Computations</h2><span id='topic+locationForGsw'></span>

<h3>Description</h3>

<p>This function is mainly intended for use within <code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity()</a></code> and
similar functions that use the <code>gsw</code> package to compute seawater
properties in the Gibbs Seawater formulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locationForGsw(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="locationForGsw_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>gsw</code> functions
require location information to be matched up with hydrographic
information.  The scheme depends on the dimensionality of the
hydrographic variables and the location variables. For example,
the <a href="#topic+ctd-class">ctd</a> stores <code>salinity</code> etc in vectors, an stores
just one longitude-latitude pair for each vector.  By contrast,
the <a href="#topic+argo-class">argo</a> stores <code>salinity</code> etc as matrices, and stores
e.g. <code>longitude</code> as a vector of length matching the first
dimension of <code>salinity</code>.
</p>
<p><a href="#topic+locationForGsw">locationForGsw</a> repeats location information as required,
returning values with dimensionality matching <code>pressure</code>.
</p>


<h3>Value</h3>

<p>A list containing <code>longitude</code> and <code>latitude</code>, with
dimensionality matching <code>pressure</code> in the <code>data</code> slot
of <code>x</code>.  If <code>x</code> lacks location information (in either
its <code>metadata</code> or <code>data</code> slot) or lacks <code>pressure</code> in its
data slot, then the returned list will hold NULL values for
both <code>longitude</code> and <code>latitude</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>

<hr>
<h2 id='lon360'>Change Longitude From -180:180 to 0:360 Convention</h2><span id='topic+lon360'></span>

<h3>Description</h3>

<p>For numerical input, including vectors, matrices and arrays,
<code><a href="#topic+lon360">lon360()</a></code> simply calls <code><a href="base.html#topic+ifelse">ifelse()</a></code> to add 360 to any negative values. For
<a href="#topic+section-class">section</a> objects, it changes <code>longitude</code> in the <code>metadata</code> slot
and then calls itself to handle the <a href="#topic+ctd-class">ctd</a> objects stored as
as the entries in <code>station</code> within the <code>data</code> slot.
For this <a href="#topic+ctd-class">ctd</a> object, and indeed for all non-<a href="#topic+section-class">section</a>
objects, <code><a href="#topic+lon360">lon360()</a></code> changes <code>longitude</code> values in the
<code>metadata</code> slot (if present) and also in the <code>data</code> slot (again, if
present). This function is not useful for dealing with coastline
data; see <code><a href="#topic+coastlineCut">coastlineCut()</a></code> for such data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lon360(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lon360_+3A_x">x</code></td>
<td>
<p>either a numeric vector or array, or an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>lon360(c(179, -179))
</code></pre>

<hr>
<h2 id='lonFormat'>Format a Longitude</h2><span id='topic+lonFormat'></span>

<h3>Description</h3>

<p>Format a longitude, using &quot;W&quot; for west longitude.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lonFormat(lon, digits = max(6, getOption("digits") - 1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lonFormat_+3A_lon">lon</code></td>
<td>
<p>longitude in <code class="reqn">^\circ</code>N east of Greenwich.</p>
</td></tr>
<tr><td><code id="lonFormat_+3A_digits">digits</code></td>
<td>
<p>the number of significant digits to use when printing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+latFormat">latFormat()</a></code> and <code><a href="#topic+latlonFormat">latlonFormat()</a></code>.
</p>

<hr>
<h2 id='longitudeTighten'>Try to Reduce Section Longitude Range</h2><span id='topic+longitudeTighten'></span>

<h3>Description</h3>

<p><a href="#topic+longitudeTighten">longitudeTighten</a> shifts some longitudes in its first argument by 360 degrees, if doing
so will reduce the overall longitude span.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>longitudeTighten(section)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="longitudeTighten_+3A_section">section</code></td>
<td>
<p>a <a href="#topic+section-class">section</a> object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be helpful in cases where the CTD stations within a section
cross the cut point of the longitude convention, which otherwise might
yield ugly plots if <code><a href="#topic+plot+2Csection-method">plot,section-method()</a></code> is used with <code>xtype="longitude"</code>.
This problem does occur with CTD objects ordered by time of sampling,
but was observed in December 2020 for a GO-SHIPS dataset downloaded from
<code style="white-space: pre;">&#8288;https://cchdo.ucsd.edu/data/15757/a10_1992_ct1&#8288;</code>.
</p>


<h3>Value</h3>

<p>A <a href="#topic+section-class">section</a> object based on its first argument, but with
longitudes shifted in its <code>metadata</code> slot, and also in the <code>metadata</code> slots
of each of the <a href="#topic+ctd-class">ctd</a> objects that are stored in the <code>station</code> item
in its <code>data</code> slot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>

<hr>
<h2 id='lonlat2map'>Convert Longitude and Latitude to X and Y</h2><span id='topic+lonlat2map'></span>

<h3>Description</h3>

<p>If a projection is already being used (e.g. as set by <code><a href="#topic+mapPlot">mapPlot()</a></code>)
then only <code>longitude</code> and <code>latitude</code> should be given, and the
other arguments will be inferred by <code>lonlat2map</code>.  This is important
because otherwise, if a new projection is called for, it will ruin any
additions to the existing plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lonlat2map(longitude, latitude, projection = "", debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lonlat2map_+3A_longitude">longitude</code></td>
<td>
<p>a numeric vector containing decimal longitudes, or a list
containing items named <code>longitude</code> and <code>latitude</code>, in which case
the indicated values are used, and next argument is ignored.</p>
</td></tr>
<tr><td><code id="lonlat2map_+3A_latitude">latitude</code></td>
<td>
<p>a numeric vector containing decimal latitude (ignored if
<code>longitude</code> is a list, as described above).</p>
</td></tr>
<tr><td><code id="lonlat2map_+3A_projection">projection</code></td>
<td>
<p>optional indication of projection.  This must be character
string in the format used by the <a href="https://CRAN.R-project.org/package=sf"><span class="pkg">sf</span></a> package;
see <code><a href="#topic+mapPlot">mapPlot()</a></code>.)</p>
</td></tr>
<tr><td><code id="lonlat2map_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing <code>x</code> and <code>y</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p><code>mapLongitudeLatitudeXY</code> is a safer alternative, if a map has
already been drawn with <code><a href="#topic+mapPlot">mapPlot()</a></code>, because that function cannot
alter an existing projection. <code><a href="#topic+map2lonlat">map2lonlat()</a></code> is an inverse to
<code>map2lonlat</code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Cape Split, in the Minas Basin of the Bay of Fundy
cs &lt;- list(longitude = -64.49657, latitude = 45.33462)
xy &lt;- lonlat2map(cs, projection = "+proj=merc")
map2lonlat(xy)

</code></pre>

<hr>
<h2 id='lonlat2utm'>Convert Longitude and Latitude to UTM</h2><span id='topic+lonlat2utm'></span>

<h3>Description</h3>

<p>Convert Longitude and Latitude to UTM
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lonlat2utm(longitude, latitude, zone, km = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lonlat2utm_+3A_longitude">longitude</code></td>
<td>
<p>numeric vector of decimal longitude.  May also be a list containing items
named <code>longitude</code> and <code>latitude</code>, in which case the indicated
values are used, and next argument is ignored.</p>
</td></tr>
<tr><td><code id="lonlat2utm_+3A_latitude">latitude</code></td>
<td>
<p>numeric vector of decimal latitude (ignored if <code>longitude</code> is a list
containing both coordinates)</p>
</td></tr>
<tr><td><code id="lonlat2utm_+3A_zone">zone</code></td>
<td>
<p>optional indication of UTM zone.  Normally this is inferred from
the longitude, but specifying it can be helpful in dealing with Landsat
images, which may cross zones and which therefore are described by a single
zone.</p>
</td></tr>
<tr><td><code id="lonlat2utm_+3A_km">km</code></td>
<td>
<p>logical value indicating whether <code>easting</code> and
<code>northing</code> are in kilometers or meters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing <code>easting</code>, <code>northing</code>, <code>zone</code> and
<code>hemisphere</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p><code style="white-space: pre;">&#8288;https://en.wikipedia.org/wiki/Universal_Transverse_Mercator_coordinate_system&#8288;</code>,
downloaded May 31, 2014.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+utm2lonlat">utm2lonlat()</a></code> does the inverse operation.  For general
projections and their inverses, use <code><a href="#topic+lonlat2map">lonlat2map()</a></code> and
<code><a href="#topic+map2lonlat">map2lonlat()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Cape Split, in the Minas Basin of the Bay of Fundy
lonlat2utm(-64.496567, 45.334626)

</code></pre>

<hr>
<h2 id='lookWithin'>Look Within the First Element of a List for Replacement Values</h2><span id='topic+lookWithin'></span>

<h3>Description</h3>

<p>This is a helper function used by some seawater functions
(with names starting with <code>sw</code>) to
facilitate the specification of water properties either with
distinct arguments, or with data stored within an <code>oce</code>
object that is the first argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lookWithin(list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lookWithin_+3A_list">list</code></td>
<td>
<p>A list of elements, typically arguments that will be used in sw functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>list[1]</code> is not an <code>oce</code> object, then the
return value of <code>lookWithin</code> is the same as the input
value, except that (a) <code>eos</code> is completed to either
<code>"gsw"</code> or <code>"unesco"</code> and (b) if <code>longitude</code>
and <code>latitude</code> are within <code>list[1]</code>, then they
are possibly lengthened, to have the same length as the first
item in the <code>data</code> slot of <code>list[1]</code>.
</p>
<p>The examples may clarify this somewhat.
</p>


<h3>Value</h3>

<p>A list with elements of the same names but possibly filled in from the first element.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. If first item is not a CTD object, just return the input
lookWithin(list(a = 1, b = 2)) # returns a list
# 2. Extract salinity from a CTD object
data(ctd)
str(lookWithin(list(salinity = ctd)))
# 3. Extract salinity and temperature. Note that the
# value specified for temperature is ignored; all that matters
# is that temperature is named.
str(lookWithin(list(salinity = ctd, temperature = NULL)))
# 4. How it is used by swRho()
rho1 &lt;- swRho(ctd, eos = "unesco")
rho2 &lt;- swRho(ctd[["salinity"]], ctd[["temperature"]], ctd[["pressure"]], eos = "unesco")
stopifnot(all.equal(rho1, rho2))
</code></pre>

<hr>
<h2 id='lowpass'>Lowpass Digital Filtering</h2><span id='topic+lowpass'></span>

<h3>Description</h3>

<p>The filter coefficients are constructed using standard definitions,
and then <code><a href="stats.html#topic+filter">stats::filter()</a></code> is
used to filter the data. This leaves <code>NA</code>
values within half the filter length of the ends of the time series, but
these may be replaced with the original <code>x</code> values, if the argument
<code>replace</code> is set to <code>TRUE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lowpass(x, filter = "hamming", n, replace = TRUE, coefficients = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lowpass_+3A_x">x</code></td>
<td>
<p>a vector to be smoothed</p>
</td></tr>
<tr><td><code id="lowpass_+3A_filter">filter</code></td>
<td>
<p>name of filter; at present, <code>"hamming"</code>, <code>"hanning"</code>, and <code>"boxcar"</code> are permitted.</p>
</td></tr>
<tr><td><code id="lowpass_+3A_n">n</code></td>
<td>
<p>length of filter (must be an odd integer exceeding 1)</p>
</td></tr>
<tr><td><code id="lowpass_+3A_replace">replace</code></td>
<td>
<p>a logical value indicating whether points near the
ends of <code>x</code> should be copied into the end regions, replacing
the <code>NA</code> values that would otherwise be placed there by
<code><a href="stats.html#topic+filter">stats::filter()</a></code>.</p>
</td></tr>
<tr><td><code id="lowpass_+3A_coefficients">coefficients</code></td>
<td>
<p>logical value indicating whether to return
the filter coefficients, instead of the filtered values. In accordance
with conventions in the literature, the returned values are not
normalized to sum to 1, although of course that normalization
is done in the actual filtering.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>By default, <code>lowpass</code> returns a filtered version
of <code>x</code>, but if <code>coefficients</code> is <code>TRUE</code> then it
returns the filter coefficients.
</p>


<h3>Caution</h3>

<p>This function was added in June of 2017,
and it may be extended during the rest of 2017. New arguments
may appear between <code>n</code> and <code>replace</code>, so users are
advised to call this function with named arguments, not positional
arguments.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
par(mfrow = c(1, 2), mar = c(4, 4, 1, 1))
coef &lt;- lowpass(n = 5, coefficients = TRUE)
plot(-2:2, coef, ylim = c(0, 1), xlab = "Lag", ylab = "Coefficient")
x &lt;- seq(-5, 5) + rnorm(11)
plot(1:11, x, type = "o", xlab = "time", ylab = "x and X")
X &lt;- lowpass(x, n = 5)
lines(1:11, X, col = 2)
points(1:11, X, col = 2)
</code></pre>

<hr>
<h2 id='magneticField'>Earth Magnetic Declination, Inclination, and Intensity</h2><span id='topic+magneticField'></span>

<h3>Description</h3>

<p>Implements the 12th and 13th generations of the
International Geomagnetic Reference Field
(IGRF), based on a reworked version of a Fortran program downloaded from a
NOAA website (see &ldquo;References&rdquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>magneticField(longitude, latitude, time, version = 13)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="magneticField_+3A_longitude">longitude</code></td>
<td>
<p>longitude in degrees east (negative for degrees west), as a
number, a vector, or a matrix.</p>
</td></tr>
<tr><td><code id="magneticField_+3A_latitude">latitude</code></td>
<td>
<p>latitude in degrees north, as a number, vector, or matrix.
The shape (length or dimensions) must conform to the dimensions of <code>longitude</code>.</p>
</td></tr>
<tr><td><code id="magneticField_+3A_time">time</code></td>
<td>
<p>The time at which the field is desired. This may be a
single value or a vector or matrix that is structured to match
<code>longitude</code> and <code>latitude</code>. The value may a decimal year,
a POSIXt time, or a Date time.</p>
</td></tr>
<tr><td><code id="magneticField_+3A_version">version</code></td>
<td>
<p>an integer that must be either 12 or 13, to specify
the version number of the formulae. Note that 13 became the default
on 2020 March 3, so to old code will need to specify <code>version=12</code>
to work as it did before that date.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The code (subroutines <code>igrf12syn</code> and <code>igrf13syn</code>) seem to have
been written by Susan Macmillan of the British Geological Survey.  Comments
in the source code <code>igrf13syn</code> (the current default used here)
indicate that its coefficients were agreed to in
December 2019 by the IAGA Working Group V-MOD.  Other comments in that code
suggest that the proposed application time interval is from years 1900 to 2025, inclusive,
but that only dates from 1945 to 2015 are to be considered definitive.
</p>


<h3>Value</h3>

<p>A list containing <code>declination</code>, <code>inclination</code>, and
<code>intensity</code>.
</p>


<h3>Historical Notes</h3>

<p>For about a decade, <code>magneticField</code> used the version 12 formulae provided
by IAGA, but the code was updated on March 3, 2020, to version 13.  Example
3 shows that the differences in declination are typically under 2 degrees
(with 95 percent of the data lying between -1.7 and 0.7 degrees).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley wrote the R code and a fortran wrapper to the
<code>igrf12.f</code> subroutine, which was written by Susan Macmillan of the
British Geological Survey and distributed &ldquo;without limitation&rdquo; (email from
SM to DK dated June 5, 2015).  This version was updated subsequent
to that date; see &lsquo;Historical Notes&rsquo;.
</p>


<h3>References</h3>


<ol>
<li><p> The underlying Fortran code for version 12 is from <code>igrf12.f</code>, downloaded the NOAA
website (<code style="white-space: pre;">&#8288;https://www.ngdc.noaa.gov/IAGA/vmod/igrf.html&#8288;</code>) on June 7,
</p>
</li>
<li><p> That for version 13 is <code>igrf13.f</code>, downloaded from the NOAA website
(<code style="white-space: pre;">&#8288;https://www.ngdc.noaa.gov/IAGA/vmod/igrf.html&#8288;</code> on March 3, 2020.
</p>
</li>
<li><p> Witze, Alexandra. &ldquo;Earth's Magnetic Field Is Acting up and Geologists Don't Know Why.&rdquo;
Nature 565 (January 9, 2019): 143.
<a href="https://doi.org/10.1038/d41586-019-00007-1">doi:10.1038/d41586-019-00007-1</a>
</p>
</li>
<li><p> Alken, P., E. Thébault, C. D. Beggan, H. Amit, J. Aubert, J. Baerenzung, T. N. Bondar, et al.
&quot;International Geomagnetic Reference Field: The Thirteenth Generation.&quot;
Earth, Planets and Space 73, no. 1 (December 2021): 49.
<a href="https://doi.org/10.1186/s40623-020-01288-x">doi:10.1186/s40623-020-01288-x</a>.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to magnetism: 
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Coce-method">applyMagneticDeclination,oce-method</a></code>,
<code><a href="#topic+applyMagneticDeclination">applyMagneticDeclination</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# 1. Today's value at Halifax NS
magneticField(-(63 + 36 / 60), 44 + 39 / 60, Sys.Date())

# 2. World map of declination in year 2000.

data(coastlineWorld)
par(mar = rep(0.5, 4)) # no axes on whole-world projection
mapPlot(coastlineWorld, projection = "+proj=robin", col = "lightgray")
# Construct matrix holding declination
lon &lt;- seq(-180, 180)
lat &lt;- seq(-90, 90)
dec2000 &lt;- function(lon, lat) {
    magneticField(lon, lat, 2000)$declination
}
dec &lt;- outer(lon, lat, dec2000) # hint: outer() is very handy!
# Contour, unlabelled for small increments, labeled for
# larger increments.
mapContour(lon, lat, dec,
    col = "blue", levels = seq(-180, -5, 5),
    lty = 3, drawlabels = FALSE
)
mapContour(lon, lat, dec, col = "blue", levels = seq(-180, -20, 20))
mapContour(lon, lat, dec,
    col = "red", levels = seq(5, 180, 5),
    lty = 3, drawlabels = FALSE
)
mapContour(lon, lat, dec, col = "red", levels = seq(20, 180, 20))
mapContour(lon, lat, dec, levels = 180, col = "black", lwd = 2, drawlabels = FALSE)
mapContour(lon, lat, dec, levels = 0, col = "black", lwd = 2)


# 3. Declination differences between versions 12 and 13

lon &lt;- seq(-180, 180)
lat &lt;- seq(-90, 90)
decDiff &lt;- function(lon, lat) {
    old &lt;- magneticField(lon, lat, 2020, version = 13)$declination
    new &lt;- magneticField(lon, lat, 2020, version = 12)$declination
    new - old
}
decDiff &lt;- outer(lon, lat, decDiff)
decDiff &lt;- ifelse(decDiff &gt; 180, decDiff - 360, decDiff)
# Overall (mean) shift -0.1deg
t.test(decDiff)
# View histogram, narrowed to small differences
par(mar = c(3.5, 3.5, 2, 2), mgp = c(2, 0.7, 0))
hist(decDiff,
    breaks = seq(-180, 180, 0.05), xlim = c(-2, 2),
    xlab = "Declination difference [deg] from version=12 to version=13",
    main = "Predictions for year 2020"
)
print(quantile(decDiff, c(0.025, 0.975)))
# Note that the large differences are at high latitudes
imagep(lon, lat, decDiff, zlim = c(-1, 1) * max(abs(decDiff)))
lines(coastlineWorld[["longitude"]], coastlineWorld[["latitude"]])

</code></pre>

<hr>
<h2 id='makeFilter'>Make a Digital Filter</h2><span id='topic+makeFilter'></span>

<h3>Description</h3>

<p>The filter is suitable for use by <code><a href="stats.html#topic+filter">filter()</a></code>,
<code><a href="stats.html#topic+convolve">convolve()</a></code> or (for the <code>asKernal=TRUE</code> case) with
<code><a href="stats.html#topic+kernapply">kernapply()</a></code>.  Note that <code><a href="stats.html#topic+convolve">convolve()</a></code> should be faster
than <code><a href="stats.html#topic+filter">filter()</a></code>, but it cannot be used if the time series has
missing values.  For the Blackman-Harris filter, the half-power frequency is
at <code>1/m</code> cycles per time unit, as shown in the &ldquo;Examples&rdquo;
section.  When using <code><a href="stats.html#topic+filter">filter()</a></code> or <code><a href="stats.html#topic+kernapply">kernapply()</a></code> with
these filters, use <code>circular=TRUE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeFilter(
  type = c("blackman-harris", "rectangular", "hamming", "hann"),
  m,
  asKernel = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makeFilter_+3A_type">type</code></td>
<td>
<p>a string indicating the type of filter to use.  (See Harris
(1978) for a comparison of these and similar filters.)
</p>

<ul>
<li> <p><code>"blackman-harris"</code> yields a modified raised-cosine filter designated
as &quot;4-Term (-92 dB) Blackman-Harris&quot; by Harris (1978; coefficients given in
the table on page 65).  This is also called &quot;minimum 4-sample Blackman
Harris&quot; by that author, in his Table 1, which lists figures of merit as
follows: highest side lobe level -92dB; side lobe fall off -6 db/octave;
coherent gain 0.36; equivalent noise bandwidth 2.00 bins; 3.0-dB bandwidth
1.90 bins; scallop loss 0.83 dB; worst case process loss 3.85 dB; 6.0-db
bandwidth 2.72 bins; overlap correlation 46 percent for 75\
for 50\
a spectral peak, so that a value of 2 indicates a cutoff frequency of
<code>1/m</code>, where <code>m</code> is as given below.
</p>
</li>
<li> <p><code>"rectangular"</code> for a flat filter.  (This is just for convenience.  Note that
<code><a href="stats.html#topic+kernel">kernel</a></code><code style="white-space: pre;">&#8288;("daniell",....)&#8288;</code> gives the same result, in kernel form.)
<code>"hamming"</code> for a Hamming filter (a raised-cosine that does not taper
to zero at the ends)
</p>
</li>
<li> <p><code>"hann"</code> (a raised cosine that tapers to zero at the ends).
</p>
</li></ul>
</td></tr>
<tr><td><code id="makeFilter_+3A_m">m</code></td>
<td>
<p>length of filter.  This should be an odd number, for any
non-rectangular filter.</p>
</td></tr>
<tr><td><code id="makeFilter_+3A_askernel">asKernel</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to get a smoothing kernel for
the return value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>asKernel</code> is <code>FALSE</code>, this returns a list of filter
coefficients, symmetric about the midpoint and summing to 1.  These may be
used with <code><a href="stats.html#topic+filter">filter()</a></code>, which should be provided with argument
<code>circular=TRUE</code> to avoid phase offsets.  If <code>asKernel</code> is
<code>TRUE</code>, the return value is a smoothing kernel, which can be applied to
a timeseries with <code><a href="stats.html#topic+kernapply">kernapply()</a></code>, whose bandwidth can be determined
with <code><a href="stats.html#topic+bandwidth.kernel">bandwidth.kernel()</a></code>, and which has both print and plot
methods.
</p>


<h3>Sample of Usage</h3>

<pre>
# need signal package for this example
r &lt;- rnorm(2048)
rh &lt;- stats::filter(r, H)
rh &lt;- rh[is.finite(rh)] # kludge to remove NA at start/end
sR &lt;- spectrum(r, plot=FALSE, spans=c(11, 5, 3))
sRH &lt;- spectrum(rh, plot=FALSE, spans=c(11, 5, 3))
par(mfrow=c(2, 1), mar=c(3, 3, 1, 1), mgp=c(2, 0.7, 0))
plot(sR$freq, sRH$spec/sR$spec, xlab="Frequency", ylab="Power Transfer",
     type="l", lwd=5, col="gray")
theory &lt;- freqz(H, n=seq(0,pi,length.out=100))
# Note we must square the modulus for the power spectrum
lines(theory$f/pi/2, Mod(theory$h)^2, lwd=1, col="red")
grid()
legend("topright", col=c("gray", "red"), lwd=c(5, 1), cex=2/3,
       legend=c("Practical", "Theory"), bg="white")
plot(log10(sR$freq), log10(sRH$spec/sR$spec),
     xlab="log10 Frequency", ylab="log10 Power Transfer",
     type="l", lwd=5, col="gray")
theory &lt;- freqz(H, n=seq(0,pi,length.out=100))
# Note we must square the modulus for the power spectrum
lines(log10(theory$f/pi/2), log10(Mod(theory$h)^2), lwd=1, col="red")
grid()
legend("topright", col=c("gray", "red"), lwd=c(5, 1), cex=2/3,
       legend=c("Practical", "Theory"), bg="white")
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>F. J. Harris, 1978.  On the use of windows for harmonic analysis
with the discrete Fourier Transform.  <em>Proceedings of the IEEE</em>, 66(1),
51-83 (<code style="white-space: pre;">&#8288;http://web.mit.edu/xiphmont/Public/windows.pdf&#8288;</code>.)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# 1. Demonstrate step-function response
y &lt;- c(rep(1, 10), rep(-1, 10))
x &lt;- seq_along(y)
plot(x, y, type = "o", ylim = c(-1.05, 1.05))
BH &lt;- makeFilter("blackman-harris", 11, asKernel = FALSE)
H &lt;- makeFilter("hamming", 11, asKernel = FALSE)
yBH &lt;- stats::filter(y, BH)
points(x, yBH, col = 2, type = "o")
yH &lt;- stats::filter(y, H)
points(yH, col = 3, type = "o")
legend("topright",
    col = 1:3, cex = 2 / 3, pch = 1,
    legend = c("input", "Blackman Harris", "Hamming")
)

# 2. Show theoretical and practical filter gain, where
#    the latter is based on random white noise, and
#    includes a particular value for the spans
#    argument of spectrum(), etc.

</code></pre>

<hr>
<h2 id='map2lonlat'>Convert X and Y to Longitude and Latitude</h2><span id='topic+map2lonlat'></span>

<h3>Description</h3>

<p>Convert from x-y coordinates to longitude and latitude. This is normally called
internally within oce; see &ldquo;Bugs&rdquo;.
A projection must already have been set up, by a call to <code><a href="#topic+mapPlot">mapPlot()</a></code>
or <code><a href="#topic+lonlat2map">lonlat2map()</a></code>. It should be noted that not all projections are
handled well; see &ldquo;Bugs&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map2lonlat(x, y, init = NULL, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="map2lonlat_+3A_x">x</code></td>
<td>
<p>vector containing the x component of points in the projected space, or
a list containing items named <code>x</code> and <code>y</code>, in which case the next
argument is ignored.</p>
</td></tr>
<tr><td><code id="map2lonlat_+3A_y">y</code></td>
<td>
<p>vector containing the y coordinate of points in the projected space
(ignored if <code>x</code> is a list, as described above).</p>
</td></tr>
<tr><td><code id="map2lonlat_+3A_init">init</code></td>
<td>
<p>vector containing the initial guesses for longitude and latitude,
presently ignored.</p>
</td></tr>
<tr><td><code id="map2lonlat_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing <code>longitude</code> and <code>latitude</code>, with <code>NA</code>
values indicating points that are off the globe as displayed.
</p>


<h3>Bugs</h3>

<p><code>oce</code> uses the <code><a href="sf.html#topic+sf_project">sf::sf_project()</a></code> function to handle projections.
Only those projections that
have inverses are permitted within <code>oce</code>, and of that subset, some are omitted
because the <code>oce</code> developers have experienced problems with them.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lonlat2map">lonlat2map()</a></code> does the inverse operation.
</p>
<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Cape Split, in the Minas Basin of the Bay of Fundy
cs &lt;- list(longitude = -64.49657, latitude = 45.33462)
xy &lt;- lonlat2map(cs, projection = "+proj=merc")
map2lonlat(xy)

</code></pre>

<hr>
<h2 id='mapArrows'>Add Arrows to a Map</h2><span id='topic+mapArrows'></span>

<h3>Description</h3>

<p>Plot arrows on an existing map, e.g. to indicate a place location.
This is not well-suited for drawing direction fields, e.g. of
velocities; for that, see <code><a href="#topic+mapDirectionField">mapDirectionField()</a></code>.
Adds arrows to an existing map, by analogy to <code><a href="graphics.html#topic+arrows">arrows()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapArrows(
  longitude0,
  latitude0,
  longitude1 = longitude0,
  latitude1 = latitude0,
  length = 0.25,
  angle = 30,
  code = 2,
  col = par("fg"),
  lty = par("lty"),
  lwd = par("lwd"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapArrows_+3A_longitude0">longitude0</code>, <code id="mapArrows_+3A_latitude0">latitude0</code></td>
<td>
<p>starting points for arrows.</p>
</td></tr>
<tr><td><code id="mapArrows_+3A_longitude1">longitude1</code>, <code id="mapArrows_+3A_latitude1">latitude1</code></td>
<td>
<p>ending points for arrows.</p>
</td></tr>
<tr><td><code id="mapArrows_+3A_length">length</code></td>
<td>
<p>length of the arrow heads, passed to <code><a href="graphics.html#topic+arrows">arrows()</a></code>.</p>
</td></tr>
<tr><td><code id="mapArrows_+3A_angle">angle</code></td>
<td>
<p>angle of the arrow heads, passed to <code><a href="graphics.html#topic+arrows">arrows()</a></code>.</p>
</td></tr>
<tr><td><code id="mapArrows_+3A_code">code</code></td>
<td>
<p>numerical code indicating the type of arrows, passed to <code><a href="graphics.html#topic+arrows">arrows()</a></code>.</p>
</td></tr>
<tr><td><code id="mapArrows_+3A_col">col</code></td>
<td>
<p>arrow color, passed to <code><a href="graphics.html#topic+arrows">arrows()</a></code>.</p>
</td></tr>
<tr><td><code id="mapArrows_+3A_lty">lty</code></td>
<td>
<p>arrow line type, passed to <code><a href="graphics.html#topic+arrows">arrows()</a></code>.</p>
</td></tr>
<tr><td><code id="mapArrows_+3A_lwd">lwd</code></td>
<td>
<p>arrow line width, passed to <code><a href="graphics.html#topic+arrows">arrows()</a></code>.</p>
</td></tr>
<tr><td><code id="mapArrows_+3A_...">...</code></td>
<td>
<p>optional arguments passed to <code><a href="graphics.html#topic+arrows">arrows()</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(coastlineWorld)
mapPlot(coastlineWorld,
    longitudelim = c(-120, -60), latitudelim = c(30, 60),
    col = "lightgray", projection = "+proj=lcc +lat_1=45 +lon_0=-100"
)
lon &lt;- seq(-120, -75, 15)
n &lt;- length(lon)
lat &lt;- 45 + rep(0, n)
# Draw meridional arrows in N America, from 45N to 60N.
mapArrows(lon, lat, lon, lat + 15, length = 0.05, col = "blue")


</code></pre>

<hr>
<h2 id='mapAxis'>Add Axis Labels to an Existing Map</h2><span id='topic+mapAxis'></span>

<h3>Description</h3>

<p>Plot axis labels on an existing map.
This is an advanced function, requiring
coordination with <code><a href="#topic+mapPlot">mapPlot()</a></code> and (possibly) also with <code><a href="#topic+mapGrid">mapGrid()</a></code>,
and so it is best avoided by novices, who may be satisfied
with the defaults used by <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapAxis(
  side = 1:2,
  longitude = TRUE,
  latitude = TRUE,
  axisStyle = 1,
  tick = TRUE,
  line = NA,
  pos = NA,
  outer = FALSE,
  font = NA,
  lty = "solid",
  lwd = 1,
  lwd.ticks = lwd,
  col = NULL,
  col.ticks = NULL,
  hadj = NA,
  padj = NA,
  tcl = -0.3,
  cex.axis = 1,
  mgp = c(0, 0.5, 0),
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapAxis_+3A_side">side</code></td>
<td>
<p>the side at which labels are to be drawn.  If not provided,
sides 1 and 2 will be used (i.e. bottom and left-hand sides).</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_longitude">longitude</code></td>
<td>
<p>either a logical value or a numeric vector of longitudes. There
are three possible cases:
(1) If <code>longitude=TRUE</code> (the default) then ticks and nearby numbers will occur at the
longitude grid established by the previous call to <code><a href="#topic+mapPlot">mapPlot()</a></code>;
(2) if <code>longitude=FALSE</code> then no longitude ticks or numbers are
drawn;
(3) if <code>longitude</code> is a vector of numerical values, then those ticks
are placed at those values, and numbers are written beside them.
Note that in cases 1 and 3, efforts are made to avoid overdrawing text,
so some longitude values might get ticks but not numbers. To get ticks
but not numbers, set <code>cex.axis=0</code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_latitude">latitude</code></td>
<td>
<p>similar to <code>longitude</code> but for latitude.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_axisstyle">axisStyle</code></td>
<td>
<p>an integer specifying the style of labels for the numbers
on axes.  The choices are:
1 for signed numbers without additional labels;
2 (the default) for unsigned numbers followed by letters indicating the hemisphere;
3 for signed numbers followed by a degree sign;
4 for unsigned numbers followed by a degree sign; and
5 for signed numbers followed by a degree sign and letters indicating the hemisphere.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_tick">tick</code></td>
<td>
<p>parameter passed to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_line">line</code></td>
<td>
<p>parameter passed to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_pos">pos</code></td>
<td>
<p>parameter passed to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_outer">outer</code></td>
<td>
<p>parameter passed to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_font">font</code></td>
<td>
<p>axis font, passed to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_lty">lty</code></td>
<td>
<p>axis line type, passed to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_lwd">lwd</code></td>
<td>
<p>axis line width, passed to <code><a href="graphics.html#topic+axis">axis()</a></code>).</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_lwd.ticks">lwd.ticks</code></td>
<td>
<p>tick line width, passed to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_col">col</code></td>
<td>
<p>axis color, passed to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_col.ticks">col.ticks</code></td>
<td>
<p>axis tick color, passed to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_hadj">hadj</code></td>
<td>
<p>an argument that is transmitted to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_padj">padj</code></td>
<td>
<p>an argument that is transmitted to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_tcl">tcl</code></td>
<td>
<p>axis-tick size (see <code><a href="graphics.html#topic+par">par()</a></code>).</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_cex.axis">cex.axis</code></td>
<td>
<p>axis-label expansion factor (see <code><a href="graphics.html#topic+par">par()</a></code>); set to 0
to prevent numbers from being placed in axes.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_mgp">mgp</code></td>
<td>
<p>three-element numerical vector describing axis-label
placement (see <code><a href="graphics.html#topic+par">par()</a></code>). It usually makes sense to set
the first and third elements to zero.</p>
</td></tr>
<tr><td><code id="mapAxis_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(coastlineWorld)
par(mar = c(2, 2, 1, 1))
lonlim &lt;- c(-180, 180)
latlim &lt;- c(70, 110)
# In mapPlot() call, note axes and grid args, to
# prevent over-plotting of defaults.  Some adjustments
# might be required to the mapGrid() arguments, to
# get agreement with the axis. This is why both
# mapGrid() and mapAxis() are best avoided; it is
# simpler to let mapPlot() handle these things.
mapPlot(coastlineWorld,
    projection = "+proj=stere +lat_0=90",
    longitudelim = lonlim, latitudelim = latlim,
    col = "tan", axes = FALSE, grid = FALSE
)
mapGrid(15, 15)
mapAxis(axisStyle = 5)


</code></pre>

<hr>
<h2 id='mapContour'>Add Contours on a Existing map</h2><span id='topic+mapContour'></span>

<h3>Description</h3>

<p>Plot contours on an existing map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapContour(
  longitude,
  latitude,
  z,
  nlevels = 10,
  levels = pretty(range(z, na.rm = TRUE), nlevels),
  labcex = 0.6,
  drawlabels = TRUE,
  underlay = "erase",
  col = par("fg"),
  lty = par("lty"),
  lwd = par("lwd"),
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapContour_+3A_longitude">longitude</code></td>
<td>
<p>numeric vector of longitudes of points to be plotted, or an object of
class <code>topo</code> (see <a href="#topic+topo-class">topo</a>), in which case
<code>longitude</code>, <code>latitude</code> and <code>z</code> are inferred from that object.</p>
</td></tr>
<tr><td><code id="mapContour_+3A_latitude">latitude</code></td>
<td>
<p>numeric vector of latitudes of points to be plotted.</p>
</td></tr>
<tr><td><code id="mapContour_+3A_z">z</code></td>
<td>
<p>matrix to be contoured. The number of rows and columns in <code>z</code>
must equal the lengths of <code>longitude</code> and <code>latitude</code>, respectively.</p>
</td></tr>
<tr><td><code id="mapContour_+3A_nlevels">nlevels</code></td>
<td>
<p>number of contour levels, if and only if <code>levels</code> is not supplied.</p>
</td></tr>
<tr><td><code id="mapContour_+3A_levels">levels</code></td>
<td>
<p>vector of contour levels.</p>
</td></tr>
<tr><td><code id="mapContour_+3A_labcex">labcex</code></td>
<td>
<p><code>cex</code> value used for contour labelling. As with
<code><a href="graphics.html#topic+contour">contour()</a></code>, this is an absolute size, not a multiple of
<code><a href="graphics.html#topic+par">par</a></code><code>("cex")</code>.</p>
</td></tr>
<tr><td><code id="mapContour_+3A_drawlabels">drawlabels</code></td>
<td>
<p>logical value or vector indicating whether to draw contour
labels.  If the length of <code>drawlabels</code> is less than the number of
levels specified, then <code><a href="base.html#topic+rep">rep()</a></code> is used to increase the length,
providing a value for each contour line. For those levels that are thus
indicated, labels are added, at a spot where the contour line is
closest to horizontal on the page. First, though, the region underneath
the label is filled with the colour given by <code><a href="graphics.html#topic+par">par</a></code><code>("bg")</code>.
See &ldquo;Limitations&rdquo; for notes on the status of contour
labelling, and its limitations.</p>
</td></tr>
<tr><td><code id="mapContour_+3A_underlay">underlay</code></td>
<td>
<p>character value relating to handling labels. If
this equals <code>"erase"</code> (which is the default), then the contour line
is drawn  first, then the area under the label is erased (filled with
white 'ink'), and then the label is drawn. This can be useful
in drawing coarsely-spaced labelled contours on top of finely-spaced
unlabelled contours. On the other hand, if <code>underlay</code> equals
<code>"interrupt"</code>, then the contour line is interrupted in the
region of the label, which is closer to the scheme used by the
base <code><a href="graphics.html#topic+contour">contour()</a></code> function.</p>
</td></tr>
<tr><td><code id="mapContour_+3A_col">col</code></td>
<td>
<p>colour of the contour line, as for <code><a href="graphics.html#topic+par">par</a></code><code>("col")</code>,
except here <code>col</code> gets lengthened by calling <code><a href="base.html#topic+rep">rep()</a></code>,
so that individual contours can be coloured distinctly.</p>
</td></tr>
<tr><td><code id="mapContour_+3A_lty">lty</code></td>
<td>
<p>type of the contour line, as for <code><a href="graphics.html#topic+par">par</a></code><code>("lty")</code>,
except for lengthening, as described for <code>col</code>.</p>
</td></tr>
<tr><td><code id="mapContour_+3A_lwd">lwd</code></td>
<td>
<p>width of the contour line, as for <code><a href="graphics.html#topic+par">par</a></code><code>("lwd")</code>,
except for lengthening, as described for <code>col</code> and <code>lty</code>.</p>
</td></tr>
<tr><td><code id="mapContour_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adds contour lines to an existing map, using <code><a href="#topic+mapLines">mapLines()</a></code>.
</p>
<p>The ability to label the contours was added in February, 2019, and
how this works may change through the summer months of that year.
Note that label placement in <code>mapContour</code> is handled differently
than in <code><a href="graphics.html#topic+contour">contour()</a></code>.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
data(coastlineWorld)
if (requireNamespace("ocedata", quietly=TRUE)) {
    data(levitus, package="ocedata")
    par(mar=rep(1, 4))
    mapPlot(coastlineWorld, projection="+proj=robin", col="lightgray")
    mapContour(levitus[["longitude"]], levitus[["latitude"]], levitus[["SST"]])
}
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>

<hr>
<h2 id='mapCoordinateSystem'>Draw a Coordinate System</h2><span id='topic+mapCoordinateSystem'></span>

<h3>Description</h3>

<p>Draws arrows on a map to indicate a coordinate system, e.g. for an
to indicate a coordinate system set up so that one axis is parallel
to a coastline.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapCoordinateSystem(longitude, latitude, L = 100, phi = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapCoordinateSystem_+3A_longitude">longitude</code></td>
<td>
<p>numeric vector of longitudes in degrees.</p>
</td></tr>
<tr><td><code id="mapCoordinateSystem_+3A_latitude">latitude</code></td>
<td>
<p>numeric vector of latitudes in degrees.</p>
</td></tr>
<tr><td><code id="mapCoordinateSystem_+3A_l">L</code></td>
<td>
<p>axis length in km.</p>
</td></tr>
<tr><td><code id="mapCoordinateSystem_+3A_phi">phi</code></td>
<td>
<p>angle, in degrees counterclockwise, that the &quot;x&quot; axis makes to a line of latitude.</p>
</td></tr>
<tr><td><code id="mapCoordinateSystem_+3A_...">...</code></td>
<td>
<p>plotting arguments, passed to <code><a href="#topic+mapArrows">mapArrows()</a></code>;
see &ldquo;Examples&rdquo; for how to control the arrow-head size.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a preliminary version of this function. It only
works if the lines of constant latitude are horizontal on the plot.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
if (requireNamespace("ocedata", quietly=TRUE)) {
    data(coastlineWorldFine, package="ocedata")
    HfxLon &lt;- -63.5752
    HfxLat &lt;- 44.6488
    mapPlot(coastlineWorldFine, proj="+proj=merc",
        longitudelim=HfxLon+c(-2,2), latitudelim=HfxLat+c(-2,2),
        col=lightgrey")
    mapCoordinateSystem(HfxLon, HfxLat, phi=45, length=0.05)
   }
</pre>


<h3>Author(s)</h3>

<p>Chantelle Layton
</p>


<h3>See Also</h3>

<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>

<hr>
<h2 id='mapDirectionField'>Add a Direction Field to an Existing Map</h2><span id='topic+mapDirectionField'></span>

<h3>Description</h3>

<p>Plot a direction field on a existing map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapDirectionField(
  longitude,
  latitude,
  u,
  v,
  scale = 1,
  length = 0.05,
  code = 2,
  col = par("fg"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapDirectionField_+3A_longitude">longitude</code>, <code id="mapDirectionField_+3A_latitude">latitude</code></td>
<td>
<p>numeric vectors of the starting points for arrows.</p>
</td></tr>
<tr><td><code id="mapDirectionField_+3A_u">u</code>, <code id="mapDirectionField_+3A_v">v</code></td>
<td>
<p>numeric vectors of the components of a vector to be shown as a direction
field.</p>
</td></tr>
<tr><td><code id="mapDirectionField_+3A_scale">scale</code></td>
<td>
<p>latitude degrees per unit of <code>u</code> or <code>v</code>.</p>
</td></tr>
<tr><td><code id="mapDirectionField_+3A_length">length</code></td>
<td>
<p>length of arrow heads, passed to <code><a href="graphics.html#topic+arrows">arrows()</a></code>.</p>
</td></tr>
<tr><td><code id="mapDirectionField_+3A_code">code</code></td>
<td>
<p>code of arrows, passed to <code><a href="graphics.html#topic+arrows">arrows()</a></code>.</p>
</td></tr>
<tr><td><code id="mapDirectionField_+3A_col">col</code></td>
<td>
<p>color of arrows.  This may be a single color, or a matrix
of colors of the same dimension as <code>u</code>.</p>
</td></tr>
<tr><td><code id="mapDirectionField_+3A_...">...</code></td>
<td>
<p>optional arguments passed to <code><a href="graphics.html#topic+arrows">arrows()</a></code>, e.g.
<code>angle</code> and <code>lwd</code> can be useful in differentiating different
fields.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adds arrows for a direction field on an existing map.  There are different
possibilities for how <code>longitude</code>, <code>latitude</code> and <code>u</code> and
<code>v</code> match up.  In one common case, all four of these are matrices, e.g.
output from a numerical model.  In another, <code>longitude</code> and
<code>latitude</code> are the coordinates along the matrices, and are thus stored in
vectors with lengths that match appropriately.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
data(coastlineWorld)
par(mar=rep(2, 4))
mapPlot(coastlineWorld, longitudelim=c(-120,-55), latitudelim=c(35, 50),
        projection="+proj=laea +lat0=40 +lat1=60 +lon_0=-110")
lon &lt;- seq(-120, -60, 15)
lat &lt;- 45 + seq(-15, 15, 5)
lonm &lt;- matrix(expand.grid(lon, lat)[, 1], nrow=length(lon))
latm &lt;- matrix(expand.grid(lon, lat)[, 2], nrow=length(lon))
# vectors pointed 45 degrees clockwise from north
u &lt;- matrix(1/sqrt(2), nrow=length(lon), ncol=length(lat))
v &lt;- matrix(1/sqrt(2), nrow=length(lon), ncol=length(lat))
mapDirectionField(lon, lat, u, v, scale=3)
mapDirectionField(lonm, latm, 0, 1, scale=3, col="red")
# Color code by longitude, using thick lines
col &lt;- colormap(lonm)$zcol
mapDirectionField(lonm, latm, 1, 0, scale=3, col=col, lwd=2)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>

<hr>
<h2 id='mapGrid'>Add a Longitude and Latitude Grid to an Existing Map</h2><span id='topic+mapGrid'></span>

<h3>Description</h3>

<p>Plot longitude and latitude grid on an existing map.
This is an advanced function, requiring
coordination with <code><a href="#topic+mapPlot">mapPlot()</a></code> and (possibly) also with <code><a href="#topic+mapAxis">mapAxis()</a></code>,
and so it is best avoided by novices, who may be satisfied
with the defaults used by <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapGrid(
  dlongitude = 15,
  dlatitude = 15,
  longitude,
  latitude,
  col = "darkgray",
  lty = "solid",
  lwd = 0.5 * par("lwd"),
  polarCircle = 0,
  longitudelim,
  latitudelim,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapGrid_+3A_dlongitude">dlongitude</code></td>
<td>
<p>increment in longitude, ignored if <code>longitude</code>
is supplied, but otherwise determines the longitude sequence.</p>
</td></tr>
<tr><td><code id="mapGrid_+3A_dlatitude">dlatitude</code></td>
<td>
<p>increment in latitude, ignored if <code>latitude</code>
is supplied, but otherwise determines the latitude sequence.</p>
</td></tr>
<tr><td><code id="mapGrid_+3A_longitude">longitude</code></td>
<td>
<p>numeric vector of longitudes, or <code>NULL</code> to prevent drawing
longitude lines.</p>
</td></tr>
<tr><td><code id="mapGrid_+3A_latitude">latitude</code></td>
<td>
<p>numeric vector of latitudes, or <code>NULL</code> to prevent drawing
latitude lines.</p>
</td></tr>
<tr><td><code id="mapGrid_+3A_col">col</code></td>
<td>
<p>color of lines</p>
</td></tr>
<tr><td><code id="mapGrid_+3A_lty">lty</code></td>
<td>
<p>line type</p>
</td></tr>
<tr><td><code id="mapGrid_+3A_lwd">lwd</code></td>
<td>
<p>line width</p>
</td></tr>
<tr><td><code id="mapGrid_+3A_polarcircle">polarCircle</code></td>
<td>
<p>a number indicating the number of degrees of latitude
extending from the poles, within which zones are not drawn.</p>
</td></tr>
<tr><td><code id="mapGrid_+3A_longitudelim">longitudelim</code></td>
<td>
<p>optional argument specifying suggested longitude limits
for the grid. If this is not supplied, grid lines are drawn for the
whole globe, which can yield excessively slow drawing speeds for
small-region plots. This, and <code>latitudelim</code>, are both set by
<code><a href="#topic+mapPlot">mapPlot()</a></code> if the arguments of the same name are passed to
that function.</p>
</td></tr>
<tr><td><code id="mapGrid_+3A_latitudelim">latitudelim</code></td>
<td>
<p>similar to <code>longitudelim</code>.</p>
</td></tr>
<tr><td><code id="mapGrid_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, 2 to go two function levels deep, or
3 to go all the way to the core functions. Any value above 3 will be
truncated to 3.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is somewhat analogous to <code><a href="graphics.html#topic+grid">grid()</a></code>, except that the
first two arguments of the latter supply the number of lines in the grid,
whereas the present function has increments for the first two arguments.
</p>


<h3>Value</h3>

<p>A <a href="base.html#topic+data.frame">data.frame</a>, returned silently, containing
<code>"side"</code>, <code>"value"</code>, <code>"type"</code>, and <code>"at"</code>.
A default call to <code><a href="#topic+mapPlot">mapPlot()</a></code> ensures agreement of grid and axes by using
this return value in a call to <code><a href="#topic+mapAxis">mapAxis()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (utils::packageVersion("sf") != "0.9.8") {
    # sf version 0.9-8 has a problem with this projection
    library(oce)
    data(coastlineWorld)
    par(mar = c(2, 2, 1, 1))
    # In mapPlot() call, note axes and grid args, to
    # prevent over-plotting of defaults.
    mapPlot(coastlineWorld,
        type = "l", projection = "+proj=ortho",
        axes = FALSE, grid = FALSE
    )
    mapGrid(15, 15)
}


</code></pre>

<hr>
<h2 id='mapImage'>Add an Image to a Map</h2><span id='topic+mapImage'></span>

<h3>Description</h3>

<p>Plot an image on an existing map that was created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapImage(
  longitude,
  latitude,
  z,
  zlim,
  zclip = FALSE,
  breaks,
  col,
  colormap,
  border = NA,
  lwd = par("lwd"),
  lty = par("lty"),
  missingColor = NA,
  filledContour = FALSE,
  gridder = "binMean2D",
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapImage_+3A_longitude">longitude</code></td>
<td>
<p>numeric vector of longitudes corresponding to <code>z</code> matrix.</p>
</td></tr>
<tr><td><code id="mapImage_+3A_latitude">latitude</code></td>
<td>
<p>numeric vector of latitudes corresponding to <code>z</code> matrix.</p>
</td></tr>
<tr><td><code id="mapImage_+3A_z">z</code></td>
<td>
<p>numeric matrix to be represented as an image.</p>
</td></tr>
<tr><td><code id="mapImage_+3A_zlim">zlim</code></td>
<td>
<p>limit for z (color).</p>
</td></tr>
<tr><td><code id="mapImage_+3A_zclip">zclip</code></td>
<td>
<p>A logical value, <code>TRUE</code> indicating that out-of-range
<code>z</code> values should be painted with <code>missingColor</code> and <code>FALSE</code>
indicating that these values should be painted with the nearest
in-range color.  If <code>zlim</code> is given then its min and max set the
range.  If <code>zlim</code> is not given but <code>breaks</code> is given, then
the min and max of <code>breaks</code> sets the range used for z.  If neither
<code>zlim</code> nor <code>breaks</code> is given, clipping is not done, i.e. the
action is as if <code>zclip</code> were <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="mapImage_+3A_breaks">breaks</code></td>
<td>
<p>The z values for breaks in the color scheme.  If this is of
length 1, the value indicates the desired number of breaks, which is
supplied to <code><a href="base.html#topic+pretty">pretty()</a></code>, in determining clean break points.</p>
</td></tr>
<tr><td><code id="mapImage_+3A_col">col</code></td>
<td>
<p>Either a vector of colors corresponding to the breaks, of length
1 plus the number of breaks, or a function specifying colors,
e.g. <code><a href="#topic+oce.colorsViridis">oce.colorsViridis()</a></code> for the Viridis scheme.</p>
</td></tr>
<tr><td><code id="mapImage_+3A_colormap">colormap</code></td>
<td>
<p>optional colormap, as created by <code><a href="#topic+colormap">colormap()</a></code>.
If a <code>colormap</code> is provided, then its properties takes precedence
over <code>breaks</code>, <code>col</code>, <code>missingColor</code>, and <code>zclip</code>
specified to <code>mapImage</code>.</p>
</td></tr>
<tr><td><code id="mapImage_+3A_border">border</code></td>
<td>
<p>Color used for borders of patches (passed to
<code><a href="graphics.html#topic+polygon">polygon()</a></code>); the default <code>NA</code> means no border.</p>
</td></tr>
<tr><td><code id="mapImage_+3A_lwd">lwd</code></td>
<td>
<p>line width, used if borders are drawn.</p>
</td></tr>
<tr><td><code id="mapImage_+3A_lty">lty</code></td>
<td>
<p>line type, used if borders are drawn.</p>
</td></tr>
<tr><td><code id="mapImage_+3A_missingcolor">missingColor</code></td>
<td>
<p>a color to be used to indicate missing data, or
<code>NA</code> to skip the drawing of such regions (which will retain
whatever material has already been drawn at the regions).</p>
</td></tr>
<tr><td><code id="mapImage_+3A_filledcontour">filledContour</code></td>
<td>
<p>either a logical value indicating whether to use
filled contours to plot the image, or a numerical value indicating the
resampling rate to be used in interpolating from lon-lat coordinates to
x-y coordinates.  See &ldquo;Details&rdquo; for how this interacts with
<code>gridder</code>.</p>
</td></tr>
<tr><td><code id="mapImage_+3A_gridder">gridder</code></td>
<td>
<p>character value specifying the gridding function to be used
if <code>filledContour</code> is <code>TRUE</code>. This can be either <code>"binMean2D"</code> to select
<code><a href="#topic+binMean2D">binMean2D()</a></code> or <code>"interp"</code> to select <code><a href="interp.html#topic+interp">interp::interp()</a></code>.  The former
produces cruder results, but the latter can be slow for large datasets.
Note that <code>"akima"</code> is taken as a synonym for <code>"interp"</code>
(see &ldquo;Historical Notes&rdquo;).</p>
</td></tr>
<tr><td><code id="mapImage_+3A_debug">debug</code></td>
<td>
<p>A flag that turns on debugging.  Set to 1 to get a
moderate amount of debugging information, or to 2 to get more.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Image data are on a regular grid in lon-lat space, but not in the projected
x-y space.  This means that <code><a href="Matrix.html#topic+image">image()</a></code> cannot be used.  Instead,
there are two approaches, depending on the value of <code>filledContour</code>.
</p>
<p>If <code>filledContour</code> is <code>FALSE</code>, the image &quot;pixels&quot; are drawn with <code><a href="graphics.html#topic+polygon">polygon()</a></code>.
This can be prohibitively slow for fine grids.  However, if <code>filledContour</code>
is <code>TRUE</code> or a numerical value, then the &quot;pixels&quot; are remapped into a regular
grid and then displayed with <code><a href="graphics.html#topic+.filled.contour">.filled.contour()</a></code>.  The remapping starts by
converting the regular lon-lat grid to an irregular x-y grid using
<code><a href="#topic+lonlat2map">lonlat2map()</a></code>.  This irregular grid is then interpolated onto a regular x-y
grid  with either <code><a href="#topic+binMean2D">binMean2D()</a></code> or <code><a href="interp.html#topic+interp">interp::interp()</a></code>, as determined by the
value of the <code>gridder</code> parameter. If <code>filledContour</code> is <code>TRUE</code>, the
dimensions of the regular x-y grid is the same as that of the original
lon-lat grid; otherwise, the number of rows and columns are multiplied by the
numerical value of <code>filledContour</code>, e.g. the value 2 means to make the grid
twice as fine.
</p>
<p>Filling contours can produce aesthetically-pleasing results, but the method
involves interpolation, so the data are not represented exactly and
analysts are advised to compare the results from the two methods (and
perhaps various grid refinement values) to guard against misinterpretation.
</p>
<p>If a <code><a href="grDevices.html#topic+png">png()</a></code> device is to be used, it is advised to supply
arguments <code>type="cairo"</code> and <code>antialias="none"</code> (see reference 1).
</p>


<h3>Historical Notes</h3>

<p>Until oce 1.7.4, the <code>gridder</code> argument could be set to <code>"akima"</code>, which used
the <code>akima</code> package.  However, that package is not released with a FOSS license,
so CRAN requested a change to <a href="https://CRAN.R-project.org/package=interp"><span class="pkg">interp</span></a>. Note that <code>drawImage()</code>
intercepts the errors that sometimes get reported by <code><a href="interp.html#topic+interp">interp::interp()</a></code>.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
data(coastlineWorld)
data(topoWorld)

# Northern polar region, with color-coded bathymetry
par(mfrow=c(1,1), mar=c(2,2,1,1))
cm &lt;- colormap(zlim=c(-5000, 0), col=oceColorsGebco)
drawPalette(colormap=cm)
mapPlot(coastlineWorld, projection="+proj=stere +lat_0=90",
        longitudelim=c(-180,180), latitudelim=c(70,110))
mapImage(topoWorld, colormap=cm)
mapGrid(15, 15, polarCircle=1, col=gray(0.2))
mapPolygon(coastlineWorld[["longitude"]], coastlineWorld[["latitude"]], col="tan")
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li> <p><code style="white-space: pre;">&#8288;https://codedocean.wordpress.com/2014/02/03/anti-aliasing-and-image-plots/&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>

<hr>
<h2 id='mapLines'>Add Lines to a Map</h2><span id='topic+mapLines'></span>

<h3>Description</h3>

<p>Plot lines on an existing map, by analogy to <code><a href="graphics.html#topic+lines">lines()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapLines(longitude, latitude, greatCircle = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapLines_+3A_longitude">longitude</code></td>
<td>
<p>numeric vector of longitudes of points to be plotted, or an
object from which longitude and latitude can be inferred (e.g. a coastline
file, or the return value from <code><a href="#topic+mapLocator">mapLocator()</a></code>), in which case the
following two arguments are ignored.</p>
</td></tr>
<tr><td><code id="mapLines_+3A_latitude">latitude</code></td>
<td>
<p>vector of latitudes of points to be plotted.</p>
</td></tr>
<tr><td><code id="mapLines_+3A_greatcircle">greatCircle</code></td>
<td>
<p>a logical value indicating whether to render line
segments as great circles.  (Ignored.)</p>
</td></tr>
<tr><td><code id="mapLines_+3A_...">...</code></td>
<td>
<p>optional arguments passed to <code><a href="graphics.html#topic+lines">lines()</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (utils::packageVersion("sf") != "0.9.8") {
    # sf version 0.9-8 has a problem with this projection
    library(oce)
    data(coastlineWorld)
    mapPlot(coastlineWorld,
        type = "l",
        longitudelim = c(-80, 10), latitudelim = c(0, 120),
        projection = "+proj=ortho +lon_0=-40"
    )
    lon &lt;- c(-63.5744, 0.1062) # Halifax CA to London UK
    lat &lt;- c(44.6479, 51.5171)
    mapPoints(lon, lat, col = "red")
    mapLines(lon, lat, col = "red")
}


</code></pre>

<hr>
<h2 id='mapLocator'>Locate Points on a Map</h2><span id='topic+mapLocator'></span>

<h3>Description</h3>

<p>Locate points on an existing map.
This uses <code><a href="#topic+map2lonlat">map2lonlat()</a></code> to infer the location in
geographical space, so it suffers the same
limitations as that function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapLocator(n = 512, type = "n", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapLocator_+3A_n">n</code></td>
<td>
<p>number of points to locate; see <code><a href="graphics.html#topic+locator">locator()</a></code>.</p>
</td></tr>
<tr><td><code id="mapLocator_+3A_type">type</code></td>
<td>
<p>type of connector for the points; see <code><a href="graphics.html#topic+locator">locator()</a></code>.</p>
</td></tr>
<tr><td><code id="mapLocator_+3A_...">...</code></td>
<td>
<p>extra arguments passed to <code><a href="graphics.html#topic+locator">locator()</a></code> (and either
<code><a href="#topic+mapPoints">mapPoints()</a></code> or <code><a href="#topic+mapLines">mapLines()</a></code>, if appropriate) if
<code>type</code> is not <code>'n'</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>

<hr>
<h2 id='mapLongitudeLatitudeXY'>Convert From Longitude and Latitude to X and Y</h2><span id='topic+mapLongitudeLatitudeXY'></span>

<h3>Description</h3>

<p>Find (x, y) values corresponding to (longitude, latitude) values, using the
present projection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapLongitudeLatitudeXY(longitude, latitude)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapLongitudeLatitudeXY_+3A_longitude">longitude</code></td>
<td>
<p>numeric vector of the longitudes of points, or an object from which
both latitude and longitude can be inferred (e.g. a coastline file, or the
return value from <code><a href="#topic+mapLocator">mapLocator()</a></code>), in which case the following
two arguments are ignored.</p>
</td></tr>
<tr><td><code id="mapLongitudeLatitudeXY_+3A_latitude">latitude</code></td>
<td>
<p>numeric vector of latitudes of points, needed only if they cannot
be inferred from the first argument.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is mainly a wrapper around <code><a href="#topic+lonlat2map">lonlat2map()</a></code>.
</p>


<h3>Value</h3>

<p>A list containing <code>x</code> and <code>y</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(coastlineWorld)
par(mfrow = c(2, 1), mar = rep(2, 4))
mapPlot(coastlineWorld, projection = "+proj=moll") # sets a projection
xy &lt;- mapLongitudeLatitudeXY(coastlineWorld)
plot(xy, type = "l", asp = 1)


</code></pre>

<hr>
<h2 id='mapPlot'>Draw a Map</h2><span id='topic+mapPlot'></span>

<h3>Description</h3>

<p>Plot coordinates as a map, using one of the subset of projections
provided by the <a href="https://CRAN.R-project.org/package=sf"><span class="pkg">sf</span></a> package.  The projection information specified
with the <code>mapPlot()</code> call is stored in a global variable that can be retrieved
by related functions, making it easy to add points, lines, text, images
or contours to an existing map.
The &ldquo;Details&rdquo; section, below, provides a list of
available projections. The &quot;Using map projections&quot; vignette
offers examples of several map plots, in addition to the single
example provided in the &ldquo;Examples&rdquo; section, below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapPlot(
  longitude,
  latitude,
  longitudelim,
  latitudelim,
  grid = TRUE,
  geographical = 0,
  bg,
  fill,
  border = NULL,
  col = NULL,
  clip = TRUE,
  type = "polygon",
  axes = TRUE,
  axisStyle = 1,
  cex,
  cex.axis = 1,
  mgp = c(0, 0.5, 0),
  drawBox = TRUE,
  showHemi = TRUE,
  polarCircle = 0,
  lonlabels = TRUE,
  latlabels = TRUE,
  projection = "+proj=moll",
  tissot = FALSE,
  trim = TRUE,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapPlot_+3A_longitude">longitude</code></td>
<td>
<p>either a numeric vector of longitudes of points to be plotted, or
something (an <code>oce</code> object, a list, or a data frame) from which both
longitude and latitude may be inferred (in which case the <code>latitude</code>
argument is ignored).  If <code>longitude</code> is missing, both it and
<code>latitude</code> are taken from the built-in <a href="#topic+coastlineWorld">coastlineWorld</a> dataset.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_latitude">latitude</code></td>
<td>
<p>numeric vector of latitudes of points to be plotted (ignored
if the first argument contains both latitude and longitude).</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_longitudelim">longitudelim</code>, <code id="mapPlot_+3A_latitudelim">latitudelim</code></td>
<td>
<p>optional numeric vectors of length two, indicating the
limits of the plot. A warning is issued if these are not specified together.
See &ldquo;Examples&rdquo; for a polar-region example, noting that
the whole-globe span of <code>longitudelim</code> is used to centre the plot
at the north pole.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_grid">grid</code></td>
<td>
<p>either a number (or pair of numbers) indicating the spacing of
longitude and latitude lines, in degrees, or a logical value (or pair of
values) indicating whether to draw an auto-scaled grid, or whether to skip
the grid drawing.  In the case of numerical values, <code>NA</code> can be used to
turn off the grid in longitude or latitude.  Grids are set up based on
examination of the scale used in middle 10 percent of the plot area, and for
most projections this works quite well.  If not, one may set
<code>grid=FALSE</code> and add a grid later with <code><a href="#topic+mapGrid">mapGrid()</a></code>.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_geographical">geographical</code></td>
<td>
<p>flag indicating the style of axes.  With
<code>geographical=0</code>, the axes are conventional, with decimal degrees as
the unit, and negative signs indicating the southern and western
hemispheres.  With <code>geographical=1</code>, the signs are dropped, with axis
values being in decreasing order within the southern and western
hemispheres.  With <code>geographical=2</code>, the signs are dropped and the axes
are labelled with degrees, minutes and seconds, as appropriate, and
hemispheres are indicated with letters. With <code>geographical=3</code>, things
are the same as for <code>geographical=2</code>, but the hemisphere indication
is omitted. Finally, with <code>geographical=4</code>, unsigned numbers are used,
followed by letters <code>N</code> in the northern hemisphere, <code>S</code> in the southern,
<code>E</code> in the eastern, and <code>W</code> in the western.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_bg">bg</code></td>
<td>
<p>color of the background (ignored).</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_fill">fill</code></td>
<td>
<p>is a deprecated argument; see <a href="#topic+oce-deprecated">oce-deprecated</a>.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_border">border</code></td>
<td>
<p>color of coastlines and international borders (ignored unless
<code>type="polygon"</code>.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_col">col</code></td>
<td>
<p>either the color for filling polygons (if <code>type="polygon"</code>)
or the color of the points and line segments (if <code>type="p"</code>,
<code>type="l"</code>, or <code>type="o"</code>). If <code>col=NULL</code> then a default
will be set: no coastline filling for the <code>type="polygon"</code> case,
or black coastlines, for <code>type="p"</code>, <code>type="l"</code>, or
<code>type="o"</code>.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_clip">clip</code></td>
<td>
<p>logical value indicating whether to trim any coastline elements that lie wholly
outside the plot region. This can prevent e.g. a problem of filling the whole plot area of
an Arctic stereopolar view, because the projected trace for Antarctica lies outside all
other regions so the whole of the world ends up being &quot;land&quot;.  Setting <code>clip=FALSE</code>
disables this action, which may be of benefit in rare instances in the line connecting
two points on a coastline may cross the plot domain, even if those points are outside
that domain.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_type">type</code></td>
<td>
<p>indication of type; may be <code>"polygon"</code>, for a filled polygon,
<code>"p"</code> for points, <code>"l"</code> for line segments, or <code>"o"</code> for points
overlain with line segments.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_axes">axes</code></td>
<td>
<p>a logical value indicating whether to draw longitude and latitude
values in the lower and left margin, respectively.  This may not work well
for some projections or scales.  See also <code>axisStyle</code>, <code>lonlabels</code>
and <code>latlabels</code>, which offer more granular control of labelling.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_axisstyle">axisStyle</code></td>
<td>
<p>an integer specifying the style of labels for the numbers
on axes.  The choices are:
1 for signed numbers without additional labels;
2 (the default) for unsigned numbers followed by letters indicating the hemisphere;
3 for signed numbers followed by a degree sign;
4 for unsigned numbers followed by a degree sign; and
5 for signed numbers followed by a degree sign and letters indicating the hemisphere.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_cex">cex</code></td>
<td>
<p>character expansion factor for plot symbols,
used if <code>type="p"</code> or any other value that yields symbols.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_cex.axis">cex.axis</code></td>
<td>
<p>axis-label expansion factor (see <code><a href="graphics.html#topic+par">par()</a></code>).</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_mgp">mgp</code></td>
<td>
<p>three-element numerical vector describing axis-label
placement, passed to <code><a href="#topic+mapAxis">mapAxis()</a></code>.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_drawbox">drawBox</code></td>
<td>
<p>logical value indicating whether to draw a box around the plot.
This is helpful for many projections at sub-global scale.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_showhemi">showHemi</code></td>
<td>
<p>logical value indicating whether to show the hemisphere in
axis tick labels.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_polarcircle">polarCircle</code></td>
<td>
<p>a number indicating the number of degrees of latitude
extending from the poles, within which zones are not drawn.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_lonlabels">lonlabels</code></td>
<td>
<p>An optional logical value or numeric vector that controls
the labelling along the horizontal axis. There are four possibilities:
(1) If <code>lonlabels</code> is <code>TRUE</code> (the default), then reasonable values are inferred
and axes are drawn with ticks and labels alongside those ticks;
(2) if <code>lonlabels</code> is <code>FALSE</code>, then ticks are drawn, but no labels;
(3) if <code>lonlabels</code> is <code>NULL</code>, then no axis ticks or labels are drawn; and
(4) if <code>lonlabels</code> is  a vector of finite numerical values, then tick marks
are placed  at those longitudes, and labels are put alongside them.
Note that R tries to avoid overwriting labels on axes, so the instructions
in case 4 might not be obeyed exactly.
See also <code>latlabels</code>, and note that setting <code>axes=FALSE</code>
ensures that no longitude or latitude axes will be drawn regardless
of the values of <code>lonlabels</code> and <code>latlabels</code>.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_latlabels">latlabels</code></td>
<td>
<p>As <code>lonlabels</code>, but for latitude, on the left
plot axis.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_projection">projection</code></td>
<td>
<p>either character value indicating the map projection, or
the output from <code><a href="sf.html#topic+st_crs">sf::st_crs()</a></code>. In the first case, see a table
in &ldquo;Details&rdquo; for the projections that are available.
In the second case, note that <code><a href="#topic+mapPlot">mapPlot()</a></code> reports an error if
a similar function from the old <code>sp</code> package is used.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_tissot">tissot</code></td>
<td>
<p>logical value indicating whether to use <code><a href="#topic+mapTissot">mapTissot()</a></code>
to plot Tissot indicatrices, i.e. ellipses at grid intersection points, which
indicate map distortion.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_trim">trim</code></td>
<td>
<p>logical value indicating whether to trim islands or lakes
containing only points that are off-scale of the current plot box.  This
solves the problem of Antarctica overfilling the entire domain, for an
Arctic-centred stereographic projection.  It is not a perfect solution,
though, because the line segment joining two off-scale points might
intersect the plotting box.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="mapPlot_+3A_...">...</code></td>
<td>
<p>optional arguments passed to some plotting functions.  This can
be useful in many ways, e.g.  Example 5 shows how to use <code>xlim</code> etc to
reproduce a scale exactly between two plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The calculations for map projections are done with
the <a href="https://CRAN.R-project.org/package=sf"><span class="pkg">sf</span></a> package.  Importantly, though, not all
the <a href="https://CRAN.R-project.org/package=sf"><span class="pkg">sf</span></a> projections are
available in <code>oce</code>, for reasons relating to limitations
of <a href="https://CRAN.R-project.org/package=sf"><span class="pkg">sf</span></a>, for example relating to inverse-projection
calculations. The <code>oce</code> choices are tabulated
below, e.g. <code>projection="+proj=aea"</code> selects the Albers equal area
projection.  (See also the warning, below, about a problem
with <a href="https://CRAN.R-project.org/package=sf"><span class="pkg">sf</span></a> version 0.9-8.)
</p>
<p>Further details of the vast array of map projections provided by PROJ
are given in reference 4.  This system has been in rapid development
since about 2018, and reference 5 provides a helpful overview of the changes
and the reasons why they were necessary.  Practical examples of map projections in
<a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a> are provided in reference 6, along with some notes on problems.
A fascinating treatment of the history of map projections is provided
in reference 7.  To get an idea of how projections are being created
nowadays, see reference 8, about the <code>eqearth</code> projection that was added
to <a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a> in August 2020.
</p>


<h3>Available Projections</h3>

<p>The following table lists projections available in <a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a>, and was generated
by reformatting a subset of the output of the unix
command <code>proj -lP</code>. Most of the arguments have default values, and many
projections also have optional arguments.  Although e.g. <code>proj -l=aea</code>
provides a little more information about particular projections,
users ought to consult reference 4 for fuller details and illustrations.</p>

<table>
<tr>
 <td style="text-align: left;">
   <strong>Projection</strong> </td><td style="text-align: left;"> <strong>Code</strong> </td><td style="text-align: left;"> <strong>Arguments</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Albers equal area </td><td style="text-align: left;"> <code>aea</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>lat_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Azimuthal equidistant </td><td style="text-align: left;"> <code>aeqd</code> </td><td style="text-align: left;"> <code>lat_0</code>, <code>guam</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Aitoff </td><td style="text-align: left;"> <code>aitoff</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Mod. stererographics of Alaska </td><td style="text-align: left;"> <code>alsk</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Bipolar conic of western hemisphere </td><td style="text-align: left;"> <code>bipc</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Bonne Werner </td><td style="text-align: left;"> <code>bonne</code> </td><td style="text-align: left;"> <code>lat_1</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Cassini </td><td style="text-align: left;"> <code>cass</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Central cylindrical </td><td style="text-align: left;"> <code>cc</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Equal area cylindrical </td><td style="text-align: left;"> <code>cea</code> </td><td style="text-align: left;"> <code>lat_ts</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Collignon </td><td style="text-align: left;"> <code>collg</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Craster parabolic Putnins P4 </td><td style="text-align: left;"> <code>crast</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Eckert I </td><td style="text-align: left;"> <code>eck1</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Eckert II </td><td style="text-align: left;"> <code>eck2</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Eckert III </td><td style="text-align: left;"> <code>eck3</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Eckert IV </td><td style="text-align: left;"> <code>eck4</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Eckert V </td><td style="text-align: left;"> <code>eck5</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Eckert VI </td><td style="text-align: left;"> <code>eck6</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Equidistant cylindrical plate (Caree) </td><td style="text-align: left;"> <code>eqc</code> </td><td style="text-align: left;"> <code>lat_ts</code>, <code>lat_0</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Equidistant conic </td><td style="text-align: left;"> <code>eqdc</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>lat_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Equal earth </td><td style="text-align: left;"> <code>eqearth</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Euler </td><td style="text-align: left;"> <code>euler</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>lat_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Extended transverse Mercator </td><td style="text-align: left;"> <code>etmerc</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Fahey </td><td style="text-align: left;"> <code>fahey</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Foucault </td><td style="text-align: left;"> <code>fouc</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Foucault sinusoidal </td><td style="text-align: left;"> <code>fouc_s</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Gall stereographic </td><td style="text-align: left;"> <code>gall</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Geostationary satellite view </td><td style="text-align: left;"> <code>geos</code> </td><td style="text-align: left;"> <code>h</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   General sinusoidal series </td><td style="text-align: left;"> <code>gn_sinu</code> </td><td style="text-align: left;"> <code>m</code>, <code>n</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Gnomonic </td><td style="text-align: left;"> <code>gnom</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Goode homolosine </td><td style="text-align: left;"> <code>goode</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Hatano asymmetrical equal area </td><td style="text-align: left;"> <code>hatano</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Interrupted Goode homolosine </td><td style="text-align: left;"> <code>igh</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Kavraisky V </td><td style="text-align: left;"> <code>kav5</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Kavraisky VII </td><td style="text-align: left;"> <code>kav7</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Lambert azimuthal equal area </td><td style="text-align: left;"> <code>laea</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Longitude and latitude </td><td style="text-align: left;"> <code>longlat</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Longitude and latitude </td><td style="text-align: left;"> <code>latlong</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Lambert conformal conic </td><td style="text-align: left;"> <code>lcc</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>lat_2</code> or <code>lat_0</code>, <code>k_0</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Lambert equal area conic </td><td style="text-align: left;"> <code>leac</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>south</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Loximuthal </td><td style="text-align: left;"> <code>loxim</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Space oblique for Landsat </td><td style="text-align: left;"> <code>lsat</code> </td><td style="text-align: left;"> <code>lsat</code>, <code>path</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   McBryde-Thomas flat-polar sine, no. 1 </td><td style="text-align: left;"> <code>mbt_s</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   McBryde-Thomas flat-polar sine, no. 2 </td><td style="text-align: left;"> <code>mbt_fps</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   McBryde-Thomas flat-polar parabolic </td><td style="text-align: left;"> <code>mbtfpp</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   McBryde-Thomas flat-polar quartic </td><td style="text-align: left;"> <code>mbtfpq</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   McBryde-Thomas flat-polar sinusoidal </td><td style="text-align: left;"> <code>mbtfps</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Mercator </td><td style="text-align: left;"> <code>merc</code> </td><td style="text-align: left;"> <code>lat_ts</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Miller oblated stereographic </td><td style="text-align: left;"> <code>mil_os</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Miller cylindrical </td><td style="text-align: left;"> <code>mill</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Mollweide </td><td style="text-align: left;"> <code>moll</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Murdoch I </td><td style="text-align: left;"> <code>murd1</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>lat_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Murdoch II </td><td style="text-align: left;"> <code>murd2</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>lat_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   murdoch III </td><td style="text-align: left;"> <code>murd3</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>lat_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Natural earth </td><td style="text-align: left;"> <code>natearth</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Nell </td><td style="text-align: left;"> <code>nell</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Nell-Hammer </td><td style="text-align: left;"> <code>nell_h</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Near-sided perspective </td><td style="text-align: left;"> <code>nsper</code> </td><td style="text-align: left;"> <code>h</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   New Zealand map grid </td><td style="text-align: left;"> <code>nzmg</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   General oblique transformation </td><td style="text-align: left;"> <code>ob_tran</code> </td><td style="text-align: left;"> <code>o_proj</code>, <code>o_lat_p</code>, <code>o_lon_p</code>, </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;">  </td><td style="text-align: left;"> <code>o_alpha</code>, <code>o_lon_c</code>, <code>o_lat_c</code>, </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;">  </td><td style="text-align: left;"> <code>o_lon_1</code>, <code>o_lat_1</code>, </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;">  </td><td style="text-align: left;"> <code>o_lon_2</code>, <code>o_lat_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Oblique cylindrical equal area </td><td style="text-align: left;"> <code>ocea</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>lat_2</code>, <code>lon_1</code>, <code>lon_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Oblated equal area </td><td style="text-align: left;"> <code>oea</code> </td><td style="text-align: left;"> <code>n</code>, <code>m</code>, <code>theta</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Oblique Mercator </td><td style="text-align: left;"> <code>omerc</code> </td><td style="text-align: left;"> <code>alpha</code>, <code>gamma</code>, <code>no_off</code>, </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;">  </td><td style="text-align: left;"> <code>lonc</code>, <code>lon_1</code>, <code>lat_1</code>, </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td><td style="text-align: left;">  </td><td style="text-align: left;"> <code>lon_2</code>, <code>lat_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Orthographic </td><td style="text-align: left;"> <code>ortho</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Polyconic American </td><td style="text-align: left;"> <code>poly</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Putnins P1 </td><td style="text-align: left;"> <code>putp1</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Putnins P2 </td><td style="text-align: left;"> <code>putp2</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Putnins P3 </td><td style="text-align: left;"> <code>putp3</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Putnins P3' </td><td style="text-align: left;"> <code>putp3p</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Putnins P4' </td><td style="text-align: left;"> <code>putp4p</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Putnins P5 </td><td style="text-align: left;"> <code>putp5</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Putnins P5' </td><td style="text-align: left;"> <code>putp5p</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Putnins P6 </td><td style="text-align: left;"> <code>putp6</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Putnins P6' </td><td style="text-align: left;"> <code>putp6p</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Quartic authalic </td><td style="text-align: left;"> <code>qua_aut</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Quadrilateralized spherical cube </td><td style="text-align: left;"> <code>qsc</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Robinson </td><td style="text-align: left;"> <code>robin</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Roussilhe stereographic </td><td style="text-align: left;"> <code>rouss</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Sinusoidal aka Sanson-Flamsteed </td><td style="text-align: left;"> <code>sinu</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Swiss. oblique Mercator </td><td style="text-align: left;"> <code>somerc</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Stereographic </td><td style="text-align: left;"> <code>stere</code> </td><td style="text-align: left;"> <code>lat_ts</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Oblique stereographic alternative </td><td style="text-align: left;"> <code>sterea</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Transverse cylindrical equal area </td><td style="text-align: left;"> <code>tcea</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Tissot </td><td style="text-align: left;"> <code>tissot</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>lat_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Transverse Mercator </td><td style="text-align: left;"> <code>tmerc</code> </td><td style="text-align: left;"> <code>approx</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Two point equidistant </td><td style="text-align: left;"> <code>tpeqd</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>lon_1</code>, <code>lat_2</code>, <code>lon_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Tilted perspective </td><td style="text-align: left;"> <code>tpers</code> </td><td style="text-align: left;"> <code>tilt</code>, <code>azi</code>, <code>h</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Universal polar stereographic </td><td style="text-align: left;"> <code>ups</code> </td><td style="text-align: left;"> <code>south</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Urmaev flat-polar sinusoidal </td><td style="text-align: left;"> <code>urmfps</code> </td><td style="text-align: left;"> <code>n</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Universal transverse Mercator </td><td style="text-align: left;"> <code>utm</code> </td><td style="text-align: left;"> <code>zone</code>, <code>south</code>, <code>approx</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   van der Grinten I </td><td style="text-align: left;"> <code>vandg</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Vitkovsky I </td><td style="text-align: left;"> <code>vitk1</code> </td><td style="text-align: left;"> <code>lat_1</code>, <code>lat_2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Wagner I Kavraisky VI </td><td style="text-align: left;"> <code>wag1</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Wagner II </td><td style="text-align: left;"> <code>wag2</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Wagner III </td><td style="text-align: left;"> <code>wag3</code> </td><td style="text-align: left;"> <code>lat_ts</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Wagner IV </td><td style="text-align: left;"> <code>wag4</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Wagner V </td><td style="text-align: left;"> <code>wag5</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Wagner VI </td><td style="text-align: left;"> <code>wag6</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Werenskiold I </td><td style="text-align: left;"> <code>weren</code> </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
   Winkel I </td><td style="text-align: left;"> <code>wink1</code> </td><td style="text-align: left;"> <code>lat_ts</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   Winkel Tripel </td><td style="text-align: left;"> <code>wintri</code> </td><td style="text-align: left;"> <code>lat_ts</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Choosing a projection</h3>

<p>The best choice of projection depends on the application.
Users may find <code>projection="+proj=moll"</code> useful for world-wide
plots, <code>ortho</code> for hemispheres viewed from the equator, <code>stere</code>
for polar views, <code>lcc</code> for wide meridional ranges in mid latitudes,
<code>merc</code> in limited-area cases where angle preservation is
important, or either <code>aea</code> or <code>eqearth</code> (on local and global
scales, respectively) where area preservation is important.
The choice becomes more important, the larger the size of the region
represented.  When it comes to publication, it can be sensible to use the
same projection as used in previous reports.
</p>


<h3>Problems</h3>

<p>Map projection is a complicated matter that is addressed here
in a limited and pragmatic way.  For example, <code>mapPlot</code> tries to draw
axes along a box containing the map, instead of trying to find spots along
the &ldquo;edge&rdquo; of the map at which to put longitude and latitude labels.
This design choice greatly simplifies the coding effort, freeing up time to
work on issues regarded as more pressing.  Chief among those issues are (a)
the occurrence of horizontal lines in maps that have prime meridians
(b) inaccurate filling of land regions that (again) occur with shifted
meridians and (c) inaccurate filling of Antarctica in some projections.
Generally, issues are tackled first for commonly used projections, such as
those used in the examples.
</p>


<h3>Historical Notes</h3>


<ul>
<li><p> 2020-12-24: complete switch from <code>rgdal</code> to <a href="https://CRAN.R-project.org/package=sf"><span class="pkg">sf</span></a>,
removing the testing scheme created on 2020-08-03.
</p>
</li>
<li><p> 2020-08-03: added support for the <code>eqearth</code> projection (like <code>robin</code> but
an equal-area method).
</p>
</li>
<li><p> 2020-08-03: dropped support for the <code>healpix</code>, <code>pconic</code>
and <code>rhealpix</code> projections, which caused errors with the
<a href="https://CRAN.R-project.org/package=sf"><span class="pkg">sf</span></a> package.  (This is not a practical loss, since these
interrupted projections were handled badly by <code>mapPlot()</code>
in any case.)
</p>
</li>
<li><p> 2020-08-03: switch from <code>rgdal</code> to <a href="https://CRAN.R-project.org/package=sf"><span class="pkg">sf</span></a> for
calculations related to map projection, owing to some
changes in the former package that broke <a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a>
code.  (To catch problems, <a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a> was set up to use
both packages temporarily, issuing warnings if the results differed
by more than 1 metre in easting or northing values.)
</p>
</li>
<li><p> 2017-11-19: <code>imw_p</code> removed, because it has problems doing
inverse calculations.
This is a also problem in the standalone PROJ.4 application version
4.9.3, downloaded and built on OSX.
See <code style="white-space: pre;">&#8288;https://github.com/dankelley/oce/issues/1319&#8288;</code> for details.
</p>
</li>
<li><p> 2017-11-17: <code>lsat</code> removed, because it does not work in
<code>rgdal</code> or in the latest standalone PROJ.4 application.
This is a also problem in the standalone PROJ.4 application version
4.9.3, downloaded and built on OSX.
See <code style="white-space: pre;">&#8288;https://github.com/dankelley/oce/issues/1337&#8288;</code> for details.
</p>
</li>
<li><p> 2017-09-30: <code>lcca</code> removed, because its inverse was
wildly inaccurate in a Pacific Antarctic-Alaska application
(see <code style="white-space: pre;">&#8288;https://github.com/dankelley/oce/issues/1303&#8288;</code>).
</p>
</li></ul>



<h3>Sample of Usage</h3>

<pre>
# Example 1.
# Mollweide (referenc 1 page 54) is an equal-area projection that works well
# for whole-globe views.
mapPlot(coastlineWorld, projection="+proj=moll", col="gray")
mtext("Mollweide", adj=1)

# Example 2.
# Note that filling is not employed (`col` is not
# given) when the prime meridian is shifted, because
# this causes a problem with Antarctica
cl180 &lt;- coastlineCut(coastlineWorld, lon_0=-180)
mapPlot(cl180, projection="+proj=moll +lon_0=-180")
mtext("Mollweide with coastlineCut", adj=1)

# Example 3.
# Orthographic projections resemble a globe, making them attractive for
# non-technical use, but they are neither conformal nor equal-area, so they
# are somewhat limited for serious use on large scales.  See Section 20 of
# reference 1. Note that filling is not employed because it causes a problem with
# Antarctica.
if (utils::packageVersion("sf") != "0.9.8") {
    # sf version 0.9-8 has a problem with this projection
    par(mar=c(3, 3, 1, 1))
    mapPlot(coastlineWorld, projection="+proj=ortho +lon_0=-180")
    mtext("Orthographic", adj=1)
}

# Example 4.
# The Lambert conformal conic projection is an equal-area projection
# recommended by reference 1, page 95, for regions of large east-west extent
# away from the equator, here illustrated for the USA and Canada.
par(mar=c(3, 3, 1, 1))
mapPlot(coastlineCut(coastlineWorld, -100),
    longitudelim=c(-130,-55), latitudelim=c(35, 60),
    projection="+proj=lcc +lat_0=30 +lat_1=60 +lon_0=-100", col="gray")
mtext("Lambert conformal", adj=1)

# Example 5.
# The stereographic projection (reference 1, page 120) in the standard
# form used NSIDC (National Snow and Ice Data Center) for the Arctic.
# (See "A Guide to NSIDC's Polar Stereographic Projection" at
# https://nsidc.org/data/user-resources/help-center.)
# Note how the latitude limit extends 20 degrees past the pole,
# symmetrically.
par(mar=c(3, 3, 1, 1))
mapPlot(coastlineWorld,
    longitudelim=c(-180, 180), latitudelim=c(70, 110),
    projection=sf::st_crs("EPSG:3413"), col="gray")
mtext("Stereographic", adj=1)

# Example 6.
# Spinning globe: create PNG files that can be assembled into a movie
if (utils::packageVersion("sf") != "0.9.8") {
    # sf version 0.9-8 has a problem with this projection
    png("globe-
    lons &lt;- seq(360, 0, -15)
    par(mar=rep(0, 4))
    for (i in seq_along(lons)) {
        p &lt;- paste("+proj=ortho +lat_0=30 +lon_0=", lons[i], sep="")
        if (i == 1) {
            mapPlot(coastlineCut(coastlineWorld, lons[i]), projection=p, col="gray")
            xlim &lt;- par("usr")[1:2]
            ylim &lt;- par("usr")[3:4]
        } else {
            mapPlot(coastlineCut(coastlineWorld, lons[i]), projection=p, col="gray",
                    xlim=xlim, ylim=ylim, xaxs="i", yaxs="i")
        }
    }
    dev.off()
}
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>


<ol>
<li><p> Snyder, John P., 1987.  Map Projections: A Working Manual.  USGS
Professional Paper: 1395
<code style="white-space: pre;">&#8288;https://pubs.er.usgs.gov/publication/pp1395&#8288;</code>
</p>
</li>
<li><p> Natural Resources Canada
<code style="white-space: pre;">&#8288;https://www.nrcan.gc.ca/earth-sciences/geography/topographic-information/maps/9805&#8288;</code>
</p>
</li>
<li><p> &quot;List of Map Projections.&quot; In Wikipedia, January 26, 2021.
<code style="white-space: pre;">&#8288;https://en.wikipedia.org/w/index.php?title=List_of_map_projections&#8288;</code>.
</p>
</li>
<li><p> PROJ contributors (2020).
&quot;PROJ Coordinate Transformation Software Library.&quot;
Open Source Geospatial Foundation, n.d.
<code style="white-space: pre;">&#8288;https://proj.org&#8288;</code>.
</p>
</li>
<li><p> Bivand, Roger (2020) Why have CRS, projections and transformations changed?
</p>
</li>
<li><p> A gallery of map plots is provided at
<code style="white-space: pre;">&#8288;https://dankelley.github.io/r/2020/08/02/oce-proj.html&#8288;</code>
</p>
</li>
<li><p> Snyder, John Parr.
Flattening the Earth: Two Thousand Years of Map Projections.
Chicago, IL: University of Chicago Press, 1993.
<code style="white-space: pre;">&#8288;https://press.uchicago.edu/ucp/books/book/chicago/F/bo3632853.html&#8288;</code>
</p>
</li>
<li><p> Šavrič, Bojan, Tom Patterson, and Bernhard Jenny.
&quot;The Equal Earth Map Projection.&quot;
International Journal of Geographical Information Science 33, no. 3 (March 4, 2019): 454-65.
<a href="https://doi.org/10.1080/13658816.2018.1504949">doi:10.1080/13658816.2018.1504949</a>
</p>
</li></ol>



<h3>See Also</h3>

<p>Points may be added to a map with <code><a href="#topic+mapPoints">mapPoints()</a></code>, lines with
<code><a href="#topic+mapLines">mapLines()</a></code>, text with <code><a href="#topic+mapText">mapText()</a></code>, polygons with
<code><a href="#topic+mapPolygon">mapPolygon()</a></code>, images with <code><a href="#topic+mapImage">mapImage()</a></code>, and scale bars
with <code><a href="#topic+mapScalebar">mapScalebar()</a></code>.  Points on a map may be determined with mouse
clicks using <code><a href="#topic+mapLocator">mapLocator()</a></code>.  Great circle paths can be calculated
with <code><a href="#topic+geodGc">geodGc()</a></code>.  See reference 8 for a demonstration of the available map
projections (with graphs).
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># NOTE: the map-projection vignette has many more examples.
library(oce)
data(coastlineWorld)
# Demonstrate a high-latitude view using a built-in "CRS" value that is used
# by the National Snow and Ice Data Center (NSIDC) for representing
# the northern-hemisphere ice zone.  The view is meant to mimic the figure
# at the top of the document entitled "A Guide to NSIDC's Polar Stereographic
# Projection" at https://nsidc.org/data/user-resources/help-center, with the
# box indicating the region of the NSIDC grid.
library(oce)
data(coastlineWorld)
projection &lt;- sf::st_crs("EPSG:3413")
cat(projection$proj4string, "\n") # see the projection details
par(mar = c(2, 2, 1, 1)) # tighten margins
mapPlot(coastlineWorld,
    projection = projection,
    col = gray(0.9), geographical = 4,
    longitudelim = c(-180, 180), latitudelim = c(10, 90)
)
# Coordinates of box from Table 6 of the NSIDC document
box &lt;- cbind(
    -360 + c(168.35, 102.34, 350.3, 279.26, 168.35),
    c(30.98, 31.37, 34.35, 33.92, 30.98)
)
mapLines(box[, 1], box[, 2], lwd = 2)

</code></pre>

<hr>
<h2 id='mapPoints'>Add Points to a Map</h2><span id='topic+mapPoints'></span>

<h3>Description</h3>

<p>Plot points on an existing map, by analogy to <code><a href="graphics.html#topic+points">points()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapPoints(longitude, latitude, debug = getOption("oceDebug"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapPoints_+3A_longitude">longitude</code></td>
<td>
<p>Longitudes of points to be plotted, or an object from which
longitude and latitude can be inferred in which case the following two
arguments are ignored.  This objects that are possible include those of type
<code>coastline</code>.</p>
</td></tr>
<tr><td><code id="mapPoints_+3A_latitude">latitude</code></td>
<td>
<p>numeric vector of latitudes of points to be plotted.</p>
</td></tr>
<tr><td><code id="mapPoints_+3A_debug">debug</code></td>
<td>
<p>A flag that turns on debugging.  Set to 1 to get a moderate amount
of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="mapPoints_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="graphics.html#topic+points">points()</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(coastlineWorld)
mapPlot(coastlineWorld,
    longitudelim = c(-80, 0), latitudelim = c(20, 50),
    col = "lightgray", projection = "+proj=laea +lon_0=-35"
)
data(section)
mapPoints(section)


</code></pre>

<hr>
<h2 id='mapPolygon'>Add a Polygon to a Map</h2><span id='topic+mapPolygon'></span>

<h3>Description</h3>

<p><code>mapPolygon</code> adds a polygon to an existing map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapPolygon(
  longitude,
  latitude,
  density = NULL,
  angle = 45,
  border = NULL,
  col = NA,
  lty = par("lty"),
  ...,
  fillOddEven = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapPolygon_+3A_longitude">longitude</code></td>
<td>
<p>numeric vector of longitudes of points defining the polygon,
to be plotted, or an object from
which both longitude and latitude can be inferred (e.g. a coastline file, or
the return value from <code><a href="#topic+mapLocator">mapLocator()</a></code>), in which case the <code>latitude</code>
argument are ignored.</p>
</td></tr>
<tr><td><code id="mapPolygon_+3A_latitude">latitude</code></td>
<td>
<p>numeric vector of latitudes of points to be plotted (ignored
if both longitude and latitude can be determined from the first argument).</p>
</td></tr>
<tr><td><code id="mapPolygon_+3A_density">density</code>, <code id="mapPolygon_+3A_angle">angle</code>, <code id="mapPolygon_+3A_border">border</code>, <code id="mapPolygon_+3A_col">col</code>, <code id="mapPolygon_+3A_lty">lty</code>, <code id="mapPolygon_+3A_...">...</code>, <code id="mapPolygon_+3A_filloddeven">fillOddEven</code></td>
<td>
<p>handled as
<code><a href="graphics.html#topic+polygon">polygon()</a></code> handles the same arguments.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(coastlineWorld)
data(topoWorld)

# Bathymetry near southeastern Canada
par(mfrow = c(1, 1), mar = c(2, 2, 1, 1))
cm &lt;- colormap(zlim = c(-5000, 0), col = oceColorsGebco)
drawPalette(colormap = cm)
lonlim &lt;- c(-60, -50)
latlim &lt;- c(40, 60)
mapPlot(coastlineWorld,
    longitudelim = lonlim,
    latitudelim = latlim, projection = "+proj=merc", grid = FALSE
)
mapImage(topoWorld, colormap = cm)
mapPolygon(coastlineWorld[["longitude"]], coastlineWorld[["latitude"]], col = "lightgray")


</code></pre>

<hr>
<h2 id='mapScalebar'>Add a Scalebar to a Map</h2><span id='topic+mapScalebar'></span>

<h3>Description</h3>

<p>Draw a scalebar on a map created by <code><a href="#topic+mapPlot">mapPlot()</a></code> or otherwise.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapScalebar(
  x,
  y = NULL,
  length,
  lwd = 1.5 * par("lwd"),
  cex = par("cex"),
  col = "black"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapScalebar_+3A_x">x</code>, <code id="mapScalebar_+3A_y">y</code></td>
<td>
<p>position of the scalebar.  Eventually this may be similar to
the corresponding arguments in <code><a href="graphics.html#topic+legend">legend()</a></code>, but at the moment
<code>y</code> must be <code>NULL</code> and <code>x</code> must be <code>"topleft"</code> or <code>"topright"</code>.</p>
</td></tr>
<tr><td><code id="mapScalebar_+3A_length">length</code></td>
<td>
<p>the distance to indicate, in kilometres.  If not provided, a
reasonable choice is made, based on the existing plot.</p>
</td></tr>
<tr><td><code id="mapScalebar_+3A_lwd">lwd</code></td>
<td>
<p>line width of the scalebar.</p>
</td></tr>
<tr><td><code id="mapScalebar_+3A_cex">cex</code></td>
<td>
<p>character expansion factor for the scalebar text.</p>
</td></tr>
<tr><td><code id="mapScalebar_+3A_col">col</code></td>
<td>
<p>color of the scalebar.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The scale is appropriate to the centre of the plot, and will become
increasingly inaccurate away from that spot, with the error depending on
the projection and the fraction of the earth that is shown.
</p>
<p>Until December 2020, it was required that the map had been drawn by <code><a href="#topic+mapPlot">mapPlot()</a></code>,
but now it can be any diagram showing longitude and latitude in degrees.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(coastlineWorld)
# Arctic Ocean
par(mar = c(2.5, 2.5, 1, 1))
mapPlot(coastlineWorld,
    latitudelim = c(60, 120), longitudelim = c(-130, -50),
    col = "lightgray", projection = "+proj=stere +lat_0=90"
)
mapScalebar()


</code></pre>

<hr>
<h2 id='mapText'>Add Text to a Map</h2><span id='topic+mapText'></span>

<h3>Description</h3>

<p>Plot text on an existing map, by analogy to <code><a href="graphics.html#topic+text">text()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapText(longitude, latitude, labels, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapText_+3A_longitude">longitude</code></td>
<td>
<p>numeric vector of longitudes of text to be plotted.</p>
</td></tr>
<tr><td><code id="mapText_+3A_latitude">latitude</code></td>
<td>
<p>numeric vector of latitudes of text to be plotted.</p>
</td></tr>
<tr><td><code id="mapText_+3A_labels">labels</code></td>
<td>
<p>vector of labels of text to be plotted.</p>
</td></tr>
<tr><td><code id="mapText_+3A_...">...</code></td>
<td>
<p>optional arguments passed to <code><a href="graphics.html#topic+text">text()</a></code>, e.g. <code>adj</code>,
<code>pos</code>, etc.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(coastlineWorld)
longitude &lt;- coastlineWorld[["longitude"]]
latitude &lt;- coastlineWorld[["latitude"]]
mapPlot(longitude, latitude,
    type = "l", grid = 5,
    longitudelim = c(-70, -50), latitudelim = c(45, 50),
    projection = "+proj=merc"
)
lon &lt;- -63.5744 # Halifax
lat &lt;- 44.6479
mapPoints(lon, lat, pch = 20, col = "red")
mapText(lon, lat, "Halifax", col = "red", pos = 1, offset = 1)


</code></pre>

<hr>
<h2 id='mapTissot'>Add Tissot Indicatrices to a Map</h2><span id='topic+mapTissot'></span>

<h3>Description</h3>

<p>Plot ellipses at grid intersection points, as a method for
indicating the distortion inherent in the projection, somewhat
analogous to the scheme used in reference 1.
(Each ellipse is drawn with 64 segments.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapTissot(grid = rep(15, 2), scale = 0.2, crosshairs = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapTissot_+3A_grid">grid</code></td>
<td>
<p>numeric vector of length 2, specifying the increment in
longitude and latitude for the grid. Indicatrices are drawn at e.g.
longitudes <code>seq(-180, 180, grid[1])</code>.</p>
</td></tr>
<tr><td><code id="mapTissot_+3A_scale">scale</code></td>
<td>
<p>numerical scale factor for ellipses. This is multiplied by
<code>min(grid)</code> and the result is the radius of the circle on the
earth, in latitude degrees.</p>
</td></tr>
<tr><td><code id="mapTissot_+3A_crosshairs">crosshairs</code></td>
<td>
<p>logical value indicating whether to draw constant-latitude
and constant-longitude crosshairs within the ellipses.  (These are drawn
with 10 line segments each.) This can be helpful in cases where it is
not desired to use <code><a href="#topic+mapGrid">mapGrid()</a></code> to draw the longitude/latitude
grid.</p>
</td></tr>
<tr><td><code id="mapTissot_+3A_...">...</code></td>
<td>
<p>extra arguments passed to plotting functions, e.g.
<code>col="red"</code> yields red indicatrices.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Snyder, John P., 1987.  Map Projections: A Working Manual.  USGS
Professional Paper: 1395
</p>
</li></ol>



<h3>See Also</h3>

<p>A map must first have been created with <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(coastlineWorld)
par(mfrow = c(1, 1), mar = c(2, 2, 1, 1))
p &lt;- "+proj=aea +lat_1=10 +lat_2=60 +lon_0=-45"
mapPlot(coastlineWorld,
    projection = p, col = "gray",
    longitudelim = c(-90, 0), latitudelim = c(0, 50)
)
mapTissot(c(15, 15), col = "red")


</code></pre>

<hr>
<h2 id='matchBytes'>Locate Byte Sequences in a Raw Vector</h2><span id='topic+matchBytes'></span>

<h3>Description</h3>

<p>Find spots in a raw vector that match a given byte sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matchBytes(input, b1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matchBytes_+3A_input">input</code></td>
<td>
<p>a vector of raw (byte) values.</p>
</td></tr>
<tr><td><code id="matchBytes_+3A_b1">b1</code></td>
<td>
<p>a vector of bytes to match (must be of length 2 or 3 at present;
for 1-byte, use <code><a href="base.html#topic+which">which()</a></code>).</p>
</td></tr>
<tr><td><code id="matchBytes_+3A_...">...</code></td>
<td>
<p>additional bytes to match for (up to 2 permitted)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of the indices of <code>input</code> that match the start of the
<code>bytes</code> sequence (see example).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>buf &lt;- as.raw(c(0xa5, 0x11, 0xaa, 0xa5, 0x11, 0x00))
match &lt;- matchBytes(buf, 0xa5, 0x11)
print(buf)
print(match)
</code></pre>

<hr>
<h2 id='matrixShiftLongitude'>Rearrange Areal Matrix so Greenwich is Near the Centre</h2><span id='topic+matrixShiftLongitude'></span>

<h3>Description</h3>

<p>Sometimes datasets are provided in matrix form, with first
index corresponding to longitudes ranging from 0 to 360.
<code>matrixShiftLongitude</code> cuts such matrices at
longitude=180, and swaps the pieces so that the dateline
is at the left of the matrix, not in the middle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrixShiftLongitude(m, longitude)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matrixShiftLongitude_+3A_m">m</code></td>
<td>
<p>The matrix to be modified.</p>
</td></tr>
<tr><td><code id="matrixShiftLongitude_+3A_longitude">longitude</code></td>
<td>
<p>A vector containing the longitude in the 0-360 convention. If missing,
this is constructed to range from 0 to 360, with as many elements as the first index of <code>m</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing <code>m</code> and <code>longitude</code>, both rearranged as appropriate.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+shiftLongitude">shiftLongitude()</a></code> and <code><a href="#topic+standardizeLongitude">standardizeLongitude()</a></code>.
</p>

<hr>
<h2 id='matrixSmooth'>Smooth a Matrix</h2><span id='topic+matrixSmooth'></span>

<h3>Description</h3>

<p>The values on the edge of the matrix are unaltered.  For interior points,
the result is defined in terms in terms of the original as follows.
<code class="reqn">r_{i,j} = (2 m_{i,j} + m_{i-1,j} + m_{i+1,j} + m_{i,j-1} +
m_{i,j+1})/6</code>.  Note that missing values propagate to neighbours.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrixSmooth(m, passes = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matrixSmooth_+3A_m">m</code></td>
<td>
<p>a matrix to be smoothed.</p>
</td></tr>
<tr><td><code id="matrixSmooth_+3A_passes">passes</code></td>
<td>
<p>an integer specifying the number of times the smoothing is to
be applied.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A smoothed matrix.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
opar &lt;- par(no.readonly = TRUE)
m &lt;- matrix(rep(seq(0, 1, length.out = 5), 5), nrow = 5, byrow = TRUE)
m[3, 3] &lt;- 2
m1 &lt;- matrixSmooth(m)
m2 &lt;- matrixSmooth(m1)
m3 &lt;- matrixSmooth(m2)
par(mfrow = c(2, 2))
image(m, col = rainbow(100), zlim = c(0, 4), main = "original image")
image(m1, col = rainbow(100), zlim = c(0, 4), main = "smoothed 1 time")
image(m2, col = rainbow(100), zlim = c(0, 4), main = "smoothed 2 times")
image(m3, col = rainbow(100), zlim = c(0, 4), main = "smoothed 3 times")
par(opar)
</code></pre>

<hr>
<h2 id='met'>Sample met Data</h2><span id='topic+met'></span>

<h3>Description</h3>

<p>This is sample <a href="#topic+met-class">met</a> object containing data for Halifax, Nova Scotia,
during September of 2003 (the period during which Hurricane Juan struck the
city).
</p>


<h3>Details</h3>

<p>The data file was downloaded
</p>
<div class="sourceCode R"><pre>metFile &lt;- download.met(id=6358, year=2003, month=9, destdir=".", type="xml")
</pre></div>
<p>Note that using <code><a href="#topic+download.met">download.met()</a></code> avoids having to navigate the
the awkward Environment Canada website, but it imposes the burden
of having to know the station ID number.  With the data in-hand,
the object was then created (and its timezone adjusted) with
</p>
<div class="sourceCode R"><pre>met &lt;- read.met(metFile)
met &lt;- oceSetData(met, "time", met[["time"]]+4*3600,
                 note="add 4h to local time to get UTC time")
</pre></div>
<p><em>Historical note.</em> The <code>data(met)</code> object was changed on October 19,
2019, based on the data provided by Environment
Canada at that time. The previous version of <code>data(met)</code>,
created in 2017, had been based on a data format that
Environment Canada no longer provided in 2019.  See the
notes on the <code>type</code> argument of <code><a href="#topic+read.met">read.met()</a></code> for more on this
shift in the Environment Canada data format.
</p>


<h3>Source</h3>

<p>Environment Canada website on October 19, 2019.
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to met data: 
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic+as.met">as.met</a>()</code>,
<code><a href="#topic+download.met">download.met</a>()</code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+read.met">read.met</a>()</code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+summary+2Cmet-method">summary,met-method</a></code>
</p>

<hr>
<h2 id='met-class'>Class to Store Meteorological Data</h2><span id='topic+met-class'></span>

<h3>Description</h3>

<p>This class stores meteorological data. For objects created with
<code><a href="#topic+read.met">read.met()</a></code>, the <code>data</code> slot will contain all the columns
within the original file (with some guesses as to units) in addition to
several calculated quantities such as <code>u</code> and <code>v</code>, which are
velocities in m/s (not the km/h stored in typical data files), and which
obey the oceanographic convention that <code>u&gt;0</code> is a wind towards the
east.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>met</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>met</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>met</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+met-class">met</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+met-class">met</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to met data: 
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic+as.met">as.met</a>()</code>,
<code><a href="#topic+download.met">download.met</a>()</code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+read.met">read.met</a>()</code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+summary+2Cmet-method">summary,met-method</a></code>
</p>

<hr>
<h2 id='metNames2oceNames'>Convert met Data Name to oce Name</h2><span id='topic+metNames2oceNames'></span>

<h3>Description</h3>

<p>Interoperability between oce functions requires that standardized data names
be used, e.g. <code>"temperature"</code> for in-situ temperature. Very few
data-file headers name the temperature column in exactly that way, however,
and this function is provided to try to guess the names. The task is complicated
by the fact that Environment Canada seems to change the names of the columns,
e.g. sometimes a symbol is used for the degree sign, other times not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metNames2oceNames(names, scheme)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metNames2oceNames_+3A_names">names</code></td>
<td>
<p>a vector of character strings with original names</p>
</td></tr>
<tr><td><code id="metNames2oceNames_+3A_scheme">scheme</code></td>
<td>
<p>an optional indication of the scheme that is employed. This may
be <code>"ODF"</code>, in which case <code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames()</a></code> is used,
or <code>"met"</code>, in which case some tentative code for met files is used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Several quantities in the returned object differ from their values in the source
file. For example, speed is converted from km/h to m/s, and angles are converted
from tens of degrees to degrees. Also, some items are created from scratch, e.g.
<code>u</code> and <code>v</code>, the eastward and northward velocity, are computed from speed
and direction. (Note that e.g. u is positive if the wind blows to the east; the
data are thus in the normal Physics convention.)
</p>


<h3>Value</h3>

<p>Vector of strings for the decoded names. If an unknown scheme is provided,
this will just be <code>names</code>.
</p>

<hr>
<h2 id='moonAngle'>Lunar Angle as Function of Space and Time</h2><span id='topic+moonAngle'></span>

<h3>Description</h3>

<p>The calculations are based on formulae provided by Meeus (1982), primarily
in chapters 6, 18, and 30.  The first step is to compute sidereal time as
formulated in Meeus (1982) chapter 7, which in turn uses Julian day computed
according to as formulae in Meeus (1982) chapter 3.  Using these quantities,
formulae in Meeus (1982) chapter 30 are then used to compute geocentric
longitude (<code class="reqn">lambda</code>, in the Meeus notation), geocentric latitude
(<code class="reqn">beta</code>), and parallax.  Then the <code>obliquity</code> of the ecliptic
is computed with Meeus (1982) equation 18.4.  Equatorial coordinates (right
ascension and declination) are computed with equations 8.3 and 8.4 from
Meeus (1982), using <code><a href="#topic+eclipticalToEquatorial">eclipticalToEquatorial()</a></code>.  The hour angle
(<code class="reqn">H</code>) is computed using the unnumbered equation preceding Meeus's
(1982) equation 8.1.  Finally, Meeus (1982) equations 8.5 and 8.6 are used
to calculate the local azimuth and altitude of the moon, using
<code><a href="#topic+equatorialToLocalHorizontal">equatorialToLocalHorizontal()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moonAngle(t, longitude = 0, latitude = 0, useRefraction = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="moonAngle_+3A_t">t</code></td>
<td>
<p>time, a POSIXt object (converted to timezone <code>"UTC"</code>,
if it is not already in that timezone), a character or numeric value that
corresponds to such a time.</p>
</td></tr>
<tr><td><code id="moonAngle_+3A_longitude">longitude</code></td>
<td>
<p>observer longitude in degrees east</p>
</td></tr>
<tr><td><code id="moonAngle_+3A_latitude">latitude</code></td>
<td>
<p>observer latitude in degrees north</p>
</td></tr>
<tr><td><code id="moonAngle_+3A_userefraction">useRefraction</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to apply a correction for
atmospheric refraction.  (Ignored at present.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following.
</p>

<ul>
<li> <p><code>time</code>
</p>
</li>
<li> <p><code>azimuth</code> moon azimuth, in degrees eastward of north, from 0 to 360.
Note: this is not the convention used by Meeus, who uses degrees westward of
South. Here, the convention is chosen to more closely match the expectation
of oceanographers.
</p>
</li>
<li> <p><code>altitude</code> moon altitude, in degrees from -90 to 90.
</p>
</li>
<li> <p><code>rightAscension</code> in' degrees.
</p>
</li>
<li> <p><code>declination</code> in degrees.
</p>
</li>
<li> <p><code>lambda</code> geocentric longitude, in degrees.
</p>
</li>
<li> <p><code>beta</code> geocentric latitude, in degrees.
</p>
</li>
<li> <p><code>diameter</code> lunar diameter, in degrees.
</p>
</li>
<li> <p><code>distance</code> earth-moon distance, in kilometers.
</p>
</li>
<li> <p><code>illuminatedFraction</code> fraction of moon's visible disk that is illuminated.
</p>
</li>
<li> <p><code>phase</code> phase of the moon, defined in equation 32.3 of Meeus (1982).
The fractional part of which is 0 for new moon, 1/4 for first
quarter, 1/2 for full moon, and 3/4 for last quarter.
</p>
</li></ul>

<p><img src="../help/figures/starCoords.png" width="400" alt="starCoords.png" />
</p>


<h3>Alternate formulations</h3>

<p>Formulae provide by Meeus (1982) are used
for all calculations here.  Meeus (1991) provides formulae that are similar,
but that differ in the 5th or 6th digits.  For example, the formula for
ephemeris time in Meeus (1991) differs from that in Meeus (1992) at the 5th
digit, and almost all of the approximately 200 coefficients in the relevant
formulae also differ in the 5th and 6th digits.  Discussion of the changing
formulations is best left to members of the astronomical community.  For the
present purpose, it may be sufficient to note that <code>moonAngle</code>, based
on Meeus (1982), reproduces the values provided in example 45.a of Meeus
(1991) to 4 significant digits, e.g. with all angles matching to under 2
minutes of arc.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, based on formulae in Meeus (1982).
</p>


<h3>References</h3>


<ul>
<li><p> Meeus, Jean. Astronomical Formulas for Calculators. Second Edition.
Richmond, Virginia, USA: Willmann-Bell, 1982.
</p>
</li>
<li><p> Meeus, Jean. Astronomical Algorithms. Second Edition.
Richmond, Virginia, USA: Willmann-Bell, 1991.
</p>
</li></ul>



<h3>See Also</h3>

<p>The equivalent function for the sun is <code><a href="#topic+sunAngle">sunAngle()</a></code>.
</p>
<p>Other things related to astronomy: 
<code><a href="#topic+angle2hms">angle2hms</a>()</code>,
<code><a href="#topic+eclipticalToEquatorial">eclipticalToEquatorial</a>()</code>,
<code><a href="#topic+equatorialToLocalHorizontal">equatorialToLocalHorizontal</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+siderealTime">siderealTime</a>()</code>,
<code><a href="#topic+sunAngle">sunAngle</a>()</code>,
<code><a href="#topic+sunDeclinationRightAscension">sunDeclinationRightAscension</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
par(mfrow = c(3, 2))
y &lt;- 2012
m &lt;- 4
days &lt;- 1:3
# Halifax sunrise/sunset (see e.g. https://www.timeanddate.com/worldclock)
rises &lt;- ISOdatetime(y, m, days, c(13, 15, 16), c(55, 04, 16), 0, tz = "UTC") + 3 * 3600 # ADT
sets &lt;- ISOdatetime(y, m, days, c(3, 4, 4), c(42, 15, 45), 0, tz = "UTC") + 3 * 3600
azrises &lt;- c(69, 75, 82)
azsets &lt;- c(293, 288, 281)
latitude &lt;- 44.65
longitude &lt;- -63.6
for (i in 1:3) {
    t &lt;- ISOdatetime(y, m, days[i], 0, 0, 0, tz = "UTC") + seq(0, 24 * 3600, 3600 / 4)
    ma &lt;- moonAngle(t, longitude, latitude)
    oce.plot.ts(t, ma$altitude, type = "l", mar = c(2, 3, 1, 1), cex = 1 / 2, ylab = "Altitude")
    abline(h = 0)
    points(rises[i], 0, col = "red", pch = 3, lwd = 2, cex = 1.5)
    points(sets[i], 0, col = "blue", pch = 3, lwd = 2, cex = 1.5)
    oce.plot.ts(t, ma$azimuth, type = "l", mar = c(2, 3, 1, 1), cex = 1 / 2, ylab = "Azimuth")
    points(rises[i], -180 + azrises[i], col = "red", pch = 3, lwd = 2, cex = 1.5)
    points(sets[i], -180 + azsets[i], col = "blue", pch = 3, lwd = 2, cex = 1.5)
}

</code></pre>

<hr>
<h2 id='numberAsHMS'>Convert a Numeric Time to Hour, Minute, and Second</h2><span id='topic+numberAsHMS'></span>

<h3>Description</h3>

<p>Convert a Numeric Time to Hour, Minute, and Second
</p>


<h3>Usage</h3>

<pre><code class='language-R'>numberAsHMS(t, default = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="numberAsHMS_+3A_t">t</code></td>
<td>
<p>a vector of factors or character strings, in the format 1200 for
12:00, 0900 for 09:00, etc.</p>
</td></tr>
<tr><td><code id="numberAsHMS_+3A_default">default</code></td>
<td>
<p>value to be used for the returned hour, minute and second if
there is something wrong with the input value (e.g. its length exceeds 4
characters, or it contains non-numeric characters)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing <code>hour</code>, <code>minute</code>, and <code>second</code>, the
last of which is always zero.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to time: 
<code><a href="#topic+ctimeToSeconds">ctimeToSeconds</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+numberAsPOSIXct">numberAsPOSIXct</a>()</code>,
<code><a href="#topic+secondsToCtime">secondsToCtime</a>()</code>,
<code><a href="#topic+unabbreviateYear">unabbreviateYear</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>t &lt;- c("0900", "1234")
numberAsHMS(t)

</code></pre>

<hr>
<h2 id='numberAsPOSIXct'>Convert a Numeric Time to a POSIXct Time</h2><span id='topic+numberAsPOSIXct'></span>

<h3>Description</h3>

<p>This converts numerical values into POSIXct times.  There are many
schemes for doing this, with the <code>type</code> parameter being used
to select between them.  See &ldquo;Details&rdquo; for a listing, broken
down by scheme.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>numberAsPOSIXct(t, type = "unix", tz = "UTC", leap = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="numberAsPOSIXct_+3A_t">t</code></td>
<td>
<p>an integer corresponding to a time, in a way that depends on
<code>type</code>.</p>
</td></tr>
<tr><td><code id="numberAsPOSIXct_+3A_type">type</code></td>
<td>
<p>character value indicating the time type.  The permitted values
are <code>"argo"</code>, <code>"epic"</code>, <code>"excel"</code>, <code>"gps"</code>, <code>"matlab"</code>, <code>"ncep1"</code>, <code>"ncep2"</code>,
<code>"sas"</code>, <code>"spss"</code>, <code>"unix"</code>, and <code>"yearday"</code>, the first of these being
the default.</p>
</td></tr>
<tr><td><code id="numberAsPOSIXct_+3A_tz">tz</code></td>
<td>
<p>a string indicating the time zone, by default <code>"UTC"</code>.</p>
</td></tr>
<tr><td><code id="numberAsPOSIXct_+3A_leap">leap</code></td>
<td>
<p>a logical value, TRUE by default, that applies only
if <code>type</code> is <code>"gps"</code>.  If <code>leap</code> is TRUE, then the built-in
dataset named <a href="base.html#topic+.leap.seconds">.leap.seconds</a> is consulted to find
of the number of leap seconds between 1980 (when the
GPS program started) and the time computed from the other
parameters, and the return value is decreased accordingly
(see Example 3).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The possible choices for <code>type</code> are as listed below.
</p>

<ul>
<li> <p><code>"unix"</code> handles Unix times, measured in seconds since the start
of the year 1970.
</p>
</li>
<li> <p><code>"matlab"</code> handles Matlab times, measured in days since what
MathWorks (reference 1) calls &ldquo;January 0, 0000&rdquo; (i.e.  <code>ISOdatetime(0, 1, 1, 0,
0, 0)</code> in R notation).
</p>
</li>
<li> <p><code>"gps"</code> handles the Global Positioning System convention. The
scheme is complicated, owing to hardware limitations of
GPS satellites.  As illustrated in Example 3, <code>t</code> may be
a matrix with either 2 or 3 columns. In the 2-column format, the
first column holds the number of weeks after 1999-08-22, modulo
1024 (approximately 19.6 years), and the second column (here
and also in the 3-column format) holds the number of seconds
in the referenced week, with leap seconds being handled with
the <code>leap</code> parameter. The modulo calculation is required
because GPS satellites dedicate only 10 bits to the
week number. The resultant ambiguity (e.g. a rollover in
2019-04-07) is addressed in the 3-column format, in which the
last column holds the number of 1024-week rollover events
since 1980-01-06. Users should set this column to 0 for times
prior to 1999-08-22, to 1 for later times prior
to 2019-04-07, to 2 for later times prior to 2038-11-21,
etc. However, there will be an exception to this rule,
when satellites start dedicating 12 bits to the week value.
For such data, the third column will need to be 0 for all times
prior to 2137-01-06.
</p>
</li>
<li> <p><code>"argo"</code> handles Argo times, measured in days since the start of
the year 1900.
</p>
</li>
<li> <p><code>"excel"</code> handles Excel times, measured in days since the start of
the year 1900. (Note that excel incorrectly regards 1900 as a leap year,
so 1 day is subtracted from <code>t</code> unless the time is less than or equal
to 1900 Feb 28.  Note that NA is returned for the day 60, which is
what excel codes for &quot;Feb 29, 1900&quot;, the non-existing day that excel
accepts.
</p>
</li>
<li> <p><code>"ncep1"</code> handles NCEP times, measured in hours since the start
of the year 1800.
</p>
</li>
<li> <p><code>"ncep2"</code> handles NCEP times, measured in days since the start of
the year 1. (Note that, for reasons that are unknown at this time, a simple
R expression of this definition is out by two days compared with the UDUNITS
library, which is used by NCEP. Therefore, a two-day offset is applied. See
references 2 and 3.)
</p>
</li>
<li> <p><code>"sas"</code> handles SAS times, indicated by <code>type="sas"</code>, have
origin at the start of 1960.
</p>
</li>
<li> <p><code>"spss"</code> handles SPSS times, in seconds after 1582-10-14.
</p>
</li>
<li> <p><code>"yearday"</code> handles a convention in which <code>t</code> is a
two-column matrix, with the first column being the year, and the second the
yearday (starting at 1 for the first second of January 1, to match the
convention used by Sea-Bird CTD software).
</p>
</li>
<li> <p><code>"epic"</code> handles a convention used in the EPIC software library,
from the Pacific Marine Environmental Laboratory, in which <code>t</code> is a
two-column matrix, with the first column being the julian Day (as defined in
<code><a href="#topic+julianDay">julianDay()</a></code>, for example), and with the second column being the
millisecond within that day. See reference 4.
</p>
</li>
<li> <p><code>"vms"</code> handles a convention used in the VMS operating system and
for Modified Julian Day, in which <code>t</code> is the number of seconds
past 1859-11-17T00:00:00 UTC. See reference 5.
</p>
</li></ul>



<h3>Value</h3>

<p>A <code><a href="base.html#topic+POSIXct">POSIXct()</a></code> time vector.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Matlab times:
<code style="white-space: pre;">&#8288;https://www.mathworks.com/help/matlab/ref/datenum.html&#8288;</code>
</p>
</li>
<li><p> NCEP times: <code style="white-space: pre;">&#8288;https://psl.noaa.gov/data/gridded/faq.html&#8288;</code>
</p>
</li>
<li><p> Problem with NCEP times:
<code style="white-space: pre;">&#8288;https://github.com/dankelley/oce/issues/738&#8288;</code>
</p>
</li>
<li><p> EPIC times: software and manuals at <code style="white-space: pre;">&#8288;https://www.pmel.noaa.gov/epic/download/index.html#epslib&#8288;</code>;
see also Denbo, Donald W., and Nancy N. Soreide. &ldquo;EPIC.&rdquo; Oceanography 9 (1996).
<a href="https://doi.org/10.5670/oceanog.1996.10">doi:10.5670/oceanog.1996.10</a>
</p>
</li>
<li><p> VMS times: https://en.wikipedia.org/wiki/Epoch_(computing)
</p>
</li>
<li><p> GPS times: https://www.labsat.co.uk/index.php/en/gps-time-calculator
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to time: 
<code><a href="#topic+ctimeToSeconds">ctimeToSeconds</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+numberAsHMS">numberAsHMS</a>()</code>,
<code><a href="#topic+secondsToCtime">secondsToCtime</a>()</code>,
<code><a href="#topic+unabbreviateYear">unabbreviateYear</a>()</code>
</p>
<p>Other things related to time: 
<code><a href="#topic+ctimeToSeconds">ctimeToSeconds</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+numberAsHMS">numberAsHMS</a>()</code>,
<code><a href="#topic+secondsToCtime">secondsToCtime</a>()</code>,
<code><a href="#topic+unabbreviateYear">unabbreviateYear</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1. default (unix)
numberAsPOSIXct(0)

# Example 2. Matlab
numberAsPOSIXct(1, type = "matlab")

# Example 3. GPS with default week rollover or with no rollover (Canada Day, year 2010)
numberAsPOSIXct(cbind(566, 345615), type = "gps")
numberAsPOSIXct(cbind(566, 345615, 1), type = "gps")
numberAsPOSIXct(cbind(1024 + 566, 345615, 0), type = "gps")
# Show how to deal with leap seconds (15 of them, in this case)
sum(as.POSIXct("1980-01-01") &lt; .leap.seconds &amp; .leap.seconds &lt;= as.POSIXct("2010-07-01"))
-15 + numberAsPOSIXct(cbind(1024 + 566, 345615, 0), type = "gps", leap = FALSE)

# Example 4. yearday
numberAsPOSIXct(cbind(2013, 1), type = "yearday") # start of 2013

# Example 5. Epic time, one hour into Canada Day of year 2018. In computing the
# Julian day, note that this starts at noon.
jd &lt;- julianDay(as.POSIXct("2018-07-01 12:00:00", tz = "UTC"))
numberAsPOSIXct(cbind(jd, 1e3 * 1 * 3600), type = "epic", tz = "UTC")

# Example 6. Julian day, note that this starts at noon.
jd &lt;- julianDay(as.POSIXct("2018-07-01 12:00:00", tz = "UTC"))
numberAsPOSIXct(cbind(jd, 1e3 * 1 * 3600), type = "epic", tz = "UTC")

</code></pre>

<hr>
<h2 id='oce-class'>Base Class for oce Objects</h2><span id='topic+oce-class'></span>

<h3>Description</h3>

<p>This is mainly used within oce to create sub-classes, although
users can use <code>new("oce")</code> to create a blank <code>oce</code>
object, if desired.
</p>


<h3>Slots</h3>


<dl>
<dt><code>metadata</code></dt><dd><p>A list containing information about the data. The
contents vary across sub-classes, e.g. an <a href="#topic+adp-class">adp</a>
object has information about beam patterns, which obviously would
not make sense for a <a href="#topic+ctd-class">ctd</a> object In addition,
all classes have items named <code>units</code> and <code>flags</code>, used
to store information on the units of the data, and the data quality.</p>
</dd>
<dt><code>data</code></dt><dd><p>A list containing the data.</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>A list containing time-stamped processing steps,
typically stored in the object by oce functions.</p>
</dd>
</dl>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>str(new("oce"))

</code></pre>

<hr>
<h2 id='oce-deprecated'>Deprecated and Defunct Elements of the oce Package</h2><span id='topic+oce-deprecated'></span><span id='topic+oce-defunct'></span>

<h3>Description</h3>

<p>Certain functions and function arguments are still provided for
compatibility with older versions of <a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a>, but will be removed soon.
The <a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a> scheme for removing functions is similar to that used by
<code>Bioconductor</code>: items are marked as &quot;deprecated&quot; in one release, marked as
&quot;defunct&quot; in the next, and removed in the next after that. This goal is to provide a
gentle migration path for users who keep their packages reasonably
up-to-date.
</p>


<h3>Details</h3>

<p>The following are marked &quot;deprecated&quot; in the present CRAN release of <a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a>.
Please use the replacement functions as listed below. The upcoming CRAN
release of <a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a> will mark these as &quot;defunct&quot;, which is the
last step before outright removal.
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Deprecated</strong>            </td><td style="text-align: left;"> <strong>Replacement</strong>   </td><td style="text-align: left;"> <strong>Deprecated</strong>  </td><td style="text-align: left;"> <strong>Defunct</strong> </td><td style="text-align: left;"> <strong>Removed</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The following are marked &quot;defunct&quot;, so calling them in the
the present version produces an error message that hints at a replacement
function. Once a function is marked &quot;defunct&quot; on one CRAN release, it will
be slated for outright deletion in some subsequent release.
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Defunct</strong>         </td><td style="text-align: left;"> <strong>Replacement</strong>                </td><td style="text-align: left;"> <strong>Version</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The following functions were removed after having been marked as &quot;deprecated&quot;
in at least one CRAN release, and possibly as &quot;defunct&quot; in at least
one CRAN release.  (The version number in the table is the first
version to lack the named function.)
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Function</strong>        </td><td style="text-align: left;"> <strong>Replacement</strong>                </td><td style="text-align: left;"> <strong>Version</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>addColumn()</code>       </td><td style="text-align: left;"> <code><a href="#topic+oceSetData">oceSetData()</a></code>                 </td><td style="text-align: left;"> 1.1-2      </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ctdAddColumn()</code>    </td><td style="text-align: left;"> <code><a href="#topic+oceSetData">oceSetData()</a></code>                 </td><td style="text-align: left;"> 1.1-2      </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ctdUpdateHeader()</code> </td><td style="text-align: left;"> <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>             </td><td style="text-align: left;"> 1.1-2      </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>findInOrdered()</code>   </td><td style="text-align: left;"> <code><a href="base.html#topic+findInterval">findInterval()</a></code>               </td><td style="text-align: left;"> 1.1-2      </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>makeSection()</code>     </td><td style="text-align: left;"> <code><a href="#topic+as.section">as.section()</a></code>                 </td><td style="text-align: left;"> 0.9.24     </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>mapMeridians()</code>    </td><td style="text-align: left;"> <code><a href="#topic+mapGrid">mapGrid()</a></code>                    </td><td style="text-align: left;"> 1.1-2      </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>mapZones()</code>        </td><td style="text-align: left;"> <code><a href="#topic+mapGrid">mapGrid()</a></code>                    </td><td style="text-align: left;"> 1.1-2      </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>oce.as.POSIXlt()</code>  </td><td style="text-align: left;"> <code><a href="lubridate.html#topic+parse_date_time">lubridate::parse_date_time()</a></code> </td><td style="text-align: left;"> 1.1-2      </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>renameData()</code>      </td><td style="text-align: left;"> <code><a href="#topic+oceRenameData">oceRenameData()</a></code>              </td><td style="text-align: left;"> 1.7-9      </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>trimString()</code>      </td><td style="text-align: left;"> <code><a href="base.html#topic+trimws">trimws()</a></code>                     </td><td style="text-align: left;"> 1.8-2      </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Several <a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a> function arguments are considered &quot;deprecated&quot;, which
means they will be marked &quot;defunct&quot; in the next CRAN release. These are normally
listed in the help page for the function in question. A few that may be
of general interest are also listed below.
</p>

<ul>
<li><p> The <code>adorn</code> argument was still being checked for (in the dots argument)
until 2020 August 11.
</p>
</li>
<li><p> The <code>eos</code> argument of <code><a href="#topic+swN2">swN2()</a></code> was removed on 2019
April 11; for details, see the &ldquo;Deprecation Notation&rdquo; section
of the documentation for <code><a href="#topic+swN2">swN2()</a></code>.
</p>
</li>
<li><p> The <code>parameters</code> argument of <code><a href="#topic+plot+2Cctd-method">plot,ctd-method()</a></code>
was deprecated on 2016-12-30.  It was once used by
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method()</a></code> but has been ignored by that
function since February 2016.
</p>
</li>
<li><p> The <code>orientation</code> argument of <code><a href="#topic+plot+2Cctd-method">plot,ctd-method()</a></code>
was deprecated on 2016-12-30.  It was once used by
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method()</a></code> but has been ignored by that
function since February 2016.
</p>
</li></ul>

<p>Several &ldquo;oce&rdquo; function arguments are considered &quot;defunct&quot;, which
means they will be removed in the next CRAN release. They are as follows.
</p>

<ul>
<li><p> The <code>fill</code> argument of <code><a href="#topic+mapPlot">mapPlot()</a></code> was confusing
to users, so it was designated as deprecated in June 2016.
(The confusion stemmed from subtle differences between
<code><a href="graphics.html#topic+plot">plot()</a></code> and <code><a href="graphics.html#topic+polygon">polygon()</a></code>, and the problem is that
<code><a href="#topic+mapPlot">mapPlot()</a></code> can use either of these functions, according
to whether coastlines are to be filled.)
The functionality is preserved, in the <code>col</code> argument.
</p>
</li></ul>



<h3>See Also</h3>

<p>The &ldquo;Bioconductor&rdquo; scheme for removing functions is
described at
<code style="white-space: pre;">&#8288;https://www.bioconductor.org/developers/how-to/deprecation/&#8288;</code> and it is
extended here to function arguments.
</p>

<hr>
<h2 id='oce.as.raw'>Version of as.raw() That Clips Data</h2><span id='topic+oce.as.raw'></span>

<h3>Description</h3>

<p>A version of as.raw() that clips data to prevent warnings
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oce.as.raw(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oce.as.raw_+3A_x">x</code></td>
<td>
<p>values to be converted to raw</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Negative values are clipped to 0, while values above 255 are clipped to 255;
the result is passed to <code><a href="base.html#topic+as.raw">as.raw()</a></code> and returned.
</p>


<h3>Value</h3>

<p>Raw values corresponding to <code>x</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(-0.1, 0, 1, 255, 255.1)
data.frame(x, oce.as.raw(x))
</code></pre>

<hr>
<h2 id='oce.axis.POSIXct'>Oce Version of axis.POSIXct</h2><span id='topic+oce.axis.POSIXct'></span>

<h3>Description</h3>

<p>A specialized variant of <code><a href="graphics.html#topic+axis.POSIXct">axis.POSIXct()</a></code> that produces
results with less ambiguity in axis labels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oce.axis.POSIXct(
  side,
  x,
  at,
  tformat,
  labels = TRUE,
  drawTimeRange,
  abbreviateTimeRange = FALSE,
  drawFrequency = FALSE,
  cex.axis = par("cex.axis"),
  cex.lab = par("cex.lab"),
  cex.main = par("cex.main"),
  mar = par("mar"),
  mgp = par("mgp"),
  main = "",
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oce.axis.POSIXct_+3A_side">side</code></td>
<td>
<p>as for <code><a href="graphics.html#topic+axis.POSIXct">axis.POSIXct()</a></code>.</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_x">x</code></td>
<td>
<p>as for <code><a href="graphics.html#topic+axis.POSIXct">axis.POSIXct()</a></code>.</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_at">at</code></td>
<td>
<p>as for <code><a href="graphics.html#topic+axis.POSIXct">axis.POSIXct()</a></code>.</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_tformat">tformat</code></td>
<td>
<p>as <code>format</code> for <code><a href="graphics.html#topic+axis.POSIXct">axis.POSIXct()</a></code> for now, but
may eventually have new features for multiline labels, e.g. day on one line
and month on another.</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_labels">labels</code></td>
<td>
<p>as for <code><a href="graphics.html#topic+axis.POSIXct">axis.POSIXct()</a></code>.</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_drawtimerange">drawTimeRange</code></td>
<td>
<p>Optional indication of whether/how to draw the time range
in the margin on the side of the the plot opposite the time axis. If this is
not supplied, it defaults to the value returned by
<code><a href="base.html#topic+getOption">getOption</a></code><code>("oceDrawTimeRange")</code>, and if that option is not set,
it defaults to <code>TRUE</code>. No time range is drawn if <code>drawTimeRange</code> is <code>FALSE</code>.
If it is <code>TRUE</code>, the range will be shown. This range refers to
range of the x axis (not the data). The format of the elements of that range is set by
<code><a href="base.html#topic+getOption">getOption</a></code><code>("oceTimeFormat")</code> (or with the default value
of an empty string, if this option has not been set). The timezone will
be indicated if the time range is under a week.  For preliminary work, it makes
sense to use <code>drawTimeRange=TRUE</code>, but for published work it can be better
to drop this label and indicate something about the time in the figure caption.</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_abbreviatetimerange">abbreviateTimeRange</code></td>
<td>
<p>boolean, <code>TRUE</code> to abbreviate the second
number in the time range, e.g. dropping the year if it is the same in the
first number.</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_drawfrequency">drawFrequency</code></td>
<td>
<p>boolean, <code>TRUE</code> to show the frequency of sampling
in the data</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_cex.axis">cex.axis</code>, <code id="oce.axis.POSIXct_+3A_cex.lab">cex.lab</code>, <code id="oce.axis.POSIXct_+3A_cex.main">cex.main</code></td>
<td>
<p>character expansion factors for axis numbers, axis names and plot titles; see <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_mar">mar</code></td>
<td>
<p>value for <code>par(mar)</code> for axis</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_mgp">mgp</code></td>
<td>
<p>value for <code>par(mgp)</code> for axis</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_main">main</code></td>
<td>
<p>title of plot</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="oce.axis.POSIXct_+3A_...">...</code></td>
<td>
<p>as for <code><a href="graphics.html#topic+axis.POSIXct">axis.POSIXct()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The tick marks are set automatically based on examination of the time range on
the axis. The scheme was devised by constructing test cases with a typical plot
size and font size, and over a wide range of time scales. In some categories,
both small tick marks are interspersed between large ones.
</p>
<p>The user may set the format of axis numbers with the <code>tformat</code> argument.
If this is not supplied, the format is set based on the time span of the axis:
</p>

<ul>
<li><p> If this time span is less than a minute, the time axis labels are in
seconds (fractional seconds, if the interval is less than 2 seconds), with
leading zeros on small integers. (Fractional seconds are enabled with a trick:
the usual R format <code>"\%S"</code> is supplemented with a new format e.g.
<code>"\%.2S"</code>, meaning to use two digits after the decimal.)
</p>
</li>
<li><p> If the time span exceeds a minute but is less than 1.5 days, the label
format is <code>"\%H:\%M:\%S"</code>.
</p>
</li>
<li><p> If the time span exceeds 1.5 days but is less than 1 year, the format is
<code>"\%b \%d"</code> (e.g. Jul 15) and, again, the tick marks are set up for several
subcategories.
</p>
</li>
<li><p> If the time span exceeds a year, the format is <code>"\%Y"</code>, i.e. the year
is displayed with 4 digits.
</p>
</li></ul>

<p>It should be noted that this scheme differs from the R approach in several
ways. First, R writes day names for some time ranges, in a convention that is
seldom seen in the literature. Second, R will write nn:mm for both HH:MM and
MM:SS, an ambiguity that might confuse readers. Third, the use of both large
and small tick marks is not something that R does.
</p>
<p>Bear in mind that <code>tformat</code> may be set to alter the number format, but
that the tick mark scheme cannot (presently) be controlled.
</p>


<h3>Value</h3>

<p>A vector of times corresponding to axis ticks is returned silently.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>This is used mainly by <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code>.
</p>

<hr>
<h2 id='oce.contour'>Oce Variant of contour</h2><span id='topic+oce.contour'></span><span id='topic+oceContour'></span>

<h3>Description</h3>

<p>This provides something analogous to <code><a href="graphics.html#topic+contour">contour()</a></code>, but with the
ability to flip x and y.
Setting <code>revy=TRUE</code> can be helpful if the <code>y</code> data represent
pressure or depth below the surface.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oce.contour(
  x,
  y,
  z,
  revx = FALSE,
  revy = FALSE,
  add = FALSE,
  tformat,
  drawTimeRange = getOption("oceDrawTimeRange"),
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oce.contour_+3A_x">x</code></td>
<td>
<p>values for x grid.</p>
</td></tr>
<tr><td><code id="oce.contour_+3A_y">y</code></td>
<td>
<p>values for y grid.</p>
</td></tr>
<tr><td><code id="oce.contour_+3A_z">z</code></td>
<td>
<p>matrix for values to be contoured.  The first dimension of <code>z</code>
must equal the number of items in <code>x</code>, etc.</p>
</td></tr>
<tr><td><code id="oce.contour_+3A_revx">revx</code></td>
<td>
<p>set to <code>TRUE</code> to reverse the order in which the labels on
the x axis are drawn</p>
</td></tr>
<tr><td><code id="oce.contour_+3A_revy">revy</code></td>
<td>
<p>set to <code>TRUE</code> to reverse the order in which the labels on
the y axis are drawn</p>
</td></tr>
<tr><td><code id="oce.contour_+3A_add">add</code></td>
<td>
<p>logical value indicating whether the contours should be added to
a pre-existing plot.</p>
</td></tr>
<tr><td><code id="oce.contour_+3A_tformat">tformat</code></td>
<td>
<p>time format; if not supplied, a reasonable choice will be
made by <code><a href="#topic+oce.axis.POSIXct">oce.axis.POSIXct()</a></code>, which draws time axes.</p>
</td></tr>
<tr><td><code id="oce.contour_+3A_drawtimerange">drawTimeRange</code></td>
<td>
<p>logical, only used if the <code>x</code> axis is a time.  If
<code>TRUE</code>, then an indication of the time range of the data (not the axis)
is indicated at the top-left margin of the graph.  This is useful because
the labels on time axes only indicate hours if the range is less than a day,
etc.</p>
</td></tr>
<tr><td><code id="oce.contour_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging; set to 1 to information about
the processing.</p>
</td></tr>
<tr><td><code id="oce.contour_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(topoWorld)
# coastline now, and in last glacial maximum
lon &lt;- topoWorld[["longitude"]]
lat &lt;- topoWorld[["latitude"]]
z &lt;- topoWorld[["z"]]
oce.contour(lon, lat, z, levels = 0, drawlabels = FALSE)
oce.contour(lon, lat, z, levels = -130, drawlabels = FALSE, col = "blue", add = TRUE)
</code></pre>

<hr>
<h2 id='oce.grid'>Add a Grid to an Existing Oce Plot</h2><span id='topic+oce.grid'></span>

<h3>Description</h3>

<p>Add a Grid to an Existing Oce Plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oce.grid(xat, yat, col = "lightgray", lty = "dotted", lwd = par("lwd"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oce.grid_+3A_xat">xat</code></td>
<td>
<p>either a list of x values at which to draw the grid, or the return value from an oce plotting function</p>
</td></tr>
<tr><td><code id="oce.grid_+3A_yat">yat</code></td>
<td>
<p>a list of y values at which to plot the grid (ignored if <code>gx</code> was a return value from an oce plotting function)</p>
</td></tr>
<tr><td><code id="oce.grid_+3A_col">col</code></td>
<td>
<p>color of grid lines (see <code><a href="graphics.html#topic+par">par()</a></code>)</p>
</td></tr>
<tr><td><code id="oce.grid_+3A_lty">lty</code></td>
<td>
<p>type for grid lines (see <code><a href="graphics.html#topic+par">par()</a></code>)</p>
</td></tr>
<tr><td><code id="oce.grid_+3A_lwd">lwd</code></td>
<td>
<p>width for grid lines (see <code><a href="graphics.html#topic+par">par()</a></code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For plots not created by oce functions, or for missing <code>xat</code> and <code>yat</code>,
this is the same as a call to <code><a href="graphics.html#topic+grid">grid()</a></code> with missing <code>nx</code> and
<code>ny</code>. However, if <code>xat</code> is the return value from certain oce functions,
a more sophisticated grid is constructed. The problem with <code><a href="graphics.html#topic+grid">grid()</a></code> is
that it cannot handle axes with non-uniform grids, e.g. those with time axes
that span months of differing lengths.
</p>
<p>As of early February 2015, <code>oce.grid</code> handles <code>xat</code> produced as the
return value from the following functions: <code><a href="#topic+imagep">imagep()</a></code> and
<code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code>, <code><a href="#topic+plot+2Cadp-method">plot,adp-method()</a></code>,
<code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method()</a></code>, and <code><a href="#topic+plotTS">plotTS()</a></code>.
It makes no sense to try to use <code>oce.grid</code> for multipanel oce plots,
e.g. the default plot from <code><a href="#topic+plot+2Cadp-method">plot,adp-method()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
i &lt;- imagep(volcano)
oce.grid(i, lwd = 2)

data(sealevel)
i &lt;- oce.plot.ts(sealevel[["time"]], sealevel[["elevation"]])
oce.grid(i, col = "red")

data(ctd)
i &lt;- plotTS(ctd)
oce.grid(i, col = "red")

data(adp)
i &lt;- plot(adp, which = 1)
oce.grid(i, col = "gray", lty = 1)

data(echosounder)
i &lt;- plot(echosounder)
oce.grid(i, col = "pink", lty = 1)

</code></pre>

<hr>
<h2 id='oce.plot.ts'>Oce Variant of plot.ts</h2><span id='topic+oce.plot.ts'></span>

<h3>Description</h3>

<p>Plot a time-series, obeying the timezone and possibly drawing the range in
the top-left margin.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oce.plot.ts(
  x,
  y,
  type = "l",
  xlim,
  ylim,
  log = "",
  logStyle = "r",
  flipy = FALSE,
  xlab,
  ylab,
  drawTimeRange,
  simplify = 2560,
  fill = FALSE,
  col = par("col"),
  pch = par("pch"),
  cex = par("cex"),
  cex.axis = par("cex.axis"),
  cex.lab = par("cex.lab"),
  cex.main = par("cex.main"),
  xaxs = par("xaxs"),
  yaxs = par("yaxs"),
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + if (nchar(xlab) &gt; 0) 1.5 else 1, mgp[1] + 1.5, mgp[2] + 1, mgp[2] +
    3/4),
  main = "",
  despike = FALSE,
  axes = TRUE,
  tformat,
  marginsAsImage = FALSE,
  grid = FALSE,
  grid.col = "lightgray",
  grid.lty = "dotted",
  grid.lwd = par("lwd"),
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oce.plot.ts_+3A_x">x</code></td>
<td>
<p>the times of observations.  If this is not a <a href="base.html#topic+POSIXt">POSIXt</a> object, then an
attempt is made to convert it to one using <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_y">y</code></td>
<td>
<p>the observations.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_type">type</code></td>
<td>
<p>plot type, <code>"l"</code> for lines, <code>"p"</code> for points.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_xlim">xlim</code></td>
<td>
<p>optional limit for x axis.  This has an additional effect,
beyond that for conventional R functions: it effectively windows the data,
so that autoscaling will yield limits for y that make sense within the
window.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_ylim">ylim</code></td>
<td>
<p>optional limit for y axis.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_log">log</code></td>
<td>
<p>a character value that must be either empty (the default) for linear
<code>y</code> axis, or <code>"y"</code> for logarithmic <code>y</code> axis.  (Unlike
<code><a href="graphics.html#topic+plot.default">plot.default()</a></code> etc., <code>oce.plot.ts</code> does not permit
logarithmic time, or <code>x</code> axis.)</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_logstyle">logStyle</code></td>
<td>
<p>a character value that indicates how to draw the y axis, if
<code>log="y"</code>.  If it is <code>"r"</code> (the default) then the conventional R style is used,
in which a logarithmic transform connects y values to position on the &quot;page&quot;
of the plot device, so that tics will be nonlinearly spaced, but not
organized by integral powers of 10.  However, if it is <code>"decade"</code>, then
the style will be that used in the scientific literature, in which large
tick marks are used for integral powers of 10, with smaller tick marks
at integral multiples of those powers, and with labels that use exponential
format for values above 100 or below 0.01.  The value of <code>logStyle</code> is passed
to <code><a href="#topic+oceAxis">oceAxis()</a></code>, which draws the axis.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_flipy">flipy</code></td>
<td>
<p>Logical, with <code>TRUE</code> indicating that the graph
should have the y axis reversed, i.e. with smaller values at
the bottom of the page.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_xlab">xlab</code></td>
<td>
<p>name for x axis; defaults to <code>""</code>.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_ylab">ylab</code></td>
<td>
<p>name for y axis; defaults to the plotted item.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_drawtimerange">drawTimeRange</code></td>
<td>
<p>an optional indication of whether/how to draw a time range,
in the top-left margin of the plot; see <code><a href="#topic+oce.axis.POSIXct">oce.axis.POSIXct()</a></code> for details.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_simplify">simplify</code></td>
<td>
<p>an integer value that indicates
whether to speed up <code>type="l"</code> plots by replacing the data
with minimum and maximum values within a subsampled time mesh.
This can speed up plots of large datasets (e.g. by factor 20 for 10^7 points),
sometimes with minor changes in appearance.
This procedure is skipped if <code>simplify</code> is <code>NA</code> or
if the number of visible data points is less than 5 times <code>simplify</code>.
Otherwise, <code>oce.plot.ts</code> creates <code>simplify</code> intervals ranging across
the visible time range. Intervals with under 2 finite
<code>y</code> data are ignored. In the rest, <code>y</code> values
are replaced with their range, and <code>x</code> values are replaced
with the repeated midpoint time. Thus, each retained sub-interval
has exactly 2 data points.
A warning is printed if this replacement is done.
The default value of <code>simplify</code> means that cases with
under 2560 visible points are plotted conventionally.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_fill">fill</code></td>
<td>
<p>boolean, set <code>TRUE</code> to fill the curve to zero (which it
does incorrectly if there are missing values in <code>y</code>).</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_col">col</code></td>
<td>
<p>The colours for points (if <code>type=="p"</code>) or lines (if <code>type=="l"</code>).
For the <code>type="p"</code> case,
if there are fewer <code>col</code> values than there are <code>x</code> values, then the <code>col</code> values
are recycled in the standard fashion.
For the <code>type="l"</code> case, the line is plotted in the first colour specified.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_pch">pch</code></td>
<td>
<p>character code, used if <code>type=="p"</code>.
If there are fewer <code>pch</code> values than there are <code>x</code> values, then the <code>pch</code> values
are recycled in the standard fashion.
See <code><a href="graphics.html#topic+points">points()</a></code> for the possible values for <code>pch</code>.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_cex">cex</code></td>
<td>
<p>numeric character expansion factor for points on plots, ignored unless
<code>type</code> is <code>"p"</code>.  This may be a single number, applied to all points, or
a vector of numbers to be applied to the points in sequence.  If there are
fewer <code>pch</code> values than there are <code>x</code> values, then the <code>pch</code> values are recycled
in the standard fashion. See <code><a href="graphics.html#topic+par">par()</a></code> for more on <code>cex</code>.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_cex.axis">cex.axis</code>, <code id="oce.plot.ts_+3A_cex.lab">cex.lab</code>, <code id="oce.plot.ts_+3A_cex.main">cex.main</code></td>
<td>
<p>numeric character expansion factors for axis numbers,
axis names and plot titles; see <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_xaxs">xaxs</code></td>
<td>
<p>control x axis ending; see <code><a href="graphics.html#topic+par">par</a></code><code>("xaxs")</code>.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_yaxs">yaxs</code></td>
<td>
<p>control y axis ending; see <code><a href="graphics.html#topic+par">par</a></code><code>("yaxs")</code>.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code>par(mgp)</code>, and also
for <code>par(mar)</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code> to set margins.
The default value uses significantly tighter margins than is the norm in R,
which gives more space for the data.  However, in doing this, the existing
<code>par("mar")</code> value is ignored, which contradicts values that may have
been set by a previous call to <code><a href="#topic+drawPalette">drawPalette()</a></code>.  To get plot with
a palette, first call <code><a href="#topic+drawPalette">drawPalette()</a></code>, then call
<code>oce.plot.ts</code> with <code>mar=par("mar")</code>.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_main">main</code></td>
<td>
<p>title of plot.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_despike">despike</code></td>
<td>
<p>boolean flag that can turn on despiking with
<code><a href="#topic+despike">despike()</a></code>.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_axes">axes</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to get axes plotted</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_tformat">tformat</code></td>
<td>
<p>optional format for labels on the time axis</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_marginsasimage">marginsAsImage</code></td>
<td>
<p>boolean indicating whether to set the right-hand
margin to the width normally taken by an image drawn with
<code><a href="#topic+imagep">imagep()</a></code>.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_grid">grid</code></td>
<td>
<p>if <code>TRUE</code>, a grid will be drawn for each panel.  (This
argument is needed, because calling <code><a href="graphics.html#topic+grid">grid()</a></code> after doing a
sequence of plots will not result in useful results for the individual
panels.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_grid.col">grid.col</code></td>
<td>
<p>color of grid</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_grid.lty">grid.lty</code></td>
<td>
<p>line type of grid</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_grid.lwd">grid.lwd</code></td>
<td>
<p>line width of grid</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="oce.plot.ts_+3A_...">...</code></td>
<td>
<p>graphical parameters passed down to <code><a href="graphics.html#topic+plot">plot()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Depending on the version of R, the standard <code><a href="graphics.html#topic+plot">plot()</a></code> and
<code><a href="stats.html#topic+plot.ts">plot.ts()</a></code> routines will not obey the time zone of the data.
This routine gets around that problem.  It can also plot the time range in
the top-left margin, if desired; this string includes the timezone, to
remove any possible confusion.
The time axis is drawn with <code><a href="#topic+oce.axis.POSIXct">oce.axis.POSIXct()</a></code>.
</p>


<h3>Value</h3>

<p>A list is silently returned, containing <code>xat</code> and <code>yat</code>,
values that can be used by <code><a href="#topic+oce.grid">oce.grid()</a></code> to add a grid to the plot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
t0 &lt;- as.POSIXct("2008-01-01", tz = "UTC")
t &lt;- seq(t0, length.out = 48, by = "30 min")
y &lt;- sin(as.numeric(t - t0) * 2 * pi / (12 * 3600))
oce.plot.ts(t, y, type = "l", xaxs = "i")
# Show how col, pch and cex get recycled
oce.plot.ts(t, y,
    type = "p", xaxs = "i",
    col = 1:3, pch = c(rep(1, 6), rep(20, 6)), cex = sqrt(1:6)
)
# Trimming x; note the narrowing of the y view
oce.plot.ts(t, y, type = "p", xlim = c(t[6], t[12]))
# Flip the y axis
oce.plot.ts(t, y, flipy = TRUE)
</code></pre>

<hr>
<h2 id='oce.write.table'>Write the Data Portion of Object to a File</h2><span id='topic+oce.write.table'></span>

<h3>Description</h3>

<p>The output has a line containing the names of the columns in <code>x$data</code>,
each enclosed in double quotes.  After that line are lines for the data
themselves.  The default is to separate data items by a single space
character, but this can be altered by using a <code>sep</code> argument in the
<code>...</code> list; see <code><a href="utils.html#topic+write.table">utils::write.table()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oce.write.table(x, file = "", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oce.write.table_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="oce.write.table_+3A_file">file</code></td>
<td>
<p>file name, as passed to <code><a href="utils.html#topic+write.table">utils::write.table()</a></code>.  Use
<code>""</code> to get a listing in the terminal window.</p>
</td></tr>
<tr><td><code id="oce.write.table_+3A_...">...</code></td>
<td>
<p>optional arguments passed to <code><a href="utils.html#topic+write.table">utils::write.table()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is little more than a thin wrapper around
<code><a href="utils.html#topic+write.table">utils::write.table()</a></code>, the only difference being that row names
are omitted here, making for a file format that is more conventional in
Oceanography.
</p>


<h3>Value</h3>

<p>The value returned by <code><a href="utils.html#topic+write.table">utils::write.table()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>'<code><a href="utils.html#topic+write.table">utils::write.table()</a></code>, which does the actual work.
</p>

<hr>
<h2 id='oceApprox'>Interpolate 1D Data with UNESCO or Reiniger-Ross Algorithm</h2><span id='topic+oceApprox'></span><span id='topic+oce.approx'></span>

<h3>Description</h3>

<p>Interpolate one-dimensional data using schemes that permit curvature but
tends minimize extrema that are not well-indicated by the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceApprox(x, y, xout, method = c("rr", "unesco"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceApprox_+3A_x">x</code></td>
<td>
<p>the independent variable (z or p, usually).</p>
</td></tr>
<tr><td><code id="oceApprox_+3A_y">y</code></td>
<td>
<p>the dependent variable.</p>
</td></tr>
<tr><td><code id="oceApprox_+3A_xout">xout</code></td>
<td>
<p>the values of the independent variable at which interpolation is
to be done.</p>
</td></tr>
<tr><td><code id="oceApprox_+3A_method">method</code></td>
<td>
<p>method to use.  See &ldquo;Details&rdquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Setting <code>method="rr"</code> yields the weighted-parabola algorithm of
Reiniger and Ross (1968).  For procedure is as follows.  First, the
interpolant for any <code>xout</code> value that is outside the range of <code>x</code>
is set to NA.  Next, linear interpolation is used for any <code>xout</code> value
that has only one smaller neighboring <code>x</code> value, or one larger
neighboring value.  For all other values of <code>xout</code>, the 4 neighboring
points <code>x</code> are sought, two smaller and two larger.  Then two parabolas
are determined, one from the two smaller points plus the nearest larger
point, and the other from the nearest smaller point and the two larger
points.  A weighted sum of these two parabolas provides the interpolated
value.  Note that, in the notation of Reiniger and Ross (1968), this
algorithm uses <code>m</code>=2 and <code>n</code>=1.  (A future version of this routine
might provide the ability to modify these values.)
</p>
<p>Setting <code>method="unesco"</code> yields the method that is used by the U.S.
National Oceanographic Data Center. It is described in pages 48-50 of
reference 2; reference 3 presumably contains the same information but it is
not as easily accessible.  The method works as follows.
</p>

<ul>
<li><p> If there are data above 5m depth, then the surface value is taken to
equal to the shallowest recorded value.
</p>
</li>
<li><p> Distance bounds are put on the four neighboring points, and the
Reiniger-Ross method is used for interpolated points with sufficiently four
close neighbors.  The bounds are described in table 15 of reference 2 only
for so-called standard depths; in the present instance they are transformed
to the following rules.  Inner neighbors must be within 5m for data above
10m, 50m above 250m 100m above 900m, 200m above 2000m, or within 1000m
otherwise.  Outer neighbors must be within 200m above 500m, 400m above
1300m, or 1000m otherwise.  If two or more points meet these criteria,
Lagrangian interpolation is used.  If not, <code>NA</code> is used as the
interpolant.
</p>
</li></ul>

<p>After these rules are applied, the interpolated value is compared with the
values immediately above and below it, and if it is outside the range,
simple linear interpolation is used.
</p>


<h3>Value</h3>

<p>A vector of interpolated values, corresponding to the <code>xout</code>
values and equal in number.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> R.F. Reiniger and C.K. Ross, 1968.  A method of interpolation with
application to oceanographic data.  <em>Deep Sea Research</em>, <strong>15</strong>,
185-193.
</p>
</li>
<li><p> Daphne R. Johnson, Tim P. Boyer, Hernan E. Garcia, Ricardo A.
Locarnini, Olga K. Baranova, and Melissa M. Zweng, 2011. World Ocean
Database 2009 Documentation.  NODC Internal report 20.  Ocean Climate
Laboratory, National Oceanographic Data Center.  Silver Spring, Maryland.
</p>
</li>
<li><p> UNESCO, 1991. Processing of oceanographic station data, 138 pp.,
Imprimerie des Presses Universitaires de France, United Nations Educational,
Scientific and Cultural Organization, France.
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
if (require(ocedata)) {
    data(RRprofile)
    zz &lt;- seq(0, 2000, 2)
    plot(RRprofile$temperature, RRprofile$depth, ylim = c(500, 0), xlim = c(2, 11))
    # Contrast two methods
    a1 &lt;- oce.approx(RRprofile$depth, RRprofile$temperature, zz, "rr")
    a2 &lt;- oce.approx(RRprofile$depth, RRprofile$temperature, zz, "unesco")
    lines(a1, zz)
    lines(a2, zz, col = "red")
    legend("bottomright", lwd = 1, col = 1:2, legend = c("rr", "unesco"), cex = 3 / 4)
}
</code></pre>

<hr>
<h2 id='oceAxis'>Draw an Axis, Possibly with Decade-style Logarithmic Scaling</h2><span id='topic+oceAxis'></span>

<h3>Description</h3>

<p>Draw an Axis, Possibly with Decade-style Logarithmic Scaling
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceAxis(side, labels = TRUE, logStyle = "r", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceAxis_+3A_side">side</code></td>
<td>
<p>an integer specifying which axis to draw, with 1 for bottom axis, 2 for left axis,
3 for top axis, and 4 for right axis (as with <code><a href="graphics.html#topic+axis">axis()</a></code>).</p>
</td></tr>
<tr><td><code id="oceAxis_+3A_labels">labels</code></td>
<td>
<p>either a vector of character values used for labels or a logical value indicating
whether to draw such labels.  The first form only works if the coordinate is not logarithmic,
and if <code>logStyle</code> is <code>"r"</code>.</p>
</td></tr>
<tr><td><code id="oceAxis_+3A_logstyle">logStyle</code></td>
<td>
<p>a character value that indicates how to draw the y axis, if
<code>log="y"</code>.  If it is <code>"r"</code> (the default) then the conventional R style is used,
in which a logarithmic transform connects y values to position on the &quot;page&quot;
of the plot device, so that tics will be nonlinearly spaced, but not
organized by integral powers of 10.  However, if it is <code>"decade"</code>, then
the style will be that used in the scientific literature, in which large
tick marks are used for integral powers of 10, with smaller tick marks
at integral multiples of those powers, and with labels that use exponential
format for values above 100 or below 0.01.</p>
</td></tr>
<tr><td><code id="oceAxis_+3A_...">...</code></td>
<td>
<p>other graphical parameters, passed to <code><a href="graphics.html#topic+axis">axis()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numerical values at which tick marks were drawn (or would have been drawn, if <code>labels</code>
specified to draw them).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
Ra &lt;- 10^seq(4, 10, 0.1)
Nu &lt;- 0.085 * Ra^(1 / 3)
plot(Ra, Nu, log = "xy", axes = FALSE)
box()
oceAxis(1, logStyle = "decade")
oceAxis(2, logStyle = "decade")

</code></pre>

<hr>
<h2 id='ocecolors'>Data That Define Some Color Palettes</h2><span id='topic+ocecolors'></span>

<h3>Description</h3>

<p>The <code>ocecolors</code> dataset is a list containing color-schemes, used
by <code><a href="#topic+oceColorsClosure">oceColorsClosure()</a></code> to create functions such as
<code><a href="#topic+oceColorsViridis">oceColorsViridis()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Authored by matplotlib contributers, packaged (with license permission) in oce by Dan Kelley
</p>


<h3>Source</h3>

<p>The data come from the matplotlib site
<code style="white-space: pre;">&#8288;https://github.com/matplotlib/matplotlib&#8288;</code>.
</p>


<h3>References</h3>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>
<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>
</p>

<hr>
<h2 id='oceColors9B'>Create Colors in a Red-Yellow-Blue Color Scheme</h2><span id='topic+oceColors9B'></span><span id='topic+oce.colors9B'></span>

<h3>Description</h3>

<p>The results are similar to those of <code><a href="#topic+oceColorsJet">oceColorsJet()</a></code>, but
with white hues in the centre, rather than green ones. The scheme
may be useful in displaying signed quantities, and thus is somewhat
analogous to <code><a href="#topic+oceColorsTwo">oceColorsTwo()</a></code>, except that some viewers
may be able to distinguish more colors with <code>oceColors9B</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColors9B(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColors9B_+3A_n">n</code></td>
<td>
<p>number of colors</p>
</td></tr>
</table>


<h3>References</h3>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
imagep(volcano,
    col = oceColors9B(128),
    zlab = "oceColors9B"
)

</code></pre>

<hr>
<h2 id='oceColorsCDOM'>Create Colors Suitable for CDOM Fields</h2><span id='topic+oceColorsCDOM'></span><span id='topic+oce.colorsCDOM'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying CDOM values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsCDOM(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsCDOM_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsCDOM(128),
    zlab="oceColorsCDOM")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceColorsChlorophyll'>Create Colors Suitable for chlorophyll Fields</h2><span id='topic+oceColorsChlorophyll'></span><span id='topic+oce.colorsChlorophyll'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying chlorophyll values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsChlorophyll(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsChlorophyll_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsChlorophyll(128),
    zlab="oceColorsChlorophyll")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceColorsClosure'>Create Color Functions</h2><span id='topic+oceColorsClosure'></span>

<h3>Description</h3>

<p>This function generates other functions that are used to specify colors.
It is used within oce to create <code><a href="#topic+oceColorsTemperature">oceColorsTemperature()</a></code>
and its many cousins. Users may also find it helpful, for creating
custom color schemes (see &ldquo;Examples&rdquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsClosure(spec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsClosure_+3A_spec">spec</code></td>
<td>
<p>Specification of the color scheme. This may be a
character string, in which case it must be the name of an item stored
in <code>data(ocecolors)</code>, or either a 3-column data frame or
matrix, in which case the columns specify red, green and blue values
(in range from 0 to 1).</p>
</td></tr>
</table>


<h3>Sample of Usage</h3>

<pre>
# Update oxygen color scheme to latest matplotlib value.
library(oce)
oxy &lt;- "https://raw.githubusercontent.com/matplotlib/cmocean/master/cmocean/rgb/oxy-rgb.txt"
oxyrgb &lt;- read.table(oxy, header=FALSE)
oceColorsOxygenUpdated &lt;- oceColorsClosure(oxyrgb)
par(mfrow=c(1, 2))
m &lt;- matrix(1:256)
imagep(m, col=oceColorsOxygen, zlab="oxygen")
imagep(m, col=oceColorsOxygenUpdated, zlab="oxygenUpdated")
</pre>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>

<hr>
<h2 id='oceColorsDensity'>Create Colors Suitable for density Fields</h2><span id='topic+oceColorsDensity'></span><span id='topic+oce.colorsDensity'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying density values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsDensity(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsDensity_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsDensity(128),
    zlab="oceColorsDensity")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceColorsFreesurface'>Create Colors Suitable for freesurface Fields</h2><span id='topic+oceColorsFreesurface'></span><span id='topic+oce.colorsFreesurface'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying freesurface values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsFreesurface(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsFreesurface_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsFreesurface(128),
    zlab="oceColorsFreesurface")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceColorsGebco'>Create Colors in a GEBCO-like Scheme</h2><span id='topic+oceColorsGebco'></span><span id='topic+oce.colorsGebco'></span>

<h3>Description</h3>

<p>The colours were determined by examination of paper
charts printed during the GEBCO Fifth Edition era.
The hues range from dark blue to light blue, then
from light brown to dark brown.  If used to show
topography in scheme centred on z=0, this means that
near-coastal regions are light in tone, with darker
colours representing both mountains and the deep sea.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsGebco(
  n = 9,
  region = c("water", "land", "both"),
  type = c("fill", "line"),
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsGebco_+3A_n">n</code></td>
<td>
<p>Number of colors to return</p>
</td></tr>
<tr><td><code id="oceColorsGebco_+3A_region">region</code></td>
<td>
<p>String indicating application region,
one of <code>"water"</code>, <code>"land"</code>, or <code>"both"</code>.</p>
</td></tr>
<tr><td><code id="oceColorsGebco_+3A_type">type</code></td>
<td>
<p>String indicating the purpose, one of <code>"fill"</code> or <code>"line"</code>.</p>
</td></tr>
<tr><td><code id="oceColorsGebco_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
imagep(volcano, col = oceColorsGebco(128, region = "both"))

</code></pre>

<hr>
<h2 id='oceColorsJet'>Create Colors Similar to the Matlab Jet Scheme</h2><span id='topic+oceColorsJet'></span><span id='topic+oce.colorsJet'></span><span id='topic+oceColors9A'></span><span id='topic+oce.colors9A'></span>

<h3>Description</h3>

<p>Create Colors Similar to the Matlab Jet Scheme
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsJet(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsJet_+3A_n">n</code></td>
<td>
<p>number of colors</p>
</td></tr>
</table>


<h3>References</h3>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
imagep(volcano, col = oceColorsJet, zlab = "oceColorsJet")

</code></pre>

<hr>
<h2 id='oceColorsOxygen'>Create Colors Suitable for oxygen Fields</h2><span id='topic+oceColorsOxygen'></span><span id='topic+oce.colorsOxygen'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying oxygen values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsOxygen(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsOxygen_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsOxygen(128),
    zlab="oceColorsOxygen")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceColorsPalette'>Create a Vector of Colors</h2><span id='topic+oceColorsPalette'></span><span id='topic+oce.colorsPalette'></span>

<h3>Description</h3>

<p>The available schemes are:
</p>

<ul>
<li> <p><code>which=1</code> for a red-white-blue scheme.
</p>
</li>
<li> <p><code>which=2</code> for a red-yellow-blue scheme.
</p>
</li>
<li> <p><code>which=9.01</code>, <code>which="9A"</code> or <code>which="jet"</code> for <code><a href="#topic+oceColorsJet">oceColorsJet</a></code><code>(n)</code>.
</p>
</li>
<li> <p><code>which=9.02</code> or <code>which="9B"</code> for <code><a href="#topic+oceColors9B">oceColors9B</a></code><code>(n)</code>.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>oceColorsPalette(n, which = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsPalette_+3A_n">n</code></td>
<td>
<p>number of colors to create</p>
</td></tr>
<tr><td><code id="oceColorsPalette_+3A_which">which</code></td>
<td>
<p>integer or character string indicating the palette
to use; see &ldquo;Details&rdquo;.</p>
</td></tr>
</table>


<h3>References</h3>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>

<hr>
<h2 id='oceColorsPAR'>Create Colors Suitable for PAR Fields</h2><span id='topic+oceColorsPAR'></span><span id='topic+oce.colorsPAR'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying PAR values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsPAR(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsPAR_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsPAR(128),
    zlab="oceColorsPAR")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceColorsPhase'>Create Colors Suitable for phase Fields</h2><span id='topic+oceColorsPhase'></span><span id='topic+oce.colorsPhase'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying phase values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsPhase(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsPhase_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsPhase(128),
    zlab="oceColorsPhase")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceColorsSalinity'>Create Colors Suitable for salinity Fields</h2><span id='topic+oceColorsSalinity'></span><span id='topic+oce.colorsSalinity'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying salinity values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsSalinity(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsSalinity_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsSalinity(128),
    zlab="oceColorsSalinity")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceColorsTemperature'>Create Colors Suitable for temperature Fields</h2><span id='topic+oceColorsTemperature'></span><span id='topic+oce.colorsTemperature'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying temperature values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsTemperature(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsTemperature_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsTemperature(128),
    zlab="oceColorsTemperature")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceColorsTurbidity'>Create Colors Suitable for turbidity Fields</h2><span id='topic+oceColorsTurbidity'></span><span id='topic+oce.colorsTurbidity'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying turbidity values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsTurbidity(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsTurbidity_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsTurbidity(128),
    zlab="oceColorsTurbidity")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceColorsTurbo'>Create Colors Similar to the Google Turbo Scheme</h2><span id='topic+oceColorsTurbo'></span><span id='topic+oce.colorsTurbo'></span>

<h3>Description</h3>

<p>This uses the coefficients published (with Apache license) by google,
as described by Mikhailo (2019).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsTurbo(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsTurbo_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Mikhailo, Anton.
&ldquo;Turbo, An Improved Rainbow Colormap for Visualization.&rdquo;
Google AI (blog), August 20, 2019.
<code style="white-space: pre;">&#8288;https://ai.googleblog.com/2019/08/turbo-improved-rainbow-colormap-for.html&#8288;</code>
</p>
<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
imagep(volcano,
    col = oceColorsTurbo(128),
    zlab = "oceColorsTurbo"
)

</code></pre>

<hr>
<h2 id='oceColorsTwo'>Create Two-Color Palette</h2><span id='topic+oceColorsTwo'></span><span id='topic+oce.colorsTwo'></span>

<h3>Description</h3>

<p>Create colors ranging between two specified limits, with white
in the middle.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsTwo(n, low = 2/3, high = 0, smax = 1, alpha = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsTwo_+3A_n">n</code></td>
<td>
<p>number of colors to generate.</p>
</td></tr>
<tr><td><code id="oceColorsTwo_+3A_low">low</code>, <code id="oceColorsTwo_+3A_high">high</code></td>
<td>
<p>numerical values (in range 0 to 1) specifying the hue
for the low and high ends of the color scale.</p>
</td></tr>
<tr><td><code id="oceColorsTwo_+3A_smax">smax</code></td>
<td>
<p>numerical value (in range 0 to 1) for the color saturation.</p>
</td></tr>
<tr><td><code id="oceColorsTwo_+3A_alpha">alpha</code></td>
<td>
<p>numerical value (in ragne 0 to 1) for the alpha (transparency)
of the colors.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
imagep(volcano - mean(range(volcano)),
    col = oceColorsTwo(128),
    zlim = "symmetric", zlab = "oceColorsTwo"
)
</code></pre>

<hr>
<h2 id='oceColorsVelocity'>Create Colors Suitable for velocity Fields</h2><span id='topic+oceColorsVelocity'></span><span id='topic+oce.colorsVelocity'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying velocity values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsVelocity(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsVelocity_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsVelocity(128),
    zlab="oceColorsVelocity")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceColorsViridis'>Create Colors Similar to the Matlab Viridis Scheme</h2><span id='topic+oceColorsViridis'></span><span id='topic+oce.colorsViridis'></span>

<h3>Description</h3>

<p>This is patterned on a matlab/python scheme that blends
from yellow to blue in a way that is designed to reproduce well
in black-and-white, and to be interpretable by those with
certain forms of color blindness.  See the references for
notes about issues of colour blindness in computer graphics.
An alternative
to <a href="#topic+oceColorsViridis">oceColorsViridis</a> is provided in the <a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated
in Example 2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsViridis(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsViridis_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>
<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsVorticity">oceColorsVorticity</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Example 1: oceColorsViridis
imagep(volcano,
    col = oceColorsViridis(128),
    zlab = "oceColorsViridis"
)

</code></pre>

<hr>
<h2 id='oceColorsVorticity'>Create Colors Suitable for vorticity Fields</h2><span id='topic+oceColorsVorticity'></span><span id='topic+oce.colorsVorticity'></span>

<h3>Description</h3>

<p>Create a set of colors for displaying vorticity values,
based on the scheme devised by Thyng et al. (2016) and
presented in a python package by Thyng (2019).
The color specifications were transliterated from
python to R on 2015-09-29, but have not been adjusted
since, even though the python source has changed.  This
is to prevent breaking old <code>oce</code> code.  To get the
latest versions of these colours or other colours,
use the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> R package
(Thyng, Richards, and Krylov, 2019) directly, as is
illustrated (with the &quot;matter&quot; scheme) in Example 2.
Note that the <a href="https://CRAN.R-project.org/package=cmocean"><span class="pkg">cmocean</span></a> core functions provide a way
to select between various versions of the colour schemes.
It is also worth considering the palettes provided by the
<a href="https://CRAN.R-project.org/package=viridis"><span class="pkg">viridis</span></a> package, as illustrated (with the
&quot;inferno&quot; scheme) in Example 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceColorsVorticity(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceColorsVorticity_+3A_n">n</code></td>
<td>
<p>number of colors to create.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of color specifications.
</p>


<h3>Author(s)</h3>

<p>Krysten M. Thyng (Python version), Dan Kelley (R transliteration)
</p>


<h3>References</h3>


<ul>
<li><p> Thyng, Kristen, Chad Greene, Robert Hetland, Heather Zimmerle, and Steven DiMarco.
&ldquo;True Colors of Oceanography: Guidelines for Effective and Accurate Colormap Selection.&rdquo;
Oceanography 29, no. 3 (September 1, 2016): 9–13.
<a href="https://doi.org/10.5670/oceanog.2016.66">doi:10.5670/oceanog.2016.66</a>
</p>
</li>
<li><p> Thyng, Kristen. Kthyng/Cmocean. Python, 2019.
<code style="white-space: pre;">&#8288;https://github.com/kthyng/cmocean&#8288;</code>.
</p>
</li>
<li><p> Thyng, Kristen, Clark Richards, and Ivan Krylov.
Cmocean: Beautiful Colour Maps for Oceanography (version 0.2), 2019.
<code style="white-space: pre;">&#8288;https://CRAN.R-project.org/package=cmocean&#8288;</code>.
</p>
</li></ul>

<p>The following references provide information on choosing
colour schemes, that are suitable for viewers who have
colour deficiencies.
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;The End of the Rainbow? Color
Schemes for Improved Data Graphics.&quot; <em>Eos, Transactions American Geophysical
Union</em> 85, no. 40 (2004): 385.
DOI: 10.1029/2004EO400002
</p>
<p>Stephenson, David B. &quot;Comment on 'Color Schemes for Improved Data
Graphics', by A Light and P.J. Bartlein.&quot; <em>Eos, Transactions American
Geophysical Union</em> 86, no.  20 (2005): 196.
DOI: 10.1029/2005EO200005
</p>
<p>Light, Adam, and Patrick J. Bartlein. &quot;Reply to 'Comment on Color
Schemes for Improved Data Graphics,' by A. Light and P.J.  Bartlein'.&quot;
<em>Eos, Transactions American Geophysical Union</em> 86, no. 20 (2005): 196–196.
DOI: 10.1029/2005EO200006
</p>


<h3>See Also</h3>

<p>Other things related to colors: 
<code><a href="#topic+colormapGMT">colormapGMT</a>()</code>,
<code><a href="#topic+colormap">colormap</a>()</code>,
<code><a href="#topic+oceColors9B">oceColors9B</a>()</code>,
<code><a href="#topic+oceColorsCDOM">oceColorsCDOM</a>()</code>,
<code><a href="#topic+oceColorsChlorophyll">oceColorsChlorophyll</a>()</code>,
<code><a href="#topic+oceColorsClosure">oceColorsClosure</a>()</code>,
<code><a href="#topic+oceColorsDensity">oceColorsDensity</a>()</code>,
<code><a href="#topic+oceColorsFreesurface">oceColorsFreesurface</a>()</code>,
<code><a href="#topic+oceColorsGebco">oceColorsGebco</a>()</code>,
<code><a href="#topic+oceColorsJet">oceColorsJet</a>()</code>,
<code><a href="#topic+oceColorsOxygen">oceColorsOxygen</a>()</code>,
<code><a href="#topic+oceColorsPAR">oceColorsPAR</a>()</code>,
<code><a href="#topic+oceColorsPalette">oceColorsPalette</a>()</code>,
<code><a href="#topic+oceColorsPhase">oceColorsPhase</a>()</code>,
<code><a href="#topic+oceColorsSalinity">oceColorsSalinity</a>()</code>,
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature</a>()</code>,
<code><a href="#topic+oceColorsTurbidity">oceColorsTurbidity</a>()</code>,
<code><a href="#topic+oceColorsTurbo">oceColorsTurbo</a>()</code>,
<code><a href="#topic+oceColorsTwo">oceColorsTwo</a>()</code>,
<code><a href="#topic+oceColorsVelocity">oceColorsVelocity</a>()</code>,
<code><a href="#topic+oceColorsViridis">oceColorsViridis</a>()</code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Example 1
imagep(volcano, col=oceColorsVorticity(128),
    zlab="oceColorsVorticity")
## Not run: 
# Example 2 (requires the cmocean package)
imagep(volcano, col=cmocean::cmocean("matter"),
    zlab="cmocean::cmocean(\"matter\")")
## End(Not run)

## Not run: 
# Example 3 (requires the viridis package)
imagep(volcano, col=viridis::inferno,
    zlab="viridis::inferno")
## End(Not run)

</code></pre>

<hr>
<h2 id='oceConvolve'>Convolve Two Time Series</h2><span id='topic+oceConvolve'></span><span id='topic+oce.convolve'></span>

<h3>Description</h3>

<p>Convolve two time series, using a backward-looking method.
This function provides a straightforward convolution, which may be useful to
those who prefer not to use <code><a href="stats.html#topic+convolve">convolve()</a></code> and <code>filter</code> in the
<code>stats</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceConvolve(x, f, end = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceConvolve_+3A_x">x</code></td>
<td>
<p>a numerical vector of observations.</p>
</td></tr>
<tr><td><code id="oceConvolve_+3A_f">f</code></td>
<td>
<p>a numerical vector of filter coefficients.</p>
</td></tr>
<tr><td><code id="oceConvolve_+3A_end">end</code></td>
<td>
<p>a flag that controls how to handle the points of the <code>x</code>
series that have indices less than the length of <code>f</code>.  If <code>end=0</code>,
the values are set to 0.  If <code>end=1</code>, the original x values are used
there.  If <code>end=2</code>, that fraction of the <code>f</code> values that overlap
with <code>x</code> are used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of the convolution output.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
t &lt;- 0:1027
n &lt;- length(t)
signal &lt;- ifelse(sin(t * 2 * pi / 128) &gt; 0, 1, 0)
tau &lt;- 10
filter &lt;- exp(-seq(5 * tau, 0) / tau)
filter &lt;- filter / sum(filter)
observation &lt;- oce.convolve(signal, filter)
plot(t, signal, type = "l")
lines(t, observation, lty = "dotted")
</code></pre>

<hr>
<h2 id='oceCRS'>Coordinate Reference System Strings for Some Oceans</h2><span id='topic+oceCRS'></span>

<h3>Description</h3>

<p>Create a coordinate reference string (CRS), suitable for use as a
<code>projection</code> argument to <code><a href="#topic+mapPlot">mapPlot()</a></code> or
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceCRS(region)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceCRS_+3A_region">region</code></td>
<td>
<p>character string indicating the region. This must be
in the following list (or a string that matches to just one entry,
with <code><a href="base.html#topic+pmatch">pmatch()</a></code>):
<code>"North Atlantic"</code>, <code>"South Atlantic"</code>, <code>"Atlantic"</code>,
<code>"North Pacific"</code>, <code>"South Pacific"</code>, <code>"Pacific"</code>,
<code>"Arctic"</code>,  and <code>"Antarctic"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>string contain a CRS, which can be used as <code>projection</code>
in <code><a href="#topic+mapPlot">mapPlot()</a></code>.
</p>


<h3>Caution</h3>

<p>This is a preliminary version of this function,
with the results being very likely to change through the autumn of 2016,
guided by real-world usage.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(coastlineWorld)
par(mar = c(2, 2, 1, 1))
plot(coastlineWorld, projection = oceCRS("Atlantic"), span = 12000)
plot(coastlineWorld, projection = oceCRS("North Atlantic"), span = 8000)
plot(coastlineWorld, projection = oceCRS("South Atlantic"), span = 8000)
plot(coastlineWorld, projection = oceCRS("Arctic"), span = 4000)
plot(coastlineWorld, projection = oceCRS("Antarctic"), span = 10000)
# Avoid ugly horizontal lines, an artifact of longitude shifting.
# Note: we cannot fill the land once we shift, either.
pacific &lt;- coastlineCut(coastlineWorld, -180)
plot(pacific, proj = oceCRS("Pacific"), span = 15000, col = NULL)
plot(pacific, proj = oceCRS("North Pacific"), span = 12000, col = NULL)
plot(pacific, proj = oceCRS("South Pacific"), span = 12000, col = NULL)

</code></pre>

<hr>
<h2 id='oceDebug'>Print a Debugging Message</h2><span id='topic+oceDebug'></span><span id='topic+oce.debug'></span>

<h3>Description</h3>

<p>Print an indented debugging message.
Many oce functions decrease the <code>debug</code> level by 1 when they call other
functions, so the effect is a nesting, with more space for deeper function
level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceDebug(debug = 0, ..., style = "plain", unindent = 0, sep = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceDebug_+3A_debug">debug</code></td>
<td>
<p>an integer, less than or equal to zero for no message, and
greater than zero for increasing levels of debugging.  Values greater than 4
are treated like 4.</p>
</td></tr>
<tr><td><code id="oceDebug_+3A_...">...</code></td>
<td>
<p>items to be supplied to <code><a href="base.html#topic+cat">cat()</a></code>, which does the
printing.  Note that no newline will be printed unless ...
contains a string with a newline character (as in the example).</p>
</td></tr>
<tr><td><code id="oceDebug_+3A_style">style</code></td>
<td>
<p>either a string or a function. If a string,
it must be <code>"plain"</code> (the default) for plain text,
<code>"bold"</code>, <code>"italic"</code>, <code>"red"</code>, <code>"green"</code> or <code>"blue"</code> (with
obvious meanings).  Note that none of these has any effect
for non-interactive use, because doing so would make it difficult
to work with R-markdown and similar documents that are to be run
through latex.
</p>
<p>If <code>style</code> is a function, it must prepend and postpend the text
with control codes, as in the cyan-coloured example; note that
<a href="https://CRAN.R-project.org/package=crayon"><span class="pkg">crayon</span></a> provides many functions that work well for <code>style</code>.</p>
</td></tr>
<tr><td><code id="oceDebug_+3A_unindent">unindent</code></td>
<td>
<p>integer giving the number of levels to un-indent,
e.g. for start and end lines from a called function.</p>
</td></tr>
<tr><td><code id="oceDebug_+3A_sep">sep</code></td>
<td>
<p>character to insert between elements of <code>...</code>, by passing
it to <code><a href="base.html#topic+cat">cat()</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>oceDebug(debug = 1, "Example", 1, "Plain text")
oceDebug(debug = 1, "Example", 2, "Bold", style = "bold")
oceDebug(debug = 1, "Example", 3, "Italic", style = "italic")
oceDebug(debug = 1, "Example", 4, "Red", style = "red")
oceDebug(debug = 1, "Example", 5, "Green", style = "green")
oceDebug(debug = 1, "Example", 6, "Blue", style = "blue")
mycyan &lt;- function(...) paste("\033[36m", paste(..., sep = " "), "\033[0m", sep = "")
oceDebug(debug = 1, "Example", 7, "User-set cyan", style = mycyan)
</code></pre>

<hr>
<h2 id='oceDeleteData'>Delete Something From the data Slot of an oce Object</h2><span id='topic+oceDeleteData'></span>

<h3>Description</h3>

<p>Return a copy of the supplied object that lacks the named
element in its <code>data</code> slot, and that has a note
about the deletion in its processing log.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceDeleteData(object, name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceDeleteData_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="oceDeleteData_+3A_name">name</code></td>
<td>
<p>String indicating the name of the item to be deleted.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to the data slot: 
<code><a href="#topic+oceGetData">oceGetData</a>()</code>,
<code><a href="#topic+oceRenameData">oceRenameData</a>()</code>,
<code><a href="#topic+oceSetData">oceSetData</a>()</code>
</p>

<hr>
<h2 id='oceDeleteMetadata'>Delete Something in an oce metadata Slot</h2><span id='topic+oceDeleteMetadata'></span>

<h3>Description</h3>

<p>Return a copy of the supplied object that lacks the named
element in its <code>metadata</code> slot, and that has a note
about the deletion in its processing log.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceDeleteMetadata(object, name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceDeleteMetadata_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="oceDeleteMetadata_+3A_name">name</code></td>
<td>
<p>String indicating the name of the item to be deleted.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to the metadata slot: 
<code><a href="#topic+oceGetMetadata">oceGetMetadata</a>()</code>,
<code><a href="#topic+oceRenameMetadata">oceRenameMetadata</a>()</code>,
<code><a href="#topic+oceSetMetadata">oceSetMetadata</a>()</code>
</p>

<hr>
<h2 id='oceEdit'>Edit an Oce Object</h2><span id='topic+oceEdit'></span><span id='topic+oce.edit'></span>

<h3>Description</h3>

<p>Edit an element of an oce object, inserting a note in the processing
log of the returned object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceEdit(
  x,
  item,
  value,
  action,
  reason = "",
  person = "",
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceEdit_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.  The exact action of <code><a href="#topic+oceEdit">oceEdit()</a></code> depends
on the sub-class of <code>x</code>.</p>
</td></tr>
<tr><td><code id="oceEdit_+3A_item">item</code></td>
<td>
<p>if supplied, a character string naming an item in the object's
<code>metadata</code> or <code>data</code> slot, the former being checked first.
An exception is if <code>item</code> starts with <code>"data@"</code> or
<code>"metadata@"</code>, in which case the named slot is updated with a changed
value of the contents of <code>item</code> after the <code>@</code> character.</p>
</td></tr>
<tr><td><code id="oceEdit_+3A_value">value</code></td>
<td>
<p>new value for <code>item</code>, if both supplied.</p>
</td></tr>
<tr><td><code id="oceEdit_+3A_action">action</code></td>
<td>
<p>optional character string containing R code to carry out some
action on the object.</p>
</td></tr>
<tr><td><code id="oceEdit_+3A_reason">reason</code></td>
<td>
<p>character string giving the reason for the change.</p>
</td></tr>
<tr><td><code id="oceEdit_+3A_person">person</code></td>
<td>
<p>character string giving the name of person making the change.</p>
</td></tr>
<tr><td><code id="oceEdit_+3A_debug">debug</code></td>
<td>
<p>an integer that specifies a level of debugging, with 0 or less
indicating no debugging, and 1 or more indicating debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are several ways to use this function.
</p>

<ol>
<li><p> If both an <code>item</code> and <code>value</code> are supplied, then
either the object's metadata or data slot may be altered. There are
two ways in which this can be done.
</p>

<ul>
<li><p> Case 1A. If the <code>item</code> string does not contain an
<code>@</code> character, then the <code>metadata</code> slot is examined
for an entry named <code>item</code>, and that is modified if so.
Alternatively, if <code>item</code> is found in <code>metadata</code>, then
that value is modified. However, if <code>item</code> is not found in
either <code>metadata</code> or <code>data</code>, then an error is reported
(see 1B for how to add something that does not yet exist).
</p>
</li>
<li><p> Case 1B. If the <code>item</code> string contains
the <code>@</code> character, then the text to the left of that character
must be either <code>"metadata"</code> or <code>"data"</code>, and it names the slot
in which the change is done. In contrast with case 1A, this will
<em>create</em> a new item, if it is not already in existence.
</p>
</li></ul>

</li>
<li><p> If <code>item</code> and <code>value</code> are not supplied, then <code>action</code> must
be supplied.  This is a character string specifying some action to be
performed on the object, e.g. a manipulation of a column.  The action must
refer to the object as <code>x</code>, as in Example 2.
</p>
</li></ol>

<p>In any case, a log entry is stored in the object, to document the change.
Indeed, this is the main benefit to using this function, instead of altering
the object directly.  The log entry will be most useful if it contains a
brief note on the <code>reason</code> for the change, and the name of the
<code>person</code> doing the work.
</p>


<h3>Value</h3>

<p>A <a href="#topic+oce-class">oce</a> object, altered appropriately, and with a log item
indicating the nature of the alteration.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
# Example 1: change latitude
ctd2 &lt;- oceEdit(ctd,
    item = "latitude", value = 47.8879,
    reason = "illustration", person = "Dan Kelley"
)
# Example 2: add 0.1 dbar to pressure
ctd3 &lt;- oceEdit(ctd, action = "x@data$pressure&lt;-x@data$pressure+0.1")
</code></pre>

<hr>
<h2 id='oceFileTrim'>Trim an oce File</h2><span id='topic+oceFileTrim'></span>

<h3>Description</h3>

<p>Create an oce file by copying the first <code>n</code> data chunks of another such file.
This can be useful in supplying small sample files for bug reports. Only
a few file types (as inferred with <code><a href="#topic+oceMagic">oceMagic()</a></code>) are permitted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceFileTrim(infile, n = 100L, outfile, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceFileTrim_+3A_infile">infile</code></td>
<td>
<p>name of an AD2CP source file.</p>
</td></tr>
<tr><td><code id="oceFileTrim_+3A_n">n</code></td>
<td>
<p>integer indicating the number of data chunks to keep. The default is
to keep 100 chunks, a common good choice for sample files.</p>
</td></tr>
<tr><td><code id="oceFileTrim_+3A_outfile">outfile</code></td>
<td>
<p>optional name of the new file to be created. If this is not
supplied, a default is used, by adding <code style="white-space: pre;">&#8288;_trimmed&#8288;</code> to the base filename, e.g.
for an AD2CP file named <code>"a.ad2cp"</code>, the constructed value of <code>outfile</code> will
be <code>a_trimmed.ad2cp</code>.</p>
</td></tr>
<tr><td><code id="oceFileTrim_+3A_debug">debug</code></td>
<td>
<p>an integer value indicating the level of debugging. If
this is 1L, then a brief indication is given of the processing steps. If it
is &gt; 1L, then information is given about each data chunk, which can yield
very extensive output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>oceFileTrim()</code> returns the name of the output file, either provided
in the <code>outfile</code> parameter or constructed by this function.
</p>


<h3>Sample of Usage</h3>

<pre>
# Can only be run by the developer, since it uses a private file.
f  &lt;- "~/Dropbox/oce_secret_data/ad2cp/byg_trimmed.ad2cp"
if (file.exists(f)) {
    oceFileTrim(f, 10L) # this file holds 100 data segments
}
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that trim data files: 
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>
</p>

<hr>
<h2 id='oceFilter'>Filter a Time Series</h2><span id='topic+oceFilter'></span><span id='topic+oce.filter'></span>

<h3>Description</h3>

<p>Filter a time-series, possibly recursively
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceFilter(x, a = 1, b, zero.phase = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceFilter_+3A_x">x</code></td>
<td>
<p>a vector of numeric values, to be filtered as a time series.</p>
</td></tr>
<tr><td><code id="oceFilter_+3A_a">a</code></td>
<td>
<p>a vector of numeric values, giving the <code class="reqn">a</code> coefficients (see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="oceFilter_+3A_b">b</code></td>
<td>
<p>a vector of numeric values, giving the <code class="reqn">b</code> coefficients (see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="oceFilter_+3A_zero.phase">zero.phase</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to run the filter forwards,
and then backwards, thus removing any phase shifts associated with the
filter.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The filter is defined as e.g.  <code class="reqn">y[i]=b[1]*x[i] + </code><code class="reqn"> b[2]*x[i-1] + b[3]*x[i-2] + ... - a[2]*y[i-1] - a[3]*y[i-2] -
</code><code class="reqn"> a[4]*y[i-3] - ...</code>, where some of the illustrated terms will be omitted if the lengths of
<code>a</code> and <code>b</code> are too small, and terms are dropped at the start of
the time series where the index on <code>x</code> would be less than 1.
</p>
<p>By contrast with the <code><a href="stats.html#topic+filter">filter()</a></code> function of R, <code>oce.filter</code>
lacks the option to do a circular filter.  As a consequence,
<code>oceFilter</code> introduces a phase lag.  One way to remove this lag is to
run the filter forwards and then backwards, as in the &ldquo;Examples&rdquo;.
However, the result is still problematic, in the sense that applying it in
the reverse order would yield a different result.  (Matlab's <code>filtfilt</code>
shares this problem.)
</p>


<h3>Value</h3>

<p>A numeric vector of the filtered results, <code class="reqn">y</code>, as denoted in
&ldquo;Details&rdquo;.
</p>


<h3>Note</h3>

<p>The first value in the <code>a</code> vector is ignored, and if
<code>length(a)</code> equals 1, a non-recursive filter results.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
par(mar = c(4, 4, 1, 1))
b &lt;- rep(1, 5) / 5
a &lt;- 1
x &lt;- seq(0, 10)
y &lt;- ifelse(x == 5, 1, 0)
f1 &lt;- oceFilter(y, a, b)
plot(x, y, ylim = c(-0, 1.5), pch = "o", type = "b")
points(x, f1, pch = "x", col = "red")

# remove the phase lag
f2 &lt;- oceFilter(y, a, b, TRUE)
points(x, f2, pch = "+", col = "blue")

legend("topleft",
    col = c("black", "red", "blue"), pch = c("o", "x", "+"),
    legend = c("data", "normal filter", "zero-phase filter")
)
mtext("note that normal filter rolls off at end")
</code></pre>

<hr>
<h2 id='oceGetData'>Extract Something From the data Slot of an oce Object</h2><span id='topic+oceGetData'></span>

<h3>Description</h3>

<p>In contrast to the various <code>[[</code> functions, this is
guaranteed to look only within the <code>data</code> slot. If
the named item is not found, <code>NULL</code> is returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceGetData(object, name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceGetData_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="oceGetData_+3A_name">name</code></td>
<td>
<p>String indicating the name of the item to be found.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to the data slot: 
<code><a href="#topic+oceDeleteData">oceDeleteData</a>()</code>,
<code><a href="#topic+oceRenameData">oceRenameData</a>()</code>,
<code><a href="#topic+oceSetData">oceSetData</a>()</code>
</p>

<hr>
<h2 id='oceGetMetadata'>Extract Something From the metadata Slot of an oce Object</h2><span id='topic+oceGetMetadata'></span>

<h3>Description</h3>

<p>In contrast to the various <code>[[</code> functions, this is
guaranteed to look only within the <code>metadata</code> slot. If
the named item is not found, <code>NULL</code> is returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceGetMetadata(object, name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceGetMetadata_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="oceGetMetadata_+3A_name">name</code></td>
<td>
<p>String indicating the name of the item to be found.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to the metadata slot: 
<code><a href="#topic+oceDeleteMetadata">oceDeleteMetadata</a>()</code>,
<code><a href="#topic+oceRenameMetadata">oceRenameMetadata</a>()</code>,
<code><a href="#topic+oceSetMetadata">oceSetMetadata</a>()</code>
</p>

<hr>
<h2 id='oceMagic'>Find the Type of an Oceanographic Data File</h2><span id='topic+oceMagic'></span>

<h3>Description</h3>

<p><code>oceMagic</code> tries to infer the file type, based on the data
within the file, the file name, or a combination of the two.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceMagic(file, encoding = "latin1", debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceMagic_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to be checked.</p>
</td></tr>
<tr><td><code id="oceMagic_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="oceMagic_+3A_debug">debug</code></td>
<td>
<p>an integer, set non-zero to turn on debugging.  Higher values
indicate more debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>oceMagic</code> was previously called <code>oce.magic</code>, but that
alias was removed in version 0.9.24; see <a href="#topic+oce-defunct">oce-defunct</a>.
</p>


<h3>Value</h3>

<p>A character string indicating the file type, or <code>"unknown"</code>, if
the type cannot be determined. If the result contains <code>"/"</code> characters,
these separate a list describing the file type, with the first element being
the general type, the second element being the manufacturer, and the third
element being the manufacturer's name for the instrument. For example,
<code>"adp/nortek/aquadopp"</code> indicates a acoustic-doppler profiler made by
NorTek, of the model type called Aquadopp.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>This is used mainly by <code><a href="#topic+read.oce">read.oce()</a></code>.
</p>

<hr>
<h2 id='oceNames2whpNames'>Translate Oce Data Names to WHP Data Names</h2><span id='topic+oceNames2whpNames'></span>

<h3>Description</h3>

<p>Translate oce-style names to WOCE names, using <code><a href="base.html#topic+gsub">gsub()</a></code>
to match patterns. For example, the pattern <code>"oxygen"</code>
is taken to mean <code>"CTDOXY"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceNames2whpNames(names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceNames2whpNames_+3A_names">names</code></td>
<td>
<p>vector of strings holding oce-style names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of strings holding WHP-style names.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Several online sources list WHP names. An example is
<code style="white-space: pre;">&#8288;https://cchdo.github.io/hdo-assets/documentation/manuals/pdf/90_1/chap4.pdf&#8288;</code>
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that interpret variable names and units from headers: 
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+unitFromStringRsk">unitFromStringRsk</a>()</code>,
<code><a href="#topic+unitFromString">unitFromString</a>()</code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>
</p>

<hr>
<h2 id='ocePmatch'>Partial Matching of Strings or Numbers</h2><span id='topic+ocePmatch'></span><span id='topic+oce.pmatch'></span>

<h3>Description</h3>

<p>An extended version of <code><a href="base.html#topic+pmatch">pmatch()</a></code> that allows <code>x</code> to be
numeric or string-based.  As with <code><a href="base.html#topic+pmatch">pmatch()</a></code>, partial string
matches are handled.
This is a wrapper that is useful mainly for <code>which</code> arguments to
plotting functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ocePmatch(x, table, nomatch = NA_integer_, duplicates.ok = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ocePmatch_+3A_x">x</code></td>
<td>
<p>a code, or vector of codes.  This may be numeric, in which case it
is simply returned without further analysis of the other arguments, or it
may be string-based, in which case <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used to find
numeric matches.</p>
</td></tr>
<tr><td><code id="ocePmatch_+3A_table">table</code></td>
<td>
<p>a list that maps strings to numbers; <code><a href="base.html#topic+pmatch">pmatch()</a></code> is
used on <code>names(table)</code>.  If the name contains characters that are
normally not permitted in a variable name, use quotes, e.g.
<code>list(salinity=1, temperature=2, "salinity+temperature"=3)</code>.</p>
</td></tr>
<tr><td><code id="ocePmatch_+3A_nomatch">nomatch</code></td>
<td>
<p>value to be returned for cases of no match (passed to
<code><a href="base.html#topic+pmatch">pmatch()</a></code>.</p>
</td></tr>
<tr><td><code id="ocePmatch_+3A_duplicates.ok">duplicates.ok</code></td>
<td>
<p>code for the handling of duplicates (passed to
<code><a href="base.html#topic+pmatch">pmatch()</a></code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A number, or vector of numbers, corresponding to the matches.
Non-matches are indicated with <code>NA</code> values, or whatever value is given
by the <code>NA</code> argument.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Since <code><a href="base.html#topic+pmatch">pmatch()</a></code> is used for the actual matching, its
documentation should be consulted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
oce.pmatch(c("s", "at", "te"), list(salinity = 1, temperature = 3.1))
</code></pre>

<hr>
<h2 id='oceProject'>Wrapper to sf::sf_project()</h2><span id='topic+oceProject'></span>

<h3>Description</h3>

<p>This function is used to isolate other oce functions from
changes to the map-projection functions that are done in the <a href="https://CRAN.R-project.org/package=sf"><span class="pkg">sf</span></a>
package.  (Until 2020 December, the <code>rgdal</code> package was used,
after a year of tests ensuring that the results of the two packages were
the same.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceProject(xy, proj, inv = FALSE, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceProject_+3A_xy">xy</code></td>
<td>
<p>two-column numeric matrix specifying locations.  If <code>inv</code> is False, then <code>xy[,1]</code> will hold
longitude and <code>xy[,2]</code> will hold latitude, but if <code>inv</code> is True, then the columns will be easting
and northing values (in metres).</p>
</td></tr>
<tr><td><code id="oceProject_+3A_proj">proj</code></td>
<td>
<p>a character value specifying the desired map projection.  See
the <code>projection</code> parameter of <code><a href="#topic+mapPlot">mapPlot()</a></code> for details, including a
historical note dated 2023-04-11 about the now-deprecated <code>sp</code> package.</p>
</td></tr>
<tr><td><code id="oceProject_+3A_inv">inv</code></td>
<td>
<p>logical value, False by default, indicating whether an inverse projection is requested.</p>
</td></tr>
<tr><td><code id="oceProject_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-column matrix, with first column holding either
<code>longitude</code> or <code>x</code>, and second column holding either
<code>latitude</code> or <code>y</code>.
</p>

<hr>
<h2 id='oceRenameData'>Rename Something in the data slot of an oce Object</h2><span id='topic+oceRenameData'></span>

<h3>Description</h3>

<p>Rename an item within the <code>data</code> slot of an <a href="#topic+oce-class">oce</a> object, also changing
<code>dataNamesOriginal</code> in the <code>metadata</code> slot, so that the <code>[[</code> accessor will
still work with the original name that was stored in the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceRenameData(object, old, new, note = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceRenameData_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="oceRenameData_+3A_old">old</code></td>
<td>
<p>character value that matches the name of an item in <code>object</code>'s <code>data</code> slot.</p>
</td></tr>
<tr><td><code id="oceRenameData_+3A_new">new</code></td>
<td>
<p>character value to be used as the new name that matches the name of an item in
<code>object</code>'s <code>data</code> slot. Thus must not be the name of something that is already in the <code>data</code>
slot. If <code>new</code> is the  same as <code>old</code>, then the object is returned unaltered.</p>
</td></tr>
<tr><td><code id="oceRenameData_+3A_note">note</code></td>
<td>
<p>character value that holds an explanation of the reason for the change. If this
is a string of non-zero length, then this is inserted in the processing log of the returned
value. If it is <code>NULL</code>, then no entry is added to the processing log.  Otherwise, the processing
log gets a new item that is constructed from the function call.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to the data slot: 
<code><a href="#topic+oceDeleteData">oceDeleteData</a>()</code>,
<code><a href="#topic+oceGetData">oceGetData</a>()</code>,
<code><a href="#topic+oceSetData">oceSetData</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
CTD &lt;- oceRenameData(ctd, "salinity", "SALT")
stopifnot(all.equal(ctd[["salinity"]], CTD[["SALT"]]))
stopifnot(all.equal(ctd[["sal00"]], CTD[["SALT"]]))

</code></pre>

<hr>
<h2 id='oceRenameMetadata'>Rename Something in the metadata Slot of an oce Object</h2><span id='topic+oceRenameMetadata'></span>

<h3>Description</h3>

<p>Rename an item within the <code>metadata</code> slot of an <a href="#topic+oce-class">oce</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceRenameMetadata(object, old, new, note = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceRenameMetadata_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="oceRenameMetadata_+3A_old">old</code></td>
<td>
<p>character value that matches the name of an item in <code>object</code>'s <code>metadata</code> slot.</p>
</td></tr>
<tr><td><code id="oceRenameMetadata_+3A_new">new</code></td>
<td>
<p>character value to be used as the new name that matches the name of an item in
<code>object</code>'s <code>metadata</code> slot. Thus must not be the name of something that is already in the <code>metadata</code>
slot. If <code>new</code> is the  same as <code>old</code>, then the object is returned unaltered.</p>
</td></tr>
<tr><td><code id="oceRenameMetadata_+3A_note">note</code></td>
<td>
<p>character value that holds an explanation of the reason for the change. If this
is a string of non-zero length, then this is inserted in the processing log of the returned
value. If it is <code>NULL</code>, then no entry is added to the processing log.  Otherwise, the processing
log gets a new item that is constructed from the function call.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to the metadata slot: 
<code><a href="#topic+oceDeleteMetadata">oceDeleteMetadata</a>()</code>,
<code><a href="#topic+oceGetMetadata">oceGetMetadata</a>()</code>,
<code><a href="#topic+oceSetMetadata">oceSetMetadata</a>()</code>
</p>

<hr>
<h2 id='oceSetData'>Set Something in the data Slot of an oce Object</h2><span id='topic+oceSetData'></span>

<h3>Description</h3>

<p>Create a copy of an object in which some element of its
<code>data</code> slot has been altered, or added.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceSetData(object, name, value, unit, originalName, note = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceSetData_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="oceSetData_+3A_name">name</code></td>
<td>
<p>String indicating the name of the <code>data</code> item to be set.</p>
</td></tr>
<tr><td><code id="oceSetData_+3A_value">value</code></td>
<td>
<p>Value for the item.</p>
</td></tr>
<tr><td><code id="oceSetData_+3A_unit">unit</code></td>
<td>
<p>An optional indication of the units for the item. This has
three possible forms (see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="oceSetData_+3A_originalname">originalName</code></td>
<td>
<p>Optional character string giving an 'original' name (e.g.
as stored in the header of a data file).</p>
</td></tr>
<tr><td><code id="oceSetData_+3A_note">note</code></td>
<td>
<p>Either empty (the default), a character string, or <code>NULL</code>,
to control additions made to the processing log of the return value. If
<code>note=""</code> then the an entry is created based on deparsing the function call.
If <code>note</code> is a non-empty string, then that string gets added added
to the processing log. Finally, if <code>note=NULL</code>, then nothing is
added to the processing log. This last form is useful in cases where
<code>oceSetData</code> is to be called many times in succession, resulting
in an overly verbose processing log; in such cases, it might help
to add a note by e.g. <code>processingLog(a) &lt;- "QC (memo dek-2018-01/31)"</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The trickiest argument to set is the <code>unit</code>.  There are three
possibilities for this:
</p>

<ol>
<li> <p><code>unit</code> is a named or unnamed <code><a href="base.html#topic+list">list()</a></code> that contains two items.
If the list is named, the names must be
<code>unit</code> and <code>scale</code>. If the list is unnamed, the stated names are assigned
to the items, in the stated order. Either way, the <code>unit</code>
item must be an <code><a href="base.html#topic+expression">expression()</a></code> that specifies the unit,
and the <code>scale</code> item must be a string that describes the scale. For
example, modern temperatures have
<code>unit=list(unit=expression(degree*C), scale="ITS-90")</code>.
</p>
</li>
<li> <p><code>unit</code> is an <code><a href="base.html#topic+expression">expression()</a></code> giving the unit as above. In this
case, the scale will be set to <code>""</code>.
</p>
</li>
<li> <p><code>unit</code> is a character string that is converted
into an expression with <a href="base.html#topic+parse">parse</a><code>(text=unit)</code>,
and the scale set to <code>""</code>.
</p>
</li></ol>



<h3>Value</h3>

<p>An <a href="#topic+oce-class">oce</a> object, the <code>data</code> slot of which has
been altered either by adding a new item or modifying an existing
item.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to the data slot: 
<code><a href="#topic+oceDeleteData">oceDeleteData</a>()</code>,
<code><a href="#topic+oceGetData">oceGetData</a>()</code>,
<code><a href="#topic+oceRenameData">oceRenameData</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ctd)
Tf &lt;- swTFreeze(ctd)
ctd &lt;- oceSetData(ctd, "freezing", Tf,
    unit = list(unit = expression(degree * C), scale = "ITS-90")
)
plotProfile(ctd, "freezing")

</code></pre>

<hr>
<h2 id='oceSetMetadata'>Set Something in the metadata Slot of an oce Object</h2><span id='topic+oceSetMetadata'></span>

<h3>Description</h3>

<p>Create a copy of an object in which some element of its
<code>metadata</code> slot has been altered, or added.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceSetMetadata(object, name, value, note = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceSetMetadata_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="oceSetMetadata_+3A_name">name</code></td>
<td>
<p>String indicating the name of the <code>metadata</code> item to be set.</p>
</td></tr>
<tr><td><code id="oceSetMetadata_+3A_value">value</code></td>
<td>
<p>Value for the item.</p>
</td></tr>
<tr><td><code id="oceSetMetadata_+3A_note">note</code></td>
<td>
<p>Either empty (the default), a character string, or <code>NULL</code>,
to control additions made to the processing log of the return value. If
<code>note=""</code> then an entry is created based on deparsing the function call.
If <code>note</code> is a non-empty string, then that string gets added added
to the processing log. Finally, if <code>note=NULL</code>, then nothing is
added to the processing log.  This last form is useful in cases where
<code>oceSetData</code> is to be called many times in succession, resulting
in an overly verbose processing log; in which case, it might helpful
to use <a href="#topic+processingLog+3C-">processingLog&lt;-</a> to add a summary entry to the object's
processing log.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+oce-class">oce</a> object, the <code>metadata</code> slot of which has
been altered either by adding a new item or modifying an existing
item.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to the metadata slot: 
<code><a href="#topic+oceDeleteMetadata">oceDeleteMetadata</a>()</code>,
<code><a href="#topic+oceGetMetadata">oceGetMetadata</a>()</code>,
<code><a href="#topic+oceRenameMetadata">oceRenameMetadata</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Add an estimate of MLD (mixed layer depth) to a ctd object
library(oce)
data(ctd)
ctdWithMLD &lt;- oceSetMetadata(ctd, "MLD", 3)
ctdWithMLD[["MLD"]] # 3

</code></pre>

<hr>
<h2 id='oceSmooth'>Smooth an oce Object</h2><span id='topic+oceSmooth'></span><span id='topic+oce.smooth'></span>

<h3>Description</h3>

<p>Each data element is smoothed as a timeseries. For ADP data, this is done
along time, not distance.  Time vectors, if any, are not smoothed.  A good
use of <code>oce.smooth</code> is for despiking noisy data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceSmooth(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceSmooth_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="oceSmooth_+3A_...">...</code></td>
<td>
<p>parameters to be supplied to <code><a href="stats.html#topic+smooth">smooth()</a></code>, which does
the actual work.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+oce-class">oce</a> object that has been smoothed appropriately.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The work is done with <code><a href="stats.html#topic+smooth">smooth()</a></code>, and the <code>...</code>
arguments are handed to it directly by <code>oce.smooth</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
d &lt;- oce.smooth(ctd)
plot(d)
</code></pre>

<hr>
<h2 id='oceSpectrum'>Normalize a Spectrum</h2><span id='topic+oceSpectrum'></span><span id='topic+oce.spectrum'></span>

<h3>Description</h3>

<p>This is a wrapper around the R <code><a href="stats.html#topic+spectrum">spectrum()</a></code> function, which
returns spectral values that are adjusted so that the integral of those
values equals the variance of the input <code>x</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceSpectrum(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceSpectrum_+3A_x">x</code></td>
<td>
<p>a univariate or multivariate time series, as for <code><a href="stats.html#topic+spectrum">spectrum()</a></code>.</p>
</td></tr>
<tr><td><code id="oceSpectrum_+3A_...">...</code></td>
<td>
<p>extra arguments passed on to <code><a href="stats.html#topic+spectrum">spectrum()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A spectrum that has values that integrate to the variance.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+spectrum">spectrum()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- rnorm(1e3)
s &lt;- spectrum(x, plot = FALSE)
ss &lt;- oce.spectrum(x, plot = FALSE)
cat("variance of x=", var(x), "\n")
cat("integral of     spectrum=", sum(s$spec) * diff(s$freq[1:2]), "\n")
cat("integral of oce.spectrum=", sum(ss$spec) * diff(ss$freq[1:2]), "\n")
</code></pre>

<hr>
<h2 id='oceUnits2whpUnits'>Translate oce Unit to WHP Unit</h2><span id='topic+oceUnits2whpUnits'></span>

<h3>Description</h3>

<p>Translate oce units to WHP-style strings,
to match patterns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oceUnits2whpUnits(units, scales)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oceUnits2whpUnits_+3A_units">units</code></td>
<td>
<p>vector of expressions for units in oce notation.</p>
</td></tr>
<tr><td><code id="oceUnits2whpUnits_+3A_scales">scales</code></td>
<td>
<p>vector of strings for scales in oce notation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of strings holding WOCE-style names.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Several online sources list WOCE names. An example is
<code style="white-space: pre;">&#8288;https://cchdo.github.io/hdo-assets/documentation/manuals/pdf/90_1/chap4.pdf&#8288;</code>
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that interpret variable names and units from headers: 
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+unitFromStringRsk">unitFromStringRsk</a>()</code>,
<code><a href="#topic+unitFromString">unitFromString</a>()</code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>
</p>

<hr>
<h2 id='odf-class'>Class to Store ODF Data</h2><span id='topic+odf-class'></span>

<h3>Description</h3>

<p>This class is for data stored in a format used at Canadian
Department of Fisheries and Oceans laboratories. It is somewhat
similar to the <a href="#topic+bremen-class">bremen</a> class, in the sense
that it does not apply just to a particular instrument.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>odf</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>odf</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>odf</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+odf-class">odf</a> objects (see <code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+odf-class">odf</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Anthony W. Isenor and David Kellow, 2011. <em>ODF Format Specification
Version 2.0</em>. (This is a .doc file obtained in June 2011 by Dan Kelley,
which no longer seems to be made available at any DFO website.)
</p>
</li>
<li><p> (Unknown authors), October 2014. <em>ODF Format Description (MLI)</em>,
<code style="white-space: pre;">&#8288;https://ogsl.ca/wp-content/uploads/ODF_format_desc_en_0.pdf&#8288;</code>,
(Link worked early on March 16, 2022, but failed later that day.)
</p>
</li>
<li><p> A sample ODF file in the DFO format is available at
<code>system.file("extdata","CTD_BCD2014666_008_1_DN.ODF.gz",package="oce")</code>
</p>
</li>
<li><p> A sample ODF file in the MLI format may be available at
<code style="white-space: pre;">&#8288;https://ogsl.ca/wp-content/uploads/ODF_file_example_en_0.pdf&#8288;</code>.
(Link worked early on March 16, 2022, but failed later that day.)
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to odf data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ODF2oce">ODF2oce</a>()</code>,
<code><a href="#topic+ODFListFromHeader">ODFListFromHeader</a>()</code>,
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.odf">read.odf</a>()</code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+summary+2Codf-method">summary,odf-method</a></code>
</p>
<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>

<hr>
<h2 id='ODF2oce'>Create ODF Object From Output of read_ODF in ODF package</h2><span id='topic+ODF2oce'></span>

<h3>Description</h3>

<p>As of August 11, 2015, <code>ODF::read_ODF</code> returns a list with 9 elements,
one named <code>DATA</code>, which is a <code><a href="base.html#topic+data.frame">data.frame()</a></code> containing the
columnar data, the others being headers of various sorts.  The present
function constructs an oce object from such data, facilitating processing
and plotting with the general oce functions.
This involves storing the 8 headers verbatim in the
<code>odfHeaders</code> in the <code>metadata</code> slot, and also
copying some of the header
information into more standard names (e.g.  <code>metadata@longitude</code> is a
copy of <code>metadata@odfHeader$EVENT_HEADER$INITIAL_LATITUDE</code>).  As for
the <code>DATA</code>, they are stored in the <code>data</code> slot, after renaming
from ODF to oce convention using <code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ODF2oce(ODF, coerce = TRUE, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ODF2oce_+3A_odf">ODF</code></td>
<td>
<p>A list as returned by <code>read_ODF</code> in the <code>ODF</code> package</p>
</td></tr>
<tr><td><code id="ODF2oce_+3A_coerce">coerce</code></td>
<td>
<p>A logical value indicating whether to coerce the return value
to an appropriate object type, if possible.</p>
</td></tr>
<tr><td><code id="ODF2oce_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An oce object, possibly coerced to a subtype.
</p>


<h3>Caution</h3>

<p>This function may change as the <code>ODF</code> package
changes.  Since <code>ODF</code> has not been released yet, this should not affect
any users except those involved in the development of <code>oce</code> and
<code>ODF</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to odf data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ODFListFromHeader">ODFListFromHeader</a>()</code>,
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.odf">read.odf</a>()</code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+summary+2Codf-method">summary,odf-method</a></code>
</p>

<hr>
<h2 id='ODFListFromHeader'>Create a List of odf Header Metadata</h2><span id='topic+ODFListFromHeader'></span>

<h3>Description</h3>

<p>Create a List of odf Header Metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ODFListFromHeader(header)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ODFListFromHeader_+3A_header">header</code></td>
<td>
<p>Vector of character strings, holding the header</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list holding the metadata, with item names matching
those in the ODF header, except that duplicates are transformed
through the use of <code><a href="#topic+unduplicateNames">unduplicateNames()</a></code>.
</p>


<h3>See Also</h3>

<p>Other things related to odf data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ODF2oce">ODF2oce</a>()</code>,
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.odf">read.odf</a>()</code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+summary+2Codf-method">summary,odf-method</a></code>
</p>

<hr>
<h2 id='ODFNames2oceNames'>Translate ODF CODE Strings to oce Variable Names</h2><span id='topic+ODFNames2oceNames'></span>

<h3>Description</h3>

<p>Translate ODF CODE strings to oce variable names. This
is done differently for data names and quality-control
(QC) names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ODFNames2oceNames(
  ODFnames,
  columns = NULL,
  PARAMETER_HEADER = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ODFNames2oceNames_+3A_odfnames">ODFnames</code></td>
<td>
<p>vector of character values that hold ODF names.</p>
</td></tr>
<tr><td><code id="ODFNames2oceNames_+3A_columns">columns</code></td>
<td>
<p>Optional list containing name correspondances, as described for
<code><a href="#topic+read.ctd.odf">read.ctd.odf()</a></code>.</p>
</td></tr>
<tr><td><code id="ODFNames2oceNames_+3A_parameter_header">PARAMETER_HEADER</code></td>
<td>
<p>Optional list containing information on the data variables.</p>
</td></tr>
<tr><td><code id="ODFNames2oceNames_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following table gives the recognized ODF code names for variables,
along with the translated names as used in oce objects. Note that the
code names are appended with strings such as <code>"_01"</code>, <code>"_02"</code>, etc,
for repeats. The converted name for an <code>"_01"</code> item is as shown below,
and for e.g. <code>"_02"</code> a suffix 2 is added to the oce name, etc.
</p>
<p>QC items (which get stored as <code>flags</code> in object's
<code>metadata</code> slots) are assigned names that match those of the
parameters to which they refer.  In parsing ODF files, it is assumed
that QC items refer to the data items that precede them.  This pattern
does not seem to be documented, but it has held in all the files
examined by the author, and a similar assumption is made in other
software systems.  QC items have <code>CODE</code> values that are
either start with <code>"QQQQ"</code> or equal <code>"Q&lt;CODE&gt;"</code>,
where <code style="white-space: pre;">&#8288;&lt;CODE&gt;&#8288;</code> matches the corresponding data item.
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>ODF Code</strong> </td><td style="text-align: left;"> <strong>Oce Name</strong>      </td><td style="text-align: left;"> <strong>Notes</strong>                                    </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ABSH</code> </td><td style="text-align: left;"> <code>humidityAbsolute</code>      </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ACO2</code> </td><td style="text-align: left;"> <code>CO2Atmosphere</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ALKW</code> </td><td style="text-align: left;"> <code>alkalinity</code>            </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ALKY</code> </td><td style="text-align: left;"> <code>alkalinityTotal</code>       </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ALP0</code> </td><td style="text-align: left;"> <code>apha0</code>                 </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ALTB</code> </td><td style="text-align: left;"> <code>altimeter</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ALTS</code> </td><td style="text-align: left;"> <code>altitude</code>              </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>AMON</code> </td><td style="text-align: left;"> <code>ammonium</code>              </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ATMP</code> </td><td style="text-align: left;"> <code>pressureAtmosphere</code>    </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ATMS</code> </td><td style="text-align: left;"> <code>pressureAtmosphereSealevel</code> </td><td style="text-align: left;">                                         </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ATRK</code> </td><td style="text-align: left;"> <code>alongTrackDisplacement</code> </td><td style="text-align: left;">                                             </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ATTU</code> </td><td style="text-align: left;"> <code>attenuation</code>           </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>AUTH</code> </td><td style="text-align: left;"> <code>authority</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>BATH</code> </td><td style="text-align: left;"> <code>barometricDepth</code>       </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>BATT</code> </td><td style="text-align: left;"> <code>batteryVoltage</code>        </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>BEAM</code> </td><td style="text-align: left;"> <code>a</code>                     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>BNO7</code> </td><td style="text-align: left;"> <code>bestNODC7Number</code>       </td><td style="text-align: left;"> That is an "oh" letter, not a zero           </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CALK</code> </td><td style="text-align: left;"> <code>carbonateAlkalinity</code>   </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CHLR</code> </td><td style="text-align: left;"> <code>chlorinity</code>            </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CHLS</code> </td><td style="text-align: left;"> <code>chlorosity</code>            </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CNDC</code> </td><td style="text-align: left;"> <code>conductivity</code>          </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CNTR</code> </td><td style="text-align: left;"> <code>scan</code>                  </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>COND</code> </td><td style="text-align: left;"> <code>conductivity</code>          </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CORG</code> </td><td style="text-align: left;"> <code>carbonOrganic</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CPHL</code> </td><td style="text-align: left;"> <code>chlorophyll</code>           </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CRAT</code> </td><td style="text-align: left;"> <code>conductivity</code>          </td><td style="text-align: left;"> Conductivity ratio (may have spurious unit)  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CMNT</code> </td><td style="text-align: left;"> <code>comment</code>               </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CNDC</code> </td><td style="text-align: left;"> <code>conductivity</code>          </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>COND</code> </td><td style="text-align: left;"> <code>conductivity</code>          </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CTOT</code> </td><td style="text-align: left;"> <code>carbonTotal</code>           </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DCHG</code> </td><td style="text-align: left;"> <code>discharge</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DENS</code> </td><td style="text-align: left;"> <code>density</code>               </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DEPH</code> </td><td style="text-align: left;"> <code>pressure</code>              </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DEWT</code> </td><td style="text-align: left;"> <code>temperatureDewpoint</code>   </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DOC_</code> </td><td style="text-align: left;"> <code>carbonOrganicDissolved</code> </td><td style="text-align: left;">                                             </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DON_</code> </td><td style="text-align: left;"> <code>nitrogenOrganicDissolved</code> </td><td style="text-align: left;">                                           </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DOXY</code> </td><td style="text-align: left;"> <code>oxygen</code>                </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DPDT</code> </td><td style="text-align: left;"> <code>dpdt</code>                  </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DRDP</code> </td><td style="text-align: left;"> <code>drogueDepth</code>           </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DPWT</code> </td><td style="text-align: left;"> <code>dryWeight</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DRYT</code> </td><td style="text-align: left;"> <code>temperatureDryBulb</code>    </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DYNH</code> </td><td style="text-align: left;"> <code>dynamicHeight</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ERRV</code> </td><td style="text-align: left;"> <code>errorVelocity</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>EWCM</code> </td><td style="text-align: left;"> <code>uMagnetic</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>EWCT</code> </td><td style="text-align: left;"> <code>u</code>                     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>FFFF</code> </td><td style="text-align: left;"> <code>overall(FFFF)</code>         </td><td style="text-align: left;"> Archaic overall flag, replaced by <code>QCFF</code>     </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>FLOR</code> </td><td style="text-align: left;"> <code>fluorometer</code>           </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>GDIR</code> </td><td style="text-align: left;"> <code>windDirectionGust</code>     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>GEOP</code> </td><td style="text-align: left;"> <code>geopotential</code>          </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>GSPD</code> </td><td style="text-align: left;"> <code>windSpeedGust</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>HCDM</code> </td><td style="text-align: left;"> <code>directionMagnetic</code>     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>HCDT</code> </td><td style="text-align: left;"> <code>directionTrue</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>HCSP</code> </td><td style="text-align: left;"> <code>speedHorizontal</code>       </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>HEAD</code> </td><td style="text-align: left;"> <code>heading</code>               </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>HSUL</code> </td><td style="text-align: left;"> <code>hydrogenSulphide</code>      </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>IDEN</code> </td><td style="text-align: left;"> <code>sampleNumber</code>          </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>LABT</code> </td><td style="text-align: left;"> <code>temperatureLaboratory</code> </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>LATD</code> </td><td style="text-align: left;"> <code>latitude</code>              </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>LHIS</code> </td><td style="text-align: left;"> <code>lifeHistory</code>           </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>LOND</code> </td><td style="text-align: left;"> <code>longitude</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>LPHT</code> </td><td style="text-align: left;"> <code>pHLaboratory</code>          </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>MNSV</code> </td><td style="text-align: left;"> <code>retentionFilterSize</code>   </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>MNSZ</code> </td><td style="text-align: left;"> <code>organismSizeMinimum</code>   </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>MODF</code> </td><td style="text-align: left;"> <code>additionalTaxonomicInformation</code> </td><td style="text-align: left;">                                     </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>MXSZ</code> </td><td style="text-align: left;"> <code>organismSizeMaximum</code>   </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>NETR</code> </td><td style="text-align: left;"> <code>netSolarRadiation</code>     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>NONE</code> </td><td style="text-align: left;"> <code>noWMOcode</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>NORG</code> </td><td style="text-align: left;"> <code>nitrogenOrganic</code>       </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>NSCM</code> </td><td style="text-align: left;"> <code>vMagnetic</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>NSCT</code> </td><td style="text-align: left;"> <code>v</code>                     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>NTOT</code> </td><td style="text-align: left;"> <code>nitrogenTotal</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>NTRA</code> </td><td style="text-align: left;"> <code>nitrate</code>               </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>NTRI</code> </td><td style="text-align: left;"> <code>nitrite</code>               </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>NTRZ</code> </td><td style="text-align: left;"> <code>nitrite+nitrate</code>       </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>NUM_</code> </td><td style="text-align: left;"> <code>scansPerAverage</code>       </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>OBKS</code> </td><td style="text-align: left;"> <code>turbidity</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>OCUR</code> </td><td style="text-align: left;"> <code>oxygenCurrent</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>OPPR</code> </td><td style="text-align: left;"> <code>oxygenPartialPressure</code> </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>OSAT</code> </td><td style="text-align: left;"> <code>oxygenSaturation</code>      </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>OTMP</code> </td><td style="text-align: left;"> <code>oxygenTemperature</code>     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>OXYG</code> </td><td style="text-align: left;"> <code>oxygenDissolved</code>       </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>OXYM</code> </td><td style="text-align: left;"> <code>oxygenDissolved</code>       </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>OXYV</code> </td><td style="text-align: left;"> <code>oxygenVoltage</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>OXV_</code> </td><td style="text-align: left;"> <code>oxygenVoltageRaw</code>      </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PCO2</code> </td><td style="text-align: left;"> <code>CO2</code>                   </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PHA_</code> </td><td style="text-align: left;"> <code>phaeopigment</code>          </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PHOS</code> </td><td style="text-align: left;"> <code>phosphate</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PHPH</code> </td><td style="text-align: left;"> <code>pH</code>                    </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PHT_</code> </td><td style="text-align: left;"> <code>pHTotal</code>               </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PIM_</code> </td><td style="text-align: left;"> <code>particulateInorganicMatter</code> </td><td style="text-align: left;">                                         </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PHY_</code> </td><td style="text-align: left;"> <code>phytoplanktonCount</code>    </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>POC_</code> </td><td style="text-align: left;"> <code>particulateOrganicCarbon</code> </td><td style="text-align: left;">                                           </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>POM_</code> </td><td style="text-align: left;"> <code>particulateOrganicMatter</code> </td><td style="text-align: left;">                                           </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PON_</code> </td><td style="text-align: left;"> <code>particulateOrganicNitrogen</code> </td><td style="text-align: left;">                                         </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>POTM</code> </td><td style="text-align: left;"> <code>theta</code>                 </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PRES</code> </td><td style="text-align: left;"> <code>pressure</code>              </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PSAL</code> </td><td style="text-align: left;"> <code>salinity</code>              </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PSAR</code> </td><td style="text-align: left;"> <code>PSAR</code>                  </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PTCH</code> </td><td style="text-align: left;"> <code>pitch</code>                 </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>QCFF</code> </td><td style="text-align: left;"> <code>overall(QCFF)</code>         </td><td style="text-align: left;"> Overall flag (see also archaic FFFF)         </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>RANG</code> </td><td style="text-align: left;"> <code>range</code>                 </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>REFR</code> </td><td style="text-align: left;"> <code>reference</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>RELH</code> </td><td style="text-align: left;"> <code>humidityRelative</code>      </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>RELP</code> </td><td style="text-align: left;"> <code>relativeTotalPressure</code> </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ROLL</code> </td><td style="text-align: left;"> <code>roll</code>                  </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SDEV</code> </td><td style="text-align: left;"> <code>standardDeviation</code>     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SECC</code> </td><td style="text-align: left;"> <code>SecchiDepth</code>           </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SEX_</code> </td><td style="text-align: left;"> <code>sex</code>                   </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SIG0</code> </td><td style="text-align: left;"> <code>sigma0</code>                </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SIGP</code> </td><td style="text-align: left;"> <code>sigmaTheta</code>            </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SIGT</code> </td><td style="text-align: left;"> <code>sigmat</code>                </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SLCA</code> </td><td style="text-align: left;"> <code>silicate</code>              </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SNCN</code> </td><td style="text-align: left;"> <code>scanCounter</code>           </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SPAR</code> </td><td style="text-align: left;"> <code>SPAR</code>                  </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SPEH</code> </td><td style="text-align: left;"> <code>humiditySpecific</code>      </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SPFR</code> </td><td style="text-align: left;"> <code>sampleFraction</code>        </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SPVO</code> </td><td style="text-align: left;"> <code>specificVolume</code>        </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SPVA</code> </td><td style="text-align: left;"> <code>specificVolumeAnomaly</code> </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>STRA</code> </td><td style="text-align: left;"> <code>stressAmplitude</code>       </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>STRD</code> </td><td style="text-align: left;"> <code>stressDirection</code>       </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>STRU</code> </td><td style="text-align: left;"> <code>stressU</code>               </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>STRV</code> </td><td style="text-align: left;"> <code>stressV</code>               </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SSAL</code> </td><td style="text-align: left;"> <code>salinity</code>              </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SVEL</code> </td><td style="text-align: left;"> <code>soundVelocity</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SYTM</code> </td><td style="text-align: left;"> <code>time</code>                  </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TAXN</code> </td><td style="text-align: left;"> <code>taxonomicName</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TE90</code> </td><td style="text-align: left;"> <code>temperature</code>           </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TEMP</code> </td><td style="text-align: left;"> <code>temperature</code>           </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TEXZT</code> </td><td style="text-align: left;"> <code>text</code>                 </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TICW</code> </td><td style="text-align: left;"> <code>totalInorganicCarbon</code>  </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TILT</code> </td><td style="text-align: left;"> <code>tilt</code>                  </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TOTP</code> </td><td style="text-align: left;"> <code>pressureAbsolute</code>      </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TPHS</code> </td><td style="text-align: left;"> <code>phosphorousTotal</code>      </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TRAN</code> </td><td style="text-align: left;"> <code>lightTransmission</code>     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TRB_</code> </td><td style="text-align: left;"> <code>turbidity</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TRBH</code> </td><td style="text-align: left;"> <code>trophicDescriptor</code>     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TSM_</code> </td><td style="text-align: left;"> <code>suspendedMatterTotal</code>  </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TSN_</code> </td><td style="text-align: left;"> <code>taxonomicSerialNumber</code> </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TURB</code> </td><td style="text-align: left;"> <code>turbidity</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>UNKN</code> </td><td style="text-align: left;"> <code>-</code>                     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>UREA</code> </td><td style="text-align: left;"> <code>urea</code>                  </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>VAIS</code> </td><td style="text-align: left;"> <code>BVFrequency</code>           </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>VCSP</code> </td><td style="text-align: left;"> <code>w</code>                     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>VMXL</code> </td><td style="text-align: left;"> <code>waveHeightMaximum</code>     </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>VRMS</code> </td><td style="text-align: left;"> <code>waveHeightMean</code>        </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>VTCA</code> </td><td style="text-align: left;"> <code>wavePeriod</code>            </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>WDIR</code> </td><td style="text-align: left;"> <code>windDirection</code>         </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>WETT</code> </td><td style="text-align: left;"> <code>temperatureWetBulb</code>    </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>WSPD</code> </td><td style="text-align: left;"> <code>windSpeed</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>WTWT</code> </td><td style="text-align: left;"> <code>wetWeight</code>             </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ZOO_</code> </td><td style="text-align: left;"> <code>zooplanktonCount</code>      </td><td style="text-align: left;">                                              </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Any code not shown in the list is transferred to the oce object without renaming, apart from
the adjustment of suffix numbers. The following code have been seen in data files from
the Bedford Institute of Oceanography: <code>ALTB</code>, <code>PHPH</code> and <code>QCFF</code>.
</p>


<h3>Value</h3>

<p>A list relating ODF names to oce names (see &ldquo;Examples&rdquo;).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>For sources that describe the ODF format, see the documentation
for the <a href="#topic+odf-class">odf</a>.
</p>


<h3>See Also</h3>

<p>Other functions that interpret variable names and units from headers: 
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+unitFromStringRsk">unitFromStringRsk</a>()</code>,
<code><a href="#topic+unitFromString">unitFromString</a>()</code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>
</p>
<p>Other things related to odf data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ODF2oce">ODF2oce</a>()</code>,
<code><a href="#topic+ODFListFromHeader">ODFListFromHeader</a>()</code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.odf">read.odf</a>()</code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+summary+2Codf-method">summary,odf-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ODFNames2oceNames("TEMP_01")$names # "temperature"

</code></pre>

<hr>
<h2 id='parseLatLon'>Parse a Latitude or Longitude String</h2><span id='topic+parseLatLon'></span>

<h3>Description</h3>

<p>Parse a latitude or longitude string, e.g. as in the header of a CTD file
The following formats are understood (for, e.g. latitude):
</p>
<div class="sourceCode"><pre>** NMEA Latitude = 47 54.760 N
** Latitude: 47 53.27 N
</pre></div>
<p>Note that <code><a href="base.html#topic+iconv">iconv()</a></code> is called to convert the string to ASCII before
decoding, to change any degree (or other non-ASCII) symbols to blanks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseLatLon(line, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parseLatLon_+3A_line">line</code></td>
<td>
<p>a character string containing an indication of latitude or
longitude.</p>
</td></tr>
<tr><td><code id="parseLatLon_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numerical value of latitude or longitude.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Used by <code><a href="#topic+read.ctd">read.ctd()</a></code>.
</p>

<hr>
<h2 id='plot+2Cadp-method'>Plot an adp Object</h2><span id='topic+plot+2Cadp-method'></span><span id='topic+plot.adp'></span>

<h3>Description</h3>

<p>Create a summary plot of data measured by an acoustic Doppler profiler.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adp'
plot(
  x,
  which,
  j,
  col,
  breaks,
  zlim,
  titles,
  lwd = par("lwd"),
  type = "l",
  ytype = c("profile", "distance"),
  drawTimeRange = getOption("oceDrawTimeRange"),
  useSmoothScatter,
  missingColor = "gray",
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1.5, mgp[1] + 1.5, 1.5, 1.5),
  mai.palette = rep(0, 4),
  tformat,
  marginsAsImage = FALSE,
  cex = par("cex"),
  cex.axis = par("cex.axis"),
  cex.lab = par("cex.lab"),
  xlim,
  ylim,
  control,
  useLayout = FALSE,
  coastline = "coastlineWorld",
  span = 300,
  main = "",
  grid = FALSE,
  grid.col = "darkgray",
  grid.lty = "dotted",
  grid.lwd = 1,
  xlab = NULL,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cadp-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_which">which</code></td>
<td>
<p>list of desired plot types.  These are graphed in panels
running down from the top of the page.  If <code>which</code> is not given,
the plot will show images of the distance-time dependence of velocity
for each beam. See &ldquo;Details&rdquo; for the meanings of various values of <code>which</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_j">j</code></td>
<td>
<p>optional string specifying a sub-class of <code>which</code>. For
Nortek Aquadopp profilers, this may either be <code>"default"</code> (or missing)
to get the main signal, or <code>"diagnostic"</code> to get a diagnostic
signal.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_col">col</code></td>
<td>
<p>optional indication of color(s) to use.  If not provided, the
default for images is <code>oce.colorsPalette(128,1)</code>, and for lines and
points is black.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_breaks">breaks</code></td>
<td>
<p>optional breaks for color scheme</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_zlim">zlim</code></td>
<td>
<p>a range to be used as the <code>zlim</code> parameter to the
<code><a href="#topic+imagep">imagep()</a></code> call that is used to create the image.  If omitted,
<code>zlim</code> is set for each panel individually, to encompass the data of the
panel and to be centred around zero.  If provided as a two-element vector,
then that is used for each panel.  If provided as a two-column matrix, then
each panel of the graph uses the corresponding row of the matrix; for
example, setting <code>zlim=rbind(c(-1,1),c(-1,1),c(-.1,.1))</code> might make
sense for <code>which=1:3</code>, so that the two horizontal velocities have one
scale, and the smaller vertical velocity has another.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_titles">titles</code></td>
<td>
<p>optional vector of character strings to be used as labels for
the plot panels.  For images, these strings will be placed in the right hand
side of the top margin.  For timeseries, these strings are ignored.  If this
is provided, its length must equal that of <code>which</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_lwd">lwd</code></td>
<td>
<p>if the plot is of a time-series or scattergraph format with
lines, this is used in the usual way; otherwise, e.g. for image formats,
this is ignored.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_type">type</code></td>
<td>
<p>if the plot is of a time-series or scattergraph format, this is
used in the usual way, e.g. <code>"l"</code> for lines, etc.; otherwise, as for
image formats, this is ignored.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_ytype">ytype</code></td>
<td>
<p>character string controlling the type of the y axis for images
(ignored for time series).  If <code>"distance"</code>, then the y axis will be
distance from the sensor head, with smaller distances nearer the bottom of
the graph.  If <code>"profile"</code>, then this will still be true for
upward-looking instruments, but the y axis will be flipped for
downward-looking instruments, so that in either case, the top of the graph
will represent the sample nearest the sea surface.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_drawtimerange">drawTimeRange</code></td>
<td>
<p>boolean that applies to panels with time as the
horizontal axis, indicating whether to draw the time range in the top-left
margin of the plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_usesmoothscatter">useSmoothScatter</code></td>
<td>
<p>boolean that indicates whether to use
<code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code> in various plots, such as <code>which="uv"</code>.  If
not provided a default is used, with <code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code> being used
if there are more than 2000 points to plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_missingcolor">missingColor</code></td>
<td>
<p>color used to indicate <code>NA</code> values in images (see
<code><a href="#topic+imagep">imagep()</a></code>); set to <code>NULL</code> to avoid this indication.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_mgp">mgp</code></td>
<td>
<p>A 3-element numerical vector used with <code><a href="graphics.html#topic+par">par</a></code><code>("mgp")</code>
to control the spacing of axis elements.  The default is tighter than the R default.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_mar">mar</code></td>
<td>
<p>A 4-element numerical vector used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>
to control the plot margins. The default is tighter than the R default.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_mai.palette">mai.palette</code></td>
<td>
<p>margins, in inches, to be added to those calculated for
the palette; alter from the default only with caution</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_tformat">tformat</code></td>
<td>
<p>optional argument passed to <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code>, for
plot types that call that function.  (See <code><a href="base.html#topic+strptime">strptime()</a></code> for the
format used.)</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_marginsasimage">marginsAsImage</code></td>
<td>
<p>boolean, <code>TRUE</code> to put a wide margin to the right
of time-series plots, even if there are no images in the <code>which</code> list.
(The margin is made wide if there are some images in the sequence.)</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_cex">cex</code></td>
<td>
<p>numeric character expansion factor for plot symbols; see <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_cex.axis">cex.axis</code>, <code id="plot+2B2Cadp-method_+3A_cex.lab">cex.lab</code></td>
<td>
<p>character expansion factors for axis numbers and axis names; see <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_xlim">xlim</code></td>
<td>
<p>optional 2-element list for <code>xlim</code>, or 2-column matrix, in
which case the rows are used, in order, for the panels of the graph.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_ylim">ylim</code></td>
<td>
<p>optional 2-element list for <code>ylim</code>, or 2-column matrix, in
which case the rows are used, in order, for the panels of the graph.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_control">control</code></td>
<td>
<p>optional list of parameters that may be used for different
plot types.  Possibilities are <code>drawBottom</code> (a boolean that indicates
whether to draw the bottom) and <code>bin</code> (a numeric giving the index of
the bin on which to act, as explained in &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_uselayout">useLayout</code></td>
<td>
<p>set to <code>FALSE</code> to prevent using <code><a href="graphics.html#topic+layout">layout()</a></code>
to set up the plot.  This is needed if the call is to be part of a sequence
set up by e.g. <code>par(mfrow)</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_coastline">coastline</code></td>
<td>
<p>a <code>coastline</code> object, or a character string naming
one.  This is used only for <code>which="map"</code>.  See notes at
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method()</a></code> for more information on built-in coastlines.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_span">span</code></td>
<td>
<p>approximate span of map in km</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_main">main</code></td>
<td>
<p>main title for plot, used just on the top panel, if there are
several panels.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_grid">grid</code></td>
<td>
<p>if <code>TRUE</code>, a grid will be drawn for each panel.  (This
argument is needed, because calling <code><a href="graphics.html#topic+grid">grid()</a></code> after doing a
sequence of plots will not result in useful results for the individual
panels.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_grid.col">grid.col</code></td>
<td>
<p>color of grid</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_grid.lty">grid.lty</code></td>
<td>
<p>line type of grid</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_grid.lwd">grid.lwd</code></td>
<td>
<p>line width of grid</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_xlab">xlab</code></td>
<td>
<p>optional character value giving the label for the x axis.
If NULL (the default) then the label is determined automatically.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadp-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.  For example,
supplying <code>despike=TRUE</code> will cause time-series panels to be de-spiked
with <code><a href="#topic+despike">despike()</a></code>.  Another common action is to set the color for
missing values on image plots, with the argument <code>missingColor</code> (see
<code><a href="#topic+imagep">imagep()</a></code>).  Note that it is an error to give <code>breaks</code> in
..., if the formal argument <code>zlim</code> was also given, because they
could contradict each other.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot may have one or more panels, with the content being controlled by
the <code>which</code> argument.
</p>

<ul>
<li> <p><code>which=1:4</code> (or <code>which="u1"</code> to <code>"u4"</code>) yield a
distance-time image plot of a velocity component.  If <code>x</code> is in
<code>beam</code> coordinates (signalled by
<code>metadata$oce.coordinate=="beam"</code>), this will be the beam velocity,
labelled <code>b[1]</code> etc.  If <code>x</code> is in xyz coordinates (sometimes
called frame coordinates, or ship coordinates), it will be the velocity
component to the right of the frame or ship (labelled <code>u</code> etc).
Finally, if <code>x</code> is in <code>"enu"</code> coordinates, the image will show the
the eastward component (labelled <code>east</code>).  If <code>x</code> is in
<code>"other"</code> coordinates, it will be component corresponding to east,
after rotation (labelled <code style="white-space: pre;">&#8288;u\'&#8288;</code>).  Note that the coordinate is set by
<code><a href="#topic+read.adp">read.adp()</a></code>, or by <code><a href="#topic+beamToXyzAdp">beamToXyzAdp()</a></code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp()</a></code>, or <code><a href="#topic+enuToOtherAdp">enuToOtherAdp()</a></code>.
</p>
</li>
<li> <p><code>which=5:8</code> (or <code>which="a1"</code> to <code>"a4"</code>) yield
distance-time images of backscatter intensity of the respective beams.  (For
data derived from Teledyne-RDI instruments, this is the item called &ldquo;echo
intensity.&rdquo;)
</p>
</li>
<li> <p><code>which=9:12</code> (or <code>which="q1"</code> to <code>"q4"</code>) yield
distance-time images of signal quality for the respective beams.  (For RDI
data derived from instruments, this is the item called &ldquo;correlation
magnitude.&rdquo;)
</p>
</li>
<li> <p><code>which=60</code> or <code>which="map"</code> draw a map of location(s).
</p>
</li>
<li> <p><code>which=70:73</code> (or <code>which="g1"</code> to <code>"g4"</code>) yield
distance-time images of percent-good for the respective beams.  (For data
derived from Teledyne-RDI instruments, which are the only instruments that
yield this item, it is called &ldquo;percent good.&rdquo;)
</p>
</li>
<li> <p><code>which=80:83</code> (or <code>which="vv"</code>, <code>which="va"</code>,
<code>which="vq"</code>, and <code>which="vg"</code>) yield distance-time
images of the vertical beam fields for a 5 beam &quot;SentinelV&quot; ADCP
from Teledyne RDI.
</p>
</li>
<li> <p><code>which="vertical"</code> yields a two panel distance-time
image of vertical beam velocity and amplitude.
</p>
</li>
<li> <p><code>which=13</code> (or <code>which="salinity"</code>) yields a time-series plot
of salinity.
</p>
</li>
<li> <p><code>which=14</code> (or <code>which="temperature"</code>) yields a time-series
plot of temperature.
</p>
</li>
<li> <p><code>which=15</code> (or <code>which="pressure"</code>) yields a time-series plot
of pressure.
</p>
</li>
<li> <p><code>which=16</code> (or <code>which="heading"</code>) yields a time-series plot
of instrument heading.
</p>
</li>
<li> <p><code>which=17</code> (or <code>which="pitch"</code>) yields a time-series plot of
instrument pitch.
</p>
</li>
<li> <p><code>which=18</code> (or <code>which="roll"</code>) yields a time-series plot of
instrument roll.
</p>
</li>
<li> <p><code>which=19</code> yields a time-series plot of distance-averaged
velocity for beam 1, rightward velocity, eastward velocity, or
rotated-eastward velocity, depending on the coordinate system.
</p>
</li>
<li> <p><code>which=20</code> yields a time-series of distance-averaged velocity for
beam 2, foreward velocity, northward velocity, or rotated-northward
velocity, depending on the coordinate system.
</p>
</li>
<li> <p><code>which=21</code> yields a time-series of distance-averaged velocity for
beam 3, up-frame velocity, upward velocity, or rotated-upward velocity,
depending on the coordinate system.
</p>
</li>
<li> <p><code>which=22</code> yields a time-series of distance-averaged velocity for
beam 4, for <code>beam</code> coordinates, or velocity estimate, for other
coordinates.  (This is ignored for 3-beam data.)
</p>
</li>
<li> <p><code>which="progressiveVector"</code> (or <code>which=23</code>) yields a progressive-vector diagram in the horizontal
plane, plotted with <code>asp=1</code>.  Normally, the depth-averaged velocity
components are used, but if the <code>control</code> list contains an item named
<code>bin</code>, then the depth bin will be used (with an error resulting if the
bin is out of range).
</p>
</li>
<li> <p><code>which=24</code> yields a time-averaged profile of the first component
of velocity (see <code>which=19</code> for the meaning of the component, in
various coordinate systems).
</p>
</li>
<li> <p><code>which=25</code> as for 24, but the second component.
</p>
</li>
<li> <p><code>which=26</code> as for 24, but the third component.
</p>
</li>
<li> <p><code>which=27</code> as for 24, but the fourth component (if that makes
sense, for the given instrument).
</p>
</li>
<li> <p><code>which=28</code> or <code>"uv"</code> yields velocity plot in the horizontal
plane, i.e. <code>u[2]</code> versus <code>u[1]</code>.  If the number of data points is small, a
scattergraph is used, but if it is large, <code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code> is
used.
</p>
</li>
<li> <p><code>which=29</code> or <code>"uv+ellipse"</code> as the <code>"uv"</code> case, but
with an added indication of the tidal ellipse, calculated from the eigen
vectors of the covariance matrix.
</p>
</li>
<li> <p><code>which=30</code> or <code>"uv+ellipse+arrow"</code> as the
<code>"uv+ellipse"</code> case, but with an added arrow indicating the mean
current.
</p>
</li>
<li> <p><code>which=40</code> or <code>"bottomRange"</code> for average bottom range from
all beams of the instrument.
</p>
</li>
<li> <p><code>which=41</code> to <code>44</code> (or <code>"bottomRange1"</code> to
<code>"bottomRange4"</code>) for bottom range from beams 1 to 4.
</p>
</li>
<li> <p><code>which=50</code> or <code>"bottomVelocity"</code> for average bottom velocity
from all beams of the instrument.
</p>
</li>
<li> <p><code>which=51</code> to <code>54</code> (or <code>"bottomVelocity1"</code> to
<code>"bottomVelocity4"</code>) for bottom velocity from beams 1 to 4.
</p>
</li>
<li> <p><code>which=55</code> (or <code>"heaving"</code>) for time-integrated,
depth-averaged, vertical velocity, i.e. a time series of heaving.
</p>
</li>
<li> <p><code>which=60</code> (or <code>"map"</code>) for a map.
</p>
</li>
<li> <p><code>which=100</code> (or <code>"soundSpeed"</code>) for a time series of sound speed.
</p>
</li>
<li> <p><code>which=200</code> (or <code>"accelerometerx"</code>) for a time-series of the x component of
the accelerometer reading.
</p>
</li>
<li> <p><code>which=201</code> (or <code>"accelerometery"</code>) for a time-series of the y component of
the accelerometer reading.
</p>
</li>
<li> <p><code>which=202</code> (or <code>"accelerometerz"</code>) for a time-series of the z component of
the accelerometer reading.
</p>
</li>
<li> <p><code>which=210</code> (or <code>"magnetometerx"</code>) for a time-series of the x component of
the magnetometer reading.
</p>
</li>
<li> <p><code>which=211</code> (or <code>"magnetometery"</code>) for a time-series of the y component of
the magnetometer reading.
</p>
</li>
<li> <p><code>which=212</code> (or <code>"magnetometerz"</code>) for a time-series of the z component of
the magnetometer reading.
</p>
</li></ul>

<p>In addition to the above, the following shortcuts are defined:
</p>

<ul>
<li> <p><code>which="velocity"</code> equivalent to <code>which=1:3</code> or <code>1:4</code>
(depending on the device) for velocity components.
</p>
</li>
<li> <p><code>which="amplitude"</code> equivalent to <code>which=5:7</code>
or <code>5:8</code> (depending on the device) for backscatter intensity
components.
</p>
</li>
<li> <p><code>which="quality"</code> equivalent to <code>which=9:11</code> or <code>9:12</code>
(depending on the device) for quality components.
</p>
</li>
<li> <p><code>which="hydrography"</code> equivalent to <code>which=14:15</code>
for temperature and pressure.
</p>
</li>
<li> <p><code>which="angles"</code> equivalent to <code>which=16:18</code> for
heading, pitch and roll.
</p>
</li>
<li> <p><code>which="accelerometer"</code> to plot a 3-panel timeseries
of acceleration, equivalent to <code>which=110:102</code>.
</p>
</li></ul>

<p>The color scheme for image plots (<code>which</code> in 1:12) is provided by the
<code>col</code> argument, which is passed to <code><a href="Matrix.html#topic+image">image()</a></code> to do the actual
plotting.  See &ldquo;Examples&rdquo; for some comparisons.
</p>
<p>A common quick-look plot to assess mooring movement is to use
<code>which=15:18</code> (pressure being included to signal the tide, and tidal
currents may dislodge a mooring or cause it to settle).
</p>
<p>By default, <code style="white-space: pre;">&#8288;plot,adp-method&#8288;</code> uses a <code>zlim</code> value for the
<code><a href="Matrix.html#topic+image">image()</a></code> that is constructed to contain all the data, but to be
symmetric about zero.  This is done on a per-panel basis, and the scale is
plotted at the top-right corner, along with the name of the variable being
plotted. You may also supply <code>zlim</code> as one of the ... arguments,
but be aware that a reasonable limit on horizontal velocity components is
unlikely to be of much use for the vertical component.
</p>
<p>A good first step in the analysis of measurements made from a moored device
(stored in <code>d</code>, say) is to do <code>plot(d, which=14:18)</code>.  This shows
time series of water properties and sensor orientation, which is helpful in
deciding which data to trim at the start and end of the deployment, because
they were measured on the dock or on the ship as it travelled to the mooring
site.
</p>


<h3>Value</h3>

<p>A list is silently returned, containing <code>xat</code> and <code>yat</code>,
values that can be used by <code><a href="#topic+oce.grid">oce.grid()</a></code> to add a grid to the plot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adp)
plot(adp, which = 1:3)
plot(adp, which = "temperature", tformat = "%H:%M")

</code></pre>

<hr>
<h2 id='plot+2Cadv-method'>Plot an adv Object</h2><span id='topic+plot+2Cadv-method'></span><span id='topic+plot.adv'></span>

<h3>Description</h3>

<p>Plot <a href="#topic+adv-class">adv</a> data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adv'
plot(
  x,
  which = c(1:3, 14, 15),
  col,
  titles,
  type = "l",
  lwd = par("lwd"),
  drawTimeRange = getOption("oceDrawTimeRange"),
  drawZeroLine = FALSE,
  useSmoothScatter,
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1.5, mgp[1] + 1.5, 1.5, 1.5),
  tformat,
  marginsAsImage = FALSE,
  cex = par("cex"),
  cex.axis = par("cex.axis"),
  cex.lab = par("cex.lab"),
  cex.main = par("cex.main"),
  xlim,
  ylim,
  brushCorrelation,
  colBrush = "red",
  main = "",
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cadv-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_which">which</code></td>
<td>
<p>List of desired plot types.  These are graphed in panels running
down from the top of the page.  See &ldquo;Details&rdquo; for the meanings of
various values of <code>which</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_col">col</code></td>
<td>
<p>Optional indication of color(s) to use.  If not provided, the
default for images is <code>oce.colorsPalette(128,1)</code>, and for lines and points
is black.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_titles">titles</code></td>
<td>
<p>Optional vector of character strings to be used as labels for the
plot panels.  For images, these strings will be placed in the right hand side
of the top margin.  For timeseries, these strings are ignored.  If this is
provided, its length must equal that of <code>which</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_type">type</code></td>
<td>
<p>Type of plot, as for <code><a href="graphics.html#topic+plot">plot()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_lwd">lwd</code></td>
<td>
<p>If the plot is of a time-series or scattergraph format with lines,
this is used in the usual way; otherwise, e.g. for image formats, this is
ignored.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_drawtimerange">drawTimeRange</code></td>
<td>
<p>Logical value that applies to panels with time as the horizontal
axis, indicating whether to draw the time range in the top-left margin of the
plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_drawzeroline">drawZeroLine</code></td>
<td>
<p>Logical value indicating whether to draw zero lines on
velocities.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_usesmoothscatter">useSmoothScatter</code></td>
<td>
<p>Logical value indicating whether to use
<code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code> in various plots, such as <code>which="uv"</code>.  If
not provided a default is used, with <code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code> being used if
there are more than 2000 points to plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical
vector to use for <code>par(mgp)</code>, and also for <code>par(mar)</code>, computed from
this.  The default is tighter than the R default, in order to use more space
for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_mar">mar</code></td>
<td>
<p>Value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_tformat">tformat</code></td>
<td>
<p>Optional argument passed to <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code>, for plot
types that call that function.  (See <code><a href="base.html#topic+strptime">strptime()</a></code> for the format
used.)</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_marginsasimage">marginsAsImage</code></td>
<td>
<p>Logical value indicating whether to put a wide margin to
the right of time-series plots, matching the space used up by a palette in an
<code><a href="#topic+imagep">imagep()</a></code> plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_cex">cex</code></td>
<td>
<p>numeric character expansion factor for plot symbols; see <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_cex.axis">cex.axis</code>, <code id="plot+2B2Cadv-method_+3A_cex.lab">cex.lab</code>, <code id="plot+2B2Cadv-method_+3A_cex.main">cex.main</code></td>
<td>
<p>character expansion factors for axis numbers, axis names and plot titles; see <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_xlim">xlim</code></td>
<td>
<p>Optional 2-element list for <code>xlim</code>, or 2-column matrix, in
which case the rows are used, in order, for the panels of the graph.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_ylim">ylim</code></td>
<td>
<p>Optional 2-element list for <code>ylim</code>, or 2-column matrix, in
which case the rows are used, in order, for the panels of the graph.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_brushcorrelation">brushCorrelation</code></td>
<td>
<p>Optional number between 0 and 100, indicating a
per-beam correlation threshold below which data are to be considered suspect.
If the plot type is <code>p</code>, the suspect points (velocity, backscatter
amplitude, or correlation) will be colored red; otherwise, this argument is
ignored.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_colbrush">colBrush</code></td>
<td>
<p>Color to use for brushed (bad) data, if
<code>brushCorrelation</code> is active.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_main">main</code></td>
<td>
<p>Main title for plot, used just on the top panel, if there are
several panels.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_debug">debug</code></td>
<td>
<p>A flag that turns on debugging.  Set to 1 to get a moderate amount
of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="plot+2B2Cadv-method_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a multi-panel summary plot of data measured by an ADV.
The panels are controlled by the <code>which</code> argument.  (Note the
gaps in the sequence, e.g. 4 and 8 are not used.)
</p>

<ul>
<li> <p><code>which=1</code> to <code>3</code> (or <code>"u1"</code> to <code>"u3"</code>)
yield timeseries of the first, second, and third components of
velocity (in beam, xyz or enu coordinates).
</p>
</li>
<li> <p><code>which=4</code> is not permitted (since ADV are 3-beam devices)
</p>
</li>
<li> <p><code>which=5</code> to <code>7</code> (or <code>"a1"</code> to <code>"a3"</code>)
yield timeseries of the amplitudes of beams 1 to 3.  (Note that
the data are called <code>data$a[,1]</code>, <code>data$a[,2]</code> and
<code>data$a[,3]</code>, for these three timeseries.)
</p>
</li>
<li> <p><code>which=8</code> is not permitted (since ADV are 3-beam devices)
</p>
</li>
<li> <p><code>which=9</code> to <code>11</code> (or <code>"q1"</code> to <code>"q3"</code>)
yield timeseries of correlation for beams 1 to 3.  (Note that the
data are called <code>data$c[,1]</code>, <code>data$c[,2]</code> and
<code>data$c[,3]</code>, for these three timeseries.)
</p>
</li>
<li> <p><code>which=12</code> is not permitted (since ADVs are 3-beam devices)
</p>
</li>
<li> <p><code>which=13</code> is not permitted (since ADVs do not measure salinity)
</p>
</li>
<li> <p><code>which=14</code> or <code>which="temperature"</code> yields a timeseries of temperature.
</p>
</li>
<li> <p><code>which=15</code> or <code>which="pressure"</code> yields a timeseries of pressure.
</p>
</li>
<li> <p><code>which=16</code> or <code>which="heading"</code> yields a timeseries of heading.
</p>
</li>
<li> <p><code>which=17</code> or <code>which="pitch"</code>yields a timeseries of pitch.
</p>
</li>
<li> <p><code>which=18</code> or <code>which="roll"</code>yields a timeseries of roll.
</p>
</li>
<li> <p><code>which=19</code> to <code>21</code> yields plots of correlation versus
amplitude, for beams 1 through 3, using <code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code>.
</p>
</li>
<li> <p><code>which=22</code> is not permitted (since ADVs are 3-beam devices)
</p>
</li>
<li> <p><code>which=23</code> or <code>"progressive vector"</code> yields a
progressive-vector diagram in the horizontal plane, plotted with
<code>asp=1</code>, and taking beam1 and beam2 as the eastward and
northward components of velocity, respectively.
</p>
</li>
<li> <p><code>which=28</code> or <code>"uv"</code> yields velocity plot in the
horizontal plane, i.e. <code>u[2]</code> versus <code>u[1]</code>.  If the number of data
points is small, a scattergraph is used, but if it is large,
<code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code> is used.
</p>
</li>
<li> <p><code>which=29</code> or <code>"uv+ellipse"</code> as the <code>"uv"</code>
case, but with an added indication of the tidal ellipse,
calculated from the eigen vectors of the covariance matrix.
</p>
</li>
<li> <p><code>which=30</code> or <code>"uv+ellipse+arrow"</code> as the
<code>"uv+ellipse"</code> case, but with an added arrow indicating the
mean current.
</p>
</li>
<li> <p><code>which=50</code> or <code>"analog1"</code> plots a time series of the
analog1 signal, if there is one.
</p>
</li>
<li> <p><code>which=51</code> or <code>"analog2"</code> plots a time series of the
analog2 signal, if there is one.
</p>
</li>
<li> <p><code>which=100</code> or <code>"voltage"</code> plots the voltage as a
timeseries, if voltage exists in the dataset.
</p>
</li></ul>

<p>In addition to the above, there are some groupings defined:
</p>

<ul>
<li> <p><code>which="velocity"</code> equivalent to <code>which=1:3</code> (three velocity components)
</p>
</li>
<li> <p><code>which="amplitude"</code> equivalent to <code>which=5:7</code> (three amplitude components)
</p>
</li>
<li> <p><code>which="backscatter"</code> equivalent to <code>which=9:11</code> (three backscatter components)
</p>
</li>
<li> <p><code>which="hydrography"</code> equivalent to <code>which=14:15</code> (temperature and pressure)
</p>
</li>
<li> <p><code>which="angles"</code> equivalent to <code>which=16:18</code> (heading, pitch and roll)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for <a href="#topic+adv-class">adv</a> explains the structure
of ADV objects, and also outlines the other functions dealing with them.
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adv)
plot(adv)

</code></pre>

<hr>
<h2 id='plot+2Camsr-method'>Plot an amsr Object</h2><span id='topic+plot+2Camsr-method'></span><span id='topic+plot.amsr'></span>

<h3>Description</h3>

<p>Plot an image of a component of an <a href="#topic+amsr-class">amsr</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'amsr'
plot(
  x,
  y,
  asp = NULL,
  breaks,
  col,
  colormap,
  zlim,
  missingColor,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Camsr-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+amsr-class">amsr</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Camsr-method_+3A_y">y</code></td>
<td>
<p>character value indicating the name of the band to plot; if not provided,
<code>SST</code> is used; see the documentation for the <a href="#topic+amsr-class">amsr</a> class for a list of bands.</p>
</td></tr>
<tr><td><code id="plot+2B2Camsr-method_+3A_asp">asp</code></td>
<td>
<p>optional numerical value giving the aspect ratio for plot.  The
default value, <code>NULL</code>, means to use an aspect ratio of 1 for world views,
and a value computed from <code>ylim</code>, if the latter is specified in the
<code>...</code> argument.</p>
</td></tr>
<tr><td><code id="plot+2B2Camsr-method_+3A_breaks">breaks</code></td>
<td>
<p>optional numeric vector of the z values for breaks in the color scheme.
If <code>colormap</code> is provided, it takes precedence over <code>breaks</code> and <code>col</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Camsr-method_+3A_col">col</code></td>
<td>
<p>optional argument, either a vector of colors corresponding to the breaks, of length
1 less than the number of breaks, or a function specifying colors.
If neither <code>col</code> or <code>colormap</code> is provided, then <code>col</code> defaults to
<code><a href="#topic+oceColorsTemperature">oceColorsTemperature()</a></code>.
If <code>colormap</code> is provided, it takes precedence over <code>breaks</code> and <code>col</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Camsr-method_+3A_colormap">colormap</code></td>
<td>
<p>a specification of the colormap to use, as created
with <code><a href="#topic+colormap">colormap()</a></code>.  If <code>colormap</code> is NULL, which is the default, then
a colormap is created to cover the range of data values, using
<a href="#topic+oceColorsTemperature">oceColorsTemperature</a> color scheme.
If <code>colormap</code> is provided, it takes precedence over <code>breaks</code> and <code>col</code>.
See &ldquo;Examples&rdquo; for an example of using the &quot;turbo&quot; color scheme.</p>
</td></tr>
<tr><td><code id="plot+2B2Camsr-method_+3A_zlim">zlim</code></td>
<td>
<p>optional numeric vector of length 2, giving the limits
of the plotted quantity.  A reasonable default is computed, if this
is not given.</p>
</td></tr>
<tr><td><code id="plot+2B2Camsr-method_+3A_missingcolor">missingColor</code></td>
<td>
<p>optional list specifying colors to use for
non-data categories.  If not provided, a default is used.  For
type 1, that default is
<code>list(land="papayaWhip", none="lightGray", bad="gray", rain="plum", ice="mediumVioletRed")</code>.  For type 2, it is
<code>list(coast="gray", land="papayaWhip", noObs="lightGray", seaIce="mediumVioletRed")</code>.  Any colors may be used in place of these,
but the names must match, and all names must be present.</p>
</td></tr>
<tr><td><code id="plot+2B2Camsr-method_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
<tr><td><code id="plot+2B2Camsr-method_+3A_...">...</code></td>
<td>
<p>extra arguments passed to <code><a href="#topic+imagep">imagep()</a></code>, e.g. to control
the view with <code>xlim</code> (for longitude) and <code>ylim</code> (for latitude).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In addition to fields named directly in the object, such as <code>SSTDay</code> and
<code>SSTNight</code>, it is also possible to plot computed fields, such as <code>SST</code>,
which combines the day and night fields.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to amsr data: 
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method</a></code>,
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+read.amsr">read.amsr</a>()</code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+summary+2Camsr-method">summary,amsr-method</a></code>
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(coastlineWorld)
data(amsr) # see ?amsr for how to read and composite such objects

# Example 1: plot with default color scheme, oceColorsTemperature()
plot(amsr, "SST")
lines(coastlineWorld[["longitude"]], coastlineWorld[["latitude"]])

# Example 2: 'turbo' color scheme
plot(amsr, "SST", col = oceColorsTurbo)
lines(coastlineWorld[["longitude"]], coastlineWorld[["latitude"]])

</code></pre>

<hr>
<h2 id='plot+2Cargo-method'>Plot an argo Object</h2><span id='topic+plot+2Cargo-method'></span><span id='topic+plot.argo'></span>

<h3>Description</h3>

<p>Plot a summary diagram for argo data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'argo'
plot(
  x,
  which = 1,
  level,
  coastline = c("best", "coastlineWorld", "coastlineWorldMedium", "coastlineWorldFine",
    "none"),
  cex = 1,
  pch = 1,
  type = "p",
  col = 1,
  fill = FALSE,
  projection = NULL,
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1.5, mgp[1] + 1.5, 1.5, 1.5),
  tformat,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cargo-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+argo-class">argo</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_which">which</code></td>
<td>
<p>list of desired plot types, one of the following. Note
that <code><a href="#topic+oce.pmatch">oce.pmatch()</a></code> is used to try to complete partial
character matches, and that an error will occur if the match is
not complete (e.g. <code>"salinity"</code> matches to both
<code>"salinity ts"</code> and <code>"salinity profile"</code>.).
</p>

<ul>
<li> <p><code>which=1</code>, <code>which="trajectory"</code> or <code>which="map"</code> gives a
plot of the argo trajectory, with the coastline, if one is provided.
</p>
</li>
<li> <p><code>which=2</code> or <code>"salinity ts"</code> gives a time series of
salinity at the indicated level(s)
</p>
</li>
<li> <p><code>which=3</code> or <code>"temperature ts"</code> gives a time series
of temperature at the indicated level(s)
</p>
</li>
<li> <p><code>which=4</code> or <code>"TS"</code> gives a TS diagram at the
indicated level(s)
</p>
</li>
<li> <p><code>which=5</code> or <code>"salinity profile"</code> gives a salinity
profile of all the data (with S and p trimmed to the 1 and 99
percentiles)
</p>
</li>
<li> <p><code>which=6</code> or <code>"temperature profile"</code> gives a
temperature profile (with T and p trimmed to the 1 and 99
percentiles)
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_level">level</code></td>
<td>
<p>depth pseudo-level to plot, for <code>which=2</code> and higher.  May be an
integer, in which case it refers to an index of depth (1 being the top)
or it may be the string &quot;all&quot; which means to plot all data.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_coastline">coastline</code></td>
<td>
<p>character string giving the coastline to be used in an Argo-location
map, or <code>"best"</code> to pick the one with highest resolution, or
<code>"none"</code> to avoid drawing the coastline.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_cex">cex</code></td>
<td>
<p>size of plotting symbols to be used if <code>type="p"</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_pch">pch</code></td>
<td>
<p>type of plotting symbols to be used if <code>type="p"</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_type">type</code></td>
<td>
<p>plot type, either <code>"l"</code> or <code>"p"</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_col">col</code></td>
<td>
<p>optional list of colors for plotting.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_fill">fill</code></td>
<td>
<p>Either a logical, indicating whether to fill the land with
light-gray, or a color name.  Owing to problems with some projections, the
default is not to fill.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_projection">projection</code></td>
<td>
<p>indication of the projection to be used
in trajectory maps. If this is <code>NULL</code>, no projection is used, although
the plot aspect ratio will be set to yield zero shape distortion at the
mean float latitude.  If <code>projection="automatic"</code>, then one
of two projections is used: stereopolar (i.e. <code>"+proj=stere +lon_0=X"</code>
where <code>X</code> is the mean longitude), or Mercator (i.e. <code>"+proj=merc"</code>)
otherwise.  Otherwise, <code>projection</code> must be a character string specifying
a projection in the notation used by <code><a href="#topic+oceProject">oceProject()</a></code> and <code><a href="#topic+mapPlot">mapPlot()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code>par(mgp)</code>, and also for
<code>par(mar)</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code>par("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_tformat">tformat</code></td>
<td>
<p>optional argument passed to <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code>, for plot
types that call that function.  (See <code><a href="base.html#topic+strptime">strptime()</a></code> for the format
used.)</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_debug">debug</code></td>
<td>
<p>debugging flag.</p>
</td></tr>
<tr><td><code id="plot+2B2Cargo-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(argo)
tc &lt;- cut(argo[["time"]], "year")
# Example 1: plot map, which reveals float trajectory.
plot(argo, pch = as.integer(tc))
year &lt;- substr(levels(tc), 1, 4)
data(topoWorld)
contour(topoWorld[["longitude"]], topoWorld[["latitude"]],
    topoWorld[["z"]],
    add = TRUE
)
legend("bottomleft", pch = seq_along(year), legend = year, bg = "white", cex = 3 / 4)

# Example 2: plot map, TS, T(z) and S(z). Note the use
# of handleFlags(), to skip over questionable data.
plot(handleFlags(argo), which = c(1, 4, 6, 5))

</code></pre>

<hr>
<h2 id='plot+2Cbremen-method'>Plot a bremen Object</h2><span id='topic+plot+2Cbremen-method'></span><span id='topic+plot.bremen'></span>

<h3>Description</h3>

<p>Plot a <a href="#topic+bremen-class">bremen</a> object.
If the first argument seems to be a CTD dataset, this uses <code><a href="#topic+plot+2Cctd-method">plot,ctd-method()</a></code>;
otherwise, that argument is assumed to be a <a href="#topic+ladp-class">ladp</a> object, and a two-panel
plot is created with <code><a href="#topic+plot+2Cladp-method">plot,ladp-method()</a></code> to show velocity variation with
pressure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'bremen'
plot(x, type, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cbremen-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+bremen-class">bremen</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cbremen-method_+3A_type">type</code></td>
<td>
<p>Optional string indicating the type to which <code>x</code> should be
coerced before plotting. The choices are <code>ctd</code> and <code>ladp</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cbremen-method_+3A_...">...</code></td>
<td>
<p>Other arguments, passed to plotting functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to bremen data: 
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+read.bremen">read.bremen</a>()</code>,
<code><a href="#topic+summary+2Cbremen-method">summary,bremen-method</a></code>
</p>

<hr>
<h2 id='plot+2Ccm-method'>Plot a cm Object</h2><span id='topic+plot+2Ccm-method'></span><span id='topic+plot.cm'></span>

<h3>Description</h3>

<p>Creates a multi-panel summary plot of data measured by a current meter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'cm'
plot(
  x,
  which = c(1:2),
  type = "l",
  xlim,
  ylim,
  xaxs = "r",
  yaxs = "r",
  drawTimeRange = getOption("oceDrawTimeRange"),
  drawZeroLine = FALSE,
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1.5, mgp[1] + 1.5, 1.5, 1.5),
  small = 2000,
  main = "",
  tformat,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Ccm-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+cm-class">cm</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_which">which</code></td>
<td>
<p>list of desired plot types.  These are graphed in panels running
down from the top of the page.  See &ldquo;Details&rdquo; for the meanings of various
values of <code>which</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_type">type</code></td>
<td>
<p>type of plot, as for <code><a href="graphics.html#topic+plot">plot()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_xlim">xlim</code>, <code id="plot+2B2Ccm-method_+3A_ylim">ylim</code></td>
<td>
<p>optional limit to the x and y axes, passed to <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code> for
time-series plots.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_xaxs">xaxs</code>, <code id="plot+2B2Ccm-method_+3A_yaxs">yaxs</code></td>
<td>
<p>optional controls over the limits of the x and y axes,
passed to <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code> for time-series plots.  These values default
to <code>"r"</code>, meaning to use the regular method of extend the plot past
its normal limits.  It is common to use <code>"i"</code> to make the graph extend
to the panel limits.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_drawtimerange">drawTimeRange</code></td>
<td>
<p>boolean that applies to panels with time as the horizontal
axis, indicating whether to draw the time range in the top-left margin of the
plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_drawzeroline">drawZeroLine</code></td>
<td>
<p>boolean that indicates whether to draw zero lines on
velocities.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code>par(mgp)</code>, and also for
<code>par(mar)</code>, computed from this.  The default is tighter than the R default,
in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_small">small</code></td>
<td>
<p>an integer indicating the size of data set to be considered
&quot;small&quot;, to be plotted with points or lines using the standard
<code><a href="graphics.html#topic+plot">plot()</a></code> function.  Data sets with more than <code>small</code> points will
be plotted with <code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code> instead.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_main">main</code></td>
<td>
<p>main title for plot, used just on the top panel, if there are
several panels.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_tformat">tformat</code></td>
<td>
<p>optional argument passed to <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code>, for plot
types that call that function.  (See <code><a href="base.html#topic+strptime">strptime()</a></code> for the format
used.)</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate amount
of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccm-method_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The panels are controlled by the <code>which</code> argument, as follows.
</p>

<ul>
<li> <p><code>which=1</code> or <code>which="u"</code> for a time-series graph of eastward
velocity, <code>u</code>, as a function of time.
</p>
</li>
<li> <p><code>which=2</code> or <code>which="v"</code> for a time-series graph of
northward velocity, <code>u</code>, as a function of time.
</p>
</li>
<li> <p><code>which=3</code> or <code>"progressive vector"</code> for progressive-vector
plot
</p>
</li>
<li> <p><code>which=4</code> or <code>"uv"</code> for a plot of <code>v</code> versus <code>u</code>.
(Dots are used for small datasets, and smoothScatter for large ones.)
</p>
</li>
<li> <p><code>which=5</code> or <code>"uv+ellipse"</code> as the <code>"uv"</code> case, but
with an added indication of the tidal ellipse, calculated from the eigen
vectors of the covariance matrix.
</p>
</li>
<li> <p><code>which=6</code> or <code>"uv+ellipse+arrow"</code> as the <code>"uv+ellipse"</code>
case, but with an added arrow indicating the mean current.
</p>
</li>
<li> <p><code>which=7</code> or <code>"pressure"</code> for pressure
</p>
</li>
<li> <p><code>which=8</code> or <code>"salinity"</code> for salinity
</p>
</li>
<li> <p><code>which=9</code> or <code>"temperature"</code> for temperature
</p>
</li>
<li> <p><code>which=10</code> or <code>"TS"</code> for a TS diagram
</p>
</li>
<li> <p><code>which=11</code> or <code>"conductivity"</code> for conductivity
</p>
</li>
<li> <p><code>which=20</code> or <code>"direction"</code> for the direction of flow
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to cm data: 
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+as.cm">as.cm</a>()</code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+read.cm">read.cm</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+summary+2Ccm-method">summary,cm-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(cm)
summary(cm)
plot(cm)

</code></pre>

<hr>
<h2 id='plot+2Ccoastline-method'>Plot a coastline Object</h2><span id='topic+plot+2Ccoastline-method'></span><span id='topic+plot.coastline'></span>

<h3>Description</h3>

<p>This function plots a coastline.  An attempt is made to fill the space of
the plot, and this is done by limiting either the longitude range or the
latitude range, as appropriate, by modifying the eastern or northern limit,
as appropriate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'coastline'
plot(
  x,
  xlab = "",
  ylab = "",
  showHemi = TRUE,
  asp,
  clongitude,
  clatitude,
  span,
  lonlabels = TRUE,
  latlabels = TRUE,
  projection = NULL,
  expand = 1,
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1, mgp[1] + 1, 1, 1),
  bg,
  fill,
  type = "polygon",
  border = NULL,
  col = NULL,
  axes = TRUE,
  cex.axis = par("cex.axis"),
  add = FALSE,
  inset = FALSE,
  geographical = 0,
  longitudelim,
  latitudelim,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+coastline-class">coastline</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_xlab">xlab</code></td>
<td>
<p>label for x axis</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_ylab">ylab</code></td>
<td>
<p>label for y axis</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_showhemi">showHemi</code></td>
<td>
<p>logical indicating whether to show the hemisphere in axis
tick labels.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_asp">asp</code></td>
<td>
<p>Aspect ratio for plot.  The default is for <code style="white-space: pre;">&#8288;plot,coastline-method&#8288;</code>
to set the aspect ratio to give natural latitude-longitude scaling somewhere
near the centre latitude on the plot. Often, it makes sense to set
<code>asp</code> yourself, e.g. to get correct shapes at 45N, use
<code>asp=1/cos(45*pi/180)</code>.  Note that the land mass is not symmetric about
the equator, so to get good world views you should set <code>asp=1</code> or set
<code>ylim</code> to be symmetric about zero. Any given value of <code>asp</code> is
ignored, if <code>clongitude</code> and <code>clatitude</code> are given (or
if the latter two are inferred from <code>projection</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_clongitude">clongitude</code>, <code id="plot+2B2Ccoastline-method_+3A_clatitude">clatitude</code></td>
<td>
<p>optional center latitude of map, in decimal
degrees.  If both <code>clongitude</code> and <code>clatitude</code> are provided,
or alternatively if they can be inferred from substrings <code>+lon_0</code>
and <code>+lat_0</code> in <code>projection</code>, then
any provided value of <code>asp</code> is ignored, and instead the plot aspect
ratio is computed based on the center latitude.  If <code>clongitude</code> and
<code>clatitude</code> are known, then <code>span</code> must also be provided, and
in this case, it is not permitted to also specify <code>longitudelim</code> and
<code>latitudelim</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_span">span</code></td>
<td>
<p>optional suggested diagonal span of the plot, in kilometers.
The plotted span is usually close to the suggestion, although the details
depend on the plot aspect ratio and other factors, so some adjustment may be
required to fine-tune a plot.  A value for <code>span</code> must be supplied, if
<code>clongitude</code> and <code>clatitude</code> are supplied
(or inferred from <code>projection</code>).</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_lonlabels">lonlabels</code>, <code id="plot+2B2Ccoastline-method_+3A_latlabels">latlabels</code></td>
<td>
<p>optional vectors of longitude and latitude to
label on the sides of plot, passed to <code><a href="#topic+mapPlot">mapPlot()</a></code> to control
axis labelling, for plots done with map projections (i.e. for
cases in which <code>projection</code> is not <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_projection">projection</code></td>
<td>
<p>optional map projection to use (see
the <code><a href="#topic+mapPlot">mapPlot()</a></code> argument of the same name).
If set to <code>FALSE</code> then no projection is used,
and the data are plotted in a cartesion frame, with aspect ratio set to
reduce distortion near the middle of the plot.  This option is useful if the
coastline produces spurious horizontal lines owing to islands crossing the
plot edges (a problem that plagues map projections).  If <code>projection</code>
is not set, a Mercator projection is used for latitudes below about 70
degrees, as if <code>projection="+proj=merc"</code> had been supplied, or a
Stereopolar one is used as if <code>projection="+proj=stere"</code>.  Otherwise,
<code>projection</code> must be a character string identifying a projection
accepted by <code><a href="#topic+mapPlot">mapPlot()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_expand">expand</code></td>
<td>
<p>numerical factor for the expansion of plot limits, showing
area outside the plot, e.g. if showing a ship track as a coastline, and then
an actual coastline to show the ocean boundary.  The value of <code>expand</code>
is ignored if either <code>xlim</code> or <code>ylim</code> is given.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code><a href="graphics.html#topic+par">par</a></code><code>("mgp")</code>, and also
for <code>par(mar)</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_bg">bg</code></td>
<td>
<p>optional color to be used for the background of the map.  This
comes in handy for drawing insets (see &ldquo;details&rdquo;).</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_fill">fill</code></td>
<td>
<p>a legacy parameter that will be permitted only temporarily; see
&ldquo;History&rdquo;.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_type">type</code></td>
<td>
<p>indication of type; may be <code>"polygon"</code>, for a filled polygon,
<code>"p"</code> for points, <code>"l"</code> for line segments, or <code>"o"</code> for points
overlain with line segments. See <code>color</code> for a note on how
the the value of <code>type</code> alters the meaning of the <code>color</code>
argument.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_border">border</code></td>
<td>
<p>color used to indicate land (if <code>type="polygon"</code>) or
the coastline and international borders (if <code>type="l"</code>).</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_col">col</code></td>
<td>
<p>either the color for filling polygons (if <code>type="polygon"</code>)
or the color of the points and line segments (if <code>type="p"</code>,
<code>type="l"</code>, or <code>type="o"</code>).</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_axes">axes</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to plot axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_cex.axis">cex.axis</code></td>
<td>
<p>value for axis font size factor.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_add">add</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to draw the coastline on an existing
plot.  Note that this retains the aspect ratio of that existing plot, so it
is important to set that correctly, e.g. with <code>asp=1/cos(lat * pi /
180)</code>, where <code>clat</code> is the central latitude of the plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_inset">inset</code></td>
<td>
<p>set to <code>TRUE</code> for use within <code><a href="#topic+plotInset">plotInset()</a></code>.  The
effect is to prevent the present function from adjusting margins, which is
necessary because margin adjustment is the basis for the method used by
<code><a href="#topic+plotInset">plotInset()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_geographical">geographical</code></td>
<td>
<p>flag indicating the style of axes.  With
<code>geographical=0</code>, the axes are conventional, with decimal degrees as
the unit, and negative signs indicating the southern and western
hemispheres.  With <code>geographical=1</code>, the signs are dropped, with axis
values being in decreasing order within the southern and western
hemispheres.  With <code>geographical=2</code>, the signs are dropped and the axes
are labelled with degrees, minutes and seconds, as appropriate, and
hemispheres are indicated with letters. With <code>geographical=3</code>, things
are the same as for <code>geographical=2</code>, but the hemisphere indication
is omitted. Finally, with <code>geographical=4</code>, unsigned numbers are used,
followed by letters <code>N</code> in the northern hemisphere, <code>S</code> in the southern,
<code>E</code> in the eastern, and <code>W</code> in the western.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_longitudelim">longitudelim</code></td>
<td>
<p>this and <code>latitudelim</code> provide a second way to
suggest plot ranges. Note that these may not be supplied if
<code>clongitude</code>, <code>clatitude</code> and <code>span</code> are given.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_latitudelim">latitudelim</code></td>
<td>
<p>see <code>longitudelim</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_debug">debug</code></td>
<td>
<p>set to <code>TRUE</code> to get debugging information during
processing.</p>
</td></tr>
<tr><td><code id="plot+2B2Ccoastline-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.  For example,
set <code>yaxp=c(-90,90,4)</code> for a plot extending from pole to pole.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>longitudelim</code>, <code>latitudelim</code> and <code>projection</code> are all given,
then these arguments are passed to <code><a href="#topic+mapPlot">mapPlot()</a></code> to produce the plot.
(The call uses <code>bg</code> for <code>col</code>, and uses <code>col</code>, <code>fill</code>
and <code>border</code> directly.) If the results need further customization,
users should use <code><a href="#topic+mapPlot">mapPlot()</a></code> directly.
</p>
<p>If <code>projection</code> is provided without <code>longitudelim</code> or <code>latitudelim</code>,
then <code><a href="#topic+mapPlot">mapPlot()</a></code> is still called, but <code>longitudelim</code> and
<code>latitudelim</code> are computed from <code>clongitude</code>, <code>clatitude</code> and <code>span</code>.
</p>
<p>If <code>projection</code> is not provided, much simpler plots are produced. These are
Cartesian, with aspect ratio set to minimize shape distortion at the central latitude.
Although these are crude, they have the benefit of always working, which cannot
be said of true map projections, which can be problematic in various ways owing
to difficulties in inverting projection calculations.
</p>
<p>To get an inset map inside another map, draw the first map, do
<code><a href="graphics.html#topic+par">par</a></code><code>(new=TRUE)</code>, and then call <code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method()</a></code> with a value of
<code>mar</code> that moves the inset plot to a desired location on the existing
plot, and with <code>bg="white"</code>.
</p>


<h3>Value</h3>

<p>None.
</p>


<h3>History</h3>

<p>Until February, 2016, <code style="white-space: pre;">&#8288;plot,coastline-method&#8288;</code> relied on a
now-defunct argument <code>fill</code> to control colors; <code>col</code> is to be
used now, instead.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for the <a href="#topic+coastline-class">coastline</a> class explains the
structure of coastline objects, and also outlines the other functions
dealing with them.
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
par(mar = c(2, 2, 1, 1))
data(coastlineWorld)
plot(coastlineWorld)
plot(coastlineWorld, clongitude = -63.6, clatitude = 44.6, span = 1000)

# Canada in Lambert projection
plot(coastlineWorld,
    clongitude = -95, clatitude = 65, span = 5500,
    grid = 10, projection = "+proj=laea +lon_0=-100 +lat_0=55"
)


</code></pre>

<hr>
<h2 id='plot+2Cctd-method'>Plot a ctd Object</h2><span id='topic+plot+2Cctd-method'></span><span id='topic+plot.ctd'></span>

<h3>Description</h3>

<p>Plot CTD data in any of many different ways. In many cases,
the best choice is to use default values for all parameters
other than the first.  This yields a 4-panel plot that
displays a basic overview of the data, with a combined
profile of salinity and temperature at the top
left, a combined plot of density and the square of buoyancy
frequency at top right, a TS diagram at bottom left, and
a map at bottom right.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ctd'
plot(
  x,
  which,
  col = par("fg"),
  fill,
  borderCoastline = NA,
  colCoastline = "lightgray",
  eos = getOption("oceEOS", default = "gsw"),
  ref.lat = NaN,
  ref.lon = NaN,
  grid = TRUE,
  coastline = "best",
  Slim,
  Clim,
  Tlim,
  plim,
  densitylim,
  sigmalim,
  N2lim,
  Rrholim,
  dpdtlim,
  timelim,
  drawIsobaths = FALSE,
  clongitude,
  clatitude,
  span,
  showHemi = TRUE,
  lonlabels = TRUE,
  latlabels = TRUE,
  latlon.pch = 20,
  latlon.cex = 1.5,
  latlon.col = "red",
  projection = NULL,
  cex = 1,
  cex.axis = par("cex.axis"),
  pch = 1,
  useSmoothScatter = FALSE,
  df,
  keepNA = FALSE,
  type,
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1.5, mgp[1] + 1.5, mgp[1] + 1.5, mgp[1] + 1),
  inset = FALSE,
  add = FALSE,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cctd-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_which">which</code></td>
<td>
<p>a numeric or character vector specifying desired plot types. If
<code>which</code> is not supplied, a default will be used. This default depends on
<code>deploymentType</code> in the <code>metadata</code> slot of <code>x</code>.  If <code>deploymentType</code> is
<code>"profile"</code> or missing, then <code>which</code> defaults to <code>c(1, 2, 3, 5)</code>.  If
<code>deploymentType</code> is <code>"moored"</code> or <code>"thermosalinograph"</code> then <code>which</code> defaults
to <code>c(30, 3, 31, 5)</code>. Finally, if <code>deploymentType</code> is <code>towyo</code> then <code>which</code>
defaults to <code>c(30, 31, 32, 3)</code>.
</p>
<p>The details of individual <code>which</code> values are as follows. Some of the entries
refer to the EOS (equation of state for seawater), which may either <code>"gsw"</code>
for the modern Gibbs Seawater system, or <code>"unesco"</code> for the older UNESCO
system. The EOS may be set with the <code>eos</code> argument to <code style="white-space: pre;">&#8288;plot,ctd-method()&#8288;</code> or
by using <code><a href="base.html#topic+options">options()</a></code>, with <code>options(oceEOS="unesco")</code> or
<code>options(oceEOS="unesco")</code>. The default EOS is <code>"gsw"</code>.
</p>

<ul>
<li> <p><code>which=1</code> or <code>which="salinity+temperature"</code> gives a combined profile of
temperature and salinity.  If the EOS is <code>"gsw"</code> then Conservative
Temperature and Absolute Salinity are shown; otherwise in-situ temperature
and practical salinity are shown.
</p>
</li>
<li> <p><code>which=2</code> or <code>which="density+N2"</code> gives a combined profile of density
anomaly, computed with <code><a href="#topic+swSigma0">swSigma0()</a></code>, along with the square of the buoyancy
frequency, computed with <code><a href="#topic+swN2">swN2()</a></code>. The <code>eos</code> parameter is passed
to each of these functions, so the desired EOS is used.
</p>
</li>
<li> <p><code>which=3</code> or <code>which="TS"</code> gives a TS plot. If the EOS is <code>"gsw"</code>, T is
Conservative Temperature and S is Absolute Salinity; otherwise, they are
in-situ temperature and practical salinity, respectively.
</p>
</li>
<li> <p><code>which=4</code> or <code>which="text"</code> gives a textual  summary of
some aspects of the data.
</p>
</li>
<li> <p><code>which=5</code> or <code>which="map"</code> gives a map plotted with
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method()</a></code>, with a dot for the station location.  Notes near
the top boundary of the map give the station number, the sampling date, and
the name of the chief scientist, if these are known. Note that the longitude
will be converted to a value between -180 and 180 before plotting.  (See also
notes about <code>span</code>.)
</p>
</li>
<li> <p><code>which=5.1</code> as for <code>which=5</code>, except that the file name
is drawn above the map.
</p>
</li>
<li> <p><code>which=6</code> or <code>which="density+dpdt"</code> gives a profile of density and
<code class="reqn">dP/dt</code>, which is useful for evaluating whether the instrument is
dropping properly through the water column.  If the EOS is <code>"gsw"</code> then
<code class="reqn">\sigma_0</code> is shown; otherwise,
<code class="reqn">\sigma_\theta</code> is shown.
</p>
</li>
<li> <p><code>which=7</code> or <code>which="density+time"</code> gives a profile of density and time.
</p>
</li>
<li> <p><code>which=8</code> or <code>which="index"</code> gives a profile of index number, which can
provide useful information for trimming with <code><a href="#topic+ctdTrim">ctdTrim()</a></code>.
</p>
</li>
<li> <p><code>which=9</code> or <code>which="salinity"</code> gives a profile of Absolute
Salinity if the EOS is <code>"gsw"</code>, or practical salinity otherwise.
</p>
</li>
<li> <p><code>which=10</code> or <code>which="temperature"</code> gives a profile
of Conservative Temperature if the EOS is <code>"gsw"</code>, or
in-situ temperature otherwise.
</p>
</li>
<li> <p><code>which=11</code> or <code>which="density"</code> gives a profile
of density as computed with <code><a href="#topic+swRho">swRho()</a></code>, to which the <code>eos</code>
parameter is passed.
</p>
</li>
<li> <p><code>which=12</code> or <code>which="N2"</code> gives an <code class="reqn">N^2</code> profile.
</p>
</li>
<li> <p><code>which=13</code> or <code>which="spice"</code> gives a spiciness profile.
</p>
</li>
<li> <p><code>which=14</code> or <code>which="tritium"</code> gives a tritium profile.
</p>
</li>
<li> <p><code>which=15</code> or <code>which="Rrho"</code> gives a diffusive-case density
ratio profile.
</p>
</li>
<li> <p><code>which=16</code> or <code>which="RrhoSF"</code> gives a salt-finger case
density ratio profile.
</p>
</li>
<li> <p><code>which=17</code> or <code>which="conductivity"</code> gives a conductivity profile.
</p>
</li>
<li> <p><code>which=20</code> or <code>which="CT"</code> gives a profile of Conservative Temperature.
</p>
</li>
<li> <p><code>which=21</code> or <code>which="SA"</code> gives a profile of Absolute Salinity.
</p>
</li>
<li> <p><code>which=30</code> or <code>which="Sts"</code> gives a time series of Salinity Absolute
Salinity if the EOS is <code>"gsw"</code> or practical salinity otherwise.
</p>
</li>
<li> <p><code>which=31</code> or <code>which="Tts"</code> gives a time series of Conservative Temperature
if the EOS is <code>"gsw"</code> or in-situ temperature otherwise.
</p>
</li>
<li> <p><code>which=32</code> or <code>which="pts"</code> gives a time series of pressure
</p>
</li>
<li> <p><code>which=33</code> or <code>which="rhots"</code> gives a time series of density anomaly,
<code class="reqn">\sigma_0</code> if the EOS is <code>"gsw"</code>
or <code class="reqn">\sigma_\theta</code> otherwise.
</p>
</li>
<li><p> otherwise, <code>which</code> is interpreted as a character value to be checked
against the <code>data</code> and <code>dataDerived</code> fields returned by <code style="white-space: pre;">&#8288;x[["?"]&#8288;</code>. If a match
is found then a profile of the corresponding quantity is plotted.  If there
is no match, an error is reported.
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_col">col</code></td>
<td>
<p>color of lines or symbols.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_fill">fill</code></td>
<td>
<p>a legacy parameter that will be permitted only temporarily; see
&ldquo;History&rdquo;.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_bordercoastline">borderCoastline</code></td>
<td>
<p>color of coastlines and international borders, passed
to <code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method()</a></code> if a map is included in <code>which</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_colcoastline">colCoastline</code></td>
<td>
<p>fill color of coastlines and international borders, passed
to <code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method()</a></code> if a map is included in <code>which</code>. Set to
<code>NULL</code> to avoid filling.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_eos">eos</code></td>
<td>
<p>character value indicating the equation of state to be used, either
<code>"unesco"</code> or <code>"gsw"</code>.  The default is to use a value stored with
<code><a href="base.html#topic+options">options()</a></code> as e.g. <code>options(oceEOS="unesco")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_ref.lat">ref.lat</code></td>
<td>
<p>latitude of reference point for distance calculation.
The permitted range is -90 to 90.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_ref.lon">ref.lon</code></td>
<td>
<p>longitude of reference point for distance calculation.
The permitted range is -180 to 180.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_grid">grid</code></td>
<td>
<p>logical value indicating whether to draw a grid on the plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_coastline">coastline</code></td>
<td>
<p>a specification of the coastline to be used for
<code>which="map"</code>.  This may be a coastline object, whether built-in or
supplied by the user, or a character string.  If the later, it may be the
name of a built-in coastline (<code>"coastlineWorld"</code>,
<code>"coastlineWorldFine"</code>, or
<code>"coastlineWorldCoarse"</code>), or <code>"best"</code>, to choose
a suitable coastline for the locale, or <code>"none"</code> to prevent
the drawing of a coastline.  There is a speed penalty for providing
<code>coastline</code> as a character string, because it forces
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method()</a></code> to load it on every call.  So, if
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method()</a></code> is to be called several times for a given
coastline, it makes sense to load it in before the first call, and to
supply the object as an argument, as opposed to the name of the object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_slim">Slim</code>, <code id="plot+2B2Cctd-method_+3A_clim">Clim</code>, <code id="plot+2B2Cctd-method_+3A_tlim">Tlim</code>, <code id="plot+2B2Cctd-method_+3A_plim">plim</code>, <code id="plot+2B2Cctd-method_+3A_densitylim">densitylim</code>, <code id="plot+2B2Cctd-method_+3A_sigmalim">sigmalim</code>, <code id="plot+2B2Cctd-method_+3A_n2lim">N2lim</code>, <code id="plot+2B2Cctd-method_+3A_rrholim">Rrholim</code>, <code id="plot+2B2Cctd-method_+3A_dpdtlim">dpdtlim</code>, <code id="plot+2B2Cctd-method_+3A_timelim">timelim</code></td>
<td>
<p>optional
numeric vectors of length 2, that give axis limits for salinity (or Absolute
Salinity, if <code>eos</code> is <code>"gsw"</code>), conductivity, in-situ or potential
temperature (or Conservative Temperature, if <code>eos</code> is &lsquo;&quot;gsw&quot;&rsquo;), pressure,
density, density anomaly (either sigma-theta or sigma0),
square of buoyancy frequency, density ratio, dp/dt, and time, respectively.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_drawisobaths">drawIsobaths</code></td>
<td>
<p>logical value indicating whether to draw depth contours on
maps, in addition to the coastline. The argument has no effect except
for panels in which the value of <code>which</code> equals <code>"map"</code> or
the equivalent numerical code, <code>5</code>. If <code>drawIsobaths</code> is
<code>FALSE</code>, then no contours are drawn. If <code>drawIsobaths</code>
is <code>TRUE</code>, then contours are selected automatically,
using <a href="base.html#topic+pretty">pretty</a><code>(c(0, 300))</code> if the station depth is
under 100m or <a href="base.html#topic+pretty">pretty</a><code>(c(0, 5500))</code> otherwise.
If <code>drawIsobaths</code> is a numerical vector,
then the indicated depths are drawn. For plots drawn with <code>projection</code>
set to <code>NULL</code>, the contours are added with <code><a href="graphics.html#topic+contour">contour()</a></code>
and otherwise <code><a href="#topic+mapContour">mapContour()</a></code> is used. To customize
the resultant contours, e.g. setting particular line types or colors,
users should call these functions directly (see e.g. Example 2).</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_clongitude">clongitude</code>, <code id="plot+2B2Cctd-method_+3A_clatitude">clatitude</code>, <code id="plot+2B2Cctd-method_+3A_span">span</code></td>
<td>
<p>controls for the map area view,
used only if <code>which="map"</code>.
<code>clongitude</code> and <code>clatitude</code> specify the centre of the view, and
<code>span</code> specifies the approximate extend of the view, in
kilometres.  (If <code>span</code> is not given, it is be determined as a small
multiple of the distance to the nearest point of land, in an attempt to show
the station in familiar geographical context.)</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_showhemi">showHemi</code>, <code id="plot+2B2Cctd-method_+3A_lonlabels">lonlabels</code>, <code id="plot+2B2Cctd-method_+3A_latlabels">latlabels</code></td>
<td>
<p>controls for axis labelling, used only if <code>which="map"</code>.
<code>showHemi</code> is logical value indicating whether to show hemisphere in axis tick
labels. <code>lonlabels</code> and <code>latlabels</code> are numeric and character values that control the
axis labelling.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_latlon.pch">latlon.pch</code>, <code id="plot+2B2Cctd-method_+3A_latlon.cex">latlon.cex</code>, <code id="plot+2B2Cctd-method_+3A_latlon.col">latlon.col</code></td>
<td>
<p>controls for station location,
used only if <code>which="map"</code>.  <code>latlon.pch</code> sets the symbol code,
<code>latlon.cex</code> sets the character expansion
factor, and <code>latlon.col</code> sets the colour.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_projection">projection</code></td>
<td>
<p>controls the map projection (if any), and ignored unless
<code>which="map"</code>. The possibilities are as follows.  (1) If <code>projection=NULL</code>
(the default) then no projection will be used; the map will simply show
longitude and latitude in a Cartesian frame, scaled to retain shapes at the
centre.  (2) If <code style="white-space: pre;">&#8288;projection=&#8288;</code>&quot;automatic&quot;<code style="white-space: pre;">&#8288;then either a Mercator or Stereographic projection will be used, depending on whether the CTD station is within 70 degrees of the equator or at higher latitudes. (3) If&#8288;</code>projection' is a string in the format used by <code><a href="#topic+mapPlot">mapPlot()</a></code>, then it is is
passed to that function.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_cex">cex</code></td>
<td>
<p>size to be used for plot symbols (see <code><a href="graphics.html#topic+par">par()</a></code>).</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_cex.axis">cex.axis</code></td>
<td>
<p>size factor for axis labels (see <code><a href="graphics.html#topic+par">par()</a></code>).</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_pch">pch</code></td>
<td>
<p>code for plotting symbol (see <code><a href="graphics.html#topic+par">par()</a></code>).</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_usesmoothscatter">useSmoothScatter</code></td>
<td>
<p>logical value indicating whether to
use <code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code> instead of <code><a href="graphics.html#topic+plot">plot()</a></code> to draw the plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_df">df</code></td>
<td>
<p>optional numeric argument that is ignored except for plotting buoyancy
frequency; in that case, it is passed to <code><a href="#topic+swN2">swN2()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_keepna">keepNA</code></td>
<td>
<p>logical value indicating whether <code>NA</code> values
will yield breaks in lines drawn if <code>type</code> is <code>b</code>, <code>l</code>, or <code>o</code>.
The default value is <code>FALSE</code>.  Setting <code>keepNA</code> to <code>TRUE</code>
can be helpful when working with multiple profiles
strung together into one <a href="#topic+ctd-class">ctd</a> object, which otherwise
would have extraneous lines joining the deepest point in one
profile to the shallowest in the next profile.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_type">type</code></td>
<td>
<p>the type of plot to draw, using the same scheme as
<code><a href="graphics.html#topic+plot">plot()</a></code>. If supplied, this is increased to be the
same length as <code>which</code>, if necessary, and then supplied to
each of the individual plot calls. If it is not supplied,
then those plot calls use defaults (e.g. using a line for
<code><a href="#topic+plotProfile">plotProfile()</a></code>, using dots for <code><a href="#topic+plotTS">plotTS()</a></code>,
etc).</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_mgp">mgp</code></td>
<td>
<p>three-element numerical vector specifying axis-label geometry,
passed to <code><a href="graphics.html#topic+par">par()</a></code>.
The default establishes tighter margins than in the usual R setup.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_mar">mar</code></td>
<td>
<p>four-element numerical vector specifying margin geometry,
passed to <code><a href="graphics.html#topic+par">par()</a></code>.
The default establishes tighter margins than in the usual R setup.
Note that the value of <code>mar</code> is ignored for the map panel
of multi-panel maps; instead, the present value of
<a href="graphics.html#topic+par">par</a><code>("mar")</code> is used, which in the default call will
make the map plot region equal that of the previously-drawn
profiles and TS plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_inset">inset</code></td>
<td>
<p>logical value indicating whether this
function is being used as an inset.  The
effect is to prevent the present function from adjusting margins, which is
necessary because margin adjustment is the basis for the method used by
<code><a href="#topic+plotInset">plotInset()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_add">add</code></td>
<td>
<p>logical value indicating whether to add to an existing plot.  This
only works if <code>length(which)=1</code>, and it will yield odd results if the
value of <code>which</code> does not match that in the previous plots.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
<tr><td><code id="plot+2B2Cctd-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The
default values of <code>which</code> and other arguments are chosen to be useful
for quick overviews of data. However, for detailed work it is common
to call the present function with just a single value of <code>which</code>, e.g.
with four calls to get four panels. The advantage of this is that it provides
much more control over the display, and also it permits the addition of extra
display elements (lines, points, margin notes, etc.) to the individual panels.
</p>
<p>Note that panels that draw more than one curve (e.g. <code>which="salinity+temperature"</code>
draws temperature and salinity profiles in one graph), the value of <a href="graphics.html#topic+par">par</a><code>("usr")</code>
is established by the second profile to have been drawn. Some experimentation will
reveal what this profile is, for each permitted <code>which</code> case, although
it seems unlikely that this will help much ... the simple fact is that drawing two
profiles in one graph is useful for a quick overview, but not useful for e.g. interactive
analysis with <code><a href="graphics.html#topic+locator">locator()</a></code> to flag bad data, etc.
</p>


<h3>History of Changes</h3>


<ul>
<li><p> January 2022:
</p>

<ul>
<li><p> Add ability to profile anything stored in the <code>data</code> slot, and anything
that can be computed from information in that slot. The list of
possibilities is found by examining the <code>data</code> and <code>dataDerived</code> elements
of <code>x[["?"]]</code>.
</p>
</li>
<li><p> Drop the <code>lonlim</code> and <code>latlim</code> parameters, marked for removal in 2014;
use <code>clongitude</code>, <code>clatitude</code> and <code>span</code> instead (see
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method()</a></code>).
</p>
</li></ul>

</li>
<li><p> February 2016:
</p>

<ul>
<li><p> Drop the <code>fill</code> parameter for land colour; use <code>colCoastline</code> instead.
</p>
</li>
<li><p> Add the <code>borderCoastline</code> argument, to control the colour of coastlines
and international boundaries.
</p>
</li></ul>

</li></ul>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for <a href="#topic+ctd-class">ctd</a> explains the structure of CTD
objects, and also outlines the other functions dealing with them.
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. simple plot
library(oce)
data(ctd)
plot(ctd)

# 2. how to customize depth contours
par(mfrow = c(1, 2))
data(section)
stn &lt;- section[["station", 105]]
plot(stn, which = "map", drawIsobaths = TRUE)
plot(stn, which = "map")
data(topoWorld)
tlon &lt;- topoWorld[["longitude"]]
tlat &lt;- topoWorld[["latitude"]]
tdep &lt;- -topoWorld[["z"]]
contour(tlon, tlat, tdep,
    drawlabels = FALSE,
    levels = seq(1000, 6000, 1000), col = "lightblue", add = TRUE
)
contour(tlon, tlat, tdep,
    vfont = c("sans serif", "bold"),
    levels = stn[["waterDepth"]], col = "red", lwd = 2, add = TRUE
)

</code></pre>

<hr>
<h2 id='plot+2Cechosounder-method'>Plot an echosounder Object</h2><span id='topic+plot+2Cechosounder-method'></span><span id='topic+plot.echosounder'></span>

<h3>Description</h3>

<p>Plot echosounder data.
Simple linear approximation is used when a <code>newx</code> value is specified
with the <code>which=2</code> method, but arguably a gridding method should be
used, and this may be added in the future.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'echosounder'
plot(
  x,
  which = 1,
  beam = "a",
  newx,
  xlab,
  ylab,
  xlim,
  ylim,
  zlim,
  type = "l",
  col,
  lwd = 2,
  despike = FALSE,
  drawBottom,
  ignore = 5,
  drawTimeRange = FALSE,
  drawPalette = TRUE,
  radius,
  coastline,
  mgp = getOption("oceMgp"),
  mar = c(mgp[1], mgp[1] + 1.5, mgp[2] + 1/2, 1/2),
  atTop,
  labelsTop,
  tformat,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+echosounder-class">echosounder</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_which">which</code></td>
<td>
<p>list of desired plot types: <code>which=1</code> or <code>which="zt
image"</code> gives a z-time image, <code>which=2</code> or <code>which="zx image"</code>
gives a z-distance image, and <code>which=3</code> or <code>which="map"</code> gives a
map showing the cruise track.  In the image plots, the display is of
<code><a href="base.html#topic+log10">log10()</a></code> of amplitude, trimmed to zero for any amplitude values
less than 1 (including missing values, which equal 0).  Add 10 to the
numeric codes to get the secondary data (non-existent for single-beam files,</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_beam">beam</code></td>
<td>
<p>a more detailed specification of the data to be plotted.  For
single-beam data, this may only be <code>"a"</code>.  For dual-beam data, this may
be <code>"a"</code> for the narrow-beam signal, or <code>"b"</code> for the wide-beam
signal.  For split-beam data, this may be <code>"a"</code> for amplitude,
<code>"b"</code> for x-angle data, or <code>"c"</code> for y-angle data.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_newx">newx</code></td>
<td>
<p>optional vector of values to appear on the horizontal axis if
<code>which=1</code>, instead of time.  This must be of the same length as the
time vector, because the image is remapped from time to <code>newx</code> using
<code><a href="stats.html#topic+approx">approx()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_xlab">xlab</code>, <code id="plot+2B2Cechosounder-method_+3A_ylab">ylab</code></td>
<td>
<p>optional labels for the horizontal and vertical axes; if
not provided, the labels depend on the value of <code>which</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_xlim">xlim</code></td>
<td>
<p>optional range for x axis.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_ylim">ylim</code></td>
<td>
<p>optional range for y axis.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_zlim">zlim</code></td>
<td>
<p>optional range for color scale.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_type">type</code></td>
<td>
<p>type of graph, <code>"l"</code> for line, <code>"p"</code> for points, or
<code>"b"</code> for both.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_col">col</code></td>
<td>
<p>a function providing the color scale for image plots.
This value is passed to <code><a href="#topic+imagep">imagep()</a></code>, which draws
the images.  Since <code><a href="#topic+imagep">imagep()</a></code> defaults <code>col</code> to
<code><a href="#topic+oceColorsViridis">oceColorsViridis()</a></code>, that is effectively also the default
for the present function. (Prior to 2023-03-18, the present
function defaulted <code>col</code> to <code><a href="#topic+oceColorsJet">oceColorsJet()</a></code>.)</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_lwd">lwd</code></td>
<td>
<p>line width (ignored if <code>type="p"</code>).</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_despike">despike</code></td>
<td>
<p>remove vertical banding by using <code><a href="stats.html#topic+smooth">smooth()</a></code> to
smooth across image columns, row by row.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_drawbottom">drawBottom</code></td>
<td>
<p>optional flag used for section images.  If <code>TRUE</code>,
then the bottom is inferred as a smoothed version of the ridge of highest
image value, and data below that are grayed out after the image is drawn.
If <code>drawBottom</code> is a color, then that color is used, instead of
white.  The bottom is detected with <code><a href="#topic+findBottom">findBottom()</a></code>, using the
<code>ignore</code> value described next.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_ignore">ignore</code></td>
<td>
<p>optional flag specifying the thickness in metres of a surface
region to be ignored during the bottom-detection process.  This is ignored
unless <code>drawBottom=TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_drawtimerange">drawTimeRange</code></td>
<td>
<p>if <code>TRUE</code>, the time range will be drawn at the
top.  Ignored except for <code>which=2</code>, i.e. distance-depth plots.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_drawpalette">drawPalette</code></td>
<td>
<p>if <code>TRUE</code>, the palette will be drawn.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_radius">radius</code></td>
<td>
<p>radius to use for maps; ignored unless <code>which=3</code> or
<code>which="map"</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_coastline">coastline</code></td>
<td>
<p>coastline to use for maps; ignored unless <code>which=3</code> or
<code>which="map"</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code><a href="graphics.html#topic+par">par</a></code><code>("mgp")</code>, and also
for <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_attop">atTop</code></td>
<td>
<p>optional vector of time values, for labels at the top of the
plot produced with <code>which=2</code>.  If <code>labelsTop</code> is provided, then it
will hold the labels.  If <code>labelsTop</code> is not provided, the labels will
be constructed with the <code><a href="base.html#topic+format">format()</a></code> function, and these may be
customized by supplying a <code>format</code> in the <code>...</code> arguments.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_labelstop">labelsTop</code></td>
<td>
<p>optional vector of character strings to be plotted above
the <code>atTop</code> times.  Ignored unless <code>atTop</code> was provided.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_tformat">tformat</code></td>
<td>
<p>optional argument passed to <code><a href="#topic+imagep">imagep()</a></code>, for plot
types that call that function.  (See <code><a href="base.html#topic+strptime">strptime()</a></code> for the format
used.)</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_debug">debug</code></td>
<td>
<p>set to an integer exceeding zero, to get debugging information
during processing.</p>
</td></tr>
<tr><td><code id="plot+2B2Cechosounder-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.  For example,
for maps, it is possible to specify the radius of the view in kilometres,
with <code>radius</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list is silently returned, containing <code>xat</code> and <code>yat</code>,
values that can be used by <code><a href="#topic+oce.grid">oce.grid()</a></code> to add a grid to the plot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, with extensive help from Clark Richards
</p>


<h3>See Also</h3>

<p>Other things related to echosounder data: 
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic+as.echosounder">as.echosounder</a>()</code>,
<code><a href="#topic+echosounder-class">echosounder-class</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+findBottom">findBottom</a>()</code>,
<code><a href="#topic+read.echosounder">read.echosounder</a>()</code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+summary+2Cechosounder-method">summary,echosounder-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(echosounder)
plot(echosounder, drawBottom = TRUE)

</code></pre>

<hr>
<h2 id='plot+2Cgps-method'>Plot a gps Object</h2><span id='topic+plot+2Cgps-method'></span><span id='topic+plot.gps'></span>

<h3>Description</h3>

<p>This function plots a gps object.  An attempt is made to use the whole space
of the plot, and this is done by limiting either the longitude range or the
latitude range, as appropriate, by modifying the eastern or northern limit,
as appropriate.
To get an inset map inside another map, draw the first map, do
<code>par(new=TRUE)</code>, and then call <code>plot.gps</code> with a value of
<code>mar</code> that moves the inset plot to a desired location on the existing
plot, and with <code>bg="white"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'gps'
plot(
  x,
  xlab = "",
  ylab = "",
  asp,
  clongitude,
  clatitude,
  span,
  projection,
  expand = 1,
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1, mgp[1] + 1, 1, 1),
  bg,
  axes = TRUE,
  cex.axis = par("cex.axis"),
  add = FALSE,
  inset = FALSE,
  geographical = 0,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cgps-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+gps-class">gps</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_xlab">xlab</code></td>
<td>
<p>label for x axis</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_ylab">ylab</code></td>
<td>
<p>label for y axis</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_asp">asp</code></td>
<td>
<p>Aspect ratio for plot.  The default is for <code>plot.gps</code> to set
the aspect ratio to give natural latitude-longitude scaling somewhere near
the centre latitude on the plot. Often, it makes sense to set <code>asp</code>
yourself, e.g. to get correct shapes at 45N, use
<code>asp=1/cos(45*pi/180)</code>.  Note that the land mass is not symmetric about
the equator, so to get good world views you should set <code>asp=1</code> or set
<code>ylim</code> to be symmetric about zero. Any given value of <code>asp</code> is
ignored, if <code>clongitude</code> and <code>clatitude</code> are given.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_clongitude">clongitude</code>, <code id="plot+2B2Cgps-method_+3A_clatitude">clatitude</code></td>
<td>
<p>optional center latitude of map, in decimal
degrees.  If both <code>clongitude</code> and <code>clatitude</code> are provided, then
any provided value of <code>asp</code> is ignored, and instead the plot aspect
ratio is computed based on the center latitude.  If <code>clongitude</code> and
<code>clatitude</code> are provided, then <code>span</code> must also be provided.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_span">span</code></td>
<td>
<p>optional suggested span of plot, in kilometers.  The suggestion
is an upper limit on the scale; depending on the aspect ratio of the
plotting device, the radius may be smaller than <code>span</code>.  A value for
<code>span</code> must be supplied, if <code>clongitude</code> and <code>clatitude</code> are
supplied.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_projection">projection</code></td>
<td>
<p>optional map projection to use (see
<code><a href="#topic+mapPlot">mapPlot()</a></code>); if not given, a cartesian frame is used, scaled so
that gps shapes near the centre of the plot are preserved.  If a projection
is provided, the coordinate system will bear an indirect relationship to
longitude and longitude, and further adornment of the plot must be done with
e.g.  <code><a href="#topic+mapPoints">mapPoints()</a></code> instead of <code><a href="graphics.html#topic+points">points()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_expand">expand</code></td>
<td>
<p>numerical factor for the expansion of plot limits, showing
area outside the plot, e.g. if showing a ship track as a gps, and then an
actual gps to show the ocean boundary.  The value of <code>expand</code> is
ignored if either <code>xlim</code> or <code>ylim</code> is given.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code>par(mgp)</code>, and also
for <code>par(mar)</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_bg">bg</code></td>
<td>
<p>optional color to be used for the background of the map.  This
comes in handy for drawing insets (see &ldquo;details&rdquo;).</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_axes">axes</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to plot axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_cex.axis">cex.axis</code></td>
<td>
<p>value for axis font size factor.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_add">add</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to draw the gps on an existing plot.
Note that this retains the aspect ratio of that existing plot, so it is
important to set that correctly, e.g. with <code>asp=1/cos(lat * pi / 180)</code>,
where <code>clat</code> is the central latitude of the plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_inset">inset</code></td>
<td>
<p>set to <code>TRUE</code> for use within <code><a href="#topic+plotInset">plotInset()</a></code>.  The
effect is to prevent the present function from adjusting margins, which is
necessary because margin adjustment is the basis for the method used by
<code><a href="#topic+plotInset">plotInset()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_geographical">geographical</code></td>
<td>
<p>flag indicating the style of axes.  If
<code>geographical=0</code>, the axes are conventional, with decimal degrees as
the unit, and negative signs indicating the southern and western
hemispheres.  If <code>geographical=1</code>, the signs are dropped, with axis
values being in decreasing order within the southern and western
hemispheres.  If <code>geographical=2</code>, the signs are dropped and the axes
are labelled with degrees, minutes and seconds, as appropriate.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_debug">debug</code></td>
<td>
<p>set to <code>TRUE</code> to get debugging information during
processing.</p>
</td></tr>
<tr><td><code id="plot+2B2Cgps-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.  For example,
set <code>yaxp=c(-90,90,4)</code> for a plot extending from pole to pole.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to gps data: 
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic+as.gps">as.gps</a>()</code>,
<code><a href="#topic+gps-class">gps-class</a></code>,
<code><a href="#topic+read.gps">read.gps</a>()</code>,
<code><a href="#topic+summary+2Cgps-method">summary,gps-method</a></code>
</p>

<hr>
<h2 id='plot+2Cladp-method'>Plot an ladp Object</h2><span id='topic+plot+2Cladp-method'></span><span id='topic+plot.ladp'></span>

<h3>Description</h3>

<p>Uses <code><a href="#topic+plotProfile">plotProfile()</a></code> to create panels of depth variation of easterly
and northerly velocity components.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ladp'
plot(x, which = c("u", "v"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cladp-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+ladp-class">ladp</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cladp-method_+3A_which">which</code></td>
<td>
<p>a character vector storing names of items to be plotted.</p>
</td></tr>
<tr><td><code id="plot+2B2Cladp-method_+3A_...">...</code></td>
<td>
<p>Other arguments, passed to plotting functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ladp data: 
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic+as.ladp">as.ladp</a>()</code>,
<code><a href="#topic+ladp-class">ladp-class</a></code>,
<code><a href="#topic+summary+2Cladp-method">summary,ladp-method</a></code>
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>

<hr>
<h2 id='plot+2Clandsat-method'>Plot a landsat Object</h2><span id='topic+plot+2Clandsat-method'></span><span id='topic+plot.landsat'></span>

<h3>Description</h3>

<p>Plot the data within a landsat image, or information computed from the
data. The second category includes possibilities such as an estimate of
surface temperature and the <code>"terralook"</code> estimate of a natural-color
view.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'landsat'
plot(
  x,
  band,
  which = 1,
  decimate = TRUE,
  zlim,
  utm = FALSE,
  col = oce.colorsPalette,
  drawPalette = TRUE,
  showBandName = TRUE,
  alpha.f = 1,
  red.f = 1.7,
  green.f = 1.5,
  blue.f = 6,
  offset = c(0, -0.05, -0.2, 0),
  transform = diag(c(red.f, green.f, blue.f, alpha.f)),
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Clandsat-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+landsat-class">landsat</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_band">band</code></td>
<td>
<p>If given, the name of the band.  For Landsat-8 data, this may be
one of: <code>"aerosol"</code>, <code>"blue"</code>, <code>"green"</code>, <code>"red"</code>,
<code>"nir"</code>, <code>"swir1"</code>, <code>"swir2"</code>, <code>"panchromatic"</code>,
<code>"cirrus"</code>, <code>"tirs1"</code>, or <code>"tirs2"</code>.  For Landsat-7 data,
this may be one of <code>"blue"</code>, <code>"green"</code>, <code>"red"</code>,
<code>"nir"</code>, <code>"swir1"</code>, <code>"tirs1"</code>, <code>"tirs2"</code>,
<code>"swir2"</code>, or <code>"panchromatic"</code>.  For Landsat data prior to
Landsat-7, this may be one of <code>"blue"</code>, <code>"green"</code>,
<code>"red"</code>, <code>"nir"</code>, <code>"swir1"</code>, <code>"tirs1"</code>,
<code>"tirs2"</code>, or <code>"swir2"</code>.  If <code>band</code> is not given, the
(<code>"tirs1"</code>) will be used if it exists in the object data, or
otherwise the first band will be used.  In addition to the above, using
<code>band="temperature"</code> will plot an estimate of at-satellite
brightness temperature, computed from the <code>tirs1</code> band, and
<code>band="terralook"</code> will plot a sort of natural color by combining
the <code>red</code>, <code>green</code>, <code>blue</code> and <code>nir</code> bands
according to the formula provided at
<code style="white-space: pre;">&#8288;https://lta.cr.usgs.gov/terralook/what_is_terralook&#8288;</code> (a
website that worked once, but failed as of Feb 2, 2017).</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_which">which</code></td>
<td>
<p>Desired plot type; 1=image, 2=histogram.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_decimate">decimate</code></td>
<td>
<p>An indication of the desired decimation,
passed to <code><a href="#topic+imagep">imagep()</a></code> for image plots.
The default yields faster plotting.  Some decimation is sensible for
full-size images, since no graphical displays can show 16 thousand pixels
on a side.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_zlim">zlim</code></td>
<td>
<p>Either a pair of numbers giving the limits for the colorscale,
or <code>"histogram"</code> to have a flattened histogram (i.e. to maximally
increase contrast throughout the domain.)  If not given, the 1 and 99
percent quantiles are calculated and used as limits.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_utm">utm</code></td>
<td>
<p>A logical value indicating whether to use UTS (easting and northing) instead
of longitude and latitude on plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_col">col</code></td>
<td>
<p>Either a function yielding colors, taking a single integer
argument with the desired number of colors, or the string
<code>"natural"</code>, which combines the information in the <code>red</code>,
<code>green</code> and <code>blue</code> bands and produces a natural-hue image.  In
the latter case, the band designation is ignored, and the object must
contain the three color bands.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_drawpalette">drawPalette</code></td>
<td>
<p>Indication of the type of palette to draw, if
any. See <code><a href="#topic+imagep">imagep()</a></code> for details.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_showbandname">showBandName</code></td>
<td>
<p>A logical indicating whether the band name is to
plotted in the top margin, near the right-hand side.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_alpha.f">alpha.f</code></td>
<td>
<p>Argument used if <code>col="natural"</code>, to adjust colors
with <code><a href="grDevices.html#topic+adjustcolor">adjustcolor()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_red.f">red.f</code></td>
<td>
<p>Argument used if <code>col="natural"</code>, to adjust colors with
<code><a href="grDevices.html#topic+adjustcolor">adjustcolor()</a></code>.  Higher values of <code>red.f</code> cause red hues
to be emphasized (e.g. dry land).</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_green.f">green.f</code></td>
<td>
<p>Argument used if <code>col="natural"</code>, to adjust colors with
<code><a href="grDevices.html#topic+adjustcolor">adjustcolor()</a></code>.  Higher values of <code>green.f</code> emphasize
green hues (e.g. forests).</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_blue.f">blue.f</code></td>
<td>
<p>Argument used if <code>band="terralook"</code>, to adjust colors with
<code><a href="grDevices.html#topic+adjustcolor">adjustcolor()</a></code>.  Higher values of <code>blue.f</code> emphasize blue
hues (e.g. ocean).</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_offset">offset</code></td>
<td>
<p>Argument used if <code>band="terralook"</code>, to adjust colors with
<code><a href="grDevices.html#topic+adjustcolor">adjustcolor()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_transform">transform</code></td>
<td>
<p>Argument used if <code>band="terralook"</code>, to adjust colors
with <code><a href="grDevices.html#topic+adjustcolor">adjustcolor()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_debug">debug</code></td>
<td>
<p>Set to a positive value to get debugging information during
processing.</p>
</td></tr>
<tr><td><code id="plot+2B2Clandsat-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For Landsat-8 data, the <code>band</code> may be
one of: <code>"aerosol"</code>, <code>"blue"</code>, <code>"green"</code>, <code>"red"</code>,
<code>"nir"</code>, <code>"swir1"</code>, <code>"swir2"</code>, <code>"panchromatic"</code>,
<code>"cirrus"</code>, <code>"tirs1"</code>, or <code>"tirs2"</code>.
</p>
<p>For Landsat-7 data, <code>band</code> may be one of <code>"blue"</code>, <code>"green"</code>, <code>"red"</code>,
<code>"nir"</code>, <code>"swir1"</code>, <code>"tirs1"</code>, <code>"tirs2"</code>,
<code>"swir2"</code>, or <code>"panchromatic"</code>.
</p>
<p>For Landsat data prior to
Landsat-7, <code>band</code> may be one of <code>"blue"</code>, <code>"green"</code>,
<code>"red"</code>, <code>"nir"</code>, <code>"swir1"</code>, <code>"tirs1"</code>,
<code>"tirs2"</code>, or <code>"swir2"</code>.
</p>
<p>If <code>band</code> is not given, the
(<code>"tirs1"</code>) will be used if it exists in the object data, or
otherwise the first band will be used.
</p>
<p>In addition to the above there are also some pseudo-bands that
can be plotted, as follows.
</p>

<ul>
<li><p> Setting <code>band="temperature"</code> will plot an estimate
of at-satellite brightness temperature, computed from the
<code>tirs1</code> band.
</p>
</li>
<li><p> Setting <code>band="terralook"</code> will plot a sort of natural
color by combining the <code>red</code>, <code>green</code>, <code>blue</code> and
<code>nir</code> bands according to the formula provided at
<code style="white-space: pre;">&#8288;https://lta.cr.usgs.gov/terralook/what_is_terralook&#8288;</code> (a
website that worked once, but failed as of Feb 2, 2017), namely
that the <code>red</code>-band data are provided as the <code>red</code>
argument of the <code><a href="grDevices.html#topic+rgb">rgb()</a></code> function, while
the <code>green</code> argument is computed as
2/3 of the <code>green</code>-band data plus 1/3 of the <code>nir</code>-band data, and
the <code>blue</code> argument is computed as 2/3 of the <code>green</code>-band
data minus 1/3 of the <code>nir</code>-band data. (This is not a typo: the
<code>blue</code> band is not used.)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to landsat data: 
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic+landsat-class">landsat-class</a></code>,
<code><a href="#topic+landsatAdd">landsatAdd</a>()</code>,
<code><a href="#topic+landsatTrim">landsatTrim</a>()</code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+read.landsat">read.landsat</a>()</code>,
<code><a href="#topic+summary+2Clandsat-method">summary,landsat-method</a></code>
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>

<hr>
<h2 id='plot+2Clisst-method'>Plot a lisst Object</h2><span id='topic+plot+2Clisst-method'></span><span id='topic+plot.lisst'></span>

<h3>Description</h3>

<p>Creates a multi-panel summary plot of data measured by LISST instrument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lisst'
plot(x, which = c(16, 37, 38), tformat, debug = getOption("oceDebug"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Clisst-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+lisst-class">lisst</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Clisst-method_+3A_which">which</code></td>
<td>
<p>list of desired plot types.  These are graphed in panels
running down from the top of the page.  See &ldquo;Details&rdquo; for the
meanings of various values of <code>which</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Clisst-method_+3A_tformat">tformat</code></td>
<td>
<p>optional argument passed to <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code>, for
plot types that call that function.  (See <code><a href="base.html#topic+strptime">strptime()</a></code> for the
format used.)</p>
</td></tr>
<tr><td><code id="plot+2B2Clisst-method_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  The value indicates the depth
within the call stack to which debugging applies.</p>
</td></tr>
<tr><td><code id="plot+2B2Clisst-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The panels are controlled by the <code>which</code> argument, as follows.
</p>

<ul>
<li> <p><code>which=1</code> to <code>32</code>, or <code>which="C1"</code> to <code>"C32"</code> for
a time-series graph of the named column (a size class).
</p>
</li>
<li> <p><code>which=33</code> or <code>which="lts"</code> for a time-series plot of laser
transmission sensor.
</p>
</li>
<li> <p><code>which=34</code> or <code>which="voltage"</code> for a time-series plot of
instrument voltage.
</p>
</li>
<li> <p><code>which=35</code> or <code>which="aux"</code> for a time-series plot of the
external auxiliary input.
</p>
</li>
<li> <p><code>which=36</code> or <code>which="lrs"</code> for a time-series plot of the
laser reference sensor.
</p>
</li>
<li> <p><code>which=37</code> or <code>which="pressure"</code> for a time-series plot of
pressure.
</p>
</li>
<li> <p><code>which=38</code> or <code>which="temperature"</code> for a time-series plot
of temperature.
</p>
</li>
<li> <p><code>which=41</code> or <code>which="transmission"</code> for a time-series plot
of transmission, in percent.
</p>
</li>
<li> <p><code>which=42</code> or <code>which="beam"</code> for a time-series plot of
beam-C, in 1/metre.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for <a href="#topic+lisst-class">lisst</a> explains the
structure of lisst objects, and also outlines the other functions dealing
with them.
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to lisst data: 
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic+as.lisst">as.lisst</a>()</code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+read.lisst">read.lisst</a>()</code>,
<code><a href="#topic+summary+2Clisst-method">summary,lisst-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(lisst)
plot(lisst)

</code></pre>

<hr>
<h2 id='plot+2Clobo-method'>Plot a lobo object</h2><span id='topic+plot+2Clobo-method'></span><span id='topic+plot.lobo'></span>

<h3>Description</h3>

<p>Plot a summary diagram for lobo data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lobo'
plot(
  x,
  which = c(1, 2, 3),
  mgp = getOption("oceMgp"),
  mar = c(mgp[2] + 1, mgp[1] + 1, 1, mgp[1] + 1.25),
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Clobo-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+lobo-class">lobo</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Clobo-method_+3A_which">which</code></td>
<td>
<p>A vector of numbers or character strings, indicating the
quantities to plot.  These are stacked in a single column.  The possible
values for <code>which</code> are as follows: <code>1</code> or <code>"temperature"</code> for
a time series of temperature; <code>2</code> or <code>"salinity"</code> for salinity;
<code>3</code> or <code>"TS"</code> for a TS diagram (which uses <code>eos="unesco"</code>),
<code>4</code> or <code>"u"</code> for a
timeseries of the u component of velocity; <code>5</code> or <code>"v"</code> for a
timeseries of the v component of velocity; <code>6</code> or <code>"nitrate"</code> for
a timeseries of nitrate concentration; <code>7</code> or <code>"fluorescence"</code> for
a timeseries of fluorescence value.</p>
</td></tr>
<tr><td><code id="plot+2B2Clobo-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code>par(mgp)</code>, and also
for <code>par(mar)</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Clobo-method_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Clobo-method_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
<tr><td><code id="plot+2B2Clobo-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to lobo data: 
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic+as.lobo">as.lobo</a>()</code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+read.lobo">read.lobo</a>()</code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+summary+2Clobo-method">summary,lobo-method</a></code>
</p>

<hr>
<h2 id='plot+2Cmet-method'>Plot a met Object</h2><span id='topic+plot+2Cmet-method'></span><span id='topic+plot.met'></span>

<h3>Description</h3>

<p>Creates a multi-panel summary plot of data measured in a meteorological data
set.  cast. The panels are controlled by the <code>which</code> argument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'met'
plot(x, which = 1:4, mgp, mar, tformat, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cmet-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+met-class">met</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cmet-method_+3A_which">which</code></td>
<td>
<p>list of desired plot types.
</p>

<ul>
<li> <p><code>which=1</code> gives a time-series plot of temperature
</p>
</li>
<li> <p><code>which=2</code> gives a time-series plot of pressure
</p>
</li>
<li> <p><code>which=3</code> gives a time-series plot of the x (eastward) component of velocity
</p>
</li>
<li> <p><code>which=4</code> gives a time-series plot of the y (northward) component of velocity
</p>
</li>
<li> <p><code>which=5</code> gives a time-series plot of speed
</p>
</li>
<li> <p><code>which=6</code> gives a time-series plot of direction (degrees clockwise from north;
note that the values returned by <code>met[["direction"]]</code> must be multiplied by 10
to get the direction plotted)
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot+2B2Cmet-method_+3A_mgp">mgp</code></td>
<td>
<p>A 3-element numerical vector used with <code><a href="graphics.html#topic+par">par</a></code><code>("mgp")</code>
to control the spacing of axis elements.  The default is tighter than the R default.</p>
</td></tr>
<tr><td><code id="plot+2B2Cmet-method_+3A_mar">mar</code></td>
<td>
<p>A 4-element numerical vector used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>
to control the plot margins. The default is tighter than the R default.</p>
</td></tr>
<tr><td><code id="plot+2B2Cmet-method_+3A_tformat">tformat</code></td>
<td>
<p>optional argument passed to <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code>, for
plot types that call that function.  (See <code><a href="base.html#topic+strptime">strptime()</a></code> for the
format used.)</p>
</td></tr>
<tr><td><code id="plot+2B2Cmet-method_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If more than one panel is drawn, then on exit from <code>plot.met</code>, the
value of <code>par</code> will be reset to the value it had before the function
call.  However, if only one panel is drawn, the adjustments to <code>par</code>
made within <code>plot.met</code> are left in place, so that further additions may
be made to the plot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to met data: 
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic+as.met">as.met</a>()</code>,
<code><a href="#topic+download.met">download.met</a>()</code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+read.met">read.met</a>()</code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+summary+2Cmet-method">summary,met-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(met)
plot(met, which = 3:4)

# Wind speed and direction during Hurricane Juan
# Compare with the final figure in a white paper by Chris Fogarty
# (available at http://www.novaweather.net/Hurricane_Juan_files/McNabs_plot.pdf
# downloaded 2017-01-02).
library(oce)
data(met)
t0 &lt;- as.POSIXct("2003-09-29 04:00:00", tz = "UTC")
dt &lt;- 12 * 3600
juan &lt;- subset(met, t0 - dt &lt;= time &amp; time &lt;= t0 + dt)
par(mfrow = c(2, 1))
plot(juan, which = 5)
abline(v = t0)
plot(juan, which = 6)
abline(v = t0)

</code></pre>

<hr>
<h2 id='plot+2Coce-method'>Plot an oce Object</h2><span id='topic+plot+2Coce-method'></span><span id='topic+plot.oce'></span>

<h3>Description</h3>

<p>This creates a <code><a href="graphics.html#topic+pairs">pairs()</a></code> plot of the elements in the <code>data</code>
slot, if there are more than 2 elements there, or a simple xy plot if 2
elements, or a histogram if 1 element.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'oce'
plot(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Coce-method_+3A_x">x</code></td>
<td>
<p>a basic <a href="#topic+oce-class">oce</a> object,
but not from any subclass that derive from this base, because
subclasses have their own plot methods, e.g. calling <code>plot()</code> on a
<a href="#topic+ctd-class">ctd</a> object dispatches to <code><a href="#topic+plot+2Cctd-method">plot,ctd-method()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Coce-method_+3A_y">y</code></td>
<td>
<p>Ignored; only present here because S4 object for generic <code>plot</code>
need to have a second parameter before the <code>...</code> parameter.</p>
</td></tr>
<tr><td><code id="plot+2B2Coce-method_+3A_...">...</code></td>
<td>
<p>Passed to <code><a href="graphics.html#topic+hist">hist()</a></code>, <code><a href="graphics.html#topic+plot">plot()</a></code>, or to
<code><a href="graphics.html#topic+pairs">pairs()</a></code>, according to whichever does the plotting.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
o &lt;- new("oce")
o &lt;- oceSetData(o, "x", rnorm(10))
o &lt;- oceSetData(o, "y", rnorm(10))
o &lt;- oceSetData(o, "z", rnorm(10))
plot(o)
</code></pre>

<hr>
<h2 id='plot+2Codf-method'>Plot an odf Object</h2><span id='topic+plot+2Codf-method'></span><span id='topic+plot.odf'></span>

<h3>Description</h3>

<p>Plot data contained within an ODF object,
using <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code> to create panels of time-series plots for
all the columns contained in the <code>odf</code> object (or just those that
contain at least one finite value, if <code>blanks</code> is <code>FALSE</code>).
If the object's <code>data</code> slot does not contain <code>time</code>, then
<code><a href="graphics.html#topic+pairs">pairs()</a></code> is used to plot all the elements in the <code>data</code>
slot that contain at least one finite value.
These actions are both
crude and there are no arguments to control the behaviour, but this
function is really just a stop-gap measure, since in practical work
<a href="#topic+odf-class">odf</a> objects are usually cast to other types, and those types
tend to have more useful plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'odf'
plot(x, blanks = TRUE, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Codf-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+odf-class">odf</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Codf-method_+3A_blanks">blanks</code></td>
<td>
<p>A logical value that indicates whether to include dummy
plots for data items that lack any finite values.</p>
</td></tr>
<tr><td><code id="plot+2B2Codf-method_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to odf data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ODF2oce">ODF2oce</a>()</code>,
<code><a href="#topic+ODFListFromHeader">ODFListFromHeader</a>()</code>,
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.odf">read.odf</a>()</code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+summary+2Codf-method">summary,odf-method</a></code>
</p>

<hr>
<h2 id='plot+2Crsk-method'>Plot a rsk Object</h2><span id='topic+plot+2Crsk-method'></span><span id='topic+plot.rsk'></span>

<h3>Description</h3>

<p>Rsk data may be in many forms, and it is not easy to devise a general plotting
strategy for all of them. The present function is quite crude, on the
assumption that users will understand their own datasets, and that they can
devise plots that are best-suited to their applications.  Sometimes, the
sensible scheme is to coerce the object into another form, e.g. using
<code>plot(as.ctd(rsk))</code> if the object contains CTD-like data.  Other times,
users should extract data from the <code>rsk</code> object and construct plots
themselves. The idea is to use the present function mainly to get an overview,
and for that reason, the default plot type (set by <code>which</code>) is a set of
time-series plots, because the one thing that is definitely known about
<code>rsk</code> objects is that they contain a <code>time</code> vector in their
<code>data</code> slot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'rsk'
plot(
  x,
  which = "timeseries",
  tlim,
  ylim,
  xlab,
  ylab,
  tformat,
  drawTimeRange = getOption("oceDrawTimeRange"),
  abbreviateTimeRange = getOption("oceAbbreviateTimeRange"),
  useSmoothScatter = FALSE,
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1.5, mgp[1] + 1.5, 1.5, 1.5),
  main = "",
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Crsk-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+rsk-class">rsk</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_which">which</code></td>
<td>
<p>character indicating desired plot types.  These are
graphed in panels running down from the top of the page.  See
&ldquo;Details&rdquo; for the meanings of various values of
<code>which</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_tlim">tlim</code></td>
<td>
<p>optional limits for time axis.  If not provided, the value will be
inferred from the data.</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_ylim">ylim</code></td>
<td>
<p>optional limits for the y axis.  If not provided, the
value will be inferred from the data.  (It is helpful to
specify this, if the auto-scaled value will be inappropriate,
e.g. if more lines are to be added later). Note that this is
ignored, unless <code>length(which) == 1</code> and <code>which</code>
corresponds to one of the data fields. If a multipanel plot of
a specific subset of the data fields is desired with
<code>ylim</code> control, it should be done panel by panel (see
Examples).</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_xlab">xlab</code></td>
<td>
<p>optional label for x axis.</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_ylab">ylab</code></td>
<td>
<p>optional label for y axis.</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_tformat">tformat</code></td>
<td>
<p>optional argument passed to <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code>, for plot
types that call that function.  (See <code><a href="base.html#topic+strptime">strptime()</a></code> for the format
used.)</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_drawtimerange">drawTimeRange</code></td>
<td>
<p>boolean that applies to panels with time as the horizontal
axis, indicating whether to draw the time range in the top-left margin of the
plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_abbreviatetimerange">abbreviateTimeRange</code></td>
<td>
<p>boolean that applies to panels with time as the
horizontal axis, indicating whether to abbreviate the second time in the time
range (e.g. skipping the year, month, day, etc. if it's the same as the start
time).</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_usesmoothscatter">useSmoothScatter</code></td>
<td>
<p>a boolean to cause <code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code> to be
used for profile plots, instead of <code><a href="graphics.html#topic+plot">plot()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code><a href="graphics.html#topic+par">par</a></code><code>("mgp")</code>, and
also for <code>par(mar)</code>, computed from this.  The default is tighter than the
R default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_main">main</code></td>
<td>
<p>main title for plot, used just on the top panel, if there are several panels.</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging, if it exceeds 0.</p>
</td></tr>
<tr><td><code id="plot+2B2Crsk-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots produced are time series plots of the data in the
object. The default, <code>which="timeseries"</code> plots all data
fields, and over-rides any other specification. Specific fields
can be plotted by naming the field,
e.g. <code>which="temperature"</code> to plot a time series of just
the temperature field.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>See Also</h3>

<p>The documentation for <a href="#topic+rsk-class">rsk</a> explains the structure of
<code>rsk</code> objects, and also outlines the other functions dealing with them.
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic+as.rsk">as.rsk</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+read.rsk">read.rsk</a>()</code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+rskPatm">rskPatm</a>()</code>,
<code><a href="#topic+rskToc">rskToc</a>()</code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(rsk)
# 1. default timeseries plot of all data fields
plot(rsk)
# 2. plot in ctd format
plot(as.ctd(rsk))

</code></pre>

<hr>
<h2 id='plot+2Csatellite-method'>Plot a satellite Object</h2><span id='topic+plot+2Csatellite-method'></span><span id='topic+plot.satellite'></span>

<h3>Description</h3>

<p>For an example using <code>g1sst</code> data, see <code><a href="#topic+read.g1sst">read.g1sst()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'satellite'
plot(x, y, asp, debug = getOption("oceDebug"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Csatellite-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+satellite-class">satellite</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Csatellite-method_+3A_y">y</code></td>
<td>
<p>String indicating the quantity to be plotted.</p>
</td></tr>
<tr><td><code id="plot+2B2Csatellite-method_+3A_asp">asp</code></td>
<td>
<p>Optional aspect ratio for plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Csatellite-method_+3A_debug">debug</code></td>
<td>
<p>A debugging flag, integer.</p>
</td></tr>
<tr><td><code id="plot+2B2Csatellite-method_+3A_...">...</code></td>
<td>
<p>extra arguments passed to <code><a href="#topic+imagep">imagep()</a></code>, e.g. set
<code>col</code> to control colors.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>

<hr>
<h2 id='plot+2Csealevel-method'>Plot a sealevel Object</h2><span id='topic+plot+2Csealevel-method'></span><span id='topic+plot.sealevel'></span>

<h3>Description</h3>

<p>Creates a plot for a sea-level dataset, in one of two varieties.  Depending
on the length of <code>which</code>, either a single-panel or multi-panel plot is
drawn.  If there is just one panel, then the value of <code>par</code> used in
<code style="white-space: pre;">&#8288;plot,sealevel-method&#8288;</code> is retained upon exit, making it convenient to add to
the plot.  For multi-panel plots, <code>par</code> is returned to the value it had
before the call.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'sealevel'
plot(
  x,
  which = 1:3,
  drawTimeRange = getOption("oceDrawTimeRange"),
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 0.5, mgp[1] + 1.5, mgp[2] + 1, mgp[2] + 3/4),
  marginsAsImage = FALSE,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Csealevel-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+sealevel-class">sealevel</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Csealevel-method_+3A_which">which</code></td>
<td>
<p>a numerical or string vector indicating desired plot types,
with possibilities 1 or <code>"all"</code> for a time-series of all the elevations, 2 or
<code>"month"</code> for a time-series of just the first month, 3 or
<code>"spectrum"</code> for a power spectrum (truncated to frequencies below 0.1
cycles per hour, or 4 or <code>"cumulativespectrum"</code> for a cumulative
integral of the power spectrum.</p>
</td></tr>
<tr><td><code id="plot+2B2Csealevel-method_+3A_drawtimerange">drawTimeRange</code></td>
<td>
<p>boolean that applies to panels with time as the
horizontal axis, indicating whether to draw the time range in the top-left
margin of the plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Csealevel-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code><a href="graphics.html#topic+par">par</a></code><code>("mgp")</code>, and also
for <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Csealevel-method_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Csealevel-method_+3A_marginsasimage">marginsAsImage</code></td>
<td>
<p>boolean, <code>TRUE</code> to put a wide margin to the right
of time-series plots, matching the space used up by a palette in an
<code><a href="#topic+imagep">imagep()</a></code> plot.</p>
</td></tr>
<tr><td><code id="plot+2B2Csealevel-method_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging, if it exceeds 0.</p>
</td></tr>
<tr><td><code id="plot+2B2Csealevel-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Historical Note</h3>

<p>Until 2020-02-06, sea-level plots had the mean value removed, and indicated
with a tick mark and margin note on the right-hand side of the plot.
This behaviour was confusing. The change did not go through the usual
deprecation process, because the margin-note behaviour had not
been documented.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>The example refers to Hurricane Juan, which caused a great deal
of damage to Halifax in 2003.  Since this was in the era of the digital
photo, a casual web search will uncover some spectacular images of damage,
from both wind and storm surge.
Landfall, within 30km of this sealevel
gauge, was between 00:10 and 00:20 Halifax local time on Monday, Sept 29,
2003.
</p>


<h3>See Also</h3>

<p>The documentation for the <a href="#topic+sealevel-class">sealevel</a> class explains the
structure of sealevel objects, and also outlines the other functions dealing
with them.
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to sealevel data: 
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic+as.sealevel">as.sealevel</a>()</code>,
<code><a href="#topic+read.sealevel">read.sealevel</a>()</code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+summary+2Csealevel-method">summary,sealevel-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(sealevel)
# local Halifax time is UTC + 4h
juan &lt;- as.POSIXct("2003-09-29 00:15:00", tz = "UTC") + 4 * 3600
plot(sealevel, which = 1, xlim = juan + 86400 * c(-7, 7))
abline(v = juan, col = "red")

</code></pre>

<hr>
<h2 id='plot+2Csection-method'>Plot a section Object</h2><span id='topic+plot+2Csection-method'></span><span id='topic+plot.section'></span>

<h3>Description</h3>

<p>Creates a summary plot for a CTD section, with one panel for each value of
<code>which</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'section'
plot(
  x,
  which = c(1, 2, 3, 99),
  eos,
  at = NULL,
  labels = TRUE,
  grid = FALSE,
  contourLevels = NULL,
  contourLabels = NULL,
  stationIndices,
  coastline = "best",
  colLand = "gray",
  xlim = NULL,
  ylim = NULL,
  zlim = NULL,
  zbreaks = NULL,
  zcol = NULL,
  map.xlim = NULL,
  map.ylim = NULL,
  clongitude,
  clatitude,
  span,
  projection = NULL,
  xtype = "distance",
  ytype = "depth",
  ztype = "contour",
  longitude0,
  latitude0,
  legend.loc = "bottomright",
  legend.text = NULL,
  showStations = FALSE,
  showStart = TRUE,
  stationTicks = TRUE,
  showBottom = TRUE,
  showSpine = TRUE,
  drawPalette = TRUE,
  axes = TRUE,
  mgp,
  mar,
  col,
  cex,
  pch,
  lwd,
  labcex = par("cex"),
  debug = getOption("oceDebug", 0),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Csection-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+section-class">section</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_which">which</code></td>
<td>
<p>a list of desired plot types, as explained in &ldquo;Details&rdquo;.
Plot types not listed in &ldquo;Details&rdquo; can be generated using the
name of the data in the <a href="#topic+section-class">section</a> object.
There may be up to four panels in total, and the desired plots are placed in
these panels, in reading order.  If only one panel is plotted, <code>par</code> is
not adjusted, which makes it easy to add to the plot with subsequent plotting
commands.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_eos">eos</code></td>
<td>
<p>Character indication of the seawater equation of state
to use. The permitted choices are <code>"gsw"</code> and <code>"unesco"</code>. If <code>eos</code> is
not supplied, it defaults to <code><a href="base.html#topic+getOption">getOption</a></code><code style="white-space: pre;">&#8288;("oceEOS",default="gsw")&#8288;</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_at">at</code></td>
<td>
<p>If <code>NULL</code> (the default), the x axis will indicate the distance
of the stations from the first in the section.  (This may give errors in the
contouring routine, if the stations are not present in a geographical order.)
If a list, then it indicates the values at which stations will be plotted.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_labels">labels</code></td>
<td>
<p>Either a logical, indicating whether to put labels on the x axis,
or a vector that is a list of labels to be placed at the x positions indicated
by <code>at</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_grid">grid</code></td>
<td>
<p>If <code>TRUE</code>, points are drawn at data locations.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_contourlevels">contourLevels</code></td>
<td>
<p>Optional contour levels.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_contourlabels">contourLabels</code></td>
<td>
<p>Optional contour labels.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_stationindices">stationIndices</code></td>
<td>
<p>Optional list of the indices of stations to use.  Note
that an index is <em>not</em> a station number, e.g. to show the first 4
stations, use <code>station.indices=1:4</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_coastline">coastline</code></td>
<td>
<p>Either a <a href="#topic+coastline-class">coastline</a> object to be used,
or a string.  In the second case, the permitted
choices are <code>"best"</code> (the default) to pick
a variant that suits the scale, <code>"coastlineWorld"</code> for the coarse
version that is provided by <a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a>,
<code>"coastlineWorldMedium"</code> or <code>"coastlineWorldFine"</code> for two
coastlines provided by the <a href="https://CRAN.R-project.org/package=ocedata"><span class="pkg">ocedata</span></a> package, or <code>"none"</code>, to avoid
drawing a coastline.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_colland">colLand</code></td>
<td>
<p>colour used to fill in land areas if <code>which</code> is <code>"map"</code>;
ignored otherwise.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_xlim">xlim</code></td>
<td>
<p>Optional limit for x axis (only in sections, not map).</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_ylim">ylim</code></td>
<td>
<p>Optional limit for y axis (only in sections, not map)</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_zlim">zlim</code>, <code id="plot+2B2Csection-method_+3A_zbreaks">zbreaks</code>, <code id="plot+2B2Csection-method_+3A_zcol">zcol</code></td>
<td>
<p>Elements that control colours for <code>image</code> and <code>points</code>
plot types, i.e. if <code>ztype</code> is either <code>"points"</code> or <code>"image"</code>.
<code>zlim</code> is a two-element numerical vector specifying the
limit on the plotted field.  If not provided, it defaults to the data
range.
<code>zbreaks</code> controls the colour breaks, in a manner that is similar to
the <code><a href="Matrix.html#topic+image">image()</a></code> parameter named <code>breaks</code>.  If not provided, <code>zbreaks</code> is
inferred from <code>zlim</code>.
<code>zcol</code>, which controls the colour scheme, may be a vector of colours
(of length 1 less than <code>zbreaks</code>), or a function that takes an
integer as its sole argument and returns that number of colours.
If not provided, <code>zcol</code> defaults to <code><a href="#topic+oceColorsViridis">oceColorsViridis()</a></code>.
These three parameters are used in Example 6, an illustration of
Atlantic salinity along 36N.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_map.xlim">map.xlim</code>, <code id="plot+2B2Csection-method_+3A_map.ylim">map.ylim</code></td>
<td>
<p>Optional limits for station map; <code>map.ylim</code> is
ignored if <code>map.xlim</code> is provided.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_clongitude">clongitude</code>, <code id="plot+2B2Csection-method_+3A_clatitude">clatitude</code>, <code id="plot+2B2Csection-method_+3A_span">span</code></td>
<td>
<p>Optional map centre position and span (km).</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_projection">projection</code></td>
<td>
<p>Parameter specifying map
projection; see <code><a href="#topic+mapPlot">mapPlot()</a></code>.  If <code>projection="automatic"</code>,
however, a projection is devised from the data, with <code>stereographic</code> if
the mean latitude exceeds 70N and <code>mollweide</code> otherwise.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_xtype">xtype</code></td>
<td>
<p>Type of x axis, for contour plots, either <code>"distance"</code> for
distance (in km) to the first point in the section, <code>"track"</code> for distance
along the cruise track, <code>"longitude"</code>, <code>"latitude"</code>,
<code>"time"</code> or <code>"spine"</code> (distance along a spine that was added
with <code><a href="#topic+addSpine">addSpine()</a></code>).  Note that if the x values are not in order, they will be put in
order, and since that might not make physical sense, a warning will be issued.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_ytype">ytype</code></td>
<td>
<p>Type of y axis for contour plots, either <code>"pressure"</code> for
pressure (in dbar, with zero at the surface) or <code>"depth"</code> for depth (in m
below the surface, calculated from pressure with <code><a href="#topic+swDepth">swDepth()</a></code>).</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_ztype">ztype</code></td>
<td>
<p>String indicating whether to how to indicate the &quot;z&quot;
data (in the R sense, i.e. this could be salinity, temperature, etc; it does
not mean the vertical coordinate) The choices are: <code>"contour"</code> for
contours, <code>"image"</code> for an image (drawn with <code><a href="#topic+imagep">imagep()</a></code> with
<code>filledContours=TRUE</code>), or <code>"points"</code> to draw points.
In the first two cases, the data must be gridded, with identical pressures at
each station.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_longitude0">longitude0</code>, <code id="plot+2B2Csection-method_+3A_latitude0">latitude0</code></td>
<td>
<p>Location of the point from which distance is measured.
These values are ignored unless <code>xtype</code> is <code>"distance"</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_legend.loc">legend.loc</code></td>
<td>
<p>Location of legend, as supplied to <code><a href="graphics.html#topic+legend">legend()</a></code>, or
set to the empty string to avoid plotting a legend.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_legend.text">legend.text</code></td>
<td>
<p>character value indicating the text for the legend.
If this is NULL (the default) then the legend is automatically
constructed by <code><a href="#topic+labelWithUnit">labelWithUnit()</a></code>, based on the value of <code>which</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_showstations">showStations</code></td>
<td>
<p>Logical indicating whether to draw station numbers on maps.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_showstart">showStart</code></td>
<td>
<p>Logical indicating whether to indicate the first station with</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_stationticks">stationTicks</code></td>
<td>
<p>A logical value indicating whether to indicate station
locations with ticks at the top margin of cross-section plots. Setting this
parameter to <code>FALSE</code> frees the user up to do their own labelling
at this spot.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_showbottom">showBottom</code></td>
<td>
<p>a value indicating whether (and how) to indicate the
ocean bottom on cross-section views.  There are three possibilities.
(a) If <code>showBottom</code> is <code>FALSE</code>, then the bottom is not rendered.  If it
is <code>TRUE</code>, then the  bottom is rendered with a gray polygon.
(b) If <code>showBottom</code> is the character value <code>"polygon"</code>, then a polygon is drawn,
and similarly lines are drawn for <code>"lines"</code>, and points for <code>"points"</code>.
(c) If <code>showBottom</code> is a <a href="#topic+topo-class">topo</a> object, then the station locations are
interpolated to that topography and the results are shown with a polygon.
See &ldquo;Examples&rdquo;.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_showspine">showSpine</code></td>
<td>
<p>logical value used if <code>which="map"</code>.  If <code>showSpine</code> is
<code>TRUE</code> and <code>section</code> has had a spine added with <code><a href="#topic+addSpine">addSpine()</a></code>, then
the spine is drawn in blue.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_drawpalette">drawPalette</code></td>
<td>
<p>logical value indicating whether to draw a palette when <code>ztype="image"</code>
ignored otherwise.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_axes">axes</code></td>
<td>
<p>Logical value indicating whether to draw axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_mgp">mgp</code></td>
<td>
<p>A 3-element numerical vector to use for <code>par(mgp)</code>, and also for
<code>par(mar)</code>, computed from this. If not provided, this defaults to
<code><a href="base.html#topic+getOption">getOption</a></code><code>("oceMgp")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_mar">mar</code></td>
<td>
<p>Value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>. If not provided,
a default is set up.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_col">col</code></td>
<td>
<p>Color for line types.  If not provided, this defaults to
<code><a href="graphics.html#topic+par">par</a></code><code>("col")</code>.  See <code>zcol</code>, for <code>ztype="image"</code> and <code>ztype="points"</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_cex">cex</code></td>
<td>
<p>Numerical character-expansion factor, which defaults to <code><a href="graphics.html#topic+par">par</a></code><code>("cex")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_pch">pch</code></td>
<td>
<p>Indication of symbol type; defaults to <code><a href="graphics.html#topic+par">par</a></code><code>("pch")</code> for
non-map or to 3 for map.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_lwd">lwd</code></td>
<td>
<p>line width; defaults to <code><a href="graphics.html#topic+par">par</a></code><code>("lwd")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_labcex">labcex</code></td>
<td>
<p>Size of characters in contour labels (passed to
<code><a href="graphics.html#topic+contour">contour()</a></code>).</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If <code>debug</code> is not supplied, it defaults to
<code><a href="base.html#topic+getOption">getOption</a></code><code>("oceDebug")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Csection-method_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to the contouring function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The type of plot is governed by <code>which</code>, as follows.
</p>

<ul>
<li> <p><code>which=0</code> or <code>"potential temperature"</code> for potential temperature contours
</p>
</li>
<li> <p><code>which=1</code> or <code>"temperature"</code> for in-situ temperature contours (the default)
</p>
</li>
<li> <p><code>which=2</code> or <code>"salinity"</code> for salinity contours
</p>
</li>
<li> <p><code>which=3</code> or <code>"sigmaTheta"</code> for sigma-theta contours
</p>
</li>
<li> <p><code>which=4</code> or <code>"nitrate"</code> for nitrate concentration contours
</p>
</li>
<li> <p><code>which=5</code> or <code>"nitrite"</code> for nitrite concentration contours
</p>
</li>
<li> <p><code>which=6</code> or <code>"oxygen"</code> for oxygen concentration  contours
</p>
</li>
<li> <p><code>which=7</code> or <code>"phosphate"</code> for phosphate concentration contours
</p>
</li>
<li> <p><code>which=8</code> or <code>"silicate"</code> for silicate concentration contours
</p>
</li>
<li> <p><code>which=9</code> or <code>"u"</code> for eastward velocity
</p>
</li>
<li> <p><code>which=10</code> or <code>"uz"</code> for vertical derivative of eastward velocity
</p>
</li>
<li> <p><code>which=11</code> or <code>"v"</code> for northward velocity
</p>
</li>
<li> <p><code>which=12</code> or <code>"vz"</code> for vertical derivative of northward velocity
</p>
</li>
<li> <p><code>which=20</code> or <code>"data"</code> for a dot for each data location
</p>
</li>
<li> <p><code>which=99</code> or <code>"map"</code> for a location map
</p>
</li></ul>

<p>The y-axis for the contours is pressure, plotted in the conventional reversed
form, so that the water surface appears at the top of the plot.  The x-axis is
more complicated. If <code>at</code> is not supplied, then the routine calculates x
as the distance between the first station in the section and each of the other
stations. (This will produce an error if the stations are not ordered
geographically, because the <code><a href="graphics.html#topic+contour">contour()</a></code> routine cannot handle
non-increasing axis coordinates.) If <code>at</code> is specified, then it is taken
to be the location, in arbitrary units, along the x-axis of labels specified by
<code>labels</code>; the way this works is designed to be the same as for
<code><a href="graphics.html#topic+axis">axis()</a></code>.
</p>


<h3>Value</h3>

<p>If the original section was gridded, the return value is that section.
Otherwise, the gridded section that was constructed for the plot is returned.
In both cases, the value is returned silently. The
purpose of returning the section is to enable subsequent processing
of the grid, including adding elements to the plot (see example 5).
</p>


<h3>Ancillary Examples</h3>

<p>The following examples were once part of the &ldquo;Examples&rdquo;
section, but were moved here in May 2022, to reduce the build-check
time for CRAN submission.
</p>
<div class="sourceCode"><pre>library(oce)
data(section)
GS &lt;- subset(section, 113&lt;=stationId&amp;stationId&lt;=129)
GSg &lt;- sectionGrid(GS, p=seq(0, 2000, 100))

# Gulf Stream, salinity data and contoured
par(mfrow=c(2, 1))
plot(GS, which=1, ylim=c(2000, 0), ztype="points",
     zbreaks=seq(0,30,2), pch=20, cex=3)
plot(GSg, which=1, ztype="image", zbreaks=seq(0,30,2))

# Gulf Stream, temperature grid (image) and data (dots)
par(mfrow=c(1, 1))
plot(GSg, which=1, ztype="image")
T &lt;- GS[["temperature"]]
col &lt;- oceColorsViridis(100)[rescale(T, rlow=1, rhigh=100)]
points(GS[["distance"]],GS[["depth"]],pch=20,cex=3,col="white")
points(GS[["distance"]],GS[["depth"]],pch=20,cex=2.5,col=col)

# 4. Image of temperature, with a high-salinity contour on top;
#    note the Mediterranean water.
sec &lt;- plot(section, which="temperature", ztype="image")
S &lt;- sec[["salinity", "grid:distance-pressure"]]
contour(S$distance, S$pressure, S$field, level=35.8, lwd=3, add=TRUE)

# 5. Contours of salinity, with dots for high pressure and spice
plot(section, which="salinity")
distance &lt;- section[["distance"]]
depth &lt;- section[["depth"]]
spice &lt;- section[["spice"]]
look &lt;- spice &gt; 1.8 &amp; depth &gt; 500
points(distance[look], depth[look], col="red")

# Image of Absolute Salinity, with 4-minute bathymetry
# It's easy to calculate the desired area for the bathymetry,
# but for brevity we'll hard-code it. Note that download.topo()
# requires the "ncdf4" package to have been installed.
if (requireNamespace("ncdf4")) {
    f &lt;- download.topo(west=-80, east=0, south=35, north=40, resolution=4)
    t &lt;- read.topo(f)
    plot(section, which="SA", xtype="longitude", ztype="image", showBottom=t)
}

# Temperature with salinity added in red
plot(GSg, which="temperature")
distance &lt;- GSg[["distance", "byStation"]]
depth &lt;- GSg[["station", 1]][["depth"]]
S &lt;- matrix(GSg[["salinity"]], byrow=TRUE, nrow=length(GSg[["station"]]))
contour(distance, depth, S, col=2, add=TRUE)

# Image with controlled colours
plot(GSg, which="salinity", ztype="image",
    zlim=c(35, 37.5),
    zbreaks=seq(35, 37.5, 0.25),
    zcol=oceColorsTurbo)
</pre></div>


<h3>Author(s)</h3>

<p>Dan Kelley, with help from Clark Richards and Chantelle Layton.
</p>


<h3>See Also</h3>

<p>The documentation for <a href="#topic+section-class">section</a> explains the
structure of section objects, and also outlines the other functions dealing
with them.
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(section)
GS &lt;- subset(section, 113 &lt;= stationId &amp; stationId &lt;= 129)
GSg &lt;- sectionGrid(GS, p = seq(0, 2000, 100))

# Gulf Stream, salinity and temperature contours
plot(GSg, which = c("salinity", "temperature"))

# Gulf Stream, Temperature image
plot(GSg,
    which = "temperature", ztype = "image",
    zbreaks = seq(0, 25, 2), zcol = oceColorsTemperature
)

</code></pre>

<hr>
<h2 id='plot+2Ctidem-method'>Plot a tidem Object</h2><span id='topic+plot+2Ctidem-method'></span><span id='topic+plot.tidem'></span>

<h3>Description</h3>

<p>Plot a summary diagram for a tidal fit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'tidem'
plot(
  x,
  which = 1,
  constituents = c("SA", "O1", "K1", "M2", "S2", "M4"),
  sides = NULL,
  col = "blue",
  log = "",
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1, mgp[1] + 1, mgp[2] + 0.25, mgp[2] + 1),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Ctidem-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+tidem-class">tidem</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctidem-method_+3A_which">which</code></td>
<td>
<p>integer flag indicating plot type, 1 for stair-case spectral, 2
for spike spectral.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctidem-method_+3A_constituents">constituents</code></td>
<td>
<p>character vector holding the names of constituents that are
to be drawn and labelled. If <code>NULL</code>, then no constituents will be shown.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctidem-method_+3A_sides">sides</code></td>
<td>
<p>an integer vector of length equal to that of <code>constituents</code>,
designating the side on which the constituent labels are to be drawn. As in
all R graphics, the value <code>1</code> indicates the bottom of the plot, and
<code>3</code> indicates the top. If <code>sides=NULL</code>, the default, then all labels
are drawn at the top. Any value of <code>sides</code> that is not either 1 or 3
is converted to 3.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctidem-method_+3A_col">col</code></td>
<td>
<p>a character vector naming colors to be used for <code>constituents</code>.
Ignored if <code>sides=3</code>. Repeated to be of the same length as
<code>constituents</code>, otherwise.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctidem-method_+3A_log">log</code></td>
<td>
<p>if set to &quot;<code>x</code>&quot;, the frequency axis will be logarithmic.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctidem-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code>par(mgp)</code>, and also
for <code>par(mar)</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctidem-method_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code>[</code>par<code style="white-space: pre;">&#8288;]("mar")&#8288;</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctidem-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions, not all
of which are obeyed.  For example, if ... contains <code>type</code>, that value will be
ignored because it is set internally, according to the value of <code>which</code>.</p>
</td></tr>
</table>


<h3>Sample of Usage</h3>

<pre>
library(oce)
data(sealevel)
tide &lt;- tidem(sealevel)
plot(tide)
</pre>


<h3>Historical note</h3>

<p>An argument named <code>labelIf</code> was removed in July 2016,
because it was discovered never to have worked as documented, and
because the more useful argument <code>constituents</code> had been added.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>

<hr>
<h2 id='plot+2Ctopo-method'>Plot a topo Object</h2><span id='topic+plot+2Ctopo-method'></span><span id='topic+plot.topo'></span>

<h3>Description</h3>

<p>This plots contours of topographic elevation.  The plot aspect ratio is set
based on the middle latitude in the plot.  The line properties, such as
<code>land.lwd</code>, may either be a single item, or a vector; in the latter case,
the length must match the length of the corresponding properties, e.g.
<code>land.z</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'topo'
plot(
  x,
  xlab = "",
  ylab = "",
  asp,
  clongitude,
  clatitude,
  span,
  expand = 1.5,
  water.z,
  col.water,
  lty.water,
  lwd.water,
  land.z,
  col.land,
  lty.land,
  lwd.land,
  geographical = FALSE,
  location = "topright",
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1, mgp[1] + 1, 1, 1),
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Ctopo-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+topo-class">topo</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_xlab">xlab</code>, <code id="plot+2B2Ctopo-method_+3A_ylab">ylab</code></td>
<td>
<p>Character strings giving a label for the x and y axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_asp">asp</code></td>
<td>
<p>Aspect ratio for plot.  The default is for <code>plot.coastline</code> to
set the aspect ratio to give natural latitude-longitude scaling somewhere near
the centre latitude on the plot. Often, it makes sense to set <code>asp</code>
yourself, e.g. to get correct shapes at 45N, use <code>asp=1/cos(45*pi/180)</code>.
Note that the land mass is not symmetric about the equator, so to get good
world views you should set <code>asp=1</code> or set <code>ylim</code> to be symmetric
about zero.  Any given value of <code>asp</code> is ignored, if <code>clongitude</code> and
<code>clatitude</code> are given.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_clongitude">clongitude</code></td>
<td>
<p>Optional center longitude of map, in degrees east; see
<code>clatitude</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_clatitude">clatitude</code></td>
<td>
<p>Optional center latitude of map, in degrees north.  If this
and <code>clongitude</code> are provided, then any provided value of <code>asp</code> is
ignored, and instead the plot aspect ratio is computed based on the center
latitude.  Also, if <code>clongitude</code> and <code>clatitude</code> are provided, then
<code>span</code> must be, also.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_span">span</code></td>
<td>
<p>Optional suggested span of plot, in kilometers (must be supplied,
if <code>clongitude</code> and <code>clatitude</code> are supplied).</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_expand">expand</code></td>
<td>
<p>Numerical factor for the expansion of plot limits, showing area
outside the plot, e.g. if showing a ship track as a coastline, and then an
actual coastline to show the ocean boundary.  The value of <code>expand</code> is
ignored if either <code>xlim</code> or <code>ylim</code> is given.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_water.z">water.z</code></td>
<td>
<p>Depths at which to plot water contours.  If not provided, these
are inferred from the data.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_col.water">col.water</code></td>
<td>
<p>Colors corresponding to <code>water.z</code> values.  If not
provided, these will be <code>"fill"</code> colors from
<code><a href="#topic+oce.colorsGebco">oce.colorsGebco()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_lty.water">lty.water</code></td>
<td>
<p>Line type(s) for water contours.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_lwd.water">lwd.water</code></td>
<td>
<p>Line width(s) for water contours.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_land.z">land.z</code></td>
<td>
<p>Depths at which to plot land contours.  If not provided, these
are inferred from the data.  If set to <code>NULL</code>, no land contours will be
plotted.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_col.land">col.land</code></td>
<td>
<p>Colors corresponding to <code>land.z</code> values.  If not
provided, these will be <code>"fill"</code> colors from
<code><a href="#topic+oce.colorsGebco">oce.colorsGebco()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_lty.land">lty.land</code></td>
<td>
<p>Line type(s) for land contours.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_lwd.land">lwd.land</code></td>
<td>
<p>Line width(s) for land contours.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_geographical">geographical</code></td>
<td>
<p>Logical, indicating whether to plot latitudes and
longitudes without minus signs.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_location">location</code></td>
<td>
<p>Location for a legend (or <code>"none"</code>, for no legend).</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code>par(mgp)</code>, and also for
<code>par(mar)</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_mar">mar</code></td>
<td>
<p>Four-element numerical vector to be used with
<code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_debug">debug</code></td>
<td>
<p>Numerical value, with positive values indicating higher levels of
debugging.</p>
</td></tr>
<tr><td><code id="plot+2B2Ctopo-method_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to plotting functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to topo data: 
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic+as.topo">as.topo</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>,
<code><a href="#topic+read.topo">read.topo</a>()</code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+summary+2Ctopo-method">summary,topo-method</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+topoInterpolate">topoInterpolate</a>()</code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(topoWorld)
plot(topoWorld, clongitude = -60, clatitude = 45, span = 10000)

</code></pre>

<hr>
<h2 id='plot+2Cwindrose-method'>Plot a windrose Object</h2><span id='topic+plot+2Cwindrose-method'></span><span id='topic+plot.windrose'></span>

<h3>Description</h3>

<p>Plot a <a href="#topic+windrose-class">windrose</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'windrose'
plot(
  x,
  type = c("count", "mean", "median", "fivenum"),
  convention = c("meteorological", "oceanographic"),
  mgp = getOption("oceMgp"),
  mar = c(mgp[1], mgp[1], 1 + mgp[1], mgp[1]),
  col,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cwindrose-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+windrose-class">windrose</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cwindrose-method_+3A_type">type</code></td>
<td>
<p>The thing to be plotted, either the number of counts in the angle
interval, the mean of the values in the interval, the median of the values, or
a <code><a href="stats.html#topic+fivenum">fivenum()</a></code> representation of the values.</p>
</td></tr>
<tr><td><code id="plot+2B2Cwindrose-method_+3A_convention">convention</code></td>
<td>
<p>String indicating whether to use meteorological convention or
oceanographic convention for the arrows that emanate from the centre of the
rose.  In meteorological convection, an arrow emanates towards the right on
the diagram if the wind is from the east; in oceanographic convention, such an
arrow indicates flow <em>to</em> the east.</p>
</td></tr>
<tr><td><code id="plot+2B2Cwindrose-method_+3A_mgp">mgp</code></td>
<td>
<p>Three-element numerical vector to use for <code><a href="graphics.html#topic+par">par</a></code><code>(mgp)</code>, and also
for <code><a href="graphics.html#topic+par">par</a></code><code>(mar)</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Cwindrose-method_+3A_mar">mar</code></td>
<td>
<p>Four-element numerical vector to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cwindrose-method_+3A_col">col</code></td>
<td>
<p>Optional list of colors to use.  If not set, the colors will be
<code>c("red", "pink", "blue", "lightgray")</code>.  For the first three types of
plot, the first color in this list is used to fill in the rose, the third is
used for the petals of the rose, and the fourth is used for grid lines. For the
<code>"fivenum"</code> type, the region from the lower hinge to the first quartile is
coloured pink, the region from the first quartile to the third quartile
is coloured red, and the region from the third quartile to the upper hinge
is coloured pink.  Then the median is drawn in black.</p>
</td></tr>
<tr><td><code id="plot+2B2Cwindrose-method_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to windrose data: 
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic+as.windrose">as.windrose</a>()</code>,
<code><a href="#topic+summary+2Cwindrose-method">summary,windrose-method</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
set.seed(1234)
theta &lt;- seq(0, 360, 0.25)
x &lt;- 1 + cos(pi / 180 * theta) + rnorm(theta)
y &lt;- sin(pi / 180 * theta) + rnorm(theta)
wr &lt;- as.windrose(x, y)
plot(wr)
plot(wr, type = "fivenum")

</code></pre>

<hr>
<h2 id='plot+2Cxbt-method'>Plot an xbt Object</h2><span id='topic+plot+2Cxbt-method'></span><span id='topic+plot.xbt'></span>

<h3>Description</h3>

<p>Plots data contained in an <a href="#topic+xbt-class">xbt</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'xbt'
plot(
  x,
  which = 1,
  type = "l",
  mgp = getOption("oceMgp"),
  mar,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot+2B2Cxbt-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+xbt-class">xbt</a> object.</p>
</td></tr>
<tr><td><code id="plot+2B2Cxbt-method_+3A_which">which</code></td>
<td>
<p>list of desired plot types; see &ldquo;Details&rdquo; for the meanings of various
values of <code>which</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cxbt-method_+3A_type">type</code></td>
<td>
<p>type of plot, as for <code><a href="graphics.html#topic+plot">plot()</a></code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cxbt-method_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code>par(mgp)</code>, and also for
<code>par(mar)</code>, computed from this.  The default is tighter than the R default,
in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plot+2B2Cxbt-method_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plot+2B2Cxbt-method_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate amount
of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="plot+2B2Cxbt-method_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The panels are controlled by the <code>which</code> argument, with choices as follows.
</p>

<ul>
<li> <p><code>which=1</code> for a temperature profile as a function of depth.
</p>
</li>
<li> <p><code>which=2</code> for a soundSpeed profile as a function of depth.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to xbt data: 
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>,
<code><a href="#topic+as.xbt">as.xbt</a>()</code>,
<code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1</a>()</code>,
<code><a href="#topic+read.xbt">read.xbt</a>()</code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>,
<code><a href="#topic+summary+2Cxbt-method">summary,xbt-method</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(xbt)
summary(xbt)
plot(xbt)

</code></pre>

<hr>
<h2 id='plotInset'>Plot an Inset Diagram</h2><span id='topic+plotInset'></span>

<h3>Description</h3>

<p>Adds an inset diagram to an existing plot.  Note that if the inset is a map
or coastline, it will be necessary to supply <code>inset=TRUE</code> to prevent
the inset diagram from occupying the whole device width.  After
<code>plotInset()</code> has been called, any further plotting will take place
within the inset, so it is essential to finish a plot before drawing an
inset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotInset(
  xleft,
  ybottom,
  xright,
  ytop,
  expr,
  mar = c(2, 2, 1, 1),
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotInset_+3A_xleft">xleft</code></td>
<td>
<p>location of left-hand of the inset diagram, in the existing
plot units.  (PROVISIONAL FEATURE: this may also be <code>"bottomleft"</code>, to
put the inset there.  Eventually, other positions may be added.)</p>
</td></tr>
<tr><td><code id="plotInset_+3A_ybottom">ybottom</code></td>
<td>
<p>location of bottom side of the inset diagram, in the existing
plot units.</p>
</td></tr>
<tr><td><code id="plotInset_+3A_xright">xright</code></td>
<td>
<p>location of right-hand side of the inset diagram, in the
existing plot units.</p>
</td></tr>
<tr><td><code id="plotInset_+3A_ytop">ytop</code></td>
<td>
<p>location of top side of the inset diagram, in the existing plot
units.</p>
</td></tr>
<tr><td><code id="plotInset_+3A_expr">expr</code></td>
<td>
<p>An expression that draws the inset plot.  This may be a single
plot command, or a sequence of commands enclosed in curly braces.</p>
</td></tr>
<tr><td><code id="plotInset_+3A_mar">mar</code></td>
<td>
<p>margins, in line heights, to be used at the four sides of the
inset diagram.  (This is often helpful to save space.)</p>
</td></tr>
<tr><td><code id="plotInset_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# power law in linear and log form
x &lt;- 1:10
y &lt;- x^2
plot(x, y, log = "xy", type = "l")
plotInset(3, 1, 10, 8,
    expr = plot(x, y, type = "l", cex.axis = 3 / 4, mgp = c(3 / 2, 1 / 2, 0)),
    mar = c(2.5, 2.5, 1, 1)
)

# CTD data with location
data(ctd)
plot(ctd, which = "TS")
plotInset(29.9, 2.7, 31, 10,
    expr = plot(ctd,
        which = "map",
        coastline = "coastlineWorld",
        span = 5000, mar = NULL, cex.axis = 3 / 4
    )
)
</code></pre>

<hr>
<h2 id='plotPolar'>Draw a Polar Plot</h2><span id='topic+plotPolar'></span>

<h3>Description</h3>

<p>Creates a crude polar plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPolar(r, theta, debug = getOption("oceDebug"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotPolar_+3A_r">r</code></td>
<td>
<p>radii of points to plot.</p>
</td></tr>
<tr><td><code id="plotPolar_+3A_theta">theta</code></td>
<td>
<p>angles of points to plot, in degrees.</p>
</td></tr>
<tr><td><code id="plotPolar_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="plotPolar_+3A_...">...</code></td>
<td>
<p>optional arguments passed to the lower-level plotting
functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
r &lt;- rnorm(50, mean = 2, sd = 0.1)
theta &lt;- runif(50, 0, 360)
plotPolar(r, theta)
</code></pre>

<hr>
<h2 id='plotProfile'>Plot a ctd Profile</h2><span id='topic+plotProfile'></span>

<h3>Description</h3>

<p>Plot a profile, showing variation of some quantity (or quantities) with
pressure, using the oceanographic convention of putting lower pressures
nearer the top of the plot. This works for any <code>oce</code> object that has a
pressure column in its <code>data</code> slot.
The colors (<code>col.salinity</code>, etc.) are only used if two profiles appear
on a plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotProfile(
  x,
  xtype = "salinity+temperature",
  ytype = "pressure",
  eos = getOption("oceEOS", default = "gsw"),
  lty = 1,
  xlab = NULL,
  ylab = NULL,
  col = "black",
  col.salinity = "darkgreen",
  col.temperature = "red",
  col.rho = "blue",
  col.N2 = "brown",
  col.dpdt = "darkgreen",
  col.time = "darkgreen",
  pt.bg = "transparent",
  grid = TRUE,
  col.grid = "lightgray",
  lty.grid = "dotted",
  Slim,
  Clim,
  Tlim,
  densitylim,
  sigmalim,
  N2lim,
  Rrholim,
  dpdtlim,
  timelim,
  plim,
  xlim,
  ylim,
  lwd = par("lwd"),
  xaxs = "r",
  yaxs = "r",
  cex = 1,
  pch = 1,
  useSmoothScatter = FALSE,
  df,
  keepNA = FALSE,
  type = "l",
  mgp = getOption("oceMgp"),
  mar,
  add = FALSE,
  inset = FALSE,
  debug = getOption("oceDebug", 0),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotProfile_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_xtype">xtype</code></td>
<td>
<p>item(s) to be plotted on the x axis, either a character
value taken from the following list, or a numeric vector of length
matching the <code>pressure</code> field stored in <code>x</code>. (In the second case,
as of version 1.7-11, a label is auto-constructed, unless the user
supplied a character value for <code>xlab</code>.)
</p>

<ul>
<li> <p><code>"salinity"</code> Profile of salinity.
</p>
</li>
<li> <p><code>"conductivity"</code> Profile of conductivity.
</p>
</li>
<li> <p><code>"temperature"</code> Profile of <em>in-situ</em> temperature.
</p>
</li>
<li> <p><code>"theta"</code> Profile of potential temperature.
</p>
</li>
<li> <p><code>"density"</code> Profile of density (expressed as <code class="reqn">\sigma_\theta</code>).
</p>
</li>
<li> <p><code>"index"</code> Index of sample (useful for working with <code><a href="#topic+ctdTrim">ctdTrim()</a></code>).
</p>
</li>
<li> <p><code>"salinity+temperature"</code> Profile of salinity and temperature within a single axis frame.
</p>
</li>
<li> <p><code>"N2"</code> Profile of square of buoyancy frequency <code class="reqn">N^2</code>,
calculated with <code><a href="#topic+swN2">swN2()</a></code> with
an optional argument setting of <code>df=length(x[["pressure"]])/4</code> to do
some smoothing.
</p>
</li>
<li> <p><code>"density+N2"</code> Profile of sigma0 and
the square of buoyancy frequency within a single axis frame.
</p>
</li>
<li> <p><code>"density+dpdt"</code> Profile of sigma0 and dP/dt for the
sensor.  The latter is useful in indicating problems with the deployment.
It is calculated by first differencing pressure and then using a smoothing
spline on the result (to avoid grid-point wiggles that result because the
SBE software only writes 3 decimal places in pressure).  Note that dP/dt may
be off by a scale factor; this should not be a problem if there is a
<code>time</code> column in the <code>data</code> slot, or a <code>sample.rate</code> in the
<code>metadata</code> slot.
</p>
</li>
<li> <p><code>"sigma0"</code>, <code>"sigma1"</code>, <code>"sigma2"</code>, <code>"sigma3"</code>
and <code>"sigma4"</code> Profile of potential density referenced
to 0dbar (i.e. the surface), 1000dbar, 2000dbar, 3000dbar, and 4000dbar.
</p>
</li>
<li> <p><code>"spice"</code> Profile of spice.
</p>
</li>
<li> <p><code>"Rrho"</code> Profile of Rrho, defined in the diffusive sense.
</p>
</li>
<li> <p><code>"RrhoSF"</code> Profile of Rrho, defined in the salt-finger sense.
</p>
</li></ul>
</td></tr>
<tr><td><code id="plotProfile_+3A_ytype">ytype</code></td>
<td>
<p>variable to use on y axis. The valid choices are:
<code>"pressure"</code> (the default), <code>"z"</code>,
<code>"depth"</code>, <code>"sigmaTheta"</code> and <code>"sigma0"</code>.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_eos">eos</code></td>
<td>
<p>equation of state to be used, either <code>"unesco"</code> or
<code>"gsw"</code>.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_lty">lty</code></td>
<td>
<p>line type for the profile.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_xlab">xlab</code></td>
<td>
<p>optional label for x axis (at top of plot). If not
provided, a label is inferred from the value of <code>xtype</code>. For
the user-supplied case, bear in mind that the easy way to get
units is to use an expression, e.g.
<code>xlab=expression("Acceleration ["*m/s^2*"]")</code>.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_ylab">ylab</code></td>
<td>
<p>optional label for y axis.  See <code>xlab</code> for a note on
units.  Setting <code>ylab=""</code> prevents labelling the axis.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_col">col</code></td>
<td>
<p>color for a general profile.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_col.salinity">col.salinity</code></td>
<td>
<p>color for salinity profile (see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_col.temperature">col.temperature</code></td>
<td>
<p>color for temperature (see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_col.rho">col.rho</code></td>
<td>
<p>color for density (see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_col.n2">col.N2</code></td>
<td>
<p>color for square of buoyancy frequency (see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_col.dpdt">col.dpdt</code></td>
<td>
<p>color for dP/dt.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_col.time">col.time</code></td>
<td>
<p>color for delta-time.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_pt.bg">pt.bg</code></td>
<td>
<p>inside color for symbols with <code>pch</code> in 21:25</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_grid">grid</code></td>
<td>
<p>logical, set to <code>TRUE</code> to get a grid.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_col.grid">col.grid</code></td>
<td>
<p>color for grid.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_lty.grid">lty.grid</code></td>
<td>
<p>line type for grid.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_slim">Slim</code></td>
<td>
<p>optional limit for the salinity axis, which can either
represent Practical Salinity or Absolute Salinity.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_clim">Clim</code></td>
<td>
<p>optional limit for the conductivity axis.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_tlim">Tlim</code></td>
<td>
<p>optional limit for the temperature axis, which
can represent in-situ temperature, potential temperature, or
Conservative Temperature.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_densitylim">densitylim</code></td>
<td>
<p>optional limit for density axis.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_sigmalim">sigmalim</code></td>
<td>
<p>optional limit for the density-anomaly
axis, which can represent <code>sigmaTheta</code>, <code>sigma0</code>, <code>sigma1</code>,
<code>sigma2</code>, <code>sigma3</code> or <code>sigma4</code>.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_n2lim">N2lim</code></td>
<td>
<p>optional limit for the N2 axis.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_rrholim">Rrholim</code></td>
<td>
<p>optional limit for the density ratio axis.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_dpdtlim">dpdtlim</code></td>
<td>
<p>optional limit for the dp/dt axis.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_timelim">timelim</code></td>
<td>
<p>optional limit for the delta-time axis.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_plim">plim</code></td>
<td>
<p>optional limit for the pressure axis, ignored unless
<code>ytype=="pressure"</code>, in which case it takes precedence over
<code>ylim</code>.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_xlim">xlim</code></td>
<td>
<p>optional limit for x axis, which can apply to any plot type.
This is ignored if the plotted x variable is something for which a limit
may be specified with an argument, e.g. <code>xlim</code> is ignored for a
salinity profile, because <code>Slim</code> ought to be given in such a case.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_ylim">ylim</code></td>
<td>
<p>optional limit for y axis, which can apply to any plot type,
although is overridden by <code>plim</code> if <code>ytype</code> is <code>"pressure"</code>
or by <code>densitylim</code> if <code>ytype</code> is <code>"sigmaTheta"</code>.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_lwd">lwd</code></td>
<td>
<p>line width value for data line</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_xaxs">xaxs</code></td>
<td>
<p>value of <code><a href="graphics.html#topic+par">par()</a></code> <code>xaxs</code> to use</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_yaxs">yaxs</code></td>
<td>
<p>value of <code><a href="graphics.html#topic+par">par()</a></code> <code>yaxs</code> to use</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_cex">cex</code></td>
<td>
<p>size to be used for plot symbols (see <code><a href="graphics.html#topic+par">par()</a></code>)</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_pch">pch</code></td>
<td>
<p>code for plotting symbol (see <code><a href="graphics.html#topic+par">par()</a></code>).</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_usesmoothscatter">useSmoothScatter</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to use
<code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code> instead of <code><a href="graphics.html#topic+plot">plot()</a></code> to draw the plot.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_df">df</code></td>
<td>
<p>optional argument, passed to <code><a href="#topic+swN2">swN2()</a></code> if provided, and
if a plot using <code class="reqn">N^2</code> is requested.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_keepna">keepNA</code></td>
<td>
<p>FALSE</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_type">type</code></td>
<td>
<p>type of plot to draw, using the same scheme as
<code><a href="graphics.html#topic+plot">plot()</a></code>.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <a href="graphics.html#topic+par">par</a><code>(mgp)</code>, and also
for <a href="graphics.html#topic+par">par</a><code>(mar)</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_mar">mar</code></td>
<td>
<p>Four-element numerical value to be used to set the plot
margins, with a call to <a href="graphics.html#topic+par">par</a><code>(mar)</code> prior to the plot.
If this is not supplied, a reasonable default will be set up.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_add">add</code></td>
<td>
<p>A logical value that controls whether to add to an existing plot.  (It
makes sense to use <code>add=TRUE</code> in the <code>panel</code> argument of a
<code><a href="graphics.html#topic+coplot">coplot()</a></code>, for example.)</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_inset">inset</code></td>
<td>
<p>A logical value indicating whether to draw an inset plot.
Setting this to <code>TRUE</code> will prevent the present function from adjusting
the margins, which is
necessary because margin adjustment is the basis for the method used by
<code><a href="#topic+plotInset">plotInset()</a></code>.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="plotProfile_+3A_...">...</code></td>
<td>
<p>optional arguments passed to other functions.  A common example
is to set <code>df</code>, for use in <code><a href="#topic+swN2">swN2()</a></code> calculations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.ctd">read.ctd()</a></code> scans ctd information from a file,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method()</a></code> is a general plotting function for <code>ctd</code>
objects, and <code><a href="#topic+plotTS">plotTS()</a></code> plots a temperature-salinity diagrams.
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
plotProfile(ctd, xtype = "temperature")

</code></pre>

<hr>
<h2 id='plotScan'>Plot a ctd Object  in a Low-Level Fashion</h2><span id='topic+plotScan'></span>

<h3>Description</h3>

<p>Plot CTD data as time-series against scan number, to help with trimming
extraneous data from a CTD cast.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotScan(
  x,
  which = 1,
  xtype = "scan",
  flipy = FALSE,
  type = "l",
  mgp = getOption("oceMgp"),
  xlim = NULL,
  ylim = NULL,
  mar = c(mgp[1] + 1.5, mgp[1] + 1.5, mgp[1], mgp[1]),
  ...,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotScan_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="plotScan_+3A_which">which</code></td>
<td>
<p>integer specifying the plot to be drawn: 1
for pressure vs 'x', 2 for <code>diff(pressure)</code> vs 'x', 3 for temperature vs
'x', and 4 for salinity vs 'x'  Here, the value of 'x' is determined by
<code>xtype</code>.</p>
</td></tr>
<tr><td><code id="plotScan_+3A_xtype">xtype</code></td>
<td>
<p>Character string indicating variable for the x axis. The
permitted values are <code>"scan"</code> (the default), <code>"time"</code> and <code>"index"</code>.
The last of these is created by using <code><a href="base.html#topic+seq_along">seq_along()</a></code> on the pressure
column (which is assumed to be present in any <a href="#topic+ctd-class">ctd</a> object).
Only <code>xtype="index"</code> is guaranteed to work for all objects, and indeed
that value is used, if either <code>"scan"</code> or <code>"time"</code> is requested, but
unavailable.</p>
</td></tr>
<tr><td><code id="plotScan_+3A_flipy">flipy</code></td>
<td>
<p>Logical value, ignored unless <code>which</code> is 1. If <code>flipy</code>
is <code>TRUE</code>, then a pressure plot will have high pressures at the bottom
of the axis.</p>
</td></tr>
<tr><td><code id="plotScan_+3A_type">type</code></td>
<td>
<p>Character indicating the line type, as for <code><a href="graphics.html#topic+plot.default">plot.default()</a></code>. The default
is <code>"l"</code>, meaning to connect data with line segments. Another good choice is
<code>"o"</code>, to add points at the data.</p>
</td></tr>
<tr><td><code id="plotScan_+3A_mgp">mgp</code></td>
<td>
<p>Three-element numerical vector to use for <a href="graphics.html#topic+par">par</a><code>(mgp)</code>, and also
for <a href="graphics.html#topic+par">par</a><code>(mar)</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plotScan_+3A_xlim">xlim</code></td>
<td>
<p>Limits on the x value. The default, <code>NULL</code>, is to select this
from the data.</p>
</td></tr>
<tr><td><code id="plotScan_+3A_ylim">ylim</code></td>
<td>
<p>Limits on the y value. The default, <code>NULL</code>, is to select this
from the data.</p>
</td></tr>
<tr><td><code id="plotScan_+3A_mar">mar</code></td>
<td>
<p>Four-element vector be used with <a href="graphics.html#topic+par">par</a><code>("mar")</code>.  If set
to <code>NULL</code>, then <a href="graphics.html#topic+par">par</a><code>("mar")</code> is used.  A good choice for a TS diagram
with a palette to the right is <code style="white-space: pre;">&#8288;mar=par("mar")+c(0, 0, 0, 1))&#8288;</code>.</p>
</td></tr>
<tr><td><code id="plotScan_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to plotting functions.</p>
</td></tr>
<tr><td><code id="plotScan_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Historical Note</h3>

<p>On 2022-12-07, <code>xtype</code> was expanded to include <code>"index"</code>, and
an undocumented multi-panel feature was removed.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctdRaw)
plotScan(ctdRaw)
abline(v = c(130, 350), col = "red") # useful for ctdTrim()

</code></pre>

<hr>
<h2 id='plotSticks'>Draw a Stick Plot</h2><span id='topic+plotSticks'></span>

<h3>Description</h3>

<p>The arrows are drawn with directions on the graph that match the directions
indicated by the <code>u</code> and <code>v</code> components. The arrow size is set
relative to the units of the <code>y</code> axis, according to the value of
<code>yscale</code>, which has the unit of <code>v</code> divided by the unit of
<code>y</code>.
The interpretation of diagrams produced by <code>plotSticks</code> can be
difficult, owing to overlap in the arrows.  For this reason, it is often
a good idea to smooth <code>u</code> and <code>v</code> before using this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSticks(
  x,
  y,
  u,
  v,
  yscale = 1,
  add = FALSE,
  length = 1/20,
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1, mgp[1] + 1, 1, 1 + par("cex")),
  xlab = "",
  ylab = "",
  col = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSticks_+3A_x">x</code></td>
<td>
<p>x coordinates of stick origins.</p>
</td></tr>
<tr><td><code id="plotSticks_+3A_y">y</code></td>
<td>
<p>y coordinates of stick origins.  If not supplied, 0 will be used;
if length is less than that of x, the first number is repeated and the rest
are ignored.</p>
</td></tr>
<tr><td><code id="plotSticks_+3A_u">u</code></td>
<td>
<p>x component of stick length.</p>
</td></tr>
<tr><td><code id="plotSticks_+3A_v">v</code></td>
<td>
<p>y component of stick length.</p>
</td></tr>
<tr><td><code id="plotSticks_+3A_yscale">yscale</code></td>
<td>
<p>scale from u and v to y (see &ldquo;Description&rdquo;).</p>
</td></tr>
<tr><td><code id="plotSticks_+3A_add">add</code></td>
<td>
<p>boolean, set <code>TRUE</code> to add to an existing plot.</p>
</td></tr>
<tr><td><code id="plotSticks_+3A_length">length</code></td>
<td>
<p>value to be provided to <code><a href="graphics.html#topic+arrows">arrows()</a></code>; here, we set a
default that is smaller than normally used, because these plots tend to be
crowded in oceanographic applications.</p>
</td></tr>
<tr><td><code id="plotSticks_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code><a href="graphics.html#topic+par">par</a></code><code>("mgp")</code>. Note
that the default <code>mar</code> is computed from the <code>mgp</code> value.
The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plotSticks_+3A_mar">mar</code></td>
<td>
<p>value to be used with <code><a href="graphics.html#topic+par">par</a></code><code>("mar")</code>.</p>
</td></tr>
<tr><td><code id="plotSticks_+3A_xlab">xlab</code>, <code id="plotSticks_+3A_ylab">ylab</code></td>
<td>
<p>labels for the plot axes. The default is not to label them.</p>
</td></tr>
<tr><td><code id="plotSticks_+3A_col">col</code></td>
<td>
<p>color of sticks, in either numerical or character format. This is
made to have length matching that of <code>x</code> by a call to <code><a href="base.html#topic+rep">rep()</a></code>,
which can be handy in e.g. colorizing a velocity field by day.</p>
</td></tr>
<tr><td><code id="plotSticks_+3A_...">...</code></td>
<td>
<p>graphical parameters passed down to <code><a href="graphics.html#topic+arrows">arrows()</a></code>.  It
is common, for example, to use smaller arrow heads than <code><a href="graphics.html#topic+arrows">arrows()</a></code>
uses; see &ldquo;Examples&rdquo;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)

# Flow from a point source
n &lt;- 16
x &lt;- rep(0, n)
y &lt;- rep(0, n)
theta &lt;- seq(0, 2 * pi, length.out = n)
u &lt;- sin(theta)
v &lt;- cos(theta)
plotSticks(x, y, u, v, xlim = c(-2, 2), ylim = c(-2, 2))
rm(n, x, y, theta, u, v)

# Oceanographic example
data(met)
t &lt;- met[["time"]]
u &lt;- met[["u"]]
v &lt;- met[["v"]]
p &lt;- met[["pressure"]]
oce.plot.ts(t, p)
plotSticks(t, 99, u, v, yscale = 25, add = TRUE)
</code></pre>

<hr>
<h2 id='plotTaylor'>Plot a Model-data Comparison Diagram</h2><span id='topic+plotTaylor'></span>

<h3>Description</h3>

<p>Creates a diagram as described by Taylor (2001).  The graph is in the form
of a semicircle, with radial lines and spokes connecting at a focus point on
the flat (lower) edge.  The radius of a point on the graph indicates the
standard deviation of the corresponding quantity, i.e. x and the columns in
y.  The angle connecting a point on the graph to the focus provides an
indication of correlation coefficient with respect to x.  The &ldquo;east&rdquo; side
of the graph indicates <code class="reqn">R=1</code>, while <code class="reqn">R=0</code> is at the
&quot;north&quot; edge and <code class="reqn">R=-1</code> is at the &quot;west&quot; side.  The <code>x</code>
data are indicated with a bullet on the graph, appearing on the lower edge
to the right of the focus at a distance indicating the standard deviation of
'x'.  The other points on the graph represent the columns of 'y',
coded automatically or with the supplied values of 'pch' and
'col'.
The example shows two tidal models of the Halifax sealevel data, computed
with <code><a href="#topic+tidem">tidem()</a></code> with just the M2 component and the S2 component;
the graph indicates that the M2 model is much better than the S2 model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTaylor(x, y, scale, pch, col, labels, pos, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTaylor_+3A_x">x</code></td>
<td>
<p>a vector of reference values of some quantity, e.g. measured over
time or space.</p>
</td></tr>
<tr><td><code id="plotTaylor_+3A_y">y</code></td>
<td>
<p>a matrix whose columns hold values of values to be compared with
those in x.  (If <code>y</code> is a vector, it is converted first to a one-column
matrix).</p>
</td></tr>
<tr><td><code id="plotTaylor_+3A_scale">scale</code></td>
<td>
<p>optional scale, interpreted as the maximum value of standard
deviation.</p>
</td></tr>
<tr><td><code id="plotTaylor_+3A_pch">pch</code></td>
<td>
<p>list of characters to plot, one for each column of <code>y</code>.</p>
</td></tr>
<tr><td><code id="plotTaylor_+3A_col">col</code></td>
<td>
<p>list of colors for points on the plot, one for each column of
<code>y</code>.</p>
</td></tr>
<tr><td><code id="plotTaylor_+3A_labels">labels</code></td>
<td>
<p>optional vector of strings to use for labelling the points.</p>
</td></tr>
<tr><td><code id="plotTaylor_+3A_pos">pos</code></td>
<td>
<p>optional vector of positions for labelling strings.  If not
provided, labels will be to the left of the symbols.</p>
</td></tr>
<tr><td><code id="plotTaylor_+3A_...">...</code></td>
<td>
<p>optional arguments passed by <code>plotTaylor</code> to more child
functions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Taylor, Karl E., 2001.  Summarizing multiple aspects of model
performance in a single diagram, <em>J. Geophys. Res.</em>, 106:D7, 7183&ndash;7192.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(sealevel)
x &lt;- sealevel[["elevation"]]
M2 &lt;- predict(tidem(sealevel, constituents = "M2"))
S2 &lt;- predict(tidem(sealevel, constituents = c("S2")))
plotTaylor(x, cbind(M2, S2))
</code></pre>

<hr>
<h2 id='plotTS'>Plot Temperature-Salinity Diagram</h2><span id='topic+plotTS'></span>

<h3>Description</h3>

<p>Creates a temperature-salinity plot for a CTD cast, with labeled isopycnals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotTS(
  x,
  inSitu = FALSE,
  type = "p",
  referencePressure = 0,
  nlevels = 6,
  levels,
  grid = TRUE,
  col.grid = "lightgray",
  lty.grid = "dotted",
  rho1000 = FALSE,
  eos = getOption("oceEOS", default = "gsw"),
  cex = par("cex"),
  col = par("col"),
  pch = par("pch"),
  bg = "white",
  pt.bg = "transparent",
  col.rho = gray(0.5),
  cex.rho = 3/4 * par("cex"),
  rotate = TRUE,
  useSmoothScatter = FALSE,
  xlab,
  ylab,
  Slim,
  Tlim,
  drawFreezing = TRUE,
  trimIsopycnals = TRUE,
  gridIsopycnals = c(30, 50),
  mgp = getOption("oceMgp"),
  mar = c(mgp[1] + 1.5, mgp[1] + 1.5, mgp[1], mgp[1]),
  lwd = par("lwd"),
  lty = par("lty"),
  lwd.rho = par("lwd"),
  lty.rho = par("lty"),
  add = FALSE,
  inset = FALSE,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotTS_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a>, <a href="#topic+argo-class">argo</a> or <a href="#topic+section-class">section</a> object, or a list
containing solely <a href="#topic+ctd-class">ctd</a> objects or <a href="#topic+argo-class">argo</a> objects.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_insitu">inSitu</code></td>
<td>
<p>A boolean indicating whether to use in-situ temperature or
(the default) potential temperature, calculated with reference pressure
given by <code>referencePressure</code>.  This is ignored if <code>eos="gsw"</code>,
because those cases the y axis is necessarily the conservative formulation
of temperature.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_type">type</code></td>
<td>
<p>representation of data, <code>"p"</code> for points, <code>"l"</code> for
connecting lines, <code>"b"</code> for spaced connecting lines, or <code>"n"</code>
for no indication.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_referencepressure">referencePressure</code></td>
<td>
<p>reference pressure, to be used in calculating
potential temperature, if <code>inSitu</code> is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_nlevels">nlevels</code></td>
<td>
<p>Number of automatically-selected isopycnal levels (ignored if
<code>levels</code> is supplied).</p>
</td></tr>
<tr><td><code id="plotTS_+3A_levels">levels</code></td>
<td>
<p>Optional vector of desired isopycnal levels.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_grid">grid</code></td>
<td>
<p>a flag that can be set to <code>TRUE</code> to get a grid.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_col.grid">col.grid</code></td>
<td>
<p>color for grid.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_lty.grid">lty.grid</code></td>
<td>
<p>line type for grid.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_rho1000">rho1000</code></td>
<td>
<p>if TRUE, label isopycnals as e.g. 1024; if FALSE, label as
e.g. 24</p>
</td></tr>
<tr><td><code id="plotTS_+3A_eos">eos</code></td>
<td>
<p>equation of state to be used, either <code>"unesco"</code> or
<code>"gsw"</code>.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_cex">cex</code></td>
<td>
<p>character-expansion factor for symbols, as in <a href="graphics.html#topic+par">par</a><code>("cex")</code>.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_col">col</code></td>
<td>
<p>color for symbols.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_pch">pch</code></td>
<td>
<p>symbol type, as in <a href="graphics.html#topic+par">par</a><code>("pch")</code>.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_bg">bg</code></td>
<td>
<p>optional color to be painted under plotting area, before
plotting.  (This is useful for cases in which <code>inset=TRUE</code>.)</p>
</td></tr>
<tr><td><code id="plotTS_+3A_pt.bg">pt.bg</code></td>
<td>
<p>inside color for symbols with <code>pch</code> in 21:25</p>
</td></tr>
<tr><td><code id="plotTS_+3A_col.rho">col.rho</code></td>
<td>
<p>color for isopycnal lines and their labels.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_cex.rho">cex.rho</code></td>
<td>
<p>size of the isopycnal labels.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_rotate">rotate</code></td>
<td>
<p>if TRUE, labels in right-hand margin are written vertically</p>
</td></tr>
<tr><td><code id="plotTS_+3A_usesmoothscatter">useSmoothScatter</code></td>
<td>
<p>if TRUE, use <code><a href="graphics.html#topic+smoothScatter">smoothScatter()</a></code> to plot the
points.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_xlab">xlab</code></td>
<td>
<p>optional label for the x axis, with default &quot;Salinity [PSU]&quot;.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_ylab">ylab</code></td>
<td>
<p>optional label for the y axis, with default &quot;Temperature [C]&quot;.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_slim">Slim</code></td>
<td>
<p>optional limits for salinity axis, otherwise inferred from visible data
(i.e. the data that have finite values for both salinity and temperature).</p>
</td></tr>
<tr><td><code id="plotTS_+3A_tlim">Tlim</code></td>
<td>
<p>as <code>Slim</code>, but for temperature.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_drawfreezing">drawFreezing</code></td>
<td>
<p>logical indication of whether to draw a freezing-point
line. This is based on zero pressure. If <code>eos="unesco"</code> then
<code><a href="#topic+swTFreeze">swTFreeze()</a></code> is used to compute the curve, whereas if
<code>eos="gsw"</code> then <code><a href="gsw.html#topic+gsw_CT_freezing">gsw::gsw_CT_freezing()</a></code> is used;
in each case, zero pressure is used.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_trimisopycnals">trimIsopycnals</code></td>
<td>
<p>logical value (<code>TRUE</code> by default) that
indicates whether to trim isopycnal curves
to the region of temperature-salinity space for which density
computations are considered to be valid in the context of the
chosen <code>eos</code>; see &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_gridisopycnals">gridIsopycnals</code></td>
<td>
<p>a parameter that controls how the isopycnals
are computed.  This may be NULL, or an integer vector of length 2.
<em>Case 1:</em> the isopycnals are drawn by tracing density
isopleths in salinity-temperature space. This method was
used as the default prior to version 1.7-11, but it was
found to yield staircase-like isopycnal curves for highly
zoomed-in plots (e.g. with millidegree temperature ranges).
<em>Case 2:</em> a grid of density is constructed, with <code>gridIsopycnals[1]</code>
salinity levels and <code>gridIsopycnals[2]</code> temperature levels, and
then <code><a href="grDevices.html#topic+contourLines">contourLines()</a></code> is used to trace the isopycnals.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_mgp">mgp</code></td>
<td>
<p>3-element numerical vector to use for <code style="white-space: pre;">&#8288;[par](mgp)&#8288;</code>, and also
for <a href="graphics.html#topic+par">par</a><code>(mar)</code>, computed from this.  The default is tighter than the R
default, in order to use more space for the data and less for the axes.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_mar">mar</code></td>
<td>
<p>value to be used with <a href="graphics.html#topic+par">par</a><code>("mar")</code>.  If set to
<code>NULL</code>, then <a href="graphics.html#topic+par">par</a><code>("mar")</code> is used.  A good choice for a TS diagram
with a palette to the right is <code style="white-space: pre;">&#8288;mar=par("mar")+c(0, 0, 0, 1))&#8288;</code>.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_lwd">lwd</code></td>
<td>
<p>line width of lines or symbols.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_lty">lty</code></td>
<td>
<p>line type of lines or symbols.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_lwd.rho">lwd.rho</code></td>
<td>
<p>line width for density curves.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_lty.rho">lty.rho</code></td>
<td>
<p>line type for density curves.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_add">add</code></td>
<td>
<p>a flag that controls whether to add to an existing plot.  (It
makes sense to use <code>add=TRUE</code> in the <code>panel</code> argument of a
<code><a href="graphics.html#topic+coplot">coplot()</a></code>, for example.)</p>
</td></tr>
<tr><td><code id="plotTS_+3A_inset">inset</code></td>
<td>
<p>set to <code>TRUE</code> for use within <code><a href="#topic+plotInset">plotInset()</a></code>.  The
effect is to prevent the present function from adjusting margins, which is
necessary because margin adjustment is the basis for the method used by
<code><a href="#topic+plotInset">plotInset()</a></code>.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="plotTS_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The isopycnal curves (along which density is constant) are
drawn with <code><a href="#topic+drawIsopycnals">drawIsopycnals()</a></code>, which also places
labels in the margins showing density minus 1000 <code class="reqn">kg/m^3</code>.
If <code>trimIsopycnals</code> is <code>TRUE</code> (which is the default), these curves
are trimmed to the region within which the results of density calculation
in the chosen equation of state (<code>eos</code>) are considered to be reliable.
</p>
<p>With <code>eos="unesco"</code> this region includes
Practical Salinity from 0 to 42 and Potential Temperature
from -2C to 40C, in accordance with Fofonoff and Millard
(1983, page 23).
</p>
<p>With <code>eos="gsw"</code> the lower
limit of Absolute Salinity validity is taken as 0 g/kg,
in accordance with both McDougall et al. (2003 section 3)
and the TEOS-10/gsw Matlab code for the so-called &quot;funnel&quot; of validity.
However, an appropriate upper limit on Absolute Salinity is not as clear.
Here, the value 42 g/kg is chosen to match the &quot;funnel&quot; Matlab code
as of July 2020, but two other choices might have been
made. One is 50 g/kg, since <code><a href="gsw.html#topic+gsw_SA_from_rho">gsw::gsw_SA_from_rho()</a></code> returns <code>NA</code> values
for Absolute Salinities exceeding that value, and another is
40 g/kg, as in McDougall et al. (2003 section 3).
The Conservative Temperature range is set to run from -2C
to 33C, as in McDougall et al. (2003 section 3), even though the
&quot;funnel&quot; imposes no upper limit on this variable.
</p>


<h3>Value</h3>

<p>A list is silently returned, containing <code>xat</code> and <code>yat</code>,
values that can be used by <code><a href="#topic+oce.grid">oce.grid()</a></code> to add a grid to the plot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ul>
<li><p> Fofonoff, N. P., and R. C. Millard.
&quot;Algorithms for Computation of Fundamental Properties of Seawater.&quot;
UNESCO Technical Papers in Marine Research. SCOR working group on Evaluation of CTD data;
UNESCO/ICES/SCOR/IAPSO Joint Panel on Oceanographic Tables and Standards, 1983.
<code style="white-space: pre;">&#8288;https://unesdoc.unesco.org/ark:/48223/pf0000059832&#8288;</code>.
</p>
</li>
<li><p> McDougall, Trevor J., David R. Jackett, Daniel G. Wright, and Rainer Feistel.
&quot;Accurate and Computationally Efficient Algorithms for Potential Temperature and Density of Seawater.&quot;
Journal of Atmospheric and Oceanic Technology 20, no. 5 (May 1, 2003): 730-41.
<code>https://journals.ametsoc.org/jtech/article/20/5/730/2543/Accurate-and-Computationally-Efficient-Algorithms</code>.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+summary+2Cctd-method">summary,ctd-method()</a></code> summarizes the information, while
<code><a href="#topic+read.ctd">read.ctd()</a></code> scans it from a file.
</p>
<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. ctd object
library(oce)
data(ctd)
plotTS(ctd)

# 2. section object (note the outlier!)
data(section)
plotTS(section)

# 3. argo object
data(argo)
plotTS(handleFlags(argo))

# 4. oxygen-based colormap
marOrig &lt;- par("mar") # so later plots with palettes have same margins
cm &lt;- colormap(section[["oxygen"]])
drawPalette(colormap = cm, zlab = "Oxygen")
plotTS(section, pch = 19, col = cm$zcol, mar = par("mar")) # the mar adjusts for the palette

# 5. waters near Gulf Stream, colour-coded for longitude.
sec &lt;- subset(section, abs(longitude + 71.6) &lt; 1)
cm &lt;- colormap(sec[["longitude", "byStation"]], col = oceColors9B)
par(mar = c(3.3, 3.3, 1, 1.5))
drawPalette(colormap = cm, zlab = "Longitude")
plotTS(sec, type = "n", xaxs = "r", mar = par("mar"))
jnk &lt;- mapply(
    function(s, col) {
        plotTS(s, type = "o", col = "gray", pt.bg = col, pch = 21, add = TRUE)
    },
    sec[["station"]],
    col = cm$zcol
)

# 6. with added spiciness contours
data(ctd)
plotTS(ctd, eos = "gsw") # MANDATORY so x=SA and y=CT
usr &lt;- par("usr")
n &lt;- 100
SAgrid &lt;- seq(usr[1], usr[2], length.out = n)
CTgrid &lt;- seq(usr[3], usr[4], length.out = n)
g &lt;- expand.grid(SA = SAgrid, CT = CTgrid)
spiciness &lt;- matrix(gsw::gsw_spiciness0(g$SA, g$CT), nrow = n)
contour(SAgrid, CTgrid, spiciness, col = 2, labcex = 1, add = TRUE)

</code></pre>

<hr>
<h2 id='predict.tidem'>Predict a Tidal Signal</h2><span id='topic+predict.tidem'></span>

<h3>Description</h3>

<p>This creates a time-series of predicted tides, based on a
tidal model object that was created by <code><a href="#topic+as.tidem">as.tidem()</a></code> or <code><a href="#topic+tidem">tidem()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'tidem'
predict(object, newdata, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict.tidem_+3A_object">object</code></td>
<td>
<p>a <a href="#topic+tidem-class">tidem</a> object.</p>
</td></tr>
<tr><td><code id="predict.tidem_+3A_newdata">newdata</code></td>
<td>
<p>vector of POSIXt times at which to make the
prediction.  For models created with <code><a href="#topic+tidem">tidem()</a></code>,
the <code>newdata</code> argument is optional, and if it is not provided, then
the predictions are at the observation times given to
<code><a href="#topic+tidem">tidem()</a></code>. However, <code>newdata</code> is required  if <code><a href="#topic+as.tidem">as.tidem()</a></code>
had been used to create <code>object</code>.</p>
</td></tr>
<tr><td><code id="predict.tidem_+3A_...">...</code></td>
<td>
<p>optional arguments passed on to children.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All the tidal constituents that are stored in <code>object</code> are used,
not just those that are statistically significant or that have amplitude
exceeding any particular value.  In this respect, <code><a href="#topic+predict.tidem">predict.tidem()</a></code>
follows a pattern established by e.g. <code><a href="stats.html#topic+predict.lm">predict.lm()</a></code>.  Note that
the constituents in <code>object</code> are straightforward if it
was constructed with <code><a href="#topic+as.tidem">as.tidem()</a></code>, but considerably more complicated
for <code><a href="#topic+tidem">tidem()</a></code>, and so the documentation for the latter ought to
be studied closely, especially with regard to the Rayleigh criterion.
</p>


<h3>Value</h3>

<p>A vector of predictions.
</p>


<h3>Sample of Usage</h3>

<pre>
# prediction at specified times
data(sealevel)
m &lt;- tidem(sealevel)
# Check fit over 2 days (interpolating to finer timescale)
look &lt;- 1:48
time &lt;- sealevel[["time"]]
elevation &lt;- sealevel[["elevation"]]
oce.plot.ts(time[look], elevation[look])
# 360s = 10 minute timescale
t &lt;- seq(from=time[1], to=time[max(look)], by=360)
lines(t, predict(m, newdata=t), col="red")
legend("topright", col=c("black","red"),
legend=c("data","model"),lwd=1)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Show non-tidal sealevel signal in Halifax Harbour during
# the year 2002. The spike resulted from Hurricane Juan.
library(oce)
data(sealevel)
time &lt;- sealevel[["time"]]
elevation &lt;- sealevel[["elevation"]]
prediction &lt;- tidem(sealevel) |&gt; predict()
oce.plot.ts(time, elevation - prediction)

</code></pre>

<hr>
<h2 id='preferAdjusted'>Set Preference for Adjusted Values</h2><span id='topic+preferAdjusted'></span>

<h3>Description</h3>

<p><a href="#topic+argo-class">argo</a> data can contain &quot;adjusted&quot; forms of data items,
which may be more trustworthy than the original
data, and <code>preferAdjusted</code> lets the user express a preference
for such adjusted data.  This means that using
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code> on the results returned by <code>preferAdjusted</code>
will (if possible) return adjusted data, and also use those adjusted
data in computations of derived quantities such as Absolute Salinity.
The preference applies also to units and to data-quality flags,
both of which can be returned by <code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>, as
discussed in &ldquo;Details&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preferAdjusted(argo, which = "all", fallback = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preferAdjusted_+3A_argo">argo</code></td>
<td>
<p>An <a href="#topic+argo-class">argo</a> object.</p>
</td></tr>
<tr><td><code id="preferAdjusted_+3A_which">which</code></td>
<td>
<p>A character vector naming the items for which
(depending also on the value of <code>fallback</code>) adjusted values
are to be sought by future calls to <code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>.
The short names are used, e.g. <code>which="oxygen"</code> means that
adjusted oxygen is to be returned in future calls
such as <code>argo[["oxygen"]]</code>.  The default,
<code>"all"</code>, means to  use adjusted values for any item in <code>argo</code>
that has adjusted values.</p>
</td></tr>
<tr><td><code id="preferAdjusted_+3A_fallback">fallback</code></td>
<td>
<p>A logical value indicating whether to fall back
to unadjusted values for any data field in which the
adjusted values are all <code>NA</code>.  The default value, <code>TRUE</code>,
avoids a problem with biogeochemical fields, where adjustment
of any one field may lead to insertion of &quot;adjusted&quot; values for
other fields that consist of nothing more than <code>NA</code>s.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>preferAdjusted()</code> merely sets two items in the <code>metadata</code> slot of the
returned <a href="#topic+argo-class">argo</a> object. The real action is carried out by
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code> but, for convenience, the details are explained here.
</p>
<p>Consider salinity, for example.
If <code>which</code> equals <code>"all"</code>, or if it is a character
vector containing <code>"salinity"</code>, then using
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code> on the returned object
will yield the adjusted forms of the salinity data,
its associated flags, or its units.  Thus, in the salinity
case,
</p>

<ul>
<li> <p><code>argo[["salinity"]]</code> will attempt to return <code>argo@data$salinityAdjusted</code>
instead of returning <code>argo@data$salinity</code>, although if the adjusted values
are all <code>NA</code> then, depending on the value of <code>fallback</code>, the
unadjusted values may be returned; similarly
</p>
</li>
<li> <p><code>argo[["salinityFlags"]]</code> will attempt to return
<code>argo@metadata$flags$salinityAdjusted</code>
instead of <code>argo@metadata$flags$salinity</code>, and
</p>
</li>
<li> <p><code>argo[["salinityUnits"]]</code> will attempt to return
<code>argo@metadata$units$salinityAdjusted</code>
instead of <code>argo@metadata$units$salinity</code>.
</p>
</li></ul>

<p>The default value, <code>which="all"</code>, indicates that this
preference for adjusted values will apply to all the
elements of the <code>data</code> slot of the returned object, along
with associated flags and units. This can be handy for quick
work, but analysts may also choose to restrict their use of
adjusted values to a subset of variables, based on their own
decisions about data quality or accuracy.
</p>
<p>The default value <code>fallback=TRUE</code> indicates that later calls to
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code> should return unadjusted values for any
data items that have <code>NA</code> for all the adjusted values.  This
condition is rare for core variables (salinity, temperature and
pressure) but is annoyingly common for biogeochemical variables; see
e.g. Section 2.2.5 of Reference 1 for a discussion of
the conditions under which Argo netcdf files contain
adjusted values. Setting <code>fallback=FALSE</code> means that adjusted
values (if they exist) will always be returned, even if they
are a useless collection of <code>NA</code> values.
</p>
<p>Error fields, such as <code>salinityAdjustedError</code>, are returned
as-is by <code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>, regardless of whether
the object was created by <code>preferAdjusted</code>.
</p>
<p>It should be noted that, regardless of whether <code>preferAdjusted</code>
has been used, the analyst can always access either unadjusted
or adjusted data directly, using the original variable names stored
in the source netcdf file.  For example, <code>argo[["PSAL"]]</code>
yields unadjusted salinity values, and
<code>argo[["PSAL_ADJUSTED"]]</code> yields adjusted values (if they exist, or
<code>NULL</code> if they do not).
Similarly, adjusted value can always be obtained by using a form
like <code>argo[["salinityAdjusted"]]</code>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+argo-class">argo</a> object its <code>metadata</code> slot altered
(in its <code>adjustedWhich</code> and <code>adjustedFallback</code> elements)
as a signal for how <code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code> should
function on the object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, based on discussions with Jaimie Harbin (with
respect to the <code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code> interface) and Clark Richards
(with respect to storing the preference in the <code>metadata</code> slot).
</p>


<h3>References</h3>


<ol>
<li><p> Argo Data Management Team. &quot;Argo User's Manual V3.3.&quot; Ifremer,
November 28, 2019.
<a href="https://doi.org/10.13155/29825">doi:10.13155/29825</a>
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(argo)
argoAdjusted &lt;- preferAdjusted(argo)
all.equal(argo[["salinityAdjusted"]], argoAdjusted[["salinity"]])
all.equal(argo[["salinityFlagsAdjusted"]], argoAdjusted[["salinityFlags"]])
all.equal(argo[["salinityUnitsAdjusted"]], argoAdjusted[["salinityUnits"]])

</code></pre>

<hr>
<h2 id='presentTime'>Get the Present Time, in a Stated Timezone</h2><span id='topic+presentTime'></span>

<h3>Description</h3>

<p>Get the Present Time, in a Stated Timezone
</p>


<h3>Usage</h3>

<pre><code class='language-R'>presentTime(tz = "UTC")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="presentTime_+3A_tz">tz</code></td>
<td>
<p>String indicating the desired timezone. The default is
to use UTC, which is used very commonly in oceanographic work. To
get the local time, use <code>tz=""</code> or <code>tz=NULL</code>,</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="base.html#topic+POSIXct">POSIXct()</a></code>-style object holding the present time, in the
indicated timezone.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>presentTime() # UTC
presentTime("") # the local timezone

</code></pre>

<hr>
<h2 id='prettyPosition'>Pretty Longitude/Latitude in Degree-Minute-Second Format</h2><span id='topic+prettyPosition'></span>

<h3>Description</h3>

<p>Round a geographical positions in degrees, minutes, and seconds
Depending on the range of values in <code>x</code>, rounding is done to degrees,
half-degrees, minutes, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prettyPosition(x, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prettyPosition_+3A_x">x</code></td>
<td>
<p>a series of one or more values of a latitude or longitude, in
decimal degrees</p>
</td></tr>
<tr><td><code id="prettyPosition_+3A_debug">debug</code></td>
<td>
<p>set to a positive value to get debugging information during
processing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of numbers that will yield good axis labels if
<code><a href="#topic+formatPosition">formatPosition()</a></code> is used.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
formatPosition(prettyPosition(10 + 1:10 / 60 + 2.8 / 3600))
</code></pre>

<hr>
<h2 id='processingLog+26lt+3B-'>Add an Item to a Processing Log</h2><span id='topic+processingLog+3C-'></span>

<h3>Description</h3>

<p>Add an Item to a Processing Log
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processingLog(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processingLog+2B26lt+2B3B-_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="processingLog+2B26lt+2B3B-_+3A_value">value</code></td>
<td>
<p>A character string with the description of the logged activity.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other things related to processing logs: 
<code><a href="#topic+processingLogAppend">processingLogAppend</a>()</code>,
<code><a href="#topic+processingLogItem">processingLogItem</a>()</code>,
<code><a href="#topic+processingLogShow">processingLogShow</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ctd)
processingLogShow(ctd)
processingLog(ctd) &lt;- "test"
processingLogShow(ctd)

</code></pre>

<hr>
<h2 id='processingLogAppend'>Append an Item to a Processing Log</h2><span id='topic+processingLogAppend'></span>

<h3>Description</h3>

<p>Append an Item to a Processing Log
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processingLogAppend(h, value = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processingLogAppend_+3A_h">h</code></td>
<td>
<p>either the <code>processingLog</code> slot of an object, or
an <code>oce</code> object from which the processingLog will be extracted</p>
</td></tr>
<tr><td><code id="processingLogAppend_+3A_value">value</code></td>
<td>
<p>A string indicating the text of the log entry.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code><a href="base.html#topic+list">list()</a></code> containing items named
<code>time</code> and <code>value</code>, i.e. the times of entries
and the text notations of those entries..
</p>


<h3>See Also</h3>

<p>Other things related to processing logs: 
<code><a href="#topic+processingLog+3C-">processingLog&lt;-</a>()</code>,
<code><a href="#topic+processingLogItem">processingLogItem</a>()</code>,
<code><a href="#topic+processingLogShow">processingLogShow</a>()</code>
</p>

<hr>
<h2 id='processingLogItem'>Create an Item That can be Inserted into a Processing Log</h2><span id='topic+processingLogItem'></span>

<h3>Description</h3>

<p>A function is used internally to initialize processing logs.
Users will probably prefer to use <code><a href="#topic+processingLogAppend">processingLogAppend()</a></code>
instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processingLogItem(value = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processingLogItem_+3A_value">value</code></td>
<td>
<p>A string that will be used for the item.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="base.html#topic+list">list()</a></code> containing <code>time</code>, which is
the time in UTC (calculated with <code><a href="#topic+presentTime">presentTime()</a></code>)
at the moment the function is called and
<code>value</code>, a string that is set to the argument of the same name.
</p>


<h3>See Also</h3>

<p>Other things related to processing logs: 
<code><a href="#topic+processingLog+3C-">processingLog&lt;-</a>()</code>,
<code><a href="#topic+processingLogAppend">processingLogAppend</a>()</code>,
<code><a href="#topic+processingLogShow">processingLogShow</a>()</code>
</p>

<hr>
<h2 id='processingLogShow'>Show the Processing Log of an oce Object</h2><span id='topic+processingLogShow'></span>

<h3>Description</h3>

<p>Show the Processing Log of an oce Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processingLogShow(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processingLogShow_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other things related to processing logs: 
<code><a href="#topic+processingLog+3C-">processingLog&lt;-</a>()</code>,
<code><a href="#topic+processingLogAppend">processingLogAppend</a>()</code>,
<code><a href="#topic+processingLogItem">processingLogItem</a>()</code>
</p>

<hr>
<h2 id='pwelch'>Welch Periodogram</h2><span id='topic+pwelch'></span>

<h3>Description</h3>

<p>Compute periodogram using the Welch (1967) method. This is
somewhat analogous to the Matlab function of the same name,
but it is <em>not</em> intended as a drop-in replacement.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pwelch(
  x,
  window,
  noverlap,
  nfft,
  fs,
  spec,
  demean = FALSE,
  detrend = TRUE,
  plot = TRUE,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pwelch_+3A_x">x</code></td>
<td>
<p>a vector or timeseries to be analyzed.  If <code>x</code> is a timeseries, then
it there is no need to <code>fs</code>, and doing so will result in an error if it does
not match the value inferred from <code>x</code>.</p>
</td></tr>
<tr><td><code id="pwelch_+3A_window">window</code></td>
<td>
<p>optional numeric vector specifying a window to be applied
to the timeseries subsamples.  This is ignored if <code>spec</code> is provided.
Otherwise, if <code>window</code> is provided, then it must either
be of the same length as <code>nfft</code> or be of length 1.  In the first case,
the vector is multiplied into the timeseries subsample, and the length
of <code>window</code> must equal <code>nfft</code> is that is supplied.
In the second then <code>window</code> is taken to be the number
of sub-intervals into which the time series is to be broken up, with a
hamming window being used for each sub-interval.  If <code>window</code> is not
specified and <code>nfft</code> is given, then the window is constructed as
a hamming window with length <code>nfft</code>.  And, if neither <code>window</code>
nor <code>nfft</code> are specified, then <code>x</code> will be broken up
into 8 portions.</p>
</td></tr>
<tr><td><code id="pwelch_+3A_noverlap">noverlap</code></td>
<td>
<p>number of points to overlap between windows.  If not
specified, this will be set to half the window length.</p>
</td></tr>
<tr><td><code id="pwelch_+3A_nfft">nfft</code></td>
<td>
<p>length of FFT. See <code>window</code> for how <code>nfft</code> interacts with
that argument.</p>
</td></tr>
<tr><td><code id="pwelch_+3A_fs">fs</code></td>
<td>
<p>frequency of time-series.  If <code>x</code> is a time-series, and if
<code>fs</code> is supplied, then time-series is altered to have frequency
<code>fs</code>.</p>
</td></tr>
<tr><td><code id="pwelch_+3A_spec">spec</code></td>
<td>
<p>optional function to be used for the computation of the spectrum,
to allow finer-grained control of the processing.
If provided, <code>spec</code> must accept a time-series as its first argument, and
must return a list containing the spectrum in <code>spec</code> and the
frequency in <code>freq</code>.
Note that no window will be applied to the data after subsampling,
and an error will be reported if <code>window</code> and <code>spec</code> are both given.
An error will be reported if <code>spec</code> is given but <code>nfft</code> is not given.
Note that the values of <code>demean</code>, <code>detrend</code> and <code>plot</code> are ignored if <code>spec</code>
is given. However, the ... argument <em>is</em> passed to <code>spec</code>.</p>
</td></tr>
<tr><td><code id="pwelch_+3A_demean">demean</code>, <code id="pwelch_+3A_detrend">detrend</code></td>
<td>
<p>logical values that can control the spectrum calculation,
in the default case of <code>spec</code>.  These are passed to <code><a href="stats.html#topic+spectrum">spectrum()</a></code> and thence
<code><a href="stats.html#topic+spec.pgram">spec.pgram()</a></code>; see the help pages for the latter for an explanation.</p>
</td></tr>
<tr><td><code id="pwelch_+3A_plot">plot</code></td>
<td>
<p>logical, set to <code>TRUE</code> to plot the spectrum.</p>
</td></tr>
<tr><td><code id="pwelch_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="pwelch_+3A_...">...</code></td>
<td>
<p>optional extra arguments to be passed to
<code><a href="stats.html#topic+spectrum">spectrum()</a></code>, or to <code>spec</code>, if the latter is given.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>First, <code>x</code> is broken up into chunks,
overlapping as specified by <code>noverlap</code>.  These chunks are then
multiplied by the window, and then
passed to <code><a href="stats.html#topic+spectrum">spectrum()</a></code>.  The resulting spectra are then averaged,
with the results being stored in <code>spec</code> of the return value.  Other
entries of the return value mimic those returned by <code><a href="stats.html#topic+spectrum">spectrum()</a></code>.
</p>
<p>It should be noted that the actions of several parameters are interlocked,
so this can be a complex function to use.  For example, if <code>window</code> is
given and has length exceeding 1, then its length must equal <code>nfft</code>, if the
latter is also provided.
</p>


<h3>Value</h3>

<p><code>pwelch</code> returns a list mimicking the return value from <code><a href="stats.html#topic+spectrum">spectrum()</a></code>,
containing frequency <code>freq</code>, spectral power <code>spec</code>, degrees of
freedom <code>df</code>, bandwidth <code>bandwidth</code>, etc.
</p>


<h3>Bugs</h3>

<p>Both bandwidth and degrees of freedom are just copied from
the values for one of the chunk spectra, and are thus incorrect.  That means
the cross indicated on the graph is also incorrect.
</p>


<h3>Historical notes</h3>


<ul>
<li> <p><strong>2021-06-26:</strong> Until this date, <code><a href="#topic+pwelch">pwelch()</a></code> passed the
subsampled timeseries portions through <code><a href="#topic+detrend">detrend()</a></code>
before applying the window. This practice was dropped
because it could lead to over-estimates of low frequency
energy (as noticed by Holger Foysi of the University of Siegen),
perhaps because <code><a href="#topic+detrend">detrend()</a></code> considers only endpoints and
therefore can yield inaccurate trend estimates.
In a related change, <code>demean</code> and <code>detrend</code> were added
as formal arguments, to avoid users having to trace the documentation
for <code><a href="stats.html#topic+spectrum">spectrum()</a></code> and then <code><a href="stats.html#topic+spec.pgram">spec.pgram()</a></code>, to learn how to
remove means and trends from data.
For more control, the <code>spec</code> argument was
added to let users sidestep <code><a href="stats.html#topic+spectrum">spectrum()</a></code> entirely, by providing
their own spectral computation functions.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Welch, P. D., 1967. The Use of Fast Fourier Transform for the
Estimation of Power Spectra: A Method Based on Time Averaging Over Short,
Modified Periodograms. <em>IEEE Transactions on Audio Electroacoustics</em>,
AU-15, 70&ndash;73.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
Fs &lt;- 1000
t &lt;- seq(0, 0.296, 1 / Fs)
x &lt;- cos(2 * pi * t * 200) + rnorm(n = length(t))
X &lt;- ts(x, frequency = Fs)
s &lt;- spectrum(X, spans = c(3, 2), main = "random + 200 Hz", log = "no")
w &lt;- pwelch(X, plot = FALSE)
lines(w$freq, w$spec, col = "red")
w2 &lt;- pwelch(X, nfft = 75, plot = FALSE)
lines(w2$freq, w2$spec, col = "green")
abline(v = 200, col = "blue", lty = "dotted")
cat("Checking spectral levels with Parseval's theorem:\n")
cat("var(x)                              = ", var(x), "\n")
cat("2 * sum(s$spec) * diff(s$freq[1:2]) = ", 2 * sum(s$spec) * diff(s$freq[1:2]), "\n")
cat("sum(w$spec) * diff(s$freq[1:2])     = ", sum(w$spec) * diff(w$freq[1:2]), "\n")
cat("sum(w2$spec) * diff(s$freq[1:2])    = ", sum(w2$spec) * diff(w2$freq[1:2]), "\n")
# co2
par(mar = c(3, 3, 2, 1), mgp = c(2, 0.7, 0))
s &lt;- spectrum(co2, plot = FALSE)
plot(log10(s$freq), s$spec * s$freq,
    xlab = expression(log[10] * Frequency), ylab = "Power*Frequency", type = "l"
)
title("Variance-preserving spectrum")
pw &lt;- pwelch(co2, nfft = 256, plot = FALSE)
lines(log10(pw$freq), pw$spec * pw$freq, col = "red")

</code></pre>

<hr>
<h2 id='rangeExtended'>Calculate Range, Extended a Little, as is Done for Axes</h2><span id='topic+rangeExtended'></span>

<h3>Description</h3>

<p>This is analogous to what is done as part of the R axis range calculation,
in the case where <code>xaxs="r"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rangeExtended(x, extend = 0.04)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rangeExtended_+3A_x">x</code></td>
<td>
<p>a numeric vector.</p>
</td></tr>
<tr><td><code id="rangeExtended_+3A_extend">extend</code></td>
<td>
<p>fraction to extend on either end</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-element vector with the extended range of <code>x</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>

<hr>
<h2 id='rangeLimit'>Substitute NA for Data Outside a Range</h2><span id='topic+rangeLimit'></span>

<h3>Description</h3>

<p>Substitute NA for data outside a range, e.g. to remove wild spikes in data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rangeLimit(x, min, max)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rangeLimit_+3A_x">x</code></td>
<td>
<p>vector of values</p>
</td></tr>
<tr><td><code id="rangeLimit_+3A_min">min</code></td>
<td>
<p>minimum acceptable value.  If not supplied, and if <code>max</code> is
also not supplied, a <code>min</code> of the 0.5 percentile will be used.</p>
</td></tr>
<tr><td><code id="rangeLimit_+3A_max">max</code></td>
<td>
<p>maximum acceptable value.  If not supplied, and if <code>min</code> is
also not supplied, a <code>min</code> of the 0.995 percentile will be used.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ten.to.twenty &lt;- rangeLimit(1:100, 10, 20)
</code></pre>

<hr>
<h2 id='read.adp'>Read an adp File</h2><span id='topic+read.adp'></span>

<h3>Description</h3>

<p>Read an ADP data file, producing an <a href="#topic+adp-class">adp</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.adp(
  file,
  from,
  to,
  by,
  tz = getOption("oceTz"),
  longitude = NA,
  latitude = NA,
  manufacturer,
  encoding = NA,
  monitor = FALSE,
  despike = FALSE,
  processingLog,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.adp_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  (For <code>read.adp.sontek.serial</code>, this is generally a list of
files, which will be concatenated.)</p>
</td></tr>
<tr><td><code id="read.adp_+3A_from">from</code></td>
<td>
<p>indication of the first profile to read.  This can be an
integer, the sequence number of the first profile to read, or a POSIXt time
before which profiles should be skipped, or a character string that converts
to a POSIXt time (assuming UTC timezone).  See &ldquo;Examples&rdquo;, and make
careful note of the use of the <code>tz</code> argument. If <code>from</code> is not
supplied, it defaults to 1.</p>
</td></tr>
<tr><td><code id="read.adp_+3A_to">to</code></td>
<td>
<p>an optional indication of the last profile to read, in a
format as described for <code>from</code>.  As a special case, <code>to=0</code> means
to read the file to the end.  If <code>to</code> is not supplied, then it defaults
to 0.</p>
</td></tr>
<tr><td><code id="read.adp_+3A_by">by</code></td>
<td>
<p>an optional indication of the stride length to use while walking through
the file.  If this is an integer, then <code>by-1</code> profiles are skipped
between each pair of profiles that is read, e.g. the default <code>by=1</code>
means to read all the data.  (For RDI files <em>only</em>, there are some
extra features to avoid running out of memory; see &ldquo;Memory considerations&rdquo;.)</p>
</td></tr>
<tr><td><code id="read.adp_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.adp_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.adp_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.adp_+3A_manufacturer">manufacturer</code></td>
<td>
<p>an optional character string indicating the manufacturer, used by
the general function <code>read.adp</code> to select a subsidiary function to use. If this
is not given, then <code><a href="#topic+oceMagic">oceMagic()</a></code> is used to try to infer the type. If this
is provided, then the value <code>"rdi"</code> will cause <code><a href="#topic+read.adp.rdi">read.adp.rdi()</a></code>
to be used, <code>"nortek"</code> will cause <code><a href="#topic+read.adp.nortek">read.adp.nortek()</a></code> to be used,
and <code>"sontek"</code> will cause <code><a href="#topic+read.adp.sontek">read.adp.sontek()</a></code> to be used.</p>
</td></tr>
<tr><td><code id="read.adp_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.adp_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.adp_+3A_despike">despike</code></td>
<td>
<p>if <code>TRUE</code>, <code><a href="#topic+despike">despike()</a></code> will be used to clean
anomalous spikes in heading, etc.</p>
</td></tr>
<tr><td><code id="read.adp_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
<tr><td><code id="read.adp_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="read.adp_+3A_...">...</code></td>
<td>
<p>optional additional arguments that some (but not all)
<code style="white-space: pre;">&#8288;read.adp.*()&#8288;</code> functions pass to lower-level functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Several file types can be handled.  Some of
these functions are wrappers that map to device names, e.g.
<code>read.aquadoppProfiler</code> does its work by calling
<code>read.adp.nortek</code>; in this context, it is worth noting that the
&ldquo;aquadopp&rdquo; instrument is a one-cell profiler that might just as well have
been documented under the heading <code><a href="#topic+read.adv">read.adv()</a></code>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
The contents of that object make sense for the particular instrument
type under study, e.g. if the data file contains
NMEA strings, then navigational data will be stored in an item
called <code>nmea</code> in the <code>data</code> slot).
</p>


<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that read adp data: 
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>
</p>

<hr>
<h2 id='read.adp.ad2cp'>Read an adp File in Nortek AD2CP Format</h2><span id='topic+read.adp.ad2cp'></span>

<h3>Description</h3>

<p>This function is under active development and may change without notice.  In
contrast with other <code>oce</code> reading functions, <code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp()</a></code> focusses just
on one data type within the source file.  Another difference is that it can
either return an object holding the data or just a data frame holding a
description of the data types in the file; indeed, the latter is the default.
See &ldquo;Details&rdquo; for more on the reasons for these departures from the
usual <code>oce</code> pattern.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.adp.ad2cp(
  file,
  from = 1L,
  to = 0L,
  by = 1L,
  dataType = NULL,
  dataSet = 1L,
  tz = getOption("oceTz"),
  longitude = NA,
  latitude = NA,
  plan,
  TOC = FALSE,
  debug = getOption("oceDebug"),
  orientation,
  distance,
  monitor,
  despike,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.adp.ad2cp_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file to
load.</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_from">from</code></td>
<td>
<p>an integer indicating the index number of the first record to
read. This must equal 1, for this version of <code>read.adp.ad2cp</code>.  (If not
provided, <code>from</code> defaults to 1.)</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_to">to</code></td>
<td>
<p>an integer indicating the final record to read. If <code>to</code> is 0L,
which is the default, then the value is changed internally to 1e9, and
reading stops at the end of the file.</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_by">by</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_datatype">dataType</code></td>
<td>
<p>an indication of the data type to be extracted.  If this is
NULL (the default) then <code>read.adp.ad2cp()</code> returns a data frame indicating
the data type occurrence rate in the file.  Otherwise, <code>dataType</code> must be
either a numeric or character value (see &ldquo;Details&rdquo;).  In the numeric
case, which includes both base-10 numbers and <code>raw</code> values, <code>dataType</code> is
converted to an integer that is taken to indicate the data type via ID. The
permitted values follow the Nortek convention, a summary of which is shown
the table at the start of the &ldquo;Details&rdquo; section.  In the character
case, it must be a string taken from that same table.</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_dataset">dataSet</code></td>
<td>
<p>a positive integer that indicates which of the possibly
several data sets stored within a file is to be focussed upon.  By
default, the first data set is chosen. Note that data sets are found
by trying to match each text data chunk against the regular expression
<code>"^GETCLOCKSTR,TIME="</code>.</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_tz">tz</code></td>
<td>
<p>a character value indicating time zone. This is used in
interpreting times stored in the file.</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_longitude">longitude</code>, <code id="read.adp.ad2cp_+3A_latitude">latitude</code></td>
<td>
<p>numerical values indicating the observation
location.</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_plan">plan</code></td>
<td>
<p>optional integer specifying which 'plan' to focus on (see</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_toc">TOC</code></td>
<td>
<p>a logical value.  If this is FALSE (the default) then
the other parameters of the function are used to select data from
the indicated <code>filename</code>, and an <a href="#topic+adp-class">adp</a> object is returned.
However, if <code>TOC</code> is TRUE, then the number of datasets held within
the file is returned.</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_debug">debug</code></td>
<td>
<p>an integer value indicating the level of debugging.  Set to 1 to
get a moderate amount of debugging information, from the R code only, to 2 to
get some debugging information from the C++ code that is used to parse the
data chunks, or to 3 for intensive debugging at both levels.</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_orientation">orientation</code>, <code id="read.adp.ad2cp_+3A_distance">distance</code>, <code id="read.adp.ad2cp_+3A_monitor">monitor</code>, <code id="read.adp.ad2cp_+3A_despike">despike</code></td>
<td>
<p>ignored, provided only for
calling compatibility with other functions that read <a href="#topic+adp-class">adp</a>
files.  A warning is issued if any of these is supplied in a call
to <code>read.adp.ad2cp()</code>.</p>
</td></tr>
<tr><td><code id="read.adp.ad2cp_+3A_...">...</code></td>
<td>
<p>ignored parameters that might be passed to <code>read.adp.ad2cp()</code>
by <code><a href="#topic+read.oce">read.oce()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Why does <code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp()</a></code> focus only on parts of the data file? The answer
lies in the AD2CP format itself, which may combine data subsets of such
differing natures as to break with the <code>oce</code> system of pairing a <code>metadata</code>
slot with a <code>data</code> slot.  For example, in a conventional ADP dataset, the
<code>metadata</code> slot has items for the sampling times, the number of beams, the
blanking distance, the cell size, the number of cells, etc.  Such items have
a natural pairing with elements of the <code>data</code> slot, and <code>oce</code> uses this
pairing in constructing plots and other items. However, an AD2CP file might
combine such data with echosounder measurements, and these will have
different values for number of beams and so forth.  This poses a challenge
in naming conventions within the <code>oce</code> object, with ripple effects for
plotting and data access.  Those ripple effects would extend beyond <code>oce</code>
itself to user code.  To avoid such problems, <code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp()</a></code>
is designed to focus on one data type at a time, relying on users to
keep track of the resultant object, perhaps to combine it with other objects
from within the AD2CP file or other files, in the normal R manner.
</p>
<p>The permitted values for <code>dataType</code> are shown in the table below;
the <code>dataType</code> argument of <code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp()</a></code> may be chosen from any
of the three columns in this table.</p>

<table>
<tr>
 <td style="text-align: right;">
   code (raw) </td><td style="text-align: right;"> code (integer) </td><td style="text-align: right;"> oce name </td>
</tr>
<tr>
 <td style="text-align: right;">
   ---------- </td><td style="text-align: right;"> -------------- </td><td style="text-align: right;"> ----------------- </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x15</code> </td><td style="text-align: right;"> 21 </td><td style="text-align: right;"> <code>burst</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x16</code> </td><td style="text-align: right;"> 22 </td><td style="text-align: right;"> <code>average</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x17</code> </td><td style="text-align: right;"> 23 </td><td style="text-align: right;"> <code>bottomTrack</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x18</code> </td><td style="text-align: right;"> 24 </td><td style="text-align: right;"> <code>interleavedBurst</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1a</code> </td><td style="text-align: right;"> 26 </td><td style="text-align: right;"> <code>burstAltimeterRaw</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1b</code> </td><td style="text-align: right;"> 27 </td><td style="text-align: right;"> <code>DVLBottomTrack</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1c</code> </td><td style="text-align: right;"> 28 </td><td style="text-align: right;"> <code>echosounder</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1d</code> </td><td style="text-align: right;"> 29 </td><td style="text-align: right;"> <code>DVLWaterTrack</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1e</code> </td><td style="text-align: right;"> 30 </td><td style="text-align: right;"> <code>altimeter</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x1f</code> </td><td style="text-align: right;"> 31 </td><td style="text-align: right;"> <code>averageAltimeter</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0x23</code> </td><td style="text-align: right;"> 35 </td><td style="text-align: right;"> <code>echosounderRaw</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
   <code>0xa0</code> </td><td style="text-align: right;"> 160 </td><td style="text-align: right;"> <code>text</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>



<h3>Value</h3>

<p><code>read.adp.ad2cp()</code> returns either an <a href="#topic+adp-class">adp</a> object or
the number of data sets within the file, according to the value
of <code>TOC</code>.
</p>


<h3>Sample of Usage</h3>

<pre>
d &lt;- read.adp.ad2cp("~/test.ad2cp", to=100) # or read.oce()
</pre>


<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Nortek AS. &ldquo;Signature Integration 55|250|500|1000kHz.&rdquo; Nortek AS,
2017.
</p>
<p>Nortek AS. &ldquo;Signature Integration 55|250|500|1000kHz.&rdquo; Nortek AS,
2018.
</p>
<p>Nortek AS. &ldquo;Signature Integration 55|250|500|1000kHz.&rdquo; Nortek AS,
March 31, 2022.
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to ad2cp data: 
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>
</p>
<p>Other functions that read adp data: 
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# You can run this within the oce directory, if you clone from github.
file &lt;- "tests/testthat/local_data/ad2cp/S102791A002_Barrow_v2.ad2cp"
if (file.exists(file)) {
    library(oce)
    d &lt;- read.oce(file)
}

</code></pre>

<hr>
<h2 id='read.adp.nortek'>Read an adp File in Nortek Format</h2><span id='topic+read.adp.nortek'></span>

<h3>Description</h3>

<p>Read an adp File in Nortek Format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.adp.nortek(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  longitude = NA,
  latitude = NA,
  type = NULL,
  orientation,
  distance,
  encoding = NA,
  monitor = FALSE,
  despike = FALSE,
  processingLog,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.adp.nortek_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  (For <code>read.adp.sontek.serial</code>, this is generally a list of
files, which will be concatenated.)</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_from">from</code></td>
<td>
<p>indication of the first profile to read.  This can be an
integer, the sequence number of the first profile to read, or a POSIXt time
before which profiles should be skipped, or a character string that converts
to a POSIXt time (assuming UTC timezone).  See &ldquo;Examples&rdquo;, and make
careful note of the use of the <code>tz</code> argument. If <code>from</code> is not
supplied, it defaults to 1.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_to">to</code></td>
<td>
<p>an optional indication of the last profile to read, in a
format as described for <code>from</code>.  As a special case, <code>to=0</code> means
to read the file to the end.  If <code>to</code> is not supplied, then it defaults
to 0.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_by">by</code></td>
<td>
<p>an optional indication of the stride length to use while walking through
the file.  If this is an integer, then <code>by-1</code> profiles are skipped
between each pair of profiles that is read, e.g. the default <code>by=1</code>
means to read all the data.  (For RDI files <em>only</em>, there are some
extra features to avoid running out of memory; see &ldquo;Memory considerations&rdquo;.)</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_type">type</code></td>
<td>
<p>a character string indicating the type of instrument.
If NULL (the default), then <code><a href="#topic+oceMagic">oceMagic()</a></code> is used to infer the type. If
non-NULL, then the value must be one of: <code>"aquadoppHR"</code>,
<code>"aquadoppProfiler"</code>, <code>"aquadopp"</code>, or <code>"aquadoppPlusMagnetometer"</code>.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_orientation">orientation</code></td>
<td>
<p>an optional character string specifying the orientation of
the sensor, provided for those cases in which it cannot be inferred from the
data file.  The valid choices are <code>"upward"</code>, <code>"downward"</code>, and
<code>"sideward"</code>.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_distance">distance</code></td>
<td>
<p>an optional vector holding the distances of bin centres from
the sensor.  This argument is ignored except for Nortek profilers, and need
not be given if the function determines the distances correctly from the
data.  The problem is that the distance is poorly documented in the Nortek
System Integrator Guide (2008 edition, page 31), so the function must rely
on word-of-mouth formulae that do not work in all cases.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_despike">despike</code></td>
<td>
<p>a logical value indicating whether to use <code><a href="#topic+despike">despike()</a></code> to remove
anomalous spikes in heading, etc.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="read.adp.nortek_+3A_...">...</code></td>
<td>
<p>optional additional arguments that some (but not all)
<code style="white-space: pre;">&#8288;read.adp.*()&#8288;</code> functions pass to lower-level functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
The contents of that object make sense for the particular instrument
type under study, e.g. if the data file contains
NMEA strings, then navigational data will be stored in an item
called <code>nmea</code> in the <code>data</code> slot).
</p>


<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Information on Nortek profilers (including the System Integrator Guide,
which explains the data format byte-by-byte) is available at
<code style="white-space: pre;">&#8288;https://www.nortekusa.com/&#8288;</code>.  (One must join the site to see the
manuals.)
</p>
</li>
<li><p> The Nortek Knowledge Center
<code style="white-space: pre;">&#8288;https://www.nortekusa.com/en/knowledge-center&#8288;</code> may be of help if
problems arise in dealing with data from Nortek instruments.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that read adp data: 
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>
</p>

<hr>
<h2 id='read.adp.rdi'>Read an adp File in Teledyne/RDI Format</h2><span id='topic+read.adp.rdi'></span>

<h3>Description</h3>

<p>Read a Teledyne/RDI ADCP file (called 'adp' in oce).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.adp.rdi(
  file,
  from,
  to,
  by,
  tz = getOption("oceTz"),
  longitude = NA,
  latitude = NA,
  type = c("workhorse"),
  which,
  encoding = NA,
  monitor = FALSE,
  despike = FALSE,
  processingLog,
  testing = FALSE,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.adp.rdi_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  (For <code>read.adp.sontek.serial</code>, this is generally a list of
files, which will be concatenated.)</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_from">from</code></td>
<td>
<p>indication of the first profile to read.  This can be an
integer, the sequence number of the first profile to read, or a POSIXt time
before which profiles should be skipped, or a character string that converts
to a POSIXt time (assuming UTC timezone).  See &ldquo;Examples&rdquo;, and make
careful note of the use of the <code>tz</code> argument. If <code>from</code> is not
supplied, it defaults to 1.</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_to">to</code></td>
<td>
<p>an optional indication of the last profile to read, in a
format as described for <code>from</code>.  As a special case, <code>to=0</code> means
to read the file to the end.  If <code>to</code> is not supplied, then it defaults
to 0.</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_by">by</code></td>
<td>
<p>an optional indication of the stride length to use while walking through
the file.  If this is an integer, then <code>by-1</code> profiles are skipped
between each pair of profiles that is read, e.g. the default <code>by=1</code>
means to read all the data.  (For RDI files <em>only</em>, there are some
extra features to avoid running out of memory; see &ldquo;Memory considerations&rdquo;.)</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_type">type</code></td>
<td>
<p>character string indicating the type of instrument.</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_which">which</code></td>
<td>
<p>optional character value.  If this is <code>"??"</code> then the
<code><a href="#topic+read.adp.rdi">read.adp.rdi()</a></code> works by locating the indices in <code>file</code> at which data
segments begin, and storing them as <code>index</code> in a list that is returned.
The other entry of the list is <code>time</code>, the time of the observation.</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_despike">despike</code></td>
<td>
<p>if <code>TRUE</code>, <code><a href="#topic+despike">despike()</a></code> will be used to clean
anomalous spikes in heading, etc.</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_testing">testing</code></td>
<td>
<p>logical value (IGNORED).</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="read.adp.rdi_+3A_...">...</code></td>
<td>
<p>optional additional arguments that some (but not all)
<code style="white-space: pre;">&#8288;read.adp.*()&#8288;</code> functions pass to lower-level functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As of 2016-09-25, this function has provisional functionality to
read data from the new &quot;SentinelV&quot; series ADCP &ndash; essentially a
combination of a 4 beam workhorse with an additional vertical
centre beam.
</p>
<p>If a heading bias had been set with the <code>EB</code> command during the setup
for the deployment, then a heading bias will have been stored in the file's
header.  This value is stored in the object's metadata as
<code>metadata$heading.bias</code>.  <strong>Importantly</strong>, this value is
subtracted from the headings stored in the file, and the result of this
subtraction is stored in the objects heading value (in <code>data$heading</code>).
It should be noted that <code>read.adp.rdi()</code> was tested for firmware
version 16.30.  For other versions, there may be problems.  For example, the
serial number is not recognized properly for version 16.28.
</p>
<p>In Teledyne/RDI ADP data files, velocities are coded to signed 2-byte integers, with a
scale factor being used to convert to velocity in metres per second.  These
two facts control the maximum recordable velocity and the velocity
resolution, values that may be retrieved for an ADP object name <code>d</code>
with <code>d[["velocityMaximum"]]</code> and <code>d[["velocityResolution"]]</code>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
The contents of that object make sense for the particular instrument
type under study, e.g. if the data file contains
NMEA strings, then navigational data will be stored in an item
called <code>nmea</code> in the <code>data</code> slot).
</p>


<h3>Handling of old file formats</h3>


<ol>
<li><p> Early PD0 file formats stored the year of sampling with a different
base year than that used in modern files.  To accommodate this,
<code>read.adp.rdi</code> examines the inferred year, and if it is greater than
2050, then 100 years are subtracted from the time. This offset was
inferred by tests with sample files, but <em>not</em> from RDI documentation,
so it is somewhat risky.  If the authors can find RDI documentation that
indicates the condition in which this century offset is required, then
a change will be made to the code.  Even if not, the method should
not cause problems for a long time.
</p>
</li></ol>



<h3>Names of items in data slot</h3>

<p>The names of items in the <code>data</code> slot are below. Not all items are present
for ll file varieties; use e.g. <code>names(d[["data"]])</code> to determine the
names used in an object named <code>d</code>. In this list, items are either
a vector (with one sample per time of measurement), a
<a href="base.html#topic+matrix">matrix</a> with first index for time and second for bin number,
or an <a href="base.html#topic+array">array</a> with first index for time, second for bin number,
and third for beam number. Items are of vector type, unless
otherwise indicated.
</p>

<table>
<tr>
 <td style="text-align: right;">
<strong>Item</strong> </td><td style="text-align: right;"> <strong>Meaning</strong> </td>
</tr>
<tr>
 <td style="text-align: right;">
<code>a</code>                                    </td><td style="text-align: right;"> signal amplitude array (units?)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>ambientTemp</code>                          </td><td style="text-align: right;"> ambient temperature (degC)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>attitude</code>                             </td><td style="text-align: right;"> attitude (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>attitudeTemp</code>                         </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>avgMagnitudeVelocityEast</code>             </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>avgMagnitudeVelocityNorth</code>            </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>avgSpeed</code>                             </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>avgTrackMagnetic</code>                     </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>avgTrackTrue</code>                         </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>avgTrueVelocityEast</code>                  </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>avgTrueVelocityNorth</code>                 </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>br</code>                                   </td><td style="text-align: right;"> bottom range matrix (m)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>bv</code>                                   </td><td style="text-align: right;"> bottom velocity matrix (m/s)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>contaminationSensor</code>                  </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>depth</code>                                </td><td style="text-align: right;"> depth (m)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>directionMadeGood</code>                    </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>distance</code>                             </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>firstLatitude</code>                        </td><td style="text-align: right;"> latitude at start of profile (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>firstLongitude</code>                       </td><td style="text-align: right;"> longitude at start of profile (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>firstTime</code>                            </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>g</code>                                    </td><td style="text-align: right;"> data goodness matrix (units?)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>heading</code>                              </td><td style="text-align: right;"> instrument heading (degrees)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>headingStd</code>                           </td><td style="text-align: right;"> instrument heading std-dev (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>lastLatitude</code>                         </td><td style="text-align: right;"> latitude at end of profile (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>lastLongitude</code>                        </td><td style="text-align: right;"> longitude at end of profile (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>lastTime</code>                             </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>numberOfHeadingSamplesAveraged</code>       </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>numberOfMagneticTrackSamplesAveraged</code> </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>numberOfPitchRollSamplesAveraged</code>     </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>numberOfSpeedSamplesAveraged</code>         </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>numberOfTrueTrackSamplesAveraged</code>     </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>pitch</code>                                </td><td style="text-align: right;"> instrument pitch (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>pitchStd</code>                             </td><td style="text-align: right;"> instrument pitch std-dev (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>pressure</code>                             </td><td style="text-align: right;"> pressure (dbar)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>pressureMinus</code>                        </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>pressurePlus</code>                         </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>pressureStd</code>                          </td><td style="text-align: right;"> pressure std-dev (dbar)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>primaryFlags</code>                         </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>q</code>                                    </td><td style="text-align: right;"> data quality array</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>roll</code>                                 </td><td style="text-align: right;"> instrument roll (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>rollStd</code>                              </td><td style="text-align: right;"> instrument roll std-dev (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>salinity</code>                             </td><td style="text-align: right;"> salinity</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>shipHeading</code>                          </td><td style="text-align: right;"> ship heading (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>shipPitch</code>                            </td><td style="text-align: right;"> ship pitch (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>shipRoll</code>                             </td><td style="text-align: right;"> ship roll (deg)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>soundSpeed</code>                           </td><td style="text-align: right;"> sound speed (m/s)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>speedMadeGood</code>                        </td><td style="text-align: right;"> speed over ground (?) (m/s)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>speedMadeGoodEast</code>                    </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>speedMadeGoodNorth</code>                   </td><td style="text-align: right;"> (FIXME add a description here)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>temperature</code>                          </td><td style="text-align: right;"> temperature (degC)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>time</code>                                 </td><td style="text-align: right;"> profile time (POSIXct)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>v</code>                                    </td><td style="text-align: right;"> velocity array (m/s)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>xmitCurrent</code>                          </td><td style="text-align: right;"> transmit current (unit?)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>xmitVoltage</code>                          </td><td style="text-align: right;"> transmit voltage</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>



<h3>Memory considerations</h3>

<p>For <code>RDI</code> files only, and only in the case where <code>by</code> is not specified,
an attempt is made to avoid running out of memory by skipping some profiles
in large input files. This only applies if <code>from</code> and <code>to</code> are both
integers; if they are times, none of the rest of this section applies.
</p>
<p>A key issue is that RDI files store velocities in 2-byte values, which is
not a format that R supports. These velocities become 8-byte (numeric) values
in R. Thus, the R object created by <code>read.adp.rdi</code> will require more memory
than that of the data file. A scale factor can be estimated by ignoring
vector quantities (e.g. time, which has just one value per profile) and concentrating on matrix properties
such as velocity, backscatter, and correlation. These three elements have equal dimensions.
Thus, each 4-byte slide in the data file (2 bytes + 1 byte + 1 byte)
corresponds to 10 bytes in the object (8 bytes + 1 byte + 1 byte).
Rounding up the resultant 10/4 to 3 for safety, we conclude that any limit on the
size of the R object corresponds to a 3X smaller limit on file size.
</p>
<p>Various things can limit the size of objects in R, but a strong upper limit
is set by the space the operating system provides to R. The least-performant machines
in typical use appear to be Microsoft-Windows systems, which limit R objects to
about 2e6 bytes (see <code>?Memory-limits</code>).  Since R routinely duplicates objects for certain tasks
(e.g. for call-by-value in function evaluation), <code>read.adp.rdi</code> uses a safety
factor in its calculation of when to auto-decimate a file. This factor is set to 3,
based partly on the developers' experience with datasets in their possession.
Multiplied by the previously stated safety factor of 3,
this suggests that the 2 GB limit on R objects corresponds to approximately a
222 MB limit on file size. In the present version of <code>read.adp.rdi</code>, this
value is lowered to 200 MB for simplicity. Larger files are considered to be &quot;big&quot;,
and are decimated unless the user supplies a value for the <code>by</code> argument.
</p>
<p>The decimation procedure has two cases.
</p>

<ol>
<li><p> If <code>from=1</code> and <code>to=0</code> (or if neither <code>from</code> or <code>to</code> is given), then the
intention is to process the full span of the data.  If the input file is
under 200 MB, then <code>by</code> defaults to 1, so that all profiles are read.
For larger files, <code>by</code> is set to the <code><a href="base.html#topic+ceiling">ceiling()</a></code> of the
ratio of input file size to 200 MB.
</p>
</li>
<li><p> If <code>from</code> exceeds 1, and/or <code>to</code> is nonzero, then
the intention is to process only an interior subset of the file. In this
case, <code>by</code> is calculated as the <code><a href="base.html#topic+ceiling">ceiling()</a></code> of
the ratio of <code>bbp*(1+to-from)</code> to 200 MB, where <code>bbp</code> is the number
of file bytes per profile. Of course, <code>by</code> is set to 1, if this
ratio is less than 1.
</p>
</li></ol>

<p>If the result of these calculations is that <code>by</code> exceeds 1, then
messages are printed to alert the user that the file will be decimated,
and also <code>monitor</code> is set to <code>TRUE</code>, so that a textual progress bar
is shown (if the session is interactive).
</p>


<h3>Development Notes</h3>

<p>An important part of the work of this function is to recognize what
will be called &quot;data chunks&quot; by two-byte ID sequences. This function is
developed in a practical way, with emphasis being focussed on
data files in the possession of the developers. Since Teledyne-RDI tends
to introduce new ID codes with new instruments, that means that
<code>read.adp.rdi</code> may not work on recently-developed instruments.
</p>
<p>The following two-byte ID codes are recognized by <code>read.adp.rdi</code>
at this time (with bytes listed in natural order, LSB byte before
MSB). Items preceded by an asterisk are recognized, but not handled,
and so produce a warning.
</p>

<table>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> <strong>Byte 1</strong> </td><td style="text-align: right;"> <strong>Byte 2</strong> </td><td style="text-align: right;"> <strong>Meaning</strong></td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x01 </td><td style="text-align: right;"> velocity</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x01 </td><td style="text-align: right;"> velocity</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x02 </td><td style="text-align: right;"> correlation</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x03 </td><td style="text-align: right;"> echo intensity</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x04 </td><td style="text-align: right;"> percent good</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x06 </td><td style="text-align: right;"> bottom track</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x0a </td><td style="text-align: right;"> Sentinel vertical beam velocity</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x0b </td><td style="text-align: right;"> Sentinel vertical beam correlation</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x0c </td><td style="text-align: right;"> Sentinel vertical beam amplitude</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x0d </td><td style="text-align: right;"> Sentinel vertical beam percent good</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x20 </td><td style="text-align: right;"> VMDASS</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x30 </td><td style="text-align: right;"> Binary Fixed Attitude header</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x32 </td><td style="text-align: right;"> Sentinel transformation matrix</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x0a </td><td style="text-align: right;"> Sentinel data</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x0b </td><td style="text-align: right;"> Sentinel correlation</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x0c </td><td style="text-align: right;"> Sentinel amplitude</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x00 </td><td style="text-align: right;"> 0x0d </td><td style="text-align: right;"> Sentinel percent good</td>
</tr>
<tr>
 <td style="text-align: right;">
</td><td style="text-align: right;"> 0x01 </td><td style="text-align: right;"> 0x0f </td><td style="text-align: right;"> ?? something to do with V series and 4-beam</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>Lacking a comprehensive Teledyne-RDI listing of ID codes,
the authors have cobbled together a listing from documents to which they
have access, as follows.
</p>

<ul>
<li><p> Table 33 of reference 3 lists codes as follows:
</p>

<table>
<tr>
 <td style="text-align: right;">
<strong>Standard ID</strong>  </td><td style="text-align: right;">  <strong>Standard plus 1D</strong> </td><td style="text-align: right;"> <strong>DESCRIPTION</strong></td>
</tr>
<tr>
 <td style="text-align: right;">
MSB    LSB  </td><td style="text-align: right;">        MSB    LSB</td><td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
---    ---  </td><td style="text-align: right;">        ---    ---</td><td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
7F     7F  </td><td style="text-align: right;">         7F     7F </td><td style="text-align: right;">   Header</td>
</tr>
<tr>
 <td style="text-align: right;">
00     00  </td><td style="text-align: right;">         00     01 </td><td style="text-align: right;">   Fixed Leader</td>
</tr>
<tr>
 <td style="text-align: right;">
00     80  </td><td style="text-align: right;">         00     81 </td><td style="text-align: right;">   Variable Leader</td>
</tr>
<tr>
 <td style="text-align: right;">
01     00  </td><td style="text-align: right;">         01     01 </td><td style="text-align: right;">   Velocity Profile Data</td>
</tr>
<tr>
 <td style="text-align: right;">
02     00  </td><td style="text-align: right;">         02     01 </td><td style="text-align: right;">   Correlation Profile Data</td>
</tr>
<tr>
 <td style="text-align: right;">
03     00  </td><td style="text-align: right;">         03     01 </td><td style="text-align: right;">   Echo Intensity Profile Data</td>
</tr>
<tr>
 <td style="text-align: right;">
04     00  </td><td style="text-align: right;">         04     01 </td><td style="text-align: right;">   Percent Good Profile Data</td>
</tr>
<tr>
 <td style="text-align: right;">
05     00  </td><td style="text-align: right;">         05     01 </td><td style="text-align: right;">   Status Profile Data</td>
</tr>
<tr>
 <td style="text-align: right;">
06     00  </td><td style="text-align: right;">         06     01 </td><td style="text-align: right;">   Bottom Track Data</td>
</tr>
<tr>
 <td style="text-align: right;">
20     00  </td><td style="text-align: right;">         20     00 </td><td style="text-align: right;">   Navigation</td>
</tr>
<tr>
 <td style="text-align: right;">
30     00  </td><td style="text-align: right;">         30     00 </td><td style="text-align: right;">   Binary Fixed Attitude</td>
</tr>
<tr>
 <td style="text-align: right;">
30  40-F0  </td><td style="text-align: right;">         30  40-F0 </td><td style="text-align: right;">   Binary Variable Attitude</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

</li>
<li><p> Table 6 on p90 of reference 4 lists &quot;Fixed Leader Navigation&quot; ID
codes (none of which are handled by <code>read.adp.rdi</code> yet)
as follows (the format is reproduced literally; note that
e.g. 0x2100 is 0x00,0x21 in the oce notation):
</p>

<table>
<tr>
 <td style="text-align: right;">
<strong>ID</strong> </td><td style="text-align: right;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: right;">
0x2100 </td><td style="text-align: right;">  $xxDBT</td>
</tr>
<tr>
 <td style="text-align: right;">
0x2101 </td><td style="text-align: right;">  $xxGGA</td>
</tr>
<tr>
 <td style="text-align: right;">
0x2102 </td><td style="text-align: right;">  $xxVTG</td>
</tr>
<tr>
 <td style="text-align: right;">
0x2103 </td><td style="text-align: right;">  $xxGSA</td>
</tr>
<tr>
 <td style="text-align: right;">
0x2104 </td><td style="text-align: right;">  $xxHDT, $xxHGD or $PRDID</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>and following pages in that manual reveal the following meanings
</p>

<table>
<tr>
 <td style="text-align: right;">
<strong>Symbol</strong> </td><td style="text-align: right;"> <strong>Meaning</strong></td>
</tr>
<tr>
 <td style="text-align: right;">
<code>DBT</code>   </td><td style="text-align: right;"> depth below transducer</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>GGA</code>   </td><td style="text-align: right;"> global positioning system</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>VTA</code>   </td><td style="text-align: right;"> track made good and ground speed</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>GSA</code>   </td><td style="text-align: right;"> GPS DOP and active satellites</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>HDT</code>   </td><td style="text-align: right;"> heading, true</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>HDG</code>   </td><td style="text-align: right;"> heading, deviation, and variation</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>PRDID</code> </td><td style="text-align: right;"> heading, pitch and roll</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

</li></ul>



<h3>Error recovery</h3>

<p>Files can sometimes be corrupted, and <code>read.adp.rdi</code> has ways to handle two types
of error that have been noticed in files supplied by users.
</p>

<ol>
<li><p> There are two bytes within each ensemble that indicate the number of bytes to check within
that ensemble, to get the checksum. Sometimes, those two bytes can be erroneous, so that
the wrong number of bytes are checked, leading to a failed checksum. As a preventative
measure, <code>read.adp.rdi</code> checks the stated ensemble length, whenever it detects a
failed checksum. If that length agrees with the length of the most recent ensemble that
had a good checksum, then the ensemble is declared as faulty and is ignored. However,
if the length differs from that of the most recent accepted ensemble, then <code>read.adp.rdi</code>
goes back to just after the start of the ensemble, and searches forward for the next two-byte
pair, namely <code style="white-space: pre;">&#8288;0x7f 0x7f&#8288;</code>, that designates the start of an ensemble.  Distinct notifications
are given about these two cases, and they give the byte numbers in the original file, as a way
to help analysts who want to look at the data stream with other tools.
</p>
</li>
<li><p> At the end of an ensemble, the next two characters ought to be <code style="white-space: pre;">&#8288;0x7f 0x7f&#8288;</code>, and if they
are not, then the next ensemble is faulty. If this error occurs, <code>read.adp.rdi</code> attempts
to recover by searching forward to the next instance of this two-byte pair, discarding any
information that is present in the mangled ensemble.
</p>
</li></ol>

<p>In each of these cases, warnings are printed about ensembles that seem problematic.
Advanced users who want to diagnose the problem further might find it helpful to
examine the original data file using other tools. To this end, <code>read.adp.rdi</code>
inserts an element named <code>ensembleInFile</code> into the <code>metadata</code> slot.
This gives the starting byte number of each inferred ensemble within the original data
file.  For example, if <code>d</code> is an object read with <code>read.adp.rdi</code>, then using
</p>
<div class="sourceCode"><pre>plot(d[["time"]][-1], diff(d[["ensembleInFile"]]))
</pre></div>
<p>can be a good way to narrow in on problems.
</p>


<h3>Changes</h3>


<ul>
<li><p> The <code>bq</code> (bottom-track quality) field was called <code>bc</code> until 2023-02-09.
See https://github.com/dankelley/oce/issues/2039 for discussion.
</p>
</li></ul>



<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>


<ol>
<li><p> Teledyne-RDI, 2007. <em>WorkHorse commands and output data
format.</em> P/N 957-6156-00 (November 2007).  (Section 5.3 h details the binary
format, e.g. the file should start with the byte <code>0x7f</code> repeated twice,
and each profile starts with the bytes <code>0x80</code>, followed by <code>0x00</code>,
followed by the sequence number of the profile, represented as a
little-endian two-byte short integer.  <code>read.adp.rdi</code> uses these
sequences to interpret data files.)
</p>
</li>
<li><p> Teledyne RD Instruments, 2015. <em>V Series monitor, sentinel Output Data Format.</em>
P/N 95D-6022-00 (May 2015). <code>SV_ODF_May15.pdf</code>
</p>
</li>
<li><p> Teledyne RD Instruments, 2014. <em>Ocean Surveyor / Ocean Observer Technical Manual.</em>
P/N 95A-6012-00 (April 2014). <code>OS_TM_Apr14.pdf</code>
</p>
</li>
<li><p> Teledyne RD Instruments, 2001. <em>WinRiver User's Guide International Version.</em>
P/N 957-6171-00 (June 2001) <code style="white-space: pre;">&#8288;WinRiver User Guide International Version.pdf.pdf&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that read adp data: 
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>adp &lt;- read.adp.rdi(system.file("extdata", "adp_rdi.000", package = "oce"))
summary(adp)

</code></pre>

<hr>
<h2 id='read.adp.sontek'>Read an adp File in Sontek Format</h2><span id='topic+read.adp.sontek'></span>

<h3>Description</h3>

<p>Read a Sontek acoustic-Doppler profiler file (see reference 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.adp.sontek(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  longitude = NA,
  latitude = NA,
  type = c("adp", "pcadp"),
  encoding = NA,
  monitor = FALSE,
  despike = FALSE,
  processingLog,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.adp.sontek_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  (For <code>read.adp.sontek.serial</code>, this is generally a list of
files, which will be concatenated.)</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_from">from</code></td>
<td>
<p>indication of the first profile to read.  This can be an
integer, the sequence number of the first profile to read, or a POSIXt time
before which profiles should be skipped, or a character string that converts
to a POSIXt time (assuming UTC timezone).  See &ldquo;Examples&rdquo;, and make
careful note of the use of the <code>tz</code> argument. If <code>from</code> is not
supplied, it defaults to 1.</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_to">to</code></td>
<td>
<p>an optional indication of the last profile to read, in a
format as described for <code>from</code>.  As a special case, <code>to=0</code> means
to read the file to the end.  If <code>to</code> is not supplied, then it defaults
to 0.</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_by">by</code></td>
<td>
<p>an optional indication of the stride length to use while walking through
the file.  If this is an integer, then <code>by-1</code> profiles are skipped
between each pair of profiles that is read, e.g. the default <code>by=1</code>
means to read all the data.  (For RDI files <em>only</em>, there are some
extra features to avoid running out of memory; see &ldquo;Memory considerations&rdquo;.)</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_type">type</code></td>
<td>
<p>A character string indicating the type of instrument.</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_despike">despike</code></td>
<td>
<p>if <code>TRUE</code>, <code><a href="#topic+despike">despike()</a></code> will be used to clean
anomalous spikes in heading, etc.</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="read.adp.sontek_+3A_...">...</code></td>
<td>
<p>optional additional arguments that some (but not all)
<code style="white-space: pre;">&#8288;read.adp.*()&#8288;</code> functions pass to lower-level functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
The contents of that object make sense for the particular instrument
type under study, e.g. if the data file contains
NMEA strings, then navigational data will be stored in an item
called <code>nmea</code> in the <code>data</code> slot).
</p>


<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>


<ol>
<li><p> Information about Sontek profilers is available at https://www.sontek.com.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that read adp data: 
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>
</p>

<hr>
<h2 id='read.adp.sontek.serial'>Read an adp File in Serial Sontek Format</h2><span id='topic+read.adp.sontek.serial'></span>

<h3>Description</h3>

<p>Read a Sontek acoustic-Doppler profiler file, in a serial form that
is possibly unique to Dalhousie University.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.adp.sontek.serial(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  longitude = NA,
  latitude = NA,
  type = c("adp", "pcadp"),
  beamAngle = 25,
  orientation,
  encoding = NA,
  monitor = FALSE,
  processingLog,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.adp.sontek.serial_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  (For <code>read.adp.sontek.serial</code>, this is generally a list of
files, which will be concatenated.)</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_from">from</code></td>
<td>
<p>indication of the first profile to read.  This can be an
integer, the sequence number of the first profile to read, or a POSIXt time
before which profiles should be skipped, or a character string that converts
to a POSIXt time (assuming UTC timezone).  See &ldquo;Examples&rdquo;, and make
careful note of the use of the <code>tz</code> argument. If <code>from</code> is not
supplied, it defaults to 1.</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_to">to</code></td>
<td>
<p>an optional indication of the last profile to read, in a
format as described for <code>from</code>.  As a special case, <code>to=0</code> means
to read the file to the end.  If <code>to</code> is not supplied, then it defaults
to 0.</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_by">by</code></td>
<td>
<p>an optional indication of the stride length to use while walking through
the file.  If this is an integer, then <code>by-1</code> profiles are skipped
between each pair of profiles that is read, e.g. the default <code>by=1</code>
means to read all the data.  (For RDI files <em>only</em>, there are some
extra features to avoid running out of memory; see &ldquo;Memory considerations&rdquo;.)</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_type">type</code></td>
<td>
<p>a character string indicating the type of instrument.</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_beamangle">beamAngle</code></td>
<td>
<p>angle between instrument axis and beams, in degrees.</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_orientation">orientation</code></td>
<td>
<p>optional character string specifying the orientation of the
sensor, provided for those cases in which it cannot be inferred from the
data file.  The valid choices are <code>"upward"</code>, <code>"downward"</code>, and
<code>"sideward"</code>.</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="read.adp.sontek.serial_+3A_...">...</code></td>
<td>
<p>optional additional arguments that some (but not all)
<code style="white-space: pre;">&#8288;read.adp.*()&#8288;</code> functions pass to lower-level functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
The contents of that object make sense for the particular instrument
type under study, e.g. if the data file contains
NMEA strings, then navigational data will be stored in an item
called <code>nmea</code> in the <code>data</code> slot).
</p>


<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that read adp data: 
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>
</p>

<hr>
<h2 id='read.adv'>Read an adv File</h2><span id='topic+read.adv'></span>

<h3>Description</h3>

<p>Read an ADV data file, producing an object of type <code>adv</code>. This
function works by transferring control to a more specialized function,
e.g. <code><a href="#topic+read.adp.nortek">read.adp.nortek()</a></code> and <code><a href="#topic+read.adp.sontek">read.adp.sontek()</a></code>,
and in many cases users will find it preferable to either use these or
the several even more specialized functions, if the file type is
known.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.adv(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  type = c("nortek", "sontek", "sontek.adr", "sontek.text"),
  header = TRUE,
  encoding = NA,
  longitude = NA,
  latitude = NA,
  start = NULL,
  deltat = NA,
  debug = getOption("oceDebug"),
  monitor = FALSE,
  processingLog = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.adv_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  It is also possible to give <code>file</code> as a vector of filenames,
to handle the case of data split into files by a data logger.  In the
multi-file case, <code>header</code> must be <code>FALSE</code>, <code>start</code> must be a
vector of times, and <code>deltat</code> must be provided.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_from">from</code></td>
<td>
<p>index number of the first profile to be read, or the time of
that profile, as created with <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code> (hint: use
<code>tz="UTC"</code>).  This argument is ignored if <code>header==FALSE</code>. See
&ldquo;Examples&rdquo;.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_to">to</code></td>
<td>
<p>indication of the last profile to read, in a format matching that
of <code>from</code>.  This is ignored if <code>header==FALSE</code>.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_by">by</code></td>
<td>
<p>an indication of the stride length to use while walking through
the file.  This is ignored if <code>header==FALSE</code>.  Otherwise, if this is
an integer, then <code>by-1</code> profiles are skipped between each pair of
profiles that is read. This may not make much sense, if the data are not
equi-spaced in time.  If <code>by</code> is a string representing a time interval,
in colon-separated format, then this interval is divided by the sampling
interval, to get the stride length. <em>BUG:</em> <code>by</code> only partially
works; see &ldquo;Bugs&rdquo;.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_type">type</code></td>
<td>
<p>character string indicating type of file, and used by
<code>read.adv</code> to dispatch to one of the speciality functions.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_header">header</code></td>
<td>
<p>A logical value indicating whether the file starts with a header.
(This will not be the case for files that are created by data loggers that
chop the raw data up into a series of sub-files, e.g. once per hour.)</p>
</td></tr>
<tr><td><code id="read.adv_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_start">start</code></td>
<td>
<p>the time of the first sample, typically created with
<code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>.  This may be a vector of times,
if <code>filename</code> is a vector of file names.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_deltat">deltat</code></td>
<td>
<p>the time between samples. (This is mandatory if
<code>header=FALSE</code>.)</p>
</td></tr>
<tr><td><code id="read.adv_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  The value indicates the depth
within the call stack to which debugging applies.  For example,
<code>read.adv.nortek()</code> calls <code>read.header.nortek()</code>, so that
<code>read.adv.nortek(...,debug=2)</code> provides information about not just the
main body of the data file, but also the details of the header.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.adv_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This parameter is typically only provided for internal calls; the default
that it provides is better for normal calls by a user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Files <em>without</em> headers may be created in experiments in which a data
logger was set up to monitor the serial data stream from an instrument.  The
lack of header information places a burden on the user, who must supply such
basic information as the times of observations, the instrument orientation,
the instrument coordinate system, etc.  Example 3 below shows how to deal
with such files.  Three things should be noted.
</p>

<ol>
<li><p> The user must choose the appropriate <code>read.adv</code> variant
corresponding to the instrument in question.  (This is necessary because
<code><a href="#topic+oceMagic">oceMagic()</a></code>, which is used by the generic <code><a href="#topic+read.oce">read.oce()</a></code>
routine, cannot determine the type of instrument by examining a file that
lacks a header.)
</p>
</li>
<li><p> The call to the <code>read</code> function must include a start time
(<code>start</code>) and the number of seconds between data (<code>deltat</code>),
again, because the instrument data stream may lack those things when the
device is set to a serial mode.  Also, of course, it is necessary to set
<code>header=FALSE</code> in the function call.
</p>
</li>
<li><p> Once the file has been read in, the user will be obliged to specify
other information, for the object to be well-formed.  For example, the
<code>read</code> function will have no way of knowing the instrument orientation,
the coordinate system being used, the transformation matrix to go from
<code>"beam"</code> to <code>"xyz"</code> coordinates, or the instrument heading, pitch,
and roll, to go from <code>"xyz"</code> coordinates to <code>"enu"</code> coordinates.
Such things are illustrated in example 3 below.
</p>
</li></ol>

<p>In ADV data files, velocities are coded to signed 2-byte integers, with a
scale factor being used to convert to velocity in metres per second.  These
two facts control the maximum recordable velocity and the velocity
resolution, values that may be retrieved for an ADV object name <code>d</code>
with <code>d[["velocityMaximum"]]</code> and <code>d[["velocityResolution"]]</code>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+adv-class">adv</a> object that contains measurements made with an ADV device.
</p>
<p>The <code>metadata</code> contains information as given in the following table.
The <code style="white-space: pre;">&#8288;Nortek name'' is the name used in the Nortek System Integrator Guide (reference 1) and the &#8288;</code>Sontek name&rdquo; is the name used in the relevant
Sontek documentation.  References are given in square brackets.
</p>

<table>
<tr>
 <td style="text-align: left;">

<strong><code>metadata</code> name</strong></td><td style="text-align: left;"> <strong>Nortek name</strong> </td><td style="text-align: left;"> <strong>Sontek name</strong> </td><td style="text-align: left;"> <strong>Meaning</strong></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>manufacturer</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Either <code>"nortek"</code> or <code>"sontek"</code></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>instrumentType</code></td><td style="text-align: left;">  - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Either <code>"vector"</code> or <code>"adv"</code></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>filename</code></td><td style="text-align: left;">  - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Name of data file(s)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>latitude</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;">  Latitude of mooring (if applicable)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>longitude</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Longitude of mooring (if applicable)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfSamples</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Number of data samples in file</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfBeams</code></td><td style="text-align: left;"> NBeams (reference 1, p18) </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Number of beams (always 3)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfBeamSequencesPerBurst</code></td><td style="text-align: left;"> NPings </td><td style="text-align: left;"> - </td><td style="text-align: left;"> number of beam sequences per burst</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>measurementInterval</code></td><td style="text-align: left;"> MeasInterval (reference 1 p31) </td><td style="text-align: left;"> - </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">

<code>samplingRate</code></td><td style="text-align: left;"> 512/(AvgInterval) (reference 1 p30; reference 4) </td><td style="text-align: left;"> - #' </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">

</td>
</tr>

</table>

<p>The <code>data</code> list contains items with names corresponding to <code>adp</code>
objects, with an exception for Nortek data.  Nortek instruments report some
things at a time interval that is longer than the velocity sampling, and
these are stored in <code>data</code> as <code>timeSlow</code>, <code>headingSlow</code>,
<code>pitchSlow</code>, <code>rollSlow</code>, and <code>temperatureSlow</code>; if burst
sampling was used, there will also be items <code>recordsBurst</code> and
<code>timeBurst</code>.
</p>
<p>The <code>processingLog</code> is in the standard format.
</p>


<h3>Nortek files</h3>

<p><strong>Sampling-rate and similar issues</strong>
</p>
<p>The data format is inferred from the System Integrator Guide (reference 1A) and System
Integrator Manual (reference 1B).  These document lacks clarity in spots, and so
<code>read.adv.nortek</code> contains some assumptions that are noted here, so
that users will be aware of possible problems.
</p>
<p>A prominent example is the specification of the sampling rate, stored in
<code>metadata$sampingRate</code> in the return value.  Repeated examination of
the System Integrator Guide (reference 1) failed to indicate where this value is
stored in the various headers contained in Vector datasets.  After some
experimentation with a few data files, <code>read.adv.nortek</code> was set up to
calculate <code>metadata$samplingRate</code> as <code>512/AvgInterval</code> where
<code>AvgInterval</code> is a part of the <code style="white-space: pre;">&#8288;User Configuration'' header (reference 1 p30), where the explanation is &#8288;</code>average interval in seconds&rdquo;).  This formula was
developed through trial and error, but it was confirmed later on the Nortek
discussion group, and it should appear in upcoming versions of (reference 1).
</p>
<p>Another basic issue is the determination of whether an instrument had
recorded in continuous mode or burst mode.  One might infer that
<code>TimCtrlReg</code> in the <code style="white-space: pre;">&#8288;User Configuration'' header (reference 1 p30) determines this, in bits 1 and 2.  However, this was the case in test files available to the author.  For this reason, `read.adv.nortek` infers the mode by reverse engineering of data files of known configuration.  The present version of `read.adv.nortek` determines the sampling mode from the ```NRecords`'' item of the &#8288;</code>Vector Velocity Data&rdquo; header, which seems
to be 0 for data collected continuously, and non-zero for data collected in
bursts.
</p>
<p>Taking these things together, we come upon the issue of how to infer
sampling times for Nortek instruments.  There do not seem to be definitive
documents on this, and so <code>read.adv.nortek</code> is based partly on
information (of unknown quality) found on Nortek discussion boards.  The
present version of <code>read.adv.nortek</code> infers the times of velocity
observations differently, depending on whether the instrument was set to
record in burst mode or continuous mode.  For burst mode, times stated in
the burst headers are used, but for continuous mode, times stated in the
&ldquo;vector system data&rdquo; are used.  On the advice found on a Nortek discussion
board, the burst-mode times are offset by 2 seconds to allow for the
instrument warm-up period.
</p>
<p><strong>Handling IMU (inertial measurement unit) data</strong>
</p>
<p>Starting in March
2016, <code>read.adv.nortek</code> has offered some support for handling IMU
(inertial measurement unit) data incorporated into Nortek binary files. This
is not described in the Nortek document named <code style="white-space: pre;">&#8288;System Integrator Guide'' (reference 1A) but it appeared in &#8288;</code>System Integrator Manual&rdquo; (reference 1B;
reference 1C). Confusingly, 1B described 3 varieties of data, whereas 1C does not
describe any of these, but describes instead a fourth variety. As of March
2016, <code>read.adv.nortek</code> handles all 4 varieties, because files in the
various schemes appear to exist. In <code>oce</code>, the varieties are named
after the byte code that flags them.  (Variety <code>c3</code> is the one
described in (reference 1C); the others were described in (reference 1B).)
The variety is stored in the <code>metadata</code> slot of the returned object as
a string named <code>IMUtype</code>.
</p>
<p>For each variety, the reader is cautioned that strong tests have not been
performed on the code.  One way to test the code is to compare with textual
data files produced by the Nortek software.  In March 2016, an <code>oce</code>
user shared a dataset of the <code>c3</code> variety, and this permitted detailed
comparison between the text file and the values inferred by
<code>read.adv.nortek</code>. The test suggested agreement (to within the
resolution printed in the text file) for velocity (<code>v</code> in the
<code>data</code> slot), signal amplitude (<code>a</code>), correlation (<code>q</code>),
pressure (<code>p</code>), the three components of IMU delta angle
(<code>IMUdeltaAngleX</code> etc), and all components of the rotation matrix
(<code>IMUrotation</code>).  However, the delta velocity signals did not match,
with <code>IMUdeltaVelocityX</code> disagreeing in the second decimal place,
<code>IMUdeltaVelocityY</code> component disagreeing in the first, and
<code>IMUdeltaVelocityZ</code> being out by a factor of about 10. This is github
issue 893 (<code style="white-space: pre;">&#8288;https://github.com/dankelley/oce/issues/893&#8288;</code>).
</p>

<ul>
<li><p> Variety <code>c3</code> (signalled by byte 5 of a sequence being
<code>0xc3</code>) provides information on what Nortek calls DeltaAngle,
DeltaVelocity and Orientation Matrix. (Apart from the orientation matrix,
Nortek provides no documentation on what these quantities mean.) In the
object returned by <code>read.adv.nortek</code>, these are stored in the
<code>data</code> slot as <code>IMUdeltaAngleX</code>, <code>IMUdeltaAngleY</code>,
<code>IMUdeltaAngleZ</code>, <code>IMUdeltaVelocityX</code>, <code>IMUdeltaVelocityY</code>,
<code>IMUdeltaVelocityZ</code>, and <code>IMUrotation</code>, all vectors except the
last, which is a 3D array. In addition to these, <code>IMUtimestamp</code> is a
time-stamp, which is not defined in the Nortek documents but seems, from IMU
documents (reference 5), to be defined based on a clock that ticks once per 16
microseconds. Caution may be required in dealing with this timestamp, since
it seemed sensible in one test case (variety <code>d3</code>) but kept reseting to
zero in another (variety <code>c3</code>). The lack of Nortek documentation on
most of these quantities is a roadblock to implementing <code>oce</code> functions
dealing with IMU-enabled datasets
</p>
</li>
<li><p> Variety <code>cc</code> (signalled by byte 5 of a sequence being
<code>0xcc</code>) provides information on acceleration, angular rotation rate,
magnetic vector and orientation matrix.  Each is a timeseries. Acceleration
is stored in the <code>data</code> slot as <code>IMUaccelX</code>, <code>IMUaccelY</code>,
<code>IMUaccelz</code>. The angular rotation components are <code>IMUngrtX</code>,
<code>IMUngrtY</code> and <code>IMUngrtz</code>. The magnetic data are in
<code>IMUmagrtx</code>, <code>IMUmagrty</code> and <code>IMUmagrtz</code>. Finally,
<code>IMUmatrix</code> is a rotation matrix made up from elements named
<code>M11</code>, <code>M12</code>, etc in the Nortek documentation.  In addition to all
of these, <code>IMUtime</code> stores time in seconds, with an origin whose
definition is not stated in reference 1B.
</p>
</li>
<li><p> Variety <code>d2</code> (signalled by byte 5 being <code>0xd2</code>) provides
information on gyro-stabilized acceleration, angular rate and magnetometer
vectors.  The data stored <code>MUaccelX</code>, <code>IMUangrtX</code>,
<code>IMUmagrtX</code>, with similar for <code>Y</code> and <code>Z</code>.  Again, time is in
<code>IMUtime</code>. This data type has not been tested as of mid-March 2016,
because the developers do not have a test file with which to test.
</p>
</li>
<li><p> Variety <code>d3</code> (signalled by byte 5 being <code>0xd3</code>) provides
information on DeltaAngle, DeltaVelocity and magnetometer vectors, stored in
<code>IMUdeltaAngleX</code>, <code>IMUdeltaVelocityX</code>, and
<code>IMUdeltaMagVectorX</code>, with similar for <code>Y</code> and <code>Z</code>. Again,
time is in <code>IMUtime</code>. This data type has not been tested as of
mid-March 2016, because the developers do not have a test file with which to
test.
</p>
</li></ul>



<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>1A. Nortek AS.  System Integrator Guide (paradopp family of products). June
2008.  (Doc No: PSI00-0101-0608).  (Users may find it helpful to also
examine newer versions of the guide.)
</p>
<p>1B. Nortek AS.  System Integrator Manual. Dec 2014.
(<code>system-integrator-manual_Dec2014_jan.pdf</code>)
</p>
<p>1C. Nortek AS.  System Integrator Manual. March 2016.
(<code>system-integrator-manual_Mar2016.pdf</code>)
</p>

<ol>
<li><p> SonTek/YSI ADVField/Hydra Acoustic Doppler Velocimeter (Field) Technical
Documentation (Sept 1, 2001).
</p>
</li>
<li><p> Appendix 2.2.3 of the Sontek ADV operation Manual, Firmware Version 4.0
(Oct 1997).
</p>
</li>
<li><p> Nortek Knowledge Center (http://www.nortekusa.com/en/knowledge-center)
</p>
</li>
<li><p> A document describing an IMU unit that seems to be close to the one named
in (references 1B and C) as being an adjunct to Nortek Vector systems is at
<code style="white-space: pre;">&#8288;http://files.microstrain.com/3DM-GX3-35-Data-Communications-Protocol.pdf&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(oce)
# A nortek Vector file
d &lt;- read.oce("/data/archive/sleiwex/2008/moorings/m05/adv/nortek_1943/raw/adv_nortek_1943.vec",
              from=as.POSIXct("2008-06-26 00:00:00", tz="UTC"),
              to=as.POSIXct("2008-06-26 00:00:10", tz="UTC"))
plot(d, which=c(1:3,15))

## End(Not run)
</code></pre>

<hr>
<h2 id='read.adv.nortek'>Read an adv File</h2><span id='topic+read.adv.nortek'></span>

<h3>Description</h3>

<p>Read an ADV data file, producing an object of type <code>adv</code>. This
function works by transferring control to a more specialized function,
e.g. <code><a href="#topic+read.adp.nortek">read.adp.nortek()</a></code> and <code><a href="#topic+read.adp.sontek">read.adp.sontek()</a></code>,
and in many cases users will find it preferable to either use these or
the several even more specialized functions, if the file type is
known.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.adv.nortek(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  header = TRUE,
  longitude = NA,
  latitude = NA,
  encoding = NA,
  type = c("vector", "aquadopp"),
  haveAnalog1 = FALSE,
  haveAnalog2 = FALSE,
  debug = getOption("oceDebug"),
  monitor = FALSE,
  processingLog = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.adv.nortek_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  It is also possible to give <code>file</code> as a vector of filenames,
to handle the case of data split into files by a data logger.  In the
multi-file case, <code>header</code> must be <code>FALSE</code>, <code>start</code> must be a
vector of times, and <code>deltat</code> must be provided.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_from">from</code></td>
<td>
<p>index number of the first profile to be read, or the time of
that profile, as created with <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code> (hint: use
<code>tz="UTC"</code>).  This argument is ignored if <code>header==FALSE</code>. See
&ldquo;Examples&rdquo;.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_to">to</code></td>
<td>
<p>indication of the last profile to read, in a format matching that
of <code>from</code>.  This is ignored if <code>header==FALSE</code>.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_by">by</code></td>
<td>
<p>an indication of the stride length to use while walking through
the file.  This is ignored if <code>header==FALSE</code>.  Otherwise, if this is
an integer, then <code>by-1</code> profiles are skipped between each pair of
profiles that is read. This may not make much sense, if the data are not
equi-spaced in time.  If <code>by</code> is a string representing a time interval,
in colon-separated format, then this interval is divided by the sampling
interval, to get the stride length. <em>BUG:</em> <code>by</code> only partially
works; see &ldquo;Bugs&rdquo;.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_header">header</code></td>
<td>
<p>A logical value indicating whether the file starts with a header.
(This will not be the case for files that are created by data loggers that
chop the raw data up into a series of sub-files, e.g. once per hour.)</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_type">type</code></td>
<td>
<p>A string indicating which type of Nortek device produced the data file, <code>vector</code>
or <code>aquadopp</code>.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_haveanalog1">haveAnalog1</code></td>
<td>
<p>A logical value indicating whether the data file has 'analog1' data.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_haveanalog2">haveAnalog2</code></td>
<td>
<p>A logical value indicating whether the data file has 'analog2' data.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  The value indicates the depth
within the call stack to which debugging applies.  For example,
<code>read.adv.nortek()</code> calls <code>read.header.nortek()</code>, so that
<code>read.adv.nortek(...,debug=2)</code> provides information about not just the
main body of the data file, but also the details of the header.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.adv.nortek_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This parameter is typically only provided for internal calls; the default
that it provides is better for normal calls by a user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Files <em>without</em> headers may be created in experiments in which a data
logger was set up to monitor the serial data stream from an instrument.  The
lack of header information places a burden on the user, who must supply such
basic information as the times of observations, the instrument orientation,
the instrument coordinate system, etc.  Example 3 below shows how to deal
with such files.  Three things should be noted.
</p>

<ol>
<li><p> The user must choose the appropriate <code>read.adv</code> variant
corresponding to the instrument in question.  (This is necessary because
<code><a href="#topic+oceMagic">oceMagic()</a></code>, which is used by the generic <code><a href="#topic+read.oce">read.oce()</a></code>
routine, cannot determine the type of instrument by examining a file that
lacks a header.)
</p>
</li>
<li><p> The call to the <code>read</code> function must include a start time
(<code>start</code>) and the number of seconds between data (<code>deltat</code>),
again, because the instrument data stream may lack those things when the
device is set to a serial mode.  Also, of course, it is necessary to set
<code>header=FALSE</code> in the function call.
</p>
</li>
<li><p> Once the file has been read in, the user will be obliged to specify
other information, for the object to be well-formed.  For example, the
<code>read</code> function will have no way of knowing the instrument orientation,
the coordinate system being used, the transformation matrix to go from
<code>"beam"</code> to <code>"xyz"</code> coordinates, or the instrument heading, pitch,
and roll, to go from <code>"xyz"</code> coordinates to <code>"enu"</code> coordinates.
Such things are illustrated in example 3 below.
</p>
</li></ol>

<p>In ADV data files, velocities are coded to signed 2-byte integers, with a
scale factor being used to convert to velocity in metres per second.  These
two facts control the maximum recordable velocity and the velocity
resolution, values that may be retrieved for an ADV object name <code>d</code>
with <code>d[["velocityMaximum"]]</code> and <code>d[["velocityResolution"]]</code>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+adv-class">adv</a> object that contains measurements made with an ADV device.
</p>
<p>The <code>metadata</code> contains information as given in the following table.
The <code style="white-space: pre;">&#8288;Nortek name'' is the name used in the Nortek System Integrator Guide (reference 1) and the &#8288;</code>Sontek name&rdquo; is the name used in the relevant
Sontek documentation.  References are given in square brackets.
</p>

<table>
<tr>
 <td style="text-align: left;">

<strong><code>metadata</code> name</strong></td><td style="text-align: left;"> <strong>Nortek name</strong> </td><td style="text-align: left;"> <strong>Sontek name</strong> </td><td style="text-align: left;"> <strong>Meaning</strong></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>manufacturer</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Either <code>"nortek"</code> or <code>"sontek"</code></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>instrumentType</code></td><td style="text-align: left;">  - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Either <code>"vector"</code> or <code>"adv"</code></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>filename</code></td><td style="text-align: left;">  - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Name of data file(s)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>latitude</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;">  Latitude of mooring (if applicable)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>longitude</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Longitude of mooring (if applicable)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfSamples</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Number of data samples in file</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfBeams</code></td><td style="text-align: left;"> NBeams (reference 1, p18) </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Number of beams (always 3)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfBeamSequencesPerBurst</code></td><td style="text-align: left;"> NPings </td><td style="text-align: left;"> - </td><td style="text-align: left;"> number of beam sequences per burst</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>measurementInterval</code></td><td style="text-align: left;"> MeasInterval (reference 1 p31) </td><td style="text-align: left;"> - </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">

<code>samplingRate</code></td><td style="text-align: left;"> 512/(AvgInterval) (reference 1 p30; reference 4) </td><td style="text-align: left;"> - #' </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">

</td>
</tr>

</table>

<p>The <code>data</code> list contains items with names corresponding to <code>adp</code>
objects, with an exception for Nortek data.  Nortek instruments report some
things at a time interval that is longer than the velocity sampling, and
these are stored in <code>data</code> as <code>timeSlow</code>, <code>headingSlow</code>,
<code>pitchSlow</code>, <code>rollSlow</code>, and <code>temperatureSlow</code>; if burst
sampling was used, there will also be items <code>recordsBurst</code> and
<code>timeBurst</code>.
</p>
<p>The <code>processingLog</code> is in the standard format.
</p>


<h3>Nortek files</h3>

<p><strong>Sampling-rate and similar issues</strong>
</p>
<p>The data format is inferred from the System Integrator Guide (reference 1A) and System
Integrator Manual (reference 1B).  These document lacks clarity in spots, and so
<code>read.adv.nortek</code> contains some assumptions that are noted here, so
that users will be aware of possible problems.
</p>
<p>A prominent example is the specification of the sampling rate, stored in
<code>metadata$sampingRate</code> in the return value.  Repeated examination of
the System Integrator Guide (reference 1) failed to indicate where this value is
stored in the various headers contained in Vector datasets.  After some
experimentation with a few data files, <code>read.adv.nortek</code> was set up to
calculate <code>metadata$samplingRate</code> as <code>512/AvgInterval</code> where
<code>AvgInterval</code> is a part of the <code style="white-space: pre;">&#8288;User Configuration'' header (reference 1 p30), where the explanation is &#8288;</code>average interval in seconds&rdquo;).  This formula was
developed through trial and error, but it was confirmed later on the Nortek
discussion group, and it should appear in upcoming versions of (reference 1).
</p>
<p>Another basic issue is the determination of whether an instrument had
recorded in continuous mode or burst mode.  One might infer that
<code>TimCtrlReg</code> in the <code style="white-space: pre;">&#8288;User Configuration'' header (reference 1 p30) determines this, in bits 1 and 2.  However, this was the case in test files available to the author.  For this reason, `read.adv.nortek` infers the mode by reverse engineering of data files of known configuration.  The present version of `read.adv.nortek` determines the sampling mode from the ```NRecords`'' item of the &#8288;</code>Vector Velocity Data&rdquo; header, which seems
to be 0 for data collected continuously, and non-zero for data collected in
bursts.
</p>
<p>Taking these things together, we come upon the issue of how to infer
sampling times for Nortek instruments.  There do not seem to be definitive
documents on this, and so <code>read.adv.nortek</code> is based partly on
information (of unknown quality) found on Nortek discussion boards.  The
present version of <code>read.adv.nortek</code> infers the times of velocity
observations differently, depending on whether the instrument was set to
record in burst mode or continuous mode.  For burst mode, times stated in
the burst headers are used, but for continuous mode, times stated in the
&ldquo;vector system data&rdquo; are used.  On the advice found on a Nortek discussion
board, the burst-mode times are offset by 2 seconds to allow for the
instrument warm-up period.
</p>
<p><strong>Handling IMU (inertial measurement unit) data</strong>
</p>
<p>Starting in March
2016, <code>read.adv.nortek</code> has offered some support for handling IMU
(inertial measurement unit) data incorporated into Nortek binary files. This
is not described in the Nortek document named <code style="white-space: pre;">&#8288;System Integrator Guide'' (reference 1A) but it appeared in &#8288;</code>System Integrator Manual&rdquo; (reference 1B;
reference 1C). Confusingly, 1B described 3 varieties of data, whereas 1C does not
describe any of these, but describes instead a fourth variety. As of March
2016, <code>read.adv.nortek</code> handles all 4 varieties, because files in the
various schemes appear to exist. In <code>oce</code>, the varieties are named
after the byte code that flags them.  (Variety <code>c3</code> is the one
described in (reference 1C); the others were described in (reference 1B).)
The variety is stored in the <code>metadata</code> slot of the returned object as
a string named <code>IMUtype</code>.
</p>
<p>For each variety, the reader is cautioned that strong tests have not been
performed on the code.  One way to test the code is to compare with textual
data files produced by the Nortek software.  In March 2016, an <code>oce</code>
user shared a dataset of the <code>c3</code> variety, and this permitted detailed
comparison between the text file and the values inferred by
<code>read.adv.nortek</code>. The test suggested agreement (to within the
resolution printed in the text file) for velocity (<code>v</code> in the
<code>data</code> slot), signal amplitude (<code>a</code>), correlation (<code>q</code>),
pressure (<code>p</code>), the three components of IMU delta angle
(<code>IMUdeltaAngleX</code> etc), and all components of the rotation matrix
(<code>IMUrotation</code>).  However, the delta velocity signals did not match,
with <code>IMUdeltaVelocityX</code> disagreeing in the second decimal place,
<code>IMUdeltaVelocityY</code> component disagreeing in the first, and
<code>IMUdeltaVelocityZ</code> being out by a factor of about 10. This is github
issue 893 (<code style="white-space: pre;">&#8288;https://github.com/dankelley/oce/issues/893&#8288;</code>).
</p>

<ul>
<li><p> Variety <code>c3</code> (signalled by byte 5 of a sequence being
<code>0xc3</code>) provides information on what Nortek calls DeltaAngle,
DeltaVelocity and Orientation Matrix. (Apart from the orientation matrix,
Nortek provides no documentation on what these quantities mean.) In the
object returned by <code>read.adv.nortek</code>, these are stored in the
<code>data</code> slot as <code>IMUdeltaAngleX</code>, <code>IMUdeltaAngleY</code>,
<code>IMUdeltaAngleZ</code>, <code>IMUdeltaVelocityX</code>, <code>IMUdeltaVelocityY</code>,
<code>IMUdeltaVelocityZ</code>, and <code>IMUrotation</code>, all vectors except the
last, which is a 3D array. In addition to these, <code>IMUtimestamp</code> is a
time-stamp, which is not defined in the Nortek documents but seems, from IMU
documents (reference 5), to be defined based on a clock that ticks once per 16
microseconds. Caution may be required in dealing with this timestamp, since
it seemed sensible in one test case (variety <code>d3</code>) but kept reseting to
zero in another (variety <code>c3</code>). The lack of Nortek documentation on
most of these quantities is a roadblock to implementing <code>oce</code> functions
dealing with IMU-enabled datasets
</p>
</li>
<li><p> Variety <code>cc</code> (signalled by byte 5 of a sequence being
<code>0xcc</code>) provides information on acceleration, angular rotation rate,
magnetic vector and orientation matrix.  Each is a timeseries. Acceleration
is stored in the <code>data</code> slot as <code>IMUaccelX</code>, <code>IMUaccelY</code>,
<code>IMUaccelz</code>. The angular rotation components are <code>IMUngrtX</code>,
<code>IMUngrtY</code> and <code>IMUngrtz</code>. The magnetic data are in
<code>IMUmagrtx</code>, <code>IMUmagrty</code> and <code>IMUmagrtz</code>. Finally,
<code>IMUmatrix</code> is a rotation matrix made up from elements named
<code>M11</code>, <code>M12</code>, etc in the Nortek documentation.  In addition to all
of these, <code>IMUtime</code> stores time in seconds, with an origin whose
definition is not stated in reference 1B.
</p>
</li>
<li><p> Variety <code>d2</code> (signalled by byte 5 being <code>0xd2</code>) provides
information on gyro-stabilized acceleration, angular rate and magnetometer
vectors.  The data stored <code>MUaccelX</code>, <code>IMUangrtX</code>,
<code>IMUmagrtX</code>, with similar for <code>Y</code> and <code>Z</code>.  Again, time is in
<code>IMUtime</code>. This data type has not been tested as of mid-March 2016,
because the developers do not have a test file with which to test.
</p>
</li>
<li><p> Variety <code>d3</code> (signalled by byte 5 being <code>0xd3</code>) provides
information on DeltaAngle, DeltaVelocity and magnetometer vectors, stored in
<code>IMUdeltaAngleX</code>, <code>IMUdeltaVelocityX</code>, and
<code>IMUdeltaMagVectorX</code>, with similar for <code>Y</code> and <code>Z</code>. Again,
time is in <code>IMUtime</code>. This data type has not been tested as of
mid-March 2016, because the developers do not have a test file with which to
test.
</p>
</li></ul>



<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>1A. Nortek AS.  System Integrator Guide (paradopp family of products). June
2008.  (Doc No: PSI00-0101-0608).  (Users may find it helpful to also
examine newer versions of the guide.)
</p>
<p>1B. Nortek AS.  System Integrator Manual. Dec 2014.
(<code>system-integrator-manual_Dec2014_jan.pdf</code>)
</p>
<p>1C. Nortek AS.  System Integrator Manual. March 2016.
(<code>system-integrator-manual_Mar2016.pdf</code>)
</p>

<ol>
<li><p> SonTek/YSI ADVField/Hydra Acoustic Doppler Velocimeter (Field) Technical
Documentation (Sept 1, 2001).
</p>
</li>
<li><p> Appendix 2.2.3 of the Sontek ADV operation Manual, Firmware Version 4.0
(Oct 1997).
</p>
</li>
<li><p> Nortek Knowledge Center (http://www.nortekusa.com/en/knowledge-center)
</p>
</li>
<li><p> A document describing an IMU unit that seems to be close to the one named
in (references 1B and C) as being an adjunct to Nortek Vector systems is at
<code style="white-space: pre;">&#8288;http://files.microstrain.com/3DM-GX3-35-Data-Communications-Protocol.pdf&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(oce)
# A nortek Vector file
d &lt;- read.oce("/data/archive/sleiwex/2008/moorings/m05/adv/nortek_1943/raw/adv_nortek_1943.vec",
              from=as.POSIXct("2008-06-26 00:00:00", tz="UTC"),
              to=as.POSIXct("2008-06-26 00:00:10", tz="UTC"))
plot(d, which=c(1:3,15))

## End(Not run)
</code></pre>

<hr>
<h2 id='read.adv.sontek.adr'>Read an adv File</h2><span id='topic+read.adv.sontek.adr'></span>

<h3>Description</h3>

<p>Read an ADV data file, producing an object of type <code>adv</code>. This
function works by transferring control to a more specialized function,
e.g. <code><a href="#topic+read.adp.nortek">read.adp.nortek()</a></code> and <code><a href="#topic+read.adp.sontek">read.adp.sontek()</a></code>,
and in many cases users will find it preferable to either use these or
the several even more specialized functions, if the file type is
known.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.adv.sontek.adr(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  header = TRUE,
  longitude = NA,
  latitude = NA,
  encoding = NA,
  debug = getOption("oceDebug"),
  monitor = FALSE,
  processingLog = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.adv.sontek.adr_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  It is also possible to give <code>file</code> as a vector of filenames,
to handle the case of data split into files by a data logger.  In the
multi-file case, <code>header</code> must be <code>FALSE</code>, <code>start</code> must be a
vector of times, and <code>deltat</code> must be provided.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.adr_+3A_from">from</code></td>
<td>
<p>index number of the first profile to be read, or the time of
that profile, as created with <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code> (hint: use
<code>tz="UTC"</code>).  This argument is ignored if <code>header==FALSE</code>. See
&ldquo;Examples&rdquo;.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.adr_+3A_to">to</code></td>
<td>
<p>indication of the last profile to read, in a format matching that
of <code>from</code>.  This is ignored if <code>header==FALSE</code>.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.adr_+3A_by">by</code></td>
<td>
<p>an indication of the stride length to use while walking through
the file.  This is ignored if <code>header==FALSE</code>.  Otherwise, if this is
an integer, then <code>by-1</code> profiles are skipped between each pair of
profiles that is read. This may not make much sense, if the data are not
equi-spaced in time.  If <code>by</code> is a string representing a time interval,
in colon-separated format, then this interval is divided by the sampling
interval, to get the stride length. <em>BUG:</em> <code>by</code> only partially
works; see &ldquo;Bugs&rdquo;.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.adr_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.adr_+3A_header">header</code></td>
<td>
<p>A logical value indicating whether the file starts with a header.
(This will not be the case for files that are created by data loggers that
chop the raw data up into a series of sub-files, e.g. once per hour.)</p>
</td></tr>
<tr><td><code id="read.adv.sontek.adr_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.adr_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.adr_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.adr_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  The value indicates the depth
within the call stack to which debugging applies.  For example,
<code>read.adv.nortek()</code> calls <code>read.header.nortek()</code>, so that
<code>read.adv.nortek(...,debug=2)</code> provides information about not just the
main body of the data file, but also the details of the header.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.adr_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.adr_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This parameter is typically only provided for internal calls; the default
that it provides is better for normal calls by a user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Files <em>without</em> headers may be created in experiments in which a data
logger was set up to monitor the serial data stream from an instrument.  The
lack of header information places a burden on the user, who must supply such
basic information as the times of observations, the instrument orientation,
the instrument coordinate system, etc.  Example 3 below shows how to deal
with such files.  Three things should be noted.
</p>

<ol>
<li><p> The user must choose the appropriate <code>read.adv</code> variant
corresponding to the instrument in question.  (This is necessary because
<code><a href="#topic+oceMagic">oceMagic()</a></code>, which is used by the generic <code><a href="#topic+read.oce">read.oce()</a></code>
routine, cannot determine the type of instrument by examining a file that
lacks a header.)
</p>
</li>
<li><p> The call to the <code>read</code> function must include a start time
(<code>start</code>) and the number of seconds between data (<code>deltat</code>),
again, because the instrument data stream may lack those things when the
device is set to a serial mode.  Also, of course, it is necessary to set
<code>header=FALSE</code> in the function call.
</p>
</li>
<li><p> Once the file has been read in, the user will be obliged to specify
other information, for the object to be well-formed.  For example, the
<code>read</code> function will have no way of knowing the instrument orientation,
the coordinate system being used, the transformation matrix to go from
<code>"beam"</code> to <code>"xyz"</code> coordinates, or the instrument heading, pitch,
and roll, to go from <code>"xyz"</code> coordinates to <code>"enu"</code> coordinates.
Such things are illustrated in example 3 below.
</p>
</li></ol>

<p>In ADV data files, velocities are coded to signed 2-byte integers, with a
scale factor being used to convert to velocity in metres per second.  These
two facts control the maximum recordable velocity and the velocity
resolution, values that may be retrieved for an ADV object name <code>d</code>
with <code>d[["velocityMaximum"]]</code> and <code>d[["velocityResolution"]]</code>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+adv-class">adv</a> object that contains measurements made with an ADV device.
</p>
<p>The <code>metadata</code> contains information as given in the following table.
The <code style="white-space: pre;">&#8288;Nortek name'' is the name used in the Nortek System Integrator Guide (reference 1) and the &#8288;</code>Sontek name&rdquo; is the name used in the relevant
Sontek documentation.  References are given in square brackets.
</p>

<table>
<tr>
 <td style="text-align: left;">

<strong><code>metadata</code> name</strong></td><td style="text-align: left;"> <strong>Nortek name</strong> </td><td style="text-align: left;"> <strong>Sontek name</strong> </td><td style="text-align: left;"> <strong>Meaning</strong></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>manufacturer</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Either <code>"nortek"</code> or <code>"sontek"</code></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>instrumentType</code></td><td style="text-align: left;">  - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Either <code>"vector"</code> or <code>"adv"</code></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>filename</code></td><td style="text-align: left;">  - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Name of data file(s)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>latitude</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;">  Latitude of mooring (if applicable)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>longitude</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Longitude of mooring (if applicable)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfSamples</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Number of data samples in file</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfBeams</code></td><td style="text-align: left;"> NBeams (reference 1, p18) </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Number of beams (always 3)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfBeamSequencesPerBurst</code></td><td style="text-align: left;"> NPings </td><td style="text-align: left;"> - </td><td style="text-align: left;"> number of beam sequences per burst</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>measurementInterval</code></td><td style="text-align: left;"> MeasInterval (reference 1 p31) </td><td style="text-align: left;"> - </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">

<code>samplingRate</code></td><td style="text-align: left;"> 512/(AvgInterval) (reference 1 p30; reference 4) </td><td style="text-align: left;"> - #' </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">

</td>
</tr>

</table>

<p>The <code>data</code> list contains items with names corresponding to <code>adp</code>
objects, with an exception for Nortek data.  Nortek instruments report some
things at a time interval that is longer than the velocity sampling, and
these are stored in <code>data</code> as <code>timeSlow</code>, <code>headingSlow</code>,
<code>pitchSlow</code>, <code>rollSlow</code>, and <code>temperatureSlow</code>; if burst
sampling was used, there will also be items <code>recordsBurst</code> and
<code>timeBurst</code>.
</p>
<p>The <code>processingLog</code> is in the standard format.
</p>


<h3>Nortek files</h3>

<p><strong>Sampling-rate and similar issues</strong>
</p>
<p>The data format is inferred from the System Integrator Guide (reference 1A) and System
Integrator Manual (reference 1B).  These document lacks clarity in spots, and so
<code>read.adv.nortek</code> contains some assumptions that are noted here, so
that users will be aware of possible problems.
</p>
<p>A prominent example is the specification of the sampling rate, stored in
<code>metadata$sampingRate</code> in the return value.  Repeated examination of
the System Integrator Guide (reference 1) failed to indicate where this value is
stored in the various headers contained in Vector datasets.  After some
experimentation with a few data files, <code>read.adv.nortek</code> was set up to
calculate <code>metadata$samplingRate</code> as <code>512/AvgInterval</code> where
<code>AvgInterval</code> is a part of the <code style="white-space: pre;">&#8288;User Configuration'' header (reference 1 p30), where the explanation is &#8288;</code>average interval in seconds&rdquo;).  This formula was
developed through trial and error, but it was confirmed later on the Nortek
discussion group, and it should appear in upcoming versions of (reference 1).
</p>
<p>Another basic issue is the determination of whether an instrument had
recorded in continuous mode or burst mode.  One might infer that
<code>TimCtrlReg</code> in the <code style="white-space: pre;">&#8288;User Configuration'' header (reference 1 p30) determines this, in bits 1 and 2.  However, this was the case in test files available to the author.  For this reason, `read.adv.nortek` infers the mode by reverse engineering of data files of known configuration.  The present version of `read.adv.nortek` determines the sampling mode from the ```NRecords`'' item of the &#8288;</code>Vector Velocity Data&rdquo; header, which seems
to be 0 for data collected continuously, and non-zero for data collected in
bursts.
</p>
<p>Taking these things together, we come upon the issue of how to infer
sampling times for Nortek instruments.  There do not seem to be definitive
documents on this, and so <code>read.adv.nortek</code> is based partly on
information (of unknown quality) found on Nortek discussion boards.  The
present version of <code>read.adv.nortek</code> infers the times of velocity
observations differently, depending on whether the instrument was set to
record in burst mode or continuous mode.  For burst mode, times stated in
the burst headers are used, but for continuous mode, times stated in the
&ldquo;vector system data&rdquo; are used.  On the advice found on a Nortek discussion
board, the burst-mode times are offset by 2 seconds to allow for the
instrument warm-up period.
</p>
<p><strong>Handling IMU (inertial measurement unit) data</strong>
</p>
<p>Starting in March
2016, <code>read.adv.nortek</code> has offered some support for handling IMU
(inertial measurement unit) data incorporated into Nortek binary files. This
is not described in the Nortek document named <code style="white-space: pre;">&#8288;System Integrator Guide'' (reference 1A) but it appeared in &#8288;</code>System Integrator Manual&rdquo; (reference 1B;
reference 1C). Confusingly, 1B described 3 varieties of data, whereas 1C does not
describe any of these, but describes instead a fourth variety. As of March
2016, <code>read.adv.nortek</code> handles all 4 varieties, because files in the
various schemes appear to exist. In <code>oce</code>, the varieties are named
after the byte code that flags them.  (Variety <code>c3</code> is the one
described in (reference 1C); the others were described in (reference 1B).)
The variety is stored in the <code>metadata</code> slot of the returned object as
a string named <code>IMUtype</code>.
</p>
<p>For each variety, the reader is cautioned that strong tests have not been
performed on the code.  One way to test the code is to compare with textual
data files produced by the Nortek software.  In March 2016, an <code>oce</code>
user shared a dataset of the <code>c3</code> variety, and this permitted detailed
comparison between the text file and the values inferred by
<code>read.adv.nortek</code>. The test suggested agreement (to within the
resolution printed in the text file) for velocity (<code>v</code> in the
<code>data</code> slot), signal amplitude (<code>a</code>), correlation (<code>q</code>),
pressure (<code>p</code>), the three components of IMU delta angle
(<code>IMUdeltaAngleX</code> etc), and all components of the rotation matrix
(<code>IMUrotation</code>).  However, the delta velocity signals did not match,
with <code>IMUdeltaVelocityX</code> disagreeing in the second decimal place,
<code>IMUdeltaVelocityY</code> component disagreeing in the first, and
<code>IMUdeltaVelocityZ</code> being out by a factor of about 10. This is github
issue 893 (<code style="white-space: pre;">&#8288;https://github.com/dankelley/oce/issues/893&#8288;</code>).
</p>

<ul>
<li><p> Variety <code>c3</code> (signalled by byte 5 of a sequence being
<code>0xc3</code>) provides information on what Nortek calls DeltaAngle,
DeltaVelocity and Orientation Matrix. (Apart from the orientation matrix,
Nortek provides no documentation on what these quantities mean.) In the
object returned by <code>read.adv.nortek</code>, these are stored in the
<code>data</code> slot as <code>IMUdeltaAngleX</code>, <code>IMUdeltaAngleY</code>,
<code>IMUdeltaAngleZ</code>, <code>IMUdeltaVelocityX</code>, <code>IMUdeltaVelocityY</code>,
<code>IMUdeltaVelocityZ</code>, and <code>IMUrotation</code>, all vectors except the
last, which is a 3D array. In addition to these, <code>IMUtimestamp</code> is a
time-stamp, which is not defined in the Nortek documents but seems, from IMU
documents (reference 5), to be defined based on a clock that ticks once per 16
microseconds. Caution may be required in dealing with this timestamp, since
it seemed sensible in one test case (variety <code>d3</code>) but kept reseting to
zero in another (variety <code>c3</code>). The lack of Nortek documentation on
most of these quantities is a roadblock to implementing <code>oce</code> functions
dealing with IMU-enabled datasets
</p>
</li>
<li><p> Variety <code>cc</code> (signalled by byte 5 of a sequence being
<code>0xcc</code>) provides information on acceleration, angular rotation rate,
magnetic vector and orientation matrix.  Each is a timeseries. Acceleration
is stored in the <code>data</code> slot as <code>IMUaccelX</code>, <code>IMUaccelY</code>,
<code>IMUaccelz</code>. The angular rotation components are <code>IMUngrtX</code>,
<code>IMUngrtY</code> and <code>IMUngrtz</code>. The magnetic data are in
<code>IMUmagrtx</code>, <code>IMUmagrty</code> and <code>IMUmagrtz</code>. Finally,
<code>IMUmatrix</code> is a rotation matrix made up from elements named
<code>M11</code>, <code>M12</code>, etc in the Nortek documentation.  In addition to all
of these, <code>IMUtime</code> stores time in seconds, with an origin whose
definition is not stated in reference 1B.
</p>
</li>
<li><p> Variety <code>d2</code> (signalled by byte 5 being <code>0xd2</code>) provides
information on gyro-stabilized acceleration, angular rate and magnetometer
vectors.  The data stored <code>MUaccelX</code>, <code>IMUangrtX</code>,
<code>IMUmagrtX</code>, with similar for <code>Y</code> and <code>Z</code>.  Again, time is in
<code>IMUtime</code>. This data type has not been tested as of mid-March 2016,
because the developers do not have a test file with which to test.
</p>
</li>
<li><p> Variety <code>d3</code> (signalled by byte 5 being <code>0xd3</code>) provides
information on DeltaAngle, DeltaVelocity and magnetometer vectors, stored in
<code>IMUdeltaAngleX</code>, <code>IMUdeltaVelocityX</code>, and
<code>IMUdeltaMagVectorX</code>, with similar for <code>Y</code> and <code>Z</code>. Again,
time is in <code>IMUtime</code>. This data type has not been tested as of
mid-March 2016, because the developers do not have a test file with which to
test.
</p>
</li></ul>



<h3>References</h3>


<ol>
<li><p> SonTek/YSI Incorporated. &quot;ADVField/Hydra Operation Manual,&quot; September 1, 2001.
</p>
</li>
<li><p> SonTek/YSI Incorporated. &quot;Argonaut Acoustic Doppler Current Meter Operation Manual Firmware Version 7.9.&quot;
SonTek/YSI, May 1, 2001.
https://eng.ucmerced.edu/snsjho/files/San_Joaquin/Sensors_and_Loggers/SonTek/SonTek_Argonaut/ArgonautXR.pdf.
</p>
</li></ol>



<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>1A. Nortek AS.  System Integrator Guide (paradopp family of products). June
2008.  (Doc No: PSI00-0101-0608).  (Users may find it helpful to also
examine newer versions of the guide.)
</p>
<p>1B. Nortek AS.  System Integrator Manual. Dec 2014.
(<code>system-integrator-manual_Dec2014_jan.pdf</code>)
</p>
<p>1C. Nortek AS.  System Integrator Manual. March 2016.
(<code>system-integrator-manual_Mar2016.pdf</code>)
</p>

<ol>
<li><p> SonTek/YSI ADVField/Hydra Acoustic Doppler Velocimeter (Field) Technical
Documentation (Sept 1, 2001).
</p>
</li>
<li><p> Appendix 2.2.3 of the Sontek ADV operation Manual, Firmware Version 4.0
(Oct 1997).
</p>
</li>
<li><p> Nortek Knowledge Center (http://www.nortekusa.com/en/knowledge-center)
</p>
</li>
<li><p> A document describing an IMU unit that seems to be close to the one named
in (references 1B and C) as being an adjunct to Nortek Vector systems is at
<code style="white-space: pre;">&#8288;http://files.microstrain.com/3DM-GX3-35-Data-Communications-Protocol.pdf&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(oce)
# A nortek Vector file
d &lt;- read.oce("/data/archive/sleiwex/2008/moorings/m05/adv/nortek_1943/raw/adv_nortek_1943.vec",
              from=as.POSIXct("2008-06-26 00:00:00", tz="UTC"),
              to=as.POSIXct("2008-06-26 00:00:10", tz="UTC"))
plot(d, which=c(1:3,15))

## End(Not run)
</code></pre>

<hr>
<h2 id='read.adv.sontek.serial'>Read an adv File</h2><span id='topic+read.adv.sontek.serial'></span>

<h3>Description</h3>

<p>Read an ADV data file, producing an object of type <code>adv</code>. This
function works by transferring control to a more specialized function,
e.g. <code><a href="#topic+read.adp.nortek">read.adp.nortek()</a></code> and <code><a href="#topic+read.adp.sontek">read.adp.sontek()</a></code>,
and in many cases users will find it preferable to either use these or
the several even more specialized functions, if the file type is
known.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.adv.sontek.serial(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  longitude = NA,
  latitude = NA,
  start = NULL,
  deltat = NULL,
  encoding = NA,
  monitor = FALSE,
  debug = getOption("oceDebug"),
  processingLog = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.adv.sontek.serial_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  It is also possible to give <code>file</code> as a vector of filenames,
to handle the case of data split into files by a data logger.  In the
multi-file case, <code>header</code> must be <code>FALSE</code>, <code>start</code> must be a
vector of times, and <code>deltat</code> must be provided.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_from">from</code></td>
<td>
<p>index number of the first profile to be read, or the time of
that profile, as created with <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code> (hint: use
<code>tz="UTC"</code>).  This argument is ignored if <code>header==FALSE</code>. See
&ldquo;Examples&rdquo;.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_to">to</code></td>
<td>
<p>indication of the last profile to read, in a format matching that
of <code>from</code>.  This is ignored if <code>header==FALSE</code>.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_by">by</code></td>
<td>
<p>an indication of the stride length to use while walking through
the file.  This is ignored if <code>header==FALSE</code>.  Otherwise, if this is
an integer, then <code>by-1</code> profiles are skipped between each pair of
profiles that is read. This may not make much sense, if the data are not
equi-spaced in time.  If <code>by</code> is a string representing a time interval,
in colon-separated format, then this interval is divided by the sampling
interval, to get the stride length. <em>BUG:</em> <code>by</code> only partially
works; see &ldquo;Bugs&rdquo;.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_start">start</code></td>
<td>
<p>the time of the first sample, typically created with
<code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>.  This may be a vector of times,
if <code>filename</code> is a vector of file names.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_deltat">deltat</code></td>
<td>
<p>the time between samples.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  The value indicates the depth
within the call stack to which debugging applies.  For example,
<code>read.adv.nortek()</code> calls <code>read.header.nortek()</code>, so that
<code>read.adv.nortek(...,debug=2)</code> provides information about not just the
main body of the data file, but also the details of the header.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.serial_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This parameter is typically only provided for internal calls; the default
that it provides is better for normal calls by a user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Files <em>without</em> headers may be created in experiments in which a data
logger was set up to monitor the serial data stream from an instrument.  The
lack of header information places a burden on the user, who must supply such
basic information as the times of observations, the instrument orientation,
the instrument coordinate system, etc.  Example 3 below shows how to deal
with such files.  Three things should be noted.
</p>

<ol>
<li><p> The user must choose the appropriate <code>read.adv</code> variant
corresponding to the instrument in question.  (This is necessary because
<code><a href="#topic+oceMagic">oceMagic()</a></code>, which is used by the generic <code><a href="#topic+read.oce">read.oce()</a></code>
routine, cannot determine the type of instrument by examining a file that
lacks a header.)
</p>
</li>
<li><p> The call to the <code>read</code> function must include a start time
(<code>start</code>) and the number of seconds between data (<code>deltat</code>),
again, because the instrument data stream may lack those things when the
device is set to a serial mode.  Also, of course, it is necessary to set
<code>header=FALSE</code> in the function call.
</p>
</li>
<li><p> Once the file has been read in, the user will be obliged to specify
other information, for the object to be well-formed.  For example, the
<code>read</code> function will have no way of knowing the instrument orientation,
the coordinate system being used, the transformation matrix to go from
<code>"beam"</code> to <code>"xyz"</code> coordinates, or the instrument heading, pitch,
and roll, to go from <code>"xyz"</code> coordinates to <code>"enu"</code> coordinates.
Such things are illustrated in example 3 below.
</p>
</li></ol>

<p>In ADV data files, velocities are coded to signed 2-byte integers, with a
scale factor being used to convert to velocity in metres per second.  These
two facts control the maximum recordable velocity and the velocity
resolution, values that may be retrieved for an ADV object name <code>d</code>
with <code>d[["velocityMaximum"]]</code> and <code>d[["velocityResolution"]]</code>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+adv-class">adv</a> object that contains measurements made with an ADV device.
</p>
<p>The <code>metadata</code> contains information as given in the following table.
The <code style="white-space: pre;">&#8288;Nortek name'' is the name used in the Nortek System Integrator Guide (reference 1) and the &#8288;</code>Sontek name&rdquo; is the name used in the relevant
Sontek documentation.  References are given in square brackets.
</p>

<table>
<tr>
 <td style="text-align: left;">

<strong><code>metadata</code> name</strong></td><td style="text-align: left;"> <strong>Nortek name</strong> </td><td style="text-align: left;"> <strong>Sontek name</strong> </td><td style="text-align: left;"> <strong>Meaning</strong></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>manufacturer</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Either <code>"nortek"</code> or <code>"sontek"</code></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>instrumentType</code></td><td style="text-align: left;">  - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Either <code>"vector"</code> or <code>"adv"</code></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>filename</code></td><td style="text-align: left;">  - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Name of data file(s)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>latitude</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;">  Latitude of mooring (if applicable)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>longitude</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Longitude of mooring (if applicable)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfSamples</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Number of data samples in file</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfBeams</code></td><td style="text-align: left;"> NBeams (reference 1, p18) </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Number of beams (always 3)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfBeamSequencesPerBurst</code></td><td style="text-align: left;"> NPings </td><td style="text-align: left;"> - </td><td style="text-align: left;"> number of beam sequences per burst</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>measurementInterval</code></td><td style="text-align: left;"> MeasInterval (reference 1 p31) </td><td style="text-align: left;"> - </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">

<code>samplingRate</code></td><td style="text-align: left;"> 512/(AvgInterval) (reference 1 p30; reference 4) </td><td style="text-align: left;"> - #' </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">

</td>
</tr>

</table>

<p>The <code>data</code> list contains items with names corresponding to <code>adp</code>
objects, with an exception for Nortek data.  Nortek instruments report some
things at a time interval that is longer than the velocity sampling, and
these are stored in <code>data</code> as <code>timeSlow</code>, <code>headingSlow</code>,
<code>pitchSlow</code>, <code>rollSlow</code>, and <code>temperatureSlow</code>; if burst
sampling was used, there will also be items <code>recordsBurst</code> and
<code>timeBurst</code>.
</p>
<p>The <code>processingLog</code> is in the standard format.
</p>


<h3>Nortek files</h3>

<p><strong>Sampling-rate and similar issues</strong>
</p>
<p>The data format is inferred from the System Integrator Guide (reference 1A) and System
Integrator Manual (reference 1B).  These document lacks clarity in spots, and so
<code>read.adv.nortek</code> contains some assumptions that are noted here, so
that users will be aware of possible problems.
</p>
<p>A prominent example is the specification of the sampling rate, stored in
<code>metadata$sampingRate</code> in the return value.  Repeated examination of
the System Integrator Guide (reference 1) failed to indicate where this value is
stored in the various headers contained in Vector datasets.  After some
experimentation with a few data files, <code>read.adv.nortek</code> was set up to
calculate <code>metadata$samplingRate</code> as <code>512/AvgInterval</code> where
<code>AvgInterval</code> is a part of the <code style="white-space: pre;">&#8288;User Configuration'' header (reference 1 p30), where the explanation is &#8288;</code>average interval in seconds&rdquo;).  This formula was
developed through trial and error, but it was confirmed later on the Nortek
discussion group, and it should appear in upcoming versions of (reference 1).
</p>
<p>Another basic issue is the determination of whether an instrument had
recorded in continuous mode or burst mode.  One might infer that
<code>TimCtrlReg</code> in the <code style="white-space: pre;">&#8288;User Configuration'' header (reference 1 p30) determines this, in bits 1 and 2.  However, this was the case in test files available to the author.  For this reason, `read.adv.nortek` infers the mode by reverse engineering of data files of known configuration.  The present version of `read.adv.nortek` determines the sampling mode from the ```NRecords`'' item of the &#8288;</code>Vector Velocity Data&rdquo; header, which seems
to be 0 for data collected continuously, and non-zero for data collected in
bursts.
</p>
<p>Taking these things together, we come upon the issue of how to infer
sampling times for Nortek instruments.  There do not seem to be definitive
documents on this, and so <code>read.adv.nortek</code> is based partly on
information (of unknown quality) found on Nortek discussion boards.  The
present version of <code>read.adv.nortek</code> infers the times of velocity
observations differently, depending on whether the instrument was set to
record in burst mode or continuous mode.  For burst mode, times stated in
the burst headers are used, but for continuous mode, times stated in the
&ldquo;vector system data&rdquo; are used.  On the advice found on a Nortek discussion
board, the burst-mode times are offset by 2 seconds to allow for the
instrument warm-up period.
</p>
<p><strong>Handling IMU (inertial measurement unit) data</strong>
</p>
<p>Starting in March
2016, <code>read.adv.nortek</code> has offered some support for handling IMU
(inertial measurement unit) data incorporated into Nortek binary files. This
is not described in the Nortek document named <code style="white-space: pre;">&#8288;System Integrator Guide'' (reference 1A) but it appeared in &#8288;</code>System Integrator Manual&rdquo; (reference 1B;
reference 1C). Confusingly, 1B described 3 varieties of data, whereas 1C does not
describe any of these, but describes instead a fourth variety. As of March
2016, <code>read.adv.nortek</code> handles all 4 varieties, because files in the
various schemes appear to exist. In <code>oce</code>, the varieties are named
after the byte code that flags them.  (Variety <code>c3</code> is the one
described in (reference 1C); the others were described in (reference 1B).)
The variety is stored in the <code>metadata</code> slot of the returned object as
a string named <code>IMUtype</code>.
</p>
<p>For each variety, the reader is cautioned that strong tests have not been
performed on the code.  One way to test the code is to compare with textual
data files produced by the Nortek software.  In March 2016, an <code>oce</code>
user shared a dataset of the <code>c3</code> variety, and this permitted detailed
comparison between the text file and the values inferred by
<code>read.adv.nortek</code>. The test suggested agreement (to within the
resolution printed in the text file) for velocity (<code>v</code> in the
<code>data</code> slot), signal amplitude (<code>a</code>), correlation (<code>q</code>),
pressure (<code>p</code>), the three components of IMU delta angle
(<code>IMUdeltaAngleX</code> etc), and all components of the rotation matrix
(<code>IMUrotation</code>).  However, the delta velocity signals did not match,
with <code>IMUdeltaVelocityX</code> disagreeing in the second decimal place,
<code>IMUdeltaVelocityY</code> component disagreeing in the first, and
<code>IMUdeltaVelocityZ</code> being out by a factor of about 10. This is github
issue 893 (<code style="white-space: pre;">&#8288;https://github.com/dankelley/oce/issues/893&#8288;</code>).
</p>

<ul>
<li><p> Variety <code>c3</code> (signalled by byte 5 of a sequence being
<code>0xc3</code>) provides information on what Nortek calls DeltaAngle,
DeltaVelocity and Orientation Matrix. (Apart from the orientation matrix,
Nortek provides no documentation on what these quantities mean.) In the
object returned by <code>read.adv.nortek</code>, these are stored in the
<code>data</code> slot as <code>IMUdeltaAngleX</code>, <code>IMUdeltaAngleY</code>,
<code>IMUdeltaAngleZ</code>, <code>IMUdeltaVelocityX</code>, <code>IMUdeltaVelocityY</code>,
<code>IMUdeltaVelocityZ</code>, and <code>IMUrotation</code>, all vectors except the
last, which is a 3D array. In addition to these, <code>IMUtimestamp</code> is a
time-stamp, which is not defined in the Nortek documents but seems, from IMU
documents (reference 5), to be defined based on a clock that ticks once per 16
microseconds. Caution may be required in dealing with this timestamp, since
it seemed sensible in one test case (variety <code>d3</code>) but kept reseting to
zero in another (variety <code>c3</code>). The lack of Nortek documentation on
most of these quantities is a roadblock to implementing <code>oce</code> functions
dealing with IMU-enabled datasets
</p>
</li>
<li><p> Variety <code>cc</code> (signalled by byte 5 of a sequence being
<code>0xcc</code>) provides information on acceleration, angular rotation rate,
magnetic vector and orientation matrix.  Each is a timeseries. Acceleration
is stored in the <code>data</code> slot as <code>IMUaccelX</code>, <code>IMUaccelY</code>,
<code>IMUaccelz</code>. The angular rotation components are <code>IMUngrtX</code>,
<code>IMUngrtY</code> and <code>IMUngrtz</code>. The magnetic data are in
<code>IMUmagrtx</code>, <code>IMUmagrty</code> and <code>IMUmagrtz</code>. Finally,
<code>IMUmatrix</code> is a rotation matrix made up from elements named
<code>M11</code>, <code>M12</code>, etc in the Nortek documentation.  In addition to all
of these, <code>IMUtime</code> stores time in seconds, with an origin whose
definition is not stated in reference 1B.
</p>
</li>
<li><p> Variety <code>d2</code> (signalled by byte 5 being <code>0xd2</code>) provides
information on gyro-stabilized acceleration, angular rate and magnetometer
vectors.  The data stored <code>MUaccelX</code>, <code>IMUangrtX</code>,
<code>IMUmagrtX</code>, with similar for <code>Y</code> and <code>Z</code>.  Again, time is in
<code>IMUtime</code>. This data type has not been tested as of mid-March 2016,
because the developers do not have a test file with which to test.
</p>
</li>
<li><p> Variety <code>d3</code> (signalled by byte 5 being <code>0xd3</code>) provides
information on DeltaAngle, DeltaVelocity and magnetometer vectors, stored in
<code>IMUdeltaAngleX</code>, <code>IMUdeltaVelocityX</code>, and
<code>IMUdeltaMagVectorX</code>, with similar for <code>Y</code> and <code>Z</code>. Again,
time is in <code>IMUtime</code>. This data type has not been tested as of
mid-March 2016, because the developers do not have a test file with which to
test.
</p>
</li></ul>



<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>1A. Nortek AS.  System Integrator Guide (paradopp family of products). June
2008.  (Doc No: PSI00-0101-0608).  (Users may find it helpful to also
examine newer versions of the guide.)
</p>
<p>1B. Nortek AS.  System Integrator Manual. Dec 2014.
(<code>system-integrator-manual_Dec2014_jan.pdf</code>)
</p>
<p>1C. Nortek AS.  System Integrator Manual. March 2016.
(<code>system-integrator-manual_Mar2016.pdf</code>)
</p>

<ol>
<li><p> SonTek/YSI ADVField/Hydra Acoustic Doppler Velocimeter (Field) Technical
Documentation (Sept 1, 2001).
</p>
</li>
<li><p> Appendix 2.2.3 of the Sontek ADV operation Manual, Firmware Version 4.0
(Oct 1997).
</p>
</li>
<li><p> Nortek Knowledge Center (http://www.nortekusa.com/en/knowledge-center)
</p>
</li>
<li><p> A document describing an IMU unit that seems to be close to the one named
in (references 1B and C) as being an adjunct to Nortek Vector systems is at
<code style="white-space: pre;">&#8288;http://files.microstrain.com/3DM-GX3-35-Data-Communications-Protocol.pdf&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(oce)
# A nortek Vector file
d &lt;- read.oce("/data/archive/sleiwex/2008/moorings/m05/adv/nortek_1943/raw/adv_nortek_1943.vec",
              from=as.POSIXct("2008-06-26 00:00:00", tz="UTC"),
              to=as.POSIXct("2008-06-26 00:00:10", tz="UTC"))
plot(d, which=c(1:3,15))

## End(Not run)
</code></pre>

<hr>
<h2 id='read.adv.sontek.text'>Read an adv File</h2><span id='topic+read.adv.sontek.text'></span>

<h3>Description</h3>

<p>Read an ADV data file, producing an object of type <code>adv</code>. This
function works by transferring control to a more specialized function,
e.g. <code><a href="#topic+read.adp.nortek">read.adp.nortek()</a></code> and <code><a href="#topic+read.adp.sontek">read.adp.sontek()</a></code>,
and in many cases users will find it preferable to either use these or
the several even more specialized functions, if the file type is
known.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.adv.sontek.text(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  originalCoordinate = "xyz",
  transformationMatrix,
  longitude = NA,
  latitude = NA,
  encoding = "latin1",
  monitor = FALSE,
  debug = getOption("oceDebug"),
  processingLog = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.adv.sontek.text_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  It is also possible to give <code>file</code> as a vector of filenames,
to handle the case of data split into files by a data logger.  In the
multi-file case, <code>header</code> must be <code>FALSE</code>, <code>start</code> must be a
vector of times, and <code>deltat</code> must be provided.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_from">from</code></td>
<td>
<p>index number of the first profile to be read, or the time of
that profile, as created with <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code> (hint: use
<code>tz="UTC"</code>).  This argument is ignored if <code>header==FALSE</code>. See
&ldquo;Examples&rdquo;.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_to">to</code></td>
<td>
<p>indication of the last profile to read, in a format matching that
of <code>from</code>.  This is ignored if <code>header==FALSE</code>.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_by">by</code></td>
<td>
<p>an indication of the stride length to use while walking through
the file.  This is ignored if <code>header==FALSE</code>.  Otherwise, if this is
an integer, then <code>by-1</code> profiles are skipped between each pair of
profiles that is read. This may not make much sense, if the data are not
equi-spaced in time.  If <code>by</code> is a string representing a time interval,
in colon-separated format, then this interval is divided by the sampling
interval, to get the stride length. <em>BUG:</em> <code>by</code> only partially
works; see &ldquo;Bugs&rdquo;.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_originalcoordinate">originalCoordinate</code></td>
<td>
<p>character string indicating coordinate system, one
of <code>"beam"</code>, <code>"xyz"</code>, <code>"enu"</code> or <code>"other"</code>.  (This is
needed for the case of multiple files that were created by a data logger,
because the header information is normally lost in such instances.)</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_transformationmatrix">transformationMatrix</code></td>
<td>
<p>transformation matrix to use in converting beam
coordinates to xyz coordinates.  This will over-ride the matrix in the file
header, if there is one.  An example is <code>rbind(c(2.710, -1.409,
-1.299), c(0.071, 2.372, -2.442), c(0.344, 0.344, 0.344))</code>.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  The value indicates the depth
within the call stack to which debugging applies.  For example,
<code>read.adv.nortek()</code> calls <code>read.header.nortek()</code>, so that
<code>read.adv.nortek(...,debug=2)</code> provides information about not just the
main body of the data file, but also the details of the header.</p>
</td></tr>
<tr><td><code id="read.adv.sontek.text_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This parameter is typically only provided for internal calls; the default
that it provides is better for normal calls by a user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Files <em>without</em> headers may be created in experiments in which a data
logger was set up to monitor the serial data stream from an instrument.  The
lack of header information places a burden on the user, who must supply such
basic information as the times of observations, the instrument orientation,
the instrument coordinate system, etc.  Example 3 below shows how to deal
with such files.  Three things should be noted.
</p>

<ol>
<li><p> The user must choose the appropriate <code>read.adv</code> variant
corresponding to the instrument in question.  (This is necessary because
<code><a href="#topic+oceMagic">oceMagic()</a></code>, which is used by the generic <code><a href="#topic+read.oce">read.oce()</a></code>
routine, cannot determine the type of instrument by examining a file that
lacks a header.)
</p>
</li>
<li><p> The call to the <code>read</code> function must include a start time
(<code>start</code>) and the number of seconds between data (<code>deltat</code>),
again, because the instrument data stream may lack those things when the
device is set to a serial mode.  Also, of course, it is necessary to set
<code>header=FALSE</code> in the function call.
</p>
</li>
<li><p> Once the file has been read in, the user will be obliged to specify
other information, for the object to be well-formed.  For example, the
<code>read</code> function will have no way of knowing the instrument orientation,
the coordinate system being used, the transformation matrix to go from
<code>"beam"</code> to <code>"xyz"</code> coordinates, or the instrument heading, pitch,
and roll, to go from <code>"xyz"</code> coordinates to <code>"enu"</code> coordinates.
Such things are illustrated in example 3 below.
</p>
</li></ol>

<p>In ADV data files, velocities are coded to signed 2-byte integers, with a
scale factor being used to convert to velocity in metres per second.  These
two facts control the maximum recordable velocity and the velocity
resolution, values that may be retrieved for an ADV object name <code>d</code>
with <code>d[["velocityMaximum"]]</code> and <code>d[["velocityResolution"]]</code>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+adv-class">adv</a> object that contains measurements made with an ADV device.
</p>
<p>The <code>metadata</code> contains information as given in the following table.
The <code style="white-space: pre;">&#8288;Nortek name'' is the name used in the Nortek System Integrator Guide (reference 1) and the &#8288;</code>Sontek name&rdquo; is the name used in the relevant
Sontek documentation.  References are given in square brackets.
</p>

<table>
<tr>
 <td style="text-align: left;">

<strong><code>metadata</code> name</strong></td><td style="text-align: left;"> <strong>Nortek name</strong> </td><td style="text-align: left;"> <strong>Sontek name</strong> </td><td style="text-align: left;"> <strong>Meaning</strong></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>manufacturer</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Either <code>"nortek"</code> or <code>"sontek"</code></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>instrumentType</code></td><td style="text-align: left;">  - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Either <code>"vector"</code> or <code>"adv"</code></td>
</tr>
<tr>
 <td style="text-align: left;">

<code>filename</code></td><td style="text-align: left;">  - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Name of data file(s)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>latitude</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;">  Latitude of mooring (if applicable)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>longitude</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Longitude of mooring (if applicable)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfSamples</code></td><td style="text-align: left;"> - </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Number of data samples in file</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfBeams</code></td><td style="text-align: left;"> NBeams (reference 1, p18) </td><td style="text-align: left;"> - </td><td style="text-align: left;"> Number of beams (always 3)</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>numberOfBeamSequencesPerBurst</code></td><td style="text-align: left;"> NPings </td><td style="text-align: left;"> - </td><td style="text-align: left;"> number of beam sequences per burst</td>
</tr>
<tr>
 <td style="text-align: left;">

<code>measurementInterval</code></td><td style="text-align: left;"> MeasInterval (reference 1 p31) </td><td style="text-align: left;"> - </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">

<code>samplingRate</code></td><td style="text-align: left;"> 512/(AvgInterval) (reference 1 p30; reference 4) </td><td style="text-align: left;"> - #' </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">

</td>
</tr>

</table>

<p>The <code>data</code> list contains items with names corresponding to <code>adp</code>
objects, with an exception for Nortek data.  Nortek instruments report some
things at a time interval that is longer than the velocity sampling, and
these are stored in <code>data</code> as <code>timeSlow</code>, <code>headingSlow</code>,
<code>pitchSlow</code>, <code>rollSlow</code>, and <code>temperatureSlow</code>; if burst
sampling was used, there will also be items <code>recordsBurst</code> and
<code>timeBurst</code>.
</p>
<p>The <code>processingLog</code> is in the standard format.
</p>


<h3>Nortek files</h3>

<p><strong>Sampling-rate and similar issues</strong>
</p>
<p>The data format is inferred from the System Integrator Guide (reference 1A) and System
Integrator Manual (reference 1B).  These document lacks clarity in spots, and so
<code>read.adv.nortek</code> contains some assumptions that are noted here, so
that users will be aware of possible problems.
</p>
<p>A prominent example is the specification of the sampling rate, stored in
<code>metadata$sampingRate</code> in the return value.  Repeated examination of
the System Integrator Guide (reference 1) failed to indicate where this value is
stored in the various headers contained in Vector datasets.  After some
experimentation with a few data files, <code>read.adv.nortek</code> was set up to
calculate <code>metadata$samplingRate</code> as <code>512/AvgInterval</code> where
<code>AvgInterval</code> is a part of the <code style="white-space: pre;">&#8288;User Configuration'' header (reference 1 p30), where the explanation is &#8288;</code>average interval in seconds&rdquo;).  This formula was
developed through trial and error, but it was confirmed later on the Nortek
discussion group, and it should appear in upcoming versions of (reference 1).
</p>
<p>Another basic issue is the determination of whether an instrument had
recorded in continuous mode or burst mode.  One might infer that
<code>TimCtrlReg</code> in the <code style="white-space: pre;">&#8288;User Configuration'' header (reference 1 p30) determines this, in bits 1 and 2.  However, this was the case in test files available to the author.  For this reason, `read.adv.nortek` infers the mode by reverse engineering of data files of known configuration.  The present version of `read.adv.nortek` determines the sampling mode from the ```NRecords`'' item of the &#8288;</code>Vector Velocity Data&rdquo; header, which seems
to be 0 for data collected continuously, and non-zero for data collected in
bursts.
</p>
<p>Taking these things together, we come upon the issue of how to infer
sampling times for Nortek instruments.  There do not seem to be definitive
documents on this, and so <code>read.adv.nortek</code> is based partly on
information (of unknown quality) found on Nortek discussion boards.  The
present version of <code>read.adv.nortek</code> infers the times of velocity
observations differently, depending on whether the instrument was set to
record in burst mode or continuous mode.  For burst mode, times stated in
the burst headers are used, but for continuous mode, times stated in the
&ldquo;vector system data&rdquo; are used.  On the advice found on a Nortek discussion
board, the burst-mode times are offset by 2 seconds to allow for the
instrument warm-up period.
</p>
<p><strong>Handling IMU (inertial measurement unit) data</strong>
</p>
<p>Starting in March
2016, <code>read.adv.nortek</code> has offered some support for handling IMU
(inertial measurement unit) data incorporated into Nortek binary files. This
is not described in the Nortek document named <code style="white-space: pre;">&#8288;System Integrator Guide'' (reference 1A) but it appeared in &#8288;</code>System Integrator Manual&rdquo; (reference 1B;
reference 1C). Confusingly, 1B described 3 varieties of data, whereas 1C does not
describe any of these, but describes instead a fourth variety. As of March
2016, <code>read.adv.nortek</code> handles all 4 varieties, because files in the
various schemes appear to exist. In <code>oce</code>, the varieties are named
after the byte code that flags them.  (Variety <code>c3</code> is the one
described in (reference 1C); the others were described in (reference 1B).)
The variety is stored in the <code>metadata</code> slot of the returned object as
a string named <code>IMUtype</code>.
</p>
<p>For each variety, the reader is cautioned that strong tests have not been
performed on the code.  One way to test the code is to compare with textual
data files produced by the Nortek software.  In March 2016, an <code>oce</code>
user shared a dataset of the <code>c3</code> variety, and this permitted detailed
comparison between the text file and the values inferred by
<code>read.adv.nortek</code>. The test suggested agreement (to within the
resolution printed in the text file) for velocity (<code>v</code> in the
<code>data</code> slot), signal amplitude (<code>a</code>), correlation (<code>q</code>),
pressure (<code>p</code>), the three components of IMU delta angle
(<code>IMUdeltaAngleX</code> etc), and all components of the rotation matrix
(<code>IMUrotation</code>).  However, the delta velocity signals did not match,
with <code>IMUdeltaVelocityX</code> disagreeing in the second decimal place,
<code>IMUdeltaVelocityY</code> component disagreeing in the first, and
<code>IMUdeltaVelocityZ</code> being out by a factor of about 10. This is github
issue 893 (<code style="white-space: pre;">&#8288;https://github.com/dankelley/oce/issues/893&#8288;</code>).
</p>

<ul>
<li><p> Variety <code>c3</code> (signalled by byte 5 of a sequence being
<code>0xc3</code>) provides information on what Nortek calls DeltaAngle,
DeltaVelocity and Orientation Matrix. (Apart from the orientation matrix,
Nortek provides no documentation on what these quantities mean.) In the
object returned by <code>read.adv.nortek</code>, these are stored in the
<code>data</code> slot as <code>IMUdeltaAngleX</code>, <code>IMUdeltaAngleY</code>,
<code>IMUdeltaAngleZ</code>, <code>IMUdeltaVelocityX</code>, <code>IMUdeltaVelocityY</code>,
<code>IMUdeltaVelocityZ</code>, and <code>IMUrotation</code>, all vectors except the
last, which is a 3D array. In addition to these, <code>IMUtimestamp</code> is a
time-stamp, which is not defined in the Nortek documents but seems, from IMU
documents (reference 5), to be defined based on a clock that ticks once per 16
microseconds. Caution may be required in dealing with this timestamp, since
it seemed sensible in one test case (variety <code>d3</code>) but kept reseting to
zero in another (variety <code>c3</code>). The lack of Nortek documentation on
most of these quantities is a roadblock to implementing <code>oce</code> functions
dealing with IMU-enabled datasets
</p>
</li>
<li><p> Variety <code>cc</code> (signalled by byte 5 of a sequence being
<code>0xcc</code>) provides information on acceleration, angular rotation rate,
magnetic vector and orientation matrix.  Each is a timeseries. Acceleration
is stored in the <code>data</code> slot as <code>IMUaccelX</code>, <code>IMUaccelY</code>,
<code>IMUaccelz</code>. The angular rotation components are <code>IMUngrtX</code>,
<code>IMUngrtY</code> and <code>IMUngrtz</code>. The magnetic data are in
<code>IMUmagrtx</code>, <code>IMUmagrty</code> and <code>IMUmagrtz</code>. Finally,
<code>IMUmatrix</code> is a rotation matrix made up from elements named
<code>M11</code>, <code>M12</code>, etc in the Nortek documentation.  In addition to all
of these, <code>IMUtime</code> stores time in seconds, with an origin whose
definition is not stated in reference 1B.
</p>
</li>
<li><p> Variety <code>d2</code> (signalled by byte 5 being <code>0xd2</code>) provides
information on gyro-stabilized acceleration, angular rate and magnetometer
vectors.  The data stored <code>MUaccelX</code>, <code>IMUangrtX</code>,
<code>IMUmagrtX</code>, with similar for <code>Y</code> and <code>Z</code>.  Again, time is in
<code>IMUtime</code>. This data type has not been tested as of mid-March 2016,
because the developers do not have a test file with which to test.
</p>
</li>
<li><p> Variety <code>d3</code> (signalled by byte 5 being <code>0xd3</code>) provides
information on DeltaAngle, DeltaVelocity and magnetometer vectors, stored in
<code>IMUdeltaAngleX</code>, <code>IMUdeltaVelocityX</code>, and
<code>IMUdeltaMagVectorX</code>, with similar for <code>Y</code> and <code>Z</code>. Again,
time is in <code>IMUtime</code>. This data type has not been tested as of
mid-March 2016, because the developers do not have a test file with which to
test.
</p>
</li></ul>



<h3>Note on file name</h3>

<p>The <code>file</code> argument does not actually name a file. It names a basename
for a file. The actual file names are created by appending suffix
<code>.hd1</code> for one file and <code>.ts1</code> for another.
</p>


<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>1A. Nortek AS.  System Integrator Guide (paradopp family of products). June
2008.  (Doc No: PSI00-0101-0608).  (Users may find it helpful to also
examine newer versions of the guide.)
</p>
<p>1B. Nortek AS.  System Integrator Manual. Dec 2014.
(<code>system-integrator-manual_Dec2014_jan.pdf</code>)
</p>
<p>1C. Nortek AS.  System Integrator Manual. March 2016.
(<code>system-integrator-manual_Mar2016.pdf</code>)
</p>

<ol>
<li><p> SonTek/YSI ADVField/Hydra Acoustic Doppler Velocimeter (Field) Technical
Documentation (Sept 1, 2001).
</p>
</li>
<li><p> Appendix 2.2.3 of the Sontek ADV operation Manual, Firmware Version 4.0
(Oct 1997).
</p>
</li>
<li><p> Nortek Knowledge Center (http://www.nortekusa.com/en/knowledge-center)
</p>
</li>
<li><p> A document describing an IMU unit that seems to be close to the one named
in (references 1B and C) as being an adjunct to Nortek Vector systems is at
<code style="white-space: pre;">&#8288;http://files.microstrain.com/3DM-GX3-35-Data-Communications-Protocol.pdf&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(oce)
# A nortek Vector file
d &lt;- read.oce("/data/archive/sleiwex/2008/moorings/m05/adv/nortek_1943/raw/adv_nortek_1943.vec",
              from=as.POSIXct("2008-06-26 00:00:00", tz="UTC"),
              to=as.POSIXct("2008-06-26 00:00:10", tz="UTC"))
plot(d, which=c(1:3,15))

## End(Not run)
</code></pre>

<hr>
<h2 id='read.amsr'>Read an amsr File</h2><span id='topic+read.amsr'></span>

<h3>Description</h3>

<p>Read an amsr file, generating an <a href="#topic+amsr-class">amsr</a> object.
Two file types are handled: type 1 is from gzipped files that were available
until perhaps the year 2022, and type 2 is from NetCDF files that
were available afterwards.
The type is stored in the <code>metadata</code> slot
as <code>type</code>, and this is detected in other functions relating to
<a href="#topic+amsr-class">amsr</a> data.  The best way to locate amsr files is to use
<code><a href="#topic+download.amsr">download.amsr()</a></code>, but if this fails, it may be necessary to search
the web for a source.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.amsr(file, encoding = NA, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.amsr_+3A_file">file</code></td>
<td>
<p>String indicating the name of a compressed file. See
&ldquo;File sources&rdquo;.</p>
</td></tr>
<tr><td><code id="read.amsr_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.amsr_+3A_debug">debug</code></td>
<td>
<p>A debugging flag, integer.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley and Chantelle Layton
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot+2Camsr-method">plot,amsr-method()</a></code> for an example.
</p>
<p>Other things related to amsr data: 
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method</a></code>,
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+summary+2Camsr-method">summary,amsr-method</a></code>
</p>

<hr>
<h2 id='read.aquadopp'>Read an adp File in Nortek Aquadopp Format</h2><span id='topic+read.aquadopp'></span>

<h3>Description</h3>

<p>The R code is based on information in the Nortek System Integrator Guide
(2017), postings on the Nortek &ldquo;knowledge center&rdquo; discussion board, and
discussions with Nortek engineers (Dec. 2020).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.aquadopp(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  longitude = NA,
  latitude = NA,
  type = "aquadopp",
  orientation,
  distance,
  monitor = FALSE,
  despike = FALSE,
  encoding = NA,
  processingLog,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.aquadopp_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  (For <code>read.adp.sontek.serial</code>, this is generally a list of
files, which will be concatenated.)</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_from">from</code></td>
<td>
<p>indication of the first profile to read.  This can be an
integer, the sequence number of the first profile to read, or a POSIXt time
before which profiles should be skipped, or a character string that converts
to a POSIXt time (assuming UTC timezone).  See &ldquo;Examples&rdquo;, and make
careful note of the use of the <code>tz</code> argument. If <code>from</code> is not
supplied, it defaults to 1.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_to">to</code></td>
<td>
<p>an optional indication of the last profile to read, in a
format as described for <code>from</code>.  As a special case, <code>to=0</code> means
to read the file to the end.  If <code>to</code> is not supplied, then it defaults
to 0.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_by">by</code></td>
<td>
<p>an optional indication of the stride length to use while walking through
the file.  If this is an integer, then <code>by-1</code> profiles are skipped
between each pair of profiles that is read, e.g. the default <code>by=1</code>
means to read all the data.  (For RDI files <em>only</em>, there are some
extra features to avoid running out of memory; see &ldquo;Memory considerations&rdquo;.)</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_type">type</code></td>
<td>
<p>Either &quot;aquadopp&quot; for a standard aquadopp file (the default), or
&quot;aquadoppPlusMagnetometer&quot; for a file which includes the raw magnetometer
data.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_orientation">orientation</code></td>
<td>
<p>Optional character string specifying the orientation of the
sensor, provided for those cases in which it cannot be inferred from the
data file.  The valid choices are <code>"upward"</code>, <code>"downward"</code>, and
<code>"sideward"</code>.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_distance">distance</code></td>
<td>
<p>Optional vector holding the distances of bin centres from the
sensor.  This argument is ignored except for Nortek profilers, and need not
be given if the function determines the distances correctly from the data.
The problem is that the distance is poorly documented in the Nortek System
Integrator Guide (2008 edition, page 31), so the function must rely on
word-of-mouth formulae that do not work in all cases.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_despike">despike</code></td>
<td>
<p>if <code>TRUE</code>, <code><a href="#topic+despike">despike()</a></code> will be used to clean
anomalous spikes in heading, etc.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="read.aquadopp_+3A_...">...</code></td>
<td>
<p>optional additional arguments that some (but not all)
<code style="white-space: pre;">&#8288;read.adp.*()&#8288;</code> functions pass to lower-level functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
The contents of that object make sense for the particular instrument
type under study, e.g. if the data file contains
NMEA strings, then navigational data will be stored in an item
called <code>nmea</code> in the <code>data</code> slot).
</p>


<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>


<ol>
<li><p> Information on Nortek profilers (including the System Integrator Guide,
which explains the data format byte-by-byte) is available at
<code style="white-space: pre;">&#8288;https://www.nortekusa.com/&#8288;</code>.  (One must join the site to see the
manuals.)
</p>
</li>
<li><p> The Nortek Knowledge Center
<code style="white-space: pre;">&#8288;https://www.nortekusa.com/en/knowledge-center&#8288;</code> may be of help if
problems arise in dealing with data from Nortek instruments.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that read adp data: 
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>
</p>

<hr>
<h2 id='read.aquadoppHR'>Read Nortek Aquadopp-HR File</h2><span id='topic+read.aquadoppHR'></span>

<h3>Description</h3>

<p>The R code is based on information in
the Nortek System Integrator Guide (2008) and on postings on the Nortek
&ldquo;knowledge center&rdquo; discussion board.  One might assume that the latter is
less authoritative than the former.  For example, the inference of cell size
follows advice found at
<code style="white-space: pre;">&#8288;https://www.nortekusa.com/en/knowledge-center/forum/hr-profilers/736804717&#8288;</code>,
which contains a typo in an early posting that is
corrected later on.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.aquadoppHR(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  longitude = NA,
  latitude = NA,
  orientation = orientation,
  distance,
  monitor = FALSE,
  despike = FALSE,
  encoding = NA,
  processingLog,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.aquadoppHR_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  (For <code>read.adp.sontek.serial</code>, this is generally a list of
files, which will be concatenated.)</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_from">from</code></td>
<td>
<p>indication of the first profile to read.  This can be an
integer, the sequence number of the first profile to read, or a POSIXt time
before which profiles should be skipped, or a character string that converts
to a POSIXt time (assuming UTC timezone).  See &ldquo;Examples&rdquo;, and make
careful note of the use of the <code>tz</code> argument. If <code>from</code> is not
supplied, it defaults to 1.</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_to">to</code></td>
<td>
<p>an optional indication of the last profile to read, in a
format as described for <code>from</code>.  As a special case, <code>to=0</code> means
to read the file to the end.  If <code>to</code> is not supplied, then it defaults
to 0.</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_by">by</code></td>
<td>
<p>an optional indication of the stride length to use while walking through
the file.  If this is an integer, then <code>by-1</code> profiles are skipped
between each pair of profiles that is read, e.g. the default <code>by=1</code>
means to read all the data.  (For RDI files <em>only</em>, there are some
extra features to avoid running out of memory; see &ldquo;Memory considerations&rdquo;.)</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_orientation">orientation</code></td>
<td>
<p>Optional character string specifying the orientation of the
sensor, provided for those cases in which it cannot be inferred from the
data file.  The valid choices are <code>"upward"</code>, <code>"downward"</code>, and
<code>"sideward"</code>.</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_distance">distance</code></td>
<td>
<p>Optional vector holding the distances of bin centres from the
sensor.  This argument is ignored except for Nortek profilers, and need not
be given if the function determines the distances correctly from the data.
The problem is that the distance is poorly documented in the Nortek System
Integrator Guide (2008 edition, page 31), so the function must rely on
word-of-mouth formulae that do not work in all cases.</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_despike">despike</code></td>
<td>
<p>if <code>TRUE</code>, <code><a href="#topic+despike">despike()</a></code> will be used to clean
anomalous spikes in heading, etc.</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="read.aquadoppHR_+3A_...">...</code></td>
<td>
<p>optional additional arguments that some (but not all)
<code style="white-space: pre;">&#8288;read.adp.*()&#8288;</code> functions pass to lower-level functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
The contents of that object make sense for the particular instrument
type under study, e.g. if the data file contains
NMEA strings, then navigational data will be stored in an item
called <code>nmea</code> in the <code>data</code> slot).
</p>


<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Information on Nortek profilers (including the System Integrator Guide,
which explains the data format byte-by-byte) is available at
<code style="white-space: pre;">&#8288;https://www.nortekusa.com/&#8288;</code>.  (One must join the site to see the
manuals.)
</p>
</li>
<li><p> The Nortek Knowledge Center
<code style="white-space: pre;">&#8288;https://www.nortekusa.com/en/knowledge-center&#8288;</code> may be of help if
problems arise in dealing with data from Nortek instruments.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that read adp data: 
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>
</p>

<hr>
<h2 id='read.aquadoppProfiler'>Read an adp File in Nortek Aquadopp Format</h2><span id='topic+read.aquadoppProfiler'></span>

<h3>Description</h3>

<p>The R code is based on information in
the Nortek System Integrator Guide (2008) and on postings on the Nortek
&ldquo;knowledge center&rdquo; discussion board.  One might assume that the latter is
less authoritative than the former.  For example, the inference of cell size
follows advice found at
<code style="white-space: pre;">&#8288;https://www.nortekusa.com/en/knowledge-center/forum/hr-profilers/736804717&#8288;</code>,
which contains a typo in an early posting that is
corrected later on.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.aquadoppProfiler(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  longitude = NA,
  latitude = NA,
  orientation,
  distance,
  monitor = FALSE,
  despike = FALSE,
  encoding = NA,
  processingLog,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.aquadoppProfiler_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  (For <code>read.adp.sontek.serial</code>, this is generally a list of
files, which will be concatenated.)</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_from">from</code></td>
<td>
<p>indication of the first profile to read.  This can be an
integer, the sequence number of the first profile to read, or a POSIXt time
before which profiles should be skipped, or a character string that converts
to a POSIXt time (assuming UTC timezone).  See &ldquo;Examples&rdquo;, and make
careful note of the use of the <code>tz</code> argument. If <code>from</code> is not
supplied, it defaults to 1.</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_to">to</code></td>
<td>
<p>an optional indication of the last profile to read, in a
format as described for <code>from</code>.  As a special case, <code>to=0</code> means
to read the file to the end.  If <code>to</code> is not supplied, then it defaults
to 0.</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_by">by</code></td>
<td>
<p>an optional indication of the stride length to use while walking through
the file.  If this is an integer, then <code>by-1</code> profiles are skipped
between each pair of profiles that is read, e.g. the default <code>by=1</code>
means to read all the data.  (For RDI files <em>only</em>, there are some
extra features to avoid running out of memory; see &ldquo;Memory considerations&rdquo;.)</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees
North.</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_orientation">orientation</code></td>
<td>
<p>Optional character string specifying the orientation of the
sensor, provided for those cases in which it cannot be inferred from the
data file.  The valid choices are <code>"upward"</code>, <code>"downward"</code>, and
<code>"sideward"</code>.</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_distance">distance</code></td>
<td>
<p>Optional vector holding the distances of bin centres from the
sensor.  This argument is ignored except for Nortek profilers, and need not
be given if the function determines the distances correctly from the data.
The problem is that the distance is poorly documented in the Nortek System
Integrator Guide (2008 edition, page 31), so the function must rely on
word-of-mouth formulae that do not work in all cases.</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_monitor">monitor</code></td>
<td>
<p>boolean value indicating whether to indicate the progress
of reading the file, by using <code><a href="utils.html#topic+txtProgressBar">txtProgressBar()</a></code> or otherwise.  The value
of <code>monitor</code> is changed to <code>FALSE</code> automatically, for non-interactive
sessions.</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_despike">despike</code></td>
<td>
<p>if <code>TRUE</code>, <code><a href="#topic+despike">despike()</a></code> will be used to clean
anomalous spikes in heading, etc.</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="read.aquadoppProfiler_+3A_...">...</code></td>
<td>
<p>optional additional arguments that some (but not all)
<code style="white-space: pre;">&#8288;read.adp.*()&#8288;</code> functions pass to lower-level functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
The contents of that object make sense for the particular instrument
type under study, e.g. if the data file contains
NMEA strings, then navigational data will be stored in an item
called <code>nmea</code> in the <code>data</code> slot).
</p>


<h3>How the binary file is decoded</h3>

<p>This file type, like other acoustic-Doppler types, is read
with a hybrid R/C++ system, for efficiency.
The processing steps are sketched below, for
users who want to inspect the code or build upon it.
</p>

<ol>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to insert the file contents into
a <a href="base.html#topic+vector">vector</a> of type <code>raw</code>.
</p>
</li>
<li><p> In C++, this raw vector is scanned byte by byte,
to find the starting indices of data &quot;chunks&quot;, or subsections of
the data that correspond to individual sampling times.
Checksum computations are also done at this stage, to detect
possible data corruption.  Warnings are issued for any bad chunks,
and they are skipped in further processing. The valid
starting points are then passed back to R as a <a href="base.html#topic+vector">vector</a> of
type <code>integer</code>.
</p>
</li>
<li><p> In R, <code><a href="base.html#topic+readBin">readBin()</a></code> is used to read the components of each chunk.
For speed, this is done in a vectorized fashion. For example,
all the velocities in the whole file are read in a single call
to <code><a href="base.html#topic+readBin">readBin()</a></code>. This process is done for each of the data
fields that are to be handled.  Importantly, these <code><a href="base.html#topic+readBin">readBin()</a></code>
calls are tailored to the data, using values of
the <code>size</code>, <code>endian</code> and <code>signed</code> parameters that are tailored
to the structure of the given component.
Scaling factors are then applied as required, to convert the
components to physical units.
</p>
</li>
<li><p> Finally, in R, the acquired items are inserted into the <code>data</code> or
<code>metadata</code> slot of the return value, according to oce convention.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Information on Nortek profilers (including the System Integrator Guide,
which explains the data format byte-by-byte) is available at
<code style="white-space: pre;">&#8288;https://www.nortekusa.com/&#8288;</code>.  (One must join the site to see the
manuals.)
</p>
</li>
<li><p> The Nortek Knowledge Center
<code style="white-space: pre;">&#8288;https://www.nortekusa.com/en/knowledge-center&#8288;</code> may be of help if
problems arise in dealing with data from Nortek instruments.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that read adp data: 
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>
</p>

<hr>
<h2 id='read.argo'>Read an Argo Data File</h2><span id='topic+read.argo'></span>

<h3>Description</h3>

<p><code>read.argo</code> is used to read an Argo file, producing an <a href="#topic+argo-class">argo</a> object.
The file must be in the ARGO-style NetCDF format described
in the Argo documentation (see references 2 and 3).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.argo(
  file,
  encoding = NA,
  debug = getOption("oceDebug"),
  processingLog,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.argo_+3A_file">file</code></td>
<td>
<p>A character string giving the name of the file to load.</p>
</td></tr>
<tr><td><code id="read.argo_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.argo_+3A_debug">debug</code></td>
<td>
<p>A flag that turns on debugging.  Set to 1 to get a moderate amount
of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="read.argo_+3A_processinglog">processingLog</code></td>
<td>
<p>If provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
<tr><td><code id="read.argo_+3A_...">...</code></td>
<td>
<p>additional arguments, passed to called routines.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See the Argo documentation (see references 2 and 3) for some details on what files contain.
Many items listed in section 2.2.3 of reference 3 are read from the
file and stored in the <code>metadata</code> slot, with the exception of
<code>longitude</code> and <code>latitude</code>, which are stored in the
<code>data</code> slot, alongside hydrographic information.  The details of storage
in the return value are somewhat complex, although the following notes
might be helpful to readers seeking to learn more.
</p>
<p><em>1. Variable renaming.</em>
</p>
<p>The names of several data parameters stored within the netCDF file
are altered to fit the oce context. For example, <code>PRES</code> becomes <code>pressure</code>,
matching the name of this variable in other oce data types.
The original names are reported by <code style="white-space: pre;">&#8288;summary,argo-method&#8288;</code>, and
data may be extracted with <code style="white-space: pre;">&#8288;[[,argo-method&#8288;</code> using those names, so
the renaming should not be too inconvenient to Argo experts who
are new to oce.
</p>
<p>Argo netcdf files employ a <code>"SNAKE_CASE"</code> naming scheme (sometimes
using lower case) that is inconsistent with the <code>"camelCase"</code> scheme
used in oce. Since argo objects are just a small part of oce, a decision
was made to rename argo items. For example, <code>"CYCLE_NUMBER"</code> in the netcdf file
becomes <code>"cycleNumber"</code> in the oce object returned by <code>read.argo</code>.
(Note that <code style="white-space: pre;">&#8288;[[,argo-method&#8288;</code> also accepts <code>"cycle"</code> for this item.)
The conversion for objects in the <code>data</code> slot often also involves
expanding on argo abbreviations, e.g. <code>"PSAL"</code> becomes <code>"salinity"</code>.
The renaming work is carried out with <code><a href="#topic+argoNames2oceNames">argoNames2oceNames()</a></code> for
handles both name expansion for several dozen special cases,
and with <code><a href="#topic+snakeToCamel">snakeToCamel()</a></code> with the <code>specialCase</code> argument
set to <code>"QC"</code>. While this results in variable names that should make
sense in the general oce context (where, for example, salinity is expected
to be stored in a variable named <code>"salinity"</code>), it may be confusing
to argo experts who are just starting to use oce.  Such people might
find it helpful to use e.g. <code>sort(names(x[["metadata"]]))</code> to get a list
of all items in the <code>metadata</code> slot (or similar with <code>"data"</code>), since working
in reverse may be easier than simply guessing at what names oce has chosen.
(Note that prior to 2020 June 24, some metadata items were stored in
<code>"SNAKE_CASE"</code>.)
</p>
<p><em>2. Metadata.</em>
</p>
<p>Several of the netCDF global attributes are also renamed before
placement in the <code>metadata</code> slot of the return value.  These include
<code>conventions</code>, <code>featureType</code>, <code>history</code>, <code>institution</code>,
<code>nParameters</code>, <code>nProfiles</code>,  <code>references</code>, <code>source</code>, <code>title</code>,
and <code>userManualVersion</code>.
These names are derived from those in the netcdf
file, and mainly follow the pattern explained in the
&ldquo;Variable renaming convention&rdquo; section.
</p>
<p>For profile data (as indicated by the NetCDF global attribute
named <code>"featureType"</code> being equal to <code>"trajectoryProfile"</code>),
the NetCDF item named <code>"STATION_PARAMETERS"</code> controls
whether variables in the source file will be stored in the
<code>metadata</code> or <code>data</code> slot of the returned object.
If <code>STATION_PARAMETERS</code> is not present, as is the case for
trajectory files (which are detected by <code>featureType</code> being
<code>"trajectory"</code>), some guesses are made as to what goes in
<code>data</code> and <code>metadata</code> slots.
</p>
<p><em>3. Data variants.</em>
</p>
<p>Each data item can have variants, as
described in Sections 2.3.4 of reference 3.
For example, if <code>"PRES"</code> is found in <code>STATION_PARAMETERS</code>,
then <code>PRES</code> (pressure) data are sought in the file, along with
<code>PRES_QC</code>, <code>PRES_ADJUSTED</code>, <code>PRES_ADJUSTED_QC</code>, and
<code>PRES_ERROR</code>. The same pattern works for other profile data. The variables
are stored with names created as explained in the
&ldquo;Variable renaming convention&rdquo; section below. Note that
flags, which are stored variables ending in <code>"_QC"</code> in the netcdf
file, are stored in the <code>flags</code> item within the <code>metadata</code> slot
of the returned object; thus, for example,
<code>PRES_QC</code> is stored as <code>pressure</code> in <code>flags</code>.
</p>
<p><em>4. How time is handled.</em>
</p>
<p>The netcdf files for profile data store time in an item named <code>juld</code>,
which holds the overall profile time, in what the Argo documentation
calls Julian days, with respect to a reference time that is also stored
in the file.  Based on this information, a <a href="base.html#topic+POSIXct">POSIXct</a> value named <code>time</code>
is stored in the <code>metadata</code> slot of the returned value, and this
may be found with e.g. <code>a[["time"]]</code>, where <code>a</code> is that returned value.
Importantly, this value matches the time listed in profile index files.
In addition, some profile data files contain a field called <code>MTIME</code>,
which holds the offset (in days) between the time of individual measurements and the
overall profile time. For such files, the measurement times may be
computed with <code>a[["time"]]+86400*a[["mtime"]]</code>. (This formula is used by
<code><a href="#topic+as.ctd">as.ctd()</a></code>, if its first argument is an <a href="#topic+argo-class">argo</a> object created
by supplying <code><a href="#topic+read.argo">read.argo()</a></code> with such a data file.)
</p>
<p><em>5. Data sources.</em>
</p>
<p>Argo data are made available at several websites. A bit of detective
work can be required to track down the data.
</p>
<p>Some servers provide data for floats that surfaced in a given ocean
on a given day, the anonymous FTP server
<code>usgodae.org/pub/outgoing/argo/geo/</code> being an example.
</p>
<p>Other servers provide data on a per-float basis. A complicating
factor is that these data tend to be categorized by &quot;dac&quot; (data
archiving centre), which makes it difficult to find a particular
float. For example,
<code>https://www.usgodae.org/ftp/outgoing/argo/</code> is the top level of
a such a repository. If the ID of a float is known but not the
&quot;dac&quot;, then a first step is to download the text file
<code>https://www.usgodae.org/ftp/outgoing/argo/ar_index_global_meta.txt</code>
and search for the ID. The first few lines of that file are header,
and after that the format is simple, with columns separated by slash
(<code>/</code>). The dac is in the first such column and the float ID in the
second. A simple search will reveal the dac.
For example <code>data(argo)</code> is based on float 6900388, and the line
containing that token is
<code style="white-space: pre;">&#8288;bodc/6900388/6900388_meta.nc,846,BO,20120225005617&#8288;</code>, from
which the dac is seen to be the British Oceanographic Data Centre
(<code>bodc</code>). Armed with that information, visit
<code>https://www.usgodae.org/ftp/outgoing/argo/dac/bodc/6900388</code>
and see a directory called <code>profiles</code> that contains a NetCDF
file for each profile the float made. These can be read with
<code>read.argo</code>. It is also possible, and probably more common,
to read a NetCDF file containing all the profiles together and for
that purpose the file
<code>https://www.usgodae.org/ftp/outgoing/argo/dac/bodc/6900388/6900388_prof.nc</code>
should be downloaded and provided as the <code>file</code> argument to
<code>read.argo</code>.  This can be automated as in Example 2,
although readers are cautioned that URL structures tend to change
over time.
</p>
<p>Similar steps can be followed on other servers.
</p>


<h3>Value</h3>

<p>An <a href="#topic+argo-class">argo</a> object.
</p>


<h3>Sample of Usage</h3>

<pre>
# Example 1: read from a local file
library(oce)
d &lt;- read.argo("~/data/OAR/6900388_prof.nc")
summary(d)
plot(d)

# Example 2: construct URL for download (brittle)
id &lt;- "6900388"
url &lt;- "https://www.usgodae.org/ftp/outgoing/argo"
if (!length(list.files(pattern="argo_index.txt")))
    download.file(paste(url, "ar_index_global_meta.txt", sep="/"), "argo_index.txt")
index &lt;- readLines("argo_index.txt")
line &lt;- grep(id, index)
if (0 == length(line))
    stop("id ", id, " not found")
if (1 &lt; length(line))
    stop("id ", id, " found multiple times")
dac &lt;- strsplit(index[line], "/")[[1]][1]
profile &lt;- paste(id, "_prof.nc", sep="")
float &lt;- paste(url, "dac", dac, id, profile, sep="/")
download.file(float, profile)
argo &lt;- read.argo(profile)
summary(argo)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li> <p><code style="white-space: pre;">&#8288;https://argo.ucsd.edu&#8288;</code>
</p>
</li>
<li><p> Argo User's Manual Version 3.2, Dec 29th, 2015, available at
<code style="white-space: pre;">&#8288;https://archimer.ifremer.fr/doc/00187/29825/&#8288;</code> online.
</p>
</li>
<li><p> User's Manual (ar-um-02-01) 13 July 2010, available at
<code style="white-space: pre;">&#8288;http://www.argodatamgt.org/content/download/4729/34634/file/argo-dm-user-manual-version-2.3.pdf&#8288;</code>,
which is the main document describing argo data.
</p>
</li></ol>



<h3>See Also</h3>

<p>The documentation for the <a href="#topic+argo-class">argo</a> class explains the structure of argo
objects, and also outlines the other functions dealing with them.
</p>
<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>

<hr>
<h2 id='read.argo.copernicus'>Read an argo File in Copernicus Format</h2><span id='topic+read.argo.copernicus'></span>

<h3>Description</h3>

<p>This function was added to read a particular file downloaded from the
Fleet Monitoring website (Reference 1).  The format was inferred
through examination of the file and a brief study of a document
(Reference 2) that describes the format.  Not all fields are read
by this function; see Reference 3 for a full list and note that
the author would be happy to add new entries (but not to spend hours
entering then all).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.argo.copernicus(
  file,
  encoding = NA,
  debug = getOption("oceDebug"),
  processingLog,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.argo.copernicus_+3A_file">file</code></td>
<td>
<p>A character string giving the name of the file to load.</p>
</td></tr>
<tr><td><code id="read.argo.copernicus_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.argo.copernicus_+3A_debug">debug</code></td>
<td>
<p>A flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or 0 (the default) for silent operation.</p>
</td></tr>
<tr><td><code id="read.argo.copernicus_+3A_processinglog">processingLog</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.argo.copernicus_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li> <p><code style="white-space: pre;">&#8288;https://fleetmonitoring.euro-argo.eu/float/4902489&#8288;</code>
</p>
</li>
<li><p> Copernicus Marine In Situ Tac Data Management Team. Copernicus Marine In
Situ NetCDF Format Manual (version V1.43). Pdf. Copernicus in situ TAC, 2021.
<code style="white-space: pre;">&#8288;https://doi.org/10.13155/59938&#8288;</code> (link checked 2022-04-11).
</p>
</li>
<li><p> Variable names are provided in files at
<code style="white-space: pre;">&#8288;https://doi.org/10.13155/53381&#8288;</code> (link checked 2022-04-12)
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>

<hr>
<h2 id='read.bremen'>Read a bremen File</h2><span id='topic+read.bremen'></span>

<h3>Description</h3>

<p>Read a file in Bremen format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.bremen(file, encoding = "latin1")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.bremen_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.</p>
</td></tr>
<tr><td><code id="read.bremen_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Velocities are assumed to be in
cm/s, and are converted to m/s to follow the oce convention. Shears
(which is what the variables named <code>uz</code> and <code>vz</code> are assumed
to represent) are assumed to be in (cm/s)/m, although they could be in 1/s
or something else; the lack of documentation is a problem here. Also,
note that the assumed shears are not just first-difference estimates
of velocity, given the results of a sample dataset:
</p>
<div class="sourceCode"><pre>&gt; head(data.frame(b[["data"]]))
  pressure      u      v       uz       vz
1        0  0.092 -0.191  0.00000  0.00000
2       10  0.092 -0.191  0.02183 -0.35412
3       20  0.092 -0.191  0.03046 -0.09458
4       30  0.026 -0.246 -0.03948  0.02169
5       40 -0.003 -0.212 -0.02614  0.03111
6       50 -0.023 -0.169 -0.03791  0.01706
</pre></div>


<h3>Value</h3>

<p>A <a href="#topic+bremen-class">bremen</a> object.
</p>


<h3>Issues</h3>

<p>This function may be renamed (or removed) without notice.
It was created to read some data being used in a particular research
project, and will be rendered useless if Bremen changes this data format.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to bremen data: 
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+summary+2Cbremen-method">summary,bremen-method</a></code>
</p>

<hr>
<h2 id='read.cm'>Read a cm File</h2><span id='topic+read.cm'></span>

<h3>Description</h3>

<p>Read a current-meter data file, producing a <a href="#topic+cm-class">cm</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.cm(
  file,
  from = 1,
  to,
  by = 1,
  tz = getOption("oceTz"),
  type = c("s4"),
  longitude = NA,
  latitude = NA,
  debug = getOption("oceDebug"),
  encoding = "latin1",
  monitor = FALSE,
  processingLog
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.cm_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file to
load.</p>
</td></tr>
<tr><td><code id="read.cm_+3A_from">from</code></td>
<td>
<p>index number of the first measurement to be read, or the time of
that measurement, as created with <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code> (hint: use
<code>tz="UTC"</code>).</p>
</td></tr>
<tr><td><code id="read.cm_+3A_to">to</code></td>
<td>
<p>indication of the last measurement to read, in a format matching that
of <code>from</code>.</p>
</td></tr>
<tr><td><code id="read.cm_+3A_by">by</code></td>
<td>
<p>an indication of the stride length to use while walking through the
file. If this is an integer, then <code>by-1</code> measurements are skipped between
each pair of profiles that is read. This may not make much sense, if the data
are not equi-spaced in time.  If <code>by</code> is a string representing a time
interval, in colon-separated format, then this interval is divided by the
sampling interval, to get the stride length. <em>BUG:</em> if the data are not
equi-spaced, then odd results will occur.</p>
</td></tr>
<tr><td><code id="read.cm_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.cm_+3A_type">type</code></td>
<td>
<p>character string indicating type of file (ignored at present).</p>
</td></tr>
<tr><td><code id="read.cm_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.cm_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees North.</p>
</td></tr>
<tr><td><code id="read.cm_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  The value indicates the depth
within the call stack to which debugging applies.</p>
</td></tr>
<tr><td><code id="read.cm_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.cm_+3A_monitor">monitor</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.cm_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.  This
parameter is typically only provided for internal calls; the default that it
provides is better for normal calls by a user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There function has been tested on only a single file, and the data-scanning
algorithm was based on visual inspection of that file.  Whether it will work
generally is an open question. It should be noted that the sample file had
several odd characteristics, some of which are listed below.
</p>

<ul>
<li><p> file contained two columns named <code>"Cond"</code>, which was guessed
to stand for conductivity. Since only the first contained data, the second was
ignored, but this may not be the case for all files.
</p>
</li>
<li><p> The unit for <code>"Cond"</code> was stated in the file to be <code>"mS"</code>,
which makes no sense, so the unit was assumed to be mS/cm.
</p>
</li>
<li><p> The file contained a column named <code>"T-Temp"</code>, which is not
something the author has seen in his career. It was assumed to stand for
in-situ temperature.
</p>
</li>
<li><p> The file contained a column named <code>"Depth"</code>, which is not something
an instrument can measure. Presumably it was calculated from pressure (with
what atmospheric offset, though?) and so pressure was inferred from it using
<code><a href="#topic+swPressure">swPressure()</a></code>.
</p>
</li>
<li><p> The file contained several columns that lacked names. These were ignored.
</p>
</li>
<li><p> The file contained several columns that seem to be derived from the
actual measured data, such as <code>"Speed"</code>, <code>"Dir"</code>, <code>"N-S Dist"</code>,
etc. These are ignored.
</p>
</li>
<li><p> The file contained several columns that were basically a mystery to the
author, e.g. <code>"Hx"</code>, <code>"Hy"</code>, <code>"Vref"</code>, etc. These were ignored.
</p>
</li></ul>

<p>Based on such considerations, <code><a href="#topic+read.cm">read.cm()</a></code> reads only the columns that
were reasonably well-understood based on the sample file. Users who need more
columns should contact the author. And a user who could produce a document
explaining the data format would be especially appreciated!
</p>


<h3>Value</h3>

<p>An <a href="#topic+cm-class">cm</a> object.
</p>
<p>The <code>data</code> slot will contain all the data in the file, with names
determined from the tokens in line 3 in that file, passed through
<code><a href="base.html#topic+make.names">make.names()</a></code>, except that
<code>Vnorth</code> is renamed <code>v</code> (after conversion from cm/s to m/s),
<code>Veast</code> is renamed <code>u</code> (after conversion from cm/s to m/s),
<code>Cond</code> is renamed <code>conductivity</code>,
<code>T.Temp</code> is renamed <code>temperature</code>
and
<code>Sal</code> is renamed <code>salinity</code>, and a new
column named <code>time</code> (a POSIX time) is constructed
from the information in the file header, and another named
<code>pressure</code> is constructed from the column named <code>Depth</code>.
At least in the single file studied in the creation of this function,
there are some columns that are unnamed in line 3 of the header;
these yield data items with names <code>X</code>, <code>X.1</code>, etc.
</p>


<h3>Historical note</h3>

<p>Prior to late July, 2016, the direction of current flow was stored in the
return value, but it is no longer stored, since it can be derived from the
<code>u</code> and <code>v</code> values.
</p>


<h3>Changes</h3>


<ul>
<li><p> On 2023-02-09 an item named <code>north</code> was added to the <code>metadata</code> slot.  This
is initialized to <code>"magnetic"</code> by <code><a href="#topic+read.cm">read.cm()</a></code>, but this is really just a
guess, and users ought to consider using <code><a href="#topic+applyMagneticDeclination">applyMagneticDeclination()</a></code> to take
magnetic declination into account.
</p>
</li></ul>



<h3>Sample of Usage</h3>

<pre>
library(oce)
cm &lt;- read.oce("cm_interocean_0811786.s4a.tab")
summary(cm)
plot(cm)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to cm data: 
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+as.cm">as.cm</a>()</code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+summary+2Ccm-method">summary,cm-method</a></code>
</p>

<hr>
<h2 id='read.coastline'>Read a coastline File</h2><span id='topic+read.coastline'></span>

<h3>Description</h3>

<p>Read a coastline file in R, Splus, mapgen, shapefile, or openstreetmap
format.
The S and R formats are identical, and consist of two columns, lon and lat,
with land-jump segments separated by lines with two NAs.
The MapGen format is of the form </p>
<pre> # -b -16.179081 28.553943
-16.244793 28.563330 </pre><p> BUG: the 'arc/info ungenerate' format is not yet
understood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.coastline(
  file,
  type = c("R", "S", "mapgen", "shapefile", "openstreetmap"),
  encoding = "latin1",
  monitor = FALSE,
  debug = getOption("oceDebug"),
  processingLog
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.coastline_+3A_file">file</code></td>
<td>
<p>name of file containing coastline data.</p>
</td></tr>
<tr><td><code id="read.coastline_+3A_type">type</code></td>
<td>
<p>type of file, one of <code>"R"</code>, <code>"S"</code>, <code>"mapgen"</code>,
<code>"shapefile"</code> or <code>"openstreetmap"</code>.</p>
</td></tr>
<tr><td><code id="read.coastline_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.coastline_+3A_monitor">monitor</code></td>
<td>
<p>print a dot for every coastline segment read (ignored except
for reading &quot;shapefile&quot; type)</p>
</td></tr>
<tr><td><code id="read.coastline_+3A_debug">debug</code></td>
<td>
<p>set to TRUE to print information about the header, etc.</p>
</td></tr>
<tr><td><code id="read.coastline_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="#topic+coastline-class">coastline</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>

<hr>
<h2 id='read.coastline.openstreetmap'>Read a coastline File in Openstreetmap Format</h2><span id='topic+read.coastline.openstreetmap'></span>

<h3>Description</h3>

<p>Read coastline data stored in the openstreetmap format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.coastline.openstreetmap(
  file,
  lonlim = c(-180, 180),
  latlim = c(-90, 90),
  monitor = FALSE,
  encoding = NA,
  debug = getOption("oceDebug"),
  processingLog
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.coastline.openstreetmap_+3A_file">file</code></td>
<td>
<p>name of file containing coastline data (a file ending in <code>.shp</code>)
or a zipfile that contains such a file, with a corresponding name.
The second scheme is useful for files downloaded from the NaturalEarth
website (see reference 2).</p>
</td></tr>
<tr><td><code id="read.coastline.openstreetmap_+3A_lonlim">lonlim</code>, <code id="read.coastline.openstreetmap_+3A_latlim">latlim</code></td>
<td>
<p>numerical vectors specifying the
west and east edges (and south and north edges) of a focus window.
Coastline polygons that do not intersect the defined box are
skipped, which can be useful in narrowing high-resolution world-scale
data to a local application.</p>
</td></tr>
<tr><td><code id="read.coastline.openstreetmap_+3A_monitor">monitor</code></td>
<td>
<p>Logical indicating whether to print an indication of progress through
the file.</p>
</td></tr>
<tr><td><code id="read.coastline.openstreetmap_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.coastline.openstreetmap_+3A_debug">debug</code></td>
<td>
<p>set to TRUE to print information about the header, etc.</p>
</td></tr>
<tr><td><code id="read.coastline.openstreetmap_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="#topic+coastline-class">coastline</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>

<hr>
<h2 id='read.coastline.shapefile'>Read a coastline File in Shapefile Format</h2><span id='topic+read.coastline.shapefile'></span>

<h3>Description</h3>

<p>Read coastline data stored in the shapefile format (see reference 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.coastline.shapefile(
  file,
  lonlim = c(-180, 180),
  latlim = c(-90, 90),
  encoding = NA,
  monitor = FALSE,
  debug = getOption("oceDebug"),
  processingLog
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.coastline.shapefile_+3A_file">file</code></td>
<td>
<p>name of file containing coastline data (a file ending in <code>.shp</code>)
or a zipfile that contains such a file, with a corresponding name.
The second scheme is useful for files downloaded from the NaturalEarth
website (see reference 2).</p>
</td></tr>
<tr><td><code id="read.coastline.shapefile_+3A_lonlim">lonlim</code>, <code id="read.coastline.shapefile_+3A_latlim">latlim</code></td>
<td>
<p>numerical vectors specifying the
west and east edges (and south and north edges) of a focus window.
Coastline polygons that do not intersect the defined box are
skipped, which can be useful in narrowing high-resolution world-scale
data to a local application.</p>
</td></tr>
<tr><td><code id="read.coastline.shapefile_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.coastline.shapefile_+3A_monitor">monitor</code></td>
<td>
<p>Logical indicating whether to print an indication of progress through
the file.</p>
</td></tr>
<tr><td><code id="read.coastline.shapefile_+3A_debug">debug</code></td>
<td>
<p>set to TRUE to print information about the header, etc.</p>
</td></tr>
<tr><td><code id="read.coastline.shapefile_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>x a <a href="#topic+coastline-class">coastline</a> object.
</p>


<h3>A hack for depth contours</h3>

<p>The following demonstrates that this
code is getting close to working with depth contours.  This should be
handled more internally, and a new object for depth contours should be
constructed, of which coastlines could be a subset.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> The &ldquo;shapefile&rdquo; format is described in
<em>ESRI Shapefile Technical Description</em>, March 1998, available at
<code style="white-space: pre;">&#8288;https://www.esri.com/content/dam/esrisites/sitecore-archive/Files/Pdfs/library/whitepapers/pdfs/shapefile.pdf&#8288;</code>
(last checked 2021-03-24).
</p>
</li>
<li><p> The NaturalEarth website <code style="white-space: pre;">&#8288;https://www.naturalearthdata.com/downloads/&#8288;</code>
provides coastline datasets in three resolutions, along with similar files
lakes and rivers, for borders, etc. It is highly recommended.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>

<hr>
<h2 id='read.ctd'>Read a ctd File in General Format</h2><span id='topic+read.ctd'></span>

<h3>Description</h3>

<p>Read a ctd File in General Format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.ctd(
  file,
  type = NULL,
  columns = NULL,
  station = NULL,
  missingValue,
  deploymentType = "unknown",
  monitor = FALSE,
  encoding = "latin1",
  debug = getOption("oceDebug"),
  processingLog,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.ctd_+3A_file">file</code></td>
<td>
<p>either a connection or a character value naming a file.
For <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.woce">read.ctd.woce()</a></code>, this may be a
wildcard (e.g. <code>"*.cnv"</code> or <code>"*.csv"</code>) in which case the return
value is a vector containing CTD objects created by reading the files from
<code><a href="base.html#topic+list.files">list.files()</a></code> with <code>pattern</code> set to the specified wildcard
pattern.</p>
</td></tr>
<tr><td><code id="read.ctd_+3A_type">type</code></td>
<td>
<p>If <code>NULL</code>, then the first line is studied, in order to
determine the file type, and control is dispatched to a specialized
function to handle that type.  Otherwise, <code>type</code> must be a string.
If <code>type="SBE19"</code> then a Seabird file format is assumed,
and control is dispatched to <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code>.
If <code>type="WOCE"</code> then a WOCE-exchange file is assumed,
and control is dispatched to <code><a href="#topic+read.ctd.woce">read.ctd.woce()</a></code>.
If <code>type="ITP"</code> then an ice-tethered profiler file is assumed,
and control is dispatched to <code><a href="#topic+read.ctd.itp">read.ctd.itp()</a></code>.
If <code>type="ODF"</code> then an ODF file (used by the Canadian Department of
Fisheries and Ocean) is assumed,
and control is dispatched to <code><a href="#topic+read.ctd.odf">read.ctd.odf()</a></code>.
Finally, if <code>type="ODV"</code> then an ODV file (used by Ocean Data View software) is assumed,
and control is dispatched to <code><a href="#topic+read.ctd.odv">read.ctd.odv()</a></code>.</p>
</td></tr>
<tr><td><code id="read.ctd_+3A_columns">columns</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that can be used to convert unrecognized
data names to resultant variable names.  This is used only by
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.odf">read.ctd.odf()</a></code>. For example,
if a data file named salinity as <code>"SAL"</code>, then using
</p>
<div class="sourceCode"><pre>d &lt;- read.ctd(f, columns=list(
    salinity=list(name="SAL",
                  unit=list(unit=expression(),
                  scale="PSS-78"))))
</pre></div>
<p>would assign the <code>"SAL"</code> column to the <code>salinity</code> entry in the data
slot of the CTD object returned by the <code style="white-space: pre;">&#8288;read.*&#8288;</code> function.</p>
</td></tr>
<tr><td><code id="read.ctd_+3A_station">station</code></td>
<td>
<p>optional character string containing an identifying name or
number for the station. This can be useful if the routine cannot determine the
name automatically, or if another name is preferred.</p>
</td></tr>
<tr><td><code id="read.ctd_+3A_missingvalue">missingValue</code></td>
<td>
<p>optional missing-value flag; data matching this value will
be set to <code>NA</code> upon reading. If this is provided, then it overrules any
missing-value flag found in the data. For Seabird (<code>.cnv</code>) files, there is
usually no need to set <code>missingValue</code>, because it can be inferred from the
header (typically as -9.990e-29). Set <code>missingValue=NULL</code> to turn off
missing-value detection, even in <code>.cnv</code> files that contain missing-value
codes in their headers. If <code>missingValue</code> is not specified,
then an attempt is made to infer such a value from the data, by testing
whether salinity and/or temperature has a minimum that is under -8 in value;
this should catch common values in files, without false positives. A warning
will be issued in this case, and a note inserted in the processing log of
the return value.</p>
</td></tr>
<tr><td><code id="read.ctd_+3A_deploymenttype">deploymentType</code></td>
<td>
<p>character string indicating the type of deployment. Use
<code>"unknown"</code> if this is not known, <code>"profile"</code> for a profile (in
which the data were acquired during a downcast, while the device was lowered
into the water column, perhaps also including an upcast; <code>"moored"</code> if
the device is installed on a fixed mooring, <code>"thermosalinograph"</code> (or
<code>"tsg"</code>) if the device is mounted on a moving vessel, to record
near-surface properties, or <code>"towyo"</code> if the device is repeatedly
lowered and raised.</p>
</td></tr>
<tr><td><code id="read.ctd_+3A_monitor">monitor</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to provide an indication of
progress.  This is useful if <code>filename</code> is a wildcard.</p>
</td></tr>
<tr><td><code id="read.ctd_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.ctd_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed.</p>
</td></tr>
<tr><td><code id="read.ctd_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This is typically only provided for internal calls; the default that it provides is
better for normal calls by a user.</p>
</td></tr>
<tr><td><code id="read.ctd_+3A_...">...</code></td>
<td>
<p>additional arguments, passed to called routines.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that read ctd data: 
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>
</p>

<hr>
<h2 id='read.ctd.aml'>Read a ctd File in AML Format</h2><span id='topic+read.ctd.aml'></span>

<h3>Description</h3>

<p><code><a href="#topic+read.ctd.aml">read.ctd.aml()</a></code> reads files that hold data acquired with an AML
Oceanographic BaseX2 CTD instrument. The SeaCast software associated with
this device can output data in several formats, of which only two are
handled, and only one is recommended (see &ldquo;Details&rdquo;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.ctd.aml(
  file,
  format,
  encoding = "UTF-8-BOM",
  debug = getOption("oceDebug"),
  processingLog,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.ctd.aml_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of
the file to load.</p>
</td></tr>
<tr><td><code id="read.ctd.aml_+3A_format">format</code></td>
<td>
<p>an integer indicating the format type.  If not supplied, the
first line is examined to determine whether the file matches the <code>format=1</code> or
<code>format=2</code> style (see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="read.ctd.aml_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.ctd.aml_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
<tr><td><code id="read.ctd.aml_+3A_processinglog">processingLog</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.ctd.aml_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The handled formats match files available to the author, both of which
diverge slightly from the format described in the AML documentation (see
&ldquo;References&rdquo;).
</p>
<p>Regardless of the format, files must contain columns named <code>Conductivity (mS/cm)</code>, <code>Temperature (C)</code> and <code>Pressure (dBar)</code>, because <a href="#topic+ctd-class">ctd</a>
objects need those quantities.  (Actually, if pressure is not found, but
<code>Depth (m)</code> is, then pressure is estimated with <code><a href="#topic+swDepth">swDepth()</a></code>, as a
workaround.) Note that other columns will be also read and stored in the
returned value, but they will not have proper units.  Attempts are made to
infer the sampling location from the file, by searching for strings like
<code style="white-space: pre;">&#8288;Latitude=&#8288;</code> in the header. Headers typically contain two values of the
location, and it is the second pair that is used by this function, with a
<code>NA</code> value being recorded if the value in the file is <code>no-lock</code>.  The
instrument serial number is also read, although the individual serial numbers
of the sensors are not read.  Position and serial number are stored in the
the <code>metadata</code> slot of the returned value.  The entire header is also stored
there, to let users glean more about dataset.
</p>
<p>Two formats are handled, as described below. Format 1 is greatly preferred,
because it is more robust (see below on <code>format=2</code>) and also because it can
be read later by the AML SeaCast software.
</p>

<ol>
<li><p> If <code>format</code> is <code>1</code> then the file is assumed to be in a format created by
selecting <em>Export As ... Seacast (.csv)</em> in AML's SeaCast software, with
settings to output pressure (or, as second-best, depth), temperature and
conductivity, and perhaps other things.  The delimiter must be comma.  If
date and time are output, their formats must be yyyy-mm-dd and UTC,
respectively.  Decoding the file proceeds as follows.  First, a check is done
to ensure that the first line consists of the string <code style="white-space: pre;">&#8288;[cast header]&#8288;</code>. Then an
attempt is made to infer location and serial number from the header.  After
this, <code><a href="#topic+read.ctd.aml">read.ctd.aml()</a></code> searches down for a line containing the string
<code style="white-space: pre;">&#8288;[data]&#8288;</code>. The first line thereafter is taken as a comma-separated list of
variable names, and lines following that are taken to hold the variable
values, separated by commas.
</p>
</li>
<li><p> If <code>format</code> is <code>2</code> then the first line must be a comma-separated list of
column names.  This may be followed by header information, which is handled
similarly as for <code>format=1</code>. The data are read from all lines that have the
same number of commas as the first line, an admittedly brittle strategy
developed as a way to handle some files that lacked other information about
the end of the header.
</p>
</li></ol>

<p>In both cases, the data columns, renamed to oce convention, are stored in the
<code>data</code> slot.  For the mandatory variables, units are also stored, as for
other <a href="#topic+ctd-class">ctd</a> objects.
</p>


<h3>Value</h3>

<p><code><a href="#topic+read.ctd.aml">read.ctd.aml()</a></code> returns a <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>AML Oceanographic. &quot;SeaCast 4 User Manual (Version 2.06).&quot; AML Oceanographic,
Mahy 2016.
<code style="white-space: pre;">&#8288;https://www.subseatechnologies.com/media/files/page/032e50ac/seacast-4-2-user-manual-sti.pdf&#8288;</code>.
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that read ctd data: 
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
f &lt;- system.file("extdata", "ctd_aml.csv.gz", package = "oce")
d &lt;- read.ctd.aml(f)
summary(d)

</code></pre>

<hr>
<h2 id='read.ctd.itp'>Read a ctd File in ITP Format</h2><span id='topic+read.ctd.itp'></span>

<h3>Description</h3>

<p>Read a ctd File in ITP Format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.ctd.itp(
  file,
  columns = NULL,
  station = NULL,
  missingValue,
  deploymentType = "unknown",
  encoding = "latin1",
  monitor = FALSE,
  debug = getOption("oceDebug"),
  processingLog,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.ctd.itp_+3A_file">file</code></td>
<td>
<p>either a connection or a character value naming a file.
For <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.woce">read.ctd.woce()</a></code>, this may be a
wildcard (e.g. <code>"*.cnv"</code> or <code>"*.csv"</code>) in which case the return
value is a vector containing CTD objects created by reading the files from
<code><a href="base.html#topic+list.files">list.files()</a></code> with <code>pattern</code> set to the specified wildcard
pattern.</p>
</td></tr>
<tr><td><code id="read.ctd.itp_+3A_columns">columns</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that can be used to convert unrecognized
data names to resultant variable names.  This is used only by
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.odf">read.ctd.odf()</a></code>. For example,
if a data file named salinity as <code>"SAL"</code>, then using
</p>
<div class="sourceCode"><pre>d &lt;- read.ctd(f, columns=list(
    salinity=list(name="SAL",
                  unit=list(unit=expression(),
                  scale="PSS-78"))))
</pre></div>
<p>would assign the <code>"SAL"</code> column to the <code>salinity</code> entry in the data
slot of the CTD object returned by the <code style="white-space: pre;">&#8288;read.*&#8288;</code> function.</p>
</td></tr>
<tr><td><code id="read.ctd.itp_+3A_station">station</code></td>
<td>
<p>optional character string containing an identifying name or
number for the station. This can be useful if the routine cannot determine the
name automatically, or if another name is preferred.</p>
</td></tr>
<tr><td><code id="read.ctd.itp_+3A_missingvalue">missingValue</code></td>
<td>
<p>optional missing-value flag; data matching this value will
be set to <code>NA</code> upon reading. If this is provided, then it overrules any
missing-value flag found in the data. For Seabird (<code>.cnv</code>) files, there is
usually no need to set <code>missingValue</code>, because it can be inferred from the
header (typically as -9.990e-29). Set <code>missingValue=NULL</code> to turn off
missing-value detection, even in <code>.cnv</code> files that contain missing-value
codes in their headers. If <code>missingValue</code> is not specified,
then an attempt is made to infer such a value from the data, by testing
whether salinity and/or temperature has a minimum that is under -8 in value;
this should catch common values in files, without false positives. A warning
will be issued in this case, and a note inserted in the processing log of
the return value.</p>
</td></tr>
<tr><td><code id="read.ctd.itp_+3A_deploymenttype">deploymentType</code></td>
<td>
<p>character string indicating the type of deployment. Use
<code>"unknown"</code> if this is not known, <code>"profile"</code> for a profile (in
which the data were acquired during a downcast, while the device was lowered
into the water column, perhaps also including an upcast; <code>"moored"</code> if
the device is installed on a fixed mooring, <code>"thermosalinograph"</code> (or
<code>"tsg"</code>) if the device is mounted on a moving vessel, to record
near-surface properties, or <code>"towyo"</code> if the device is repeatedly
lowered and raised.</p>
</td></tr>
<tr><td><code id="read.ctd.itp_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.ctd.itp_+3A_monitor">monitor</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to provide an indication of
progress.  This is useful if <code>filename</code> is a wildcard.</p>
</td></tr>
<tr><td><code id="read.ctd.itp_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed.</p>
</td></tr>
<tr><td><code id="read.ctd.itp_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This is typically only provided for internal calls; the default that it provides is
better for normal calls by a user.</p>
</td></tr>
<tr><td><code id="read.ctd.itp_+3A_...">...</code></td>
<td>
<p>additional arguments, passed to called routines.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>
<p><code><a href="#topic+read.ctd.itp">read.ctd.itp()</a></code> reads ice-tethered-profiler data that are stored
in a format files used by WHOI servers as of 2016-2017. Lacking
documentation on the format, the author constructed this function
to work with some files that were on-hand. Whether the function will
prove robust is an open question.
</p>
<p>Dan Kelley
</p>


<h3>References</h3>

<p>Information about ice-tethered profile data is provided at
<code>https://www.whoi.edu/page.do?pid=23096</code>, which also provides a link for
downloading data.  Note that the present version only handles data in
profiler-mode, not fixed-depth mode.
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that read ctd data: 
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>
</p>

<hr>
<h2 id='read.ctd.odf'>Read a ctd File in odf Format</h2><span id='topic+read.ctd.odf'></span>

<h3>Description</h3>

<p>Read a ctd File in odf Format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.ctd.odf(
  file,
  columns = NULL,
  station = NULL,
  missingValue,
  deploymentType = "unknown",
  monitor = FALSE,
  exclude = NULL,
  encoding = "latin1",
  debug = getOption("oceDebug"),
  processingLog,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.ctd.odf_+3A_file">file</code></td>
<td>
<p>either a connection or a character value naming a file.
For <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.woce">read.ctd.woce()</a></code>, this may be a
wildcard (e.g. <code>"*.cnv"</code> or <code>"*.csv"</code>) in which case the return
value is a vector containing CTD objects created by reading the files from
<code><a href="base.html#topic+list.files">list.files()</a></code> with <code>pattern</code> set to the specified wildcard
pattern.</p>
</td></tr>
<tr><td><code id="read.ctd.odf_+3A_columns">columns</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that can be used to convert unrecognized
data names to resultant variable names.  This is used only by
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.odf">read.ctd.odf()</a></code>. For example,
if a data file named salinity as <code>"SAL"</code>, then using
</p>
<div class="sourceCode"><pre>d &lt;- read.ctd(f, columns=list(
    salinity=list(name="SAL",
                  unit=list(unit=expression(),
                  scale="PSS-78"))))
</pre></div>
<p>would assign the <code>"SAL"</code> column to the <code>salinity</code> entry in the data
slot of the CTD object returned by the <code style="white-space: pre;">&#8288;read.*&#8288;</code> function.</p>
</td></tr>
<tr><td><code id="read.ctd.odf_+3A_station">station</code></td>
<td>
<p>optional character string containing an identifying name or
number for the station. This can be useful if the routine cannot determine the
name automatically, or if another name is preferred.</p>
</td></tr>
<tr><td><code id="read.ctd.odf_+3A_missingvalue">missingValue</code></td>
<td>
<p>optional missing-value flag; data matching this value will
be set to <code>NA</code> upon reading. If this is provided, then it overrules any
missing-value flag found in the data. For Seabird (<code>.cnv</code>) files, there is
usually no need to set <code>missingValue</code>, because it can be inferred from the
header (typically as -9.990e-29). Set <code>missingValue=NULL</code> to turn off
missing-value detection, even in <code>.cnv</code> files that contain missing-value
codes in their headers. If <code>missingValue</code> is not specified,
then an attempt is made to infer such a value from the data, by testing
whether salinity and/or temperature has a minimum that is under -8 in value;
this should catch common values in files, without false positives. A warning
will be issued in this case, and a note inserted in the processing log of
the return value.</p>
</td></tr>
<tr><td><code id="read.ctd.odf_+3A_deploymenttype">deploymentType</code></td>
<td>
<p>character string indicating the type of deployment. Use
<code>"unknown"</code> if this is not known, <code>"profile"</code> for a profile (in
which the data were acquired during a downcast, while the device was lowered
into the water column, perhaps also including an upcast; <code>"moored"</code> if
the device is installed on a fixed mooring, <code>"thermosalinograph"</code> (or
<code>"tsg"</code>) if the device is mounted on a moving vessel, to record
near-surface properties, or <code>"towyo"</code> if the device is repeatedly
lowered and raised.</p>
</td></tr>
<tr><td><code id="read.ctd.odf_+3A_monitor">monitor</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to provide an indication of
progress.  This is useful if <code>filename</code> is a wildcard.</p>
</td></tr>
<tr><td><code id="read.ctd.odf_+3A_exclude">exclude</code></td>
<td>
<p>either a character value holding a regular
expression that is used with <code><a href="base.html#topic+grep">grep()</a></code> to remove lines from the
header before processing, or <code>NULL</code> (the default), meaning
not to exclude any such lines.  The purpose of this argument
is to solve problems with some files, which can have
thousands of lines that indicate details that are may be of
little value in processing.  For example, some files have thousands
of lines that would be excluded by using
<code>exclude="PROCESS='Nulled the .* value"</code> in the function call.</p>
</td></tr>
<tr><td><code id="read.ctd.odf_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.ctd.odf_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed.</p>
</td></tr>
<tr><td><code id="read.ctd.odf_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This is typically only provided for internal calls; the default that it provides is
better for normal calls by a user.</p>
</td></tr>
<tr><td><code id="read.ctd.odf_+3A_...">...</code></td>
<td>
<p>additional arguments, passed to called routines.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>read.ctd.odf</code> reads files stored in Ocean Data Format, used in
some Canadian hydrographic databases.
</p>


<h3>Value</h3>

<p>This function returns a <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Caution</h3>

<p>Lacking detailed documentation of the ODF file format, the <code><a href="#topic+read.odf">read.odf()</a></code> and
<code><a href="#topic+read.ctd.odf">read.ctd.odf()</a></code> functions were crafted based on inspection of data files, and
so some guesses had to be made.
</p>
<p>The <code>PARAMETER_HEADER</code> chunks describing quality-control flags are
a case in point.  These contain <code>NAME</code> components that
refer to other <code>PARAMETER_HEADER</code> chunks that hold measured data.  However,
those references are not always matched well with the data names, and
even if they do match, the cross-reference syntax used by
the Bedford Institute of Oceanography differs from that used by
l’Institut Maurice-Lamontagne. To simplify coding, it was assumed that
each quality-control sequence applies to the data sequence
immediately preceding it.  (This assumption is made in other
analysis systems.)
</p>
<p>It is also prudent to pay attention to the units decoding,
which <code><a href="#topic+read.odf">read.odf()</a></code> handles by calling <code><a href="#topic+unitFromString">unitFromString()</a></code>.
Be on the lookout for incorrect temperature scales, which
are sometimes reported with nonstandard strings in ODF files.
Also, note that you may see warnings about conductivity ratios,
which some ODF files incorrectly suggest have dimensions.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>For sources that describe the ODF format, see the documentation
for the <a href="#topic+odf-class">odf</a> class.
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other things related to odf data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ODF2oce">ODF2oce</a>()</code>,
<code><a href="#topic+ODFListFromHeader">ODFListFromHeader</a>()</code>,
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+read.odf">read.odf</a>()</code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+summary+2Codf-method">summary,odf-method</a></code>
</p>
<p>Other functions that read ctd data: 
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>
</p>

<hr>
<h2 id='read.ctd.odv'>Read a &quot;ctd&quot; File in ODV Format</h2><span id='topic+read.ctd.odv'></span>

<h3>Description</h3>

<p>Read a &quot;ctd&quot; File in ODV Format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.ctd.odv(
  file,
  columns = NULL,
  station = NULL,
  missingValue,
  deploymentType,
  encoding = "latin1",
  monitor = FALSE,
  debug = getOption("oceDebug"),
  processingLog,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.ctd.odv_+3A_file">file</code></td>
<td>
<p>either a connection or a character value naming a file.
For <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.woce">read.ctd.woce()</a></code>, this may be a
wildcard (e.g. <code>"*.cnv"</code> or <code>"*.csv"</code>) in which case the return
value is a vector containing CTD objects created by reading the files from
<code><a href="base.html#topic+list.files">list.files()</a></code> with <code>pattern</code> set to the specified wildcard
pattern.</p>
</td></tr>
<tr><td><code id="read.ctd.odv_+3A_columns">columns</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that can be used to convert unrecognized
data names to resultant variable names.  This is used only by
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.odf">read.ctd.odf()</a></code>. For example,
if a data file named salinity as <code>"SAL"</code>, then using
</p>
<div class="sourceCode"><pre>d &lt;- read.ctd(f, columns=list(
    salinity=list(name="SAL",
                  unit=list(unit=expression(),
                  scale="PSS-78"))))
</pre></div>
<p>would assign the <code>"SAL"</code> column to the <code>salinity</code> entry in the data
slot of the CTD object returned by the <code style="white-space: pre;">&#8288;read.*&#8288;</code> function.</p>
</td></tr>
<tr><td><code id="read.ctd.odv_+3A_station">station</code></td>
<td>
<p>optional character string containing an identifying name or
number for the station. This can be useful if the routine cannot determine the
name automatically, or if another name is preferred.</p>
</td></tr>
<tr><td><code id="read.ctd.odv_+3A_missingvalue">missingValue</code></td>
<td>
<p>optional missing-value flag; data matching this value will
be set to <code>NA</code> upon reading. If this is provided, then it overrules any
missing-value flag found in the data. For Seabird (<code>.cnv</code>) files, there is
usually no need to set <code>missingValue</code>, because it can be inferred from the
header (typically as -9.990e-29). Set <code>missingValue=NULL</code> to turn off
missing-value detection, even in <code>.cnv</code> files that contain missing-value
codes in their headers. If <code>missingValue</code> is not specified,
then an attempt is made to infer such a value from the data, by testing
whether salinity and/or temperature has a minimum that is under -8 in value;
this should catch common values in files, without false positives. A warning
will be issued in this case, and a note inserted in the processing log of
the return value.</p>
</td></tr>
<tr><td><code id="read.ctd.odv_+3A_deploymenttype">deploymentType</code></td>
<td>
<p>character string indicating the type of deployment. Use
<code>"unknown"</code> if this is not known, <code>"profile"</code> for a profile (in
which the data were acquired during a downcast, while the device was lowered
into the water column, perhaps also including an upcast; <code>"moored"</code> if
the device is installed on a fixed mooring, <code>"thermosalinograph"</code> (or
<code>"tsg"</code>) if the device is mounted on a moving vessel, to record
near-surface properties, or <code>"towyo"</code> if the device is repeatedly
lowered and raised.</p>
</td></tr>
<tr><td><code id="read.ctd.odv_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.ctd.odv_+3A_monitor">monitor</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to provide an indication of
progress.  This is useful if <code>filename</code> is a wildcard.</p>
</td></tr>
<tr><td><code id="read.ctd.odv_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed.</p>
</td></tr>
<tr><td><code id="read.ctd.odv_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This is typically only provided for internal calls; the default that it provides is
better for normal calls by a user.</p>
</td></tr>
<tr><td><code id="read.ctd.odv_+3A_...">...</code></td>
<td>
<p>additional arguments, passed to called routines.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+read.ctd.odv">read.ctd.odv()</a></code> attempts to read files stored in ODV format,
used by some European data providers. This works only crudely, as of
2020-05-17. In particular, the translation from ODV parameter names to oce
names is <em>very</em> limited.  For example, only one of the dozens of possibilities
for variants of phosphate is handled at the moment, and that is because
this was the variant supplied in a test file sent to the author on
2020-05-16.  It is unlikely that full support of ODV files will
become available in <code><a href="#topic+read.ctd.odv">read.ctd.odv()</a></code>, given the lack of a comprehensive source
listing ODV variable names and their meanings, and low user
interest.
</p>


<h3>Value</h3>

<p>This function returns a <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li> <p><code style="white-space: pre;">&#8288;https://www.bodc.ac.uk/resources/delivery_formats/odv_format/&#8288;</code> describes
the <code>ODV</code> format.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;https://vocab.nerc.ac.uk/collection/P07/current/&#8288;</code> may be
worth consulting for variable names.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>

<hr>
<h2 id='read.ctd.saiv'>Read a ctd File in SAIV Format</h2><span id='topic+read.ctd.saiv'></span>

<h3>Description</h3>

<p><code><a href="#topic+read.ctd.saiv">read.ctd.saiv()</a></code> reads files that hold data acquired with a
SAIV model SD204 CTD profiler (reference 1). Since no
documentation on the format was available to the author,
this function was written based on examination of a particular
data file.  This almost certainly will yield limitations
for other files, in particular for those with data names
that differ from those in the sample file (see
&ldquo;Details&rdquo; for this and other limitations).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.ctd.saiv(
  file,
  encoding = "latin1",
  debug = getOption("oceDebug"),
  processingLog,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.ctd.saiv_+3A_file">file</code></td>
<td>
<p>a character string naming the file to be read.</p>
</td></tr>
<tr><td><code id="read.ctd.saiv_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.ctd.saiv_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
<tr><td><code id="read.ctd.saiv_+3A_processinglog">processingLog</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.ctd.saiv_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Some variable names are change to the oce convention, e.g.
<code>"Sal."</code> becomes <code>"salinity"</code>, <code>"Temp"</code> becomes
<code>"temperature"</code>, etc.  In the first version of the code,
this renaming was done based on examination of a single file.
This list was expanded after a user kindly supplied a one-page
document that explains the variable names and units. As with
other functions for reading <a href="#topic+oce-class">oce</a> data, <code><a href="#topic+read.ctd.saiv">read.ctd.saiv()</a></code>
resolves duplicate variable names by appending 2 to the second
instance, 3 to the third, etc.
</p>
<p>As with other <a href="#topic+ctd-class">ctd</a> objects, the <code>[[</code> operator handles
both the original name from the file, and the converted oce
name.
</p>
<p>It is worth noting the following oddities that were present
in the sample file upon which <code><a href="#topic+read.ctd.saiv">read.ctd.saiv()</a></code> was based.
</p>

<ol>
<li><p> The header line that names the data columns ends with a tab,
indicating the presence of 12 columns (the last unnamed), but the
data contain only 11 columns.  Therefore, the last tab character is
ignored by <code><a href="#topic+read.ctd.saiv">read.ctd.saiv()</a></code>.
</p>
</li>
<li><p> The test file lacked longitude and latitude information.  This
means that modern quantities like Absolute Salinity and Conservative
Temperature cannot be computed.  Users who know the location information
ought to insert values into the object returned by <code><a href="#topic+read.ctd.saiv">read.ctd.saiv()</a></code>
using <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>.
</p>
</li>
<li><p> Further to the previous point, it is not possible to compute pressure
accurately from depth (which is what the header suggests the file
contains) unless the latitude is known. In <code><a href="#topic+read.ctd.saiv">read.ctd.saiv()</a></code>,
latitude is assumed to be 45 degrees north, which is the default
used by <code><a href="#topic+swPressure">swPressure()</a></code>.
</p>
</li></ol>



<h3>Value</h3>

<p><code><a href="#topic+read.ctd.saiv">read.ctd.saiv()</a></code> returns a <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, with help from the github member with
the handle 'Rdescoteaux', who kindly supplied a sample file
and a document listing SAIV variable names.
</p>


<h3>References</h3>


<ol>
<li> <p><code style="white-space: pre;">&#8288;https://saiv.no/sd204-ctd-profiler&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that read ctd data: 
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>
</p>

<hr>
<h2 id='read.ctd.sbe'>Read a ctd File in Seabird Format</h2><span id='topic+read.ctd.sbe'></span>

<h3>Description</h3>

<p>Read a ctd File in Seabird Format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.ctd.sbe(
  file,
  columns = NULL,
  station = NULL,
  missingValue,
  deploymentType = "unknown",
  btl = FALSE,
  monitor = FALSE,
  encoding = "latin1",
  debug = getOption("oceDebug"),
  processingLog,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.ctd.sbe_+3A_file">file</code></td>
<td>
<p>either a connection or a character value naming a file.
For <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.woce">read.ctd.woce()</a></code>, this may be a
wildcard (e.g. <code>"*.cnv"</code> or <code>"*.csv"</code>) in which case the return
value is a vector containing CTD objects created by reading the files from
<code><a href="base.html#topic+list.files">list.files()</a></code> with <code>pattern</code> set to the specified wildcard
pattern.</p>
</td></tr>
<tr><td><code id="read.ctd.sbe_+3A_columns">columns</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that can be used to convert unrecognized
data names to resultant variable names.  This is used only by
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.odf">read.ctd.odf()</a></code>. For example,
if a data file named salinity as <code>"SAL"</code>, then using
</p>
<div class="sourceCode"><pre>d &lt;- read.ctd(f, columns=list(
    salinity=list(name="SAL",
                  unit=list(unit=expression(),
                  scale="PSS-78"))))
</pre></div>
<p>would assign the <code>"SAL"</code> column to the <code>salinity</code> entry in the data
slot of the CTD object returned by the <code style="white-space: pre;">&#8288;read.*&#8288;</code> function.</p>
</td></tr>
<tr><td><code id="read.ctd.sbe_+3A_station">station</code></td>
<td>
<p>optional character string containing an identifying name or
number for the station. This can be useful if the routine cannot determine the
name automatically, or if another name is preferred.</p>
</td></tr>
<tr><td><code id="read.ctd.sbe_+3A_missingvalue">missingValue</code></td>
<td>
<p>optional missing-value flag; data matching this value will
be set to <code>NA</code> upon reading. If this is provided, then it overrules any
missing-value flag found in the data. For Seabird (<code>.cnv</code>) files, there is
usually no need to set <code>missingValue</code>, because it can be inferred from the
header (typically as -9.990e-29). Set <code>missingValue=NULL</code> to turn off
missing-value detection, even in <code>.cnv</code> files that contain missing-value
codes in their headers. If <code>missingValue</code> is not specified,
then an attempt is made to infer such a value from the data, by testing
whether salinity and/or temperature has a minimum that is under -8 in value;
this should catch common values in files, without false positives. A warning
will be issued in this case, and a note inserted in the processing log of
the return value.</p>
</td></tr>
<tr><td><code id="read.ctd.sbe_+3A_deploymenttype">deploymentType</code></td>
<td>
<p>character string indicating the type of deployment. Use
<code>"unknown"</code> if this is not known, <code>"profile"</code> for a profile (in
which the data were acquired during a downcast, while the device was lowered
into the water column, perhaps also including an upcast; <code>"moored"</code> if
the device is installed on a fixed mooring, <code>"thermosalinograph"</code> (or
<code>"tsg"</code>) if the device is mounted on a moving vessel, to record
near-surface properties, or <code>"towyo"</code> if the device is repeatedly
lowered and raised.</p>
</td></tr>
<tr><td><code id="read.ctd.sbe_+3A_btl">btl</code></td>
<td>
<p>a logical value, with <code>TRUE</code> indicating that this is a <code>.BTL</code> file and <code>FALSE</code>
(the default) indicating a <code>.CNV</code> file.  Note that if <code>btl</code> is <code>TRUE</code>, the data column
names are taken directly from the file (without e.g. translating to <code>"Sal00"</code>
to <code>"salinity"</code>.  Also, the &quot;avg&quot; and &quot;sdev&quot; columns are blended together, with
all the latter named as in the file, but with <code>"_sdev"</code> appended.</p>
</td></tr>
<tr><td><code id="read.ctd.sbe_+3A_monitor">monitor</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to provide an indication of
progress.  This is useful if <code>filename</code> is a wildcard.</p>
</td></tr>
<tr><td><code id="read.ctd.sbe_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.ctd.sbe_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed.</p>
</td></tr>
<tr><td><code id="read.ctd.sbe_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This is typically only provided for internal calls; the default that it provides is
better for normal calls by a user.</p>
</td></tr>
<tr><td><code id="read.ctd.sbe_+3A_...">...</code></td>
<td>
<p>additional arguments, passed to called routines.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function reads files stored in Seabird <code>.cnv</code> format.
Note that these files can contain multiple sensors for a given field. For example,
the file might contain a column named <code>t090C</code> for one
temperature sensor and <code>t190C</code> for a second. The first will be denoted
<code>temperature</code> in the <code>data</code> slot of the return value, and the second
will be denoted <code>temperature1</code>. This means that the first sensor
will be used in any future processing that accesses <code>temperature</code>. This
is for convenience of processing, and it does not pose a limitation, because the
data from the second sensor are also available as e.g. <code>x[["temperature1"]]</code>,
where <code>x</code> is the name of the returned value.  For the details of the
mapping from <code>.cnv</code> names to <code>ctd</code> names, see <code><a href="#topic+cnvName2oceName">cnvName2oceName()</a></code>.
</p>
<p>The names of the elements in the <code>data</code> slot of the returned value depend on
the file type, as signalled by the <code>btl</code> argument.  For the default case of <code>.cnv</code> files,
the original data names as stored in <code>file</code> are stored within the <code>metadata</code>
slot as <code>dataNamesOriginal</code>, and are displayed with <code>summary</code> alongside the
numerical summary; see the Appendix VI of reference 2 for the meanings of these
names (in the &quot;Short Name&quot; column of the table spanning pages 161 through 172).
However, for the case of <code>.btl</code> files, the column names are as described
in the documentation entry for the <code>btl</code> argument.
</p>


<h3>Value</h3>

<p>This function returns a <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>A note on hand-entered headers</h3>

<p>CNV files may have a section that contains human-entered information. This is
detected by <code>read.ctd.sbe()</code> as lines that begin with two asterisks. Decoding
this information can be tricky, because humans have many ways of writing
things.
</p>
<p>For example, consider the <code>date</code> item in the <code>metadata</code> slot of the returned
value.  <code>read.ctd.sbe()</code> infers this value in one of two ways.  First, if
there is a header line staring with
</p>
<div class="sourceCode"><pre>* NMEA UTC (Time) =
</pre></div>
<p>then that value is decoded and used for <code>date</code>.  This header line, preceded
by a single asterisk, is not human-entered, and so there is reason to hope
for a uniform format that can be handled by <code>read.ctd.sbe()</code>.  However, if
there is no NMEA header line, then <code>read.ctd.sbe()</code> will look for a line
starting with
</p>
<div class="sourceCode"><pre>** Date:
</pre></div>
<p>which was human-entered. This is the second choice, because humans write
dates in a bewildering variety of ways, and <code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>, which
<code>read.ctd.sbe</code> uses to parse the date, cannot handle them all. If there is a
problem, <code>read.ctd.sbe()</code> issues a warning and stores NA in <code>date</code>.
</p>
<p>A similar error-detection procedure is used for human-entered location data,
which appear in lines starting with either
</p>
<div class="sourceCode"><pre>** Longitude:
</pre></div>
<p>or
</p>
<div class="sourceCode"><pre>** Latitude:
</pre></div>
<p>which often take forms that <code>read.ctd.sbe()</code> cannot parse.
</p>
<p>It is important to note that, even if no warnings are issued, there is a
reasonably high chance that human-entered data will be scanned incorrectly.
(Did the operator remember to indicate the hemisphere? Does 123.456 indicate
decimal degrees, or 123 degrees plus 45.6 minutes? Is hemisphere indicated by
sign or by letter, and, if the latter, where does it appear?)
</p>
<p>In deep-sea work, a ship might steam for 6 hours between CTD stations, so the
ship-time cost of each CTD file can be several thousand dollars.  Surely it
is not unreasonable for an analyst to take a minute to glance at the CNV
file, to ascertain whether <code>read.ctd.sbe()</code> inferred correct values.
</p>
<p><code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code> is helpful for correcting problems with individual files,
but if many files are systematically problematic, say for a whole cruise or
perhaps even for a whole institution, then it might sense to set up a wrapper
function to correct deficiencies in the CNV files. As an example, the
following handles dates specified in a particular nonstandard way.
</p>
<div class="sourceCode"><pre>read.ctd.sbe.wrapper &lt;- function(cnv)
{
    lines &lt;- readLines(cnv)
    # Change month-day-year to year-month-day, so as.POSIXct() can parse it.
    lines &lt;- gsub("^\\*\\* Date: (.*)-(.*)-(.*)", "** Date: \\3-\\1-\\2", lines)
    read.ctd.sbe(textConnection(lines))
}
</pre></div>


<h3>A note on sampling times</h3>

<p>Until November of 2018, there was a possibility for great confusion in the
storage of the time entries within the <code>data</code> slot, because <code>read.ctd.sbe</code>
renamed each of the ten variants of time (see reference 2 for a list) as
<code>"time"</code> in the <code>data</code> slot of the returned value.  For CTD profiles, this
was perhaps not a great problem, but it could lead to significant confusion
for moored data. Therefore, a change to <code>read.ctd.sbe</code> was made, so that it
would Seabird times, using the <code>start_time</code> entry in the CNV file header
(which is stored as <code>startTime</code> in the object <code>metadata</code> slot), along with
specific time columns as follows (and as documented, with uneven clarity, in
the SBE Seasoft data processing manual, revision 7.26.8, Appendix VI):
</p>

<table>
<tr>
 <td style="text-align: right;">
<strong>Item</strong> </td><td style="text-align: left;"> <strong>Meaning</strong></td>
</tr>
<tr>
 <td style="text-align: right;">
<code>timeS</code>   </td><td style="text-align: left;"> seconds elapsed since <code>start_time</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<code>timeM</code>   </td><td style="text-align: left;"> minutes elapsed since <code>start_time</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<code>timeH</code>   </td><td style="text-align: left;"> hours elapsed since <code>start_time</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<code>timeJ</code>   </td><td style="text-align: left;"> Julian days since the start of the year of the first observation</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>timeN</code>   </td><td style="text-align: left;"> NMEA-based time, in seconds past Jan 1, 1970</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>timeQ</code>   </td><td style="text-align: left;"> NMEA-based time, in seconds past Jan 1, 2000</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>timeK</code>   </td><td style="text-align: left;"> NMEA-based time, in seconds past Jan 1, 2000</td>
</tr>
<tr>
 <td style="text-align: right;">
<code>timeJV2</code> </td><td style="text-align: left;"> as <code>timeJ</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<code>timeSCP</code> </td><td style="text-align: left;"> as <code>timeJ</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<code>timeY</code>   </td><td style="text-align: left;"> computer time, in seconds past Jan 1, 1970</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>NOTE: not all of these times have been tested properly, and so users
are asked to report incorrect times, so that <code>read.ctd.sbe</code> can
be improved.
</p>


<h3>A note on scales</h3>

<p>The user might encounter data files with a variety of scales for temperature and
salinity. Oce keeps track of these scales in the units it sets up for the stored
variables. For example, if <code>A</code> is a CTD object, then
<code>A[["temperatureUnit"]]$scale</code> is a character string that will indicate the scale.
Modern-day data will have <code>"ITS-90"</code> for that scale, and old data may have
<code>"IPTS-68"</code>. The point of saving the scale in this way is so that the various
formulas that deal with water properties can account for the scale, e.g. converting
from numerical values saved on the <code>"IPTS-68"</code> scale to the newer scale, using
<code><a href="#topic+T90fromT68">T90fromT68()</a></code> before doing calculations that are expressed in
terms of the <code>"ITS-90"</code> scale. This is taken care of by retrieving temperatures
with the accessor function, e.g. writing <code>A[["temperature"]]</code> will either
retrieve the stored values (if the scale is ITS-90) or converted values (if
the scale is IPTS-68). Even though this procedure should work, users who
really care about the details of their data are well-advised to do a couple
of tests after examining the first data line of their data file in an editor.
Note that reading a file that contains IPTS-68 temperatures produces a warning.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>


<ol>
<li><p> The Sea-Bird SBE 19plus profiler is described at
<code style="white-space: pre;">&#8288;http://www.seabird.com/products/spec_sheets/19plusdata.htm&#8288;</code>.  Some more
information is given in the Sea-Bird data-processing manual
(next item).
</p>
</li>
<li><p> A SBE data processing manual was once at
<code style="white-space: pre;">&#8288;http://www.seabird.com/document/sbe-data-processing-manual&#8288;</code>,
but as of summer 2018, this no longer seems to be provided by SeaBird.
A web search will turn up copies of the manual that have been put
online by various research groups and data-archiving agencies.
As of 2018-07-05, the latest version was named
<code>SBEDataProcessing_7.26.4.pdf</code> and had release date 12/08/2017,
and this was the reference version used in coding <code>oce</code>.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that read ctd data: 
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>f &lt;- system.file("extdata", "ctd.cnv.gz", package = "oce")
d &lt;- read.ctd(f)

</code></pre>

<hr>
<h2 id='read.ctd.ssda'>Read a ctd File in SSDA Format</h2><span id='topic+read.ctd.ssda'></span>

<h3>Description</h3>

<p><code><a href="#topic+read.ctd.ssda">read.ctd.ssda()</a></code> reads CTD files in Sea &amp; Sun Technology's Standard Data
Acquisition (SSDA) format. This function is somewhat preliminary, in the
sense that header information is not scanned fully, and some guesses have
been made about the meanings of variables and units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.ctd.ssda(
  file,
  encoding = "latin1",
  debug = getOption("oceDebug"),
  processingLog
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.ctd.ssda_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file to
load.</p>
</td></tr>
<tr><td><code id="read.ctd.ssda_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.ctd.ssda_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. If nonzero, some information
is printed.</p>
</td></tr>
<tr><td><code id="read.ctd.ssda_+3A_processinglog">processingLog</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code><a href="#topic+read.ctd.ssda">read.ctd.ssda()</a></code> returns a <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, with help from Liam MacNeil
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that read ctd data: 
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>
</p>

<hr>
<h2 id='read.ctd.woce'>Read a ctd File in WOCE-Exchange Format</h2><span id='topic+read.ctd.woce'></span>

<h3>Description</h3>

<p>This reads WOCE exchange files that start with the string <code>"CTD"</code>.
There are two variants: one in which the first 4 characters are
<code>"CTD,"</code> and the other in which the first 3 characters are
again <code>"CTD"</code> but no other non-whitespace characters occur on
the line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.ctd.woce(
  file,
  columns = NULL,
  station = NULL,
  missingValue,
  deploymentType = "unknown",
  monitor = FALSE,
  encoding = "latin1",
  debug = getOption("oceDebug"),
  processingLog,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.ctd.woce_+3A_file">file</code></td>
<td>
<p>either a connection or a character value naming a file.
For <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.woce">read.ctd.woce()</a></code>, this may be a
wildcard (e.g. <code>"*.cnv"</code> or <code>"*.csv"</code>) in which case the return
value is a vector containing CTD objects created by reading the files from
<code><a href="base.html#topic+list.files">list.files()</a></code> with <code>pattern</code> set to the specified wildcard
pattern.</p>
</td></tr>
<tr><td><code id="read.ctd.woce_+3A_columns">columns</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that can be used to convert unrecognized
data names to resultant variable names.  This is used only by
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.odf">read.ctd.odf()</a></code>. For example,
if a data file named salinity as <code>"SAL"</code>, then using
</p>
<div class="sourceCode"><pre>d &lt;- read.ctd(f, columns=list(
    salinity=list(name="SAL",
                  unit=list(unit=expression(),
                  scale="PSS-78"))))
</pre></div>
<p>would assign the <code>"SAL"</code> column to the <code>salinity</code> entry in the data
slot of the CTD object returned by the <code style="white-space: pre;">&#8288;read.*&#8288;</code> function.</p>
</td></tr>
<tr><td><code id="read.ctd.woce_+3A_station">station</code></td>
<td>
<p>optional character string containing an identifying name or
number for the station. This can be useful if the routine cannot determine the
name automatically, or if another name is preferred.</p>
</td></tr>
<tr><td><code id="read.ctd.woce_+3A_missingvalue">missingValue</code></td>
<td>
<p>optional missing-value flag; data matching this value will
be set to <code>NA</code> upon reading. If this is provided, then it overrules any
missing-value flag found in the data. For Seabird (<code>.cnv</code>) files, there is
usually no need to set <code>missingValue</code>, because it can be inferred from the
header (typically as -9.990e-29). Set <code>missingValue=NULL</code> to turn off
missing-value detection, even in <code>.cnv</code> files that contain missing-value
codes in their headers. If <code>missingValue</code> is not specified,
then an attempt is made to infer such a value from the data, by testing
whether salinity and/or temperature has a minimum that is under -8 in value;
this should catch common values in files, without false positives. A warning
will be issued in this case, and a note inserted in the processing log of
the return value.</p>
</td></tr>
<tr><td><code id="read.ctd.woce_+3A_deploymenttype">deploymentType</code></td>
<td>
<p>character string indicating the type of deployment. Use
<code>"unknown"</code> if this is not known, <code>"profile"</code> for a profile (in
which the data were acquired during a downcast, while the device was lowered
into the water column, perhaps also including an upcast; <code>"moored"</code> if
the device is installed on a fixed mooring, <code>"thermosalinograph"</code> (or
<code>"tsg"</code>) if the device is mounted on a moving vessel, to record
near-surface properties, or <code>"towyo"</code> if the device is repeatedly
lowered and raised.</p>
</td></tr>
<tr><td><code id="read.ctd.woce_+3A_monitor">monitor</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to provide an indication of
progress.  This is useful if <code>filename</code> is a wildcard.</p>
</td></tr>
<tr><td><code id="read.ctd.woce_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.ctd.woce_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed.</p>
</td></tr>
<tr><td><code id="read.ctd.woce_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This is typically only provided for internal calls; the default that it provides is
better for normal calls by a user.</p>
</td></tr>
<tr><td><code id="read.ctd.woce_+3A_...">...</code></td>
<td>
<p>additional arguments, passed to called routines.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>The WOCE-exchange format was once described at
<code style="white-space: pre;">&#8288;http://woce.nodc.noaa.gov/woce_v3/wocedata_1/whp/exchange/exchange_format_desc.htm&#8288;</code>
although that link is no longer working as of December 2020.
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that read ctd data: 
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>
</p>

<hr>
<h2 id='read.ctd.woce.other'>Read a ctd File in WOCE-Exchange EXPOCODE Format</h2><span id='topic+read.ctd.woce.other'></span>

<h3>Description</h3>

<p>This reads WOCE exchange files that start with the string <code>"EXPOCODE"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.ctd.woce.other(
  file,
  columns = NULL,
  station = NULL,
  missingValue,
  deploymentType = "unknown",
  monitor = FALSE,
  encoding = "latin1",
  debug = getOption("oceDebug"),
  processingLog,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.ctd.woce.other_+3A_file">file</code></td>
<td>
<p>either a connection or a character value naming a file.
For <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.woce">read.ctd.woce()</a></code>, this may be a
wildcard (e.g. <code>"*.cnv"</code> or <code>"*.csv"</code>) in which case the return
value is a vector containing CTD objects created by reading the files from
<code><a href="base.html#topic+list.files">list.files()</a></code> with <code>pattern</code> set to the specified wildcard
pattern.</p>
</td></tr>
<tr><td><code id="read.ctd.woce.other_+3A_columns">columns</code></td>
<td>
<p>an optional <a href="base.html#topic+list">list</a> that can be used to convert unrecognized
data names to resultant variable names.  This is used only by
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> and <code><a href="#topic+read.ctd.odf">read.ctd.odf()</a></code>. For example,
if a data file named salinity as <code>"SAL"</code>, then using
</p>
<div class="sourceCode"><pre>d &lt;- read.ctd(f, columns=list(
    salinity=list(name="SAL",
                  unit=list(unit=expression(),
                  scale="PSS-78"))))
</pre></div>
<p>would assign the <code>"SAL"</code> column to the <code>salinity</code> entry in the data
slot of the CTD object returned by the <code style="white-space: pre;">&#8288;read.*&#8288;</code> function.</p>
</td></tr>
<tr><td><code id="read.ctd.woce.other_+3A_station">station</code></td>
<td>
<p>optional character string containing an identifying name or
number for the station. This can be useful if the routine cannot determine the
name automatically, or if another name is preferred.</p>
</td></tr>
<tr><td><code id="read.ctd.woce.other_+3A_missingvalue">missingValue</code></td>
<td>
<p>optional missing-value flag; data matching this value will
be set to <code>NA</code> upon reading. If this is provided, then it overrules any
missing-value flag found in the data. For Seabird (<code>.cnv</code>) files, there is
usually no need to set <code>missingValue</code>, because it can be inferred from the
header (typically as -9.990e-29). Set <code>missingValue=NULL</code> to turn off
missing-value detection, even in <code>.cnv</code> files that contain missing-value
codes in their headers. If <code>missingValue</code> is not specified,
then an attempt is made to infer such a value from the data, by testing
whether salinity and/or temperature has a minimum that is under -8 in value;
this should catch common values in files, without false positives. A warning
will be issued in this case, and a note inserted in the processing log of
the return value.</p>
</td></tr>
<tr><td><code id="read.ctd.woce.other_+3A_deploymenttype">deploymentType</code></td>
<td>
<p>character string indicating the type of deployment. Use
<code>"unknown"</code> if this is not known, <code>"profile"</code> for a profile (in
which the data were acquired during a downcast, while the device was lowered
into the water column, perhaps also including an upcast; <code>"moored"</code> if
the device is installed on a fixed mooring, <code>"thermosalinograph"</code> (or
<code>"tsg"</code>) if the device is mounted on a moving vessel, to record
near-surface properties, or <code>"towyo"</code> if the device is repeatedly
lowered and raised.</p>
</td></tr>
<tr><td><code id="read.ctd.woce.other_+3A_monitor">monitor</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to provide an indication of
progress.  This is useful if <code>filename</code> is a wildcard.</p>
</td></tr>
<tr><td><code id="read.ctd.woce.other_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.ctd.woce.other_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed.</p>
</td></tr>
<tr><td><code id="read.ctd.woce.other_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This is typically only provided for internal calls; the default that it provides is
better for normal calls by a user.</p>
</td></tr>
<tr><td><code id="read.ctd.woce.other_+3A_...">...</code></td>
<td>
<p>additional arguments, passed to called routines.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that read ctd data: 
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>
</p>

<hr>
<h2 id='read.echosounder'>Read an echosounder File</h2><span id='topic+read.echosounder'></span>

<h3>Description</h3>

<p>Reads a biosonics echosounder file.  This function was written for and
tested with single-beam, dual-beam, and split-beam Biosonics files of type
V3, and may not work properly with other file formats.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.echosounder(
  file,
  channel = 1,
  soundSpeed,
  tz = getOption("oceTz"),
  encoding = NA,
  debug = getOption("oceDebug"),
  processingLog
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.echosounder_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.</p>
</td></tr>
<tr><td><code id="read.echosounder_+3A_channel">channel</code></td>
<td>
<p>sequence number of channel to extract, for multi-channel
files.</p>
</td></tr>
<tr><td><code id="read.echosounder_+3A_soundspeed">soundSpeed</code></td>
<td>
<p>sound speed, in m/s. If not provided, this is calculated
using <code><a href="#topic+swSoundSpeed">swSoundSpeed</a></code><code style="white-space: pre;">&#8288;(35,15,30,eos="unesco")&#8288;</code>.  (In theory,
it could be calculated using the temperature and salinity that are stored
in the data file, but these will just be nominal values, anyway.</p>
</td></tr>
<tr><td><code id="read.echosounder_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.echosounder_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.echosounder_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="read.echosounder_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log,
typically only provided for internal calls.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+echosounder-class">echosounder</a> object.
</p>


<h3>Bugs</h3>

<p>Only the amplitude information (in counts) is determined.  A
future version of this function may provide conversion to dB, etc.  The
handling of dual-beam and split-beam files is limited.  In the dual-beam
cse, only the wide beam signal is processed (I think ... it could be the
narrow beam, actually, given the confusing endian tricks being played).  In
the split-beam case, only amplitude is read, with the x-axis and y-axis
angle data being ignored.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, with help from Clark Richards
</p>


<h3>References</h3>

<p>Various echosounder instruments provided by BioSonics are
described at the company website, <code style="white-space: pre;">&#8288;https://www.biosonicsinc.com/&#8288;</code>.  The
document listed as reference 1 below was provided to the author of this function in
November 2011, which suggests that the data format was not changed since
July 2010.
</p>

<ol>
<li><p> Biosonics, 2010.  DT4 Data File Format Specification.  BioSonics
Advanced Digital Hydroacoustics. July, 2010.  SOFTWARE AND ENGINEERING
LIBRARY REPORT BS&amp;E-2004-07-0009-2.0.
</p>
</li></ol>



<h3>See Also</h3>

<p>The documentation for <a href="#topic+echosounder-class">echosounder</a> explains the
structure of <code>ctd</code> objects, and also outlines the other functions
dealing with them.
</p>
<p>Other things related to echosounder data: 
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic+as.echosounder">as.echosounder</a>()</code>,
<code><a href="#topic+echosounder-class">echosounder-class</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+findBottom">findBottom</a>()</code>,
<code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+summary+2Cechosounder-method">summary,echosounder-method</a></code>
</p>

<hr>
<h2 id='read.g1sst'>Read a g1sst File</h2><span id='topic+read.g1sst'></span>

<h3>Description</h3>

<p>Read a G1SST file in the netcdf format provided by the ERDDAP server
(see reference 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.g1sst(file, encoding = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.g1sst_+3A_file">file</code></td>
<td>
<p>character value containing the name of a netcdf file containing G1SST data.</p>
</td></tr>
<tr><td><code id="read.g1sst_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As noted in the documentation for the <a href="#topic+g1sst-class">g1sst</a> class, one
must be aware of the incorporation of model simulations in the
<code>g1sst</code> product. For example, the code presented below
might lead one to believe that the mapped field represents
observations, whereas in fact it can be verified by
consulting reference 2 (clicking and unclicking the radio button to
show just the data) that the field mostly derives from simulation.
</p>


<h3>Value</h3>

<p>A <a href="#topic+g1sst-class">g1sst</a> object.
</p>


<h3>Sample of Usage</h3>

<pre>
# Construct query, making it easier to understand and modify.
day &lt;- "2016-01-02"
lon0 &lt;- -66.5
lon1 &lt;- -64.0
lat0 &lt;- 44
lat1 &lt;- 46
source &lt;- paste("https://coastwatch.pfeg.noaa.gov/erddap/griddap/",
    "jplG1SST.nc?",
    "SST
    "
    "
    "
if (!length(list.files(pattern="^a.nc$")))
    download.file(source, "a.nc")
d &lt;- read.g1sst("a.nc")
plot(d, "SST", col=oceColorsTemperature)
if (requireNamespace("ocedata", quietly=TRUE)) {
    data(coastlineWorldFine, package="ocedata")
    lines(coastlineWorldFine[["longitude"]],coastlineWorldFine[["latitude"]])
}
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> ERDDAP Portal <code style="white-space: pre;">&#8288;https://coastwatch.pfeg.noaa.gov/erddap/&#8288;</code>
</p>
</li>
<li><p> JPO OurOcean Portal <code style="white-space: pre;">&#8288;https://ourocean.jpl.nasa.gov/SST/&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to g1sst data: 
<code><a href="#topic++5B+5B+2Cg1sst-method">[[,g1sst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cg1sst-method">[[&lt;-,g1sst-method</a></code>,
<code><a href="#topic+g1sst-class">g1sst-class</a></code>
</p>

<hr>
<h2 id='read.gps'>Read a gps File</h2><span id='topic+read.gps'></span>

<h3>Description</h3>

<p>Reads GPX format files by simply finding all longitudes and latitudes; in
other words, information on separate tracks, or waypoints, etc., is lost.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.gps(
  file,
  type = NULL,
  encoding = "latin1",
  debug = getOption("oceDebug"),
  processingLog
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.gps_+3A_file">file</code></td>
<td>
<p>name of file containing gps data.</p>
</td></tr>
<tr><td><code id="read.gps_+3A_type">type</code></td>
<td>
<p>type of file, which will be inferred from examination of the
data if not supplied.  In the present version, the only choice for
<code>type</code> is <code>"gpx"</code>.</p>
</td></tr>
<tr><td><code id="read.gps_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.gps_+3A_debug">debug</code></td>
<td>
<p>set to TRUE to print information about the header, etc.</p>
</td></tr>
<tr><td><code id="read.gps_+3A_processinglog">processingLog</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+gps-class">gps</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to gps data: 
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic+as.gps">as.gps</a>()</code>,
<code><a href="#topic+gps-class">gps-class</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+summary+2Cgps-method">summary,gps-method</a></code>
</p>

<hr>
<h2 id='read.index'>Read a NOAA Ocean Index File</h2><span id='topic+read.index'></span>

<h3>Description</h3>

<p>Read an ocean index file, in the format used by NOAA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.index(
  file,
  format,
  missingValue,
  tz = getOption("oceTz"),
  encoding = "latin1",
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.index_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  May be a URL.</p>
</td></tr>
<tr><td><code id="read.index_+3A_format">format</code></td>
<td>
<p>optional character string indicating the format type. If not
supplied, a guess will be made, based on examination of the first few lines
of the file. If supplied, the possibilities are <code>"noaa"</code> and
<code>"ucar"</code>. See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="read.index_+3A_missingvalue">missingValue</code></td>
<td>
<p>If supplied, this is a numerical value that indicates
invalid data. In some datasets, this is -99.9, but other values may be used.
If <code>missingValue</code> is not supplied, any data that have value equal
to or less than -99 are considered invalid. Set <code>missingValue</code>
to <code>TRUE</code> to turn of the processing of missing values.</p>
</td></tr>
<tr><td><code id="read.index_+3A_tz">tz</code></td>
<td>
<p>character string indicating time zone to be assumed in the data.</p>
</td></tr>
<tr><td><code id="read.index_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.index_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging, ignored in the present version
of the function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads a text-format index file, in either of two formats. If
<code>format</code> is missing, then the first line of the file is examined. If
that line contains 2 (whitespace-separated) tokens, then <code>"noaa"</code>
format is assumed. If it contains 13 tokens, then <code>"ucar"</code> format is assumed.
Otherwise, an error is reported.
</p>
<p>In the <code>"noaa"</code> format, the two tokens on the first line are taken to
be the start year and the end year, respectively. The second line
must contain a single token, the missing value. All further lines must contain
12 tokens, for the values in January, February, etc.
</p>
<p>In the <code>"ucar"</code> format, all data lines must contain the 13 tokens,
the first of which is the year, with the following 12 being the values
for January, February, etc.
</p>


<h3>Value</h3>

<p>A data frame containing <code>t</code>, a POSIX time, and <code>index</code>,
the numerical index.  The times are set to the 15th day of each month, which
is a guess that may need to be changed if so indicated by documentation (yet
to be located).
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
par(mfrow=c(2, 1))
# 1. AO, Arctic oscillation
#  Note that data used to be at https://www.esrl.noaa.gov/psd/data/correlation/ao.data
ao &lt;- read.index("https://psl.noaa.gov/data/correlation/ao.data")
aorecent &lt;- subset(ao, t &gt; as.POSIXct("2000-01-01"))
oce.plot.ts(aorecent$t, aorecent$index)
# 2. SOI, probably more up-to-date then data(soi, package="ocedata")
soi &lt;- read.index("https://www.cgd.ucar.edu/cas/catalog/climind/SOI.signal.ascii")
soirecent &lt;- subset(soi, t &gt; as.POSIXct("2000-01-01"))
oce.plot.ts(soirecent$t, soirecent$index)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>See <code style="white-space: pre;">&#8288;https://psl.noaa.gov/data/climateindices/list/&#8288;</code>
for a list of indices.
</p>

<hr>
<h2 id='read.landsat'>Read a landsat File Directory</h2><span id='topic+read.landsat'></span>

<h3>Description</h3>

<p>Read a landsat data file, producing an object of <a href="#topic+landsat-class">landsat</a>.
The actual reading is done with <code><a href="tiff.html#topic+readTIFF">tiff::readTIFF()</a></code> in the
<a href="https://CRAN.R-project.org/package=tiff"><span class="pkg">tiff</span></a> package, so that package must be installed for
<code>read.landsat</code> to work.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.landsat(
  file,
  band = "all",
  emissivity = 0.984,
  decimate,
  encoding = "latin1",
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.landsat_+3A_file">file</code></td>
<td>
<p>A connection or a character string giving the name of the file to
load.  This is a directory name containing the data.</p>
</td></tr>
<tr><td><code id="read.landsat_+3A_band">band</code></td>
<td>
<p>The bands to be read, by default all of the bands.  Use
<code>band=NULL</code> to skip the reading of bands, instead reading only the
image metadata, which is often enough to check if the image is of
interest in a given study. See &ldquo;Details&rdquo; for the names of the
bands, some of which are pseudo-bands, computed from the actual data.</p>
</td></tr>
<tr><td><code id="read.landsat_+3A_emissivity">emissivity</code></td>
<td>
<p>Value of the emissivity of the surface, stored as
<code>emissivity</code> in the <code>metadata</code> slot of the
resultant object. This is used in the
calculation of surface temperature, as explained in the discussion of
accessor functions for <a href="#topic+landsat-class">landsat</a>. The default value is
from Konda et al.  (1994). These authors suggest an uncertainty of 0.04,
but a wider range of values can be found in the literature.  The value of
<code>metadata$emissivity</code> is easy to alter, either as a single value or
as a matrix, yielding flexibility of calculation.</p>
</td></tr>
<tr><td><code id="read.landsat_+3A_decimate">decimate</code></td>
<td>
<p>optional positive integer indicating the degree to which
the data should be subsampled after reading and before storage. Setting
this to 10 can speed up reading by a factor of 3 or more, but higher values
have diminishing effect.  In exploratory work, it is useful to set
<code>decimate=10</code>, to plot the image to determine a subregion
of interest, and then to use <code><a href="#topic+landsatTrim">landsatTrim()</a></code> to trim the image.</p>
</td></tr>
<tr><td><code id="read.landsat_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.landsat_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Landsat data are provided in directories that contain TIFF files and header
information, and <code>read.landsat</code> relies on a strict convention for the
names of the files in those directories.  Those file names were found by
inspection of some data, on the assumption that similar patterns will hold for
other datasets for any given satellite. This is a brittle approach and it
should be born in mind if <code>read.landsat</code> fails for a given dataset.
</p>
<p>For Landsat 8, there are 11 bands, with names <code>"aerosol"</code> (band 1),
<code>"blue"</code> (band 2), <code>"green"</code> (band 3), <code>"red"</code> (band 4),
<code>"nir"</code> (band 5), <code>"swir1"</code> (band 6), <code>"swir2"</code> (band 7),
<code>"panchromatic"</code> (band 8), <code>"cirrus"</code> (band 9), <code>"tirs1"</code> (band
10), and <code>"tirs2"</code> (band 11).
In addition to the above, setting <code>band="terralook"</code> may be used as
an abbreviation for <code>band=c("red", "green", "nir")</code>.
</p>
<p>For Landsat 7, there 8 bands, with names <code>"blue"</code> (band 1), <code>"green"</code>
(band 2), <code>"red"</code> (band 3), <code>"nir"</code> (band 4), <code>"swir1"</code> (band
5), <code>"tir1"</code> (band 6A), <code>"tir2"</code> (band 6B), <code>"swir2"</code> (band 7)
and <code>"panchromatic"</code> (band 8).
</p>
<p>For Landsat 4 and 5, the bands similar to Landsat 7 but without
<code>"panchromatic"</code> (band 8).
</p>


<h3>Value</h3>

<p>A <a href="#topic+landsat-class">landsat</a> object, with the conventional Oce
slots <code>metadata</code>, <code>data</code> and <code>processingLog</code>.  The
<code>metadata</code> is mainly intended for use by Oce functions, but for generality
it also contains an entry named <code>header</code> that represents the full image
header in a list (with names made lower-case).  The <code>data</code> slot holds
matrices of the data in the requested bands, and users may add extra matrices
if desired, e.g. to store calculated quantities.
</p>


<h3>Storage requirements</h3>

<p>Landsat data files (directories, really) are large, accounting for
approximately 1 gigabyte each.  The storage of the Oce object is
similar (see <a href="#topic+landsat-class">landsat</a>).  In R, many operations involving
copying data, so that dealing with full-scale landsat images can overwhelm
computers with storage under 8GB.  For this reason, it is typical to read just
the bands that are of interest.  It is also helpful to use
<code><a href="#topic+landsatTrim">landsatTrim()</a></code> to trim the data to a geographical range, or
to use <code><a href="#topic+decimate">decimate()</a></code> to get a coarse view of the domain, especially
early in an analysis.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Konda, M. Imasato N., Nishi, K., and T. Toda, 1994.  Measurement of the Sea
Surface Emissivity.  <em>Journal of Oceanography</em>, 50, 17:30.
<a href="https://doi.org/10.1007/BF02233853">doi:10.1007/BF02233853</a>
</p>
</li></ol>



<h3>See Also</h3>

<p>See the documentation for the <a href="#topic+landsat-class">landsat</a> class
for more information on <code>landsat</code> objects,
especially band information.  Use <code><a href="#topic+landsatTrim">landsatTrim()</a></code> to trim Landsat
objects geographically and <code><a href="#topic+landsatAdd">landsatAdd()</a></code> to add new &ldquo;bands.&rdquo;  The
accessor operator (<code>[[</code>) is used to access band information, full or
decimated, and to access certain derived quantities.  A sample dataset named
<code><a href="#topic+landsat">landsat()</a></code> is provided by the <a href="https://CRAN.R-project.org/package=oce"><span class="pkg">oce</span></a> package.
</p>
<p>Other things related to landsat data: 
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic+landsat-class">landsat-class</a></code>,
<code><a href="#topic+landsatAdd">landsatAdd</a>()</code>,
<code><a href="#topic+landsatTrim">landsatTrim</a>()</code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+summary+2Clandsat-method">summary,landsat-method</a></code>
</p>

<hr>
<h2 id='read.lisst'>Read a lisst File</h2><span id='topic+read.lisst'></span>

<h3>Description</h3>

<p>Read a LISST data file.
The file should contain 42 columns, with no header.  If there are fewer than
42 columns, an error results.  If there are more, only the first 42 are
used.  Note that <code><a href="#topic+read.oce">read.oce()</a></code> can recognize LISST files by their
having a name ending in <code>".asc"</code> and by having 42 elements on the first
line.  Even so, it is preferred to use the present function, because it
gives the opportunity to specify the year and timezone, so that times can be
calculated properly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.lisst(
  file,
  year = 0,
  tz = "UTC",
  longitude = NA,
  latitude = NA,
  encoding = "latin1"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.lisst_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.</p>
</td></tr>
<tr><td><code id="read.lisst_+3A_year">year</code></td>
<td>
<p>year in which the measurement of the series was made.</p>
</td></tr>
<tr><td><code id="read.lisst_+3A_tz">tz</code></td>
<td>
<p>time zone.</p>
</td></tr>
<tr><td><code id="read.lisst_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (stored in metadata)</p>
</td></tr>
<tr><td><code id="read.lisst_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (stored in metadata)</p>
</td></tr>
<tr><td><code id="read.lisst_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>x A <a href="#topic+lisst-class">lisst</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to lisst data: 
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic+as.lisst">as.lisst</a>()</code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+summary+2Clisst-method">summary,lisst-method</a></code>
</p>

<hr>
<h2 id='read.lobo'>Read a lobo File</h2><span id='topic+read.lobo'></span>

<h3>Description</h3>

<p>Read a data file created by a LOBO instrument.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.lobo(file, cols = 7, encoding = "latin1", processingLog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.lobo_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.</p>
</td></tr>
<tr><td><code id="read.lobo_+3A_cols">cols</code></td>
<td>
<p>number of columns in dataset.</p>
</td></tr>
<tr><td><code id="read.lobo_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.lobo_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This version of <code>read.lobo</code> is really quite crude, having been
developed mainly for a &ldquo;predict the Spring bloom&rdquo; contest at Dalhousie
University.  In particular, the function assumes that the data columns are
exactly as specified in the Examples section; if you reorder the columns or
add new ones, this function is unlikely to work correctly. Furthermore, it
should be noted that the file format was inferred simply by downloading
files; the supplier makes no claims that the format will be fixed in time.
It is also worth noting that there is no <code><a href="#topic+read.oce">read.oce()</a></code> equivalent
to <code>read.lobo</code>, because the file format has no recognizable header.
</p>


<h3>Value</h3>

<p>A <a href="#topic+lobo-class">lobo</a> object.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
uri &lt;- paste("http://lobo.satlantic.com/cgi-bin/nph-data.cgi?",
    "min_date=20070220&amp;max_date=20070305",
    "&amp;x=date&amp;",
    "y=current_across1,current_along1,nitrate,fluorescence,salinity,temperature&amp;",
    "data_format=text", sep="")
lobo &lt;- read.lobo(uri)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to lobo data: 
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic+as.lobo">as.lobo</a>()</code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+summary+2Clobo-method">summary,lobo-method</a></code>
</p>

<hr>
<h2 id='read.met'>Read a met File</h2><span id='topic+read.met'></span>

<h3>Description</h3>

<p>Reads some meteorological file formats used by the Environment Canada (see
reference 1).  Since the agency does not publish the data formats, this
function has to be adjusted every few years, when a user finds that the
format has changed. <strong>Caution:</strong> as of March 2022, this function fails on some
on Windows machines, for reasons that seem to be related to the handling of
both file encoding and system encoding. Adjusting the <code>encoding</code> parameter
of this function might help.  If not, try reading the data with
<code><a href="utils.html#topic+read.csv">read.csv()</a></code> and then using <code><a href="#topic+as.met">as.met()</a></code> to create a <a href="#topic+met-class">met</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.met(
  file,
  type = NULL,
  skip = NULL,
  encoding = "latin1",
  tz = getOption("oceTz"),
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.met_+3A_file">file</code></td>
<td>
<p>a character string naming a file that holds met data.</p>
</td></tr>
<tr><td><code id="read.met_+3A_type">type</code></td>
<td>
<p>if <code>NULL</code>, which is the default, then an attempt is
made to infer the type from the file contents. If this fails, it
will be necessary for the user to provide a value for the <code>type</code>
argument.  The permitted choices are: (a) <code>"csv"</code> or <code>"csv1"</code> for an
old CSV format no longer provided as of October 2019,
(b) <code>"csv2"</code> for a CSV format noticed on the Environment Canada
website in October 2019 (but note that the paired metadata file
is ignored), (c) <code>"csv3"</code> for a CSV format noticed on the
Environment Canada website in January 2020,
and (d) <code>"xml2"</code> for an XML format that was noticed
on the Environment Canada website in October 2019.</p>
</td></tr>
<tr><td><code id="read.met_+3A_skip">skip</code></td>
<td>
<p>integer giving the number of header lines that precede the
data.  This is ignored unless <code>type</code> is <code>"csv"</code> or <code>"csv1"</code>, in which case
a non-<code>NULL</code> value of <code>skip</code> is taken as the number of lines preceding
the columnar data. Specifying <code>skip</code> is usually only needed if <code><a href="#topic+read.met">read.met()</a></code>
cannot find a line starting with <code>"Date/Time"</code> (or a similar string).</p>
</td></tr>
<tr><td><code id="read.met_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.met_+3A_tz">tz</code></td>
<td>
<p>timezone assumed for time data.  This defaults to
<code>getOption("oceTz")</code>, which is very likely to be wrong.  In
a scientific context, where UTC is typically used for oceanographic
measurement, it makes sense to set <code>tz="UTC"</code>.  Note that these
data files do not contain timezone information, instead giving
data in Local Standard Time (LST).  Since LST differs from city
to city, users must make corrections to the time, as
shown in the &ldquo;Examples&rdquo;, which use data for
Halifax Nova Scotia, where LST is UTC-4.</p>
</td></tr>
<tr><td><code id="read.met_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  Set to 1 to get a moderate
amount of debugging information, or to 2 to get more.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+met-class">met</a> object.
</p>


<h3>Sample of Usage</h3>

<pre>
# Example 1: "csv1" Environment Canada format (found to be obsolete as of Oct 2019)
csv1 &lt;- read.met(system.file("extdata", "test_met_vsn1.csv", package="oce"))
csv1 &lt;- oceSetData(csv1, "time", csv1[["time"]]+4*3600,
    note="add 4h to local time to get UTC time")

# Example 2: "csv2" Environment Canada format (found to be obsolete as of Jan 2022)
csv2 &lt;- read.met(system.file("extdata", "test_met_vsn2.csv", package="oce"))
csv2 &lt;- oceSetData(csv2, "time", csv2[["time"]]+4*3600,
    note="add 4h to local time to get UTC time")

# Example 3: "csv3" Environment Canada format. Note timezone correction
csv3 &lt;- read.met(system.file("extdata", "test_met_vsn3.csv", package="oce"))
csv3 &lt;- oceSetData(csv3, "time", csv3[["time"]]+4*3600,
    note="add 4h to local time to get UTC time")

# Example 4: "xml2" format. (Uncertain timezone, so not corrected.)
if (requireNamespace("XML", quietly=TRUE))
    xml2 &lt;- read.met(system.file("extdata", "test_met_xml2.xml", package="oce"))

# Example 5: download and plot
library(oce)
# Recreate data(met) and plot u(t) and v(t)
metFile &lt;- download.met(id=6358, year=2003, month=9, destdir=".")
met &lt;- read.met(metFile)
met &lt;- oceSetData(met, "time", met[["time"]]+4*3600,
    note="add 4h to local time to get UTC time")
plot(met)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Environment Canada website for Historical Climate Data
<code style="white-space: pre;">&#8288;https://climate.weather.gc.ca/index_e.html&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to met data: 
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic+as.met">as.met</a>()</code>,
<code><a href="#topic+download.met">download.met</a>()</code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+summary+2Cmet-method">summary,met-method</a></code>
</p>

<hr>
<h2 id='read.netcdf'>Read a NetCDF File</h2><span id='topic+read.netcdf'></span>

<h3>Description</h3>

<p>Read a netcdf file, trying to interpret its contents sensibly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.netcdf(file, ..., encoding = NA, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.netcdf_+3A_file">file</code></td>
<td>
<p>the name of a file</p>
</td></tr>
<tr><td><code id="read.netcdf_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
<tr><td><code id="read.netcdf_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.netcdf_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is important to note that this is a preliminary version of
this function, and much about it may change without notice.
Indeed, it may be removed entirely.
</p>
<p>Below are some features that may be changed.
</p>

<ol>
<li><p> The names of data items are not changed from those in the netcdf
file on the assumption that this will offer the least surprise to
the user.
</p>
</li>
<li><p> An attempt is made to find some common metadata from global
attributes in the netcdf file. These attributes include
<code>Longitude</code>, <code>Latitude</code>, <code>Ship</code> and <code>Cruise</code>.
Before they are stored in the metadata, they are converted to
lower case, since that is the oce convention.
</p>
</li></ol>



<h3>Value</h3>

<p>An <a href="#topic+oce-class">oce</a> object.
</p>

<hr>
<h2 id='read.oce'>Read an Oceanographic Data File</h2><span id='topic+read.oce'></span>

<h3>Description</h3>

<p>Read an oceanographic data file, auto-discovering the file type from the
first line of the file.
This function tries to infer the file type from the first line, using
<code><a href="#topic+oceMagic">oceMagic()</a></code>.  If it can be discovered, then an
instrument-specific file reading function is called, with the <code>file</code>
and with any additional arguments being supplied.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.oce(file, ..., encoding = "latin1")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.oce_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.</p>
</td></tr>
<tr><td><code id="read.oce_+3A_...">...</code></td>
<td>
<p>arguments to be handed to whichever instrument-specific reading
function is selected, based on the header.</p>
</td></tr>
<tr><td><code id="read.oce_+3A_encoding">encoding</code></td>
<td>
<p>a character string giving the file encoding.  This defaults
to <code style="white-space: pre;">&#8288;"latin1&#8288;</code>&quot;, which seems to work for files available to the authors, but
be aware that a different setting may be required for files that contain
unusual accents or characters.  (Try <code>"UTF-8"</code> if the default produces
errors.) Note that <code>encoding</code> is ignored in binary files, and also
in some text-based files, as well.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+oce-class">oce</a> object of that is
specialized to the data type, e.g. <a href="#topic+ctd-class">ctd</a>,
if the data file contains <code>ctd</code> data.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The file type is determined by <code><a href="#topic+oceMagic">oceMagic()</a></code>.  If the file
type can be determined, then one of the following is called:
<code><a href="#topic+read.ctd">read.ctd()</a></code>, <code><a href="#topic+read.coastline">read.coastline()</a></code>
<code><a href="#topic+read.lobo">read.lobo()</a></code>, <code><a href="#topic+read.rsk">read.rsk()</a></code>,
<code><a href="#topic+read.sealevel">read.sealevel()</a></code>, etc.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
x &lt;- read.oce(system.file("extdata", "ctd.cnv.gz", package = "oce"))
plot(x) # summary with TS and profiles
plotTS(x) # just the TS
</code></pre>

<hr>
<h2 id='read.odf'>Read an odf File</h2><span id='topic+read.odf'></span>

<h3>Description</h3>

<p>ODF (Ocean Data Format) is a
format developed at the Bedford Institute of Oceanography and also used
at other Canadian Department of Fisheries and Oceans (DFO) facilities
(see references 1 and 2).
It can hold various types of time-series data, which includes a variety
of instrument types. Thus, <code><a href="#topic+read.odf">read.odf()</a></code>
is used by <code>read.ctd.odf</code> for CTD data, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.odf(
  file,
  columns = NULL,
  header = "list",
  exclude = NULL,
  encoding = "latin1",
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.odf_+3A_file">file</code></td>
<td>
<p>the file containing the data.</p>
</td></tr>
<tr><td><code id="read.odf_+3A_columns">columns</code></td>
<td>
<p>An optional <a href="base.html#topic+list">list</a> that can be used to convert unrecognized
data names to resultant variable names.  For example,
<code style="white-space: pre;">&#8288;columns=list(salinity=list(name="salt", unit=list(unit=expression(), scale="PSS-78"))&#8288;</code>
states that a short-name of <code>"salt"</code> represents salinity, and that the unit is
as indicated. This is passed to <code><a href="#topic+cnvName2oceName">cnvName2oceName()</a></code> or <code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames()</a></code>,
as appropriate, and takes precedence over the lookup table in that function.</p>
</td></tr>
<tr><td><code id="read.odf_+3A_header">header</code></td>
<td>
<p>An indication of whether, or how, to store the entire
ODF file header in the <code>metadata</code> slot of the returned object.
There are three choices for the <code>header</code> argument.
(1) If it is <code>NULL</code>, then the ODF header is not stored in
the <code>metadata</code> slot (although some of its contents are).
(2) If it is <code>"character"</code>, the header is stored within
the <code>metadata</code> as a vector named <code>header</code>, comprising
a character string for each line of the header within the ODF file.
(3) If it is <code>"list"</code>, then the <code>metadata</code> slot of the
returned object will contain a <code>list</code> named <code>header</code> that
has lists as its entries. (The sub-lists are in the form of
key-value pairs.) The naming of list entries is patterned on
that in the ODF header, except that <code><a href="#topic+unduplicateNames">unduplicateNames()</a></code>
is used to transform repeated names by adding numerical suffices.
Note: on June 6, 2019, the default value of <code>header</code> was
changed from <code>NULL</code> to <code>"list"</code>; in addition, the resultant
list was made to contain every single item in the ODF header, with
<code><a href="#topic+unduplicateNames">unduplicateNames()</a></code> being used to append integers to
distinguish between repeated names in the ODF format.</p>
</td></tr>
<tr><td><code id="read.odf_+3A_exclude">exclude</code></td>
<td>
<p>either a character value holding a regular
expression that is used with <code><a href="base.html#topic+grep">grep()</a></code> to remove lines from the
header before processing, or <code>NULL</code> (the default), meaning
not to exclude any such lines.  The purpose of this argument
is to solve problems with some files, which can have
thousands of lines that indicate details that are may be of
little value in processing.  For example, some files have thousands
of lines that would be excluded by using
<code>exclude="PROCESS='Nulled the .* value"</code> in the function call.</p>
</td></tr>
<tr><td><code id="read.odf_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.odf_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that some elements of the metadata are particular to ODF objects,
e.g. <code>depthMin</code>, <code>depthMax</code> and <code>sounding</code>, which
are inferred from ODF items named <code>MIN_DEPTH</code>, <code>MAX_DEPTH</code>
and <code>SOUNDING</code>, respectively. In addition, the more common metadata
item <code>waterDepth</code>, which is used in <code>ctd</code> objects to refer to
the total water depth, is set to <code>sounding</code> if that is finite,
or to <code>maxDepth</code> otherwise.
</p>
<p>The function <code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames()</a></code> is used to translate
data names from the ODF file to standard <code>oce</code> names.
</p>


<h3>Value</h3>

<p>An <a href="#topic+oce-class">oce</a> object.
</p>


<h3>Metadata conventions</h3>

<p>Some metadata items may be specific to certain instruments, and
certain research groups. It can be important for analysts to be aware of
the conventions used in datasets that are under study.
For example, as of June 2018, <code>adp</code>
objects created at the Bedford Institute of Oceanography may
have a metadata item named <code>depthOffBottom</code> (called
<code>DEPTH_OFF_BOTTOM</code> in ODF files), which is not typically
present in <code>ctd</code> files. This item illustrates the renaming
convention, from the CAMEL_CASE used in ODF files to the snakeCase
used in oce. Bearing this conversion in mind, users should not
find it difficult to understand the meaning of items that <code><a href="#topic+read.odf">read.odf()</a></code>
stores within the <code>metadata</code> slot. Users should bear in mind
that the whole ODF header is saved as a list by
calling the function with <code>header="list"</code>, after which
e.g. <a href="utils.html#topic+str">str</a><code>(rval[["header"]])</code> or <a href="utils.html#topic+View">View</a><code>(rval[["header"]])</code>
can be used to isolate any information of interest (but bear in mind
that suffices are used to disambiguate sibling items of identical
name in the ODF header).
</p>


<h3>Handling of temperature scales</h3>

<p><code>read.odf()</code> stores temperature data directly as read from the file, which
might mean the IPTS-68 scale.  These values should not be used to calculate
other seawater quantities, because formulae are generally based in ITS90
temperatures. To avoid problems, the accessor function converts to the modern
scale, e.g. <code>x[["temperature"]]</code> yields temperature in the ITS90
scale, whether temperatures in the original file were reported on that scale
or the older IPTS-68 scale.
</p>


<h3>Caution</h3>

<p>Lacking detailed documentation of the ODF file format, the <code><a href="#topic+read.odf">read.odf()</a></code> and
<code><a href="#topic+read.ctd.odf">read.ctd.odf()</a></code> functions were crafted based on inspection of data files, and
so some guesses had to be made.
</p>
<p>The <code>PARAMETER_HEADER</code> chunks describing quality-control flags are
a case in point.  These contain <code>NAME</code> components that
refer to other <code>PARAMETER_HEADER</code> chunks that hold measured data.  However,
those references are not always matched well with the data names, and
even if they do match, the cross-reference syntax used by
the Bedford Institute of Oceanography differs from that used by
l’Institut Maurice-Lamontagne. To simplify coding, it was assumed that
each quality-control sequence applies to the data sequence
immediately preceding it.  (This assumption is made in other
analysis systems.)
</p>
<p>It is also prudent to pay attention to the units decoding,
which <code><a href="#topic+read.odf">read.odf()</a></code> handles by calling <code><a href="#topic+unitFromString">unitFromString()</a></code>.
Be on the lookout for incorrect temperature scales, which
are sometimes reported with nonstandard strings in ODF files.
Also, note that you may see warnings about conductivity ratios,
which some ODF files incorrectly suggest have dimensions.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, with help from Chantelle Layton
</p>


<h3>References</h3>

<p>For sources that describe the ODF format, see the documentation
for the <a href="#topic+odf-class">odf</a> class.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ODF2oce">ODF2oce()</a></code> will be an alternative to this, once (or perhaps if) a <code>ODF</code>
package is released by the Canadian Department of Fisheries and Oceans.
</p>
<p>Other things related to odf data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ODF2oce">ODF2oce</a>()</code>,
<code><a href="#topic+ODFListFromHeader">ODFListFromHeader</a>()</code>,
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+summary+2Codf-method">summary,odf-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
#
# 1. Read a CTD cast made on the Scotian Shelf. Note that the file's metadata
# states that conductivity is in S/m, but it is really conductivity ratio,
# so we must alter the unit before converting to a CTD object. Note that
# read.odf() on this data file produces a warning suggesting that the user
# repair the unit, using the method outlined here.
odf &lt;- read.odf(system.file("extdata", "CTD_BCD2014666_008_1_DN.ODF.gz", package = "oce"))
ctd &lt;- as.ctd(odf) # so we can e.g. extract potential temperature
ctd[["conductivityUnit"]] &lt;- list(unit = expression(), scale = "")
#
# 2. Make a CTD, and plot (with span to show NS)
plot(ctd, span = 500)
#
# 3. Highlight bad data on TS diagram. (Note that the eos
# is specified, because we will extract practical-salinity and
# UNESCO-defined potential temperatures for the added points.)
plotTS(ctd, type = "o", eos = "unesco") # use a line to show loops
bad &lt;- ctd[["QCFlag"]] != 0
points(ctd[["salinity"]][bad], ctd[["theta"]][bad], col = "red", pch = 20)

</code></pre>

<hr>
<h2 id='read.rsk'>Read a rsk File</h2><span id='topic+read.rsk'></span>

<h3>Description</h3>

<p>Read an RBR rsk or txt file, e.g. as produced by an RBR logger, producing an
object of class <code>rsk</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.rsk(
  file,
  from = 1,
  to,
  by = 1,
  type,
  encoding = NA,
  tz = getOption("oceTz", default = "UTC"),
  tzOffsetLocation,
  patm = FALSE,
  allTables = TRUE,
  processingLog,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.rsk_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the RSK file to
load. Note that <code>file</code> must be a character string, because connections are
not used in that case, which is instead handled with database calls.</p>
</td></tr>
<tr><td><code id="read.rsk_+3A_from">from</code></td>
<td>
<p>indication of the first datum to read.  This can a positive integer
to indicate sequence number, the POSIX time of the first datum, or a character
string that can be converted to a POSIX time.  (For POSIX times, be careful
about the <code>tz</code> argument.)</p>
</td></tr>
<tr><td><code id="read.rsk_+3A_to">to</code></td>
<td>
<p>an indication of the last datum to be read, in the same format as
<code>from</code>.  If <code>to</code> is missing, data will be read to the end of the file.</p>
</td></tr>
<tr><td><code id="read.rsk_+3A_by">by</code></td>
<td>
<p>an indication of the stride length to use while walking through the
file.  If this is an integer, then <code>by-1</code> samples are skipped between each
pair of samples that is read.  If this is a string representing a time interval,
in colon-separated format (HH:MM:SS or MM:SS), then this interval is divided by
the sampling interval, to get the stride length.</p>
</td></tr>
<tr><td><code id="read.rsk_+3A_type">type</code></td>
<td>
<p>optional file type, presently can be <code>rsk</code> or <code>txt</code> (for a
text export of an RBR rsk or hex file). If this argument is not provided, an
attempt will be made to infer the type from the file name and contents.</p>
</td></tr>
<tr><td><code id="read.rsk_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.rsk_+3A_tz">tz</code></td>
<td>
<p>the timezone assumed for the time values stored in the
data file.  Unless the user has set an alternative value in the
<code style="white-space: pre;">&#8288;~/.Rprofile&#8288;</code> file, the default will be <code>"UTC"</code>; see the
&ldquo;Altering oce Defaults&rdquo; vignette for more on the use
of the <code style="white-space: pre;">&#8288;~/.Rprofile&#8288;</code> file.</p>
</td></tr>
<tr><td><code id="read.rsk_+3A_tzoffsetlocation">tzOffsetLocation</code></td>
<td>
<p>offset, in hours, between the CTD clock and
the clock in the controlling computer/tablet/phone (if one was used during
the sampling). This offset is required to relate location information from the
controller to hydrographic information from the CTD, using timestamps as an
index (see &quot;A note on location information&quot; in &ldquo;Details&rdquo;).
If the user supplies a value for <code>tzOffsetLocation</code>, then that is used.
If not, an attempt is made to infer it from an item named <code>UTCdelta</code>
that might be present within a table named <code>epochs</code> in the file. If no
value can be inferred from either of these two methods, then
<code>tzOffsetLocation</code> is set to zero.</p>
</td></tr>
<tr><td><code id="read.rsk_+3A_patm">patm</code></td>
<td>
<p>controls the handling of atmospheric pressure, an important issue
for RBR instruments that record absolute pressure; see &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="read.rsk_+3A_alltables">allTables</code></td>
<td>
<p>logical value, TRUE by default, indicating whether to save
all the non-empty tables in the database (pruned of <code>blob</code> columns) in the
<code>metadata</code> slot of the returned object. This may be useful for detailed
analysis.</p>
</td></tr>
<tr><td><code id="read.rsk_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
This is typically only provided for internal calls; the default that it provides
is better for normal calls by a user.</p>
</td></tr>
<tr><td><code id="read.rsk_+3A_debug">debug</code></td>
<td>
<p>a flag that can be set to <code>TRUE</code> to turn on debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This can read files produced by several RBR instruments.  At the moment, five
styles are understood: (1) text file produced as an export of an RBR <code>hex</code>
or <code>rsk</code> file; (2) text file with columns for temperature and pressure
(with sampling times indicated in the header); (3) text file with four columns,
in which the date the time of day are given in the first two columns, followed
by the temperature, and pressure; (4) text file with five columns, in which
depth in the water column is given after the pressure; (5) an SQLite-based
database format. The first four options are provided mainly for historical
reasons, since RBR instruments at the date of writing commonly use the SQLite
format, though the first option is common for all instruments that produce a
<code>hex</code> file that can be read using Ruskin.
Options 2-4 are mostly obsolete, and will be removed from future versions.
</p>
<p><strong>A note on location information.</strong>
It is possible to couple RBR CTD devices with smart phones or tablets
(see RBR-global, 2020), with the location data from the latter being stored
in the output <code>.rsk</code> file. The format does not seem to be documented by RBR,
but <code>read.rsk()</code> attempts to infer location nevertheless.
The procedure involves comparing the <code>tstamp</code> (time-stamp)
field from the hydrographic part of the dataset (which is in a database
table named <code>data</code>) with the <code>tstamp</code> field in the geographical part of the
dataset (in a table named <code>geodata</code>). (The <code>geodata</code> entries are filtered to
those for which the <code>origin</code> field equals <code>"auto"</code>. This seems to align
with times shown for &quot;LOCATION&quot; data in RBR-provided viewing software.)
The connection between the two fields is done with <code><a href="stats.html#topic+approx">approx()</a></code>, after
adding <code>tzOffsetLocation</code> (with units converted appropriately) to the
time inferred from <code>geodata$tstamp</code> field, to account for the fact that phones and tablets
may be set to local time. If the procedure succeeds, then <code>longitude</code> and
<code>latitude</code> are inserted into the <code>metadata</code> slot of the return value, in the
form of vectors with the same length as <code>pressure</code> in the <code>data</code> slot.
</p>
<p><strong>A note on conductivity.</strong>
RBR devices record conductivity in mS/cm, and it
is this value that is stored in the object returned by <code>read.rsk</code>. This can
be converted to conductivity ratio (which is what many other instruments report)
by dividing by 42.914 (see Culkin and Smith, 1980) which will be necessary in
any seawater-related function that takes conductivity ratio as an argument (see
&ldquo;Examples&rdquo;).
</p>
<p><strong>A note on pressure.</strong>
RBR devices tend to record absolute pressure (i.e.
sea pressure plus atmospheric pressure), unlike most oceanographic instruments
that record sea pressure (or an estimate thereof).  The handling of pressure
is controlled with the <code>patm</code> argument, for which there are three
possibilities.  (1) If <code>patm</code> is <code>FALSE</code> (the default), then
pressure read from the data file is stored in the <code>data</code> slot of return
value, and the <code>metadata</code> item <code>pressureType</code> is set to the string
<code>"absolute"</code>.  (2) If <code>patm</code> is <code>TRUE</code>, then an estimate of
atmospheric pressure is subtracted from the raw data. For data files in the
SQLite format (i.e.  <code style="white-space: pre;">&#8288;*.rsk&#8288;</code> files), this estimate will be the value read
from the file, or the &ldquo;standard atmosphere&rdquo; value 10.1325 dbar, if the file
lacks this information.  (3) If <code>patm</code> is a numerical value (or list of
values, one for each sampling time), then 'patm' is subtracted from the
raw data.  In cases 2 and 3, an additional column named
'pressureOriginal' is added to the 'data' slot to store the value
contained in the data file, and 'pressureType' is set to a string
starting with '&quot;sea&quot;'.  See <code><a href="#topic+as.ctd">as.ctd()</a></code> for details of how this
setup facilitates the conversion of <a href="#topic+rsk-class">rsk</a> objects to
<a href="#topic+ctd-class">ctd</a> objects.
</p>


<h3>Value</h3>

<p>An <a href="#topic+rsk-class">rsk</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>

<p>Culkin, F., and Norman D. Smith, 1980. Determination of the concentration of
potassium chloride solution having the same electrical conductivity, at 15 C and
infinite frequency, as standard seawater of salinity 35.0000 ppt (Chlorinity
19.37394 ppt). <em>IEEE Journal of Oceanic Engineering</em>, <strong>5</strong>, pp 22-23.
</p>
<p>RBR-global.com, 2020. &quot;Ruskin User Guide.&quot; RBR, August 18, 2020.
Revision RBR#0006105revH.
</p>


<h3>See Also</h3>

<p>The documentation for <a href="#topic+rsk-class">rsk</a> explains the structure of
<code>rsk</code> objects, and also outlines other functions dealing with them.  Since
RBR has a wide variety of instruments, <code>rsk</code> datasets can be quite general,
and it is common to coerce <code>rsk</code> objects to other forms for specialized
work, e.g. <code><a href="#topic+as.ctd">as.ctd()</a></code> can be used to create CTD object, so that the
generic plot obeys the CTD format.
</p>
<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic+as.rsk">as.rsk</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+rskPatm">rskPatm</a>()</code>,
<code><a href="#topic+rskToc">rskToc</a>()</code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method</a></code>
</p>

<hr>
<h2 id='read.sealevel'>Read a sealevel File</h2><span id='topic+read.sealevel'></span>

<h3>Description</h3>

<p>Read a data file holding sea level data.  BUG: the time vector assumes GMT,
regardless of the GMT.offset value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.sealevel(
  file,
  tz = getOption("oceTz"),
  encoding = "latin1",
  processingLog,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.sealevel_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file
to load.  See Details for the types of files that are recognized.</p>
</td></tr>
<tr><td><code id="read.sealevel_+3A_tz">tz</code></td>
<td>
<p>time zone.  The default value, <code>oceTz</code>, is set to <code>UTC</code>
at setup.  (If a time zone is present in the file header, this will
supercede the value given here.)</p>
</td></tr>
<tr><td><code id="read.sealevel_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.sealevel_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.
(Typically only provided for internal calls; the default that it provides is
better for normal calls by a user.)</p>
</td></tr>
<tr><td><code id="read.sealevel_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function starts by scanning the first line of the file, from which it
determines whether the file is in one of two known formats: type 1, the
format used at the Hawaii archive centre, and type 2, the
comma-separated-value format used by the Marine Environmental Data Service.
The file type is inferred by examination of its first line. If that contains
the string <code>Station_Name</code> the file is of type 2. If
the file is in neither of these formats, the user might wish to scan it
directly, and then to use <code><a href="#topic+as.sealevel">as.sealevel()</a></code> to create a
<code>sealevel</code> object.
The Hawaii archive site at
<code style="white-space: pre;">&#8288;http://ilikai.soest.hawaii.edu/uhslc/datai.html&#8288;</code> at one time provided a graphical
interface for downloading sealevel data in Type 1, with format that was once
described at <code style="white-space: pre;">&#8288;http://ilikai.soest.hawaii.edu/rqds/hourly.fmt&#8288;</code> (although that link
was observed to no longer work, on December 4, 2016).
Examination of data retrieved from what seems to be a replacement Hawaii server
(https://uhslc.soest.hawaii.edu/data/?rq) in September 2019 indicated that the
format had been changed to what is called Type 3 by <code>read.sealevel</code>.
Web searches did not uncover documentation on this format, so the
decoding scheme was developed solely through examination of
data files, which means that it might be not be correct.
The MEDS repository (<code>http://www.isdm-gdsi.gc.ca/isdm-gdsi/index-eng.html</code>)
provides Type 2 data.
</p>


<h3>Value</h3>

<p>A <a href="#topic+sealevel-class">sealevel</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to sealevel data: 
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic+as.sealevel">as.sealevel</a>()</code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+summary+2Csealevel-method">summary,sealevel-method</a></code>
</p>

<hr>
<h2 id='read.section'>Read a section File</h2><span id='topic+read.section'></span>

<h3>Description</h3>

<p>Read a file that contains a series of <code>ctd</code> profiles that make up an
oceanographic section.
Only <em>exchange BOT</em> comma-separated value format is permitted at this time,
but other formats may be added later.  It should also be noted that the parsing
scheme was developed after inspection of the A03 data set (see Examples). This
may cause problems if the format is not universal. For example, the header must
name the salinity column &quot;<code>CTDSAL</code>&quot;; if not, salinity values will not be
read from the file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.section(
  file,
  directory,
  sectionId = "",
  flags,
  ship = "",
  scientist = "",
  institute = "",
  missingValue = -999,
  encoding = "latin1",
  debug = getOption("oceDebug"),
  processingLog
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.section_+3A_file">file</code></td>
<td>
<p>A file containing a set of CTD observations.  At present, only the
<em>exchange BOT</em> format is accepted (see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="read.section_+3A_directory">directory</code></td>
<td>
<p>A character string indicating the name of a  directory that
contains a set of CTD files that hold individual stations in the section.</p>
</td></tr>
<tr><td><code id="read.section_+3A_sectionid">sectionId</code></td>
<td>
<p>Optional string indicating the name for the section.  If not
provided, the section ID is determined by examination of the file header.</p>
</td></tr>
<tr><td><code id="read.section_+3A_flags">flags</code></td>
<td>
<p>Ignored, and deprecated (will be disallowed in a future version).</p>
</td></tr>
<tr><td><code id="read.section_+3A_ship">ship</code></td>
<td>
<p>Name of the ship carrying out the sampling.</p>
</td></tr>
<tr><td><code id="read.section_+3A_scientist">scientist</code></td>
<td>
<p>Name of chief scientist aboard ship.</p>
</td></tr>
<tr><td><code id="read.section_+3A_institute">institute</code></td>
<td>
<p>Name of chief scientist's institute.</p>
</td></tr>
<tr><td><code id="read.section_+3A_missingvalue">missingValue</code></td>
<td>
<p>Numerical value used to indicate missing data.</p>
</td></tr>
<tr><td><code id="read.section_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.section_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
<tr><td><code id="read.section_+3A_processinglog">processingLog</code></td>
<td>
<p>If provided, the action item to be stored in the log.  This
is typically only provided for internal calls; the default that it provides is
better for normal calls by a user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+section-class">section</a> object.
</p>


<h3>Disambiguating salinity</h3>

<p>WOCE datasets commonly have a column named <code>CTDSAL</code> for salinity inferred
from a CTD and <code>SALNTY</code> (not a typo) for salinity derived from bottle data.
If only one of these is present in the data file, the data will be called
<code>salinity</code> in the <code>data</code> slot of the return value. However, if both
are present, then <code>CTDSAL</code> is stored as <code>salinity</code> and <code>SALNTY</code>
is stored as <code>salinityBottle</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Several repository sites provide section data. A reasonably stable example is
<code style="white-space: pre;">&#8288;https://cchdo.ucsd.edu&#8288;</code>, but a search on <code>"WOCE bottle data"</code> should
turn up other sites, if this ceases to exist. Only
the so-called <em>exchange BOT</em> data format can be processed by <code><a href="#topic+read.section">read.section()</a></code>
at this time. Data names are inferred from column headings using
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames()</a></code>.
</p>


<h3>See Also</h3>

<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>

<hr>
<h2 id='read.topo'>Read a topo File</h2><span id='topic+read.topo'></span>

<h3>Description</h3>

<p>Read a file that contains topographic data in the ETOPO dataset, as was once provided by
the NOAA website (see <code><a href="#topic+download.topo">download.topo()</a></code> for a good server for such
files. (As of May, 2020, there does not seem to be a way to download these
files from the NOAA website.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.topo(file, encoding = "latin1", debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.topo_+3A_file">file</code></td>
<td>
<p>Name of a file containing an ETOPO-format dataset. Three
types are permitted; see &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="read.topo_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
<tr><td><code id="read.topo_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The three permitted file types are as follows.
</p>

<ol>
<li><p> An ascii type
in which line 1 holds a label (which is ignored), whitespace, and then
the number of columns in the matrix (i.e. the number of longitude values),
line 2 is similar but for latitude, line 3 is similar but for the westernmost
longitude, line 4 is similar but for southernmost latitude, line 5
is similar but for cell size, and lines after that hold the grid.
</p>
</li>
<li><p> A NetCDF format that was once described by NOAA as &quot;GMT NetCDF&quot;.
</p>
</li>
<li><p> A NetCDF format that was once described by NOAA as &quot;NetCDF&quot;.
</p>
</li></ol>



<h3>Value</h3>

<p>A <a href="#topic+topo-class">topo</a> object.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
topoMaritimes &lt;- read.topo("topoMaritimes.asc")
plot(topographyMaritimes)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to topo data: 
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic+as.topo">as.topo</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+summary+2Ctopo-method">summary,topo-method</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+topoInterpolate">topoInterpolate</a>()</code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>
</p>

<hr>
<h2 id='read.woa'>Read a World Ocean Atlas NetCDF File</h2><span id='topic+read.woa'></span>

<h3>Description</h3>

<p>Read a World Ocean Atlas NetCDF File
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.woa(file, name, positive = FALSE, encoding = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.woa_+3A_file">file</code></td>
<td>
<p>character string naming the file</p>
</td></tr>
<tr><td><code id="read.woa_+3A_name">name</code></td>
<td>
<p>of variable to extract. If not provided, an
error message is issued that lists the names of data in the file.</p>
</td></tr>
<tr><td><code id="read.woa_+3A_positive">positive</code></td>
<td>
<p>logical value indicating whether <code>longitude</code> should
be converted to be in the range from 0 to 360, with <code>name</code>
being shuffled accordingly. This is set to <code>FALSE</code> by default,
because the usual oce convention is for longitude to range between -180
to +180.</p>
</td></tr>
<tr><td><code id="read.woa_+3A_encoding">encoding</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing vectors <code>longitude</code>, <code>latitude</code>,
<code>depth</code>, and an array with the specified name. If <code>positive</code>
is true, then <code>longitude</code> will be converted to range from 0
to 360, and the array will be shuffled accordingly.
</p>


<h3>Sample of Usage</h3>

<pre>
# Mean SST at 5-degree spatial resolution
tmn &lt;- read.woa("~/data/woa13/woa13_decav_t00_5dv2.nc", "t_mn")
imagep(tmn$longitude, tmn$latitude, tmn$t_mn[, , 1], zlab="SST")
</pre>

<hr>
<h2 id='read.xbt'>Read an xbt file</h2><span id='topic+read.xbt'></span>

<h3>Description</h3>

<p>Two file types are handled: (1) the <code>"sippican"</code> format, used
for Sippican XBT files, handled with <code><a href="#topic+read.xbt.edf">read.xbt.edf()</a></code>, and (2)
the <code>"noaa1"</code> format, handled with <code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1()</a></code>. The first of
these is recognized by <code><a href="#topic+read.oce">read.oce()</a></code>, but the second must be called
directly with <code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.xbt(
  file,
  type = "sippican",
  longitude = NA,
  latitude = NA,
  encoding = "latin1",
  debug = getOption("oceDebug"),
  processingLog
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.xbt_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file to
load.</p>
</td></tr>
<tr><td><code id="read.xbt_+3A_type">type</code></td>
<td>
<p>character string indicating type of file, with valid choices being
<code>"sippican"</code> and <code>"noaa1"</code>.</p>
</td></tr>
<tr><td><code id="read.xbt_+3A_longitude">longitude</code>, <code id="read.xbt_+3A_latitude">latitude</code></td>
<td>
<p>optional signed numbers indicating the longitude in degrees
East and latitude in degrees North. These values are used if <code>type="sippican"</code>,
but ignored if <code>type="noaa1"</code>, because those files contain location information.</p>
</td></tr>
<tr><td><code id="read.xbt_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.xbt_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  The value indicates the depth
within the call stack to which debugging applies.</p>
</td></tr>
<tr><td><code id="read.xbt_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.  This
parameter is typically only provided for internal calls; the default that it
provides is better for normal calls by a user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+xbt-class">xbt</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Sippican, Inc. &quot;Bathythermograph Data Acquisition System: Installation, Operation and Maintenance
Manual (P/N 308195, Rev. A),&quot; 2003.
https://pages.uoregon.edu/drt/MGL0910_Science_Report/attachments/MK21_ISA_Manual_Rev_A.pdf.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to xbt data: 
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>,
<code><a href="#topic+as.xbt">as.xbt</a>()</code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1</a>()</code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>,
<code><a href="#topic+summary+2Cxbt-method">summary,xbt-method</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
xbt &lt;- read.oce(system.file("extdata", "xbt.edf", package = "oce"))
summary(xbt)
plot(xbt)

</code></pre>

<hr>
<h2 id='read.xbt.edf'>Read an xbt File in Sippican Format</h2><span id='topic+read.xbt.edf'></span>

<h3>Description</h3>

<p>The function was written by inspection of a particular file, and might
be wrong for other files; see &ldquo;Details&rdquo; for a note on character
translation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.xbt.edf(
  file,
  longitude = NA,
  latitude = NA,
  encoding = "latin1",
  debug = getOption("oceDebug"),
  processingLog
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.xbt.edf_+3A_file">file</code></td>
<td>
<p>a connection or a character string giving the name of the file to
load.</p>
</td></tr>
<tr><td><code id="read.xbt.edf_+3A_longitude">longitude</code></td>
<td>
<p>optional signed number indicating the longitude in degrees
East.</p>
</td></tr>
<tr><td><code id="read.xbt.edf_+3A_latitude">latitude</code></td>
<td>
<p>optional signed number indicating the latitude in degrees North.</p>
</td></tr>
<tr><td><code id="read.xbt.edf_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.xbt.edf_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  The value indicates the depth
within the call stack to which debugging applies.</p>
</td></tr>
<tr><td><code id="read.xbt.edf_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.  This
parameter is typically only provided for internal calls; the default that it
provides is better for normal calls by a user.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The header is converted to ASCII format prior to storage in
the <code>metadata</code> slot, so that e.g. a degree sign in the original file will
become a <code style="white-space: pre;">&#8288;?&#8288;</code> character in the <code>header</code>.  This is to prevent problems
with submission of <code>oce</code> to the CRAN system, which produces NOTEs
about UTF-8 strings in data (on some build machines, evidently depending
on the locale on those machines).  This character substitution
is at odds with the <code>oce</code> philosophy of leaving data intact, so
it will be reverted, if CRAN policy changes or if the developers
can find a way to otherwise silence the NOTE.
</p>


<h3>Value</h3>

<p>An <a href="#topic+xbt-class">xbt</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
xbt &lt;- read.oce(system.file("extdata", "xbt.edf", package = "oce"))
summary(xbt)
plot(xbt)

</code></pre>

<hr>
<h2 id='read.xbt.noaa1'>Read an xbt File in NOAA Format</h2><span id='topic+read.xbt.noaa1'></span>

<h3>Description</h3>

<p>This file format, described at <code>https://www.aoml.noaa.gov/phod/dhos/axbt.php</code>, contains a header
line, followed by data lines.  For example, a particular file at this site has first
three lines as follows.
</p>
<div class="sourceCode"><pre>181.589 20100709 140820  -85.336  25.290 N42RF GL10 14    2010-190-15:49:18
  -0.0 27.52 -9.99
  -1.5 27.52 -9.99
</pre></div>
<p>where the items on the header line are (1) a year-day (ignored here), (2)
YYYYMMDD, (3) HHMMSS, (4) longitude, (5) latitude, (6) aircraft wing
number, (7) project name, (8) AXBT channel and (9) AXBT ID.  The other lines hold vertical
coordinate in metres, temperature and temperature error; -9.99 is a missing-value
code.  (This formatting information is extracted from a file named <code>readme.axbt</code> that
is provided with the data.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.xbt.noaa1(
  file,
  debug = getOption("oceDebug"),
  missingValue = -9.99,
  encoding = "latin1",
  processingLog
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.xbt.noaa1_+3A_file">file</code></td>
<td>
<p>character value naming a file, or a file connection, containing the data.</p>
</td></tr>
<tr><td><code id="read.xbt.noaa1_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.  The value indicates the depth
within the call stack to which debugging applies.</p>
</td></tr>
<tr><td><code id="read.xbt.noaa1_+3A_missingvalue">missingValue</code></td>
<td>
<p>numerical value that is to be interpreted as <code>NA</code></p>
</td></tr>
<tr><td><code id="read.xbt.noaa1_+3A_encoding">encoding</code></td>
<td>
<p>a character value that indicates the encoding to be used for
this data file, if it is textual.  The default value for most functions is
<code>"latin1"</code>, which seems to be suitable for files containing text written in
English and French.</p>
</td></tr>
<tr><td><code id="read.xbt.noaa1_+3A_processinglog">processingLog</code></td>
<td>
<p>if provided, the action item to be stored in the log.  This
parameter is typically only provided for internal calls; the default that it
provides is better for normal calls by a user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+xbt-class">xbt</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to xbt data: 
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>,
<code><a href="#topic+as.xbt">as.xbt</a>()</code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+read.xbt">read.xbt</a>()</code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>,
<code><a href="#topic+summary+2Cxbt-method">summary,xbt-method</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>

<hr>
<h2 id='rescale'>Rescale Values to lie in a Given Range</h2><span id='topic+rescale'></span>

<h3>Description</h3>

<p>This is helpful in e.g. developing a color scale for an image plot.  It is
not necessary that <code>rlow</code> be less than <code>rhigh</code>, and in fact
reversing them is a good way to get a reversed color scale for a plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rescale(x, xlow, xhigh, rlow = 0, rhigh = 1, clip = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rescale_+3A_x">x</code></td>
<td>
<p>a numeric vector.</p>
</td></tr>
<tr><td><code id="rescale_+3A_xlow">xlow</code></td>
<td>
<p><code>x</code> value to correspond to <code>rlow</code>.  If not given, it
will be calculated as the minimum value of <code>x</code></p>
</td></tr>
<tr><td><code id="rescale_+3A_xhigh">xhigh</code></td>
<td>
<p><code>x</code> value to correspond to <code>rhigh</code>.  If not given, it
will be calculated as the maximum value of <code>x</code></p>
</td></tr>
<tr><td><code id="rescale_+3A_rlow">rlow</code></td>
<td>
<p>value of the result corresponding to <code>x</code> equal to
<code>xlow</code>.</p>
</td></tr>
<tr><td><code id="rescale_+3A_rhigh">rhigh</code></td>
<td>
<p>value of the result corresponding to <code>x</code> equal to
<code>xhigh</code>.</p>
</td></tr>
<tr><td><code id="rescale_+3A_clip">clip</code></td>
<td>
<p>logical, set to <code>TRUE</code> to clip the result to the range
spanned by <code>rlow</code> and <code>rhigh</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new vector, which has minimum <code>lim[1]</code> and maximum <code>lim[2]</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Fake tow-yow data
t &lt;- seq(0, 600, 5)
x &lt;- 0.5 * t
z &lt;- 50 * (-1 + sin(2 * pi * t / 360))
T &lt;- 5 + 10 * exp(z / 100)
palette &lt;- oce.colorsViridis(100)
zlim &lt;- range(T)
drawPalette(zlim = zlim, col = palette)
plot(x, z,
    type = "p", pch = 20, cex = 3,
    col = palette[rescale(T, xlow = zlim[1], xhigh = zlim[2], rlow = 1, rhigh = 100)]
)
</code></pre>

<hr>
<h2 id='resizableLabel'>Variable Names in Adjustable Sizes</h2><span id='topic+resizableLabel'></span>

<h3>Description</h3>

<p>Provide axis names in adjustable sizes, e.g. using T instead of Temperature,
and including units as appropriate.
Used by e.g. <code><a href="#topic+plot+2Cctd-method">plot,ctd-method()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resizableLabel(
  item,
  axis = "x",
  sep,
  unit = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="resizableLabel_+3A_item">item</code></td>
<td>
<p>code for the label. The following common values are recognized:
<code>"absolute salinity"</code>, <code>"along-spine distance km"</code>, <code>"along-track distance km"</code>,
<code>"C"</code>, <code>"conductivity mS/cm"</code>, <code>"conductivity S/m"</code>, <code>"conservative temperature"</code>,
<code>"CT"</code>, <code>"depth"</code>, <code>"direction"</code>, <code>"distance"</code>, <code>"distance km"</code>, <code>"eastward"</code>,
<code>"elevation"</code>, <code>"fluorescence"</code>, <code>"frequency cph"</code>, <code>"heading"</code>, <code>"latitude"</code>,
<code>"longitude"</code>, <code>"N2"</code>, <code>"nitrate"</code>, <code>"nitrite"</code>, <code>"northward"</code>, <code>"oxygen"</code>,
<code>"oxygen mL/L"</code>, <code>"oxygen saturation"</code>, <code>"oxygen umol/kg"</code>, <code>"oxygen umol/L"</code>,
<code>"p"</code>, <code>"phosphate"</code>, <code>"pitch"</code>, <code>"roll"</code>, <code>"S"</code>, <code>"SA"</code>,
<code>"sigma0"</code>, <code>"sigma1"</code>, <code>"sigma2"</code>, <code>"sigma3"</code>, <code>"sigma4"</code>,
<code>"sigmaTheta"</code>,
<code>"silicate"</code>, <code>"sound speed"</code>, <code>"spectral density m2/cph"</code>, <code>"speed"</code>,
<code>"spice"</code>, <code>"T"</code>, <code>"theta"</code>, <code>"tritium"</code>, <code>"u"</code>, <code>"v"</code>, <code>"w"</code>, or <code>"z"</code>.
Other values may also be recognized, and if an unrecognized item is
given, then it is returned, unaltered.</p>
</td></tr>
<tr><td><code id="resizableLabel_+3A_axis">axis</code></td>
<td>
<p>a string indicating which axis to use; must be <code>x</code> or
<code>y</code>.</p>
</td></tr>
<tr><td><code id="resizableLabel_+3A_sep">sep</code></td>
<td>
<p>optional character string inserted between the unit and the
parentheses or brackets that enclose it. If not provided, then
<code><a href="base.html#topic+getOption">getOption</a></code><code>("oceUnitSep")</code> is checked. If that exists, then it is
used as the separator; if not, no separator is used.</p>
</td></tr>
<tr><td><code id="resizableLabel_+3A_unit">unit</code></td>
<td>
<p>optional unit to use, if the default is not satisfactory. This
might be the case if for example temperature was not measured in Celcius.</p>
</td></tr>
<tr><td><code id="resizableLabel_+3A_debug">debug</code></td>
<td>
<p>optional debugging flag. Setting to 0 turns off debugging,
while setting to 1 causes some debugging information to be printed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string or expression, in either a long or a shorter
format, for use in the indicated axis at the present plot size.  Whether the
unit is enclosed in parentheses or square brackets is determined by the
value of <code>getOption("oceUnitBracket")</code>, which may be <code>"["</code> or
<code>"("</code>.  Whether spaces are used between the unit and these deliminators
is set by <code>psep</code> or <code><a href="base.html#topic+getOption">getOption</a></code><code>("oceUnitSep")</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that create labels: 
<code><a href="#topic+labelWithUnit">labelWithUnit</a>()</code>
</p>

<hr>
<h2 id='retime'>Adjust The Time Within an oce Object</h2><span id='topic+retime'></span>

<h3>Description</h3>

<p>This function compensates for drifting instrument clocks, according to
<code class="reqn">t'=t + a + b (t-t0)</code>, where <code class="reqn">t'</code> is the
true time and <code class="reqn">t</code> is the time stored in the object.  A single check
on time mismatch can be described by a simple time offset, with a non-zero
value of <code>a</code>, a zero value of <code>b</code>, and an arbitrary value of
<code>t0</code>.  Checking the mismatch before and after an experiment yields
sufficient information to specify a linear drift, via <code>a</code>, <code>b</code>,
and <code>t0</code>.  Note that <code>t0</code> is just a convenience parameter, which
avoids the user having to know the &quot;zero time&quot; used in R and clarifies the
values of the other two parameters.  It makes sense for <code>t0</code> to have
the same timezone as the time within <code>x</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retime(x, a, b, t0, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="retime_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="retime_+3A_a">a</code></td>
<td>
<p>intercept (in seconds) in linear model of time drift (see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="retime_+3A_b">b</code></td>
<td>
<p>slope (unitless) in linear model of time drift (unitless) (see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="retime_+3A_t0">t0</code></td>
<td>
<p>reference time (in <code><a href="base.html#topic+POSIXct">POSIXct()</a></code> format) used in linear model of time
drift (see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="retime_+3A_debug">debug</code></td>
<td>
<p>a flag that, if nonzero, turns on debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The returned object is computed by linear interpolation, using
<code><a href="stats.html#topic+approx">approx()</a></code> with <code>rule=2</code>, to avoid <code>NA</code> values at the
start or end.  The new time will be as given by the formula above. Note that
if the drift is large enough, the sampling rate will be changed.  It is a
good idea to start with an object that has an extended time range, so that,
after this is called, <code><a href="base.html#topic+subset">subset()</a></code> can be used to trim to a desired
time range.
</p>


<h3>Value</h3>

<p>A new object, with time and other data adjusted.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adv)
adv2 &lt;- retime(adv, 0, 1e-4, as.POSIXct("2008-07-01 00:00:00", tz = "UTC"))
plot(adv[["time"]], adv2[["time"]] - adv[["time"]], type = "l")
</code></pre>

<hr>
<h2 id='rotateAboutZ'>Rotate Velocity Components Within an oce Object</h2><span id='topic+rotateAboutZ'></span>

<h3>Description</h3>

<p>Alter the horizontal components of velocities in <code>adp</code>,
<code>adv</code> or <code>cm</code> objects, by applying a rotation about
the vertical axis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rotateAboutZ(x, angle)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rotateAboutZ_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a>, <a href="#topic+adv-class">adv</a>, or <a href="#topic+cm-class">cm</a> object.</p>
</td></tr>
<tr><td><code id="rotateAboutZ_+3A_angle">angle</code></td>
<td>
<p>The rotation angle about the z axis, in degrees counterclockwise.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to cm data: 
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+as.cm">as.cm</a>()</code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+read.cm">read.cm</a>()</code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+summary+2Ccm-method">summary,cm-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
par(mfcol = c(2, 3))
# adp (acoustic Doppler profiler)
data(adp)
plot(adp, which = "uv")
mtext("adp", side = 3, line = 0, adj = 1, cex = 0.7)
adpRotated &lt;- rotateAboutZ(adp, 30)
plot(adpRotated, which = "uv")
mtext("adp rotated 30 deg", side = 3, line = 0, adj = 1, cex = 0.7)
# adv (acoustic Doppler velocimeter)
data(adv)
plot(adv, which = "uv")
mtext("adv", side = 3, line = 0, adj = 1, cex = 0.7)
advRotated &lt;- rotateAboutZ(adv, 125)
plot(advRotated, which = "uv")
mtext("adv rotated 125 deg", side = 3, line = 0, adj = 1, cex = 0.7)
# cm (current meter)
data(cm)
plot(cm, which = "uv")
mtext("cm", side = 3, line = 0, adj = 1, cex = 0.7)
cmRotated &lt;- rotateAboutZ(cm, 30)
plot(cmRotated, which = "uv")
mtext("cm rotated 30 deg", side = 3, line = 0, adj = 1, cex = 0.7)

</code></pre>

<hr>
<h2 id='rsk'>Sample rsk Data</h2><span id='topic+rsk'></span>

<h3>Description</h3>

<p>A sample <code>rsk</code> object derived from a Concerto CTD manufactured by RBR Ltd.
</p>


<h3>Details</h3>

<p>The data were obtained September 2015, off the west coast
of Greenland, by Matt Rutherford and Nicole Trenholm of the
Ocean Research Project, in collaboration with RBR and with the
NASA Oceans Melting Greenland project. The <code>rsk</code> object was
created with <code><a href="#topic+read.rsk">read.rsk()</a></code> with <code>allTables=FALSE</code>, after which
some metadata were added and the samples were trimmed to
just the downcast portion.
</p>


<h3>References</h3>

<p><code style="white-space: pre;">&#8288;https://rbr-global.com/&#8288;</code>
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic+as.rsk">as.rsk</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+read.rsk">read.rsk</a>()</code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+rskPatm">rskPatm</a>()</code>,
<code><a href="#topic+rskToc">rskToc</a>()</code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(rsk)
# The object doesn't "know" it is CTD until told so
plot(rsk)
plot(as.ctd(rsk))

</code></pre>

<hr>
<h2 id='rsk-class'>Class to Store Rsk Data</h2><span id='topic+rsk-class'></span>

<h3>Description</h3>

<p>This class stores Ruskin data, from RBR (see reference 1).
</p>


<h3>Details</h3>

<p>A <a href="#topic+rsk-class">rsk</a> object may be read with <code><a href="#topic+read.rsk">read.rsk()</a></code> or created with
<code><a href="#topic+as.rsk">as.rsk()</a></code>, but the former method is much preferred because it
retains the entirety of the information in the file.
Plots can be made with <code><a href="#topic+plot+2Crsk-method">plot,rsk-method()</a></code>, while
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method()</a></code> produces statistical summaries and <code>show</code>
produces overviews.   If atmospheric pressure has not been removed from the
data, <code><a href="#topic+rskPatm">rskPatm()</a></code> may provide guidance as to its value,
but this is no equal to record-keeping at sea.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>rsk</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>rsk</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>rsk</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+rsk-class">rsk</a> objects (see <code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+rsk-class">rsk</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>


<ol>
<li><p> RBR website (https://www.rbr-global.com/products)
</p>
</li></ol>



<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic+as.rsk">as.rsk</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+read.rsk">read.rsk</a>()</code>,
<code><a href="#topic+rskPatm">rskPatm</a>()</code>,
<code><a href="#topic+rskToc">rskToc</a>()</code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method</a></code>
</p>

<hr>
<h2 id='rsk2ctd'>Create a ctd Object from an rsk Object</h2><span id='topic+rsk2ctd'></span>

<h3>Description</h3>

<p>A new <code>ctd</code> object is assembled from the contents of the <code>rsk</code> object.
The data and metadata are mostly unchanged, with an important exception: the
<code>pressure</code> item in the <code>data</code> slot may altered, because <code>rsk</code>
instruments measure total pressure, not sea pressure; see &ldquo;Details&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rsk2ctd(
  x,
  pressureAtmospheric = 0,
  longitude = NULL,
  latitude = NULL,
  ship = NULL,
  cruise = NULL,
  station = NULL,
  deploymentType = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rsk2ctd_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+rsk-class">rsk</a> object.</p>
</td></tr>
<tr><td><code id="rsk2ctd_+3A_pressureatmospheric">pressureAtmospheric</code></td>
<td>
<p>A numerical value (a constant or a vector),
that is subtracted from the pressure in <code>object</code> before storing it in the return value.</p>
</td></tr>
<tr><td><code id="rsk2ctd_+3A_longitude">longitude</code></td>
<td>
<p>numerical value of longitude, in degrees East.</p>
</td></tr>
<tr><td><code id="rsk2ctd_+3A_latitude">latitude</code></td>
<td>
<p>numerical value of latitude, in degrees North.</p>
</td></tr>
<tr><td><code id="rsk2ctd_+3A_ship">ship</code></td>
<td>
<p>optional string containing the ship from which the observations were made.</p>
</td></tr>
<tr><td><code id="rsk2ctd_+3A_cruise">cruise</code></td>
<td>
<p>optional string containing a cruise identifier.</p>
</td></tr>
<tr><td><code id="rsk2ctd_+3A_station">station</code></td>
<td>
<p>optional string containing a station identifier.</p>
</td></tr>
<tr><td><code id="rsk2ctd_+3A_deploymenttype">deploymentType</code></td>
<td>
<p>character string indicating the type of deployment (see
<code><a href="#topic+as.ctd">as.ctd()</a></code>).</p>
</td></tr>
<tr><td><code id="rsk2ctd_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>pressureType</code> element of the
<code>metadata</code> of <code>rsk</code> objects defines the pressure type, and this controls
how pressure is set up in the returned object. If <code>object@metadata$pressureType</code>
is <code>"absolute"</code> (or <code>NULL</code>) then the resultant pressure will be adjusted
to make it into <code>"sea"</code> pressure. To do this, the value of
<code>object@metadata$pressureAtmospheric</code> is inspected. If this is present, then
it is subtracted from <code>pressure</code>. If this is missing, then
standard pressure (10.1325 dbar) will be subtracted. At this stage, the
pressure should be near zero at the ocean surface, but some additional adjustment
might be necessary, and this may be indicated by setting the argument <code>pressureAtmospheric</code> to
a non-zero value to be subtracted from pressure.
</p>

<hr>
<h2 id='rskPatm'>Estimate Atmospheric Pressure in an rsk Object</h2><span id='topic+rskPatm'></span>

<h3>Description</h3>

<p>Estimate atmospheric pressure in an <a href="#topic+rsk-class">rsk</a> object.  Pressures must be in
decibars for this to work.  First, a subset of pressures is created, in which
the range is <code>sap-dp</code> to <code>sap+dp</code>.  Here, <code>sap</code>=10.1325 dbar is standard
sealevel atmospheric pressure.  Within this window, three measures of central
tendency are calculated: the median, the mean, and a weighted mean that has
weight given by
<code class="reqn">exp(-2*((p-sap)/dp)^2)</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rskPatm(x, dp = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rskPatm_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+rsk-class">rsk</a> object.</p>
</td></tr>
<tr><td><code id="rskPatm_+3A_dp">dp</code></td>
<td>
<p>Half-width of pressure window to be examined (in decibars).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of four estimates: <code>sap</code>, the median, the mean, and the weighted mean.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for <a href="#topic+rsk-class">rsk</a> explains the structure of
<code>rsk</code> objects, and also outlines the other functions dealing with them.
</p>
<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic+as.rsk">as.rsk</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+read.rsk">read.rsk</a>()</code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+rskToc">rskToc</a>()</code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(rsk)
print(rskPatm(rsk))

</code></pre>

<hr>
<h2 id='rskToc'>Decode Table-of-Contents From an rsk File</h2><span id='topic+rskToc'></span>

<h3>Description</h3>

<p>Decode table-of-contents file from a rsk file, of the sort used by some
researchers at Dalhousie University.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rskToc(dir, from, to, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rskToc_+3A_dir">dir</code></td>
<td>
<p>name of a directory containing a single table-of-contents file, with
<code>.TBL</code> at the end of its file name.</p>
</td></tr>
<tr><td><code id="rskToc_+3A_from">from</code></td>
<td>
<p>optional <code><a href="base.html#topic+POSIXct">POSIXct()</a></code> time, indicating the beginning of a
data interval of interest.  This must have timezone <code>"UTC"</code>.</p>
</td></tr>
<tr><td><code id="rskToc_+3A_to">to</code></td>
<td>
<p>optional <code><a href="base.html#topic+POSIXct">POSIXct()</a></code> time, indicating the end of a data
interval of interest.  This must have timezone <code>"UTC"</code>.</p>
</td></tr>
<tr><td><code id="rskToc_+3A_debug">debug</code></td>
<td>
<p>optional integer to control debugging, with positive values
indicating to print information about the processing.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is assumed that the <code>.TBL</code> file contains lines of the form <code>"File
  \day179\SL08A179.023 started at Fri Jun 27 22:00:00 2008"</code> The first step is
to parse these lines to get day and hour information, i.e.  179 and 023 in the
line above.  Then, recognizing that it is common to change the names of such
files, the rest of the file-name information in the line is ignored, and instead
a new file name is constructed based on the data files that are found in the
directory.  (In other words, the <code>"\\day179\\SL08A"</code> portion of the line is
replaced.)  Once the file list is complete, with all times put into R format,
then (optionally) the list is trimmed to the time interval indicated by
<code>from</code> and <code>to</code>.  It is important that <code>from</code> and <code>to</code> be in
the <code>UTC</code> time zone, because that time zone is used in decoding the lines
in the <code>.TBL</code> file.
</p>


<h3>Value</h3>

<p>A list with two elements: <code>filename</code>, a vector of file names, and
<code>startTime</code>, a vector of <code><a href="base.html#topic+POSIXct">POSIXct()</a></code> times indicating the (real)
times of the first datum in the corresponding files.
</p>


<h3>Sample of Usage</h3>

<pre>
file &lt;- "~/data/archive/sleiwex/2008/moorings/m05/adv/sontek_202h/raw"
table &lt;- rskToc(file,
    from=as.POSIXct("2008-07-01 00:00:00", tz="UTC"),
    to=as.POSIXct("2008-07-01 12:00:00", tz="UTC"))
print(table)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic+as.rsk">as.rsk</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+read.rsk">read.rsk</a>()</code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+rskPatm">rskPatm</a>()</code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method</a></code>
</p>

<hr>
<h2 id='runlm'>Calculate Running Linear Models</h2><span id='topic+runlm'></span>

<h3>Description</h3>

<p>The linear model is calculated from the slope of a localized least-squares
regression model y=y(x).  The localization is defined by the x difference
from the point in question, with data at distance exceeding L/2 being
ignored.  With a <code>boxcar</code> window, all data within the local domain are
treated equally, while with a <code>hanning</code> window, a raised-cosine
weighting function is used; the latter produces smoother derivatives, which
can be useful for noisy data.  The function is based on internal
calculation, not on <code><a href="stats.html#topic+lm">lm()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runlm(x, y, xout, window = c("hanning", "boxcar"), L, deriv)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runlm_+3A_x">x</code></td>
<td>
<p>a vector holding x values.</p>
</td></tr>
<tr><td><code id="runlm_+3A_y">y</code></td>
<td>
<p>a vector holding y values.</p>
</td></tr>
<tr><td><code id="runlm_+3A_xout">xout</code></td>
<td>
<p>optional vector of x values at which the derivative is to be
found.  If not provided, <code>x</code> is used.</p>
</td></tr>
<tr><td><code id="runlm_+3A_window">window</code></td>
<td>
<p>type of weighting function used to weight data within the
window; see &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="runlm_+3A_l">L</code></td>
<td>
<p>width of running window, in x units.  If not provided, a reasonable
default will be used.</p>
</td></tr>
<tr><td><code id="runlm_+3A_deriv">deriv</code></td>
<td>
<p>an optional indicator of the desired return value; see
&ldquo;Examples&rdquo;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>deriv</code> is not specified, a list containing vectors of
output values <code>y</code> and <code>y</code>, derivative (<code>dydx</code>), along with
the scalar length scale <code>L</code>.  If <code>deriv=0</code>, a vector of values is
returned, and if <code>deriv=1</code>, a vector of derivatives is returned.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)

# Case 1: smooth a noisy signal
x &lt;- 1:100
y &lt;- 1 + x / 100 + sin(x / 5)
yn &lt;- y + rnorm(100, sd = 0.1)
L &lt;- 4
calc &lt;- runlm(x, y, L = L)
plot(x, y, type = "l", lwd = 7, col = "gray")
points(x, yn, pch = 20, col = "blue")
lines(x, calc$y, lwd = 2, col = "red")

# Case 2: square of buoyancy frequency
data(ctd)
par(mfrow = c(1, 1))
plot(ctd, which = "N2")
rho &lt;- swRho(ctd)
z &lt;- swZ(ctd)
zz &lt;- seq(min(z), max(z), 0.1)
N2 &lt;- -9.8 / mean(rho) * runlm(z, rho, zz, deriv = 1)
lines(N2, -zz, col = "red")
legend("bottomright",
    lwd = 2, bg = "white",
    col = c("black", "red"),
    legend = c("swN2()", "using runlm()")
)
</code></pre>

<hr>
<h2 id='satellite-class'>Class to Store Satellite Data</h2><span id='topic+satellite-class'></span>

<h3>Description</h3>

<p>This class holds satellite data of various types, including
<a href="#topic+amsr-class">amsr</a> and <a href="#topic+g1sst-class">g1sst</a>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Chantelle Layton
</p>


<h3>See Also</h3>

<p>Other classes holding satellite data: 
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+g1sst-class">g1sst-class</a></code>,
<code><a href="#topic+landsat-class">landsat-class</a></code>
</p>

<hr>
<h2 id='sealevel'>Sample sealevel Data (Halifax Harbour)</h2><span id='topic+sealevel'></span>

<h3>Description</h3>

<p>This sample sea-level dataset is the 2003 record from Halifax Harbour in
Nova Scotia, Canada.  For reasons that are not mentioned on the data archive
website, the record ends on the 8th of October.
</p>


<h3>Details</h3>

<p>See <code><a href="#topic+predict.tidem">predict.tidem()</a></code> for an example that reveals the storm surge
that resulted from Hurricane Juan, in this year.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Source</h3>

<p>The data were created as </p>
<pre> sealevel &lt;-
read.oce("490-01-JAN-2003_slev.csv") sealevel &lt;- oce.edit(sealevel,
"longitude", -sealevel[["longitude"]], reason="Fix longitude hemisphere") </pre>
<p>where the csv file was downloaded from reference 1. Note the correction of longitude
sign, which is required because the data file has no indication that this is
the western hemisphere.
</p>


<h3>References</h3>


<ol>
<li><p> Fisheries and Oceans Canada <code>http://www.meds-sdmm.dfo-mpo.gc.ca/isdm-gdsi/index-eng.html</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to sealevel data: 
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic+as.sealevel">as.sealevel</a>()</code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+read.sealevel">read.sealevel</a>()</code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+summary+2Csealevel-method">summary,sealevel-method</a></code>
</p>

<hr>
<h2 id='sealevel-class'>Class to Store Sealevel Data</h2><span id='topic+sealevel-class'></span>

<h3>Description</h3>

<p>This class stores sealevel data, e.g. from a tide gauge.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>sealevel</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
The key items stored in this slot are <code>time</code> and <code>elevation</code>.</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>sealevel</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
An example of the former might be the location at which a <code>sealevel</code> measurement was made, stored in <code>longitude</code> and <code>latitude</code>, and of the latter might be <code>filename</code>, the name of the data source.</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>sealevel</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+sealevel-class">sealevel</a> objects (see <code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+sealevel-class">sealevel</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to sealevel data: 
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic+as.sealevel">as.sealevel</a>()</code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+read.sealevel">read.sealevel</a>()</code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+summary+2Csealevel-method">summary,sealevel-method</a></code>
</p>

<hr>
<h2 id='sealevelTuktoyaktuk'>Sample sealevel Data (Tuktoyaktuk)</h2><span id='topic+sealevelTuktoyaktuk'></span>

<h3>Description</h3>

<p>This sea-level dataset is provided with in Appendix 7.2 of Foreman (1977)
and also with the <code>T_TIDE</code> package (Pawlowicz et al., 2002). It results
from measurements made in 1975 at Tuktoyaktuk, Northwest Territories,
Canada.
</p>


<h3>Details</h3>

<p>The data set contains 1584 points, some of which have NA for sea-level
height.
</p>
<p>Although Foreman's Appendix 7.2 states that times are in Mountain standard
time, the timezone is set to <code>UTC</code> in the present case, so that the
results will be similar to those he provides in his Appendix 7.3.
</p>


<h3>Historical note</h3>

<p>Until Jan 6, 2018, the time in this dataset had been increased
by 7 hours. However, this alteration was removed on this date,
to make for simpler comparison of amplitude and phase output with
the results obtained by Foreman (1977) and Pawlowicz et al. (2002).
</p>


<h3>Source</h3>

<p>The data were based on the <code>T_TIDE</code> dataset, which in turn
seems to be based on Appendix 7.2 of Foreman (1977).  Minor editing was on
file format, and then the <code>sealevelTuktoyaktuk</code> object was created
using <code><a href="#topic+as.sealevel">as.sealevel()</a></code>.
</p>


<h3>References</h3>

<p>Foreman, M. G. G., 1977.  Manual for tidal heights analysis and
prediction.  Pacific Marine Science Report 77-10, Institute of Ocean
Sciences, Patricia Bay, Sidney, BC, 58pp.
</p>
<p>Pawlowicz, Rich, Bob Beardsley, and Steve Lentz, 2002.  Classical tidal
harmonic analysis including error estimates in MATLAB using <code>T_TIDE</code>.
Computers and Geosciences, 28, 929-937.
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to sealevel data: 
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic+as.sealevel">as.sealevel</a>()</code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+read.sealevel">read.sealevel</a>()</code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+summary+2Csealevel-method">summary,sealevel-method</a></code>
</p>

<hr>
<h2 id='secondsToCtime'>Express Time Interval as Colon-Separated String</h2><span id='topic+secondsToCtime'></span>

<h3>Description</h3>

<p>Convert a time interval to a colon-separated string
</p>


<h3>Usage</h3>

<pre><code class='language-R'>secondsToCtime(sec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="secondsToCtime_+3A_sec">sec</code></td>
<td>
<p>length of time interval in seconds.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string with a colon-separated time interval.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+ctimeToSeconds">ctimeToSeconds()</a></code>, the inverse of this.
</p>
<p>Other things related to time: 
<code><a href="#topic+ctimeToSeconds">ctimeToSeconds</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+numberAsHMS">numberAsHMS</a>()</code>,
<code><a href="#topic+numberAsPOSIXct">numberAsPOSIXct</a>()</code>,
<code><a href="#topic+unabbreviateYear">unabbreviateYear</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
cat("   10 s = ", secondsToCtime(10), "\n", sep = "")
cat("   61 s = ", secondsToCtime(61), "\n", sep = "")
cat("86400 s = ", secondsToCtime(86400), "\n", sep = "")
</code></pre>

<hr>
<h2 id='section'>Sample section Data</h2><span id='topic+section'></span>

<h3>Description</h3>

<p>This is line A03 (ExpoCode 90CT40_1, with nominal sampling date 1993-09-11).
The chief scientist was Tereschenkov of SOI, working aboard the Russian ship
Multanovsky, undertaking a westward transect from the Mediterranean outflow
region across to North America, with a change of heading in the last few dozen
stations to run across the nominal Gulf Stream axis.
The data flags follow the &quot;WHP Bottle&quot;convention, set by
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method()</a></code> to <code>"WHP bottle"</code>.  This convention
used to be described at the link
<code style="white-space: pre;">&#8288;https://www.nodc.noaa.gov/woce/woce_v3/wocedata_1/whp/exchange/exchange_format_desc.htm&#8288;</code>
but that was found to fail in December 2020.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(section)
</code></pre>


<h3>Speculation on a timing error</h3>

<p>In May 2022, it was discovered that the times in this dataset are not fully
sequential, at two spots.  This might be a reporting error. Station 41 has
time listed as 1993-10-03T00:06:00 and that leads to a time reversal.
However, if that time were actually on the day before, then
the time reversal would vanish, and the inter-station timing of
about 5 to 6 hours would be recovered. A similar pattern is seen at station
45.  Of course, this hypothesis of incorrect recording is difficult to test,
for data taken thirty years ago.
</p>


<h3>Source</h3>

<p>This is based on the WOCE file named <code>a03_hy1.csv</code>, downloaded
from <code style="white-space: pre;">&#8288;https://cchdo.ucsd.edu/cruise/90CT40_1&#8288;</code>, 13 April 2015.
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Gulf Stream
data(section)
GS &lt;- subset(section, 113 &lt;= stationId &amp; stationId &lt;= 129)
GSg &lt;- sectionGrid(GS, p = seq(0, 5000, 100))
plot(GSg, span = 1500) # increase span to show more coastline

</code></pre>

<hr>
<h2 id='section-class'>Class to Store Hydrographic Section Data</h2><span id='topic+section-class'></span>

<h3>Description</h3>

<p>This class stores data from oceanographic section surveys.
</p>


<h3>Details</h3>

<p>Sections can be read with <code><a href="#topic+read.section">read.section()</a></code> or created with
<code><a href="#topic+read.section">read.section()</a></code> or created from CTD objects by using
<code><a href="#topic+as.section">as.section()</a></code> or by adding a ctd station to an existing section with
<code><a href="#topic+sectionAddStation">sectionAddStation()</a></code>.
</p>
<p>Sections may be sorted with <code><a href="#topic+sectionSort">sectionSort()</a></code>, subsetted with
<code><a href="#topic+subset+2Csection-method">subset,section-method()</a></code>, smoothed with <code><a href="#topic+sectionSmooth">sectionSmooth()</a></code>, and
gridded with <code><a href="#topic+sectionGrid">sectionGrid()</a></code>.  A &quot;spine&quot; may be added to a section
with <code><a href="#topic+addSpine">addSpine()</a></code>.  Sections may be summarized with
<code><a href="#topic+summary+2Csection-method">summary,section-method()</a></code> and plotted
with <code><a href="#topic+plot+2Csection-method">plot,section-method()</a></code>.
</p>
<p>The sample dataset <code><a href="#topic+section">section()</a></code> contains data along WOCE line A03.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>section</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>section</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
Examples that are of common interest include <code>stationId</code>, <code>longitude</code>, <code>latitude</code> and <code>time</code>.</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>section</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+section-class">section</a> objects (see <code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+section-class">section</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(section)
plot(section[["station", 1]])
pairs(cbind(z = -section[["pressure"]], T = section[["temperature"]], S = section[["salinity"]]))
# T profiles for first few stations in section, at common scale
par(mfrow = c(3, 3))
Tlim &lt;- range(section[["temperature"]])
ylim &lt;- rev(range(section[["pressure"]]))
for (stn in section[["station", 1:9]]) {
    plotProfile(stn, xtype = "potential temperature", ylim = ylim, Tlim = Tlim)
}

</code></pre>

<hr>
<h2 id='sectionAddStation'>Add a ctd Profile to a section Object</h2><span id='topic+sectionAddStation'></span><span id='topic+sectionAddCtd'></span>

<h3>Description</h3>

<p>Add a CTD profile to an existing section.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sectionAddStation(section, station)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sectionAddStation_+3A_section">section</code></td>
<td>
<p>A section to which a station is to be added.</p>
</td></tr>
<tr><td><code id="sectionAddStation_+3A_station">station</code></td>
<td>
<p>A ctd object holding data for the station to be added.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+section-class">section</a> object.
</p>


<h3>Historical note</h3>

<p>Until March 2015, this operation was carried out with the <code>+</code> operator,
but at that time, the syntax was flagged by the development version of R, so it
was changed to the present form.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
ctd2 &lt;- ctd
ctd2[["temperature"]] &lt;- ctd2[["temperature"]] + 0.5
ctd2[["latitude"]] &lt;- ctd2[["latitude"]] + 0.1
section &lt;- as.section(c("ctd", "ctd2"))
ctd3 &lt;- ctd
ctd3[["temperature"]] &lt;- ctd[["temperature"]] + 1
ctd3[["latitude"]] &lt;- ctd[["latitude"]] + 0.1
ctd3[["station"]] &lt;- "Stn 3"
sectionAddStation(section, ctd3)

</code></pre>

<hr>
<h2 id='sectionGrid'>Grid a Section in Pressure Space</h2><span id='topic+sectionGrid'></span>

<h3>Description</h3>

<p>Grid a section, by interpolating to fixed pressure levels.  The
<code>"approx"</code>, <code>"boxcar"</code> and <code>"lm"</code> methods are described in the
documentation for <code><a href="#topic+ctdDecimate">ctdDecimate()</a></code>, which is used to do this
processing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sectionGrid(
  section,
  p,
  method = "approx",
  trim = TRUE,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sectionGrid_+3A_section">section</code></td>
<td>
<p>A <code>section</code> object containing the section to be gridded.</p>
</td></tr>
<tr><td><code id="sectionGrid_+3A_p">p</code></td>
<td>
<p>Optional indication of the pressure levels to which interpolation
should be done.  If this is not supplied, the pressure levels will be
calculated based on the typical spacing in the ctd profiles stored within
<code>section</code>.  If <code>p="levitus"</code>, then pressures will be set to be those
of the Levitus atlas, given by <code><a href="#topic+standardDepths">standardDepths()</a></code>.
If <code>p</code> is a single numerical value,
it is taken as the number of subdivisions to use in a call to <code><a href="base.html#topic+seq">seq()</a></code>
that has range from 0 to the maximum pressure in <code>section</code>.  Finally, if a
vector numerical values is provided, perhaps as constructed with <code><a href="base.html#topic+seq">seq()</a></code>
or <code><a href="#topic+standardDepths">standardDepths</a></code><code>(5)</code> (as in the examples),
then it is used as is, after trimming any values that exceed the maximum
pressure in the station data stored within <code>section</code>.</p>
</td></tr>
<tr><td><code id="sectionGrid_+3A_method">method</code></td>
<td>
<p>The method to use to decimate data within the stations; see
<code><a href="#topic+ctdDecimate">ctdDecimate()</a></code>, which is used for the decimation.</p>
</td></tr>
<tr><td><code id="sectionGrid_+3A_trim">trim</code></td>
<td>
<p>Logical value indicating whether to trim gridded pressures
to the range of the data in <code>section</code>.</p>
</td></tr>
<tr><td><code id="sectionGrid_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
<tr><td><code id="sectionGrid_+3A_...">...</code></td>
<td>
<p>Optional arguments to be supplied to <code><a href="#topic+ctdDecimate">ctdDecimate()</a></code>,
e.g. <code>rule</code> controls extrapolation beyond the observed pressure range,
in the case where <code>method</code> equals <code>"approx"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default <code>"approx"</code> method is best for bottle data, the
<code>"boxcar"</code> is best for ctd data, and the <code>"lm"</code> method is probably
too slow to recommend for exploratory work, in which it is common to do trials
with a variety of <code>"p"</code> values.
</p>
<p>The stations in the returned value have flags with names that match those
of the corresponding stations in the original <code>section</code>, but the values
of these flags are all set to <code>NA</code>. This recognizes that it makes
no sense to grid flag values, but that there is merit in initializing
a flag system, for possible use in later processing steps.
</p>


<h3>Value</h3>

<p>A <a href="#topic+section-class">section</a> object that contains stations in which
the pressure values match identically, and that has all
flags set to <code>NA</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Gulf Stream
library(oce)
data(section)
GS &lt;- subset(section, 113 &lt;= stationId &amp; stationId &lt;= 129)
GSg &lt;- sectionGrid(GS, p = seq(0, 5000, 100))
plot(GSg, which = "temperature")
## Show effects of various depth schemes

</code></pre>

<hr>
<h2 id='sectionSmooth'>Smooth a Section</h2><span id='topic+sectionSmooth'></span>

<h3>Description</h3>

<p>Smooth a section, in any of several ways, working either in the vertical
direction or in both the vertical and lateral directions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sectionSmooth(
  section,
  method = "spline",
  x,
  xg,
  yg,
  xgl,
  ygl,
  xr,
  yr,
  df,
  gamma = 0.5,
  iterations = 2,
  trim = 0,
  pregrid = FALSE,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sectionSmooth_+3A_section">section</code></td>
<td>
<p>A <code>section</code> object containing the section to be smoothed.
For <code>method="spline"</code>, the pressure levels must match for each station in
the section.</p>
</td></tr>
<tr><td><code id="sectionSmooth_+3A_method">method</code></td>
<td>
<p>A string or a function that specifies the method to use; see &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="sectionSmooth_+3A_x">x</code></td>
<td>
<p>Optional numerical vector, of the same length as the number of stations in <code>section</code>,
which will be used in gridding in the lateral direction. If not provided, this
defaults to <code><a href="#topic+geodDist">geodDist</a></code><code>(section)</code>.</p>
</td></tr>
<tr><td><code id="sectionSmooth_+3A_xg">xg</code>, <code id="sectionSmooth_+3A_xgl">xgl</code></td>
<td>
<p>ignored in the <code>method="spline"</code> case, but passed to
<code><a href="#topic+interpBarnes">interpBarnes()</a></code> if <code>method="barnes"</code>, to kriging
functions if <code>method="kriging"</code>, or to <code>method</code> itself, if it
is a function.
If <code>xg</code> is supplied, it defines the x component of the grid, which by
default is the terms of station distances, x, along the track of the section. (Note
that the grid <code>xg</code> is trimmed to the range of the data <code>x</code>, because otherwise
it would be impossible to interpolate between stations to infer water depth,
longitude, and latitude, which are all stored within the stations in the
returned <code>section</code> object.)
Alternatively, if <code>xgl</code> is supplied, the x grid is established using <code><a href="base.html#topic+seq">seq()</a></code>,
to span the data with <code>xgl</code> elements. If neither of these is supplied, the output
x grid will equal the input x grid.</p>
</td></tr>
<tr><td><code id="sectionSmooth_+3A_yg">yg</code>, <code id="sectionSmooth_+3A_ygl">ygl</code></td>
<td>
<p>similar to <code>xg</code> and <code>xgl</code>, but for pressure. (Note that
trimming to the input <code>y</code> is not done, as it is for <code>xg</code> and <code>x</code>.)
If <code>yg</code> is not given, it is determined from the deepest station in the section.
If <code>ygl</code> was not given, then a grid is constructed to span the pressures
of that deepest station with <code>ygl</code> elements. On the other hand,
if <code>ygl</code> was not given, then the y grid will constructed from the
pressure levels in the deepest station.</p>
</td></tr>
<tr><td><code id="sectionSmooth_+3A_xr">xr</code>, <code id="sectionSmooth_+3A_yr">yr</code></td>
<td>
<p>influence ranges in x (along-section distance) and y (pressure),
passed to <code><a href="#topic+interpBarnes">interpBarnes()</a></code> if <code>method="barnes"</code> or to
<code>method</code>, if the latter is a function. If missing, <code>xr</code> defaults to
1.5X the median inter-station distance and <code>yr</code> defaults to 0.2X
the pressure range. Since these defaults have changed over the evolution
of <code>sectionSmooth</code>, analysts ought to supply <code>xr</code> and <code>yr</code>
in the function call, tailoring them to particular applications, and
making the code more resistant to changes in <code>sectionSmooth</code>.</p>
</td></tr>
<tr><td><code id="sectionSmooth_+3A_df">df</code></td>
<td>
<p>Degree-of-freedom parameter, passed to <code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code>
if <code>method="spline"</code>, and ignored otherwise. If <code>df</code> is not provided,
it defaults to 1/5-th of the number of stations containing non-<code>NA</code>
data at the particular pressure level being processed, as <code>sectionSmooth</code>
works its way through the water column.</p>
</td></tr>
<tr><td><code id="sectionSmooth_+3A_gamma">gamma</code>, <code id="sectionSmooth_+3A_iterations">iterations</code>, <code id="sectionSmooth_+3A_trim">trim</code>, <code id="sectionSmooth_+3A_pregrid">pregrid</code></td>
<td>
<p>Values passed to
<code><a href="#topic+interpBarnes">interpBarnes()</a></code>, if <code>method="barnes"</code>, and
ignored otherwise. <code>gamma</code> is the factor by which
<code>xr</code> and <code>yr</code> are reduced on each of succeeding iterations.
<code>iterations</code> is the number of iterations to do.
<code>trim</code> controls whether the gridded data are set to
<code>NA</code> in regions with sparse data
coverage. <code>pregrid</code> controls whether data are to be
pre-gridded with <code><a href="#topic+binMean2D">binMean2D()</a></code> before being passed to
<code><a href="#topic+interpBarnes">interpBarnes()</a></code>.</p>
</td></tr>
<tr><td><code id="sectionSmooth_+3A_debug">debug</code></td>
<td>
<p>A flag that turns on debugging.  Set to 1 to get a moderate amount
of debugging information, or to 2 to get more.</p>
</td></tr>
<tr><td><code id="sectionSmooth_+3A_...">...</code></td>
<td>
<p>Optional extra arguments, passed to either
<code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code>, if <code>method="spline"</code>, and ignored otherwise.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function produces smoothed fields that might be useful in
simplifying graphical elements created with <code><a href="#topic+plot+2Csection-method">plot,section-method()</a></code>.
As with any smoothing method, a careful analyst will compare the results
against the raw data, e.g. using <code><a href="#topic+plot+2Csection-method">plot,section-method()</a></code>.
In addition the problem of falsely widening narrow features such as
fronts, there is also the potential to get unphysical results with
spars sampling near topographic features such as bottom slopes and ridges.
</p>
<p>The <code>method</code> argument selects between three possible methods.
</p>

<ul>
<li><p> For <code>method="spline"</code>, i.e. the default, the section is smoothed
laterally, using <code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code> on individual pressure levels.
(If the pressure levels do not match up, <code><a href="#topic+sectionGrid">sectionGrid()</a></code> should
be used first to create a <code>section</code> object that can be used here.)
The <code>df</code> argument sets the degree of freedom of the spline, with
larger values indicating less smoothing.
</p>
</li>
<li><p> For <code>method="barnes"</code>, smoothing is done across
both horizontal and vertical coordinates, using <code><a href="#topic+interpBarnes">interpBarnes()</a></code>.
The output station locations are computed by linear interpolation of
input locations, using <code><a href="stats.html#topic+approx">approx()</a></code> on the original
longitudes and longitudes of stations, with the independent variable
being the distance along the track, computed with <code><a href="#topic+geodDist">geodDist()</a></code>.
The values of <code>xg</code>, <code>yg</code>, <code>xgl</code> and <code>ygl</code> control
the smoothing.
</p>
</li>
<li><p> For <code>method="kriging"</code>, smoothing is done across
both horizontal and vertical coordinates, using <code>autoKrige()</code> from
the <a href="https://CRAN.R-project.org/package=automap"><span class="pkg">automap</span></a> package (along with support from the
<a href="https://CRAN.R-project.org/package=sp"><span class="pkg">sp</span></a> package to format the data).  Note that the format of
the value returned by <code>autoKrige()</code> has changed over the years,
and <code>method="kriging"</code> can only handle two particular formats,
one of which is the result from version 1.1.9 of
<a href="https://CRAN.R-project.org/package=automap"><span class="pkg">automap</span></a>.
</p>
</li>
<li><p> If <code>method</code> is a function, then that function is applied to
the (distance, pressure) data for each variable at a grid defined by
<code>xg</code>, <code>xgl</code>, <code>yg</code> and <code>ygl</code>. The function must
be of the form <code style="white-space: pre;">&#8288;function(x,y,z,xg,xr,yg,yr)&#8288;</code>, and must
return a matrix of the gridded result, with first index indicating
the &quot;grid&quot; station number and second index indicating &quot;grid&quot; pressure.
The <code>x</code> value that is supplied to this function is set as
the distance along the section, as computed with <code><a href="#topic+geodDist">geodDist()</a></code>,
and repeated for each of the points at each station.  The corresponding
pressures are provided in <code>y</code>, and the value to be gridded is
in <code>z</code>, which may be <code>temperature</code> on one call to the function,
<code>salinity</code> on another call, etc. The other quantities
have the meanings as described below.
</p>
</li></ul>



<h3>Value</h3>

<p>A <a href="#topic+section-class">section</a> object of that has been smoothed in some way.
Every data field that is in even a single station of the input object
is inserted into every station in the returned value, and therefore
the units represent all the units in any of the stations, as do the
flag names. However, the flags are all set to <code>NA</code> values.
</p>


<h3>Sample of Usage</h3>

<pre>
# I have seen problems with kriging as the automap package has
# evolved, so please be aware that the following may fail.
if (requireNamespace("automap", quietly=TRUE)
       &amp;&amp; requireNamespace("sf", quietly=TRUE)) {
    gsKriging &lt;- sectionSmooth(gs, "kriging", xr=50, yr=200)
    plot(gsKriging, which="temperature")
    mtext("sectionSmooth(..., method=\"kriging\")", line=0.5)
}
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Unsmoothed (Gulf Stream)
library(oce)
data(section)
gs &lt;- subset(section, 115 &lt;= stationId &amp; stationId &lt;= 125)
par(mfrow = c(2, 2))

plot(gs, which = "temperature")
mtext("Original data, without smoothing", line = 0.5)

# Spline
gsg &lt;- sectionGrid(gs, p = seq(0, 5000, 100))
gsSpline &lt;- sectionSmooth(gsg, "spline")
plot(gsSpline, which = "temperature")
mtext("sectionSmooth(..., method=\"spline\")", line = 0.5)

# Barnes
gsBarnes &lt;- sectionSmooth(gs, "barnes", xr = 50, yr = 200)
plot(gsBarnes, which = "temperature")
mtext("sectionSmooth(..., method=\"barnes\")", line = 0.5)

</code></pre>

<hr>
<h2 id='sectionSort'>Sort a Section</h2><span id='topic+sectionSort'></span>

<h3>Description</h3>

<p>Sections created with <code><a href="#topic+as.section">as.section()</a></code> have &quot;stations&quot; that are in the
order of the CTD objects (or filenames for such objects) provided.  Sometimes,
this is not the desired order, e.g. if file names discovered with
<code><a href="base.html#topic+dir">dir()</a></code> are in an order that makes no sense.  (For example, a
practioner might name stations <code>"stn1"</code>, <code>"stn2"</code>, etc., not
realizing that this will yield an unhelpful ordering, by file name, if there
are more than 9 stations.) The purpose of <code>sectionSort</code> is to permit
reordering the constituent stations in sensible ways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sectionSort(section, by, decreasing = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sectionSort_+3A_section">section</code></td>
<td>
<p>A <code>section</code> object containing the section whose stations
are to be sorted.</p>
</td></tr>
<tr><td><code id="sectionSort_+3A_by">by</code></td>
<td>
<p>An optional string indicating how to reorder.  If not provided,
<code>"stationID"</code> will be assumed.  Other choices are <code>"distance"</code>, for
distance from the first station, <code>"longitude"</code>, for longitude,
<code>"latitude"</code> for latitude, and <code>"time"</code>, for time.</p>
</td></tr>
<tr><td><code id="sectionSort_+3A_decreasing">decreasing</code></td>
<td>
<p>A logical value indicating whether to sort in decreasing
order.  The default is FALSE. (Thanks to Martin Renner for adding this
parameter.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object A <a href="#topic+section-class">section</a> object that has been smoothed,
so its data fields will station-to-station variation than
is the case for the input section, <code>x</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(section)
sectionByLongitude &lt;- sectionSort(section, by = "longitude")
head(section)
head(sectionByLongitude)

</code></pre>

<hr>
<h2 id='setFlags'>Set Data-Quality Flags within a oce Object</h2><span id='topic+setFlags'></span>

<h3>Description</h3>

<p>This function changes specified entries in the data-quality
flags of a oce object, which are stored within
a list named <code>flags</code> that resides in the <code>metadata</code>
slot. If the object already has a flag set up for <code>name</code>,
then only the specified entries are altered. If not, the flag
entry is first created and its entries set to <code>default</code>,
after which the entries specified by <code>i</code>
are changed to <code>value</code>.
</p>
<p>The specification is made with <code>i</code>, the form of which
is determined by the data item in question. Generally,
the rules are as follows:
</p>

<ol>
<li><p> If the data item is a vector, then <code>i</code> must be (a)
an integer vector specifying indices to be set to <code>value</code>,
(b) a logical vector of length matching the data item, with
<code>TRUE</code> meaning to set the flag to <code>value</code>, or (c)
a function that takes an <code>oce</code> object as its single
argument, and returns a vector in either of the forms
just described.
</p>
</li>
<li><p> If the data item is an array, then <code>i</code> must be
(a) a data frame of integers whose rows specify spots to change
(where the number of columns matches the number of dimensions
of the data item), (b) a logical array that has dimension equal to
that of the data item, or (c) a function that takes an <code>oce</code>
object as its single input and returns such a data frame or array.
</p>
</li></ol>

<p>See &ldquo;Details&rdquo; for the particular case of <a href="#topic+oce-class">oce</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setFlags(object, name = NULL, i = NULL, value = NULL, debug = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setFlags_+3A_object">object</code></td>
<td>
<p>An oce object.</p>
</td></tr>
<tr><td><code id="setFlags_+3A_name">name</code></td>
<td>
<p>Character string indicating the name of the variable to be flagged. If
this variable is not contained in the object's <code>data</code> slot, an error is reported.</p>
</td></tr>
<tr><td><code id="setFlags_+3A_i">i</code></td>
<td>
<p>Indication of where to insert the flags; see &ldquo;Description&rdquo; for
general rules and &ldquo;Details&rdquo; for rules for <a href="#topic+oce-class">oce</a>
objects.</p>
</td></tr>
<tr><td><code id="setFlags_+3A_value">value</code></td>
<td>
<p>The value to be inserted in the flag.</p>
</td></tr>
<tr><td><code id="setFlags_+3A_debug">debug</code></td>
<td>
<p>Integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This generic function is overridden by specialized functions for some object classes.
</p>


<h3>Value</h3>

<p>An object with flags set as indicated.
</p>


<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>
</p>

<hr>
<h2 id='setFlags+2Cadp-method'>Set Data-Quality Flags within a adp Object</h2><span id='topic+setFlags+2Cadp-method'></span>

<h3>Description</h3>

<p>This function changes specified entries in the data-quality
flags of a adp object, which are stored within
a list named <code>flags</code> that resides in the <code>metadata</code>
slot. If the object already has a flag set up for <code>name</code>,
then only the specified entries are altered. If not, the flag
entry is first created and its entries set to <code>default</code>,
after which the entries specified by <code>i</code>
are changed to <code>value</code>.
</p>
<p>The specification is made with <code>i</code>, the form of which
is determined by the data item in question. Generally,
the rules are as follows:
</p>

<ol>
<li><p> If the data item is a vector, then <code>i</code> must be (a)
an integer vector specifying indices to be set to <code>value</code>,
(b) a logical vector of length matching the data item, with
<code>TRUE</code> meaning to set the flag to <code>value</code>, or (c)
a function that takes an <code>oce</code> object as its single
argument, and returns a vector in either of the forms
just described.
</p>
</li>
<li><p> If the data item is an array, then <code>i</code> must be
(a) a data frame of integers whose rows specify spots to change
(where the number of columns matches the number of dimensions
of the data item), (b) a logical array that has dimension equal to
that of the data item, or (c) a function that takes an <code>oce</code>
object as its single input and returns such a data frame or array.
</p>
</li></ol>

<p>See &ldquo;Details&rdquo; for the particular case of <a href="#topic+adp-class">adp</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adp'
setFlags(
  object,
  name = NULL,
  i = NULL,
  value = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setFlags+2B2Cadp-method_+3A_object">object</code></td>
<td>
<p>An oce object.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Cadp-method_+3A_name">name</code></td>
<td>
<p>Character string indicating the name of the variable to be flagged. If
this variable is not contained in the object's <code>data</code> slot, an error is reported.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Cadp-method_+3A_i">i</code></td>
<td>
<p>Indication of where to insert the flags; see &ldquo;Description&rdquo; for
general rules and &ldquo;Details&rdquo; for rules for <a href="#topic+adp-class">adp</a>
objects.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Cadp-method_+3A_value">value</code></td>
<td>
<p>The value to be inserted in the flag.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Cadp-method_+3A_debug">debug</code></td>
<td>
<p>Integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The only flag that may be set is <code>v</code>, for the array holding velocity. See &ldquo;Indexing rules&rdquo;, noting that adp data are stored in 3D arrays; Example 1 shows using a data frame for <code>i</code>, while Example 2 shows using an array.
</p>


<h3>Value</h3>

<p>An object with flags set as indicated.
</p>


<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adp)

# Example 1: flag first 10 samples in a mid-depth bin of beam 1
i1 &lt;- data.frame(1:20, 40, 1)
adpQC &lt;- initializeFlags(adp, "v", 2)
adpQC &lt;- setFlags(adpQC, "v", i1, 3)
adpClean1 &lt;- handleFlags(adpQC, flags = list(3), actions = list("NA"))
par(mfrow = c(2, 1))
# Top: original, bottom: altered
plot(adp, which = "u1")
plot(adpClean1, which = "u1")

# Example 2: percent-good and error-beam scheme
v &lt;- adp[["v"]]
i2 &lt;- array(FALSE, dim = dim(v))
g &lt;- adp[["g", "numeric"]]
# Thresholds on percent "goodness" and error "velocity"
G &lt;- 25
V4 &lt;- 0.45
for (k in 1:3) {
    i2[, , k] &lt;- ((g[, , k] + g[, , 4]) &lt; G) | (v[, , 4] &gt; V4)
}
adpQC2 &lt;- initializeFlags(adp, "v", 2)
adpQC2 &lt;- setFlags(adpQC2, "v", i2, 3)
adpClean2 &lt;- handleFlags(adpQC2, flags = list(3), actions = list("NA"))
# Top: original, bottom: altered
plot(adp, which = "u1")
plot(adpClean2, which = "u1") # differs at 8h and 20h

</code></pre>

<hr>
<h2 id='setFlags+2Cctd-method'>Set Data-Quality Flags within a ctd Object</h2><span id='topic+setFlags+2Cctd-method'></span>

<h3>Description</h3>

<p>This function changes specified entries in the data-quality
flags of a ctd object, which are stored within
a list named <code>flags</code> that resides in the <code>metadata</code>
slot. If the object already has a flag set up for <code>name</code>,
then only the specified entries are altered. If not, the flag
entry is first created and its entries set to <code>default</code>,
after which the entries specified by <code>i</code>
are changed to <code>value</code>.
</p>
<p>The specification is made with <code>i</code>, the form of which
is determined by the data item in question. Generally,
the rules are as follows:
</p>

<ol>
<li><p> If the data item is a vector, then <code>i</code> must be (a)
an integer vector specifying indices to be set to <code>value</code>,
(b) a logical vector of length matching the data item, with
<code>TRUE</code> meaning to set the flag to <code>value</code>, or (c)
a function that takes an <code>oce</code> object as its single
argument, and returns a vector in either of the forms
just described.
</p>
</li>
<li><p> If the data item is an array, then <code>i</code> must be
(a) a data frame of integers whose rows specify spots to change
(where the number of columns matches the number of dimensions
of the data item), (b) a logical array that has dimension equal to
that of the data item, or (c) a function that takes an <code>oce</code>
object as its single input and returns such a data frame or array.
</p>
</li></ol>

<p>See &ldquo;Details&rdquo; for the particular case of <a href="#topic+ctd-class">ctd</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ctd'
setFlags(
  object,
  name = NULL,
  i = NULL,
  value = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setFlags+2B2Cctd-method_+3A_object">object</code></td>
<td>
<p>An oce object.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Cctd-method_+3A_name">name</code></td>
<td>
<p>Character string indicating the name of the variable to be flagged. If
this variable is not contained in the object's <code>data</code> slot, an error is reported.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Cctd-method_+3A_i">i</code></td>
<td>
<p>Indication of where to insert the flags; see &ldquo;Description&rdquo; for
general rules and &ldquo;Details&rdquo; for rules for <a href="#topic+ctd-class">ctd</a>
objects.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Cctd-method_+3A_value">value</code></td>
<td>
<p>The value to be inserted in the flag.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Cctd-method_+3A_debug">debug</code></td>
<td>
<p>Integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Since all the entries in the <code>data</code> slot of ctd objects are vectors, <code>i</code> must be a vector (either logical as in Example 1 or integer as in Example 2), or a function taking a <code>ctd</code> object and returning such a vector (see &ldquo;Indexing rules&rdquo;).
</p>


<h3>Value</h3>

<p>An object with flags set as indicated.
</p>


<h3>Sample of Usage</h3>

<pre>
# Example 2: Interactive flag assignment based on TS plot, using
# WHP scheme to define 'acceptable' and 'bad' codes
options(eos="gsw")
data(ctd)
qc &lt;- ctd
qc &lt;- initializeFlagScheme(qc, "WHP CTD")
qc &lt;- initializeFlags(qc, "salinity", 2)
Sspan &lt;- diff(range(qc[["SA"]]))
Tspan &lt;- diff(range(qc[["CT"]]))
n &lt;- length(qc[["SA"]])
par(mfrow=c(1, 1))
plotTS(qc, type="o")
message("Click on bad points; quit by clicking to right of plot")
for (i in seq_len(n)) {
    xy &lt;- locator(1)
    if (xy$x &gt; par("usr")[2])
        break
    i &lt;- which.min(abs(qc[["SA"]] - xy$x)/Sspan + abs(qc[["CT"]] - xy$y)/Tspan)
    qc &lt;- setFlags(qc, "salinity", i=i, value=4)
    qc &lt;- handleFlags(qc, flags=list(salinity=4))
    plotTS(qc, type="o")
}
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Coce-method">setFlags,oce-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Example 1: Range-check salinity
data(ctdRaw)
# Salinity and temperature range checks
qc &lt;- ctdRaw
# Initialize flags to 2, meaning good data in the default
# scheme for handleFlags(ctd).
qc &lt;- initializeFlags(qc, "salinity", 2)
qc &lt;- initializeFlags(qc, "temperature", 2)
# Flag bad salinities as 4
oddS &lt;- with(qc[["data"]], salinity &lt; 25 | 40 &lt; salinity)
qc &lt;- setFlags(qc, name = "salinity", i = oddS, value = 4)
# Flag bad temperatures as 4
oddT &lt;- with(qc[["data"]], temperature &lt; -2 | 40 &lt; temperature)
qc &lt;- setFlags(qc, name = "temperature", i = oddT, value = 4)
# Compare results in TS space
par(mfrow = c(2, 1))
plotTS(ctdRaw)
plotTS(handleFlags(qc, flags = c(1, 3:9)))

</code></pre>

<hr>
<h2 id='setFlags+2Coce-method'>Set Data-Quality Flags within a oce Object</h2><span id='topic+setFlags+2Coce-method'></span>

<h3>Description</h3>

<p>This function changes specified entries in the data-quality
flags of a oce object, which are stored within
a list named <code>flags</code> that resides in the <code>metadata</code>
slot. If the object already has a flag set up for <code>name</code>,
then only the specified entries are altered. If not, the flag
entry is first created and its entries set to <code>default</code>,
after which the entries specified by <code>i</code>
are changed to <code>value</code>.
</p>
<p>The specification is made with <code>i</code>, the form of which
is determined by the data item in question. Generally,
the rules are as follows:
</p>

<ol>
<li><p> If the data item is a vector, then <code>i</code> must be (a)
an integer vector specifying indices to be set to <code>value</code>,
(b) a logical vector of length matching the data item, with
<code>TRUE</code> meaning to set the flag to <code>value</code>, or (c)
a function that takes an <code>oce</code> object as its single
argument, and returns a vector in either of the forms
just described.
</p>
</li>
<li><p> If the data item is an array, then <code>i</code> must be
(a) a data frame of integers whose rows specify spots to change
(where the number of columns matches the number of dimensions
of the data item), (b) a logical array that has dimension equal to
that of the data item, or (c) a function that takes an <code>oce</code>
object as its single input and returns such a data frame or array.
</p>
</li></ol>

<p>See &ldquo;Details&rdquo; for the particular case of <a href="#topic+oce-class">oce</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'oce'
setFlags(
  object,
  name = NULL,
  i = NULL,
  value = NULL,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setFlags+2B2Coce-method_+3A_object">object</code></td>
<td>
<p>An oce object.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Coce-method_+3A_name">name</code></td>
<td>
<p>Character string indicating the name of the variable to be flagged. If
this variable is not contained in the object's <code>data</code> slot, an error is reported.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Coce-method_+3A_i">i</code></td>
<td>
<p>Indication of where to insert the flags; see &ldquo;Description&rdquo; for
general rules and &ldquo;Details&rdquo; for rules for <a href="#topic+oce-class">oce</a>
objects.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Coce-method_+3A_value">value</code></td>
<td>
<p>The value to be inserted in the flag.</p>
</td></tr>
<tr><td><code id="setFlags+2B2Coce-method_+3A_debug">debug</code></td>
<td>
<p>Integer set to 0 for quiet action or to 1 for some debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This generic function is overridden by specialized functions for some object classes.
</p>


<h3>Value</h3>

<p>An object with flags set as indicated.
</p>


<h3>See Also</h3>

<p>Other functions relating to data-quality flags: 
<code><a href="#topic+defaultFlags">defaultFlags</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+handleFlags+2Coce-method">handleFlags,oce-method</a></code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+handleFlags">handleFlags</a>()</code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Coce-method">initializeFlagScheme,oce-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+initializeFlagSchemeInternal">initializeFlagSchemeInternal</a>()</code>,
<code><a href="#topic+initializeFlagScheme">initializeFlagScheme</a>()</code>,
<code><a href="#topic+initializeFlags+2Cadp-method">initializeFlags,adp-method</a></code>,
<code><a href="#topic+initializeFlags+2Coce-method">initializeFlags,oce-method</a></code>,
<code><a href="#topic+initializeFlagsInternal">initializeFlagsInternal</a>()</code>,
<code><a href="#topic+initializeFlags">initializeFlags</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+setFlags">setFlags</a>()</code>
</p>

<hr>
<h2 id='shiftLongitude'>Shift Longitude to Range -180 to 180</h2><span id='topic+shiftLongitude'></span>

<h3>Description</h3>

<p>This is a utility function used by <code><a href="#topic+mapGrid">mapGrid()</a></code>. It works
simply by subtracting 180 from each longitude, if any longitude
in the vector exceeds 180.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shiftLongitude(longitudes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shiftLongitude_+3A_longitudes">longitudes</code></td>
<td>
<p>numerical vector of longitudes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of longitudes, shifted to the desired range.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+matrixShiftLongitude">matrixShiftLongitude()</a></code> and <code><a href="#topic+standardizeLongitude">standardizeLongitude()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>

<hr>
<h2 id='showMetadataItem'>Show an Item in the metadata Slot of an oce Object</h2><span id='topic+showMetadataItem'></span>

<h3>Description</h3>

<p>This is a helper function for various <code>summary</code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>showMetadataItem(
  object,
  name,
  label = "",
  postlabel = "",
  isdate = FALSE,
  quote = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="showMetadataItem_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="showMetadataItem_+3A_name">name</code></td>
<td>
<p>name of item</p>
</td></tr>
<tr><td><code id="showMetadataItem_+3A_label">label</code></td>
<td>
<p>label to print before item</p>
</td></tr>
<tr><td><code id="showMetadataItem_+3A_postlabel">postlabel</code></td>
<td>
<p>label to print after item</p>
</td></tr>
<tr><td><code id="showMetadataItem_+3A_isdate">isdate</code></td>
<td>
<p>boolean indicating whether the item is a time</p>
</td></tr>
<tr><td><code id="showMetadataItem_+3A_quote">quote</code></td>
<td>
<p>boolean indicating whether to enclose the item in quotes</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
showMetadataItem(ctd, "ship", "ship")
</code></pre>

<hr>
<h2 id='siderealTime'>Convert From POSIXt Time to Sidereal Time</h2><span id='topic+siderealTime'></span>

<h3>Description</h3>

<p>Convert a POSIXt time to a sidereal time, using
the method in Chapter 7 of reference 1.  The small correction
that he discusses after his equation 7.1 is not applied here.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>siderealTime(t)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="siderealTime_+3A_t">t</code></td>
<td>
<p>a time, in POSIXt format, e.g. as created by
<code><a href="base.html#topic+as.POSIXct">as.POSIXct()</a></code>, <code><a href="base.html#topic+as.POSIXlt">as.POSIXlt()</a></code>, or
<code><a href="#topic+numberAsPOSIXct">numberAsPOSIXct()</a></code>.  If this is provided, the other arguments are
ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A sidereal time, in hours in the range from 0 to 24.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ul>
<li><p> Meeus, Jean. Astronomical Formulas for Calculators. Second Edition.
Richmond, Virginia, USA: Willmann-Bell, 1982.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other things related to astronomy: 
<code><a href="#topic+angle2hms">angle2hms</a>()</code>,
<code><a href="#topic+eclipticalToEquatorial">eclipticalToEquatorial</a>()</code>,
<code><a href="#topic+equatorialToLocalHorizontal">equatorialToLocalHorizontal</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+moonAngle">moonAngle</a>()</code>,
<code><a href="#topic+sunAngle">sunAngle</a>()</code>,
<code><a href="#topic+sunDeclinationRightAscension">sunDeclinationRightAscension</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
t &lt;- ISOdatetime(1978, 11, 13, 0, 0, 0, tz = "UTC")
print(siderealTime(t))

</code></pre>

<hr>
<h2 id='snakeToCamel'>Convert From Snake-Case to Camel-Case Notation</h2><span id='topic+snakeToCamel'></span>

<h3>Description</h3>

<p><code>snakeToCamel</code> converts &quot;snake-case&quot; characters such as <code>"NOVA_SCOTIA"</code>
to &quot;camel-case&quot; values, such as <code>"NovaScotia"</code>.  It was written for
use by <code><a href="#topic+read.argo">read.argo()</a></code>, but it also may prove helpful in other contexts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snakeToCamel(s, specialCases = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snakeToCamel_+3A_s">s</code></td>
<td>
<p>A vector of character values.</p>
</td></tr>
<tr><td><code id="snakeToCamel_+3A_specialcases">specialCases</code></td>
<td>
<p>A vector of character values that tell which
special-cases to apply, or <code>NULL</code> (the default) to turn off special
cases.  The only permitted special case at the moment is <code>"QC"</code> (see
&ldquo;Details&rdquo;) but the idea of this argument is that other cases
can be added later, if needed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The basic procedure is to chop the string up into substrings separated by
the underline character, then to upper-case the first letter of
all substrings except the first, and then to paste the substrings
together.
</p>
<p>However, there are exceptions.  First, any upper-case string that contains no
underlines is converted to lower case, but any mixed-case string with no
underlines is returned as-is (see the second example). Second, if
the <code>specialCases</code> argument contains <code>"QC"</code>, then the
<code>QC</code> is passed through directly (since it is an acronym) and
if the first letter of remaining text is upper-cased (contrast
see the four examples).
</p>


<h3>Value</h3>

<p>A vector of character values
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
snakeToCamel("PARAMETER_DATA_MODE") # "parameterDataMode"
snakeToCamel("PARAMETER") # "parameter"
snakeToCamel("HISTORY_QCTEST") # "historyQctest"
snakeToCamel("HISTORY_QCTEST", "QC") # "historyQCTest"
snakeToCamel("PROFILE_DOXY_QC") # "profileDoxyQc"
snakeToCamel("PROFILE_DOXY_QC", "QC") # "profileDoxyQC"
</code></pre>

<hr>
<h2 id='standardDepths'>Standard Oceanographic Depths</h2><span id='topic+standardDepths'></span>

<h3>Description</h3>

<p>This returns a vector of numbers that build upon the shorter lists
provided in Chapter 10 of reference 1 and the more modern World
Ocean Atlases (e.g. reference 2).
With the default call,
i.e. with <code>n=0</code>, the result is
<code>c(0, 10, 20, 30, 40, 50, 75, 100, 125, 150, 200, 250,
seq(300, 1500, by=100), 1750, seq(2000, 10000, by=500))</code>.
For higher values of <code>n</code>, progressively more and more values
are added between each pair in this sequence.
See the documentation for
<code><a href="#topic+sectionGrid">sectionGrid()</a></code> for how <code>standardDepths</code> can be used
in gridding data for section plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>standardDepths(n = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="standardDepths_+3A_n">n</code></td>
<td>
<p>Integer specifying the number of subdivisions to insert between
each of the stated levels. For exmple, setting <code>n=1</code> puts a 5m level
between the 0 and 10m levels, and <code>n=2</code> puts 3.33 and 6.66 between
0 and 10m.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of depths that are more closely spaced for small values,
i.e. a finer grid near the ocean surface.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Sverdrup, H U, Martin W Johnson, and Richard H Fleming. The Oceans,
Their Physics, Chemistry, and General Biology. New York: Prentice-Hall, 1942.
<code>https://publishing.cdlib.org/ucpressebooks/view?docId=kt167nb66r</code>
</p>
</li></ol>

<p>2.Locarnini, R. A., A. V. Mishonov, J. I. Antonov, T. P. Boyer,
H. E. Garcia, O. K. Baranova, M. M. Zweng, D. R. Johnson, and
S. Levitus. &ldquo;World Ocean Atlas 2009 Temperature.&rdquo;
US Government printing Office, 2010.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>depth &lt;- standardDepths()
depth1 &lt;- standardDepths(1)
plot(depth, depth)
points(depth1, depth1, col = 2, pch = 20, cex = 1 / 2)

</code></pre>

<hr>
<h2 id='standardizeLongitude'>Put Longitude in the Range From -180 to 180</h2><span id='topic+standardizeLongitude'></span>

<h3>Description</h3>

<p>Put Longitude in the Range From -180 to 180
</p>


<h3>Usage</h3>

<pre><code class='language-R'>standardizeLongitude(longitude)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="standardizeLongitude_+3A_longitude">longitude</code></td>
<td>
<p>in degrees East, possibly exceeding 180</p>
</td></tr>
</table>


<h3>Value</h3>

<p>longitude in signed degrees East
</p>


<h3>See Also</h3>

<p><code><a href="#topic+matrixShiftLongitude">matrixShiftLongitude()</a></code> and <code><a href="#topic+shiftLongitude">shiftLongitude()</a></code> are more
powerful relatives to <code>standardizeLongitude</code>.
</p>

<hr>
<h2 id='subset+2Cadp-method'>Subset an adp Object</h2><span id='topic+subset+2Cadp-method'></span>

<h3>Description</h3>

<p>Subset an adp (acoustic Doppler profile) object, in a manner that is function
is somewhat analogous to <code><a href="base.html#topic+subset.data.frame">subset.data.frame()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adp'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Cadp-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Cadp-method_+3A_subset">subset</code></td>
<td>
<p>A condition to be applied to the <code>data</code> portion of
<code>x</code>.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="subset+2B2Cadp-method_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For any data type,
subsetting can be by <code>time</code>, <code>ensembleNumber</code>, or <code>distance</code>.
These may not be combined, but it is easy to use a string of calls to
carry out combined operations, e.g.
<code>subset(subset(adp,distance&lt;d0), time&lt;t0)</code>
</p>


<h3>Value</h3>

<p>An <a href="#topic+adp-class">adp</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adp)
# 1. Look at first part of time series, organized by time
earlyTime &lt;- subset(adp, time &lt; mean(range(adp[["time"]])))
plot(earlyTime)

# 2. Look at first ten ensembles (AKA profiles)
en &lt;- adp[["ensembleNumber"]]
firstTen &lt;- subset(adp, ensembleNumber &lt; en[11])
plot(firstTen)

</code></pre>

<hr>
<h2 id='subset+2Cadv-method'>Subset an adv Object</h2><span id='topic+subset+2Cadv-method'></span>

<h3>Description</h3>

<p>Subset an adv (acoustic Doppler profile) object.  This function is somewhat
analogous to <code><a href="base.html#topic+subset.data.frame">subset.data.frame()</a></code>, except that subsets can only be
specified in terms of <code>time</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adv'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Cadv-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Cadv-method_+3A_subset">subset</code></td>
<td>
<p>a condition to be applied to the <code>data</code> portion of <code>x</code>.
See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="subset+2B2Cadv-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <a href="#topic+adv-class">adv</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adv)
plot(adv)
plot(subset(adv, time &lt; mean(range(adv[["time"]]))))

</code></pre>

<hr>
<h2 id='subset+2Camsr-method'>Subset an amsr Object</h2><span id='topic+subset+2Camsr-method'></span>

<h3>Description</h3>

<p>Return a subset of a <a href="#topic+amsr-class">amsr</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'amsr'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Camsr-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+amsr-class">amsr</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Camsr-method_+3A_subset">subset</code></td>
<td>
<p>an expression indicating how to subset <code>x</code>.</p>
</td></tr>
<tr><td><code id="subset+2B2Camsr-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to subset data within an <a href="#topic+amsr-class">amsr</a>
object by <code>longitude</code> or by <code>latitude</code>.  These two methods cannot
be combined in a single call, so two calls are required, as shown
in the Example.
</p>


<h3>Value</h3>

<p>An <a href="#topic+amsr-class">amsr</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to amsr data: 
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method</a></code>,
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+read.amsr">read.amsr</a>()</code>,
<code><a href="#topic+summary+2Camsr-method">summary,amsr-method</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(amsr) # see ?amsr for how to read and composite such objects
sub &lt;- subset(amsr, -75 &lt; longitude &amp; longitude &lt; -45)
sub &lt;- subset(sub, 40 &lt; latitude &amp; latitude &lt; 50)
plot(sub)
data(coastlineWorld)
lines(coastlineWorld[["longitude"]], coastlineWorld[["latitude"]])

</code></pre>

<hr>
<h2 id='subset+2Cargo-method'>Subset an argo Object</h2><span id='topic+subset+2Cargo-method'></span><span id='topic+subset.argo'></span>

<h3>Description</h3>

<p>Subset an argo object, either by selecting just the &quot;adjusted&quot; data
or by subsetting by pressure or other variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'argo'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Cargo-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+argo-class">argo</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Cargo-method_+3A_subset">subset</code></td>
<td>
<p>An expression indicating how to subset <code>x</code>.</p>
</td></tr>
<tr><td><code id="subset+2B2Cargo-method_+3A_...">...</code></td>
<td>
<p>optional arguments, of which only the first is examined. The
only possibility is <code>within</code>, a polygon enclosing data to be
retained. This must be either a list or data frame, containing items
named either <code>x</code> and <code>y</code> or <code>longitude</code> and
<code>latitude</code>; see Example 4.  If <code>within</code> is given,
then <code>subset</code> is ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>subset</code> is the string <code>"adjusted"</code>, then <code>subset</code>
replaces the station variables with their adjusted counterparts. In
the argo notation, e.g. <code>PSAL</code> is replaced with <code>PSAL_ADJUSTED</code>;
in the present notation, this means that <code>salinity</code> in the <code>data</code>
slot is replaced with <code>salinityAdjusted</code>, and the latter is deleted.
Similar replacements are also done with the flags stored in the <code>metadata</code>
slot.
</p>
<p>If <code>subset</code> is an expression, then the action is somewhat similar
to other <code>subset</code> functions, but with the restriction that
only one independent variable may be
used in in any call to the function, so that
repeated calls will be necessary to subset based on more than one
independent variable.  Subsetting may be done by anything
stored in the data, e.g. <code>time</code>,
<code>latitude</code>, <code>longitude</code>, <code>profile</code>, <code>dataMode</code>,
or <code>pressure</code> or by <code>profile</code> (a made-up variable),
<code>id</code> (from the <code>metadata</code> slot) or <code>ID</code> (a synonym for <code>id</code>).
Note that subsetting by <code>pressure</code>
preserves matrix shape, by setting discarded values to <code>NA</code>, as opposed
to dropping data (as is the case with <code>time</code>, for example).
</p>


<h3>Value</h3>

<p>An <a href="#topic+argo-class">argo</a> object.
</p>


<h3>Sample of Usage</h3>

<pre>
# Example 2: restrict attention to delayed-mode profiles.
par(mfrow=c(1, 1))
plot(subset(argo, dataMode == "D"))

# Example 3: contrast adjusted and unadjusted data
par(mfrow=c(1, 2))
plotTS(argo)
plotTS(subset(argo, "adjusted"))

# Example 2. Subset by a polygon determined with locator()
par(mfrow=c(1, 2))
plot(argo, which="map")
# Can get a boundary with e.g. locator(4)
boundary &lt;- list(x=c(-65, -40, -40, -65), y=c(65, 65, 45, 45))
argoSubset &lt;- subset(argo, within=boundary)
plot(argoSubset, which="map")
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+summary+2Cargo-method">summary,argo-method</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(argo)

# Example 1: subset by time, longitude, and pressure
par(mfrow = c(2, 2))
plot(argo)
plot(subset(argo, time &gt; mean(time)))
plot(subset(argo, longitude &gt; mean(longitude)))
plot(subset(argoGrid(argo), pressure &gt; 500 &amp; pressure &lt; 1000), which = 5)

</code></pre>

<hr>
<h2 id='subset+2Ccm-method'>Subset a cm Object</h2><span id='topic+subset+2Ccm-method'></span>

<h3>Description</h3>

<p>This function is somewhat analogous to <code><a href="base.html#topic+subset.data.frame">subset.data.frame()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'cm'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Ccm-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+cm-class">cm</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Ccm-method_+3A_subset">subset</code></td>
<td>
<p>a condition to be applied to the <code>data</code> portion of <code>x</code>.
See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="subset+2B2Ccm-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>cm</code> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to cm data: 
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+as.cm">as.cm</a>()</code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+read.cm">read.cm</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+summary+2Ccm-method">summary,cm-method</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(cm)
plot(cm)
plot(subset(cm, time &lt; mean(range(cm[["time"]]))))

</code></pre>

<hr>
<h2 id='subset+2Ccoastline-method'>Subset a coastline Object</h2><span id='topic+subset+2Ccoastline-method'></span><span id='topic+subset.coastline'></span>

<h3>Description</h3>

<p>Subsets a coastline object according to limiting values for longitude
and latitude.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'coastline'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Ccoastline-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+coastline-class">coastline</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Ccoastline-method_+3A_subset">subset</code></td>
<td>
<p>An expression indicating how to subset <code>x</code>. See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="subset+2B2Ccoastline-method_+3A_...">...</code></td>
<td>
<p>optional additional arguments, the only one of which is considered
is one named <code>debug</code>, an integer that controls the level of debugging. If
this is not supplied, <code>debug</code> is assumed to be 0, meaning no debugging. If
it is 1, the steps of determining the bounding box are shown. If it is 2 or larger,
then additional processing steps are shown, including the extraction of every
polygon involved in the final result.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As illustrated in the &ldquo;Examples&rdquo;, <code>subset</code> must be an
expression that indicates limits on <code>latitude</code> and
<code>longitude</code>. The individual elements are provided in R notation,
not mathematical notation (i.e. <code style="white-space: pre;">&#8288;30&lt;latitude&lt;60&#8288;</code> would not work).
Ampersands must be used to combine the limits.  The simplest way
to understand this is to copy the example directly, and then modify
the stated limits. Note that <code>&gt;</code> comparison is not permitted,
and that <code>&lt;</code> is converted to <code>&lt;=</code> in the calculation.
Similarly, <code>&amp;&amp;</code> is converted to <code>&amp;</code>. Spaces in the
expression are ignored. For convenience, <code>longitude</code> and
and <code>latitude</code> may be abbreviated as <code>lon</code> and <code>lat</code>,
as in the &ldquo;Examples&rdquo;.
</p>


<h3>Value</h3>

<p>A <code>coastline</code> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+summary+2Ccoastline-method">summary,coastline-method</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(coastlineWorld)
# Subset to a box centred on Nova Scotia, Canada
if (requireNamespace("sf")) {
    cl &lt;- subset(coastlineWorld, -80 &lt; lon &amp; lon &lt;- 50 &amp; 30 &lt; lat &amp; lat &lt; 60)
    # The plot demonstrates that the trimming is as requested.
    plot(cl, clon = -65, clat = 45, span = 6000)
    rect(-80, 30, -50, 60, bg = "transparent", border = "red")
}
</code></pre>

<hr>
<h2 id='subset+2Cctd-method'>Subset a ctd Object</h2><span id='topic+subset+2Cctd-method'></span>

<h3>Description</h3>

<p>Return a subset of a <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ctd'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Cctd-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Cctd-method_+3A_subset">subset</code></td>
<td>
<p>An expression indicating how to subset <code>x</code>.</p>
</td></tr>
<tr><td><code id="subset+2B2Cctd-method_+3A_...">...</code></td>
<td>
<p>optional arguments, of which only the first is examined. The only
possibility is that this argument be named <code>indices</code>. See &ldquo;Details&rdquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to subset data within
a <a href="#topic+ctd-class">ctd</a> object. There are two ways of working. If
<code>subset</code> is supplied, then it is a logical expression
that is evaluated within the environment of the <code>data</code>
slot of the object (see Example 1). Alternatively, if the
<code>...</code> list contains an expression defining <code>indices</code>,
then that expression is used to subset each item within the
<code>data</code> slot (see Example 2).
</p>


<h3>Value</h3>

<p>A <a href="#topic+ctd-class">ctd</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
plot(ctd)
# Example 1
plot(subset(ctd, pressure &lt; 10))
# Example 2
plot(subset(ctd, indices = 1:10))

</code></pre>

<hr>
<h2 id='subset+2Cechosounder-method'>Subset an echosounder Object</h2><span id='topic+subset+2Cechosounder-method'></span>

<h3>Description</h3>

<p>This function is somewhat analogous to <code><a href="base.html#topic+subset.data.frame">subset.data.frame()</a></code>.
Subsetting can be by <code>time</code> or <code>depth</code>, but these may not be
combined; use a sequence of calls to subset by both.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'echosounder'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Cechosounder-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+echosounder-class">echosounder</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Cechosounder-method_+3A_subset">subset</code></td>
<td>
<p>a condition to be applied to the <code>data</code> portion of
<code>x</code>.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="subset+2B2Cechosounder-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+echosounder-class">echosounder</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to echosounder data: 
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic+as.echosounder">as.echosounder</a>()</code>,
<code><a href="#topic+echosounder-class">echosounder-class</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+findBottom">findBottom</a>()</code>,
<code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method</a></code>,
<code><a href="#topic+read.echosounder">read.echosounder</a>()</code>,
<code><a href="#topic+summary+2Cechosounder-method">summary,echosounder-method</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(echosounder)
plot(echosounder)
plot(subset(echosounder, depth &lt; 10))
plot(subset(echosounder, time &lt; mean(range(echosounder[["time"]]))))

</code></pre>

<hr>
<h2 id='subset+2Clobo-method'>Subset a lobo Object</h2><span id='topic+subset+2Clobo-method'></span>

<h3>Description</h3>

<p>Subset an lobo object, in a way that is somewhat
analogous to <code><a href="base.html#topic+subset.data.frame">subset.data.frame()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lobo'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Clobo-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+lobo-class">lobo</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Clobo-method_+3A_subset">subset</code></td>
<td>
<p>a condition to be applied to the <code>data</code> portion of
<code>x</code>.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="subset+2B2Clobo-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+lobo-class">lobo</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to lobo data: 
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic+as.lobo">as.lobo</a>()</code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+read.lobo">read.lobo</a>()</code>,
<code><a href="#topic+summary+2Clobo-method">summary,lobo-method</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>

<hr>
<h2 id='subset+2Cmet-method'>Subset a met Object</h2><span id='topic+subset+2Cmet-method'></span>

<h3>Description</h3>

<p>This function is somewhat analogous to <code><a href="base.html#topic+subset.data.frame">subset.data.frame()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'met'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Cmet-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+met-class">met</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Cmet-method_+3A_subset">subset</code></td>
<td>
<p>An expression indicating how to subset <code>x</code>.</p>
</td></tr>
<tr><td><code id="subset+2B2Cmet-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+met-class">met</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to met data: 
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic+as.met">as.met</a>()</code>,
<code><a href="#topic+download.met">download.met</a>()</code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+read.met">read.met</a>()</code>,
<code><a href="#topic+summary+2Cmet-method">summary,met-method</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(met)
# Few days surrounding Hurricane Juan
plot(subset(met, time &gt; as.POSIXct("2003-09-27", tz = "UTC")))

</code></pre>

<hr>
<h2 id='subset+2Coce-method'>Subset an oce Object</h2><span id='topic+subset+2Coce-method'></span>

<h3>Description</h3>

<p>This is a basic class for general oce objects.  It has specialised
versions for most sub-classes, e.g. <code><a href="#topic+subset+2Cctd-method">subset,ctd-method()</a></code>
for <code>ctd</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'oce'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Coce-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Coce-method_+3A_subset">subset</code></td>
<td>
<p>a logical expression indicating how to take the subset; the form depends on the sub-class.</p>
</td></tr>
<tr><td><code id="subset+2B2Coce-method_+3A_...">...</code></td>
<td>
<p>optional arguments, used in some specialized methods, e.g. <code><a href="#topic+subset+2Csection-method">subset,section-method()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An oce object.
</p>


<h3>See Also</h3>

<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
# Select just the top 10 metres (pressure less than 10 dbar)
top10 &lt;- subset(ctd, pressure &lt; 10)
par(mfrow = c(1, 2))
plotProfile(ctd)
plotProfile(top10)
</code></pre>

<hr>
<h2 id='subset+2Codf-method'>Subset an odf Object</h2><span id='topic+subset+2Codf-method'></span>

<h3>Description</h3>

<p>This function is somewhat analogous to <code><a href="base.html#topic+subset.data.frame">subset.data.frame()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'odf'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Codf-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+odf-class">odf</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Codf-method_+3A_subset">subset</code></td>
<td>
<p>a condition to be applied to the <code>data</code> portion of
<code>x</code>.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="subset+2B2Codf-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It seems likely that users will first convert the odf object into
another class (e.g. ctd) and use the subset method of that class;
note that some of those methods interpret the ... argument.
</p>


<h3>Value</h3>

<p>An <a href="#topic+odf-class">odf</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to odf data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ODF2oce">ODF2oce</a>()</code>,
<code><a href="#topic+ODFListFromHeader">ODFListFromHeader</a>()</code>,
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.odf">read.odf</a>()</code>,
<code><a href="#topic+summary+2Codf-method">summary,odf-method</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>

<hr>
<h2 id='subset+2Crsk-method'>Subset a rsk Object</h2><span id='topic+subset+2Crsk-method'></span>

<h3>Description</h3>

<p>Subset a rsk object.  This function is somewhat analogous to
<code><a href="base.html#topic+subset.data.frame">subset.data.frame()</a></code>, but subsetting is only permitted by time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'rsk'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Crsk-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+rsk-class">rsk</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Crsk-method_+3A_subset">subset</code></td>
<td>
<p>a condition to be applied to the <code>data</code> portion of <code>x</code>.
See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="subset+2B2Crsk-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+rsk-class">rsk</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic+as.rsk">as.rsk</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+read.rsk">read.rsk</a>()</code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+rskPatm">rskPatm</a>()</code>,
<code><a href="#topic+rskToc">rskToc</a>()</code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+summary+2Crsk-method">summary,rsk-method</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(rsk)
plot(rsk)
plot(subset(rsk, time &lt; mean(range(rsk[["time"]]))))

</code></pre>

<hr>
<h2 id='subset+2Csealevel-method'>Subset a sealevel Object</h2><span id='topic+subset+2Csealevel-method'></span>

<h3>Description</h3>

<p>This function is somewhat analogous to <code><a href="base.html#topic+subset.data.frame">subset.data.frame()</a></code>, but
subsetting is only permitted by time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'sealevel'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Csealevel-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+sealevel-class">sealevel</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Csealevel-method_+3A_subset">subset</code></td>
<td>
<p>a condition to be applied to the <code>data</code> portion of
<code>x</code>.</p>
</td></tr>
<tr><td><code id="subset+2B2Csealevel-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>sealevel</code> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to sealevel data: 
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic+as.sealevel">as.sealevel</a>()</code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+read.sealevel">read.sealevel</a>()</code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+summary+2Csealevel-method">summary,sealevel-method</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(sealevel)
plot(sealevel)
plot(subset(sealevel, time &lt; mean(range(sealevel[["time"]]))))

</code></pre>

<hr>
<h2 id='subset+2Csection-method'>Subset a section Object</h2><span id='topic+subset+2Csection-method'></span><span id='topic+subset.section'></span>

<h3>Description</h3>

<p>Return a subset of a section object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'section'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Csection-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+section-class">section</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Csection-method_+3A_subset">subset</code></td>
<td>
<p>an optional indication of either the stations to be kept,
or the data to be kept within the stations.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="subset+2B2Csection-method_+3A_...">...</code></td>
<td>
<p>optional arguments, of which only the first is examined. The
possibilities for this argument are <code>indices</code>, which must be a
vector of station indices (see Example 6), or <code>within</code>, which must be
a list or data frame, containing items named either <code>x</code> and <code>y</code>
or <code>longitude</code> and <code>latitude</code> (see Example 7). If <code>within</code>
is given, then <code>subset</code> is ignored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is used to subset data within the
stations of a section, or to choose a subset of the stations
themselves. The first case is handled with the <code>subset</code> argument,
while the second is handled if <code>...</code> contains a vector named
<code>indices</code>. Either <code>subset</code> or <code>indices</code> must be
provided, but not both.
</p>
<p><strong>In the &quot;subset&quot; method</strong>, <code>subset</code> indicates
either stations to be kept, or data to be kept within the stations.
</p>
<p>The first step in processing is to check for the presence of certain
key words in the <code>subset</code> expression. If <code>distance</code>
is present, then stations are selected according to a condition on the
distance (in km) from the first station to the given station (Example 1).
If either <code>longitude</code> or <code>latitude</code> is given, then
stations are selected according to the stated condition (Example 2).
If <code>stationId</code> is present, then selection is in terms of the
station ID (not the sequence number) is used (Example 3). In all of these
cases, stations are either selected in their entirety or dropped
in their entirety.
</p>
<p>If none of these keywords is present, then the <code>subset</code> expression is
evaluated in the context of the <code>data</code> slot of each of the CTD stations
stored within <code>x</code>. (Note that this slot does not normally
contain any of the keywords that are listed in the previous
paragraph; it does, then odd results may occur.) Each station is examined
in turn, with <code>subset</code> being evaluated individually in each. The evaluation
produces a logical vector. If that vector has length 1 (Examples 4 and 5)
then the station is retained or discarded, accordingly. If the vector is longer,
then the logical vector is used as a sieve to subsample that individual CTD
profile.
</p>
<p><strong>In the &quot;indices&quot; method</strong>, stations are selected using
<code>indices</code>, which may be a vector of integers that indicate sequence
number, or a logical vector, again indicating which stations to keep.
</p>


<h3>Value</h3>

<p>A <a href="#topic+section-class">section</a> object.
</p>


<h3>Sample of Usage</h3>

<pre>
# Example 7. Subset by a polygon determined with locator()
par(mfrow=c(2, 1))
plot(section, which="map")
bdy &lt;- locator(4) # choose a polygon near N. America
GS &lt;- subset(section, within=bdy)
plot(GS, which="map")
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>
<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+summary+2Csection-method">summary,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(section)

# Example 1. Stations within 500 km of the first station
starting &lt;- subset(section, distance &lt; 500)

# Example 2. Stations east of 50W
east &lt;- subset(section, longitude &gt; (-50))

# Example 3. Gulf Stream
GS &lt;- subset(section, 113 &lt;= stationId &amp; stationId &lt;= 129)

# Example 4. Only stations with more than 5 pressure levels
long &lt;- subset(section, length(pressure) &gt; 5)

# Example 5. Only stations that have some data in top 50 dbar
surfacing &lt;- subset(section, min(pressure) &lt; 50)

# Example 6. Similar to #4, but done in more detailed way
long &lt;- subset(section,
    indices = unlist(lapply(
        section[["station"]],
        function(s) 5 &lt; length(s[["pressure"]])
    ))
)

</code></pre>

<hr>
<h2 id='subset+2Ctopo-method'>Subset a topo Object</h2><span id='topic+subset+2Ctopo-method'></span>

<h3>Description</h3>

<p>This function is somewhat analogous to <code><a href="base.html#topic+subset.data.frame">subset.data.frame()</a></code>.
Subsetting can be by <code>time</code> or <code>distance</code>, but these may not be
combined; use a sequence of calls to subset by both.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'topo'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Ctopo-method_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+topo-class">topo</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Ctopo-method_+3A_subset">subset</code></td>
<td>
<p>A condition to be applied to the <code>data</code> portion of <code>x</code>.
See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="subset+2B2Ctopo-method_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <a href="#topic+topo-class">topo</a> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to topo data: 
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic+as.topo">as.topo</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+read.topo">read.topo</a>()</code>,
<code><a href="#topic+summary+2Ctopo-method">summary,topo-method</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+topoInterpolate">topoInterpolate</a>()</code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># northern hemisphere
library(oce)
data(topoWorld)
plot(subset(topoWorld, latitude &gt; 0))

</code></pre>

<hr>
<h2 id='subset+2Cxbt-method'>Subset an xbt Object</h2><span id='topic+subset+2Cxbt-method'></span>

<h3>Description</h3>

<p>This function is somewhat analogous to <code><a href="base.html#topic+subset.data.frame">subset.data.frame()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'xbt'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset+2B2Cxbt-method_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+xbt-class">xbt</a> object.</p>
</td></tr>
<tr><td><code id="subset+2B2Cxbt-method_+3A_subset">subset</code></td>
<td>
<p>a condition to be applied to the <code>data</code> portion of <code>x</code>.
See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="subset+2B2Cxbt-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>xbt</code> object.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to xbt data: 
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>,
<code><a href="#topic+as.xbt">as.xbt</a>()</code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1</a>()</code>,
<code><a href="#topic+read.xbt">read.xbt</a>()</code>,
<code><a href="#topic+summary+2Cxbt-method">summary,xbt-method</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other functions that subset oce objects: 
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>,
<code><a href="#topic+subset+2Coce-method">subset,oce-method</a></code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(xbt)
plot(xbt)
plot(subset(xbt, depth &lt; mean(range(xbt[["depth"]]))))

</code></pre>

<hr>
<h2 id='subtractBottomVelocity'>Subtract Bottom Velocity From an adp Object</h2><span id='topic+subtractBottomVelocity'></span>

<h3>Description</h3>

<p>Subtracts bottom tracking velocities from an <code>"adp"</code> object. Works for
all coordinate systems (<code>beam</code>, <code>xyz</code>, and <code>enu</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subtractBottomVelocity(x, despike = FALSE, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subtractBottomVelocity_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object that contains bottom-tracking velocities.</p>
</td></tr>
<tr><td><code id="subtractBottomVelocity_+3A_despike">despike</code></td>
<td>
<p>either a logical value or a univariate function. This
controls whether the bottom velocity (<code>bv</code>) values should be altered before they are
subtracted from the beam velocities. If it is <code>TRUE</code> then the <code>bv</code> values are despiked
first by calling <code><a href="#topic+despike">despike()</a></code>. If it is a function, then that function is used instead of
<code><a href="#topic+despike">despike()</a></code>, e.g. <code>function(x) despike(x, reference="smooth")</code> would change the reference
function for despiking from its default of <code>"median"</code>.</p>
</td></tr>
<tr><td><code id="subtractBottomVelocity_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+read.adp">read.adp()</a></code> for notes on functions relating to
<code>"adp"</code> objects, and <a href="#topic+adp-class">adp</a> for notes on the ADP
object class.
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='summary+2Cadp-method'>Summarize an adp Object</h2><span id='topic+summary+2Cadp-method'></span><span id='topic+summary.adp'></span><span id='topic+summary+2Cadp+2Cmissing-method'></span>

<h3>Description</h3>

<p>Summarize data in an <code>adp</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adp'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Cadp-method_+3A_object">object</code></td>
<td>
<p>an object of class <code>"adp"</code>, usually, a result of a call
to <code><a href="#topic+read.oce">read.oce()</a></code>, <code><a href="#topic+read.adp.rdi">read.adp.rdi()</a></code>, or
<code><a href="#topic+read.adp.nortek">read.adp.nortek()</a></code>.</p>
</td></tr>
<tr><td><code id="summary+2B2Cadp-method_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pertinent summary information is presented.
</p>


<h3>Value</h3>

<p>A matrix containing statistics of the elements of the <code>data</code>
slot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='summary+2Cadv-method'>Summarize an adv Object</h2><span id='topic+summary+2Cadv-method'></span>

<h3>Description</h3>

<p>Summarize data in an <code>adv</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'adv'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Cadv-method_+3A_object">object</code></td>
<td>
<p>an object of class <code>"adv"</code>, usually, a result of a call to
<code><a href="#topic+read.adv">read.adv()</a></code>.</p>
</td></tr>
<tr><td><code id="summary+2B2Cadv-method_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adv)
summary(adv)

</code></pre>

<hr>
<h2 id='summary+2Camsr-method'>Summarize an amsr Object</h2><span id='topic+summary+2Camsr-method'></span>

<h3>Description</h3>

<p>Print a summary of key components of the object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'amsr'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Camsr-method_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+amsr-class">amsr</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Camsr-method_+3A_...">...</code></td>
<td>
<p>ignored.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to amsr data: 
<code><a href="#topic++5B+5B+2Camsr-method">[[,amsr-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Camsr-method">[[&lt;-,amsr-method</a></code>,
<code><a href="#topic+amsr-class">amsr-class</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+composite+2Camsr-method">composite,amsr-method</a></code>,
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+read.amsr">read.amsr</a>()</code>,
<code><a href="#topic+subset+2Camsr-method">subset,amsr-method</a></code>
</p>

<hr>
<h2 id='summary+2Cargo-method'>Summarize an argo Object</h2><span id='topic+summary+2Cargo-method'></span><span id='topic+summary.argo'></span>

<h3>Description</h3>

<p>Summarizes some of the data in an <code>argo</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'argo'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Cargo-method_+3A_object">object</code></td>
<td>
<p>an object of class <code>"argo"</code>, usually, a result of a call to <code><a href="#topic+read.argo">read.argo()</a></code>.</p>
</td></tr>
<tr><td><code id="summary+2B2Cargo-method_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Pertinent summary information is presented.
</p>


<h3>Value</h3>

<p>A matrix containing statistics of the elements of the <code>data</code> slot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to argo data: 
<code><a href="#topic++5B+5B+2Cargo-method">[[,argo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cargo-method">[[&lt;-,argo-method</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+argoGrid">argoGrid</a>()</code>,
<code><a href="#topic+argoNames2oceNames">argoNames2oceNames</a>()</code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+as.argo">as.argo</a>()</code>,
<code><a href="#topic+handleFlags+2Cargo-method">handleFlags,argo-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+read.argo.copernicus">read.argo.copernicus</a>()</code>,
<code><a href="#topic+read.argo">read.argo</a>()</code>,
<code><a href="#topic+subset+2Cargo-method">subset,argo-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(argo)
summary(argo)

</code></pre>

<hr>
<h2 id='summary+2Cbremen-method'>Summarize a bremen Object</h2><span id='topic+summary+2Cbremen-method'></span>

<h3>Description</h3>

<p>Pertinent summary information is presented, including the station name,
sampling location, data ranges, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'bremen'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Cbremen-method_+3A_object">object</code></td>
<td>
<p>a <a href="#topic+bremen-class">bremen</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Cbremen-method_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to bremen data: 
<code><a href="#topic++5B+5B+2Cbremen-method">[[,bremen-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cbremen-method">[[&lt;-,bremen-method</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+read.bremen">read.bremen</a>()</code>
</p>

<hr>
<h2 id='summary+2Ccm-method'>Summarize a cm Object</h2><span id='topic+summary+2Ccm-method'></span>

<h3>Description</h3>

<p>Summarizes some of the data in a <code>cm</code> object, presenting such information
as the station name, sampling location, data ranges, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'cm'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Ccm-method_+3A_object">object</code></td>
<td>
<p>A <a href="#topic+cm-class">cm</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Ccm-method_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for the <a href="#topic+cm-class">cm</a> class explains the structure
of <code>cm</code> objects, and also outlines the other functions dealing with them.
</p>
<p>Other things related to cm data: 
<code><a href="#topic++5B+5B+2Ccm-method">[[,cm-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccm-method">[[&lt;-,cm-method</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Ccm-method">applyMagneticDeclination,cm-method</a></code>,
<code><a href="#topic+as.cm">as.cm</a>()</code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+read.cm">read.cm</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Ccm-method">subset,cm-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(cm)
summary(cm)

</code></pre>

<hr>
<h2 id='summary+2Ccoastline-method'>Summarize a coastline Object</h2><span id='topic+summary+2Ccoastline-method'></span><span id='topic+summary.coastline'></span>

<h3>Description</h3>

<p>Summarizes coastline length, bounding box, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'coastline'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Ccoastline-method_+3A_object">object</code></td>
<td>
<p>a <a href="#topic+coastline-class">coastline</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Ccoastline-method_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to coastline data: 
<code><a href="#topic++5B+5B+2Ccoastline-method">[[,coastline-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ccoastline-method">[[&lt;-,coastline-method</a></code>,
<code><a href="#topic+as.coastline">as.coastline</a>()</code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+coastlineBest">coastlineBest</a>()</code>,
<code><a href="#topic+coastlineCut">coastlineCut</a>()</code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+download.coastline">download.coastline</a>()</code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+read.coastline.openstreetmap">read.coastline.openstreetmap</a>()</code>,
<code><a href="#topic+read.coastline.shapefile">read.coastline.shapefile</a>()</code>,
<code><a href="#topic+subset+2Ccoastline-method">subset,coastline-method</a></code>
</p>

<hr>
<h2 id='summary+2Cctd-method'>Summarize a ctd Object</h2><span id='topic+summary+2Cctd-method'></span><span id='topic+summary.ctd'></span>

<h3>Description</h3>

<p>Summarizes some of the data in a <code>ctd</code> object, presenting such information
as the station name, sampling location, data ranges, etc. If the object was read
from a <code>.cnv</code> file or a <code>.rsk</code> file, then the <code>OriginalName</code>
column for the data summary will contain the original names of data within
the source file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ctd'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Cctd-method_+3A_object">object</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Cctd-method_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
summary(ctd)

</code></pre>

<hr>
<h2 id='summary+2Cechosounder-method'>Summarize an echosounder Object</h2><span id='topic+summary+2Cechosounder-method'></span>

<h3>Description</h3>

<p>Summarizes some of the data in an <a href="#topic+echosounder-class">echosounder</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'echosounder'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Cechosounder-method_+3A_object">object</code></td>
<td>
<p>an object of class <code>"echosounder"</code>, usually, a result of
a call to <code><a href="#topic+read.echosounder">read.echosounder()</a></code>, <code><a href="#topic+read.oce">read.oce()</a></code>, or
<code><a href="#topic+as.echosounder">as.echosounder()</a></code>.</p>
</td></tr>
<tr><td><code id="summary+2B2Cechosounder-method_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to echosounder data: 
<code><a href="#topic++5B+5B+2Cechosounder-method">[[,echosounder-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cechosounder-method">[[&lt;-,echosounder-method</a></code>,
<code><a href="#topic+as.echosounder">as.echosounder</a>()</code>,
<code><a href="#topic+echosounder-class">echosounder-class</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+findBottom">findBottom</a>()</code>,
<code><a href="#topic+plot+2Cechosounder-method">plot,echosounder-method</a></code>,
<code><a href="#topic+read.echosounder">read.echosounder</a>()</code>,
<code><a href="#topic+subset+2Cechosounder-method">subset,echosounder-method</a></code>
</p>

<hr>
<h2 id='summary+2Cgps-method'>Summarize a gps Object</h2><span id='topic+summary+2Cgps-method'></span>

<h3>Description</h3>

<p>Summarize a <a href="#topic+gps-class">gps</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'gps'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Cgps-method_+3A_object">object</code></td>
<td>
<p>an object of class <code>"gps"</code></p>
</td></tr>
<tr><td><code id="summary+2B2Cgps-method_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to gps data: 
<code><a href="#topic++5B+5B+2Cgps-method">[[,gps-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cgps-method">[[&lt;-,gps-method</a></code>,
<code><a href="#topic+as.gps">as.gps</a>()</code>,
<code><a href="#topic+gps-class">gps-class</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+read.gps">read.gps</a>()</code>
</p>

<hr>
<h2 id='summary+2Cladp-method'>Summarize an ladp Object</h2><span id='topic+summary+2Cladp-method'></span>

<h3>Description</h3>

<p>Pertinent summary information is presented, including the station name,
sampling location, data ranges, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ladp'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Cladp-method_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+ladp-class">ladp</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Cladp-method_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing statistics of the elements of the <code>data</code> slot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ladp data: 
<code><a href="#topic++5B+5B+2Cladp-method">[[,ladp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cladp-method">[[&lt;-,ladp-method</a></code>,
<code><a href="#topic+as.ladp">as.ladp</a>()</code>,
<code><a href="#topic+ladp-class">ladp-class</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>
</p>

<hr>
<h2 id='summary+2Clandsat-method'>Summarize a landsat Object</h2><span id='topic+summary+2Clandsat-method'></span>

<h3>Description</h3>

<p>Provides a summary of a some information about a <a href="#topic+landsat-class">landsat</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'landsat'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Clandsat-method_+3A_object">object</code></td>
<td>
<p>A <a href="#topic+landsat-class">landsat</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Clandsat-method_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to landsat data: 
<code><a href="#topic++5B+5B+2Clandsat-method">[[,landsat-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clandsat-method">[[&lt;-,landsat-method</a></code>,
<code><a href="#topic+landsat-class">landsat-class</a></code>,
<code><a href="#topic+landsatAdd">landsatAdd</a>()</code>,
<code><a href="#topic+landsatTrim">landsatTrim</a>()</code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+read.landsat">read.landsat</a>()</code>
</p>

<hr>
<h2 id='summary+2Clisst-method'>Summarize a lisst Object</h2><span id='topic+summary+2Clisst-method'></span>

<h3>Description</h3>

<p>Summarizes some of the data in a <code>lisst</code> object, presenting such information
as the station name, sampling location, data ranges, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lisst'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Clisst-method_+3A_object">object</code></td>
<td>
<p>a <a href="#topic+lisst-class">lisst</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Clisst-method_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to lisst data: 
<code><a href="#topic++5B+5B+2Clisst-method">[[,lisst-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clisst-method">[[&lt;-,lisst-method</a></code>,
<code><a href="#topic+as.lisst">as.lisst</a>()</code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+read.lisst">read.lisst</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(lisst)
summary(lisst)

</code></pre>

<hr>
<h2 id='summary+2Clobo-method'>Summarize a lobo Object</h2><span id='topic+summary+2Clobo-method'></span>

<h3>Description</h3>

<p>Pertinent summary information is presented, including the sampling interval,
data ranges, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lobo'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Clobo-method_+3A_object">object</code></td>
<td>
<p>a <a href="#topic+lobo-class">lobo</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Clobo-method_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing statistics of the elements of the <code>data</code>
slot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for <a href="#topic+lobo-class">lobo</a> explains the
structure of LOBO objects, and also outlines the other functions dealing
with them.
</p>
<p>Other things related to lobo data: 
<code><a href="#topic++5B+5B+2Clobo-method">[[,lobo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Clobo-method">[[&lt;-,lobo-method</a></code>,
<code><a href="#topic+as.lobo">as.lobo</a>()</code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+read.lobo">read.lobo</a>()</code>,
<code><a href="#topic+subset+2Clobo-method">subset,lobo-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(lobo)
summary(lobo)
</code></pre>

<hr>
<h2 id='summary+2Cmet-method'>Summarize a met Object</h2><span id='topic+summary+2Cmet-method'></span>

<h3>Description</h3>

<p>Pertinent summary information is presented, including the sampling location,
data ranges, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'met'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Cmet-method_+3A_object">object</code></td>
<td>
<p>a <a href="#topic+met-class">met</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Cmet-method_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to met data: 
<code><a href="#topic++5B+5B+2Cmet-method">[[,met-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cmet-method">[[&lt;-,met-method</a></code>,
<code><a href="#topic+as.met">as.met</a>()</code>,
<code><a href="#topic+download.met">download.met</a>()</code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+read.met">read.met</a>()</code>,
<code><a href="#topic+subset+2Cmet-method">subset,met-method</a></code>
</p>

<hr>
<h2 id='summary+2Coce-method'>Summarize an oce Object</h2><span id='topic+summary+2Coce-method'></span>

<h3>Description</h3>

<p>Provide a textual summary of some pertinent aspects of the object, including
selected components of its <code>metadata</code> slot, statistical and
dimensional information on the entries in the <code>data</code> slot,
and a listing of the contents of its <code>processingLog</code> slot.
The details depend on the class of the object, especially for
the <code>metadata</code> slot, so it can help to consult the specialized
documentation, e.g. <a href="#topic+summary+2Cctd-method">summary,ctd-method</a> for CTD objects
(i.e. objects inheriting from the <a href="#topic+ctd-class">ctd</a> class.)
It is important to note that this is not
a good way to learn the details of the object contents. Instead,
for an object named <code>object</code>, say, one might use <code>str(object)</code>
to learn about all the contents, or <code>str(object[["metadata"]])</code>
to learn about the <code>metadata</code>, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'oce'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Coce-method_+3A_object">object</code></td>
<td>
<p>The object to be summarized.</p>
</td></tr>
<tr><td><code id="summary+2B2Coce-method_+3A_...">...</code></td>
<td>
<p>Extra arguments (ignored)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>o &lt;- new("oce")
summary(o)
</code></pre>

<hr>
<h2 id='summary+2Codf-method'>Summarize an odf Object</h2><span id='topic+summary+2Codf-method'></span>

<h3>Description</h3>

<p>Pertinent summary information is presented, including the station name,
sampling location, data ranges, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'odf'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Codf-method_+3A_object">object</code></td>
<td>
<p>an <a href="#topic+odf-class">odf</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Codf-method_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing statistics of the elements of the <code>data</code>
slot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to odf data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+ODF2oce">ODF2oce</a>()</code>,
<code><a href="#topic+ODFListFromHeader">ODFListFromHeader</a>()</code>,
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic++5B+5B+2Codf-method">[[,odf-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Codf-method">[[&lt;-,odf-method</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.odf">read.odf</a>()</code>,
<code><a href="#topic+subset+2Codf-method">subset,odf-method</a></code>
</p>

<hr>
<h2 id='summary+2Crsk-method'>Summarize a rsk Object</h2><span id='topic+summary+2Crsk-method'></span>

<h3>Description</h3>

<p>Summarizes some of the data in a <a href="#topic+rsk-class">rsk</a> object, presenting such information
as the station name, sampling location, data ranges, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'rsk'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Crsk-method_+3A_object">object</code></td>
<td>
<p>An <a href="#topic+rsk-class">rsk</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Crsk-method_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for <a href="#topic+rsk-class">rsk</a> explains the structure
of CTD objects, and also outlines the other functions dealing with them.
</p>
<p>Other things related to rsk data: 
<code><a href="#topic++5B+5B+2Crsk-method">[[,rsk-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Crsk-method">[[&lt;-,rsk-method</a></code>,
<code><a href="#topic+as.rsk">as.rsk</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+read.rsk">read.rsk</a>()</code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+rskPatm">rskPatm</a>()</code>,
<code><a href="#topic+rskToc">rskToc</a>()</code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+subset+2Crsk-method">subset,rsk-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(rsk)
summary(rsk)

</code></pre>

<hr>
<h2 id='summary+2Csatellite-method'>Summarize a satellite Object</h2><span id='topic+summary+2Csatellite-method'></span>

<h3>Description</h3>

<p>Summarize a satellite Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'satellite'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Csatellite-method_+3A_object">object</code></td>
<td>
<p>a <a href="#topic+satellite-class">satellite</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Csatellite-method_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>

<hr>
<h2 id='summary+2Csealevel-method'>Summarize a sealevel Object</h2><span id='topic+summary+2Csealevel-method'></span>

<h3>Description</h3>

<p>Summarizes some of the data in a sealevel object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'sealevel'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Csealevel-method_+3A_object">object</code></td>
<td>
<p>A <a href="#topic+sealevel-class">sealevel</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Csealevel-method_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing statistics of the elements of the <code>data</code>
slot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to sealevel data: 
<code><a href="#topic++5B+5B+2Csealevel-method">[[,sealevel-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csealevel-method">[[&lt;-,sealevel-method</a></code>,
<code><a href="#topic+as.sealevel">as.sealevel</a>()</code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+read.sealevel">read.sealevel</a>()</code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+subset+2Csealevel-method">subset,sealevel-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(sealevel)
summary(sealevel)

</code></pre>

<hr>
<h2 id='summary+2Csection-method'>Summarize a section Object</h2><span id='topic+summary+2Csection-method'></span><span id='topic+summary.section'></span>

<h3>Description</h3>

<p>Pertinent summary information is presented, including station locations,
distance along trac, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'section'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Csection-method_+3A_object">object</code></td>
<td>
<p>An object of class <code>"section"</code>, usually, a result of a call
to <code><a href="#topic+read.section">read.section()</a></code>, <code><a href="#topic+read.oce">read.oce()</a></code>, or
<code><a href="#topic+as.section">as.section()</a></code>.</p>
</td></tr>
<tr><td><code id="summary+2B2Csection-method_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>NULL</code>
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to section data: 
<code><a href="#topic++5B+5B+2Csection-method">[[,section-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Csection-method">[[&lt;-,section-method</a></code>,
<code><a href="#topic+as.section">as.section</a>()</code>,
<code><a href="#topic+handleFlags+2Csection-method">handleFlags,section-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Csection-method">initializeFlagScheme,section-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+read.section">read.section</a>()</code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+sectionAddStation">sectionAddStation</a>()</code>,
<code><a href="#topic+sectionGrid">sectionGrid</a>()</code>,
<code><a href="#topic+sectionSmooth">sectionSmooth</a>()</code>,
<code><a href="#topic+sectionSort">sectionSort</a>()</code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+subset+2Csection-method">subset,section-method</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(section)
summary(section)

</code></pre>

<hr>
<h2 id='summary+2Ctidem-method'>Summarize a tidem Object</h2><span id='topic+summary+2Ctidem-method'></span>

<h3>Description</h3>

<p>By default, all fitted constituents are plotted, but it is quite useful to
set e.g. p=0.05 To see just those constituents that are significant at the 5
percent level.
Note that the p values are estimated as the average of the p values for the
sine and cosine components at a given frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'tidem'
summary(object, p = 1, constituent, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Ctidem-method_+3A_object">object</code></td>
<td>
<p>an object of class <a href="#topic+tidem">tidem</a>, as created by
<code><a href="#topic+as.tidem">as.tidem()</a></code> or <code><a href="#topic+tidem">tidem()</a></code>.</p>
</td></tr>
<tr><td><code id="summary+2B2Ctidem-method_+3A_p">p</code></td>
<td>
<p>optional value of the maximum p value for the display of an
individual coefficient.  If not given, all coefficients are shown.</p>
</td></tr>
<tr><td><code id="summary+2B2Ctidem-method_+3A_constituent">constituent</code></td>
<td>
<p>optional character vector holding the names
of constituents on which to focus.</p>
</td></tr>
<tr><td><code id="summary+2B2Ctidem-method_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>NULL</code>
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
data(sealevel)
tide &lt;- tidem(sealevel)
summary(tide)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>

<hr>
<h2 id='summary+2Ctopo-method'>Summarize a topo Object</h2><span id='topic+summary+2Ctopo-method'></span>

<h3>Description</h3>

<p>Pertinent summary information is presented, including the longitude and
latitude range, and the range of elevation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'topo'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Ctopo-method_+3A_object">object</code></td>
<td>
<p>A <a href="#topic+topo-class">topo</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Ctopo-method_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing statistics of the elements of the <code>data</code> slot.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to topo data: 
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic+as.topo">as.topo</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+read.topo">read.topo</a>()</code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+topoInterpolate">topoInterpolate</a>()</code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(topoWorld)
summary(topoWorld)

</code></pre>

<hr>
<h2 id='summary+2Cwindrose-method'>Summarize a windrose Object</h2><span id='topic+summary+2Cwindrose-method'></span>

<h3>Description</h3>

<p>Summarizes some of the data in a <a href="#topic+windrose-class">windrose</a> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'windrose'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Cwindrose-method_+3A_object">object</code></td>
<td>
<p>A <a href="#topic+windrose-class">windrose</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Cwindrose-method_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to windrose data: 
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic+as.windrose">as.windrose</a>()</code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>
</p>

<hr>
<h2 id='summary+2Cxbt-method'>Summarize an xbt Object</h2><span id='topic+summary+2Cxbt-method'></span>

<h3>Description</h3>

<p>Summarizes some of the data in a <code>xbt</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'xbt'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary+2B2Cxbt-method_+3A_object">object</code></td>
<td>
<p>A <a href="#topic+xbt-class">xbt</a> object.</p>
</td></tr>
<tr><td><code id="summary+2B2Cxbt-method_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The documentation for the <a href="#topic+xbt-class">xbt</a> class explains the structure
of <code>xbt</code> objects, and also outlines the other functions dealing with them.
</p>
<p>Other things related to xbt data: 
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>,
<code><a href="#topic+as.xbt">as.xbt</a>()</code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1</a>()</code>,
<code><a href="#topic+read.xbt">read.xbt</a>()</code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>

<hr>
<h2 id='sunAngle'>Solar Angle as Function of Space and Time</h2><span id='topic+sunAngle'></span>

<h3>Description</h3>

<p>This calculates solar angle, based on a NASA-provided Fortran
program, which (according to comments in the code) is in turn
based on &quot;The Astronomical Almanac&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sunAngle(t, longitude = 0, latitude = 0, useRefraction = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sunAngle_+3A_t">t</code></td>
<td>
<p>time, a POSIXt object (converted to timezone <code>"UTC"</code>,
if it is not already in that timezone), a character or numeric value that
corresponds to such a time.  This may be a single value or a vector
of values.  In the latter case, <code>longitude</code>, <code>latitude</code> and <code>useRefraction</code>
are all made to be of the same length as <code>time</code>, by calling <code><a href="base.html#topic+rep">rep()</a></code>.</p>
</td></tr>
<tr><td><code id="sunAngle_+3A_longitude">longitude</code></td>
<td>
<p>observer longitude in degrees east.</p>
</td></tr>
<tr><td><code id="sunAngle_+3A_latitude">latitude</code></td>
<td>
<p>observer latitude in degrees north.</p>
</td></tr>
<tr><td><code id="sunAngle_+3A_userefraction">useRefraction</code></td>
<td>
<p>boolean, set to <code>TRUE</code> to apply a correction for
atmospheric refraction.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following:
</p>

<ul>
<li> <p><code>time</code> the time
</p>
</li>
<li> <p><code>azimuth</code>, in degrees eastward of north, from 0 to 360.
</p>
</li>
<li> <p><code>altitude</code>, in degrees above the horizon,  ranging from -90 to 90.
</p>
</li>
<li> <p><code>diameter</code>, solar diameter, in degrees.
</p>
</li>
<li> <p><code>distance</code> to sun, in astronomical units.
</p>
</li>
<li> <p><code>declination</code> angle in degrees, computed with <code><a href="#topic+sunDeclinationRightAscension">sunDeclinationRightAscension()</a></code>.
</p>
</li>
<li> <p><code>rightAscension</code> angle in degrees, computed with <code><a href="#topic+sunDeclinationRightAscension">sunDeclinationRightAscension()</a></code>.
<img src="../help/figures/starCoords.png" width="400" alt="starCoords.png" />
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Regarding <code>declination</code> and <code>rightAscension</code>, see
references in the documentation for <code><a href="#topic+sunDeclinationRightAscension">sunDeclinationRightAscension()</a></code>.
The other items are based on Fortran code retrieved from
the file <code>sunae.f</code>, downloaded from the ftp site
<code>climate1.gsfc.nasa.gov/wiscombe/Solar_Rad/SunAngles</code>
on 2009-11-1.  Comments in that code list as references:
</p>
<p>Michalsky, J., 1988: The Astronomical Almanac's algorithm for approximate
solar position (1950-2050), Solar Energy 40, 227-235
</p>
<p>The Astronomical Almanac, U.S. Gov't Printing Office, Washington, D.C.
(published every year).
</p>
<p>The code comments suggest that the appendix in Michalsky (1988) contains
errors, and declares the use of the following formulae in the 1995 version
the Almanac:
</p>

<ul>
<li><p> p. A12: approximation to sunrise/set times
</p>
</li>
<li><p> p. B61: solar altitude (AKA elevation) and azimuth
</p>
</li>
<li><p> p. B62: refraction correction
</p>
</li>
<li><p> p. C24: mean longitude, mean anomaly, ecliptic
longitude, obliquity of ecliptic, right ascension, declination, Earth-Sun
distance, angular diameter of Sun
</p>
</li>
<li><p> p. L2: Greenwich mean sidereal time (ignoring T^2, T^3 terms)
</p>
</li></ul>

<p>The code lists authors as Dr. Joe Michalsky and Dr. Lee Harrison (State
University of New York), with modifications by Dr. Warren Wiscombe (NASA
Goddard Space Flight Center).
</p>


<h3>See Also</h3>

<p>The corresponding function for the moon is <code><a href="#topic+moonAngle">moonAngle()</a></code>.
</p>
<p>Other things related to astronomy: 
<code><a href="#topic+angle2hms">angle2hms</a>()</code>,
<code><a href="#topic+eclipticalToEquatorial">eclipticalToEquatorial</a>()</code>,
<code><a href="#topic+equatorialToLocalHorizontal">equatorialToLocalHorizontal</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+moonAngle">moonAngle</a>()</code>,
<code><a href="#topic+siderealTime">siderealTime</a>()</code>,
<code><a href="#topic+sunDeclinationRightAscension">sunDeclinationRightAscension</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rise &lt;- as.POSIXct("2011-03-03 06:49:00", tz = "UTC") + 4 * 3600
set &lt;- as.POSIXct("2011-03-03 18:04:00", tz = "UTC") + 4 * 3600
mismatch &lt;- function(lonlat) {
    sunAngle(rise, lonlat[1], lonlat[2])$altitude^2 + sunAngle(set, lonlat[1], lonlat[2])$altitude^2
}
result &lt;- optim(c(1, 1), mismatch)
lonHfx &lt;- (-63.55274)
latHfx &lt;- 44.65
dist &lt;- geodDist(result$par[1], result$par[2], lonHfx, latHfx)
cat(sprintf(
    "Infer Halifax latitude %.2f and longitude %.2f; distance mismatch %.0f km",
    result$par[2], result$par[1], dist
))

</code></pre>

<hr>
<h2 id='sunDeclinationRightAscension'>Sun Declination and Right Ascension</h2><span id='topic+sunDeclinationRightAscension'></span>

<h3>Description</h3>

<p>The formulae are from Meeus (1991), chapter 24 (which uses chapter 21).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sunDeclinationRightAscension(time, apparent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sunDeclinationRightAscension_+3A_time">time</code></td>
<td>
<p>a POSIXct time. This ought to be in UTC timezone; if not,
the behaviour of this function is unlikely to be correct.</p>
</td></tr>
<tr><td><code id="sunDeclinationRightAscension_+3A_apparent">apparent</code></td>
<td>
<p>logical value indicating whether to return
the 'apparent' angles.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing <code>declination</code> and <code>rightAscension</code>, in degrees.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley, based on formulae in Meeus (1991).
</p>


<h3>References</h3>


<ul>
<li><p> Meeus, Jean. Astronomical Algorithms. Second Edition.
Richmond, Virginia, USA: Willmann-Bell, 1991.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other things related to astronomy: 
<code><a href="#topic+angle2hms">angle2hms</a>()</code>,
<code><a href="#topic+eclipticalToEquatorial">eclipticalToEquatorial</a>()</code>,
<code><a href="#topic+equatorialToLocalHorizontal">equatorialToLocalHorizontal</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+moonAngle">moonAngle</a>()</code>,
<code><a href="#topic+siderealTime">siderealTime</a>()</code>,
<code><a href="#topic+sunAngle">sunAngle</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 24.a in Meeus (1991) (page 158 PDF, 153 print)
time &lt;- as.POSIXct("1992-10-13 00:00:00", tz = "UTC")
a &lt;- sunDeclinationRightAscension(time, apparent = TRUE)
stopifnot(abs(a$declination - (-7.78507)) &lt; 0.00004)
stopifnot(abs(a$rightAscension - (-161.61919)) &lt; 0.00003)
b &lt;- sunDeclinationRightAscension(time)
# check against previous results, to protect aginst code-drift errors
stopifnot(abs(b$declination - (-7.785464443)) &lt; 0.000000001)
stopifnot(abs(b$rightAscension - (-161.6183305)) &lt; 0.0000001)

</code></pre>

<hr>
<h2 id='swAbsoluteSalinity'>Seawater Absolute Salinity (GSW Formulation)</h2><span id='topic+swAbsoluteSalinity'></span>

<h3>Description</h3>

<p>Compute the seawater Absolute Salinity, according to the GSW/TEOS-10
formulation with <code><a href="gsw.html#topic+gsw_SA_from_SP">gsw::gsw_SA_from_SP()</a></code> in the <a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a> package.
Typically, this is a fraction of a unit
higher than practical salinity as defined in the UNESCO formulae.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swAbsoluteSalinity(
  salinity,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swAbsoluteSalinity_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object (in
which case <code>salinity</code>, etc. are inferred from the object).</p>
</td></tr>
<tr><td><code id="swAbsoluteSalinity_+3A_pressure">pressure</code></td>
<td>
<p>pressure in dbar.</p>
</td></tr>
<tr><td><code id="swAbsoluteSalinity_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation.</p>
</td></tr>
<tr><td><code id="swAbsoluteSalinity_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Absolute Salinity in <code class="reqn">g/kg</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>McDougall, T.J. and P.M. Barker, 2011: Getting started with
TEOS-10 and the Gibbs Seawater (GSW) Oceanographic Toolbox, 28pp.,
SCOR/IAPSO WG127, ISBN 978-0-646-55621-5.
</p>


<h3>See Also</h3>

<p>The related TEOS-10 quantity &ldquo;conservative temperature&rdquo; may be
computed with <code><a href="#topic+swConservativeTemperature">swConservativeTemperature()</a></code>.  For a ctd object,
absolute salinity may also be recovered by indexing as e.g.
<code>ctd[["absoluteSalinity"]]</code> or <code>ctd[["SA"]]</code>.
</p>
<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>swAbsoluteSalinity(35.5, 300, 260, 16) # 35.67136

</code></pre>

<hr>
<h2 id='swAlpha'>Seawater Thermal Expansion Coefficient</h2><span id='topic+swAlpha'></span>

<h3>Description</h3>

<p>Compute <code class="reqn">\alpha</code>, the thermal expansion coefficient for seawater.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swAlpha(
  salinity,
  temperature = NULL,
  pressure = 0,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swAlpha_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object (in
which case <code>salinity</code>, etc. are inferred from the object).</p>
</td></tr>
<tr><td><code id="swAlpha_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale; see &ldquo;Temperature units&rdquo; in the documentation for
<code><a href="#topic+swRho">swRho()</a></code>.</p>
</td></tr>
<tr><td><code id="swAlpha_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swAlpha_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swAlpha_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swAlpha_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> or <code>"gsw"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Value in 1/degC.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>The <code>eos="unesco"</code> formulae are based on the UNESCO
equation of state, but are formulated empirically by Trevor J. McDougall,
1987, Neutral Surfaces, Journal of Physical Oceanography, volume 17, pages
1950-1964. The <code>eos="gsw"</code> formulae come from GSW; see references in
the <code><a href="#topic+swRho">swRho()</a></code> documentation.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>

<hr>
<h2 id='swAlphaOverBeta'>Ratio of Seawater Thermal Expansion Coefficient to Haline Contraction Coefficient</h2><span id='topic+swAlphaOverBeta'></span>

<h3>Description</h3>

<p>Compute <code class="reqn">\alpha/\beta</code> using McDougall's (1987) algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swAlphaOverBeta(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swAlphaOverBeta_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object (in
which case <code>salinity</code>, etc. are inferred from the object).</p>
</td></tr>
<tr><td><code id="swAlphaOverBeta_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C)</p>
</td></tr>
<tr><td><code id="swAlphaOverBeta_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swAlphaOverBeta_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swAlphaOverBeta_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swAlphaOverBeta_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> or <code>"gsw"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Value in psu/<code class="reqn">^\circ</code>C.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>The <code>eos="unesco"</code> formulae are based on the UNESCO
equation of state, but are formulated empirically by Trevor J. McDougall,
1987, Neutral Surfaces, Journal of Physical Oceanography, volume 17, pages
1950-1964. The <code>eos="gsw"</code> formulae come from GSW; see references in
the <code><a href="#topic+swRho">swRho()</a></code> documentation.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>swAlphaOverBeta(40, 10, 4000, eos = "unesco") # 0.3476

</code></pre>

<hr>
<h2 id='swBeta'>Seawater Haline Contraction Coefficient</h2><span id='topic+swBeta'></span>

<h3>Description</h3>

<p>Compute <code class="reqn">\beta</code>, the haline contraction coefficient for seawater.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swBeta(
  salinity,
  temperature = NULL,
  pressure = 0,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swBeta_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object (in
which case <code>salinity</code>, etc. are inferred from the object).</p>
</td></tr>
<tr><td><code id="swBeta_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale; see &ldquo;Temperature units&rdquo; in the documentation for
<code><a href="#topic+swRho">swRho()</a></code>.</p>
</td></tr>
<tr><td><code id="swBeta_+3A_pressure">pressure</code></td>
<td>
<p>seawater pressure (dbar)</p>
</td></tr>
<tr><td><code id="swBeta_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swBeta_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swBeta_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> or <code>"gsw"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Value in 1/psu.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>The <code>eos="unesco"</code> formulae are based on the UNESCO
equation of state, but are formulated empirically by Trevor J. McDougall,
1987, Neutral Surfaces, Journal of Physical Oceanography, volume 17, pages
1950-1964. The <code>eos="gsw"</code> formulae come from GSW; see references in
the <code><a href="#topic+swRho">swRho()</a></code> documentation.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>

<hr>
<h2 id='swConservativeTemperature'>Seawater Conservative Temperature (GSW Formulation)</h2><span id='topic+swConservativeTemperature'></span>

<h3>Description</h3>

<p>Compute seawater Conservative Temperature, according to the GSW/TEOS-10
formulation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swConservativeTemperature(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swConservativeTemperature_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object (in
which case <code>salinity</code>, etc. are inferred from the object).</p>
</td></tr>
<tr><td><code id="swConservativeTemperature_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale; see &ldquo;Temperature units&rdquo; in the documentation for
<code><a href="#topic+swRho">swRho()</a></code>.</p>
</td></tr>
<tr><td><code id="swConservativeTemperature_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swConservativeTemperature_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation.</p>
</td></tr>
<tr><td><code id="swConservativeTemperature_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the first argument is an <code>oce</code> object, then values for salinity,
etc., are extracted from it, and used for the calculation, and the
corresponding arguments to the present function are ignored.
</p>
<p>The conservative temperature is calculated using the TEOS-10 function
<a href="gsw.html#topic+gsw_CT_from_t">gsw::gsw_CT_from_t</a> from the <a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a> package.
</p>


<h3>Value</h3>

<p>Conservative temperature in degrees Celcius.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>McDougall, T.J. and P.M. Barker, 2011: Getting started with
TEOS-10 and the Gibbs Seawater (GSW) Oceanographic Toolbox, 28pp.,
SCOR/IAPSO WG127, ISBN 978-0-646-55621-5.
</p>


<h3>See Also</h3>

<p>The related TEOS-10 quantity &ldquo;absolute salinity&rdquo; may be computed
with <code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity()</a></code>.  For a ctd object, conservative
temperature may also be recovered by indexing as e.g.
<code>ctd[["conservativeTemperature"]]</code> or <code>ctd[["CT"]]</code>.
</p>
<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>swConservativeTemperature(35, 10, 1000, 188, 4) # 9.86883

</code></pre>

<hr>
<h2 id='swCSTp'>Electrical Conductivity Ratio From Salinity, Temperature and Pressure</h2><span id='topic+swCSTp'></span>

<h3>Description</h3>

<p>Compute electrical conductivity ratio based on salinity, temperature, and
pressure (relative to the conductivity of seawater with salinity=35,
temperature68=15, and pressure=0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swCSTp(
  salinity,
  temperature = 15,
  pressure = 0,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swCSTp_+3A_salinity">salinity</code></td>
<td>
<p>practical salinity, or a CTD object (in which case its
temperature and pressure are used, and the next two arguments are ignored)</p>
</td></tr>
<tr><td><code id="swCSTp_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale; see the examples, as well as the
&ldquo;Temperature units&rdquo; section in the documentation for <code><a href="#topic+swRho">swRho()</a></code>.</p>
</td></tr>
<tr><td><code id="swCSTp_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swCSTp_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> or <code>"gsw"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>eos="unesco"</code>, the calculation is done by a bisection root search
on the UNESCO formula relating salinity to conductivity, temperature, and
pressure (see <code><a href="#topic+swSCTp">swSCTp()</a></code>).  If it is <code>"gsw"</code> then the
Gibbs-SeaWater formulation is used, via <code><a href="gsw.html#topic+gsw_C_from_SP">gsw_C_from_SP()</a></code>.
</p>


<h3>Value</h3>

<p>Conductivity ratio (unitless), i.e. the ratio of conductivity to the
conductivity at salinity=35, temperature=15 (IPTS-68 scale) and pressure=0,
which has numerical value 42.9140 mS/cm = 4.29140 S/m (see
Culkin and Smith, 1980, in the regression result cited at the bottom of
the left-hand column on page 23).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Fofonoff, P. and R. C. Millard Jr, 1983. Algorithms for
computation of fundamental properties of seawater. <em>Unesco Technical
Papers in Marine Science</em>, <em>44</em>, 53 pp.
</p>
</li>
<li><p> Culkin, F., and Norman D. Smith, 1980. Determination of the concentration of
potassium chloride solution having the same electrical conductivity, at 15 C
and infinite frequency, as standard seawater of salinity 35.0000 ppt
(Chlorinity 19.37394 ppt). <em>IEEE Journal of Oceanic Engineering</em>,
<em>5</em>, pp 22-23.
</p>
</li></ol>



<h3>See Also</h3>

<p>For thermal (as opposed to electrical) conductivity, see
<code><a href="#topic+swThermalConductivity">swThermalConductivity()</a></code>. For computation of salinity from
electrical conductivity, see <code><a href="#topic+swSCTp">swSCTp()</a></code>.
</p>
<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stopifnot(abs(1.0 - swCSTp(35, T90fromT68(15), 0, eos = "unesco")) &lt; 1e-7)
stopifnot(abs(1.0 - swCSTp(34.25045, T90fromT68(15), 2000, eos = "unesco")) &lt; 1e-7)
stopifnot(abs(1.0 - swCSTp(34.25045, T90fromT68(15), 2000, eos = "gsw")) &lt; 1e-7)

</code></pre>

<hr>
<h2 id='swDepth'>Water Depth</h2><span id='topic+swDepth'></span>

<h3>Description</h3>

<p>Compute depth below the surface (i.e. a positive number within the
water column) based on pressure and latitude. (Use <code><a href="#topic+swZ">swZ()</a></code>
to get the vertical coordinate, which is negative within the water
column.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swDepth(pressure, latitude = 45, eos = getOption("oceEOS", default = "gsw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swDepth_+3A_pressure">pressure</code></td>
<td>
<p>either pressure (dbar), in which case <code>lat</code> must also
be given, or a <code>ctd</code> object, in which case <code>lat</code> will be inferred
from the object.</p>
</td></tr>
<tr><td><code id="swDepth_+3A_latitude">latitude</code></td>
<td>
<p>Latitude in <code class="reqn">^\circ</code>N.</p>
</td></tr>
<tr><td><code id="swDepth_+3A_eos">eos</code></td>
<td>
<p>indication of formulation to be used, either <code>"unesco"</code> or
<code>"gsw"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>eos="unesco"</code> then depth is calculated from pressure using Saunders
and Fofonoff's method, with the formula refitted for 1980 UNESCO equation of
state (reference 1).  If <code>eos="gsw"</code>, then <code><a href="gsw.html#topic+gsw_z_from_p">gsw::gsw_z_from_p()</a></code> from
the <a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a> package (references 2 and 3) is used.
</p>


<h3>Value</h3>

<p>Depth below the ocean surface, in metres.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Unesco 1983. Algorithms for computation of fundamental
properties of seawater, 1983. <em>Unesco Tech. Pap. in Mar. Sci.</em>, No. 44,
53 pp.
</p>
</li>
<li><p> IOC, SCOR, and IAPSO (2010). The international thermodynamic equation of
seawater-2010: Calculation and use of thermodynamic properties.  Technical
Report 56, Intergovernmental Oceanographic Commission, Manuals and Guide.
</p>
</li>
<li><p> McDougall, T.J. and P.M. Barker, 2011: Getting started with TEOS-10 and
the Gibbs Seawater (GSW) Oceanographic Toolbox, 28pp., SCOR/IAPSO WG127,
ISBN 978-0-646-55621-5.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>d &lt;- swDepth(10, 45)

</code></pre>

<hr>
<h2 id='swDynamicHeight'>Dynamic Height of a Seawater Profile</h2><span id='topic+swDynamicHeight'></span>

<h3>Description</h3>

<p>Compute the dynamic height of a column of seawater.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swDynamicHeight(
  x,
  referencePressure = 2000,
  subdivisions = 500,
  rel.tol = .Machine$double.eps^0.25,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swDynamicHeight_+3A_x">x</code></td>
<td>
<p>a <a href="#topic+section-class">section</a> object.</p>
</td></tr>
<tr><td><code id="swDynamicHeight_+3A_referencepressure">referencePressure</code></td>
<td>
<p>reference pressure (dbar). If this exceeds the
highest pressure supplied to <code><a href="#topic+swDynamicHeight">swDynamicHeight()</a></code>, then that highest
pressure is used, instead of the supplied value of
<code>referencePressure</code>.</p>
</td></tr>
<tr><td><code id="swDynamicHeight_+3A_subdivisions">subdivisions</code></td>
<td>
<p>number of subdivisions for call to
<code><a href="stats.html#topic+integrate">integrate()</a></code>.  (The default value is considerably larger than the
default for <code><a href="stats.html#topic+integrate">integrate()</a></code>, because otherwise some test profiles
failed to integrate.</p>
</td></tr>
<tr><td><code id="swDynamicHeight_+3A_rel.tol">rel.tol</code></td>
<td>
<p>absolute tolerance for call to <code><a href="stats.html#topic+integrate">integrate()</a></code>.  Note
that this call is made in scaled coordinates, i.e. pressure is divided by
its maximum value, and dz/dp is also divided by its maximum.</p>
</td></tr>
<tr><td><code id="swDynamicHeight_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> or <code>"gsw"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the first argument is a <code>section</code>, then dynamic height is calculated
for each station within a section, and returns a list containing distance
along the section along with dynamic height.
</p>
<p>If the first argument is a <code>ctd</code>, then this returns just a single
value, the dynamic height.
</p>
<p>If <code>eos="unesco"</code>, processing is as follows.  First, a piecewise-linear
model of the density variation with pressure is developed using
<code><a href="stats.html#topic+approxfun">stats::approxfun()</a></code>. (The option <code>rule=2</code> is used to
extrapolate the uppermost density up to the surface, preventing a possible a
bias for bottle data, in which the first depth may be a few metres below the
surface.) A second function is constructed as the density of water with
salinity 35PSU, temperature of 0<code class="reqn">^\circ</code>C, and pressure as in the
<code>ctd</code>. The difference of the reciprocals of these densities, is then
integrated with <code><a href="stats.html#topic+integrate">stats::integrate()</a></code> with pressure limits <code>0</code>
to <code>referencePressure</code>.  (For improved numerical results, the variables
are scaled before the integration, making both independent and dependent
variables be of order one.)
</p>
<p>If <code>eos="gsw"</code>, <code><a href="gsw.html#topic+gsw_geo_strf_dyn_height">gsw::gsw_geo_strf_dyn_height()</a></code> is used
to calculate a result in m^2/s^2, and this is divided by
9.7963<code class="reqn">m/s^2</code>.
If pressures are out of order, the data are sorted. If any pressure
is repeated, only the first level is used.
If there are under 4 remaining distinct
pressures, <code>NA</code> is returned, with a warning.
</p>


<h3>Value</h3>

<p>In the first form, a list containing <code>distance</code>, the distance
(km( from the first station in the section and <code>height</code>, the dynamic
height (m). In the second form, a single value, containing the
dynamic height (m).
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
data(section)

# Dynamic height and geostrophy
par(mfcol=c(2, 2))
par(mar=c(4.5, 4.5, 2, 1))

# Left-hand column: whole section
# (The smoothing lowers Gulf Stream speed greatly)
westToEast &lt;- subset(section, 1&lt;=stationId&amp;stationId&lt;=123)
dh &lt;- swDynamicHeight(westToEast)
plot(dh$distance, dh$height, type="p", xlab="", ylab="dyn. height [m]")
ok &lt;- !is.na(dh$height)
smu &lt;- supsmu(dh$distance, dh$height)
lines(smu, col="blue")
f &lt;- coriolis(section[["station", 1]][["latitude"]])
g &lt;- gravity(section[["station", 1]][["latitude"]])
v &lt;- diff(smu$y)/diff(smu$x) * g / f / 1e3 # 1e3 converts to m
plot(smu$x[-1], v, type="l", col="blue", xlab="distance [km]", ylab="velocity (m/s)")

# right-hand column: gulf stream region, unsmoothed
gs &lt;- subset(section, 102&lt;=stationId&amp;stationId&lt;=124)
dh.gs &lt;- swDynamicHeight(gs)
plot(dh.gs$distance, dh.gs$height, type="b", xlab="", ylab="dyn. height [m]")
v &lt;- diff(dh.gs$height)/diff(dh.gs$distance) * g / f / 1e3
plot(dh.gs$distance[-1], v, type="l", col="blue",
  xlab="distance [km]", ylab="velocity (m/s)")
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Gill, A.E., 1982. <em>Atmosphere-ocean Dynamics</em>, Academic
Press, New York, 662 pp.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>

<hr>
<h2 id='swLapseRate'>Seawater Lapse Rate</h2><span id='topic+swLapseRate'></span>

<h3>Description</h3>

<p>Compute adiabatic lapse rate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swLapseRate(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swLapseRate_+3A_salinity">salinity</code></td>
<td>
<p>either salinity (PSU) (in which case <code>temperature</code> and
<code>pressure</code> must be provided) <em>or</em> a <code>ctd</code> object (in which
case <code>salinity</code>, <code>temperature</code> and <code>pressure</code> are determined
from the object, and must not be provided in the argument list).</p>
</td></tr>
<tr><td><code id="swLapseRate_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale; see &ldquo;Temperature units&rdquo; in the documentation for
<code><a href="#topic+swRho">swRho()</a></code>.</p>
</td></tr>
<tr><td><code id="swLapseRate_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swLapseRate_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swLapseRate_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swLapseRate_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2)
or <code>"gsw"</code> (references 3 and 4).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>eos="unesco"</code>, the density is calculated using the UNESCO equation
of state for seawater (references 1 and 2), and if <code>eos="gsw"</code>, the GSW formulation
(references 3 and 4) is used.
</p>


<h3>Value</h3>

<p>Lapse rate (<code class="reqn">deg</code>C/m).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Fofonoff, P. and R. C. Millard Jr, 1983. Algorithms for
computation of fundamental properties of seawater.
<em>Unesco Technical Papers in Marine Science</em>,
<em>44</em>, 53 pp.  (Section 7, pages 38-40)
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lr &lt;- swLapseRate(40, 40, 10000) # 3.255976e-4

</code></pre>

<hr>
<h2 id='swN2'>Squared Buoyancy Frequency for Seawater</h2><span id='topic+swN2'></span>

<h3>Description</h3>

<p>Compute <code class="reqn">N^2</code>, the square of the buoyancy frequency for a seawater
profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swN2(
  pressure,
  sigmaTheta = NULL,
  derivs,
  df,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swN2_+3A_pressure">pressure</code></td>
<td>
<p>either pressure (dbar) (in which case <code>sigmaTheta</code> must
be provided) <em>or</em> an object of class <code>ctd</code> object (in which case
<code>sigmaTheta</code> is inferred from the object.</p>
</td></tr>
<tr><td><code id="swN2_+3A_sigmatheta">sigmaTheta</code></td>
<td>
<p>Surface-referenced potential density minus 1000
(kg/m<code class="reqn">^3</code>).</p>
</td></tr>
<tr><td><code id="swN2_+3A_derivs">derivs</code></td>
<td>
<p>optional argument to control how the derivative
<code class="reqn">d\sigma_\theta/dp</code> is calculated.  This may
be a character string or a function of two arguments.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="swN2_+3A_df">df</code></td>
<td>
<p>argument passed to <code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code> if this function is
used for smoothing; set to <code>NA</code> to prevent smoothing.</p>
</td></tr>
<tr><td><code id="swN2_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
<tr><td><code id="swN2_+3A_...">...</code></td>
<td>
<p>additional argument, passed to <code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code>, in
the case that <code>derivs="smoothing"</code>.  See &ldquo;Details&rdquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Smoothing is often useful prior to computing buoyancy frequency, and so this
may optionally be done with <code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code>, unless
<code>df=NA</code>, in which case raw data are used.  If <code>df</code> is not
provided, a possibly reasonable value computed from an analysis of the
profile, based on the number of pressure levels.
</p>
<p>The core of the method involves computing potential density referenced to median
pressure, using the UNESCO-style <a href="#topic+swSigmaTheta">swSigmaTheta</a> function, and then differentiating
this with respect to pressure. The <code>derivs</code> argument is used
to control how this is done, as follows.
</p>

<ul>
<li><p> If <code>derivs</code> is not supplied, the action is as though it were
given as the string <code>"smoothing"</code>
</p>
</li>
<li><p> If <code>derivs</code> equals <code>"simple"</code>, then the derivative of
density with respect to pressure is calculated as the ratio of first-order
derivatives of density and pressure, each calculated using
<code><a href="base.html#topic+diff">diff()</a></code>.  (A zero is appended at the top level.)
</p>
</li>
<li><p> If <code>derivs</code> equals <code>"smoothing"</code>, then the processing
depends on the number of data in the profile, and on whether <code>df</code> is
given as an optional argument.  When the number of points exceeds 4, and
when <code>df</code> exceeds 1, <code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code> is used to calculate
smoothing spline representation the variation of density as a function of
pressure, and derivatives are extracted from the spline using
<code>predict</code>.  Otherwise, density is smoothed using <code><a href="stats.html#topic+smooth">smooth()</a></code>,
and derivatives are calculated as with the <code>"simple"</code> method.
</p>
</li>
<li><p> If <code>derivs</code> is a function taking two arguments (first pressure,
then density) then that function is called directly to calculate the
derivative, and no smoothing is done before or after that call.
</p>
</li></ul>

<p>For precise work, it makes sense to skip <code>swN2</code> entirely, choosing
whether, what, and how to smooth based on an understanding of fundamental
principles as well as data practicalities.
</p>


<h3>Value</h3>

<p>Square of buoyancy frequency (<code class="reqn">radian^2/s^2</code>).
</p>


<h3>Deprecation Notice</h3>

<p>Until 2019 April 11, <code>swN2</code> had an argument named <code>eos</code>. However,
this did not work as stated, unless the first argument was a <code>ctd</code>
object. Besides, the argument name was inherently deceptive, because the UNESCO
scheme does not specify how N2 is to be calculated.
Nothing is really lost by making this change, because the new default is the
same as was previously available with the <code>eos="unesco"</code>
setup, and the gsw-formulated estimate of N2 is provided
by <code><a href="gsw.html#topic+gsw_Nsquared">gsw::gsw_Nsquared()</a></code> in the <a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a> package.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>The <code><a href="gsw.html#topic+gsw_Nsquared">gsw::gsw_Nsquared()</a></code> function of the <a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a>
provides an alternative to this, as formulated in the GSW system. It
has a more sophisticated treatment of potential density, but it is based
on simple first-difference derivatives, so its results may require
smoothing, depending on the dataset and purpose of the analysis.
</p>
<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(oce)
data(ctd)
# Left panel: density
p &lt;- ctd[["pressure"]]
ylim &lt;- rev(range(p))
par(mfrow = c(1, 2), mar = c(3, 3, 1, 1), mgp = c(2, 0.7, 0))
plot(ctd[["sigmaTheta"]], p, ylim = ylim, type = "l", xlab = expression(sigma[theta]))
# Right panel: N2, with default settings (black) and with df=2 (red)
N2 &lt;- swN2(ctd)
plot(N2, p, ylim = ylim, xlab = "N2 [1/s^2]", ylab = "p", type = "l")
lines(swN2(ctd, df = 3), p, col = 2)

</code></pre>

<hr>
<h2 id='swPressure'>Water Pressure</h2><span id='topic+swPressure'></span>

<h3>Description</h3>

<p>Compute seawater pressure from depth by inverting <code><a href="#topic+swDepth">swDepth()</a></code>
using <code><a href="stats.html#topic+uniroot">uniroot()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swPressure(depth, latitude = 45, eos = getOption("oceEOS", default = "gsw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swPressure_+3A_depth">depth</code></td>
<td>
<p>distance below the surface in metres.</p>
</td></tr>
<tr><td><code id="swPressure_+3A_latitude">latitude</code></td>
<td>
<p>Latitude in <code class="reqn">^\circ</code>N.</p>
</td></tr>
<tr><td><code id="swPressure_+3A_eos">eos</code></td>
<td>
<p>indication of formulation to be used, either <code>"unesco"</code> or
<code>"gsw"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>eos="unesco"</code> this is done by numerical inversion of
<code><a href="#topic+swDepth">swDepth()</a></code> is done using <code><a href="stats.html#topic+uniroot">uniroot()</a></code>. If
<code>eos="gsw"</code>, it is done using <code><a href="gsw.html#topic+gsw_p_from_z">gsw::gsw_p_from_z()</a></code> in the
<a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a> package.
</p>


<h3>Value</h3>

<p>Pressure in dbar.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Unesco 1983. Algorithms for computation of fundamental
properties of seawater, 1983. <em>Unesco Tech. Pap. in Mar. Sci.</em>, No. 44,
53 pp.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>swPressure(9712.653, 30, eos = "unesco") # 10000
swPressure(9712.653, 30, eos = "gsw") #  9998.863

</code></pre>

<hr>
<h2 id='swRho'>Seawater Density</h2><span id='topic+swRho'></span>

<h3>Description</h3>

<p>Compute <code class="reqn">\rho</code>, the <em>in-situ</em> density of seawater.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swRho(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swRho_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object, in
which case <code>salinity</code>, <code>temperature</code> (in the ITS-90 scale; see
next item), etc. are inferred from the object, ignoring the
other parameters, if they are supplied.</p>
</td></tr>
<tr><td><code id="swRho_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.  This scale is used by GSW-style calculation (as
requested by setting <code>eos="gsw"</code>), and is the value contained within
<code>ctd</code> objects (and probably most other objects created with data
acquired in the past decade or two). Since the UNESCO-style calculation is
based on IPTS-68, the temperature is converted within the present function,
using <code><a href="#topic+T68fromT90">T68fromT90()</a></code>.</p>
</td></tr>
<tr><td><code id="swRho_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swRho_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swRho_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swRho_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2)
or <code>"gsw"</code> (references 3 and 4).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>eos="unesco"</code>, the density is calculated using the UNESCO equation
of state for seawater (references 1 and 2), and if <code>eos="gsw"</code>, the GSW formulation
(references 3 and 4) is used.
</p>


<h3>Value</h3>

<p><em>In-situ</em> density (kg/m<code class="reqn">^3</code>).
</p>


<h3>Temperature units</h3>

<p>The UNESCO formulae are defined in terms of
temperature measured on the IPTS-68 scale, whereas the replacement GSW
formulae are based on the ITS-90 scale. Prior to the addition of GSW
capabilities, the various <code style="white-space: pre;">&#8288;sw*&#8288;</code> functions took temperature to be in
IPTS-68 units. As GSW capabilities were added in early 2015, the assumed
unit of <code>temperature</code> was taken to be ITS-90.  This change means that
old code has to be modified, by replacing e.g. <code>swRho(S, T, p)</code> with
<code>swRho(S, T90fromT68(T), p)</code>. At typical oceanic values, the difference
between the two scales is a few millidegrees.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Fofonoff, P. and R. C. Millard Jr, 1983. Algorithms for computation of
fundamental properties of seawater.
<em>Unesco Technical Papers in Marine Science</em>,
<em>44</em>, 53 pp.
</p>
</li>
<li><p> Gill, A.E., 1982. <em>Atmosphere-ocean Dynamics</em>, Academic Press, New
York, 662 pp.
</p>
</li>
<li><p> IOC, SCOR, and IAPSO (2010). The international thermodynamic equation of
seawater-2010: Calculation and use of thermodynamic properties.  Technical
Report 56, Intergovernmental Oceanographic Commission, Manuals and Guide.
</p>
</li>
<li><p> McDougall, T.J. and P.M. Barker, 2011: Getting started with TEOS-10 and
the Gibbs Seawater (GSW) Oceanographic Toolbox, 28pp., SCOR/IAPSO WG127,
ISBN 978-0-646-55621-5.
</p>
</li></ol>



<h3>See Also</h3>

<p>Related density routines include <code><a href="#topic+swSigma0">swSigma0()</a></code> (and
equivalents at other pressure horizons), <code><a href="#topic+swSigmaT">swSigmaT()</a></code>, and
<code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code>.
</p>
<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# The numbers in the comments are the check values listed in reference 1;
# note that temperature in that reference was on the T68 scale, but that
# the present function works with the ITS-90 scale, so a conversion
# is required.
swRho(35, T90fromT68(5), 0, eos = "unesco") # 1027.67547
swRho(35, T90fromT68(5), 10000, eos = "unesco") # 1069.48914
swRho(35, T90fromT68(25), 0, eos = "unesco") # 1023.34306
swRho(35, T90fromT68(25), 10000, eos = "unesco") # 1062.53817

</code></pre>

<hr>
<h2 id='swRrho'>Density Ratio</h2><span id='topic+swRrho'></span>

<h3>Description</h3>

<p>Compute density ratio
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swRrho(
  ctd,
  sense = c("diffusive", "finger"),
  smoothingLength = 10,
  df,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swRrho_+3A_ctd">ctd</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object that holds <code>salinity</code>, <code>temperature</code>, and
<code>pressure</code>.  If <code>eos</code> is <code>"gsw"</code>, then it must also hold <code>longitude</code> and
<code>latitude</code>.</p>
</td></tr>
<tr><td><code id="swRrho_+3A_sense">sense</code></td>
<td>
<p>an indication of the sense of double diffusion under study and
therefore of the definition of Rrho; see &ldquo;Details&rdquo;</p>
</td></tr>
<tr><td><code id="swRrho_+3A_smoothinglength">smoothingLength</code></td>
<td>
<p>ignored if <code>df</code> supplied, but otherwise the
latter is calculated as the number of data points, divided by the number
within a depth interval of <code>smoothingLength</code> metres.</p>
</td></tr>
<tr><td><code id="swRrho_+3A_df">df</code></td>
<td>
<p>if given, this is provided to <code><a href="stats.html#topic+smooth.spline">smooth.spline()</a></code>.</p>
</td></tr>
<tr><td><code id="swRrho_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> or <code>"gsw"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This computes Rrho (density ratio) from a <code>ctd</code> object.
</p>
<p>If <code>eos="unesco"</code>, this is done by calculating salinity and
potential-temperature derivatives from smoothing splines whose properties
are governed by <code>smoothingLength</code> or <code>df</code>.  If
<code>sense="diffusive"</code> the definition is
<code class="reqn">(beta*dS/dz)/(alpha*d(theta)/dz)</code> and
the reciprocal for <code>"finger"</code>.
</p>
<p>If <code>eos="gsw"</code>, this is done by extracting absolute salinity and
conservative temperature, smoothing with a smoothing spline as in the
<code>"unesco"</code> case, and then calling <code><a href="gsw.html#topic+gsw_Turner_Rsubrho">gsw::gsw_Turner_Rsubrho()</a></code>
on these smoothed fields. Since the gsw function works on mid-point
pressures, <code><a href="stats.html#topic+approx">approx()</a></code> is used to interpolate back to the original
pressures.
</p>
<p>If the default arguments are acceptable, <code>ctd[["Rrho"]]</code> may be used
instead of <code>swRrho(ctd)</code>.
</p>


<h3>Value</h3>

<p>Density ratio defined in either the <code>"diffusive"</code> or
<code>"finger"</code> sense.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Chantelle Layton
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(ctd)
u &lt;- swRrho(ctd, eos = "unesco")
g &lt;- swRrho(ctd, eos = "gsw")
p &lt;- ctd[["p"]]
plot(u, p, ylim = rev(range(p)), type = "l", xlab = expression(R[rho]))
lines(g, p, lty = 2, col = "red")
legend("topright", lty = 1:2, legend = c("unesco", "gsw"), col = c("black", "red"))

</code></pre>

<hr>
<h2 id='swSCTp'>Practical Salinity From Electrical Conductivity, Temperature and Pressure</h2><span id='topic+swSCTp'></span>

<h3>Description</h3>

<p>Calculate salinity from what is actually measured by a CTD, <em>i.e.</em>
conductivity, <em>in-situ</em> temperature and pressure.  Often this is done
by the CTD processing software, but sometimes it is helpful to do this
directly, <em>e.g.</em> when there is a concern about mismatches in sensor
response times.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSCTp(
  conductivity,
  temperature = NULL,
  pressure = NULL,
  conductivityUnit,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSCTp_+3A_conductivity">conductivity</code></td>
<td>
<p>a measure of conductivity (see also <code>conductivityUnit</code>)
or an <code>oce</code> object holding hydrographic information. In the second case,
all the other arguments to <code>swSCTp</code> are ignored.</p>
</td></tr>
<tr><td><code id="swSCTp_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale; see &ldquo;Temperature units&rdquo; in the documentation for
<code><a href="#topic+swRho">swRho()</a></code>.</p>
</td></tr>
<tr><td><code id="swSCTp_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar).</p>
</td></tr>
<tr><td><code id="swSCTp_+3A_conductivityunit">conductivityUnit</code></td>
<td>
<p>string indicating the unit used for conductivity.
This may be <code>"ratio"</code> or <code>""</code> (meaning conductivity ratio),
<code>"mS/cm"</code> or <code>"S/m"</code>.  Note that the ratio mode assumes that
measured conductivity has been divided by the standard conductivity
of 4.2914 S/m.  In dealing with unfamiliar data for which the measurement
unit has not been recorded, it can be sensible to try all three possibilities
for <code>conductivityUnit</code>, to see which yields the most sensible salinities.</p>
</td></tr>
<tr><td><code id="swSCTp_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> or <code>"gsw"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two variants are provided. First, if <code>eos</code> is
<code>"unesco"</code>, then salinity is calculated using
the UNESCO algorithm described by Fofonoff and Millard (1983) as in
reference 1. Second, if <code>eos</code> is <code>"gsw"</code>, then the
Gibbs-SeaWater formulation is used, via <code><a href="gsw.html#topic+gsw_SP_from_C">gsw::gsw_SP_from_C()</a></code>
in the <a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a> package. The latter starts with the same formula
as the former, but if this yields a Practical Salinity less than 2,
then the result is instead calculated using
formulae provided by Hill et al. (1986; reference 2), modified to match the
<code>"unesco"</code> value at Practical salinity equal to 2 (reference 3).
</p>


<h3>Value</h3>

<p>Practical Salinity.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Fofonoff, P. and R. C. Millard Jr, 1983. Algorithms for
computation of fundamental properties of seawater. <em>Unesco Technical
Papers in Marine Science</em>, <em>44</em>, 53 pp.
</p>
</li>
<li><p> K. Hill, T. Dauphinee, and D. Woods. &ldquo;The Extension of the Practical
Salinity Scale 1978 to Low Salinities.&rdquo; IEEE Journal of Oceanic Engineering 11,
no. 1 (January 1986): 109-12.
<a href="https://doi.org/10.1109/JOE.1986.1145154">doi:10.1109/JOE.1986.1145154</a>
</p>
</li>
<li> <p><code>gsw_from_SP</code> online documentation, available at
<code style="white-space: pre;">&#8288;http://www.teos-10.org/pubs/gsw/html/gsw_C_from_SP.html&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>For thermal (as opposed to electrical) conductivity, see
<code><a href="#topic+swThermalConductivity">swThermalConductivity()</a></code>.  For computation of electrical
conductivity from salinity, see <code><a href="#topic+swCSTp">swCSTp()</a></code>.
</p>
<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Demonstrate agreement with test value in UNESCO documents
swSCTp(1, T90fromT68(15), 0, eos = "unesco") # expect 35
# 2. Demonstrate agreement of gsw and unesco, S&gt;2 case
swSCTp(1, T90fromT68(15), 0, eos = "gsw") # again, expect 35
# 3. Demonstrate close values even in very brackish water
swSCTp(0.02, 10, 100, eos = "gsw") # 0.6013981
swSCTp(0.02, 10, 100, eos = "unesco") # 0.6011721

</code></pre>

<hr>
<h2 id='swSigma'>Seawater Density Anomaly</h2><span id='topic+swSigma'></span>

<h3>Description</h3>

<p>Compute <code class="reqn">\sigma_\theta</code>, the density of seawater, minus 1000
kg/m<code class="reqn">^3</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSigma(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSigma_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object, in
which case <code>salinity</code>, <code>temperature</code> (in the ITS-90 scale; see
next item), etc. are inferred from the object, ignoring the
other parameters, if they are supplied.</p>
</td></tr>
<tr><td><code id="swSigma_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.  This scale is used by GSW-style calculation (as
requested by setting <code>eos="gsw"</code>), and is the value contained within
<code>ctd</code> objects (and probably most other objects created with data
acquired in the past decade or two). Since the UNESCO-style calculation is
based on IPTS-68, the temperature is converted within the present function,
using <code><a href="#topic+T68fromT90">T68fromT90()</a></code>.</p>
</td></tr>
<tr><td><code id="swSigma_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSigma_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2)
or <code>"gsw"</code> (references 3 and 4).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Density anomaly (kg/m<code class="reqn">^3</code>), as computed with <code><a href="#topic+swRho">swRho()</a></code>, minus-
1000 kg/m<code class="reqn">^3</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>See citations provided in the <code><a href="#topic+swRho">swRho()</a></code> documentation.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
swSigma(35, 13, 1000, longitude = 300, latitude = 30, eos = "gsw") # 30.82374
swSigma(35, T90fromT68(13), 1000, eos = "unesco") # 30.8183

</code></pre>

<hr>
<h2 id='swSigma0'>Seawater Potential Density Anomaly Referenced to Surface Pressure</h2><span id='topic+swSigma0'></span>

<h3>Description</h3>

<p>Compute the potential density of seawater (minus
1000 kg/m<code class="reqn">^3</code>), referenced to surface pressure. This is done using
<code><a href="gsw.html#topic+gsw_sigma0">gsw::gsw_sigma0()</a></code> if <code>eos="gsw"</code>, or using <code><a href="#topic+swSigmaTheta">swSigmaTheta()</a></code> if it is
<code>"unesco"</code>. (The difference between the formulations is typically
under 0.01 kg/m^3, corresponding to a few millidegrees of temperature.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSigma0(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSigma0_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object, in
which case <code>salinity</code>, <code>temperature</code> (in the ITS-90 scale; see
next item), etc. are inferred from the object, ignoring the
other parameters, if they are supplied.</p>
</td></tr>
<tr><td><code id="swSigma0_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.  This scale is used by GSW-style calculation (as
requested by setting <code>eos="gsw"</code>), and is the value contained within
<code>ctd</code> objects (and probably most other objects created with data
acquired in the past decade or two). Since the UNESCO-style calculation is
based on IPTS-68, the temperature is converted within the present function,
using <code><a href="#topic+T68fromT90">T68fromT90()</a></code>.</p>
</td></tr>
<tr><td><code id="swSigma0_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSigma0_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma0_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma0_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2)
or <code>"gsw"</code> (references 3 and 4).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Potential density anomaly (kg/m<code class="reqn">^3</code>).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>See citations provided in the <code><a href="#topic+swRho">swRho()</a></code> documentation.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>

<hr>
<h2 id='swSigma1'>Seawater Potential Density Anomaly Referenced to 1000db Pressure</h2><span id='topic+swSigma1'></span>

<h3>Description</h3>

<p>This is analogous to <code><a href="#topic+swSigma0">swSigma0()</a></code>, but referenced to 1000db pressure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSigma1(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSigma1_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object, in
which case <code>salinity</code>, <code>temperature</code> (in the ITS-90 scale; see
next item), etc. are inferred from the object, ignoring the
other parameters, if they are supplied.</p>
</td></tr>
<tr><td><code id="swSigma1_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.  This scale is used by GSW-style calculation (as
requested by setting <code>eos="gsw"</code>), and is the value contained within
<code>ctd</code> objects (and probably most other objects created with data
acquired in the past decade or two). Since the UNESCO-style calculation is
based on IPTS-68, the temperature is converted within the present function,
using <code><a href="#topic+T68fromT90">T68fromT90()</a></code>.</p>
</td></tr>
<tr><td><code id="swSigma1_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSigma1_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma1_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma1_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2)
or <code>"gsw"</code> (references 3 and 4).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Potential density anomaly (kg/m<code class="reqn">^3</code>).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>See citations provided in the <code><a href="#topic+swRho">swRho()</a></code> documentation.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>

<hr>
<h2 id='swSigma2'>Seawater Potential Density Anomaly Referenced to 2000db Pressure</h2><span id='topic+swSigma2'></span>

<h3>Description</h3>

<p>This is analogous to <code><a href="#topic+swSigma0">swSigma0()</a></code>, but referenced to 2000db pressure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSigma2(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSigma2_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object, in
which case <code>salinity</code>, <code>temperature</code> (in the ITS-90 scale; see
next item), etc. are inferred from the object, ignoring the
other parameters, if they are supplied.</p>
</td></tr>
<tr><td><code id="swSigma2_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.  This scale is used by GSW-style calculation (as
requested by setting <code>eos="gsw"</code>), and is the value contained within
<code>ctd</code> objects (and probably most other objects created with data
acquired in the past decade or two). Since the UNESCO-style calculation is
based on IPTS-68, the temperature is converted within the present function,
using <code><a href="#topic+T68fromT90">T68fromT90()</a></code>.</p>
</td></tr>
<tr><td><code id="swSigma2_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSigma2_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma2_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma2_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2)
or <code>"gsw"</code> (references 3 and 4).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Potential density anomaly (kg/m<code class="reqn">^3</code>).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>See citations provided in the <code><a href="#topic+swRho">swRho()</a></code> documentation.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>

<hr>
<h2 id='swSigma3'>Seawater Potential Density Anomaly Referenced to 3000db Pressure</h2><span id='topic+swSigma3'></span>

<h3>Description</h3>

<p>This is analogous to <code><a href="#topic+swSigma0">swSigma0()</a></code>, but referenced to 3000db pressure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSigma3(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSigma3_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object, in
which case <code>salinity</code>, <code>temperature</code> (in the ITS-90 scale; see
next item), etc. are inferred from the object, ignoring the
other parameters, if they are supplied.</p>
</td></tr>
<tr><td><code id="swSigma3_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.  This scale is used by GSW-style calculation (as
requested by setting <code>eos="gsw"</code>), and is the value contained within
<code>ctd</code> objects (and probably most other objects created with data
acquired in the past decade or two). Since the UNESCO-style calculation is
based on IPTS-68, the temperature is converted within the present function,
using <code><a href="#topic+T68fromT90">T68fromT90()</a></code>.</p>
</td></tr>
<tr><td><code id="swSigma3_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSigma3_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma3_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma3_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2)
or <code>"gsw"</code> (references 3 and 4).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Potential density anomaly (kg/m<code class="reqn">^3</code>).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>See citations provided in the <code><a href="#topic+swRho">swRho()</a></code> documentation.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>

<hr>
<h2 id='swSigma4'>Seawater Potential Density Anomaly Referenced to 4000db Pressure</h2><span id='topic+swSigma4'></span>

<h3>Description</h3>

<p>This is analogous to <code><a href="#topic+swSigma0">swSigma0()</a></code>, but referenced to 4000db pressure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSigma4(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSigma4_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object, in
which case <code>salinity</code>, <code>temperature</code> (in the ITS-90 scale; see
next item), etc. are inferred from the object, ignoring the
other parameters, if they are supplied.</p>
</td></tr>
<tr><td><code id="swSigma4_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.  This scale is used by GSW-style calculation (as
requested by setting <code>eos="gsw"</code>), and is the value contained within
<code>ctd</code> objects (and probably most other objects created with data
acquired in the past decade or two). Since the UNESCO-style calculation is
based on IPTS-68, the temperature is converted within the present function,
using <code><a href="#topic+T68fromT90">T68fromT90()</a></code>.</p>
</td></tr>
<tr><td><code id="swSigma4_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSigma4_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma4_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigma4_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2)
or <code>"gsw"</code> (references 3 and 4).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Potential density anomaly (kg/m<code class="reqn">^3</code>).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>See citations provided in the <code><a href="#topic+swRho">swRho()</a></code> documentation.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>

<hr>
<h2 id='swSigmaT'>Seawater Quasi-Potential Density Anomaly</h2><span id='topic+swSigmaT'></span>

<h3>Description</h3>

<p>Compute <code class="reqn">\sigma_t</code>, a rough estimate of potential density of
seawater, minus 1000 kg/m<code class="reqn">^3</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSigmaT(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSigmaT_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object, in
which case <code>salinity</code>, <code>temperature</code> (in the ITS-90 scale; see
next item), etc. are inferred from the object, ignoring the
other parameters, if they are supplied.</p>
</td></tr>
<tr><td><code id="swSigmaT_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.  This scale is used by GSW-style calculation (as
requested by setting <code>eos="gsw"</code>), and is the value contained within
<code>ctd</code> objects (and probably most other objects created with data
acquired in the past decade or two). Since the UNESCO-style calculation is
based on IPTS-68, the temperature is converted within the present function,
using <code><a href="#topic+T68fromT90">T68fromT90()</a></code>.</p>
</td></tr>
<tr><td><code id="swSigmaT_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSigmaT_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigmaT_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigmaT_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2)
or <code>"gsw"</code> (references 3 and 4).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the first argument is an <a href="#topic+oce-class">oce</a> object, then salinity, etc., are
extracted from it, and used for the calculation.
</p>


<h3>Value</h3>

<p>Quasi-potential density anomaly (kg/m<code class="reqn">^3</code>), defined as the
density calculated with pressure set to zero.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>See citations provided in the <code><a href="#topic+swRho">swRho()</a></code> documentation.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>swSigmaT(35, 13, 1000, longitude = 300, latitude = 30, eos = "gsw") # 26.39623
swSigmaT(35, T90fromT68(13), 1000, eos = "unesco") # 26.39354

</code></pre>

<hr>
<h2 id='swSigmaTheta'>Seawater Potential Density Anomaly</h2><span id='topic+swSigmaTheta'></span>

<h3>Description</h3>

<p>Compute the potential density (minus 1000 kg/m^3) that seawater would have if raised
adiabatically to the surface.  In the UNESCO system, this quantity is
is denoted <code class="reqn">\sigma_\theta</code> (hence the function name), but in
the GSW system, a somewhat related quantity is denoted <code>sigma0</code>. (In a
deep-water CTD cast, the RMS deviation between sigma-theta and sigma0 is
typically of order 0.0003 kg/m^3, corresponding to a temperature shift of
about 0.002C, so the distinction between the quantities is not large.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSigmaTheta(
  salinity,
  temperature = NULL,
  pressure = NULL,
  referencePressure = 0,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSigmaTheta_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object, in
which case <code>salinity</code>, <code>temperature</code> (in the ITS-90 scale; see
next item), etc. are inferred from the object, ignoring the
other parameters, if they are supplied.</p>
</td></tr>
<tr><td><code id="swSigmaTheta_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.  This scale is used by GSW-style calculation (as
requested by setting <code>eos="gsw"</code>), and is the value contained within
<code>ctd</code> objects (and probably most other objects created with data
acquired in the past decade or two). Since the UNESCO-style calculation is
based on IPTS-68, the temperature is converted within the present function,
using <code><a href="#topic+T68fromT90">T68fromT90()</a></code>.</p>
</td></tr>
<tr><td><code id="swSigmaTheta_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSigmaTheta_+3A_referencepressure">referencePressure</code></td>
<td>
<p>The reference pressure, in dbar.</p>
</td></tr>
<tr><td><code id="swSigmaTheta_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigmaTheta_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSigmaTheta_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2)
or <code>"gsw"</code> (references 3 and 4).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the first argument is an <code>oce</code> object, then salinity, etc., are
extracted from it, and used for the calculation instead of any values
provided in the other arguments.
</p>


<h3>Value</h3>

<p>Potential density anomaly (kg/m<code class="reqn">^3</code>), defined as
<code class="reqn">\sigma_\theta=\rho(S,\theta(S,t,p),0</code>
</p>

<ul>
<li><p> 1000 kg/m<code class="reqn">^3</code>.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>See citations provided in the <code><a href="#topic+swRho">swRho()</a></code> documentation.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stopifnot(abs(26.4212790994 - swSigmaTheta(35, 13, 1000, eos = "unesco")) &lt; 1e-7)

</code></pre>

<hr>
<h2 id='swSoundAbsorption'>Seawater Sound Absorption</h2><span id='topic+swSoundAbsorption'></span>

<h3>Description</h3>

<p>Compute the sound absorption of seawater, in dB/m
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSoundAbsorption(
  frequency,
  salinity,
  temperature,
  pressure,
  pH = 8,
  formulation = c("fisher-simmons", "francois-garrison")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSoundAbsorption_+3A_frequency">frequency</code></td>
<td>
<p>The frequency of sound, in Hz.</p>
</td></tr>
<tr><td><code id="swSoundAbsorption_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object, in
which case <code>salinity</code>, <code>temperature</code> (in the ITS-90 scale; see
next item), etc. are inferred from the object, ignoring the
other parameters, if they are supplied.</p>
</td></tr>
<tr><td><code id="swSoundAbsorption_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.  This scale is used by GSW-style calculation (as
requested by setting <code>eos="gsw"</code>), and is the value contained within
<code>ctd</code> objects (and probably most other objects created with data
acquired in the past decade or two). Since the UNESCO-style calculation is
based on IPTS-68, the temperature is converted within the present function,
using <code><a href="#topic+T68fromT90">T68fromT90()</a></code>.</p>
</td></tr>
<tr><td><code id="swSoundAbsorption_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSoundAbsorption_+3A_ph">pH</code></td>
<td>
<p>seawater pH</p>
</td></tr>
<tr><td><code id="swSoundAbsorption_+3A_formulation">formulation</code></td>
<td>
<p>character string indicating the formulation to use, either
of <code>"fischer-simmons"</code> or <code>"francois-garrison"</code>; see &ldquo;References&rdquo;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Salinity and pH are ignored in this formulation.  Several formulae may be
found in the literature, and they give results differing by 10 percent, as
shown in reference 3 for example.  For this reason, it is likely that more
formulations will be added to this function, and entirely possible that the
default may change.
</p>


<h3>Value</h3>

<p>Sound absorption in dB/m.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> F. H. Fisher and V. P. Simmons, 1977.  Sound absorption in
sea water.  Journal of the Acoustical Society of America, 62(3), 558-564.
</p>
</li>
<li><p> R. E. Francois and G. R. Garrison, 1982.  Sound absorption based on
ocean measurements.  Part II: Boric acid contribution and equation for total
absorption.  Journal of the Acoustical Society of America, 72(6):1879-1890.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;http://resource.npl.co.uk/acoustics/techguides/seaabsorption/&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Fisher &amp; Simmons (1977 table IV) gives 0.52 dB/km for 35 PSU, 5 degC, 500 atm
# (4990 dbar of water)a and 10 kHz
alpha &lt;- swSoundAbsorption(35, 4, 4990, 10e3)

# reproduce part of Fig 8 of Francois and Garrison (1982 Fig 8)
f &lt;- 1e3 * 10^(seq(-1, 3, 0.1)) # in KHz
plot(f / 1000, 1e3 * swSoundAbsorption(f, 35, 10, 0, formulation = "fr"),
    xlab = " Freq [kHz]", ylab = " dB/km", type = "l", log = "xy"
)
lines(f / 1000, 1e3 * swSoundAbsorption(f, 0, 10, 0, formulation = "fr"), lty = "dashed")
legend("topleft", lty = c("solid", "dashed"), legend = c("S=35", "S=0"))

</code></pre>

<hr>
<h2 id='swSoundSpeed'>Seawater Sound Speed</h2><span id='topic+swSoundSpeed'></span>

<h3>Description</h3>

<p>Compute the seawater speed of sound.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSoundSpeed(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSoundSpeed_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object, in
which case <code>salinity</code>, <code>temperature</code> (in the ITS-90 scale; see
next item), etc. are inferred from the object, ignoring the
other parameters, if they are supplied.</p>
</td></tr>
<tr><td><code id="swSoundSpeed_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.  This scale is used by GSW-style calculation (as
requested by setting <code>eos="gsw"</code>), and is the value contained within
<code>ctd</code> objects (and probably most other objects created with data
acquired in the past decade or two). Since the UNESCO-style calculation is
based on IPTS-68, the temperature is converted within the present function,
using <code><a href="#topic+T68fromT90">T68fromT90()</a></code>.</p>
</td></tr>
<tr><td><code id="swSoundSpeed_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSoundSpeed_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSoundSpeed_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSoundSpeed_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2)
or <code>"gsw"</code> (references 3 and 4).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>eos="unesco"</code>, the sound speed is calculated using the formulation
in section 9 of Fofonoff and Millard (1983). If <code>eos="gsw"</code>, then the
<code><a href="gsw.html#topic+gsw_sound_speed">gsw::gsw_sound_speed()</a></code> function from the
<a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a> package is used.
</p>


<h3>Value</h3>

<p>Sound speed (m/s).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Fofonoff, P. and R. C. Millard Jr, 1983. Algorithms for
computation of fundamental properties of seawater.
<em>Unesco Technical Papers in Marine Science</em>,
<em>44</em>, 53 pp.  (See section 9.)
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>swSoundSpeed(40, T90fromT68(40), 10000) # 1731.995 (p48 of Fofonoff + Millard 1983)

</code></pre>

<hr>
<h2 id='swSpecificHeat'>Seawater Specific Heat</h2><span id='topic+swSpecificHeat'></span>

<h3>Description</h3>

<p>Compute specific heat of seawater.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSpecificHeat(
  salinity,
  temperature = NULL,
  pressure = 0,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSpecificHeat_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case <code>temperature</code>
and <code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object (in
which case <code>salinity</code>, etc. are inferred from the object).</p>
</td></tr>
<tr><td><code id="swSpecificHeat_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale.</p>
</td></tr>
<tr><td><code id="swSpecificHeat_+3A_pressure">pressure</code></td>
<td>
<p>seawater pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSpecificHeat_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSpecificHeat_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSpecificHeat_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> or <code>"gsw"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the first argument is a <code>ctd</code> object, then salinity, etc, are
extracted from it, and used for the calculation.
</p>


<h3>Value</h3>

<p>Specific heat (J/kg/degC).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Millero et. al., J. Geophys. Res. 78 (1973), 4499-4507
</p>
<p>Millero et. al., UNESCO report 38 (1981), 99-188.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>swSpecificHeat(40, T90fromT68(40), 10000, eos = "unesco") # 3949.499

</code></pre>

<hr>
<h2 id='swSpice'>Seawater Spiciness</h2><span id='topic+swSpice'></span>

<h3>Description</h3>

<p>Compute seawater &quot;spice&quot;, also called &quot;spiciness&quot; (a variable orthogonal
to density in TS space), in either of two formulations, depending on
the value of the <code>eos</code> argument. If <code>eos="unesco"</code> then
Flament's (reference 1) formulation is used. If <code>eos="gsw"</code>
then the Gibbs SeaWater formulation for &quot;spiciness0&quot; is used
(see reference 2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSpice(
  salinity,
  temperature = NULL,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSpice_+3A_salinity">salinity</code></td>
<td>
<p>either salinity (PSU) (in which case <code>temperature</code> and
<code>pressure</code> must be provided) <em>or</em> a <code>ctd</code> object (in which
case <code>salinity</code>, <code>temperature</code> and <code>pressure</code> are determined
from the object, and must not be provided in the argument list).</p>
</td></tr>
<tr><td><code id="swSpice_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C) on the
ITS-90 scale; see &ldquo;Temperature units&rdquo; in the documentation for
<code><a href="#topic+swRho">swRho()</a></code>.</p>
</td></tr>
<tr><td><code id="swSpice_+3A_pressure">pressure</code></td>
<td>
<p>Seawater pressure (dbar) (only used if <code>eos</code> is
<code>"gsw"</code>); see &ldquo;Details&rdquo;..</p>
</td></tr>
<tr><td><code id="swSpice_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos</code> is
<code>"gsw"</code>; see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSpice_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos</code> is
<code>"gsw"</code>; see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swSpice_+3A_eos">eos</code></td>
<td>
<p>Character value specifying the equation of state,
either <code>"unesco"</code> (for the Flament formulation, although this
is not actually part of UNESCO)
or <code>"gsw"</code> for the Gibbs SeaWater formulation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the first argument is a <code>ctd</code> object, then salinity, temperature and
pressure values are extracted from it, and used for the calculation. (For
the <code>eos="gsw"</code> case, longitude and latitude are also extracted, because
these are required for the formulation of spiciness0.
</p>
<p>Roughly speaking, seawater with a high spiciness is relatively warm and
salty compared with less spicy water. Another interpretation is that spice
is a variable measuring distance orthogonal to isopycnal lines on TS
diagrams (if the diagrams are scaled to make the isopycnals run at 45
degrees). Note that pressure, longitude and latitude are all
ignored in the Flament definition.
</p>


<h3>Value</h3>

<p>Flament-formulated spice <code class="reqn">kg/m^3</code> if <code>eos</code> is <code>"unesco"</code>
or surface-referenced GSW spiciness0 <code class="reqn">kg/m^3</code> if <code>eos</code> is <code>"gsw"</code>,
the latter provided by <code><a href="gsw.html#topic+gsw_spiciness0">gsw::gsw_spiciness0()</a></code>, and hence aimed
at application within the top half-kilometre of the ocean.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley coded this, merely an interface to the code described
by references 1 and 2.
</p>


<h3>References</h3>


<ol>
<li><p> Flament, P. &ldquo;A State Variable for Characterizing Water Masses and Their
Diffusive Stability: Spiciness.&rdquo; Progress in Oceanography, Observations of the
1997-98 El Nino along the West Coast of North America, 54, no. 1
(July 1, 2002):493-501.
<a href="https://doi.org/10.1016/S0079-6611%2802%2900065-4">doi:10.1016/S0079-6611(02)00065-4</a>
</p>
</li></ol>

<p>2.McDougall, Trevor J., and Oliver A. Krzysik. &ldquo;Spiciness.&rdquo;
Journal of Marine Research 73, no. 5 (September 1, 2015): 141-52.
doi: <code>10.1357/002224015816665589</code>
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Contrast the two formulations.
library(oce)
data(ctd)
p &lt;- ctd[["pressure"]]
plot(swSpice(ctd, eos = "unesco"), p,
    xlim = c(-2.7, -1.5), ylim = rev(range(p)),
    xlab = "Spice", ylab = "Pressure (dbar)"
)
points(swSpice(ctd, eos = "gsw"), p, col = 2)
mtext("black=unesco, red=gsw")

</code></pre>

<hr>
<h2 id='swSR'>Seawater Reference Salinity (GSW Formulation)</h2><span id='topic+swSR'></span>

<h3>Description</h3>

<p>Compute seawater Reference Salinity (SR), according to the GSW/TEOS-10
formulation with <code><a href="gsw.html#topic+gsw_SR_from_SP">gsw::gsw_SR_from_SP()</a></code> in the <a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSR(salinity)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSR_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity or an <code>oce</code> object
that holds salinity in its data slot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Reference Salinity, SR, in <code class="reqn">g/kg</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>McDougall, T.J. and P.M. Barker, 2011: Getting started with
TEOS-10 and the Gibbs Seawater (GSW) Oceanographic Toolbox, 28pp.,
SCOR/IAPSO WG127, ISBN 978-0-646-55621-5.
</p>


<h3>See Also</h3>

<p>For some objects, SR may also be recovered by indexing as e.g.
<code>ctd[["SR"]]</code>.
</p>
<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SR &lt;- swSR(35.0) # 35.16504

</code></pre>

<hr>
<h2 id='swSstar'>Seawater Preformed Salinity (GSW Formulation)</h2><span id='topic+swSstar'></span>

<h3>Description</h3>

<p>Compute seawater Preformed Salinity (S*), according to the GSW/TEOS-10
formulation with <code><a href="gsw.html#topic+gsw_Sstar_from_SA">gsw::gsw_Sstar_from_SA()</a></code> in the <a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSstar(salinity, pressure = NULL, longitude = NULL, latitude = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSstar_+3A_salinity">salinity</code></td>
<td>
<p>either practical salinity (in which case
<code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object with
<code>salinity</code> and <code>pressure</code> in its data slot, and with
<code>longitude</code> and <code>latitude</code> either there, or in the metadata
slot.</p>
</td></tr>
<tr><td><code id="swSstar_+3A_pressure">pressure</code></td>
<td>
<p>pressure in dbar.</p>
</td></tr>
<tr><td><code id="swSstar_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation.</p>
</td></tr>
<tr><td><code id="swSstar_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Preformed Salinity, S*, in <code class="reqn">g/kg</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>McDougall, T.J. and P.M. Barker, 2011: Getting started with
TEOS-10 and the Gibbs Seawater (GSW) Oceanographic Toolbox, 28pp.,
SCOR/IAPSO WG127, ISBN 978-0-646-55621-5.
</p>


<h3>See Also</h3>

<p>For some objects, S-star may also be recovered by
indexing as e.g. <code>ctd[["Sstar"]]</code>.
</p>
<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>swSstar(35.5, 300, 260, 16) # 35.66601

</code></pre>

<hr>
<h2 id='swSTrho'>Seawater Salinity From Temperature and Density</h2><span id='topic+swSTrho'></span>

<h3>Description</h3>

<p>Compute Practical or Absolute Salinity, given in-situ or Conservative
Temperature, density, and pressure. This is mainly used to draw isopycnal
lines on TS diagrams, hence the dual meanings for salinity and temperature,
depending on the value of <code>eos</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swSTrho(
  temperature,
  density,
  pressure,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swSTrho_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale; see &ldquo;Temperature units&rdquo; in the documentation for
<code><a href="#topic+swRho">swRho()</a></code>.</p>
</td></tr>
<tr><td><code id="swSTrho_+3A_density">density</code></td>
<td>
<p><em>in-situ</em> density or sigma value (<code class="reqn">kg/m^3</code>)</p>
</td></tr>
<tr><td><code id="swSTrho_+3A_pressure">pressure</code></td>
<td>
<p><em>in-situ</em> pressure (dbar)</p>
</td></tr>
<tr><td><code id="swSTrho_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (see references 1 and 2)
or <code>"gsw"</code> (see references 3 and 4).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code>eos="unesco"</code>, finds the practical salinity that yields the given
density, with the given in-situ temperature and pressure.  The method is a
bisection search with a salinity tolerance of 0.001.  For <code>eos="gsw"</code>,
the function <code><a href="gsw.html#topic+gsw_SA_from_rho">gsw::gsw_SA_from_rho()</a></code> in the <a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a>
package is used
to infer Absolute Salinity from Conservative Temperature.
</p>


<h3>Value</h3>

<p>Practical Salinity, if <code>eos="unesco"</code>, or Absolute Salinity, if
<code>eos="gsw"</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Fofonoff, P. and R. C. Millard Jr, 1983. Algorithms for computation of
fundamental properties of seawater. <em>Unesco Technical Papers in Marine
Science</em>, <em>44</em>, 53 pp
</p>
</li>
<li><p> Gill, A.E., 1982. <em>Atmosphere-ocean Dynamics</em>, Academic Press, New
York, 662 pp.
</p>
</li>
<li><p> IOC, SCOR, and IAPSO (2010). The international thermodynamic equation of
seawater-2010: Calculation and use of thermodynamic properties.  Technical
Report 56, Intergovernmental Oceanographic Commission, Manuals and Guide.
</p>
</li>
<li><p> McDougall, T.J. and P.M. Barker, 2011: Getting started with TEOS-10 and
the Gibbs Seawater (GSW) Oceanographic Toolbox, 28pp., SCOR/IAPSO WG127,
ISBN 978-0-646-55621-5.
</p>
</li></ol>



<h3>See Also</h3>

<p><code><a href="#topic+swTSrho">swTSrho()</a></code>
</p>
<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>swSTrho(10, 22, 0, eos = "gsw") # 28.76285
swSTrho(10, 22, 0, eos = "unesco") # 28.651625

</code></pre>

<hr>
<h2 id='swTFreeze'>Seawater Freezing Temperature</h2><span id='topic+swTFreeze'></span>

<h3>Description</h3>

<p>Compute in-situ freezing temperature of seawater, using either the UNESCO
formulation (computed as in Section 5 of Fofonoff and Millard, 1983) or the
GSW formulation (computed by using <code><a href="gsw.html#topic+gsw_SA_from_SP">gsw::gsw_SA_from_SP()</a></code> to get Absolute
Salinity, and then <code><a href="gsw.html#topic+gsw_t_freezing">gsw::gsw_t_freezing()</a></code> to get the freezing temperature).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swTFreeze(
  salinity,
  pressure = NULL,
  longitude = NULL,
  latitude = NULL,
  saturation_fraction = 1,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swTFreeze_+3A_salinity">salinity</code></td>
<td>
<p>Either practical salinity (PSU) or a <code>ctd</code> object from which
practical salinity and pressure (plus in the <code>eos="gsw"</code> case,
longitude and latitude) are inferred.</p>
</td></tr>
<tr><td><code id="swTFreeze_+3A_pressure">pressure</code></td>
<td>
<p>Seawater pressure (dbar).</p>
</td></tr>
<tr><td><code id="swTFreeze_+3A_longitude">longitude</code></td>
<td>
<p>Longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swTFreeze_+3A_latitude">latitude</code></td>
<td>
<p>Latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swTFreeze_+3A_saturation_fraction">saturation_fraction</code></td>
<td>
<p>The saturation fraction of dissolved air in seawater,
ignored if <code>eos="unesco"</code>).</p>
</td></tr>
<tr><td><code id="swTFreeze_+3A_eos">eos</code></td>
<td>
<p>The equation of state, either <code>"unesco"</code> (Fofonoff and Millard,
1983; Gill 1982) or <code>"gsw"</code> (IOC, SCOR and IAPSO 2010; McDougall and Barker
2011).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the first argument is an <code>oce</code> object, and if the <code>pressure</code> argument is
<code>NULL</code>, then the pressure is sought within the first argument. In the case of
<code>eos="gsw"</code>, then a similar procedure also applies to the <code>longitude</code> and
<code>latitude</code> arguments.
</p>


<h3>Value</h3>

<p>Temperature (degC), defined on the ITS-90 scale.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Fofonoff, N. P., and R. C. Millard. Algorithms for Computation of
Fundamental Properties of Seawater. UNESCO Technical Papers in Marine
Research. SCOR working group on Evaluation of CTD data; UNESCO/ICES/SCOR/IAPSO
Joint Panel on Oceanographic Tables and Standards, 1983.
</p>
<p>Gill, A E. Atmosphere-Ocean Dynamics. New York, NY, USA: Academic Press,
1982.
</p>
<p>IOC, SCOR, and IAPSO (2010). The international thermodynamic equation of
seawater-2010: Calculation and use of thermodynamic properties.  Technical
Report 56, Intergovernmental Oceanographic Commission, Manuals and Guide, 2010.
</p>
<p>McDougall, Trevor J., and Paul M. Barker. Getting Started with TEOS-10 and
the Gibbs Seawater (GSW) Oceanographic Toolbox. SCOR/IAPSO WG127, 2011.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 1. Test for a check-value given in reference 1. This value, -2.588567 degC,
# is in the 1968 temperature scale (IPTS-68), but swTFreeze reports
# in the newer ITS-90 scale, so we must convert before checking.
Tcheck &lt;- -2.588567 # IPTS-68
T &lt;- swTFreeze(salinity = 40, pressure = 500, eos = "unesco")
stopifnot(abs(Tcheck - T68fromT90(T)) &lt; 1e-6)

# 2. Compare unesco and gsw formulations.
data(ctd)
p &lt;- ctd[["pressure"]]
par(mfrow = c(1, 2), mar = c(3, 3, 1, 2), mgp = c(2, 0.7, 0))
plot(swTFreeze(ctd, eos = "unesco"),
    p,
    xlab = "unesco", ylim = rev(range(p))
)
plot(swTFreeze(ctd, eos = "unesco") - swTFreeze(ctd, eos = "gsw"),
    p,
    xlab = "unesco-gsw", ylim = rev(range(p))
)

</code></pre>

<hr>
<h2 id='swThermalConductivity'>Seawater Thermal Conductivity</h2><span id='topic+swThermalConductivity'></span>

<h3>Description</h3>

<p>Compute seawater thermal conductivity, in <code class="reqn">W
m^{-1\circ}C^{-1}</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swThermalConductivity(salinity, temperature = NULL, pressure = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swThermalConductivity_+3A_salinity">salinity</code></td>
<td>
<p>salinity (PSU), or a <code>ctd</code> object, in which case
<code>temperature</code> and <code>pressure</code> will be ignored.</p>
</td></tr>
<tr><td><code id="swThermalConductivity_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale; see &ldquo;Temperature units&rdquo; in the documentation for
<code><a href="#topic+swRho">swRho()</a></code>.</p>
</td></tr>
<tr><td><code id="swThermalConductivity_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Caldwell's (1974) detailed formulation is used.  To be specific, his
equation 6 to calculate K, and his two sentences above that equation are
used to infer this to be K(0,T,S) in his notation of equation 7. Then,
application of his equations 7 and 8 is straightforward. He states an
accuracy for this method of 0.3 percent.  (See the check against his Table 1
in the &ldquo;Examples&rdquo;.)
</p>


<h3>Value</h3>

<p>Conductivity of seawater in <code class="reqn">W m^{-1} {^\circ} C^{-1}</code>.
To calculate thermal diffusivity in <code class="reqn">m^2/s^2</code>, divide by the
product of density and specific heat, as in the example.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Caldwell, Douglas R., 1974. Thermal conductivity of seawater,
<em>Deep-sea Research</em>, <em>21</em>, 131-137.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Values in m^2/s, a unit that is often used instead of W/(m*degC).
swThermalConductivity(35, 10, 100) / (swRho(35, 10, 100) * swSpecificHeat(35, 10, 100)) # ocean
swThermalConductivity(0, 20, 0) / (swRho(0, 20, 0) * swSpecificHeat(0, 20, 0)) # lab
# Caldwell Table 1 gives 1478e-6 cal/(cm*sec*degC) at 31.5 o/oo, 10degC, 1kbar
joulePerCalorie &lt;- 4.18400
cmPerM &lt;- 100
swThermalConductivity(31.5, 10, 1000) / joulePerCalorie / cmPerM

</code></pre>

<hr>
<h2 id='swTheta'>Seawater Potential Temperature (UNESCO Version)</h2><span id='topic+swTheta'></span>

<h3>Description</h3>

<p>Compute the potential temperature of seawater, denoted <code class="reqn">\theta</code>
in the UNESCO system, and <code>pt</code> in the GSW system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swTheta(
  salinity,
  temperature = NULL,
  pressure = NULL,
  referencePressure = 0,
  longitude = NULL,
  latitude = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swTheta_+3A_salinity">salinity</code></td>
<td>
<p>either salinity (PSU) (in which case <code>temperature</code> and
<code>pressure</code> must be provided) <em>or</em> an <code>oce</code> object (in which
case <code>salinity</code>, etc. are inferred from the object).</p>
</td></tr>
<tr><td><code id="swTheta_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale; see &ldquo;Temperature units&rdquo; in the documentation for
<code><a href="#topic+swRho">swRho()</a></code>, and the examples below.</p>
</td></tr>
<tr><td><code id="swTheta_+3A_pressure">pressure</code></td>
<td>
<p>pressure (dbar)</p>
</td></tr>
<tr><td><code id="swTheta_+3A_referencepressure">referencePressure</code></td>
<td>
<p>reference pressure (dbar)</p>
</td></tr>
<tr><td><code id="swTheta_+3A_longitude">longitude</code></td>
<td>
<p>longitude of observation (only used if <code>eos="gsw"</code>;
see &ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swTheta_+3A_latitude">latitude</code></td>
<td>
<p>latitude of observation (only used if <code>eos="gsw"</code>; see
&ldquo;Details&rdquo;).</p>
</td></tr>
<tr><td><code id="swTheta_+3A_eos">eos</code></td>
<td>
<p>equation of state, either <code>"unesco"</code> (references 1 and 2) or <code>"gsw"</code>
(references 3 and 4).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Different formulae are used depending on the equation of state. If <code>eos</code>
is <code>"unesco"</code>, the method of Fofonoff <em>et al.</em> (1983) is used
(see references 1 and 2).
Otherwise, <code>swTheta</code> uses <code><a href="gsw.html#topic+gsw_pt_from_t">gsw::gsw_pt_from_t()</a></code> from
the <a href="https://CRAN.R-project.org/package=gsw"><span class="pkg">gsw</span></a> package.
</p>
<p>If the first argument is a <code>ctd</code> or <code>section</code> object, then values
for salinity, etc., are extracted from it, and used for the calculation, and
the corresponding arguments to the present function are ignored.
</p>


<h3>Value</h3>

<p>Potential temperature (<code class="reqn">^\circ</code>C) of seawater, referenced
to pressure <code>referencePressure</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Fofonoff, P. and R. C. Millard Jr, 1983. Algorithms for computation of
fundamental properties of seawater.
<em>Unesco Technical Papers in Marine Science</em>, <em>44</em>, 53 pp
</p>
</li>
<li><p> Gill, A.E., 1982. <em>Atmosphere-ocean Dynamics</em>, Academic Press, New
York, 662 pp.
</p>
</li>
<li><p> IOC, SCOR, and IAPSO (2010). The international thermodynamic equation of
seawater-2010: Calculation and use of thermodynamic properties.  Technical
Report 56, Intergovernmental Oceanographic Commission, Manuals and Guide.
</p>
</li>
<li><p> McDougall, T.J. and P.M. Barker, 2011: Getting started with TEOS-10 and
the Gibbs Seawater (GSW) Oceanographic Toolbox, 28pp., SCOR/IAPSO WG127,
ISBN 978-0-646-55621-5.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Example 1: test value from Fofonoff et al., 1983
stopifnot(abs(36.8818748026 - swTheta(40, T90fromT68(40), 10000, 0, eos = "unesco")) &lt; 0.0000000001)

# Example 2: a deep-water station. Note that theta and CT are
# visually identical on this scale.
data(section)
stn &lt;- section[["station", 70]]
plotProfile(stn, "temperature", ylim = c(6000, 1000))
lines(stn[["theta"]], stn[["pressure"]], col = 2)
lines(stn[["CT"]], stn[["pressure"]], col = 4, lty = 2)
legend("bottomright",
    lwd = 1, col = c(1, 2, 4), lty = c(1, 1, 2),
    legend = c("in-situ", "theta", "CT"),
    title = sprintf("MAD(theta-CT)=%.4f", mean(abs(stn[["theta"]] - stn[["CT"]])))
)

</code></pre>

<hr>
<h2 id='swTSrho'>Seawater Temperature from Salinity and Density</h2><span id='topic+swTSrho'></span>

<h3>Description</h3>

<p>Compute <em>in-situ</em> temperature, given salinity, density, and pressure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swTSrho(
  salinity,
  density,
  pressure = NULL,
  eos = getOption("oceEOS", default = "gsw")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swTSrho_+3A_salinity">salinity</code></td>
<td>
<p><em>in-situ</em> salinity (PSU)</p>
</td></tr>
<tr><td><code id="swTSrho_+3A_density">density</code></td>
<td>
<p><em>in-situ</em> density or sigma value (kg/m<code class="reqn">^3</code>)</p>
</td></tr>
<tr><td><code id="swTSrho_+3A_pressure">pressure</code></td>
<td>
<p><em>in-situ</em> pressure (dbar)</p>
</td></tr>
<tr><td><code id="swTSrho_+3A_eos">eos</code></td>
<td>
<p>equation of state to be used, either <code>"unesco"</code> or
<code>"gsw"</code> (ignored at present).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Finds the temperature that yields the given density, with the given salinity
and pressure.  The method is a bisection search with temperature tolerance
0.001 <code class="reqn">^\circ</code>C.
</p>


<h3>Value</h3>

<p><em>In-situ</em> temperature in <code class="reqn">^\circ</code>C on the ITS-90
scale.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Fofonoff, P. and R. C. Millard Jr, 1983. Algorithms for
computation of fundamental properties of seawater. <em>Unesco Technical
Papers in Marine Science</em>, <em>44</em>, 53 pp
</p>
<p>Gill, A.E., 1982. <em>Atmosphere-ocean Dynamics</em>, Academic Press, New
York, 662 pp.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+swSTrho">swSTrho()</a></code>
</p>
<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>swTSrho(35, 23, 0, eos = "unesco") # 26.11301

</code></pre>

<hr>
<h2 id='swViscosity'>Seawater Viscosity</h2><span id='topic+swViscosity'></span>

<h3>Description</h3>

<p>Compute viscosity of seawater, in <code class="reqn">Pa\cdot s</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swViscosity(salinity, temperature)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swViscosity_+3A_salinity">salinity</code></td>
<td>
<p>either salinity (PSU) (in which case <code>temperature</code> and
<code>pressure</code> must be provided) <em>or</em> a <code>ctd</code> object (in which
case <code>salinity</code>, <code>temperature</code> and <code>pressure</code> are determined
from the object, and must not be provided in the argument list).</p>
</td></tr>
<tr><td><code id="swViscosity_+3A_temperature">temperature</code></td>
<td>
<p><em>in-situ</em> temperature (<code class="reqn">^\circ</code>C), defined
on the ITS-90 scale; see &ldquo;Temperature units&rdquo; in the documentation for
<code><a href="#topic+swRho">swRho()</a></code>, and the examples below.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the first argument is a <code>ctd</code> object, then salinity, temperature and
pressure values are extracted from it, and used for the calculation.
</p>
<p>The result is determined from a regression of the data provided in Table 87
of Dorsey (1940). The fit matches the table to within 0.2 percent at worst,
and with average absolute error of 0.07 percent. The maximum deviation from
the table is one unit in the last decimal place.
</p>
<p>No pressure dependence was reported by Dorsey (1940).
</p>


<h3>Value</h3>

<p>Viscosity of seawater in <code class="reqn">Pa\cdot s</code>.  Divide by density
to get kinematic viscosity in <code class="reqn">m^2/s</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>N. Ernest Dorsey (1940),
<em>Properties of ordinary Water-substance</em>,
American Chemical Society Monograph Series.  Reinhold
Publishing.
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>swViscosity(30, 10) # 0.001383779

</code></pre>

<hr>
<h2 id='swZ'>Vertical Coordinate</h2><span id='topic+swZ'></span>

<h3>Description</h3>

<p>Compute height above the surface. This is the negative of depth,
and so is defined simply in terms of <code><a href="#topic+swDepth">swDepth()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swZ(pressure, latitude = 45, eos = getOption("oceEOS", default = "gsw"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swZ_+3A_pressure">pressure</code></td>
<td>
<p>either pressure (dbar), in which case <code>lat</code> must also
be given, or a <code>ctd</code> object, in which case <code>lat</code> will be inferred
from the object.</p>
</td></tr>
<tr><td><code id="swZ_+3A_latitude">latitude</code></td>
<td>
<p>Latitude in <code class="reqn">^\circ</code>N.</p>
</td></tr>
<tr><td><code id="swZ_+3A_eos">eos</code></td>
<td>
<p>indication of formulation to be used, either <code>"unesco"</code> or
<code>"gsw"</code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>
</p>

<hr>
<h2 id='T68fromT90'>Convert From ITS-90 to IPTS-68 Temperature</h2><span id='topic+T68fromT90'></span>

<h3>Description</h3>

<p>Today's instruments typically record in the ITS-90 scale, but some old
datasets will be in the IPTS-68 scale. <code>T90fromT68()</code> converts from the
IPTS-68 to the ITS-90 scale, using Saunders' (1990) formula, while
<code>T68fromT90()</code> does the reverse. The difference between IPTS-68 and
ITS-90 values is typically a few millidegrees (see &lsquo;Examples&rsquo;), which
is seldom visible on a typical temperature profile, but may be of interest
in some precise work.  Mostly for historical interest, <code>T90fromT48()</code>
is provided to convert from the ITS-48 system to ITS-90.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>T68fromT90(temperature)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="T68fromT90_+3A_temperature">temperature</code></td>
<td>
<p>numeric vector of temperatures]in <code class="reqn">^\circ</code>C on the ITS-90 scale.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Corresponding temperatures in <code class="reqn">^\circ</code>C on the IPTS-68 scale.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>P. M. Saunders, 1990. The international temperature scale of
1990, ITS-90.  WOCE Newsletter, volume 10, September 1990, page 10.
http://www.nodc.noaa.gov/woce/wdiu/wocedocs/newsltr/news10/contents.htm
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
T68 &lt;- seq(3, 20, 1)
T90 &lt;- T90fromT68(T68)
sqrt(mean((T68-T90)^2))

</code></pre>

<hr>
<h2 id='T90fromT48'>Convert From ITS-48 to ITS-90 Temperature</h2><span id='topic+T90fromT48'></span>

<h3>Description</h3>

<p>Today's instruments typically record in the ITS-90 scale, but some old
datasets will be in the IPTS-68 scale. <code>T90fromT68()</code> converts from the
IPTS-68 to the ITS-90 scale, using Saunders' (1990) formula, while
<code>T68fromT90()</code> does the reverse. The difference between IPTS-68 and
ITS-90 values is typically a few millidegrees (see &lsquo;Examples&rsquo;), which
is seldom visible on a typical temperature profile, but may be of interest
in some precise work.  Mostly for historical interest, <code>T90fromT48()</code>
is provided to convert from the ITS-48 system to ITS-90.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>T90fromT48(temperature)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="T90fromT48_+3A_temperature">temperature</code></td>
<td>
<p>Vector of temperatures in <code class="reqn">^\circ</code>C on the IPTS-48 scale.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Corresponding temperatures in <code class="reqn">^\circ</code>C on the ITS-90 scale.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>P. M. Saunders, 1990. The international temperature scale of
1990, ITS-90.  WOCE Newsletter, volume 10, September 1990, page 10.
http://www.nodc.noaa.gov/woce/wdiu/wocedocs/newsltr/news10/contents.htm
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT68">T90fromT68</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
T68 &lt;- seq(3, 20, 1)
T90 &lt;- T90fromT68(T68)
sqrt(mean((T68-T90)^2))

</code></pre>

<hr>
<h2 id='T90fromT68'>Convert From IPTS-68 to ITS-90 Temperature</h2><span id='topic+T90fromT68'></span>

<h3>Description</h3>

<p>Today's instruments typically record in the ITS-90 scale, but some old
datasets will be in the IPTS-68 scale. <code>T90fromT68()</code> converts from the
IPTS-68 to the ITS-90 scale, using Saunders' (1990) formula, while
<code>T68fromT90()</code> does the reverse. The difference between IPTS-68 and
ITS-90 values is typically a few millidegrees (see &lsquo;Examples&rsquo;), which
is seldom visible on a typical temperature profile, but may be of interest
in some precise work.  Mostly for historical interest, <code>T90fromT48()</code>
is provided to convert from the ITS-48 system to ITS-90.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>T90fromT68(temperature)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="T90fromT68_+3A_temperature">temperature</code></td>
<td>
<p>numeric vector of temperatures in <code class="reqn">^\circ</code>C on the IPTS-68 scale.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Corresponding temperatures in <code class="reqn">^\circ</code>C on the ITS-90 scale.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>P. M. Saunders, 1990. The international temperature scale of
1990, ITS-90.  WOCE Newsletter, volume 10, September 1990, page 10.
http://www.nodc.noaa.gov/woce/wdiu/wocedocs/newsltr/news10/contents.htm
</p>


<h3>See Also</h3>

<p>Other functions that calculate seawater properties: 
<code><a href="#topic+T68fromT90">T68fromT90</a>()</code>,
<code><a href="#topic+T90fromT48">T90fromT48</a>()</code>,
<code><a href="#topic+computableWaterProperties">computableWaterProperties</a>()</code>,
<code><a href="#topic+locationForGsw">locationForGsw</a>()</code>,
<code><a href="#topic+swAbsoluteSalinity">swAbsoluteSalinity</a>()</code>,
<code><a href="#topic+swAlphaOverBeta">swAlphaOverBeta</a>()</code>,
<code><a href="#topic+swAlpha">swAlpha</a>()</code>,
<code><a href="#topic+swBeta">swBeta</a>()</code>,
<code><a href="#topic+swCSTp">swCSTp</a>()</code>,
<code><a href="#topic+swConservativeTemperature">swConservativeTemperature</a>()</code>,
<code><a href="#topic+swDepth">swDepth</a>()</code>,
<code><a href="#topic+swDynamicHeight">swDynamicHeight</a>()</code>,
<code><a href="#topic+swLapseRate">swLapseRate</a>()</code>,
<code><a href="#topic+swN2">swN2</a>()</code>,
<code><a href="#topic+swPressure">swPressure</a>()</code>,
<code><a href="#topic+swRho">swRho</a>()</code>,
<code><a href="#topic+swRrho">swRrho</a>()</code>,
<code><a href="#topic+swSCTp">swSCTp</a>()</code>,
<code><a href="#topic+swSR">swSR</a>()</code>,
<code><a href="#topic+swSTrho">swSTrho</a>()</code>,
<code><a href="#topic+swSigma0">swSigma0</a>()</code>,
<code><a href="#topic+swSigma1">swSigma1</a>()</code>,
<code><a href="#topic+swSigma2">swSigma2</a>()</code>,
<code><a href="#topic+swSigma3">swSigma3</a>()</code>,
<code><a href="#topic+swSigma4">swSigma4</a>()</code>,
<code><a href="#topic+swSigmaTheta">swSigmaTheta</a>()</code>,
<code><a href="#topic+swSigmaT">swSigmaT</a>()</code>,
<code><a href="#topic+swSigma">swSigma</a>()</code>,
<code><a href="#topic+swSoundAbsorption">swSoundAbsorption</a>()</code>,
<code><a href="#topic+swSoundSpeed">swSoundSpeed</a>()</code>,
<code><a href="#topic+swSpecificHeat">swSpecificHeat</a>()</code>,
<code><a href="#topic+swSpice">swSpice</a>()</code>,
<code><a href="#topic+swSstar">swSstar</a>()</code>,
<code><a href="#topic+swTFreeze">swTFreeze</a>()</code>,
<code><a href="#topic+swTSrho">swTSrho</a>()</code>,
<code><a href="#topic+swThermalConductivity">swThermalConductivity</a>()</code>,
<code><a href="#topic+swTheta">swTheta</a>()</code>,
<code><a href="#topic+swViscosity">swViscosity</a>()</code>,
<code><a href="#topic+swZ">swZ</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
T68 &lt;- seq(3, 20, 1)
T90 &lt;- T90fromT68(T68)
sqrt(mean((T68-T90)^2))

</code></pre>

<hr>
<h2 id='tail.oce'>Extract the End of an Oce Object</h2><span id='topic+tail.oce'></span>

<h3>Description</h3>

<p>Extract the End of an Oce Object
</p>
<p>This function handles the following object classes directly:
<a href="#topic+adp-class">adp</a>,
<a href="#topic+adv-class">adv</a>,
<a href="#topic+argo-class">argo</a> (selection by profile),
<a href="#topic+coastline-class">coastline</a>,
<a href="#topic+ctd-class">ctd</a>,
<a href="#topic+echosounder-class">echosounder</a> (selection by ping),
<a href="#topic+section-class">section</a> (selection by station)
and
<a href="#topic+topo-class">topo</a> (selection by longitude and latitude).
It does not handle
<a href="#topic+amsr-class">amsr</a> or
<a href="#topic+landsat-class">landsat</a>
yet, instead issuing a warning and returning <code>x</code> in those cases.
For all other classes, it calls <code><a href="utils.html#topic+tail">tail()</a></code>
with <code>n</code> as provided, for each item in the <code>data</code>
slot, issuing a warning if that item is not a vector; the
author is quite aware that this may not work well for all classes.
The plan is to handle all appropriate classes by July 2018.
Please contact the author if there is a class you need handled
before that date.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'oce'
tail(x, n = 6L, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tail.oce_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="tail.oce_+3A_n">n</code></td>
<td>
<p>Number of elements to extract, as for <code><a href="utils.html#topic+tail">tail()</a></code>.</p>
</td></tr>
<tr><td><code id="tail.oce_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p><code><a href="#topic+head.oce">head.oce()</a></code>, which yields the start of an <code>oce</code> object.
</p>

<hr>
<h2 id='threenum'>Calculate Minimum, Mean, and Maximum Values</h2><span id='topic+threenum'></span>

<h3>Description</h3>

<p>This is a simpler cousin of the standard <code><a href="stats.html#topic+fivenum">fivenum()</a></code> function,
used in <code><a href="base.html#topic+summary">summary()</a></code> functions for <code>oce</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>threenum(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="threenum_+3A_x">x</code></td>
<td>
<p>a vector or matrix of numerical values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of three values: the minimum, the mean, the
maximum.
</p>


<h3>Historical note</h3>

<p>On Aug 5, 2019, the dimension was dropped as the fourth column, and
this function returned to the original intention (revealed by its
name).  Another change is that the function now returns numerical
results, leaving the task of setting the number of digits to
<code><a href="base.html#topic+summary">summary()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
threenum(1:10)
</code></pre>

<hr>
<h2 id='tidalCurrent'>Tidal Current Dataset</h2><span id='topic+tidalCurrent'></span>

<h3>Description</h3>

<p>The <code>tidalCurrent</code> dataset contains tidal velocities reported in
Foreman's (1978) report (reference 1) on his Fortran code for the analysis of
tidal currents and provided in an associated webpage (reference 2).
Here, <code>tidalCurrent</code> is data frame containing
</p>

<ul>
<li> <p><code>time</code> a POSIXct time.
</p>
</li>
<li> <p><code>u</code> the eastward component of velocity in m/s.
</p>
</li>
<li> <p><code>v</code> the northward component of velocity in m/s.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Dan Kelley (reformatting data provided by Michael Foreman)
</p>


<h3>Source</h3>

<p>The data come from the <code>tide8.dat</code> and <code>tide9.dat</code> files provided
at reference 2.
</p>


<h3>References</h3>


<ol>
<li><p> Foreman, M. G. G. &quot;Manual for Tidal Currents Analysis and Prediction.&quot;
Pacific Marine Science Report.
British Columbia, Canada: Institute of Ocean Sciences, Patricia Bay, 1978.
</p>
</li>
<li> <p><code>https://www.dfo-mpo.gc.ca/science/documents/data-donnees/tidal-marees/tidpack.zip</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(tidalCurrent)
par(mfrow = c(2, 1))
oce.plot.ts(tidalCurrent$time, tidalCurrent$u, ylab = "u [m/s]")
abline(h = 0, col = 2)
oce.plot.ts(tidalCurrent$time, tidalCurrent$v, ylab = "v [m/s]")
abline(h = 0, col = 2)

</code></pre>

<hr>
<h2 id='tidedata'>Tidal Constituent Information</h2><span id='topic+tidedata'></span>

<h3>Description</h3>

<p>The <code>tidedata</code> dataset contains Tide-constituent information that is
use by <code><a href="#topic+tidem">tidem()</a></code> to fit tidal models.  <code>tidedata</code> is a list
containing
</p>

<dl>
<dt><code>const</code></dt><dd>
<p>a list containing vectors
<code>name</code> (a string with constituent name),
<code>freq</code> (the frequency, in cycles per hour),
<code>kmpr</code> (a string naming the comparison constituent, blank if there is none),
<code>ikmpr</code> (index of comparison constituent, or <code>0</code> if there is none),
<code>df</code> (frequency difference between constituent and its
comparison, used in the Rayleigh criterion),
<code>d1</code> through <code>d6</code> (the first through sixth Doodson numbers),
<code>semi</code>,
<code>nsat</code> (number of satellite constituents),
<code>ishallow</code>,
<code>nshallow</code>,
<code>doodsonamp</code>,
and
<code>doodsonspecies</code>.
</p>
</dd>
<dt><code>sat</code></dt><dd>
<p>a list containing vectors
<code>deldood</code>,
<code>phcorr</code>,
<code>amprat</code>,
<code>ilatfac</code>,
and
<code>iconst</code>.
</p>
</dd>
<dt><code>shallow</code></dt><dd>
<p>a list containing vectors
<code>iconst</code>,
<code>coef</code>,
and
<code>iname</code>.
</p>
</dd>
</dl>

<p>Apart from the use of <code>d1</code> through <code>d6</code>, the naming and content
follows <code>T_TIDE</code> (see Pawlowicz et al. 2002), which in turn builds upon
the analysis of Foreman (1978).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Source</h3>

<p>The data come from the <code>tide3.dat</code> file of the <code>T_TIDE</code>
package (Pawlowicz et al., 2002), and derive from Appendices provided by
Foreman (1978).  The data are scanned using &lsquo;<span class="file">tests/tide.R</span>&rsquo; in this
package, which also performs some tests using <code>T_TIDE</code> values as a
reference.
</p>


<h3>References</h3>

<p>Foreman, M. G. G., 1978. Manual for Tidal Currents Analysis and Prediction.
Pacific Marine Science Report. British Columbia, Canada: Institute of Ocean
Sciences, Patricia Bay.
</p>
<p>Pawlowicz, Rich, Bob Beardsley, and Steve Lentz, 2002.  Classical tidal
harmonic analysis including error estimates in MATLAB using <code>T_TIDE</code>.
Computers and Geosciences, 28, 929-937.
</p>


<h3>See Also</h3>

<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>

<hr>
<h2 id='tidem'>Fit a Tidal Model to a Timeseries</h2><span id='topic+tidem'></span>

<h3>Description</h3>

<p>The fit is done in terms of sine and cosine components at the indicated
tidal frequencies (after possibly pruning to satisfy the Rayleigh criterion,
as explained in phase 2 of the procedure outlined in
'&lsquo;Details&rsquo;), with the amplitude and phase being
calculated from the resultant coefficients on the sine and cosine terms.
The scheme was devised for hourly data; for other sampling schemes,
see &ldquo;Application to non-hourly data&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tidem(
  t,
  x,
  constituents,
  infer = NULL,
  latitude = NULL,
  rc = 1,
  regress = lm,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tidem_+3A_t">t</code></td>
<td>
<p>either a vector of times or a <a href="#topic+sealevel-class">sealevel</a> object
(as created with <code><a href="#topic+read.sealevel">read.sealevel()</a></code> or <code><a href="#topic+as.sealevel">as.sealevel()</a></code>).
In the first case, <code>x</code> must be provided.  In the second
case, though, any <code>x</code> that is provided will be ignored,
because sealevel objects contain both <code>time</code> and water
<code>elevation</code>, and the latter is used for <code>x</code>.</p>
</td></tr>
<tr><td><code id="tidem_+3A_x">x</code></td>
<td>
<p>an optional numerical vector holding something that varies with
time.  This is ignored if <code>t</code> is a <a href="#topic+sealevel-class">sealevel</a> object,
because it is inferred automatically, using <code>t[["elevation"]]</code>.</p>
</td></tr>
<tr><td><code id="tidem_+3A_constituents">constituents</code></td>
<td>
<p>an optional character vector holding the names
of tidal constituents to which the fit is done; see &ldquo;Details&rdquo;
and &ldquo;Constituent Naming Convention&rdquo;.</p>
</td></tr>
<tr><td><code id="tidem_+3A_infer">infer</code></td>
<td>
<p>a list of constituents to be inferred from
fitted constituents according to the method outlined
in Section 2.3.4 of Foreman (1978).
If <code>infer</code> is <code>NULL</code>, the default, then
no such inferences are made. Otherwise, some constituents
are computed based on other constituents, instead of being
determined by regression at the proper frequency.
If provided, <code>infer</code> must be a list containing
four elements:
<code>name</code>, a vector of strings naming the constituents to be
inferred; <code>from</code>, a vector of strings naming the fitted
constituents used as the sources for those inferences (these
source constituents are added to the regression list, if they
are not already there);
<code>amp</code>, a numerical vector of factors to be applied to the
source amplitudes; and <code>phase</code>, a numerical vector of angles,
in degrees, to be subtracted from the source phases. For example,
Following Foreman (1998), if any of the <code>name</code> items
have already been computed, then the suggested inference is ignored,
and the already-computed values are used.
</p>
<pre>
infer=list(name=c("P1","K2"),
           from=c("K1", "S2"),
           amp=c(0.33093, 0.27215),
           phase=c(-7.07, -22.4)
</pre>
<p>means that the amplitude of <code>P1</code> will be set as 0.33093 times the calculated amplitude
of <code>K1</code>, and that the <code>P1</code> phase will be set to the <code>K1</code> phase,
minus an offset of <code>-7.07</code> degrees.
(This example is used in the Foreman (1978) discussion of a
Fortran analysis code and also in Pawlowicz et al. (2002) discussion
of the T_TIDE Matlab code.
Rounded to the 0.1mm resolution of values reported in Foreman (1978)
and Pawlowicz et al. (2002),
the <code>tidem</code> results have root-mean-square amplitude difference
to Foreman's (1978) Appendix 7.3 of 0.06mm; by comparison,
the results in Table 1 of Pawlowicz et al. (2002) agree with Foreman's
results to RMS difference 0.04mm.)</p>
</td></tr>
<tr><td><code id="tidem_+3A_latitude">latitude</code></td>
<td>
<p>if provided, the latitude of the observations.  If not
provided, <code>tidem</code> will try to infer this from the first argument,
if it is a <a href="#topic+sealevel-class">sealevel</a> object.</p>
</td></tr>
<tr><td><code id="tidem_+3A_rc">rc</code></td>
<td>
<p>the value of the coefficient in the Rayleigh criterion.</p>
</td></tr>
<tr><td><code id="tidem_+3A_regress">regress</code></td>
<td>
<p>function to be used for regression, by default
<code><a href="stats.html#topic+lm">lm()</a></code>, but could be for example <code>rlm</code> from the
<code>MASS</code> package.</p>
</td></tr>
<tr><td><code id="tidem_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A summary of constituents used by <code><a href="#topic+tidem">tidem()</a></code> may be found with:
</p>
<pre>
data(tidedata)
print(tidedata$const)
</pre>
<p>A multi-stage procedure is used to establish the list of tidal
constituents to be used in the fit.
</p>
<p><em>Phase 1: initial selection.</em>
</p>
<p>The initial list tidal constituents (prior to the pruning phase described
below) to be used in the analysis are specified as follows;
see &ldquo;Constituent Naming Convention&rdquo;.
</p>

<ol>
<li><p> If <code>constituents</code> is not provided, then the constituent
list will be made up of the 69 constituents designated by Foreman as &quot;standard&quot;.
These include astronomical frequencies and some shallow-water frequencies,
and are as follows: <code>c("Z0", "SA", "SSA", "MSM", "MM", "MSF", "MF", "ALP1", "2Q1", "SIG1", "Q1", "RHO1", "O1", "TAU1", "BET1", "NO1", "CHI1", "PI1", "P1", "S1", "K1", "PSI1", "PHI1", "THE1", "J1", "SO1", "OO1", "UPS1", "OQ2", "EPS2", "2N2", "MU2", "N2", "NU2", "GAM2", "H1", "M2", "H2", "MKS2", "LDA2", "L2", "T2", "S2", "R2", "K2", "MSN2", "ETA2", "MO3", "M3", "SO3", "MK3", "SK3", "MN4", "M4", "SN4", "MS4", "MK4", "S4", "SK4", "2MK5", "2SK5", "2MN6", "M6", "2MS6", "2MK6", "2SM6", "MSK6", "3MK7", "M8")</code>.
</p>
</li>
<li><p> If the first item in <code>constituents</code> is the string
<code>"standard"</code>, then a provisional list is set up as in Case 1, and then
the (optional) rest of the elements of <code>constituents</code> are examined, in
order.  Each of these constituents is based on the name of a tidal
constituent in the Foreman (1978) notation.  (To get the list, execute
<code>data(tidedata)</code> and then execute <code>cat(tideData$name)</code>.)  Each
named constituent is added to the existing list, if it is not already there.
But, if the constituent is preceded by a minus sign, then it is removed
from the list (if it is already there).  Thus, for example, a user
might specify <code>constituents=c("standard", "-M2", "ST32")</code> to remove the M2
constituent and add the ST32 constituent.
</p>
</li>
<li><p> If the first item is not <code>"standard"</code>, then the list of
constituents is processed as in Case 2, but without starting with the
standard list. As an example, <code>constituents=c("K1", "M2")</code> would fit
for just the K1 and M2 components. (It would be strange to use a minus sign
to remove items from the list, but the function allows that.)
</p>
</li></ol>

<p>In each of the above cases, the list is reordered in frequency prior to the
analysis, so that the results of <code><a href="#topic+summary+2Ctidem-method">summary,tidem-method()</a></code> will be in a
familiar form.
</p>
<p><em>Phase 2: pruning based on Rayleigh's criterion.</em>
</p>
<p>Once the initial constituent list is determined during Phase 1,
<code>tidem</code> applies the Rayleigh criterion, which holds that
constituents of frequencies <code class="reqn">f_1</code> and <code class="reqn">f_2</code> cannot be
resolved unless the time series spans a time interval of at least
<code class="reqn">rc/(f_1-f_2)</code>.  The details of the decision of which
constituent to prune is fairly complicated, involving decisions
based on a comparison table.  The details of this process are provided
by Foreman (1978).
</p>
<p><em>Phase 3: optionally overruling Rayleigh's criterion.</em>
</p>
<p>The pruning list from phase 2 can be overruled by the user. This
is done by retaining any constituents that the user has named
in the <code>constituents</code> parameter.  This action
was added on 2017-12-27, to make <code>tidem</code> behave the same
way as the Foreman (1978) code, as illustrated in his
Appendices 7.2 and 7.3. (As an aside, his Appendix 7.3 has some errors,
e.g. the frequency for the 2SK5 constituent is listed there (p58) as
0.20844743, but it is listed as 0.2084474129 in his Appendix 7.1 (p41).
For this reason, the frequency comparison is relaxed to a <code>tol</code>
value of <code>1e-7</code> in a portion of the oce test suite
(see <code>tests/testthat/test_tidem.R</code> in the source).
</p>
<p><em>Comments on phases 2 and 3</em>
</p>
<p>A specific example may be of help in understanding the removal of unresolvable
constituents. For example, the <code>data(sealevel)</code> dataset is of length
6718 hours, and this is too short to resolve the full list of constituents,
with the conventional (and, really, necessary) limit of <code>rc=1</code>.
From Table 1 of Foreman (1978), this timeseries is too short to resolve the
<code>SA</code> constituent, so that <code>SA</code> will not be in the resultant.
Similarly, Table 2 of Foreman (1978) dictates the removal of
<code>PI1</code>, <code>S1</code> and <code>PSI1</code> from the list. And, finally,
Table 3 of Foreman (1978) dictates the removal of
<code>H1</code>, <code>H2</code>, <code>T2</code> and <code>R2</code>, and since that document
suggests that <code>GAM2</code> be subsumed into <code>H1</code>,
then if <code>H1</code> is already being deleted, then
<code>GAM2</code> will also be deleted.
</p>


<h3>Value</h3>

<p>An object of <a href="#topic+tidem-class">tidem</a>, consisting of
</p>
<table>
<tr><td><code>const</code></td>
<td>
<p>constituent number, e.g. 1 for <code>Z0</code>, 1 for <code>SA</code>,
etc.</p>
</td></tr> <tr><td><code>model</code></td>
<td>
<p>the regression model</p>
</td></tr> <tr><td><code>name</code></td>
<td>
<p>a vector of constituent
names, in non-subscript format, e.g. &quot;<code>M2</code>&quot;.</p>
</td></tr> <tr><td><code>frequency</code></td>
<td>
<p>a vector
of constituent frequencies, in inverse hours.</p>
</td></tr> <tr><td><code>amplitude</code></td>
<td>
<p>a vector of
fitted constituent amplitudes, in metres.</p>
</td></tr> <tr><td><code>phase</code></td>
<td>
<p>a vector of fitted
constituent phase.  NOTE: The definition of phase is likely to change as
this function evolves.  For now, it is phase with respect to the first data
sample.</p>
</td></tr> <tr><td><code>p</code></td>
<td>
<p>a vector containing a sort of p value for each
constituent.  This is calculated as the average of the p values for the
sine() and cosine() portions used in fitting; whether it makes any sense is
an open question.</p>
</td></tr>
</table>


<h3>Application to non-hourly data</h3>

<p>The framework on which <code><a href="#topic+tidem">tidem()</a></code> rests on the assumption of data
that have been sampled on a 1-hour interval (see e.g. Foreman, 1977).
Since regression (as opposed to spectral analysis) is used to infer
the amplitude and phase of tidal constituents, data gaps do not pose
a serious problem. Sampling intervals under an hour are also not a
problem. However, trying to use <code><a href="#topic+tidem">tidem()</a></code> on time series that are
sampled at uniform intervals that exceed 1 hour can lead to results
that are difficult to interpret.  For example, some drifter data are
sampled at a 6-hour interval.  This makes it impossible to fit for the
S4 component (which has exactly 6 cycles per day), because the method
works by constructing sine and cosine series at tidal frequencies and
using these as the basis for regression.  Each of these series will have
a constant value through the constructed time, and regression cannot handle
that (in addition to a constant-value constructed series that is used to fit
for the Z0 constituent).  <code><a href="#topic+tidem">tidem()</a></code> tries to handle such problems by examining
the range of the constructed sine and cosine time-series, omitting any
constituents that yield near-constant values in either of these. Messages are
issued if this problem is encountered.  This prevents failure of the regression,
and the predictions of the regression seem to represent the data reasonably well,
but the inferred constituent amplitudes are not physically reasonable. Cautious
use of <code><a href="#topic+tidem">tidem()</a></code> to infer individual constituents might be warranted, but
users must be aware that the results will be difficult to interpret. The tool
is simply not designed for this use.
</p>


<h3>Bugs</h3>


<ol>
<li><p> This function is not fully developed yet, and both the
form of the call and the results of the calculation may change.
</p>
</li>
<li><p> The reported <code>p</code> value may make no sense at all, and it might be
removed in a future version of this function. Perhaps a significance level
should be presented, as in the software developed by both Foreman and
Pawlowicz.
</p>
</li></ol>



<h3>Constituent Naming Convention</h3>

<p><code>tidem</code> uses constituent names that follow the convention
set by Foreman (1978). This convention is slightly different
from that used in the T-TIDE package of Pawlowicz et al.
(2002), with Foreman's <code>UPS1</code> and <code>M8</code> becoming
<code>UPSI</code> and <code>MS</code> in T-TIDE. To permit the use of either notation,
<code><a href="#topic+tidem">tidem()</a></code> uses <code><a href="#topic+tidemConstituentNameFix">tidemConstituentNameFix()</a></code> to
convert from T-TIDE names to the
Foreman names, issuing warnings when doing so.
</p>


<h3>Agreement with <code>T_TIDE</code> results</h3>

<p>The <code>tidem</code> amplitude and phase results, obtained with
</p>
<pre>
tidem(sealevelTuktoyaktuk, constituents=c("standard", "M10"),
    infer=list(name=c("P1", "K2"),
        from=c("K1", "S2"),
        amp=c(0.33093, 0.27215),
        phase=c(-7.07, -22.40)))
</pre>
<p>match the <code>T_TIDE</code> values listed in
Table 1 of Pawlowicz et al. (2002),
after rounding amplitude and phase to 4 and 2 digits past
the decimal place, respectively, to match the format of
that table.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Foreman, M G., 1977 (revised 1996).
Manual for Tidal Heights Analysis and Prediction.
Pacific Marine Science Report 77-10.
British Columbia, Canada: Institute of Ocean Sciences, Patricia Bay.
</p>
<p>Foreman, M. G. G., 1978.
Manual for Tidal Currents Analysis and Prediction.
Pacific Marine Science Report 78-6.
British Columbia, Canada: Institute of Ocean Sciences, Patricia Bay,
</p>
<p>Foreman, M. G. G., Neufeld, E. T., 1991.
Harmonic tidal analyses of long time series.
International Hydrographic Review, 68 (1), 95-108.
</p>
<p>Leffler, K. E. and D. A. Jay, 2009.  Enhancing tidal harmonic analysis:
Robust (hybrid) solutions.  Continental Shelf Research, 29(1):78-88.
</p>
<p>Pawlowicz, Rich, Bob Beardsley, and Steve Lentz, 2002.  Classical tidal
harmonic analysis including error estimates in MATLAB using <code>T_TIDE</code>.
Computers and Geosciences, 28, 929-937.
</p>


<h3>See Also</h3>

<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# The demonstration time series from Foreman (1978),
# also used in T_TIDE (Pawlowicz, 2002).
data(sealevelTuktoyaktuk)
tide &lt;- tidem(sealevelTuktoyaktuk)
summary(tide)

# AIC analysis
extractAIC(tide[["model"]])

# Fake data at M2
library(oce)
data("tidedata")
M2 &lt;- with(tidedata$const, freq[name == "M2"])
t &lt;- seq(0, 10 * 86400, 3600)
eta &lt;- sin(M2 * t * 2 * pi / 3600)
sl &lt;- as.sealevel(eta)
m &lt;- tidem(sl)
summary(m)

</code></pre>

<hr>
<h2 id='tidem-class'>Class to Store Tidal Models</h2><span id='topic+tidem-class'></span>

<h3>Description</h3>

<p>This class stores tidal-constituent models.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>tidem</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>tidem</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>tidem</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+tidem-class">tidem</a> objects (see <code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+tidem-class">tidem</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions that plot oce data: 
<code><a href="#topic+download.amsr">download.amsr</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+plot+2Camsr-method">plot,amsr-method</a></code>,
<code><a href="#topic+plot+2Cargo-method">plot,argo-method</a></code>,
<code><a href="#topic+plot+2Cbremen-method">plot,bremen-method</a></code>,
<code><a href="#topic+plot+2Ccm-method">plot,cm-method</a></code>,
<code><a href="#topic+plot+2Ccoastline-method">plot,coastline-method</a></code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plot+2Cgps-method">plot,gps-method</a></code>,
<code><a href="#topic+plot+2Cladp-method">plot,ladp-method</a></code>,
<code><a href="#topic+plot+2Clandsat-method">plot,landsat-method</a></code>,
<code><a href="#topic+plot+2Clisst-method">plot,lisst-method</a></code>,
<code><a href="#topic+plot+2Clobo-method">plot,lobo-method</a></code>,
<code><a href="#topic+plot+2Cmet-method">plot,met-method</a></code>,
<code><a href="#topic+plot+2Codf-method">plot,odf-method</a></code>,
<code><a href="#topic+plot+2Crsk-method">plot,rsk-method</a></code>,
<code><a href="#topic+plot+2Csatellite-method">plot,satellite-method</a></code>,
<code><a href="#topic+plot+2Csealevel-method">plot,sealevel-method</a></code>,
<code><a href="#topic+plot+2Csection-method">plot,section-method</a></code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>
</p>
<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>

<hr>
<h2 id='tidemAstron'>Astronomical Calculations for tidem</h2><span id='topic+tidemAstron'></span>

<h3>Description</h3>

<p>Do some astronomical calculations for <code><a href="#topic+tidem">tidem()</a></code>.  This function is based directly
on <code>t_astron</code> in the <code>T_TIDE</code> Matlab package (see Pawlowicz et al. 2002),
which inherits from the Fortran code described by Foreman (1978).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tidemAstron(t)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tidemAstron_+3A_t">t</code></td>
<td>
<p>Either a time in <code>POSIXct</code> format (with <code>"UTC"</code> timezone,
or else odd behaviours may result),
or an integer. In the second case, it is converted to a time with
<code><a href="#topic+numberAsPOSIXct">numberAsPOSIXct()</a></code>, using <code>tz="UTC"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>list</code> containing items named
<code>astro</code> and <code>ader</code> (see the <code>T_TIDE</code> documentation).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley translated this from the <code>t_astron</code> function
of the <code>T_TIDE</code> Matlab package (see Pawlowicz et al. 2002).
</p>


<h3>References</h3>


<ul>
<li><p> Foreman, M. G. G., 1978. Manual for Tidal Currents Analysis and Prediction.
Pacific Marine Science Report. British Columbia, Canada: Institute of Ocean
Sciences, Patricia Bay.
</p>
</li>
<li><p> Pawlowicz, Rich, Bob Beardsley, and Steve Lentz, 2002.  Classical tidal
harmonic analysis including error estimates in MATLAB using <code>T_TIDE</code>.
Computers and Geosciences, 28, 929-937.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tidemAstron(as.POSIXct("2008-01-22 18:50:24"))

</code></pre>

<hr>
<h2 id='tidemConstituentNameFix'>Change Tidal Constituent Name from T-TIDE to Foreman Convention</h2><span id='topic+tidemConstituentNameFix'></span>

<h3>Description</h3>

<p>This is used by <code><a href="#topic+tidem">tidem()</a></code> to permit users to specify constituent names in either
the T-TIDE convention (see Pawlowicz et al. 2002) or Foreman convention
(see Foreman (1978). There are only two such instances:
<code>"MS"</code>, which gets translated to <code>"M8"</code>, and <code>"UPSI"</code>,
which gets translated to <code>"UPS1"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tidemConstituentNameFix(names, debug = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tidemConstituentNameFix_+3A_names">names</code></td>
<td>
<p>a vector of character values, holding constituent names</p>
</td></tr>
<tr><td><code id="tidemConstituentNameFix_+3A_debug">debug</code></td>
<td>
<p>an integer controlling warnings. If this is zero, then no warnings
are issued during processing; otherwise, as is the default, warnings are
issued for each conversion that is required.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of character values of tidal constituent names, in the Foreman naming convention.
</p>


<h3>References</h3>

<p>Foreman, M. G. G., 1978. Manual for Tidal Currents Analysis and Prediction.
Pacific Marine Science Report. British Columbia, Canada: Institute of Ocean
Sciences, Patricia Bay.
</p>
<p>Pawlowicz, Rich, Bob Beardsley, and Steve Lentz, 2002.  Classical tidal
harmonic analysis including error estimates in MATLAB using <code>T_TIDE</code>.
Computers and Geosciences, 28, 929-937.
</p>

<hr>
<h2 id='tidemVuf'>Nodal Modulation Calculations for Tidal Analyses</h2><span id='topic+tidemVuf'></span>

<h3>Description</h3>

<p>Carry out nodal modulation calculations for <code><a href="#topic+tidem">tidem()</a></code>. This function is based directly
on <code>t_vuf</code> in the <code>T_TIDE</code> Matlab package (Pawlowicz et al., 2002),
which inherits from the Fortran code described by Foreman (1978).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tidemVuf(t, j, latitude = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tidemVuf_+3A_t">t</code></td>
<td>
<p>a single time in <code><a href="base.html#topic+POSIXct">POSIXct()</a></code> format, with timezone <code>"UTC"</code>.</p>
</td></tr>
<tr><td><code id="tidemVuf_+3A_j">j</code></td>
<td>
<p>integer vector, giving indices of tidal constituents to use.</p>
</td></tr>
<tr><td><code id="tidemVuf_+3A_latitude">latitude</code></td>
<td>
<p>optional numerical value containing the latitude in degrees North.
If not provided, <code>u</code> in the return value will be a vector consisting of
repeated 0 value, and <code>f</code> will be a vector of repeated 1 value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>list</code> containing
items named <code>v</code>, <code>u</code> and <code>f</code> as described in the <code>T_TIDE</code> documentation,
as well in Pawlowicz et al. (2002) and Foreman (1978).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley translated this from the <code>t_vuf</code> function
of the <code>T_TIDE</code> Matlab package (see Pawlowicz et al. 2002).
</p>


<h3>References</h3>


<ul>
<li><p> Foreman, M. G. G., 1978. Manual for Tidal Currents Analysis and Prediction.
Pacific Marine Science Report. British Columbia, Canada: Institute of Ocean
Sciences, Patricia Bay.
</p>
</li>
<li><p> Pawlowicz, Rich, Bob Beardsley, and Steve Lentz, 2002.  Classical tidal
harmonic analysis including error estimates in MATLAB using <code>T_TIDE</code>.
Computers and Geosciences, 28, 929-937.
</p>
</li></ul>



<h3>See Also</h3>

<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>,
<code><a href="#topic+webtide">webtide</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Look up values for the M2 constituent in Halifax Harbour, Canada.
library(oce)
data("tidedata")
j &lt;- with(tidedata$const, which(name == "M2"))
tidemVuf(t = as.POSIXct("2008-01-22 18:50:24"), j = j, lat = 44.63)

</code></pre>

<hr>
<h2 id='timeToArgoJuld'>Convert Time to Argo Julian Day (juld)</h2><span id='topic+timeToArgoJuld'></span>

<h3>Description</h3>

<p>This converts a POSIXct time into an Argo julian day, with the convention
that juld=0 at 1950-01-01.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>timeToArgoJuld(t)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="timeToArgoJuld_+3A_t">t</code></td>
<td>
<p>A POSIXct time or a string that can be converted to a POSIXct time</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jaimie Harbin and Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>timeToArgoJuld("2020-07-01")

</code></pre>

<hr>
<h2 id='titleCase'>Capitalize First Letter of Each of a Vector of Words</h2><span id='topic+titleCase'></span>

<h3>Description</h3>

<p>This is used in making labels for data names in some ctd functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>titleCase(w)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="titleCase_+3A_w">w</code></td>
<td>
<p>vector of character strings</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of strings patterned on <code>w</code> but with first letter
in upper case and others in lower case
</p>

<hr>
<h2 id='toEnu'>Rotate Acoustic-Doppler Data to the ENU Coordinate System</h2><span id='topic+toEnu'></span>

<h3>Description</h3>

<p>Rotate Acoustic-Doppler Data to the ENU Coordinate System
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toEnu(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="toEnu_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> or <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="toEnu_+3A_...">...</code></td>
<td>
<p>extra arguments that are passed on to <code><a href="#topic+toEnuAdp">toEnuAdp()</a></code>
or <code><a href="#topic+toEnuAdv">toEnuAdv()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same class as <code>x</code>, but with velocities
in the enu coordinate system
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='toEnuAdp'>Convert an adp Object to ENU Coordinates</h2><span id='topic+toEnuAdp'></span>

<h3>Description</h3>

<p>Convert an adp Object to ENU Coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toEnuAdp(x, declination = 0, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="toEnuAdp_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="toEnuAdp_+3A_declination">declination</code></td>
<td>
<p>magnetic declination to be added to the heading, to get
ENU with N as &quot;true&quot; north.</p>
</td></tr>
<tr><td><code id="toEnuAdp_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p><code style="white-space: pre;">&#8288;https://nortek.zendesk.com/hc/en-us/articles/360029820971-How-is-a-Coordinate-transformation-done-&#8288;</code>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+read.adp">read.adp()</a></code> for notes on functions relating to
<code>"adp"</code> objects.  Also, see <code><a href="#topic+beamToXyzAdp">beamToXyzAdp()</a></code> and
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp()</a></code>.
</p>
<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='toEnuAdv'>Convert an adv Object to ENU Coordinates</h2><span id='topic+toEnuAdv'></span>

<h3>Description</h3>

<p>Convert an adv Object to ENU Coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toEnuAdv(x, declination = 0, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="toEnuAdv_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="toEnuAdv_+3A_declination">declination</code></td>
<td>
<p>magnetic declination to be added to the heading, to get
ENU with N as &quot;true&quot; north.</p>
</td></tr>
<tr><td><code id="toEnuAdv_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li> <p><code style="white-space: pre;">&#8288;https://nortek.zendesk.com/hc/en-us/articles/360029820971-How-is-a-Coordinate-transformation-done-&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>See <code><a href="#topic+read.adv">read.adv()</a></code> for notes on functions relating to
<code>"adv"</code> objects.  Also, see <code><a href="#topic+beamToXyzAdv">beamToXyzAdv()</a></code> and
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv()</a></code>.
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='topo-class'>Class to Store Topographic Data</h2><span id='topic+topo-class'></span>

<h3>Description</h3>

<p>This class stores topographic data, as read with
<code><a href="#topic+read.topo">read.topo()</a></code> or assembled with <code><a href="#topic+as.topo">as.topo()</a></code>.
Plotting is handled with <code><a href="#topic+plot+2Ctopo-method">plot,topo-method()</a></code>
and summaries with <code><a href="#topic+summary+2Ctopo-method">summary,topo-method()</a></code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>topo</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
The key items stored in this slot are: <code>longititude</code>, <code>latitude</code>, and <code>z</code>.</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>topo</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>topo</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+topo-class">topo</a> objects (see <code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+topo-class">topo</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to topo data: 
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic+as.topo">as.topo</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+read.topo">read.topo</a>()</code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+summary+2Ctopo-method">summary,topo-method</a></code>,
<code><a href="#topic+topoInterpolate">topoInterpolate</a>()</code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>
</p>

<hr>
<h2 id='topoInterpolate'>Interpolate Within a topo Object</h2><span id='topic+topoInterpolate'></span>

<h3>Description</h3>

<p>Bilinear interpolation is used so that values will vary smoothly within a
longitude-latitude grid cell. Note that the sign convention for
<code>longitude</code> and <code>latitude</code> must match that in <code>topo</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>topoInterpolate(longitude, latitude, topo)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="topoInterpolate_+3A_longitude">longitude</code></td>
<td>
<p>Vector of longitudes (in the same sign convention as used in
<code>topo</code>).</p>
</td></tr>
<tr><td><code id="topoInterpolate_+3A_latitude">latitude</code></td>
<td>
<p>Vector of latitudes (in the same sign convention as used in
<code>topo</code>).</p>
</td></tr>
<tr><td><code id="topoInterpolate_+3A_topo">topo</code></td>
<td>
<p>A <a href="#topic+topo-class">topo</a> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of heights giving the elevation of the earth above means sea
level at the indicated location on the earth.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to topo data: 
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic+as.topo">as.topo</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+read.topo">read.topo</a>()</code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+summary+2Ctopo-method">summary,topo-method</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(topoWorld)
# "The Gully", approx. 400m deep, connects Gulf of St Lawrence with North Atlantic
topoInterpolate(45, -57, topoWorld)

</code></pre>

<hr>
<h2 id='topoWorld'>Global Topographic Data (at Half-degree Resolution)</h2><span id='topic+topoWorld'></span>

<h3>Description</h3>

<p>Global topographic dataset at half-degree resolution, downloaded from
a NOAA server on May 18, 2019.  Longitude, accessible as
<code>topoWorld[["longitude"]]</code>, ranges from -179.75 to 129.75 degrees north.
Latitude (<code>topoWorld[["latitude"]]</code>) ranges from -89.75 to 89.75 degrees east.
Height (<code>topoWorld[["z"]]</code>) is measured in metres above nominal sea level.
</p>
<p>The coarse resolution can be a problem in plotting depth contours along with
coastlines in regions of steep topography. For example, near the southeast
corner of Newfoundland, a 200m contour will overlap a coastline drawn with
<code>coastlineWorldFine</code> from the <a href="https://CRAN.R-project.org/package=ocedata"><span class="pkg">ocedata</span></a> package. The solution in such cases is to
download a higher-resolution topography file, perhaps using
<code><a href="#topic+download.topo">download.topo()</a></code>, and then use <code><a href="#topic+read.topo">read.topo()</a></code>
to create another <code>topo</code> object.  (With other data
sources, <code><a href="#topic+as.topo">as.topo()</a></code> may be helpful.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(topoWorld)
</code></pre>


<h3>Historical note</h3>

<p>From late 2009 until May 18, 2019, the <code>topoWorld</code> dataset was created
with a fairly complicated code that read a binary file downloaded from NOAA
(&lsquo;<span class="samp">&#8288;http://www.ngdc.noaa.gov/mgg/global/relief/ETOPO5/TOPO/ETOPO5&#8288;</span>&rsquo;),
decoded, decimated from 1/12th degree resolution to 1/2 degree resolution, and
passed through <code><a href="#topic+matrixShiftLongitude">matrixShiftLongitude()</a></code> to put longitude
between -180 and 180 degrees. The new scheme for creating the dataset,
(see &ldquo;Source&rdquo;) is much simpler, and also a much better model
of how users are likely to deal with topography files in the more
modern netCDF format. Note that the new version differs from the old one
in longitude and latitude being shifted by 1/4 degree,
and by a mean  elevation difference of under 10m. The old and new
versions appear identical when plotted at the global scale that is
the recommended for such a coarse topographic file.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
data(topoWorld)
par(mfrow=c(2, 1))
plot(topoWorld, location=NULL)
imagep(topoWorld)
</pre>


<h3>Source</h3>

<p>This is created with <code><a href="#topic+read.topo">read.topo()</a></code>, using a file downloaded with
</p>
<pre>
topoFile &lt;- download.topo(west=-180, east=180, south=-90, north=90,
   resolution=30, destdir=".")
</pre>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+wind">wind</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other things related to topo data: 
<code><a href="#topic++5B+5B+2Ctopo-method">[[,topo-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctopo-method">[[&lt;-,topo-method</a></code>,
<code><a href="#topic+as.topo">as.topo</a>()</code>,
<code><a href="#topic+download.topo">download.topo</a>()</code>,
<code><a href="#topic+plot+2Ctopo-method">plot,topo-method</a></code>,
<code><a href="#topic+read.topo">read.topo</a>()</code>,
<code><a href="#topic+subset+2Ctopo-method">subset,topo-method</a></code>,
<code><a href="#topic+summary+2Ctopo-method">summary,topo-method</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+topoInterpolate">topoInterpolate</a>()</code>
</p>

<hr>
<h2 id='unabbreviateYear'>Determine Year From Various Abbreviations</h2><span id='topic+unabbreviateYear'></span>

<h3>Description</h3>

<p>Various data files may contain various abbreviations for years.  For
example, 99 refers to 1999, and 8 refers to 2008.  Sometimes, even 108
refers to 2008 (the idea being that the &quot;zero&quot; year was 1900).  This
function deals with the three cases mentioned.  It will fail if someone
supplies 60, meaning year 2060 as opposed to 1960.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unabbreviateYear(year)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unabbreviateYear_+3A_year">year</code></td>
<td>
<p>a year, or vector of years, possibly abbreviated</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to time: 
<code><a href="#topic+ctimeToSeconds">ctimeToSeconds</a>()</code>,
<code><a href="#topic+julianCenturyAnomaly">julianCenturyAnomaly</a>()</code>,
<code><a href="#topic+julianDay">julianDay</a>()</code>,
<code><a href="#topic+numberAsHMS">numberAsHMS</a>()</code>,
<code><a href="#topic+numberAsPOSIXct">numberAsPOSIXct</a>()</code>,
<code><a href="#topic+secondsToCtime">secondsToCtime</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fullYear &lt;- unabbreviateYear(c(99, 8, 108))
</code></pre>

<hr>
<h2 id='undriftTime'>Correct for Drift in an Instrument Clock</h2><span id='topic+undriftTime'></span>

<h3>Description</h3>

<p>It is assumed that the instrument clock matches the real time at the start
of the sampling, and that the clock drifts linearly (i.e. is uniformly fast
or slow) over the sampling interval.  Linear interpolation is used to infer
the values of all variables in the <code>data</code> slot.  The data length is
altered in this process, e.g. a slow instrument clock (positive
<code>slowEnd</code>) takes too few samples in a given time interval, so
<code>undriftTime</code> will increase the number of data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>undriftTime(x, slowEnd = 0, tname = "time")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="undriftTime_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="undriftTime_+3A_slowend">slowEnd</code></td>
<td>
<p>number of seconds to add to final instrument time, to get the
correct time of the final sample.  This will be a positive number, for a
&quot;slow&quot; instrument clock.</p>
</td></tr>
<tr><td><code id="undriftTime_+3A_tname">tname</code></td>
<td>
<p>Character string indicating the name of the time column in the
<code>data</code> slot of <code>x</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same class as <code>x</code>, with the <code>data</code> slot
adjusted appropriately.
</p>


<h3>Sample of Usage</h3>

<pre>
library(oce)
file &lt;- "~/data/archive/sleiwex/2008/moorings/m08/pt/rbr_011855/raw/pt_rbr_011855.dat"
rbr011855 &lt;- read.oce(file)
d &lt;- subset(rbr011855, time &lt; as.POSIXct("2008-06-25 10:05:00"))
x &lt;- undriftTime(d, 1)   # clock lost 1 second over whole experiment
summary(d)
summary(x)
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>

<hr>
<h2 id='unduplicateNames'>Rename Duplicated Character Strings</h2><span id='topic+unduplicateNames'></span>

<h3>Description</h3>

<p>Append numeric suffices to character strings, to avoid repeats.
This is used by various data
input functions, to handle the fact that several oceanographic data
formats permit the reuse of variable names within a given file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unduplicateNames(strings, style = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unduplicateNames_+3A_strings">strings</code></td>
<td>
<p>Vector of character strings.</p>
</td></tr>
<tr><td><code id="unduplicateNames_+3A_style">style</code></td>
<td>
<p>An integer giving the style. If <code>style</code>
is <code>1</code>, then e.g. a triplicate of <code>"a"</code> yields
<code>"a"</code>, <code>"a1"</code>, and <code>"a2"</code>.
If <code>style</code> is <code>2</code>, then the same input yields
<code>"a_001"</code>, <code>"a_002"</code>, and <code>"a_003"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of strings with repeats distinguished by suffix.
</p>


<h3>See Also</h3>

<p>Used by <code><a href="#topic+read.ctd.sbe">read.ctd.sbe()</a></code> with <code>style=1</code> to
rename repeated data elements (e.g. for multiple temperature sensors)
in CTD data, and by <code><a href="#topic+read.odf">read.odf()</a></code> with <code>style=2</code> on
key-value pairs within ODF metadata.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>unduplicateNames(c("a", "b", "a", "c", "b"))
unduplicateNames(c("a", "b", "a", "c", "b"), style = 2)
</code></pre>

<hr>
<h2 id='ungrid'>Extract (x, y, z) From (x, y, grid)</h2><span id='topic+ungrid'></span>

<h3>Description</h3>

<p>Extract the grid points from a grid, returning columns.
This is useful for e.g. gridding large datasets, in which the first step
might be to use <code><a href="#topic+binMean2D">binMean2D()</a></code>, followed by
<code><a href="#topic+interpBarnes">interpBarnes()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ungrid(x, y, grid)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ungrid_+3A_x">x</code></td>
<td>
<p>a vector holding the x coordinates of grid.</p>
</td></tr>
<tr><td><code id="ungrid_+3A_y">y</code></td>
<td>
<p>a vector holding the y coordinates of grid.</p>
</td></tr>
<tr><td><code id="ungrid_+3A_grid">grid</code></td>
<td>
<p>a matrix holding the grid.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing three vectors: <code>x</code>, the grid x values,
<code>y</code>, the grid y values, and <code>grid</code>, the grid values.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(wind)
u &lt;- interpBarnes(wind$x, wind$y, wind$z)
contour(u$xg, u$yg, u$zg)
U &lt;- ungrid(u$xg, u$yg, u$zg)
points(U$x, U$y, col = oce.colorsViridis(100)[rescale(U$grid, rlow = 1, rhigh = 100)], pch = 20)
</code></pre>

<hr>
<h2 id='unitFromString'>Decode Units From Strings</h2><span id='topic+unitFromString'></span>

<h3>Description</h3>

<p>This is mainly intended for internal use within the package, e.g. by
<code><a href="#topic+read.odf">read.odf()</a></code>, and so the list of string-to-unit mappings is not
documented, since developers can learn it from simple examination
of the code.  The focus of <code>unitFromString()</code> is on strings that are
found in oceanographic files available to the author, <em>not</em> on all
possible units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unitFromString(unit, scale = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unitFromString_+3A_unit">unit</code></td>
<td>
<p>a character value indicating the unit. These
are matched according to rules developed to work with actual
data files, and so the list is not by any means exhaustive.</p>
</td></tr>
<tr><td><code id="unitFromString_+3A_scale">scale</code></td>
<td>
<p>a character value indicating the scale.  The default value
of <code>NULL</code> dictates that the scale is to be inferred from the unit. If
a non-<code>NULL</code> value is supplied, it will be used, even if it makes no sense
in relation to value of <code>unit</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="base.html#topic+list">list()</a></code> of two items: <code>unit</code> which is an
<code><a href="base.html#topic+expression">expression()</a></code>, and <code>scale</code>, which is a string.
</p>


<h3>See Also</h3>

<p>Other functions that interpret variable names and units from headers: 
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+unitFromStringRsk">unitFromStringRsk</a>()</code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>unitFromString("dbar") # dbar (no scale)
unitFromString("deg c") # modern temperature (ITS-90 scale)
</code></pre>

<hr>
<h2 id='unitFromStringRsk'>Infer rsk Units From a Vector of Strings</h2><span id='topic+unitFromStringRsk'></span>

<h3>Description</h3>

<p>This is used by <code><a href="#topic+read.rsk">read.rsk()</a></code> to infer the units of data, based
on strings stored in <code>.rsk</code> files. Lacking a definitive guide
to the format of these file, this function was based on visual inspection
of the data contained within a few sample files; unusual sensors may
not be handled properly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unitFromStringRsk(s)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unitFromStringRsk_+3A_s">s</code></td>
<td>
<p>Vector of character strings, holding the <code>units</code> entry in the
<code>channels</code> table of the <code>.rsk</code> database.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of unit lists.
</p>


<h3>See Also</h3>

<p>Other functions that interpret variable names and units from headers: 
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+unitFromString">unitFromString</a>()</code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>
</p>

<hr>
<h2 id='unwrapAngle'>Unwrap an Angle That Suffers Modulo-360 Problems</h2><span id='topic+unwrapAngle'></span>

<h3>Description</h3>

<p>This is mostly used for instrument heading angles, in cases where the
instrument is aligned nearly northward, so that small variations in heading
(e.g. due to mooring motion) can yield values that swing from small angles
to large angles, because of the modulo-360 cut point.
The method is to use the cosine and sine of the angle, to construct &quot;x&quot; and
&quot;y&quot; values on a unit circle, then to find means and medians of x and y
respectively, and finally to use <code><a href="base.html#topic+atan2">atan2()</a></code> to infer the angles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unwrapAngle(angle)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unwrapAngle_+3A_angle">angle</code></td>
<td>
<p>an angle (in degrees) that is thought be near 360 degrees, with
added noise</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two estimates: <code>mean</code> is based on an arithmetic
mean, and <code>median</code> is based on the median. Both are mapped to the range
0 to 360.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
true &lt;- 355
a &lt;- true + rnorm(100, sd = 10)
a &lt;- ifelse(a &gt; 360, a - 360, a)
a2 &lt;- unwrapAngle(a)
par(mar = c(3, 3, 5, 3))
hist(a, breaks = 360)
abline(v = a2$mean, col = "blue", lty = "dashed")
abline(v = true, col = "blue")
mtext("true (solid)\n estimate (dashed)", at = true, side = 3, col = "blue")
abline(v = mean(a), col = "red")
mtext("mean", at = mean(a), side = 3, col = "red")
</code></pre>

<hr>
<h2 id='useHeading'>Replace the Heading for One Instrument With That of Another</h2><span id='topic+useHeading'></span>

<h3>Description</h3>

<p>Replace the heading angles in one oce object with that from another,
possibly with a constant adjustment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>useHeading(b, g, add = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="useHeading_+3A_b">b</code></td>
<td>
<p>object holding data from an instrument whose heading is bad, but
whose other data are good.</p>
</td></tr>
<tr><td><code id="useHeading_+3A_g">g</code></td>
<td>
<p>object holding data from an instrument whose heading is good, and
should be interpolated to the time base of <code>b</code>.</p>
</td></tr>
<tr><td><code id="useHeading_+3A_add">add</code></td>
<td>
<p>an angle, in degrees, to be added to the heading.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A copy of <code>b</code>, but with <code>b$data$heading</code> replaced with
heading angles that result from linear interpolation of the headings in
<code>g</code>, and then adding the angle <code>add</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>

<hr>
<h2 id='usrLonLat'>Calculate Geographic Coordinates of Plot Box</h2><span id='topic+usrLonLat'></span>

<h3>Description</h3>

<p>Trace along the plot box, converting from xy coordinates to lonlat
coordinates. The results are used by <code><a href="#topic+mapGrid">mapGrid()</a></code>
and <code><a href="#topic+mapAxis">mapAxis()</a></code> to ignore out-of-frame grid
lines and axis labels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>usrLonLat(n = 25, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="usrLonLat_+3A_n">n</code></td>
<td>
<p>number of points to check along each side of the plot box</p>
</td></tr>
<tr><td><code id="usrLonLat_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Some projections, such as <code>"wintri"</code>, have trouble inverting
points that are &quot;off the globe&quot;.  In such cases,
the returned value has <code>lonmin</code>, <code>lonmax</code>, <code>latmin</code> and
<code>latmax</code> set to <code>NA</code>, and <code>ok</code> set to <code>FALSE</code>.
</p>


<h3>Value</h3>

<p>A list containing numerical
values <code>lonmin</code>, <code>lonmax</code>, <code>latmin</code>, and <code>latmax</code>, along
with logical value <code>ok</code>. The last of these indicates
whether at least one point on the plot box is invertible.
Note that longitudes are in the range from -180 to 180 degrees.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+utm2lonlat">utm2lonlat</a>()</code>
</p>

<hr>
<h2 id='utm2lonlat'>Convert UTM to Longitude and Latitude</h2><span id='topic+utm2lonlat'></span>

<h3>Description</h3>

<p>Convert UTM to Longitude and Latitude
</p>


<h3>Usage</h3>

<pre><code class='language-R'>utm2lonlat(easting, northing, zone = 1, hemisphere = "N", km = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="utm2lonlat_+3A_easting">easting</code></td>
<td>
<p>easting coordinate (in km or m, depending on value of
<code>km</code>).  Alternatively, a list containing items named <code>easting</code>,
<code>northing</code>, and <code>zone</code>, in which case these are taken from the
list and the arguments named <code>northing</code>, <code>zone</code> and are ignored.</p>
</td></tr>
<tr><td><code id="utm2lonlat_+3A_northing">northing</code></td>
<td>
<p>northing coordinate (in km or m, depending on value of
<code>km</code>).</p>
</td></tr>
<tr><td><code id="utm2lonlat_+3A_zone">zone</code></td>
<td>
<p>UTM zone</p>
</td></tr>
<tr><td><code id="utm2lonlat_+3A_hemisphere">hemisphere</code></td>
<td>
<p>indication of hemisphere; <code>"N"</code> for North, anything
else for South.</p>
</td></tr>
<tr><td><code id="utm2lonlat_+3A_km">km</code></td>
<td>
<p>logical value indicating whether <code>easting</code> and
<code>northing</code> are in kilometers or meters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing <code>longitude</code> and <code>latitude</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p><code style="white-space: pre;">&#8288;https://en.wikipedia.org/wiki/Universal_Transverse_Mercator_coordinate_system&#8288;</code>,
downloaded May 31, 2014.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lonlat2utm">lonlat2utm()</a></code> does the inverse operation.  For general
projections and their inverses, use <code><a href="#topic+lonlat2map">lonlat2map()</a></code> and
<code><a href="#topic+map2lonlat">map2lonlat()</a></code>.
</p>
<p>Other functions related to maps: 
<code><a href="#topic+formatPosition">formatPosition</a>()</code>,
<code><a href="#topic+lonlat2map">lonlat2map</a>()</code>,
<code><a href="#topic+lonlat2utm">lonlat2utm</a>()</code>,
<code><a href="#topic+map2lonlat">map2lonlat</a>()</code>,
<code><a href="#topic+mapArrows">mapArrows</a>()</code>,
<code><a href="#topic+mapAxis">mapAxis</a>()</code>,
<code><a href="#topic+mapContour">mapContour</a>()</code>,
<code><a href="#topic+mapCoordinateSystem">mapCoordinateSystem</a>()</code>,
<code><a href="#topic+mapDirectionField">mapDirectionField</a>()</code>,
<code><a href="#topic+mapGrid">mapGrid</a>()</code>,
<code><a href="#topic+mapImage">mapImage</a>()</code>,
<code><a href="#topic+mapLines">mapLines</a>()</code>,
<code><a href="#topic+mapLocator">mapLocator</a>()</code>,
<code><a href="#topic+mapLongitudeLatitudeXY">mapLongitudeLatitudeXY</a>()</code>,
<code><a href="#topic+mapPlot">mapPlot</a>()</code>,
<code><a href="#topic+mapPoints">mapPoints</a>()</code>,
<code><a href="#topic+mapPolygon">mapPolygon</a>()</code>,
<code><a href="#topic+mapScalebar">mapScalebar</a>()</code>,
<code><a href="#topic+mapText">mapText</a>()</code>,
<code><a href="#topic+mapTissot">mapTissot</a>()</code>,
<code><a href="#topic+oceCRS">oceCRS</a>()</code>,
<code><a href="#topic+shiftLongitude">shiftLongitude</a>()</code>,
<code><a href="#topic+usrLonLat">usrLonLat</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
# Cape Split, in the Minas Basin of the Bay of Fundy
utm2lonlat(852863, 5029997, 19)

</code></pre>

<hr>
<h2 id='vectorShow'>Show Some Values From a List, Vector or Matrix</h2><span id='topic+vectorShow'></span>

<h3>Description</h3>

<p>This is similar to <code><a href="utils.html#topic+str">str()</a></code>, but it shows data at the first and last of the
vector, which can be quite helpful in debugging.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vectorShow(
  v,
  msg = "",
  postscript = "",
  digits = 5L,
  n = 2L,
  showNA = FALSE,
  showNewline = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vectorShow_+3A_v">v</code></td>
<td>
<p>the item to be summarized.  If this is a list of a vector of named
items, then information is provided for each element. Otherwise, information
is provided for the first and last <code>n</code> values.</p>
</td></tr>
<tr><td><code id="vectorShow_+3A_msg">msg</code></td>
<td>
<p>optional character value indicating a message to show,
introducing the vector.  If not provided, then
a message is created from <code>v</code>. If <code>msg</code> is a non-empty string,
then that string is pasted together with a colon (unless <code>msg</code> already
contains a colon), before pasting a summary of data values.</p>
</td></tr>
<tr><td><code id="vectorShow_+3A_postscript">postscript</code></td>
<td>
<p>optional character value indicating an optional message
to append at the end of the return value.</p>
</td></tr>
<tr><td><code id="vectorShow_+3A_digits">digits</code></td>
<td>
<p>for numerical values of <code>v</code>, this is the number of digits
to use, in formatting the numbers with <code><a href="base.html#topic+format">format()</a></code>; otherwise,
<code>digits</code> is ignored.</p>
</td></tr>
<tr><td><code id="vectorShow_+3A_n">n</code></td>
<td>
<p>number of elements to show at start and end. If <code>n</code>
is negative, then all the elements are shown.</p>
</td></tr>
<tr><td><code id="vectorShow_+3A_showna">showNA</code></td>
<td>
<p>logical value indicating whether to show the number
of <code>NA</code> values. This is done only if the output contains ellipses,
meaning that some values are skipped, because if all values are shown,
it will be perfectly obvious whether there are any <code>NA</code> values.</p>
</td></tr>
<tr><td><code id="vectorShow_+3A_shownewline">showNewline</code></td>
<td>
<p>logical value indicating whether to put a newline
character at the end of the output string.  The default, TRUE, is
convenient for printing, but using FALSE makes more sense if
the result is to be used with, e.g. <code><a href="graphics.html#topic+mtext">mtext()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string ending in a newline character, suitable for
display with <code><a href="base.html#topic+cat">cat()</a></code> or <code><a href="#topic+oceDebug">oceDebug()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Examples</h3>

<pre><code class='language-R'># List
limits &lt;- list(low = 0, high = 1)
vectorShow(limits)

# Vector of named items
planktonCount &lt;- c(phytoplankton = 100, zooplankton = 20)
vectorShow(planktonCount)

# Vector
vectorShow(pi)

# Matrix
vectorShow(volcano)

# Other arguments
knot2mps &lt;- 0.5144444
vectorShow(knot2mps, postscript = "knots per m/s")
vectorShow("January", msg = "The first month is")

</code></pre>

<hr>
<h2 id='velocityStatistics'>Report Statistics of adp or adv Velocities</h2><span id='topic+velocityStatistics'></span>

<h3>Description</h3>

<p>Report statistics of ADP or ADV velocities, such as means and variance
ellipses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>velocityStatistics(x, control, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="velocityStatistics_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> or <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="velocityStatistics_+3A_control">control</code></td>
<td>
<p>An optional <a href="base.html#topic+list">list</a> used to specify more information.
This is presently ignored for <code>adv</code> objects.  For <code>adp</code> objects, if
<code>control$bin</code> is an integer, it is taken as the bin to be selected
(otherwise, an average across bins is used).</p>
</td></tr>
<tr><td><code id="velocityStatistics_+3A_...">...</code></td>
<td>
<p>additional arguments that are used in the call to
<code><a href="base.html#topic+mean">mean()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing items the major and minor axes of the covariance
ellipse (<code>ellipseMajor</code> and <code>ellipseMinor</code>), the angle of the
major axis anticlockwise of the horizontal axis (<code>ellipseAngle</code>), and
the x and y components of the mean velocity (<code>uMean</code> and <code>vMean</code>).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adp)
a &lt;- velocityStatistics(adp)
print(a)
t &lt;- seq(0, 2 * pi, length.out = 100)
theta &lt;- a$ellipseAngle * pi / 180
y &lt;- a$ellipseMajor * cos(t) * sin(theta) + a$ellipseMinor * sin(t) * cos(theta)
x &lt;- a$ellipseMajor * cos(t) * cos(theta) - a$ellipseMinor * sin(t) * sin(theta)
plot(adp, which = "uv+ellipse+arrow")
lines(x, y, col = "blue", lty = "dashed", lwd = 5)
arrows(0, 0, a$uMean, a$vMean, lwd = 5, length = 1 / 10, col = "blue", lty = "dashed")

</code></pre>

<hr>
<h2 id='webtide'>Get a Tidal Prediction From a WebTide Database</h2><span id='topic+webtide'></span>

<h3>Description</h3>

<p>Get a tidal prediction from a WebTide database. This only
works if the standalone WebTide application is installed,
and if it is installed in a standard location. The details
of installation are not within the oce purview.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>webtide(
  action = c("map", "predict"),
  longitude,
  latitude,
  node,
  time,
  basedir = getOption("webtide"),
  region = "nwatl",
  plot = TRUE,
  tformat,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="webtide_+3A_action">action</code></td>
<td>
<p>An indication of the action, either <code>action="map"</code> to
draw a map or <code>action="predict"</code> to get a prediction; see
&ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="webtide_+3A_longitude">longitude</code>, <code id="webtide_+3A_latitude">latitude</code></td>
<td>
<p>optional location at which prediction is required (ignored if
<code>node</code> is given).</p>
</td></tr>
<tr><td><code id="webtide_+3A_node">node</code></td>
<td>
<p>optional integer relating to a node in the database. If <code>node</code>
is given, then neither <code>latitude</code> nor <code>longitude</code> may be given.
If <code>node</code> is positive, then specifies indicates the node. If it is negative,
<code><a href="graphics.html#topic+locator">locator()</a></code> is called so that the user can click (once) on the map, after
which the node is displayed on the map.</p>
</td></tr>
<tr><td><code id="webtide_+3A_time">time</code></td>
<td>
<p>a vector of times (in the UTC timezone)
at which prediction is to be made.
If not supplied, this will be the week starting at the present time,
computed with <code><a href="#topic+presentTime">presentTime()</a></code>, with a 15 minute increment.</p>
</td></tr>
<tr><td><code id="webtide_+3A_basedir">basedir</code></td>
<td>
<p>directory containing the <code>WebTide</code> application.</p>
</td></tr>
<tr><td><code id="webtide_+3A_region">region</code></td>
<td>
<p>database region, given as a directory name in the WebTide
directory.  For example, <code>h3o</code> is for Halifax Harbour, <code>nwatl</code> is
for the northwest Atlantic, and <code>sshelf</code> is for the Scotian Shelf and
Gulf of Maine.</p>
</td></tr>
<tr><td><code id="webtide_+3A_plot">plot</code></td>
<td>
<p>boolean indicating whether to plot.</p>
</td></tr>
<tr><td><code id="webtide_+3A_tformat">tformat</code></td>
<td>
<p>optional argument passed to <code><a href="#topic+oce.plot.ts">oce.plot.ts()</a></code>, for
plot types that call that function.  (See <code><a href="base.html#topic+strptime">strptime()</a></code> for the
format used.)</p>
</td></tr>
<tr><td><code id="webtide_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
<tr><td><code id="webtide_+3A_...">...</code></td>
<td>
<p>optional arguments passed to plotting functions. A common
example is to set <code>xlim</code> and <code>ylim</code>, to focus a map region.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are two methods of using this function.
<em>Case 1:</em> <code>action="map"</code>. In this case, if
<code>plot</code> is <code>FALSE</code>, a list is returned, containing
all the <code>node</code>s in the selected database, along with all
the <code>latitude</code>s and <code>longitude</code>s. This value is
also returned (silently) if <code>plot</code> is true, but in that case,
a plot is drawn to indicate the node locations. If <code>latitude</code> and
<code>longitude</code> are given, then the node nearest that spot is indicated on
the map; otherwise, if <code>node</code> is given, then the location of that
node is indicated. There is also a special case: if <code>node</code> is negative
and <code>interactive()</code> is <code>TRUE</code>,
then <code><a href="graphics.html#topic+locator">locator()</a></code> is called, and the node nearest the spot
where the user clicks the mouse is indicated in the plot and in the
return value.
</p>
<p><em>Case 2:</em> <code>action="predict"</code>. If <code>plot</code> is <code>FALSE</code>,
then a list is returned, indicating <code>time</code>, predicted
<code>elevation</code>, velocity components <code>u</code> and <code>v</code>,
<code>node</code> number, the name of the <code>basedir</code>, and
the <code>region</code>. If <code>plot</code> is <code>TRUE</code>, this list is returned
silently, and time-series plots are drawn for elevation, u, and v.
</p>
<p>Naturally, <code>webtide</code> will not work unless WebTide has been installed on
the computer.
</p>


<h3>Value</h3>

<p>The value depends on <code>action</code>:
</p>

<ul>
<li><p> If <code>action="map"</code> the return value is a
list containing the index of the nearest node, along with the
<code>latitude</code> and <code>longitude</code> of that node.  If
<code>plot</code> is <code>FALSE</code>, this value is returned invisibly.
</p>
</li>
<li><p> If <code>action="predict"</code>, the return value is a list containing a vector
of times (<code>time</code>), as well as vectors of the predicted <code>elevation</code>
in metres and the predicted horizontal components of velocity, <code>u</code> and
<code>v</code>, along with the <code>node</code> number, and the <code>basedir</code> and
<code>region</code> as supplied to this function. If <code>plot</code> is <code>FALSE</code>,
this value is returned invisibly.
</p>
</li></ul>



<h3>Caution</h3>

<p>WebTide is not an open-source application, so the present function was
designed based on little more than guesses about the WebTide file structure.
Users should be on the lookout for odd results.
</p>


<h3>Sample of Usage</h3>

<pre>
# needs WebTide at the system level
library(oce)
# 1. prediction at Halifax NS
longitude &lt;- -63.57
latitude &lt;- 44.65
prediction &lt;- webtide("predict", longitude=longitude, latitude=latitude)
mtext(paste0("prediction at ", latitude, "N and ", longitude, "E"), line=0.75, side=3)
# 2. map
webtide(lon=-63.57,lat=44.65,xlim=c(-64,-63),ylim=c(43.0,46))
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>Source</h3>

<p>The WebTide software may be downloaded for free at the
Department of Fisheries and Oceans (Canada) website at
<code style="white-space: pre;">&#8288;http://www.bio.gc.ca/science/research-recherche/ocean/webtide/index-en.php&#8288;</code>
(checked February 2016 and May 2017).
</p>


<h3>See Also</h3>

<p>Other things related to tides: 
<code><a href="#topic++5B+5B+2Ctidem-method">[[,tidem-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Ctidem-method">[[&lt;-,tidem-method</a></code>,
<code><a href="#topic+as.tidem">as.tidem</a>()</code>,
<code><a href="#topic+plot+2Ctidem-method">plot,tidem-method</a></code>,
<code><a href="#topic+predict.tidem">predict.tidem</a>()</code>,
<code><a href="#topic+summary+2Ctidem-method">summary,tidem-method</a></code>,
<code><a href="#topic+tidalCurrent">tidalCurrent</a></code>,
<code><a href="#topic+tidedata">tidedata</a></code>,
<code><a href="#topic+tidem-class">tidem-class</a></code>,
<code><a href="#topic+tidemAstron">tidemAstron</a>()</code>,
<code><a href="#topic+tidemVuf">tidemVuf</a>()</code>,
<code><a href="#topic+tidem">tidem</a></code>
</p>

<hr>
<h2 id='wind'>Sample Wind Data</h2><span id='topic+wind'></span>

<h3>Description</h3>

<p>Wind data inferred from Figure 5 of Koch et al. (1983), provided to
illustrate the <code><a href="#topic+interpBarnes">interpBarnes()</a></code> function.
Columns <code>wind$x</code> and <code>wind$y</code> are location, while <code>wind$z</code> is
the wind speed, in m/s.
</p>


<h3>References</h3>

<p>S. E.  Koch and M.  DesJardins and P. J. Kocin, 1983.  &ldquo;An
interactive Barnes objective map analysis scheme for use with satellite and
conventional data,&rdquo; <em>J.  Climate Appl.  Met.</em>, vol 22, p. 1487-1503.
</p>


<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>

<hr>
<h2 id='window.oce'>Window an oce Object by Time or Distance</h2><span id='topic+window.oce'></span>

<h3>Description</h3>

<p>Windows <code>x</code> on either time or distance, depending on the value of
<code>which</code>.  In each case, values of <code>start</code> and <code>end</code> may be
integers, to indicate a portion of the time or distance range.  If
<code>which</code> is <code>"time"</code>, then the <code>start</code> and <code>end</code> values
may also be provided as POSIX times, or character strings indicating times
(in time zone given by the value of <code>getOption("oceTz")</code>).
Note that <code><a href="base.html#topic+subset">subset()</a></code> may be more useful than this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'oce'
window(
  x,
  start = NULL,
  end = NULL,
  frequency = NULL,
  deltat = NULL,
  extend = FALSE,
  which = c("time", "distance"),
  indexReturn = FALSE,
  debug = getOption("oceDebug"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="window.oce_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+oce-class">oce</a> object.</p>
</td></tr>
<tr><td><code id="window.oce_+3A_start">start</code></td>
<td>
<p>the start time (or distance) of the time (or space) region of
interest.  This may be a single value or a vector.</p>
</td></tr>
<tr><td><code id="window.oce_+3A_end">end</code></td>
<td>
<p>the end time (or distance) of the time (or space) region of
interest.  This may be a single value or a vector.</p>
</td></tr>
<tr><td><code id="window.oce_+3A_frequency">frequency</code></td>
<td>
<p>not permitted yet.</p>
</td></tr>
<tr><td><code id="window.oce_+3A_deltat">deltat</code></td>
<td>
<p>not permitted yet</p>
</td></tr>
<tr><td><code id="window.oce_+3A_extend">extend</code></td>
<td>
<p>not permitted yet</p>
</td></tr>
<tr><td><code id="window.oce_+3A_which">which</code></td>
<td>
<p>string containing the name of the quantity on which sampling is
done.  Possibilities are <code>"time"</code>, which applies the windowing on the
<code>time</code> entry of the <code>data</code> slot, and <code>"distance"</code>, for the
<code>distance</code> entry (for those objects, such as <code>adp</code>, that have this
entry).</p>
</td></tr>
<tr><td><code id="window.oce_+3A_indexreturn">indexReturn</code></td>
<td>
<p>boolean flag indicating whether to return a list of the
&quot;kept&quot; indices for the <code>time</code> entry of the <code>data</code> slot, as well as
the <code>timeSlow</code> entry, if there is one..  Either of these lists will be
<code>NULL</code>, if the object lacks the relevant items.</p>
</td></tr>
<tr><td><code id="window.oce_+3A_debug">debug</code></td>
<td>
<p>a flag that turns on debugging.</p>
</td></tr>
<tr><td><code id="window.oce_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Normally, this is new <code>oce</code> object.  However, if
<code>indexReturn=TRUE</code>, the return value is two-element list containing
items named <code>index</code> and <code>indexSlow</code>, which are the indices for the
<code>time</code> entry of the <code>data</code> slot (and the <code>timeSlow</code>, if it
exists).
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+subset">subset()</a></code> provides more flexible selection of subsets.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(adp)
plot(adp)
early &lt;- window(adp, start = "2008-06-26 00:00:00", end = "2008-06-26 12:00:00")
plot(early)
bottom &lt;- window(adp, start = 0, end = 20, which = "distance")
plot(bottom)
</code></pre>

<hr>
<h2 id='windrose-class'>Class to Store windrose Data</h2><span id='topic+windrose-class'></span>

<h3>Description</h3>

<p>This class stores <code>windrose</code> objects, which
store statistical information about winds, mainly for
plotting as &quot;wind rose&quot; plots with <code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method()</a></code>.
Unlike most other <a href="#topic+oce-class">oce</a> objects, there is no reading
method for <code>windrose</code> objects, because there is no standard way to store
wind data in files; instead, <code><a href="#topic+as.windrose">as.windrose()</a></code> is provided
to construct <code>windrose</code> objects.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>windrose</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>windrose</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>windrose</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+windrose-class">windrose</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+windrose-class">windrose</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>See Also</h3>

<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>
</p>
<p>Other things related to windrose data: 
<code><a href="#topic++5B+5B+2Cwindrose-method">[[,windrose-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cwindrose-method">[[&lt;-,windrose-method</a></code>,
<code><a href="#topic+as.windrose">as.windrose</a>()</code>,
<code><a href="#topic+plot+2Cwindrose-method">plot,windrose-method</a></code>,
<code><a href="#topic+summary+2Cwindrose-method">summary,windrose-method</a></code>
</p>

<hr>
<h2 id='woceNames2oceNames'>Translate WOCE Data Names to Oce Data Names</h2><span id='topic+woceNames2oceNames'></span>

<h3>Description</h3>

<p>Translate WOCE-style names to <code>oce</code> names, using <code><a href="base.html#topic+gsub">gsub()</a></code>
to match patterns. For example, the pattern <code>"CTDOXY.*"</code> is taken
to mean <code>oxygen</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>woceNames2oceNames(names)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="woceNames2oceNames_+3A_names">names</code></td>
<td>
<p>vector of strings holding WOCE-style names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of strings holding <code>oce</code>-style names.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>Several online sources list WOCE names. An example is
<code style="white-space: pre;">&#8288;https://cchdo.github.io/hdo-assets/documentation/manuals/pdf/90_1/chap4.pdf&#8288;</code>
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that interpret variable names and units from headers: 
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+unitFromStringRsk">unitFromStringRsk</a>()</code>,
<code><a href="#topic+unitFromString">unitFromString</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>
</p>

<hr>
<h2 id='woceUnit2oceUnit'>Translate WOCE Units to oce Units</h2><span id='topic+woceUnit2oceUnit'></span>

<h3>Description</h3>

<p>Translate WOCE-style units to oce units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>woceUnit2oceUnit(woceUnit)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="woceUnit2oceUnit_+3A_woceunit">woceUnit</code></td>
<td>
<p>string holding a WOCE unit</p>
</td></tr>
</table>


<h3>Value</h3>

<p>expression in oce unit form
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>See Also</h3>

<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+write.ctd">write.ctd</a>()</code>
</p>
<p>Other functions that interpret variable names and units from headers: 
<code><a href="#topic+ODFNames2oceNames">ODFNames2oceNames</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+unitFromStringRsk">unitFromStringRsk</a>()</code>,
<code><a href="#topic+unitFromString">unitFromString</a>()</code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>
</p>

<hr>
<h2 id='write.ctd'>Save a ctd Object in a CSV File</h2><span id='topic+write.ctd'></span>

<h3>Description</h3>

<p>Writes a comma-separated file containing the data frame stored in
the <code>data</code> slot of the first argument.  The file is suitable
for reading with a spreadsheet, or
with <code><a href="utils.html#topic+read.csv">read.csv()</a></code>.  This output file will contain
some of the metadata in <code>x</code>, if <code>metadata</code> is <code>TRUE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.ctd(object, file, metadata = TRUE, flags = TRUE, format = "csv")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.ctd_+3A_object">object</code></td>
<td>
<p>a <a href="#topic+ctd-class">ctd</a> object.</p>
</td></tr>
<tr><td><code id="write.ctd_+3A_file">file</code></td>
<td>
<p>Either a character string (the file name) or a connection. If not
provided, <code>file</code> defaults to <code><a href="base.html#topic+stdout">stdout()</a></code>.</p>
</td></tr>
<tr><td><code id="write.ctd_+3A_metadata">metadata</code></td>
<td>
<p>a logical value indicating whether to put some selected
metadata elements at the start of the output file.</p>
</td></tr>
<tr><td><code id="write.ctd_+3A_flags">flags</code></td>
<td>
<p>a logical value indicating whether to show data-quality flags
as well as data.</p>
</td></tr>
<tr><td><code id="write.ctd_+3A_format">format</code></td>
<td>
<p>string indicating the format to use. This may be <code>"csv"</code>
for a simple CSV format, or <code>"whp"</code> for the World Hydrographic
Program format, described in reference 1 and exemplified in reference 2.</p>
</td></tr>
</table>


<h3>Sample of Usage</h3>

<pre>
library(oce)
data(ctd)
write.ctd(ctd, "ctd.csv")
d &lt;- read.csv("ctd.csv")
plot(as.ctd(d$salinity, d$temperature, d$pressure))
</pre>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>

<p>The following links used to work, but failed as of December 2020.
</p>

<ol>
<li> <p><code style="white-space: pre;">&#8288;https://www.nodc.noaa.gov/woce/woce_v3/wocedata_1/whp/exchange/exchange_format_desc.htm&#8288;</code>
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;https://www.nodc.noaa.gov/woce/woce_v3/wocedata_1/whp/exchange/example_ct1.csv&#8288;</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>The documentation for <a href="#topic+ctd-class">ctd</a> explains the structure
of CTD objects.
</p>
<p>Other things related to ctd data: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic++5B+5B+2Cctd-method">[[,ctd-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cctd-method">[[&lt;-,ctd-method</a></code>,
<code><a href="#topic+as.ctd">as.ctd</a>()</code>,
<code><a href="#topic+cnvName2oceName">cnvName2oceName</a>()</code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctdDecimate">ctdDecimate</a>()</code>,
<code><a href="#topic+ctdFindProfilesRBR">ctdFindProfilesRBR</a>()</code>,
<code><a href="#topic+ctdFindProfiles">ctdFindProfiles</a>()</code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctdRepair">ctdRepair</a>()</code>,
<code><a href="#topic+ctdTrim">ctdTrim</a>()</code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>,
<code><a href="#topic+handleFlags+2Cctd-method">handleFlags,ctd-method</a></code>,
<code><a href="#topic+initialize+2Cctd-method">initialize,ctd-method</a></code>,
<code><a href="#topic+initializeFlagScheme+2Cctd-method">initializeFlagScheme,ctd-method</a></code>,
<code><a href="#topic+oceNames2whpNames">oceNames2whpNames</a>()</code>,
<code><a href="#topic+oceUnits2whpUnits">oceUnits2whpUnits</a>()</code>,
<code><a href="#topic+plot+2Cctd-method">plot,ctd-method</a></code>,
<code><a href="#topic+plotProfile">plotProfile</a>()</code>,
<code><a href="#topic+plotScan">plotScan</a>()</code>,
<code><a href="#topic+plotTS">plotTS</a>()</code>,
<code><a href="#topic+read.ctd.aml">read.ctd.aml</a>()</code>,
<code><a href="#topic+read.ctd.itp">read.ctd.itp</a>()</code>,
<code><a href="#topic+read.ctd.odf">read.ctd.odf</a>()</code>,
<code><a href="#topic+read.ctd.odv">read.ctd.odv</a>()</code>,
<code><a href="#topic+read.ctd.saiv">read.ctd.saiv</a>()</code>,
<code><a href="#topic+read.ctd.sbe">read.ctd.sbe</a>()</code>,
<code><a href="#topic+read.ctd.ssda">read.ctd.ssda</a>()</code>,
<code><a href="#topic+read.ctd.woce.other">read.ctd.woce.other</a>()</code>,
<code><a href="#topic+read.ctd.woce">read.ctd.woce</a>()</code>,
<code><a href="#topic+read.ctd">read.ctd</a>()</code>,
<code><a href="#topic+setFlags+2Cctd-method">setFlags,ctd-method</a></code>,
<code><a href="#topic+subset+2Cctd-method">subset,ctd-method</a></code>,
<code><a href="#topic+summary+2Cctd-method">summary,ctd-method</a></code>,
<code><a href="#topic+woceNames2oceNames">woceNames2oceNames</a>()</code>,
<code><a href="#topic+woceUnit2oceUnit">woceUnit2oceUnit</a>()</code>
</p>

<hr>
<h2 id='xbt'>Sample xbt Data</h2><span id='topic+xbt'></span>

<h3>Description</h3>

<p>An <a href="#topic+xbt-class">xbt</a> object created by using <code><a href="#topic+read.xbt">read.xbt()</a></code> on a Sippican file created by extracting the near-surface
fraction of the sample provided in Section 5.5.6 of reference 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(xbt)
</code></pre>


<h3>References</h3>


<ol>
<li><p> Sippican, Inc. &quot;Bathythermograph Data Acquisition System: Installation, Operation and Maintenance
Manual (P/N 308195, Rev. A),&quot; 2003.
https://pages.uoregon.edu/drt/MGL0910_Science_Report/attachments/MK21_ISA_Manual_Rev_A.pdf.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other datasets provided with oce: 
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+amsr">amsr</a></code>,
<code><a href="#topic+argo">argo</a></code>,
<code><a href="#topic+cm">cm</a></code>,
<code><a href="#topic+coastlineWorld">coastlineWorld</a></code>,
<code><a href="#topic+ctdRaw">ctdRaw</a></code>,
<code><a href="#topic+ctd">ctd</a></code>,
<code><a href="#topic+echosounder">echosounder</a></code>,
<code><a href="#topic+landsat">landsat</a></code>,
<code><a href="#topic+lisst">lisst</a></code>,
<code><a href="#topic+lobo">lobo</a></code>,
<code><a href="#topic+met">met</a></code>,
<code><a href="#topic+ocecolors">ocecolors</a></code>,
<code><a href="#topic+rsk">rsk</a></code>,
<code><a href="#topic+sealevelTuktoyaktuk">sealevelTuktoyaktuk</a></code>,
<code><a href="#topic+sealevel">sealevel</a></code>,
<code><a href="#topic+section">section</a></code>,
<code><a href="#topic+topoWorld">topoWorld</a></code>,
<code><a href="#topic+wind">wind</a></code>
</p>
<p>Other things related to xbt data: 
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>,
<code><a href="#topic+as.xbt">as.xbt</a>()</code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1</a>()</code>,
<code><a href="#topic+read.xbt">read.xbt</a>()</code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>,
<code><a href="#topic+summary+2Cxbt-method">summary,xbt-method</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(oce)
data(xbt)
summary(xbt)
plot(xbt)

</code></pre>

<hr>
<h2 id='xbt-class'>Class to Store XBT (Expendable Bathythermograph) Data</h2><span id='topic+xbt-class'></span>

<h3>Description</h3>

<p>This class stores expendable bathythermograph (XBT) data, e.g. from a Sippican
device.  Reference 1 gives some information on Sippican
devices, and reference 2 is a useful introduction to the
modern literature on XBTs in general.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>As with all <code>oce</code> objects, the <code>data</code> slot
for <code>xbt</code> objects is a
<a href="base.html#topic+list">list</a> containing the main data for the object.
The key items stored in this slot are <code>depth</code> (or <code>z</code>) and <code>temperature</code>, although some datasets also have <code>soundSpeed</code>.  Note that <code>depth</code> and <code>z</code> are inferred from time in water, using an empirical formula for instrument descent rate, and that <code>soundSpeed</code> is #' calculated using a fixed  practical salinity of 35. Note that the <code>[[</code> accessor will compute any of <code>depth</code>, <code>z</code> or <code>pressure</code>, based on whatever is in the data object.  Similarly, <code>soundspeed</code> will compute sound speed (assuming a practical salinity of 35), if that that item is present in the <code>data</code> slot.</p>
</dd>
<dt><code>metadata</code></dt><dd><p>As with all <code>oce</code> objects, the <code>metadata</code> slot
for <code>xbt</code> objects is a <a href="base.html#topic+list">list</a> containing
information about the <code>data</code> or about the object itself.
</p>
</dd>
<dt><code>processingLog</code></dt><dd><p>As with all <code>oce</code> objects, the <code>processingLog</code> slot
for <code>xbt</code> objects is a
<a href="base.html#topic+list">list</a> with entries describing the creation and evolution
of the object. The contents are updated by various <code>oce</code> functions to
keep a record of processing steps.  Object summaries and
<code><a href="#topic+processingLogShow">processingLogShow()</a></code> both display the log.</p>
</dd>
</dl>


<h3>Modifying slot contents</h3>

<p>Although the <code>[[&lt;-</code> operator may permit modification of the contents
of <a href="#topic+xbt-class">xbt</a> objects (see <code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>),
it is better to use <code><a href="#topic+oceSetData">oceSetData()</a></code> and <code><a href="#topic+oceSetMetadata">oceSetMetadata()</a></code>,
because those functions save an entry in the <code>processingLog</code>
that describes the change.
</p>


<h3>Retrieving slot contents</h3>

<p>The full contents of the <code>data</code> and <code>metadata</code> slots of a <a href="#topic+xbt-class">xbt</a>
object may be retrieved in the standard R way using <code><a href="methods.html#topic+slot">slot()</a></code>. For
example <code>slot(o,"data")</code> returns the <code>data</code> slot of an object named <code>o</code>,
and similarly <code>slot(o,"metadata")</code> returns
the <code>metadata</code> slot.
</p>
<p>The slots may also be obtained with the <code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>
operator, as e.g. <code>o[["data"]]</code> and <code>o[["metadata"]]</code>, respectively.
</p>
<p>The <code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code> operator can also
be used to retrieve items from within the <code>data</code> and <code>metadata</code> slots.
For example, <code>o[["temperature"]]</code> can be used to retrieve temperature
from an object containing that quantity. The rule is that a named
quantity is sought first within the object's <code>metadata</code> slot,
with the <code>data</code> slot being checked only if <code>metadata</code> does not
contain the item. This <code>[[</code> method can also be used to get
certain derived quantities, if the object contains sufficient
information to calculate them. For example, an object that holds
(practical) salinity, temperature and pressure, along with
longitude and latitude, has sufficient information to compute
Absolute Salinity, and so <code>o[["SA"]]</code> will yield the
calculated Absolute Salinity.
</p>
<p>It is also possible to find items more directly, using <code><a href="#topic+oceGetData">oceGetData()</a></code> and
<code><a href="#topic+oceGetMetadata">oceGetMetadata()</a></code>, but neither of these functions can
retrieve derived items.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Sippican, Inc. &quot;Bathythermograph Data Acquisition System: Installation, Operation and Maintenance
Manual (P/N 308195, Rev. A),&quot; 2003.
https://pages.uoregon.edu/drt/MGL0910_Science_Report/attachments/MK21_ISA_Manual_Rev_A.pdf.
</p>
</li>
<li><p> Cheng, Lijing, John Abraham, Gustavo Goni, Timothy Boyer, Susan Wijffels, Rebecca
Cowley, Viktor Gouretski, et al. &quot;XBT Science: Assessment of Instrumental Biases and Errors.&quot;
Bulletin of the American Meteorological Society 97, no. 6 (June 2016): 924-33.
<code>10.1175/BAMS-D-15-00031.1</code>
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to xbt data: 
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>,
<code><a href="#topic+as.xbt">as.xbt</a>()</code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1</a>()</code>,
<code><a href="#topic+read.xbt">read.xbt</a>()</code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>,
<code><a href="#topic+summary+2Cxbt-method">summary,xbt-method</a></code>,
<code><a href="#topic+xbt.edf">xbt.edf</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>
<p>Other classes provided by oce: 
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+argo-class">argo-class</a></code>,
<code><a href="#topic+bremen-class">bremen-class</a></code>,
<code><a href="#topic+cm-class">cm-class</a></code>,
<code><a href="#topic+coastline-class">coastline-class</a></code>,
<code><a href="#topic+ctd-class">ctd-class</a></code>,
<code><a href="#topic+lisst-class">lisst-class</a></code>,
<code><a href="#topic+lobo-class">lobo-class</a></code>,
<code><a href="#topic+met-class">met-class</a></code>,
<code><a href="#topic+oce-class">oce-class</a></code>,
<code><a href="#topic+odf-class">odf-class</a></code>,
<code><a href="#topic+rsk-class">rsk-class</a></code>,
<code><a href="#topic+sealevel-class">sealevel-class</a></code>,
<code><a href="#topic+section-class">section-class</a></code>,
<code><a href="#topic+topo-class">topo-class</a></code>,
<code><a href="#topic+windrose-class">windrose-class</a></code>
</p>

<hr>
<h2 id='xbt.edf'>Sample xbt File in .edf Format</h2><span id='topic+xbt.edf'></span>

<h3>Description</h3>

<p>Sample xbt File in .edf Format
</p>


<h3>See Also</h3>

<p>Other raw datasets: 
<code><a href="#topic+CTD_BCD2014666_008_1_DN.ODF.gz">CTD_BCD2014666_008_1_DN.ODF.gz</a></code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+ctd.cnv.gz">ctd.cnv.gz</a></code>,
<code><a href="#topic+ctd_aml.csv.gz">ctd_aml.csv.gz</a></code>,
<code><a href="#topic+d200321-001.ctd.gz">d200321-001.ctd.gz</a></code>,
<code><a href="#topic+d201211_0011.cnv.gz">d201211_0011.cnv.gz</a></code>
</p>
<p>Other things related to xbt data: 
<code><a href="#topic++5B+5B+2Cxbt-method">[[,xbt-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cxbt-method">[[&lt;-,xbt-method</a></code>,
<code><a href="#topic+as.xbt">as.xbt</a>()</code>,
<code><a href="#topic+plot+2Cxbt-method">plot,xbt-method</a></code>,
<code><a href="#topic+read.xbt.noaa1">read.xbt.noaa1</a>()</code>,
<code><a href="#topic+read.xbt">read.xbt</a>()</code>,
<code><a href="#topic+subset+2Cxbt-method">subset,xbt-method</a></code>,
<code><a href="#topic+summary+2Cxbt-method">summary,xbt-method</a></code>,
<code><a href="#topic+xbt-class">xbt-class</a></code>,
<code><a href="#topic+xbt">xbt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xbt &lt;- read.oce(system.file("extdata", "xbt.edf", package="oce"))

</code></pre>

<hr>
<h2 id='xyzToEnu'>Convert Acoustic-Doppler Data From XYZ to ENU Coordinates</h2><span id='topic+xyzToEnu'></span>

<h3>Description</h3>

<p>Convert Acoustic-Doppler Data From XYZ to ENU Coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xyzToEnu(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xyzToEnu_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> or <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="xyzToEnu_+3A_...">...</code></td>
<td>
<p>extra arguments that are passed on to <code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp()</a></code>
or <code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv()</a></code>; see the documentation for those functions,
for the details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same class as <code>x</code>, but with velocities
in east-north-up coordinates instead of xyz coordinates.
</p>


<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdv">xyzToEnuAdv</a>()</code>
</p>

<hr>
<h2 id='xyzToEnuAdp'>Convert adp Object From XYZ to ENU Coordinates</h2><span id='topic+xyzToEnuAdp'></span>

<h3>Description</h3>

<p>Convert ADP velocity components from a xyz-based coordinate system to
an enu-based coordinate system, by using the instrument's recording of
information relating to heading, pitch, and roll. The action is based
on what is stored in the data, and so it depends greatly on instrument type
and the style of original data format. This function handles data from
RDI Teledyne, Sontek, and some Nortek instruments directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xyzToEnuAdp(x, declination = 0, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xyzToEnuAdp_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object.</p>
</td></tr>
<tr><td><code id="xyzToEnuAdp_+3A_declination">declination</code></td>
<td>
<p>magnetic declination to be added to the heading after
&quot;righting&quot; (see below), to get ENU with N as &quot;true&quot; north.  If this
is set to NULL, then the returned object is set up without adjusting
the compass for declination.  That means that <code>north</code> in its <code>metadata</code>
slot will be set to <code>"magnetic"</code>, and also that there will be no item
named <code>declination</code> in that slot.  Note that <code><a href="#topic+applyMagneticDeclination">applyMagneticDeclination()</a></code>
can be used later, to set a declination.</p>
</td></tr>
<tr><td><code id="xyzToEnuAdp_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The first step is to convert the (x,y,z) velocity components (stored in the
three columns of <code>x[["v"]][,,1:3]</code>) into what RDI (reference 1, pages 11 and 12)
calls &quot;ship&quot; (or &quot;righted&quot;) components.  For example, the z coordinate,
which may point upwards or downwards depending on instrument orientation, is
mapped onto a &quot;mast&quot; coordinate that points more nearly upwards than
downward.  The other ship coordinates are called &quot;starboard&quot; and &quot;forward&quot;,
the meanings of which will be clear to mariners.  Once the (x,y,z)
velocities are converted to ship velocities, the orientation of the
instrument is extracted from heading, pitch, and roll vectors stored in the
object.  These angles are defined differently for RDI and Sontek profilers.
</p>
<p>The code handles every case individually, based on the table given below.
The table comes from Clark Richards, a former PhD student at Dalhousie
University (reference 2), who developed it based on instrument documentation,
discussion on user groups, and analysis of measurements acquired with RDI
and Sontek acoustic current profilers in the SLEIWEX experiment.  In the
table, (X, Y, Z) denote instrument-coordinate velocities, (S, F, M) denote
ship-coordinate velocities, and (H, P, R) denote heading, pitch, and roll.
</p>

<table>
<tr>
 <td style="text-align: right;"> <strong>Case</strong> </td><td style="text-align: right;"> <strong>Mfr.</strong> </td><td style="text-align: right;">
<strong>Instr.</strong> <strong>Orient.</strong> </td><td style="text-align: right;"> <strong>H</strong> </td><td style="text-align: right;"> <strong>P</strong> </td><td style="text-align: right;">
<strong>R</strong> </td><td style="text-align: right;"> <strong>S</strong> </td><td style="text-align: right;"> <strong>F</strong> </td><td style="text-align: right;"> <strong>M</strong></td>
</tr>
<tr>
 <td style="text-align: right;"> 1 </td><td style="text-align: right;"> RDI
</td><td style="text-align: right;"> ADCP </td><td style="text-align: right;"> up </td><td style="text-align: right;"> H </td><td style="text-align: right;"> arctan(tan(P)*cos(R)) </td><td style="text-align: right;"> R </td><td style="text-align: right;"> -X </td><td style="text-align: right;"> Y
</td><td style="text-align: right;"> -Z</td>
</tr>
<tr>
 <td style="text-align: right;"> 2 </td><td style="text-align: right;"> RDI </td><td style="text-align: right;"> ADCP </td><td style="text-align: right;"> down </td><td style="text-align: right;"> H </td><td style="text-align: right;"> arctan(tan(P)*cos(R))
</td><td style="text-align: right;"> -R </td><td style="text-align: right;"> X </td><td style="text-align: right;"> Y </td><td style="text-align: right;"> Z</td>
</tr>
<tr>
 <td style="text-align: right;"> 3 </td><td style="text-align: right;"> Nortek </td><td style="text-align: right;"> ADP </td><td style="text-align: right;"> up </td><td style="text-align: right;"> H-90
</td><td style="text-align: right;"> R </td><td style="text-align: right;"> -P </td><td style="text-align: right;"> X </td><td style="text-align: right;"> Y </td><td style="text-align: right;"> Z</td>
</tr>
<tr>
 <td style="text-align: right;"> 4 </td><td style="text-align: right;"> Nortek </td><td style="text-align: right;"> ADP </td><td style="text-align: right;"> down </td><td style="text-align: right;">
H-90 </td><td style="text-align: right;"> R </td><td style="text-align: right;"> -P </td><td style="text-align: right;"> X </td><td style="text-align: right;"> -Y </td><td style="text-align: right;"> -Z</td>
</tr>
<tr>
 <td style="text-align: right;"> 5 </td><td style="text-align: right;"> Sontek </td><td style="text-align: right;"> ADP </td><td style="text-align: right;"> up
</td><td style="text-align: right;"> H-90 </td><td style="text-align: right;"> -P </td><td style="text-align: right;"> -R </td><td style="text-align: right;"> X </td><td style="text-align: right;"> Y </td><td style="text-align: right;"> Z</td>
</tr>
<tr>
 <td style="text-align: right;"> 6 </td><td style="text-align: right;"> Sontek </td><td style="text-align: right;"> ADP
</td><td style="text-align: right;"> down </td><td style="text-align: right;"> H-90 </td><td style="text-align: right;"> -P </td><td style="text-align: right;"> -R </td><td style="text-align: right;"> X </td><td style="text-align: right;"> Y </td><td style="text-align: right;"> Z</td>
</tr>
<tr>
 <td style="text-align: right;"> 7 </td><td style="text-align: right;"> Sontek
</td><td style="text-align: right;"> PCADP </td><td style="text-align: right;"> up </td><td style="text-align: right;"> H-90 </td><td style="text-align: right;"> R </td><td style="text-align: right;"> -P </td><td style="text-align: right;"> X </td><td style="text-align: right;"> Y </td><td style="text-align: right;"> Z</td>
</tr>
<tr>
 <td style="text-align: right;"> 8 </td><td style="text-align: right;">
Sontek </td><td style="text-align: right;"> PCADP </td><td style="text-align: right;"> down </td><td style="text-align: right;"> H-90 </td><td style="text-align: right;"> R </td><td style="text-align: right;"> -P </td><td style="text-align: right;"> X </td><td style="text-align: right;"> Y </td><td style="text-align: right;"> Z</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>Finally, a standardized rotation matrix is used to convert from ship
coordinates to earth coordinates.  As described in the RDI coordinate
transformation manual (reference 1, pages 13 and 14), this matrix is based on sines
and cosines of heading, pitch, and roll If <code>CH</code> and <code>SH</code> denote
cosine and sine of heading (after adjusting for declination), with similar
terms for pitch and roll using second letters <code>P</code> and <code>R</code>, the
rotation matrix is
</p>
<pre> rbind(c( CH*CR + SH*SP*SR, SH*CP, CH*SR - SH*SP*CR), c(-SH*CR
+ CH*SP*SR, CH*CP, -SH*SR - CH*SP*CR), c( -CP*SR, SP, CP*CR)) </pre>
<p>This matrix is left-multiplied by a matrix with three rows, the top a vector
of &quot;starboard&quot; values, the middle a vector of &quot;forward&quot; values, and the
bottom a vector of &quot;mast&quot; values.  Finally, the columns of
<code>data$v[,,1:3]</code> are filled in with the result of the matrix
multiplication.
</p>


<h3>Value</h3>

<p>An object with <code>data$v[,,1:3]</code> altered appropriately, and
<code>x[["oceCoordinate"]]</code> changed from <code>xyz</code> to <code>enu</code>.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley and Clark Richards
</p>


<h3>References</h3>


<ol>
<li><p> Teledyne RD Instruments. &ldquo;ADCP Coordinate Transformation: Formulas and Calculations,&rdquo;
January 2010. P/N 951-6079-00.
</p>
</li>
<li><p> Clark Richards, 2012, PhD Dalhousie University Department of
Oceanography.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdpAD2CP">xyzToEnuAdpAD2CP</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='xyzToEnuAdpAD2CP'>Convert adp Object of AD2CP type From XYZ to ENU Coordinates</h2><span id='topic+xyzToEnuAdpAD2CP'></span>

<h3>Description</h3>

<p>This function is in active development,
and both the methodology and user interface may change
without notice. Only developers (or invitees) should be trying to
use this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xyzToEnuAdpAD2CP(x, declination = 0, debug = getOption("oceDebug"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xyzToEnuAdpAD2CP_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adp-class">adp</a> object created by <code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp()</a></code>.</p>
</td></tr>
<tr><td><code id="xyzToEnuAdpAD2CP_+3A_declination">declination</code></td>
<td>
<p>IGNORED at present, but will be used at some later time.</p>
</td></tr>
<tr><td><code id="xyzToEnuAdpAD2CP_+3A_debug">debug</code></td>
<td>
<p>an integer specifying whether debugging information is
to be printed during the processing. This is a general parameter that
is used by many <code>oce</code> functions. Generally, setting <code>debug=0</code>
turns off the printing, while higher values suggest that more information
be printed. If one function calls another, it usually reduces the value of
<code>debug</code> first, so that a user can often obtain deeper debugging
by specifying higher <code>debug</code> values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with <code>data$v[,,1:3]</code> altered appropriately, and
<code>x[["oceCoordinate"]]</code> changed from <code>xyz</code> to <code>enu</code>.
</p>


<h3>Limitations</h3>

<p>This only works if the instrument orientation is <code>"AHRS"</code>, and even
that is not tested yet. Plus, as noted, the declination is ignored.
</p>


<h3>Author(s)</h3>

<p>Dan Kelley
</p>


<h3>References</h3>


<ol>
<li><p> Nortek AS. &ldquo;Signature Integration 55|250|500|1000kHz.&rdquo; Nortek AS, 2017.
</p>
</li>
<li><p> Nortek AS. &ldquo;Signature Integration 55|250|500|1000kHz.&rdquo; Nortek AS, 2018.
https://www.nortekgroup.com/assets/software/N3015-007-Integrators-Guide-AD2CP_1018.pdf.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other things related to adp data: 
<code><a href="#topic++5B+5B+2Cadp-method">[[,adp-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadp-method">[[&lt;-,adp-method</a></code>,
<code><a href="#topic+ad2cpCodeToName">ad2cpCodeToName</a>()</code>,
<code><a href="#topic+ad2cpHeaderValue">ad2cpHeaderValue</a>()</code>,
<code><a href="#topic+adp-class">adp-class</a></code>,
<code><a href="#topic+adpAd2cpFileTrim">adpAd2cpFileTrim</a>()</code>,
<code><a href="#topic+adpConvertRawToNumeric">adpConvertRawToNumeric</a>()</code>,
<code><a href="#topic+adpEnsembleAverage">adpEnsembleAverage</a>()</code>,
<code><a href="#topic+adpFlagPastBoundary">adpFlagPastBoundary</a>()</code>,
<code><a href="#topic+adpRdiFileTrim">adpRdiFileTrim</a>()</code>,
<code><a href="#topic+adp_rdi.000">adp_rdi.000</a></code>,
<code><a href="#topic+adp">adp</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadp-method">applyMagneticDeclination,adp-method</a></code>,
<code><a href="#topic+as.adp">as.adp</a>()</code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyzAdpAD2CP">beamToXyzAdpAD2CP</a>()</code>,
<code><a href="#topic+beamToXyzAdp">beamToXyzAdp</a>()</code>,
<code><a href="#topic+beamToXyzAdv">beamToXyzAdv</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+beamUnspreadAdp">beamUnspreadAdp</a>()</code>,
<code><a href="#topic+binmapAdp">binmapAdp</a>()</code>,
<code><a href="#topic+enuToOtherAdp">enuToOtherAdp</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+handleFlags+2Cadp-method">handleFlags,adp-method</a></code>,
<code><a href="#topic+is.ad2cp">is.ad2cp</a>()</code>,
<code><a href="#topic+plot+2Cadp-method">plot,adp-method</a></code>,
<code><a href="#topic+read.adp.ad2cp">read.adp.ad2cp</a>()</code>,
<code><a href="#topic+read.adp.nortek">read.adp.nortek</a>()</code>,
<code><a href="#topic+read.adp.rdi">read.adp.rdi</a>()</code>,
<code><a href="#topic+read.adp.sontek.serial">read.adp.sontek.serial</a>()</code>,
<code><a href="#topic+read.adp.sontek">read.adp.sontek</a>()</code>,
<code><a href="#topic+read.adp">read.adp</a>()</code>,
<code><a href="#topic+read.aquadoppHR">read.aquadoppHR</a>()</code>,
<code><a href="#topic+read.aquadoppProfiler">read.aquadoppProfiler</a>()</code>,
<code><a href="#topic+read.aquadopp">read.aquadopp</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+setFlags+2Cadp-method">setFlags,adp-method</a></code>,
<code><a href="#topic+subset+2Cadp-method">subset,adp-method</a></code>,
<code><a href="#topic+subtractBottomVelocity">subtractBottomVelocity</a>()</code>,
<code><a href="#topic+summary+2Cadp-method">summary,adp-method</a></code>,
<code><a href="#topic+toEnuAdp">toEnuAdp</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnuAdp">xyzToEnuAdp</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

<hr>
<h2 id='xyzToEnuAdv'>Convert an adv Object From XYZ to ENU Coordinates</h2><span id='topic+xyzToEnuAdv'></span>

<h3>Description</h3>

<p>Convert ADV velocity components from a xyz-based coordinate system to
an enu-based coordinate system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>xyzToEnuAdv(
  x,
  declination = 0,
  cabled = FALSE,
  horizontalCase,
  sensorOrientation,
  debug = getOption("oceDebug")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="xyzToEnuAdv_+3A_x">x</code></td>
<td>
<p>an <a href="#topic+adv-class">adv</a> object.</p>
</td></tr>
<tr><td><code id="xyzToEnuAdv_+3A_declination">declination</code></td>
<td>
<p>magnetic declination to be added to the heading after
&quot;righting&quot; (see below), to get ENU with N as &quot;true&quot; north.  If this
is set to NULL, then the returned object is set up without adjusting
the compass for declination.  That means that <code>north</code> in its <code>metadata</code>
slot will be set to <code>"magnetic"</code>, and also that there will be no item
named <code>declination</code> in that slot.  Note that <code><a href="#topic+applyMagneticDeclination">applyMagneticDeclination()</a></code>
can be used later, to set a declination.</p>
</td></tr>
<tr><td><code id="xyzToEnuAdv_+3A_cabled">cabled</code></td>
<td>
<p>boolean value indicating whether the sensor head is connected
to the pressure case with a cable.  If <code>cabled=FALSE</code>, then
<code>horizontalCase</code> is ignored.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="xyzToEnuAdv_+3A_horizontalcase">horizontalCase</code></td>
<td>
<p>optional boolean value indicating whether the sensor
case is oriented horizontally.  Ignored unless <code>cabled</code> is <code>TRUE</code>.
See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="xyzToEnuAdv_+3A_sensororientation">sensorOrientation</code></td>
<td>
<p>optional string indicating the direction in which
the sensor points.  The value, which must be <code>"upward"</code> or
<code>"downward"</code>, over-rides the value of <code>orientation</code>,
in the <code>metadata</code> slot,
which will have been set by <code><a href="#topic+read.adv">read.adv()</a></code>, <em>provided</em> that the
data file contained the full header.  See &ldquo;Details&rdquo;.</p>
</td></tr>
<tr><td><code id="xyzToEnuAdv_+3A_debug">debug</code></td>
<td>
<p>a flag that, if non-zero, turns on debugging.  Higher values
yield more extensive debugging.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The coordinate transformation is done using the heading, pitch, and roll
information contained within <code>x</code>.  The algorithm is similar to that
used for Teledyne/RDI ADCP units, taking into account the different
definitions of heading, pitch, and roll as they are defined for the
velocimeters.
</p>
<p>Generally, the transformation must be done on a time-by-time basis, which is
a slow operation.  However, this function checks whether the vectors for
heading, pitch and roll, are all of unit length, and in that case, the
calculation is altered, resulting in shorter execution times.  Note that
the angles are held in (<code>data$timeSlow</code>, <code>data$headingSlow</code>, ...)
for Nortek instruments and (<code>data$time</code>, <code>data$heading</code>, ...) for
Sontek instruments.
</p>
<p>Since the documentation provided by instrument manufacturers can be vague on
the coordinate transformations, the method used here had to be developed
indirectly.  (This is in contrast to the RDI ADCP instruments, for which
there are clear instructions.)  documents that manufacturers provide.  If
results seem incorrect (e.g. if currents go east instead of west), users
should examine the code in detail for the case at hand.  The first step is
to set <code>debug</code> to 1, so that the processing will print a trail of
processing steps.  The next step should be to consult the table below, to
see if it matches the understanding (or empirical tests) of the user.  It
should not be difficult to tailor the code, if needed.
</p>
<p>The code handles every case individually, based on the table given below.
The table comes from Clark Richards, a former PhD student at Dalhousie
University (reference 2), who developed it based on instrument documentation,
discussion on user groups, and analysis of measurements acquired with Nortek
and Sontek velocimeters in the SLEIWEX experiment.
</p>
<p>The column labelled <code style="white-space: pre;">&#8288;Cabled'' indicates whether the sensor and the pressure case are connected with a flexible cable, and the column labelled &#8288;</code>H.case&rdquo;
indicates whether the pressure case is oriented horizontally.  These two
properties are not discoverable in the headers of the data files, and so
they must be supplied with the arguments <code>cabled</code> and
<code>horizontalCase</code>.  The source code refers to the information in this
table by case numbers.  (Cases 5 and 6 are not handled.)  Angles are
abbreviated as follows:: heading <code style="white-space: pre;">&#8288;H,'' pitch &#8288;</code>P,&rdquo; and roll &ldquo;R&rdquo;.
Entries X, Y and Z refer to instrument coordinates of the same names.
Entries S, F and M refer to so-called ship coordinates starboard, forward,
and mast; it is these that are used together with a rotation matrix to get
velocity components in the east, north, and upward directions.
</p>

<table>
<tr>
 <td style="text-align: right;">
<strong><code>Case</code></strong> </td><td style="text-align: right;"> <strong><code>Mfr.</code></strong> </td><td style="text-align: right;">
<strong><code>Instr.</code></strong> </td><td style="text-align: right;"> <strong><code>Cabled</code></strong> </td><td style="text-align: right;"> <strong><code style="white-space: pre;">&#8288;H. case&#8288;</code></strong> </td><td style="text-align: right;">
<strong><code>Orient.</code></strong> </td><td style="text-align: right;"> <strong><code>H</code></strong> </td><td style="text-align: right;"> <strong><code>P</code></strong> </td><td style="text-align: right;"> <strong><code>R</code></strong> </td><td style="text-align: right;">
<strong><code>S</code></strong> </td><td style="text-align: right;"> <strong><code>F</code></strong> </td><td style="text-align: right;"> <strong><code>M</code></strong></td>
</tr>
<tr>
 <td style="text-align: right;">

1 </td><td style="text-align: right;"> Nortek </td><td style="text-align: right;"> vector </td><td style="text-align: right;">
no </td><td style="text-align: right;"> - </td><td style="text-align: right;"> up </td><td style="text-align: right;"> H-90 </td><td style="text-align: right;"> R </td><td style="text-align: right;"> -P </td><td style="text-align: right;"> X </td><td style="text-align: right;"> -Y </td><td style="text-align: right;"> -Z</td>
</tr>
<tr>
 <td style="text-align: right;">

2 </td><td style="text-align: right;"> Nortek </td><td style="text-align: right;"> vector </td><td style="text-align: right;"> no </td><td style="text-align: right;"> - </td><td style="text-align: right;"> down </td><td style="text-align: right;"> H-90 </td><td style="text-align: right;"> R </td><td style="text-align: right;"> -P </td><td style="text-align: right;"> X
</td><td style="text-align: right;"> Y </td><td style="text-align: right;"> Z</td>
</tr>
<tr>
 <td style="text-align: right;">

3 </td><td style="text-align: right;"> Nortek </td><td style="text-align: right;"> vector </td><td style="text-align: right;"> yes </td><td style="text-align: right;"> yes </td><td style="text-align: right;"> up </td><td style="text-align: right;">
H-90 </td><td style="text-align: right;"> R </td><td style="text-align: right;"> -P </td><td style="text-align: right;"> X </td><td style="text-align: right;"> Y </td><td style="text-align: right;"> Z</td>
</tr>
<tr>
 <td style="text-align: right;">

4 </td><td style="text-align: right;"> Nortek </td><td style="text-align: right;"> vector </td><td style="text-align: right;">
yes </td><td style="text-align: right;"> yes </td><td style="text-align: right;"> down </td><td style="text-align: right;"> H-90 </td><td style="text-align: right;"> R </td><td style="text-align: right;"> P </td><td style="text-align: right;"> X </td><td style="text-align: right;"> -Y </td><td style="text-align: right;"> -Z</td>
</tr>
<tr>
 <td style="text-align: right;">

5 </td><td style="text-align: right;"> Nortek </td><td style="text-align: right;"> vector </td><td style="text-align: right;"> yes </td><td style="text-align: right;"> no </td><td style="text-align: right;"> up </td><td style="text-align: right;"> - </td><td style="text-align: right;"> - </td><td style="text-align: right;"> - </td><td style="text-align: right;"> -
</td><td style="text-align: right;"> - </td><td style="text-align: right;"> -</td>
</tr>
<tr>
 <td style="text-align: right;">

6 </td><td style="text-align: right;"> Nortek </td><td style="text-align: right;"> vector </td><td style="text-align: right;"> yes </td><td style="text-align: right;"> no </td><td style="text-align: right;"> down </td><td style="text-align: right;"> -
</td><td style="text-align: right;"> - </td><td style="text-align: right;"> - </td><td style="text-align: right;"> - </td><td style="text-align: right;"> - </td><td style="text-align: right;"> -</td>
</tr>
<tr>
 <td style="text-align: right;">

7 </td><td style="text-align: right;"> Sontek </td><td style="text-align: right;"> adv </td><td style="text-align: right;"> - </td><td style="text-align: right;"> -
</td><td style="text-align: right;"> up </td><td style="text-align: right;"> H-90 </td><td style="text-align: right;"> R </td><td style="text-align: right;"> -P </td><td style="text-align: right;"> X </td><td style="text-align: right;"> -Y </td><td style="text-align: right;"> -Z</td>
</tr>
<tr>
 <td style="text-align: right;">

8 </td><td style="text-align: right;"> Sontek </td><td style="text-align: right;"> adv </td><td style="text-align: right;"> - </td><td style="text-align: right;"> - </td><td style="text-align: right;"> down </td><td style="text-align: right;"> H-90 </td><td style="text-align: right;"> R </td><td style="text-align: right;"> -P </td><td style="text-align: right;"> X </td><td style="text-align: right;"> Y </td><td style="text-align: right;"> Z</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Dan Kelley, in collaboration with Clark Richards
</p>


<h3>References</h3>


<ol>
<li> <p><code style="white-space: pre;">&#8288;https://nortek.zendesk.com/hc/en-us/articles/360029820971-How-is-a-Coordinate-transformation-done-&#8288;</code>
</p>
</li>
<li><p> Clark Richards, 2012, PhD Dalhousie University Department of
Oceanography.
</p>
</li></ol>



<h3>See Also</h3>

<p>See <code><a href="#topic+read.adv">read.adv()</a></code> for notes on functions relating to
<code>adv</code> objects.
</p>
<p>Other things related to adv data: 
<code><a href="#topic++5B+5B+2Cadv-method">[[,adv-method</a></code>,
<code><a href="#topic++5B+5B+3C-+2Cadv-method">[[&lt;-,adv-method</a></code>,
<code><a href="#topic+adv-class">adv-class</a></code>,
<code><a href="#topic+advSontekAdrFileTrim">advSontekAdrFileTrim</a>()</code>,
<code><a href="#topic+adv">adv</a></code>,
<code><a href="#topic+applyMagneticDeclination+2Cadv-method">applyMagneticDeclination,adv-method</a></code>,
<code><a href="#topic+beamName">beamName</a>()</code>,
<code><a href="#topic+beamToXyz">beamToXyz</a>()</code>,
<code><a href="#topic+enuToOtherAdv">enuToOtherAdv</a>()</code>,
<code><a href="#topic+enuToOther">enuToOther</a>()</code>,
<code><a href="#topic+plot+2Cadv-method">plot,adv-method</a></code>,
<code><a href="#topic+read.adv.nortek">read.adv.nortek</a>()</code>,
<code><a href="#topic+read.adv.sontek.adr">read.adv.sontek.adr</a>()</code>,
<code><a href="#topic+read.adv.sontek.serial">read.adv.sontek.serial</a>()</code>,
<code><a href="#topic+read.adv.sontek.text">read.adv.sontek.text</a>()</code>,
<code><a href="#topic+read.adv">read.adv</a>()</code>,
<code><a href="#topic+rotateAboutZ">rotateAboutZ</a>()</code>,
<code><a href="#topic+subset+2Cadv-method">subset,adv-method</a></code>,
<code><a href="#topic+summary+2Cadv-method">summary,adv-method</a></code>,
<code><a href="#topic+toEnuAdv">toEnuAdv</a>()</code>,
<code><a href="#topic+toEnu">toEnu</a>()</code>,
<code><a href="#topic+velocityStatistics">velocityStatistics</a>()</code>,
<code><a href="#topic+xyzToEnu">xyzToEnu</a>()</code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
