<!DOCTYPE html><html lang="en"><head><title>Help for package ODT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ODT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#drug_response_w12'><p>drug_response_w12 data</p></a></li>
<li><a href='#drug_response_w34'><p>drug_response_w34 data</p></a></li>
<li><a href='#expression_w12'><p>expression_w12 Data Set</p></a></li>
<li><a href='#expression_w34'><p>expression_w34 Data Set</p></a></li>
<li><a href='#InternalFunctions'><p>ODT Internal Functions</p></a></li>
<li><a href='#mutations_w12'><p>mutations_w12 Data Set</p></a></li>
<li><a href='#mutations_w34'><p>mutations_w34 Data Set</p></a></li>
<li><a href='#niceTree'><p>niceTree function</p></a></li>
<li><a href='#predictTree'><p>Predict Treatment Outcomes with a Trained Decision Tree</p></a></li>
<li><a href='#trainTree'><p>trainTree Function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Optimal Decision Trees Algorithm</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Katyna Sada Del Real &lt;ksada@unav.es&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements a tree-based method specifically designed for personalized medicine applications. By using genomic and mutational data, 'ODT' efficiently identifies optimal drug recommendations tailored to individual patient profiles. The 'ODT' algorithm constructs decision trees that bifurcate at each node, selecting the most relevant markers (discrete or continuous) and corresponding treatments, thus ensuring that recommendations are both personalized and statistically robust. This iterative approach enhances therapeutic decision-making by refining treatment suggestions until a predefined group size is achieved. Moreover, the simplicity and interpretability of the resulting trees make the method accessible to healthcare professionals. Includes functions for training the decision tree, making predictions on new samples or patients, and visualizing the resulting tree. For detailed insights into the methodology, please refer to Gimeno et al. (2023) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbad200">doi:10.1093/bib/bbad200</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0), matrixStats, partykit, data.tree, stats</td>
</tr>
<tr>
<td>Imports:</td>
<td>magick, DiagrammeRsvg, grDevices, DiagrammeR, rsvg</td>
</tr>
<tr>
<td>Suggests:</td>
<td>RUnit, Matrix, rmarkdown, robustbase, knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-17 10:19:25 UTC; katyna</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-18 10:50:43 UTC</td>
</tr>
<tr>
<td>Author:</td>
<td>Maddi Eceiza [aut],
  Lucia Ruiz [aut],
  Angel Rubio [aut],
  Katyna Sada Del Real [aut, cre]</td>
</tr>
</table>
<hr>
<h2 id='drug_response_w12'>drug_response_w12 data</h2><span id='topic+drug_response_w12'></span>

<h3>Description</h3>

<p>A matrix containing drug response values (IC50 values) from patients retrieved from waves 1 and 2. Used as a toy example in <code>trainTree</code> and <code>predictTree</code> and <code>niceTree</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("drug_response_w12")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:247, 1:119] 2.710983 2.8755433 3.4390103 2.6527257...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(drug_response_w12)
</code></pre>

<hr>
<h2 id='drug_response_w34'>drug_response_w34 data</h2><span id='topic+drug_response_w34'></span>

<h3>Description</h3>

<p>A matrix containing drug response values (IC50 values) from patients retrieved from waves 3 and 4. Used as a toy example in <code>trainTree</code> and <code>predictTree</code> and <code>niceTree</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("drug_response_w34")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:247, 1:119] 3.4156359 3.2345985 3.1836058 3.7874252...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(drug_response_w34)
</code></pre>

<hr>
<h2 id='expression_w12'>expression_w12 Data Set</h2><span id='topic+expression_w12'></span>

<h3>Description</h3>

<p>A dataframe containing gene expression values (with different types of genes) from patients retrieved from waves 1 and 2. Used as a toy example in <code>trainTree</code> and <code>predictTree</code> and <code>niceTree</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("expression_w12")</code></pre>


<h3>Format</h3>

<p>A data frame with 22843 observations on the following 247 variables, 
where there are 247 patients from the waves and 22843 different genes   are considered.
</p>
<p>The format is:
22843 obs. 247 variables 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expression_w12)
</code></pre>

<hr>
<h2 id='expression_w34'>expression_w34 Data Set</h2><span id='topic+expression_w34'></span>

<h3>Description</h3>

<p>A dataframe containing gene expression values (with different types of genes) from patients retrieved from waves 3 and 4. Used as a toy example in <code>trainTree</code> and <code>predictTree</code> and <code>niceTree</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("expression_w34")</code></pre>


<h3>Format</h3>

<p>A data frame with 22843 observations on the following 142 variables, 
where there are 142 patients from the waves and 22843 different genes   are considered.
</p>
<p>The format is:
22843 obs. 142 variables 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expression_w34)
</code></pre>

<hr>
<h2 id='InternalFunctions'>ODT Internal Functions</h2><span id='topic+InternalFunctions'></span><span id='topic+getTreatment'></span><span id='topic+getSplit'></span><span id='topic+getsumic50v3'></span><span id='topic+findsplitExp'></span><span id='topic+growtreeExp'></span><span id='topic+findsplitMut'></span><span id='topic+growtreeMut'></span>

<h3>Description</h3>

<p>Internal functions used by 'trainTree' in the various
steps of the algorithm. These functions are not intended for
direct use by end users and are primarily for internal logic
and calculations within the package.
</p>
<p>This internal function calculates the optimal treatment based on patient sensitivity data.
</p>
<p>This internal function calculates the optimal split point based on a specified gene in the expression data.
</p>
<p>This internal function calculates the summed IC50 value based on the response of patients to a specific gene.
</p>
<p>This internal function identifies the optimal split point in expression data based on patient sensitivity (IC50).
</p>
<p>This internal function recursively builds a decision tree based on expression data and patient sensitivity (IC50).
</p>
<p>This internal function identifies the optimal split point in a mutation matrix based on
patient sensitivity data (IC50) and the presence of specific mutations.
</p>
<p>This internal function recursively builds a decision tree based on patient data and
drug sensitivity responses, specifically designed for mutation data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTreatment(PatientSensitivity, weights)

getSplit(gene, PatientData, PatientSensitivity)

getsumic50v3(genePatientResponse, PatientSensitivity)

findsplitExp(
  PatientSensitivity,
  PatientData,
  minimum = 1,
  weights = NULL,
  verbose = FALSE
)

growtreeExp(
  id = 1L,
  PatientSensitivity,
  PatientData,
  minbucket = 10,
  weights = NULL
)

findsplitMut(PatientSensitivity, X, minimum = 1, weights)

growtreeMut(
  id = 1L,
  PatientSensitivity,
  PatientData,
  minbucket = 10,
  weights = NULL,
  findsplit = findsplitMut
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="InternalFunctions_+3A_patientsensitivity">PatientSensitivity</code></td>
<td>
<p>A matrix representing drug response values (e.g., IC50), where rows correspond to patients/samples and columns correspond to drugs.</p>
</td></tr>
<tr><td><code id="InternalFunctions_+3A_weights">weights</code></td>
<td>
<p>A numeric vector indicating which samples are under study; defaults to NULL, meaning all samples are considered.</p>
</td></tr>
<tr><td><code id="InternalFunctions_+3A_gene">gene</code></td>
<td>
<p>The index of the gene used for splitting the data.</p>
</td></tr>
<tr><td><code id="InternalFunctions_+3A_patientdata">PatientData</code></td>
<td>
<p>A matrix representing patient features, where rows correspond to patients/samples and columns correspond to genes/features.</p>
</td></tr>
<tr><td><code id="InternalFunctions_+3A_genepatientresponse">genePatientResponse</code></td>
<td>
<p>A numeric vector representing the response of patients to a particular gene.</p>
</td></tr>
<tr><td><code id="InternalFunctions_+3A_minimum">minimum</code></td>
<td>
<p>An integer specifying the minimum number of samples required for a valid split (default is 1).</p>
</td></tr>
<tr><td><code id="InternalFunctions_+3A_verbose">verbose</code></td>
<td>
<p>A logical value indicating whether to print additional information during execution (default is FALSE).</p>
</td></tr>
<tr><td><code id="InternalFunctions_+3A_id">id</code></td>
<td>
<p>A unique identifier for the node in the decision tree (default is 1L).</p>
</td></tr>
<tr><td><code id="InternalFunctions_+3A_minbucket">minbucket</code></td>
<td>
<p>An integer specifying the minimum number of samples required in a child node for further splitting (default is 10).</p>
</td></tr>
<tr><td><code id="InternalFunctions_+3A_x">X</code></td>
<td>
<p>(PatientData) A matrix representing patient features, where rows correspond to patients/samples and columns correspond to genes/features.</p>
</td></tr>
<tr><td><code id="InternalFunctions_+3A_findsplit">findsplit</code></td>
<td>
<p>A function used to find the best split; defaults to 'findsplitMut'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of internal outputs generated by these functions.
</p>
<p>An integer indicating the index of the optimal treatment based on the minimum sum of sensitivity values.
</p>
<p>A list containing:
- 'sumic50': The minimum summed IC50 values for the two groups.
- 'Treatment1': The treatment associated with the first group.
- 'Treatment2': The treatment associated with the second group.
- 'split': The index of the optimal split point.
- 'expressionSplit': The expression value at the split point adjusted by a small epsilon.
</p>
<p>A numeric value representing the minimum summed IC50 value derived from the patient responses.
</p>
<p>A 'partysplit' object representing the optimal split based on the specified patient sensitivity and expression data.
</p>
<p>A 'partynode' object representing the current node and its children in the decision tree.
</p>
<p>A 'partysplit' object if a valid split is found, or 'NULL' if no valid split can be determined.
</p>
<p>A 'partynode' object representing the current node and its children in the decision tree.
</p>

<hr>
<h2 id='mutations_w12'>mutations_w12 Data Set</h2><span id='topic+mutations_w12'></span>

<h3>Description</h3>

<p>A binary matrix containing mutation values, wether the mutation is present or not in the patient from patients retrieved from waves 1 and 2. Used as a toy example in <code>trainTree</code> and <code>predictTree</code> and <code>niceTree</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("mutations_w12")</code></pre>


<h3>Format</h3>

<p>A binary matrix with,where there are 247 patients from the waves and    70 different mutations   are considered.
</p>
<p>The format is:
num [1:247, 1:70] 0 0 0 0 0 1...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mutations_w12)
</code></pre>

<hr>
<h2 id='mutations_w34'>mutations_w34 Data Set</h2><span id='topic+mutations_w34'></span>

<h3>Description</h3>

<p>A binary matrix containing mutation values, wether the mutation is present or not in the patient from patients retrieved from waves 3 and 4. Used as a toy example in <code>trainTree</code> and <code>predictTree</code> and <code>niceTree</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("mutations_w34")</code></pre>


<h3>Format</h3>

<p>A binary matrix with,where there are 142 patients from the waves and    70 different mutations   are considered.
</p>
<p>The format is:
num [1:142, 1:70] 0 0 0 1 0 0...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mutations_w34)
</code></pre>

<hr>
<h2 id='niceTree'>niceTree function</h2><span id='topic+niceTree'></span>

<h3>Description</h3>

<p>A graphical display of the tree. It can also be saved as an image in the selected directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>niceTree(
  tree,
  folder = NULL,
  colors = c("", "#367592", "#39A7AE", "#96D6B6", "#FDE5B0", "#F3908B", "#E36192",
    "#8E4884", "#A83333"),
  fontname = "Roboto",
  fontstyle = "plain",
  shape = "diamond",
  output_format = "png"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="niceTree_+3A_tree">tree</code></td>
<td>
<p>A party of the trained tree with the treatments assigned to each node.</p>
</td></tr>
<tr><td><code id="niceTree_+3A_folder">folder</code></td>
<td>
<p>Directory to save the image (default is the current working directory).</p>
</td></tr>
<tr><td><code id="niceTree_+3A_colors">colors</code></td>
<td>
<p>A vector of colors for the boxes. Can include hex color codes (e.g., &quot;#FFFFFF&quot;).</p>
</td></tr>
<tr><td><code id="niceTree_+3A_fontname">fontname</code></td>
<td>
<p>The name of the font to use for the text labels (default is &quot;Roboto&quot;).</p>
</td></tr>
<tr><td><code id="niceTree_+3A_fontstyle">fontstyle</code></td>
<td>
<p>The style of the font (e.g., &quot;plain&quot;, &quot;italic&quot;, &quot;bold&quot;).</p>
</td></tr>
<tr><td><code id="niceTree_+3A_shape">shape</code></td>
<td>
<p>The format of the boxes for the different genes (e.g., &quot;diamond&quot;, &quot;box&quot;).</p>
</td></tr>
<tr><td><code id="niceTree_+3A_output_format">output_format</code></td>
<td>
<p>The image format for saving (e.g., &quot;png&quot;, &quot;jpg&quot;, &quot;svg&quot;, &quot;pdf&quot;).</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p> The user has already defined a style for the plot; the parameters
are set if not modified when calling niceTree.
</p>
</li></ul>



<h3>Value</h3>

<p>(Invisibly) returns a list. The representation of the tree in the command window and the plot of the tree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Basic example of how to perform niceTree:
  data("mutations_w12")
  data("drug_response_w12")
  ODTmut &lt;- trainTree(PatientData = mutations_w12,
                       PatientSensitivity = drug_response_w12, minbucket = 10)
  niceTree(ODTmut)

  # Example for plotting the tree trained for gene expressions:
  data("expression_w34")
  data("drug_response_w34")
  ODTExp &lt;- trainTree(PatientData = expression_w34,
                       PatientSensitivity = drug_response_w34, minbucket = 20)
  niceTree(ODTExp)


</code></pre>

<hr>
<h2 id='predictTree'>Predict Treatment Outcomes with a Trained Decision Tree</h2><span id='topic+predictTree'></span>

<h3>Description</h3>

<p>This function utilizes a trained decision tree model (ODT) to predict treatment
outcomes for test data based on patient sensitivity data and features, such as
mutations or gene expression profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predictTree(tree, PatientData, PatientSensitivityTrain)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predictTree_+3A_tree">tree</code></td>
<td>
<p>A trained decision tree object created by the 'trainTree' function.</p>
</td></tr>
<tr><td><code id="predictTree_+3A_patientdata">PatientData</code></td>
<td>
<p>A matrix representing patient features, where rows correspond to patients/samples
and columns correspond to genes/features. This matrix can contain:
</p>

<ul>
<li><p> Binary mutation data (e.g., presence/absence of mutations).
</p>
</li>
<li><p> Continuous data from gene expression profiles (e.g., expression levels).
</p>
</li></ul>
</td></tr>
<tr><td><code id="predictTree_+3A_patientsensitivitytrain">PatientSensitivityTrain</code></td>
<td>
<p>A matrix containing the drug response values of the **training dataset**. 
In this matrix, rows correspond to patients, and columns correspond to drugs. 
This matrix is used solely for extracting treatment names and is not 
used in the prediction process itself.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A factor representing the assigned treatment for each node in the
decision tree based on the provided patient data and sensitivity.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # Example 1: Prediction using mutation data
  data("mutations_w12")
  data("mutations_w34")
  data("drug_response_w12")
  ODTmut &lt;- trainTree(PatientData = mutations_w12, 
                      PatientSensitivity = drug_response_w12, 
                      minbucket = 10)
  ODTmut
  ODT_mutpred &lt;- predictTree(tree = ODTmut, 
                              PatientSensitivityTrain = drug_response_w12, 
                              PatientData = mutations_w34)

  # Example 2: Prediction using gene expression data
  data("expression_w34")
  data("expression_w12")
  data("drug_response_w34")
  ODTExp &lt;- trainTree(PatientData = expression_w34, 
                       PatientSensitivity = drug_response_w34, 
                       minbucket = 20)
  ODTExp
  ODT_EXPpred &lt;- predictTree(tree = ODTExp, 
                              PatientSensitivityTrain = drug_response_w34, 
                              PatientData = expression_w12)


</code></pre>

<hr>
<h2 id='trainTree'>trainTree Function</h2><span id='topic+trainTree'></span>

<h3>Description</h3>

<p>This function trains a decision tree model based on patient data, which can either be gene expression levels or a binary matrix indicating mutations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trainTree(PatientData, PatientSensitivity, minbucket = 20)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trainTree_+3A_patientdata">PatientData</code></td>
<td>
<p>A matrix representing patient features, where rows correspond to patients/samples
and columns correspond to genes/features. This matrix can contain:
</p>

<ul>
<li><p> Binary mutation data (e.g., presence/absence of mutations).
</p>
</li>
<li><p> Continuous data from gene expression profiles (e.g., expression levels).
</p>
</li></ul>
</td></tr>
<tr><td><code id="trainTree_+3A_patientsensitivity">PatientSensitivity</code></td>
<td>
<p>A matrix representing drug response values, where rows correspond to patients
in the same order as in 'PatientData', and columns correspond to drugs. Higher values indicate greater drug resistance and, consequently, 
lower sensitivity to treatment. This matrix can represent various measures of drug 
response, such as IC50 values or area under the drug response curve (AUC). Depending 
on the interpretation of these values, users may need to adjust the sign of this data.</p>
</td></tr>
<tr><td><code id="trainTree_+3A_minbucket">minbucket</code></td>
<td>
<p>An integer specifying the minimum number of patients required in a node to allow for a split.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class 'party' representing the trained decision tree, with the assigned treatments for each node.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

  # Basic example of using the trainTree function with mutational data
  data("drug_response_w12")
  data("mutations_w12")
  ODTmut &lt;- trainTree(PatientData = mutations_w12, 
                      PatientSensitivity = drug_response_w12,
                      minbucket = 10)
  plot(ODTmut)

  # Example using gene expression data instead
  data("drug_response_w34")
  data("expression_w34")
  ODTExp &lt;- trainTree(PatientData = expression_w34,
                      PatientSensitivity = drug_response_w34,
                      minbucket = 20)
  plot(ODTExp)


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
