<!DOCTYPE html><html><head><title>Help for package ofGEM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ofGEM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#forest_plot'><p>Forest plot</p></a></li>
<li><a href='#ofGEM'><p>A meta-analysis approach with filtering for identifying gene-level</p>
gene-environment interactions with genetic association data</a></li>
<li><a href='#sim_dat'><p>A simulated data example</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>A Meta-Analysis Approach with Filtering for Identifying
Gene-Level Gene-Environment Interactions with Genetic
Association Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-07-06</td>
</tr>
<tr>
<td>Author:</td>
<td>Jiebiao Wang, Qianying Liu and Lin S. Chen</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jiebiao Wang &lt;randel.wang@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Offers a gene-based meta-analysis test with filtering to detect gene-environment interactions (GxE) with association data, proposed by Wang et al. (2018) &lt;<a href="https://doi.org/10.1002%2Fgepi.22115">doi:10.1002/gepi.22115</a>&gt;. It first conducts a meta-filtering test to filter out unpromising SNPs by combining all samples in the consortia data. It then runs a test of omnibus-filtering-based GxE meta-analysis (ofGEM) that combines the strengths of the fixed- and random-effects meta-analysis with meta-filtering. It can also analyze data from multiple ethnic groups. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Depends:</td>
<td>CompQuadForm, MASS, forestplot</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/randel/ofGEM">https://github.com/randel/ofGEM</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/randel/ofGEM/issues">https://github.com/randel/ofGEM/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-07-06 20:31:56 UTC; rande</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-07-13 14:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='forest_plot'>Forest plot</h2><span id='topic+forest_plot'></span>

<h3>Description</h3>

<p>It makes a forest plot for gene-environment interactions (GxE) across
multiple sub-studies for all the SNPs in a set (e.g., a gene).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>forest_plot(coef, se, sort = TRUE, exp = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="forest_plot_+3A_coef">coef</code></td>
<td>
<p>a matrix contains the GxE effect sizes. Each row corresponds to
a SNP in a set, and each column represents a study. If the matrix has
colnames and/or rownames for SNPs and studies respectively, they will be
used as labels in the forest plot.</p>
</td></tr>
<tr><td><code id="forest_plot_+3A_se">se</code></td>
<td>
<p>a matrix contains the standard errors for the GxE effect
estimates. Each row corresponds to a SNP in a set, and each column
represents a study.</p>
</td></tr>
<tr><td><code id="forest_plot_+3A_sort">sort</code></td>
<td>
<p>logical. If TRUE, the SNPs are sorted by averaged effect sizes
across different studies. The default is TRUE.</p>
</td></tr>
<tr><td><code id="forest_plot_+3A_exp">exp</code></td>
<td>
<p>logical. If TRUE, coef will be exponentiated. This option is
mainly used for log odds ratios obtained from logistic regressions. The
default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A forest plot across multiple sub-studies for all the SNPs in a set
(e.g., a gene).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

coef = matrix(rnorm(6 * 6), 6, 6)
se = matrix(abs(runif(6 * 6, 0.1, 0.15)), 6, 6)

forest_plot(coef, se)

</code></pre>

<hr>
<h2 id='ofGEM'>A meta-analysis approach with filtering for identifying gene-level
gene-environment interactions with genetic association data</h2><span id='topic+ofGEM'></span>

<h3>Description</h3>

<p>This function first conducts a meta-filtering test to filter out unpromising
SNPs. It then runs a test of omnibus-filtering-based GxE meta-analysis
(ofGEM) that combines the strengths of the fixed- and random-effects
meta-analysis with meta-filtering. It can also analyze data from multiple
ethnic groups. The p-values are calculated using a sequential sampling
approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ofGEM(Z, X, R, weight, threshold = 0.1, maxSim = 1e+06, tol = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ofGEM_+3A_z">Z</code></td>
<td>
<p>a matrix of test statistics for gene-environment interactions (GxE)
from consortium data. Each row corresponds to a SNP in a set (e.g., a gene),
and each column represents a study. For multi-ethnic groups, Z is a list
with each element being the matrix for each ethnic group.</p>
</td></tr>
<tr><td><code id="ofGEM_+3A_x">X</code></td>
<td>
<p>a matrix of filtering statistics for GxE. Each row corresponds to a
SNP in a set, and each column represents a study. For multi-ethnic groups, X
is a list with each element being the matrix for each ethnic group.</p>
</td></tr>
<tr><td><code id="ofGEM_+3A_r">R</code></td>
<td>
<p>the correlation matrix of test statistics for SNPs in a set. One
may use the genotype LD matrix for the set of SNPs to approximate it. This
matrix is used when sampling correlated testing and filtering statistics
under the null hypothesis and to obtain the null meta-analysis statistics.
For multi-ethnic groups, R is a list with each element being the correlation
matrix for each ethnic group.</p>
</td></tr>
<tr><td><code id="ofGEM_+3A_weight">weight</code></td>
<td>
<p>the weight vector for each study, or the weight matrix for
each SNP and each study. If the weight is the same across SNPs, it is a
vector with length equaling to the number of studies. If the weight is
different for different SNPs, it is a matrix with each row corresponding to
each SNP and each column representing each study.</p>
</td></tr>
<tr><td><code id="ofGEM_+3A_threshold">threshold</code></td>
<td>
<p>a fixed p-value threshold for filtering test. The default
is 0.1.</p>
</td></tr>
<tr><td><code id="ofGEM_+3A_maxsim">maxSim</code></td>
<td>
<p>the maximum number of samplings performed in obtaining the
sets of correlated testing and filtering statistics under the null. The
default is 1e6. This number determines the precision of the p-value
calculation.</p>
</td></tr>
<tr><td><code id="ofGEM_+3A_tol">tol</code></td>
<td>
<p>the tolerance number to stop the sequential sampling procedure.
The default is 10. We count the number of sampling-based null
meta-statistics that is more extreme than the observed meta-statistics. We
sequentially increase the number of sampling with an increment of 100. The
sequential sampling will stop if the cumulative count reaches tol. The idea
is to stop pursuing a higher precision with more sampling of null if the
p-value appears to be not significant. If tol = 0, the number of samplings
equals to maxSim.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing </p>
<table>
<tr><td><code>pval_random_mf</code></td>
<td>
<p>the p-value based on the
random-effects meta-analysis test with its corresponding meta-filtering.</p>
</td></tr>
<tr><td><code>pval_fixed_mf</code></td>
<td>
<p>the p-value based on the fixed-effects meta-analysis
test with its corresponding meta-filtering.</p>
</td></tr> <tr><td><code>pval_ofGEM</code></td>
<td>
<p>the p-value
based on using Fisher's method to aggregating the p-values of fixed- and
random-effects meta-analysis tests with meta-filtering</p>
</td></tr> <tr><td><code>nsim</code></td>
<td>
<p>the
number of samplings being performed.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Wang, Jiebiao, Qianying Liu, Brandon L. Pierce, Dezheng Huo, 
Olufunmilayo I. Olopade, Habibul Ahsan, and Lin S. Chen. 
&quot;A meta-analysis approach with filtering for identifying gene-level gene-environment interactions.&quot;
Genetic epidemiology (2018). https://doi.org/10.1002/gepi.22115
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(sim_dat)

pval = ofGEM(Z = sim_dat$Z, X = sim_dat$X, R = sim_dat$R, weight = rep(1/6, 6))


</code></pre>

<hr>
<h2 id='sim_dat'>A simulated data example</h2><span id='topic+sim_dat'></span>

<h3>Description</h3>

<p>This simulated data list is for demonstration.
</p>


<h3>Value</h3>

<p>A list containing </p>
<table>
<tr><td><code>Z</code></td>
<td>
<p>a simulated matrix of testing statistics
for GxE. Each row corresponds to a SNP in a set, and each column represents
a study.</p>
</td></tr> <tr><td><code>X</code></td>
<td>
<p>a simulated matrix of filtering statistics for GxE. Each
row corresponds to a SNP, and each column represents a study.</p>
</td></tr> <tr><td><code>R</code></td>
<td>
<p>a
simulated correlation matrix for testing (or filtering) statistics in a set
under the null. The testing statistics for SNPs in LD will be also
correlated. The correlations can be approximated by the LD matrix. We
resample SNP-level testing and filtering statistics from multivariate normal
distributions to calculate the null meta-analysis statistics.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data(sim_dat)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
