<!DOCTYPE html><html lang="en"><head><title>Help for package ogrdbstats</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ogrdbstats}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#example_rep'><p>Example repertoire data</p></a></li>
<li><a href='#generate_ogrdb_report'><p>Generate OGRDB reports from specified files.</p></a></li>
<li><a href='#genotype_statistics_cmd'><p>Collect parameters from the command line and use them to create a report and CSV file</p></a></li>
<li><a href='#make_barplot_grobs'><p>Create a barplot for each allele, showing number of reads distributed by mutation count</p></a></li>
<li><a href='#make_haplo_grobs'><p>Create haplotyping plots</p></a></li>
<li><a href='#make_novel_base_grobs'><p>Create plots showing base usage at selected locations in sequences based on novel alleles</p></a></li>
<li><a href='#read_input_files'><p>Read input files into memory</p></a></li>
<li><a href='#write_genotype_file'><p>Write the genotype file required by OGRDB</p></a></li>
<li><a href='#write_plot_file'><p>Create the OGRDB style plot file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysis of Adaptive Immune Receptor Repertoire Germ Line
Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.2</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/airr-community/ogrdbstats">https://github.com/airr-community/ogrdbstats</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/airr-community/ogrdbstats/issues">https://github.com/airr-community/ogrdbstats/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Multiple tools are now available for inferring the personalised 
    germ line set from an adaptive immune receptor repertoire. 
    Output from these tools is converted to 
    a single format and supplemented with rich data such as usage and 
    characterisation of 'novel' germ line alleles. This data can be 
    particularly useful when considering the validity of novel inferences. Use
    of the analysis provided is described in &lt;<a href="https://doi.org/10.3389%2Ffimmu.2019.00435">doi:10.3389/fimmu.2019.00435</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://creativecommons.org/licenses/by-sa/4.0">CC BY-SA 4.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr (&ge; 0.8.3), ggplot2 (&ge; 3.2.1), magrittr, tigger (&ge;
0.4.0), alakazam (&ge; 0.3.0), stringr (&ge; 1.4.0), data.table,
gridExtra (&ge; 2.3), tidyr (&ge; 1.0.0), stringdist (&ge; 0.9.5.2),
RColorBrewer (&ge; 1.1-2), Biostrings (&ge; 2.52.0), argparser (&ge;
0.4), ComplexHeatmap, bookdown, scales,</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-02 14:56:39 UTC; William</td>
</tr>
<tr>
<td>Author:</td>
<td>William Lees <a href="https://orcid.org/0000-0001-9834-6840"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>William Lees &lt;william@lees.org.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-03 06:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='example_rep'>Example repertoire data</h2><span id='topic+example_rep'></span>

<h3>Description</h3>

<p>A small example of the analytical datasets created by ogrdbstats from
repertoires and reference sets.
The dataset can be created by running the example shown for the function
read_input_data().
The dataset is created from example files provided with the package.
The repertoire data is taken from Rubelt et al. 2016,
&lt;doi: 10.1038/ncomms11112&gt;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_rep
</code></pre>


<h3>Format</h3>

<p>## 'example_rep' - a named list containing the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
ref_genes  </td><td style="text-align: left;"> named list of IMGT-gapped reference genes </td>
</tr>
<tr>
 <td style="text-align: left;">
inferred_seqs </td><td style="text-align: left;"> named list of IMGT-gapped inferred (novel) sequences. </td>
</tr>
<tr>
 <td style="text-align: left;">
input_sequences </td><td style="text-align: left;"> data frame with one row per annotated read, with
      CHANGEO-style column names. The column SEG_CALL is the gene call
     for the segment under analysis. Hence if segment is 'V', 'V_CALL' will
     be renamed 'SEG_CALL' whereas is segment is 'J', 'J_CALL' is renamed
     'SEG_CALL'. This simplifies downstream processing. Rows in the input
     file with ambiguous SEG_CALLs, or no call, are removed. </td>
</tr>
<tr>
 <td style="text-align: left;">
genotype_db </td><td style="text-align: left;"> named list of gene sequences referenced in the annotated
     reads (both reference and novel sequences) </td>
</tr>
<tr>
 <td style="text-align: left;">
haplo_details </td><td style="text-align: left;"> data used for haplotype analysis, showing allelic ratios
     calculated with various potential haplotyping genes </td>
</tr>
<tr>
 <td style="text-align: left;">
genotype </td><td style="text-align: left;"> data frame containing information provided in the OGRDB
     genotype csv file </td>
</tr>
<tr>
 <td style="text-align: left;">
calculated_NC </td><td style="text-align: left;"> a boolean that is TRUE if mutation counts were calculated
     by this library, FALSE if they were read from the annotated read file </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Source</h3>

<p>&lt;doi: 10.1038/ncomms11112&gt;
</p>

<hr>
<h2 id='generate_ogrdb_report'>Generate OGRDB reports from specified files.</h2><span id='topic+generate_ogrdb_report'></span>

<h3>Description</h3>

<p>This creates the genotype report
(suffixed _ogrdb_report.csv) and the plot file (suffixed _ogrdb_plos.pdf).
Both are created in the directory holding the annotated read file, and the
file names are prefixed by the name of the annotated read file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_ogrdb_report(
  ref_filename,
  inferred_filename,
  species,
  filename,
  chain,
  hap_gene,
  segment,
  chain_type,
  plot_unmutated,
  all_inferred = FALSE,
  format = "pdf"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_ogrdb_report_+3A_ref_filename">ref_filename</code></td>
<td>
<p>Name of file containing IMGT-aligned reference genes in FASTA format</p>
</td></tr>
<tr><td><code id="generate_ogrdb_report_+3A_inferred_filename">inferred_filename</code></td>
<td>
<p>Name of file containing sequences of inferred novel alleles, or '-' if none</p>
</td></tr>
<tr><td><code id="generate_ogrdb_report_+3A_species">species</code></td>
<td>
<p>Species name used in field 3 of the IMGT germline header with spaces omitted, if the reference file is from IMGT. Otherwise &rdquo;</p>
</td></tr>
<tr><td><code id="generate_ogrdb_report_+3A_filename">filename</code></td>
<td>
<p>Name of file containing annotated reads in AIRR, CHANGEO or IgDiscover format. The format is detected automatically</p>
</td></tr>
<tr><td><code id="generate_ogrdb_report_+3A_chain">chain</code></td>
<td>
<p>one of IGHV, IGKV, IGLV, IGHD, IGHJ, IGKJ, IGLJ, TRAV, TRAj, TRBV, TRBD, TRBJ, TRGV, TRGj, TRDV, TRDD, TRDJ</p>
</td></tr>
<tr><td><code id="generate_ogrdb_report_+3A_hap_gene">hap_gene</code></td>
<td>
<p>The haplotyping columns will be completed based on the usage of the two most frequent alleles of this gene. If NA, the column will be blank</p>
</td></tr>
<tr><td><code id="generate_ogrdb_report_+3A_segment">segment</code></td>
<td>
<p>one of V, D, J</p>
</td></tr>
<tr><td><code id="generate_ogrdb_report_+3A_chain_type">chain_type</code></td>
<td>
<p>one of H, L</p>
</td></tr>
<tr><td><code id="generate_ogrdb_report_+3A_plot_unmutated">plot_unmutated</code></td>
<td>
<p>Plot base composition using only unmutated sequences (V-chains only)</p>
</td></tr>
<tr><td><code id="generate_ogrdb_report_+3A_all_inferred">all_inferred</code></td>
<td>
<p>Treat all alleles as novel</p>
</td></tr>
<tr><td><code id="generate_ogrdb_report_+3A_format">format</code></td>
<td>
<p>The format for the plot file ('pdf', 'html' or 'none')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'># prepare files for example
reference_set = system.file("extdata/ref_gapped.fasta", package = "ogrdbstats")
inferred_set = system.file("extdata/novel_gapped.fasta", package = "ogrdbstats")
repertoire = system.file("extdata/ogrdbstats_example_repertoire.tsv", package = "ogrdbstats")
file.copy(repertoire, tempdir())
repfile = file.path(tempdir(), 'ogrdbstats_example_repertoire.tsv')

generate_ogrdb_report(reference_set, inferred_set, 'Homosapiens',
          repfile, 'IGHV', NA, 'V', 'H', FALSE, format='none')

#clean up
outfile = file.path(tempdir(), 'ogrdbstats_example_repertoire_ogrdb_report.csv')
file.remove(repfile)
file.remove(outfile)
</code></pre>

<hr>
<h2 id='genotype_statistics_cmd'>Collect parameters from the command line and use them to create a report and CSV file</h2><span id='topic+genotype_statistics_cmd'></span>

<h3>Description</h3>

<p>Collect parameters from the command line and use them to create a report and CSV file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotype_statistics_cmd(args = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genotype_statistics_cmd_+3A_args">args</code></td>
<td>
<p>A string vector containing the command line arguments. If NULL, will take them from the command line</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Prepare files for example
reference_set = system.file("extdata/ref_gapped.fasta", package = "ogrdbstats")
inferred_set = system.file("extdata/novel_gapped.fasta", package = "ogrdbstats")
repertoire = system.file("extdata/ogrdbstats_example_repertoire.tsv", package = "ogrdbstats")
file.copy(repertoire, tempdir())
repfile = file.path(tempdir(), 'repertoire.tsv')

genotype_statistics_cmd(c(
              reference_set,
              'Homosapiens',
              repfile,
              'IGHV',
              '--inf_file', inferred_set,
              '--format', 'none'))

# clean up
outfile = file.path(tempdir(), 'repertoire_ogrdb_report.csv')
plotdir = file.path(tempdir(), 'repertoire_ogrdb_plots')
file.remove(repfile)
file.remove(outfile)
unlink(plotdir, recursive=TRUE)
</code></pre>

<hr>
<h2 id='make_barplot_grobs'>Create a barplot for each allele, showing number of reads distributed by mutation count</h2><span id='topic+make_barplot_grobs'></span>

<h3>Description</h3>

<p>Create a barplot for each allele, showing number of reads distributed by mutation count
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_barplot_grobs(
  input_sequences,
  genotype_db,
  inferred_seqs,
  genotype,
  segment,
  calculated_NC
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_barplot_grobs_+3A_input_sequences">input_sequences</code></td>
<td>
<p>the input_sequences data frame</p>
</td></tr>
<tr><td><code id="make_barplot_grobs_+3A_genotype_db">genotype_db</code></td>
<td>
<p>named list of gene sequences in the personalised genotype</p>
</td></tr>
<tr><td><code id="make_barplot_grobs_+3A_inferred_seqs">inferred_seqs</code></td>
<td>
<p>named list of novel gene sequences</p>
</td></tr>
<tr><td><code id="make_barplot_grobs_+3A_genotype">genotype</code></td>
<td>
<p>data frame created by calc_genotype</p>
</td></tr>
<tr><td><code id="make_barplot_grobs_+3A_segment">segment</code></td>
<td>
<p>one of V, D, J</p>
</td></tr>
<tr><td><code id="make_barplot_grobs_+3A_calculated_nc">calculated_NC</code></td>
<td>
<p>a boolean, TRUE if mutation counts had to be calculated, FALSE otherwise</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of grobs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>barplot_grobs = make_barplot_grobs(
                      example_rep$input_sequences,
                      example_rep$genotype_db,
                      example_rep$inferred_seqs,
                      example_rep$genotype,
                      'V',
                      example_rep$calculated_NC
               )

</code></pre>

<hr>
<h2 id='make_haplo_grobs'>Create haplotyping plots</h2><span id='topic+make_haplo_grobs'></span>

<h3>Description</h3>

<p>Create haplotyping plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_haplo_grobs(segment, haplo_details)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_haplo_grobs_+3A_segment">segment</code></td>
<td>
<p>one of V, D, J</p>
</td></tr>
<tr><td><code id="make_haplo_grobs_+3A_haplo_details">haplo_details</code></td>
<td>
<p>Data structure created by create_haplo_details</p>
</td></tr>
</table>


<h3>Value</h3>

<p>named list containing the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
    a_allele_plot </td><td style="text-align: left;">   plot showing allele usage for each potential haplotyping gene </td>
</tr>
<tr>
 <td style="text-align: left;">
    haplo_grobs </td><td style="text-align: left;">     differential plot of allele usage for each usable haplotyping gene </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>haplo_grobs = make_haplo_grobs('V', example_rep$haplo_details)
</code></pre>

<hr>
<h2 id='make_novel_base_grobs'>Create plots showing base usage at selected locations in sequences based on novel alleles</h2><span id='topic+make_novel_base_grobs'></span>

<h3>Description</h3>

<p>Create plots showing base usage at selected locations in sequences based on novel alleles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_novel_base_grobs(inferred_seqs, input_sequences, segment, all_inferred)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="make_novel_base_grobs_+3A_inferred_seqs">inferred_seqs</code></td>
<td>
<p>named list of novel gene sequences</p>
</td></tr>
<tr><td><code id="make_novel_base_grobs_+3A_input_sequences">input_sequences</code></td>
<td>
<p>the input_sequences data frame</p>
</td></tr>
<tr><td><code id="make_novel_base_grobs_+3A_segment">segment</code></td>
<td>
<p>one of V, D, J</p>
</td></tr>
<tr><td><code id="make_novel_base_grobs_+3A_all_inferred">all_inferred</code></td>
<td>
<p>true if user has requested all alleles in reference set plotted - will suppress some warnings</p>
</td></tr>
</table>


<h3>Value</h3>

<p>named list containing the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
    cdr3_dist </td><td style="text-align: left;"> cdr3 length distribution plots </td>
</tr>
<tr>
 <td style="text-align: left;">
    whole </td><td style="text-align: left;">     whole-length usage plots </td>
</tr>
<tr>
 <td style="text-align: left;">
    end  </td><td style="text-align: left;">     3' end usage plots </td>
</tr>
<tr>
 <td style="text-align: left;">
    conc </td><td style="text-align: left;">   3' end consensus composition plots </td>
</tr>
<tr>
 <td style="text-align: left;">
    triplet </td><td style="text-align: left;">   3' end triplet usage plots </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>base_grobs = make_novel_base_grobs(
                 example_rep$inferred_seqs,
                 example_rep$input_sequences,
                 'V',
                 FALSE
             )
</code></pre>

<hr>
<h2 id='read_input_files'>Read input files into memory</h2><span id='topic+read_input_files'></span>

<h3>Description</h3>

<p>Read input files into memory
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_input_files(
  ref_filename,
  inferred_filename,
  species,
  filename,
  chain,
  hap_gene,
  segment,
  chain_type,
  all_inferred
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_input_files_+3A_ref_filename">ref_filename</code></td>
<td>
<p>Name of file containing IMGT-aligned reference genes in FASTA format</p>
</td></tr>
<tr><td><code id="read_input_files_+3A_inferred_filename">inferred_filename</code></td>
<td>
<p>Name of file containing sequences of inferred novel alleles, or '-' if none</p>
</td></tr>
<tr><td><code id="read_input_files_+3A_species">species</code></td>
<td>
<p>Species name used in field 3 of the IMGT germline header with spaces omitted, if the reference file is from IMGT. Otherwise &rdquo;</p>
</td></tr>
<tr><td><code id="read_input_files_+3A_filename">filename</code></td>
<td>
<p>Name of file containing annotated reads in AIRR, CHANGEO or IgDiscover format. The format is detected automatically</p>
</td></tr>
<tr><td><code id="read_input_files_+3A_chain">chain</code></td>
<td>
<p>one of IGHV, IGKV, IGLV, IGHD, IGHJ, IGKJ, IGLJ, TRAV, TRAj, TRBV, TRBD, TRBJ, TRGV, TRGj, TRDV, TRDD, TRDJ</p>
</td></tr>
<tr><td><code id="read_input_files_+3A_hap_gene">hap_gene</code></td>
<td>
<p>The haplotyping columns will be completed based on the usage of the two most frequent alleles of this gene. If NA, the column will be blank</p>
</td></tr>
<tr><td><code id="read_input_files_+3A_segment">segment</code></td>
<td>
<p>one of V, D, J</p>
</td></tr>
<tr><td><code id="read_input_files_+3A_chain_type">chain_type</code></td>
<td>
<p>one of H, L</p>
</td></tr>
<tr><td><code id="read_input_files_+3A_all_inferred">all_inferred</code></td>
<td>
<p>Treat all alleles as novel</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list containing the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
ref_genes  </td><td style="text-align: left;"> named list of IMGT-gapped reference genes </td>
</tr>
<tr>
 <td style="text-align: left;">
inferred_seqs </td><td style="text-align: left;"> named list of IMGT-gapped inferred (novel) sequences. </td>
</tr>
<tr>
 <td style="text-align: left;">
input_sequences </td><td style="text-align: left;"> data frame with one row per annotated read, with CHANGEO-style column names
                          One key point: the column SEG_CALL is the gene call for the segment under analysis. Hence if segment is 'V',
                          'V_CALL' will be renamed 'SEG_CALL' whereas is segment is 'J', 'J_CALL' is renamed 'SEG_CALL'. This simplifies
                          downstream processing.
                          Rows in the input file with ambiguous SEG_CALLs, or no call, are removed. </td>
</tr>
<tr>
 <td style="text-align: left;">
genotype_db </td><td style="text-align: left;"> named list of gene sequences referenced in the annotated reads (both reference and novel sequences) </td>
</tr>
<tr>
 <td style="text-align: left;">
haplo_details </td><td style="text-align: left;"> data used for haplotype analysis, showing allelic ratios calculated with various potential haplotyping genes </td>
</tr>
<tr>
 <td style="text-align: left;">
genotype </td><td style="text-align: left;"> data frame containing information provided in the OGRDB genotype csv file </td>
</tr>
<tr>
 <td style="text-align: left;">
calculated_NC </td><td style="text-align: left;"> a boolean that is TRUE if mutation counts were calculated by this library, FALSE if they were read from the annotated read file </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'># Create the analysis data set from example files provided with the package
#(this dataset is also provided in the package as example_rep)
reference_set = system.file("extdata/ref_gapped.fasta", package = "ogrdbstats")
inferred_set = system.file("extdata/novel_gapped.fasta", package = "ogrdbstats")
repertoire = system.file("extdata/ogrdbstats_example_repertoire.tsv", package = "ogrdbstats")

example_data = read_input_files(reference_set, inferred_set, 'Homosapiens',
       repertoire, 'IGHV', NA, 'V', 'H', FALSE)
</code></pre>

<hr>
<h2 id='write_genotype_file'>Write the genotype file required by OGRDB</h2><span id='topic+write_genotype_file'></span>

<h3>Description</h3>

<p>Write the genotype file required by OGRDB
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_genotype_file(filename, segment, chain_type, genotype)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_genotype_file_+3A_filename">filename</code></td>
<td>
<p>name of file to create (csv)</p>
</td></tr>
<tr><td><code id="write_genotype_file_+3A_segment">segment</code></td>
<td>
<p>one of V, D, J</p>
</td></tr>
<tr><td><code id="write_genotype_file_+3A_chain_type">chain_type</code></td>
<td>
<p>one of H, L</p>
</td></tr>
<tr><td><code id="write_genotype_file_+3A_genotype">genotype</code></td>
<td>
<p>genotype data frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotype_file = tempfile("ogrdb_genotype")
write_genotype_file(genotype_file, 'V', 'H', example_rep$genotype)
file.remove(genotype_file)
</code></pre>

<hr>
<h2 id='write_plot_file'>Create the OGRDB style plot file</h2><span id='topic+write_plot_file'></span>

<h3>Description</h3>

<p>Create the OGRDB style plot file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_plot_file(
  filename,
  input_sequences,
  cdr3_dist_grobs,
  end_composition_grobs,
  cons_composition_grobs,
  whole_composition_grobs,
  triplet_composition_grobs,
  barplot_grobs,
  a_allele_plot,
  haplo_grobs,
  message,
  format
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_plot_file_+3A_filename">filename</code></td>
<td>
<p>name of file to create (pdf)</p>
</td></tr>
<tr><td><code id="write_plot_file_+3A_input_sequences">input_sequences</code></td>
<td>
<p>the input_sequences data frame</p>
</td></tr>
<tr><td><code id="write_plot_file_+3A_cdr3_dist_grobs">cdr3_dist_grobs</code></td>
<td>
<p>cdr3 length distribution grobs created by make_novel_base_grob</p>
</td></tr>
<tr><td><code id="write_plot_file_+3A_end_composition_grobs">end_composition_grobs</code></td>
<td>
<p>end composition grobs created by make_novel_base_grobs</p>
</td></tr>
<tr><td><code id="write_plot_file_+3A_cons_composition_grobs">cons_composition_grobs</code></td>
<td>
<p>consensus composition grobs created by make_novel_base_grobs</p>
</td></tr>
<tr><td><code id="write_plot_file_+3A_whole_composition_grobs">whole_composition_grobs</code></td>
<td>
<p>whole composition grobs created by make_novel_base_grobs</p>
</td></tr>
<tr><td><code id="write_plot_file_+3A_triplet_composition_grobs">triplet_composition_grobs</code></td>
<td>
<p>triplet composition grobs created by make_novel_base_grobs</p>
</td></tr>
<tr><td><code id="write_plot_file_+3A_barplot_grobs">barplot_grobs</code></td>
<td>
<p>barplot grobs created by make_barplot_grons</p>
</td></tr>
<tr><td><code id="write_plot_file_+3A_a_allele_plot">a_allele_plot</code></td>
<td>
<p>a_allele_plot grob created by make_haplo_grobs</p>
</td></tr>
<tr><td><code id="write_plot_file_+3A_haplo_grobs">haplo_grobs</code></td>
<td>
<p>haplo_grobs created by make_haplo_grobs</p>
</td></tr>
<tr><td><code id="write_plot_file_+3A_message">message</code></td>
<td>
<p>text message to display at end of report</p>
</td></tr>
<tr><td><code id="write_plot_file_+3A_format">format</code></td>
<td>
<p>Format of report ('pdf', 'html' or 'none')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_file = tempfile(pattern = 'ogrdb_plots')

base_grobs = make_novel_base_grobs(
                 example_rep$inferred_seqs,
                 example_rep$input_sequences,
                 'V',
                 FALSE
             )
barplot_grobs = make_barplot_grobs(
                      example_rep$input_sequences,
                      example_rep$genotype_db,
                      example_rep$inferred_seqs,
                      example_rep$genotype,
                      'V',
                      example_rep$calculated_NC
               )
haplo_grobs = make_haplo_grobs('V', example_rep$haplo_details)

write_plot_file(
    plot_file,
    example_rep$input_sequences,
    base_grobs$cdr3_dist,
    base_grobs$end,
    base_grobs$conc,
    base_grobs$whole,
    base_grobs$triplet,
    barplot_grobs,
    haplo_grobs$aplot,
    haplo_grobs$haplo,
    "Notes on this analysis",
    'none'
)

file.remove(plot_file)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
