<!DOCTYPE html><html><head><title>Help for package OmicSense</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {OmicSense}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#os.clean'><p>Clean data by eliminating predictors with many missing values</p></a></li>
<li><a href='#os.pca'><p>Visualize predictors using principal coordinate analysis</p></a></li>
<li><a href='#os.pred'><p>Construct and apply the OmicSense model with your own data</p></a></li>
<li><a href='#os.rank'><p>Visualize R-squared value distribution in target-predictor relationship</p></a></li>
<li><a href='#os.sort'><p>Sort and select predictors according to the strength of target-predictor relationship</p></a></li>
<li><a href='#Pinus'><p>Transcriptomes of Pinus roots under a Temperature Gradient</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Biosensor Development using Omics Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Takahiko Koizumi &lt;takahiko.koizumi@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A method for the quantitative prediction using omics data. 
    This package provides functions to construct the quantitative prediction 
    model using omics data. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>URL:</td>
<td>&lt;<a href="https://github.com/takakoizumi/OmicSense&amp;gt;">https://github.com/takakoizumi/OmicSense&gt;</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, kernlab</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-22 21:14:43 UTC; taka</td>
</tr>
<tr>
<td>Author:</td>
<td>Takahiko Koizumi [aut, cre],
  Kenta Suzuki [ctb],
  Yasunori Ichihashi [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-22 21:30:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='os.clean'>Clean data by eliminating predictors with many missing values</h2><span id='topic+os.clean'></span>

<h3>Description</h3>

<p>Clean data by eliminating predictors with many missing values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>os.clean(x, missing = 0.1, lowest = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="os.clean_+3A_x">x</code></td>
<td>
<p>A data matrix (raw: samples, col: predictors).</p>
</td></tr>
<tr><td><code id="os.clean_+3A_missing">missing</code></td>
<td>
<p>A ratio of missing values in each column allowed to be remained in the data.</p>
</td></tr>
<tr><td><code id="os.clean_+3A_lowest">lowest</code></td>
<td>
<p>The lowest value to be leaved in the data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data matrix (raw: samples, col: qualified predictors)
</p>


<h3>Author(s)</h3>

<p>Takahiko Koizumi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pinus)
train.raw &lt;- Pinus$train
ncol(train.raw)

train &lt;- os.clean(train.raw)
ncol(train)
</code></pre>

<hr>
<h2 id='os.pca'>Visualize predictors using principal coordinate analysis</h2><span id='topic+os.pca'></span>

<h3>Description</h3>

<p>Visualize predictors using principal coordinate analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>os.pca(x, y, method = "linear", thresh = 0, n.pred = ncol(x), size = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="os.pca_+3A_x">x</code></td>
<td>
<p>A data matrix (row: samples, col: predictors).</p>
</td></tr>
<tr><td><code id="os.pca_+3A_y">y</code></td>
<td>
<p>A vector of target value.</p>
</td></tr>
<tr><td><code id="os.pca_+3A_method">method</code></td>
<td>
<p>A string to specify the regression function for calculating R-squared values.
&quot;linear&quot; (default), &quot;quadratic&quot; or &quot;cubic&quot; function can be specified.</p>
</td></tr>
<tr><td><code id="os.pca_+3A_thresh">thresh</code></td>
<td>
<p>The lower threshold of R-squared value to be indicated in a PCA plot (default: 0).</p>
</td></tr>
<tr><td><code id="os.pca_+3A_n.pred">n.pred</code></td>
<td>
<p>The number of candidate predictors for prediction model to be indicated in a PCA plot (default: ncol(x)).</p>
</td></tr>
<tr><td><code id="os.pca_+3A_size">size</code></td>
<td>
<p>The size of symbols in a PCA plot (default: 1).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A PCA plot
</p>


<h3>Author(s)</h3>

<p>Takahiko Koizumi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pinus)
train &lt;- os.clean(Pinus$train)
target &lt;- Pinus$target
os.pca(train, target)
</code></pre>

<hr>
<h2 id='os.pred'>Construct and apply the OmicSense model with your own data</h2><span id='topic+os.pred'></span>

<h3>Description</h3>

<p>Construct and apply the OmicSense model with your own data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>os.pred(x, y, newx = x, method = "linear", thresh = 0, n.pred = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="os.pred_+3A_x">x</code></td>
<td>
<p>A data matrix (row: samples, col: predictors).</p>
</td></tr>
<tr><td><code id="os.pred_+3A_y">y</code></td>
<td>
<p>A vector of target value.</p>
</td></tr>
<tr><td><code id="os.pred_+3A_newx">newx</code></td>
<td>
<p>A data matrix (row: samples, col: predictors).</p>
</td></tr>
<tr><td><code id="os.pred_+3A_method">method</code></td>
<td>
<p>A string to specify the regression function for calculating R-squared values.
&quot;linear&quot; (default), &quot;quadratic&quot; or &quot;cubic&quot; function can be specified.</p>
</td></tr>
<tr><td><code id="os.pred_+3A_thresh">thresh</code></td>
<td>
<p>The lower threshold of R-squared value to be leaved in prediction model (default: 0).</p>
</td></tr>
<tr><td><code id="os.pred_+3A_n.pred">n.pred</code></td>
<td>
<p>The number of candidate predictors to be leaved in prediction model (default: 30).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of the environment in which the samples of newx were collected
</p>


<h3>Author(s)</h3>

<p>Takahiko Koizumi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pinus)
train &lt;- os.clean(Pinus$train)
test &lt;- Pinus$test
test &lt;- test[, colnames(train)]
target &lt;- Pinus$target
cor(target, os.pred(train, target, newx = test, method = "cubic"))
</code></pre>

<hr>
<h2 id='os.rank'>Visualize R-squared value distribution in target-predictor relationship</h2><span id='topic+os.rank'></span>

<h3>Description</h3>

<p>Visualize R-squared value distribution in target-predictor relationship
</p>


<h3>Usage</h3>

<pre><code class='language-R'>os.rank(
  x,
  y,
  method = "linear",
  thresh = 0,
  n.pred = ncol(x),
  upper.xlim = ncol(x)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="os.rank_+3A_x">x</code></td>
<td>
<p>A data matrix (row: samples, col: predictors).</p>
</td></tr>
<tr><td><code id="os.rank_+3A_y">y</code></td>
<td>
<p>A vector of target value.</p>
</td></tr>
<tr><td><code id="os.rank_+3A_method">method</code></td>
<td>
<p>A string to specify the regression function for calculating R-squared values.
&quot;linear&quot; (default), &quot;quadratic&quot; or &quot;cubic&quot; function can be specified.</p>
</td></tr>
<tr><td><code id="os.rank_+3A_thresh">thresh</code></td>
<td>
<p>The lower threshold of R-squared value to be leaved in prediction model (default: 0).</p>
</td></tr>
<tr><td><code id="os.rank_+3A_n.pred">n.pred</code></td>
<td>
<p>The number of predictors to be leaved in prediction model (default: ncol(x)).</p>
</td></tr>
<tr><td><code id="os.rank_+3A_upper.xlim">upper.xlim</code></td>
<td>
<p>The upper limitation of x axis (i.e., the number of predictors) in the resulted figure (default: ncol(x)).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A rank order plot
</p>


<h3>Author(s)</h3>

<p>Takahiko Koizumi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pinus)
train &lt;- os.clean(Pinus$train)
target &lt;- Pinus$target
train &lt;- os.sort(train, target)
os.rank(train, target)
</code></pre>

<hr>
<h2 id='os.sort'>Sort and select predictors according to the strength of target-predictor relationship</h2><span id='topic+os.sort'></span>

<h3>Description</h3>

<p>Sort and select predictors according to the strength of target-predictor relationship
</p>


<h3>Usage</h3>

<pre><code class='language-R'>os.sort(x, y, method = "linear", n.pred = ncol(x), thresh = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="os.sort_+3A_x">x</code></td>
<td>
<p>A data matrix (raw: samples, col: predictors).</p>
</td></tr>
<tr><td><code id="os.sort_+3A_y">y</code></td>
<td>
<p>A vector of target value.</p>
</td></tr>
<tr><td><code id="os.sort_+3A_method">method</code></td>
<td>
<p>A string to specify the regression function for calculating R-squared values.
&quot;linear&quot; (default), &quot;quadratic&quot; or &quot;cubic&quot; function can be specified.</p>
</td></tr>
<tr><td><code id="os.sort_+3A_n.pred">n.pred</code></td>
<td>
<p>The number of predictors to be leaved in prediction model (default: ncol(x)).</p>
</td></tr>
<tr><td><code id="os.sort_+3A_thresh">thresh</code></td>
<td>
<p>The lower threshold of R-squared value to be leaved in prediction model (default: 1).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data matrix (raw: samples, col: sorted predictors)
</p>


<h3>Author(s)</h3>

<p>Takahiko Koizumi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pinus)
train &lt;- os.clean(Pinus$train)
target &lt;- Pinus$target
cor(target, train[, 1])

train &lt;- os.sort(train, target, thresh = 0.5)
cor(target, train[, 1])
</code></pre>

<hr>
<h2 id='Pinus'>Transcriptomes of Pinus roots under a Temperature Gradient</h2><span id='topic+Pinus'></span>

<h3>Description</h3>

<p>This dataset gives the TPM values of 200 selected genes obtained from 60 Pinus root
samples (30 samples each for training and test data) under a temperature gradient,
generated by RNA-seq.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Pinus</code></pre>


<h3>Details</h3>

<p>A gene expression data matrix of 30 root samples of P. thunbergii under five
temperature conditions (8, 13, 18, 23, 28 Â°C) with six biological replicates is in
the first element of the list.
</p>
<p>A gene expression data matrix of another 30 root samples of P. thunbergii under
the same condition is in the second one.
</p>
<p>Temperature conditions where 30 root samples in each data matrix were generated are
in the third one.
</p>
<p>Gene expressions are normalized in the TPM value.
</p>


<h3>Source</h3>

<p>original (not published)</p>


<h3>References</h3>

<p>original (not published)</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
