<!DOCTYPE html><html><head><title>Help for package OmicNavigator</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {OmicNavigator}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#addAnnotations'><p>Add annotations</p></a></li>
<li><a href='#addAssays'><p>Add assays</p></a></li>
<li><a href='#addBarcodes'><p>Add barcode plot metadata</p></a></li>
<li><a href='#addEnrichments'><p>Add enrichment results</p></a></li>
<li><a href='#addEnrichmentsLinkouts'><p>Add linkouts to external resources in the enrichments table</p></a></li>
<li><a href='#addFeatures'><p>Add feature metadata</p></a></li>
<li><a href='#addMapping'><p>Add mapping object</p></a></li>
<li><a href='#addMetaFeatures'><p>Add meta-feature metadata</p></a></li>
<li><a href='#addMetaFeaturesLinkouts'><p>Add linkouts to external resources in the metaFeatures table</p></a></li>
<li><a href='#addModels'><p>Add models</p></a></li>
<li><a href='#addOverlaps'><p>Add overlaps between annotation gene sets</p></a></li>
<li><a href='#addPlots'><p>Add custom plotting functions</p></a></li>
<li><a href='#addReports'><p>Add reports</p></a></li>
<li><a href='#addResults'><p>Add inference results</p></a></li>
<li><a href='#addResultsLinkouts'><p>Add linkouts to external resources in the results table</p></a></li>
<li><a href='#addSamples'><p>Add sample metadata</p></a></li>
<li><a href='#addTests'><p>Add tests</p></a></li>
<li><a href='#basal.vs.lp'><p>basal.vs.lp from Bioconductor workflow RNAseq123</p></a></li>
<li><a href='#basal.vs.ml'><p>basal.vs.ml from Bioconductor workflow RNAseq123</p></a></li>
<li><a href='#cam.BasalvsLP'><p>cam.BasalvsLP from Bioconductor workflow RNAseq123</p></a></li>
<li><a href='#cam.BasalvsML'><p>cam.BasalvsML from Bioconductor workflow RNAseq123</p></a></li>
<li><a href='#combineStudies'><p>Combine two or more studies</p></a></li>
<li><a href='#createStudy'><p>Create a study</p></a></li>
<li><a href='#exportStudy'><p>Export a study</p></a></li>
<li><a href='#getAnnotations'><p>Get annotations from a study</p></a></li>
<li><a href='#getAssays'><p>Get assays from a study</p></a></li>
<li><a href='#getBarcodeData'><p>Get data for barcode and violin plots</p></a></li>
<li><a href='#getBarcodes'><p>Get barcodes from a study</p></a></li>
<li><a href='#getEnrichments'><p>Get enrichments from a study</p></a></li>
<li><a href='#getEnrichmentsIntersection'><p>getEnrichmentsIntersection</p></a></li>
<li><a href='#getEnrichmentsLinkouts'><p>Get enrichments table linkouts from a study</p></a></li>
<li><a href='#getEnrichmentsNetwork'><p>Get enrichments network from a study</p></a></li>
<li><a href='#getEnrichmentsTable'><p>Get enrichments table from a study</p></a></li>
<li><a href='#getEnrichmentsUpset'><p>getEnrichmentsUpset</p></a></li>
<li><a href='#getFavicons'><p>Get favicon URLs for table linkouts</p></a></li>
<li><a href='#getFeatures'><p>Get features from a study</p></a></li>
<li><a href='#getInstalledStudies'><p>Get installed OmicNavigator studies</p></a></li>
<li><a href='#getLinkFeatures'><p>Get the shared features in a network link</p></a></li>
<li><a href='#getMapping'><p>Get mapping object from a study</p></a></li>
<li><a href='#getMetaFeatures'><p>Get metaFeatures from a study</p></a></li>
<li><a href='#getMetaFeaturesLinkouts'><p>Get metaFeatures table linkouts from a study</p></a></li>
<li><a href='#getMetaFeaturesTable'><p>Get metaFeatures for a given feature</p></a></li>
<li><a href='#getModels'><p>Get models from a study</p></a></li>
<li><a href='#getNodeFeatures'><p>Get the features in a network node</p></a></li>
<li><a href='#getOverlaps'><p>Get overlaps from a study</p></a></li>
<li><a href='#getPackageVersion'><p>Get version of OmicNavigator package</p></a></li>
<li><a href='#getPlots'><p>Get plots from a study</p></a></li>
<li><a href='#getPlottingData'><p>Get plotting data</p></a></li>
<li><a href='#getReportLink'><p>Get link to report</p></a></li>
<li><a href='#getReports'><p>Get reports from a study</p></a></li>
<li><a href='#getResults'><p>Get results from a study</p></a></li>
<li><a href='#getResultsIntersection'><p>getResultsIntersection</p></a></li>
<li><a href='#getResultsLinkouts'><p>Get results table linkouts from a study</p></a></li>
<li><a href='#getResultsTable'><p>Get results table from a study</p></a></li>
<li><a href='#getResultsUpset'><p>getResultsUpset</p></a></li>
<li><a href='#getSamples'><p>Get samples from a study</p></a></li>
<li><a href='#getTests'><p>Get tests from a study</p></a></li>
<li><a href='#getUpsetCols'><p>getUpsetCols</p></a></li>
<li><a href='#group'><p>group from Bioconductor workflow RNAseq123</p></a></li>
<li><a href='#importStudy'><p>Import a study package</p></a></li>
<li><a href='#installApp'><p>Install the OmicNavigator web app</p></a></li>
<li><a href='#installStudy'><p>Install a study as an R package</p></a></li>
<li><a href='#lane'><p>lane from Bioconductor workflow RNAseq123</p></a></li>
<li><a href='#lcpm'><p>lcpm from Bioconductor workflow RNAseq123</p></a></li>
<li><a href='#listStudies'><p>List available studies and their metadata</p></a></li>
<li><a href='#Mm.c2'><p>Mm.c2 from Bioconductor workflow RNAseq123</p></a></li>
<li><a href='#OmicNavigator-package'><p>OmicNavigator</p></a></li>
<li><a href='#plotStudy'><p>Plot a feature using a custom plotting function</p></a></li>
<li><a href='#removeStudy'><p>Remove an installed study R package</p></a></li>
<li><a href='#samplenames'><p>samplenames from Bioconductor workflow RNAseq123</p></a></li>
<li><a href='#shared-add'><p>Shared parameters for add functions</p></a></li>
<li><a href='#shared-data'><p>Shared sections for data objects</p></a></li>
<li><a href='#shared-get'><p>Shared parameters for get functions</p></a></li>
<li><a href='#shared-upset'><p>Shared parameters for upset functions</p></a></li>
<li><a href='#startApp'><p>Start app on local machine</p></a></li>
<li><a href='#summary.onStudy'><p>Summarize elements of OmicNavigator study</p></a></li>
<li><a href='#validateStudy'><p>Validate a study</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Open-Source Software for 'Omic' Data Analysis and Visualization</td>
</tr>
<tr>
<td>Description:</td>
<td>
  A tool for interactive exploration of the results from 'omics'
  experiments to facilitate novel discoveries from high-throughput biology. The
  software includes R functions for the 'bioinformatician' to deposit study
  metadata and the outputs from statistical analyses (e.g. differential
  expression, enrichment). These results are then exported to an interactive
  JavaScript dashboard that can be interrogated on the user's local machine or
  deployed online to be explored by collaborators. The dashboard includes
  'sortable' tables, interactive plots including network visualization, and
  fine-grained filtering based on statistical significance.</td>
</tr>
<tr>
<td>Version:</td>
<td>1.13.13</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/abbvie-external/OmicNavigator">https://github.com/abbvie-external/OmicNavigator</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/abbvie-external/OmicNavigator/issues">https://github.com/abbvie-external/OmicNavigator/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>License_restricts_use:</td>
<td>no</td>
</tr>
<tr>
<td>License_is_FOSS:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table (&ge; 1.12.4), graphics, jsonlite, stats, tools,
utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>faviconPlease, ggplot2, opencpu, plotly, tinytest (&ge; 1.2.3),
ttdo (&ge; 0.0.6), UpSetR</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-25 19:39:29 UTC; jdblischak</td>
</tr>
<tr>
<td>Author:</td>
<td>Terrence Ernst [aut] (Web application),
  John Blischak <a href="https://orcid.org/0000-0003-2634-9879"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Paul Nordlund [aut] (Web application),
  Justin Moore [aut] (UpSet-related functions and web application),
  Joe Dalen [aut] (Barcode functionality and web application),
  Akshay Bhamidipati [aut] (Web application),
  Brett Engelmann [aut],
  Marco Curado [aut] (Improved plotting capabilities),
  Joe LoGrasso [aut] (Support for plotly),
  AbbVie Inc. [cph, fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>John Blischak &lt;jdblischak@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-25 20:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='addAnnotations'>Add annotations</h2><span id='topic+addAnnotations'></span>

<h3>Description</h3>

<p>Add annotations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addAnnotations(study, annotations, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addAnnotations_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addAnnotations_+3A_annotations">annotations</code></td>
<td>
<p>The annotations used for the enrichment analyses. The
input is a nested list. The top-level list contains one entry per
annotation database, e.g. reactome. The names correspond to the name of
each annotation database. Each of these elements should be a list that
contains more information about each annotation database. Specifically the
sublist should contain 1) <code>description</code>, a character vector that
describes the resource, 2) <code>featureID</code>, the name of the column in the
features table that was used for the enrichment analysis, and 3) <code>terms</code>,
a list of annotation terms. The names of <code>terms</code> sublist correspond to
the name of the annotation terms. Each of the annotation terms should be a
character vector of featureIDs.</p>
</td></tr>
<tr><td><code id="addAnnotations_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>

<hr>
<h2 id='addAssays'>Add assays</h2><span id='topic+addAssays'></span>

<h3>Description</h3>

<p>Add assays
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addAssays(study, assays, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addAssays_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addAssays_+3A_assays">assays</code></td>
<td>
<p>The assays from the study. The input object is a list of data
frames (one per model). The row names should correspond to the featureIDs
(<code><a href="#topic+addFeatures">addFeatures</a></code>). The column names should correspond to the
sampleIDs (<code><a href="#topic+addSamples">addSamples</a></code>). The data frame should only contain
numeric values. To share a data frame across multiple models, use the
modelID &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="addAssays_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>

<hr>
<h2 id='addBarcodes'>Add barcode plot metadata</h2><span id='topic+addBarcodes'></span>

<h3>Description</h3>

<p>The app can display a barcode plot of the enrichment results for a given
annotation term. The metadata in 'barcodes' instructs the app how to create
and label the barcode plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addBarcodes(study, barcodes, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addBarcodes_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addBarcodes_+3A_barcodes">barcodes</code></td>
<td>
<p>The metadata variables that describe the barcode plot.
The input object is a list of lists (one per model). Each sublist must
contain the element <code>statistic</code>, which is the column name in the
results table to use to construct the barcode plot. Each sublist may
additionally contain any of the following optional elements:
1) <code>absolute</code> - Should the statistic be converted to its absolute
value (default is <code>TRUE</code>).
2) <code>logFoldChange</code> - The column name in the results table that contains
the log fold change values.
3) <code>labelStat</code> - The x-axis label to describe the statistic.
4) <code>labelLow</code> - The left-side label to describe low values of the statistic.
5) <code>labelHigh</code> - The right-side label to describe high values of the statistic.
6) <code>featureDisplay</code> - The feature variable to use to label the barcode plot
on hover.
To share metadata across multiple models, use the modelID &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="addBarcodes_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>

<hr>
<h2 id='addEnrichments'>Add enrichment results</h2><span id='topic+addEnrichments'></span>

<h3>Description</h3>

<p>Add enrichment results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addEnrichments(study, enrichments, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addEnrichments_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addEnrichments_+3A_enrichments">enrichments</code></td>
<td>
<p>The enrichment results from each model. The input is a
nested named list. The names of the list correspond to the model names.
Each list element should be a list of the annotation databases tested
(<code><a href="#topic+addAnnotations">addAnnotations</a></code>). The names of the list correspond to the
annotation databases. Each list element should be another list of tests
(<code><a href="#topic+addTests">addTests</a></code>). The names correspond to the tests performed. Each
of these elements should be a data frame with enrichment results. Each
table must contain the following columns: &quot;termID&quot;, &quot;description&quot;,
&quot;nominal&quot; (the nominal statistics), and &quot;adjusted&quot; (the statistics after
adjusting for multiple testing). Any additional columns are ignored.</p>
</td></tr>
<tr><td><code id="addEnrichments_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>

<hr>
<h2 id='addEnrichmentsLinkouts'>Add linkouts to external resources in the enrichments table</h2><span id='topic+addEnrichmentsLinkouts'></span>

<h3>Description</h3>

<p>You can provide additional information on the annotation terms in your study
by providing linkouts to external resources. These will be embedded directly
in the enrichments table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addEnrichmentsLinkouts(study, enrichmentsLinkouts, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addEnrichmentsLinkouts_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addEnrichmentsLinkouts_+3A_enrichmentslinkouts">enrichmentsLinkouts</code></td>
<td>
<p>The URL patterns that describe linkouts to
external resources (see Details below). The input object is a named list.
The names of the list correspond to the annotation names. Each element of
the list is a character vector of linkouts for that annotationID.</p>
</td></tr>
<tr><td><code id="addEnrichmentsLinkouts_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each linkout, the URL pattern you provide will be concatenated with the
value of the termID column. As an example, if you used the annotation
database <a href="https://amigo.geneontology.org/">AmiGO 2</a> for your enrichments
analysis, you can provide a linkout for each termID using the following
pattern:
</p>
<pre>go = "https://amigo.geneontology.org/amigo/term/"</pre>
<p>As another example, if you used the annotation database
<a href="https://reactome.org/">Reactome</a> for your enrichments analysis, you can
provide a linkout for each termID using the following pattern:
</p>
<pre>reactome = "https://reactome.org/content/detail/"</pre>
<p>Note that you can provide more than one linkout per termID.
</p>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addAnnotations">addAnnotations</a></code>, <code><a href="#topic+addEnrichments">addEnrichments</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  study &lt;- createStudy("example")
  enrichmentsLinkouts &lt;- list(
    gobp = c("https://amigo.geneontology.org/amigo/term/",
             "https://www.ebi.ac.uk/QuickGO/term/"),
    reactome = "https://reactome.org/content/detail/"
  )
  study &lt;- addEnrichmentsLinkouts(study, enrichmentsLinkouts)

</code></pre>

<hr>
<h2 id='addFeatures'>Add feature metadata</h2><span id='topic+addFeatures'></span>

<h3>Description</h3>

<p>Add feature metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addFeatures(study, features, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addFeatures_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addFeatures_+3A_features">features</code></td>
<td>
<p>The metadata variables that describe the features in the
study. The input object is a list of data frames (one per model). The first
column of each data frame is used as the featureID, so it must contain
unique values. To share a data frame across multiple models, use the
modelID &quot;default&quot;. All columns will be coerced to character strings.</p>
</td></tr>
<tr><td><code id="addFeatures_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>

<hr>
<h2 id='addMapping'>Add mapping object</h2><span id='topic+addMapping'></span>

<h3>Description</h3>

<p>Add mapping object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addMapping(study, mapping, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addMapping_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addMapping_+3A_mapping">mapping</code></td>
<td>
<p>Feature IDs from models. The input object is a list of named
data frames. For each data frame, column names indicate model names while
rows indicate featureIDs per model. Features with same index position across
columns are treated as mapped across models. For each model, feature IDs must
match feature IDs available in the results object of the respective model.
1:N relationships are allowed.
</p>
<p>Mapping list elements are required to be named as 'default' or after a model
name as provided in addModels(). If a single data frame is provided, this
list element is recommended to be named 'default'. For multiple list
elements, each with its own data frame, list elements should be named after
model name(s) (a single element may still be  named 'default'). In that case,
when navigating in ON front-end (FE), mapping element related to the selected
model in the FE will be used in multimodel plots. If a selected model in FE
does not have a corresponding mapping list element, it may still use the
mapping list element called 'default' if this is available.
</p>
<p>E.g., if in a study there are models &quot;transcriptomics&quot; and &quot;proteomics&quot; and
the user wants to create a plot based on data from both, a mapping list
should be provided with addMapping(). In this case, the mapping list element
may be named 'default'. This should contain a data frame with column names
'transcriptomics' and 'proteomics', where feature IDs that map across models
are found in the same row.</p>
</td></tr>
<tr><td><code id="addMapping_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getPlottingData">getPlottingData</a></code>, <code><a href="#topic+plotStudy">plotStudy</a></code>
</p>

<hr>
<h2 id='addMetaFeatures'>Add meta-feature metadata</h2><span id='topic+addMetaFeatures'></span>

<h3>Description</h3>

<p>The meta-features table is useful anytime there are metadata variables that
cannot be mapped 1:1 to your features. For example, a peptide may be
associated with multiple proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addMetaFeatures(study, metaFeatures, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addMetaFeatures_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addMetaFeatures_+3A_metafeatures">metaFeatures</code></td>
<td>
<p>The metadata variables that describe the meta-features in
the study. The input object is a list of data frames (one per model). The
first column of each data frame is used as the featureID, so it must
contain the same IDs as the corresponding features data frame
(<code><a href="#topic+addFeatures">addFeatures</a></code>). To share a data frame across multiple models,
use the modelID &quot;default&quot;. All columns will be coerced to character
strings.</p>
</td></tr>
<tr><td><code id="addMetaFeatures_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>

<hr>
<h2 id='addMetaFeaturesLinkouts'>Add linkouts to external resources in the metaFeatures table</h2><span id='topic+addMetaFeaturesLinkouts'></span>

<h3>Description</h3>

<p>You can provide additional information on the metaFeatures in your study by
providing linkouts to external resources. These will be embedded directly in
the metaFeatures table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addMetaFeaturesLinkouts(study, metaFeaturesLinkouts, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addMetaFeaturesLinkouts_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addMetaFeaturesLinkouts_+3A_metafeatureslinkouts">metaFeaturesLinkouts</code></td>
<td>
<p>The URL patterns that describe linkouts to external
resources (see Details below). The input object is a nested named list. The
names of the list correspond to the model names. Each element of the list
is a named list of character vectors. The names of this nested list must
correspond to the column names of the matching metaFeatures table (<code><a href="#topic+addMetaFeatures">addMetaFeatures</a></code>). To share
linkouts across multiple models, use the modelID &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="addMetaFeaturesLinkouts_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each linkout, the URL pattern you provide will be concatenated with the
value of that column for each row. As an example, if your metaFeatures table
included a column named <code>"ensembl"</code> that contained the Ensembl Gene ID
for each feature, you could create a linkout to Ensembl using the following
pattern:
</p>
<pre>ensembl = "https://ensembl.org/Homo_sapiens/Gene/Summary?g="</pre>
<p>As another example, if you had a column named <code>"entrez"</code> that contained
the Entrez Gene ID  for each feature, you could create a linkout to Entrez
using the following pattern:
</p>
<pre>entrez = "https://www.ncbi.nlm.nih.gov/gene/"</pre>
<p>Note that you can provide more than one linkout per column.
</p>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addMetaFeatures">addMetaFeatures</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  study &lt;- createStudy("example")
  metaFeaturesLinkouts &lt;- list(
    default = list(
      ensembl = c("https://ensembl.org/Homo_sapiens/Gene/Summary?g=",
                  "https://www.genome.ucsc.edu/cgi-bin/hgGene?hgg_gene="),
      entrez = "https://www.ncbi.nlm.nih.gov/gene/"
    )
  )
  study &lt;- addMetaFeaturesLinkouts(study, metaFeaturesLinkouts)

</code></pre>

<hr>
<h2 id='addModels'>Add models</h2><span id='topic+addModels'></span>

<h3>Description</h3>

<p>Add models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addModels(study, models, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addModels_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addModels_+3A_models">models</code></td>
<td>
<p>The models analyzed in the study. The input is a named list.
The names correspond to the names of the models. The elements correspond to
the descriptions of the models. Alternatively, instead of a single
character string, you can provide a list of metadata fields about each
model. The field &quot;description&quot; will be used to derive the tooltip displayed
in the app.</p>
</td></tr>
<tr><td><code id="addModels_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  study &lt;- createStudy("example")
  models &lt;- list(
    model_01 = "Name of first model",
    model_02 = "Name of second model"
  )
  study &lt;- addModels(study, models)

  # Alternative: provide additional metadata about each model
  models &lt;- list(
    model_01 = list(
      description = "Name of first model",
      data_type = "transcriptomics"
    ),
    model_02 = list(
      description = "Name of second model",
      data_type = "proteomics"
    )
  )

</code></pre>

<hr>
<h2 id='addOverlaps'>Add overlaps between annotation gene sets</h2><span id='topic+addOverlaps'></span>

<h3>Description</h3>

<p>The app's network view of the enrichments results requires pairwise overlap
metrics between all the terms of each annotation in order to draw the edges
between the nodes/terms. These overlaps are calculated automatically when
installing or exporting an OmicNavigator study. If you'd like, you can
manually calculate these pairwise overlaps by calling <code>addOverlaps</code>
prior to installing or exporting your study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addOverlaps(study, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addOverlaps_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addOverlaps_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>

<hr>
<h2 id='addPlots'>Add custom plotting functions</h2><span id='topic+addPlots'></span>

<h3>Description</h3>

<p>Include custom plots that the app will display when a feature is selected by
the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addPlots(study, plots, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addPlots_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addPlots_+3A_plots">plots</code></td>
<td>
<p>Custom plotting functions for the study. The input object is a
nested list. The first list corresponds to the modelID(s). The second list
corresponds to the name(s) of the function(s) defined in the current R
session. The third list provides metadata to describe each plot. The only
required metadata element is <code>displayName</code>, which controls how the
plot will be named in the app. You are encouraged to also specify the
<code>plotType</code>, e.g. <code>"singleFeature"</code>, <code>"multiFeature"</code>,
<code>"multiTest"</code>, <code>"multiModel"</code>. PlotType accepts vector of
entries, whenever applicable, e.g., plotType = c(<code>"multiFeature"</code>,
<code>"multiTest"</code>). If you do not specify the <code>plotType</code>, the plot
will be assumed to be <code>"singleFeature"</code> and <code>"singleTest"</code>.
Optionally, if the plotting function requires external packages, these can
be defined in the element <code>packages</code>. To share plots across multiple
models, use the modelID &quot;default&quot;. To add a plotting function that returns
an interactive plotly plot, add &quot;plotly&quot; to the <code>plotType</code> vector.</p>
</td></tr>
<tr><td><code id="addPlots_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Custom plotting functions are passed a list of data frames: <code>assays</code>
with the measurements, <code>features</code> with the feature data,
<code>samples</code> with the sample data, and <code>results</code> with test results
data. Note that <code>assays</code>, <code>features</code> and <code>results</code>
only include data for the specified featureID(s) (and
re-ordered so their rows match). Thus your custom plotting function must have
at least one argument. It can have additional arguments if you wish, but
these must be provided with default values, because <code>plotStudy</code> only
passes the plotting data to the first argument.
</p>
<p>Note that any ggplot2 plots will require extra care. This is because the
plotting code will be inserted into a study package, and thus must follow the
<a href="https://ggplot2.tidyverse.org/articles/ggplot2-in-packages.html#using-aes-and-vars-in-a-package-function-1">best
practices for using ggplot2 within packages</a>. Specifically, when you refer to
columns of the data frame, e.g. <code>aes(x = group)</code>, you need to prefix it
with <code>.data$</code>, so that it becomes <code>aes(x = .data$group)</code>.
Fortunately this latter code will also run fine as you interactively develop
the function.
</p>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getPlottingData">getPlottingData</a></code>, <code><a href="#topic+plotStudy">plotStudy</a></code>
</p>

<hr>
<h2 id='addReports'>Add reports</h2><span id='topic+addReports'></span>

<h3>Description</h3>

<p>You can include reports of the analyses you performed to generate the
results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addReports(study, reports, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addReports_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addReports_+3A_reports">reports</code></td>
<td>
<p>The analysis report(s) that explain how the study results were
generated. The input object is a list of character vectors (one per model).
Each element should be either a URL or a path to a file on your computer.
If it is a path to a file, this file will be included in the exported study
package. To share a report across multiple models, use the modelID
&quot;default&quot;.</p>
</td></tr>
<tr><td><code id="addReports_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>

<hr>
<h2 id='addResults'>Add inference results</h2><span id='topic+addResults'></span>

<h3>Description</h3>

<p>Add inference results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addResults(study, results, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addResults_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addResults_+3A_results">results</code></td>
<td>
<p>The inference results from each model. The input is a
nested named list. The names of the list correspond to the model names.
Each element in the list should be a list of data frames with inference
results, one for each test. In each data frame, the featureID must be in
the first column, and all other columns must be numeric.</p>
</td></tr>
<tr><td><code id="addResults_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>

<hr>
<h2 id='addResultsLinkouts'>Add linkouts to external resources in the results table</h2><span id='topic+addResultsLinkouts'></span>

<h3>Description</h3>

<p>You can provide additional information on the features in your study by
providing linkouts to external resources. These will be embedded directly in
the results table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addResultsLinkouts(study, resultsLinkouts, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addResultsLinkouts_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addResultsLinkouts_+3A_resultslinkouts">resultsLinkouts</code></td>
<td>
<p>The URL patterns that describe linkouts to external
resources (see Details below). The input object is a nested named list. The
names of the list correspond to the model names. Each element of the list
is a named list of character vectors. The names of this nested list must
correspond to the column names of the matching features table. To share
linkouts across multiple models, use the modelID &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="addResultsLinkouts_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each linkout, the URL pattern you provide will be concatenated with the
value of that column for each row. As an example, if your features table
included a column named <code>"ensembl"</code> that contained the Ensembl Gene ID
for each feature, you could create a linkout to Ensembl using the following
pattern:
</p>
<pre>ensembl = "https://ensembl.org/Homo_sapiens/Gene/Summary?g="</pre>
<p>As another example, if you had a column named <code>"entrez"</code> that contained
the Entrez Gene ID  for each feature, you could create a linkout to Entrez
using the following pattern:
</p>
<pre>entrez = "https://www.ncbi.nlm.nih.gov/gene/"</pre>
<p>Note that you can provide more than one linkout per column.
</p>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addFeatures">addFeatures</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  study &lt;- createStudy("example")
  resultsLinkouts &lt;- list(
    default = list(
      ensembl = c("https://ensembl.org/Homo_sapiens/Gene/Summary?g=",
                  "https://www.genome.ucsc.edu/cgi-bin/hgGene?hgg_gene="),
      entrez = "https://www.ncbi.nlm.nih.gov/gene/"
    )
  )
  study &lt;- addResultsLinkouts(study, resultsLinkouts)

</code></pre>

<hr>
<h2 id='addSamples'>Add sample metadata</h2><span id='topic+addSamples'></span>

<h3>Description</h3>

<p>Add sample metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addSamples(study, samples, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addSamples_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addSamples_+3A_samples">samples</code></td>
<td>
<p>The metadata variables that describe the samples in the study.
The input object is a named list of data frames (one per model). The first
column of each data frame is used as the sampleID, so it must contain
unique values. To share a data frame across multiple models, use the
modelID &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="addSamples_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>

<hr>
<h2 id='addTests'>Add tests</h2><span id='topic+addTests'></span>

<h3>Description</h3>

<p>Add tests
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addTests(study, tests, reset = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addTests_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="addTests_+3A_tests">tests</code></td>
<td>
<p>The tests from the study. The input object is a list of lists.
Each element of the top-level list is a model. The names should be the
modelIDs. For each modelID, each element of the nested list is a test. The
names should be the testIDs. The value should be a single character string
describing the testID. To share tests across multiple models, use the
modelID &quot;default&quot;. Instead of a single character string, you can provide a
list of metadata fields about each test. The field &quot;description&quot; will be
used to derive the tooltip displayed in the app.</p>
</td></tr>
<tr><td><code id="addTests_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  study &lt;- createStudy("example")
  tests &lt;- list(
    default = list(
      test_01 = "Name of first test",
      test_02 = "Name of second test"
    )
  )
  study &lt;- addTests(study, tests)

  # Alternative: provide additional metadata about each test
  tests &lt;- list(
    default = list(
      test_01 = list(
        description = "Name of first test",
        comparison_type = "treatment vs control",
        effect_size = "beta"
      ),
      test_02 = list(
        description = "Name of second test",
        comparison_type = "treatment vs control",
        effect_size = "logFC"
      )
    )
  )

</code></pre>

<hr>
<h2 id='basal.vs.lp'>basal.vs.lp from Bioconductor workflow RNAseq123</h2><span id='topic+basal.vs.lp'></span>

<h3>Description</h3>

<p>A subset of the object <code>basal.vs.lp</code> from Bioconductor workflow RNAseq123.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basal.vs.lp
</code></pre>


<h3>Format</h3>

<p>A data frame with 24 rows and 8 columns:
</p>

<dl>
<dt>ENTREZID</dt><dd><p>Entrez ID of mouse gene</p>
</dd>
<dt>SYMBOL</dt><dd><p>Symbol of mouse gene</p>
</dd>
<dt>TXCHROM</dt><dd><p>Chromosome location of mouse gene</p>
</dd>
<dt>logFC</dt><dd><p>Log fold change</p>
</dd>
<dt>AveExpr</dt><dd><p>Average expression level of the gene across all samples</p>
</dd>
<dt>t</dt><dd><p>Moderated t-statistic</p>
</dd>
<dt>P.Value</dt><dd><p>p-value</p>
</dd>
<dt>adj.P.Val</dt><dd><p>Adjusted p-value</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html">https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html</a>
</p>


<h3>References</h3>

<p>Law CW, Alhamdoosh M, Su S, Dong X, Tian L, Smyth GK, Ritchie ME.
<a href="https://f1000research.com/articles/5-1408/v3">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR [version 3; peer review: 3 approved].</a>
F1000Research 2018, 5:1408
<a href="https://doi.org/10.12688/f1000research.9005.3">doi:10.12688/f1000research.9005.3</a>
</p>
<p>Sheridan, J.M., Ritchie, M.E., Best, S.A. et al.
<a href="https://bmccancer.biomedcentral.com/articles/10.1186/s12885-015-1187-z">A pooled shRNA screen for regulators of primary mammary stem and progenitor cells identifies roles for <em>Asap1</em> and <em>Prox1</em>.</a>
BMC Cancer 2015, 15:221
<a href="https://doi.org/10.1186/s12885-015-1187-z">doi:10.1186/s12885-015-1187-z</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(basal.vs.lp)
  str(basal.vs.lp)

</code></pre>

<hr>
<h2 id='basal.vs.ml'>basal.vs.ml from Bioconductor workflow RNAseq123</h2><span id='topic+basal.vs.ml'></span>

<h3>Description</h3>

<p>A subset of the object <code>basal.vs.ml</code> from Bioconductor workflow RNAseq123.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>basal.vs.ml
</code></pre>


<h3>Format</h3>

<p>A data frame with 24 rows and 8 columns:
</p>

<dl>
<dt>ENTREZID</dt><dd><p>Entrez ID of mouse gene</p>
</dd>
<dt>SYMBOL</dt><dd><p>Symbol of mouse gene</p>
</dd>
<dt>TXCHROM</dt><dd><p>Chromosome location of mouse gene</p>
</dd>
<dt>logFC</dt><dd><p>Log fold change</p>
</dd>
<dt>AveExpr</dt><dd><p>Average expression level of the gene across all samples</p>
</dd>
<dt>t</dt><dd><p>Moderated t-statistic</p>
</dd>
<dt>P.Value</dt><dd><p>p-value</p>
</dd>
<dt>adj.P.Val</dt><dd><p>Adjusted p-value</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html">https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html</a>
</p>


<h3>References</h3>

<p>Law CW, Alhamdoosh M, Su S, Dong X, Tian L, Smyth GK, Ritchie ME.
<a href="https://f1000research.com/articles/5-1408/v3">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR [version 3; peer review: 3 approved].</a>
F1000Research 2018, 5:1408
<a href="https://doi.org/10.12688/f1000research.9005.3">doi:10.12688/f1000research.9005.3</a>
</p>
<p>Sheridan, J.M., Ritchie, M.E., Best, S.A. et al.
<a href="https://bmccancer.biomedcentral.com/articles/10.1186/s12885-015-1187-z">A pooled shRNA screen for regulators of primary mammary stem and progenitor cells identifies roles for <em>Asap1</em> and <em>Prox1</em>.</a>
BMC Cancer 2015, 15:221
<a href="https://doi.org/10.1186/s12885-015-1187-z">doi:10.1186/s12885-015-1187-z</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(basal.vs.ml)
  str(basal.vs.ml)

</code></pre>

<hr>
<h2 id='cam.BasalvsLP'>cam.BasalvsLP from Bioconductor workflow RNAseq123</h2><span id='topic+cam.BasalvsLP'></span>

<h3>Description</h3>

<p>A subset of the object <code>cam.BasalvsLP</code> from Bioconductor workflow RNAseq123.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cam.BasalvsLP
</code></pre>


<h3>Format</h3>

<p>A data frame with 4 rows and 4 columns:
</p>

<dl>
<dt>NGenes</dt><dd><p>Number of genes in each term</p>
</dd>
<dt>Direction</dt><dd><p>Direction of the enrichment</p>
</dd>
<dt>PValue</dt><dd><p>Nominal p-value</p>
</dd>
<dt>FDR</dt><dd><p>Multiple-testing adjusted p-value</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html">https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html</a>
</p>


<h3>References</h3>

<p>Law CW, Alhamdoosh M, Su S, Dong X, Tian L, Smyth GK, Ritchie ME.
<a href="https://f1000research.com/articles/5-1408/v3">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR [version 3; peer review: 3 approved].</a>
F1000Research 2018, 5:1408
<a href="https://doi.org/10.12688/f1000research.9005.3">doi:10.12688/f1000research.9005.3</a>
</p>
<p>Sheridan, J.M., Ritchie, M.E., Best, S.A. et al.
<a href="https://bmccancer.biomedcentral.com/articles/10.1186/s12885-015-1187-z">A pooled shRNA screen for regulators of primary mammary stem and progenitor cells identifies roles for <em>Asap1</em> and <em>Prox1</em>.</a>
BMC Cancer 2015, 15:221
<a href="https://doi.org/10.1186/s12885-015-1187-z">doi:10.1186/s12885-015-1187-z</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(cam.BasalvsLP)
  str(cam.BasalvsLP)

</code></pre>

<hr>
<h2 id='cam.BasalvsML'>cam.BasalvsML from Bioconductor workflow RNAseq123</h2><span id='topic+cam.BasalvsML'></span>

<h3>Description</h3>

<p>A subset of the object <code>cam.BasalvsML</code> from Bioconductor workflow RNAseq123.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cam.BasalvsML
</code></pre>


<h3>Format</h3>

<p>A data frame with 4 rows and 4 columns:
</p>

<dl>
<dt>NGenes</dt><dd><p>Number of genes in each term</p>
</dd>
<dt>Direction</dt><dd><p>Direction of the enrichment</p>
</dd>
<dt>PValue</dt><dd><p>Nominal p-value</p>
</dd>
<dt>FDR</dt><dd><p>Multiple-testing adjusted p-value</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html">https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html</a>
</p>


<h3>References</h3>

<p>Law CW, Alhamdoosh M, Su S, Dong X, Tian L, Smyth GK, Ritchie ME.
<a href="https://f1000research.com/articles/5-1408/v3">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR [version 3; peer review: 3 approved].</a>
F1000Research 2018, 5:1408
<a href="https://doi.org/10.12688/f1000research.9005.3">doi:10.12688/f1000research.9005.3</a>
</p>
<p>Sheridan, J.M., Ritchie, M.E., Best, S.A. et al.
<a href="https://bmccancer.biomedcentral.com/articles/10.1186/s12885-015-1187-z">A pooled shRNA screen for regulators of primary mammary stem and progenitor cells identifies roles for <em>Asap1</em> and <em>Prox1</em>.</a>
BMC Cancer 2015, 15:221
<a href="https://doi.org/10.1186/s12885-015-1187-z">doi:10.1186/s12885-015-1187-z</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(cam.BasalvsML)
  str(cam.BasalvsML)

</code></pre>

<hr>
<h2 id='combineStudies'>Combine two or more studies</h2><span id='topic+combineStudies'></span>

<h3>Description</h3>

<p>Create a new OmicNavigator study by combining two or more existing study
objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combineStudies(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combineStudies_+3A_...">...</code></td>
<td>
<p>Two or more objects of class <code>onStudy</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a convenience function to quickly and conveniently combine studies.
However, it is naive, and you will likely need to edit the new study after
combining. When there are conflicting elements (e.g. different study names or
different maintainers), then the value for the latter study is kept. As a
concrete example, if you combined 5 studies, the name of the combined study
would be the name of the 5th study.
</p>
<p>The behavior is more complex for study elements that are nested lists of data
frames (e.g. results). If the 5 studies included a results table for the same
modelID/testID combination, then only the results from the 5th study would be
retained. However, if they each defined a different modelID, then the results
for all 5 modelIDs would be included in the combined study. Please note that
you should be extra cautious in the situation where the studies have the same
modelID/testID combination. Ideally they should all have the same column
names. Since a data frame is technically a list, the workhorse function
<code><a href="utils.html#topic+modifyList">modifyList</a></code> will retain any uniquely named
columns from earlier studies along with the columns from the final study.
</p>
<p>Note that as a shortcut you can also combine studies using the S3 method
<code><a href="base.html#topic+c">c</a></code>.
</p>
<p>If a study you would like to combine is already installed, you can convert it
to a study object by importing it with <code><a href="#topic+importStudy">importStudy</a></code>.
</p>


<h3>Value</h3>

<p>Returns a new combined OmicNavigator study object, which is a named
nested list with class <code>onStudy</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+createStudy">createStudy</a></code>,
<code><a href="#topic+importStudy">importStudy</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Define threee study objects
studyOne &lt;- createStudy(name = "One",
                        description = "First study",
                        studyMeta = list(metafield1 = "metavalue1"))

studyTwo &lt;- createStudy(name = "Two",
                        description = "Second study",
                        maintainer = "The Maintainer",
                        studyMeta = list(metafield2 = "metavalue2"))

studyThree &lt;- createStudy(name = "Three",
                          description = "Third study",
                          studyMeta = list(metafield3 = "metavalue3"))

# Combine the three studies
combineStudies(studyOne, studyTwo, studyThree)

# Equivalently, can use c()
c(studyOne, studyTwo, studyThree)

</code></pre>

<hr>
<h2 id='createStudy'>Create a study</h2><span id='topic+createStudy'></span>

<h3>Description</h3>

<p>Create a new OmicNavigator study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>createStudy(
  name,
  description = name,
  samples = list(),
  features = list(),
  models = list(),
  assays = list(),
  tests = list(),
  annotations = list(),
  results = list(),
  enrichments = list(),
  metaFeatures = list(),
  plots = list(),
  mapping = list(),
  barcodes = list(),
  reports = list(),
  resultsLinkouts = list(),
  enrichmentsLinkouts = list(),
  metaFeaturesLinkouts = list(),
  version = NULL,
  maintainer = NULL,
  maintainerEmail = NULL,
  studyMeta = list()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="createStudy_+3A_name">name</code></td>
<td>
<p>Name of the study</p>
</td></tr>
<tr><td><code id="createStudy_+3A_description">description</code></td>
<td>
<p>Description of the study</p>
</td></tr>
<tr><td><code id="createStudy_+3A_samples">samples</code></td>
<td>
<p>The metadata variables that describe the samples in the study.
The input object is a named list of data frames (one per model). The first
column of each data frame is used as the sampleID, so it must contain
unique values. To share a data frame across multiple models, use the
modelID &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_features">features</code></td>
<td>
<p>The metadata variables that describe the features in the
study. The input object is a list of data frames (one per model). The first
column of each data frame is used as the featureID, so it must contain
unique values. To share a data frame across multiple models, use the
modelID &quot;default&quot;. All columns will be coerced to character strings.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_models">models</code></td>
<td>
<p>The models analyzed in the study. The input is a named list.
The names correspond to the names of the models. The elements correspond to
the descriptions of the models. Alternatively, instead of a single
character string, you can provide a list of metadata fields about each
model. The field &quot;description&quot; will be used to derive the tooltip displayed
in the app.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_assays">assays</code></td>
<td>
<p>The assays from the study. The input object is a list of data
frames (one per model). The row names should correspond to the featureIDs
(<code><a href="#topic+addFeatures">addFeatures</a></code>). The column names should correspond to the
sampleIDs (<code><a href="#topic+addSamples">addSamples</a></code>). The data frame should only contain
numeric values. To share a data frame across multiple models, use the
modelID &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_tests">tests</code></td>
<td>
<p>The tests from the study. The input object is a list of lists.
Each element of the top-level list is a model. The names should be the
modelIDs. For each modelID, each element of the nested list is a test. The
names should be the testIDs. The value should be a single character string
describing the testID. To share tests across multiple models, use the
modelID &quot;default&quot;. Instead of a single character string, you can provide a
list of metadata fields about each test. The field &quot;description&quot; will be
used to derive the tooltip displayed in the app.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_annotations">annotations</code></td>
<td>
<p>The annotations used for the enrichment analyses. The
input is a nested list. The top-level list contains one entry per
annotation database, e.g. reactome. The names correspond to the name of
each annotation database. Each of these elements should be a list that
contains more information about each annotation database. Specifically the
sublist should contain 1) <code>description</code>, a character vector that
describes the resource, 2) <code>featureID</code>, the name of the column in the
features table that was used for the enrichment analysis, and 3) <code>terms</code>,
a list of annotation terms. The names of <code>terms</code> sublist correspond to
the name of the annotation terms. Each of the annotation terms should be a
character vector of featureIDs.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_results">results</code></td>
<td>
<p>The inference results from each model. The input is a
nested named list. The names of the list correspond to the model names.
Each element in the list should be a list of data frames with inference
results, one for each test. In each data frame, the featureID must be in
the first column, and all other columns must be numeric.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_enrichments">enrichments</code></td>
<td>
<p>The enrichment results from each model. The input is a
nested named list. The names of the list correspond to the model names.
Each list element should be a list of the annotation databases tested
(<code><a href="#topic+addAnnotations">addAnnotations</a></code>). The names of the list correspond to the
annotation databases. Each list element should be another list of tests
(<code><a href="#topic+addTests">addTests</a></code>). The names correspond to the tests performed. Each
of these elements should be a data frame with enrichment results. Each
table must contain the following columns: &quot;termID&quot;, &quot;description&quot;,
&quot;nominal&quot; (the nominal statistics), and &quot;adjusted&quot; (the statistics after
adjusting for multiple testing). Any additional columns are ignored.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_metafeatures">metaFeatures</code></td>
<td>
<p>The metadata variables that describe the meta-features in
the study. The input object is a list of data frames (one per model). The
first column of each data frame is used as the featureID, so it must
contain the same IDs as the corresponding features data frame
(<code><a href="#topic+addFeatures">addFeatures</a></code>). To share a data frame across multiple models,
use the modelID &quot;default&quot;. All columns will be coerced to character
strings.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_plots">plots</code></td>
<td>
<p>Custom plotting functions for the study. The input object is a
nested list. The first list corresponds to the modelID(s). The second list
corresponds to the name(s) of the function(s) defined in the current R
session. The third list provides metadata to describe each plot. The only
required metadata element is <code>displayName</code>, which controls how the
plot will be named in the app. You are encouraged to also specify the
<code>plotType</code>, e.g. <code>"singleFeature"</code>, <code>"multiFeature"</code>,
<code>"multiTest"</code>, <code>"multiModel"</code>. PlotType accepts vector of
entries, whenever applicable, e.g., plotType = c(<code>"multiFeature"</code>,
<code>"multiTest"</code>). If you do not specify the <code>plotType</code>, the plot
will be assumed to be <code>"singleFeature"</code> and <code>"singleTest"</code>.
Optionally, if the plotting function requires external packages, these can
be defined in the element <code>packages</code>. To share plots across multiple
models, use the modelID &quot;default&quot;. To add a plotting function that returns
an interactive plotly plot, add &quot;plotly&quot; to the <code>plotType</code> vector.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_mapping">mapping</code></td>
<td>
<p>Feature IDs from models. The input object is a list of named
data frames. For each data frame, column names indicate model names while
rows indicate featureIDs per model. Features with same index position across
columns are treated as mapped across models. For each model, feature IDs must
match feature IDs available in the results object of the respective model.
1:N relationships are allowed.
</p>
<p>Mapping list elements are required to be named as 'default' or after a model
name as provided in addModels(). If a single data frame is provided, this
list element is recommended to be named 'default'. For multiple list
elements, each with its own data frame, list elements should be named after
model name(s) (a single element may still be  named 'default'). In that case,
when navigating in ON front-end (FE), mapping element related to the selected
model in the FE will be used in multimodel plots. If a selected model in FE
does not have a corresponding mapping list element, it may still use the
mapping list element called 'default' if this is available.
</p>
<p>E.g., if in a study there are models &quot;transcriptomics&quot; and &quot;proteomics&quot; and
the user wants to create a plot based on data from both, a mapping list
should be provided with addMapping(). In this case, the mapping list element
may be named 'default'. This should contain a data frame with column names
'transcriptomics' and 'proteomics', where feature IDs that map across models
are found in the same row.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_barcodes">barcodes</code></td>
<td>
<p>The metadata variables that describe the barcode plot.
The input object is a list of lists (one per model). Each sublist must
contain the element <code>statistic</code>, which is the column name in the
results table to use to construct the barcode plot. Each sublist may
additionally contain any of the following optional elements:
1) <code>absolute</code> - Should the statistic be converted to its absolute
value (default is <code>TRUE</code>).
2) <code>logFoldChange</code> - The column name in the results table that contains
the log fold change values.
3) <code>labelStat</code> - The x-axis label to describe the statistic.
4) <code>labelLow</code> - The left-side label to describe low values of the statistic.
5) <code>labelHigh</code> - The right-side label to describe high values of the statistic.
6) <code>featureDisplay</code> - The feature variable to use to label the barcode plot
on hover.
To share metadata across multiple models, use the modelID &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_reports">reports</code></td>
<td>
<p>The analysis report(s) that explain how the study results were
generated. The input object is a list of character vectors (one per model).
Each element should be either a URL or a path to a file on your computer.
If it is a path to a file, this file will be included in the exported study
package. To share a report across multiple models, use the modelID
&quot;default&quot;.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_resultslinkouts">resultsLinkouts</code></td>
<td>
<p>The URL patterns that describe linkouts to external
resources (see Details below). The input object is a nested named list. The
names of the list correspond to the model names. Each element of the list
is a named list of character vectors. The names of this nested list must
correspond to the column names of the matching features table. To share
linkouts across multiple models, use the modelID &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_enrichmentslinkouts">enrichmentsLinkouts</code></td>
<td>
<p>The URL patterns that describe linkouts to
external resources (see Details below). The input object is a named list.
The names of the list correspond to the annotation names. Each element of
the list is a character vector of linkouts for that annotationID.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_metafeatureslinkouts">metaFeaturesLinkouts</code></td>
<td>
<p>The URL patterns that describe linkouts to external
resources (see Details below). The input object is a nested named list. The
names of the list correspond to the model names. Each element of the list
is a named list of character vectors. The names of this nested list must
correspond to the column names of the matching metaFeatures table (<code><a href="#topic+addMetaFeatures">addMetaFeatures</a></code>). To share
linkouts across multiple models, use the modelID &quot;default&quot;.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_version">version</code></td>
<td>
<p>(Optional) Include a version number to track the updates to
your study package. If you export the study to a package, the version is
used as the package version.</p>
</td></tr>
<tr><td><code id="createStudy_+3A_maintainer">maintainer</code></td>
<td>
<p>(Optional) Include the name of the study package's
maintainer</p>
</td></tr>
<tr><td><code id="createStudy_+3A_maintaineremail">maintainerEmail</code></td>
<td>
<p>(Optional) Include the email of the study package's
maintainer</p>
</td></tr>
<tr><td><code id="createStudy_+3A_studymeta">studyMeta</code></td>
<td>
<p>(Optional) Define metadata about your study. The input is a
list of key:value pairs. See below for more details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can add metadata to describe your study by passing a named list to to the
argument <code>studyMeta</code>. The names of the list cannot contain spaces or
colons, and they can't start with <code>#</code> or <code>-</code>. The values of each
list should be a single value. Also, your metadata fields cannot use any of
the
<a href="https://gist.github.com/jdblischak/f9d946327c9991fb57dde1e6f2bff1c2">reserved
fields for R's DESCRIPTION file</a>.
</p>


<h3>Value</h3>

<p>Returns a new OmicNavigator study object, which is a named nested
list with class <code>onStudy</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addSamples">addSamples</a></code>,
<code><a href="#topic+addFeatures">addFeatures</a></code>,
<code><a href="#topic+addModels">addModels</a></code>,
<code><a href="#topic+addAssays">addAssays</a></code>,
<code><a href="#topic+addTests">addTests</a></code>,
<code><a href="#topic+addAnnotations">addAnnotations</a></code>,
<code><a href="#topic+addResults">addResults</a></code>,
<code><a href="#topic+addEnrichments">addEnrichments</a></code>,
<code><a href="#topic+addMetaFeatures">addMetaFeatures</a></code>,
<code><a href="#topic+addPlots">addPlots</a></code>,
<code><a href="#topic+addMapping">addMapping</a></code>,
<code><a href="#topic+addBarcodes">addBarcodes</a></code>,
<code><a href="#topic+addReports">addReports</a></code>,
<code><a href="#topic+addResultsLinkouts">addResultsLinkouts</a></code>,
<code><a href="#topic+addEnrichmentsLinkouts">addEnrichmentsLinkouts</a></code>,
<code><a href="#topic+addMetaFeaturesLinkouts">addMetaFeaturesLinkouts</a></code>,
<code><a href="#topic+exportStudy">exportStudy</a></code>,
<code><a href="#topic+installStudy">installStudy</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
study &lt;- createStudy(name = "ABC",
                     description = "An analysis of ABC")

# Define a version and study metadata
study &lt;- createStudy(name = "ABC",
                     description = "An analysis of ABC",
                     version = "0.1.0",
                     maintainer = "My Name",
                     maintainerEmail = "me@email.com",
                     studyMeta = list(department = "immunology",
                                      organism = "Mus musculus"))

</code></pre>

<hr>
<h2 id='exportStudy'>Export a study</h2><span id='topic+exportStudy'></span>

<h3>Description</h3>

<p>Export a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exportStudy(
  study,
  type = c("tarball", "package"),
  path = NULL,
  requireValid = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exportStudy_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study</p>
</td></tr>
<tr><td><code id="exportStudy_+3A_type">type</code></td>
<td>
<p>Export study as a package tarball (&quot;tarball&quot;) or as a package
directory (&quot;package&quot;)</p>
</td></tr>
<tr><td><code id="exportStudy_+3A_path">path</code></td>
<td>
<p>Optional file path to save the object</p>
</td></tr>
<tr><td><code id="exportStudy_+3A_requirevalid">requireValid</code></td>
<td>
<p>Require that study is valid before exporting</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns the name of the tarball file (&quot;tarball&quot;) or the
path to the package directory (&quot;package&quot;)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+validateStudy">validateStudy</a></code>
</p>

<hr>
<h2 id='getAnnotations'>Get annotations from a study</h2><span id='topic+getAnnotations'></span>

<h3>Description</h3>

<p>Get annotations from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAnnotations(study, annotationID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAnnotations_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getAnnotations_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getAnnotations_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getAnnotations_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getAssays'>Get assays from a study</h2><span id='topic+getAssays'></span>

<h3>Description</h3>

<p>Get assays from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAssays(study, modelID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAssays_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getAssays_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getAssays_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getAssays_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getBarcodeData'>Get data for barcode and violin plots</h2><span id='topic+getBarcodeData'></span>

<h3>Description</h3>

<p>Get data for barcode and violin plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBarcodeData(study, modelID, testID, annotationID, termID)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getBarcodeData_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getBarcodeData_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getBarcodeData_+3A_testid">testID</code></td>
<td>
<p>Filter by testID</p>
</td></tr>
<tr><td><code id="getBarcodeData_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getBarcodeData_+3A_termid">termID</code></td>
<td>
<p>Filter by termID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr><td><code>data</code></td>
<td>
<p>Data frame with the differential statistics to plot</p>
</td></tr>
<tr><td><code>highest</code></td>
<td>
<p>(numeric) The largest differential statistic, rounded up to
the next integer</p>
</td></tr>
<tr><td><code>labelStat</code></td>
<td>
<p>(character) The x-axis label to describe the differential
statistic</p>
</td></tr>
<tr><td><code>labelLow</code></td>
<td>
<p>(character) The vertical axis label on the left to describe
smaller values (default is &quot;Low&quot;)</p>
</td></tr>
<tr><td><code>labelHigh</code></td>
<td>
<p>(character) The vertical axis label on the right to
describe larger values (default is &quot;High&quot;)</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+addBarcodes">addBarcodes</a></code>, <code><a href="#topic+getBarcodes">getBarcodes</a></code>
</p>

<hr>
<h2 id='getBarcodes'>Get barcodes from a study</h2><span id='topic+getBarcodes'></span>

<h3>Description</h3>

<p>Get barcodes from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getBarcodes(study, modelID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getBarcodes_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getBarcodes_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getBarcodes_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getBarcodes_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getEnrichments'>Get enrichments from a study</h2><span id='topic+getEnrichments'></span>

<h3>Description</h3>

<p>Get enrichments from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getEnrichments(
  study,
  modelID = NULL,
  annotationID = NULL,
  testID = NULL,
  quiet = FALSE,
  libraries = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getEnrichments_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getEnrichments_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getEnrichments_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getEnrichments_+3A_testid">testID</code></td>
<td>
<p>Filter by testID</p>
</td></tr>
<tr><td><code id="getEnrichments_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getEnrichments_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getEnrichmentsIntersection'>getEnrichmentsIntersection</h2><span id='topic+getEnrichmentsIntersection'></span>

<h3>Description</h3>

<p>getEnrichmentsIntersection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getEnrichmentsIntersection(
  study,
  modelID,
  annotationID,
  mustTests,
  notTests,
  sigValue,
  operator,
  type
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getEnrichmentsIntersection_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getEnrichmentsIntersection_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getEnrichmentsIntersection_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getEnrichmentsIntersection_+3A_musttests">mustTests</code></td>
<td>
<p>The testIDs for which a featureID (or termID for enrichment)
must pass the filters</p>
</td></tr>
<tr><td><code id="getEnrichmentsIntersection_+3A_nottests">notTests</code></td>
<td>
<p>The testIDs for which a featureID (or termID for enrichment)
must <b>not</b> pass the filters. In other words, if a featureID passes the
filter for a testID specified in notTests, that featureID is removed from
the output</p>
</td></tr>
<tr><td><code id="getEnrichmentsIntersection_+3A_sigvalue">sigValue</code></td>
<td>
<p>The numeric significance value to use as a cutoff for each
column</p>
</td></tr>
<tr><td><code id="getEnrichmentsIntersection_+3A_operator">operator</code></td>
<td>
<p>The comparison operators for each column, e.g. <code>"&lt;"</code></p>
</td></tr>
<tr><td><code id="getEnrichmentsIntersection_+3A_type">type</code></td>
<td>
<p>Type of p-value: (<code>"nominal"</code> or <code>"adjusted"</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame with the enrichments, similar to
<code><a href="#topic+getEnrichmentsTable">getEnrichmentsTable</a></code>. Only rows that pass all the filters are
included.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getEnrichmentsTable">getEnrichmentsTable</a></code>
</p>

<hr>
<h2 id='getEnrichmentsLinkouts'>Get enrichments table linkouts from a study</h2><span id='topic+getEnrichmentsLinkouts'></span>

<h3>Description</h3>

<p>Get enrichments table linkouts from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getEnrichmentsLinkouts(
  study,
  annotationID = NULL,
  quiet = FALSE,
  libraries = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getEnrichmentsLinkouts_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getEnrichmentsLinkouts_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getEnrichmentsLinkouts_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getEnrichmentsLinkouts_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getEnrichmentsNetwork'>Get enrichments network from a study</h2><span id='topic+getEnrichmentsNetwork'></span>

<h3>Description</h3>

<p>Get enrichments network from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getEnrichmentsNetwork(study, modelID, annotationID, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getEnrichmentsNetwork_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getEnrichmentsNetwork_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getEnrichmentsNetwork_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getEnrichmentsNetwork_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list with the following components:
</p>
<table>
<tr><td><code>tests</code></td>
<td>
<p>(character) Vector of testIDs</p>
</td></tr>
<tr><td><code>nodes</code></td>
<td>
<p>(data frame) The description of each annotation term (i.e.
node). The nominal and adjusted p-values are in list-columns.</p>
</td></tr>
<tr><td><code>links</code></td>
<td>
<p>(list) The statistics for each pairwise overlap between the
annotation terms (i.e. nodes)</p>
</td></tr>
</table>

<hr>
<h2 id='getEnrichmentsTable'>Get enrichments table from a study</h2><span id='topic+getEnrichmentsTable'></span>

<h3>Description</h3>

<p>Get enrichments table from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getEnrichmentsTable(
  study,
  modelID,
  annotationID,
  type = "nominal",
  libraries = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getEnrichmentsTable_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getEnrichmentsTable_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getEnrichmentsTable_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getEnrichmentsTable_+3A_type">type</code></td>
<td>
<p>Type of p-value: (<code>"nominal"</code> or <code>"adjusted"</code>)</p>
</td></tr>
<tr><td><code id="getEnrichmentsTable_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of enrichments with the following columns:
</p>
<table>
<tr><td><code>termID</code></td>
<td>
<p>The unique ID for the annotation term</p>
</td></tr>
<tr><td><code>description</code></td>
<td>
<p>The description of the annotation term</p>
</td></tr>
<tr><td><code>...</code></td>
<td>
<p>One column for each of the enrichments</p>
</td></tr>
</table>

<hr>
<h2 id='getEnrichmentsUpset'>getEnrichmentsUpset</h2><span id='topic+getEnrichmentsUpset'></span>

<h3>Description</h3>

<p>getEnrichmentsUpset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getEnrichmentsUpset(
  study,
  modelID,
  annotationID,
  sigValue,
  operator,
  type,
  tests = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getEnrichmentsUpset_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getEnrichmentsUpset_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getEnrichmentsUpset_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getEnrichmentsUpset_+3A_sigvalue">sigValue</code></td>
<td>
<p>The numeric significance value to use as a cutoff for each
column</p>
</td></tr>
<tr><td><code id="getEnrichmentsUpset_+3A_operator">operator</code></td>
<td>
<p>The comparison operators for each column, e.g. <code>"&lt;"</code></p>
</td></tr>
<tr><td><code id="getEnrichmentsUpset_+3A_type">type</code></td>
<td>
<p>Type of p-value: (<code>"nominal"</code> or <code>"adjusted"</code>)</p>
</td></tr>
<tr><td><code id="getEnrichmentsUpset_+3A_tests">tests</code></td>
<td>
<p>Restrict UpSet plot to only include these tests</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. This function is called for the side effect of
creating an UpSet plot.
</p>

<hr>
<h2 id='getFavicons'>Get favicon URLs for table linkouts</h2><span id='topic+getFavicons'></span>

<h3>Description</h3>

<p>To enhance the display of the linkouts in the app's tables, it can fetch the
favicon URL for each website.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getFavicons(linkouts)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFavicons_+3A_linkouts">linkouts</code></td>
<td>
<p>Character vector or (potentially nested) list of character
vectors containing the URLs for the table linkouts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The URLs to the favicons for each linkout. The output returned will
always be the same class and structure as the input.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getResultsLinkouts">getResultsLinkouts</a></code>,
<code><a href="#topic+getEnrichmentsLinkouts">getEnrichmentsLinkouts</a></code>
</p>

<hr>
<h2 id='getFeatures'>Get features from a study</h2><span id='topic+getFeatures'></span>

<h3>Description</h3>

<p>Get features from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getFeatures(study, modelID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getFeatures_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getFeatures_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getFeatures_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getFeatures_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame (if <code>modelID</code> is specified) or a list of data
frames. All the columns will be character strings, even if the values
appear numeric.
</p>

<hr>
<h2 id='getInstalledStudies'>Get installed OmicNavigator studies</h2><span id='topic+getInstalledStudies'></span>

<h3>Description</h3>

<p>Get installed OmicNavigator studies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getInstalledStudies(libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getInstalledStudies_+3A_libraries">libraries</code></td>
<td>
<p>Character vector of library directories to search for study
packages. If <code>NULL</code>, uses <code>.libPaths</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character vector of the installed OmicNavigator study
packages
</p>

<hr>
<h2 id='getLinkFeatures'>Get the shared features in a network link</h2><span id='topic+getLinkFeatures'></span>

<h3>Description</h3>

<p>Get the shared features in a network link
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getLinkFeatures(study, annotationID, termID1, termID2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getLinkFeatures_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Only accepts name of installed study
package.</p>
</td></tr>
<tr><td><code id="getLinkFeatures_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getLinkFeatures_+3A_termid1">termID1</code>, <code id="getLinkFeatures_+3A_termid2">termID2</code></td>
<td>
<p>Linked terms to find overlapping features</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character vector with the features included in both termIDs
(i.e. the intersection)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getNodeFeatures">getNodeFeatures</a></code>
</p>

<hr>
<h2 id='getMapping'>Get mapping object from a study</h2><span id='topic+getMapping'></span>

<h3>Description</h3>

<p>Get mapping object from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMapping(study, modelID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMapping_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getMapping_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getMapping_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getMapping_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getMetaFeatures'>Get metaFeatures from a study</h2><span id='topic+getMetaFeatures'></span>

<h3>Description</h3>

<p>Get metaFeatures from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMetaFeatures(study, modelID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMetaFeatures_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getMetaFeatures_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getMetaFeatures_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getMetaFeatures_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getMetaFeaturesLinkouts'>Get metaFeatures table linkouts from a study</h2><span id='topic+getMetaFeaturesLinkouts'></span>

<h3>Description</h3>

<p>Get metaFeatures table linkouts from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMetaFeaturesLinkouts(study, modelID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMetaFeaturesLinkouts_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getMetaFeaturesLinkouts_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getMetaFeaturesLinkouts_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getMetaFeaturesLinkouts_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getMetaFeaturesTable'>Get metaFeatures for a given feature</h2><span id='topic+getMetaFeaturesTable'></span>

<h3>Description</h3>

<p>Get metaFeatures for a given feature
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMetaFeaturesTable(study, modelID, featureID)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getMetaFeaturesTable_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getMetaFeaturesTable_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getMetaFeaturesTable_+3A_featureid">featureID</code></td>
<td>
<p>Filter by featureID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame with the metaFeatures for the provided
featureID. If the featureID is not found in the metaFeatures table, the
data frame will have zero rows.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addMetaFeatures">addMetaFeatures</a></code>, <code><a href="#topic+getMetaFeatures">getMetaFeatures</a></code>
</p>

<hr>
<h2 id='getModels'>Get models from a study</h2><span id='topic+getModels'></span>

<h3>Description</h3>

<p>Get models from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getModels(study, modelID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getModels_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getModels_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getModels_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getModels_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getNodeFeatures'>Get the features in a network node</h2><span id='topic+getNodeFeatures'></span>

<h3>Description</h3>

<p>Get the features in a network node
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNodeFeatures(study, annotationID, termID, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getNodeFeatures_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Only accepts name of installed study
package.</p>
</td></tr>
<tr><td><code id="getNodeFeatures_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getNodeFeatures_+3A_termid">termID</code></td>
<td>
<p>Filter by termID</p>
</td></tr>
<tr><td><code id="getNodeFeatures_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character vector with the features in the termID
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getLinkFeatures">getLinkFeatures</a></code>
</p>

<hr>
<h2 id='getOverlaps'>Get overlaps from a study</h2><span id='topic+getOverlaps'></span>

<h3>Description</h3>

<p>Get overlaps from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getOverlaps(study, annotationID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getOverlaps_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getOverlaps_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getOverlaps_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getOverlaps_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getPackageVersion'>Get version of OmicNavigator package</h2><span id='topic+getPackageVersion'></span>

<h3>Description</h3>

<p>This is a convenience function for the app. It is easier to always call the
OmicNavigator package functions via OpenCPU than to call the utils package
for this one endpoint.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPackageVersion()
</code></pre>


<h3>Value</h3>

<p>Returns a one-element character vector with the version of the
currently installed OmicNavigator R package
</p>

<hr>
<h2 id='getPlots'>Get plots from a study</h2><span id='topic+getPlots'></span>

<h3>Description</h3>

<p>Get plots from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPlots(study, modelID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPlots_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getPlots_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getPlots_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getPlots_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getPlottingData'>Get plotting data</h2><span id='topic+getPlottingData'></span>

<h3>Description</h3>

<p>This function creates the input data that <code><a href="#topic+plotStudy">plotStudy</a></code> passes to
custom plotting functions added with <code><a href="#topic+addPlots">addPlots</a></code>. You can use it
directly when you are interactively creating your custom plotting functions.
Note that for multiModel plots testID is required to be a named vector, with
each testID named after the related modelID.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPlottingData(study, modelID, featureID, testID = NULL, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPlottingData_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getPlottingData_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getPlottingData_+3A_featureid">featureID</code></td>
<td>
<p>Filter by featureID</p>
</td></tr>
<tr><td><code id="getPlottingData_+3A_testid">testID</code></td>
<td>
<p>Filter by testID</p>
</td></tr>
<tr><td><code id="getPlottingData_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of 4 data frames:
</p>
<table>
<tr><td><code>assays</code></td>
<td>
<p>A data frame that contains the assay measurements,
filtered to only include the row(s) corresponding to the input featureID(s)
(see <code><a href="#topic+getAssays">getAssays</a></code>). If multiple featureIDs are requested, the rows
are reordered to match the order of this input. The column order is
unchanged.</p>
</td></tr>
<tr><td><code>samples</code></td>
<td>
<p>A data frame that contains the sample metadata for the
given modelID (see <code><a href="#topic+getSamples">getSamples</a></code>). The rows are reordered to match
the columns of the assays data frame.</p>
</td></tr>
<tr><td><code>features</code></td>
<td>
<p>A data frame that contains the feature metadata,
filtered to only include the row(s) corresponding to the input featureID(s)
(see <code><a href="#topic+getFeatures">getFeatures</a></code>). If multiple featureIDs are requested, the
rows are reordered to match the order of this input (and thus match the order
of the assays data frame).</p>
</td></tr>
<tr><td><code>results</code></td>
<td>
<p>A data frame that contains the test results,
filtered to only include the row(s) corresponding to the input featureID(s).
If multiple featureIDs are requested, the rows are reordered to match the
order of this input. The column order is unchanged. If multiple testIDs are
provided, they are stored in a list object.</p>
</td></tr>
</table>
<p>The data frame <code>results</code> is only returned if you pass a testID. By
default the app will always pass the currently selected testID. To make
<code>results</code> a list of data frames (one for each testID for the currently
selected modelID), set the plotType to be &quot;multiTest&quot; when adding the plot
with <code><a href="#topic+addPlots">addPlots</a></code>. For &quot;multiModel&quot; plots, testID and modelID
should be vectors of same length, where the index position indicate which
test in testID relate to which model in modelID.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addPlots">addPlots</a></code>, <code><a href="#topic+plotStudy">plotStudy</a></code>
</p>

<hr>
<h2 id='getReportLink'>Get link to report</h2><span id='topic+getReportLink'></span>

<h3>Description</h3>

<p>Get link to report
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getReportLink(study, modelID)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getReportLink_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getReportLink_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a one-element character vector with either a path to a report
file or a URL to a report web page. If no report is available for the
modelID, an empty character vector is returned.
</p>

<hr>
<h2 id='getReports'>Get reports from a study</h2><span id='topic+getReports'></span>

<h3>Description</h3>

<p>Get reports from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getReports(study, modelID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getReports_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getReports_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getReports_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getReports_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getResults'>Get results from a study</h2><span id='topic+getResults'></span>

<h3>Description</h3>

<p>Get results from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getResults(
  study,
  modelID = NULL,
  testID = NULL,
  quiet = FALSE,
  libraries = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getResults_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getResults_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getResults_+3A_testid">testID</code></td>
<td>
<p>Filter by testID</p>
</td></tr>
<tr><td><code id="getResults_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getResults_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getResultsIntersection'>getResultsIntersection</h2><span id='topic+getResultsIntersection'></span>

<h3>Description</h3>

<p>getResultsIntersection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getResultsIntersection(
  study,
  modelID,
  anchor,
  mustTests,
  notTests,
  sigValue,
  operator,
  column
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getResultsIntersection_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getResultsIntersection_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getResultsIntersection_+3A_anchor">anchor</code></td>
<td>
<p>The primary testID to filter the results</p>
</td></tr>
<tr><td><code id="getResultsIntersection_+3A_musttests">mustTests</code></td>
<td>
<p>The testIDs for which a featureID (or termID for enrichment)
must pass the filters</p>
</td></tr>
<tr><td><code id="getResultsIntersection_+3A_nottests">notTests</code></td>
<td>
<p>The testIDs for which a featureID (or termID for enrichment)
must <b>not</b> pass the filters. In other words, if a featureID passes the
filter for a testID specified in notTests, that featureID is removed from
the output</p>
</td></tr>
<tr><td><code id="getResultsIntersection_+3A_sigvalue">sigValue</code></td>
<td>
<p>The numeric significance value to use as a cutoff for each
column</p>
</td></tr>
<tr><td><code id="getResultsIntersection_+3A_operator">operator</code></td>
<td>
<p>The comparison operators for each column, e.g. <code>"&lt;"</code></p>
</td></tr>
<tr><td><code id="getResultsIntersection_+3A_column">column</code></td>
<td>
<p>The columns to apply the filters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame with the results, similar to
<code><a href="#topic+getResultsTable">getResultsTable</a></code>. Only rows that pass all the filters are
included. The new column <code>Set_Membership</code> is a comma-separated field
that includes the testIDs in which the featureID passed the filters.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getResultsTable">getResultsTable</a></code>
</p>

<hr>
<h2 id='getResultsLinkouts'>Get results table linkouts from a study</h2><span id='topic+getResultsLinkouts'></span>

<h3>Description</h3>

<p>Get results table linkouts from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getResultsLinkouts(study, modelID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getResultsLinkouts_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getResultsLinkouts_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getResultsLinkouts_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getResultsLinkouts_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getResultsTable'>Get results table from a study</h2><span id='topic+getResultsTable'></span>

<h3>Description</h3>

<p>Get results table from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getResultsTable(
  study,
  modelID,
  testID,
  annotationID = NULL,
  termID = NULL,
  libraries = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getResultsTable_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getResultsTable_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getResultsTable_+3A_testid">testID</code></td>
<td>
<p>Filter by testID</p>
</td></tr>
<tr><td><code id="getResultsTable_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="getResultsTable_+3A_termid">termID</code></td>
<td>
<p>Filter by termID</p>
</td></tr>
<tr><td><code id="getResultsTable_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame which includes the columns from the features table
followed by the columns from the results table. All the columns from the
features table will be character strings, even if the values appear
numeric.
</p>
<p>If the optional arguments <code>annotationID</code> and <code>termID</code> are
provided, the table will be filtered to only include features in that
annotation term.
</p>

<hr>
<h2 id='getResultsUpset'>getResultsUpset</h2><span id='topic+getResultsUpset'></span>

<h3>Description</h3>

<p>getResultsUpset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getResultsUpset(study, modelID, sigValue, operator, column, legacy = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getResultsUpset_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getResultsUpset_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getResultsUpset_+3A_sigvalue">sigValue</code></td>
<td>
<p>The numeric significance value to use as a cutoff for each
column</p>
</td></tr>
<tr><td><code id="getResultsUpset_+3A_operator">operator</code></td>
<td>
<p>The comparison operators for each column, e.g. <code>"&lt;"</code></p>
</td></tr>
<tr><td><code id="getResultsUpset_+3A_column">column</code></td>
<td>
<p>The columns to apply the filters</p>
</td></tr>
<tr><td><code id="getResultsUpset_+3A_legacy">legacy</code></td>
<td>
<p>Use legacy code (for testing purposes only)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns the output from <code><a href="UpSetR.html#topic+upset">upset</a></code>
</p>

<hr>
<h2 id='getSamples'>Get samples from a study</h2><span id='topic+getSamples'></span>

<h3>Description</h3>

<p>Get samples from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSamples(study, modelID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSamples_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getSamples_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getSamples_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getSamples_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getTests'>Get tests from a study</h2><span id='topic+getTests'></span>

<h3>Description</h3>

<p>Get tests from a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getTests(study, modelID = NULL, testID = NULL, quiet = FALSE, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getTests_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getTests_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="getTests_+3A_testid">testID</code></td>
<td>
<p>Filter by testID</p>
</td></tr>
<tr><td><code id="getTests_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="getTests_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='getUpsetCols'>getUpsetCols</h2><span id='topic+getUpsetCols'></span>

<h3>Description</h3>

<p>Determine the common columns across all tests of a model that are available
for filtering with UpSet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getUpsetCols(study, modelID)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getUpsetCols_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="getUpsetCols_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character vector with the names of the common columns
</p>

<hr>
<h2 id='group'>group from Bioconductor workflow RNAseq123</h2><span id='topic+group'></span>

<h3>Description</h3>

<p>A subset of the object <code>group</code> from Bioconductor workflow RNAseq123.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group
</code></pre>


<h3>Format</h3>

<p>A factor with 3 levels:
</p>

<dl>
<dt>Basal</dt><dd><p>Basal cells</p>
</dd>
<dt>LP</dt><dd><p>Luminal progenitor cells</p>
</dd>
<dt>ML</dt><dd><p>Mature luminal cells</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html">https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html</a>
</p>


<h3>References</h3>

<p>Law CW, Alhamdoosh M, Su S, Dong X, Tian L, Smyth GK, Ritchie ME.
<a href="https://f1000research.com/articles/5-1408/v3">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR [version 3; peer review: 3 approved].</a>
F1000Research 2018, 5:1408
<a href="https://doi.org/10.12688/f1000research.9005.3">doi:10.12688/f1000research.9005.3</a>
</p>
<p>Sheridan, J.M., Ritchie, M.E., Best, S.A. et al.
<a href="https://bmccancer.biomedcentral.com/articles/10.1186/s12885-015-1187-z">A pooled shRNA screen for regulators of primary mammary stem and progenitor cells identifies roles for <em>Asap1</em> and <em>Prox1</em>.</a>
BMC Cancer 2015, 15:221
<a href="https://doi.org/10.1186/s12885-015-1187-z">doi:10.1186/s12885-015-1187-z</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  table(group)
  str(group)

</code></pre>

<hr>
<h2 id='importStudy'>Import a study package</h2><span id='topic+importStudy'></span>

<h3>Description</h3>

<p>Create an onStudy object by importing an installed study package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importStudy(study, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importStudy_+3A_study">study</code></td>
<td>
<p>Named of an installed OmicNavigator study</p>
</td></tr>
<tr><td><code id="importStudy_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the <code>onStudy</code> object imported from the OmicNavigator
study package
</p>

<hr>
<h2 id='installApp'>Install the OmicNavigator web app</h2><span id='topic+installApp'></span>

<h3>Description</h3>

<p>In order to run the OmicNavigator web app on your local machine, the app must
be installed in the <code>www/</code> subdirectory of the R package. If you
installed the release tarball from the GitHub Releases page, then you already
have the app installed. If you installed directly from GitHub with
<code>install_github</code>, or if you want to use a different version of the app,
you can manually download and install the app.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>installApp(version = NULL, overwrite = FALSE, lib.loc = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="installApp_+3A_version">version</code></td>
<td>
<p>Version of the web app to install, e.g. <code>"1.0.0"</code></p>
</td></tr>
<tr><td><code id="installApp_+3A_overwrite">overwrite</code></td>
<td>
<p>Should an existing installation of the app be overwritten?</p>
</td></tr>
<tr><td><code id="installApp_+3A_lib.loc">lib.loc</code></td>
<td>
<p>a character vector with path names of <span class="rlang"><b>R</b></span> libraries.
See &lsquo;Details&rsquo; for the meaning of the default value of <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="installApp_+3A_...">...</code></td>
<td>
<p>Passed to <code><a href="utils.html#topic+download.file">download.file</a></code>. If the download
fails, you may need to adjust the download settings for your operating
system. For example, to download with <code>wget</code>, pass the argument
<code>method = "wget"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A one-element character vector with the absolute path to the
directory in which the app files were installed
</p>

<hr>
<h2 id='installStudy'>Install a study as an R package</h2><span id='topic+installStudy'></span>

<h3>Description</h3>

<p>Install a study as an R package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>installStudy(study, library = .libPaths()[1])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="installStudy_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study to install (class <code>onStudy</code>)</p>
</td></tr>
<tr><td><code id="installStudy_+3A_library">library</code></td>
<td>
<p>Directory to install package. Defaults to first directory
returned by <code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns the original <code>onStudy</code> object that was passed
to the argument <code>study</code>
</p>

<hr>
<h2 id='lane'>lane from Bioconductor workflow RNAseq123</h2><span id='topic+lane'></span>

<h3>Description</h3>

<p>A subset of the object <code>lane</code> from Bioconductor workflow RNAseq123.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lane
</code></pre>


<h3>Format</h3>

<p>A factor with 3 levels:
</p>

<dl>
<dt>L004</dt><dd><p>Sample sequenced on lane 4</p>
</dd>
<dt>L006</dt><dd><p>Sample sequenced on lane 6</p>
</dd>
<dt>L008</dt><dd><p>Sample sequenced on lane 8</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html">https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html</a>
</p>


<h3>References</h3>

<p>Law CW, Alhamdoosh M, Su S, Dong X, Tian L, Smyth GK, Ritchie ME.
<a href="https://f1000research.com/articles/5-1408/v3">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR [version 3; peer review: 3 approved].</a>
F1000Research 2018, 5:1408
<a href="https://doi.org/10.12688/f1000research.9005.3">doi:10.12688/f1000research.9005.3</a>
</p>
<p>Sheridan, J.M., Ritchie, M.E., Best, S.A. et al.
<a href="https://bmccancer.biomedcentral.com/articles/10.1186/s12885-015-1187-z">A pooled shRNA screen for regulators of primary mammary stem and progenitor cells identifies roles for <em>Asap1</em> and <em>Prox1</em>.</a>
BMC Cancer 2015, 15:221
<a href="https://doi.org/10.1186/s12885-015-1187-z">doi:10.1186/s12885-015-1187-z</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  table(lane)
  str(lane)

</code></pre>

<hr>
<h2 id='lcpm'>lcpm from Bioconductor workflow RNAseq123</h2><span id='topic+lcpm'></span>

<h3>Description</h3>

<p>A subset of the object <code>lcpm</code> from Bioconductor workflow RNAseq123.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lcpm
</code></pre>


<h3>Format</h3>

<p>A matrix with 24 rows and 9 columns
</p>


<h3>Source</h3>

<p><a href="https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html">https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html</a>
</p>


<h3>References</h3>

<p>Law CW, Alhamdoosh M, Su S, Dong X, Tian L, Smyth GK, Ritchie ME.
<a href="https://f1000research.com/articles/5-1408/v3">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR [version 3; peer review: 3 approved].</a>
F1000Research 2018, 5:1408
<a href="https://doi.org/10.12688/f1000research.9005.3">doi:10.12688/f1000research.9005.3</a>
</p>
<p>Sheridan, J.M., Ritchie, M.E., Best, S.A. et al.
<a href="https://bmccancer.biomedcentral.com/articles/10.1186/s12885-015-1187-z">A pooled shRNA screen for regulators of primary mammary stem and progenitor cells identifies roles for <em>Asap1</em> and <em>Prox1</em>.</a>
BMC Cancer 2015, 15:221
<a href="https://doi.org/10.1186/s12885-015-1187-z">doi:10.1186/s12885-015-1187-z</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(lcpm)
  str(lcpm)

</code></pre>

<hr>
<h2 id='listStudies'>List available studies and their metadata</h2><span id='topic+listStudies'></span>

<h3>Description</h3>

<p>List available studies and their metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>listStudies(libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="listStudies_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a nested list with one element per installed OmicNavigator
study package. Each study package entry has the following sublist components:
</p>
<table>
<tr><td><code>name</code></td>
<td>
<p>(character) Name of the study</p>
</td></tr>
<tr><td><code>package</code></td>
<td>
<p>(list) The fields from <code>DESCRIPTION</code></p>
</td></tr>
<tr><td><code>results</code></td>
<td>
<p>(nested list) The testIDs available for each modelID</p>
</td></tr>
<tr><td><code>enrichments</code></td>
<td>
<p>(nested list) The annotationIDs available for each modelID</p>
</td></tr>
<tr><td><code>plots</code></td>
<td>
<p>(nested list) The plotIDs available for each modelID</p>
</td></tr>
</table>

<hr>
<h2 id='Mm.c2'>Mm.c2 from Bioconductor workflow RNAseq123</h2><span id='topic+Mm.c2'></span>

<h3>Description</h3>

<p>A subset of the object <code>Mm.c2</code> from Bioconductor workflow RNAseq123.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Mm.c2
</code></pre>


<h3>Format</h3>

<p>A list of 4 character vectors
</p>


<h3>Source</h3>

<p><a href="https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html">https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html</a>
</p>


<h3>References</h3>

<p>Law CW, Alhamdoosh M, Su S, Dong X, Tian L, Smyth GK, Ritchie ME.
<a href="https://f1000research.com/articles/5-1408/v3">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR [version 3; peer review: 3 approved].</a>
F1000Research 2018, 5:1408
<a href="https://doi.org/10.12688/f1000research.9005.3">doi:10.12688/f1000research.9005.3</a>
</p>
<p>Sheridan, J.M., Ritchie, M.E., Best, S.A. et al.
<a href="https://bmccancer.biomedcentral.com/articles/10.1186/s12885-015-1187-z">A pooled shRNA screen for regulators of primary mammary stem and progenitor cells identifies roles for <em>Asap1</em> and <em>Prox1</em>.</a>
BMC Cancer 2015, 15:221
<a href="https://doi.org/10.1186/s12885-015-1187-z">doi:10.1186/s12885-015-1187-z</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  Mm.c2[[1]]
  str(Mm.c2)

</code></pre>

<hr>
<h2 id='OmicNavigator-package'>OmicNavigator</h2><span id='topic+OmicNavigator'></span><span id='topic+OmicNavigator-package'></span>

<h3>Description</h3>

<p>Package options to control package-wide behavior are described below.
</p>


<h3>Details</h3>

<p>The default prefix for OmicNavigator study packages is &quot;ONstudy&quot;. If you
would prefer to use a different prefix, you can change the package option
<code>OmicNavigator.prefix</code>. For example, to use the prefix &quot;OmicNavigatorStudy&quot;,
you could add the following line to your <code>.Rprofile</code> file.
</p>
<pre>
options(OmicNavigator.prefix = "OmicNavigatorStudy")
</pre>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: John Blischak <a href="mailto:jdblischak@gmail.com">jdblischak@gmail.com</a> (<a href="https://orcid.org/0000-0003-2634-9879">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Terrence Ernst (Web application)
</p>
</li>
<li><p> Paul Nordlund (Web application)
</p>
</li>
<li><p> Justin Moore (UpSet-related functions and web application)
</p>
</li>
<li><p> Joe Dalen (Barcode functionality and web application)
</p>
</li>
<li><p> Akshay Bhamidipati (Web application)
</p>
</li>
<li><p> Brett Engelmann <a href="mailto:brett.engelmann@abbvie.com">brett.engelmann@abbvie.com</a>
</p>
</li>
<li><p> Marco Curado (Improved plotting capabilities)
</p>
</li>
<li><p> Joe LoGrasso (Support for plotly)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> AbbVie Inc. [copyright holder, funder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/abbvie-external/OmicNavigator">https://github.com/abbvie-external/OmicNavigator</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/abbvie-external/OmicNavigator/issues">https://github.com/abbvie-external/OmicNavigator/issues</a>
</p>
</li></ul>


<hr>
<h2 id='plotStudy'>Plot a feature using a custom plotting function</h2><span id='topic+plotStudy'></span>

<h3>Description</h3>

<p>Plot a feature using a custom plotting function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotStudy(study, modelID, featureID, plotID, testID = NULL, libraries = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotStudy_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="plotStudy_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="plotStudy_+3A_featureid">featureID</code></td>
<td>
<p>Filter by featureID</p>
</td></tr>
<tr><td><code id="plotStudy_+3A_plotid">plotID</code></td>
<td>
<p>Filter by plotID</p>
</td></tr>
<tr><td><code id="plotStudy_+3A_testid">testID</code></td>
<td>
<p>Filter by testID</p>
</td></tr>
<tr><td><code id="plotStudy_+3A_libraries">libraries</code></td>
<td>
<p>The directories to search for installed study packages. If
left as <code>NULL</code> (the default), then
<code><a href="utils.html#topic+installed.packages">installed.packages</a></code> will use the result of
<code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The arguments <code>study</code>, <code>modelID</code>, <code>featureID</code>, and
<code>testID</code> are passed to the function <code><a href="#topic+getPlottingData">getPlottingData</a></code>, and
the nested list returned by this function is passed as the first argument to
your custom plotting function.
</p>


<h3>Value</h3>

<p>This function is called for the side effect of creating a plot. It
invisibly returns the result from the custom plotting function specified by
<code>plotID</code>. Previously it invisibly returned the study object. It's
unlikely you relied on this behavior. For a ggplot2 plot, the return value
will be the plotting object with class <code>"ggplot"</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+addPlots">addPlots</a></code>, <code><a href="#topic+getPlottingData">getPlottingData</a></code>
</p>

<hr>
<h2 id='removeStudy'>Remove an installed study R package</h2><span id='topic+removeStudy'></span>

<h3>Description</h3>

<p>Remove an installed study R package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeStudy(study, library = .libPaths()[1])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeStudy_+3A_study">study</code></td>
<td>
<p>The name of the study or an <code>onStudy</code> object. Do <strong>not</strong> include
the prefix of the installed package, e.g. <code>ONstudy</code>.</p>
</td></tr>
<tr><td><code id="removeStudy_+3A_library">library</code></td>
<td>
<p>Directory where the study package is installed. Defaults to first directory
returned by <code><a href="base.html#topic+.libPaths">.libPaths</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns the path of the removed study package
</p>

<hr>
<h2 id='samplenames'>samplenames from Bioconductor workflow RNAseq123</h2><span id='topic+samplenames'></span>

<h3>Description</h3>

<p>A subset of the object <code>samplenames</code> from Bioconductor workflow RNAseq123.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samplenames
</code></pre>


<h3>Format</h3>

<p>A character vector containing the unique sample identifiers
</p>


<h3>Source</h3>

<p><a href="https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html">https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html</a>
</p>


<h3>References</h3>

<p>Law CW, Alhamdoosh M, Su S, Dong X, Tian L, Smyth GK, Ritchie ME.
<a href="https://f1000research.com/articles/5-1408/v3">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR [version 3; peer review: 3 approved].</a>
F1000Research 2018, 5:1408
<a href="https://doi.org/10.12688/f1000research.9005.3">doi:10.12688/f1000research.9005.3</a>
</p>
<p>Sheridan, J.M., Ritchie, M.E., Best, S.A. et al.
<a href="https://bmccancer.biomedcentral.com/articles/10.1186/s12885-015-1187-z">A pooled shRNA screen for regulators of primary mammary stem and progenitor cells identifies roles for <em>Asap1</em> and <em>Prox1</em>.</a>
BMC Cancer 2015, 15:221
<a href="https://doi.org/10.1186/s12885-015-1187-z">doi:10.1186/s12885-015-1187-z</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  head(samplenames)
  str(samplenames)

</code></pre>

<hr>
<h2 id='shared-add'>Shared parameters for add functions</h2><span id='topic+shared-add'></span>

<h3>Description</h3>

<p>Shared parameters for add functions
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="shared-add_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study created with <code><a href="#topic+createStudy">createStudy</a></code></p>
</td></tr>
<tr><td><code id="shared-add_+3A_reset">reset</code></td>
<td>
<p>Reset the data prior to adding the new data (default:
<code>FALSE</code>). The default is to add to or modify any previously added data
(if it exists). Setting <code>reset = TRUE</code> enables you to remove existing
data you no longer want to include in the study.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original <code>onStudy</code> object passed to the argument
<code>study</code>, but modified to include the newly added data
</p>

<hr>
<h2 id='shared-data'>Shared sections for data objects</h2><span id='topic+shared-data'></span>

<h3>Description</h3>

<p>Shared sections for data objects
</p>


<h3>Source</h3>

<p><a href="https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html">https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html</a>
</p>


<h3>References</h3>

<p>Law CW, Alhamdoosh M, Su S, Dong X, Tian L, Smyth GK, Ritchie ME.
<a href="https://f1000research.com/articles/5-1408/v3">RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR [version 3; peer review: 3 approved].</a>
F1000Research 2018, 5:1408
<a href="https://doi.org/10.12688/f1000research.9005.3">doi:10.12688/f1000research.9005.3</a>
</p>
<p>Sheridan, J.M., Ritchie, M.E., Best, S.A. et al.
<a href="https://bmccancer.biomedcentral.com/articles/10.1186/s12885-015-1187-z">A pooled shRNA screen for regulators of primary mammary stem and progenitor cells identifies roles for <em>Asap1</em> and <em>Prox1</em>.</a>
BMC Cancer 2015, 15:221
<a href="https://doi.org/10.1186/s12885-015-1187-z">doi:10.1186/s12885-015-1187-z</a>
</p>

<hr>
<h2 id='shared-get'>Shared parameters for get functions</h2><span id='topic+shared-get'></span>

<h3>Description</h3>

<p>Shared parameters for get functions
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="shared-get_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study. Either an object of class <code>onStudy</code>,
or the name of an installed study package.</p>
</td></tr>
<tr><td><code id="shared-get_+3A_modelid">modelID</code></td>
<td>
<p>Filter by modelID</p>
</td></tr>
<tr><td><code id="shared-get_+3A_testid">testID</code></td>
<td>
<p>Filter by testID</p>
</td></tr>
<tr><td><code id="shared-get_+3A_annotationid">annotationID</code></td>
<td>
<p>Filter by annotationID</p>
</td></tr>
<tr><td><code id="shared-get_+3A_termid">termID</code></td>
<td>
<p>Filter by termID</p>
</td></tr>
<tr><td><code id="shared-get_+3A_featureid">featureID</code></td>
<td>
<p>Filter by featureID</p>
</td></tr>
<tr><td><code id="shared-get_+3A_plotid">plotID</code></td>
<td>
<p>Filter by plotID</p>
</td></tr>
<tr><td><code id="shared-get_+3A_quiet">quiet</code></td>
<td>
<p>Suppress messages (default: <code>FALSE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The object returned depends on the data available and any filters
(e.g. the argument  <code>modelID</code>):
</p>
<p>If no filters are specified, then the object returned is a nested list,
similar to the original input object.
</p>
<p>If one or more filters are applied, then only a subset of the original nested
list is returned. Technically, each filter applied is used to subset the
original nested list using <code><a href="base.html#topic+Extract">[[</a></code>.
</p>
<p>If no data is available, an empty list is returned (<code>list()</code>).
</p>

<hr>
<h2 id='shared-upset'>Shared parameters for upset functions</h2><span id='topic+shared-upset'></span>

<h3>Description</h3>

<p>Shared parameters for upset functions
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="shared-upset_+3A_anchor">anchor</code></td>
<td>
<p>The primary testID to filter the results</p>
</td></tr>
<tr><td><code id="shared-upset_+3A_musttests">mustTests</code></td>
<td>
<p>The testIDs for which a featureID (or termID for enrichment)
must pass the filters</p>
</td></tr>
<tr><td><code id="shared-upset_+3A_nottests">notTests</code></td>
<td>
<p>The testIDs for which a featureID (or termID for enrichment)
must <b>not</b> pass the filters. In other words, if a featureID passes the
filter for a testID specified in notTests, that featureID is removed from
the output</p>
</td></tr>
<tr><td><code id="shared-upset_+3A_sigvalue">sigValue</code></td>
<td>
<p>The numeric significance value to use as a cutoff for each
column</p>
</td></tr>
<tr><td><code id="shared-upset_+3A_operator">operator</code></td>
<td>
<p>The comparison operators for each column, e.g. <code>"&lt;"</code></p>
</td></tr>
<tr><td><code id="shared-upset_+3A_column">column</code></td>
<td>
<p>The columns to apply the filters</p>
</td></tr>
<tr><td><code id="shared-upset_+3A_type">type</code></td>
<td>
<p>Type of p-value: (<code>"nominal"</code> or <code>"adjusted"</code>)</p>
</td></tr>
<tr><td><code id="shared-upset_+3A_tests">tests</code></td>
<td>
<p>Restrict UpSet plot to only include these tests</p>
</td></tr>
<tr><td><code id="shared-upset_+3A_legacy">legacy</code></td>
<td>
<p>Use legacy code (for testing purposes only)</p>
</td></tr>
</table>

<hr>
<h2 id='startApp'>Start app on local machine</h2><span id='topic+startApp'></span>

<h3>Description</h3>

<p>After you have installed at least one OmicNavigator study package with
<code><a href="#topic+installStudy">installStudy</a></code>, you can explore the results in the app. The
function <code>startApp</code> starts a local instance of the app running on your
current machine. It will automatically open the app in your default browser.
For the best experience, use Google Chrome. From the dropdown menu, you will
be able to select from any of the studies you have installed on your machine.
When you are finished, you can stop the web server by returning to the R
console and pressing the Esc key (Windows) or Ctrl-C (Linux, macOS).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>startApp(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="startApp_+3A_...">...</code></td>
<td>
<p>extra parameters passed to <a href="opencpu.html#topic+ocpu_start_server">ocpu_start_server</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the app can't be run from within RStudio Server.
</p>
<p>The app requires some additional R packages to run. If you receive an error
about a missing package, please install it with
<code><a href="utils.html#topic+install.packages">install.packages</a></code>. To ensure you have all the extra
packages installed, you can run the command below:
</p>
<pre>
install.packages(c("faviconPlease", "opencpu", "UpSetR"))
</pre>


<h3>Value</h3>

<p>No return value. This function is only called for the side effect of
running a local instance of the app.
</p>

<hr>
<h2 id='summary.onStudy'>Summarize elements of OmicNavigator study</h2><span id='topic+summary.onStudy'></span>

<h3>Description</h3>

<p>Displays a tree-like summary of the elements that have been added to an
OmicNavigator study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'onStudy'
summary(object, elements = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.onStudy_+3A_object">object</code></td>
<td>
<p>OmicNavigator study object (class <code>onStudy</code>)</p>
</td></tr>
<tr><td><code id="summary.onStudy_+3A_elements">elements</code></td>
<td>
<p>Subset the output to only include specific elements of the
study, e.g. <code>c("results", "enrichments")</code></p>
</td></tr>
<tr><td><code id="summary.onStudy_+3A_...">...</code></td>
<td>
<p>Currently unused</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns the original <code>onStudy</code> object
</p>

<hr>
<h2 id='validateStudy'>Validate a study</h2><span id='topic+validateStudy'></span>

<h3>Description</h3>

<p>Validate a study
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validateStudy(study)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="validateStudy_+3A_study">study</code></td>
<td>
<p>An OmicNavigator study object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For a valid study object, the logical value <code>TRUE</code> is invisibly
returned. For an invalid study object, there is no return value because an
error is thrown.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
