<!DOCTYPE html><html><head><title>Help for package omics</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {omics}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cpma'><p>Cross Phenotype Meta-Analysis</p></a></li>
<li><a href='#mlm'><p>Multiple Linear Models</p></a></li>
<li><a href='#mlmer'><p>Multiple Linear Mixed-Effects Models</p></a></li>
<li><a href='#na.count'><p>Count Missing Values</p></a></li>
<li><a href='#pqq'>
<p>Quantile-Quantile Plot for <code class="reqn">p</code>-values</p></a></li>
<li><a href='#ranef.ranks'><p>Extract Random Effect Rank Distributions</p></a></li>
<li><a href='#ranks.heatmap'><p>Plot Random Effect Rank Distributions</p></a></li>
<li><a href='#re.match'><p>Pattern Matching and Extraction</p></a></li>
<li><a href='#setutils'><p>Set Operations on Multiple Vectors</p></a></li>
<li><a href='#var.components'><p>Extract Variance Components</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.1-5</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-11-27</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Title:</td>
<td>'&ndash;omics' Data Analysis Toolbox</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of functions to analyse '&ndash;omics' datasets such as DNA
    methylation and gene expression profiles.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0), methods</td>
</tr>
<tr>
<td>Imports:</td>
<td>lme4, pheatmap</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-11-27 18:19:59 UTC; gc</td>
</tr>
<tr>
<td>Author:</td>
<td>Gianluca Campanella [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gianluca Campanella &lt;gianluca@campanella.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-11-27 22:43:22</td>
</tr>
</table>
<hr>
<h2 id='cpma'>Cross Phenotype Meta-Analysis</h2><span id='topic+cpma'></span>

<h3>Description</h3>

<p>Performs (signed) cross phenotype meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    cpma(ps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cpma_+3A_ps">ps</code></td>
<td>
<p>a (non-empty) numeric vector of <code class="reqn">p</code>-values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with class <code>htest</code> containing the following components:
</p>
<table>
<tr><td><code>statistic</code></td>
<td>
<p>the value of the <code class="reqn">\chi^2</code> test statistic.</p>
</td></tr>
<tr><td><code>parameter</code></td>
<td>
<p>the estimated rate <code class="reqn">\lambda</code>.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>the <code class="reqn">p</code>-value for the test.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>a character string indicating the type of test performed.</p>
</td></tr>
<tr><td><code>data.name</code></td>
<td>
<p>a character string giving the name of the data.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Cotsapas, C., et al. 2011
Pervasive Sharing of Genetic Effects in Autoimmune Disease.
<em>PLOS Genetics</em> <b>7</b>(8):e1002254.
</p>

<hr>
<h2 id='mlm'>Multiple Linear Models</h2><span id='topic+mlm'></span>

<h3>Description</h3>

<p>Fits multiple linear models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    mlm(formula, data, vars, save.residuals=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mlm_+3A_formula">formula</code></td>
<td>
<p>an object of class <code>formula</code> (or one that can be coerced
to that class): a symbolic description of the models to be
fitted.</p>
</td></tr>
<tr><td><code id="mlm_+3A_data">data</code></td>
<td>
<p>an optional data frame, list or environment (or object coercible
by <code>as.data.frame</code> to a data frame) containing the
variables in the model.</p>
</td></tr>
<tr><td><code id="mlm_+3A_vars">vars</code></td>
<td>
<p>a character vector of variables of interest.</p>
</td></tr>
<tr><td><code id="mlm_+3A_save.residuals">save.residuals</code></td>
<td>
<p>whether model residuals should be stored and
returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with elements:
</p>
<table>
<tr><td><code>coefficients</code></td>
<td>
<p>if 'vars' is missing or non-empty, an array (or data
frame, if <code>vars</code> contains a single element) of
regression coefficient estimates (<code>coef</code>), standard
errors (<code>coef.se</code>), and corresponding
<code class="reqn">p</code>-values (<code>pval</code>).</p>
</td></tr>
<tr><td><code>residuals</code></td>
<td>
<p>if <code>save.residuals</code> is <code>TRUE</code>, a matrix of model
residuals.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm">lm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    coefs &lt;- matrix(runif(6*10), 6, 10)
    X &lt;- matrix(rnorm(100*5), 100, 5)
    Y &lt;- cbind(1, X) 

    models &lt;- mlm(Y ~ X)
    str(models)
</code></pre>

<hr>
<h2 id='mlmer'>Multiple Linear Mixed-Effects Models</h2><span id='topic+mlmer'></span>

<h3>Description</h3>

<p>Fits multiple linear mixed-effects models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    mlmer(formula, data, vars, lrt=TRUE, save.residuals=FALSE, save.ranks=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mlmer_+3A_formula">formula</code></td>
<td>
<p>an object of class <code>formula</code> (or one that can be coerced
to that class): a symbolic description of the models to be
fitted.</p>
</td></tr>
<tr><td><code id="mlmer_+3A_data">data</code></td>
<td>
<p>an optional data frame, list or environment (or object coercible
by <code>as.data.frame</code> to a data frame) containing the
variables in the model.</p>
</td></tr>
<tr><td><code id="mlmer_+3A_vars">vars</code></td>
<td>
<p>a character vector of variables of interest.</p>
</td></tr>
<tr><td><code id="mlmer_+3A_lrt">lrt</code></td>
<td>
<p>whether <code class="reqn">p</code>-values should be computed using likelihood-ratio
tests.</p>
</td></tr>
<tr><td><code id="mlmer_+3A_save.residuals">save.residuals</code></td>
<td>
<p>whether model residuals should be stored and
returned.</p>
</td></tr>
<tr><td><code id="mlmer_+3A_save.ranks">save.ranks</code></td>
<td>
<p>whether random effect ranks should be stored and
returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with elements:
</p>
<table>
<tr><td><code>coefficients</code></td>
<td>
<p>if 'vars' is missing or non-empty, an array (or data
frame, if <code>vars</code> contains a single element) of
regression coefficient estimates (<code>coef</code>), standard
errors (<code>coef.se</code>), and corresponding
<code class="reqn">p</code>-values (<code>pval</code>).</p>
</td></tr>
<tr><td><code>residuals</code></td>
<td>
<p>if <code>save.residuals</code> is <code>TRUE</code>, a matrix of model
residuals.</p>
</td></tr>
<tr><td><code>ranef.ranks</code></td>
<td>
<p>if <code>save.ranks</code> is <code>TRUE</code> (default), a list of
rank matrices (one per random effect) that can be
visualized using <code><a href="#topic+ranks.heatmap">ranks.heatmap</a></code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="lme4.html#topic+lmer">lmer</a></code> and <code><a href="#topic+mlm">mlm</a></code>
</p>

<hr>
<h2 id='na.count'>Count Missing Values</h2><span id='topic+na.count'></span>

<h3>Description</h3>

<p>Counts the number of missing values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    na.count(X, margin, fraction=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="na.count_+3A_x">X</code></td>
<td>
<p>an array, including a matrix.</p>
</td></tr>
<tr><td><code id="na.count_+3A_margin">margin</code></td>
<td>
<p>a vector giving the subscripts to count over.</p>
</td></tr>
<tr><td><code id="na.count_+3A_fraction">fraction</code></td>
<td>
<p>whether to return the fraction of missing values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number (or fraction) of missing values.
</p>

<hr>
<h2 id='pqq'>
Quantile-Quantile Plot for <code class="reqn">p</code>-values
</h2><span id='topic+pqq'></span><span id='topic+pqq.ci'></span>

<h3>Description</h3>

<p>Computes value to produce a quantile-quantile plot in <code class="reqn">-log_10</code> scale of
the given <code class="reqn">p</code>-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    pqq(ps)

    pqq.ci(n, level=0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pqq_+3A_ps">ps</code></td>
<td>
<p>a (non-empty) numeric vector of <code class="reqn">p</code>-values.</p>
</td></tr>
<tr><td><code id="pqq_+3A_n">n</code></td>
<td>
<p>number of <code class="reqn">p</code>-values.</p>
</td></tr>
<tr><td><code id="pqq_+3A_level">level</code></td>
<td>
<p>significance level.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>For <code>pqq</code>, a list containing the following components (all in
<code class="reqn">-log_10</code> scale):
</p>
<table>
<tr><td><code>x</code></td>
<td>
<p>a vector of (sorted) expected <code class="reqn">p</code>-values.</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>a vector of (sorted) observed <code class="reqn">p</code>-values.</p>
</td></tr>
</table>
<p>For <code>pqq.ci</code>, a matrix with the lower and upper bounds for the
confidence interval at the given significance level.
</p>

<hr>
<h2 id='ranef.ranks'>Extract Random Effect Rank Distributions</h2><span id='topic+ranef.ranks'></span><span id='topic+ranef.ranks.merMod'></span>

<h3>Description</h3>

<p>Extracts random effect rank distributions from a linear mixed-effects model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    ## S3 method for class 'merMod'
ranef.ranks(model, groups)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ranef.ranks_+3A_model">model</code></td>
<td>
<p>a linear mixed-effects model estimated using
<code><a href="lme4.html#topic+lmer">lmer</a></code>.</p>
</td></tr>
<tr><td><code id="ranef.ranks_+3A_groups">groups</code></td>
<td>
<p>a character vector of random effects of interest.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>which</code> is missing, a list of rank distribution vectors for all
random intercepts in the model.
Otherwise, a list of rank distribution vectors for random intercepts
specified by <code>which</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mlmer">mlmer</a></code>
</p>

<hr>
<h2 id='ranks.heatmap'>Plot Random Effect Rank Distributions</h2><span id='topic+ranks.heatmap'></span>

<h3>Description</h3>

<p>Plots the rank distribution matrix for a random effect generated by
<code><a href="#topic+mlmer">mlmer</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    ranks.heatmap(x, col="red")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ranks.heatmap_+3A_x">x</code></td>
<td>
<p>a matrix with the rank distribution for a random effect generated
by <code><a href="#topic+mlmer">mlmer</a></code>.</p>
</td></tr>
<tr><td><code id="ranks.heatmap_+3A_col">col</code></td>
<td>
<p>the color corresponding to &quot;higher-than-expected&quot; counts.</p>
</td></tr>
</table>


<h3>Side Effects</h3>

<p>Generates a new plot.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mlmer">mlmer</a></code>
</p>

<hr>
<h2 id='re.match'>Pattern Matching and Extraction</h2><span id='topic+re.match'></span>

<h3>Description</h3>

<p>Matches and extracts substrings using regular expressions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    re.match(pattern, x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="re.match_+3A_pattern">pattern</code></td>
<td>
<p>character string containing a regular expression.</p>
</td></tr>
<tr><td><code id="re.match_+3A_x">x</code></td>
<td>
<p>a character vector where matches are sought, or an object which can
be coerced by <code>as.character</code> to a character vector.</p>
</td></tr>
<tr><td><code id="re.match_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code>regexec</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a wrapper for consecutive calls to <code>regexec</code> and
<code>regmatches</code>.
</p>


<h3>Value</h3>

<p>A matrix of matched substrings.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+regexec">regexec</a></code> and <code><a href="base.html#topic+regmatches">regmatches</a></code>
</p>

<hr>
<h2 id='setutils'>Set Operations on Multiple Vectors</h2><span id='topic+mintersect'></span><span id='topic+munion'></span>

<h3>Description</h3>

<p>Performs set operations on multiple vectors.</p>


<h3>Usage</h3>

<pre><code class='language-R'>    mintersect(..., sorted=FALSE)

    munion(..., sorted=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setutils_+3A_...">...</code></td>
<td>
<p>input vectors</p>
</td></tr>
<tr><td><code id="setutils_+3A_sorted">sorted</code></td>
<td>
<p>whether the result should be sorted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Set intersection or union of the given vectors
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+intersect">intersect</a></code> and <code><a href="base.html#topic+union">union</a></code>
</p>

<hr>
<h2 id='var.components'>Extract Variance Components</h2><span id='topic+var.components'></span><span id='topic+var.components.merMod'></span>

<h3>Description</h3>

<p>Extracts variance components from a linear mixed-effects model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>    ## S3 method for class 'merMod'
var.components(model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="var.components_+3A_model">model</code></td>
<td>
<p>a linear mixed-effects model estimated using
<code><a href="lme4.html#topic+lmer">lmer</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with elements:
</p>
<table>
<tr><td><code>group</code></td>
<td>
<p>the name of the random effect.</p>
</td></tr>
<tr><td><code>var.name</code></td>
<td>
<p>the name of the variable.</p>
</td></tr>
<tr><td><code>var</code></td>
<td>
<p>the estimated variance.</p>
</td></tr>
<tr><td><code>var.prop</code></td>
<td>
<p>the estimated proportion of variance.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+mlmer">mlmer</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
