<!DOCTYPE html><html lang="en"><head><title>Help for package omicsTools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {omicsTools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#impute'><p>Impute function</p></a></li>
<li><a href='#normalize'><p>Normalize function</p></a></li>
<li><a href='#run_app'><p>Run the Shiny Application</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Omics Data Process Toolbox</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-06-29</td>
</tr>
<tr>
<td>Description:</td>
<td>Processing and analyzing omics data from genomics, transcriptomics, proteomics, and metabolomics platforms. It provides functions for preprocessing, normalization, visualization, and statistical analysis, as well as machine learning algorithms for predictive modeling. 'omicsTools' is an essential tool for researchers working with high-throughput omics data in fields such as biology, bioinformatics, and medicine.The QC-RLSC (quality controlâ€“based robust LOESS signal correction) algorithm is used for normalization. Dunn et al. (2011) &lt;<a href="https://doi.org/10.1038%2Fnprot.2011.335">doi:10.1038/nprot.2011.335</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Imports:</td>
<td>bs4Dash, config (&ge; 0.3.1), dplyr, DT, golem (&ge; 0.3.5),
magrittr, readr, shiny (&ge; 1.7.2), tibble</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>spelling, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/YaoxiangLi/omicsTools">https://github.com/YaoxiangLi/omicsTools</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/YaoxiangLi/omicsTools/issues">https://github.com/YaoxiangLi/omicsTools/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-06-30 18:20:54 UTC; bach</td>
</tr>
<tr>
<td>Author:</td>
<td>Yaoxiang Li <a href="https://orcid.org/0000-0001-9200-1016"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre,
    aut],
  Zihao Zhang [aut],
  Meth Jayatilake [aut],
  Amrita Cheema <a href="https://orcid.org/0000-0003-4877-7583"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yaoxiang Li &lt;liyaoxiang@outlook.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-07-03 16:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling 'rhs(lhs)'.
</p>

<hr>
<h2 id='impute'>Impute function</h2><span id='topic+impute'></span>

<h3>Description</h3>

<p>This function performs data cleaning and imputation on a given data matrix.
It removes blank and NIST samples, features with NA values more than the specified threshold,
and imputes remaining NA values with half of the smallest non-NA value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>impute(data, percent = 0.2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="impute_+3A_data">data</code></td>
<td>
<p>A data frame containing the sample data. The first column should contain
the sample identifiers, and the rest of the columns contain the peaks.</p>
</td></tr>
<tr><td><code id="impute_+3A_percent">percent</code></td>
<td>
<p>A numeric value between 0 and 1 representing the threshold of the
percentage of NA values a feature should have for it to be removed from the dataset.
Default value is 0.2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the first column as the sample identifiers and
the rest of the columns containing the cleaned and imputed peak intensities.
</p>


<h3>Author(s)</h3>

<p>Yaoxiang Li <a href="mailto:yl814@georgetown.edu">yl814@georgetown.edu</a>
</p>
<p>Georgetown University, USA
</p>
<p>License: GPL (&gt;= 3)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load the CSV data
data_file &lt;- system.file("extdata", "example1.csv", package = "omicsTools")
data &lt;- readr::read_csv(data_file)
# Apply the impute function
imputed_data &lt;- omicsTools::impute(data, percent = 0.2)


# Write the imputed data to a new CSV file
readr::write_csv(imputed_data, paste0(tempdir(), "/imputed_data.csv"))

</code></pre>

<hr>
<h2 id='normalize'>Normalize function</h2><span id='topic+normalize'></span>

<h3>Description</h3>

<p>This function performs normalization on the input data matrix using
the loess regression method. Normalization is done based on Quality Control
(QC) samples in the data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="normalize_+3A_data">data</code></td>
<td>
<p>A data frame containing the sample data. The first column
should contain the sample identifiers, and the rest of the columns
contain the peaks to be normalized. QC samples should be indicated
in the sample identifiers with 'QC'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the first column as the sample identifiers and
the rest of the columns containing the normalized peak intensities.
</p>


<h3>Author(s)</h3>

<p>Yaoxiang Li <a href="mailto:yl814@georgetown.edu">yl814@georgetown.edu</a>
</p>
<p>Georgetown University, USA
</p>
<p>License: GPL (&gt;= 3)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load the CSV data
data_file &lt;- system.file("extdata", "example2.csv", package = "omicsTools")
data &lt;- readr::read_csv(data_file)
# Apply the normalize function
normalized_data &lt;- omicsTools::normalize(data)


# Write the normalized data to a new CSV file
readr::write_csv(normalized_data, paste0(tempdir(), "/normalized_data.csv"))

</code></pre>

<hr>
<h2 id='run_app'>Run the Shiny Application</h2><span id='topic+run_app'></span>

<h3>Description</h3>

<p>Run the Shiny Application
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_app(
  onStart = NULL,
  options = list(),
  enableBookmarking = NULL,
  uiPattern = "/",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_app_+3A_onstart">onStart</code></td>
<td>
<p>A function that will be called before the app is actually run.
This is only needed for <code>shinyAppObj</code>, since in the <code>shinyAppDir</code>
case, a <code>global.R</code> file can be used for this purpose.</p>
</td></tr>
<tr><td><code id="run_app_+3A_options">options</code></td>
<td>
<p>Named options that should be passed to the <code>runApp</code> call
(these can be any of the following: &quot;port&quot;, &quot;launch.browser&quot;, &quot;host&quot;, &quot;quiet&quot;,
&quot;display.mode&quot; and &quot;test.mode&quot;). You can also specify <code>width</code> and
<code>height</code> parameters which provide a hint to the embedding environment
about the ideal height/width for the app.</p>
</td></tr>
<tr><td><code id="run_app_+3A_enablebookmarking">enableBookmarking</code></td>
<td>
<p>Can be one of <code>"url"</code>, <code>"server"</code>, or
<code>"disable"</code>. The default value, <code>NULL</code>, will respect the setting from
any previous calls to  <code><a href="shiny.html#topic+enableBookmarking">enableBookmarking()</a></code>. See <code><a href="shiny.html#topic+enableBookmarking">enableBookmarking()</a></code>
for more information on bookmarking your app.</p>
</td></tr>
<tr><td><code id="run_app_+3A_uipattern">uiPattern</code></td>
<td>
<p>A regular expression that will be applied to each <code>GET</code>
request to determine whether the <code>ui</code> should be used to handle the
request. Note that the entire request path must match the regular
expression in order for the match to be considered successful.</p>
</td></tr>
<tr><td><code id="run_app_+3A_...">...</code></td>
<td>
<p>arguments to pass to golem_opts.
See '?golem::get_golem_options' for more details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for launch the application.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
