<!DOCTYPE html><html><head><title>Help for package omicwas</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {omicwas}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ctassoc'><p>Cell-Type-Specific Association Testing</p></a></li>
<li><a href='#ctcisQTL'><p>Cell-Type-Specific QTL analysis</p></a></li>
<li><a href='#GSE42861small'><p>Small Subset of GSE42861 Dataset From GEO</p></a></li>
<li><a href='#GSE79262small'><p>Small Subset of GSE79262 Dataset From GEO</p></a></li>
<li><a href='#GTExsmall'><p>Small Subset of GTEx Dataset</p></a></li>
<li><a href='#rrs.fit'><p>Fitting reduced-rank ridge regression with given rank and shrinkage penalty</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Cell-Type-Specific Association Testing in Bulk Omics Experiments</td>
</tr>
<tr>
<td>Version:</td>
<td>0.8.0</td>
</tr>
<tr>
<td>Description:</td>
<td>In bulk epigenome/transcriptome experiments, molecular expression
    is measured in a tissue, which is a mixture of multiple types of cells.
    This package tests association of a disease/phenotype with a molecular marker
    for each cell type.
    The proportion of cell types in each sample needs to be given as input.
    The package is applicable to epigenome-wide association study (EWAS) and
    differential gene expression analysis.
    Takeuchi and Kato (submitted)
    "omicwas: cell-type-specific epigenome-wide and transcriptome association study".</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/fumi-github/omicwas">https://github.com/fumi-github/omicwas</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/fumi-github/omicwas/issues">https://github.com/fumi-github/omicwas/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>broom, data.table, dplyr, ff, glmnet, magrittr, MASS,
matrixStats, parallel, purrr, rlang, tidyr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-10-08 09:40:09 UTC; fumi</td>
</tr>
<tr>
<td>Author:</td>
<td>Fumihiko Takeuchi <a href="https://orcid.org/0000-0003-3185-5661"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Fumihiko Takeuchi &lt;fumihiko@takeuchi.name&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-10-08 12:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='ctassoc'>Cell-Type-Specific Association Testing</h2><span id='topic+ctassoc'></span>

<h3>Description</h3>

<p>Cell-Type-Specific Association Testing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctassoc(
  X,
  W,
  Y,
  C = NULL,
  test = "full",
  regularize = FALSE,
  num.cores = 1,
  chunk.size = 1000,
  seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctassoc_+3A_x">X</code></td>
<td>
<p>Matrix (or vector) of traits; samples x traits.</p>
</td></tr>
<tr><td><code id="ctassoc_+3A_w">W</code></td>
<td>
<p>Matrix of cell type composition; samples x cell types.</p>
</td></tr>
<tr><td><code id="ctassoc_+3A_y">Y</code></td>
<td>
<p>Matrix (or vector) of bulk omics measurements; markers x samples.</p>
</td></tr>
<tr><td><code id="ctassoc_+3A_c">C</code></td>
<td>
<p>Matrix (or vector) of covariates; samples x covariates.
X, W, Y, C should be numeric.</p>
</td></tr>
<tr><td><code id="ctassoc_+3A_test">test</code></td>
<td>
<p>Statistical test to apply; either <code>"full"</code>, <code>"marginal"</code>,
<code>"nls.identity"</code>, <code>"nls.log"</code>, <code>"nls.logit"</code>,
<code>"propdiff.identity"</code>, <code>"propdiff.log"</code>, <code>"propdiff.logit"</code>
or <code>"reducedrankridge"</code>.</p>
</td></tr>
<tr><td><code id="ctassoc_+3A_regularize">regularize</code></td>
<td>
<p>Whether to apply Tikhonov (ie ridge) regularization
to <code class="reqn">\beta_{h j k}</code>.
The regularization parameter is chosen automatically according to
an unbiased version of (Lawless &amp; Wang, 1976).
Effective for <code>nls.*</code> and <code>propdiff.*</code> tests.</p>
</td></tr>
<tr><td><code id="ctassoc_+3A_num.cores">num.cores</code></td>
<td>
<p>Number of CPU cores to use.
Full, marginal and propdiff tests are run in serial, thus num.cores is ignored.</p>
</td></tr>
<tr><td><code id="ctassoc_+3A_chunk.size">chunk.size</code></td>
<td>
<p>The size of job for a CPU core in one batch.
If you have many cores but limited memory, and there is a memory failure,
decrease num.cores and/or chunk.size.</p>
</td></tr>
<tr><td><code id="ctassoc_+3A_seed">seed</code></td>
<td>
<p>Seed for random number generation.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Let the indexes be
<code class="reqn">h</code> for cell type, <code class="reqn">i</code> for sample,
<code class="reqn">j</code> for marker (CpG site or gene),
<code class="reqn">k</code> for each trait that has cell-type-specific effect,
and <code class="reqn">l</code> for each trait that has a uniform effect across cell types.
The input data are <code class="reqn">X_{i k}</code>, <code class="reqn">C_{i l}</code>, <code class="reqn">W_{i h}</code> and <code class="reqn">Y_{j i}</code>,
where <code class="reqn">C_{i l}</code> can be omitted.
<code class="reqn">X_{i k}</code> and <code class="reqn">C_{i l}</code> are the values for two types of traits,
showing effects that are cell-type-specific or not, respectively.
Thus, calling <code class="reqn">X_{i k}</code> and <code class="reqn">C_{i l}</code> as &quot;traits&quot; and &quot;covariates&quot;
gives a rough idea, but is not strictly correct.
<code class="reqn">W_{i h}</code> represents the cell type composition and
<code class="reqn">Y_{j i}</code> represents the marker level,
such as methylation or gene expression.
For each tissue sample, the cell type proportion <code class="reqn">W_{i h}</code>
is the proportion of each cell type in the bulk tissue,
which is measured or imputed beforehand.
The marker level <code class="reqn">Y_{j i}</code> in bulk tissue is measured and provided as input.
</p>
<p>The parameters we estimate are
the cell-type-specific trait effect <code class="reqn">\beta_{h j k}</code>,
the tissue-uniform trait effect <code class="reqn">\gamma_{j l}</code>,
and the basal marker level <code class="reqn">\alpha_{h j}</code> in each cell type.
</p>
<p>We first describe the conventional linear regression models.
For marker <code class="reqn">j</code> in sample <code class="reqn">i</code>,
the maker level specific to cell type <code class="reqn">h</code> is
</p>
<p style="text-align: center;"><code class="reqn">\alpha_{h j} + \sum_k \beta_{h j k} * X_{i k}.</code>
</p>

<p>This is a representative value rather than a mean, because we do not model
a probability distribution for cell-type-specific expression.
The bulk tissue marker level is the average weighted by <code class="reqn">W_{i h}</code>,
</p>
<p style="text-align: center;"><code class="reqn">\mu_{j i} = \sum_h W_{i h} [ \alpha_{h j} + \sum_k \beta_{h j k} * X_{i k} ] +
                  \sum_l \gamma_{j l} C_{i l}.</code>
</p>

<p>The statistical model is
</p>
<p style="text-align: center;"><code class="reqn">Y_{j i} = \mu_{j i} + \epsilon_{j i},</code>
</p>

<p style="text-align: center;"><code class="reqn">\epsilon_{j i} ~ N(0, \sigma^2_j).</code>
</p>

<p>The error of the marker level is is noramlly distributed with variance
<code class="reqn">\sigma^2_j</code>, independently among samples.
</p>
<p>The <code>full</code> model is the linear regression
</p>
<p style="text-align: center;"><code class="reqn">Y_{j i} = (\sum_h \alpha_{h j} * W_{i h}) +
                (\sum_{h k} \beta_{h j k} * W_{i h} * X_{i k}) +
                (\sum_l \gamma_{j l} * C_{i l}) +
                error.</code>
</p>

<p>The <code>marginal</code> model tests the trait association only in one
cell type <code class="reqn">h</code>, under the linear regression,
</p>
<p style="text-align: center;"><code class="reqn">Y_{j i} = (\sum_{h'} \alpha_{h' j} * W_{i h'}) +
                (\sum_k \beta_{h j k} * W_{i h} * X_{i k}) +
                (\sum_l \gamma_{j l} * C_{i l}) +
                error.</code>
</p>

<p>The nonlinear model simultaneously analyze cell type composition in
linear scale and differential expression/methylation in log/logit scale.
The normalizing function is the natural logarithm <code class="reqn">f</code> = log for gene
expression, and <code class="reqn">f</code> = logit for methylation. Conventional linear regression
can be formulated by defining <code class="reqn">f</code> as the identity function. The three models
are named <code>nls.log</code>, <code>nls.logit</code> and <code>nls.identity</code>.
We denote the inverse function of <code class="reqn">f</code> by <code class="reqn">g</code>; <code class="reqn">g</code> = exp for
gene expression, and <code class="reqn">g</code> = logistic for methylation.
The mean normalized marker level of marker <code class="reqn">j</code> in sample <code class="reqn">i</code> becomes
</p>
<p style="text-align: center;"><code class="reqn">\mu_{j i} = f(\sum_h W_{i h} g( \alpha_{h j} + \sum_k \beta_{h j k} * X_{i k} )) +
                  \sum_l \gamma_{j l} C_{i l}.</code>
</p>

<p>The statistical model is
</p>
<p style="text-align: center;"><code class="reqn">f(Y_{j i}) = \mu_{j i} + \epsilon_{j i},</code>
</p>

<p style="text-align: center;"><code class="reqn">\epsilon_{j i} ~ N(0, \sigma^2_j).</code>
</p>

<p>The error of the marker level is is noramlly distributed with variance
<code class="reqn">\sigma^2_j</code>, independently among samples.
</p>
<p>The ridge regression aims to cope with multicollinearity of
the interacting terms <code class="reqn">W_{i h} * X_{i k}</code>.
Ridge regression is fit by minimizing the residual sum of squares (RSS) plus
<code class="reqn">\lambda \sum_{h k} \beta_{h j k}^2</code>, where <code class="reqn">\lambda &gt; 0</code> is the
regularization parameter.
</p>
<p>The propdiff tests try to cope with multicollinearity by, roughly speaking,
using mean-centered <code class="reqn">W_{i h}</code>.
We obtain, instead of <code class="reqn">\beta_{h j k}</code>, the deviation of
<code class="reqn">\beta_{h j k}</code> from the average across cell types.
Accordingly, the null hypothesis changes.
The original null hypothesis was <code class="reqn">\beta_{h j k} = 0</code>.
The null hypothesis when centered is
<code class="reqn">\beta_{h j k} - (\sum_{i h'} W_{i h'} \beta_{h' j k}) / (\sum_{i h'} W_{i h'}) = 0</code>.
It becomes difficult to detect a signal for a major cell type,
because <code class="reqn">\beta_{h j k}</code> would be close to the average across cell types.
The tests <code>propdiff.log</code> and <code>propdiff.logit</code> include
an additional preprocessing step that converts <code class="reqn">Y_{j i}</code> to <code class="reqn">f(Y_{j i})</code>.
Apart from the preprocessing, the computations are performed in linear scale.
As the preprocessing distorts the linearity between the dependent variable
and (the centered) <code class="reqn">W_{i h}</code>,
I actually think <code>propdiff.identity</code> is better.
</p>


<h3>Value</h3>

<p>A list with one element, which is named &quot;coefficients&quot;.
The element gives the estimate, statistic, p.value in tibble format.
In order to transform the estimate for <code class="reqn">\alpha_{h j}</code> to the original scale,
apply <code>plogis</code> for <code>test = nls.logit</code> and
<code>exp</code> for <code>test = nls.log</code>.
The estimate for <code class="reqn">\beta_{h j k}</code> by <code>test = nls.log</code> is
the natural logarithm of fold-change, not the log2.
If numerical convergence fails, <code>NA</code> is returned for that marker.
</p>


<h3>References</h3>

<p>Lawless, J. F., &amp; Wang, P. (1976). A simulation study of ridge and other
regression estimators.
Communications in Statistics - Theory and Methods, 5(4), 307–323.
<a href="https://doi.org/10.1080/03610927608827353">https://doi.org/10.1080/03610927608827353</a>
</p>


<h3>See Also</h3>

<p>ctcisQTL
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(GSE42861small)
X = GSE42861small$X
W = GSE42861small$W
Y = GSE42861small$Y
C = GSE42861small$C
result = ctassoc(X, W, Y, C = C)
result$coefficients


</code></pre>

<hr>
<h2 id='ctcisQTL'>Cell-Type-Specific QTL analysis</h2><span id='topic+ctcisQTL'></span>

<h3>Description</h3>

<p>Cell-Type-Specific QTL analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ctcisQTL(
  X,
  Xpos,
  W,
  Y,
  Ypos,
  C = NULL,
  max.pos.diff = 1e+06,
  outdir = tempdir(),
  outfile = "ctcisQTL.out.txt"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ctcisQTL_+3A_x">X</code></td>
<td>
<p>Matrix (or vector) of SNP genotypes; SNPs x samples.</p>
</td></tr>
<tr><td><code id="ctcisQTL_+3A_xpos">Xpos</code></td>
<td>
<p>Vector of the physical position of X</p>
</td></tr>
<tr><td><code id="ctcisQTL_+3A_w">W</code></td>
<td>
<p>Matrix of cell type composition; samples x cell types.</p>
</td></tr>
<tr><td><code id="ctcisQTL_+3A_y">Y</code></td>
<td>
<p>Matrix (or vector) of bulk omics measurements; markers x samples.</p>
</td></tr>
<tr><td><code id="ctcisQTL_+3A_ypos">Ypos</code></td>
<td>
<p>Vector of the physical position of Y</p>
</td></tr>
<tr><td><code id="ctcisQTL_+3A_c">C</code></td>
<td>
<p>Matrix (or vector) of covariates; samples x covariates.
X, Xpos, W, Y, Ypos, C should be numeric.</p>
</td></tr>
<tr><td><code id="ctcisQTL_+3A_max.pos.diff">max.pos.diff</code></td>
<td>
<p>Maximum positional difference to compute cis-QTL.
Association analysis is performed between a row of X and a row of Y,
only when they are within this limit.
Since the limiting is only by position, the function needs to be run
separately for each chromosome.</p>
</td></tr>
<tr><td><code id="ctcisQTL_+3A_outdir">outdir</code></td>
<td>
<p>Output directory.</p>
</td></tr>
<tr><td><code id="ctcisQTL_+3A_outfile">outfile</code></td>
<td>
<p>Output file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A function for analyses of QTL, such as eQTL, mQTL, pQTL.
The statistical test is almost identical to
<code>ctassoc(test =  "nls.identity", regularize = "TRUE")</code>.
Association analysis is performed between each row of Y and each row of X.
Usually, the former will be a methylation/expression marker,
and the latter will be a SNP.
To cope with the large number of combinations,
the testing is limited to pairs whose position is within
the difference specified by <code>max.pos.diff</code>; i.e., limited to cis-QTL.
In detail, this function performs linear ridge regression,
whereas <code>ctassoc(test =  "nls.identity", regularize = "TRUE")</code>
actually is nonlinear regression
but with <code class="reqn">f</code> = identity as normalizing transformation.
In order to speed up computation, first, the parameters <code class="reqn">\alpha_{h j}</code> and
<code class="reqn">\gamma_{j l}</code> are fit by ordinary linear regression assuming <code class="reqn">\beta_{h j k} = 0</code>.
Next, <code class="reqn">\beta_{h j k}</code> are fit and tested by
linear ridge regression (see documentation for <a href="#topic+ctassoc">ctassoc</a>).
</p>


<h3>Value</h3>

<p>The estimate, statistic, p.value are written to the specified file.
</p>


<h3>See Also</h3>

<p>ctassoc
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(GSE79262small)
X    = GSE79262small$X
Xpos = GSE79262small$Xpos
W    = GSE79262small$W
Y    = GSE79262small$Y
Ypos = GSE79262small$Ypos
C    = GSE79262small$C
X    = X[seq(1, 3601, 100), ] # for brevity
Xpos = Xpos[seq(1, 3601, 100)]
ctcisQTL(X, Xpos, W, Y, Ypos, C = C)


</code></pre>

<hr>
<h2 id='GSE42861small'>Small Subset of GSE42861 Dataset From GEO</h2><span id='topic+GSE42861small'></span>

<h3>Description</h3>

<p>The dataset includes 336 rheumatoid arthritis cases and 322 controls.
A subset of 500 CpG sites were randomly selected from the original EWAS dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GSE42861small)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 4.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE42861">GEO</a>
</p>


<h3>See Also</h3>

<p>ctassoc
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GSE42861small)
X = GSE42861small$X
W = GSE42861small$W
Y = GSE42861small$Y
Y = Y[seq(1, 20), ] # for brevity
C = GSE42861small$C
result = ctassoc(X, W, Y, C = C)
result$coefficients
</code></pre>

<hr>
<h2 id='GSE79262small'>Small Subset of GSE79262 Dataset From GEO</h2><span id='topic+GSE79262small'></span>

<h3>Description</h3>

<p>The dataset includes 53 samples.
A subset of 737 CpG sites and 3624 SNPs within Chr1:100,000,000-110,000,000
were selected from the original EWAS dataset.
DNA methylation was measured in T cells.
The estimated proportion of CD4T, CD8T, NK cells are saved in W.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GSE79262small)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 6.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE79262">GEO</a>
</p>


<h3>See Also</h3>

<p>ctcisQTL
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GSE79262small)
X    = GSE79262small$X
Xpos = GSE79262small$Xpos
W    = GSE79262small$W
Y    = GSE79262small$Y
Ypos = GSE79262small$Ypos
C    = GSE79262small$C
X    = X[seq(1, 3001, 100), ] # for brevity
Xpos = Xpos[seq(1, 3001, 100)]
Y    = Y[seq(1, 501, 100), ]
Ypos = Ypos[seq(1, 501, 100)]
ctcisQTL(X, Xpos, W, Y, Ypos, C = C)
</code></pre>

<hr>
<h2 id='GTExsmall'>Small Subset of GTEx Dataset</h2><span id='topic+GTExsmall'></span>

<h3>Description</h3>

<p>The dataset includes gene expression measured in whole blood for 389 samples.
A subset of 500 genes were randomly selected from the original dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GTExsmall)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 4.
</p>


<h3>Source</h3>

<p><a href="https://gtexportal.org">GTEx</a>
</p>


<h3>See Also</h3>

<p>ctassoc
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GTExsmall)
X = GTExsmall$X
W = GTExsmall$W
Y = GTExsmall$Y + 1
Y = Y[seq(1, 20), ] # for brevity
C = GTExsmall$C
result = ctassoc(X, W, Y, C = C)
result$coefficients
</code></pre>

<hr>
<h2 id='rrs.fit'>Fitting reduced-rank ridge regression with given rank and shrinkage penalty</h2><span id='topic+rrs.fit'></span>

<h3>Description</h3>

<p>Fitting reduced-rank ridge regression with given rank and shrinkage penalty
This is a modification of rrs.fit in rrpack version 0.1-6.
In order to handle extremely large q = ncol(Y),
generation of a q by q matrix is avoided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rrs.fit(Y, X, nrank = min(ncol(Y), ncol(X)), lambda = 1, coefSVD = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rrs.fit_+3A_y">Y</code></td>
<td>
<p>a matrix of response (n by q)</p>
</td></tr>
<tr><td><code id="rrs.fit_+3A_x">X</code></td>
<td>
<p>a matrix of covariate (n by p)</p>
</td></tr>
<tr><td><code id="rrs.fit_+3A_nrank">nrank</code></td>
<td>
<p>an integer specifying the desired rank</p>
</td></tr>
<tr><td><code id="rrs.fit_+3A_lambda">lambda</code></td>
<td>
<p>tunging parameter for the ridge penalty</p>
</td></tr>
<tr><td><code id="rrs.fit_+3A_coefsvd">coefSVD</code></td>
<td>
<p>logical indicating the need for SVD for the
coeffient matrix int the output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>S3 <code>rrr</code> object, a list consisting of
</p>
<table>
<tr><td><code>coef</code></td>
<td>
<p>coefficient of rrs</p>
</td></tr>
<tr><td><code>coef.ls</code></td>
<td>
<p>coefficient of least square</p>
</td></tr>
<tr><td><code>fitted</code></td>
<td>
<p>fitted value of rrs</p>
</td></tr>
<tr><td><code>fitted.ls</code></td>
<td>
<p>fitted value of least square</p>
</td></tr>
<tr><td><code>A</code></td>
<td>
<p>right singular matrix</p>
</td></tr>
<tr><td><code>Ad</code></td>
<td>
<p>sigular value vector</p>
</td></tr>
<tr><td><code>nrank</code></td>
<td>
<p>rank of the fitted rrr</p>
</td></tr>
</table>


<h3>References</h3>

<p>Mukherjee, A. and Zhu, J. (2011) Reduced rank ridge regression and its
kernal extensions.
</p>
<p>Mukherjee, A., Chen, K., Wang, N. and Zhu, J. (2015) On the degrees of
freedom of reduced-rank estimators in multivariate
regression. <em>Biometrika</em>, 102, 457&ndash;477.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Y &lt;- matrix(rnorm(400), 100, 4)
X &lt;- matrix(rnorm(800), 100, 8)
rfit &lt;- rrs.fit(Y, X)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
