<!DOCTYPE html><html><head><title>Help for package OmnibusFisher</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {OmnibusFisher}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#example_data'><p>This is the data for examples</p></a></li>
<li><a href='#OmnibusFisher'><p>A modified Fisher’s method (Omnibus-Fisher) to combine separate p-values of SNPs, RNA expressions and DNA methylations into an overall gene-level p-value</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Modified Fisher’s Method to Test Overall Gene-Level Effect</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Qi Yan</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Qi Yan &lt;qiy17@pitt.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The separate p-values of SNPs, RNA expressions and DNA
    methylations are calculated by KM regression. The correlation between
    different omics data are taken into account. This method can be applied to
    either samples with all three types of omics data or samples with two types.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>CompQuadForm, stringr</td>
</tr>
<tr>
<td>Imports:</td>
<td>survey</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.0</td>
</tr>
<tr>
<td>Collate:</td>
<td>'OmnibusFisher_iteration_GM.R' 'OmnibusFisher_iteration_GMR.R'
'OmnibusFisher_iteration_GR.R' 'OmnibusFisher_iteration_MR.R'
'OmnibusFisher_outer.R' 'example_data.R'</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-12-10 15:52:18 UTC; qiyan</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-12-19 08:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='example_data'>This is the data for examples</h2><span id='topic+All_header'></span><span id='topic+G'></span><span id='topic+M'></span><span id='topic+R'></span><span id='topic+example_data'></span><span id='topic+pheno'></span>

<h3>Description</h3>


<ul>
<li><p> All_header. ID for all samples that are available in at least one type of data
</p>
</li>
<li><p> pheno. phenotype file. 1st column is ID, 2nd column is disease status, 3rd column is age, 4th column is gender
</p>
</li>
<li><p> G. genotypes for 3 genes. G[[i]] is for the ith genes. In each G[[i]], the 1st column is ID and the rest columns are genotypes
</p>
</li>
<li><p> M. methylated sites 3 genes. M[[i]] is for the ith genes. In each M[[i]], the 1st column is ID and the rest columns are methylated sites
</p>
</li>
<li><p> R. RNA expression for 3 genes. R[[i]] is for the ith genes. In each R[[i]], the 1st column is ID and the rest columns (usually one column) are RNA expression
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(example_data)
</code></pre>

<hr>
<h2 id='OmnibusFisher'>A modified Fisher’s method (Omnibus-Fisher) to combine separate p-values of SNPs, RNA expressions and DNA methylations into an overall gene-level p-value</h2><span id='topic+OmnibusFisher'></span>

<h3>Description</h3>

<p>* Each sample does not have to have all three types of omics data; each gene needs to have all three types of omics data mapped to it. <br />
* For example, 1,000 samples have SNPs mapped to 20,000 genes; 500 samples have methylated sites mapped to 18,000 genes; 300 samples have 16,000 expressed genes. <br />
* Then, all 1,000 samples (500 and 300 samples are included in the total 1,000 samples) contribute to the test. We are interested in the overlapped genes (e.g., 16,000 genes with SNPs, methylated sites and expression genes mapped to them)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OmnibusFisher(pheno, full_id, G = NULL, M = NULL, R = NULL,
  exprs_G = NULL, exprs_M = NULL, exprs_R = NULL, type, optimal = FALSE,
  perturb_iteration = NULL, method = "kuonen")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OmnibusFisher_+3A_pheno">pheno</code></td>
<td>
<p>A matrix of sample ID, trait (i.e., y) and covariates (class: data.frame).</p>
</td></tr>
<tr><td><code id="OmnibusFisher_+3A_full_id">full_id</code></td>
<td>
<p>A vector of sample ID. This vector should include all IDs. In other words, samples with all 3 types of omics data, with 2 types and with 1 type should have their IDs in the vector (class: data.frame).</p>
</td></tr>
<tr><td><code id="OmnibusFisher_+3A_g">G</code></td>
<td>
<p>A matrix of genotypes in a gene. The 1st column is sample ID. Each column is a SNP in the following columns (class: data.frame).</p>
</td></tr>
<tr><td><code id="OmnibusFisher_+3A_m">M</code></td>
<td>
<p>A matrix of methylated sites in a gene. The 1st column is sample ID. Each column is a methylated site in the following columns (class: data.frame).</p>
</td></tr>
<tr><td><code id="OmnibusFisher_+3A_r">R</code></td>
<td>
<p>A matrix of RNA expression probes in a gene (for microarray, one gene could have multiple probes mapped; for RNAseq, one gene always has one value). The 1st column is sample ID. For microarray, each column is a probe in the following columns; for RNAseq, the 2nd column is the expression value (only two columns) (class: data.frame).</p>
</td></tr>
<tr><td><code id="OmnibusFisher_+3A_exprs_g">exprs_G</code></td>
<td>
<p>Regression model for SNPs under the null hypothesis (i.e., SNP effect is zero), y = cov1 + cov2 + ... + covp.</p>
</td></tr>
<tr><td><code id="OmnibusFisher_+3A_exprs_m">exprs_M</code></td>
<td>
<p>Regression model for DNA methylation under the null hypothesis (i.e., methylation effect is zero), y = cov1 + cov2 + ... + covp.</p>
</td></tr>
<tr><td><code id="OmnibusFisher_+3A_exprs_r">exprs_R</code></td>
<td>
<p>Regression model for RNA expression under the null hypothesis (i.e., RNA expression effect is zero), y = cov1 + cov2 + ... + covp.</p>
</td></tr>
<tr><td><code id="OmnibusFisher_+3A_type">type</code></td>
<td>
<p>Either type=&quot;binary&quot; or type=&quot;continuous&quot; for binary or continuous traits.</p>
</td></tr>
<tr><td><code id="OmnibusFisher_+3A_optimal">optimal</code></td>
<td>
<p>Whether use optimal method to automatically search for the disease model (perturbation invovled).</p>
</td></tr>
<tr><td><code id="OmnibusFisher_+3A_perturb_iteration">perturb_iteration</code></td>
<td>
<p>The number of perturbation iterations when using optimal method. For example, 1,000,000, then the lowest p-value can be obtained for perturbation method is 1/1,000,001.</p>
</td></tr>
<tr><td><code id="OmnibusFisher_+3A_method">method</code></td>
<td>
<p>Method used to approximately calculate p-values: &quot;kuonen&quot; or &quot;davies&quot;. Default &quot;kuonen&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>1. pval_GMR_pert:   the overall gene-level p-value automatically searching for the optimal disease model, when inputting three types of data. <br />
2. pval_GMR_tri:   the overall gene-level p-value assuming all three types of data in the disease model, when inputting three types of data. <br />
3. pval_GM_tri/pval_GR_tri/pval_MR_tri:   the overall gene-level p-value automatically searching for the optimal disease model, when inputting two types of data. <br />
4. pval_GM_tri/pval_GR_tri/pval_MR_tri:   the overall gene-level p-value assuming two types of data in the disease model, when inputting two types of data. <br />
5. pval_G/pval_M/pval_R:   the gene-level p-value for single type of data. <br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>################
### Examples ###
################
data("example_data")
set.seed(123)
exprs_G = exprs_M = exprs_R = "aff ~ age + sex"
### SNPs (G), DNA methylations (M) and RNA expressions (R) ###
results&lt;-list()
for(i in 1:1){ #change to 1:3 for 3 genes
 results[[i]]&lt;-OmnibusFisher(pheno=pheno, full_id=All_header, G=G[[i]], M=M[[i]],
 R=R[[i]], exprs_G=exprs_G, exprs_M=exprs_M, exprs_R=exprs_R, type="binary")
 # G[[1]] includes SNPs in gene1;
 # M[[1]] includes methylated sites in gene1;
 # R[[1]] includes gene expression probes in gene1 (or single gene1 expression value).
}
### SNPs (G) and DNA methylations (M) ###
results&lt;-list()
for(i in 1:1){
 results[[i]]&lt;-OmnibusFisher(pheno=pheno, full_id=All_header, G=G[[i]], M=M[[i]],
 exprs_G=exprs_G, exprs_M=exprs_M, type="binary")
}
### SNPs (G) and RNA expressions (R) ###
# results[[i]]&lt;-OmnibusFisher(pheno=pheno, full_id=All_header, G=G[[i]], R=R[[i]],
# exprs_G=exprs_G, exprs_R=exprs_R, type="binary")
### DNA methylations (M) and RNA expressions (R) ###
# results[[i]]&lt;-OmnibusFisher(pheno=pheno, full_id=All_header, R=R[[i]], M=M[[i]],
# exprs_R=exprs_R, exprs_M=exprs_M, type="binary")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
