<!DOCTYPE html><html><head><title>Help for package omu</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {omu}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#assign_hierarchy'><p>Assign hierarchy metadata</p></a></li>
<li><a href='#c57_nos2KO_mouse_countDF'><p>c57b6J nos2KO metabolomics count matrix</p></a></li>
<li><a href='#c57_nos2KO_mouse_metadata'><p>c57b6J nos2KO meta data</p></a></li>
<li><a href='#check_zeros'><p>Check data for zeros across samples within factor levels.</p>
Will determine if there are more zeros than a user specified threshold within any given factor level(s). Returns
a vector of Metabolites that are 0 above the threshold in any given factor level.</a></li>
<li><a href='#count_fold_changes'><p>Get counts for significant fold changes by metabolite class.</p></a></li>
<li><a href='#get_seqs'><p>Get nucleotide and amino acid sequences for genes</p></a></li>
<li><a href='#KEGG_gather'><p>Gather metadata from KEGG for metabolites</p></a></li>
<li><a href='#make_omelette'><p>Get metadata from KEGG API</p></a></li>
<li><a href='#omu_anova'><p>Perform anova</p></a></li>
<li><a href='#omu_summary'><p>omu_summary</p>
Performs comparison of means between two independent variables, standard deviation, standard error, FDR correction, fold change, log2FoldChange.
The order effects the fold change values</a></li>
<li><a href='#PCA_plot'><p>Create a PCA plot</p></a></li>
<li><a href='#pie_chart'><p>Create a pie chart</p></a></li>
<li><a href='#plate_omelette'><p>plate_omelette</p>
Internal method for KEGG_Gather which parses flat text files</a></li>
<li><a href='#plate_omelette_rxnko'><p>Clean up orthology metadata</p></a></li>
<li><a href='#plot_bar'><p>Create a bar plot</p></a></li>
<li><a href='#plot_boxplot'><p>Create a box plot</p></a></li>
<li><a href='#plot_heatmap'><p>Create a heatmap</p></a></li>
<li><a href='#plot_rf_PCA'><p>plot_rf_PCA</p></a></li>
<li><a href='#plot_variable_importance'><p>plot_variable_importance</p></a></li>
<li><a href='#plot_volcano'><p>Create a volcano plot</p></a></li>
<li><a href='#ra_table'><p>Creates a ratio table from the count_fold_changes function output.</p></a></li>
<li><a href='#random_forest'><p>random_forest</p>
Perform a classification or regression random forest model</a></li>
<li><a href='#read_metabo'><p>Import a metabolomics count data frame</p></a></li>
<li><a href='#transform_metabolites'><p>transform_metabolites</p></a></li>
<li><a href='#transform_samples'><p>transform_samples</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>A Metabolomics Analysis Tool for Intuitive Figures and
Convenient Metadata Collection</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Facilitates the creation of intuitive figures to describe metabolomics data by utilizing Kyoto Encyclopedia of Genes and Genomes (KEGG) hierarchy data, and gathers functional orthology and gene data from the KEGG-REST API.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>plyr, dplyr, stringr, httr, ggfortify, ggplot2, magrittr,
tidyr, broom, FSA, rstatix, randomForest, caret</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/connor-reid-tiffany/Omu">https://github.com/connor-reid-tiffany/Omu</a>,
<a href="https://www.kegg.jp/kegg/rest/keggapi.html">https://www.kegg.jp/kegg/rest/keggapi.html</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/connor-reid-tiffany/Omu/issues">https://github.com/connor-reid-tiffany/Omu/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-06 16:12:49 UTC; connor</td>
</tr>
<tr>
<td>Author:</td>
<td>Connor Tiffany [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Connor Tiffany &lt;tiffanyc1@chop.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-06 23:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='assign_hierarchy'>Assign hierarchy metadata</h2><span id='topic+assign_hierarchy'></span>

<h3>Description</h3>

<p>Assigns hierarchy metadata to a metabolomics count matrix using identifier values.
It can assign KEGG compound hierarchy, orthology hierarchy, or organism hierarchy data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assign_hierarchy(count_data, keep_unknowns, identifier)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assign_hierarchy_+3A_count_data">count_data</code></td>
<td>
<p>a metabolomics count data frame with either a KEGG compound, orthology,
or a gene identifier column</p>
</td></tr>
<tr><td><code id="assign_hierarchy_+3A_keep_unknowns">keep_unknowns</code></td>
<td>
<p>a boolean of either TRUE or FALSE. TRUE keeps unannotated compounds,
FALSE removes them</p>
</td></tr>
<tr><td><code id="assign_hierarchy_+3A_identifier">identifier</code></td>
<td>
<p>a string that is either &quot;KEGG&quot; for metabolite, &quot;KO&quot; for orthology,
&quot;Prokaryote&quot; for organism, or &quot;Eukaryote&quot; for organism</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>assign_hierarchy(count_data = c57_nos2KO_mouse_countDF, keep_unknowns = TRUE, identifier = "KEGG")
</code></pre>

<hr>
<h2 id='c57_nos2KO_mouse_countDF'>c57b6J nos2KO metabolomics count matrix</h2><span id='topic+c57_nos2KO_mouse_countDF'></span>

<h3>Description</h3>

<p>A dataset containing metabolomics counts for an experiment done using
c57b6J wild type and c57b6J nos2 knockout mice
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c57_nos2KO_mouse_countDF
</code></pre>


<h3>Format</h3>

<p>A data frame with 668 rows and 36 variables:
</p>

<hr>
<h2 id='c57_nos2KO_mouse_metadata'>c57b6J nos2KO meta data</h2><span id='topic+c57_nos2KO_mouse_metadata'></span>

<h3>Description</h3>

<p>A a meta data file for the c57b6J metabolomics count matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c57_nos2KO_mouse_metadata
</code></pre>


<h3>Format</h3>

<p>A data frame with 29 rows and 4 variables:
</p>

<hr>
<h2 id='check_zeros'>Check data for zeros across samples within factor levels.
Will determine if there are more zeros than a user specified threshold within any given factor level(s). Returns
a vector of Metabolites that are 0 above the threshold in any given factor level.</h2><span id='topic+check_zeros'></span>

<h3>Description</h3>

<p>Check data for zeros across samples within factor levels.
Will determine if there are more zeros than a user specified threshold within any given factor level(s). Returns
a vector of Metabolites that are 0 above the threshold in any given factor level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_zeros(
  count_data,
  metadata,
  numerator = NULL,
  denominator = NULL,
  threshold = 25,
  response_variable = "Metabolite",
  Factor
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_zeros_+3A_count_data">count_data</code></td>
<td>
<p>A metabolomics count data frame</p>
</td></tr>
<tr><td><code id="check_zeros_+3A_metadata">metadata</code></td>
<td>
<p>Metadata dataframe for the metabolomics count data frame</p>
</td></tr>
<tr><td><code id="check_zeros_+3A_numerator">numerator</code></td>
<td>
<p>String of the first independent variable you wish to test. Defualt is NULL</p>
</td></tr>
<tr><td><code id="check_zeros_+3A_denominator">denominator</code></td>
<td>
<p>String of the second independent variable you wish to test. Default is NULL.</p>
</td></tr>
<tr><td><code id="check_zeros_+3A_threshold">threshold</code></td>
<td>
<p>Integer. A percentage threshold for the number of zeros in a Metabolite. Default is 25.</p>
</td></tr>
<tr><td><code id="check_zeros_+3A_response_variable">response_variable</code></td>
<td>
<p>String of the column header for the response variables,
usually &quot;Metabolite&quot;</p>
</td></tr>
<tr><td><code id="check_zeros_+3A_factor">Factor</code></td>
<td>
<p>A factor with levels to test for zeros.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
check_zeros(count_data = c57_nos2KO_mouse_countDF, metadata = c57_nos2KO_mouse_metadata,
Factor = "Treatment")

check_zeros(count_data = c57_nos2KO_mouse_countDF, metadata = c57_nos2KO_mouse_metadata,
Factor = "Treatment",numerator = "Strep", denominator = "Mock", threshold = 10)
</code></pre>

<hr>
<h2 id='count_fold_changes'>Get counts for significant fold changes by metabolite class.</h2><span id='topic+count_fold_changes'></span>

<h3>Description</h3>

<p>Takes an input data frame from the output of omu_summary and creates a
data frame of counts for significantly changed metabolites by class hierarchy data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_fold_changes(count_data, column, sig_threshold, keep_unknowns)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_fold_changes_+3A_count_data">count_data</code></td>
<td>
<p>Output dataframe from the omu_summary function or omu_anova.</p>
</td></tr>
<tr><td><code id="count_fold_changes_+3A_column">column</code></td>
<td>
<p>Metabolite metadata you want to group by, i.e. &quot;Class&quot;, &quot;Subclass_1&quot;.</p>
</td></tr>
<tr><td><code id="count_fold_changes_+3A_sig_threshold">sig_threshold</code></td>
<td>
<p>Significance threshold for compounds that go towars the count,
sig_threshold = 0.05</p>
</td></tr>
<tr><td><code id="count_fold_changes_+3A_keep_unknowns">keep_unknowns</code></td>
<td>
<p>TRUE or FALSE for whether to drop compounds that weren't assigned
hierarchy metadata</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>c57_nos2KO_mouse_countDF &lt;- assign_hierarchy(c57_nos2KO_mouse_countDF, TRUE, "KEGG")

t_test_df &lt;- omu_summary(count_data = c57_nos2KO_mouse_countDF,
metadata = c57_nos2KO_mouse_metadata,
numerator = "Strep", denominator = "Mock", response_variable = "Metabolite",
Factor = "Treatment", log_transform = TRUE, p_adjust = "BH", test_type = "welch")

fold_change_counts &lt;- count_fold_changes(count_data = t_test_df,
column = "Class", sig_threshold = 0.05, keep_unknowns = "FALSE")
</code></pre>

<hr>
<h2 id='get_seqs'>Get nucleotide and amino acid sequences for genes</h2><span id='topic+get_seqs'></span>

<h3>Description</h3>

<p>Function that gets nt and aa seqs for gene data from KEGG_gather
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_seqs(gene_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_seqs_+3A_gene_data">gene_data</code></td>
<td>
<p>A dataframe with genes from KEGG_gather, with class seqs</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gene_data &lt;- c57_nos2KO_mouse_countDF[(1:2),]

gene_data &lt;- KEGG_gather(gene_data)

gene_data &lt;- KEGG_gather(gene_data)
gene_data &lt;- gene_data[1:2,]

gene_data &lt;- get_seqs(gene_data)

## End(Not run)
</code></pre>

<hr>
<h2 id='KEGG_gather'>Gather metadata from KEGG for metabolites</h2><span id='topic+KEGG_gather'></span><span id='topic+KEGG_gather.cpd'></span><span id='topic+KEGG_gather.rxn'></span><span id='topic+KEGG_gather.KO'></span>

<h3>Description</h3>

<p>Method for gathering metadata from the KEGG API.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KEGG_gather(count_data)

## S3 method for class 'cpd'
KEGG_gather(count_data)

## S3 method for class 'rxn'
KEGG_gather(count_data)

## S3 method for class 'KO'
KEGG_gather(count_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KEGG_gather_+3A_count_data">count_data</code></td>
<td>
<p>A metabolomics count dataframe with a KEGG identifier columns</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
count_data &lt;- assign_hierarchy(count_data = c57_nos2KO_mouse_countDF,
keep_unknowns = TRUE, identifier = "KEGG")

count_data &lt;- subset(count_data, Subclass_2=="Aldoses")

count_data &lt;- KEGG_gather(count_data = count_data)

## End(Not run)
</code></pre>

<hr>
<h2 id='make_omelette'>Get metadata from KEGG API</h2><span id='topic+make_omelette'></span>

<h3>Description</h3>

<p>Internal function for KEGG_Gather
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_omelette(count_data, column, first_char)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_omelette_+3A_count_data">count_data</code></td>
<td>
<p>The metabolomics count data</p>
</td></tr>
<tr><td><code id="make_omelette_+3A_column">column</code></td>
<td>
<p>The name of the KEGG identifier being sent to the KEGG API</p>
</td></tr>
<tr><td><code id="make_omelette_+3A_first_char">first_char</code></td>
<td>
<p>firct character in number being fed to KEGG database</p>
</td></tr>
</table>

<hr>
<h2 id='omu_anova'>Perform anova</h2><span id='topic+omu_anova'></span>

<h3>Description</h3>

<p>Performs an anova across all response variables, followed by a Tukeys test on every possible
contrast in your model and calculates group means and fold changes for each contrast. Returns a list of
data frames for each contrast, and includes a dataframe of model residuals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>omu_anova(
  count_data,
  metadata,
  response_variable = "Metabolite",
  model,
  log_transform = FALSE,
  method = "anova"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="omu_anova_+3A_count_data">count_data</code></td>
<td>
<p>A metabolomics count data frame</p>
</td></tr>
<tr><td><code id="omu_anova_+3A_metadata">metadata</code></td>
<td>
<p>Metadata dataframe for the metabolomics count data frame</p>
</td></tr>
<tr><td><code id="omu_anova_+3A_response_variable">response_variable</code></td>
<td>
<p>String of the column header for the response variables,
usually &quot;Metabolite&quot;</p>
</td></tr>
<tr><td><code id="omu_anova_+3A_model">model</code></td>
<td>
<p>A formual class object, see ?formula for more info on formulas in R.
an interaction between independent variables. Optional parameter</p>
</td></tr>
<tr><td><code id="omu_anova_+3A_log_transform">log_transform</code></td>
<td>
<p>Boolean of TRUE or FALSE for whether or not you wish to log transform
your metabolite counts</p>
</td></tr>
<tr><td><code id="omu_anova_+3A_method">method</code></td>
<td>
<p>A string of 'anova', 'kruskal', or 'welch'. anova performs an anova with a post hoc
tukeys test, kruskal performs a kruskal wallis with a post hoc dunn test, welch performs a
welch's anova with a post hoc games howell test</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
anova_df &lt;- omu_anova(count_data = c57_nos2KO_mouse_countDF, metadata = c57_nos2KO_mouse_metadata,
response_variable = "Metabolite", model = ~ Treatment, log_transform = TRUE)

anova_df &lt;- omu_anova(count_data = c57_nos2KO_mouse_countDF, metadata = c57_nos2KO_mouse_metadata,
response_variable = "Metabolite", model = ~ Treatment + Background, log_transform = TRUE)

anova_df &lt;- omu_anova(count_data = c57_nos2KO_mouse_countDF, metadata = c57_nos2KO_mouse_metadata,
response_variable = "Metabolite", model = ~ Treatment + Background + Treatment*Background,
log_transform = TRUE)

</code></pre>

<hr>
<h2 id='omu_summary'>omu_summary
Performs comparison of means between two independent variables, standard deviation, standard error, FDR correction, fold change, log2FoldChange.
The order effects the fold change values</h2><span id='topic+omu_summary'></span>

<h3>Description</h3>

<p>omu_summary
Performs comparison of means between two independent variables, standard deviation, standard error, FDR correction, fold change, log2FoldChange.
The order effects the fold change values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>omu_summary(
  count_data,
  metadata,
  numerator,
  denominator,
  response_variable = "Metabolite",
  Factor,
  log_transform = FALSE,
  p_adjust = "BH",
  test_type = "welch",
  paired = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="omu_summary_+3A_count_data">count_data</code></td>
<td>
<p>should be a metabolomics count data frame</p>
</td></tr>
<tr><td><code id="omu_summary_+3A_metadata">metadata</code></td>
<td>
<p>is meta data</p>
</td></tr>
<tr><td><code id="omu_summary_+3A_numerator">numerator</code></td>
<td>
<p>is the variable you wish to compare against the denominator, in quotes</p>
</td></tr>
<tr><td><code id="omu_summary_+3A_denominator">denominator</code></td>
<td>
<p>see above, in quotes</p>
</td></tr>
<tr><td><code id="omu_summary_+3A_response_variable">response_variable</code></td>
<td>
<p>the name of the column with your response variables</p>
</td></tr>
<tr><td><code id="omu_summary_+3A_factor">Factor</code></td>
<td>
<p>the column name for your independent variables</p>
</td></tr>
<tr><td><code id="omu_summary_+3A_log_transform">log_transform</code></td>
<td>
<p>TRUE or FALSE value for whether or not log transformation of data is performed
before the t test</p>
</td></tr>
<tr><td><code id="omu_summary_+3A_p_adjust">p_adjust</code></td>
<td>
<p>Method for adjusting the p value, i.e. &quot;BH&quot;</p>
</td></tr>
<tr><td><code id="omu_summary_+3A_test_type">test_type</code></td>
<td>
<p>One of &quot;mwu&quot;, &quot;students&quot;, or &quot;welch&quot; to determine which model to use</p>
</td></tr>
<tr><td><code id="omu_summary_+3A_paired">paired</code></td>
<td>
<p>A boolean of TRUE or FALSE. If TRUE, performs a paired sample test. To perform a paired sample test,
metadata must have a column named 'ID' containing the subject IDs.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
omu_summary(count_data = c57_nos2KO_mouse_countDF, metadata = c57_nos2KO_mouse_metadata,
numerator = "Strep", denominator = "Mock", response_variable = "Metabolite", Factor = "Treatment",
log_transform = TRUE, p_adjust = "BH", test_type = "welch")
</code></pre>

<hr>
<h2 id='PCA_plot'>Create a PCA plot</h2><span id='topic+PCA_plot'></span>

<h3>Description</h3>

<p>Performs an ordination and outputs a PCA plot using a metabolomics
count data frame and metabolomics metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCA_plot(
  count_data,
  metadata,
  variable,
  color,
  response_variable = "Metabolite",
  label = FALSE,
  size = 2,
  ellipse = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PCA_plot_+3A_count_data">count_data</code></td>
<td>
<p>Metabolomics count data</p>
</td></tr>
<tr><td><code id="PCA_plot_+3A_metadata">metadata</code></td>
<td>
<p>Metabolomics metadata</p>
</td></tr>
<tr><td><code id="PCA_plot_+3A_variable">variable</code></td>
<td>
<p>The independent variable you wish to compare and contrast</p>
</td></tr>
<tr><td><code id="PCA_plot_+3A_color">color</code></td>
<td>
<p>String of what you want to color by. Usually should be the same as variable.</p>
</td></tr>
<tr><td><code id="PCA_plot_+3A_response_variable">response_variable</code></td>
<td>
<p>String of the response_variable, usually should be &quot;Metabolite&quot;</p>
</td></tr>
<tr><td><code id="PCA_plot_+3A_label">label</code></td>
<td>
<p>TRUE or FALSE, whether to add point labels or not</p>
</td></tr>
<tr><td><code id="PCA_plot_+3A_size">size</code></td>
<td>
<p>An integer for point size.</p>
</td></tr>
<tr><td><code id="PCA_plot_+3A_ellipse">ellipse</code></td>
<td>
<p>TRUE or FALSE, whether to add confidence interval ellipses or not.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>PCA_plot(count_data = c57_nos2KO_mouse_countDF, metadata = c57_nos2KO_mouse_metadata,
variable = "Treatment", color = "Treatment", response_variable = "Metabolite")
</code></pre>

<hr>
<h2 id='pie_chart'>Create a pie chart</h2><span id='topic+pie_chart'></span>

<h3>Description</h3>

<p>Creates a pie chart as ggplot2 object using the output from ra_table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pie_chart(ratio_data, variable, column, color)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pie_chart_+3A_ratio_data">ratio_data</code></td>
<td>
<p>a dataframe object of percents. output from ra_table function</p>
</td></tr>
<tr><td><code id="pie_chart_+3A_variable">variable</code></td>
<td>
<p>The metadata variable you are measuring, i.e. &quot;Class&quot;</p>
</td></tr>
<tr><td><code id="pie_chart_+3A_column">column</code></td>
<td>
<p>either &quot;Increase&quot;, &quot;Decrease&quot;, or &quot;Significant_Changes&quot;</p>
</td></tr>
<tr><td><code id="pie_chart_+3A_color">color</code></td>
<td>
<p>string denoting color for outline. use NA for no outline</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>c57_nos2KO_mouse_countDF &lt;- assign_hierarchy(c57_nos2KO_mouse_countDF, TRUE, "KEGG")

t_test_df &lt;- omu_summary(count_data = c57_nos2KO_mouse_countDF,
metadata = c57_nos2KO_mouse_metadata,
numerator = "Strep", denominator = "Mock", response_variable = "Metabolite",
Factor = "Treatment",
log_transform = TRUE, p_adjust = "BH", test_type = "welch")

fold_change_counts &lt;- count_fold_changes(count_data = t_test_df,
column = "Class", sig_threshold = 0.05, keep_unknowns = FALSE)

ra_table &lt;- ra_table(fc_data = fold_change_counts, variable = "Class")

pie_chart(ratio_data = ra_table, variable = "Class", column = "Decrease", color = "black")
</code></pre>

<hr>
<h2 id='plate_omelette'>plate_omelette
Internal method for KEGG_Gather which parses flat text files</h2><span id='topic+plate_omelette'></span><span id='topic+plate_omelette.rxn'></span><span id='topic+plate_omelette.genes'></span><span id='topic+plate_omelette.KO'></span>

<h3>Description</h3>

<p>plate_omelette
Internal method for KEGG_Gather which parses flat text files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plate_omelette(output)

## S3 method for class 'rxn'
plate_omelette(output)

## S3 method for class 'genes'
plate_omelette(output)

## S3 method for class 'KO'
plate_omelette(output)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plate_omelette_+3A_output">output</code></td>
<td>
<p>The metabolomics count dataframe</p>
</td></tr>
</table>

<hr>
<h2 id='plate_omelette_rxnko'>Clean up orthology metadata</h2><span id='topic+plate_omelette_rxnko'></span>

<h3>Description</h3>

<p>Internal function for KEGG_Gather.rxn method
KEGG_Gather.rxn requires dispatch on multiple elements, so
There was no way to incorporate as a method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plate_omelette_rxnko(output)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plate_omelette_rxnko_+3A_output">output</code></td>
<td>
<p>output from plate_omelette</p>
</td></tr>
</table>

<hr>
<h2 id='plot_bar'>Create a bar plot</h2><span id='topic+plot_bar'></span>

<h3>Description</h3>

<p>Creates a ggplot2 object using the output file from the count_fold_changes function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_bar(fc_data, fill, size = c(1, 1), outline_color = c("black", "black"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_bar_+3A_fc_data">fc_data</code></td>
<td>
<p>The output file from Count_Fold_Changes</p>
</td></tr>
<tr><td><code id="plot_bar_+3A_fill">fill</code></td>
<td>
<p>A character vector of length 2 containing colors for filling the bars,
the first color is for the &quot;Decrease&quot; bar while the second
is for &quot;Increase&quot;</p>
</td></tr>
<tr><td><code id="plot_bar_+3A_size">size</code></td>
<td>
<p>A numeric vector of 2 numbers for the size of the bar outlines.</p>
</td></tr>
<tr><td><code id="plot_bar_+3A_outline_color">outline_color</code></td>
<td>
<p>A character vector of length 2 containing colors for the bar outlines</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>c57_nos2KO_mouse_countDF &lt;- assign_hierarchy(c57_nos2KO_mouse_countDF, TRUE, "KEGG")

t_test_df &lt;- omu_summary(count_data = c57_nos2KO_mouse_countDF,
metadata = c57_nos2KO_mouse_metadata, numerator = "Strep", denominator = "Mock",
response_variable = "Metabolite", Factor = "Treatment",
log_transform = TRUE, p_adjust = "BH", test_type = "welch")

fold_change_counts &lt;- count_fold_changes(count_data = t_test_df,
column = "Class", sig_threshold = 0.05, keep_unknowns = FALSE)

plot_bar(fc_data = fold_change_counts, fill = c("firebrick2", "dodgerblue2"),
outline_color = c("black", "black"), size = c(1,1))
</code></pre>

<hr>
<h2 id='plot_boxplot'>Create a box plot</h2><span id='topic+plot_boxplot'></span>

<h3>Description</h3>

<p>Takes a metabolomics count data frame and creates boxplots. It is recommended to
either subset, truncate, or agglomerate by hierarchical metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_boxplot(
  count_data,
  metadata,
  aggregate_by,
  log_transform = FALSE,
  Factor,
  response_variable = "Metabolite",
  fill_list
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_boxplot_+3A_count_data">count_data</code></td>
<td>
<p>A metabolomics count data frame, either from read_metabo or omu_summary</p>
</td></tr>
<tr><td><code id="plot_boxplot_+3A_metadata">metadata</code></td>
<td>
<p>The descriptive meta data for the samples</p>
</td></tr>
<tr><td><code id="plot_boxplot_+3A_aggregate_by">aggregate_by</code></td>
<td>
<p>Hierarchical metadata value to sum metabolite values by, i.e. &quot;Class&quot;</p>
</td></tr>
<tr><td><code id="plot_boxplot_+3A_log_transform">log_transform</code></td>
<td>
<p>TRUE or FALSE. Recommended for visualization purposes. If true data is
transformed by the natural log</p>
</td></tr>
<tr><td><code id="plot_boxplot_+3A_factor">Factor</code></td>
<td>
<p>The column name for the experimental variable</p>
</td></tr>
<tr><td><code id="plot_boxplot_+3A_response_variable">response_variable</code></td>
<td>
<p>The response variable for the data, i.e. &quot;Metabolite&quot;</p>
</td></tr>
<tr><td><code id="plot_boxplot_+3A_fill_list">fill_list</code></td>
<td>
<p>Colors for the plot which is colored by Factor, in the form of c(&quot;&quot;)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>c57_nos2KO_mouse_countDF &lt;- c57_nos2KO_mouse_countDF[1:5,]
c57_nos2KO_mouse_countDF &lt;- assign_hierarchy(c57_nos2KO_mouse_countDF, TRUE, "KEGG")

plot_boxplot(count_data = c57_nos2KO_mouse_countDF, metadata = c57_nos2KO_mouse_metadata,
log_transform = TRUE, Factor = "Treatment", response_variable = "Metabolite",
aggregate_by = "Subclass_2", fill_list = c("darkgoldenrod1", "dodgerblue2"))
</code></pre>

<hr>
<h2 id='plot_heatmap'>Create a heatmap</h2><span id='topic+plot_heatmap'></span>

<h3>Description</h3>

<p>Takes a metabolomics count data frame and creates a heatmap. It is recommended to
either subset, truncate, or agglomerate by metabolite metadata to improve legibility.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heatmap(
  count_data,
  metadata,
  Factor,
  response_variable,
  log_transform = FALSE,
  high_color,
  low_color,
  aggregate_by
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_heatmap_+3A_count_data">count_data</code></td>
<td>
<p>A metabolomics count data frame.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_metadata">metadata</code></td>
<td>
<p>The descriptive meta data for the samples.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_factor">Factor</code></td>
<td>
<p>The column name for the independent variable in your metadata.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_response_variable">response_variable</code></td>
<td>
<p>The response variable for the data, i.e. &quot;Metabolite&quot;</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_log_transform">log_transform</code></td>
<td>
<p>TRUE or FALSE. Recommended for visualization purposes. If true data is
transformed by the natural log.</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_high_color">high_color</code></td>
<td>
<p>Color for high abundance values</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_low_color">low_color</code></td>
<td>
<p>Color for low abundance values</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_aggregate_by">aggregate_by</code></td>
<td>
<p>Hierarchical metadata value to sum metabolite values by, i.e. &quot;Class&quot;</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>c57_nos2KO_mouse_countDF &lt;- assign_hierarchy(c57_nos2KO_mouse_countDF, TRUE, "KEGG")

plot_heatmap(count_data = c57_nos2KO_mouse_countDF, metadata = c57_nos2KO_mouse_metadata,
log_transform = TRUE, Factor = "Treatment", response_variable = "Metabolite",
aggregate_by = "Subclass_2", high_color = "darkgoldenrod1", low_color = "dodgerblue2")
</code></pre>

<hr>
<h2 id='plot_rf_PCA'>plot_rf_PCA</h2><span id='topic+plot_rf_PCA'></span>

<h3>Description</h3>

<p>PCA plot of the proximity matrix from a random forest classification model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_rf_PCA(rf_list, color, size, ellipse = FALSE, label = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_rf_PCA_+3A_rf_list">rf_list</code></td>
<td>
<p>The output from the random_forest function. This only works on classification models.</p>
</td></tr>
<tr><td><code id="plot_rf_PCA_+3A_color">color</code></td>
<td>
<p>A grouping factor. Use the one that was the LHS of your model parameter in the random_forest funciton</p>
</td></tr>
<tr><td><code id="plot_rf_PCA_+3A_size">size</code></td>
<td>
<p>The number for point size in the plot</p>
</td></tr>
<tr><td><code id="plot_rf_PCA_+3A_ellipse">ellipse</code></td>
<td>
<p>TRUE or FALSE. Whether to plot with confidence interval ellipses or not.</p>
</td></tr>
<tr><td><code id="plot_rf_PCA_+3A_label">label</code></td>
<td>
<p>TRUE or FALSE. Whether to include point labels or not.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>rf_list &lt;- random_forest(c57_nos2KO_mouse_countDF,c57_nos2KO_mouse_metadata,
Treatment ~.,c(60,40),500)
plot_rf_PCA(rf_list = rf_list, color = "Treatment", size = 1.5)
</code></pre>

<hr>
<h2 id='plot_variable_importance'>plot_variable_importance</h2><span id='topic+plot_variable_importance'></span>

<h3>Description</h3>

<p>Plot the variable importance from a random forest model. Mean Decrease Gini for Classification and
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_variable_importance(rf_list, color = "Class", n_metabolites = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_variable_importance_+3A_rf_list">rf_list</code></td>
<td>
<p>The output from the random_forest function</p>
</td></tr>
<tr><td><code id="plot_variable_importance_+3A_color">color</code></td>
<td>
<p>Metabolite metadata to color by</p>
</td></tr>
<tr><td><code id="plot_variable_importance_+3A_n_metabolites">n_metabolites</code></td>
<td>
<p>The number of metabolites to include. Metabolites are sorted by decreasing importance.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>rf_list &lt;- random_forest(c57_nos2KO_mouse_countDF,c57_nos2KO_mouse_metadata,
Treatment ~.,c(60,40),500)
plot_variable_importance(rf_list = rf_list, color = "Class", n_metabolites = 10)
</code></pre>

<hr>
<h2 id='plot_volcano'>Create a volcano plot</h2><span id='topic+plot_volcano'></span>

<h3>Description</h3>

<p>Creates a volcano plot as ggplot2 object using the output of omu_summary
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_volcano(
  count_data,
  column,
  size,
  strpattern,
  fill,
  sig_threshold,
  alpha,
  shape,
  color
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_volcano_+3A_count_data">count_data</code></td>
<td>
<p>The output file from the omu_summary function.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_column">column</code></td>
<td>
<p>The column with metadata you want to highlight points in the plot with,
i.e. &quot;Class&quot;</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_size">size</code></td>
<td>
<p>Size of the points in the plot</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_strpattern">strpattern</code></td>
<td>
<p>A character vector of levels of the column you want the plot to focus on,
i.e. strpattern = c(&quot;Carbohydrates&quot;, &quot;Organicacids&quot;)</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_fill">fill</code></td>
<td>
<p>A character vector of colors you want your points to be. Must be of length
1 + length(strpattern) to account for points not in strpattern.
Levels of a factor are organzed alphabetically. All levels not in the
strpattern argument will be set to NA.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_sig_threshold">sig_threshold</code></td>
<td>
<p>An integer. Creates a horizontal dashed line for a significance threshold.
i.e. sig_threshold = 0.05. Defaut value is 0.05</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_alpha">alpha</code></td>
<td>
<p>A character vector for setting transparency of factor levels.Must be of length
1 + length(strpattern) to account for points not in strpattern.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_shape">shape</code></td>
<td>
<p>A character vector for setting the shapes for your column levels.
Must be of length 1 + length(strpattern) to account for points not in strpattern.
See ggplot2 for an index of shape values.</p>
</td></tr>
<tr><td><code id="plot_volcano_+3A_color">color</code></td>
<td>
<p>A character vector of colors for the column levels.
Must be of length 1 + length(strpattern) to account for points not in strpattern.
If you choose to use shapes
with outlines, this list will set the outline colors.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>c57_nos2KO_mouse_countDF &lt;- assign_hierarchy(c57_nos2KO_mouse_countDF, TRUE, "KEGG")

t_test_df &lt;-  omu_summary(count_data = c57_nos2KO_mouse_countDF,
metadata = c57_nos2KO_mouse_metadata, numerator = "Strep", denominator = "Mock",
response_variable = "Metabolite", Factor = "Treatment",
log_transform = TRUE, p_adjust = "BH", test_type = "welch")

plot_volcano(count_data = t_test_df, column = "Class", strpattern = c("Carbohydrates"),
fill = c("firebrick2", "white"), sig_threshold = 0.05, alpha = c(1,1),
shape = c(1,24), color = c("black", "black"), size = 2)

plot_volcano(count_data = t_test_df, sig_threshold = 0.05, size = 2)
</code></pre>

<hr>
<h2 id='ra_table'>Creates a ratio table from the count_fold_changes function output.</h2><span id='topic+ra_table'></span>

<h3>Description</h3>

<p>Create a ratio table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ra_table(fc_data, variable)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ra_table_+3A_fc_data">fc_data</code></td>
<td>
<p>data frame output from the count_fold_changes function</p>
</td></tr>
<tr><td><code id="ra_table_+3A_variable">variable</code></td>
<td>
<p>metadata from count_fold_changes, i.e. &quot;Class&quot;</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>c57_nos2KO_mouse_countDF &lt;- assign_hierarchy(c57_nos2KO_mouse_countDF, TRUE, "KEGG")

t_test_df &lt;- omu_summary(count_data = c57_nos2KO_mouse_countDF,
metadata = c57_nos2KO_mouse_metadata, numerator = "Strep", denominator = "Mock",
response_variable = "Metabolite", Factor = "Treatment",
log_transform = TRUE, p_adjust = "BH", test_type = "welch")

fold_change_counts &lt;- count_fold_changes(count_data = t_test_df,
column = "Class", sig_threshold = 0.05, keep_unknowns = FALSE)

ra_table(fc_data = fold_change_counts, variable = "Class")
</code></pre>

<hr>
<h2 id='random_forest'>random_forest
Perform a classification or regression random forest model</h2><span id='topic+random_forest'></span>

<h3>Description</h3>

<p>a wrapper built around the randomForest function from package randomForest.
Returns a list with a randomForest object list, training data set, testing data set,
metabolite metadata, and confusion matrices for training and testing data
(if type was classification).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_forest(
  count_data,
  metadata,
  model,
  training_proportion = c(80, 20),
  n_tree = 500
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="random_forest_+3A_count_data">count_data</code></td>
<td>
<p>Metabolomics data</p>
</td></tr>
<tr><td><code id="random_forest_+3A_metadata">metadata</code></td>
<td>
<p>sample data</p>
</td></tr>
<tr><td><code id="random_forest_+3A_model">model</code></td>
<td>
<p>a model of format variable ~.</p>
</td></tr>
<tr><td><code id="random_forest_+3A_training_proportion">training_proportion</code></td>
<td>
<p>a numeric vector of length 2, first element is the percent of
samples to use for training the model, second element is the percent of samples used to
test the models accuracy</p>
</td></tr>
<tr><td><code id="random_forest_+3A_n_tree">n_tree</code></td>
<td>
<p>number of decision trees to create</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>rf_list &lt;- random_forest(count_data = c57_nos2KO_mouse_countDF,metadata = c57_nos2KO_mouse_metadata,
model = Treatment ~.,training_proportion = c(60,40),n_tree = 500)
</code></pre>

<hr>
<h2 id='read_metabo'>Import a metabolomics count data frame</h2><span id='topic+read_metabo'></span>

<h3>Description</h3>

<p>Wrapper for read.csv that appends the &quot;cpd&quot; class and sets blank cells to NA. Used to
import metabolomics count data into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_metabo(filepath)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_metabo_+3A_filepath">filepath</code></td>
<td>
<p>a file path to your metabolomics count data</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>filepath_to_yourdata = paste0(system.file(package = "omu"), "/extdata/read_metabo_test.csv")
count_data &lt;- read_metabo(filepath_to_yourdata)
</code></pre>

<hr>
<h2 id='transform_metabolites'>transform_metabolites</h2><span id='topic+transform_metabolites'></span>

<h3>Description</h3>

<p>A functional to transform metabolomics data across metabolites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transform_metabolites(count_data, func)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transform_metabolites_+3A_count_data">count_data</code></td>
<td>
<p>Metabolomics data</p>
</td></tr>
<tr><td><code id="transform_metabolites_+3A_func">func</code></td>
<td>
<p>a function to transform metabolites by. can be an anonymous function</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data_pareto_scaled &lt;- transform_samples(count_data = c57_nos2KO_mouse_countDF,
function(x) x/sqrt(sd(x)))
</code></pre>

<hr>
<h2 id='transform_samples'>transform_samples</h2><span id='topic+transform_samples'></span>

<h3>Description</h3>

<p>A functional to transform metabolomics data across samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transform_samples(count_data, func)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transform_samples_+3A_count_data">count_data</code></td>
<td>
<p>Metabolomics data</p>
</td></tr>
<tr><td><code id="transform_samples_+3A_func">func</code></td>
<td>
<p>a function to transform samples by. can be an anonymous function</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data_ln &lt;- transform_samples(count_data = c57_nos2KO_mouse_countDF, log)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
