<!DOCTYPE html><html><head><title>Help for package oncoPredict</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {oncoPredict}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calcPhenotype'><p>This function predicts a phenotype (drug sensitivity score) when provided with microarray or bulk RNAseq gene expression data of different platforms.</p>
The imputations are performed using ridge regression, training on a gene expression matrix where phenotype is already known.
This function integrates training and testing datasets via a user-defined procedure, and power transforming the known phenotype.</a></li>
<li><a href='#completeMatrix'><p>This function performs an iterative matrix completion algorithm to predict drug response for pre-clinical data when there are missing ('NA') values.</p></a></li>
<li><a href='#doVariableSelection'><p>This function performs variable selection on gene expression matrices.</p>
It can, for instance, remove genes with low variation.</a></li>
<li><a href='#glds'><p>This function determines drug-gene associations for pre-clinical data.</p></a></li>
<li><a href='#homogenizeData'><p>This function takes two gene expression matrices (like trainExprMat and testExprMat) and returns homogenized versions of the matrices by employing the homogenization method specified.</p>
By default, the Combat method from the sva library is used.
In both matrices, genes are row names and samples are column names.
It will deal with duplicated gene names, as it subsets and orders the matrices correctly.</a></li>
<li><a href='#idwas'><p>This function will test every drug against every CNV or somatic mutation for your cancer type.</p></a></li>
<li><a href='#map_cnv'><p>This function maps cnv data to genes. The output of this function is a .RData file called map.RData; this file contains theCnvQuantVecList_mat (rows are genes, and columns are samples) and tumorSamps (indicates which samples are primary tumor samples, 01A).</p></a></li>
<li><a href='#summarizeGenesByMean'><p>This function takes a gene expression matrix and if duplicate genes are measured, summarizes them by their means.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Drug and Biomarker Discovery</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Bridges in vitro drug screening with in vivo drug and biomarker discovery. Specifically, predicts in vivo or cancer patient drug response and biomarkers to enrich for response from cell line screening data. Builds model using ridge regression, and enables biomarker discovery by imputing drug response in large cancer molecular datasets. It also enables drug specific biomarker identification by correcting for general level of drug sensitivity shared among the population. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>biocViews:</td>
<td>sva, preprocessCore, stringr, biomaRt, genefilter,
GenomicFeatures, genefilter, BiocGenerics, GenomicRanges,
IRanges, S4Vectors, org.Hs.eg.db,
TxDb.Hsapiens.UCSC.hg19.knownGene, maftools</td>
</tr>
<tr>
<td>Imports:</td>
<td>parallel, ridge, car, glmnet, pls, sva, preprocessCore,
GenomicFeatures, genefilter, gdata, tidyverse, readxl,
BiocGenerics, GenomicRanges, IRanges, S4Vectors, org.Hs.eg.db,
TxDb.Hsapiens.UCSC.hg19.knownGene, maftools</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-24 02:27:46 UTC; Dani</td>
</tr>
<tr>
<td>Author:</td>
<td>Danielle Maeser <a href="https://orcid.org/0000-0002-3890-887X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Robert Gruener [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Danielle Maeser &lt;maese005@umn.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-24 08:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='calcPhenotype'>This function predicts a phenotype (drug sensitivity score) when provided with microarray or bulk RNAseq gene expression data of different platforms.
The imputations are performed using ridge regression, training on a gene expression matrix where phenotype is already known.
This function integrates training and testing datasets via a user-defined procedure, and power transforming the known phenotype.</h2><span id='topic+calcPhenotype'></span>

<h3>Description</h3>

<p>This function predicts a phenotype (drug sensitivity score) when provided with microarray or bulk RNAseq gene expression data of different platforms.
The imputations are performed using ridge regression, training on a gene expression matrix where phenotype is already known.
This function integrates training and testing datasets via a user-defined procedure, and power transforming the known phenotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcPhenotype(
  trainingExprData,
  trainingPtype,
  testExprData,
  batchCorrect,
  powerTransformPhenotype = TRUE,
  removeLowVaryingGenes = 0.2,
  minNumSamples,
  selection = 1,
  printOutput,
  pcr = FALSE,
  removeLowVaringGenesFrom,
  report_pc = FALSE,
  cc = FALSE,
  percent = 80,
  rsq = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcPhenotype_+3A_trainingexprdata">trainingExprData</code></td>
<td>
<p>The training data. A matrix of expression levels. rownames() are genes, colnames() are samples (cell line names or cosmic ides, etc.). rownames() must be specified and must contain the same type of gene ids as &quot;testExprData&quot;</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_trainingptype">trainingPtype</code></td>
<td>
<p>The known phenotype for &quot;trainingExprData&quot;. This data must be a matrix of cell lines/rows or cosmic ids/rows x drugs/columns. This matrix can contain NA values, that is ok (they are removed in the calcPhenotype() function).</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_testexprdata">testExprData</code></td>
<td>
<p>The test data where the phenotype will be estimated. It is a matrix of expression levels, rows contain genes and columns contain samples, &quot;rownames()&quot; must be specified and must contain the same type of gene ids as &quot;trainingExprData&quot;.</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_batchcorrect">batchCorrect</code></td>
<td>
<p>How should training and test data matrices be homogenized. Choices are &quot;eb&quot; (default) for ComBat, &quot;qn&quot; for quantiles normalization or &quot;none&quot; for no homogenization.</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_powertransformphenotype">powerTransformPhenotype</code></td>
<td>
<p>Should the phenotype be power transformed before we fit the regression model? Default to TRUE, set to FALSE if the phenotype is already known to be highly normal.</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_removelowvaryinggenes">removeLowVaryingGenes</code></td>
<td>
<p>What proportion of low varying genes should be removed? 20 percent be default</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_minnumsamples">minNumSamples</code></td>
<td>
<p>How many training and test samples are required. Print an error if below this threshold</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_selection">selection</code></td>
<td>
<p>How should duplicate gene ids be handled. Default is -1 which asks the user. 1 to summarize by their or 2 to disguard all duplicates.</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_printoutput">printOutput</code></td>
<td>
<p>Set to FALSE to supress output.</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_pcr">pcr</code></td>
<td>
<p>Indicates whether or not you'd like to use pcr for feature (gene) reduction. Options are 'TRUE' and 'FALSE'. If you indicate 'report_pc=TRUE' you need to also indicate 'pcr=TRUE'</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_removelowvaringgenesfrom">removeLowVaringGenesFrom</code></td>
<td>
<p>Determine method to remove low varying genes. Options are 'homogenizeData' and 'rawData'.</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_report_pc">report_pc</code></td>
<td>
<p>Indicates whether you want to output the training principal components. Options are 'TRUE' and 'FALSE'.</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_cc">cc</code></td>
<td>
<p>Indicate if you want correlation coefficients for biomarker discovery.</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_percent">percent</code></td>
<td>
<p>Indicate percent variability (of the training data) you'd like principal components to reflect if pcr=TRUE. Default is 80 for 80%
These are the correlations between a given gene of interest across all samples vs. a given drug response across samples.
These correlations can be ranked to obtain a ranked correlation to determine highly correlated drug-gene associations.</p>
</td></tr>
<tr><td><code id="calcPhenotype_+3A_rsq">rsq</code></td>
<td>
<p>Indicate whether or not you want to output the R^2 values for the data you train on from true and predicted values.
These values represent the percentage in which the optimal model accounts for the variance in the training data.
Options are 'TRUE' and 'FALSE'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>.txt files will be saved into your working directory. Depending on the parameter specified, the .txt file outputs of this function can include the estimated phenotype/sensitivity predictions,
the R^2 data, and the correlation coefficients. Principal components are stored as .RData files for each drug in your drug dataset.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
try(calcPhenotype(trainingExprData=trainingExprData,
trainingPtype=trainingPtype,
testExprData=testExprData,
batchCorrect=batchCorrect,
powerTransformPhenotype=powerTransformPhenotype,
removeLowVaryingGenes=removeLowVaryingGenes,
minNumSamples=minNumSamples,
selection=selection,
printOutput=printOutput,
pcr=pcr,
removeLowVaringGenesFrom=removeLowVaringGenesFrom,
report_pc=report_pc,
cc=cc,
percent=percent,
rsq=rsq))

</code></pre>

<hr>
<h2 id='completeMatrix'>This function performs an iterative matrix completion algorithm to predict drug response for pre-clinical data when there are missing ('NA') values.</h2><span id='topic+completeMatrix'></span>

<h3>Description</h3>

<p>This function performs an iterative matrix completion algorithm to predict drug response for pre-clinical data when there are missing ('NA') values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>completeMatrix(senMat, nPerms = 50)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="completeMatrix_+3A_senmat">senMat</code></td>
<td>
<p>A matrix of drug sensitivity data with missing ('NA') values. rownames() are samples (e.g. cell lines), and colnames() are drugs.</p>
</td></tr>
<tr><td><code id="completeMatrix_+3A_nperms">nPerms</code></td>
<td>
<p>The number of iterations that the EM-algorithm (expectation maximization approach)  run. The default is 50, as previous findings recommend 50 iterations (https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1050-9)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of drug sensitivity scores without missing values. rownames() are samples, and colnames are drugs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
try(completeMatrix(senMat, nPerms = 50))

</code></pre>

<hr>
<h2 id='doVariableSelection'>This function performs variable selection on gene expression matrices.
It can, for instance, remove genes with low variation.</h2><span id='topic+doVariableSelection'></span>

<h3>Description</h3>

<p>This function performs variable selection on gene expression matrices.
It can, for instance, remove genes with low variation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>doVariableSelection(exprMat, removeLowVaryingGenes = 0.2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="doVariableSelection_+3A_exprmat">exprMat</code></td>
<td>
<p>A matrix of gene expression levels. rownames() are genes, and colnames() are samples.</p>
</td></tr>
<tr><td><code id="doVariableSelection_+3A_removelowvaryinggenes">removeLowVaryingGenes</code></td>
<td>
<p>The proportion of low varying genes to be removed.The default is .2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of row/genes to keep.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
try(doVariableSelection(exprMat, removeLowVaryingGenes = 0.2))

</code></pre>

<hr>
<h2 id='glds'>This function determines drug-gene associations for pre-clinical data.</h2><span id='topic+glds'></span>

<h3>Description</h3>

<p>This function determines drug-gene associations for pre-clinical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glds(
  drugMat,
  drugRelatedness,
  markerMat,
  minMuts = 5,
  additionalCovariateMatrix = NULL,
  expression = NULL,
  threshold = 0.7
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glds_+3A_drugmat">drugMat</code></td>
<td>
<p>A matrix of drug sensitivity data. rownames() are pre-clinical samples, and colnames() are drug names.</p>
</td></tr>
<tr><td><code id="glds_+3A_drugrelatedness">drugRelatedness</code></td>
<td>
<p>A matrix in which column 1 contains a list of compounds, and column 2 contains a list of their corresponding target pathways. Given the subjective nature of
drug classification, please ensure these pathways are as specific as possible for accurate results.</p>
</td></tr>
<tr><td><code id="glds_+3A_markermat">markerMat</code></td>
<td>
<p>A matrix containing the data for which you are looking for an association with drug sensitivity (e.g. a matrix of somatic mutation data). rownames() are marker names (e.g. gene names), and colnames() are samples.</p>
</td></tr>
<tr><td><code id="glds_+3A_minmuts">minMuts</code></td>
<td>
<p>The minimum number of non-zero entries required so that a p-value can be calculated (e.g. how many somatic mutations must be present). The default is 5.</p>
</td></tr>
<tr><td><code id="glds_+3A_additionalcovariatematrix">additionalCovariateMatrix</code></td>
<td>
<p>A matrix containing covariates to be fit in the drug biomarker association models. This could be, for example, tissue of origin or cancer type. Columns are sample names. The default is NULL.</p>
</td></tr>
<tr><td><code id="glds_+3A_expression">expression</code></td>
<td>
<p>A matrix of expression data. rownames() are genes, and colnames() are the same pre-clinical samples as those in the drugMat (also in the same order).
The default is NULL. If expression data is provided, a gene signature will be obtained.</p>
</td></tr>
<tr><td><code id="glds_+3A_threshold">threshold</code></td>
<td>
<p>Determine the correlation coefficient. Drugs with a correlation coefficient greater than or equal to this number with the drug under scrutiny will be removed from the negative control group.
The default is 0.7</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Naive and corrected p-values and beta-values, as well as a gene signature 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>try(glds(
drugMat,
drugRelatedness,
markerMat,
minMuts = 5,
additionalCovariateMatrix = NULL,
expression = NULL,
threshold = 0.7))

</code></pre>

<hr>
<h2 id='homogenizeData'>This function takes two gene expression matrices (like trainExprMat and testExprMat) and returns homogenized versions of the matrices by employing the homogenization method specified.
By default, the Combat method from the sva library is used.
In both matrices, genes are row names and samples are column names.
It will deal with duplicated gene names, as it subsets and orders the matrices correctly.</h2><span id='topic+homogenizeData'></span>

<h3>Description</h3>

<p>This function takes two gene expression matrices (like trainExprMat and testExprMat) and returns homogenized versions of the matrices by employing the homogenization method specified.
By default, the Combat method from the sva library is used.
In both matrices, genes are row names and samples are column names.
It will deal with duplicated gene names, as it subsets and orders the matrices correctly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>homogenizeData(
  testExprMat,
  trainExprMat,
  batchCorrect = "eb",
  selection = -1,
  printOutput = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="homogenizeData_+3A_testexprmat">testExprMat</code></td>
<td>
<p>A gene expression matrix for samples on which we wish to predict a phenotype.Genes are rows, samples are columns.</p>
</td></tr>
<tr><td><code id="homogenizeData_+3A_trainexprmat">trainExprMat</code></td>
<td>
<p>A gene expression matrix for samples for which the phenotype is already known.Genes are rows, samples are columns.</p>
</td></tr>
<tr><td><code id="homogenizeData_+3A_batchcorrect">batchCorrect</code></td>
<td>
<p>The type of batch correction to be used. Options are 'eb' for Combat, 'none', or 'qn' for quantile normalization.
#The default is 'eb'.</p>
</td></tr>
<tr><td><code id="homogenizeData_+3A_selection">selection</code></td>
<td>
<p>This parameter can be used to specify how duplicates are handled. The default value of -1 means to ask the user.
#Other options include '1' to summarize duplicates by their mean, and '2'to discard all duplicated genes.</p>
</td></tr>
<tr><td><code id="homogenizeData_+3A_printoutput">printOutput</code></td>
<td>
<p>To suppress output, set to false. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two entries $train and $test, which are the homogenized input matrices.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
try(homogenizeData(testExprMat,trainExprMat,batchCorrect = "eb",selection = -1,printOutput = TRUE))

</code></pre>

<hr>
<h2 id='idwas'>This function will test every drug against every CNV or somatic mutation for your cancer type.</h2><span id='topic+idwas'></span>

<h3>Description</h3>

<p>This function will test every drug against every CNV or somatic mutation for your cancer type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>idwas(drug_prediction, data, n = 10, cnv)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="idwas_+3A_drug_prediction">drug_prediction</code></td>
<td>
<p>The drug prediction data. Must be a data frame. rownames are samples, colnames are drugs. Make sure sample names are of the same form as the sample names in your cnv or mutation data. e.g. if the rownames() are TCGA barcodes of the form TCGA-##-####-###, make sure your cnv/mutation data also uses samples in the form TCGA-##-####-###</p>
</td></tr>
<tr><td><code id="idwas_+3A_data">data</code></td>
<td>
<p>The cnv or mutation data. Must be a data frame. If you wish to use cnv data, use the output from map_cnv(), transpose it so that colnames() are samples. Or use data of similar form. If you wish to use mutation data, use the method for downloading mutation data outlined in the vignette, and make sure the TCGA barcodes use '-' instead of '.'; if you use another dataset (and don't download data from TCGA), make sure your data file includes the following columns: 'Variant_Classification', 'Hugo_Symbol', 'Tumor_Sample_Barcode'.</p>
</td></tr>
<tr><td><code id="idwas_+3A_n">n</code></td>
<td>
<p>The minimum number of samples you want CNVs or mutations to be amplified in. The default is 10 (arbitrarily chosen).</p>
</td></tr>
<tr><td><code id="idwas_+3A_cnv">cnv</code></td>
<td>
<p>TRUE or FALSE. Indicate whether or not you would like to test cnv data. If TRUE, you will test cnv data. If FALSE, you will test mutation data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Raw p-value and beta-values for cnv and somatic mutations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
try(idwas(drug_prediction, data, n = 10, cnv))

</code></pre>

<hr>
<h2 id='map_cnv'>This function maps cnv data to genes. The output of this function is a .RData file called map.RData; this file contains theCnvQuantVecList_mat (rows are genes, and columns are samples) and tumorSamps (indicates which samples are primary tumor samples, 01A).</h2><span id='topic+map_cnv'></span>

<h3>Description</h3>

<p>This function maps cnv data to genes. The output of this function is a .RData file called map.RData; this file contains theCnvQuantVecList_mat (rows are genes, and columns are samples) and tumorSamps (indicates which samples are primary tumor samples, 01A).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map_cnv(Cnvs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="map_cnv_+3A_cnvs">Cnvs</code></td>
<td>
<p>The cnv data. A table with the following colnames: Sample (named using the TCGA patient barcode), Chromosome, Start, End, Num_Probes, and Segment_Mean.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A .RData file called, map.RData, which stores two objects: theCnvQuantVecList_mat (rows are genes, columns are samples), tumorSamps (indicates which samples are primary tumor/01A). This output will serve as the input for test().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
try(map_cnv(Cnvs))

</code></pre>

<hr>
<h2 id='summarizeGenesByMean'>This function takes a gene expression matrix and if duplicate genes are measured, summarizes them by their means.</h2><span id='topic+summarizeGenesByMean'></span>

<h3>Description</h3>

<p>This function takes a gene expression matrix and if duplicate genes are measured, summarizes them by their means.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarizeGenesByMean(exprMat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summarizeGenesByMean_+3A_exprmat">exprMat</code></td>
<td>
<p>A gene expression matrix with genes as rownames() and samples as colnames().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A gene expression matrix that does not contain duplicate genes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
try(summarizeGenesByMean(exprMat))

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
