<!DOCTYPE html><html lang="en"><head><title>Help for package OncoSubtype</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {OncoSubtype}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#centroids_subtype'><p>Predict the subtypes of selected cancer type based published papers</p></a></li>
<li><a href='#example_fpkm'><p>example FPKM data</p></a></li>
<li><a href='#get_hvg'><p>select highly variable genes from a expression matrix</p></a></li>
<li><a href='#get_median_centered'><p>convert expression matrix to median-centered</p></a></li>
<li><a href='#get_rf_pred'><p>Predict the subtypes of selected cancer type</p></a></li>
<li><a href='#hnsc_centroids'><p>HNSC predictor centroids</p></a></li>
<li><a href='#load_dataset_from_github'><p>Load Dataset from GitHub Repository</p></a></li>
<li><a href='#luad_centroids'><p>LUAD predictor centroids</p></a></li>
<li><a href='#lusc_centroids'><p>LUSC predictor centroids</p></a></li>
<li><a href='#ml_subtype'><p>Predict the subtypes of selected cancer type using machine learning</p></a></li>
<li><a href='#PlotHeat'><p>Plot heatmap of the train set or test set</p></a></li>
<li><a href='#SubtypeClass-class'><p>Set the SubtypeClass</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Predict Cancer Subtypes Based on TCGA Data using Machine
Learning Method</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Dadong Zhang &lt;dadong.zhang.shared@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dadong Zhang &lt;dadong.zhang.shared@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provide functionality for cancer subtyping using nearest centroids or machine learning methods based on TCGA data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/DadongZ/OncoSubtype">https://github.com/DadongZ/OncoSubtype</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/DadongZ/OncoSubtype/issues">https://github.com/DadongZ/OncoSubtype/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>Imports:</td>
<td>caret, randomForest, methods, e1071, pheatmap, tibble, dplyr,
limma, rlang, Rdpack</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>Depends:</td>
<td>SummarizedExperiment, R (&ge; 3.63),</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-22 16:31:11 UTC; dzhang4</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-22 17:30:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='centroids_subtype'>Predict the subtypes of selected cancer type based published papers</h2><span id='topic+centroids_subtype'></span>

<h3>Description</h3>

<p>Predict the subtypes of selected cancer type based published papers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>centroids_subtype(data, disease = "LUSC")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="centroids_subtype_+3A_data">data</code></td>
<td>
<p>data set to predict the subtypes which is a numeric matrix with row names of features and column names of samples</p>
</td></tr>
<tr><td><code id="centroids_subtype_+3A_disease">disease</code></td>
<td>
<p>character string of the disease to predict subtypes, currently support 'LUSC', 'LUAD'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class &quot;SubtypeClass&quot; with four slots: genes used for predictiong, predicted subtypes of samples, a matrix of predicting scores, and the method.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(OncoSubtype)
data &lt;- get_median_centered(example_fpkm)
data &lt;- assays(data)$centered
rownames(data) &lt;- rowData(example_fpkm)$external_gene_name
centroids_subtype(data, disease = 'HNSC')

## End(Not run)
</code></pre>

<hr>
<h2 id='example_fpkm'>example FPKM data</h2><span id='topic+example_fpkm'></span>

<h3>Description</h3>

<p>example FPKM data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_fpkm
</code></pre>


<h3>Format</h3>

<p>SummarizedExperiment object
</p>

<hr>
<h2 id='get_hvg'>select highly variable genes from a expression matrix</h2><span id='topic+get_hvg'></span>

<h3>Description</h3>

<p>select highly variable genes from a expression matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_hvg(data, top = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_hvg_+3A_data">data</code></td>
<td>
<p>a (normalized) matrix with rownames of features and colnames of samples</p>
</td></tr>
<tr><td><code id="get_hvg_+3A_top">top</code></td>
<td>
<p>number of top highly variable genes to output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>subset with top ranked genes by the variances
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(OncoSubtype)
data &lt;- get_median_centered(example_fpkm)
data &lt;- assays(data)$centered
get_hvg(data)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_median_centered'>convert expression matrix to median-centered</h2><span id='topic+get_median_centered'></span>

<h3>Description</h3>

<p>convert expression matrix to median-centered
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_median_centered(data, log2 = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_median_centered_+3A_data">data</code></td>
<td>
<p>a numeric matrix or 'S4' object</p>
</td></tr>
<tr><td><code id="get_median_centered_+3A_log2">log2</code></td>
<td>
<p>logical, if 'TRUE' <code class="reqn">log2(x+1)</code> will be applied before median centering. Defaut 'TRUE'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>median-centered express matrix or an object with new slot &quot;centered&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
get_median_centered(example_fpkm)

## End(Not run)

</code></pre>

<hr>
<h2 id='get_rf_pred'>Predict the subtypes of selected cancer type</h2><span id='topic+get_rf_pred'></span>

<h3>Description</h3>

<p>Predict the subtypes of selected cancer type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_rf_pred(train_set, test_set, method = "rf", seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_rf_pred_+3A_train_set">train_set</code></td>
<td>
<p>training set with rownames of samples, first column named 'mRNA_subtype' and the rest of features and expression values.</p>
</td></tr>
<tr><td><code id="get_rf_pred_+3A_test_set">test_set</code></td>
<td>
<p>test set with rownames of features and colnames of samples.</p>
</td></tr>
<tr><td><code id="get_rf_pred_+3A_method">method</code></td>
<td>
<p>character string of the method to use currently support 'rf'.</p>
</td></tr>
<tr><td><code id="get_rf_pred_+3A_seed">seed</code></td>
<td>
<p>integer seed to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix with column names of subtypes and predicted probabilities.
</p>

<hr>
<h2 id='hnsc_centroids'>HNSC predictor centroids</h2><span id='topic+hnsc_centroids'></span>

<h3>Description</h3>

<p>HNSC predictor centroids from <a href="https://www.nature.com/articles/nature14129">https://www.nature.com/articles/nature14129</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hnsc_centroids
</code></pre>


<h3>Format</h3>

<p>A <code>tibble</code> with 728 features and four subtypes.
</p>

<hr>
<h2 id='load_dataset_from_github'>Load Dataset from GitHub Repository</h2><span id='topic+load_dataset_from_github'></span>

<h3>Description</h3>

<p>Downloads a specified dataset from a GitHub repository if it is not already
present in the specified local directory, then loads the dataset into the global
environment. This function is designed to help manage package size by storing
data externally and loading it on-demand.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_dataset_from_github(disease, local_dir = path.expand(getwd()))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="load_dataset_from_github_+3A_disease">disease</code></td>
<td>
<p>A character string specifying the disease, which corresponds
to the name of the dataset to be loaded (e.g., &quot;LUSC&quot;). The function constructs
the filename as <code>tolower(disease)_tcga.rda</code> and attempts to load this dataset.</p>
</td></tr>
<tr><td><code id="load_dataset_from_github_+3A_local_dir">local_dir</code></td>
<td>
<p>An optional character string specifying the path to the directory
where datasets should be stored locally. If not provided, defaults to a
subdirectory named <code>your_package_name_data</code> within the user's home directory.
Users can specify their own directory path if they prefer to store data in a different
location.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisible NULL. The function is primarily used for its side effect of
loading a dataset into the global environment. However, the function itself
does not return the dataset directly.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  load_dataset_from_github("LUSC")

## End(Not run)

</code></pre>

<hr>
<h2 id='luad_centroids'>LUAD predictor centroids</h2><span id='topic+luad_centroids'></span>

<h3>Description</h3>

<p>LUAD predictor centroids from Wilkerson (2012)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>luad_centroids
</code></pre>


<h3>Format</h3>

<p>A <code>tibble</code> with 506 features and three subtypes bronchioid, magnoid, and squamoid.
</p>

<hr>
<h2 id='lusc_centroids'>LUSC predictor centroids</h2><span id='topic+lusc_centroids'></span>

<h3>Description</h3>

<p>LUSC predictor centroids from Wilkerson (2010)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lusc_centroids
</code></pre>


<h3>Format</h3>

<p>A <code>tibble</code> with 208 features and four subtypes: primitive, classical, secretory, and basal.
</p>

<hr>
<h2 id='ml_subtype'>Predict the subtypes of selected cancer type using machine learning</h2><span id='topic+ml_subtype'></span>

<h3>Description</h3>

<p>Predict the subtypes of selected cancer type using machine learning
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ml_subtype(
  data,
  disease = "LUSC",
  method = "rf",
  removeBatch = TRUE,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ml_subtype_+3A_data">data</code></td>
<td>
<p>data set to predict the subtypes which is a numeric matrix with row names of features and column names of samples</p>
</td></tr>
<tr><td><code id="ml_subtype_+3A_disease">disease</code></td>
<td>
<p>character string of the disease to predict subtypes, currently support 'LUSC', 'LUAD', and 'BLCA'.</p>
</td></tr>
<tr><td><code id="ml_subtype_+3A_method">method</code></td>
<td>
<p>character string of the method to use currently support 'rf'.</p>
</td></tr>
<tr><td><code id="ml_subtype_+3A_removebatch">removeBatch</code></td>
<td>
<p>whether do batch effect correction using <code>limma::removeBatchEffect</code>, default TRUE.</p>
</td></tr>
<tr><td><code id="ml_subtype_+3A_seed">seed</code></td>
<td>
<p>integer seed to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class &quot;SubtypeClass&quot; with four slots: genes used for predictiong, predicted subtypes of samples, a matrix of predicting scores, and the method.
</p>


<h3>References</h3>


<ol>
<li> <p>Wilkerson MJ, Yin X, Hayes D, et al. (2010).
&ldquo;Lung Squamous Cell Carcinoma mRNA Expression Subtypes Are Reproducible, Clinically Important, and Correspond to Normal Cell Types.&rdquo;
<em>Clin Cancer Res</em>, <b>16</b>(19), 4864-4875.
</p>
</li>
<li> <p>Wilkerson MJ, Yin X, Hayes D, et al. (2012).
&ldquo;Differential pathogenesis of lung adenocarcinoma subtypes involving sequence mutations, copy number, chromosomal instability, and methylation.&rdquo;
<em>Plos One</em>, <b>7</b>(5), e36530.
</p>
</li>
<li> <p>Network TCGA (2015).
&ldquo;Comprehensive genomic characterization of head and neck squamous cell carcinomas.&rdquo;
<em>Nature</em>, <b>517</b>, e36530.
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(OncoSubtype)
data &lt;- get_median_centered(example_fpkm)
data &lt;- assays(data)$centered
rownames(data) &lt;- rowData(example_fpkm)$external_gene_name
ml_subtype(data, disease = 'LUAD', method = 'rf', seed = 123)

## End(Not run)
</code></pre>

<hr>
<h2 id='PlotHeat'>Plot heatmap of the train set or test set</h2><span id='topic+PlotHeat'></span>

<h3>Description</h3>

<p>Plot heatmap of the train set or test set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotHeat(object, set = "test", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotHeat_+3A_object">object</code></td>
<td>
<p>a SubtypeClass object</p>
</td></tr>
<tr><td><code id="PlotHeat_+3A_set">set</code></td>
<td>
<p>options could be 'test', 'train' or 'both'. Default 'test'.</p>
</td></tr>
<tr><td><code id="PlotHeat_+3A_...">...</code></td>
<td>
<p>Parameters passed to <code>pheatmap</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a pheatmap object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(OncoSubtype)
data &lt;- get_median_centered(example_fpkm)
data &lt;- assays(data)$centered
rownames(data) &lt;- rowData(example_fpkm)$external_gene_name
object &lt;- MLSubtype(data, disease = 'LUSC')
PlotHeat(object, set = 'both', fontsize = 10, show_rownames = FALSE, show_colnames = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='SubtypeClass-class'>Set the SubtypeClass</h2><span id='topic+SubtypeClass-class'></span><span id='topic+SubtypeClass'></span>

<h3>Description</h3>

<p>Set the SubtypeClass
</p>


<h3>Value</h3>

<p>an object of SubtypeClass with three empty solts
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
