<!DOCTYPE html><html><head><title>Help for package ondisc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ondisc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#create_ondisc_matrix_from_mtx'><p>Create an <code>ondisc_matrix</code> from a .mtx file.</p></a></li>
<li><a href='#dim'><p>Get dimension</p></a></li>
<li><a href='#extract-odm'><p>Pull a submatrix into memory using the <code>[[</code> operator.</p></a></li>
<li><a href='#get-names'><p>Get cell barcodes, feature names, and feature IDs</p></a></li>
<li><a href='#head,ondisc_matrix-method'><p>head</p></a></li>
<li><a href='#metadata_ondisc_matrix'><p><code>metadata_ondisc_matrix</code> class</p></a></li>
<li><a href='#multimodal_ondisc_matrix'><p><code>multimodal_ondisc_matrix</code> class</p></a></li>
<li><a href='#ondisc'><p>ondisc: A package for out-of-memory computing on single-cell data</p></a></li>
<li><a href='#ondisc_matrix'><p><code>ondisc_matrix</code> class</p></a></li>
<li><a href='#show'><p>Print basic information to the console</p></a></li>
<li><a href='#subset-odm'><p>Subset using the <code>[</code> operator.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Fast, Universal, and Intuitive Computing on Large-Scale
Single-Cell Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Single-cell datasets are growing in size, posing challenges as well as opportunities for biology researchers. 'ondisc' (short for "on-disk single cell") enables users to easily and efficiently analyze large-scale single-cell data. 'ondisc' makes computing on large-scale single-cell data FUN: Fast, Universal, and iNtuitive.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://timothy-barry.github.io/ondisc/">https://timothy-barry.github.io/ondisc/</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, covr</td>
</tr>
<tr>
<td>Imports:</td>
<td>readr, methods, magrittr, rhdf5, data.table, Matrix, Rcpp,
crayon, dplyr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, Rhdf5lib</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-03-04 03:57:43 UTC; timbarry</td>
</tr>
<tr>
<td>Author:</td>
<td>Timothy Barry <a href="https://orcid.org/0000-0002-4356-627X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Eugene Katsevich <a href="https://orcid.org/0000-0003-0598-2050"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ths],
  Kathryn Roeder [ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Timothy Barry &lt;tbarry2@andrew.cmu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-03-05 09:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='create_ondisc_matrix_from_mtx'>Create an <code>ondisc_matrix</code> from a .mtx file.</h2><span id='topic+create_ondisc_matrix_from_mtx'></span>

<h3>Description</h3>

<p>Initializes an <code>ondisc_matrix</code> from a .mtx file, a features.tsv file, and a barcodes.tsv file. Returns an <code>ondisc_matrix</code> along with cell-specific and feature-specific covariate matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ondisc_matrix_from_mtx(
  mtx_fp,
  barcodes_fp,
  features_fp,
  n_lines_per_chunk = 3e+08,
  on_disk_dir = NULL,
  file_name = NULL,
  return_metadata_ondisc_matrix = FALSE,
  progress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_ondisc_matrix_from_mtx_+3A_mtx_fp">mtx_fp</code></td>
<td>
<p>file path to a .mtx file storing the expression data. The .mtx file can represent either an integer matrix or a logical (i.e., binary) matrix. If the .mtx file contains only two columns (after the initial three-column row of metadata), then the .mtx file is assumed to represent a logical matrix.</p>
</td></tr>
<tr><td><code id="create_ondisc_matrix_from_mtx_+3A_barcodes_fp">barcodes_fp</code></td>
<td>
<p>file path to the .tsv file containing the cell barcodes.</p>
</td></tr>
<tr><td><code id="create_ondisc_matrix_from_mtx_+3A_features_fp">features_fp</code></td>
<td>
<p>file path to the features.tsv file. The first column (required) contains the feature IDs (e.g., ENSG00000186092), and the second column (optional) contains the human-readable feature names (e.g., OR4F5). Subsequent columns are discarded.</p>
</td></tr>
<tr><td><code id="create_ondisc_matrix_from_mtx_+3A_n_lines_per_chunk">n_lines_per_chunk</code></td>
<td>
<p>(optional) number of lines in .mtx file to process per chunk. Defaults to 3e+08.</p>
</td></tr>
<tr><td><code id="create_ondisc_matrix_from_mtx_+3A_on_disk_dir">on_disk_dir</code></td>
<td>
<p>(optional) directory in which to store the on-disk portion of the ondisc_matrix. Defaults to the directory in which the .mtx file is located.</p>
</td></tr>
<tr><td><code id="create_ondisc_matrix_from_mtx_+3A_file_name">file_name</code></td>
<td>
<p>(optional) name of the file in which to store the .h5 data on-disk. Defaults to ondisc_matrix_x.h5, where x is a unique integer starting at 1.</p>
</td></tr>
<tr><td><code id="create_ondisc_matrix_from_mtx_+3A_return_metadata_ondisc_matrix">return_metadata_ondisc_matrix</code></td>
<td>
<p>(optional) return the output as a metadata_ondisc_matrix (instead of a list)? Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="create_ondisc_matrix_from_mtx_+3A_progress">progress</code></td>
<td>
<p>(optional; default FALSE) print progress messages?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function can compute the following cell-specific and feature-specific covariates:
</p>

<ul>
<li><p> cell-specific: (i) total number of features expressed in cell (n_nonzero_cell), (ii) total UMI count (n_umis_cell), and (iii) percentage of UMIs that map to mitochondrial genes (p_mito_cell).
</p>
</li>
<li><p> feature-specific: (i) total number of cells in which feature is expressed (n_nonzero_feature), (ii) mean expression of feature across cells (mean_expression_feature), (iii) coefficient of variation of feature expression across cells (coef_of_variation_feature).
</p>
</li></ul>

<p>The function decides which covariates to compute given the input; in general, the function computes the maximum set of covariates possible.
</p>


<h3>Value</h3>

<p>A list containing (i) an ondisc_matrix, (ii) a cell-specific covariate matrix, and (iii) a feature-specific covariate matrix; if the parameter return_metadata_ondisc_matrix set to TRUE, converts the list to a metadata_ondisc_matrix before returning.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# First example: initialize a metadata_ondisc_matrix
# using simulated expression data; store output in tempdir()
file_locs &lt;- system.file("extdata",package = "ondisc",
c("gene_expression.mtx", "genes.tsv", "cell_barcodes.tsv"))
names(file_locs) &lt;- c("expressions", "features", "barcodes")
expression_data &lt;- create_ondisc_matrix_from_mtx(mtx_fp = file_locs[["expressions"]],
barcodes_fp = file_locs[["barcodes"]],
features_fp = file_locs[["features"]],
on_disk_dir = tempdir(),
file_name = "expressions",
return_metadata_ondisc_matrix = TRUE)
saveRDS(object = expression_data, file = paste0(tempdir(), "/expressions.rds"))

# Second example: initialize a metadata_ondisc_matrix using simulated
# gRNA perturbation data; store in tempdir()
file_locs &lt;- system.file("extdata", package = "ondisc",
c("perturbation.mtx", "guides.tsv", "cell_barcodes.tsv"))
names(file_locs) &lt;- c("perturbations", "features", "barcodes")
perturbation_data &lt;- create_ondisc_matrix_from_mtx(mtx_fp = file_locs[["perturbations"]],
barcodes_fp = file_locs[["barcodes"]],
features_fp = file_locs[["features"]],
on_disk_dir = tempdir(),
file_name = "perturbations",
return_metadata_ondisc_matrix = TRUE)
saveRDS(object = perturbation_data, file = paste0(tempdir(), "/perturbations.rds"))

## End(Not run)
</code></pre>

<hr>
<h2 id='dim'>Get dimension</h2><span id='topic+dim'></span><span id='topic+dim+2Condisc_matrix-method'></span><span id='topic+dim+2Cmetadata_ondisc_matrix-method'></span><span id='topic+dim+2Cmultimodal_ondisc_matrix-method'></span><span id='topic+ncol'></span><span id='topic+nrow'></span><span id='topic+ncol+2Cmultimodal_ondisc_matrix-method'></span><span id='topic+nrow+2Cmultimodal_ondisc_matrix-method'></span>

<h3>Description</h3>

<p>Return the dimension of an <code>ondisc_matrix</code>, <code>metadata_ondisc_matrix</code>, or <code>multimodal_ondisc_matrix</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ondisc_matrix'
dim(x)

## S4 method for signature 'metadata_ondisc_matrix'
dim(x)

## S4 method for signature 'multimodal_ondisc_matrix'
dim(x)

ncol(x)

nrow(x)

## S4 method for signature 'multimodal_ondisc_matrix'
ncol(x)

## S4 method for signature 'multimodal_ondisc_matrix'
nrow(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dim_+3A_x">x</code></td>
<td>
<p>an <code>ondisc_matrix</code>, <code>metadata_ondisc_matrix</code>, or <code>multimodal_ondisc_matrix</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>x</code> is an <code>ondisc_matrix</code> or <code>metadata_ondisc_matrix</code>, length-two integer vector containing
the dimension of <code>x</code>; if <code>x</code> is a <code>multimodal_ondisc_matrix</code>, a list of integer vectors containing the dimensions
of the constituent modalities of <code>x</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># NOTE: You must create the RDS files "expressions.rds" and
# "perturbations.rds" to run this example. Navigate to the help file of
# "create_ondisc_matrix_from_mtx" (via ?create_ondisc_matrix_from_mtx),
# and execute both code blocks.

# dimension of an ondisc_matrix
h5_fp &lt;- paste0(tempdir(), "/expressions.h5")
if (file.exists(h5_fp)) {
odm &lt;- ondisc_matrix(h5_file = h5_fp)
dim(odm)
}

# dimension of a metadata_ondic_matrix
expressions_fp &lt;- paste0(tempdir(), "/expressions.rds")
if (file.exists(expressions_fp)) {
expressions &lt;- readRDS(expressions_fp)
dim(expressions)
}

# dimension of a multimodal_ondisc_matrix
expression_fp &lt;- paste0(tempdir(), "/expressions.rds")
perturbations_fp &lt;- paste0(tempdir(), "/perturbations.rds")
if (file.exists(expression_fp) &amp;&amp; file.exists(perturbations_fp)) {
    crispr_experiment &lt;- multimodal_ondisc_matrix(list(expressions = readRDS(expression_fp),
    perturbations = readRDS(perturbations_fp)))
    dim(crispr_experiment)
}
</code></pre>

<hr>
<h2 id='extract-odm'>Pull a submatrix into memory using the <code>[[</code> operator.</h2><span id='topic+extract-odm'></span><span id='topic++5B+5B+2Condisc_matrix+2Cmissing+2Cmissing-method'></span><span id='topic++5B+5B+2Condisc_matrix+2CANY+2Cmissing-method'></span><span id='topic++5B+5B+2Condisc_matrix+2Cmissing+2CANY-method'></span><span id='topic++5B+5B+2Condisc_matrix+2CANY+2CANY-method'></span><span id='topic++5B+5B+2Cmetadata_ondisc_matrix+2CANY+2CANY-method'></span><span id='topic++5B+5B+2Cmultimodal_ondisc_matrix+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>Apply the <code>[[</code> operator to an <code>ondisc_matrix</code> to pull a submatrix into memory. You can pass logical, character,
or numeric vectors to <code>[[</code>; character vectors are assumed to refer to feature IDs (for rows) and cell barcodes
(for columns).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ondisc_matrix,missing,missing'
x[[i, j]]

## S4 method for signature 'ondisc_matrix,ANY,missing'
x[[i, j]]

## S4 method for signature 'ondisc_matrix,missing,ANY'
x[[i, j]]

## S4 method for signature 'ondisc_matrix,ANY,ANY'
x[[i, j]]

## S4 method for signature 'metadata_ondisc_matrix,ANY,ANY'
x[[i, j]]

## S4 method for signature 'multimodal_ondisc_matrix,ANY,ANY'
x[[i, j]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract-odm_+3A_x">x</code></td>
<td>
<p>an <code>ondisc_matrix</code> object.</p>
</td></tr>
<tr><td><code id="extract-odm_+3A_i">i</code></td>
<td>
<p>a vector (numeric, logical, or character) indicating features to pull into memory.</p>
</td></tr>
<tr><td><code id="extract-odm_+3A_j">j</code></td>
<td>
<p>a vector (numeric, logical, or character) indicating cells to pull into memory.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can apply <code>[[</code> to <code>ondisc_matrix</code> objects only. You cannot apply <code>[[</code> to <code>metadata_ondisc_matrix</code> or
<code>multimodal_ondisc_matrix</code> objects, because in the latter case the data to be accessed is ambiguous.
</p>
<p>You can remember the difference between <code>[</code> and <code>[[</code> by thinking about R lists: <code>[</code> is used to subset a list, and
<code>[[</code> is used to access elements stored <em>inside</em> a list. Similarly, <code>[</code> is used to subset an <code>ondisc_matrix</code>, and
<code>[[</code> is used to access a submatrix usable within R.
</p>


<h3>Value</h3>

<p>a matrix (as implemented by the Matrix package).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># NOTE: You must create the HDF5 file "expressions.h5" to run this example.
# Navigate to the help file of "create_ondisc_matrix_from_mtx"
# (via ?create_ondisc_matrix_from_mtx), and execute the code in the first code block.

h5_fp &lt;- paste0(tempdir(), "/expressions.h5")
if (file.exists(h5_fp)) {
odm &lt;- ondisc_matrix(h5_file = h5_fp)
# extract cells 100-110:
x &lt;- odm[[,100:110]]
# extract genes ENSG00000188305, ENSG00000257284, ENSG00000251655:
x &lt;- odm[[c("ENSG00000188305", "ENSG00000257284", "ENSG00000251655"),]]
# extract cells CTTAGGACACTGGCGT-1 and AAAGGATTCACATCAG-1:
x &lt;- odm[[,c("CTTAGGACACTGGCGT-1", "AAAGGATTCACATCAG-1")]]
}
</code></pre>

<hr>
<h2 id='get-names'>Get cell barcodes, feature names, and feature IDs</h2><span id='topic+get-names'></span><span id='topic+get_feature_ids'></span><span id='topic+get_feature_names'></span><span id='topic+get_cell_barcodes'></span><span id='topic+get_feature_ids+2Condisc_matrix-method'></span><span id='topic+get_feature_names+2Condisc_matrix-method'></span><span id='topic+get_cell_barcodes+2Condisc_matrix-method'></span><span id='topic+get_feature_ids+2Cmetadata_ondisc_matrix-method'></span><span id='topic+get_feature_names+2Cmetadata_ondisc_matrix-method'></span><span id='topic+get_cell_barcodes+2Cmetadata_ondisc_matrix-method'></span><span id='topic+get_feature_ids+2Cmultimodal_ondisc_matrix-method'></span><span id='topic+get_feature_names+2Cmultimodal_ondisc_matrix-method'></span><span id='topic+get_cell_barcodes+2Cmultimodal_ondisc_matrix-method'></span>

<h3>Description</h3>

<p>Obtain cell barcodes, feature names, and feature IDs of an <code>ondisc_matrix</code>, <code>metadata_ondisc_matrix</code>,
or <code>multimodal_ondisc_matrix</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_feature_ids(x)

get_feature_names(x)

get_cell_barcodes(x)

## S4 method for signature 'ondisc_matrix'
get_feature_ids(x)

## S4 method for signature 'ondisc_matrix'
get_feature_names(x)

## S4 method for signature 'ondisc_matrix'
get_cell_barcodes(x)

## S4 method for signature 'metadata_ondisc_matrix'
get_feature_ids(x)

## S4 method for signature 'metadata_ondisc_matrix'
get_feature_names(x)

## S4 method for signature 'metadata_ondisc_matrix'
get_cell_barcodes(x)

## S4 method for signature 'multimodal_ondisc_matrix'
get_feature_ids(x)

## S4 method for signature 'multimodal_ondisc_matrix'
get_feature_names(x)

## S4 method for signature 'multimodal_ondisc_matrix'
get_cell_barcodes(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get-names_+3A_x">x</code></td>
<td>
<p>an object of class <code>ondisc_matrix</code>, <code>covaraite_ondisc_matrix</code>, or <code>multimodal_ondisc_matrix</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following functions can be used to obtain feature and cell identifiers:
</p>

<ul>
<li> <p><code>get_cell_barcodes</code>: return the cell barcodes.
</p>
</li>
<li> <p><code>get_feature_names</code>: return the feature names.
</p>
</li>
<li> <p><code>get_feature_ids</code>: return the IDs of the features.
</p>
</li></ul>

<p>In general, these functions return a character vector containing the requested identifiers. When
<code>get_feature_names</code> or <code>get_feature_ids</code> is called on a <code>multimodal_ondisc_matrix</code>, the function instead
returns a list containing the feature names and feature IDs, respectively, of the modalities contained
within the <code>multimodal_ondisc_matrix</code>.
</p>


<h3>Value</h3>

<p>A character vector or list of character vectors containing the requested identifiers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># NOTE: You must create the RDS files "expressions.rds" and
# "perturbations.rds" to run this example. Navigate to the help file of
# "create_ondisc_matrix_from_mtx" (via ?create_ondisc_matrix_from_mtx),
# and execute both code blocks.

# ondisc_matrix
h5_fp &lt;- paste0(tempdir(), "/expressions.h5")
if (file.exists(h5_fp)) {
odm &lt;- ondisc_matrix(h5_file = h5_fp)
barcodes &lt;- get_cell_barcodes(odm)
feature_names &lt;- get_feature_names(odm)
feature_ids &lt;- get_feature_ids(odm)
}

# metadata_ondic_matrix
expressions_fp &lt;- paste0(tempdir(), "/expressions.rds")
if (file.exists(expressions_fp)) {
expressions &lt;- readRDS(expressions_fp)
barcodes &lt;- get_cell_barcodes(odm)
feature_names &lt;- get_feature_names(odm)
feature_ids &lt;- get_feature_ids(odm)
}

# multimodal_ondisc_matrix
expression_fp &lt;- paste0(tempdir(), "/expressions.rds")
perturbations_fp &lt;- paste0(tempdir(), "/perturbations.rds")
if (file.exists(expression_fp) &amp;&amp; file.exists(perturbations_fp)) {
    crispr_experiment &lt;- multimodal_ondisc_matrix(list(expressions = readRDS(expression_fp),
    perturbations = readRDS(perturbations_fp)))
    barcodes &lt;- get_cell_barcodes(crispr_experiment)
    feature_ids &lt;- get_feature_ids(crispr_experiment)
}
</code></pre>

<hr>
<h2 id='head+2Condisc_matrix-method'>head</h2><span id='topic+head+2Condisc_matrix-method'></span>

<h3>Description</h3>

<p>Print the first few rows and columns of an <code>ondisc_matrix</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ondisc_matrix'
head(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="head+2B2Condisc_matrix-method_+3A_x">x</code></td>
<td>
<p>an <code>ondisc_matrix</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL; called for printing
</p>


<h3>Examples</h3>

<pre><code class='language-R'># NOTE: You must create the HDF5 file "expressions.h5" to run this example.
# Navigate to the help file of "create_ondisc_matrix_from_mtx"
# (via ?create_ondisc_matrix_from_mtx), and execute the code in the first code block.

h5_fp &lt;- paste0(tempdir(), "/expressions.h5")
if (file.exists(h5_fp)) {
odm &lt;- ondisc_matrix(h5_file = h5_fp)
head(odm)
}
</code></pre>

<hr>
<h2 id='metadata_ondisc_matrix'><code>metadata_ondisc_matrix</code> class</h2><span id='topic+metadata_ondisc_matrix'></span>

<h3>Description</h3>

<p>A <code>metadata_ondisc_matrix</code> stores an <code>ondisc_matrix</code>, along with cell-specific and feature-specific covariate matrices.
</p>
<p>Construct a <code>metadata_ondisc_matrix</code> by passing an <code>ondisc_matrix</code>, along with its associated <code>cell_covariates</code> and <code>feature_covariates</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metadata_ondisc_matrix(ondisc_matrix, cell_covariates, feature_covariates)

metadata_ondisc_matrix(ondisc_matrix, cell_covariates, feature_covariates)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metadata_ondisc_matrix_+3A_ondisc_matrix">ondisc_matrix</code></td>
<td>
<p>an <code>ondisc_matrix</code>.</p>
</td></tr>
<tr><td><code id="metadata_ondisc_matrix_+3A_cell_covariates">cell_covariates</code></td>
<td>
<p>a data frame storing the cell-specific covariates.</p>
</td></tr>
<tr><td><code id="metadata_ondisc_matrix_+3A_feature_covariates">feature_covariates</code></td>
<td>
<p>a data frame storing the feature-specific covariates.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>metadata_ondisc_matrix</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>ondisc_matrix</code></dt><dd><p>an ondisc_matrix.</p>
</dd>
<dt><code>cell_covariates</code></dt><dd><p>a data frame of cell covariates.</p>
</dd>
<dt><code>feature_covariates</code></dt><dd><p>a data frame of feature covariates.</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'># NOTE: You must create the HDF5 file "expressions.h5" and the RDS file
# "expressions.rds" to run this example. Navigate to the help file of
# "create_ondisc_matrix_from_mtx" (via ?create_ondisc_matrix_from_mtx),
# and execute the code in the first code block.
covariates_fp &lt;- paste0(tempdir(), "/expressions.rds")
h5_fp &lt;-  paste0(tempdir(), "/expressions.h5")
if (file.exists(covariates_fp) &amp;&amp; file.exists(h5_fp)) {
covariate_odm &lt;- readRDS(covariates_fp)
cell_covariate_matrix &lt;- covariate_odm@cell_covariates
feature_covariate_matrix &lt;- covariate_odm@feature_covariates
covariate_odm_copy &lt;- metadata_ondisc_matrix(
ondisc_matrix = ondisc_matrix(h5_file = h5_fp),
cell_covariates = cell_covariate_matrix,
feature_covariates = feature_covariate_matrix)
}
</code></pre>

<hr>
<h2 id='multimodal_ondisc_matrix'><code>multimodal_ondisc_matrix</code> class</h2><span id='topic+multimodal_ondisc_matrix'></span>

<h3>Description</h3>

<p>A <code>multimodal_ondisc_matrix</code> represents multimodal data.
</p>
<p>Construct a <code>multimodal_ondisc_matrix</code> from a list of <code>metadata_ondisc_matrix</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multimodal_ondisc_matrix(metadata_ondisc_matrix_list)

multimodal_ondisc_matrix(metadata_ondisc_matrix_list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multimodal_ondisc_matrix_+3A_metadata_ondisc_matrix_list">metadata_ondisc_matrix_list</code></td>
<td>
<p>a named list containing <code>metadata_ondisc_matrices</code>; the names are taken to be the names of the modalities.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a multimodal_ondisc_matrix
</p>


<h3>Slots</h3>


<dl>
<dt><code>modalities</code></dt><dd><p>a list containing <code>metadata_ondisc_matrix</code> objects representing different modalities.</p>
</dd>
<dt><code>global_cell_covariates</code></dt><dd><p>a data frame containing the cell-specific covariates pooled across all modalities.</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'># NOTE: You must create the RDS files "expressions.rds" and
# "perturbations.rds" to run this example. Navigate to the help file of
# "create_ondisc_matrix_from_mtx" (via ?create_ondisc_matrix_from_mtx),
# and execute both code blocks.
expression_fp &lt;- paste0(tempdir(), "/expressions.rds")
perturbations_fp &lt;- paste0(tempdir(), "/perturbations.rds")
if (file.exists(expression_fp) &amp;&amp; file.exists(perturbations_fp)) {
    expressions &lt;- readRDS(expression_fp)
    perturbations &lt;- readRDS(perturbations_fp)
    crispr_experiment &lt;- multimodal_ondisc_matrix(list(expressions = expressions,
    perturbations = perturbations))
}
</code></pre>

<hr>
<h2 id='ondisc'>ondisc: A package for out-of-memory computing on single-cell data</h2><span id='topic+ondisc'></span>

<h3>Description</h3>

<p>Single-cell datasets are large and are growing in size as sequencing costs drop. The ondisc package is designed to facilitate large-scale computing on single-cell expression data by providing access to expression matrices out-of-memory. ondisc is functional (i.e., all objects are persistent) and efficient (i.e., all algorithms are theoretically optimal in time).
</p>

<hr>
<h2 id='ondisc_matrix'><code>ondisc_matrix</code> class</h2><span id='topic+ondisc_matrix'></span>

<h3>Description</h3>

<p>An <code>ondisc_matrix</code> represents a feature-by-cell expression matrix stored on-disk.
</p>
<p>Construct an <code>ondisc_matrix</code> from an initialized .h5 file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ondisc_matrix(h5_file)

ondisc_matrix(h5_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ondisc_matrix_+3A_h5_file">h5_file</code></td>
<td>
<p>a .h5 file storing the on-disk portion of an initialized <code>ondisc_matrix</code> object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is best to avoid interacting with the slots of an <code>ondisc_matrix</code> directly. Instead, use the functions
and operators provided by the package.
</p>


<h3>Value</h3>

<p>an initialized <code>ondisc_matrix</code> object.
</p>


<h3>Slots</h3>


<dl>
<dt><code>h5_file</code></dt><dd><p>path to an initialized .h5 file stored on-disk.</p>
</dd>
<dt><code>logical_mat</code></dt><dd><p>logical value indicating whether the matrix is logical.</p>
</dd>
<dt><code>cell_subset</code></dt><dd><p>integer vector recording the cells currently in use.</p>
</dd>
<dt><code>feature_subset</code></dt><dd><p>integer vector recording the features currently in use.</p>
</dd>
<dt><code>underlying_dimension</code></dt><dd><p>the dimension of the (unsubset) expression matrix.</p>
</dd>
</dl>


<h3>Examples</h3>

<pre><code class='language-R'># NOTE: You must create the HDF5 file "expressions.h5" to run this example.
# Navigate to the help file of "create_ondisc_matrix_from_mtx"
# (via ?create_ondisc_matrix_from_mtx), and execute the code in the first code block.
h5_fp &lt;- paste0(tempdir(), "/expressions.h5")
if (file.exists(h5_fp)) {
  odm &lt;- ondisc_matrix(h5_file = h5_fp)
}
</code></pre>

<hr>
<h2 id='show'>Print basic information to the console</h2><span id='topic+show'></span><span id='topic+show+2Condisc_matrix-method'></span><span id='topic+show+2Cmetadata_ondisc_matrix-method'></span><span id='topic+show+2Cmultimodal_ondisc_matrix-method'></span>

<h3>Description</h3>

<p>Print basic information to the console
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ondisc_matrix'
show(object)

## S4 method for signature 'metadata_ondisc_matrix'
show(object)

## S4 method for signature 'multimodal_ondisc_matrix'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show_+3A_object">object</code></td>
<td>
<p>an object of class ondisc_matrix, covaraite_ondisc_matrix, or multimodal_ondisc_matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL; called for printing
</p>


<h3>Examples</h3>

<pre><code class='language-R'># NOTE: You must create the HDF5 file "expressions.h5" to run this example.
# Navigate to the help file of "create_ondisc_matrix_from_mtx"
# (via ?create_ondisc_matrix_from_mtx), and execute the code in the first code block.

h5_fp &lt;- paste0(tempdir(), "/expressions.h5")
if (file.exists(h5_fp)) {
odm &lt;- ondisc_matrix(h5_file = h5_fp)
show(odm)
}
</code></pre>

<hr>
<h2 id='subset-odm'>Subset using the <code>[</code> operator.</h2><span id='topic+subset-odm'></span><span id='topic++5B+2Condisc_matrix+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Condisc_matrix+2CANY+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Condisc_matrix+2Cmissing+2CANY+2Cmissing-method'></span><span id='topic++5B+2Condisc_matrix+2CANY+2CANY+2Cmissing-method'></span><span id='topic++5B+2Cmetadata_ondisc_matrix+2CANY+2CANY+2Cmissing-method'></span><span id='topic++5B+2Cmetadata_ondisc_matrix+2CANY+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cmetadata_ondisc_matrix+2Cmissing+2CANY+2Cmissing-method'></span><span id='topic++5B+2Cmetadata_ondisc_matrix+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cmultimodal_ondisc_matrix+2Cmissing+2Cmissing+2Cmissing-method'></span><span id='topic++5B+2Cmultimodal_ondisc_matrix+2Cmissing+2CANY+2Cmissing-method'></span><span id='topic++5B+2Cmultimodal_ondisc_matrix+2CANY+2CANY+2CANY-method'></span>

<h3>Description</h3>

<p>Apply the <code>[</code> operator to an <code>ondisc_matrix</code>, <code>metadata_ondisc_matrix</code>, or <code>multimodal_ondisc_matrix</code>
to subset the object. You can pass logical, character, or numeric vectors to <code>[</code>; character
vectors are assumed to refer to feature IDs (for rows) and cell barcodes (for columns).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'ondisc_matrix,missing,missing,missing'
x[i, j, drop]

## S4 method for signature 'ondisc_matrix,ANY,missing,missing'
x[i, j]

## S4 method for signature 'ondisc_matrix,missing,ANY,missing'
x[i, j]

## S4 method for signature 'ondisc_matrix,ANY,ANY,missing'
x[i, j]

## S4 method for signature 'metadata_ondisc_matrix,ANY,ANY,missing'
x[i, j, drop]

## S4 method for signature 'metadata_ondisc_matrix,ANY,missing,missing'
x[i, j, drop]

## S4 method for signature 'metadata_ondisc_matrix,missing,ANY,missing'
x[i, j, drop]

## S4 method for signature 'metadata_ondisc_matrix,missing,missing,missing'
x[i, j, drop]

## S4 method for signature 'multimodal_ondisc_matrix,missing,missing,missing'
x[i, j, drop]

## S4 method for signature 'multimodal_ondisc_matrix,missing,ANY,missing'
x[i, j, drop]

## S4 method for signature 'multimodal_ondisc_matrix,ANY,ANY,ANY'
x[i, j, drop]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset-odm_+3A_x">x</code></td>
<td>
<p>an <code>ondisc_matrix</code>, <code>metadata_ondisc_matrix</code>, or <code>multimodal_ondisc_matrix</code> object.</p>
</td></tr>
<tr><td><code id="subset-odm_+3A_i">i</code></td>
<td>
<p>a vector (numeric, logical, or character) indicating features to keep.</p>
</td></tr>
<tr><td><code id="subset-odm_+3A_j">j</code></td>
<td>
<p>a vector (numeric, logical, or character) indicating cells to keep.</p>
</td></tr>
<tr><td><code id="subset-odm_+3A_drop">drop</code></td>
<td>
<p>not used</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can subset an <code>ondisc_matrix</code> and a <code>metadata_ondisc_matrix</code> by cell and/or feature. You can subset a
<code>multimodal_ondisc_matrix</code> by cell only (because the features differ across modalities).
</p>


<h3>Value</h3>

<p>An appropriately subset object of the same class as <code>x</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># NOTE: You must create the RDS files "expressions.rds" and
# "perturbations.rds" to run this example. Navigate to the help file of
# "create_ondisc_matrix_from_mtx" (via ?create_ondisc_matrix_from_mtx),
# and execute both code blocks.

# subset an ondisc_matrix
h5_fp &lt;- paste0(tempdir(), "/expressions.h5")
if (file.exists(h5_fp)) {
odm &lt;- ondisc_matrix(h5_file = h5_fp)
# keep cells 100-110
x &lt;- odm[,100:110]
# keep all cells except 50, 100, 150
x &lt;- odm[,-c(50, 100, 150)]
# keep genes ENSG00000188305, ENSG00000257284, and ENSG00000251655:
x &lt;- odm[c("ENSG00000188305", "ENSG00000257284", "ENSG00000251655"),]
# keep the cells CTTAGGACACTGGCGT-1 and AAAGGATTCACATCAG-1:
x &lt;- odm[,c("CTTAGGACACTGGCGT-1", "AAAGGATTCACATCAG-1")]
# keep all genes except ENSG00000188305 and ENSG00000257284
x &lt;- odm[!(get_feature_ids(odm) %in% c("ENSG00000188305", "ENSG00000257284")),]
}

# subset a metadata_ondic_matrix
expressions_fp &lt;- paste0(tempdir(), "/expressions.rds")
if (file.exists(expressions_fp)) {
expressions &lt;- readRDS(expressions_fp)
# keep cells 100-110
x &lt;- expressions[,100:110]
# keep genes ENSG00000188305, ENSG00000257284, and ENSG00000251655
x &lt;- expressions[c("ENSG00000188305", "ENSG00000257284", "ENSG00000251655"),]
}

# subset a multimodal ondisc_matrix
expression_fp &lt;- paste0(tempdir(), "/expressions.rds")
perturbations_fp &lt;- paste0(tempdir(), "/perturbations.rds")
if (file.exists(expression_fp) &amp;&amp; file.exists(perturbations_fp)) {
    expressions &lt;- readRDS(expression_fp)
    perturbations &lt;- readRDS(expression_fp)
    crispr_experiment &lt;- multimodal_ondisc_matrix(list(expressions = expressions,
    perturbations = perturbations))
    # Keep all cells except 10,100, and 105.
    x &lt;- crispr_experiment[,-c(10,100,105)]
    # Keep the first 5 cells
    x &lt;- crispr_experiment[,1:5]
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
