<!DOCTYPE html><html lang="en"><head><title>Help for package ontologySimilarity</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ontologySimilarity}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ontologySimilarity-package'>
<p>Functions for Calculating Ontological Similarities</p></a></li>
<li><a href='#create_sim_index'><p>Create similarity index for list of term sets</p></a></li>
<li><a href='#descendants_IC'><p>Get information content based on number of descendants each term has</p></a></li>
<li><a href='#gene_GO_terms'><p>Gene Ontology annotation of genes</p></a></li>
<li><a href='#get_asym_sim_grid'><p>Get asymmetrical similarity matrix</p></a></li>
<li><a href='#get_profile_sims'><p>Get similarities of term sets to profile</p></a></li>
<li><a href='#get_sim'><p>Calculate the group similarity of a set of row/column indices</p></a></li>
<li><a href='#get_sim_grid'><p>Get similarity matrix of pairwise similarities of term sets.</p></a></li>
<li><a href='#get_sim_p'><p>Get similarity p-value</p></a></li>
<li><a href='#get_sim_p_from_ontology'><p>Get similarity p-value for subgroup term sets</p></a></li>
<li><a href='#get_similarity_rank_matrix'><p>Get matrix of similarity rank from similarity matrix</p></a></li>
<li><a href='#get_term_set_to_term_sims'><p>Get &lsquo;term sets to term&rsquo; similarity matrix</p></a></li>
<li><a href='#get_term_sim_mat'><p>Get term-term similarity matrix</p></a></li>
<li><a href='#GO_IC'><p>Gene Ontology terms information content.</p></a></li>
<li><a href='#group_term_enrichment'><p>Identify enriched terms in subgroup</p></a></li>
<li><a href='#lin'><p>Calculate Lin similarity score of two term sets</p></a></li>
<li><a href='#resnik'><p>Calculate Resnik similarity score of two term sets</p></a></li>
<li><a href='#sample_group_sim'><p>Draw sample of group similarities</p></a></li>
<li><a href='#sample_group_sim_from_ontology'><p>Draw sample of group similarities</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Calculating Ontological Similarities</td>
</tr>
<tr>
<td>Version:</td>
<td>2.7</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel Greene</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel Greene &lt;dg333@cam.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Calculate similarity between ontological terms and sets of ontological terms based on term information content and assess statistical significance of similarity in the context of a collection of terms sets - Greene et al. 2017 &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw763">doi:10.1093/bioinformatics/btw763</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.0), ontologyIndex (&ge; 2.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, paintmap</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-29 05:16:35 UTC; dg</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-29 05:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ontologySimilarity-package'>
Functions for Calculating Ontological Similarities
</h2><span id='topic+ontologySimilarity-package'></span><span id='topic+ontologySimilarity'></span>

<h3>Description</h3>

<p>Functions for calculating semantic similarities between ontological terms or sets of ontological terms based on term information content and assessing statistical significance of similarity in the context of a collection of sets of ontological terms.
</p>


<h3>Details</h3>

<p>Semantic similarity and similarity significance functions based on Resnik and Lin's measures of similarity. Computationally intensive functions are written in C++ for performance.
</p>


<h3>Author(s)</h3>

<p>Daniel Greene &lt;dg333@cam.ac.uk&gt;
</p>
<p>Maintainer: Daniel Greene &lt;dg333@cam.ac.uk&gt;
</p>


<h3>References</h3>

<p>Greene D, Richardson S, Turro E (2017). 'ontologyX: a suite of R packages for working with ontological data. _Bioinformatics_, 33(7), 1104&ndash;1106.
</p>
<p>Westbury SK, Turro E, Greene D, Lentaigne C, Kelly AM, Bariana TK, Simeoni I, Pillois X, Attwood A, Austin S, Jansen SB, Bakchoul T, Crisp-Hihn A, Erber WN, Favier R, Foad N, Gattens M, Jolley JD, Liesner R, Meacham S, Millar CM, Nurden AT, Peerlinck K, Perry DJ, Poudel P, Schulman S, Schulze H, Stephens JC, Furie B, Robinson PN, Geet Cv, Rendon A, Gomez K, Laffan MA, Lambert MP, Nurden P, Ouwehand WH, Richardson S, Mumford AD and Freson K (2015). &lsquo;Human phenotype ontology annotation and cluster analysis to unravel genetic defects in 707 cases with unexplained bleeding and platelet disorders.&rsquo; _Genome Med_, *7*(1), pp. 36.
</p>
<p>Kohler S, Doelken SC, Mungall CJ, Bauer S, Firth HV, Bailleul-Forestier I, Black GC, Brown DL, Brudno M, Campbell J, FitzPatrick DR, Eppig JT, Jackson AP, Freson K, Girdea M, Helbig I, Hurst JA, Jahn J, Jackson LG, Kelly AM, Ledbetter DH, Mansour S, Martin CL, Moss C, Mumford A, Ouwehand WH, Park SM, Riggs ER, Scott RH, Sisodiya S, Van Vooren S, Wapner RJ, Wilkie AO, Wright CF, Vulto-van Silfhout A, de Leeuw N, de Vries B, Washingthon NL, Smith CL, Westerfield M, Schofield P, Ruef BJ, Gkoutos GV, Haendel M, Smedley D, Lewis SE and Robinson PN (2014). &lsquo;The Human Phenotype Ontology project: linking molecular biology and disease through phenotype data.&rsquo; _Nucleic Acids Res._, *42*(Database issue), pp. D966-974.
</p>
<p>Resnik, P. (1995). &lsquo;Using information content to evaluate semantic similarity in a taxonomy&rsquo;. Proceedings of the 14th IJCAI 1, 448-453.
</p>
<p>Lin D (1998). &lsquo;An Information-Theoretic Definition of Similarity.&rsquo; In Shavlik JW (ed.), _Proceedings of the Fifteenth International Conference on Machine Learning (ICML 1998), Madison, Wisconsin, USA, July 24-27, 1998_, pp. 296-304.
</p>

<hr>
<h2 id='create_sim_index'>Create similarity index for list of term sets</h2><span id='topic+create_sim_index'></span>

<h3>Description</h3>

<p>Create light-weight similarity index for fast lookups of between term set similarity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_sim_index(
  ontology,
  term_sets,
  information_content = descendants_IC(ontology),
  term_sim_method = "lin",
  combine = "average"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_sim_index_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="create_sim_index_+3A_term_sets">term_sets</code></td>
<td>
<p>List of character vectors of ontological term IDs.</p>
</td></tr>
<tr><td><code id="create_sim_index_+3A_information_content">information_content</code></td>
<td>
<p>Numeric vector of information contents of terms (named by term)</p>
</td></tr>
<tr><td><code id="create_sim_index_+3A_term_sim_method">term_sim_method</code></td>
<td>
<p>Character string equalling either &quot;lin&quot; or &quot;resnik&quot; to use Lin or Resnik's expression for the similarity of terms.</p>
</td></tr>
<tr><td><code id="create_sim_index_+3A_combine">combine</code></td>
<td>
<p>Character string - either &quot;average&quot; or &quot;product&quot;, indicating whether to use the best-match-product' method, or function accepting two arguments - the first, the similarity matrix obtained by averaging across term sets in <code>term_sets</code>, and the second averaging across those in <code>term_sets2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code>sim_index</code>.
</p>


<h3>See Also</h3>

<p><code>link{get_sim}</code> <code><a href="#topic+get_sim_p">get_sim_p</a></code> <code><a href="#topic+sample_group_sim">sample_group_sim</a></code>
</p>

<hr>
<h2 id='descendants_IC'>Get information content based on number of descendants each term has</h2><span id='topic+descendants_IC'></span>

<h3>Description</h3>

<p>Calculate information content of terms based on frequency with which it is an ancestor of other terms. Useful as a default if there is no population frequency information available as it captures the structure of the ontology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>descendants_IC(ontology)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="descendants_IC_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of information contents named by term.
</p>

<hr>
<h2 id='gene_GO_terms'>Gene Ontology annotation of genes</h2><span id='topic+gene_GO_terms'></span>

<h3>Description</h3>

<p><code>list</code> object containing character vectors of term IDs of GO terms annotating each gene, named by gene. Users can select a list of annotations for a subset of the annotated genes using a character vector of gene symbols, e.g. <code>gene_GO_terms[c("ACTN1", "TUBB1")]</code>, which can then be used in functions for calculating similarities, e.g. <code><a href="#topic+get_sim_grid">get_sim_grid</a></code>. Note that these annotation vectors contain annotation from all major branches of the Gene Ontology, however one can simply extract the terms only relevant to one by calling the function in the <code>ontologyIndex</code> package: <code>intersection_with_descendants</code>.
</p>


<h3>Format</h3>

<p>List of character vectors.
</p>


<h3>References</h3>

<p>Annotation downloaded from Gene Ontology consortium website, http://geneontology.org/, dated 20/02/2024.
</p>

<hr>
<h2 id='get_asym_sim_grid'>Get asymmetrical similarity matrix</h2><span id='topic+get_asym_sim_grid'></span>

<h3>Description</h3>

<p>Create a numeric matrix of similarities between two lists of term sets, but only averaging over the terms in sets from <code>A</code> the similarities of the best matches in sets from <code>B</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_asym_sim_grid(A, B, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_asym_sim_grid_+3A_a">A</code></td>
<td>
<p>List of term sets.</p>
</td></tr>
<tr><td><code id="get_asym_sim_grid_+3A_b">B</code></td>
<td>
<p>List of term sets.</p>
</td></tr>
<tr><td><code id="get_asym_sim_grid_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed to <code><a href="#topic+get_sim_grid">get_sim_grid</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric matrix of similarities
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_sim_grid">get_sim_grid</a></code> <code><a href="#topic+get_profile_sims">get_profile_sims</a></code>
</p>

<hr>
<h2 id='get_profile_sims'>Get similarities of term sets to profile</h2><span id='topic+get_profile_sims'></span>

<h3>Description</h3>

<p>Get numeric vector of similarities between each item in a list of term sets and another &lsquo;ontological profile&rsquo;, i.e. a single term set. Similarity averaging over terms in <code>term_sets</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_profile_sims(profile, term_sets, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_profile_sims_+3A_profile">profile</code></td>
<td>
<p>Character vector of term IDs.</p>
</td></tr>
<tr><td><code id="get_profile_sims_+3A_term_sets">term_sets</code></td>
<td>
<p>List of character vectors of ontological term IDs.</p>
</td></tr>
<tr><td><code id="get_profile_sims_+3A_...">...</code></td>
<td>
<p>Other arguments to pass to <code><a href="#topic+get_sim_grid">get_sim_grid</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of profile similarities.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_asym_sim_grid">get_asym_sim_grid</a></code> <code><a href="#topic+get_sim_grid">get_sim_grid</a></code>
</p>

<hr>
<h2 id='get_sim'>Calculate the group similarity of a set of row/column indices</h2><span id='topic+get_sim'></span><span id='topic+get_sim.integer'></span><span id='topic+get_sim.numeric'></span><span id='topic+get_sim.matrix'></span><span id='topic+get_sim.sim_index'></span><span id='topic+get_sim.default'></span>

<h3>Description</h3>

<p>Calculates the similarity of a group within a population by applying the function specified by <code>group_sim</code> to the pairwise similarities of group members.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sim(pop_sim, ...)

## S3 method for class 'integer'
get_sim(pop_sim, ...)

## S3 method for class 'numeric'
get_sim(pop_sim, group = seq(length(pop_sim)), ...)

## S3 method for class 'matrix'
get_sim(pop_sim, group = seq(nrow(pop_sim)), ...)

## S3 method for class 'sim_index'
get_sim(pop_sim, group = seq(pop_sim[["N"]]), ...)

## Default S3 method:
get_sim(pop_sim, group, type, group_sim = "average", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_sim_+3A_pop_sim">pop_sim</code></td>
<td>
<p>An object representing the similarities of an indexed population of objects.</p>
</td></tr>
<tr><td><code id="get_sim_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed to <code>get_sim</code>.</p>
</td></tr>
<tr><td><code id="get_sim_+3A_group">group</code></td>
<td>
<p>Character or integer vector specifying names/indices of subgroup for which to calculate a group similarity p-value.</p>
</td></tr>
<tr><td><code id="get_sim_+3A_type">type</code></td>
<td>
<p>Either &quot;matrix&quot;, &quot;sim_index&quot; or &quot;numeric&quot; - the type of the <code>pop_sim</code> object.</p>
</td></tr>
<tr><td><code id="get_sim_+3A_group_sim">group_sim</code></td>
<td>
<p>String Either &quot;average&quot; or &quot;min&quot;, determining how to calculate the similarity of a group of term sets over all pairwise combinations of group members</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of group similarity
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_sim_p">get_sim_p</a></code> <code><a href="#topic+sample_group_sim">sample_group_sim</a></code>
</p>

<hr>
<h2 id='get_sim_grid'>Get similarity matrix of pairwise similarities of term sets.</h2><span id='topic+get_sim_grid'></span>

<h3>Description</h3>

<p>Using either an <code>ontology_index</code> object and numeric vector of information content per term - or a matrix of between-term similarities (e.g. the output of <code><a href="#topic+get_term_sim_mat">get_term_sim_mat</a></code>), create a numeric matrix of &lsquo;between-term set&rsquo; similarities. Either the &lsquo;best-match-average&rsquo; or &lsquo;best-match-product&rsquo; approach (i.e. where the 2 scores obtained by applying the asymmetric &lsquo;best-match&rsquo; similarity function to two term sets in each order are combined by taking the average or the product respectively). Either Lin's (default) or Resnik's definition of term similarity can be used. If <code>information_content</code> is not specified, a default value from <code><a href="#topic+descendants_IC">descendants_IC</a></code> is generated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sim_grid(
  ontology,
  information_content,
  term_sim_method,
  term_sim_mat,
  term_sets,
  term_sets2 = term_sets,
  combine = "average"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_sim_grid_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="get_sim_grid_+3A_information_content">information_content</code></td>
<td>
<p>Numeric vector of information contents of terms (named by term)</p>
</td></tr>
<tr><td><code id="get_sim_grid_+3A_term_sim_method">term_sim_method</code></td>
<td>
<p>Character string equalling either &quot;lin&quot; or &quot;resnik&quot; to use Lin or Resnik's expression for the similarity of terms.</p>
</td></tr>
<tr><td><code id="get_sim_grid_+3A_term_sim_mat">term_sim_mat</code></td>
<td>
<p>Numeric matrix with rows and columns corresponding to (and named by) term IDs, and cells containing the similarity between the row and column term</p>
</td></tr>
<tr><td><code id="get_sim_grid_+3A_term_sets">term_sets</code></td>
<td>
<p>List of character vectors of ontological term IDs.</p>
</td></tr>
<tr><td><code id="get_sim_grid_+3A_term_sets2">term_sets2</code></td>
<td>
<p>Second set of term sets.</p>
</td></tr>
<tr><td><code id="get_sim_grid_+3A_combine">combine</code></td>
<td>
<p>Character string - either &quot;average&quot; or &quot;product&quot;, indicating whether to use the best-match-product' method, or function accepting two arguments - the first, the similarity matrix obtained by averaging across term sets in <code>term_sets</code>, and the second averaging across those in <code>term_sets2</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that if any term set within <code>term_sets</code> has 0 terms associated with it, it will get a similarity of 0 to any other set. If you do not want to compare term sets with no annotation, take care to filter out empty sets first, e.g. by 'term_sets=term_sets[sapply(term_sets, length) &gt; 0]'.
</p>


<h3>Value</h3>

<p>Numeric matrix of pairwise term set similarities.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_term_sim_mat">get_term_sim_mat</a></code> <code><a href="#topic+get_sim_p">get_sim_p</a></code> <code><a href="#topic+get_asym_sim_grid">get_asym_sim_grid</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ontologyIndex)
data(hpo)
term_sets &lt;- list(
`case1`=c("HP:0001873", "HP:0011877"),
`case2`=c("HP:0001872", "HP:0001892"),
`case3`="HP:0001873")
get_sim_grid(ontology=hpo, term_sets=term_sets)
</code></pre>

<hr>
<h2 id='get_sim_p'>Get similarity p-value</h2><span id='topic+get_sim_p'></span><span id='topic+get_sim_p.integer'></span><span id='topic+get_sim_p.numeric'></span><span id='topic+get_sim_p.matrix'></span><span id='topic+get_sim_p.sim_index'></span><span id='topic+get_sim_p.default'></span>

<h3>Description</h3>

<p>p-value of group similarity, calculated by estimating the proportion by random sampling of groups the same size as <code>group</code> which have at least as great group similarity than does <code>group</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sim_p(pop_sim, ...)

## S3 method for class 'integer'
get_sim_p(pop_sim, ...)

## S3 method for class 'numeric'
get_sim_p(pop_sim, group, ...)

## S3 method for class 'matrix'
get_sim_p(pop_sim, group, ...)

## S3 method for class 'sim_index'
get_sim_p(pop_sim, group, ...)

## Default S3 method:
get_sim_p(
  pop_sim,
  group,
  type,
  min_its = 1000,
  max_its = 1e+05,
  signif = 0.05,
  log_dismiss = log(1e-06),
  group_sim = "average",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_sim_p_+3A_pop_sim">pop_sim</code></td>
<td>
<p>An object representing the similarities of an indexed population of objects.</p>
</td></tr>
<tr><td><code id="get_sim_p_+3A_...">...</code></td>
<td>
<p>Arguments for <code>get_sim_p</code>.</p>
</td></tr>
<tr><td><code id="get_sim_p_+3A_group">group</code></td>
<td>
<p>Character or integer vector specifying names/indices of subgroup for which to calculate a group similarity p-value.</p>
</td></tr>
<tr><td><code id="get_sim_p_+3A_type">type</code></td>
<td>
<p>Either &quot;matrix&quot;, &quot;sim_index&quot; or &quot;numeric&quot; - the type of the <code>pop_sim</code> object.</p>
</td></tr>
<tr><td><code id="get_sim_p_+3A_min_its">min_its</code></td>
<td>
<p>Minimum number of simulated group similarities to calculate</p>
</td></tr>
<tr><td><code id="get_sim_p_+3A_max_its">max_its</code></td>
<td>
<p>Maximum number of simulated group similarities to calculate</p>
</td></tr>
<tr><td><code id="get_sim_p_+3A_signif">signif</code></td>
<td>
<p>Threshold p-value of statistical significance</p>
</td></tr>
<tr><td><code id="get_sim_p_+3A_log_dismiss">log_dismiss</code></td>
<td>
<p>Threshold of log probability, below which to trigger return of current estimated p-value</p>
</td></tr>
<tr><td><code id="get_sim_p_+3A_group_sim">group_sim</code></td>
<td>
<p>String Either &quot;average&quot; or &quot;min&quot;, determining how to calculate the similarity of a group of term sets over all pairwise combinations of group members</p>
</td></tr>
</table>


<h3>Value</h3>

<p>p-value.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_sim">get_sim</a></code> <code><a href="#topic+sample_group_sim">sample_group_sim</a></code>
</p>

<hr>
<h2 id='get_sim_p_from_ontology'>Get similarity p-value for subgroup term sets</h2><span id='topic+get_sim_p_from_ontology'></span>

<h3>Description</h3>

<p>Compute a similarity p-value by permutation for subgroup of a list of term sets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_sim_p_from_ontology(
  ontology,
  term_sets,
  information_content = descendants_IC(ontology),
  term_sim_method = "lin",
  combine = "average",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_sim_p_from_ontology_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="get_sim_p_from_ontology_+3A_term_sets">term_sets</code></td>
<td>
<p>List of character vectors of ontological term IDs.</p>
</td></tr>
<tr><td><code id="get_sim_p_from_ontology_+3A_information_content">information_content</code></td>
<td>
<p>Numeric vector of information contents of terms (named by term)</p>
</td></tr>
<tr><td><code id="get_sim_p_from_ontology_+3A_term_sim_method">term_sim_method</code></td>
<td>
<p>Character string equalling either &quot;lin&quot; or &quot;resnik&quot; to use Lin or Resnik's expression for the similarity of terms.</p>
</td></tr>
<tr><td><code id="get_sim_p_from_ontology_+3A_combine">combine</code></td>
<td>
<p>Character string - either &quot;average&quot; or &quot;product&quot;, indicating whether to use the best-match-product' method, or function accepting two arguments - the first, the similarity matrix obtained by averaging across term sets in <code>term_sets</code>, and the second averaging across those in <code>term_sets2</code>.</p>
</td></tr>
<tr><td><code id="get_sim_p_from_ontology_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed to <code><a href="#topic+get_sim_p">get_sim_p</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_sim_p">get_sim_p</a></code> <code><a href="#topic+create_sim_index">create_sim_index</a></code>
</p>

<hr>
<h2 id='get_similarity_rank_matrix'>Get matrix of similarity rank from similarity matrix</h2><span id='topic+get_similarity_rank_matrix'></span>

<h3>Description</h3>

<p>Given a lower triangular similarity matrix, construct a distance matrix where the rows are the ranks of the column cases with respect to similarity to the row case. If relative similarity is of interest, this rank-transformation may reduce bias in favour of high similarity scores in downstream analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_similarity_rank_matrix(similarity_matrix, symmetric = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_similarity_rank_matrix_+3A_similarity_matrix">similarity_matrix</code></td>
<td>
<p>Lower triangular numeric matrix of similarities, where the rownames and colnames are identical to the case IDs.</p>
</td></tr>
<tr><td><code id="get_similarity_rank_matrix_+3A_symmetric">symmetric</code></td>
<td>
<p>Logical value determining whether to &lsquo;symmetrify&rsquo; resultant matrix by averaging rank similarity of A -&gt; B and B -&gt; A.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of rank similarities.
</p>

<hr>
<h2 id='get_term_set_to_term_sims'>Get &lsquo;term sets to term&rsquo; similarity matrix</h2><span id='topic+get_term_set_to_term_sims'></span>

<h3>Description</h3>

<p>Create a numeric matrix of similarities between term sets and individual terms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_term_set_to_term_sims(term_sets, terms, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_term_set_to_term_sims_+3A_term_sets">term_sets</code></td>
<td>
<p>List of character vectors of ontological term IDs.</p>
</td></tr>
<tr><td><code id="get_term_set_to_term_sims_+3A_terms">terms</code></td>
<td>
<p>Character vector of ontological terms.</p>
</td></tr>
<tr><td><code id="get_term_set_to_term_sims_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed to <code><a href="#topic+get_sim_grid">get_sim_grid</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric matrix of term set-to-term similarities
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_sim_grid">get_sim_grid</a></code>
</p>

<hr>
<h2 id='get_term_sim_mat'>Get term-term similarity matrix</h2><span id='topic+get_term_sim_mat'></span>

<h3>Description</h3>

<p>Get matrix of pairwise similarity of individual terms based on Lin's (default) or Resnik's information content-based expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_term_sim_mat(
  ontology,
  information_content,
  method = "lin",
  row_terms = names(information_content),
  col_terms = names(information_content)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_term_sim_mat_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="get_term_sim_mat_+3A_information_content">information_content</code></td>
<td>
<p>Numeric vector of information contents of terms (named by term)</p>
</td></tr>
<tr><td><code id="get_term_sim_mat_+3A_method">method</code></td>
<td>
<p>Character value equalling either &quot;lin&quot; or &quot;resnik&quot; to use Lin or Resnik's expression for similarity of terms respectively.</p>
</td></tr>
<tr><td><code id="get_term_sim_mat_+3A_row_terms">row_terms</code></td>
<td>
<p>Character vector of term IDs to appear as rows of result matrix.</p>
</td></tr>
<tr><td><code id="get_term_sim_mat_+3A_col_terms">col_terms</code></td>
<td>
<p>Character vector of term IDs to appear as cols of result matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric matrix of pairwise term similarities.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_sim_grid">get_sim_grid</a></code> <code><a href="#topic+resnik">resnik</a></code>, <code><a href="#topic+lin">lin</a></code>
</p>

<hr>
<h2 id='GO_IC'>Gene Ontology terms information content.</h2><span id='topic+GO_IC'></span>

<h3>Description</h3>

<p>Numeric vector containing the information content of Gene Ontology terms based on frequencies of annotation data object <code>gene_GO_terms</code>. The object can be derived using the function <code>get_term_info_content</code> and data object <code>go</code> from the <code>ontologyIndex</code> package.
</p>


<h3>Format</h3>

<p>List of character vectors.
</p>

<hr>
<h2 id='group_term_enrichment'>Identify enriched terms in subgroup</h2><span id='topic+group_term_enrichment'></span>

<h3>Description</h3>

<p>Create a table of terms ranked by their significance of occurrence in a set of term sets amongst an enclosing set, with p-values computed by permutation. Terms are subselected so that only the minimal set of non-redundant terms at each level of frequency within the group are retained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>group_term_enrichment(
  ontology,
  term_sets,
  group,
  permutations = 1000L,
  min_terms = 2L,
  mc.cores = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="group_term_enrichment_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="group_term_enrichment_+3A_term_sets">term_sets</code></td>
<td>
<p>List of character vectors of ontological term IDs.</p>
</td></tr>
<tr><td><code id="group_term_enrichment_+3A_group">group</code></td>
<td>
<p>Integer/logical/character vector specifying indices/positions/names of subgroup for which to calculate a group similarity p-value.</p>
</td></tr>
<tr><td><code id="group_term_enrichment_+3A_permutations">permutations</code></td>
<td>
<p>Number of permutations to test against, or if <code>NULL</code>, perform no permutations and return the unadjusted p-values for the occurrence of each term.</p>
</td></tr>
<tr><td><code id="group_term_enrichment_+3A_min_terms">min_terms</code></td>
<td>
<p>Minimum number of times a term should occur within the given group to be eligible for inclusion in the results.</p>
</td></tr>
<tr><td><code id="group_term_enrichment_+3A_mc.cores">mc.cores</code></td>
<td>
<p>If not null and greater than on, the number of cores use calculating permutations (passed to <code>mclapply</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>data.frame</code> containing columns: <code>term</code> (with the term ID); <code>name</code> (term readable name); <code>in_term</code> (number of sets in the given group of containing the term); <code>in_no_term</code> (number of sets in the given group not containing the term); <code>out_term</code> and <code>out_no_term</code> (equivalently for the sets not in the given group); <code>p</code> (the p-values calculated by permutation for seeing a term with such a strong association, measured using Fisher's exact test, in a group of term sets the size of the given group among <code>term_sets</code>). Rows ordered by significance (i.e. the <code>p</code> columns).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sample_group_sim">sample_group_sim</a></code> <code><a href="#topic+create_sim_index">create_sim_index</a></code>
</p>

<hr>
<h2 id='lin'>Calculate Lin similarity score of two term sets</h2><span id='topic+lin'></span>

<h3>Description</h3>

<p>Warning! This function is slow - performing large numbers of &lsquo;between term-set&rsquo; similarity calculations should be done using <code><a href="#topic+get_sim_grid">get_sim_grid</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lin(ontology, information_content, term_set_1, term_set_2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lin_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="lin_+3A_information_content">information_content</code></td>
<td>
<p>Numeric vector of information contents of terms (named by term)</p>
</td></tr>
<tr><td><code id="lin_+3A_term_set_1">term_set_1</code></td>
<td>
<p>Character vector of terms.</p>
</td></tr>
<tr><td><code id="lin_+3A_term_set_2">term_set_2</code></td>
<td>
<p>Character vector of terms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value.
</p>


<h3>References</h3>

<p>Lin D (1998). &lsquo;An Information-Theoretic Definition of Similarity.&rsquo; In Shavlik JW (ed.), _Proceedings of the Fifteenth International Conference on Machine Learning (ICML 1998), Madison, Wisconsin, USA, July 24-27, 1998_, pp. 296-304.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+resnik">resnik</a></code>, <code><a href="#topic+get_term_sim_mat">get_term_sim_mat</a></code>
</p>

<hr>
<h2 id='resnik'>Calculate Resnik similarity score of two term sets</h2><span id='topic+resnik'></span>

<h3>Description</h3>

<p>Warning! This function is slow - performing large numbers of &lsquo;between term-set&rsquo; similarity calculations should be done using <code><a href="#topic+get_sim_grid">get_sim_grid</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>resnik(ontology, information_content, term_set_1, term_set_2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="resnik_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="resnik_+3A_information_content">information_content</code></td>
<td>
<p>Numeric vector of information contents of terms (named by term)</p>
</td></tr>
<tr><td><code id="resnik_+3A_term_set_1">term_set_1</code></td>
<td>
<p>Character vector of terms.</p>
</td></tr>
<tr><td><code id="resnik_+3A_term_set_2">term_set_2</code></td>
<td>
<p>Character vector of terms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value.
</p>


<h3>References</h3>

<p>Resnik, P. (1995). &lsquo;Using information content to evaluate semantic similarity in a taxonomy&rsquo;. Proceedings of the 14th IJCAI 1, 448-453.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lin">lin</a></code>, <code><a href="#topic+get_term_sim_mat">get_term_sim_mat</a></code>
</p>

<hr>
<h2 id='sample_group_sim'>Draw sample of group similarities</h2><span id='topic+sample_group_sim'></span><span id='topic+sample_group_sim.integer'></span><span id='topic+sample_group_sim.numeric'></span><span id='topic+sample_group_sim.matrix'></span><span id='topic+sample_group_sim.sim_index'></span><span id='topic+sample_group_sim.default'></span>

<h3>Description</h3>

<p>Draw sample of group similarities of groups of given size
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_group_sim(pop_sim, ...)

## S3 method for class 'integer'
sample_group_sim(pop_sim, ...)

## S3 method for class 'numeric'
sample_group_sim(pop_sim, ...)

## S3 method for class 'matrix'
sample_group_sim(pop_sim, ...)

## S3 method for class 'sim_index'
sample_group_sim(pop_sim, ...)

## Default S3 method:
sample_group_sim(
  pop_sim,
  type,
  group_size,
  group_sim = "average",
  sample_size = 10000,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sample_group_sim_+3A_pop_sim">pop_sim</code></td>
<td>
<p>An object representing the similarities of an indexed population of objects.</p>
</td></tr>
<tr><td><code id="sample_group_sim_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed to <code>sample_group_sim</code>.</p>
</td></tr>
<tr><td><code id="sample_group_sim_+3A_type">type</code></td>
<td>
<p>Either &quot;matrix&quot;, &quot;sim_index&quot; or &quot;numeric&quot; - the type of the <code>pop_sim</code> object.</p>
</td></tr>
<tr><td><code id="sample_group_sim_+3A_group_size">group_size</code></td>
<td>
<p>Integer giving the number of members of a group.</p>
</td></tr>
<tr><td><code id="sample_group_sim_+3A_group_sim">group_sim</code></td>
<td>
<p>String Either &quot;average&quot; or &quot;min&quot;, determining how to calculate the similarity of a group of term sets over all pairwise combinations of group members</p>
</td></tr>
<tr><td><code id="sample_group_sim_+3A_sample_size">sample_size</code></td>
<td>
<p>Number of samples to draw.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of random group similarities.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_sim">get_sim</a></code> <code><a href="#topic+get_sim_p">get_sim_p</a></code>
</p>

<hr>
<h2 id='sample_group_sim_from_ontology'>Draw sample of group similarities</h2><span id='topic+sample_group_sim_from_ontology'></span>

<h3>Description</h3>

<p>ample of group similarities for random groups of given drawn from the given <code>ontology</code> argument
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_group_sim_from_ontology(
  ontology,
  term_sets,
  information_content = descendants_IC(ontology),
  term_sim_method = "lin",
  combine = "average",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sample_group_sim_from_ontology_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="sample_group_sim_from_ontology_+3A_term_sets">term_sets</code></td>
<td>
<p>List of character vectors of ontological term IDs.</p>
</td></tr>
<tr><td><code id="sample_group_sim_from_ontology_+3A_information_content">information_content</code></td>
<td>
<p>Numeric vector of information contents of terms (named by term)</p>
</td></tr>
<tr><td><code id="sample_group_sim_from_ontology_+3A_term_sim_method">term_sim_method</code></td>
<td>
<p>Character string equalling either &quot;lin&quot; or &quot;resnik&quot; to use Lin or Resnik's expression for the similarity of terms.</p>
</td></tr>
<tr><td><code id="sample_group_sim_from_ontology_+3A_combine">combine</code></td>
<td>
<p>Character string - either &quot;average&quot; or &quot;product&quot;, indicating whether to use the best-match-product' method, or function accepting two arguments - the first, the similarity matrix obtained by averaging across term sets in <code>term_sets</code>, and the second averaging across those in <code>term_sets2</code>.</p>
</td></tr>
<tr><td><code id="sample_group_sim_from_ontology_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed to <code><a href="#topic+get_sim_p">get_sim_p</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of group similarities.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sample_group_sim">sample_group_sim</a></code> <code><a href="#topic+create_sim_index">create_sim_index</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
