<!DOCTYPE html><html><head><title>Help for package ontologyIndex</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ontologyIndex}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ontologyIndex-package'>
<p>Functions for Reading Ontologies into R</p></a></li>
<li><a href='#as.data.frame.ontology_index'><p>Case <code>ontology_index</code> object to <code>data.frame</code></p></a></li>
<li><a href='#check'><p>Check an ontology index object</p></a></li>
<li><a href='#exclude_descendants'><p>Exclude terms descending from any in a given set of root terms</p></a></li>
<li><a href='#get_ancestors'><p>Get set of terms containing all ancestors of terms in a given set</p></a></li>
<li><a href='#get_descendants'><p>Get set of terms containing all descendants of terms in a given set</p></a></li>
<li><a href='#get_ontology'><p>Read ontology from OBO file into R</p></a></li>
<li><a href='#get_relation_names'><p>Get names of relations used in OBO file</p></a></li>
<li><a href='#get_term_descendancy_matrix'><p>Get logical descendancy matrix for set of terms</p></a></li>
<li><a href='#get_term_frequencies'><p>Get frequency of each term in a set of phenotypes</p></a></li>
<li><a href='#get_term_info_content'><p>Get information content of each term in a set of phenotypes</p></a></li>
<li><a href='#get_term_property'><p>Get property of individual ontological term</p></a></li>
<li><a href='#go'><p>GO index</p></a></li>
<li><a href='#hpo'><p>HPO index</p></a></li>
<li><a href='#intersection_with_descendants'><p>Intersect a set of terms with the descendants of a given set of roots</p></a></li>
<li><a href='#minimal_set'><p>Remove redundant/implied terms from a set of terms</p></a></li>
<li><a href='#mpo'><p>MPO index</p></a></li>
<li><a href='#ontology_index'><p>Create <code>ontology_index</code> object from vectors and lists of term properties</p></a></li>
<li><a href='#print.ontology_index'><p>Print <code>ontology_index</code> object</p></a></li>
<li><a href='#propagate_relations'><p>Select terms by propagating relations from a set of terms</p></a></li>
<li><a href='#prune_descendants'><p>Exclude terms descending from a given set of roots but include those roots which were originally implicitly present.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Reading Ontologies into R</td>
</tr>
<tr>
<td>Version:</td>
<td>2.12</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel Greene &lt;dg333@cam.ac.uk&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel Greene &lt;dg333@cam.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for reading ontologies into R as lists and manipulating sets of ontological terms - 'ontologyX: A suite of R packages for working with ontological data', Greene et al 2017 &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw763">doi:10.1093/bioinformatics/btw763</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-20 21:52:39 UTC; dg</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-27 00:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ontologyIndex-package'>
Functions for Reading Ontologies into R	
</h2><span id='topic+ontologyIndex-package'></span><span id='topic+ontologyIndex'></span>

<h3>Description</h3>

<p>Functions for reading ontologies into R and manipulating sets of ontological terms.
</p>


<h3>Author(s)</h3>

<p>Daniel Greene &lt;dg333@cam.ac.uk&gt;
</p>
<p>Maintainer: Daniel Greene &lt;dg333@cam.ac.uk&gt;
</p>


<h3>References</h3>

<p>Greene D, Richardson S, Turro E (2017). 'ontologyX: a suite of R packages for working with ontological data. _Bioinformatics_, 33(7), 1104&ndash;1106.
</p>
<p>Westbury SK, Turro E, Greene D, Lentaigne C, Kelly AM, Bariana TK, Simeoni I, Pillois X, Attwood A, Austin S, Jansen SB, Bakchoul T, Crisp-Hihn A, Erber WN, Favier R, Foad N, Gattens M, Jolley JD, Liesner R, Meacham S, Millar CM, Nurden AT, Peerlinck K, Perry DJ, Poudel P, Schulman S, Schulze H, Stephens JC, Furie B, Robinson PN, Geet Cv, Rendon A, Gomez K, Laffan MA, Lambert MP, Nurden P, Ouwehand WH, Richardson S, Mumford AD and Freson K (2015). &lsquo;Human phenotype ontology annotation and cluster analysis to unravel genetic defects in 707 cases with unexplained bleeding and platelet disorders.&rsquo; _Genome Med_, *7*(1), pp. 36.
</p>
<p>Kohler S, Doelken SC, Mungall CJ, Bauer S, Firth HV, Bailleul-Forestier I, Black GC, Brown DL, Brudno M, Campbell J, FitzPatrick DR, Eppig JT, Jackson AP, Freson K, Girdea M, Helbig I, Hurst JA, Jahn J, Jackson LG, Kelly AM, Ledbetter DH, Mansour S, Martin CL, Moss C, Mumford A, Ouwehand WH, Park SM, Riggs ER, Scott RH, Sisodiya S, Van Vooren S, Wapner RJ, Wilkie AO, Wright CF, Vulto-van Silfhout A, de Leeuw N, de Vries B, Washingthon NL, Smith CL, Westerfield M, Schofield P, Ruef BJ, Gkoutos GV, Haendel M, Smedley D, Lewis SE and Robinson PN (2014). &lsquo;The Human Phenotype Ontology project: linking molecular biology and disease through phenotype data.&rsquo; _Nucleic Acids Res._, *42*(Database issue), pp. D966-974.
</p>
<p>Ashburner et al. Gene ontology: tool for the unification of biology (2000) Nat Genet 25(1):25-9
</p>
<p>Smith, Cynthia L., Carroll-Ann W. Goldsmith, and Janan T. Eppig. &lsquo;The Mammalian Phenotype Ontology as a tool for annotating, analyzing and comparing phenotypic information.&rsquo; Genome biology 6.1 (2004): 1.
</p>

<hr>
<h2 id='as.data.frame.ontology_index'>Case <code>ontology_index</code> object to <code>data.frame</code></h2><span id='topic+as.data.frame.ontology_index'></span>

<h3>Description</h3>

<p>Case <code>ontology_index</code> object to <code>data.frame</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ontology_index'
as.data.frame(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.data.frame.ontology_index_+3A_x">x</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="as.data.frame.ontology_index_+3A_...">...</code></td>
<td>
<p>Arguments pass to <code>as.data.frame</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>data.frame</code> containing information about an ontology, rows corresponding to terms and columns to properties.
</p>

<hr>
<h2 id='check'>Check an ontology index object</h2><span id='topic+check'></span>

<h3>Description</h3>

<p>Perform simple consistency checks on <code>ontology_index</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check(ontology, stop_if_invalid = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="check_+3A_stop_if_invalid">stop_if_invalid</code></td>
<td>
<p>Logical value determining whether the function should call <code>stop</code> and print an error message upon finding that the given <code>ontology_index</code> is in valid.</p>
</td></tr>
</table>

<hr>
<h2 id='exclude_descendants'>Exclude terms descending from any in a given set of root terms</h2><span id='topic+exclude_descendants'></span>

<h3>Description</h3>

<p>Exclude from set <code>terms</code>, any terms that are either in, or descend from one of, the set <code>roots</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exclude_descendants(ontology, roots, terms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exclude_descendants_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="exclude_descendants_+3A_roots">roots</code></td>
<td>
<p>Character vector of IDs for terms with respect to which descendants are to be defined.</p>
</td></tr>
<tr><td><code id="exclude_descendants_+3A_terms">terms</code></td>
<td>
<p>Character vector of ontological terms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of terms
</p>


<h3>See Also</h3>

<p><code><a href="#topic+intersection_with_descendants">intersection_with_descendants</a></code>, <code><a href="#topic+prune_descendants">prune_descendants</a></code>
</p>

<hr>
<h2 id='get_ancestors'>Get set of terms containing all ancestors of terms in a given set</h2><span id='topic+get_ancestors'></span>

<h3>Description</h3>

<p>Get set of terms containing all ancestors of terms in a given set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ancestors(ontology, terms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_ancestors_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="get_ancestors_+3A_terms">terms</code></td>
<td>
<p>Character vector of ontological terms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of all terms which are an ancestor of at least one term in <code>terms</code>, including the terms themselves
</p>


<h3>See Also</h3>

<p><code>link{get_descendants}</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hpo)
get_ancestors(hpo, c("HP:0001873", "HP:0011877"))
</code></pre>

<hr>
<h2 id='get_descendants'>Get set of terms containing all descendants of terms in a given set</h2><span id='topic+get_descendants'></span>

<h3>Description</h3>

<p>Get set of terms containing all descendants of terms in a given set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_descendants(ontology, roots, exclude_roots = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_descendants_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="get_descendants_+3A_roots">roots</code></td>
<td>
<p>Character vector of IDs for terms with respect to which descendants are to be defined.</p>
</td></tr>
<tr><td><code id="get_descendants_+3A_exclude_roots">exclude_roots</code></td>
<td>
<p>Boolean determining whether to remove the given <code>roots</code> terms from the result.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of terms
</p>


<h3>See Also</h3>

<p><code>link{get_ancestors}</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hpo)
get_descendants(hpo, roots="HP:0001873")
</code></pre>

<hr>
<h2 id='get_ontology'>Read ontology from OBO file into R</h2><span id='topic+get_ontology'></span><span id='topic+get_OBO'></span>

<h3>Description</h3>

<p>Read ontology from OBO file into R
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_ontology(
  file,
  propagate_relationships = "is_a",
  extract_tags = "minimal",
  merge_equivalent_terms = TRUE
)

get_OBO(
  file,
  propagate_relationships = "is_a",
  extract_tags = "minimal",
  merge_equivalent_terms = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_ontology_+3A_file">file</code></td>
<td>
<p>File path of OBO formatted file.</p>
</td></tr>
<tr><td><code id="get_ontology_+3A_propagate_relationships">propagate_relationships</code></td>
<td>
<p>Character vector of relations</p>
</td></tr>
<tr><td><code id="get_ontology_+3A_extract_tags">extract_tags</code></td>
<td>
<p>Character value: either &quot;minimal&quot; or &quot;everything&quot;, determining whether to extract only the properties of terms which are required to run functions in the package - i.e. <code>"id", "name", "parents", "children"</code> and <code>"ancestors"</code> - or extract all properties provided in the file. Term properties are named in the resulting <code>ontology_index</code> as their corresponding tags in the OBO file (except <code>"parents"</code>, <code>"children"</code> and <code>"ancestors"</code> which are appended with <code>"_OBO"</code> to avoid clashing with standard <code>ontology_index</code> properties. Defaults to <code>"minimal"</code>.</p>
</td></tr>
<tr><td><code id="get_ontology_+3A_merge_equivalent_terms">merge_equivalent_terms</code></td>
<td>
<p>Logical value determining whether terms that are marked <code>"equivalent_to"</code> a target term should be merged, retaining properties of the target term when the property should have one value, e.g. the term ID and name. Defaults to <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>ontology_index</code> object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_relation_names">get_relation_names</a></code>
</p>

<hr>
<h2 id='get_relation_names'>Get names of relations used in OBO file</h2><span id='topic+get_relation_names'></span>

<h3>Description</h3>

<p>Get names of relations used in OBO file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_relation_names(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_relation_names_+3A_file">file</code></td>
<td>
<p>File path of OBO formatted file.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+get_ontology">get_ontology</a></code>
</p>

<hr>
<h2 id='get_term_descendancy_matrix'>Get logical descendancy matrix for set of terms</h2><span id='topic+get_term_descendancy_matrix'></span>

<h3>Description</h3>

<p>Get logical descendancy matrix for set of terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_term_descendancy_matrix(ontology, terms = NULL, rows = terms, cols = terms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_term_descendancy_matrix_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="get_term_descendancy_matrix_+3A_terms">terms</code></td>
<td>
<p>Character vector of ontological terms.</p>
</td></tr>
<tr><td><code id="get_term_descendancy_matrix_+3A_rows">rows</code></td>
<td>
<p>Rows for resultant matrix (defaults to <code>terms</code>).</p>
</td></tr>
<tr><td><code id="get_term_descendancy_matrix_+3A_cols">cols</code></td>
<td>
<p>Cols for resultant matrix (defaults to <code>terms</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical square matrix of with <code>length(terms)</code> columns and rows. <code>result[row_term,col_term] == TRUE</code> if <code>row_term</code> is an ancestor (and not the same as) of <code>col_term</code>.
</p>
<p>A logical matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hpo)
get_term_descendancy_matrix(hpo, c("HP:0001873", "HP:0011877"))
</code></pre>

<hr>
<h2 id='get_term_frequencies'>Get frequency of each term in a set of phenotypes</h2><span id='topic+get_term_frequencies'></span>

<h3>Description</h3>

<p>Get frequency of each term in a set of phenotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_term_frequencies(ontology, term_sets, patch_missing = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_term_frequencies_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="get_term_frequencies_+3A_term_sets">term_sets</code></td>
<td>
<p>List of character vectors of ontological term IDs.</p>
</td></tr>
<tr><td><code id="get_term_frequencies_+3A_patch_missing">patch_missing</code></td>
<td>
<p>Logical indicating whether to include whole ontology even if they're not present in the <code>term_sets</code> as if they had occurred once</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of information contents, named by corresponding terms. Takes into account ancestors, in the sense that all ancestor terms implied by the phenotypes are considered 'on'
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_term_info_content">get_term_info_content</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hpo)
get_term_frequencies(hpo, list("HP:0001873"))
</code></pre>

<hr>
<h2 id='get_term_info_content'>Get information content of each term in a set of phenotypes</h2><span id='topic+get_term_info_content'></span>

<h3>Description</h3>

<p>Get information content of each term in a set of phenotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_term_info_content(ontology, term_sets, patch_missing = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_term_info_content_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="get_term_info_content_+3A_term_sets">term_sets</code></td>
<td>
<p>List of character vectors of ontological term IDs.</p>
</td></tr>
<tr><td><code id="get_term_info_content_+3A_patch_missing">patch_missing</code></td>
<td>
<p>Logical indicating whether to include all ontology terms even if they're not present in the <code>term_sets</code> as if they had occurred once</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of information contents, named by corresponding terms. Takes into account ancestors, in the sense that all ancestor terms implied by the phenotypes are considered 'on'
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hpo)
get_term_info_content(hpo, list("HP:0001873"))
</code></pre>

<hr>
<h2 id='get_term_property'>Get property of individual ontological term</h2><span id='topic+get_term_property'></span>

<h3>Description</h3>

<p>Get property of individual ontological term
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_term_property(ontology, property_name, term, as_names = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_term_property_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="get_term_property_+3A_property_name">property_name</code></td>
<td>
<p>Name of property.</p>
</td></tr>
<tr><td><code id="get_term_property_+3A_term">term</code></td>
<td>
<p>Character value of term ID.</p>
</td></tr>
<tr><td><code id="get_term_property_+3A_as_names">as_names</code></td>
<td>
<p>Logical value determining whether to return character vector of names (defaults to <code>FALSE</code>).</p>
</td></tr>
</table>

<hr>
<h2 id='go'>GO index</h2><span id='topic+go'></span>

<h3>Description</h3>

<p><code>ontology_index</code> object encapsulating structure of the Gene Ontology (HPO) comprising a <code>list</code> of lists/vectors of properties of GO terms indexed by term ID (downloaded from https://current.geneontology.org/ontology/go-basic.obo).
</p>


<h3>Format</h3>

<p>List of lists and vectors
</p>

<hr>
<h2 id='hpo'>HPO index</h2><span id='topic+hpo'></span>

<h3>Description</h3>

<p><code>ontology_index</code> object encapsulating structure of the Human Phenotype Ontology (HPO) comprising a <code>list</code> of lists/vectors of properties of HPO terms indexed by term ID (downloaded from https://hpo.jax.org/app/data/ontology).
</p>


<h3>Format</h3>

<p>List of lists and vectors
</p>

<hr>
<h2 id='intersection_with_descendants'>Intersect a set of terms with the descendants of a given set of roots</h2><span id='topic+intersection_with_descendants'></span>

<h3>Description</h3>

<p>Intersect a set of terms with the descendants of a given set of roots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intersection_with_descendants(ontology, roots, terms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intersection_with_descendants_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="intersection_with_descendants_+3A_roots">roots</code></td>
<td>
<p>Character vector of IDs for terms with respect to which descendants are to be defined.</p>
</td></tr>
<tr><td><code id="intersection_with_descendants_+3A_terms">terms</code></td>
<td>
<p>Character vector of ontological terms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of terms
</p>


<h3>See Also</h3>

<p><code><a href="#topic+exclude_descendants">exclude_descendants</a></code>, <code><a href="#topic+prune_descendants">prune_descendants</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hpo)
intersection_with_descendants(hpo, c("HP:0001872", "HP:0000707"), c("HP:0001873", "HP:0011877"))
</code></pre>

<hr>
<h2 id='minimal_set'>Remove redundant/implied terms from a set of terms</h2><span id='topic+minimal_set'></span>

<h3>Description</h3>

<p>Remove redundant/implied terms from a set of terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minimal_set(ontology, terms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="minimal_set_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="minimal_set_+3A_terms">terms</code></td>
<td>
<p>Character vector of ontological terms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of terms
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hpo)
minimal_set(hpo, c("HP:0001873", "HP:0001872"))
</code></pre>

<hr>
<h2 id='mpo'>MPO index</h2><span id='topic+mpo'></span>

<h3>Description</h3>

<p><code>ontology_index</code> object encapsulating structure of the Mammalian Phenotype Ontology (MPO) comprising a <code>list</code> of lists/vectors of properties of MPO terms indexed by term ID (downloaded from https://www.informatics.jax.org/downloads/reports/MPheno_OBO.ontology).
</p>


<h3>Format</h3>

<p>List of lists and vectors
</p>

<hr>
<h2 id='ontology_index'>Create <code>ontology_index</code> object from vectors and lists of term properties</h2><span id='topic+ontology_index'></span>

<h3>Description</h3>

<p>Create <code>ontology_index</code> object from vectors and lists of term properties
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ontology_index(
  parents,
  id = names(parents),
  name = id,
  obsolete = setNames(nm = id, rep(FALSE, length(id))),
  version = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ontology_index_+3A_parents">parents</code></td>
<td>
<p>List of character vectors of parents per term.</p>
</td></tr>
<tr><td><code id="ontology_index_+3A_id">id</code></td>
<td>
<p>Character vector of term IDs. Defaults to the <code>"names"</code> attribute of the <code>parents</code> argument and must be the same length as <code>parents</code>.</p>
</td></tr>
<tr><td><code id="ontology_index_+3A_name">name</code></td>
<td>
<p>Character vector of term labels.</p>
</td></tr>
<tr><td><code id="ontology_index_+3A_obsolete">obsolete</code></td>
<td>
<p>Logical vector indicating whether given terms are obsolete.</p>
</td></tr>
<tr><td><code id="ontology_index_+3A_version">version</code></td>
<td>
<p>Version information about the ontology.</p>
</td></tr>
<tr><td><code id="ontology_index_+3A_...">...</code></td>
<td>
<p>Additional arguments, each of which should be either a vector or list of term properties, each with the same length as <code>id</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>animal_superclasses &lt;- list(animal=character(0), mammal="animal", cat="mammal", fish="animal")
animal_ontology &lt;- ontology_index(parents=animal_superclasses)
unclass(animal_ontology)
</code></pre>

<hr>
<h2 id='print.ontology_index'>Print <code>ontology_index</code> object</h2><span id='topic+print.ontology_index'></span>

<h3>Description</h3>

<p>Print <code>ontology_index</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ontology_index'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.ontology_index_+3A_x">x</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="print.ontology_index_+3A_...">...</code></td>
<td>
<p>Unused parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Prints a summary
</p>

<hr>
<h2 id='propagate_relations'>Select terms by propagating relations from a set of terms</h2><span id='topic+propagate_relations'></span>

<h3>Description</h3>

<p>An 'ontology_index' can contain multiple relations (for example in the case of the Gene Ontology, <code>"is_a"</code> and <code>"part_of"</code> could be stored as separate properties in an &lsquo;ontology_index'). Transitive relations (i.e. relations such that x related to y and y related to z implies x related to z, for example the relation &rsquo;is an ancestor of') stored by an &lsquo;ontology_index' can be propagated using this function. The &rsquo;inverse relations' (i.e. x inversely related to y if y related to x) can also be propagated by setting the <code>use_inverse_relations</code> parameter to <code>TRUE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>propagate_relations(
  ontology,
  roots,
  relations,
  use_inverse_relations = FALSE,
  exclude_roots = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="propagate_relations_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="propagate_relations_+3A_roots">roots</code></td>
<td>
<p>Character vector of term IDs from which relations will be propagated.</p>
</td></tr>
<tr><td><code id="propagate_relations_+3A_relations">relations</code></td>
<td>
<p>Character vector given names of transitive relations to be propagated.</p>
</td></tr>
<tr><td><code id="propagate_relations_+3A_use_inverse_relations">use_inverse_relations</code></td>
<td>
<p>Boolean vector indicating whether to propagate inverse relations. If <code>use_inverse_relations</code> is the same length as <code>relations</code>, each element determines whether the corresponding relation in <code>relations</code> is inverted.</p>
</td></tr>
<tr><td><code id="propagate_relations_+3A_exclude_roots">exclude_roots</code></td>
<td>
<p>Boolean determining whether to remove the given <code>roots</code> terms from the result.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of terms
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_ancestors">get_ancestors</a></code>, <code><a href="#topic+get_descendants">get_descendants</a></code>
</p>

<hr>
<h2 id='prune_descendants'>Exclude terms descending from a given set of roots but include those roots which were originally implicitly present.</h2><span id='topic+prune_descendants'></span>

<h3>Description</h3>

<p>Given two sets of terms, <code>roots</code> and <code>terms</code>, construct a set of terms containing those in <code>terms</code> which do not descend from any term in <code>roots</code>, and also any terms in <code>roots</code> which are ancestors of any term in <code>terms</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prune_descendants(ontology, roots, terms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prune_descendants_+3A_ontology">ontology</code></td>
<td>
<p><code>ontology_index</code> object.</p>
</td></tr>
<tr><td><code id="prune_descendants_+3A_roots">roots</code></td>
<td>
<p>Character vector of IDs for terms with respect to which descendants are to be defined.</p>
</td></tr>
<tr><td><code id="prune_descendants_+3A_terms">terms</code></td>
<td>
<p>Character vector of ontological terms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of terms
</p>


<h3>See Also</h3>

<p><code><a href="#topic+exclude_descendants">exclude_descendants</a></code>, <code><a href="#topic+intersection_with_descendants">intersection_with_descendants</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
