<!DOCTYPE html><html><head><title>Help for package oompaData</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {oompaData}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#clinical.info'><p>Experimental info for the prostate cancer data set</p></a></li>
<li><a href='#expression.data'><p>Microarray expression data on prostate cancer</p></a></li>
<li><a href='#gene.info'><p>Gene information for the prostate cancer data set</p></a></li>
<li><a href='#lungData-data'><p>Lung Cancer Gene Expression Dataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Data to Illustrate OOMPA Algorithms</td>
</tr>
<tr>
<td>Version:</td>
<td>3.1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-12-02</td>
</tr>
<tr>
<td>Author:</td>
<td>Kevin R. Coombes</td>
</tr>
<tr>
<td>Description:</td>
<td>This is a data-only package to provide example data for
  other packages that are part of the "Object-Oriented Microrray and
  Proteomics Analysis" suite of packages. These are described in more
  detail at the package URL.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kevin R. Coombes &lt;krc@silicovore.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (== 2.0)</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://oompa.r-forge.r-project.org/">http://oompa.r-forge.r-project.org/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-12-02 14:57:21 UTC; kevin</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-02 15:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='clinical.info'>Experimental info for the prostate cancer data set</h2><span id='topic+clinical.info'></span>

<h3>Description</h3>

<p>This data set provides experimental and clinical information about the
(partial) prostate cancer data set included for demonstration purposes
as part of the tail.rank.test package. The experiments were two-color
glass microarrays printed at Stanford.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(clinical.info)</code></pre>


<h3>Format</h3>

<p>A data frame with 112 observations on the following 6 variables.
</p>

<dl>
<dt>Arrays</dt><dd><p>A factor containing the barcode of the microarray on
which the experiment was performed. Each of the 112 entries should
be distinct.</p>
</dd>
<dt>Reference</dt><dd><p>A factor describing the reference sample used in
each experiment. This was a common reference, so the identifiers
here are not meaningful.</p>
</dd>
<dt>Sample</dt><dd><p>A factor identifying the test sample in each
experiment. These match the codes published in the original
paper.</p>
</dd>
<dt>Status</dt><dd><p>A factor with three levels identifying normal prostate
(<code>N</code>), prostate cancer (<code>T</code>), or lymph node metastasis
(<code>L</code>).</p>
</dd>
<dt>Subgroups</dt><dd><p>A factor with five levels: <code>I</code> <code>II</code>
<code>III</code> <code>N</code> <code>O</code>. These correspond to the groups found
in the original paper using clustering.</p>
</dd>
<dt>ChipType</dt><dd><p>a factor with levels <code>new</code> or <code>old</code>. At
least two different print designs of microarrays were used in this
experiment; this factor identifies the design.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The data was originally described in the paper by Lapointe et al., and
downloaded from the Stanford Microarray Database
<a href="https://bio.tools/stanfordmicroarraydb">https://bio.tools/stanfordmicroarraydb</a>.
</p>


<h3>References</h3>

<p>Lapointe J et al. (2004) Gene expression profiling identifies
clinically relevant subtypes of prostate cancer. 
<em>Proc Natl Acad Sci U S A</em>, 101, 811&ndash;816.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+expression.data">expression.data</a></code>,
<code><a href="#topic+gene.info">gene.info</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(clinical.info)
summary(clinical.info) 
</code></pre>

<hr>
<h2 id='expression.data'>Microarray expression data on prostate cancer</h2><span id='topic+expression.data'></span>

<h3>Description</h3>

<p>A subset of the microarray data from a study of prostate
cancer at Stanford is supplied as demo data with the
tail.rank.test package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(expression.data)</code></pre>


<h3>Format</h3>

<p>A data frame with 2000 observations on the 112 variables.
Each column represent a different patient sample, as
described in the accompanying data.frame called
<code><a href="#topic+clinical.info">clinical.info</a></code>.
</p>


<h3>Details</h3>

<p>This data set contains normalized microarray expression
data on 2000 randomly selected genes from a prostate
cancer data set. The study was originially described
in a publication by Lapointe et al. The experiments
were performed on two-color glass microarrays printed
at Stanford and available from the Stanford Microarray
Database.  We downloaded the raw data and preprocessed
it. In particular,after background correction and loess
normalization, we computed log ratios between the channels.
We then randomly selected 2000 of the 42129 spots to
include as demonstration data here.
</p>


<h3>Source</h3>

<p><a href="https://bio.tools/stanfordmicroarraydb">https://bio.tools/stanfordmicroarraydb</a>
</p>


<h3>References</h3>

<p>Lapointe J et al. (2004) Gene expression profiling identifies
clinically relevant subtypes of prostate cancer. 
<em>Proc Natl Acad Sci U S A</em>, 101, 811&ndash;816.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clinical.info">clinical.info</a></code>,
<code><a href="#topic+gene.info">gene.info</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(expression.data)
summary(expression.data)
</code></pre>

<hr>
<h2 id='gene.info'>Gene information for the prostate cancer data set</h2><span id='topic+gene.info'></span>

<h3>Description</h3>

<p>This data set provides information about the genes included with the
(partial) prostate cancer data set as part of the tail.rank.test
package. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene.info)</code></pre>


<h3>Format</h3>

<p>A data frame with 2000 observations on the following 6 variables.
</p>

<dl>
<dt>ArrayI.Spot</dt><dd><p>a numeric vector; where is this clone spotted on
the old arrays</p>
</dd>
<dt>ArrayII.Spot</dt><dd><p>a numeric vector; where is this clone spotted on
the new arrays</p>
</dd>
<dt>Clone.ID</dt><dd><p>a factor; the IMAGE clone identifier</p>
</dd> 
<dt>Gene.Symbol</dt><dd><p>a factor; the official gene symbol</p>
</dd>
<dt>Cluster.ID</dt><dd><p>a factor; the UniGene cluster number</p>
</dd>
<dt>Accession</dt><dd><p>a factor; the GenBanlk accession number</p>
</dd> 
</dl>



<h3>Source</h3>

<p>The data was originally described in the paper by Lapointe et al., and
downloaded from the Stanford Microarray Database
<a href="https://bio.tools/stanfordmicroarraydb">https://bio.tools/stanfordmicroarraydb</a>. We randomly selected 2000 of
the 42129 spots to include as demonstration data here.
</p>


<h3>References</h3>

<p>Lapointe J et al. (2004) Gene expression profiling identifies
clinically relevant subtypes of prostate cancer. 
<em>Proc Natl Acad Sci U S A</em>, 101, 811&ndash;816.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+clinical.info">clinical.info</a></code>,
<code><a href="#topic+expression.data">expression.data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gene.info)
summary(gene.info)
</code></pre>

<hr>
<h2 id='lungData-data'>Lung Cancer Gene Expression Dataset</h2><span id='topic+lungData-data'></span><span id='topic+lungData'></span><span id='topic+lung.clinical'></span><span id='topic+lung.dataset'></span>

<h3>Description</h3>

<p>This data set contains clinical annotations and the log expression of
150 genes for a set of 444 lung cancer patients.  The 150 genes were
selected randomly from a larger Affymetrix U133A dataset. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lungData)
</code></pre>


<h3>Format</h3>

<p>A data matrix (<code>lung.dataset</code>) containing the log
expression of 150 genes (rows) in 444 lung tumor samples (columns),
along with a data frame (lung.clinical) containing clinical
annotations of the patients.</p>


<h3>Source</h3>

<p>Supporting data for the Nature Medicine paper by Shedden et al. was
downloaded from the (now defunct) caArray web site. The original data
are still available by FTP from 
<a href="ftp://caftpd.nci.nih.gov/pub/caARRAY/experiments/caArray_beer-00153/">ftp://caftpd.nci.nih.gov/pub/caARRAY/experiments/caArray_beer-00153/</a>
and in the Gene Expression Omnibus at
<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE68571">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE68571</a>.
The data were log transformed by mapping the expression value x to
log2(1+x). A subset of genes and of clinical annotation columns were
selected to form this data set.
</p>


<h3>References</h3>

<p>Abrams ZB, Zucker M, Wang M, Asiaee Taheri A, Abruzzo LV, Coombes KR.<br />
<em>Thirty biologically interpretable clusters of transcription
factors distinguish cancer type.</em><br />
BMC Genomics. 2018 Oct 11;19(1):738. doi: 10.1186/s12864-018-5093-z.
</p>
<p>Asiaee A, Abrams ZB, Nakayiza S, Sampath D, Coombes KR.<br />
<em>Explaining Gene Expression Using Twenty-One MicroRNAs.</em><br />
J Comput Biol. 2020 Jul;27(7):1157-1170. doi: 10.1089/cmb.2019.0321.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
