<!DOCTYPE html><html lang="en"><head><title>Help for package oosse</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {oosse}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#boot632'><p>The .632 bootstrap estimation of the MSE</p></a></li>
<li><a href='#boot632multiple'><p>Repeated .632 bootstrapa</p></a></li>
<li><a href='#bootOob'><p>The oob bootstrap (smooths leave-one-out CV)</p></a></li>
<li><a href='#Brassica'><p>Gene expression and phenotypes of Brassica napus (rapeseed) plants</p></a></li>
<li><a href='#buildConfInt'><p>Calculate a confidence interval for R², MSE and MST</p></a></li>
<li><a href='#checkFitFun'><p>Check whether supplied prediction function meets the requirements</p></a></li>
<li><a href='#estCorMSEMST'><p>Estimate correlation between MSE and MST estimators</p></a></li>
<li><a href='#estMSE'><p>Estimate MSE and its standard error</p></a></li>
<li><a href='#formatSeconds'><p>Format seconds into human readable format</p></a></li>
<li><a href='#getSEsNested'><p>Calculate standard error on MSE from nested CV results</p></a></li>
<li><a href='#isPD'><p>Helper function to check if matrix is positive definite</p></a></li>
<li><a href='#processOob'><p>Process the out-of-bag bootstraps to get to standard errors following Efron 1997</p></a></li>
<li><a href='#R2oosse'><p>Estimate out-of-sample R² and its standard error</p></a></li>
<li><a href='#RsquaredSE'><p>Calculate out-of-sample R² and its standard error based on MSE estimates</p></a></li>
<li><a href='#simpleCV'><p>Perform simple CV, and return the MSE estimate</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Out-of-Sample R² with Standard Error Estimation</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.11</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Stijn Hawinkel &lt;stijn.hawinkel@psb.ugent.be&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimates out-of-sample R² through bootstrap or cross-validation as a measure of predictive performance. In addition, a standard error for this point estimate is provided, and confidence intervals are constructed.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.0</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, BiocParallel, Matrix, methods, doParallel, Rdpack</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr,rmarkdown,testthat,randomForest,glmnet</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/sthawinke/oosse">https://github.com/sthawinke/oosse</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-07 11:03:01 UTC; sthaw</td>
</tr>
<tr>
<td>Author:</td>
<td>Stijn Hawinkel <a href="https://orcid.org/0000-0002-4501-5180"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-07 11:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='boot632'>The .632 bootstrap estimation of the MSE</h2><span id='topic+boot632'></span>

<h3>Description</h3>

<p>The .632 bootstrap estimation of the MSE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boot632(y, x, id, fitFun, predFun)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="boot632_+3A_y">y</code></td>
<td>
<p>The vector of outcome values</p>
</td></tr>
<tr><td><code id="boot632_+3A_x">x</code></td>
<td>
<p>The matrix of predictors</p>
</td></tr>
<tr><td><code id="boot632_+3A_id">id</code></td>
<td>
<p>the sample indices resampled with replacement</p>
</td></tr>
<tr><td><code id="boot632_+3A_fitfun">fitFun</code></td>
<td>
<p>The function for fitting the prediction model</p>
</td></tr>
<tr><td><code id="boot632_+3A_predfun">predFun</code></td>
<td>
<p>The function for evaluating the prediction model</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The implementation follows (Efron and Tibshirani 1997)
</p>


<h3>Value</h3>

<p>The MSE estimate
</p>


<h3>References</h3>

<p>Efron B, Tibshirani R (1997).
&ldquo;Improvements on cross-validation: The 632+ bootstrap method.&rdquo;
<em>J. Am. Stat. Assoc.</em>, <b>92</b>(438), 548 - 560.
</p>


<h3>See Also</h3>

<p><a href="#topic+estMSE">estMSE</a> <a href="#topic+bootOob">bootOob</a>
</p>

<hr>
<h2 id='boot632multiple'>Repeated .632 bootstrapa</h2><span id='topic+boot632multiple'></span>

<h3>Description</h3>

<p>Repeated .632 bootstrapa
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boot632multiple(nBootstraps, y, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="boot632multiple_+3A_nbootstraps">nBootstraps</code></td>
<td>
<p>The number of .632 bootstraps</p>
</td></tr>
<tr><td><code id="boot632multiple_+3A_y">y</code></td>
<td>
<p>The vector of outcome values</p>
</td></tr>
<tr><td><code id="boot632multiple_+3A_...">...</code></td>
<td>
<p>passed onto boot632</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The estimated MSE
</p>

<hr>
<h2 id='bootOob'>The oob bootstrap (smooths leave-one-out CV)</h2><span id='topic+bootOob'></span>

<h3>Description</h3>

<p>The oob bootstrap (smooths leave-one-out CV)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootOob(y, x, id, fitFun, predFun)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bootOob_+3A_y">y</code></td>
<td>
<p>The vector of outcome values</p>
</td></tr>
<tr><td><code id="bootOob_+3A_x">x</code></td>
<td>
<p>The matrix of predictors</p>
</td></tr>
<tr><td><code id="bootOob_+3A_id">id</code></td>
<td>
<p>sample indices sampled with replacement</p>
</td></tr>
<tr><td><code id="bootOob_+3A_fitfun">fitFun</code></td>
<td>
<p>The function for fitting the prediction model</p>
</td></tr>
<tr><td><code id="bootOob_+3A_predfun">predFun</code></td>
<td>
<p>The function for evaluating the prediction model</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The implementation follows (Efron and Tibshirani 1997)
</p>


<h3>Value</h3>

<p>matrix of errors and inclusion times
</p>


<h3>References</h3>

<p>Efron B, Tibshirani R (1997).
&ldquo;Improvements on cross-validation: The 632+ bootstrap method.&rdquo;
<em>J. Am. Stat. Assoc.</em>, <b>92</b>(438), 548 - 560.
</p>


<h3>See Also</h3>

<p><a href="#topic+estMSE">estMSE</a> <a href="#topic+boot632">boot632</a>
</p>

<hr>
<h2 id='Brassica'>Gene expression and phenotypes of Brassica napus (rapeseed) plants</h2><span id='topic+Brassica'></span>

<h3>Description</h3>

<p>RNA-sequencing data of genetically identical Brassica napus plants in autumn, with 5 phenotypes next spring, as published by De Meyer S, Cruz DF, De Swaef T, Lootens P, Block JD, Bird K, Sprenger H, Van de Voorde M, Hawinkel S, Van Hautegem T, Inzé D, Nelissen H, Roldán-Ruiz I, Maere S (2022).
&ldquo;Predicting yield traits of individual field-grown Brassica napus plants from rosette-stage leaf gene expression.&rdquo;
<em>bioRxiv</em>.
<a href="https://doi.org/10.1101/2022.10.21.513275">doi:10.1101/2022.10.21.513275</a>, https://www.biorxiv.org/content/early/2022/10/23/2022.10.21.513275.full.pdf..
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Brassica
</code></pre>


<h3>Format</h3>

<p>A list with two components Expr and Pheno
</p>

<dl>
<dt>Expr</dt><dd><p>Matrix with Rlog values of 1000 most expressed genes</p>
</dd>
<dt>Pheno</dt><dd><p>Data frame with 5 phenotypes and x and y coordinates of the plants in the field</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://doi.org/10.1101/2022.10.21.513275">doi:10.1101/2022.10.21.513275</a>
</p>


<h3>References</h3>

<p>(De Meyer et al. 2022)
</p>

<hr>
<h2 id='buildConfInt'>Calculate a confidence interval for R², MSE and MST</h2><span id='topic+buildConfInt'></span>

<h3>Description</h3>

<p>Calculate a confidence interval for R², MSE and MST
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildConfInt(oosseObj, what = c("R2", "MSE", "MST"), conf = 0.95)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildConfInt_+3A_oosseobj">oosseObj</code></td>
<td>
<p>The result of the R2oosse call</p>
</td></tr>
<tr><td><code id="buildConfInt_+3A_what">what</code></td>
<td>
<p>For which property should the ci be found: R² (default), MSE or MST</p>
</td></tr>
<tr><td><code id="buildConfInt_+3A_conf">conf</code></td>
<td>
<p>the confidence level required</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The upper bound of the interval is truncated at 1 for the R² and
the lower bound at 0 for the MSE
</p>
<p>The confidence intervals for R² and the MSE are based on standard errors and normal approximations.
The confidence interval for the MST is based on the chi-squared distribution as in equation (16) of (Harding et al. 2014),
but with inflation by a factor (n+1)/n. All quantities are out-of-sample.
</p>


<h3>Value</h3>

<p>A vector of length 2 with lower and upper bound of the confidence interval
</p>


<h3>References</h3>

<p>Harding B, Tremblay C, Cousineau D (2014).
&ldquo;Standard errors: A review and evaluation of standard error estimators using Monte Carlo simulations.&rdquo;
<em>The Quantitative Methods for Psychology</em>, <b>10</b>(2), 107 - 123.
</p>


<h3>See Also</h3>

<p><a href="#topic+R2oosse">R2oosse</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Brassica)
fitFunLM = function(y, x){lm.fit(y = y, x = cbind(1, x))}
predFunLM = function(mod, x) {cbind(1,x) %*% mod$coef}
R2lm = R2oosse(y = Brassica$Pheno$Leaf_8_width, x = Brassica$Expr[, 1:10],
fitFun = fitFunLM, predFun = predFunLM, nFolds = 10)
buildConfInt(R2lm)
buildConfInt(R2lm, what = "MSE")
buildConfInt(R2lm, what = "MST")
</code></pre>

<hr>
<h2 id='checkFitFun'>Check whether supplied prediction function meets the requirements</h2><span id='topic+checkFitFun'></span>

<h3>Description</h3>

<p>Check whether supplied prediction function meets the requirements
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkFitFun(fitFun, reqArgs = c("y", "x"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="checkFitFun_+3A_fitfun">fitFun</code></td>
<td>
<p>The prediction function, or its name as character string</p>
</td></tr>
<tr><td><code id="checkFitFun_+3A_reqargs">reqArgs</code></td>
<td>
<p>The vector of required arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Throws an error when requirements not met, otherwise returns the function
</p>

<hr>
<h2 id='estCorMSEMST'>Estimate correlation between MSE and MST estimators</h2><span id='topic+estCorMSEMST'></span>

<h3>Description</h3>

<p>Estimate correlation between MSE and MST estimators
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estCorMSEMST(
  y,
  x,
  fitFun,
  predFun,
  methodMSE,
  methodCor,
  nBootstrapsCor,
  nFolds,
  nBootstraps
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estCorMSEMST_+3A_y">y</code></td>
<td>
<p>The vector of outcome values</p>
</td></tr>
<tr><td><code id="estCorMSEMST_+3A_x">x</code></td>
<td>
<p>The matrix of predictors</p>
</td></tr>
<tr><td><code id="estCorMSEMST_+3A_fitfun">fitFun</code></td>
<td>
<p>The function for fitting the prediction model</p>
</td></tr>
<tr><td><code id="estCorMSEMST_+3A_predfun">predFun</code></td>
<td>
<p>The function for evaluating the prediction model</p>
</td></tr>
<tr><td><code id="estCorMSEMST_+3A_methodmse">methodMSE</code></td>
<td>
<p>The method to estimate the MSE, either &quot;CV&quot; for cross-validation or &quot;bootstrap&quot; for .632 bootstrap</p>
</td></tr>
<tr><td><code id="estCorMSEMST_+3A_methodcor">methodCor</code></td>
<td>
<p>The method to estimate the correlation between MSE and MST estimators, either &quot;nonparametric&quot; or &quot;jackknife&quot;</p>
</td></tr>
<tr><td><code id="estCorMSEMST_+3A_nbootstrapscor">nBootstrapsCor</code></td>
<td>
<p>The number of bootstraps to estimate the correlation</p>
</td></tr>
<tr><td><code id="estCorMSEMST_+3A_nfolds">nFolds</code></td>
<td>
<p>The number of outer folds for cross-validation</p>
</td></tr>
<tr><td><code id="estCorMSEMST_+3A_nbootstraps">nBootstraps</code></td>
<td>
<p>The number of .632 bootstraps</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the estimated correlation
</p>

<hr>
<h2 id='estMSE'>Estimate MSE and its standard error</h2><span id='topic+estMSE'></span>

<h3>Description</h3>

<p>Estimate MSE and its standard error
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estMSE(
  y,
  x,
  fitFun,
  predFun,
  methodMSE,
  nFolds,
  nInnerFolds,
  cvReps,
  nBootstraps
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estMSE_+3A_y">y</code></td>
<td>
<p>The vector of outcome values</p>
</td></tr>
<tr><td><code id="estMSE_+3A_x">x</code></td>
<td>
<p>The matrix of predictors</p>
</td></tr>
<tr><td><code id="estMSE_+3A_fitfun">fitFun</code></td>
<td>
<p>The function for fitting the prediction model</p>
</td></tr>
<tr><td><code id="estMSE_+3A_predfun">predFun</code></td>
<td>
<p>The function for evaluating the prediction model</p>
</td></tr>
<tr><td><code id="estMSE_+3A_methodmse">methodMSE</code></td>
<td>
<p>The method to estimate the MSE, either &quot;CV&quot; for cross-validation or &quot;bootstrap&quot; for .632 bootstrap</p>
</td></tr>
<tr><td><code id="estMSE_+3A_nfolds">nFolds</code></td>
<td>
<p>The number of outer folds for cross-validation</p>
</td></tr>
<tr><td><code id="estMSE_+3A_ninnerfolds">nInnerFolds</code></td>
<td>
<p>The number of inner cross-validation folds</p>
</td></tr>
<tr><td><code id="estMSE_+3A_cvreps">cvReps</code></td>
<td>
<p>The number of repeats for the cross-validation</p>
</td></tr>
<tr><td><code id="estMSE_+3A_nbootstraps">nBootstraps</code></td>
<td>
<p>The number of .632 bootstraps</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The nested cross-validation scheme follows (Bates et al. 2023),
the .632 bootstrap is implemented as in (Efron and Tibshirani 1997)
</p>


<h3>Value</h3>

<p>A vector with MSE estimate and its standard error
</p>


<h3>References</h3>

<p>Bates S, Hastie T, Tibshirani R (2023).
&ldquo;Cross-validation: What does it estimate and how well does it do it?&rdquo;
<em>J. Am. Stat. Assoc.</em>, <b>118</b>(ja), 1 - 22.
<a href="https://doi.org/10.1080/01621459.2023.2197686">doi:10.1080/01621459.2023.2197686</a>, https://doi.org/10.1080/01621459.2023.2197686.<br /><br /> Efron B, Tibshirani R (1997).
&ldquo;Improvements on cross-validation: The 632+ bootstrap method.&rdquo;
<em>J. Am. Stat. Assoc.</em>, <b>92</b>(438), 548 - 560.
</p>

<hr>
<h2 id='formatSeconds'>Format seconds into human readable format</h2><span id='topic+formatSeconds'></span>

<h3>Description</h3>

<p>Format seconds into human readable format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>formatSeconds(seconds, digits = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="formatSeconds_+3A_seconds">seconds</code></td>
<td>
<p>The number of seconds to be formatted</p>
</td></tr>
<tr><td><code id="formatSeconds_+3A_digits">digits</code></td>
<td>
<p>the number of digits for rounding</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector expressing time in human readable format
</p>

<hr>
<h2 id='getSEsNested'>Calculate standard error on MSE from nested CV results</h2><span id='topic+getSEsNested'></span>

<h3>Description</h3>

<p>Calculate standard error on MSE from nested CV results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSEsNested(cvSplitReps, nOuterFolds, n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getSEsNested_+3A_cvsplitreps">cvSplitReps</code></td>
<td>
<p>The list of outer and inner CV results</p>
</td></tr>
<tr><td><code id="getSEsNested_+3A_nouterfolds">nOuterFolds</code></td>
<td>
<p>The number of outer folds</p>
</td></tr>
<tr><td><code id="getSEsNested_+3A_n">n</code></td>
<td>
<p>The sample size</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The calculation of the standard error of the MSE as proposed by (Bates et al. 2023)
</p>


<h3>Value</h3>

<p>The estimate of the MSE and its standard error
</p>


<h3>References</h3>

<p>Bates S, Hastie T, Tibshirani R (2023).
&ldquo;Cross-validation: What does it estimate and how well does it do it?&rdquo;
<em>J. Am. Stat. Assoc.</em>, <b>118</b>(ja), 1 - 22.
<a href="https://doi.org/10.1080/01621459.2023.2197686">doi:10.1080/01621459.2023.2197686</a>, https://doi.org/10.1080/01621459.2023.2197686.
</p>


<h3>See Also</h3>

<p><a href="#topic+estMSE">estMSE</a>
</p>

<hr>
<h2 id='isPD'>Helper function to check if matrix is positive definite</h2><span id='topic+isPD'></span>

<h3>Description</h3>

<p>Helper function to check if matrix is positive definite
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isPD(mat, tol = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isPD_+3A_mat">mat</code></td>
<td>
<p>The matrix</p>
</td></tr>
<tr><td><code id="isPD_+3A_tol">tol</code></td>
<td>
<p>The tolerance</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A boolean indicating positive definiteness
</p>

<hr>
<h2 id='processOob'>Process the out-of-bag bootstraps to get to standard errors following Efron 1997</h2><span id='topic+processOob'></span>

<h3>Description</h3>

<p>Process the out-of-bag bootstraps to get to standard errors following Efron 1997
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processOob(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="processOob_+3A_x">x</code></td>
<td>
<p>the list with out=of=bag bootstrap results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>out-of-bag MSE estimate and standard error
</p>

<hr>
<h2 id='R2oosse'>Estimate out-of-sample R² and its standard error</h2><span id='topic+R2oosse'></span>

<h3>Description</h3>

<p>Estimate out-of-sample R² and its standard error
</p>


<h3>Usage</h3>

<pre><code class='language-R'>R2oosse(
  y,
  x,
  fitFun,
  predFun,
  methodMSE = c("CV", "bootstrap"),
  methodCor = c("nonparametric", "jackknife"),
  printTimeEstimate = TRUE,
  nFolds = 10L,
  nInnerFolds = nFolds - 1L,
  cvReps = 200L,
  nBootstraps = 200L,
  nBootstrapsCor = 50L,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="R2oosse_+3A_y">y</code></td>
<td>
<p>The vector of outcome values</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_x">x</code></td>
<td>
<p>The matrix of predictors</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_fitfun">fitFun</code></td>
<td>
<p>The function for fitting the prediction model</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_predfun">predFun</code></td>
<td>
<p>The function for evaluating the prediction model</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_methodmse">methodMSE</code></td>
<td>
<p>The method to estimate the MSE, either &quot;CV&quot; for cross-validation or &quot;bootstrap&quot; for .632 bootstrap</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_methodcor">methodCor</code></td>
<td>
<p>The method to estimate the correlation between MSE and MST estimators, either &quot;nonparametric&quot; or &quot;jackknife&quot;</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_printtimeestimate">printTimeEstimate</code></td>
<td>
<p>A boolean, should an estimate of the running time be printed?</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_nfolds">nFolds</code></td>
<td>
<p>The number of outer folds for cross-validation</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_ninnerfolds">nInnerFolds</code></td>
<td>
<p>The number of inner cross-validation folds</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_cvreps">cvReps</code></td>
<td>
<p>The number of repeats for the cross-validation</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_nbootstraps">nBootstraps</code></td>
<td>
<p>The number of .632 bootstraps</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_nbootstrapscor">nBootstrapsCor</code></td>
<td>
<p>The number of bootstraps to estimate the correlation</p>
</td></tr>
<tr><td><code id="R2oosse_+3A_...">...</code></td>
<td>
<p>passed onto fitFun and predFun</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Implements the calculation of the R² and its standard error by (Hawinkel et al. 2023).
Multithreading is used as provided by the BiocParallel or doParallel packages,
A rough estimate of expected computation time is printed when printTimeEstimate is true, but this is purely indicative.
The options to estimate the mean squared error (MSE) are cross-validation (Bates et al. 2023) or the .632 bootstrap (Efron and Tibshirani 1997).
</p>


<h3>Value</h3>

<p>A list with components
</p>
<table role = "presentation">
<tr><td><code>R2</code></td>
<td>
<p>Estimate of the R² with standard error</p>
</td></tr>
<tr><td><code>MSE</code></td>
<td>
<p>Estimate of the MSE with standard error</p>
</td></tr>
<tr><td><code>MST</code></td>
<td>
<p>Estimate of the MST with standard error</p>
</td></tr>
<tr><td><code>corMSEMST</code></td>
<td>
<p>Estimated correlation between MSE and MST estimators</p>
</td></tr>
<tr><td><code>params</code></td>
<td>
<p>List of parameters used</p>
</td></tr>
<tr><td><code>fullModel</code></td>
<td>
<p>The model trained on the entire dataset using fitFun</p>
</td></tr>
<tr><td><code>n</code></td>
<td>
<p>The sample size of the training data</p>
</td></tr>
</table>


<h3>References</h3>

<p>Bates S, Hastie T, Tibshirani R (2023).
&ldquo;Cross-validation: What does it estimate and how well does it do it?&rdquo;
<em>J. Am. Stat. Assoc.</em>, <b>118</b>(ja), 1 - 22.
<a href="https://doi.org/10.1080/01621459.2023.2197686">doi:10.1080/01621459.2023.2197686</a>, https://doi.org/10.1080/01621459.2023.2197686.<br /><br /> Efron B, Tibshirani R (1997).
&ldquo;Improvements on cross-validation: The 632+ bootstrap method.&rdquo;
<em>J. Am. Stat. Assoc.</em>, <b>92</b>(438), 548 - 560.<br /><br /> Hawinkel S, Waegeman W, Maere S (2023).
&ldquo;Out-of-sample R2: Estimation and inference.&rdquo;
<em>Am. Stat.</em>, 1 - 16.
<a href="https://doi.org/10.1080/00031305.2023.2216252">doi:10.1080/00031305.2023.2216252</a>, https://doi.org/10.1080/00031305.2023.2216252.
</p>


<h3>See Also</h3>

<p><a href="#topic+buildConfInt">buildConfInt</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Brassica)
#Linear model
fitFunLM = function(y, x){lm.fit(y = y, x = cbind(1, x))}
predFunLM = function(mod, x) {cbind(1,x) %*% mod$coef}
y = Brassica$Pheno$Leaf_8_width
R2lm = R2oosse(y = Brassica$Pheno$Leaf_8_width, x = Brassica$Expr[, 1:10],
fitFun = fitFunLM, predFun = predFunLM, nFolds = 10)
</code></pre>

<hr>
<h2 id='RsquaredSE'>Calculate out-of-sample R² and its standard error based on MSE estimates</h2><span id='topic+RsquaredSE'></span>

<h3>Description</h3>

<p>Calculate out-of-sample R² and its standard error based on MSE estimates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RsquaredSE(MSE, margVar, SEMSE, n, corMSEMST)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RsquaredSE_+3A_mse">MSE</code></td>
<td>
<p>An estimate of the mean squared error (MSE)</p>
</td></tr>
<tr><td><code id="RsquaredSE_+3A_margvar">margVar</code></td>
<td>
<p>The marginal variance of the outcome, not scaled by (n+1)/n</p>
</td></tr>
<tr><td><code id="RsquaredSE_+3A_semse">SEMSE</code></td>
<td>
<p>The standard error on the MSE estimate</p>
</td></tr>
<tr><td><code id="RsquaredSE_+3A_n">n</code></td>
<td>
<p>the sample size of the training data</p>
</td></tr>
<tr><td><code id="RsquaredSE_+3A_cormsemst">corMSEMST</code></td>
<td>
<p>The correlation between MSE and marginal variance estimates</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is exported to allow the user to estimate the MSE and its standard error
and the correlation between MSE and MST estimators himself.
The marginal variance is scaled by (n+1)/n to the out-of-sample MST, so the user does not need to do this.
</p>


<h3>Value</h3>

<p>A vector with the R² and standard error estimates
</p>


<h3>References</h3>

<p>Hawinkel S, Waegeman W, Maere S (2023).
&ldquo;Out-of-sample R2: Estimation and inference.&rdquo;
<em>Am. Stat.</em>, 1 - 16.
<a href="https://doi.org/10.1080/00031305.2023.2216252">doi:10.1080/00031305.2023.2216252</a>, https://doi.org/10.1080/00031305.2023.2216252.
</p>


<h3>See Also</h3>

<p><a href="#topic+R2oosse">R2oosse</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The out-of-sample R² calculated using externally provided estimates
RsquaredSE(MSE = 3, margVar = 4, SEMSE = 0.4, n = 50, corMSEMST = 0.75)
</code></pre>

<hr>
<h2 id='simpleCV'>Perform simple CV, and return the MSE estimate</h2><span id='topic+simpleCV'></span>

<h3>Description</h3>

<p>Perform simple CV, and return the MSE estimate
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simpleCV(y, x, fitFun, predFun, nFolds)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simpleCV_+3A_y">y</code></td>
<td>
<p>The vector of outcome values</p>
</td></tr>
<tr><td><code id="simpleCV_+3A_x">x</code></td>
<td>
<p>The matrix of predictors</p>
</td></tr>
<tr><td><code id="simpleCV_+3A_fitfun">fitFun</code></td>
<td>
<p>The function for fitting the prediction model</p>
</td></tr>
<tr><td><code id="simpleCV_+3A_predfun">predFun</code></td>
<td>
<p>The function for evaluating the prediction model</p>
</td></tr>
<tr><td><code id="simpleCV_+3A_nfolds">nFolds</code></td>
<td>
<p>The number of outer folds for cross-validation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The MSE estimate
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
