<!DOCTYPE html><html lang="en"><head><title>Help for package OpEnHiMR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {OpEnHiMR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#pred_hmc'><p>Prediction of Histone Modification for Multiclass Data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Optimization Based Ensemble Model for Prediction of Histone
Modifications in Rice</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>The comprehensive knowledge of epigenetic modifications in plants, encompassing histone modifications in regulating gene expression, is not completely ingrained. It is noteworthy that histone deacetylation and histone H3 lysine 27 trimethylation (H3K27me3) play a role in repressing transcription in eukaryotes. In contrast, histone acetylation (H3K9ac) and H3K4me3 have been inevitably linked to the stimulation of gene expression, which significantly influences plant development and plays a role in plant responses to biotic and abiotic stresses. To our knowledge this the first multiclass classifier for predicting histone modification in plants. &lt;<a href="https://doi.org/10.1186%2Fs12864-019-5489-4">doi:10.1186/s12864-019-5489-4</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings, devtools, tidyverse, seqinr, splitstackshape,
entropy, party, e1071, caret, randomForest, gbm, stats,
stringr, ftrCOOL, dplyr, RCurl</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-29 18:22:38 UTC; dipro</td>
</tr>
<tr>
<td>Author:</td>
<td>Dipro Sinha [aut, cre],
  Sneha Murmu [aut],
  Girish Kumar Jha [aut],
  Md Yeasin [aut],
  Saikath Das [aut],
  Sougata Bhattacharjee [aut],
  Dwijesh Chandra Mishra [aut],
  Neeraj Budhlakoti [aut],
  Sudhir Srivastava [aut],
  Sunil Archak [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dipro Sinha &lt;diprosinha@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-30 07:00:34 UTC</td>
</tr>
</table>
<hr>
<h2 id='pred_hmc'>Prediction of Histone Modification for Multiclass Data</h2><span id='topic+pred_hmc'></span>

<h3>Description</h3>

<p>Prediction of H3K27me3, H3K9ac and H3K4me3 modification in rice.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pred_hmc(fasta_file_path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pred_hmc_+3A_fasta_file_path">fasta_file_path</code></td>
<td>
<p>Sequence file path (.fasta format)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Modications: sequences with their modifications (H3K27me3, H3K9ac and H3K4me3) or no modification.
</p>


<h3>References</h3>

<p>Yin, Q., Wu, M., Liu, Q. et al. DeepHistone: a deep learning approach to predicting histone modifications. BMC Genomics 20 (Suppl 2), 193 (2019).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
example_path &lt;- system.file("exdata/test.fasta", package = "OpEnHiMR")
pred &lt;- pred_hmc(fasta_file_path = example_path)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
