<!DOCTYPE html><html><head><title>Help for package optBiomarker</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {optBiomarker}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#classificationError'><p>Estimation of misclassification errors (generalisation errors)</p>
based on statistical and various machine learning methods</a></li>
<li><a href='#errorDbase'>
<p>Database of leave-one-out cross validation errors for various</p>
combinations of data characteristics</a></li>
<li><a href='#internal'><p>Internal functions for optBiomarker package.</p></a></li>
<li><a href='#optBiomarker-package'>
<p>R package for estimating optimal number of biomarkers at a given error tolerance level</p>
for various classification rules</a></li>
<li><a href='#optimiseBiomarker'>
<p>Estimates optimal number of biomarkers at a given error tolerance level</p>
for various classification rules</a></li>
<li><a href='#realBiomarker'>
<p>A set of 54359 median gene expressions in log (base 2) scale</p></a></li>
<li><a href='#simData'><p>Simulation of microarray data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimation of Optimal Number of Biomarkers for Two-Group
Microarray Based Classifications at a Given Error Tolerance
Level for Various Classification Rules</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0-28</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-12-23</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), rpanel</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, randomForest, e1071,ipred, msm, rgl, Matrix</td>
</tr>
<tr>
<td>Author:</td>
<td>Mizanur Khondoker &lt;mizanur.khondoker@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mizanur Khondoker &lt;mizanur.khondoker@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimates optimal number of biomarkers for two-group
        classification based on microarray data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-18 20:13:49 UTC; mizan</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-18 23:50:02 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
</table>
<hr>
<h2 id='classificationError'>Estimation of misclassification errors (generalisation errors)
based on statistical and various machine learning methods</h2><span id='topic+classificationError'></span><span id='topic+classificationError.data.frame'></span><span id='topic+classificationError.default'></span>

<h3>Description</h3>

<p>Estimates misclassification errors (generalisation errors), sensitivity and specificity using cross-validation, 
bootstrap and <code>632plus</code> bias corrected bootstrap methods based on Random Forest, 
Support Vector Machines, Linear Discriminant Analysis and k-Nearest Neighbour methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'data.frame'
classificationError(
          formula,
          data, 
          method=c("RF","SVM","LDA","KNN"), 
          errorType = c("cv", "boot", "six32plus"),
	  senSpec=TRUE,
          negLevLowest=TRUE,
	  na.action=na.omit, 
          control=control.errorest(k=NROW(na.action(data)),nboot=100),
          ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="classificationError_+3A_formula">formula</code></td>
<td>
<p>A formula of the form <code>lhs ~ rhs</code> relating response (class) 
variable and the explanatory variables. See <code><a href="stats.html#topic+lm">lm</a></code> for 
more detail.</p>
</td></tr>
<tr><td><code id="classificationError_+3A_data">data</code></td>
<td>
<p>A data frame containing the response (class membership) variable and 
the explanatory variables in the formula.</p>
</td></tr>
<tr><td><code id="classificationError_+3A_method">method</code></td>
<td>
<p>A character vector of length <code>1</code> to <code>4</code> representing the classification 
methods to be used. Can be one or more of <code>"RF"</code> (Random Forest), <code>"SVM"</code> 
(Support Vector Machines), <code>"LDA"</code> (Linear Discriminant Analysis) and <code>"KNN"</code> 
(k-Nearest Neighbour). Defaults to all four methods.</p>
</td></tr>
<tr><td><code id="classificationError_+3A_errortype">errorType</code></td>
<td>
<p>A character vector of length <code>1</code> to <code>3</code> representing the type of 
estimators  to be used for computing misclassification errors. 
Can be one or more of the <code>"cv"</code> (cross-validation), <code>"boot"</code> 
(bootstrap) and <code>"632plus"</code> (632plus bias corrected bootstrap) estimators.
Defaults to all three estimators.</p>
</td></tr>
<tr><td><code id="classificationError_+3A_senspec">senSpec</code></td>
<td>
<p>Logical. Should sensitivity and specificity (for cross-validation estimator only)
be computed? Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="classificationError_+3A_neglevlowest">negLevLowest</code></td>
<td>
<p>Logical. Is the lowest of the ordered levels of the class variable represnts
the negative control? Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="classificationError_+3A_na.action">na.action</code></td>
<td>
<p>Function which indicates what should happen when the data
contains <code>NA</code>'s, defaults to <code><a href="stats.html#topic+na.omit">na.omit</a></code>.</p>
</td></tr> 
<tr><td><code id="classificationError_+3A_control">control</code></td>
<td>
<p>Control parameters of the the function <code><a href="ipred.html#topic+errorest">errorest</a></code>.</p>
</td></tr>
<tr><td><code id="classificationError_+3A_...">...</code></td>
<td>
<p>additional parameters to <code>method</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the current version of the package, estimation of sensitivity and
specificity is limited to cross-validation estimator only. For LDA
sample size must be greater than the number of explanatory variables to
avoid singularity. The function <code>classificationError</code> does not
check if this is satisfied, but the underlying function
<code><a href="MASS.html#topic+lda">lda</a></code> produces warnings if this condition is violated.
</p>


<h3>Value</h3>

<p>Returns an object of class <code>classificationError</code> with components
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>The call of the <code>classificationError</code> function.</p>
</td></tr>
<tr><td><code>errorRate</code></td>
<td>
<p>A <code>length(errorType)</code> by <code>length(method)</code>
matrix  of classification errors.</p>
</td></tr>
<tr><td><code>rocData</code></td>
<td>
<p>A <code>2</code> by <code>length(method)</code>  matrix  of
sensitivities (first row) and specificities (second row).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mizanur Khondoker, Till Bachmann, Peter Ghazal <br />
Maintainer: Mizanur Khondoker <a href="mailto:mizanur.khondoker@gmail.com">mizanur.khondoker@gmail.com</a>.</p>


<h3>References</h3>

<p>Khondoker, M. R., Till T. Bachmann, T. T., Mewissen, M., Dickinson, P. <em>et al.</em>(2010).
Multi-factorial analysis of class prediction error: estimating optimal number of biomarkers for various classification rules.
<em>Journal of Bioinformatics and Computational Biology</em>, <b>8</b>, 945-965.
</p>
<p>Breiman, L. (2001). <em>Random Forests</em>, Machine Learning
<b>45</b>(1), 5&ndash;32.
</p>
<p>Chang, Chih-Chung and Lin, Chih-Jen: <em>LIBSVM: a library for Support
Vector Machines</em>, <a href="https://www.csie.ntu.edu.tw/~cjlin/libsvm/">https://www.csie.ntu.edu.tw/~cjlin/libsvm/</a>. 
</p>
<p>Ripley, B. D. (1996). <em>Pattern Recognition and Neural
Networks</em>.Cambridge: Cambridge University Press.
</p>
<p>Efron, B.  and Tibshirani, R. (1997). Improvements on Cross-Validation:
The .632+ Bootstrap Estimator. <em>Journal of the American Statistical
Association</em> <b>92</b>(438), 548&ndash;560.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simData">simData</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
mydata&lt;-simData(nTrain=30,nBiom=3)$data
classificationError(formula=class~., data=mydata)

## End(Not run)

</code></pre>

<hr>
<h2 id='errorDbase'>
Database of leave-one-out cross validation errors for various
combinations of data characteristics
</h2><span id='topic+errorDbase'></span>

<h3>Description</h3>

<p>This is a 7-dimensional array (database) of  leave-one-out cross
validation errors for Random Forest, Support Vector Machines, Linear
Discriminant Analysis and k-Nearest Neighbour classifiers. The
database is the basis for estimating the optimal number of biomarkers
at a given error tolerance level using <code><a href="#topic+optimiseBiomarker">optimiseBiomarker</a></code>
function. See <b>Details</b> for more information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(errorDbase)</code></pre>


<h3>Format</h3>

<p>7-dimensional numeric array.</p>


<h3>Details</h3>

<p>The following table gives the dimension names, lengths and values/levels
of the data object <code>errorDbase</code>.
</p>

<table>
<tr>
 <td style="text-align: left;">
      <b>Dimension name</b> </td><td style="text-align: left;"> <b>Length</b> </td><td style="text-align: left;"> <b>Values/Levels</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
      No. of biomarkers     </td><td style="text-align: left;"> 14</td><td style="text-align: left;"> (1-6, 7, 9, 11, 15, 20, 30, 40,
      50, 100) </td>
</tr>
<tr>
 <td style="text-align: left;">
      Size of replication   </td><td style="text-align: left;"> 5 </td><td style="text-align: left;"> (1, 3, 5, 7, 10) </td>
</tr>
<tr>
 <td style="text-align: left;">
      Biological variation (<code class="reqn">\sigma_b</code>)</td><td style="text-align: left;"> 4</td><td style="text-align: left;"> (0.5, 1.0, 1.5,
      2.5) </td>
</tr>
<tr>
 <td style="text-align: left;">
      Experimental variation (<code class="reqn">\sigma_e</code>) </td><td style="text-align: left;"> 4 </td><td style="text-align: left;"> (0.1, 0.5,
      1.0, 1.5) </td>
</tr>
<tr>
 <td style="text-align: left;">
      Minimum (Average) fold change </td><td style="text-align: left;"> 4 </td><td style="text-align: left;"> (1 (1.73), 2(2.88), 3(4.03),
      5(6.33))</td>
</tr>
<tr>
 <td style="text-align: left;">
      Training set size   </td><td style="text-align: left;"> 5 </td><td style="text-align: left;"> (10, 20, 50, 100, 250) </td>
</tr>
<tr>
 <td style="text-align: left;">
      Classification method </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> (Random Forest, Support Vector
      Machine, k-Nearest Neighbour)</td>
</tr>
<tr>
 <td style="text-align: left;">
      </td>
</tr>

</table>

<p>We have a plan  to expand the database to a 8-dimensional one by
adding  another dimension to store error rates at different level
of correlation between  biomarkers. Length of each dimension will
also be increased leading to a bigger database with a wider coverage
of the parameter space. Current version of the database contain error rates
for independent (correlation = 0) biomarkers only. Also, it does not
contain error rates for Linear Discriminant Analysis, which we plan
to implement in the next release of the package. With the current
version of the database, optimal number of biomarkers can be
estimated using the  <code><a href="#topic+optimiseBiomarker">optimiseBiomarker</a></code>
function for any intermediate values of the factors represented by
the dimensions of the database. 
</p>


<h3>Author(s)</h3>

<p>Mizanur Khondoker, Till Bachmann, Peter Ghazal <br />
Maintainer: Mizanur Khondoker <a href="mailto:mizanur.khondoker@gmail.com">mizanur.khondoker@gmail.com</a>.</p>


<h3>References</h3>

<p>Khondoker, M. R., Till T. Bachmann, T. T., Mewissen, M., Dickinson, P. <em>et al.</em>(2010).
Multi-factorial analysis of class prediction error: estimating optimal number of biomarkers for various classification rules.
<em>Journal of Bioinformatics and Computational Biology</em>, <b>8</b>, 945-965.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+optimiseBiomarker">optimiseBiomarker</a></code>
</p>

<hr>
<h2 id='internal'>Internal functions for optBiomarker package.</h2><span id='topic+tnormAvg'></span><span id='topic+objFun'></span><span id='topic+avg2Lower'></span><span id='topic+plot3dFun'></span><span id='topic+dimSelect'></span><span id='topic+yapprox'></span><span id='topic+matapprox'></span><span id='topic+classpredict.lda'></span><span id='topic+classpredict.knn'></span>

<h3>Description</h3>

<p>Internal functions for <code>optBiomarker</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tnormAvg(a1=-Inf, a2=Inf,mu=0,sigma=1)
avg2Lower(truncAvg=log2(2), mu=0,sigma=1)
objFun(para, mu=0,sigma=1,truncAvg=log2(2))
plot3dFun(panel,...)
dimSelect(x, val)
yapprox(y, x, xout, method="linear", n=50,
          yleft, yright, rule = 1, f = 0, ties = mean)
matapprox(mat, rx, cx, nr=100,nc=20,rout, cout)

classpredict.lda(object, newdata)
classpredict.knn(object, newdata) 

</code></pre>


<h3>Details</h3>

<p>These functions are not to be called by the user.
</p>

<hr>
<h2 id='optBiomarker-package'>
R package for estimating optimal number of biomarkers at a given error tolerance level
for various classification rules
</h2><span id='topic+optBiomarker-package'></span><span id='topic+optBiomarker'></span>

<h3>Description</h3>

<p>Using interactive control panel  (<code><a href="rpanel.html#topic+rpanel.package">rpanel</a></code>)  and 3D
real-time rendering system  (<code><a href="rgl.html#topic+rgl-package">rgl</a></code>) , this package provides a
user friendly GUI for estimating the minimum  number of biomarkers
(variables) needed to achieve  a given level of accuracy for two-group
classification problems based on microarray data. 
</p>


<h3>Details</h3>

<p>The function <code><a href="#topic+optimiseBiomarker">optimiseBiomarker</a></code> is a user friendly GUI for
interrogating the database of leave-one-out cross-validation errors, 
<code><a href="#topic+errorDbase">errorDbase</a></code>, to estimate optimal number of biomarkers for
microarray based classifications. The database is built on the basis of
simulated data using the <code><a href="#topic+classificationError">classificationError</a></code> function. The
function <code><a href="#topic+simData">simData</a></code> is used for simulating microarray data
for various combinations of factors such as the number of biomarkers,
training set size, biological variation, experimental variation, fold
change, replication, and correlation.
</p>


<h3>Author(s)</h3>

<p>Mizanur Khondoker, Till Bachmann, Peter Ghazal <br />
</p>
<p>Maintainer: Mizanur Khondoker <a href="mailto:mizanur.khondoker@gmail.com">mizanur.khondoker@gmail.com</a>.
</p>


<h3>References</h3>

<p>Khondoker, M. R., Till T. Bachmann, T. T., Mewissen, M., Dickinson, P. <em>et al.</em>(2010).
Multi-factorial analysis of class prediction error: estimating optimal number of biomarkers for various classification rules.
<em>Journal of Bioinformatics and Computational Biology</em>, <b>8</b>, 945-965.
</p>
<p>Breiman, L. (2001). <em>Random Forests</em>, Machine Learning
<b>45</b>(1), 5&ndash;32.
</p>
<p>Chang, Chih-Chung and Lin, Chih-Jen: <em>LIBSVM: a library for Support
Vector Machines</em>, <a href="https://www.csie.ntu.edu.tw/~cjlin/libsvm/">https://www.csie.ntu.edu.tw/~cjlin/libsvm/</a>. 
</p>
<p>Ripley, B. D. (1996). <em>Pattern Recognition and Neural
Networks</em>.Cambridge: Cambridge University Press.
</p>
<p>Efron, B.  and Tibshirani, R. (1997). Improvements on Cross-Validation:
The .632+ Bootstrap Estimator. <em>Journal of the American Statistical
Association</em> <b>92</b>(438), 548&ndash;560.
</p>
<p>Bowman, A., Crawford, E., Alexander, G. and Bowman,
R. W. (2007). rpanel: Simple interactive controls for R functions using
the tcltk package. <em>Journal of Statistical Software</em> <b>17</b>(9). 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simData">simData</a></code>
<code><a href="#topic+classificationError">classificationError</a></code>
<code><a href="#topic+optimiseBiomarker">optimiseBiomarker</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()){
data(errorDbase)
optimiseBiomarker(error=errorDbase)
}
</code></pre>

<hr>
<h2 id='optimiseBiomarker'>
Estimates optimal number of biomarkers at a given error tolerance level
for various classification rules
</h2><span id='topic+optimiseBiomarker'></span>

<h3>Description</h3>

<p>Using interactive control panel (see <code><a href="rpanel.html#topic+rpanel.package">rpanel</a></code>) and 3D
real-time rendering system (<code><a href="rgl.html#topic+rgl-package">rgl</a></code>), this package provides a
user friendly GUI for estimating the minimum  number of biomarkers
(variables) needed to achieve  a given level of accuracy for two-group
classification problems  based on microarray data. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optimiseBiomarker (error,
                   errorTol = 0.05,
                   method = "RF", nTrain = 100, 
                   sdB = 1.5,
                   sdW = 1,
                   foldAvg = 2.88,
                   nRep = 3) 

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="optimiseBiomarker_+3A_error">error</code></td>
<td>
<p>The database of classification errors. See
<code>errorDbase</code> for details.</p>
</td></tr>
<tr><td><code id="optimiseBiomarker_+3A_errortol">errorTol</code></td>
<td>
<p>Error tolerance limit.</p>
</td></tr>
<tr><td><code id="optimiseBiomarker_+3A_method">method</code></td>
<td>
<p>Classification method. Can be one of <code>"RF"</code>,
<code>"SVM"</code>, and <code>"KNN"</code> for Random Forest,
Support Vector Machines, Linear Discriminant Analysis and k-Nearest
Neighbour respectively.</p>
</td></tr>
<tr><td><code id="optimiseBiomarker_+3A_ntrain">nTrain</code></td>
<td>
<p>Training set size, i.e., the total number of biological
samples in group 1 and group 2.</p>
</td></tr>
<tr><td><code id="optimiseBiomarker_+3A_sdb">sdB</code></td>
<td>
<p>Biological variation (<code class="reqn">\sigma_b</code>) of  data in
log (base 2) scale.</p>
</td></tr>
<tr><td><code id="optimiseBiomarker_+3A_sdw">sdW</code></td>
<td>
<p>Experimental (technical) variation (<code class="reqn">\sigma_e</code>) of  data in
log (base 2) scale.</p>
</td></tr>
<tr><td><code id="optimiseBiomarker_+3A_foldavg">foldAvg</code></td>
<td>
<p>Average fold change of the biomarkers.</p>
</td></tr>
<tr><td><code id="optimiseBiomarker_+3A_nrep">nRep</code></td>
<td>
<p>Number of technical replications.</p>
</td></tr>

</table>


<h3>Details</h3>

<p>The function <code><a href="#topic+optimiseBiomarker">optimiseBiomarker</a></code> is a user friendly GUI for
interrogating the database of leave-one-out cross-validation errors,
<code><a href="#topic+errorDbase">errorDbase</a></code>, to estimate optimal number of biomarkers for
microarray based classifications. The database is built on the basis of
simulated data using the <code><a href="#topic+classificationError">classificationError</a></code> function. The
function <code><a href="#topic+simData">simData</a></code> is used for simulating microarray data
for various combinations of factors such as the number of biomarkers,
training set size, biological variation, experimental variation, fold
change, replication, and correlation.
</p>


<h3>Author(s)</h3>

<p>Mizanur Khondoker, Till Bachmann, Peter Ghazal <br />
</p>
<p>Maintainer: Mizanur Khondoker <a href="mailto:mizanur.khondoker@gmail.com">mizanur.khondoker@gmail.com</a>.
</p>


<h3>References</h3>

<p>Khondoker, M. R., Till T. Bachmann, T. T., Mewissen, M., Dickinson, P. <em>et al.</em>(2010).
Multi-factorial analysis of class prediction error: estimating optimal number of biomarkers for various classification rules.
<em>Journal of Bioinformatics and Computational Biology</em>, <b>8</b>, 945-965.
</p>
<p>Breiman, L. (2001). <em>Random Forests</em>, Machine Learning
<b>45</b>(1), 5&ndash;32.
</p>
<p>Chang, Chih-Chung and Lin, Chih-Jen: <em>LIBSVM: a library for Support
Vector Machines</em>, <a href="https://www.csie.ntu.edu.tw/~cjlin/libsvm/">https://www.csie.ntu.edu.tw/~cjlin/libsvm/</a>. 
</p>
<p>Ripley, B. D. (1996). <em>Pattern Recognition and Neural
Networks</em>.Cambridge: Cambridge University Press.
</p>
<p>Efron, B.  and Tibshirani, R. (1997). Improvements on Cross-Validation:
The .632+ Bootstrap Estimator. <em>Journal of the American Statistical
Association</em> <b>92</b>(438), 548&ndash;560.
</p>
<p>Bowman, A., Crawford, E., Alexander, G. and Bowman,
R. W. (2007). rpanel: Simple interactive controls for R functions using
the tcltk package. <em>Journal of Statistical Software</em> <b>17</b>(9). 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simData">simData</a></code>
<code><a href="#topic+classificationError">classificationError</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()){
data(errorDbase)
optimiseBiomarker(error=errorDbase)
}
</code></pre>

<hr>
<h2 id='realBiomarker'>
A set of 54359 median gene expressions in log (base 2) scale 
</h2><span id='topic+realBiomarker'></span>

<h3>Description</h3>

<p>This data set contains a set of 54359 log base 2 gene expression values from  a neonatal 
whole blood gene expression study described in Smith <em>et al.</em> (2007). The data represent
the median of 28 microarrays corresponding to 28 control (healthy) patients of the neonatal study.
This data set is used as a base expressions set for simulating
biomarker data using <code><a href="#topic+simData">simData</a></code> function of the <code><a href="#topic+optBiomarker">optBiomarker</a></code> package.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(realBiomarker)</code></pre>


<h3>Format</h3>

<p>A vector of 54359 gene expressions in log (base 2) scale.</p>


<h3>References</h3>

<p>Smith, C. L., Dickinson, P., Forster, T., Khondoker, M. R., Craigon, M., Ross, A., Storm, P., Burgess, S., Lacaze, P., Stenson, B. J.and  Ghazal, P. (2007). 
Quantitative assessment of whole blood RNA as a potential biomarker for infectious disease.
<em>Analyst</em> <b>132</b>, 1200&ndash;1209.   
</p>
<p>Khondoker, M. R., Till T. Bachmann, T. T., Mewissen, M., Dickinson, P. <em>et al.</em>(2010).
Multi-factorial analysis of class prediction error: estimating optimal number of biomarkers for various classification rules.
<em>Journal of Bioinformatics and Computational Biology</em>, <b>8</b>, 945-965.
</p>

<hr>
<h2 id='simData'>Simulation of microarray data</h2><span id='topic+simData'></span>

<h3>Description</h3>

<p>The function simulates microarray data for two-group comparison with
user supplied parameters such as  number of biomarkers (genes or proteins),
sample size, biological and experimental (technical) variation,
replication, differential expression, and correlation between biomarkers. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simData(nTrain=100,
        nGr1=floor(nTrain/2),
        nBiom=50,nRep=3,
        sdW=1.0,
        sdB=1.0,
        rhoMax=NULL, rhoMin=NULL, nBlock=NULL,bsMin=3, bSizes=NULL, gamma=NULL,
        sigma=0.1,diffExpr=TRUE,
        foldMin=2,
        orderBiom=TRUE,
        baseExpr=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simData_+3A_ntrain">nTrain</code></td>
<td>
<p>Training set size,.i.e., the total number of biological
samples in group 1 (<code>nGr1</code>) and group 2.</p>
</td></tr>
<tr><td><code id="simData_+3A_ngr1">nGr1</code></td>
<td>
<p>Size of group 1. Defaults to <code>floor(nTrain/2)</code>.</p>
</td></tr>
<tr><td><code id="simData_+3A_nbiom">nBiom</code></td>
<td>
<p>Number of biomarkers (genes, probes or proteins).</p>
</td></tr>
<tr><td><code id="simData_+3A_nrep">nRep</code></td>
<td>
<p>Number of technical replications.</p>
</td></tr>
<tr><td><code id="simData_+3A_sdw">sdW</code></td>
<td>
<p>Experimental (technical) variation (<code class="reqn">\sigma_e</code>) of  data in
log (base 2) scale.</p>
</td></tr>
<tr><td><code id="simData_+3A_sdb">sdB</code></td>
<td>
<p>Biological variation (<code class="reqn">\sigma_b</code>) of  data in
log (base 2) scale.</p>
</td></tr>
<tr><td><code id="simData_+3A_rhomax">rhoMax</code></td>
<td>
<p>Maximum Pearson's correlation coefficient between
biomarkers. To ensure positive definiteness, allowed values are
restricted between 0 and 0.95 inclusive.  If <code>NULL</code>, set to
<code>runif(1,min=0.6,max=0.8)</code>.</p>
</td></tr>
<tr><td><code id="simData_+3A_rhomin">rhoMin</code></td>
<td>
<p>Minimum Pearson's correlation coefficient between
biomarkers. To ensure positive definiteness, allowed values are
restricted between 0 and 0.95 inclusive. If <code>NULL</code>, set to
<code>runif(1,min=0.2,max=0.4)</code>.</p>
</td></tr>
<tr><td><code id="simData_+3A_nblock">nBlock</code></td>
<td>
<p>Number of blocks in the block diagonal (Hub-Toeplitz)
correlation matrix.  If <code>NULL</code>, set to 1 for <code>nBiom&lt;5</code> and
randomly selected from <code>c(1:floor(nBiom/bsMin))</code> for  <code>nBiom&gt;=5</code>.</p>
</td></tr>
<tr><td><code id="simData_+3A_bsmin">bsMin</code></td>
<td>
<p>Minimum block size. <code>bsMin=3</code> by default.</p>
</td></tr>
<tr><td><code id="simData_+3A_bsizes">bSizes</code></td>
<td>
<p>A vector of length <code>nBlock</code> representing the block sizes
(should sum to <code>nBlock</code>). If <code>NULL</code>, set to
<code>c(bs+mod,rep(bs,nBlock-1)</code>, where <code>bs</code> is the integer
part of <code>nBiom/nBlock</code> and <code>mod</code> is the remainder after
integer division.</p>
</td></tr>
<tr><td><code id="simData_+3A_gamma">gamma</code></td>
<td>
<p>Specifies a correlation structure. If <code>NULL</code>, assumes
independence.<code>gamma=0</code> indicates a single block exchangeable
correlation marix with constant correlation
<code>rho=0.5*(rhoMin+rhoMax)</code>. A value greater than zero indicates
block diagonal (Hub-Toeplitz) correlation matrix with decline rate
determined by the value of <code>gamma</code>. Decline rate is linear for
<code>gamma=1</code>.</p>
</td></tr>
<tr><td><code id="simData_+3A_sigma">sigma</code></td>
<td>
<p>Standard deviation of the normal distribution (before
truncation) where fold changes are generated from. See details.</p>
</td></tr>
<tr><td><code id="simData_+3A_diffexpr">diffExpr</code></td>
<td>
<p>Logical. Should systematic difference be introduced
between the data of the two groups?</p>
</td></tr>
<tr><td><code id="simData_+3A_foldmin">foldMin</code></td>
<td>
<p>Minimum value of fold changes. See details.</p>
</td></tr>
<tr><td><code id="simData_+3A_orderbiom">orderBiom</code></td>
<td>
<p>Logical. Should columns (biomarkers) be
arranged in order of differential expression?</p>
</td></tr>
<tr><td><code id="simData_+3A_baseexpr">baseExpr</code></td>
<td>
<p>A vector of length <code>nBiom</code> to be used as base
expressions <code class="reqn">\mu</code>. See <code>realBiomarker</code> for details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Differential expressions are introduced by adding <code class="reqn">z\delta</code> to the data
of group 2 where <code class="reqn">\delta</code> values are generated from a truncated normal
distribution and <code class="reqn">z</code> is randomly selected from <code>(-1,1)</code> to
characterise up- or down-regulation.
</p>
<p>Assuming that <code class="reqn">Y ~is~  N(\mu, \sigma^2)</code>, and <code class="reqn">A=[a_1,a_2]</code>, a subset of
<code class="reqn">-Inf &lt;y &lt; Inf</code>, the conditional distribution of <code class="reqn">Y</code> given <code class="reqn">A</code>
is called  truncated normal  distribution:
</p>
<p style="text-align: center;"><code class="reqn">f(y, \mu, \sigma)= (1/\sigma) \phi((y-\mu)/\sigma) / (\Phi((a2-\mu)/\sigma) -
\Phi((a_1-\mu)/\sigma))</code>
</p>

<p>for <code class="reqn">a_1 &lt;= y &lt;= a_2</code>, and 0 otherwise,
</p>
<p>where <code class="reqn">\mu</code> is the mean of the original Normal distribution before truncation,
<code class="reqn">\sigma</code> is the corresponding standard deviation,<code class="reqn">a_2</code> is the upper truncation point,
<code class="reqn">a_1</code> is the lower truncation point, <code class="reqn">\phi(x)</code> is the density of the
standard normal distribution, and <code class="reqn">\Phi(x)</code> is the distribution function
of the standard normal distribution. For <code>simData</code> function, we
consider <code class="reqn">a_1=log_2(\code{foldMin})</code> and <code class="reqn">a_2=Inf</code>. This ensures that the
biomarkers are differentially expressed by a fold change of
<code>foldMin</code> or more.
</p>


<h3>Value</h3>

<p>A dataframe of dimension <code>nTrain</code> by <code>nBiom+1</code>. The first
column is a factor (<code>class</code>) representing the group memberships of
the samples.
</p>


<h3>Author(s)</h3>

<p>Mizanur Khondoker, Till Bachmann, Peter Ghazal <br />
Maintainer: Mizanur Khondoker <a href="mailto:mizanur.khondoker@gmail.com">mizanur.khondoker@gmail.com</a>.</p>


<h3>References</h3>

<p>Khondoker, M. R., Till T. Bachmann, T. T., Mewissen, M., Dickinson, P. <em>et al.</em>(2010).
Multi-factorial analysis of class prediction error: estimating optimal number of biomarkers for various classification rules.
<em>Journal of Bioinformatics and Computational Biology</em>, <b>8</b>, 945-965.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classificationError">classificationError</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
simData(nTrain=10,nBiom=3)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
