<!DOCTYPE html><html><head><title>Help for package opticskxi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {opticskxi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Magrittr pipe operator</p></a></li>
<li><a href='#%&lt;&gt;%'><p>Magrittr pipe-assign operator</p></a></li>
<li><a href='#%$%'><p>Magrittr pipe-with operator</p></a></li>
<li><a href='#contingency_table'><p>Contingency table</p></a></li>
<li><a href='#crohn'><p>Crohn's disease data</p></a></li>
<li><a href='#fortify_dimred'><p>Fortify a dimension reduction object</p></a></li>
<li><a href='#fortify_ica'><p>Get and fortify ICA</p></a></li>
<li><a href='#fortify_pca'><p>Get and fortify PCA</p></a></li>
<li><a href='#get_best_kxi'><p>Get best k-Xi model</p></a></li>
<li><a href='#ggpairs'><p>Plot multiple axes of a data frame or a fortified dimension reduction.</p></a></li>
<li><a href='#ggplot_kxi_metrics'><p>Ggplot OPTICS k-Xi metrics</p></a></li>
<li><a href='#ggplot_optics'><p>Ggplot optics</p></a></li>
<li><a href='#gtable_kxi_profiles'><p>Gtable OPTICS k-Xi distance profiles</p></a></li>
<li><a href='#hla'><p>The HLA data</p></a></li>
<li><a href='#multishapes'><p>A dataset containing clusters of multiple shapes</p></a></li>
<li><a href='#nice_palette'><p>Nice palette</p></a></li>
<li><a href='#opticskxi'><p>OPTICS k-Xi clustering algorithm</p></a></li>
<li><a href='#opticskxi_pipeline'><p>OPTICS k-Xi models comparison pipeline</p></a></li>
<li><a href='#print_table'><p>Print table</p></a></li>
<li><a href='#residuals_table'><p>Residuals table</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>OPTICS K-Xi Density-Based Clustering</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas Charlon &lt;charlon@protonmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomas Charlon &lt;charlon@protonmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a novel density-based cluster extraction method, OPTICS k-Xi, and a framework to compare k-Xi models using distance-based metrics to investigate datasets with unknown number of clusters.</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, magrittr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>amap, dbscan, cowplot, fastICA, fpc, ggrepel, grid,
grDevices, gtable, knitr, parallel, plyr, reshape2, stats,
testthat, utils</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> | file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-07-15 16:24:20 UTC; root</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-07-18 22:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Magrittr pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>Magrittr pipe operator
</p>

<hr>
<h2 id='+25+26lt+3B+26gt+3B+25'>Magrittr pipe-assign operator</h2><span id='topic++25+3C+3E+25'></span>

<h3>Description</h3>

<p>Magrittr pipe-assign operator
</p>

<hr>
<h2 id='+25+24+25'>Magrittr pipe-with operator</h2><span id='topic++25+24+25'></span>

<h3>Description</h3>

<p>Magrittr pipe-with operator
</p>

<hr>
<h2 id='contingency_table'>Contingency table</h2><span id='topic+contingency_table'></span>

<h3>Description</h3>

<p>Include NAs and add totals to table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contingency_table(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="contingency_table_+3A_...">...</code></td>
<td>
<p>Passed to table</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table object
</p>

<hr>
<h2 id='crohn'>Crohn's disease data</h2><span id='topic+crohn'></span>

<h3>Description</h3>

<p>The data set consist of 103 common (&gt;5% minor allele frequency) SNPs 
genotyped in 129 trios from an European-derived population. These 
SNPs are in a 500-kb region on human chromosome 5q31 implicated as 
containing a genetic risk factor for Crohn disease.
</p>
<p>Imported from the gap R package.
</p>
<p>An example use of the data is with the following paper,
Kelly M. Burkett, Celia M. T. Greenwood, BradMcNeney, Jinko Graham.
Gene genealogies for genetic association mapping, with application to Crohn's disease.
Fron Genet 2013, 4(260) doi: 10.3389/fgene.2013.00260
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(crohn)</code></pre>


<h3>Format</h3>

<p>A data frame containing 387 rows and 212 columns</p>


<h3>Source</h3>

<p>MJ Daly, JD Rioux, SF Schaffner, TJ Hudson, ES Lander (2001) 
High-resolution haplotype structure in the human genome Nature 
Genetics 29:229-232
</p>

<hr>
<h2 id='fortify_dimred'>Fortify a dimension reduction object</h2><span id='topic+fortify_dimred'></span>

<h3>Description</h3>

<p>Fortify a dimension reduction object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fortify_dimred(m_dimred, m_vars = NULL, v_variance = NULL,
  sup_vars = NULL, var_digits = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fortify_dimred_+3A_m_dimred">m_dimred</code></td>
<td>
<p>Projection matrix</p>
</td></tr>
<tr><td><code id="fortify_dimred_+3A_m_vars">m_vars</code></td>
<td>
<p>Rotation matrix (optional)</p>
</td></tr>
<tr><td><code id="fortify_dimred_+3A_v_variance">v_variance</code></td>
<td>
<p>Explained variance (optional)</p>
</td></tr>
<tr><td><code id="fortify_dimred_+3A_sup_vars">sup_vars</code></td>
<td>
<p>Optional supplementary variables</p>
</td></tr>
<tr><td><code id="fortify_dimred_+3A_var_digits">var_digits</code></td>
<td>
<p>Explained variance percent digits</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame
</p>


<h3>See Also</h3>

<p><a href="#topic+fortify_pca">fortify_pca</a>, <a href="#topic+fortify_ica">fortify_ica</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pca &lt;- prcomp(iris[-5])
df_pca &lt;- fortify_dimred(pca$x)
</code></pre>

<hr>
<h2 id='fortify_ica'>Get and fortify ICA</h2><span id='topic+fortify_ica'></span>

<h3>Description</h3>

<p>Get and fortify ICA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fortify_ica(m_data, ..., sup_vars = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fortify_ica_+3A_m_data">m_data</code></td>
<td>
<p>Input matrix</p>
</td></tr>
<tr><td><code id="fortify_ica_+3A_...">...</code></td>
<td>
<p>Passed to fastICA::fastICA</p>
</td></tr>
<tr><td><code id="fortify_ica_+3A_sup_vars">sup_vars</code></td>
<td>
<p>Optional supplementary variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fortified dimension reduction
</p>


<h3>See Also</h3>

<p><a href="#topic+fortify_dimred">fortify_dimred</a>, <a href="#topic+fortify_pca">fortify_pca</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_ica &lt;- fortify_ica(iris[-5], n.comp = 2)
</code></pre>

<hr>
<h2 id='fortify_pca'>Get and fortify PCA</h2><span id='topic+fortify_pca'></span>

<h3>Description</h3>

<p>Get and fortify PCA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fortify_pca(m_data, ..., sup_vars = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fortify_pca_+3A_m_data">m_data</code></td>
<td>
<p>Input matrix</p>
</td></tr>
<tr><td><code id="fortify_pca_+3A_...">...</code></td>
<td>
<p>Passed to stats::prcomp</p>
</td></tr>
<tr><td><code id="fortify_pca_+3A_sup_vars">sup_vars</code></td>
<td>
<p>Optional supplementary variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fortified dimension reduction
</p>


<h3>See Also</h3>

<p><a href="#topic+fortify_dimred">fortify_dimred</a>, <a href="#topic+fortify_ica">fortify_ica</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_pca &lt;- fortify_pca(iris[-5])
df_pca &lt;- fortify_pca(iris[-5], sup_vars = iris[5])
</code></pre>

<hr>
<h2 id='get_best_kxi'>Get best k-Xi model</h2><span id='topic+get_best_kxi'></span>

<h3>Description</h3>

<p>Select k-Xi clustering model based on a metric and a rank
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_best_kxi(df_kxi, metric = "avg.silwidth", rank = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_best_kxi_+3A_df_kxi">df_kxi</code></td>
<td>
<p>Data frame returned by opticsxi_pipeline</p>
</td></tr>
<tr><td><code id="get_best_kxi_+3A_metric">metric</code></td>
<td>
<p>Metric to choose best model</p>
</td></tr>
<tr><td><code id="get_best_kxi_+3A_rank">rank</code></td>
<td>
<p>Rank(s) of model to choose, ordered by decreasing metric</p>
</td></tr>
</table>


<h3>Value</h3>

<p>df_kxi row with specified metric and rank,
simplified to a list if only one rank selected
</p>


<h3>See Also</h3>

<p><a href="#topic+opticskxi_pipeline">opticskxi_pipeline</a>
</p>

<hr>
<h2 id='ggpairs'>Plot multiple axes of a data frame or a fortified dimension reduction.</h2><span id='topic+ggpairs'></span>

<h3>Description</h3>

<p>Plot multiple axes of a data frame or a fortified dimension reduction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggpairs(df_data, group = NULL, axes = 1:2, variables = FALSE,
  n_vars = 0, ellipses = FALSE, ..., title = NULL, colors = if
  (!is.null(group)) nice_palette(df_data[[group]]))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggpairs_+3A_df_data">df_data</code></td>
<td>
<p>Data frame</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_group">group</code></td>
<td>
<p>Column name of the grouping of observations</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_axes">axes</code></td>
<td>
<p>Axes to plot. If more than 2, plots all pair combinations</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_variables">variables</code></td>
<td>
<p>Logical, plot variable contributions of the dimension
reduction to the selected axes, only for 2 axes</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_n_vars">n_vars</code></td>
<td>
<p>Maximum number of variable contributions to plot. By
default 0, for all variables.</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_ellipses">ellipses</code></td>
<td>
<p>Logical, plot ellipses of groups</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_...">...</code></td>
<td>
<p>Passed to ggplot2 stat_ellipse if ellipses are requested</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_title">title</code></td>
<td>
<p>String to add as title, default NULL</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_colors">colors</code></td>
<td>
<p>Vector of colors for each group</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggmatrix
</p>


<h3>See Also</h3>

<p><a href="#topic+fortify_pca">fortify_pca</a>, <a href="#topic+fortify_ica">fortify_ica</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_pca &lt;- fortify_pca(iris[-5])
ggpairs(df_pca)
df_pca &lt;- fortify_pca(iris[-5], sup_vars = iris[5])
ggpairs(df_pca, group = 'Species', ellipses = TRUE, variables = TRUE)
</code></pre>

<hr>
<h2 id='ggplot_kxi_metrics'>Ggplot OPTICS k-Xi metrics</h2><span id='topic+ggplot_kxi_metrics'></span>

<h3>Description</h3>

<p>Plot metrics of a kxi_pipeline output
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_kxi_metrics(df_kxi, metric = c("avg.silwidth", "bw.ratio"),
  n = 8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggplot_kxi_metrics_+3A_df_kxi">df_kxi</code></td>
<td>
<p>Data frame returned by opticskxi_pipeline</p>
</td></tr>
<tr><td><code id="ggplot_kxi_metrics_+3A_metric">metric</code></td>
<td>
<p>Vector of metrics to display from the df_kxi object</p>
</td></tr>
<tr><td><code id="ggplot_kxi_metrics_+3A_n">n</code></td>
<td>
<p>Number of best models for the first metric to display</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p><a href="#topic+opticskxi_pipeline">opticskxi_pipeline</a>
</p>

<hr>
<h2 id='ggplot_optics'>Ggplot optics</h2><span id='topic+ggplot_optics'></span>

<h3>Description</h3>

<p>Plot OPTICS reachability plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_optics(optics_obj, groups = NULL, colors = if (!is.null(groups))
  nice_palette(groups), segment_size = 300/nrow(df_optics))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggplot_optics_+3A_optics_obj">optics_obj</code></td>
<td>
<p>dbscan::optics object</p>
</td></tr>
<tr><td><code id="ggplot_optics_+3A_groups">groups</code></td>
<td>
<p>Optional vector defining groups of OPTICS observations</p>
</td></tr>
<tr><td><code id="ggplot_optics_+3A_colors">colors</code></td>
<td>
<p>If groups specified, vector of colors for each group</p>
</td></tr>
<tr><td><code id="ggplot_optics_+3A_segment_size">segment_size</code></td>
<td>
<p>Size for geom_segment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p><a href="#topic+opticskxi">opticskxi</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('multishapes')
optics_obj &lt;- dbscan::optics(multishapes[1:2])
ggplot_optics(optics_obj)
ggplot_optics(optics_obj,
  groups = opticskxi(optics_obj, n_xi = 5, pts = 30))
</code></pre>

<hr>
<h2 id='gtable_kxi_profiles'>Gtable OPTICS k-Xi distance profiles</h2><span id='topic+gtable_kxi_profiles'></span>

<h3>Description</h3>

<p>Plot OPTICS distance profiles of k-Xi clustering models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtable_kxi_profiles(df_kxi, metric = "avg.silwidth", rank = 1:4, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gtable_kxi_profiles_+3A_df_kxi">df_kxi</code></td>
<td>
<p>Data frame returned by opticskxi_pipeline</p>
</td></tr>
<tr><td><code id="gtable_kxi_profiles_+3A_metric">metric</code></td>
<td>
<p>Metric to choose best clustering model</p>
</td></tr>
<tr><td><code id="gtable_kxi_profiles_+3A_rank">rank</code></td>
<td>
<p>Ranks of models to plot, ordered by decreasing model metric</p>
</td></tr>
<tr><td><code id="gtable_kxi_profiles_+3A_...">...</code></td>
<td>
<p>Passed to ggplot_kxi_profile</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="#topic+opticskxi_pipeline">opticskxi_pipeline</a>
</p>

<hr>
<h2 id='hla'>The HLA data</h2><span id='topic+hla'></span>

<h3>Description</h3>

<p>This data set contains HLA markers DRB, DQA, DQB and phenotypes
of 271 Schizophrenia patients (y=1) and controls (y=0). Genotypes for 3 HLA
loci have prefixes name (e.g., &quot;DQB&quot;) and a suffix for each of two alleles 
(&quot;.a1&quot; and &quot;.a2&quot;).
</p>
<p>Imported from the gap package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hla)</code></pre>


<h3>Format</h3>

<p>A data frame containing 271 rows and 8 columns</p>


<h3>Source</h3>

<p>Dr Padraig Wright of Pfizer</p>

<hr>
<h2 id='multishapes'>A dataset containing clusters of multiple shapes</h2><span id='topic+multishapes'></span>

<h3>Description</h3>

<p>Data containing clusters of any shapes. Useful for comparing
density-based clustering (DBSCAN) and standard partitioning methods such as
k-means clustering. Imported from the factoextra package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("multishapes")
</code></pre>


<h3>Format</h3>

<p>A data frame with 1100 observations on the following 3 variables. 
</p>

<dl>
<dt><code>x</code></dt><dd><p>a numeric vector containing the x coordinates of observations</p>
</dd>
<dt><code>y</code></dt><dd><p>a numeric vector containing the y coordinates of observations</p>
</dd>
<dt><code>shape</code></dt><dd><p>a numeric vector corresponding to the cluster number of each observations.</p>
</dd>
</dl>


<h3>Details</h3>

<p>The dataset contains 5 clusters and some outliers/noises.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('multishapes')
plot(multishapes[, 1], multishapes[, 2],
    col = multishapes[, 3], pch = 19, cex = 0.8)

</code></pre>

<hr>
<h2 id='nice_palette'>Nice palette</h2><span id='topic+nice_palette'></span>

<h3>Description</h3>

<p>Color palette
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nice_palette(groups, rainbow = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nice_palette_+3A_groups">groups</code></td>
<td>
<p>Vector, each unique value will get a color</p>
</td></tr>
<tr><td><code id="nice_palette_+3A_rainbow">rainbow</code></td>
<td>
<p>If TRUE, rainbow-like colors, else differentiate successive
values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of colors
</p>

<hr>
<h2 id='opticskxi'>OPTICS k-Xi clustering algorithm</h2><span id='topic+opticskxi'></span>

<h3>Description</h3>

<p>For each largest distance differences on the OPTICS profile,
consecutive observations left and right on the OPTICS profile
(i.e. lower and higher OPTICS id) will be assigned to 2 different clusters
if their distance is below the distance of the edge point.
If above, observations are NA.
The pts parameter defines a minimum number of observations to form a
valley (i.e. cluster). If the number of observations in one valley is
smaller than pts, observations are set to NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opticskxi(optics_obj, n_xi, pts = optics_obj$minPts, max_loop = 50,
  verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="opticskxi_+3A_optics_obj">optics_obj</code></td>
<td>
<p>Data frame returned by optics</p>
</td></tr>
<tr><td><code id="opticskxi_+3A_n_xi">n_xi</code></td>
<td>
<p>Number of clusters to define</p>
</td></tr>
<tr><td><code id="opticskxi_+3A_pts">pts</code></td>
<td>
<p>Minimum number of points per clusters</p>
</td></tr>
<tr><td><code id="opticskxi_+3A_max_loop">max_loop</code></td>
<td>
<p>Maximum iterations to find n_xi clusters</p>
</td></tr>
<tr><td><code id="opticskxi_+3A_verbose">verbose</code></td>
<td>
<p>Print the ids of the largest difference considered and
cluster information if they define one</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of clusters
</p>


<h3>See Also</h3>

<p><a href="#topic+opticskxi_pipeline">opticskxi_pipeline</a>, <a href="#topic+ggplot_optics">ggplot_optics</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('multishapes')
optics_shapes &lt;- dbscan::optics(multishapes[1:2])
kxi_shapes &lt;- opticskxi(optics_shapes, n_xi = 5, pts = 30)
ggplot_optics(optics_shapes, groups = kxi_shapes)
ggpairs(cbind(multishapes[1:2], kXi = kxi_shapes), group = 'kXi')
</code></pre>

<hr>
<h2 id='opticskxi_pipeline'>OPTICS k-Xi models comparison pipeline</h2><span id='topic+opticskxi_pipeline'></span>

<h3>Description</h3>

<p>Computes OPTICS k-Xi models based on a parameter grid, binds results in a
data frame, and computes distance based metrics for each model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opticskxi_pipeline(m_data, df_params = expand.grid(n_xi = 1:10, pts =
  c(20, 30, 40), dist = c("euclidean", "abscorrelation"), dim_red =
  c("identity", "PCA", "ICA"), n_dimred_comp = c(5, 10, 20)),
  n_cores = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="opticskxi_pipeline_+3A_m_data">m_data</code></td>
<td>
<p>Data matrix</p>
</td></tr>
<tr><td><code id="opticskxi_pipeline_+3A_df_params">df_params</code></td>
<td>
<p>Parameter grid for the OPTICS k-Xi function call and
optional dimension reduction.
Required columns: n_xi, pts, dist.
Optonal columns: dim_red, n_dim_red.</p>
</td></tr>
<tr><td><code id="opticskxi_pipeline_+3A_n_cores">n_cores</code></td>
<td>
<p>Number of cores</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Input parameter data frame with with results binded in columns
optics, clusters and metrics.
</p>


<h3>See Also</h3>

<p><a href="#topic+get_best_kxi">get_best_kxi</a>, <a href="#topic+ggplot_kxi_metrics">ggplot_kxi_metrics</a>,
<a href="#topic+gtable_kxi_profiles">gtable_kxi_profiles</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('hla')
m_hla &lt;- hla[-c(1:2)] %&gt;% scale
df_params_hla &lt;- expand.grid(n_xi = 3:5, pts = c(20, 30),
  dist = c('manhattan', 'euclidean'))
df_kxi_hla &lt;- opticskxi_pipeline(m_hla, df_params_hla)
ggplot_kxi_metrics(df_kxi_hla, n = 8)
gtable_kxi_profiles(df_kxi_hla) %&gt;% plot

best_kxi_hla &lt;- get_best_kxi(df_kxi_hla, rank = 2)
clusters_hla &lt;- best_kxi_hla$clusters
fortify_pca(m_hla, sup_vars = data.frame(Clusters = clusters_hla)) %&gt;%
  ggpairs('Clusters', ellipses = TRUE, variables = TRUE)
</code></pre>

<hr>
<h2 id='print_table'>Print table</h2><span id='topic+print_table'></span>

<h3>Description</h3>

<p>Print knitr::kable latex table with legend at bottom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_table(table_obj, label)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_table_+3A_table_obj">table_obj</code></td>
<td>
<p>Table object</p>
</td></tr>
<tr><td><code id="print_table_+3A_label">label</code></td>
<td>
<p>Latex label</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None
</p>

<hr>
<h2 id='residuals_table'>Residuals table</h2><span id='topic+residuals_table'></span>

<h3>Description</h3>

<p>Bind contingency table and Pearson Chi-squared residuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>residuals_table(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="residuals_table_+3A_...">...</code></td>
<td>
<p>Passed to contingency_table and chisq.test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
