<!DOCTYPE html><html lang="en"><head><title>Help for package opticskxi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {opticskxi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+25+26gt+3B+25'><p>Pipe</p></a></li>
<li><a href='#+25+26lt+3B+26gt+3B+25'><p>Assignment pipe</p></a></li>
<li><a href='#+25+24+25'><p>Exposition pipe</p></a></li>
<li><a href='#contingency_table'><p>Contingency table</p></a></li>
<li><a href='#cosine_simi'><p>Cosine similarity between vectors and/or matrices.</p></a></li>
<li><a href='#crohn'><p>Crohn's disease data</p></a></li>
<li><a href='#dist_matrix'><p>dist_matrix</p></a></li>
<li><a href='#ensemble_metrics'><p>Compute ensemble metrics</p></a></li>
<li><a href='#ensemble_metrics_bootstrap'><p>Select models based on ensemble metrics</p></a></li>
<li><a href='#ensemble_models'><p>Select best models based on ensemble metrics</p></a></li>
<li><a href='#fortify_dimred'><p>Fortify a dimension reduction object</p></a></li>
<li><a href='#fortify_ica'><p>Get and fortify ICA</p></a></li>
<li><a href='#fortify_pca'><p>Get and fortify PCA</p></a></li>
<li><a href='#get_best_kxi'><p>Get best k-Xi model</p></a></li>
<li><a href='#ggpairs'><p>Plot multiple axes of a data frame or a fortified dimension reduction.</p></a></li>
<li><a href='#ggplot_kxi_metrics'><p>Ggplot OPTICS k-Xi metrics</p></a></li>
<li><a href='#ggplot_optics'><p>Ggplot optics</p></a></li>
<li><a href='#gtable_kxi_profiles'><p>Gtable OPTICS k-Xi distance profiles</p></a></li>
<li><a href='#hla'><p>The HLA data</p></a></li>
<li><a href='#m_psych_embeds'><p>A dataset containing the embeddings matrix of psychological related words</p></a></li>
<li><a href='#multishapes'><p>A dataset containing clusters of multiple shapes</p></a></li>
<li><a href='#nice_palette'><p>Nice palette</p></a></li>
<li><a href='#norm_inprod'><p>norm_inprod</p></a></li>
<li><a href='#normalize'><p>Matrix normalization</p></a></li>
<li><a href='#opticskxi'><p>OPTICS k-Xi clustering algorithm</p></a></li>
<li><a href='#opticskxi_pipeline'><p>OPTICS k-Xi models comparison pipeline</p></a></li>
<li><a href='#print_vignette_table'><p>Print vignette table</p></a></li>
<li><a href='#residuals_table'><p>Residuals table</p></a></li>
<li><a href='#stddev_mean'><p>stddev_mean</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>OPTICS K-Xi Density-Based Clustering</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Density-based clustering methods are well adapted to the clustering of high-dimensional data and enable the discovery of core groups of various shapes despite large amounts of noise. This package provides a novel density-based cluster extraction method, OPTICS k-Xi, and a framework to compare k-Xi models using distance-based metrics to investigate datasets with unknown number of clusters. The vignette first introduces density-based algorithms with simulated datasets, then presents and evaluates the k-Xi cluster extraction method. Finally, the models comparison framework is described and experimented on 2 genetic datasets to identify groups and their discriminating features. The k-Xi algorithm is a novel OPTICS cluster extraction method that specifies directly the number of clusters and does not require fine-tuning of the steepness parameter as the OPTICS Xi method. Combined with a framework that compares models with varying parameters, the OPTICS k-Xi method can identify groups in noisy datasets with unknown number of clusters. Results on summarized genetic data of 1,200 patients are in Charlon T. (2019) &lt;<a href="https://doi.org/10.13097%2Farchive-ouverte%2Funige%3A161795">doi:10.13097/archive-ouverte/unige:161795</a>&gt;. A short video tutorial can be found at <a href="https://www.youtube.com/watch?v=P2XAjqI5Lc4/">https://www.youtube.com/watch?v=P2XAjqI5Lc4/</a>.</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, magrittr, Matrix, rlang</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>amap, dbscan, cowplot, fastICA, fpc, ggrepel, grid,
grDevices, gtable, knitr, parallel, plyr, reshape2, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://gitlab.com/thomaschln/opticskxi">https://gitlab.com/thomaschln/opticskxi</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://gitlab.com/thomaschln/opticskxi/-/issues">https://gitlab.com/thomaschln/opticskxi/-/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-09 22:37:16 UTC; root</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas Charlon <a href="https://orcid.org/0000-0001-7497-0470"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomas Charlon &lt;charlon@protonmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-09 22:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>Pipe an object forward into a function or call expression.
Magrittr imported function, see details and examples in the magrittr package.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Result of rhs applied to lhs, see details in magrittr package.
</p>

<hr>
<h2 id='+25+26lt+3B+26gt+3B+25'>Assignment pipe</h2><span id='topic++25+3C+3E+25'></span>

<h3>Description</h3>

<p>Pipe an object forward into a function or call expression and update the
'lhs' object with the resulting value.
Magrittr imported function, see details and examples in the magrittr package.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B25+2B26lt+2B3B+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>An object which serves both as the initial value and as target.</p>
</td></tr>
<tr><td><code id="+2B25+2B26lt+2B3B+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>a function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None, used to update the value of lhs.
</p>

<hr>
<h2 id='+25+24+25'>Exposition pipe</h2><span id='topic++25+24+25'></span>

<h3>Description</h3>

<p>Expose the names in 'lhs' to the 'rhs' expression.
Magrittr imported function, see details and examples in the magrittr package.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B25+2B24+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A list, environment, or a data.frame.</p>
</td></tr>
<tr><td><code id="+2B25+2B24+2B25_+3A_rhs">rhs</code></td>
<td>
<p>An expression where the names in lhs is available.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Result of rhs applied to one or several names of lhs.
</p>

<hr>
<h2 id='contingency_table'>Contingency table</h2><span id='topic+contingency_table'></span>

<h3>Description</h3>

<p>Include NAs and add totals to table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contingency_table(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="contingency_table_+3A_...">...</code></td>
<td>
<p>Passed to table</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Table object
</p>

<hr>
<h2 id='cosine_simi'>Cosine similarity between vectors and/or matrices.</h2><span id='topic+cosine_simi'></span>

<h3>Description</h3>

<p>Inputs will be L2 normalized, then matrix multiplied (y is transposed).
If second input is missing, first input will be recycled, which enables to
efficiently compute cosine similarities between the rows of a rectangular
matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cosine_simi(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cosine_simi_+3A_x">x</code></td>
<td>
<p>Numeric vector or matrix</p>
</td></tr>
<tr><td><code id="cosine_simi_+3A_y">y</code></td>
<td>
<p>Numeric vector or matrix. If missing, copied from parameter x.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Symmetric numeric similarity matrix
</p>

<hr>
<h2 id='crohn'>Crohn's disease data</h2><span id='topic+crohn'></span>

<h3>Description</h3>

<p>The data set consist of 103 common (&gt;5% minor allele frequency) SNPs 
genotyped in 129 trios from an European-derived population. These 
SNPs are in a 500-kb region on human chromosome 5q31 implicated as 
containing a genetic risk factor for Crohn disease.
</p>
<p>Imported from the gap R package.
</p>
<p>An example use of the data is with the following paper,
Kelly M. Burkett, Celia M. T. Greenwood, BradMcNeney, Jinko Graham.
Gene genealogies for genetic association mapping, with application to Crohn's disease.
Fron Genet 2013, 4(260) doi: 10.3389/fgene.2013.00260
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(crohn)</code></pre>


<h3>Format</h3>

<p>A data frame containing 387 rows and 212 columns</p>


<h3>Source</h3>

<p>MJ Daly, JD Rioux, SF Schaffner, TJ Hudson, ES Lander (2001) 
High-resolution haplotype structure in the human genome Nature 
Genetics 29:229-232
</p>

<hr>
<h2 id='dist_matrix'>dist_matrix</h2><span id='topic+dist_matrix'></span>

<h3>Description</h3>

<p>Dispatch of amap::Dist, cosine_dist, and norm_inprod methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_matrix(data, method = "euclidean", n_cores = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dist_matrix_+3A_data">data</code></td>
<td>
<p>Rectangular numeric matrix [Observations, Features]</p>
</td></tr>
<tr><td><code id="dist_matrix_+3A_method">method</code></td>
<td>
<p>Methods accepted by amap::Dist or cosine and norm_inprod</p>
</td></tr>
<tr><td><code id="dist_matrix_+3A_n_cores">n_cores</code></td>
<td>
<p>Number of cores</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Distance symmetric matrix
</p>

<hr>
<h2 id='ensemble_metrics'>Compute ensemble metrics</h2><span id='topic+ensemble_metrics'></span>

<h3>Description</h3>

<p>Use models' rankings over several metrics to select best model.
Several approaches can be taken to sum the models' rankings, and instead of
summing the ranks of all models over all metrics, we prefer to rank only the
top models for each metrics, and set 0 to all other. This behavior is
controlled by the n_top parameter. In a second step, we sum the ranks and
return only the top models, and this is controlled by the n_models
parameter.
The output is a list of the rankings matrix, for quality control purposes,
and the selected models' parameters data frame, which is used by the
ensemble_models function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ensemble_metrics(
  n_top = 0,
  df_params,
  metrics = NULL,
  metrics_exclude = NULL,
  n_models = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ensemble_metrics_+3A_n_top">n_top</code></td>
<td>
<p>Threshold of number of models to rank</p>
</td></tr>
<tr><td><code id="ensemble_metrics_+3A_df_params">df_params</code></td>
<td>
<p>Output of opticskxi_pipeline</p>
</td></tr>
<tr><td><code id="ensemble_metrics_+3A_metrics">metrics</code></td>
<td>
<p>Names of metrics to use. Any of those computed by
opticskxi_pipeline, e.g. 'sindex', 'ch', 'dunn',
'dunn2', 'widestgap', 'entropy' etc.
NULL for all (8).</p>
</td></tr>
<tr><td><code id="ensemble_metrics_+3A_metrics_exclude">metrics_exclude</code></td>
<td>
<p>Names of metrics to exclude.
Typically used with metrics = NULL. E.g. 'entropy'.</p>
</td></tr>
<tr><td><code id="ensemble_metrics_+3A_n_models">n_models</code></td>
<td>
<p>Number of best models to return</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of metrics' rankings matrix and best models' parameters data
frame.
</p>

<hr>
<h2 id='ensemble_metrics_bootstrap'>Select models based on ensemble metrics</h2><span id='topic+ensemble_metrics_bootstrap'></span>

<h3>Description</h3>

<p>Typically we will call ensemble_metrics with varying numbers of ranks to
consider and this function will sum up the ranks from those calls.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ensemble_metrics_bootstrap(l_ensemble_metrics, n_models = 4)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ensemble_metrics_bootstrap_+3A_l_ensemble_metrics">l_ensemble_metrics</code></td>
<td>
<p>Output of function ensemble_metrics</p>
</td></tr>
<tr><td><code id="ensemble_metrics_bootstrap_+3A_n_models">n_models</code></td>
<td>
<p>Number of best models to return</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of parameters of best models
</p>

<hr>
<h2 id='ensemble_models'>Select best models based on ensemble metrics</h2><span id='topic+ensemble_models'></span>

<h3>Description</h3>

<p>Call ensemble_metrics with varying numbers of rank thresholds to
consider and sum up the ranks from those calls.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ensemble_models(
  df_kxi,
  n_models = 4,
  metrics = NULL,
  metrics_exclude = NULL,
  model_subsample = c(0.1, 0.2, 0.5),
  n_models_subsample = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ensemble_models_+3A_df_kxi">df_kxi</code></td>
<td>
<p>Output of opticskxi_pipeline function.
Dataframe with models' parameters and OPTICS k-Xi results</p>
</td></tr>
<tr><td><code id="ensemble_models_+3A_n_models">n_models</code></td>
<td>
<p>Number of best models to return</p>
</td></tr>
<tr><td><code id="ensemble_models_+3A_metrics">metrics</code></td>
<td>
<p>Names of metrics to use. Any of those computed by
opticskxi_pipeline, e.g. 'sindex', 'ch', 'dunn',
'dunn2', 'widestgap', 'entropy' etc.
NULL for all (8).</p>
</td></tr>
<tr><td><code id="ensemble_models_+3A_metrics_exclude">metrics_exclude</code></td>
<td>
<p>Names of metrics to exclude.
Typically used with metrics = NULL. E.g. 'entropy'.</p>
</td></tr>
<tr><td><code id="ensemble_models_+3A_model_subsample">model_subsample</code></td>
<td>
<p>Ratios of best models to consider.</p>
</td></tr>
<tr><td><code id="ensemble_models_+3A_n_models_subsample">n_models_subsample</code></td>
<td>
<p>Number of best models when subsampling.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Input object df_kxi subsetted to best models according to ensemble
metrics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('m_psych_embeds')
m_psych_embeds = m_psych_embeds[1:200, 1:20]

df_params = expand.grid(n_xi = 4:5, pts = c(5, 10), dist = 'cosine',
                        dim_red = 'ICA', n_dimred_comp = 5)

df_kxi = opticskxi_pipeline(m_psych_embeds, df_params,
                            metrics_dist = 'cosine',
                            n_min_clusters = 2, n_cores = 1,
                            metrics = c('avg.silwidth', 'dunn'))

df_kxi = ensemble_models(df_kxi, n_models = 4,
                         model_subsample = c(0.4, 0.6),
                         n_models_subsample = 4)

</code></pre>

<hr>
<h2 id='fortify_dimred'>Fortify a dimension reduction object</h2><span id='topic+fortify_dimred'></span>

<h3>Description</h3>

<p>Fortify a dimension reduction object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fortify_dimred(
  m_dimred,
  m_vars = NULL,
  v_variance = NULL,
  sup_vars = NULL,
  var_digits = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_dimred_+3A_m_dimred">m_dimred</code></td>
<td>
<p>Projection matrix</p>
</td></tr>
<tr><td><code id="fortify_dimred_+3A_m_vars">m_vars</code></td>
<td>
<p>Rotation matrix (optional)</p>
</td></tr>
<tr><td><code id="fortify_dimred_+3A_v_variance">v_variance</code></td>
<td>
<p>Explained variance (optional)</p>
</td></tr>
<tr><td><code id="fortify_dimred_+3A_sup_vars">sup_vars</code></td>
<td>
<p>Optional supplementary variables</p>
</td></tr>
<tr><td><code id="fortify_dimred_+3A_var_digits">var_digits</code></td>
<td>
<p>Explained variance percent digits</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame
</p>


<h3>See Also</h3>

<p><a href="#topic+fortify_pca">fortify_pca</a>, <a href="#topic+fortify_ica">fortify_ica</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pca &lt;- prcomp(iris[-5])
df_pca &lt;- fortify_dimred(pca$x)
</code></pre>

<hr>
<h2 id='fortify_ica'>Get and fortify ICA</h2><span id='topic+fortify_ica'></span>

<h3>Description</h3>

<p>Get and fortify ICA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fortify_ica(m_data, ..., sup_vars = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_ica_+3A_m_data">m_data</code></td>
<td>
<p>Input matrix</p>
</td></tr>
<tr><td><code id="fortify_ica_+3A_...">...</code></td>
<td>
<p>Passed to fastICA::fastICA</p>
</td></tr>
<tr><td><code id="fortify_ica_+3A_sup_vars">sup_vars</code></td>
<td>
<p>Optional supplementary variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fortified dimension reduction
</p>


<h3>See Also</h3>

<p><a href="#topic+fortify_dimred">fortify_dimred</a>, <a href="#topic+fortify_pca">fortify_pca</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_ica &lt;- fortify_ica(iris[-5], n.comp = 2)
</code></pre>

<hr>
<h2 id='fortify_pca'>Get and fortify PCA</h2><span id='topic+fortify_pca'></span>

<h3>Description</h3>

<p>Get and fortify PCA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fortify_pca(m_data, ..., sup_vars = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fortify_pca_+3A_m_data">m_data</code></td>
<td>
<p>Input matrix</p>
</td></tr>
<tr><td><code id="fortify_pca_+3A_...">...</code></td>
<td>
<p>Passed to stats::prcomp</p>
</td></tr>
<tr><td><code id="fortify_pca_+3A_sup_vars">sup_vars</code></td>
<td>
<p>Optional supplementary variables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fortified dimension reduction
</p>


<h3>See Also</h3>

<p><a href="#topic+fortify_dimred">fortify_dimred</a>, <a href="#topic+fortify_ica">fortify_ica</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_pca &lt;- fortify_pca(iris[-5])
df_pca &lt;- fortify_pca(iris[-5], sup_vars = iris[5])
</code></pre>

<hr>
<h2 id='get_best_kxi'>Get best k-Xi model</h2><span id='topic+get_best_kxi'></span>

<h3>Description</h3>

<p>Select k-Xi clustering model based on a metric and a rank
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_best_kxi(df_kxi, metric = "avg.silwidth", rank = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_best_kxi_+3A_df_kxi">df_kxi</code></td>
<td>
<p>Data frame returned by opticsxi_pipeline</p>
</td></tr>
<tr><td><code id="get_best_kxi_+3A_metric">metric</code></td>
<td>
<p>Metric to choose best model</p>
</td></tr>
<tr><td><code id="get_best_kxi_+3A_rank">rank</code></td>
<td>
<p>Rank(s) of model to choose, ordered by decreasing metric</p>
</td></tr>
</table>


<h3>Value</h3>

<p>df_kxi row with specified metric and rank,
simplified to a list if only one rank selected
</p>


<h3>See Also</h3>

<p><a href="#topic+opticskxi_pipeline">opticskxi_pipeline</a>
</p>

<hr>
<h2 id='ggpairs'>Plot multiple axes of a data frame or a fortified dimension reduction.</h2><span id='topic+ggpairs'></span>

<h3>Description</h3>

<p>Plot multiple axes of a data frame or a fortified dimension reduction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggpairs(
  df_data,
  group = NULL,
  axes = 1:2,
  variables = FALSE,
  n_vars = 0,
  ellipses = FALSE,
  ...,
  title = NULL,
  colors = if (!is.null(group)) nice_palette(df_data[[group]])
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggpairs_+3A_df_data">df_data</code></td>
<td>
<p>Data frame</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_group">group</code></td>
<td>
<p>Column name of the grouping of observations</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_axes">axes</code></td>
<td>
<p>Axes to plot. If more than 2, plots all pair combinations</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_variables">variables</code></td>
<td>
<p>Logical, plot variable contributions of the dimension
reduction to the selected axes, only for 2 axes</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_n_vars">n_vars</code></td>
<td>
<p>Maximum number of variable contributions to plot. By
default 0, for all variables.</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_ellipses">ellipses</code></td>
<td>
<p>Logical, plot ellipses of groups</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_...">...</code></td>
<td>
<p>Passed to ggplot2 stat_ellipse if ellipses are requested</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_title">title</code></td>
<td>
<p>String to add as title, default NULL</p>
</td></tr>
<tr><td><code id="ggpairs_+3A_colors">colors</code></td>
<td>
<p>Vector of colors for each group</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggmatrix
</p>


<h3>See Also</h3>

<p><a href="#topic+fortify_pca">fortify_pca</a>, <a href="#topic+fortify_ica">fortify_ica</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_pca &lt;- fortify_pca(iris[-5])
ggpairs(df_pca)
df_pca &lt;- fortify_pca(iris[-5], sup_vars = iris[5])
ggpairs(df_pca, group = 'Species', ellipses = TRUE, variables = TRUE)
</code></pre>

<hr>
<h2 id='ggplot_kxi_metrics'>Ggplot OPTICS k-Xi metrics</h2><span id='topic+ggplot_kxi_metrics'></span>

<h3>Description</h3>

<p>Plot metrics of a kxi_pipeline output
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_kxi_metrics(df_kxi, metric = c("avg.silwidth", "bw.ratio"), n = 8)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_kxi_metrics_+3A_df_kxi">df_kxi</code></td>
<td>
<p>Data frame returned by opticskxi_pipeline</p>
</td></tr>
<tr><td><code id="ggplot_kxi_metrics_+3A_metric">metric</code></td>
<td>
<p>Vector of metrics to display from the df_kxi object</p>
</td></tr>
<tr><td><code id="ggplot_kxi_metrics_+3A_n">n</code></td>
<td>
<p>Number of best models for the first metric to display</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p><a href="#topic+opticskxi_pipeline">opticskxi_pipeline</a>
</p>

<hr>
<h2 id='ggplot_optics'>Ggplot optics</h2><span id='topic+ggplot_optics'></span>

<h3>Description</h3>

<p>Plot OPTICS reachability plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggplot_optics(
  optics_obj,
  groups = NULL,
  colors = if (!is.null(groups)) nice_palette(groups),
  segment_size = 300/nrow(df_optics)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggplot_optics_+3A_optics_obj">optics_obj</code></td>
<td>
<p>dbscan::optics object</p>
</td></tr>
<tr><td><code id="ggplot_optics_+3A_groups">groups</code></td>
<td>
<p>Optional vector defining groups of OPTICS observations</p>
</td></tr>
<tr><td><code id="ggplot_optics_+3A_colors">colors</code></td>
<td>
<p>If groups specified, vector of colors for each group</p>
</td></tr>
<tr><td><code id="ggplot_optics_+3A_segment_size">segment_size</code></td>
<td>
<p>Size for geom_segment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p><a href="#topic+opticskxi">opticskxi</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('multishapes')
optics_obj &lt;- dbscan::optics(multishapes[1:2])
ggplot_optics(optics_obj)
ggplot_optics(optics_obj,
  groups = opticskxi(optics_obj, n_xi = 5, pts = 30))
</code></pre>

<hr>
<h2 id='gtable_kxi_profiles'>Gtable OPTICS k-Xi distance profiles</h2><span id='topic+gtable_kxi_profiles'></span>

<h3>Description</h3>

<p>Plot OPTICS distance profiles of k-Xi clustering models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gtable_kxi_profiles(df_kxi, metric = "avg.silwidth", rank = 1:4, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gtable_kxi_profiles_+3A_df_kxi">df_kxi</code></td>
<td>
<p>Data frame returned by opticskxi_pipeline</p>
</td></tr>
<tr><td><code id="gtable_kxi_profiles_+3A_metric">metric</code></td>
<td>
<p>Metric to choose best clustering model</p>
</td></tr>
<tr><td><code id="gtable_kxi_profiles_+3A_rank">rank</code></td>
<td>
<p>Ranks of models to plot, ordered by decreasing model metric</p>
</td></tr>
<tr><td><code id="gtable_kxi_profiles_+3A_...">...</code></td>
<td>
<p>Passed to ggplot_kxi_profile</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><a href="#topic+opticskxi_pipeline">opticskxi_pipeline</a>
</p>

<hr>
<h2 id='hla'>The HLA data</h2><span id='topic+hla'></span>

<h3>Description</h3>

<p>This data set contains HLA markers DRB, DQA, DQB and phenotypes
of 271 Schizophrenia patients (y=1) and controls (y=0). Genotypes for 3 HLA
loci have prefixes name (e.g., &quot;DQB&quot;) and a suffix for each of two alleles 
(&quot;.a1&quot; and &quot;.a2&quot;).
</p>
<p>Imported from the gap package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hla)</code></pre>


<h3>Format</h3>

<p>A data frame containing 271 rows and 8 columns</p>


<h3>Source</h3>

<p>Dr Padraig Wright of Pfizer</p>

<hr>
<h2 id='m_psych_embeds'>A dataset containing the embeddings matrix of psychological related words</h2><span id='topic+m_psych_embeds'></span>

<h3>Description</h3>

<p>Data containing Glove embeddings of psychological related
words, useful for demonstrating the use of ensemble metrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("m_psych_embeds")
</code></pre>


<h3>Format</h3>

<p>A matrix with 831 words in rows and 100 embedding dimensions in
columns.
</p>


<h3>Details</h3>

<p>The dataset contains groups of related words among other irrelevant words.
</p>

<hr>
<h2 id='multishapes'>A dataset containing clusters of multiple shapes</h2><span id='topic+multishapes'></span>

<h3>Description</h3>

<p>Data containing clusters of any shapes. Useful for comparing
density-based clustering (DBSCAN) and standard partitioning methods such as
k-means clustering. Imported from the factoextra package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("multishapes")
</code></pre>


<h3>Format</h3>

<p>A data frame with 1100 observations on the following 3 variables. 
</p>

<dl>
<dt><code>x</code></dt><dd><p>a numeric vector containing the x coordinates of observations</p>
</dd>
<dt><code>y</code></dt><dd><p>a numeric vector containing the y coordinates of observations</p>
</dd>
<dt><code>shape</code></dt><dd><p>a numeric vector corresponding to the cluster number of each observations.</p>
</dd>
</dl>



<h3>Details</h3>

<p>The dataset contains 5 clusters and some outliers/noises.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('multishapes')
plot(multishapes[, 1], multishapes[, 2],
    col = multishapes[, 3], pch = 19, cex = 0.8)

</code></pre>

<hr>
<h2 id='nice_palette'>Nice palette</h2><span id='topic+nice_palette'></span>

<h3>Description</h3>

<p>Color palette
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nice_palette(groups, rainbow = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nice_palette_+3A_groups">groups</code></td>
<td>
<p>Vector, each unique value will get a color</p>
</td></tr>
<tr><td><code id="nice_palette_+3A_rainbow">rainbow</code></td>
<td>
<p>If TRUE, rainbow-like colors, else differentiate successive
values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of colors
</p>

<hr>
<h2 id='norm_inprod'>norm_inprod</h2><span id='topic+norm_inprod'></span>

<h3>Description</h3>

<p>Normalized inner product with transposed input matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>norm_inprod(m)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="norm_inprod_+3A_m">m</code></td>
<td>
<p>Numeric matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric matrix
</p>

<hr>
<h2 id='normalize'>Matrix normalization</h2><span id='topic+normalize'></span>

<h3>Description</h3>

<p>normalize matrix rows using given norm. Copied from text2vec
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize(m, norm = c("l1", "l2", "none"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="normalize_+3A_m">m</code></td>
<td>
<p><code>matrix</code> (sparse or dense).</p>
</td></tr>
<tr><td><code id="normalize_+3A_norm">norm</code></td>
<td>
<p><code>character</code> the method used to normalize term vectors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>normalized matrix
</p>

<hr>
<h2 id='opticskxi'>OPTICS k-Xi clustering algorithm</h2><span id='topic+opticskxi'></span>

<h3>Description</h3>

<p>For each largest distance differences on the OPTICS profile,
consecutive observations left and right on the OPTICS profile
(i.e. lower and higher OPTICS id) will be assigned to 2 different clusters
if their distance is below the distance of the edge point.
If above, observations are NA.
The pts parameter defines a minimum number of observations to form a
valley (i.e. cluster). If the number of observations in one valley is
smaller than pts, observations are set to NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opticskxi(
  optics_obj,
  n_xi,
  pts = optics_obj$minPts,
  max_loop = 50,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="opticskxi_+3A_optics_obj">optics_obj</code></td>
<td>
<p>Data frame returned by optics</p>
</td></tr>
<tr><td><code id="opticskxi_+3A_n_xi">n_xi</code></td>
<td>
<p>Number of clusters to define</p>
</td></tr>
<tr><td><code id="opticskxi_+3A_pts">pts</code></td>
<td>
<p>Minimum number of points per clusters</p>
</td></tr>
<tr><td><code id="opticskxi_+3A_max_loop">max_loop</code></td>
<td>
<p>Maximum iterations to find n_xi clusters</p>
</td></tr>
<tr><td><code id="opticskxi_+3A_verbose">verbose</code></td>
<td>
<p>Print the ids of the largest difference considered and
cluster information if they define one</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of clusters
</p>


<h3>See Also</h3>

<p><a href="#topic+opticskxi_pipeline">opticskxi_pipeline</a>, <a href="#topic+ggplot_optics">ggplot_optics</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('multishapes')
optics_shapes &lt;- dbscan::optics(multishapes[1:2])
kxi_shapes &lt;- opticskxi(optics_shapes, n_xi = 5, pts = 30)
ggplot_optics(optics_shapes, groups = kxi_shapes)
ggpairs(cbind(multishapes[1:2], kXi = kxi_shapes), group = 'kXi')
</code></pre>

<hr>
<h2 id='opticskxi_pipeline'>OPTICS k-Xi models comparison pipeline</h2><span id='topic+opticskxi_pipeline'></span>

<h3>Description</h3>

<p>Computes OPTICS k-Xi models based on a parameter grid, binds results in a
data frame, and computes distance based metrics for each model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opticskxi_pipeline(
  m_data,
  df_params = expand.grid(n_xi = 1:10, pts = c(20, 30, 40), dist = c("euclidean",
    "abscorrelation"), dim_red = c("identity", "PCA", "ICA"), n_dimred_comp = c(5, 10,
    20)),
  metrics_dist = c("euclidean", "cosine"),
  max_size_ratio = 1,
  n_min_clusters = 0,
  n_cores = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="opticskxi_pipeline_+3A_m_data">m_data</code></td>
<td>
<p>Data matrix</p>
</td></tr>
<tr><td><code id="opticskxi_pipeline_+3A_df_params">df_params</code></td>
<td>
<p>Parameter grid for the OPTICS k-Xi function call and
optional dimension reduction.
Required columns: n_xi, pts, dist.
Optonal columns: dim_red, n_dim_red.</p>
</td></tr>
<tr><td><code id="opticskxi_pipeline_+3A_metrics_dist">metrics_dist</code></td>
<td>
<p>Distance used for metrics, either euclidean or cosine.</p>
</td></tr>
<tr><td><code id="opticskxi_pipeline_+3A_max_size_ratio">max_size_ratio</code></td>
<td>
<p>Maximum size ratio of clusters.
E.g. for 0.8, if a cluster is larger than 80% of
points it will be removed.</p>
</td></tr>
<tr><td><code id="opticskxi_pipeline_+3A_n_min_clusters">n_min_clusters</code></td>
<td>
<p>Minimum number of clusters. Ignored if 0.</p>
</td></tr>
<tr><td><code id="opticskxi_pipeline_+3A_n_cores">n_cores</code></td>
<td>
<p>Number of cores</p>
</td></tr>
<tr><td><code id="opticskxi_pipeline_+3A_...">...</code></td>
<td>
<p>Passed to get_kxi_metrics</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Input parameter data frame with with results binded in columns
optics, clusters and metrics.
</p>


<h3>See Also</h3>

<p><a href="#topic+get_best_kxi">get_best_kxi</a>, <a href="#topic+ggplot_kxi_metrics">ggplot_kxi_metrics</a>,
<a href="#topic+gtable_kxi_profiles">gtable_kxi_profiles</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data('hla')
m_hla &lt;- hla[-c(1:2)] %&gt;% scale

df_params_hla &lt;- expand.grid(n_xi = 3:5, pts = c(20, 30),
  dist = c('manhattan', 'euclidean'))

df_kxi_hla &lt;- opticskxi_pipeline(m_hla, df_params_hla)

ggplot_kxi_metrics(df_kxi_hla, n = 8)
gtable_kxi_profiles(df_kxi_hla) %&gt;% plot

best_kxi_hla &lt;- get_best_kxi(df_kxi_hla, rank = 2)
clusters_hla &lt;- best_kxi_hla$clusters

fortify_pca(m_hla, sup_vars = data.frame(Clusters = clusters_hla)) %&gt;%
  ggpairs('Clusters', ellipses = TRUE, variables = TRUE)

</code></pre>

<hr>
<h2 id='print_vignette_table'>Print vignette table</h2><span id='topic+print_vignette_table'></span>

<h3>Description</h3>

<p>Print knitr::kable latex table with legend at bottom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_vignette_table(table_obj, label)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print_vignette_table_+3A_table_obj">table_obj</code></td>
<td>
<p>Table object</p>
</td></tr>
<tr><td><code id="print_vignette_table_+3A_label">label</code></td>
<td>
<p>Latex label</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None, side-effect prints a Latex table
</p>

<hr>
<h2 id='residuals_table'>Residuals table</h2><span id='topic+residuals_table'></span>

<h3>Description</h3>

<p>Bind contingency table and Pearson Chi-squared residuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>residuals_table(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="residuals_table_+3A_...">...</code></td>
<td>
<p>Passed to contingency_table and chisq.test</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix
</p>

<hr>
<h2 id='stddev_mean'>stddev_mean</h2><span id='topic+stddev_mean'></span>

<h3>Description</h3>

<p>Get mean of standard deviations of matrix columns
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stddev_mean(m)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stddev_mean_+3A_m">m</code></td>
<td>
<p>Numeric matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Mean of standard deviations of matrix columns
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
