<!DOCTYPE html><html lang="en"><head><title>Help for package optRF</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {optRF}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#estimate_numtrees'><p>Estimate the required number of trees</p></a></li>
<li><a href='#estimate_stability'><p>Estimate the stability of random forest</p></a></li>
<li><a href='#opt_importance'><p>Optimise random forest for estimation of variable importance</p></a></li>
<li><a href='#opt_prediction'><p>Optimise random forest for prediction</p></a></li>
<li><a href='#plot_stability'><p>Plot random forest stability</p></a></li>
<li><a href='#SNPdata'><p>Simulated data of wheat yield and genomic markers</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Optimising Random Forest Stability by Determining the Optimal
Number of Trees</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomas Martin Lange &lt;thomas.lange@uni-goettingen.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Calculating the stability of random forest with certain numbers of trees. The non-linear relationship between stability and numbers of trees is described using a logistic regression model and used to estimate the optimal number of trees.</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/tmlange/optRF/issues">https://github.com/tmlange/optRF/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/tmlange/optRF">https://github.com/tmlange/optRF</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>minpack.lm (&ge; 1.2-2), ranger (&ge; 0.12.0), irr (&ge; 0.82),
graphics, methods, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, knitr, rmarkdown, spelling, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-03 11:02:34 UTC; felix.heinrich</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas Martin Lange
    <a href="https://orcid.org/0000-0003-4351-7950"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre, aut],
  Felix Heinrich <a href="https://orcid.org/0000-0002-6093-8522"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-03 14:30:01 UTC</td>
</tr>
</table>
<hr>
<h2 id='estimate_numtrees'>Estimate the required number of trees</h2><span id='topic+estimate_numtrees'></span>

<h3>Description</h3>

<p>Estimate the number of trees required to achieve certain stability of random forest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_numtrees(
  optRF_object,
  measure = c("selection", "importance", "prediction"),
  for_stability = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimate_numtrees_+3A_optrf_object">optRF_object</code></td>
<td>
<p>An optRF_object, either the result from the <a href="#topic+opt_importance">opt_importance</a> or the <a href="#topic+opt_prediction">opt_prediction</a> function.</p>
</td></tr>
<tr><td><code id="estimate_numtrees_+3A_measure">measure</code></td>
<td>
<p>A character string indicating which stability measure is to be analysed. One of &quot;selection&quot; (default, analyses selection stability), &quot;prediction&quot; (analyses prediction stability) or &quot;importance&quot; (analyses variable importance stability).</p>
</td></tr>
<tr><td><code id="estimate_numtrees_+3A_for_stability">for_stability</code></td>
<td>
<p>Either a single stability value or a vector containing multiple stability values for which the number of trees should be estimated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame summarising the estimated stability and run time in seconds for the given num.trees values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(SNPdata)
set.seed(123)
result_optpred = opt_prediction(y = SNPdata[,1], X=SNPdata[,-1]) # optimise random forest
estimate_numtrees(result_optpred, measure="prediction", for_stability=0.95)

## End(Not run)

</code></pre>

<hr>
<h2 id='estimate_stability'>Estimate the stability of random forest</h2><span id='topic+estimate_stability'></span>

<h3>Description</h3>

<p>Estimate the stability of random forest with certain numbers of trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_stability(
  optRF_object,
  with_num.trees = c(1000, 5000, 10000, 50000, 1e+05)
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimate_stability_+3A_optrf_object">optRF_object</code></td>
<td>
<p>An optRF_object, either the result from the <a href="#topic+opt_importance">opt_importance</a> or the <a href="#topic+opt_prediction">opt_prediction</a> function.</p>
</td></tr>
<tr><td><code id="estimate_stability_+3A_with_num.trees">with_num.trees</code></td>
<td>
<p>Either a single num.trees value or a vector containing multiple num.trees values for which the stability should be estimated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame summarising the estimated stability and run time in seconds for the given num.trees values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(SNPdata)
set.seed(123)
result_optpred = opt_prediction(y = SNPdata[,1], X=SNPdata[,-1]) # optimise random forest
estimate_stability(result_optpred, with_num.trees=c(1000, 5000, 10000, 50000, 100000))

## End(Not run)

</code></pre>

<hr>
<h2 id='opt_importance'>Optimise random forest for estimation of variable importance</h2><span id='topic+opt_importance'></span>

<h3>Description</h3>

<p>Optimising random forest for estimating the importance of variables by calculating the variable importance stability with certain numbers of trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opt_importance(
  y,
  X,
  number.repetitions = 10,
  alpha = 0.05,
  num.trees_values = c(250, 500, 750, 1000, 2000),
  importance = c("permutation", "impurity", "impurity_corrected"),
  visualisation = c("none", "importance", "selection"),
  recommendation = c("importance", "selection", "none"),
  rec.thresh = 1e-06,
  round.recommendation = c("thousand", "hundred", "ten", "none"),
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="opt_importance_+3A_y">y</code></td>
<td>
<p>A vector containing the response variable.</p>
</td></tr>
<tr><td><code id="opt_importance_+3A_x">X</code></td>
<td>
<p>A data frame containing the explanatory variables. The number of rows must be equal to the number of elements in y.</p>
</td></tr>
<tr><td><code id="opt_importance_+3A_number.repetitions">number.repetitions</code></td>
<td>
<p>Number of repetitions of random forest to estimate the stability.</p>
</td></tr>
<tr><td><code id="opt_importance_+3A_alpha">alpha</code></td>
<td>
<p>The amount of most important variables to be selected based on their estimated variable importance. If &lt; 1, alpha will be considered the relative amount of variables in the data set.</p>
</td></tr>
<tr><td><code id="opt_importance_+3A_num.trees_values">num.trees_values</code></td>
<td>
<p>A vector containing the numbers of trees to be analysed. If not specified, 250, 500, 750, 1000, and 2000 trees will be analysed.</p>
</td></tr>
<tr><td><code id="opt_importance_+3A_importance">importance</code></td>
<td>
<p>Variable importance mode, one of &quot;permutation&quot; (default), &quot;impurity&quot; or &quot;impurity_corrected&quot;. The &quot;impurity&quot; measure is the Gini index for classification and the variance of the responses for regression.</p>
</td></tr>
<tr><td><code id="opt_importance_+3A_visualisation">visualisation</code></td>
<td>
<p>Can be set to &quot;importance&quot; to draw a plot of the variable importance stability or to &quot;selection&quot; to draw a plot of the selection stability for the numbers of trees to be analysed.</p>
</td></tr>
<tr><td><code id="opt_importance_+3A_recommendation">recommendation</code></td>
<td>
<p>If set to &quot;importance&quot; (default) or &quot;selection&quot;, a recommendation will be given based on optimised variable importance or selection stability. If set to be &quot;none&quot;, the function will analyse the stability of random forest with the inserted numbers of trees without giving a recommendation.</p>
</td></tr>
<tr><td><code id="opt_importance_+3A_rec.thresh">rec.thresh</code></td>
<td>
<p>If the number of trees leads to an increase of stability smaller or equal to the value specified, this number of trees will be recommended. Default is 1e-6.</p>
</td></tr>
<tr><td><code id="opt_importance_+3A_round.recommendation">round.recommendation</code></td>
<td>
<p>Setting to what number the recommended number of trees should be rounded to. Options: &quot;none&quot;, &quot;ten&quot;, &quot;hundred&quot;, &quot;thousand&quot; (default).</p>
</td></tr>
<tr><td><code id="opt_importance_+3A_verbose">verbose</code></td>
<td>
<p>Show computation status</p>
</td></tr>
<tr><td><code id="opt_importance_+3A_...">...</code></td>
<td>
<p>Any other argument from the ranger function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An opt_importance_object containing the recommended number of trees, based on which measure the recommendation was given (importance or selection), a matrix summarising the estimated stability and computation time of a random forest with the recommended numbers of trees, a matrix containing the calculated stability and computation time for the analysed numbers of trees, and the parameters used to model the relationship between stability and numbers of trees.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(SNPdata)
set.seed(123)
result_optimp = opt_importance(y = SNPdata[,1], X=SNPdata[,-1]) # optimise random forest
summary(result_optimp)

## End(Not run)

</code></pre>

<hr>
<h2 id='opt_prediction'>Optimise random forest for prediction</h2><span id='topic+opt_prediction'></span>

<h3>Description</h3>

<p>Optimising random forest predictions by calculating the prediction stability with certain numbers of trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opt_prediction(
  y,
  X,
  X_Test = NULL,
  number.repetitions = 10,
  alpha = 0.15,
  num.trees_values = c(250, 500, 750, 1000, 2000),
  visualisation = c("none", "prediction", "selection"),
  select_for = c("high", "low", "zero"),
  recommendation = c("prediction", "selection", "none"),
  rec.thresh = 1e-06,
  round.recommendation = c("thousand", "hundred", "ten", "none"),
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="opt_prediction_+3A_y">y</code></td>
<td>
<p>A vector containing the response variable in the training data set.</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_x">X</code></td>
<td>
<p>A data frame containing the explanatory variables in the training data set. The number of rows must be equal to the number of elements in y.</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_x_test">X_Test</code></td>
<td>
<p>A data frame containing the explanatory variables of the test data set. If not entered, the out of bag data will be used.</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_number.repetitions">number.repetitions</code></td>
<td>
<p>Number of repetitions of random forest to estimate the stability.</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_alpha">alpha</code></td>
<td>
<p>The number of best individuals to be selected in the test data set based on their predicted response values. If &lt; 1, alpha will be considered to be the relative amount of individuals in the test data set.</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_num.trees_values">num.trees_values</code></td>
<td>
<p>A vector containing the numbers of trees to be analysed. If not specified, 250, 500, 750, 1000, and 2000 trees will be analysed.</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_visualisation">visualisation</code></td>
<td>
<p>Can be set to &quot;prediction&quot; to draw a plot of the prediction stability or &quot;selection&quot; to draw a plot of the selection stability for the numbers of trees to be analysed.</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_select_for">select_for</code></td>
<td>
<p>What should be selected? In random forest classification, this must be set to a vector containing the values of the desired classes. In random forest regression, this can be set as &quot;high&quot; (default) to select the individuals with the highest predicted value, &quot;low&quot; to select the individuals with the lowest predicted value, or &quot;zero&quot; to select the individuals which predicted value is closest to zero.</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_recommendation">recommendation</code></td>
<td>
<p>If set to &quot;prediction&quot; (default) or &quot;selection&quot;, a recommendation will be given based on optimised prediction or selection stability. If set to be &quot;none&quot;, the function will analyse the stability of random forest with the inserted numbers of trees without giving a recommendation.</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_rec.thresh">rec.thresh</code></td>
<td>
<p>If the number of trees leads to an increase of stability smaller or equal to the value specified, this number of trees will be recommended. Default is 1e-6.</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_round.recommendation">round.recommendation</code></td>
<td>
<p>Setting to what number the recommended number of trees should be rounded to. Options: &quot;none&quot;, &quot;ten&quot;, &quot;hundred&quot;, &quot;thousand&quot; (default).</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_verbose">verbose</code></td>
<td>
<p>Show computation status</p>
</td></tr>
<tr><td><code id="opt_prediction_+3A_...">...</code></td>
<td>
<p>Any other argument from the ranger function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An opt_prediction_object containing the recommended number of trees, based on which measure the recommendation was given (prediction or selection), a matrix summarising the estimated stability and computation time of a random forest with the recommended numbers of trees, a matrix containing the calculated stability and computation time for the analysed numbers of trees, and the parameters used to model the relationship between stability and numbers of trees.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(SNPdata)
set.seed(123)
result_optpred = opt_prediction(y = SNPdata[,1], X=SNPdata[,-1]) # optimise random forest
summary(result_optpred)

## End(Not run)

</code></pre>

<hr>
<h2 id='plot_stability'>Plot random forest stability</h2><span id='topic+plot_stability'></span>

<h3>Description</h3>

<p>Plot the estimated stability of random forest against certain numbers of trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_stability(
  optRF_object,
  measure = c("selection", "importance", "prediction"),
  from = 0,
  to = 1e+05,
  add.recommendation = TRUE,
  add = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_stability_+3A_optrf_object">optRF_object</code></td>
<td>
<p>An optRF_object, either the result from the <a href="#topic+opt_importance">opt_importance</a> or the <a href="#topic+opt_prediction">opt_prediction</a> function.</p>
</td></tr>
<tr><td><code id="plot_stability_+3A_measure">measure</code></td>
<td>
<p>A character string indicating which stability measure is to be plotted. One of &quot;selection&quot; (default, visualises selection stability), &quot;prediction&quot; (visualises prediction stability) or &quot;importance&quot; (visualises variable importance stability).</p>
</td></tr>
<tr><td><code id="plot_stability_+3A_from">from</code></td>
<td>
<p>Smallest num.trees value to be plotted.</p>
</td></tr>
<tr><td><code id="plot_stability_+3A_to">to</code></td>
<td>
<p>Greatest num.trees value to be plotted.</p>
</td></tr>
<tr><td><code id="plot_stability_+3A_add.recommendation">add.recommendation</code></td>
<td>
<p>When set as TRUE, if a recommendation was stated within the opt_prediction or opt_importance function, the recommended num.trees value as well as the expected random forest stability will be highlighted in the graph</p>
</td></tr>
<tr><td><code id="plot_stability_+3A_add">add</code></td>
<td>
<p>If FALSE, a new plot will be created, if TRUE, the graph will be added to an existing plot.</p>
</td></tr>
<tr><td><code id="plot_stability_+3A_...">...</code></td>
<td>
<p>Any other arguments from the plot function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot showing the estimated stability of random forest for the given num.trees values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(SNPdata)
set.seed(123)
result_optpred = opt_prediction(y = SNPdata[,1], X=SNPdata[,-1]) # optimise random forest
plot_stability(result_optpred, measure = "prediction", add.recommendation = TRUE, add=FALSE)
plot_stability(result_optpred, measure = "selection",  add.recommendation = FALSE, add=TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='SNPdata'>Simulated data of wheat yield and genomic markers</h2><span id='topic+SNPdata'></span>

<h3>Description</h3>

<p>Data set containing simulated data of wheat yield in g/m^2 of 250 wheat lines and 5,000 SNP markers
being coded as 0 for homozygous form of the major allele and 2 for homozygous form of the
minor allele.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SNPdata)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"data.frame"</code>
</p>

<dl>
<dt>yield</dt><dd><p>Simulated wheat yield in g/m^2</p>
</dd>
<dt>SNP_0001 to SNP_5000</dt><dd><p>Simulated values for 5,000 single nucleotide polymorphism (SNP) markers</p>
</dd>
</dl>



<h3>References</h3>

<p>This artificial data set was created for the optRF package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(SNPdata)
SNPdata[1:5,1:5]

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
