<!DOCTYPE html><html lang="en"><head><title>Help for package ordinalgmifs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ordinalgmifs}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ordinalgmifs-package'>
<p>Ordinal Response Regression for High-Dimensional Data</p></a></li>
<li><a href='#coef.ordinalgmifs'>
<p>Extract Model Coefficients</p></a></li>
<li><a href='#eyedisease'>
<p>Eye Disease Risk Factors</p></a></li>
<li><a href='#hccframe'>
<p>Liver Cancer Methylation Data</p></a></li>
<li><a href='#ordinalgmifs'>
<p>Ordinal Generalized Monotone Incremental Forward Stagewise Regression</p></a></li>
<li><a href='#ordinalgmifs-internal'>
<p>Functions Called by ordinalgmifs Functions, Not by the User</p></a></li>
<li><a href='#plot.ordinalgmifs'>
<p>Plot Solution Path for Ordinal GMIFS Fitted Model.</p></a></li>
<li><a href='#predict.ordinalgmifs'>
<p>Predicted Probabilities and Class for Ordinal GMIFS Fit.</p></a></li>
<li><a href='#print.ordinalgmifs'>
<p>Print the Contents of an Ordinal GMIFS Fitted Object.</p></a></li>
<li><a href='#summary.ordinalgmifs'>
<p>Summarize an Ordinal GMIFS Object.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.0.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-05-01</td>
</tr>
<tr>
<td>Title:</td>
<td>Ordinal Regression for High-Dimensional Data</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0), survival</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a function for fitting cumulative link, adjacent category, forward and backward continuation ratio, and stereotype ordinal response models when the number of parameters exceeds the sample size, using the the generalized monotone incremental forward stagewise method.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods</td>
</tr>
<tr>
<td>BuildResaveData:</td>
<td>best</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++11</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>BuildVignettes:</td>
<td>TRUE</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-04 12:01:04 UTC; archer.43</td>
</tr>
<tr>
<td>Author:</td>
<td>Kellie J. Archer <a href="https://orcid.org/0000-0003-1555-5781"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Jiayi Hou [aut],
  Qing Zhou [aut],
  Kyle Ferber [aut],
  John G. Layne [com, ctr],
  Amanda Gentry [rev]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kellie J. Archer &lt;archer.43@osu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-05 07:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='ordinalgmifs-package'>
Ordinal Response Regression for High-Dimensional Data
</h2><span id='topic+ordinalgmifs-package'></span>

<h3>Description</h3>

<p>This package provides a function, ordinalgmifs, for fitting cumulative link, adjacent category, forward and backward continuation ratio, and stereotype ordinal response models when the number of parameters exceeds the sample size, using the the generalized monotone incremental forward stagewise method.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> ordinalgmifs</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.8</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2023-05-01</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Ordinal Regression for High-Dimensional Data</td>
</tr>
<tr>
 <td style="text-align: left;">
Authors@R: </td><td style="text-align: left;"> c(person(c("Kellie", "J."), "Archer", email
          = "archer.43@osu.edu", role =
            c("aut", "cre"), comment = c(ORCID = "0000-0003-1555-5781")), 
   person("Jiayi", "Hou", role = "aut"), 
   person("Qing", "Zhou", role = "aut"), 
   person("Kyle","Ferber", role = "aut"), 
   person(c("John", "G."), "Layne", role = c("com","ctr")), 
   person("Amanda", "Gentry", role = "rev") )</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R (&gt;= 4.2.0), survival</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Provides a function for fitting cumulative link, adjacent category, forward and backward continuation ratio, and stereotype ordinal response models when the number of parameters exceeds the sample size, using the the generalized monotone incremental forward stagewise method.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> methods</td>
</tr>
<tr>
 <td style="text-align: left;">
BuildResaveData: </td><td style="text-align: left;"> best</td>
</tr>
<tr>
 <td style="text-align: left;">
SystemRequirements: </td><td style="text-align: left;"> C++11</td>
</tr>
<tr>
 <td style="text-align: left;">
NeedsCompilation: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
BuildVignettes: </td><td style="text-align: left;"> TRUE</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyData: </td><td style="text-align: left;"> true</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Kellie J. Archer [aut, cre] (&lt;https://orcid.org/0000-0003-1555-5781&gt;),
  Jiayi Hou [aut],
  Qing Zhou [aut],
  Kyle Ferber [aut],
  John G. Layne [com, ctr],
  Amanda Gentry [rev]</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Kellie J. Archer &lt;archer.43@osu.edu&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<p>Index of help topics:
</p>
<pre>
coef.ordinalgmifs       Extract Model Coefficients
eyedisease              Eye Disease Risk Factors
hccframe                Liver Cancer Methylation Data
ordinalgmifs            Ordinal Generalized Monotone Incremental
                        Forward Stagewise Regression
ordinalgmifs-package    Ordinal Response Regression for
                        High-Dimensional Data
plot.ordinalgmifs       Plot Solution Path for Ordinal GMIFS Fitted
                        Model.
predict.ordinalgmifs    Predicted Probabilities and Class for Ordinal
                        GMIFS Fit.
print.ordinalgmifs      Print the Contents of an Ordinal GMIFS Fitted
                        Object.
summary.ordinalgmifs    Summarize an Ordinal GMIFS Object.
</pre>
<p>This package contains generic methods (coef, plot, predict, print, summary) that can be invoked for an object fitted using ordinalgmifs. 
</p>


<h3>Author(s)</h3>

<p>NA Kellie J. Archer, Jiayi Hou, Qing Zhou, Kyle Ferber, John G. Layne, Amanda Gentry
</p>
<p>Maintainer: NA Kellie J. Archer &lt;archer.43@osu.edu&gt;
</p>


<h3>References</h3>

<p>Hastie T., Taylor J., Tibshirani R., and Walther G. (2007) Forward stagewise regression and the monotone lasso. <em>Electronic Journal of Statistics</em>, 1, 1-29.
</p>


<h3>See Also</h3>

<p>See Also <code><a href="#topic+ordinalgmifs">ordinalgmifs</a></code>. For models where no predictor is penalized see <code><a href="VGAM.html#topic+VGAM">vglm</a></code> 
</p>

<hr>
<h2 id='coef.ordinalgmifs'>
Extract Model Coefficients
</h2><span id='topic+coef.ordinalgmifs'></span>

<h3>Description</h3>

<p><code>coef.ordinalgmifs</code> is a generic function which extracts the model coefficients from a fitted model object fit using <code>ordinalgmifs</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ordinalgmifs'
coef(object, model.select = "AIC", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coef.ordinalgmifs_+3A_object">object</code></td>
<td>

<p>an <code>ordinalgmifs</code> object.
</p>
</td></tr>
<tr><td><code id="coef.ordinalgmifs_+3A_model.select">model.select</code></td>
<td>

<p>when <code>x</code> is specified any model along the solution path can be selected. The default is <code>model.select="AIC"</code> which extracts the coefficients from the model having the lowest AIC. Other options are <code>model.select="BIC"</code> or any numeric value from the solution path.
</p>
</td></tr>
<tr><td><code id="coef.ordinalgmifs_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Coefficients extracted from the model object.
</p>


<h3>Author(s)</h3>

<p>Kellie J. Archer
</p>


<h3>References</h3>

<p>Hastie T., Taylor J., Tibshirani R., and Walther G. (2007) Forward stagewise regression and the monotone lasso. <em>Electronic Journal of Statistics</em>, 1, 1-29.
</p>


<h3>See Also</h3>

<p>See Also <code><a href="#topic+ordinalgmifs">ordinalgmifs</a></code>, <code><a href="#topic+summary.ordinalgmifs">summary.ordinalgmifs</a></code>, <code><a href="#topic+plot.ordinalgmifs">plot.ordinalgmifs</a></code>, <code><a href="#topic+predict.ordinalgmifs">predict.ordinalgmifs</a></code>
</p>

<hr>
<h2 id='eyedisease'>
Eye Disease Risk Factors
</h2><span id='topic+eyedisease'></span>

<h3>Description</h3>

<p>Eye Disease Risk Factors data from Section 9.1 of Agresti's Analysis of Ordinal Categorical Data. The primary data are from the Wisconsin Epidemiological Study of Diabetic Retinopathy. The primary outcome is severity of retinopathy which was measured in the left and right eye of every subject.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eyedisease)</code></pre>


<h3>Format</h3>

<p>A data frame with 720 observations on the following 19 variables.
</p>

<dl>
<dt><code>rme</code></dt><dd><p>right eye macular oedema (absent = 0, present = 1)</p>
</dd>
<dt><code>lme</code></dt><dd><p>left eye macular oedema (absent = 0, present = 1)</p>
</dd>
<dt><code>rre</code></dt><dd><p>right eye refraction index</p>
</dd>
<dt><code>lre</code></dt><dd><p>left eye refraction index</p>
</dd>
<dt><code>riop</code></dt><dd><p>right eye intraocular eye pressure</p>
</dd>
<dt><code>liop</code></dt><dd><p>left eye intraocular eye pressure</p>
</dd>
<dt><code>age</code></dt><dd><p>age</p>
</dd>
<dt><code>diab</code></dt><dd><p>duration of diabetes (in years)</p>
</dd>
<dt><code>gh</code></dt><dd><p>glycosylated haemoglobin level</p>
</dd>
<dt><code>sbp</code></dt><dd><p>systolic blood pressure</p>
</dd>
<dt><code>dbp</code></dt><dd><p>diastolic blood pressure</p>
</dd>
<dt><code>bmi</code></dt><dd><p>body mass index</p>
</dd>
<dt><code>pr</code></dt><dd><p>pulse rate?</p>
</dd>
<dt><code>sex</code></dt><dd><p>gender (male=1, female=2)</p>
</dd>
<dt><code>prot</code></dt><dd><p>proteinuria (absent = 0, present = 1)</p>
</dd>
<dt><code>dose</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rerl</code></dt><dd><p>right eye severity of retinopathy, an ordered factor with levels <code>None</code> &lt; <code>Mild</code> &lt; <code>Moderate</code> &lt; <code>Proliferative</code></p>
</dd>
<dt><code>lerl</code></dt><dd><p>left eye severity of retinopathy, an ordered factor with levels <code>None</code> &lt; <code>Mild</code> &lt; <code>Moderate</code> &lt; <code>Proliferative</code></p>
</dd>
<dt><code>id</code></dt><dd><p>subject identifier</p>
</dd>
</dl>



<h3>References</h3>

<p>R. Klein and B.E.K. Klein and S.E. Moss and M.D. Davis and D.L. DeMets. (1984) The Wisconsin Epidemiologic Study of Diabetic Retinopathy II. Prevalence and risk of diabetic retinopathy when age at diagnosis is less than 30 years. <em>Archives of Opthalmology</em> 101, 520-526.
</p>
<p>J. Williamson and K. Kim. (1996) A global odds ratio regression model for bivariate ordered categorical data from opthalmologic studies. <em>Statistics in Medicine</em> 15: 1507-1518.
</p>
<p>A. Agresti. (2010) <em>Analysis of Ordered Categorical Data</em>, Second Edition. Wiley. Hoboken, NJ. 
</p>


<h3>See Also</h3>

<p>See Also as <code> <a href="#topic+ordinalgmifs">ordinalgmifs</a> </code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(eyedisease)
</code></pre>

<hr>
<h2 id='hccframe'>
Liver Cancer Methylation Data
</h2><span id='topic+hccframe'></span>

<h3>Description</h3>

<p>These data are a subset of subjects and CpG sites reported in the original paper where liver samples were assayed using the Illumina GoldenGate Methylation BeadArray Cancer Panel I. Technical replicate samples
were removed to ensure all samples were independent. The matched cirrhotic samples from subjects with hepatocellular carcinoma (HCC, labeled Tumor) were also excluded. Therefore methylation levels in liver tissue are provided for independent subjects whose liver was Normal (N=20),
cirrhotic but not having HCC (N=16, Cirrhosis non-HCC), and HCC (N=20, Tumor). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hccframe)</code></pre>


<h3>Format</h3>

<p>A data frame with 56 observations on the following 46 variables.
</p>

<dl>
<dt><code>group</code></dt><dd><p>an ordered factor with levels <code>Normal</code> &lt; <code>Cirrhosis non-HCC</code> &lt; <code>Tumor</code></p>
</dd>
<dt><code>CDKN2B_seq_50_S294_F</code></dt><dd><p>a numeric vector representing a  CpG site proportion methylation for CDKN2B</p>
</dd>
<dt><code>DDIT3_P1313_R</code></dt><dd><p>a numeric vector representing a  CpG site proportion methylation for DDIT3</p>
</dd>
<dt><code>ERN1_P809_R</code></dt><dd><p>a numeric vector representing a  CpG site proportion methylation for ERN1</p>
</dd>
<dt><code>GML_E144_F</code></dt><dd><p>a numeric vector representing a  CpG site proportion methylation for GML</p>
</dd>
<dt><code>HDAC9_P137_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for HDAC9</p>
</dd>
<dt><code>HLA.DPA1_P205_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for HLA.DPA1</p>
</dd>
<dt><code>HOXB2_P488_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for HOXB2</p>
</dd>
<dt><code>IL16_P226_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for IL16</p>
</dd>
<dt><code>IL16_P93_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for IL16</p>
</dd>
<dt><code>IL8_P83_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for IL8</p>
</dd>
<dt><code>MPO_E302_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for MPO</p>
</dd>
<dt><code>MPO_P883_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for MPO</p>
</dd>
<dt><code>PADI4_P1158_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for PADI4</p>
</dd>
<dt><code>SOX17_P287_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for SOX17</p>
</dd>
<dt><code>TJP2_P518_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for TJP2</p>
</dd>
<dt><code>WRN_E57_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for WRN</p>
</dd>
<dt><code>CRIP1_P874_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for CRIP1</p>
</dd>
<dt><code>SLC22A3_P634_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for SLC22A3</p>
</dd>
<dt><code>CCNA1_P216_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for CCNA1</p>
</dd>
<dt><code>SEPT9_P374_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for SEPT9</p>
</dd>
<dt><code>ITGA2_E120_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for ITGA2</p>
</dd>
<dt><code>ITGA6_P718_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for ITGA6</p>
</dd>
<dt><code>HGF_P1293_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for HGF</p>
</dd>
<dt><code>DLG3_E340_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for DLG3</p>
</dd>
<dt><code>APP_E8_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for APP</p>
</dd>
<dt><code>SFTPB_P689_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for SFTPB</p>
</dd>
<dt><code>PENK_P447_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for PENK</p>
</dd>
<dt><code>COMT_E401_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for COMT</p>
</dd>
<dt><code>NOTCH1_E452_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for NOTCH1</p>
</dd>
<dt><code>EPHA8_P456_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for EPHA8</p>
</dd>
<dt><code>WT1_P853_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for WT1</p>
</dd>
<dt><code>KLK10_P268_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for KLK10</p>
</dd>
<dt><code>PCDH1_P264_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for PCDH1</p>
</dd>
<dt><code>TDGF1_P428_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for TDGF1</p>
</dd>
<dt><code>EFNB3_P442_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for EFNB3</p>
</dd>
<dt><code>MMP19_P306_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for MMP19</p>
</dd>
<dt><code>FGFR2_P460_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for FGFR2</p>
</dd>
<dt><code>RAF1_P330_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for RAF1</p>
</dd>
<dt><code>BMPR2_E435_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for BMPR2</p>
</dd>
<dt><code>GRB10_P496_R</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for GRB10</p>
</dd>
<dt><code>CTSH_P238_F</code></dt><dd><p>a numeric vector representing a  CpG site  proportion methylation for CTSH</p>
</dd>
<dt><code>SLC6A8_seq_28_S227_F</code></dt><dd><p>a numeric vector representing a  CpG site proportion methylation for SLC6A8</p>
</dd>
<dt><code>PLXDC1_P236_F</code></dt><dd><p>a numeric vector representing a  CpG site proportion methylation for PLXDC1</p>
</dd>
<dt><code>TFE3_P421_F</code></dt><dd><p>a numeric vector representing a  CpG site proportion methylation for TFE3</p>
</dd>
<dt><code>TSG101_P139_R</code></dt><dd><p>a numeric vector representing a  CpG site proportion methylation for TSG101</p>
</dd>
</dl>



<h3>Source</h3>

<p>The full dataset is available as GSE18081 from Gene Expression Omnibus at 
<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE18081">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE18081</a>
</p>


<h3>References</h3>

<p>Archer KJ, Mas VR, Maluf DG, Fisher RA. High-throughput assessment of CpG site methylation for distinguishing between HCV-cirrhosis and HCV-associated hepatocellular carcinoma. Molecular Genetics and Genomics, 283(4): 341-349, 2010.
</p>


<h3>See Also</h3>

<p>See Also as <code> <a href="#topic+ordinalgmifs">ordinalgmifs</a> </code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hccframe)
</code></pre>

<hr>
<h2 id='ordinalgmifs'>
Ordinal Generalized Monotone Incremental Forward Stagewise Regression
</h2><span id='topic+ordinalgmifs'></span>

<h3>Description</h3>

<p>This function can fit a cumulative link, adjacent category, forward and backward continuation ratio, and stereotype ordinal response model when the number of parameters exceeds the sample size, using the the generalized monotone incremental forward stagewise method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ordinalgmifs(formula, data, x = NULL, subset, epsilon = 0.001, tol = 1e-05, 
	scale = TRUE, probability.model = "Cumulative", link = "logit", 
	verbose=FALSE, assumption=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ordinalgmifs_+3A_formula">formula</code></td>
<td>
<p>an object of class &quot;<code>formula</code>&quot; (or one that can be coerced to that class): a symbolic description of the model to be fitted. The left side of the formula is the ordinal outcome while the variables on the right side of the formula are the covariates that are not included in the penalization process. Note that if all variables in the model are to be penalized, an intercept only model formula should be specified.</p>
</td></tr>
<tr><td><code id="ordinalgmifs_+3A_data">data</code></td>
<td>
<p>an optional data frame, list or environment (or object coercible by <code>as.data.frame</code> to a data frame) containing the variables in the model. </p>
</td></tr>
<tr><td><code id="ordinalgmifs_+3A_x">x</code></td>
<td>
<p>an optional matrix of predictors that are to be penalized in the model fitting process.</p>
</td></tr>
<tr><td><code id="ordinalgmifs_+3A_subset">subset</code></td>
<td>
<p>an optional vector specifying a subset of observations to be used in the fitting process.</p>
</td></tr>
<tr><td><code id="ordinalgmifs_+3A_epsilon">epsilon</code></td>
<td>
<p>small incremental amount used to update a coefficient at a given step.</p>
</td></tr>
<tr><td><code id="ordinalgmifs_+3A_tol">tol</code></td>
<td>
<p>the iterative process stops when the difference between successive log-likelihoods is less than this specified level of tolerance.</p>
</td></tr>
<tr><td><code id="ordinalgmifs_+3A_scale">scale</code></td>
<td>
<p>logical, if TRUE the penalized predictors are centered and scaled.</p>
</td></tr>
<tr><td><code id="ordinalgmifs_+3A_probability.model">probability.model</code></td>
<td>
<p>the type of ordinal response model to be fit. Can be <code>"Cumulative"</code>, <code>"AdjCategory"</code>, <code>"ForwardCR"</code>, <code>"BackwardCR"</code>, or <code>"Stereotype"</code></p>
</td></tr>
<tr><td><code id="ordinalgmifs_+3A_link">link</code></td>
<td>
<p>the link function used. Allowable links for <code>"Cumulative"</code>, <code>"ForwardCR"</code>, and <code>"BackwardCR"</code> are <code>"logit"</code>, <code>"probit"</code>, and <code>"cloglog"</code>. For an <code>"AdjCategory"</code> model only a <code>"loge"</code> link is allowed; for a <code>"Stereotype"</code> model only a <code>"logit"</code> link is allowed.</p>
</td></tr>
<tr><td><code id="ordinalgmifs_+3A_verbose">verbose</code></td>
<td>
<p>logical, if TRUE the step number is printed to the console (default is FALSE).</p>
</td></tr>
<tr><td><code id="ordinalgmifs_+3A_assumption">assumption</code></td>
<td>
<p>integer, only use with <code>probability.model = "ForwardCR"</code> and <code>link = "cloglog"</code> to denote the assumption to use for discrete censored survival modeling. If <code>assumption = 1</code>, assume the observation was censored at the end of the discrete time interval in which the censoring occurred; if <code>assumption = 2</code>, assume the observation was censored at the beginning of the interval in which censoring occurred; if <code>assumption = 3</code>, assume constant hazard rate within the interval in which the censoring occurred; if no censoring occurs, do not specify a value for <code>assumption</code>.</p>
</td></tr>
<tr><td><code id="ordinalgmifs_+3A_...">...</code></td>
<td>
<p>additional arguments </p>
</td></tr>
</table>


<h3>Details</h3>

<p>A model specified as <code>response~terms, x=penalized.terms</code> where <code>response</code> is the ordinal response vector and <code>terms</code> is the series of variables in the model that are not to be penalized and <code>x</code> is a matrix of variables that are to be penalized. For example, <code>terms</code> may include the variables age and gender while <code>x</code> includes hundreds to thousands of features from a high-throughput genomic experiment. In the event that no baseline demographic/clinical characteristics/subject level variables are available or needed in <code>terms</code> (all variables are to be penalized) then the model is specified as <code>response~1, x=penalized.terms</code>. 
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>AIC</code></td>
<td>
<p>a vector of AIC values for each step (if <code>x</code> is specified).</p>
</td></tr>
<tr><td><code>BIC</code></td>
<td>
<p>a vector of BIC values for each step (if <code>x</code> is specified).</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>the ordinal threshold estimates for the fitted model.</p>
</td></tr>
<tr><td><code>theta</code></td>
<td>
<p>the coefficient estimates for the unpenalized variables (if <code>terms</code> are specified on the right hand side of the model formula).</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>the coefficient estimates for the penalized variables (if <code>x</code> is specified in the model).</p>
</td></tr>
<tr><td><code>phi</code></td>
<td>
<p>the scaling coefficient estimates (if a <code>"Stereotype"</code> logit model is fit).</p>
</td></tr>
<tr><td><code>logLik</code></td>
<td>
<p>a vector of log-likelihood values for each step(if <code>terms</code> are specified on the right hand side of the model formula).</p>
</td></tr>
<tr><td><code>link</code></td>
<td>
<p>the link function used in the model fit.</p>
</td></tr>
<tr><td><code>model.select</code></td>
<td>
<p>the step at which the minimum AIC was observed (if <code>terms</code> are specified on the right hand side of the model formula).</p>
</td></tr>
<tr><td><code>probability.model</code></td>
<td>
<p>the model fit.</p>
</td></tr>
<tr><td><code>scale</code></td>
<td>
<p>logical indicating whether penalized variables were centered and scaled.</p>
</td></tr>
<tr><td><code>w</code></td>
<td>
<p>the unpenalized variables in the model (if any).</p>
</td></tr>
<tr><td><code>x</code></td>
<td>
<p>the penalized variables in the model (if any).</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>the ordinal response.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kellie J. Archer, Jiayi Hou, Qing Zhou, Kyle Ferber, John G. Layne, Amanda Gentry
</p>


<h3>References</h3>

<p>Hastie T., Taylor J., Tibshirani R., and Walther G. (2007) Forward stagewise regression and the monotone lasso. <em>Electronic Journal of Statistics</em>, 1, 1-29.
</p>


<h3>See Also</h3>

<p>See Also <code><a href="#topic+coef.ordinalgmifs">coef.ordinalgmifs</a></code>, <code><a href="#topic+summary.ordinalgmifs">summary.ordinalgmifs</a></code>, <code><a href="#topic+plot.ordinalgmifs">plot.ordinalgmifs</a></code>, <code><a href="#topic+predict.ordinalgmifs">predict.ordinalgmifs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hccframe)
# To minimize processing time, MPO_E302_R is coerced into the model and only a subset of 
# two CpG sites (DDIT3_P1313_R and HDAC9_P137_R) are included as penalized covariates
# in this demonstration, and epsilon is set to 0.01
hcc.fit &lt;- ordinalgmifs(group ~ MPO_E302_R, x = c("DDIT3_P1313_R", "HDAC9_P137_R"), 
	data = hccframe, epsilon = 0.01)
coef(hcc.fit)
summary(hcc.fit)
phat &lt;- predict(hcc.fit)
head(phat$predicted)
table(phat$class, hccframe$group)
</code></pre>

<hr>
<h2 id='ordinalgmifs-internal'>
Functions Called by ordinalgmifs Functions, Not by the User
</h2><span id='topic+ordinalgmifs-internal'></span><span id='topic+du.adjcat'></span><span id='topic+du.bcr'></span><span id='topic+du.cum'></span><span id='topic+du.fcr'></span><span id='topic+du.stereo'></span><span id='topic+fn.acat'></span><span id='topic+fn.bcr'></span><span id='topic+fn.cum'></span><span id='topic+fn.fcr'></span><span id='topic+fn.stereo'></span><span id='topic+G'></span><span id='topic+gradient'></span>

<h3>Description</h3>

<p>These functions are called my other <code>ordinalgmifs</code> functions and are not intended to be directly called by the user.
</p>


<h3>Details</h3>

<p>The <code>du.adjcat</code>, <code>du.bcr</code>, <code>du.cum</code>, <code>du.fcr</code>, and <code>du.stereo</code> functions calculate the derivatives at the current step for the adjacent category, backward CR, cumulative link, forward CR, and stereotype logit models, respectively, are used to identify which penalized parameter is updated. The <code>fn.acat</code>, <code>fn.bcr</code>, <code>fn.cum</code>, <code>fn.fcr</code>, and <code>fn.stereo</code> are the log-likelihood functions for the adjacent category, backward CR, cumulative link, forward CR, and stereotype logit models, respectively, are used to estimate the thresholds and non-penalized subset parameters (if included) at each step of the algorithm.
The <code>G</code> function returns the probability for the indicated link function. The <code>gradient</code> function returns the gradient of the log-likelihood for the cumulative link models and is used for the cumulative link constrained optimization.
</p>


<h3>Value</h3>

<p>these functions are called for intermediate results used 
internally by user-invoked functions</p>


<h3>Author(s)</h3>

<p>Kellie J. Archer, <a href="mailto:archer.43@osu.edu">archer.43@osu.edu</a>
</p>


<h3>See Also</h3>

<p>See Also as <code> <a href="#topic+ordinalgmifs">ordinalgmifs</a> </code>
</p>

<hr>
<h2 id='plot.ordinalgmifs'>
Plot Solution Path for Ordinal GMIFS Fitted Model.
</h2><span id='topic+plot.ordinalgmifs'></span>

<h3>Description</h3>

<p>This function plots either the coefficient path, the AIC, or the log-likelihood for a fitted <code>ordinalgmifs</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ordinalgmifs'
plot(x, type = "trace", xlab=NULL, ylab=NULL, main=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.ordinalgmifs_+3A_x">x</code></td>
<td>

<p>an <code>ordinalgmifs</code> object.
</p>
</td></tr>
<tr><td><code id="plot.ordinalgmifs_+3A_type">type</code></td>
<td>

<p>default is <code>"trace"</code> which plots the coefficient path for the fitted object. Also available are <code>"AIC"</code>, <code>"BIC"</code>, and <code>"logLik"</code>.
</p>
</td></tr>
<tr><td><code id="plot.ordinalgmifs_+3A_xlab">xlab</code></td>
<td>

<p>a default x-axis label will be used which can be changed by specifying a user-defined x-axis label.
</p>
</td></tr>
<tr><td><code id="plot.ordinalgmifs_+3A_ylab">ylab</code></td>
<td>

<p>a default y-axis label will be used which can be changed by specifying a user-defined y-axis label.
</p>
</td></tr>
<tr><td><code id="plot.ordinalgmifs_+3A_main">main</code></td>
<td>

<p>a default main title will be used which can be changed by specifying a user-defined main title.
</p>
</td></tr>
<tr><td><code id="plot.ordinalgmifs_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects</p>


<h3>Author(s)</h3>

<p>Kellie J. Archer
</p>


<h3>See Also</h3>

<p>See Also <code><a href="#topic+ordinalgmifs">ordinalgmifs</a></code>, <code><a href="#topic+coef.ordinalgmifs">coef.ordinalgmifs</a></code>, <code><a href="#topic+summary.ordinalgmifs">summary.ordinalgmifs</a></code>, <code><a href="#topic+predict.ordinalgmifs">predict.ordinalgmifs</a></code>
</p>

<hr>
<h2 id='predict.ordinalgmifs'>
Predicted Probabilities and Class for Ordinal GMIFS Fit.
</h2><span id='topic+predict.ordinalgmifs'></span><span id='topic+fitted.ordinalgmifs'></span>

<h3>Description</h3>

<p>This function returns a list the includes the predicted probabilities as well as the predicted class for an <code>ordinalgmifs</code> fitted object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ordinalgmifs'
predict(object, neww = NULL, newdata, newx = NULL, model.select = "AIC", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.ordinalgmifs_+3A_object">object</code></td>
<td>
<p>an <code>ordinalgmifs</code> fitted object.</p>
</td></tr>
<tr><td><code id="predict.ordinalgmifs_+3A_neww">neww</code></td>
<td>
<p>an optional formula that includes the unpenalized variables to use for predicting the response. If omitted, the training data are used.</p>
</td></tr>
<tr><td><code id="predict.ordinalgmifs_+3A_newdata">newdata</code></td>
<td>
<p>an optional data.frame that minimally includes the unpenalized variables to use for predicting the response. If omitted, the training data are used.</p>
</td></tr>
<tr><td><code id="predict.ordinalgmifs_+3A_newx">newx</code></td>
<td>
<p>an optional matrix of penalized variables to use for predicting the response. If omitted, the training data are used.</p>
</td></tr>
<tr><td><code id="predict.ordinalgmifs_+3A_model.select">model.select</code></td>
<td>
<p>when <code>x</code> is specified any model along the solution path can be selected. The default is <code>model.select="AIC"</code> which calculates the predicted values using the coefficients from the model having the lowest AIC. Other options are <code>model.select="BIC"</code> or any numeric value from the solution path.</p>
</td></tr>
<tr><td><code id="predict.ordinalgmifs_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>predicted</code></td>
<td>
<p>a matrix of predicted probabilities from the fitted model.</p>
</td></tr>
<tr><td><code>class</code></td>
<td>
<p>a vector containing the predicted class taken as that class having the largest predicted probability.</p>
</td></tr>
<tr><td><code>...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kellie J. Archer, Jiayi Hou, Qing Zhou, Kyle Ferber, John G. Layne, Amanda Gentry
</p>


<h3>See Also</h3>

<p>See Also <code><a href="#topic+ordinalgmifs">ordinalgmifs</a></code>, <code><a href="#topic+coef.ordinalgmifs">coef.ordinalgmifs</a></code>, <code><a href="#topic+summary.ordinalgmifs">summary.ordinalgmifs</a></code>, <code><a href="#topic+plot.ordinalgmifs">plot.ordinalgmifs</a></code> 
</p>

<hr>
<h2 id='print.ordinalgmifs'>
Print the Contents of an Ordinal GMIFS Fitted Object.
</h2><span id='topic+print.ordinalgmifs'></span>

<h3>Description</h3>

<p>This function prints the names of the list objects from an <code>ordinalgmifs</code> fitted model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ordinalgmifs'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.ordinalgmifs_+3A_x">x</code></td>
<td>
<p>an <code>ordinalgmifs</code> object.</p>
</td></tr>
<tr><td><code id="print.ordinalgmifs_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns the object names in the fitted ordinalgmifs object</p>


<h3>Note</h3>

<p>The contents of an <code>ordinalgmifs</code> fitted object differ depending upon whether <code>x</code> is specified in the <code>ordinalgmifs</code> model (i.e., penalized variables are included in
the model fit hence a solution path is returned) or only <code>terms</code> on the right hand side of the equation are included (unpenalized variables). In the
latter case, we recommend using the VGAM package.
</p>


<h3>Author(s)</h3>

<p>Kellie J. Archer
</p>


<h3>See Also</h3>

<p>See Also <code><a href="#topic+ordinalgmifs">ordinalgmifs</a></code>, <code><a href="#topic+coef.ordinalgmifs">coef.ordinalgmifs</a></code>, <code><a href="#topic+summary.ordinalgmifs">summary.ordinalgmifs</a></code>, <code><a href="#topic+plot.ordinalgmifs">plot.ordinalgmifs</a></code>, <code><a href="#topic+predict.ordinalgmifs">predict.ordinalgmifs</a></code>
</p>

<hr>
<h2 id='summary.ordinalgmifs'>
Summarize an Ordinal GMIFS Object.
</h2><span id='topic+summary.ordinalgmifs'></span>

<h3>Description</h3>

<p><code>summary</code> method for class <code>ordinalgmifs</code>.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ordinalgmifs'
summary(object, model.select = "AIC", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.ordinalgmifs_+3A_object">object</code></td>
<td>

<p>an <code>ordinalgmifs</code> object.
</p>
</td></tr>
<tr><td><code id="summary.ordinalgmifs_+3A_model.select">model.select</code></td>
<td>

<p>when <code>x</code> is specified any model along the solution path can be selected. The default is <code>model.select="AIC"</code> which extracts the model having the lowest AIC. Other options are <code>model.select="BIC"</code> or any numeric value from the solution path.
</p>
</td></tr>
<tr><td><code id="summary.ordinalgmifs_+3A_...">...</code></td>
<td>
<p>other arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Prints the following items extracted from the fitted <code>ordinalgmifs</code> object:
the probability model and link used and model parameter estimates. For models that include
<code>x</code>, the parameter estimates, AIC, BIC, and log-likelihood are printed for indicated <code>model.select</code> step or if <code>model.select</code> is not supplied the step at which the minimum AIC was observed.
</p>


<h3>Value</h3>

<p>extracts the relevant information from the step in the solution 
path that attained the minimum AIC (default) or at the user-defined 
<code>model.select</code> step</p>


<h3>Author(s)</h3>

<p>Kellie J. Archer
</p>


<h3>See Also</h3>

<p>See Also <code><a href="#topic+ordinalgmifs">ordinalgmifs</a></code>, <code><a href="#topic+coef.ordinalgmifs">coef.ordinalgmifs</a></code>, <code><a href="#topic+plot.ordinalgmifs">plot.ordinalgmifs</a></code>, <code><a href="#topic+predict.ordinalgmifs">predict.ordinalgmifs</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
