<!DOCTYPE html><html><head><title>Help for package OrgMassSpecR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {OrgMassSpecR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#OrgMassSpecR-package'>
<p>Organic Mass Spectrometry</p></a></li>
<li><a href='#ConvertConcentration'><p>Convert Concentration Basis</p></a></li>
<li><a href='#ConvertPeptide'><p>Convert peptide sequence.</p></a></li>
<li><a href='#DeadVolume'><p>Internal volume of tubing.</p></a></li>
<li><a href='#Digest'><p>Predict Peptides Resulting from Enzymatic Digest</p></a></li>
<li><a href='#DrawChromatogram'><p>Plot a Chromatogram</p></a></li>
<li><a href='#example.chromatogram.single'><p>Example Chromatograms</p></a></li>
<li><a href='#example.sequence'><p>Human Serum Albumin Amino Acid Sequence</p></a></li>
<li><a href='#example.spectrum.labeled'><p>Example MALDI-TOF Spectrum of Nitrogen-15 Labeled Peptide</p></a></li>
<li><a href='#example.spectrum.peptide'><p>Example Peptide Fragmentation Spectrum</p></a></li>
<li><a href='#example.spectrum.unknown'><p>Example EI Fragmentation Spectra</p></a></li>
<li><a href='#ExchangeableAmides'><p>Determine the Number of Backbone Amide Hydrogens</p></a></li>
<li><a href='#FlowTime'><p>Solvent transit time.</p></a></li>
<li><a href='#FragmentPeptide'><p>Predict Peptide Fragment Ions</p></a></li>
<li><a href='#IsotopicDistribution'><p>Isotopic Distribution of an Organic Molecule.</p></a></li>
<li><a href='#IsotopicDistributionHDX'><p>Isotopic Distribution of a Peptide Undergoing H-D Exchange</p></a></li>
<li><a href='#IsotopicDistributionN'><p>Isotopic Distribution of a Nitrogen-15 Labeled Peptide.</p></a></li>
<li><a href='#ListFormula'><p>Convert an Elemental Formula to a List</p></a></li>
<li><a href='#MolecularWeight'><p>Calculate the molecular weight of an organic molecule.</p></a></li>
<li><a href='#MonoisotopicMass'><p>Calculate the monoisotopic mass or monoisotopic m/z value of an organic molecule.</p></a></li>
<li><a href='#PeptideSpectrum'><p>Plot Annotated Peptide Fragmentation Mass Spectrum</p></a></li>
<li><a href='#ReadMspDirectory'><p>Read in MSP Formatted Mass Spectra</p></a></li>
<li><a href='#ReadMspFile'><p>Read in a MSP Formatted Mass Spectrum</p></a></li>
<li><a href='#RetentionIndex'><p>Non-Isothermal Retention Index Calculation</p></a></li>
<li><a href='#SpectrumSimilarity'><p>Similarity Between Two Mass Spectra</p></a></li>
<li><a href='#WriteMspFile'><p>Write Spectra to the NIST MSP Text Format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Organic Mass Spectrometry</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5-3</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-08-12</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0), grid</td>
</tr>
<tr>
<td>Suggests:</td>
<td>lattice, knitr, rmarkdown</td>
</tr>
<tr>
<td>Description:</td>
<td>Organic/biological mass spectrometry data analysis.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-2-Clause">BSD_2_clause</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://OrgMassSpec.github.io/">http://OrgMassSpec.github.io/</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-08-12 17:08:13 UTC; natha</td>
</tr>
<tr>
<td>Author:</td>
<td>Nathan Dodder [cre, aut, cph],
  Katharine Mullen [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nathan Dodder &lt;ndodder@mail.sdsu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-08-13 09:18:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='OrgMassSpecR-package'>
Organic Mass Spectrometry
</h2><span id='topic+OrgMassSpecR-package'></span><span id='topic+OrgMassSpecR'></span>

<h3>Description</h3>

<p>Organic/biological mass spectrometry data analysis.
</p>
<p>http://OrgMassSpec.github.io/
</p>

<hr>
<h2 id='ConvertConcentration'>Convert Concentration Basis</h2><span id='topic+ConvertConcentration'></span>

<h3>Description</h3>

<p>Change the unit basis for a sample concentration, such as
ng/g wet weight to ng/g dry weight, or pg/g lipid weight to
pg/g wet weight.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConvertConcentration(x, convert, percent)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ConvertConcentration_+3A_x">x</code></td>
<td>
<p>numeric value. The concentration to be converted.</p>
</td></tr>
<tr><td><code id="ConvertConcentration_+3A_convert">convert</code></td>
<td>
<p>character string. One of <code>wet.to.dry</code>,
<code>dry.to.wet</code>, <code>wet.to.lipid</code>, or <code>lipid.to.wet</code>. See
details.</p>
</td></tr>
<tr><td><code id="ConvertConcentration_+3A_percent">percent</code></td>
<td>
<p>numeric value. Either the percent moisture or percent
lipid. See details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>convert</code> argument specifies the type of conversion,
for example <code>wet.to.dry</code> is wet weight basis to dry weight basis.
</p>
<p>Conversion between wet weight basis and dry weight basis requires
<code>percent</code> to equal the percent moisture in the sample. Conversion
between wet weight basis and lipid weight basis requires
<code>percent</code> to equal the percent lipid in the sample. The following
definitions for percent moisture and percent lipid are used.
</p>
<p style="text-align: center;"><code class="reqn">percent moisture = \frac{wet weight - dry weight}{wet weight} *
  100</code>
</p>

<p style="text-align: center;"><code class="reqn">percent lipid = \frac{lipid weight}{wet weight} * 100</code>
</p>
 
<p>The unit prefixes (for example ng/g or pg/g) are not changed by the
conversion.
</p>


<h3>Value</h3>

<p>Numeric vector of length 1. The unit prefixes are the same as for
<code>x</code>, only the concentration basis is converted. 
</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Convert a concentration of 15.3 ng/g wet weight,
## 5 percent lipid, to ng/g lipid weight.

ConvertConcentration(15.3, "wet.to.lipid", 5)
</code></pre>

<hr>
<h2 id='ConvertPeptide'>Convert peptide sequence.</h2><span id='topic+ConvertPeptide'></span>

<h3>Description</h3>

<p>Convert single amino acid codes to an elemental formula or three letter codes.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConvertPeptide(sequence, output = "elements", IAA = TRUE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ConvertPeptide_+3A_sequence">sequence</code></td>
<td>
<p>a character string representing the amino acid sequence.</p>
</td></tr>
<tr><td><code id="ConvertPeptide_+3A_output">output</code></td>
<td>
<p>a character string specifying the output.  Options are <code>"elements"</code> (default) for the elemental formula or <code>"3letter"</code> for the three letter codes.</p>
</td></tr>
<tr><td><code id="ConvertPeptide_+3A_iaa">IAA</code></td>
<td>
<p>logical. <code>TRUE</code> specifies iodoacetylated cysteine and <code>FALSE</code> specifies unmodified cystine.  Used only in determining the elemental formula, not the three letter codes.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The amino acid residues must be specified by the one letter codes defined in the help for <code><a href="#topic+Digest">Digest</a></code>.
</p>
<p>The argument <code>IAA</code> specifies treatment of the protein with iodoacetamide to break the disulfide bonds.  This treatment produces iodoacetylated cysteine residues (elemental formula C5H8N2O2S).
</p>


<h3>Value</h3>

<p>If <code>output = "elements"</code>, the value is a list specifying the number of each element.  This list can be used as input to other functions, see the examples.  If output = <code>"3letter"</code>, the value is a vector of length 1, containing the amino acid sequence in three letter codes.
</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>Examples</h3>

<pre><code class='language-R'>ConvertPeptide("SEQENCE", output = "3letter")

# as input to MonoisotopicMass
MonoisotopicMass(formula = ConvertPeptide("SEQENCE"), charge = 1)

# as input to MolecularWeight
MolecularWeight(formula = ConvertPeptide("SEQENCE"))
</code></pre>

<hr>
<h2 id='DeadVolume'>Internal volume of tubing.</h2><span id='topic+DeadVolume'></span>

<h3>Description</h3>

<p>Calculate the internal volume of a defined length of tubing.</p>


<h3>Usage</h3>

<pre><code class='language-R'>DeadVolume(internalDiameterMicrometers = 24, tubeLengthCentimeters = 45)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DeadVolume_+3A_internaldiametermicrometers">internalDiameterMicrometers</code></td>
<td>
<p>the internal diameter of the tubing, in micrometers; numeric value.</p>
</td></tr>
<tr><td><code id="DeadVolume_+3A_tubelengthcentimeters">tubeLengthCentimeters</code></td>
<td>
<p>the length of tubing, in centimeters; numeric value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of the dead volume in microliters.</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>See Also</h3>

<p><code><a href="#topic+FlowTime">FlowTime</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>DeadVolume(internalDiameterMicrometers = 47, tubeLengthCentimeters = 33)</code></pre>

<hr>
<h2 id='Digest'>Predict Peptides Resulting from Enzymatic Digest</h2><span id='topic+Digest'></span>

<h3>Description</h3>

<p>Cleave an amino acid sequence (a protein or peptide) according to enzyme specific rules and calculate the precursor ion <em>m/z</em> values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Digest(sequence, enzyme = "trypsin", missed = 0, IAA = TRUE, 
       N15 = FALSE, custom = list())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Digest_+3A_sequence">sequence</code></td>
<td>
<p>a character string representing the amino acid sequence to be cleaved by the enzyme.</p>
</td></tr>
<tr><td><code id="Digest_+3A_enzyme">enzyme</code></td>
<td>
<p>a character string specifying the rules for cleavage.  Options are <code>"trypsin"</code> (default),                    <code>"trypsin.strict"</code> (see details), or <code>"pepsin"</code>.</p>
</td></tr>
<tr><td><code id="Digest_+3A_missed">missed</code></td>
<td>
<p>the maximum number of missed cleavages.  Must be an integer of 0 (default) or greater.  An error will result if the specified number of missed cleavages is greater than the maximum possible number of missed cleavages.</p>
</td></tr>
<tr><td><code id="Digest_+3A_iaa">IAA</code></td>
<td>
<p>logical. <code>TRUE</code> specifies iodoacetylated cysteine and <code>FALSE</code> specifies unmodified cystine.</p>
</td></tr>
<tr><td><code id="Digest_+3A_n15">N15</code></td>
<td>
<p>logical indicating if the nitrogen-15 isotope should be used in place of the default nitrogen-14 isotope.</p>
</td></tr>
<tr><td><code id="Digest_+3A_custom">custom</code></td>
<td>
<p>a list specifying user defined residues as <code>custom = list(code, mass)</code>, where <code>code</code> is a vector of one letter characters and <code>mass</code> is a vector of the respective monoisotopic masses.  See Details and Examples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The amino acid residues must be specified by one letter codes.  The predefined residues are: 
</p>

<table>
<tr>
 <td style="text-align: left;">
A = alanine         </td><td style="text-align: left;"> L = leucine</td>
</tr>
<tr>
 <td style="text-align: left;"> 
R = arginine        </td><td style="text-align: left;"> K = lysine</td>
</tr>
<tr>
 <td style="text-align: left;">
N = asparagine      </td><td style="text-align: left;"> M = methionine</td>
</tr>
<tr>
 <td style="text-align: left;">
D = aspartic acid   </td><td style="text-align: left;"> F = phenylalanine</td>
</tr>
<tr>
 <td style="text-align: left;">
C = cysteine        </td><td style="text-align: left;"> P = proline</td>
</tr>
<tr>
 <td style="text-align: left;">
E = glutamic acid   </td><td style="text-align: left;"> S = serine</td>
</tr>
<tr>
 <td style="text-align: left;">
Q = glutamine       </td><td style="text-align: left;"> T = threonine</td>
</tr>
<tr>
 <td style="text-align: left;">
G = glycine         </td><td style="text-align: left;"> W = tryptophan</td>
</tr>
<tr>
 <td style="text-align: left;">
H = histidine       </td><td style="text-align: left;"> Y = tyrosine</td>
</tr>
<tr>
 <td style="text-align: left;">
I = isoleucine      </td><td style="text-align: left;"> V = valine
</td>
</tr>

</table>

<p>If <code>"trypsin"</code> is specified, the sequence is cleaved on the c-terminal side of K and R residues, except if K or R is followed by P.  If <code>"trypsin.strict"</code> is specified, the sequence is cleaved on the c-terminal side of K and R residues. If <code>"pepsin"</code> is specified, the sequence is cleaved on the c-terminal side of F, L, W, Y, A, E, and Q residues.  This rule is specific to pepsin at pH <code class="reqn">&gt;</code> 2, as used in hydrogen-deuterium exchange experiments.
</p>
<p>When <code>"trypsin"</code> is specified, KP and RP are not considered missed cleavages when <code>missed</code> is <code class="reqn">&gt;</code> 0.
</p>
<p>The argument <code>IAA</code> specifies treatment of the protein with iodoacetamide.  This treatment produces iodoacetylated cysteine residues (elemental formula C5H8N2O2S).
</p>
<p>If <code>TRUE</code>, the argument <code>N15</code> specifies 100% nitrogen-15 incorporation.  It is intended for proteins grown with a nitrogen-15 labeled food source.  (Although the experiment itself may grow a protein with less than 100% nitrogen-15 incorporation).  Setting <code>N15 = TRUE</code> does not modify the mass of a custom residue, or the mass of the nitrogen(s) added if <code>IAA = TRUE</code>.
</p>
<p>If a custom residue code is identical to a predefined residue code, the custom residue mass will be used in place of the predefined mass.  
</p>
<p>The error message &ldquo;<code>object "mass" not found</code>&rdquo; indicates the input sequence contains an undefined residue(s).
</p>


<h3>Value</h3>

<p>A data frame with the following column names.
</p>
<table>
<tr><td><code>peptide</code></td>
<td>
<p>resulting peptides.</p>
</td></tr>
<tr><td><code>start</code></td>
<td>
<p>beginning residue positions in the the original sequence.</p>
</td></tr>
<tr><td><code>end</code></td>
<td>
<p>ending residue positions in the the original sequence.</p>
</td></tr>
<tr><td><code>mc</code></td>
<td>
<p>number of missed cleavages.</p>
</td></tr>
<tr><td><code>mz1</code></td>
<td>
<p>monoisotopic <em>m/z</em> values for the <code class="reqn">[M + H]^{1+}</code> ions (where M is the precursor mass).</p>
</td></tr>
<tr><td><code>mz2</code></td>
<td>
<p>monoisotopic <em>m/z</em> values for the <code class="reqn">[M + 2H]^{2+}</code> ions.</p>
</td></tr>
<tr><td><code>mz3</code></td>
<td>
<p>monoisotopic <em>m/z</em> values for the <code class="reqn">[M + 3H]^{3+}</code> ions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>References</h3>

<p>The relative atomic masses of the isotopes are from the NIST Physical Reference Data Website <a href="http://physics.nist.gov/PhysRefData/Compositions/">http://physics.nist.gov/PhysRefData/Compositions/</a>.  The molar mass of a proton (H+) is from the NIST CODATA Website <a href="http://physics.nist.gov/cuu/Constants/index.html">http://physics.nist.gov/cuu/Constants/index.html</a>.  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MonoisotopicMass">MonoisotopicMass</a></code>, <code><a href="#topic+FragmentPeptide">FragmentPeptide</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## digest human serum albumin with 0 and 1 missed cleavages
Digest(example.sequence, missed = 1)

## digest human serum albumin with a phosphoserine at position 58
## and all methionines oxidized
modifiedHsaSequence &lt;- strsplit(example.sequence, split = "")[[1]]
modifiedHsaSequence[58] &lt;- "s"   # insert code for phosphoserine  
modifiedHsaSequence &lt;- paste(modifiedHsaSequence, collapse = "")
Digest(modifiedHsaSequence, custom = list(code = c("s","M"), 
       mass = c(MonoisotopicMass(list(C=3, H=6, N=1, O=5, P=1)),
                MonoisotopicMass(list(C=5, H=9, N=1, O=2, S=1)))))
                
## digest human serum albumin with strict rules
Digest(example.sequence, enzyme = "trypsin.strict")
</code></pre>

<hr>
<h2 id='DrawChromatogram'>Plot a Chromatogram</h2><span id='topic+DrawChromatogram'></span>

<h3>Description</h3>

<p>Plot a chromatogram, color the area under specified peak(s), and calculate the peak area(s).</p>


<h3>Usage</h3>

<pre><code class='language-R'>DrawChromatogram(time, intensity, range = list(start, stop), color = "blue", 
                 xlab = "retention time", ylab = "intensity", 
                 ylim = c(0, max(intensity) * 1.1), las = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DrawChromatogram_+3A_time">time</code></td>
<td>
<p>numeric vector containing the time points (the x-axis).</p>
</td></tr>
<tr><td><code id="DrawChromatogram_+3A_intensity">intensity</code></td>
<td>
<p>numeric vector containing the respective signal intensities at each time point (the y-axis).</p>
</td></tr>
<tr><td><code id="DrawChromatogram_+3A_range">range</code></td>
<td>
<p>list describing the start and stop time points for each peak, defined as <code>range = list(start, stop)</code>, where <code>start</code> is the numeric vector of starting time points and <code>stop</code> is the numeric vector of the respective ending time points.</p>
</td></tr>
<tr><td><code id="DrawChromatogram_+3A_color">color</code></td>
<td>
<p>vector of character strings specifying the color for each peak given in <code>range</code>.  If a single color is specified, it is applied to all peaks.</p>
</td></tr>
<tr><td><code id="DrawChromatogram_+3A_xlab">xlab</code></td>
<td>
<p>character string specifying the x-axis label.</p>
</td></tr>
<tr><td><code id="DrawChromatogram_+3A_ylab">ylab</code></td>
<td>
<p>character string specifying the y-axis label.</p>
</td></tr>
<tr><td><code id="DrawChromatogram_+3A_ylim">ylim</code></td>
<td>
<p>numeric vector of length 2 specifying the range of the y axis.</p>
</td></tr>
<tr><td><code id="DrawChromatogram_+3A_las">las</code></td>
<td>
<p>numeric value specifying the rotation of the axis labels, see par for options.</p>
</td></tr>
<tr><td><code id="DrawChromatogram_+3A_...">...</code></td>
<td>
<p>additional parameters to be passed to plot().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The area under the peak(s) is rendered using the <code><a href="graphics.html#topic+polygon">polygon</a></code> function.  The area calculation assumes that the polygon does not self-intersect.</p>


<h3>Value</h3>

<p>A data frame with the following column names.
</p>
<table>
<tr><td><code>retention.time</code></td>
<td>
<p>retention times of the peaks specified in <code>range</code></p>
</td></tr>
<tr><td><code>area</code></td>
<td>
<p>the respective peak areas</p>
</td></tr>
<tr><td><code>apex.intensity</code></td>
<td>
<p>the respective intensities at each peak apex</p>
</td></tr> 
</table>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>Examples</h3>

<pre><code class='language-R'>## single peak
x &lt;- DrawChromatogram(example.chromatogram.single$time,  
                      example.chromatogram.single$intensity/100, 
                      range = list(start = 25.4, stop = 26.1), 
                      main = "example chromatogram 1",
                      ylab = "intensity x 100")
# label peak with retention time and area
text(x$retentionTime, x$apexIntensity + 1500, 
     labels = paste("RT = ", round(x$retentionTime, digits = 1),
                    ", Area = ", round(x$peakArea), sep = ""), cex = 0.9) 

## multiple peaks 
y &lt;- DrawChromatogram(example.chromatogram.multiple$time, 
                      example.chromatogram.multiple$intensity / 1000, 
                      range = list(start = c(21.5, 21.925, 23.1, 25.5, 27.35), 
                        stop = c(21.925, 22.4, 23.6, 26.2, 28.0)),
                      color = c("blue", "red", "green", "yellow", "orange"), 
                      xlab = "retention time (min)", 
                      ylab = "intensity x 1000 (cps)", 
                      main = "Example Chromatogram")
## label peaks
text(y$retentionTime, y$apexIntensity + 50, labels = c("a", "b", "c", "d", "e"))
</code></pre>

<hr>
<h2 id='example.chromatogram.single'>Example Chromatograms</h2><span id='topic+example.chromatogram.single'></span><span id='topic+example.chromatogram.multiple'></span>

<h3>Description</h3>

<p>Examples for <code><a href="#topic+DrawChromatogram">DrawChromatogram</a></code>.</p>


<h3>Usage</h3>

<pre><code class='language-R'>example.chromatogram.single
example.chromatogram.multiple
</code></pre>


<h3>Format</h3>

<p>A data frame consisting of time (minutes) in the first column and intensity (counts per second) in the second.</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>

<hr>
<h2 id='example.sequence'>Human Serum Albumin Amino Acid Sequence</h2><span id='topic+example.sequence'></span>

<h3>Description</h3>

<p>Used in examples.</p>


<h3>Usage</h3>

<pre><code class='language-R'>example.sequence</code></pre>


<h3>Format</h3>

<p>Character sequence.</p>


<h3>Source</h3>

<p>Downloaded from the ExPASy Proteomics Server (P02768) on 2009-02-07.  Residues 25-609.</p>

<hr>
<h2 id='example.spectrum.labeled'>Example MALDI-TOF Spectrum of Nitrogen-15 Labeled Peptide</h2><span id='topic+example.spectrum.labeled'></span>

<h3>Description</h3>

<p>Example used in Vinegette.</p>


<h3>Usage</h3>

<pre><code class='language-R'>example.spectrum.labeled</code></pre>


<h3>Format</h3>

<p>A data frame consisting of <em>m/z</em> values in the first column and intensity in the second.</p>


<h3>Author(s)</h3>

<p>Protein expressed by Wei-Li Liao.</p>


<h3>Source</h3>

<p>Precursor ion spectrum of peptide YEVQGEVFTKPQLWP acquired on a MALDI-TOF/TOF.  The peptide is labeled with approximately 99% nitrogen-15.
</p>

<hr>
<h2 id='example.spectrum.peptide'>Example Peptide Fragmentation Spectrum</h2><span id='topic+example.spectrum.peptide'></span>

<h3>Description</h3>

<p>Example for <code><a href="#topic+PeptideSpectrum">PeptideSpectrum</a></code>.</p>


<h3>Usage</h3>

<pre><code class='language-R'>example.spectrum.peptide
</code></pre>


<h3>Format</h3>

<p>A data frame consisting of (centroid) <em>m/z</em> values in the first column and peak intensities in the second.</p>


<h3>Source</h3>

<p>Peptide sequence NIDALSGMEGR.  Acquired on a hybrid triple quadruploe/linear ion trap mass spectrometer in collision induced dissociation (CID) mode.</p>

<hr>
<h2 id='example.spectrum.unknown'>Example EI Fragmentation Spectra</h2><span id='topic+example.spectrum.unknown'></span><span id='topic+example.spectrum.authentic'></span>

<h3>Description</h3>

<p>Examples for <code><a href="#topic+SpectrumSimilarity">SpectrumSimilarity</a></code>.</p>


<h3>Usage</h3>

<pre><code class='language-R'>example.spectrum.unknown
example.spectrum.authentic
</code></pre>


<h3>Format</h3>

<p>A data frame consisting of (centroid) <em>m/z</em> values in the first column and peak intensities in the second.</p>


<h3>Author(s)</h3>

<p>Spectra acquired by Gauthier Eppe.</p>


<h3>Source</h3>

<p>Mass spectra of alanine derivatized with N-Methyl-N-[tert-butyldimethyl-silyl]trifluoroacetimide (MTBSTFA) collected in electron impact (EI) mode on a 2 dimensional gas chromatography time of flight (GCxGC-TOF) mass spectrometer.  example.spectrum.unknown was acquired from a human plasma extract.  example.spectrum.authentic was acquired from an authentic material.</p>

<hr>
<h2 id='ExchangeableAmides'>Determine the Number of Backbone Amide Hydrogens</h2><span id='topic+ExchangeableAmides'></span>

<h3>Description</h3>

<p>Determine the number of backbone amide hydrogens given a protein/peptide sequence.  Used in hydrogen-deuterium exchange experiments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExchangeableAmides(sequence)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ExchangeableAmides_+3A_sequence">sequence</code></td>
<td>
<p>character vector containing one or more amino acid sequences.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The number of backbone amide hydrogens in an amino acid sequence is the number of residues, minus the number of prolines, minus 1.
</p>


<h3>Value</h3>

<p>A numeric vector containing the number of exchangeable hydrogens.</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>See Also</h3>

<p><code><a href="#topic+IsotopicDistributionHDX">IsotopicDistributionHDX</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>ExchangeableAmides(c("VDVMCTA", "VSTPTL"))

## find the number of exchangeable amides for 
## each peptide in a digest of human serum albumin
x &lt;- Digest(example.sequence, enzyme = "pepsin", IAA = FALSE, missed = 4)
transform(x, exchange = ExchangeableAmides(x$peptide))
</code></pre>

<hr>
<h2 id='FlowTime'>Solvent transit time.</h2><span id='topic+FlowTime'></span>

<h3>Description</h3>

<p>Calculate the time required for a liquid to flow through a defined length of tubing.  Intended for conventional-flow or nano-flow liquid chromatography.</p>


<h3>Usage</h3>

<pre><code class='language-R'>FlowTime(internalDiameterMicrometers = 24, tubingLengthCentimeters = 45, 
                flowRateMicrolitersPerMinute = 0.3)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FlowTime_+3A_internaldiametermicrometers">internalDiameterMicrometers</code></td>
<td>
<p>the internal diameter of the tubing, in micrometers; numeric value.</p>
</td></tr>
<tr><td><code id="FlowTime_+3A_tubinglengthcentimeters">tubingLengthCentimeters</code></td>
<td>
<p>the length of tubing, in centimeters; numeric value.</p>
</td></tr>
<tr><td><code id="FlowTime_+3A_flowratemicrolitersperminute">flowRateMicrolitersPerMinute</code></td>
<td>
<p>the flow rate of the solvent in microliters per minute; numeric value</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The calculation assumes the viscosity of the liquid is negligible.</p>


<h3>Value</h3>

<p>Vector of the flow time in seconds.</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>See Also</h3>

<p><code><a href="#topic+DeadVolume">DeadVolume</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>FlowTime(internalDiameterMicrometers = 47, tubingLengthCentimeters = 33, 
                   flowRateMicrolitersPerMinute = 5)</code></pre>

<hr>
<h2 id='FragmentPeptide'>Predict Peptide Fragment Ions</h2><span id='topic+FragmentPeptide'></span>

<h3>Description</h3>

<p>Determine the b- and y-ions or c- and z-ions produced by the fragmentation of a peptide by tandem mass spectrometry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FragmentPeptide(sequence, fragments = "by", IAA = TRUE, 
                N15 = FALSE, custom = list())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FragmentPeptide_+3A_sequence">sequence</code></td>
<td>
<p>a vector of character strings representing the amino acid sequences to be fragmented by the mass spectrometer.</p>
</td></tr>
<tr><td><code id="FragmentPeptide_+3A_fragments">fragments</code></td>
<td>
<p>character string specifying the fragmentation rules.  Options are <code>"by"</code> (default) for the b- and y-ions, or <code>"cz"</code> for the c- and z-ions.</p>
</td></tr>
<tr><td><code id="FragmentPeptide_+3A_iaa">IAA</code></td>
<td>
<p>logical. <code>TRUE</code> specifies iodoacetylated cysteine and <code>FALSE</code> specifies unmodified cystine.</p>
</td></tr>
<tr><td><code id="FragmentPeptide_+3A_n15">N15</code></td>
<td>
<p>logical indicating if the nitrogen-15 isotope should be used in place of the default nitrogen-14 isotope.</p>
</td></tr>
<tr><td><code id="FragmentPeptide_+3A_custom">custom</code></td>
<td>
<p>a list specifying user defined residues as <code>custom = list(code, mass)</code>, where <code>code</code> is a vector of one letter characters and <code>mass</code> is a vector of the respective monoisotopic masses.  See Details and Examples.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The amino acid residues must be specified by the one letter codes defined in the help for <code><a href="#topic+Digest">Digest</a></code>. 
</p>
<p>The fragmentation rules can be set for collision induced dissociation (b- and y-ions) or electron transfer dissociation (c- and z-ions). 
</p>
<p>The argument <code>IAA</code> specifies treatment of the protein with iodoacetamide to break the disulfide bonds.  This treatment produces iodoacetylated cysteine residues (elemental formula C5H8N2O2S).
</p>
<p>If TRUE, the argument <code>N15</code> specifies 100% nitrogen-15 incorporation.  It is intended for proteins grown with a nitrogen-15 labeled food source.  (Although the experiment itself may grow a protein with less than 100% nitrogen-15 incorporation).  Setting <code>N15 = TRUE</code> does not modify the mass of a custom residue, or the mass of the nitrogen(s) added if <code>IAA = TRUE</code>.
</p>
<p>If a custom residue code is identical to a predefined residue code, the custom residue mass will be used in place of the predefined mass.
</p>


<h3>Value</h3>

<p>A data frame with the following column names.  The data frame is arranged this way to facilitate selection of product-precursor ion pairs. 
</p>
<table>
<tr><td><code>ms1seq</code></td>
<td>
<p>precursor ion sequence.</p>
</td></tr>
<tr><td><code>ms1z1</code></td>
<td>
<p>monoisotopic <em>m/z</em> value for the <code class="reqn">[M + H]^{1+}</code> precursor ions (where M is the precursor mass).</p>
</td></tr>
<tr><td><code>ms1z2</code></td>
<td>
<p>monoisotopic <em>m/z</em> value for the <code class="reqn">[M + 2H]^{2+}</code> precursor ions.</p>
</td></tr>
<tr><td><code>ms1z3</code></td>
<td>
<p>monoisotopic <em>m/z</em> value for the <code class="reqn">[M + 3H]^{3+}</code> precursor ions.</p>
</td></tr>
<tr><td><code>ms2seq</code></td>
<td>
<p>product ion sequence.</p>
</td></tr>
<tr><td><code>ms2type</code></td>
<td>
<p>the type and charge state of the product ions.</p>
</td></tr>
<tr><td><code>ms2mz</code></td>
<td>
<p>monoisotopic <em>m/z</em> values for the product ions.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nathan G. Dodder and Katharine M. Mullen</p>


<h3>References</h3>

<p>The relative atomic masses of the isotopes are from the NIST Physical Reference Data Website <a href="http://physics.nist.gov/PhysRefData/Compositions/">http://physics.nist.gov/PhysRefData/Compositions/</a>.  The molar mass of a proton (H+) is from the NIST CODATA Website <a href="http://physics.nist.gov/cgi-bin/cuu/Value?mmp">http://physics.nist.gov/cgi-bin/cuu/Value?mmp</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MonoisotopicMass">MonoisotopicMass</a></code>, <code><a href="#topic+Digest">Digest</a></code>, <code><a href="#topic+PeptideSpectrum">PeptideSpectrum</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## fragment unlabeled peptide
FragmentPeptide("NECFLQHK")

## fragment peptide with carbon-13 labeled lysine
k.mass &lt;- MonoisotopicMass(formula = list(C = 6, H = 12, N = 2, O = 1),
                       isotopes = list(C = 13.0033548378)) 
FragmentPeptide("NECFLQHk", custom = list(code = "k", mass = k.mass))

## fragment peptide with two modifications
m.mass &lt;- MonoisotopicMass(formula = list(C=5, H=9, N=1, O=2, S=1))
FragmentPeptide("NDmELWk", custom = list(code = c("m", "k"), mass = c(m.mass, k.mass)))

## fragment a vector of peptides produced by Digest
x &lt;- Digest(example.sequence)
y &lt;- subset(x, nchar(x$peptide) &gt; 5 &amp; nchar(x$peptide) &lt; 12)
FragmentPeptide(y$peptide)
</code></pre>

<hr>
<h2 id='IsotopicDistribution'>Isotopic Distribution of an Organic Molecule.</h2><span id='topic+IsotopicDistribution'></span>

<h3>Description</h3>

<p>Simulate the isotopic distribution of an organic molecule using the natural abundances of the isotopes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IsotopicDistribution(formula = list(), charge = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IsotopicDistribution_+3A_formula">formula</code></td>
<td>
<p>a list describing the charged elemental formula.  The
allowed elements are C, H, N, O, S, P, Br, Cl, F, Si.</p>
</td></tr>
<tr><td><code id="IsotopicDistribution_+3A_charge">charge</code></td>
<td>
<p>an integer specifying the number of positive or negative
charges. A charge of zero is not allowed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The elemental formula should be that of the charged molecule; i.e, the
<code>charge</code> argument does not change the elemental formula.
</p>
<p>The algorithm used in this function is based on <code><a href="base.html#topic+sample">sample</a></code>,
and will give a slightly different result each time it is run.
</p>
<p>This function is intended for simulating the isotopic distributions of
small molecules (&lt; approximately 3000 amu and &lt;= 3 charges) as measured on low resolution mass spectrometers. 
</p>


<h3>Value</h3>

<p>A data frame with the following column names.
</p>
<table>
<tr><td><code>mz</code></td>
<td>
<p>the <em>m/z</em> value</p>
</td></tr>
<tr><td><code>intensity</code></td>
<td>
<p>the number of counts at each <em>m/z</em> value.  The total is 10000.</p>
</td></tr>
<tr><td><code>percent</code></td>
<td>
<p>the intensity at each <em>m/z</em> value, expressed as a percent of the maximum intensity.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>References</h3>

<p>The relative atomic masses of the isotopes are from the NIST Physical Reference Data Website <a href="http://physics.nist.gov/PhysRefData/Compositions/">http://physics.nist.gov/PhysRefData/Compositions/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+IsotopicDistributionN">IsotopicDistributionN</a></code>, <code><a href="#topic+IsotopicDistributionHDX">IsotopicDistributionHDX</a></code>, <code><a href="base.html#topic+sample">sample</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- IsotopicDistribution(formula = list(C = 9, H = 4, Br = 5, Cl = 1, N = 2))
library(lattice)
xyplot(percent ~ mz, data = x,
  type = "h",
  xlab = "m/z",
  ylab = "intensity (%)",
  main = "Isotopic Distribution, C9H4Br5ClN2")
</code></pre>

<hr>
<h2 id='IsotopicDistributionHDX'>Isotopic Distribution of a Peptide Undergoing H-D Exchange</h2><span id='topic+IsotopicDistributionHDX'></span>

<h3>Description</h3>

<p>Simulates the isotopic distribution of a peptide undergoing hydrogen-deuterium exchange.  Only the peptide backbone amides are labeled with deuterium.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IsotopicDistributionHDX(sequence, incorp, charge = 1, 
                        custom = list(code = NULL, elements = NULL))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IsotopicDistributionHDX_+3A_sequence">sequence</code></td>
<td>
<p>character vector specifying the amino acid sequence.</p>
</td></tr>
<tr><td><code id="IsotopicDistributionHDX_+3A_incorp">incorp</code></td>
<td>
<p>numeric value from 0 to 1, specifying the fraction of deuterium incorporation in the backbone amides.</p>
</td></tr>
<tr><td><code id="IsotopicDistributionHDX_+3A_charge">charge</code></td>
<td>
<p>numeric value specifying the number of positive charges (<code class="reqn">[M + nH]^{n+}</code>)</p>
</td></tr>
<tr><td><code id="IsotopicDistributionHDX_+3A_custom">custom</code></td>
<td>
<p>a list specifying user defined residues as <code>custom = list(code, mass)</code>, where <code>code</code> is a vector of one letter characters and <code>mass</code> is a vector of the respective monoisotopic masses.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The amino acid residues must be specified by the one letter codes defined in the help for <code><a href="#topic+Digest">Digest</a></code>.  If a custom residue code is identical to a predefined residue code, the custom residue mass will be used in place of the predefined mass.
</p>
<p>The natural incorporation of deuterium is 0.000115.
</p>
<p>The algorithm used in this function is based on <code><a href="base.html#topic+sample">sample</a></code>, and will give a slightly different result each time it is run.  <code>IsotopicDistributionHDX</code> has not been tested for sequences over approximately 3000 amu and charges greater that 3+.
</p>


<h3>Value</h3>

<p>A data frame with the following column names.
</p>
<table>
<tr><td><code>mz</code></td>
<td>
<p>the <em>m/z</em> value</p>
</td></tr>
<tr><td><code>intensity</code></td>
<td>
<p>the number of counts at each <em>m/z</em> value.  The total is 10000.</p>
</td></tr>
<tr><td><code>percent</code></td>
<td>
<p>the intensity at each <em>m/z</em> value, expressed as a percent of the maximum intensity.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>References</h3>

<p>The relative atomic masses of the isotopes are from the NIST Physical Reference Data Website <a href="http://physics.nist.gov/PhysRefData/Compositions/">http://physics.nist.gov/PhysRefData/Compositions/</a>.  The molar mass of a proton (H+) is from the NIST CODATA Website <a href="http://physics.nist.gov/cuu/Constants/index.html">http://physics.nist.gov/cuu/Constants/index.html</a>.  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+IsotopicDistribution">IsotopicDistribution</a></code>, <code><a href="#topic+IsotopicDistributionN">IsotopicDistributionN</a></code>, <code><a href="base.html#topic+sample">sample</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate a peptide with 0.0115 and 100 percent incorporation and plot
x &lt;- IsotopicDistributionHDX("NECFLQHK", incorp = 0.000115)
x$t &lt;- "incorp = 0.0115%"
y &lt;- IsotopicDistributionHDX("NECFLQHK", incorp = 1)
y$t &lt;- "incorp = 100%"
z &lt;- rbind(x, y)
library(lattice)
print(xyplot(percent ~ mz | t, data = z,
       type = "h", main = "H-D exchange simulation",
       xlab = "m/z", ylab = "intensity (%)"))
</code></pre>

<hr>
<h2 id='IsotopicDistributionN'>Isotopic Distribution of a Nitrogen-15 Labeled Peptide.</h2><span id='topic+IsotopicDistributionN'></span>

<h3>Description</h3>

<p>Simulates the isotopic distribution of a nitrogen-15 labeled peptide.  Intended for peptides from proteins grown in media with, for example, nitrogen-15 labeled ammonium chloride as the only nitrogen source.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IsotopicDistributionN(sequence, incorp, IAA = TRUE, charge = 1, 
                      custom = list(code = NULL, elements = NULL))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IsotopicDistributionN_+3A_sequence">sequence</code></td>
<td>
<p>character vector specifying the amino acid sequence.</p>
</td></tr>
<tr><td><code id="IsotopicDistributionN_+3A_incorp">incorp</code></td>
<td>
<p>numeric value from 0 to 1, specifying the fraction of nitrogen-15.</p>
</td></tr>
<tr><td><code id="IsotopicDistributionN_+3A_iaa">IAA</code></td>
<td>
<p>logical. <code>TRUE</code> specifies iodoacetylated cysteine and <code>FALSE</code> specifies unmodified cystine.</p>
</td></tr>
<tr><td><code id="IsotopicDistributionN_+3A_charge">charge</code></td>
<td>
<p>numeric value specifying the number of positive
charges. Must be 1, 2, or 3. One proton per charge is added to the
amino acid sequence (<code class="reqn">[M + nH]^{n+}</code>).</p>
</td></tr>
<tr><td><code id="IsotopicDistributionN_+3A_custom">custom</code></td>
<td>
<p>a list specifying a user defined residue as <code>custom = list(code, elements)</code>, where <code>code</code> is a one letter character vector representing the custom residue and <code>elements</code> is a vector of the number of elements in the character vector, with the indicies equal to C,H,N,O,S,or P.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The amino acid residues must be specified by the one letter codes defined in the help for <code><a href="#topic+Digest">Digest</a></code>.  If a custom residue code is identical to a predefined residue code, the custom residue will NOT be used in place of the predefined residue; doing so will cause errors.
</p>
<p>The natural nitrogen-15 incorporation is 0.00368. 
</p>
<p>The argument <code>IAA</code> specifies treatment of the protein with iodoacetamide.  This treatment produces iodoacetylated cysteine residues (elemental formula C5H8N2O2S).  The nitrogen(s) added is not labeled.
</p>
<p>The algorithm used in this function is based on <code><a href="base.html#topic+sample">sample</a></code>, and will give a slightly different result each time it is run.  It is similar to the algorithm used in reference 1.  <code>IsotopicDistributionN</code> has not been tested for sequences over approximately 3000 amu and charges greater that 3+.
</p>


<h3>Value</h3>

<p>A data frame with the following column names.
</p>
<table>
<tr><td><code>mz</code></td>
<td>
<p>the <em>m/z</em> value</p>
</td></tr>
<tr><td><code>intensity</code></td>
<td>
<p>the number of counts at each <em>m/z</em> value.  The total is 10000.</p>
</td></tr>
<tr><td><code>percent</code></td>
<td>
<p>the intensity at each <em>m/z</em> value, expressed as a percent of the maximum intensity.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>References</h3>

<p>1. &quot;Method for Estimating the Isotopic Distributions of Metabolically Labeled Proteins by MALDI-TOFMS: Application to NMR Samples.&quot; Choudhary K, Spicer VL, Donald LJ, Duckworth HW, Ens W, Loewen PC, Standing KG, <em>Analytical Chemistry</em>, 2006, 78, 5419-5423.
</p>
<p>2. &quot;Perturbation and Interpretation of Nitrogen Isotope Distribution Patterns in Proteomics&quot; Snijders APL, de Koning B, Wright PC, <em>Journal of Proteome Research</em>, 2005, 4, 2185-2191.
</p>
<p>3. &quot;Measurement of the Isotope Enrichment of Stable Isotope-Labeled Proteins Using High-Resolution Mass Spectra of Peptides&quot; MacCoss MJ, Wu CC, Matthews DE, and Yates III JR, <em>Analytical Chemistry</em>, 2005, 77, 7646-7653.
</p>
<p>The relative atomic masses of the isotopes are from the NIST Physical Reference Data Website <a href="http://physics.nist.gov/PhysRefData/Compositions/">http://physics.nist.gov/PhysRefData/Compositions/</a>.  The molar mass of a proton (H+) is from the NIST CODATA Website <a href="http://physics.nist.gov/cuu/Constants/index.html">http://physics.nist.gov/cuu/Constants/index.html</a>.
</p>
<p>Note that in reference 3 &quot;a biologically relevant 13C/12C isotope ratio of 1.096% was used instead of the more carbonate specific ratio of 1.112%.&quot;  The 13C/12C ratio used by <code>IsotopicDistributionN</code> is 1.081%.  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+IsotopicDistribution">IsotopicDistribution</a></code>, <code><a href="#topic+IsotopicDistributionHDX">IsotopicDistributionHDX</a></code>, <code><a href="base.html#topic+sample">sample</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate a peptide with 0.368 to 100 percent incorporation and plot
dn &lt;- IsotopicDistributionN("NECFLQHK", incorp = 0.00368)
dn$t &lt;- "incorp = 0.368%"
d50 &lt;- IsotopicDistributionN("NECFLQHK", incorp = 0.5)
d50$t &lt;- "incorp = 50%"
d99 &lt;- IsotopicDistributionN("NECFLQHK", incorp = 0.99)
d99$t &lt;- "incorp = 99%"
x &lt;- rbind(dn, d50, d99)
library(lattice)
xyplot(percent ~ mz | t, data = x,
    type = "h", 
    xlab = "m/z", 
    ylab = "intensity (%)",
    main = "Isotopic distribution of peptide NECFLQHK with 
            varying nitrogen-15 incorporation")
## Not run:          
## simulate peptide DVFLGMFLYPYAR with oxidized methionine
IsotopicDistributionN("DVFLGmFLYPYAR", incorp = 0.5, 
                      custom = list(code = "m", 
                                    elements = c(C=5, H=9, N=1, O=2, S=1)))

## End(Not run)


</code></pre>

<hr>
<h2 id='ListFormula'>Convert an Elemental Formula to a List</h2><span id='topic+ListFormula'></span>

<h3>Description</h3>

<p>Convert a character string representing an elemental
formula to a list representing the elemental formula.  The list can be
used as input to other functions.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ListFormula(elemental.formula)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ListFormula_+3A_elemental.formula">elemental.formula</code></td>
<td>
<p>character string representing the elemental formula.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To maintain compatibility with <code><a href="#topic+MolecularWeight">MolecularWeight</a></code>,
<code><a href="#topic+MonoisotopicMass">MonoisotopicMass</a></code>, and <code><a href="#topic+IsotopicDistribution">IsotopicDistribution</a></code>
the elemental formula can contain only C,H,N,O,S,P,Br,Cl,F,I,Si, and
Sn (I and Sn are default elements in <code>MolecularWeight</code> only). Elements not in this set will be ignored and a warning will be
generated. The function can handle repeated elements, but not element
multiplication indicated by parenthesis.
</p>
<p>Known issue: Lower case letters after numbers, such as the &ldquo;o&rdquo; in
&ldquo;C12oBr5H5&rdquo; will not be caught with a warning and the resulting
list will contain the incorrect number of elements.
</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>See Also</h3>

<p><code><a href="#topic+ConvertPeptide">ConvertPeptide</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>ListFormula("C14H8Cl4")
ListFormula("C6H5OH")
ListFormula("C15H12I3NO4")
</code></pre>

<hr>
<h2 id='MolecularWeight'>Calculate the molecular weight of an organic molecule.</h2><span id='topic+MolecularWeight'></span>

<h3>Description</h3>

<p>Given an elemental formula and the average relative atomic masses of the elements, determine the molecular weight.</p>


<h3>Usage</h3>

<pre><code class='language-R'>MolecularWeight(formula = list(), amu = list())</code></pre>


<h3>Arguments</h3>

  
<table>
<tr><td><code id="MolecularWeight_+3A_formula">formula</code></td>
<td>
<p>a list describing the elemental formula.  The allowed
elements are C, H, N, O, S, P, Br, Cl, F, I, Si, Sn, and a user defined &quot;x&quot;.  See Examples.</p>
</td></tr>
<tr><td><code id="MolecularWeight_+3A_amu">amu</code></td>
<td>
<p>a list specifing user defined standard atomic masses of the elements (in atomic mass units).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user defined <code>x</code> in the argument lists can be used to define an additional element, such as a metal.  See example.
</p>
<p>Due to apparent rounding differences at the elemental level, other sources of molecular weight information may vary in the second or third decimal place.
</p>
<p>This function will accept values that do not correspond to known physical reality, such as a fractional number of elements or the wrong standard atomic mass for an element.
</p>


<h3>Value</h3>

<p>The molecular weight of the molecule.</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>References</h3>

<p>The relative atomic masses of the elements are from the NIST Physical Reference Data Website <a href="https://www.nist.gov/pml/atomic-weights-and-isotopic-compositions-relative-atomic-masses">https://www.nist.gov/pml/atomic-weights-and-isotopic-compositions-relative-atomic-masses</a>.  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MonoisotopicMass">MonoisotopicMass</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>MolecularWeight(formula = list(C=2, H=4))

## Molecular weight of cyanocobalamin (C63H88CoN14O14P) with user defined cobalt
MolecularWeight(formula = list(C=63, H=88, N=14, O=14, P=1, x=1),
                amu = list(x = 58.933200))
                
## Molecular weight of triiodothyronine (C15H12I3NO4) using output from ListFormula
MolecularWeight(formula = ListFormula("C15H12I3NO4"))               
</code></pre>

<hr>
<h2 id='MonoisotopicMass'>Calculate the monoisotopic mass or monoisotopic m/z value of an organic molecule.</h2><span id='topic+MonoisotopicMass'></span>

<h3>Description</h3>

<p>Given an elemental formula, the relative atomic masses of the isotopes, and the charge state, determine the monoisotopic mass or monoisotopic m/z value.</p>


<h3>Usage</h3>

<pre><code class='language-R'>MonoisotopicMass(formula = list(), isotopes = list(), charge = 0)</code></pre>


<h3>Arguments</h3>

  
<table>
<tr><td><code id="MonoisotopicMass_+3A_formula">formula</code></td>
<td>
<p>a list describing the uncharged elemental formula.  The allowed elements are C, H, N, O, S, P, Br, Cl, F, Si, and a user defined &quot;x&quot;.  See Examples.</p>
</td></tr>
<tr><td><code id="MonoisotopicMass_+3A_isotopes">isotopes</code></td>
<td>
<p>a list specifing the relative atomic masses of the isotopes.</p>
</td></tr>
<tr><td><code id="MonoisotopicMass_+3A_charge">charge</code></td>
<td>
<p>an integer specifying the number of positive or
negative charges.  If a charge is specified, the m/z of the
molecule is calculated by adding or subtracting the specified
number of protons.  If <code>charge = 0</code>, the monoisotopic mass is
returned. See details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The elemental formula describes the uncharged molecule; the charge
argument will add or remove hydrogens as needed.  This assumes an
electrospray or MALDI type ionization. In electron impact ionization a
positive charge is generated by loss of an electron, not additon of H+,
therefore to calculate the correct m/z, the charge should be set to zero
as a workaround.
</p>
<p>The relative atomic masses of the most abundant
isotopes (carbon-12, hydrogen-1, nitrogen-14, and oxygen-16) are set as
the default values.  For reference, the relative atomic masses of some
common isotopic labels are:
</p>

<table>
<tr>
 <td style="text-align: left;">
carbon-13    </td><td style="text-align: left;"> 13.0033548378</td>
</tr>
<tr>
 <td style="text-align: left;">
hydrogen-2   </td><td style="text-align: left;"> 2.0141017780</td>
</tr>
<tr>
 <td style="text-align: left;">
nitrogen-15  </td><td style="text-align: left;"> 15.0001088984</td>
</tr>
<tr>
 <td style="text-align: left;">
oxygen-18    </td><td style="text-align: left;"> 17.9991604
</td>
</tr>

</table>

<p>The user defined element <code>x</code> can be used to define an additional element, such as a metal, or to specify a certain number of isotopically labeled atoms in a molecule.  See Examples.
</p>
<p>This function will accept values that do not correspond to known physical reality, such as a fractional number of elements, a fractional charge, or the wrong relative atomic mass value for an isotope.
</p>


<h3>Value</h3>

<p>The monoisotopic mass of the unchaged molecule or the monoisotopic m/z value of the charged molecule.</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder and Katharine M. Mullen</p>


<h3>References</h3>

<p>The relative atomic masses of the isotopes are from the NIST Physical Reference Data Website <a href="http://physics.nist.gov/PhysRefData/Compositions/">http://physics.nist.gov/PhysRefData/Compositions/</a>.  The molar mass of a proton (H+) is from the NIST CODATA Website <a href="http://physics.nist.gov/cuu/Constants/index.html">http://physics.nist.gov/cuu/Constants/index.html</a>.  
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Digest">Digest</a></code>, <code><a href="#topic+FragmentPeptide">FragmentPeptide</a></code>, <code><a href="#topic+MolecularWeight">MolecularWeight</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## monoisotopic m/z of creatinine (C4H7N3O), +1 charge
## unlabeled
MonoisotopicMass(formula = list(C=4, H=7, N=3, O=1), charge = 1)
## with all carbon-13 atoms
MonoisotopicMass(formula = list(C=4, H=7, N=3, O=1), 
                 isotopes = list(C = 13.0033548378),
                 charge = 1) 
## with 2 carbon-12 atoms and 2 carbon-13 atoms
MonoisotopicMass(formula = list(C=2, H=7, N=3, O=1, x=2), 
                 isotopes = list(x = 13.0033548378),
                 charge = 1)

## monoisotopic mass of cyanocobalamin (C63H88CoN14O14P)
MonoisotopicMass(formula = list(C=63, H=88, N=14, O=14, P=1, x=1),
                 isotopes = list(x = 58.9332002))
</code></pre>

<hr>
<h2 id='PeptideSpectrum'>Plot Annotated Peptide Fragmentation Mass Spectrum</h2><span id='topic+PeptideSpectrum'></span>

<h3>Description</h3>

<p>Identify the b- and y-ions or c- and z-ions in a peptide fragmentation mass spectrum given the sequence.  Generate a plot and table identifying the fragment ions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PeptideSpectrum(expt, theory, t = 0.4, b = 5, label = "", xlim = c(100, 1500), 
                supress = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PeptideSpectrum_+3A_expt">expt</code></td>
<td>
<p>data frame containing the experimental MS/MS peak list with the m/z values in the first column and corresponding intensities in the second</p>
</td></tr>
<tr><td><code id="PeptideSpectrum_+3A_theory">theory</code></td>
<td>
<p>data frame of theoretical fragment ions generated by <code><a href="#topic+FragmentPeptide">FragmentPeptide</a></code></p>
</td></tr>
<tr><td><code id="PeptideSpectrum_+3A_t">t</code></td>
<td>
<p>numeric value specifying the m/z tolerance for matching theoretical fragment ions to experimental peaks</p>
</td></tr>
<tr><td><code id="PeptideSpectrum_+3A_b">b</code></td>
<td>
<p>numeric value specifying the baseline threshold for peak identification.  Expressed as a percent of the maximum intensity.</p>
</td></tr>
<tr><td><code id="PeptideSpectrum_+3A_label">label</code></td>
<td>
<p>character string to label the spectrum.</p>
</td></tr>
<tr><td><code id="PeptideSpectrum_+3A_xlim">xlim</code></td>
<td>
<p>numeric vector of length 2, defining the beginning and ending values of the x-axis.</p>
</td></tr>
<tr><td><code id="PeptideSpectrum_+3A_supress">supress</code></td>
<td>
<p>logical specifying whether or not to supress the peak annotations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A graphics window of width = 10 and height = 5.5 seems to work well.  A crude attempt is made to prevent overlapping peak annotations.  If this is unsuccessful, use <code>supress = TRUE</code> and then add the annotations using the <code><a href="graphics.html#topic+text">text</a></code> function.  If two or more peak identifications are made within the set tolerance, all identifications will be printed, overlapping, on the plot.
</p>


<h3>Value</h3>

<p>Generates an annotated spectrum and a data frame showing the matches between the experimental peaks and the theoretical fragment ions.  The column names refer to the experimental <em>m/z</em> value (<code>expt_mz</code>), the experimental peak intensity (<code>expt_int</code>) and the error between the experimental and theoretical <em>m/z</em> values (<code>error</code>).  The other column names are the same as for <code><a href="#topic+FragmentPeptide">FragmentPeptide</a></code>.
</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>See Also</h3>

<p><code><a href="#topic+FragmentPeptide">FragmentPeptide</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## plot spectrum of peptide NIDALSGMEGR
t &lt;- FragmentPeptide("NIDALSGMEGR")   # generate theoretical fragment ions
PeptideSpectrum(example.spectrum.peptide, t, label = "CID", xlim = c(100, 1200))
mtext("Peptide fragmentation mass spectrum", line = 1)
</code></pre>

<hr>
<h2 id='ReadMspDirectory'>Read in MSP Formatted Mass Spectra</h2><span id='topic+ReadMspDirectory'></span>

<h3>Description</h3>

<p>Reads in all <code>.msp</code> files within a directory and makes a single concatenated data frame of <em>m/z</em> values and intensities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadMspDirectory(directory, skip = 2, comment.char = "", 
                 remove.placeholders = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadMspDirectory_+3A_directory">directory</code></td>
<td>
<p>character string. The path to the directory
containing the <code>.msp</code> files. Can be &ldquo;./&rdquo; for the current
directory on unix-like systems.</p>
</td></tr>
<tr><td><code id="ReadMspDirectory_+3A_skip">skip</code></td>
<td>
<p>integer. The number of lines at the top of the input file to skip before reading. Passed internally to read.table.</p>
</td></tr>
<tr><td><code id="ReadMspDirectory_+3A_comment.char">comment.char</code></td>
<td>
<p>a character vector of length one containing a single character or an empty string. Use &quot;&quot; to turn off the interpretation of comments altogether. Passed internally to read.table.</p>
</td></tr>
<tr><td><code id="ReadMspDirectory_+3A_remove.placeholders">remove.placeholders</code></td>
<td>
<p>TRUE or FALSE specifying if zero intensity peaks in the MSP file should be removed from the resulting data frame.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MSP is a National Institute of Standards and Technology (NIST) text format for centroid spectra.  This format is exported by some instrument software.
</p>
<p>The spectral files do not need to have the extension <code>.msp</code>. But an attempt will be made to process all files in the directory, regardless of the extension.  The presense of other file types may cause errors.
</p>
<p><code>ReadMspDirectory</code> is a wrapper for <code><a href="#topic+ReadMspFile">ReadMspFile</a></code>.
</p>


<h3>Value</h3>

<p>A data frame with the following column names:
</p>
<table>
<tr><td><code>mz</code></td>
<td>
<p>the centroid <em>m/z</em> values.</p>
</td></tr>
<tr><td><code>intensity</code></td>
<td>
<p>the respective intensities.</p>
</td></tr>
<tr><td><code>filename</code></td>
<td>
<p>the corresponding filename (used to identify the spectra within the data frame).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>See Also</h3>

<p><code><a href="#topic+ReadMspFile">ReadMspFile</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>## The package directory msp contains two .msp files 
ReadMspDirectory(paste(system.file(package = "OrgMassSpecR"), 
  "/extdata/msp", sep = ""))
</code></pre>

<hr>
<h2 id='ReadMspFile'>Read in a MSP Formatted Mass Spectrum</h2><span id='topic+ReadMspFile'></span>

<h3>Description</h3>

<p>Reads in a <code>.msp</code> file and makes a data frame of <em>m/z</em> values and intensities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadMspFile(file, skip = 2, comment.char = "", 
            remove.placeholders = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadMspFile_+3A_file">file</code></td>
<td>
<p>character string. The <code>.msp</code> filename.  Include the path to the filename if necessary.</p>
</td></tr>
<tr><td><code id="ReadMspFile_+3A_skip">skip</code></td>
<td>
<p>integer. The number of lines at the top of the input file to skip before reading. Passed internally to read.table.</p>
</td></tr>
<tr><td><code id="ReadMspFile_+3A_comment.char">comment.char</code></td>
<td>
<p>a character vector of length one containing a single character or an empty string. Use &quot;&quot; to turn off the interpretation of comments altogether. Passed internally to read.table.</p>
</td></tr>
<tr><td><code id="ReadMspFile_+3A_remove.placeholders">remove.placeholders</code></td>
<td>
<p>TRUE or FALSE specifying if zero intensity peaks in the MSP file should be removed from the resulting data frame.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MSP is a National Institute of Standards and Technology (NIST) ASCII format for centroid spectra. This format is exported by some instrument software.
</p>
<p>The spectral file does not need to have the extension <code>.msp</code>.
</p>


<h3>Value</h3>

<p>A data frame with the following column names:
</p>
<table>
<tr><td><code>mz</code></td>
<td>
<p>the centroid <em>m/z</em> values.</p>
</td></tr>
<tr><td><code>intensity</code></td>
<td>
<p>the respective intensities.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>See Also</h3>

<p><code><a href="#topic+ReadMspDirectory">ReadMspDirectory</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>ReadMspFile(paste(system.file(package = "OrgMassSpecR"), "/extdata/msp/pentaBDE.msp", sep = ""))
</code></pre>

<hr>
<h2 id='RetentionIndex'>Non-Isothermal Retention Index Calculation</h2><span id='topic+RetentionIndex'></span>

<h3>Description</h3>

<p>Determine the non-isothermal (temperature-programmed) gas chromatographic retention index of a target compound based on the retention times of adjacent n-alkanes.</p>


<h3>Usage</h3>

<pre><code class='language-R'>RetentionIndex(n, target, preceding, following)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RetentionIndex_+3A_n">n</code></td>
<td>
<p>the number of carbon atoms in the n-alkane eluting prior to the target compound.</p>
</td></tr>
<tr><td><code id="RetentionIndex_+3A_target">target</code></td>
<td>
<p>the retention time of the target compound</p>
</td></tr>
<tr><td><code id="RetentionIndex_+3A_preceding">preceding</code></td>
<td>
<p>the retention time of the n-alkane eluting prior to the target compound.</p>
</td></tr>
<tr><td><code id="RetentionIndex_+3A_following">following</code></td>
<td>
<p>the retention time of the n-alkane eluting after the target compound.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The retention time units can be either seconds or minutes, but should be consistent. The n-alkanes eluting before and after the target compound should differ by one carbon atom.  This equation does not apply to isothermal separations or when the n-alkane series contains only odd or even carbon atoms.</p>


<h3>Value</h3>

<p>A vector giving the retention index.</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>References</h3>

<p>Van Den Dool H, Kratz PD, <em>Journal of Chromatography</em>, 1963, 11, 463-471.</p>


<h3>Examples</h3>

<pre><code class='language-R'>RetentionIndex(8, 30, 28, 33)</code></pre>

<hr>
<h2 id='SpectrumSimilarity'>Similarity Between Two Mass Spectra</h2><span id='topic+SpectrumSimilarity'></span>

<h3>Description</h3>

<p>Generate a head-to-tail plot of the two mass spectra and calculate a similarity score.</p>


<h3>Usage</h3>

<pre><code class='language-R'>SpectrumSimilarity(spec.top, spec.bottom, t = 0.25, b = 10, 
                   top.label = NULL, bottom.label = NULL, 
                   xlim = c(50, 1200), x.threshold = 0, 
                   print.alignment = FALSE, print.graphic = TRUE,
                   output.list = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SpectrumSimilarity_+3A_spec.top">spec.top</code></td>
<td>
<p>data frame containing the experimental spectrum's peak list with the <em>m/z</em> values in the first column and corresponding intensities in the second</p>
</td></tr>
<tr><td><code id="SpectrumSimilarity_+3A_spec.bottom">spec.bottom</code></td>
<td>
<p>data frame containing the reference spectrum's peak list with the <em>m/z</em> values in the first column and corresponding intensities in the second</p>
</td></tr>
<tr><td><code id="SpectrumSimilarity_+3A_t">t</code></td>
<td>
<p>numeric value specifying the tolerance used to align the <em>m/z</em> values of the two spectra.</p>
</td></tr>
<tr><td><code id="SpectrumSimilarity_+3A_b">b</code></td>
<td>
<p>numeric value specifying the baseline threshold for peak identification.  Expressed as a percent of the maximum intensity.</p>
</td></tr>
<tr><td><code id="SpectrumSimilarity_+3A_top.label">top.label</code></td>
<td>
<p>character string to label the top spectrum.</p>
</td></tr>
<tr><td><code id="SpectrumSimilarity_+3A_bottom.label">bottom.label</code></td>
<td>
<p>character string to label the bottom spectrum.</p>
</td></tr>
<tr><td><code id="SpectrumSimilarity_+3A_xlim">xlim</code></td>
<td>
<p>numeric vector of length 2, defining the beginning and ending values of the x-axis.</p>
</td></tr>
<tr><td><code id="SpectrumSimilarity_+3A_x.threshold">x.threshold</code></td>
<td>
<p>numeric value of length 1 specifying the <em>m/z</em> threshold used for the similarity score calculation. Only peaks with <em>m/z</em> values above the threshold are used in the calculation. This can be used to exclude noise and/or non-specific ions at the low end of the spectrum. By default all ions are used.</p>
</td></tr>
<tr><td><code id="SpectrumSimilarity_+3A_print.alignment">print.alignment</code></td>
<td>
<p>TRUE or FALSE value specifying if a data frame showing the aligned <em>m/z</em> values should be printed. Default is FALSE. The data frame contains aligned peak intensities from the top and bottom spectra, and is used to check the results. It contains all peaks, not only those subsetted by the <code>b</code> and <code>x.threshold</code> arguments.</p>
</td></tr>
<tr><td><code id="SpectrumSimilarity_+3A_print.graphic">print.graphic</code></td>
<td>
<p>TRUE or FALSE value specifying if the head-to-tail plot should be printed. The defaut value is TRUE. The unaligned spectra are shown in the plot.</p>
</td></tr>
<tr><td><code id="SpectrumSimilarity_+3A_output.list">output.list</code></td>
<td>
<p>TRUE or FALSE. If TRUE, will return a list with the similarity score, alignment dataframe, and plot. The plot is a gTree (grid graphics) object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The mass spectral similarity score is calculated as (where <code class="reqn">\cdot</code> is the dot product) </p>
<p style="text-align: center;"><code class="reqn">\cos \theta = \frac{u \cdot v}{\sqrt{u \cdot u} \sqrt{v \cdot v}}</code>
</p>
<p> where <code class="reqn">u</code> and <code class="reqn">v</code> are the aligned intensity vectors of the two spectra, as subsetted by the <code>b</code> and <code>x.threshold</code> arguments.  The <code>t</code> argument is used to align the intensities.  The bottom spectum is used as the reference spectrum, and the <em>m/z</em> values of peaks in the top spectrum that are within <code>t</code> of a reference <em>m/z</em> value are paired with that reference peak.  Ideally, a single peak from the top spectrum should be paired with a single peak from the reference spectrum. Peaks without a match are paired with an intensity of zero.
</p>
<p>Note that, although both are based on the cosine of the two intensity vectors, the spectral similarity score given by <code>SpectrumSimilarity</code> is not the same as that given by the NIST MS Search program, described in the reference below. 
</p>


<h3>Value</h3>

<p>A vector containing the similarity score. By default a base graphics head-to-tail plot of the two mass spectra is printed, and optionally the data frame showing the peak alignment is printed. Alternatively, if <code>output.list = TRUE</code> the function will return a list with the similarity score, alignment dataframe, and plot. In this case the plot is a gTree (grid graphics) object.</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>References</h3>

<p>&quot;Optimization and Testing of Mass Spectral Library Search Algorithms for Compound Identification,&quot; Stein SE, Scott DR, <em>Journal of the American Society for Mass Spectrometry</em>, 1994, 5, 859-866.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PeptideSpectrum">PeptideSpectrum</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
## With output.list = FALSE (default)
SpectrumSimilarity(example.spectrum.unknown, example.spectrum.authentic,
                   top.label = "unknown, electron impact", 
                   bottom.label = "derivatized alanine, electron impact",
                   xlim = c(25, 350))
# label peaks
plot.window(xlim = c(25,350), ylim = c(-125, 125))
text(c(73, 147, 158, 232, 260), c(100, 23, 44, 22, 15) + 10,
     c(73, 147, 158, 232, 260), cex = 0.75)
text(c(73, 147, 158, 232, 260), -c(100, 47, 74, 33, 20) - 10, 
     c(73, 147, 158, 232, 260), cex = 0.75)
mtext("Spectrum similarity", line = 1)

## With output.list = TRUE
x &lt;- SpectrumSimilarity(example.spectrum.unknown, example.spectrum.authentic,
                   top.label = "unknown, electron impact", 
                   bottom.label = "derivatized alanine, electron impact",
                   xlim = c(25, 350), output.list = TRUE)

print(x)
grid.newpage()
grid.draw(x$plot) # draw to device
</code></pre>

<hr>
<h2 id='WriteMspFile'>Write Spectra to the NIST MSP Text Format</h2><span id='topic+WriteMspFile'></span>

<h3>Description</h3>

<p>Writes all spectra in the OrgMassSpecR data frame format to
a single text file in NIST MSP format. The MSP files can be imported
into the NIST MS Search program to make a custom seachable library. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WriteMspFile(spectra, metadata, filename = "library.msp", comment = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WriteMspFile_+3A_spectra">spectra</code></td>
<td>
<p>data frame containing the mass spectra.</p>
</td></tr>
<tr><td><code id="WriteMspFile_+3A_metadata">metadata</code></td>
<td>
<p>data frame containing the metadata.</p>
</td></tr>
<tr><td><code id="WriteMspFile_+3A_filename">filename</code></td>
<td>
<p>character string. Name (including path if needed) of
the MSP text file.</p>
</td></tr>
<tr><td><code id="WriteMspFile_+3A_comment">comment</code></td>
<td>
<p>charater string applied to the COMMENT field of the MSP
file. The comment is applied to all spectra.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The spectra and metadata data frame format is
specified in the help files for the <code>LibraryReport</code> functions
in the spectral library packages, for example see
<code>SpecLibExample::LibraryReport</code>. The spectra data frame must
contain at least columns: <code>filename</code>, <code>mz</code>, and
<code>intensity</code>. The metadata data frame must contain at least columns: <code>filename</code> and <code>compound</code>.
</p>
<p>The NAME field for each spectrum in the MSP file is assigned
based on <code>compound</code> in the metadata data frame. 
</p>
<p>Newlines are CR+LF since the NIST MS Search program is
Windows based.
</p>


<h3>Value</h3>

<p>An MSP file is written using a <code><a href="base.html#topic+file">file</a></code> connection.
</p>


<h3>Author(s)</h3>

<p>Nathan G. Dodder</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Simple example to illustrate the input data formats. 

spectra &lt;- data.frame(filename = c(rep("Spectrum A", 2), rep("Spectrum B", 2)),
                      mz = c(50, 51, 100, 101),
                      intensity = c(70, 71, 90, 91))

metadata &lt;- data.frame(filename = c("Spectrum A", "Spectrum B"),
                       compound = c("Compound A", "Compound B"))

## Not run: 
WriteMspFile(spectra = spectra, 
             metadata = metadata,
             filename = "Test.txt",
             comment = "Test Comment")

## End(Not run)

## Example using SpecLibExample package located at http://OrgMassSpec.github.io/libraries.html.
## Not run: 
library(SpecLibExample)
WriteMspFile(spectra = spec,
             metadata = meta,
             comment = "example output")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
