<!DOCTYPE html><html><head><title>Help for package ORIClust</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ORIClust}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ORIClust-package'>
<p>Order-restricted Information Criterion-based Clustering Algorithm</p></a></li>
<li><a href='#Breast'><p>Breast cancer cell line data</p></a></li>
<li><a href='#complete.profile'><p>complete.profile</p></a></li>
<li><a href='#cyclical.max.min'><p>cyclical.max.min</p></a></li>
<li><a href='#cyclical.min.max'><p>cyclical.min.max</p></a></li>
<li><a href='#decreasing'><p>decreasing</p></a></li>
<li><a href='#down.up'><p>down.up</p></a></li>
<li><a href='#flat.pattern'><p>flat.pattern</p></a></li>
<li><a href='#increasing'><p>increasing</p></a></li>
<li><a href='#isodecre'><p>isodecre</p></a></li>
<li><a href='#isoincre'><p>isoincre</p></a></li>
<li><a href='#ORICC1'><p>One-stage ORICC</p></a></li>
<li><a href='#ORICC2'>
<p>Two-stage ORICC</p></a></li>
<li><a href='#up.down'><p>up.down</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Order-Restricted Information Criterion-Based Clustering
Algorithm</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0-2</td>
</tr>
<tr>
<td>Date:</td>
<td>2009-09-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tianqing Liu &lt;tianqingliu@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A user-friendly R-based software package for
        gene clustering. Clusters are given by genes matched to
        prespecified profiles across various ordered treatment groups.
        It is particularly useful for analyzing data obtained from
        short time-course or dose-response microarray experiments.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-06-23 06:56:25 UTC; hornik</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-06-23 07:18:42 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
</table>
<hr>
<h2 id='ORIClust-package'>
Order-restricted Information Criterion-based Clustering Algorithm
</h2><span id='topic+ORIClust-package'></span>

<h3>Description</h3>

<p>ORIClust is a user-friendly R-based software package
for gene clustering. Clusters are given by genes matched to prespecified
profiles across various ordered treatment groups. It is
particularly useful for analyzing data obtained from short time-course
or dose-response microarray experiments.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> ORIClust</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2009-05-24</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The main functions are <code><a href="#topic+ORICC1">ORICC1</a></code> and <code><a href="#topic+ORICC2">ORICC2</a></code>, see the documentation files with examples.
</p>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:  Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments, BMC Bioinformatics, 10: 146.
</p>

<hr>
<h2 id='Breast'>Breast cancer cell line data</h2><span id='topic+Breast'></span>

<h3>Description</h3>

<p> This data set comes from a breast cancer cell line
microarray study. The experiment was done as
follows. First, the MCF-7 breast cancer cell line was treated
with 17 beta-estradiol or ethanol (vehicle control). Then,
samples were harvested at 1, 4, 12, 24, 36 and 48 hours
after treatment. At each time point, M = 8 replicate arrays
were prepared with each array consisting of G = 1901
genes.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Breast</code></pre>


<h3>Format</h3>

<p>A matrix containing 1901 rows and 50 columns.</p>


<h3>References</h3>

<p>Lobenhofer, E., Bennett, L., Cable, P., Li, L., Bushel, P., and Afshari, C. (2002), <em>Regulation of DNA replication fork genes by 17 beta-estradiol.</em>
Molec. Endocrin., <b>16</b>, 1215-1229.
</p>

<hr>
<h2 id='complete.profile'>complete.profile</h2><span id='topic+complete.profile'></span>

<h3>Description</h3>

 
<p>Returns the log-maximum likelihood and the estimator of the mean when there is no inequality constraint.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
complete.profile(data,x,n.rep)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="complete.profile_+3A_data">data</code></td>
<td>

<p>A vector containing the expressions of one gene.
</p>
</td></tr>
<tr><td><code id="complete.profile_+3A_x">x</code></td>
<td>
<p>A vector consisting of the average expression at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="complete.profile_+3A_n.rep">n.rep</code></td>
<td>
<p>A vector consisting of the number of replicate arrays at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>logelr</code></td>
<td>
<p> Log-maximum likelihood 
</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>A vector containing the estimator of the mean
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:Tianqing Liu &lt;tianqingliu@gmail.com&gt; 
</p>

<hr>
<h2 id='cyclical.max.min'>cyclical.max.min</h2><span id='topic+cyclical.max.min'></span>

<h3>Description</h3>

 
<p>Returns the log-maximum likelihood and the estimator of the mean under cyclical profile with maximum at <code>max1</code> and minimum at <code>min1</code> (<code>max1</code> &lt; <code>min1</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
cyclical.max.min(data,x,n.rep,max1,min1)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cyclical.max.min_+3A_data">data</code></td>
<td>

<p>A vector containing the expressions of one gene.
</p>
</td></tr>
<tr><td><code id="cyclical.max.min_+3A_x">x</code></td>
<td>
<p>A vector consisting of the average expression at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="cyclical.max.min_+3A_n.rep">n.rep</code></td>
<td>
<p>A vector consisting of the number of replicate arrays at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="cyclical.max.min_+3A_max1">max1</code></td>
<td>
<p>Cyclical profile with maximum at <code>max1</code>. 
</p>
</td></tr>
<tr><td><code id="cyclical.max.min_+3A_min1">min1</code></td>
<td>
<p>Cyclical profile with minimum at <code>min1</code>. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>logelr</code></td>
<td>
<p> Log-maximum likelihood 
</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>A vector containing the estimator of the mean
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:  Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), <em>Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments.</em> BMC Bioinformatics, <b>10</b>: 146.
</p>
<p>Robertson, T., Wright, F. T. and Dykstra, R. L. (1988). <em>Order restricted statistical inference.</em> New York: Wiley.
</p>
<p>Shi, N., Gao, W. and Zhang, B. (2001). <em>One sided estimation and testing
problems for location models from grouped samples.</em>
Commun Statist-Simula, <b>30</b>: 885-898.
</p>

<hr>
<h2 id='cyclical.min.max'>cyclical.min.max</h2><span id='topic+cyclical.min.max'></span>

<h3>Description</h3>

 
<p>Returns the log-maximum likelihood and the estimator of the mean under cyclical profile with minimum at <code>min1</code> and maximum at <code>max1</code> (<code>min1</code> &lt; <code>max1</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cyclical.min.max(data,x,n.rep,min1,max1)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cyclical.min.max_+3A_data">data</code></td>
<td>

<p>A vector containing the expressions of one gene.
</p>
</td></tr>
<tr><td><code id="cyclical.min.max_+3A_x">x</code></td>
<td>
<p> A vector consisting of the average expression at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="cyclical.min.max_+3A_n.rep">n.rep</code></td>
<td>
<p> A vector consisting of the number of replicate arrays at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="cyclical.min.max_+3A_min1">min1</code></td>
<td>
<p> Cyclical profile with minimum at <code>min1</code>
</p>
</td></tr>
<tr><td><code id="cyclical.min.max_+3A_max1">max1</code></td>
<td>
<p> cyclical profile with maximum at <code>max1</code> 
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>logelr</code></td>
<td>
<p> Log-maximum likelihood 
</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>A vector containing the estimator of the mean
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:  Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), <em>Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments.</em> BMC Bioinformatics, <b>10</b>: 146.
</p>
<p>Robertson, T., Wright, F. T. and Dykstra, R. L. (1988). <em>Order restricted statistical inference.</em> New York: Wiley.
</p>
<p>Shi, N., Gao, W. and Zhang, B. (2001). <em>One sided estimation and testing
problems for location models from grouped samples.</em>
Commun Statist-Simula, <b>30</b>: 885-898.
</p>

<hr>
<h2 id='decreasing'>decreasing</h2><span id='topic+decreasing'></span>

<h3>Description</h3>

 
<p>Returns the log-maximum likelihood and the estimator of the mean under the monotone decreasing profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
decreasing(data,x,n.rep)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="decreasing_+3A_data">data</code></td>
<td>

<p>A vector containing the expressions of one gene.
</p>
</td></tr>
<tr><td><code id="decreasing_+3A_x">x</code></td>
<td>
<p>A vector consisting of the average expression at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="decreasing_+3A_n.rep">n.rep</code></td>
<td>
<p>A vector consisting of the number of replicate arrays at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>logelr</code></td>
<td>
<p> Log-maximum likelihood 
</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>A vector containing the estimator of the mean
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:  Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), <em>Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments.</em> BMC Bioinformatics, <b>10</b>: 146.
</p>
<p>Robertson, T., Wright, F. T. and Dykstra, R. L. (1988). <em>Order restricted statistical inference.</em> New York: Wiley.
</p>
<p>Shi, N., Gao, W. and Zhang, B. (2001). <em>One sided estimation and testing
problems for location models from grouped samples.</em>
Commun Statist-Simula, <b>30</b>: 885-898.
</p>

<hr>
<h2 id='down.up'>down.up</h2><span id='topic+down.up'></span>

<h3>Description</h3>

 
<p>Returns the log-maximum likelihood and the estimator of the mean under down-up profile with minimum at <code>h</code>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
down.up(data,x,n.rep,h)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="down.up_+3A_data">data</code></td>
<td>

<p>A vector containing the expressions of one gene.
</p>
</td></tr>
<tr><td><code id="down.up_+3A_x">x</code></td>
<td>
<p>A vector consisting of the average expression at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="down.up_+3A_n.rep">n.rep</code></td>
<td>
<p>A vector consisting of the number of replicate arrays at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="down.up_+3A_h">h</code></td>
<td>
<p>Down-up profile with minimum at <code>h</code>. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>logelr</code></td>
<td>
<p> Log-maximum likelihood 
</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>A vector containing the estimator of the mean
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:  Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), <em>Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments.</em> BMC Bioinformatics, <b>10</b>: 146.
</p>
<p>Robertson, T., Wright, F. T. and Dykstra, R. L. (1988). <em>Order restricted statistical inference.</em> New York: Wiley.
</p>
<p>Shi, N., Gao, W. and Zhang, B. (2001). <em>One sided estimation and testing
problems for location models from grouped samples.</em>
Commun Statist-Simula, <b>30</b>: 885-898.
</p>

<hr>
<h2 id='flat.pattern'>flat.pattern</h2><span id='topic+flat.pattern'></span>

<h3>Description</h3>

 
<p>Returns the log-maximum likelihood and the estimator of the mean under the equality constraint that all means are equal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>flat.pattern(data,x,n.rep)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="flat.pattern_+3A_data">data</code></td>
<td>

<p>A vector containing the expressions of one gene.
</p>
</td></tr>
<tr><td><code id="flat.pattern_+3A_x">x</code></td>
<td>
<p>A vector consisting of the average expression at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="flat.pattern_+3A_n.rep">n.rep</code></td>
<td>
<p>A vector consisting of the number of replicate arrays at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>logelr</code></td>
<td>
<p> Log-maximum likelihood 
</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>A vector containing the estimator of the mean
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:  Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), <em>Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments.</em> BMC Bioinformatics, <b>10</b>: 146.
</p>
<p>Robertson, T., Wright, F. T. and Dykstra, R. L. (1988). <em>Order restricted statistical inference.</em> New York: Wiley.
</p>
<p>Shi, N., Gao, W. and Zhang, B. (2001). <em>One sided estimation and testing
problems for location models from grouped samples.</em>
Commun Statist-Simula, <b>30</b>: 885-898.
</p>

<hr>
<h2 id='increasing'>increasing</h2><span id='topic+increasing'></span>

<h3>Description</h3>

 
<p>Returns the log-maximum likelihood and the estimator of the mean under the monotone increasing profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
increasing(data,x,n.rep)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="increasing_+3A_data">data</code></td>
<td>

<p>A vector containing the expressions of one gene.
</p>
</td></tr>
<tr><td><code id="increasing_+3A_x">x</code></td>
<td>
<p>A vector consisting of the average expression at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="increasing_+3A_n.rep">n.rep</code></td>
<td>
<p>A vector consisting of the number of replicate arrays at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>logelr</code></td>
<td>
<p> Log-maximum likelihood 
</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>A vector containing the estimator of the mean
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:  Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), <em>Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments.</em> BMC Bioinformatics, <b>10</b>: 146.
</p>
<p>Robertson, T., Wright, F. T. and Dykstra, R. L. (1988). <em>Order restricted statistical inference.</em> New York: Wiley.
</p>
<p>Shi, N., Gao, W. and Zhang, B. (2001). <em>One sided estimation and testing
problems for location models from grouped samples.</em>
Commun Statist-Simula, <b>30</b>: 885-898.
</p>

<hr>
<h2 id='isodecre'>isodecre</h2><span id='topic+isodecre'></span>

<h3>Description</h3>

 
<p>Isotonic regression of <code>a</code> with weights <code>w</code> under monotone decreasing profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isodecre(a, w)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isodecre_+3A_a">a</code></td>
<td>
<p>A vector consisting of the average expression at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="isodecre_+3A_w">w</code></td>
<td>
<p> The weights.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>is</code></td>
<td>
<p>A vector containing the estimator of the mean
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:  Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), <em>Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments.</em> BMC Bioinformatics, <b>10</b>: 146.
</p>
<p>Robertson, T., Wright, F. T. and Dykstra, R. L. (1988). <em>Order restricted statistical inference.</em> New York: Wiley.
</p>
<p>Shi, N., Gao, W. and Zhang, B. (2001). <em>One sided estimation and testing
problems for location models from grouped samples.</em>
Commun Statist-Simula, <b>30</b>: 885-898.
</p>

<hr>
<h2 id='isoincre'>isoincre</h2><span id='topic+isoincre'></span>

<h3>Description</h3>

 
<p>Isotonic regression of <code>a</code> with weights <code>w</code> under monotone increasing profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isoincre(a, w)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isoincre_+3A_a">a</code></td>
<td>
<p>A vector consisting of the average expression at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="isoincre_+3A_w">w</code></td>
<td>
<p> The weights.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>is</code></td>
<td>
<p>A vector containing the estimator of the mean
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:  Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), <em>Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments.</em> BMC Bioinformatics, <b>10</b>: 146.
</p>
<p>Robertson, T., Wright, F. T. and Dykstra, R. L. (1988). <em>Order restricted statistical inference.</em> New York: Wiley.
</p>
<p>Shi, N., Gao, W. and Zhang, B. (2001). <em>One sided estimation and testing
problems for location models from grouped samples.</em>
Commun Statist-Simula, <b>30</b>: 885-898.
</p>

<hr>
<h2 id='ORICC1'>One-stage ORICC</h2><span id='topic+ORICC1'></span>

<h3>Description</h3>

<p> One-stage ORICC is a computationally efficient information criterion-based clustering algorithm for selecting and
clustering genes according to their time-course or dose-response
profiles. This algorithm takes account of the ordering in time-course or dose-response experiments by 
embedding the order-restricted inference into a model selection framework. This algorithm mainly consist of two steps. In the first step,
candidate  profiles are defined in terms of inequalities
among mean expression levels at different time points or doses levels. In the
second step, genes are assigned to the best matched profiles
determined by an information criterion for
order-restricted inference.</p>


<h3>Usage</h3>

<pre><code class='language-R'>
ORICC1(data,data.col,id.col,n.rep,n.top,transform,
       name.profile,cyclical.profile,complete.profile,
       onefile,plot.format)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ORICC1_+3A_data">data</code></td>
<td>

<p>A matrix containing the gene expressions.
</p>
</td></tr>
<tr><td><code id="ORICC1_+3A_data.col">data.col</code></td>
<td>
<p>Column indices of the gene expression data.
</p>
</td></tr>
<tr><td><code id="ORICC1_+3A_id.col">id.col</code></td>
<td>
<p>Column index of the gene ID. Defaults to 1. 
</p>
</td></tr>
<tr><td><code id="ORICC1_+3A_n.rep">n.rep</code></td>
<td>
<p>A vector consisting of the number of replicate arrays at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="ORICC1_+3A_n.top">n.top</code></td>
<td>
<p>The number of genes kept for the final clustering result. Genes are ranked based on expression variation across time or dose levels. Defaults to all genes ORICC1 selects
</p>
</td></tr>
<tr><td><code id="ORICC1_+3A_transform">transform</code></td>
<td>
<p>Transformation of the original data:
</p>
<p>0=None, 1=natural log, 2=square root, 3=cubic root.  Defaults to 0.      
</p>
</td></tr>
<tr><td><code id="ORICC1_+3A_name.profile">name.profile</code></td>
<td>
<p>A character string specifying the collection of candidate profiles. This option only supports monotone, up-down and down-up profiles specified as by 
</p>
<p><code>"decreasing";</code> 
</p>
<p><code>"increasing".</code>
</p>
<p><code>paste("up down max at",i,sep=" ");</code>
</p>
<p><code>paste("down up min at",j,sep=" ");</code>
</p>
<p>If <code>name.profile="all"</code>, the &lsquo;decreasing&rsquo;, &lsquo;increasing&rsquo; and all &lsquo;up-down&rsquo; and &lsquo;down-up&rsquo; profiles will be included. 
</p>
<p>If <code>name.profile=NULL</code>, &lsquo;decreasing&rsquo;, &lsquo;increasing&rsquo; and all &lsquo;up-down&rsquo; and &lsquo;down-up&rsquo; profiles will be absent. Defaults to NULL.
</p>
<p>One can also specify  several up-down or down-up profiles together as follows. 
</p>
<p><code>profile1=paste("up down max at",c(2,4),sep=" ");</code>
</p>
<p><code>profile2=paste("down up min at",c(3,5),sep=" ");</code>
</p>
<p><code>name.profile=c(profile1,profile2);</code>
</p>
<p>then up-down profile with maxima at 2 and 4 as well as down-up profile with  minima  at 3 and 5 will be included. 
</p>
</td></tr>
<tr><td><code id="ORICC1_+3A_cyclical.profile">cyclical.profile</code></td>
<td>
<p>A matrix with 2 columns. Each element of the matrix must be  a number in the set {2,3,...,T-1}.  Each row of the matrix represents a cyclical profile with minima at the first entry of the row and maxima at the 2nd entry. As a result, two elements in the same row must be different.  For example, if 
</p>
<p><code>cyclical.profile=matrix(c(2,3,4,3),2,2,byrow=T),</code>
then the cyclical profile with minima at 2 and maxima at 3 and the cyclical profile with minima at 4 and maxima at 3 will be included as candidate profiles.
</p>
<p>If <code>cyclical.profile=NULL</code>, all cyclical profiles will be absent. Defaults to NULL.
</p>
</td></tr>
<tr><td><code id="ORICC1_+3A_complete.profile">complete.profile</code></td>
<td>
<p>The <code>complete.profile</code> means a profile in which there is no defined inequality constraint. 
</p>
<p>If the <code>complete.profile</code> is a candidate profile, 
</p>
<p><code>complete.profile=1</code>, otherwise, 
</p>
<p><code>complete.profile=NULL</code>. Defaults to NULL.
</p>
</td></tr>
<tr><td><code id="ORICC1_+3A_onefile">onefile</code></td>
<td>
<p>logical: if true (the default) multiple figures for different clusters are output in one file. If FALSE, each cluster is plotted in a seperate file. Defaults to TRUE. 
</p>
</td></tr>
<tr><td><code id="ORICC1_+3A_plot.format">plot.format</code></td>
<td>
<p>The format of the output file containing plots of gene clusters.Users can choose between &lsquo;eps&rsquo; and &lsquo;jpg&rsquo;. Defaults to &lsquo;eps&rsquo;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The gene expression dataset should be in a tab-delimited txt file, in which the first two columns contain the gene names and their
accession numbers or descriptions, and the remaining columns, in their orders, are the geneexpression data (contain multiple columns, i.e. <code>data.col</code>).The dataset is assumed to have been processed so that each row contains the expressions of only one gene.
</p>


<h3>Value</h3>

<p>The results are displayed in a graphical form. The graphics can be stored in a JPG or EPS format. Both the raw gene expression values and the estimated mean expressions are output to external files &lsquo;cluster of raw data.txt&rsquo; and &lsquo;cluster of fitted mean data.txt&rsquo;, respectively. 
</p>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer: Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), <em>Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments.</em> BMC Bioinformatics, <b>10</b>: 146.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Breast)
ORICC1(Breast,data.col=3:50,id.col=1,n.rep=rep(8,6),
       n.top=50,transform=1,name.profile="all",plot.format="eps")
</code></pre>

<hr>
<h2 id='ORICC2'>
Two-stage ORICC
</h2><span id='topic+ORICC2'></span>

<h3>Description</h3>

<p>It is a computationally efficient two-stage algorithm by adding a pre-screening stage. It first screens out genes that show no significant changes over time, and then applies the one-stage algorithm to a much smaller set of remained genes. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
ORICC2(data,data.col,id.col,n.rep,n.top,transform, 
       name.profile,cyclical.profile, 
       onefile,plot.format)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ORICC2_+3A_data">data</code></td>
<td>

<p>A matrix containing the gene expressions.
</p>
</td></tr>
<tr><td><code id="ORICC2_+3A_data.col">data.col</code></td>
<td>
<p>Column indices of the gene expression data.
</p>
</td></tr>
<tr><td><code id="ORICC2_+3A_id.col">id.col</code></td>
<td>
<p>Column index of the gene ID. Defaults to 1. 
</p>
</td></tr>
<tr><td><code id="ORICC2_+3A_n.rep">n.rep</code></td>
<td>
<p>A vector consisting of the number of replicate arrays at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="ORICC2_+3A_n.top">n.top</code></td>
<td>
<p>The number of genes kept for the final clustering result. Genes are ranked based on expression variation across time or dose levels. Defaults to all genes ORICC2 selects
</p>
</td></tr>
<tr><td><code id="ORICC2_+3A_transform">transform</code></td>
<td>
<p>Transformation of the original data:
</p>
<p>0=None, 1=natural log, 2=square root, 3=cubic root.  Defaults to 0.      
</p>
</td></tr>
<tr><td><code id="ORICC2_+3A_name.profile">name.profile</code></td>
<td>
<p>A character string specifying the collection of candidate profiles. This option only supports monotone, up-down and down-up profiles specified as by 
</p>
<p><code>"decreasing";</code> 
</p>
<p><code>"increasing".</code>
</p>
<p><code>paste("up down max at",i,sep=" ");</code>
</p>
<p><code>paste("down up min at",j,sep=" ");</code>
</p>
<p>If <code>name.profile="all"</code>, the &lsquo;decreasing&rsquo;, &lsquo;increasing&rsquo; and all &lsquo;up-down&rsquo; and &lsquo;down-up&rsquo; profiles will be included. 
</p>
<p>If <code>name.profile=NULL</code>, &lsquo;decreasing&rsquo;, &lsquo;increasing&rsquo; and all &lsquo;up-down&rsquo; and &lsquo;down-up&rsquo; profiles will be absent. Defaults to NULL.
</p>
<p>One can also specify  several up-down or down-up profiles together as follows. 
</p>
<p><code>profile1=paste("up down max at",c(2,4),sep=" ");</code>
</p>
<p><code>profile2=paste("down up min at",c(3,5),sep=" ");</code>
</p>
<p><code>name.profile=c(profile1,profile2);</code>
</p>
<p>then up-down profile with maxima at 2 and 4 as well as down-up profile with  minima  at 3 and 5 will be included. 
</p>
</td></tr>
<tr><td><code id="ORICC2_+3A_cyclical.profile">cyclical.profile</code></td>
<td>
<p>A matrix with 2 columns. Each element of the matrix must be  a number in the set {2,3,...,T-1 }.  Each row of the matrix represents a cyclical profile with minima at the first entry of the row and maxima at the 2nd entry. As a result, two elements in the same row must be different.  For example, if 
</p>
<p><code>cyclical.profile=matrix(c(2,3,4,3),2,2,byrow=T),</code>
then the cyclical profile with minima at 2 and maxima at 3 and the cyclical profile with minima at 4 and maxima at 3 will be included as candidate profiles.
</p>
<p>If <code>cyclical.profile=NULL</code>, all cyclical profiles will be absent. Defaults to NULL.
</p>
</td></tr>
<tr><td><code id="ORICC2_+3A_onefile">onefile</code></td>
<td>
<p>logical: if true (the default) multiple figures for different clusters are output in one file. If FALSE, each cluster is plotted in a seperate file. Defaults to TRUE. 
</p>
</td></tr>
<tr><td><code id="ORICC2_+3A_plot.format">plot.format</code></td>
<td>
<p>The format of the output file containing plots of gene clusters.Users can choose between &lsquo;eps&rsquo; and &lsquo;jpg&rsquo;. Defaults to &lsquo;eps&rsquo;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The gene expression dataset should be in a tab-delimited txt file, in which the first two columns contain the gene names and their
accession numbers or descriptions, and the remaining columns, in their orders, are the geneexpression data (contain multiple columns, i.e. <code>data.col</code>).The dataset is assumed to have been processed so that each row contains the expressions of only one gene.
</p>


<h3>Value</h3>

<p>The results are displayed in a graphical form. The graphics can be stored in a JPG or EPS format. Both the raw gene expression values and the estimated mean expressions are output to external files &lsquo;cluster of raw data.txt&rsquo; and &lsquo;cluster of fitted mean data.txt&rsquo;, respectively. 
</p>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:  Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), <em>Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments.</em> BMC Bioinformatics, <b>10</b>: 146.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(Breast)
ORICC2(Breast,data.col=3:50,id.col=1,n.rep=rep(8,6),
       n.top=50,transform=1,name.profile="all",plot.format="eps")
</code></pre>

<hr>
<h2 id='up.down'>up.down</h2><span id='topic+up.down'></span>

<h3>Description</h3>

 
<p>Returns the log-maximum likelihood and the estimator of the mean under under up-down profile with maximum at <code>h</code>. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
up.down(data,x,n.rep,h)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="up.down_+3A_data">data</code></td>
<td>

<p>A vector containing the expressions of one gene.
</p>
</td></tr>
<tr><td><code id="up.down_+3A_x">x</code></td>
<td>
<p>A vector consisting of the average expression at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="up.down_+3A_n.rep">n.rep</code></td>
<td>
<p>A vector consisting of the number of replicate arrays at time points (1, 2,... ,T), where T is the total number of time points.
</p>
</td></tr>
<tr><td><code id="up.down_+3A_h">h</code></td>
<td>
<p> Up-down profile with maximum at <code>h</code>. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>logelr</code></td>
<td>
<p> Log-maximum likelihood 
</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>A vector containing the estimator of the mean
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Tianqing Liu, Nan Lin, Ningzhong Shi and Baoxue Zhang 
</p>
<p>Maintainer:  Tianqing Liu &lt;tianqingliu@gmail.com&gt;
</p>


<h3>References</h3>

<p>Liu, T., Lin, N., Shi, N. and Zhang, B. (2009), <em>Information criterion-based clustering with order-restricted candidate profiles in short time-course microarray experiments.</em> BMC Bioinformatics, <b>10</b>: 146.
</p>
<p>Robertson, T., Wright, F. T. and Dykstra, R. L. (1988). <em>Order restricted statistical inference.</em> New York: Wiley.
</p>
<p>Shi, N., Gao, W. and Zhang, B. (2001). <em>One sided estimation and testing
problems for location models from grouped samples.</em>
Commun Statist-Simula, <b>30</b>: 885-898.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
