<!DOCTYPE html><html lang="en"><head><title>Help for package orthGS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {orthGS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#A_selected'><p>Adjacency Matrix for Orthology Graph</p></a></li>
<li><a href='#agf'><p>Angiosperms Gymnosperms Ferns</p></a></li>
<li><a href='#AngGym'><p>Angiosperms Gymnosperms</p></a></li>
<li><a href='#coltips'><p>Colouring Tree Tips</p></a></li>
<li><a href='#gapless_msa'><p>Remove Gaps in a MSA</p></a></li>
<li><a href='#getseqGS'><p>Get the GS Sequence</p></a></li>
<li><a href='#madRoot'><p>Find The Root of a Phylogenetic Tree Using MAD Method</p></a></li>
<li><a href='#mapTrees'><p>Map Gene Tree into Species Tree</p></a></li>
<li><a href='#mltree'><p>Build Up a ML Tree</p></a></li>
<li><a href='#msa'><p>Multiple Sequence Alignment</p></a></li>
<li><a href='#orthG'><p>Infer GS OrthoGroups Within a Set of Species</p></a></li>
<li><a href='#orthP'><p>Search Orthologous of a Given Protein</p></a></li>
<li><a href='#sdf'><p>Seed Plants and Ferns GS</p></a></li>
<li><a href='#selected_tr'><p>Ultrametric Rooted Seed Plants Tree</p></a></li>
<li><a href='#speciesGS'><p>Map Species Names</p></a></li>
<li><a href='#subsetGS'><p>GS Proteins Report</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Orthology vs Paralogy Relationships among Glutamine Synthetase
from Plants</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.6</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools to analyze and infer orthology and paralogy relationships between glutamine synthetase proteins in seed plants.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, bio3d, castor, igraph, phangorn, phytools, seqinr,
TreeTools</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BiocManager, Biostrings, knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-30 10:08:12 UTC; jcaledo</td>
</tr>
<tr>
<td>Author:</td>
<td>Elena Aledo [aut, cre, cph],
  Juan-Carlos Aledo <a href="https://orcid.org/0000-0002-3497-9945"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Elena Aledo &lt;elenaaledoesteban@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-12-02 10:00:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='A_selected'>Adjacency Matrix for Orthology Graph</h2><span id='topic+A_selected'></span>

<h3>Description</h3>

<p>155 x 155 square matrix (155 GS proteins from 45 seed plant species)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>A_selected
</code></pre>


<h3>Format</h3>

<p>A matrix with 155 rows and 155 columns
</p>


<h3>Source</h3>

<p>It has been generated using the function orthG::mapTrees()
and the reconciliation output file 'selected'.
Verbigracia: orthG::mapTrees('./inst/extdata/selected')
The reconciliation was carried out using RANGER-DTL with parameters D = 1, T = 10 and L = 1.
</p>

<hr>
<h2 id='agf'>Angiosperms Gymnosperms Ferns</h2><span id='topic+agf'></span>

<h3>Description</h3>

<p>Angiosperms Gymnosperms Ferns
</p>


<h3>Usage</h3>

<pre><code class='language-R'>agf
</code></pre>


<h3>Format</h3>

<p>A dataframe with 275 rows (GS proteins) and 23 columns:
</p>

<dl>
<dt>n</dt><dd><p>Reference number</p>
</dd>
<dt>phylo_id</dt><dd><p>Unique identification label of the protein/gen</p>
</dd>
<dt>species</dt><dd><p>Species</p>
</dd>
<dt>taxon</dt><dd><p>Acrogymnospermae, Angiospermae, Polypodiopsida</p>
</dd>
<dt>dna</dt><dd><p>CDS sequence</p>
</dd>
<dt>prot</dt><dd><p>Protein sequence</p>
</dd>
<dt>short</dt><dd><p>Unique three letter identification of the species</p>
</dd>
<dt>gs</dt><dd><p>GS2, GS1a or GS1b_Ang, GS1b_Gym</p>
</dd>
<dt>pI</dt><dd><p>isoelectric point</p>
</dd>
<dt>factor</dt><dd><p>Ferns, GS2, GS1a, GS1b_Ang, GS1b_Gym</p>
</dd>
<dt>size</dt><dd><p>number of residues</p>
</dd>
<dt>CSpos</dt><dd><p>position signal</p>
</dd>
<dt>prediction</dt><dd><p>prediction</p>
</dd>
<dt>Lk_SP</dt><dd><p>seq pep</p>
</dd>
<dt>Lk_mTP</dt><dd><p>mit</p>
</dd>
<dt>Lk_cTP</dt><dd><p>chl</p>
</dd>
<dt>Lk_Thylak</dt><dd><p>thy</p>
</dd>
<dt>secAa</dt><dd><p>amino acid at position 2</p>
</dd>
<dt>core</dt><dd><p>core</p>
</dd>
<dt>dabase</dt><dd><p>db</p>
</dd>
<dt>acc</dt><dd><p>acc</p>
</dd>
<dt>up_id</dt><dd><p>uniprot</p>
</dd>
<dt>note</dt><dd><p>note</p>
</dd>
</dl>



<h3>Source</h3>

<p>It has been manually curated by the authors
</p>

<hr>
<h2 id='AngGym'>Angiosperms Gymnosperms</h2><span id='topic+AngGym'></span>

<h3>Description</h3>

<p>Angiosperms Gymnosperms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AngGym
</code></pre>


<h3>Format</h3>

<p>A dataframe with 155 rows (GS proteins) and 10 columns:
</p>

<dl>
<dt>n</dt><dd><p>Reference number</p>
</dd>
<dt>phylo_id</dt><dd><p>Unique identification label of the protein/gen</p>
</dd>
<dt>species</dt><dd><p>Species</p>
</dd>
<dt>taxon</dt><dd><p>Acrogymnospermae or Angiospermae</p>
</dd>
<dt>class</dt><dd><p>Angiosperms: Amborellopsida, Liliopsida, Magnoliopsida; Gymnosperms: Ginkgoopsida, Cycadopsida, Gnetopsida, Pinopsida</p>
</dd>
<dt>dna</dt><dd><p>CDS sequence</p>
</dd>
<dt>prot</dt><dd><p>Protein sequence</p>
</dd>
<dt>short</dt><dd><p>Unique three letter identification of the species</p>
</dd>
<dt>gsLineage</dt><dd><p>Either GS2, GS1a or GS1b</p>
</dd>
<dt>plant_group</dt><dd><p>Primitive angiosperms, Modern angiosperms, Ginkgo-Cycadales, Gnetales, Pinacea, Conifer II</p>
</dd>
</dl>



<h3>Source</h3>

<p>It has been manually curated by the authors
</p>

<hr>
<h2 id='coltips'>Colouring Tree Tips</h2><span id='topic+coltips'></span>

<h3>Description</h3>

<p>Make a color vector for colouring tree tips
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coltips(phy)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coltips_+3A_phy">phy</code></td>
<td>
<p>tree as a phylo object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each tip is given a color according to the nature of the isoform: green (GS2), blue (GS1a), brown (GS1b Gym), salmon (GS1b Ang), purple (other).
</p>


<h3>Value</h3>

<p>a color vector as long as the number of tips
</p>


<h3>Examples</h3>

<pre><code class='language-R'>coltips(ape::read.tree(text = "((Bdi, Sly), (Pp, Ap));"))
</code></pre>

<hr>
<h2 id='gapless_msa'>Remove Gaps in a MSA</h2><span id='topic+gapless_msa'></span>

<h3>Description</h3>

<p>Removes gaps in a given msa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gapless_msa(msa, seqtype = 'AA', df = TRUE, sfile = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gapless_msa_+3A_msa">msa</code></td>
<td>
<p>input alignment.</p>
</td></tr>
<tr><td><code id="gapless_msa_+3A_seqtype">seqtype</code></td>
<td>
<p>the nature of the sequences: 'DNA' or 'AA'.</p>
</td></tr>
<tr><td><code id="gapless_msa_+3A_df">df</code></td>
<td>
<p>logical. When TRUE msa should be a matrix, when FALSE msa should be a string giving the path to a fasta file containing the alignment.</p>
</td></tr>
<tr><td><code id="gapless_msa_+3A_sfile">sfile</code></td>
<td>
<p>if different to FALSE, then it should be a string indicating the path to save a fasta alignment file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It should be noted that this function does not carry out the alignment itself.
</p>


<h3>Value</h3>

<p>an alignment without gaps in form of matrix or a file containing such an alignment in fasta format.
</p>


<h3>See Also</h3>

<p>msa
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1:
aln &lt;- matrix(c("A",  "P",  "G",  "W",  "-",  "-",
                "-",  "A",  "G",  "W",  "C",  "-",
                "-",  "-",  "C",  "W",  "G",  "A" ), nrow = 3, byrow =TRUE)
gapless_msa(aln)
# Example 2:
## Not run: gapless_msa(msa(sequences = c("APGW", "AGWC", "CWGA"),
                     ids = c("a", "b", "c"))$ali)
## End(Not run)
</code></pre>

<hr>
<h2 id='getseqGS'>Get the GS Sequence</h2><span id='topic+getseqGS'></span>

<h3>Description</h3>

<p>Provides the requested GS sequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getseqGS(phylo_id, molecule = "Prot")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getseqGS_+3A_phylo_id">phylo_id</code></td>
<td>
<p>the unique sequence identifier</p>
</td></tr>
<tr><td><code id="getseqGS_+3A_molecule">molecule</code></td>
<td>
<p>either &quot;Prot&quot; or &quot;CDS&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The identifier should be one of the 'phylo_id' from data(agf).
</p>


<h3>Value</h3>

<p>The requested sequence as a character string.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>getseqGS("Pp_GS1b_2")
</code></pre>

<hr>
<h2 id='madRoot'>Find The Root of a Phylogenetic Tree Using MAD Method</h2><span id='topic+madRoot'></span>

<h3>Description</h3>

<p>Finds the root of an unrooted phylogenetic tree by minimizing the relative deviation from the molecular clock.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>madRoot(tree, output_mode = 'phylo')
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="madRoot_+3A_tree">tree</code></td>
<td>
<p>unrooted tree string in newick format or a tree object of class 'phylo'.</p>
</td></tr>
<tr><td><code id="madRoot_+3A_output_mode">output_mode</code></td>
<td>
<p>amount of information to return. If 'phylo' (default) only the rooted tree is returned. If 'stats' also a structure with the ambiguity index, clock cv, the minimum ancestor deviation and the number of roots. If 'full' also an unrooted tree object, the index of the root branch, the branch ancestor deviations and a rooted tree object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a slight modification of the code provided by Tria et al at https://www.mikrobio.uni-kiel.de/de/ag-dagan/ressourcen.
</p>


<h3>Value</h3>

<p>a rooted tree and supplementary information if required.
</p>


<h3>Author(s)</h3>

<p>Tria, F. D. K., Landan, G. and Dagan, T.
</p>


<h3>References</h3>

<p>Tria, F. D. K., Landan, G. and Dagan, T. Nat. Ecol. Evol. 1, 0193 (2017).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1:
madRoot("(c:1.182246599,b:0.4169984702,a:0.1582465793);")
# Example 2:
## Not run: 
a &lt;- msa(sequences=c("RAPGT", "KMPGT", "ESGGT"), ids = letters[1:3])$ali
tr &lt;- mltree(a)$tree
rtr &lt;- madRoot(tr) 
## End(Not run)
</code></pre>

<hr>
<h2 id='mapTrees'>Map Gene Tree into Species Tree</h2><span id='topic+mapTrees'></span>

<h3>Description</h3>

<p>Maps a gene/protein tree into a species tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapTrees(path2rec)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mapTrees_+3A_path2rec">path2rec</code></td>
<td>
<p>path to the file containing the reconciliation output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Mapping gene tree into species tree allow to infer the sequence of events (Duplication, Speciation, Transfer).
</p>


<h3>Value</h3>

<p>A list with three elements. The first one is a 'phylo' object where the nodelabels indicate the event: D, duplication or T transfer. If no label is shown is because the event correspond to speciation. The second element is a dataframe (the first column is the label of the internal nodes in the gene tree; the second column is the label of the internal nodes in the species tree, and the third and fourth columns label each internal node according to the inferred event). The third element of the list is an adjacency matrix: 1 when two proteins are orthologous, 0 if they are paralogous.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file_path &lt;- system.file("extdata", "representatives", package = "orthGS")
mapTrees(file_path)
</code></pre>

<hr>
<h2 id='mltree'>Build Up a ML Tree</h2><span id='topic+mltree'></span>

<h3>Description</h3>

<p>Given an alignment builds an ML tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mltree(msa, df = TRUE, gapl = TRUE, model = "WAG")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mltree_+3A_msa">msa</code></td>
<td>
<p>input alignment.</p>
</td></tr>
<tr><td><code id="mltree_+3A_df">df</code></td>
<td>
<p>logical. When TRUE msa should be a dataframe, when FALSE msa should be a string giving the path to a fasta file containing the alignment.</p>
</td></tr>
<tr><td><code id="mltree_+3A_gapl">gapl</code></td>
<td>
<p>logical, when TRUE a gapless alignment is used.</p>
</td></tr>
<tr><td><code id="mltree_+3A_model">model</code></td>
<td>
<p>allows to choose an amino acid models (see the function phangorn::as.pml)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function makes a NJ tree and then improvove it using an optimization procedure based on ML.
</p>


<h3>Value</h3>

<p>a ML optimized tree (and parameters)
</p>


<h3>See Also</h3>

<p>gapless_msa
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example 1:
mltree(matrix(c("R","K","E","A","M","S","P","P","G"), nrow=3,
       dimnames = list(letters[1:3], 1:3)))$tree
# Example 2:
## Not run: 
a &lt;- msa(sequences=c("RAPGT", "KMPGT", "ESGGT"), ids = letters[1:3])$ali
mltree(a)$tree

## End(Not run)
</code></pre>

<hr>
<h2 id='msa'>Multiple Sequence Alignment</h2><span id='topic+msa'></span>

<h3>Description</h3>

<p>Aligns multiple protein, DNA or CDS sequences using inhouse software.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>msa(sequences, ids = names(sequences), seqtype = "prot", method, sfile = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="msa_+3A_sequences">sequences</code></td>
<td>
<p>vector containing the sequences as strings.</p>
</td></tr>
<tr><td><code id="msa_+3A_ids">ids</code></td>
<td>
<p>character vector containing the sequences' ids.</p>
</td></tr>
<tr><td><code id="msa_+3A_seqtype">seqtype</code></td>
<td>
<p>it should be either &quot;prot&quot; of &quot;dna&quot; or &quot;cds&quot; (see details).</p>
</td></tr>
<tr><td><code id="msa_+3A_method">method</code></td>
<td>
<p>the software to be used for the alignment, as invoked in your system. For instance, &quot;muscle3&quot; or &quot;clustalo&quot;.</p>
</td></tr>
<tr><td><code id="msa_+3A_sfile">sfile</code></td>
<td>
<p>if different to FALSE, then it should be a string indicating the path to save a fasta alignment file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Either Clustal Omega or MUSCLE must be installed, and their executable be in your system's PATH. If seqtype is set to &quot;cds&quot; the sequences must not contain stop codons and they will be translated using the standard code. Afterward, the amino acid alignment will be used to lead the codon alignment.
</p>


<h3>Value</h3>

<p>Returns a list of four elements. The first one ($seq) provides the sequences analyzed, the second element ($id) returns the identifiers, the third element ($aln) provides the alignment in fasta format and the fourth element ($ali) gives the alignment in matrix format.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: msa(sequences = c("APGW", "AGWC", "CWGA"),
                             ids = c("a", "b", "c"))
## End(Not run)
</code></pre>

<hr>
<h2 id='orthG'>Infer GS OrthoGroups Within a Set of Species</h2><span id='topic+orthG'></span>

<h3>Description</h3>

<p>Infers GS orthogroups using tree reconciliation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orthG(set = "all")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="orthG_+3A_set">set</code></td>
<td>
<p>set of species of interest provided as a character vector either with the binomial or short code of the species (see data(sdf)).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When set = &quot;all&quot;, all the species in the database will be included.
</p>


<h3>Value</h3>

<p>A list with two elements. The first one is the adjacency matrix (1 for orthologous, 0 for paralogous). The second element is an orthogroup graph.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>orthG(set = c("Pp", "Psy", "Psm", "Ap"))
</code></pre>

<hr>
<h2 id='orthP'>Search Orthologous of a Given Protein</h2><span id='topic+orthP'></span>

<h3>Description</h3>

<p>Searchs orthologous of a given protein within a set of selected species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orthP(phylo_id, set = "all")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="orthP_+3A_phylo_id">phylo_id</code></td>
<td>
<p>phylo_id of the query protein</p>
</td></tr>
<tr><td><code id="orthP_+3A_set">set</code></td>
<td>
<p>set of species of interest provided as a character vector, either with the binomial or short code of the species (see details).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When set = &quot;all&quot;, the search will be carry out against all the species in the database.
</p>


<h3>Value</h3>

<p>A list with thee elements: 1. subtree of the relevant proteins; 2. vector color; 3. phylo_ids of the orthologous found.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>orthP(phylo_id = "Pp_GS1a", set = c("Pp", "Psy", "Psm", "Ap"))
</code></pre>

<hr>
<h2 id='sdf'>Seed Plants and Ferns GS</h2><span id='topic+sdf'></span>

<h3>Description</h3>

<p>155 GS proteins from 25 seed plants species and 41 GS proteins from 11 fern species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sdf
</code></pre>


<h3>Format</h3>

<p>A dataframe with 196 rows (GS proteins) and 7 columns:
</p>

<dl>
<dt>n</dt><dd><p>Reference number</p>
</dd>
<dt>Sec.Name_</dt><dd><p>Unique identification label of the protein</p>
</dd>
<dt>species</dt><dd><p>Species</p>
</dd>
<dt>taxon</dt><dd><p>Acrogymnospermae, Angiospermae  or Polypodiopsida</p>
</dd>
<dt>short</dt><dd><p>Unique three letter identification of the species</p>
</dd>
<dt>gs</dt><dd><p>Either GS2, GS1a, GS1b_Gym or GS1b_Ang. Here the ferns proteins have been forced to be either GS1a or GS2</p>
</dd>
<dt>tax_group</dt><dd><p>Taxonomic group</p>
</dd>
</dl>



<h3>Source</h3>

<p>It has been handly curated by the authors
</p>

<hr>
<h2 id='selected_tr'>Ultrametric Rooted Seed Plants Tree</h2><span id='topic+selected_tr'></span>

<h3>Description</h3>

<p>155 GS proteins from 45 seed plants species
Rooted using MAD (Minimal Ancestor Deviation)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selected_tr
</code></pre>


<h3>Format</h3>

<p>An phylo object
</p>


<h3>Source</h3>

<p>It has been manually curated by the authors
</p>

<hr>
<h2 id='speciesGS'>Map Species Names</h2><span id='topic+speciesGS'></span>

<h3>Description</h3>

<p>Map binomial species name to short code species name and vice versa
</p>


<h3>Usage</h3>

<pre><code class='language-R'>speciesGS(sp)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="speciesGS_+3A_sp">sp</code></td>
<td>
<p>set of species of interest (either binomial or short code name)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The species set should be given as a character vector (see example)
</p>


<h3>Value</h3>

<p>A datafrane containing the information for the requested species.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>speciesGS(c("Pinus pinaster", "Ath"))
</code></pre>

<hr>
<h2 id='subsetGS'>GS Proteins Report</h2><span id='topic+subsetGS'></span>

<h3>Description</h3>

<p>Assembles a report regarding the GS proteins found in the indicated subset of species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetGS(sp)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subsetGS_+3A_sp">sp</code></td>
<td>
<p>set of species of interest (either binomial or short code name)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns the protein and DNA sequences of the different isoforms found in each species, along with other relevant data.
</p>


<h3>Value</h3>

<p>A dataframe with the information for the requested species.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>subsetGS(c("Pinus pinaster", "Ath"))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
