<!DOCTYPE html><html><head><title>Help for package otargen</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {otargen}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#colocalisationsForGene'><p>Retrieves Colocalisation Statistics for a Given Gene</p></a></li>
<li><a href='#geneInfo'><p>Retrieve gene summary information.</p></a></li>
<li><a href='#genesForVariant'><p>Retrieves variant-surrounding genes with calculated statistics for causal gene prioritization.</p></a></li>
<li><a href='#getLociGenes'><p>Retrieve summary information of the genes in a locus</p></a></li>
<li><a href='#gwasColocalisation'><p>Retrieve calculated GWAS colocalisation data</p></a></li>
<li><a href='#gwasColocalisationForRegion'><p>Retrieve GWAS colocalisation data for a genomic region .</p></a></li>
<li><a href='#gwasCredibleSet'><p>Retrieve GWAS credible set data</p></a></li>
<li><a href='#gwasRegional'><p>Retrieve GWAS summary statistics for a genomic region.</p></a></li>
<li><a href='#indexVariantsAndStudiesForTagVariant'><p>Retrieve population-level summary GWAS statistics.</p></a></li>
<li><a href='#manhattan'><p>Retrieve GWAS summary statistics to create a Manhattan plot</p></a></li>
<li><a href='#overlapInfoForStudy'><p>Retrieves overlap info for a study and a list of studies</p></a></li>
<li><a href='#pheWAS'><p>Retrieve PheWAS (Phenome Wide Association Studies) data for a variant.</p></a></li>
<li><a href='#plot_coloc'><p>Scatter plot of colocalisations for gene and reported traits.</p></a></li>
<li><a href='#plot_l2g'><p>Radar plot for L2G partial scores from <code>studiesAndLeadVariantsForGeneByL2G()</code></p></a></li>
<li><a href='#plot_manhattan'><p>Plot results from <code>manhattan()</code></p></a></li>
<li><a href='#plot_phewas'><p>Plot <code>PheWAS()</code> results.</p></a></li>
<li><a href='#qtlColocalisationVariantQuery'><p>Retrieve  QTL colocalisation results of a variant</p></a></li>
<li><a href='#qtlCredibleSet'><p>Retrieve calculated QTL summary statistics for credible variant set.</p></a></li>
<li><a href='#run_custom_query'><p>Running custom GraphQL queries</p></a></li>
<li><a href='#studiesAndLeadVariantsForGeneByL2G'><p>Retrieve &quot;locus-to-gene&quot; (L2G) model summary data for a gene.</p></a></li>
<li><a href='#studyInfo'><p>Retrieve study summary information.</p></a></li>
<li><a href='#studyLocus2GeneTable'><p>Retrieve the locus-to-gene (L2G) data table for loci genes.</p></a></li>
<li><a href='#studyVariants'><p>Retrieves all variants for a study.</p></a></li>
<li><a href='#tagVariantsAndStudiesForIndexVariant'><p>Retrieves tag variants and studies for a given index variant.</p></a></li>
<li><a href='#topOverlappedStudies'><p>Retrieve top studies having overlap in identified loci.</p></a></li>
<li><a href='#variantInfo'><p>Retrieves information about a variant.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Access Open Target Genetics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-09</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Amir Feizi &lt;afeizi@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Interact seamlessly with Open Target Genetics' GraphQL endpoint to query and retrieve tidy data tables, facilitating the analysis of genetic data. For more information about the Open Target Genetics API(<a href="https://genetics.opentargets.org/api">https://genetics.opentargets.org/api</a>).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ghql, janitor, ggiraphExtra, cli, dplyr, ggplot2, ggrepel,
jsonlite, rlang, magrittr, stringr, tibble, tidyr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0),</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://amirfeizi.github.io/otargen/">https://amirfeizi.github.io/otargen/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-21 08:08:05 UTC; amir.feizi</td>
</tr>
<tr>
<td>Author:</td>
<td>Amir Feizi [aut, cre],
  Kamalika Ray [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-21 08:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='colocalisationsForGene'>Retrieves Colocalisation Statistics for a Given Gene</h2><span id='topic+colocalisationsForGene'></span>

<h3>Description</h3>

<p>Retrieves Colocalisation statistics for a gene using ENSEMBL gene IDs or gene symbol.
Colocalisation analysis is performed between all studies in the Portal with at least one overlapping
associated locus. This analysis tests whether two independent associations at the same locus are
consistent with having a shared causal variant. The function supports multiple gene IDs as a list.
The returned data frame (tibble format) includes studies that have evidence of colocalisation
with molecular QTLs for the selected gene(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colocalisationsForGene(genes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colocalisationsForGene_+3A_genes">genes</code></td>
<td>
<p>Character: Gene ENSEMBL ID (e.g. ENSG00000169174) or gene symbol (e.g. PCSK9). Multiple gene IDs are supported as a character vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame (tibble format) including the colocalisation data for the query gene(s).
</p>
<p>The output tibble contains the following columns:
</p>

<ul>
<li><p><code>Study</code>: Character vector. Study identifier.
</p>
</li>
<li><p><code>Trait_reported</code>: Character vector. Reported trait associated with the colocalisation.
</p>
</li>
<li><p><code>Lead_variant</code>: Character vector. Lead variant associated with the colocalisation.
</p>
</li>
<li><p><code>Molecular_trait</code>: Character vector. Molecular trait associated with the colocalisation.
</p>
</li>
<li><p><code>Gene_symbol</code>: Character vector. Gene symbol associated with the colocalisation.
</p>
</li>
<li><p><code>Tissue</code>: Character vector. Tissue where the colocalisation occurs.
</p>
</li>
<li><p><code>Source</code>: Character vector. Source of the colocalisation data.
</p>
</li>
<li><p><code>H3</code>: Numeric vector. H3 value associated with the colocalisation.
</p>
</li>
<li><p><code>log2_H4_H3</code>: Numeric vector. Log2 ratio of H4 to H3 values.
</p>
</li>
<li><p><code>Title</code>: Character vector. Title of the study.
</p>
</li>
<li><p><code>Author</code>: Character vector. Author(s) of the study.
</p>
</li>
<li><p><code>Has_sumstats</code>: Logical vector. Indicates if the study has summary statistics.
</p>
</li>
<li><p><code>numAssocLoci</code>: Numeric vector. Number of associated loci in the study.
</p>
</li>
<li><p><code>nInitial_cohort</code>: Numeric vector. Number of samples in the initial cohort.
</p>
</li>
<li><p><code>study_nReplication</code>: Numeric vector. Number of samples in the replication cohort.
</p>
</li>
<li><p><code>study_nCases</code>: Numeric vector. Number of cases in the study.
</p>
</li>
<li><p><code>Publication_date</code>: Character vector. Publication date of the study.
</p>
</li>
<li><p><code>Journal</code>: Character vector. Journal where the study was published.
</p>
</li>
<li><p><code>Pubmed_id</code>: Character vector. PubMed identifier of the study.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result1 &lt;- colocalisationsForGene(gene = c("ENSG00000163946", "ENSG00000169174", "ENSG00000143001"))
result2 &lt;- colocalisationsForGene(gene = "ENSG00000169174")
result3 &lt;- colocalisationsForGene(gene = c("TP53", "TASOR"))
result4 &lt;- colocalisationsForGene(gene = "TP53") 
## End(Not run)

</code></pre>

<hr>
<h2 id='geneInfo'>Retrieve gene summary information.</h2><span id='topic+geneInfo'></span>

<h3>Description</h3>

<p>This function takes a gene ENSEMBL id (e.g.ENSG00000169174) or a gene name (e.g.PCSK9) and returns a table containing details about
input genes , such as genomic location, gene structure and its BioType.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneInfo(gene)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geneInfo_+3A_gene">gene</code></td>
<td>
<p>Character: an ENSEMBL gene identifier (e.g.ENSG00000169174) or gene name (e.g. PCSK9).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame (tibble format) with the following data structure:
</p>

<ol>
<li><p> id
</p>
</li>
<li><p> symbol
</p>
</li>
<li><p> bioType
</p>
</li>
<li><p> description
</p>
</li>
<li><p> chromosome
</p>
</li>
<li><p> tss
</p>
</li>
<li><p> start
</p>
</li>
<li><p> end
</p>
</li>
<li><p> fwdStrand
</p>
</li>
<li><p> exons
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- geneInfo(gene="ENSG00000169174")
result &lt;- geneInfo(gene="PCSK9")

## End(Not run)
</code></pre>

<hr>
<h2 id='genesForVariant'>Retrieves variant-surrounding genes with calculated statistics for causal gene prioritization.</h2><span id='topic+genesForVariant'></span>

<h3>Description</h3>

<p>This function retrieves all calculated prioritizing scores for surrounding genes of a specific variant based on the Open Target Genetics locus-to-gene (L2G) ML scoring pipeline.
It provides detailed insights into the relationship between genetic variants and genes, allowing users to explore the impact of variants on gene expression, colocalization scores, chromatin interactions, and predicted functional effects.
The function returns the information in a structured format, making it easier to analyze and interpret the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genesForVariant(variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genesForVariant_+3A_variant_id">variant_id</code></td>
<td>
<p>A character string specifying the ID of the variant for which to fetch gene information.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following components:
</p>

<ul>
<li> <p><code>v2g</code>: A data frame with all variant-to-gene information with the following data structure:
- <code>gene.symbol</code>: character
- <code>variant</code>: character
- <code>overallScore</code>: numeric
- <code>gene.id</code>: character
</p>
</li>
<li> <p><code>tssd</code>: A data frame with all details on colocalization scores effect of variant in expression of the genes across tissues with the following data structure:
- <code>gene.symbol</code>: character
- <code>variant</code>: character
- <code>typeId</code>: character
- <code>sourceId</code>: character
- <code>aggregatedScore</code>: numeric
- <code>tissues_distance</code>: integer
- <code>tissues_score</code>: numeric
- <code>tissues_quantile</code>: numeric
- <code>tissues_id</code>: character
- <code>tissues_name</code>: character
</p>
</li>
<li> <p><code>qtls</code>: List of QTL associations between genes and variants across analyzed tissues with the following data structure:
- <code>gene.symbol</code>: character
- <code>variant</code>: character
- <code>typeId</code>: character
- <code>aggregatedScore</code>: numeric
- <code>tissues_quantile</code>: numeric
- <code>tissues_beta</code>: numeric
- <code>tissues_pval</code>: numeric
- <code>tissues_id</code>: character
- <code>tissues_name</code>: character
</p>
</li>
<li> <p><code>chromatin</code>: A data frame including all information on chromatin interactions effect involving genes and variants with the following data structure:
- <code>gene.symbol</code>: character
- <code>variant</code>: character
- <code>typeId</code>: character
- <code>sourceId</code>: character
- <code>aggregatedScore</code>: numeric
- <code>tissues_quantile</code>: numeric
- <code>tissues_score</code>: numeric
- <code>tissues_id</code>: character
- <code>tissues_name</code>: character
</p>
</li>
<li> <p><code>functionalpred</code>: A data frame including predicted functional effects of variants on genes across tissues with the following data structure:
- <code>gene.symbol</code>: character
- <code>variant</code>: character
- <code>typeId</code>: character
- <code>sourceId</code>: character
- <code>aggregatedScore</code>: numeric
- <code>tissues_maxEffectLabel</code>: character
- <code>tissues_maxEffectScore</code>: numeric
- <code>tissues_id</code>: character
- <code>tissues_name</code>: character
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  result &lt;- genesForVariant(variant_id = "1_154453788_C_T")
  result &lt;- genesForVariant(variant_id = "rs4129267")
  print(result)

## End(Not run)
</code></pre>

<hr>
<h2 id='getLociGenes'>Retrieve summary information of the genes in a locus</h2><span id='topic+getLociGenes'></span>

<h3>Description</h3>

<p>This function retrieves information for a specified region on a chromosome, including overlapping genes and their
associated details such as ID, symbol, biotype, description,
transcription start site (TSS), start position, end position, strand direction, and exons structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getLociGenes(chromosome, start, end)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getLociGenes_+3A_chromosome">chromosome</code></td>
<td>
<p>Character: Chromosome number as a string.</p>
</td></tr>
<tr><td><code id="getLociGenes_+3A_start">start</code></td>
<td>
<p>Integer: Start position of the specified region on the chromosome.</p>
</td></tr>
<tr><td><code id="getLociGenes_+3A_end">end</code></td>
<td>
<p>Integer: End position of the specified region on the chromosome.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a tibble data frame of all the overlapping genes in the specified region with the following data structure:
</p>

<ul>
<li><p><code>id</code>: <em>Character</em>. ID of the gene.
</p>
</li>
<li><p><code>symbol</code>: <em>Character</em>. Symbol of the gene.
</p>
</li>
<li><p><code>bioType</code>: <em>Character</em>. Biotype of the gene.
</p>
</li>
<li><p><code>description</code>: <em>Character</em>. Description of the gene.
</p>
</li>
<li><p><code>chromosome</code>: <em>Character</em>. Chromosome of the gene.
</p>
</li>
<li><p><code>tss</code>: <em>Integer</em>. Transcription start site of the gene.
</p>
</li>
<li><p><code>start</code>: <em>Integer</em>. Start position of the gene.
</p>
</li>
<li><p><code>end</code>: <em>Integer</em>. End position of the gene.
</p>
</li>
<li><p><code>fwdStrand</code>: <em>Logical</em>. Strand direction of the gene.
</p>
</li>
<li><p><code>exons</code>: <em>List</em>. List of exons of the gene.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- get_genes(chromosome = "2", start = 239634984, end = 241634984)

## End(Not run)
</code></pre>

<hr>
<h2 id='gwasColocalisation'>Retrieve calculated GWAS colocalisation data</h2><span id='topic+gwasColocalisation'></span>

<h3>Description</h3>

<p>This function retrieves colocalisation data for a specific variant from a study with other GWAS studies.
It returns a data frame of the studies that colocalise with the input variant and study,
including details on the study and reported trait, index variant, and calculated coloc method (see Ref. below) outputs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gwasColocalisation(study_id, variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gwasColocalisation_+3A_study_id">study_id</code></td>
<td>
<p>Character: Open Target Genetics generated ID for the GWAS study.</p>
</td></tr>
<tr><td><code id="gwasColocalisation_+3A_variant_id">variant_id</code></td>
<td>
<p>Character: Open Target Genetics generated ID for the variant (CHRPOSITION_REFALLELE_ALTALLELE or rsID).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame of the studies that colocalise with the input variant and study. The table consists of the following data structure:
</p>

<ul>
<li><p><code>study.studyId</code>: <em>Character vector</em>. Study identifier.
</p>
</li>
<li><p><code>study.traitReported</code>: <em>Character vector</em>. Reported trait associated with the colocalisation.
</p>
</li>
<li><p><code>study.traitCategory</code>: <em>Character vector</em>. Trait category.
</p>
</li>
<li><p><code>indexVariant.id</code>: <em>Character vector</em>. Index variant identifier.
</p>
</li>
<li><p><code>indexVariant.position</code>: <em>Integer vector</em>. Index variant position.
</p>
</li>
<li><p><code>indexVariant.chromosome</code>: <em>Character vector</em>. Index variant chromosome.
</p>
</li>
<li><p><code>indexVariant.rsId</code>: <em>Character vector</em>. Index variant rsID.
</p>
</li>
<li><p><code>beta</code>: <em>Numeric vector</em>. Beta value associated with the colocalisation.
</p>
</li>
<li><p><code>h3</code>: <em>Numeric vector</em>. H3 value associated with the colocalisation.
</p>
</li>
<li><p><code>h4</code>: <em>Numeric vector</em>. H4 value associated with the colocalisation.
</p>
</li>
<li><p><code>log2h4h3</code>: <em>Numeric vector</em>. Log2 ratio of H4 to H3 values.
</p>
</li></ul>



<h3>References</h3>

<p>Giambartolomei, Claudia et al. “Bayesian test for colocalisation between pairs of genetic association studies using summary statistics.” PLoS genetics vol. 10,5 e1004383. 15 May. 2014, doi:10.1371/journal.pgen.1004383
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
colocalisation_data &lt;- gwasColocalisation(study_id = "GCST90002357", variant_id = "1_154119580_C_A")
colocalisation_data &lt;- gwasColocalisation(study_id = "GCST90002357", variant_id = "rs2494663")

## End(Not run)
</code></pre>

<hr>
<h2 id='gwasColocalisationForRegion'>Retrieve GWAS colocalisation data for a genomic region .</h2><span id='topic+gwasColocalisationForRegion'></span>

<h3>Description</h3>

<p>By providing a genomic region (chromosome name with start and end position), this function returns information about colocalisation
between GWAS studies and associated loci within a specified genomic region. It provides details on the
studies that have at least one overlapping associated locus within the region, allowing for the assessment
of potential shared causal variants.
The query output includes data such as the study identifiers, traits,
loci information, and other relevant attributes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gwasColocalisationForRegion(chromosome, start, end)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gwasColocalisationForRegion_+3A_chromosome">chromosome</code></td>
<td>
<p>String: Chromosome number as a string.</p>
</td></tr>
<tr><td><code id="gwasColocalisationForRegion_+3A_start">start</code></td>
<td>
<p>Long: Start position of the specified chromosome.</p>
</td></tr>
<tr><td><code id="gwasColocalisationForRegion_+3A_end">end</code></td>
<td>
<p>Long: End position of the specified chromosome.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame with the following columns:
</p>

<ul>
<li><p><code>leftVariant.id</code>: <em>Character vector</em>. ID of the left variant.
</p>
</li>
<li><p><code>leftVariant.position</code>: <em>Integer vector</em>. Position of the left variant.
</p>
</li>
<li><p><code>leftVariant.chromosome</code>: <em>Character vector</em>. Chromosome of the left variant.
</p>
</li>
<li><p><code>leftVariant.rsId</code>: <em>Character vector</em>. rsID of the left variant.
</p>
</li>
<li><p><code>leftStudy.studyId</code>: <em>Character vector</em>. Study identifier for the left study.
</p>
</li>
<li><p><code>leftStudy.traitReported</code>: <em>Character vector</em>. Reported trait associated with the colocalisation in the left study.
</p>
</li>
<li><p><code>leftStudy.traitCategory</code>: <em>Character vector</em>. Category of the reported trait in the left study.
</p>
</li>
<li><p><code>rightVariant.id</code>: <em>Character vector</em>. ID of the right variant.
</p>
</li>
<li><p><code>rightVariant.position</code>: <em>Integer vector</em>. Position of the right variant.
</p>
</li>
<li><p><code>rightVariant.chromosome</code>: <em>Character vector</em>. Chromosome of the right variant.
</p>
</li>
<li><p><code>rightVariant.rsId</code>: <em>Character vector</em>. rsID of the right variant.
</p>
</li>
<li><p><code>rightStudy.studyId</code>: <em>Character vector</em>. Study identifier for the right study.
</p>
</li>
<li><p><code>rightStudy.traitReported</code>: <em>Character vector</em>. Reported trait associated with the colocalisation in the right study.
</p>
</li>
<li><p><code>rightStudy.traitCategory</code>: <em>Character vector</em>. Category of the reported trait in the right study.
</p>
</li>
<li><p><code>h3</code>: <em>Numeric vector</em>. H3 value associated with the colocalisation.
</p>
</li>
<li><p><code>h4</code>: <em>Numeric vector</em>. H4 value associated with the colocalisation.
</p>
</li>
<li><p><code>log2h4h3</code>: <em>Numeric vector</em>. Log2 ratio of H4 to H3 values associated with the colocalisation.
</p>
</li></ul>



<h3>References</h3>

<p>- Giambartolomei, Claudia et al. “Bayesian test for colocalisation between pairs of
genetic association studies using summary statistics.” PLoS genetics vol. 10,5 e1004383. 15 May. 2014, doi:10.1371/journal.pgen.1004383
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- gwasColocalisationForRegion(chromosome = "1", start = 153992685, end = 154155116)

## End(Not run)
</code></pre>

<hr>
<h2 id='gwasCredibleSet'>Retrieve GWAS credible set data</h2><span id='topic+gwasCredibleSet'></span>

<h3>Description</h3>

<p>Provided with a study ID and a lead variant ID, this function returns a
data frame consisting of all the associated credible set tag variants
with the corresponding statistical data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gwasCredibleSet(study_id, variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gwasCredibleSet_+3A_study_id">study_id</code></td>
<td>
<p>Character: Study ID(s) generated by Open Targets Genetics (e.g GCST90002357).</p>
</td></tr>
<tr><td><code id="gwasCredibleSet_+3A_variant_id">variant_id</code></td>
<td>
<p>Character: generated ID for variants by Open Targets Genetics (e.g. 1_154119580_C_A) or rsId (rs2494663).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame of results from the credible set of variants for a specific lead variant with the following columns:
</p>

<ul>
<li><p><code>tagVariant.id</code>: <em>Data frame</em>. A table of IDs of the tag variant.
</p>
</li>
<li><p><code>tagVariant.rsId</code>: <em>Character vector</em>. rsID of the tag variant.
</p>
</li>
<li><p><code>beta</code>: <em>Numeric</em>. Beta value.
</p>
</li>
<li><p><code>postProb</code>: <em>Numeric</em>. Posterior probability.
</p>
</li>
<li><p><code>pval</code>: <em>Numeric</em>. P-value.
</p>
</li>
<li><p><code>se</code>: <em>Numeric</em>. Standard error.
</p>
</li>
<li><p><code>MultisignalMethod</code>: <em>Character vector</em>. Multisignal method.
</p>
</li>
<li><p><code>logABF</code>: <em>Numeric</em>. Logarithm of approximate Bayes factor.
</p>
</li>
<li><p><code>is95</code>: <em>Logical</em>. Indicates if the variant has a 95
</p>
</li>
<li><p><code>is99</code>: <em>Logical</em>. Indicates if the variant has a 99
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- gwasCredibleSet(study_id="GCST90002357", variant_id="1_154119580_C_A")
result &lt;- gwasCredibleSet(study_id="GCST90002357", variant_id="rs2494663")

## End(Not run)
</code></pre>

<hr>
<h2 id='gwasRegional'>Retrieve GWAS summary statistics for a genomic region.</h2><span id='topic+gwasRegional'></span>

<h3>Description</h3>

<p>For a given study ID and chromosomal region information, this function returns  data frame(tibble format) with all variants and their GWAS summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gwasRegional(study_id, chromosome, start, end)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gwasRegional_+3A_study_id">study_id</code></td>
<td>
<p>Character: Open Target Genetics generated ID for the GWAS study.</p>
</td></tr>
<tr><td><code id="gwasRegional_+3A_chromosome">chromosome</code></td>
<td>
<p>Character: Chromosome number as a string.</p>
</td></tr>
<tr><td><code id="gwasRegional_+3A_start">start</code></td>
<td>
<p>Integer: Start position of the specified chromosome.</p>
</td></tr>
<tr><td><code id="gwasRegional_+3A_end">end</code></td>
<td>
<p>Integer: End position of the specified chromosome.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data table of variant information and p-values with the following columns:
</p>

<ul>
<li><p><code>variant.id</code>: <em>Character vector</em>. Variant identifier.
</p>
</li>
<li><p><code>variant.chromosome</code>: <em>Character vector</em>. Chromosome of the variant.
</p>
</li>
<li><p><code>variant.position</code>: <em>Integer vector</em>. Position of the variant.
</p>
</li>
<li><p><code>pval</code>: <em>Numeric vector</em>. P-value.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- gwasRegional(study_id = "GCST90002357",
chromosome = "1", start = 153992685, end = 154155116)

## End(Not run)
</code></pre>

<hr>
<h2 id='indexVariantsAndStudiesForTagVariant'>Retrieve population-level summary GWAS statistics.</h2><span id='topic+indexVariantsAndStudiesForTagVariant'></span>

<h3>Description</h3>

<p>For an input tag variant ID, this function returns a data frame(tibble format) with population-level summary statistics data across various GWAS studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>indexVariantsAndStudiesForTagVariant(variant_id, pageindex = 0, pagesize = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="indexVariantsAndStudiesForTagVariant_+3A_variant_id">variant_id</code></td>
<td>
<p>Character: generated ID for variants by Open Targets Genetics (e.g. 1_154119580_C_A) or rsId (rs2494663).</p>
</td></tr>
<tr><td><code id="indexVariantsAndStudiesForTagVariant_+3A_pageindex">pageindex</code></td>
<td>
<p>Integer: Index of the current page, pagination index &gt;= 0.</p>
</td></tr>
<tr><td><code id="indexVariantsAndStudiesForTagVariant_+3A_pagesize">pagesize</code></td>
<td>
<p>Integer: Number of records in a page, pagination size &gt; 0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame containing the variant associated with the input tag variant. The table consists of the following columns:
</p>

<ul>
<li><p><code>index_variant</code>: <em>Data frame</em>. Data frame of index variants with the following columns:
</p>

<ul>
<li><p><code>id</code>: <em>Character vector</em>. Variant ID.
</p>
</li>
<li><p><code>rsId</code>: <em>Character vector</em>. rsID of the variant.
</p>
</li></ul>

</li>
<li><p><code>study</code>: <em>Data frame</em>. Data frame of studies with the following columns:
</p>

<ul>
<li><p><code>studyId</code>: <em>Character vector</em>. Study identifier.
</p>
</li>
<li><p><code>traitReported</code>: <em>Character vector</em>. Reported trait associated with the colocalisation.
</p>
</li>
<li><p><code>traitCategory</code>: <em>Character vector</em>. Trait category.
</p>
</li></ul>

</li>
<li><p><code>pval</code>: <em>Numeric vector</em>. P-value.
</p>
</li>
<li><p><code>pval_mantissa</code>: <em>Numeric vector</em>. Mantissa of the p-value.
</p>
</li>
<li><p><code>pval_exponent</code>: <em>Integer vector</em>. Exponent of the p-value.
</p>
</li>
<li><p><code>n_total</code>: <em>Integer vector</em>. Total number of samples.
</p>
</li>
<li><p><code>n_cases</code>: <em>Integer vector</em>. Number of cases.
</p>
</li>
<li><p><code>overall_r2</code>: <em>Numeric vector</em>. Overall R-squared value.
</p>
</li>
<li><p><code>afr1000g_prop</code>: <em>Numeric vector</em>. Proportion in African population in 1000 Genomes.
</p>
</li>
<li><p><code>amr1000g_prop</code>: <em>Numeric vector</em>. Proportion in Admixed American population in 1000 Genomes.
</p>
</li>
<li><p><code>eas1000g_prop</code>: <em>Numeric vector</em>. Proportion in East Asian population in 1000 Genomes.
</p>
</li>
<li><p><code>eur1000g_prop</code>: <em>Numeric vector</em>. Proportion in European population in 1000 Genomes.
</p>
</li>
<li><p><code>sas1000g_prop</code>: <em>Numeric vector</em>. Proportion in South Asian population in 1000 Genomes.
</p>
</li>
<li><p><code>log10abf</code>: <em>Numeric vector</em>. Log10 ABF (Approximate Bayes Factor).
</p>
</li>
<li><p><code>posterior_probability</code>: <em>Numeric vector</em>. Posterior probability.
</p>
</li>
<li><p><code>odds_ratio</code>: <em>Numeric vector</em>. Odds ratio.
</p>
</li>
<li><p><code>odds_ratio_ci_lower</code>: <em>Numeric vector</em>. Lower confidence interval of the odds ratio.
</p>
</li>
<li><p><code>odds_ratio_ci_upper</code>: <em>Numeric vector</em>. Upper confidence interval of the odds ratio.
</p>
</li>
<li><p><code>beta</code>: <em>Numeric vector</em>. Beta value.
</p>
</li>
<li><p><code>beta_ci_lower</code>: <em>Numeric vector</em>. Lower confidence interval of the beta value.
</p>
</li>
<li><p><code>beta_ci_upper</code>: <em>Numeric vector</em>. Upper confidence interval of the beta value.
</p>
</li>
<li><p><code>direction</code>: <em>Character vector</em>. Direction of the effect.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- indexVariantsAndStudiesForTagVariant(variant_id = "1_109274968_G_T")
result &lt;- indexVariantsAndStudiesForTagVariant(variant_id = "rs12740374",
 pageindex = 1, pagesize = 50)

## End(Not run)
</code></pre>

<hr>
<h2 id='manhattan'>Retrieve GWAS summary statistics to create a Manhattan plot</h2><span id='topic+manhattan'></span>

<h3>Description</h3>

<p>This function retrieves summary statistics for a given GWAS study ID, which are used to generate a Manhattan plot.
The Manhattan plot is a graphical representation of genetic association studies, particularly in genome-wide association studies (GWAS).
It displays the results of statistical associations between genetic variants and a trait or disease of interest across the genome.
This function returns a data frame of the underlying data, which can be used to recreate the Manhattan plot using the <code>plot_manhattan()</code> function,
or for custom plots and downstream analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattan(study_id, pageindex = 0, pagesize = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="manhattan_+3A_study_id">study_id</code></td>
<td>
<p>Character: Open Targets Genetics generated ID for the GWAS study.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_pageindex">pageindex</code></td>
<td>
<p>Int: Index of the current page (pagination index &gt;= 0).</p>
</td></tr>
<tr><td><code id="manhattan_+3A_pagesize">pagesize</code></td>
<td>
<p>Int: Number of records in a page (pagination size &gt; 0).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame containing the Manhattan associations for the input study ID. The table consists of the following columns:
</p>

<ul>
<li><p><code>pval_mantissa</code>: Numeric vector. Mantissa of the p-value.
</p>
</li>
<li><p><code>pval_exponent</code>: Integer vector. Exponent of the p-value.
</p>
</li>
<li><p><code>credible_set_size</code>: Integer vector. Size of the credible set.
</p>
</li>
<li><p><code>ld_set_size</code>: Integer vector. Size of the LD set.
</p>
</li>
<li><p><code>total_set_size</code>: Integer vector. Total size of the set.
</p>
</li>
<li><p><code>pval</code>: Numeric vector. P-value.
</p>
</li>
<li><p><code>odds_ratio</code>: Logical vector. Odds ratio.
</p>
</li>
<li><p><code>odds_ratio_ci_lower</code>: Logical vector. Lower confidence interval of the odds ratio.
</p>
</li>
<li><p><code>odds_ratio_ci_upper</code>: Logical vector. Upper confidence interval of the odds ratio.
</p>
</li>
<li><p><code>beta</code>: Numeric vector. Beta value.
</p>
</li>
<li><p><code>beta_ci_lower</code>: Numeric vector. Lower confidence interval of the beta value.
</p>
</li>
<li><p><code>beta_ci_upper</code>: Numeric vector. Upper confidence interval of the beta value.
</p>
</li>
<li><p><code>direction</code>: Character vector. Direction of the effect.
</p>
</li>
<li><p><code>best_genes_score</code>: Numeric vector. Score of the best genes.
</p>
</li>
<li><p><code>best_genes_gene_id</code>: Character vector. Gene ID of the best genes.
</p>
</li>
<li><p><code>best_genes_gene_symbol</code>: Character vector. Gene symbol of the best genes.
</p>
</li>
<li><p><code>best_coloc_genes_score</code>: Numeric vector. Score of the best colocated genes.
</p>
</li>
<li><p><code>best_coloc_genes_gene_id</code>: Character vector. Gene ID of the best colocated genes.
</p>
</li>
<li><p><code>best_coloc_genes_gene_symbol</code>: Character vector. Gene symbol of the best colocated genes.
</p>
</li>
<li><p><code>best_locus2genes_score</code>: Numeric vector. Score of the best locus-to-genes.
</p>
</li>
<li><p><code>best_locus2genes_gene_id</code>: Character vector. Gene ID of the best locus-to-genes.
</p>
</li>
<li><p><code>best_locus2genes_gene_symbol</code>: Character vector. Gene symbol of the best locus-to-genes.
</p>
</li>
<li><p><code>variant_id</code>: Character vector. Variant ID.
</p>
</li>
<li><p><code>variant_position</code>: Integer vector. Variant position.
</p>
</li>
<li><p><code>variant_chromosome</code>: Character vector. Variant chromosome.
</p>
</li>
<li><p><code>variant_rs_id</code>: Character vector. Variant rsID.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- manhattan(study_id = "GCST90002357")
result &lt;- manhattan(study_id = "GCST90002357", pageindex = 2, pagesize = 50)

## End(Not run)
</code></pre>

<hr>
<h2 id='overlapInfoForStudy'>Retrieves overlap info for a study and a list of studies</h2><span id='topic+overlapInfoForStudy'></span>

<h3>Description</h3>

<p>For an input study ID and a list of other study IDs, this function returns two elements.
One contains the overlap information in a table format, and the other element is the variant intersection set,
representing an overlap between two variants of the two given studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>overlapInfoForStudy(study_id, study_ids = list())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="overlapInfoForStudy_+3A_study_id">study_id</code></td>
<td>
<p>Character: Study ID(s) generated by Open Targets Genetics (e.g GCST90002357).</p>
</td></tr>
<tr><td><code id="overlapInfoForStudy_+3A_study_ids">study_ids</code></td>
<td>
<p>Character: generated ID for variants by Open Targets Genetics (e.g. 1_154119580_C_A) or rsId (rs2494663).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing a data frame of overlap information and the variant intersection set.
The overlap information table (overlap_info) consists of the following columns:
</p>

<ul>
<li><p><code>studyId</code>: <em>Character vector</em>. Study ID.
</p>
</li>
<li><p><code>traitReported</code>: <em>Character vector</em>. Reported trait.
</p>
</li>
<li><p><code>traitCategory</code>: <em>Character vector</em>. Trait category.
</p>
</li>
<li><p><code>variantIdA</code>: <em>Character vector</em>. Variant ID from study A.
</p>
</li>
<li><p><code>variantIdB</code>: <em>Character vector</em>. Variant ID from study B.
</p>
</li>
<li><p><code>overlapAB</code>: <em>Integer vector</em>. Number of overlaps between variants A and B.
</p>
</li>
<li><p><code>distinctA</code>: <em>Integer vector</em>. Number of distinct variants in study A.
</p>
</li>
<li><p><code>distinctB</code>: <em>Integer vector</em>. Number of distinct variants in study B.
</p>
</li>
<li><p><code>study.studyId</code>: <em>Character vector</em>. Study ID from study list.
</p>
</li>
<li><p><code>study.traitReported</code>: <em>Character vector</em>. Reported trait from study list.
</p>
</li>
<li><p><code>study.traitCategory</code>: <em>Character vector</em>. Trait category from study list.
</p>
</li></ul>

<p>The variant intersection set (variant_intersection_set) is a character vector representing the intersection of variants.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- overlapInfoForStudy(study_id = "GCST90002357",
 study_ids = list("GCST90025975", "GCST90025962"))

## End(Not run)
</code></pre>

<hr>
<h2 id='pheWAS'>Retrieve PheWAS (Phenome Wide Association Studies) data for a variant.</h2><span id='topic+pheWAS'></span>

<h3>Description</h3>

<p>PheWAS (<em>Phenome-wide association study</em>) is a method that investigates the relationships
between genetic variants and traits or phenotypes, helping in the study of their
potential influence on multiple traits or diseases concurrently. This function retrieves
the traits associated with a given variant in the <em>UK Biobank</em>, <em>FinnGen</em>,
and/or <em>GWAS Catalog</em> summary statistics repository (only traits with a p-value
less than 0.005 are returned).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pheWAS(variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pheWAS_+3A_variant_id">variant_id</code></td>
<td>
<p>Character: generated ID for variants by Open Targets Genetics (e.g. 1_154119580_C_A) or rsId (rs2494663).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with PheWAS associations.
</p>
<p>The output data frame contains the following columns:
</p>

<ul>
<li><p><code>totalGWASStudies</code>: An integer indicating the total number of GWAS studies where the variant is associated.
</p>
</li>
<li><p><code>pval</code>: A numeric value representing the p-value of the association between the variant and the trait.
</p>
</li>
<li><p><code>beta</code>: A numeric value representing the beta value, which represents the effect size of the variant on the trait.
</p>
</li>
<li><p><code>oddsRatio</code>: A numeric value representing the odds ratio, measuring the association between the variant and the trait.
</p>
</li>
<li><p><code>nTotal</code>: An integer indicating the total number of participants in the study.
</p>
</li>
<li><p><code>study.studyId</code>: A character vector representing the study ID.
</p>
</li>
<li><p><code>study.source</code>: A character vector representing the source of the study.
</p>
</li>
<li><p><code>study.pmid</code>: A character vector representing the PubMed ID (PMID) of the study.
</p>
</li>
<li><p><code>study.pubDate</code>: A character vector representing the publication date of the study.
</p>
</li>
<li><p><code>study.traitReported</code>: A character vector representing the reported trait associated with the variant.
</p>
</li>
<li><p><code>study.traitCategory</code>: A character vector representing the category of the trait.
</p>
</li></ul>



<h3>References</h3>

<p>Pendergrass, S A et al. “The use of phenome-wide association studies (PheWAS) for exploration of
novel genotype-phenotype relationships and pleiotropy discovery.” Genetic epidemiology vol. 35,5 (2011): 410-22. doi:10.1002/gepi.20589
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- pheWAS(variant_id = "1_154549918_C_A")
result &lt;- pheWAS(variant_id = "rs72698179")

## End(Not run)

</code></pre>

<hr>
<h2 id='plot_coloc'>Scatter plot of colocalisations for gene and reported traits.</h2><span id='topic+plot_coloc'></span>

<h3>Description</h3>

<p>Generates a scatter plot using the results from <code>colocalisationsForGene()</code>
function as an input. The reported trait in each study are shown on the
<em>x-axis</em> and plotted against their corresponding <code>-log2(H4/H3)</code>
values on the <em>y-axis</em>, indicating the evidence of colocalisation between
the molecular QTLs reported in each study and the explored gene. The molecular
QTLs are mapped to the colors of the points. If the results of
<code>colocalisationsForGene()</code> includes the data for multiple genes, they will
be plotted in separate panels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_coloc(data, biobank = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_coloc_+3A_data">data</code></td>
<td>
<p>Data Frame: result of colocalisationsForGene function in data
frame format, contacting the phewas information for a variant id</p>
</td></tr>
<tr><td><code id="plot_coloc_+3A_biobank">biobank</code></td>
<td>
<p>Logical: <code>TRUE</code> and <code>FALSE</code> variable, with the default value of
FALSE which will keep the data that are from UKbioBank beside the published GWAS
data. In case, this parameter is set to <code>TRUE</code>, only UKbioBank data will be kept
which has not been published.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A horizontal bar plot for colocalisation of information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
plot_out &lt;- colocalisationsForGene(genes = "ENSG00000169174") %&gt;%
          plot_coloc(biobank = FALSE)
plot_out &lt;- colocalisationsForGene(genes = "PCSK9") %&gt;%
          plot_coloc(biobank = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='plot_l2g'>Radar plot for L2G partial scores from <code>studiesAndLeadVariantsForGeneByL2G()</code></h2><span id='topic+plot_l2g'></span>

<h3>Description</h3>

<p>This function returns a radar plot to compare the partial scores,
important for prioritising the causal genes that are obtained
from the <code>studiesAndLeadVariantsForGeneByL2G()</code> function. The user can
decide to plot only for a specific disease by specifying an <code>EFO</code> ID
for the <code>disease</code> argument, otherwise the returned plot will will facet
based on existing traits/diseases in the outputs from <code>studiesAndLeadVariantsForGeneByL2G()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_l2g(data, disease_efo = NULL, l2g_cutoff = 0.5, top_n_disease = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_l2g_+3A_data">data</code></td>
<td>
<p>Data frame: result of <code>studiesAndLeadVariantsForGeneByL2G</code> function.</p>
</td></tr>
<tr><td><code id="plot_l2g_+3A_disease_efo">disease_efo</code></td>
<td>
<p>Character: Input EFO id to filter the L2G data for a particular disease.</p>
</td></tr>
<tr><td><code id="plot_l2g_+3A_l2g_cutoff">l2g_cutoff</code></td>
<td>
<p>Numeric: Sets the minimum L2G score threshold for diseases to be considered in the plot.</p>
</td></tr>
<tr><td><code id="plot_l2g_+3A_top_n_disease">top_n_disease</code></td>
<td>
<p>Numeric: Determines the number of top diseases to plot for each gene, ranked by L2G score.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A radar plot for the input disease and the genes associated with that disease.
The variables shown include L2G score, chromatin interaction, variant pathogenicity and distance.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- studiesAndLeadVariantsForGeneByL2G(list("ENSG00000167207","ENSG00000096968",
  "ENSG00000138821", "ENSG00000125255")) %&gt;% plot_l2g(disease = "EFO_0003767")
p

## End(Not run)

</code></pre>

<hr>
<h2 id='plot_manhattan'>Plot results from <code>manhattan()</code></h2><span id='topic+plot_manhattan'></span>

<h3>Description</h3>

<p>This function generates a Manhattan plot using the statistical summary data
obtained from the <code>manhattan()</code> function. Top 3 genes (based on p-value) are annotated per chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_manhattan(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_manhattan_+3A_data">data</code></td>
<td>
<p>Data frame containing the necessary columns from <code>manhattan()</code> output for plotting:
- <code>variant_position</code>: Variant position
- <code>variant_chromosome</code>: Variant chromosome
- <code>pval</code>: P-value
- <code>variant_id</code>: Variant ID
- <code>best_locus2genes_score</code>: Best locus2genes score
- <code>best_locus2genes_gene_symbol</code>: Best locus2genes gene symbol</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Manhattan plot visualizing the GWAS results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- manhattan(study_id = "GCST003044") %&gt;% plot_manhattan()

p

## End(Not run)
</code></pre>

<hr>
<h2 id='plot_phewas'>Plot <code>PheWAS()</code> results.</h2><span id='topic+plot_phewas'></span>

<h3>Description</h3>

<p>This plot visualizes which traits are associated with the user's selected variant in the UK Biobank,
FinnGen, and/or GWAS Catalog summary statistics repository based on PheWAS analysis.
The associated traits are mapped onto the <em>x-axis</em>, and their corresponding -log10(p-value) values are plotted on the <em>y-axis</em>.
A horizontal line is shown at a p-value cutoff of 0.005 to indicate significant associations.
Associations above this cutoff are labeled with the trait's name, and the sources of the associations are color-coded as points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_phewas(
  data,
  disease = TRUE,
  source = c("GCST", "FINNGEN", "NEALE", "SAIGE")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_phewas_+3A_data">data</code></td>
<td>
<p>Data Frame: The result of the <code>PheWAS()</code> function in data frame format, containing the PheWAS information for a selected variant ID.</p>
</td></tr>
<tr><td><code id="plot_phewas_+3A_disease">disease</code></td>
<td>
<p>Logical: A logical variable indicating whether to filter the PheWAS data for disease (default: TRUE).</p>
</td></tr>
<tr><td><code id="plot_phewas_+3A_source">source</code></td>
<td>
<p>Character vector: Choices for the data sources of PheWAS analysis, including FINNGEN, GCST, NEALE (UKBiobank), and SAIGE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot to prioritize variants based on their -log10(p-value).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p &lt;- pheWAS(variant_id = "14_87978408_G_A") %&gt;%
     plot_phewas(disease = TRUE)
p

## End(Not run)
</code></pre>

<hr>
<h2 id='qtlColocalisationVariantQuery'>Retrieve  QTL colocalisation results of a variant</h2><span id='topic+qtlColocalisationVariantQuery'></span>

<h3>Description</h3>

<p>The colocalisation analysis in Open Target Genetics is performed using the coloc method (Giambartolomei et al., 2014).
Coloc is a Bayesian method which, for two traits, integrates evidence over all variants
at a locus to evaluate the following hypotheses:
- H0: No association with either trait
- H1: Association with trait 1, not with trait 2
- H2: Association with trait 2, not with trait 1
- H3: Association with trait 1 and trait 2, two independent SNPs
- H4: Association with trait 1 and trait 2, one shared SNP
This analysis tests whether two independent associations at the same locus are consistent
with having a shared causal variant. Colocalisation of two independent associations from two GWAS studies may suggest a shared causal mechanism.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qtlColocalisationVariantQuery(study_id, variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qtlColocalisationVariantQuery_+3A_study_id">study_id</code></td>
<td>
<p>Character:  Study ID(s) generated by Open Targets Genetics (e.g GCST90002357).</p>
</td></tr>
<tr><td><code id="qtlColocalisationVariantQuery_+3A_variant_id">variant_id</code></td>
<td>
<p>Character: generated ID for variants by Open Targets Genetics (e.g. 1_154119580_C_A) or rsId (rs2494663).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame of the colocalisation information for a lead variant in a specific study.
The output is a tidy data frame with the following data structure:
</p>

<ul>
<li><p><code>qtlStudyName</code>: <em>Character vector</em>. QTL study name.
</p>
</li>
<li><p><code>phenotypeId</code>: <em>Character vector</em>. Phenotype ID.
</p>
</li>
<li><p><code>gene.id</code>: <em>Character vector</em>. Gene ID.
</p>
</li>
<li><p><code>gene.symbol</code>: <em>Character vector</em>. Gene symbol.
</p>
</li>
<li><p><code>name</code>: <em>Character vector</em>. Tissue name.
</p>
</li>
<li><p><code>indexVariant.id</code>: <em>Character vector</em>. Index variant ID.
</p>
</li>
<li><p><code>indexVariant.rsId</code>: <em>Character vector</em>. Index variant rsID.
</p>
</li>
<li><p><code>beta</code>: <em>Numeric</em>. Beta value.
</p>
</li>
<li><p><code>h4</code>: <em>Numeric</em>. h4 value.
</p>
</li>
<li><p><code>h3</code>: <em>Numeric</em>. h3 value.
</p>
</li>
<li><p><code>log2h4h3</code>: <em>Numeric</em>. Log2(h4/h3) value.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- qtlColocalisationVariantQuery(study_id = "GCST90002357", variant_id = "1_154119580_C_A")
result &lt;- qtlColocalisationVariantQuery(study_id = "GCST90002357", variant_id = "rs2494663")

## End(Not run)
</code></pre>

<hr>
<h2 id='qtlCredibleSet'>Retrieve calculated QTL summary statistics for credible variant set.</h2><span id='topic+qtlCredibleSet'></span>

<h3>Description</h3>

<p>In Open Targets Genetics, the lead variants are expanded into a more comprehensive set of candidate causal variants referred to as the tag variants.
This function retrieves calculated summary statistics for tag variants included in a lead variant colocalization analysis for a
given study (which links a top loci with a trait). The user can filter the results by desired biofeature (e.g tissue, cell type,...)
the function obtains tag variant information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qtlCredibleSet(study_id, variant_id, gene, biofeature)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qtlCredibleSet_+3A_study_id">study_id</code></td>
<td>
<p>Character:  Study ID(s) generated by Open Targets Genetics (e.g GCST90002357).</p>
</td></tr>
<tr><td><code id="qtlCredibleSet_+3A_variant_id">variant_id</code></td>
<td>
<p>Character: generated ID for variants by Open Targets Genetics (e.g. 1_154119580_C_A) or rsId (rs2494663).</p>
</td></tr>
<tr><td><code id="qtlCredibleSet_+3A_gene">gene</code></td>
<td>
<p>Character: Gene ENSEMBL ID (e.g. ENSG00000169174) or gene symbol (e.g. PCSK9).</p>
</td></tr>
<tr><td><code id="qtlCredibleSet_+3A_biofeature">biofeature</code></td>
<td>
<p>Character: Represents either a tissue, cell type, aggregation type, protein type, etc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame of results from the QTL credible set of variants consisting of the following columns:
</p>

<ul>
<li><p><code>tagVariant.id</code>: <em>Character vector</em>. Tag variant ID.
</p>
</li>
<li><p><code>tagVariant.rsId</code>: <em>Character vector</em>. Tag variant rsID.
</p>
</li>
<li><p><code>pval</code>: <em>Numeric</em>. P-value.
</p>
</li>
<li><p><code>se</code>: <em>Numeric</em>. Standard error.
</p>
</li>
<li><p><code>beta</code>: <em>Numeric</em>. Beta value.
</p>
</li>
<li><p><code>postProb</code>: <em>Numeric</em>. Posterior probability.
</p>
</li>
<li><p><code>MultisignalMethod</code>: <em>Character vector</em>. Multisignal method.
</p>
</li>
<li><p><code>logABF</code>: <em>Numeric</em>. Logarithm of approximate Bayes factor.
</p>
</li>
<li><p><code>is95</code>: <em>Logical</em>. Indicates if the variant has a 95
</p>
</li>
<li><p><code>is99</code>: <em>Logical</em>. Indicates if the variant has a 99
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- qtlCredibleSet(study_id = "Braineac2", variant_id = "1_55053079_C_T",
    gene = "ENSG00000169174", biofeature = "SUBSTANTIA_NIGRA")
result &lt;- qtlCredibleSet(study_id = "Braineac2", variant_id = "rs7552841",
    gene = "PCSK9", biofeature = "SUBSTANTIA_NIGRA")

## End(Not run)

</code></pre>

<hr>
<h2 id='run_custom_query'>Running custom GraphQL queries</h2><span id='topic+run_custom_query'></span>

<h3>Description</h3>

<p>Running custom GraphQL queries
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_custom_query(variableList, query, query_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_custom_query_+3A_variablelist">variableList</code></td>
<td>
<p>is a list format which includes the key value pair list of genes/variants/study ids to be queries.</p>
</td></tr>
<tr><td><code id="run_custom_query_+3A_query">query</code></td>
<td>
<p>is a GraphQL desired query body to be run.</p>
</td></tr>
<tr><td><code id="run_custom_query_+3A_query_name">query_name</code></td>
<td>
<p>is a string format of the query name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a flatten json file format
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
otargen::run_custom_query (variableList, query, query_name)

## End(Not run)

</code></pre>

<hr>
<h2 id='studiesAndLeadVariantsForGeneByL2G'>Retrieve &quot;locus-to-gene&quot; (L2G) model summary data for a gene.</h2><span id='topic+studiesAndLeadVariantsForGeneByL2G'></span>

<h3>Description</h3>

<p>The &quot;locus-to-gene&quot; (L2G) model derives features to prioritize likely causal genes at each GWAS
locus based on genetic and functional genomics features. The main categories of predictive features are:
</p>

<ul>
<li><p> Distance: Distance from credible set variants to the gene.
</p>
</li>
<li><p> Molecular QTL colocalization: Colocalization with molecular QTLs.
</p>
</li>
<li><p> Chromatin interaction: Interactions, such as promoter-capture Hi-C.
</p>
</li>
<li><p> Variant pathogenicity: Pathogenicity scores from VEP (Variant Effect Predictor).
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>studiesAndLeadVariantsForGeneByL2G(gene, l2g = NA, pvalue = NA, vtype = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="studiesAndLeadVariantsForGeneByL2G_+3A_gene">gene</code></td>
<td>
<p>Character: Gene ENSEMBL ID (e.g. ENSG00000169174) or gene symbol (e.g. PCSK9). This argument can take a list of genes too.</p>
</td></tr>
<tr><td><code id="studiesAndLeadVariantsForGeneByL2G_+3A_l2g">l2g</code></td>
<td>
<p>Numeric: Locus-to-gene (L2G) cutoff score. (Default: NA)</p>
</td></tr>
<tr><td><code id="studiesAndLeadVariantsForGeneByL2G_+3A_pvalue">pvalue</code></td>
<td>
<p>Character: P-value cutoff. (Default: NA)</p>
</td></tr>
<tr><td><code id="studiesAndLeadVariantsForGeneByL2G_+3A_vtype">vtype</code></td>
<td>
<p>Character: Most severe consequence to filter the variant types, including &quot;intergenic_variant&quot;,
&quot;upstream_gene_variant&quot;, &quot;intron_variant&quot;, &quot;missense_variant&quot;, &quot;5_prime_UTR_variant&quot;,
&quot;non_coding_transcript_exon_variant&quot;, &quot;splice_region_variant&quot;. (Default: NULL)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function also provides additional filtering parameters to narrow the results based following parameters (see below)
</p>


<h3>Value</h3>

<p>Returns a data frame containing the input gene ID and its data for the L2G model. The table consists of the following columns:
</p>

<ul>
<li><p><code>yProbaModel</code>: <em>Numeric</em>. L2G score.
</p>
</li>
<li><p><code>yProbaDistance</code>: <em>Numeric</em>. Distance.
</p>
</li>
<li><p><code>yProbaInteraction</code>: <em>Numeric</em>. Chromatin interaction.
</p>
</li>
<li><p><code>yProbaMolecularQTL</code>: <em>Numeric</em>. Molecular QTL.
</p>
</li>
<li><p><code>yProbaPathogenicity</code>: <em>Numeric</em>. Pathogenicity.
</p>
</li>
<li><p><code>pval</code>: <em>Numeric</em>. P-value.
</p>
</li>
<li><p><code>beta.direction</code>: <em>Character</em>. Beta direction.
</p>
</li>
<li><p><code>beta.betaCI</code>: <em>Numeric</em>. Beta confidence interval.
</p>
</li>
<li><p><code>beta.betaCILower</code>: <em>Numeric</em>. Lower bound of the beta confidence interval.
</p>
</li>
<li><p><code>beta.betaCIUpper</code>: <em>Numeric</em>. Upper bound of the beta confidence interval.
</p>
</li>
<li><p><code>odds.oddsCI</code>: <em>Numeric</em>. Odds ratio confidence interval.
</p>
</li>
<li><p><code>odds.oddsCILower</code>: <em>Numeric</em>. Lower bound of the odds ratio confidence interval.
</p>
</li>
<li><p><code>odds.oddsCIUpper</code>: <em>Numeric</em>. Upper bound of the odds ratio confidence interval.
</p>
</li>
<li><p><code>study.studyId</code>: <em>Character</em>. Study ID.
</p>
</li>
<li><p><code>study.traitReported</code>: <em>Character</em>. Reported trait.
</p>
</li>
<li><p><code>study.traitCategory</code>: <em>Character</em>. Trait category.
</p>
</li>
<li><p><code>study.pubDate</code>: <em>Character</em>. Publication date.
</p>
</li>
<li><p><code>study.pubTitle</code>: <em>Character</em>. Publication title.
</p>
</li>
<li><p><code>study.pubAuthor</code>: <em>Character</em>. Publication author.
</p>
</li>
<li><p><code>study.pubJournal</code>: <em>Character</em>. Publication journal.
</p>
</li>
<li><p><code>study.pmid</code>: <em>Character</em>. PubMed ID.
</p>
</li>
<li><p><code>study.hasSumstats</code>: <em>Logical</em>. Indicates if the study has summary statistics.
</p>
</li>
<li><p><code>study.nCases</code>: <em>Integer</em>. Number of cases in the study.
</p>
</li>
<li><p><code>study.numAssocLoci</code>: <em>Integer</em>. Number of associated loci.
</p>
</li>
<li><p><code>study.nTotal</code>: <em>Integer</em>. Total number of samples in the study.
</p>
</li>
<li><p><code>study.traitEfos</code>: <em>Character</em>. Trait EFOs.
</p>
</li>
<li><p><code>variant.id</code>: <em>Character</em>. Variant ID.
</p>
</li>
<li><p><code>variant.rsId</code>: <em>Character</em>. Variant rsID.
</p>
</li>
<li><p><code>variant.chromosome</code>: <em>Character</em>. Variant chromosome.
</p>
</li>
<li><p><code>variant.position</code>: <em>Integer</em>. Variant position.
</p>
</li>
<li><p><code>variant.refAllele</code>: <em>Character</em>. Variant reference allele.
</p>
</li>
<li><p><code>variant.altAllele</code>: <em>Character</em>. Variant alternate allele.
</p>
</li>
<li><p><code>variant.nearestCodingGeneDistance</code>: <em>Integer</em>. Distance to the nearest coding gene.
</p>
</li>
<li><p><code>variant.nearestGeneDistance</code>: <em>Integer</em>. Distance to the nearest gene.
</p>
</li>
<li><p><code>variant.mostSevereConsequence</code>: <em>Character</em>. Most severe consequence.
</p>
</li>
<li><p><code>variant.nearestGene.id</code>: <em>Character</em>. Nearest gene ID.
</p>
</li>
<li><p><code>variant.nearestCodingGene.id</code>: <em>Character</em>. Nearest coding gene ID.
</p>
</li>
<li><p><code>ensembl_id</code>: <em>Character</em>. Ensembl ID.
</p>
</li>
<li><p><code>gene_symbol</code>: <em>Character</em>. Gene symbol.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- studiesAndLeadVariantsForGeneByL2G(genes = c("ENSG00000163946",
     "ENSG00000169174", "ENSG00000143001"), l2g = 0.7)
result &lt;- studiesAndLeadVariantsForGeneByL2G(genes = "ENSG00000169174",
     l2g = 0.6, pvalue = 1e-8, vtype = c("intergenic_variant", "intron_variant"))
result &lt;- studiesAndLeadVariantsForGeneByL2G(genes = "TMEM61")

## End(Not run)
</code></pre>

<hr>
<h2 id='studyInfo'>Retrieve study summary information.</h2><span id='topic+studyInfo'></span>

<h3>Description</h3>

<p>For a given study id, this function returns a data frame of relevant information about
the GWAS study, such as PubMed ID, studied trait EFO ID, case/control size, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>studyInfo(study_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="studyInfo_+3A_study_id">study_id</code></td>
<td>
<p>Character:  Study ID(s) generated by Open Targets Genetics (e.g GCST90002357).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame (in tibble format) containing the summary iformation about a GWAS study. The data frame has the following data structure:
</p>

<ul>
<li><p><code>studyId</code>: <em>Character</em>. Study ID.
</p>
</li>
<li><p><code>traitReported</code>: <em>Character</em>. Reported trait.
</p>
</li>
<li><p><code>source</code>: <em>Character</em>. Source.
</p>
</li>
<li><p><code>traitEfos</code>: <em>Character</em>. Trait EFO ID.
</p>
</li>
<li><p><code>pmid</code>: <em>Character</em>. PubMed ID.
</p>
</li>
<li><p><code>pubDate</code>: <em>Character</em>. Publication date.
</p>
</li>
<li><p><code>pubJournal</code>: <em>Character</em>. Publication journal.
</p>
</li>
<li><p><code>pubTitle</code>: <em>Character</em>. Publication title.
</p>
</li>
<li><p><code>pubAuthor</code>: <em>Character</em>. Publication author.
</p>
</li>
<li><p><code>hasSumstats</code>: <em>Character</em>. Indicates if the study has summary statistics.
</p>
</li>
<li><p><code>ancestryInitial</code>: <em>Character</em>. Initial ancestry.
</p>
</li>
<li><p><code>nInitial</code>: <em>Character</em>. Initial sample size.
</p>
</li>
<li><p><code>nReplication</code>: <em>Character</em>. Replication sample size.
</p>
</li>
<li><p><code>traitCategory</code>: <em>Character</em>. Trait category.
</p>
</li>
<li><p><code>numAssocLoci</code>: <em>Character</em>. Number of associated loci.
</p>
</li>
<li><p><code>nTotal</code>: <em>Character</em>. Total sample size.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- studyInfo(study_id = "GCST90002357")

## End(Not run)
</code></pre>

<hr>
<h2 id='studyLocus2GeneTable'>Retrieve the locus-to-gene (L2G) data table for loci genes.</h2><span id='topic+studyLocus2GeneTable'></span>

<h3>Description</h3>

<p>This function fetches the locus-to-gene (L2G) pipeline summary data table for
the neighboring genes of a variant in a GWAS study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>studyLocus2GeneTable(study_id, variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="studyLocus2GeneTable_+3A_study_id">study_id</code></td>
<td>
<p>Character: Study ID(s) generated by Open Targets Genetics (e.g GCST90002357).</p>
</td></tr>
<tr><td><code id="studyLocus2GeneTable_+3A_variant_id">variant_id</code></td>
<td>
<p>Character: generated ID for variants by Open Targets Genetics (e.g. 1_154119580_C_A) or rsId (rs2494663).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame with the summary statistics of the study and a data table containing various
calculated scores and features for any lead variant. The output table has the following data structure:
</p>

<ul>
<li><p><code>studyId</code>: <em>Character</em>. Study ID.
</p>
</li>
<li><p><code>variant.id</code>: <em>Character</em>. Variant ID.
</p>
</li>
<li><p><code>variant.rsId</code>: <em>Character</em>. Variant rsID.
</p>
</li>
<li><p><code>yProbaDistance</code>: <em>Numeric</em>. Distance score.
</p>
</li>
<li><p><code>yProbaModel</code>: <em>Numeric</em>. Model score.
</p>
</li>
<li><p><code>yProbaMolecularQTL</code>: <em>Numeric</em>. Molecular QTL score.
</p>
</li>
<li><p><code>yProbaPathogenicity</code>: <em>Numeric</em>. Pathogenicity score.
</p>
</li>
<li><p><code>yProbaInteraction</code>: <em>Numeric</em>. Interaction score.
</p>
</li>
<li><p><code>hasColoc</code>: <em>Logical</em>. Indicates if colocalization data is available.
</p>
</li>
<li><p><code>distanceToLocus</code>: <em>Numeric</em>. Distance to the locus.
</p>
</li>
<li><p><code>gene.id</code>: <em>Character</em>. Gene ID.
</p>
</li>
<li><p><code>gene.symbol</code>: <em>Character</em>. Gene symbol.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- studyLocus2GeneTable(study_id = "GCST90002357", variant_id = "1_154119580_C_A")
result &lt;- studyLocus2GeneTable(study_id = "GCST90002357", variant_id = "rs2494663")

## End(Not run)
</code></pre>

<hr>
<h2 id='studyVariants'>Retrieves all variants for a study.</h2><span id='topic+studyVariants'></span>

<h3>Description</h3>

<p>For an input study ID, this function returns information of all variants across associated loci. The output also includes
information about the associated genes within the each loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>studyVariants(study_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="studyVariants_+3A_study_id">study_id</code></td>
<td>
<p>Character: Study ID(s) generated by Open Targets Genetics (e.g GCST90002357).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of two data frames.
</p>
<p>the first data frame (tibble format) includes the loci data frame with following data structure:
</p>

<ul>
<li><p><code>variant.id</code>: <em>Character</em>. Variant ID.
</p>
</li>
<li><p><code>pval</code>: <em>Numeric</em>. P-value.
</p>
</li>
<li><p><code>variant.nearestCodingGene.symbol</code>: <em>Character</em>. Symbol of the nearest coding gene to the variant.
</p>
</li>
<li><p><code>variant.rsId</code>: <em>Character</em>. Variant rsID.
</p>
</li>
<li><p><code>variant.chromosome</code>: <em>Character</em>. Chromosome of the variant.
</p>
</li>
<li><p><code>variant.position</code>: <em>Integer</em>. Position of the variant.
</p>
</li>
<li><p><code>variant.nearestCodingGeneDistance</code>: <em>Integer</em>. Distance to the nearest coding gene.
</p>
</li>
<li><p><code>credibleSetSize</code>: <em>Integer</em>. Size of the credible set.
</p>
</li>
<li><p><code>ldSetSize</code>: <em>Integer</em>. Size of the LD set.
</p>
</li>
<li><p><code>oddsRatio</code>: <em>Numeric</em>. Odds ratio.
</p>
</li>
<li><p><code>beta</code>: <em>Numeric</em>. Beta value.
</p>
</li></ul>

<p>The second data frame includes gene information with following data structure:
</p>

<ul>
<li><p><code>score</code>: <em>Numeric</em>. Gene score.
</p>
</li>
<li><p><code>gene.id</code>: <em>Character</em>. Gene ID.
</p>
</li>
<li><p><code>gene.symbol</code>: <em>Character</em>. Gene symbol.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- studyVariants(study_id = "GCST003155")

## End(Not run)
</code></pre>

<hr>
<h2 id='tagVariantsAndStudiesForIndexVariant'>Retrieves tag variants and studies for a given index variant.</h2><span id='topic+tagVariantsAndStudiesForIndexVariant'></span>

<h3>Description</h3>

<p>For an input index variant ID, this function fetches information about the tag variants
and associated studies, including scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tagVariantsAndStudiesForIndexVariant(variant_id, pageindex = 0, pagesize = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tagVariantsAndStudiesForIndexVariant_+3A_variant_id">variant_id</code></td>
<td>
<p>Character: Open Targets Genetics generated ID for a variant (CHRPOSITION_REFALLELE_ALTALLELE or rsId).</p>
</td></tr>
<tr><td><code id="tagVariantsAndStudiesForIndexVariant_+3A_pageindex">pageindex</code></td>
<td>
<p>Integer: Index of the current page for pagination (&gt;= 0).</p>
</td></tr>
<tr><td><code id="tagVariantsAndStudiesForIndexVariant_+3A_pagesize">pagesize</code></td>
<td>
<p>Integer: Number of records in a page for pagination (&gt; 0).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame containing the variant associations connected to the input index variant.
The columns in the data frame are as follows:
</p>

<ul>
<li><p><code>tagVariant.id</code>: <em>Character</em>. Tag variant ID.
</p>
</li>
<li><p><code>tagVariant.chromosome</code>: <em>Character</em>. Chromosome of the tag variant.
</p>
</li>
<li><p><code>tagVariant.rsId</code>: <em>Character</em>. rsID of the tag variant.
</p>
</li>
<li><p><code>tagVariant.position</code>: <em>Integer</em>. Position of the tag variant.
</p>
</li>
<li><p><code>study.studyId</code>: <em>Character</em>. Study ID.
</p>
</li>
<li><p><code>study.traitReported</code>: <em>Character</em>. Reported trait of the study.
</p>
</li>
<li><p><code>study.traitCategory</code>: <em>Character</em>. Category of the trait in the study.
</p>
</li>
<li><p><code>pval</code>: <em>Numeric</em>. P-value.
</p>
</li>
<li><p><code>pvalMantissa</code>: <em>Numeric</em>. Mantissa of the p-value.
</p>
</li>
<li><p><code>pvalExponent</code>: <em>Integer</em>. Exponent of the p-value.
</p>
</li>
<li><p><code>nTotal</code>: <em>Integer</em>. Total number of samples.
</p>
</li>
<li><p><code>nCases</code>: <em>Integer</em>. Number of cases in the study.
</p>
</li>
<li><p><code>overallR2</code>: <em>Numeric</em>. Overall R-squared value.
</p>
</li>
<li><p><code>afr1000GProp</code>: <em>Numeric</em>. Proportion in African 1000 Genomes population.
</p>
</li>
<li><p><code>amr1000GProp</code>: <em>Numeric</em>. Proportion in Admixed American 1000 Genomes population.
</p>
</li>
<li><p><code>eas1000GProp</code>: <em>Numeric</em>. Proportion in East Asian 1000 Genomes population.
</p>
</li>
<li><p><code>eur1000GProp</code>: <em>Numeric</em>. Proportion in European 1000 Genomes population.
</p>
</li>
<li><p><code>sas1000GProp</code>: <em>Numeric</em>. Proportion in South Asian 1000 Genomes population.
</p>
</li>
<li><p><code>oddsRatio</code>: <em>Numeric</em>. Odds ratio.
</p>
</li>
<li><p><code>oddsRatioCILower</code>: <em>Numeric</em>. Lower bound of the odds ratio confidence interval.
</p>
</li>
<li><p><code>oddsRatioCIUpper</code>: <em>Numeric</em>. Upper bound of the odds ratio confidence interval.
</p>
</li>
<li><p><code>posteriorProbability</code>: <em>Numeric</em>. Posterior probability.
</p>
</li>
<li><p><code>beta</code>: <em>Numeric</em>. Beta value.
</p>
</li>
<li><p><code>betaCILower</code>: <em>Numeric</em>. Lower bound of the beta value confidence interval.
</p>
</li>
<li><p><code>betaCIUpper</code>: <em>Numeric</em>. Upper bound of the beta value confidence interval.
</p>
</li>
<li><p><code>direction</code>: <em>Character</em>. Direction of the effect.
</p>
</li>
<li><p><code>log10Abf</code>: <em>Numeric</em>. Log base 10 of the approximate Bayes factor.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- tagVariantsAndStudiesForIndexVariant(variant_id = "1_109274968_G_T")
result &lt;- tagVariantsAndStudiesForIndexVariant(variant_id = "1_109274968_G_T"
    ,pageindex = 1, pagesize = 50)
    
## End(Not run)
</code></pre>

<hr>
<h2 id='topOverlappedStudies'>Retrieve top studies having overlap in identified loci.</h2><span id='topic+topOverlappedStudies'></span>

<h3>Description</h3>

<p>For a provided study ID, the function, retrieves top studies with overlap in their identified loci with the queried study loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>topOverlappedStudies(study_id, pageindex = 0, pagesize = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="topOverlappedStudies_+3A_study_id">study_id</code></td>
<td>
<p>Character: Open Targets Genetics generated ID for a GWAS study.</p>
</td></tr>
<tr><td><code id="topOverlappedStudies_+3A_pageindex">pageindex</code></td>
<td>
<p>Integer: Index of the current page for pagination (&gt;= 0).</p>
</td></tr>
<tr><td><code id="topOverlappedStudies_+3A_pagesize">pagesize</code></td>
<td>
<p>Integer: Number of records in a page for pagination (&gt; 0).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame with the top studies containing the following columns:
</p>

<ul>
<li><p><code>study.studyId</code>: <em>Character</em>. Study ID of the input study.
</p>
</li>
<li><p><code>study.traitReported</code>: <em>Character</em>. Reported trait of the input study.
</p>
</li>
<li><p><code>study.traitCategory</code>: <em>Character</em>. Category of the trait in the input study.
</p>
</li>
<li><p><code>topStudiesByLociOverlap.studyId</code>: <em>Character</em>. Study ID of the top associated studies.
</p>
</li>
<li><p><code>topStudiesByLociOverlap.study.studyId</code>: <em>Character</em>. Study ID of the top associated studies.
</p>
</li>
<li><p><code>topStudiesByLociOverlap.study.traitReported</code>: <em>Character</em>. Reported trait of the top associated studies.
</p>
</li>
<li><p><code>topStudiesByLociOverlap.study.traitCategory</code>: <em>Character</em>. Category of the trait in the top associated studies.
</p>
</li>
<li><p><code>topStudiesByLociOverlap.numOverlapLoci</code>: <em>Integer</em>. Number of loci overlapped with the input study.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- topOverlappedStudies(study_id = "GCST006614_3")
result &lt;- topOverlappedStudies(study_id = "NEALE2_6177_1", pageindex = 1, pagesize = 50)

## End(Not run)
</code></pre>

<hr>
<h2 id='variantInfo'>Retrieves information about a variant.</h2><span id='topic+variantInfo'></span>

<h3>Description</h3>

<p>For a given variant ID, this function retrieves information about the variant, including its chromosome,
position, reference allele, alternative allele, rsID, nearest gene, most severe consequence, and allele
frequencies in different populations from gnomAD databse. The Genome Aggregation Database (gnomAD) is a
resource developed by an international coalition of investigators, with the goal of aggregating and
harmonizing both exome and genome sequencing data from a wide variety of large-scale sequencing projects,
and making summary data available for the wider scientific community (see the reference).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variantInfo(variant_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variantInfo_+3A_variant_id">variant_id</code></td>
<td>
<p>Character: generated ID for variants by Open Targets Genetics (e.g. 1_154119580_C_A) or rsId (rs2494663).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a data frame (in tibble format) containing information about the variant.
The data frame has the following structure:
</p>

<ul>
<li><p><code>chromosome</code>: <em>Character</em>. Chromosome of the variant.
</p>
</li>
<li><p><code>position</code>: <em>Integer</em>. Position of the variant.
</p>
</li>
<li><p><code>refAllele</code>: <em>Character</em>. Reference allele.
</p>
</li>
<li><p><code>altAllele</code>: <em>Character</em>. Alternative allele.
</p>
</li>
<li><p><code>rsId</code>: <em>Character</em>. Variant rsID.
</p>
</li>
<li><p><code>chromosomeB37</code>: <em>Character</em>. Chromosome of the variant in build 37 coordinates.
</p>
</li>
<li><p><code>positionB37</code>: <em>Integer</em>. Position of the variant in build 37 coordinates.
</p>
</li>
<li><p><code>id</code>: <em>Character</em>. Variant ID.
</p>
</li>
<li><p><code>nearestGene.id</code>: <em>Character</em>. ID of the nearest gene to the variant.
</p>
</li>
<li><p><code>nearestGene.symbol</code>: <em>Character</em>. Symbol of the nearest gene to the variant.
</p>
</li>
<li><p><code>nearestGeneDistance</code>: <em>Integer</em>. Distance to the nearest gene.
</p>
</li>
<li><p><code>nearestCodingGene.id</code>: <em>Character</em>. ID of the nearest coding gene to the variant.
</p>
</li>
<li><p><code>nearestCodingGene.symbol</code>: <em>Character</em>. Symbol of the nearest coding gene to the variant.
</p>
</li>
<li><p><code>nearestCodingGeneDistance</code>: <em>Integer</em>. Distance to the nearest coding gene.
</p>
</li>
<li><p><code>mostSevereConsequence</code>: <em>Character</em>. Most severe consequence of the variant.
</p>
</li>
<li><p><code>caddRaw</code>: <em>Numeric</em>. CADD raw score.
</p>
</li>
<li><p><code>caddPhred</code>: <em>Numeric</em>. CADD phred score.
</p>
</li>
<li><p><code>gnomadAFR</code>: <em>Numeric</em>. Allele frequency in the African/African-American population in gnomAD.
</p>
</li>
<li><p><code>gnomadAMR</code>: <em>Numeric</em>. Allele frequency in the Latino/Admixed American population in gnomAD.
</p>
</li>
<li><p><code>gnomadASJ</code>: <em>Numeric</em>. Allele frequency in the Ashkenazi Jewish population in gnomAD.
</p>
</li>
<li><p><code>gnomadEAS</code>: <em>Numeric</em>. Allele frequency in the East Asian population in gnomAD.
</p>
</li>
<li><p><code>gnomadFIN</code>: <em>Numeric</em>. Allele frequency in the Finnish population in gnomAD.
</p>
</li>
<li><p><code>gnomadNFE</code>: <em>Numeric</em>. Allele frequency in the Non-Finnish European population in gnomAD.
</p>
</li>
<li><p><code>gnomadNFEEST</code>: <em>Numeric</em>. Allele frequency in the Estonian population in gnomAD.
</p>
</li>
<li><p><code>gnomadNFENWE</code>: <em>Numeric</em>. Allele frequency in the Northwest European population in gnomAD.
</p>
</li>
<li><p><code>gnomadNFESEU</code>: <em>Numeric</em>. Allele frequency in the Southern European population in gnomAD.
</p>
</li>
<li><p><code>gnomadNFEONF</code>: <em>Numeric</em>. Allele frequency in the Other Non-Finnish European population in gnomAD.
</p>
</li>
<li><p><code>gnomadOTH</code>: <em>Numeric</em>. Allele frequency in other populations in gnomAD.
</p>
</li></ul>



<h3>References</h3>

<p>https://gnomad.broadinstitute.org/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result &lt;- variantInfo(variant_id = "rs2494663")

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
