<!DOCTYPE html><html><head><title>Help for package otuSummary</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {otuSummary}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alphaDiversity'>
<p>Calculate the alpha diversity indices</p></a></li>
<li><a href='#bplot'>
<p>Generate barplot with custom controls on the x axis labels</p></a></li>
<li><a href='#calc_bc'>
<p>calculate the Bray-Curtis dissimilarity</p></a></li>
<li><a href='#contrib'>
<p>Contribution of rare/abundant biosphere to the total Bray-Curtis dissimilarity</p></a></li>
<li><a href='#fillTax'>
<p>reshape the heirarchical taxonomy</p></a></li>
<li><a href='#fillTax2'>
<p>reshape the heirarchical taxonomy</p></a></li>
<li><a href='#matrixConvert'>
<p>Convert lower triangular distance matrix into data frame</p></a></li>
<li><a href='#otu4type'>
<p>An example OTU table with samples and taxonomy in rows and otus in columns</p></a></li>
<li><a href='#otuCollap'>
<p>collapse a OTU table at given level.</p></a></li>
<li><a href='#otumothur'>
<p>OTU table generated from 8 lakes over 4 years</p></a></li>
<li><a href='#otuqiime'>
<p>OTU table generated from 8 lakes over 4 years</p></a></li>
<li><a href='#otuReport'>
<p>Summarize the community structure and abundance with OTU table</p></a></li>
<li><a href='#pbray'>
<p>Partition the Bray-Curtis distance dissimilarity</p></a></li>
<li><a href='#rareBiosphere'>
<p>Summarize different subgroups in the rare biosphere</p></a></li>
<li><a href='#slimTax'>
<p>reshape heirarchical taxonomy</p></a></li>
<li><a href='#subOTU'>
<p>Subset a OTU table</p></a></li>
<li><a href='#typeConvert'>
<p>Transpose the data frame if there is data type conversion.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Summarizing OTU Table Regarding the Composition, Abundance and
Beta Diversity of Abundant and Rare Biospheres</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-8-13</td>
</tr>
<tr>
<td>Author:</td>
<td>Sizhong Yang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sizhong Yang &lt;yanglzu@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Summarizes the taxonomic composition, diversity contribution of the rare and abundant community by using OTU (operational taxonomic unit) table which was generated by analyzing pipeline of 'QIIME' or 'mothur'. The rare biosphere in this package is subset by the relative abundance threshold (for details about rare biosphere please see Lynch and Neufeld (2015) &lt;<a href="https://doi.org/10.1038%2Fnrmicro3400">doi:10.1038/nrmicro3400</a>&gt;).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>reshape2 (&ge; 1.4)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/cam315/otuSummary">https://github.com/cam315/otuSummary</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-05 21:29:33 UTC; syang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-05 22:00:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='alphaDiversity'>
Calculate the alpha diversity indices
</h2><span id='topic+alphaDiversity'></span>

<h3>Description</h3>

<p>This function will calculate the alpha diversity indices for the total, abundant and rare biospheres.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alphaDiversity(otutab, siteInCol = FALSE, taxhead = NULL, threshold = 1,
    percent = FALSE, write = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alphaDiversity_+3A_otutab">otutab</code></td>
<td>

<p>A OTU table of microbial community, which can contain a taxonomic column (if siteInCol) or row (if site in rows). The OTU table should be given in numeric (integer) counts.
</p>
</td></tr>
<tr><td><code id="alphaDiversity_+3A_siteincol">siteInCol</code></td>
<td>

<p>Logical, if &quot;TRUE&quot;, the OTU table contains samples in columns and taxa in rows. By default in this function, the siteInCol is FALSE, meaning the samples in rows.
</p>
</td></tr>
<tr><td><code id="alphaDiversity_+3A_taxhead">taxhead</code></td>
<td>

<p>Character, specify the header of taxonomy if there is a taxonomic column in your data. By default this argument is NULL.
</p>
</td></tr>
<tr><td><code id="alphaDiversity_+3A_threshold">threshold</code></td>
<td>

<p>Numeric, the threshold of relative abundance upon which the rare biosphere will be subset.
</p>
</td></tr>
<tr><td><code id="alphaDiversity_+3A_percent">percent</code></td>
<td>

<p>Logical, whether the input OTU table are given in relative abundance. FALSE means that the input OTU table is in numeric counts.
</p>
</td></tr>
<tr><td><code id="alphaDiversity_+3A_write">write</code></td>
<td>

<p>Logical, if TRUE, the result will be written out in a Tab separated data frame.
</p>
</td></tr>
<tr><td><code id="alphaDiversity_+3A_...">...</code></td>
<td>

<p>arguments to be passed to write.table().
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The rare biosphere is defined by the relative abundance cutoffs (which is the &quot;threshold&quot; argument in this function) (Lynch and Neufeld, 2015). This update (version 0.1.2) removed the dependencies on functions &quot;specnumber&quot;, &quot;diversity&quot; and &quot;estimateR&quot; from the R package &quot;vegan&quot; (Oksanen et al, 2013), and removed the &quot;gini&quot; function from package 'reldist' (http://www.stat.ucla.edu/~handcock/RelDist).
</p>


<h3>Value</h3>

<p>The function will return a list of length 3, including indices of observed, shannon, simpson, invsimpson, chao1, chao2, and evenness.

</p>
<table>
<tr><td><code>allBio</code></td>
<td>
<p>The alpha diversity indices for the whole community</p>
</td></tr>
<tr><td><code>abundBio</code></td>
<td>
<p>The alpha diversity indices for the abundant population</p>
</td></tr>
<tr><td><code>rareBio</code></td>
<td>
<p>The alpha diversity indices for the rare biosphere</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sizhong Yang &lt;yanglzu@163.com&gt;
</p>


<h3>References</h3>

<p>Lynch MDJ, Neufeld JD (2015). Ecology and exploration of the rare biosphere. Nature Reviews Microbiology 13: 217-229.
</p>
<p>Oksanen J, Blanchet FG, Kindt R, Legendre P, Minchin PR, O'Hara RB et al (2013). vegan: Community Ecology Package. R package version 2.0-7. http://CRAN.R-project.org/package=vegan.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otumothur)

test1 &lt;- alphaDiversity(otutab = otumothur, siteInCol = TRUE,
    taxhead = "taxonomy", threshold = 1, percent = FALSE, write = FALSE)

test2 &lt;- alphaDiversity(otutab = otumothur[,-ncol(otumothur)], siteInCol = TRUE,
    taxhead = NULL, threshold = 1, percent = FALSE, write = FALSE)
</code></pre>

<hr>
<h2 id='bplot'>
Generate barplot with custom controls on the x axis labels
</h2><span id='topic+bplot'></span>

<h3>Description</h3>

<p>barplot with custom controls on the x axis labels, e.g. rotation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bplot(data, srt = 45, yoff = 0.05, dataoff = 0.025, barcol = "grey", grid = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bplot_+3A_data">data</code></td>
<td>

<p>Numeric, vector to plot in barplot.
</p>
</td></tr>
<tr><td><code id="bplot_+3A_srt">srt</code></td>
<td>

<p>Numeric, rotation degree of the x axis labels.
</p>
</td></tr>
<tr><td><code id="bplot_+3A_yoff">yoff</code></td>
<td>

<p>Numeric, vertical offset of x axis labels.
</p>
</td></tr>
<tr><td><code id="bplot_+3A_dataoff">dataoff</code></td>
<td>

<p>Numeric, vertical offset of data labels in relation to the bar height.
</p>
</td></tr>
<tr><td><code id="bplot_+3A_barcol">barcol</code></td>
<td>

<p>Character, color of bar. By default is grey.
</p>
</td></tr>
<tr><td><code id="bplot_+3A_grid">grid</code></td>
<td>

<p>Logical, whether show the grids in the plot.
</p>
</td></tr>
<tr><td><code id="bplot_+3A_...">...</code></td>
<td>

<p>arguments to be passed to/from other methods.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(otumothur)

summaryInfo &lt;- otuReport(otutab = otumothur, siteInCol = TRUE, taxhead = "taxonomy",
    platform = "qiime", percent = FALSE, taxlevel = "phylum", collap = ";")

length(summaryInfo)
names(summaryInfo)
summaryInfo[[1]]

bplot(summaryInfo[["taxaFreqs"]])
</code></pre>

<hr>
<h2 id='calc_bc'>
calculate the Bray-Curtis dissimilarity
</h2><span id='topic+calc_bc'></span>

<h3>Description</h3>

<p>This function will calculate the Bray-Curtis distance matrix for community data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_bc(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_bc_+3A_df">df</code></td>
<td>

<p>data frame or matrix, i.e. OTU or ASV table with sites in rows and species in columns.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will calculate the dissimilarity and output as lower triangular distance matrix.
</p>


<h3>Value</h3>

<p>The function returns lower triangular distrance matrix.




</p>


<h3>Author(s)</h3>

<p>Sizhong Yang &lt;yanglzu@163.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otuqiime)
mat &lt;- calc_bc(t(otuqiime[,-ncol(otuqiime)]))
mat2 &lt;- as.matrix(mat)
dim(mat2)
</code></pre>

<hr>
<h2 id='contrib'>
Contribution of rare/abundant biosphere to the total Bray-Curtis dissimilarity
</h2><span id='topic+contrib'></span>

<h3>Description</h3>

<p>function to calculate the contribution (in fraction) of the abundant or rare biosphere to the Bray-Curtis dissimilarity of the whole community.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>contrib(otutab, siteInCol = TRUE, taxhead = NULL, threshold = 1, percent = FALSE,
    check = "rare", write = FALSE, plot = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="contrib_+3A_otutab">otutab</code></td>
<td>

<p>An OTU table of microbial community, which can contain taxonomy in a column or a row.
</p>
</td></tr>
<tr><td><code id="contrib_+3A_siteincol">siteInCol</code></td>
<td>

<p>Logical, if &quot;TRUE&quot;, the OTU table contains samples in columns and taxa in rows. The function will decide whether to transpose the OTU table based on this parameter.
</p>
</td></tr>
<tr><td><code id="contrib_+3A_taxhead">taxhead</code></td>
<td>

<p>Character, specify the header of taxonomy, e.g. &quot;taxonomy&quot; if there is a taxonomy column or row. It is NULL by default.
</p>
</td></tr>
<tr><td><code id="contrib_+3A_threshold">threshold</code></td>
<td>

<p>Numeric, the threshold relative abundance cutoff upon which the rare biosphere will be subset.
</p>
</td></tr>
<tr><td><code id="contrib_+3A_percent">percent</code></td>
<td>

<p>Logical, whether the input OTU table are in relative abundance.
</p>
</td></tr>
<tr><td><code id="contrib_+3A_check">check</code></td>
<td>

<p>Character, either &quot;rare&quot; or &quot;abundant&quot;, telling the function which biosphere to be check.
</p>
</td></tr>
<tr><td><code id="contrib_+3A_write">write</code></td>
<td>

<p>Logical, if TRUE, the result will be written out as &quot;txt&quot; file, default is FALSE.
</p>
</td></tr>
<tr><td><code id="contrib_+3A_plot">plot</code></td>
<td>

<p>Logical, whether the contribution result to be visualized in boxplot. By default is FALSE.
</p>
</td></tr>
<tr><td><code id="contrib_+3A_...">...</code></td>
<td>

<p>arguments to be passed to/from other methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this function, the rare biosphere is defined by the relative abundance cutoffs (argument threshold). The Bray-Curtis distance between pairwise samples was partitioned. The Bray-Curtis measure is a scaled summation of abundance differences between two communities and can thus be partitioned for a subset population from the community (Shade et al 2014, Yang et al 2017).
</p>


<h3>Value</h3>

<p>The function will return a data frame of five columns. The first two columns specify the sample names whose Bray-Curtis distance were calculated. The third and forth columns give the distances respectively based on the whole community OTU data or the subset data. The last column gives the contribution (in fraction, not percentage) of the subset data for each pair of samples.




</p>


<h3>Author(s)</h3>

<p>Sizhong Yang &lt;yanglzu@163.com&gt;
</p>


<h3>References</h3>

<p>Shade A, Jones SE, Caporaso JG, Handelsman J, Knight R, Fierer N et al (2014). Conditionally rare taxa disproportionately contribute to temporal changes in microbial diversity. Mbio 5: e01371-01314.
</p>
<p>Yang S, Winkel M, Wagner D, Liebner S (2017). Community structure of rare methanogenic archaea: insight from a single functional group. FEMS Microbiology Ecology: fix126.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otuqiime)

result &lt;- contrib(otutab = otuqiime, siteInCol = TRUE, taxhead = "taxonomy",
    threshold = 1, percent = FALSE, check = "abund", plot = FALSE)

names(result)
head(result)
</code></pre>

<hr>
<h2 id='fillTax'>
reshape the heirarchical taxonomy
</h2><span id='topic+fillTax'></span>

<h3>Description</h3>

<p>reformat the taxonomy with the last clear assignment and changing the prefix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fillTax(x, split = ';', prefix=TRUE, fillAll=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fillTax_+3A_x">x</code></td>
<td>

<p>Character, structured strings giving the heirarchical rank of taxonomy, e.g., d__Bacteria;
p__Planctomycetota;c__Planctomycetes;o__Gemmatales;f__Gemmataceae;g__;s__</p>
</td></tr>
<tr><td><code id="fillTax_+3A_split">split</code></td>
<td>

<p>Character, the seperator for the heirarchical taxonomy.
</p>
</td></tr>
<tr><td><code id="fillTax_+3A_prefix">prefix</code></td>
<td>

<p>Logical, whether contains prefix in the taxonomic strings, default TRUE.
</p>
</td></tr>
<tr><td><code id="fillTax_+3A_fillall">fillAll</code></td>
<td>

<p>Logical, whether to fill all taxonic level, default TRUE.
</p>
</td></tr></table>


<h3>See Also</h3>

<p>fillTax2, slimTax
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  test = 'd__Bacteria;p__Planctomycetota;c__Planctomycetes;o__Gemmatales;f__Gemmataceae;g__;s__'
  fillTax(x = test, split = ';', prefix=TRUE, fillAll=TRUE)
</code></pre>

<hr>
<h2 id='fillTax2'>
reshape the heirarchical taxonomy
</h2><span id='topic+fillTax2'></span>

<h3>Description</h3>

<p>this function is similar to fillTax, except that this function has no parameter of &quot;fillAll&quot; 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fillTax2(x, split = ';', prefix=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fillTax2_+3A_x">x</code></td>
<td>

<p>Character, structured strings giving the heirarchical rank of taxonomy. Please convert to character object for your input taxonomic information before using.
</p>
</td></tr>
<tr><td><code id="fillTax2_+3A_split">split</code></td>
<td>

<p>Character, the seperator for the heirarchical taxonomy.
</p>
</td></tr>
<tr><td><code id="fillTax2_+3A_prefix">prefix</code></td>
<td>

<p>Logical, whether contains prefix in the taxonomic strings, default TRUE.
</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>fillTax, slimTax
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  test = 'd__Bacteria;p__Planctomycetota;c__Planctomycetes;o__Gemmatales;f__Gemmataceae;g__;s__'
  fillTax2(x = test, split = ';', prefix=TRUE)
</code></pre>

<hr>
<h2 id='matrixConvert'>
Convert lower triangular distance matrix into data frame
</h2><span id='topic+matrixConvert'></span>

<h3>Description</h3>

<p>This function will convert lower triangular distance matrix into a 3-column, long-format data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrixConvert(triMatrix, colname = c("sp1", "sp2", "dist"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matrixConvert_+3A_trimatrix">triMatrix</code></td>
<td>

<p>Matrix, the input matrix should be lower triangular matrix.
</p>
</td></tr>
<tr><td><code id="matrixConvert_+3A_colname">colname</code></td>
<td>

<p>Character, a vector of length 3 to specify the column names of the converted data frame.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will call the &quot;melt&quot; function in the reshape2 package, and convert the pairwise values in the lower triangular distance matrix into long-format data frame.
</p>


<h3>Value</h3>

<p>The function returns long format of data frame, with 3 columns. The first two columns give the pairwise names and the third column contains values in the matrix.




</p>


<h3>Author(s)</h3>

<p>Sizhong Yang &lt;yanglzu@163.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otuqiime)
mat &lt;- calc_bc(t(otuqiime[,-ncol(otuqiime)]))
mat.m &lt;- matrixConvert(mat, colname = c("sp1", "sp2", "bray"))
</code></pre>

<hr>
<h2 id='otu4type'>
An example OTU table with samples and taxonomy in rows and otus in columns
</h2><span id='topic+otu4type'></span>

<h3>Description</h3>

<p>A data set containing bacterial counts from the North Temperate Lakes Microbial Observatory. Due to the last row contain taxonomy, read.table function with default setting will treat the type of each column as factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("otu4type")</code></pre>


<h3>Format</h3>

<p>A data frame with 591 columns (OTUs) and 454 rows (453 samples plus 1 taxonomy).
</p>


<h3>Details</h3>

<p>This data show examples of &quot;SiteInRow&quot; if there is a taxonomy rows. When read in, the column with numeric counts will be marked as &quot;factor&quot;. This data set could be transposed to correct type with &quot;typeConvert&quot; function.
</p>


<h3>Source</h3>

<p>see the entire dataset at https://github.com/cran/OTUtable/tree/master/data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otu4type)
sapply(otu4type, class)
new &lt;- typeConvert(otu4type)
sapply(new, class)
</code></pre>

<hr>
<h2 id='otuCollap'>
collapse a OTU table at given level.
</h2><span id='topic+otuCollap'></span>

<h3>Description</h3>

<p>The function will collapse a structured OTU table at given taxonomic level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>otuCollap(otutab, taxto, siteInCol = TRUE, taxhead = "taxonomy",
    pattern = ";", collap = ";")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="otuCollap_+3A_otutab">otutab</code></td>
<td>

<p>A OTU table of microbial community, which must contain a taxonomic column (if siteInCol) or row (if site in rows). The otu table can be given in numeric counts or in relative abundance.
</p>
</td></tr>
<tr><td><code id="otuCollap_+3A_taxto">taxto</code></td>
<td>

<p>numeric, collapse the otutable at the taxonomic level which start position (index) from the left  of taxonomy. For example, in &quot;Archaea; Euryarchaeota; Methanomicrobia; Methanomicrobiales; Methanoregulaceae&quot;, the indices of &quot;Euryarchaeota&quot; and &quot;Methanomicrobiales&quot; are 2 and 4, respectively.
</p>
</td></tr>
<tr><td><code id="otuCollap_+3A_siteincol">siteInCol</code></td>
<td>

<p>Logical, if &quot;TRUE&quot;, the OTU table contains samples in columns and taxa in rows. The function will decide whether to transpose the otu table based on this parameter.
</p>
</td></tr>
<tr><td><code id="otuCollap_+3A_taxhead">taxhead</code></td>
<td>

<p>Character, specify the header of taxonomy. By default we assume your taxonomic column is entitled &quot;taxonomy&quot;.
</p>
</td></tr>
<tr><td><code id="otuCollap_+3A_pattern">pattern</code></td>
<td>

<p>Character, specify the separation of taxonomy. By default, the taxonomy is separated by semicolon (&quot;;&quot;).
</p>
</td></tr>
<tr><td><code id="otuCollap_+3A_collap">collap</code></td>
<td>

<p>Character, tell the function about the separation for the hierarchical order in the output.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will directly collapse the otu table according to numeric position of structured taxonony. This function can also collapse data with structured format like the example OTU table.
</p>


<h3>Value</h3>

<p>The function will return a collapsed OTU table.


</p>


<h3>Author(s)</h3>

<p>Sizhong Yang &lt;yanglzu@163.com&gt;
</p>


<h3>See Also</h3>

<p>otuReport
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(otuqiime)
dim(otuqiime)

result &lt;- otuCollap(otutab = otuqiime, taxto = 2, siteInCol = TRUE,
    taxhead = "taxonomy", pattern = ";")

dim(result)

</code></pre>

<hr>
<h2 id='otumothur'>
OTU table generated from 8 lakes over 4 years
</h2><span id='topic+otumothur'></span>

<h3>Description</h3>

<p>A data set containing bacterial counts from the North Temperate Lakes Microbial Observatory, the taxonomy column is in mothur format. The data set is published in Msphere (Linz et al, 2017). The data set is identical to otuqiime except for the taxonomy column in qiime format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("otumothur")</code></pre>


<h3>Format</h3>

<p>A data frame with 454 columns (453 samples plus 1 taxonomy) and 591 rows (OTUs). The taxonomy is given in mothur format, with hierarchical taxonomy from kingdom to species separated with semiclone.




</p>


<h3>Details</h3>

<p>The full version of the data set is published in Msphere (Linz et al 2017). The first two letters of sample names denote the sampling site (e.g. &quot;CB&quot;), followed with epilimnion or hypolimnion (&quot;E&quot; or &quot;H&quot;) and sampling date (&quot;01OCT07&quot;). The original data set have replicates (&quot;R1&quot; and &quot;R2&quot;), this data set only extracted the subset of &quot;R2&quot; (with extension &quot;.R2&quot; in sample names). The data set is identical to otuqiime except that the taxonomy is in format generated by software 'mothur' (Schloss et al 2009).
</p>


<h3>Source</h3>

<p>see the entire dataset at https://github.com/cran/OTUtable/tree/master/data
</p>


<h3>References</h3>

<p>Linz AM, Crary BC, Shade A, Owens S, Gilbert JA, Knight R et al (2017). Bacterial community composition and dynamics spanning five years in freshwater bog lakes. Msphere 2: e00169-00117.
Schloss PD, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB et al (2009). Introducing mothur: open-source, platform-independent, community-supported software for describing and comparing microbial communities. Applied and Environmental Microbiology 75: 7537-7541.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otumothur)
dim(otumothur)
sapply(otumothur, class)
head(otumothur$taxonomy)
</code></pre>

<hr>
<h2 id='otuqiime'>
OTU table generated from 8 lakes over 4 years
</h2><span id='topic+otuqiime'></span>

<h3>Description</h3>

<p>A data set containing bacterial counts from the North Temperate Lakes Microbial Observatory, the taxonomy column is in format generated by software platfor 'QIIME' (Caporaso et al 2010). The data set is published in Msphere (Linz et al 2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("otuqiime")</code></pre>


<h3>Format</h3>

<p>A data frame with 454 columns (453 samples plus 1 taxonomy) and 591 rows (OTUs).




</p>


<h3>Details</h3>

<p>The first two letters of sample names denote the sampling site (e.g. &quot;CB&quot;), followed with epilimnion or hypolimnion (&quot;E&quot; or &quot;H&quot;) and sampling date (&quot;01OCT07&quot;). The original data set have replicates (&quot;R1&quot; and &quot;R2&quot;), this data set only extracted the subset of &quot;R2&quot; (with extension &quot;.R2&quot; in sample names).
</p>


<h3>Source</h3>

<p>see https://github.com/cran/OTUtable/tree/master/data
</p>


<h3>References</h3>

<p>Caporaso JG, Kuczynski J, Stombaugh J, Bittinger K, Bushman FD, Costello EK et al (2010). QIIME allows analysis of high-throughput community sequencing data. Nature Methods 7: 335-336.
Linz AM, Crary BC, Shade A, Owens S, Gilbert JA, Knight R et al (2017). Bacterial community composition and dynamics spanning five years in freshwater bog lakes. Msphere 2: e00169-00117.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otuqiime)
dim(otuqiime)
names(otuqiime)[1:10]
rownames(otuqiime)[1:10]
head(otuqiime$taxonomy)
</code></pre>

<hr>
<h2 id='otuReport'>
Summarize the community structure and abundance with OTU table
</h2><span id='topic+otuReport'></span>

<h3>Description</h3>

<p>The function will summarize the frequency, abundance at given taxonomic level for the input OTU table. This function could quickly give the summary information for user when adding these values in describing the community structure in a paper.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>otuReport(otutab, siteInCol = TRUE, taxhead = "taxonomy", platform = "mothur",
    pattern = ";", prefix = TRUE, percent = FALSE, taxlevel = "phylum",
    collap = ";")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="otuReport_+3A_otutab">otutab</code></td>
<td>

<p>A OTU table of microbial community, which must contain a taxonomic column (if siteInCol) or row (if site in rows). The otu table can be given in numeric counts or in relative abundance.
</p>
</td></tr>
<tr><td><code id="otuReport_+3A_siteincol">siteInCol</code></td>
<td>

<p>Logical, if &quot;TRUE&quot;, the OTU table contains samples in columns and taxa in rows. The function will decide whether to transpose the otu table based on this parameter.
</p>
</td></tr>
<tr><td><code id="otuReport_+3A_taxhead">taxhead</code></td>
<td>

<p>Character, specify the header of taxonomy. By default we assume your taxonomic column is entitled &quot;taxonomy&quot;.
</p>
</td></tr>
<tr><td><code id="otuReport_+3A_platform">platform</code></td>
<td>

<p>Character, argument to specify the platform generating the otu table. Currently, the function support otu table generate by &quot;mothur&quot; or &quot;qiime&quot;.
</p>
</td></tr>
<tr><td><code id="otuReport_+3A_pattern">pattern</code></td>
<td>

<p>Character, specify the separation of taxonomy. By default, the taxonomy is separated by semicolon (&quot;;&quot;).
</p>
</td></tr>
<tr><td><code id="otuReport_+3A_prefix">prefix</code></td>
<td>

<p>Logical, tell the function whether the output will include the prefix like &quot;p__&quot;, &quot;c__&quot; for the corresponding taxonomic levels.
</p>
</td></tr>
<tr><td><code id="otuReport_+3A_percent">percent</code></td>
<td>

<p>Logical, whether the input otu table are given in relative abundance. FALSE means that the input otu table is in numeric counts.
</p>
</td></tr>
<tr><td><code id="otuReport_+3A_taxlevel">taxlevel</code></td>
<td>

<p>Character, specify the taxonomic level at which you want to know for the otu table. The valid choice are c(&quot;kingdom&quot;, &quot;phylum&quot;, &quot;class&quot;, &quot;order&quot;, &quot;family&quot;, &quot;genus&quot;, &quot;species&quot;).
</p>
</td></tr>
<tr><td><code id="otuReport_+3A_collap">collap</code></td>
<td>

<p>Character, tell the function about the separation for the hierarchical order in the output.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function was designed according to the structured taxonomy generated by mothur or qiime. So far, the function support the 7 levels of hierarchical taxonomy from kingdom to species.
</p>


<h3>Value</h3>

<p>If the input otu table is in counts, the function will return a list of results summarizing 9 different aspects for the microbial community as follows:

</p>
<table>
<tr><td><code>whatTaxa</code></td>
<td>
<p>The &quot;whatTaxa&quot; will give which lineages are present at given taxonomic level in the community.</p>
</td></tr>
<tr><td><code>taxaFreqs</code></td>
<td>
<p>The element of &quot;taxaFreqs&quot; in the list is the frequency table of each lineage when community table were collapsed at a given taxonomic level.</p>
</td></tr>
<tr><td><code>taxaFrac</code></td>
<td>
<p>The &quot;taxaFrac&quot; element summarizes the fraction of each lineage among the total lineages at a given taxonomic level.</p>
</td></tr>
<tr><td><code>reads</code></td>
<td>
<p>The &quot;reads&quot; table is the otu table in absolute counts which has been collapsed at a given taxonomic level.</p>
</td></tr>
<tr><td><code>readSum</code></td>
<td>
<p>The &quot;readSum&quot; gives the total amount of reads in each sample after the community was collapsed at given taxonomic level.</p>
</td></tr>
<tr><td><code>readFrac</code></td>
<td>
<p>The &quot;readFrac&quot; table give the fraction of reads in relation to the total counts of the whole community at given taxonomic level.</p>
</td></tr>
<tr><td><code>readFracSum</code></td>
<td>
<p>The &quot;readFracSum&quot; table give sum of reads fraction by different lineages at given taxonomic level.</p>
</td></tr>
<tr><td><code>Relabund</code></td>
<td>
<p>The &quot;Relabund&quot; is the relative abundance table in percentage at given taxonomic level. If the input otu table is in absolute counts, the reads will be normalized by the total amount of reads of each sample, not the total amount of the whole community.</p>
</td></tr>
<tr><td><code>RelabundMean</code></td>
<td>
<p>The data &quot;RelabundMean&quot; in the list return the mean relative abundance of each lineage at given taxonomic level across all samples.</p>
</td></tr>
</table>
<p>For the relative abundance otu input, the function will omit four summary table regarding absolute reads (&quot;reads&quot;, &quot;readSum&quot;, &quot;readFrac&quot; and &quot;readFracSum&quot;).

</p>


<h3>Author(s)</h3>

<p>Sizhong Yang &lt;yanglzu@163.com&gt;
</p>


<h3>See Also</h3>

<p>subOTU, otuCollap
</p>


<h3>Examples</h3>

<pre><code class='language-R'># summary the otu table in qiime format

data(otuqiime)

summaryInfo &lt;- otuReport(otutab = otuqiime, siteInCol = TRUE, taxhead = "taxonomy",
    platform = "qiime", pattern = ";", prefix = TRUE, percent = FALSE, taxlevel = "class")

length(summaryInfo)
names(summaryInfo)
summaryInfo[[1]]

# summary otu table in mothur format

data(otumothur)

summaryInfo &lt;- otuReport(otutab = otumothur, siteInCol = TRUE, taxhead = "taxonomy",
    platform = "mothur", pattern = ";", percent = FALSE, taxlevel = "phylum", collap = ";")

length(summaryInfo)
names(summaryInfo)
summaryInfo[[1]]

op &lt;- par(mar = c(8,6,2,1)+0.1)
bplot(summaryInfo[["taxaFreqs"]])
par(op)

# summary otu table of relative abundance

per &lt;- subOTU(otutab = otuqiime, siteInCol = TRUE, taxhead = "taxonomy",
    percent = FALSE, choose = "all", outype = "Relabund", sort = TRUE)

summaryInfo &lt;- otuReport(otutab = per, siteInCol = TRUE, taxhead = "taxonomy",
    platform = "qiime", pattern = ";", percent = TRUE, taxlevel = "class")

length(summaryInfo)
names(summaryInfo)
</code></pre>

<hr>
<h2 id='pbray'>
Partition the Bray-Curtis distance dissimilarity
</h2><span id='topic+pbray'></span>

<h3>Description</h3>

<p>The pbray function partitions the Bray-Curtis distance matrix based on the who community and the subset of the community data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbray(allComm, subComm, tolower = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pbray_+3A_allcomm">allComm</code></td>
<td>

<p>A otu table of microbial community which contains sample in rows and taxa in column. The otu table should not contain a taxonomic column.
</p>
</td></tr>
<tr><td><code id="pbray_+3A_subcomm">subComm</code></td>
<td>

<p>A subset of the otu table, with the same samples as allComm.
</p>
</td></tr>
<tr><td><code id="pbray_+3A_tolower">tolower</code></td>
<td>

<p>Logical, by default the function returns the lower triangular matrix.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Bray-Curtis dissimilarity is a scaled summation of abundance differences between two communities, it is thus could be partitioned between two samples attributable to a subset of the community (Shade et al., 2014; Yang et al., 2017).
Note, the pbray function requires the input for &quot;allComm&quot; and &quot;subComm&quot; keep consistent either in either counts or relative abundance.
If the input for &quot;allComm&quot; and &quot;subComm&quot; are the same data, the function will return the Bray-Curtis matrix for the whole community.
</p>


<h3>Value</h3>

<p>The function returns a distance matrix by using the subset community against the whole community data.




</p>


<h3>Author(s)</h3>

<p>Sizhong Yang &lt;yanglzu@163.com&gt;
</p>


<h3>References</h3>

<p>Shade A, Jones SE, Caporaso JG, Handelsman J, Knight R, Fierer N, and Gilbert JA. Conditionally rare taxa disproportionately contribute to temporal changes in microbial diversity. Mbio, 2014, 5(4): e01371-01314.
</p>
<p>Yang S, Winkel M, Wagner D, and Liebner S. Community structure of rare methanogenic archaea: insight from a single functional group. FEMS Microbiology Ecology, 2017: fix126.
</p>


<h3>See Also</h3>

<p>function contrib(), matrixConvert().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otumothur)

subotus &lt;- subOTU(otutab = otumothur, taxhead = "taxonomy", siteInCol = TRUE,
    percent = FALSE, choose = "rare", outype = "counts", sort = FALSE)
pRare &lt;- pbray(allComm = t(otumothur[,-454]), subComm = t(subotus[,-454]))
class(pRare)

# convert to long format data frame

longdist &lt;- matrixConvert(pRare, colname = c("sp1", "sp2", "bray"))
</code></pre>

<hr>
<h2 id='rareBiosphere'>
Summarize different subgroups in the rare biosphere
</h2><span id='topic+rareBiosphere'></span>

<h3>Description</h3>

<p>The function discriminates different fractions of rare biosphere from the input community data, based on the ratio between the maximum and minimum abundance (for details, please see Yang et al. 2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rareBiosphere(otutab, siteInCol = TRUE, taxhead= NULL, percent = FALSE,
    threshold = 1, cutRatio = 100, cutPERare = 5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rareBiosphere_+3A_otutab">otutab</code></td>
<td>

<p>An OTU table of microbial community, which can contain a taxonomic column (if siteInCol) or row (if site in rows). Note this function requires the otu table must be given in absolute counts, not in relative abundance.
</p>
</td></tr>
<tr><td><code id="rareBiosphere_+3A_siteincol">siteInCol</code></td>
<td>

<p>Logical, by default it is &quot;TRUE&quot;, meaning the OTU table contains samples in columns and taxa in rows. Otherwise, the otu table will be transposed.
</p>
</td></tr>
<tr><td><code id="rareBiosphere_+3A_taxhead">taxhead</code></td>
<td>

<p>Character, specify the header of taxonomy. By default the taxonomic column is NULL.
</p>
</td></tr>
<tr><td><code id="rareBiosphere_+3A_percent">percent</code></td>
<td>

<p>Logical, whether the input otu table is in relative abundance. The default is FALSE.
</p>
</td></tr>
<tr><td><code id="rareBiosphere_+3A_threshold">threshold</code></td>
<td>

<p>Numeric, the threshold specify the relative abundance cutoff upon which the rare biosphere is subset.
</p>
</td></tr>
<tr><td><code id="rareBiosphere_+3A_cutratio">cutRatio</code></td>
<td>

<p>Numeric, the cutRatio parameter is the ratio between the maximum and minimum non-zero absolute abundance, which specify the threshold of conditionally rare taxa (CRT).
</p>
</td></tr>
<tr><td><code id="rareBiosphere_+3A_cutperare">cutPERare</code></td>
<td>

<p>Numeric, the argument cutPERare specify the threshold of permanently rare taxa (PERare) in the community according to the ratio as mentioned above.
</p>
</td></tr>
<tr><td><code id="rareBiosphere_+3A_...">...</code></td>
<td>

<p>arguments to be passed to/from other methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The rare biopshere constitutes different fractions of rarity. The conditionally rare taxa (CRT) are generally rare across samples but can become abundant in some samples. In contrast, the permanently rare taxa (PERare) are persistently low in abundance. Different levels of rarity may have practical implications and suggest potential roles in metabolism and ecological functions (Lynch and Neufeld, 2015; Yang et al., 2017).
</p>
<p>This function will filter the rare biophere from the input OTU table and then discriminate different rare fractions based on the ratio between the maximum and minimum non-zero absolute abundance, which has been as an technical alternative of skewness (please see Yang et al. 2017 for details).
</p>
<p>The function only works with otu table with absolute abundance, for the absolute reads enable to identify of absolute singletons and doubletons.
</p>


<h3>Value</h3>

<p>The function returns a list of 4 data frames.

</p>
<table>
<tr><td><code>summaryTable</code></td>
<td>
<p>The element &quot;summaryTable&quot; returns a table containing the maximum and the minimum relative abundance, the ratio of the two abundance, the grouping of rarity, whether the taxa is singleton or doubletons, with additional column of taxonomy if it is included in the input otu table.</p>
</td></tr>
<tr><td><code>CRT</code></td>
<td>
<p>Table &quot;CRT&quot; gives the subset of the above &quot;summaryTable&quot; only for the conditionally rare taxa.</p>
</td></tr>
<tr><td><code>PERare</code></td>
<td>
<p>Table &quot;PERare&quot; shows the information of permanently rare taxa.</p>
</td></tr>
<tr><td><code>otherRare</code></td>
<td>
<p>Table &quot;otherRare&quot; summarizes the rare taxa outside the &quot;CRT&quot; and &quot;PERare&quot;.</p>
</td></tr>

</table>


<h3>Author(s)</h3>

<p>Sizhong Yang &lt;yanglzu@163.com&gt;
</p>


<h3>References</h3>

<p>Lynch MDJ, Neufeld JD (2015). Ecology and exploration of the rare biosphere. Nature Reviews Microbiology 13: 217-229.
</p>
<p>Yang S, Winkel M, Wagner D, Liebner S (2017). Community structure of rare methanogenic archaea: insight from a single functional group. FEMS Microbiology Ecology: fix126.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otumothur)

example &lt;- rareBiosphere(otutab = otumothur, siteInCol = TRUE, taxhead = "taxonomy",
    percent = FALSE, threshold = 1, cutRatio = 100, cutPERare = 5)
length(example)
names(example)
head(example[["summaryTable"]])
head(example[["CRT"]])

example2 &lt;- rareBiosphere(otutab = otumothur[,-454], siteInCol = TRUE,
    taxhead = NULL, percent=FALSE, threshold = 1, cutRatio = 100, cutPERare = 5)
length(example2)
names(example2)
</code></pre>

<hr>
<h2 id='slimTax'>
reshape heirarchical taxonomy
</h2><span id='topic+slimTax'></span>

<h3>Description</h3>

<p>The function will trunck the heirarchical taxonomy by &quot;from&quot; and &quot;to&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slimTax(x, from, to, separator =';', jump=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="slimTax_+3A_x">x</code></td>
<td>

<p>Character, structured strings to give the heirarchical taxonomy. If x is not character, please use as.character() function to convert first.
</p>
</td></tr>
<tr><td><code id="slimTax_+3A_from">from</code></td>
<td>

<p>numeric, valid values within 1,2,...7, which represent the tax rank of domain, phylum, class, order, family, genus and species, respectively.
</p>
</td></tr>
<tr><td><code id="slimTax_+3A_to">to</code></td>
<td>

<p>numeric, similar to parameter &quot;from&quot;. Please note that value of &quot;to&quot; must not be smaller than &quot;from&quot;.
</p>
</td></tr>
<tr><td><code id="slimTax_+3A_separator">separator</code></td>
<td>

<p>Character, for example semicolon &quot;;&quot; which states the seperator of taxonomic heirarchy.
</p>
</td></tr>
<tr><td><code id="slimTax_+3A_jump">jump</code></td>
<td>

<p>Logic, Setting &quot;jump&quot; to TRUE means only take the tax at indices &quot;from&quot; and &quot;to&quot;, excluding those in between.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will reformat the taxonomy by the taxonomic ranks specified by &quot;from&quot; and &quot;to&quot;.
</p>


<h3>Value</h3>

<p>The function will return the reshaped taxonomy.


</p>


<h3>Author(s)</h3>

<p>Sizhong Yang &lt;yanglzu@163.com&gt;
</p>


<h3>See Also</h3>

<p>fillTax
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
test = 'd__Bacteria;p__Planctomycetota;c__Planctomycetes;o__Gemmatales;f__Gemmataceae;g__;s__'
slimTax(test, from=2, to=5)

data(otuqiime)
dim(otuqiime)

result &lt;- slimTax(x = as.character(otuqiime$taxonomy), from=2, to=5, separator =';', jump=FALSE)

dim(result)

</code></pre>

<hr>
<h2 id='subOTU'>
Subset a OTU table
</h2><span id='topic+subOTU'></span>

<h3>Description</h3>

<p>This function subset a OTU table according to the specified threshold of mean relative abundance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subOTU(otutab, siteInCol = TRUE, taxhead = NULL, percent = TRUE, choose = "rare",
       threshold = 1, outype = "Relabund", sort = TRUE, write = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subOTU_+3A_otutab">otutab</code></td>
<td>

<p>An OTU table of microbial community, which can contain a taxonomic column (if siteInCol) or row (if site in rows). The OTU table can be given in numeric counts or in relative abundance.
</p>
</td></tr>
<tr><td><code id="subOTU_+3A_siteincol">siteInCol</code></td>
<td>

<p>Logical, if &quot;TRUE&quot;, the OTU table contains samples in columns and taxa in rows. The function will decide whether to transpose the OTU table based on this parameter.
</p>
</td></tr>
<tr><td><code id="subOTU_+3A_taxhead">taxhead</code></td>
<td>

<p>Character, specify the header of taxonomy. By default the taxonomic column is NULL.
</p>
</td></tr>
<tr><td><code id="subOTU_+3A_percent">percent</code></td>
<td>

<p>Logical, whether the input OTU table is in relative abundance. The default is TRUE.
</p>
</td></tr>
<tr><td><code id="subOTU_+3A_choose">choose</code></td>
<td>

<p>Character, tell the function which part to subset out. The valid choice are &quot;rare&quot;,&quot;abundant&quot; and &quot;all&quot;, which will specify the rare biosphere, the abundant population and keep the whole community data.
</p>
</td></tr>
<tr><td><code id="subOTU_+3A_threshold">threshold</code></td>
<td>

<p>Numeric, the threshold specify the relative abundance cutoff upon which the rare biosphere is subset.
</p>
</td></tr>
<tr><td><code id="subOTU_+3A_outype">outype</code></td>
<td>

<p>Character, specify whether the output otu subset should in relative abundance (outype=&quot;Relabund&quot; or &quot;relabund&quot;) or in absolute counts (outype=&quot;counts&quot; or &quot;Counts&quot;). This function also support partial match of the parameter.
</p>
</td></tr>
<tr><td><code id="subOTU_+3A_sort">sort</code></td>
<td>

<p>Logical, be default, the output OTU subset is sorted according to the descending order of mean relative abundance across samples.
</p>
</td></tr>
<tr><td><code id="subOTU_+3A_write">write</code></td>
<td>

<p>Logical, whether the output OTU table will be written out when running this function. The default is FALSE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function will subset the OTU table to abundant (choose = &quot;abundant&quot;) or rare biosphere (choose = &quot;rare&quot;) according to the given relative abundance threshold of rare biosphere (Lynch and Neufeld, 2015). It will also keep the whole community without subsetting, if choose = &quot;all&quot;. The output could be relative abundance (outype = &quot;Relabund&quot;) or absolute counts (outype = &quot;counts&quot;). If sort is TRUE, the output result will be sorted by the descending order of mean relative abundance across samples.
</p>


<h3>Value</h3>

<p>This function will return an OTU table (data frame) according to the specified arguments.
</p>


<h3>Author(s)</h3>

<p>Sizhong Yang &lt;yanglzu@163.com&gt;
</p>


<h3>References</h3>

<p>Lynch MDJ, Neufeld JD (2015). Ecology and exploration of the rare biosphere. Nature Reviews Microbiology 13: 217-229.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otuqiime)

example1 &lt;- subOTU(otutab = otuqiime, siteInCol = TRUE, taxhead ="taxonomy",
    percent = FALSE, choose = "abundant", threshold = 1, outype = "Relabund")
dim(example1)

example2 &lt;- subOTU(otutab = otuqiime[,-454], siteInCol = TRUE, taxhead = NULL,
    percent = FALSE, choose = "rare", threshold = 1, outype = "counts")
dim(example2)
</code></pre>

<hr>
<h2 id='typeConvert'>
Transpose the data frame if there is data type conversion.
</h2><span id='topic+typeConvert'></span>

<h3>Description</h3>

<p>function to transpose the data frame if there is data type conversion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>typeConvert(otutab,taxhead = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="typeConvert_+3A_otutab">otutab</code></td>
<td>

<p>An OTU table, which can contain no taxonomy. The OTU table can be given in numeric counts or in relative abundance.
</p>
</td></tr>
<tr><td><code id="typeConvert_+3A_taxhead">taxhead</code></td>
<td>

<p>Character, specify the header of taxonomy. By default the taxonomic column is NULL.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is to convert the numeric values into right type so that the downstream numeric calculation could be processed without type error.
</p>


<h3>Value</h3>

<p>This function returns a transposed OTU table. In the source OTU table the numeric values showing &quot;character&quot; or &quot;factor&quot; will be converted to right types.




</p>


<h3>Author(s)</h3>

<p>Sizhong Yang &lt;yanglzu@163.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otu4type)
sapply(otu4type, class)
new &lt;- typeConvert(as.data.frame(t(otu4type)), taxhead = "taxonomy")
sapply(new, class)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
