<!DOCTYPE html><html><head><title>Help for package PAC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PAC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aggregateData'><p>Aggregates results from the clustering and merging step.</p></a></li>
<li><a href='#annotateClades'><p>Creates annotation matrix for the clades in aggregated format. The matrix contains average signals of each dimension for each clade in each sample</p></a></li>
<li><a href='#annotationMatrix_withSubpopProp'><p>Adds subpopulation proportion for the annotation matrix for the clades</p></a></li>
<li><a href='#BSPLeaveCenter'><p>Finds N Leaf centers in the data</p></a></li>
<li><a href='#constellationPlot'><p>Makes constellation plot, in which the centroids are clusters are embedded in the t-SNE 2D plane and the cross-sample relationships are plotted as lines connecting related sample clusters (clades).</p></a></li>
<li><a href='#fmeasure'><p>F-measure Calculation</p></a></li>
<li><a href='#getAverageSpreadOf2SubpopClades'><p>Calculate the (global) average spread of subpopulations in clades with 2 subpopulations on the constellation plot.</p></a></li>
<li><a href='#getExtraneousCladeSubpopulations'><p>Calculates subpopulations in clades (with two or more subpopulations) that are too far away from other subpopulations (within the same clade) on the constellation plot; these far away subpopulations should be pruned away from the original clades.</p></a></li>
<li><a href='#getRepresentativeNetworks'><p>Representative Networks</p></a></li>
<li><a href='#heatmapInput'><p>Creates the matrix that can be easily plotted with a heatmap function available in an R package</p></a></li>
<li><a href='#JaccardSM'><p>Calculates the Jaccard similarity matrix.</p></a></li>
<li><a href='#MAN'><p>Creates network alignments using network constructed from subpopulations after PAC</p></a></li>
<li><a href='#MINetwork_matrix_topEdges'><p>Mutual information network connection matrix generation (mrnet algorithm) using the parmigene package. Mutual information calculated with infotheo package.</p></a></li>
<li><a href='#MINetwork_simplified_topEdges'><p>Outputs the vectorized summary of a network based on the number of edges connected to a node</p></a></li>
<li><a href='#MINetworkPlot_topEdges'><p>Plots mutual information network (mrnet algorithm) connection using the parmigene package. Mutual information calculated with infotheo package.</p></a></li>
<li><a href='#outputNetworks_topEdges_matrix'><p>Wrapper to output the mutual information networks for subpopulations with size larger than a desired threshold.</p></a></li>
<li><a href='#outputRepresentativeNetworks_topEdges'><p>Outputs the representative/clade networks (plots and summary vectors) for subpopulations with size larger than a desired threshold. Saves the networks and the data matrices without the smaller subpopulations.</p></a></li>
<li><a href='#PAC'><p>Partition Assisted Clustering</p>
PAC 1) utilizes dsp or bsp-ll to recursively partition the data space and 2) applies a short round of kmeans style postprocessing to efficiently output clustered labels of data points.</a></li>
<li><a href='#recordWithinClusterSpread'><p>Calculates the within cluster spread</p></a></li>
<li><a href='#refineSubpopulationLabels'><p>Refines the subpopulation labels from PAC using network alignment and small subpopulation information. Outputs a new set of files containing the representative labels.</p></a></li>
<li><a href='#renamePrunedSubpopulations'><p>Prune away specified subpopulations in clades that are far away.</p></a></li>
<li><a href='#runElbowPointAnalysis'><p>Runs elbow point analysis to find the practical optimal number of clades to output. Outputs the average within sample cluster spread for all samples and the elbow point analysis plot with loess line fitted through the results.</p></a></li>
<li><a href='#samplePass'><p>Run PAC for Specified Samples</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Partition-Assisted Clustering and Multiple Alignments of
Networks</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-02-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Ye Henry Li, Dangna Li</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ye Henry Li &lt;hlowl2@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements partition-assisted clustering and multiple alignments of networks. It 1) utilizes partition-assisted clustering to find robust and accurate clusters and 2) discovers coherent relationships of clusters across multiple samples. It is particularly useful for analyzing single-cell data set. Please see Li et al. (2017) &lt;<a href="https://doi.org/10.1371%2Fjournal.pcbi.1005875">doi:10.1371/journal.pcbi.1005875</a>&gt; for detail method description.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://doi.org/10.1371/journal.pcbi.1005875">https://doi.org/10.1371/journal.pcbi.1005875</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.2),igraph,parmigene,infotheo,dplyr, Rtsne,
ggplot2, ggrepel</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++11</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-02-18 06:20:59 UTC; henryli</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-02-18 07:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='aggregateData'>Aggregates results from the clustering and merging step.</h2><span id='topic+aggregateData'></span>

<h3>Description</h3>

<p>Aggregates results from the clustering and merging step.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aggregateData(dataInput, labelsInput)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aggregateData_+3A_datainput">dataInput</code></td>
<td>
<p>Data matrix, with first column being SampleID.</p>
</td></tr>
<tr><td><code id="aggregateData_+3A_labelsinput">labelsInput</code></td>
<td>
<p>cluster labels from PAC.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The aggregated data of <code>dataInput</code>, with average signal levels for all clusters and sample combinations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n = 5e3                       # number of observations
p = 1                         # number of dimensions
K = 3                         # number of clusters
w = rep(1,K)/K                # component weights
mu &lt;- c(0,2,4)                # component means
sd &lt;- rep(1,K)/K              # component standard deviations
g &lt;- sample(1:K,prob=w,size=n,replace=TRUE)   # ground truth for clustering
X &lt;- as.matrix(rnorm(n=n,mean=mu[g],sd=sd[g]))
y &lt;- PAC(X, K)
X2&lt;-as.matrix(rnorm(n=n,mean=mu[g],sd=sd[g]))
y2&lt;-PAC(X2,K)
X&lt;-cbind("Sample1", as.data.frame(X)); colnames(X)&lt;-c("SampleID", "Value")
X2&lt;-cbind("Sample2", as.data.frame(X2)); colnames(X2)&lt;-c("SampleID", "Value")
aggregateData(rbind(X,X2),c(y,y2))
</code></pre>

<hr>
<h2 id='annotateClades'>Creates annotation matrix for the clades in aggregated format. The matrix contains average signals of each dimension for each clade in each sample</h2><span id='topic+annotateClades'></span>

<h3>Description</h3>

<p>Creates annotation matrix for the clades in aggregated format. The matrix contains average signals of each dimension for each clade in each sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotateClades(sampleIDs, topHubs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotateClades_+3A_sampleids">sampleIDs</code></td>
<td>
<p>sampleID vector</p>
</td></tr>
<tr><td><code id="annotateClades_+3A_tophubs">topHubs</code></td>
<td>
<p>number of top ranked genes to output for annotation; annotation is a concatenated list of top ranked genes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Annotated clade matrix
</p>

<hr>
<h2 id='annotationMatrix_withSubpopProp'>Adds subpopulation proportion for the annotation matrix for the clades</h2><span id='topic+annotationMatrix_withSubpopProp'></span>

<h3>Description</h3>

<p>Adds subpopulation proportion for the annotation matrix for the clades
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotationMatrix_withSubpopProp(aggregateMatrix_withAnnotation)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotationMatrix_withSubpopProp_+3A_aggregatematrix_withannotation">aggregateMatrix_withAnnotation</code></td>
<td>
<p>the annotated clade matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Annotated clade matrix with subpopulation proportions
</p>

<hr>
<h2 id='BSPLeaveCenter'>Finds N Leaf centers in the data</h2><span id='topic+BSPLeaveCenter'></span>

<h3>Description</h3>

<p>Finds N Leaf centers in the data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BSPLeaveCenter(data, N = 40, method = "dsp")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BSPLeaveCenter_+3A_data">data</code></td>
<td>
<p>a n x p data matrix</p>
</td></tr>
<tr><td><code id="BSPLeaveCenter_+3A_n">N</code></td>
<td>
<p>number of leaves centers</p>
</td></tr>
<tr><td><code id="BSPLeaveCenter_+3A_method">method</code></td>
<td>
<p>partition method, either &quot;dsp (discrepancy based partition)&quot;,  or &quot;ll (bayesian sequantial partition limited-look ahead)&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>leafctr  N leaves centers
</p>

<hr>
<h2 id='constellationPlot'>Makes constellation plot, in which the centroids are clusters are embedded in the t-SNE 2D plane and the cross-sample relationships are plotted as lines connecting related sample clusters (clades).</h2><span id='topic+constellationPlot'></span>

<h3>Description</h3>

<p>Makes constellation plot, in which the centroids are clusters are embedded in the t-SNE 2D plane and the cross-sample relationships are plotted as lines connecting related sample clusters (clades).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>constellationPlot(pacman_results, perplexity, max_iter, seed,
  plotTitle = "Constellations of Clades", nudge_x = 0.3, nudge_y = 0.3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="constellationPlot_+3A_pacman_results">pacman_results</code></td>
<td>
<p>PAC-MAN analysis result matrix that contains network annotation, clade IDs and mean (centroid) clade expression levels.</p>
</td></tr>
<tr><td><code id="constellationPlot_+3A_perplexity">perplexity</code></td>
<td>
<p>perplexity setting for running t-SNE</p>
</td></tr>
<tr><td><code id="constellationPlot_+3A_max_iter">max_iter</code></td>
<td>
<p>max_iter setting for running t-SNE</p>
</td></tr>
<tr><td><code id="constellationPlot_+3A_seed">seed</code></td>
<td>
<p>set seed to make t-SNE and consetllation plot to be reproducible</p>
</td></tr>
<tr><td><code id="constellationPlot_+3A_plottitle">plotTitle</code></td>
<td>
<p>max_iter setting for running t-SNE</p>
</td></tr>
<tr><td><code id="constellationPlot_+3A_nudge_x">nudge_x</code></td>
<td>
<p>nudge on x coordinate of centroid labels</p>
</td></tr>
<tr><td><code id="constellationPlot_+3A_nudge_y">nudge_y</code></td>
<td>
<p>nudge on y coordinate of centroid labels</p>
</td></tr>
</table>

<hr>
<h2 id='fmeasure'>F-measure Calculation</h2><span id='topic+fmeasure'></span>

<h3>Description</h3>

<p>Compute the F measure between the ground truth and the estimated label
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fmeasure(g, t)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fmeasure_+3A_g">g</code></td>
<td>
<p>the ground truth</p>
</td></tr>
<tr><td><code id="fmeasure_+3A_t">t</code></td>
<td>
<p>estimated labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>f  the F measure
</p>

<hr>
<h2 id='getAverageSpreadOf2SubpopClades'>Calculate the (global) average spread of subpopulations in clades with 2 subpopulations on the constellation plot.</h2><span id='topic+getAverageSpreadOf2SubpopClades'></span>

<h3>Description</h3>

<p>Calculate the (global) average spread of subpopulations in clades with 2 subpopulations on the constellation plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAverageSpreadOf2SubpopClades(tsneResults, pacman_results)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAverageSpreadOf2SubpopClades_+3A_tsneresults">tsneResults</code></td>
<td>
<p>t-SNE output of clade centroids' embedding.</p>
</td></tr>
<tr><td><code id="getAverageSpreadOf2SubpopClades_+3A_pacman_results">pacman_results</code></td>
<td>
<p>PAC-MAN analysis result matrix that contains network annotation, clade IDs and mean (centroid) clade expression levels.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns global average of 2-subpopulation clade spread on the constellation plot.
</p>

<hr>
<h2 id='getExtraneousCladeSubpopulations'>Calculates subpopulations in clades (with two or more subpopulations) that are too far away from other subpopulations (within the same clade) on the constellation plot; these far away subpopulations should be pruned away from the original clades.</h2><span id='topic+getExtraneousCladeSubpopulations'></span>

<h3>Description</h3>

<p>Calculates subpopulations in clades (with two or more subpopulations) that are too far away from other subpopulations (within the same clade) on the constellation plot; these far away subpopulations should be pruned away from the original clades.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExtraneousCladeSubpopulations(tsneResults, pacman_results,
  threshold_multiplier, max_threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getExtraneousCladeSubpopulations_+3A_tsneresults">tsneResults</code></td>
<td>
<p>t-SNE output of clade centroids' embedding.</p>
</td></tr>
<tr><td><code id="getExtraneousCladeSubpopulations_+3A_pacman_results">pacman_results</code></td>
<td>
<p>PAC-MAN analysis result matrix that contains network annotation, clade IDs and mean (centroid) clade expression levels.</p>
</td></tr>
<tr><td><code id="getExtraneousCladeSubpopulations_+3A_threshold_multiplier">threshold_multiplier</code></td>
<td>
<p>how many times the threshold ( (a) spread from center of clade for clades with three or more sample subpopulations and (b) distance from each subpopulation centroid for clades with exactly two subpopulations).</p>
</td></tr>
<tr><td><code id="getExtraneousCladeSubpopulations_+3A_max_threshold">max_threshold</code></td>
<td>
<p>the maximum distance (on t-SNE plane) allowed for sample subpopulations to be categorized into the same clade.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns clade subpopulations to be pruned.
</p>

<hr>
<h2 id='getRepresentativeNetworks'>Representative Networks</h2><span id='topic+getRepresentativeNetworks'></span>

<h3>Description</h3>

<p>Outputs representative networks for clades/subpopulations larger than a size filter (very small subpopulations are not considered in downstream analyses)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRepresentativeNetworks(sampleIDs, dim_subset, SubpopSizeFilter,
  num_networkEdge)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getRepresentativeNetworks_+3A_sampleids">sampleIDs</code></td>
<td>
<p>sampleID vector</p>
</td></tr>
<tr><td><code id="getRepresentativeNetworks_+3A_dim_subset">dim_subset</code></td>
<td>
<p>a string vector of string names to subset the data columns for PAC; set to NULL to use all columns</p>
</td></tr>
<tr><td><code id="getRepresentativeNetworks_+3A_subpopsizefilter">SubpopSizeFilter</code></td>
<td>
<p>the cutoff for small subpopulations. Smaller subpopulations have unstable covariance structure, so no network structure is calculated</p>
</td></tr>
<tr><td><code id="getRepresentativeNetworks_+3A_num_networkedge">num_networkEdge</code></td>
<td>
<p>the number of edges to draw for each subpopulation mutual information network</p>
</td></tr>
</table>

<hr>
<h2 id='heatmapInput'>Creates the matrix that can be easily plotted with a heatmap function available in an R package</h2><span id='topic+heatmapInput'></span>

<h3>Description</h3>

<p>Creates the matrix that can be easily plotted with a heatmap function available in an R package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heatmapInput(aggregateMatrix_withAnnotation)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heatmapInput_+3A_aggregatematrix_withannotation">aggregateMatrix_withAnnotation</code></td>
<td>
<p>the annotated clade matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the heatmap input matrix
</p>

<hr>
<h2 id='JaccardSM'>Calculates the Jaccard similarity matrix.</h2><span id='topic+JaccardSM'></span>

<h3>Description</h3>

<p>Calculates the Jaccard similarity matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JaccardSM(network1, network2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="JaccardSM_+3A_network1">network1</code></td>
<td>
<p>first network matrix input</p>
</td></tr>
<tr><td><code id="JaccardSM_+3A_network2">network2</code></td>
<td>
<p>second network matrix input</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the alignment/co-occurene score
</p>

<hr>
<h2 id='MAN'>Creates network alignments using network constructed from subpopulations after PAC</h2><span id='topic+MAN'></span>

<h3>Description</h3>

<p>Creates network alignments using network constructed from subpopulations after PAC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MAN(sampleIDs, num_PACSupop, smallSubpopCutoff, k_clades)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MAN_+3A_sampleids">sampleIDs</code></td>
<td>
<p>sampleID vector</p>
</td></tr>
<tr><td><code id="MAN_+3A_num_pacsupop">num_PACSupop</code></td>
<td>
<p>number of subpopulations learned in PAC step for each sample</p>
</td></tr>
<tr><td><code id="MAN_+3A_smallsubpopcutoff">smallSubpopCutoff</code></td>
<td>
<p>Population size cutoff for subpopulations in clade calculation. The small subpopulations will be considered in the refinement step.</p>
</td></tr>
<tr><td><code id="MAN_+3A_k_clades">k_clades</code></td>
<td>
<p>number of clades to output before refinement</p>
</td></tr>
</table>


<h3>Value</h3>

<p>clades_network_only the clades constructed without small subpopulations (by cutoff) using mutual information network alignments
</p>

<hr>
<h2 id='MINetwork_matrix_topEdges'>Mutual information network connection matrix generation (mrnet algorithm) using the parmigene package. Mutual information calculated with infotheo package.</h2><span id='topic+MINetwork_matrix_topEdges'></span>

<h3>Description</h3>

<p>Mutual information network connection matrix generation (mrnet algorithm) using the parmigene package. Mutual information calculated with infotheo package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MINetwork_matrix_topEdges(dataMatrix, threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MINetwork_matrix_topEdges_+3A_datamatrix">dataMatrix</code></td>
<td>
<p>data matrix</p>
</td></tr>
<tr><td><code id="MINetwork_matrix_topEdges_+3A_threshold">threshold</code></td>
<td>
<p>the number of edges to draw for each subpopulation mutual information network</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the mutual information network connection matrix with top edges
</p>

<hr>
<h2 id='MINetwork_simplified_topEdges'>Outputs the vectorized summary of a network based on the number of edges connected to a node</h2><span id='topic+MINetwork_simplified_topEdges'></span>

<h3>Description</h3>

<p>Outputs the vectorized summary of a network based on the number of edges connected to a node
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MINetwork_simplified_topEdges(dataMatrix, threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MINetwork_simplified_topEdges_+3A_datamatrix">dataMatrix</code></td>
<td>
<p>data matrix</p>
</td></tr>
<tr><td><code id="MINetwork_simplified_topEdges_+3A_threshold">threshold</code></td>
<td>
<p>the number of edges to draw for each subpopulation mutual information network</p>
</td></tr>
</table>

<hr>
<h2 id='MINetworkPlot_topEdges'>Plots mutual information network (mrnet algorithm) connection using the parmigene package. Mutual information calculated with infotheo package.</h2><span id='topic+MINetworkPlot_topEdges'></span>

<h3>Description</h3>

<p>Plots mutual information network (mrnet algorithm) connection using the parmigene package. Mutual information calculated with infotheo package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MINetworkPlot_topEdges(dataMatrix, threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MINetworkPlot_topEdges_+3A_datamatrix">dataMatrix</code></td>
<td>
<p>data matrix</p>
</td></tr>
<tr><td><code id="MINetworkPlot_topEdges_+3A_threshold">threshold</code></td>
<td>
<p>the maximum number of edges to draw for each subpopulation mutual information network</p>
</td></tr>
</table>

<hr>
<h2 id='outputNetworks_topEdges_matrix'>Wrapper to output the mutual information networks for subpopulations with size larger than a desired threshold.</h2><span id='topic+outputNetworks_topEdges_matrix'></span>

<h3>Description</h3>

<p>Wrapper to output the mutual information networks for subpopulations with size larger than a desired threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>outputNetworks_topEdges_matrix(dataMatrix, subpopulationLabels, threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="outputNetworks_topEdges_matrix_+3A_datamatrix">dataMatrix</code></td>
<td>
<p>data matrix with first column being the sample ID</p>
</td></tr>
<tr><td><code id="outputNetworks_topEdges_matrix_+3A_subpopulationlabels">subpopulationLabels</code></td>
<td>
<p>the subpopulation labels</p>
</td></tr>
<tr><td><code id="outputNetworks_topEdges_matrix_+3A_threshold">threshold</code></td>
<td>
<p>the number of edges to draw for each subpopulation mutual information network</p>
</td></tr>
</table>

<hr>
<h2 id='outputRepresentativeNetworks_topEdges'>Outputs the representative/clade networks (plots and summary vectors) for subpopulations with size larger than a desired threshold. Saves the networks and the data matrices without the smaller subpopulations.</h2><span id='topic+outputRepresentativeNetworks_topEdges'></span>

<h3>Description</h3>

<p>Outputs the representative/clade networks (plots and summary vectors) for subpopulations with size larger than a desired threshold. Saves the networks and the data matrices without the smaller subpopulations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>outputRepresentativeNetworks_topEdges(dataMatrix, subpopulationLabels,
  threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="outputRepresentativeNetworks_topEdges_+3A_datamatrix">dataMatrix</code></td>
<td>
<p>data matrix with first column being the sample ID</p>
</td></tr>
<tr><td><code id="outputRepresentativeNetworks_topEdges_+3A_subpopulationlabels">subpopulationLabels</code></td>
<td>
<p>the subpopulation labels</p>
</td></tr>
<tr><td><code id="outputRepresentativeNetworks_topEdges_+3A_threshold">threshold</code></td>
<td>
<p>the number of edges to draw for each subpopulation mutual information network</p>
</td></tr>
</table>

<hr>
<h2 id='PAC'>Partition Assisted Clustering
PAC 1) utilizes dsp or bsp-ll to recursively partition the data space and 2) applies a short round of kmeans style postprocessing to efficiently output clustered labels of data points.</h2><span id='topic+PAC'></span>

<h3>Description</h3>

<p>Partition Assisted Clustering
PAC 1) utilizes dsp or bsp-ll to recursively partition the data space and 2) applies a short round of kmeans style postprocessing to efficiently output clustered labels of data points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PAC(data, K, maxlevel = 40, method = "dsp", max.iter = 50)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PAC_+3A_data">data</code></td>
<td>
<p>a n x p data matrix</p>
</td></tr>
<tr><td><code id="PAC_+3A_k">K</code></td>
<td>
<p>number of final clusters in the output</p>
</td></tr>
<tr><td><code id="PAC_+3A_maxlevel">maxlevel</code></td>
<td>
<p>the maximum level of the partition</p>
</td></tr>
<tr><td><code id="PAC_+3A_method">method</code></td>
<td>
<p>partition method, either &quot;dsp(discrepancy based partition)&quot;,  or &quot;bsp(bayesian sequantial partition)&quot;</p>
</td></tr>
<tr><td><code id="PAC_+3A_max.iter">max.iter</code></td>
<td>
<p>maximum iteration for the kmeans step</p>
</td></tr>
</table>


<h3>Value</h3>

<p>y        cluter labels for the input
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n = 5e3                       # number of observations
p = 1                         # number of dimensions
K = 3                         # number of clusters
w = rep(1,K)/K                # component weights
mu &lt;- c(0,2,4)                # component means
sd &lt;- rep(1,K)/K              # component standard deviations
g &lt;- sample(1:K,prob=w,size=n,replace=TRUE)   # ground truth for clustering
X &lt;- as.matrix(rnorm(n=n,mean=mu[g],sd=sd[g]))
y &lt;- PAC(X, K)
print(fmeasure(g,y))
</code></pre>

<hr>
<h2 id='recordWithinClusterSpread'>Calculates the within cluster spread</h2><span id='topic+recordWithinClusterSpread'></span>

<h3>Description</h3>

<p>Calculates the within cluster spread
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recordWithinClusterSpread(sampleIDs, dim_subset = NULL, SubpopSizeFilter)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="recordWithinClusterSpread_+3A_sampleids">sampleIDs</code></td>
<td>
<p>A vector of sample names.</p>
</td></tr>
<tr><td><code id="recordWithinClusterSpread_+3A_dim_subset">dim_subset</code></td>
<td>
<p>a string vector of string names to subset the data columns for PAC; set to NULL to use all columns.</p>
</td></tr>
<tr><td><code id="recordWithinClusterSpread_+3A_subpopsizefilter">SubpopSizeFilter</code></td>
<td>
<p>threshold to filter out very small clusters with too few points; these very small subpopulations may not be outliers and not biologically relevant.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the sample within cluster spread
</p>

<hr>
<h2 id='refineSubpopulationLabels'>Refines the subpopulation labels from PAC using network alignment and small subpopulation information. Outputs a new set of files containing the representative labels.</h2><span id='topic+refineSubpopulationLabels'></span>

<h3>Description</h3>

<p>Refines the subpopulation labels from PAC using network alignment and small subpopulation information. Outputs a new set of files containing the representative labels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refineSubpopulationLabels(sampleIDs, dim_subset, clades_network_only,
  expressionGroupClamp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refineSubpopulationLabels_+3A_sampleids">sampleIDs</code></td>
<td>
<p>sampleID vector</p>
</td></tr>
<tr><td><code id="refineSubpopulationLabels_+3A_dim_subset">dim_subset</code></td>
<td>
<p>a string vector of string names to subset the data columns for PAC; set to NULL to use all columns</p>
</td></tr>
<tr><td><code id="refineSubpopulationLabels_+3A_clades_network_only">clades_network_only</code></td>
<td>
<p>the alignment results from MAN; used to translate the original sample-specific labels into clade labels</p>
</td></tr>
<tr><td><code id="refineSubpopulationLabels_+3A_expressiongroupclamp">expressionGroupClamp</code></td>
<td>
<p>clamps the subpopulations into desired number of expression groups for assigning small subpopulations into larger groups or their own groups.</p>
</td></tr>
</table>

<hr>
<h2 id='renamePrunedSubpopulations'>Prune away specified subpopulations in clades that are far away.</h2><span id='topic+renamePrunedSubpopulations'></span>

<h3>Description</h3>

<p>Prune away specified subpopulations in clades that are far away.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>renamePrunedSubpopulations(pacman_results, subpopulationsToPrune)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="renamePrunedSubpopulations_+3A_pacman_results">pacman_results</code></td>
<td>
<p>PAC-MAN analysis result matrix that contains network annotation, clade IDs and mean (centroid) clade expression levels.</p>
</td></tr>
<tr><td><code id="renamePrunedSubpopulations_+3A_subpopulationstoprune">subpopulationsToPrune</code></td>
<td>
<p>A vector of clade IDs; these clades will be pruned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns PAC-MAN analysis result matrix with pruned clades. The pruning process creates new clades to replace the original clade ID of the specified subpopulations.
</p>

<hr>
<h2 id='runElbowPointAnalysis'>Runs elbow point analysis to find the practical optimal number of clades to output. Outputs the average within sample cluster spread for all samples and the elbow point analysis plot with loess line fitted through the results.</h2><span id='topic+runElbowPointAnalysis'></span>

<h3>Description</h3>

<p>Runs elbow point analysis to find the practical optimal number of clades to output. Outputs the average within sample cluster spread for all samples and the elbow point analysis plot with loess line fitted through the results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runElbowPointAnalysis(ks, sampleIDs, dim_subset, num_PACSupop,
  smallSubpopCutoff, expressionGroupClamp, SubpopSizeFilter)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runElbowPointAnalysis_+3A_ks">ks</code></td>
<td>
<p>Vector that is a sequence of clade sizes.</p>
</td></tr>
<tr><td><code id="runElbowPointAnalysis_+3A_sampleids">sampleIDs</code></td>
<td>
<p>A vector of sample names.</p>
</td></tr>
<tr><td><code id="runElbowPointAnalysis_+3A_dim_subset">dim_subset</code></td>
<td>
<p>a string vector of string names to subset the data columns for PAC; set to NULL to use all columns.</p>
</td></tr>
<tr><td><code id="runElbowPointAnalysis_+3A_num_pacsupop">num_PACSupop</code></td>
<td>
<p>Number of PAC subpopulation explored in each sample.</p>
</td></tr>
<tr><td><code id="runElbowPointAnalysis_+3A_smallsubpopcutoff">smallSubpopCutoff</code></td>
<td>
<p>Cutoff of minor subpopulation not used in multiple alignments of networks</p>
</td></tr>
<tr><td><code id="runElbowPointAnalysis_+3A_expressiongroupclamp">expressionGroupClamp</code></td>
<td>
<p>clamps the subpopulations into desired number of expression groups for assigning small subpopulations into larger groups or their own groups.</p>
</td></tr>
<tr><td><code id="runElbowPointAnalysis_+3A_subpopsizefilter">SubpopSizeFilter</code></td>
<td>
<p>threshold to filter out very small clusters with too few points in the calculation of cluster spreads; these very small subpopulations may be outliers and not biologically relevant.</p>
</td></tr>
</table>

<hr>
<h2 id='samplePass'>Run PAC for Specified Samples</h2><span id='topic+samplePass'></span>

<h3>Description</h3>

<p>A wrapper to run PAC and output subpopulation mutual information networks. Please use the PAC function itself for individual samples or if the MAN step is not needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samplePass(sampleIDs, dim_subset, hyperrectangles, num_PACSupop, max.iter,
  num_networkEdge)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="samplePass_+3A_sampleids">sampleIDs</code></td>
<td>
<p>sampleID vector</p>
</td></tr>
<tr><td><code id="samplePass_+3A_dim_subset">dim_subset</code></td>
<td>
<p>a string vector of string names to subset the data columns for PAC; set to NULL to use all columns</p>
</td></tr>
<tr><td><code id="samplePass_+3A_hyperrectangles">hyperrectangles</code></td>
<td>
<p>number of hyperrectangles to learn for each sample</p>
</td></tr>
<tr><td><code id="samplePass_+3A_num_pacsupop">num_PACSupop</code></td>
<td>
<p>number of subpopulations to output for each sample using PAC</p>
</td></tr>
<tr><td><code id="samplePass_+3A_max.iter">max.iter</code></td>
<td>
<p>postprocessing kmeans iterations</p>
</td></tr>
<tr><td><code id="samplePass_+3A_num_networkedge">num_networkEdge</code></td>
<td>
<p>a threshold on the number of edges to output for each subpopulation mutual information network</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
