<!DOCTYPE html><html><head><title>Help for package PACVr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PACVr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PACVr-package'><p>PACVr: Plastome Assembly Coverage Visualization</p></a></li>
<li><a href='#PACVr.complete'><p>Execute the complete pipeline of <span class="pkg">PACVr</span></p></a></li>
<li><a href='#RCircos.Env'>
<p>Export the custom environment 'RCircos.Env'</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.0.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-28</td>
</tr>
<tr>
<td>Title:</td>
<td>Plastome Assembly Coverage Visualization</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Gruenstaeudl &lt;m_gruenstaeudl@fhsu.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings (&ge; 2.48.0), dplyr, GenomicAlignments (&ge; 1.18.1),
GenomicRanges, IRanges, logger, read.gb (&ge; 2.2), RCircos (&ge;
1.2.0), grDevices, stats, utils, tidyr</td>
</tr>
<tr>
<td>Description:</td>
<td>Visualizes the coverage depth of a complete plastid genome as well as the equality of its inverted repeat regions in relation to the circular, quadripartite genome structure and the location of individual genes. For more information, please see Gruenstaeudl and Jenke (2020) &lt;<a href="https://doi.org/10.1186%2Fs12859-020-3475-0">doi:10.1186/s12859-020-3475-0</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-3-Clause">BSD 3-clause License</a> + file LICENSE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/michaelgruenstaeudl/PACVr">https://github.com/michaelgruenstaeudl/PACVr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/michaelgruenstaeudl/PACVr/issues">https://github.com/michaelgruenstaeudl/PACVr/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), rentrez</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-28 00:59:58 UTC; gruenstaeudl</td>
</tr>
<tr>
<td>Author:</td>
<td>Gregory Smith [ctb],
  Nils Jenke [ctb],
  Michael Gruenstaeudl
    <a href="https://orcid.org/0000-0002-1666-1773"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-28 03:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='PACVr-package'>PACVr: Plastome Assembly Coverage Visualization</h2><span id='topic+PACVr'></span><span id='topic+PACVr-package'></span>

<h3>Description</h3>

<p>Visualizes the coverage depth of a complete plastid genome as well as the equality of its inverted repeat regions in relation to the circular, quadripartite genome structure and the location of individual genes. For more information, please see Gruenstaeudl and Jenke (2020) <a href="https://doi.org/10.1186/s12859-020-3475-0">doi:10.1186/s12859-020-3475-0</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Michael Gruenstaeudl <a href="mailto:m_gruenstaeudl@fhsu.edu">m_gruenstaeudl@fhsu.edu</a> (<a href="https://orcid.org/0000-0002-1666-1773">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Gregory Smith [contributor]
</p>
</li>
<li><p> Nils Jenke [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/michaelgruenstaeudl/PACVr">https://github.com/michaelgruenstaeudl/PACVr</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/michaelgruenstaeudl/PACVr/issues">https://github.com/michaelgruenstaeudl/PACVr/issues</a>
</p>
</li></ul>


<hr>
<h2 id='PACVr.complete'>Execute the complete pipeline of <span class="pkg">PACVr</span></h2><span id='topic+PACVr.complete'></span>

<h3>Description</h3>

<p>This function executes the complete pipeline of <span class="pkg">PACVr</span> 
via a single command.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PACVr.complete(
  gbkFile,
  bamFile,
  windowSize = 250,
  logScale = FALSE,
  threshold = 0.5,
  IRCheck = NA,
  relative = TRUE,
  textSize = 0.5,
  verbose = FALSE,
  output = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PACVr.complete_+3A_gbkfile">gbkFile</code></td>
<td>
<p>a character string that specifies the name of, and path to, 
the GenBank input file; alternatively, a character string of GenBank data</p>
</td></tr>
<tr><td><code id="PACVr.complete_+3A_bamfile">bamFile</code></td>
<td>
<p>a character string that specifies the name of, and path to, 
the BAM input file</p>
</td></tr>
<tr><td><code id="PACVr.complete_+3A_windowsize">windowSize</code></td>
<td>
<p>a numeric value that specifies window size in which the 
coverage is calculated</p>
</td></tr>
<tr><td><code id="PACVr.complete_+3A_logscale">logScale</code></td>
<td>
<p>a boolean that specifies if the coverage depth is to be 
log-transformed before visualizing it</p>
</td></tr>
<tr><td><code id="PACVr.complete_+3A_threshold">threshold</code></td>
<td>
<p>a numeric value that specifies the threshold for plotting 
coverage depth bars in red as opposed to the default black</p>
</td></tr>
<tr><td><code id="PACVr.complete_+3A_ircheck">IRCheck</code></td>
<td>
<p>a numeric value that specifies if tests 
for IRs of input genome should be performed, and - if IRs are present - 
which line type to be used for visualizing gene synteny between IRs;
0 = IR presence test but no synteny visualization, 
1 = IR presence test and synteny visualization, with ribbon lines between IRs,
2 = IR presence test and synteny visualization, with solid lines between IRs,
otherwise = neither IR presence test nor synteny visualization</p>
</td></tr>
<tr><td><code id="PACVr.complete_+3A_relative">relative</code></td>
<td>
<p>a boolean that specifies whether the threshold is a relative 
value of the average coverage instead of an absolute value</p>
</td></tr>
<tr><td><code id="PACVr.complete_+3A_textsize">textSize</code></td>
<td>
<p>a numeric value that specifies the relative font size of the 
text element in the visualization</p>
</td></tr>
<tr><td><code id="PACVr.complete_+3A_verbose">verbose</code></td>
<td>
<p>a boolean, that when TRUE, generates additional files with
detailed genomic region information</p>
</td></tr>
<tr><td><code id="PACVr.complete_+3A_output">output</code></td>
<td>
<p>a character string that specifies the name of, and path to, 
the output file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A file in pdf format containing a circular visualization of the 
input plastid genome and its sequence reads. 
As a function, returns 0 in case of visualization success.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gbkFile &lt;- system.file("extdata", "NC_045072/NC_045072.gb", package="PACVr")
bamFile &lt;- system.file("extdata", "NC_045072/NC_045072_subsampled.bam", package="PACVr")
outFile &lt;- paste(tempdir(), "/NC_045072__all_reads.pdf", sep="")
PACVr.complete(gbkFile=gbkFile, bamFile=bamFile, windowSize=250, logScale=FALSE,
               threshold=0.5, IRCheck=1, relative=TRUE, textSize=0.5,
               verbose=FALSE, output=outFile)

## End(Not run)
## Not run: 
gbkFile &lt;- system.file("extdata", "MG936619/MG936619.gb", package="PACVr")
bamFile &lt;- system.file("extdata", "MG936619/MG936619_subsampled.bam", package="PACVr")
outFile &lt;- paste(tempdir(), "/MG936619_CoverageViz.pdf", sep="")
PACVr.complete(gbkFile=gbkFile, bamFile=bamFile, windowSize=50, logScale=FALSE,
               threshold=0.5, IRCheck=NA, relative=TRUE, textSize=0.5,
               verbose=FALSE, output=outFile)

## End(Not run)
</code></pre>

<hr>
<h2 id='RCircos.Env'>
Export the custom environment 'RCircos.Env'
</h2><span id='topic+RCircos.Env'></span>

<h3>Description</h3>

<p>This function exports the custom environment 'RCircos.Env' used by <span class="pkg">RCircos</span>.
</p>


<h3>Details</h3>

<p><span class="pkg">PACVr</span> employs <span class="pkg">RCircos</span> as its visualization engine. In its operation, <span class="pkg">RCircos</span> defines a custom environment (called 'RCircos.Env') and reads/writes variables to this environment from various of its functions. In order to make this environment accessible to <span class="pkg">RCircos</span> within <span class="pkg">PACVr</span> and, simultaneously, fulfil the requirements of CRAN, this export command was created. For more information, please see the Stackoverflow post at <a href="https://stackoverflow.com/questions/56875962/r-package-transferring-environment-from-imported-package">https://stackoverflow.com/questions/56875962/r-package-transferring-environment-from-imported-package</a>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
