<!DOCTYPE html><html><head><title>Help for package pafr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pafr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#as_paf'><p>Coerce a data.frame or tibble into a pafr object</p></a></li>
<li><a href='#chrom_sizes'><p>Extract the sizes of all sequences in a paf alignment</p></a></li>
<li><a href='#dotplot'><p>Generate a dot plot from a paf alignment</p></a></li>
<li><a href='#filter_secondary_alignments'><p>Remove secondary alignments from a pafr alignment</p></a></li>
<li><a href='#Gb_lab'><p>Number formatters for scales in base pairs</p></a></li>
<li><a href='#highlight_query'><p>Highlight segments of a query or target genome in a dot plot</p></a></li>
<li><a href='#plot_coverage'><p>Plot the regions of one genome that are covered by alignments in a paf file</p></a></li>
<li><a href='#plot_synteny'><p>Plot synteny between a query and target sequence in a PAF alignment</p></a></li>
<li><a href='#read_bed'><p>Read genomic intervals in bed format</p></a></li>
<li><a href='#read_paf'><p>Read a genomic alignment in PAF format</p></a></li>
<li><a href='#theme_coverage_plot'><p>A minimalistic ggplot2 theme designed for use with genome coverage plots</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Read, Manipulate and Visualize 'Pairwise mApping Format' Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions to read, process and visualize pairwise sequence 
 alignments in the 'PAF' format used by 'minimap2' and other whole-genome aligners. 
 'minimap2' is described by Li H. (2018) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbty191">doi:10.1093/bioinformatics/bty191</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.4), ggplot2,</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, tibble, stringr, rlang</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, covr, knitr, ggpubr, rmarkdown, microbenchmark</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://dwinter.github.io/pafr/">https://dwinter.github.io/pafr/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/dwinter/pafr/issues">https://github.com/dwinter/pafr/issues</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-12-06 19:59:37 UTC; david</td>
</tr>
<tr>
<td>Author:</td>
<td>David Winter <a href="https://orcid.org/0000-0002-6165-0029"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Kate Lee <a href="https://orcid.org/0000-0002-0886-3746"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Murray Cox <a href="https://orcid.org/0000-0003-1936-0236"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>David Winter &lt;david.winter@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-12-08 10:20:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='as_paf'>Coerce a data.frame or tibble into a pafr object</h2><span id='topic+as_paf'></span>

<h3>Description</h3>

<p>The main reason to use this function is speed up the process of reading in
a large paf file that has no tags. Functions like read.table, read_delim
(reader) and fread (data.table) can process a 12 column file  more quickly
than pafr's read_paf. If you you do not need tag data for your analyses or
visualizations, it might make sense to use a fast reading function to get a
12 column data.frame,  convert that data.frame into a 'pafr  object with this
function. The 'pafr' object can then work easily with the functions in this package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_paf(paf_data_frame)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_paf_+3A_paf_data_frame">paf_data_frame</code></td>
<td>
<p>a data.frame object with 12 columns. Column names and
types will be over-written in the returned object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a pafr object
</p>


<h3>See Also</h3>

<p>read_paf
</p>

<hr>
<h2 id='chrom_sizes'>Extract the sizes of all sequences in a paf alignment</h2><span id='topic+chrom_sizes'></span>

<h3>Description</h3>

<p>Extract the sizes of all sequences in a paf alignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chrom_sizes(ali)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chrom_sizes_+3A_ali">ali</code></td>
<td>
<p>pafr or tibble containing the genome alignment (as returned by
<code><a href="#topic+read_paf">read_paf</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with two elements (tlens and qlens)  Each element is a
dataframe with one column of sequence names and another column containing
the length of each sequence
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ali &lt;- read_paf( system.file("extdata", "fungi.paf", package="pafr") )
chrom_sizes(ali)
</code></pre>

<hr>
<h2 id='dotplot'>Generate a dot plot from a paf alignment</h2><span id='topic+dotplot'></span>

<h3>Description</h3>

<p>Generate a dot plot from a paf alignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dotplot(
  ali,
  order_by = c("size", "qstart", "provided"),
  label_seqs = FALSE,
  dashes = TRUE,
  ordering = list(),
  alignment_colour = "black",
  xlab = "query",
  ylab = "target",
  line_size = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dotplot_+3A_ali">ali</code></td>
<td>
<p>pafr or tibble containing the genome alignment (as returned by
<code><a href="#topic+read_paf">read_paf</a></code>)</p>
</td></tr>
<tr><td><code id="dotplot_+3A_order_by">order_by</code></td>
<td>
<p>How the query and target sequences should be ordered in the
dot plot. Option must be one of 'size' (smallest-to-largest), 'qstart' (query organised
smallest to largest, target by first match in the query genome) or 'provided'
(ordering as specified in the <code>ordering</code> argument)</p>
</td></tr>
<tr><td><code id="dotplot_+3A_label_seqs">label_seqs</code></td>
<td>
<p>boolean  If TRUE, label centre of query and target
sequences in margins of the dot plot</p>
</td></tr>
<tr><td><code id="dotplot_+3A_dashes">dashes</code></td>
<td>
<p>boolean  If TRUE, add dashes to borders of query and
target sequences in the dot plot</p>
</td></tr>
<tr><td><code id="dotplot_+3A_ordering">ordering</code></td>
<td>
<p>If <code>order_by</code> is set to TRUE,
this variable should be a list with two elements specifying the order of query
and then target sequences in the dot plot. This option is ignored if
<code>order_by</code> is set to other values</p>
</td></tr>
<tr><td><code id="dotplot_+3A_alignment_colour">alignment_colour</code></td>
<td>
<p>character  The colour used to draw each aligned
section in the dot plot (defaults to black)</p>
</td></tr>
<tr><td><code id="dotplot_+3A_xlab">xlab</code></td>
<td>
<p>character  The x-axis label (defaults to 'query')</p>
</td></tr>
<tr><td><code id="dotplot_+3A_ylab">ylab</code></td>
<td>
<p>character  The y-axis label (defaults to 'target')</p>
</td></tr>
<tr><td><code id="dotplot_+3A_line_size">line_size</code></td>
<td>
<p>The width of the line used to represent an alignment in the
dot plot (defaults to 2)</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>ali &lt;- read_paf( system.file("extdata", "fungi.paf", package="pafr") )
dotplot(ali)
dotplot(ali) + theme_bw()
dotplot(ali, label_seqs=TRUE, order_by="qstart", alignment_colour="blue")
</code></pre>

<hr>
<h2 id='filter_secondary_alignments'>Remove secondary alignments from a pafr alignment</h2><span id='topic+filter_secondary_alignments'></span>

<h3>Description</h3>

<p>Remove secondary alignments from a pafr alignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_secondary_alignments(ali, remove_inversions = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_secondary_alignments_+3A_ali">ali</code></td>
<td>
<p>Genomic alignment in <code>pafr</code> or <code>tbl_df</code> format, as
returned by <code><a href="#topic+read_paf">read_paf</a></code></p>
</td></tr>
<tr><td><code id="filter_secondary_alignments_+3A_remove_inversions">remove_inversions</code></td>
<td>
<p>logical  If TRUE, also remove inversions (tp
flag 'I' or 'i') from the alignment</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>ali &lt;- read_paf( system.file("extdata", "fungi.paf", package="pafr") ) 
ali
filter_secondary_alignments(ali)
</code></pre>

<hr>
<h2 id='Gb_lab'>Number formatters for scales in base pairs</h2><span id='topic+Gb_lab'></span><span id='topic+Mb_lab'></span><span id='topic+Kb_lab'></span>

<h3>Description</h3>

<p>For use with <code><a href="ggplot2.html#topic+ggplot2">ggplot2</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Gb_lab(x)

Mb_lab(x)

Kb_lab(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Gb_lab_+3A_x">x</code></td>
<td>
<p>The data (in base pairs) to be formatted as Gb, Mb or Kb</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector with scale labels
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ali &lt;- read_paf(system.file("extdata", "fungi.paf", package="pafr"))
doplot(ali) + scale_x_continuous("Genomic position", label=Mb_lab)

## End(Not run)
</code></pre>

<hr>
<h2 id='highlight_query'>Highlight segments of a query or target genome in a dot plot</h2><span id='topic+highlight_query'></span><span id='topic+highlight_target'></span>

<h3>Description</h3>

<p>This plot is intended to be used in conjunction with <code>link{dotplot}</code>.
Adding <code>higlight_query</code> or <code>highlight_target</code> to a dotplot function call
(see examples below) will add a rectangular 'highlight' corresponding to a
particular genomic interval in the corresponding genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>highlight_query(bed, fill = "yellow", colour = "black", alpha = 0.6)

highlight_target(bed, fill = "yellow", colour = "black", alpha = 0.6)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="highlight_query_+3A_bed">bed</code></td>
<td>
<p><code>data.frame</code> or <code>tbl_df</code> containing a bed file, as returned by
<code><a href="#topic+read_bed">read_bed</a></code>. Should contain three columns named 'chrom', 'start' 
and 'end'</p>
</td></tr>
<tr><td><code id="highlight_query_+3A_fill">fill</code></td>
<td>
<p>character  Fill colour for highlight segment</p>
</td></tr>
<tr><td><code id="highlight_query_+3A_colour">colour</code></td>
<td>
<p>character  Outline colour for highlight segment</p>
</td></tr>
<tr><td><code id="highlight_query_+3A_alpha">alpha</code></td>
<td>
<p>character  Opacity ([0-1]) for highlight segment</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>ali &lt;- read_paf( system.file("extdata", "fungi.paf", package="pafr") )
cen &lt;- read_bed(system.file("extdata", "Q_centro.bed", package="pafr"))
dotplot(ali) + highlight_query(cen)
interval &lt;- data.frame(chrom="T_chr3", start=2000000, end=3000000)
dotplot(ali, label_seqs=TRUE) + 
   highlight_target(interval)
</code></pre>

<hr>
<h2 id='plot_coverage'>Plot the regions of one genome that are covered by alignments in a paf file</h2><span id='topic+plot_coverage'></span>

<h3>Description</h3>

<p>Each sequence in the focal genome is displayed as a rectangle, with regions
covered by an alignment shaded as per the <code>fill</code> argument described
below. Uncovered regions remain white.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_coverage(
  ali,
  target = TRUE,
  fill = "forestgreen",
  direct_label = TRUE,
  label_colour = "black",
  xlab = "Position in sequence",
  x_labeller = Mb_lab
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_coverage_+3A_ali">ali</code></td>
<td>
<p>alignment  An alignment as read by <code><a href="#topic+read_paf">read_paf</a></code></p>
</td></tr>
<tr><td><code id="plot_coverage_+3A_target">target</code></td>
<td>
<p>logical  If TRUE, display coverage for the target
genome; if FALSE, display coverage for the query</p>
</td></tr>
<tr><td><code id="plot_coverage_+3A_fill">fill</code></td>
<td>
<p>character  How to colour the alignment blocks. If the character
provided is the name of a column in the alignment, this column will be passed
to <code><a href="ggplot2.html#topic+ggplot2">ggplot2</a></code> to shade alignment blocks. Otherwise, the character
is treated as a single colour to be used for all alignment blocks.</p>
</td></tr>
<tr><td><code id="plot_coverage_+3A_direct_label">direct_label</code></td>
<td>
<p>logical  If TRUE, use geom_text to directly label the
name of the focal sequences; if FALSE, no direct labels are drawn</p>
</td></tr>
<tr><td><code id="plot_coverage_+3A_label_colour">label_colour</code></td>
<td>
<p>character  Colour used for direct labels</p>
</td></tr>
<tr><td><code id="plot_coverage_+3A_xlab">xlab</code></td>
<td>
<p>string  Name for the x-axis</p>
</td></tr>
<tr><td><code id="plot_coverage_+3A_x_labeller">x_labeller</code></td>
<td>
<p>function  Function to be used to label the x-axis (defaults to
<code><a href="#topic+Mb_lab">Mb_lab</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this function uses <code><a href="#topic+theme_coverage_plot">theme_coverage_plot</a></code> to style the
graph. Using another ggplot theme on the plot may produce unexpected results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ali &lt;- read_paf( system.file("extdata", "fungi.paf", package="pafr") )
plot_coverage(ali)
plot_coverage(ali, fill='qname', direct_label=FALSE) + 
   scale_fill_brewer(palette="Set1")
</code></pre>

<hr>
<h2 id='plot_synteny'>Plot synteny between a query and target sequence in a PAF alignment</h2><span id='topic+plot_synteny'></span>

<h3>Description</h3>

<p>Plot synteny between a query and target sequence in a PAF alignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_synteny(
  ali,
  q_chrom,
  t_chrom,
  centre = TRUE,
  rc = FALSE,
  xlab = "Position in query",
  ylab = "",
  x_labeller = Mb_lab
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_synteny_+3A_ali">ali</code></td>
<td>
<p>pafr or tibble containing the genome alignment (as returned by
<code><a href="#topic+read_paf">read_paf</a></code>)</p>
</td></tr>
<tr><td><code id="plot_synteny_+3A_q_chrom">q_chrom</code></td>
<td>
<p>character  Name for the query sequence</p>
</td></tr>
<tr><td><code id="plot_synteny_+3A_t_chrom">t_chrom</code></td>
<td>
<p>character  Name for the target sequence</p>
</td></tr>
<tr><td><code id="plot_synteny_+3A_centre">centre</code></td>
<td>
<p>logical  If TRUE (default), adjust the position of the target 
sequence, so it is centred on the query. If not, both sequences start at
position zero</p>
</td></tr>
<tr><td><code id="plot_synteny_+3A_rc">rc</code></td>
<td>
<p>logical  If TRUE, use the reverse and complement for the target
sequence</p>
</td></tr>
<tr><td><code id="plot_synteny_+3A_xlab">xlab</code></td>
<td>
<p>string  Name for the x-axis</p>
</td></tr>
<tr><td><code id="plot_synteny_+3A_ylab">ylab</code></td>
<td>
<p>string  Name for the y-axis</p>
</td></tr>
<tr><td><code id="plot_synteny_+3A_x_labeller">x_labeller</code></td>
<td>
<p>Function to be used to label the x-axis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object that displays synteny between query and target sequences
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ali &lt;- read_paf( system.file("extdata", "fungi.paf", package="pafr") )
long_ali &lt;- subset(ali, alen &gt; 1e4)
plot_synteny(long_ali, q_chrom="Q_chr3", t_chrom="T_chr4", centre=TRUE)
plot_synteny(long_ali, q_chrom="Q_chr5", t_chrom="T_chr5", centre=TRUE)
plot_synteny(long_ali, q_chrom="Q_chr5", t_chrom="T_chr5", centre=TRUE, rc=TRUE)
</code></pre>

<hr>
<h2 id='read_bed'>Read genomic intervals in bed format</h2><span id='topic+read_bed'></span>

<h3>Description</h3>

<p>The first three columns of the file specified by <code>file_name</code> must
contain data in the standard bed format (i.e., a genomic interval
represented by 0-based half-open interval with seq-id, start and end position). 
These columns will be renamed to 'chrom', 'start' and 'end', respectively. Any
other columns present in the data will be left unmodified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_bed(file_name, tibble = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_bed_+3A_file_name">file_name</code></td>
<td>
<p>Path to the bed file to be read in</p>
</td></tr>
<tr><td><code id="read_bed_+3A_tibble">tibble</code></td>
<td>
<p>logical  If TRUE, the genomic intervals are returned as
a tidy <code>tbl_df</code>.</p>
</td></tr>
<tr><td><code id="read_bed_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code>read.table</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The file is read into memory with <code><a href="utils.html#topic+read.table">read.table</a></code>, with the
argument <code>sep</code> set to <code>'\t'</code> and <code>stringsAsFactors</code> set to
FALSE. All other arguments are left as default, but arguments can be passed
from <code>read_bed</code> to <code>read.table</code>.
</p>


<h3>Value</h3>

<p>Either a <code>data.frame</code> or a <code>tbl_df</code> with at least three
columns named 'chrom', 'start' and 'end'
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bed_path &lt;- system.file("extdata", "Q_centro.bed", package="pafr")
centro &lt;- read_bed(bed_path)
centro
# Can pass arguments to read.table
miss_two &lt;- read_bed(bed_path, skip=2)
miss_two 
</code></pre>

<hr>
<h2 id='read_paf'>Read a genomic alignment in PAF format</h2><span id='topic+read_paf'></span>

<h3>Description</h3>

<p>See the package vignette for detailed information on the file format and its
representation as an R object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_paf(file_name, tibble = FALSE, include_tags = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_paf_+3A_file_name">file_name</code></td>
<td>
<p>Path to the .paf file</p>
</td></tr>
<tr><td><code id="read_paf_+3A_tibble">tibble</code></td>
<td>
<p>logical  If TRUE, the genomic alignments are returned as
a tidy <code>tbl_df</code></p>
</td></tr>
<tr><td><code id="read_paf_+3A_include_tags">include_tags</code></td>
<td>
<p>logical if TRUE (default) read additional information
about each alignment encoded as PAF tags. Setting this to FALSE will speed up
parsing of paf alignments, specially those with large CIGAR strings/</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either a <code>pafr</code> object, which acts as a <code>data.frame</code>, or a
<code>tbl_df</code> containing information on genomic alignments. The contents of
this table are described in detail in the pafr package vingette.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ali &lt;- read_paf( system.file("extdata", "fungi.paf", package="pafr") )
ali
</code></pre>

<hr>
<h2 id='theme_coverage_plot'>A minimalistic ggplot2 theme designed for use with genome coverage plots</h2><span id='topic+theme_coverage_plot'></span>

<h3>Description</h3>

<p>This theme is used as the default when <code><a href="#topic+plot_coverage">plot_coverage</a></code> is called,
so you should usually only call this function to modify the appearance of the
coverage plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_coverage_plot(facet_labs = TRUE, show_legend = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="theme_coverage_plot_+3A_facet_labs">facet_labs</code></td>
<td>
<p>logical  If TRUE (default), label sequences using the facet
labels; if FALSE, sequences are labeled directly using 
<code><a href="ggplot2.html#topic+geom_text">geom_text</a></code></p>
</td></tr>
<tr><td><code id="theme_coverage_plot_+3A_show_legend">show_legend</code></td>
<td>
<p>logical  If TRUE (default), label display any legend
associated with the fill parameter of <code>plot_coverage</code>; if FALSE, 
do not display a legend</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>ali &lt;- read_paf( system.file("extdata", "fungi.paf", package="pafr") )
plot_coverage(ali) + theme_coverage_plot(show_legend=FALSE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
