<!DOCTYPE html><html><head><title>Help for package pagoo</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pagoo}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#load_pangenomeRDS'><p>Load A Pagoo Pangenome</p></a></li>
<li><a href='#pagoo'><p>Create a Pagoo Object</p></a></li>
<li><a href='#panaroo_2_pagoo'><p>Read panaroos's output into a pagoo's R6 class object</p></a></li>
<li><a href='#PgR6'><p>PgR6 basic class</p></a></li>
<li><a href='#PgR6M'><p>PgR6 class with methods.</p></a></li>
<li><a href='#PgR6MS'><p>PgR6 class with Methods and Sequences.</p></a></li>
<li><a href='#roary_2_pagoo'><p>Read roary's output into a pagoo's R6 class object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.3.17</td>
</tr>
<tr>
<td>Title:</td>
<td>Analyze Bacterial Pangenomes in R with 'Pagoo'</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides an encapsulated, object-oriented class system for
  analyzing bacterial pangenomes. For a definition of this concept, see
  Tettelin, et al. (2005) &lt;<a href="https://doi.org/10.1073%2Fpnas.0506758102">doi:10.1073/pnas.0506758102</a>&gt;. It uses the R6
  package as backend. It was designed in order to facilitate and speed-up
  the comparative analysis of multiple bacterial genomes, standardizing and
  optimizing routine tasks performed everyday. There are a handful of things
  done everyday when working with bacterial pangenomes: subset, summarize,
  extract, visualize and store data. So, 'pagoo' is intended to facilitate these
  tasks as much as possible. For a description of the implemented data structure
  and methods, see Ferres &amp; Iraola (2020), &lt;<a href="https://doi.org/10.1101%2F2020.07.29.226951">doi:10.1101/2020.07.29.226951</a>&gt;.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ignacio Ferres &lt;iferres@pasteur.edu.uy&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://iferres.github.io/pagoo/">https://iferres.github.io/pagoo/</a>, <a href="https://github.com/iferres/pagoo">https://github.com/iferres/pagoo</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/iferres/pagoo/issues">https://github.com/iferres/pagoo/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), S4Vectors, Biostrings, ggplot2</td>
</tr>
<tr>
<td>Imports:</td>
<td>R6, reshape2, vegan, GenomicRanges, BiocGenerics,
shinyWidgets, shinydashboard, DT, plotly, magrittr, heatmaply,
dendextend, ggfortify, shiny</td>
</tr>
<tr>
<td>Suggests:</td>
<td>micropan, patchwork, ape, phangorn, pegas, DECIPHER,
rhierbaps, IRanges, knitr, rmarkdown, testthat, covr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-11-18 18:24:46 UTC; iferres</td>
</tr>
<tr>
<td>Author:</td>
<td>Ignacio Ferres <a href="https://orcid.org/0000-0003-0910-6568"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Gregorio Iraola <a href="https://orcid.org/0000-0002-6516-3404"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Institut Pasteur de Montevideo [fnd]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-11-18 18:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='load_pangenomeRDS'>Load A Pagoo Pangenome</h2><span id='topic+load_pangenomeRDS'></span>

<h3>Description</h3>

<p>This function loads a pagoo pangenome from a '.RDS' file generated
by the 'save_pangenomeRDS' method. Objects loaded by this functions keep their
states, i.e : dropped/recovered organisms are conserved, as well as the
'core_level' setted when the object was originally saved.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_pangenomeRDS(file, pkg, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_pangenomeRDS_+3A_file">file</code></td>
<td>
<p>The path to the pangenome '.RDS' file.</p>
</td></tr>
<tr><td><code id="load_pangenomeRDS_+3A_pkg">pkg</code></td>
<td>
<p>The package to use to load the object. Shouldn't be necessary to
provide, but may be useful in some cases.</p>
</td></tr>
<tr><td><code id="load_pangenomeRDS_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to the pagoo object. <code>sep</code> and
<code>core_level</code> overwrite the values stored in the file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>PgR6MS</code> class object, or a <code>PgR6M</code> object (with or
without sequences, respectively).
</p>

<hr>
<h2 id='pagoo'>Create a Pagoo Object</h2><span id='topic+pagoo'></span>

<h3>Description</h3>

<p>This is the main function to load a pagoo object. It's safer and
more friendly than using pagoo's class constructors (<a href="#topic+PgR6">PgR6</a>,
<a href="#topic+PgR6M">PgR6M</a>, and <a href="#topic+PgR6MS">PgR6MS</a>). This function returns either a
<code><a href="#topic+PgR6M">PgR6M</a></code> class object, or a <code><a href="#topic+PgR6MS">PgR6MS</a></code> class object,
depending on the parameters provided. If sequences are provided, it returns
the latter. See below for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pagoo(
  data,
  org_meta,
  cluster_meta,
  sequences,
  core_level = 95,
  sep = "__",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pagoo_+3A_data">data</code></td>
<td>
<p>A <code>data.frame</code> or
<code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> containing at least the
following columns: <code>gene</code> (gene name), <code>org</code> (organism name to
which the gene belongs to), and <code>cluster</code> (group of orthologous to
which the gene belongs to). More columns can be added as metadata for each
gene.</p>
</td></tr>
<tr><td><code id="pagoo_+3A_org_meta">org_meta</code></td>
<td>
<p>(optional) A <code>data.frame</code> or
<code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> containing additional
metadata for organisms. This <code>data.frame</code> must have a column named
&quot;org&quot; with valid organisms names (that is, they should match with those
provided in <code>data</code>, column <code>org</code>), and additional columns will be
used as metadata. Each row should correspond to each organism.</p>
</td></tr>
<tr><td><code id="pagoo_+3A_cluster_meta">cluster_meta</code></td>
<td>
<p>(optional) A <code>data.frame</code> or
<code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> containing additional
metadata for clusters. This <code>data.frame</code> must have a column named
&quot;cluster&quot; with valid organisms names (that is, they should match with those
provided in <code>data</code>, column <code>cluster</code>), and additional columns
will be used as metadata. Each row should correspond to each cluster.</p>
</td></tr>
<tr><td><code id="pagoo_+3A_sequences">sequences</code></td>
<td>
<p>(optional) Can accept: 1) a named <code>list</code> of named
<code>character</code> vector. Name of list are names of organisms, names of
character vector are gene names; or 2) a named <code>list</code> of
<code><a href="Biostrings.html#topic+XStringSet-class">DNAStringSetList</a></code> objects (same
requirements as (1), but with BStringSet names as gene names); or 3) a
<code><a href="Biostrings.html#topic+XStringSetList-class">DNAStringSetList</a></code> (same
requirements as (2) but <code>DNAStringSetList</code> names are organisms names).
If this parameter is used, then a <code><a href="#topic+PgR6MS">PgR6MS</a></code> class object is
returned.</p>
</td></tr>
<tr><td><code id="pagoo_+3A_core_level">core_level</code></td>
<td>
<p>The initial core_level (that's the percentage of organisms a
core cluster must be in to be considered as part of the core genome). Must
be a number between 100 and 85, (default: 95). You can change it later by
using the <code>$core_level</code> field once the object was created.</p>
</td></tr>
<tr><td><code id="pagoo_+3A_sep">sep</code></td>
<td>
<p>A separator. By default is '__'(two underscores). It will be used
to create a unique <code>gid</code> (gene identifier) for each gene. <code>gid</code>s
are created by pasting <code>org</code> to <code>gene</code>, separated by <code>sep</code>.</p>
</td></tr>
<tr><td><code id="pagoo_+3A_verbose">verbose</code></td>
<td>
<p><code>logical</code>. Whether to display progress messages when
loading class.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This package uses
[R6](https://r6.r-lib.org/articles/Introduction.html) classes to provide a
unified, comprehensive, standardized, but at the same time flexible, way to
analyze a pangenome. The idea is to have a single object which contains both
the data and the basic methods to analyze them, as well as manipulate fields,
explore, and to use in harmony with the already existing and extensive list of R
packages available created for comparative genomics and genetics.
</p>
<p>For more information, tutorials, and resources, please visit https://iferres.github.io/pagoo/ .
</p>


<h3>Index</h3>



<h4>Active Bindings</h4>


<ul>
<li> <p><a href="#field-pan_matrix"><code>$pan_matrix</code></a>
</p>
</li>
<li> <p><a href="#field-organisms"><code>$organisms</code></a>
</p>
</li>
<li> <p><a href="#field-clusters"><code>$clusters</code></a>
</p>
</li>
<li> <p><a href="#field-genes"><code>$genes</code></a>
</p>
</li>
<li> <p><a href="#field-sequences"><code>$sequences</code></a>
</p>
</li>
<li> <p><a href="#field-core_level"><code>$core_level</code></a>
</p>
</li>
<li> <p><a href="#field-core_genes"><code>$core_genes</code></a>
</p>
</li>
<li> <p><a href="#field-core_clusters"><code>$core_clusters</code></a>
</p>
</li>
<li> <p><a href="#field-core_sequences"><code>$core_sequences</code></a>
</p>
</li>
<li> <p><a href="#field-shell_genes"><code>$shell_genes</code></a>
</p>
</li>
<li> <p><a href="#field-shell_clusters"><code>$shell_clusters</code></a>
</p>
</li>
<li> <p><a href="#field-shell_sequences"><code>$shell_sequences</code></a>
</p>
</li>
<li> <p><a href="#field-cloud_genes"><code>$cloud_genes</code></a>
</p>
</li>
<li> <p><a href="#field-cloud_clusters"><code>$cloud_clusters</code></a>
</p>
</li>
<li> <p><a href="#field-cloud_sequences"><code>$cloud_sequences</code></a>
</p>
</li></ul>




<h3>Active bindings</h3>


<a id="field-pan_matrix"></a>
<ul>
<li><p><b><code>$pan_matrix</code></b> The panmatrix. Rows are organisms, and
columns are groups of orthologous. Cells indicates the presence (&gt;=1) or
absence (0) of a given gene, in a given organism. Cells can have values
greater than 1 if contain in-paralogs.
<a id="field-organisms"></a>
</p>
</li>
<li><p><b><code>$organisms</code></b> A <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> with available
organism names, and organism number identifier as <code>rownames()</code>. (Dropped
organisms will not be displayed in this field, see <code>$dropped</code> below).
Additional metadata will be shown if provided, as additional columns.
<a id="field-clusters"></a>
</p>
</li>
<li><p><b><code>$clusters</code></b> A <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> with the groups
of orthologous (clusters). Additional metadata will be shown as additional columns,
if provided before. Each row corresponds to each cluster.
<a id="field-genes"></a>
</p>
</li>
<li><p><b><code>$genes</code></b> A <code><a href="IRanges.html#topic+DataFrameList-class">SplitDataFrameList</a></code> object with
one entry per cluster. Each element contains a <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code>
with gene ids (<code>&lt;gid&gt;</code>) and additional metadata, if provided. <code>gid</code> are
created by <code>paste</code>ing organism and gene names, so duplication in gene names
are avoided.
<a id="field-sequences"></a>
</p>
</li>
<li><p><b><code>$sequences</code></b> A <code><a href="Biostrings.html#topic+XStringSetList-class">DNAStringSetList</a></code> with the
set of sequences grouped by cluster. Each group is accessible as were a list. All
<code>Biostrings</code> methods are available.
<a id="field-core_level"></a>
</p>
</li>
<li><p><b><code>$core_level</code></b> The percentage of organisms a gene must be in
to be considered as part of the coregenome. <code>core_level = 95</code> by default.
Can't be set above 100, and below 85 raises a warning.
<a id="field-core_genes"></a>
</p>
</li>
<li><p><b><code>$core_genes</code></b> Like <code>genes</code>, but only showing core genes.
<a id="field-core_clusters"></a>
</p>
</li>
<li><p><b><code>$core_clusters</code></b> Like <code>$clusters</code>, but only showing core
clusters.
<a id="field-core_sequences"></a>
</p>
</li>
<li><p><b><code>$core_sequences</code></b> Like <code>$sequences</code>, but only showing core
sequences.
<a id="field-cloud_genes"></a>
</p>
</li>
<li><p><b><code>$cloud_genes</code></b> Like <code>genes</code>, but only showing cloud genes.
These are defined as those clusters which contain a single gene (singletons), plus
those which have more than one but its organisms are probably clonal due to identical
general gene content. Colloquially defined as strain-specific genes.
<a id="field-cloud_clusters"></a>
</p>
</li>
<li><p><b><code>$cloud_clusters</code></b> Like <code>$clusters</code>, but only showing cloud
clusters as defined above.
<a id="field-cloud_sequences"></a>
</p>
</li>
<li><p><b><code>$cloud_sequences</code></b> Like <code>$sequences</code>, but only showing cloud
sequences as defined above.
<a id="field-shell_genes"></a>
</p>
</li>
<li><p><b><code>$shell_genes</code></b> Like <code>genes</code>, but only showing shell genes.
These are defined as those clusters than don't belong neither to the core genome,
nor to cloud genome. Colloquially defined as genes that are present in some but not
all strains, and that aren't strain-specific.
<a id="field-shell_clusters"></a>
</p>
</li>
<li><p><b><code>$shell_clusters</code></b> Like <code>$clusters</code>, but only showing shell
clusters, as defined above.
<a id="field-shell_sequences"></a>
</p>
</li>
<li><p><b><code>$shell_sequences</code></b> Like <code>$sequences</code>, but only showing shell
sequences, as defined above.
<a id="field-summary_stats"></a>
</p>
</li>
<li><p><b><code>$summary_stats</code></b> A <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> with
information about the number of core, shell, and cloud clusters, as well as the
total number of clusters.
<a id="field-random_seed"></a>
</p>
</li>
<li><p><b><code>$random_seed</code></b> The last <code>.Random.seed</code>. Used for
reproducibility purposes only.
<a id="field-dropped"></a>
</p>
</li>
<li><p><b><code>$dropped</code></b> A <code>character</code> vector with dropped organism
names, and organism number identifier as <code>names()</code>
</p>
</li></ul>




<h3>Methods</h3>


<p>Below is a comprehensive description of all the methods provided by the object.
</p>
<a id="method-add_metadata"></a>
<dl>
<dt><b>Add metadata</b></dt><dd>


<h4>Description:</h4>

<p>Add metadata to the object. You can add metadata to each organism, to each
group of orthologous, or to each gene. Elements with missing data should be filled
by <code>NA</code> (dimensions of the provided data.frame must be coherent with object
data).
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$add_metadata(map = 'org', df)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>map</code></b>: <code>character</code> identifying the metadata to map. Can
be one of <code>"org"</code>, <code>"group"</code>, or <code>"gid"</code>.
</p>
</li>
<li><p><b><code>df</code></b>: <code>data.frame</code> or <code>DataFrame</code> with the metadata to
add. For each case, a column named as <code>"map"</code> must exists, which should
contain identifiers for each element. In the case of adding gene (<code>gid</code>)
metadata,each gene should be referenced by the name of the organism and the name
of the gene as provided in the <code>"data"</code> data.frame, separated by the
<code>"sep"</code> argument.
</p>
</li></ul>




<h4>Return:</h4>

<p><code>self</code> invisibly, but with additional metadata.
</p>

</dd>
</dl>
<a id="method-drop"></a>
<dl>
<dt><b>Drop an organism</b></dt><dd>


<h4>Description:</h4>

<p>Drop an organism from the dataset. This method allows to hide an organism from
the real dataset, ignoring it in downstream analyses. All the fields and
methods will behave as it doesn't exist. For instance, if you decide to drop
organism 1, the <code>$pan_matrix</code> field (see below) would not show it when
called.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$drop(x)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>x</code></b>:  <code>character</code> or <code>numeric</code>. The name of the
organism wanted to be dropped, or its numeric id as returned in
<code>$organism</code> field (see below).
</p>
</li></ul>




<h4>Return:</h4>

<p><code>self</code> invisibly, but with <code>x</code> dropped. It isn't necessary
to assign the function call to a new object, nor to re-write it as R6 objects
are mutable.
</p>

</dd>
</dl>
<a id="method-recover"></a>
<dl>
<dt><b>Recover a dropped organism</b></dt><dd>


<h4>Description:</h4>

<p>Recover a previously <code>$drop()</code>ped organism (see above). All fields
and methods will start to behave considering this organism again.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$recover(x)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>x</code></b>: <code>character</code> or <code>numeric</code>. The name of the
organism wanted to be recover, or its numeric id as returned in
<code>$dropped</code> field (see below).
</p>
</li></ul>




<h4>Return:</h4>

<p><code>self</code> invisibly, but with <code>x</code> recovered. It isn't necessary
to assign the function call to a new object, nor to re-write it as R6 objects
are mutable.
</p>

</dd>
</dl>
<a id="method-write_pangenome"></a>
<dl>
<dt><b>Write a pangenome as flat (text) files.</b></dt><dd>


<h4>Description:</h4>

<p>Write the pangenome data as flat tables (text). Is not the most recommended way
to save a pangenome, since you can loose information as numeric precision,
column classes (factor, numeric, integer), and the state of the object itself
(i.e. dropped organisms, or core_level), loosing reproducibility. Use
<code>save_pangenomeRDS</code> for a more precise way of saving a pagoo object.
Still, it is useful if you want to work with the data outside R, just keep
the above in mind.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$write_pangenome(dir = "pangenome", force = FALSE)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>dir</code></b>: The unexisting directory name where to put the data files. Default
is &quot;pangenome&quot;.
</p>
</li>
<li><p><b><code>force</code></b>: <code>logical</code>. Whether to overwrite the directory if it already
exists. Default: <code>FALSE</code>.
</p>
</li></ul>




<h4>Return:</h4>

<p>A directory with at least 3 files. &quot;data.tsv&quot; contain the basic
pangenome data as it is provided to the <code>data</code> argument in the
initialization method (<code>$new(...)</code>). &quot;clusters.tsv&quot; contain any metadata
associated to the clusters. &quot;organisms.tsv&quot; contain any metadata associated to
the organisms. The latter 2 files will contain a single column if no metadata
was provided.
</p>

</dd>
</dl>
<a id="method-save_pangenomeRDS"></a>
<dl>
<dt><b>Save a pangenome as a RDS (binary) file.</b></dt><dd>


<h4>Description:</h4>

<p>Save a pagoo pangenome object. This function provides a method for saving a pagoo
object and its state into a &quot;RDS&quot; file. To load the pangenome, use the
<code>load_pangenomeRDS</code> function in this package. It *should* be compatible between
pagoo versions, so you could update pagoo and still recover the same pangenome. Even
<code>sep</code> and <code>core_level</code> are restored unless the user provides those
arguments in <code>load_pangenomeRDS</code>. <code>dropped</code> organisms also kept hidden, as
you where working with the original object.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$save_pangenomeRDS(file = "pangenome.rds")</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>file</code></b>: The name of the file to save. Default: &quot;pangenome.rds&quot;.
</p>
</li></ul>




<h4>Return:</h4>

<p>Writes a list with all the information needed to restore the object by
using the load_pangenomeRDS function, into an RDS (binary) file.
</p>

</dd>
</dl>
<a id="method-clone"></a>
<dl>
<dt><b>Clone a pagoo object.</b></dt><dd>


<h4>Description:</h4>

<p>The objects of this class are clonable with this method.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$clone(deep = FALSE)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>deep</code></b>: <code>character</code> identifying the metadata to map. Can
be one of <code>"org"</code>, <code>"group"</code>, or <code>"gid"</code>.
</p>
</li></ul>




<h4>Return:</h4>

<p>Whether to make a deep clone.
</p>

</dd>
</dl>
<a id="method-dist"></a>
<dl>
<dt><b>Compute distances</b></dt><dd>


<h4>Description:</h4>

<p>Compute distance between all pairs of genomes. The default dist method is
<code>"bray"</code> (Bray-Curtis distance). Another used distance method is <code>"jaccard"</code>,
but you should set <code>binary = FALSE</code> (see below) to obtain a meaningful result.
See <code><a href="vegan.html#topic+vegdist">vegdist</a></code> for details, this is just a wrapper function.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$dist(
                  method = "bray",
                  binary = FALSE,
                  diag = FALSE,
                  upper = FALSE,
                  na.rm = FALSE,
                  ...
                  )</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>method</code></b>: The distance method to use. See <a href="vegan.html#topic+vegdist">vegdist</a>
for available methods, and details for each one.
</p>
</li>
<li><p><b><code>binary</code></b>: Transform abundance matrix into a presence/absence
matrix before computing distance.
</p>
</li>
<li><p><b><code>diag</code></b>: Compute diagonals.
</p>
</li>
<li><p><b><code>upper</code></b>: Return only the upper diagonal.
</p>
</li>
<li><p><b><code>na.rm</code></b>: Pairwise deletion of missing observations when
computing dissimilarities.
</p>
</li>
<li><p><b><code>...</code></b>: Other parameters. See <a href="vegan.html#topic+vegdist">vegdist</a> for details.
</p>
</li></ul>




<h4>Return:</h4>

<p>A <code>dist</code> object containing all pairwise dissimilarities between genomes.
</p>

</dd>
</dl>
<a id="method-pan_pca"></a>
<dl>
<dt><b>Compute a Principal Component Analysis</b></dt><dd>


<h4>Description:</h4>

<p>Performs a principal components analysis on the panmatrix.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$pan_pca( center = TRUE, scale. = FALSE, ...)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>center</code></b>: a logical value indicating whether the variables should be shifted
to be zero centered. Alternately, a vector of length equal the number of columns of x can be
supplied. The value is passed to scale.
</p>
</li>
<li><p><b><code>scale.</code></b>: a logical value indicating whether the variables should be scaled
to have unit variance before the analysis takes place. The default is TRUE.
</p>
</li>
<li><p><b><code>...</code></b>: Other arguments. See <a href="stats.html#topic+prcomp">prcomp</a>
</p>
</li></ul>




<h4>Return:</h4>

<p>Returns a list with class &quot;prcomp&quot;. See <a href="stats.html#topic+prcomp">prcomp</a> for more information.
</p>

</dd>
</dl>
<a id="method-pg_power_law_fit"></a>
<dl>
<dt><b>Fit a Power Law Function for the Pangenome</b></dt><dd>


<h4>Description:</h4>

<p>Fits a power law curve for the pangenome rarefaction simulation.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$pg_power_law_fit(raref, ...)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>raref</code></b>: (Optional) A rarefaction matrix, as returned by <code>rarefact()</code>.
</p>
</li>
<li><p><b><code>...</code></b>: Further arguments to be passed to <code>rarefact()</code>. If <code>raref</code>
is missing, it will be computed with default arguments, or with the ones provided here.
</p>
</li></ul>




<h4>Return:</h4>

<p>A <code>list</code> of two elements: <code>$formula</code> with a fitted function, and <code>$params</code>
with fitted parameters. An attribute <code>"alpha"</code> is also returned (If
<code>alpha&gt;1</code>, then the pangenome is closed, otherwise is open.
</p>

</dd>
</dl>
<a id="method-cg_exp_decay_fit"></a>
<dl>
<dt><b>Fit an Exponential Decay Function for the Coregenome</b></dt><dd>


<h4>Description:</h4>

<p>Fits an exponential decay curve for the coregenome rarefaction simulation.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$cg_exp_decay_fit(raref, pcounts = 10, ...)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>raref</code></b>: (Optional) A rarefaction matrix, as returned by <code>rarefact()</code>.
</p>
</li>
<li><p><b><code>pcounts</code></b>: An integer of pseudo-counts. This is used to better fit the function
at small numbers, as the linearization method requires to subtract a constant C, which is the
coregenome size, from <code>y</code>. As <code>y</code> becomes closer to the coregenome size, this operation
tends to 0, and its logarithm goes crazy. By default <code>pcounts=10</code>.
</p>
</li>
<li><p><b><code>...</code></b>: Further arguments to be passed to <code>rarefact()</code>. If <code>raref</code>
is missing, it will be computed with default arguments, or with the ones provided here.
</p>
</li></ul>




<h4>Return:</h4>

<p>A <code>list</code> of two elements: <code>$formula</code> with a fitted function, and <code>$params</code>
with fitted intercept and decay parameters.
</p>

</dd>
</dl>
<a id="method-fluidity"></a>
<dl>
<dt><b>Compute Genomic Fluidity</b></dt><dd>


<h4>Description:</h4>

<p>Computes the genomic fluidity, which is a measure of population
diversity. See <code><a href="micropan.html#topic+fluidity">fluidity</a></code> for more details.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$fluidity(nsim = 10)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>nsim</code></b>:An integer specifying the number of random samples
to use in the computations.
</p>
</li></ul>




<h4>Return:</h4>

<p>A list with two elements, the mean fluidity and its sample standard
deviation over the n.sim computed values.
</p>

</dd>
</dl>
<a id="method-gg_barplot"></a>
<dl>
<dt><b>Plot Accessory Frequency Plot</b></dt><dd>


<h4>Description:</h4>

<p>Plot a barplot with the frequency of genes within the total number of
genomes.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$gg_barplot()</code>
</p>



<h4>Return:</h4>

<p>A barplot, and a <code>gg</code> object (<code>ggplot2</code> package) invisibly.
</p>

</dd>
</dl>
<a id="method-gg_dist"></a>
<dl>
<dt><b>Plot a Distance Heatmap</b></dt><dd>


<h4>Description:</h4>

<p>Plot a heatmap showing the computed distance between all pairs of organisms.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$gg_dist(method = "bray", ...)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>method</code></b>: Distance method. One of &quot;Jaccard&quot; (default), or
&quot;Manhattan&quot;,see above.
</p>
</li>
<li><p><b><code>...</code></b>: More arguments to be passed to <code><a href="micropan.html#topic+distManhattan">distManhattan</a></code>
</p>
</li></ul>




<h4>Return:</h4>

<p>A heatmap (<code>ggplot2::geom_tile()</code>), and a <code>gg</code> object (<code>ggplot2</code>
package) invisibly.
</p>

</dd>
</dl>
<a id="method-gg_binmap"></a>
<dl>
<dt><b>Plot a Pangenome Binary Map</b></dt><dd>


<h4>Description:</h4>

<p>Plot a pangenome binary map representing the presence/absence of each
gene within each organism.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$gg_binmap()</code>
</p>



<h4>Return:</h4>

<p>A binary map (<code>ggplot2::geom_raster()</code>), and a <code>gg</code> object (<code>ggplot2</code>
package) invisibly.
</p>

</dd>
</dl>
<a id="method-gg_pca"></a>
<dl>
<dt><b>Plot a PCA</b></dt><dd>


<h4>Description:</h4>

<p>Plot a scatter plot of a Principal Components Analysis.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$gg_pca(colour = NULL, ...))</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>colour</code></b>:The name of the column in <code>$organisms</code> field
from which points will take color (if provided). <code>NULL</code> (default) renders
black points.
</p>
</li>
<li><p><b><code>...</code></b>: More arguments to be passed to <code>ggplot2::autoplot()</code>.
</p>
</li></ul>




<h4>Return:</h4>

<p>A scatter plot (<code>ggplot2::autoplot()</code>), and a <code>gg</code> object
(<code>ggplot2</code> package) invisibly.
</p>

</dd>
</dl>
<a id="method-gg_pie"></a>
<dl>
<dt><b>Plot a Pie with Pangenome Categories</b></dt><dd>


<h4>Description:</h4>

<p>Plot a pie chart showing the number of clusters of each pangenome category: core,
shell, or cloud.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$gg_pie()</code>
</p>



<h4>Return:</h4>

<p>A pie chart (<code>ggplot2::geom_bar() + coord_polar()</code>), and a
<code>gg</code> object (<code>ggplot2</code> package) invisibly.
</p>

</dd>
</dl>
<a id="method-gg_curves"></a>
<dl>
<dt><b>Plot Pangenome Curves</b></dt><dd>


<h4>Description:</h4>

<p>Plot pangenome and/or coregenome curves with the fitted functions returned by
<code>pg_power_law_fit()</code> and <code>cg_exp_decay_fit()</code>. You can add points by
adding <code>+ geom_points()</code>, of ggplot2 package.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$gg_curves(what = c("pangenome", "coregenome", ...)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>what</code></b>:  <code>"pangenome"</code> and/or <code>"coregenome"</code>.
</p>
</li>
<li><p><b><code>...</code></b>: ignored
</p>
</li></ul>




<h4>Return:</h4>

<p>A scatter plot, and a <code>gg</code> object (<code>ggplot2</code> package) invisibly.
</p>

</dd>
</dl>
<a id="method-runShinyApp"></a>
<dl>
<dt><b>Run a Shiny App</b></dt><dd>


<h4>Description:</h4>

<p>Launch an interactive shiny app. It contains a sidebar
with controls and switches to interact with the pagoo
object. You can drop/recover organisms from the dataset,
modify the core_level, visualize statistics, plots, and
browse cluster and gene information. In the main body, it
contains 2 tabs to switch between summary statistics plots
and core genome information on one side, and accessory
genome plots and information on the other.
</p>
<p>The lower part of each tab contains two tables, side by
side. On the &quot;Summary&quot; tab, the left one contain
information about core clusters, with one cluster per row.
When one of them is selected (click), the one on the right
is updated to show information about its genes (if
provided), one gene per row. On the &quot;Accessory&quot; tab, a
similar configuration is shown, but on this case only
accessory clusters/genes are displayed. There is a slider
on the sidebar where one can select the accessory
frequency range to display.
</p>
<p>Give it a try!
</p>
<p>Take into account that big pangenomes can slow down the
performance of the app. More than 50-70 organisms often
leads to a delay in the update of the plots/tables.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$runShinyApp()</code>
</p>



<h4>Return:</h4>

<p>Opens a shiny app on the browser.
</p>

</dd>
</dl>
<a id="method-core_seqs_4_phylo"></a>
<dl>
<dt><b>Retrieve Core Genes for Phylogeny</b></dt><dd>


<h4>Description:</h4>

<p>A field for obtaining core gene sequences is available (see below),
but for creating a phylogeny with this sets is useful to: 1) have the
possibility of extracting just one sequence of each organism on each
cluster, in case paralogues are present, and 2) filling gaps with
empty sequences in case the core_level was set below 100%, allowing
more genes (some not in 100% of organisms) to be incorporated to the
phylogeny. That is the purpose of this special function.
</p>



<h4>Usage:</h4>

<p><code style="white-space: pre;">&#8288;                  &#8288;</code><code>$core_seqs_4_phylo(max_per_org = 1, fill = TRUE)</code>
</p>



<h4>Arguments:</h4>


<ul>
<li><p><b><code>max_per_org</code></b>:  Maximum number of sequences of each organism
to be taken from each cluster.
</p>
</li>
<li><p><b><code>fill</code></b>: <code>logical</code>. If fill <code>DNAStringSet</code> with
empty <code>DNAString</code> in cases where <code>core_level</code> is set below 100%,
and some clusters with missing organisms are also considered.
</p>
</li></ul>




<h4>Return:</h4>

<p>A <code>DNAStringSetList</code> with core genes. Order of organisms on each cluster
is conserved, so it is easier to concatenate them into a super-gene suitable
for phylogenetic inference.
</p>

</dd>
</dl>


<hr>
<h2 id='panaroo_2_pagoo'>Read panaroos's output into a pagoo's R6 class object</h2><span id='topic+panaroo_2_pagoo'></span>

<h3>Description</h3>

<p>This function handle conversion of <a href="https://gtonkinhill.github.io/panaroo/#/">panaroo</a>'s output files into
a pagoo R6 class object. It takes the &quot;gene_presence_absence.csv&quot; file and
(optionally but recommended) gff input file paths, and returns an object of
class <code><a href="#topic+PgR6MS">PgR6MS</a></code> (or <code><a href="#topic+PgR6M">PgR6M</a></code> if left
empty the <code>gffs</code> argument). Panaroo identifies some genes with unusual
lengths tagging them with 'stop', 'length', or 'refound' labels. In the
current version, this function discards those genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>panaroo_2_pagoo(gene_presence_absence_csv, gffs, sep = "__")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="panaroo_2_pagoo_+3A_gene_presence_absence_csv">gene_presence_absence_csv</code></td>
<td>
<p><code>character</code>, path to the
&quot;gene_presence_absence.csv&quot; file. (Do not confuse with the file with the
same name but with <code>.Rtab</code> extension).</p>
</td></tr>
<tr><td><code id="panaroo_2_pagoo_+3A_gffs">gffs</code></td>
<td>
<p>A <code>character</code> vector with paths to original gff files used
as roary's input. Typically the return value of <code>list.files()</code> function.
This parameter is optional but highly recommended if you want to manipulate
sequences.</p>
</td></tr>
<tr><td><code id="panaroo_2_pagoo_+3A_sep">sep</code></td>
<td>
<p><code>character</code>. Default: <code>"__"</code> (two underscores). See
<a href="#topic+PgR6MS">PgR6MS</a> for a more detail argument description.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A pagoo's R6 class object. Either <a href="#topic+PgR6M">PgR6M</a>, if <code>gffs</code>
argument is left empty, or <a href="#topic+PgR6MS">PgR6MS</a> if path to gff files is
provided.
</p>


<h3>References</h3>

<p>Tonkin-Hill, G., MacAlasdair, N., Ruis, C. et al. Producing polished
prokaryotic pangenomes with the Panaroo pipeline. Genome Biol 21, 180 (2020).
https://doi.org/10.1186/s13059-020-02090-4
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gffs &lt;- list.files(path = "path/to/gffs/",
                   pattern = "[.]gff$",
                   full.names = TRUE)
gpa_csv &lt;- "path/to/gene_presence_absence.csv"

library(pagoo)
pg &lt;- panaroo_2_pagoo(gene_presence_absence_csv = gpa_csv,
                    gffs = gffs)

## End(Not run)
</code></pre>

<hr>
<h2 id='PgR6'>PgR6 basic class</h2><span id='topic+PgR6'></span>

<h3>Description</h3>

<p>A basic <code>PgR6</code> class constructor. It contains basic fields
and subset functions to handle a pangenome. Final users should use <code><a href="#topic+pagoo">pagoo</a></code>
instead of this, since is more easy to understand.
</p>


<h3>Active bindings</h3>

<div class="r6-active-bindings">

<dl>
<dt><code>pan_matrix</code></dt><dd><p>The panmatrix. Rows are organisms, and
columns are groups of orthologous. Cells indicates the presence (&gt;=1) or
absence (0) of a given gene, in a given organism. Cells can have values
greater than 1 if contain in-paralogs.</p>
</dd>
<dt><code>organisms</code></dt><dd><p>A <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> with available
organism names, and organism number identifier as <code>rownames()</code>. (Dropped
organisms will not be displayed in this field, see <code>$dropped</code> below).
Additional metadata will be shown if provided, as additional columns.</p>
</dd>
<dt><code>genes</code></dt><dd><p>A <code><a href="IRanges.html#topic+DataFrameList-class">SplitDataFrameList</a></code> object with
one entry per cluster. Each element contains a <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code>
with gene ids (<code>&lt;gid&gt;</code>) and additional metadata, if provided. <code>gid</code> are
created by <code>paste</code>ing organism and gene names, so duplication in gene names
are avoided.</p>
</dd>
<dt><code>clusters</code></dt><dd><p>A <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> with the groups
of orthologous (clusters). Additional metadata will be shown as additional columns,
if provided before. Each row corresponds to each cluster.</p>
</dd>
<dt><code>core_level</code></dt><dd><p>The percentage of organisms a gene must be in
to be considered as part of the coregenome. <code>core_level = 95</code> by default.
Can't be set above 100, and below 85 raises a warning.</p>
</dd>
<dt><code>core_genes</code></dt><dd><p>Like <code>genes</code>, but only showing core genes.</p>
</dd>
<dt><code>core_clusters</code></dt><dd><p>Like <code>$clusters</code>, but only showing core
clusters.</p>
</dd>
<dt><code>cloud_genes</code></dt><dd><p>Like <code>genes</code>, but only showing cloud genes.
These are defined as those clusters which contain a single gene (singletons), plus
those which have more than one but its organisms are probably clonal due to identical
general gene content. Colloquially defined as strain-specific genes.</p>
</dd>
<dt><code>cloud_clusters</code></dt><dd><p>Like <code>$clusters</code>, but only showing cloud
clusters as defined above.</p>
</dd>
<dt><code>shell_genes</code></dt><dd><p>Like <code>genes</code>, but only showing shell genes.
These are defined as those clusters than don't belong neither to the core genome,
nor to cloud genome. Colloquially defined as genes that are present in some but not
all strains, and that aren't strain-specific.</p>
</dd>
<dt><code>shell_clusters</code></dt><dd><p>Like <code>$clusters</code>, but only showing shell
clusters, as defined above.</p>
</dd>
<dt><code>summary_stats</code></dt><dd><p>A <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> with
information about the number of core, shell, and cloud clusters, as well as the
total number of clusters.</p>
</dd>
<dt><code>random_seed</code></dt><dd><p>The last <code>.Random.seed</code>. Used for
reproducibility purposes only.</p>
</dd>
<dt><code>dropped</code></dt><dd><p>A <code>character</code> vector with dropped organism
names, and organism number identifier as <code>names()</code></p>
</dd>
</dl>

</div>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-PgR6-new"><code>PgR6$new()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6-add_metadata"><code>PgR6$add_metadata()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6-drop"><code>PgR6$drop()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6-recover"><code>PgR6$recover()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6-write_pangenome"><code>PgR6$write_pangenome()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6-save_pangenomeRDS"><code>PgR6$save_pangenomeRDS()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6-clone"><code>PgR6$clone()</code></a>
</p>
</li></ul>


<hr>
<a id="method-PgR6-new"></a>



<h4>Method <code>new()</code></h4>

<p>A basic <code>PgR6</code> class constructor. It contains basic fields
and subset functions to handle a pangenome.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6$new(
  data,
  org_meta,
  cluster_meta,
  core_level = 95,
  sep = "__",
  verbose = TRUE,
  DF,
  group_meta
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A <code>data.frame</code> or <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> containing at least the
following columns: <code>gene</code> (gene name), <code>org</code> (organism name to which the gene belongs to),
and <code>cluster</code> (group of orthologous to which the gene belongs to). More columns can be added as metadata
for each gene.</p>
</dd>
<dt><code>org_meta</code></dt><dd><p>(optional) A <code>data.frame</code> or <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code>
containing additional metadata for organisms. This <code>data.frame</code> must have a column named &quot;org&quot; with
valid organisms names (that is, they should match with those provided in <code>data</code>, column <code>org</code>), and
additional columns will be used as metadata. Each row should correspond to each organism.</p>
</dd>
<dt><code>cluster_meta</code></dt><dd><p>(optional) A <code>data.frame</code> or <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code>
containing additional metadata for clusters. This <code>data.frame</code> must have a column named &quot;cluster&quot; with
valid organisms names (that is, they should match with those provided in <code>data</code>, column <code>cluster</code>), and
additional columns will be used as metadata. Each row should correspond to each cluster.</p>
</dd>
<dt><code>core_level</code></dt><dd><p>The initial core_level (that's the percentage of organisms a core cluster must be in to be
considered as part of the core genome). Must be a number between 100 and 85, (default: 95). You can change it
later by using the <code>$core_level</code> field once the object was created.</p>
</dd>
<dt><code>sep</code></dt><dd><p>A separator. By default is '__'(two underscores). It will be used to
create a unique <code>gid</code> (gene identifier) for each gene. <code>gid</code>s are created by pasting
<code>org</code> to <code>gene</code>, separated by <code>sep</code>.</p>
</dd>
<dt><code>verbose</code></dt><dd><p><code>logical</code>. Whether to display progress messages when loading class.</p>
</dd>
<dt><code>DF</code></dt><dd><p>Deprecated. Use <code>data</code> instead.</p>
</dd>
<dt><code>group_meta</code></dt><dd><p>Deprecated. Use <code>cluster_meta</code> instead.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>An R6 object of class PgR6. It contains basic fields and methods for
analyzing a pangenome.
</p>


<hr>
<a id="method-PgR6-add_metadata"></a>



<h4>Method <code>add_metadata()</code></h4>

<p>Add metadata to the object. You can add metadata to each organism, to each
group of orthologous (cluster), or to each gene. Elements with missing data should be filled
by <code>NA</code> (dimensions of the provided data.frame must be coherent with object
data).
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6$add_metadata(map = "org", data)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>map</code></dt><dd><p><code>character</code> identifying the metadata to map. Can
be one of <code>"org"</code>, <code>"cluster"</code>, or <code>"gid"</code>.</p>
</dd>
<dt><code>data</code></dt><dd><p><code>data.frame</code> or <code>DataFrame</code> with the metadata to
add. For each case, a column named as <code>"map"</code> must exists, which should
contain identifiers for each element. In the case of adding gene (<code>gid</code>)
metadata,each gene should be referenced by the name of the organism and the name
of the gene as provided in the <code>"data"</code> data.frame, separated by the
<code>"sep"</code> argument.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p><code>self</code> invisibly, but with additional metadata.
</p>


<hr>
<a id="method-PgR6-drop"></a>



<h4>Method <code>drop()</code></h4>

<p>Drop an organism from the dataset. This method allows to hide an organism
from the real dataset, ignoring it in downstream analyses. All the fields and
methods will behave as it doesn't exist. For instance, if you decide to drop
organism 1, the <code>$pan_matrix</code> field (see below) would not show it when
called.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6$drop(x)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>x</code></dt><dd><p><code>character</code> or <code>numeric</code>. The name of the
organism wanted to be dropped, or its numeric id as returned in
<code>$organism</code> field (see below).</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p><code>self</code> invisibly, but with <code>x</code> dropped. It isn't necessary
to assign the function call to a new object, nor to re-write it as R6 objects
are mutable.
</p>


<hr>
<a id="method-PgR6-recover"></a>



<h4>Method <code>recover()</code></h4>

<p>Recover a previously <code>$drop()</code>ped organism (see above). All fields
and methods will start to behave considering this organism again.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6$recover(x)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>x</code></dt><dd><p><code>character</code> or <code>numeric</code>. The name of the
organism wanted to be recover, or its numeric id as returned in
<code>$dropped</code> field (see below).</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p><code>self</code> invisibly, but with <code>x</code> recovered. It isn't necessary
to assign the function call to a new object, nor to re-write it as R6 objects
are mutable.
</p>


<hr>
<a id="method-PgR6-write_pangenome"></a>



<h4>Method <code>write_pangenome()</code></h4>

<p>Write the pangenome data as flat tables (text). Is not the most recommended way
to save a pangenome, since you can loose information as numeric precision,
column classes (factor, numeric, integer), and the state of the object itself
(i.e. dropped organisms, or core_level), loosing reproducibility. Use
<code>$save_pangenomeRDS</code> for a more precise way of saving a pagoo object.
Still, it is useful if you want to work with the data outside R, just keep
the above in mind.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6$write_pangenome(dir = "pangenome", force = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>dir</code></dt><dd><p>The non-existing directory name where to put the data files. Default
is &quot;pangenome&quot;.</p>
</dd>
<dt><code>force</code></dt><dd><p><code>logical</code>. Whether to overwrite the directory if it already
exists. Default: <code>FALSE</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A directory with at least 3 files. &quot;data.tsv&quot; contain the basic
pangenome data as it is provided to the <code>data</code> argument in the
initialization method (<code>$new(...)</code>). &quot;clusters.tsv&quot; contain any metadata
associated to the clusters. &quot;organisms.tsv&quot; contain any metadata associated to
the organisms. The latter 2 files will contain a single column if no metadata
was provided.
</p>


<hr>
<a id="method-PgR6-save_pangenomeRDS"></a>



<h4>Method <code>save_pangenomeRDS()</code></h4>

<p>Save a pagoo pangenome object. This function provides a method for saving a pagoo
object and its state into a &quot;RDS&quot; file. To load the pangenome, use the
<code>load_pangenomeRDS</code> function in this package. It *should* be compatible between
pagoo versions, so you could update pagoo and still recover the same pangenome. Even
<code>sep</code> and <code>core_level</code> are restored unless the user provides those
arguments in <code>load_pangenomeRDS</code>. <code>dropped</code> organisms also kept hidden, as
you where working with the original object.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6$save_pangenomeRDS(file = "pangenome.rds")</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>file</code></dt><dd><p>The name of the file to save. Default: &quot;pangenome.rds&quot;.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>Writes a list with all the information needed to restore the object by
using the load_pangenomeRDS function, into an RDS (binary) file.
</p>


<hr>
<a id="method-PgR6-clone"></a>



<h4>Method <code>clone()</code></h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt><dd><p>Whether to make a deep clone.</p>
</dd>
</dl>

</div>



<hr>
<h2 id='PgR6M'>PgR6 class with methods.</h2><span id='topic+PgR6M'></span>

<h3>Description</h3>

<p>PgR6 with Methods. Final users should use <code><a href="#topic+pagoo">pagoo</a></code>
instead of this, since is more easy to understand.
Inherits: <code><a href="#topic+PgR6">PgR6</a></code>
</p>


<h3>Super class</h3>

<p><code><a href="#topic+PgR6">pagoo::PgR6</a></code> -&gt; <code>PgR6M</code>
</p>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-PgR6M-new"><code>PgR6M$new()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-rarefact"><code>PgR6M$rarefact()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-dist"><code>PgR6M$dist()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-pan_pca"><code>PgR6M$pan_pca()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-pg_power_law_fit"><code>PgR6M$pg_power_law_fit()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-cg_exp_decay_fit"><code>PgR6M$cg_exp_decay_fit()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-gg_barplot"><code>PgR6M$gg_barplot()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-gg_binmap"><code>PgR6M$gg_binmap()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-gg_dist"><code>PgR6M$gg_dist()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-gg_pca"><code>PgR6M$gg_pca()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-gg_pie"><code>PgR6M$gg_pie()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-gg_curves"><code>PgR6M$gg_curves()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-runShinyApp"><code>PgR6M$runShinyApp()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6M-clone"><code>PgR6M$clone()</code></a>
</p>
</li></ul>



<details open><summary>Inherited methods</summary>
<ul>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="add_metadata"><a href='../../pagoo/html/PgR6.html#method-PgR6-add_metadata'><code>pagoo::PgR6$add_metadata()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="drop"><a href='../../pagoo/html/PgR6.html#method-PgR6-drop'><code>pagoo::PgR6$drop()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="recover"><a href='../../pagoo/html/PgR6.html#method-PgR6-recover'><code>pagoo::PgR6$recover()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="save_pangenomeRDS"><a href='../../pagoo/html/PgR6.html#method-PgR6-save_pangenomeRDS'><code>pagoo::PgR6$save_pangenomeRDS()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="write_pangenome"><a href='../../pagoo/html/PgR6.html#method-PgR6-write_pangenome'><code>pagoo::PgR6$write_pangenome()</code></a></span></li>
</ul>
</details>

<hr>
<a id="method-PgR6M-new"></a>



<h4>Method <code>new()</code></h4>

<p>Create a <code>PgR6M</code> object.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$new(
  data,
  org_meta,
  cluster_meta,
  core_level = 95,
  sep = "__",
  verbose = TRUE,
  DF,
  group_meta
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A <code>data.frame</code> or <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> containing at least the
following columns: <code>gene</code> (gene name), <code>org</code> (organism name to which the gene belongs to),
and <code>cluster</code> (group of orthologous to which the gene belongs to). More columns can be added as metadata
for each gene.</p>
</dd>
<dt><code>org_meta</code></dt><dd><p>(optional) A <code>data.frame</code> or <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code>
containing additional metadata for organisms. This <code>data.frame</code> must have a column named &quot;org&quot; with
valid organisms names (that is, they should match with those provided in <code>data</code>, column <code>org</code>), and
additional columns will be used as metadata. Each row should correspond to each organism.</p>
</dd>
<dt><code>cluster_meta</code></dt><dd><p>(optional) A <code>data.frame</code> or <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code>
containing additional metadata for clusters. This <code>data.frame</code> must have a column named &quot;cluster&quot; with
valid organisms names (that is, they should match with those provided in <code>data</code>, column <code>cluster</code>), and
additional columns will be used as metadata. Each row should correspond to each cluster.</p>
</dd>
<dt><code>core_level</code></dt><dd><p>The initial core_level (that's the percentage of organisms a core cluster must be in to be
considered as part of the core genome). Must be a number between 100 and 85, (default: 95). You can change it
later by using the <code>$core_level</code> field once the object was created.</p>
</dd>
<dt><code>sep</code></dt><dd><p>A separator. By default is '__'(two underscores). It will be used to
create a unique <code>gid</code> (gene identifier) for each gene. <code>gid</code>s are created by pasting
<code>org</code> to <code>gene</code>, separated by <code>sep</code>.</p>
</dd>
<dt><code>verbose</code></dt><dd><p><code>logical</code>. Whether to display progress messages when loading class.</p>
</dd>
<dt><code>DF</code></dt><dd><p>Deprecated. Use <code>data</code> instead.</p>
</dd>
<dt><code>group_meta</code></dt><dd><p>Deprecated. Use <code>cluster_meta</code> instead.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>An R6 object of class PgR6M. It contains basic fields and methods for analyzing a pangenome. It also
contains additional statistical methods for analyze it, and methods to make basic
exploratory plots.
</p>


<hr>
<a id="method-PgR6M-rarefact"></a>



<h4>Method <code>rarefact()</code></h4>

<p>Rarefact pangenome or corgenome. Compute the number of genes which belong to
the pangenome or to the coregenome, for a number of random permutations of
increasingly bigger sample of genomes.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$rarefact(what = "pangenome", n.perm = 10)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>what</code></dt><dd><p>One of <code>"pangenome"</code> or <code>"coregenome"</code>.</p>
</dd>
<dt><code>n.perm</code></dt><dd><p>The number of permutations to compute (default: 10).</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A <code>matrix</code>, rows are the number of genomes added, columns are
permutations, and the cell number is the number of genes in each category.
</p>


<hr>
<a id="method-PgR6M-dist"></a>



<h4>Method <code>dist()</code></h4>

<p>Compute distance between all pairs of genomes. The default dist method is
<code>"bray"</code> (Bray-Curtis distance). Another used distance method is <code>"jaccard"</code>,
but you should set <code>binary = FALSE</code> (see below) to obtain a meaningful result.
See <code><a href="vegan.html#topic+vegdist">vegdist</a></code> for details, this is just a wrapper function.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$dist(
  method = "bray",
  binary = FALSE,
  diag = FALSE,
  upper = FALSE,
  na.rm = FALSE,
  ...
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>method</code></dt><dd><p>The distance method to use. See <a href="vegan.html#topic+vegdist">vegdist</a>
for available methods, and details for each one.</p>
</dd>
<dt><code>binary</code></dt><dd><p>Transform abundance matrix into a presence/absence
matrix before computing distance.</p>
</dd>
<dt><code>diag</code></dt><dd><p>Compute diagonals.</p>
</dd>
<dt><code>upper</code></dt><dd><p>Return only the upper diagonal.</p>
</dd>
<dt><code>na.rm</code></dt><dd><p>Pairwise deletion of missing observations when
computing dissimilarities.</p>
</dd>
<dt><code>...</code></dt><dd><p>Other parameters. See <a href="vegan.html#topic+vegdist">vegdist</a> for details.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A <code>dist</code> object containing all pairwise dissimilarities between genomes.
</p>


<hr>
<a id="method-PgR6M-pan_pca"></a>



<h4>Method <code>pan_pca()</code></h4>

<p>Performs a principal components analysis on the panmatrix
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$pan_pca(center = TRUE, scale. = FALSE, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>center</code></dt><dd><p>a logical value indicating whether the variables should be shifted
to be zero centered. Alternately, a vector of length equal the number of columns of x can be
supplied. The value is passed to scale.</p>
</dd>
<dt><code>scale.</code></dt><dd><p>a logical value indicating whether the variables should be scaled
to have unit variance before the analysis takes place. The default is TRUE.</p>
</dd>
<dt><code>...</code></dt><dd><p>Other arguments. See <a href="stats.html#topic+prcomp">prcomp</a></p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>Returns a list with class &quot;prcomp&quot;. See <a href="stats.html#topic+prcomp">prcomp</a> for more information.
</p>


<hr>
<a id="method-PgR6M-pg_power_law_fit"></a>



<h4>Method <code>pg_power_law_fit()</code></h4>

<p>Fits a power law curve for the pangenome rarefaction simulation.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$pg_power_law_fit(raref, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>raref</code></dt><dd><p>(Optional) A rarefaction matrix, as returned by <code>rarefact()</code>.</p>
</dd>
<dt><code>...</code></dt><dd><p>Further arguments to be passed to <code>rarefact()</code>. If <code>raref</code>
is missing, it will be computed with default arguments, or with the ones provided here.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A <code>list</code> of two elements: <code>$formula</code> with a fitted function, and <code>$params</code>
with fitted parameters. An attribute <code>"alpha"</code> is also returned (If
<code>alpha&gt;1</code>, then the pangenome is closed, otherwise is open.
</p>


<hr>
<a id="method-PgR6M-cg_exp_decay_fit"></a>



<h4>Method <code>cg_exp_decay_fit()</code></h4>

<p>Fits an exponential decay curve for the coregenome rarefaction simulation.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$cg_exp_decay_fit(raref, pcounts = 10, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>raref</code></dt><dd><p>(Optional) A rarefaction matrix, as returned by <code>rarefact()</code>.</p>
</dd>
<dt><code>pcounts</code></dt><dd><p>An integer of pseudo-counts. This is used to better fit the function
at small numbers, as the linearization method requires to subtract a constant C, which is the
coregenome size, from <code>y</code>. As <code>y</code> becomes closer to the coregenome size, this operation
tends to 0, and its logarithm goes crazy. By default <code>pcounts=10</code>.</p>
</dd>
<dt><code>...</code></dt><dd><p>Further arguments to be passed to <code>rarefact()</code>. If <code>raref</code>
is missing, it will be computed with default arguments, or with the ones provided here.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A <code>list</code> of two elements: <code>$formula</code> with a fitted function, and <code>$params</code>
with fitted intercept and decay parameters.
</p>


<hr>
<a id="method-PgR6M-gg_barplot"></a>



<h4>Method <code>gg_barplot()</code></h4>

<p>Plot a barplot with the frequency of genes within the total number of
genomes.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$gg_barplot()</pre></div>



<h5>Returns</h5>

<p>A barplot, and a <code>gg</code> object (<code>ggplot2</code> package) invisibly.
</p>


<hr>
<a id="method-PgR6M-gg_binmap"></a>



<h4>Method <code>gg_binmap()</code></h4>

<p>Plot a pangenome binary map representing the presence/absence of each
gene within each organism.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$gg_binmap()</pre></div>



<h5>Returns</h5>

<p>A binary map (<code>ggplot2::geom_raster()</code>), and a <code>gg</code> object (<code>ggplot2</code>
package) invisibly.
</p>


<hr>
<a id="method-PgR6M-gg_dist"></a>



<h4>Method <code>gg_dist()</code></h4>

<p>Plot a heatmap showing the computed distance between all pairs of organisms.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$gg_dist(method = "bray", ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>method</code></dt><dd><p>Distance method. One of &quot;Jaccard&quot; (default), or &quot;Manhattan&quot;,
see above.</p>
</dd>
<dt><code>...</code></dt><dd><p>More arguments to be passed to <code><a href="micropan.html#topic+distManhattan">distManhattan</a></code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A heatmap (<code>ggplot2::geom_tile()</code>), and a <code>gg</code> object (<code>ggplot2</code>
package) invisibly.
</p>


<hr>
<a id="method-PgR6M-gg_pca"></a>



<h4>Method <code>gg_pca()</code></h4>

<p>Plot a scatter plot of a Principal Components Analysis.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$gg_pca(colour = NULL, ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>colour</code></dt><dd><p>The name of the column in <code>$organisms</code> field from which points will take
colour (if provided). <code>NULL</code> (default) renders black points.</p>
</dd>
<dt><code>...</code></dt><dd><p>More arguments to be passed to <code>ggplot2::autoplot()</code>.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A scatter plot (<code>ggplot2::autoplot()</code>), and a <code>gg</code> object (<code>ggplot2</code>
package) invisibly.
</p>


<hr>
<a id="method-PgR6M-gg_pie"></a>



<h4>Method <code>gg_pie()</code></h4>

<p>Plot a pie chart showing the number of clusters of each pangenome category: core,
shell, or cloud.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$gg_pie()</pre></div>



<h5>Returns</h5>

<p>A pie chart (<code>ggplot2::geom_bar() + coord_polar()</code>), and a <code>gg</code> object
(<code>ggplot2</code> package) invisibly.
</p>


<hr>
<a id="method-PgR6M-gg_curves"></a>



<h4>Method <code>gg_curves()</code></h4>

<p>Plot pangenome and/or coregenome curves with the fitted functions returned by <code>pg_power_law_fit()</code>
and <code>cg_exp_decay_fit()</code>. You can add points by adding <code>+ geom_points()</code>, of ggplot2 package
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$gg_curves(what = c("pangenome", "coregenome"), ...)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>what</code></dt><dd><p>One of <code>"pangenome"</code> or <code>"coregenome"</code>.</p>
</dd>
<dt><code>...</code></dt><dd><p>????</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A scatter plot, and a <code>gg</code> object (<code>ggplot2</code> package) invisibly.
</p>


<hr>
<a id="method-PgR6M-runShinyApp"></a>



<h4>Method <code>runShinyApp()</code></h4>

<p>Launch an interactive shiny app. It contains a sidebar
with controls and switches to interact with the pagoo
object. You can drop/recover organisms from the dataset,
modify the core_level, visualize statistics, plots, and
browse cluster and gene information. In the main body, it
contains 2 tabs to switch between summary statistics plots
and core genome information on one side, and accessory
genome plots and information on the other.
</p>
<p>The lower part of each tab contains two tables, side by
side. On the &quot;Summary&quot; tab, the left one contain
information about core clusters, with one cluster per row.
When one of them is selected (click), the one on the right
is updated to show information about its genes (if
provided), one gene per row. On the &quot;Accessory&quot; tab, a
similar configuration is shown, but on this case only
accessory clusters/genes are displayed. There is a slider
on the sidebar where one can select the accessory
frequency range to display.
</p>
<p>Give it a try!
</p>
<p>Take into account that big pangenomes can slow down the
performance of the app. More than 50-70 organisms often
leads to a delay in the update of the plots/tables.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$runShinyApp()</pre></div>


<hr>
<a id="method-PgR6M-clone"></a>



<h4>Method <code>clone()</code></h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6M$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt><dd><p>Whether to make a deep clone.</p>
</dd>
</dl>

</div>



<hr>
<h2 id='PgR6MS'>PgR6 class with Methods and Sequences.</h2><span id='topic+PgR6MS'></span>

<h3>Description</h3>

<p>PgR6 with Methods and Sequences. Final users should use <code><a href="#topic+pagoo">pagoo</a></code>
instead of this, since is more easy to understand.
Inherits: <code><a href="#topic+PgR6M">PgR6M</a></code>
</p>


<h3>Super classes</h3>

<p><code><a href="#topic+PgR6">pagoo::PgR6</a></code> -&gt; <code><a href="#topic+PgR6M">pagoo::PgR6M</a></code> -&gt; <code>PgR6MS</code>
</p>


<h3>Active bindings</h3>

<div class="r6-active-bindings">

<dl>
<dt><code>sequences</code></dt><dd><p>A <code><a href="Biostrings.html#topic+XStringSetList-class">DNAStringSetList</a></code> with the
set of sequences grouped by cluster. Each group is accessible as were a list. All
<code>Biostrings</code> methods are available.</p>
</dd>
<dt><code>core_sequences</code></dt><dd><p>Like <code>$sequences</code>, but only showing core
sequences.</p>
</dd>
<dt><code>cloud_sequences</code></dt><dd><p>Like <code>$sequences</code>, but only showing cloud
sequences as defined above.</p>
</dd>
<dt><code>shell_sequences</code></dt><dd><p>Like <code>$sequences</code>, but only showing shell
sequences, as defined above.</p>
</dd>
</dl>

</div>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-PgR6MS-new"><code>PgR6MS$new()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6MS-core_seqs_4_phylo"><code>PgR6MS$core_seqs_4_phylo()</code></a>
</p>
</li>
<li> <p><a href="#method-PgR6MS-clone"><code>PgR6MS$clone()</code></a>
</p>
</li></ul>



<details><summary>Inherited methods</summary>
<ul>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="add_metadata"><a href='../../pagoo/html/PgR6.html#method-PgR6-add_metadata'><code>pagoo::PgR6$add_metadata()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="drop"><a href='../../pagoo/html/PgR6.html#method-PgR6-drop'><code>pagoo::PgR6$drop()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="recover"><a href='../../pagoo/html/PgR6.html#method-PgR6-recover'><code>pagoo::PgR6$recover()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="save_pangenomeRDS"><a href='../../pagoo/html/PgR6.html#method-PgR6-save_pangenomeRDS'><code>pagoo::PgR6$save_pangenomeRDS()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6" data-id="write_pangenome"><a href='../../pagoo/html/PgR6.html#method-PgR6-write_pangenome'><code>pagoo::PgR6$write_pangenome()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="cg_exp_decay_fit"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-cg_exp_decay_fit'><code>pagoo::PgR6M$cg_exp_decay_fit()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="dist"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-dist'><code>pagoo::PgR6M$dist()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_barplot"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-gg_barplot'><code>pagoo::PgR6M$gg_barplot()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_binmap"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-gg_binmap'><code>pagoo::PgR6M$gg_binmap()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_curves"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-gg_curves'><code>pagoo::PgR6M$gg_curves()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_dist"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-gg_dist'><code>pagoo::PgR6M$gg_dist()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_pca"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-gg_pca'><code>pagoo::PgR6M$gg_pca()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="gg_pie"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-gg_pie'><code>pagoo::PgR6M$gg_pie()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="pan_pca"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-pan_pca'><code>pagoo::PgR6M$pan_pca()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="pg_power_law_fit"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-pg_power_law_fit'><code>pagoo::PgR6M$pg_power_law_fit()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="rarefact"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-rarefact'><code>pagoo::PgR6M$rarefact()</code></a></span></li>
<li><span class="pkg-link" data-pkg="pagoo" data-topic="PgR6M" data-id="runShinyApp"><a href='../../pagoo/html/PgR6M.html#method-PgR6M-runShinyApp'><code>pagoo::PgR6M$runShinyApp()</code></a></span></li>
</ul>
</details>

<hr>
<a id="method-PgR6MS-new"></a>



<h4>Method <code>new()</code></h4>

<p>Create a <code>PgR6MS</code> object.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6MS$new(
  data,
  org_meta,
  cluster_meta,
  core_level = 95,
  sep = "__",
  DF,
  group_meta,
  sequences,
  verbose = TRUE
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>data</code></dt><dd><p>A <code>data.frame</code> or <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code> containing at least the
following columns: <code>gene</code> (gene name), <code>org</code> (organism name to which the gene belongs to),
and <code>cluster</code> (group of orthologous to which the gene belongs to). More columns can be added as metadata
for each gene.</p>
</dd>
<dt><code>org_meta</code></dt><dd><p>(optional) A <code>data.frame</code> or <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code>
containing additional metadata for organisms. This <code>data.frame</code> must have a column named &quot;org&quot; with
valid organisms names (that is, they should match with those provided in <code>data</code>, column <code>org</code>), and
additional columns will be used as metadata. Each row should correspond to each organism.</p>
</dd>
<dt><code>cluster_meta</code></dt><dd><p>(optional) A <code>data.frame</code> or <code><a href="S4Vectors.html#topic+DataFrame-class">DataFrame</a></code>
containing additional metadata for clusters. This <code>data.frame</code> must have a column named &quot;cluster&quot; with
valid organisms names (that is, they should match with those provided in <code>data</code>, column <code>cluster</code>), and
additional columns will be used as metadata. Each row should correspond to each cluster.</p>
</dd>
<dt><code>core_level</code></dt><dd><p>The initial core_level (that's the percentage of organisms a core cluster must be in to be
considered as part of the core genome). Must be a number between 100 and 85, (default: 95). You can change it
later by using the <code>$core_level</code> field once the object was created.</p>
</dd>
<dt><code>sep</code></dt><dd><p>A separator. By default is '__'(two underscores). It will be used to
create a unique <code>gid</code> (gene identifier) for each gene. <code>gid</code>s are created by pasting
<code>org</code> to <code>gene</code>, separated by <code>sep</code>.</p>
</dd>
<dt><code>DF</code></dt><dd><p>Deprecated. Use <code>data</code> instead.</p>
</dd>
<dt><code>group_meta</code></dt><dd><p>Deprecated. Use <code>cluster_meta</code> instead.</p>
</dd>
<dt><code>sequences</code></dt><dd><p>Can accept: 1) a named <code>list</code> of named <code>character</code> vector. Name of list
are names of organisms, names of character vector are gene names; or 2) a named <code>list</code> of <code><a href="Biostrings.html#topic+XStringSet-class">DNAStringSetList</a></code>
objects (same requirements as (1), but with BStringSet names as gene names); or 3) a <code><a href="Biostrings.html#topic+XStringSetList-class">DNAStringSetList</a></code>
(same requirements as (2) but <code>DNAStringSetList</code> names are organisms names).</p>
</dd>
<dt><code>verbose</code></dt><dd><p><code>logical</code>. Whether to display progress messages when loading class.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>An R6 object of class PgR6MS. It contains basic fields and methods for analyzing a pangenome. It also
contains additional statistical methods for analyze it, methods to make basic exploratory plots, and methods
for sequence manipulation.
</p>


<hr>
<a id="method-PgR6MS-core_seqs_4_phylo"></a>



<h4>Method <code>core_seqs_4_phylo()</code></h4>

<p>A field for obtaining core gene sequences is available (see below), but for creating a phylogeny with this
sets is useful to: 1) have the possibility of extracting just one sequence of each organism on each cluster, in
case paralogues are present, and 2) filling gaps with empty sequences in case the core_level was set below 100%,
allowing more genes (some not in 100% of organisms) to be incorporated to the phylogeny. That is the purpose of
this special function.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6MS$core_seqs_4_phylo(max_per_org = 1, fill = TRUE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>max_per_org</code></dt><dd><p>Maximum number of sequences of each organism to be taken from each cluster.</p>
</dd>
<dt><code>fill</code></dt><dd><p><code>logical</code>. If fill <code>DNAStringSet</code> with empty <code>DNAString</code> in cases where
<code>core_level</code> is set below 100%, and some clusters with missing organisms are also considered.</p>
</dd>
</dl>

</div>



<h5>Returns</h5>

<p>A <code>DNAStringSetList</code> with core genes. Order of organisms on each cluster is conserved, so it is easier
to concatenate them into a super-gene suitable for phylogenetic inference.
</p>


<hr>
<a id="method-PgR6MS-clone"></a>



<h4>Method <code>clone()</code></h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>PgR6MS$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt><dd><p>Whether to make a deep clone.</p>
</dd>
</dl>

</div>



<hr>
<h2 id='roary_2_pagoo'>Read roary's output into a pagoo's R6 class object</h2><span id='topic+roary_2_pagoo'></span>

<h3>Description</h3>

<p>This function handle conversion of <a href="https://sanger-pathogens.github.io/Roary/">roary</a>'s output files into
a pagoo R6 class object. It takes the &quot;gene_presence_absence.csv&quot; file and
(optionally but recommended) gff input file paths, and returns an object of
class <code><a href="#topic+PgR6MS">PgR6MS</a></code> (or <code><a href="#topic+PgR6M">PgR6M</a></code> if left
empty the <code>gffs</code> argument).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>roary_2_pagoo(gene_presence_absence_csv, gffs, sep = "__", paralog_sep = "\t")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="roary_2_pagoo_+3A_gene_presence_absence_csv">gene_presence_absence_csv</code></td>
<td>
<p><code>character</code>, path to the
&quot;gene_presence_absence.csv&quot; file. (Do not confuse with the file with the
same name but with <code>.Rtab</code> extension).</p>
</td></tr>
<tr><td><code id="roary_2_pagoo_+3A_gffs">gffs</code></td>
<td>
<p>A <code>character</code> vector with paths to original gff files used
as roary's input. Typically the return value of <code>list.files()</code> function.
This parameter is optional but highly recommended if you want to manipulate
sequences.</p>
</td></tr>
<tr><td><code id="roary_2_pagoo_+3A_sep">sep</code></td>
<td>
<p><code>character</code>. Default: <code>"__"</code> (two underscores). See
<a href="#topic+PgR6MS">PgR6MS</a> for a more detail argument description.</p>
</td></tr>
<tr><td><code id="roary_2_pagoo_+3A_paralog_sep">paralog_sep</code></td>
<td>
<p><code>character</code>. A gene separator for cases where the
clusters have in-paralogs. (Default: &quot;\t&quot; - tab).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A pagoo's R6 class object. Either <a href="#topic+PgR6M">PgR6M</a>, if <code>gffs</code>
argument is left empty, or <a href="#topic+PgR6MS">PgR6MS</a> if path to gff files is
provided.
</p>


<h3>References</h3>

<p>Andrew J. Page, Carla A. Cummins, Martin Hunt, Vanessa K. Wong, Sandra Reuter,
Matthew T. G. Holden, Maria Fookes, Daniel Falush, Jacqueline A. Keane, Julian
Parkhill, &quot;Roary: Rapid large-scale prokaryote pan genome analysis&quot;,
Bioinformatics, 2015;31(22):3691-3693
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
gffs &lt;- list.files(path = "path/to/gffs/",
                   pattern = "[.]gff$",
                   full.names = TRUE)
gpa_csv &lt;- "path/to/gene_presence_absence.csv"

library(pagoo)
pg &lt;- roary_2_pagoo(gene_presence_absence_csv = gpa_csv,
                    gffs = gffs)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
