<!DOCTYPE html><html><head><title>Help for package PAGWAS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PAGWAS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PAGWAS-package'>
<p>Pathway Analysis Methods for Genomewide Association Data</p></a></li>
<li><a href='#create.pathway.df'>
<p>Creates a pathway data frame</p></a></li>
<li><a href='#FM.chi.pvalue'>
<p>Calculates the Fisher's method p-value for each tested pathway</p></a></li>
<li><a href='#genes'>
<p>A data frame of 20 artificial genes with their chromosomes and positions on the genome</p></a></li>
<li><a href='#genotypes'>
<p>Genotypes for 100 SNPs and 75 individuals</p></a></li>
<li><a href='#NBF'>
<p>Normal/Bayes factors method for finding associated pathways</p></a></li>
<li><a href='#pathways'>
<p>A list of 2 pathways with their gene members</p></a></li>
<li><a href='#SNAL'><p>Sparse Normal/Adaptive lasso method for finding associated pathways</p></a></li>
<li><a href='#SNAL.calculation'>
<p>Sparse Normal/Adaptive lasso method for finding associated variables. The SNAL method is applied to the linear regression Y= Phi beta + epsilon</p></a></li>
<li><a href='#SNPs'>
<p>A data frame of 100 artificial SNPs with their chromosomes and positions on the genome</p></a></li>
<li><a href='#snps.to.genes'>
<p>Assigns SNPs to genes</p></a></li>
<li><a href='#snps.to.pathways'>
<p>Assigns SNPs to pathways</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Pathway Analysis Methods for Genomewide Association Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-12-07</td>
</tr>
<tr>
<td>Author:</td>
<td>Marina Evangelou</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Marina Evangelou &lt;m.evangelou@ic.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Bayesian hierarchical methods for pathway analysis of genomewide association data: Normal/Bayes factors and Sparse Normal/Adaptive lasso. The Frequentist Fisher's product method is included as well.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>lars, foreach, mnormt</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-12-08 12:35:35 UTC; marinaevangelou</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-12-08 14:53:45</td>
</tr>
</table>
<hr>
<h2 id='PAGWAS-package'>
Pathway Analysis Methods for Genomewide Association Data</h2><span id='topic+PAGWAS-package'></span><span id='topic+PAGWAS'></span>

<h3>Description</h3>

<p>Bayesian hierarchical methods for pathway analysis of genomewide association data: Normal/Bayes factors and Sparse Normal/Adaptive lasso. The Frequentist Fisher's method is included as well.</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> PAGWAS</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2015-12-02</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;=2) </td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Marina Evangelou
Maintainer: Marina Evangelou &lt;m.evangelou@ic.ac.ukt&gt;
</p>


<h3>References</h3>

<p>Evangelou, M., Dudbridge, F., Wernisch, L. (2014). Two novel pathway analysis methods based on a hierarchical model. Bioinformatics 30(5), 690 - 697
</p>
<p>Evangelou, M., Rendon, A., Ouhewand, W. H., Wernisch, L., Dudbridge, F. (2012) Comparison of methods for competitive tests of pathway analysis. Plos One 7(7): e41018
</p>

<hr>
<h2 id='create.pathway.df'>
Creates a pathway data frame
</h2><span id='topic+create.pathway.df'></span>

<h3>Description</h3>

<p>Returns a data frame with L rows and M columns. L is the number of SNPs in the genotypes data frame and M is the number of tested pathways. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create.pathway.df(genotypes,snps.paths)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create.pathway.df_+3A_genotypes">genotypes</code></td>
<td>

<p>Genotype matrix, with L SNPs (columns) and N individuals (rows)</p>
</td></tr>
<tr><td><code id="create.pathway.df_+3A_snps.paths">snps.paths</code></td>
<td>

<p>A list with entries the SNP members of each pathway. The size of the list is M</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns equal to the number of pathways in the pathway.snps list and rows equal to the number of tested SNPs
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SNPs">SNPs</a></code>, <code><a href="#topic+genes">genes</a></code>, <code><a href="#topic+snps.to.pathways">snps.to.pathways</a></code>
<code><a href="#topic+snps.to.genes">snps.to.genes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)
data(genes)
data(pathways)
data(genotypes)
snps.genes &lt;- snps.to.genes(snp.info=SNPs,gene.info=genes, distance=0)
pathway.snps &lt;- snps.to.pathways(pathways,snps.genes)
P &lt;- create.pathway.df(genotypes=genotypes,snps.paths=pathway.snps)
</code></pre>

<hr>
<h2 id='FM.chi.pvalue'>
Calculates the Fisher's method p-value for each tested pathway
</h2><span id='topic+FM.chi.pvalue'></span>

<h3>Description</h3>

<p>Calculates the Fisher's method <em>p</em>-value for a set of <em>p</em>-values. It returns both the <em>p</em>-value and the test statistic value of the Fisher's product method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FM.chi.pvalue(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FM.chi.pvalue_+3A_x">x</code></td>
<td>

<p>A vector of <em>p</em>-values. These <em>p</em>-values can be either gene or SNP <em>p</em>-values of a tested pathway</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>FMstatistic</code></td>
<td>
<p>Fisher's product method test statistic</p>
</td></tr>
<tr><td><code>FMpvalue</code></td>
<td>
<p>Fisher's method <em>p</em>-value, computed using the exact distribution of the Fisher's method test statistic which is a <em>Chi^2</em> distribution with degrees of freedom twice the size of vector x</p>
</td></tr>
</table>


<h3>References</h3>

<p>Evangelou M, Rendon A, Ouwehand WH, Wernisch L, Dudbridge F (2012) Comparison of Methods for Competitive Tests of Pathway Analysis. PLoS ONE 7(7): e41018. doi:10.1371/journal.pone.0041018</p>


<h3>See Also</h3>

<p><code><a href="#topic+pathways">pathways</a></code>, <code><a href="#topic+snps.to.pathways">snps.to.pathways</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>FM.chi.pvalue(x=c(0.05,0.1))</code></pre>

<hr>
<h2 id='genes'>
A data frame of 20 artificial genes with their chromosomes and positions on the genome
</h2><span id='topic+genes'></span>

<h3>Description</h3>

<p>A data frame with 20 rows and 4 columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(genes)</code></pre>


<h3>Format</h3>

<p>Column names:
</p>

<dl>
<dt><code>Name</code></dt><dd><p>Name of gene</p>
</dd>
<dt><code>Start</code></dt><dd><p>Start position of gene on the genome</p>
</dd>
<dt><code>End</code></dt><dd><p>End position of gene on the genome</p>
</dd>
<dt><code>Chr</code></dt><dd><p>Chromosome of gene</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+SNPs">SNPs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genes)
print(genes[1:5,])
</code></pre>

<hr>
<h2 id='genotypes'>
Genotypes for 100 SNPs and 75 individuals
</h2><span id='topic+genotypes'></span>

<h3>Description</h3>

<p>A data frame with 75 rows (individuals) and 100 columns (SNPs). The entries of the genotype matrix are 0, 1 and 2. There are no missing values. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(genotypes)</code></pre>


<h3>See Also</h3>

 <p><code><a href="#topic+SNPs">SNPs</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genotypes)
</code></pre>

<hr>
<h2 id='NBF'>
Normal/Bayes factors method for finding associated pathways</h2><span id='topic+NBF'></span>

<h3>Description</h3>

<p>A vector of the computed Bayes factors for the tested pathways.</p>


<h3>Usage</h3>

<pre><code class='language-R'>NBF(y, G, P, a, b, s2, nu)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NBF_+3A_y">y</code></td>
<td>

<p>Response vector of length N</p>
</td></tr>
<tr><td><code id="NBF_+3A_g">G</code></td>
<td>

<p>Genotype matrix, with N rows and L columns (number of tested SNPs) </p>
</td></tr>
<tr><td><code id="NBF_+3A_p">P</code></td>
<td>

<p>Pathway matrix, with L columns and M columns (number of tested pathways) </p>
</td></tr>
<tr><td><code id="NBF_+3A_a">a</code></td>
<td>

<p>Hyper-parameter of the variance assumed for the integrated out SNP effects</p>
</td></tr>
<tr><td><code id="NBF_+3A_b">b</code></td>
<td>

<p>Hyper-parameter of the variance assumed for the pathway effects</p>
</td></tr>
<tr><td><code id="NBF_+3A_s2">s2</code></td>
<td>

<p>Hyper-parameter of the Inverse-Chi-squared distribution assumed for the variance of the response vector </p>
</td></tr>
<tr><td><code id="NBF_+3A_nu">nu</code></td>
<td>

<p>Hyper-parameter of the Inverse-Chi-squared distribution assumed for the variance of the response vector </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of the computed Bayes factors of the same length as the number of tested pathways 
</p>


<h3>References</h3>

<p>Evangelou, M., Dudbridge, F., Wernisch, L. (2014). Two novel pathway analysis methods based on a hierarchical model. Bioinformatics, 30(5), 690 - 697.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
	data(genotypes)
	G=genotypes
	data(pathways)
	data(SNPs)
	data(genes)
	snps.genes=snps.to.genes(SNPs,genes,distance=0)
	snps.paths=snps.to.pathways(pathways,snps.genes)
	P=create.pathway.df(G,snps.paths)
	y=rnorm(nrow(G),mean=0,sd=10)
	NBF(y,G,P,a,b,s2,nu)
## End(Not run)
</code></pre>

<hr>
<h2 id='pathways'>
A list of 2 pathways with their gene members
</h2><span id='topic+pathways'></span>

<h3>Description</h3>

<p>A list of two pathways. The gene members of each pathway are given.</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pathways)</code></pre>


<h3>See Also</h3>

<p><code><a href="#topic+genes">genes</a></code>, <code><a href="#topic+SNPs">SNPs</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pathways)
</code></pre>

<hr>
<h2 id='SNAL'>Sparse Normal/Adaptive lasso method for finding associated pathways</h2><span id='topic+SNAL'></span>

<h3>Description</h3>

<p>Sparse Normal/Adaptive lasso method applied for finding the associated pathways. The iterative algorithm suggested by Wipf and Nagarajan (2008) is applied. A vector equal to the number of tested pathways is returned, the zero entries of the vector correspond to the pathways that are not associated. The posterior estimates of the beta coefficients are also returned as they are described by Wipf and Nagarajan (2008).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNAL(y, G, P, a, s2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNAL_+3A_y">y</code></td>
<td>

<p>Response vector of length N</p>
</td></tr>
<tr><td><code id="SNAL_+3A_g">G</code></td>
<td>

<p>Genotype matrix, with N rows and L columns (number of tested SNPs) </p>
</td></tr>
<tr><td><code id="SNAL_+3A_p">P</code></td>
<td>

<p>Pathway matrix, with L columns and M columns (number of tested pathways) </p>
</td></tr>
<tr><td><code id="SNAL_+3A_a">a</code></td>
<td>

<p>Hyper-parameter of the variance assumed for the integrated out SNP effects</p>
</td></tr>
<tr><td><code id="SNAL_+3A_s2">s2</code></td>
<td>

<p>Variance assumed for the response variable, the tuning parameter of adaptive lasso</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>gamma.star</code></td>
<td>
<p>Estimates of gamma hyper-parameters</p>
</td></tr>
<tr><td><code>ARD</code></td>
<td>
<p>Posterior estimates of beta coefficients</p>
</td></tr>
</table>


<h3>References</h3>

<p>Evangelou, M., Dudbridge, F., Wernisch, L. (2014). Two novel pathway analysis methods based on a hierarchical model. Bioinformatics, 30(5), 690 - 697.
</p>
<p>Wipf, D. and Nagarajan, S. (2008). A new view of automatic relevance determination. Advances in Neural Information Processing Systems, 20 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SNAL.calculation">SNAL.calculation</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
	data(genotypes)
	G=genotypes
	data(pathways)
	data(SNPs)
	data(genes)
	snps.genes=snps.to.genes(SNPs,genes,distance=0)
	snps.paths=snps.to.pathways(pathways,snps.genes)
	P=create.pathway.df(G,snps.paths)
	y=rnorm(nrow(G),mean=0,sd=10)
	SNAL(y,G,P,a,s2)
## End(Not run)
</code></pre>

<hr>
<h2 id='SNAL.calculation'>
Sparse Normal/Adaptive lasso method for finding associated variables. The SNAL method is applied to the linear regression Y= Phi beta + epsilon
</h2><span id='topic+SNAL.calculation'></span>

<h3>Description</h3>

<p>For more details please read <code><a href="#topic+SNAL">SNAL</a>.</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNAL.calculation(Y, Phi, s2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNAL.calculation_+3A_y">Y</code></td>
<td>
<p>Response vector of length N</p>
</td></tr>
<tr><td><code id="SNAL.calculation_+3A_phi">Phi</code></td>
<td>
<p>Design matrix, with N rows and M columns (number of tested variables)</p>
</td></tr>
<tr><td><code id="SNAL.calculation_+3A_s2">s2</code></td>
<td>

<p>Variance assumed for the response variable, the tuning parameter of the adaptive lasso problem</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>gamma.star</code></td>
<td>
<p>Estimates of gamma hyper-parameters</p>
</td></tr>
<tr><td><code>ARD</code></td>
<td>
<p>Posterior estimates of beta coefficients</p>
</td></tr>
</table>


<h3>References</h3>

<p>Evangelou, M., Dudbridge, F., Wernisch, L. (2014). Two novel pathway analysis methods based on a hierarchical model. Bioinformatics, 30(5), 690 - 697
</p>
<p>Wipf, D. and Nagarajan, S. (2008). A new view of automatic relevance determination. Advances in Neural Information Processing Systems, 20  </p>


<h3>See Also</h3>

<p><code><a href="#topic+SNAL">SNAL</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: SNAL.calculation(Y,Phi,s2=0.5)
</code></pre>

<hr>
<h2 id='SNPs'>
A data frame of 100 artificial SNPs with their chromosomes and positions on the genome
</h2><span id='topic+SNPs'></span>

<h3>Description</h3>

 
<p>A data frame with 100 rows and 3 columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SNPs)</code></pre>


<h3>Format</h3>

<p>Column names:
</p>

<dl>
<dt><code>Name</code></dt><dd><p>SNP name</p>
</dd>
<dt><code>Position</code></dt><dd><p>Position of SNP on the genome</p>
</dd>
<dt><code>Chr</code></dt><dd><p>Chromosome of the SNP</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+genes">genes</a></code>, <code><a href="#topic+genotypes">genotypes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)
print(SNPs[1:5,])
</code></pre>

<hr>
<h2 id='snps.to.genes'>
Assigns SNPs to genes
</h2><span id='topic+snps.to.genes'></span>

<h3>Description</h3>

<p>Assigns SNPs to genes based on their physical distance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snps.to.genes(snp.info, gene.info, distance)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snps.to.genes_+3A_snp.info">snp.info</code></td>
<td>
<p>A data frame with 3 columns with names:  Name, Position and Chr that correspond to the SNP name, its position on the genome and its chromosome, respectively
</p>
</td></tr>
<tr><td><code id="snps.to.genes_+3A_gene.info">gene.info</code></td>
<td>
<p>A data frame with 4 columns with names: Name, Start, End and Chr that correspond to the gene name, start and end positions on the genome and its chromosome, respectively
</p>
</td></tr>
<tr><td><code id="snps.to.genes_+3A_distance">distance</code></td>
<td>
<p>A number that corresponds to the distance below and above the Start and End positions of the gene that all SNPs in that region should be assigned to the gene
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the same size as the number of genes of the gene.info data frame. The names of the SNPs assigned to each gene are returned
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SNPs">SNPs</a></code>, <code><a href="#topic+genes">genes</a></code>, <code><a href="#topic+snps.to.pathways">snps.to.pathways</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)
data(genes)
snps.to.genes(snp.info=SNPs,gene.info=genes,distance=50)
</code></pre>

<hr>
<h2 id='snps.to.pathways'>
Assigns SNPs to pathways
</h2><span id='topic+snps.to.pathways'></span>

<h3>Description</h3>

<p>Assigns SNPs to pathways, using the pathway gene members and the SNPs assigned to each gene.</p>


<h3>Usage</h3>

<pre><code class='language-R'>snps.to.pathways(pathways,gene.snps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snps.to.pathways_+3A_pathways">pathways</code></td>
<td>
<p>A list of pathways with their gene members</p>
</td></tr>
<tr><td><code id="snps.to.pathways_+3A_gene.snps">gene.snps</code></td>
<td>
<p>A list of genes with the SNPs assigned to them according to their physical distance on the genome</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the same size as the number of pathways in the pathway list. The names of the SNPs assigned to each pathway are returned. Empty pathways are also returned.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SNPs">SNPs</a></code>, <code><a href="#topic+genes">genes</a></code>, <code><a href="#topic+snps.to.genes">snps.to.genes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SNPs)
data(genes)
data(pathways)
snps.genes &lt;- snps.to.genes(snp.info=SNPs,gene.info=genes, distance=50)
pathway.snps &lt;- snps.to.pathways(pathways,snps.genes)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
