<!DOCTYPE html><html><head><title>Help for package pald</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pald}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aggregation'><p>Aggregation</p></a></li>
<li><a href='#any_isolated'><p>Any isolated</p></a></li>
<li><a href='#as_cohesion_matrix'><p>Coerce a matrix to a cohesion matrix object</p></a></li>
<li><a href='#cognate_dist'><p>Cognate Data Distance Matrix</p></a></li>
<li><a href='#cohesion_matrix'><p>Cohesion Matrix</p></a></li>
<li><a href='#cohesion_strong'><p>Cohesion Matrix: Strong Ties</p></a></li>
<li><a href='#community_clusters'><p>Community clusters</p></a></li>
<li><a href='#community_graphs'><p>Community Graphs</p></a></li>
<li><a href='#cultures'><p>Cultures pairwise dissimilarities</p></a></li>
<li><a href='#dist_cohesion_plot'><p>Distance Cohesion Plot</p></a></li>
<li><a href='#exdata1'><p>Example Data 1</p></a></li>
<li><a href='#exdata2'><p>Example Data 2</p></a></li>
<li><a href='#exdata3'><p>Example Data 3</p></a></li>
<li><a href='#local_depths'><p>Local (Community) Depths</p></a></li>
<li><a href='#noisy_circles'><p>Noisy circles</p></a></li>
<li><a href='#noisy_moons'><p>Noisy moons</p></a></li>
<li><a href='#pald'><p>Partitioned Local Depth (PaLD)</p></a></li>
<li><a href='#pald_colors'><p>PaLD Color Palette</p></a></li>
<li><a href='#plot_community_graphs'><p>Plot Community Graphs</p></a></li>
<li><a href='#strong_threshold'><p>Cohesion Threshold for Strong Ties</p></a></li>
<li><a href='#tissue_dist'><p>Tissue Data Distance Matrix</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Partitioned Local Depth for Community Structure in Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of the Partitioned Local Depth (PaLD) 
    approach which provides a measure of local depth and the cohesion of a point 
    to another which  (together with a universal threshold for distinguishing 
    strong and weak ties) may be used to reveal local and global structure in 
    data, based on methods described in Berenhaut, Moore, and Melvin (2022)
    &lt;<a href="https://doi.org/10.1073%2Fpnas.2003634119">doi:10.1073/pnas.2003634119</a>&gt;. No extraneous inputs, distributional 
    assumptions, iterative procedures nor optimization criteria are employed. 
    This package includes functions for computing local depths and cohesion as
    well as flexible functions for plotting community networks and displays of 
    cohesion against distance. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>igraph, graphics, glue</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/LucyMcGowan/pald">https://github.com/LucyMcGowan/pald</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/LucyMcGowan/pald/issues">https://github.com/LucyMcGowan/pald/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-28 19:48:37 UTC; lucymcgowan</td>
</tr>
<tr>
<td>Author:</td>
<td>Katherine Moore <a href="https://orcid.org/0000-0001-6943-2416"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Kenneth Berenhaut [aut],
  Lucy D'Agostino McGowan
    <a href="https://orcid.org/0000-0001-7297-9359"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lucy D'Agostino McGowan &lt;lucydagostino@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-28 20:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='aggregation'>Aggregation</h2><span id='topic+aggregation'></span>

<h3>Description</h3>

<p>A synthetic data set of two-dimensional points created by Gionis et al. to
demonstrate clustering aggregation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aggregation
</code></pre>


<h3>Format</h3>

<p>A data frame with 788 rows and 2 columns, <code>x1</code> and <code>x2</code>.
</p>


<h3>References</h3>

<p>A. Gionis, H. Mannila, and P. Tsaparas, Clustering aggregation.
ACM Transactions on Knowledge Discovery from Data (TKDD), 2007. 1(1): p. 1-30.
</p>

<hr>
<h2 id='any_isolated'>Any isolated</h2><span id='topic+any_isolated'></span>

<h3>Description</h3>

<p>Checks for isolated points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>any_isolated(c)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="any_isolated_+3A_c">c</code></td>
<td>
<p>A <code>cohesion_matrix</code> object, a matrix of cohesion values
(see <code><a href="#topic+cohesion_matrix">cohesion_matrix</a></code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical, indicating whether any points are isolated.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>d &lt;- data.frame(
  x1 = c(1, 2, 3, 6),
  x2 = c(2, 1, 3, 10)
  )
D &lt;- dist(d)
C &lt;- cohesion_matrix(D)
any_isolated(C)
</code></pre>

<hr>
<h2 id='as_cohesion_matrix'>Coerce a matrix to a cohesion matrix object</h2><span id='topic+as_cohesion_matrix'></span>

<h3>Description</h3>

<p><code>as_cohesion_matrix()</code> converts an existing matrix into an object of class
<code>cohesion_matrix</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_cohesion_matrix(c)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_cohesion_matrix_+3A_c">c</code></td>
<td>
<p>A matrix of cohesion values (see <code><a href="#topic+cohesion_matrix">cohesion_matrix</a></code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code>cohesion_matrix</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>C &lt;- matrix(
  c(0.25, 0.125, 0.125, 0,
   0.125, 0.25, 0, 0.125,
   0.125, 0, 0.25, 0.125,
   0, 0.125, 0.125, 0.25
), nrow = 4, byrow = TRUE)

class(C)

C &lt;- as_cohesion_matrix(C)
class(C)
</code></pre>

<hr>
<h2 id='cognate_dist'>Cognate Data Distance Matrix</h2><span id='topic+cognate_dist'></span>

<h3>Description</h3>

<p>A <code><a href="stats.html#topic+dist">dist</a></code> object describing distances between 87 Indo-European languages from
the perspective of cognates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cognate_dist
</code></pre>


<h3>Format</h3>

<p>A <code><a href="stats.html#topic+dist">dist</a></code> object for 87 Indo-European languages.
</p>


<h3>Details</h3>

<p>Cognate relationships from a collection of essential words were collected
from Dyen et al. and encoded in a 87x2665 binary matrix from which this
distance matrix was derived (using Euclidean distance).
</p>


<h3>References</h3>

<p>I. Dyen, J. B. Kruskal, P. Black, An Indoeuropean
classification: A lexicostatistical experiment. Trans. Am. Phil. Soc. 82,
iii-132 (1992).
</p>

<hr>
<h2 id='cohesion_matrix'>Cohesion Matrix</h2><span id='topic+cohesion_matrix'></span>

<h3>Description</h3>

<p>Creates a matrix of (pairwise) cohesion values from a matrix of pairwise
distances or a <code><a href="stats.html#topic+dist">dist</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cohesion_matrix(d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cohesion_matrix_+3A_d">d</code></td>
<td>
<p>A matrix of pairwise distances or a <code><a href="stats.html#topic+dist">dist</a></code> object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Computes the matrix of (pairwise) cohesion values, C_xw, from a matrix of
pairwise distances or a <code><a href="stats.html#topic+dist">dist</a></code> object. Cohesion is an interpretable probability
that reflects the strength of alignment of a point, <code>w</code>, to another point, <code>x</code>.
The rows of the cohesion matrix can be seen as providing neighborhood
weights.  These values may be used for defining associated weighted graphs
(for the purpose of community analysis) as in  Berenhaut, Moore, and
Melvin (2022).
</p>
<p>Given an n x n distance matrix, the sum of the entries in the resulting
cohesion matrix is always equal to n/2.
Cohesion is partitioned local depth (see <code><a href="#topic+local_depths">local_depths</a></code>) and thus the row
sums of the cohesion matrix provide a measure of local depth centrality.
</p>
<p>If you have a matrix that is already a cohesion matrix and you would like to
add the class, see <code><a href="#topic+as_cohesion_matrix">as_cohesion_matrix()</a></code>.
</p>


<h3>Value</h3>

<p>The matrix of cohesion values. An object of class <code>cohesion_matrix</code>.
</p>


<h3>References</h3>

<p>K. S. Berenhaut, K. E. Moore, R. L. Melvin, A social perspective
on perceived distances reveals deep community structure. Proc. Natl.
Acad. Sci., 119(4), 2022.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
plot(exdata1)
text(exdata1 + .08, lab = 1:8)

D &lt;- dist(exdata1)
C &lt;- cohesion_matrix(D)
C

## neighbor weights (provided by cohesion) for the 8th point in exdata1
C[8, ]
localdepths &lt;- rowSums(C)

</code></pre>

<hr>
<h2 id='cohesion_strong'>Cohesion Matrix: Strong Ties</h2><span id='topic+cohesion_strong'></span>

<h3>Description</h3>

<p>Provides the symmetrized and thresholded matrix of cohesion values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cohesion_strong(c, symmetric = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cohesion_strong_+3A_c">c</code></td>
<td>
<p>A <code>cohesion_matrix</code> object, a matrix of cohesion values
(see <code><a href="#topic+cohesion_matrix">cohesion_matrix</a></code>).</p>
</td></tr>
<tr><td><code id="cohesion_strong_+3A_symmetric">symmetric</code></td>
<td>
<p>Logical. Whether the returned matrix should be made
symmetric (using the minimum); the default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The threshold is that provided by strong_threshold (and is equal to half of
the average of the diagonal of <code>c</code>).
Values of the cohesion matrix which are less than the threshold are set to
zero.
The symmetrization, if desired, is computed using the entry-wise (parallel)
minimum of C and its transpose (i.e., <code>min(C_ij, C_ji)</code>).
The matrix provided by cohesion_strong (with default <code>symmetric = TRUE</code>) is
the adjacency matrix for the graph of strong ties (the cluster graph), see
<code><a href="#topic+community_graphs">community_graphs</a></code> and <code><a href="#topic+pald">pald</a></code>.
</p>


<h3>Value</h3>

<p>The symmetrized cohesion matrix in which all entries corresponding
to weak ties are set to zero.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>C &lt;- cohesion_matrix(dist(exdata2))
strong_threshold(C)
cohesion_strong(C)

## To illustrate the calculation performed
C_strong &lt;- C

## C_strong is equal to cohesion_strong(C, symmetric = FALSE)
C_strong[C &lt; strong_threshold(C)] &lt;- 0

## C_strong_sym is equal to cohesion_strong(C)
C_strong_sym &lt;- pmin(C_strong, t(C_strong))

## The (cluster) graph whose adjacency matrix, CS,
## is the matrix of strong ties
CS &lt;- cohesion_strong(C)

if (requireNamespace("igraph", quietly = TRUE)) {
G_strong &lt;- igraph::simplify(
  igraph::graph.adjacency(CS, weighted = TRUE, mode = "undirected")
  )
plot(G_strong)
}
</code></pre>

<hr>
<h2 id='community_clusters'>Community clusters</h2><span id='topic+community_clusters'></span>

<h3>Description</h3>

<p>Community clusters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>community_clusters(c)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="community_clusters_+3A_c">c</code></td>
<td>
<p>A <code>cohesion_matrix</code> object, a matrix of cohesion values
(see <code><a href="#topic+cohesion_matrix">cohesion_matrix</a></code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with two columns:
</p>

<ul>
<li> <p><code>point</code>: The points from cohesion matrix <code>c</code>
</p>
</li>
<li> <p><code>community</code>: The community cluster labels
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>D &lt;- dist(exdata2)
C &lt;- cohesion_matrix(D)
community_clusters(C)
</code></pre>

<hr>
<h2 id='community_graphs'>Community Graphs</h2><span id='topic+community_graphs'></span>

<h3>Description</h3>

<p>Provides the graphs whose edge weights are (mutual) cohesion, together with
a graph layout.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>community_graphs(c)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="community_graphs_+3A_c">c</code></td>
<td>
<p>A <code>cohesion_matrix</code> object, a matrix of cohesion values
(see <code><a href="#topic+cohesion_matrix">cohesion_matrix</a></code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Constructs the graphs whose edge weights are (mutual) cohesion
(see <code><a href="#topic+cohesion_matrix">cohesion_matrix</a></code>), self-loops are removed.
The graph G has adjacency matrix equal to the symmetrized cohesion matrix
(using the entry-wise parallel minimum of C and its transpose).
The graph G_strong has adjacency matrix equal to the thresholded and
symmetrized cohesion matrix (see <code><a href="#topic+cohesion_strong">cohesion_strong</a></code>).  The threshold is
equal to half of the average of the diagonal of the
cohesion matrix (see <code><a href="#topic+strong_threshold">strong_threshold</a></code>).
</p>
<p>A layout is also computed using the Fruchterman-Reingold (FR)
force-directed graph drawing algorithm.  As a result, it may provide a
somewhat different layout each time it is run.
</p>


<h3>Value</h3>

<p>A list consisting of:
</p>

<ul>
<li> <p><code>G</code>: the weighted (community) graph whose edge weights are mutual
cohesion
</p>
</li>
<li> <p><code>G_strong</code>: the weighted (community) graph consisting of edges
for which mutual cohesion is greater than the threshold for strong
ties (see <code><a href="#topic+strong_threshold">strong_threshold</a></code>)
</p>
</li>
<li> <p><code>layout</code>: the layout, using the Fruchterman Reingold (FR)
force-directed graph drawing for the graph <code>G</code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>C &lt;- cohesion_matrix(dist(exdata2))
plot(community_graphs(C)$G_strong)
plot(community_graphs(C)$G_strong, layout = community_graphs(C)$layout)
</code></pre>

<hr>
<h2 id='cultures'>Cultures pairwise dissimilarities</h2><span id='topic+cultures'></span>

<h3>Description</h3>

<p>Pairwise dissimilarities are given by the cultural fixation index obtained
from World Values Survey responses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cultures
</code></pre>


<h3>Format</h3>

<p>A <code style="white-space: pre;">&#8288;59x59 matrix&#8288;</code> of dissimilarities
</p>


<h3>References</h3>

<p>M. Muthukrishna, et al., Beyond western, educated, industrial,
rich, and democratic (WEIRD) psychology: measuring and mapping scales of
cultural and psychological distance. Psychol. Sci. 1, 24 (2020).
</p>
<p>R. Inglehart et al, World Values Survey: All Rounds-Country-Pooled Datafile
1981-2014, (JD Systems Institute, Madrid 2014).
</p>

<hr>
<h2 id='dist_cohesion_plot'>Distance Cohesion Plot</h2><span id='topic+dist_cohesion_plot'></span>

<h3>Description</h3>

<p>Provides a plot of cohesion against distance, with the threshold indicated
by a horizontal line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_cohesion_plot(
  d,
  mutual = FALSE,
  xlim_max = NULL,
  cex = 1,
  colors = NULL,
  weak_gray = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist_cohesion_plot_+3A_d">d</code></td>
<td>
<p>A matrix of pairwise distances or a <code><a href="stats.html#topic+dist">dist</a></code> object.</p>
</td></tr>
<tr><td><code id="dist_cohesion_plot_+3A_mutual">mutual</code></td>
<td>
<p>Set to <code>TRUE</code> to consider mutual cohesion (i.e., symmetrized
using the minimum); the default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="dist_cohesion_plot_+3A_xlim_max">xlim_max</code></td>
<td>
<p>If desired, set the maximum value of distance which
is displayed on the x-axis.</p>
</td></tr>
<tr><td><code id="dist_cohesion_plot_+3A_cex">cex</code></td>
<td>
<p>Factor by which points should be scaled relative to the default.</p>
</td></tr>
<tr><td><code id="dist_cohesion_plot_+3A_colors">colors</code></td>
<td>
<p>A vector of color names, if none is given a default is
provided.</p>
</td></tr>
<tr><td><code id="dist_cohesion_plot_+3A_weak_gray">weak_gray</code></td>
<td>
<p>Set to <code>TRUE</code> to display the plot with all weak ties plotted
in gray; the default is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot of cohesion against distance provides a visualization for the
manner in which distance is transformed.
The threshold distinguishing strong and weak ties is indicated by a
horizontal line.
When there are separated regions with different density, one can often
observe vertical bands of color, see example below and Berenhaut, Moore, and
Melvin (2022).  For each distance pair in <code>d</code>, the corresponding value of
cohesion is computed.  If the pair is within a single cluster, the point is
colored (with the same color provided by the <code><a href="#topic+pald">pald</a></code> and
<code><a href="#topic+plot_community_graphs">plot_community_graphs</a></code> functions).  Weak ties appear below the threshold.
</p>
<p>Note that cohesion is not symmetric, and so all <code>n^2</code> points are plotted.
A gray point above the threshold corresponds to a pair in which the value
of cohesion is greater than the threshold in only one direction.  If one
only wants to observe mutual cohesion (i.e., cohesion made symmetric via
the minimum), set <code>mutual = TRUE</code>.
</p>


<h3>Value</h3>

<p>A plot of cohesion against distance with threshold indicated by a
horizontal line.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D &lt;- dist(exdata2)
dist_cohesion_plot(D)
dist_cohesion_plot(D, mutual = TRUE)
C &lt;- cohesion_matrix(D)
threshold &lt;- strong_threshold(C) #the horizontal line
dist_cohesion_plot(D, mutual = TRUE, weak_gray = TRUE)
</code></pre>

<hr>
<h2 id='exdata1'>Example Data 1</h2><span id='topic+exdata1'></span>

<h3>Description</h3>

<p>A data set consisting of 8 points (in 2-dimensional Euclidean space)
to provide a simple illustrative example.  This data is displayed in Figure
1 in Berenhaut, Moore, and Melvin (2022).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exdata1
</code></pre>


<h3>Format</h3>

<p>A data frame with 8 rows and 2 columns, <code>x1</code> and <code>x2</code>
</p>


<h3>References</h3>

<p>K. S. Berenhaut, K. E. Moore, R. L. Melvin, A social perspective
on perceived distances reveals deep community structure. Proc. Natl.
Acad. Sci., 119(4), 2022.
</p>

<hr>
<h2 id='exdata2'>Example Data 2</h2><span id='topic+exdata2'></span>

<h3>Description</h3>

<p>A data set consisting of 16 points (in 2-dimensional Euclidean space)
to provide an illustrative example.  This data is displayed in Figure 2
in Berenhaut, Moore, and Melvin (2022).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exdata2
</code></pre>


<h3>Format</h3>

<p>A data frame with 16 rows and 2 columns, <code>x1</code> and <code>x2</code>
</p>


<h3>References</h3>

<p>K. S. Berenhaut, K. E. Moore, R. L. Melvin, A social perspective
on perceived distances reveals deep community structure. Proc. Natl.
Acad. Sci., 119(4), 2022.
</p>

<hr>
<h2 id='exdata3'>Example Data 3</h2><span id='topic+exdata3'></span>

<h3>Description</h3>

<p>A data set consisting of 240 points (in 2-dimensional Euclidean space)
to provide an illustrative example.  Points were generated from bivariate
normal distributions with varying mean and variance (with covariance
matrix cI). This data is displayed in Figure 4D in Berenhaut, Moore, and
Melvin (2022).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exdata3
</code></pre>


<h3>Format</h3>

<p>A data frame with 240 rows and 2 columns, <code>x1</code> and <code>x2</code>
</p>


<h3>References</h3>

<p>K. S. Berenhaut, K. E. Moore, R. L. Melvin, A social perspective
on perceived distances reveals deep community structure. Proc. Natl.
Acad. Sci., 119(4), 2022.
</p>

<hr>
<h2 id='local_depths'>Local (Community) Depths</h2><span id='topic+local_depths'></span>

<h3>Description</h3>

<p>Creates a vector of local depths from a matrix of distances (or <code>dist</code>
object).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>local_depths(d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="local_depths_+3A_d">d</code></td>
<td>
<p>A matrix of pairwise distances, a <code><a href="stats.html#topic+dist">dist</a></code> object, or a
<code><a href="#topic+cohesion_matrix">cohesion_matrix</a></code> object.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Local depth is an interpretable probability which reflects aspects of
relative position and centrality via distance comparisons
(i.e., d(z, x) &lt; d(z, y)).
</p>
<p>The average of the local depth values is always 1/2.  Cohesion is
partitioned local depth (see <code><a href="#topic+cohesion_matrix">cohesion_matrix</a></code>); the row-sums of the
cohesion matrix are the values of local depth.
</p>


<h3>Value</h3>

<p>A vector of local depths.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
D &lt;- dist(exdata1)
local_depths(D)
C &lt;- cohesion_matrix(D)
local_depths(C)

## local depths are the row sums of the cohesion matrix
rowSums(C)

## cognate distance data

ld_lang &lt;- sort(local_depths(cognate_dist))
</code></pre>

<hr>
<h2 id='noisy_circles'>Noisy circles</h2><span id='topic+noisy_circles'></span>

<h3>Description</h3>

<p>Noisy circles data generated from scikit-learn
</p>


<h3>Usage</h3>

<pre><code class='language-R'>noisy_circles
</code></pre>


<h3>Format</h3>

<p>A dataframe with 500 rows and 2 columns, <code>x1</code> and <code>x2</code>.
</p>


<h3>Source</h3>

<p><a href="https://scikit-learn.org/stable/modules/clustering.html#clustering">https://scikit-learn.org/stable/modules/clustering.html#clustering</a>
</p>

<hr>
<h2 id='noisy_moons'>Noisy moons</h2><span id='topic+noisy_moons'></span>

<h3>Description</h3>

<p>Noisy moons data generated from scikit-learn
</p>


<h3>Usage</h3>

<pre><code class='language-R'>noisy_moons
</code></pre>


<h3>Format</h3>

<p>A dataframe with 500 rows and 2 columns, <code>x1</code> and <code>x2</code>.
</p>


<h3>Source</h3>

<p><a href="https://scikit-learn.org/stable/modules/clustering.html#clustering">https://scikit-learn.org/stable/modules/clustering.html#clustering</a>
</p>

<hr>
<h2 id='pald'>Partitioned Local Depth (PaLD)</h2><span id='topic+pald'></span>

<h3>Description</h3>

<p>A wrapper function which computes the cohesion matrix, local depths,
community graphs and provides a plot of the community graphs with connected
components of the graph of strong ties colored by connected component.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pald(
  d,
  show_plot = TRUE,
  show_labels = TRUE,
  only_strong = FALSE,
  emph_strong = 2,
  edge_width_factor = 50,
  colors = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pald_+3A_d">d</code></td>
<td>
<p>A matrix of pairwise distances or a <code><a href="stats.html#topic+dist">dist</a></code> object.</p>
</td></tr>
<tr><td><code id="pald_+3A_show_plot">show_plot</code></td>
<td>
<p>Set to <code>TRUE</code> to display plot; the default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="pald_+3A_show_labels">show_labels</code></td>
<td>
<p>Set to <code>FALSE</code> to omit vertex labels (to display a subset
of labels, use optional parameter <code>vertex.label</code> to modify the label list).
Default: <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="pald_+3A_only_strong">only_strong</code></td>
<td>
<p>Set to <code>TRUE</code> if only strong ties, G_strong, should be
displayed; the default <code>FALSE</code> will show both strong (colored by connected
component) and weak ties (in gray).</p>
</td></tr>
<tr><td><code id="pald_+3A_emph_strong">emph_strong</code></td>
<td>
<p>Numeric. The numeric factor by which the edge widths of
strong ties are emphasized in the display; the default is <code>2</code>.</p>
</td></tr>
<tr><td><code id="pald_+3A_edge_width_factor">edge_width_factor</code></td>
<td>
<p>Numeric. Modify to change displayed edge widths.
Default: <code>50</code>.</p>
</td></tr>
<tr><td><code id="pald_+3A_colors">colors</code></td>
<td>
<p>A vector of display colors, if none is given a default list
(of length 24) is provided.</p>
</td></tr>
<tr><td><code id="pald_+3A_...">...</code></td>
<td>
<p>Optional parameters to pass to the <code><a href="igraph.html#topic+plot.igraph">igraph::plot.igraph</a></code>.
function. Some commonly passed arguments include:
</p>

<ul>
<li> <p><code>layout</code>  A layout for the graph.  If none is specified, FR-graph
drawing algorithm is used.
</p>
</li>
<li> <p><code>vertex.label</code> A vector containing label names. If none is given,
the rownames of <code>c</code> are used
</p>
</li>
<li> <p><code>vertex.size</code> A numeric value for vertex size (default = <code>1</code>)
</p>
</li>
<li> <p><code>vertex.color.vec</code> A vector of color names for coloring the vertices
</p>
</li>
<li> <p><code>vertex.label.cex</code> A numeric value for modifying the vertex label size.
(default = <code>1</code>)
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This function re-computes the cohesion matrix each time it is run.
To avoid unnecessary computation when creating visualizations, use the
function <code><a href="#topic+cohesion_matrix">cohesion_matrix</a></code> to compute the cohesion matrix which may then
be taken as input for <code><a href="#topic+local_depths">local_depths</a></code>, <code><a href="#topic+strong_threshold">strong_threshold</a></code>,
<code><a href="#topic+cohesion_strong">cohesion_strong</a></code>, <code><a href="#topic+community_graphs">community_graphs</a></code>, and <code><a href="#topic+plot_community_graphs">plot_community_graphs</a></code>.
For further details regarding each component, see the documentation for
each of the above functions.
</p>


<h3>Value</h3>

<p>A list consisting of:
</p>

<ul>
<li> <p><code>C</code>: the matrix of cohesion values
</p>
</li>
<li>  <p><code>local_depths</code>: a vector of local depths
</p>
</li>
<li> <p><code>clusters</code>: a vector of (community) cluster labels
</p>
</li>
<li>  <p><code>threshold</code>: the threshold above which cohesion is considered
particularly strong
</p>
</li>
<li>  <p><code>C_strong</code>: the thresholded matrix of cohesion values
</p>
</li>
<li>   <p><code>G</code>: the graph whose edges weights are mutual cohesion
</p>
</li>
<li>  <p><code>G_strong</code>: the weighted graph whose edges are those for
which cohesion is particularly strong
</p>
</li>
<li>  <p><code>layout</code>: a FR force-directed layout associated with G
</p>
</li></ul>



<h3>References</h3>

<p>K. S. Berenhaut, K. E. Moore, R. L. Melvin, A social perspective
on perceived distances reveals deep community structure. Proc. Natl. Acad.
Sci., 119(4), 2022.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>D &lt;- dist(exdata2)
pald_results &lt;- pald(D)
pald_results$local_depths
pald(D, layout = as.matrix(exdata2), show_labels = FALSE)

C &lt;- cohesion_matrix(D)
local_depths(C)
plot_community_graphs(C, layout = as.matrix(exdata2), show_labels = FALSE)

pald_languages &lt;- pald(cognate_dist)
head(pald_languages$local_depths)

</code></pre>

<hr>
<h2 id='pald_colors'>PaLD Color Palette</h2><span id='topic+pald_colors'></span>

<h3>Description</h3>

<p>A vector of colors to use if comparing other clustering methods. These are
the default colors used in the plotting functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pald_colors
</code></pre>


<h3>Format</h3>

<p>A vector of 24 colors
</p>

<hr>
<h2 id='plot_community_graphs'>Plot Community Graphs</h2><span id='topic+plot_community_graphs'></span>

<h3>Description</h3>

<p>Provides a plot of the community graphs, with connected components of the
graph of strong ties colored by connected component.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_community_graphs(
  c,
  show_labels = TRUE,
  only_strong = FALSE,
  emph_strong = 2,
  edge_width_factor = 50,
  colors = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_community_graphs_+3A_c">c</code></td>
<td>
<p>A <code>cohesion_matrix</code> object, a matrix of cohesion values
(see <code><a href="#topic+cohesion_matrix">cohesion_matrix</a></code>).</p>
</td></tr>
<tr><td><code id="plot_community_graphs_+3A_show_labels">show_labels</code></td>
<td>
<p>Set to <code>FALSE</code> to omit vertex labels (to display a subset
of labels, use optional parameter <code>vertex.label</code> to modify the label list).
Default: <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_community_graphs_+3A_only_strong">only_strong</code></td>
<td>
<p>Set to <code>TRUE</code> if only strong ties, G_strong, should be
displayed; the default <code>FALSE</code> will show both strong (colored by connected
component) and weak ties (in gray).</p>
</td></tr>
<tr><td><code id="plot_community_graphs_+3A_emph_strong">emph_strong</code></td>
<td>
<p>Numeric. The numeric factor by which the edge widths of
strong ties are emphasized in the display; the default is <code>2</code>.</p>
</td></tr>
<tr><td><code id="plot_community_graphs_+3A_edge_width_factor">edge_width_factor</code></td>
<td>
<p>Numeric. Modify to change displayed edge widths.
Default: <code>50</code>.</p>
</td></tr>
<tr><td><code id="plot_community_graphs_+3A_colors">colors</code></td>
<td>
<p>A vector of display colors, if none is given a default list
(of length 24) is provided.</p>
</td></tr>
<tr><td><code id="plot_community_graphs_+3A_...">...</code></td>
<td>
<p>Optional parameters to pass to the <code><a href="igraph.html#topic+plot.igraph">igraph::plot.igraph</a></code>.
function. Some commonly passed arguments include:
</p>

<ul>
<li> <p><code>layout</code>  A layout for the graph.  If none is specified, FR-graph
drawing algorithm is used.
</p>
</li>
<li> <p><code>vertex.label</code> A vector containing label names. If none is given,
the rownames of <code>c</code> are used
</p>
</li>
<li> <p><code>vertex.size</code> A numeric value for vertex size (default = <code>1</code>)
</p>
</li>
<li> <p><code>vertex.color.vec</code> A vector of color names for coloring the vertices
</p>
</li>
<li> <p><code>vertex.label.cex</code> A numeric value for modifying the vertex label size.
(default = <code>1</code>)
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the community graph, G, with the sub-graph of strong ties emphasized
and colored by connected component.  If no layout is provided, the
Fruchterman-Reingold (FR) graph drawing algorithm is used.
Note that the FR graph drawing algorithm may provide a somewhat different
layout each time it is run.  You can also access and save a given graph
layout using <code>community_graphs(C)$layout</code>.
The example below shows how to display only a subset of vertex labels.
</p>
<p>Note that the parameter <code>emph_strong</code> is for visualization purposes
only and does not influence the network layout.
</p>


<h3>Value</h3>

<p>A plot of the community graphs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>C &lt;- cohesion_matrix(dist(exdata1))
plot_community_graphs(C, emph_strong = 1, layout = as.matrix(exdata1))
plot_community_graphs(C, only_strong = TRUE)

C2 &lt;- cohesion_matrix(cognate_dist)
subset_lang_names &lt;- rownames(C2)
subset_lang_names[sample(1:87, 60)] &lt;- ""
plot_community_graphs(C2, vertex.label = subset_lang_names, vertex.size = 3)
</code></pre>

<hr>
<h2 id='strong_threshold'>Cohesion Threshold for Strong Ties</h2><span id='topic+strong_threshold'></span>

<h3>Description</h3>

<p>Given a cohesion matrix, provides the value of the threshold above which
values of cohesion are considered &quot;particularly strong&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strong_threshold(c)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="strong_threshold_+3A_c">c</code></td>
<td>
<p>A <code>cohesion_matrix</code> object, a matrix of cohesion values
(see <code><a href="#topic+cohesion_matrix">cohesion_matrix</a></code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The threshold considered in Berenhaut, Moore, and Melvin (2022) which may be
used for distinguishing between strong and weak ties.
The threshold is equal to half the average of the diagonal of the cohesion
matrix, see Berenhaut, Moore, and Melvin (2022).
</p>


<h3>Value</h3>

<p>The value of the threshold.
</p>


<h3>References</h3>

<p>K. S. Berenhaut, K. E. Moore, R. L. Melvin, A social perspective
on perceived distances reveals deep community structure. Proc. Natl. Acad.
Sci., 119(4), 2022.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>C &lt;- cohesion_matrix(dist(exdata1))
strong_threshold(C)
mean(diag(C)) / 2

## points whose cohesion are greater than the threshold may be considered
## (strong) neighbors
which(C[3, ] &gt; strong_threshold(C))

## note that the number of (strongly-cohesive) neighbors varies across the
## space
which(C[4, ] &gt; strong_threshold(C))
C[4, c(2, 3, 4, 6)] # cohesion values can provide neighbor weights

</code></pre>

<hr>
<h2 id='tissue_dist'>Tissue Data Distance Matrix</h2><span id='topic+tissue_dist'></span>

<h3>Description</h3>

<p>A <code><a href="stats.html#topic+dist">dist</a></code> object describing distances from a subset of tissue gene
expression data from the following papers:
</p>

<ul>
<li><p> http://www.ncbi.nlm.nih.gov/pubmed/17906632
</p>
</li>
<li><p> http://www.ncbi.nlm.nih.gov/pubmed/21177656
</p>
</li>
<li><p> http://www.ncbi.nlm.nih.gov/pubmed/24271388
obtained from the <strong>tissuesGeneExpression</strong> bioconductor package.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>tissue_dist
</code></pre>


<h3>Format</h3>

<p>A <code><a href="stats.html#topic+dist">dist</a></code> object of 189 tissue types
</p>


<h3>Details</h3>

<p>The original data frame had 189 rows,  each with a corresponding tissue,
such as <code>colon</code>, <code>kidney</code> or <code>cerebellum</code>.
There were 22,215 columns corresponding to gene expression data from each of
these rows. This was then converted into a distance matrix.
</p>


<h3>References</h3>

<p>M. Love and R. Irizarry. tissueGeneExpression. Bioconductor
Package
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
