<!DOCTYPE html><html lang="en"><head><title>Help for package paleoDiv</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {paleoDiv}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ab.gg'><p>Make a data.frame() that can be used to plot diversity data with density plots, e.g. in ggplot2</p></a></li>
<li><a href='#abdistr_'><p>Count number of entries in occurrence or collection data.frame for specific points in geological time</p></a></li>
<li><a href='#add.alpha'><p>Add transparency to any color</p></a></li>
<li><a href='#ages_archosauria'><p>ages_archosauria</p></a></li>
<li><a href='#archosauria'><p>archosauria</p></a></li>
<li><a href='#convert.sptab'><p>Convert geological ages in taxon-range tables as constructed by mk.sptab() for plotting alongside a time-calibrated phylogeny.</p></a></li>
<li><a href='#darken'><p>Darken or lighten colors by adding/subtracting to or hsv channel values</p></a></li>
<li><a href='#div.gg'><p>Make a data.frame() that can be used to plot diversity data with density plots, e.g. in ggplot2</p></a></li>
<li><a href='#divdistr_'><p>Calculate total species diversity for any point in time based on a taxon-range table</p></a></li>
<li><a href='#divdistr_int'><p>Count number of taxon records overlapping a specific time interval.</p></a></li>
<li><a href='#diversity_table'><p>diversity_table</p></a></li>
<li><a href='#ggcol'><p>Replicate the standard color scheme from ggplot2</p></a></li>
<li><a href='#jitterp'><p>plot data as a jitter-plot</p></a></li>
<li><a href='#mk.sptab'><p>Generate a taxon-range table based on an occurrence dataset.</p></a></li>
<li><a href='#multijitter'><p>Wrapper around jitterp that plots multiple jitter plots on the same plotting device (analogous to violins())</p></a></li>
<li><a href='#occ.cleanup'><p>Clean up occurrence dataset by removing commonly used character combinations in the identified name that will result in different factor levels for the same taxon.</p></a></li>
<li><a href='#pdb'><p>Download data from the paleobiology database.</p></a></li>
<li><a href='#pdb.autodiv'><p>A wrapper around pdb(), occ.cleanup() and mk.sptab() to automatically download and clean occurrence data from the paleobiology database and build species-level taxon-range tables for multiple taxa in one step.</p></a></li>
<li><a href='#pdb.diff'><p>Subtract one occurrence data.frame from another, for disentangling overlapping taxonomies or quantifying stem-lineage diversity.</p></a></li>
<li><a href='#pdb.union'><p>Form the union of two occurrence data.frames or remove duplicates from occurrence data.frame. Useful if parts of a clade are not included in the downloaded dataset and need to be added separately.</p></a></li>
<li><a href='#phylo.spindles'><p>Plots a phylogenetic tree with spindle-diagrams, optimized for showing taxonomic diversity.</p></a></li>
<li><a href='#redraw.phylo'><p>Redraw the lines of a phylogenetic tree.</p></a></li>
<li><a href='#rmean'><p>Calculate a rolling mean for a vector x.</p></a></li>
<li><a href='#rmeana'><p>Calculate a rolling mean based on distance within a second variable.</p></a></li>
<li><a href='#stax.sel'><p>Extract subsets of an occurrence data.frame.</p></a></li>
<li><a href='#synonymize'><p>Combine selected entries in a taxon-range table to remove duplicates</p></a></li>
<li><a href='#tree_archosauria'><p>tree_archosauria</p></a></li>
<li><a href='#tree.age.combine'><p>Combine two calibration matrixes and fill in NA values in one with values from another</p></a></li>
<li><a href='#tree.ages'><p>Automatically build matrix for time-calibration of phylogenetic trees using occurrence data.</p></a></li>
<li><a href='#tree.ages.spp'><p>Automatically build matrix for time-calibration of phylogenetic trees using occurrence data.</p></a></li>
<li><a href='#ts.periods'><p>Add a horizontal, period-level phanerozoic timescale to any plot, especially calibrated phylogenies plotted with ape.</p></a></li>
<li><a href='#ts.stages'><p>Add a horizontal, stage-level phanerozoic timescale to any plot, especially calibrated phylogenies plotted with ape.</p></a></li>
<li><a href='#tsconv'><p>Convert geological ages for accurate plotting alongside a calibrated phylogeny</p></a></li>
<li><a href='#viol'><p>Generate a violin plot</p></a></li>
<li><a href='#violins'><p>Wrapper around viol() to conveniently plot multiple violins on a single plot, analogous to the behavior of boxplot()</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Extracting and Visualizing Paleobiodiversity</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains various tools for conveniently downloading and editing taxon-specific datasets from the Paleobiology Database <a href="https://paleobiodb.org">https://paleobiodb.org</a>, extracting information on abundance, temporal distribution of subtaxa and taxonomic diversity through deep time, and visualizing these data in relation to phylogeny and stratigraphy.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), strap, divDyn, ggplot2, knitr, rmarkdown</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), ape, stringr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-17 13:12:02 UTC; suirad</td>
</tr>
<tr>
<td>Author:</td>
<td>Darius Nau <a href="https://orcid.org/0009-0000-4343-6830"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Darius Nau &lt;dariusnau@gmx.at&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-17 13:30:04 UTC</td>
</tr>
</table>
<hr>
<h2 id='ab.gg'>Make a data.frame() that can be used to plot diversity data with density plots, e.g. in ggplot2</h2><span id='topic+ab.gg'></span>

<h3>Description</h3>

<p>Make a data.frame() that can be used to plot diversity data with density plots, e.g. in ggplot2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ab.gg(data, taxa = NULL, agerange = c(252, 66), precision_ma = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ab.gg_+3A_data">data</code></td>
<td>
<p>list()-object containing occurrence data.frames or single occurrence data.frame()</p>
</td></tr>
<tr><td><code id="ab.gg_+3A_taxa">taxa</code></td>
<td>
<p>Selection of taxa to include. If NULL, then abundance is tabulated for each unique factor level of data$tna</p>
</td></tr>
<tr><td><code id="ab.gg_+3A_agerange">agerange</code></td>
<td>
<p>Range of geological ages to include in data.frame()</p>
</td></tr>
<tr><td><code id="ab.gg_+3A_precision_ma">precision_ma</code></td>
<td>
<p>Size of intervals (in ma) at which to calculate diversity within the age range.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each taxon receives one entry per occurrence per time interval. The number of entries per taxon at any given point is thus proportional to the abundance of the taxon in the fossil record, and can be used for plotting with frequency- or density-based functions (e.g. hist(), ggplot2::geom_violin(), etc.). Note that using age values in the original occurrence table instead of this function will often be fully sufficient if the number of occurrences is considered an adequate proxy for abundance. However, instead using the ab.gg() and thus visualizing the results of the abdistr_() function has the benefit of the ability to account for a column of abundance values within the occurrence dataset, if available.
</p>


<h3>Value</h3>

<p>A data.frame() with two columns: ma, for the numerical age, and tax, for the taxon.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
ab.gg(data=archosauria, taxa=c("Ankylosauria","Stegosauria"))-&gt;thyreophora
library(ggplot2)
ggplot(data=thyreophora, aes(x=tax, y=ma, col=tax))+ylim(252,0)+geom_violin(scale="count")
</code></pre>

<hr>
<h2 id='abdistr_'>Count number of entries in occurrence or collection data.frame for specific points in geological time</h2><span id='topic+abdistr_'></span>

<h3>Description</h3>

<p>Count number of entries in occurrence or collection data.frame for specific points in geological time
</p>


<h3>Usage</h3>

<pre><code class='language-R'>abdistr_(
  x,
  table = NULL,
  ab.val = table$abund_value,
  ab.val.na = 1,
  smooth = 0,
  max = table$eag,
  min = table$lag,
  w = rep(1, length(x))
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="abdistr__+3A_x">x</code></td>
<td>
<p>A numeric vector giving the times (in ma) at which to determine the number of overlapping records.</p>
</td></tr>
<tr><td><code id="abdistr__+3A_table">table</code></td>
<td>
<p>An occurrence or collection dataset</p>
</td></tr>
<tr><td><code id="abdistr__+3A_ab.val">ab.val</code></td>
<td>
<p>Abundance value to be used. Default is table$abund_value. If set to 1, each occurrence is treated as representing one specimen. If NULL (e.g. because this column does not exist) or NA, each occurrence is treated as the number of specimens specified under ab.val.na</p>
</td></tr>
<tr><td><code id="abdistr__+3A_ab.val.na">ab.val.na</code></td>
<td>
<p>Value to substitute for missing entries in abundance values. Defaults to 1. Either a single numeric or a function to be applied to all non-missing entries of ab.val (e.g. mean()  or median()).</p>
</td></tr>
<tr><td><code id="abdistr__+3A_smooth">smooth</code></td>
<td>
<p>The smoothing margin, in units of ma. Corresponds to the plusminus parameter of rmeana(). Defaults to 0, i.e. no smoothing (beyond the resolution determined by the resolution of x)</p>
</td></tr>
<tr><td><code id="abdistr__+3A_max">max</code></td>
<td>
<p>Vector or column containing maximum age of each occurrence or collection</p>
</td></tr>
<tr><td><code id="abdistr__+3A_min">min</code></td>
<td>
<p>Vector or column containing minimum age of each occurrence or collection</p>
</td></tr>
<tr><td><code id="abdistr__+3A_w">w</code></td>
<td>
<p>A Vector of weights. Must be of same length as x</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of the same length as x, giving the estimated number of occurrence records (if ab.val==FALSE) or specimens (if ab.val==TRUE), or the estimated number of collections (if collection data are used instead of occurrences) overlapping each temporal value given in x
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
abdistr_(x=c(170:120), table=archosauria$Stegosauria)
</code></pre>

<hr>
<h2 id='add.alpha'>Add transparency to any color</h2><span id='topic+add.alpha'></span>

<h3>Description</h3>

<p>Add transparency to any color
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add.alpha(col, alpha = 0.5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add.alpha_+3A_col">col</code></td>
<td>
<p>Color value or vector of colors</p>
</td></tr>
<tr><td><code id="add.alpha_+3A_alpha">alpha</code></td>
<td>
<p>Opacity value to apply to the color(s)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing color hex codes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>add.alpha("red",0.8)
</code></pre>

<hr>
<h2 id='ages_archosauria'>ages_archosauria</h2><span id='topic+ages_archosauria'></span>

<h3>Description</h3>

<p>A dataset containing earliest and latest occurrence dates for clades shown in the example phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ages_archosauria
</code></pre>


<h3>Format</h3>

<p>A matrix with 13 rows and 2 collumns containing:
</p>

<dl>
<dt>FAD</dt><dd><p>Earliest occurrence age</p>
</dd>
<dt>LAD</dt><dd><p>Latest occurrence age</p>
</dd>
</dl>

<p>â€¦for each taxon
</p>

<hr>
<h2 id='archosauria'>archosauria</h2><span id='topic+archosauria'></span>

<h3>Description</h3>

<p>A dataset of stratigraphic ranges of species within the clades in tree_archosauria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>archosauria
</code></pre>


<h3>Format</h3>

<p>A list() object containing 2 occurrence data.frames, 1 collections data.frame and 15 species-range tables (all as data.frames) with the following data in each:
</p>

<dl>
<dt>tna</dt><dd><p>taxon names (species names)</p>
</dd>
<dt>max</dt><dd><p>maximum ages</p>
</dd>
<dt>min</dt><dd><p>minimum ages</p>
</dd>
<dt>ma</dt><dd><p>mean ages</p>
</dd>
</dl>



<h3>Source</h3>

<p>Generated from data downloaded from the paleobiology database <a href="https://paleobiodb.org">https://paleobiodb.org</a> using the functions pdb(), occ.cleanup() and mk.sptab()
</p>

<hr>
<h2 id='convert.sptab'>Convert geological ages in taxon-range tables as constructed by mk.sptab() for plotting alongside a time-calibrated phylogeny.</h2><span id='topic+convert.sptab'></span>

<h3>Description</h3>

<p>Convert geological ages in taxon-range tables as constructed by mk.sptab() for plotting alongside a time-calibrated phylogeny.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert.sptab(sptab, tree = NULL, root.time = tree$root.time)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert.sptab_+3A_sptab">sptab</code></td>
<td>
<p>Taxon-range table to convert</p>
</td></tr>
<tr><td><code id="convert.sptab_+3A_tree">tree</code></td>
<td>
<p>Optional phylogenetic tree to draw root.time from</p>
</td></tr>
<tr><td><code id="convert.sptab_+3A_root.time">root.time</code></td>
<td>
<p>Root time of the tree, used for converting ages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame() object in the format of the original taxon-range table, but with geological ages converted for plotting alongside the the phylogenetic tree.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
data(tree_archosauria)
convert.sptab(archosauria$sptab_Coelophysoidea,tree_archosauria)
</code></pre>

<hr>
<h2 id='darken'>Darken or lighten colors by adding/subtracting to or hsv channel values</h2><span id='topic+darken'></span>

<h3>Description</h3>

<p>Darken or lighten colors by adding/subtracting to or hsv channel values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>darken(x, add = 0, abs = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="darken_+3A_x">x</code></td>
<td>
<p>Color value or vector of colors</p>
</td></tr>
<tr><td><code id="darken_+3A_add">add</code></td>
<td>
<p>Value to be added to the third hsv-channel. Can be a vector of length x, or a vector of any length if length(x)==1</p>
</td></tr>
<tr><td><code id="darken_+3A_abs">abs</code></td>
<td>
<p>Value to substitute for the third hsv-channel. If set, this overrides the setting for parameter add. Can be a vector of length x, or a vector of any length if length(x)==1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A color value or vector of color values of length x (or, if length(x)==1, the length of add or abs)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>darken(ggcol(3),abs=0.5)
</code></pre>

<hr>
<h2 id='div.gg'>Make a data.frame() that can be used to plot diversity data with density plots, e.g. in ggplot2</h2><span id='topic+div.gg'></span>

<h3>Description</h3>

<p>Make a data.frame() that can be used to plot diversity data with density plots, e.g. in ggplot2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>div.gg(data, taxa, agerange = c(252, 66), precision_ma = 1, prefix = "sptab_")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="div.gg_+3A_data">data</code></td>
<td>
<p>list()-object containing taxon-range tables</p>
</td></tr>
<tr><td><code id="div.gg_+3A_taxa">taxa</code></td>
<td>
<p>Selection of taxa to include</p>
</td></tr>
<tr><td><code id="div.gg_+3A_agerange">agerange</code></td>
<td>
<p>Range of geological ages to include in data.frame()</p>
</td></tr>
<tr><td><code id="div.gg_+3A_precision_ma">precision_ma</code></td>
<td>
<p>Size of intervals (in ma) at which to calculate diversity within the age range.</p>
</td></tr>
<tr><td><code id="div.gg_+3A_prefix">prefix</code></td>
<td>
<p>Prefix under which to find taxon-range tables in data</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each taxon receives one entry per subtaxon (e.g. species) occurring for each time interval at which it occurs. The number of entries per taxon at any given point is thus proportional to the diversity of the taxon, and can be used to trick density functions (e.g. hist(), density()) into plotting diversity diagrams of various types. This is most useful when using ggplot2::geom_violin(), geom_histogram() or geom_density() functions.
A simpler alternative to achieve a similar result would be to use the taxon-range-tables directly with these functions. However, this will lead to a relative underestimate of diversity for taxa with long-lived subtaxa, since each subtaxon will only be counted once. The div.gg()-function circumvents this problem by representing each taxon for each time interval in which it occurs, i.e. the relative number of entries in the returned data.frame will be proportional to the relative number of taxa with ranges overlapping each point in time.
</p>


<h3>Value</h3>

<p>A data.frame() with two columns: ma, for the numerical age, and tax, for the taxon.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
div.gg(archosauria, taxa=c("Pterosauria","Aves"), agerange=c(252,0),precision_ma=1)-&gt;flyers
library(ggplot2)
ggplot(data=flyers, aes(x=tax, y=ma))+ylim(252,0)+geom_violin(scale="count")
ggplot(data=flyers, aes(col=tax, x=ma))+xlim(252,0)+geom_density(adjust=0.5)
</code></pre>

<hr>
<h2 id='divdistr_'>Calculate total species diversity for any point in time based on a taxon-range table</h2><span id='topic+divdistr_'></span>

<h3>Description</h3>

<p>Calculate total species diversity for any point in time based on a taxon-range table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>divdistr_(
  x,
  table = NULL,
  w = rep(1, length(x)),
  smooth = 0,
  max = table$max,
  min = table$min
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="divdistr__+3A_x">x</code></td>
<td>
<p>A point in time or vector of points in time, in ma, at which species diversity is to be determined.</p>
</td></tr>
<tr><td><code id="divdistr__+3A_table">table</code></td>
<td>
<p>A taxon-range table to be used, usually the output of mk.sptab()</p>
</td></tr>
<tr><td><code id="divdistr__+3A_w">w</code></td>
<td>
<p>A vector of weights to apply to the estimated (raw) diversity figures. This vector needs to be of the same length as x. Each raw diversity estimate will then be multiplied by the weight. Can be used to account for differences in collection intensity/sampling biases, if these can be quantified (e.g. by analyzing collection records.</p>
</td></tr>
<tr><td><code id="divdistr__+3A_smooth">smooth</code></td>
<td>
<p>The smoothing margin, in units of ma. Corresponds to the plusminus parameter of rmeana(). Defaults to 0, i.e. no smoothing (beyond the resolution determined by the resolution of x)</p>
</td></tr>
<tr><td><code id="divdistr__+3A_max">max</code></td>
<td>
<p>Vector or column containing the maximum age of each entry in the taxon-range table. Defaults to table$max</p>
</td></tr>
<tr><td><code id="divdistr__+3A_min">min</code></td>
<td>
<p>Vector or column containing the minimum age of each entry in the taxon-range table. Defaults to table$min</p>
</td></tr>
</table>


<h3>Details</h3>

<p>divdistr_() produces a &quot;maximum&quot; estimate of taxonomic diversity at any given point in time in the fossil record. This function is based on the principle of counting the number of taxon ranges (from the provided range table) that overlap each age provided in x. As a result of uncertainty of age estimates, this may lead to an overestimation of the actual fossil diversity at each point in time, especially at the points of overlap between taxon-specific ranges. Moreover this represents a &quot;raw&quot;, uncorrected diversity estimate that does not account for differences in sampling intensity throughout the time interval that is investigated. A rudimentary functionality for using such a correction exists in the form of the w argument, which allows the user to provide a vector of weights (of the same length as x) to be multiplied with the raw diversity estimates. Such weights can, for instance, be based on (the inverse of) the number of collections overlapping any given age in x, which can be calculated using the same basic approach as the raw diversity, by downloading collections instead of occurrence data.
</p>


<h3>Value</h3>

<p>A numeric vector containing taxon diversity (at the chosen taxonomic level used in the generation of the range table) at the provided ages.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
divdistr_(c(170:140),table=archosauria$sptab_Stegosauria)
curve(divdistr_(x,archosauria$sptab_Stegosauria), xlim=c(200,100),ylim=c(-5,35))
ts.stages(ylim=c(-6,-1),alpha=0.3,border=add.alpha("grey"))
ts.periods(ylim=c(-6,-1),alpha=0.0)
</code></pre>

<hr>
<h2 id='divdistr_int'>Count number of taxon records overlapping a specific time interval.</h2><span id='topic+divdistr_int'></span>

<h3>Description</h3>

<p>Count number of taxon records overlapping a specific time interval.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>divdistr_int(x, table = NULL, ids = FALSE, max = table$max, min = table$min)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="divdistr_int_+3A_x">x</code></td>
<td>
<p>A numeric vector of length 2 specifying the start and end (in ma) of the time interval in question.</p>
</td></tr>
<tr><td><code id="divdistr_int_+3A_table">table</code></td>
<td>
<p>Taxon-range table to use</p>
</td></tr>
<tr><td><code id="divdistr_int_+3A_ids">ids</code></td>
<td>
<p>Logical whether to return ids of entries in taxon-range table (defaults to FALSE) or their number</p>
</td></tr>
<tr><td><code id="divdistr_int_+3A_max">max</code></td>
<td>
<p>Vector or column containing the maximum age of each entry in the taxon-range table. Defaults to table$max</p>
</td></tr>
<tr><td><code id="divdistr_int_+3A_min">min</code></td>
<td>
<p>Vector or column containing the minimum age of each entry in the taxon-range table. Defaults to table$min</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single numeric giving the number of entries in table overlapping the specified interval, or a numeric vector giving their indices.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
divdistr_int(x=c(201,220), table=archosauria$sptab_Coelophysoidea)
</code></pre>

<hr>
<h2 id='diversity_table'>diversity_table</h2><span id='topic+diversity_table'></span>

<h3>Description</h3>

<p>A dataset of diversity by stage, exemplifying the output produced by the divDyn-package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diversity_table
</code></pre>


<h3>Format</h3>

<p>A data.frame() containing mean ages and diversity figures by stage.
</p>

<dl>
<dt>x_orig</dt><dd><p>ages for each stage in the phanerozoic</p>
</dd>
<dt>x</dt><dd><p>ages converted for plotting on tree_archosauria, using the tsconv()-function</p>
</dd>
<dt>Sauroporomorpha</dt><dd><p>diversity by stage for Sauropodomorpha</p>
</dd>
<dt>etc</dt><dd><p>diversity by stage for each of the taxa represented in tree_archosauria</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='ggcol'>Replicate the standard color scheme from ggplot2</h2><span id='topic+ggcol'></span>

<h3>Description</h3>

<p>Replicate the standard color scheme from ggplot2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggcol(n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggcol_+3A_n">n</code></td>
<td>
<p>Length of color vector to return.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing color hex codes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ggcol(3)
</code></pre>

<hr>
<h2 id='jitterp'>plot data as a jitter-plot</h2><span id='topic+jitterp'></span>

<h3>Description</h3>

<p>plot data as a jitter-plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jitterp(x, y, width, col = "black", alpha = 0.5, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="jitterp_+3A_x">x</code></td>
<td>
<p>x values to plot (if single value and y is a vector, plot is vertical)</p>
</td></tr>
<tr><td><code id="jitterp_+3A_y">y</code></td>
<td>
<p>y value at which to plot (if single value, plot is horizontal)</p>
</td></tr>
<tr><td><code id="jitterp_+3A_width">width</code></td>
<td>
<p>standard deviation for jitter</p>
</td></tr>
<tr><td><code id="jitterp_+3A_col">col</code></td>
<td>
<p>color for points</p>
</td></tr>
<tr><td><code id="jitterp_+3A_alpha">alpha</code></td>
<td>
<p>opacity for points</p>
</td></tr>
<tr><td><code id="jitterp_+3A_...">...</code></td>
<td>
<p>other parameters to be passed on to points()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>adds the points to the open plotting device as a jitter plot and returns an invisible list()-object containing the positions of all points
</p>


<h3>Examples</h3>

<pre><code class='language-R'>c(1,2,3,2,3,2,3,4,4)-&gt;tmp
hist(tmp)
jitterp(x=tmp, y=1, width=0.1)
</code></pre>

<hr>
<h2 id='mk.sptab'>Generate a taxon-range table based on an occurrence dataset.</h2><span id='topic+mk.sptab'></span>

<h3>Description</h3>

<p>Generate a taxon-range table based on an occurrence dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mk.sptab(
  xx = NULL,
  taxa = xx$tna,
  earliest = xx$eag,
  latest = xx$lag,
  tax = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mk.sptab_+3A_xx">xx</code></td>
<td>
<p>A data.frame() of occurrence records, containing at least the following columns: taxonomic name at level at which ranges are to be determined (e.g. species or genus), earliest possible age for each occurrence and latest possible age for each occurrence. If xx==NULL, then each column or vector must be specified individualy using the following parameters</p>
</td></tr>
<tr><td><code id="mk.sptab_+3A_taxa">taxa</code></td>
<td>
<p>column/vector containing the taxonomic variable. Defaults to xx$tna</p>
</td></tr>
<tr><td><code id="mk.sptab_+3A_earliest">earliest</code></td>
<td>
<p>column/vector containing the earliest age estimate. Defaults to xx$eag.</p>
</td></tr>
<tr><td><code id="mk.sptab_+3A_latest">latest</code></td>
<td>
<p>column/vector containing the latest age estimate. Defaults to xx$lag.</p>
</td></tr>
<tr><td><code id="mk.sptab_+3A_tax">tax</code></td>
<td>
<p>Optional. A single character string containing the taxon name, to be added as another column to the range table (useful for categorization, should several range tables be concatenated, e.g. using rbind()).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame() containing the taxon names, the maximum and minimum age for each taxon, and (optionally) a column with the name of the higher-level taxon.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
mk.sptab(archosauria$Stegosauria)-&gt;sptab_Stegosauria
</code></pre>

<hr>
<h2 id='multijitter'>Wrapper around jitterp that plots multiple jitter plots on the same plotting device (analogous to violins())</h2><span id='topic+multijitter'></span>

<h3>Description</h3>

<p>Wrapper around jitterp that plots multiple jitter plots on the same plotting device (analogous to violins())
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multijitter(
  x,
  data = NULL,
  group = NULL,
  horiz = FALSE,
  order = NULL,
  xlab = "",
  ylab = "",
  col = "black",
  pch = 16,
  spaces = "_",
  width = 0.1,
  xlim = NULL,
  ylim = NULL,
  add = TRUE,
  ax = FALSE,
  srt = 45,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multijitter_+3A_x">x</code></td>
<td>
<p>plotting statistic (numeric vector) or formula object from which a plotting statistic and grouping variable can be extracter (i.e. of form x~group)</p>
</td></tr>
<tr><td><code id="multijitter_+3A_data">data</code></td>
<td>
<p>data.frame object containing x and y</p>
</td></tr>
<tr><td><code id="multijitter_+3A_group">group</code></td>
<td>
<p>grouping variable</p>
</td></tr>
<tr><td><code id="multijitter_+3A_horiz">horiz</code></td>
<td>
<p>logical indicating whether to plot horizontally</p>
</td></tr>
<tr><td><code id="multijitter_+3A_order">order</code></td>
<td>
<p>order of factor levels of categorical factor</p>
</td></tr>
<tr><td><code id="multijitter_+3A_xlab">xlab</code></td>
<td>
<p>x axis label</p>
</td></tr>
<tr><td><code id="multijitter_+3A_ylab">ylab</code></td>
<td>
<p>y axis label</p>
</td></tr>
<tr><td><code id="multijitter_+3A_col">col</code></td>
<td>
<p>vector of border colors</p>
</td></tr>
<tr><td><code id="multijitter_+3A_pch">pch</code></td>
<td>
<p>vector of symbols</p>
</td></tr>
<tr><td><code id="multijitter_+3A_spaces">spaces</code></td>
<td>
<p>character string in group to replace with spaces for labels, if not NULL</p>
</td></tr>
<tr><td><code id="multijitter_+3A_width">width</code></td>
<td>
<p>standard deviation for jitter</p>
</td></tr>
<tr><td><code id="multijitter_+3A_xlim">xlim</code></td>
<td>
<p>x limits (data limits used if NULL)</p>
</td></tr>
<tr><td><code id="multijitter_+3A_ylim">ylim</code></td>
<td>
<p>y limits (data limits used if NULL)</p>
</td></tr>
<tr><td><code id="multijitter_+3A_add">add</code></td>
<td>
<p>logical whether to add to existing plot (default: TRUE)</p>
</td></tr>
<tr><td><code id="multijitter_+3A_ax">ax</code></td>
<td>
<p>whether to plot axes</p>
</td></tr>
<tr><td><code id="multijitter_+3A_srt">srt</code></td>
<td>
<p>angle for categorical axis text rotation</p>
</td></tr>
<tr><td><code id="multijitter_+3A_...">...</code></td>
<td>
<p>other arguments to pass on to jitterp() and plot()</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data.frame(p=rnorm(50), cat=rep(c("A","B","B","B","B"),10))-&gt;d
multijitter(p~cat,d, add=FALSE)
</code></pre>

<hr>
<h2 id='occ.cleanup'>Clean up occurrence dataset by removing commonly used character combinations in the identified name that will result in different factor levels for the same taxon.</h2><span id='topic+occ.cleanup'></span>

<h3>Description</h3>

<p>Clean up occurrence dataset by removing commonly used character combinations in the identified name that will result in different factor levels for the same taxon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>occ.cleanup(x, remove = NULL, return.df = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="occ.cleanup_+3A_x">x</code></td>
<td>
<p>A occurrence data.frame or character vector containing the variable to clean up (defaults to x$tna)</p>
</td></tr>
<tr><td><code id="occ.cleanup_+3A_remove">remove</code></td>
<td>
<p>Which values to remove. If NULL, a default set of commonly occurring character combinations is used (&quot;n. gen.&quot;, &quot;n. sp.&quot;, &quot;cf.&quot;,&quot;aff.&quot;, punctuation, as well as double, leading and ending spaces). If user-defined, remove needs to be formatted as a character vector with the values to be removed as names, i.e. in the format of c(&quot;remove_this&quot; = &quot;&quot;, &quot;removethistoo&quot;=&quot;&quot;)</p>
</td></tr>
<tr><td><code id="occ.cleanup_+3A_return.df">return.df</code></td>
<td>
<p>A logical indicating whether to return the entire data.frame (if TRUE) or just the column of taxonomic names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the cleaned up taxonomic names or a dataframe with cleaned-up tna column (if return.df==TRUE).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
occ.cleanup(archosauria$Stegosauria)-&gt;archosauria$Stegosauria
</code></pre>

<hr>
<h2 id='pdb'>Download data from the paleobiology database.</h2><span id='topic+pdb'></span>

<h3>Description</h3>

<p>Download data from the paleobiology database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pdb(
  taxon,
  interval = "all",
  what = "occs",
  full = FALSE,
  base = "https://paleobiodb.org/data1.2/",
  file = "list.csv",
  cc = NULL,
  envtype = NULL,
  append_additional = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pdb_+3A_taxon">taxon</code></td>
<td>
<p>A taxon (base_name) for which to download records.</p>
</td></tr>
<tr><td><code id="pdb_+3A_interval">interval</code></td>
<td>
<p>A character string indicating over which temporal interval to download data (defaults to &quot;all&quot;), e.g. &quot;Phanerozoic&quot; or &quot;Jurassic&quot;.</p>
</td></tr>
<tr><td><code id="pdb_+3A_what">what</code></td>
<td>
<p>The type of data to download (for details, see https://paleobiodb.org/data1.2/). Defaults to &quot;occs&quot;, which downloads occurrence data. Setting this parameter to &quot;colls&quot; will instead download collection data.</p>
</td></tr>
<tr><td><code id="pdb_+3A_full">full</code></td>
<td>
<p>A logical indicating whether or not the full dataset is to be downloaded (defaults to FALSE). At the expense of larger file size, the full dataset contains a large number of additional columns containing data such as stratigraphy, phylogeny and (paleo)geography, which is useful for various purposes but not strictly necessary for graphing paleodiversity.</p>
</td></tr>
<tr><td><code id="pdb_+3A_base">base</code></td>
<td>
<p>Character string containing base url to use. Defaults to https://paleobiodb.org/data1.2/. Entering &quot;dev&quot; serves as a shortcut to use https://dev.paleobiodb.org/data1.2/ instead (can sometimes be helpful if one of the two is unavailable).</p>
</td></tr>
<tr><td><code id="pdb_+3A_file">file</code></td>
<td>
<p>Character string containing which file name to look for. Defaults to list.csv.</p>
</td></tr>
<tr><td><code id="pdb_+3A_cc">cc</code></td>
<td>
<p>Selection for continent (e.g. EUR for Europe, see paleobiodb.org documentation)</p>
</td></tr>
<tr><td><code id="pdb_+3A_envtype">envtype</code></td>
<td>
<p>Selection for environment type (e.g. marine)</p>
</td></tr>
<tr><td><code id="pdb_+3A_append_additional">append_additional</code></td>
<td>
<p>Any additional character string to append to URL for pdb dataset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame() containing the downloaded paleobioDB dataset. The column &quot;identified_name&quot; will be copied into the column &quot;tna&quot;, and (if what==occs) the columns &quot;max_ma&quot; and &quot;min_ma&quot; will be copied into the columns named &quot;eag&quot; and &quot;lag&quot; respectively, maintaining compatibility with the output of the deprecated package &quot;paleobioDB&quot; for those variable names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pdb("Stegosauria")-&gt;Stegosauria
</code></pre>

<hr>
<h2 id='pdb.autodiv'>A wrapper around pdb(), occ.cleanup() and mk.sptab() to automatically download and clean occurrence data from the paleobiology database and build species-level taxon-range tables for multiple taxa in one step.</h2><span id='topic+pdb.autodiv'></span>

<h3>Description</h3>

<p>A wrapper around pdb(), occ.cleanup() and mk.sptab() to automatically download and clean occurrence data from the paleobiology database and build species-level taxon-range tables for multiple taxa in one step.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pdb.autodiv(taxa, cleanup = TRUE, interval = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pdb.autodiv_+3A_taxa">taxa</code></td>
<td>
<p>Either a character vector of valid taxonomic names, or an object of class &quot;phylo&quot; whose tip.labels to use instead.</p>
</td></tr>
<tr><td><code id="pdb.autodiv_+3A_cleanup">cleanup</code></td>
<td>
<p>Logical indicating whether to apply occ.cleanup() to occurrence data after download (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="pdb.autodiv_+3A_interval">interval</code></td>
<td>
<p>Stratigraphic interval for which to download data (defaults to NULL, which downloads data for all intervals)</p>
</td></tr>
<tr><td><code id="pdb.autodiv_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed on to pdb()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list() object containing occurrence data (saved under the taxon names given) and species-level taxon-range tables (saved with the prefix &quot;sptab_&quot; before the taxon names).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pdb.autodiv("Coelophysoidea")-&gt;coelo
</code></pre>

<hr>
<h2 id='pdb.diff'>Subtract one occurrence data.frame from another, for disentangling overlapping taxonomies or quantifying stem-lineage diversity.</h2><span id='topic+pdb.diff'></span>

<h3>Description</h3>

<p>Subtract one occurrence data.frame from another, for disentangling overlapping taxonomies or quantifying stem-lineage diversity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pdb.diff(x, subtract, id_col = x$occurrence_no)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pdb.diff_+3A_x">x</code></td>
<td>
<p>Occurrence data from which to subtract.</p>
</td></tr>
<tr><td><code id="pdb.diff_+3A_subtract">subtract</code></td>
<td>
<p>Occurrence data frame or vector of occurrence numbers to subtract from x</p>
</td></tr>
<tr><td><code id="pdb.diff_+3A_id_col">id_col</code></td>
<td>
<p>Vector or column of x containing id to be used for determining which values are also found in subtract or subtract$occurrence_no</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame() containing the difference between the two occurrence datasets, i.e. all entries that are in x but not in subtract.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
pdb.union(rbind(archosauria$Ankylosauria, archosauria$Stegosauria))-&gt;Eurypoda
pdb.diff(Eurypoda, subtract=archosauria$Stegosauria)
</code></pre>

<hr>
<h2 id='pdb.union'>Form the union of two occurrence data.frames or remove duplicates from occurrence data.frame. Useful if parts of a clade are not included in the downloaded dataset and need to be added separately.</h2><span id='topic+pdb.union'></span>

<h3>Description</h3>

<p>Form the union of two occurrence data.frames or remove duplicates from occurrence data.frame. Useful if parts of a clade are not included in the downloaded dataset and need to be added separately.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pdb.union(x, id_col = x$occurrence_no)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pdb.union_+3A_x">x</code></td>
<td>
<p>Concatenated occurrence data.frames to be merged</p>
</td></tr>
<tr><td><code id="pdb.union_+3A_id_col">id_col</code></td>
<td>
<p>Vector or column of x containing id to be used for determining which values contain occurrence numbers to be used for matching entries</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame() containing the first entry for each unique occurrence to be represented in x.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
pdb.union(rbind(archosauria$Ankylosauria, archosauria$Stegosauria))-&gt;Eurypoda
</code></pre>

<hr>
<h2 id='phylo.spindles'>Plots a phylogenetic tree with spindle-diagrams, optimized for showing taxonomic diversity.</h2><span id='topic+phylo.spindles'></span>

<h3>Description</h3>

<p>Plots a phylogenetic tree with spindle-diagrams, optimized for showing taxonomic diversity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylo.spindles(
  phylo0,
  occ,
  stat = divdistr_,
  prefix = "sptab_",
  pos = NULL,
  ages = NULL,
  xlimits = NULL,
  ylimits = NULL,
  res = 1,
  weights = 1,
  dscale = 0.002,
  col = add.alpha("black"),
  fill = col,
  lwd = 1,
  lty = 1,
  cex.txt = 1,
  col.txt = add.alpha(col, 1),
  axis = TRUE,
  labels = TRUE,
  txt.y = 0.5,
  txt.x = NULL,
  adj.x = NULL,
  add = FALSE,
  tbmar = 0.2,
  smooth = 0,
  italicize = character()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phylo.spindles_+3A_phylo0">phylo0</code></td>
<td>
<p>A time-calibrated phylogenetic tree to plot with spindle diagrams, or a character vector of taxonomic names for which to plot spindle diagrams.</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_occ">occ</code></td>
<td>
<p>Either a list()-object containing taxon-range tables for plotting diversity, or a matrix() or data.frame()-object that contains numerical plotting statistics. If the latter is provided, the default use of divdistr_() is overridden and the function will look for a column named &quot;x&quot; and columns matching the phylogeny tip.labels to plot the spindles.</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_stat">stat</code></td>
<td>
<p>Plotting statistic to be passed on to viol(). Defaults to use divdistr_().</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_prefix">prefix</code></td>
<td>
<p>Prefix for taxon-range tables in occ. Defaults to &quot;sptab_&quot;</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_pos">pos</code></td>
<td>
<p>Position at which to draw spindles. If NULL (default), then spindles are drawn at c(1:n) where n is the number of taxa in phylo0.</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_ages">ages</code></td>
<td>
<p>Optional matrix with lower and upper age limits for each spindle, formatted like the output of tree.ages() (most commonly the same calibration matrix used to time-calibrate the tree)</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_xlimits">xlimits</code></td>
<td>
<p>Limits for plotting on the x axis.</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_ylimits">ylimits</code></td>
<td>
<p>Limits for plotting on the y axis. If NULL (default) or not a numeric vector of length 2, the y limits are instead constructed from the tbmar parameter and the number of entries in the phylogeny or taxon list.</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_res">res</code></td>
<td>
<p>Temporal resolution of diversity estimation (if occ is a matrix or data.frame containing plotting statistics, this is ignored)</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_weights">weights</code></td>
<td>
<p>Weights for diversity estimation. Must have the same length as the range of xlimits divided by res. For details, see divdistr_()</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_dscale">dscale</code></td>
<td>
<p>Scale value of the spindles on the y axis. Should be adjusted manually to optimize visibility of results.</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_col">col</code></td>
<td>
<p>Color to use for the border of the plotted spindles</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_fill">fill</code></td>
<td>
<p>Color to use for the fill of the plotted spindles. Defaults to col.</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_lwd">lwd</code></td>
<td>
<p>Line width for the plotted spindles.</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_lty">lty</code></td>
<td>
<p>Line type for the plotted spindles.</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_cex.txt">cex.txt</code></td>
<td>
<p>Adjustment for tip label text size</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_col.txt">col.txt</code></td>
<td>
<p>tip label text color, defaults to be same as col, but with no transparency</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_axis">axis</code></td>
<td>
<p>Logical indicating whether to plot (temporal) x axis (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_labels">labels</code></td>
<td>
<p>Logical indicating whether to plot tip labels of phylogeny (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_txt.y">txt.y</code></td>
<td>
<p>y axis alignment of tip labels</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_txt.x">txt.x</code></td>
<td>
<p>x coordinates for plotting tip labels. Can be a single value applicable to all labels, or a vector of the same length as phylo0$tip.label. If NULL (default), the right margin of the plot is used with right-hand alignment for the text.</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_adj.x">adj.x</code></td>
<td>
<p>Numeric value giving alignment on x axis. If NULL (default) this defaults to 0 (left-aligned) but can also any other adjustment value (e.g. 0.5 for centered, 1 for right-aligned).</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_add">add</code></td>
<td>
<p>Logical indicating whether to add to an existing plot, in which case only the spindles are plotted on top of an existing phylogeny, or not, in which case the phylogeny is plotted along with the spindles.</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_tbmar">tbmar</code></td>
<td>
<p>Top and bottom margin around the plot. Numeric of either length 1 or 2</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_smooth">smooth</code></td>
<td>
<p>Smoothing parameter to be passed on to divdistr_()</p>
</td></tr>
<tr><td><code id="phylo.spindles_+3A_italicize">italicize</code></td>
<td>
<p>Character or numeric vector specifying which labels to italicize, if any.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The phylo.spindles() function allows the plotting of a phylogeny with spindle diagrams at each of its terminal branches. Various data can be represented (e.g. disparity, abundance, various diversity measures, such as those output by the divDyn package, etc.) depending on the settings for occ and stat, but the function is optimized to plot the results of divdistr_() and does so by default.
If another function is used as an argument to stat, it has to be able to take the sequence resulting from xlimits and res as its first, and occ as its 'table' argument and return a vector of the same length as range(xlimits)/res to be plotted. If occ is a list() object containing multiple dataframes, occurrence datasets or taxon range tables are automatically converted to work with abdistr_() or divdistr_() respectively (if the plot contains a phylogeny). If occ is a matrix or data.frame, the x values must already be converted (e.g. using tsconv()) to match the phylogeny.
</p>


<h3>Value</h3>

<p>A plotted phylogeny with spindle diagrams plotted at each of its terminal branches.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
data(tree_archosauria)
data(ages_archosauria)
data(diversity_table)
phylo.spindles(tree_archosauria,occ=archosauria,dscale=0.005,ages=ages_archosauria,txt.x=66)
phylo.spindles(tree_archosauria,occ=diversity_table,dscale=0.005,ages=ages_archosauria,txt.x=66)
</code></pre>

<hr>
<h2 id='redraw.phylo'>Redraw the lines of a phylogenetic tree.</h2><span id='topic+redraw.phylo'></span>

<h3>Description</h3>

<p>Redraw the lines of a phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>redraw.phylo(
  saved_plot = NULL,
  col = "black",
  lwd = 1,
  lty = 1,
  lend = 2,
  arrow.l = 0,
  arrow.angle = 45,
  arrow.code = 2,
  indices = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="redraw.phylo_+3A_saved_plot">saved_plot</code></td>
<td>
<p>Optional saved plot (e.g. using get(&quot;last_plot.phylo&quot;, envir = ape::.PlotPhyloEnv)) to be used instead of currently active plot.</p>
</td></tr>
<tr><td><code id="redraw.phylo_+3A_col">col</code></td>
<td>
<p>Color to be used for redrawing tree edges.</p>
</td></tr>
<tr><td><code id="redraw.phylo_+3A_lwd">lwd</code></td>
<td>
<p>Line width to be used for redrawing tree edges.</p>
</td></tr>
<tr><td><code id="redraw.phylo_+3A_lty">lty</code></td>
<td>
<p>Line type to be used for redrawing tree edges.</p>
</td></tr>
<tr><td><code id="redraw.phylo_+3A_lend">lend</code></td>
<td>
<p>Style of line ends to be used for redrawing tree edges.</p>
</td></tr>
<tr><td><code id="redraw.phylo_+3A_arrow.l">arrow.l</code></td>
<td>
<p>Length of arrow ends to be used for plotting. Defaults to 0, i.e. no visible arrow.</p>
</td></tr>
<tr><td><code id="redraw.phylo_+3A_arrow.angle">arrow.angle</code></td>
<td>
<p>Angle of arrow ends to be used for plotting. Defaults to 45 degrees.</p>
</td></tr>
<tr><td><code id="redraw.phylo_+3A_arrow.code">arrow.code</code></td>
<td>
<p>Arrow code to be used for plotting. For details, see ?arrows</p>
</td></tr>
<tr><td><code id="redraw.phylo_+3A_indices">indices</code></td>
<td>
<p>Optional indices which edges to redraw. Can be used to highlight specific edges in different color or style.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing (redraws selected edges of the phylogeny on the active plot device)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_archosauria)
ape::plot.phylo(tree_archosauria)
redraw.phylo(col="darkred",lwd=3,indices=c(19:24))
redraw.phylo(col="red",lwd=3,indices=c(18),arrow.l=0.1)
</code></pre>

<hr>
<h2 id='rmean'>Calculate a rolling mean for a vector x.</h2><span id='topic+rmean'></span>

<h3>Description</h3>

<p>Calculate a rolling mean for a vector x.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmean(x, width = 11)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rmean_+3A_x">x</code></td>
<td>
<p>Numeric vector for which to calculate the rolling mean.</p>
</td></tr>
<tr><td><code id="rmean_+3A_width">width</code></td>
<td>
<p>Width of the interval over which to calculate rolling mean values. Should be an uneven number (even numbers are coerced into the next-higher uneven number)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of the same length as x containing the calculated rolling means, with the first and last few values being NA (depending on the setting for width)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rmean(x=c(1,2,3,4,5,6),width=5)
</code></pre>

<hr>
<h2 id='rmeana'>Calculate a rolling mean based on distance within a second variable.</h2><span id='topic+rmeana'></span>

<h3>Description</h3>

<p>Calculate a rolling mean based on distance within a second variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rmeana(x0, y0, x1 = NULL, plusminus = 5, weighting = FALSE, weightdiff = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rmeana_+3A_x0">x0</code></td>
<td>
<p>Numeric independent variable at which rolling mean is to be calculated.</p>
</td></tr>
<tr><td><code id="rmeana_+3A_y0">y0</code></td>
<td>
<p>Numeric variable of which mean is to be calculated.</p>
</td></tr>
<tr><td><code id="rmeana_+3A_x1">x1</code></td>
<td>
<p>Optional. New x values at which rolling mean of y0 is to be calculated. If x1==NULL, calculation will take place at original (x0) values.</p>
</td></tr>
<tr><td><code id="rmeana_+3A_plusminus">plusminus</code></td>
<td>
<p>Criterium for the width (in x0) of the interval over which rolling mean values are to be calculated. Value represents the margin as calculated from every value of x1 or x0, i.e. for a plusminus==5, the interval over which the means are drawn will range from values with x-x_i=5 to x-x_i=-5.</p>
</td></tr>
<tr><td><code id="rmeana_+3A_weighting">weighting</code></td>
<td>
<p>Whether or not to apply weighting. If weighting==TRUE, then means are calculated as weighted means with weighting decreasing linearly towards the margins of the interval over which the mean is to be drawn.</p>
</td></tr>
<tr><td><code id="rmeana_+3A_weightdiff">weightdiff</code></td>
<td>
<p>Minimum weight to be added to all weights if weighting==TRUE. Defaults to 0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of the same length as either x1 (if not NULL) or x0, containing the calculated rolling means.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rmeana(x0=c(1,2,3,4,5,6), y0=c(2,3,3,4,5,6))
</code></pre>

<hr>
<h2 id='stax.sel'>Extract subsets of an occurrence data.frame.</h2><span id='topic+stax.sel'></span>

<h3>Description</h3>

<p>Extract subsets of an occurrence data.frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stax.sel(taxa, rank = x$class, x = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stax.sel_+3A_taxa">taxa</code></td>
<td>
<p>A vector containing subtaxa (or any other entries matching entries of rank) to be returned</p>
</td></tr>
<tr><td><code id="stax.sel_+3A_rank">rank</code></td>
<td>
<p>Vector or column of x in which to look for entries matching taxa. defaults to x$class, for selecting class-level subtaxa from large datasets (only works if pdb(...,full=TRUE))</p>
</td></tr>
<tr><td><code id="stax.sel_+3A_x">x</code></td>
<td>
<p>Optional occurrence data.frame. If set, a data.frame with the selected entries will be returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If is.null(x) (default), a vector giving the indices of values matching taxa in rank. Otherwise, an occurrence data.frame() containing only the selected taxa or values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
archosauria$Stegosauria-&gt;stegos
stax.sel(c("Stegosaurus"), rank=stegos$genus,x=stegos)-&gt;Stegosaurus
</code></pre>

<hr>
<h2 id='synonymize'>Combine selected entries in a taxon-range table to remove duplicates</h2><span id='topic+synonymize'></span>

<h3>Description</h3>

<p>Combine selected entries in a taxon-range table to remove duplicates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>synonymize(x, table = NULL, ids = table$tna, max = table$max, min = table$min)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="synonymize_+3A_x">x</code></td>
<td>
<p>Indices or values (taxon names) to combine</p>
</td></tr>
<tr><td><code id="synonymize_+3A_table">table</code></td>
<td>
<p>Taxon-range table</p>
</td></tr>
<tr><td><code id="synonymize_+3A_ids">ids</code></td>
<td>
<p>Vector or column of taxon names (used for matching taxon names in x). Defaults to table$tna</p>
</td></tr>
<tr><td><code id="synonymize_+3A_max">max</code></td>
<td>
<p>Vector or column containing maximum ages</p>
</td></tr>
<tr><td><code id="synonymize_+3A_min">min</code></td>
<td>
<p>Vector or column containing minimum ages</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is meant as an aid to manually editing species tables and remove synonyms or incorrect spellings of taxonomic name that result in an inflated number of distinct taxa being represented.
</p>


<h3>Value</h3>

<p>A data.frame containing taxon names, maximum, minimum and mean ages, with ranges for the selected entries merged and superfluous entries removed (note that the first taxon indicated by x is kept as valid).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
sp&lt;-archosauria$sptab_Stegosauria
synonymize(c(32,33),sp)-&gt;sp
synonymize(grep("stenops",sp$tna),sp)-&gt;sp
synonymize(c("Hesperosaurus mjosi","Stegosaurus mjosi"),sp)-&gt;sp
</code></pre>

<hr>
<h2 id='tree_archosauria'>tree_archosauria</h2><span id='topic+tree_archosauria'></span>

<h3>Description</h3>

<p>A time-calibrated phylogenetic tree of Archosauria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree_archosauria
</code></pre>


<h3>Format</h3>

<p>An object of class==phylo with 13 tips and 12 internal nodes.
</p>

<hr>
<h2 id='tree.age.combine'>Combine two calibration matrixes and fill in NA values in one with values from another</h2><span id='topic+tree.age.combine'></span>

<h3>Description</h3>

<p>Combine two calibration matrixes and fill in NA values in one with values from another
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree.age.combine(ages0, ages1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree.age.combine_+3A_ages0">ages0</code></td>
<td>
<p>First matrix, NA values in which to replace with values from second matrix</p>
</td></tr>
<tr><td><code id="tree.age.combine_+3A_ages1">ages1</code></td>
<td>
<p>matrix from which to take replacement values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>tree.age.combine builds the union of two calibration matrices if some of the values in one of them are NAs. If exact matches for some entries cannot be found, a relaxed search matching only the first word (i.e. usually the genus name) in each taxon name is run, in order to fill in as much of the age matrix as possible with non-NA values. It is highly recommended to manually inspect the resulting table for accuracy.
</p>


<h3>Value</h3>

<p>A two-column matrix containing earliest and latest occurrences for each taxon in taxa, with taxon names as row names
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
data(tree_archosauria)
tree.ages.spp(tree_archosauria,data=archosauria$sptab_Ornithopoda)-&gt;ages_A
tree.ages.spp(tree_archosauria,data=archosauria$sptab_Allosauroidea)-&gt;ages_B
tree.age.combine(ages_A,ages_B)-&gt;ages
</code></pre>

<hr>
<h2 id='tree.ages'>Automatically build matrix for time-calibration of phylogenetic trees using occurrence data.</h2><span id='topic+tree.ages'></span>

<h3>Description</h3>

<p>Automatically build matrix for time-calibration of phylogenetic trees using occurrence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree.ages(phylo0 = NULL, data = NULL, taxa = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree.ages_+3A_phylo0">phylo0</code></td>
<td>
<p>Either an object of class phylo, or a character vector containing taxon names for building the matrix</p>
</td></tr>
<tr><td><code id="tree.ages_+3A_data">data</code></td>
<td>
<p>Optional list()-object containing either taxon-range tables or occurrence datasets for all taxa. If NULL, data will be automatically downloaded via the pdb()-function</p>
</td></tr>
<tr><td><code id="tree.ages_+3A_taxa">taxa</code></td>
<td>
<p>Deprecated argument; vector containing taxa to include in calibration matrix (can now be provided directly as phylo0)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>tree.ages works best for getting occurrence dates for higher-level taxa (genus-level and up) that can be used as a base_name in a call to the paleobiology database and will return NAs for species names (or any other taxon that cannot be found in the paleobiology database or the provided list object). For a function optimized to recover taxon ranges for genera and species, see tree.ages.spp().  It is highly recommended to manually inspect the resulting table for accuracy.
</p>


<h3>Value</h3>

<p>A two-column matrix containing earliest and latest occurrences for each taxon in taxa, with taxon names as row names
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
data(tree_archosauria)
tree.ages(tree_archosauria,data=archosauria)-&gt;ages
</code></pre>

<hr>
<h2 id='tree.ages.spp'>Automatically build matrix for time-calibration of phylogenetic trees using occurrence data.</h2><span id='topic+tree.ages.spp'></span>

<h3>Description</h3>

<p>Automatically build matrix for time-calibration of phylogenetic trees using occurrence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tree.ages.spp(phylo0, data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tree.ages.spp_+3A_phylo0">phylo0</code></td>
<td>
<p>Either an object of class phylo, or a character vector containing taxon names for building the matrix</p>
</td></tr>
<tr><td><code id="tree.ages.spp_+3A_data">data</code></td>
<td>
<p>A higher-level taxon name to get data for in the paleobiology database, or a data.frame containing a species table containing entries for the taxa in question.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>tree.ages looks for the taxon names in the tna column of a taxon-range table (as produced by mk.sptab()), so it will only recover ages for taxa that can be found there. For a function optimized for higher-level taxa that might not be represented in such a table, see tree.ages().  It is highly recommended to manually inspect the resulting table for accuracy.
</p>


<h3>Value</h3>

<p>A two-column matrix containing earliest and latest occurrences for each taxon in taxa, with taxon names as row names
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(archosauria)
data(tree_archosauria)
tree.ages.spp(tree_archosauria,data=archosauria$sptab_Ornithopoda)-&gt;ages
</code></pre>

<hr>
<h2 id='ts.periods'>Add a horizontal, period-level phanerozoic timescale to any plot, especially calibrated phylogenies plotted with ape.</h2><span id='topic+ts.periods'></span>

<h3>Description</h3>

<p>Add a horizontal, period-level phanerozoic timescale to any plot, especially calibrated phylogenies plotted with ape.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ts.periods(
  phylo = NULL,
  alpha = 1,
  names = TRUE,
  exclude = c("Quarternary"),
  col.txt = NULL,
  border = NA,
  ylim = NULL,
  adj.txt = c(0.5, 0.5),
  txt.y = mean,
  bw = FALSE,
  update = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ts.periods_+3A_phylo">phylo</code></td>
<td>
<p>Optional (calibrated) phylogeny to which to add timescale. If phylogeny is provided, the $root.time variable is used to convert ages so that the time scale will fit the phylogeny.</p>
</td></tr>
<tr><td><code id="ts.periods_+3A_alpha">alpha</code></td>
<td>
<p>Opacity value to use for the fill of the time scale</p>
</td></tr>
<tr><td><code id="ts.periods_+3A_names">names</code></td>
<td>
<p>Logical indicating whether to plot period names (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="ts.periods_+3A_exclude">exclude</code></td>
<td>
<p>Character vector listing periods for which to not plot the names, if names==TRUE</p>
</td></tr>
<tr><td><code id="ts.periods_+3A_col.txt">col.txt</code></td>
<td>
<p>Color(s) to use for labels.</p>
</td></tr>
<tr><td><code id="ts.periods_+3A_border">border</code></td>
<td>
<p>Color to use for the border of the timescale</p>
</td></tr>
<tr><td><code id="ts.periods_+3A_ylim">ylim</code></td>
<td>
<p>Setting for height of the timescale. Can either be one single value giving the height of the timescale, in which case the function attempts to use the lower limit of the current plot as the lower margin, or a vector of length 2 containing the lower and upper limits of the timescale.</p>
</td></tr>
<tr><td><code id="ts.periods_+3A_adj.txt">adj.txt</code></td>
<td>
<p>Numeric vector of length==2 giving horizontal and vertical label alignment (defaults to centered, i.e. 0.5 for both values)</p>
</td></tr>
<tr><td><code id="ts.periods_+3A_txt.y">txt.y</code></td>
<td>
<p>Function to use to determine the vertical text position (defaults to mean, i.e. centered)</p>
</td></tr>
<tr><td><code id="ts.periods_+3A_bw">bw</code></td>
<td>
<p>Logical whether to plot in black and white (defaults to FALSE). If TRUE, time scale is drawn with a white background</p>
</td></tr>
<tr><td><code id="ts.periods_+3A_update">update</code></td>
<td>
<p>Character string giving the filename of a .csv table for providing an updated timescale. If provided, the values for plotting the time scale are taken from the csv file instead of the internally provided values. Table must have columns named periods, bottom, top and col, giving the period names, start time in ma, end time in ma and a valid color value, respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots a timescale on the currently active plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_archosauria)
ape::plot.phylo(tree_archosauria)
ts.periods(tree_archosauria, alpha=0.5)
</code></pre>

<hr>
<h2 id='ts.stages'>Add a horizontal, stage-level phanerozoic timescale to any plot, especially calibrated phylogenies plotted with ape.</h2><span id='topic+ts.stages'></span>

<h3>Description</h3>

<p>Add a horizontal, stage-level phanerozoic timescale to any plot, especially calibrated phylogenies plotted with ape.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ts.stages(
  phylo = NULL,
  alpha = 1,
  names = FALSE,
  col.txt = NULL,
  border = NA,
  ylim = NULL,
  adj.txt = c(0.5, 0.5),
  txt.y = mean,
  bw = FALSE,
  update = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ts.stages_+3A_phylo">phylo</code></td>
<td>
<p>Optional (calibrated) phylogeny to which to add timescale. If phylogeny is provided, the $root.time variable is used to convert ages so that the time scale will fit the phylogeny.</p>
</td></tr>
<tr><td><code id="ts.stages_+3A_alpha">alpha</code></td>
<td>
<p>Opacity value to use for the fill of the time scale</p>
</td></tr>
<tr><td><code id="ts.stages_+3A_names">names</code></td>
<td>
<p>Logical indicating whether to plot stage names (defaults to FALSE)</p>
</td></tr>
<tr><td><code id="ts.stages_+3A_col.txt">col.txt</code></td>
<td>
<p>Color(s) to use for labels.</p>
</td></tr>
<tr><td><code id="ts.stages_+3A_border">border</code></td>
<td>
<p>Color to use for the border of the timescale</p>
</td></tr>
<tr><td><code id="ts.stages_+3A_ylim">ylim</code></td>
<td>
<p>Setting for height of the timescale. Can either be one single value giving the height of the timescale, in which case the function attempts to use the lower limit of the current plot as the lower margin, or a vector of length 2 containing the lower and upper limits of the timescale.</p>
</td></tr>
<tr><td><code id="ts.stages_+3A_adj.txt">adj.txt</code></td>
<td>
<p>Numeric vector of length==2 giving horizontal and vertical label alignment (defaults to centered, i.e. 0.5 for both values)</p>
</td></tr>
<tr><td><code id="ts.stages_+3A_txt.y">txt.y</code></td>
<td>
<p>Function to use to determine the vertical text position (defaults to mean, i.e. centered)</p>
</td></tr>
<tr><td><code id="ts.stages_+3A_bw">bw</code></td>
<td>
<p>Logical whether to plot in black and white (defaults to FALSE). If TRUE, time scale is drawn with a white background</p>
</td></tr>
<tr><td><code id="ts.stages_+3A_update">update</code></td>
<td>
<p>Character string giving the filename of a .csv table for providing an updated timescale. If provided, the values for plotting the time scale are taken from the csv file instead of the internally provided values. Table must have columns named stage, bottom, top and col, giving the stage names, start time in ma, end time in ma and a valid color value, respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots a timescale on the currently active plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(tree_archosauria)
ape::plot.phylo(tree_archosauria)
ts.stages(tree_archosauria, alpha=0.7)
ts.periods(tree_archosauria, alpha=0)
</code></pre>

<hr>
<h2 id='tsconv'>Convert geological ages for accurate plotting alongside a calibrated phylogeny</h2><span id='topic+tsconv'></span>

<h3>Description</h3>

<p>Convert geological ages for accurate plotting alongside a calibrated phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsconv(x, phylo0 = NULL, root.time = phylo0$root.time)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tsconv_+3A_x">x</code></td>
<td>
<p>A vector of geological ages to be converted.</p>
</td></tr>
<tr><td><code id="tsconv_+3A_phylo0">phylo0</code></td>
<td>
<p>Phylogeny from which to take root.age</p>
</td></tr>
<tr><td><code id="tsconv_+3A_root.time">root.time</code></td>
<td>
<p>Numeric root age, if not taken from a phylogeny</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric() containing the converted geological ages
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tsconv(c(252,201,66), root.time=300)
</code></pre>

<hr>
<h2 id='viol'>Generate a violin plot</h2><span id='topic+viol'></span>

<h3>Description</h3>

<p>Generate a violin plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>viol(
  x,
  pos = 0,
  x2 = NULL,
  stat = density,
  dscale = 1,
  cutoff = range(x),
  horiz = TRUE,
  add = TRUE,
  lim = cutoff,
  xlab = "",
  ylab = "",
  fill = "grey",
  col = "black",
  lwd = 1,
  lty = 1,
  na.rm = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="viol_+3A_x">x</code></td>
<td>
<p>Variable for which to plot violin.</p>
</td></tr>
<tr><td><code id="viol_+3A_pos">pos</code></td>
<td>
<p>Position at which to place violin in the axis perpendicular to x. Defaults to 0</p>
</td></tr>
<tr><td><code id="viol_+3A_x2">x2</code></td>
<td>
<p>Optional variable to override the use of x as input variable for the plotting statistic. If x2 is set, the function (default: density()) used to calculate the plotting statistic is run on x2 instead of x, but the results are plotted at the corresponding values of x.</p>
</td></tr>
<tr><td><code id="viol_+3A_stat">stat</code></td>
<td>
<p>The plotting statistic. Details to the density() function, as in a standard violin plot, but can be overridden with another function that can take x or x2 as its first argument. Stat can also be a numeric vector of the same length as x, in which case the values in this vectors are used instead of the function output and plotted against x as an independent variable.</p>
</td></tr>
<tr><td><code id="viol_+3A_dscale">dscale</code></td>
<td>
<p>The scale to apply to the values for density (or another plotting statistic). Defaults to 1, but adjustment may be needed depending on the scale of the plot the violin is to be added to.</p>
</td></tr>
<tr><td><code id="viol_+3A_cutoff">cutoff</code></td>
<td>
<p>Setting for cropping the violin. Can be either a single value, in which case the input is interpreted as number of standard deviations from the mean, or a numeric vector of length 2, giving the lower and upper cutoff value directly.</p>
</td></tr>
<tr><td><code id="viol_+3A_horiz">horiz</code></td>
<td>
<p>Logical indicating whether to plot horizontally (defaults to TRUE) or vertically</p>
</td></tr>
<tr><td><code id="viol_+3A_add">add</code></td>
<td>
<p>Logical indicating whether to add to an existing plot (defaults to TRUE) or generate a new plot.</p>
</td></tr>
<tr><td><code id="viol_+3A_lim">lim</code></td>
<td>
<p>Limits (in the dimensions of x) used for plotting, if add==FALSE. Defaults to cutoff, but can be manually set as a numeric vector of length 2, giving the lower and upper limits of the plot.</p>
</td></tr>
<tr><td><code id="viol_+3A_xlab">xlab</code></td>
<td>
<p>x axis label</p>
</td></tr>
<tr><td><code id="viol_+3A_ylab">ylab</code></td>
<td>
<p>y axis label</p>
</td></tr>
<tr><td><code id="viol_+3A_fill">fill</code></td>
<td>
<p>Fill color for the plotted violin</p>
</td></tr>
<tr><td><code id="viol_+3A_col">col</code></td>
<td>
<p>Line color for the plotted violin</p>
</td></tr>
<tr><td><code id="viol_+3A_lwd">lwd</code></td>
<td>
<p>Line width for the plotted violin</p>
</td></tr>
<tr><td><code id="viol_+3A_lty">lty</code></td>
<td>
<p>Line width for the plotted violin</p>
</td></tr>
<tr><td><code id="viol_+3A_na.rm">na.rm</code></td>
<td>
<p>logical indicating whether to remove NA values from input data.</p>
</td></tr>
<tr><td><code id="viol_+3A_...">...</code></td>
<td>
<p>Other arguments to be passed on to function in parameter stat</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Viol provides a versatile function for generating violin plots and adding them to r base graphics. The default plotting statistic is density(), resulting in the standard violin plot. However, density can be overridden by entering any function that can take x or x2 as its first argument, or any numeric vector containing the data to be plotted, as long as this vector is the same length as x.
</p>


<h3>Value</h3>

<p>A violin plot and a data.frame containing the original and modified plotting statistic and independent variable against which it is plotted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>viol(x=c(1,2,2,2,3,4,4,3,2,2,3,3,4,5,3,3,2,2,1,6,7,6,9),pos=1, add=FALSE)
viol(c(1:10), width=9, stat=rmean, pos=0, add=FALSE)
viol(c(1:10), stat=c(11:20), pos=0, add=FALSE)
</code></pre>

<hr>
<h2 id='violins'>Wrapper around viol() to conveniently plot multiple violins on a single plot, analogous to the behavior of boxplot()</h2><span id='topic+violins'></span>

<h3>Description</h3>

<p>Wrapper around viol() to conveniently plot multiple violins on a single plot, analogous to the behavior of boxplot()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>violins(
  x,
  data = NULL,
  group = NULL,
  horiz = FALSE,
  order = NULL,
  xlab = "",
  ylab = "",
  col = "black",
  fill = "grey",
  lwd = 1,
  lty = 1,
  dscale = 1,
  xlim = NULL,
  ylim = NULL,
  spaces = "_",
  add = FALSE,
  ax = TRUE,
  srt = 45,
  na.rm = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="violins_+3A_x">x</code></td>
<td>
<p>plotting statistic (numeric vector) or formula object from which a plotting statistic and grouping variable can be extracted (i.e. of form x~group)</p>
</td></tr>
<tr><td><code id="violins_+3A_data">data</code></td>
<td>
<p>data.frame object containing x and y</p>
</td></tr>
<tr><td><code id="violins_+3A_group">group</code></td>
<td>
<p>grouping variable</p>
</td></tr>
<tr><td><code id="violins_+3A_horiz">horiz</code></td>
<td>
<p>logical indicating whether to plot horizontally</p>
</td></tr>
<tr><td><code id="violins_+3A_order">order</code></td>
<td>
<p>order of factor levels of categorical factor</p>
</td></tr>
<tr><td><code id="violins_+3A_xlab">xlab</code></td>
<td>
<p>x axis label</p>
</td></tr>
<tr><td><code id="violins_+3A_ylab">ylab</code></td>
<td>
<p>y axis label</p>
</td></tr>
<tr><td><code id="violins_+3A_col">col</code></td>
<td>
<p>vector of border colors</p>
</td></tr>
<tr><td><code id="violins_+3A_fill">fill</code></td>
<td>
<p>vector of fill colors</p>
</td></tr>
<tr><td><code id="violins_+3A_lwd">lwd</code></td>
<td>
<p>vector of line widths</p>
</td></tr>
<tr><td><code id="violins_+3A_lty">lty</code></td>
<td>
<p>vector of line types</p>
</td></tr>
<tr><td><code id="violins_+3A_dscale">dscale</code></td>
<td>
<p>density scaling factors (numeric) to apply to individual violins</p>
</td></tr>
<tr><td><code id="violins_+3A_xlim">xlim</code></td>
<td>
<p>x limits (data limits used if NULL)</p>
</td></tr>
<tr><td><code id="violins_+3A_ylim">ylim</code></td>
<td>
<p>y limits (data limits used if NULL)</p>
</td></tr>
<tr><td><code id="violins_+3A_spaces">spaces</code></td>
<td>
<p>character string in group to replace with spaces for labels, if not NULL</p>
</td></tr>
<tr><td><code id="violins_+3A_add">add</code></td>
<td>
<p>logical whether to add to existing plot (default: FALSE)</p>
</td></tr>
<tr><td><code id="violins_+3A_ax">ax</code></td>
<td>
<p>whether to plot axes</p>
</td></tr>
<tr><td><code id="violins_+3A_srt">srt</code></td>
<td>
<p>angle for categorical axis text rotation</p>
</td></tr>
<tr><td><code id="violins_+3A_na.rm">na.rm</code></td>
<td>
<p>logical indicating whether to tell viol() to remove NA values (defaults to TRUE)</p>
</td></tr>
<tr><td><code id="violins_+3A_...">...</code></td>
<td>
<p>other arguments to pass on to paleoDiv::viol() and plot()</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data.frame(p=rnorm(50), cat=rep(c("A","B","B","B","B"),10))-&gt;d
violins(p~cat,d)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
