<!DOCTYPE html><html lang="en"><head><title>Help for package PANACEA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PANACEA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PANACEA'><p>PANACEA: Personalized Network-based Anti-Cancer Therapy Evaluation</p></a></li>
<li><a href='#add_drugs_as_nodes'><p>Add Drugs as Nodes</p></a></li>
<li><a href='#adj_list2mat'><p>Turn Adjacency List into Adjacency Matrix</p></a></li>
<li><a href='#convert2alias'><p>Convert Input Gene Symbols to Alias</p></a></li>
<li><a href='#DGIdb_interactions_df'><p>DGIdb Interactions Expert-curated Sources</p></a></li>
<li><a href='#example_driveR_res'><p>Example driveR Result</p></a></li>
<li><a href='#example_scores_dist'><p>Example PANACEA &quot;distance-based&quot; Method Result</p></a></li>
<li><a href='#example_scores_RWR'><p>Example PANACEA &quot;RWR&quot; Method Result</p></a></li>
<li><a href='#Laplacian.norm'><p>Graph Laplacian Normalization</p></a></li>
<li><a href='#network_propagation'><p>Network Propagation (Random-walk with Restart)</p></a></li>
<li><a href='#process_drug_target_interactions'><p>Process Drug-Target Interactions</p></a></li>
<li><a href='#score_drugs'><p>Scoring of Drugs via Network-based Methods</p></a></li>
<li><a href='#score_drugs_distance_based'><p>Distance-based Scoring of Drugs</p></a></li>
<li><a href='#score_drugs_RWR_based'><p>RWR-based Scoring of Drugs</p></a></li>
<li><a href='#STRING_adj_df'><p>Adjacency List for STRING v11.5 - High Confidence Interactions</p></a></li>
<li><a href='#toy_W_mat'><p>Toy Adjacency Matrix (for examples)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Personalized Network-Based Anti-Cancer Therapy Evaluation</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ege Ulgen &lt;egeulgen@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Identification of the most appropriate pharmacotherapy for each 
    patient based on genomic alterations is a major challenge in personalized 
    oncology. 'PANACEA' is a collection of personalized anti-cancer drug 
    prioritization approaches utilizing network methods. The methods utilize 
    personalized "driverness" scores from 'driveR' to rank drugs, mapping these
    onto a protein-protein interaction network. The "distance-based" method 
    scores each drug based on these scores and distances between drugs and genes
    to rank given drugs. The "RWR" method propagates these scores via a 
    random-walk with restart framework to rank the drugs. The methods are 
    described in detail in Ulgen E, Ozisik O, Sezerman OU. 2023. PANACEA: 
    network-based methods for pharmacotherapy prioritization in personalized 
    oncology. Bioinformatics &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtad022">doi:10.1093/bioinformatics/btad022</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/egeulgen/PANACEA">https://github.com/egeulgen/PANACEA</a>,
<a href="https://egeulgen.github.io/PANACEA/">https://egeulgen.github.io/PANACEA/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/egeulgen/PANACEA/issues">https://github.com/egeulgen/PANACEA/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>org.Hs.eg.db, DBI, igraph, reshape2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, testthat (&ge; 3.0.0), covr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-19 12:57:02 UTC; egeulgen</td>
</tr>
<tr>
<td>Author:</td>
<td>Ege Ulgen <a href="https://orcid.org/0000-0003-2090-3621"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-19 13:22:36 UTC</td>
</tr>
</table>
<hr>
<h2 id='PANACEA'>PANACEA: Personalized Network-based Anti-Cancer Therapy Evaluation</h2><span id='topic+PANACEA'></span><span id='topic+PANACEA-package'></span>

<h3>Description</h3>

<p>Identification of the most appropriate pharmacotherapy for each patient based
on genomic alterations is a major challenge in personalized oncology.
<code>PANACEA</code> is a collection of personalized anti-cancer drug prioritization
approaches utilizing network methods. The methods utilize personalized
&quot;driverness&quot; scores from 'driveR' to rank drugs, mapping these onto a
protein-protein interaction network (PIN). The &quot;distance-based&quot; method scores
each drug based on these scores and distances between drugs and genes to rank
given drugs. The &quot;RWR&quot; method propagates these scores via a random-walk with
restart framework to rank the drugs.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Ege Ulgen <a href="mailto:egeulgen@gmail.com">egeulgen@gmail.com</a> (<a href="https://orcid.org/0000-0003-2090-3621">ORCID</a>) [copyright holder]
</p>


<h3>See Also</h3>

<p><code><a href="#topic+score_drugs">score_drugs</a></code> for the wrapper function for scoring of
drugs via network-based methods
</p>

<hr>
<h2 id='add_drugs_as_nodes'>Add Drugs as Nodes</h2><span id='topic+add_drugs_as_nodes'></span>

<h3>Description</h3>

<p>Add Drugs as Nodes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_drugs_as_nodes(W_mat, drug_target_interactions, edge_weight = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_drugs_as_nodes_+3A_w_mat">W_mat</code></td>
<td>
<p>adjacency matrix for the chosen PIN</p>
</td></tr>
<tr><td><code id="add_drugs_as_nodes_+3A_drug_target_interactions">drug_target_interactions</code></td>
<td>
<p>data frame containing (processed) drugs and target genes</p>
</td></tr>
<tr><td><code id="add_drugs_as_nodes_+3A_edge_weight">edge_weight</code></td>
<td>
<p>edge weight for drug-target gene interaction (default = 1000)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>adjacency matrix with the drugs added as nodes
</p>

<hr>
<h2 id='adj_list2mat'>Turn Adjacency List into Adjacency Matrix</h2><span id='topic+adj_list2mat'></span>

<h3>Description</h3>

<p>Turn Adjacency List into Adjacency Matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adj_list2mat(adj_list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adj_list2mat_+3A_adj_list">adj_list</code></td>
<td>
<p>Adjacency list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Adjacency matrix
</p>

<hr>
<h2 id='convert2alias'>Convert Input Gene Symbols to Alias</h2><span id='topic+convert2alias'></span>

<h3>Description</h3>

<p>Convert Input Gene Symbols to Alias
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert2alias(input_genes, target_genes)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert2alias_+3A_input_genes">input_genes</code></td>
<td>
<p>vector of input genes</p>
</td></tr>
<tr><td><code id="convert2alias_+3A_target_genes">target_genes</code></td>
<td>
<p>vector of target genes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of converted gene symbols (if any alias in target genes)
</p>

<hr>
<h2 id='DGIdb_interactions_df'>DGIdb Interactions Expert-curated Sources</h2><span id='topic+DGIdb_interactions_df'></span>

<h3>Description</h3>

<p>Data frame containing drug-gene interactions from expert-curated sources
(CancerCommons, CGI, ChemblInteractions, CIViC, ClearityFoundationBiomarkers,
ClearityFoundationClinicalTrial, COSMIC, DoCM, MyCancerGenome,
MyCancerGenomeClinicalTrial, TALC, TdgClinicalTrial, TEND)
from DGIdb.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DGIdb_interactions_df
</code></pre>


<h3>Format</h3>

<p>a data frame containing 11323 rows and 2 variables:
</p>

<dl>
<dt>drug_name</dt><dd><p>Drug name</p>
</dd>
<dt>gene_name</dt><dd><p>HGNC gene symbol for the interacting gene</p>
</dd>
</dl>


<hr>
<h2 id='example_driveR_res'>Example driveR Result</h2><span id='topic+example_driveR_res'></span>

<h3>Description</h3>

<p>Data frame containing 'driveR' results for a lung adenocarcinoma case.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_driveR_res
</code></pre>


<h3>Format</h3>

<p>a data frame containing 106 rows and 3 variables:
</p>

<dl>
<dt>gene_symbol</dt><dd><p>HGNC gene symbol</p>
</dd>
<dt>driverness_prob</dt><dd><p>'driverness' probability</p>
</dd>
<dt>prediction</dt><dd><p>driveR's prediction for whether the gene is a 'driver' or 'non-driver'</p>
</dd>
</dl>


<hr>
<h2 id='example_scores_dist'>Example PANACEA &quot;distance-based&quot; Method Result</h2><span id='topic+example_scores_dist'></span>

<h3>Description</h3>

<p>Vector containing 'PANACEA' &quot;distance-based&quot; results for a lung adenocarcinoma case.
Names are drug names, values are scores
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_scores_dist
</code></pre>


<h3>Format</h3>

<p>named vector of 1423 values
</p>

<hr>
<h2 id='example_scores_RWR'>Example PANACEA &quot;RWR&quot; Method Result</h2><span id='topic+example_scores_RWR'></span>

<h3>Description</h3>

<p>Vector containing 'PANACEA' &quot;RWR&quot; results for a lung adenocarcinoma case.
Names are drug names, values are scores
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_scores_RWR
</code></pre>


<h3>Format</h3>

<p>named vector of 1423 values
</p>

<hr>
<h2 id='Laplacian.norm'>Graph Laplacian Normalization</h2><span id='topic+Laplacian.norm'></span>

<h3>Description</h3>

<p>Graph Laplacian Normalization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Laplacian.norm(W)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Laplacian.norm_+3A_w">W</code></td>
<td>
<p>square symmetric adjacency matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>normalized adjacency matrix
</p>

<hr>
<h2 id='network_propagation'>Network Propagation (Random-walk with Restart)</h2><span id='topic+network_propagation'></span>

<h3>Description</h3>

<p>Network Propagation (Random-walk with Restart)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_propagation(prior_vec, W_prime, alpha, max.iter = 1000, eps = 1e-04)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="network_propagation_+3A_prior_vec">prior_vec</code></td>
<td>
<p>vector of prior knowledge on selected genes (names are gene symbols)</p>
</td></tr>
<tr><td><code id="network_propagation_+3A_w_prime">W_prime</code></td>
<td>
<p>(Laplacian-normalized, symmetric) adjacency matrix</p>
</td></tr>
<tr><td><code id="network_propagation_+3A_alpha">alpha</code></td>
<td>
<p>restart parameter, controlling trade-off between prior information and network smoothing</p>
</td></tr>
<tr><td><code id="network_propagation_+3A_max.iter">max.iter</code></td>
<td>
<p>maximum allowed number of iterations (default = 1000)</p>
</td></tr>
<tr><td><code id="network_propagation_+3A_eps">eps</code></td>
<td>
<p>epsilon value to assess the L2 norm of the difference between iterations (default = 1e-4)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Implementing RWR following the following publications:
Cowen L, Ideker T, Raphael BJ, Sharan R. Network propagation: a universal amplifier of genetic associations. Nat Rev Genet. 2017 Sep;18(9):551–62.
Shnaps O, Perry E, Silverbush D, Sharan R. Inference of personalized drug targets via network propagation. Pac Symp Biocomput. 2016;21:156–67.
</p>


<h3>Value</h3>

<p>vector of propagation values
</p>

<hr>
<h2 id='process_drug_target_interactions'>Process Drug-Target Interactions</h2><span id='topic+process_drug_target_interactions'></span>

<h3>Description</h3>

<p>Process Drug-Target Interactions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_drug_target_interactions(
  drug_target_interactions,
  PIN_genes,
  drug_name_col = "drug_name",
  target_col = "gene_name"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="process_drug_target_interactions_+3A_drug_target_interactions">drug_target_interactions</code></td>
<td>
<p>data frame containing drugs and target genes</p>
</td></tr>
<tr><td><code id="process_drug_target_interactions_+3A_pin_genes">PIN_genes</code></td>
<td>
<p>gene symbols for the chosen PIN</p>
</td></tr>
<tr><td><code id="process_drug_target_interactions_+3A_drug_name_col">drug_name_col</code></td>
<td>
<p>name of the column containing drug names (default = &quot;drug_name&quot;)</p>
</td></tr>
<tr><td><code id="process_drug_target_interactions_+3A_target_col">target_col</code></td>
<td>
<p>name of the column containing drug targets (default = &quot;converted_target_gene&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>processed drug-target interactions. Processing involves converting
symbols missing in the PIN, merging drugs that have the same target gene(s)
</p>

<hr>
<h2 id='score_drugs'>Scoring of Drugs via Network-based Methods</h2><span id='topic+score_drugs'></span>

<h3>Description</h3>

<p>Scoring of Drugs via Network-based Methods
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score_drugs(driveR_res, drug_interactions_df, W_mat, method, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="score_drugs_+3A_driver_res">driveR_res</code></td>
<td>
<p>data frame of driveR results</p>
</td></tr>
<tr><td><code id="score_drugs_+3A_drug_interactions_df">drug_interactions_df</code></td>
<td>
<p>data frame of drug-gene interactions</p>
</td></tr>
<tr><td><code id="score_drugs_+3A_w_mat">W_mat</code></td>
<td>
<p>adjacency matrix for the PIN</p>
</td></tr>
<tr><td><code id="score_drugs_+3A_method">method</code></td>
<td>
<p>scoring method (one of 'distance-based' or 'RWR')</p>
</td></tr>
<tr><td><code id="score_drugs_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="#topic+score_drugs_distance_based">score_drugs_distance_based</a></code> or
<code><a href="#topic+score_drugs_RWR_based">score_drugs_RWR_based</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is the wrapper function for the two proposed methods for
personalized scoring of drugs for individual cancer samples via network-based
methods. The available methods are 'distance-based' and 'RWR'. For the
'distance-based' method, the score between a gene (g) and drug (d) is formulated
as:
</p>
<p style="text-align: center;"><code class="reqn">score(g, d) = driver(g) / (d(g, d) + 1)^2</code>
</p>

<p>where driver(g) is the driverness probability of gene g, as predicted by
'driveR' and d(g, d) is the distance withing the PIN between gene g and drug d.
The final score of the drug d is then the average of the scores between each
altered gene and d:
</p>
<p style="text-align: center;"><code class="reqn">score(d) = \Sigma{score(g, d)} / |genes|</code>
</p>

<p>For the 'RWR' method, a random-walk with restart framework is used to propagate
the driverness probabilities.
</p>
<p>By default <code><a href="#topic+DGIdb_interactions_df">DGIdb_interactions_df</a></code> is used as the
<code>drug_interactions_df</code>.
</p>
<p>If the <code>W_mat</code> argument is not supplied, the built-in STRNG data
<code><a href="#topic+STRING_adj_df">STRING_adj_df</a></code> is used to generate <code>W_mat</code>.
</p>


<h3>Value</h3>

<p>vector of scores per drug.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>toy_data &lt;- data.frame(
  gene_symbol = c("TP53", "EGFR", "KDR", "ATM"),
  driverness_prob = c(0.94, 0.92, 0.84, 0.72)
)
toy_interactions &lt;- DGIdb_interactions_df[1:25, ]
res &lt;- score_drugs(
  driveR_res = toy_data,
  drug_interactions_df = toy_interactions, # leave blank for default
  W_mat = toy_W_mat, # leave blank for default
  method = "distance-based",
  verbose = FALSE
)
</code></pre>

<hr>
<h2 id='score_drugs_distance_based'>Distance-based Scoring of Drugs</h2><span id='topic+score_drugs_distance_based'></span>

<h3>Description</h3>

<p>Distance-based Scoring of Drugs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score_drugs_distance_based(
  driveR_res,
  drug_interactions_df,
  W_mat,
  driver_prob_cutoff = 0.05,
  drug_name_col = "drug_name",
  target_col = "gene_name",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="score_drugs_distance_based_+3A_driver_res">driveR_res</code></td>
<td>
<p>data frame of driveR results</p>
</td></tr>
<tr><td><code id="score_drugs_distance_based_+3A_drug_interactions_df">drug_interactions_df</code></td>
<td>
<p>data frame of drug-gene interactions</p>
</td></tr>
<tr><td><code id="score_drugs_distance_based_+3A_w_mat">W_mat</code></td>
<td>
<p>adjacency matrix for the PIN</p>
</td></tr>
<tr><td><code id="score_drugs_distance_based_+3A_driver_prob_cutoff">driver_prob_cutoff</code></td>
<td>
<p>cut-off value for 'driverness_prob' (default = 0.05)</p>
</td></tr>
<tr><td><code id="score_drugs_distance_based_+3A_drug_name_col">drug_name_col</code></td>
<td>
<p>for 'drug_interactions_df', the column name containing drug names/identifiers</p>
</td></tr>
<tr><td><code id="score_drugs_distance_based_+3A_target_col">target_col</code></td>
<td>
<p>for 'drug_interactions_df', the column name containing target gene symbols</p>
</td></tr>
<tr><td><code id="score_drugs_distance_based_+3A_verbose">verbose</code></td>
<td>
<p>boolean to control verbosity (default = <code>TRUE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of scores per drug. Drugs with the same target gene(s) are merged
(via <code><a href="#topic+process_drug_target_interactions">process_drug_target_interactions</a></code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>toy_data &lt;- data.frame(
  gene_symbol = c("TP53", "EGFR", "KDR", "ATM"),
  driverness_prob = c(0.94, 0.92, 0.84, 0.72)
)
toy_interactions &lt;- DGIdb_interactions_df[1:100, ]
res &lt;- score_drugs_distance_based(
  driveR_res = toy_data,
  drug_interactions_df = toy_interactions,
  W_mat = toy_W_mat, verbose = FALSE
)
</code></pre>

<hr>
<h2 id='score_drugs_RWR_based'>RWR-based Scoring of Drugs</h2><span id='topic+score_drugs_RWR_based'></span>

<h3>Description</h3>

<p>RWR-based Scoring of Drugs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score_drugs_RWR_based(
  driveR_res,
  drug_interactions_df,
  W_mat,
  alpha = 0.05,
  max.iter = 1000,
  eps = 1e-04,
  drug_name_col = "drug_name",
  target_col = "gene_name",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="score_drugs_RWR_based_+3A_driver_res">driveR_res</code></td>
<td>
<p>data frame of driveR results</p>
</td></tr>
<tr><td><code id="score_drugs_RWR_based_+3A_drug_interactions_df">drug_interactions_df</code></td>
<td>
<p>data frame of drug-gene interactions</p>
</td></tr>
<tr><td><code id="score_drugs_RWR_based_+3A_w_mat">W_mat</code></td>
<td>
<p>adjacency matrix for the PIN</p>
</td></tr>
<tr><td><code id="score_drugs_RWR_based_+3A_alpha">alpha</code></td>
<td>
<p>restart parameter, controlling trade-off between prior information and network smoothing</p>
</td></tr>
<tr><td><code id="score_drugs_RWR_based_+3A_max.iter">max.iter</code></td>
<td>
<p>maximum allowed number of iterations (default = 1000)</p>
</td></tr>
<tr><td><code id="score_drugs_RWR_based_+3A_eps">eps</code></td>
<td>
<p>epsilon value to assess the L2 norm of the difference between iterations (default = 1e-4)</p>
</td></tr>
<tr><td><code id="score_drugs_RWR_based_+3A_drug_name_col">drug_name_col</code></td>
<td>
<p>for 'drug_interactions_df', the column name containing drug names/identifiers</p>
</td></tr>
<tr><td><code id="score_drugs_RWR_based_+3A_target_col">target_col</code></td>
<td>
<p>for 'drug_interactions_df', the column name containing target gene symbols</p>
</td></tr>
<tr><td><code id="score_drugs_RWR_based_+3A_verbose">verbose</code></td>
<td>
<p>boolean to control verbosity (default = <code>TRUE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of scores per drug. Drugs with the same target gene(s) are merged
(via <code><a href="#topic+process_drug_target_interactions">process_drug_target_interactions</a></code>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>toy_data &lt;- data.frame(
  gene_symbol = c("TP53", "EGFR", "KDR", "ATM"),
  driverness_prob = c(0.94, 0.92, 0.84, 0.72)
)
toy_interactions &lt;- DGIdb_interactions_df[1:100, ]
res &lt;- score_drugs_RWR_based(
  driveR_res = toy_data,
  drug_interactions_df = toy_interactions,
  W_mat = toy_W_mat, verbose = FALSE
)
</code></pre>

<hr>
<h2 id='STRING_adj_df'>Adjacency List for STRING v11.5 - High Confidence Interactions</h2><span id='topic+STRING_adj_df'></span>

<h3>Description</h3>

<p>Data frame of adjacency list for STRING v11.5 interactions with combined
score &gt; 700 (high confidence)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>STRING_adj_df
</code></pre>


<h3>Format</h3>

<p>a data frame with 887797 rows and 3 variables:
</p>

<dl>
<dt>protein1</dt><dd><p>Interactor 1</p>
</dd>
<dt>protein2</dt><dd><p>Interactor 2</p>
</dd>
<dt>value</dt><dd><p>edge weight(combined score)</p>
</dd>
</dl>


<hr>
<h2 id='toy_W_mat'>Toy Adjacency Matrix (for examples)</h2><span id='topic+toy_W_mat'></span>

<h3>Description</h3>

<p>Symmetric matrix containing example adjacency data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>toy_W_mat
</code></pre>


<h3>Format</h3>

<p>matrix of 84 rows and 84 columns
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
