<!DOCTYPE html><html><head><title>Help for package paramlink2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {paramlink2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#diseaseModel'><p>Disease models for linkage analysis</p></a></li>
<li><a href='#dominant1'><p>Dominant linkage analysis example</p></a></li>
<li><a href='#linkres'><p>S3 methods for class 'linkres'.</p></a></li>
<li><a href='#lod'><p>Singlepoint LOD score</p></a></li>
<li><a href='#lodPeaks'><p>LOD score peaks</p></a></li>
<li><a href='#merlinLod'><p>Multipoint LOD scores by MERLIN</p></a></li>
<li><a href='#paramlink2'><p>paramlink2: Parametric linkage analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Parametric Linkage Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Parametric linkage analysis of monogenic traits in medical
    pedigrees. Features include singlepoint analysis, multipoint analysis
    via 'MERLIN' (Abecasis et al. (2002) &lt;<a href="https://doi.org/10.1038%2Fng786">doi:10.1038/ng786</a>&gt;),
    visualisation of log of the odds (LOD) scores and summaries of linkage
    peaks. Disease models may be specified to accommodate phenocopies,
    reduced penetrance and liability classes. 'paramlink2' is part of the
    'ped suite' package ecosystem, presented in 'Pedigree Analysis in R'
    (Vigeland, 2021, ISBN:9780128244302).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/magnusdv/paramlink2">https://github.com/magnusdv/paramlink2</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/magnusdv/paramlink2/issues">https://github.com/magnusdv/paramlink2/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1), pedtools</td>
</tr>
<tr>
<td>Imports:</td>
<td>pedprobr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>spelling, testthat</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>MERLIN (https://csg.sph.umich.edu/abecasis/merlin/)
for multipoint linkage analysis.</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-11 09:46:07 UTC; magnusdv</td>
</tr>
<tr>
<td>Author:</td>
<td>Magnus Dehli Vigeland
    <a href="https://orcid.org/0000-0002-9134-4962"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Magnus Dehli Vigeland &lt;m.d.vigeland@medisin.uio.no&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-11 10:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='diseaseModel'>Disease models for linkage analysis</h2><span id='topic+diseaseModel'></span>

<h3>Description</h3>

<p>Create a disease model in form of a <code>disModel</code> object, for use in e.g.
<code><a href="#topic+lod">lod()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diseaseModel(model = NULL, chrom = NULL, penetrances = NULL, dfreq = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diseaseModel_+3A_model">model</code></td>
<td>
<p>An existing <code>disModel</code> object (to be modified by other
arguments), or one of the following two-letter keywords:
</p>

<ul>
<li> <p><code>AD</code> = autosomal dominant
</p>
</li>
<li> <p><code>AR</code> = autosomal recessive
</p>
</li>
<li> <p><code>XD</code> = X-linked dominant
</p>
</li>
<li> <p><code>XR</code> = X-linked recessive
</p>
</li></ul>

<p>In all of the above, the disease is assumed to be fully penetrant, and with
a disease allele frequency of 1e-5.</p>
</td></tr>
<tr><td><code id="diseaseModel_+3A_chrom">chrom</code></td>
<td>
<p>Either &quot;AUTOSOMAL&quot; or &quot;X&quot;. Lower case versions are allowed and
will be converted automatically.</p>
</td></tr>
<tr><td><code id="diseaseModel_+3A_penetrances">penetrances</code></td>
<td>
<p>For autosomal models, a numeric of length 3 corresponding
to <code style="white-space: pre;">&#8288;(f0, f1, f2)&#8288;</code>, where <code>fi</code> is the probability of being affected given
<code>i</code> disease alleles. It can also be a matrix with 3 columns, in which case
each row represents a liability class.
</p>
<p>For X-linked models, a list of two vectors named <code>male</code> and <code>female</code>, of
lengths 2 and 3 respectively: <code style="white-space: pre;">&#8288;(f0, f1)&#8288;</code> for males and <code style="white-space: pre;">&#8288;(f0, f1, f2)&#8288;</code> for
females. Alternatively, each list entry may be a matrix or data frame (with
the same number of columns) where each row represents a liability class.</p>
</td></tr>
<tr><td><code id="diseaseModel_+3A_dfreq">dfreq</code></td>
<td>
<p>A number in <code style="white-space: pre;">&#8288;[0,1]&#8288;</code>: The population frequency of the disease
allele.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>disModel</code>, which is a list with entries <code>chrom</code>,
<code>penetrances</code> and <code>dfreq</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lod">lod()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Fully penetrant AD model:
m1 = diseaseModel(model = "AD")

# The above is equivalent to
m2 = diseaseModel(chrom = "Aut", penetrances = c(0,1,1), dfreq = 1e-5)
stopifnot(identical(m1, m2))

# X-linked recessive model:
m3 = diseaseModel(model = "XR", dfreq = 0.01)

# Long version of the above:
m4 = diseaseModel(chrom = "X", penetrances = list(male = c(0,1), female = c(0,0,1)),
              dfreq = 0.01)
stopifnot(identical(m3, m4))

</code></pre>

<hr>
<h2 id='dominant1'>Dominant linkage analysis example</h2><span id='topic+dominant1'></span>

<h3>Description</h3>

<p>A dataset with SNP genotypes for 14 members of a pedigree affected with a dominant disorder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dominant1
</code></pre>


<h3>Format</h3>

<p>A list with 3 elements:
</p>

<ul>
<li> <p><code>ped</code>: A <a href="pedtools.html#topic+ped">pedtools::ped</a> object describing a pedigree with 19 individuals, including genotypes for 14 members at 248 markers on Chromosome 1.
</p>
</li>
<li> <p><code>aff</code>: A vector indicating the affected pedigree members.
</p>
</li>
<li> <p><code>map</code>: A data frame with 3 columns (<code>chrom</code>, <code>marker</code>, <code>cm</code>) describing the centiMorgan positions of the markers.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
data(dominant1)
ped = dominant1$ped
aff = dominant1$aff
model = diseaseModel("AD")

# Compute singlepoint LODs
lods = lod(ped, aff = aff, model = model)

# LOD score graph
plot(lods)


</code></pre>

<hr>
<h2 id='linkres'>S3 methods for class 'linkres'.</h2><span id='topic+linkres'></span><span id='topic+print.linkres'></span><span id='topic+summary.linkres'></span><span id='topic+plot.linkres'></span><span id='topic+points.linkres'></span>

<h3>Description</h3>

<p>Functions for printing, summarizing and plotting the results of a linkage
analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'linkres'
print(x, ...)

## S3 method for class 'linkres'
summary(object, ...)

## S3 method for class 'linkres'
plot(
  x,
  chrom = NULL,
  type = "l",
  lwd = NA,
  ylim = NULL,
  xlab = NULL,
  ylab = NULL,
  ...
)

## S3 method for class 'linkres'
points(x, chrom = NULL, type = "l", lwd = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linkres_+3A_x">x</code>, <code id="linkres_+3A_object">object</code></td>
<td>
<p>A <code>linkres</code> object (normally produced by <code><a href="#topic+lod">lod()</a></code> or
<code><a href="#topic+merlinLod">merlinLod()</a></code>).</p>
</td></tr>
<tr><td><code id="linkres_+3A_...">...</code></td>
<td>
<p>further arguments.</p>
</td></tr>
<tr><td><code id="linkres_+3A_chrom">chrom</code></td>
<td>
<p>(Optional) A numeric indicating which chromosomes to be included
in the plot.</p>
</td></tr>
<tr><td><code id="linkres_+3A_lwd">lwd</code>, <code id="linkres_+3A_type">type</code>, <code id="linkres_+3A_ylim">ylim</code></td>
<td>
<p>Graphical parameters passed on to <code>plot()</code>.</p>
</td></tr>
<tr><td><code id="linkres_+3A_xlab">xlab</code>, <code id="linkres_+3A_ylab">ylab</code></td>
<td>
<p>Axis labels.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>These functions are called for their side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lod">lod()</a></code>, <code><a href="#topic+merlinLod">merlinLod()</a></code>, <code><a href="#topic+lodPeaks">lodPeaks()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Pedigree with 5 simulated SNP markers
x = nuclearPed(3)

x = setMarkers(x, alleleMatrix = cbind(
  m1 = c("1/1", "1/2", "1/2", "1/2", "1/2"),
  m2 = c("1/2", "1/2", "1/2", "1/2", "1/2"),
  m3 = c("1/1", "1/2", "1/2", "1/2", "1/1")), sep="/")

# Mother and all children affected
aff = c(1, 2, 2, 2, 2)

# LOD scores under autosomal dominant model
lods = lod(x, aff, model = diseaseModel(model = "AD"))

summary(lods)
as.data.frame(lods)
plot(lods)

</code></pre>

<hr>
<h2 id='lod'>Singlepoint LOD score</h2><span id='topic+lod'></span>

<h3>Description</h3>

<p>Calculates the singlepoint log of the odds (LOD) scores of a pedigree for the
specified markers, assuming a fixed recombination rate between the disease
and each marker locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lod(
  x,
  aff,
  model,
  rho = 0,
  liability = NULL,
  markers = NULL,
  maxOnly = NA,
  loopBreakers = NULL,
  peelOrder = NULL,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lod_+3A_x">x</code></td>
<td>
<p>A <code>ped</code> object.</p>
</td></tr>
<tr><td><code id="lod_+3A_aff">aff</code></td>
<td>
<p>A vector naming the affected pedigree members, or a numeric vector
of length <code>pedsize(x)</code> with affection statuses for all pedigree members (2
= affected; 1 = unaffected; 0 = unknown). Alternatively, <code>aff</code> can be a
list of vectors, named <code>affected</code>, <code>unaffected</code> and <code>unknown</code>. It suffices
to give (any) two of these vectors.</p>
</td></tr>
<tr><td><code id="lod_+3A_model">model</code></td>
<td>
<p>A <code>disModel</code> object, typically created with <code><a href="#topic+diseaseModel">diseaseModel()</a></code>.</p>
</td></tr>
<tr><td><code id="lod_+3A_rho">rho</code></td>
<td>
<p>A number between 0 and 0.5 (inclusive); the hypothesised
recombination ratio between the marker and the disease locus.</p>
</td></tr>
<tr><td><code id="lod_+3A_liability">liability</code></td>
<td>
<p>NULL (default) or a vector of length <code>pedsize(x)</code> indicating
the liability class (a row number of <code>model$penetrances</code>) of each
individual.</p>
</td></tr>
<tr><td><code id="lod_+3A_markers">markers</code></td>
<td>
<p>A vector of marker names or indices referring to markers
attached to <code>x</code>. By default all markers are included.</p>
</td></tr>
<tr><td><code id="lod_+3A_maxonly">maxOnly</code></td>
<td>
<p>a logical indicating whether only the maximum LOD score should
be returned. By default this is always done if the number of markers is 1.</p>
</td></tr>
<tr><td><code id="lod_+3A_loopbreakers">loopBreakers</code></td>
<td>
<p>A vector of ID labels indicating loop breakers. (Only
relevant for inbred pedigrees.)</p>
</td></tr>
<tr><td><code id="lod_+3A_peelorder">peelOrder</code></td>
<td>
<p>For internal use.</p>
</td></tr>
<tr><td><code id="lod_+3A_verbose">verbose</code></td>
<td>
<p>a logical: verbose output or not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The LOD score of a marker is defined as </p>
<p style="text-align: center;"><code class="reqn">LOD(\rho) = \log
\frac{L(\rho)}{L(0.5)}</code>
</p>
<p> where the
logarithms are base 10, and <code class="reqn">L(\rho)</code> denotes the likelihood of the
observed marker genotypes given a recombination ratio <code class="reqn">\rho</code> between the
marker and the disease locus.
</p>
<p>The likelihoods are computed with the <strong>pedprobr</strong> package.
</p>


<h3>Value</h3>

<p>If the number of markers is 1, or if <code>maxOnly = TRUE</code>, a single
number is returned.
</p>
<p>Otherwise a <code>linkres</code> object, which is basically a data frame with columns
<code>CHROM</code>, <code>MARKER</code>, <code>MB</code> and <code>LOD</code>.
</p>


<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>See Also</h3>

<p><a href="#topic+linkres">linkres</a>, <code><a href="#topic+merlinLod">merlinLod()</a></code>, <code><a href="#topic+diseaseModel">diseaseModel()</a></code>, <code><a href="#topic+lodPeaks">lodPeaks()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
x = nuclearPed(2) |&gt;
  addMarker(geno = c("1/2", "1/1", "1/2", "1/2"))

aff = c(2,1,2,2)
model = diseaseModel(model = "AD")

lod(x, aff, model)

</code></pre>

<hr>
<h2 id='lodPeaks'>LOD score peaks</h2><span id='topic+lodPeaks'></span><span id='topic+peakSummary'></span>

<h3>Description</h3>

<p>Identify and summarise LOD score peaks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lodPeaks(x, threshold, width = 1)

peakSummary(x, threshold, width = 1, physmap = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lodPeaks_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+linkres">linkres</a> object, or data frame with columns <code>CHROM</code>, <code>MB</code>, <code>LOD</code>.</p>
</td></tr>
<tr><td><code id="lodPeaks_+3A_threshold">threshold</code></td>
<td>
<p>A single number</p>
</td></tr>
<tr><td><code id="lodPeaks_+3A_width">width</code></td>
<td>
<p>A positive integer</p>
</td></tr>
<tr><td><code id="lodPeaks_+3A_physmap">physmap</code></td>
<td>
<p>A matrix or data frame with three columns: Marker
name, chromosome and physical position.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A peak is defined as a run of at least <code>width</code> consecutive markers with LOD
score above or equal to <code>threshold</code>. If possible, one flanking marker is
included on each side of the peak.
</p>


<h3>Value</h3>

<p>A list of data frames.
</p>


<h3>See Also</h3>

<p><a href="#topic+linkres">linkres</a>, <code><a href="#topic+lod">lod()</a></code>, <code><a href="#topic+merlinLod">merlinLod()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Use built-in dataset `dominant1`
lods = lod(x = dominant1$ped,
           aff = dominant1$aff,
           model = diseaseModel("AD"))

# All peaks above LOD = 1.5
lodPeaks(lods, threshold = 1.5)
peakSummary(lods, threshold = 1.5)

</code></pre>

<hr>
<h2 id='merlinLod'>Multipoint LOD scores by MERLIN</h2><span id='topic+merlinLod'></span>

<h3>Description</h3>

<p>This function is a wrapper for the parametric linkage functionality of the
MERLIN software. For this to work, MERLIN must be installed and correctly
pointed to in the PATH environment variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merlinLod(
  x,
  aff,
  model,
  map = NULL,
  markers = NULL,
  rho = 0,
  liability = NULL,
  maxOnly = NA,
  options = "",
  dir = tempdir(),
  cleanup = TRUE,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merlinLod_+3A_x">x</code></td>
<td>
<p>A <code>ped</code> object.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_aff">aff</code></td>
<td>
<p>A vector naming the affected pedigree members, or a numeric vector
of length <code>pedsize(x)</code> with affection statuses for all pedigree members (2
= affected; 1 = unaffected; 0 = unknown). Alternatively, <code>aff</code> can be a
list of vectors, named <code>affected</code>, <code>unaffected</code> and <code>unknown</code>. It suffices
to give (any) two of these vectors.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_model">model</code></td>
<td>
<p>A <code>disModel</code> object, typically created with <code><a href="#topic+diseaseModel">diseaseModel()</a></code>.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_map">map</code></td>
<td>
<p>A data frame with columns according to MERLIN format for map
files.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_markers">markers</code></td>
<td>
<p>A vector of marker names or indices referring to markers
attached to <code>x</code>. By default all markers are included.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_rho">rho</code></td>
<td>
<p>A numeric with values between 0 and 0.5: The recombination
value(s) for which the LOD score is computed. The value of <code>rho</code> is
converted to centiMorgans using Haldane's map function and included in the
MERLIN command using the <code>--position</code> parameter.
</p>
<p>Note: This parameter only works when <code>markers</code> has length 1.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_liability">liability</code></td>
<td>
<p>NULL (default) or a vector of length <code>pedsize(x)</code> indicating
the liability class (a row number of <code>model$penetrances</code>) of each
individual.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_maxonly">maxOnly</code></td>
<td>
<p>a logical indicating whether only the maximum LOD score should
be returned. By default this is always done if the number of markers is 1.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_options">options</code></td>
<td>
<p>A character with additional options to the MERLIN command.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_dir">dir</code></td>
<td>
<p>Path to a directory where the MERLIN input files should be
created.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_cleanup">cleanup</code></td>
<td>
<p>A logical indicating if MERLIN files should be removed after
use. Default: TRUE.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_verbose">verbose</code></td>
<td>
<p>a logical: verbose output or not.</p>
</td></tr>
<tr><td><code id="merlinLod_+3A_...">...</code></td>
<td>
<p>Further arguments passed on to <code><a href="pedprobr.html#topic+merlin">pedprobr::merlin()</a></code>. In
particular, <code>merlinpath</code> should be supplied if the merlin executables are
not on the system's search path.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default the following MERLIN command is run (via a call to <code><a href="base.html#topic+system">system()</a></code>)
after creating appropriate files in a temporary (or user specified) directory:
</p>
<pre>
_merlin.freq --model _merlin.model --tabulate --markerNames --quiet</pre>
<p>The resulting multipoint LOD scores are extracted from the output and
returned in R as a <a href="#topic+linkres">linkres</a> object.
</p>


<h3>Value</h3>

<p>If the number of markers is 1, or if <code>maxOnly = TRUE</code>, a single
number is returned.
</p>
<p>Otherwise a <code>linkres</code> object similar to the output of <code><a href="#topic+lod">lod()</a></code>, but with an
additional column <code>CM</code>.
</p>


<h3>References</h3>

<p><a href="https://csg.sph.umich.edu/abecasis/merlin/">https://csg.sph.umich.edu/abecasis/merlin/</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lod">lod()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#---------------------------------
# Requires MERLIN to be installed
#---------------------------------

if(pedprobr::checkMerlin()) {

# Pedigree with a single marker
x = nuclearPed(3, sex = c(1,2,2)) |&gt;
  addMarker(geno = c("1/1", "1/2", "1/2", "1/2", "1/2"))

# Simple AD model
merlinLod(x, aff = 2:5, model = diseaseModel("AD"))

# With liability classes
mod = diseaseModel("AD", penetrances = cbind(f0 = 0, f1 = 1:0, f2 = 1:0))
merlinLod(x, aff = 2:4, mod, liability = c(1,1,1,1,2))

# X
merlinLod(x, aff = 3:5, model = diseaseModel("XR"))

}

</code></pre>

<hr>
<h2 id='paramlink2'>paramlink2: Parametric linkage analysis</h2><span id='topic+paramlink2'></span>

<h3>Description</h3>

<p>Parametric linkage analysis of monogenic traits in medical pedigrees.
Features include singlepoint analysis, multipoint analysis via 'MERLIN'
(Abecasis et al. (2002) <a href="https://doi.org/10.1038/ng786">doi:10.1038/ng786</a>), visualisation of log of the
odds (LOD) scores and summaries of linkage peaks. Disease models may be
specified to  accommodate phenocopies, reduced penetrance and liability
classes. 'paramlink2' is part of the 'ped suite' package ecosystem, presented
in 'Pedigree Analysis in R' (Vigeland, 2021, ISBN:9780128244302).
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
