<!DOCTYPE html><html><head><title>Help for package partitionMetric</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {partitionMetric}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AhRs'><p>Sample data for partitionMetric</p></a></li>
<li><a href='#partitionMetric'><p>Compute a distance metric between two partitions of a set</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Compute a distance metric between two partitions of a set</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2014-03-01</td>
</tr>
<tr>
<td>Author:</td>
<td>David Weisman, Dan Simovici</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>David Weisman &lt;David.Weisman@acm.org&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10.1)</td>
</tr>
<tr>
<td>Description:</td>
<td>partitionMetric computes a distance between two partitions
        of a set.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-2-Clause">BSD_2_clause</a> + file LICENSE</td>
</tr>
<tr>
<td>Copyright:</td>
<td>Copyright (c) 2010, David Weisman and Dan Simovici, All
rights reserved.</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2012-10-29 08:59:23 UTC; ripley</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2014-03-02 14:03:31</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
</table>
<hr>
<h2 id='AhRs'>Sample data for partitionMetric</h2><span id='topic+AhRs'></span>

<h3>Description</h3>

<p>This small dataset contains aligned protein sequences for seven alleles of the
aryl hydrocarbon receptor (AhR).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(AhRs)</code></pre>


<h3>Format</h3>

<p>The format is a character matrix in which column <code class="reqn">i</code> represents
the <code class="reqn">i</code>'th position in the alignment, and contains an amino
acid code or &quot;-&quot; indicating an indel.  Row names contain the
animal species. 
</p>


<h3>Details</h3>

<p> A DNA or protein sequence has an associated index set
<code class="reqn">\{1,2,\ldots,n\}</code> that labels the <code class="reqn">n</code>
positions of the nucleotides or amino acids (AA).
This index set can be partitioned such that all members referring to
the same AA share a homogeneous partition.
For example, given the sequence <code style="white-space: pre;">&#8288;ATGTA&#8288;</code> and its index
set <code class="reqn">\{1,2,\ldots,5\}</code>, the &quot;A&quot; partition
contains the subset <code class="reqn">\{1,5\}</code>, the &quot;T&quot; partition contains
<code class="reqn">\{2,4\}</code>, and so on.
</p>
<p>Given two aligned sequences and their respective partitions of the
index set, a metric distance between these partitions can be computed.  See
<code><a href="#topic+partitionMetric">partitionMetric</a></code> for such a metric, along with an example
of clustering this AhR dataset.  </p>


<h3>Source</h3>

<p>This dataset was derived from NCBI HomoloGene:1224.

</p>


<h3>References</h3>

<p>Mark Hahn, Aryl hydrocarbon receptors: diversity and evolution. <em>Chem
Biol Interact</em>, 2002, <strong>141</strong>, 131-160 
</p>

<hr>
<h2 id='partitionMetric'>Compute a distance metric between two partitions of a set</h2><span id='topic+partitionMetric'></span>

<h3>Description</h3>

<p>Given a set partitioned in two ways, compute a distance metric between
the partitions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>partitionMetric(B, C, beta = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="partitionMetric_+3A_b">B</code></td>
<td>


<p>B and C are vectors that represents partitions of a single set, with
each element representing a member of the set.  <code class="reqn">B_i</code>
corresponds to <code class="reqn">C_i</code>, and the two vectors must be the same
length.  The data types of B and C must be identical and convertable to a
factor data type.
</p>
<p>See examples below for more information.
</p>
</td></tr> 
<tr><td><code id="partitionMetric_+3A_c">C</code></td>
<td>
<p>See B above.

</p>
</td></tr>
<tr><td><code id="partitionMetric_+3A_beta">beta</code></td>
<td>

<p><code class="reqn">\beta</code> is the nonlinear parameter used to compute the
distance metric.  See the publication referenced below for full details.

</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The return value is a nonnegative real number representing the distance
between the two partition of the set.  Full details are in the paper
referenced below.





</p>


<h3>Author(s)</h3>

<p>David Weisman, Dan Simovici

</p>


<h3>References</h3>

<p>David Weisman and Dan Simovici, 
Several Remarks on the Metric Space of Genetic Codes.
<em>International
Journal of Data Mining and Bioinformatics</em>, 2012(6).
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+as.dist">as.dist</a></code>, <code><a href="stats.html#topic+hclust">hclust</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Define several partitions of a 4-element set
gender &lt;- c('boy',   'girl', 'girl',   'boy')
height &lt;- c('short', 'tall', 'medium', 'tall')
age    &lt;- c(7,       6,      5,        4)    

## Compute some distances
(dGG &lt;- partitionMetric (gender, gender))
(dGH &lt;- partitionMetric (gender, height))
(dHG &lt;- partitionMetric (height, gender))
(dGA &lt;- partitionMetric (gender, age))
(dHA &lt;- partitionMetric (height, age))

## These properties must hold for any metric 
dGG == 0
dGH == dHG
dGA &lt;= dGH + dHA

## Note that the partition names are irrelevant, and only need to be
## self-consistent within each B and C.  It follows that these two set
## partitions are identical and have distance 0.
partitionMetric (c(1,8,8), c(7,3,3)) == 0


## Use the set partition to measure amino acid acid sequence differences
## between several alleles of the aryl hydrocarbon receptor.

data(AhRs)
dim(AhRs)
AhRs[,1:10]

distanceMatrix &lt;-
  matrix(nrow=nrow(AhRs), ncol=nrow(AhRs), 0,
         dimnames=list(rownames(AhRs), rownames(AhRs)))

for (pair in combn(rownames(AhRs), 2, simplify=FALSE)) {
  d &lt;- partitionMetric (AhRs[pair[1],], AhRs[pair[2],], beta=1.01)
  distanceMatrix[pair[1],pair[2]] &lt;- distanceMatrix[pair[2],pair[1]] &lt;- d
}

hc &lt;- hclust(as.dist(distanceMatrix))
plot(hc,
     sub=sprintf('Cophenentic correlation between distances and tree is %0.2f',
       cor(as.dist(distanceMatrix), cophenetic(hc))))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
