<!DOCTYPE html><html lang="en"><head><title>Help for package PAS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PAS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#binmod.plot'>
<p>plot function</p></a></li>
<li><a href='#binmod.predict'>
<p>predict</p></a></li>
<li><a href='#PAS-package'>
<p>Polygenic Analysis System (PAS)</p></a></li>
<li><a href='#PAS.binmod'>
<p>binmod</p></a></li>
<li><a href='#PAS.binmod.beef'>
<p>beef data</p></a></li>
<li><a href='#PAS.binmod.print'>
<p>binmod.print</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Polygenic Analysis System (PAS)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-06-13</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhiqiu Hu; Shizhong Xu; Zhiquan Wang; Rongcai Yang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zhiqiu Hu &lt;zhiqiu.hu@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>An R package for polygenic trait analysis. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), glmnet</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://statgen.ucr.edu">http://statgen.ucr.edu</a>, <a href="http://www.ualberta.ca/~zhiqiu1">http://www.ualberta.ca/~zhiqiu1</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-06-13 17:30:08 UTC; statgen</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-06-13 17:40:35 UTC</td>
</tr>
</table>
<hr>
<h2 id='binmod.plot'>
plot function
</h2><span id='topic+PAS.binmod.plot'></span><span id='topic+binmod.plot'></span><span id='topic+plot'></span><span id='topic+plot.binmod'></span>

<h3>Description</h3>

<p>Generate figures using an object created by the binmod function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'binmod'
plot(x, file=NULL, width=7, height=5, getdata=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="binmod.plot_+3A_x">x</code></td>
<td>
<p>An object generated by the binnod function.</p>
</td></tr>
<tr><td><code id="binmod.plot_+3A_file">file</code></td>
<td>
<p>The prefix of the figure files to be saved.</p>
</td></tr>
<tr><td><code id="binmod.plot_+3A_width">width</code></td>
<td>
<p>width of the figures (inch).</p>
</td></tr>
<tr><td><code id="binmod.plot_+3A_height">height</code></td>
<td>
<p>height of the figures (inch).</p>
</td></tr>
<tr><td><code id="binmod.plot_+3A_getdata">getdata</code></td>
<td>
<p>A logic indicator. The default value is FALSE, which mean not to return the data for plotting.</p>
</td></tr>
<tr><td><code id="binmod.plot_+3A_...">...</code></td>
<td>
<p>Further graphical parameters may also be supplied as arguments.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#load PAS library
library (PAS)
#load the demo data
data(beef)
#conduct bin model analysis and plotting the result.
plot(binmod(x, y, map))
</code></pre>

<hr>
<h2 id='binmod.predict'>
predict
</h2><span id='topic+PAS.binmod.predict'></span><span id='topic+binmod.predict'></span><span id='topic+predict'></span><span id='topic+predict.binmod'></span>

<h3>Description</h3>

<p>Extract predicted genomic breeding values from the 10-fold cross-validation result that has been saved in a binmod object, or predict the breeding values for a new sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'binmod'
predict(object, newx=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="binmod.predict_+3A_object">object</code></td>
<td>
<p>An object generated by the binnod function.</p>
</td></tr>
<tr><td><code id="binmod.predict_+3A_newx">newx</code></td>
<td>

<p>The numeric genotype indicator matrix of a new sample, which need to be coded in the same way as the genotypicdata generating the binmod object.
</p>
</td></tr>
<tr><td><code id="binmod.predict_+3A_...">...</code></td>
<td>
<p>Further parameters may also be supplied as arguments.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#load PAS library
library(PAS)
#load the demo data
data(beef)
#conduct bin model analysis.
binmod.result=binmod(x, y, map)
#generate a new sample by sampling 20 individuals from the demo data
x0=x[sample(1:NROW(x) , 20), ]
#predict the genomic values of the new sample.
predict(binmod.result, newx=x0)
</code></pre>

<hr>
<h2 id='PAS-package'>
Polygenic Analysis System (PAS)
</h2><span id='topic+PAS'></span>

<h3>Description</h3>

<p>The PAS package was developed to implement the method and algorithm developed by Zhiqiu Hu, Shizhong Xu, Zhiquan Wang, and Rongcai Yang for genomic value prediction. Although the current version of the package only provided functions for the bin model analysis (Hu et al., 2012), the package will be developed continuously to incorporate new methods of genomic value prediction that will be introduced by the authors in the near future.<br /><br />
<b>updates:</b><br />
1. A new option <em>foldid</em> was added into the binmod function to allow users assigning foldid for cross-validations;<br />
2. A new output item <em>obj$optimal$map.binsnp</em> was added bridging the bin map and the snp map in a binmod object.<br />
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> PAS</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2012-04-12</td>
</tr>
<tr>
 <td style="text-align: left;">
License: GPL (&gt;2.0)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Zhiqiu Hu, Shizhong Xu, Zhiquan Wang, Rong-cai Yang
</p>
<p>Maintainer: Zhiqiu Hu &lt;Zhiqiu.hu@gmail.com&gt;
</p>


<h3>References</h3>

<p><code>bin model</code><br /> Zhiqiu Hu, Zhiquan Wang, and Shizhong Xu (2012) An infinitesimal model for quantitative trait genomic value prediction. PloS ONE 7: e41336.
</p>

<hr>
<h2 id='PAS.binmod'>
binmod
</h2><span id='topic+PAS.binmod'></span><span id='topic+binmod'></span>

<h3>Description</h3>

<p>This is the main function for bin model analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>binmod(x, y, map, beta0=NA, binsizelist=-1, full.search=FALSE, foldid=NA, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PAS.binmod_+3A_x">x</code></td>
<td>

<p>input matrix, of dimensions nobs*nvars; each row is a observation vector of an individual and each column is a genotypic indicator vector for a molecular marker.
</p>
</td></tr>
<tr><td><code id="PAS.binmod_+3A_y">y</code></td>
<td>
<p>a matrix of response variable (phenotypic observations), of dimensions nobs*1.</p>
</td></tr>
<tr><td><code id="PAS.binmod_+3A_map">map</code></td>
<td>
<p>A data frame for linkage map or physical map.</p>
</td></tr>
<tr><td><code id="PAS.binmod_+3A_beta0">beta0</code></td>
<td>
<p>Estimated SNP effects obtained by univariate analysis. By default, the glm function in R will be called by the binmod to calculate the estimates of effects.</p>
</td></tr>
<tr><td><code id="PAS.binmod_+3A_binsizelist">binsizelist</code></td>
<td>
<p>A list of binsizes to be considered in the analysis. A default list will be generated if the option was ignored or an invalid list has been specified.</p>
</td></tr>
<tr><td><code id="PAS.binmod_+3A_full.search">full.search</code></td>
<td>

<p>A logic indicator selecting search strategies. If FALSE was assigned, the binmod will complete the running as soon as the optimal binsize was found. Otherwise, analysis will be conducted for all binsizes on the list. 
</p>
</td></tr>
<tr><td><code id="PAS.binmod_+3A_foldid">foldid</code></td>
<td>

<p>An optional vector of values between 1 and nfold identifying what fold each observation is in. If not supplied, a random vector is generated under nfold=10.
</p>
</td></tr>
<tr><td><code id="PAS.binmod_+3A_...">...</code></td>
<td>

<p>Other parameters need to be passed to glmnet/r and glm/r. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function invokes binmod analysis for genomic value prediction. The default settings are strongly suggested for new users.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>grid</code></td>
<td>
<p>information of all searched binsizes</p>
</td></tr>
<tr><td><code>grid$mselist</code></td>
<td>
<p>a 'data.frame': nbinsizes of 4 variables # A list of mean square errors</p>
</td></tr>
<tr><td><code>grid$mselist$binsize</code></td>
<td>
<p>size settings of the bins, eight in bp or cM.</p>
</td></tr>
<tr><td><code>grid$mselist$mse</code></td>
<td>
<p>mean square error</p>
</td></tr>
<tr><td><code>grid$mselist$mse_std</code></td>
<td>
<p>the standard deviation of MSEs</p>
</td></tr>
<tr><td><code>grid$mselist$nbin</code></td>
<td>
<p>number of bins under the binsize setting</p>
</td></tr>
<tr><td><code>grid$optbinsize</code></td>
<td>
<p>optimal binsize</p>
</td></tr>
<tr><td><code>grid$optid</code></td>
<td>
<p>order of the optimal binsize in the grid</p>
</td></tr>
<tr><td><code>optimal</code></td>
<td>
<p>result obtained under the optimal binsize</p>
</td></tr>
<tr><td><code>optimal$predict</code></td>
<td>
<p>phentypic values and its' predicted values under the optimal model.</p>
</td></tr>
<tr><td><code>optimal$predict$y</code></td>
<td>
<p>original phenotypic observations</p>
</td></tr>
<tr><td><code>optimal$predict$yp_cv</code></td>
<td>
<p>predictions by 10-fold cross-validation.</p>
</td></tr>
<tr><td><code>optimal$beta</code></td>
<td>
<p>estimated bin parameters</p>
</td></tr>
<tr><td><code>optimal$beta$beta</code></td>
<td>
<p>bin effect</p>
</td></tr>
<tr><td><code>optimal$beta$SSx</code></td>
<td>
<p>sum of square of bin indicator</p>
</td></tr>
<tr><td><code>optimal$beta$Se</code></td>
<td>
<p>residual error</p>
</td></tr>
<tr><td><code>optimal$beta$Sb</code></td>
<td>
<p>estimating error of bin effect</p>
</td></tr>
<tr><td><code>optimal$beta$Wald</code></td>
<td>
<p>Wald-test statistics</p>
</td></tr>
<tr><td><code>optimal$beta$LOD</code></td>
<td>
<p>LOD-test statistics</p>
</td></tr>
<tr><td><code>optimal$xbin</code></td>
<td>
<p>indicator matrix of the bins under the optimal binsize</p>
</td></tr>
<tr><td><code>optimal$map</code></td>
<td>
<p>'data.frame':	of  5 variables: #bin map</p>
</td></tr>
<tr><td><code>optimal$map$chr</code></td>
<td>
<p>chromosome id</p>
</td></tr>
<tr><td><code>optimal$map$pos</code></td>
<td>
<p>bin position</p>
</td></tr>
<tr><td><code>optimal$map$pos_id</code></td>
<td>
<p>mean of the orders of markers in the bin</p>
</td></tr>
<tr><td><code>optimal$map$start_id</code></td>
<td>
<p>the order the first maker in a bin</p>
</td></tr>
<tr><td><code>optimal$map$end_id</code></td>
<td>
<p>the order the last maker in a bin</p>
</td></tr>
<tr><td><code>optimal$binsize</code></td>
<td>
<p>optimal binsize</p>
</td></tr>
<tr><td><code>optimal$cv</code></td>
<td>
<p>cross-validation results</p>
</td></tr>
<tr><td><code>optimal$cv$binsize</code></td>
<td>
<p>binsize</p>
</td></tr>
<tr><td><code>optimal$cv$nbin</code></td>
<td>
<p>number of bins under the binsize setting</p>
</td></tr>
<tr><td><code>optimal$cv$mse</code></td>
<td>
<p>mean squared error obtained from cross-validation</p>
</td></tr>
<tr><td><code>optimal$cv$r</code></td>
<td>
<p>Pearson's correlation coefficient obtained from cross-validation</p>
</td></tr>
<tr><td><code>snp</code></td>
<td>
<p>SNP information</p>
</td></tr>
<tr><td><code>snp$map</code></td>
<td>
<p>linkage map or physical map</p>
</td></tr>
<tr><td><code>snp$map$chr</code></td>
<td>
<p>chromosome id</p>
</td></tr>
<tr><td><code>snp$map$pos</code></td>
<td>
<p>marker position</p>
</td></tr>
<tr><td><code>snp$map$pos_id</code></td>
<td>
<p>marker order</p>
</td></tr>
<tr><td><code>snp$effect</code></td>
<td>
<p>single marker analysis result</p>
</td></tr>
<tr><td><code>snp$effect$beta</code></td>
<td>
<p>SNP effect</p>
</td></tr>
<tr><td><code>snp$effect$SSx</code></td>
<td>
<p>sum of square of genotypic indicator</p>
</td></tr>
<tr><td><code>snp$effect$Se</code></td>
<td>
<p>residual variance</p>
</td></tr>
<tr><td><code>snp$effect$Sb</code></td>
<td>
<p>estimating error of marker effect</p>
</td></tr>
<tr><td><code>snp$effect$Wald</code></td>
<td>
<p>Wald-test statistics</p>
</td></tr>
<tr><td><code>snp$effect$LOD</code></td>
<td>
<p>LOD test statistics</p>
</td></tr>
<tr><td><code>snp$mapinfo</code></td>
<td>
<p>a brief summary of the map</p>
</td></tr>
<tr><td><code>snp$mapinfo$chr</code></td>
<td>
<p>chromosome id</p>
</td></tr>
<tr><td><code>snp$mapinfo$start</code></td>
<td>
<p>the position of the first marker on the chromosome</p>
</td></tr>
<tr><td><code>snp$mapinfo$end</code></td>
<td>
<p>the position of the last marker on the chromosome</p>
</td></tr>
<tr><td><code>snp$mapinfo$length</code></td>
<td>
<p>length of the chromosome</p>
</td></tr>
<tr><td><code>snp$mapinfo$nmark</code></td>
<td>
<p>number of markers on the chromosome</p>
</td></tr>
<tr><td><code>snp$mapinfo$aver</code></td>
<td>
<p>average interval of the chromosome</p>
</td></tr>
<tr><td><code>snp$mapinfo$min.interval</code></td>
<td>
<p>the smallest interval size on the chromosome</p>
</td></tr>
<tr><td><code>cvfit</code></td>
<td>
<p>A cv.glmnet project. See manual of glmnet for details.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Zhiqiu Hu, Zhiquan Wang, and Shizhong Xu (2012) An infinitesimal model for quantitative trait genomic value prediction. PloS ONE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load PAS library
library (PAS)
#load the demo data
data (beef)
#perform binmod analysis under the default settings.
binmod.result=binmod (x , y , map)
#plot binmod result
plot(binmod.result)
str(binmod.result)
#Output the predicted phenotypic values that was obtained
#by 10-fold cross validation .
predict(binmod.result)
#predict the phenotypic values for new individuals
x1=x[sample(1:NROW(x) , 20), ]
bin.pred.x1=predict(binmod.result, newx=x1)
str(bin.pred.x1)
</code></pre>

<hr>
<h2 id='PAS.binmod.beef'>
beef data
</h2><span id='topic+PAS.binmod.beef'></span><span id='topic+data'></span><span id='topic+x'></span><span id='topic+y'></span><span id='topic+map'></span>

<h3>Description</h3>

<p>The data are provide for demonstration purpose only.
</p>


<h3>Value</h3>


<dl>
<dt><code>x</code></dt><dd><p>genotypic data. int [1:836, 1:300] 0 0 0 -1 -1 -1 -1 0 0 0 ...</p>
</dd>
<dt><code>y</code></dt><dd><p>phenotypic data. int [1:836, 1] 768 157 508 614 590 777 505 243 509 351 ...</p>
</dd>
<dt><code>map</code></dt><dd><p>physical map: 'data.frame':	300 obs. of  2 variables:<br />
$ chr: num  1 1 1 1 ...<br />
$ pos: int  113641 244698 369418 447277 ..<br /></p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>#load PAS library
library (PAS)
#load example data
data(beef)
str(x)
str(y)
str(map)
</code></pre>

<hr>
<h2 id='PAS.binmod.print'>
binmod.print
</h2><span id='topic+PAS.binmod.print'></span><span id='topic+binmod.print'></span><span id='topic+print'></span><span id='topic+print.binmod'></span>

<h3>Description</h3>

<p>Show a terse summarize for a binmod object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'binmod'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PAS.binmod.print_+3A_x">x</code></td>
<td>
<p>An binmod object.</p>
</td></tr>
<tr><td><code id="PAS.binmod.print_+3A_...">...</code></td>
<td>
<p>Further parameters may also be supplied as arguments.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>#load PAS library
library (PAS)
#load the demo data
data (beef)
#conduct bin model analysis.
binmod.result=binmod (x, y, map)
print(binmod.result)
#show structure of a binmod object
str(binmod.result)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
