<!DOCTYPE html><html lang="en"><head><title>Help for package PathwaySpace</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PathwaySpace}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PathwaySpace-package'><p>PathwaySpace: Spatial Projection of Network Signals along Geodesic Paths</p></a></li>
<li><a href='#buildPathwaySpace'><p>Constructor of PathwaySpace-class objects.</p></a></li>
<li><a href='#CGC_20211118'><p>COSMIC-CGC genes mapped to PathwaySpace images.</p></a></li>
<li><a href='#circularProjection+2CPathwaySpace-method'><p>Creating 2D-landscape images from graph objects.</p></a></li>
<li><a href='#expDecay'><p>Exponential decay function.</p></a></li>
<li><a href='#getPathwaySpace+2CPathwaySpace-method'><p>Accessors for fetching slots from a PathwaySpace object.</p></a></li>
<li><a href='#gimage'><p>An image matrix.</p></a></li>
<li><a href='#gtoys'><p>Toy 'igraph' objects.</p></a></li>
<li><a href='#Hallmarks_v2023_1_Hs_symbols'><p>A list with Hallmark gene sets (v2023.1).</p></a></li>
<li><a href='#length+2CPathwaySpace-method'><p>Accessor function for PathwaySpace objects.</p></a></li>
<li><a href='#linearDecay'><p>A simple linear decay function.</p></a></li>
<li><a href='#names+2CPathwaySpace-method'><p>Accessor functions for PathwaySpace objects.</p></a></li>
<li><a href='#pathDistances'><p>Calculate a pathway space distance between two vectors.</p></a></li>
<li><a href='#PathwaySpace-class'><p>PathwaySpace: An S4 class for signal propagation on image spaces.</p></a></li>
<li><a href='#PCv12_pruned_igraph'><p>A pruned and laid out igraph object from Pathway Commons V12.</p></a></li>
<li><a href='#plotPathDistances'><p>Accessory function to plot pathway space distances.</p></a></li>
<li><a href='#plotPathwaySpace+2CPathwaySpace-method'><p>Plotting 2D-landscape images for the PathwaySpace package.</p></a></li>
<li><a href='#polarProjection+2CPathwaySpace-method'><p>Creating 2D-landscape images from graph objects.</p></a></li>
<li><a href='#pspace.cols'><p>A simple vector of colors for PathwaySpace images.</p></a></li>
<li><a href='#silhouetteMapping+2CPathwaySpace-method'><p>Decorating PathwaySpace images with graph silhouettes.</p></a></li>
<li><a href='#summitMapping+2CPathwaySpace-method'><p>Mapping summits on PathwaySpace images.</p></a></li>
<li><a href='#summitWatershed'><p>Variation of the watershed algorithm for summit detection.</p></a></li>
<li><a href='#vertexSignal+2CPathwaySpace-method'><p>Accessor functions for fetching slots from a PathwaySpace object.</p></a></li>
<li><a href='#vertexWeight+2CPathwaySpace-method'><p>Accessor functions for fetching slots from a PathwaySpace object.</p></a></li>
<li><a href='#weibullDecay'><p>Weibull decay function.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Spatial Projection of Network Signals along Geodesic Paths</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 4.4), methods, RGraphSpace</td>
</tr>
<tr>
<td>Imports:</td>
<td>grDevices, stats, scales, igraph, RANN, ggplot2, ggrepel</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat, RedeR</td>
</tr>
<tr>
<td>Description:</td>
<td>For a given graph containing vertices, edges, and a signal associated with the vertices, the 'PathwaySpace' package performs a convolution operation, which involves a weighted combination of neighboring vertices and their associated signals. The package then uses a decay function to project these signals, creating geodesic paths on a 2D-image space. 'PathwaySpace' could have various applications, such as visualizing and analyzing network data in a graphical format that highlights the relationships and signal strengths between vertices. It can be particularly useful for understanding the influence of signals through complex networks. By combining graph theory, signal processing, and visualization, the 'PathwaySpace' package provides a novel way of representing and analyzing graph data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/sysbiolab/PathwaySpace">https://github.com/sysbiolab/PathwaySpace</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/sysbiolab/PathwaySpace/issues">https://github.com/sysbiolab/PathwaySpace/issues</a></td>
</tr>
<tr>
<td>Collate:</td>
<td>pspaceChecks.R pspaceSupplements.R pspaceMisc.R
pspaceWatershed.R pspaceDecay.R pspaceClasses.R
pspaceGenerics.R pspaceMethods.R pspacePlots.R</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-12 01:31:31 UTC; maac</td>
</tr>
<tr>
<td>Author:</td>
<td>Victor Apolonio [ctr],
  Vinicius Chagas [ctr],
  Mauro Castro <a href="https://orcid.org/0000-0003-4942-8131"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mauro Castro &lt;mauro.a.castro@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-12 03:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='PathwaySpace-package'>PathwaySpace: Spatial Projection of Network Signals along Geodesic Paths</h2><span id='topic+PathwaySpace'></span><span id='topic+PathwaySpace-package'></span>

<h3>Description</h3>

<p>For a given graph containing vertices, edges, and a signal associated with
the vertices, the PathwaySpace package performs a convolution operation,
which involves a weighted combination of neighboring vertices and their
associated signals. The package then uses a decay function to propagate
these signals, creating geodesic paths on a 2D-image space.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> PathwaySpace</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Software</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> Artistic-2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Mauro Castro <a href="mailto:mauro.a.castro@gmail.com">mauro.a.castro@gmail.com</a></td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Index</h3>


<table>
<tr>
 <td style="text-align: left;">
<a href="#topic+PathwaySpace-class">PathwaySpace-class</a>: 
</td><td style="text-align: left;"> An S4 class for signal propagation on pathway spaces.</td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+buildPathwaySpace">buildPathwaySpace</a>: 
</td><td style="text-align: left;"> Constructor of PathwaySpace-class objects.</td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+circularProjection">circularProjection</a>: 
</td><td style="text-align: left;"> Creating 2D-landscape images from graph objects.</td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+polarProjection">polarProjection</a>: 
</td><td style="text-align: left;"> Creating 2D-landscape images from graph objects.</td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+silhouetteMapping">silhouetteMapping</a>: 
</td><td style="text-align: left;"> Mapping graph silhouettes on PathwaySpace images.</td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+summitMapping">summitMapping</a>: 
</td><td style="text-align: left;"> Mapping summits on a 2D-landscape image.</td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+getPathwaySpace">getPathwaySpace</a>: 
</td><td style="text-align: left;"> Accessory method for fetching slots from a PathwaySpace object.</td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+plotPathwaySpace">plotPathwaySpace</a>: 
</td><td style="text-align: left;"> Plotting 2D-landscape images for the PathwaySpace package.</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Further information is available in the vignettes by typing
<code>vignette('PathwaySpace')</code>. Documented topics are also available in
HTML by typing <code>help.start()</code> and selecting the PathwaySpace package
from the menu.
</p>


<h3>Author(s)</h3>

<p>The Cancer Genome Atlas Analysis Network.
</p>


<h3>References</h3>

<p>The Cancer Genome Atlas Analysis Network (2023). PathwaySpace: Spatial propagation 
of network signals along geodesic paths. R package version 0.99.
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/sysbiolab/PathwaySpace">https://github.com/sysbiolab/PathwaySpace</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/sysbiolab/PathwaySpace/issues">https://github.com/sysbiolab/PathwaySpace/issues</a>
</p>
</li></ul>


<hr>
<h2 id='buildPathwaySpace'>Constructor of PathwaySpace-class objects.</h2><span id='topic+buildPathwaySpace'></span>

<h3>Description</h3>

<p><code>buildPathwaySpace</code> is a constructor of
PathwaySpace-class objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildPathwaySpace(g, nrc = 500, mar = 0.075, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="buildPathwaySpace_+3A_g">g</code></td>
<td>
<p>An <code>igraph</code> object. It must include graph
layout information, with vertex coordinates assigned to <code>x</code> 
and <code>y</code> vertex attributes. It must also include vertex labels 
assigned to the <code>name</code> vertex attribute.</p>
</td></tr>
<tr><td><code id="buildPathwaySpace_+3A_nrc">nrc</code></td>
<td>
<p>A single positive integer indicating the number of rows and 
columns (in pixels) for a square image matrix. This argument will 
affect the resulting image size and resolution.</p>
</td></tr>
<tr><td><code id="buildPathwaySpace_+3A_mar">mar</code></td>
<td>
<p>A single numeric value (in <code>[0,1]</code>) indicating the size of
the outer margins as a fraction of the image matrix.</p>
</td></tr>
<tr><td><code id="buildPathwaySpace_+3A_verbose">verbose</code></td>
<td>
<p>A single logical value specifying to display detailed 
messages (when <code>verbose=TRUE</code>) or not (when <code>verbose=FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A preprocessed <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.
</p>


<h3>Author(s)</h3>

<p>Vinicius Chagas, Victor Apolonio, Mauro Castro,
and TCGA Network.
</p>


<h3>See Also</h3>

<p><code><a href="igraph.html#topic+undirected_graph">undirected_graph</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load a demo igraph
data('gtoy1', package = 'PathwaySpace')

# Create a new PathwaySpace object
pts &lt;- buildPathwaySpace(gtoy1, nrc = 100)
# note: adjust 'nrc' to increase image resolution

</code></pre>

<hr>
<h2 id='CGC_20211118'>COSMIC-CGC genes mapped to PathwaySpace images.</h2><span id='topic+CGC_20211118'></span>

<h3>Description</h3>

<p>A data frame listing 'GeneSymbol' and 'Entrez' IDs from the
COSMIC-CGC database (Sondka et al., 2020). These genes are used to
demonstrate the PathwaySpace's summit mapping pipeline, which assigns 
summits to an image space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(CGC_20211118)
</code></pre>


<h3>Format</h3>

<p>data.frame
</p>


<h3>Value</h3>

<p>A data.frame object.
</p>


<h3>Source</h3>

<p>COSMIC-CGC database (release v95, tier 1 collection).
</p>


<h3>References</h3>

<p>Sondka et al. The COSMIC Cancer Gene Census: describing genetic
dysfunction across all human cancers. Nat Rev Cancer 18, 696-705, 2018.
Doi: 10.1038/s41568-018-0060-1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(CGC_20211118)
</code></pre>

<hr>
<h2 id='circularProjection+2CPathwaySpace-method'>Creating 2D-landscape images from graph objects.</h2><span id='topic+circularProjection+2CPathwaySpace-method'></span><span id='topic+circularProjection'></span>

<h3>Description</h3>

<p><code>circularProjection</code> implements a convolution
algorithm to project signal across a 2D-coordinate system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'PathwaySpace'
circularProjection(
  pts,
  knn = 8,
  pdist = 0.15,
  rescale = TRUE,
  verbose = TRUE,
  decay_fun = weibullDecay,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circularProjection+2B2CPathwaySpace-method_+3A_pts">pts</code></td>
<td>
<p>A <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.</p>
</td></tr>
<tr><td><code id="circularProjection+2B2CPathwaySpace-method_+3A_knn">knn</code></td>
<td>
<p>A single positive integer determining the k-nearest signal 
sources used in the signal convolution operation.</p>
</td></tr>
<tr><td><code id="circularProjection+2B2CPathwaySpace-method_+3A_pdist">pdist</code></td>
<td>
<p>A term (in <code>[0,1]</code>) determining a distance unit for the signal
convolution related to the image space. This distance will affect the
extent over which the convolution operation projects the signal between
source- and destination points.</p>
</td></tr>
<tr><td><code id="circularProjection+2B2CPathwaySpace-method_+3A_rescale">rescale</code></td>
<td>
<p>A single logical value indicating whether to rescale 
the signal. If the signal <code>&gt;=0</code>, then it will be rescaled to 
<code>[0,1]</code>; if the signal <code>&lt;=0</code>, then it will be rescaled to 
<code>[-1,0]</code>; and if the signal in <code>(-Inf,+Inf)</code>, then it will be 
rescaled to <code>[-1,1]</code>.</p>
</td></tr>
<tr><td><code id="circularProjection+2B2CPathwaySpace-method_+3A_verbose">verbose</code></td>
<td>
<p>A single logical value specifying to display detailed 
messages (when <code>verbose=TRUE</code>) or not (when <code>verbose=FALSE</code>).</p>
</td></tr>
<tr><td><code id="circularProjection+2B2CPathwaySpace-method_+3A_decay_fun">decay_fun</code></td>
<td>
<p>A signal decay function. Available: 'Weibull',
'exponential', and 'linear' functions (see <code><a href="#topic+weibullDecay">weibullDecay</a></code>).</p>
</td></tr>
<tr><td><code id="circularProjection+2B2CPathwaySpace-method_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the decay function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A preprocessed <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.
</p>


<h3>Author(s)</h3>

<p>Vinicius Chagas, Victor Apolonio, Mauro Castro, 
and TCGA Network.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+buildPathwaySpace">buildPathwaySpace</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load a demo igraph
data('gtoy1', package = 'PathwaySpace')

# Create a new PathwaySpace object
pts &lt;- buildPathwaySpace(gtoy1, nrc = 100)
# note: adjust 'nrc' to increase image resolution

# Create a 2D-landscape image
pts &lt;- circularProjection(pts)

</code></pre>

<hr>
<h2 id='expDecay'>Exponential decay function.</h2><span id='topic+expDecay'></span>

<h3>Description</h3>

<p>The <code>expDecay</code> function is used by PathwaySpace's methods
for signal convolution and projection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expDecay(x, signal, decay = 0.999, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="expDecay_+3A_x">x</code></td>
<td>
<p>A numeric vector of distances (in [0,1]).</p>
</td></tr>
<tr><td><code id="expDecay_+3A_signal">signal</code></td>
<td>
<p>A single numeric value representing a signal.</p>
</td></tr>
<tr><td><code id="expDecay_+3A_decay">decay</code></td>
<td>
<p>The rate (in [0,1]) at which the signal decays.
This term indicates how much the <code>signal</code> decreases at a certain
distance in <code>x</code>. At the distance defined by the <code>pdist</code> term
(see <code><a href="#topic+circularProjection">circularProjection</a></code>), the signal's value will
correspond to the initial signal multiplied by <code>1 - decay</code>.</p>
</td></tr>
<tr><td><code id="expDecay_+3A_...">...</code></td>
<td>
<p>Not used; argument implemented for call compatibility with
the <code><a href="#topic+weibullDecay">weibullDecay</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector.
</p>


<h3>Author(s)</h3>

<p>Vinicius Chagas, Victor Apolonio, and
Mauro Castro (<a href="mailto:mauro.castro@ufpr.br">mauro.castro@ufpr.br</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+weibullDecay">weibullDecay</a></code>, <code><a href="#topic+linearDecay">linearDecay</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(1:100) / 100
y &lt;- expDecay(x, 1)
plot(x, y)

</code></pre>

<hr>
<h2 id='getPathwaySpace+2CPathwaySpace-method'>Accessors for fetching slots from a PathwaySpace object.</h2><span id='topic+getPathwaySpace+2CPathwaySpace-method'></span><span id='topic+getPathwaySpace'></span>

<h3>Description</h3>

<p><code>getPathwaySpace</code> retrives information from
individual slots available in a PathwaySpace object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'PathwaySpace'
getPathwaySpace(pts, what = "status")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getPathwaySpace+2B2CPathwaySpace-method_+3A_pts">pts</code></td>
<td>
<p>A preprocessed <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object</p>
</td></tr>
<tr><td><code id="getPathwaySpace+2B2CPathwaySpace-method_+3A_what">what</code></td>
<td>
<p>A single character value specifying which information should 
be retrieved from the slots.
Options: 'graph','gxy','gxyz','pars','misc','status','summits',
'summit_mask', and 'summit_contour'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Content from slots in the <a href="#topic+PathwaySpace-class">PathwaySpace</a> object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load a demo igraph
data('gtoy1', package = 'PathwaySpace')

# Create a new PathwaySpace object
pts &lt;- buildPathwaySpace(gtoy1, nrc = 100)
# note: adjust 'nrc' to increase image resolution

# Get the 'status' slot in pts
status &lt;- getPathwaySpace(pts, what = 'status')

</code></pre>

<hr>
<h2 id='gimage'>An image matrix.</h2><span id='topic+gimage'></span>

<h3>Description</h3>

<p>An image matrix used for workflow demonstrations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gimage)
</code></pre>


<h3>Format</h3>

<p>matrix
</p>


<h3>Value</h3>

<p>An image matrix.
</p>


<h3>Source</h3>

<p>This package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gimage)
</code></pre>

<hr>
<h2 id='gtoys'>Toy 'igraph' objects.</h2><span id='topic+gtoys'></span><span id='topic+gtoy1'></span><span id='topic+gtoy2'></span>

<h3>Description</h3>

<p>Small 'igraph' objects used for workflow demonstrations.
All graphs include 'x', 'y', and 'name' vertex attributes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gtoy1)
</code></pre>


<h3>Format</h3>

<p>igraph
</p>


<h3>Value</h3>

<p>A pre-processed igraph object.
</p>


<h3>Source</h3>

<p>This package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gtoy1)
data(gtoy2)
</code></pre>

<hr>
<h2 id='Hallmarks_v2023_1_Hs_symbols'>A list with Hallmark gene sets (v2023.1).</h2><span id='topic+Hallmarks_v2023_1_Hs_symbols'></span><span id='topic+hallmarks'></span>

<h3>Description</h3>

<p>A list with Human gene symbols from the MSigDB's Hallmark gene set 
collection (Liberzon et al., 2015). These gene sets are used to
demonstrate the PathwaySpace's summit mapping pipeline, which assigns 
summits to an image space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Hallmarks_v2023_1_Hs_symbols)
</code></pre>


<h3>Format</h3>

<p>list
</p>


<h3>Value</h3>

<p>A list object.
</p>


<h3>Source</h3>

<p>MSigDB database (v2023.1).
</p>


<h3>References</h3>

<p>Liberzon et al. The Molecular Signatures Database (MSigDB) hallmark 
gene set collection. Cell Systems 1(5):417-425, 2015
Doi: 10.1016/j.cels.2015.12.004
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Hallmarks_v2023_1_Hs_symbols)
</code></pre>

<hr>
<h2 id='length+2CPathwaySpace-method'>Accessor function for PathwaySpace objects.</h2><span id='topic+length+2CPathwaySpace-method'></span><span id='topic+length'></span>

<h3>Description</h3>

<p>Get length of a PathwaySpace object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'PathwaySpace'
length(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="length+2B2CPathwaySpace-method_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A non-negative integer of length 1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('gtoy1', package = 'PathwaySpace')
pts &lt;- buildPathwaySpace(gtoy1, nrc = 100)
length(pts)

</code></pre>

<hr>
<h2 id='linearDecay'>A simple linear decay function.</h2><span id='topic+linearDecay'></span>

<h3>Description</h3>

<p>The <code>linearDecay</code> function is used by PathwaySpace's
methods for signal convolution and projection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linearDecay(x, signal, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="linearDecay_+3A_x">x</code></td>
<td>
<p>A numeric vector of distances (in [0,1]).</p>
</td></tr>
<tr><td><code id="linearDecay_+3A_signal">signal</code></td>
<td>
<p>A single numeric value representing a signal.</p>
</td></tr>
<tr><td><code id="linearDecay_+3A_...">...</code></td>
<td>
<p>Not used; argument implemented for call compatibility with
<code><a href="#topic+weibullDecay">weibullDecay</a></code> and <code><a href="#topic+expDecay">expDecay</a></code> functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector.
</p>


<h3>Author(s)</h3>

<p>Vinicius Chagas, Victor Apolonio, and
Mauro Castro (<a href="mailto:mauro.castro@ufpr.br">mauro.castro@ufpr.br</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+weibullDecay">weibullDecay</a></code>, <code><a href="#topic+expDecay">expDecay</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(1:100) / 100
y &lt;- linearDecay(x, 1)
plot(x, y)

</code></pre>

<hr>
<h2 id='names+2CPathwaySpace-method'>Accessor functions for PathwaySpace objects.</h2><span id='topic+names+2CPathwaySpace-method'></span><span id='topic+names'></span>

<h3>Description</h3>

<p>Get and set 'vertex' names of a <a href="#topic+PathwaySpace-class">PathwaySpace</a>
class object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'PathwaySpace'
names(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="names+2B2CPathwaySpace-method_+3A_x">x</code></td>
<td>
<p>A <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('gtoy1', package = 'PathwaySpace')
pts &lt;- buildPathwaySpace(gtoy1, nrc = 100)
names(pts)

</code></pre>

<hr>
<h2 id='pathDistances'>Calculate a pathway space distance between two vectors.</h2><span id='topic+pathDistances'></span>

<h3>Description</h3>

<p>Calculate a pathway space distance between two vectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathDistances(gdist, from, to, nperm = 1000, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pathDistances_+3A_gdist">gdist</code></td>
<td>
<p>A distance matrix computed by the igraph's <code>distances</code> 
function. Rows and columns must be named with vertex labels as listed in
the 'igraph' object.</p>
</td></tr>
<tr><td><code id="pathDistances_+3A_from">from</code></td>
<td>
<p>A vector with valid vertex names.</p>
</td></tr>
<tr><td><code id="pathDistances_+3A_to">to</code></td>
<td>
<p>A vector with valid vertex names.</p>
</td></tr>
<tr><td><code id="pathDistances_+3A_nperm">nperm</code></td>
<td>
<p>Number of permutations.</p>
</td></tr>
<tr><td><code id="pathDistances_+3A_verbose">verbose</code></td>
<td>
<p>A single logical value specifying to display detailed 
messages (when <code>verbose=TRUE</code>) or not (when <code>verbose=FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with pathway space distances and a 'ggplot' object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotPathwaySpace">plotPathwaySpace</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load a vertex-wise distance matrix (distance between nodes in a graph)
data("gdist.toy", package = "PathwaySpace")

# Get two vertex lists
from &lt;- sample(colnames(gdist.toy), 50)
to &lt;- sample(colnames(gdist.toy), 50)

# Calculate distances between lists, and between random lists
res &lt;- pathDistances(gdist.toy, from, to)
names(res)
# "p_dist"  "z_score"

</code></pre>

<hr>
<h2 id='PathwaySpace-class'>PathwaySpace: An S4 class for signal propagation on image spaces.</h2><span id='topic+PathwaySpace-class'></span>

<h3>Description</h3>

<p>PathwaySpace: An S4 class for signal propagation on image spaces.
</p>


<h3>Value</h3>

<p>An S4 class object.
</p>


<h3>Slots</h3>


<dl>
<dt><code>vertex</code></dt><dd><p>A character vector with vertex names.</p>
</dd>
<dt><code>vsignal</code></dt><dd><p>A numerical vector with vertex signals.</p>
</dd>
<dt><code>vweight</code></dt><dd><p>A numerical vector with vertex weights.</p>
</dd>
<dt><code>edges</code></dt><dd><p>A data frame with edges.</p>
</dd>
<dt><code>gxy</code></dt><dd><p>A data frame with xy-vertex coordinates (numerical).</p>
</dd>
<dt><code>gxyz</code></dt><dd><p>A numerical matrix with x-cols and y-rows coordinates,
and a z-signal.</p>
</dd>
<dt><code>pars</code></dt><dd><p>A list with parameters.</p>
</dd>
<dt><code>misc</code></dt><dd><p>A list with intermediate objects for downstream methods.</p>
</dd>
<dt><code>status</code></dt><dd><p>A vector containing the processing status of the
PathwaySpace object.</p>
</dd>
</dl>


<h3>Constructor</h3>

<p>see <code><a href="#topic+buildPathwaySpace">buildPathwaySpace</a></code> constructor.
</p>


<h3>Author(s)</h3>

<p>Mauro Castro, <a href="mailto:mauro.castro@ufpr.br">mauro.castro@ufpr.br</a>
</p>

<hr>
<h2 id='PCv12_pruned_igraph'>A pruned and laid out igraph object from Pathway Commons V12.</h2><span id='topic+PCv12_pruned_igraph'></span><span id='topic+gdist.toy'></span>

<h3>Description</h3>

<p>This igraph object was created from a 'sif' file available from 
the Pathway Commons V12 (Rodchenkov et al., 2020), which was filtered to 
keep interactions from the following sources: CTD, Recon, HumanCyc, 
DrugBank, MSigDB, DIP, BioGRID, IntAct, BIND, and PhosphoSite. The igraph 
was additionally pruned and laid out by a force-directed algorithm aiming 
signal projection on PathwaySpace's images. Edges with the smallest 
betweenness centrality were pruned using 'backward elimination' and
'forward selection' strategies. The resulting graph represents the 
main connected component with the minimum number of edges.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(PCv12_pruned_igraph)
</code></pre>


<h3>Format</h3>

<p>igraph
</p>


<h3>Value</h3>

<p>An igraph object.
</p>


<h3>Author(s)</h3>

<p>Chris Wong, Mauro Castro, and TCGA Network.
</p>


<h3>Source</h3>

<p>Pathway Commons V12.
</p>


<h3>References</h3>

<p>Rodchenkov et al. Pathway Commons 2019 Update: integration, analysis and
exploration of pathway data. Nucleic Acids Research 48(D1):D489â€“D497, 2020.
<a href="https://doi.org/10.1093/nar/gkz946">doi:10.1093/nar/gkz946</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(PCv12_pruned_igraph)
## Suggestion to vizualize this igraph in R:
library(RGraphSpace)
plotGraphSpace(PCv12_pruned_igraph)
</code></pre>

<hr>
<h2 id='plotPathDistances'>Accessory function to plot pathway space distances.</h2><span id='topic+plotPathDistances'></span>

<h3>Description</h3>

<p>Accessory function to plot pathway space distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotPathDistances(pdist, z.transform = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotPathDistances_+3A_pdist">pdist</code></td>
<td>
<p>A list generated by the <code><a href="#topic+pathDistances">pathDistances</a></code> function.</p>
</td></tr>
<tr><td><code id="plotPathDistances_+3A_z.transform">z.transform</code></td>
<td>
<p>A single logical value specifying to convert pathway 
distances into z-score values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggplot' object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load a vertex-wise distance matrix (distance between nodes in a graph)
data("gdist.toy", package = "PathwaySpace")

# Get two gene lists
from &lt;- sample(colnames(gdist.toy), 50)
to &lt;- sample(colnames(gdist.toy), 50)

# Calculate distances between lists, and between random lists
res &lt;- pathDistances(gdist.toy, from, to)

# Plot observed and null distances
plotPathDistances(res)

</code></pre>

<hr>
<h2 id='plotPathwaySpace+2CPathwaySpace-method'>Plotting 2D-landscape images for the PathwaySpace package.</h2><span id='topic+plotPathwaySpace+2CPathwaySpace-method'></span><span id='topic+plotPathwaySpace'></span>

<h3>Description</h3>

<p><code>plotPathwaySpace</code> is a wrapper function to 
create dedicated ggplot graphics for PathwaySpace-class objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'PathwaySpace'
plotPathwaySpace(
  pts,
  colors = pspace.cols(),
  trim.colors = c(3, 2, 1, 2, 3),
  bg.color = "grey85",
  theme.name = c("th0", "th1", "th2", "th3"),
  title = "PathwaySpace",
  font.size = 1,
  font.color = "white",
  xlab = "Pathway coordinates 1",
  ylab = "Pathway coordinates 2",
  zlab = "Density",
  zlim = NULL,
  slices = 25,
  add.grid = TRUE,
  grid.color = "white",
  add.contour = TRUE,
  contour.color = "white",
  label.summits = TRUE,
  marks = FALSE,
  mark.size = 3,
  mark.color = "white",
  mark.padding = 0.5,
  mark.line.width = 0.5,
  use.dotmark = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_pts">pts</code></td>
<td>
<p>A <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_colors">colors</code></td>
<td>
<p>A vector of colors. Each color is a specific tone used to 
create a customized color palette that matches the signal type projected 
in the image space, such as binary or numeric values in (-Inf, +Inf). The 
palette will be created with 5-color tones, interpolated according to the
provided sequence in the vector of colors. The proportion of each color 
tone can be adjusted by the 'trim.colors' argument. The primary objective 
of this color palette construction is to enhance the visibility of summits 
and valleys in the image space. If you wish to bypass the automatic palette 
generation and use the 'colors' input as-is, simply set 'trim.colors' 
to NULL.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_trim.colors">trim.colors</code></td>
<td>
<p>An vector with 5 positive integer numbers. This argument
can be used to adjust the proportion of each color tone in the palette.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_bg.color">bg.color</code></td>
<td>
<p>A single color for background.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_theme.name">theme.name</code></td>
<td>
<p>Name of a custom PathwaySpace theme. These themes 
(from 'th0' to 'th3') consist mainly of preconfigured ggplot settings, 
which the user can subsequently fine-tune within the resulting 
ggplot object.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_title">title</code></td>
<td>
<p>A string for the title.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_font.size">font.size</code></td>
<td>
<p>A single numeric value passed to ggplot themes.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_font.color">font.color</code></td>
<td>
<p>A single color passed to ggplot themes.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_xlab">xlab</code></td>
<td>
<p>The title for the 'x' axis of a 2D-image space.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_ylab">ylab</code></td>
<td>
<p>The title for the 'y' axis of a 2D-image space.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_zlab">zlab</code></td>
<td>
<p>The title for the 'z' axis of the image signal.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_zlim">zlim</code></td>
<td>
<p>The 'z' limits of the plot (a numeric vector with two numbers).
If NULL, limits are determined from the range of the input values.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_slices">slices</code></td>
<td>
<p>A single positive integer value used to split 
the image signal into equally-spaced intervals.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_add.grid">add.grid</code></td>
<td>
<p>A logical value indicating whether to add gridlines to 
the image space. However, gridlines will only appear when the image 
is decorated with graph silhouettes (see <code><a href="#topic+silhouetteMapping">silhouetteMapping</a></code>).</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_grid.color">grid.color</code></td>
<td>
<p>A color passed to <code><a href="ggplot2.html#topic+geom_point">geom_point</a></code>.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_add.contour">add.contour</code></td>
<td>
<p>A logical value indicating whether to add contour 
lines to 'summits' (see <code><a href="#topic+summitMapping">summitMapping</a></code>).</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_contour.color">contour.color</code></td>
<td>
<p>A color passed to <code><a href="ggplot2.html#topic+geom_tile">geom_tile</a></code>.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_label.summits">label.summits</code></td>
<td>
<p>A logical value indicating whether to label summits,
(when summits are available).</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_marks">marks</code></td>
<td>
<p>A logical value indicating whether to add 'marks' to vertex 
positions. Alternatively, this could be a vector listing vertex names.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_mark.size">mark.size</code></td>
<td>
<p>A font size argument passed to 
<code><a href="ggplot2.html#topic+geom_text">geom_text</a></code>.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_mark.color">mark.color</code></td>
<td>
<p>A color passed to <code><a href="ggrepel.html#topic+geom_text_repel">geom_text_repel</a></code>.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_mark.padding">mark.padding</code></td>
<td>
<p>A box padding argument passed to 
<code><a href="ggrepel.html#topic+geom_text_repel">geom_text_repel</a></code>.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_mark.line.width">mark.line.width</code></td>
<td>
<p>A line width argument passed to 
<code><a href="ggrepel.html#topic+geom_text_repel">geom_text_repel</a></code>.</p>
</td></tr>
<tr><td><code id="plotPathwaySpace+2B2CPathwaySpace-method_+3A_use.dotmark">use.dotmark</code></td>
<td>
<p>A logical value indicating whether &quot;marks&quot; should be 
represented as dots.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot-class object.
</p>


<h3>Author(s)</h3>

<p>Vinicius Chagas, Victor Apolonio, Mauro Castro, 
and TCGA Network.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+circularProjection">circularProjection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load a demo igraph
data('gtoy1', package = 'PathwaySpace')

# Create a new PathwaySpace object
pts &lt;- buildPathwaySpace(gtoy1, nrc = 100)
# note: adjust 'nrc' to increase image resolution

# Create a 2D-landscape image
pts &lt;- circularProjection(pts)

# Map graph silhouette
pts &lt;- silhouetteMapping(pts)

# Plot a 2D-landscape image
plotPathwaySpace(pts)

</code></pre>

<hr>
<h2 id='polarProjection+2CPathwaySpace-method'>Creating 2D-landscape images from graph objects.</h2><span id='topic+polarProjection+2CPathwaySpace-method'></span><span id='topic+polarProjection'></span>

<h3>Description</h3>

<p><code>polarProjection</code> implements a convolution algorithm
to project a signal across a 2D-coordinate system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'PathwaySpace'
polarProjection(
  pts,
  knn = 8,
  pdist = 0.5,
  rescale = TRUE,
  theta = 180,
  directional = FALSE,
  verbose = TRUE,
  decay_fun = weibullDecay,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="polarProjection+2B2CPathwaySpace-method_+3A_pts">pts</code></td>
<td>
<p>A <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.</p>
</td></tr>
<tr><td><code id="polarProjection+2B2CPathwaySpace-method_+3A_knn">knn</code></td>
<td>
<p>A single positive integer determining the k-nearest signal 
sources used in the signal convolution operation.</p>
</td></tr>
<tr><td><code id="polarProjection+2B2CPathwaySpace-method_+3A_pdist">pdist</code></td>
<td>
<p>A term (in <code>[0,1]</code>) determining a distance unit for the
signal convolution related to length between any two connected vertices. 
This distance will affect the extent over which the convolution operation 
projects the signal between source- and destination points along the 
polar coordinates of the edges.</p>
</td></tr>
<tr><td><code id="polarProjection+2B2CPathwaySpace-method_+3A_rescale">rescale</code></td>
<td>
<p>A single logical value indicating whether to rescale 
the signal. If the signal <code>&gt;=0</code>, then it will be rescaled to 
<code>[0,1]</code>; if the signal <code>&lt;=0</code>, then it will be rescaled to 
<code>[-1,0]</code>; and if the signal in <code>(-Inf,+Inf)</code>, then it will be 
rescaled to <code>[-1,1]</code>.</p>
</td></tr>
<tr><td><code id="polarProjection+2B2CPathwaySpace-method_+3A_theta">theta</code></td>
<td>
<p>Angle of projection (degrees in <code>(0,360]</code>).</p>
</td></tr>
<tr><td><code id="polarProjection+2B2CPathwaySpace-method_+3A_directional">directional</code></td>
<td>
<p>If directional edges are available, this argument can 
be used to orientate the signal projection on directed graphs.</p>
</td></tr>
<tr><td><code id="polarProjection+2B2CPathwaySpace-method_+3A_verbose">verbose</code></td>
<td>
<p>A single logical value specifying to display detailed 
messages (when <code>verbose=TRUE</code>) or not (when <code>verbose=FALSE</code>).</p>
</td></tr>
<tr><td><code id="polarProjection+2B2CPathwaySpace-method_+3A_decay_fun">decay_fun</code></td>
<td>
<p>A signal decay function. Available: 'Weibull', 
'exponential', and 'linear' functions (see <code><a href="#topic+weibullDecay">weibullDecay</a></code>).</p>
</td></tr>
<tr><td><code id="polarProjection+2B2CPathwaySpace-method_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the decay function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A preprocessed <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.
</p>


<h3>Author(s)</h3>

<p>Vinicius Chagas, Victor Apolonio, Mauro Castro,
and TCGA Network.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+buildPathwaySpace">buildPathwaySpace</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load a demo igraph
data('gtoy1', package = 'PathwaySpace')

# Create a new PathwaySpace object
pts &lt;- buildPathwaySpace(gtoy1, nrc = 100)
# note: adjust 'nrc' to increase image resolution

# Create a 2D-landscape image
pts &lt;- polarProjection(pts)

</code></pre>

<hr>
<h2 id='pspace.cols'>A simple vector of colors for PathwaySpace images.</h2><span id='topic+pspace.cols'></span>

<h3>Description</h3>

<p>A simple vector of colors for PathwaySpace images.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pspace.cols(n = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pspace.cols_+3A_n">n</code></td>
<td>
<p>Number of colors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with hexadecimal color codes.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotPathwaySpace">plotPathwaySpace</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pspace.cols()

</code></pre>

<hr>
<h2 id='silhouetteMapping+2CPathwaySpace-method'>Decorating PathwaySpace images with graph silhouettes.</h2><span id='topic+silhouetteMapping+2CPathwaySpace-method'></span><span id='topic+silhouetteMapping'></span>

<h3>Description</h3>

<p><code>silhouetteMapping</code> constructs an image baseline used
to outline the graph layout in a PathwaySpace image.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'PathwaySpace'
silhouetteMapping(pts, baseline = 0.01, pdist = 0.05, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="silhouetteMapping+2B2CPathwaySpace-method_+3A_pts">pts</code></td>
<td>
<p>A <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.</p>
</td></tr>
<tr><td><code id="silhouetteMapping+2B2CPathwaySpace-method_+3A_baseline">baseline</code></td>
<td>
<p>A fraction (in <code>[0,1]</code>) of the signal scale of a 
PathwaySpace image. This term only affects the image baseline projection, 
which represents a silhouette of the graph's layout outlined in the 
resulting image. When <code>baseline = 0</code> (i.e. lower level of the signal 
scale), the baseline will extend over the entire image space, so no 
silhouette will be visible.</p>
</td></tr>
<tr><td><code id="silhouetteMapping+2B2CPathwaySpace-method_+3A_pdist">pdist</code></td>
<td>
<p>A term (in <code>0,1</code>) determining a distance unit for
the signal convolution related to the image space. This distance
will affect the extent over which the convolution operation
projects the image baseline.</p>
</td></tr>
<tr><td><code id="silhouetteMapping+2B2CPathwaySpace-method_+3A_verbose">verbose</code></td>
<td>
<p>A single logical value specifying to display detailed 
messages (when <code>verbose=TRUE</code>) or not (when <code>verbose=FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A preprocessed <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.
</p>


<h3>Author(s)</h3>

<p>Vinicius Chagas, Victor Apolonio, Mauro Castro,
and TCGA Network.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+circularProjection">circularProjection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load a demo igraph
data('gtoy1', package = 'PathwaySpace')

# Create a new PathwaySpace object
pts &lt;- buildPathwaySpace(gtoy1, nrc = 100)
# note: adjust 'nrc' to increase image resolution

# Create a 2D-landscape image
pts &lt;- circularProjection(pts)

# Map graph silhouette
pts &lt;- silhouetteMapping(pts)

</code></pre>

<hr>
<h2 id='summitMapping+2CPathwaySpace-method'>Mapping summits on PathwaySpace images.</h2><span id='topic+summitMapping+2CPathwaySpace-method'></span><span id='topic+summitMapping'></span>

<h3>Description</h3>

<p>The <code>summitMapping</code> method implements a segmentation
strategy to identify summits on a 2D-landscape image 
(see <code><a href="#topic+summitWatershed">summitWatershed</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'PathwaySpace'
summitMapping(
  pts,
  maxset = 30,
  minsize = 30,
  threshold = 0.5,
  verbose = TRUE,
  segm_fun = summitWatershed,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summitMapping+2B2CPathwaySpace-method_+3A_pts">pts</code></td>
<td>
<p>A <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.</p>
</td></tr>
<tr><td><code id="summitMapping+2B2CPathwaySpace-method_+3A_maxset">maxset</code></td>
<td>
<p>A single positive integer indicating the maximum number 
of summits to be returned by the segmentation function.</p>
</td></tr>
<tr><td><code id="summitMapping+2B2CPathwaySpace-method_+3A_minsize">minsize</code></td>
<td>
<p>A single positive integer indicating the minimum size 
of the summits.</p>
</td></tr>
<tr><td><code id="summitMapping+2B2CPathwaySpace-method_+3A_threshold">threshold</code></td>
<td>
<p>A threshold provided as a fraction (in <code>[0,1]</code>) of the
max signal intensity.</p>
</td></tr>
<tr><td><code id="summitMapping+2B2CPathwaySpace-method_+3A_verbose">verbose</code></td>
<td>
<p>A single logical value specifying to display detailed 
messages (when <code>verbose=TRUE</code>) or not (when <code>verbose=FALSE</code>).</p>
</td></tr>
<tr><td><code id="summitMapping+2B2CPathwaySpace-method_+3A_segm_fun">segm_fun</code></td>
<td>
<p>A segmentation function used to detect summits
(see <code><a href="#topic+summitWatershed">summitWatershed</a></code>).</p>
</td></tr>
<tr><td><code id="summitMapping+2B2CPathwaySpace-method_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to the segmentation function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A preprocessed <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.
</p>


<h3>Author(s)</h3>

<p>Vinicius Chagas, Victor Apolonio, Mauro Castro, 
and TCGA Network.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+circularProjection">circularProjection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load a demo igraph
data('gtoy1', package = 'PathwaySpace')

# Create a new PathwaySpace object
pts &lt;- buildPathwaySpace(gtoy1, nrc = 100)
# note: adjust 'nrc' to increase image resolution

# Create a 2D-landscape image
pts &lt;- circularProjection(pts)

# Map summits in a 2D-landscape image
pts &lt;- summitMapping(pts)

</code></pre>

<hr>
<h2 id='summitWatershed'>Variation of the watershed algorithm for summit detection.</h2><span id='topic+summitWatershed'></span>

<h3>Description</h3>

<p>The <code>summitWatershed</code> function implements a segmentation
strategy to identify summits within a landscape image generated by the
PathwaySpace package. This function is entirely coded in R, which helps
alleviating users from the task of loading an excessive number of
dependencies. Nonetheless, while this novel implementation prevents
the burden a 'dependency heaviness', it still requires optimization
as it currently exhibits slower performance compared to well-established
implementations such as the watershed function from the EBImage package.
The <code>summitWatershed</code> maintain a certain level of compatibility
with the EBImage's watershed function, and both can be used in the
PathwaySpace package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summitWatershed(x, tolerance = 0.1, ext = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summitWatershed_+3A_x">x</code></td>
<td>
<p>A 2D-numeric array in which each point represents the coordinates
of a signal in a landscape image.</p>
</td></tr>
<tr><td><code id="summitWatershed_+3A_tolerance">tolerance</code></td>
<td>
<p>The minimum signal intensity of a summit (in [0,1]),
representing a fraction of the maximum signal intensity.</p>
</td></tr>
<tr><td><code id="summitWatershed_+3A_ext">ext</code></td>
<td>
<p>Radius (in pixels) for detecting neighboring objects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with labeled summits.
</p>


<h3>Author(s)</h3>

<p>Vinicius Chagas, Victor Apolonio, and
Mauro Castro (<a href="mailto:mauro.castro@ufpr.br">mauro.castro@ufpr.br</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summitMapping">summitMapping</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load a demo landscape image
data('gimage', package = 'PathwaySpace')

# Scale down the image for a quicker demonstration
gimage &lt;- gimage[200:300, 200:300]

# Check signal range
range(gimage, na.rm = TRUE)
# [1] 0 1

# Check image

image(gimage)


# Threshold the signal intensity, for example:
gimage[gimage &lt; 0.5] &lt;- 0

# Run summit segmentation
gmask &lt;- summitWatershed(x = gimage)

# Check resulting image mask

image(gimage)


</code></pre>

<hr>
<h2 id='vertexSignal+2CPathwaySpace-method'>Accessor functions for fetching slots from a PathwaySpace object.</h2><span id='topic+vertexSignal+2CPathwaySpace-method'></span><span id='topic+vertexSignal'></span><span id='topic+vertexSignal+3C-'></span><span id='topic+vertexSignal+3C-+2CPathwaySpace-method'></span>

<h3>Description</h3>

<p>Get or set 'signal' for a <a href="#topic+PathwaySpace-class">PathwaySpace</a>
class object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'PathwaySpace'
vertexSignal(pts)

## S4 replacement method for signature 'PathwaySpace'
vertexSignal(pts) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vertexSignal+2B2CPathwaySpace-method_+3A_pts">pts</code></td>
<td>
<p>A <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.</p>
</td></tr>
<tr><td><code id="vertexSignal+2B2CPathwaySpace-method_+3A_value">value</code></td>
<td>
<p>A numeric vector with values representing signal
intensities. This vector should be aligned to the &quot;vertex&quot; slot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('gtoy1', package = 'PathwaySpace')
pts &lt;- buildPathwaySpace(gtoy1, nrc = 100)
vertexSignal(pts)

</code></pre>

<hr>
<h2 id='vertexWeight+2CPathwaySpace-method'>Accessor functions for fetching slots from a PathwaySpace object.</h2><span id='topic+vertexWeight+2CPathwaySpace-method'></span><span id='topic+vertexWeight'></span><span id='topic+vertexWeight+3C-'></span><span id='topic+vertexWeight+3C-+2CPathwaySpace-method'></span>

<h3>Description</h3>

<p>Get or set 'weights' for a <a href="#topic+PathwaySpace-class">PathwaySpace</a>
class object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'PathwaySpace'
vertexWeight(pts)

## S4 replacement method for signature 'PathwaySpace'
vertexWeight(pts) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vertexWeight+2B2CPathwaySpace-method_+3A_pts">pts</code></td>
<td>
<p>A <a href="#topic+PathwaySpace-class">PathwaySpace</a> class object.</p>
</td></tr>
<tr><td><code id="vertexWeight+2B2CPathwaySpace-method_+3A_value">value</code></td>
<td>
<p>A numeric vector with values representing vertex weights.
This vector should be aligned to the &quot;vertex&quot; slot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data('gtoy1', package = 'PathwaySpace')
pts &lt;- buildPathwaySpace(gtoy1, nrc = 100)
vertexWeight(pts)

</code></pre>

<hr>
<h2 id='weibullDecay'>Weibull decay function.</h2><span id='topic+weibullDecay'></span>

<h3>Description</h3>

<p>The <code>weibullDecay</code> function is used by
PathwaySpace's methods for signal convolution and projection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weibullDecay(x, signal, decay = 0.999, shape = 1.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="weibullDecay_+3A_x">x</code></td>
<td>
<p>A numeric vector of distances (in [0,1]).</p>
</td></tr>
<tr><td><code id="weibullDecay_+3A_signal">signal</code></td>
<td>
<p>A single numeric value representing a signal.</p>
</td></tr>
<tr><td><code id="weibullDecay_+3A_decay">decay</code></td>
<td>
<p>The rate (in [0,1]) at which the signal decays.
This term indicates how much the <code>signal</code> decreases at a certain
distance in <code>x</code>. At the distance defined by the <code>pdist</code> term
(see <code><a href="#topic+circularProjection">circularProjection</a></code>), the signal's value will
correspond to the initial signal multiplied by <code>1 - decay</code>.</p>
</td></tr>
<tr><td><code id="weibullDecay_+3A_shape">shape</code></td>
<td>
<p>A parameter (&gt;=1) of a Weibull function. When <code>shape=1</code>
the Weibull decay follows an exponential decay. When <code>shape&gt;1</code>
the function is first convex, then concave with an inflection point.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector.
</p>


<h3>Author(s)</h3>

<p>Vinicius Chagas, Victor Apolonio, and
Mauro Castro (<a href="mailto:mauro.castro@ufpr.br">mauro.castro@ufpr.br</a>)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+expDecay">expDecay</a></code>, <code><a href="#topic+linearDecay">linearDecay</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- c(1:100) / 100
y &lt;- weibullDecay(x, 1)
plot(x, y)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
