<!DOCTYPE html><html><head><title>Help for package pathfindR.data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pathfindR.data}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#biocarta_descriptions'><p>BioCarta Pathways - Descriptions</p></a></li>
<li><a href='#biocarta_genes'><p>BioCarta Pathways - Gene Sets</p></a></li>
<li><a href='#cell_markers_descriptions'><p>Human Cell Markers - Descriptions</p></a></li>
<li><a href='#cell_markers_gsets'><p>Human Cell Markers - Gene Sets</p></a></li>
<li><a href='#example_active_snws'><p>Example Active Subnetworks</p></a></li>
<li><a href='#example_comparison_output'><p>Second Example Output for the pathfindR Enrichment Workflow (H.sapiens. - Rheumatoid Arthritis data)</p></a></li>
<li><a href='#example_custom_genesets_result'><p>Custom Gene Set Enrichment Results</p></a></li>
<li><a href='#example_experiment_matrix'><p>Example Experiment Matrix for pathfindR - Enriched Term Scoring</p></a></li>
<li><a href='#example_mmu_input'><p>Example Input for Mus musculus - Myeloma Analysis</p></a></li>
<li><a href='#example_mmu_output'><p>Example Output for Mus musculus - Myeloma Analysis</p></a></li>
<li><a href='#example_pathfindR_input'><p>Example Input for the pathfindR Enrichment Workflow - Rheumatoid Arthritis (H.sapiens)</p></a></li>
<li><a href='#example_pathfindR_output'><p>Example Output for the pathfindR Enrichment Workflow - Rheumatoid Arthritis</p></a></li>
<li><a href='#example_pathfindR_output_clustered'><p>Example Output for the pathfindR Clustering Workflow - Rheumatoid Arthritis</p></a></li>
<li><a href='#go_all_genes'><p>Gene Ontology - All Gene Ontology Gene Sets</p></a></li>
<li><a href='#kegg_descriptions'><p>KEGG Pathways - Descriptions</p></a></li>
<li><a href='#kegg_genes'><p>KEGG Pathways - Gene Sets</p></a></li>
<li><a href='#mmu_kegg_descriptions'><p>Mus Musculus KEGG Pathways - Descriptions</p></a></li>
<li><a href='#mmu_kegg_genes'><p>Mus Musculus KEGG Pathways - Gene Sets</p></a></li>
<li><a href='#pathfindR.data_updates'><p>Table of Data for pathfindR</p></a></li>
<li><a href='#reactome_descriptions'><p>Reactome Pathways - Descriptions</p></a></li>
<li><a href='#reactome_genes'><p>Reactome Pathways - Gene Sets</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Data Package for 'pathfindR'</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ege Ulgen &lt;egeulgen@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>This is a data-only package, containing data needed to run the CRAN 
    package 'pathfindR', a package for enrichment analysis utilizing active 
    subnetworks. This package contains protein-protein interaction network data, 
    data related to gene sets and example input/output data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/egeulgen/pathfindR.data">https://github.com/egeulgen/pathfindR.data</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/egeulgen/pathfindR.data/issues">https://github.com/egeulgen/pathfindR.data/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-27 18:55:55 UTC; egeulgen</td>
</tr>
<tr>
<td>Author:</td>
<td>Ege Ulgen <a href="https://orcid.org/0000-0003-2090-3621"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre,
    cph],
  Ozan Ozisik <a href="https://orcid.org/0000-0001-5980-8002"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-27 22:50:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='biocarta_descriptions'>BioCarta Pathways - Descriptions</h2><span id='topic+biocarta_descriptions'></span>

<h3>Description</h3>

<p>A named vector containing the descriptions for each human BioCarta pathway.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biocarta_descriptions
</code></pre>


<h3>Format</h3>

<p>named vector containing 292 character values, the descriptions for the given
pathways.
</p>

<hr>
<h2 id='biocarta_genes'>BioCarta Pathways - Gene Sets</h2><span id='topic+biocarta_genes'></span>

<h3>Description</h3>

<p>A list containing the genes involved in each human BioCarta pathway. Each
element is a vector of gene symbols located in the given pathway.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biocarta_genes
</code></pre>


<h3>Format</h3>

<p>list containing 292 vectors of gene symbols. Each vector corresponds
to a gene set.
</p>

<hr>
<h2 id='cell_markers_descriptions'>Human Cell Markers - Descriptions</h2><span id='topic+cell_markers_descriptions'></span>

<h3>Description</h3>

<p>A named vector containing descriptions of different cell types from different
tissues in human. Names of the vectors are Cell Ontology IDs (if available)
of the cell types in the following format: &quot;tissue type, cancer type, cell name&quot;
For more information, refer to the article:
Hu C, Li T, Xu Y, Zhang X, Li F, Bai J, et al. CellMarker 2.0: an updated
database of manually curated cell markers in human/mouse and web tools based
on scRNA-seq data. Nucleic Acids Res. 2022 Oct 27;gkac947.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cell_markers_descriptions
</code></pre>


<h3>Format</h3>

<p>named vector containing 1986 character values, the descriptions for
the given human cell types.
</p>

<hr>
<h2 id='cell_markers_gsets'>Human Cell Markers - Gene Sets</h2><span id='topic+cell_markers_gsets'></span>

<h3>Description</h3>

<p>A list containing the sets of genes that are cell markers of different cell
types from different tissues in human. Each element is a vector of cell marker
gene symbols for the given cell type. Names correspond to the Cell Ontology ID
(if available) of the cell type. For more information, refer to the article:
Hu C, Li T, Xu Y, Zhang X, Li F, Bai J, et al. CellMarker 2.0: an updated
database of manually curated cell markers in human/mouse and web tools based
on scRNA-seq data. Nucleic Acids Res. 2022 Oct 27;gkac947.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cell_markers_gsets
</code></pre>


<h3>Format</h3>

<p>list containing 1986 vectors. Each vector corresponds to a cell marker
gene set for a given human cell type.
</p>

<hr>
<h2 id='example_active_snws'>Example Active Subnetworks</h2><span id='topic+example_active_snws'></span>

<h3>Description</h3>

<p>A list of vectors containing genes for each active subnetwork that passed
the filtering step.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_active_snws
</code></pre>


<h3>Format</h3>

<p>list containing 150 vectors. Each vector is the set of genes for the
given active subnetwork.
</p>

<hr>
<h2 id='example_comparison_output'>Second Example Output for the pathfindR Enrichment Workflow (H.sapiens. - Rheumatoid Arthritis data)</h2><span id='topic+example_comparison_output'></span>

<h3>Description</h3>

<p>The data frame containing the results of pathfindR's active-subnetwork-oriented
enrichment workflow performed on the rheumatoid arthritis dataset GSE84074
<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE84074">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE84074</a>. Analysis via
<code>run_pathfindR</code> was performed using the default settings.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_comparison_output
</code></pre>


<h3>Format</h3>

<p>A data frame with 38 rows and 9 columns:
</p>

<dl>
<dt>ID</dt><dd><p>ID of the enriched term</p>
</dd>
<dt>Term_Description</dt><dd><p>Description of the enriched term</p>
</dd>
<dt>Fold_Enrichment</dt><dd><p>Fold enrichment value for the enriched term</p>
</dd>
<dt>occurrence</dt><dd><p>the number of iterations that the given term was found to enriched over all iterations</p>
</dd>
<dt>support</dt><dd><p>the median support (proportion of active subnetworks leading to enrichment within an iteration) over all iterations</p>
</dd>
<dt>lowest_p</dt><dd><p>the lowest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>highest_p</dt><dd><p>the highest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>Up_regulated</dt><dd><p>the up-regulated genes in the input involved in the given term, comma-separated</p>
</dd>
<dt>Down_regulated</dt><dd><p>the down-regulated genes in the input involved in the given term, comma-separated</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+example_pathfindR_input">example_pathfindR_input</a></code>  for the RA differentially-expressed genes data frame
<code><a href="#topic+example_pathfindR_output">example_pathfindR_output</a></code> for the RA example pathfindR enrichment output
<code><a href="#topic+example_pathfindR_output_clustered">example_pathfindR_output_clustered</a></code> for the RA example pathfindR clustering output
<code><a href="#topic+example_experiment_matrix">example_experiment_matrix</a></code> for the RA differentially-expressed genes expression matrix
<code><a href="pathfindR.html#topic+run_pathfindR">run_pathfindR</a></code> for details on the pathfindR enrichment analysis
</p>

<hr>
<h2 id='example_custom_genesets_result'>Custom Gene Set Enrichment Results</h2><span id='topic+example_custom_genesets_result'></span>

<h3>Description</h3>

<p>A data frame consisting of pathfindR enrichment analysis results on the
example TF target genes data (target gene sets of CREB and MYC).
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_custom_genesets_result
</code></pre>


<h3>Format</h3>

<p>data frame containing 2 rows and 9 columns. Each row is a gene set (the TF target gene sets).
</p>

<hr>
<h2 id='example_experiment_matrix'>Example Experiment Matrix for pathfindR - Enriched Term Scoring</h2><span id='topic+example_experiment_matrix'></span>

<h3>Description</h3>

<p>A matrix containing the log<sub>2</sub>-transformed
and quantile-normalized expression values of the differentially-expressed
genes for 18 rheumatoid arthritis (RA) patients and 15 healthy subjects. The
matrix contains expression values of  572 significantly
differentially-expressed genes (see <code><a href="#topic+example_pathfindR_input">example_pathfindR_input</a></code>) with adj.P.Val &lt;= 0.05.
<em>Generated on 28 Sep 2019.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_experiment_matrix
</code></pre>


<h3>Format</h3>

<p>A matrix with 572 rows and 33 columns.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE15573">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE15573</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+example_pathfindR_input">example_pathfindR_input</a></code> for the RA differentially-expressed genes data frame
<code><a href="#topic+example_pathfindR_output">example_pathfindR_output</a></code> for the RA example pathfindR enrichment output
<code><a href="pathfindR.html#topic+score_terms">score_terms</a></code> for details on calculating agglomerated scores of enriched terms
</p>

<hr>
<h2 id='example_mmu_input'>Example Input for Mus musculus - Myeloma Analysis</h2><span id='topic+example_mmu_input'></span>

<h3>Description</h3>

<p>A dataset containing the differentially-expressed genes and adjusted p-values
for the GEO dataset GSE99393. The RNA microarray experiment was perform to
detail the global program of gene expression underlying polarization of
myeloma-associated macrophages by CSF1R antibody treatment. The samples were
6 murine bone marrow derived macrophages co-cultured with myeloma cells
(myeloma-associated  macrophages), 3 of which were treated with CSF1R
antibody (treatment group) and the rest were treated with control IgG
antibody (control group). In this dataset, differentially-expressed genes
with |logFC| &gt;= 2 and FDR &lt; 0.05 are presented.
<em>Generated on 1 Nov 2019.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_mmu_input
</code></pre>


<h3>Format</h3>

<p>A data frame with 45 rows and 2 variables: </p>

<dl>
<dt>Gene_Symbol</dt><dd><p>MGI gene symbols of the differentially-expressed genes</p>
</dd>
<dt>FDR</dt><dd><p>adjusted p values, via the Benjamini &amp; Hochberg (1995) method</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE99393">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE99393</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+example_mmu_output">example_mmu_output</a></code> for the example mmu enrichment output.
<code><a href="pathfindR.html#topic+run_pathfindR">run_pathfindR</a></code> for details on the pathfindR enrichment analysis.
</p>

<hr>
<h2 id='example_mmu_output'>Example Output for Mus musculus - Myeloma Analysis</h2><span id='topic+example_mmu_output'></span>

<h3>Description</h3>

<p>A dataset containing the results of pathfindR's active-subnetwork-oriented
enrichment workflow performed on the Mus musculus myeloma
differential expression dataset <code><a href="#topic+example_mmu_input">example_mmu_input</a></code>.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_mmu_output
</code></pre>


<h3>Format</h3>

<p>A data frame with 34 rows and 9 columns:
</p>

<dl>
<dt>ID</dt><dd><p>ID of the enriched term</p>
</dd>
<dt>Term_Description</dt><dd><p>Description of the enriched term</p>
</dd>
<dt>Fold_Enrichment</dt><dd><p>Fold enrichment value for the enriched term</p>
</dd>
<dt>occurrence</dt><dd><p>the number of iterations that the given term was found to enriched over all iterations</p>
</dd>
<dt>support</dt><dd><p>the median support (proportion of active subnetworks leading to enrichment within an iteration) over all iterations</p>
</dd>
<dt>lowest_p</dt><dd><p>the lowest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>highest_p</dt><dd><p>the highest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>Up_regulated</dt><dd><p>the up-regulated genes in the input involved in the given term, comma-separated</p>
</dd>
<dt>Down_regulated</dt><dd><p>the down-regulated genes in the input involved in the given term, comma-separated</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+example_mmu_input">example_mmu_input</a></code> for the example mmu input.
<code><a href="pathfindR.html#topic+run_pathfindR">run_pathfindR</a></code> for details on the pathfindR enrichment workflow.
</p>

<hr>
<h2 id='example_pathfindR_input'>Example Input for the pathfindR Enrichment Workflow - Rheumatoid Arthritis (H.sapiens)</h2><span id='topic+example_pathfindR_input'></span>

<h3>Description</h3>

<p>A dataset containing the differentially-expressed genes along with the
associated log<sub>2</sub>(fold-change)
values and FDR adjusted p-values for the GEO dataset GSE15573. This microarray
dataset aimed to characterize gene expression profiles in the peripheral
blood mononuclear cells of 18 rheumatoid arthritis (RA) patients versus 15
healthy subjects. Differentially-expressed genes with adj.P.Val &lt; 0.05 are
presented in this data frame.
<em>Generated on 1 Nov 2019.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_pathfindR_input
</code></pre>


<h3>Format</h3>

<p>A data frame with 572 rows and 3 variables: </p>

<dl>
<dt>Gene.symbol</dt><dd><p>HGNC gene symbols of the differentially-expressed genes</p>
</dd>
<dt>logFC</dt><dd>log<sub>2</sub><p>(fold-change) values</p>
</dd>
<dt>adj.P.Val</dt><dd><p>adjusted p values, via the Benjamini &amp; Hochberg (1995) method</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE15573">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE15573</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+example_pathfindR_output">example_pathfindR_output</a></code> for the RA example pathfindR enrichment output
<code><a href="#topic+example_pathfindR_output_clustered">example_pathfindR_output_clustered</a></code> for the RA example pathfindR clustering output
<code><a href="#topic+example_experiment_matrix">example_experiment_matrix</a></code> for the RA differentially-expressed genes expression matrix
<code><a href="pathfindR.html#topic+run_pathfindR">run_pathfindR</a></code> for details on the pathfindR enrichment analysis
</p>

<hr>
<h2 id='example_pathfindR_output'>Example Output for the pathfindR Enrichment Workflow - Rheumatoid Arthritis</h2><span id='topic+example_pathfindR_output'></span>

<h3>Description</h3>

<p>The data frame containing the results of pathfindR's active-subnetwork-oriented
enrichment workflow performed on the rheumatoid arthritis
differential-expression data frame <code><a href="#topic+example_pathfindR_input">example_pathfindR_input</a></code>. Analysis via
<code>run_pathfindR</code> was performed using the default settings.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_pathfindR_output
</code></pre>


<h3>Format</h3>

<p>A data frame with 121 rows and 9 columns:
</p>

<dl>
<dt>ID</dt><dd><p>ID of the enriched term</p>
</dd>
<dt>Term_Description</dt><dd><p>Description of the enriched term</p>
</dd>
<dt>Fold_Enrichment</dt><dd><p>Fold enrichment value for the enriched term</p>
</dd>
<dt>occurrence</dt><dd><p>the number of iterations that the given term was found to enriched over all iterations</p>
</dd>
<dt>support</dt><dd><p>the median support (proportion of active subnetworks leading to enrichment within an iteration) over all iterations</p>
</dd>
<dt>lowest_p</dt><dd><p>the lowest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>highest_p</dt><dd><p>the highest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>Up_regulated</dt><dd><p>the up-regulated genes in the input involved in the given term, comma-separated</p>
</dd>
<dt>Down_regulated</dt><dd><p>the down-regulated genes in the input involved in the given term, comma-separated</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+example_pathfindR_input">example_pathfindR_input</a></code>  for the RA differentially-expressed genes data frame
<code><a href="#topic+example_pathfindR_output_clustered">example_pathfindR_output_clustered</a></code> for the RA example pathfindR clustering outputs
<code><a href="#topic+example_experiment_matrix">example_experiment_matrix</a></code> for the RA differentially-expressed genes expression matrix
<code><a href="pathfindR.html#topic+run_pathfindR">run_pathfindR</a></code> for details on the pathfindR enrichment analysis
</p>

<hr>
<h2 id='example_pathfindR_output_clustered'>Example Output for the pathfindR Clustering Workflow - Rheumatoid Arthritis</h2><span id='topic+example_pathfindR_output_clustered'></span>

<h3>Description</h3>

<p>A dataset containing the results of pathfindR's clustering and
partitioning workflow performed on the rheumatoid arthritis
enrichment results <code><a href="#topic+example_pathfindR_output">example_pathfindR_output</a></code>. The clustering and partitioning
function <code><a href="pathfindR.html#topic+cluster_enriched_terms">cluster_enriched_terms</a></code> was used with the default settings
(i.e. hierarchical clustering was performed and the agglomeration method
was &quot;average&quot;).
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_pathfindR_output_clustered
</code></pre>


<h3>Format</h3>

<p>A data frame with 121 rows and 11 columns:
</p>

<dl>
<dt>ID</dt><dd><p>ID of the enriched term</p>
</dd>
<dt>Term_Description</dt><dd><p>Description of the enriched term</p>
</dd>
<dt>Fold_Enrichment</dt><dd><p>Fold enrichment value for the enriched term</p>
</dd>
<dt>occurrence</dt><dd><p>the number of iterations that the given term was found to enriched over all iterations</p>
</dd>
<dt>support</dt><dd><p>the median support (proportion of active subnetworks leading to enrichment within an iteration) over all iterations</p>
</dd></dl>
<p>s
</p>
<dl>
<dt>lowest_p</dt><dd><p>the lowest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>highest_p</dt><dd><p>the highest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>Up_regulated</dt><dd><p>the up-regulated genes in the input involved in the given term, comma-separated</p>
</dd>
<dt>Down_regulated</dt><dd><p>the down-regulated genes in the input involved in the given term, comma-separated</p>
</dd>
<dt>Cluster</dt><dd><p>the cluster to which the enriched term is assigned</p>
</dd>
<dt>Status</dt><dd><p>whether the enriched term is the &quot;Representative&quot; term in its cluster or only a &quot;Member&quot;</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+example_pathfindR_input">example_pathfindR_input</a></code>  for the RA differentially-expressed genes data frame
<code><a href="#topic+example_experiment_matrix">example_experiment_matrix</a></code> for the RA differentially-expressed genes expression matrix
<code><a href="pathfindR.html#topic+run_pathfindR">run_pathfindR</a></code> for details on the pathfindR enrichment analysis
<code><a href="#topic+example_pathfindR_output">example_pathfindR_output</a></code> for the RA example pathfindR enrichment output
<code><a href="pathfindR.html#topic+cluster_enriched_terms">cluster_enriched_terms</a></code> for details on clustering methods
</p>

<hr>
<h2 id='go_all_genes'>Gene Ontology - All Gene Ontology Gene Sets</h2><span id='topic+go_all_genes'></span>

<h3>Description</h3>

<p>A list containing the genes involved in each GO ontology term. Each
element is a vector of gene symbols located in the given gene set.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>go_all_genes
</code></pre>


<h3>Format</h3>

<p>list containing 15450 vectors of gene symbols. Each vector corresponds
to a GO gene set.
</p>

<hr>
<h2 id='kegg_descriptions'>KEGG Pathways - Descriptions</h2><span id='topic+kegg_descriptions'></span>

<h3>Description</h3>

<p>A named vector containing the descriptions for each Homo sapiens KEGG pathway. Names of the
vector correspond to the KEGG ID of the pathway. Pathways that did not contain
any genes were discarded.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kegg_descriptions
</code></pre>


<h3>Format</h3>

<p>named vector containing 358 character values, the descriptions for the given
pathways.
</p>

<hr>
<h2 id='kegg_genes'>KEGG Pathways - Gene Sets</h2><span id='topic+kegg_genes'></span>

<h3>Description</h3>

<p>A list containing the genes involved in each Homo sapiens KEGG pathway. Each element
is a vector of gene symbols located in the given pathway. Names correspond to
the KEGG ID of the pathway. Pathways that did not contain any genes were
discarded.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kegg_genes
</code></pre>


<h3>Format</h3>

<p>list containing 358 vectors of gene symbols. Each vector corresponds
to a pathway.
</p>

<hr>
<h2 id='mmu_kegg_descriptions'>Mus Musculus KEGG Pathways - Descriptions</h2><span id='topic+mmu_kegg_descriptions'></span>

<h3>Description</h3>

<p>A named vector containing the descriptions for each Mus musculus KEGG pathway. Names of the
vector correspond to the KEGG ID of the pathway. Pathways that did not contain
any genes were discarded.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmu_kegg_descriptions
</code></pre>


<h3>Format</h3>

<p>named vector containing 355 character values, the descriptions for the given
pathways.
</p>

<hr>
<h2 id='mmu_kegg_genes'>Mus Musculus KEGG Pathways - Gene Sets</h2><span id='topic+mmu_kegg_genes'></span>

<h3>Description</h3>

<p>A list containing the genes involved in each Mus musculus KEGG pathway. Each element
is a vector of gene symbols located in the given pathway. Names correspond to
the KEGG ID of the pathway. Pathways that did not contain any genes were
discarded.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmu_kegg_genes
</code></pre>


<h3>Format</h3>

<p>list containing 355 vectors of gene symbols. Each vector corresponds
to a pathway.
</p>

<hr>
<h2 id='pathfindR.data_updates'>Table of Data for pathfindR</h2><span id='topic+pathfindR.data_updates'></span>

<h3>Description</h3>

<p>Data frame containing all the data for pathfindR along with descriptions and
last update dates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathfindR.data_updates
</code></pre>


<h3>Format</h3>

<p>A data frame with 30 rows and 6 columns:
</p>

<dl>
<dt>Category</dt><dd><p>Category of the data</p>
</dd>
<dt>Name</dt><dd><p>Name of the data</p>
</dd>
<dt>Description</dt><dd><p>Description of the data</p>
</dd>
<dt>Source</dt><dd><p>Source of the data</p>
</dd>
<dt>Version</dt><dd><p>Version of the data (if applicable)</p>
</dd>
<dt>Last Update</dt><dd><p>Last update date</p>
</dd>
</dl>


<hr>
<h2 id='reactome_descriptions'>Reactome Pathways - Descriptions</h2><span id='topic+reactome_descriptions'></span>

<h3>Description</h3>

<p>A named vector containing the descriptions for each human Reactome pathway. Names of the
vector correspond to the Reactome ID of the pathway.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reactome_descriptions
</code></pre>


<h3>Format</h3>

<p>named vector containing 2681 character values, the descriptions for the given
pathways.
</p>

<hr>
<h2 id='reactome_genes'>Reactome Pathways - Gene Sets</h2><span id='topic+reactome_genes'></span>

<h3>Description</h3>

<p>A list containing the genes involved in each human Reactome pathway. Each
element is a vector of gene symbols located in the given pathway. Names
correspond to the Reactome ID of the pathway.
<em>Generated on 27 Apr 2024.</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reactome_genes
</code></pre>


<h3>Format</h3>

<p>list containing 2681 vectors of gene symbols. Each vector corresponds
to a pathway.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
