<!DOCTYPE html><html><head><title>Help for package pathfindR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pathfindR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#pathfindR'><p>pathfindR: A package for Enrichment Analysis Utilizing Active Subnetworks</p></a></li>
<li><a href='#active_snw_enrichment_wrapper'><p>Wrapper for Active Subnetwork Search + Enrichment over Single/Multiple Iteration(s)</p></a></li>
<li><a href='#active_snw_search'><p>Perform Active Subnetwork Search</p></a></li>
<li><a href='#annotate_term_genes'><p>Annotate the Affected Genes in the Provided Enriched Terms</p></a></li>
<li><a href='#check_java_version'><p>Check Java Version</p></a></li>
<li><a href='#cluster_enriched_terms'><p>Cluster Enriched Terms</p></a></li>
<li><a href='#cluster_graph_vis'><p>Graph Visualization of Clustered Enriched Terms</p></a></li>
<li><a href='#color_kegg_pathway'><p>Color hsa KEGG pathway</p></a></li>
<li><a href='#combine_pathfindR_results'><p>Combine 2 pathfindR Results</p></a></li>
<li><a href='#combined_results_graph'><p>Combined Results Graph</p></a></li>
<li><a href='#configure_output_dir'><p>Configure Output Directory Name</p></a></li>
<li><a href='#create_HTML_report'><p>Create HTML Report of pathfindR Results</p></a></li>
<li><a href='#create_kappa_matrix'><p>Create Kappa Statistics Matrix</p></a></li>
<li><a href='#enrichment'><p>Perform Enrichment Analysis for a Single Gene Set</p></a></li>
<li><a href='#enrichment_analyses'><p>Perform Enrichment Analyses on the Input Subnetworks</p></a></li>
<li><a href='#enrichment_chart'><p>Create Bubble Chart of Enrichment Results</p></a></li>
<li><a href='#fetch_gene_set'><p>Fetch Gene Set Objects</p></a></li>
<li><a href='#fetch_java_version'><p>Obtain Java Version</p></a></li>
<li><a href='#filterActiveSnws'><p>Parse Active Subnetwork Search Output File and Filter the Subnetworks</p></a></li>
<li><a href='#fuzzy_term_clustering'><p>Heuristic Fuzzy Multiple-linkage Partitioning of Enriched Terms</p></a></li>
<li><a href='#get_biogrid_pin'><p>Retrieve the Requested Release of Organism-specific BioGRID PIN</p></a></li>
<li><a href='#get_gene_sets_list'><p>Retrieve Organism-specific Gene Sets List</p></a></li>
<li><a href='#get_kegg_gsets'><p>Retrieve Organism-specific KEGG Pathway Gene Sets</p></a></li>
<li><a href='#get_mgsigdb_gsets'><p>Retrieve Organism-specific MSigDB Gene Sets</p></a></li>
<li><a href='#get_pin_file'><p>Retrieve Organism-specific PIN data</p></a></li>
<li><a href='#get_reactome_gsets'><p>Retrieve Reactome Pathway Gene Sets</p></a></li>
<li><a href='#gset_list_from_gmt'><p>Retrieve Gene Sets from GMT-format File</p></a></li>
<li><a href='#hierarchical_term_clustering'><p>Hierarchical Clustering of Enriched Terms</p></a></li>
<li><a href='#hyperg_test'><p>Hypergeometric Distribution-based Hypothesis Testing</p></a></li>
<li><a href='#input_processing'><p>Process Input</p></a></li>
<li><a href='#input_testing'><p>Input Testing</p></a></li>
<li><a href='#isColor'><p>Check if value is a valid color</p></a></li>
<li><a href='#plot_scores'><p>Plot the Heatmap of Score Matrix of Enriched Terms per Sample</p></a></li>
<li><a href='#process_pin'><p>Process Data frame of Protein-protein Interactions</p></a></li>
<li><a href='#return_pin_path'><p>Return The Path to Given Protein-Protein Interaction Network (PIN)</p></a></li>
<li><a href='#run_pathfindR'><p>Wrapper Function for pathfindR - Active-Subnetwork-Oriented Enrichment Workflow</p></a></li>
<li><a href='#score_terms'><p>Calculate Agglomerated Scores of Enriched Terms for Each Subject</p></a></li>
<li><a href='#single_iter_wrapper'><p>Active Subnetwork Search + Enrichment Analysis Wrapper for a Single Iteration</p></a></li>
<li><a href='#summarize_enrichment_results'><p>Summarize Enrichment Results</p></a></li>
<li><a href='#term_gene_graph'><p>Create Term-Gene Graph</p></a></li>
<li><a href='#term_gene_heatmap'><p>Create Terms by Genes Heatmap</p></a></li>
<li><a href='#UpSet_plot'><p>Create UpSet Plot of Enriched Terms</p></a></li>
<li><a href='#visualize_active_subnetworks'><p>Visualize Active Subnetworks</p></a></li>
<li><a href='#visualize_KEGG_diagram'><p>Visualize Human KEGG Pathways</p></a></li>
<li><a href='#visualize_term_interactions'><p>Visualize Interactions of Genes Involved in the Given Enriched Terms</p></a></li>
<li><a href='#visualize_terms'><p>Create Diagrams for Enriched Terms</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Enrichment Analysis Utilizing Active Subnetworks</td>
</tr>
<tr>
<td>Version:</td>
<td>2.4.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ege Ulgen &lt;egeulgen@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Enrichment analysis enables researchers to uncover mechanisms 
    underlying a phenotype. However, conventional methods for enrichment 
    analysis do not take into account protein-protein interaction information, 
    resulting in incomplete conclusions. pathfindR is a tool for enrichment 
    analysis utilizing active subnetworks. The main function identifies active 
    subnetworks in a protein-protein interaction network using a user-provided 
    list of genes and associated p values. It then performs enrichment analyses 
    on the identified subnetworks, identifying enriched terms (i.e. pathways or, 
    more broadly, gene sets) that possibly underlie the phenotype of interest.
    pathfindR also offers functionalities to cluster the enriched terms and 
    identify representative terms in each cluster, to score the enriched terms 
    per sample and to visualize analysis results. The enrichment, clustering and 
    other methods implemented in pathfindR are described in detail in 
    Ulgen E, Ozisik O, Sezerman OU. 2019. pathfindR: An R Package for 
    Comprehensive Identification of Enriched Pathways in Omics Data Through 
    Active Subnetworks. Front. Genet. &lt;<a href="https://doi.org/10.3389%2Ffgene.2019.00858">doi:10.3389/fgene.2019.00858</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://egeulgen.github.io/pathfindR/">https://egeulgen.github.io/pathfindR/</a>,
<a href="https://github.com/egeulgen/pathfindR">https://github.com/egeulgen/pathfindR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/egeulgen/pathfindR/issues">https://github.com/egeulgen/pathfindR/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>Java (&gt;= 8.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>DBI, AnnotationDbi, doParallel, foreach, rmarkdown,
org.Hs.eg.db, ggplot2, ggraph, ggupset, fpc, ggkegg, grDevices,
httr, igraph, R.utils, msigdbr, knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0), pathfindR.data (&ge; 2.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 2.3.2), covr, mockery</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-04 13:25:40 UTC; egeulgen</td>
</tr>
<tr>
<td>Author:</td>
<td>Ege Ulgen <a href="https://orcid.org/0000-0003-2090-3621"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre,
    cph],
  Ozan Ozisik <a href="https://orcid.org/0000-0001-5980-8002"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-04 15:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='pathfindR'>pathfindR: A package for Enrichment Analysis Utilizing Active Subnetworks</h2><span id='topic+pathfindR-package'></span><span id='topic+pathfindR'></span>

<h3>Description</h3>

<p>pathfindR is a tool for active-subnetwork-oriented gene set enrichment analysis.
The main aim of the package is to identify active subnetworks in a
protein-protein interaction network using a user-provided list of genes
and associated p values then performing enrichment analyses on the identified
subnetworks, discovering enriched terms (i.e. pathways, gene ontology, TF target
gene sets etc.) that possibly underlie the phenotype of interest.
</p>


<h3>Details</h3>

<p>For analysis on non-Homo sapiens organisms, pathfindR offers utility functions
for obtaining organism-specific PIN data and organism-specific gene sets data.
</p>
<p>pathfindR also offers functionalities to cluster the enriched terms and
identify representative terms in each cluster, to score the enriched terms
per sample and to visualize analysis results.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Ege Ulgen <a href="mailto:egeulgen@gmail.com">egeulgen@gmail.com</a> (<a href="https://orcid.org/0000-0003-2090-3621">ORCID</a>) [copyright holder]
</p>
<p>Authors:
</p>

<ul>
<li><p> Ozan Ozisik <a href="mailto:ozanytu@gmail.com">ozanytu@gmail.com</a> (<a href="https://orcid.org/0000-0001-5980-8002">ORCID</a>)
</p>
</li></ul>



<h3>See Also</h3>

<p>See <code><a href="#topic+run_pathfindR">run_pathfindR</a></code> for details on the pathfindR
active-subnetwork-oriented enrichment analysis
See <code><a href="#topic+cluster_enriched_terms">cluster_enriched_terms</a></code> for details on methods of enriched
terms clustering to define clusters of biologically-related terms
See <code><a href="#topic+score_terms">score_terms</a></code> for details on agglomerated score calculation
for enriched terms to investigate how a gene set is altered in a given sample
(or in cases vs. controls)
See <code><a href="#topic+term_gene_heatmap">term_gene_heatmap</a></code> for details on visualization of the heatmap
of enriched terms by involved genes
See <code><a href="#topic+term_gene_graph">term_gene_graph</a></code> for details on visualizing terms and
term-related genes as a graph to determine the degree of overlap between the
enriched terms by identifying shared and/or distinct significant genes
See <code><a href="#topic+UpSet_plot">UpSet_plot</a></code> for details on creating an UpSet plot of the
enriched terms.
See <code><a href="#topic+get_pin_file">get_pin_file</a></code> for obtaining organism-specific PIN data and
<code><a href="#topic+get_gene_sets_list">get_gene_sets_list</a></code> for obtaining organism-specific gene sets data
</p>

<hr>
<h2 id='active_snw_enrichment_wrapper'>Wrapper for Active Subnetwork Search + Enrichment over Single/Multiple Iteration(s)</h2><span id='topic+active_snw_enrichment_wrapper'></span>

<h3>Description</h3>

<p>Wrapper for Active Subnetwork Search + Enrichment over Single/Multiple Iteration(s)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>active_snw_enrichment_wrapper(
  input_processed,
  pin_path,
  gset_list,
  enrichment_threshold,
  list_active_snw_genes,
  adj_method = "bonferroni",
  search_method = "GR",
  disable_parallel = FALSE,
  use_all_positives = FALSE,
  iterations = 10,
  n_processes = NULL,
  score_quan_thr = 0.8,
  sig_gene_thr = 0.02,
  saTemp0 = 1,
  saTemp1 = 0.01,
  saIter = 10000,
  gaPop = 400,
  gaIter = 200,
  gaThread = 5,
  gaCrossover = 1,
  gaMut = 0,
  grMaxDepth = 1,
  grSearchDepth = 1,
  grOverlap = 0.5,
  grSubNum = 1000,
  silent_option = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_input_processed">input_processed</code></td>
<td>
<p>processed input data frame</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_pin_path">pin_path</code></td>
<td>
<p>path/to/PIN/file</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_gset_list">gset_list</code></td>
<td>
<p>list for gene sets</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_enrichment_threshold">enrichment_threshold</code></td>
<td>
<p>adjusted-p value threshold used when filtering
enrichment results (default = 0.05)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_list_active_snw_genes">list_active_snw_genes</code></td>
<td>
<p>boolean value indicating whether or not to report
the non-significant active subnetwork genes for the active subnetwork which was enriched for
the given term with the lowest p value (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_adj_method">adj_method</code></td>
<td>
<p>correction method to be used for adjusting p-values.
(default = 'bonferroni')</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_search_method">search_method</code></td>
<td>
<p>algorithm to use when performing active subnetwork
search. Options are greedy search (GR), simulated annealing (SA) or genetic
algorithm (GA) for the search (default = 'GR').</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_disable_parallel">disable_parallel</code></td>
<td>
<p>boolean to indicate whether to disable parallel runs
via <code>foreach</code> (default = FALSE)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_use_all_positives">use_all_positives</code></td>
<td>
<p>if TRUE: in GA, adds an individual with all positive
nodes. In SA, initializes candidate solution with all positive nodes. (default = FALSE)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_iterations">iterations</code></td>
<td>
<p>number of iterations for active subnetwork search and
enrichment analyses (Default = 10)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_n_processes">n_processes</code></td>
<td>
<p>optional argument for specifying the number of processes
used by foreach. If not specified, the function determines this
automatically (Default == NULL. Gets set to 1 for Genetic Algorithm)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_score_quan_thr">score_quan_thr</code></td>
<td>
<p>active subnetwork score quantile threshold. Must be
between 0 and 1 or set to -1 for not filtering. (Default = 0.8)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_sig_gene_thr">sig_gene_thr</code></td>
<td>
<p>threshold for the minimum proportion of significant genes in
the subnetwork (Default = 0.02) If the number of genes to use as threshold is
calculated to be &lt; 2 (e.g. 50 signif. genes x 0.01 = 0.5), the threshold number
is set to 2</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_satemp0">saTemp0</code></td>
<td>
<p>Initial temperature for SA (default = 1.0)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_satemp1">saTemp1</code></td>
<td>
<p>Final temperature for SA (default = 0.01)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_saiter">saIter</code></td>
<td>
<p>Iteration number for SA (default = 10000)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_gapop">gaPop</code></td>
<td>
<p>Population size for GA (default = 400)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_gaiter">gaIter</code></td>
<td>
<p>Iteration number for GA (default = 200)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_gathread">gaThread</code></td>
<td>
<p>Number of threads to be used in GA (default = 5)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_gacrossover">gaCrossover</code></td>
<td>
<p>Applies crossover with the given probability in GA (default = 1, i.e. always perform crossover)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_gamut">gaMut</code></td>
<td>
<p>For GA, applies mutation with given mutation rate (default = 0, i.e. mutation off)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_grmaxdepth">grMaxDepth</code></td>
<td>
<p>Sets max depth in greedy search, 0 for no limit (default = 1)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_grsearchdepth">grSearchDepth</code></td>
<td>
<p>Search depth in greedy search (default = 1)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_groverlap">grOverlap</code></td>
<td>
<p>Overlap threshold for results of greedy search (default = 0.5)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_grsubnum">grSubNum</code></td>
<td>
<p>Number of subnetworks to be presented in the results (default = 1000)</p>
</td></tr>
<tr><td><code id="active_snw_enrichment_wrapper_+3A_silent_option">silent_option</code></td>
<td>
<p>boolean value indicating whether to print the messages
to the console (FALSE) or not (TRUE, this will print to a temp. file) during
active subnetwork search (default = TRUE). This option was added because
during parallel runs, the console messages get disorderly printed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame of combined pathfindR enrichment results
</p>

<hr>
<h2 id='active_snw_search'>Perform Active Subnetwork Search</h2><span id='topic+active_snw_search'></span>

<h3>Description</h3>

<p>Perform Active Subnetwork Search
</p>


<h3>Usage</h3>

<pre><code class='language-R'>active_snw_search(
  input_for_search,
  pin_name_path = "Biogrid",
  snws_file = "active_snws",
  dir_for_parallel_run = NULL,
  score_quan_thr = 0.8,
  sig_gene_thr = 0.02,
  search_method = "GR",
  seedForRandom = 1234,
  silent_option = TRUE,
  use_all_positives = FALSE,
  geneInitProbs = 0.1,
  saTemp0 = 1,
  saTemp1 = 0.01,
  saIter = 10000,
  gaPop = 400,
  gaIter = 10000,
  gaThread = 5,
  gaCrossover = 1,
  gaMut = 0,
  grMaxDepth = 1,
  grSearchDepth = 1,
  grOverlap = 0.5,
  grSubNum = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="active_snw_search_+3A_input_for_search">input_for_search</code></td>
<td>
<p>input the input data that active subnetwork search uses. The input
must be a data frame containing at least these 2 columns: </p>

<dl>
<dt>GENE</dt><dd><p>Gene Symbol</p>
</dd>
<dt>P_VALUE</dt><dd><p>p value obtained through a test, e.g. differential expression/methylation</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="active_snw_search_+3A_pin_name_path">pin_name_path</code></td>
<td>
<p>Name of the chosen PIN or absolute/path/to/PIN.sif. If PIN name,
must be one of c('Biogrid', 'STRING', 'GeneMania', 'IntAct', 'KEGG', 'mmu_STRING'). If
path/to/PIN.sif, the file must comply with the PIN specifications. (Default = 'Biogrid')</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_snws_file">snws_file</code></td>
<td>
<p>name for active subnetwork search output data
<strong>without file extension</strong> (default = 'active_snws')</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_dir_for_parallel_run">dir_for_parallel_run</code></td>
<td>
<p>(previously created) directory for a parallel run iteration.
Used in the wrapper function (see ?run_pathfindR) (Default = NULL)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_score_quan_thr">score_quan_thr</code></td>
<td>
<p>active subnetwork score quantile threshold. Must be
between 0 and 1 or set to -1 for not filtering. (Default = 0.8)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_sig_gene_thr">sig_gene_thr</code></td>
<td>
<p>threshold for the minimum proportion of significant genes in
the subnetwork (Default = 0.02) If the number of genes to use as threshold is
calculated to be &lt; 2 (e.g. 50 signif. genes x 0.01 = 0.5), the threshold number
is set to 2</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_search_method">search_method</code></td>
<td>
<p>algorithm to use when performing active subnetwork
search. Options are greedy search (GR), simulated annealing (SA) or genetic
algorithm (GA) for the search (default = 'GR').</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_seedforrandom">seedForRandom</code></td>
<td>
<p>seed for reproducibility while running the java modules (applies for GR and SA)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_silent_option">silent_option</code></td>
<td>
<p>boolean value indicating whether to print the messages
to the console (FALSE) or not (TRUE, this will print to a temp. file) during
active subnetwork search (default = TRUE). This option was added because
during parallel runs, the console messages get disorderly printed.</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_use_all_positives">use_all_positives</code></td>
<td>
<p>if TRUE: in GA, adds an individual with all positive
nodes. In SA, initializes candidate solution with all positive nodes. (default = FALSE)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_geneinitprobs">geneInitProbs</code></td>
<td>
<p>For SA and GA, probability of adding a gene in initial solution (default = 0.1)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_satemp0">saTemp0</code></td>
<td>
<p>Initial temperature for SA (default = 1.0)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_satemp1">saTemp1</code></td>
<td>
<p>Final temperature for SA (default = 0.01)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_saiter">saIter</code></td>
<td>
<p>Iteration number for SA (default = 10000)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_gapop">gaPop</code></td>
<td>
<p>Population size for GA (default = 400)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_gaiter">gaIter</code></td>
<td>
<p>Iteration number for GA (default = 200)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_gathread">gaThread</code></td>
<td>
<p>Number of threads to be used in GA (default = 5)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_gacrossover">gaCrossover</code></td>
<td>
<p>Applies crossover with the given probability in GA (default = 1, i.e. always perform crossover)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_gamut">gaMut</code></td>
<td>
<p>For GA, applies mutation with given mutation rate (default = 0, i.e. mutation off)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_grmaxdepth">grMaxDepth</code></td>
<td>
<p>Sets max depth in greedy search, 0 for no limit (default = 1)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_grsearchdepth">grSearchDepth</code></td>
<td>
<p>Search depth in greedy search (default = 1)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_groverlap">grOverlap</code></td>
<td>
<p>Overlap threshold for results of greedy search (default = 0.5)</p>
</td></tr>
<tr><td><code id="active_snw_search_+3A_grsubnum">grSubNum</code></td>
<td>
<p>Number of subnetworks to be presented in the results (default = 1000)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of genes in every identified active subnetwork that has a score greater than
the 'score_quan_thr'th quantile and that has at least 'sig_gene_thr' affected genes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>processed_df &lt;- example_pathfindR_input[1:15, -2]
colnames(processed_df) &lt;- c('GENE', 'P_VALUE')
GR_snws &lt;- active_snw_search(
  input_for_search = processed_df,
  pin_name_path = 'KEGG',
  search_method = 'GR',
  score_quan_thr = 0.8
)
# clean-up
unlink('active_snw_search', recursive = TRUE)
</code></pre>

<hr>
<h2 id='annotate_term_genes'>Annotate the Affected Genes in the Provided Enriched Terms</h2><span id='topic+annotate_term_genes'></span>

<h3>Description</h3>

<p>Function to annotate the involved affected (input) genes in each term.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>annotate_term_genes(
  result_df,
  input_processed,
  genes_by_term = pathfindR.data::kegg_genes
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="annotate_term_genes_+3A_result_df">result_df</code></td>
<td>
<p>data frame of enrichment results.
The only must-have column is 'ID'.</p>
</td></tr>
<tr><td><code id="annotate_term_genes_+3A_input_processed">input_processed</code></td>
<td>
<p>input data processed via <code><a href="#topic+input_processing">input_processing</a></code></p>
</td></tr>
<tr><td><code id="annotate_term_genes_+3A_genes_by_term">genes_by_term</code></td>
<td>
<p>List that contains genes for each gene set. Names of
this list are gene set IDs (default = kegg_genes)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The original data frame with two additional columns:  </p>

<dl>
<dt>Up_regulated</dt><dd><p>the up-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>Down_regulated</dt><dd><p>the down-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>example_gene_data &lt;- example_pathfindR_input
colnames(example_gene_data) &lt;- c('GENE', 'CHANGE', 'P_VALUE')

annotated_result &lt;- annotate_term_genes(
  result_df = example_pathfindR_output,
  input_processed = example_gene_data
)
</code></pre>

<hr>
<h2 id='check_java_version'>Check Java Version</h2><span id='topic+check_java_version'></span>

<h3>Description</h3>

<p>Check Java Version
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_java_version(version = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_java_version_+3A_version">version</code></td>
<td>
<p>character vector containing the output of 'java -version'. If
NULL, result of <code><a href="#topic+fetch_java_version">fetch_java_version</a></code> is used (default = NULL)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>this function was adapted from the CRAN package <code>sparklyr</code>
</p>


<h3>Value</h3>

<p>only parses and checks whether the java version is &gt;= 1.8
</p>

<hr>
<h2 id='cluster_enriched_terms'>Cluster Enriched Terms</h2><span id='topic+cluster_enriched_terms'></span>

<h3>Description</h3>

<p>Cluster Enriched Terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluster_enriched_terms(
  enrichment_res,
  method = "hierarchical",
  plot_clusters_graph = TRUE,
  use_description = FALSE,
  use_active_snw_genes = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cluster_enriched_terms_+3A_enrichment_res">enrichment_res</code></td>
<td>
<p>data frame of pathfindR enrichment results. Must-have
columns are 'Term_Description' (if <code>use_description = TRUE</code>) or 'ID'
(if <code>use_description = FALSE</code>), 'Down_regulated', and 'Up_regulated'.
If <code>use_active_snw_genes = TRUE</code>, 'non_Signif_Snw_Genes' must also be
provided.</p>
</td></tr>
<tr><td><code id="cluster_enriched_terms_+3A_method">method</code></td>
<td>
<p>Either 'hierarchical' or 'fuzzy'. Details of clustering are
provided in the corresponding functions <code><a href="#topic+hierarchical_term_clustering">hierarchical_term_clustering</a></code>,
and <code><a href="#topic+fuzzy_term_clustering">fuzzy_term_clustering</a></code></p>
</td></tr>
<tr><td><code id="cluster_enriched_terms_+3A_plot_clusters_graph">plot_clusters_graph</code></td>
<td>
<p>boolean value indicate whether or not to plot
the graph diagram of clustering results (default = TRUE)</p>
</td></tr>
<tr><td><code id="cluster_enriched_terms_+3A_use_description">use_description</code></td>
<td>
<p>Boolean argument to indicate whether term descriptions
(in the 'Term_Description' column) should be used. (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="cluster_enriched_terms_+3A_use_active_snw_genes">use_active_snw_genes</code></td>
<td>
<p>boolean to indicate whether or not to use
non-input active subnetwork genes in the calculation of kappa statistics
(default = FALSE, i.e. only use affected genes)</p>
</td></tr>
<tr><td><code id="cluster_enriched_terms_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="#topic+hierarchical_term_clustering">hierarchical_term_clustering</a></code>,
<code><a href="#topic+fuzzy_term_clustering">fuzzy_term_clustering</a></code> and <code><a href="#topic+cluster_graph_vis">cluster_graph_vis</a></code>.
See documentation of these functions for more details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame of clustering results. For 'hierarchical', the cluster
assignments (Cluster) and whether the term is representative of its cluster
(Status) is added as columns. For 'fuzzy', terms that are in multiple
clusters are provided for each cluster. The cluster assignments (Cluster)
and whether the term is representative of its cluster (Status) is
added as columns.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+hierarchical_term_clustering">hierarchical_term_clustering</a></code> for hierarchical
clustering of enriched terms.
See <code><a href="#topic+fuzzy_term_clustering">fuzzy_term_clustering</a></code> for fuzzy clustering of enriched terms.
See <code><a href="#topic+cluster_graph_vis">cluster_graph_vis</a></code> for graph visualization of clustering.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_clustered &lt;- cluster_enriched_terms(
  example_pathfindR_output[1:3, ],
  plot_clusters_graph = FALSE
)
example_clustered &lt;- cluster_enriched_terms(
  example_pathfindR_output[1:3, ],
  method = 'fuzzy', plot_clusters_graph = FALSE
)
</code></pre>

<hr>
<h2 id='cluster_graph_vis'>Graph Visualization of Clustered Enriched Terms</h2><span id='topic+cluster_graph_vis'></span>

<h3>Description</h3>

<p>Graph Visualization of Clustered Enriched Terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluster_graph_vis(
  clu_obj,
  kappa_mat,
  enrichment_res,
  kappa_threshold = 0.35,
  use_description = FALSE,
  vertex.label.cex = 0.7,
  vertex.size.scaling = 2.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cluster_graph_vis_+3A_clu_obj">clu_obj</code></td>
<td>
<p>clustering result (either a matrix obtained via
<code><a href="#topic+hierarchical_term_clustering">hierarchical_term_clustering</a></code> or <code><a href="#topic+fuzzy_term_clustering">fuzzy_term_clustering</a></code>
'fuzzy_term_clustering' or a vector obtained via 'hierarchical_term_clustering')</p>
</td></tr>
<tr><td><code id="cluster_graph_vis_+3A_kappa_mat">kappa_mat</code></td>
<td>
<p>matrix of kappa statistics (output of <code><a href="#topic+create_kappa_matrix">create_kappa_matrix</a></code>)</p>
</td></tr>
<tr><td><code id="cluster_graph_vis_+3A_enrichment_res">enrichment_res</code></td>
<td>
<p>data frame of pathfindR enrichment results. Must-have
columns are 'Term_Description' (if <code>use_description = TRUE</code>) or 'ID'
(if <code>use_description = FALSE</code>), 'Down_regulated', and 'Up_regulated'.
If <code>use_active_snw_genes = TRUE</code>, 'non_Signif_Snw_Genes' must also be
provided.</p>
</td></tr>
<tr><td><code id="cluster_graph_vis_+3A_kappa_threshold">kappa_threshold</code></td>
<td>
<p>threshold for kappa statistics, defining strong
relation (default = 0.35)</p>
</td></tr>
<tr><td><code id="cluster_graph_vis_+3A_use_description">use_description</code></td>
<td>
<p>Boolean argument to indicate whether term descriptions
(in the 'Term_Description' column) should be used. (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="cluster_graph_vis_+3A_vertex.label.cex">vertex.label.cex</code></td>
<td>
<p>font size for vertex labels; it is interpreted as a multiplication factor of some device-dependent base font size (default = 0.7)</p>
</td></tr>
<tr><td><code id="cluster_graph_vis_+3A_vertex.size.scaling">vertex.size.scaling</code></td>
<td>
<p>scaling factor for the node size (default = 2.5)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plots a graph diagram of clustering results. Each node is an enriched term
from 'enrichment_res'. Size of node corresponds to -log(lowest_p). Thickness
of the edges between nodes correspond to the kappa statistic between the two
terms. Color of each node corresponds to distinct clusters. For fuzzy
clustering, if a term is in multiple clusters, multiple colors are utilized.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
cluster_graph_vis(clu_obj, kappa_mat, enrichment_res)

## End(Not run)
</code></pre>

<hr>
<h2 id='color_kegg_pathway'>Color hsa KEGG pathway</h2><span id='topic+color_kegg_pathway'></span>

<h3>Description</h3>

<p>Color hsa KEGG pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>color_kegg_pathway(
  pw_id,
  change_vec,
  scale_vals = TRUE,
  node_cols = NULL,
  legend.position = "top"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="color_kegg_pathway_+3A_pw_id">pw_id</code></td>
<td>
<p>hsa KEGG pathway id (e.g. hsa05012)</p>
</td></tr>
<tr><td><code id="color_kegg_pathway_+3A_change_vec">change_vec</code></td>
<td>
<p>vector of change values, names should be hsa KEGG gene ids</p>
</td></tr>
<tr><td><code id="color_kegg_pathway_+3A_scale_vals">scale_vals</code></td>
<td>
<p>should change values be scaled? (default = <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="color_kegg_pathway_+3A_node_cols">node_cols</code></td>
<td>
<p>low, middle and high color values for coloring the pathway nodes
(default = <code>NULL</code>). If <code>node_cols=NULL</code>, the low, middle and high color
are set as 'green', 'gray' and 'red'. If all change values are 1e6 (in case no
changes are supplied, this dummy value is assigned by
<code><a href="#topic+input_processing">input_processing</a></code>), only one color ('#F38F18' if NULL) is used.</p>
</td></tr>
<tr><td><code id="color_kegg_pathway_+3A_legend.position">legend.position</code></td>
<td>
<p>the default position of legends (&quot;none&quot;, &quot;left&quot;,
&quot;right&quot;, &quot;bottom&quot;, &quot;top&quot;, &quot;inside&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot object containing the colored KEGG pathway diagram visualization
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pw_id &lt;- 'hsa00010'
change_vec &lt;- c(-2, 4, 6)
names(change_vec) &lt;- c('hsa:2821', 'hsa:226', 'hsa:229')
result &lt;- pathfindR:::color_kegg_pathway(pw_id, change_vec)

## End(Not run)
</code></pre>

<hr>
<h2 id='combine_pathfindR_results'>Combine 2 pathfindR Results</h2><span id='topic+combine_pathfindR_results'></span>

<h3>Description</h3>

<p>Combine 2 pathfindR Results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_pathfindR_results(result_A, result_B, plot_common = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_pathfindR_results_+3A_result_a">result_A</code></td>
<td>
<p>data frame of first pathfindR enrichment results</p>
</td></tr>
<tr><td><code id="combine_pathfindR_results_+3A_result_b">result_B</code></td>
<td>
<p>data frame of second pathfindR enrichment results</p>
</td></tr>
<tr><td><code id="combine_pathfindR_results_+3A_plot_common">plot_common</code></td>
<td>
<p>boolean to indicate whether or not to plot the term-gene
graph of the common terms (default=<code>TRUE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame of combined pathfindR enrichment results. Columns are: </p>

<dl>
<dt>ID</dt><dd><p>ID of the enriched term</p>
</dd>
<dt>Term_Description</dt><dd><p>Description of the enriched term</p>
</dd>
<dt>Fold_Enrichment_A</dt><dd><p>Fold enrichment value for the enriched term (Calculated using ONLY the input genes)</p>
</dd>
<dt>occurrence_A</dt><dd><p>the number of iterations that the given term was found to enriched over all iterations</p>
</dd>
<dt>lowest_p_A</dt><dd><p>the lowest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>highest_p_A</dt><dd><p>the highest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>Up_regulated_A</dt><dd><p>the up-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>Down_regulated_A</dt><dd><p>the down-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>Fold_Enrichment_B</dt><dd><p>Fold enrichment value for the enriched term (Calculated using ONLY the input genes)</p>
</dd>
<dt>occurrence_B</dt><dd><p>the number of iterations that the given term was found to enriched over all iterations</p>
</dd>
<dt>lowest_p_B</dt><dd><p>the lowest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>highest_p_B</dt><dd><p>the highest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>Up_regulated_B</dt><dd><p>the up-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>Down_regulated_B</dt><dd><p>the down-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>combined_p</dt><dd><p>the combined p value (via Fisher's method)</p>
</dd>
<dt>status</dt><dd><p>whether the term is found in both analyses ('common'), found only in the first ('A only') or found only in the second ('B only)</p>
</dd>
</dl>

<p>By default, the function also displays the term-gene graph of the common terms
</p>


<h3>Examples</h3>

<pre><code class='language-R'>combined_results &lt;- combine_pathfindR_results(example_pathfindR_output, example_comparison_output)
</code></pre>

<hr>
<h2 id='combined_results_graph'>Combined Results Graph</h2><span id='topic+combined_results_graph'></span>

<h3>Description</h3>

<p>Combined Results Graph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combined_results_graph(
  combined_df,
  selected_terms = "common",
  use_description = FALSE,
  layout = "stress",
  node_size = "num_genes"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combined_results_graph_+3A_combined_df">combined_df</code></td>
<td>
<p>Data frame of combined pathfindR enrichment results</p>
</td></tr>
<tr><td><code id="combined_results_graph_+3A_selected_terms">selected_terms</code></td>
<td>
<p>the vector of selected terms for creating the graph
(either IDs or term descriptions). If set to <code>'common'</code>, all of the
common terms are used. (default = 'common')</p>
</td></tr>
<tr><td><code id="combined_results_graph_+3A_use_description">use_description</code></td>
<td>
<p>Boolean argument to indicate whether term descriptions
(in the 'Term_Description' column) should be used. (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="combined_results_graph_+3A_layout">layout</code></td>
<td>
<p>The type of layout to create (see <code><a href="ggraph.html#topic+ggraph">ggraph</a></code> for details. Default = 'stress')</p>
</td></tr>
<tr><td><code id="combined_results_graph_+3A_node_size">node_size</code></td>
<td>
<p>Argument to indicate whether to use number of significant genes ('num_genes')
or the -log10(lowest p value) ('p_val') for adjusting the node sizes (default = 'num_genes')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a  <code><a href="ggraph.html#topic+ggraph">ggraph</a></code> object containing the combined term-gene graph.
Each node corresponds to an enriched term (orange if common, different shades of blue otherwise),
an up-regulated gene (green), a down-regulated gene (red) or
a conflicting (i.e. up in one analysis, down in the other or vice versa) gene
(gray). An edge between a term and a gene indicates
that the given term involves the gene. Size of a term node is proportional
to either the number of genes (if <code>node_size = 'num_genes'</code>) or
the -log10(lowest p value) (if <code>node_size = 'p_val'</code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>combined_results &lt;- combine_pathfindR_results(
  example_pathfindR_output,
  example_comparison_output,
  plot_common = FALSE
)
g &lt;- combined_results_graph(combined_results, selected_terms = sample(combined_results$ID, 3))
</code></pre>

<hr>
<h2 id='configure_output_dir'>Configure Output Directory Name</h2><span id='topic+configure_output_dir'></span>

<h3>Description</h3>

<p>Configure Output Directory Name
</p>


<h3>Usage</h3>

<pre><code class='language-R'>configure_output_dir(output_dir = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="configure_output_dir_+3A_output_dir">output_dir</code></td>
<td>
<p>the directory to be created where the output and intermediate
files are saved (default = <code>NULL</code>, a temporary directory is used)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>/path/to/output/dir
</p>

<hr>
<h2 id='create_HTML_report'>Create HTML Report of pathfindR Results</h2><span id='topic+create_HTML_report'></span>

<h3>Description</h3>

<p>Create HTML Report of pathfindR Results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_HTML_report(input, input_processed, final_res, dir_for_report)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_HTML_report_+3A_input">input</code></td>
<td>
<p>the input data that pathfindR uses. The input must be a data
frame with three columns: </p>

<ol>
<li><p> Gene Symbol (Gene Symbol)
</p>
</li>
<li><p> Change value, e.g. log(fold change) (OPTIONAL)
</p>
</li>
<li><p> p value, e.g. adjusted p value associated with differential expression
</p>
</li></ol>
</td></tr>
<tr><td><code id="create_HTML_report_+3A_input_processed">input_processed</code></td>
<td>
<p>processed input data frame</p>
</td></tr>
<tr><td><code id="create_HTML_report_+3A_final_res">final_res</code></td>
<td>
<p>final pathfindR result data frame</p>
</td></tr>
<tr><td><code id="create_HTML_report_+3A_dir_for_report">dir_for_report</code></td>
<td>
<p>directory to render the report in</p>
</td></tr>
</table>

<hr>
<h2 id='create_kappa_matrix'>Create Kappa Statistics Matrix</h2><span id='topic+create_kappa_matrix'></span>

<h3>Description</h3>

<p>Create Kappa Statistics Matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_kappa_matrix(
  enrichment_res,
  use_description = FALSE,
  use_active_snw_genes = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_kappa_matrix_+3A_enrichment_res">enrichment_res</code></td>
<td>
<p>data frame of pathfindR enrichment results. Must-have
columns are 'Term_Description' (if <code>use_description = TRUE</code>) or 'ID'
(if <code>use_description = FALSE</code>), 'Down_regulated', and 'Up_regulated'.
If <code>use_active_snw_genes = TRUE</code>, 'non_Signif_Snw_Genes' must also be
provided.</p>
</td></tr>
<tr><td><code id="create_kappa_matrix_+3A_use_description">use_description</code></td>
<td>
<p>Boolean argument to indicate whether term descriptions
(in the 'Term_Description' column) should be used. (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="create_kappa_matrix_+3A_use_active_snw_genes">use_active_snw_genes</code></td>
<td>
<p>boolean to indicate whether or not to use
non-input active subnetwork genes in the calculation of kappa statistics
(default = FALSE, i.e. only use affected genes)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix of kappa statistics between each term in the
enrichment results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sub_df &lt;- example_pathfindR_output[1:3, ]
create_kappa_matrix(sub_df)
</code></pre>

<hr>
<h2 id='enrichment'>Perform Enrichment Analysis for a Single Gene Set</h2><span id='topic+enrichment'></span>

<h3>Description</h3>

<p>Perform Enrichment Analysis for a Single Gene Set
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enrichment(
  input_genes,
  genes_by_term = pathfindR.data::kegg_genes,
  term_descriptions = pathfindR.data::kegg_descriptions,
  adj_method = "bonferroni",
  enrichment_threshold = 0.05,
  sig_genes_vec,
  background_genes
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="enrichment_+3A_input_genes">input_genes</code></td>
<td>
<p>The set of gene symbols to be used for enrichment
analysis. In the scope of this package, these are genes that were
identified for an active subnetwork</p>
</td></tr>
<tr><td><code id="enrichment_+3A_genes_by_term">genes_by_term</code></td>
<td>
<p>List that contains genes for each gene set. Names of
this list are gene set IDs (default = kegg_genes)</p>
</td></tr>
<tr><td><code id="enrichment_+3A_term_descriptions">term_descriptions</code></td>
<td>
<p>Vector that contains term descriptions for the
gene sets. Names of this vector are gene set IDs (default = kegg_descriptions)</p>
</td></tr>
<tr><td><code id="enrichment_+3A_adj_method">adj_method</code></td>
<td>
<p>correction method to be used for adjusting p-values.
(default = 'bonferroni')</p>
</td></tr>
<tr><td><code id="enrichment_+3A_enrichment_threshold">enrichment_threshold</code></td>
<td>
<p>adjusted-p value threshold used when filtering
enrichment results (default = 0.05)</p>
</td></tr>
<tr><td><code id="enrichment_+3A_sig_genes_vec">sig_genes_vec</code></td>
<td>
<p>vector of significant gene symbols. In the scope of this
package, these are the input genes that were used for active subnetwork search</p>
</td></tr>
<tr><td><code id="enrichment_+3A_background_genes">background_genes</code></td>
<td>
<p>vector of background genes. In the scope of this package,
the background genes are taken as all genes in the PIN
(see <code><a href="#topic+enrichment_analyses">enrichment_analyses</a></code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame that contains enrichment results
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+p.adjust">p.adjust</a></code> for adjustment of p values. See
<code><a href="#topic+run_pathfindR">run_pathfindR</a></code> for the wrapper function of the pathfindR
workflow. <code><a href="#topic+hyperg_test">hyperg_test</a></code> for the details on hypergeometric
distribution-based hypothesis testing.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>enrichment(
  input_genes = c('PER1', 'PER2', 'CRY1', 'CREB1'),
  sig_genes_vec = 'PER1',
  background_genes = unlist(pathfindR.data::kegg_genes)
)
</code></pre>

<hr>
<h2 id='enrichment_analyses'>Perform Enrichment Analyses on the Input Subnetworks</h2><span id='topic+enrichment_analyses'></span>

<h3>Description</h3>

<p>Perform Enrichment Analyses on the Input Subnetworks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enrichment_analyses(
  snws,
  sig_genes_vec,
  pin_name_path = "Biogrid",
  genes_by_term = pathfindR.data::kegg_genes,
  term_descriptions = pathfindR.data::kegg_descriptions,
  adj_method = "bonferroni",
  enrichment_threshold = 0.05,
  list_active_snw_genes = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="enrichment_analyses_+3A_snws">snws</code></td>
<td>
<p>a list of subnetwork genes (i.e., vectors of genes for each subnetwork)</p>
</td></tr>
<tr><td><code id="enrichment_analyses_+3A_sig_genes_vec">sig_genes_vec</code></td>
<td>
<p>vector of significant gene symbols. In the scope of this
package, these are the input genes that were used for active subnetwork search</p>
</td></tr>
<tr><td><code id="enrichment_analyses_+3A_pin_name_path">pin_name_path</code></td>
<td>
<p>Name of the chosen PIN or absolute/path/to/PIN.sif. If PIN name,
must be one of c('Biogrid', 'STRING', 'GeneMania', 'IntAct', 'KEGG', 'mmu_STRING'). If
path/to/PIN.sif, the file must comply with the PIN specifications. (Default = 'Biogrid')</p>
</td></tr>
<tr><td><code id="enrichment_analyses_+3A_genes_by_term">genes_by_term</code></td>
<td>
<p>List that contains genes for each gene set. Names of
this list are gene set IDs (default = kegg_genes)</p>
</td></tr>
<tr><td><code id="enrichment_analyses_+3A_term_descriptions">term_descriptions</code></td>
<td>
<p>Vector that contains term descriptions for the
gene sets. Names of this vector are gene set IDs (default = kegg_descriptions)</p>
</td></tr>
<tr><td><code id="enrichment_analyses_+3A_adj_method">adj_method</code></td>
<td>
<p>correction method to be used for adjusting p-values.
(default = 'bonferroni')</p>
</td></tr>
<tr><td><code id="enrichment_analyses_+3A_enrichment_threshold">enrichment_threshold</code></td>
<td>
<p>adjusted-p value threshold used when filtering
enrichment results (default = 0.05)</p>
</td></tr>
<tr><td><code id="enrichment_analyses_+3A_list_active_snw_genes">list_active_snw_genes</code></td>
<td>
<p>boolean value indicating whether or not to report
the non-significant active subnetwork genes for the active subnetwork which was enriched for
the given term with the lowest p value (default = <code>FALSE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of combined enrichment results. Columns are: </p>

<dl>
<dt>ID</dt><dd><p>ID of the enriched term</p>
</dd>
<dt>Term_Description</dt><dd><p>Description of the enriched term</p>
</dd>
<dt>Fold_Enrichment</dt><dd><p>Fold enrichment value for the enriched term</p>
</dd>
<dt>p_value</dt><dd><p>p value of enrichment</p>
</dd>
<dt>adj_p</dt><dd><p>adjusted p value of enrichment</p>
</dd>
<dt>support</dt><dd><p>the support (proportion of active subnetworks leading to enrichment over all subnetworks) for the gene set</p>
</dd>
<dt>non_Signif_Snw_Genes (OPTIONAL)</dt><dd><p>the non-significant active subnetwork genes, comma-separated</p>
</dd>
</dl>



<h3>See Also</h3>

<p><code><a href="#topic+enrichment">enrichment</a></code> for the enrichment analysis for a single gene set
</p>


<h3>Examples</h3>

<pre><code class='language-R'>enr_res &lt;- enrichment_analyses(
  snws = example_active_snws[1:2],
  sig_genes_vec = example_pathfindR_input$Gene.symbol[1:25],
  pin_name_path = 'KEGG'
)
</code></pre>

<hr>
<h2 id='enrichment_chart'>Create Bubble Chart of Enrichment Results</h2><span id='topic+enrichment_chart'></span>

<h3>Description</h3>

<p>This function is used to create a ggplot2 bubble chart displaying the
enrichment results.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enrichment_chart(
  result_df,
  top_terms = 10,
  plot_by_cluster = FALSE,
  num_bubbles = 4,
  even_breaks = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="enrichment_chart_+3A_result_df">result_df</code></td>
<td>
<p>a data frame that must contain the following columns: </p>

<dl>
<dt>Term_Description</dt><dd><p>Description of the enriched term</p>
</dd>
<dt>Fold_Enrichment</dt><dd><p>Fold enrichment value for the enriched term</p>
</dd>
<dt>lowest_p</dt><dd><p>the lowest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>Up_regulated</dt><dd><p>the up-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>Down_regulated</dt><dd><p>the down-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>Cluster(OPTIONAL)</dt><dd><p>the cluster to which the enriched term is assigned</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="enrichment_chart_+3A_top_terms">top_terms</code></td>
<td>
<p>number of top terms (according to the 'lowest_p' column)
to plot (default = 10). If <code>plot_by_cluster = TRUE</code>, selects the top
<code>top_terms</code> terms per each cluster. Set <code>top_terms = NULL</code> to plot
for all terms.If the total number of terms is less than <code>top_terms</code>,
all terms are plotted.</p>
</td></tr>
<tr><td><code id="enrichment_chart_+3A_plot_by_cluster">plot_by_cluster</code></td>
<td>
<p>boolean value indicating whether or not to group the
enriched terms by cluster (works if <code>result_df</code> contains a
'Cluster' column).</p>
</td></tr>
<tr><td><code id="enrichment_chart_+3A_num_bubbles">num_bubbles</code></td>
<td>
<p>number of sizes displayed in the legend <code># genes</code>
(Default = 4)</p>
</td></tr>
<tr><td><code id="enrichment_chart_+3A_even_breaks">even_breaks</code></td>
<td>
<p>whether or not to set even breaks for the number of sizes
displayed in the legend <code># genes</code>. If <code>TRUE</code> (default), sets
equal breaks and the number of displayed bubbles may be different than the
number set by <code>num_bubbles</code>. If the exact number set by
<code>num_bubbles</code> is required, set this argument to <code>FALSE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code><a href="ggplot2.html#topic+ggplot2">ggplot2</a></code> object containing the bubble chart.
The x-axis corresponds to fold enrichment values while the y-axis indicates
the enriched terms. Size of the bubble indicates the number of significant
genes in the given enriched term. Color indicates the -log10(lowest-p) value.
The closer the color is to red, the more significant the enrichment is.
Optionally, if 'Cluster' is a column of <code>result_df</code> and
<code>plot_by_cluster == TRUE</code>, the enriched terms are grouped by clusters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g &lt;- enrichment_chart(example_pathfindR_output)
</code></pre>

<hr>
<h2 id='fetch_gene_set'>Fetch Gene Set Objects</h2><span id='topic+fetch_gene_set'></span>

<h3>Description</h3>

<p>Function for obtaining the gene sets per term and the term descriptions to
be used for enrichment analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fetch_gene_set(
  gene_sets = "KEGG",
  min_gset_size = 10,
  max_gset_size = 300,
  custom_genes = NULL,
  custom_descriptions = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fetch_gene_set_+3A_gene_sets">gene_sets</code></td>
<td>
<p>Name of the gene sets to be used for enrichment analysis.
Available gene sets are 'KEGG', 'Reactome', 'BioCarta', 'GO-All',
'GO-BP', 'GO-CC', 'GO-MF', 'cell_markers', 'mmu_KEGG' or 'Custom'.
If 'Custom', the arguments <code>custom_genes</code> and <code>custom_descriptions</code>
must be specified. (Default = 'KEGG')</p>
</td></tr>
<tr><td><code id="fetch_gene_set_+3A_min_gset_size">min_gset_size</code></td>
<td>
<p>minimum number of genes a term must contain (default = 10)</p>
</td></tr>
<tr><td><code id="fetch_gene_set_+3A_max_gset_size">max_gset_size</code></td>
<td>
<p>maximum number of genes a term must contain (default = 300)</p>
</td></tr>
<tr><td><code id="fetch_gene_set_+3A_custom_genes">custom_genes</code></td>
<td>
<p>a list containing the genes involved in each custom
term. Each element is a vector of gene symbols located in the given custom
term. Names should correspond to the IDs of the custom terms.</p>
</td></tr>
<tr><td><code id="fetch_gene_set_+3A_custom_descriptions">custom_descriptions</code></td>
<td>
<p>A vector containing the descriptions for each
custom  term. Names of the vector should correspond to the IDs of the custom
terms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing 2 elements </p>

<dl>
<dt>genes_by_term</dt><dd><p>list of vectors of genes contained in each term</p>
</dd>
<dt>term_descriptions</dt><dd><p>vector of descriptions per each term</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>KEGG_gset &lt;- fetch_gene_set()
GO_MF_gset &lt;- fetch_gene_set('GO-MF', min_gset_size = 20, max_gset_size = 100)
</code></pre>

<hr>
<h2 id='fetch_java_version'>Obtain Java Version</h2><span id='topic+fetch_java_version'></span>

<h3>Description</h3>

<p>Obtain Java Version
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fetch_java_version()
</code></pre>


<h3>Details</h3>

<p>this function was adapted from the CRAN package <code>sparklyr</code>
</p>


<h3>Value</h3>

<p>character vector containing the output of 'java -version'
</p>

<hr>
<h2 id='filterActiveSnws'>Parse Active Subnetwork Search Output File and Filter the Subnetworks</h2><span id='topic+filterActiveSnws'></span>

<h3>Description</h3>

<p>Parse Active Subnetwork Search Output File and Filter the Subnetworks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterActiveSnws(
  active_snw_path,
  sig_genes_vec,
  score_quan_thr = 0.8,
  sig_gene_thr = 0.02
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterActiveSnws_+3A_active_snw_path">active_snw_path</code></td>
<td>
<p>path to the output of an Active Subnetwork Search</p>
</td></tr>
<tr><td><code id="filterActiveSnws_+3A_sig_genes_vec">sig_genes_vec</code></td>
<td>
<p>vector of significant gene symbols. In the scope of this
package, these are the input genes that were used for active subnetwork search</p>
</td></tr>
<tr><td><code id="filterActiveSnws_+3A_score_quan_thr">score_quan_thr</code></td>
<td>
<p>active subnetwork score quantile threshold. Must be
between 0 and 1 or set to -1 for not filtering. (Default = 0.8)</p>
</td></tr>
<tr><td><code id="filterActiveSnws_+3A_sig_gene_thr">sig_gene_thr</code></td>
<td>
<p>threshold for the minimum proportion of significant genes in
the subnetwork (Default = 0.02) If the number of genes to use as threshold is
calculated to be &lt; 2 (e.g. 50 signif. genes x 0.01 = 0.5), the threshold number
is set to 2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing <code>subnetworks</code>: a list of of genes in every
active subnetwork that has a score greater than the <code>score_quan_thr</code>th
quantile and that contains at least <code>sig_gene_thr</code> of significant genes
and <code>scores</code> the score of each filtered active subnetwork
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+run_pathfindR">run_pathfindR</a></code> for the wrapper function of the
pathfindR enrichment workflow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path2snw_list &lt;- system.file(
  'extdata/resultActiveSubnetworkSearch.txt',
  package = 'pathfindR'
)
filtered &lt;- filterActiveSnws(
  active_snw_path = path2snw_list,
  sig_genes_vec = example_pathfindR_input$Gene.symbol
)
</code></pre>

<hr>
<h2 id='fuzzy_term_clustering'>Heuristic Fuzzy Multiple-linkage Partitioning of Enriched Terms</h2><span id='topic+fuzzy_term_clustering'></span>

<h3>Description</h3>

<p>Heuristic Fuzzy Multiple-linkage Partitioning of Enriched Terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fuzzy_term_clustering(
  kappa_mat,
  enrichment_res,
  kappa_threshold = 0.35,
  use_description = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fuzzy_term_clustering_+3A_kappa_mat">kappa_mat</code></td>
<td>
<p>matrix of kappa statistics (output of <code><a href="#topic+create_kappa_matrix">create_kappa_matrix</a></code>)</p>
</td></tr>
<tr><td><code id="fuzzy_term_clustering_+3A_enrichment_res">enrichment_res</code></td>
<td>
<p>data frame of pathfindR enrichment results. Must-have
columns are 'Term_Description' (if <code>use_description = TRUE</code>) or 'ID'
(if <code>use_description = FALSE</code>), 'Down_regulated', and 'Up_regulated'.
If <code>use_active_snw_genes = TRUE</code>, 'non_Signif_Snw_Genes' must also be
provided.</p>
</td></tr>
<tr><td><code id="fuzzy_term_clustering_+3A_kappa_threshold">kappa_threshold</code></td>
<td>
<p>threshold for kappa statistics, defining strong
relation (default = 0.35)</p>
</td></tr>
<tr><td><code id="fuzzy_term_clustering_+3A_use_description">use_description</code></td>
<td>
<p>Boolean argument to indicate whether term descriptions
(in the 'Term_Description' column) should be used. (default = <code>FALSE</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The fuzzy clustering algorithm was implemented based on:
Huang DW, Sherman BT, Tan Q, et al. The DAVID Gene Functional
Classification Tool: a novel biological module-centric algorithm to
functionally analyze large gene lists. Genome Biol. 2007;8(9):R183.
</p>


<h3>Value</h3>

<p>a boolean matrix of cluster assignments. Each row corresponds to an
enriched term, each column corresponds to a cluster.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
fuzzy_term_clustering(kappa_mat, enrichment_res)
fuzzy_term_clustering(kappa_mat, enrichment_res, kappa_threshold = 0.45)

## End(Not run)
</code></pre>

<hr>
<h2 id='get_biogrid_pin'>Retrieve the Requested Release of Organism-specific BioGRID PIN</h2><span id='topic+get_biogrid_pin'></span>

<h3>Description</h3>

<p>Retrieve the Requested Release of Organism-specific BioGRID PIN
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_biogrid_pin(org = "Homo_sapiens", path2pin, release = "latest")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_biogrid_pin_+3A_org">org</code></td>
<td>
<p>organism name. BioGRID naming requires underscores for spaces so
'Homo sapiens' becomes 'Homo_sapiens', 'Mus musculus' becomes 'Mus_musculus'
etc. See <a href="https://wiki.thebiogrid.org/doku.php/statistics">https://wiki.thebiogrid.org/doku.php/statistics</a> for a full
list of available organisms (default = 'Homo_sapiens')</p>
</td></tr>
<tr><td><code id="get_biogrid_pin_+3A_path2pin">path2pin</code></td>
<td>
<p>the path of the file to save the PIN data. By default, the
PIN data is saved in a temporary file</p>
</td></tr>
<tr><td><code id="get_biogrid_pin_+3A_release">release</code></td>
<td>
<p>the requested BioGRID release (default = 'latest')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the path of the file in which the PIN data was saved. If
<code>path2pin</code> was not supplied by the user, the PIN data is saved in a
temporary file
</p>

<hr>
<h2 id='get_gene_sets_list'>Retrieve Organism-specific Gene Sets List</h2><span id='topic+get_gene_sets_list'></span>

<h3>Description</h3>

<p>Retrieve Organism-specific Gene Sets List
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gene_sets_list(
  source = "KEGG",
  org_code = "hsa",
  species = "Homo sapiens",
  collection,
  subcollection = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_gene_sets_list_+3A_source">source</code></td>
<td>
<p>As of this version, either 'KEGG', 'Reactome' or 'MSigDB' (default = 'KEGG')</p>
</td></tr>
<tr><td><code id="get_gene_sets_list_+3A_org_code">org_code</code></td>
<td>
<p>(Used for 'KEGG' only) KEGG organism code for the selected organism. For a full list
of all available organisms, see <a href="https://www.genome.jp/kegg/catalog/org_list.html">https://www.genome.jp/kegg/catalog/org_list.html</a></p>
</td></tr>
<tr><td><code id="get_gene_sets_list_+3A_species">species</code></td>
<td>
<p>(Used for 'MSigDB' only) species name, such as Homo sapiens, Mus musculus, etc.
See <code><a href="msigdbr.html#topic+msigdbr_show_species">msigdbr_show_species</a></code> for all the species available in
the msigdbr package (default = 'Homo sapiens')</p>
</td></tr>
<tr><td><code id="get_gene_sets_list_+3A_collection">collection</code></td>
<td>
<p>(Used for 'MSigDB' only) collection. i.e., H, C1, C2, C3, C4, C5, C6, C7.</p>
</td></tr>
<tr><td><code id="get_gene_sets_list_+3A_subcollection">subcollection</code></td>
<td>
<p>(Used for 'MSigDB' only)  sub-collection, such as CGP, MIR, BP, etc. (default = NULL,
i.e. list all gene sets in collection)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing 2 elements: </p>

<ul>
<li><p>gene_sets - A list containing the genes involved in each gene set
</p>
</li>
<li><p>descriptions - A named vector containing the descriptions for each gene set
</p>
</li></ul>
<p>. For 'KEGG' and 'MSigDB', it is possible to choose a specific organism. For a full list
of all available KEGG organisms, see <a href="https://www.genome.jp/kegg/catalog/org_list.html">https://www.genome.jp/kegg/catalog/org_list.html</a>.
See <code><a href="msigdbr.html#topic+msigdbr_show_species">msigdbr_show_species</a></code> for all the species available in
the msigdbr package used for obtaining 'MSigDB' gene sets.
For Reactome, there is only one collection of pathway gene sets.
</p>

<hr>
<h2 id='get_kegg_gsets'>Retrieve Organism-specific KEGG Pathway Gene Sets</h2><span id='topic+get_kegg_gsets'></span>

<h3>Description</h3>

<p>Retrieve Organism-specific KEGG Pathway Gene Sets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_kegg_gsets(org_code = "hsa")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_kegg_gsets_+3A_org_code">org_code</code></td>
<td>
<p>KEGG organism code for the selected organism. For a full list
of all available organisms, see <a href="https://www.genome.jp/kegg/catalog/org_list.html">https://www.genome.jp/kegg/catalog/org_list.html</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing 2 elements: </p>

<ul>
<li><p>gene_sets - A list containing KEGG IDs for the genes involved in each KEGG pathway
</p>
</li>
<li><p>descriptions - A named vector containing the descriptions for each KEGG pathway
</p>
</li></ul>


<hr>
<h2 id='get_mgsigdb_gsets'>Retrieve Organism-specific MSigDB Gene Sets</h2><span id='topic+get_mgsigdb_gsets'></span>

<h3>Description</h3>

<p>Retrieve Organism-specific MSigDB Gene Sets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mgsigdb_gsets(species = "Homo sapiens", collection, subcollection = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_mgsigdb_gsets_+3A_species">species</code></td>
<td>
<p>species name, such as Homo sapiens, Mus musculus, etc.
See <code><a href="msigdbr.html#topic+msigdbr_show_species">msigdbr_show_species</a></code> for all the species available in
the msigdbr package</p>
</td></tr>
<tr><td><code id="get_mgsigdb_gsets_+3A_collection">collection</code></td>
<td>
<p>collection. i.e., H, C1, C2, C3, C4, C5, C6, C7.</p>
</td></tr>
<tr><td><code id="get_mgsigdb_gsets_+3A_subcollection">subcollection</code></td>
<td>
<p>sub-collection, such as CGP, BP, etc. (default = NULL,
i.e. list all gene sets in collection)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>this function utilizes the function <code><a href="msigdbr.html#topic+msigdbr">msigdbr</a></code>
from the <code>msigdbr</code> package to retrieve the 'Molecular Signatures Database'
(MSigDB) gene sets (Subramanian et al. 2005 &lt;doi:10.1073/pnas.0506580102&gt;,
Liberzon et al. 2015 &lt;doi:10.1016/j.cels.2015.12.004&gt;).
Available collections are: H: hallmark gene sets, C1: positional gene sets,
C2: curated gene sets, C3: motif gene sets, C4: computational gene sets,
C5: GO gene sets, C6: oncogenic signatures and C7: immunologic signatures
</p>


<h3>Value</h3>

<p>Retrieves the MSigDB gene sets and returns a list containing 2 elements: </p>

<ul>
<li><p>gene_sets - A list containing the genes involved in each of the selected MSigDB gene sets
</p>
</li>
<li><p>descriptions - A named vector containing the descriptions for each selected MSigDB gene set
</p>
</li></ul>


<hr>
<h2 id='get_pin_file'>Retrieve Organism-specific PIN data</h2><span id='topic+get_pin_file'></span>

<h3>Description</h3>

<p>Retrieve Organism-specific PIN data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pin_file(source = "BioGRID", org = "Homo_sapiens", path2pin, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pin_file_+3A_source">source</code></td>
<td>
<p>As of this version, this function is implemented to get data
from 'BioGRID' only. This argument (and this wrapper function) was implemented
for future utility</p>
</td></tr>
<tr><td><code id="get_pin_file_+3A_org">org</code></td>
<td>
<p>organism name. BioGRID naming requires underscores for spaces so
'Homo sapiens' becomes 'Homo_sapiens', 'Mus musculus' becomes 'Mus_musculus'
etc. See <a href="https://wiki.thebiogrid.org/doku.php/statistics">https://wiki.thebiogrid.org/doku.php/statistics</a> for a full
list of available organisms (default = 'Homo_sapiens')</p>
</td></tr>
<tr><td><code id="get_pin_file_+3A_path2pin">path2pin</code></td>
<td>
<p>the path of the file to save the PIN data. By default, the
PIN data is saved in a temporary file</p>
</td></tr>
<tr><td><code id="get_pin_file_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="#topic+get_biogrid_pin">get_biogrid_pin</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the path of the file in which the PIN data was saved. If
<code>path2pin</code> was not supplied by the user, the PIN data is saved in a
temporary file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pin_path &lt;- get_pin_file()

## End(Not run)
</code></pre>

<hr>
<h2 id='get_reactome_gsets'>Retrieve Reactome Pathway Gene Sets</h2><span id='topic+get_reactome_gsets'></span>

<h3>Description</h3>

<p>Retrieve Reactome Pathway Gene Sets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_reactome_gsets()
</code></pre>


<h3>Value</h3>

<p>Gets the latest Reactome pathways gene sets in gmt format. Parses the
gmt file and returns a list containing 2 elements: </p>

<ul>
<li><p>gene_sets - A list containing the genes involved in each Reactome pathway
</p>
</li>
<li><p>descriptions - A named vector containing the descriptions for each Reactome pathway
</p>
</li></ul>


<hr>
<h2 id='gset_list_from_gmt'>Retrieve Gene Sets from GMT-format File</h2><span id='topic+gset_list_from_gmt'></span>

<h3>Description</h3>

<p>Retrieve Gene Sets from GMT-format File
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gset_list_from_gmt(path2gmt, descriptions_idx = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gset_list_from_gmt_+3A_path2gmt">path2gmt</code></td>
<td>
<p>path to the gmt file</p>
</td></tr>
<tr><td><code id="gset_list_from_gmt_+3A_descriptions_idx">descriptions_idx</code></td>
<td>
<p>index for descriptions (default = 2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list containing 2 elements: </p>

<ul>
<li><p>gene_sets - A list containing the genes involved in each gene set
</p>
</li>
<li><p>descriptions - A named vector containing the descriptions for each gene set
</p>
</li></ul>


<hr>
<h2 id='hierarchical_term_clustering'>Hierarchical Clustering of Enriched Terms</h2><span id='topic+hierarchical_term_clustering'></span>

<h3>Description</h3>

<p>Hierarchical Clustering of Enriched Terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hierarchical_term_clustering(
  kappa_mat,
  enrichment_res,
  num_clusters = NULL,
  use_description = FALSE,
  clu_method = "average",
  plot_hmap = FALSE,
  plot_dend = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hierarchical_term_clustering_+3A_kappa_mat">kappa_mat</code></td>
<td>
<p>matrix of kappa statistics (output of <code><a href="#topic+create_kappa_matrix">create_kappa_matrix</a></code>)</p>
</td></tr>
<tr><td><code id="hierarchical_term_clustering_+3A_enrichment_res">enrichment_res</code></td>
<td>
<p>data frame of pathfindR enrichment results. Must-have
columns are 'Term_Description' (if <code>use_description = TRUE</code>) or 'ID'
(if <code>use_description = FALSE</code>), 'Down_regulated', and 'Up_regulated'.
If <code>use_active_snw_genes = TRUE</code>, 'non_Signif_Snw_Genes' must also be
provided.</p>
</td></tr>
<tr><td><code id="hierarchical_term_clustering_+3A_num_clusters">num_clusters</code></td>
<td>
<p>number of clusters to be formed (default = <code>NULL</code>).
If <code>NULL</code>, the optimal number of clusters is determined as the number
which yields the highest average silhouette width.</p>
</td></tr>
<tr><td><code id="hierarchical_term_clustering_+3A_use_description">use_description</code></td>
<td>
<p>Boolean argument to indicate whether term descriptions
(in the 'Term_Description' column) should be used. (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="hierarchical_term_clustering_+3A_clu_method">clu_method</code></td>
<td>
<p>the agglomeration method to be used
(default = 'average', see <code><a href="stats.html#topic+hclust">hclust</a></code>)</p>
</td></tr>
<tr><td><code id="hierarchical_term_clustering_+3A_plot_hmap">plot_hmap</code></td>
<td>
<p>boolean to indicate whether to plot the kappa statistics
clustering heatmap or not (default = FALSE)</p>
</td></tr>
<tr><td><code id="hierarchical_term_clustering_+3A_plot_dend">plot_dend</code></td>
<td>
<p>boolean to indicate whether to plot the clustering
dendrogram partitioned into the optimal number of clusters (default = TRUE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function initially performs hierarchical clustering
of the enriched terms in <code>enrichment_res</code> using the kappa statistics
(defining the distance as <code>1 - kappa_statistic</code>). Next,
the clustering dendrogram is cut into k = 2, 3, ..., n - 1 clusters
(where n is the number of terms). The optimal number of clusters is
determined as the k value which yields the highest average silhouette width.
(if <code>num_clusters</code> not specified)
</p>


<h3>Value</h3>

<p>a vector of clusters for each enriched term in the enrichment results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
hierarchical_term_clustering(kappa_mat, enrichment_res)
hierarchical_term_clustering(kappa_mat, enrichment_res, method = 'complete')

## End(Not run)
</code></pre>

<hr>
<h2 id='hyperg_test'>Hypergeometric Distribution-based Hypothesis Testing</h2><span id='topic+hyperg_test'></span>

<h3>Description</h3>

<p>Hypergeometric Distribution-based Hypothesis Testing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hyperg_test(term_genes, chosen_genes, background_genes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hyperg_test_+3A_term_genes">term_genes</code></td>
<td>
<p>vector of genes in the selected term gene set</p>
</td></tr>
<tr><td><code id="hyperg_test_+3A_chosen_genes">chosen_genes</code></td>
<td>
<p>vector containing the set of input genes</p>
</td></tr>
<tr><td><code id="hyperg_test_+3A_background_genes">background_genes</code></td>
<td>
<p>vector of background genes (i.e. universal set of
genes in the experiment)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To determine whether the <code>chosen_genes</code> are enriched
(compared to a background pool of genes) in the <code>term_genes</code>, the
hypergeometric distribution is assumed and the appropriate p value
(the value under the right tail) is calculated and returned.
</p>


<h3>Value</h3>

<p>the p-value as determined using the hypergeometric distribution.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hyperg_test(letters[1:5], letters[2:5], letters)
hyperg_test(letters[1:5], letters[2:10], letters)
hyperg_test(letters[1:5], letters[2:13], letters)
</code></pre>

<hr>
<h2 id='input_processing'>Process Input</h2><span id='topic+input_processing'></span>

<h3>Description</h3>

<p>Process Input
</p>


<h3>Usage</h3>

<pre><code class='language-R'>input_processing(
  input,
  p_val_threshold = 0.05,
  pin_name_path = "Biogrid",
  convert2alias = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="input_processing_+3A_input">input</code></td>
<td>
<p>the input data that pathfindR uses. The input must be a data
frame with three columns: </p>

<ol>
<li><p> Gene Symbol (Gene Symbol)
</p>
</li>
<li><p> Change value, e.g. log(fold change) (OPTIONAL)
</p>
</li>
<li><p> p value, e.g. adjusted p value associated with differential expression
</p>
</li></ol>
</td></tr>
<tr><td><code id="input_processing_+3A_p_val_threshold">p_val_threshold</code></td>
<td>
<p>the p value threshold to use when filtering
the input data frame. Must a numeric value between 0 and 1. (default = 0.05)</p>
</td></tr>
<tr><td><code id="input_processing_+3A_pin_name_path">pin_name_path</code></td>
<td>
<p>Name of the chosen PIN or absolute/path/to/PIN.sif. If PIN name,
must be one of c('Biogrid', 'STRING', 'GeneMania', 'IntAct', 'KEGG', 'mmu_STRING'). If
path/to/PIN.sif, the file must comply with the PIN specifications. (Default = 'Biogrid')</p>
</td></tr>
<tr><td><code id="input_processing_+3A_convert2alias">convert2alias</code></td>
<td>
<p>boolean to indicate whether or not to convert gene symbols
in the input that are not found in the PIN to an alias symbol found in the PIN
(default = TRUE) IMPORTANT NOTE: the conversion uses human gene symbols/alias symbols.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function first filters the input so that all p values are less
than or equal to the threshold. Next, gene symbols that are not found in
the PIN are identified. If aliases of these gene symbols are found in the
PIN, the symbols are converted to the corresponding aliases. The
resulting data frame containing the original gene symbols, the updated
symbols, change values and p values is then returned.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+run_pathfindR">run_pathfindR</a></code> for the wrapper function of the
pathfindR workflow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>processed_df &lt;- input_processing(
  input = example_pathfindR_input[1:5, ],
  pin_name_path = 'KEGG'
)
processed_df &lt;- input_processing(
  input = example_pathfindR_input[1:10, ],
  pin_name_path = 'KEGG',
  convert2alias = FALSE
)
</code></pre>

<hr>
<h2 id='input_testing'>Input Testing</h2><span id='topic+input_testing'></span>

<h3>Description</h3>

<p>Input Testing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>input_testing(input, p_val_threshold = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="input_testing_+3A_input">input</code></td>
<td>
<p>the input data that pathfindR uses. The input must be a data
frame with three columns: </p>

<ol>
<li><p> Gene Symbol (Gene Symbol)
</p>
</li>
<li><p> Change value, e.g. log(fold change) (OPTIONAL)
</p>
</li>
<li><p> p value, e.g. adjusted p value associated with differential expression
</p>
</li></ol>
</td></tr>
<tr><td><code id="input_testing_+3A_p_val_threshold">p_val_threshold</code></td>
<td>
<p>the p value threshold to use when filtering
the input data frame. Must a numeric value between 0 and 1. (default = 0.05)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Only checks if the input and the threshold follows the required
specifications.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+run_pathfindR">run_pathfindR</a></code> for the wrapper function of the
pathfindR workflow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>input_testing(example_pathfindR_input, 0.05)
</code></pre>

<hr>
<h2 id='isColor'>Check if value is a valid color</h2><span id='topic+isColor'></span>

<h3>Description</h3>

<p>Check if value is a valid color
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isColor(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="isColor_+3A_x">x</code></td>
<td>
<p>value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if x is a valid color, otherwise FALSE
</p>

<hr>
<h2 id='plot_scores'>Plot the Heatmap of Score Matrix of Enriched Terms per Sample</h2><span id='topic+plot_scores'></span>

<h3>Description</h3>

<p>Plot the Heatmap of Score Matrix of Enriched Terms per Sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_scores(
  score_matrix,
  cases = NULL,
  label_samples = TRUE,
  case_title = "Case",
  control_title = "Control",
  low = "green",
  mid = "black",
  high = "red"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_scores_+3A_score_matrix">score_matrix</code></td>
<td>
<p>Matrix of agglomerated enriched term scores per sample. Columns are
samples, rows are enriched terms</p>
</td></tr>
<tr><td><code id="plot_scores_+3A_cases">cases</code></td>
<td>
<p>(Optional) A vector of sample names that are cases in the
case/control experiment. (default = NULL)</p>
</td></tr>
<tr><td><code id="plot_scores_+3A_label_samples">label_samples</code></td>
<td>
<p>Boolean value to indicate whether or not to label the
samples in the heatmap plot (default = TRUE)</p>
</td></tr>
<tr><td><code id="plot_scores_+3A_case_title">case_title</code></td>
<td>
<p>Naming of the 'Case' group (as in <code>cases</code>) (default = 'Case')</p>
</td></tr>
<tr><td><code id="plot_scores_+3A_control_title">control_title</code></td>
<td>
<p>Naming of the 'Control' group (default = 'Control')</p>
</td></tr>
<tr><td><code id="plot_scores_+3A_low">low</code></td>
<td>
<p>a string indicating the color of 'low' values in the coloring gradient (default = 'green')</p>
</td></tr>
<tr><td><code id="plot_scores_+3A_mid">mid</code></td>
<td>
<p>a string indicating the color of 'mid' values in the coloring gradient (default = 'black')</p>
</td></tr>
<tr><td><code id="plot_scores_+3A_high">high</code></td>
<td>
<p>a string indicating the color of 'high' values in the coloring gradient (default = 'red')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 'ggplot2' object containing the heatmap plot. x-axis indicates
the samples. y-axis indicates the enriched terms. 'Score' indicates the
score of the term in a given sample. If <code>cases</code> are provided, the plot is
divided into 2 facets, named by <code>case_title</code> and <code>control_title</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>score_matrix &lt;- score_terms(
  example_pathfindR_output,
  example_experiment_matrix,
  plot_hmap = FALSE
)
hmap &lt;- plot_scores(score_matrix)
</code></pre>

<hr>
<h2 id='process_pin'>Process Data frame of Protein-protein Interactions</h2><span id='topic+process_pin'></span>

<h3>Description</h3>

<p>Process Data frame of Protein-protein Interactions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_pin(pin_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process_pin_+3A_pin_df">pin_df</code></td>
<td>
<p>data frame of protein-protein interactions with 2 columns:
'Interactor_A' and 'Interactor_B'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>processed PIN data frame (removes self-interactions and
duplicated interactions)
</p>

<hr>
<h2 id='return_pin_path'>Return The Path to Given Protein-Protein Interaction Network (PIN)</h2><span id='topic+return_pin_path'></span>

<h3>Description</h3>

<p>This function returns the absolute path/to/PIN.sif. While the default PINs are
'Biogrid', 'STRING', 'GeneMania', 'IntAct', 'KEGG' and 'mmu_STRING'. The user can also
use any other PIN by specifying the 'path/to/PIN.sif'. All PINs to be used
in this package must formatted as SIF files: i.e. have 3 columns with no
header, no row names and be tab-separated. Columns 1 and 3 must be
interactors' gene symbols, column 2 must be a column with all
rows consisting of 'pp'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>return_pin_path(pin_name_path = "Biogrid")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="return_pin_path_+3A_pin_name_path">pin_name_path</code></td>
<td>
<p>Name of the chosen PIN or absolute/path/to/PIN.sif. If PIN name,
must be one of c('Biogrid', 'STRING', 'GeneMania', 'IntAct', 'KEGG', 'mmu_STRING'). If
path/to/PIN.sif, the file must comply with the PIN specifications. (Default = 'Biogrid')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The absolute path to chosen PIN.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+run_pathfindR">run_pathfindR</a></code> for the wrapper function of the
pathfindR workflow
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pin_path &lt;- return_pin_path('GeneMania')

## End(Not run)
</code></pre>

<hr>
<h2 id='run_pathfindR'>Wrapper Function for pathfindR - Active-Subnetwork-Oriented Enrichment Workflow</h2><span id='topic+run_pathfindR'></span>

<h3>Description</h3>

<p><code>run_pathfindR</code> is the wrapper function for the pathfindR workflow
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_pathfindR(
  input,
  gene_sets = "KEGG",
  min_gset_size = 10,
  max_gset_size = 300,
  custom_genes = NULL,
  custom_descriptions = NULL,
  pin_name_path = "Biogrid",
  p_val_threshold = 0.05,
  enrichment_threshold = 0.05,
  convert2alias = TRUE,
  plot_enrichment_chart = TRUE,
  output_dir = NULL,
  list_active_snw_genes = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_pathfindR_+3A_input">input</code></td>
<td>
<p>the input data that pathfindR uses. The input must be a data
frame with three columns: </p>

<ol>
<li><p> Gene Symbol (Gene Symbol)
</p>
</li>
<li><p> Change value, e.g. log(fold change) (OPTIONAL)
</p>
</li>
<li><p> p value, e.g. adjusted p value associated with differential expression
</p>
</li></ol>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_gene_sets">gene_sets</code></td>
<td>
<p>Name of the gene sets to be used for enrichment analysis.
Available gene sets are 'KEGG', 'Reactome', 'BioCarta', 'GO-All',
'GO-BP', 'GO-CC', 'GO-MF', 'cell_markers', 'mmu_KEGG' or 'Custom'.
If 'Custom', the arguments <code>custom_genes</code> and <code>custom_descriptions</code>
must be specified. (Default = 'KEGG')</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_min_gset_size">min_gset_size</code></td>
<td>
<p>minimum number of genes a term must contain (default = 10)</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_max_gset_size">max_gset_size</code></td>
<td>
<p>maximum number of genes a term must contain (default = 300)</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_custom_genes">custom_genes</code></td>
<td>
<p>a list containing the genes involved in each custom
term. Each element is a vector of gene symbols located in the given custom
term. Names should correspond to the IDs of the custom terms.</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_custom_descriptions">custom_descriptions</code></td>
<td>
<p>A vector containing the descriptions for each
custom  term. Names of the vector should correspond to the IDs of the custom
terms.</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_pin_name_path">pin_name_path</code></td>
<td>
<p>Name of the chosen PIN or absolute/path/to/PIN.sif. If PIN name,
must be one of c('Biogrid', 'STRING', 'GeneMania', 'IntAct', 'KEGG', 'mmu_STRING'). If
path/to/PIN.sif, the file must comply with the PIN specifications. (Default = 'Biogrid')</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_p_val_threshold">p_val_threshold</code></td>
<td>
<p>the p value threshold to use when filtering
the input data frame. Must a numeric value between 0 and 1. (default = 0.05)</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_enrichment_threshold">enrichment_threshold</code></td>
<td>
<p>adjusted-p value threshold used when filtering
enrichment results (default = 0.05)</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_convert2alias">convert2alias</code></td>
<td>
<p>boolean to indicate whether or not to convert gene symbols
in the input that are not found in the PIN to an alias symbol found in the PIN
(default = TRUE) IMPORTANT NOTE: the conversion uses human gene symbols/alias symbols.</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_plot_enrichment_chart">plot_enrichment_chart</code></td>
<td>
<p>boolean value. If TRUE, a bubble chart displaying
the enrichment results is plotted. (default = TRUE)</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_output_dir">output_dir</code></td>
<td>
<p>the directory to be created where the output and intermediate
files are saved (default = <code>NULL</code>, a temporary directory is used)</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_list_active_snw_genes">list_active_snw_genes</code></td>
<td>
<p>boolean value indicating whether or not to report
the non-significant active subnetwork genes for the active subnetwork which was enriched for
the given term with the lowest p value (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="run_pathfindR_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="#topic+active_snw_enrichment_wrapper">active_snw_enrichment_wrapper</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function takes in a data frame consisting of Gene Symbol, log-fold-change
and adjusted-p values. After input testing, any gene symbols that are not in
the PIN are converted to alias symbols if the alias is in the PIN. Next,
active subnetwork search is performed. Enrichment analysis is
performed using the genes in each of the active subnetworks. Terms with
adjusted-p values lower than <code>enrichment_threshold</code> are discarded. The
lowest adjusted-p value (over all subnetworks) for each term is kept. This
process of active subnetwork search and enrichment is repeated  for a selected
number of <code>iterations</code>, which is done in parallel. Over all iterations,
the lowest and the highest adjusted-p values, as well as number of occurrences
are reported for each enriched term.
</p>


<h3>Value</h3>

<p>Data frame of pathfindR enrichment results. Columns are: </p>

<dl>
<dt>ID</dt><dd><p>ID of the enriched term</p>
</dd>
<dt>Term_Description</dt><dd><p>Description of the enriched term</p>
</dd>
<dt>Fold_Enrichment</dt><dd><p>Fold enrichment value for the enriched term (Calculated using ONLY the input genes)</p>
</dd>
<dt>occurrence</dt><dd><p>the number of iterations that the given term was found to enriched over all iterations</p>
</dd>
<dt>support</dt><dd><p>the median support (proportion of active subnetworks leading to enrichment within an iteration) over all iterations</p>
</dd>
<dt>lowest_p</dt><dd><p>the lowest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>highest_p</dt><dd><p>the highest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>non_Signif_Snw_Genes (OPTIONAL)</dt><dd><p>the non-significant active subnetwork genes, comma-separated</p>
</dd>
<dt>Up_regulated</dt><dd><p>the up-regulated genes (as determined by &lsquo;change value' &gt; 0, if the 'change column' was provided) in the input involved in the given term&rsquo;s gene set, comma-separated. If change column not provided, all affected are listed here.</p>
</dd>
<dt>Down_regulated</dt><dd><p>the down-regulated genes (as determined by &lsquo;change value' &lt; 0, if the 'change column' was provided) in the input involved in the given term&rsquo;s gene set, comma-separated</p>
</dd>
</dl>

<p>The function also creates an HTML report with the pathfindR enrichment
results linked to the visualizations of the enriched terms in addition to
the table of converted gene symbols. This report can be found in
'<code>output_dir</code>/results.html' under the current working directory.
</p>
<p>By default, a bubble chart of top 10 enrichment results are plotted. The x-axis
corresponds to fold enrichment values while the y-axis indicates the enriched
terms. Sizes of the bubbles indicate the number of significant genes in the given terms.
Color indicates the -log10(lowest-p) value; the more red it is, the more
significant the enriched term is. See <code><a href="#topic+enrichment_chart">enrichment_chart</a></code>.
</p>


<h3>Warning</h3>

<p>Especially depending on the protein interaction network,
the algorithm and the number of iterations you choose, 'active subnetwork
search + enrichment' component of <code>run_pathfindR</code> may take a long time to finish.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+input_testing">input_testing</a></code> for input testing, <code><a href="#topic+input_processing">input_processing</a></code> for input processing,
<code><a href="#topic+active_snw_search">active_snw_search</a></code> for active subnetwork search and subnetwork filtering,
<code><a href="#topic+enrichment_analyses">enrichment_analyses</a></code> for enrichment analysis (using the active subnetworks),
<code><a href="#topic+summarize_enrichment_results">summarize_enrichment_results</a></code> for summarizing the active-subnetwork-oriented enrichment results,
<code><a href="#topic+annotate_term_genes">annotate_term_genes</a></code> for annotation of affected genes in the given gene sets,
<code><a href="#topic+visualize_terms">visualize_terms</a></code> for visualization of enriched terms,
<code><a href="#topic+enrichment_chart">enrichment_chart</a></code> for a visual summary of the pathfindR enrichment results,
<code><a href="foreach.html#topic+foreach">foreach</a></code> for details on parallel execution of looping constructs,
<code><a href="#topic+cluster_enriched_terms">cluster_enriched_terms</a></code> for clustering the resulting enriched terms and partitioning into clusters.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
run_pathfindR(example_pathfindR_input)

## End(Not run)
</code></pre>

<hr>
<h2 id='score_terms'>Calculate Agglomerated Scores of Enriched Terms for Each Subject</h2><span id='topic+score_terms'></span>

<h3>Description</h3>

<p>Calculate Agglomerated Scores of Enriched Terms for Each Subject
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score_terms(
  enrichment_table,
  exp_mat,
  cases = NULL,
  use_description = FALSE,
  plot_hmap = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="score_terms_+3A_enrichment_table">enrichment_table</code></td>
<td>
<p>a data frame that must contain the 3 columns below: </p>

<dl>
<dt>Term_Description</dt><dd><p>Description of the enriched term (necessary if <code>use_description = TRUE</code>)</p>
</dd>
<dt>ID</dt><dd><p>ID of the enriched term (necessary if <code>use_description = FALSE</code>)</p>
</dd>
<dt>Up_regulated</dt><dd><p>the up-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>Down_regulated</dt><dd><p>the down-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="score_terms_+3A_exp_mat">exp_mat</code></td>
<td>
<p>the experiment (e.g., gene expression/methylation) matrix.
Columns are samples and rows are genes. Column names must contain sample
names and row names must contain the gene symbols.</p>
</td></tr>
<tr><td><code id="score_terms_+3A_cases">cases</code></td>
<td>
<p>(Optional) A vector of sample names that are cases in the
case/control experiment. (default = NULL)</p>
</td></tr>
<tr><td><code id="score_terms_+3A_use_description">use_description</code></td>
<td>
<p>Boolean argument to indicate whether term descriptions
(in the 'Term_Description' column) should be used. (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="score_terms_+3A_plot_hmap">plot_hmap</code></td>
<td>
<p>Boolean value to indicate whether or not to draw the
heatmap plot of the scores. (default = TRUE)</p>
</td></tr>
<tr><td><code id="score_terms_+3A_...">...</code></td>
<td>
<p>Additional arguments for <code><a href="#topic+plot_scores">plot_scores</a></code> for aesthetics
of the heatmap plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of agglomerated scores of each enriched term per sample.
Columns are samples, rows are enriched terms. Optionally, displays a heatmap
of this matrix.
</p>


<h3>Conceptual Background</h3>

<p>For an experiment matrix (containing expression, methylation, etc. values),
the rows of which are genes and the columns of which are samples,
we denote: </p>

<ul>
<li><p> E as a matrix of size m x n
</p>
</li>
<li><p> G as the set of all genes in the experiment G = E<sub>i.</sub>,  i &#8712; [1, m]
</p>
</li>
<li><p> S as the set of all samples in the experiment S = E<sub>.j</sub>,  i &#8712; [1, n]
</p>
</li></ul>

<p>We next define the gene score matrix GS (the standardized experiment matrix,
also of size m x n) as:
</p>
GS<sub>gs</sub> = (E<sub>gs</sub> - &#x113;<sub>g</sub>) / s<sub>g</sub>
<p>where g &#8712; G, s &#8712; S,
&#x113;<sub>g</sub> is the mean of
all values for gene g and s<sub>g</sub>
is the standard deviation of all values for gene g.
</p>
<p>We next denote T to be a set of terms (where each t &#8712; T
is a set of term-related genes, i.e.,
t = {g<sub>x</sub>, ..., g<sub>y</sub>} &sub; G)
and finally define the agglomerated term scores matrix TS (where rows
correspond to genes and columns corresponds to samples s.t. the matrix has size
|T| x n) as:
</p>
TS<sub>ts</sub> = 1/|t| &#x2211; <sub>g &#8712; t</sub> GS<sub>gs</sub><p>,
where t &#8712; T and s &#8712; S.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>score_matrix &lt;- score_terms(
  example_pathfindR_output,
  example_experiment_matrix,
  plot_hmap = FALSE
)
</code></pre>

<hr>
<h2 id='single_iter_wrapper'>Active Subnetwork Search + Enrichment Analysis Wrapper for a Single Iteration</h2><span id='topic+single_iter_wrapper'></span>

<h3>Description</h3>

<p>Active Subnetwork Search + Enrichment Analysis Wrapper for a Single Iteration
</p>


<h3>Usage</h3>

<pre><code class='language-R'>single_iter_wrapper(
  i = NULL,
  dirs,
  input_processed,
  pin_path,
  score_quan_thr,
  sig_gene_thr,
  search_method,
  silent_option,
  use_all_positives,
  geneInitProbs,
  saTemp0,
  saTemp1,
  saIter,
  gaPop,
  gaIter,
  gaThread,
  gaCrossover,
  gaMut,
  grMaxDepth,
  grSearchDepth,
  grOverlap,
  grSubNum,
  gset_list,
  adj_method,
  enrichment_threshold,
  list_active_snw_genes
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="single_iter_wrapper_+3A_i">i</code></td>
<td>
<p>current iteration index (default = <code>NULL</code>)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_dirs">dirs</code></td>
<td>
<p>vector of directories for parallel runs</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_input_processed">input_processed</code></td>
<td>
<p>processed input data frame</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_pin_path">pin_path</code></td>
<td>
<p>path/to/PIN/file</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_score_quan_thr">score_quan_thr</code></td>
<td>
<p>active subnetwork score quantile threshold. Must be
between 0 and 1 or set to -1 for not filtering. (Default = 0.8)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_sig_gene_thr">sig_gene_thr</code></td>
<td>
<p>threshold for the minimum proportion of significant genes in
the subnetwork (Default = 0.02) If the number of genes to use as threshold is
calculated to be &lt; 2 (e.g. 50 signif. genes x 0.01 = 0.5), the threshold number
is set to 2</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_search_method">search_method</code></td>
<td>
<p>algorithm to use when performing active subnetwork
search. Options are greedy search (GR), simulated annealing (SA) or genetic
algorithm (GA) for the search (default = 'GR').</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_silent_option">silent_option</code></td>
<td>
<p>boolean value indicating whether to print the messages
to the console (FALSE) or not (TRUE, this will print to a temp. file) during
active subnetwork search (default = TRUE). This option was added because
during parallel runs, the console messages get disorderly printed.</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_use_all_positives">use_all_positives</code></td>
<td>
<p>if TRUE: in GA, adds an individual with all positive
nodes. In SA, initializes candidate solution with all positive nodes. (default = FALSE)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_geneinitprobs">geneInitProbs</code></td>
<td>
<p>For SA and GA, probability of adding a gene in initial solution (default = 0.1)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_satemp0">saTemp0</code></td>
<td>
<p>Initial temperature for SA (default = 1.0)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_satemp1">saTemp1</code></td>
<td>
<p>Final temperature for SA (default = 0.01)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_saiter">saIter</code></td>
<td>
<p>Iteration number for SA (default = 10000)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_gapop">gaPop</code></td>
<td>
<p>Population size for GA (default = 400)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_gaiter">gaIter</code></td>
<td>
<p>Iteration number for GA (default = 200)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_gathread">gaThread</code></td>
<td>
<p>Number of threads to be used in GA (default = 5)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_gacrossover">gaCrossover</code></td>
<td>
<p>Applies crossover with the given probability in GA (default = 1, i.e. always perform crossover)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_gamut">gaMut</code></td>
<td>
<p>For GA, applies mutation with given mutation rate (default = 0, i.e. mutation off)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_grmaxdepth">grMaxDepth</code></td>
<td>
<p>Sets max depth in greedy search, 0 for no limit (default = 1)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_grsearchdepth">grSearchDepth</code></td>
<td>
<p>Search depth in greedy search (default = 1)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_groverlap">grOverlap</code></td>
<td>
<p>Overlap threshold for results of greedy search (default = 0.5)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_grsubnum">grSubNum</code></td>
<td>
<p>Number of subnetworks to be presented in the results (default = 1000)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_gset_list">gset_list</code></td>
<td>
<p>list for gene sets</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_adj_method">adj_method</code></td>
<td>
<p>correction method to be used for adjusting p-values.
(default = 'bonferroni')</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_enrichment_threshold">enrichment_threshold</code></td>
<td>
<p>adjusted-p value threshold used when filtering
enrichment results (default = 0.05)</p>
</td></tr>
<tr><td><code id="single_iter_wrapper_+3A_list_active_snw_genes">list_active_snw_genes</code></td>
<td>
<p>boolean value indicating whether or not to report
the non-significant active subnetwork genes for the active subnetwork which was enriched for
the given term with the lowest p value (default = <code>FALSE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame of enrichment results using active subnetwork search results
</p>

<hr>
<h2 id='summarize_enrichment_results'>Summarize Enrichment Results</h2><span id='topic+summarize_enrichment_results'></span>

<h3>Description</h3>

<p>Summarize Enrichment Results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarize_enrichment_results(enrichment_res, list_active_snw_genes = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summarize_enrichment_results_+3A_enrichment_res">enrichment_res</code></td>
<td>
<p>a dataframe of combined enrichment results. Columns are: </p>

<dl>
<dt>ID</dt><dd><p>ID of the enriched term</p>
</dd>
<dt>Term_Description</dt><dd><p>Description of the enriched term</p>
</dd>
<dt>Fold_Enrichment</dt><dd><p>Fold enrichment value for the enriched term</p>
</dd>
<dt>p_value</dt><dd><p>p value of enrichment</p>
</dd>
<dt>adj_p</dt><dd><p>adjusted p value of enrichment</p>
</dd>
<dt>non_Signif_Snw_Genes (OPTIONAL)</dt><dd><p>the non-significant active subnetwork genes, comma-separated</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="summarize_enrichment_results_+3A_list_active_snw_genes">list_active_snw_genes</code></td>
<td>
<p>boolean value indicating whether or not to report
the non-significant active subnetwork genes for the active subnetwork which was enriched for
the given term with the lowest p value (default = <code>FALSE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe of summarized enrichment results (over multiple iterations). Columns are: </p>

<dl>
<dt>ID</dt><dd><p>ID of the enriched term</p>
</dd>
<dt>Term_Description</dt><dd><p>Description of the enriched term</p>
</dd>
<dt>Fold_Enrichment</dt><dd><p>Fold enrichment value for the enriched term</p>
</dd>
<dt>occurrence</dt><dd><p>the number of iterations that the given term was found to enriched over all iterations</p>
</dd>
<dt>support</dt><dd><p>the median support (proportion of active subnetworks leading to enrichment within an iteration) over all iterations</p>
</dd>
<dt>lowest_p</dt><dd><p>the lowest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>highest_p</dt><dd><p>the highest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>non_Signif_Snw_Genes (OPTIONAL)</dt><dd><p>the non-significant active subnetwork genes, comma-separated</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
summarize_enrichment_results(enrichment_res)

## End(Not run)
</code></pre>

<hr>
<h2 id='term_gene_graph'>Create Term-Gene Graph</h2><span id='topic+term_gene_graph'></span>

<h3>Description</h3>

<p>Create Term-Gene Graph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>term_gene_graph(
  result_df,
  num_terms = 10,
  layout = "stress",
  use_description = FALSE,
  node_size = "num_genes",
  node_colors = c("#E5D7BF", "green", "red")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="term_gene_graph_+3A_result_df">result_df</code></td>
<td>
<p>A dataframe of pathfindR results that must contain the following columns: </p>

<dl>
<dt>Term_Description</dt><dd><p>Description of the enriched term (necessary if <code>use_description = TRUE</code>)</p>
</dd>
<dt>ID</dt><dd><p>ID of the enriched term (necessary if <code>use_description = FALSE</code>)</p>
</dd>
<dt>lowest_p</dt><dd><p>the lowest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>Up_regulated</dt><dd><p>the up-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>Down_regulated</dt><dd><p>the down-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="term_gene_graph_+3A_num_terms">num_terms</code></td>
<td>
<p>Number of top enriched terms to use while creating the graph. Set to <code>NULL</code> to use
all enriched terms (default = 10, i.e. top 10 terms)</p>
</td></tr>
<tr><td><code id="term_gene_graph_+3A_layout">layout</code></td>
<td>
<p>The type of layout to create (see <code><a href="ggraph.html#topic+ggraph">ggraph</a></code> for details. Default = 'stress')</p>
</td></tr>
<tr><td><code id="term_gene_graph_+3A_use_description">use_description</code></td>
<td>
<p>Boolean argument to indicate whether term descriptions
(in the 'Term_Description' column) should be used. (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="term_gene_graph_+3A_node_size">node_size</code></td>
<td>
<p>Argument to indicate whether to use number of significant genes ('num_genes')
or the -log10(lowest p value) ('p_val') for adjusting the node sizes (default = 'num_genes')</p>
</td></tr>
<tr><td><code id="term_gene_graph_+3A_node_colors">node_colors</code></td>
<td>
<p>vector of 3 colors to be used for coloring nodes (colors for term nodes, up, and down, respectively)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function (adapted from the Gene-Concept network visualization
by the R package <code>enrichplot</code>) can be utilized to visualize which input
genes are involved in the enriched terms as a graph. The term-gene graph
shows the links between genes and biological terms and allows for the
investigation of multiple terms to which significant genes are related. The
graph also enables determination of the overlap between the enriched terms
by identifying shared and distinct significant term-related genes.
</p>


<h3>Value</h3>

<p>a  <code><a href="ggraph.html#topic+ggraph">ggraph</a></code> object containing the term-gene graph.
Each node corresponds to an enriched term (beige), an up-regulated gene (green)
or a down-regulated gene (red). An edge between a term and a gene indicates
that the given term involves the gene. Size of a term node is proportional
to either the number of genes (if <code>node_size = 'num_genes'</code>) or
the -log10(lowest p value) (if <code>node_size = 'p_val'</code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p &lt;- term_gene_graph(example_pathfindR_output)
p &lt;- term_gene_graph(example_pathfindR_output, num_terms = 5)
p &lt;- term_gene_graph(example_pathfindR_output, node_size = 'p_val')
</code></pre>

<hr>
<h2 id='term_gene_heatmap'>Create Terms by Genes Heatmap</h2><span id='topic+term_gene_heatmap'></span>

<h3>Description</h3>

<p>Create Terms by Genes Heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>term_gene_heatmap(
  result_df,
  genes_df,
  num_terms = 10,
  use_description = FALSE,
  low = "red",
  mid = "black",
  high = "green",
  legend_title = "change",
  sort_terms_by_p = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="term_gene_heatmap_+3A_result_df">result_df</code></td>
<td>
<p>A dataframe of pathfindR results that must contain the following columns: </p>

<dl>
<dt>Term_Description</dt><dd><p>Description of the enriched term (necessary if <code>use_description = TRUE</code>)</p>
</dd>
<dt>ID</dt><dd><p>ID of the enriched term (necessary if <code>use_description = FALSE</code>)</p>
</dd>
<dt>lowest_p</dt><dd><p>the highest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>Up_regulated</dt><dd><p>the up-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>Down_regulated</dt><dd><p>the down-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="term_gene_heatmap_+3A_genes_df">genes_df</code></td>
<td>
<p>the input data that was used with <code><a href="#topic+run_pathfindR">run_pathfindR</a></code>.
It must be a data frame with 3 columns: </p>

<ol>
<li><p> Gene Symbol (Gene Symbol)
</p>
</li>
<li><p> Change value, e.g. log(fold change) (optional)
</p>
</li>
<li><p> p value, e.g. adjusted p value associated with differential expression
</p>
</li></ol>
<p> The change values in this data frame are used to color the affected genes</p>
</td></tr>
<tr><td><code id="term_gene_heatmap_+3A_num_terms">num_terms</code></td>
<td>
<p>Number of top enriched terms to use while creating the plot. Set to <code>NULL</code> to use
all enriched terms (default = 10)</p>
</td></tr>
<tr><td><code id="term_gene_heatmap_+3A_use_description">use_description</code></td>
<td>
<p>Boolean argument to indicate whether term descriptions
(in the 'Term_Description' column) should be used. (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="term_gene_heatmap_+3A_low">low</code></td>
<td>
<p>a string indicating the color of 'low' values in the coloring gradient (default = 'green')</p>
</td></tr>
<tr><td><code id="term_gene_heatmap_+3A_mid">mid</code></td>
<td>
<p>a string indicating the color of 'mid' values in the coloring gradient (default = 'black')</p>
</td></tr>
<tr><td><code id="term_gene_heatmap_+3A_high">high</code></td>
<td>
<p>a string indicating the color of 'high' values in the coloring gradient (default = 'red')</p>
</td></tr>
<tr><td><code id="term_gene_heatmap_+3A_legend_title">legend_title</code></td>
<td>
<p>legend title (default = 'change')</p>
</td></tr>
<tr><td><code id="term_gene_heatmap_+3A_sort_terms_by_p">sort_terms_by_p</code></td>
<td>
<p>boolean to indicate whether to sort terms by 'lowest_p'
(<code>TRUE</code>) or by number of genes (<code>FALSE</code>) (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="term_gene_heatmap_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="#topic+input_processing">input_processing</a></code> (used if
<code>genes_df</code> is provided)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot2 object of a heatmap where rows are enriched terms and
columns are involved input genes. If <code>genes_df</code> is provided, colors of
the tiles indicate the change values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>term_gene_heatmap(example_pathfindR_output, num_terms = 3)
</code></pre>

<hr>
<h2 id='UpSet_plot'>Create UpSet Plot of Enriched Terms</h2><span id='topic+UpSet_plot'></span>

<h3>Description</h3>

<p>Create UpSet Plot of Enriched Terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UpSet_plot(
  result_df,
  genes_df,
  num_terms = 10,
  method = "heatmap",
  use_description = FALSE,
  low = "red",
  mid = "black",
  high = "green",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="UpSet_plot_+3A_result_df">result_df</code></td>
<td>
<p>A dataframe of pathfindR results that must contain the following columns: </p>

<dl>
<dt>Term_Description</dt><dd><p>Description of the enriched term (necessary if <code>use_description = TRUE</code>)</p>
</dd>
<dt>ID</dt><dd><p>ID of the enriched term (necessary if <code>use_description = FALSE</code>)</p>
</dd>
<dt>lowest_p</dt><dd><p>the highest adjusted-p value of the given term over all iterations</p>
</dd>
<dt>Up_regulated</dt><dd><p>the up-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
<dt>Down_regulated</dt><dd><p>the down-regulated genes in the input involved in the given term's gene set, comma-separated</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="UpSet_plot_+3A_genes_df">genes_df</code></td>
<td>
<p>the input data that was used with <code><a href="#topic+run_pathfindR">run_pathfindR</a></code>.
It must be a data frame with 3 columns: </p>

<ol>
<li><p> Gene Symbol (Gene Symbol)
</p>
</li>
<li><p> Change value, e.g. log(fold change) (optional)
</p>
</li>
<li><p> p value, e.g. adjusted p value associated with differential expression
</p>
</li></ol>
<p> The change values in this data frame are used to color the affected genes</p>
</td></tr>
<tr><td><code id="UpSet_plot_+3A_num_terms">num_terms</code></td>
<td>
<p>Number of top enriched terms to use while creating the plot. Set to <code>NULL</code> to use
all enriched terms (default = 10)</p>
</td></tr>
<tr><td><code id="UpSet_plot_+3A_method">method</code></td>
<td>
<p>the option for producing the plot. Options include 'heatmap',
'boxplot' and 'barplot'. (default = 'heatmap')</p>
</td></tr>
<tr><td><code id="UpSet_plot_+3A_use_description">use_description</code></td>
<td>
<p>Boolean argument to indicate whether term descriptions
(in the 'Term_Description' column) should be used. (default = <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="UpSet_plot_+3A_low">low</code></td>
<td>
<p>a string indicating the color of 'low' values in the coloring gradient (default = 'green')</p>
</td></tr>
<tr><td><code id="UpSet_plot_+3A_mid">mid</code></td>
<td>
<p>a string indicating the color of 'mid' values in the coloring gradient (default = 'black')</p>
</td></tr>
<tr><td><code id="UpSet_plot_+3A_high">high</code></td>
<td>
<p>a string indicating the color of 'high' values in the coloring gradient (default = 'red')</p>
</td></tr>
<tr><td><code id="UpSet_plot_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="#topic+input_processing">input_processing</a></code> (used if
<code>genes_df</code> is provided)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>UpSet plots are plots of the intersections of sets as a matrix. This
function creates a ggplot object of an UpSet plot where the x-axis is the
UpSet plot of intersections of enriched terms. By default (i.e.
<code>method = 'heatmap'</code>) the main plot is a heatmap of genes at the
corresponding intersections, colored by up/down regulation (if
<code>genes_df</code> is provided, colored by change values). If
<code>method = 'barplot'</code>, the main plot is bar plots of the number of genes
at the corresponding intersections. Finally, if <code>method = 'boxplot'</code> and
if <code>genes_df</code> is provided, then the main plot displays the boxplots of
change values of the genes at the corresponding intersections.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>UpSet_plot(example_pathfindR_output)
</code></pre>

<hr>
<h2 id='visualize_active_subnetworks'>Visualize Active Subnetworks</h2><span id='topic+visualize_active_subnetworks'></span>

<h3>Description</h3>

<p>Visualize Active Subnetworks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_active_subnetworks(
  active_snw_path,
  genes_df,
  pin_name_path = "Biogrid",
  num_snws,
  layout = "stress",
  score_quan_thr = 0.8,
  sig_gene_thr = 0.02,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_active_subnetworks_+3A_active_snw_path">active_snw_path</code></td>
<td>
<p>path to the output of an Active Subnetwork Search</p>
</td></tr>
<tr><td><code id="visualize_active_subnetworks_+3A_genes_df">genes_df</code></td>
<td>
<p>the input data that was used with <code><a href="#topic+run_pathfindR">run_pathfindR</a></code>.
It must be a data frame with 3 columns: </p>

<ol>
<li><p> Gene Symbol (Gene Symbol)
</p>
</li>
<li><p> Change value, e.g. log(fold change) (optional)
</p>
</li>
<li><p> p value, e.g. adjusted p value associated with differential expression
</p>
</li></ol>
<p> The change values in this data frame are used to color the affected genes</p>
</td></tr>
<tr><td><code id="visualize_active_subnetworks_+3A_pin_name_path">pin_name_path</code></td>
<td>
<p>Name of the chosen PIN or absolute/path/to/PIN.sif. If PIN name,
must be one of c('Biogrid', 'STRING', 'GeneMania', 'IntAct', 'KEGG', 'mmu_STRING'). If
path/to/PIN.sif, the file must comply with the PIN specifications. (Default = 'Biogrid')</p>
</td></tr>
<tr><td><code id="visualize_active_subnetworks_+3A_num_snws">num_snws</code></td>
<td>
<p>number of top subnetworks to be visualized (leave blank if
you want to visualize all subnetworks)</p>
</td></tr>
<tr><td><code id="visualize_active_subnetworks_+3A_layout">layout</code></td>
<td>
<p>The type of layout to create (see <code><a href="ggraph.html#topic+ggraph">ggraph</a></code> for details. Default = 'stress')</p>
</td></tr>
<tr><td><code id="visualize_active_subnetworks_+3A_score_quan_thr">score_quan_thr</code></td>
<td>
<p>active subnetwork score quantile threshold. Must be
between 0 and 1 or set to -1 for not filtering. (Default = 0.8)</p>
</td></tr>
<tr><td><code id="visualize_active_subnetworks_+3A_sig_gene_thr">sig_gene_thr</code></td>
<td>
<p>threshold for the minimum proportion of significant genes in
the subnetwork (Default = 0.02) If the number of genes to use as threshold is
calculated to be &lt; 2 (e.g. 50 signif. genes x 0.01 = 0.5), the threshold number
is set to 2</p>
</td></tr>
<tr><td><code id="visualize_active_subnetworks_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="#topic+input_processing">input_processing</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of ggplot objects of graph visualizations of identified active
subnetworks. Green nodes are down-regulated genes, reds are up-regulated genes
and yellows are non-input genes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>path2snw_list &lt;- system.file(
  'extdata/resultActiveSubnetworkSearch.txt',
  package = 'pathfindR'
)
# visualize top 2 active subnetworks
g_list &lt;- visualize_active_subnetworks(
  active_snw_path = path2snw_list,
  genes_df = example_pathfindR_input[1:10, ],
  pin_name_path = 'KEGG',
  num_snws = 2
)
</code></pre>

<hr>
<h2 id='visualize_KEGG_diagram'>Visualize Human KEGG Pathways</h2><span id='topic+visualize_KEGG_diagram'></span>

<h3>Description</h3>

<p>Visualize Human KEGG Pathways
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_KEGG_diagram(
  kegg_pw_ids,
  input_processed,
  scale_vals = TRUE,
  node_cols = NULL,
  legend.position = "top"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_KEGG_diagram_+3A_kegg_pw_ids">kegg_pw_ids</code></td>
<td>
<p>KEGG ids of pathways to be colored and visualized</p>
</td></tr>
<tr><td><code id="visualize_KEGG_diagram_+3A_input_processed">input_processed</code></td>
<td>
<p>input data processed via <code><a href="#topic+input_processing">input_processing</a></code></p>
</td></tr>
<tr><td><code id="visualize_KEGG_diagram_+3A_scale_vals">scale_vals</code></td>
<td>
<p>should change values be scaled? (default = <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="visualize_KEGG_diagram_+3A_node_cols">node_cols</code></td>
<td>
<p>low, middle and high color values for coloring the pathway nodes
(default = <code>NULL</code>). If <code>node_cols=NULL</code>, the low, middle and high color
are set as 'green', 'gray' and 'red'. If all change values are 1e6 (in case no
changes are supplied, this dummy value is assigned by
<code><a href="#topic+input_processing">input_processing</a></code>), only one color ('#F38F18' if NULL) is used.</p>
</td></tr>
<tr><td><code id="visualize_KEGG_diagram_+3A_legend.position">legend.position</code></td>
<td>
<p>the default position of legends (&quot;none&quot;, &quot;left&quot;,
&quot;right&quot;, &quot;bottom&quot;, &quot;top&quot;, &quot;inside&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Creates colored visualizations of the enriched human KEGG pathways
and returns them as a list of ggplot objects, named by Term ID.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+visualize_terms">visualize_terms</a></code> for the wrapper function for
creating enriched term diagrams. See <code><a href="#topic+run_pathfindR">run_pathfindR</a></code> for the
wrapper function of the pathfindR enrichment workflow.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
input_processed &lt;- data.frame(
  GENE = c("PKLR", "GPI", "CREB1", "INS"),
  CHANGE = c(1.5, -2, 3, 5)
)
gg_list &lt;- visualize_KEGG_diagram(c("hsa00010", "hsa04911"), input_processed)

## End(Not run)
</code></pre>

<hr>
<h2 id='visualize_term_interactions'>Visualize Interactions of Genes Involved in the Given Enriched Terms</h2><span id='topic+visualize_term_interactions'></span>

<h3>Description</h3>

<p>Visualize Interactions of Genes Involved in the Given Enriched Terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_term_interactions(result_df, pin_name_path, show_legend = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_term_interactions_+3A_result_df">result_df</code></td>
<td>
<p>Data frame of enrichment results. Must-have columns
are: 'Term_Description', 'Up_regulated' and 'Down_regulated'</p>
</td></tr>
<tr><td><code id="visualize_term_interactions_+3A_pin_name_path">pin_name_path</code></td>
<td>
<p>Name of the chosen PIN or absolute/path/to/PIN.sif. If PIN name,
must be one of c('Biogrid', 'STRING', 'GeneMania', 'IntAct', 'KEGG', 'mmu_STRING'). If
path/to/PIN.sif, the file must comply with the PIN specifications. (Default = 'Biogrid')</p>
</td></tr>
<tr><td><code id="visualize_term_interactions_+3A_show_legend">show_legend</code></td>
<td>
<p>Boolean to indicate whether to display the legend (<code>TRUE</code>)
or not (<code>FALSE</code>) (default: <code>TRUE</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The following steps are performed for the visualization of interactions
of genes involved for each enriched term: </p>

<ol>
<li><p> shortest paths between all affected genes are determined (via <code><a href="igraph.html#topic+igraph">igraph</a></code>)
</p>
</li>
<li><p> the nodes of all shortest paths are merged
</p>
</li>
<li><p> the PIN is subsetted using the merged nodes (genes)
</p>
</li>
<li><p> using the PIN subset, the graph showing the interactions is generated
</p>
</li>
<li><p> the final graph is visualized using <code><a href="igraph.html#topic+igraph">igraph</a></code>, colored by changed
status (if provided)
</p>
</li></ol>



<h3>Value</h3>

<p>list of ggplot objects (named by Term ID) visualizing the interactions of genes involved
in the given enriched terms (annotated in the <code>result_df</code>) in the PIN used
for enrichment analysis (specified by <code>pin_name_path</code>).
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+visualize_terms">visualize_terms</a></code> for the wrapper function
for creating enriched term diagrams. See <code><a href="#topic+run_pathfindR">run_pathfindR</a></code> for the
wrapper function of the pathfindR enrichment workflow.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
result_df &lt;- example_pathfindR_output[1:2, ]
gg_list &lt;- visualize_term_interactions(result_df, pin_name_path = 'IntAct')

## End(Not run)
</code></pre>

<hr>
<h2 id='visualize_terms'>Create Diagrams for Enriched Terms</h2><span id='topic+visualize_terms'></span>

<h3>Description</h3>

<p>Create Diagrams for Enriched Terms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualize_terms(
  result_df,
  input_processed = NULL,
  is_KEGG_result = TRUE,
  pin_name_path = "Biogrid",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualize_terms_+3A_result_df">result_df</code></td>
<td>
<p>Data frame of enrichment results. Must-have columns for
KEGG human pathway diagrams (<code>is_KEGG_result = TRUE</code>) are: 'ID' and 'Term_Description'.
Must-have columns for the rest are: 'Term_Description', 'Up_regulated' and
'Down_regulated'</p>
</td></tr>
<tr><td><code id="visualize_terms_+3A_input_processed">input_processed</code></td>
<td>
<p>input data processed via <code><a href="#topic+input_processing">input_processing</a></code>,
not necessary when <code>is_KEGG_result = FALSE</code></p>
</td></tr>
<tr><td><code id="visualize_terms_+3A_is_kegg_result">is_KEGG_result</code></td>
<td>
<p>boolean to indicate whether KEGG gene sets were used for
enrichment analysis or not (default = <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="visualize_terms_+3A_pin_name_path">pin_name_path</code></td>
<td>
<p>Name of the chosen PIN or absolute/path/to/PIN.sif. If PIN name,
must be one of c('Biogrid', 'STRING', 'GeneMania', 'IntAct', 'KEGG', 'mmu_STRING'). If
path/to/PIN.sif, the file must comply with the PIN specifications. (Default = 'Biogrid')</p>
</td></tr>
<tr><td><code id="visualize_terms_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="#topic+visualize_KEGG_diagram">visualize_KEGG_diagram</a></code> (used
when <code>is_KEGG_result = TRUE</code>) or <code><a href="#topic+visualize_term_interactions">visualize_term_interactions</a></code>
(used when <code>is_KEGG_result = FALSE</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code>is_KEGG_result = TRUE</code>, KEGG pathway diagrams are created,
affected nodes colored by up/down regulation status.
For other gene sets, interactions of affected genes are determined (via a shortest-path
algorithm) and are visualized (colored by change status) using igraph.
</p>


<h3>Value</h3>

<p>Depending on the argument <code>is_KEGG_result</code>, creates visualization of
interactions of genes involved in the list of enriched terms in
<code>result_df</code>. Returns a list of ggplot objects named by Term ID.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+visualize_KEGG_diagram">visualize_KEGG_diagram</a></code> for the visualization function
of KEGG diagrams. See <code><a href="#topic+visualize_term_interactions">visualize_term_interactions</a></code> for the
visualization function that generates diagrams showing the interactions of
input genes in the PIN. See <code><a href="#topic+run_pathfindR">run_pathfindR</a></code> for the wrapper
function of the pathfindR workflow.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
input_processed &lt;- data.frame(
  GENE = c("PARP1", "NDUFA1", "STX6", "SNAP23"),
  CHANGE = c(1.5, -2, 3, 5)
)
result_df &lt;- example_pathfindR_output[1:2, ]

gg_list &lt;- visualize_terms(result_df, input_processed)
gg_list2 &lt;- visualize_terms(result_df, is_KEGG_result = FALSE, pin_name_path = 'IntAct')

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
