<!DOCTYPE html><html lang="en-US"><head><title>Help for package pathwayTMB</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pathwayTMB}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#final_character'><p>final_character, the example's final signature</p></a></li>
<li><a href='#gene_path'><p>gene_path, the pathways geneset</p></a></li>
<li><a href='#GenePathwayOncoplots'><p>draw an GenePathwayOncoplots</p></a></li>
<li><a href='#genesmbol'><p>genesmbol, the coding genes' length</p></a></li>
<li><a href='#get_final_signature'><p>Filter cancer-specific dysfunction pathways.</p></a></li>
<li><a href='#get_mut_matrix'><p>Converts MAF into mutation matrix.</p></a></li>
<li><a href='#get_PTMB'><p>Calculate the Pathway-based Tumor Mutational Burden.</p></a></li>
<li><a href='#mut_matrix'><p>mut_matrix, mutations matrix</p></a></li>
<li><a href='#plotKMcurves'><p>Drawing Kaplan Meier Survival Curves Using the final survival-related PTMB.</p></a></li>
<li><a href='#plotMutInteract'><p>Exact tests to detect mutually exclusive, co-occuring and altered genesets or pathways.</p></a></li>
<li><a href='#plotROC'><p>plot the ROC curve</p></a></li>
<li><a href='#PTMB_matrix'><p>PTMB_matrix, the Pathway-based Tumor Mutational Burden matrix</p></a></li>
<li><a href='#sur'><p>sur, the samples' survival data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Pathway Based Tumor Mutational Burden</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A systematic bioinformatics tool to develop a new pathway-based gene panel for tumor mutational burden (TMB) assessment (pathway-based tumor mutational burden, PTMB), using somatic mutations files in an efficient manner from either The Cancer Genome Atlas sources or any in-house studies as long as the data is in mutation annotation file (MAF) format. Besides, we develop a multiple machine learning method using the sample's PTMB profiles to identify cancer-specific dysfunction pathways, which can be a biomarker of prognostic and predictive for cancer immunotherapy.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>BiocGenerics, purrr, utils, glmnet, randomForest, stats,
survival, survminer, caret, data.table, RColorBrewer,
grDevices, pROC, graphics, maftools, clusterProfiler</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr,rmarkdown,qpdf</td>
</tr>
<tr>
<td>Suggests:</td>
<td>stringi, knitr, rmarkdown, testthat, BiocManager, xfun,
e1071, qpdf, tinytex, spelling</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-01 13:54:24 UTC; DELL</td>
</tr>
<tr>
<td>Author:</td>
<td>Junwei Han [aut, cre, cph],
  Xiangmei Li [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-09 13:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='final_character'>final_character, the example's final signature</h2><span id='topic+final_character'></span>

<h3>Description</h3>

<p>final_character, a potential marker for cancer prognostic and immunotherapy, generated by 'get_final_signature'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>final_character
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 2.
</p>

<hr>
<h2 id='gene_path'>gene_path, the pathways geneset</h2><span id='topic+gene_path'></span>

<h3>Description</h3>

<p>gene_path, a list of KEGG Non-metabolic pathways geneset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_path
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 27.
</p>

<hr>
<h2 id='GenePathwayOncoplots'>draw an GenePathwayOncoplots</h2><span id='topic+GenePathwayOncoplots'></span>

<h3>Description</h3>

<p>takes output generated by read.maf and draws an GenePathwayOncoplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenePathwayOncoplots(
  maffile,
  gene_path,
  freq_matrix,
  risk_score,
  cut_off,
  final_character,
  isTCGA = FALSE,
  top = 20,
  clinicalFeatures = "sample_group",
  annotationColor = c("red", "green"),
  sortByAnnotation = TRUE,
  removeNonMutated = FALSE,
  drawRowBar = TRUE,
  drawColBar = TRUE,
  leftBarData = NULL,
  leftBarLims = NULL,
  rightBarData = NULL,
  rightBarLims = NULL,
  topBarData = NULL,
  logColBar = FALSE,
  draw_titv = FALSE,
  showTumorSampleBarcodes = FALSE,
  fill = TRUE,
  showTitle = TRUE,
  titleText = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GenePathwayOncoplots_+3A_maffile">maffile</code></td>
<td>
<p>an MAF object generated by read.maf.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_gene_path">gene_path</code></td>
<td>
<p>User input pathways geneset list.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_freq_matrix">freq_matrix</code></td>
<td>
<p>The mutations matrix,generated by 'get_mut_matrix'.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_risk_score">risk_score</code></td>
<td>
<p>Samples' PTMB-related risk score,which could be a biomarker for survival analysis and immunotherapy prediction.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_cut_off">cut_off</code></td>
<td>
<p>A threshold value(the median risk score as the default value).Using this value to divide the sample into high and low risk groups with different overall survival.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_final_character">final_character</code></td>
<td>
<p>The pathway signature,use to map gene in the GenePathwayOncoplots.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_istcga">isTCGA</code></td>
<td>
<p>Is input MAF file from TCGA source. If TRUE uses only first 12 characters from Tumor_Sample_Barcode.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_top">top</code></td>
<td>
<p>how many top genes to be drawn,genes are arranged from high to low depending on the frequency of mutations. defaults to 20.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_clinicalfeatures">clinicalFeatures</code></td>
<td>
<p>columns names from 'clinical.data' slot of MAF to be drawn in the plot. Dafault &quot;sample_group&quot;.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_annotationcolor">annotationColor</code></td>
<td>
<p>Custom colors to use for sample annotation-&quot;sample_group&quot;. Must be a named list containing a named vector of colors. Default &quot;red&quot; and &quot;green&quot;.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_sortbyannotation">sortByAnnotation</code></td>
<td>
<p>logical sort oncomatrix (samples) by provided 'clinicalFeatures'. Sorts based on first 'clinicalFeatures'. Defaults to TRUE. column-sort.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_removenonmutated">removeNonMutated</code></td>
<td>
<p>Logical. If TRUE removes samples with no mutations in the GenePathwayOncoplots for better visualization. Default FALSE.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_drawrowbar">drawRowBar</code></td>
<td>
<p>logical. Plots righ barplot for each gene. Default TRUE.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_drawcolbar">drawColBar</code></td>
<td>
<p>logical plots top barplot for each sample. Default TRUE.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_leftbardata">leftBarData</code></td>
<td>
<p>Data for leftside barplot. Must be a data.frame with two columns containing gene names and values. Default 'NULL'.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_leftbarlims">leftBarLims</code></td>
<td>
<p>limits for 'leftBarData'. Default 'NULL'.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_rightbardata">rightBarData</code></td>
<td>
<p>Data for rightside barplot. Must be a data.frame with two columns containing to gene names and values. Default 'NULL' which draws distibution by variant classification. This option is applicable when only 'drawRowBar' is TRUE.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_rightbarlims">rightBarLims</code></td>
<td>
<p>limits for 'rightBarData'. Default 'NULL'.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_topbardata">topBarData</code></td>
<td>
<p>Default 'NULL' which draws absolute number of mutation load for each sample. Can be overridden by choosing one clinical indicator(Numeric) or by providing a two column data.frame contaning sample names and values for each sample. This option is applicable when only 'drawColBar' is TRUE.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_logcolbar">logColBar</code></td>
<td>
<p>Plot top bar plot on log10 scale. Default FALSE.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_draw_titv">draw_titv</code></td>
<td>
<p>logical Includes TiTv plot. Default FALSE</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_showtumorsamplebarcodes">showTumorSampleBarcodes</code></td>
<td>
<p>logical to include sample names.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_fill">fill</code></td>
<td>
<p>Logical. If TRUE draws genes and samples as blank grids even when they are not altered.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_showtitle">showTitle</code></td>
<td>
<p>Default TRUE.</p>
</td></tr>
<tr><td><code id="GenePathwayOncoplots_+3A_titletext">titleText</code></td>
<td>
<p>Custom title. Default 'NULL'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get the path of the mutation annotation file and samples' survival data
maf&lt;-system.file("extdata","data_mutations_extended.txt",package = "pathwayTMB")
sur_path&lt;-system.file("extdata","sur.csv",package = "pathwayTMB")
sur&lt;-read.csv(sur_path,header=TRUE,row.names = 1)
#perform the function 'get_mut_matrix'
mut_matrix&lt;-get_mut_matrix(maffile=maf,mut_fre = 0.01,is.TCGA=FALSE,sur=sur)
#perform the function `get_PTMB`
PTMB_matrix&lt;-get_PTMB(freq_matrix=mut_matrix,genesmbol=genesmbol,gene_path=gene_path)
set.seed(1)
final_character&lt;-get_final_signature(PTMB=PTMB_matrix,sur=sur)
#calculate the risksciore
riskscore&lt;-plotKMcurves(t(PTMB_matrix[final_character,]),sur=sur,plots=FALSE)$risk_score
cut&lt;-median(riskscore)
GenePathwayOncoplots(maf,gene_path,mut_matrix,riskscore,cut,final_character)

</code></pre>

<hr>
<h2 id='genesmbol'>genesmbol, the coding genes' length</h2><span id='topic+genesmbol'></span>

<h3>Description</h3>

<p>genesmbol,a list of coding genes' length, generated by 'get_gene_length'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genesmbol
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 34931.
</p>

<hr>
<h2 id='get_final_signature'>Filter cancer-specific dysfunction pathways.</h2><span id='topic+get_final_signature'></span>

<h3>Description</h3>

<p>The function 'get_final_signature' , using to filter cancer-specific dysfunction pathways (a potential marker for cancer prognostic and immunotherapy), is the main function of our analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_final_signature(PTMB, sur, pval_cutoff = 0.01)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_final_signature_+3A_ptmb">PTMB</code></td>
<td>
<p>The pathway tumor mutation burden matrix,generated by'get_PTMB'.</p>
</td></tr>
<tr><td><code id="get_final_signature_+3A_sur">sur</code></td>
<td>
<p>A nx2 data frame of samples' survival data,the first line is samples' survival event and the second line is samples' overall survival.</p>
</td></tr>
<tr><td><code id="get_final_signature_+3A_pval_cutoff">pval_cutoff</code></td>
<td>
<p>A threshold value (0.01 as the default value) to identify the differential PTMB pathway.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return the final PTMB signature,could be a potential marker for prognostic and immunotherapy prediction.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get the path of the mutation annotation file and samples' survival data
maf&lt;-system.file("extdata","data_mutations_extended.txt",package = "pathwayTMB")
sur_path&lt;-system.file("extdata","sur.csv",package = "pathwayTMB")
sur&lt;-read.csv(sur_path,header=TRUE,row.names = 1)
#perform the function 'get_mut_matrix'
mut_matrix&lt;-get_mut_matrix(maffile=maf,mut_fre = 0.01,is.TCGA=FALSE,sur=sur)
#perform the function `get_PTMB`
PTMB_matrix&lt;-get_PTMB(freq_matrix=mut_matrix,genesmbol=genesmbol,gene_path=gene_path)
set.seed(1)
final_character&lt;-get_final_signature(PTMB=PTMB_matrix,sur=sur)
</code></pre>

<hr>
<h2 id='get_mut_matrix'>Converts MAF into mutation matrix.</h2><span id='topic+get_mut_matrix'></span>

<h3>Description</h3>

<p>The function 'get_mut_matrix' converts mutation annotation file (MAF) format data into a mutations matrix.Then use the fisher exact test to select the geneset with higher mutation frequency in alive sample group.Finally return the higher mutation frequency matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mut_matrix(
  maffile,
  is.TCGA = TRUE,
  mut_fre = 0,
  nonsynonymous = TRUE,
  cut_Cox.pval = 1,
  cut_HR = 1,
  sur
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mut_matrix_+3A_maffile">maffile</code></td>
<td>
<p>Input mutation annotation file (MAF) format data. It must be an absolute path or the name relatived to the current working directory.</p>
</td></tr>
<tr><td><code id="get_mut_matrix_+3A_is.tcga">is.TCGA</code></td>
<td>
<p>Is input MAF file from TCGA source. If TRUE uses only first 15 characters from Tumor_Sample_Barcode.</p>
</td></tr>
<tr><td><code id="get_mut_matrix_+3A_mut_fre">mut_fre</code></td>
<td>
<p>A threshold value(zero as the default value). The genes with a given mutation frequency equal or greater than the threshold value are retained for the following analysis.</p>
</td></tr>
<tr><td><code id="get_mut_matrix_+3A_nonsynonymous">nonsynonymous</code></td>
<td>
<p>Logical,tell if extract the non-synonymous somatic mutations (nonsense mutation, missense mutation, frame-shif indels, splice site, nonstop mutation, translation start site, inframe indels).</p>
</td></tr>
<tr><td><code id="get_mut_matrix_+3A_cut_cox.pval">cut_Cox.pval</code></td>
<td>
<p>The significant cut_off pvalue for the univariate Cox regression.</p>
</td></tr>
<tr><td><code id="get_mut_matrix_+3A_cut_hr">cut_HR</code></td>
<td>
<p>The cut_off HR for the univariate Cox regression, uses to select the genes with survival benefit mutations.</p>
</td></tr>
<tr><td><code id="get_mut_matrix_+3A_sur">sur</code></td>
<td>
<p>A nx2 data frame of samples' survival data,the first line is samples' survival event and the second line is samples' overall survival.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The survival-related mutations matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get the path of the mutation annotation file and samples' survival data
maf&lt;-system.file("extdata","data_mutations_extended.txt",package = "pathwayTMB")
sur_path&lt;-system.file("extdata","sur.csv",package = "pathwayTMB")
sur&lt;-read.csv(sur_path,header=TRUE,row.names = 1)
#perform the function 'get_mut_matrix'
mut_matrix&lt;-get_mut_matrix(maffile=maf,mut_fre = 0.01,is.TCGA=FALSE,sur=sur)
</code></pre>

<hr>
<h2 id='get_PTMB'>Calculate the Pathway-based Tumor Mutational Burden.</h2><span id='topic+get_PTMB'></span>

<h3>Description</h3>

<p>The function 'get_PTMB' uses to calculate the Pathway-based Tumor Mutational Burden (PTMB). PTMB is defined as pathway-based tumor mutational burden corrected by genes’ length and number.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_PTMB(freq_matrix, genesmbol, path_mut_cutoff = 0, gene_path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_PTMB_+3A_freq_matrix">freq_matrix</code></td>
<td>
<p>The mutations matrix,generated by 'get_mut_matrix'.</p>
</td></tr>
<tr><td><code id="get_PTMB_+3A_genesmbol">genesmbol</code></td>
<td>
<p>The genes' length matrix,generated by 'get_gene_length'.</p>
</td></tr>
<tr><td><code id="get_PTMB_+3A_path_mut_cutoff">path_mut_cutoff</code></td>
<td>
<p>A threshold value(zero percent as the default value).Pathways with a given mutation frequency equal or greater than the threshold value are retained for the following analysis.</p>
</td></tr>
<tr><td><code id="get_PTMB_+3A_gene_path">gene_path</code></td>
<td>
<p>User input pathways geneset list.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return the Pathway-based Tumor Mutational Burden matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get the path of the mutation annotation file and samples' survival data
maf&lt;-system.file("extdata","data_mutations_extended.txt",package = "pathwayTMB")
sur_path&lt;-system.file("extdata","sur.csv",package = "pathwayTMB")
sur&lt;-read.csv(sur_path,header=TRUE,row.names = 1)
#perform the function 'get_mut_matrix'
mut_matrix&lt;-get_mut_matrix(maffile=maf,mut_fre = 0.01,is.TCGA=FALSE,sur=sur)
#perform the function `get_PTMB`
PTMB_matrix&lt;-get_PTMB(freq_matrix=mut_matrix,genesmbol=genesmbol,gene_path=gene_path)
</code></pre>

<hr>
<h2 id='mut_matrix'>mut_matrix, mutations matrix</h2><span id='topic+mut_matrix'></span>

<h3>Description</h3>

<p>mut_matrix, the mutations matrix,generated by 'get_mut_matrix'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mut_matrix
</code></pre>


<h3>Format</h3>

<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 673 rows and 35 columns.
</p>

<hr>
<h2 id='plotKMcurves'>Drawing Kaplan Meier Survival Curves Using the final survival-related PTMB.</h2><span id='topic+plotKMcurves'></span>

<h3>Description</h3>

<p>The function 'plotKMcurves' uses to draw Kaplan-Meier Survival Curves based on PTMB-related riskscore.The riskscore is generated by the signature's PTMB and the coefficient of &quot;Univariate&quot; or &quot;Multivariate&quot; cox regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotKMcurves(
  sig_PTMB,
  sur,
  method = "Multivariate",
  returnAll = TRUE,
  pval = TRUE,
  color = NULL,
  plots = TRUE,
  palette = NULL,
  linetype = 1,
  conf.int = FALSE,
  pval.method = FALSE,
  test.for.trend = FALSE,
  surv.median.line = "none",
  risk.table = FALSE,
  cumevents = FALSE,
  cumcensor = FALSE,
  tables.height = 0.25,
  add.all = FALSE,
  ggtheme = theme_survminer()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotKMcurves_+3A_sig_ptmb">sig_PTMB</code></td>
<td>
<p>The signature's PTMB matrix,which rows are samples and columns are pathways.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_sur">sur</code></td>
<td>
<p>A nx2 data frame of samples' survival data,the first line is samples' survival event and the second line is samples' overall survival.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_method">method</code></td>
<td>
<p>Method must be one of &quot;Univariate&quot; and &quot;Multivariate&quot;.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_returnall">returnAll</code></td>
<td>
<p>Logicalvalue.Default is TRUE. If TRUE, return the riskscore and the coefficient of cox regression.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_pval">pval</code></td>
<td>
<p>Logical value, a numeric or a string. If logical and TRUE, the p-value is added on the plot. If numeric, than the computet p-value is substituted with the one passed with this parameter. If character, then the customized string appears on the plot.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_color">color</code></td>
<td>
<p>Color to be used for the survival curves.If the number of strata/group (n.strata) = 1, the expected value is the color name. For example color = &quot;blue&quot;.If n.strata &gt; 1, the expected value is the grouping variable name. By default, survival curves are colored by strata using the argument color = &quot;strata&quot;, but you can also color survival curves by any other grouping variables used to fit the survival curves. In this case, it's possible to specify a custom color palette by using the argument palette.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_plots">plots</code></td>
<td>
<p>logical value.Default is TRUE.If TRUE,plot the Kaplan Meier Survival Curves.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_palette">palette</code></td>
<td>
<p>the color palette to be used. Allowed values include &quot;hue&quot; for the default hue color scale; &quot;grey&quot; for grey color palettes; brewer palettes e.g. &quot;RdBu&quot;, &quot;Blues&quot;, ...; or custom color palette e.g. c(&quot;blue&quot;, &quot;red&quot;); and scientific journal palettes from ggsci R package, e.g.: &quot;npg&quot;, &quot;aaas&quot;, &quot;lancet&quot;, &quot;jco&quot;, &quot;ucscgb&quot;, &quot;uchicago&quot;, &quot;simpsons&quot; and &quot;rickandmorty&quot;. See details section for more information. Can be also a numeric vector of length(groups); in this case a basic color palette is created using the function palette.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_linetype">linetype</code></td>
<td>
<p>line types. Allowed values includes i) &quot;strata&quot; for changing linetypes by strata (i.e. groups); ii) a numeric vector (e.g., c(1, 2)) or a character vector c(&quot;solid&quot;, &quot;dashed&quot;).</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_conf.int">conf.int</code></td>
<td>
<p>logical value. If TRUE, plots confidence interval.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_pval.method">pval.method</code></td>
<td>
<p>whether to add a text with the test name used for calculating the pvalue, that corresponds to survival curves' comparison - used only when pval=TRUE</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_test.for.trend">test.for.trend</code></td>
<td>
<p>logical value. Default is FALSE. If TRUE, returns the test for trend p-values. Tests for trend are designed to detect ordered differences in survival curves. That is, for at least one group. The test for trend can be only performed when the number of groups is &gt; 2.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_surv.median.line">surv.median.line</code></td>
<td>
<p>character vector for drawing a horizontal/vertical line at median survival. Allowed values include one of c(&quot;none&quot;, &quot;hv&quot;, &quot;h&quot;, &quot;v&quot;). v: vertical, h:horizontal.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_risk.table">risk.table</code></td>
<td>
<p>Allowed values include:(1)TRUE or FALSE specifying whether to show or not the risk table. Default is FALSE.(2)&quot;absolute&quot; or &quot;percentage&quot;. Shows the absolute number and the percentage of subjects at risk by time, respectively.(3)&quot;abs_pct&quot; to show both absolute number and percentage.(4)&quot;nrisk_cumcensor&quot; and &quot;nrisk_cumevents&quot;. Show the number at risk and, the cumulative number of censoring and events, respectively.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_cumevents">cumevents</code></td>
<td>
<p>logical value specifying whether to show or not the table of the cumulative number of events. Default is FALSE.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_cumcensor">cumcensor</code></td>
<td>
<p>logical value specifying whether to show or not the table of the cumulative number of censoring. Default is FALSE.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_tables.height">tables.height</code></td>
<td>
<p>numeric value (in [0 - 1]) specifying the general height of all tables under the main survival plot.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_add.all">add.all</code></td>
<td>
<p>a logical value. If TRUE, add the survival curve of pooled patients (null model) onto the main plot.</p>
</td></tr>
<tr><td><code id="plotKMcurves_+3A_ggtheme">ggtheme</code></td>
<td>
<p>function, ggplot2 theme name. Default value is theme_survminer. Allowed values include ggplot2 official themes: see theme.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a list of riskscore and coefficient of cox regression.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get the path of the mutation annotation file and samples' survival data
maf&lt;-system.file("extdata","data_mutations_extended.txt",package = "pathwayTMB")
sur_path&lt;-system.file("extdata","sur.csv",package = "pathwayTMB")
sur&lt;-read.csv(sur_path,header=TRUE,row.names = 1)
#perform the function 'get_mut_matrix'
mut_matrix&lt;-get_mut_matrix(maffile=maf,mut_fre = 0.01,is.TCGA=FALSE,sur=sur)
#perform the function `get_PTMB`
PTMB_matrix&lt;-get_PTMB(freq_matrix=mut_matrix,genesmbol=genesmbol,gene_path=gene_path)
set.seed(1)
final_character&lt;-get_final_signature(PTMB=PTMB_matrix,sur=sur)
#plot the K-M survival curve
plotKMcurves(t(PTMB_matrix[final_character,]),sur=sur,risk.table = TRUE)

</code></pre>

<hr>
<h2 id='plotMutInteract'>Exact tests to detect mutually exclusive, co-occuring and altered genesets or pathways.</h2><span id='topic+plotMutInteract'></span>

<h3>Description</h3>

<p>Performs Pair-wise Fisher's Exact test to detect mutually exclusive or co-occuring events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotMutInteract(
  freq_matrix,
  genes,
  pvalue = c(0.05, 0.01),
  returnAll = TRUE,
  fontSize = 0.8,
  showSigSymbols = TRUE,
  showCounts = FALSE,
  countStats = "all",
  countType = "all",
  countsFontSize = 0.8,
  countsFontColor = "black",
  colPal = "BrBG",
  nShiftSymbols = 5,
  sigSymbolsSize = 2,
  sigSymbolsFontSize = 0.9,
  pvSymbols = c(46, 42),
  limitColorBreaks = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotMutInteract_+3A_freq_matrix">freq_matrix</code></td>
<td>
<p>The mutations matrix,generated by 'get_mut_matrix'.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_genes">genes</code></td>
<td>
<p>List of genes or pathways among which interactions should be tested.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_pvalue">pvalue</code></td>
<td>
<p>Default c(0.05, 0.01) p-value threshold. You can provide two values for upper and lower threshold.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_returnall">returnAll</code></td>
<td>
<p>If TRUE returns test statistics for all pair of tested genes. Default FALSE, returns for only genes below pvalue threshold.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_fontsize">fontSize</code></td>
<td>
<p>cex for gene names. Default 0.8.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_showsigsymbols">showSigSymbols</code></td>
<td>
<p>Default TRUE. Heighlight significant pairs.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_showcounts">showCounts</code></td>
<td>
<p>Default TRUE. Include number of events in the plot.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_countstats">countStats</code></td>
<td>
<p>Default 'all'. Can be 'all' or 'sig'.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_counttype">countType</code></td>
<td>
<p>Default 'cooccur'. Can be 'all', 'cooccur', 'mutexcl'.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_countsfontsize">countsFontSize</code></td>
<td>
<p>Default 0.8.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_countsfontcolor">countsFontColor</code></td>
<td>
<p>Default 'black'.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_colpal">colPal</code></td>
<td>
<p>colPalBrewer palettes. See RColorBrewer::display.brewer.all() for details.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_nshiftsymbols">nShiftSymbols</code></td>
<td>
<p>shift if positive shift SigSymbols by n to the left, default = 5.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_sigsymbolssize">sigSymbolsSize</code></td>
<td>
<p>size of symbols in the matrix and in legend.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_sigsymbolsfontsize">sigSymbolsFontSize</code></td>
<td>
<p>size of font in legends.</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_pvsymbols">pvSymbols</code></td>
<td>
<p>vector of pch numbers for symbols of p-value for upper and lower thresholds c(upper, lower).</p>
</td></tr>
<tr><td><code id="plotMutInteract_+3A_limitcolorbreaks">limitColorBreaks</code></td>
<td>
<p>limit color to extreme values. Default TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of data.tables
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get the path of the mutation annotation file and samples' survival data
maf&lt;-system.file("extdata","data_mutations_extended.txt",package = "pathwayTMB")
sur_path&lt;-system.file("extdata","sur.csv",package = "pathwayTMB")
sur&lt;-read.csv(sur_path,header=TRUE,row.names = 1)
#perform the function 'get_mut_matrix'
mut_matrix&lt;-get_mut_matrix(maffile=maf,mut_fre = 0.01,is.TCGA=FALSE,sur=sur)
#perform the function `get_PTMB`
PTMB_matrix&lt;-get_PTMB(freq_matrix=mut_matrix,genesmbol=genesmbol,gene_path=gene_path)
set.seed(1)
final_character&lt;-get_final_signature(PTMB=PTMB_matrix,sur=sur)
plotMutInteract(freq_matrix=PTMB_matrix, genes=final_character,nShiftSymbols =0.3)
</code></pre>

<hr>
<h2 id='plotROC'>plot the ROC curve</h2><span id='topic+plotROC'></span>

<h3>Description</h3>

<p>This function uses to plot a ROC curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotROC(
  riskscore,
  response,
  main,
  add = FALSE,
  col = par("col"),
  legacy.axes = TRUE,
  print.auc = FALSE,
  grid = FALSE,
  auc.polygon = FALSE,
  auc.polygon.col = "skyblue",
  max.auc.polygon = FALSE,
  max.auc.polygon.col = "#EEEEEE"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotROC_+3A_riskscore">riskscore</code></td>
<td>
<p>a numeric vector of the same length than response, containing the predicted value of each observation.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_response">response</code></td>
<td>
<p>a factor, numeric or character vector of responses (true class), typically encoded with 0 (controls) and 1 (cases). Only two classes can be used in a ROC curve.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_main">main</code></td>
<td>
<p>the title of the ROC curve</p>
</td></tr>
<tr><td><code id="plotROC_+3A_add">add</code></td>
<td>
<p>if TRUE, the ROC curve will be added to an existing plot. If FALSE (default), a new plot will be created.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_col">col</code></td>
<td>
<p>the color of the ROC curve</p>
</td></tr>
<tr><td><code id="plotROC_+3A_legacy.axes">legacy.axes</code></td>
<td>
<p>a logical indicating if the specificity axis (x axis) must be plotted as as decreasing “specificity” (FALSE) or increasing “1 - specificity” (TRUE, the default) as in most legacy software. This affects only the axis, not the plot coordinates.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_print.auc">print.auc</code></td>
<td>
<p>boolean. Should the numeric value of AUC be printed on the plot?</p>
</td></tr>
<tr><td><code id="plotROC_+3A_grid">grid</code></td>
<td>
<p>boolean or numeric vector of length 1 or 2. Should a background grid be added to the plot? Numeric: show a grid with the specified interval between each line; Logical: show the grid or not. Length 1: same values are taken for horizontal and vertical lines. Length 2: grid value for vertical (grid[1]) and horizontal (grid[2]). Note that these values are used to compute grid.v and grid.h. Therefore if you specify a grid.h and grid.v, it will be ignored.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_auc.polygon">auc.polygon</code></td>
<td>
<p>boolean. Whether or not to display the area as a polygon.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_auc.polygon.col">auc.polygon.col</code></td>
<td>
<p>color (col) for the AUC polygon.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_max.auc.polygon">max.auc.polygon</code></td>
<td>
<p>boolean. Whether or not to display the maximal possible area as a polygon.</p>
</td></tr>
<tr><td><code id="plotROC_+3A_max.auc.polygon.col">max.auc.polygon.col</code></td>
<td>
<p>color (col) for the maximum AUC polygon.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#get the path of the mutation annotation file and samples' survival data
maf&lt;-system.file("extdata","data_mutations_extended.txt",package = "pathwayTMB")
sur_path&lt;-system.file("extdata","sur.csv",package = "pathwayTMB")
sur&lt;-read.csv(sur_path,header=TRUE,row.names = 1)
#perform the function 'get_mut_matrix'
mut_matrix&lt;-get_mut_matrix(maffile=maf,mut_fre = 0.01,is.TCGA=FALSE,sur=sur)
#perform the function `get_PTMB`
PTMB_matrix&lt;-get_PTMB(freq_matrix=mut_matrix,genesmbol=genesmbol,gene_path=gene_path)
set.seed(1)
final_character&lt;-get_final_signature(PTMB=PTMB_matrix,sur=sur)
#calculate the risksciore
riskscore&lt;-plotKMcurves(t(PTMB_matrix[final_character,]),sur=sur,plots=FALSE)$risk_score
#get the path of samples' immunotherapy response data
res_path&lt;- system.file("extdata","response.csv",package = "pathwayTMB")
response&lt;-read.csv(res_path,header=TRUE,stringsAsFactors =FALSE,row.name=1)
plotROC(riskscore=riskscore,response=response,main="Objective Response",print.auc=TRUE)

</code></pre>

<hr>
<h2 id='PTMB_matrix'>PTMB_matrix, the Pathway-based Tumor Mutational Burden matrix</h2><span id='topic+PTMB_matrix'></span>

<h3>Description</h3>

<p>PTMB_matrix, the pathway tumor mutation burden matrix,generated by'get_PTMB'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PTMB_matrix
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 27 rows and 35 columns.
</p>

<hr>
<h2 id='sur'>sur, the samples' survival data</h2><span id='topic+sur'></span>

<h3>Description</h3>

<p>sur, a nx2 data frame, the samples' survival data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sur
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 110 rows and 2 columns.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
