<!DOCTYPE html><html><head><title>Help for package pavo</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pavo}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#pavo-package'><p>pavo: Perceptual Analysis, Visualization and Organization of Spectral Colour Data</p></a></li>
<li><a href='#adjacent'><p>Run an adjacency and boundary strength analysis</p></a></li>
<li><a href='#aggplot'><p>Plot aggregated reflectance spectra</p></a></li>
<li><a href='#aggspec'><p>Aggregate reflectance spectra</p></a></li>
<li><a href='#as.rimg'><p>Convert data to an rimg object</p></a></li>
<li><a href='#as.rspec'><p>Convert data to an rspec object</p></a></li>
<li><a href='#axistetra'><p>Plot reference axes in a static tetrahedral colourspace</p></a></li>
<li><a href='#bgandilum'><p>Default background and illuminant data</p></a></li>
<li><a href='#bootcoldist'><p>Bootstrap colour distance confidence intervals</p></a></li>
<li><a href='#categorical'><p>Categorical fly-visual model</p></a></li>
<li><a href='#catplot'><p>Plot the categorical colour vision model</p></a></li>
<li><a href='#cie'><p>CIE colour spaces</p></a></li>
<li><a href='#cieplot'><p>CIE plot</p></a></li>
<li><a href='#classify'><p>Identify colour classes in an image for adjacency analyses</p></a></li>
<li><a href='#coc'><p>Color opponent coding model</p></a></li>
<li><a href='#cocplot'><p>Plot the colour opponent coding diagram</p></a></li>
<li><a href='#coldist'><p>Colour distances</p></a></li>
<li><a href='#coldist2mat'><p>Convert coldist to distance matrix</p></a></li>
<li><a href='#colspace'><p>Model spectra in a colorspace</p></a></li>
<li><a href='#diplot'><p>Plot a dichromat segment</p></a></li>
<li><a href='#dispace'><p>Dichromatic colour space</p></a></li>
<li><a href='#explorespec'><p>Plot spectral curves</p></a></li>
<li><a href='#find_astar'><p>Compute the <code class="reqn">\alpha^*</code> value</p></a></li>
<li><a href='#flowers'><p>Reflectance spectra from a suite of native Australian flowers, collected</p>
around Cairns, Queensland.</a></li>
<li><a href='#getimg'><p>Import image data</p></a></li>
<li><a href='#getspec'><p>Import spectra files</p></a></li>
<li><a href='#hexagon'><p>Colour hexagon</p></a></li>
<li><a href='#hexplot'><p>Plot a colour hexagon</p></a></li>
<li><a href='#img_conversion'><p>Convert images between class rimg and cimg or magick-image</p></a></li>
<li><a href='#irrad2flux'><p>Converts between irradiance and photon (quantum) flux</p></a></li>
<li><a href='#is.colspace'><p>Test if object is of class 'colspace'</p></a></li>
<li><a href='#is.vismodel'><p>Test if object is of class 'vismodel'</p></a></li>
<li><a href='#jnd2xyz'><p>Convert JND distances into perceptually-corrected Cartesian coordinates</p></a></li>
<li><a href='#jndplot'><p>Perceptually-corrected chromaticity diagrams</p></a></li>
<li><a href='#jndrot'><p>Rotate Cartesian coordinates obtained from <code>jnd2xyz()</code></p></a></li>
<li><a href='#legendtetra'><p>Add legend to a static tetrahedral colourspace</p></a></li>
<li><a href='#merge.rspec'><p>Merge two rspec objects</p></a></li>
<li><a href='#peakshape'><p>Peak shape descriptors</p></a></li>
<li><a href='#plot.colspace'><p>Plot spectra in a colourspace</p></a></li>
<li><a href='#plot.rimg'><p>Plot unprocessed or colour-classified images</p></a></li>
<li><a href='#plot.rspec'><p>Plot spectra</p></a></li>
<li><a href='#plot.sensmod'><p>Plot absorbance spectra from <code>sensmodel()</code></p></a></li>
<li><a href='#plotsmooth'><p>Plot loess smoothed curves</p></a></li>
<li><a href='#points.colspace'><p>Plot points in a colourspace</p></a></li>
<li><a href='#procimg'><p>Process images</p></a></li>
<li><a href='#procspec'><p>Process spectra</p></a></li>
<li><a href='#projplot'><p>2D projection of a tetrahedral colourspace</p></a></li>
<li><a href='#segplot'><p>Plot the segment-analysis model</p></a></li>
<li><a href='#segspace'><p>Segment classification</p></a></li>
<li><a href='#sensdata'><p>Retrieve or plot in-built spectral sensitivity data</p></a></li>
<li><a href='#sensmodel'><p>Modeling spectral sensitivity</p></a></li>
<li><a href='#sicalis'><p>Spectral curves from three body regions of stripe-tailed yellow finch</p>
(<em>Sicalis citrina</em>) males</a></li>
<li><a href='#simulate_spec'><p>Simulate a spectrum</p></a></li>
<li><a href='#spec2rgb'><p>Spectrum to rgb colour conversion</p></a></li>
<li><a href='#subset.rspec'><p>Subset rspec, vismodel, and colspace objects</p></a></li>
<li><a href='#summary.colspace'><p>Colourspace data summary</p></a></li>
<li><a href='#summary.rimg'><p>Image summary</p></a></li>
<li><a href='#summary.rspec'><p>Colourimetric variables</p></a></li>
<li><a href='#summary.vismodel'><p>Visual model summary</p></a></li>
<li><a href='#tcspace'><p>Tetrahedral colourspace</p></a></li>
<li><a href='#tcsplot'><p>Interactive plot of a tetrahedral colourspace</p></a></li>
<li><a href='#teal'><p>Angle-resolved reflectance data for the iridescent wing patch of a male green-winged</p>
teal (<em>Anas carolinensis</em>)</a></li>
<li><a href='#tetraplot'><p>Plot a static tetrahedral colorspace</p></a></li>
<li><a href='#transmissiondata'><p>Default ocular transmission data</p></a></li>
<li><a href='#triplot'><p>Plot a Maxwell triangle</p></a></li>
<li><a href='#trispace'><p>Trichromatic colour space</p></a></li>
<li><a href='#ttvertex'><p>vertex for the tetrahedral color space</p></a></li>
<li><a href='#vismodel'><p>Visual models</p></a></li>
<li><a href='#vissyst'><p>Animal visual systems data</p></a></li>
<li><a href='#vol'><p>Plot a tetrahedral colour space</p></a></li>
<li><a href='#voloverlap'><p>Colour volume overlap</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Perceptual Analysis, Visualization and Organization of Spectral
Colour Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.9.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A cohesive framework for the spectral and spatial analysis of 
    colour described in Maia, Eliason, Bitton, Doucet &amp; Shawkey (2013) 
    &lt;<a href="https://doi.org/10.1111%2F2041-210X.12069">doi:10.1111/2041-210X.12069</a>&gt; and Maia, Gruson, Endler &amp; White (2019)
    &lt;<a href="https://doi.org/10.1111%2F2041-210X.13174">doi:10.1111/2041-210X.13174</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://pavo.colrverse.com">http://pavo.colrverse.com</a>, <a href="https://github.com/rmaia/pavo/">https://github.com/rmaia/pavo/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/rmaia/pavo/issues">https://github.com/rmaia/pavo/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>cluster, future.apply, geometry(&ge; 0.4.0), lightr(&ge; 1.0),
magick, farver, plot3D, progressr, sf, viridisLite</td>
</tr>
<tr>
<td>Suggests:</td>
<td>alphashape3d, imager, knitr, mapproj, rgl, rmarkdown,
testthat(&ge; 2.99.0), vdiffr, digest</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/Needs/website:</td>
<td>pkgdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-24 09:43:04 UTC; twhite</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas White <a href="https://orcid.org/0000-0002-3976-1734"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Rafael Maia <a href="https://orcid.org/0000-0002-7563-9795"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Hugo Gruson <a href="https://orcid.org/0000-0002-4094-1476"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  John Endler [aut],
  Chad Eliason [aut],
  Pierre-Paul Bitton
    <a href="https://orcid.org/0000-0001-5984-2331"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomas White &lt;thomas.white026@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-24 10:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='pavo-package'>pavo: Perceptual Analysis, Visualization and Organization of Spectral Colour Data</h2><span id='topic+pavo'></span><span id='topic+pavo-package'></span>

<h3>Description</h3>

<p>A cohesive framework for the spectral and spatial analysis of colour described in Maia, Eliason, Bitton, Doucet &amp; Shawkey (2013) <a href="https://doi.org/10.1111/2041-210X.12069">doi:10.1111/2041-210X.12069</a> and Maia, Gruson, Endler &amp; White (2019) <a href="https://doi.org/10.1111/2041-210X.13174">doi:10.1111/2041-210X.13174</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a> (<a href="https://orcid.org/0000-0002-3976-1734">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Rafael Maia <a href="mailto:rafa.maia@gmail.com">rafa.maia@gmail.com</a> (<a href="https://orcid.org/0000-0002-7563-9795">ORCID</a>)
</p>
</li>
<li><p> Hugo Gruson <a href="mailto:hugo.gruson+R@normalesup.org">hugo.gruson+R@normalesup.org</a> (<a href="https://orcid.org/0000-0002-4094-1476">ORCID</a>)
</p>
</li>
<li><p> John Endler <a href="mailto:john.endler@deakin.edu.au">john.endler@deakin.edu.au</a>
</p>
</li>
<li><p> Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>
</li>
<li><p> Pierre-Paul Bitton <a href="mailto:bittonp@uwindsor.ca">bittonp@uwindsor.ca</a> (<a href="https://orcid.org/0000-0001-5984-2331">ORCID</a>)
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="http://pavo.colrverse.com">http://pavo.colrverse.com</a>
</p>
</li>
<li> <p><a href="https://github.com/rmaia/pavo/">https://github.com/rmaia/pavo/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/rmaia/pavo/issues">https://github.com/rmaia/pavo/issues</a>
</p>
</li></ul>


<hr>
<h2 id='adjacent'>Run an adjacency and boundary strength analysis</h2><span id='topic+adjacent'></span>

<h3>Description</h3>

<p>Calculate summary variables from the adjacency (Endler 2012) and
boundary-strength (Endler et al. 2018) analyses, along with overall pattern
contrast (Endler &amp; Mielke 2005).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjacent(
  classimg,
  xpts = NULL,
  xscale = NULL,
  bkgID = NULL,
  polygon = NULL,
  exclude = c("none", "background", "object"),
  coldists = NULL,
  hsl = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adjacent_+3A_classimg">classimg</code></td>
<td>
<p>(required) an xyz matrix, or list of matrices, in which x and
y correspond to spatial (e.g. pixel) coordinates, and z is a numeric code
specifying a colour-class. Preferably the result of <code><a href="#topic+classify">classify()</a></code>, or
constructed from grid-sampled spectra that have been visually modelled and
clustered (as per Endler 2012).</p>
</td></tr>
<tr><td><code id="adjacent_+3A_xpts">xpts</code></td>
<td>
<p>an integer specifying the number of sample points along the x
axis, from which the evenly-spaced sampling grid is constructed (if
required). Defaults to the smallest dimension of <code>classimg</code>, though this
should be carefully considered.</p>
</td></tr>
<tr><td><code id="adjacent_+3A_xscale">xscale</code></td>
<td>
<p>(required) an integer or list of integers equal in length to classimg()
specifying the true length of the x-axis, in preferred units. Not required,
and ignored, only if image scales have been set via <code><a href="#topic+procimg">procimg()</a></code>.</p>
</td></tr>
<tr><td><code id="adjacent_+3A_bkgid">bkgID</code></td>
<td>
<p>an integer or vector specifying the colour-class ID number(s) of
pertaining to the background alone, for relatively homogeneous and
uniquely-identified backgrounds (e.g. the matte background of pinned
specimens). Examine the attributes of, or call <code>summary</code> on, the result of
<code><a href="#topic+classify">classify()</a></code> to visualise the RGB values corresponding to colour-class ID
numbers for classified images. Ignored if the focal object and background
has been identified using <code><a href="#topic+procimg">procimg()</a></code>.</p>
</td></tr>
<tr><td><code id="adjacent_+3A_polygon">polygon</code></td>
<td>
<p>a data.frame of x-y coordinates delineating a closed polygon
that separates the focal object from the background. Not required, and
ignored, if the focal object outline is specified using <code><a href="#topic+procimg">procimg()</a></code>.</p>
</td></tr>
<tr><td><code id="adjacent_+3A_exclude">exclude</code></td>
<td>
<p>the portion of the scene to be excluded from the analysis, if
any.
</p>

<ul>
<li> <p><code>'none'</code>: default
</p>
</li>
<li> <p><code>'background'</code>: exclude everything <em>outside</em> the closed polygon specified
using <code><a href="#topic+procimg">procimg()</a></code>, or the argument <code>polygon</code>. Alternatively, if the
background is relatively homogeneous the colour-class ID(s) uniquely
corresponding to the background can be specified via <code>bkgID</code>, and
subsequently excluded.
</p>
</li>
<li> <p><code>'object'</code>: exclude everything <em>inside</em> the closed polygon specified
using <code><a href="#topic+procimg">procimg()</a></code>, or the argument <code>polygon</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="adjacent_+3A_coldists">coldists</code></td>
<td>
<p>a data.frame specifying the visually-modelled chromatic (dS)
and/or achromatic (dL) distances between colour-categories. The first two
columns should be named 'c1' and 'c2', and specify all possible
combinations of numeric colour-class ID's (viewable by calling
<code>summary(image, plot = TRUE)</code> on a colour classified image), with the
remaining columns named dS (for chromatic distances) and/or dL (for
achromatic distances). See <code><a href="#topic+vismodel">vismodel()</a></code> and <code><a href="#topic+colspace">colspace()</a></code> for visual
modelling with spectral data.</p>
</td></tr>
<tr><td><code id="adjacent_+3A_hsl">hsl</code></td>
<td>
<p>data.frame specifying the hue, saturation, and luminance of color
patch elements, as might be estimated via <code><a href="#topic+vismodel">vismodel()</a></code> and <code><a href="#topic+colspace">colspace()</a></code>.
The first column, named 'patch', should contain numeric color category IDs,
with the remaining columns specifying one or more of 'hue' (angle, in
radians), 'sat', and/or 'lum'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can customise the type of parallel processing used by this function with
the <code><a href="future.html#topic+plan">future::plan()</a></code> function. This works on all operating systems, as well
as high performance computing (HPC) environment. Similarly, you can customise
the way progress is shown with the <code><a href="progressr.html#topic+handlers">progressr::handlers()</a></code> functions
(progress bar, acoustic feedback, nothing, etc.)
</p>


<h3>Value</h3>

<p>a data frame of summary variables:
</p>

<ul>
<li> <p><code>'k'</code>: The number of user-specified colour and/or luminance classes.
</p>
</li>
<li> <p><code>'N'</code>: The grand total (sum of diagonal and off-diagonal) transitions.
</p>
</li>
<li> <p><code>'n_off'</code>: The total off-diagonal transitions.
</p>
</li>
<li> <p><code>'p_i'</code>: The overall frequency of colour class <em>i</em>.
</p>
</li>
<li> <p><code>'q_i_j'</code>: The frequency of transitions between <em>all</em> colour classes <em>i</em>
and <em>j</em>, such that <code>sum(q_i_j) = 1</code>.
</p>
</li>
<li> <p><code>'t_i_j'</code>: The frequency of off-diagonal (i.e. class-change transitions)
transitions <em>i</em> and <em>j</em>, such that <code>sum(t_i_j) = 1</code>.
</p>
</li>
<li> <p><code>'m'</code>: The overall transition density (mean transitions), in units
specified in the argument <code>xscale</code>.
</p>
</li>
<li> <p><code>'m_r'</code>: The row-wise transition density (mean row transitions), in
user-specified units.
</p>
</li>
<li> <p><code>'m_c'</code>: The column-wise transition density (mean column transitions), in
user-specified units.
</p>
</li>
<li> <p><code>'A'</code>: The transition aspect ratio (&lt; 1 = wide, &gt; 1 = tall).
</p>
</li>
<li> <p><code>'Sc'</code>: Simpson colour class diversity, <code>Sc = 1/(sum(p_i^2))</code>. If all
colour and luminance classes are equal in relative area, then <code>Sc = k</code>.
</p>
</li>
<li> <p><code>'St'</code>: Simpson transition diversity, <code>St = 1/sum(t_i_j^2)</code>.
</p>
</li>
<li> <p><code>'Jc'</code>: Simpson colour class diversity relative to its achievable maximum.
<code>Jc = Sc/k</code>.
</p>
</li>
<li> <p><code>'Jt'</code>: Simpson transition diversity relative to its achievable maximum.
<code>Jt = St/(k*(k-1)/2)</code>.
</p>
</li>
<li> <p><code>'B'</code>: The animal/background transition ratio, or the ratio of class-change
transitions entirely within the focal object and those involving the object
and background,
<code>B = sum(O_a_a / O_a_b)</code>.
</p>
</li>
<li> <p><code>'Rt'</code>: Ratio of animal-animal and animal-background transition
diversities, <code>Rt = St_a_a / St_a_b</code>.
</p>
</li>
<li> <p><code>'Rab'</code>: Ratio of animal-animal and background-background transition
diversities, <code>Rt = St_a_a / St_b_b</code>.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;'m_dS', 's_dS', 'cv_dS'&#8288;</code>: weighted mean, sd, and coefficient of variation
of the chromatic boundary strength.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;'m_dL', 's_dL', 'cv_dL'&#8288;</code>: weighted mean, sd, and coefficient of variation
of the achromatic boundary strength.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;'m_hue', 's_hue', 'var_hue'&#8288;</code>: circular mean, sd, and variance of overall
pattern hue (in radians).
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;'m_sat', 's_sat', 'cv_sat'&#8288;</code>: weighted mean, sd, and coefficient variation
of overall pattern saturation.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;'m_lum', 's_lum', 'cv_lum'&#8288;</code>: weighted mean, sd, and coefficient variation
of overall pattern luminance.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Endler, J. A. (2012). A framework for analysing colour pattern
geometry: adjacent colours. Biological Journal Of The Linnean Society,
107(2), 233-253.
</p>
<p>Endler, J. A., Cole G., Kranz A. (2018). Boundary Strength
Analysis: Combining color pattern geometry and coloured patch visual
properties for use in predicting behaviour and fitness. Methods in Ecology
and Evolution, 9(12), 2334-2348.
</p>
<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour
patterns  as birds see them. Biological Journal Of The Linnean Society,
86(4), 405-431.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+classify">classify()</a></code>, <code><a href="#topic+summary.rimg">summary.rimg()</a></code>, <code><a href="#topic+procimg">procimg()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Set a seed, for reproducibility
set.seed(153)

# Run the adjacency analysis on a single image of a butterfly
papilio &lt;- getimg(system.file("testdata/images/butterflies/papilio.png", package = "pavo"))
papilio_class &lt;- classify(papilio, kcols = 4)
papilio_adj &lt;- adjacent(papilio_class, xscale = 100)

# Expand on the above, by including (fake) color distances and hsl values
# of colour elements in the image

# Generate fake color distances
distances &lt;- data.frame(
  c1 = c(1, 1, 1, 2, 2, 3),
  c2 = c(2, 3, 4, 3, 4, 4),
  dS = c(5.3, 3.5, 5.7, 2.9, 6.1, 3.2),
  dL = c(5.5, 6.6, 3.3, 2.2, 4.4, 6.6)
)

# Generate some fake hue, saturation, luminance values
hsl_vals &lt;- data.frame(
  patch = seq_len(4),
  hue = c(1.5, 2.2, 1.0, 0.5),
  lum = c(10, 5, 7, 3),
  sat = c(3.5, 1.1, 6.3, 1.3)
)

# Run the full analysis, including the white background's ID
papilio_adj &lt;- adjacent(papilio_class,
  xscale = 100, bkgID = 1,
  coldists = distances, hsl = hsl_vals
)

# Run an adjacency analysis on multiple images.
# First load some images of coral snake colour patterns
snakes &lt;- getimg(system.file("testdata/images/snakes", package = "pavo"))

# Automatically colour-classify the coral snake patterns
snakes_class &lt;- classify(snakes, kcols = 3)

# Run the adjacency analysis, with varying real-world scales for each image
snakes_adj &lt;- adjacent(snakes_class, xpts = 120, xscale = c(50, 55))

</code></pre>

<hr>
<h2 id='aggplot'>Plot aggregated reflectance spectra</h2><span id='topic+aggplot'></span>

<h3>Description</h3>

<p>Combines and plots spectra (by taking the average and the standard deviation,
for example) according to an index or a vector of identities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aggplot(
  rspecdata,
  by = NULL,
  FUN.center = mean,
  FUN.error = sd,
  lcol = NULL,
  shadecol = NULL,
  alpha = 0.2,
  legend = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aggplot_+3A_rspecdata">rspecdata</code></td>
<td>
<p>(required) a data frame, possibly of class <code>rspec</code>, which
contains a column containing a wavelength range, named 'wl', and spectra
data in remaining columns.</p>
</td></tr>
<tr><td><code id="aggplot_+3A_by">by</code></td>
<td>
<p>(required) either a single value specifying the range of spectra
within the data frame to be combined (for example, <code>by</code> = 3 indicates the
function will be applied to groups of 3 consecutive columns in the spectra
data frame); a vector containing identifications for the columns in the
spectra data frame (in which case the function will be applied to each
group of spectra sharing the same identification); or a list of vectors,
e.g., <code>by = list(sex, species)</code>.</p>
</td></tr>
<tr><td><code id="aggplot_+3A_fun.center">FUN.center</code></td>
<td>
<p>the function to be applied to the groups of spectra,
calculating a measure of central tendency (defaults to <code><a href="base.html#topic+mean">base::mean()</a></code>).</p>
</td></tr>
<tr><td><code id="aggplot_+3A_fun.error">FUN.error</code></td>
<td>
<p>the function to be applied to the groups of spectra,
calculating a measure of variation (defaults to <code><a href="stats.html#topic+sd">stats::sd()</a></code>).</p>
</td></tr>
<tr><td><code id="aggplot_+3A_lcol">lcol</code></td>
<td>
<p>colour of plotted lines indicating central tendency.</p>
</td></tr>
<tr><td><code id="aggplot_+3A_shadecol">shadecol</code></td>
<td>
<p>colour of shaded areas indicating variance measure.</p>
</td></tr>
<tr><td><code id="aggplot_+3A_alpha">alpha</code></td>
<td>
<p>transparency of the shaded areas.</p>
</td></tr>
<tr><td><code id="aggplot_+3A_legend">legend</code></td>
<td>
<p>automatically add a legend.</p>
</td></tr>
<tr><td><code id="aggplot_+3A_...">...</code></td>
<td>
<p>additional graphical parameters to be passed to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot containing the lines and shaded areas of the groups of spectra.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>
<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Montgomerie R (2006) Analyzing colors. In: Hill G, McGraw K (eds)
Bird coloration. Harvard University Press, Cambridge, pp 90-147.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load reflectance data
data(sicalis)

# Create grouping variable based on spec names
bysic &lt;- gsub("^ind[0-9].", "", names(sicalis)[-1])

# Plot using various error functions and options
aggplot(sicalis, bysic)
aggplot(sicalis, bysic, FUN.error = function(x) quantile(x, c(0.0275, 0.975)))
aggplot(sicalis, bysic, shadecol = spec2rgb(sicalis), lcol = 1)
aggplot(sicalis, bysic, lcol = 1, FUN.error = function(x) sd(x) / sqrt(length(x)))

</code></pre>

<hr>
<h2 id='aggspec'>Aggregate reflectance spectra</h2><span id='topic+aggspec'></span>

<h3>Description</h3>

<p>Combines spectra (by taking the average, for example) according to an index
or a vector of identities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aggspec(rspecdata, by = NULL, FUN = mean, trim = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aggspec_+3A_rspecdata">rspecdata</code></td>
<td>
<p>(required) a data frame, possibly of class <code>rspec</code>, which
contains a column containing a wavelength range, named 'wl', and spectra
data in remaining columns.</p>
</td></tr>
<tr><td><code id="aggspec_+3A_by">by</code></td>
<td>
<p>(required) either a single value specifying the range of spectra
within the data frame to be combined (for example, <code>by</code> = 3 indicates the
function will be applied to groups of 3 consecutive columns in the spectra
data frame); a vector containing identifications for the columns in the
spectra data frame (in which case the function will be applied to each
group of spectra sharing the same identification); or a list of vectors,
e.g., <code>by = list(sex, species)</code>.</p>
</td></tr>
<tr><td><code id="aggspec_+3A_fun">FUN</code></td>
<td>
<p>the function to be applied to the groups of spectra. (defaults to
<code><a href="base.html#topic+mean">mean()</a></code>)</p>
</td></tr>
<tr><td><code id="aggspec_+3A_trim">trim</code></td>
<td>
<p>logical. if <code>TRUE</code> (default), the function will try to identify
and remove numbers at the end of the names of the columns in the new rspec
object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of class <code>rspec</code> containing the spectra after applying
the aggregating function.
</p>


<h3>Author(s)</h3>

<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Montgomerie R (2006) Analyzing colors. In: Hill G, McGraw K (eds)
Bird coloration. Harvard University Press, Cambridge, pp 90-147.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(teal)

# Average every two spectra
teal.sset1 &lt;- aggspec(teal, by = 2)
plot(teal.sset1)

# Create factor and average spectra by levels 'a' and 'b'
ind &lt;- rep(c("a", "b"), times = 6)
teal.sset2 &lt;- aggspec(teal, by = ind)

plot(teal.sset2)
</code></pre>

<hr>
<h2 id='as.rimg'>Convert data to an rimg object</h2><span id='topic+as.rimg'></span><span id='topic+as.rimg.default'></span><span id='topic+as.rimg.cimg'></span><span id='topic+is.rimg'></span>

<h3>Description</h3>

<p>Converts an array of RGB values, a <code>cimg</code> object, or a <code>magick-image</code> object,
to an <code>rimg</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.rimg(object, name = "img")

## Default S3 method:
as.rimg(object, name = "img")

## S3 method for class 'cimg'
as.rimg(object, name = "img")

is.rimg(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.rimg_+3A_object">object</code></td>
<td>
<p>(required) a three-dimensional array containing RGB values.</p>
</td></tr>
<tr><td><code id="as.rimg_+3A_name">name</code></td>
<td>
<p>the name(s) of the image(s).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code>rimg</code> for use in further <code>pavo</code> functions
</p>
<p>a logical value indicating whether the object is of class <code>rimg</code>
</p>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Hugo Gruson <a href="mailto:hugo.gruson+R@normalesup.org">hugo.gruson+R@normalesup.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate some fake image data
fake &lt;- array(c(
  as.matrix(rep(c(0.2, 0.4, 0.6), each = 250)),
  as.matrix(rep(c(0.4, 0.7, 0.8), each = 250)),
  as.matrix(rep(c(0.6, 0.1, 0.2), each = 250))
),
dim = c(750, 750, 3)
)

# Inspect it
head(fake)

# Determine if is an rimg object
is.rimg(fake)

# Convert to rimg object and check again
fake2 &lt;- as.rimg(fake)
is.rimg(fake2)

</code></pre>

<hr>
<h2 id='as.rspec'>Convert data to an rspec object</h2><span id='topic+as.rspec'></span><span id='topic+is.rspec'></span>

<h3>Description</h3>

<p>Converts data frames or matrices containing spectral data to <code>rspec</code>
object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.rspec(
  object,
  whichwl = NULL,
  interp = TRUE,
  lim = NULL,
  exceed.range = TRUE
)

is.rspec(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.rspec_+3A_object">object</code></td>
<td>
<p>(required) a data frame or matrix containing spectra to
process.</p>
</td></tr>
<tr><td><code id="as.rspec_+3A_whichwl">whichwl</code></td>
<td>
<p>a numeric or character vector specifying which column contains
wavelengths. If <code>NULL</code> (default), function searches for column containing
equally spaced numbers and sets it as wavelengths &quot;wl&quot;. If no wavelengths
are found or <code>whichwl</code> is not given, returns arbitrary index values.</p>
</td></tr>
<tr><td><code id="as.rspec_+3A_interp">interp</code></td>
<td>
<p>whether to interpolate wavelengths in 1-nm bins (defaults to
<code>TRUE</code>). It is rarely recommended to turn off this option, as
uninterpolated spectra are incompatible with some downstream analyses,
including notably colour vision models.</p>
</td></tr>
<tr><td><code id="as.rspec_+3A_lim">lim</code></td>
<td>
<p>vector specifying wavelength range to interpolate over (e.g.
<code>c(300, 700)</code>).</p>
</td></tr>
<tr><td><code id="as.rspec_+3A_exceed.range">exceed.range</code></td>
<td>
<p>logical. Should data be interpolated to the limits
specified by <code>lim</code> if <code>lim</code> exceeds the range of the actual data? Useful,
and relatively safe, when the data range falls slightly within <code>lim</code> (e.g.
300.1 - 699 nm), but will produce spurious results if <code>lim</code> far exceeds the
range of input data. Defaults to <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code>rspec</code> for use in further <code>pavo</code> functions
</p>
<p>a logical value indicating whether the object is of class <code>rspec</code>
</p>


<h3>Author(s)</h3>

<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Generate some fake reflectance data
fakedat &lt;- data.frame(wl = 300:700, refl1 = rnorm(401), refl2 = rnorm(401))
head(fakedat)

# Determine if is rspec object
is.rspec(fakedat)

# Convert to rspec object
fakedat2 &lt;- as.rspec(fakedat)
is.rspec(fakedat2)
head(fakedat2)
</code></pre>

<hr>
<h2 id='axistetra'>Plot reference axes in a static tetrahedral colourspace</h2><span id='topic+axistetra'></span>

<h3>Description</h3>

<p>Plots reference x, y and z arrows showing the direction of
the axes in a static tetrahedral colourspace plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>axistetra(
  x = 0,
  y = 1.3,
  size = 0.1,
  arrowhead = 0.05,
  col = par("fg"),
  lty = par("lty"),
  lwd = par("lwd"),
  label = TRUE,
  adj.label = list(x = c(0.003, 0), y = c(0.003, 0.003), z = c(0, 0.003)),
  label.cex = 1,
  label.col = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="axistetra_+3A_x">x</code>, <code id="axistetra_+3A_y">y</code></td>
<td>
<p>position of the legend relative to plot limits
(usually a value between 0 and 1, but because of the perspective distortion,
values greater than 1 are possible)</p>
</td></tr>
<tr><td><code id="axistetra_+3A_size">size</code></td>
<td>
<p>length of the arrows. Can be either a single value
(applied for x, y and z) or a vector of 3 separate values for each axis.</p>
</td></tr>
<tr><td><code id="axistetra_+3A_arrowhead">arrowhead</code></td>
<td>
<p>size of the arrowhead.</p>
</td></tr>
<tr><td><code id="axistetra_+3A_col">col</code>, <code id="axistetra_+3A_lty">lty</code>, <code id="axistetra_+3A_lwd">lwd</code></td>
<td>
<p>graphical parameters for the arrows.</p>
</td></tr>
<tr><td><code id="axistetra_+3A_label">label</code></td>
<td>
<p>logical, include x, y and z labels (defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="axistetra_+3A_adj.label">adj.label</code></td>
<td>
<p>position adjustment for the labels. a list of 3 named objects
for x, y and z arrows, each with 2 values for x and y adjustment.</p>
</td></tr>
<tr><td><code id="axistetra_+3A_label.cex">label.cex</code>, <code id="axistetra_+3A_label.col">label.col</code></td>
<td>
<p>graphical parameters for the labels.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>axistetra</code> adds reference arrows showing the direction of the
3-dimensional axes in a static tetrahedral colourspace plot.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sicalis)
vis.sicalis &lt;- vismodel(sicalis, visual = "avg.uv")
tcs.sicalis &lt;- colspace(vis.sicalis, space = "tcs")
plot(tcs.sicalis)
axistetra()
</code></pre>

<hr>
<h2 id='bgandilum'>Default background and illuminant data</h2><span id='topic+bgandilum'></span>

<h3>Description</h3>

<p>Default background and illuminant data
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Endler, J. (1993). The Color of Light in Forests and Its Implications. Ecological Monographs, 63, 1-27.
</p>

<hr>
<h2 id='bootcoldist'>Bootstrap colour distance confidence intervals</h2><span id='topic+bootcoldist'></span>

<h3>Description</h3>

<p>Uses a bootstrap procedure to generate confidence intervals
for the mean colour distance between two or more samples of colours
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootcoldist(vismodeldata, by, boot.n = 1000, alpha = 0.95, raw = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bootcoldist_+3A_vismodeldata">vismodeldata</code></td>
<td>
<p>(required) quantum catch colour data.
Can be the result from <code><a href="#topic+vismodel">vismodel()</a></code>, or <code><a href="#topic+colspace">colspace()</a></code>. Data may also be
independently calculated quantum catches, in the form of a data frame with
columns representing photoreceptors.</p>
</td></tr>
<tr><td><code id="bootcoldist_+3A_by">by</code></td>
<td>
<p>(required) a numeric or character vector indicating the group to which each row from
the object belongs to.</p>
</td></tr>
<tr><td><code id="bootcoldist_+3A_boot.n">boot.n</code></td>
<td>
<p>number of bootstrap replicates (defaults to 1000)</p>
</td></tr>
<tr><td><code id="bootcoldist_+3A_alpha">alpha</code></td>
<td>
<p>the confidence level for the confidence intervals (defaults to 0.95)</p>
</td></tr>
<tr><td><code id="bootcoldist_+3A_raw">raw</code></td>
<td>
<p>should the full set of bootstrapped distances (equal in length to boot.n)
be returned, instead of the summary distances and CI's? Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="bootcoldist_+3A_...">...</code></td>
<td>
<p>other arguments to be passed to <code><a href="#topic+coldist">coldist()</a></code>. Must at minimum
include <code>n</code> and <code>weber</code>. See <code><a href="#topic+coldist">coldist()</a></code> for details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can customise the type of parallel processing used by this function with
the <code><a href="future.html#topic+plan">future::plan()</a></code> function. This works on all operating systems, as well
as high performance computing (HPC) environment. Similarly, you can customise
the way progress is shown with the <code><a href="progressr.html#topic+handlers">progressr::handlers()</a></code> functions
(progress bar, acoustic feedback, nothing, etc.)
</p>


<h3>Value</h3>

<p>a matrix including the empirical mean and bootstrapped
confidence limits for dS (and dL if <code>achromatic = TRUE</code>), or a data.frame
of raw bootstraped dS (and dL if <code>achromatic = TRUE</code>) values equal in length to boot.n.
</p>


<h3>References</h3>

<p>Maia, R., White, T. E., (2018) Comparing colors using visual models.
Behavioral Ecology, ary017 <a href="https://doi.org/10.1093/beheco/ary017">doi:10.1093/beheco/ary017</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Run the receptor-noise limited model, using the visual phenotype
# of the blue tit
data(sicalis)
vm &lt;- vismodel(sicalis, achromatic = "bt.dc", relative = FALSE)
gr &lt;- gsub("ind..", "", rownames(vm))
bootcoldist(vm, by = gr, n = c(1, 2, 2, 4), weber = 0.1, weber.achro = 0.1)

# Run the same again, though as a simple colourspace model
data(sicalis)
vm &lt;- vismodel(sicalis, achromatic = "bt.dc")
space &lt;- colspace(vm)
gr &lt;- gsub("ind..", "", rownames(space))
bootcoldist(space, by = gr)

# Estimate bootstrapped colour-distances for a more 'specialised' model,
# like the colour hexagon
data(flowers)
vis.flowers &lt;- vismodel(flowers,
  visual = "apis", qcatch = "Ei", relative = FALSE,
  vonkries = TRUE, achromatic = "l", bkg = "green"
)
flowers.hex &lt;- colspace(vis.flowers, space = "hexagon")
pop_group &lt;- c(rep("pop_1", nrow(flowers.hex) / 2), rep("pop_2", nrow(flowers.hex) / 2))
bootcoldist(flowers.hex, by = pop_group)


</code></pre>

<hr>
<h2 id='categorical'>Categorical fly-visual model</h2><span id='topic+categorical'></span>

<h3>Description</h3>

<p>Applies the categorical colour vision model of Troje (1993)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>categorical(vismodeldata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="categorical_+3A_vismodeldata">vismodeldata</code></td>
<td>
<p>(required) quantum catch color data. Can be either the result
from <code><a href="#topic+vismodel">vismodel()</a></code> or independently calculated data (in the form of a data frame
with four columns named 'u' ,'s', 'm', 'l', representing a tetrachromatic viewer).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code>colspace</code> consisting of the following columns:
</p>

<ul>
<li> <p><code style="white-space: pre;">&#8288;R7p, R7y, R8p, R8y&#8288;</code>: the quantum catch data used to
calculate the remaining variables.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;x, y&#8288;</code>: cartesian coordinates in the categorical colour space.
</p>
</li>
<li> <p><code>r.vec</code>: the r vector (saturation, distance from the center).
</p>
</li>
<li> <p><code>h.theta</code>: angle theta (in radians), a continuous measure of stimulus hue.
</p>
</li>
<li> <p><code>category</code>: fly-colour category. One of <code style="white-space: pre;">&#8288;p-y-&#8288;</code>, <code style="white-space: pre;">&#8288;p-y+&#8288;</code>, <code style="white-space: pre;">&#8288;p+y-&#8288;</code>, <code style="white-space: pre;">&#8288;p+y+&#8288;</code>.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Troje N. (1993). Spectral categories in the learning behaviour
of blowflies. Zeitschrift fur Naturforschung C, 48, 96-96.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers, visual = "musca", achromatic = "md.r1")
cat.flowers &lt;- colspace(vis.flowers, space = "categorical")
</code></pre>

<hr>
<h2 id='catplot'>Plot the categorical colour vision model</h2><span id='topic+catplot'></span>

<h3>Description</h3>

<p>Produces a plot based on Troje's (1993) categorical colour model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>catplot(catdata, labels = TRUE, labels.cex = 0.9, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="catplot_+3A_catdata">catdata</code></td>
<td>
<p>(required) a data frame, possibly a result from the
<code><a href="#topic+colspace">colspace()</a></code> or <code><a href="#topic+categorical">categorical()</a></code> function, containing values for 'x' and 'y'
coordinates as columns (labeled as such).</p>
</td></tr>
<tr><td><code id="catplot_+3A_labels">labels</code></td>
<td>
<p>plot category labels inside? Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="catplot_+3A_labels.cex">labels.cex</code></td>
<td>
<p>size of the arrow labels.</p>
</td></tr>
<tr><td><code id="catplot_+3A_...">...</code></td>
<td>
<p>additional graphical options. See <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Troje N. (1993). Spectral categories in the learning behaviour
of blowflies. Zeitschrift fur Naturforschung C, 48, 96-96.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers, qcatch = "Qi", visual = "musca", achro = "none", relative = TRUE)
cat.flowers &lt;- colspace(vis.flowers, space = "categorical")
plot(cat.flowers)
</code></pre>

<hr>
<h2 id='cie'>CIE colour spaces</h2><span id='topic+cie'></span>

<h3>Description</h3>

<p>Calculates coordinates and colorimetric variables that represent reflectance spectra
in either the CIEXYZ (1931), CIELAB (1971), or CIELCh (1971) colourspaces.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cie(
  vismodeldata,
  space = c("XYZ", "LAB", "LCh"),
  visual = c("cie2", "cie10"),
  illum = c("D65", "bluesky", "forestshade")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cie_+3A_vismodeldata">vismodeldata</code></td>
<td>
<p>(required) quantum catch color data. Can be either the result
from <code><a href="#topic+vismodel">vismodel()</a></code> or independently calculated data (in the form of a
data frame with three columns representing trichromatic viewer).</p>
</td></tr>
<tr><td><code id="cie_+3A_space">space</code></td>
<td>
<p>(required) Choice between XYZ (default), LAB, or LCh colour models.</p>
</td></tr>
<tr><td><code id="cie_+3A_visual">visual</code></td>
<td>
<p>the visual system used when estimating XYZ values, if <code>vismodeldata</code> are
not the result of a call to <code>vismodel()</code> (otherwise the argument is ignored). Options are:
</p>

<ul>
<li><p> a data frame such as one produced containing by <code><a href="#topic+sensmodel">sensmodel()</a></code>, containing
user-defined sensitivity data for the receptors involved in colour vision.
The data frame must contain a <code>'wl'</code> column with the range of wavelengths included,
and the sensitivity for each other cone as a column.
</p>
</li>
<li> <p><code>'cie2'</code>: 2-degree colour matching functions for CIE models of human
colour vision. Functions are linear transformations of the 2-degree cone fundamentals
of Stockman &amp; Sharpe (2000), as ratified by the CIE (2006).
</p>
</li>
<li> <p><code>'cie10'</code>: 10-degree colour matching functions for CIE models of human
colour vision. Functions are linear transformations of the 10-degree cone fundamentals
of Stockman &amp; Sharpe (2000), as ratified by the CIE (2006).
</p>
</li></ul>
</td></tr>
<tr><td><code id="cie_+3A_illum">illum</code></td>
<td>
<p>the illuminant used when estimating XYZ values, if <code>vismodeldata</code> are
not the result of a call to <code>vismodel()</code> (otherwise the argument is ignored).
Either a data frame containing a <code>'wl'</code> column and the illuminant spectrum, or
one of the built-in options:
</p>

<ul>
<li> <p><code>'D65'</code>: standard daylight.
</p>
</li>
<li> <p><code>'bluesky'</code> open blue sky.
</p>
</li>
<li> <p><code>'forestshade'</code> forest shade.
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>Object of class <code><a href="#topic+colspace">colspace</a></code> containing:
</p>

<ul>
<li> <p><code style="white-space: pre;">&#8288;X, Y, Z&#8288;</code>: Tristimulus values.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;x, y, z&#8288;</code>: Cartesian coordinates, when using <code>space = XYZ</code>.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;L, a, b&#8288;</code>: Lightness, <code>L</code>, and colour-opponent <code>a</code> (redness-greenness) and
<code>b</code> (yellowness-blueness) values, in a Cartesian coordinate space. Returned
when using <code>space = LAB</code>.
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;L, a, b, C, h&#8288;</code>: Lightness, <code>L</code>, colour-opponent <code>a</code> (redness-greenness)
and <code>b</code> (yellowness-blueness) values, as well as chroma <code>C</code> and hue-angle <code>h</code>
(degrees), the latter of which are cylindrical representations of <code>a</code> and <code>b</code>
from the CIELAB model. Returned when using <code>space = LCh</code>.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Smith T, Guild J. (1932) The CIE colorimetric standards and their use.
Transactions of the Optical Society, 33(3), 73-134.
</p>
<p>Westland S, Ripamonti C, Cheung V. (2012). Computational colour science
using MATLAB. John Wiley &amp; Sons.
</p>
<p>Stockman, A., &amp; Sharpe, L. T. (2000). Spectral sensitivities of
the middle- and long-wavelength sensitive cones derived from measurements in
observers of known genotype. Vision Research, 40, 1711-1737.
</p>
<p>CIE (2006). Fundamental chromaticity diagram with physiological axes.
Parts 1 and 2. Technical Report 170-1. Vienna: Central Bureau of the Commission
Internationale de l Eclairage.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load floral reflectance spectra
data(flowers)

# Estimate quantum catches, using the cie10-degree viewer matching function
vis.flowers &lt;- vismodel(flowers, visual = "cie10", illum = "D65", vonkries = TRUE, relative = FALSE)

# Model floral spectra in the CIEXYZ space
flowers.ciexyz &lt;- colspace(vis.flowers, space = "ciexyz")

# Model floral spectra in the CIELab space
flowers.cielab &lt;- colspace(vis.flowers, space = "cielab")

# Model floral spectra in the CIELch space
flowers.cielch &lt;- colspace(vis.flowers, space = "cielch")

</code></pre>

<hr>
<h2 id='cieplot'>CIE plot</h2><span id='topic+cieplot'></span>

<h3>Description</h3>

<p>Plot a CIE (XYZ, LAB, or LCH) chromaticity diagram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cieplot(
  ciedata,
  mono = TRUE,
  out.lwd = NULL,
  out.lcol = "black",
  out.lty = 1,
  theta = 45,
  phi = 10,
  r = 1e+06,
  zoom = 1,
  box = FALSE,
  ciebg = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cieplot_+3A_ciedata">ciedata</code></td>
<td>
<p>(required). a data frame, possibly a result from the
<code><a href="#topic+colspace">colspace()</a></code> or <code><a href="#topic+cie">cie()</a></code> function, containing values for 'x', 'y' and 'z'
coordinates for the CIEXYZ model, or LAB coordinates for the CIELAB (or
CIELCh models), as columns (labeled as such).</p>
</td></tr>
<tr><td><code id="cieplot_+3A_mono">mono</code></td>
<td>
<p>should the monochromatic loci (the 'horseshoe') be plotted when
<code>space = "ciexyz"</code>? Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="cieplot_+3A_out.lwd">out.lwd</code>, <code id="cieplot_+3A_out.lcol">out.lcol</code>, <code id="cieplot_+3A_out.lty">out.lty</code></td>
<td>
<p>graphical parameters for the plot outline.</p>
</td></tr>
<tr><td><code id="cieplot_+3A_theta">theta</code></td>
<td>
<p>angle to rotate the plot in the xy plane  when
<code>space = "cielab"</code> (defaults to 10).</p>
</td></tr>
<tr><td><code id="cieplot_+3A_phi">phi</code></td>
<td>
<p>angle to rotate the plot in the yz plane  when <code>space = "cielab"</code>
(defaults to 45).</p>
</td></tr>
<tr><td><code id="cieplot_+3A_r">r</code></td>
<td>
<p>the distance of the eyepoint from the center of the plotting box
when <code>space = "cielab"</code>. Very high values approximate an orthographic
projection (defaults to 1e6). See <code><a href="graphics.html#topic+persp">graphics::persp()</a></code> for details.</p>
</td></tr>
<tr><td><code id="cieplot_+3A_zoom">zoom</code></td>
<td>
<p>zooms in (values greater than 1) or out (values between 0 and 1)
from the plotting area when <code>space = "cielab"</code>.</p>
</td></tr>
<tr><td><code id="cieplot_+3A_box">box</code></td>
<td>
<p>logical. Should the plot area box and axes be plotted? (defaults
to <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="cieplot_+3A_ciebg">ciebg</code></td>
<td>
<p>should the colour background be plotted for CIEXYZ plot?
(defaults to <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="cieplot_+3A_...">...</code></td>
<td>
<p>additional graphical options. See <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Smith T, Guild J. (1932) The CIE colorimetric standards and their use.
Transactions of the Optical Society, 33(3), 73-134.
</p>
<p>Westland S, Ripamonti C, Cheung V. (2012). Computational colour science
using MATLAB. John Wiley &amp; Sons.
</p>
<p>Stockman, A., &amp; Sharpe, L. T. (2000). Spectral sensitivities of
the middle- and long-wavelength sensitive cones derived from measurements in
observers of known genotype. Vision Research, 40, 1711-1737.
</p>
<p>CIE (2006). Fundamental chromaticity diagram with physiological axes.
Parts 1 and 2. Technical Report 170-1. Vienna: Central Bureau of the Commission
Internationale de l Eclairage.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load floral reflectance spectra
data(flowers)

# CIEXYZ
# Estimate quantum catches, using the cie10-degree viewer matching function
vis.flowers &lt;- vismodel(flowers, visual = "cie10", illum = "D65", vonkries = TRUE, relative = FALSE)

# Run the ciexyz model
xyz.flowers &lt;- colspace(vis.flowers, space = "ciexyz")

# Visualise the floral spectra in a ciexyz chromaticity diagram
plot(xyz.flowers)

# CIELAB
# Using the quantum catches above, instead model the spectra in the CIELab
# space
lab.flowers &lt;- colspace(vis.flowers, space = "cielab")

# And plot in Lab space
plot(lab.flowers)

</code></pre>

<hr>
<h2 id='classify'>Identify colour classes in an image for adjacency analyses</h2><span id='topic+classify'></span>

<h3>Description</h3>

<p>Classify image pixels into discrete colour classes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>classify(
  imgdat,
  method = c("kMeans", "kMedoids"),
  kcols = NULL,
  refID = NULL,
  interactive = FALSE,
  plotnew = FALSE,
  col = "red",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="classify_+3A_imgdat">imgdat</code></td>
<td>
<p>(required) image data. Either a single image, or a series of images
stored in a list. Preferably the result of <code><a href="#topic+getimg">getimg()</a></code>.</p>
</td></tr>
<tr><td><code id="classify_+3A_method">method</code></td>
<td>
<p>methods for image segmentation/classification.
</p>

<ul>
<li> <p><code>'kMeans'</code>: k-means clustering (default)
</p>
</li>
<li> <p><code>'kMedoids'</code>: k-medoids clustering, using the partitioning-around-medoids ('pam')
algorithm for large datasets.
</p>
</li></ul>
</td></tr>
<tr><td><code id="classify_+3A_kcols">kcols</code></td>
<td>
<p>the number of discrete colour classes present in the input image(s).
Can be a single integer when only a single image is present, or if kcols is identical for all
images. When passing a list of images, <code>kcols</code> can also be a vector the same length
as <code>imgdat</code>, or a data.frame with two columns specifying image file names and
corresponding kcols. This argument can optionally be disregarded when <code>interactive = TRUE</code>,
and kcols will be inferred from the number of selections.</p>
</td></tr>
<tr><td><code id="classify_+3A_refid">refID</code></td>
<td>
<p>either the numeric index or name of a 'reference' image, for use when passing
a list of images. Other images will be k-means classified using centres identified
in the single reference image, thus helping to ensure that homologous pattern elements
will be reliably classified between images, if so desired.</p>
</td></tr>
<tr><td><code id="classify_+3A_interactive">interactive</code></td>
<td>
<p>interactively specify the colour-category 'centers', for k-means clustering.
When <code>TRUE</code>, the user is asked to click a number of points (equal to <code>kcols</code>,
if specified, otherwise user-determined) that represent the distinct colours of interest.
If a reference image is specified, it will be the only image presented.</p>
</td></tr>
<tr><td><code id="classify_+3A_plotnew">plotnew</code></td>
<td>
<p>Should plots be opened in a new window when <code>interactive = TRUE</code>?
Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="classify_+3A_col">col</code></td>
<td>
<p>the color of the marker points, when <code>interactive = TRUE</code>.</p>
</td></tr>
<tr><td><code id="classify_+3A_...">...</code></td>
<td>
<p>additional graphical parameters when <code>interactive = TRUE</code>.
Also see <code><a href="graphics.html#topic+par">graphics::par()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can customise the type of parallel processing used by this function with
the <code><a href="future.html#topic+plan">future::plan()</a></code> function. This works on all operating systems, as well
as high performance computing (HPC) environment. Similarly, you can customise
the way progress is shown with the <code><a href="progressr.html#topic+handlers">progressr::handlers()</a></code> functions
(progress bar, acoustic feedback, nothing, etc.)
</p>


<h3>Value</h3>

<p>A matrix, or list of matrices, of class <code>rimg</code> containing the colour
class classifications ID at each pixel location. The RGB values corresponding to
cluster centres (i.e. colour classes) are stored as object attributes.
</p>


<h3>Note</h3>

<p>Since the <code>kmeans</code> process draws on random numbers to find initial
cluster centres when <code>interactive = FALSE</code>, use <code><a href="base.html#topic+set.seed">set.seed()</a></code> if reproducible
cluster ID's are desired between runs.
</p>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>See Also</h3>

<p><a href="stats.html#topic+kmeans">stats::kmeans</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Single image
papilio &lt;- getimg(system.file("testdata/images/butterflies/papilio.png", package = "pavo"))
papilio_class &lt;- classify(papilio, kcols = 4)

# Multiple images, with interactive classification and a reference image
snakes &lt;- getimg(system.file("testdata/images/snakes", package = "pavo"))
if (interactive()) {
  snakes_class &lt;- classify(snakes, refID = "snake_01", interactive = TRUE)
}
</code></pre>

<hr>
<h2 id='coc'>Color opponent coding model</h2><span id='topic+coc'></span>

<h3>Description</h3>

<p>Calculates coordinates and colorimetric variables that represent reflectance spectra
in the color opponent coding model of hymenopteran vision.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coc(vismodeldata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coc_+3A_vismodeldata">vismodeldata</code></td>
<td>
<p>(required) quantum catch colour data. Can be either the result
from <code><a href="#topic+vismodel">vismodel()</a></code> or independently calculated data (in the form of a data frame
with three columns named 's', 'm', 'l', representing a trichromatic viewer).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of class <code><a href="#topic+colspace">colspace</a></code> consisting of the following columns:
</p>

<ul>
<li> <p><code>s</code>, <code>m</code>, <code>l</code>: the quantum catch data used to calculate the remaining
variables.
</p>
</li>
<li> <p><code>x</code>, <code>y</code>: coordinates for the points in coc space
</p>
</li>
<li> <p><code>r.vec</code>: the r vector (saturation, distance from the center using a
city-block metric).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Backhaus W. (1991). Color opponent coding in the visual system
of the honeybee. Vision Research, 31, 1381-1397.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers, visual = "apis", qcatch = "Ei", relative = FALSE, vonkries = TRUE)
coc.flowers &lt;- colspace(vis.flowers, space = "coc")
</code></pre>

<hr>
<h2 id='cocplot'>Plot the colour opponent coding diagram</h2><span id='topic+cocplot'></span>

<h3>Description</h3>

<p>Produces a plot based on the colour opponent coding diagram of Backhaus
(1991).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cocplot(
  cocdata,
  labels = TRUE,
  labels.cex = 0.9,
  tick.loc = c(-12, -9, -6, -3, 3, 6, 9, 12),
  achro = FALSE,
  achrosize = 0.8,
  achrocol = "grey",
  square = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cocplot_+3A_cocdata">cocdata</code></td>
<td>
<p>(required) a data frame, possibly a result from the
<code><a href="#topic+colspace">colspace()</a></code> or <code><a href="#topic+categorical">categorical()</a></code> function, containing values for 'x' and 'y'
coordinates as columns (labeled as such).</p>
</td></tr>
<tr><td><code id="cocplot_+3A_labels">labels</code></td>
<td>
<p>logical. Should the name of each cone be printed next to the
corresponding vertex?</p>
</td></tr>
<tr><td><code id="cocplot_+3A_labels.cex">labels.cex</code></td>
<td>
<p>size of the arrow labels.</p>
</td></tr>
<tr><td><code id="cocplot_+3A_tick.loc">tick.loc</code></td>
<td>
<p>a numeric vector specifying the location of tick marks on x &amp;
y axes.</p>
</td></tr>
<tr><td><code id="cocplot_+3A_achro">achro</code></td>
<td>
<p>should a point be plotted at the origin (defaults to <code>TRUE</code>)?</p>
</td></tr>
<tr><td><code id="cocplot_+3A_achrosize">achrosize</code></td>
<td>
<p>size of the point at the origin when <code>achro = TRUE</code>
(defaults to <code>0.8</code>).</p>
</td></tr>
<tr><td><code id="cocplot_+3A_achrocol">achrocol</code></td>
<td>
<p>color of the point at the origin <code>achro = TRUE</code> (defaults to
<code>'grey'</code>).</p>
</td></tr>
<tr><td><code id="cocplot_+3A_square">square</code></td>
<td>
<p>logical. Should the aspect ratio of the plot be held to 1:1?
(defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="cocplot_+3A_...">...</code></td>
<td>
<p>additional graphical options. See <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Backhaus W. (1991). Color opponent coding in the visual system
of the honeybee. Vision Research, 31, 1381-1397.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers, visual = "apis", qcatch = "Ei", relative = FALSE, vonkries = TRUE)
coc.flowers &lt;- colspace(vis.flowers, space = "coc")
plot(coc.flowers)
</code></pre>

<hr>
<h2 id='coldist'>Colour distances</h2><span id='topic+coldist'></span>

<h3>Description</h3>

<p>Calculates colour distances. When data are the result of <code><a href="#topic+vismodel">vismodel()</a></code>, it
applies the receptor-noise model of Vorobyev et al. (1998) to calculate
colour distances with noise based on relative photoreceptor densities. It
also accepts <code><a href="#topic+colspace">colspace()</a></code> data in which case unweighted Euclidean distances,
CIE2000 distances (cielab and cielch only), or Manhattan distances (coc model only)
are returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coldist(
  modeldata,
  noise = c("neural", "quantum"),
  subset = NULL,
  achromatic = FALSE,
  qcatch = NULL,
  n = c(1, 2, 2, 4),
  weber = 0.1,
  weber.ref = "longest",
  weber.achro = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coldist_+3A_modeldata">modeldata</code></td>
<td>
<p>(required) quantum catch colour data. Can be the result from
<code><a href="#topic+vismodel">vismodel()</a></code> for noise-weighted Euclidean distances, or <code><a href="#topic+colspace">colspace()</a></code> for
unweighted (typically) Euclidean distances. Data may also be independently
calculated quantum catches, in the form of a data frame with columns
representing photoreceptors.</p>
</td></tr>
<tr><td><code id="coldist_+3A_noise">noise</code></td>
<td>
<p>how the noise will be calculated (ignored for <code>colspace</code>
objects):
</p>

<ul>
<li> <p><code>neural</code> (default): noise is proportional to the Weber fraction and is
independent of the intensity of the signal received (i.e. assumes bright
conditions).
</p>
</li>
<li> <p><code>quantum</code>: noise is the sum of the neural noise and receptor noise, and
is thus proportional to the Weber fraction and inversely proportional to
the intensity of the signal received (the quantum catches).
Note that the <code>quantum</code> option will only work with objects of
class <code>vismodel</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="coldist_+3A_subset">subset</code></td>
<td>
<p>If only some of the comparisons should be returned, a character
vector of length 1 or 2 can be provided, indicating which samples are
desired. The subset vector must match the labels of the input samples, but
partial matching (and regular expressions) are supported.</p>
</td></tr>
<tr><td><code id="coldist_+3A_achromatic">achromatic</code></td>
<td>
<p>Logical. If <code>TRUE</code>, last column of the data frame is used
to calculate the achromatic contrast, the form of which will depend on the
input data and will be indicated by a message during execution. For
noise-weighted distances, noise is based on the Weber fraction given by the
argument <code>weber.achro</code>.</p>
</td></tr>
<tr><td><code id="coldist_+3A_qcatch">qcatch</code></td>
<td>
<p>if the object is of class <code><a href="#topic+vismodel">vismodel</a></code> or <code><a href="#topic+colspace">colspace</a></code>, this
argument is ignored. If the object is a data frame of quantal catches from
another source, this argument is used to specify what type of quantum catch
is being used, so that the noise can be calculated accordingly: * <code>Qi</code>:
Quantum catch for each photoreceptor * <code>fi</code>: Quantum catch according to
Fechner's law (the signal of the receptor channel is proportional to the
logarithm of the quantum catch)</p>
</td></tr>
<tr><td><code id="coldist_+3A_n">n</code></td>
<td>
<p>photoreceptor densities for the cones used in visual modeling. must
have same length as number of columns (excluding achromatic receptor if
used; defaults to the Pekin robin <em>Leiothrix lutea</em> densities:
<code>c(1,2,2,4)</code>). Ignored for <code><a href="#topic+colspace">colspace</a></code> objects.</p>
</td></tr>
<tr><td><code id="coldist_+3A_weber">weber</code></td>
<td>
<p>The Weber fraction(s) to be used (often also referred to as
receptor noise, or <em>e</em>). The noise-to-signal ratio <code>v</code> is unknown, and
therefore must be calculated based on the empirically estimated Weber
fraction of one or (more rarely) all the cone classes. When noise is only
known for one receptor, as is typical, <code>v</code> is then applied to estimate the
Weber fraction of the other cones. By default, the value of 0.1 is used
(the empirically estimated value for the LWS cone from <em>Leiothrix lutea</em>).
See Olsson et al. 2017 for a review of published values in the literature.
Ignored for <code>colspace</code> objects.</p>
</td></tr>
<tr><td><code id="coldist_+3A_weber.ref">weber.ref</code></td>
<td>
<p>the cone class used to obtain the empirical estimate of the
Weber fraction used for the <code>weber</code> argument, if a single value is
specified. By default, <code>n4</code> is used, representing the LWS cone for
<em>Leiothrix lutea</em>. Ignored for <code>colspace</code> objects.</p>
</td></tr>
<tr><td><code id="coldist_+3A_weber.achro">weber.achro</code></td>
<td>
<p>the Weber fraction to be used to calculate achromatic
contrast, when <code>achromatic = TRUE</code>. Defaults to 0.1. Ignored for <code>colspace</code>
objects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing up to 4 columns. The first two
(<code style="white-space: pre;">&#8288;patch1, patch2&#8288;</code>) refer to the two colors being contrasted; <code>dS</code> is the
chromatic contrast (delta S) and <code>dL</code> is the achromatic contrast (delta L).
Units of <code>dS</code> JND's in the receptor-noise model, unweighted Euclidean
distances in colorspace models, and Manhattan distances in the
colour-opponent-coding space. Units of <code>dL</code> vary, and are either simple
contrast, Weber contrast, or Michelson contrast, as indicated by the output
message.
</p>


<h3>Note on previous versions</h3>

<p>Generic di- tri- and tetra-chromatic
<code>colspace</code> objects were previously passed through the receptor-noise
limited model to return noise-weighted Euclidean distances. This behaviour
has been amended, and generic spaces now return unweighted Euclidean
distances. Equivalent results to the former behaviour can be attained by
sending the results of <code><a href="#topic+vismodel">vismodel()</a></code> directly to <code><a href="#topic+coldist">coldist()</a></code> , as
previously, which also offers greater flexibility and reliability. Thus
<code><a href="#topic+coldist">coldist()</a></code> now returns unweighted Euclidean distances for <code>colspace</code>
objects (with the exception of Manhattan distances for the coc space, and CIE2000,
distances for CIELab and CIELCh spaces), and noise-weighted Euclidean distances
for <code>vismodel</code> objects.
</p>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Vorobyev, M., Osorio, D., Bennett, A., Marshall, N., &amp; Cuthill,
I. (1998). Tetrachromacy, oil droplets and bird plumage colours. Journal Of
Comparative Physiology A-Neuroethology Sensory Neural And Behavioral
Physiology, 183(5), 621-633.
</p>
<p>Hart, N. S. (2001). The visual ecology of avian photoreceptors.
Progress In Retinal And Eye Research, 20(5), 675-703.
</p>
<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour
patterns as birds see them. Biological Journal Of The Linnean Society,
86(4), 405-431.
</p>
<p>Olsson, P., Lind, O., &amp; Kelber, A. (2015) Bird colour vision:
behavioural thresholds reveal receptor noise. Journal of Experimental
Biology, 218, 184-193.
</p>
<p>Lind, O. (2016) Colour vision and background adaptation in a
passerine bird, the zebra finch (Taeniopygia guttata). Royal Society Open
Science, 3, 160383.
</p>
<p>Olsson, P., Lind, O., &amp; Kelber, A. (2017) Chromatic and
achromatic vision: parameter choice and limitations for reliable model
predictions. Behavioral Ecology, <a href="https://doi.org/10.1093/beheco/arx133">doi:10.1093/beheco/arx133</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Dichromat
data(flowers)
vis.flowers &lt;- vismodel(flowers, visual = "canis", relative = FALSE)
didist.flowers &lt;- coldist(vis.flowers, n = c(1, 2))

# Trichromat
vis.flowers &lt;- vismodel(flowers, visual = "apis", relative = FALSE)
tridist.flowers &lt;- coldist(vis.flowers, n = c(1, 2, 1))

# Trichromat, colour-hexagon model (euclidean distances)
vis.flowers &lt;- vismodel(flowers,
  visual = "apis", qcatch = "Ei",
  relative = FALSE, vonkries = TRUE, achromatic = "l", bkg = "green"
)
hex.flowers &lt;- colspace(vis.flowers, space = "hexagon")
hexdist.flowers &lt;- coldist(hex.flowers)

# Trichromat, colour-opponent-coding model (manhattan distances)
vis.flowers &lt;- vismodel(flowers, visual = "apis", qcatch = "Ei", relative = FALSE, vonkries = TRUE)
coc.flowers &lt;- colspace(vis.flowers, space = "coc")
hexdist.flowers &lt;- coldist(coc.flowers)

# Tetrachromat
data(sicalis)
vis.sicalis &lt;- vismodel(sicalis, visual = "avg.uv", relative = FALSE)
tetradist.sicalis.n &lt;- coldist(vis.sicalis)


</code></pre>

<hr>
<h2 id='coldist2mat'>Convert coldist to distance matrix</h2><span id='topic+coldist2mat'></span>

<h3>Description</h3>

<p>Converts a <code><a href="#topic+coldist">coldist()</a></code> output into a distance matrix where samples
are rows and columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coldist2mat(coldistres)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coldist2mat_+3A_coldistres">coldistres</code></td>
<td>
<p>(required) the output from a <code><a href="#topic+coldist">coldist()</a></code> call.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing one or two matrices, for dS and dL, depending
if the original object had dS and dL columns
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Maia, R., White, T. E., (2018) Comparing colors using visual models.
Behavioral Ecology, ary017 <a href="https://doi.org/10.1093/beheco/ary017">doi:10.1093/beheco/ary017</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers, achromatic = "l")
cd.flowers &lt;- coldist(vis.flowers, achromatic = TRUE)
coldist2mat(cd.flowers)[["dS"]]
coldist2mat(cd.flowers)[["dL"]]
</code></pre>

<hr>
<h2 id='colspace'>Model spectra in a colorspace</h2><span id='topic+colspace'></span>

<h3>Description</h3>

<p>Models reflectance spectra in a colorspace. For information on plotting
arguments and graphical parameters, see <code><a href="#topic+plot.colspace">plot.colspace()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colspace(
  vismodeldata,
  space = c("auto", "di", "tri", "tcs", "hexagon", "coc", "categorical", "ciexyz",
    "cielab", "cielch", "segment"),
  qcatch = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="colspace_+3A_vismodeldata">vismodeldata</code></td>
<td>
<p>(required) quantum catch color data. Can be either the
result from <code><a href="#topic+vismodel">vismodel()</a></code> or independently calculated data (in the form of a
data frame with columns representing quantum catches).</p>
</td></tr>
<tr><td><code id="colspace_+3A_space">space</code></td>
<td>
<p>Which colorspace/model to use. Options are:
</p>

<ul>
<li> <p><code>auto</code>: if data is a result from <code><a href="#topic+vismodel">vismodel()</a></code>, applies <code>di</code>, <code>tri</code> or <code>tcs</code>
if input visual model had two, three or four cones, respectively.
</p>
</li>
<li> <p><code>di</code>: dichromatic colourspace. See <code><a href="#topic+dispace">dispace()</a></code> for details.
(<a href="#topic+diplot">plotting arguments</a>)
</p>
</li>
<li> <p><code>tri</code>: trichromatic colourspace (i.e. Maxwell triangle). See <code><a href="#topic+trispace">trispace()</a></code>
for details. (<a href="#topic+triplot">plotting arguments</a>)
</p>
</li>
<li> <p><code>tcs</code>: tetrahedral colourspace. See <code><a href="#topic+tcspace">tcspace()</a></code> for details.
(<a href="#topic+tetraplot">plotting arguments</a>)
</p>
</li>
<li> <p><code>hexagon</code>: the trichromatic colour-hexagon of Chittka (1992). See
<code><a href="#topic+hexagon">hexagon()</a></code> for details. (<a href="#topic+hexplot">plotting arguments</a>)
</p>
</li>
<li> <p><code>coc</code>: the trichromatic colour-opponent-coding model of Backhaus (1991).
See <code><a href="#topic+coc">coc()</a></code> for details. (<a href="#topic+cocplot">plotting arguments</a>)
</p>
</li>
<li> <p><code>categorical</code>: the tetrachromatic categorical fly-model of Troje (1993).
See <code><a href="#topic+categorical">categorical()</a></code> for details. (<a href="#topic+catplot">plotting arguments</a>)
</p>
</li>
<li> <p><code>ciexyz</code>: CIEXYZ space. See <code><a href="#topic+cie">cie()</a></code> for details.
(<a href="#topic+cieplot">plotting arguments</a>)
</p>
</li>
<li> <p><code>cielab</code>: CIELAB space. See <code><a href="#topic+cie">cie()</a></code> for details.
(<a href="#topic+cieplot">plotting arguments</a>)
</p>
</li>
<li> <p><code>cielch</code>: CIELCh space. See <code><a href="#topic+cie">cie()</a></code> for details.
(<a href="#topic+cieplot">plotting arguments</a>)
</p>
</li>
<li> <p><code>segment</code>: segment analysis of Endler (1990). See <code><a href="#topic+segspace">segspace()</a></code> for details.
(<a href="#topic+segplot">plotting arguments</a>)
</p>
</li></ul>
</td></tr>
<tr><td><code id="colspace_+3A_qcatch">qcatch</code></td>
<td>
<p>Which quantal catch metric is being inputted. Only used when
input data is NOT an output from <code><a href="#topic+vismodel">vismodel()</a></code>. Must be <code>Qi</code>, <code>fi</code> or <code>Ei</code>.</p>
</td></tr>
<tr><td><code id="colspace_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code><a href="#topic+cie">cie()</a></code> for non <code>vismodel()</code> data.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>
<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Smith T, Guild J. (1932) The CIE colorimetric standards and their
use. Transactions of the Optical Society, 33(3), 73-134.
</p>
<p>Westland S, Ripamonti C, Cheung V. (2012). Computational colour
science using MATLAB. John Wiley &amp; Sons.
</p>
<p>Chittka L. (1992). The colour hexagon: a chromaticity diagram
based on photoreceptor excitations as a generalized representation of
colour opponency. Journal of Comparative Physiology A, 170(5), 533-543.
</p>
<p>Chittka L, Shmida A, Troje N, Menzel R. (1994). Ultraviolet as a
component of flower reflections, and the colour perception of Hymenoptera.
Vision research, 34(11), 1489-1508.
</p>
<p>Troje N. (1993). Spectral categories in the learning behaviour of
blowflies. Zeitschrift fur Naturforschung C, 48, 96-96.
</p>
<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage
color in a tetrahedral color space: A phylogenetic analysis of new world
buntings. The American Naturalist, 171(6), 755-776.
</p>
<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour
patterns as birds see them. Biological Journal Of The Linnean Society,
86(4), 405-431.
</p>
<p>Kelber A, Vorobyev M, Osorio D. (2003). Animal colour vision -
behavioural tests and physiological concepts. Biological Reviews, 78, 81 -
118.
</p>
<p>Backhaus W. (1991). Color opponent coding in the visual system of
the honeybee. Vision Research, 31, 1381-1397.
</p>
<p>Endler, J. A. (1990) On the measurement and classification of
color in studies of animal color patterns. Biological Journal of the
Linnean Society, 41, 315-352.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)

# Model a dichromat viewer in a segment colourspace
vis.flowers &lt;- vismodel(flowers, visual = "canis")
di.flowers &lt;- colspace(vis.flowers, space = "di")

# Model a honeybee viewer in the colour hexagon
vis.flowers &lt;- vismodel(flowers,
  visual = "apis", qcatch = "Ei", relative = FALSE,
  vonkries = TRUE, achromatic = "l", bkg = "green"
)
hex.flowers &lt;- colspace(vis.flowers, space = "hexagon")

# Model a trichromat (the honeybee) in a Maxwell triangle
vis.flowers &lt;- vismodel(flowers, visual = "apis")
tri.flowers &lt;- colspace(vis.flowers, space = "tri")
plot(tri.flowers)

# Model a tetrachromat (the Blue Tit) in a tetrahedral colourspace
vis.flowers &lt;- vismodel(flowers, visual = "bluetit")
tcs.flowers &lt;- colspace(vis.flowers, space = "tcs")

# Model a housefly in the 'categorical' colourspace
vis.flowers &lt;- vismodel(flowers, visual = "musca", achro = "md.r1")
cat.flowers &lt;- colspace(vis.flowers, space = "categorical")

</code></pre>

<hr>
<h2 id='diplot'>Plot a dichromat segment</h2><span id='topic+diplot'></span>

<h3>Description</h3>

<p>Produces a dichromat segment plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diplot(
  didata,
  labels = TRUE,
  achro = TRUE,
  achrocol = "grey",
  achrosize = 0.8,
  labels.cex = 1,
  out.lwd = 1,
  out.lcol = "black",
  out.lty = 1,
  square = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diplot_+3A_didata">didata</code></td>
<td>
<p>(required) a data frame, possibly a result from the
<code><a href="#topic+colspace">colspace()</a></code> or <code><a href="#topic+dispace">dispace()</a></code> function, containing values for the 'x'
coordinates as a column (labeled as such).</p>
</td></tr>
<tr><td><code id="diplot_+3A_labels">labels</code></td>
<td>
<p>logical. Should the name of each cone be printed next to the
corresponding vertex?</p>
</td></tr>
<tr><td><code id="diplot_+3A_achro">achro</code></td>
<td>
<p>should a point be plotted at the origin (defaults to <code>TRUE</code>)?</p>
</td></tr>
<tr><td><code id="diplot_+3A_achrocol">achrocol</code></td>
<td>
<p>color of the point at the origin <code>achro = TRUE</code> (defaults to
<code>'grey'</code>).</p>
</td></tr>
<tr><td><code id="diplot_+3A_achrosize">achrosize</code></td>
<td>
<p>size of the point at the origin when <code>achro = TRUE</code>
(defaults to <code>0.8</code>).</p>
</td></tr>
<tr><td><code id="diplot_+3A_labels.cex">labels.cex</code></td>
<td>
<p>size of the arrow labels.</p>
</td></tr>
<tr><td><code id="diplot_+3A_out.lwd">out.lwd</code>, <code id="diplot_+3A_out.lcol">out.lcol</code>, <code id="diplot_+3A_out.lty">out.lty</code></td>
<td>
<p>graphical parameters for the plot outline.</p>
</td></tr>
<tr><td><code id="diplot_+3A_square">square</code></td>
<td>
<p>logical. Should the aspect ratio of the plot be held to 1:1?
(defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="diplot_+3A_...">...</code></td>
<td>
<p>additional graphical options. See <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Kelber A, Vorobyev M, Osorio D. (2003). Animal colour vision -
behavioural tests and physiological concepts. Biological Reviews, 78,
81 - 118.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers, visual = "canis")
di.flowers &lt;- colspace(vis.flowers, space = "di")
plot(di.flowers)
</code></pre>

<hr>
<h2 id='dispace'>Dichromatic colour space</h2><span id='topic+dispace'></span>

<h3>Description</h3>

<p>Calculates coordinates and colorimetric variables that represent reflectance spectra
in a dichromatic colour space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dispace(vismodeldata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dispace_+3A_vismodeldata">vismodeldata</code></td>
<td>
<p>(required) quantum catch color data. Can be either the result
from <code><a href="#topic+vismodel">vismodel()</a></code> or independently calculated data (in the form of a data frame
with two columns named 's' and 'l', representing a dichromatic viewer's receptors).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of class <code><a href="#topic+colspace">colspace</a></code> consisting of the following columns:
</p>

<ul>
<li> <p><code>s</code>, <code>l</code>: the quantum catch data used to calculate
the remaining variables.
</p>
</li>
<li> <p><code>x</code>: the coordinate of the stimulus along a segment
</p>
</li>
<li> <p><code>r.vec</code>: the r vector (saturation, distance from the center).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Kelber A, Vorobyev M, Osorio D. (2003). Animal colour vision -
behavioural tests and physiological concepts. Biological Reviews, 78,
81 - 118.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers, visual = "canis")
di.flowers &lt;- colspace(vis.flowers, space = "di")
</code></pre>

<hr>
<h2 id='explorespec'>Plot spectral curves</h2><span id='topic+explorespec'></span>

<h3>Description</h3>

<p>Plots one or multiple spectral curves in the same graph to rapidly
compare groups of spectra.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>explorespec(
  rspecdata,
  by = NULL,
  scale = c("equal", "free"),
  legpos = "topright",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="explorespec_+3A_rspecdata">rspecdata</code></td>
<td>
<p>(required) a data frame, possibly of class <code>rspec</code>, which
contains a column containing a wavelength range, named 'wl', and spectra
data in remaining columns.</p>
</td></tr>
<tr><td><code id="explorespec_+3A_by">by</code></td>
<td>
<p>number of spectra to include in each graph (defaults to 1)</p>
</td></tr>
<tr><td><code id="explorespec_+3A_scale">scale</code></td>
<td>
<p>defines how the y-axis should be scaled. <code>"free"</code>: panels can vary in
the range of the y-axis; <code>"equal"</code>: all panels have the y-axis with the same range.</p>
</td></tr>
<tr><td><code id="explorespec_+3A_legpos">legpos</code></td>
<td>
<p>legend position control. Either a vector containing <code>x</code> and <code>y</code> coordinates
or a single keyword from the list: <code>"bottomright"</code>, <code>"bottom"</code>, <code>"bottomleft"</code>,
<code>"left"</code>, <code>"topleft"</code>, <code>"top"</code>, <code>"topright"</code>, <code>"right"</code> and <code>"center"</code>.</p>
</td></tr>
<tr><td><code id="explorespec_+3A_...">...</code></td>
<td>
<p>additional parameters to be passed to plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Spectral curve plots
</p>


<h3>Note</h3>

<p>Number of plots presented per page depends on the number of graphs produced.
</p>


<h3>Author(s)</h3>

<p>Pierre-Paul Bitton <a href="mailto:bittonp@uwindsor.ca">bittonp@uwindsor.ca</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sicalis)
explorespec(sicalis, 3)
explorespec(sicalis, 3, ylim = c(0, 100), legpos = c(500, 80))
</code></pre>

<hr>
<h2 id='find_astar'>Compute the <code class="reqn">\alpha^*</code> value</h2><span id='topic+find_astar'></span>

<h3>Description</h3>

<p>Compute the <code class="reqn">\alpha^*</code> value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_astar(coords)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find_astar_+3A_coords">coords</code></td>
<td>
<p>coordinates of the points for which you want to compute the
alphashape</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The <code class="reqn">\alpha^*</code> value
as defined in Gruson (2020).
</p>


<h3>References</h3>

<p>Gruson H. (2020). Estimation of colour volumes as concave hypervolumes using
<code class="reqn">\alpha</code>-shapes. Methods in Ecology and Evolution, 11(8), 955-963
<a href="https://doi.org/10.1111/2041-210X.13398">doi:10.1111/2041-210X.13398</a>
</p>


<h3>See Also</h3>

<p><code><a href="alphashape3d.html#topic+ashape3d">alphashape3d::ashape3d()</a></code>
</p>

<hr>
<h2 id='flowers'>Reflectance spectra from a suite of native Australian flowers, collected
around Cairns, Queensland.</h2><span id='topic+flowers'></span>

<h3>Description</h3>

<p>Dataset containing reflectance measurements from 36 native Australian
angiosperm species, indicated by column names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(flowers)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>rspec</code> (inherits from <code>data.frame</code>) with 401 rows and 37 columns.
</p>


<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Dalrymple, R., L., Kemp, D. J., Flores-Moreno, H., Laffan, S. W.,
White, T. E., Hemmings, F. A., Tindall, M. L., &amp; Moles, A. T. (2015). Birds,
butterflies and flowers in the tropics are not more colourful than those at higher
latitudes. Global Ecology and Biogeography, 24(12), 1424-1432. <a href="https://doi.org/10.1111/geb.12368">doi:10.1111/geb.12368</a>
</p>
<p>White, T. E., Dalrymple, R. L., Herberstein, M. E., &amp; Kemp, D. J. (2017).
The perceptual similarity of orb-spider prey lures and flowers colours. Evolutionary
Ecology, 31(1), 1-20. <a href="https://doi.org/10.1007/s10682-016-9876-x">doi:10.1007/s10682-016-9876-x</a>
</p>
<p>Dalrymple, R. L., Flores-Moreno, H., Kemp, D. J., White, T. E.,
Laffan, S. W., Hemmings, F. A., Tindall, M. L., &amp; Moles, A. T. (2018). Abiotic
and biotic predictors of macroecological patterns in bird and butterfly coloration.
Ecological Monographs, 88(2), 204-224.
</p>

<hr>
<h2 id='getimg'>Import image data</h2><span id='topic+getimg'></span>

<h3>Description</h3>

<p>Finds and imports PNG, JPEG, and/or BMP images.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getimg(imgpath = getwd(), subdir = FALSE, subdir.names = FALSE, max.size = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getimg_+3A_imgpath">imgpath</code></td>
<td>
<p>(required) either the full file-path or URL to an image (including extension),
or the path to a folder in which multiple image files are located. Mixed file formats
within a folder are accepted.</p>
</td></tr>
<tr><td><code id="getimg_+3A_subdir">subdir</code></td>
<td>
<p>should subdirectories within the <code>imgpath</code> folder be
included in the search? (defaults to <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="getimg_+3A_subdir.names">subdir.names</code></td>
<td>
<p>should subdirectory path be included in the name of the
images? (defaults to <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="getimg_+3A_max.size">max.size</code></td>
<td>
<p>maximum size of all images to be allowed in memory, in GB. Defaults to
<code>1</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a image, or list of images, of class <code>rimg</code>, for use in further
<code>pavo</code> functions.
</p>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Single image
papilio &lt;- getimg(system.file("testdata/images/butterflies/papilio.png", package = "pavo"))

# Multiple images
snakes &lt;- getimg(system.file("testdata/images/snakes", package = "pavo"))
</code></pre>

<hr>
<h2 id='getspec'>Import spectra files</h2><span id='topic+getspec'></span>

<h3>Description</h3>

<p>Finds and imports spectra files from a folder. Currently works
for reflectance files generated in Ocean Optics SpectraSuite (USB2000,
USB4000 and Jaz spectrometers), CRAIC software (after exporting) and
Avantes (before or after exporting).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getspec(
  where = getwd(),
  ext = "txt",
  lim = c(300, 700),
  decimal = ".",
  sep = NULL,
  subdir = FALSE,
  subdir.names = FALSE,
  ignore.case = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getspec_+3A_where">where</code></td>
<td>
<p>Folder in which files are located (defaults to current working
directory).</p>
</td></tr>
<tr><td><code id="getspec_+3A_ext">ext</code></td>
<td>
<p>File extension to be searched for, without the &quot;.&quot; (defaults to
<code>txt</code>). You can also use a character vector to specify multiple file
extensions.</p>
</td></tr>
<tr><td><code id="getspec_+3A_lim">lim</code></td>
<td>
<p>A vector with two numbers determining the wavelength limits to be
considered (defaults to <code>c(300, 700)</code>).</p>
</td></tr>
<tr><td><code id="getspec_+3A_decimal">decimal</code></td>
<td>
<p>Character to be used to identify decimal plates
(defaults to <code>.</code>).</p>
</td></tr>
<tr><td><code id="getspec_+3A_sep">sep</code></td>
<td>
<p>Column delimiting characters to be considered in addition to the
default (which are: tab, space, and &quot;;&quot;)</p>
</td></tr>
<tr><td><code id="getspec_+3A_subdir">subdir</code></td>
<td>
<p>Should subdirectories within the <code>where</code> folder be included in
the search? (defaults to <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="getspec_+3A_subdir.names">subdir.names</code></td>
<td>
<p>Should subdirectory path be included in the name of the
spectra? (defaults to <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="getspec_+3A_ignore.case">ignore.case</code></td>
<td>
<p>Should the extension search be case insensitive? (defaults
to <code>TRUE</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You can customise the type of parallel processing used by this function with
the <code><a href="future.html#topic+plan">future::plan()</a></code> function. This works on all operating systems, as well
as high performance computing (HPC) environment. Similarly, you can customise
the way progress is shown with the <code><a href="progressr.html#topic+handlers">progressr::handlers()</a></code> functions
(progress bar, acoustic feedback, nothing, etc.)
</p>


<h3>Value</h3>

<p>A data frame, of class <code>rspec</code>, containing individual imported
spectral files as columns.
Reflectance values are interpolated to the nearest wavelength integer.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>
<p>Hugo Gruson <a href="mailto:hugo.gruson+R@normalesup.org">hugo.gruson+R@normalesup.org</a>
</p>


<h3>References</h3>

<p>Gruson H, White TE, Maia R (2019) lightr: import spectral data
and metadata in R. Journal of Open Source Software, 4(43), 1857,
<a href="https://doi.org/10.21105/joss.01857">doi:10.21105/joss.01857</a>.
</p>


<h3>See Also</h3>

<p><code><a href="lightr.html#topic+lr_get_spec">lightr::lr_get_spec()</a></code> for a more flexible version of this function
(e.g. uninterpolated wavelengths), and <code><a href="lightr.html#topic+lr_get_metadata">lightr::lr_get_metadata()</a></code> for the retrieval
and import of spectral metadata.
See <a href="https://docs.ropensci.org/lightr/">https://docs.ropensci.org/lightr/</a> for the complete, and up-to-date, list
of supported file formats.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Import and inspect example spectral data with a range of set to 400-700nm.
rspecdata &lt;- getspec(system.file("testdata", package = "lightr"), ext = "ttt", lim = c(400, 700))
head(rspecdata)

</code></pre>

<hr>
<h2 id='hexagon'>Colour hexagon</h2><span id='topic+hexagon'></span>

<h3>Description</h3>

<p>Calculates coordinates and colourimetric variables that represent reflectance spectra
in the hymenopteran colour hexagon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hexagon(vismodeldata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hexagon_+3A_vismodeldata">vismodeldata</code></td>
<td>
<p>(required) quantum catch colour data. Can be either the result
from <code><a href="#topic+vismodel">vismodel()</a></code> or independently calculated data (in the form of a data frame
with three columns named 's', 'm', 'l', representing a trichromatic viewer).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of class <code>colspace</code> consisting of the following columns:
</p>

<ul>
<li> <p><code>s</code>, <code>m</code>, <code>l</code>: the quantum catch data used to calculate
the remaining variables
</p>
</li>
<li> <p><code>x</code>, <code>y</code>: cartesian coordinates in the colour hexagon.
</p>
</li>
<li> <p><code>h.theta</code>: hue angle theta (in degrees), with 0-degrees at the 1200
angle, increasing clockwise.
</p>
</li>
<li> <p><code>r.vec</code>: the r vector (saturation, distance from the center).
</p>
</li>
<li> <p><code>sec.fine</code>: fine 'hue sector', wherein the full hexagon is composed
of 36 10-degree sectors, with 0-degrees at the 1200 angle.
</p>
</li>
<li> <p><code>sec.coarse</code>: coarse 'hue sector', wherein the full hexagon is
composed of five sectors: blue, bluegreen, green, uvgreen, uv, and uvblue.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Chittka L. (1992). The colour hexagon: a chromaticity diagram
based on photoreceptor excitations as a generalized representation of
colour opponency. Journal of Comparative Physiology A, 170(5), 533-543.
</p>
<p>Chittka L, Shmida A, Troje N, Menzel R. (1994). Ultraviolet as a
component of flower reflections, and the colour perception of Hymenoptera.
Vision research, 34(11), 1489-1508.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers,
  visual = "apis", qcatch = "Ei", relative = FALSE,
  vonkries = TRUE, achromatic = "l", bkg = "green"
)
flowers.hex &lt;- colspace(vis.flowers, space = "hexagon")
</code></pre>

<hr>
<h2 id='hexplot'>Plot a colour hexagon</h2><span id='topic+hexplot'></span>

<h3>Description</h3>

<p>Produces a colour hexagon plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hexplot(
  hexdata,
  achro = TRUE,
  labels = TRUE,
  sectors = c("none", "fine", "coarse"),
  sec.col = "grey",
  out.lwd = 1,
  out.lty = 1,
  out.lcol = "black",
  labels.cex = 1,
  achrosize = 0.8,
  achrocol = "grey",
  square = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hexplot_+3A_hexdata">hexdata</code></td>
<td>
<p>(required) a data frame, possibly a result from the
<code><a href="#topic+colspace">colspace()</a></code> or <code><a href="#topic+hexagon">hexagon()</a></code> function, containing values for the 'x' and 'y'
coordinates as columns (labeled as such)</p>
</td></tr>
<tr><td><code id="hexplot_+3A_achro">achro</code></td>
<td>
<p>should a point be plotted at the origin (defaults to <code>TRUE</code>)?</p>
</td></tr>
<tr><td><code id="hexplot_+3A_labels">labels</code></td>
<td>
<p>logical. Should the name of each cone be printed next to the
corresponding vertex?</p>
</td></tr>
<tr><td><code id="hexplot_+3A_sectors">sectors</code></td>
<td>
<p>plot the bee-hue sector dividers? Options are:
</p>

<ul>
<li> <p><code>"none"</code>: No sectors (default)
</p>
</li>
<li> <p><code>"fine"</code>: 36 10-degree sectors
</p>
</li>
<li> <p><code>"coarse"</code>: six bee-hue sectors (blue, blue-green, green, uv-green, uv,
uv-blue).
</p>
</li></ul>
</td></tr>
<tr><td><code id="hexplot_+3A_sec.col">sec.col</code></td>
<td>
<p>line colour of hue sector dividers. Defaults to <code>"grey"</code>.</p>
</td></tr>
<tr><td><code id="hexplot_+3A_out.lwd">out.lwd</code>, <code id="hexplot_+3A_out.lcol">out.lcol</code>, <code id="hexplot_+3A_out.lty">out.lty</code></td>
<td>
<p>graphical parameters for the plot outline.</p>
</td></tr>
<tr><td><code id="hexplot_+3A_labels.cex">labels.cex</code></td>
<td>
<p>size of the arrow labels.</p>
</td></tr>
<tr><td><code id="hexplot_+3A_achrosize">achrosize</code></td>
<td>
<p>size of the point at the origin when <code>achro = TRUE</code>
(defaults to <code>0.8</code>).</p>
</td></tr>
<tr><td><code id="hexplot_+3A_achrocol">achrocol</code></td>
<td>
<p>color of the point at the origin <code>achro = TRUE</code> (defaults to
<code>'grey'</code>).</p>
</td></tr>
<tr><td><code id="hexplot_+3A_square">square</code></td>
<td>
<p>logical. Should the aspect ratio of the plot be held to 1:1?
(defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="hexplot_+3A_...">...</code></td>
<td>
<p>additional graphical options. See <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Chittka L. (1992). The colour hexagon: a chromaticity diagram
based on photoreceptor excitations as a generalized representation of
colour opponency. Journal of Comparative Physiology A, 170(5), 533-543.
</p>
<p>Chittka L, Shmida A, Troje N, Menzel R. (1994). Ultraviolet as a
component of flower reflections, and the colour perception of Hymenoptera.
Vision research, 34(11), 1489-1508.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers,
  visual = "apis", qcatch = "Ei", relative = FALSE,
  vonkries = TRUE, achromatic = "l", bkg = "green"
)
hex.flowers &lt;- colspace(vis.flowers, space = "hexagon")
plot(hex.flowers)
</code></pre>

<hr>
<h2 id='img_conversion'>Convert images between class rimg and cimg or magick-image</h2><span id='topic+img_conversion'></span><span id='topic+rimg2cimg'></span><span id='topic+rimg2magick'></span>

<h3>Description</h3>

<p>Conveniently convert single objects of class <code>rimg</code> to class <code>cimg</code> (from the
package <code>imager</code>) or <code>magick-image</code> (from the package <code>magick</code>), both of which
contains a suite of useful image-processing capabilities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rimg2cimg(image)

rimg2magick(image)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="img_conversion_+3A_image">image</code></td>
<td>
<p>an object of class <code>rimg</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>an image of the specified class
</p>


<h3>Note</h3>

<p>Attributes (e.g. scales, color-classes) will not be preserved following
conversion from class <code>rimg</code>, so it's best to use early in the analysis workflow.
</p>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Hugo Gruson <a href="mailto:hugo.gruson+R@normalesup.org">hugo.gruson+R@normalesup.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>papilio &lt;- getimg(system.file("testdata/images/butterflies/papilio.png", package = "pavo"))

# Convert from class rimg to cimg
if (requireNamespace("imager", quiety = TRUE)) {
  papilio_cimg &lt;- rimg2cimg(papilio)
  class(papilio_cimg)
}


# Convert from class rimg to magick-image
papilio_magick &lt;- rimg2magick(papilio)
class(papilio_magick)

</code></pre>

<hr>
<h2 id='irrad2flux'>Converts between irradiance and photon (quantum) flux</h2><span id='topic+irrad2flux'></span><span id='topic+flux2irrad'></span>

<h3>Description</h3>

<p>Some spectrometers will give illuminant values in units of irradiance
(&mu;Watt.cm<sup>-2</sup>),
but physiological models require illuminants in units of photon (quantum) flux
(&mu;mol.s<sup>-1</sup>.m<sup>-2</sup>).
The functions <code><a href="#topic+irrad2flux">irrad2flux()</a></code> and <code><a href="#topic+flux2irrad">flux2irrad()</a></code> allows for easy conversion of
<code>rspec</code> objects between these units.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>irrad2flux(rspecdata)

flux2irrad(rspecdata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="irrad2flux_+3A_rspecdata">rspecdata</code></td>
<td>
<p>(required) a rspec object containing illuminant values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a converted <code>rspec</code> object.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>

<hr>
<h2 id='is.colspace'>Test if object is of class 'colspace'</h2><span id='topic+is.colspace'></span>

<h3>Description</h3>

<p>Test if object is of class 'colspace'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.colspace(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.colspace_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a logical value indicating whether the object is of class <code>colspace</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+colspace">colspace()</a></code>
</p>

<hr>
<h2 id='is.vismodel'>Test if object is of class 'vismodel'</h2><span id='topic+is.vismodel'></span>

<h3>Description</h3>

<p>Test if object is of class 'vismodel'
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.vismodel(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.vismodel_+3A_object">object</code></td>
<td>
<p>an <span class="rlang"><b>R</b></span> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a logical value indicating whether the object is of class <code>vismodel</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+vismodel">vismodel()</a></code>
</p>

<hr>
<h2 id='jnd2xyz'>Convert JND distances into perceptually-corrected Cartesian coordinates</h2><span id='topic+jnd2xyz'></span>

<h3>Description</h3>

<p>Converts a <code><a href="#topic+coldist">coldist()</a></code> output into Cartesian coordinates that are
perceptually-corrected (i.e. noise-weighted Euclidean distances)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jnd2xyz(
  coldistres,
  center = TRUE,
  rotate = TRUE,
  rotcenter = c("mean", "achro"),
  ref1 = "l",
  ref2 = "u",
  axis1 = c(1, 1, 0),
  axis2 = c(0, 0, 1)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jnd2xyz_+3A_coldistres">coldistres</code></td>
<td>
<p>(required) the output from a <code><a href="#topic+coldist">coldist()</a></code> call.</p>
</td></tr>
<tr><td><code id="jnd2xyz_+3A_center">center</code></td>
<td>
<p>logical indicating if the data should be centered on its centroid
(defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="jnd2xyz_+3A_rotate">rotate</code></td>
<td>
<p>logical indicating if the data should be rotated (defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="jnd2xyz_+3A_rotcenter">rotcenter</code></td>
<td>
<p>should the vectors for rotation be centered in the achromatic
center (&quot;achro&quot;) or the data centroid (&quot;mean&quot;, the default)?</p>
</td></tr>
<tr><td><code id="jnd2xyz_+3A_ref1">ref1</code></td>
<td>
<p>the cone to be used as a the first reference. May be <code>NULL</code>
(for no first rotation in the 3-dimensional case) or must match name
in the original data that was used for <code><a href="#topic+coldist">coldist()</a></code>. Defaults to 'l'.</p>
</td></tr>
<tr><td><code id="jnd2xyz_+3A_ref2">ref2</code></td>
<td>
<p>the cone to be used as a the second reference. May be <code>NULL</code>
(for no first rotation in the 3-dimensional case) or must match name
in the original data that was used for <code><a href="#topic+coldist">coldist()</a></code>. Defaults to 'u'.
(only used if data has 3 dimensions).</p>
</td></tr>
<tr><td><code id="jnd2xyz_+3A_axis1">axis1</code></td>
<td>
<p>A vector of length 3 composed of 0's and 1's, with
1's representing the axes (x, y, z) to rotate around. Defaults to c(1, 1, 0), such
that the rotation aligns with the xy plane (only used if data has 2 or 3 dimensions).
Ignored if <code>ref1</code> is <code>NULL</code> (in 3-dimensional case only)</p>
</td></tr>
<tr><td><code id="jnd2xyz_+3A_axis2">axis2</code></td>
<td>
<p>A vector of length 3 composed of 0's and 1's, with
1's representing the axes (x, y, z) to rotate around. Defaults to c(0, 0, 1), such
that the rotation aligns with the z axis (only used if data has 3 dimensions).
Ignored if <code>ref2</code> is <code>NULL</code> (in 3-dimensional case only)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Pike, T.W. (2012). Preserving perceptual distances in chromaticity diagrams.
Behavioral Ecology, 23, 723-728.
</p>
<p>Maia, R., White, T. E., (2018) Comparing colors using visual models.
Behavioral Ecology, ary017 <a href="https://doi.org/10.1093/beheco/ary017">doi:10.1093/beheco/ary017</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load floral reflectance spectra
data(flowers)

# Estimate quantum catches visual phenotype of a Blue Tit
vis.flowers &lt;- vismodel(flowers, visual = 'bluetit')

# Estimate noise-weighted colour distances between all flowers
cd.flowers &lt;- coldist(vis.flowers)

# Convert points to Cartesian coordinates in which Euclidean distances are
# noise-weighted.
jnd2xyz(cd.flowers)
</code></pre>

<hr>
<h2 id='jndplot'>Perceptually-corrected chromaticity diagrams</h2><span id='topic+jndplot'></span>

<h3>Description</h3>

<p>Plot options for <code><a href="#topic+jnd2xyz">jnd2xyz</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jndplot(
  x,
  arrow = c("relative", "absolute", "none"),
  achro = FALSE,
  arrow.labels = TRUE,
  arrow.col = "darkgrey",
  arrow.p = 1,
  labels.cex = 1,
  margin = "recommended",
  square = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jndplot_+3A_x">x</code></td>
<td>
<p>(required) the output from a <code><a href="#topic+jnd2xyz">jnd2xyz()</a></code> call.</p>
</td></tr>
<tr><td><code id="jndplot_+3A_arrow">arrow</code></td>
<td>
<p>If and how arrows indicating receptor vectors should be drawn.
Options are <code>"relative"</code> (default), <code>"absolute"</code> or <code>"none"</code>. See
description.</p>
</td></tr>
<tr><td><code id="jndplot_+3A_achro">achro</code></td>
<td>
<p>Logical. Should the achromatic variable be plotted as a
dimension? (only available for dichromats and trichromats, defaults to
<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="jndplot_+3A_arrow.labels">arrow.labels</code></td>
<td>
<p>Logical. Should labels be plotted for receptor arrows?
(defaults to <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="jndplot_+3A_arrow.col">arrow.col</code></td>
<td>
<p>color of the arrows and labels.</p>
</td></tr>
<tr><td><code id="jndplot_+3A_arrow.p">arrow.p</code></td>
<td>
<p>scaling factor for arrows.</p>
</td></tr>
<tr><td><code id="jndplot_+3A_labels.cex">labels.cex</code></td>
<td>
<p>size of the arrow labels.</p>
</td></tr>
<tr><td><code id="jndplot_+3A_margin">margin</code></td>
<td>
<p>accepts either <code>"recommended"</code>, where the function will choose
margin attributes, or a numerical vector of the form <code>c(bottom, left, top, right)</code> which gives the number of lines of margin to be specified on the
four sides of the plot. (Default varies depending on plot dimensionality).</p>
</td></tr>
<tr><td><code id="jndplot_+3A_square">square</code></td>
<td>
<p>logical. Should the aspect ratio of the plot be held to 1:1?
(defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="jndplot_+3A_...">...</code></td>
<td>
<p>additional parameters to be passed to <code><a href="base.html#topic+plot">plot()</a></code>, <code><a href="graphics.html#topic+arrows">arrows()</a></code> and
<code><a href="graphics.html#topic+persp">graphics::persp()</a></code> (for 3D plots).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Creates a plot, details of the plot depend on the input data.
</p>


<h3>Note</h3>

<p>the <code>arrow</code> argument accepts three options:
</p>

<ul>
<li> <p><code>"relative"</code>: With this option, arrows will be made relative to the data.
Arrows will be centered on the data centroid, and will have an arbitrary
length of half the average pairwise distance between points, which can be
scaled with the <code>arrow.p</code> argument.
</p>
</li>
<li> <p><code>"absolute"</code>: With this option, arrows will be made to reflect the visual
system underlying the data. Arrows will be centered on the achromatic point
in colourspace, and will have length equal to the distance to a
monochromatic point (i.e. a colour that stimulates approximately 99.9% of
that receptor alone). Arrows can still be scaled using the <code>arrow.p</code>
argument, in which case they cannot be interpreted as described.
</p>
</li>
<li> <p><code>"none"</code>: no arrows will be included.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Pike, T.W. (2012). Preserving perceptual distances in
chromaticity diagrams. Behavioral Ecology, 23, 723-728.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load floral reflectance spectra
data(flowers)

# Estimate quantum catches visual phenotype of a Blue Tit
vis.flowers &lt;- vismodel(flowers, visual = 'bluetit')

# Estimate noise-weighted colour distances between all flowers
cd.flowers &lt;- coldist(vis.flowers)

# Convert points to Cartesian coordinates in which Euclidean distances are
# noise-weighted.
propxyz &lt;- jnd2xyz(cd.flowers)

# Plot the floral spectra in 'noise-corrected' space
plot(propxyz)

</code></pre>

<hr>
<h2 id='jndrot'>Rotate Cartesian coordinates obtained from <code><a href="#topic+jnd2xyz">jnd2xyz()</a></code></h2><span id='topic+jndrot'></span>

<h3>Description</h3>

<p>Rotate Cartesian coordinates obtained from <code><a href="#topic+jnd2xyz">jnd2xyz()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jndrot(
  jnd2xyzres,
  center = c("mean", "achro"),
  ref1 = "l",
  ref2 = "u",
  axis1 = c(1, 1, 0),
  axis2 = c(0, 0, 1)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jndrot_+3A_jnd2xyzres">jnd2xyzres</code></td>
<td>
<p>(required) the output from a <code><a href="#topic+jnd2xyz">jnd2xyz()</a></code> call.</p>
</td></tr>
<tr><td><code id="jndrot_+3A_center">center</code></td>
<td>
<p>should the vectors for rotation be centered in the achromatic
center (&quot;achro&quot;) or the data centroid (&quot;mean&quot;, the default)?</p>
</td></tr>
<tr><td><code id="jndrot_+3A_ref1">ref1</code></td>
<td>
<p>the cone to be used as a the first reference. May be <code>NULL</code>
(for no first rotation in the 3-dimensional case) or must match name
in the original data that was used for <code><a href="#topic+coldist">coldist()</a></code>. Defaults to 'l'.</p>
</td></tr>
<tr><td><code id="jndrot_+3A_ref2">ref2</code></td>
<td>
<p>the cone to be used as a the second reference. May be <code>NULL</code>
(for no first rotation in the 3-dimensional case) or must match name
in the original data that was used for <code><a href="#topic+coldist">coldist()</a></code>. Defaults to 'u'.
(only used if data has 3 dimensions).</p>
</td></tr>
<tr><td><code id="jndrot_+3A_axis1">axis1</code></td>
<td>
<p>A vector of length 3 composed of 0's and 1's, with
1's representing the axes (x, y, z) to rotate around. Defaults to c(1, 1, 0), such
that the rotation aligns with the xy plane (only used if data has 2 or 3 dimensions).
Ignored if <code>ref1</code> is <code>NULL</code> (in 3-dimensional case only)</p>
</td></tr>
<tr><td><code id="jndrot_+3A_axis2">axis2</code></td>
<td>
<p>A vector of length 3 composed of 0's and 1's, with
1's representing the axes (x, y, z) to rotate around. Defaults to c(0, 0, 1), such
that the rotation aligns with the z axis (only used if data has 3 dimensions).
Ignored if <code>ref2</code> is <code>NULL</code> (in 3-dimensional case only)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load floral reflectance spectra
data(flowers)

# Estimate quantum catches visual phenotype of a Blue Tit
vis.flowers &lt;- vismodel(flowers, visual = 'bluetit')

# Estimate noise-weighted colour distances between all flowers
cd.flowers &lt;- coldist(vis.flowers)

# Convert points to Cartesian coordinates in which Euclidean distances are
# noise-weighted, before rotating them about the data centroid
jndrot(jnd2xyz(cd.flowers))

</code></pre>

<hr>
<h2 id='legendtetra'>Add legend to a static tetrahedral colourspace</h2><span id='topic+legendtetra'></span>

<h3>Description</h3>

<p>Adds a legend to a static tetrahedral colourspace plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legendtetra(x = 0.8, y = 1.2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="legendtetra_+3A_x">x</code>, <code id="legendtetra_+3A_y">y</code></td>
<td>
<p>position of the legend relative to plot limits
(usually a value between 0 and 1, but because of the perspective distortion,
values greater than 1 are possible)</p>
</td></tr>
<tr><td><code id="legendtetra_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code><a href="graphics.html#topic+legend">legend()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code><a href="#topic+legendtetra">legendtetra()</a></code> adds a legend to a static tetrahedral colourspace plot.
for additional information on which arguments are necessary and how they are used,
see <code><a href="graphics.html#topic+legend">legend()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sicalis)

vis_sicalis &lt;- vismodel(sicalis)
tcs_sicalis &lt;- colspace(vis_sicalis)

cols &lt;- c("#1B9E77", "#D95F02", "#7570B3")
plot(tcs_sicalis, col = cols)
legendtetra(
  legend = c("Crown", "Throat", "Breast"),
  col = cols, pch = 16
)
</code></pre>

<hr>
<h2 id='merge.rspec'>Merge two rspec objects</h2><span id='topic+merge.rspec'></span>

<h3>Description</h3>

<p>Merges two <code>rspec</code> objects into a single <code>rspec</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rspec'
merge(x, y, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merge.rspec_+3A_x">x</code>, <code id="merge.rspec_+3A_y">y</code></td>
<td>
<p>(required) <code>rspec</code> objects to merge.</p>
</td></tr>
<tr><td><code id="merge.rspec_+3A_...">...</code></td>
<td>
<p>additional class arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code>rspec</code> for use with <code>pavo</code> functions.
Will use <code>by = "wl"</code> if unspecified, or automatically append <code>wl</code> to the
<code>by</code> argument if one is specified.
</p>


<h3>Author(s)</h3>

<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+as.rspec">as.rspec()</a></code>, <code><a href="#topic+aggspec">aggspec()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load angle-resolved reflectance data for a green-winged teal, and
# split it in two
data(teal)
teal1 &lt;- teal[, c(1, 2:5)]
teal2 &lt;- teal[, c(1, 6:13)]

# Merge the two split datasets back into one, with a shared 'wl' column
teal.mer &lt;- merge(teal1, teal2, by = "wl")

# Examine the results, and compare the original to the (identical)
# reconstructed version
plot(teal.mer)
plot(teal)
identical(teal.mer, teal)

# Or an equivalent method, which also allows for the merging of more than one rspec
# object at a time (simply add further objects to the list())
teal.mer2 &lt;- do.call(merge, list(teal1, teal2))

# Check equivalence
identical(teal.mer2, teal)

</code></pre>

<hr>
<h2 id='peakshape'>Peak shape descriptors</h2><span id='topic+peakshape'></span>

<h3>Description</h3>

<p>Calculates height, location and width of peak at the reflectance midpoint (FWHM).
Note: bounds should be set wide enough to incorporate all minima in spectra. Smoothing
spectra using <code><a href="#topic+procspec">procspec()</a></code> is also recommended.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakshape(
  rspecdata,
  select = NULL,
  lim = NULL,
  plot = TRUE,
  ask = FALSE,
  absolute.min = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakshape_+3A_rspecdata">rspecdata</code></td>
<td>
<p>(required) a data frame, possibly of class <code>rspec</code>, which
contains a column containing a wavelength range, named 'wl', and spectra
data in remaining columns.</p>
</td></tr>
<tr><td><code id="peakshape_+3A_select">select</code></td>
<td>
<p>specification of which spectra to plot. Can be a numeric vector
or factor (e.g., <code>sex == "male"</code>)</p>
</td></tr>
<tr><td><code id="peakshape_+3A_lim">lim</code></td>
<td>
<p>a vector specifying the wavelength range to analyze.</p>
</td></tr>
<tr><td><code id="peakshape_+3A_plot">plot</code></td>
<td>
<p>logical. Should plots indicating calculated parameters be returned?
(Defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="peakshape_+3A_ask">ask</code></td>
<td>
<p>logical, specifies whether user input needed to plot multiple plots
when number of spectra to analyze is greater than 1 (defaults to <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="peakshape_+3A_absolute.min">absolute.min</code></td>
<td>
<p>logical. If <code>TRUE</code>, full width at half maximum will be
calculated using the absolute minimum reflectance of the spectrum, even if
that value falls outside the range specified by <code>lim</code>. (defaults to <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="peakshape_+3A_...">...</code></td>
<td>
<p>additional arguments to be passed to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing column names (id); peak height (max value, B3), location (hue, H1) and full width
at half maximum (FWHM), as well as half widths on left (HWHM.l) and right side of peak (HWHM.r). Incl.min column
indicates whether user-defined bounds incorporate the actual minima of the spectra.
Function will return a warning if not.
</p>


<h3>Author(s)</h3>

<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>
<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>
<p>Hugo Gruson <a href="mailto:hugo.gruson+R@normalesup.org">hugo.gruson+R@normalesup.org</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+procspec">procspec()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(teal)

peakshape(teal, select = 3)
peakshape(teal, select = 10)

# Use wavelength bounds to narrow in on peak of interest
peakshape(teal, select = 10, lim = c(400, 550))
</code></pre>

<hr>
<h2 id='plot.colspace'>Plot spectra in a colourspace</h2><span id='topic+plot.colspace'></span>

<h3>Description</h3>

<p>Plots reflectance spectra in the appropriate colourspace.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'colspace'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.colspace_+3A_x">x</code></td>
<td>
<p>(required) an object of class <code>colspace</code>.</p>
</td></tr>
<tr><td><code id="plot.colspace_+3A_...">...</code></td>
<td>
<p>additional graphical options, which vary by modeled <code>space</code>. Refer
to their individual documentation:
</p>

<ul>
<li> <p><code><a href="#topic+diplot">diplot()</a></code>: dichromat space
</p>
</li>
<li> <p><code><a href="#topic+triplot">triplot()</a></code>: trichromat space
</p>
</li>
<li> <p><code><a href="#topic+tetraplot">tetraplot()</a></code>: tetrahedral space
</p>
</li>
<li> <p><code><a href="#topic+catplot">catplot()</a></code>: categorical space
</p>
</li>
<li> <p><code><a href="#topic+hexplot">hexplot()</a></code>: colour hexagon
</p>
</li>
<li> <p><code><a href="#topic+cocplot">cocplot()</a></code>: colour-opponent-coding space
</p>
</li>
<li> <p><code><a href="#topic+cieplot">cieplot()</a></code>: cie spaces
</p>
</li>
<li> <p><code><a href="#topic+segplot">segplot()</a></code>: segment analysis space
</p>
</li>
<li> <p><code><a href="#topic+jndplot">jndplot()</a></code>: perceptual, 'noise corrected' space (for the results of <code><a href="#topic+jnd2xyz">jnd2xyz()</a></code>)
</p>
</li></ul>

<p>Also see <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A colourspace plot appropriate to the input data.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>
<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Smith T, Guild J. (1932) The CIE colorimetric standards and their
use. Transactions of the Optical Society, 33(3), 73-134.
</p>
<p>Westland S, Ripamonti C, Cheung V. (2012). Computational colour
science using MATLAB. John Wiley &amp; Sons.
</p>
<p>Chittka L. (1992). The colour hexagon: a chromaticity diagram
based on photoreceptor excitations as a generalized representation of
colour opponency. Journal of Comparative Physiology A, 170(5), 533-543.
</p>
<p>Chittka L, Shmida A, Troje N, Menzel R. (1994). Ultraviolet as a
component of flower reflections, and the colour perception of Hymenoptera.
Vision research, 34(11), 1489-1508.
</p>
<p>Troje N. (1993). Spectral categories in the learning behaviour of
blowflies. Zeitschrift fur Naturforschung C, 48, 96-96.
</p>
<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage
color in a tetrahedral color space: A phylogenetic analysis of new world
buntings. The American Naturalist, 171(6), 755-776.
</p>
<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour
patterns as birds see them. Biological Journal Of The Linnean Society,
86(4), 405-431.
</p>
<p>Kelber A, Vorobyev M, Osorio D. (2003). Animal colour vision -
behavioural tests and physiological concepts. Biological Reviews, 78, 81 -
118.
</p>
<p>Backhaus W. (1991). Color opponent coding in the visual system of
the honeybee. Vision Research, 31, 1381-1397.
</p>
<p>Endler, J. A. (1990) On the measurement and classification of
color in studies of animal color patterns. Biological Journal of the
Linnean Society, 41, 315-352.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+plot">plot()</a></code>, <code><a href="#topic+points.colspace">points.colspace()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
data(sicalis)

# Dichromat
vis.flowers &lt;- vismodel(flowers, visual = "canis")
di.flowers &lt;- colspace(vis.flowers, space = "di")
plot(di.flowers)

# Colour hexagon
vis.flowers &lt;- vismodel(flowers,
  visual = "apis", qcatch = "Ei", relative = FALSE,
  vonkries = TRUE, achromatic = "l", bkg = "green"
)
hex.flowers &lt;- colspace(vis.flowers, space = "hexagon")
plot(hex.flowers, sectors = "coarse")

# Tetrahedron (static)
vis.sicalis &lt;- vismodel(sicalis, visual = "avg.uv")
tcs.sicalis &lt;- colspace(vis.sicalis, space = "tcs")
plot(tcs.sicalis)


# Tetrahedron (interactive)
vis.sicalis &lt;- vismodel(sicalis, visual = "avg.uv")
tcs.sicalis &lt;- colspace(vis.sicalis, space = "tcs")
tcsplot(tcs.sicalis, size = 0.005)

## Add points to interactive tetrahedron
patch &lt;- rep(c("C", "T", "B"), 7)
tcs.crown &lt;- subset(tcs.sicalis, "C")
tcs.breast &lt;- subset(tcs.sicalis, "B")
tcsplot(tcs.crown, col = "blue")
tcspoints(tcs.breast, col = "red")

## Plot convex hull in interactive tetrahedron
tcsplot(tcs.sicalis, col = "blue", size = 0.005)
tcsvol(tcs.sicalis)

</code></pre>

<hr>
<h2 id='plot.rimg'>Plot unprocessed or colour-classified images</h2><span id='topic+plot.rimg'></span>

<h3>Description</h3>

<p>Plot unprocessed or colour-classified image data. If the
images are in a list, they will be stepped through
one by one.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rimg'
plot(x, axes = TRUE, col = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.rimg_+3A_x">x</code></td>
<td>
<p>(required) an image of class rimg, or list thereof.</p>
</td></tr>
<tr><td><code id="plot.rimg_+3A_axes">axes</code></td>
<td>
<p>should axes be drawn? (defaults to <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="plot.rimg_+3A_col">col</code></td>
<td>
<p>optional vector of colours when plotting colour-classified images.
Defaults to the mean RGB values of the k-means centres (i.e. the average 'original'
colours).</p>
</td></tr>
<tr><td><code id="plot.rimg_+3A_...">...</code></td>
<td>
<p>additional graphical parameters. Also see <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an image plot.
</p>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>papilio &lt;- getimg(system.file("testdata/images/butterflies/papilio.png", package = "pavo"))
plot(papilio)

papilio_class &lt;- classify(papilio, kcols = 4)
plot(papilio_class)


# Multiple images
snakes &lt;- getimg(system.file("testdata/images/snakes", package = "pavo"))
plot(snakes)

snakes_class &lt;- classify(snakes, kcols = 3)
plot(snakes_class)


</code></pre>

<hr>
<h2 id='plot.rspec'>Plot spectra</h2><span id='topic+plot.rspec'></span>

<h3>Description</h3>

<p>Plots reflectance spectra in different arrangements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rspec'
plot(
  x,
  select = NULL,
  type = c("overlay", "stack", "heatmap"),
  varying = NULL,
  n = 100,
  labels = FALSE,
  labels.stack = NULL,
  labels.cex = 1,
  wl.guide = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.rspec_+3A_x">x</code></td>
<td>
<p>(required) a data frame, possibly an object of class <code>rspec</code>, with a
column with wavelength data, named 'wl', and the remaining column
containing spectra to plot.</p>
</td></tr>
<tr><td><code id="plot.rspec_+3A_select">select</code></td>
<td>
<p>specification of which spectra to plot. Can be a numeric vector
or factor (e.g., <code>sex == "male"</code>)</p>
</td></tr>
<tr><td><code id="plot.rspec_+3A_type">type</code></td>
<td>
<p>what type of plot should be drawn. Possibilities are:
</p>

<ul>
<li> <p><code>overlay</code> (default) for plotting multiple spectra in a single panel with
a common y-axis.
</p>
</li>
<li> <p><code>stack</code> for plotting multiple spectra in a vertical arrangement.
</p>
</li>
<li> <p><code>heatmap</code> for plotting reflectance values by wavelength and a third
variable (<code>varying</code>).
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot.rspec_+3A_varying">varying</code></td>
<td>
<p>a numeric vector giving values for y-axis in
<code>type = "heatmap"</code>.</p>
</td></tr>
<tr><td><code id="plot.rspec_+3A_n">n</code></td>
<td>
<p>number of bins with which to interpolate colors and <code>varying</code> for
the heatplot.</p>
</td></tr>
<tr><td><code id="plot.rspec_+3A_labels">labels</code></td>
<td>
<p>logical. Add labels identifying each spectrum to the outer plot
margin? Defaults to <code>FALSE</code>. Ignored when <code>type = 'heatmap'</code>.</p>
</td></tr>
<tr><td><code id="plot.rspec_+3A_labels.stack">labels.stack</code></td>
<td>
<p>a vector of labels for spectra when <code>labels = TRUE</code>.
Defaults to the column names from spectral data. Note you will likely want
to adjust the plot margins to accommodate the text labels. See <code>?par()</code> for
guidance on setting margins.</p>
</td></tr>
<tr><td><code id="plot.rspec_+3A_labels.cex">labels.cex</code></td>
<td>
<p>size of the text labels when <code>labels = TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot.rspec_+3A_wl.guide">wl.guide</code></td>
<td>
<p>logical determining whether visible light spectrum should be
added to the x-axis.</p>
</td></tr>
<tr><td><code id="plot.rspec_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code><a href="base.html#topic+plot">plot()</a></code> (or <code><a href="graphics.html#topic+image">image()</a></code> for
<code>"heatmap"</code>).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Hugo Gruson <a href="mailto:hugo.gruson+R@normalesup.org">hugo.gruson+R@normalesup.org</a>
</p>
<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+spec2rgb">spec2rgb()</a></code>, <code><a href="graphics.html#topic+image">image()</a></code>, <code><a href="base.html#topic+plot">plot()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load angle-resolved reflectance data for a green-winged teal
data(teal)

# Create an overlay plot (default)
plot(teal)

# Create an stacked spectral plot
plot(teal, type = "stack")

# Create a reflectance heatmap
plot(teal, type = "heatmap")

</code></pre>

<hr>
<h2 id='plot.sensmod'>Plot absorbance spectra from <code><a href="#topic+sensmodel">sensmodel()</a></code></h2><span id='topic+plot.sensmod'></span>

<h3>Description</h3>

<p>Plot absorbance spectra from <code><a href="#topic+sensmodel">sensmodel()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'sensmod'
plot(
  x,
  select = NULL,
  type = c("overlay", "stack", "heatmap"),
  varying = NULL,
  n = 100,
  labels = FALSE,
  labels.stack = NULL,
  labels.cex = 1,
  wl.guide = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.sensmod_+3A_x">x</code></td>
<td>
<p>(required) a data frame, possibly an object of class <code>rspec</code>, with a
column with wavelength data, named 'wl', and the remaining column
containing spectra to plot.</p>
</td></tr>
<tr><td><code id="plot.sensmod_+3A_select">select</code></td>
<td>
<p>specification of which spectra to plot. Can be a numeric vector
or factor (e.g., <code>sex == "male"</code>)</p>
</td></tr>
<tr><td><code id="plot.sensmod_+3A_type">type</code></td>
<td>
<p>what type of plot should be drawn. Possibilities are:
</p>

<ul>
<li> <p><code>overlay</code> (default) for plotting multiple spectra in a single panel with
a common y-axis.
</p>
</li>
<li> <p><code>stack</code> for plotting multiple spectra in a vertical arrangement.
</p>
</li>
<li> <p><code>heatmap</code> for plotting reflectance values by wavelength and a third
variable (<code>varying</code>).
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot.sensmod_+3A_varying">varying</code></td>
<td>
<p>a numeric vector giving values for y-axis in
<code>type = "heatmap"</code>.</p>
</td></tr>
<tr><td><code id="plot.sensmod_+3A_n">n</code></td>
<td>
<p>number of bins with which to interpolate colors and <code>varying</code> for
the heatplot.</p>
</td></tr>
<tr><td><code id="plot.sensmod_+3A_labels">labels</code></td>
<td>
<p>logical. Add labels identifying each spectrum to the outer plot
margin? Defaults to <code>FALSE</code>. Ignored when <code>type = 'heatmap'</code>.</p>
</td></tr>
<tr><td><code id="plot.sensmod_+3A_labels.stack">labels.stack</code></td>
<td>
<p>a vector of labels for spectra when <code>labels = TRUE</code>.
Defaults to the column names from spectral data. Note you will likely want
to adjust the plot margins to accommodate the text labels. See <code>?par()</code> for
guidance on setting margins.</p>
</td></tr>
<tr><td><code id="plot.sensmod_+3A_labels.cex">labels.cex</code></td>
<td>
<p>size of the text labels when <code>labels = TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot.sensmod_+3A_wl.guide">wl.guide</code></td>
<td>
<p>logical determining whether visible light spectrum should be
added to the x-axis.</p>
</td></tr>
<tr><td><code id="plot.sensmod_+3A_...">...</code></td>
<td>
<p>additional arguments passed to <code><a href="base.html#topic+plot">plot()</a></code> (or <code><a href="graphics.html#topic+image">image()</a></code> for
<code>"heatmap"</code>).</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+plot.rspec">plot.rspec()</a></code>, <code><a href="#topic+sensmodel">sensmodel()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Blue tit visual system based on Hart et al (2000)
bluesens &lt;- sensmodel(c(371, 448, 502, 563),
  beta = FALSE,
  lambdacut = c(330, 413, 507, 572),
  oiltype = c("T", "C", "Y", "R"), om = TRUE
)
plot(bluesens)

# Alternatively, you can specify your own ylab
plot(bluesens, ylab = "absor.")

</code></pre>

<hr>
<h2 id='plotsmooth'>Plot loess smoothed curves</h2><span id='topic+plotsmooth'></span>

<h3>Description</h3>

<p>Plots spectral curves with various levels of loess smoothing to help decide which
loess parameters are best for subsequently smoothing the data (e.g. via <code><a href="#topic+procspec">procspec()</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotsmooth(
  rspecdata,
  minsmooth = 0.05,
  maxsmooth = 0.2,
  curves = 5,
  specnum = "ALL",
  ask = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotsmooth_+3A_rspecdata">rspecdata</code></td>
<td>
<p>(required) a data frame, possibly of class <code>rspec</code>, which
contains a column containing a wavelength range, named 'wl', and spectra
data in remaining columns.</p>
</td></tr>
<tr><td><code id="plotsmooth_+3A_minsmooth">minsmooth</code></td>
<td>
<p>the minimum f value of the loess function to visualize (defaults to <code>0.05</code>).</p>
</td></tr>
<tr><td><code id="plotsmooth_+3A_maxsmooth">maxsmooth</code></td>
<td>
<p>the maximum f value of the loess function to visualize (defaults to <code>0.20</code>).</p>
</td></tr>
<tr><td><code id="plotsmooth_+3A_curves">curves</code></td>
<td>
<p>the number of curves to display on the same plot (defaults to <code>5</code>).</p>
</td></tr>
<tr><td><code id="plotsmooth_+3A_specnum">specnum</code></td>
<td>
<p>the number of spectral curves, from the data frame, to visualize (defaults to <code>ALL</code>).</p>
</td></tr>
<tr><td><code id="plotsmooth_+3A_ask">ask</code></td>
<td>
<p>logical. if <code>TRUE</code>, asks for user input before changing plot pages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Series of plot with curves processed with varying level of loess smoothing
</p>


<h3>Author(s)</h3>

<p>Pierre-Paul Bitton <a href="mailto:bittonp@uwindsor.ca">bittonp@uwindsor.ca</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+procspec">procspec()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load reflectance spectra
data(sicalis)

# Visualise the spectral reflectance curves across a range of smoothing levels
plotsmooth(sicalis, minsmooth = 0.05, maxsmooth = 0.1, curves = 7, specnum = 6)

</code></pre>

<hr>
<h2 id='points.colspace'>Plot points in a colourspace</h2><span id='topic+points.colspace'></span>

<h3>Description</h3>

<p>Add points to a colourspace plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'colspace'
points(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="points.colspace_+3A_x">x</code></td>
<td>
<p>(required) an object of class <code>colspace</code>.</p>
</td></tr>
<tr><td><code id="points.colspace_+3A_...">...</code></td>
<td>
<p>additional graphical options. See <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>points.colspace</code> adds points to a colourspace plot.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>
<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.colspace">plot.colspace()</a></code>
</p>

<hr>
<h2 id='procimg'>Process images</h2><span id='topic+procimg'></span>

<h3>Description</h3>

<p>Specify scales, resize, and/or define focal objects within images.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>procimg(
  image,
  resize = NULL,
  rotate = NULL,
  scaledist = NULL,
  outline = FALSE,
  reclass = NULL,
  smooth = FALSE,
  iterations = 1L,
  col = "red",
  obj_dist = NULL,
  obj_width = NULL,
  eye_res = NULL,
  plotnew = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="procimg_+3A_image">image</code></td>
<td>
<p>(required) image data. Either a single image array, or a number of images
stored in a list. Preferably the result of <code><a href="#topic+getimg">getimg()</a></code>.</p>
</td></tr>
<tr><td><code id="procimg_+3A_resize">resize</code></td>
<td>
<p>an integer specifying a percentage for resizing images, if so desired.
E.g. 50 to half the size of an image, 200 to double it.</p>
</td></tr>
<tr><td><code id="procimg_+3A_rotate">rotate</code></td>
<td>
<p>an integer specifying the angle of image rotation, in degrees. Images
are rotated around the centre, and linearly interpolated.</p>
</td></tr>
<tr><td><code id="procimg_+3A_scaledist">scaledist</code></td>
<td>
<p>an integer, or numeric vector equal in length to the number of images,
specifying the length of the scale in the image(s). Image(s) will then be presented,
and the user asked to select either end of the scale corresponding to the input value.</p>
</td></tr>
<tr><td><code id="procimg_+3A_outline">outline</code></td>
<td>
<p>interactively specify the focal object in
an image by clicking around its outline. The xy-coordinates of the resulting
closed polygon are saved as an attribute, for use in generating a masking layer &amp;
separating animals/plants from backgrounds in further analyses. This is particularly
useful when backgrounds are complex, such as in natural settings.</p>
</td></tr>
<tr><td><code id="procimg_+3A_reclass">reclass</code></td>
<td>
<p>interactively specify an area on a colour-classified image that is
to be reclassified as the numeric value provided. e.g. when <code>reclass = 1</code>, the user
will be asked to select a polygon on the image, within which all colour-category values will be
changes to <code>1</code>.</p>
</td></tr>
<tr><td><code id="procimg_+3A_smooth">smooth</code></td>
<td>
<p>should the polygon specified when <code>outline = TRUE</code> be smoothed
using Chaikin's corner-cuting algorithm? Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="procimg_+3A_iterations">iterations</code></td>
<td>
<p>the number of smoothing iterations, when <code>smooth = TRUE</code>.
Defaults to <code>1</code>.</p>
</td></tr>
<tr><td><code id="procimg_+3A_col">col</code></td>
<td>
<p>the color of the marker points and/or line, when using interactive options.</p>
</td></tr>
<tr><td><code id="procimg_+3A_obj_dist">obj_dist</code>, <code id="procimg_+3A_obj_width">obj_width</code>, <code id="procimg_+3A_eye_res">eye_res</code></td>
<td>
<p>blur the image to model the visual acuity of non-human animals
as per Caves &amp; Johnsen (2018)'s AcuityView 2.0 algorithm. The procedure requires three arguments;
obj_dist is the real-world distance between the viewer and the focal object in the image in the image,
obj_width is the real-world width of the entire image; eye_res is the minimum resolvable angle of the viewer
in degrees. All three arguments are numeric, and any units of measurement are suitable for
obj_dist and obj_width, but they must match. Note that this is the more flexible v2.0 implementation meaning
that any rectangular image is suitable; it need not be square with dimensions a power of 2.
If using this capability, please cite Caves &amp; Johnsen (2018), as per the included reference,
and see note below.</p>
</td></tr>
<tr><td><code id="procimg_+3A_plotnew">plotnew</code></td>
<td>
<p>should plots be opened in a new window? Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="procimg_+3A_...">...</code></td>
<td>
<p>additional graphical parameters. Also see <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an image, or list of images, for use in further
<code>pavo</code> functions.
</p>


<h3>Note</h3>

<p>There are several caveats that should be considered when using the AcuityView
algorithm. First and foremost, the converted image is not what the animal actually sees.
For example, it does not account for edge enhancement and other processing by the retina and
brain that may alter an image. It does, however, show what spatial information
can be detected and then processed by the visual system. Second, the converted
image is static, which does not allow one to assess how movement may reveal the
presence of an otherwise indiscernible object. Third, AcuityView makes several
assumptions about the Modulation Transfer Function (MTF), which describes how
the optical system affects image contrast as a function of the level of detail.
These assumptions include that the MTF is constant over the region of the retina that views the
scene, is circularly symmetrical, and is wavelength independent. For a full discussion and details,
please do read Caves &amp; Johnsen (2018).
</p>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Caves, E. M., &amp; Johnsen, S. (2018). AcuityView: An r package
for portraying the effects of visual acuity on scenes observed by an animal.
Methods in Ecology and Evolution, 9(3), 793-797 <a href="https://doi.org/10.1111/2041-210X.12911">doi:10.1111/2041-210X.12911</a>.
</p>
<p>Chaikin, G. 1974. An algorithm for high speed curve generation.
Computer Graphics and Image Processing 3, 346-349.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  # Interactively add a scale to a single image
  papilio &lt;- getimg(system.file("testdata/images/butterflies/papilio.png", package = "pavo"))
  papilio &lt;- procimg(papilio, scaledist = 10)

  # Interactively assign individual scales to each image,
  # after slightly reducing their size (to 90% of original).
  snakes &lt;- getimg(system.file("testdata/images/snakes", package = "pavo"))
  snakes &lt;- procimg(snakes, scaledist = c(10, 14), resize = 90)

  # Model the appearance of a butterfly given the reduced visual acuity of another
  # animal viewer as per the AcuityView algorithm. Here our butterfly is 60 cm away,
  # the image width is 10 cm, and the minimum resolvable angle of the viewer is 0.2-degrees.
  tiger &lt;- getimg(system.file("testdata/images/tiger.png", package = "pavo"))
  tiger_acuity &lt;- procimg(tiger, obj_dist = 60, obj_width = 10, eye_res = 0.2)
}
</code></pre>

<hr>
<h2 id='procspec'>Process spectra</h2><span id='topic+procspec'></span>

<h3>Description</h3>

<p>Applies normalization and/or smoothing to spectra for further analysis or plotting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>procspec(
  rspecdata,
  opt = c("none", "smooth", "maximum", "minimum", "bin", "sum", "center"),
  fixneg = c("none", "addmin", "zero"),
  span = 0.25,
  bins = 20
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="procspec_+3A_rspecdata">rspecdata</code></td>
<td>
<p>(required) a data frame, possibly of class <code>rspec</code>, which
contains a column containing a wavelength range, named 'wl', and spectra
data in remaining columns.</p>
</td></tr>
<tr><td><code id="procspec_+3A_opt">opt</code></td>
<td>
<p>what type of processing options to apply. User can select multiple
options by providing a vector. Possibilities are:
</p>

<ul>
<li> <p><code>"none"</code> does not perform any processing (default).
</p>
</li>
<li> <p><code>"smooth"</code> applies LOESS smoothing to each spectrum using <code><a href="stats.html#topic+loess.smooth">loess.smooth()</a></code>.
Optimal smoothing parameter can be assessed by using <code><a href="#topic+plotsmooth">plotsmooth()</a></code>.
</p>
</li>
<li> <p><code>"minimum"</code> subtracts the minimum from each individual spectra.
</p>
</li>
<li> <p><code>"maximum"</code> divides each spectrum by its maximum value.
</p>
</li>
<li> <p><code>"sum"</code> divides each spectrum by summed values.
</p>
</li>
<li> <p><code>"bin"</code> bins each spectrum into the specified number of bins. <code>bins</code>
argument must be set.
</p>
</li>
<li> <p><code>"center"</code> centers individual spectra by subtracting mean reflectance from
all values.
</p>
</li></ul>
</td></tr>
<tr><td><code id="procspec_+3A_fixneg">fixneg</code></td>
<td>
<p>how to handle negative values. Possibilities are:
</p>

<ul>
<li> <p><code>"none"</code> does not perform negative value correction (default).
</p>
</li>
<li> <p><code>"zero"</code> sets all negative values to zero.
</p>
</li>
<li> <p><code>"addmin"</code> adds the absolute value of the maximally negative values of each
spectra to the reflectance at all other wavelengths (setting the minimum
value to zero, but scaling other values accordingly).
</p>
</li></ul>
</td></tr>
<tr><td><code id="procspec_+3A_span">span</code></td>
<td>
<p>sets the smoothing parameter used by <code><a href="stats.html#topic+loess.smooth">loess.smooth()</a></code>.</p>
</td></tr>
<tr><td><code id="procspec_+3A_bins">bins</code></td>
<td>
<p>sets the number of equally sized wavelength bins for <code>opt = "bin"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of class <code>rspec</code> with the processed data.
</p>


<h3>Author(s)</h3>

<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Cuthill, I., Bennett, A. T. D., Partridge, J. &amp; Maier, E. 1999.
Plumage reflectance and the objective assessment of avian sexual dichromatism.
The American Naturalist, 153, 183-200.
</p>
<p>Montgomerie R. 2006. Analyzing colors. In Hill, G.E, and McGraw,
K.J., eds. Bird Coloration. Volume 1 Mechanisms and measurements. Harvard
University Press, Cambridge, Massachusetts.
</p>
<p>White, T. E., Dalrymple, R. L., Noble D. W. A., O'Hanlon, J. C.,
Zurek, D. B., Umbers, K. D. L. 2015. Reproducible research in the study
of biological coloration. Animal Behaviour, 106, 51-57.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+loess.smooth">loess.smooth()</a></code>, <code><a href="#topic+plotsmooth">plotsmooth()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(teal)
plot(teal, select = 10)

# Smooth data to remove noise
teal.sm &lt;- procspec(teal, opt = "smooth", span = 0.25)
plot(teal.sm, select = 10)

# Normalize to max of unity
teal.max &lt;- procspec(teal, opt = c("max"))
plot(teal.max, select = 10)
</code></pre>

<hr>
<h2 id='projplot'>2D projection of a tetrahedral colourspace</h2><span id='topic+projplot'></span><span id='topic+projpoints'></span>

<h3>Description</h3>

<p>Produces a 2D projection plot of points in a tetrahedral colour space
</p>
<p>Adds points to a tetrahedral colorspace projection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>projplot(tcsdata, ...)

projpoints(tcsdata, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="projplot_+3A_tcsdata">tcsdata</code></td>
<td>
<p>(required) tetrahedral color space coordinates, possibly a result from <code><a href="#topic+colspace">colspace()</a></code>,
containing values for the 'h.theta' and 'h.phi' coordinates as columns (labeled as such).</p>
</td></tr>
<tr><td><code id="projplot_+3A_...">...</code></td>
<td>
<p>additional parameters to be passed to the plotting of data points.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code><a href="#topic+projplot">projplot()</a></code> creates a 2D plot  of color points projected from the tetrahedron
to its encapsulating sphere, and is ideal to visualize differences in hue.
</p>
<p><code><a href="#topic+projpoints">projpoints()</a></code> creates points in a projection color space plot
produced by <code><a href="#topic+projplot">projplot()</a></code>.
</p>


<h3>Note</h3>

<p><code><a href="#topic+projplot">projplot()</a></code> uses the Mollweide projection, and not the Robinson projection, which
has been used in the past. Among other advantages, the Mollweide projection preserves area
relationships within latitudes without distortion.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage
color in a tetrahedral color space: A phylogenetic analysis of new world buntings.
The American Naturalist, 171(6), 755-776.
</p>
<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour patterns
as birds see them. Biological Journal Of The Linnean Society, 86(4), 405-431.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sicalis)
vis.sicalis &lt;- vismodel(sicalis, visual = "avg.uv")
tcs.sicalis &lt;- colspace(vis.sicalis, space = "tcs")
projplot(tcs.sicalis, pch = 16, col = setNames(rep(seq_len(3), 7), rep(c("C", "T", "B"), 7)))
</code></pre>

<hr>
<h2 id='segplot'>Plot the segment-analysis model</h2><span id='topic+segplot'></span>

<h3>Description</h3>

<p>Produces a plot based on Endler's (1990) segment analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>segplot(
  segdata,
  labels = TRUE,
  lab.cex = 0.9,
  out.lwd = 1,
  out.lty = 1,
  out.lcol = "black",
  tick.loc = c(-1, -0.5, 0.5, 1),
  square = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="segplot_+3A_segdata">segdata</code></td>
<td>
<p>(required) a data frame, possibly a result from the
<code><a href="#topic+colspace">colspace()</a></code> or <code><a href="#topic+segspace">segspace()</a></code> function, containing values for 'LM' and 'MS'
as columns (labeled as such).</p>
</td></tr>
<tr><td><code id="segplot_+3A_labels">labels</code></td>
<td>
<p>logical. Should the name of each cone be printed next to the
corresponding vertex?</p>
</td></tr>
<tr><td><code id="segplot_+3A_tick.loc">tick.loc</code></td>
<td>
<p>a numeric vector specifying the location of tick marks on x &amp;
y axes.</p>
</td></tr>
<tr><td><code id="segplot_+3A_square">square</code></td>
<td>
<p>logical. Should the aspect ratio of the plot be held to 1:1?
(defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="segplot_+3A_...">...</code></td>
<td>
<p>additional graphical options. See <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Endler, J. A. (1990) On the measurement and classification of
colour in studies of animal colour patterns. Biological Journal of the
Linnean Society, 41, 315-352.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers, visual = "segment", achromatic = "all")
seg.flowers &lt;- colspace(vis.flowers, space = "segment")
plot(seg.flowers)
</code></pre>

<hr>
<h2 id='segspace'>Segment classification</h2><span id='topic+segspace'></span>

<h3>Description</h3>

<p>Calculates segment classification measures as defined in Endler (1990).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>segspace(vismodeldata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="segspace_+3A_vismodeldata">vismodeldata</code></td>
<td>
<p>(required) quantum catch color data. Can be either the
result from <code><a href="#topic+vismodel">vismodel()</a></code> or independently calculated data (in the form of a
data frame with columns named 'S1', 'S2', 'S3', 'S4', and, optionally,
'lum', representing a generic 'tetrachromatic' viewer).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of class <code><a href="#topic+colspace">colspace</a></code> consisting of the following columns:
</p>

<ul>
<li> <p><code>S1</code>, <code>S2</code>, <code>S3</code>, <code>S4</code>: the relative reflectance at each of the four
segments.
</p>
</li>
<li> <p><code>LM</code>, <code>MS</code>: segment scores
</p>
</li>
<li> <p><code>C</code>, <code>H</code>, <code>B</code>: 'chroma', 'hue' (degrees), and 'brightness' in the segment
classification space
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Pierre-Paul Bitton <a href="mailto:bittonp@uwindsor.ca">bittonp@uwindsor.ca</a>
</p>


<h3>References</h3>

<p>Endler, J. A. (1990) On the measurement and classification of
colour in studies of animal colour patterns. Biological Journal of the
Linnean Society, 41, 315-352.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sicalis)
vis.sic &lt;- vismodel(sicalis, visual = "segment", achromatic = "all")
seg.sic &lt;- colspace(vis.sic, space = "segment")

</code></pre>

<hr>
<h2 id='sensdata'>Retrieve or plot in-built spectral sensitivity data</h2><span id='topic+sensdata'></span>

<h3>Description</h3>

<p>Retrieve (as an rspec object) or plot pavo's in-built spectral sensitivity data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sensdata(
  visual = c("none", "all", "avg.uv", "avg.v", "bluetit", "ctenophorus", "star", "pfowl",
    "apis", "canis", "cie2", "cie10", "musca", "drosophila", "habronattus",
    "rhinecanthus"),
  achromatic = c("none", "all", "bt.dc", "ch.dc", "st.dc", "md.r1", "dm.r1", "ra.dc",
    "cf.r"),
  illum = c("none", "all", "bluesky", "D65", "forestshade"),
  trans = c("none", "all", "bluetit", "blackbird"),
  bkg = c("none", "all", "green"),
  plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sensdata_+3A_visual">visual</code></td>
<td>
<p>visual systems. Options are:
</p>

<ul>
<li> <p><code>"none"</code>: no visual sensitivity data.
</p>
</li>
<li> <p><code>"all"</code>: all visual sensitivity data.
</p>
</li>
<li> <p><code>"apis"</code>: Honeybee <em>Apis mellifera</em> visual system.
</p>
</li>
<li> <p><code>"avg.uv"</code>: average avian UV system.
</p>
</li>
<li> <p><code>"avg.v"</code>: average avian V system.
</p>
</li>
<li> <p><code>"bluetit"</code>: Blue tit <em>Cyanistes caeruleus</em> visual system.
</p>
</li>
<li> <p><code>"canis"</code>: Canid <em>Canis familiaris</em> visual system.
</p>
</li>
<li> <p><code>"cie2"</code>: 2-degree colour matching functions for CIE models of human
colour vision. Functions are linear transformations of the 2-degree cone fundamentals
of Stockman &amp; Sharpe (2000), as ratified by the CIE (2006).
</p>
</li>
<li> <p><code>"cie10"</code>: 10-degree colour matching functions for CIE models of human
colour vision. Functions are linear transformations of the 10-degree cone fundamentals
of Stockman &amp; Sharpe (2000), as ratified by the CIE (2006).
</p>
</li>
<li> <p><code>"ctenophorus"</code>: Ornate dragon lizard <em>Ctenophorus ornatus</em>.
</p>
</li>
<li> <p><code>"musca"</code>: Housefly <em>Musca domestica</em> visual system.
</p>
</li>
<li> <p><code>'drosophila'</code>: Vinegar fly <em>Drosophila melanogaster</em> (Sharkey et al. 2020).
</p>
</li>
<li> <p><code>"pfowl"</code>: Peafowl <em>Pavo cristatus</em> visual system.
</p>
</li>
<li> <p><code>"star"</code>: Starling <em>Sturnus vulgaris</em> visual system.
</p>
</li>
<li> <p><code>"habronattus"</code>: Jumping spider <em>Habronattus pyrrithrix</em>.
</p>
</li>
<li> <p><code>"rhinecanthus"</code>: Triggerfish <em>Rhinecanthus aculeatus</em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="sensdata_+3A_achromatic">achromatic</code></td>
<td>
<p>the sensitivity data used to calculate luminance (achromatic)
receptor stimulation. Options are:
</p>

<ul>
<li> <p><code>"none"</code>: no achromatic sensitivity data.
</p>
</li>
<li> <p><code>"all"</code>: all achromatic sensitivity data.
</p>
</li>
<li> <p><code>"bt.dc"</code>: Blue tit <em>Cyanistes caeruleus</em> double cone.
</p>
</li>
<li> <p><code>"ch.dc"</code>: Chicken <em>Gallus gallus</em> double cone.
</p>
</li>
<li> <p><code>"st.dc"</code>: Starling <em>Sturnus vulgaris</em> double cone.
</p>
</li>
<li> <p><code>"cf.r"</code>: Canid <em>Canis familiaris</em> rod
</p>
</li>
<li> <p><code>"md.r1"</code>: Housefly <em>Musca domestica</em> R1-6 photoreceptor.
</p>
</li>
<li> <p><code>'dm.r1'</code>: Vinegar fly <em>Drosophila melanogaster</em> R1-6 photoreceptor.
</p>
</li>
<li> <p><code>"ra.dc"</code>: Triggerfish <em>Rhinecanthus aculeatus</em> double cone.
</p>
</li></ul>
</td></tr>
<tr><td><code id="sensdata_+3A_illum">illum</code></td>
<td>
<p>illuminants. Options are:
</p>

<ul>
<li> <p><code>"none"</code>: no illuminant data.
</p>
</li>
<li> <p><code>"all"</code>: all background spectral data.
</p>
</li>
<li> <p><code>"bluesky"</code> open blue sky.
</p>
</li>
<li> <p><code>"D65"</code>: standard daylight.
</p>
</li>
<li> <p><code>"forestshade"</code> forest shade.
</p>
</li></ul>
</td></tr>
<tr><td><code id="sensdata_+3A_trans">trans</code></td>
<td>
<p>Ocular transmission data. Options are:
</p>

<ul>
<li> <p><code>"none"</code>: no transmission data.
</p>
</li>
<li> <p><code>"all"</code>: all transmission data.
</p>
</li>
<li> <p><code>"bluetit"</code>: blue tit <em>Cyanistes caeruleus</em> ocular transmission (from Hart et al. 2000).
</p>
</li>
<li> <p><code>"blackbird"</code>: blackbird <em>Turdus merula</em> ocular transmission (from Hart et al. 2000).
</p>
</li></ul>
</td></tr>
<tr><td><code id="sensdata_+3A_bkg">bkg</code></td>
<td>
<p>background spectra. Options are:
</p>

<ul>
<li> <p><code>"none"</code>: no background spectral data.
</p>
</li>
<li> <p><code>"all"</code>: all background spectral data.
</p>
</li>
<li> <p><code>"green"</code>: green foliage.
</p>
</li></ul>
</td></tr>
<tr><td><code id="sensdata_+3A_plot">plot</code></td>
<td>
<p>should the spectral data be plotted, or returned instead (defaults to <code>FALSE</code>)?</p>
</td></tr>
<tr><td><code id="sensdata_+3A_...">...</code></td>
<td>
<p>additional graphical options passed to <code><a href="#topic+plot.rspec">plot.rspec()</a></code> when <code>plot = TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>rspec</code> (when <code>plot = FALSE</code>), containing
a wavelength column <code>"wl"</code> and spectral data binned at 1 nm intervals from 300-700 nm.
</p>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Sharkey, C. R., Blanco, J., Leibowitz, M. M., Pinto-Benito, D.,
&amp; Wardill, T. J. (2020). The spectral sensitivity of Drosophila photoreceptors.
Scientific reports, 10(1), 1-13.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Plot the honeybee's receptors
sensdata(visual = "apis", ylab = "Absorbance", plot = TRUE)

# Plot the average UV vs V avian receptors
sensdata(visual = c("avg.v", "avg.uv"), ylab = "Absorbance", plot = TRUE)

# Retrieve the CIE colour matching functions as an rspec object
ciedat &lt;- sensdata(visual = c("cie2", "cie10"))

</code></pre>

<hr>
<h2 id='sensmodel'>Modeling spectral sensitivity</h2><span id='topic+sensmodel'></span>

<h3>Description</h3>

<p>Models spectral sensitivity (with oil droplets; optional) based on peak cone
sensitivity according to the models of Govardovskii et al. (2000) and Hart &amp;
Vorobyev (2005).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sensmodel(
  peaksens,
  range = c(300, 700),
  lambdacut = NULL,
  Bmid = NULL,
  oiltype = NULL,
  beta = TRUE,
  om = NULL,
  integrate = TRUE,
  sensnames = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sensmodel_+3A_peaksens">peaksens</code></td>
<td>
<p>(required) a vector with peak sensitivities for the cones to
model.</p>
</td></tr>
<tr><td><code id="sensmodel_+3A_range">range</code></td>
<td>
<p>a vector of length 2 for the range over which to calculate the
spectral sensitivities (defaults to 300nm to 700nm).</p>
</td></tr>
<tr><td><code id="sensmodel_+3A_lambdacut">lambdacut</code></td>
<td>
<p>a vector of same length as peaksens that lists the cut-off
wavelength value for oil droplets. Needs either <code>Bmid</code> or <code>oiltype</code> to also
be entered. See Hart and Vorobyev (2005).</p>
</td></tr>
<tr><td><code id="sensmodel_+3A_bmid">Bmid</code></td>
<td>
<p>a vector of same length as peaksens that lists the gradient of
line tangent to the absorbance spectrum of the oil droplets. See Hart and
Vorobyev (2005).</p>
</td></tr>
<tr><td><code id="sensmodel_+3A_oiltype">oiltype</code></td>
<td>
<p>a list of same length as peaksens that lists the oil droplet
types (currently accepts only &quot;T&quot;, C&quot;, &quot;Y&quot;, &quot;R&quot;, &quot;P&quot;) when Bmid is not
known. Calculates Bmid based on the regression equations found in Hart ad
Vorobyev (2005).</p>
</td></tr>
<tr><td><code id="sensmodel_+3A_beta">beta</code></td>
<td>
<p>logical. If <code>TRUE</code> the sensitivities will include the beta peak
See Govardovskii et al.(2000) (defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="sensmodel_+3A_om">om</code></td>
<td>
<p>a vector of same length as <code>range1</code>-<code>range2</code> that contains ocular
media transmission data. If included, cone sensitivity will be corrected
for ocular media transmission. Currently accepts &quot;bird&quot; using values from
Hart et al. (2005), or user-defined values.</p>
</td></tr>
<tr><td><code id="sensmodel_+3A_integrate">integrate</code></td>
<td>
<p>logical. If <code>TRUE</code>, each curve is transformed to have a
total area under the curve of 1 (best for visual models; defaults to
<code>TRUE</code>). NOTE: integration is applied before any effects of ocular media
are considered, for compatibility with visual model procedures.</p>
</td></tr>
<tr><td><code id="sensmodel_+3A_sensnames">sensnames</code></td>
<td>
<p>A vector equal in length to <code>peaksens</code>, specifying custom
names for the resulting sensitivity curves (e.g. c('s', 'm', 'l') for
short-, medium- and long-wavelength sensitive receptors.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of class <code>rspec</code> containing each cone model as a column.
</p>


<h3>Author(s)</h3>

<p>Pierre-Paul Bitton <a href="mailto:bittonp@uwindsor.ca">bittonp@uwindsor.ca</a>
</p>
<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Govardovskii VI, Fyhrquist N, Reuter T, Kuzmin DG and Donner K.
2000. In search of the visual pigment template. Visual Neuroscience
17:509-528, <a href="https://doi.org/10.1017/S0952523800174036">doi:10.1017/S0952523800174036</a>
</p>
<p>Hart NS, and Vorobyev M. 2005. Modeling oil droplet absorption
spectra and spectral sensitivities of bird cone photoreceptors. Journal of
Comparative Physiology A. 191: 381-392, <a href="https://doi.org/10.1007/s00359-004-0595-3">doi:10.1007/s00359-004-0595-3</a>
</p>
<p>Hart NS, Partridge JC, Cuthill IC, Bennett AT (2000) Visual
pigments, oil droplets, ocular media and cone photoreceptor distribution in
two species of passerine bird: the blue tit (<em>Parus caeruleus</em> L.) and the
blackbird (<em>Turdus merula</em> L.). J Comp Physiol A 186:375-387,
<a href="https://doi.org/10.1007/s003590050437">doi:10.1007/s003590050437</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Blue tit visual system based on Hart et al (2000)
bluesens &lt;- sensmodel(c(371, 448, 502, 563),
  beta = FALSE,
  lambdacut = c(330, 413, 507, 572),
  oiltype = c("T", "C", "Y", "R"), om = TRUE
)

# Danio aequipinnatus based on Govardovskii et al. (2000)
daniosens &lt;- sensmodel(c(357, 411, 477, 569))

</code></pre>

<hr>
<h2 id='sicalis'>Spectral curves from three body regions of stripe-tailed yellow finch
(<em>Sicalis citrina</em>) males</h2><span id='topic+sicalis'></span>

<h3>Description</h3>

<p>Dataset containing reflectance measurements from 3 body parts (&quot;C&quot;: crown,
&quot;B&quot;: breast, &quot;T&quot;: throat) from seven male stripe-tailed yellow finches
(<em>Sicalis citrina</em>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sicalis)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>rspec</code> (inherits from <code>data.frame</code>) with 401 rows and 22 columns.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>

<hr>
<h2 id='simulate_spec'>Simulate a spectrum</h2><span id='topic+simulate_spec'></span>

<h3>Description</h3>

<p>Simulate a naturalistic reflectance, radiance, irradiance, or transmission spectrum.
Curves may have sigmoidal (s-shaped) and/or Gaussian (bell-shaped) features.
Multiple Gaussian and sigmoidal curves can be combined in a single spectrum,
to simulate more complex spectral functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_spec(
  wl_inflect = NULL,
  wl_peak = NULL,
  width_sig = 20,
  width_gauss = 70,
  skew_gauss = 0,
  xlim = c(300, 700),
  ylim = c(0, 100)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_spec_+3A_wl_inflect">wl_inflect</code></td>
<td>
<p>A numeric vector specifying the wavelength location (in nm)
for one or more inflection point(s) for a 'sigmoid' shaped curve, if desired.</p>
</td></tr>
<tr><td><code id="simulate_spec_+3A_wl_peak">wl_peak</code></td>
<td>
<p>A numeric vector specifying the wavelength location (in nm)
for one or more inflection point(s) for a 'Gaussian' (or 'bell') shaped curve,
if desired.</p>
</td></tr>
<tr><td><code id="simulate_spec_+3A_width_sig">width_sig</code></td>
<td>
<p>A numeric value or vector (if multiple wl_inflect values are specified)
specifying the steepness of the change, for any sigmoidal curve(s). Required when
<code>wl_peak</code> is specified. Defaults to 20 nm.</p>
</td></tr>
<tr><td><code id="simulate_spec_+3A_width_gauss">width_gauss</code></td>
<td>
<p>A numeric value or vector specifying the the full-width at half-maximum
of any Gaussian curve(s). Required when <code>wl_peak</code> is specified. Defaults to 70 nm.</p>
</td></tr>
<tr><td><code id="simulate_spec_+3A_skew_gauss">skew_gauss</code></td>
<td>
<p>Skewness parameter for controlling the direction and magnitude of
skew, when simulating for Gaussian curves (ignored when simulating only sigmoidal curves).
Curves will have no skew when <code>skew_gauss = 0</code> (default), right skew when <code>skew_gauss &gt; 0</code>,
and left skew when <code>skew_gauss &lt; 0</code>. The parameter corresponds to 'alpha' in the
the skew-normal distribution (Azzalini 1985).</p>
</td></tr>
<tr><td><code id="simulate_spec_+3A_xlim">xlim</code></td>
<td>
<p>A vector specifying the wavelength range of the simulated spectra. Defaults to
300-700nm (<code>c(300, 700)</code>).</p>
</td></tr>
<tr><td><code id="simulate_spec_+3A_ylim">ylim</code></td>
<td>
<p>A vector specifying the minimum and maximum reflectance values of the resulting
curve. Defaults to 0 - 100 % (<code>c(0, 100)</code>). Note: sigmoidal curves, by default, move
from low to high reflectance. But if high-to-low sigmoidal curves are desired this
can be controlled by the ordering of the values given to <code>ylim()</code>. E.g. <code>c(0, 100)</code>
will generate a low-to-high sigmoidal curve, whereas <code>c(100, 0)</code> will generate a high-
to-low curve. The ordering of values has no effect on the Gaussian portions of the
final curve.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>rspec</code>.
</p>


<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Hugo Gruson <a href="mailto:hugo.gruson+R@normalesup.org">hugo.gruson+R@normalesup.org</a>
</p>


<h3>References</h3>

<p>Azzalini A (1985). A class of distributions which includes the
normal ones. Scan. J. Stat. 171-178.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Single ideal 'grey' reflectance spectrum, with 50% reflectance across 300 - 700 nm.
reflect0 &lt;- simulate_spec(ylim = c(0, 50))

# Single sigmoidal spectrum, with a low-to-high inflection at 550 nm.
reflect1 &lt;- simulate_spec(wl_inflect = 550)

# Single Gaussian spectrum, with a peak at 400 nm
reflect2 &lt;- simulate_spec(wl_peak = 400)

# Combination of both Gaussian (with peak at 340 nm) and sigmoidal (with inflection
# at 560 nm)
reflect3 &lt;- simulate_spec(wl_inflect = 560, wl_peak = 340)

# Double-Gaussian peaks of differing widths
reflect4 &lt;- simulate_spec(wl_peak = c(340, 560), width_gauss = c(12, 40))

# Complex spectrum with single sigmoidal peak and multi-Gaussian peaks
reflect5 &lt;- simulate_spec(wl_inflect = 575, wl_peak = c(340, 430), width_gauss = c(20, 60))

# Simulate a set of Gaussian reflectance curves with peaks varying between 400-600nm
# in increments of 10, then combine into a single rspec object, and plot the result
peaks &lt;- seq(400, 600, 10)  # Peak locations
reflectances &lt;- lapply(seq_along(peaks), function(x) simulate_spec(wl_peak = peaks[x]))  # Simulate
reflectances &lt;- Reduce(merge, reflectances)  # Combine
plot(reflectances)  # Plot

# Simulate a set of Gaussian reflectance curves with a single peak at 500 nm, but
# with maximum reflectance varying from 30 to 90% in 10% increments, then combine
# into a single rspec object, and plot the result
ymax &lt;- lapply(seq(30, 90, 10), function(x) c(0, x))  # Varying reflectance maxima
reflectances2 &lt;- lapply(ymax, function(x) simulate_spec(wl_peak = 500, ylim = x))  # Simulate
reflectances2 &lt;- Reduce(merge, reflectances2)  # Combine
plot(reflectances2)  # Plot

# To simulate non-reflectance spectra (like irradiances or radiances), it's often useful
# to explore more 'extreme' parameters. Here's a simple example which simulates
# natural daylight, as represented by the D65 standard daylight spectrum.
D65_real &lt;- procspec(sensdata(illum = 'D65'), opt = 'smooth')  # Official D65 daylight spectrum
D65_sim &lt;- simulate_spec(wl_peak = 400,
                         width_gauss = 1300,
                         skew_gauss = 10,
                         ylim = c(0, 1))  # Simulated D65
cor.test(D65_real$D65, D65_sim$spec_p400)  # &gt;0.99 correlation
plot(merge(D65_real, D65_sim), lty = 1:2, ylab = 'Irradiance (%)')  # Merge and plot the two spectra

</code></pre>

<hr>
<h2 id='spec2rgb'>Spectrum to rgb colour conversion</h2><span id='topic+spec2rgb'></span>

<h3>Description</h3>

<p>Calculates rgb values from spectra based on human colour matching functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>spec2rgb(rspecdata, alpha = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="spec2rgb_+3A_rspecdata">rspecdata</code></td>
<td>
<p>(required) a data frame, possibly of class <code>rspec</code>, which
contains a column containing a wavelength range, named 'wl', and spectra
data in remaining columns.</p>
</td></tr>
<tr><td><code id="spec2rgb_+3A_alpha">alpha</code></td>
<td>
<p>alpha value to use for colours (defaults to 1, opaque).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector consisting of hexadecimal colour values
for passing to further plotting functions.
</p>


<h3>Author(s)</h3>

<p>Hugo Gruson <a href="mailto:hugo.gruson+R@normalesup.org">hugo.gruson+R@normalesup.org</a>
</p>
<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>CIE(1932). Commission Internationale de l'Eclairage Proceedings, 1931. Cambridge: Cambridge University Press.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(teal)
spec2rgb(teal)

# Plot data using estimated perceived colour
plot(teal, col = spec2rgb(teal), type = "overlay")
</code></pre>

<hr>
<h2 id='subset.rspec'>Subset rspec, vismodel, and colspace objects</h2><span id='topic+subset.rspec'></span><span id='topic+subset.colspace'></span><span id='topic+subset.vismodel'></span>

<h3>Description</h3>

<p>Subsets various object types based on a given vector or grep partial matching
of data names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rspec'
subset(x, subset, ...)

## S3 method for class 'colspace'
subset(x, subset, ...)

## S3 method for class 'vismodel'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset.rspec_+3A_x">x</code></td>
<td>
<p>(required) an object of class <code>rspec</code>, <code>vismodel</code>, or
<code>colspace</code>, containing spectra, visual model output or colourspace data
to subset.</p>
</td></tr>
<tr><td><code id="subset.rspec_+3A_subset">subset</code></td>
<td>
<p>a string used for partial matching of observations.</p>
</td></tr>
<tr><td><code id="subset.rspec_+3A_...">...</code></td>
<td>
<p>additional attributes passed to <code>grep</code>. Ignored if
<code>subset</code> is logical.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a subsetted object of the same class as the input object.
</p>


<h3>Note</h3>

<p>if more than one value is given to <code>subset</code>, any spectra that
matches <em>either</em> condition will be included. It's a union, not an
intersect.
</p>


<h3>Author(s)</h3>

<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sicalis)
vis.sicalis &lt;- vismodel(sicalis)
tcs.sicalis &lt;- colspace(vis.sicalis, space = "tcs")

# Subset all 'crown' patches (C in file names)
head(subset(sicalis, "C"))
head(subset(sicalis, c("B", "C")))
head(subset(sicalis, "T", invert = TRUE))
subset(vis.sicalis, "C")
subset(tcs.sicalis, "C")[, seq_len(5)]
</code></pre>

<hr>
<h2 id='summary.colspace'>Colourspace data summary</h2><span id='topic+summary.colspace'></span>

<h3>Description</h3>

<p>Returns the attributes of <code>colspace</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'colspace'
summary(object, by = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.colspace_+3A_object">object</code></td>
<td>
<p>(required) a <code>colspace</code> object.</p>
</td></tr>
<tr><td><code id="summary.colspace_+3A_by">by</code></td>
<td>
<p>when the input is in <code>tcs</code> colourspace, <code>by</code> is either
a single value specifying the range of colour points for which
summary tetrahedral-colourspace variables should be calculated (for example, <code>by</code> = 3
indicates summary will be calculated for groups of 3 consecutive colour points (rows)
in the quantum catch colour data frame) or a vector containing identifications for
the rows in the quantum catch colour data frame (in which case summaries will be
calculated for each group of points sharing the same identification). If <code>by</code>
is left blank, the summary statistics are calculated across all colour points in the
data.</p>
</td></tr>
<tr><td><code id="summary.colspace_+3A_...">...</code></td>
<td>
<p>class consistency (ignored).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns all attributes of the data as mapped to the selected colourspace, including
options specified when calculating the visual model. Also return the default
<code>data.frame</code> summary, except when the object is the result of <code><a href="#topic+tcspace">tcspace()</a></code>,
in which case the following variables are output instead:
</p>

<ul>
<li> <p><code style="white-space: pre;">&#8288;centroid.u, .s, .m, .l&#8288;</code> the centroids of <code>usml</code> coordinates of points.
</p>
</li>
<li> <p><code>c.vol</code> the total volume occupied by the points, computed with a convex
hull.
</p>
</li>
<li> <p><code>rel.c.vol</code> volume occupied by the points (convex hull volume) relative to
the tetrahedron volume.
</p>
</li>
<li> <p><code>colspan.m</code> the mean hue span.
</p>
</li>
<li> <p><code>colspan.v</code> the variance in hue span.
</p>
</li>
<li> <p><code>huedisp.m</code> the mean hue disparity.
</p>
</li>
<li> <p><code>huedisp.v</code> the variance in hue disparity.
</p>
</li>
<li> <p><code>mean.ra</code> mean saturation.
</p>
</li>
<li> <p><code>max.ra</code> maximum saturation achieved by the group of points.
</p>
</li>
<li> <p><code>a.vol</code> colour volume computed with <code class="reqn">\alpha</code>-shapes.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage
color in a tetrahedral color space: A phylogenetic analysis of new world buntings.
The American Naturalist, 171(6), 755-776.
</p>
<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour patterns
as birds see them. Biological Journal Of The Linnean Society, 86(4), 405-431.
</p>
<p>Gruson H. (2020). Estimation of colour volumes as concave hypervolumes using
<code class="reqn">\alpha</code>-shapes. Methods in Ecology and Evolution, 11(8), 955-963
<a href="https://doi.org/10.1111/2041-210X.13398">doi:10.1111/2041-210X.13398</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Colour hexagon
data(flowers)
vis.flowers &lt;- vismodel(flowers,
  visual = "apis", qcatch = "Ei", relative = FALSE,
  vonkries = TRUE, bkg = "green"
)
flowers.hex &lt;- hexagon(vis.flowers)
summary(flowers.hex)

# Tetrahedral model
data(sicalis)
vis.sicalis &lt;- vismodel(sicalis, visual = "avg.uv")
csp.sicalis &lt;- colspace(vis.sicalis)
summary(csp.sicalis, by = rep(c("C", "T", "B"), 7))
</code></pre>

<hr>
<h2 id='summary.rimg'>Image summary</h2><span id='topic+summary.rimg'></span>

<h3>Description</h3>

<p>Returns the attributes of, and optionally plots, an image.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rimg'
summary(object, plot = FALSE, axes = TRUE, col = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.rimg_+3A_object">object</code></td>
<td>
<p>(required) an image of class <code>rimg</code>, or list thereof.</p>
</td></tr>
<tr><td><code id="summary.rimg_+3A_plot">plot</code></td>
<td>
<p>logical; plot the image and, if the image is color-classified, the colours
corresponding to colour class categories side-by-side? Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="summary.rimg_+3A_axes">axes</code></td>
<td>
<p>should axes be drawn when <code>plot = TRUE</code>? (defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="summary.rimg_+3A_col">col</code></td>
<td>
<p>optional vector of colours when plotting colour-classified images with <code>plot = TRUE</code>.
Defaults to the mean RGB values of the k-means centres (i.e. the 'original' colours).</p>
</td></tr>
<tr><td><code id="summary.rimg_+3A_...">...</code></td>
<td>
<p>additional graphical options when <code>plot = TRUE</code>. Also see <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either the RGB values of the k-means centres from the colour-classified image,
or a plot of both the image and specified colours (when <code>plot = TRUE</code>).
</p>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
papilio &lt;- getimg(system.file("testdata/images/butterflies/papilio.png", package = "pavo"))
papilio_class &lt;- classify(papilio, kcols = 4)
summary(papilio_class)

# Plot the colour-classified image alongside the colour class palette
summary(papilio_class, plot = TRUE)

# Multiple images
snakes &lt;- getimg(system.file("testdata/images/snakes", package = "pavo"))
snakes_class &lt;- classify(snakes, kcols = 3)
summary(snakes_class, plot = TRUE)


</code></pre>

<hr>
<h2 id='summary.rspec'>Colourimetric variables</h2><span id='topic+summary.rspec'></span>

<h3>Description</h3>

<p>Calculates all 23 colourimetric variables reviewed in Montgomerie (2006).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rspec'
summary(object, subset = FALSE, lim = NULL, wlmin = NULL, wlmax = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.rspec_+3A_object">object</code></td>
<td>
<p>(required) a data frame, possibly an object of class <code>rspec</code>,
with a column with wavelength data, named 'wl', and the remaining column containing
spectra to process.</p>
</td></tr>
<tr><td><code id="summary.rspec_+3A_subset">subset</code></td>
<td>
<p>Either <code>FALSE</code> (the default), <code>TRUE</code>, or a character vector.
If <code>FALSE</code>, all variables calculated are returned. If <code>TRUE</code>, only a subset
of the complete output (composed of B2, S8 and H1; the variables described in
Andersson and Prager 2006) are returned. Finally, a user-specified string of variable
names can be used in order to filter and show only those variables.</p>
</td></tr>
<tr><td><code id="summary.rspec_+3A_lim">lim</code></td>
<td>
<p>The range of wavelengths used in calculations. The default is to use
the entire range in the <code>rspec</code> object (typically equivalent to <code>lim = c(300, 700)</code>).</p>
</td></tr>
<tr><td><code id="summary.rspec_+3A_wlmin">wlmin</code>, <code id="summary.rspec_+3A_wlmax">wlmax</code></td>
<td>
<p>Deprecated. Use the <code>lim</code> argument instead.</p>
</td></tr>
<tr><td><code id="summary.rspec_+3A_...">...</code></td>
<td>
<p>class consistency (ignored)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing either 23 or 5 (<code>subset = TRUE</code>) variables described
in Montgomerie (2006) with spectra name as row names.
The colorimetric variables calculated by this function are
described in Montgomerie (2006) with corrections included in the README CLR
file from the May 2008 distribution of the CLR software. Authors should reference
both this package,Montgomerie (2006), and the original reference(s).
Description and notes on the measures:
</p>
<p>B1 (Total brightness): Sum of the relative reflectance over the entire spectral
range (area under the curve). Frequently used but should be discouraged because
values are difficult to compare across studies (B2 is preferred). REF 1-3, 7, 9-11,
13
</p>
<p>B2 (Mean brightness): Mean relative reflectance over the entire spectral range.
This is preferred to B1 since values are easier to compare across studies. REF 4, 12
</p>
<p>B3 (Intensity): Maximum relative reflectance (Reflectance at wavelength of maximum
reflectance). Note that may be sensitive to noise near the peak. REF 1, 5, 6
</p>
<p>S1 (Chroma): Relative contribution of a spectral range to the total brightness (B1)
S1 is arbitrarily divided in 6 measures of chroma based on the wavelength ranges
normally associated with specific hues. The values are calculated using the
following ranges: S1U (UV, if applicable): lambda min-400nm;
S1V (Violet) lambda min-415nm; S1B (Blue) 400nm-510nm; S1G (Green) 510nm-605nm;
S1Y (Yellow) 550nm-625nm; S1R (Red) 605nm-lambda max. REF 2, 7, 8, 11-13
</p>
<p>S2 (Spectral saturation): Rmax/Rmin This measure is sensitive to spectral noise.
Proper interpretation of this value may be difficult for spectra with multiple
peaks in the range of interest. REF 1
</p>
<p>S3 (Chroma): Reflectance over the Rmax +- 50nm range divided by B1. Values for peaks
within 50nm of either the minimum or maximum range of the data will not be comparable
since the area under the curve for the area of interest will not always
be based on the same wavelength range. Therefore, S3 should be interpreted
with caution for peaks in the UV or Red range. REF 11
</p>
<p>S4 (Spectral purity): |bmaxneg| , calculated by approximating the derivative
of the spectral curve. As such, it is very sensitive to noise and should only
be considered when data is adequately smoothed. NAs are returned for curves which
do not, at any range of wavelength, decrease in intensity. Therefore, reflectance
curves for brown and red surfaces, for example, should not generate a values. REF 1
</p>
<p>S5 (Chroma): Similar in design to segment classification measures (see Montgomerie 2006
for details). REF 10
</p>
<p>S6 (Contrast): Rmax - Rmin. Because it uses both Rmin and Rmax, this measure may be
sensitive to spectral noise. REF 5, 6
</p>
<p>S7 (Spectral saturation): Difference between the relative reflectance before and after
the wavelength at which reflectance is halfway between its minimum (Rmin)
and its maximum (Rmax). Somewhat sensitive
to noise and can be misleading when more than one maxima and/or minima are present.
REF 3, 9
</p>
<p>S8 (Chroma): (Rmax - Rmin)/B2. Because it uses both Rmin and Rmax, this measure may be
sensitive to spectral noise. REF 3, 13
</p>
<p>S9 (Carotenoid chroma): (R700 - R450)/R700. Should only be used when the colour
of the surface is clearly due to carotenoid pigmentation and R450 is lower than
R700. Could be sensitive to noise. REF 8
</p>
<p>S10 (Peaky chroma): (Rmax - Rmin)/B2 x |bmaxneg|. Should be used with properly
smoothed curves. REF 7
</p>
<p>H1 (Peak wavelength, hue): Wavelength of maximum reflectance. May be sensitive to noise
and may be variable if there is more than one maxima. REF 1, 2, 4, 6, 7, 10-13
</p>
<p>H2 (Hue): Wavelength at bmaxneg. Should be calculated using smoothed data. REF 2, 13
</p>
<p>H3 (Hue): Wavelength at Rmid. Sensitive to noisy spectra and may be variable if there are
more than one maxima and minima. REF 3, 9, 13
</p>
<p>H4 (Hue): Similar in design to segment classification measures see Montgomerie
(2006) for details. REF 10
</p>
<p>H5 (Hue): Wavelength at bmax. Sensitive to noise and may be variable if there is
more than one maxima and minima. REF 5
</p>


<h3>Note</h3>

<p>If minimum wavelength is over 400, UV chroma is not computed.
</p>
<p>Variables which compute bmax and bmaxneg should be used with caution, for they
rely on smoothed curves to remove noise, which would otherwise result in spurious
results. Make sure chosen smoothing parameters are adequate.
</p>
<p>Smoothing affects only B3, S2, S4, S6, S10, H2, and H5 calculation. All other
variables can be reliably extracted using non-smoothed data.
</p>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Pierre-Paul Bitton <a href="mailto:bittonp@windsor.ca">bittonp@windsor.ca</a>
</p>
<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Montgomerie R. 2006. Analyzing colors. In Hill, G.E, and McGraw, K.J., eds.
Bird Coloration. Volume 1 Mechanisms and measurements. Harvard University Press, Cambridge, Massachusetts.
</p>
<p>References describing variables:
</p>
<p>1- Andersson, S. 1999. Morphology of uv reflectance in a whistling-thrush: Implications for the study
of structural colour signalling in birds. Journal of Avian Biology 30:193-204.
</p>
<p>2- Andersson, S., J. Ornborg, and M. Andersson. 1998. Ultraviolet sexual dimorphism and assortative
mating in blue tits. Proceedings of the Royal Society B 265:445-450.
</p>
<p>3- Andersson, S., S. Pryke, J. Ornborg, M. Lawes, and M. Andersson. 2002. Multiple receivers, multiple
ornaments, and a trade-off between agonistic and epigamic signaling in a widowbird. American
Naturalist 160:683-691.
</p>
<p>4- Delhey, K., A. Johnsen, A. Peters, S. Andersson, and B. Kempenaers. 2003. Paternity analysis reveals
opposing selection pressures on crown coloration in the blue tit (Parus caeruleus). Proceedings
of the Royal Society B 270:2057-2063.
</p>
<p>5- Keyser, A. and G. Hill. 1999. Condition-dependent variation in the blue-ultraviolet coloration of a
structurally based plumage ornament. Proceedings of the Royal Society B 266:771-777.
</p>
<p>6- Keyser, A.J. and G. Hill. 2000. Structurally based plumage coloration is an honest signal of quality in
male blue grosbeaks. Behavioural Ecology 11:202-209.
</p>
<p>7- Ornborg, J., S. Andersson, S. Griffith, and B. Sheldon. 2002. Seasonal changes in a ultraviolet
structural colour signal in blue tits, Parus caeruleus. Biological Journal of the Linnean Society
76:237-245.
</p>
<p>8- Peters, A., A. Denk, K. Delhey, and B. Kempenaers. 2004. Carotenoid-based bill colour as an
indicator of immunocompetence and sperm performance in male mallards. Journal of
Evolutionary Biology 17:1111-1120.
</p>
<p>9- Pryke, S., M. Lawes, and S. Andersson. 2001. Agonistic carotenoid signalling in male red-collared
widowbirds: Aggression related to the colour signal of both the territory owner and model
intruder. Animal Behaviour 62:695-704.
</p>
<p>10- Saks, L., K. Mcgraw, and P. Horak. 2003. How feather colour reflects its carotenoid content.
Functional Ecology 17:555-561.
</p>
<p>11- Shawkey, M., A. Estes, L. Siefferman, and G. Hill. 2003. Nanostructure predicts intraspecific
variation in ultraviolet-blue plumage colour. Proceedings of the Royal Society B
270:1455-1460.
</p>
<p>12- Siefferman, L. and G. Hill. 2005. UV-blue structural coloration and competition for nestboxes in male
eastern bluebirds. Animal Behaviour 69:67-72.
</p>
<p>13- Smiseth, P., J. Ornborg, S. Andersson, and T. Amundsen. 2001. Is male plumage reflectance
correlated with paternal care in bluethroats? Behavioural Ecology 12:164-170.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load data
data(sicalis)

# Calculate and display all spectral summary variables
summary(sicalis)

# Calculate only subset of B2, S8 and H1 as per Andersson (1999)
summary(sicalis, subset = TRUE)

# Calculate user-specified subset of B1 and H4
summary(sicalis, subset = c("B1", "H4"))

</code></pre>

<hr>
<h2 id='summary.vismodel'>Visual model summary</h2><span id='topic+summary.vismodel'></span>

<h3>Description</h3>

<p>Returns the attributes used when calculating a visual model using <code><a href="#topic+vismodel">vismodel()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'vismodel'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.vismodel_+3A_object">object</code></td>
<td>
<p>(required) Results of <code><a href="#topic+vismodel">vismodel()</a></code></p>
</td></tr>
<tr><td><code id="summary.vismodel_+3A_...">...</code></td>
<td>
<p>class consistency (ignored)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns all attributes chosen when calculating the visual model, as well as the
default <code>data.frame</code> summary
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Vorobyev, M., Osorio, D., Bennett, A., Marshall, N., &amp; Cuthill, I. (1998).
Tetrachromacy, oil droplets and bird plumage colours. Journal Of Comparative
Physiology A-Neuroethology Sensory Neural And Behavioral Physiology, 183(5), 621-633.
</p>
<p>Hart, N. S. (2001). The visual ecology of avian photoreceptors.
Progress In Retinal And Eye Research, 20(5), 675-703.
</p>
<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage
color in a tetrahedral color space: A phylogenetic analysis of new world buntings.
The American Naturalist, 171(6), 755-776.
</p>
<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour patterns
as birds see them. Biological Journal Of The Linnean Society, 86(4), 405-431.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sicalis)
vis.sicalis &lt;- vismodel(sicalis, visual = "avg.uv")
summary(vis.sicalis)
</code></pre>

<hr>
<h2 id='tcspace'>Tetrahedral colourspace</h2><span id='topic+tcspace'></span>

<h3>Description</h3>

<p>Calculates coordinates and colorimetric variables that represent reflectance spectra
in the avian tetrahedral color space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tcspace(vismodeldata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tcspace_+3A_vismodeldata">vismodeldata</code></td>
<td>
<p>(required) quantum catch color data. Can be either the result
from <code><a href="#topic+vismodel">vismodel()</a></code> or independently calculated data (in the form of a data frame
with four columns name 'u', 's', 'm', 'l', representing the avian cones).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of class <code><a href="#topic+colspace">colspace</a></code> consisting of the following columns:
</p>

<ul>
<li> <p><code>u</code>, <code>s</code>, <code>m</code>, <code>l</code>: the quantum catch data used to
calculate the remaining variables. NOTE: even if visual system is of type V-VIS,
the output column will be labeled <code>u</code>.
</p>
</li>
<li> <p><code>u.r</code>, <code>s.r</code>, <code>m.r</code>, <code>l.r</code>: relative cone stimulation,
for a given hue, as a function of saturation. See Stoddard &amp; Prum (2008) for details.
</p>
</li>
<li> <p><code>x</code>, <code>y</code>, <code>z</code>: cartesian coordinates for the points in the
tetrahedral color space.
</p>
</li>
<li> <p><code>h.theta</code>, <code>h.phi</code>: angles theta and phi, in radians, determining
the hue of the color.
</p>
</li>
<li> <p><code>r.vec</code>: the r vector (saturation, distance from the achromatic center).
</p>
</li>
<li> <p><code>r.max</code>: the maximum r vector achievable for the colour's hue.
</p>
</li>
<li> <p><code>r.achieved</code>: the relative r distance from the achromatic center, in
relation to the maximum distance achievable (<code>r.vec/r.max</code>).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage
color in a tetrahedral color space: A phylogenetic analysis of new world buntings.
The American Naturalist, 171(6), 755-776.
</p>
<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour patterns
as birds see them. Biological Journal Of The Linnean Society, 86(4), 405-431.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sicalis)
vis.sicalis &lt;- vismodel(sicalis, visual = "avg.uv")
tcs.sicalis &lt;- colspace(vis.sicalis, space = "tcs")
</code></pre>

<hr>
<h2 id='tcsplot'>Interactive plot of a tetrahedral colourspace</h2><span id='topic+tcsplot'></span><span id='topic+tcspoints'></span><span id='topic+tcsvol'></span>

<h3>Description</h3>

<p>Produces an interactive 3D plot of a tetrahedral colourspace using OpenGL
capabilities.
</p>
<p>Plots points in a tetrahedral colour space
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tcsplot(
  tcsdata,
  size = 0.02,
  alpha = 1,
  col = "black",
  vertexsize = 0.02,
  achro = TRUE,
  achrosize = 0.01,
  achrocol = "grey",
  lwd = 1,
  lcol = "lightgrey",
  new = FALSE,
  hspin = FALSE,
  vspin = FALSE,
  floor = TRUE,
  gamut = FALSE
)

tcspoints(tcsdata, size = 0.02, col = "black", alpha = 1)

tcsvol(
  tcsdata,
  type = c("convex", "alpha"),
  avalue = "auto",
  col = "black",
  alpha = 0.2,
  grid.alpha = 1,
  grid = TRUE,
  fill = TRUE,
  lwd = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tcsplot_+3A_tcsdata">tcsdata</code></td>
<td>
<p>(required) a data frame, possibly a result from the
<code><a href="#topic+colspace">colspace()</a></code> or <code><a href="#topic+tcspace">tcspace()</a></code> function, containing values for the 'x', 'y'
and 'z' coordinates as columns (labeled as such).</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_size">size</code></td>
<td>
<p>size of the points in the plot (defaults to 0.02)</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_alpha">alpha</code></td>
<td>
<p>transparency of points (or volume fill in <code><a href="#topic+tcsvol">tcsvol()</a></code>)</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_col">col</code></td>
<td>
<p>colour of the points in the plot (defaults to black)</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_vertexsize">vertexsize</code></td>
<td>
<p>size of the points at the vertices</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_achro">achro</code></td>
<td>
<p>should a point be plotted at the origin (defaults to <code>TRUE</code>)?</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_achrosize">achrosize</code></td>
<td>
<p>size of the point at the origin when <code>achro = TRUE</code>
(defaults to <code>0.8</code>).</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_achrocol">achrocol</code></td>
<td>
<p>color of the point at the origin <code>achro = TRUE</code> (defaults to
<code>'grey'</code>).</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_lwd">lwd</code>, <code id="tcsplot_+3A_lcol">lcol</code></td>
<td>
<p>graphical parameters for the edges of the tetrahedron.</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_new">new</code></td>
<td>
<p>should a new 3D plot be called (defaults to <code>FALSE</code>)?</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_hspin">hspin</code></td>
<td>
<p>if <code>TRUE</code>, the graphic will spin horizontally (around the 'z' axis)(defaults to <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_vspin">vspin</code></td>
<td>
<p>if <code>TRUE</code>, the graphic will spin vertically (around the 'x' axis)(defaults to <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_floor">floor</code></td>
<td>
<p>if <code>TRUE</code>, a reference xy plane is plotted under the tetrahedron (defaults to <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_gamut">gamut</code></td>
<td>
<p>logical. Should the polygon showing the possible colours given
visual system and illuminant used in the analysis (defaults to <code>FALSE</code>).
This option currently only works when <code>qcatch = Qi</code>.</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_type">type</code></td>
<td>
<p>accepts a vector of length 1 or 2 with 'p' for points and/or 'l'
for lines from the point to the base of the tetrahedron.</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_avalue">avalue</code></td>
<td>
<p>if <code>type = "alpha"</code>, which alpha parameter value should be used
to compute the alphashape. <code>avalue = "auto"</code> (default) finds and use the
<code class="reqn">\alpha^*</code> value as defined in Gruson (2020).</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_grid.alpha">grid.alpha</code></td>
<td>
<p>transparency of the volume polygon grid lines</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_grid">grid</code></td>
<td>
<p>if <code>TRUE</code>, connects the polygon outlining the volume occupied by points (defaults to <code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="tcsplot_+3A_fill">fill</code></td>
<td>
<p>if <code>TRUE</code>, fills the volume occupied by points (WARNING: transparency
is not saved properly if exported using <code>rgl.postscript</code>)(defaults to <code>TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code><a href="#topic+tcsplot">tcsplot()</a></code> creates a 3D plot using functions of the package <span class="pkg">rgl</span>,
based on openGL capabilities. Plot is interactive and can be manipulated with the mouse
(left button: rotate along 'z' axis; right button: rotate along 'x' axis;
third button: zoom).
</p>
<p><code><a href="#topic+tcspoints">tcspoints()</a></code> adds points to the plot. Points are currently plotted
only as spheres to maintain export capabilities.
</p>
<p><code><a href="#topic+tcsvol">tcsvol()</a></code> creates a 3D colour volume within a <code>tcsplot</code> object.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage
color in a tetrahedral color space: A phylogenetic analysis of new world buntings.
The American Naturalist, 171(6), 755-776.
</p>
<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour patterns
as birds see them. Biological Journal Of The Linnean Society, 86(4), 405-431.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# For plotting
data(sicalis)
vis.sicalis &lt;- vismodel(sicalis, visual = "avg.uv")
tcs.sicalis &lt;- colspace(vis.sicalis, space = "tcs")
tcsplot(tcs.sicalis, size = 0.005)
rgl::rgl.postscript("testplot.pdf", fmt = "pdf")
rgl::rgl.snapshot("testplot.png")

# For adding points
patch &lt;- rep(c("C", "T", "B"), 7)
tcs.crown &lt;- subset(tcs.sicalis, "C")
tcs.breast &lt;- subset(tcs.sicalis, "B")
tcsplot(tcs.crown, col = "blue")
tcspoints(tcs.breast, col = "red")

# For plotting convex hull
tcsplot(tcs.sicalis, col = "blue", size = 0.005)
tcsvol(tcs.sicalis)

</code></pre>

<hr>
<h2 id='teal'>Angle-resolved reflectance data for the iridescent wing patch of a male green-winged
teal (<em>Anas carolinensis</em>)</h2><span id='topic+teal'></span>

<h3>Description</h3>

<p>Dataset containing reflectance measurements from the wing patch of a single
male at different incident angles (15-75 degrees in 5-degree increments).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(teal)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>rspec</code> (inherits from <code>data.frame</code>) with 401 rows and 13 columns.
</p>


<h3>Author(s)</h3>

<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>

<hr>
<h2 id='tetraplot'>Plot a static tetrahedral colorspace</h2><span id='topic+tetraplot'></span>

<h3>Description</h3>

<p>Produces a static 3D tetrahedral plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tetraplot(
  tcsdata,
  theta = 45,
  phi = 10,
  perspective = FALSE,
  range = c(1, 2),
  r = 1e+06,
  zoom = 1,
  achro = TRUE,
  achro.col = "grey",
  achro.size = 1,
  achro.line = FALSE,
  achro.lwd = 1,
  achro.lty = 3,
  tetrahedron = TRUE,
  vert.cex = 1,
  vert.range = c(1, 2),
  out.lwd = 1,
  out.lcol = "darkgrey",
  type = "p",
  labels = FALSE,
  gamut = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tetraplot_+3A_tcsdata">tcsdata</code></td>
<td>
<p>(required) a data frame, possibly a result from the
<code><a href="#topic+colspace">colspace()</a></code> or <code><a href="#topic+tcspace">tcspace()</a></code> function, containing values for the 'x', 'y'
and 'z' coordinates as columns (labeled as such).</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_theta">theta</code></td>
<td>
<p>angle to rotate the plot in the xy plane (defaults to 45).</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_phi">phi</code></td>
<td>
<p>angle to rotate the plot in the yz plane (defaults to 10).</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_perspective">perspective</code></td>
<td>
<p>logical, should perspective be forced by using point size
to reflect distance from the plane of view? (defaults to 'FALSE&ldquo;)</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_range">range</code></td>
<td>
<p>vert.range multiplier applied to <code>cex</code> and <code>vert.cex</code>,
respectively, to indicate the size range variation reflecting the distance
from the plane of view.</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_r">r</code></td>
<td>
<p>the distance of the eyepoint from the center of the plotting box.
Very high values approximate an orthographic projection (defaults to 1e6).
See <code><a href="graphics.html#topic+persp">persp()</a></code> for details.</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_zoom">zoom</code></td>
<td>
<p>zooms in (values greater than 1) or out (values between 0 and 1)
from the plotting area.</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_achro">achro</code></td>
<td>
<p>should a point be plotted at the origin (defaults to <code>TRUE</code>)?</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_achro.line">achro.line</code></td>
<td>
<p>logical. Should the achromatic line be plotted? (defaults
to <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_achro.lwd">achro.lwd</code>, <code id="tetraplot_+3A_achro.lty">achro.lty</code></td>
<td>
<p>graphical parameters for the achromatic
coordinates.</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_tetrahedron">tetrahedron</code></td>
<td>
<p>logical. Should the tetrahedron be plotted? (defaults to
<code>TRUE</code>)</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_vert.cex">vert.cex</code></td>
<td>
<p>size of the points at the vertices (defaults to 1).</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_out.lwd">out.lwd</code>, <code id="tetraplot_+3A_out.lcol">out.lcol</code></td>
<td>
<p>graphical parameters for the tetrahedral outline.</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_type">type</code></td>
<td>
<p>accepts a vector of length 1 or 2 with 'p' for points and/or 'l'
for lines from the point to the base of the tetrahedron.</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_labels">labels</code></td>
<td>
<p>logical. Should the name of each cone be printed next to the
corresponding vertex?</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_gamut">gamut</code></td>
<td>
<p>logical. Should the polygon showing the possible colours given
visual system and illuminant used in the analysis (defaults to <code>FALSE</code>).
This option currently only works when <code>qcatch = Qi</code>.</p>
</td></tr>
<tr><td><code id="tetraplot_+3A_...">...</code></td>
<td>
<p>additional graphical options. See <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code><a href="#topic+tetraplot">tetraplot()</a></code> creates a non-interactive 3D plot.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>
<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Chad Eliason <a href="mailto:cme16@zips.uakron.edu">cme16@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage
color in a tetrahedral color space: A phylogenetic analysis of new world
buntings. The American Naturalist, 171(6), 755-776.
</p>
<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour
patterns as birds see them. Biological Journal Of The Linnean Society,
86(4), 405-431.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# For plotting
data(sicalis)
vis.sicalis &lt;- vismodel(sicalis, visual = "avg.uv")
tcs.sicalis &lt;- colspace(vis.sicalis, space = "tcs")
plot(tcs.sicalis)
</code></pre>

<hr>
<h2 id='transmissiondata'>Default ocular transmission data</h2><span id='topic+transmissiondata'></span>

<h3>Description</h3>

<p>Default ocular transmission data
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Hart, N. S., Partridge, J. C., Cuthill, I. C., Bennett, A. T. D. (2000).
Visual pigments, oil droplets, ocular media and cone photoreceptor distribution in two
species of passerine
</p>

<hr>
<h2 id='triplot'>Plot a Maxwell triangle</h2><span id='topic+triplot'></span>

<h3>Description</h3>

<p>Produces a Maxwell triangle plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>triplot(
  tridata,
  labels = TRUE,
  achro = TRUE,
  achrocol = "grey",
  achrosize = 0.8,
  labels.cex = 1,
  out.lwd = 1,
  out.lcol = "black",
  out.lty = 1,
  square = TRUE,
  gamut = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="triplot_+3A_tridata">tridata</code></td>
<td>
<p>(required) a data frame, possibly a result from the
<code><a href="#topic+colspace">colspace()</a></code> or <code><a href="#topic+trispace">trispace()</a></code> function, containing values for the 'x' and
'y' coordinates as columns (labeled as such).</p>
</td></tr>
<tr><td><code id="triplot_+3A_labels">labels</code></td>
<td>
<p>logical. Should the name of each cone be printed next to the
corresponding vertex?</p>
</td></tr>
<tr><td><code id="triplot_+3A_achro">achro</code></td>
<td>
<p>should a point be plotted at the origin (defaults to <code>TRUE</code>)?</p>
</td></tr>
<tr><td><code id="triplot_+3A_achrocol">achrocol</code></td>
<td>
<p>color of the point at the origin <code>achro = TRUE</code> (defaults to
<code>'grey'</code>).</p>
</td></tr>
<tr><td><code id="triplot_+3A_achrosize">achrosize</code></td>
<td>
<p>size of the point at the origin when <code>achro = TRUE</code>
(defaults to <code>0.8</code>).</p>
</td></tr>
<tr><td><code id="triplot_+3A_labels.cex">labels.cex</code></td>
<td>
<p>size of the arrow labels.</p>
</td></tr>
<tr><td><code id="triplot_+3A_out.lwd">out.lwd</code>, <code id="triplot_+3A_out.lcol">out.lcol</code>, <code id="triplot_+3A_out.lty">out.lty</code></td>
<td>
<p>graphical parameters for the plot outline.</p>
</td></tr>
<tr><td><code id="triplot_+3A_square">square</code></td>
<td>
<p>logical. Should the aspect ratio of the plot be held to 1:1?
(defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="triplot_+3A_gamut">gamut</code></td>
<td>
<p>logical. Should the polygon showing the possible colours given
visual system and illuminant used in the analysis (defaults to <code>FALSE</code>).
This option currently only works when <code>qcatch = Qi</code>.</p>
</td></tr>
<tr><td><code id="triplot_+3A_...">...</code></td>
<td>
<p>additional graphical options. See <code><a href="graphics.html#topic+par">par()</a></code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Thomas White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Maxwell JC. (1970). On color vision. In: Macadam, D. L. (ed)
Sources of Color Science. Cambridge, MIT Press, 1872 - 1873.
</p>
<p>Kelber A, Vorobyev M, Osorio D. (2003). Animal colour vision -
behavioural tests and physiological concepts. Biological Reviews, 78,
81 - 118.
</p>
<p>MacLeod DIA, Boynton RM. (1979). Chromaticity diagram showing
cone excitation by stimuli of equal luminance. Journal of the Optical
Society of America, 69, 1183-1186.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers, visual = "apis")
tri.flowers &lt;- colspace(vis.flowers, space = "tri")
plot(tri.flowers)
</code></pre>

<hr>
<h2 id='trispace'>Trichromatic colour space</h2><span id='topic+trispace'></span>

<h3>Description</h3>

<p>Calculates coordinates and colourimetric variables that represent reflectance spectra
in a trichromatic chromaticity space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trispace(vismodeldata)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trispace_+3A_vismodeldata">vismodeldata</code></td>
<td>
<p>(required) quantum catch colour data. Can be either the result
from <code><a href="#topic+vismodel">vismodel()</a></code> or independently calculated data (in the form of a data frame
with three columns named 's', 'm', 'l', representing a trichromatic viewer).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of class <code><a href="#topic+colspace">colspace</a></code> consisting of the following columns:
</p>

<ul>
<li> <p><code>s</code>, <code>m</code>, <code>l</code>: the quantum catch data used to calculate
the remaining variables.
</p>
</li>
<li> <p><code>x</code>, <code>y</code>: cartesian coordinates for the points in the
Maxwell triangle.
</p>
</li>
<li> <p><code>h.theta</code>: angle theta, in radians, determining the hue of the colour.
</p>
</li>
<li> <p><code>r.vec</code>: the r vector (saturation, distance from the center).
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>


<h3>References</h3>

<p>Maxwell JC. (1970). On color vision. In: Macadam, D. L. (ed)
Sources of Color Science. Cambridge, MIT Press, 1872 - 1873.
</p>
<p>Kelber A, Vorobyev M, Osorio D. (2003). Animal colour vision -
behavioural tests and physiological concepts. Biological Reviews, 78,
81 - 118.
</p>
<p>MacLeod DIA, Boynton RM. (1979). Chromaticity diagram showing
cone excitation by stimuli of equal luminance. Journal of the Optical
Society of America, 69, 1183-1186.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flowers)
vis.flowers &lt;- vismodel(flowers, visual = "apis", achromatic = "l")
tri.flowers &lt;- colspace(vis.flowers, space = "tri")

</code></pre>

<hr>
<h2 id='ttvertex'>vertex for the tetrahedral color space</h2><span id='topic+ttvertex'></span>

<h3>Description</h3>

<p>internal data for plotting devices.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage
color in a tetrahedral color space: A phylogenetic analysis of new world
buntings. The American Naturalist, 171(6), 755-776.
</p>

<hr>
<h2 id='vismodel'>Visual models</h2><span id='topic+vismodel'></span>

<h3>Description</h3>

<p>Calculates quantum catches at each photoreceptor. Both raw and relative values
can be returned, for use in a suite of colourspace and non-colourspace models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vismodel(
  rspecdata,
  visual = c("avg.uv", "avg.v", "bluetit", "ctenophorus", "star", "pfowl", "apis",
    "canis", "cie2", "cie10", "musca", "drosophila", "segment", "habronattus",
    "rhinecanthus"),
  achromatic = c("none", "bt.dc", "ch.dc", "st.dc", "md.r1", "dm.r1", "ra.dc", "cf.r",
    "ml", "l", "all"),
  illum = c("ideal", "bluesky", "D65", "forestshade"),
  trans = c("ideal", "bluetit", "blackbird"),
  qcatch = c("Qi", "fi", "Ei"),
  bkg = c("ideal", "green"),
  vonkries = FALSE,
  scale = 1,
  relative = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vismodel_+3A_rspecdata">rspecdata</code></td>
<td>
<p>(required) a data frame, possibly of class <code>rspec</code>, which
contains a column containing a wavelength range, named 'wl', and spectra
data in remaining columns.</p>
</td></tr>
<tr><td><code id="vismodel_+3A_visual">visual</code></td>
<td>
<p>the visual system to be used. Options are:
</p>

<ul>
<li><p> a data frame such as one produced containing by <code><a href="#topic+sensmodel">sensmodel()</a></code>, containing
user-defined sensitivity data for the receptors involved in colour vision.
The data frame must contain a <code>'wl'</code> column with the range of wavelengths included,
and the sensitivity for each other cone as a column.
</p>
</li>
<li> <p><code>'apis'</code>: Honeybee <em>Apis mellifera</em>.
</p>
</li>
<li> <p><code>'avg.uv'</code>: average avian UV system (default).
</p>
</li>
<li> <p><code>'avg.v'</code>: average avian V system.
</p>
</li>
<li> <p><code>'bluetit'</code>: Blue tit <em>Cyanistes caeruleus</em>.
</p>
</li>
<li> <p><code>'canis'</code>: Canid <em>Canis familiaris</em>.
</p>
</li>
<li> <p><code>'cie2'</code>: 2-degree colour matching functions for CIE models of human
colour vision. Functions are linear transformations of the 2-degree cone fundamentals
of Stockman &amp; Sharpe (2000), as ratified by the CIE (2006).
</p>
</li>
<li> <p><code>'cie10'</code>: 10-degree colour matching functions for CIE models of human
colour vision. Functions are linear transformations of the 10-degree cone fundamentals
of Stockman &amp; Sharpe (2000), as ratified by the CIE (2006).
</p>
</li>
<li> <p><code>'ctenophorus'</code>: Ornate dragon lizard <em>Ctenophorus ornatus</em>.
</p>
</li>
<li> <p><code>'musca'</code>: Housefly <em>Musca domestica</em>.
</p>
</li>
<li> <p><code>'drosophila'</code>: Vinegar fly <em>Drosophila melanogaster</em> (Sharkey et al. 2020).
</p>
</li>
<li> <p><code>'pfowl'</code>: Peafowl <em>Pavo cristatus</em>.
</p>
</li>
<li> <p><code>'segment'</code>: Generic tetrachromat 'viewer' for use in the segment analysis of Endler (1990).
</p>
</li>
<li> <p><code>'star'</code>: Starling <em>Sturnus vulgaris</em>.
</p>
</li>
<li> <p><code>'habronattus'</code>: Jumping spider <em>Habronattus pyrrithrix</em>.
</p>
</li>
<li> <p><code>'rhinecanthus'</code>: Triggerfish <em>Rhinecanthus aculeatus</em>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="vismodel_+3A_achromatic">achromatic</code></td>
<td>
<p>the sensitivity data to be used to calculate luminance (achromatic)
receptor stimulation. Either a vector containing the sensitivity for a single receptor,
or one of the options:
</p>

<ul>
<li> <p><code>'none'</code>: no achromatic stimulation calculated (default).
</p>
</li>
<li> <p><code>'bt.dc'</code>: Blue tit <em>Cyanistes caeruleus</em> double cone.
</p>
</li>
<li> <p><code>'ch.dc'</code>: Chicken <em>Gallus gallus</em> double cone.
</p>
</li>
<li> <p><code>'st.dc'</code>: Starling <em>Sturnus vulgaris</em> double cone.
</p>
</li>
<li> <p><code>'md.r1'</code>: Housefly <em>Musca domestica</em> R1-6 photoreceptor.
</p>
</li>
<li> <p><code>'dm.r1'</code>: Vinegar fly <em>Drosophila melanogaster</em> R1-6 photoreceptor.
</p>
</li>
<li> <p><code>'ra.dc'</code>: Triggerfish <em>Rhinecanthus aculeatus</em> double cone.
</p>
</li>
<li> <p><code>'cf.r'</code>: Canid <em>Canis familiaris</em> cone.
</p>
</li>
<li> <p><code>'ml'</code>: the summed response of the two longest-wavelength photoreceptors.
</p>
</li>
<li> <p><code>'l'</code>: the longest-wavelength photoreceptor.
</p>
</li>
<li> <p><code>'all'</code>: the summed response of all photoreceptors.
</p>
</li></ul>
</td></tr>
<tr><td><code id="vismodel_+3A_illum">illum</code></td>
<td>
<p>either a vector containing the illuminant, or one of the options:
</p>

<ul>
<li> <p><code>'ideal'</code>: homogeneous illuminance of 1 across wavelengths (default)
</p>
</li>
<li> <p><code>'bluesky'</code> open blue sky.
</p>
</li>
<li> <p><code>'D65'</code>: standard daylight.
</p>
</li>
<li> <p><code>'forestshade'</code> forest shade.
</p>
</li></ul>
</td></tr>
<tr><td><code id="vismodel_+3A_trans">trans</code></td>
<td>
<p>either a vector containing the ocular or environmental transmission
spectra, or one of the options:
</p>

<ul>
<li> <p><code>'ideal'</code>: homogeneous transmission of 1 across all wavelengths (default)
</p>
</li>
<li> <p><code>'bluetit'</code>: blue tit <em>Cyanistes caeruleus</em>
ocular transmission (from Hart et al. 2000).
</p>
</li>
<li> <p><code>'blackbird'</code>: blackbird <em>Turdus merula</em>
ocular transmission (from Hart et al. 2000).
</p>
</li></ul>
</td></tr>
<tr><td><code id="vismodel_+3A_qcatch">qcatch</code></td>
<td>
<p>Which quantal catch metric to return. Options are:
</p>

<ul>
<li> <p><code>'Qi'</code>: Quantum catch for each photoreceptor (default)
</p>
</li>
<li> <p><code>'fi'</code>: Quantum catch according to Fechner's law (the signal of the receptor
channel is proportional to the logarithm of the quantum catch)
</p>
</li>
<li> <p><code>'Ei'</code>: Hyperbolic-transformed quantum catch, where Ei = Qi / (Qi + 1).
</p>
</li></ul>
</td></tr>
<tr><td><code id="vismodel_+3A_bkg">bkg</code></td>
<td>
<p>background spectrum. Note that this will have no effect when <code>vonkries = FALSE</code>.
Either a vector containing the spectral data, or one of the options:
</p>

<ul>
<li> <p><code>'ideal'</code>: homogeneous illuminance of 1 across all wavelengths (default).
</p>
</li>
<li> <p><code>'green'</code>: green foliage.
</p>
</li></ul>
</td></tr>
<tr><td><code id="vismodel_+3A_vonkries">vonkries</code></td>
<td>
<p>logical. Should the von Kries colour correction transformation be applied?
(defaults to <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="vismodel_+3A_scale">scale</code></td>
<td>
<p>a value by which the illuminant will be multiplied. Useful for when the
illuminant is a relative value (i.e. transformed to a maximum of 1 or to a percentage),
and does not correspond to quantum flux units
(&mu;mol.s<sup>-1</sup>.m<sup>-2</sup>).
Useful values are, for example, 500 (for dim light) and 10000 (for bright
illumination). Note that if <code>vonkries = TRUE</code> this transformation has no effect.</p>
</td></tr>
<tr><td><code id="vismodel_+3A_relative">relative</code></td>
<td>
<p>should relative quantum catches be returned (i.e. is it a colour
space model? Defaults to <code>TRUE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>vismodel</code> containing the photon catches for each of the
photoreceptors considered. Information on the parameters used in the calculation are also
stored and can be called using the <code><a href="#topic+summary.vismodel">summary.vismodel()</a></code> function.
</p>


<h3>Note</h3>

<p>Built-in <code>visual</code>, <code>achromatic</code>, <code>illum</code>, <code>bkg</code> and <code>trans</code> are only defined
on the 300 to 700nm wavelength range. If you wish to work outside this range,
you will need to provide your own data.
</p>


<h3>Author(s)</h3>

<p>Thomas E. White <a href="mailto:thomas.white026@gmail.com">thomas.white026@gmail.com</a>
</p>
<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Vorobyev, M., Osorio, D., Bennett, A., Marshall, N., &amp; Cuthill, I.
(1998). Tetrachromacy, oil droplets and bird plumage colours. Journal Of Comparative
Physiology A-Neuroethology Sensory Neural And Behavioral Physiology, 183(5), 621-633.
</p>
<p>Hart, N. S., Partridge, J. C., Cuthill, I. C., Bennett, A. T. D. (2000).
Visual pigments, oil droplets, ocular media and cone photoreceptor distribution in two
species of passerine bird: the blue tit (Parus caeruleus L.) and the blackbird
(Turdus merula L.). Journal of Comparative Physiology A, 186, 375-387.
</p>
<p>Hart, N. S. (2001). The visual ecology of avian photoreceptors. Progress
In Retinal And Eye Research, 20(5), 675-703.
</p>
<p>Barbour H. R., Archer, M. A., Hart, N. S., Thomas, N., Dunlop, S. A.,
Beazley, L. D, Shand, J. (2002). Retinal characteristics of the Ornate Dragon
Lizard, Ctenophorus ornatus.
</p>
<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage
color in a tetrahedral color space: A phylogenetic analysis of new world buntings.
The American Naturalist, 171(6), 755-776.
</p>
<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour patterns
as birds see them. Biological Journal Of The Linnean Society, 86(4), 405-431.
</p>
<p>Chittka L. (1992). The colour hexagon: a chromaticity diagram
based on photoreceptor excitations as a generalized representation of
colour opponency. Journal of Comparative Physiology A, 170(5), 533-543.
</p>
<p>Stockman, A., &amp; Sharpe, L. T. (2000). Spectral sensitivities of
the middle- and long-wavelength sensitive cones derived from measurements in
observers of known genotype. Vision Research, 40, 1711-1737.
</p>
<p>CIE (2006). Fundamental chromaticity diagram with physiological axes.
Parts 1 and 2. Technical Report 170-1. Vienna: Central Bureau of the Commission
Internationale de l' Eclairage.
</p>
<p>Neitz, J., Geist, T., Jacobs, G.H. (1989) Color vision in the dog.
Visual Neuroscience, 3, 119-125.
</p>
<p>Sharkey, C. R., Blanco, J., Leibowitz, M. M., Pinto-Benito, D.,
&amp; Wardill, T. J. (2020). The spectral sensitivity of Drosophila photoreceptors.
Scientific reports, 10(1), 1-13.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sensdata">sensdata()</a></code> to retrieve or plot in-built spectral sensitivity data
used in <code><a href="#topic+vismodel">vismodel()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Dichromat (dingo)
data(flowers)
vis.dingo &lt;- vismodel(flowers, visual = "canis")
di.dingo &lt;- colspace(vis.dingo, space = "di")

# Trichromat (honeybee)
data(flowers)
vis.bee &lt;- vismodel(flowers, visual = "apis")
tri.bee &lt;- colspace(vis.bee, space = "tri")

# Tetrachromat (blue tit)
data(sicalis)
vis.bluetit &lt;- vismodel(sicalis, visual = "bluetit")
tcs.bluetit &lt;- colspace(vis.bluetit, space = "tcs")

# Tetrachromat (starling), receptor-noise model
data(sicalis)
vis.star &lt;- vismodel(sicalis, visual = "star", achromatic = "bt.dc", relative = FALSE)
dist.star &lt;- coldist(vis.star, achromatic = TRUE)

# Estimate quantum catches using a custom trichromatic visual phenotype
custom &lt;- sensmodel(c(330, 440, 550))
names(custom) &lt;- c("wl", "s", "m", "l")
vis.custom &lt;- vismodel(flowers, visual = custom)
tri.custom &lt;- colspace(vis.custom, space = "tri")

</code></pre>

<hr>
<h2 id='vissyst'>Animal visual systems data</h2><span id='topic+vissyst'></span>

<h3>Description</h3>

<p>Internal data for visual model calculations.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>


<h3>References</h3>

<p>Endler, J. A., &amp; Mielke, P. (2005). Comparing entire colour
patterns as birds see them. Biological Journal Of The Linnean Society,
86(4), 405-431.
</p>

<hr>
<h2 id='vol'>Plot a tetrahedral colour space</h2><span id='topic+vol'></span>

<h3>Description</h3>

<p>Produces a 3D colour volume in tetrahedral colour space when plotting a
non-interactive tetrahedral plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vol(
  tcsdata,
  type = c("convex", "alpha"),
  avalue = "auto",
  alpha = 0.2,
  grid = TRUE,
  fill = TRUE,
  new = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vol_+3A_tcsdata">tcsdata</code></td>
<td>
<p>(required) a data frame, possibly a result from the
<code><a href="#topic+colspace">colspace()</a></code> or <code><a href="#topic+tcspace">tcspace()</a></code> function, containing values for the 'x', 'y'
and 'z' coordinates as columns (labeled as such).</p>
</td></tr>
<tr><td><code id="vol_+3A_type">type</code></td>
<td>
<p>if &quot;convex&quot;, the colour volume is plotted using a convex hull
and if &quot;alpha&quot;, it is plotted using alphashapes.</p>
</td></tr>
<tr><td><code id="vol_+3A_avalue">avalue</code></td>
<td>
<p>if <code>type = "alpha"</code>, which alpha parameter value should be used
to compute the alphashape. <code>avalue = "auto"</code> (default) finds and use the
<code class="reqn">\alpha^*</code> value as defined in Gruson (2020).</p>
</td></tr>
<tr><td><code id="vol_+3A_alpha">alpha</code></td>
<td>
<p>transparency of volume (if <code>fill = TRUE</code>).</p>
</td></tr>
<tr><td><code id="vol_+3A_grid">grid</code></td>
<td>
<p>logical. if <code>TRUE</code> (default), draws the polygon outline defined by the points.</p>
</td></tr>
<tr><td><code id="vol_+3A_fill">fill</code></td>
<td>
<p>logical. if <code>TRUE</code> (default), fills the volume defined by the points.</p>
</td></tr>
<tr><td><code id="vol_+3A_new">new</code></td>
<td>
<p>logical. Should a new plot be started or draw over an open plot?
(defaults to <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="vol_+3A_...">...</code></td>
<td>
<p>additional graphical options. See <code><a href="graphics.html#topic+polygon">polygon()</a></code> and <code><a href="#topic+tetraplot">tetraplot()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code><a href="#topic+vol">vol()</a></code> creates a 3D colour volume within a static tetrahedral plot.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>
<p>Hugo Gruson
</p>


<h3>References</h3>

<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage color
in a tetrahedral color space: A phylogenetic analysis of new world buntings. The
American Naturalist, 171(6), 755-776.
</p>
<p>Stoddard, M. C., &amp; Stevens, M. (2011). Avian vision and the evolution of
egg color mimicry in the common cuckoo. Evolution, 65(7), 2004-2013.
</p>
<p>Maia, R., White, T. E., (2018) Comparing colors using visual models.
Behavioral Ecology, ary017 <a href="https://doi.org/10.1093/beheco/ary017">doi:10.1093/beheco/ary017</a>
</p>
<p>Gruson H. (2020). Estimation of colour volumes as concave hypervolumes using
<code class="reqn">\alpha</code>-shapes. Methods in Ecology and Evolution, 11(8), 955-963
<a href="https://doi.org/10.1111/2041-210X.13398">doi:10.1111/2041-210X.13398</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# For plotting
data(sicalis)
vis.sicalis &lt;- vismodel(sicalis, visual = "avg.uv")
tcs.sicalis &lt;- colspace(vis.sicalis, space = "tcs")
plot(tcs.sicalis)

# Convex hull
vol(tcs.sicalis, type = "convex")

# Alpha-shape
if (require("alphashape3d")) {
  vol(tcs.sicalis, type = "alpha", avalue = 1)
}
</code></pre>

<hr>
<h2 id='voloverlap'>Colour volume overlap</h2><span id='topic+voloverlap'></span>

<h3>Description</h3>

<p>Calculates the overlap between the volumes defined by two sets of points in cartesian
space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>voloverlap(
  colsp1,
  colsp2,
  type = c("convex", "alpha"),
  avalue = "auto",
  plot = FALSE,
  interactive = FALSE,
  col = c("blue", "red", "darkgrey"),
  fill = FALSE,
  new = TRUE,
  nsamp = 1000,
  psize = 0.001,
  lwd = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="voloverlap_+3A_colsp1">colsp1</code>, <code id="voloverlap_+3A_colsp2">colsp2</code></td>
<td>
<p>(required) data frame, possibly a result from the <code><a href="#topic+colspace">colspace()</a></code>
function, containing
values for the 'x', 'y' (and possibly 'z') coordinates as columns (labeled as such)</p>
</td></tr>
<tr><td><code id="voloverlap_+3A_type">type</code></td>
<td>
<p>if &quot;convex&quot;, the colour volume is plotted using a convex hull
and if &quot;alpha&quot;, it is plotted using alphashapes.</p>
</td></tr>
<tr><td><code id="voloverlap_+3A_avalue">avalue</code></td>
<td>
<p>if <code>type = alpha</code>, the alpha parameter values for <code>colsp1</code> and
<code>colsp2</code> respectively to compute the alphashapes. Can be a numeric of length
one if the same value is used in both cases. <code>avalue = "auto"</code> (default)
finds and use the <code class="reqn">\alpha^*</code> value as defined in Gruson (2020).</p>
</td></tr>
<tr><td><code id="voloverlap_+3A_plot">plot</code></td>
<td>
<p>logical. Should the volumes and points be plotted? (defaults to <code>FALSE</code>).
This only works for tetrahedral colourspaces at the moment.</p>
</td></tr>
<tr><td><code id="voloverlap_+3A_interactive">interactive</code></td>
<td>
<p>logical. If <code>TRUE</code>, uses the rgl engine for interactive plotting;
if <code>FALSE</code> then a static plot is generated.</p>
</td></tr>
<tr><td><code id="voloverlap_+3A_col">col</code></td>
<td>
<p>a vector of length 3 with the colours for (in order) the first volume,
the second volume, and the overlap.</p>
</td></tr>
<tr><td><code id="voloverlap_+3A_fill">fill</code></td>
<td>
<p>logical. should the two volumes be filled in the plot? (defaults to <code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="voloverlap_+3A_new">new</code></td>
<td>
<p>logical. Should a new plot window be called? If <code>FALSE</code>, volumes and their
overlap are plotted over the current plot (defaults to <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="voloverlap_+3A_nsamp">nsamp</code></td>
<td>
<p>if <code>type = "alpha"</code>, the number of points to be sampled for the
Monte Carlo computation. Stoddard &amp; Stevens(2011) use around 750,000 points,
but more or fewer might be required depending on the
degree of overlap.</p>
</td></tr>
<tr><td><code id="voloverlap_+3A_psize">psize</code></td>
<td>
<p>if <code>type = "alpha"</code> and <code>plot = TRUE</code>, sets the size to plot the points
used in the Monte Carlo computation.</p>
</td></tr>
<tr><td><code id="voloverlap_+3A_lwd">lwd</code></td>
<td>
<p>if <code>plot = TRUE</code>, sets the line width for volume grids.</p>
</td></tr>
<tr><td><code id="voloverlap_+3A_...">...</code></td>
<td>
<p>additional arguments passed to the plot. See <code><a href="#topic+vol">vol()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Calculates the overlap between the volumes defined by two set of points in
colourspace. The volume from the overlap is then given relative to:
</p>

<ul>
<li> <p><code>vsmallest</code> the volume of the overlap divided by the smallest of that defined
by the the two input sets of colour points. Thus, if one of the volumes is entirely
contained within the other, this overlap will be <code>vsmallest = 1</code>.
</p>
</li>
<li> <p><code>vboth</code> the volume of the overlap divided by the combined volume of both
input sets of colour points.
If <code>type = "alpha"</code>, If used, the output will be different:
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;s_in1, s_in2&#8288;</code> the number of sampled points that fall within each of the volumes
individually.
</p>
</li>
<li> <p><code>s_inboth</code> the number of sampled points that fall within both volumes.
</p>
</li>
<li> <p><code>s_ineither</code> the number of points that fall within either of the volumes.
</p>
</li>
<li> <p><code>psmallest</code> the proportion of points that fall within both volumes divided by the
number of points that fall within the smallest volume.
</p>
</li>
<li> <p><code>pboth</code> the proportion of points that fall within both volumes divided by the total
number of points that fall within both volumes.
</p>
</li></ul>



<h3>Note</h3>

<p>Stoddard &amp; Stevens (2011) originally obtained the volume overlap through Monte Carlo
simulations of points within the range of the volumes, and obtaining the frequency of
simulated values that fall inside the volumes defined by both sets of colour points.
</p>
<p>Stoddard &amp; Stevens (2011) also return the value of the overlap relative to one of
the volumes (in that case, the host species). However, for other applications
this value may not be what one expects to obtain if (1) the two
volumes differ considerably in size, or (2) one of the volumes is entirely contained
within the other. For this reason, we also report the volume relative to the union of
the two input volumes, which may be more adequate in most cases.
</p>


<h3>Author(s)</h3>

<p>Rafael Maia <a href="mailto:rm72@zips.uakron.edu">rm72@zips.uakron.edu</a>
</p>
<p>Hugo Gruson <a href="mailto:hugo.gruson+R@normalesup.org">hugo.gruson+R@normalesup.org</a>
</p>


<h3>References</h3>

<p>Stoddard, M. C., &amp; Prum, R. O. (2008). Evolution of avian plumage color
in a tetrahedral color space: A phylogenetic analysis of new world buntings. The
American Naturalist, 171(6), 755-776.
</p>
<p>Stoddard, M. C., &amp; Stevens, M. (2011). Avian vision and the evolution of
egg color mimicry in the common cuckoo. Evolution, 65(7), 2004-2013.
</p>
<p>Maia, R., White, T. E., (2018) Comparing colors using visual models.
Behavioral Ecology, ary017 <a href="https://doi.org/10.1093/beheco/ary017">doi:10.1093/beheco/ary017</a>
</p>
<p>Gruson H. (2020). Estimation of colour volumes as concave hypervolumes using
<code class="reqn">\alpha</code>-shapes. Methods in Ecology and Evolution, 11(8), 955-963
<a href="https://doi.org/10.1111/2041-210X.13398">doi:10.1111/2041-210X.13398</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sicalis)
tcs.sicalis.C &lt;- subset(colspace(vismodel(sicalis)), "C")
tcs.sicalis.T &lt;- subset(colspace(vismodel(sicalis)), "T")
tcs.sicalis.B &lt;- subset(colspace(vismodel(sicalis)), "B")

# Convex hull volume
voloverlap(tcs.sicalis.T, tcs.sicalis.B, type = "convex")
voloverlap(tcs.sicalis.T, tcs.sicalis.C, type = "convex", plot = TRUE)
voloverlap(tcs.sicalis.T, tcs.sicalis.C, type = "convex", plot = TRUE, col = seq_len(3))

# Alpha-shape volume
if (require("alphashape3d")) {
  voloverlap(tcs.sicalis.T, tcs.sicalis.B, type = "alpha", avalue = 1)
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
