<!DOCTYPE html><html><head><title>Help for package pbatR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pbatR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#affectionPhe'><p>Affection Phenotype Object</p></a></li>
<li><a href='#c2bat'><p>C2BAT</p></a></li>
<li><a href='#clean'><p>Cleaning pbat(), pbat.last(), pbat.m() output.</p></a></li>
<li><a href='#cluster'><p>Cluster result functions</p></a></li>
<li><a href='#cped'><p>Phenotype Object</p></a></li>
<li><a href='#generic'><p>&lsquo;pbat&rsquo; Object Generic Routines</p></a></li>
<li><a href='#obfuscate'><p>Obfuscating Pedigree and Phenotype Files</p></a></li>
<li><a href='#pbat'><p>PBAT Graphical and Command Line Interface</p></a></li>
<li><a href='#pbat.help'><p>Help and Reporting Information</p></a></li>
<li><a href='#pbat.set'><p>Set Pbat Parameters</p></a></li>
<li><a href='#pbat.status'><p>PBAT Status of Run</p></a></li>
<li><a href='#pbat.work'><p>pbatRwork Temp Directory</p></a></li>
<li><a href='#ped'><p>Pedigree Object</p></a></li>
<li><a href='#phe'><p>Phenotype Object</p></a></li>
<li><a href='#power'><p>Power Exploration</p></a></li>
<li><a href='#top'><p>top</p></a></li>
<li><a href='#write.pbat'><p>Saving &lsquo;pbat&rsquo; Object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.2-17</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-22</td>
</tr>
<tr>
<td>Title:</td>
<td>Pedigree/Family-Based Genetic Association Tests Analysis and
Power</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas Hoffmann &lt;tjhoffm@gmail.com&gt;, with contributions from Christoph Lange &lt;clange@hsph.harvard.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomas Hoffmann &lt;tjhoffm@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>survival, rootSolve</td>
</tr>
<tr>
<td>Suggests:</td>
<td>kinship2, tcltk</td>
</tr>
<tr>
<td>Description:</td>
<td>This R package provides power calculations via internal simulation methods. The package also provides a frontend to the now abandoned PBAT program (developed by Christoph Lange), and reads in the corresponding output and displays results and figures when appropriate. The license of this R package itself is GPL. However, to have the program interact with the PBAT program for some functionality of the R package, users must additionally obtain the PBAT program from Christoph Lange, and accept his license. Both the data analysis and power calculations have command line and graphical interfaces using tcltk.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://academic.oup.com/bioinformatics/article-abstract/22/24/3103/208723">https://academic.oup.com/bioinformatics/article-abstract/22/24/3103/208723</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-28 00:27:49 UTC; tom</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-29 10:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='affectionPhe'>Affection Phenotype Object</h2><span id='topic+affectionPhe'></span>

<h3>Description</h3>

<p>Creates a phenotype object from a pedigree object (see &lsquo;ped&rsquo; and
&lsquo;phe&rsquo;), preserving affection status.  Mostly used internally, or as a
substitute offset fix for the GUI.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>affectionPhe(ped, trait="affected", offset=0.0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="affectionPhe_+3A_ped">ped</code></td>
<td>
<p>An object of class <code>ped</code>.</p>
</td></tr>
<tr><td><code id="affectionPhe_+3A_trait">trait</code></td>
<td>
<p>Name for the trait in the generated phenotype file.</p>
</td></tr>
<tr><td><code id="affectionPhe_+3A_offset">offset</code></td>
<td>
<p>The offset to be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returns an object of class <code>phe</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.phe">read.phe</a></code>,
<code><a href="#topic+read.ped">read.ped</a></code>
</p>

<hr>
<h2 id='c2bat'>C2BAT</h2><span id='topic+c2bat'></span>

<h3>Description</h3>

<p>C2BAT (c) Matt McQueen, Jessica Su, Christoph Lange.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c2bat(
  snps,
  case.aa, case.Aa, case.AA,
  control.aa, control.Aa, control.AA,
  replicates=0,
  statistic="armitage",
  method="exact")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c2bat_+3A_snps">snps</code></td>
<td>
<p>Names of the snps (vector). Can be strings.</p>
</td></tr>
<tr><td><code id="c2bat_+3A_case.aa">case.aa</code></td>
<td>
<p>Number of cases with aa genotype (a vector
corresponding to the snps).</p>
</td></tr>
<tr><td><code id="c2bat_+3A_case.aa">case.Aa</code></td>
<td>
<p>Number of cases with Aa genotype.</p>
</td></tr>
<tr><td><code id="c2bat_+3A_case.aa">case.AA</code></td>
<td>
<p>Number of cases with AA genotype.</p>
</td></tr>
<tr><td><code id="c2bat_+3A_control.aa">control.aa</code></td>
<td>
<p>Number of controls with aa genotype.</p>
</td></tr>
<tr><td><code id="c2bat_+3A_control.aa">control.Aa</code></td>
<td>
<p>Number of controls with Aa genotype.</p>
</td></tr>
<tr><td><code id="c2bat_+3A_control.aa">control.AA</code></td>
<td>
<p>Number of controls with AA genotype.</p>
</td></tr>
<tr><td><code id="c2bat_+3A_replicates">replicates</code></td>
<td>
<p>This value should be set to zero for the first
analysis. Then, once the top snps have been selected, this should be
rerun with replicates&gt;1000 for those selected snps.</p>
</td></tr>
<tr><td><code id="c2bat_+3A_statistic">statistic</code></td>
<td>

<p><code>"armitage"</code> = the Armitage trend test.
<code>"logrank"</code> = logrank test
</p>
</td></tr>
<tr><td><code id="c2bat_+3A_method">method</code></td>
<td>

<p><code>"exact"</code> = exact likelihood function
<code>"approximate"</code> = approximate likelihood function
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output is formatted as follows:
SNP, case.aa, case.Aa, case.AA, control.aa, control.Aa, control.AA,
Monte-Carlo p-value, p-value, non-centrality parameter (for power
calculations), modelc2-based OR, allelic OR.
</p>
<p>Note that the non-centrality parameter and the ORs are independent of
the pvalues.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
c2bat( snps=c("snp1","snp2"),
       case.aa=c(10,11), case.Aa=c(30,31), case.AA=c(14,44),
       control.aa=c(33,32), control.Aa=c(100,99), control.AA=c(0,0) )

## End(Not run)
</code></pre>

<hr>
<h2 id='clean'>Cleaning pbat(), pbat.last(), pbat.m() output.</h2><span id='topic+pbat.clean'></span>

<h3>Description</h3>

<p>Removes many of the temporary output files from communicating with
PBAT. Make sure you save how PBAT was run for future reference. Most
useful for simulations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbat.clean(pbatObj, plots=TRUE, all.output=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clean_+3A_pbatobj">pbatObj</code></td>
<td>
<p>'pbat' object</p>
</td></tr>
<tr><td><code id="clean_+3A_plots">plots</code></td>
<td>
<p>TRUE/FALSE - delete plot output</p>
</td></tr>
<tr><td><code id="clean_+3A_all.output">all.output</code></td>
<td>
<p>DEPRECATED: No longer does anything, deleting files from other (potentially running) PBAT runs can be dangerous.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+pbat">pbat</a></code>
</p>

<hr>
<h2 id='cluster'>Cluster result functions</h2><span id='topic+is.finished'></span><span id='topic+pbat.load'></span><span id='topic+pbat.concatenate'></span>

<h3>Description</h3>

<p>Options when 'clusterRefresh' has been set to 0 (so that the current
process is batched). See <code><a href="#topic+pbat.set">pbat.set</a></code> for details.
</p>
<p>is.finished() returns if PBAT execution has terminated.
</p>
<p>pbat.load() loads in PBAT output.
</p>
<p>pbat.concatenate() concatenates output files (without reading them
into an R object).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.finished(pbatObj=NULL, clean=TRUE)
pbat.load(pbatObj=NULL)
pbat.concatenate(pbatObj=NULL, 
  filename="myResults.txt", clean=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cluster_+3A_pbatobj">pbatObj</code></td>
<td>
<p>'pbat' object (result of pbat.m, pbat.obj,
pbat.files). If NULL / left empty, then the results are taken from
what was run in the GUI interface.
</p>
</td></tr>
<tr><td><code id="cluster_+3A_clean">clean</code></td>
<td>

<p><em>pbat.load</em>:  If TRUE it will remove intermediate files that setup the
job submission interface (but no other results, and not the
command-files sent to pbat); see also pbat.clean.  If FALSE, nothing
else is done.  NOTE, when TRUE (which is the default), if you run
this more than once, you will get TRUE only once, and then FALSE the
rest of the times.
</p>
<p><em>pbat.concatenate</em>:  If clean is TRUE, then output that went into the
concatenated file will be deleted.  Default is FALSE, as this has
been a primary area of continual changes in PBAT, and so you may not
want to do this until you have checked the output.
</p>
</td></tr>
<tr><td><code id="cluster_+3A_filename">filename</code></td>
<td>
<p>output filename</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+pbat.set">pbat.set</a></code>
</p>
<p><code><a href="#topic+pbat">pbat</a></code>
</p>

<hr>
<h2 id='cped'>Phenotype Object</h2><span id='topic+as.cped'></span><span id='topic+is.cped'></span><span id='topic+read.cped'></span><span id='topic+fread.cped'></span><span id='topic+write.cped'></span><span id='topic+sort.cped'></span><span id='topic+plotCPed'></span>

<h3>Description</h3>

<p>Creates, tests, reads, or writes an object of class <code>cped</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.cped(x,
  pid="pid", id="id", idfath="idfath",
  idmoth="idmoth", sex="sex", affection="AffectionStatus",
  clearSym=FALSE)

is.cped(obj)

read.cped(filename, lowercase=TRUE, sym=TRUE, max=100, ...)
fread.cped(filename, ...)

write.cped(file, cped)

## S3 method for class 'cped'
sort(x,decreasing=FALSE,...)

plotCPed( cped, sink=NULL )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cped_+3A_x">x</code></td>
<td>

<p>An object of class <code>cped</code> or <code>data.frame</code> as described below.
</p>
<p>If <code>x</code> is of class <code>cped</code>, no other options are used.
</p>
<p>When <code>x</code> is of class <code>data.frame</code>, the columns have entries
that match the string parameters <code>idped,...,AffectionStatus</code>;
copy number variation measurements markers consist of <em>subsequent</em> columns
formated as follows: e.g. two copy number variants &lsquo;cnv1&rsquo; and &lsquo;cnv2&rsquo; with three intensities would need a total of six columns (three per cnv) past the first six named 'cnv1.1', 'cnv1.2', 'cnv1.3', 'cnv2.1', 'cnv2.2', 'cnv2.3'. See the examples below.
</p>
<p>A slightly different format can be used when reading data from disk, that is more or less consistent with how a pedigree file is loaded from disk. See the details below.
</p>
</td></tr>
<tr><td><code id="cped_+3A_pid">pid</code></td>
<td>
<p>String corresponding to column name for pedigree id.</p>
</td></tr>
<tr><td><code id="cped_+3A_id">id</code></td>
<td>
<p>String corresponding to column name for subject id.</p>
</td></tr>
<tr><td><code id="cped_+3A_idfath">idfath</code></td>
<td>
<p>String corresponding to column name for father id.</p>
</td></tr>
<tr><td><code id="cped_+3A_idmoth">idmoth</code></td>
<td>
<p>String corresponding to column name for mother id.</p>
</td></tr>
<tr><td><code id="cped_+3A_sex">sex</code></td>
<td>
<p>String corresponding to column name for sex.</p>
</td></tr>
<tr><td><code id="cped_+3A_affection">affection</code></td>
<td>
<p>String corresponding to column name for affection
status.</p>
</td></tr>
<tr><td><code id="cped_+3A_filename">filename</code></td>
<td>
<p>Filename to open; does not need .phe extension. See the details below for the file format.</p>
</td></tr>
<tr><td><code id="cped_+3A_lowercase">lowercase</code></td>
<td>
<p>When TRUE (and sym is FALSE), enforces all headers to
lowercase for convenience.</p>
</td></tr>
<tr><td><code id="cped_+3A_...">...</code></td>
<td>
<p>Options for <code><a href="utils.html#topic+read.table">read.table</a></code>, used only when sym is
FALSE.  Do <em>not</em> put in <code>header=TRUE</code>, as this will
cause an error, as the header is automatically loaded.
</p>
<p>With the proper file formatting, this should not be used.
</p>
</td></tr>
<tr><td><code id="cped_+3A_file">file</code></td>
<td>
<p>string representing filename, or a connection for file output</p>
</td></tr>
<tr><td><code id="cped_+3A_cped">cped</code></td>
<td>
<p>an object of class <code>cped</code></p>
</td></tr>
<tr><td><code id="cped_+3A_obj">obj</code></td>
<td>
<p>an object</p>
</td></tr>
<tr><td><code id="cped_+3A_sym">sym</code></td>
<td>
<p>When TRUE, only the header of the file is read in; only
PBAT will load in the file.  When FALSE, the entire file will be
read in, and can be modified before using with PBAT.</p>
</td></tr>
<tr><td><code id="cped_+3A_max">max</code></td>
<td>
<p>When sym is TRUE, the amount of headers to read in before
going pure symbolic (so that the SNP usage consistency will not be
assessed by pbatR, only by PBAT).</p>
</td></tr>
<tr><td><code id="cped_+3A_clearsym">clearSym</code></td>
<td>
<p>When TRUE, if a symbolic file is found, it will be
read in; otherwise, it will stay symbolic.</p>
</td></tr>
<tr><td><code id="cped_+3A_decreasing">decreasing</code></td>
<td>
<p>Whether to sort in decreasing/increasing order.</p>
</td></tr>
<tr><td><code id="cped_+3A_sink">sink</code></td>
<td>
<p>For &lsquo;plot.cped&rsquo;, this is the name of a pdf file to output all of the plots to (there will be one plot per page).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When reading in a file on disk using <code>read.cped</code>, a &lsquo;.cped&rsquo; file should have the following format. The file should be formatted as follows. The first six columns are <em>unlabeled</em> (1) the pedigree id, (2) the individual id, (3) the father id, (4) the mother id, (5) sex [0=missing, 1=male, 2=female], and (6) AffectionStatus [0=missing, 1=unaffected, 2=affected]. The subsequent columns correspond to the intensities. So, suppose we have cnv1 and cnv2. The first line of the file would contain &lsquo;cnv1 cnv2&rsquo;. Then the subsequent lines would correspond to each individual, the first six columns being as described, and then NUMINTENSITY columns per cnv for a total of 6+2*NUMINTENSITY data columns. NUMINTENSITY is just however many intensities there are per cnv, you will need to specify this number at analysis time. NOTE: MISSING DATA in a cped file should be coded as &lsquo;-1234.0&rsquo;, rather than the usual &lsquo;.&rsquo; or &lsquo;-&rsquo; (technically the &lsquo;.&rsquo; and &lsquo;-&rsquo; should still work with <code>fread.cped</code>, and when <code>sym=FALSE</code>).
</p>
<p>The best way to see all of this in action is to run the code in the examples below, and look at the cped file produced from it.
</p>
<p>&lsquo;plotCPed&rsquo; plots the data similar to the &lsquo;plotPed&rsquo; routine (in fact it transforms the data to use it).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.ped">read.ped</a></code>,
<code><a href="#topic+read.cped">read.cped</a></code>
<code><a href="#topic+write.cped">write.cped</a></code>,
<code><a href="#topic+plotCPed">plotCPed</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>###################
## First Example ##

## A highly artificial example with not enough subjects to be run;
##  however, it demonstrates how to put data in it.
## We have two cnvs here, cnv1 and cnv2.
## The data is just completely random.
set.seed(13)
x &lt;- data.frame( pid       = c(1,1,1,1,1),
                 id        = c(1,2,3,4,5),
                 idfath    = c(4,4,4,0,0),
                 idmoth    = c(5,5,5,0,0),
                 sex       = c(1,2,1,1,2),
                 AffectionStatus = c(1,0,0,1,0),
                 cnv1.1     = runif(5),
                 cnv1.2     = runif(5),
                 cnv1.3     = runif(5),
                 cnv2.1     = runif(5),
                 cnv2.2     = runif(5),
                 cnv2.3     = runif(5) )
x
myCPed &lt;- as.cped( x ) # Mark it with the class 'cped'
myCPed

## Not run: 
####################
## Second Example ##

## Again, a completely random dataset.
## Here we go through an analysis of it.
## However, see pbat.m for many more details on all of the options.
## Create a completely random dataset with one cnv.
set.seed(13)
NUMTRIOS &lt;- 500
## The data is completely random, it does not really make any sense.
cped &lt;- as.cped(data.frame(
  pid             = kronecker(1:NUMTRIOS, rep(1,3)),
  id              = rep(1:3, NUMTRIOS),
  idfath          = rep(c(0,0,1), NUMTRIOS),
  idmoth          = rep(c(0,0,2), NUMTRIOS),
  sex             = rep(c(2,1,1), NUMTRIOS),
  AffectionStatus = rep(c(0,0,2), NUMTRIOS),
  cnv1.1          = runif(3*NUMTRIOS),
  cnv1.2          = runif(3*NUMTRIOS),
  cnv1.3          = runif(3*NUMTRIOS)))

## Print out part of the dataset
print(head(cped))

## Command line run
pbat.work() ## Makes the intermediate files go in ./pbatRwork directory

## - Analyzing the first intensity
res1 &lt;- pbat.m(AffectionStatus ~ NONE, ped=cped, phe=NULL, fbat="gee",
                cnv.intensity=1, cnv.intensity.num=3, offset="none")
pbat.clean(res1, all.output=TRUE) ## Removes all intermediate files
## - Analyzing the second intensity
res2 &lt;- pbat.m(AffectionStatus ~ NONE, ped=cped, phe=NULL, fbat="gee",
                cnv.intensity=2, cnv.intensity.num=3, offset="none")
pbat.clean(res2, all.output=TRUE)
## - Analyzing the third intensity
res3 &lt;- pbat.m(AffectionStatus ~ NONE, ped=cped, phe=NULL, fbat="gee",
                cnv.intensity=3, cnv.intensity.num=3, offset="none")
pbat.clean(res3, all.output=TRUE)

pbat.unwork() ## Close up work (head to original working directory)

## Print all of the results
print(res1$results)
print(res2$results)
print(res3$results)

## Or put all the results together and write to file
res1$results &lt;- rbind(res1$results, res2$results, res3$results)
write.pbat(res1, "cpedResults.csv")

## Otherwise, we could write the data to disk, 
##  and run with the GUI interface
## Write the data to disk:
write.cped("cped.cped", cped)

## End(Not run)
</code></pre>

<hr>
<h2 id='generic'>&lsquo;pbat&rsquo; Object Generic Routines</h2><span id='topic+plot.pbat'></span><span id='topic+print.pbat'></span><span id='topic+summary.pbat'></span>

<h3>Description</h3>

<p>Summary routines for the results of &lsquo;pbat&rsquo;, &lsquo;pbat.last&rsquo;, &lsquo;pbat.m&rsquo;,
&lsquo;pbat.obj&rsquo;, and &lsquo;pbat.files&rsquo;.
</p>
<p>Only a logrank analysis can be plotted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pbat'
print(x,...)

## S3 method for class 'pbat'
summary(object,...)

## S3 method for class 'pbat'
plot(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generic_+3A_x">x</code></td>
<td>
<p>&lsquo;pbat&rsquo; object (result of pbat.m(...) or pbat().</p>
</td></tr>
<tr><td><code id="generic_+3A_object">object</code></td>
<td>
<p>&lsquo;pbat&rsquo; object (result of pbat.m(...) or pbat().</p>
</td></tr>
<tr><td><code id="generic_+3A_...">...</code></td>
<td>
<p>Ignored.  Needed for S3 generic method consistency.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Jiang, H., Harrington, D., Raby, B. A., Bertram, L., Blacker, D., Weiss, S. T. &amp; Lange,
C. (2005) Family-based association test for time-to-onset data with time-dependent
differences between the hazard functions. Genetic Epidemiology.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pbat">pbat</a></code>,
<code><a href="#topic+pbat.last">pbat.last</a></code>
</p>

<hr>
<h2 id='obfuscate'>Obfuscating Pedigree and Phenotype Files</h2><span id='topic+obfuscate'></span>

<h3>Description</h3>

<p>Randomly permutes the data in a pedigree or phenotype file and changes
column headers so as to mangle the data. Used for debugging requests,
in the hopes that the mangled data will produce the same bad output,
but will not be identifiable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>obfuscate(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="obfuscate_+3A_obj">obj</code></td>
<td>
<p>object of class &lsquo;ped&rsquo; or &lsquo;phe&rsquo; to be used (the same object
type is returned, only mangled; you must write this out to file).</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+pbat">pbat</a></code>
<code><a href="#topic+pbat.help">pbat.help</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ped &lt;- read.ped( "myped" );  ## reads in myped.ped
oped &lt;- obfuscate( "myped" );
write.ped( "obfuscate.ped", ped );

phe &lt;- read.phe( "myphe" );  ## reads in myphe.phe
ophe &lt;- obfuscate( "myphe" );
write.phe( "obfuscate.phe", ped );

## End(Not run)
</code></pre>

<hr>
<h2 id='pbat'>PBAT Graphical and Command Line Interface</h2><span id='topic+pbat'></span><span id='topic+pbat.last'></span><span id='topic+pbat.last.rawResults'></span><span id='topic+pbat.m'></span><span id='topic+pbat.obj'></span><span id='topic+pbat.files'></span><span id='topic+pbat.create.commandfile'></span>

<h3>Description</h3>

<p>The following routines are for the graphical and command line pbat
interface.  The command line interfaces are listed in an order of
suggested usage.  Most users of the command line will only want to use
<code>pbat.m</code>.
</p>
<p><code>pbat</code> runs a GUI (Graphical User Interface) for pbat.
</p>
<p><code>pbat.last</code> returns an object of class <code>pbat</code> of the last
command file run from running <code>pbat()</code>. Note this is also
returned from <code>pbat</code>.
However, this command is provided because rerunning a command in
pbat can be a very time-consuming process).
</p>
<p><code>pbat.last.rawResults</code> prints out the raw text file of the
output (particularly useful if the output of pbat cannot be parsed
properly, in the unexpected event the output could not be parsed
correctly). This should work even with the new option of not loading
the output in.
</p>
<p><code>pbat.m</code> runs pbat according to an expression, from <code>phe</code> class
(phenotype information), <code>ped</code> class (pedigree information), and
various options.
</p>
<p><code>pbat.obj</code> runs pbat with a <code>ped</code> class object (pedigree
information), a &lsquo;phe&rsquo; class object (phenotype information), and
various other options.
</p>
<p><code>pbat.files</code> runs pbat according to a set of filenames and
commands.
</p>
<p><code>pbat.create.commandfile</code> creates a command file for Christoph
Lange's pbat software with respect to two files on disk (.phe, .ped).
</p>
<p>Some options are only available for the respective pbat-gee (G),
pbat-pc (P), pbat-logrank (L).
If a parameter is &lsquo;R&rsquo;equired for a specific version, it will be denoted,
for example, by (G-R).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbat()

pbat.last()

pbat.last.rawResults()

pbat.m( 
  formula, phe, ped, fbat="",
  max.pheno=1, min.pheno=1,
  null="no linkage, no association", alpha=0.05,
  trans.pheno="none", trans.pred="none", trans.inter="none",
  scan.pred="all", scan.inter="all",
  scan.genetic="additive",
  offset="gee",
  screening="conditional power", distribution="default",
  logfile="",
  max.gee=1,
  max.ped=14, min.info=0,
  incl.ambhaplos=TRUE, infer.mis.snp=FALSE,
  sub.haplos=FALSE, length.haplos=2, adj.snps=TRUE,
  overall.haplo=FALSE, cutoff.haplo=FALSE,
  output="normal",
  max.mating.types=10000,
  commandfile="",
  future.expansion=NULL,
  LOAD.OUTPUT=TRUE,
  monte=0,
  mminsnps=NULL, mmaxsnps=NULL,
  mminphenos=NULL, mmaxphenos=NULL,
  env.cor.adjust=FALSE,
  gwa=FALSE,
  snppedfile=FALSE,
  extended.pedigree.snp.fix=FALSE,
  new.ped.algo=FALSE,
  cnv.intensity=2, cnv.intensity.num=3 )

pbat.obj(
  phe, ped, file.prefix, phenos="", 
  offset="gee", LOAD.OUTPUT=TRUE, ...)

pbat.files( 
  pedfile, phefile, fbat="gee",
  commandfile="",
  logrank.outfile="",
  preds="", preds.order="",
  max.pheno=1,
  LOAD.OUTPUT=TRUE,
  ...)

pbat.create.commandfile(
  pedfile, phefile="",
  snps="",
  phenos="", time="", # (set only one)
  preds="", preds.order="",
  inters="",
  groups.var="", groups="",
  fbat="gee",
  censor="",
  max.pheno=1, min.pheno=1,
  null="no linkage, no association", alpha=0.05,
  trans.pheno="none", trans.pred="none", trans.inter="none",
  scan.pred="all", scan.inter="all",
  scan.genetic="additive",
  offset="gee",
  screening="conditional power", distribution="default",
  logfile="",
  max.gee=1,
  max.ped=7, min.info=0,
  haplos=NULL, incl.ambhaplos=TRUE, infer.mis.snp=FALSE,
  sub.haplos=FALSE, length.haplos=2, adj.snps=TRUE,
  overall.haplo=FALSE, cutoff.haplo=FALSE,
  output="normal",
  max.mating.types=10000,
  commandfile="",
  future.expansion=NULL,
  LOGFILE.OVERRIDE=TRUE,
  monte=0,
  mminsnps=NULL, mmaxsnps=NULL,
  mminphenos=NULL, mmaxphenos=NULL,
  env.cor.adjust=FALSE,
  gwa=FALSE,
  snppedfile=FALSE,
  extended.pedigree.snp.fix=FALSE,
  new.ped.algo=FALSE,
  cnv.intensity=2, cnv.intensity.num=3 )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pbat_+3A_formula">formula</code></td>
<td>
<p>Symbolic expression describing what should be
processed.  See &lsquo;examples&rsquo; for more information.</p>
</td></tr>
<tr><td><code id="pbat_+3A_phe">phe</code></td>
<td>
<p>&lsquo;phe&rsquo; object as described in <code><a href="#topic+write.phe">write.phe</a></code>. If
you do not have a phe file set this to NULL (i.e when you are only
using AffectionStatus from the pedigree).</p>
</td></tr>
<tr><td><code id="pbat_+3A_ped">ped</code></td>
<td>
<p>&lsquo;ped&rsquo; object as described in <code><a href="#topic+write.ped">write.ped</a></code>.</p>
</td></tr>
<tr><td><code id="pbat_+3A_file.prefix">file.prefix</code></td>
<td>
<p>Prefix of the output datafile (phe &amp; ped must match)</p>
</td></tr>
<tr><td><code id="pbat_+3A_pedfile">pedfile</code></td>
<td>
<p>Name of the pedigree file (.ped/.pped/.cped) in PBAT-format
(extension &lsquo;.ped&rsquo; is optional).</p>
</td></tr>
<tr><td><code id="pbat_+3A_phefile">phefile</code></td>
<td>
<p>Name of the phenotype file (.phe) in PBAT-format.
The default assumes the same prefix as that in 'pedfile'. Leave
empty or set to the empty string &quot;&quot; if you do not have a phenotype
file (i.e. you are only using AffecitonStatus). In the case of no
phenotype file, one must be created; it will be in <code>empty_phe.phe</code>,
and requires loading in the pedigree file into R.</p>
</td></tr>
<tr><td><code id="pbat_+3A_...">...</code></td>
<td>
<p>Options in higher level functions to be passed to
'pbat.create.commandfile'.</p>
</td></tr>
<tr><td><code id="pbat_+3A_fbat">fbat</code></td>
<td>
<p>Selects the fbat statistic used the data analysis.
</p>
<p><code>"gee"</code> = The FBAT-GEE statistic simplifies to the standard
univariate FBAT-statistic. If several phenotypes are selected, all
phenotypes are tested simultaneously, using FBAT-GEE. The FBAT-GEE
statistic can handle any type of multivariate data.
</p>
<p><code>"pc"</code> = FBAT extension for longitudinal phenotypes and repeated
measurements.
</p>
<p><code>"logrank"</code> = FBAT-extensions of the classical LOGRANK and WILCOXON
tests for time-on-onset data. Kaplan-Meier plots for the
analyzed data set will be generated and plotted.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_max.pheno">max.pheno</code></td>
<td>
<p>(G,P) The maximum number of phenotypes that will be
analyzed in the FBAT-statistic.</p>
</td></tr>
<tr><td><code id="pbat_+3A_min.pheno">min.pheno</code></td>
<td>
<p>(G,P) The minimum number of phenotypes that will be
analyzed in the FBAT-statistic.</p>
</td></tr>
<tr><td><code id="pbat_+3A_null">null</code></td>
<td>
<p>Specification of the null-hypothesis.
</p>
<p><code>"no linkage, no association"</code> = Null-hypothesis of no linkage and no
association.
</p>
<p><code>"linkage, no association"</code> = Null-hypothesis of linkage, but no
association.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_alpha">alpha</code></td>
<td>
<p>Specification of the significance level.</p>
</td></tr>
<tr><td><code id="pbat_+3A_trans.pheno">trans.pheno</code></td>
<td>
<p>Transformation of the selected phenotypes.
</p>
<p><code>"none"</code> = no transformation
</p>
<p><code>"ranks"</code> = transformation to ranks
</p>
<p><code>"normal score"</code> = transformation to normal score
</p>
<p>The default choice is <code>"none"</code>, although it recommended to use
transformation to normal scores for quantitative phenotypes.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_trans.pred">trans.pred</code></td>
<td>
<p>Transformation of the selected predictor
variables/covariates:
</p>
<p><code>"none"</code> = no transformation
</p>
<p><code>"ranks"</code> = transformation to ranks
</p>
<p><code>"normal score"</code> = transformation to normal score
</p>
<p>The default choice is <code>"none"</code>, although it recommended to use
transformation to normal scores for quantitative covariates.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_trans.inter">trans.inter</code></td>
<td>
<p>Transformation of the selected interaction variables
</p>
<p><code>"none"</code> = no transformation
</p>
<p><code>"ranks"</code> = transformation to ranks
</p>
<p><code>"normal score"</code> = transformation to normal score
</p>
<p>The default choice is <code>"none"</code>, although it recommended to use
transformation to normal scores for quantitative interaction
variables.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_scan.pred">scan.pred</code></td>
<td>
<p>(G,P) Computation of all covariate sub-models:
</p>
<p><code>"all"</code> = The selected FBAT statistic is computed with
adjustment for all selected covariates/predictors.
</p>
<p><code>"subsets"</code> = The selected FBAT statistic is computed for all
possible subsets of the selected covariates/predictor variables.  The
command is particularly useful to examine the dependence of
significant results on the selection of a covariate model.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_scan.inter">scan.inter</code></td>
<td>
<p>(G,P) Computation of all interaction sub-models:
</p>
<p><code>"all"</code> = The selected FBAT statistic is computed including all
selected interaction variables.
</p>
<p><code>"subsets"</code> = The selected FBAT statistic is computed for all
posible subsets of the interaction variables.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_scan.genetic">scan.genetic</code></td>
<td>
<p>Specification of the mode of inheritance:
</p>
<p><code>"additive"</code> = Additive model
</p>
<p><code>"dominant"</code> = Dominant model
</p>
<p><code>"recessive"</code> = Recessive model
</p>
<p><code>"heterozygous advantage"</code> = Heterozygous advantage model
</p>
<p><code>"all"</code> = The FBAT-statistics are computed for all 4 genetic models
</p>
</td></tr>
<tr><td><code id="pbat_+3A_offset">offset</code></td>
<td>
<p>Specification of the covariate/predictor variables
adjustment:
</p>
<p><code>"none"</code> = No adjustments for covariates/predictor
variables. You need to select this for dichotomous traits.
</p>
<p><code>"max power"</code> = Offset (=FBAT adjustment for 
covariates and interaction variables) that maximizes 
the power of the FBAT-statistic (computationally slow,
efficiency dependent on the correct choice of
the mode of inheritance)
</p>
<p><code>"gee + marker score"</code> = Offset (=FBAT adjustment for
covariates and interaction variables) based on standard 
phenotypic residuals obtained by GEE-estimation including
the expected marker score (E(X|H0)), all
covariates and interaction variables.
</p>
<p><code>"gee"</code> = Offset (=FBAT adjustment for covariates 
and interaction variables) based on standard phenotypic
residuals obtained by GEE-estimation including all
covariates and interaction variables. (default - most
of the time, with the exception of selecting from
the gui interface (not the command line) AffectionStatus)
</p>
<p><code>(numeric value)</code> = This only works for AffectionStatus;
set a numeric value (i.e. &lsquo;0.13&rsquo; without the '' marks) to this.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_screening">screening</code></td>
<td>

<p>Specification of the screening methods to handle the multiple
comparison problem for multiple SNPs/haplotypes and a set of
phenotypes.
</p>
<p><code>"conditional power"</code> = Screening based on conditional power
(parametric approach)
</p>
<p><code>"wald"</code> = Screening based on Wald-tests (non-parametric approach)
</p>
</td></tr>
<tr><td><code id="pbat_+3A_distribution">distribution</code></td>
<td>

<p>Screening specification of the empirical phenotypic distribution
</p>
<p><code>"default"</code>
</p>
<p><code>"jiang"</code> = Approach by Jiang et al (2006)
</p>
<p><code>"murphy"</code> = Approach by Murphy et al (2006)
</p>
<p><code>"naive"</code> = Naive allele freq estimator
</p>
<p><code>"observed"</code> = Observed allele frequencies
</p>
</td></tr>
<tr><td><code id="pbat_+3A_logfile">logfile</code></td>
<td>
<p>Specification of the log-file. By default, PBAT
selects an unique file-name for the log-file, i.e. &quot;pbatlog...&quot;.</p>
</td></tr>
<tr><td><code id="pbat_+3A_max.gee">max.gee</code></td>
<td>
<p>(G) Specification of the maximal number of iterations
in the GEE-estimation procedure.</p>
</td></tr>
<tr><td><code id="pbat_+3A_max.ped">max.ped</code></td>
<td>
<p>Specification of the maximal number of proband in one
extended pedigrees.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_min.info">min.info</code></td>
<td>
<p>Specification of the minimum number of informative
families required for the computation of the FBAT-statistics.</p>
</td></tr>
<tr><td><code id="pbat_+3A_incl.ambhaplos">incl.ambhaplos</code></td>
<td>
<p>This command defines the handling of ambiguous
haplotypes in the haplotypes analysis. Choices:
</p>
<p><code>TRUE</code> = Ambiguous haplotypes (phase can not be inferred) are included
in the analysis and are weighted according to their estimated
frequencies in the probands.
</p>
<p><code>FALSE</code> = Ambiguous haplotypes are excluded from the analysis.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_infer.mis.snp">infer.mis.snp</code></td>
<td>
<p>Handling of missing genotype information in the
haplotypes analysis.
</p>
<p><code>FALSE</code> = Individuals with missing genotype information are
excluded from the analysis. This is the analysis also implemented in
the HBAT option of the FBAT-program.
</p>
<p><code>TRUE</code> = Individuals with missing genotype information are
included in the analysis. The algorithm of Horvath et al (2004) is
applied to all individuals, even if they have missing genotype
information. This results in more ambiguous haplotypes.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_sub.haplos">sub.haplos</code></td>
<td>

<p><code>FALSE</code> = The haplotypes defined by the all SNPs given in the
haplotype-block definition are analyzed.
</p>
<p><code>TRUE</code> = All haplotypes are analyzed that are defined by any subset of
SNPs in the haplotypes block definition.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_length.haplos">length.haplos</code></td>
<td>
<p>Defines the haplotype length when subhaplos=<code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="pbat_+3A_adj.snps">adj.snps</code></td>
<td>
<p>Takes effect when subhaplos=<code>TRUE</code>.
</p>
<p><code>FALSE</code> = All sub-haplotypes are analyzed
</p>
<p><code>TRUE</code> = Only the sub-haplotypes are analyzed for which the first
constituting SNPs are adjacent.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_overall.haplo">overall.haplo</code></td>
<td>
<p>Specification of an overall haplotypes test. When
this command is included in the batch-file, only one level of the
<code>"groups"</code> variable can be specified.
</p>
<p><code>FALSE</code> = no overall test
</p>
<p><code>TRUE</code> = an overall test is computed testing all haplotypes defined by
the same set of SNPs simultaneously. This option can not be applied
when sub.haplos=<code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_cutoff.haplo">cutoff.haplo</code></td>
<td>
<p>The minimum haplotypes frequency so that a
haplotypes is included in the overall test.</p>
</td></tr>
<tr><td><code id="pbat_+3A_output">output</code></td>
<td>

<p><code>"normal"</code> = Normal PBAT output.
</p>
<p><code>"short"</code> = Shorter output.  This is mostly for use in
conjunction with 'gwa', where there is a lot of output.
</p>
<p><code>"detailed"</code> = Detailed output for each family is created.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_max.mating.types">max.mating.types</code></td>
<td>
<p>Maximal number of mating types in the haplotype
analysis.</p>
</td></tr>
<tr><td><code id="pbat_+3A_commandfile">commandfile</code></td>
<td>
<p>Name of the temporary command file that will
be created to send to the pbat.  It is suggested to leave
this blank, and an appropriate name will be chosen with a time
stamp.</p>
</td></tr>
<tr><td><code id="pbat_+3A_future.expansion">future.expansion</code></td>
<td>
<p>(Only included for future expansion of pbat.)
A vector of strings for extra lines to write to the batchfile for
pbat.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_logrank.outfile">logrank.outfile</code></td>
<td>
<p>(L) Name of the file to store the R source code to
generate the plots for logrank analysis.</p>
</td></tr>
<tr><td><code id="pbat_+3A_snps">snps</code></td>
<td>
<p>Vector of strings for the SNPs to process. Default
processes all of the SNPs.</p>
</td></tr>
<tr><td><code id="pbat_+3A_phenos">phenos</code></td>
<td>
<p>(G,P) Vector of strings for the phenotypes/traits for the
analysis.  If none are specified, then all are analyzed.  (Note:
this <em>must be left empty for logrank analysis</em>, instead specify
the time to onset with the time variable. </p>
</td></tr>
<tr><td><code id="pbat_+3A_time">time</code></td>
<td>
<p>(L-R) Time to onset variable.  &lsquo;phenos&rsquo; cannot be specified
when this is used, but it <em>must be set for logrank</em>.</p>
</td></tr>
<tr><td><code id="pbat_+3A_preds">preds</code></td>
<td>
<p>Vector of strings for the covariates for the test
statistic.</p>
</td></tr>
<tr><td><code id="pbat_+3A_preds.order">preds.order</code></td>
<td>
<p>Vector of integers indicating the order of 'preds'
- the order for the vector of covariates for the test
statistic.</p>
</td></tr>
<tr><td><code id="pbat_+3A_inters">inters</code></td>
<td>
<p>Vector of strings for the interaction variables.</p>
</td></tr>
<tr><td><code id="pbat_+3A_groups.var">groups.var</code></td>
<td>
<p>String for the grouping variable.</p>
</td></tr>
<tr><td><code id="pbat_+3A_groups">groups</code></td>
<td>
<p>Vector of strings corresponding to the groups of the
grouping variable (groupsVar).</p>
</td></tr>
<tr><td><code id="pbat_+3A_censor">censor</code></td>
<td>
<p>(L-R) String of the censoring variables. In the
corresponding data, this variable has to be binary.</p>
</td></tr>
<tr><td><code id="pbat_+3A_haplos">haplos</code></td>
<td>
<p>List of string vectors representing the haplotype
blocks for the haplotype analysis.
For example,
<code>list(block1=c("m1","m2"), block2=c("m3","m4"))</code>
defines 2 haplotype-blocks where the first block
is defined by SNPs m1 and m2, and the second by SNPs
m3 and m4.
</p>
</td></tr>
<tr><td><code id="pbat_+3A_logfile.override">LOGFILE.OVERRIDE</code></td>
<td>
<p>When using the 'sym' option in read.ped and
read.phe, when this is set to TRUE (default), the PBAT logfile is
put in the current working directory; if FALSE, then it is put in
the same directory as the datafile.</p>
</td></tr>
<tr><td><code id="pbat_+3A_load.output">LOAD.OUTPUT</code></td>
<td>
<p>When TRUE, loads the output into R (generally
recommended). When FALSE, it leaves it in the output left from PBAT
(in case output is too large to load into memory).</p>
</td></tr>
<tr><td><code id="pbat_+3A_monte">monte</code></td>
<td>
<p>When this is nonzero, monte-carlo based methods are used
to compute the p-values instead, according to the number of
iterations supplied. 1000 iterations is suggested.</p>
</td></tr>
<tr><td><code id="pbat_+3A_mminsnps">mminsnps</code></td>
<td>
<p>Multi-marker multi-phenotype tests: the minimum number
of snps to be tested.</p>
</td></tr>
<tr><td><code id="pbat_+3A_mmaxsnps">mmaxsnps</code></td>
<td>
<p>Multi-marker multi-phenotype tests: the maximum number
of snps to be tested.</p>
</td></tr>
<tr><td><code id="pbat_+3A_mminphenos">mminphenos</code></td>
<td>
<p>Multi-marker multi-phenotype tests: the minimum number
of phenotypes to be tested.</p>
</td></tr>
<tr><td><code id="pbat_+3A_mmaxphenos">mmaxphenos</code></td>
<td>
<p>Multi-marker multi-phenotype tests: the maximum number
of phenotypes to be tested.</p>
</td></tr>
<tr><td><code id="pbat_+3A_env.cor.adjust">env.cor.adjust</code></td>
<td>
<p>Whether to adjust for environmental
correlation.</p>
</td></tr>
<tr><td><code id="pbat_+3A_gwa">gwa</code></td>
<td>
<p>Whether to use (g)enome (w)ide (a)cceleration mode.  This
is faster for genome-wide association tests, and has slightly less
output.</p>
</td></tr>
<tr><td><code id="pbat_+3A_snppedfile">snppedfile</code></td>
<td>
<p>Whether the pedigree file contains just snps. When
this is true, it employs a more optimal storage technique and uses
much less memory. It is especially advantageous for genome-wide
studies.</p>
</td></tr>
<tr><td><code id="pbat_+3A_extended.pedigree.snp.fix">extended.pedigree.snp.fix</code></td>
<td>
<p>Set to TRUE when you are using a
dataset with large extended pedigrees. This will not work with any
mode but &lsquo;single&rsquo; mode currently [see pbat.set(...)].
This is also sometimes necessary for multi-allelic markers (i.e. not
binary markers).
</p>
</td></tr>
<tr><td><code id="pbat_+3A_new.ped.algo">new.ped.algo</code></td>
<td>
<p>Set to TRUE (default is FALSE) to use the new, 10-100 times faster and more memory efficient algorithm. Somewhat experimental with extended pedigrees, so use with caution.</p>
</td></tr>
<tr><td><code id="pbat_+3A_cnv.intensity">cnv.intensity</code></td>
<td>
<p>The CNV intensity number that should be analyzed.</p>
</td></tr>
<tr><td><code id="pbat_+3A_cnv.intensity.num">cnv.intensity.num</code></td>
<td>
<p>The number of CNV intensities per CNV in the .cped file.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>IF YOU ARE HAVING PROBLEMS: Try setting extended.pedigree.snp.fix, the slowest but most robust method.
</p>
<p>INTERPRITING THE OUTPUT:
</p>
<p>1) I make every attempt to try to properly header the output, but sometimes this is not possible. You will often see a warning message to this regards, which is generally safe to ignore.
</p>
<p>2) 'a'=additive, 'd'=dominant, 'r'=recessive, 'h'=heterozygous advantage
</p>
<p>FURTHER USEFUL COMMENTS:
</p>
<p>These commands require &lsquo;pbatdata.txt&rsquo; to be in the working directory;
if not found, the program will attempt to (1) copy the file from the
directory where pbat is, (2) copy it from anywhere in the path, or
(3) error and exit.
</p>
<p>Linux warning: the file &lsquo;pbatdata.txt&rsquo; appears not to have shipped
with the current (as of writing this) linux version; to fix this just
download the windows version as well and copy the file from there to
the same directory as pbat.
</p>
<p>It is recommended to set 'LOAD.OUTPUT' to 'FALSE' when dealing with
very large numbers of SNPs.
</p>
<p>These commands will also generate a lot of output files in the current
working directory when interfacing with pbat.  These files will be
time-stamped so concurrent analysis in the same directory can be run.
<em>Race condition</em>: if two logrank analysis finish at <em>exactly the
same time</em>, then the plots for one might be lost and/or get linked
to the wrong analysis.  This should be a rather rare occurence, and is
an unpreventable result of pbat always sending this output to only one
filename.  Workaround to race condition: create another directory and
use that as your current working directory instead.
</p>
<p>Note that multi-marker / multi-phenotype mode is not supported in
parallel at this time, so if you are having problems try running the
command <code>pbat.setmode("single")</code>, or setting it to single from
the graphical interface before running these tests.
</p>
<p>WARNING: Note the 'extended.pedigree.snp.fix' option, which is
important for getting more accurate results in very extended
pedigrees.  It uses a slower but more accurate pedigree reconstruction
method.
</p>


<h3>Value</h3>

<p>&lsquo;pbat&rsquo;, &lsquo;pbat.last&rsquo;, &lsquo;pbat.m&rsquo;, &lsquo;pbat.obj&rsquo;, and &lsquo;pbat.files&rsquo;
return an object of class <code>pbat</code>.  Methods supported by  this include
<code>plot(...)</code>, <code>summary(...)</code>, and <code>print(...)</code>.  Follow
the first three links in the 'see also' section of this file for more
details.
</p>


<h3>References</h3>

<p>This was taken with only slight modification to accomodate the
interface from Christoph Lange's description of the commands for the
pbat program, (which was available with the software at the time of
this writing).
</p>
<p>P2BAT webpage.
</p>
<p>FBAT webpage (lists a lot of references
in relation to both of these programs).
</p>
<p>More pbat references:
</p>
<p>Hoffmann, T. and Lange, C. (2006) P2BAT: a massive parallel
implementation of PBAT for genome-wide association studies in
R. Bioinformatics. Dec 15;22(24):3103-5.
</p>
<p>Jiang, H., et al. (2006) Family-based association test for
time-to-onset data with time-dependent differences between the hazard
functions. Genet. Epidemiol, 30, 124-132.
</p>
<p>Laird, N.M. and Lange, C. (2006) Family-based designs in the age of
large-scale gene-association studies. Nat. Rev. Genet, 7.
</p>
<p>Lange, C., et al. (2003) Using the noninformative families in
family-based association tests: a powerful new testing
strategy. Am. J. Hum. Genet, 73, 801-811.
</p>
<p>Lange, C., et al. (2004a) A family-based association test for
repeatedly measured quantitative traits adjusting for unknown
environmental and/or polygenic effects. Stat. Appl. Genet. Mol. Biol,
3.
</p>
<p>Lange, C., et al. (2004b) Family-based association tests for survival
and times-to-onset analysis. Stat. Med, 23, 179-189.
</p>
<p>Van Steen, K., et al. (2005) Genomic screening and replication using
the same data set in family-based association testing. Nat. Genet, 37,
683-691.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.pbat">summary.pbat</a></code>,
<code><a href="#topic+plot.pbat">plot.pbat</a></code>,
<code><a href="#topic+print.pbat">print.pbat</a></code>,
</p>
<p><code><a href="#topic+as.ped">as.ped</a></code>,
<code><a href="#topic+as.pedlist">as.pedlist</a></code>,
<code><a href="#topic+read.ped">read.ped</a></code>
</p>
<p><code><a href="#topic+as.phe">as.phe</a></code>,
<code><a href="#topic+read.phe">read.phe</a></code>,
</p>
<p><code><a href="#topic+top">top</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##########################
## pbat.m(...) examples ##
##########################

## Not run: 

## Note, when you run the example (or anything else) you will generally
##  get a warning message that the column headers were guessed.
## This means they were guessed, and while I've tried to catch most
##  cases, the warning stands for ones I might have missed.

## These cannot be run verbatim, and are just meant to be examples.

##############################
## Further formula examples ##
##############################


# load in the data
# Here we assume that:
#  data.phe contains 'preds1', 'preds2', 'preds3', 'time',
#                     'censor', 'phenos1', ... 'phenos4'
#  data.ped contains 'snp1', 'snp2', 'snp3',
#                     'block1snp1','block1snp2',
#                     'block2snp1','block2snp2'
data.phe &lt;- read.phe( "data" )
data.ped &lt;- read.ped( "data" )

# This model does just the affection status (always given as
#  AffectionStatus) as the phenotype, no predictor covariates, and all
#  the snps for a snps analysis.
# Since affection status is dichotomous, we additionally set
#  distribution='categorical'
#  offset='none'
# NONE is a special keyword to indicate none, and can be only used in
#  this case (note that it is _case_ _sensative_);
#  otherwise one specifies values from the phenotype object, after and
#  including AffectionStatus.
res &lt;- pbat.m( AffectionStatus ~ NONE, phe, ped, fbat="gee",
               distribution='categorical', offset='none', ... )
summary( res )
res  # equivalent to print(res)

# basic model with one phenotype, does all snps (if none specified)
pbat.m( phenos1 ~ preds1, phe, ped, fbat="gee" )

# same model, but with more phenotypes; here we test them all at once
pbat.m( phenos1 + phenos2 + phenos3 ~ preds1, phe, ped, fbat="gee" )

# same model as just before, but now supposing that these phenotypes are
#  instead from a longitudinal study
pbat.m( phenos1 + phenos2 + phenos3 ~ preds1, phe, ped, fbat="pc" )

# like our second model, but the mi() tells it should be a marker
#  interaction
pbat.m( phenos1 ~ mi(preds1), phe, ped, fbat="gee" )

# logrank analysis - fbat need not be set
# uses more than one predictor variable
res &lt;- pbat.m( time &amp; censor ~ preds1 + preds2 + preds3, phe, ped )
plot( res )

# single snp analysis (because each snp is seperated by a vertical bar
#  '|'), and stratified by group (presence of censor auto-indicates
#  log-rank analysis).  Note that the group is at the end of the
#  expression, and _must_ be at the end of the expression
res &lt;- pbat.m( time &amp; censor ~ preds1^3 + preds2 | snp1 | snp2 |
         snp3 / group, temp )
plot( res )

# haplotype analysis, stratified by group
res &lt;- pbat.m( time &amp; censor ~ preds1^2 + preds2^3 | block1snp1
               + block1snp2 | block2snp1 + block2snp2 / group, temp )

# set any of the various options
res &lt;- pbat.m( phenos ~ preds, phe, ped, fbat="pc",
               null="linkage, no association", alpha=0.1 )

## New multimarker test (as described above)
# mmaxphenos and mmaxsnps are set to the minimum if not specified
res &lt;- pbat.m( phenos1 + phenos2 + phenos3 ~ preds | m1 | m2 | m3 | m4,
               phe, ped, fbat="pc", mminphenos=2, mminsnps=2 )

## And the top markers by conditional power
top( res )

## End(Not run)


############################
## pbat.obj(...) examples ##
############################

## Not run: 
# These will not function; they only serve as examples.

# ... just indicates there are various options to be put here!
res &lt;- pbat.obj("pedfile", snps=c("snp1,snp2"), preds="pred1", ... )
summary(res)
res

# plot is only available for "logrank"
res &lt;- pbat.obj(..., fbat="logrank")
plot( res )

## End(Not run)


##############################
## pbat.files(...) examples ##
##############################

## Not run: 
# These will not function, but only serve as examples.

# Note in the following example, both "pedfile.ped" and "pedfile.phe"
#  must exist.  If the names differed, then you must specify the
#  option 'phe="phefile.phe"', for example.
res &lt;- pbat.files( "pedfile", phenos=c("phenos1","phenos2"),
                   screening="conditional power" )
summary(res)
res

## End(Not run)

</code></pre>

<hr>
<h2 id='pbat.help'>Help and Reporting Information</h2><span id='topic+pbat.help'></span><span id='topic+pbat.firsttime'></span>

<h3>Description</h3>

<p>Provides some helpful suggestions and information if you are having
some troubles getting the package set up (often tends to be version
related). Information useful for reporting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbat.help(bug=FALSE, full=TRUE, ped=NULL, phe=NULL, lib.loc=NULL)
pbat.firsttime()
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pbat.help_+3A_bug">bug</code></td>
<td>
<p>Whether you think a bug was found on a run. Will attempt to
generate some useful information for the developers.</p>
</td></tr>
<tr><td><code id="pbat.help_+3A_full">full</code></td>
<td>
<p>When bug=TRUE, runs a full bug report (slower, reruns
analysis, but recommended.</p>
</td></tr>
<tr><td><code id="pbat.help_+3A_ped">ped</code></td>
<td>
<p>Pedigree object that can be obfuscated (see &lsquo;obfuscate&rsquo;).</p>
</td></tr>
<tr><td><code id="pbat.help_+3A_phe">phe</code></td>
<td>
<p>Phenotype object that can be obfuscated (see &lsquo;obfuscate&rsquo;).</p>
</td></tr>
<tr><td><code id="pbat.help_+3A_lib.loc">lib.loc</code></td>
<td>
<p>a character vector describing the location of R library
trees to search through, or 'NULL', i.e what/if you specified for
this when the library was loaded.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+pbat">pbat</a></code>
<code><a href="#topic+obfuscate">obfuscate</a></code>
</p>

<hr>
<h2 id='pbat.set'>Set Pbat Parameters</h2><span id='topic+pbat.set'></span><span id='topic+pbat.get'></span><span id='topic+pbat.setmode'></span><span id='topic+pbat.setmode.defaults'></span><span id='topic+pbat.getmode'></span><span id='topic+pbat.setVersionCheck'></span><span id='topic+pbat.setwine'></span><span id='topic+pbat.getwine'></span>

<h3>Description</h3>

<p>All of these values are retained, even after closing and restarting R.
</p>
<p><code>pbat.set</code> sets the name of the pbat executable.
</p>
<p><code>pbat.get</code> gets the current stored name of the pbat executable.
</p>
<p><code>pbat.setmode</code> sets the mode (single, multiple, or cluster), with
<code>pbat.setmode.defaults</code> returning it to default values.
</p>
<p><code>pbat.getmode</code> returns the current mode specifications.
</p>
<p><code>pbat.setVersionCheck</code> sets whether to check the version on startup.
</p>
<p><code>pbat.setwine</code> sets pbat up for use with wine. Intended for mac OS (darwine), or linux 32-bit version (temporarily with wine). Download the windows version of pbat, and use pbat.set to point it to pbat*.exe. See details below.
<code>pbat.getwine</code> returns the value, leave empty for windows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbat.set(executableStr="", CLEAR=FALSE)
pbat.get()

pbat.setmode.defaults(save=TRUE)
pbat.setmode(mode=NULL, jobs=NULL, clusterCommand=NULL,
             clusterRefresh=NULL )
pbat.getmode()

pbat.setVersionCheck(check=TRUE)

pbat.setwine(wineStr="", CLEAR=FALSE)
pbat.getwine()
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pbat.set_+3A_executablestr">executableStr</code></td>
<td>
<p>String of the pbat executable name,
e.g. &ldquo;c:/pbat/pbat25.exe&rdquo;.  If the string is omitted, then a
file-choose dialogue will appear to selet the file (suggested).</p>
</td></tr>
<tr><td><code id="pbat.set_+3A_clear">CLEAR</code></td>
<td>
<p>If set to TRUE, the executable name is cleared,
regardless of the string passed in the other option.</p>
</td></tr>
<tr><td><code id="pbat.set_+3A_save">save</code></td>
<td>
<p>Used internally.</p>
</td></tr>
<tr><td><code id="pbat.set_+3A_mode">mode</code></td>
<td>
<p>'single', 'multiple', or 'cluster'. See details.</p>
</td></tr>
<tr><td><code id="pbat.set_+3A_jobs">jobs</code></td>
<td>
<p>The number of jobs to partition into for 'multiple' or
'cluster'.</p>
</td></tr>
<tr><td><code id="pbat.set_+3A_clustercommand">clusterCommand</code></td>
<td>
<p>The command used to submit a job to your
cluster. The default is 'bsub'.</p>
</td></tr>
<tr><td><code id="pbat.set_+3A_clusterrefresh">clusterRefresh</code></td>
<td>
<p>How often (in seconds) to check if jobs have
finished.</p>
</td></tr>
<tr><td><code id="pbat.set_+3A_check">check</code></td>
<td>
<p>Whether to perform the version check on startup
(recommended).</p>
</td></tr>
<tr><td><code id="pbat.set_+3A_winestr">wineStr</code></td>
<td>
<p>Executable binary for wine. For linux 32-bit (64-bit is natively compiled), &quot;wine&quot; should suffice, assuming it is installed. For Mac, assuming you have installed Darwine (as detailed in the package), you should set it to be &quot;/Applications/Darwine/Wine.bundle/Contents/bin/wine&quot;, NOT the wineHelper.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default (upon installation) assumes that the executable is 'pbat'
and is in the path; <code>pbat.set</code> allows you to set the version you
wish to use, and easily toggle between other versions of the software
(in case you are testing the software it is easy to toggle to an older
version).
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
</p>
<p>The 'single' mode is meant for
only one processor on a users personal computer (see also
cluster for some special features of that mode). However, with
putting several cores on one chip, it may be more advantageous to use
'multiple' or 'cluster'.
</p>
<p>WARNING: if the number of jobs is 1, it will always be in this
'single' mode.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
</p>
<p>The 'multiple' mode is meant for a single multiprocessor system, or a
cluster that acts like one.  You can then set 'jobs' to be however
many processors that you have, or more depending on some of the newer
technologies.  If you have a cluster that acts like a
multi-processor system, then this can be more efficient than the 'cluster'
mode <em>if and only if you are waiting for the output</em>.  The
'multiple' mode will work in all operating systems.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
</p>
<p>In 'cluster' mode, you must specify the command and any additional
flags (to keep it generic for all platforms)
to submit a job that is represented in a file (i.e
'bsub -q normal sh' on some clusters, 'batch -f' or 'at now -f' in a very basic unix
environment - see 'man batch' from the command line or search for it
online).
</p>
<p>When the 'refresh' option is set to '0' (zero), then the
output is batched, and R will not wait for the output to be
finished.  If you do this, be sure to save your workspace when
exiting; then you can reload your output back in.  See
<code><a href="#topic+is.finished">is.finished</a></code>,
<code><a href="#topic+pbat.load">pbat.load</a></code>, and
<code><a href="#topic+pbat.concatenate">pbat.concatenate</a></code> for more information on this.
</p>
<p>When 'refresh' is set to an integer greater than zero, it indicates
the seconds that should elapse before the current R
session checks to see if the other processes are done.  The goal was
to keep this as generic as possible.
</p>
<p>Cluster mode may work in windows, but I am unfamiliar with any
batching command system available in windows.  Would there actually be
any interest in this?  Please e-mail me if this might be useful.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;
</p>
<p>System administrators could copy the '.pbat.Rmeta' file to the users
home directory after using <code>pbat.set</code> to set it themselves.
</p>


<h3>Value</h3>

<p><code>pbat.get</code> returns a string of the currently stored name of the
pbat executable.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pbat.m">pbat.m</a></code>
</p>
<p><code><a href="#topic+is.finished">is.finished</a></code>,
<code><a href="#topic+pbat.load">pbat.load</a></code>,
<code><a href="#topic+pbat.concatenate">pbat.concatenate</a></code>
</p>

<hr>
<h2 id='pbat.status'>PBAT Status of Run</h2><span id='topic+pbat.status'></span>

<h3>Description</h3>

<p>Provides status information, i.e. if there were any errors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbat.status(n=1,workFirst=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pbat.status_+3A_n">n</code></td>
<td>
<p>Returns the last &lsquo;n&rsquo; lines from the status file.
If n=0, all lines are returned.</p>
</td></tr>
<tr><td><code id="pbat.status_+3A_workfirst">workFirst</code></td>
<td>
<p>Whether to check for 
&lsquo;./pbatRwork/pbatstatus.txt&rsquo; or
&lsquo;./pbatstatus.txt&rsquo; first.
Generally only semi-important internally,
but set this to true if you just typed
<code>pbat.unwork()</code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+pbat">pbat</a></code>
</p>

<hr>
<h2 id='pbat.work'>pbatRwork Temp Directory</h2><span id='topic+pbat.work'></span><span id='topic+pbat.unwork'></span>

<h3>Description</h3>

<p>Creates a temporary 'pbatRwork' directory, and moves to it. This can be created in the same directory as a pedigree or phenotype file by passing those as parameters instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbat.work( pedOrPhe=NULL )
pbat.unwork( cur=NULL )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pbat.work_+3A_pedorphe">pedOrPhe</code></td>
<td>
<p>If a &lsquo;ped&rsquo; or &lsquo;phe&rsquo; symbolic object (read in without &lsquo;sym=TRUE&rsquo; which is the default), then uses that directory + '/pbatRwork' as the current working directory (so all output will go there).  Otherwise it just creates a &lsquo;pbatRwork&rsquo; directory in the current working directory.</p>
</td></tr>
<tr><td><code id="pbat.work_+3A_cur">cur</code></td>
<td>
<p>Pass in the output of pbat.work(...), see example below.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
ped &lt;- read.ped("mydata")
phe &lt;- read.phe("myphe")

cur &lt;- pbat.work( ped )

res &lt;- pbat.m(trait ~ NONE, ped=ped, phe=phe, fbat="gee")
...
## whatever you want to do with the results

pbat.unwork( cur ) ## sends you back to where you were

## End(Not run)
</code></pre>

<hr>
<h2 id='ped'>Pedigree Object</h2><span id='topic+as.ped'></span><span id='topic+as.pedlist'></span><span id='topic+is.ped'></span><span id='topic+is.pedlist'></span><span id='topic+read.ped'></span><span id='topic+fread.ped'></span><span id='topic+write.ped'></span><span id='topic+is.pped'></span><span id='topic+as.pped'></span><span id='topic+read.pped'></span><span id='topic+sort.ped'></span><span id='topic+plotPed'></span><span id='topic+pedMarkerNames'></span>

<h3>Description</h3>

<p>Creates, tests, reads, or writes objects of type <code>ped</code> or
<code>pedlist</code> to be used with the pbat commands.
</p>
<p>The <code>ped</code> class inherits the <code>data.frame</code> structure, and
is almost identical to the <code>data.frame</code> object described to
create it, only with some special reserved names.
</p>
<p>The <code>pedlist</code> class inherits the <code>list</code> structure, and is almost
identical to the <code>list</code> object described to create it, only with
some special reserved names.
</p>
<p>The &lsquo;pped&rsquo; functions provide support for a more compressed form of
input, that can be read in faster, and so may be faster when running
in clustered mode.
</p>
<p>COMPATIBILITY ISSUE 2024-01-27: The function <code>ped.markerNames</code> was renamed and is now <code>pedMarkerNames</code>. Failed new S3 check (which enforced that, as named, needed to be an S3 method).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.ped(x,
  pid="pid", id="id", idfath="idfath",
  idmoth="idmoth", sex="sex", affection="AffectionStatus",
  clearSym=FALSE)

as.pedlist(x,
  pid="pid", id="id", idfath="idfath",
  idmoth="idmoth", sex="sex", affection="AffectionStatus",
  clearSym=FALSE)

is.ped(obj, pure.ped=FALSE)

is.pedlist(obj)

read.ped(filename, format="ped", lowercase=TRUE, sym=TRUE, max=100, ...)
fread.ped(filename, ...)

write.ped(file, ped)

is.pped(obj)

read.pped(filename, max=100)

as.pped(ped, ppedname="")

## S3 method for class 'ped'
sort(x,decreasing=FALSE,...)

plotPed(ped, sink=NULL, geno="percent", phe=NULL, pheCols=NULL)

pedMarkerNames(ped)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ped_+3A_x">x</code></td>
<td>

<p>An object of class <code>ped</code>, <code>pedlist</code>, <code>data.frame</code>, or <code>list</code> as
described below.
</p>
<p>If <code>x</code> is of class <code>ped</code> or <code>pedlist</code>, no other options are used.
</p>
<p>When <code>x</code> is of class <code>data.frame</code>, the columns have entries
that match the string parameters <code>idped,...,AffectionStatus</code>;
genetic markers consist of <em>subsequent</em> columns
formated as follows: e.g. marker &lsquo;p5&rsquo; would need
two columns known as &lsquo;p5.a&rsquo; &lsquo;p5.b&rsquo; (or really
just where the characters a and b are different).
</p>
<p>When <code>x</code> is of class <code>list</code>, the
entries still include <code>idped,...,censor</code> as discussed
previously, and genetic markers consist of a list object of two
vectors for each unphased haplotype.
</p>
</td></tr>
<tr><td><code id="ped_+3A_pid">pid</code></td>
<td>
<p>String corresponding to column name for pedigree id.</p>
</td></tr>
<tr><td><code id="ped_+3A_id">id</code></td>
<td>
<p>String corresponding to column name for subject id.</p>
</td></tr>
<tr><td><code id="ped_+3A_idfath">idfath</code></td>
<td>
<p>String corresponding to column name for father id.</p>
</td></tr>
<tr><td><code id="ped_+3A_idmoth">idmoth</code></td>
<td>
<p>String corresponding to column name for mother id.</p>
</td></tr>
<tr><td><code id="ped_+3A_sex">sex</code></td>
<td>
<p>String corresponding to column name for sex.</p>
</td></tr>
<tr><td><code id="ped_+3A_affection">affection</code></td>
<td>
<p>String corresponding to column name for affection
status.</p>
</td></tr>
<tr><td><code id="ped_+3A_filename">filename</code></td>
<td>
<p>Filename to open; does not need .phe extension.</p>
</td></tr>
<tr><td><code id="ped_+3A_format">format</code></td>
<td>
<p>Toggles the return structure, set to &quot;ped&quot; or &quot;pedlist&quot;.</p>
</td></tr>
<tr><td><code id="ped_+3A_lowercase">lowercase</code></td>
<td>
<p>When TRUE (and sym is FALSE), enforces all headers to
lowercase for convenience.</p>
</td></tr>
<tr><td><code id="ped_+3A_...">...</code></td>
<td>
<p>Options for <code><a href="utils.html#topic+read.table">read.table</a></code>, used only when sym is
FALSE.  Do <em>not</em> put in <code>header=TRUE</code>, as this will
cause an error, as the header is automatically loaded.
</p>
<p>With the proper file formatting, this should not be used.
</p>
</td></tr>
<tr><td><code id="ped_+3A_file">file</code></td>
<td>
<p>string representing filename, or a connection for file output</p>
</td></tr>
<tr><td><code id="ped_+3A_ped">ped</code></td>
<td>
<p>an object of class <code>ped</code> or <code>pedlist</code> (see
<code>as.ped</code> or <code>as.pedlist</code>)</p>
</td></tr>
<tr><td><code id="ped_+3A_obj">obj</code></td>
<td>
<p>an object</p>
</td></tr>
<tr><td><code id="ped_+3A_sym">sym</code></td>
<td>
<p>When TRUE, only the header of the file is read in; only
PBAT will load in the file.  When FALSE, the entire file will be
read in, and can be modified before using with PBAT.</p>
</td></tr>
<tr><td><code id="ped_+3A_max">max</code></td>
<td>
<p>When sym is TRUE, the amount of headers to read in before
going pure symbolic (so that the SNP usage consistency will not be
assessed by pbatR, only by PBAT).</p>
</td></tr>
<tr><td><code id="ped_+3A_clearsym">clearSym</code></td>
<td>
<p>When TRUE, if a symbolic file is found, it will be
read in; otherwise, it will stay symbolic.</p>
</td></tr>
<tr><td><code id="ped_+3A_pure.ped">pure.ped</code></td>
<td>
<p>When FALSE, tests if an object is a &lsquo;ped&rsquo; or
&lsquo;pped&rsquo;. When TRUE, tests only if the object is a &lsquo;ped&rsquo;.</p>
</td></tr>
<tr><td><code id="ped_+3A_ppedname">ppedname</code></td>
<td>
<p>Name of the &lsquo;pped&rsquo; file.  If a symbolic ped, it
defaults to that name except with a pped extension; otherwise, it
defaults to &lsquo;pped.pped&rsquo;.</p>
</td></tr>
<tr><td><code id="ped_+3A_decreasing">decreasing</code></td>
<td>
<p>Whether to sort in decreasing/increasing order.</p>
</td></tr>
<tr><td><code id="ped_+3A_sink">sink</code></td>
<td>
<p>For &lsquo;plot.ped&rsquo;, this is the name of a pdf file to output
all of the plots to (there will be one plot per page).</p>
</td></tr>
<tr><td><code id="ped_+3A_geno">geno</code></td>
<td>
<p>For 'plot.ped'; &quot;none&quot; includes no genotype summary information, &quot;percent&quot; gives an overall summary of how much each individual is genotyped, i.e. non-missing (the default), and &quot;each&quot; gives all of the genotypes of each individual.</p>
</td></tr>
<tr><td><code id="ped_+3A_phe">phe</code></td>
<td>
<p>For 'plot.ped', a phenotype object for extra information.</p>
</td></tr>
<tr><td><code id="ped_+3A_phecols">pheCols</code></td>
<td>
<p>For 'plot.ped', names in the phenotype object of columns that should be displayed in the pedigree.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When reading in a file on disk using <code>read.ped</code>, a &lsquo;.ped&rsquo; file should
have the following format (taken from the PBAT web-page).
The first line of the PBAT pedigree file contains the names of the
markers.
Each subsequent line stands for one individual/subject, starting with
the pedigree id, followed by the individual/subject id, the id of the
father, the id of the mother, the individual's sex and affection
status. After this information, for each marker, both marker alleles
are listed. The order of the markers has to correspond to the order of
the marker names in the first line of the file. Missing values here
must be encoded with a &lsquo;0&rsquo;, unlike the phenotype file.
Examples of this type of file can be found on the PBAT webpage.
</p>
<p>The usage of <code>as.ped</code> and <code>as.pedlist</code> should also follow
the same missingness convention.
</p>
<p>&lsquo;plot.ped&rsquo; attempts to make use of the &lsquo;kinship&rsquo; package to draw the
pedigrees. In my personal experience, this package cannot handle all
pedigrees. My preferred alternative would be to use Madeline, which
makes beautiful pictures (amongst other things):
http://eyegene.ophthy.med.umich.edu/
</p>
<p><code>ped.markerNames</code> returns the names of the markers, without the '.a' and '.b' extension for markers (and not duplicated for markers).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.ped">read.ped</a></code>,
<code><a href="#topic+write.ped">write.ped</a></code>,
<code><a href="#topic+as.pedlist">as.pedlist</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A highly artificial example with not enough subjects to be run;
#  however, it demonstrates how to put data in it.
x &lt;- data.frame( pid       = c(1,1,1,1,1),
                 id        = c(1,2,3,4,5),
                 idfath    = c(4,4,4,0,0),
                 idmoth    = c(5,5,5,0,0),
                 sex       = c(1,2,1,1,2),
                 AffectionStatus = c(1,0,0,1,0),
                 m1.a      = c(1,1,1,1,1),
                 m1.b      = c(1,2,1,1,2),
                 m2.a      = c(4,4,4,4,4),
                 m2.b      = c(3,3,3,4,3) )
x
myPed &lt;- as.ped( x )          # Mark it with the class 'ped'
myPedlist &lt;- as.pedlist( x )  # Instead mark it with 'pedlist'
myPed
myPedlist

# an alternate example of creating
names( x )[1:6] &lt;- c( "mypedid", "subid", "fathid",
                      "mothid", "gender", "affection" );
x
myPed &lt;- as.ped( x, pid="mypedid", id="subid", idfath="fathid",
                 idmoth="mothid", sex="gender", affection="affection" )
myPed  # Note it's the same as before!

myPed &lt;- as.ped( myPedlist )       # Easy conversion back
myPedlist &lt;- as.pedlist( myPed )   #  and forth between formats.
</code></pre>

<hr>
<h2 id='phe'>Phenotype Object</h2><span id='topic+as.phe'></span><span id='topic+is.phe'></span><span id='topic+read.phe'></span><span id='topic+fread.phe'></span><span id='topic+write.phe'></span><span id='topic+sort.phe'></span>

<h3>Description</h3>

<p>Creates, tests, reads, or writes an object of class <code>phe</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.phe(df, pid="pid", id="id")

is.phe(obj)

read.phe(
  filename, na.strings=c("-",".","NA"), 
  lowercase=TRUE, sym=TRUE, ...)
fread.phe(filename, ...)

write.phe(file, phe)

## S3 method for class 'phe'
sort(x,decreasing=FALSE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phe_+3A_df">df</code></td>
<td>
<p>Dataframe with the data</p>
</td></tr>
<tr><td><code id="phe_+3A_pid">pid</code></td>
<td>
<p>String for the column header for 'pid' - pedigree ID.</p>
</td></tr>
<tr><td><code id="phe_+3A_id">id</code></td>
<td>
<p>String for the column header for 'id' - subject ID.</p>
</td></tr>
<tr><td><code id="phe_+3A_obj">obj</code></td>
<td>
<p>any object</p>
</td></tr>
<tr><td><code id="phe_+3A_filename">filename</code></td>
<td>
<p>Filename to open; does not need .phe extension.</p>
</td></tr>
<tr><td><code id="phe_+3A_na.strings">na.strings</code></td>
<td>
<p>Strings that represent NA; defaults should be fine
here.</p>
</td></tr>
<tr><td><code id="phe_+3A_lowercase">lowercase</code></td>
<td>
<p>When TRUE (default), enforces all headers to
lowercase for convenience.</p>
</td></tr>
<tr><td><code id="phe_+3A_sym">sym</code></td>
<td>
<p>When TRUE, only the header of the file is read in; only
PBAT will load in the file (* - see exception).  When FALSE, the
entire file will be
read in, and can be modified before using with PBAT.</p>
</td></tr>
<tr><td><code id="phe_+3A_...">...</code></td>
<td>
<p>Options for <code><a href="utils.html#topic+read.table">read.table</a></code>.  Do <em>not</em> put in
<code>header=TRUE</code>, as this will
cause an error, as the header is automatically loaded.
</p>
<p>With the proper file formatting, this should not be used.
</p>
</td></tr>
<tr><td><code id="phe_+3A_file">file</code></td>
<td>
<p>string representing filename, or a connection for file output</p>
</td></tr>
<tr><td><code id="phe_+3A_phe">phe</code></td>
<td>
<p>An object of class 'phe' (see <code><a href="#topic+as.phe">as.phe</a></code>).</p>
</td></tr>
<tr><td><code id="phe_+3A_x">x</code></td>
<td>
<p>An object of class 'phe' (see <code><a href="#topic+as.phe">as.phe</a></code>).</p>
</td></tr>
<tr><td><code id="phe_+3A_decreasing">decreasing</code></td>
<td>
<p>Whether to sort in decreasing/increasing order.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When reading in a file on disk using <code>read.ped</code>, a &lsquo;.phe&rsquo; file
should have the following format (taken from the PBAT web-page).
The first line contains
the names of the covariates and phenotypes and the subsequent lines
contain the pedigree id, the id of the subject, followed by the values
of the covariates and phenotypes for that subject.  Here missing
values must be indicate with a &lsquo;.&rsquo; or &lsquo;-&rsquo;, unlike the pedigree file.
Examples of this type of file can be found on the PBAT webpage.
</p>
<p>Once the dataset is read in, missing values are converted into the
usual R format of NA (see NA in the help files).
</p>
<p>When using <code>as.phe</code>, missing values should be in the native R
format.
The <code>write.ped</code> function will convert back into the missing
format necessary for PBAT.
</p>
<p>(*) Exception. If symbolic is true, the dataset will be temporarily
read in under one special circumstance.  This is when a stratification
variable is used in pbat.m, pbat.obj, or pbat.files.  This is because
the grouping values must be read in.  Alternatively, you can specify
these values when calling those functions (see the groups.* options)
in pbat.
</p>


<h3>Note</h3>

<p><b>&lsquo;read.phe&rsquo; mild warning</b>: This function <em>might change
the names of headers</em>, so they may not correspond to what pbat is
looking for if you call any of
the pbat*files() commands (<em>even</em> if lowercase=FALSE, as some
symbols might be modified by R naming
conventions).  In general this should not happen.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.ped">read.ped</a></code>,
<code><a href="#topic+write.ped">write.ped</a></code>,
<code><a href="#topic+as.ped">as.ped</a></code>,
<code><a href="#topic+as.pedlist">as.pedlist</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A highly artificial example just to get you used to the syntax
#  of using 'as.phe'.
x &lt;- data.frame( pid    = c(1,1,2,2,2),
                 id     = c(1,2,3,4,5),
                 age    = c(14,45,33,22,21),
                 weight = c(150,100,180,185,110) )
x
myPhe &lt;- as.phe( x );
myPhe

# And just another e.g. capitalizing on prior code
names(x)[1:2] &lt;- c("thepid","theid")
x
myPhe &lt;- as.phe( x, pid="thepid", id="theid" )
myPhe  # same as before!
</code></pre>

<hr>
<h2 id='power'>Power Exploration</h2><span id='topic+pbat.power'></span><span id='topic+pbat.powerCmd'></span>

<h3>Description</h3>

<p>Power has been completely rewritten from scratch, and
is all done via monte carlo simulation internally now.  These routines
do not require pbat, and should run on any machine.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pbat.power(mode="continuous")

pbat.powerCmd( 
  numOffspring=1, numParents=2, numFamilies=500,
  additionalOffspringPhenos=TRUE,
  ascertainment="affected",
  modelGen="additive", modelTest=modelGen,
  afreqMarker=NA,
  penAA=0.8, penAB=0.5, penBB=0.3,
  heritability=0.0, contsAscertainmentLower=0.0,
  contsAscertainmentUpper=1.0,
  pDiseaseAlleleGivenMarkerAllele=1.0, afreqDSL=0.1,
  alpha=0.01,
  offset="default",
  numSim=1000,
  ITERATION_KILLER=200 )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="power_+3A_mode">mode</code></td>
<td>
<p>&quot;continuous&quot; or &quot;dichotomous&quot;</p>
</td></tr>
<tr><td><code id="power_+3A_numoffspring">numOffspring</code></td>
<td>
<p>Family - number of offspring</p>
</td></tr>
<tr><td><code id="power_+3A_numparents">numParents</code></td>
<td>
<p>Family - number of parents (0,1,2)</p>
</td></tr>
<tr><td><code id="power_+3A_numfamilies">numFamilies</code></td>
<td>
<p>Family - number of families</p>
</td></tr>
<tr><td><code id="power_+3A_additionaloffspringphenos">additionalOffspringPhenos</code></td>
<td>
<p>Only used when you have missing parents;
additional offspring phenotypes. 1 for yes, 0 for no.</p>
</td></tr>
<tr><td><code id="power_+3A_ascertainment">ascertainment</code></td>
<td>
<p>'unaffected', 'affected', or 'na' for anyone</p>
</td></tr>
<tr><td><code id="power_+3A_modelgen">modelGen</code></td>
<td>
<p>The model used when generating the simulated data - one of 'additive', 'dominant', 'recessive'.</p>
</td></tr>
<tr><td><code id="power_+3A_modeltest">modelTest</code></td>
<td>
<p>The model used to test the simulated data.</p>
</td></tr>
<tr><td><code id="power_+3A_afreqmarker">afreqMarker</code></td>
<td>
<p>allele frequency at the marker</p>
</td></tr>
<tr><td><code id="power_+3A_penaa">penAA</code></td>
<td>
<p>penetrance of AA genotype</p>
</td></tr>
<tr><td><code id="power_+3A_penab">penAB</code></td>
<td>
<p>penetrance of AB genotype</p>
</td></tr>
<tr><td><code id="power_+3A_penbb">penBB</code></td>
<td>
<p>penetrance of BB genotype</p>
</td></tr>
<tr><td><code id="power_+3A_heritability">heritability</code></td>
<td>
<p>heritibility - when this is zero, a binary trait
according to the previously defined parameters is used; when it is
nonzero, a continuous trait is used.</p>
</td></tr>
<tr><td><code id="power_+3A_contsascertainmentlower">contsAscertainmentLower</code></td>
<td>
<p>Lower bound for affected ascertainment
with a continuous trait, this is a vector for each member after the
proband, defaulting to &lsquo;0&rsquo;. It represents the quantiles, so 0.05
would indicate that the lower 5 percent of the phenotypes should be
removed.</p>
</td></tr>
<tr><td><code id="power_+3A_contsascertainmentupper">contsAscertainmentUpper</code></td>
<td>
<p>Upper bound, defaults to &lsquo;1&rsquo;.</p>
</td></tr>
<tr><td><code id="power_+3A_pdiseaseallelegivenmarkerallele">pDiseaseAlleleGivenMarkerAllele</code></td>
<td>
<p>Pr(Disease allele A|marker allele A)</p>
</td></tr>
<tr><td><code id="power_+3A_afreqdsl">afreqDSL</code></td>
<td>
<p>allele frequency at DSL, defaults to marker frequency.</p>
</td></tr>
<tr><td><code id="power_+3A_alpha">alpha</code></td>
<td>
<p>significance level</p>
</td></tr>
<tr><td><code id="power_+3A_offset">offset</code></td>
<td>
<p>&quot;default&quot; uses the population prevalence for dichotomous
traits and the population mean for continuous traits. If a number is
specified, then that number is used as the offset.</p>
</td></tr>
<tr><td><code id="power_+3A_numsim">numSim</code></td>
<td>
<p>Number of monte-carlo simulations. I'd use a smaller
number while starting out with it, and then turn it up to a much
higher number of iterations later on.</p>
</td></tr>
<tr><td><code id="power_+3A_iteration_killer">ITERATION_KILLER</code></td>
<td>
<p>Controls how many times to try to draw data when simulating, before giving up. A value of 0 indicates to never stop. This is useful if you are playing around and are considering a situation that is too difficult for this program to be able to simulate.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>pbat.powerCmd(...) does not really do any range checking, primarily
because I don't expect most will use it directly, and instead will use
the friendly GUI interface for power exploration.
</p>
<p>Be careful with the number of simulations! When you are first
exploring, you can keep this low, but you should turn this all the way
up before doing your final computation.
</p>
<p>Note that some values of &lsquo;pDiseaseAlleleGivenMarkerAllele&rsquo; in
combination with &lsquo;afreqMarker&rsquo; are not possible. These will return
negative values (these are error codes for the GUI, which will provide
more helpful messages).
</p>
<p>Lastly, you might want to look into something like
set.seed(1)
e.g., if you want the results to be reproducable (set it to any
number, but make note of this number, see set.seed for more details).
</p>


<h3>References</h3>

<p>Hoffmann, T. and Lange, C. (2006) P2BAT: a massive parallel
implementation of PBAT for genome-wide association studies in
R. Bioinformatics. Dec 15;22(24):3103-5.
</p>
<p>Horvath, Steve, Xu, Xin, and Laird, Nan M. The family based association test method: computing means and variances for general statistics. Tech Report.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pbat">pbat</a></code>,
<code><a href="#topic+pbat.last">pbat.last</a></code>
</p>

<hr>
<h2 id='top'>top</h2><span id='topic+top'></span>

<h3>Description</h3>

<p>The top n markers by conditional power.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>top(pbatObj, n=10, sortBy=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="top_+3A_pbatobj">pbatObj</code></td>
<td>
<p>An object of class 'pbat', i.e. the results of
'pbat.m(...)'.</p>
</td></tr>
<tr><td><code id="top_+3A_n">n</code></td>
<td>
<p>How many markers to print. 0 means all will be returned.</p>
</td></tr>
<tr><td><code id="top_+3A_sortby">sortBy</code></td>
<td>
<p>Name of conditional power column to sort by, for
instance if you wanted to look at the GxE, you might specify
&quot;FBATGxE&quot; instead.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Prints out the top markers according to conditional power.
</p>

<hr>
<h2 id='write.pbat'>Saving &lsquo;pbat&rsquo; Object</h2><span id='topic+write.pbat'></span><span id='topic+write.pbat.csv'></span>

<h3>Description</h3>

<p>Writes a &lsquo;pbat&rsquo; object (the result of calling <code>pbat.m</code>) to disk.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.pbat(x, filename, resultsOnly=FALSE)
write.pbat.csv(x, filename, resultsOnly=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.pbat_+3A_x">x</code></td>
<td>
<p>Object of class <code>pbat</code>.</p>
</td></tr>
<tr><td><code id="write.pbat_+3A_filename">filename</code></td>
<td>
<p>Filename to save to.</p>
</td></tr>
<tr><td><code id="write.pbat_+3A_resultsonly">resultsOnly</code></td>
<td>
<p>When TRUE, this only prints the results to file,
and not how the function was called. Typically not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>With <code>write.pbat</code>, a text file is written out with an &lsquo;&amp;&rsquo; symbol
delimiting the output section. Both the calling and results section are
preserved. If the file has the &lsquo;.csv&rsquo; extension, then
<code>write.pbat.csv</code> is called.
</p>
<p>With <code>write.pbat.csv</code>, the text file is of the standard &lsquo;csv&rsquo;
format to delimit the text file. This should be readable with any
spreadsheet program. Note that to use the R output (for the logrank
plots), you should open the file in a spreadsheet program and just
copy and paste it into R.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
