<!DOCTYPE html><html><head><title>Help for package PBImisc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PBImisc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PBImisc-package'>
<p>Set of supplementary datasets and functions</p></a></li>
<li><a href='#AML'>
<p>Acute myeloid leukemia AML study</p></a></li>
<li><a href='#apartments'><p>Apartment prices in Warsaw in years 2007-2009</p></a></li>
<li><a href='#boxplotpp'><p>boxplot plus plus</p></a></li>
<li><a href='#corn'><p>A datasets relatead to gene expression in corn</p></a></li>
<li><a href='#dementia'><p>A set of datasets relatead to dementia</p></a></li>
<li><a href='#Drosophila'>
<p>Drosophila datasets and QTL mapping study</p></a></li>
<li><a href='#ecap'>
<p>Epidemiology of Allergic Disease in Poland</p></a></li>
<li><a href='#eden'><p>European day hospital evaluation</p></a></li>
<li><a href='#elastase'><p>Relation between graft function and elastase</p></a></li>
<li><a href='#endometriosis'><p>Endometriosis study</p></a></li>
<li><a href='#eunomia'><p>European Evaluation of Coercion in Psychiatry and Harmonisation of Best Clinical Practise</p></a></li>
<li><a href='#flu'>
<p>Numbers of flu occurences in the 10 years period in the Poland.</p></a></li>
<li><a href='#genomes'><p>724 bacterial genomes data</p></a></li>
<li><a href='#heights'>
<p>Husband and Wife heights</p></a></li>
<li><a href='#histpp'><p> hist plus plus</p></a></li>
<li><a href='#kidney'><p>Graft function after kidney transplantation</p></a></li>
<li><a href='#Likelihood displacements'>
<p>Log-likelihood displacements for single observation and single grouping variable</p></a></li>
<li><a href='#milk'><p>Milk yield data</p></a></li>
<li><a href='#milkgene'><p>Mutation in BTN3A1 gene and milk yield</p></a></li>
<li><a href='#musculus'><p>A dataset relatead to mice musculus growth which depends on diet and genetic structure</p></a></li>
<li><a href='#Plot results from post hoc testing'>
<p>A function for visual representation of pairwise testing (both for pairwise.t.test and pairwise.wilcox.test)</p></a></li>
<li><a href='#schizophrenia'><p>Genetic backgroud of schizophrenia</p></a></li>
<li><a href='#score.cardiovascular'>
<p>SCORE for Cardiovascular Risk</p></a></li>
<li><a href='#SejmSenat'>
<p>SejmSenat</p></a></li>
<li><a href='#vaccination'><p>Effective dose study</p></a></li>
<li><a href='#YXZ'>
<p>Artificial dataset which shows the differences between tests type I and III (sequential vs. marginal)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Set of Datasets Used in My Classes or in the Book 'Modele
Liniowe i Mieszane w R, Wraz z Przykladami w Analizie Danych'</td>
</tr>
<tr>
<td>Author:</td>
<td>Przemyslaw Biecek &lt;przemyslaw.biecek@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Przemyslaw Biecek &lt;przemyslaw.biecek@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A set of datasets and functions used in the book
  'Modele liniowe i mieszane w R, wraz z przykladami w analizie danych'.
  Datasets either come from real studies or are created to be as similar 
  as possible to real studies.</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://www.biecek.pl/R/">http://www.biecek.pl/R/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.8.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>lme4, Matrix</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggplot2, ca, lattice</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-02-25 21:00:34 UTC; pbiecek</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-02-25 22:46:59</td>
</tr>
</table>
<hr>
<h2 id='PBImisc-package'>
Set of supplementary datasets and functions
</h2><span id='topic+PBImisc-package'></span><span id='topic+PBImisc'></span>

<h3>Description</h3>

<p>A set of datasets and functions used in the book
,,Modele liniowe i mieszane w R, wraz z przykladami w analizie danych&rdquo;
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> PBImisc</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-02-15</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>General Description</h3>

<p>A set of datasets some of them are my original ones, some are taken from other packages of literature.
</p>


<h3>Author(s)</h3>

<p>Przemyslaw Biecek
</p>
<p>Maintainer: You should complain to Przemyslaw Biecek  &lt;przemyslaw.biecek@gmail.com&gt;
</p>


<h3>References</h3>

<p>Przemyslaw Biecek ,,Modele liniowe i mieszane w R, wraz z przykladami w analizie danych&rdquo; 2013, Wydawnictwo PWN
</p>


<h3>Examples</h3>

<pre><code class='language-R'># here you will find some examples
#
</code></pre>

<hr>
<h2 id='AML'>
Acute myeloid leukemia AML study
</h2><span id='topic+AML'></span>

<h3>Description</h3>

<p>This dataset bases on blood samples for patients with Acute myeloid leukemia.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(AML)</code></pre>


<h3>Format</h3>

<p>data.frame with 66 obs. and 5 variables
</p>

<dl>
<dt><code>Mutation</code></dt><dd><p>Factor w/ 4 levels CBFbeta, FLT3, None, Other</p>
</dd>
<dt><code>CD14.control</code></dt><dd><p>CD14 level in the control group</p>
</dd>
<dt><code>CD14.D3</code></dt><dd><p>CD14 level after D3 treatment</p>
</dd>
<dt><code>CD14.1906</code></dt><dd><p>CD14 level after D3 homolog 1906 treatment</p>
</dd>
<dt><code>CD14.2191</code></dt><dd><p>CD14 level after D3 homolog 2191 treatment</p>
</dd>
</dl>



<h3>Details</h3>

<p>Mutation - mutated gene that causes leucemia, one of following CBFbeta, FLT3, None, Other
CD14.control, CD14.D3, CD14.1906, CD14.2191 - effects in vitamin D3 or its homologues
</p>


<h3>Source</h3>

<p>Artificial dataset generated to be consistent with Ewa M. study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(lattice)
data(AML)
AML2 = reshape(AML, direction="long", varying=colnames(AML)[2:5])
bwplot(CD14~time|Mutation, AML2)
interaction.plot(AML2$time,AML2$Mutation, AML2$CD14)
</code></pre>

<hr>
<h2 id='apartments'>Apartment prices in Warsaw in years 2007-2009</h2><span id='topic+apartments'></span>

<h3>Description</h3>

<p>Dataset downloaded from website http://www.oferty.net/. Dataset contains offer and transictional prices for apartments sold in in Warsaw in years 2007-2009.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(apartments)</code></pre>


<h3>Format</h3>

<p>data.frame with 973 obs. and 16 variables
</p>

<dl>
<dt><code>year</code></dt><dd><p>data year of the transaction</p>
</dd>
<dt><code>month</code></dt><dd><p>data month of the transaction</p>
</dd>
<dt><code>surface</code></dt><dd><p>apartment area in m2</p>
</dd>
<dt><code>city</code></dt><dd><p>city (all transactions are from Warsaw)</p>
</dd>
<dt><code>district</code></dt><dd><p>district in which the apartment is located, factor with 28 levels</p>
</dd>
<dt><code>street</code></dt><dd><p>steet in which the apartment is located</p>
</dd>
<dt><code>n.rooms</code></dt><dd><p>number of rooms</p>
</dd>
<dt><code>floor</code></dt><dd><p>floor</p>
</dd>
<dt><code>construction.date</code></dt><dd><p>the construction year</p>
</dd>
<dt><code>type</code></dt><dd><p>ownership rights</p>
</dd>
<dt><code>offer.price</code></dt><dd><p>price in the offer</p>
</dd>
<dt><code>transaction.price</code></dt><dd><p>declared price in the transaction</p>
</dd>
<dt><code>m2.price</code></dt><dd><p>price per m2</p>
</dd>
<dt><code>condition</code></dt><dd><p>apartment condition, factor with 5 levels</p>
</dd>
<dt><code>lat</code>, <code>lon</code></dt><dd><p>latitude and longitude coordinates for district center</p>
</dd>
</dl>



<h3>Details</h3>

<p>This and other related dataset you may find here http://www.oferty.net/. 
</p>


<h3>Source</h3>

<p>website http://www.oferty.net/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(apartments)
library(lattice)
xyplot(m2.price~construction.date|district, apartments, type=c("g","p"))

# 
# apartments2 = na.omit(apartments[,c(13,1,3,5,7,8,9,10,14,15,16)])
# wsp = (bincombinations(10)==1)[-1,]
# params = matrix(0, nrow(wsp), 3)
# for (i in 1:nrow(wsp)) {
# 	  model = lm(m2.price~., data=apartments2[,c(TRUE,wsp[i,])])
#   	params[i,1] = AIC(model, k=log(nrow(apartments2)))
#  	  params[i,2] = model$rank
#  	  params[i,3] = summary(model)$adj.r.squared
# }
# plot(params[,2], params[,3], xlab="no. of regressors", ylab="adj R^2")
# 
</code></pre>

<hr>
<h2 id='boxplotpp'>boxplot plus plus </h2><span id='topic+boxplotpp'></span><span id='topic+boxplotInTime'></span>

<h3>Description</h3>

<p>boxplotpp
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boxplotpp(x, xname=seq(1:ncol(x)), utitle="", addLines=TRUE, 
           color = ifelse(addLines, "white","lightgrey"), ...) 

boxplotInTime(x, xname, additional=T, color = ifelse(additional, 
     "white","lightgrey"), main="", ylim=range(unlist(x),na.rm=T), ..., 
     points = dim(x)[2], at = 1:points)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="boxplotpp_+3A_x">x</code></td>
<td>
<p>  TODO </p>
</td></tr>
<tr><td><code id="boxplotpp_+3A_xname">xname</code></td>
<td>
<p> TODO </p>
</td></tr>
<tr><td><code id="boxplotpp_+3A_utitle">utitle</code></td>
<td>
<p> TODO </p>
</td></tr>
<tr><td><code id="boxplotpp_+3A_addlines">addLines</code></td>
<td>
<p> TODO </p>
</td></tr>
<tr><td><code id="boxplotpp_+3A_color">color</code></td>
<td>
<p> TODO </p>
</td></tr>
<tr><td><code id="boxplotpp_+3A_additional">additional</code></td>
<td>
<p> TODO </p>
</td></tr>
<tr><td><code id="boxplotpp_+3A_main">main</code></td>
<td>
<p> TODO </p>
</td></tr>
<tr><td><code id="boxplotpp_+3A_points">points</code></td>
<td>
<p> TODO </p>
</td></tr>
<tr><td><code id="boxplotpp_+3A_at">at</code></td>
<td>
<p> TODO </p>
</td></tr>
<tr><td><code id="boxplotpp_+3A_ylim">ylim</code></td>
<td>
<p> TODO </p>
</td></tr>
<tr><td><code id="boxplotpp_+3A_...">...</code></td>
<td>
<p> TODO </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

<p> Przemyslaw Biecek</p>


<h3>Examples</h3>

<pre><code class='language-R'>#TODO
</code></pre>

<hr>
<h2 id='corn'>A datasets relatead to gene expression in corn</h2><span id='topic+corn'></span>

<h3>Description</h3>

<p>Dataset from the book ,,Modele liniowe i mieszane w R, wraz z przykladami w analizie danych&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(corn)</code></pre>


<h3>Format</h3>

<p>data.frame with 5339 obs. and 36 variables
</p>

<p>A dataset with expression of 5339 genes. Each column corresponds to a single experiment. Column name codes the setup of experiment. For example DH.C.1 is related to line DH in the condition C and it is a first technical replicate of this set of conditions.
</p>
<p>Note that a noise injection was added to this data, in order to obtain the original dataset please contact with the package maintainer.

</p>


<h3>Details</h3>

<p>Dataset from the book ,,Modele liniowe i mieszane w R, wraz z przykladami w analizie danych&rdquo;.
</p>
<p>Used as an example of modeling of data from expression microarrays with the use of models with mixed effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

require(lme4)

names &lt;- colnames(corn)
X &lt;- t(matrix(unlist(strsplit(names, ".", fixed=T)), 3, 36))
X &lt;- data.frame(X)
colnames(X) &lt;- c("spec", "temp", "plant")

summary(X)

y &lt;- corn[4662,]
lmer(y~spec*temp + (1|plant:spec:temp), data=X)


## End(Not run)
</code></pre>

<hr>
<h2 id='dementia'>A set of datasets relatead to dementia</h2><span id='topic+dementia'></span>

<h3>Description</h3>

<p>Dataset from the book ,,Modele liniowe i mieszane w R, wraz z przykladami w analizie danych&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dementia)</code></pre>


<h3>Format</h3>

<p>data.frame with 1000 obs. and 4 variables
</p>

<dl>
<dt><code>demscore</code></dt><dd><p>score of dementia</p>
</dd>
<dt><code>age</code></dt><dd><p>age, a factor with two levels</p>
</dd>
<dt><code>sex</code></dt><dd><p>sex, a factor with two levels</p>
</dd>
<dt><code>study</code></dt><dd><p>a source of data, a factor with 10 levels</p>
</dd>
</dl>



<h3>Details</h3>

<p>Dataset from the book ,,Modele liniowe i mieszane w R, wraz z przykladami w analizie danych&rdquo;.
</p>
<p>Used as an example of mixed modeling in meta analysis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  require(lme4)
  modelFullI &lt;- lmer(demscore~age*sex+(age*sex|study), data=dementia,
                   REML=FALSE)
  summary(modelFullI)

## End(Not run)
</code></pre>

<hr>
<h2 id='Drosophila'>
Drosophila datasets and QTL mapping study
</h2><span id='topic+Drosophila'></span>

<h3>Description</h3>

<p>Two datasets with genotypes and phenotypes for backcrossed Drosophilas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Drosophila)</code></pre>


<h3>Format</h3>

<p>Two datasets with genotypes and phenotypes for backcrossed Drosophilas.
</p>
<p>The set of 41 markers describes genotypes while 5 variables describe phenotypes. See references for more details.
</p>

<dl>
<dt><code>bm</code></dt><dd><p>A data.frame with 370 obs. and 46 variables, first 41 are genotypes of gene markers, last five describes genotypes</p>
</dd>
<dt><code>bs</code></dt><dd><p>A data.frame with 402 obs. and 46 variables, first 41 are genotypes of gene markers, last five describes genotypes</p>
</dd>
<dt><code>chr</code></dt><dd><p>Factor w/ 4 levels CBFbeta, FLT3, None, Other</p>
</dd>
<dt><code>pos</code></dt><dd><p>Markers position on chromosom in centimorgnas</p>
</dd>
</dl>



<h3>Details</h3>

<p>The phonotype pc1 is nicely described by genotype in both backcrossed datasets.
</p>


<h3>Source</h3>

<p>Genetic Architecture of a Morphological Shape Difference Between Two Drosophila Species
Zhao-Bang Zenga, Jianjun Liu, Lynn F. Stamb, Chen-Hung Kao, John M. Mercer, Cathy C. Laurie
Genetics, Vol. 154, 299-310, January 2000
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Drosophila)
library(lattice)
# calculate log likelihoods
pval1 = numeric(41)
for (i in 1:41) {
  y = Drosophila$bm$pc1
  x = factor(Drosophila$bm[,i])
  pval1[i] = logLik(lm(y~x))
}
# loglikelihood plot
xyplot(pval1~pos|chr, data=Drosophila, type=c("p","l"), 
     pch=19, ylab="log likelihood")
</code></pre>

<hr>
<h2 id='ecap'>
Epidemiology of Allergic Disease in Poland 
</h2><span id='topic+ecap'></span>

<h3>Description</h3>

<p>This dataset touch one particular aspect from ECAP dataset. The original dataset is much more richer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ecap)</code></pre>


<h3>Format</h3>

<p>data.frame with 2102 obs. and 9 variables
</p>

<dl>
<dt><code>city</code>, <code>district</code></dt><dd><p>City and district, city is a factor with nine levels, the district effect is nested in the city effect</p>
</dd>
<dt><code>sex</code></dt><dd><p>Sex</p>
</dd>
<dt><code>weight</code>, <code>height</code></dt><dd><p>Weight and height</p>
</dd>
<dt><code>house.surface</code></dt><dd><p>Surface of house in which the pearson live</p>
</dd>
<dt><code>PNIF</code></dt><dd><p>Peak Nasal Inspiratory Flow</p>
</dd>
<dt><code>age</code></dt><dd><p>Age of the pearson</p>
</dd>
<dt><code>allergenes</code></dt><dd><p>Number of allergens</p>
</dd>
</dl>



<h3>Details</h3>

<p>PNIF stands for Peak Nasal Inspiratory Flow
</p>


<h3>Source</h3>

<p>Artificial dataset generated to be consistent with ECAP (Epidemiologia Chorob Alergicznych w Polsce) study
http://www.ecap.pl/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ecap)
library(lattice)
xyplot(PNIF~age|city, data=ecap, type=c("p","g","smooth"))
</code></pre>

<hr>
<h2 id='eden'>European day hospital evaluation</h2><span id='topic+eden'></span>

<h3>Description</h3>

<p>This dataset bases on origical study of European day hospital evaluation
</p>
<p>Artificial dataset (subset from real dataset with some random modifications). Do not use it for derivation of real conclusions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eden)</code></pre>


<h3>Format</h3>

<p>data.frame with 642 obs. and 12 variables
</p>

<dl>
<dt><code>mdid</code></dt><dd><p>Medical doctor id, there are 24 different MDs which examine patients</p>
</dd>
<dt><code>center</code></dt><dd><p>City in which the examination takes place</p>
</dd>
<dt><code>BPRS.Maniac</code>, <code>BPRS.Negative</code>, <code>BPRS.Positive</code>, <code>BPRS.Depression</code></dt><dd><p>BPRS stands for Brief Psychiatric Rating Scale, scores are averaged in four subscales</p>
</dd>
<dt><code>BPRS.Average</code></dt><dd><p>Average from 24 questions</p>
</dd>
<dt><code>MANSA</code></dt><dd><p>Scale which measures Quality of Life (Manchester Short Assessment of Quality of Life)</p>
</dd>
<dt><code>sex</code></dt><dd><p>Sex</p>
</dd>
<dt><code>children</code></dt><dd><p>Number of childs</p>
</dd>
<dt><code>years.of.education</code></dt><dd><p>Number of years of education</p>
</dd>
<dt><code>day</code></dt><dd><p>Hospitalization mode, day or stationary</p>
</dd>
</dl>



<h3>Details</h3>

<p>This dataset touch one particular aspect from EDEN dataset. The original dataset is much more richer.
</p>


<h3>Source</h3>

<p>Artificial dataset generated to be consistent with Joanna R. study.
</p>
<p>Bases on European day hospital evaluation, http://www.edenstudy.com/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(eden)
library(lattice)
xyplot(BPRS.Average~MANSA|center, data=eden, type=c("p","g","smooth"))
</code></pre>

<hr>
<h2 id='elastase'>Relation between graft function and elastase</h2><span id='topic+elastase'></span>

<h3>Description</h3>

<p>Relation between graft function and elastase from nephrology study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(elastase)</code></pre>


<h3>Format</h3>

<p>data.frame with 54 obs. and 5 variables
</p>

<dl>
<dt><code>sex</code>, <code>age</code>, <code>weight</code></dt><dd><p>Patient's sex, age and weight</p>
</dd>
<dt><code>elastase</code></dt><dd><p>Elastase concentration</p>
</dd>
<dt><code>GFR</code></dt><dd><p>Patient's GFR (glomerular filtration rate)</p>
</dd>
</dl>



<h3>Details</h3>

<p>Artificial dataset (real one with some random modifications). Do not use it for medical reasoning.
</p>


<h3>Source</h3>

<p>Artificial dataset generated to be consistent with Malgorzata L. study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(elastase)
library(lattice)
xyplot(GFR~elastase, data=elastase, type=c("p","r","g"))
</code></pre>

<hr>
<h2 id='endometriosis'>Endometriosis study</h2><span id='topic+endometriosis'></span>

<h3>Description</h3>

<p>How the endometriosis affects concetration of alpha and beta factors in the blood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(endometriosis)</code></pre>


<h3>Format</h3>

<p>data.frame with 165 obs. and 4 variables
</p>

<dl>
<dt><code>disease</code></dt><dd><p>disease, blood samples were taken from women with endometriosis of from healthy ones</p>
</dd>
<dt><code>phase</code></dt><dd><p>phase in the menstrual cycle as the examination day (proliferative or secretory)</p>
</dd>
<dt><code>alpha.factor</code>, <code>beta.factor</code></dt><dd><p>concentration of alpha and beta factors in blood</p>
</dd>
</dl>



<h3>Details</h3>

<p>Dataset used as example of ANCOVA 
</p>


<h3>Source</h3>

<p>Artificial dataset generated to be consistent with Ula S. study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(endometriosis)
library(lattice)
xyplot(log(alpha.factor)~log(beta.factor)|disease*phase, 
            data=endometriosis, type=c("p", "r"))
summary(aov(alpha.factor~beta.factor*disease*phase, data=endometriosis))
</code></pre>

<hr>
<h2 id='eunomia'>European Evaluation of Coercion in Psychiatry and Harmonisation of Best Clinical Practise</h2><span id='topic+eunomia'></span>

<h3>Description</h3>

<p>This dataset touch one particular aspect from EUNOMIA dataset. The original dataset is much more richer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eunomia)</code></pre>


<h3>Format</h3>

<p>data.frame with 2008 obs. and 15 variables
</p>

<dl>
<dt><code>CENTRE13</code></dt><dd><p>Center in which the patient is hospitalized, factor with 13 levels</p>
</dd>
<dt><code>SUBJECT</code></dt><dd><p>Patients ID</p>
</dd>
<dt><code>GENDER</code>, <code>AGE</code>, <code>NUM.HOSP</code></dt><dd><p>Gender, age and number of hospitalizations of given patient</p>
</dd>
<dt><code>CAT.T1</code>, <code>CAT.T2</code>, <code>CAT.T3</code></dt><dd><p>Clients Scale for Assessment of Treatment, short assessment, which measures the impact of COPD on a patients life, measured in times: T1, T2 and T3</p>
</dd>
<dt><code>BPRS.T1</code>, <code>BPRS.T2</code>, <code>BPRS.T3</code></dt><dd><p>Average score for Brief Psychiatric Rating Scale, measured in times: T1, T2 and T3</p>
</dd>
<dt><code>MANSA.T1</code>, <code>MANSA.T2</code>, <code>MANSA.T3</code></dt><dd><p>Scale which measures Quality of Life (Manchester Short Assessment of Quality of Life), measured in times: T1, T2 and T3</p>
</dd>
<dt><code>ICD10</code></dt><dd><p>International Statistical Classification of Diseases and Related Health Problems 10th Revision (ICD-10)</p>
</dd>
</dl>



<h3>Details</h3>

<p>Artificial dataset generated to be consistent with Eunomia study (European Evaluation of Coercion in Psychiatry and Harmonisation of Best Clinical Practise) 
</p>


<h3>Source</h3>

<p>Artificial dataset generated to be consistent with Joanna R. study.
</p>
<p>Eunomia dataset, http://www.eunomia-study.net/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(eunomia)
library(lattice)
bwplot(CENTRE13~BPRS.T1, data=eunomia)
xyplot(BPRS.T1~MANSA.T1|CENTRE13, data=eunomia, type=c("p","g","smooth"))
</code></pre>

<hr>
<h2 id='flu'>
Numbers of flu occurences in the 10 years period in the Poland.
</h2><span id='topic+flu'></span>

<h3>Description</h3>

<p>Data from National Institute of Hygiene reports.
Each row correspond to one record from NIH institute. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(flu)</code></pre>


<h3>Format</h3>

<p>data.frame with 6384 obs. and 11 variables
</p>

<dl>
<dt><code>region</code></dt><dd><p>Region for which given report was taken. A factor with 16 levels</p>
</dd>
<dt><code>inception.no</code></dt><dd><p>Number of flu occurences in given region for given report period (one or two weeks)</p>
</dd>
<dt><code>inception.no</code></dt><dd><p>Number of flu occurences in given region for given report period (one or two weeks)</p>
</dd>
<dt><code>inception.rate</code></dt><dd><p>Number of flu occurences normalized to 100k people</p>
</dd>
<dt><code>inception.no.0-14</code>, <code>inception.no.15+</code>, <code>inception.rate.0-14</code>, <code>inception.rate.15+</code></dt><dd><p>Absolute and normalized numbers of flu occurences calculated for age group 0-14 or 15+</p>
</dd>
<dt><code>date</code></dt><dd><p>Date of given report</p>
</dd>
<dt><code>date.id</code></dt><dd><p>Report id, there is 38 reports per year</p>
</dd>
<dt><code>latitude</code>, <code>longitude</code></dt><dd><p>Geographical coordinates for region</p>
</dd>
</dl>



<h3>Details</h3>

<p>Dataset used during the third edition of WZUR conference, see http://www.biecek.pl/WZUR3/wzurDane.html for more information.
</p>


<h3>Source</h3>

<p>Reports from National Institute of Public Health - National Institute of Hygiene, see: http://www.pzh.gov.pl
</p>
<p>More information: http://www.biecek.pl/WZUR3/wzurDane.html
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(flu)
library(ggplot2)
subflu = flu[flu$region=="Mazowieckie", ]
# linear scale
qplot(date, inception.rate,data=subflu, geom="line")+scale_y_sqrt() +theme_bw()

# polar coordinates
qplot(1 + date.id*12/38, inception.rate,data=subflu, geom="path", xlab="month")+
        scale_y_sqrt()+geom_smooth(span=0.1,se=FALSE, size=2,col="red") + 
        coord_polar() +theme_bw()
</code></pre>

<hr>
<h2 id='genomes'>724 bacterial genomes data</h2><span id='topic+genomes'></span>

<h3>Description</h3>

<p>Few parameters gathered for 724 bacterial species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(genomes)</code></pre>


<h3>Format</h3>

<p>data.frame with 724 obs. and 7 variables
</p>

<dl>
<dt><code>organism</code></dt><dd><p>Organism name, unique value for every row</p>
</dd>
<dt><code>group</code></dt><dd><p>Group, a factor with 22 levels</p>
</dd>
<dt><code>size</code></dt><dd><p>Genome size in Mbp</p>
</dd>
<dt><code>CG</code></dt><dd><p>GC content for genome sequence</p>
</dd>
<dt><code>habitat</code>, <code>temp.group</code>, <code>temperature</code></dt><dd><p>Where does this bacteria live?</p>
</dd>
</dl>



<h3>Details</h3>

<p>This dataset is prepared by Pawel M., data are taken from NCBI repository.
</p>
<p>See http://www.ncbi.nlm.nih.gov/ for more details
</p>


<h3>Source</h3>

<p>Pawel M. study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genomes)
library(ggplot2)
# is this relation linear ?
qplot(size,GC, data=genomes) + theme_bw()
# or linear in log scales?
qplot(size,GC, data=genomes, log="xy") + theme_bw()
</code></pre>

<hr>
<h2 id='heights'>
Husband and Wife heights
</h2><span id='topic+heights'></span>

<h3>Description</h3>

<p>A dataset from ,,A modern approach to regression with R&rdquo;. Simon J. Sheather 2009 .
Paired heights for husbands and wifes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(heights)</code></pre>


<h3>Format</h3>

<p>data.frame with 96 obs. and 2 variables
</p>

<dl>
<dt><code>Husband</code>, <code>Wife</code></dt><dd><p>Height of husband and wife.</p>
</dd>
</dl>



<h3>Details</h3>

<p>The dataset from ,,A modern approach to regression with R&rdquo;. Simon J. Sheather 2009 
</p>


<h3>Source</h3>

<p>A modern approach to regression with R. Simon J. Sheather 2009 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(heights)
plot(Husband~Wife, data=heights, pch=19)
abline(lm(Husband~Wife, data=heights), col="red")
abline(lm(Husband~Wife-1, data=heights), col="blue")
</code></pre>

<hr>
<h2 id='histpp'> hist plus plus </h2><span id='topic+histpp'></span>

<h3>Description</h3>

<p>histpp
</p>


<h3>Usage</h3>

<pre><code class='language-R'>histpp(x, xname="", utitle="")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="histpp_+3A_x">x</code></td>
<td>
<p> TODO  </p>
</td></tr>
<tr><td><code id="histpp_+3A_xname">xname</code></td>
<td>
<p> TODO </p>
</td></tr>
<tr><td><code id="histpp_+3A_utitle">utitle</code></td>
<td>
<p> TODO </p>
</td></tr>
</table>


<h3>Details</h3>

<p>TODO
</p>


<h3>Value</h3>

<p>TODO
</p>


<h3>Author(s)</h3>

<p> Przemyslaw Biecek</p>


<h3>References</h3>

 
<p>TODO
</p>


<h3>Examples</h3>

<pre><code class='language-R'># TODO
</code></pre>

<hr>
<h2 id='kidney'>Graft function after kidney transplantation</h2><span id='topic+kidney'></span>

<h3>Description</h3>

<p>Artificial dataset (subset from real dataset with some random modifications)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(kidney)</code></pre>


<h3>Format</h3>

<p>data.frame with 334 obs. and 16 variables
</p>

<dl>
<dt><code>recipient.age</code>, <code>donor.age</code></dt><dd><p>Age od donor and recipient</p>
</dd>
<dt><code>CIT</code></dt><dd><p>Cold ischemia time</p>
</dd>
<dt><code>discrepancy.AB</code>, <code>discrepancy.DR</code></dt><dd><p>discrepancies in AB and DR antibodies</p>
</dd>
<dt><code>therapy</code></dt><dd><p>scheme of immunosuppression</p>
</dd>
<dt><code>diabetes</code></dt><dd><p>diabetes</p>
</dd>
<dt><code>bpl.drugs</code></dt><dd><p>number of drugs for blood pressure lowering</p>
</dd>
<dt><code>MDRD7</code>, <code>MDRD30</code>, <code>MDRD3</code>, <code>MDRD6</code>, <code>MDRD12</code>, <code>MDRD24</code>, <code>MDRD36</code>, <code>MDRD60</code></dt><dd><p> MDRD (Modification of Diet in Renal Disease) as a estiamtor of glomerular filtration rate (GFR) from serum creatinine, measured 7, 30 days and 3, 6, 12, 24, 36 and 60 months after kidney transplantation</p>
</dd>
</dl>



<h3>Details</h3>

<p>Example of longitudinal study, note that graft for all patients survives 5 years after kidney transplantation.
</p>


<h3>Source</h3>

<p>Artificial dataset generated to be consistent with Maria M. study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(kidney)
boxplotInTime(kidney[,9:16], colnames(kidney[,9:16]), additional=TRUE)
</code></pre>

<hr>
<h2 id='Likelihood+20displacements'>
Log-likelihood displacements for single observation and single grouping variable
</h2><span id='topic+recalculateLogLik'></span><span id='topic+obsDisp'></span><span id='topic+groupDisp'></span>

<h3>Description</h3>

<p>Functions for log-likelihood displacements for each observation or each level of given factor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recalculateLogLik(model, fixef = fixef(model), vcor = VarCorr(model)) 

groupDisp(formula, data, var) 

obsDisp(formula, data, inds=1:nrow(data)) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Likelihood+2B20displacements_+3A_model">model</code></td>
<td>
<p>a mixed model of the class mer,</p>
</td></tr>
<tr><td><code id="Likelihood+2B20displacements_+3A_fixef">fixef</code>, <code id="Likelihood+2B20displacements_+3A_vcor">vcor</code></td>
<td>
<p>model parameters log-likelihood evaluation, if not provided then the estimates extracted from the 'model' parameter will be used</p>
</td></tr>
<tr><td><code id="Likelihood+2B20displacements_+3A_formula">formula</code></td>
<td>
<p>a model formula that will be passes to the nlme function</p>
</td></tr>
<tr><td><code id="Likelihood+2B20displacements_+3A_data">data</code></td>
<td>
<p>a data frame</p>
</td></tr>
<tr><td><code id="Likelihood+2B20displacements_+3A_var">var</code></td>
<td>
<p>a name of grouping variable (factor) for which the group log-likelihood displacement will be performed</p>
</td></tr>
<tr><td><code id="Likelihood+2B20displacements_+3A_inds">inds</code></td>
<td>
<p>indexes of observations for which observation log-likelihood displacement will be performed</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Likelihood displacement is defined as a difference of likelihoods calculated on full dataset for two models with different sets of parameters.  The first model is a model with ML estimates obtained for full dataset, while the second model is a model with ML estimates obtained on dataset without a selected observation or group of observations.
</p>
<p>Likelihood displacements are used in model diagnostic.
</p>
<p>Note that these functions reestimate coefficients in a set of model may be a time consuming.
</p>
<p>The function recalculateLogLik() calculated a log-likelihood for model defined by the object model and model parameters defined in following function arguments.
</p>
<p>The functions groupDisp() and obsDisp() calculates how the log-likelihood will decrees if selected groups or selected observations will not be used for parameter estimates. Note that log-likelihood is calculated on full dataset.
</p>


<h3>Author(s)</h3>

<p>Przemyslaw Biecek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(eunomia)
require(lme4)
set.seed(1313)
eunomias &lt;- eunomia[sample(1:2000,100),]
groupDisp(formula = BPRS.T2~ (1|CENTRE13), data=eunomias, var="CENTRE13")
 
obsDisp(formula = BPRS.T2~ (1|CENTRE13), data=eunomias, inds = 1:10)
 
obsDisp(formula = BPRS.T2~ (1|CENTRE13), data=eunomias)
 
</code></pre>

<hr>
<h2 id='milk'>Milk yield data</h2><span id='topic+milk'></span>

<h3>Description</h3>

<p>Milk yield data for 10 unrelated cows
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(milk)</code></pre>


<h3>Format</h3>

<p>data.frame with 40 obs. and 2 variables
</p>

<dl>
<dt><code>cow</code></dt><dd><p>cow id, a factor with 10 levels</p>
</dd>
<dt><code>milk.amount</code></dt><dd><p>milk amount in kgs per week</p>
</dd>
</dl>



<h3>Details</h3>

<p>Weekly milk yield amount for 10 cows. For every cow 5 measurements are taken.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(milk)
library(lattice)
# change the order of levels
milk$cow = reorder(milk$cow, milk$milk.amount, mean)
#plot it
dotplot(cow~milk.amount, data=milk)
</code></pre>

<hr>
<h2 id='milkgene'>Mutation in BTN3A1 gene and milk yield</h2><span id='topic+milkgene'></span>

<h3>Description</h3>

<p>It is known that BTN3A1 (Butyrophilin subfamily 3 member A1) has a crucial function in the secretion of lipids into milk. 
Doeas the SNP mutation in it change the average milk yield?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(milkgene)</code></pre>


<h3>Format</h3>

<p>data.frame with 1000 obs. and 5 variables
</p>

<dl>
<dt><code>cow.id</code></dt><dd><p>cow id, there is 465 cows in this study</p>
</dd>
<dt><code>btn3a1</code></dt><dd><p>btn3a1 genotype, a factor with two levels</p>
</dd>
<dt><code>lactation</code></dt><dd><p>for some cows there are milk yileds for four lactations for other only for the first one</p>
</dd>
<dt><code>milk</code>, <code>fat</code></dt><dd><p>milk and fat amount in kgs per lactation</p>
</dd>
</dl>



<h3>Details</h3>

<p>Milk and fat yields for 465 cows. For every cow also the genotype of btn3a1 is measured.
</p>


<h3>Source</h3>

<p>Artificial dataset generated to be consistent with Joanna Sz. study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(milkgene)
library(lattice)
xyplot(milk~fat, data=milkgene)
bwplot(milk~lactation, data=milkgene)
</code></pre>

<hr>
<h2 id='musculus'>A dataset relatead to mice musculus growth which depends on diet and genetic structure</h2><span id='topic+musculus'></span><span id='topic+muscle'></span>

<h3>Description</h3>

<p>Dataset from the book ,,Modele liniowe i mieszane w R, wraz z przykladami w analizie danych&rdquo;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(musculus)</code></pre>


<h3>Format</h3>

<p>data.frame with 30 obs. and 10 variables
</p>

<dl>
<dt><code>id</code></dt><dd><p>an individual id</p>
</dd>
<dt><code>dadid</code></dt><dd><p>id of father, 0 for founders</p>
</dd>
<dt><code>momid</code></dt><dd><p>id of mother, 0 for founders</p>
</dd>
<dt><code>sex</code></dt><dd><p>sex</p>
</dd>
<dt><code>sigma</code></dt><dd><p>maximal stress</p>
</dd>
<dt><code>diet</code></dt><dd><p>diet, D1 or D2</p>
</dd>
<dt><code>k1</code></dt><dd><p>resilience coefficient in point 1</p>
</dd>
<dt><code>k2</code></dt><dd><p>resilience coefficient in point 2</p>
</dd>
<dt><code>E1</code></dt><dd><p>Younga module in point 1</p>
</dd>
<dt><code>E2</code></dt><dd><p>Younga module in point 2</p>
</dd>
</dl>



<h3>Details</h3>

<p>Dataset from the book ,,Modele liniowe i mieszane w R, wraz z przykladami w analizie danych&rdquo;.
</p>
<p>Used as an example of model with mixed effects where random effects have know dependency structure, here related to the kinship coefficient.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
   require(kinship2)
   pedmus &lt;- pedigree(musculus$id, musculus$dadid, musculus$momid, musculus$sex)
   plot(pedmus, affected=musculus$diet)

   fam  &lt;- makefamid(musculus$id, musculus$dadid, musculus$momid)
   kmatrix &lt;- makekinship(fam, musculus$id, musculus$dadid, musculus$momid)
   kmatrix[1:5,1:15]

## End(Not run)
</code></pre>

<hr>
<h2 id='Plot+20results+20from+20post+20hoc+20testing'>
A function for visual representation of pairwise testing (both for pairwise.t.test and pairwise.wilcox.test)
</h2><span id='topic+plotPairwiseTests'></span>

<h3>Description</h3>

<p>Plot sets of groups in which means of medians are not significantly different.
</p>
<p>On the veritical axis the means are marked. Then in a greedy fashion means that are not significantly different are linked by a line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> plotPairwiseTests(p.vals, means, alpha=0.05, digits=3, mar=c(2,10,3,1), ...) 
 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Plot+2B20results+2B20from+2B20post+2B20hoc+2B20testing_+3A_p.vals">p.vals</code></td>
<td>
<p>A slot <code>$p.value</code> from the object returned by pairwise.*.test function</p>
</td></tr>
<tr><td><code id="Plot+2B20results+2B20from+2B20post+2B20hoc+2B20testing_+3A_means">means</code></td>
<td>
<p>A vector of means or medians corresponding to p.vals object (the order of groups should be the same in both objects)</p>
</td></tr>
<tr><td><code id="Plot+2B20results+2B20from+2B20post+2B20hoc+2B20testing_+3A_alpha">alpha</code></td>
<td>
<p>A threshold for p.value</p>
</td></tr>
<tr><td><code id="Plot+2B20results+2B20from+2B20post+2B20hoc+2B20testing_+3A_digits">digits</code></td>
<td>
<p>Number of significant digits to be ploted with means.</p>
</td></tr>
<tr><td><code id="Plot+2B20results+2B20from+2B20post+2B20hoc+2B20testing_+3A_mar">mar</code></td>
<td>
<p>Figure margins, left margin should be large enought to handle names of groups</p>
</td></tr>
<tr><td><code id="Plot+2B20results+2B20from+2B20post+2B20hoc+2B20testing_+3A_...">...</code></td>
<td>
<p>These arguments are passed to the plot function. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Przemyslaw Biecek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
tmp1 &lt;- pairwise.wilcox.test(iris$Sepal.Width, iris$Species)
tmp2 &lt;- tapply(iris$Sepal.Width, iris$Species, median, na.rm=TRUE)
plotPairwiseTests(tmp1$p.value, tmp2, alpha=0.001)

</code></pre>

<hr>
<h2 id='schizophrenia'>Genetic backgroud of schizophrenia</h2><span id='topic+schizophrenia'></span>

<h3>Description</h3>

<p>Dataset with genotypes and phenotypes for 98 patients with schizophrenia disorder. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(schizophrenia)</code></pre>


<h3>Format</h3>

<p>data.frame with 98 obs. and 9 variables
</p>

<dl>
<dt><code>NfkB</code>, <code>CD28</code>, <code>IFN</code></dt><dd><p>Genotypes for SNP mutations in selected three genes</p>
</dd>
<dt><code>Dikeos.manic</code>, <code>Dikeos.reality.distortion</code>, <code>Dikeos.depression</code>, <code>Dikeos.disorganization</code>, <code>Dikeos.negative</code></dt><dd><p>Dikeos scores for schizophrenia measured in five domains </p>
</dd>
<dt><code>Dikeos.sum</code></dt><dd><p>Sum of Dikeos scores</p>
</dd>
</dl>



<h3>Details</h3>

<p>Alleles for two SNPs in genes: Nuclear Factor-Kappa Beta (NfkB) and Cluster of Differentiation 28 (CD28) were examined as well as mental health described by five scales (see Dikeos 2008 for more details).
</p>


<h3>Source</h3>

<p>Artificial dataset generated to be consistent with Dorota F. study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(schizophrenia)
attach(schizophrenia)
interaction.plot(CD28, NfkB, Dikeos.sum)
interaction.plot(NfkB, CD28, Dikeos.sum)
model.tables(aov(Dikeos.sum~NfkB*CD28))
</code></pre>

<hr>
<h2 id='score.cardiovascular'>
SCORE for Cardiovascular Risk
</h2><span id='topic+score.cardiovascular'></span><span id='topic+calculateScoreEur'></span>

<h3>Description</h3>

<p>Calculation of risk SCORE for use in the clinical management of cardiovascular risk in European.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateScoreEur(age, cholesterol, SBP, currentSmoker, 
  gender = "Men", risk = "Low risk")

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="score.cardiovascular_+3A_age">age</code></td>
<td>
<p>  age in years </p>
</td></tr>
<tr><td><code id="score.cardiovascular_+3A_cholesterol">cholesterol</code></td>
<td>
<p> in mmol/L </p>
</td></tr>
<tr><td><code id="score.cardiovascular_+3A_sbp">SBP</code></td>
<td>
<p> Systolic blood pressure in mmHg </p>
</td></tr>
<tr><td><code id="score.cardiovascular_+3A_currentsmoker">currentSmoker</code></td>
<td>
<p> the current smoker status,  1 for current smokers, 0 for non smokers</p>
</td></tr>
<tr><td><code id="score.cardiovascular_+3A_gender">gender</code></td>
<td>
<p> &quot;Men&quot; or &quot;Women&quot; </p>
</td></tr>
<tr><td><code id="score.cardiovascular_+3A_risk">risk</code></td>
<td>
<p> is it &quot;Low risk&quot; or &quot;High risk&quot; group</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculation of SCORE based on the paper
</p>
<p>,,Estimation of ten-year risk of fatal cardiovascular disease in Europe: the SCORE project&rdquo;
R.M. Conroy et. al.
Eur Heart J (2003) 24 (11): 987-1003. doi: 10.1016/S0195-668X(03)00114-3
</p>


<h3>Author(s)</h3>

<p> Przemyslaw Biecek</p>

<hr>
<h2 id='SejmSenat'>
SejmSenat
</h2><span id='topic+SejmSenat'></span>

<h3>Description</h3>

<p>Changes in word usage in consecutive Sejm and Senate cadencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SejmSenat)</code></pre>


<h3>Format</h3>

<p>contingency matrix with 973 27 rows and 8 columns
</p>

<dl>
<dt><code>Sejm.I</code>, <code>Sejm.II</code>, <code>Sejm.III</code>, <code>Sejm.IV</code>, </dt><dd><p>summary of records from four Sejm cadencies</p>
</dd>
<dt><code>Senat.II</code>, <code>Senat.III</code>, <code>Senat.IV</code>, <code>Senat.V</code>, </dt><dd><p>summary of records from four Senate cadencies</p>
</dd>
<dt><code>adj</code>,   <code>adja</code>,  <code>adjp</code>,  <code>adv</code>,   <code>aglt</code>,  <code>bedzie</code>,<code>conj</code>,  <code>depr</code>,  <code>fin</code>,   <code>ger</code>,   <code>ign</code>,   <code>imps</code>,  <code>impt</code>,  <code>inf</code>,   <code>interp</code>,<code>num</code>,   <code>pact</code>,  <code>pant</code>,  <code>pcon</code>,  <code>ppas</code>,  <code>praet</code>, <code>pred</code>,  <code>prep</code>,  <code>qub</code>,   <code>siebie</code>,<code>subst</code>, <code>winien</code></dt><dd><p>word modes</p>
</dd>
</dl>



<h3>Details</h3>

<p>Word usage statistics generated from Sejm and Senat records
</p>


<h3>Source</h3>

<p>The IPI PAN Corpus
webpage http://korpus.pl/ 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SejmSenat)
library(ca)
# can you see some patterns?
plot(ca(SejmSenat[-15,]), mass =c(TRUE,TRUE), arrows =c(FALSE,TRUE))
</code></pre>

<hr>
<h2 id='vaccination'>Effective dose study</h2><span id='topic+vaccination'></span>

<h3>Description</h3>

<p>What is the minimal dose that is effective?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(vaccination)</code></pre>


<h3>Format</h3>

<p>data.frame with 100 obs. and 2 variables
</p>

<dl>
<dt><code>response</code></dt><dd><p>a reaction effect</p>
</dd>
<dt><code>dose</code></dt><dd><p>a dose that was applied</p>
</dd>
</dl>



<h3>Details</h3>

<p>Responses for different doses of treatment.
</p>


<h3>Source</h3>

<p>Artificial dataset generated to be consistent with Karolina P. study
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(vaccination)
library(lattice)
bwplot(response~dose, data=vaccination)
</code></pre>

<hr>
<h2 id='YXZ'>
Artificial dataset which shows the differences between tests type I and III (sequential vs. marginal)
</h2><span id='topic+YXZ'></span>

<h3>Description</h3>

<p>Artificial dataset, shows inconsistency for test type I and III
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(YXZ)</code></pre>


<h3>Format</h3>

<p>data.frame with 100 obs. and 3 variables
</p>

<dl>
<dt><code>X</code>, <code>Z</code></dt><dd><p>explanatory variables</p>
</dd>
<dt><code>Y</code></dt><dd><p>response variable</p>
</dd>
</dl>



<h3>Details</h3>

<p>See the example, results for staistical tests are inconsistet due to correlation between X and Z variables
</p>


<h3>Source</h3>

<p>Artificial dataset, generated by PBI
</p>


<h3>Examples</h3>

<pre><code class='language-R'>attach(YXZ)
summary(lm(Y~X+Z))
anova(lm(Y~Z+X))
anova(lm(Y~X))
anova(lm(Y~Z))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
