<!DOCTYPE html><html lang="en-US"><head><title>Help for package pcadapt</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pcadapt}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bed2matrix'><p>Convert a bed to a matrix</p></a></li>
<li><a href='#get_statistics'><p>pcadapt statistics</p></a></li>
<li><a href='#get.pc'><p>Get the principal component the most associated with a genetic marker</p></a></li>
<li><a href='#get.pop.names'><p>Retrieve population names</p></a></li>
<li><a href='#get.score.color'><p>Population colorization</p></a></li>
<li><a href='#hist_plot'><p>Neutral Distribution Estimation</p></a></li>
<li><a href='#manhattan_plot'><p>Manhattan Plot</p></a></li>
<li><a href='#pcadapt'><p>Principal Component Analysis for outlier detection</p></a></li>
<li><a href='#ped2pcadapt'><p>Convert ped files</p></a></li>
<li><a href='#plot.pcadapt'><p>pcadapt visualization tool</p></a></li>
<li><a href='#print_convert'><p>Summary</p></a></li>
<li><a href='#qq_plot'><p>p-values Q-Q Plot</p></a></li>
<li><a href='#read.pcadapt'><p>File Converter</p></a></li>
<li><a href='#run.pcadapt'><p>Shiny app</p></a></li>
<li><a href='#score_plot'><p>Principal Components Analysis Scores Plot</p></a></li>
<li><a href='#scree_plot'><p>Principal Components Analysis Scree Plot</p></a></li>
<li><a href='#vcf_convert'><p>Convert vcfR genotype matrices</p></a></li>
<li><a href='#vcf2pcadapt'><p>vcfR-based converter</p></a></li>
<li><a href='#writeBed'><p>Write PLINK files</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Fast Principal Component Analysis for Outlier Detection</td>
</tr>
<tr>
<td>Version:</td>
<td>4.4.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-09-24</td>
</tr>
<tr>
<td>Description:</td>
<td>Methods to detect genetic markers involved in biological
    adaptation. 'pcadapt' provides statistical tools for outlier detection based on
    Principal Component Analysis. Implements the method described in (Luu, 2016)
    &lt;<a href="https://doi.org/10.1111%2F1755-0998.12592">doi:10.1111/1755-0998.12592</a>&gt; and later revised in (Privé, 2020) 
    &lt;<a href="https://doi.org/10.1093%2Fmolbev%2Fmsaa053">doi:10.1093/molbev/msaa053</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>bigutilsr (&ge; 0.3), data.table, ggplot2, magrittr, mmapcharr
(&ge; 0.3), Rcpp (&ge; 0.12.8), RSpectra</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>mmapcharr, Rcpp, rmio</td>
</tr>
<tr>
<td>Suggests:</td>
<td>plotly, shiny, spelling, testthat, vcfR</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/bcm-uga/pcadapt">https://github.com/bcm-uga/pcadapt</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/bcm-uga/pcadapt/issues">https://github.com/bcm-uga/pcadapt/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-24 12:52:57 UTC; au639593</td>
</tr>
<tr>
<td>Author:</td>
<td>Keurcien Luu [aut],
  Michael Blum [aut],
  Florian Privé [aut, cre],
  Eric Bazin [ctb],
  Nicolas Duforet-Frebourg [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Florian Privé &lt;florian.prive.21@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-24 13:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='bed2matrix'>Convert a bed to a matrix</h2><span id='topic+bed2matrix'></span>

<h3>Description</h3>

<p>Convert a bed to a matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bed2matrix(bedfile, n = NULL, p = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bed2matrix_+3A_bedfile">bedfile</code></td>
<td>
<p>Path to a bed file.</p>
</td></tr>
<tr><td><code id="bed2matrix_+3A_n">n</code></td>
<td>
<p>Number of samples. Default reads it from corresponding fam file.</p>
</td></tr>
<tr><td><code id="bed2matrix_+3A_p">p</code></td>
<td>
<p>Number of SNPs. Default reads it from corresponding bim file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bedfile &lt;- system.file("extdata", "geno3pops.bed", package = "pcadapt")
mat &lt;- bed2matrix(bedfile)
dim(mat)
table(mat)
</code></pre>

<hr>
<h2 id='get_statistics'>pcadapt statistics</h2><span id='topic+get_statistics'></span>

<h3>Description</h3>

<p><code>get_statistics</code> returns chi-squared distributed statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_statistics(zscores, method, pass)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_statistics_+3A_zscores">zscores</code></td>
<td>
<p>a numeric matrix containing the z-scores.</p>
</td></tr>
<tr><td><code id="get_statistics_+3A_method">method</code></td>
<td>
<p>a character string specifying the method to be used to compute
the p-values. Two statistics are currently available, <code>"mahalanobis"</code>,
and <code>"componentwise"</code>.</p>
</td></tr>
<tr><td><code id="get_statistics_+3A_pass">pass</code></td>
<td>
<p>a boolean vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value is a list containing the test statistics and the 
associated p-values.
</p>

<hr>
<h2 id='get.pc'>Get the principal component the most associated with a genetic marker</h2><span id='topic+get.pc'></span>

<h3>Description</h3>

<p><code>get.pc</code> returns a data frame such that each row contains the index of
the genetic marker and the principal component the most correlated with it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.pc(x, list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.pc_+3A_x">x</code></td>
<td>
<p>an object of class 'pcadapt'.</p>
</td></tr>
<tr><td><code id="get.pc_+3A_list">list</code></td>
<td>
<p>a list of integers corresponding to the indices of the markers of interest.</p>
</td></tr>
</table>

<hr>
<h2 id='get.pop.names'>Retrieve population names</h2><span id='topic+get.pop.names'></span>

<h3>Description</h3>

<p><code>get.pop.names</code> retrieves the population names from the population file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.pop.names(pop)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.pop.names_+3A_pop">pop</code></td>
<td>
<p>a list of integers or strings specifying which population the
individuals belong to.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## see also ?pcadapt for examples

</code></pre>

<hr>
<h2 id='get.score.color'>Population colorization</h2><span id='topic+get.score.color'></span>

<h3>Description</h3>

<p><code>get.score.color</code> allows the user to display individuals of the same
predefined population with the same color when using the option
<code>"scores"</code> in <code>pcadapt</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.score.color(pop)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.score.color_+3A_pop">pop</code></td>
<td>
<p>a list of integers or strings specifying which population the
individuals belong to.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## see also ?pcadapt for examples

</code></pre>

<hr>
<h2 id='hist_plot'>Neutral Distribution Estimation</h2><span id='topic+hist_plot'></span>

<h3>Description</h3>

<p><code>hist_plot</code> plots the histogram of the chi-squared statistics, as well 
as the estimated null distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hist_plot(x, K)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hist_plot_+3A_x">x</code></td>
<td>
<p>an output from <code>outlier</code> containing the chi-squared statistics.</p>
</td></tr>
<tr><td><code id="hist_plot_+3A_k">K</code></td>
<td>
<p>an integer indicating which principal component the histogram will 
be associated with.</p>
</td></tr>
</table>

<hr>
<h2 id='manhattan_plot'>Manhattan Plot</h2><span id='topic+manhattan_plot'></span>

<h3>Description</h3>

<p><code>manhattan_plot</code> displays a Manhattan plot which represents the p-values 
for each SNP for a particular test statistic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattan_plot(x, chr.info, snp.info, plt.pkg = "ggplot", K = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="manhattan_plot_+3A_x">x</code></td>
<td>
<p>an object of class &quot;pcadapt&quot; generated with <code>pcadapt</code> 
containing the p-values of interest.</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_chr.info">chr.info</code></td>
<td>
<p>a list containing the chromosome information for each marker.</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_snp.info">snp.info</code></td>
<td>
<p>a list containing the names of all genetic markers present in
the input.</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_plt.pkg">plt.pkg</code></td>
<td>
<p>a character string specifying the package to be used to 
display the graphical outputs. Use <code>"plotly"</code> for interactive plots, or 
<code>"ggplot"</code> for static plots.</p>
</td></tr>
<tr><td><code id="manhattan_plot_+3A_k">K</code></td>
<td>
<p>an integer specifying which principal component to display when 
<code>method="componentwise"</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='pcadapt'>Principal Component Analysis for outlier detection</h2><span id='topic+pcadapt'></span><span id='topic+pcadapt.pcadapt_matrix'></span><span id='topic+pcadapt.pcadapt_bed'></span><span id='topic+pcadapt.pcadapt_pool'></span>

<h3>Description</h3>

<p><code>pcadapt</code> performs principal component analysis and computes p-values to
test for outliers. The test for outliers is based on the correlations between
genetic variation and the first <code>K</code> principal components. <code>pcadapt</code>
also handles Pool-seq data for which the statistical analysis is performed on
the genetic markers frequencies. Returns an object of class <code>pcadapt</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcadapt(
  input,
  K = 2,
  method = "mahalanobis",
  min.maf = 0.05,
  ploidy = 2,
  LD.clumping = NULL,
  pca.only = FALSE,
  tol = 1e-04
)

## S3 method for class 'pcadapt_matrix'
pcadapt(
  input,
  K = 2,
  method = c("mahalanobis", "componentwise"),
  min.maf = 0.05,
  ploidy = 2,
  LD.clumping = NULL,
  pca.only = FALSE,
  tol = 1e-04
)

## S3 method for class 'pcadapt_bed'
pcadapt(
  input,
  K = 2,
  method = c("mahalanobis", "componentwise"),
  min.maf = 0.05,
  ploidy = 2,
  LD.clumping = NULL,
  pca.only = FALSE,
  tol = 1e-04
)

## S3 method for class 'pcadapt_pool'
pcadapt(
  input,
  K = (nrow(input) - 1),
  method = "mahalanobis",
  min.maf = 0.05,
  ploidy = NULL,
  LD.clumping = NULL,
  pca.only = FALSE,
  tol
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pcadapt_+3A_input">input</code></td>
<td>
<p>The output of function <code>read.pcadapt</code>.</p>
</td></tr>
<tr><td><code id="pcadapt_+3A_k">K</code></td>
<td>
<p>an integer specifying the number of principal components to retain.</p>
</td></tr>
<tr><td><code id="pcadapt_+3A_method">method</code></td>
<td>
<p>a character string specifying the method to be used to compute
the p-values. Two statistics are currently available, <code>"mahalanobis"</code>,
and <code>"componentwise"</code>.</p>
</td></tr>
<tr><td><code id="pcadapt_+3A_min.maf">min.maf</code></td>
<td>
<p>Threshold of minor allele frequencies above which p-values are 
computed. Default is <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="pcadapt_+3A_ploidy">ploidy</code></td>
<td>
<p>Number of trials, parameter of the binomial distribution. 
Default is 2, which corresponds to diploidy, such as for the human genome.</p>
</td></tr>
<tr><td><code id="pcadapt_+3A_ld.clumping">LD.clumping</code></td>
<td>
<p>Default is <code>NULL</code> and doesn't use any SNP thinning.
If you want to use SNP thinning, provide a named list with parameters 
<code>$size</code> and <code>$thr</code> which corresponds respectively to the window 
radius and the squared correlation threshold. A good default value would 
be <code>list(size = 500, thr = 0.1)</code>.</p>
</td></tr>
<tr><td><code id="pcadapt_+3A_pca.only">pca.only</code></td>
<td>
<p>a logical value indicating whether PCA results should be 
returned (before computing any statistic).</p>
</td></tr>
<tr><td><code id="pcadapt_+3A_tol">tol</code></td>
<td>
<p>Convergence criterion of <code>RSpectra::svds()</code>. 
Default is <code>1e-4</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>First, a principal component analysis is performed on the scaled and 
centered genotype data. Depending on the specified <code>method</code>, different 
test  statistics can be used.
</p>
<p><code>mahalanobis</code> (default): the robust Mahalanobis distance is computed for 
each genetic marker using a robust estimate of both mean and covariance 
matrix between the <code>K</code> vectors of z-scores.
</p>
<p><code>communality</code>: the communality statistic measures the proportion of 
variance explained by the first <code>K</code> PCs. Deprecated in version 4.0.0.
</p>
<p><code>componentwise</code>: returns a matrix of z-scores.
</p>
<p>To compute p-values, test statistics (<code>stat</code>) are divided by a genomic 
inflation factor (<code>gif</code>) when <code>method="mahalanobis"</code>. When using 
<code>method="mahalanobis"</code>, the scaled statistics 
(<code>chi2_stat</code>) should follow a chi-squared distribution with <code>K</code> 
degrees of freedom. When using <code>method="componentwise"</code>, the z-scores 
should follow a chi-squared distribution with <code>1</code> degree of freedom. For 
Pool-seq data, <code>pcadapt</code> provides p-values based on the Mahalanobis 
distance for each SNP.
</p>


<h3>Value</h3>

<p>The returned value is an object of class <code>pcadapt</code>.
</p>

<hr>
<h2 id='ped2pcadapt'>Convert ped files</h2><span id='topic+ped2pcadapt'></span>

<h3>Description</h3>

<p><code>ped2pcadapt</code> converts <code>ped</code> files to the format <code>pcadapt</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ped2pcadapt(input, output)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ped2pcadapt_+3A_input">input</code></td>
<td>
<p>a character string specifying the name of the file to be converted.</p>
</td></tr>
<tr><td><code id="ped2pcadapt_+3A_output">output</code></td>
<td>
<p>a character string specifying the name of the output file.</p>
</td></tr>
</table>

<hr>
<h2 id='plot.pcadapt'>pcadapt visualization tool</h2><span id='topic+plot.pcadapt'></span>

<h3>Description</h3>

<p><code>plot.pcadapt</code> is a method designed for objects of class <code>pcadapt</code>.
It provides plotting options for quick visualization of <code>pcadapt</code> 
objects. Different options are currently available : <code>"screeplot"</code>, 
<code>"scores"</code>, <code>"stat.distribution"</code>, <code>"manhattan"</code> and 
<code>"qqplot"</code>. <code>"screeplot"</code> shows the decay of the genotype matrix 
singular values and provides a figure to help with the choice of <code>K</code>.
<code>"scores"</code> plots the projection of the individuals onto the first two 
principal components. <code>"stat.distribution"</code> displays the histogram of 
the selected test statistics, as well as the estimated distribution for the 
neutral SNPs. <code>"manhattan"</code> draws the Manhattan plot of the p-values 
associated with the statistic of interest. <code>"qqplot"</code> draws a Q-Q plot 
of the p-values associated with the statistic of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pcadapt'
plot(
  x,
  ...,
  option = "manhattan",
  i = 1,
  j = 2,
  pop,
  col,
  chr.info = NULL,
  snp.info = NULL,
  plt.pkg = "ggplot",
  K = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.pcadapt_+3A_x">x</code></td>
<td>
<p>an object of class &quot;pcadapt&quot; generated with <code>pcadapt</code>.</p>
</td></tr>
<tr><td><code id="plot.pcadapt_+3A_...">...</code></td>
<td>
<p>...</p>
</td></tr>
<tr><td><code id="plot.pcadapt_+3A_option">option</code></td>
<td>
<p>a character string specifying the figures to be displayed. If 
<code>NULL</code> (the default), all three plots are printed.</p>
</td></tr>
<tr><td><code id="plot.pcadapt_+3A_i">i</code></td>
<td>
<p>an integer indicating onto which principal component the individuals
are projected when the &quot;scores&quot; option is chosen.
Default value is set to <code>1</code>.</p>
</td></tr>
<tr><td><code id="plot.pcadapt_+3A_j">j</code></td>
<td>
<p>an integer indicating onto which principal component the individuals 
are projected when the &quot;scores&quot; option is chosen.
Default value is set to <code>2</code>.</p>
</td></tr>
<tr><td><code id="plot.pcadapt_+3A_pop">pop</code></td>
<td>
<p>a list of integers or strings specifying which subpopulation the 
individuals belong to.</p>
</td></tr>
<tr><td><code id="plot.pcadapt_+3A_col">col</code></td>
<td>
<p>a list of colors to be used in the score plot.</p>
</td></tr>
<tr><td><code id="plot.pcadapt_+3A_chr.info">chr.info</code></td>
<td>
<p>a list containing the chromosome information for each marker.</p>
</td></tr>
<tr><td><code id="plot.pcadapt_+3A_snp.info">snp.info</code></td>
<td>
<p>a list containing the names of all genetic markers present in
the input.</p>
</td></tr>
<tr><td><code id="plot.pcadapt_+3A_plt.pkg">plt.pkg</code></td>
<td>
<p>a character string specifying the package to be used to 
display the graphical outputs. Use <code>"plotly"</code> for interactive plots, or 
<code>"ggplot"</code> for static plots.</p>
</td></tr>
<tr><td><code id="plot.pcadapt_+3A_k">K</code></td>
<td>
<p>an integer specifying the principal component of interest. <code>K</code> 
has to be specified only when using the <code>"componentwise"</code> method.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## see ?pcadapt for examples

</code></pre>

<hr>
<h2 id='print_convert'>Summary</h2><span id='topic+print_convert'></span>

<h3>Description</h3>

<p><code>print_convert</code> prints out a summary of the file conversion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_convert(input, output, M, N, pool)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print_convert_+3A_input">input</code></td>
<td>
<p>a genotype matrix or a character string specifying the name of the file to be converted.</p>
</td></tr>
<tr><td><code id="print_convert_+3A_output">output</code></td>
<td>
<p>a character string specifying the name of the output file.</p>
</td></tr>
<tr><td><code id="print_convert_+3A_m">M</code></td>
<td>
<p>an integer specifying the number of genetic markers present in the data.</p>
</td></tr>
<tr><td><code id="print_convert_+3A_n">N</code></td>
<td>
<p>an integer specifying the number of individuals present in the data.</p>
</td></tr>
<tr><td><code id="print_convert_+3A_pool">pool</code></td>
<td>
<p>an integer specifying the type of data. '0' for genotype data, '1' for pooled data.</p>
</td></tr>
</table>

<hr>
<h2 id='qq_plot'>p-values Q-Q Plot</h2><span id='topic+qq_plot'></span>

<h3>Description</h3>

<p><code>qq_plot</code> plots a Q-Q plot of the p-values computed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qq_plot(x, K = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qq_plot_+3A_x">x</code></td>
<td>
<p>an output from <code>outlier</code> containing the p-values of interest.</p>
</td></tr>
<tr><td><code id="qq_plot_+3A_k">K</code></td>
<td>
<p>an integer specifying which principal component to display when <code>method="componentwise"</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='read.pcadapt'>File Converter</h2><span id='topic+read.pcadapt'></span>

<h3>Description</h3>

<p><code>read.pcadapt</code> converts genotype matrices or files to an appropriate
format readable by <code>pcadapt</code>. For a file as input, you can choose to
return either a matrix or convert it in bed/bim/fam files. 
For a matrix as input, this returns a matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.pcadapt(
  input,
  type = c("pcadapt", "lfmm", "vcf", "bed", "ped", "pool", "example"),
  type.out = c("bed", "matrix"),
  allele.sep = c("/", "|"),
  pop.sizes,
  ploidy,
  local.env,
  blocksize
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read.pcadapt_+3A_input">input</code></td>
<td>
<p>A genotype matrix or a character string specifying the name of 
the file to be converted. Matrices should use NAs to encode missing values. 
To encode missing values in 'pcadapt' and 'lfmm' files, 9s should be used.</p>
</td></tr>
<tr><td><code id="read.pcadapt_+3A_type">type</code></td>
<td>
<p>A character string specifying the type of data to be converted 
from. Converters from 'vcf' and 'ped' formats are not maintained anymore;
if you have any issue with those, please use PLINK &gt;= 1.9 to convert them
to the 'bed' format.</p>
</td></tr>
<tr><td><code id="read.pcadapt_+3A_type.out">type.out</code></td>
<td>
<p>Either a bed file or a standard R matrix. 
If the input is a matrix, then the output is automatically a matrix 
(so that you don't need to specify this parameter). 
If the input is a bed file, then the output is also a bed file.</p>
</td></tr>
<tr><td><code id="read.pcadapt_+3A_allele.sep">allele.sep</code></td>
<td>
<p>a vector of characters indicating what delimiters are used 
in VCF files. By default, only &quot;|&quot; and &quot;/&quot; are recognized. 
So, this argument is only useful for <code>type = "vcf"</code>.</p>
</td></tr>
<tr><td><code id="read.pcadapt_+3A_pop.sizes">pop.sizes</code></td>
<td>
<p>deprecated argument.</p>
</td></tr>
<tr><td><code id="read.pcadapt_+3A_ploidy">ploidy</code></td>
<td>
<p>deprecated argument.</p>
</td></tr>
<tr><td><code id="read.pcadapt_+3A_local.env">local.env</code></td>
<td>
<p>deprecated argument.</p>
</td></tr>
<tr><td><code id="read.pcadapt_+3A_blocksize">blocksize</code></td>
<td>
<p>deprecated argument.</p>
</td></tr>
</table>

<hr>
<h2 id='run.pcadapt'>Shiny app</h2><span id='topic+run.pcadapt'></span>

<h3>Description</h3>

<p><code>pcadapt</code> comes with a Shiny interface.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run.pcadapt()
</code></pre>

<hr>
<h2 id='score_plot'>Principal Components Analysis Scores Plot</h2><span id='topic+score_plot'></span>

<h3>Description</h3>

<p><code>"score_plot"</code> plots the projection of the individuals onto the 
first two principal components.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>score_plot(x, i = 1, j = 2, pop, col, plt.pkg = "ggplot")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="score_plot_+3A_x">x</code></td>
<td>
<p>an output from <code>pcadapt</code> containing the scores.</p>
</td></tr>
<tr><td><code id="score_plot_+3A_i">i</code></td>
<td>
<p>an integer indicating onto which principal component the individuals
are projected when the &quot;scores&quot; option is chosen.
Default value is set to <code>1</code>.</p>
</td></tr>
<tr><td><code id="score_plot_+3A_j">j</code></td>
<td>
<p>an integer indicating onto which principal component the individuals 
are projected when the &quot;scores&quot; option is chosen.
Default value is set to <code>2</code>.</p>
</td></tr>
<tr><td><code id="score_plot_+3A_pop">pop</code></td>
<td>
<p>a list of integers or strings specifying which subpopulation the 
individuals belong to.</p>
</td></tr>
<tr><td><code id="score_plot_+3A_plt.pkg">plt.pkg</code></td>
<td>
<p>a character string specifying the package to be used to 
display the graphical outputs. Use <code>"plotly"</code> for interactive plots, or 
<code>"ggplot"</code> for static plots.</p>
</td></tr>
</table>

<hr>
<h2 id='scree_plot'>Principal Components Analysis Scree Plot</h2><span id='topic+scree_plot'></span>

<h3>Description</h3>

<p><code>scree_plot</code> plots the scree plot associated with the principal components 
analysis performed on the dataset. NB : <code>pcadapt</code> has to be run on the
dataset in order to get an output readable by <code>plot.screePlot</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scree_plot(x, K = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scree_plot_+3A_x">x</code></td>
<td>
<p>an output from <code>pcadapt</code> containing the singular values.</p>
</td></tr>
<tr><td><code id="scree_plot_+3A_k">K</code></td>
<td>
<p>an integer specifying the number of components to plot.</p>
</td></tr>
</table>

<hr>
<h2 id='vcf_convert'>Convert vcfR genotype matrices</h2><span id='topic+vcf_convert'></span>

<h3>Description</h3>

<p><code>vcf_convert</code> converts outputs of <code>extract.gt</code> to the format 
<code>pcadapt</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf_convert(string_geno, output, allele_sep)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vcf_convert_+3A_string_geno">string_geno</code></td>
<td>
<p>a genotype matrix extracted from a VCF file with 'vcfR'.</p>
</td></tr>
<tr><td><code id="vcf_convert_+3A_output">output</code></td>
<td>
<p>a character string indicating the name of the output file.</p>
</td></tr>
<tr><td><code id="vcf_convert_+3A_allele_sep">allele_sep</code></td>
<td>
<p>a vector of characters indicating what delimiters are used 
to separate alleles.</p>
</td></tr>
</table>

<hr>
<h2 id='vcf2pcadapt'>vcfR-based converter</h2><span id='topic+vcf2pcadapt'></span>

<h3>Description</h3>

<p><code>vcf2pcadapt</code> uses the package <code>vcfR</code> to extract the genotype information 
from a vcf file and exports it under the format required by <code>pcadapt</code>.
Note that this VCF converter is not maintained anymore. Please use PLINK.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf2pcadapt(input, output = "tmp.pcadapt", allele.sep = c("/", "|"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vcf2pcadapt_+3A_input">input</code></td>
<td>
<p>a character string specifying the name of the file to be converted.</p>
</td></tr>
<tr><td><code id="vcf2pcadapt_+3A_output">output</code></td>
<td>
<p>a character string indicating the name of the output file.</p>
</td></tr>
<tr><td><code id="vcf2pcadapt_+3A_allele.sep">allele.sep</code></td>
<td>
<p>a vector of characters indicating what delimiters are used
to separate alleles.</p>
</td></tr>
</table>

<hr>
<h2 id='writeBed'>Write PLINK files</h2><span id='topic+writeBed'></span>

<h3>Description</h3>

<p>Function to write bed/bim/fam files from a pcadapt or an lfmm file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeBed(file, is.pcadapt)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="writeBed_+3A_file">file</code></td>
<td>
<p>A pcadapt or lfmm file.</p>
</td></tr>
<tr><td><code id="writeBed_+3A_is.pcadapt">is.pcadapt</code></td>
<td>
<p>a boolean value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input 'bedfile' path.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
