<!DOCTYPE html><html lang="en"><head><title>Help for package PCFAM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PCFAM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PCFAM-package'>
<p>Computation of ancestry scores with mixed families and unrelated</p>
individuals</a></li>
<li><a href='#colcenter'><p>column centering of the data matrix</p></a></li>
<li><a href='#cov.function'>
<p>Sample covariance calculator</p></a></li>
<li><a href='#familyave'>
<p>Family average approach</p></a></li>
<li><a href='#fastcov'>
<p>Fast covariance function</p></a></li>
<li><a href='#findfamilies'>
<p>Find families</p></a></li>
<li><a href='#gr.pca'>
<p>The geometric rotation approach</p></a></li>
<li><a href='#ms.pca'>
<p>The matrix substitution approach</p></a></li>
<li><a href='#mysqrtm'>
<p>Matrix square root function</p></a></li>
<li><a href='#perfectwhiten'>
<p>The covariance preserving whitening function.</p></a></li>
<li><a href='#residualize'>
<p>Residualization and scale of the original genotype data</p></a></li>
<li><a href='#rowcol'>
<p>Indicators for of the row and column of the original matrix</p></a></li>
<li><a href='#rowscale'>
<p>Scale each row of a matrix</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Computation of Ancestry Scores with Mixed Families and Unrelated
Individuals</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-03-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Yi-Hui Zhou</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yi-Hui Zhou &lt;yihui_zhou@ncsu.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>We provide several algorithms to compute the genotype ancestry scores (such as eigenvector projections) in the case where highly correlated individuals are involved.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-03-24 18:30:41 UTC; yzhou19</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-03-24 19:10:36 UTC</td>
</tr>
</table>
<hr>
<h2 id='PCFAM-package'>
Computation of ancestry scores with mixed families and unrelated
individuals
</h2><span id='topic+PCFAM-package'></span><span id='topic+PCFAM'></span>

<h3>Description</h3>

<p>This package provides ancestry scores based on genotype data, and is robust to the presence of close-degree family members. Four main novel algorithms are represented: (i) Geometric rotation (within-family data orthogonalization); (ii) matrix substitution based on the decomposition of a target family-orthogonalized covariance matrix; (iii) covariance-preserving whitening, retaining covariances between unrelated pairs while orthogonalizing family members (Note: the function <a href="#topic+perfectwhiten">perfectwhiten</a> generates a new dataset which keeps the same covariance structure as the original set); (iv) using family-averaged data to obtain loadings for projection of family members.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> PCFAM</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-10-11</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL 2</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">

</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Yi-Hui Zhou
</p>
<p>Maintainer: Yi-Hui Zhou &lt;yihui_zhou@ncsu.edu&gt;
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals. arXiv:1606.08416
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- matrix(rbinom(1000*20,2,0.4),1000,20)
X[,1]=X[,2]*0.9
X=rowscale(X)
Xresid=residualize(X)
corXresid=cor(Xresid)
myfam=findfamilies(corXresid,0.1)
K=3
myms.pca=ms.pca(X,corXresid,0.1,K)
familyave.result=familyave(X,myfam,top=K)
</code></pre>

<hr>
<h2 id='colcenter'>column centering of the data matrix
</h2><span id='topic+colcenter'></span>

<h3>Description</h3>

<p>This function centerizes each column of the data matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colcenter(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="colcenter_+3A_x">X</code></td>
<td>
<p>input data matrix
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return the data matrix with each column centered
</p>


<h3>Author(s)</h3>

<p>Yi-Hui Zhou
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals. Yi-Hui Zhou, J.S. Marron, Fred Wright, arXiv:1606.08416.
</p>

<hr>
<h2 id='cov.function'>
Sample covariance calculator
</h2><span id='topic+cov.function'></span>

<h3>Description</h3>

<p>Obtain a sample covariance matrix 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cov.function(data.matrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cov.function_+3A_data.matrix">data.matrix</code></td>
<td>

<p>Input mxn data matrix
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>return the nxn sample covariance matrix
</p>


<h3>Author(s)</h3>

<p>Yi-Hui Zhou
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals.  arXiv:1606.08416.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- matrix(rbinom(1000*20,2,0.4),1000,20)
cov.X=cov.function(X)
</code></pre>

<hr>
<h2 id='familyave'>
Family average approach
</h2><span id='topic+familyave'></span>

<h3>Description</h3>

<p>This function implements the family-averaging algorithm, with loadings based on the combined data from singletons and family averages, then projected to all.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>familyave(Xall,myfam, top = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="familyave_+3A_xall">Xall</code></td>
<td>

<p>The original input genotype dataset
</p>
</td></tr>
<tr><td><code id="familyave_+3A_myfam">myfam</code></td>
<td>

<p>The identified family IDs. Each singleton forms his/her own family.
</p>
</td></tr>
<tr><td><code id="familyave_+3A_top">top</code></td>
<td>

<p>The number ancestry scores desired.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function averages the genotype information in each family, re-inflates to have appropriate variability, andtreats as a 'singleton' for the purpose of loading calculation. Ancestry scores are obtained by projection to all.
</p>


<h3>Value</h3>

<p>Output the top ancestry scores by combining family data with singletons</p>


<h3>Author(s)</h3>

<p>Yi-Hui Zhou
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals.  arXiv:1606.08416.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- matrix(rbinom(1000*20,2,0.4),1000,20)
X[,1]=X[,2]*0.9
X=rowscale(X)
Xresid=residualize(X)
corXresid=cor(Xresid)
myfam=findfamilies(corXresid,0.1)
K=3
familyave.result=familyave(X,myfam,top=K)

</code></pre>

<hr>
<h2 id='fastcov'>
Fast covariance function
</h2><span id='topic+fastcov'></span>

<h3>Description</h3>

<p>This function can generate covariance matrix faster than the regular cov() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fastcov(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fastcov_+3A_x">X</code></td>
<td>

<p>input mxn data matrix
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output nxn covariance matrix
</p>


<h3>Note</h3>

<p>The input data matrix has to be column scaled in advance.
</p>


<h3>Author(s)</h3>

<p>Yi-Hui Zhou,
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals.  arXiv:1606.08416.
</p>

<hr>
<h2 id='findfamilies'>
Find families
</h2><span id='topic+findfamilies'></span>

<h3>Description</h3>

<p>This function searches for pairs of individuals with high kinship based on the genotype correlation matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findfamilies(x, threshold = 0.4)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="findfamilies_+3A_x">x</code></td>
<td>

<p>The nxn correlation matrix of the input dataset.
</p>
</td></tr>
<tr><td><code id="findfamilies_+3A_threshold">threshold</code></td>
<td>

<p>This threshold is used to identify close-degree relatives. Recommended values are 0.4 to identify first-degree relatives, and 0.15 to identify first- and second-degree relatives.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output numerical family ID for each individual. Individuals with the same ID are judged to be family members.
</p>


<h3>Author(s)</h3>

<p>Yi-Hui Zhou
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals. arXiv:1606.08416.
</p>

<hr>
<h2 id='gr.pca'>
The geometric rotation approach
</h2><span id='topic+gr.pca'></span>

<h3>Description</h3>

<p>This algorithm rotates scaled genotypes among family members so that they are mutually orthogonal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gr.pca(data.input, index.family, myfam, weight, top, family.size, inflation)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gr.pca_+3A_data.input">data.input</code></td>
<td>

<p>Input dataset, each row is for a genetic feature (SNP), each column is for individual. Data are typically
number of minor alleles, possibly imputed.
</p>
</td></tr>
<tr><td><code id="gr.pca_+3A_index.family">index.family</code></td>
<td>

<p>Index vector to indicate the family id of each individual.
</p>
</td></tr>
<tr><td><code id="gr.pca_+3A_myfam">myfam</code></td>
<td>

<p>This value comes directly from the output of findfamilies().
</p>
</td></tr>
<tr><td><code id="gr.pca_+3A_weight">weight</code></td>
<td>

<p>Weight is 0 by default. This is a deprecated weight value that can be used to control the amount of rotation performed.  A weight of zero performs full orthogonalization, while a weight of 1 keeps the data unchanged.
</p>
</td></tr>
<tr><td><code id="gr.pca_+3A_top">top</code></td>
<td>

<p>The number of eigenvectors to be used.
</p>
</td></tr>
<tr><td><code id="gr.pca_+3A_family.size">family.size</code></td>
<td>

<p>The number of members in each family. Used to determine rotation angles.
</p>
</td></tr>
<tr><td><code id="gr.pca_+3A_inflation">inflation</code></td>
<td>

<p>The inflation of the data value is 0 under default. Deprecated.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>data.new</code></td>
<td>
<p>The new datamatrix after the geometric rotation</p>
</td></tr>
<tr><td><code>topPCs</code></td>
<td>
<p>The top eigenvectors</p>
</td></tr>
<tr><td><code>topEigenvalue</code></td>
<td>
<p>The top eigenvalues.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yi-Hui Zhou
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals.  arXiv:1606.08416.
</p>

<hr>
<h2 id='ms.pca'>
The matrix substitution approach
</h2><span id='topic+ms.pca'></span>

<h3>Description</h3>

<p>This function provides the matrix subsitution algorithm. The main idea is to replace the high covariance value entries in the covariance matrix which are produced by family members by a small value (e.g. median covariance). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ms.pca(X, corXresid, threshold, top)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ms.pca_+3A_x">X</code></td>
<td>

<p>The input data matrix
</p>
</td></tr>
<tr><td><code id="ms.pca_+3A_corxresid">corXresid</code></td>
<td>

<p>The correlation of the genotypes after residualization for any evidence of larger scale ancestry. Used to identify close-degree family members in a manner robust to large-scale ancestry.
</p>
</td></tr>
<tr><td><code id="ms.pca_+3A_threshold">threshold</code></td>
<td>

<p>Covariance values of identified family members are set to the threshold.
</p>
</td></tr>
<tr><td><code id="ms.pca_+3A_top">top</code></td>
<td>

<p>The number of ancestry scores to obtain.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>eigenvector</code></td>
<td>
<p>Eigenvectors after using the matrix substitution method</p>
</td></tr>
<tr><td><code>myeigen</code></td>
<td>
<p>The top eigenvalues and eigenvectors</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yi-Hui Zhou
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals.  arXiv:1606.08416.
</p>

<hr>
<h2 id='mysqrtm'>
Matrix square root function
</h2><span id='topic+mysqrtm'></span>

<h3>Description</h3>

<p>This function can find the matrix square root, without requiring a new package and often faster than other code.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mysqrtm(a, symmetric = F)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mysqrtm_+3A_a">a</code></td>
<td>

<p>The input matrix
</p>
</td></tr>
<tr><td><code id="mysqrtm_+3A_symmetric">symmetric</code></td>
<td>

<p>Default=FALSE. This argument indicates whether the input matrix is symmetric.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Matrix B is said to be a square root of A if the matrix product BB is equal to A.
</p>


<h3>Value</h3>

<p>returns the square root matrix B
</p>

<hr>
<h2 id='perfectwhiten'>
The covariance preserving whitening function.
</h2><span id='topic+perfectwhiten'></span>

<h3>Description</h3>

<p>This algorithm generates a new scaled 'genotype' dataset which keeps the same covariance structure as the original data, except that family members have been made orthogonal to each other, and singletons are unchanged.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perfectwhiten(Xun, Xfam, delta = 3e-04, threshold = 0.35, eta = NULL, addfuzz = F)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="perfectwhiten_+3A_xun">Xun</code></td>
<td>

<p>A matrix of (possibly scaled) genotypes, (number of SNPs)*(number of singletons)
</p>
</td></tr>
<tr><td><code id="perfectwhiten_+3A_xfam">Xfam</code></td>
<td>

<p>A matrix of (possibly scaled) genotypes, (number of SNPs)*(number of individuals belonging to families)
</p>
</td></tr>
<tr><td><code id="perfectwhiten_+3A_delta">delta</code></td>
<td>

<p>A slight offset used to ensure that the target covariance matrix is of full rank
</p>
</td></tr>
<tr><td><code id="perfectwhiten_+3A_threshold">threshold</code></td>
<td>

<p>The correlation threshold used to determine pairs of relatives. The choice should be less than the degree desired. For example, 0.35 captures first degree relatives (expected correlation 0.5), 0.15 captures first and second degree relatives (expected correlation for second degree relatives is 0.25).
</p>
</td></tr>
<tr><td><code id="perfectwhiten_+3A_eta">eta</code></td>
<td>

<p>This argument is the replacement value used for matrix substitution. The default is NULL, resulting in substitution by the median. 
</p>
</td></tr>
<tr><td><code id="perfectwhiten_+3A_addfuzz">addfuzz</code></td>
<td>

<p>The default is FALSE. Deprecated.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Xplusscaled</code></td>
<td>
<p>The row-scaled full genotype data, including both singletons and family members</p>
</td></tr> 
<tr><td><code>Y</code></td>
<td>
<p>The (scaled) genotype matrix after whitening, and should have a covariance matrix very close to Mtarget. Column means are zero</p>
</td></tr>
<tr><td><code>Ynotcolcentered</code></td>
<td>
<p>The same as Y, but with column means matching those of Xplusscaled</p>
</td></tr>
<tr><td><code>M</code></td>
<td>
<p>The covariance matrix of the full data</p>
</td></tr>
<tr><td><code>Mtilde</code></td>
<td>
<p>The covariance matrix after matrix substitution of all family pairs identified with correlations exceedingeta</p>
</td></tr>
<tr><td><code>whichbig</code></td>
<td>
<p>The set of indexes of M that have correlation exceeding threshold</p>
</td></tr>
<tr><td><code>covY</code></td>
<td>
<p>The covariance matrix of Y, useful to compare to M or to Mtarget</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yi-Hui ZHou, Fred A. Wright
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals.  arXiv:1606.08416.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- matrix(rbinom(1000*20,2,0.4),1000,20)
X[,1]=X[,2]*0.9
X=rowscale(X)
Xresid=residualize(X)
library(PCFAM)
corXresid=cor(Xresid)
myfam=findfamilies(corXresid,0.1)
K=3
perfect.result=perfectwhiten(X[,which(myfam==0)],X[,which(myfam==1)])


</code></pre>

<hr>
<h2 id='residualize'>
Residualization and scale of the original genotype data
</h2><span id='topic+residualize'></span>

<h3>Description</h3>

<p>Thus function performs a simple residualization of a row-scaled genotype dataset, removing
large-scale population stratification. Output is a residualized dataset appropriate for computing
correlations such that family members can be easily identified.
The function assumes X is row-scaled
</p>


<h3>Usage</h3>

<pre><code class='language-R'>residualize(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="residualize_+3A_x">X</code></td>
<td>

<p>The original input genotype dataset
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function pre-treatment the data before applying the findfamily function.
</p>


<h3>Value</h3>

<p>Outputs the new row-scaled genotype matrix after residualization</p>


<h3>Author(s)</h3>

<p>Yi-Hui Zhou
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals.  arXiv:1606.08416.
</p>

<hr>
<h2 id='rowcol'>
Indicators for of the row and column of the original matrix
</h2><span id='topic+rowcol'></span>

<h3>Description</h3>

<p>This function identifies the rows and columns of elements in a matrix, e.g. the family members identified based on the correlation matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rowcol(I, J, elements)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rowcol_+3A_i">I</code></td>
<td>
<p>The number of rows of the matrix (scalar)</p>
</td></tr>
<tr><td><code id="rowcol_+3A_j">J</code></td>
<td>
<p>The number of columns of the matrix (scalar)</p>
</td></tr>
<tr><td><code id="rowcol_+3A_elements">elements</code></td>
<td>
<p>A vector of matrix element indexes</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>whichrow</code></td>
<td>
<p>The rows of elements in the matrix</p>
</td></tr>
<tr><td><code>whichcol</code></td>
<td>
<p>The columns of elements in the matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yi-Hui ZHou, Fred A. Wright
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals.  arXiv:1606.08416.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X &lt;- matrix(rbinom(1000*20,2,0.4),1000,20)
X[,1]=X[,2]*0.9
X=rowscale(X)
</code></pre>

<hr>
<h2 id='rowscale'>
Scale each row of a matrix
</h2><span id='topic+rowscale'></span>

<h3>Description</h3>

<p>This function scales the input matrix so that each row mean is 0 and each row (sample) variance is 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rowscale(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rowscale_+3A_x">X</code></td>
<td>
<p>input data matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Output the row-scaled matrix.
</p>


<h3>Author(s)</h3>

<p>Yi-Hui ZHou, Fred A. Wright
</p>


<h3>References</h3>

<p>Computation of ancestry scores with mixed families and unrelated individuals.  arXiv:1606.08416.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
