<!DOCTYPE html><html><head><title>Help for package pcr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pcr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.pcr_amount'><p>Calculate the amounts</p></a></li>
<li><a href='#.pcr_average'><p>Get vector average by a variable</p></a></li>
<li><a href='#.pcr_cv'><p>Get vector coefficient of variance by a variable</p></a></li>
<li><a href='#.pcr_error'><p>Propage two vectors</p></a></li>
<li><a href='#.pcr_intercept'><p>Calculate the intercept of a line</p></a></li>
<li><a href='#.pcr_normalize'><p>Normalize vector by another</p></a></li>
<li><a href='#.pcr_plot_analyze'><p>Plotting function</p></a></li>
<li><a href='#.pcr_plot_assess'><p>Plot quality assessment graphs</p></a></li>
<li><a href='#.pcr_relative'><p>Raise two to a vector power</p></a></li>
<li><a href='#.pcr_rsquared'><p>Calculate R squared</p></a></li>
<li><a href='#.pcr_sd'><p>Get vector standard deviation by a variable</p></a></li>
<li><a href='#.pcr_slope'><p>Calculate the slope of a line</p></a></li>
<li><a href='#ct1'><p><code class="reqn">C_T</code> values from qPCR (separate tubes)</p></a></li>
<li><a href='#ct2'><p><code class="reqn">C_T</code> values from qPCR (same tubes)</p></a></li>
<li><a href='#ct3'><p><code class="reqn">C_T</code> values from qPCR (Serial dilutions)</p></a></li>
<li><a href='#ct4'><p><code class="reqn">C_T</code> values from qPCR (Serial dilutions)</p></a></li>
<li><a href='#pcr'><p><code>pcr</code> package</p></a></li>
<li><a href='#pcr_analyze'><p>Apply qPCR analysis methods</p></a></li>
<li><a href='#pcr_assess'><p>Assess qPCR data quality</p></a></li>
<li><a href='#pcr_curve'><p>Calculate the standard curve model</p></a></li>
<li><a href='#pcr_dct'><p>Calculate the delta_ct model</p></a></li>
<li><a href='#pcr_ddct'><p>Calculate the delta_delta_ct model</p></a></li>
<li><a href='#pcr_efficiency'><p>Calculate amplification efficiency</p></a></li>
<li><a href='#pcr_lm'><p>Linear regression qPCR data</p></a></li>
<li><a href='#pcr_standard'><p>Calculate the standard curve</p></a></li>
<li><a href='#pcr_test'><p>Statistical testing of PCR data</p></a></li>
<li><a href='#pcr_ttest'><p>t-test qPCR data</p></a></li>
<li><a href='#pcr_wilcox'><p>Wilcoxon test qPCR data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.2.2</td>
</tr>
<tr>
<td>Title:</td>
<td>Analyzing Real-Time Quantitative PCR Data</td>
</tr>
<tr>
<td>Description:</td>
<td>Calculates the amplification efficiency and curves from real-time 
  quantitative PCR (Polymerase Chain Reaction) data. Estimates the relative 
  expression from PCR data using the double delta CT and the standard curve 
  methods Livak &amp; Schmittgen (2001) &lt;<a href="https://doi.org/10.1006%2Fmeth.2001.1262">doi:10.1006/meth.2001.1262</a>&gt;. Tests for 
  statistical significance using two-group tests and linear regression 
  Yuan et al. (2006) &lt;<a href="https://doi.org/10.1186%2F1471-2105-7-85">doi:10.1186/1471-2105-7-85</a>&gt;.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mahmoud Ahmed &lt;mahmoud.s.fahmy@students.kasralainy.edu.eg&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/MahShaaban/pcr">https://github.com/MahShaaban/pcr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/MahShaaban/pcr/issues">https://github.com/MahShaaban/pcr/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, covr, cowplot</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-04-01 05:30:14 UTC; rstudio</td>
</tr>
<tr>
<td>Author:</td>
<td>Mahmoud Ahmed <a href="https://orcid.org/0000-0002-4377-6541"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-04-01 06:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='.pcr_amount'>Calculate the amounts</h2><span id='topic+.pcr_amount'></span>

<h3>Description</h3>

<p>Calculate the amounts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_amount(vec, a, b)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_amount_+3A_vec">vec</code></td>
<td>
<p>A vector of numerics</p>
</td></tr>
<tr><td><code id=".pcr_amount_+3A_a">a</code></td>
<td>
<p>A numeric</p>
</td></tr>
<tr><td><code id=".pcr_amount_+3A_b">b</code></td>
<td>
<p>A numeric</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of numerics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vec &lt;- rnorm(6, 30, 1)
pcr:::.pcr_amount(vec, 1, 1)
</code></pre>

<hr>
<h2 id='.pcr_average'>Get vector average by a variable</h2><span id='topic+.pcr_average'></span>

<h3>Description</h3>

<p>Get vector average by a variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_average(vec, var)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_average_+3A_vec">vec</code></td>
<td>
<p>A vector of numerics</p>
</td></tr>
<tr><td><code id=".pcr_average_+3A_var">var</code></td>
<td>
<p>A grouping variable</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of numerics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vec &lt;- rnorm(6, 30, 1)
var &lt;- rep(c('group1', 'group2'), 3)
pcr:::.pcr_average(vec, var)

</code></pre>

<hr>
<h2 id='.pcr_cv'>Get vector coefficient of variance by a variable</h2><span id='topic+.pcr_cv'></span>

<h3>Description</h3>

<p>Get vector coefficient of variance by a variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_cv(vec, var)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_cv_+3A_vec">vec</code></td>
<td>
<p>A vector of numerics</p>
</td></tr>
<tr><td><code id=".pcr_cv_+3A_var">var</code></td>
<td>
<p>A grouping variable</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of numerics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vec &lt;- rnorm(6, 30, 1)
var &lt;- rep(c('group1', 'group2'), 3)
pcr:::.pcr_cv(vec, var)

</code></pre>

<hr>
<h2 id='.pcr_error'>Propage two vectors</h2><span id='topic+.pcr_error'></span>

<h3>Description</h3>

<p>Propage two vectors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_error(vec, ref)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_error_+3A_vec">vec</code></td>
<td>
<p>A vector of numerics</p>
</td></tr>
<tr><td><code id=".pcr_error_+3A_ref">ref</code></td>
<td>
<p>A numeric vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of numerics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vec &lt;- rnorm(6, 30, 1)
ref &lt;- rnorm(6, 30, .1)
pcr:::.pcr_error(vec, ref)
</code></pre>

<hr>
<h2 id='.pcr_intercept'>Calculate the intercept of a line</h2><span id='topic+.pcr_intercept'></span>

<h3>Description</h3>

<p>Calculate the intercept of a line
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_intercept(vec, var)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_intercept_+3A_vec">vec</code></td>
<td>
<p>A vector of numerics</p>
</td></tr>
<tr><td><code id=".pcr_intercept_+3A_var">var</code></td>
<td>
<p>A grouping variable</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vec &lt;- rnorm(6, 30, 1)
var &lt;- rep(c(.1, .5), 3)
pcr:::.pcr_intercept(vec, var)

</code></pre>

<hr>
<h2 id='.pcr_normalize'>Normalize vector by another</h2><span id='topic+.pcr_normalize'></span>

<h3>Description</h3>

<p>Normalize vector by another
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_normalize(vec, ref, mode = "subtract")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_normalize_+3A_vec">vec</code></td>
<td>
<p>A vector of numerics</p>
</td></tr>
<tr><td><code id=".pcr_normalize_+3A_ref">ref</code></td>
<td>
<p>A numeric vector</p>
</td></tr>
<tr><td><code id=".pcr_normalize_+3A_mode">mode</code></td>
<td>
<p>Either 'subtract' or 'divide'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of numerics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vec &lt;- rnorm(6, 30, 1)
ref &lt;- rnorm(6, 30, .1)
pcr:::.pcr_normalize(vec, ref)
pcr:::.pcr_normalize(vec, ref, mode = 'divide')

</code></pre>

<hr>
<h2 id='.pcr_plot_analyze'>Plotting function</h2><span id='topic+.pcr_plot_analyze'></span>

<h3>Description</h3>

<p>Plotting function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_plot_analyze(df, method, facets = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_plot_analyze_+3A_df">df</code></td>
<td>
<p>A data.frame such as this returned by <a href="#topic+pcr_analyze">pcr_analyze</a></p>
</td></tr>
<tr><td><code id=".pcr_plot_analyze_+3A_method">method</code></td>
<td>
<p>A character string. Possible input includes 'delta_delta_ct',
'delta_ct' or 'relative_curve'</p>
</td></tr>
<tr><td><code id=".pcr_plot_analyze_+3A_facets">facets</code></td>
<td>
<p>A logical of whether or not to use facets when applicable</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot object. A bar graph of caculated ge
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## locate and read raw ct data
fl &lt;- system.file('extdata', 'ct1.csv', package = 'pcr')
ct1 &lt;- read.csv(fl)

# add grouping variable
group_var &lt;- rep(c('brain', 'kidney'), each = 6)

# calculate all delta_delta_ct model
df &lt;- pcr_ddct(ct1,
               group_var = group_var,
               reference_gene = 'GAPDH',
               reference_group = 'brain')

# make a plot
pcr:::.pcr_plot_analyze(df, method = 'delta_delta_ct')

# make a data.frame of two identical columns
pcr_hk &lt;- data.frame(
  GAPDH1 = ct1$GAPDH,
  GAPDH2 = ct1$GAPDH
  )

# calculate delta_ct model
df &lt;- pcr_dct(pcr_hk,
              group_var = group_var,
              reference_group = 'brain')

# make a plot
pcr:::.pcr_plot_analyze(df, method = 'delta_ct')
pcr:::.pcr_plot_analyze(df, method = 'delta_ct', facet = TRUE)

# calculate curve
# locate and read data
fl &lt;- system.file('extdata', 'ct3.csv', package = 'pcr')
ct3 &lt;- read.csv(fl)

# make a vector of RNA amounts
amount &lt;- rep(c(1, .5, .2, .1, .05, .02, .01), each = 3)

standard_curve &lt;- pcr_assess(ct3,
                             amount = amount,
                             method = 'standard_curve')
intercept &lt;- standard_curve$intercept
slope &lt;- standard_curve$slope

# calculate the rellative_curve model
df &lt;- pcr_curve(ct1,
                group_var = group_var,
                reference_gene = 'GAPDH',
                reference_group = 'brain',
                intercept = intercept,
                slope = slope)

# make a plot
pcr:::.pcr_plot_analyze(df, method = 'relative_curve')

</code></pre>

<hr>
<h2 id='.pcr_plot_assess'>Plot quality assessment graphs</h2><span id='topic+.pcr_plot_assess'></span>

<h3>Description</h3>

<p>Plot quality assessment graphs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_plot_assess(df, amount, reference_gene, method)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_plot_assess_+3A_df">df</code></td>
<td>
<p>A data.frame</p>
</td></tr>
<tr><td><code id=".pcr_plot_assess_+3A_amount">amount</code></td>
<td>
<p>A numeric vector</p>
</td></tr>
<tr><td><code id=".pcr_plot_assess_+3A_reference_gene">reference_gene</code></td>
<td>
<p>A character string</p>
</td></tr>
<tr><td><code id=".pcr_plot_assess_+3A_method">method</code></td>
<td>
<p>A character string; 'efficiency' or 'standard_curve'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'># locate and read file
fl &lt;- system.file('extdata', 'ct3.csv', package = 'pcr')
ct3 &lt;- read.csv(fl)

# make amount/dilution variable
amount &lt;- rep(c(1, .5, .2, .1, .05, .02, .01), each = 3)

# plot the standard curves
pcr:::.pcr_plot_assess(ct3,
                amount = amount,
                reference_gene = 'GAPDH',
                method = 'standard_curve')

# plot amplification efficiency
pcr:::.pcr_plot_assess(ct3,
                       amount = amount,
                       reference_gene = 'GAPDH',
                       method = 'efficiency')

</code></pre>

<hr>
<h2 id='.pcr_relative'>Raise two to a vector power</h2><span id='topic+.pcr_relative'></span>

<h3>Description</h3>

<p>Raise two to a vector power
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_relative(vec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_relative_+3A_vec">vec</code></td>
<td>
<p>A vector of numerics</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vec &lt;- rnorm(6, 30, 1)
pcr:::.pcr_relative(vec)
</code></pre>

<hr>
<h2 id='.pcr_rsquared'>Calculate R squared</h2><span id='topic+.pcr_rsquared'></span>

<h3>Description</h3>

<p>Calculate R squared
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_rsquared(vec, var)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_rsquared_+3A_vec">vec</code></td>
<td>
<p>A vector of numerics</p>
</td></tr>
<tr><td><code id=".pcr_rsquared_+3A_var">var</code></td>
<td>
<p>A grouping variable</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vec &lt;- rnorm(6, 30, 1)
var &lt;- rep(c(.1, .5), 3)
pcr:::.pcr_rsquared(vec, var)

</code></pre>

<hr>
<h2 id='.pcr_sd'>Get vector standard deviation by a variable</h2><span id='topic+.pcr_sd'></span>

<h3>Description</h3>

<p>Get vector standard deviation by a variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_sd(vec, var)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_sd_+3A_vec">vec</code></td>
<td>
<p>A vector of numerics</p>
</td></tr>
<tr><td><code id=".pcr_sd_+3A_var">var</code></td>
<td>
<p>A grouping variable</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of numerics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vec &lt;- rnorm(6, 30, 1)
var &lt;- rep(c('group1', 'group2'), 3)
pcr:::.pcr_sd(vec, var)

</code></pre>

<hr>
<h2 id='.pcr_slope'>Calculate the slope of a line</h2><span id='topic+.pcr_slope'></span>

<h3>Description</h3>

<p>Calculate the slope of a line
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.pcr_slope(vec, var)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".pcr_slope_+3A_vec">vec</code></td>
<td>
<p>A vector of numerics</p>
</td></tr>
<tr><td><code id=".pcr_slope_+3A_var">var</code></td>
<td>
<p>A grouping variable</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vec &lt;- rnorm(6, 30, 1)
var &lt;- rep(c(.1, .5), 3)
pcr:::.pcr_slope(vec, var)

</code></pre>

<hr>
<h2 id='ct1'><code class="reqn">C_T</code> values from qPCR (separate tubes)</h2><span id='topic+ct1'></span>

<h3>Description</h3>

<p>A dataset containing the <code class="reqn">C_T</code> values of two genes from a qPCR experiment.
Samples were prepared from human tissues; Brain and kidney (n = 6) each.
Primers for each genes were run in separate reaction tubes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ct1
</code></pre>


<h3>Format</h3>

<p>A data.frame with 12 rows and 2 variables:
</p>

<dl>
<dt>c_myc</dt><dd><p><code class="reqn">C_T</code> values of the target gene <strong>c-myc</strong></p>
</dd>
<dt>GAPDH</dt><dd><p><code class="reqn">C_T</code> values of the control gene <strong>GAPDH</strong></p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="http://www3.appliedbiosystems.com/cms/groups/mcb_support/documents/generaldocuments/cms_040980.pdf">http://www3.appliedbiosystems.com/cms/groups/mcb_support/documents/generaldocuments/cms_040980.pdf</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ct2">ct2</a></code>
</p>
<p><code><a href="#topic+ct3">ct3</a></code>
</p>

<hr>
<h2 id='ct2'><code class="reqn">C_T</code> values from qPCR (same tubes)</h2><span id='topic+ct2'></span>

<h3>Description</h3>

<p>A dataset containing the <code class="reqn">C_T</code> values of two genes from a qPCR experiment.
Samples were prepared from human tissues; Brain and kidney (n = 6) each.
Primers for both genes were run in the same tubes with different reporting dyes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ct2
</code></pre>


<h3>Format</h3>

<p>A data.frame with 12 rows and 2 variables:
</p>

<dl>
<dt>c_myc</dt><dd><p><code class="reqn">C_T</code> values of the target gene <strong>c-myc</strong></p>
</dd>
<dt>GAPDH</dt><dd><p><code class="reqn">C_T</code> values of the control gene <strong>GAPDH</strong></p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="http://www3.appliedbiosystems.com/cms/groups/mcb_support/documents/generaldocuments/cms_040980.pdf">http://www3.appliedbiosystems.com/cms/groups/mcb_support/documents/generaldocuments/cms_040980.pdf</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ct1">ct1</a></code>
</p>
<p><code><a href="#topic+ct3">ct3</a></code>
</p>

<hr>
<h2 id='ct3'><code class="reqn">C_T</code> values from qPCR (Serial dilutions)</h2><span id='topic+ct3'></span>

<h3>Description</h3>

<p>A dataset containing the <code class="reqn">C_T</code> values of two genes from a serial dilution
qPCR experiment. The original dataset shows only the averages and standard
deviations of each of the 7 different dilutions (1, .5, .2, .1, .05, .02 and .01).
These summaries were used to regenerate 3 replicates for each of the dilutions
to be used in testing and examples of the different functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ct3
</code></pre>


<h3>Format</h3>

<p>A data.frame with 21 rows and 2 variables:
</p>

<dl>
<dt>c_myc</dt><dd><p><code class="reqn">C_T</code> values of the target gene <strong>c-myc</strong></p>
</dd>
<dt>GAPDH</dt><dd><p><code class="reqn">C_T</code> values of the control gene <strong>GAPDH</strong></p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="http://www3.appliedbiosystems.com/cms/groups/mcb_support/documents/generaldocuments/cms_040980.pdf">http://www3.appliedbiosystems.com/cms/groups/mcb_support/documents/generaldocuments/cms_040980.pdf</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ct1">ct1</a></code>
</p>
<p><code><a href="#topic+ct2">ct2</a></code>
</p>

<hr>
<h2 id='ct4'><code class="reqn">C_T</code> values from qPCR (Serial dilutions)</h2><span id='topic+ct4'></span>

<h3>Description</h3>

<p>A dataset containing the <code class="reqn">C_T</code> values of two genes from a controlled serial
dilution qPCR experiment. The data were prepared from four different dilutions
(10, 2, 0.4 and 0.08) and two control groups; control and treatment (n = 12) each.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ct4
</code></pre>


<h3>Format</h3>

<p>A data.frame with 24 rows and 2 variables:
</p>

<dl>
<dt>ref</dt><dd><p><code class="reqn">C_T</code> values of the reference gene</p>
</dd>
<dt>target</dt><dd><p><code class="reqn">C_T</code> values of the target gene</p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1395339/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1395339/</a>
</p>

<hr>
<h2 id='pcr'><code>pcr</code> package</h2><span id='topic+pcr'></span>

<h3>Description</h3>

<p>Analyzing real-time quantitative PCR data
</p>


<h3>Details</h3>

<p>Calculates the amplification efficiency and curves from real-time
quantitative PCR (Polymerase Chain Reaction) data. Estimates the relative
expression from PCR data using the double delta CT and the standard curve
methods Livak &amp; Schmittgen (2001) &lt;doi:10.1006/meth.2001.1262&gt;. Tests for
statistical significance using two-group tests and linear regression
Yuan et al. (2006) &lt;doi: 10.1186/1471-2105-7-85&gt;.
</p>

<hr>
<h2 id='pcr_analyze'>Apply qPCR analysis methods</h2><span id='topic+pcr_analyze'></span>

<h3>Description</h3>

<p>A unified interface to invoke different analysis methods of qPCR data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcr_analyze(df, method = "delta_delta_ct", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcr_analyze_+3A_df">df</code></td>
<td>
<p>A data.frame of <code class="reqn">C_T</code> values with genes in the columns and
samples in rows rows</p>
</td></tr>
<tr><td><code id="pcr_analyze_+3A_method">method</code></td>
<td>
<p>A character string; 'delta_delta_ct' default, 'delta_ct' or
'relative_curve' for invoking a certain analysis model</p>
</td></tr>
<tr><td><code id="pcr_analyze_+3A_...">...</code></td>
<td>
<p>Arguments passed to the methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The different analysis methods can be invoked using the
argument method with 'delta_delta_ct' default, 'delta_ct' or
'relative_curve' for the double delta <code class="reqn">C_T</code>, delta ct or the standard curve
model respectively. Alternatively, the same methods can be applied by using
the corresponding functions directly: <a href="#topic+pcr_ddct">pcr_ddct</a>, <a href="#topic+pcr_dct">pcr_dct</a> or
<a href="#topic+pcr_curve">pcr_curve</a>
</p>


<h3>Value</h3>

<p>A data.frame by default, when <code>plot</code> is TRUE returns a plot.
For details; <a href="#topic+pcr_ddct">pcr_ddct</a>, <a href="#topic+pcr_dct">pcr_dct</a> and <a href="#topic+pcr_curve">pcr_curve</a>.
</p>


<h3>References</h3>

<p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. “Analysis of
Relative Gene Expression Data Using Real-Time Quantitative PCR and the
Double Delta CT Method.” Methods 25 (4). ELSEVIER.
doi:10.1006/meth.2001.1262.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># applying the delta delta ct method
## locate and read raw ct data
fl &lt;- system.file('extdata', 'ct1.csv', package = 'pcr')
ct1 &lt;- read.csv(fl)

# add grouping variable
group_var &lt;- rep(c('brain', 'kidney'), each = 6)

# calculate all values and errors in one step
pcr_analyze(ct1,
            group_var = group_var,
            reference_gene = 'GAPDH',
            reference_group = 'brain',
            method = 'delta_delta_ct')

# return a plot
pcr_analyze(ct1,
            group_var = group_var,
            reference_gene = 'GAPDH',
            reference_group = 'brain',
            method = 'delta_delta_ct',
            plot = TRUE)

# applying the delta ct method
# make a data.frame of two identical columns
pcr_hk &lt;- data.frame(
  GAPDH1 = ct1$GAPDH,
  GAPDH2 = ct1$GAPDH
  )

# calculate fold change
pcr_analyze(pcr_hk,
            group_var = group_var,
            reference_group = 'brain',
            method = 'delta_ct')

# return a plot
pcr_analyze(pcr_hk,
            group_var = group_var,
            reference_group = 'brain',
            method = 'delta_ct',
            plot = TRUE)

# applying the standard curve method
# locate and read file
fl &lt;- system.file('extdata', 'ct3.csv', package = 'pcr')
ct3 &lt;- read.csv(fl)

# make a vector of RNA amounts
amount &lt;- rep(c(1, .5, .2, .1, .05, .02, .01), each = 3)

# calculate curve
standard_curve &lt;- pcr_assess(ct3, amount = amount, method = 'standard_curve')
intercept &lt;- standard_curve$intercept
slope &lt;- standard_curve$slope

# apply the standard curve method
pcr_analyze(ct1,
           group_var = group_var,
           reference_gene = 'GAPDH',
           reference_group = 'brain',
           intercept = intercept,
           slope = slope,
           method = 'relative_curve')

# return a plot
pcr_analyze(ct1,
           group_var = group_var,
           reference_gene = 'GAPDH',
           reference_group = 'brain',
           intercept = intercept,
           slope = slope,
           method = 'relative_curve',
           plot = TRUE)

</code></pre>

<hr>
<h2 id='pcr_assess'>Assess qPCR data quality</h2><span id='topic+pcr_assess'></span>

<h3>Description</h3>

<p>A unified interface to invoke different quality assessment methods of qPCR
data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcr_assess(df, method = "standard_curve", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcr_assess_+3A_df">df</code></td>
<td>
<p>A data.frame of <code class="reqn">C_T</code> values with genes in the columns and
samples in rows rows. Each sample are replicates of a known input/dilution
given by amount</p>
</td></tr>
<tr><td><code id="pcr_assess_+3A_method">method</code></td>
<td>
<p>A character string; 'standard_curve' (default) or 'efficiency'
for invoking a certain quality assessment model</p>
</td></tr>
<tr><td><code id="pcr_assess_+3A_...">...</code></td>
<td>
<p>Arguments passed to the methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The different quality assessment methods can be invoked using the
argument method with 'standard_curve' or 'efficiency'. Alternatively, the
same methods can be applied by using the corresponding functions:
<a href="#topic+pcr_standard">pcr_standard</a> or <a href="#topic+pcr_efficiency">pcr_efficiency</a> for calculating the
amplification efficiency of a PCR reaction or the individual standard
curves respectively. Unlike the amplification efficiency calculation when,
using the double delta ct model, the standard curves are required in
calculating the standard curve analysis model.
</p>


<h3>Value</h3>

<p>A data.frame or a plot. For details; <a href="#topic+pcr_standard">pcr_standard</a> and
<a href="#topic+pcr_efficiency">pcr_efficiency</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#' # locate and read file
fl &lt;- system.file('extdata', 'ct3.csv', package = 'pcr')
ct3 &lt;- read.csv(fl)

# make amount/dilution variable
amount &lt;- rep(c(1, .5, .2, .1, .05, .02, .01), each = 3)

# calculate the standard curve
pcr_assess(ct3,
           amount = amount,
           method = 'standard_curve')

# retrun a plot
pcr_assess(ct3,
           amount = amount,
           method = 'standard_curve',
           plot = TRUE)

# calculate amplification efficiency
pcr_assess(ct3,
           amount = amount,
           reference_gene = 'GAPDH',
           method = 'efficiency')

# return a plot
pcr_assess(ct3,
           amount = amount,
           reference_gene = 'GAPDH',
           method = 'efficiency',
           plot = TRUE)

</code></pre>

<hr>
<h2 id='pcr_curve'>Calculate the standard curve model</h2><span id='topic+pcr_curve'></span>

<h3>Description</h3>

<p>Uses the <code class="reqn">C_T</code> values and a reference gene and a group, in addition to the
intercept and slope of each gene form a serial dilution experiment, to calculate
the standard curve model and estimate the normalized relative expression of the
target genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcr_curve(
  df,
  group_var,
  reference_gene,
  reference_group,
  mode = "separate_tube",
  intercept,
  slope,
  plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcr_curve_+3A_df">df</code></td>
<td>
<p>A data.frame of <code class="reqn">C_T</code> values with genes in the columns and
samples in rows rows</p>
</td></tr>
<tr><td><code id="pcr_curve_+3A_group_var">group_var</code></td>
<td>
<p>A character vector of a grouping variable. The length of
this variable should equal the number of rows of df</p>
</td></tr>
<tr><td><code id="pcr_curve_+3A_reference_gene">reference_gene</code></td>
<td>
<p>A character string of the column name of a control
gene</p>
</td></tr>
<tr><td><code id="pcr_curve_+3A_reference_group">reference_group</code></td>
<td>
<p>A character string of the control group in group_var</p>
</td></tr>
<tr><td><code id="pcr_curve_+3A_mode">mode</code></td>
<td>
<p>A character string of; 'separate_tube' (default) or 'same_tube'.
This is to indicate whether the different genes were run in separate or the
same PCR tube</p>
</td></tr>
<tr><td><code id="pcr_curve_+3A_intercept">intercept</code></td>
<td>
<p>A numeric vector of intercept and length equals the number of genes</p>
</td></tr>
<tr><td><code id="pcr_curve_+3A_slope">slope</code></td>
<td>
<p>A numeric vector of slopes length equals the number of genes</p>
</td></tr>
<tr><td><code id="pcr_curve_+3A_plot">plot</code></td>
<td>
<p>A logical (default is FALSE)</p>
</td></tr>
<tr><td><code id="pcr_curve_+3A_...">...</code></td>
<td>
<p>Arguments passed to customize plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>this model doesn't assume perfect amplification but rather actively
use the amplification in calculating the relative expression. So when the
amplification efficiency of all genes are 100% both methods should give
similar results. The standard curve method is applied using two steps.
First, serial dilutions of the mRNAs from the samples of interest are used
as input to the PCR reaction. The linear trend of the log input amount and
the resulting <code class="reqn">C_T</code> values for each gene are used to calculate an intercept
and a slope. Secondly, these intercepts and slopes are used to calculate the
amounts of mRNA of the genes of interest and the control/reference in the
samples of interest and the control sample/reference. These amounts are
finally used to calculate the relative expression.
</p>


<h3>Value</h3>

<p>A data.frame of 7 columns
</p>

<ul>
<li><p> group The unique entries in group_var
</p>
</li>
<li><p> gene The column names of df
</p>
</li>
<li><p> normalized The normalized expression of target genes relative to a reference_gene
</p>
</li>
<li><p> calibrated The calibrated expression of target genes relative to a reference_group
</p>
</li>
<li><p> error The standard deviation of normalized relative expression
</p>
</li>
<li><p> lower The lower interval of the normalized relative expression
</p>
</li>
<li><p> upper The upper interval of the normalized relative expression
</p>
</li></ul>

<p>When <code>plot</code> is TRUE, returns a bar graph of the calibrated expression
of the genes in the column and the groups in the column group. Error bars
are drawn using the columns lower and upper. When more one gene are plotted
the default in dodge bars. When the argument facet is TRUE a separate
panel is drawn for each gene.
</p>


<h3>References</h3>

<p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. “Analysis of
Relative Gene Expression Data Using Real-Time Quantitative PCR and the
Double Delta CT Method.” Methods 25 (4). ELSEVIER.
doi:10.1006/meth.2001.1262.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># locate and read file
fl &lt;- system.file('extdata', 'ct3.csv', package = 'pcr')
ct3 &lt;- read.csv(fl)

fl &lt;- system.file('extdata', 'ct1.csv', package = 'pcr')
ct1 &lt;- read.csv(fl)

# make a vector of RNA amounts
amount &lt;- rep(c(1, .5, .2, .1, .05, .02, .01), each = 3)

# calculate curve
standard_curve &lt;- pcr_assess(ct3, amount = amount, method = 'standard_curve')
intercept &lt;- standard_curve$intercept
slope &lt;- standard_curve$slope

# make grouping variable
group &lt;- rep(c('brain', 'kidney'), each = 6)

# apply the standard curve method
pcr_curve(ct1,
          group_var = group,
          reference_gene = 'GAPDH',
          reference_group = 'brain',
          intercept = intercept,
          slope = slope)

# returns a plot
pcr_curve(ct1,
          group_var = group,
          reference_gene = 'GAPDH',
          reference_group = 'brain',
          intercept = intercept,
          slope = slope,
          plot = TRUE)

</code></pre>

<hr>
<h2 id='pcr_dct'>Calculate the delta_ct model</h2><span id='topic+pcr_dct'></span>

<h3>Description</h3>

<p>Uses the <code class="reqn">C_T</code> values and a reference group to calculate the delta <code class="reqn">C_T</code>
model to estimate the relative fold change of a gene between groups
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcr_dct(
  df,
  group_var,
  reference_group,
  mode = "separate_tube",
  plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcr_dct_+3A_df">df</code></td>
<td>
<p>A data.frame of <code class="reqn">C_T</code> values with genes in the columns and
samples in rows rows</p>
</td></tr>
<tr><td><code id="pcr_dct_+3A_group_var">group_var</code></td>
<td>
<p>A character vector of a grouping variable. The length of
this variable should equal the number of rows of df</p>
</td></tr>
<tr><td><code id="pcr_dct_+3A_reference_group">reference_group</code></td>
<td>
<p>A character string of the control group in group_var</p>
</td></tr>
<tr><td><code id="pcr_dct_+3A_mode">mode</code></td>
<td>
<p>A character string of; 'separate_tube' (default) or 'same_tube'.
This is to indicate whether the different genes were run in separate or the
same PCR tube</p>
</td></tr>
<tr><td><code id="pcr_dct_+3A_plot">plot</code></td>
<td>
<p>A logical (default is FALSE)</p>
</td></tr>
<tr><td><code id="pcr_dct_+3A_...">...</code></td>
<td>
<p>Arguments passed to customize plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method is a variation of the double delta <code class="reqn">C_T</code> model,
<code><a href="#topic+pcr_ddct">pcr_ddct</a></code>. It can be used to calculate the fold change
of in one sample relative to the others. For example, it can be used to
compare and choosing a control/reference genes.
</p>


<h3>Value</h3>

<p>A data.frame of 7 columns
</p>

<ul>
<li><p> group The unique entries in group_var
</p>
</li>
<li><p> gene The column names of df
</p>
</li>
<li><p> calibrated The average <code class="reqn">C_T</code> value of target genes after
subtracting that of the reference_group
</p>
</li>
<li><p> fold_change The fold change of genes relative to a reference_group
</p>
</li>
<li><p> error The standard deviation of the fold_change
</p>
</li>
<li><p> lower The lower interval of the fold_change
</p>
</li>
<li><p> upper The upper interval of the fold_change
</p>
</li></ul>

<p>When <code>plot</code> is TRUE, returns a bar graph of the fold change of
the genes in the column and the groups in the column group. Error bars are
drawn using the columns lower and upper. When more one gene are plotted the
default in dodge bars. When the argument facet is TRUE a separate panel is
drawn for each gene.
</p>


<h3>References</h3>

<p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. “Analysis of
Relative Gene Expression Data Using Real-Time Quantitative PCR and the
Double Delta CT Method.” Methods 25 (4). ELSEVIER.
doi:10.1006/meth.2001.1262.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># locate and read file
fl &lt;- system.file('extdata', 'ct1.csv', package = 'pcr')
ct1 &lt;- read.csv(fl)

# make a data.frame of two identical columns
pcr_hk &lt;- data.frame(
  GAPDH1 = ct1$GAPDH,
  GAPDH2 = ct1$GAPDH
  )

# add grouping variable
group_var &lt;- rep(c('brain', 'kidney'), each = 6)

# calculate caliberation
pcr_dct(pcr_hk,
        group_var = group_var,
        reference_group = 'brain')

# returns a plot
pcr_dct(pcr_hk,
        group_var = group_var,
        reference_group = 'brain',
        plot = TRUE)

# returns a plot with facets
pcr_dct(pcr_hk,
        group_var = group_var,
        reference_group = 'brain',
        plot = TRUE,
        facet = TRUE)

</code></pre>

<hr>
<h2 id='pcr_ddct'>Calculate the delta_delta_ct model</h2><span id='topic+pcr_ddct'></span>

<h3>Description</h3>

<p>Uses the <code class="reqn">C_T</code> values and a reference gene and a group to calculate the
delta delta <code class="reqn">C_T</code> model to estimate the normalized relative expression
of target genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcr_ddct(
  df,
  group_var,
  reference_gene,
  reference_group,
  mode = "separate_tube",
  plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcr_ddct_+3A_df">df</code></td>
<td>
<p>A data.frame of <code class="reqn">C_T</code> values with genes in the columns and
samples in rows rows</p>
</td></tr>
<tr><td><code id="pcr_ddct_+3A_group_var">group_var</code></td>
<td>
<p>A character vector of a grouping variable. The length of
this variable should equal the number of rows of df</p>
</td></tr>
<tr><td><code id="pcr_ddct_+3A_reference_gene">reference_gene</code></td>
<td>
<p>A character string of the column name of a control
gene</p>
</td></tr>
<tr><td><code id="pcr_ddct_+3A_reference_group">reference_group</code></td>
<td>
<p>A character string of the control group in group_var</p>
</td></tr>
<tr><td><code id="pcr_ddct_+3A_mode">mode</code></td>
<td>
<p>A character string of; 'separate_tube' (default) or 'same_tube'.
This is to indicate whether the different genes were run in separate or the
same PCR tube</p>
</td></tr>
<tr><td><code id="pcr_ddct_+3A_plot">plot</code></td>
<td>
<p>A logical (default is FALSE)</p>
</td></tr>
<tr><td><code id="pcr_ddct_+3A_...">...</code></td>
<td>
<p>Arguments passed to customize plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The comparative <code class="reqn">C_T</code> methods assume that the cDNA templates of
the gene/s of interest as well as the control/reference gene have similar
amplification efficiency. And that this amplification efficiency is near
perfect. Meaning, at a certain threshold during the linear portion of the
PCR reaction, the amount of the gene of the interest and the control double
each cycle. Another assumptions is that, the expression difference between
two genes or two samples can be captured by subtracting one (gene or sample
of interest) from another (reference).  This final assumption requires also
that these references don't change with the treatment or the course in
question.
</p>


<h3>Value</h3>

<p>A data.frame of 8 columns:
</p>

<ul>
<li><p> group The unique entries in group_var
</p>
</li>
<li><p> gene The column names of df. reference_gene is dropped
</p>
</li>
<li><p> normalized The <code class="reqn">C_T</code> value (or the average <code class="reqn">C_T</code> value) of
target genes after subtracting that of the reference_gene
</p>
</li>
<li><p> calibrated The normalized average <code class="reqn">C_T</code> value of target genes
after subtracting that of the reference_group
</p>
</li>
<li><p> relative_expression The expression of target genes normalized by
a reference_gene and calibrated by a reference_group
</p>
</li>
<li><p> error The standard deviation of the relative_expression
</p>
</li>
<li><p> lower The lower interval of the relative_expression
</p>
</li>
<li><p> upper The upper interval of the relative_expression
</p>
</li></ul>

<p>When <code>plot</code> is TRUE, returns a bar graph of the relative expression of
the genes in the column and the groups in the column group. Error bars are
drawn using the columns lower and upper. When more one gene are plotted the
default in dodge bars. When the argument facet is TRUE a separate panel is
drawn for each gene.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## locate and read raw ct data
fl &lt;- system.file('extdata', 'ct1.csv', package = 'pcr')
ct1 &lt;- read.csv(fl)

# add grouping variable
group_var &lt;- rep(c('brain', 'kidney'), each = 6)

# calculate all values and errors in one step
pcr_ddct(ct1,
         group_var = group_var,
         reference_gene = 'GAPDH',
         reference_group = 'brain')

# return a plot
pcr_ddct(ct1,
         group_var = group_var,
         reference_gene = 'GAPDH',
         reference_group = 'brain',
         plot = TRUE)

</code></pre>

<hr>
<h2 id='pcr_efficiency'>Calculate amplification efficiency</h2><span id='topic+pcr_efficiency'></span>

<h3>Description</h3>

<p>Uses the <code class="reqn">C_T</code> values from a serial dilution experiment to calculate the
amplification efficiency of a PCR reaction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcr_efficiency(df, amount, reference_gene, plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcr_efficiency_+3A_df">df</code></td>
<td>
<p>A data.frame of <code class="reqn">C_T</code> values with genes in the columns and
samples in rows rows. Each sample are replicates of a known input/dilution
given by amount</p>
</td></tr>
<tr><td><code id="pcr_efficiency_+3A_amount">amount</code></td>
<td>
<p>A numeric vector of the input amounts or dilutions. The length
of this vector should equal the row number of df</p>
</td></tr>
<tr><td><code id="pcr_efficiency_+3A_reference_gene">reference_gene</code></td>
<td>
<p>A character string of the column name of a control
gene</p>
</td></tr>
<tr><td><code id="pcr_efficiency_+3A_plot">plot</code></td>
<td>
<p>A logical (default FALSE) to indicate whether to return a
data.frame or a plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fortunately, regardless of the method used in the analysis of qPCR
data, The quality assessment are done in a similar way. It requires an
experiment similar to that of calculating the standard curve. Serial
dilutions of the genes of interest and controls are used as input to the
reaction and different calculations are made. The amplification efficiency
is approximated be the linear trend between the difference between the
<code class="reqn">C_T</code> value of a gene of interest and a control/reference
(<code class="reqn">\Delta C_T</code>) and the log input amount. This piece of information is
required when using the <code class="reqn">\Delta \Delta C_T</code> model. Typically, the slope
of the curve should be very small and the <code class="reqn">R^2</code> value should be very
close to one. Other analysis methods are recommended when this is not the
case.
</p>


<h3>Value</h3>

<p>When plot is FALSE returns a data.frame of 4 columns describing the
line between the <code class="reqn">\Delta C_T</code> of target genes and the log of amount
</p>

<ul>
<li><p> gene The column names of df. reference_gene is dropped
</p>
</li>
<li><p> intercept The intercept of the line
</p>
</li>
<li><p> slope The slope of the line
</p>
</li>
<li><p> r_squared The squared correlation
</p>
</li></ul>

<p>When plot is TRUE returns a graph instead shows the average and
standard deviation of of the <code class="reqn">\Delta C_T</code> at different input amounts.
In addition, a linear trend line is drawn.
</p>


<h3>References</h3>

<p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. “Analysis of
Relative Gene Expression Data Using Real-Time Quantitative PCR and the
Double Delta CT Method.” Methods 25 (4). ELSEVIER.
doi:10.1006/meth.2001.1262.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># locate and read file
fl &lt;- system.file('extdata', 'ct3.csv', package = 'pcr')
ct3 &lt;- read.csv(fl)

# make amount/dilution variable
amount &lt;- rep(c(1, .5, .2, .1, .05, .02, .01), each = 3)

# calculate amplification efficiency
pcr_efficiency(ct3,
               amount = amount,
               reference_gene = 'GAPDH')

# plot amplification efficiency
pcr_efficiency(ct3,
               amount = amount,
               reference_gene = 'GAPDH',
               plot = TRUE)

</code></pre>

<hr>
<h2 id='pcr_lm'>Linear regression qPCR data</h2><span id='topic+pcr_lm'></span>

<h3>Description</h3>

<p>Linear regression qPCR data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcr_lm(
  df,
  group_var,
  reference_gene,
  reference_group,
  model_matrix = NULL,
  mode = "subtract",
  tidy = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcr_lm_+3A_df">df</code></td>
<td>
<p>A data.frame of <code class="reqn">C_T</code> values with genes in the columns and
samples in rows rows</p>
</td></tr>
<tr><td><code id="pcr_lm_+3A_group_var">group_var</code></td>
<td>
<p>A character vector of a grouping variable. The length of
this variable should equal the number of rows of df</p>
</td></tr>
<tr><td><code id="pcr_lm_+3A_reference_gene">reference_gene</code></td>
<td>
<p>A character string of the column name of a control
gene</p>
</td></tr>
<tr><td><code id="pcr_lm_+3A_reference_group">reference_group</code></td>
<td>
<p>A character string of the control group in group_var</p>
</td></tr>
<tr><td><code id="pcr_lm_+3A_model_matrix">model_matrix</code></td>
<td>
<p>A model matrix for advanced experimental design. for
constructing such a matrix with different variables check
<code><a href="stats.html#topic+model.matrix">model.matrix</a></code></p>
</td></tr>
<tr><td><code id="pcr_lm_+3A_mode">mode</code></td>
<td>
<p>A character string for the normalization mode. Possible values
are &quot;subtract&quot; (default) or &quot;divide&quot;.</p>
</td></tr>
<tr><td><code id="pcr_lm_+3A_tidy">tidy</code></td>
<td>
<p>A <code>logical</code> whether to return a <code>list</code> of
<code><a href="stats.html#topic+lm">lm</a></code> or a tidy <code>data.frame</code>. Default TRUE.</p>
</td></tr>
<tr><td><code id="pcr_lm_+3A_...">...</code></td>
<td>
<p>Other arguments to <code><a href="stats.html#topic+lm">lm</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame of 6 columns
</p>

<ul>
<li><p> term The term being tested
</p>
</li>
<li><p> gene The column names of df. reference_gene is dropped
</p>
</li>
<li><p> estimate The estimate for each term
</p>
</li>
<li><p> p_value The p-value for each term
</p>
</li>
<li><p> lower The low 95% confidence interval
</p>
</li>
<li><p> upper The high 95% confidence interval
</p>
</li></ul>

<p>When <code>tidy</code> is FALSE, returns a <code>list</code> of <code><a href="stats.html#topic+lm">lm</a></code>
objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># locate and read data
fl &lt;- system.file('extdata', 'ct4.csv', package = 'pcr')
ct4 &lt;- read.csv(fl)

# make group variable
group &lt;- rep(c('control', 'treatment'), each = 12)

# test
pcr_lm(ct4,
       group_var = group,
       reference_gene = 'ref',
       reference_group = 'control')

# testing using lm method
pcr_test(ct4,
         group_var = group,
         reference_gene = 'ref',
         reference_group = 'control',
         test = 'lm')

</code></pre>

<hr>
<h2 id='pcr_standard'>Calculate the standard curve</h2><span id='topic+pcr_standard'></span>

<h3>Description</h3>

<p>Uses the <code class="reqn">C_T</code> values from a serial dilution experiment to calculate the
a curve for each gene and the log of the input amount
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcr_standard(df, amount, plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcr_standard_+3A_df">df</code></td>
<td>
<p>A data.frame of <code class="reqn">C_T</code> values with genes in the columns and
samples in rows rows. Each sample are replicates of a known input/dilution
given by amount</p>
</td></tr>
<tr><td><code id="pcr_standard_+3A_amount">amount</code></td>
<td>
<p>A numeric vector of the input amounts or dilutions. The length
of this vector should equal the row number of df</p>
</td></tr>
<tr><td><code id="pcr_standard_+3A_plot">plot</code></td>
<td>
<p>A logical (default FALSE) to indicate whether to return a
data.frame or a plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fortunately, regardless of the method used in the analysis of qPCR
data, The quality assessment are done in a similar way. It requires an
experiment similar to that of calculating the standard curve. Serial
dilutions of the genes of interest and controls are used as input to
the reaction and different calculations are made.
Curves are required for each gene using the $C_T$ value and the log of the
input amount. In this case, a separate slope and intercept are required for
the calculation of the relative expression when applying the standard curve
model.
</p>


<h3>Value</h3>

<p>When plot is FALSE returns a data.frame of 4 columns describing the
line between the <code class="reqn">C_T</code> of each gene and the log of amount
</p>

<ul>
<li><p> gene The column names of df
</p>
</li>
<li><p> intercept The intercept of the line
</p>
</li>
<li><p> slope The slope of the line
</p>
</li>
<li><p> r_squared The squared correlation
</p>
</li></ul>

<p>When plot is TRUE returns a graph instead shows the average and
standard deviation of of the <code class="reqn">C_T</code> at different input amounts.
</p>


<h3>References</h3>

<p>Livak, Kenneth J, and Thomas D Schmittgen. 2001. “Analysis of
Relative Gene Expression Data Using Real-Time Quantitative PCR and the
Double Delta CT Method.” Methods 25 (4). ELSEVIER.
doi:10.1006/meth.2001.1262.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># locate and read file
fl &lt;- system.file('extdata', 'ct3.csv', package = 'pcr')
ct3 &lt;- read.csv(fl)

# make amount/dilution variable
amount &lt;- rep(c(1, .5, .2, .1, .05, .02, .01), each = 3)

# calculate the standard curve
pcr_standard(ct3,
             amount = amount)

# plot the standard curve
pcr_standard(ct3,
             amount = amount,
             plot = TRUE)

</code></pre>

<hr>
<h2 id='pcr_test'>Statistical testing of PCR data</h2><span id='topic+pcr_test'></span>

<h3>Description</h3>

<p>A unified interface to different statistical significance tests for qPCR
data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcr_test(df, test = "t.test", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcr_test_+3A_df">df</code></td>
<td>
<p>A data.frame of <code class="reqn">C_T</code> values with genes in the columns and
samples in rows rows</p>
</td></tr>
<tr><td><code id="pcr_test_+3A_test">test</code></td>
<td>
<p>A character string; 't.test' default, 'wilcox.test' or 'lm'</p>
</td></tr>
<tr><td><code id="pcr_test_+3A_...">...</code></td>
<td>
<p>Other arguments for the testing methods</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The simple t-test can be used to test the significance of the
difference between two conditions <code class="reqn">\Delta C_T</code>. t-test assumes in
addition, that the input <code class="reqn">C_T</code> values are normally distributed and the
variance between conditions are comparable. Wilcoxon test can be used when
sample size is small and those two last assumptions are hard to achieve.
</p>
<p>Two use the linear regression here. A null hypothesis is formulated as
following,
</p>
<p style="text-align: center;"><code class="reqn">
  C_{T, target, treatment} - C_{T, control, treatment} =
  C_{T, target, control} - C_{T, control, control}
  \quad \textrm{or} \quad  \Delta\Delta C_T
</code>
</p>

<p>This is exactly the <code class="reqn">\Delta\Delta C_T</code> as explained earlier. So the
<code class="reqn">\Delta\Delta C_T</code> is estimated and the null is rejected when
<code class="reqn">\Delta\Delta C_T \ne 0</code>.
</p>


<h3>Value</h3>

<p>A data.frame of 5 columns in addition to term when test == 'lm'
</p>

<ul>
<li><p> term The linear regression comparison terms
</p>
</li>
<li><p> gene The column names of df. reference_gene is dropped
</p>
</li>
<li><p> estimate The estimate for each term
</p>
</li>
<li><p> p_value The p-value for each term
</p>
</li>
<li><p> lower The low 95% confidence interval
</p>
</li>
<li><p> upper The high 95% confidence interval
</p>
</li></ul>

<p>For details about the test methods themselves and different parameters,
consult <code><a href="stats.html#topic+t.test">t.test</a></code>, <code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code>
and <code><a href="stats.html#topic+lm">lm</a></code>
</p>


<h3>References</h3>

<p>Yuan, Joshua S, Ann Reed, Feng Chen, and Neal Stewart. 2006.
“Statistical Analysis of Real-Time PCR Data.” BMC Bioinformatics 7 (85).
BioMed Central. doi:10.1186/1471-2105-7-85.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># locate and read data
fl &lt;- system.file('extdata', 'ct4.csv', package = 'pcr')
ct4 &lt;- read.csv(fl)

# make group variable
group &lt;- rep(c('control', 'treatment'), each = 12)

# test using t-test
pcr_test(ct4,
         group_var = group,
         reference_gene = 'ref',
         reference_group = 'control',
         test = 't.test')

# test using wilcox.test
pcr_test(ct4,
         group_var = group,
         reference_gene = 'ref',
         reference_group = 'control',
         test = 'wilcox.test')

# testing using lm
pcr_test(ct4,
         group_var = group,
         reference_gene = 'ref',
         reference_group = 'control',
         test = 'lm')

# testing advanced designs using a model matrix
# make a model matrix
group &lt;- relevel(factor(group), ref = 'control')
dose &lt;- rep(c(100, 80, 60, 40), each = 3, times = 2)
mm &lt;- model.matrix(~group:dose, data = data.frame(group, dose))

# test using lm
pcr_test(ct4,
         reference_gene = 'ref',
         model_matrix = mm,
         test = 'lm')

# using linear models to check the effect of RNA quality
# make a model matrix
group &lt;- relevel(factor(group), ref = 'control')
set.seed(1234)
quality &lt;- scale(rnorm(n = 24, mean = 1.9, sd = .1))
mm &lt;- model.matrix(~group + group:quality, data = data.frame(group, quality))

# testing using lm
pcr_test(ct4,
         reference_gene = 'ref',
         model_matrix = mm,
         test = 'lm')

# using linear model to check the effects of mixing separate runs
# make a model matrix
group &lt;- relevel(factor(group), ref = 'control')
run &lt;- factor(rep(c(1:3), 8))
mm &lt;- model.matrix(~group + group:run, data = data.frame(group, run))

# test using lm
pcr_test(ct4,
         reference_gene = 'ref',
         model_matrix = mm,
         test = 'lm')

</code></pre>

<hr>
<h2 id='pcr_ttest'>t-test qPCR data</h2><span id='topic+pcr_ttest'></span>

<h3>Description</h3>

<p>t-test qPCR data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcr_ttest(df, group_var, reference_gene, reference_group, tidy = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcr_ttest_+3A_df">df</code></td>
<td>
<p>A data.frame of <code class="reqn">C_T</code> values with genes in the columns and
samples in rows rows</p>
</td></tr>
<tr><td><code id="pcr_ttest_+3A_group_var">group_var</code></td>
<td>
<p>A character vector of a grouping variable. The length of
this variable should equal the number of rows of df</p>
</td></tr>
<tr><td><code id="pcr_ttest_+3A_reference_gene">reference_gene</code></td>
<td>
<p>A character string of the column name of a control
gene</p>
</td></tr>
<tr><td><code id="pcr_ttest_+3A_reference_group">reference_group</code></td>
<td>
<p>A character string of the control group in group_var</p>
</td></tr>
<tr><td><code id="pcr_ttest_+3A_tidy">tidy</code></td>
<td>
<p>A <code>logical</code> whether to return a <code>list</code> of <code>htest</code>
or a tidy <code>data.frame</code>. Default TRUE.</p>
</td></tr>
<tr><td><code id="pcr_ttest_+3A_...">...</code></td>
<td>
<p>Other arguments to <code><a href="stats.html#topic+t.test">t.test</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame of 5 columns
</p>

<ul>
<li><p> gene The column names of df. reference_gene is dropped
</p>
</li>
<li><p> estimate The estimate for each term
</p>
</li>
<li><p> p_value The p-value for each term
</p>
</li>
<li><p> lower The low 95% confidence interval
</p>
</li>
<li><p> upper The high 95% confidence interval
</p>
</li></ul>

<p>When <code>tidy</code> is FALSE, returns a <code>list</code> of <code>htest</code> objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># locate and read data
fl &lt;- system.file('extdata', 'ct4.csv', package = 'pcr')
ct4 &lt;- read.csv(fl)

# make group variable
group &lt;- rep(c('control', 'treatment'), each = 12)

# test
pcr_ttest(ct4,
          group_var = group,
          reference_gene = 'ref',
          reference_group = 'control')

# test using t.test method
pcr_test(ct4,
         group_var = group,
         reference_gene = 'ref',
         reference_group = 'control',
         test = 't.test')

</code></pre>

<hr>
<h2 id='pcr_wilcox'>Wilcoxon test qPCR data</h2><span id='topic+pcr_wilcox'></span>

<h3>Description</h3>

<p>Wilcoxon test qPCR data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pcr_wilcox(df, group_var, reference_gene, reference_group, tidy = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pcr_wilcox_+3A_df">df</code></td>
<td>
<p>A data.frame of <code class="reqn">C_T</code> values with genes in the columns and
samples in rows rows</p>
</td></tr>
<tr><td><code id="pcr_wilcox_+3A_group_var">group_var</code></td>
<td>
<p>A character vector of a grouping variable. The length of
this variable should equal the number of rows of df</p>
</td></tr>
<tr><td><code id="pcr_wilcox_+3A_reference_gene">reference_gene</code></td>
<td>
<p>A character string of the column name of a control
gene</p>
</td></tr>
<tr><td><code id="pcr_wilcox_+3A_reference_group">reference_group</code></td>
<td>
<p>A character string of the control group in group_var</p>
</td></tr>
<tr><td><code id="pcr_wilcox_+3A_tidy">tidy</code></td>
<td>
<p>A <code>logical</code> whether to return a <code>list</code> of <code>htest</code>
or a tidy <code>data.frame</code>. Default TRUE.</p>
</td></tr>
<tr><td><code id="pcr_wilcox_+3A_...">...</code></td>
<td>
<p>Other arguments to <code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame of 5 columns
</p>

<ul>
<li><p> gene The column names of df. reference_gene is dropped
</p>
</li>
<li><p> estimate The estimate for each term
</p>
</li>
<li><p> p_value The p-value for each term
</p>
</li>
<li><p> lower The low 95% confidence interval
</p>
</li>
<li><p> upper The high 95% confidence interval
</p>
</li></ul>

<p>When <code>tidy</code> is FALSE, returns a <code>list</code> of <code>htest</code> objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># locate and read data
fl &lt;- system.file('extdata', 'ct4.csv', package = 'pcr')
ct4 &lt;- read.csv(fl)

# make group variable
group &lt;- rep(c('control', 'treatment'), each = 12)

# test
pcr_wilcox(ct4,
           group_var = group,
           reference_gene = 'ref',
           reference_group = 'control')

# test using wilcox.test method
pcr_test(ct4,
         group_var = group,
         reference_gene = 'ref',
         reference_group = 'control',
         test = 'wilcox.test')

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
