<!DOCTYPE html><html><head><title>Help for package pctax</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pctax}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#pctax-package'><p>pctax: Professional Comprehensive Omics Data Analysis</p></a></li>
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#+25+26lt+3B+26gt+3B+25'><p>Assignment pipe</p></a></li>
<li><a href='#a_diversity'><p>Calculate a_diversity of otutab</p></a></li>
<li><a href='#add_strip'><p>add strips for a tree plot</p></a></li>
<li><a href='#add_tax'><p>Add taxonomy for a pc_otu object</p></a></li>
<li><a href='#ALDEX'><p>ALDEX</p></a></li>
<li><a href='#all_ec_info'><p>all element cycle information.</p></a></li>
<li><a href='#ann_tree'><p>Annotate a tree</p></a></li>
<li><a href='#aor'><p>Calculate Abundance-occupancy_relationship</p></a></li>
<li><a href='#as.b_dist'><p>Transfer dist to b_dist</p></a></li>
<li><a href='#as.dist.b_dist'><p>Transfer b_dist to dist</p></a></li>
<li><a href='#b_analyse'><p>Beta_diversity Ordination: dimensionality reduction</p></a></li>
<li><a href='#b_NTI1'><p>Calculate beta_NTI</p></a></li>
<li><a href='#b_res_3d'><p>3D plot for b_res</p></a></li>
<li><a href='#bbtt'><p>ggdotchart for diff analysis</p></a></li>
<li><a href='#before_tree'><p>Before df2tree check</p></a></li>
<li><a href='#check_taxonkit'><p>Check taxonkit</p></a></li>
<li><a href='#cor_net'><p>Correlation network, species-interaction network for omics</p></a></li>
<li><a href='#df2tree'><p>From a dataframe to construct a phylo</p></a></li>
<li><a href='#df2tree1'><p>From a dataframe to construct a phylo (save nwk)</p></a></li>
<li><a href='#diff_da'><p>Difference analysis</p></a></li>
<li><a href='#download_taxonkit_dataset'><p>Download taxonkit dataset</p></a></li>
<li><a href='#envfitt'><p>Envfit test for RDA result</p></a></li>
<li><a href='#geo_sim'><p>Lm for sample similarity and geographical distance</p></a></li>
<li><a href='#get_diff_type'><p>Get mean and type</p></a></li>
<li><a href='#gp_dis_density'><p>Group inter-intra density</p></a></li>
<li><a href='#grap_p_test'><p>Performs graph-based permutation tests</p></a></li>
<li><a href='#install_taxonkit'><p>Install taxonkit</p></a></li>
<li><a href='#kwtest'><p>KW test</p></a></li>
<li><a href='#load_N_data'><p>Load N-cycle data</p></a></li>
<li><a href='#mat_dist'><p>Calculate distance for otutab</p></a></li>
<li><a href='#micro_sbatch'><p>Microbiome sbatch</p></a></li>
<li><a href='#multi_bar'><p>Difference analysis</p></a></li>
<li><a href='#myRDA'><p>RDA</p></a></li>
<li><a href='#name_or_id2df'><p>Transfer taxon name or taxid to the lineage dataframe</p></a></li>
<li><a href='#ncm'><p>Sloan Neutral Model</p></a></li>
<li><a href='#nst'><p>Calculate NST for each group</p></a></li>
<li><a href='#nti_rc'><p>Calculate b_NTI and RC_bray for each group</p></a></li>
<li><a href='#pc_otu'><p>Create a pc_otu class object</p></a></li>
<li><a href='#pc_tax1'><p>test data (pc_otu class) for pc_tax package.</p></a></li>
<li><a href='#pc_valid'><p>Judge pc_otu is valid or not</p></a></li>
<li><a href='#permanova'><p>Permanova between a otutab and a variable</p></a></li>
<li><a href='#plot_element_cycle'><p>Plot element cycle</p></a></li>
<li><a href='#plot_N_cycle'><p>Plot the N-cycling pathway and genes</p></a></li>
<li><a href='#plot.a_res'><p>Plot a_res object</p></a></li>
<li><a href='#plot.b_res'><p>Plot a b_res</p></a></li>
<li><a href='#plot.g_test'><p>Plot g_test</p></a></li>
<li><a href='#plot.pro_res'><p>Plot pro_res</p></a></li>
<li><a href='#plot.time_cm'><p>Plot time_cm</p></a></li>
<li><a href='#pre_fastp'><p>Prepare the result from fastp (.json file)</p></a></li>
<li><a href='#pre_tax_table'><p>Complete a taxonomy table</p></a></li>
<li><a href='#print.pc_otu'><p>Print</p></a></li>
<li><a href='#procrustes_analyse'><p>Procrustes Rotation of Two Configurations and PROTEST</p></a></li>
<li><a href='#rare_curve_sample'><p>Rare the sample</p></a></li>
<li><a href='#rare_curve_species'><p>Rare the species</p></a></li>
<li><a href='#rarefaction'><p>Rarefy a otutab</p></a></li>
<li><a href='#RCbray1'><p>Calculate RCbray-curtis</p></a></li>
<li><a href='#RDA_plot'><p>Plot RDA res</p></a></li>
<li><a href='#suijisenlin'><p>RandomForest</p></a></li>
<li><a href='#summary.pc_otu'><p>Summary pc_otu</p></a></li>
<li><a href='#taxonkit_filter'><p>Filter TaxIDs based on Taxonomic Ranks</p></a></li>
<li><a href='#taxonkit_lca'><p>Compute Lowest Common Ancestor (LCA) of TaxIDs</p></a></li>
<li><a href='#taxonkit_lineage'><p>Retrieve Taxonomic Lineage using taxonkit</p></a></li>
<li><a href='#taxonkit_list'><p>Taxonkit list</p></a></li>
<li><a href='#taxonkit_name2taxid'><p>Convert Taxonomic Names to TaxIDs</p></a></li>
<li><a href='#taxonkit_reformat'><p>Reformat Taxonomic Lineage using taxonkit</p></a></li>
<li><a href='#time_by_cm'><p>Time series analysis</p></a></li>
<li><a href='#volcano_p'><p>Volcano plot for difference analysis</p></a></li>
<li><a href='#z_diversity'><p>Calculate Zeta Diversity</p></a></li>
<li><a href='#z_diversity_decay'><p>Calculate Zeta Diversity with Distance</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Professional Comprehensive Omics Data Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a comprehensive suite of tools for analyzing omics data. It includes
          functionalities for alpha diversity analysis, beta diversity analysis,
          differential abundance analysis, community assembly analysis, visualization of
          phylogenetic tree, and functional enrichment analysis. With a progressive
          approach, the package offers a range of analysis methods to explore and understand
          the complex communities. It is designed to support researchers and practitioners
          in conducting in-depth and professional omics data analysis.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>pcutils (&ge; 0.2.5), dplyr, ggplot2 (&ge; 3.2.0), vegan,
magrittr, grDevices, RColorBrewer, ggrepel, reshape2, stats,
tibble, utils, ggpubr, ggnewscale, ade4, scales</td>
</tr>
<tr>
<td>Suggests:</td>
<td>picante, ggpmisc, NST, permute, aplot, pheatmap, MASS, Rtsne,
mixOmics, geosphere, phyloseq, phyloseqGraphTest, plotly, umap,
Hmisc, minpack.lm, bbmle, snow, foreach, doSNOW, patchwork,
tidytree, ggtree, ggtreeExtra, vctrs, zoo, ape, DESeq2, limma,
ALDEx2, Mfuzz, edgeR, methods, randomForest, knitr, rmarkdown,
MetaNet, jsonlite, prettydoc, clipr, zetadiv, ggforce, readxl</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Asa12138/pctax/issues">https://github.com/Asa12138/pctax/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Asa12138/pctax">https://github.com/Asa12138/pctax</a></td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Microbiome, Software, Visualization</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-06 15:16:13 UTC; asa</td>
</tr>
<tr>
<td>Author:</td>
<td>Chen Peng <a href="https://orcid.org/0000-0002-9449-7606"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chen Peng &lt;pengchen2001@zju.edu.cn&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-10 17:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='pctax-package'>pctax: Professional Comprehensive Omics Data Analysis</h2><span id='topic+pctax'></span><span id='topic+pctax-package'></span>

<h3>Description</h3>

<p>Provides a comprehensive suite of tools for analyzing omics data. It includes functionalities for alpha diversity analysis, beta diversity analysis, differential abundance analysis, community assembly analysis, visualization of phylogenetic tree, and functional enrichment analysis. With a progressive approach, the package offers a range of analysis methods to explore and understand the complex communities. It is designed to support researchers and practitioners in conducting in-depth and professional omics data analysis.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Chen Peng <a href="mailto:pengchen2001@zju.edu.cn">pengchen2001@zju.edu.cn</a> (<a href="https://orcid.org/0000-0002-9449-7606">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/Asa12138/pctax">https://github.com/Asa12138/pctax</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/Asa12138/pctax/issues">https://github.com/Asa12138/pctax/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='+25+26lt+3B+26gt+3B+25'>Assignment pipe</h2><span id='topic++25+3C+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&lt;&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&lt;&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26lt+2B3B+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26lt+2B3B+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='a_diversity'>Calculate a_diversity of otutab</h2><span id='topic+a_diversity'></span><span id='topic+a_diversity.data.frame'></span><span id='topic+a_diversity.pc_otu'></span><span id='topic+a_diversity.numeric'></span>

<h3>Description</h3>

<p>Calculate a_diversity of otutab
</p>


<h3>Usage</h3>

<pre><code class='language-R'>a_diversity(otutab, ...)

## S3 method for class 'data.frame'
a_diversity(
  otutab,
  method = c("richness", "shannon"),
  tree = NULL,
  digits = 4,
  ...
)

## S3 method for class 'pc_otu'
a_diversity(otutab, method = "all", tbl = "otutab", ...)

## S3 method for class 'numeric'
a_diversity(otutab, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="a_diversity_+3A_otutab">otutab</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="a_diversity_+3A_...">...</code></td>
<td>
<p>pass to <code>a_diversity.data.frame</code></p>
</td></tr>
<tr><td><code id="a_diversity_+3A_method">method</code></td>
<td>
<p>one of &quot;all&quot;,&quot;richness&quot;,&quot;chao1&quot;,&quot;ace&quot;,&quot;gc&quot;,&quot;shannon&quot;,&quot;simpson&quot;,&quot;pd&quot;,&quot;pielou&quot;</p>
</td></tr>
<tr><td><code id="a_diversity_+3A_tree">tree</code></td>
<td>
<p>a iphylo object match the rownames of otutab</p>
</td></tr>
<tr><td><code id="a_diversity_+3A_digits">digits</code></td>
<td>
<p>maintance how many digits</p>
</td></tr>
<tr><td><code id="a_diversity_+3A_tbl">tbl</code></td>
<td>
<p>which table</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a a_res object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
a_diversity(otutab) -&gt; a_res
plot(a_res, "Group", metadata)
</code></pre>

<hr>
<h2 id='add_strip'>add strips for a tree plot</h2><span id='topic+add_strip'></span>

<h3>Description</h3>

<p>add strips for a tree plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_strip(trp, some_tax, flat_n = 5, strip_params = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_strip_+3A_trp">trp</code></td>
<td>
<p>tree plot from <code>ggtree</code></p>
</td></tr>
<tr><td><code id="add_strip_+3A_some_tax">some_tax</code></td>
<td>
<p>some tax you want to add strip</p>
</td></tr>
<tr><td><code id="add_strip_+3A_flat_n">flat_n</code></td>
<td>
<p>flat the text when taxa number more than <code>flat_n</code>.</p>
</td></tr>
<tr><td><code id="add_strip_+3A_strip_params">strip_params</code></td>
<td>
<p>parameters parse to <code><a href="ggtree.html#topic+geom_strip">geom_strip</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>tree plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(otutab, package = "pcutils")
# run yourself
if (interactive()) {
  ann_tree(taxonomy, otutab) -&gt; tree
  easy_tree(tree) -&gt; p
  some_tax &lt;- table(taxonomy$Phylum) %&gt;%
    sort(decreasing = TRUE) %&gt;%
    head(5) %&gt;%
    names()
  add_strip(p, some_tax)
}

</code></pre>

<hr>
<h2 id='add_tax'>Add taxonomy for a pc_otu object</h2><span id='topic+add_tax'></span>

<h3>Description</h3>

<p>Add taxonomy for a pc_otu object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_tax(pc, taxonomy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_tax_+3A_pc">pc</code></td>
<td>
<p>a pc_otu object</p>
</td></tr>
<tr><td><code id="add_tax_+3A_taxonomy">taxonomy</code></td>
<td>
<p>a taxomomy data.frame, look out the rownames of taxonomy and otutab should matched!</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pc_otu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
pc_tax1 &lt;- pc_otu(otutab, metadata)
pc_tax1 &lt;- add_tax(pc_tax1, taxonomy)
</code></pre>

<hr>
<h2 id='ALDEX'>ALDEX</h2><span id='topic+ALDEX'></span>

<h3>Description</h3>

<p>ALDEX
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ALDEX(otutab, group_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ALDEX_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="ALDEX_+3A_group_df">group_df</code></td>
<td>
<p>a dataframe with rowname same to dist and one group column</p>
</td></tr>
</table>


<h3>Value</h3>

<p>diff
</p>


<h3>References</h3>

<p><a href="https://cloud.tencent.com/developer/article/1621879">https://cloud.tencent.com/developer/article/1621879</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("ALDEx2")) {
  data(otutab, package = "pcutils")
  ALDEX(otutab, metadata["Group"]) -&gt; res
  res %&gt;%
    dplyr::top_n(9, -glm.eBH) %&gt;%
    .[, "tax"] -&gt; sig
  data.frame(t(otutab[sig, ])) %&gt;% pcutils::group_box(., "Group", metadata)
}

</code></pre>

<hr>
<h2 id='all_ec_info'>all element cycle information.</h2><span id='topic+all_ec_info'></span>

<h3>Description</h3>

<p>all element cycle information.
</p>


<h3>Format</h3>

<p>a list contains four tables.
</p>

<dl>
<dt>ec_node</dt><dd><p>chemicals</p>
</dd>
<dt>ec_link</dt><dd><p>reactions</p>
</dd>
<dt>ec_gene</dt><dd><p>genes</p>
</dd>
<dt>ec_path</dt><dd><p>reactions labels</p>
</dd>
</dl>


<hr>
<h2 id='ann_tree'>Annotate a tree</h2><span id='topic+ann_tree'></span><span id='topic+easy_tree'></span>

<h3>Description</h3>

<p>Annotate a tree
</p>
<p>Easy way to plot a phylogenetic tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ann_tree(f_tax, otutab = NULL, level = ncol(f_tax))

easy_tree(
  tree,
  highlight = "Phylum",
  colorfill = "color",
  pal = NULL,
  name_prefix = FALSE,
  basic_params = NULL,
  add_abundance = TRUE,
  color_name = "abundance",
  add_tiplab = TRUE,
  fontsize = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ann_tree_+3A_f_tax">f_tax</code></td>
<td>
<p>taxonomy dataframe</p>
</td></tr>
<tr><td><code id="ann_tree_+3A_otutab">otutab</code></td>
<td>
<p>otutab, rowname==rowname(taxonomy)</p>
</td></tr>
<tr><td><code id="ann_tree_+3A_level">level</code></td>
<td>
<p>1~7</p>
</td></tr>
<tr><td><code id="ann_tree_+3A_tree">tree</code></td>
<td>
<p>result from <code>ann_tree</code></p>
</td></tr>
<tr><td><code id="ann_tree_+3A_highlight">highlight</code></td>
<td>
<p>highlight which level, one of <code>tree$level</code></p>
</td></tr>
<tr><td><code id="ann_tree_+3A_colorfill">colorfill</code></td>
<td>
<p>&quot;color&quot; or &quot;fill&quot;</p>
</td></tr>
<tr><td><code id="ann_tree_+3A_pal">pal</code></td>
<td>
<p>color pal</p>
</td></tr>
<tr><td><code id="ann_tree_+3A_name_prefix">name_prefix</code></td>
<td>
<p>keep the prefix like &quot;k__&quot; or &quot;p__&quot; in the label? Default: FALSE</p>
</td></tr>
<tr><td><code id="ann_tree_+3A_basic_params">basic_params</code></td>
<td>
<p>parameters parse to <code><a href="ggtree.html#topic+ggtree">ggtree</a></code></p>
</td></tr>
<tr><td><code id="ann_tree_+3A_add_abundance">add_abundance</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="ann_tree_+3A_color_name">color_name</code></td>
<td>
<p>color name</p>
</td></tr>
<tr><td><code id="ann_tree_+3A_add_tiplab">add_tiplab</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="ann_tree_+3A_fontsize">fontsize</code></td>
<td>
<p>tip label fontsize</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a treedata
</p>
<p>a ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {
  data(otutab, package = "pcutils")
  ann_tree(taxonomy, otutab) -&gt; tree
  # run yourself
  easy_tree(tree, add_abundance = FALSE) -&gt; p
  p
}
</code></pre>

<hr>
<h2 id='aor'>Calculate Abundance-occupancy_relationship</h2><span id='topic+aor'></span><span id='topic+aor.data.frame'></span><span id='topic+plot.AOR'></span>

<h3>Description</h3>

<p>Calculate Abundance-occupancy_relationship
</p>
<p>Plot a AOR
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aor(otutab, ...)

## S3 method for class 'data.frame'
aor(
  otutab,
  top_r = 0.7,
  ocup_n = ceiling(0.8 * ncol(otutab)),
  special_n = ceiling(0.1 * ncol(otutab)),
  ...
)

## S3 method for class 'AOR'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aor_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="aor_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
<tr><td><code id="aor_+3A_top_r">top_r</code></td>
<td>
<p>percentage of top relative abundance</p>
</td></tr>
<tr><td><code id="aor_+3A_ocup_n">ocup_n</code></td>
<td>
<p>percentage of top occupied</p>
</td></tr>
<tr><td><code id="aor_+3A_special_n">special_n</code></td>
<td>
<p>how many occupancy define as specialists</p>
</td></tr>
<tr><td><code id="aor_+3A_x">x</code></td>
<td>
<p>AOR object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>AOR
</p>
<p>ggplot
</p>


<h3>References</h3>

<p>Barberán, A., Bates, S. T., Casamayor, E. &amp; Fierer, N. (2012) Using network analysis to explore co-occurrence patterns in soil microbial communities.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
aor(otutab) -&gt; AOR
plot(AOR)
</code></pre>

<hr>
<h2 id='as.b_dist'>Transfer dist to b_dist</h2><span id='topic+as.b_dist'></span><span id='topic+plot.dist'></span><span id='topic+plot.b_dist'></span>

<h3>Description</h3>

<p>Transfer dist to b_dist
</p>
<p>Plot dist
</p>
<p>Plot b_dist
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.b_dist(dist, group_df = NULL)

## S3 method for class 'dist'
plot(x, group_df = NULL, ...)

## S3 method for class 'b_dist'
plot(x, mode = 1, c_group = "inter", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.b_dist_+3A_dist">dist</code></td>
<td>
<p>a dist object</p>
</td></tr>
<tr><td><code id="as.b_dist_+3A_group_df">group_df</code></td>
<td>
<p>a dataframe with rowname same to dist and one group column</p>
</td></tr>
<tr><td><code id="as.b_dist_+3A_x">x</code></td>
<td>
<p>a b_dist</p>
</td></tr>
<tr><td><code id="as.b_dist_+3A_...">...</code></td>
<td>
<p>additional</p>
</td></tr>
<tr><td><code id="as.b_dist_+3A_mode">mode</code></td>
<td>
<p>1~3</p>
</td></tr>
<tr><td><code id="as.b_dist_+3A_c_group">c_group</code></td>
<td>
<p>&quot;inter&quot; or &quot;intra&quot; or both to plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a b_dist with annotation by group
</p>
<p>a pheatmap
</p>
<p>a ggplot or pheatmap
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
mat_dist(otutab) %&gt;% as.b_dist(., group_df = metadata["Group"]) -&gt; aa
plot(aa)
plot(aa, mode = 2)
</code></pre>

<hr>
<h2 id='as.dist.b_dist'>Transfer b_dist to dist</h2><span id='topic+as.dist.b_dist'></span>

<h3>Description</h3>

<p>Transfer b_dist to dist
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'b_dist'
as.dist(m, diag = FALSE, upper = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.dist.b_dist_+3A_m">m</code></td>
<td>
<p>a b_dist object</p>
</td></tr>
<tr><td><code id="as.dist.b_dist_+3A_diag">diag</code></td>
<td>
<p>logical value indicating whether the diagonal of the distance matrix should be printed by <code>print.dist</code>.</p>
</td></tr>
<tr><td><code id="as.dist.b_dist_+3A_upper">upper</code></td>
<td>
<p>logical value indicating whether the upper triangle of the distance matrix should be printed by <code>print.dist</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dist
</p>

<hr>
<h2 id='b_analyse'>Beta_diversity Ordination: dimensionality reduction</h2><span id='topic+b_analyse'></span><span id='topic+b_analyse.data.frame'></span>

<h3>Description</h3>

<p>Species abundance data can be preprocessed with Hellinger transformation or chord transformation data before PCA analysis. Because the Hellinger distance or chord distance with-without data is equal to <code class="reqn">\sqrt2\sqrt{1-Ochiai\ similarity}</code>, therefore, the sorting diagram (type 1 scale) of PCA analysis after Hellinger transformation or chord transformation with-without data is internal sample The distance between the squares is the Ochiai distance. <code class="reqn">\sqrt2\sqrt{1-Ochiai\ similarity}</code> is a distance measure, which is also suitable for the analysis of species data. The processed data is then used for pca without norm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>b_analyse(otutab, ...)

## S3 method for class 'data.frame'
b_analyse(
  otutab,
  norm = TRUE,
  method = c("pca", "nmds"),
  group = NULL,
  dist = "bray",
  ndim = 2,
  scale = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="b_analyse_+3A_otutab">otutab</code></td>
<td>
<p>an otutab data.frame, samples are columns, taxs are rows.</p>
</td></tr>
<tr><td><code id="b_analyse_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
<tr><td><code id="b_analyse_+3A_norm">norm</code></td>
<td>
<p>should normalized or not? (hellinger)</p>
</td></tr>
<tr><td><code id="b_analyse_+3A_method">method</code></td>
<td>
<p>one of &quot;pca&quot;,&quot;pcoa&quot;,&quot;ca&quot;,&quot;dca&quot;,&quot;nmds&quot;,&quot;plsda&quot;,&quot;tsne&quot;,&quot;umap&quot;,&quot;lda&quot;,&quot;all&quot;</p>
</td></tr>
<tr><td><code id="b_analyse_+3A_group">group</code></td>
<td>
<p>if needed, give a group vector</p>
</td></tr>
<tr><td><code id="b_analyse_+3A_dist">dist</code></td>
<td>
<p>if use pcoa or nmds, your can choose a dist method (default: bray) or input a distance matrix.</p>
</td></tr>
<tr><td><code id="b_analyse_+3A_ndim">ndim</code></td>
<td>
<p>how many dimension be kept? (default:2). 3 for b_res_3d()</p>
</td></tr>
<tr><td><code id="b_analyse_+3A_scale">scale</code></td>
<td>
<p>scale, default: FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>b_res object
</p>


<h3>References</h3>

<p><a href="https://www.jianshu.com/p/9694c0b6302d">https://www.jianshu.com/p/9694c0b6302d</a>
<a href="https://zhuanlan.zhihu.com/p/25501130">https://zhuanlan.zhihu.com/p/25501130</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
b_analyse(otutab, method = "pca") -&gt; b_res
plot(b_res, "Group", metadata)
</code></pre>

<hr>
<h2 id='b_NTI1'>Calculate beta_NTI</h2><span id='topic+b_NTI1'></span>

<h3>Description</h3>

<p>Calculate beta_NTI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>b_NTI1(
  phylo,
  otutab,
  beta.reps = 9,
  weighted = TRUE,
  threads = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="b_NTI1_+3A_phylo">phylo</code></td>
<td>
<p>a phylo object</p>
</td></tr>
<tr><td><code id="b_NTI1_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="b_NTI1_+3A_beta.reps">beta.reps</code></td>
<td>
<p>how many simulation performed?</p>
</td></tr>
<tr><td><code id="b_NTI1_+3A_weighted">weighted</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="b_NTI1_+3A_threads">threads</code></td>
<td>
<p>use how many threads to calculate (default:4)</p>
</td></tr>
<tr><td><code id="b_NTI1_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dist: b_NTI
</p>

<hr>
<h2 id='b_res_3d'>3D plot for b_res</h2><span id='topic+b_res_3d'></span>

<h3>Description</h3>

<p>3D plot for b_res
</p>


<h3>Usage</h3>

<pre><code class='language-R'>b_res_3d(b_res, Group, metadata = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="b_res_3d_+3A_b_res">b_res</code></td>
<td>
<p>a b_res object</p>
</td></tr>
<tr><td><code id="b_res_3d_+3A_group">Group</code></td>
<td>
<p>group vector for color</p>
</td></tr>
<tr><td><code id="b_res_3d_+3A_metadata">metadata</code></td>
<td>
<p>metadata contain Group</p>
</td></tr>
<tr><td><code id="b_res_3d_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plotly list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("plotly")) {
  data(otutab, package = "pcutils")
  b_analyse(otutab, method = "pca", ndim = 3) -&gt; b_res
  b_res_3d(b_res, "Group", metadata)
}
</code></pre>

<hr>
<h2 id='bbtt'>ggdotchart for diff analysis</h2><span id='topic+bbtt'></span>

<h3>Description</h3>

<p>ggdotchart for diff analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bbtt(res, pvalue = "glm.eBH", topN = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bbtt_+3A_res">res</code></td>
<td>
<p>result of ALDEX or kwtest</p>
</td></tr>
<tr><td><code id="bbtt_+3A_pvalue">pvalue</code></td>
<td>
<p>the name of pvaule</p>
</td></tr>
<tr><td><code id="bbtt_+3A_topn">topN</code></td>
<td>
<p>topN</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>

<hr>
<h2 id='before_tree'>Before df2tree check</h2><span id='topic+before_tree'></span>

<h3>Description</h3>

<p>Before df2tree check
</p>


<h3>Usage</h3>

<pre><code class='language-R'>before_tree(f_tax)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="before_tree_+3A_f_tax">f_tax</code></td>
<td>
<p>table</p>
</td></tr>
</table>


<h3>Value</h3>

<p>table
</p>


<h3>Examples</h3>

<pre><code class='language-R'>wrong_taxdf &lt;- data.frame(
  kingdom = c(rep(c("A", "B"), each = 4), "C", NA),
  "phylum" = c("A", "a", "b", "c", "c", "c", "d", NA, NA, "e")
)
before_tree(wrong_taxdf)

</code></pre>

<hr>
<h2 id='check_taxonkit'>Check taxonkit</h2><span id='topic+check_taxonkit'></span>

<h3>Description</h3>

<p>Check taxonkit
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_taxonkit(print = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_taxonkit_+3A_print">print</code></td>
<td>
<p>print</p>
</td></tr>
</table>


<h3>Value</h3>

<p>taxonkit path
</p>

<hr>
<h2 id='cor_net'>Correlation network, species-interaction network for omics</h2><span id='topic+cor_net'></span>

<h3>Description</h3>

<p>Correlation network, species-interaction network for omics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cor_net()
</code></pre>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='df2tree'>From a dataframe to construct a phylo</h2><span id='topic+df2tree'></span>

<h3>Description</h3>

<p>NOTE: this function will do <code>before_tree</code> first.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>df2tree(data, edge_df = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="df2tree_+3A_data">data</code></td>
<td>
<p>dataframe</p>
</td></tr>
<tr><td><code id="df2tree_+3A_edge_df">edge_df</code></td>
<td>
<p>if the data is edge_df ?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>phylo object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
df2tree(taxonomy) -&gt; tax_tree
print(tax_tree)
# check all nodes matched!
if (requireNamespace("picante")) {
  picante::match.phylo.comm(tax_tree, t(otutab)) -&gt; nn
  nrow(nn$comm) == nrow(t(otutab))
}
</code></pre>

<hr>
<h2 id='df2tree1'>From a dataframe to construct a phylo (save nwk)</h2><span id='topic+df2tree1'></span>

<h3>Description</h3>

<p>NOTE: this function will transfer all space to <code style="white-space: pre;">&#8288;_&#8288;</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>df2tree1(taxa)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="df2tree1_+3A_taxa">taxa</code></td>
<td>
<p>dataframe</p>
</td></tr>
</table>


<h3>Value</h3>

<p>phylo object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
df2tree(taxonomy) -&gt; tax_tree
print(tax_tree)
</code></pre>

<hr>
<h2 id='diff_da'>Difference analysis</h2><span id='topic+diff_da'></span>

<h3>Description</h3>

<p>Difference analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diff_da(
  otutab,
  group_df,
  ctrl = NULL,
  method = "deseq2",
  log = TRUE,
  add_mini = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diff_da_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="diff_da_+3A_group_df">group_df</code></td>
<td>
<p>a dataframe with rowname same to dist and one group column</p>
</td></tr>
<tr><td><code id="diff_da_+3A_ctrl">ctrl</code></td>
<td>
<p>the control group, one level of groups</p>
</td></tr>
<tr><td><code id="diff_da_+3A_method">method</code></td>
<td>
<p>one of &quot;deseq2&quot;,&quot;edger&quot;,&quot;limma&quot;,&quot;t.test&quot;,&quot;wilcox.test&quot;</p>
</td></tr>
<tr><td><code id="diff_da_+3A_log">log</code></td>
<td>
<p>do log transfer for limma?</p>
</td></tr>
<tr><td><code id="diff_da_+3A_add_mini">add_mini</code></td>
<td>
<p>add_mini when calculate the logFC. e.g (10+0.1)/(0+0.1), default <code>0.5*min(abundance)</code></p>
</td></tr>
<tr><td><code id="diff_da_+3A_...">...</code></td>
<td>
<p>other parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("limma")) {
  data(otutab, package = "pcutils")
  diff_da(otutab, metadata["Group"], method = "limma") -&gt; res
  volcano_p(res)
  volcano_p(res, mode = 2)
}

</code></pre>

<hr>
<h2 id='download_taxonkit_dataset'>Download taxonkit dataset</h2><span id='topic+download_taxonkit_dataset'></span>

<h3>Description</h3>

<p>Download taxonkit dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>download_taxonkit_dataset(make_sure = FALSE, taxdump_tar_gz = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="download_taxonkit_dataset_+3A_make_sure">make_sure</code></td>
<td>
<p>make sure to do this</p>
</td></tr>
<tr><td><code id="download_taxonkit_dataset_+3A_taxdump_tar_gz">taxdump_tar_gz</code></td>
<td>
<p>your download taxdump_tar_gz file from https://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='envfitt'>Envfit test for RDA result</h2><span id='topic+envfitt'></span>

<h3>Description</h3>

<p>Envfit test for RDA result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>envfitt(phy.rda, env, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="envfitt_+3A_phy.rda">phy.rda</code></td>
<td>
<p>a rda result</p>
</td></tr>
<tr><td><code id="envfitt_+3A_env">env</code></td>
<td>
<p>environmental factors</p>
</td></tr>
<tr><td><code id="envfitt_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>g_test object
</p>


<h3>See Also</h3>

<p><code><a href="vegan.html#topic+envfit">envfit</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
env &lt;- metadata[, 6:10]
# RDA
myRDA(otutab, env) -&gt; phy.rda
envfitt(phy.rda, env) -&gt; envfit_res
plot(envfit_res)
</code></pre>

<hr>
<h2 id='geo_sim'>Lm for sample similarity and geographical distance</h2><span id='topic+geo_sim'></span>

<h3>Description</h3>

<p>Lm for sample similarity and geographical distance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geo_sim(otutab, geo, method = "bray", spe_nwk = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geo_sim_+3A_otutab">otutab</code></td>
<td>
<p>an otutab data.frame, samples are columns, taxs are rows.</p>
</td></tr>
<tr><td><code id="geo_sim_+3A_geo">geo</code></td>
<td>
<p>a two-columns dataframe, first is latitude, second is longitude</p>
</td></tr>
<tr><td><code id="geo_sim_+3A_method">method</code></td>
<td>
<p>Dissimilarity index, partial match to &quot;bray&quot;, &quot;euclidean&quot;...see <code><a href="vegan.html#topic+vegdist">vegdist</a>;<a href="picante.html#topic+unifrac">unifrac</a></code></p>
</td></tr>
<tr><td><code id="geo_sim_+3A_spe_nwk">spe_nwk</code></td>
<td>
<p>a phylo tree if use unifrac...</p>
</td></tr>
<tr><td><code id="geo_sim_+3A_...">...</code></td>
<td>
<p>additional</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot
</p>


<h3>References</h3>

<p>Graco-Roza, C. et al. (2022) Distance decay 2.0 - A global synthesis of taxonomic and functional turnover in ecological communities. Glob Ecol Biogeogr 31, 1399–1421.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("NST") &amp;&amp; requireNamespace("geosphere")) {
  library(ggplot2)
  data(otutab, package = "pcutils")
  metadata[, c("lat", "long")] -&gt; geo
  geo_sim(otutab, geo) -&gt; geo_res
  pcutils::my_lm(geo_res[4], "dis.geo", geo_res)
}
</code></pre>

<hr>
<h2 id='get_diff_type'>Get mean and type</h2><span id='topic+get_diff_type'></span>

<h3>Description</h3>

<p>Get mean and type
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_diff_type(otutab, group_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_diff_type_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="get_diff_type_+3A_group_df">group_df</code></td>
<td>
<p>a dataframe with rowname same to dist and one group column</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='gp_dis_density'>Group inter-intra density</h2><span id='topic+gp_dis_density'></span>

<h3>Description</h3>

<p>Group inter-intra density
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gp_dis_density(otutab, group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gp_dis_density_+3A_otutab">otutab</code></td>
<td>
<p>an otutab data.frame, samples are columns, taxs are rows.</p>
</td></tr>
<tr><td><code id="gp_dis_density_+3A_group">group</code></td>
<td>
<p>group vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
gp_dis_density(otutab, metadata["Group"])
</code></pre>

<hr>
<h2 id='grap_p_test'>Performs graph-based permutation tests</h2><span id='topic+grap_p_test'></span>

<h3>Description</h3>

<p>Performs graph-based permutation tests
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grap_p_test(otutab, metadata, group = "Group", nperm = 999, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grap_p_test_+3A_otutab">otutab</code></td>
<td>
<p>an otutab data.frame, samples are columns, taxs are rows.</p>
</td></tr>
<tr><td><code id="grap_p_test_+3A_metadata">metadata</code></td>
<td>
<p>metadata</p>
</td></tr>
<tr><td><code id="grap_p_test_+3A_group">group</code></td>
<td>
<p>one group name in columns of metadata</p>
</td></tr>
<tr><td><code id="grap_p_test_+3A_nperm">nperm</code></td>
<td>
<p>numbers of permutations to perform</p>
</td></tr>
<tr><td><code id="grap_p_test_+3A_...">...</code></td>
<td>
<p>additional</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("phyloseqGraphTest") &amp;&amp; requireNamespace("phyloseq")) {
  data(otutab, package = "pcutils")
  grap_p_test(otutab, metadata, "Group")
}

</code></pre>

<hr>
<h2 id='install_taxonkit'>Install taxonkit</h2><span id='topic+install_taxonkit'></span>

<h3>Description</h3>

<p>Install taxonkit
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_taxonkit(make_sure = FALSE, taxonkit_tar_gz = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="install_taxonkit_+3A_make_sure">make_sure</code></td>
<td>
<p>make sure to do this</p>
</td></tr>
<tr><td><code id="install_taxonkit_+3A_taxonkit_tar_gz">taxonkit_tar_gz</code></td>
<td>
<p>your download taxonkit_tar_gz file from https://github.com/shenwei356/taxonkit/releases/</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='kwtest'>KW test</h2><span id='topic+kwtest'></span>

<h3>Description</h3>

<p>KW test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kwtest(otutab, group_df, method = "kruskal.test")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kwtest_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="kwtest_+3A_group_df">group_df</code></td>
<td>
<p>a dataframe with rowname same to dist and one group column</p>
</td></tr>
<tr><td><code id="kwtest_+3A_method">method</code></td>
<td>
<p>&quot;kruskal.test&quot;, see <code><a href="ggpubr.html#topic+compare_means">compare_means</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>res
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(otutab, package = "pcutils")
kwtest(otutab, metadata["Group"]) -&gt; res
bbtt(res, pvalue = "p.format")

</code></pre>

<hr>
<h2 id='load_N_data'>Load N-cycle data</h2><span id='topic+load_N_data'></span>

<h3>Description</h3>

<p>Load N-cycle data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_N_data()
</code></pre>


<h3>Value</h3>

<p>list
</p>


<h3>References</h3>

<p>Tu, Q., Lin, L., Cheng, L., Deng, Y. &amp; He, Z. (2019) NCycDB: a curated integrative database for fast and accurate metagenomic profiling of nitrogen cycling genes. Bioinformatics 35, 1040–1048.
Kuypers, M. M. M., Marchant, H. K. &amp; Kartal, B. (2018) The microbial nitrogen-cycling network. Nat Rev Microbiol 16, 263–276.
</p>

<hr>
<h2 id='mat_dist'>Calculate distance for otutab</h2><span id='topic+mat_dist'></span>

<h3>Description</h3>

<p>Calculate distance for otutab
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mat_dist(otutab, method = "bray", spe_nwk = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mat_dist_+3A_otutab">otutab</code></td>
<td>
<p>an otutab data.frame, samples are columns, taxs are rows.</p>
</td></tr>
<tr><td><code id="mat_dist_+3A_method">method</code></td>
<td>
<p>Dissimilarity index, partial match to &quot;bray&quot;, &quot;euclidean&quot;...see <code><a href="vegan.html#topic+vegdist">vegdist</a>;<a href="picante.html#topic+unifrac">unifrac</a></code></p>
</td></tr>
<tr><td><code id="mat_dist_+3A_spe_nwk">spe_nwk</code></td>
<td>
<p>a phylo tree if use unifrac...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dist
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
mat_dist(otutab)
</code></pre>

<hr>
<h2 id='micro_sbatch'>Microbiome sbatch</h2><span id='topic+micro_sbatch'></span>

<h3>Description</h3>

<p>Microbiome sbatch
</p>


<h3>Usage</h3>

<pre><code class='language-R'>micro_sbatch(
  work_dir = "/share/home/jianglab/pengchen/work/asthma/",
  step = "fastp",
  all_sample_num = 40,
  array = 1,
  partition = "cpu",
  cpus_per_task = 1,
  mem_per_cpu = "2G"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="micro_sbatch_+3A_work_dir">work_dir</code></td>
<td>
<p>work_dir</p>
</td></tr>
<tr><td><code id="micro_sbatch_+3A_step">step</code></td>
<td>
<p>&quot;fastp&quot;,&quot;rm_human&quot;,&quot;megahit&quot;,&quot;prodigal&quot;,&quot;salmon-quant&quot;,...</p>
</td></tr>
<tr><td><code id="micro_sbatch_+3A_all_sample_num">all_sample_num</code></td>
<td>
<p>all sample number</p>
</td></tr>
<tr><td><code id="micro_sbatch_+3A_array">array</code></td>
<td>
<p>array number</p>
</td></tr>
<tr><td><code id="micro_sbatch_+3A_partition">partition</code></td>
<td>
<p>partition</p>
</td></tr>
<tr><td><code id="micro_sbatch_+3A_cpus_per_task">cpus_per_task</code></td>
<td>
<p>cpus_per_task</p>
</td></tr>
<tr><td><code id="micro_sbatch_+3A_mem_per_cpu">mem_per_cpu</code></td>
<td>
<p>mem_per_cpu, &quot;2G&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='multi_bar'>Difference analysis</h2><span id='topic+multi_bar'></span>

<h3>Description</h3>

<p>Difference analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi_bar(
  otutab,
  group_df,
  mode = 1,
  text_df = NULL,
  text_x = NULL,
  text_angle = -90,
  errorbar = "bottom"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multi_bar_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="multi_bar_+3A_group_df">group_df</code></td>
<td>
<p>a dataframe with rowname same to dist and one group column</p>
</td></tr>
<tr><td><code id="multi_bar_+3A_mode">mode</code></td>
<td>
<p>1~2</p>
</td></tr>
<tr><td><code id="multi_bar_+3A_text_df">text_df</code></td>
<td>
<p>text_df</p>
</td></tr>
<tr><td><code id="multi_bar_+3A_text_x">text_x</code></td>
<td>
<p>text_x</p>
</td></tr>
<tr><td><code id="multi_bar_+3A_text_angle">text_angle</code></td>
<td>
<p>text_angle</p>
</td></tr>
<tr><td><code id="multi_bar_+3A_errorbar">errorbar</code></td>
<td>
<p>top, bottom, none</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
multi_bar(otutab[1:10, ], metadata["Group"])
</code></pre>

<hr>
<h2 id='myRDA'>RDA</h2><span id='topic+myRDA'></span><span id='topic+myCCA'></span><span id='topic+myCAP'></span>

<h3>Description</h3>

<p>RDA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>myRDA(
  otutab,
  env,
  norm = TRUE,
  scale = FALSE,
  choose_var = FALSE,
  direction = "forward",
  nperm = 499,
  verbose = TRUE,
  method = "rda",
  dist = "bray"
)

myCCA(
  otutab,
  env,
  norm = TRUE,
  scale = FALSE,
  choose_var = FALSE,
  nperm = 499,
  verbose = TRUE
)

myCAP(
  otutab,
  env,
  norm = TRUE,
  scale = FALSE,
  choose_var = FALSE,
  nperm = 499,
  verbose = TRUE,
  dist = "bray"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="myRDA_+3A_otutab">otutab</code></td>
<td>
<p>an otutab data.frame, samples are columns, taxs are rows.</p>
</td></tr>
<tr><td><code id="myRDA_+3A_env">env</code></td>
<td>
<p>environmental factors</p>
</td></tr>
<tr><td><code id="myRDA_+3A_norm">norm</code></td>
<td>
<p>should normalize? (default:TRUE)</p>
</td></tr>
<tr><td><code id="myRDA_+3A_scale">scale</code></td>
<td>
<p>should scale species? (default:FALSE)</p>
</td></tr>
<tr><td><code id="myRDA_+3A_choose_var">choose_var</code></td>
<td>
<p>should choose variables? use forward step</p>
</td></tr>
<tr><td><code id="myRDA_+3A_direction">direction</code></td>
<td>
<p>The direction of the stepwise selection, &quot;both&quot;, &quot;forward&quot; or &quot;backward&quot;, default is &quot;forward&quot;</p>
</td></tr>
<tr><td><code id="myRDA_+3A_nperm">nperm</code></td>
<td>
<p>number of permutation</p>
</td></tr>
<tr><td><code id="myRDA_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
<tr><td><code id="myRDA_+3A_method">method</code></td>
<td>
<p>&quot;rda&quot;, &quot;cca&quot;, &quot;cap&quot;, &quot;dbrda&quot;</p>
</td></tr>
<tr><td><code id="myRDA_+3A_dist">dist</code></td>
<td>
<p>The name of the dissimilarity (or distance) index for &quot;cap&quot; or &quot;dbrda&quot;, for <code><a href="vegan.html#topic+vegdist">vegdist</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>rda/cca
</p>


<h3>See Also</h3>

<p><code><a href="vegan.html#topic+vegdist">vegdist</a>;<a href="picante.html#topic+unifrac">unifrac</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
env &lt;- metadata[, 6:10]
# RDA
myRDA(otutab, env) -&gt; phy.rda
RDA_plot(phy.rda, "Group", metadata)
</code></pre>

<hr>
<h2 id='name_or_id2df'>Transfer taxon name or taxid to the lineage dataframe</h2><span id='topic+name_or_id2df'></span>

<h3>Description</h3>

<p>Transfer taxon name or taxid to the lineage dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>name_or_id2df(
  name_or_id,
  mode = "name",
  add_prefix = TRUE,
  fill_miss_rank = TRUE,
  data_dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="name_or_id2df_+3A_name_or_id">name_or_id</code></td>
<td>
<p>name or taxid</p>
</td></tr>
<tr><td><code id="name_or_id2df_+3A_mode">mode</code></td>
<td>
<p>&quot;id&quot; or &quot;name&quot;</p>
</td></tr>
<tr><td><code id="name_or_id2df_+3A_add_prefix">add_prefix</code></td>
<td>
<p>add_prefix</p>
</td></tr>
<tr><td><code id="name_or_id2df_+3A_fill_miss_rank">fill_miss_rank</code></td>
<td>
<p>fill_miss_rank</p>
</td></tr>
<tr><td><code id="name_or_id2df_+3A_data_dir">data_dir</code></td>
<td>
<p>directory containing nodes.dmp and names.dmp (default &quot;/Users/asa/.taxonkit&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
name_or_id2df(c("Homo sapiens", "Akkermansia muciniphila ATCC BAA-835"))

## End(Not run)
</code></pre>

<hr>
<h2 id='ncm'>Sloan Neutral Model</h2><span id='topic+ncm'></span><span id='topic+plot.ncm_res'></span>

<h3>Description</h3>

<p>Sloan Neutral Model
</p>
<p>Plot ncm_res
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ncm(otutab, model = "nls")

## S3 method for class 'ncm_res'
plot(
  x,
  mycols = c(Above = "#069870", Below = "#e29e02", In = "#1e353a"),
  text_position = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ncm_+3A_otutab">otutab</code></td>
<td>
<p>an otutab data.frame, samples are columns, taxs are rows.</p>
</td></tr>
<tr><td><code id="ncm_+3A_model">model</code></td>
<td>
<p>fit method, one of &quot;nls&quot;,&quot;mle&quot;</p>
</td></tr>
<tr><td><code id="ncm_+3A_x">x</code></td>
<td>
<p>a ncm_res object</p>
</td></tr>
<tr><td><code id="ncm_+3A_mycols">mycols</code></td>
<td>
<p>mycols</p>
</td></tr>
<tr><td><code id="ncm_+3A_text_position">text_position</code></td>
<td>
<p>text_position</p>
</td></tr>
<tr><td><code id="ncm_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ncm_res
</p>
<p>ggplot
</p>


<h3>References</h3>

<p>Sloan, W. TRUE. et al. (2006) Quantifying the roles of immigration and chance in shaping prokaryote community structure. Environmental Microbiology 8, 732–740.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("Hmisc") &amp;&amp; requireNamespace("minpack.lm")) {
  data(otutab, package = "pcutils")
  ncm(otutab) -&gt; ncm_res
  plot(ncm_res)
}

</code></pre>

<hr>
<h2 id='nst'>Calculate NST for each group</h2><span id='topic+nst'></span>

<h3>Description</h3>

<p>Calculate NST for each group
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nst(otutab, group_df, threads = 1, file = NULL, rep = 20, save = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nst_+3A_otutab">otutab</code></td>
<td>
<p>an otutab data.frame, samples are columns, taxs are rows.</p>
</td></tr>
<tr><td><code id="nst_+3A_group_df">group_df</code></td>
<td>
<p>a dataframe with rowname and one group column</p>
</td></tr>
<tr><td><code id="nst_+3A_threads">threads</code></td>
<td>
<p>default:4</p>
</td></tr>
<tr><td><code id="nst_+3A_file">file</code></td>
<td>
<p>filename to save</p>
</td></tr>
<tr><td><code id="nst_+3A_rep">rep</code></td>
<td>
<p>repeat numbers: suggest 999</p>
</td></tr>
<tr><td><code id="nst_+3A_save">save</code></td>
<td>
<p>save the file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a b_dist object, dis is MSTij
</p>


<h3>References</h3>

<p>Ning, D., Deng, Y., Tiedje, J. M. &amp; Zhou, J. (2019) A general framework for quantitatively assessing ecological stochasticity. Proceedings of the National Academy of Sciences 116, 16892–16898.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("NST")) {
  library(ggplot2)
  data(otutab, package = "pcutils")
  nst(otutab, metadata["Group"]) -&gt; nst_res
  plot(nst_res, c_group = "intra") + geom_hline(yintercept = 0.5, lty = 2) + ylab("NST")
}

</code></pre>

<hr>
<h2 id='nti_rc'>Calculate b_NTI and RC_bray for each group</h2><span id='topic+nti_rc'></span><span id='topic+plot.NTI_RC'></span>

<h3>Description</h3>

<p>Calculate b_NTI and RC_bray for each group
</p>
<p>Plot NTI_RC object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nti_rc(
  otutab,
  phylo,
  group_df,
  threads = 1,
  file = NULL,
  rep = 20,
  save = FALSE
)

## S3 method for class 'NTI_RC'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nti_rc_+3A_otutab">otutab</code></td>
<td>
<p>an otutab data.frame, samples are columns, taxs are rows.</p>
</td></tr>
<tr><td><code id="nti_rc_+3A_phylo">phylo</code></td>
<td>
<p>a phylo object</p>
</td></tr>
<tr><td><code id="nti_rc_+3A_group_df">group_df</code></td>
<td>
<p>a dataframe with rowname and one group column</p>
</td></tr>
<tr><td><code id="nti_rc_+3A_threads">threads</code></td>
<td>
<p>default:4</p>
</td></tr>
<tr><td><code id="nti_rc_+3A_file">file</code></td>
<td>
<p>filename to save</p>
</td></tr>
<tr><td><code id="nti_rc_+3A_rep">rep</code></td>
<td>
<p>repeat numbers: suggest 999</p>
</td></tr>
<tr><td><code id="nti_rc_+3A_save">save</code></td>
<td>
<p>save the file</p>
</td></tr>
<tr><td><code id="nti_rc_+3A_x">x</code></td>
<td>
<p>NTI_RC object</p>
</td></tr>
<tr><td><code id="nti_rc_+3A_...">...</code></td>
<td>
<p>pass to <code><a href="pcutils.html#topic+stackplot">stackplot</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a b_dist object, dis is MSTij
</p>
<p>ggplot
</p>


<h3>References</h3>

<p>Ning, D., Deng, Y., Tiedje, J. M. &amp; Zhou, J. (2019) A general framework for quantitatively assessing ecological stochasticity. Proceedings of the National Academy of Sciences 116, 16892–16898.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("NST") &amp;&amp; requireNamespace("pctax")) {
  data(otutab, package = "pcutils")
  pctax::df2tree(taxonomy) -&gt; phylo
  nti_rc(otutab, phylo, metadata["Group"]) -&gt; nti_res
  plot(nti_res)
}

</code></pre>

<hr>
<h2 id='pc_otu'>Create a pc_otu class object</h2><span id='topic+pc_otu'></span>

<h3>Description</h3>

<p>Create a pc_otu class object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pc_otu(otutab = data.frame(), metadata = data.frame(), taxonomy = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pc_otu_+3A_otutab">otutab</code></td>
<td>
<p>an otutab data.frame, samples are columns, taxs are rows.</p>
</td></tr>
<tr><td><code id="pc_otu_+3A_metadata">metadata</code></td>
<td>
<p>a metadata data.frame, samples are rows</p>
</td></tr>
<tr><td><code id="pc_otu_+3A_taxonomy">taxonomy</code></td>
<td>
<p>a taxomomy data.frame, look out the rowname of taxonomy and otutab should matched!</p>
</td></tr>
<tr><td><code id="pc_otu_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pc_otu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
pc_tax1 &lt;- pc_otu(otutab, metadata)
print(pc_tax1)
</code></pre>

<hr>
<h2 id='pc_tax1'>test data (pc_otu class) for pc_tax package.</h2><span id='topic+pc_tax1'></span>

<h3>Description</h3>

<p>an otutab, metadata and a taxonomy table.
</p>


<h3>Format</h3>

<p>a pc_otu contains an otutab, metadata and a taxonomy table.
</p>

<dl>
<dt>tbls</dt><dd><p>contians otutable rawdata</p>
</dd>
<dt>metas</dt><dd><p>contians metadata</p>
</dd>
<dt>otus</dt><dd><p>contians taxomomy table</p>
</dd>
</dl>


<hr>
<h2 id='pc_valid'>Judge pc_otu is valid or not</h2><span id='topic+pc_valid'></span>

<h3>Description</h3>

<p>Judge pc_otu is valid or not
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pc_valid(pc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pc_valid_+3A_pc">pc</code></td>
<td>
<p>a pc_otu object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>logical
</p>

<hr>
<h2 id='permanova'>Permanova between a otutab and a variable</h2><span id='topic+permanova'></span>

<h3>Description</h3>

<p>Permanova between a otutab and a variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>permanova(
  otutab,
  envs,
  norm = TRUE,
  each = TRUE,
  method = "adonis",
  dist = "bray",
  nperm = 999,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="permanova_+3A_otutab">otutab</code></td>
<td>
<p>an otutab data.frame, samples are columns, taxs are rows.</p>
</td></tr>
<tr><td><code id="permanova_+3A_envs">envs</code></td>
<td>
<p>factors need to test</p>
</td></tr>
<tr><td><code id="permanova_+3A_norm">norm</code></td>
<td>
<p>should normalize?(default:TRUE)</p>
</td></tr>
<tr><td><code id="permanova_+3A_each">each</code></td>
<td>
<p>test factor one by one, rather than whole</p>
</td></tr>
<tr><td><code id="permanova_+3A_method">method</code></td>
<td>
<p>adonis/mrpp/anosim/mantel</p>
</td></tr>
<tr><td><code id="permanova_+3A_dist">dist</code></td>
<td>
<p>if use pcoa or nmds, your can choose a dist method (default: bray)</p>
</td></tr>
<tr><td><code id="permanova_+3A_nperm">nperm</code></td>
<td>
<p>numbers of permutations to perform</p>
</td></tr>
<tr><td><code id="permanova_+3A_...">...</code></td>
<td>
<p>additional</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a g_test object with these columns
</p>
<table>
<tr><td><code>group</code></td>
<td>
<p>the test group or factor</p>
</td></tr>
<tr><td><code>r</code></td>
<td>
<p>relationship</p>
</td></tr>
<tr><td><code>r2</code></td>
<td>
<p>model R-square</p>
</td></tr>
<tr><td><code>p_value</code></td>
<td>
<p>model test p_value</p>
</td></tr>
<tr><td><code>sig</code></td>
<td>
<p>whether significant</p>
</td></tr>
</table>


<h3>References</h3>

<p><a href="https://blog.csdn.net/qq_42458954/article/details/110390488">https://blog.csdn.net/qq_42458954/article/details/110390488</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
permanova(otutab, metadata[, c(2:10)]) -&gt; adonis_res
print(adonis_res)
plot(adonis_res)
</code></pre>

<hr>
<h2 id='plot_element_cycle'>Plot element cycle</h2><span id='topic+plot_element_cycle'></span>

<h3>Description</h3>

<p>Plot element cycle
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_element_cycle(
  cycle = "Nitrogen cycle",
  anno_df = NULL,
  only_anno = FALSE,
  cell_fill = NA,
  cell_color = "orange",
  chemical_size = 7,
  chemical_bold = TRUE,
  chemical_color = "black",
  chemical_label = TRUE,
  reaction_width = 1,
  reaction_arrow_size = 4,
  reaction_arrow_closed = TRUE,
  gene_or_ko = "gene",
  gene_size = 3,
  gene_x_offset = 0.3,
  gene_y_offset = 0.15,
  gene_label = TRUE,
  gene_color = NULL,
  gene_bold = TRUE,
  gene_italic = TRUE,
  gene_label_fill = "white"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_element_cycle_+3A_cycle">cycle</code></td>
<td>
<p>one of c(&quot;Carbon cycle&quot;,&quot;Nitrogen cycle&quot;,&quot;Phosphorus cycle&quot;,&quot;Sulfur cycle&quot;,&quot;Iron cycle&quot;)</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_anno_df">anno_df</code></td>
<td>
<p>anno_df, columns should contains Gene or KO and Group</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_only_anno">only_anno</code></td>
<td>
<p>only show genes in anno_df?</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_cell_fill">cell_fill</code></td>
<td>
<p>cell fill color</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_cell_color">cell_color</code></td>
<td>
<p>cell border color</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_chemical_size">chemical_size</code></td>
<td>
<p>chemical text size</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_chemical_bold">chemical_bold</code></td>
<td>
<p>chemical text bold</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_chemical_color">chemical_color</code></td>
<td>
<p>chemical text color</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_chemical_label">chemical_label</code></td>
<td>
<p>chemical text in geom_label or geom_text?</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_reaction_width">reaction_width</code></td>
<td>
<p>reaction line width</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_reaction_arrow_size">reaction_arrow_size</code></td>
<td>
<p>reaction arrow size</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_reaction_arrow_closed">reaction_arrow_closed</code></td>
<td>
<p>reaction arrow closed?</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_gene_or_ko">gene_or_ko</code></td>
<td>
<p>&quot;gene&quot; or &quot;ko&quot;</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_gene_size">gene_size</code></td>
<td>
<p>gene text size</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_gene_x_offset">gene_x_offset</code></td>
<td>
<p>gene_x_offset</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_gene_y_offset">gene_y_offset</code></td>
<td>
<p>gene_y_offset</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_gene_label">gene_label</code></td>
<td>
<p>gene text in geom_label or geom_text?</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_gene_color">gene_color</code></td>
<td>
<p>gene text color</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_gene_bold">gene_bold</code></td>
<td>
<p>gene text blod?</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_gene_italic">gene_italic</code></td>
<td>
<p>gene text italic?</p>
</td></tr>
<tr><td><code id="plot_element_cycle_+3A_gene_label_fill">gene_label_fill</code></td>
<td>
<p>gene label fill color</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("ggforce")) plot_element_cycle()
</code></pre>

<hr>
<h2 id='plot_N_cycle'>Plot the N-cycling pathway and genes</h2><span id='topic+plot_N_cycle'></span>

<h3>Description</h3>

<p>Plot the N-cycling pathway and genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_N_cycle(
  my_N_genes = NULL,
  just_diff = FALSE,
  path_col = NULL,
  type_col = c(up = "red", down = "blue", none = NA),
  fill_alpha = 0.5,
  arrow_size = 0.1,
  line_width = 1,
  title = "Nitrogen cycling",
  legend.position = c(0.85, 0.15)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_N_cycle_+3A_my_n_genes">my_N_genes</code></td>
<td>
<p>dataframe, &quot;Gene_families&quot;,&quot;type&quot; should in colnames of my_N_genes</p>
</td></tr>
<tr><td><code id="plot_N_cycle_+3A_just_diff">just_diff</code></td>
<td>
<p>logical, just plot the different genes?</p>
</td></tr>
<tr><td><code id="plot_N_cycle_+3A_path_col">path_col</code></td>
<td>
<p>colors of pathways</p>
</td></tr>
<tr><td><code id="plot_N_cycle_+3A_type_col">type_col</code></td>
<td>
<p>colors of types</p>
</td></tr>
<tr><td><code id="plot_N_cycle_+3A_fill_alpha">fill_alpha</code></td>
<td>
<p>alpha, default 0.5</p>
</td></tr>
<tr><td><code id="plot_N_cycle_+3A_arrow_size">arrow_size</code></td>
<td>
<p>arrow_size, default 0.1</p>
</td></tr>
<tr><td><code id="plot_N_cycle_+3A_line_width">line_width</code></td>
<td>
<p>line_width, default 1</p>
</td></tr>
<tr><td><code id="plot_N_cycle_+3A_title">title</code></td>
<td>
<p>title, default &quot;Nitrogen cycling&quot;</p>
</td></tr>
<tr><td><code id="plot_N_cycle_+3A_legend.position">legend.position</code></td>
<td>
<p>default c(0.85,0.15)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>N_data &lt;- load_N_data()
my_N_genes &lt;- data.frame(
  `Gene_families` = sample(N_data$N_genes$Gene_families, 10, replace = FALSE),
  change = rnorm(10), check.names = FALSE
)
my_N_genes &lt;- dplyr::mutate(my_N_genes,
  type = ifelse(change &gt; 0, "up", ifelse(change &lt; 0, "down", "none"))
)
plot_N_cycle(my_N_genes, just_diff = FALSE, fill_alpha = 0.2)
# ggsave(filename = "test.pdf", width = 14, height = 10)
</code></pre>

<hr>
<h2 id='plot.a_res'>Plot a_res object</h2><span id='topic+plot.a_res'></span>

<h3>Description</h3>

<p>Plot a_res object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'a_res'
plot(x, group, metadata, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.a_res_+3A_x">x</code></td>
<td>
<p>a a_res object</p>
</td></tr>
<tr><td><code id="plot.a_res_+3A_group">group</code></td>
<td>
<p>one of colname of metadata</p>
</td></tr>
<tr><td><code id="plot.a_res_+3A_metadata">metadata</code></td>
<td>
<p>metadata</p>
</td></tr>
<tr><td><code id="plot.a_res_+3A_...">...</code></td>
<td>
<p>addditional parameters for <code><a href="pcutils.html#topic+group_box">group_box</a></code> or <code><a href="pcutils.html#topic+my_lm">my_lm</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>patchwork object,you can change theme with &amp;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+a_diversity">a_diversity</a></code>
</p>

<hr>
<h2 id='plot.b_res'>Plot a b_res</h2><span id='topic+plot.b_res'></span>

<h3>Description</h3>

<p>Plot a b_res
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'b_res'
plot(
  x,
  Group,
  metadata = NULL,
  Group2 = NULL,
  mode = 1,
  bi = FALSE,
  Topn = 10,
  rate = 1,
  margin = FALSE,
  margin_label = TRUE,
  permanova_res = NULL,
  text_param = list(),
  box_margin = TRUE,
  box_param = list(),
  pal = NULL,
  sample_label = TRUE,
  stat_ellipse = TRUE,
  coord_fix = FALSE,
  bi_text_size = 3,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.b_res_+3A_x">x</code></td>
<td>
<p>a b_res object</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_group">Group</code></td>
<td>
<p>group vector for color</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_metadata">metadata</code></td>
<td>
<p>metadata contain Group</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_group2">Group2</code></td>
<td>
<p>mapping point shape</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_mode">mode</code></td>
<td>
<p>plot mode:1~3</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_bi">bi</code></td>
<td>
<p>plot variables segments?</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_topn">Topn</code></td>
<td>
<p>how many variables to show?</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_rate">rate</code></td>
<td>
<p>segments length rate</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_margin">margin</code></td>
<td>
<p>plot the margin boxplot?</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_margin_label">margin_label</code></td>
<td>
<p>margin_label, TRUE</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_permanova_res">permanova_res</code></td>
<td>
<p>permanova result</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_text_param">text_param</code></td>
<td>
<p>text_param for <code><a href="ggplot2.html#topic+annotate">annotate</a></code></p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_box_margin">box_margin</code></td>
<td>
<p>margin plot box or density?</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_box_param">box_param</code></td>
<td>
<p>box_param for <code><a href="pcutils.html#topic+group_box">group_box</a></code></p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_pal">pal</code></td>
<td>
<p>colors for group</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_sample_label">sample_label</code></td>
<td>
<p>plot the labels of samples?</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_stat_ellipse">stat_ellipse</code></td>
<td>
<p>plot the stat_ellipse?</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_coord_fix">coord_fix</code></td>
<td>
<p>fix the coordinates y/x ratio</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_bi_text_size">bi_text_size</code></td>
<td>
<p>biplot text size</p>
</td></tr>
<tr><td><code id="plot.b_res_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+b_analyse">b_analyse</a></code>
</p>

<hr>
<h2 id='plot.g_test'>Plot g_test</h2><span id='topic+plot.g_test'></span>

<h3>Description</h3>

<p>Plot g_test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'g_test'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.g_test_+3A_x">x</code></td>
<td>
<p>a g_test object</p>
</td></tr>
<tr><td><code id="plot.g_test_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+permanova">permanova</a></code>
</p>

<hr>
<h2 id='plot.pro_res'>Plot pro_res</h2><span id='topic+plot.pro_res'></span>

<h3>Description</h3>

<p>Plot pro_res
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pro_res'
plot(x, group, metadata = NULL, pal = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.pro_res_+3A_x">x</code></td>
<td>
<p>pro_res</p>
</td></tr>
<tr><td><code id="plot.pro_res_+3A_group">group</code></td>
<td>
<p>group</p>
</td></tr>
<tr><td><code id="plot.pro_res_+3A_metadata">metadata</code></td>
<td>
<p>metadata</p>
</td></tr>
<tr><td><code id="plot.pro_res_+3A_pal">pal</code></td>
<td>
<p>pal</p>
</td></tr>
<tr><td><code id="plot.pro_res_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot
</p>

<hr>
<h2 id='plot.time_cm'>Plot time_cm</h2><span id='topic+plot.time_cm'></span>

<h3>Description</h3>

<p>Plot time_cm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'time_cm'
plot(x, mem_thr = 0.6, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.time_cm_+3A_x">x</code></td>
<td>
<p>time_cm</p>
</td></tr>
<tr><td><code id="plot.time_cm_+3A_mem_thr">mem_thr</code></td>
<td>
<p>membership threshold</p>
</td></tr>
<tr><td><code id="plot.time_cm_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>

<hr>
<h2 id='pre_fastp'>Prepare the result from fastp (.json file)</h2><span id='topic+pre_fastp'></span>

<h3>Description</h3>

<p>Prepare the result from fastp (.json file)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pre_fastp(jsonfiles, prefix = c("Raw", "Clean"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pre_fastp_+3A_jsonfiles">jsonfiles</code></td>
<td>
<p>the directory contains .json file</p>
</td></tr>
<tr><td><code id="pre_fastp_+3A_prefix">prefix</code></td>
<td>
<p>default c(&quot;Raw&quot;,&quot;Clean&quot;), for the before filtering and after filtering.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>

<hr>
<h2 id='pre_tax_table'>Complete a taxonomy table</h2><span id='topic+pre_tax_table'></span>

<h3>Description</h3>

<p>Complete a taxonomy table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pre_tax_table(
  tax_table,
  tax_levels = c("k", "p", "c", "o", "f", "g", "s", "st"),
  na_tax = "Unclassified|uncultured|Ambiguous|Unknown|unknown|metagenome|Unassig",
  ignore.case = TRUE,
  na_repalce = "Unknown"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pre_tax_table_+3A_tax_table">tax_table</code></td>
<td>
<p>taxonomy table</p>
</td></tr>
<tr><td><code id="pre_tax_table_+3A_tax_levels">tax_levels</code></td>
<td>
<p>a vector whose length longer than <code>ncol(taxdf)</code>, use to be prefix. Default: c(&quot;k&quot;, &quot;p&quot;, &quot;c&quot;, &quot;o&quot;, &quot;f&quot;, &quot;g&quot;,&quot;s&quot;, &quot;st&quot;)</p>
</td></tr>
<tr><td><code id="pre_tax_table_+3A_na_tax">na_tax</code></td>
<td>
<p>grepl some words and turn to <code>na_repalce</code>, default: &quot;Unclassified|uncultured|Ambiguous|Unknown|unknown|metagenome|Unassig&quot;</p>
</td></tr>
<tr><td><code id="pre_tax_table_+3A_ignore.case">ignore.case</code></td>
<td>
<p>ignore.case for <code>na_tax</code></p>
</td></tr>
<tr><td><code id="pre_tax_table_+3A_na_repalce">na_repalce</code></td>
<td>
<p>defalut: Unknown</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a good taxonomy table
</p>


<h3>References</h3>

<p><code>MicrobiotaProcess</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>taxmat &lt;- matrix(sample("onelevel", 7 * 2, replace = TRUE), nrow = 2, ncol = 7) %&gt;% as.data.frame()
colnames(taxmat) &lt;- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
pre_tax_table(taxmat)
</code></pre>

<hr>
<h2 id='print.pc_otu'>Print</h2><span id='topic+print.pc_otu'></span>

<h3>Description</h3>

<p>Print
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pc_otu'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.pc_otu_+3A_x">x</code></td>
<td>
<p>pc_otu</p>
</td></tr>
<tr><td><code id="print.pc_otu_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='procrustes_analyse'>Procrustes Rotation of Two Configurations and PROTEST</h2><span id='topic+procrustes_analyse'></span>

<h3>Description</h3>

<p>Procrustes Rotation of Two Configurations and PROTEST
</p>


<h3>Usage</h3>

<pre><code class='language-R'>procrustes_analyse(b_res1, b_res2, nperm = 999, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="procrustes_analyse_+3A_b_res1">b_res1</code></td>
<td>
<p>Target matrix</p>
</td></tr>
<tr><td><code id="procrustes_analyse_+3A_b_res2">b_res2</code></td>
<td>
<p>Matrix to be rotated</p>
</td></tr>
<tr><td><code id="procrustes_analyse_+3A_nperm">nperm</code></td>
<td>
<p>numbers of permutations to perform</p>
</td></tr>
<tr><td><code id="procrustes_analyse_+3A_...">...</code></td>
<td>
<p>additional</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pro_res
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
b_analyse(otutab, method = "pca") -&gt; b_res1
b_analyse(otutab * abs(rnorm(10)), method = "pca") -&gt; b_res2
pro_res &lt;- procrustes_analyse(b_res1, b_res2)
plot(pro_res, "Group", metadata)
</code></pre>

<hr>
<h2 id='rare_curve_sample'>Rare the sample</h2><span id='topic+rare_curve_sample'></span><span id='topic+plot.rare_res'></span>

<h3>Description</h3>

<p>Rare the sample
</p>
<p>Plot a rare curve
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rare_curve_sample(otutab, rep = 30, count_cutoff = 1)

## S3 method for class 'rare_res'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rare_curve_sample_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="rare_curve_sample_+3A_rep">rep</code></td>
<td>
<p>repeats number</p>
</td></tr>
<tr><td><code id="rare_curve_sample_+3A_count_cutoff">count_cutoff</code></td>
<td>
<p>cutoff to be 0</p>
</td></tr>
<tr><td><code id="rare_curve_sample_+3A_x">x</code></td>
<td>
<p>AOR object</p>
</td></tr>
<tr><td><code id="rare_curve_sample_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>
<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
a &lt;- rare_curve_sample(otutab)
plot(a)
</code></pre>

<hr>
<h2 id='rare_curve_species'>Rare the species</h2><span id='topic+rare_curve_species'></span>

<h3>Description</h3>

<p>Rare the species
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rare_curve_species(
  otutab,
  step = 2000,
  method = "richness",
  mode = 2,
  reps = 3,
  threads = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rare_curve_species_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="rare_curve_species_+3A_step">step</code></td>
<td>
<p>default 2000</p>
</td></tr>
<tr><td><code id="rare_curve_species_+3A_method">method</code></td>
<td>
<p>one of &quot;richness&quot;,&quot;chao1&quot;,&quot;ace&quot;,&quot;gc&quot;,&quot;shannon&quot;,&quot;simpson&quot;,&quot;pd&quot;,&quot;pielou&quot;</p>
</td></tr>
<tr><td><code id="rare_curve_species_+3A_mode">mode</code></td>
<td>
<p>1 for little table, 2 for big</p>
</td></tr>
<tr><td><code id="rare_curve_species_+3A_reps">reps</code></td>
<td>
<p>reps</p>
</td></tr>
<tr><td><code id="rare_curve_species_+3A_threads">threads</code></td>
<td>
<p>use how many threads to calculate (default:1)</p>
</td></tr>
<tr><td><code id="rare_curve_species_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
a &lt;- rare_curve_species(otutab, mode = 1)
plot(a)
</code></pre>

<hr>
<h2 id='rarefaction'>Rarefy a otutab</h2><span id='topic+rarefaction'></span>

<h3>Description</h3>

<p>Rarefy a otutab
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rarefaction(otutab, sample = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rarefaction_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="rarefaction_+3A_sample">sample</code></td>
<td>
<p>number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a rarefied otutab
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
rarefaction(otutab)
</code></pre>

<hr>
<h2 id='RCbray1'>Calculate RCbray-curtis</h2><span id='topic+RCbray1'></span>

<h3>Description</h3>

<p>Calculate RCbray-curtis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RCbray1(
  otutab,
  reps = 9,
  threads = 1,
  classic_metric = TRUE,
  split_ties = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RCbray1_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="RCbray1_+3A_reps">reps</code></td>
<td>
<p>how many simulation performed?</p>
</td></tr>
<tr><td><code id="RCbray1_+3A_threads">threads</code></td>
<td>
<p>use how many threads to calculate (default:4)</p>
</td></tr>
<tr><td><code id="RCbray1_+3A_classic_metric">classic_metric</code></td>
<td>
<p>standardizes the metric to range from -1 to 1</p>
</td></tr>
<tr><td><code id="RCbray1_+3A_split_ties">split_ties</code></td>
<td>
<p>adds half of the number of null observations that are equal to the observed number of shared species to the calculation- this is highly recommended</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Parallelized version of the Raup-Crick algorithm for &quot;abundance&quot; data (Stegen et al. 2013).
</p>


<h3>Value</h3>

<p>a dist
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("picante")) {
  data(otutab, package = "pcutils")
  df2tree(taxonomy) -&gt; phylo
  b_NTI1(phylo, otutab) -&gt; bnti_res
  RCbray1(otutab, reps = 9) -&gt; rc_res

  data.frame(
    type = factor(c("Homo_S", "Heter_S", "Homo_D", "D_limit", "Undominated"),
      levels = c("Homo_S", "Heter_S", "Homo_D", "D_limit", "Undominated")
    ),
    number = c(
      sum(bnti_res &lt; (-2)), sum(bnti_res &gt; 2),
      sum((abs(bnti_res) &lt; 2) &amp; (abs(rc_res) &lt; 0.95)),
      sum((abs(bnti_res) &lt; 2) &amp; (rc_res &lt; (-0.95))),
      sum((abs(bnti_res) &lt; 2) &amp; (rc_res &gt; 0.95))
    )
  ) -&gt; com_pro
  pcutils::gghuan(com_pro, reorder = FALSE)
}

</code></pre>

<hr>
<h2 id='RDA_plot'>Plot RDA res</h2><span id='topic+RDA_plot'></span>

<h3>Description</h3>

<p>Plot RDA res
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RDA_plot(
  phy.rda,
  Group,
  metadata = NULL,
  Group2 = NULL,
  env_rate = 1,
  mode = 1,
  tri = FALSE,
  Topn = 10,
  rate = 1,
  margin = FALSE,
  box = TRUE,
  pal = NULL,
  sample_label = TRUE,
  stat_ellipse = TRUE,
  coord_fix = FALSE,
  bi_text_size = 3,
  env_text_param = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RDA_plot_+3A_phy.rda">phy.rda</code></td>
<td>
<p>rda/cca object</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_group">Group</code></td>
<td>
<p>group vector for color</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_metadata">metadata</code></td>
<td>
<p>metadata contain Group</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_group2">Group2</code></td>
<td>
<p>mapping point shape</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_env_rate">env_rate</code></td>
<td>
<p>default 1</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_mode">mode</code></td>
<td>
<p>plot mode:1~3</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_tri">tri</code></td>
<td>
<p>plot variables segments?</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_topn">Topn</code></td>
<td>
<p>how many variables to show?</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_rate">rate</code></td>
<td>
<p>segments length rate</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_margin">margin</code></td>
<td>
<p>plot the margin boxplot?</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_box">box</code></td>
<td>
<p>margin plot box or density?</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_pal">pal</code></td>
<td>
<p>colors for group</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_sample_label">sample_label</code></td>
<td>
<p>plot the labels of samples?</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_stat_ellipse">stat_ellipse</code></td>
<td>
<p>plot the stat_ellipse?</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_coord_fix">coord_fix</code></td>
<td>
<p>fix the coordinates y/x ratio</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_bi_text_size">bi_text_size</code></td>
<td>
<p>biplot text size</p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_env_text_param">env_text_param</code></td>
<td>
<p>parameters pass to <code><a href="ggplot2.html#topic+geom_text">geom_text</a></code></p>
</td></tr>
<tr><td><code id="RDA_plot_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+myRDA">myRDA</a></code>
</p>

<hr>
<h2 id='suijisenlin'>RandomForest</h2><span id='topic+suijisenlin'></span>

<h3>Description</h3>

<p>RandomForest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>suijisenlin(otutab, group_df, topN = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="suijisenlin_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="suijisenlin_+3A_group_df">group_df</code></td>
<td>
<p>a dataframe with rowname same to dist and one group column</p>
</td></tr>
<tr><td><code id="suijisenlin_+3A_topn">topN</code></td>
<td>
<p>default: 10</p>
</td></tr>
</table>


<h3>Value</h3>

<p>diff
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("randomForest")) {
  data(otutab, package = "pcutils")
  suijisenlin(otutab, metadata["Group"]) -&gt; rf_res
}
</code></pre>

<hr>
<h2 id='summary.pc_otu'>Summary pc_otu</h2><span id='topic+summary.pc_otu'></span>

<h3>Description</h3>

<p>Summary pc_otu
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pc_otu'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.pc_otu_+3A_object">object</code></td>
<td>
<p>pc_otu</p>
</td></tr>
<tr><td><code id="summary.pc_otu_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("pc_tax1")
summary(pc_tax1)

</code></pre>

<hr>
<h2 id='taxonkit_filter'>Filter TaxIDs based on Taxonomic Ranks</h2><span id='topic+taxonkit_filter'></span>

<h3>Description</h3>

<p>This function uses the &quot;taxonkit filter&quot; command to filter TaxIDs based on taxonomic ranks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonkit_filter(
  file_path,
  black_list = NULL,
  discard_noranks = FALSE,
  discard_root = FALSE,
  equal_to = NULL,
  higher_than = NULL,
  lower_than = NULL,
  rank_file = NULL,
  root_taxid = NULL,
  save_predictable_norank = FALSE,
  taxid_field = NULL,
  text = FALSE,
  data_dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxonkit_filter_+3A_file_path">file_path</code></td>
<td>
<p>The path to the input file containing TaxIDs. Or file text (text=TRUE)</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_black_list">black_list</code></td>
<td>
<p>A character vector specifying the ranks to discard.</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_discard_noranks">discard_noranks</code></td>
<td>
<p>Logical value indicating whether to discard all ranks without order (default is FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_discard_root">discard_root</code></td>
<td>
<p>Logical value indicating whether to discard the root taxid (default is FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_equal_to">equal_to</code></td>
<td>
<p>A character vector specifying the ranks for which TaxIDs should be equal to.</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_higher_than">higher_than</code></td>
<td>
<p>The rank above which the TaxIDs should be (exclusive).</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_lower_than">lower_than</code></td>
<td>
<p>The rank below which the TaxIDs should be (exclusive).</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_rank_file">rank_file</code></td>
<td>
<p>The path to a user-defined ordered taxonomic ranks file.</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_root_taxid">root_taxid</code></td>
<td>
<p>The root taxid (default is 1).</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_save_predictable_norank">save_predictable_norank</code></td>
<td>
<p>Logical value indicating whether to save some special ranks without order when using lower_than (default is FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_taxid_field">taxid_field</code></td>
<td>
<p>The field index of the taxid in the input file (default is 1).</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_text">text</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="taxonkit_filter_+3A_data_dir">data_dir</code></td>
<td>
<p>directory containing nodes.dmp and names.dmp (default &quot;/Users/asa/.taxonkit&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the output of the &quot;taxonkit filter&quot; command.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
taxids2 &lt;- system.file("extdata/taxids2.txt", package = "pctax")
taxonkit_filter(taxids2, lower_than = "genus")

## End(Not run)
</code></pre>

<hr>
<h2 id='taxonkit_lca'>Compute Lowest Common Ancestor (LCA) of TaxIDs</h2><span id='topic+taxonkit_lca'></span>

<h3>Description</h3>

<p>This function uses the &quot;taxonkit lca&quot; command to compute the Lowest Common Ancestor (LCA) of TaxIDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonkit_lca(
  file_path,
  buffer_size = "1M",
  separator = " ",
  skip_deleted = FALSE,
  skip_unfound = FALSE,
  taxids_field = NULL,
  text = FALSE,
  data_dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxonkit_lca_+3A_file_path">file_path</code></td>
<td>
<p>The path to the input file containing TaxIDs. Or file text (text=TRUE)</p>
</td></tr>
<tr><td><code id="taxonkit_lca_+3A_buffer_size">buffer_size</code></td>
<td>
<p>The size of the line buffer (supported units: K, M, G).</p>
</td></tr>
<tr><td><code id="taxonkit_lca_+3A_separator">separator</code></td>
<td>
<p>The separator for TaxIDs.</p>
</td></tr>
<tr><td><code id="taxonkit_lca_+3A_skip_deleted">skip_deleted</code></td>
<td>
<p>Whether to skip deleted TaxIDs and compute with the remaining ones.</p>
</td></tr>
<tr><td><code id="taxonkit_lca_+3A_skip_unfound">skip_unfound</code></td>
<td>
<p>Whether to skip unfound TaxIDs and compute with the remaining ones.</p>
</td></tr>
<tr><td><code id="taxonkit_lca_+3A_taxids_field">taxids_field</code></td>
<td>
<p>The field index of TaxIDs. Input data should be tab-separated (default 1).</p>
</td></tr>
<tr><td><code id="taxonkit_lca_+3A_text">text</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="taxonkit_lca_+3A_data_dir">data_dir</code></td>
<td>
<p>directory containing nodes.dmp and names.dmp (default &quot;/Users/asa/.taxonkit&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the computed LCAs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
taxonkit_lca("239934, 239935, 349741", text = TRUE, separator = ", ")

## End(Not run)
</code></pre>

<hr>
<h2 id='taxonkit_lineage'>Retrieve Taxonomic Lineage using taxonkit</h2><span id='topic+taxonkit_lineage'></span>

<h3>Description</h3>

<p>Retrieve Taxonomic Lineage using taxonkit
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonkit_lineage(
  file_path,
  delimiter = ";",
  no_lineage = FALSE,
  show_lineage_ranks = FALSE,
  show_lineage_taxids = FALSE,
  show_name = FALSE,
  show_rank = FALSE,
  show_status_code = FALSE,
  taxid_field = 1,
  text = FALSE,
  data_dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxonkit_lineage_+3A_file_path">file_path</code></td>
<td>
<p>The path to the input file with taxonomic IDs. Or file text (text=TRUE)</p>
</td></tr>
<tr><td><code id="taxonkit_lineage_+3A_delimiter">delimiter</code></td>
<td>
<p>The field delimiter in the lineage (default &quot;;&quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_lineage_+3A_no_lineage">no_lineage</code></td>
<td>
<p>Logical, indicating whether to exclude lineage information (default: FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_lineage_+3A_show_lineage_ranks">show_lineage_ranks</code></td>
<td>
<p>Logical, indicating whether to append ranks of all levels in the lineage (default: FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_lineage_+3A_show_lineage_taxids">show_lineage_taxids</code></td>
<td>
<p>Logical, indicating whether to append lineage consisting of taxids (default: FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_lineage_+3A_show_name">show_name</code></td>
<td>
<p>Logical, indicating whether to append scientific name (default: FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_lineage_+3A_show_rank">show_rank</code></td>
<td>
<p>Logical, indicating whether to append rank of taxids (default: FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_lineage_+3A_show_status_code">show_status_code</code></td>
<td>
<p>Logical, indicating whether to show status code before lineage (default: FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_lineage_+3A_taxid_field">taxid_field</code></td>
<td>
<p>The field index of taxid. Input data should be tab-separated (default: 1).</p>
</td></tr>
<tr><td><code id="taxonkit_lineage_+3A_text">text</code></td>
<td>
<p>logical,</p>
</td></tr>
<tr><td><code id="taxonkit_lineage_+3A_data_dir">data_dir</code></td>
<td>
<p>directory containing nodes.dmp and names.dmp (default &quot;/Users/asa/.taxonkit&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the taxonomic lineage information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
taxonkit_lineage("9606\n63221", show_name = TRUE, show_rank = TRUE, text = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='taxonkit_list'>Taxonkit list</h2><span id='topic+taxonkit_list'></span>

<h3>Description</h3>

<p>This function uses Taxonkit to perform the &quot;list&quot; operation, which retrieves
information about taxa based on their TaxIDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonkit_list(
  ids,
  indent = "  ",
  json = FALSE,
  show_name = FALSE,
  show_rank = FALSE,
  data_dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxonkit_list_+3A_ids">ids</code></td>
<td>
<p>A character vector of TaxIDs to retrieve information for.</p>
</td></tr>
<tr><td><code id="taxonkit_list_+3A_indent">indent</code></td>
<td>
<p>The indentation string to use for pretty-printing the output. Default is &quot;  &quot;.</p>
</td></tr>
<tr><td><code id="taxonkit_list_+3A_json">json</code></td>
<td>
<p>Logical value indicating whether to output the result in JSON format. Default is FALSE.</p>
</td></tr>
<tr><td><code id="taxonkit_list_+3A_show_name">show_name</code></td>
<td>
<p>Logical value indicating whether to show the scientific names of taxa. Default is FALSE.</p>
</td></tr>
<tr><td><code id="taxonkit_list_+3A_show_rank">show_rank</code></td>
<td>
<p>Logical value indicating whether to show the ranks of taxa. Default is FALSE.</p>
</td></tr>
<tr><td><code id="taxonkit_list_+3A_data_dir">data_dir</code></td>
<td>
<p>directory containing nodes.dmp and names.dmp (default &quot;/Users/asa/.taxonkit&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output of the Taxonkit list operation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
taxonkit_list(ids = c(9605), indent = "-", show_name = TRUE, show_rank = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='taxonkit_name2taxid'>Convert Taxonomic Names to TaxIDs</h2><span id='topic+taxonkit_name2taxid'></span>

<h3>Description</h3>

<p>This function uses the &quot;taxonkit taxonkit_name2taxid&quot; command to convert taxonomic names to corresponding taxonomic IDs (TaxIDs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonkit_name2taxid(
  file_path,
  name_field = NULL,
  sci_name = FALSE,
  show_rank = FALSE,
  text = FALSE,
  data_dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxonkit_name2taxid_+3A_file_path">file_path</code></td>
<td>
<p>The path to the input file containing taxonomic names. Or file text (text=TRUE)</p>
</td></tr>
<tr><td><code id="taxonkit_name2taxid_+3A_name_field">name_field</code></td>
<td>
<p>The field index of the taxonomic name in the input file (default is 1).</p>
</td></tr>
<tr><td><code id="taxonkit_name2taxid_+3A_sci_name">sci_name</code></td>
<td>
<p>Logical value indicating whether to search only for scientific names (default is FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_name2taxid_+3A_show_rank">show_rank</code></td>
<td>
<p>Logical value indicating whether to show the taxonomic rank in the output (default is FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_name2taxid_+3A_text">text</code></td>
<td>
<p>Logical</p>
</td></tr>
<tr><td><code id="taxonkit_name2taxid_+3A_data_dir">data_dir</code></td>
<td>
<p>directory containing nodes.dmp and names.dmp (default &quot;/Users/asa/.taxonkit&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the output of the &quot;taxonkit_name2taxid&quot; command.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
names &lt;- system.file("extdata/name.txt", package = "pctax")
taxonkit_name2taxid(names, name_field = 1, sci_name = FALSE, show_rank = FALSE)
"Homo sapiens" %&gt;% taxonkit_name2taxid(text = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='taxonkit_reformat'>Reformat Taxonomic Lineage using taxonkit</h2><span id='topic+taxonkit_reformat'></span>

<h3>Description</h3>

<p>Reformat Taxonomic Lineage using taxonkit
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxonkit_reformat(
  file_path,
  delimiter = NULL,
  add_prefix = FALSE,
  prefix_kingdom = "K__",
  prefix_phylum = "p__",
  prefix_class = "c__",
  prefix_order = "o__",
  prefix_family = "f__",
  prefix_genus = "g__",
  prefix_species = "s__",
  prefix_subspecies = "t__",
  prefix_strain = "T__",
  fill_miss_rank = FALSE,
  format_string = "",
  miss_rank_repl_prefix = "unclassified ",
  miss_rank_repl = "",
  miss_taxid_repl = "",
  output_ambiguous_result = FALSE,
  lineage_field = 2,
  taxid_field = NULL,
  pseudo_strain = FALSE,
  trim = FALSE,
  text = FALSE,
  data_dir = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxonkit_reformat_+3A_file_path">file_path</code></td>
<td>
<p>The path to the input file with taxonomic lineages. Or file text (text=TRUE)</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_delimiter">delimiter</code></td>
<td>
<p>The field delimiter in the input lineage (default &quot;;&quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_add_prefix">add_prefix</code></td>
<td>
<p>Logical, indicating whether to add prefixes for all ranks (default: FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_prefix_kingdom">prefix_kingdom</code></td>
<td>
<p>The prefix for kingdom, used along with &ndash;add-prefix (default: &quot;K__&quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_prefix_phylum">prefix_phylum</code></td>
<td>
<p>The prefix for phylum, used along with &ndash;add-prefix (default: &quot;p__&quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_prefix_class">prefix_class</code></td>
<td>
<p>The prefix for class, used along with &ndash;add-prefix (default: &quot;c__&quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_prefix_order">prefix_order</code></td>
<td>
<p>The prefix for order, used along with &ndash;add-prefix (default: &quot;o__&quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_prefix_family">prefix_family</code></td>
<td>
<p>The prefix for family, used along with &ndash;add-prefix (default: &quot;f__&quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_prefix_genus">prefix_genus</code></td>
<td>
<p>The prefix for genus, used along with &ndash;add-prefix (default: &quot;g__&quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_prefix_species">prefix_species</code></td>
<td>
<p>The prefix for species, used along with &ndash;add-prefix (default: &quot;s__&quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_prefix_subspecies">prefix_subspecies</code></td>
<td>
<p>The prefix for subspecies, used along with &ndash;add-prefix (default: &quot;t__&quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_prefix_strain">prefix_strain</code></td>
<td>
<p>The prefix for strain, used along with &ndash;add-prefix (default: &quot;T__&quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_fill_miss_rank">fill_miss_rank</code></td>
<td>
<p>Logical, indicating whether to fill missing rank with lineage information of the next higher rank (default: FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_format_string">format_string</code></td>
<td>
<p>The output format string with placeholders for each rank.</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_miss_rank_repl_prefix">miss_rank_repl_prefix</code></td>
<td>
<p>The prefix for estimated taxon level for missing rank (default: &quot;unclassified &quot;).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_miss_rank_repl">miss_rank_repl</code></td>
<td>
<p>The replacement string for missing rank.</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_miss_taxid_repl">miss_taxid_repl</code></td>
<td>
<p>The replacement string for missing taxid.</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_output_ambiguous_result">output_ambiguous_result</code></td>
<td>
<p>Logical, indicating whether to output one of the ambiguous result (default: FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_lineage_field">lineage_field</code></td>
<td>
<p>The field index of lineage. Input data should be tab-separated (default: 2).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_taxid_field">taxid_field</code></td>
<td>
<p>The field index of taxid. Input data should be tab-separated. It overrides -i/&ndash;lineage-field.</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_pseudo_strain">pseudo_strain</code></td>
<td>
<p>Logical, indicating whether to use the node with lowest rank as strain name (default: FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_trim">trim</code></td>
<td>
<p>Logical, indicating whether to not fill missing rank lower than current rank (default: FALSE).</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_text">text</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="taxonkit_reformat_+3A_data_dir">data_dir</code></td>
<td>
<p>directory containing nodes.dmp and names.dmp (default &quot;/Users/asa/.taxonkit&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing the reformatted taxonomic lineages.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# Use taxid
taxids2 &lt;- system.file("extdata/taxids2.txt", package = "pctax")
reformatted_lineages &lt;- taxonkit_reformat(taxids2,
  add_prefix = TRUE, taxid_field = 1, fill_miss_rank = TRUE
)
reformatted_lineages
taxonomy &lt;- strsplit2(reformatted_lineages, "\t")
taxonomy &lt;- strsplit2(taxonomy$V2, ";")

# Use lineage result
taxonkit_lineage("9606\n63221", show_name = TRUE, show_rank = TRUE, text = TRUE) %&gt;%
  taxonkit_reformat(text = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='time_by_cm'>Time series analysis</h2><span id='topic+time_by_cm'></span>

<h3>Description</h3>

<p>Time series analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>time_by_cm(otu_time, n_cluster = 6, min.std = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="time_by_cm_+3A_otu_time">otu_time</code></td>
<td>
<p>otutab hebing by a time variable</p>
</td></tr>
<tr><td><code id="time_by_cm_+3A_n_cluster">n_cluster</code></td>
<td>
<p>number of clusters</p>
</td></tr>
<tr><td><code id="time_by_cm_+3A_min.std">min.std</code></td>
<td>
<p>min.std</p>
</td></tr>
</table>


<h3>Value</h3>

<p>time_cm
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (interactive()) {
  data(otutab, package = "pcutils")
  otu_time &lt;- pcutils::hebing(otutab, metadata$Group)
  time_by_cm(otu_time, n_cluster = 4) -&gt; time_cm_res
  plot(time_cm_res)
}

</code></pre>

<hr>
<h2 id='volcano_p'>Volcano plot for difference analysis</h2><span id='topic+volcano_p'></span>

<h3>Description</h3>

<p>Volcano plot for difference analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volcano_p(
  res,
  logfc = 1,
  adjp = 0.05,
  text = TRUE,
  repel = TRUE,
  mode = 1,
  number = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="volcano_p_+3A_res">res</code></td>
<td>
<p>result of <code>diff_da</code> which have colnames: tax, log2FoldChange, padj, compare, sig</p>
</td></tr>
<tr><td><code id="volcano_p_+3A_logfc">logfc</code></td>
<td>
<p>log_fold_change threshold</p>
</td></tr>
<tr><td><code id="volcano_p_+3A_adjp">adjp</code></td>
<td>
<p>adjust_p_value threshold</p>
</td></tr>
<tr><td><code id="volcano_p_+3A_text">text</code></td>
<td>
<p>text, TRUE</p>
</td></tr>
<tr><td><code id="volcano_p_+3A_repel">repel</code></td>
<td>
<p>repel, TRUE</p>
</td></tr>
<tr><td><code id="volcano_p_+3A_mode">mode</code></td>
<td>
<p>1:normal; 2:multi_contrast</p>
</td></tr>
<tr><td><code id="volcano_p_+3A_number">number</code></td>
<td>
<p>show the tax number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+diff_da">diff_da</a></code>
</p>

<hr>
<h2 id='z_diversity'>Calculate Zeta Diversity</h2><span id='topic+z_diversity'></span><span id='topic+plot.zeta_res'></span>

<h3>Description</h3>

<p>This function calculates Zeta diversity for each group in the provided otutab.
</p>
<p>This function plots the Zeta diversity results obtained from the z_diversity function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>z_diversity(otutab, group_df = NULL, zetadiv_params = list())

## S3 method for class 'zeta_res'
plot(x, lm_model = c("exp", "pl")[1], ribbon = FALSE, text = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="z_diversity_+3A_otutab">otutab</code></td>
<td>
<p>A matrix or data frame containing OTU (Operational Taxonomic Unit) counts.</p>
</td></tr>
<tr><td><code id="z_diversity_+3A_group_df">group_df</code></td>
<td>
<p>A data frame containing group information.</p>
</td></tr>
<tr><td><code id="z_diversity_+3A_zetadiv_params">zetadiv_params</code></td>
<td>
<p>Additional parameters to be passed to the Zeta.decline.mc function from the zetadiv package.</p>
</td></tr>
<tr><td><code id="z_diversity_+3A_x">x</code></td>
<td>
<p>Zeta diversity results obtained from z_diversity function.</p>
</td></tr>
<tr><td><code id="z_diversity_+3A_lm_model">lm_model</code></td>
<td>
<p>The linear model to be used for fitting ('exp' or 'pl').</p>
</td></tr>
<tr><td><code id="z_diversity_+3A_ribbon">ribbon</code></td>
<td>
<p>Logical, whether to add a ribbon to the plot for standard deviation.</p>
</td></tr>
<tr><td><code id="z_diversity_+3A_text">text</code></td>
<td>
<p>Logical, whether to add R-squared and p-value text annotations.</p>
</td></tr>
<tr><td><code id="z_diversity_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to ggplot2 functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>zeta_res
</p>
<p>A ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("zetadiv")) {
  data(otutab, package = "pcutils")
  zeta_result &lt;- z_diversity(otutab, metadata["Group"], zetadiv_params = list(sam = 10))
  plot(zeta_result, lm_model = "exp", text = TRUE)
}
</code></pre>

<hr>
<h2 id='z_diversity_decay'>Calculate Zeta Diversity with Distance</h2><span id='topic+z_diversity_decay'></span><span id='topic+plot.zeta_decay'></span>

<h3>Description</h3>

<p>This function calculates Zeta diversity for each group in the provided otutab.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>z_diversity_decay(otutab, xy_df, group_df = NULL, zetadiv_params = list())

## S3 method for class 'zeta_decay'
plot(x, ribbon = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="z_diversity_decay_+3A_otutab">otutab</code></td>
<td>
<p>A matrix or data frame containing OTU (Operational Taxonomic Unit) counts.</p>
</td></tr>
<tr><td><code id="z_diversity_decay_+3A_xy_df">xy_df</code></td>
<td>
<p>Site coordinates.</p>
</td></tr>
<tr><td><code id="z_diversity_decay_+3A_group_df">group_df</code></td>
<td>
<p>A data frame containing group information.</p>
</td></tr>
<tr><td><code id="z_diversity_decay_+3A_zetadiv_params">zetadiv_params</code></td>
<td>
<p>Additional parameters to be passed to the Zeta.ddecay function from the zetadiv package.</p>
</td></tr>
<tr><td><code id="z_diversity_decay_+3A_x">x</code></td>
<td>
<p>Zeta diversity results obtained from z_diversity_decay function.</p>
</td></tr>
<tr><td><code id="z_diversity_decay_+3A_ribbon">ribbon</code></td>
<td>
<p>Logical, whether to add a ribbon to the plot for standard deviation.</p>
</td></tr>
<tr><td><code id="z_diversity_decay_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to ggplot2 functions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>zeta_decay
</p>
<p>A ggplot object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("zetadiv")) {
  data(otutab, package = "pcutils")
  zeta_decay_result &lt;- z_diversity_decay(otutab, metadata[, c("lat", "long")],
    metadata["Group"],
    zetadiv_params = list(sam = 10)
  )
  plot(zeta_decay_result)
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
