<!DOCTYPE html><html><head><title>Help for package PEACH</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PEACH}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#KEGG'><p>KEGG pathways</p></a></li>
<li><a href='#peach'><p>Pareto Enrichment Analysis for Combining Heterogeneous datasets</p></a></li>
<li><a href='#TCGA.input'><p>A sample of 16 cancers gene differential expression analysis t statistic score data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Pareto Enrichment Analysis for Combining Heterogeneous Datasets</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Jinyan Chan [aut, cre], Jinghua Gu [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jinyan Chan &lt;Jinyan_Chan@baylor.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A meta gene set analysis tool developed based on principles of Pareto dominance (William B T Mock (2011) &lt;<a href="https://doi.org/10.1007%2F978-1-4020-9160-5_341">doi:10.1007/978-1-4020-9160-5_341</a>&gt;). It is designed to combine gene set analysis p-values from multiple transcriptome datasets (e.g., microarray and RNA-Seq). The novel Pareto method for p-value combination allows PEACH to properly model heterogeneity and correlation in Omics datasets.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0),MASS (&ge; 7.3-50), mnormt (&ge; 1.5-6), metap (&ge;
1.3)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-12-19 02:33:05 UTC; jinyanchan</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-07 17:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='KEGG'>KEGG pathways</h2><span id='topic+KEGG'></span>

<h3>Description</h3>

<p>This data set gives the geneset list of KEGG pathways
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KEGG
</code></pre>


<h3>Format</h3>

<p>A large list containing 186 pathways.</p>


<h3>Source</h3>

<p>KEGG genesets from GSEA MSigDB Collections</p>

<hr>
<h2 id='peach'>Pareto Enrichment Analysis for Combining Heterogeneous datasets</h2><span id='topic+peach'></span>

<h3>Description</h3>

<p>This function is for pathway enrichment meta-analysis with Pareto dominance based method. The input of this function is the gene level test-statistics (e.g. t statistics) from multiple datasets on which meta analysis will be performed, and a pathway (or geneset) list. It outputs the pathway p-values for each individual dataset as well as the pareto combined pathway p-values in a data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peach(
  input.data = NULL,
  nsample = 1000,
  input.pathway = NULL,
  direction = "up",
  is.Fisher.Stouffer = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peach_+3A_input.data">input.data</code></td>
<td>
<p>The test statistics of each gene from multiple datasets (the test statistics is from the case versus control statistical test, e.g. t-test). The rows are genes where the rownames are gene names (official gene symbols). The columns are the individual datasets.</p>
</td></tr>
<tr><td><code id="peach_+3A_nsample">nsample</code></td>
<td>
<p>The number of random sampling times for Pareto meta-analysis p-value calculation. As Pareto based meta-analysis is a non-parametric method, this parameter decides the NULL distribution size of meta-pathway p-value computing.</p>
</td></tr>
<tr><td><code id="peach_+3A_input.pathway">input.pathway</code></td>
<td>
<p>the pathways or genesets in the format of lists. The pathways or genesets should be defined by official gene symbols. An example KEGG pathway can be obtained with data('KEGG'). (The pathway input format is the same with the output from the 'gmtPathways' function from fgsea package.)</p>
</td></tr>
<tr><td><code id="peach_+3A_direction">direction</code></td>
<td>
<p>&quot;up&quot; or &quot;down&quot; denoting if the pathway p-value is calculated by accounting for pathway up-regulation or down-regulation. The default is &quot;up&quot;, which means the peach function calculated combined p-value indicates if a pathway is up-regulated across the datasets being combined.</p>
</td></tr>
<tr><td><code id="peach_+3A_is.fisher.stouffer">is.Fisher.Stouffer</code></td>
<td>
<p>Logical indication. If TRUE, peach function will output the combined meta-pathway-p-value from non-parametric Fisher's and Stouffer's method. The combined p-value will not be the same with the original Fisher's or Stouffer's method, as this version has the Monte Carlo implementation of these two methods that accounts for the correlation from the input dataset.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## load example input data (TCGA cancer versus control test t statistics)
data('TCGA.input')
## load the KEGG pathways
data('KEGG')
## Run peach

res = peach(input.data=TCGA.input,input.pathway=KEGG,direction ="up",is.Fisher.Stouffer = TRUE)

</code></pre>

<hr>
<h2 id='TCGA.input'>A sample of 16 cancers gene differential expression analysis t statistic score data</h2><span id='topic+TCGA.input'></span>

<h3>Description</h3>

<p>This data set gives the TCGA 16 cancer types tumor versus normal t-test scores of each gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TCGA.input
</code></pre>


<h3>Format</h3>

<p>A matrix containing the t statistics of 20501 genes/rows and 16 cancer types/columns</p>


<h3>Source</h3>

<p>TCGA</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
