<!DOCTYPE html><html><head><title>Help for package PeakSegDP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PeakSegDP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calc.grad.list'><p>calc grad list</p></a></li>
<li><a href='#calc.loss.from.lp.list'><p>calc loss from lp list</p></a></li>
<li><a href='#calc.loss.list'><p>calc loss list</p></a></li>
<li><a href='#cDPA'><p>cDPA</p></a></li>
<li><a href='#chr11ChIPseq'>
<p>ChIP-seq aligned read coverage for 4 samples on a subset of chr11</p></a></li>
<li><a href='#chr11first'>
<p>Counts of first base of aligned reads</p></a></li>
<li><a href='#derivs'><p>derivs</p></a></li>
<li><a href='#GeomTallRect'><p>GeomTallRect</p></a></li>
<li><a href='#getPath'><p>getPath</p></a></li>
<li><a href='#H3K36me3.AM.immune.19'>
<p>Several ChIP-seq profiles, some of which have few data points</p></a></li>
<li><a href='#H3K36me3.TDH.other.chunk3.cluster4'>
<p>8 profiles of H3K36me3 data</p></a></li>
<li><a href='#H3K4me3.TDH.immune.chunk12.cluster4'>
<p>Histone ChIP-seq data, 26 samples, chr1 subset</p></a></li>
<li><a href='#PeakSegDP'><p>PeakSegDP</p></a></li>
<li><a href='#phi.list'><p>phi list</p></a></li>
<li><a href='#PoissonLoss'><p>PoissonLoss</p></a></li>
<li><a href='#regression.funs'><p>regression funs</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Maintainer:</td>
<td>Toby Dylan Hocking &lt;toby.hocking@r-project.org&gt;</td>
</tr>
<tr>
<td>Author:</td>
<td>Toby Dylan Hocking, Guillem Rigaill</td>
</tr>
<tr>
<td>Version:</td>
<td>2024.1.24</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/tdhock/PeakSegDP">https://github.com/tdhock/PeakSegDP</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/tdhock/PeakSegDP/issues">https://github.com/tdhock/PeakSegDP/issues</a></td>
</tr>
<tr>
<td>Title:</td>
<td>Dynamic Programming Algorithm for Peak Detection in ChIP-Seq
Data</td>
</tr>
<tr>
<td>Description:</td>
<td>A quadratic time dynamic programming algorithm
 can be used to compute an approximate solution to the problem of
 finding the most likely changepoints
 with respect to the Poisson likelihood, subject
 to a constraint on the number of segments, and the changes which must
 alternate: up, down, up, down, etc. For more info read
 <a href="http://proceedings.mlr.press/v37/hocking15.html">http://proceedings.mlr.press/v37/hocking15.html</a>
 "PeakSeg: constrained optimal segmentation and supervised penalty learning
 for peak detection in count data" by TD Hocking et al,
 proceedings of ICML2015.</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggplot2 (&ge; 2.0), testthat, penaltyLearning</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-24 18:18:03 UTC; th798</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-24 19:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='calc.grad.list'>calc grad list</h2><span id='topic+calc.grad.list'></span>

<h3>Description</h3>

<p>List of calc.grad functions: x, features, limits -&gt; gradient.</p>


<h3>Usage</h3>

<pre><code class='language-R'>"calc.grad.list"</code></pre>

<hr>
<h2 id='calc.loss.from.lp.list'>calc loss from lp list</h2><span id='topic+calc.loss.from.lp.list'></span>

<h3>Description</h3>

<p>if we have already calculated the linear predictor using
fit$predict, this function can be useful.</p>


<h3>Usage</h3>

<pre><code class='language-R'>"calc.loss.from.lp.list"</code></pre>

<hr>
<h2 id='calc.loss.list'>calc loss list</h2><span id='topic+calc.loss.list'></span>

<h3>Description</h3>

<p>List of interval regression loss functions: x, feat, lim =&gt;
numeric.</p>


<h3>Usage</h3>

<pre><code class='language-R'>"calc.loss.list"</code></pre>

<hr>
<h2 id='cDPA'>cDPA</h2><span id='topic+cDPA'></span>

<h3>Description</h3>

<p>A constrained dynamic programming algorithm (cDPA) can be used to
compute the best segmentation with respect to the Poisson
likelihood, subject to a constraint on the number of segments, and
the changes which must alternate: up, down, up, down, ...</p>


<h3>Usage</h3>

<pre><code class='language-R'>cDPA(count, weight = rep(1, 
    length(count)), maxSegments)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cDPA_+3A_count">count</code></td>
<td>
<p>Integer vector of <code>count</code> data to segment.</p>
</td></tr>
<tr><td><code id="cDPA_+3A_weight">weight</code></td>
<td>
<p>Data weights (normally this is the number of base pairs).</p>
</td></tr>
<tr><td><code id="cDPA_+3A_maxsegments">maxSegments</code></td>
<td>
<p>Maximum number of segments to consider.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Toby Dylan Hocking, Guillem Rigaill</p>


<h3>Examples</h3>

<pre><code class='language-R'>fit &lt;- cDPA(c(0, 10, 11, 1), maxSegments=3)
stopifnot(fit$ends[3,4] == 3)
stopifnot(fit$ends[2,3] == 1)
</code></pre>

<hr>
<h2 id='chr11ChIPseq'>
ChIP-seq aligned read coverage for 4 samples on a subset of chr11
</h2><span id='topic+chr11ChIPseq'></span>

<h3>Description</h3>

<p>A ChIP-seq experiment was performed to locate the genomic positions of
a histone (H3K4me3) in 2 B cell samples (McGill0091, McGill0322) and
2 T cell samples (McGill0002, McGill0004). The short sequence reads
(about 100 base pairs each) were aligned to the hg19 reference genome,
and the &quot;coverage&quot; in this data set contains
the total count of aligned reads at each base pair. It also contains
annotated regions determined by an expert who examined scatterplots of
the coverage profiles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("chr11ChIPseq")</code></pre>


<h3>Format</h3>

<p>A named list of 2 data.frames: regions contains annotations
about which regions contain or do not contain peaks, and coverage
contains the noisy signal.</p>


<h3>Source</h3>

<p>H3K4me3_TDH_immune chunk 5 in
http://cbio.ensmp.fr/~thocking/chip-seq-chunk-db/
which in turn comes from
http://epigenomesportal.ca/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(chr11ChIPseq)
library(ggplot2)
ann.colors &lt;-
  c(noPeaks="#f6f4bf",
    peakStart="#ffafaf",
    peakEnd="#ff4c4c",
    peaks="#a445ee")

if(interactive() &amp;&amp; require(ggplot2)){

ggplot()+
  scale_fill_manual("annotation", values=ann.colors,
                    breaks=names(ann.colors))+
  penaltyLearning::geom_tallrect(aes(xmin=chromStart/1e3, xmax=chromEnd/1e3,
                    fill=annotation),
                data=chr11ChIPseq$regions, alpha=1/2)+
  theme_bw()+
  theme(panel.margin=grid::unit(0, "cm"))+
  facet_grid(sample.id ~ ., scales="free")+
  geom_step(aes(chromStart/1e3, count), data=chr11ChIPseq$coverage)+
  xlab("position on chr11 (kilo base pairs)")

}

</code></pre>

<hr>
<h2 id='chr11first'>
Counts of first base of aligned reads
</h2><span id='topic+chr11first'></span>

<h3>Description</h3>

<p>For 4 samples on chr11 (hg19), this data set counts the first base pair of
aligned reads at each genomic position. In contrast, chr11ChIPseq counts
every base pair in each read (and each read is about 100bp, so that
means there is some auto-correlation in chr11ChIPseq, but not in chr11first).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("chr11first")</code></pre>


<h3>Format</h3>

<p>A data frame with 23252 observations on the following 4 variables.
</p>

<dl>
<dt><code>sample.id</code></dt><dd><p>a factor with levels for each of 4 samples</p>
</dd>
<dt><code>chromStart</code></dt><dd><p>integer vector: base before, on chr11</p>
</dd>
<dt><code>chromEnd</code></dt><dd><p>integer vector: last base on chr11</p>
</dd>
<dt><code>count</code></dt><dd><p>integer: aligned first base read counts</p>
</dd>
</dl>



<h3>Source</h3>

<p>H3K4me3_TDH_immune chunk 5 in
http://cbio.ensmp.fr/~thocking/chip-seq-chunk-db/
which in turn comes from
http://epigenomesportal.ca/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(chr11ChIPseq)
data(chr11first)
library(ggplot2)
ann.colors &lt;-
  c(noPeaks="#f6f4bf",
    peakStart="#ffafaf",
    peakEnd="#ff4c4c",
    peaks="#a445ee")
both &lt;- list(coverage=chr11ChIPseq$coverage, first=chr11first)
representations &lt;- NULL
one.sample &lt;- "McGill0322"
for(data.type in names(both)){
  one &lt;- subset(both[[data.type]], sample.id==one.sample)
  representations &lt;- rbind(representations, data.frame(data.type, one))
}
one.sample.regions &lt;- subset(
  chr11ChIPseq$regions, sample.id==one.sample)

if(interactive() &amp;&amp; require(ggplot2)){

ggplot()+
  scale_fill_manual("annotation", values=ann.colors,
                    breaks=names(ann.colors))+
  penaltyLearning::geom_tallrect(aes(xmin=chromStart/1e3, xmax=chromEnd/1e3,
                    fill=annotation),
                data=one.sample.regions, alpha=1/2)+
  theme_bw()+
  theme(panel.margin=grid::unit(0, "cm"))+
  facet_grid(data.type ~ ., scales="free")+
  geom_step(aes(chromStart/1e3, count), data=representations)+
  xlab("position on chr11 (kilo base pairs)")

}

</code></pre>

<hr>
<h2 id='derivs'>derivs</h2><span id='topic+derivs'></span>

<h3>Description</h3>

<p>List of functions, each a derivative of a phi loss.</p>


<h3>Usage</h3>

<pre><code class='language-R'>"derivs"</code></pre>

<hr>
<h2 id='GeomTallRect'>GeomTallRect</h2><span id='topic+GeomTallRect'></span>

<h3>Description</h3>

<p>ggproto object for geom_tallrect</p>


<h3>Usage</h3>

<pre><code class='language-R'>"GeomTallRect"</code></pre>

<hr>
<h2 id='getPath'>getPath</h2><span id='topic+getPath'></span>

<h3>Description</h3>

<p>Extract endpoint matrix from <code><a href="#topic+cDPA">cDPA</a></code> result.</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPath(A)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPath_+3A_a">A</code></td>
<td>
<p>A </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Toby Dylan Hocking, Guillem Rigaill</p>

<hr>
<h2 id='H3K36me3.AM.immune.19'>
Several ChIP-seq profiles, some of which have few data points
</h2><span id='topic+H3K36me3.AM.immune.19'></span>

<h3>Description</h3>

<p>These data are used to test the PeakSegDP algorithm,
to make sure it gives sensible results, even when there are few data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("H3K36me3.AM.immune.19")</code></pre>


<h3>Format</h3>

<p>Named list of 21 data.frames, each with columns chromStart, chromEnd, count.
</p>


<h3>Source</h3>

<p>http://cbio.ensmp.fr/~thocking/chip-seq-chunk-db/
data set H3K36me3_AM_immune, chunk id 19
</p>

<hr>
<h2 id='H3K36me3.TDH.other.chunk3.cluster4'>
8 profiles of H3K36me3 data
</h2><span id='topic+H3K36me3.TDH.other.chunk3.cluster4'></span>

<h3>Description</h3>

<p>these data caused a bug in multiSampleSegHeuristic.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("H3K36me3.TDH.other.chunk3.cluster4")</code></pre>


<h3>Format</h3>

<p>A data frame with 36914 observations on the following 4 variables.
</p>

<dl>
<dt><code>sample.id</code></dt><dd><p>a factor with 8 levels</p>
</dd>
<dt><code>chromStart</code></dt><dd><p>integer vector</p>
</dd>
<dt><code>chromEnd</code></dt><dd><p>integer vector</p>
</dd>
<dt><code>count</code></dt><dd><p>integer vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>http://cbio.ensmp.fr/~thocking/chip-seq-chunk-db/
data set H3K36me3_TDH_other
chunk 3.
</p>

<hr>
<h2 id='H3K4me3.TDH.immune.chunk12.cluster4'>
Histone ChIP-seq data, 26 samples, chr1 subset
</h2><span id='topic+H3K4me3.TDH.immune.chunk12.cluster4'></span>

<h3>Description</h3>

<p>26 samples, each with the same overlapping peak(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("H3K4me3.TDH.immune.chunk12.cluster4")</code></pre>


<h3>Format</h3>

<p>A data frame.
</p>


<h3>Source</h3>

<p>http://cbio.ensmp.fr/~thocking/chip-seq-chunk-db H3K4me3_TDH_immune data
set, chunk.id=12.
</p>

<hr>
<h2 id='PeakSegDP'>PeakSegDP</h2><span id='topic+PeakSegDP'></span>

<h3>Description</h3>

<p>Compute the PeakSeg model on a data.frame of <code>compressed</code> sequence
reads.</p>


<h3>Usage</h3>

<pre><code class='language-R'>PeakSegDP(compressed, 
    maxPeaks)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PeakSegDP_+3A_compressed">compressed</code></td>
<td>
<p>data.frame with columns chromStart, chromEnd, count.</p>
</td></tr>
<tr><td><code id="PeakSegDP_+3A_maxpeaks">maxPeaks</code></td>
<td>
<p>maximum number of peaks to consider.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Toby Dylan Hocking, Guillem Rigaill</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(PeakSegDP)
data(chr11ChIPseq, envir=environment())
one &lt;- subset(chr11ChIPseq$coverage, sample.id=="McGill0002")[10000:12000,]
fit &lt;- PeakSegDP(one, 3L)

if(interactive() &amp;&amp; require(ggplot2)){

  ggplot()+
    geom_step(aes(chromStart/1e3, count), data=one)+
    geom_segment(aes(chromStart/1e3, mean,
                     xend=chromEnd/1e3, yend=mean),
                 data=fit$segments, color="green")+
    geom_segment(aes(chromStart/1e3, 0,
                     xend=chromEnd/1e3, yend=0),
                 data=subset(fit$segments, status=="peak"),
                 size=3, color="deepskyblue")+
    theme_bw()+
    theme(panel.margin=grid::unit(0, "cm"))+
    facet_grid(peaks ~ ., scales="free", labeller=function(df){
      s &lt;- ifelse(df$peaks==1, "", "s")
      df$peaks &lt;- paste0(df$peaks, " peak", s)
      df
    })

}

</code></pre>

<hr>
<h2 id='phi.list'>phi list</h2><span id='topic+phi.list'></span>

<h3>Description</h3>

<p>List of functions, each a phi loss.</p>


<h3>Usage</h3>

<pre><code class='language-R'>"phi.list"</code></pre>

<hr>
<h2 id='PoissonLoss'>PoissonLoss</h2><span id='topic+PoissonLoss'></span>

<h3>Description</h3>

<p>Compute the weighted Poisson loss function, which is <code>seg.mean</code> -
<code>count</code> * log(<code>seg.mean</code>). The edge case is when the mean is zero, in
which case the probability mass function takes a value of 1 when
the data is 0 (and 0 otherwise). Thus the log-likelihood of a
maximum likelihood segment with mean zero must be zero.</p>


<h3>Usage</h3>

<pre><code class='language-R'>PoissonLoss(count, seg.mean, 
    weight = 1)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PoissonLoss_+3A_count">count</code></td>
<td>
<p>count </p>
</td></tr>
<tr><td><code id="PoissonLoss_+3A_seg.mean">seg.mean</code></td>
<td>
<p>seg.mean </p>
</td></tr>
<tr><td><code id="PoissonLoss_+3A_weight">weight</code></td>
<td>
<p>weight </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Toby Dylan Hocking, Guillem Rigaill</p>


<h3>Examples</h3>

<pre><code class='language-R'>PoissonLoss(1, 1)
PoissonLoss(0, 0)
PoissonLoss(1, 0)
PoissonLoss(0, 1)
</code></pre>

<hr>
<h2 id='regression.funs'>regression funs</h2><span id='topic+regression.funs'></span>

<h3>Description</h3>

<p>List of regression functions: features, limits -&gt; list.</p>


<h3>Usage</h3>

<pre><code class='language-R'>"regression.funs"</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
