<!DOCTYPE html><html><head><title>Help for package PeakError</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PeakError}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#checkChrom'><p>checkChrom</p></a></li>
<li><a href='#checkPositions'><p>checkPositions</p></a></li>
<li><a href='#PeakError'><p>PeakError</p></a></li>
<li><a href='#PeakErrorChrom'><p>PeakErrorChrom</p></a></li>
<li><a href='#Peaks'><p>Peaks</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Maintainer:</td>
<td>Toby Dylan Hocking &lt;toby.hocking@r-project.org&gt;</td>
</tr>
<tr>
<td>Author:</td>
<td>Toby Dylan Hocking</td>
</tr>
<tr>
<td>Version:</td>
<td>2023.9.4</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Title:</td>
<td>Compute the Label Error of Peak Calls</td>
</tr>
<tr>
<td>Description:</td>
<td>Chromatin immunoprecipitation DNA sequencing results in genomic
    tracks that show enriched regions or peaks where proteins are bound.
    This package implements fast C code that computes the true and false
    positives with respect to a database of annotated region labels.</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, ggplot2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-06 04:49:01 UTC; tdhock</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-06 05:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='checkChrom'>checkChrom</h2><span id='topic+checkChrom'></span>

<h3>Description</h3>

<p>Check for a valid data.frame with chrom names.</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkChrom(df)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkChrom_+3A_df">df</code></td>
<td>
<p>df </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Toby Dylan Hocking</p>

<hr>
<h2 id='checkPositions'>checkPositions</h2><span id='topic+checkPositions'></span>

<h3>Description</h3>

<p>Check for a valid data.frame with chromStart, chromEnd.</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkPositions(df)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkPositions_+3A_df">df</code></td>
<td>
<p>df </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Toby Dylan Hocking</p>

<hr>
<h2 id='PeakError'>PeakError</h2><span id='topic+PeakError'></span>

<h3>Description</h3>

<p>Compute true and false positive peak calls, with respect to a
database of annotated <code>regions</code>.</p>


<h3>Usage</h3>

<pre><code class='language-R'>PeakError(peaks, regions)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PeakError_+3A_peaks">peaks</code></td>
<td>
<p>data.frame with columns chrom, chromStart, chromEnd. NOTE:
chromStart should be 0-based and chromEnd should be
1-based. EXAMPLE: the first 100 base of of a chromosome are
chromStart=0, chromEnd=100. The second 100 bases are
chromStart=100, chromEnd=200.</p>
</td></tr>
<tr><td><code id="PeakError_+3A_regions">regions</code></td>
<td>
<p>data.frame with columns chrom, chromStart, chromEnd, annotation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame for each region with additional counts of true
positives (tp, possible.tp), false positives (fp, possible.fp,
fp.status), and false negatives (fn, fn.status).</p>


<h3>Author(s)</h3>

<p>Toby Dylan Hocking</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- seq(5, 85, by=5)
peaks &lt;- rbind(
  Peaks("chr2", x, x+3),
  Peaks("chr3", c(25, 38, 57), c(33, 54, 75)),
  Peaks("chr4", c(5, 32, 38, 65), c(15, 35, 55, 85)),
  Peaks("chr5", c(12, 26, 56, 75), c(16, 54, 59, 85)))
regions.list &lt;- list()
for(chr in 1:5){
  regions.list[[chr]] &lt;- data.frame(
    chrom=paste0("chr", chr),
    chromStart=c(10, 30, 50, 70),
    chromEnd=c(20, 40, 60, 80),
    annotation=c("noPeaks", "peakStart", "peakEnd", "peaks"))
}
regions &lt;- do.call(rbind, regions.list)
err &lt;- PeakError(peaks, regions)
ann.colors &lt;- c(
  noPeaks="#f6f4bf",
  peakStart="#ffafaf",
  peakEnd="#ff4c4c",
  peaks="#a445ee")
if(require(ggplot2)){
  ggplot()+
    geom_rect(aes(
      xmin=chromStart+1/2, xmax=chromEnd+1/2,
      ymin=-1, ymax=1,
      fill=annotation,
      linetype=fn.status,
      size=fp.status),
      data=err, color="black")+
    scale_y_continuous("", breaks=NULL)+
    scale_linetype_manual(
      values=c("false negative"="dotted", correct="solid"))+
    scale_size_manual(
      values=c("false positive"=3, correct=1))+
    scale_fill_manual(
      values=ann.colors,
      breaks=names(ann.colors))+
    facet_grid(chrom ~ .)+
    theme_bw()+
    guides(
      fill=guide_legend(order=1),
      linetype=guide_legend(order=2, override.aes=list(fill="white")),
      size=guide_legend(order=3, override.aes=list(fill="white")))+
    theme(panel.margin=grid::unit(0, "cm"))+
    geom_segment(aes(
      chromStart+1/2, 1/2, xend=chromEnd+1/2, yend=1/2),
      data=peaks, color="deepskyblue", size=2)+
    scale_x_continuous(
      "position on chromosome",
      breaks=seq(10, 90, by=10))+
    geom_text(aes(
      base, -1/2, label="N"),
      data.frame(base=10:90),
      color="deepskyblue")
}
</code></pre>

<hr>
<h2 id='PeakErrorChrom'>PeakErrorChrom</h2><span id='topic+PeakErrorChrom'></span>

<h3>Description</h3>

<p>Compute the <code><a href="#topic+PeakError">PeakError</a></code> assuming that <code>peaks</code> and <code>regions</code> are on the
same chrom.</p>


<h3>Usage</h3>

<pre><code class='language-R'>PeakErrorChrom(peaks, 
    regions)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PeakErrorChrom_+3A_peaks">peaks</code></td>
<td>
<p>data.frame with columns chromStart, chromEnd. NOTE: chromStart
should be 0-based and chromEnd should be 1-based. EXAMPLE: the
first 100 base of of a chromosome are chromStart=0,
chromEnd=100. The second 100 bases are chromStart=100,
chromEnd=200.</p>
</td></tr>
<tr><td><code id="PeakErrorChrom_+3A_regions">regions</code></td>
<td>
<p>data.frame with columns chromStart, chromEnd.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with 1 row for each region and error columns.</p>


<h3>Author(s)</h3>

<p>Toby Dylan Hocking</p>

<hr>
<h2 id='Peaks'>Peaks</h2><span id='topic+Peaks'></span>

<h3>Description</h3>

<p>Make a data.frame that represents a list of peaks.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Peaks(chrom = factor(), 
    base.before = integer(), 
    last.base = integer())</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Peaks_+3A_chrom">chrom</code></td>
<td>
<p>character or factor with <code>chrom</code> name for example &quot;chr22&quot;</p>
</td></tr>
<tr><td><code id="Peaks_+3A_base.before">base.before</code></td>
<td>
<p>integer, base before peak.</p>
</td></tr>
<tr><td><code id="Peaks_+3A_last.base">last.base</code></td>
<td>
<p>integer, last base of peak.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame with columns <code>chrom</code>, chromStart, chromEnd.</p>


<h3>Author(s)</h3>

<p>Toby Dylan Hocking</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
