<!DOCTYPE html><html><head><title>Help for package pedbuildr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pedbuildr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#buildPeds'><p>Build a list of pedigrees</p></a></li>
<li><a href='#pedbuildr-package'><p>pedbuildr: Pedigree Reconstruction</p></a></li>
<li><a href='#reconstruct'><p>Pedigree reconstruction</p></a></li>
<li><a href='#trioData'><p>Reconstruction example with three individuals</p></a></li>
<li><a href='#Tutankhamun'><p>Pedigree of Tutankhamun</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Pedigree Reconstruction</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Reconstruct pedigrees from genotype data, by optimising the
    likelihood over all possible pedigrees subject to given restrictions.
    Tailor-made plots facilitate evaluation of the output. This package is
    part of the 'pedsuite' ecosystem for pedigree analysis. In
    particular, it imports 'pedprobr' for calculating pedigree likelihoods
    and 'forrel' for estimating pairwise relatedness.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/magnusdv/pedbuildr">https://github.com/magnusdv/pedbuildr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/magnusdv/pedbuildr/issues">https://github.com/magnusdv/pedbuildr/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>pedtools (&ge; 2.2.0), R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>forrel (&ge; 1.5.0), glue, pedmut, pedprobr, ribd</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-22 05:23:15 UTC; magnusdv</td>
</tr>
<tr>
<td>Author:</td>
<td>Magnus Dehli Vigeland
    <a href="https://orcid.org/0000-0002-9134-4962"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Magnus Dehli Vigeland &lt;m.d.vigeland@medisin.uio.no&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-22 08:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='buildPeds'>Build a list of pedigrees</h2><span id='topic+buildPeds'></span>

<h3>Description</h3>

<p>Build all pedigrees between a set of individuals, subject to given
restrictions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildPeds(
  labs,
  sex = 1,
  extra = "parents",
  age = NULL,
  knownPO = NULL,
  knownSub = NULL,
  allKnown = FALSE,
  notPO = NULL,
  noChildren = NULL,
  connected = TRUE,
  maxInbreeding = 1/16,
  linearInb = FALSE,
  sexSymmetry = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="buildPeds_+3A_labs">labs</code></td>
<td>
<p>A character vector of ID labels.</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_sex">sex</code></td>
<td>
<p>A vector of the same length as <code>labs</code>, with entries 1 (male) or 2
(female).</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_extra">extra</code></td>
<td>
<p>Either the word &quot;parents&quot; (default), or a non-negative integer.
See Details.</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_age">age</code></td>
<td>
<p>A numeric or character vector. If numeric, and <code>age[i] &lt; age[j]</code>,
then individual <code>i</code> will not be an ancestor of individual <code>j</code>. The numbers
themselves are irrelevant, only the partial ordering. (No inference is made
about individuals of equal age.) Alternatively, for finer control, <code>age</code>
may be a character vector of inequalities, e.g., <code>age = c("1&gt;2", "1&gt;3")</code>.</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_knownpo">knownPO</code></td>
<td>
<p>A list of vectors of length 2, containing the ID labels of
pairs known to be parent-offspring. By default, both directions are
considered; use <code>age</code> to force a specific direction.</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_knownsub">knownSub</code></td>
<td>
<p>A <code>ped</code> object involving a subset of the <code>labs</code> individuals.</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_allknown">allKnown</code></td>
<td>
<p>A logical. If TRUE, no other pairs than <code>knownPO</code> will be
assigned as parent-offspring. If FALSE (default), all pairs except those in
<code>notPO</code> are treated as potential parent-offspring.</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_notpo">notPO</code></td>
<td>
<p>A list of vectors of length 2, containing the ID labels of pairs
known <em>not</em> to be parent-offspring.</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_nochildren">noChildren</code></td>
<td>
<p>A vector of ID labels, indicating individuals without
children of their own.</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_connected">connected</code></td>
<td>
<p>A logical. If TRUE (default), only connected pedigrees are
returned.</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_maxinbreeding">maxInbreeding</code></td>
<td>
<p>A single numeric indicating the highest permitted
inbreeding coefficient. Default: 1/16 (as with first-cousin parents.)</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_linearinb">linearInb</code></td>
<td>
<p>A parameter controlling the maximum separation of linearly
related spouses. Either TRUE (allow all linear inbreeding), FALSE (disallow
all) or a non-negative integer. For example, <code>linearInb = 1</code> allows
parent/child mating, but not grandparent/grandchild or more distant linear
relatives. Default: FALSE.</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_sexsymmetry">sexSymmetry</code></td>
<td>
<p>A logical. If TRUE (default), pedigrees which are equal
except for the gender distribution of the <em>added</em> parents, are regarded as
equivalent, and only one of each equivalence class is returned. Example:
paternal vs. maternal half sibs.</p>
</td></tr>
<tr><td><code id="buildPeds_+3A_verbose">verbose</code></td>
<td>
<p>A logical.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The parameter <code>extra</code> controls which of two algorithms are used to create the
pedigree list.
</p>
<p>If <code>extra</code> is a nonnegative integer, it determines the number of extra
individuals allowed in the iterative pedigree construction. These extras
start off with undetermined sex, meaning that both males and females are
used. It should be noted that the final pedigrees may contain additional
extras, since missing parents are added at the end.
</p>
<p>If <code>extra</code> is the word &quot;parents&quot;, the algorithm is not iterative. It first
generates all directed acyclic graphs between the original individuals. Then
their parents are added and merged in all possible ways. This option has the
advantage of not requiring an explicit/ad hoc number of &quot;extras&quot;, but works
best in smaller cases.
</p>


<h3>Value</h3>

<p>A list of (possibly disconnected) pedigrees.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Two individuals + 1 extra
plist = buildPeds(1:2, extra = 1, age = "1&gt;2")
plot(plist)

# Allow disconnected
plist2 = buildPeds(1:2, extra = 1, age = "1&gt;2", connected = FALSE)
plot(plist2, frames = TRUE)

# Note that full sibs require 2 extras
plist3 = buildPeds(1:2, extra = 2, age = "1&gt;2")
plot(plist3)

# With 2 extras, allowing any inbreeding
plist4 = buildPeds(1:2, extra = 2, age = "1&gt;2", maxInbreeding = 1)
plot(plist4)

# Full sibs are also included when `extra = "parents"`
plist5 = buildPeds(1:2, extra = "parents", age = "1&gt;2")
plot(plist5)


</code></pre>

<hr>
<h2 id='pedbuildr-package'>pedbuildr: Pedigree Reconstruction</h2><span id='topic+pedbuildr'></span><span id='topic+pedbuildr-package'></span>

<h3>Description</h3>

<p>Reconstruct pedigrees from genotype data, by optimising the likelihood over all possible pedigrees subject to given restrictions. Tailor-made plots facilitate evaluation of the output. This package is part of the 'pedsuite' ecosystem for pedigree analysis. In particular, it imports 'pedprobr' for calculating pedigree likelihoods and 'forrel' for estimating pairwise relatedness.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Magnus Dehli Vigeland <a href="mailto:m.d.vigeland@medisin.uio.no">m.d.vigeland@medisin.uio.no</a> (<a href="https://orcid.org/0000-0002-9134-4962">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/magnusdv/pedbuildr">https://github.com/magnusdv/pedbuildr</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/magnusdv/pedbuildr/issues">https://github.com/magnusdv/pedbuildr/issues</a>
</p>
</li></ul>


<hr>
<h2 id='reconstruct'>Pedigree reconstruction</h2><span id='topic+reconstruct'></span>

<h3>Description</h3>

<p>Reconstructs the most likely pedigree from genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reconstruct(
  x,
  ids,
  extra = "parents",
  alleleMatrix = NULL,
  loci = NULL,
  pedlist = NULL,
  inferPO = FALSE,
  sex = NULL,
  age = NULL,
  knownPO = NULL,
  knownSub = NULL,
  allKnown = FALSE,
  notPO = NULL,
  noChildren = NULL,
  connected = TRUE,
  maxInbreeding = 1/16,
  linearInb = FALSE,
  sexSymmetry = TRUE,
  sortResults = TRUE,
  founderInb = 0,
  numCores = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reconstruct_+3A_x">x</code></td>
<td>
<p>A <code>pedtools::ped</code> object or a list of such.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_ids">ids</code></td>
<td>
<p>A vector of ID labels from <code>x</code>. By default, the genotyped members
of <code>x</code> are used.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_extra">extra</code></td>
<td>
<p>Either the word &quot;parents&quot; (default), or a non-negative integer.
See Details.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_allelematrix">alleleMatrix</code></td>
<td>
<p>A matrix with two columns for each marker. By default
extracted from <code>x</code></p>
</td></tr>
<tr><td><code id="reconstruct_+3A_loci">loci</code></td>
<td>
<p>A list of marker attributes. By default extracted from <code>x</code>.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_pedlist">pedlist</code></td>
<td>
<p>A list of pedigrees to optimise over. If NULL, <code><a href="#topic+buildPeds">buildPeds()</a></code>
is used to generate a list.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_inferpo">inferPO</code></td>
<td>
<p>A logical. If TRUE, an initial stage of pairwise IBD
estimation is done to infer high-confidence parent/child pairs, and also
<em>non</em>-parent/child pairs. When this option is used, arguments to <code>knownPO</code>
and <code>notPO</code> are ignored.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_sex">sex</code></td>
<td>
<p>A vector of the same length as <code>labs</code>, with entries 1 (male) or 2
(female).</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_age">age</code></td>
<td>
<p>A numeric or character vector. If numeric, and <code>age[i] &lt; age[j]</code>,
then individual <code>i</code> will not be an ancestor of individual <code>j</code>. The numbers
themselves are irrelevant, only the partial ordering. (No inference is made
about individuals of equal age.) Alternatively, for finer control, <code>age</code>
may be a character vector of inequalities, e.g., <code>age = c("1&gt;2", "1&gt;3")</code>.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_knownpo">knownPO</code></td>
<td>
<p>A list of vectors of length 2, containing the ID labels of
pairs known to be parent-offspring. By default, both directions are
considered; use <code>age</code> to force a specific direction.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_knownsub">knownSub</code></td>
<td>
<p>A <code>ped</code> object involving a subset of the <code>labs</code> individuals.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_allknown">allKnown</code></td>
<td>
<p>A logical. If TRUE, no other pairs than <code>knownPO</code> will be
assigned as parent-offspring. If FALSE (default), all pairs except those in
<code>notPO</code> are treated as potential parent-offspring.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_notpo">notPO</code></td>
<td>
<p>A list of vectors of length 2, containing the ID labels of pairs
known <em>not</em> to be parent-offspring.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_nochildren">noChildren</code></td>
<td>
<p>A vector of ID labels, indicating individuals without
children of their own.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_connected">connected</code></td>
<td>
<p>A logical. If TRUE (default), only connected pedigrees are
returned.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_maxinbreeding">maxInbreeding</code></td>
<td>
<p>A single numeric indicating the highest permitted
inbreeding coefficient. Default: 1/16 (as with first-cousin parents.)</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_linearinb">linearInb</code></td>
<td>
<p>A parameter controlling the maximum separation of linearly
related spouses. Either TRUE (allow all linear inbreeding), FALSE (disallow
all) or a non-negative integer. For example, <code>linearInb = 1</code> allows
parent/child mating, but not grandparent/grandchild or more distant linear
relatives. Default: FALSE.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_sexsymmetry">sexSymmetry</code></td>
<td>
<p>A logical. If TRUE (default), pedigrees which are equal
except for the gender distribution of the <em>added</em> parents, are regarded as
equivalent, and only one of each equivalence class is returned. Example:
paternal vs. maternal half sibs.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_sortresults">sortResults</code></td>
<td>
<p>A logical. If TRUE (default), the output is sorted so that
the most likely pedigree comes first.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_founderinb">founderInb</code></td>
<td>
<p>A number in the interval <code style="white-space: pre;">&#8288;[0,1]&#8288;</code>, used as background
inbreeding level in all founders. Default: 0.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_numcores">numCores</code></td>
<td>
<p>A positive integer. The number of cores used in
parallelisation. Default: 1.</p>
</td></tr>
<tr><td><code id="reconstruct_+3A_verbose">verbose</code></td>
<td>
<p>A logical.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The parameter <code>extra</code> controls which of two algorithms are used to create the
pedigree list.
</p>
<p>If <code>extra</code> is a nonnegative integer, it determines the number of extra
individuals allowed in the iterative pedigree construction. These extras
start off with undetermined sex, meaning that both males and females are
used. It should be noted that the final pedigrees may contain additional
extras, since missing parents are added at the end.
</p>
<p>If <code>extra</code> is the word &quot;parents&quot;, the algorithm is not iterative. It first
generates all directed acyclic graphs between the original individuals. Then
their parents are added and merged in all possible ways. This option has the
advantage of not requiring an explicit/ad hoc number of &quot;extras&quot;, but works
best in smaller cases.
</p>


<h3>Value</h3>

<p>An object of class <code>pedrec</code>, which is essentially list with the
following entries:
</p>

<ul>
<li> <p><code>labs</code>: The individual labels as given in <code>ids</code>.
</p>
</li>
<li> <p><code>pedlist</code>: A list of pedigrees, either built by <code><a href="#topic+buildPeds">buildPeds()</a></code> or as
supplied in the input argument <code>pedlist</code>. If <code>sortResults = TRUE</code>, the list
is sorted so that the most likely pedigrees come first
</p>
</li>
<li> <p><code>logliks</code>: A numerical vector of pedigree log-likelihoods
</p>
</li>
<li> <p><code>kappa</code>: A data frame with pairwise estimates (if <code>inferPO = TRUE</code>)
</p>
</li>
<li> <p><code>alleleMatrix</code>: A matrix of marker alleles
</p>
</li>
<li> <p><code>loci</code>: A list of marker locus attributes
</p>
</li>
<li> <p><code>errPeds</code>: A list of pedigrees for which the likelihood calculation
failed
</p>
</li>
<li> <p><code>errIdx</code>: The indices of pedigrees in <code>errPeds</code> as elements of <code>pedlist</code>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>#-----------------
# Example 1: Trio
#-----------------

# Built-in dataset `trioData`
x = list(singleton(1), singleton(2), singleton(3)) |&gt;
  setMarkers(alleleMatrix = trioData, locusAttributes ="snp12")

res = reconstruct(x, inferPO = TRUE, age = "1 &gt; 2")

# Plot most likely pedigrees
plot(res, top = 6)


#--------------------
# Example 2: Siblings
#--------------------
library(forrel)

ids = c("s1", "s2")

# Create pedigree and simulate profiles with 20 STR markers
y = nuclearPed(children = ids) |&gt;
  profileSim(markers = NorwegianFrequencies[1:20], ids = ids, seed = 123)

# Reconstruct allowing 2 extra individuals and any inbreeding
res2 = reconstruct(y, extra = 2, maxInb = 1)
plot(res2)

# With mutation modelling
y = setMutmod(y, model = "equal", rate = 0.01)
res3 = reconstruct(y, extra = 2, maxInb = 1)
plot(res3)

</code></pre>

<hr>
<h2 id='trioData'>Reconstruction example with three individuals</h2><span id='topic+trioData'></span>

<h3>Description</h3>

<p>This dataset contains simulated genotypes for 3 individuals at 100 SNP
markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trioData
</code></pre>


<h3>Format</h3>

<p>A matrix with 3 rows and 100 columns. Each entry contains a genotype
in the form a/b.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>trioData[, 1:10]

x = list(singleton(1), singleton(2), singleton(3)) |&gt;
  setMarkers(alleleMatrix = trioData, locusAttributes ="snp12")

x

</code></pre>

<hr>
<h2 id='Tutankhamun'>Pedigree of Tutankhamun</h2><span id='topic+Tutankhamun'></span>

<h3>Description</h3>

<p>A reconstructed pedigree of the Egyptian Pharaoh Tutankhamun, with genotypes
for 8 STR markers, as published by Hawass et al.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Tutankhamun
</code></pre>


<h3>Format</h3>

<p>A data frame with 7 rows and 12 columns:
</p>

<ul>
<li> <p><code>id</code>,<code>fid</code>,<code>mid</code>,<code>sex</code>: Pedigree columns in standard format
</p>
</li>
<li> <p><code>D13S317</code>, ...: Genotype columns for 8 markers
</p>
</li></ul>



<h3>Source</h3>

<p>Hawass et al. <em>Ancestry and pathology in King Tutankhamun's family</em>.
Jama (2010).
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Pedigree as published
plot(Tutankhamun)

# Simple reconstruction, assuming all directly related
res = reconstruct(Tutankhamun, extra = 0, inferPO = TRUE, maxInbreeding = 1)
plot(res, top = 4)

# Published ped is most likely (with these assumptions)
identical(res[[1]], Tutankhamun)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
