<!DOCTYPE html><html><head><title>Help for package pedgene</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pedgene}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#example.geno'><p>Example datasets for pedgene</p></a></li>
<li><a href='#format_relation'>
<p>Internal utility to format the matrix used to specify special</p>
relationships between pairs of individuals</a></li>
<li><a href='#pedgene'>
<p>Compute Kernel and Burden Statistics for Pedigree Data (possibly</p>
with unrelated subjects)</a></li>
<li><a href='#pedigreeChecks'>
<p>Validity checks on pedigree data</p></a></li>
<li><a href='#quadfactor'>
<p>Internal utility to calculate a constant quadratic factor for</p>
gene-level statistics variances, over all pedigrees</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Gene-Level Variant Association Tests for Pedigree Data</td>
</tr>
<tr>
<td>Version:</td>
<td>3.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-04</td>
</tr>
<tr>
<td>Description:</td>
<td>Gene-level variant association tests with disease status for pedigree data: kernel and burden association statistics.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0), stats, Matrix, CompQuadForm, survey, kinship2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://analytictools.mayo.edu/research/pedgene/">https://analytictools.mayo.edu/research/pedgene/</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-04 18:06:39 UTC; sinnwell</td>
</tr>
<tr>
<td>Author:</td>
<td>Schaid Daniel [aut],
  Visconti Alessia [ctb],
  Jason Sinnwell [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jason Sinnwell &lt;sinnwell.jason@mayo.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-05 05:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='example.geno'>Example datasets for pedgene </h2><span id='topic+example.geno'></span><span id='topic+example.ped'></span><span id='topic+example.map'></span><span id='topic+example.relation'></span>

<h3>Description</h3>

<p>example.geno: a data frame with minor allele count for subjects (rows)
at variant positions (columns); example.ped: pedigree and trait data for subjects in
example.geno; example.map: gene and chromosome for variant positions in example.geno;  
example.relation: special (twin) relationships for individuals in example.ped
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(example.geno)
data(example.ped)
data(example.map)
data(example.relation)
</code></pre>


<h3>Format</h3>

<p><code>example.geno</code> -data frame with minor allele count for 20 variant
positions:
</p>

<dl>
<dt><code>ped </code></dt><dd><p>pedigree ID, character or numeric</p>
</dd>
<dt><code>person </code></dt><dd><p>person ID, used with ped to match subjects to
their row in example.ped</p>
</dd>
<dt><code>AA.1-AA.10,AX.1-AX.10 </code></dt><dd><p>genotype columns at 10 positions for
each of 2 simulated genes</p>
</dd>
</dl>

<p><code>example.ped</code> -data frame with pedigree structure and trait
values in the following columns:
</p>

<dl>
<dt><code>ped </code></dt><dd><p> pedigree ID, character or numeric</p>
</dd>
<dt><code>person </code></dt><dd><p> person ID, a unique ID within each pedigree</p>
</dd>
<dt><code>father </code></dt><dd><p>father ID, 0 if no father</p>
</dd>
<dt><code>mother </code></dt><dd><p>mother ID, 0 if no mother</p>
</dd>
<dt><code>sex </code></dt><dd><p>coded as 1 for male, 2 for female</p>
</dd>
<dt><code>trait </code></dt><dd><p>phenotype, either case-control status coded as 1
for affected and 0 for unaffected, or a continuous
value. Subjects with missing (NA) will be removed from the analysis</p>
</dd>
<dt><code>trait.adjusted </code></dt><dd><p>an optional variable for covariate-adjusted
trait. If trait.adjusted is present in the data.frame, then
gene-level tests are adjusted for covariates using
residuals = (trait - trait.adjusted). Otherwise, gene-level tests
are not adjusted for covariates, in which case residuals = trait - mean(trait)</p>
</dd>
</dl>

<p><code>example.map </code> - data frame with columns for gene name and chromosome:
</p>

<dl>
<dt><code>chrom </code></dt><dd><p>chromosome code (1-23,X allowed) where the gene is located</p>
</dd>
<dt><code>gene </code></dt><dd><p>gene identifier</p>
</dd>
</dl>

<p><code>example.relation </code> - matrix specifying special relationships between pairs 
of individuals in the following columns:
</p>

<dl>
<dt><code>ped </code></dt><dd><p>pedigree ID, character or numeric</p>
</dd>
<dt><code>id1 </code></dt><dd><p>person1 ID, used with ped to match subjects to
their row in example.ped</p>
</dd>
<dt><code>id2 </code></dt><dd><p>person2 ID, used with ped to match subjects to
their row in example.ped</p>
</dd>
<dt><code>code </code></dt><dd><p>any of the following: 1=Monozygotic twin, 2=Dizygotic twin, 
3=Twin of unknown zygosity, and 4=Spouse</p>
</dd>
</dl>



<h3>Source</h3>

<p>Simulated data for testing purposes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example.geno)
data(example.ped)
data(example.map)
data(example.relation)
</code></pre>

<hr>
<h2 id='format_relation'>
Internal utility to format the matrix used to specify special 
relationships between pairs of individuals
</h2><span id='topic+format_relation'></span>

<h3>Description</h3>

<p>Internal utility to format the matrix used to specify special 
relationships between pairs of individuals. 
See kinship2::pedigree for details. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>format_relation(relation)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="format_relation_+3A_relation">relation</code></td>
<td>
<p>A matrix for one or more special relationships</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix for one or more special relationships as required by pedgene  
</p>


<h3>Author(s)</h3>

<p>Alessia Visconti, King's College London (alessia.visconti@kcl.ac.uk).
</p>


<h3>See Also</h3>

<p>pedgene, kinship
</p>

<hr>
<h2 id='pedgene'>
Compute Kernel and Burden Statistics for Pedigree Data (possibly
with unrelated subjects) 
</h2><span id='topic+pedgene'></span><span id='topic+pedgene.stats'></span><span id='topic+print.pedgene'></span><span id='topic+summary.pedgene'></span>

<h3>Description</h3>

<p>Compute linear kernel and burden statistics for gene-level analysis of
data that includes pedigree-related subjects, and possibly unrelated
subjects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedgene(ped, geno, map=NULL, male.dose=2,
                    checkpeds=TRUE, verbose.return=FALSE,
                    weights=NULL, weights.beta=c(1,25),
                    weights.mb=FALSE, relation=NULL,
                    method="kounen", acc.davies=1e-5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedgene_+3A_ped">ped</code></td>
<td>
<p>A data.frame with variables that define the pedigree structure
(typical format used by LINKAGE and PLINK), trait (phenotype), and
optionally a covariate-adjusted trait (for covariate-adjusted gene
level statistics). The columns in the data.frame must be named as
follows:
</p>

<ul>
<li><p>famid: pedigree ID, character or numeric allowed
</p>
</li>
<li><p>person: person ID, a unique ID within each pedigree, numeric
or character allowed
</p>
</li>
<li><p>father: father ID, 0 if no father
</p>
</li>
<li><p>mother: mother ID, 0 if no mother
</p>
</li>
<li><p>sex: coded as 1 for male, 2 for female
</p>
</li>
<li><p>trait: phenotype, either case-control status coded as 1
for affected and 0 for unaffected, or a continuous
value. Subjects with missing (NA) will be removed from the analysis
</p>
</li>
<li><p>trait.adjusted: an optional variable for covariate-adjusted
trait. If trait.adjusted is present in the data.frame, then
gene-level tests are adjusted for covariates using
residuals = (trait - trait.adjusted). Otherwise, gene-level tests
are not adjusted for covariates, in which case
residuals = trait - mean(trait) where the mean is taken on all
subjects passed into pedgene before removing subjects who do not
have genotype data.
</p>
</li></ul>

</td></tr>
<tr><td><code id="pedgene_+3A_geno">geno</code></td>
<td>

<p>Data.frame or matrix with genotypes for subjects (rows) at each variant position
(columns). The first two columns are required to be named <code>famid</code> and
<code>person</code>, which are used to match subjects to their data in the
<code>ped</code> data.frame. The genotypes are coded as 0, 1, 2 for autosomal
markers (typically a count of the number of the less-frequent
allele). For X-chromosome markers, females are coded 0, 1, 2, and
males coded 0, 1. Missing genotypes (NA) are allowed.
</p>
</td></tr>
<tr><td><code id="pedgene_+3A_map">map</code></td>
<td>

<p>Optional data.frame with columns &quot;chrom&quot; and  &quot;gene&quot;, one row per
variant column in geno. The gene name can be any identifier for the
gene. The chromosome can be either numeric or character, where the
calculations will differ between autosomes vs X chromosome (allow
&quot;X&quot;/&quot;x&quot;/23, converted to &quot;X&quot; in results)
</p>
</td></tr>
<tr><td><code id="pedgene_+3A_male.dose">male.dose</code></td>
<td>
<p>When analyzing the X-chromosome, male.dose defines
how male genotypes should be analyzed. male.dose can be between 0
and 2, but is typically either 1 or 2. Ozbek and Clayton show that
male.dose = 2 is powerful in the presence of X-chromosome dosage
compensation in females.
</p>
</td></tr>
<tr><td><code id="pedgene_+3A_checkpeds">checkpeds</code></td>
<td>
<p>logical, if FALSE, the method will skip the pedigree
checking step, which can be intensive for large studies</p>
</td></tr>
<tr><td><code id="pedgene_+3A_verbose.return">verbose.return</code></td>
<td>
<p>logical, if TRUE, return the pedigree, geno,
and map objects used in the tests after initial cleanup, e.g., the
removal of monomorphic variants. They are returned in the pedgene
object in a list called &quot;save&quot;</p>
</td></tr>
<tr><td><code id="pedgene_+3A_weights.beta">weights.beta</code></td>
<td>
<p> Weights based on a function of the minor allele
frequency (maf) and the Beta distribution</p>
</td></tr> 
<tr><td><code id="pedgene_+3A_weights">weights</code></td>
<td>
<p>optional user-specified weights, a vector of weights
for each variant column of geno. If none given, the Beta weights
are applied</p>
</td></tr>
<tr><td><code id="pedgene_+3A_weights.mb">weights.mb</code></td>
<td>
<p>logical, if TRUE and no user-given weights, apply the
Madsen-Browning weights per variant: 1/sqrt(maf*(1-maf)). The
hierarchy of weights used is 1) user-specified weights, 2)
Madsen-Browning if weights.mb=TRUE and weights=NULL, 3) Beta density
weights (default if no other weights are set)</p>
</td></tr>
<tr><td><code id="pedgene_+3A_relation">relation</code></td>
<td>

<p>Optional data.frame/matrix with 4 columns (id, person1, person2, code) specifying 
special relationships between pairs of individuals and used by the kinship 
function. Codes are : 
1=Monozygotic twin, 
2=Dizygotic twin, 
3=Twin of unknown zygosity, and
4=Spouse. 
The last is necessary in order to place a marriage with no children into the 
plot. See kinship2::pedigree for details.
</p>
</td></tr>
<tr><td><code id="pedgene_+3A_method">method</code></td>
<td>
<p>method for calculating the kernel test p-value. Kounen's
saddlepoint approximation (default) is based on the survey package,
and has been found to have less faults (e.g., returned missing
value) than  Davies' method (see Chen et al., 2012). The Davies method, 
which computes an exact p-value for a mixture of chi-square
distributions, is also provided. The accuracy of the Davies method
depends on the numerical accuracy parameter (acc.davies), which can
be difficult to specify ahead of time.</p>
</td></tr>
<tr><td><code id="pedgene_+3A_acc.davies">acc.davies</code></td>
<td>
<p>Numerical accuracy parameter used in the Davies'
method for calculating the kernel test p-value. In some instances, a
p-value from the kernel test is out of range, in which case the
p-value is set to 0 or 1, depending on which direction the p-value
was out of range.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The pedgene function is a wrapper function to call pedgene.stats on
one gene at a time. The pedgene.stats function calculates gene-level
tests for associations with a trait among subjects, accounting for relationships
among subjects based on known pedigree relationships (see Schaid et al). 
This is achieved by the kinship function in the kinship2 package. The kernel
association statistic uses a weighted linear kernel, with default
weights based on the beta distribution and the sample minor allele
frequency. The burden statistic is based on a weighted sum of variants. If
a gene only has one variant, the kernel test reduces to the burden statistic.
Variant positions that have zero variance are removed from the
analysis because they do not contribute information.
</p>
<p>Note that if ped contains extra people that are not necessary to
define relationships of people with genotype data, their trait value will
still be used in mean(trait) in calculating trait.adjusted if trait.adjusted is
not given as a column in ped. 
</p>


<h3>Value</h3>

<p>An object of the pedgene S3 class, with the following elements:
</p>
<table>
<tr><td><code>call:</code></td>
<td>
<p>function call</p>
</td></tr>
<tr><td><code>pgdf:</code></td>
<td>
<p>data.frame with gene name, chromosome, n-variants per
gene(after removing uncessary variants), n-variants removed per
gene, kernel and burden test statistics and p-values. Kernel
p-values are based on either Kuonen (1999) or Davies (1980) method.
The burden statistic has a standard normal distribution, so the sign
of the burden statistic gives information on the direction of
association (positive value implies large burden score are positively 
associated with larger trait values).  When a gene has only 1 variant, 
the kernel test reduces to the burden test. In this instance, the
kernel statistic (chi-square) is the square of the burden statistic
(standard normal), with both having the same p-value. When a gene
has no markers after removing zero-variance markers, the gene test
stastistics and p-values are all NA. 
</p>
</td></tr>
<tr><td><code>save:</code></td>
<td>
<p>If verbose.return was set to TRUE, a list containing the
cleaned form of the data that was input to pedgene and is used
in the tests: ped, geno, and map</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Daniel J. Schaid, Jason P. Sinnwell, Mayo Clinic (contact: schaid@mayo.edu).
Alessia Visconti, King's College London (alessia.visconti@kcl.ac.uk).
</p>


<h3>References</h3>

<p>Schaid DJ, McDonnell SK, Sinnwell JP, Thibodeau SN (2013)
Multiple Genetic Variant Association Testing by Collapsing and Kernel
Methods With Pedigree or Population Structured Data. Genetic
Epidemiology, 37(5):409-18.
</p>
<p>Ozbek U (2012) Statistics for X-chromosome association. 62nd Annual
Meeting of The American Society of Human Genetics; Program #22. San
Francisco, California.
</p>
<p>Clayton D (2008) Testing for association on the X chromosome.
Biostatistics 9:593-600
</p>
<p>Chen H, Meigs J, Dupuis J (2012) Sequence kernel association test for
quantitative traits in family samples. Genetic Epidem 37:196-204
</p>
<p>Kounen D (1999) Saddle point approximatinos for distributions of
quadratic forms in normal variables. Biometrika 86:929 -935
</p>
<p>Davies RB (1980) Algorithm AS 155: The Distribution of a Linear
Combination of chi-2 Random Variables, Journal of the Royal
Statistical Society. Series C (Applied Statistics), 29(3):323-33
</p>


<h3>See Also</h3>

<p>pedigreeChecks, example.ped
</p>


<h3>Examples</h3>

<pre><code class='language-R'># example data with the same 10 variants for an autosome and X chromosome
# pedigree data on 39 subjects including 3 families and unrelateds
data(example.ped)
data(example.geno)
data(example.map)

# gene tests (chroms 1 and X) with male.dose=2
pg.m2 &lt;- pedgene(example.ped, example.geno, example.map, male.dose=2)
# same genes, with male.dose=1
pg.m1 &lt;- pedgene(example.ped, example.geno, example.map, male.dose=1)

## print and summary methods
print(pg.m2, digits=3)
summary(pg.m1, digits=3)
</code></pre>

<hr>
<h2 id='pedigreeChecks'>
Validity checks on pedigree data
</h2><span id='topic+pedigreeChecks'></span>

<h3>Description</h3>

<p>Checks for valid IDs, sex codes for data in a single pedigree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigreeChecks(pedigree, male.code = 1, female.code = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigreeChecks_+3A_pedigree">pedigree</code></td>
<td>
<p>data frame with variables named person, father, mother, sex.</p>
</td></tr>
<tr><td><code id="pedigreeChecks_+3A_male.code">male.code</code></td>
<td>
<p>sex code for males</p>
</td></tr>
<tr><td><code id="pedigreeChecks_+3A_female.code">female.code</code></td>
<td>
<p>sex code for females</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A series of basic pedigree checks
</p>


<h3>Value</h3>

<p>valid = TRUE or FALSE for validity of pedigree data
</p>


<h3>Author(s)</h3>

<p>Daniel J. Schaid (schaid@mayo.edu).
</p>

<hr>
<h2 id='quadfactor'>
Internal utility to calculate a constant quadratic factor for
gene-level statistics variances, over all pedigrees
</h2><span id='topic+quadfactor'></span>

<h3>Description</h3>

<p>Internal utility to calculate a constant quadratic factor for
gene-level statistics
variances over all pedigrees, for either autosomes or X chromosome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quadfactor(kinmat, chrom, resid, sex, male.dose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="quadfactor_+3A_kinmat">kinmat</code></td>
<td>
<p>A kinship matrix for one or more pedigrees</p>
</td></tr>
<tr><td><code id="quadfactor_+3A_chrom">chrom</code></td>
<td>
<p>character string for chromosome number, if &quot;X&quot;, the
method accounts for sex code</p>
</td></tr>
<tr><td><code id="quadfactor_+3A_resid">resid</code></td>
<td>
<p>the residual based on the trait minus the group mean or
adjusted trait value for each subject</p>
</td></tr>
<tr><td><code id="quadfactor_+3A_sex">sex</code></td>
<td>
<p>See pedgene</p>
</td></tr>
<tr><td><code id="quadfactor_+3A_male.dose">male.dose</code></td>
<td>
<p>See pedgene</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Constant quadratic factor for gene-level statistics variances, for
autosomes and X chromosome 
</p>


<h3>Author(s)</h3>

<p>Daniel J. Schaid, Mayo Clinic (schaid@mayo.edu).
</p>


<h3>See Also</h3>

<p>pedgene, kinship
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
