<!DOCTYPE html><html lang="en-US"><head><title>Help for package pedtricks</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pedtricks}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#pedtricks-package'><p>pedtricks: Visualize, Summarize and Simulate Data from Pedigrees</p></a></li>
<li><a href='#convert_ped'><p>Converts a pedigree with individuals specified as factors to a numeric</p>
pedigree</a></li>
<li><a href='#draw_ped'><p>Produce a graphical representation of a pedigree</p></a></li>
<li><a href='#draw_pedA'><p>Produce a graphical representation of the relatedness matrix of a pedigree</p></a></li>
<li><a href='#extractA'><p>Allele Frequencies</p></a></li>
<li><a href='#fix_ped'><p>Manipulating pedigrees to prepare them for requirements of</p>
subsequent analyses
Prepares a pedigree to conform with requirements of many softwares
used in quantitative genetic analysis, as well as for many of the
functions in pedtricks.</a></li>
<li><a href='#genome_sim'><p>A function to simulate QTL and/or SNP data.</p></a></li>
<li><a href='#ggpedigree'><p>ggpedigree: Plotting tool for simple and complex pedigrees.</p></a></li>
<li><a href='#gryphons'><p>Example dataset for pedtricks examples and tutorial</p></a></li>
<li><a href='#makemakePedigreeFactor'><p>Deprecated functions from package <span class="pkg">pedantics</span>.</p></a></li>
<li><a href='#micro_sim'><p>Simulates microsatellite data across a pedigree.</p></a></li>
<li><a href='#ped_stats'><p>Calculates a range of statistics of pedigrees</p></a></li>
<li><a href='#pedantics-defunct'><p>Defunct functions from package <span class="pkg">pedantics</span>.</p></a></li>
<li><a href='#phen_sim'><p>A function to simulated phenotypic data</p></a></li>
<li><a href='#plot.ped_stats'><p>Plot output from ped_stats</p></a></li>
<li><a href='#summary.ped_stats'><p>Post-processes output from ped_stats</p></a></li>
<li><a href='#WarblerG'><p>Seychelles Warbler Genotypes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Visualize, Summarize and Simulate Data from Pedigrees</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Sensitivity and power analysis,
  for calculating statistics describing pedigrees from wild populations,
  and for visualizing pedigrees.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://juliengamartin.github.io/pedtricks/">https://juliengamartin.github.io/pedtricks/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/JulienGAMartin/pedtricks/issues">https://github.com/JulienGAMartin/pedtricks/issues</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, dplyr, kinship2, tidyr, Matrix, mvtnorm, MCMCglmm,
nadiv, genetics, methods, igraph</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-04 00:28:02 UTC; julien</td>
</tr>
<tr>
<td>Author:</td>
<td>Julien Martin <a href="https://orcid.org/0000-0001-7726-6809"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Matthew Wolak <a href="https://orcid.org/0000-0002-7962-0071"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Susan Johnston <a href="https://orcid.org/0000-0002-5623-8902"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Michael Morrissey <a href="https://orcid.org/0000-0001-6209-0177"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Julien Martin &lt;julien.martin@uottawa.ca&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-09 08:50:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='pedtricks-package'>pedtricks: Visualize, Summarize and Simulate Data from Pedigrees</h2><span id='topic+pedtricks'></span><span id='topic+pedtricks-package'></span>

<h3>Description</h3>

<p>Sensitivity and power analysis, for calculating statistics describing pedigrees from wild populations, and for visualizing pedigrees.
</p>


<h3>Details</h3>


<ul>
<li> <p><code>pedigreeStats()</code> and <code>pedStatSummary()</code> have been replaced by <code>ped_stats()</code> with a <code>summary()</code> and <code>plot()</code> methods to simplify the workflow and allow to get the plots without running the statistics each time
</p>
</li>
<li> <p><code>makePedigreeFactor()</code> and <code>makePedigreeNumeric()</code> have been combined in <code>convert_ped()</code> which convert a pedigree to numeric or factor
</p>
</li>
<li> <p><code>fixPedigree()</code> is now <code>fix_ped()</code>
</p>
</li>
<li> <p><code>genomesim()</code>, <code>microsim()</code> and <code>phensim()</code> have been renamed as <code>genome_sim()</code>, <code>micro_sim()</code> and <code>phen_sim()</code>
</p>
</li></ul>

<p>For backward compatibility with code using <code>pedantics</code> older names are still usable but not recommended.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Julien Martin <a href="mailto:julien.martin@uottawa.ca">julien.martin@uottawa.ca</a> (<a href="https://orcid.org/0000-0001-7726-6809">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Matthew Wolak <a href="mailto:mew0099@auburn.edu">mew0099@auburn.edu</a> (<a href="https://orcid.org/0000-0002-7962-0071">ORCID</a>)
</p>
</li>
<li><p> Susan Johnston <a href="mailto:Susan.Johnston@ed.ac.uk">Susan.Johnston@ed.ac.uk</a> (<a href="https://orcid.org/0000-0002-5623-8902">ORCID</a>)
</p>
</li>
<li><p> Michael Morrissey <a href="mailto:michael.morrissey@st-andrews.ac.uk">michael.morrissey@st-andrews.ac.uk</a> (<a href="https://orcid.org/0000-0001-6209-0177">ORCID</a>)
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://juliengamartin.github.io/pedtricks/">https://juliengamartin.github.io/pedtricks/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/JulienGAMartin/pedtricks/issues">https://github.com/JulienGAMartin/pedtricks/issues</a>
</p>
</li></ul>


<hr>
<h2 id='convert_ped'>Converts a pedigree with individuals specified as factors to a numeric
pedigree</h2><span id='topic+convert_ped'></span>

<h3>Description</h3>

<p>Some internal <code>pedtricks</code> modules require that pedigrees be
specified only by numerical values, or including numerical values
for missing data. This function provides the conversion to numeric but also
back to factors if needed
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_ped(type = "numeric", id, sire, dam, missingVal = NA, key = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_ped_+3A_type">type</code></td>
<td>
<p>define how to convert the pedigree so &quot;numeric&quot; or &quot;factor&quot;</p>
</td></tr>
<tr><td><code id="convert_ped_+3A_id">id</code></td>
<td>
<p>Individual identifiers - pass using <code>as.character()</code></p>
</td></tr>
<tr><td><code id="convert_ped_+3A_sire">sire</code></td>
<td>
<p>Sire codes - pass <code>using as.character()</code></p>
</td></tr>
<tr><td><code id="convert_ped_+3A_dam">dam</code></td>
<td>
<p>Dam codes - pass <code>using as.character()</code></p>
</td></tr>
<tr><td><code id="convert_ped_+3A_missingval">missingVal</code></td>
<td>
<p>the indicator that should be substituted for missing values</p>
</td></tr>
<tr><td><code id="convert_ped_+3A_key">key</code></td>
<td>
<p>A dataframe, as produced by <code>convert_ped</code>, specifying factor codes for numeric values in id, sire, and dam</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>numericPedigree</code></td>
<td>
<p>The factor pedigree in numeric form</p>
</td></tr>
<tr><td><code>idKey</code></td>
<td>
<p>A key to facilitate conversion back to the original
identifiers</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>pedigree &lt;- as.data.frame(matrix(c(
  "m1",   NA,     NA,
  "m2",   NA,     NA,
  "m3",   NA,     NA,
  "d4",   NA,     NA,
  "d5",   NA,     NA,
  "o6",   "m1",   "d4",
  "o7",   "m1",   "d4",
  "o8",   "m1",   "d4",
  "o9",   "m1",   "d4",
  "o10",  "m2",   "d5",
  "o11",  "m2",   "d5",
  "o12",  "m2",   "d5",
  "o13",  "m2",   "d5",
  "o14",  "m3",   "d5",
  "o15",  "m3",   "d5",
  "o16",  "m3",   "d5",
  "o17",  "m3",   "d5"
), 17, 3, byrow = TRUE))
names(pedigree) &lt;- c("id", "dam", "sire")
for (x in 1:3) pedigree[, x] &lt;- as.factor(pedigree[, x])

## make the test pedigree numeric with NAs denoted by -1
convert_ped(
  type = "numeric",
  id = as.character(pedigree[, 1]),
  dam = as.character(pedigree[, 2]),
  sire = as.character(pedigree[, 3]),
  missingVal = -1
)
</code></pre>

<hr>
<h2 id='draw_ped'>Produce a graphical representation of a pedigree</h2><span id='topic+draw_ped'></span>

<h3>Description</h3>

<p>Plots a pedigree, with options specific to considerations for pedigrees used
to for quantitative genetic inference in natural populations.  Pedigrees
containing only those individuals that are informative with respect to
(genetic) variation in an arbitrary trait can be plotted, potentially
overlain on a complete pedigree.
Functions also exist to plot various types of pedigree links associated with
focal individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_ped(
  Ped,
  cohorts = NULL,
  sex = NULL,
  dat = NULL,
  dots = "n",
  plotfull = "y",
  writeCohortLabels = "n",
  links = "all",
  sexInd = c(0, 1),
  dotSize = 0.001,
  dataDots = "n",
  dataDots.cex = 2,
  cohortLabs.cex = 1,
  retain = "informative",
  focal = NULL,
  sexColours = c("red", "blue"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="draw_ped_+3A_ped">Ped</code></td>
<td>
<p>An ordered pedigree with 3 columns: id, dam, sire</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_cohorts">cohorts</code></td>
<td>
<p>An optional numeric vector of the same length as the pedigree
designating, for example cohort affinities or birth years</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_sex">sex</code></td>
<td>
<p>An optional numeric vector of the same length as the pedigree
containing the sexes (may be unknown) of all individuals with entries in the
pedigree.  Defaults (modifiable with <code>sexInd</code>) are 0=male and 1=female</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_dat">dat</code></td>
<td>
<p>An optional vector or data frame containing indicators of data availability.  If dat contains only ones and zeros, then any individual with any entry of one will be considered as having data records.  If data contains values other than ones and zeros, individuals in the pedigree with rows in data that contain at least one available record, i.e., one data record is not NA, will be treated as having data.</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_dots">dots</code></td>
<td>
<p>If 'y', then a dot will be printed representing each individual in the pedigree.  If sexes are available, dots will be colour coded by sex.</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_plotfull">plotfull</code></td>
<td>
<p>To be used when dat is supplied.  If 'y' (the default), individuals in the pedigree that are uninformative with respect to the available data have their pedigree links plotted in gray.</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_writecohortlabels">writeCohortLabels</code></td>
<td>
<p>To be used when cohorts is used.  Will plot the cohort values on the left hand side of the pedigree image.</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_links">links</code></td>
<td>
<p>Default is 'all', other values are 'mums' to print only maternal pedigree links and 'dads' to print only paternal pedigree links.</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_sexind">sexInd</code></td>
<td>
<p>To be used with if sex is supplied and if the vector of sex specifiers differs from the default.</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_dotsize">dotSize</code></td>
<td>
<p>Set the dot size bigger or smaller</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_datadots">dataDots</code></td>
<td>
<p>Will print dots over the dots denoting individuals, but denoting individuals with available data as indicated by dat.</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_datadots.cex">dataDots.cex</code></td>
<td>
<p>controls the size of dataDots relative to dots.</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_cohortlabs.cex">cohortLabs.cex</code></td>
<td>
<p>controls the size of cohort labels.</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_retain">retain</code></td>
<td>
<p>When those pedigree links only informative relative to phenotypic data availability are to be plotted, this controls whether or not a pruned pedigree based on phenotypic data is plotted (if set to &quot;pruned&quot;), or whether strictly only those informative pedigree links are plotted (the default)</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_focal">focal</code></td>
<td>
<p>An optional list containing the id of an individual and the kinds of relatives of the focal individual to which to plot pedigree links.  Available types are 'offspring','descendants','parents',,ancestors', and 'kin'.</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_sexcolours">sexColours</code></td>
<td>
<p>The colours that will be used to draw points and or lines associated with males and females.</p>
</td></tr>
<tr><td><code id="draw_ped_+3A_...">...</code></td>
<td>
<p>Additional graphical parameters.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>output a plot of the pedigree, and does not return a value
</p>


<h3>Author(s)</h3>

<p>Michael Morrissey <a href="mailto:michael.morrissey@st-andrews.ac.uk">michael.morrissey@st-andrews.ac.uk</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+fix_ped">fix_ped</a></code> to prepare pedigrees that may not explicitly contain records for all individuals (i.e., where founding individuals may only appear in the dam or sire column).)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gryphons)
pedigree &lt;- fix_ped(gryphons[, 1:3])

## draw the gryphon pedigree by pedigree depth
draw_ped(pedigree)

## draw the gryphon pedigree by cohort
draw_ped(pedigree,
  cohorts = gryphons$cohort, writeCohortLabels = "y",
  cohortLabs.cex = 1
)

## draw the gryphon pedigree by cohort with only maternal links
draw_ped(pedigree, cohorts = gryphons$cohort, links = "mums")

## draw the gryphon pedigree by cohort with colour only for those
## indiduals that are informative relative to the quantitative
## genetics of a hypothetical trait only measured for individuals
## in the last two cohorts, emphasize the phenotyped individuals
## with large black dots, and all other individuals with dots
## colour coded by sex:
dataAvailability &lt;- (gryphons$cohort &gt;= (max(gryphons$cohort) - 1)) + 0

draw_ped(pedigree,
  cohorts = gryphons$cohort, sex = gryphons$sex,
  dots = "y", dat = dataAvailability, writeCohortLabels = "y", dataDots = "y"
)

</code></pre>

<hr>
<h2 id='draw_pedA'>Produce a graphical representation of the relatedness matrix of a pedigree</h2><span id='topic+draw_pedA'></span>

<h3>Description</h3>

<p>Creates the object needed to plot a pedigree's numerator relatedness matrix
given a few different choices for ordering. The resulting image for a pedigree
of size n can be visualized as a n x n grid of colored squares based on
values of the numerator relatedness matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_pedA(
  pedigree,
  order = c("original", "generation", "user"),
  grp = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="draw_pedA_+3A_pedigree">pedigree</code></td>
<td>
<p>A data.frame of a pedigree with 3 columns: id, dam, sire</p>
</td></tr>
<tr><td><code id="draw_pedA_+3A_order">order</code></td>
<td>
<p>A character expression for how the pedigree should be ordered
for visualization. See Details.</p>
</td></tr>
<tr><td><code id="draw_pedA_+3A_grp">grp</code></td>
<td>
<p>A character expression for the column name in pedigree indicating
how to order the pedigree for visualization.</p>
</td></tr>
<tr><td><code id="draw_pedA_+3A_...">...</code></td>
<td>
<p>Additional plotting arguments passed to <code><a href="Matrix.html#topic+image">image</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class &ldquo;trellis&rdquo;.
</p>


<h3>Author(s)</h3>

<p><a href="mailto:mew0099@auburn.edu">mew0099@auburn.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(gryphons)
pedigree &lt;- fix_ped(gryphons[, 1:3])

## draw the gryphon pedigree
draw_pedA(pedigree, order = "original")

## draw the gryphon pedigree by function assigned generation
(Agen &lt;- draw_pedA(pedigree, order = "generation"))



## draw the gryphon pedigree by cohort in the dataset
## add cohort back from original data
pedigree$cohort &lt;- NA
pedigree$cohort[match(gryphons$id, pedigree[, 1])] &lt;- gryphons$cohort
(Achrt &lt;- draw_pedA(pedigree, order = "user", grp = "cohort"))

## show two images of the same pedigree in different orders
### (i.e., plotting multiple trellis objects in the same figure)
plot(Agen,
  position = c(xmin = 0, ymin = 0, xmax = 0.45, ymax = 1),
  more = TRUE
)
plot(Achrt, position = c(xmin = 0.55, ymin = 0, xmax = 1, ymax = 1))


</code></pre>

<hr>
<h2 id='extractA'>Allele Frequencies</h2><span id='topic+extractA'></span>

<h3>Description</h3>

<p>Extracts allele frequencies from genotype data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extractA(G, marker.type = "MSW")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extractA_+3A_g">G</code></td>
<td>
<p>data frame or list of <code>genotype</code> objects</p>
</td></tr>
<tr><td><code id="extractA_+3A_marker.type">marker.type</code></td>
<td>
<p><code>"MSW"</code> or <code>"MSC"</code> for co-dominant markers with Wang's (2004) model of genotyping error or CERVUS's model of genotyping error (Marshall, 1998) or <code>"AFLP"</code> for dominant markers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of allele frequencies at each loci
</p>


<h3>Author(s)</h3>

<p>Jarrod Hadfield <a href="mailto:j.hadfield@ed.ac.uk">j.hadfield@ed.ac.uk</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(WarblerG)

A &lt;- extractA(WarblerG)
A[[1]]


</code></pre>

<hr>
<h2 id='fix_ped'>Manipulating pedigrees to prepare them for requirements of
subsequent analyses
Prepares a pedigree to conform with requirements of many softwares
used in quantitative genetic analysis, as well as for many of the
functions in pedtricks.</h2><span id='topic+fix_ped'></span>

<h3>Description</h3>

<p>Manipulating pedigrees to prepare them for requirements of
subsequent analyses
Prepares a pedigree to conform with requirements of many softwares
used in quantitative genetic analysis, as well as for many of the
functions in pedtricks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fix_ped(ped, dat = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fix_ped_+3A_ped">ped</code></td>
<td>
<p>An ordered pedigree with 3 columns: id, dam, sire</p>
</td></tr>
<tr><td><code id="fix_ped_+3A_dat">dat</code></td>
<td>
<p>An optional data frame, the same length as the pedigree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a pedigree in which all individuals that exist in the dam
and sire columns are represented by their own record lines, occurring
before the records of their first offspring.  If data are supplied,
then fix_ped will return a dataframe, the first three columns are
the 'fixed' pedigree, and the following columns of which contain
appropriately reordered data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##  a valid pedigree, i.e., no loops, no bisexuality, etc.,
## but where not all parents have a record line, and where
## parents do not necessarily occur before their offspring:
pedigree &lt;- as.data.frame(matrix(c(
  10, 1, 2,
  11, 1, 2,
  12, 1, 3,
  13, 1, 3,
  14, 4, 5,
  15, 6, 7,
  4, NA, NA,
  5, NA, NA,
  6, NA, NA,
  7, NA, NA
), 10, 3, byrow = TRUE))
names(pedigree) &lt;- c("id", "dam", "sire")
pedigree
fixed_pedigree &lt;- fix_ped(ped = pedigree)
fixed_pedigree

</code></pre>

<hr>
<h2 id='genome_sim'>A function to simulate QTL and/or SNP data.</h2><span id='topic+genome_sim'></span>

<h3>Description</h3>

<p>Simulates a chromosome of arbitrary length with arbitrary numbers, types, and spacings of genetic loci over arbitrary pedigrees.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genome_sim(
  pedigree,
  founders = NULL,
  positions = NULL,
  initHe = NULL,
  mutationType = NULL,
  mutationRate = NULL,
  phenotyped = NULL,
  founderHaplotypes = NULL,
  genotyped = NULL,
  returnG = "n",
  initFreqs = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="genome_sim_+3A_pedigree">pedigree</code></td>
<td>
<p>A pedigree</p>
</td></tr>
<tr><td><code id="genome_sim_+3A_founders">founders</code></td>
<td>
<p>A vector of indicator variables denoting founder status (1=founder, 0=non-founder)</p>
</td></tr>
<tr><td><code id="genome_sim_+3A_positions">positions</code></td>
<td>
<p>Genome locations in cM for markers</p>
</td></tr>
<tr><td><code id="genome_sim_+3A_inithe">initHe</code></td>
<td>
<p>Initial levels of expected heterozygosity</p>
</td></tr>
<tr><td><code id="genome_sim_+3A_mutationtype">mutationType</code></td>
<td>
<p>A vector of locus types - see details</p>
</td></tr>
<tr><td><code id="genome_sim_+3A_mutationrate">mutationRate</code></td>
<td>
<p>A vector of mutation rates</p>
</td></tr>
<tr><td><code id="genome_sim_+3A_phenotyped">phenotyped</code></td>
<td>
<p>A vector of IDs of those individuals for which to return phenotypic data</p>
</td></tr>
<tr><td><code id="genome_sim_+3A_founderhaplotypes">founderHaplotypes</code></td>
<td>
<p>A matrix or dataframe containing founder haplotypes</p>
</td></tr>
<tr><td><code id="genome_sim_+3A_genotyped">genotyped</code></td>
<td>
<p>A vector of IDs of those individuals for which to return genotypic data</p>
</td></tr>
<tr><td><code id="genome_sim_+3A_returng">returnG</code></td>
<td>
<p>If 'y' then genotypic data for all loci (including <code>cIAM</code> loci) will be returned.</p>
</td></tr>
<tr><td><code id="genome_sim_+3A_initfreqs">initFreqs</code></td>
<td>
<p>A list of allele frequencies for all loci.  If <code>initFreqs</code> is specified, it will override information from <code>initHe</code>.  <code>extractA</code> from package <code>MasterBayes</code> can be used to obtain obtain <code>initFreqs</code> form a sample of genotypes.  For <code>cIAM</code> loci, allele names in <code>initFreqs</code> should be allelic substitution effects.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Valid mutation types are <code style="white-space: pre;">&#8288;Micro', &#8288;</code>Dom', <code style="white-space: pre;">&#8288;dIAM' and &#8288;</code>cIAM', for microsatellite, dominant (AFLP), discrete infinite alleles mutation model loci (SNPs), and continuous infinite alleles mutation model loci (polymorphisms effecting phenotypic variation).  cIAM loci have mutational allelic substitution effects taken drawn from a normal distribution with mean 0 and variance 1.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Phenotypes</code></td>
<td>
<p>A vector of phenotypes.  Calculated as the sum of all allelic effects.  Scaling is currently left to be done post-hoc.</p>
</td></tr>
<tr><td><code>MarkerData</code></td>
<td>
<p>A vector of marker genotypes, i.e. alleles at all loci except those designated &lsquo;cIAM&rsquo;</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+phen_sim">phen_sim</a></code>, <code><a href="#topic+micro_sim">micro_sim</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>testData &lt;- as.data.frame(matrix(
  c(
    1,      NA,     NA,     1,      1,      1,      2,      2,
    2,      NA,     NA,     1,      1,      1,      2,      2,
    3,      NA,     NA,     1,      1,      1,      2,      2,
    4,      NA,     NA,     1,      0,      1,      2,      2,
    5,      NA,     NA,     1,      0,      1,      2,      2,
    6,      1,      4,      0,      -1,     2,      3,      3,
    7,      1,      4,      0,      -1,     2,      3,      3,
    8,      1,      4,      0,      -1,     2,      3,      3,
    9,      1,      4,      0,      -1,     2,      3,      3,
    10,     2,      5,      0,      -1,     2,      3,      3,
    11,     2,      5,      0,      -1,     2,      3,      3,
    12,     2,      5,      0,      -1,     2,      3,      3,
    13,     2,      5,      0,      -1,     2,      3,      3,
    14,     3,      5,      0,      -1,     2,      3,      3,
    15,     3,      5,      0,      -1,     2,      3,      3,
    16,     3,      5,      0,      -1,     2,      3,      3,
    17,     3,      5,      0,      -1,     2,      3,      3
  ),
  17, 8,
  byrow = TRUE
))

names(testData) &lt;- c(
  "id", "dam", "sire", "founder", "sex",
  "cohort", "first", "last"
)
pedigree &lt;- as.data.frame(cbind(
  testData$id, testData$dam,
  testData$sire
))
for (x in 1:3) pedigree[, x] &lt;- as.factor(pedigree[, x])
names(pedigree) &lt;- c("id", "dam", "sire")
pedigree

## make up some microsatellite and gene allele frquencies:
sampleGenotypes &lt;- as.data.frame(matrix(c(
  1, 2, -1.32, 0.21, 2, 1, 0.21, 0.21
), 2, 4, byrow = TRUE))
testFreqs &lt;- extractA(sampleGenotypes)

## note that alleles at the gene locus are given as their
## allelic substitution effects:
testFreqs

## simulate data for these indivdiuals based on a single QTL
## with two equally alleles with balanced frequencies in the
## founders, linked (2 cM) to a highly  polymorphic microsatellite:
genome_sim(
  pedigree = pedigree, founders = testData$founder, positions = c(0, 2),
  mutationType = c("Micro", "cIAM"), mutationRate = c(0, 0),
  initFreqs = testFreqs, returnG = "y"
)
## since we specified returnG='y', we can check that
## the phenotypes add up to the
## allelic substitution effects for the second locus.

</code></pre>

<hr>
<h2 id='ggpedigree'>ggpedigree: Plotting tool for simple and complex pedigrees.</h2><span id='topic+ggpedigree'></span>

<h3>Description</h3>

<p>This function plots simple and complex pedigrees, with options specific to
the types of pedigrees used for quantitative genetic inference in natural
populations. This function is flexible to missing parents and can be
customized to visualise specific cohorts, sexes, and/or phenotype
availability. Pedigree layout is optimized using a Sugiyama algorithm. For
simpler pedigrees, visualisation may be improved by specifying
<code>spread_x_coordinates = FALSE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggpedigree(
  .data,
  ids,
  mothers,
  fathers,
  cohort,
  sex,
  pheno,
  sex_code = NULL,
  id_labels = FALSE,
  remove_singletons = TRUE,
  plot_unknown_cohort = FALSE,
  spread_x_coordinates = TRUE,
  print_cohort_labels = TRUE,
  return_plot_tables = FALSE,
  line_col_mother = "#E41A1C",
  line_col_father = "#377EB8",
  line_col_no_pheno = "#aaaaaa",
  line_alpha = 0.3,
  point_size = 1,
  point_colour = "black",
  point_alpha = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ggpedigree_+3A_.data">.data</code></td>
<td>
<p>a data frame object with all the pedigree information</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_ids">ids</code></td>
<td>
<p>a column of .data of individual identifiers</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_mothers">mothers</code></td>
<td>
<p>A column of .data of mothers corresponding to ids. Missing
values are 0 or NA.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_fathers">fathers</code></td>
<td>
<p>A column of .data of fathers corresponding to ids. Missing
values are 0 or NA.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_cohort">cohort</code></td>
<td>
<p>integer. Default NULL. A optional column of .data assigning a
cohort to each id. If NULL, then <code>kinship2::kindepth</code> is used to assign
cohorts to ids.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_sex">sex</code></td>
<td>
<p>integer or character. Default NULL. An optional column of .data
assigning a sex to each id. When using this option, <code>sex_code</code> must be
specified. Any values not matching values in <code>sex_code</code> will be treated as
unknown sex.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_pheno">pheno</code></td>
<td>
<p>integer or character. Default NULL. An optional column of .data
assigning a phenotype to each id. Links originating from parents that have
<code>NA</code> values for this argument will be plotted with a grey line, unless
otherwise specified in <code>line_col_no_pheno</code>.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_sex_code">sex_code</code></td>
<td>
<p>Default NULL. A vector of length 2, indicating the value used
in <code>sex</code> for females and males respectively. Females are plotted as
circles, males as squares, and unknown values as triangles.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_id_labels">id_labels</code></td>
<td>
<p>logical. Default FALSE. Print the ids on the pedigree plot.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_remove_singletons">remove_singletons</code></td>
<td>
<p>logical. Default TRUE. Remove ids with no relatives
i.e., no offspring or parents assigned.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_plot_unknown_cohort">plot_unknown_cohort</code></td>
<td>
<p>logical. Default FALSE. Plots ids of unknown
cohorts. These are plotted in an &quot;Unknown&quot; cohort at the top of the
pedigree. Be aware that any mothers and fathers of these individuals will
be plotted below them.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_spread_x_coordinates">spread_x_coordinates</code></td>
<td>
<p>logical. Default TRUE. Evenly spreads the x-axis
(horizontal) distribution of points within each cohort. If FALSE, this will
plot the direct outcome of <code>igraph::layout_with_sugiyama</code>; the FALSE option
is only recommended for small pedigrees and/or less connected pedigrees.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_print_cohort_labels">print_cohort_labels</code></td>
<td>
<p>logical. Default TRUE. Prints cohort levels on the
left hand side of plot.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_return_plot_tables">return_plot_tables</code></td>
<td>
<p>logical. Default FALSE. Returns an object with the
line and point data used for the plot, but the plot will not be generated</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_line_col_mother">line_col_mother</code></td>
<td>
<p>Default = &quot;#E41A1C&quot;. Line colour for maternal links.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_line_col_father">line_col_father</code></td>
<td>
<p>Default = &quot;#377EB8&quot;. Line colour for paternal links.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_line_col_no_pheno">line_col_no_pheno</code></td>
<td>
<p>Default = &quot;#aaaaaa&quot;. Line colour for parents with
<code>NA</code> values in <code>pheno</code>.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_line_alpha">line_alpha</code></td>
<td>
<p>Default = 0.3. Line alpha (transparency) value for maternal
and paternal links.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_point_size">point_size</code></td>
<td>
<p>Default = 1. Point size for ids.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_point_colour">point_colour</code></td>
<td>
<p>Default = &quot;black&quot;. Point colour for ids.</p>
</td></tr>
<tr><td><code id="ggpedigree_+3A_point_alpha">point_alpha</code></td>
<td>
<p>Default = 1. Point alpha for ids.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>output a ggplot object or
a list of tables if <code>return_plot_tables = TRUE</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gryphons)
pedigree &lt;- fix_ped(gryphons[, 1:3])

## draw the gryphon pedigree by pedigree depth
ggpedigree(pedigree)


# specifying the column names for id, mother and father
ggpedigree(pedigree, id, dam, sire)

# with cohort and sex
ggpedigree(gryphons, cohort = cohort, sex = sex, sex_code = c(1, 0))

#' with cohort, sex, and pheno
gryphons$pheno &lt;- 1
gryphons$pheno[sample(length(gryphons$pheno), 1000)] &lt;- NA
ggpedigree(gryphons, cohort = cohort, sex = sex, sex_code = c(1, 0), pheno = pheno)


</code></pre>

<hr>
<h2 id='gryphons'>Example dataset for pedtricks examples and tutorial</h2><span id='topic+gryphons'></span>

<h3>Description</h3>

<p>This contains pedigree and life history data of a fictional
population.  The data are relevant to power and sensitivity
analyses for quantitative genetic studies of natural populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gryphons
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 4918 rows and 9 columns.
</p>


<h3>Details</h3>


<dl>
<dt>id</dt><dd><p>what</p>
</dd>
<dt>dam</dt><dd><p>what</p>
</dd>
<dt>sire</dt><dd><p>what</p>
</dd>
<dt>sex</dt><dd><p>what</p>
</dd>
<dt>cohort</dt><dd><p>what</p>
</dd>
<dt>fatherErrorProb</dt><dd><p>what</p>
</dd>
<dt>fatherSampledProb</dt><dd><p>what</p>
</dd>
<dt>firstReproCohort</dt><dd><p>what</p>
</dd>
<dt>lastReproCohort</dt><dd><p>what</p>
</dd>
</dl>


<hr>
<h2 id='makemakePedigreeFactor'>Deprecated functions from package <span class="pkg">pedantics</span>.</h2><span id='topic+makemakePedigreeFactor'></span><span id='topic+makePedigreeNumeric'></span><span id='topic+drawPedigree'></span><span id='topic+fixPedigree'></span><span id='topic+genomesim'></span><span id='topic+microsim'></span><span id='topic+pedigreeStat'></span><span id='topic+pedStatSummary'></span><span id='topic+pedantics-deprecated'></span><span id='topic+phensim'></span>

<h3>Description</h3>

<p>The functions listed below are deprecated and are available for
backward compatibility only. They might be defunct in a future version
When possible, alternative functions with similar
functionality are also mentioned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makemakePedigreeFactor(id, sire, dam, key = NULL)

makePedigreeNumeric(id, sire, dam, missingVal = NA)

drawPedigree(
  Ped,
  cohorts = NULL,
  sex = NULL,
  dat = NULL,
  dots = "n",
  plotfull = "y",
  writeCohortLabels = "n",
  links = "all",
  sexInd = c(0, 1),
  dotSize = 0.001,
  dataDots = "n",
  dataDots.cex = 2,
  cohortLabs.cex = 1,
  retain = "informative",
  focal = NULL,
  sexColours = c("red", "blue"),
  ...
)

fixPedigree(ped, dat = NULL)

genomesim(
  pedigree,
  founders = NULL,
  positions = NULL,
  initHe = NULL,
  mutationType = NULL,
  mutationRate = NULL,
  phenotyped = NULL,
  founderHaplotypes = NULL,
  genotyped = NULL,
  returnG = "n",
  initFreqs = NULL
)

microsim(
  pedigree,
  genFreqs = NULL,
  genotypesSample = NULL,
  knownGenotypes = NULL,
  records = NULL,
  eRate1 = 0,
  eRate2 = 0,
  eRate3 = 0
)

pedigreeStat()

pedStatSummary()

phensim(
  pedigree,
  traits = 1,
  randomA = NULL,
  randomE = NULL,
  parentalA = NULL,
  parentalE = NULL,
  sampled = NULL,
  records = NULL,
  returnAllEffects = FALSE
)
</code></pre>


<h3>Value</h3>

<p>dependant on the deprecated function see pedantics help for details
</p>


<h3><code>makePedigreeFactor</code></h3>

<p>the function is now a special case of the <code>convert_ped</code> function when type is factor
</p>


<h3><code>makePedigreeNumeric</code></h3>

<p>the function is now a special case of the <code>convert_ped</code> function when type is numeric
</p>


<h3><code>drawPedigree</code></h3>

<p>the function has just been renamed with no other changes for the moment, but will soon be replace by &quot;ggpedigree&quot; function
</p>


<h3><code>fixPedigree</code></h3>

<p>the function has just been renamed with no
other changes for the moment
</p>


<h3><code>genomesim</code></h3>

<p>the function has just been renamed with no other changes for the moment
</p>


<h3><code>microsim</code></h3>

<p>the function has just been renamed with no other changes for the moment
</p>


<h3><code>pedigreeStats</code></h3>

<p>the function has been simplified but only the functionality are still available via <code>ped_stats</code> and its summary and plot methods
</p>


<h3><code>pedStatSummary</code></h3>

<p>the function has been simplified but only the functionality are still available via <code>ped_stats</code> and its summary and plot methods
</p>


<h3><code>phensim</code></h3>

<p>the function has just been renamed with no other changes for the moment
</p>

<hr>
<h2 id='micro_sim'>Simulates microsatellite data across a pedigree.</h2><span id='topic+micro_sim'></span>

<h3>Description</h3>

<p>Uses a pedigree with parents identified for all non-founding
individuals and simulates microsatellite genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>micro_sim(
  pedigree,
  genFreqs = NULL,
  genotypesSample = NULL,
  knownGenotypes = NULL,
  records = NULL,
  eRate1 = 0,
  eRate2 = 0,
  eRate3 = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="micro_sim_+3A_pedigree">pedigree</code></td>
<td>
<p>A pedigree</p>
</td></tr>
<tr><td><code id="micro_sim_+3A_genfreqs">genFreqs</code></td>
<td>
<p>(optional) A list of allele frequencies, can be produced with <code>extractA</code></p>
</td></tr>
<tr><td><code id="micro_sim_+3A_genotypessample">genotypesSample</code></td>
<td>
<p>(required if <code>genFreqs</code> is not supplied) a sample of genotypes from which to estimate population allele frequencies</p>
</td></tr>
<tr><td><code id="micro_sim_+3A_knowngenotypes">knownGenotypes</code></td>
<td>
<p>(not yet implemented) a data frame of genotypes for (potentially a subset) of founder individuals</p>
</td></tr>
<tr><td><code id="micro_sim_+3A_records">records</code></td>
<td>
<p>Record availability, see details.</p>
</td></tr>
<tr><td><code id="micro_sim_+3A_erate1">eRate1</code></td>
<td>
<p>The rate of genotypic substitution errors, i.e., when a true genotype at a given locus is replaced by a pair of alleles selected at random based on the population allele frequencies</p>
</td></tr>
<tr><td><code id="micro_sim_+3A_erate2">eRate2</code></td>
<td>
<p>The rate of allelic substitution errors, i.e. when an allele is erroneously replaced at a given locus by an allele chosen at random based on the population allele frequencies</p>
</td></tr>
<tr><td><code id="micro_sim_+3A_erate3">eRate3</code></td>
<td>
<p>The rate of large allele dropouts, simulated by setting the value of the larger allele at a locus to the value of the smaller allele</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Error rates and data availability rates can be specified as either (1) single values to be applied to all individuals and all loci, (2) as a vector the same length as the number of loci, representing locus-specific rates to be applied uniformly to all individuals, or (3) as data frames with rows for each individual and columns for each locus.  In the third option, observed patterns of data availability can be simulated by supplying 0s and 1s for missing and available individual genotypes, respectively.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>trueGenotypes</code></td>
<td>
<p>A data frame of true genotypes</p>
</td></tr>
<tr><td><code>observedGenotypes</code></td>
<td>
<p>A data frame of plausible observed genotypes, given specified patterns of missingness and errors.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+phen_sim">phen_sim</a></code>, <code><a href="#topic+genome_sim">genome_sim</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
pedigree &lt;- as.data.frame(matrix(c(
  "m1",   NA,     NA,
  "m2",   NA,     NA,
  "m3",   NA,     NA,
  "d4",   NA,     NA,
  "d5",   NA,     NA,
  "o6",   "m1",   "d4",
  "o7",   "m1",   "d4",
  "o8",   "m1",   "d4",
  "o9",   "m1",   "d4",
  "o10",  "m2",   "d5",
  "o11",  "m2",   "d5",
  "o12",  "m2",   "d5",
  "o13",  "m2",   "d5",
  "o14",  "m3",   "d5",
  "o15",  "m3",   "d5",
  "o16",  "m3",   "d5",
  "o17",  "m3",   "d5"
), 17, 3, byrow = TRUE))
names(pedigree) &lt;- c("id", "dam", "sire")
for (x in 1:3) pedigree[, x] &lt;- as.factor(pedigree[, x])

## some sample genotypes, very simple, two markers with He = 0.5
sampleGenotypes &lt;- as.data.frame(matrix(c(
  1, 2, 1, 2, 2, 1, 2, 1
), 2, 4, byrow = TRUE))
## locus names
names(sampleGenotypes) &lt;- c("loc1a", "loc1b", "loc2a", "loc2b")

## simulate some genotypes
micro_sim(pedigree = pedigree, genotypesSample = sampleGenotypes)

</code></pre>

<hr>
<h2 id='ped_stats'>Calculates a range of statistics of pedigrees</h2><span id='topic+ped_stats'></span><span id='topic+pedigreeStats'></span>

<h3>Description</h3>

<p>Statistics are those that will hopefully be useful for describing pedigrees to be used in quantitative genetic analyses of natural populations.  This module will be most useful when cohort affinities for all individuals can be provided.  All outputs are produced in a numerical form as well as in graphical summaries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ped_stats(
  Ped,
  cohorts = NULL,
  dat = NULL,
  retain = "informative",
  includeA = TRUE,
  lowMem = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ped_stats_+3A_ped">Ped</code></td>
<td>
<p>A pedigree</p>
</td></tr>
<tr><td><code id="ped_stats_+3A_cohorts">cohorts</code></td>
<td>
<p>(Optional) Cohort affinities for members of the pedigree</p>
</td></tr>
<tr><td><code id="ped_stats_+3A_dat">dat</code></td>
<td>
<p>(Optional) Available data based upon which the pedigree can be pruned for just informative individuals</p>
</td></tr>
<tr><td><code id="ped_stats_+3A_retain">retain</code></td>
<td>
<p>The default value ('informative') results in pedigree being pruned to only those individuals who's records contribute to estimation of quantitative genetic parameters with respect to the available data specified in <code>dat</code>.  Otherwise, specifying a value of 'ancestors' will result in the inclusion of all ancestors of phenotyped individuals.</p>
</td></tr>
<tr><td><code id="ped_stats_+3A_includea">includeA</code></td>
<td>
<p>If TRUE, additive genetic relatedness matrix is returned.</p>
</td></tr>
<tr><td><code id="ped_stats_+3A_lowmem">lowMem</code></td>
<td>
<p>If TRUE, then stats based on calculation of A are not performed.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>totalMaternities</code></td>
<td>
<p>Total number of maternities defined by the pedigree.</p>
</td></tr>
<tr><td><code>totalPaternities</code></td>
<td>
<p>Total number of paternities defined by the pedigree.</p>
</td></tr>
<tr><td><code>totalFullSibs</code></td>
<td>
<p>Total number of pair-wise full sib relationships defined by the pedigree.</p>
</td></tr>
<tr><td><code>totalMaternalSibs</code></td>
<td>
<p>Total number of pair-wise maternal sib relationships defined by the pedigree.  To get the number of maternal half sibs, subtract totalFullSibs.</p>
</td></tr>
<tr><td><code>totalPaternalSibs</code></td>
<td>
<p>Total number of pair-wise paternal sib relationships defined by the pedigree.  To get the number of paternal half sibs, subtract totalFullSibs.</p>
</td></tr>
<tr><td><code>totalMaternalGrandmothers</code></td>
<td>
<p>Total number of maternal grandmothers defined by the pedigree.</p>
</td></tr>
<tr><td><code>totalMaternalGrandfathers</code></td>
<td>
<p>Total number of maternal grandfathers defined by the pedigree.</p>
</td></tr>
<tr><td><code>totalPaternalGrandmothers</code></td>
<td>
<p>Total number of paternal grandmothers defined by the pedigree.</p>
</td></tr>
<tr><td><code>totalPaternalGrandfathers</code></td>
<td>
<p>Total number of paternal grandfathers defined by the pedigree.</p>
</td></tr>
<tr><td><code>pedigreeDepth</code></td>
<td>
<p>The pedigree depth, i.e. maximum number of ancestral generations, for each individual.</p>
</td></tr>
<tr><td><code>inbreedingCoefficients</code></td>
<td>
<p>Individual inbreeding coefficients</p>
</td></tr>
<tr><td><code>maternalSibships</code></td>
<td>
<p>Sibship size of each individual appearing the the dam column of the pedigree.</p>
</td></tr>
<tr><td><code>paternalSibships</code></td>
<td>
<p>Sibship size of each individual appearing the the sire column of the pedigree.</p>
</td></tr>
<tr><td><code>cumulativeRelatedness</code></td>
<td>
<p>Proportion of pair-wise relatedness values less than values ranging from 0 to 1.</p>
</td></tr>
<tr><td><code>relatednessCategories</code></td>
<td>
<p>Discretized distribution of relatedness.</p>
</td></tr>
<tr><td><code>analyzedPedigree</code></td>
<td>
<p>Returns the pedigree.</p>
</td></tr>
<tr><td><code>sampleSizesByCohort</code></td>
<td>
<p>(Optional) Number of individuals belonging to each cohort.</p>
</td></tr>
<tr><td><code>maternitiesByCohort</code></td>
<td>
<p>(Optional) Number of assigned maternities by offspring cohort.</p>
</td></tr>
<tr><td><code>paternitiesByCohort</code></td>
<td>
<p>(Optional) Number of assigned paternities by offspring cohort.</p>
</td></tr>
<tr><td><code>fullSibsByCohort</code></td>
<td>
<p>(Optional) Number of pair-wise full sib relationships by cohort - note the sum of these need not be equal to totalFullSibs in pedigrees of long-lived organisms.</p>
</td></tr>
<tr><td><code>maternalSibsByCohort</code></td>
<td>
<p>(Optional) Number of pair-wise maternal sib relationships by cohort - note the sum of these need not be equal to totalMaternalSibs in pedigrees of long-lived organisms.</p>
</td></tr>
<tr><td><code>paternalSibsByCohort</code></td>
<td>
<p>(Optional) Number of pair-wise paternal sib relationships by cohort - note the sum of these need not be equal to totalPaternalSibs in pedigrees of long-lived organisms.</p>
</td></tr>
<tr><td><code>maternalGrandmothersByCohort</code></td>
<td>
<p>(Optional) Numbers of maternal grandmother assignments by offspring cohort.</p>
</td></tr>
<tr><td><code>maternalGrandfathersByCohort</code></td>
<td>
<p>(Optional) Numbers of maternal grandmother assignments by offspring cohort.</p>
</td></tr>
<tr><td><code>paternalGrandmothersByCohort</code></td>
<td>
<p>(Optional) Numbers of paternal grandfather assignments by offspring cohort.</p>
</td></tr>
<tr><td><code>paternalGrandfathersByCohort</code></td>
<td>
<p>(Optional) Numbers of paternal grandfather assignments by offspring cohort.</p>
</td></tr>
<tr><td><code>cumulativePedigreeDepth</code></td>
<td>
<p>(Optional) Distributions of pedigree depth by cohort.</p>
</td></tr>
<tr><td><code>meanRelatednessAmongCohorts</code></td>
<td>
<p>(Optional) Mean relatedness among cohorts.</p>
</td></tr>
<tr><td><code>cohorts</code></td>
<td>
<p>(Optional) Returns cohort designations.</p>
</td></tr>
</table>
<p>Graphical summaries of a number of these summary statistics are printed to the console when <code>graphicalReports=='y'</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(gryphons)
pedigree &lt;- gryphons[, 1:3]

gryphons_ped_stats &lt;- ped_stats(pedigree,
  cohorts = gryphons$cohort
)

gryphons_ped_stats$totalMaternities
gryphons_ped_stats$paternitiesByCohort

summary(gryphons_ped_stats)

plot(gryphons_ped_stats)

</code></pre>

<hr>
<h2 id='pedantics-defunct'>Defunct functions from package <span class="pkg">pedantics</span>.</h2><span id='topic+pedantics-defunct'></span>

<h3>Description</h3>

<p>The functions listed below have not been transferred from pedantics to
pedtricks with no short-term plan to implement them since they were relying
on C code that need to be heavily updated.
</p>


<h3>Details</h3>

<p>In addition, The questions about rates of pedigree error typically
encountered, and their effects on inference of elementary QG parameters are
basically settled (Charmantier &amp; Réale 2005, Morrissey <em>et al.</em> 2007,
Firth <em>et al.</em> 2015, Bourret &amp; Garant 2017), and outstanding questions will
not plausibly be supported by these functions or any straightforward
modifications thereof.
</p>
<p>Bourret, A., and D. Garant. 2017. An assessment of the reliability of
quantitative genetics estimates in study systems with high rate of extra
pair paternity and low recruitment.  Heredity 118: 229-38.
</p>
<p>Charmantier, A., and D. Réale. 2005. How do misassigned paternities affect
estimation of heritability in the wild? Molecular Ecology 14: 2839-50
</p>
<p>Firth, J.A. J.D. Hadfield, A.W. Santure, J. Slate and B.C. Sheldon. 2015.
The influence of non-random extra pair paternity on heritability estimates
derived from wild pedigrees. Evolution 69: 1336-44.
</p>
<p>Morrissey, M.B., A.J. Wilson, J.M. Pemberton, and M.M. Ferguson. 2007. A
framework for power and sensitivity analysis for quantitative genetic studies
of natural populations, and a case study in Soay sheep (Ovis aries). Journal
of Evolutionary Biology 20: 2309-21.
</p>


<h3><code>rpederr</code></h3>

<p>Permutes a pedigree to create a plausible complete pedigree
</p>


<h3><code>fpederr</code></h3>

<p>Simulates a pedigree with errors and missing data from a complete pedigree.
</p>

<hr>
<h2 id='phen_sim'>A function to simulated phenotypic data</h2><span id='topic+phen_sim'></span>

<h3>Description</h3>

<p>Simulates phenotypic data across arbitrary pedigrees.  <code> phen_sim</code> simulate direct, maternal and paternal genetic and environmental effects for an arbitrary number of traits with arbitrary patterns of missing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phen_sim(
  pedigree,
  traits = 1,
  randomA = NULL,
  randomE = NULL,
  parentalA = NULL,
  parentalE = NULL,
  sampled = NULL,
  records = NULL,
  returnAllEffects = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phen_sim_+3A_pedigree">pedigree</code></td>
<td>
<p>A pedigree</p>
</td></tr>
<tr><td><code id="phen_sim_+3A_traits">traits</code></td>
<td>
<p>The number of traits for which data should be simulated.</p>
</td></tr>
<tr><td><code id="phen_sim_+3A_randoma">randomA</code></td>
<td>
<p>An additive genetic covariance matrix, with dimensions a multiple of traits - see details</p>
</td></tr>
<tr><td><code id="phen_sim_+3A_randome">randomE</code></td>
<td>
<p>An additive environmental covariance matrix, with dimensions a multiple of traits - see details</p>
</td></tr>
<tr><td><code id="phen_sim_+3A_parentala">parentalA</code></td>
<td>
<p>A vector indicating which effects in <code>randomA</code> (if any) to treat as parental effects</p>
</td></tr>
<tr><td><code id="phen_sim_+3A_parentale">parentalE</code></td>
<td>
<p>A vector indicating which effects in <code>randomE</code> (if any) to treat as parental effects</p>
</td></tr>
<tr><td><code id="phen_sim_+3A_sampled">sampled</code></td>
<td>
<p>A vector indicating which individuals are sampled</p>
</td></tr>
<tr><td><code id="phen_sim_+3A_records">records</code></td>
<td>
<p>A single value, array of matrix specifying data record availability - see details</p>
</td></tr>
<tr><td><code id="phen_sim_+3A_returnalleffects">returnAllEffects</code></td>
<td>
<p>If <code>TRUE</code> then all individual breeding values and environmental effects are returned</p>
</td></tr>
<tr><td><code id="phen_sim_+3A_verbose">verbose</code></td>
<td>
<p>If <code>TRUE</code> provide a progress bar and messages, Default: TRUE</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>randomA</code> and <code>randomE</code> are square matrices with dimension equal to the sum of the number direct and indirect effects.  This must be a multiple of the number of traits, i.e. if an indirect effect is to be simulated for only one of multiple traits, those traits with no indirect effect should be included with (co)variances of zero.
</p>
<p><code>parentalA</code> and <code>parentalE</code> are optional vectors of characters indicating which trait positions in <code>randomA</code> and <code>randomE</code> are to be treated as indirect effects, and which effects to treat as maternal or paternal.  Valid values are 'd', 'm', and 'p', for direct, maternal indirect and paternal indirect effects, respectively.
</p>
<p><code>records</code> can be specified either (1) as a single value to be applied to all individuals and traits, (2) as a vector the same length as the number of traits, representing trait-specific rates to be applied uniformly to all individuals, or (3) as data frames with rows for each individual and columns for each trait.  In the third option, observed patterns of data availability can be simulated by supplying 0s and 1s for missing and available individual genotypes, respectively.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>phenotypes</code></td>
<td>
<p>A dataframe containing phenotypes for all individuals specified to have records.</p>
</td></tr>
<tr><td><code>allEffects</code></td>
<td>
<p>(optional) A dataframe with all direct and indirect genetic and environmental effects.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+micro_sim">micro_sim</a></code>, <code><a href="#topic+genome_sim">genome_sim</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## make up a pedigree
id &lt;- c("a1", "a2", "a3", "a4", "a5", "a6", "a7", "a8", "a9")
dam &lt;- c(NA, NA, NA, "a1", "a1", "a1", "a4", "a4", "a4")
sire &lt;- c(NA, NA, NA, "a2", "a2", "a2", "a5", "a6", "a6")
pedigree &lt;- as.data.frame(cbind(id, sire, dam))

traits &lt;- 2
## no correlations
randomA &lt;- diag(4)
randomE &lt;- diag(4)
parentalA &lt;- c("d", "d", "m", "m")
parentalE &lt;- c("d", "d", "m", "m")

## generate phenoypic data based on this architecture
phen_sim(
  pedigree = pedigree, traits = 2, randomA = randomA, randomE = randomE,
  parentalA = parentalA, parentalE = parentalE
)

## let's do it again but see how the phenotypes were composed
phen_sim(
  pedigree = pedigree, traits = 2, randomA = randomA, randomE = randomE,
  parentalA = parentalA, parentalE = parentalE, returnAllEffects = TRUE
)

</code></pre>

<hr>
<h2 id='plot.ped_stats'>Plot output from ped_stats</h2><span id='topic+plot.ped_stats'></span>

<h3>Description</h3>

<p>Generates a manageable summary of pedigree-wide statistics
reported by ped_stats, either for a single pedigree or
for a comparison between pedigrees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ped_stats'
plot(x, lowMem = FALSE, grContrast = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.ped_stats_+3A_x">x</code></td>
<td>
<p>An object of class ped_stats generated by <code>ped_stats</code></p>
</td></tr>
<tr><td><code id="plot.ped_stats_+3A_lowmem">lowMem</code></td>
<td>
<p>If TRUE, then stats based on calculation of A are not performed.</p>
</td></tr>
<tr><td><code id="plot.ped_stats_+3A_grcontrast">grContrast</code></td>
<td>
<p>If TRUE, then uglier shades of red and blue are used to denote male and female statistics in graphical reports, but these colours provide better contrast in greyscale.</p>
</td></tr>
<tr><td><code id="plot.ped_stats_+3A_...">...</code></td>
<td>
<p>extra arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a table of numbers of records, maternities,
paternities, pairwise sibship relationships, numbers
of different classes of grand-parental relationships,
pedigree depth, number of founders, mean sibship sizes,
simple statistics of numbers of inbred and non-inbred
individuals, and proportions of pairwise relationship
coefficients equal to or greater than several
thresholds.
</p>

<hr>
<h2 id='summary.ped_stats'>Post-processes output from ped_stats</h2><span id='topic+summary.ped_stats'></span>

<h3>Description</h3>

<p>Generates a manageable summary of pedigree-wide statistics
reported by ped_stats, either for a single pedigree or
for a comparison between pedigrees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ped_stats'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.ped_stats_+3A_object">object</code></td>
<td>
<p>An object of class ped_stats generated by <code>ped_stats</code></p>
</td></tr>
<tr><td><code id="summary.ped_stats_+3A_...">...</code></td>
<td>
<p>extra arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a table of numbers of records, maternities,
paternities, pairwise sibship relationships, numbers
of different classes of grand-parental relationships,
pedigree depth, number of founders, mean sibship sizes,
simple statistics of numbers of inbred and non-inbred
individuals, and proportions of pairwise relationship
coefficients equal to or greater than several
thresholds.
</p>

<hr>
<h2 id='WarblerG'>Seychelles Warbler Genotypes</h2><span id='topic+WarblerG'></span>

<h3>Description</h3>

<p>Genotype data collected by David Richardson from Cousin Island in 1999.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WarblerG
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 307 rows and 29 columns.
</p>


<h3>References</h3>

<p>Richardson <em>et.al.</em> (2001) Molecular Ecology 10 2263-2273
Hadfield J.D. <em>et al</em> (2006) Molecular Ecology 15 3715-31
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
