<!DOCTYPE html><html><head><title>Help for package Peptides</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Peptides}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aaComp'><p>Compute the amino acid composition of a protein sequence</p></a></li>
<li><a href='#AAdata'><p>Properties, scales and indices for the 20 naturally occurring amino acids from various sources</p></a></li>
<li><a href='#aaDescriptors'><p>Compute 66 descriptors for each amino acid of a protein sequence.</p></a></li>
<li><a href='#aaList'><p>Return a vector with the 20 standard amino acids in upper case</p></a></li>
<li><a href='#aaSMILES'><p>Create Smiles String from aminoacid sequences</p></a></li>
<li><a href='#aIndex'><p>Compute the aliphatic index of a protein sequence</p></a></li>
<li><a href='#autoCorrelation'><p>Compute the auto-correlation index of a protein sequence</p></a></li>
<li><a href='#autoCovariance'><p>Compute the auto-covariance index of a protein sequence</p></a></li>
<li><a href='#blosumIndices'><p>Compute the BLOSUM62 derived indices of a protein sequence</p></a></li>
<li><a href='#boman'><p>Compute the Boman (Potential Protein Interaction) index</p></a></li>
<li><a href='#charge'><p>Compute the theoretical net charge of a protein sequence</p></a></li>
<li><a href='#crossCovariance'><p>Compute the cross-covariance index of a protein sequence</p></a></li>
<li><a href='#crucianiProperties'><p>Compute the Cruciani properties of a protein sequence</p></a></li>
<li><a href='#fasgaiVectors'><p>Compute the FASGAI vectors of a protein sequence</p></a></li>
<li><a href='#hmoment'><p>Compute the hydrophobic moment of a protein sequence</p></a></li>
<li><a href='#hydrophobicity'><p>Compute the hydrophobicity index of a protein sequence</p></a></li>
<li><a href='#instaIndex'><p>Compute the instability index of a protein sequence</p></a></li>
<li><a href='#kideraFactors'><p>Compute the Kidera factors of a protein sequence</p></a></li>
<li><a href='#lengthpep'><p>Compute the amino acid length of a protein sequence</p></a></li>
<li><a href='#massShift'><p>Calculate the mass difference of modified peptides.</p></a></li>
<li><a href='#membpos'><p>Compute theoretically the class of a protein sequence</p></a></li>
<li><a href='#mswhimScores'><p>Compute the MS-WHIM scores of a protein sequence</p></a></li>
<li><a href='#mw'><p>Compute the molecular weight of a protein sequence</p></a></li>
<li><a href='#mz'><p>Calculate the m/z for peptides.</p></a></li>
<li><a href='#pepdata'><p>Physicochemical properties and indices from 100 amino acid sequences</p></a></li>
<li><a href='#pI'><p>Compute the isoelectic point (pI) of a protein sequence</p></a></li>
<li><a href='#plotXVG'><p>Plot time series from GROMACS XVG files</p></a></li>
<li><a href='#protFP'><p>Compute the protFP descriptors of a protein sequence</p></a></li>
<li><a href='#readXVG'><p>Read output data from a XVG format file.</p></a></li>
<li><a href='#stScales'><p>Compute the ST-scales of a protein sequence</p></a></li>
<li><a href='#tScales'><p>Compute the T-scales of a protein sequence</p></a></li>
<li><a href='#vhseScales'><p>Compute the VHSE-scales of a protein sequence</p></a></li>
<li><a href='#zScales'><p>Compute the Z-scales of a protein sequence</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>2.4.6</td>
</tr>
<tr>
<td>Title:</td>
<td>Calculate Indices and Theoretical Physicochemical Properties of
Protein Sequences</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/dosorio/Peptides/">https://github.com/dosorio/Peptides/</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 2.1.0)</td>
</tr>
<tr>
<td>Description:</td>
<td>Includes functions to calculate several physicochemical properties and indices for amino-acid sequences as well as to read and plot 'XVG' output files from the 'GROMACS' molecular dynamics package.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-13 17:30:20 UTC; osoriod</td>
</tr>
<tr>
<td>Author:</td>
<td>Daniel Osorio <a href="https://orcid.org/0000-0003-4424-8422"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Paola Rondon-Villarreal
    <a href="https://orcid.org/0000-0001-8209-3885"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, ths],
  Rodrigo Torres <a href="https://orcid.org/0000-0003-1113-3020"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ths],
  J. Sebastian Paez <a href="https://orcid.org/0000-0002-0065-1474"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Luis Pedro Coelho <a href="https://orcid.org/0000-0002-9280-7885"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Rich√®l J.C. Bilderbeek
    <a href="https://orcid.org/0000-0003-1107-7049"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Florian C. Sigloch
    <a href="https://orcid.org/0000-0001-8130-6885"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Daniel Osorio &lt;daniecos@uio.no&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-13 18:00:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='aaComp'>Compute the amino acid composition of a protein sequence</h2><span id='topic+aaComp'></span>

<h3>Description</h3>

<p>This function calculates the amount of amino acids of a particular class and classified as: Tiny, Small, Aliphatic, Aromatic, Non-polar, Polar, Charged, Basic and Acidic based on their size and R-groups using same function implemented in EMBOSS 'pepstat'. 
The output is a matrix with the number and percentage of amino acids of a particular class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aaComp(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aaComp_+3A_seq">seq</code></td>
<td>
<p>An amino-acid sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Amino acids are zwitterionic molecules with an amine and a carboxyl group present in their structure. 
Some amino acids possess side chains with specific properties that allow grouping them in different ways. 
The <code>aaComp</code> function classifies amino acids based on their size, side chains, hydrophobicity, charge and their response to pH 7.
</p>


<h3>Value</h3>

<p>The output is a matrix with the number and percentage of amino acids of a particular class:
</p>

<ul>
<li><p> Tiny (A + C + G + S + T)
</p>
</li>
<li><p> Small (A + B + C + D + G + N + P + S + T + V)
</p>
</li>
<li><p> Aliphatic (A + I + L + V)
</p>
</li>
<li><p> Aromatic (F + H + W + Y)
</p>
</li>
<li><p> Non-polar (A + C + F + G + I + L + M + P + V + W + Y)
</p>
</li>
<li><p> Polar (D + E + H + K + N + Q + R + S + T + Z)
</p>
</li>
<li><p> Charged (B + D + E + H + K + R + Z)
</p>
</li>
<li><p> Basic (H + K + R)
</p>
</li>
<li><p> Acidic (B + D + E + Z)
</p>
</li></ul>



<h3>Note</h3>

<p>This function was originally written by Alan Bleasby (ajb@ebi.ac.uk) for the EMBOSS package. 
Further information: http://emboss.sourceforge.net/apps/cvs/emboss/apps/pepstats.html
</p>


<h3>References</h3>

<p>Rice, Peter, Ian Longden, and Alan Bleasby. &quot;EMBOSS: the European molecular biology open software suite.&quot; Trends in genetics 16.6 (2000): 276-277.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COMPARED TO PEPSTATS
# http://emboss.bioinformatics.nl/cgi-bin/emboss/pepstats
# Property      Residues              Number  Mole%
# Tiny          (A+C+G+S+T)             4   19.048
# Small         (A+B+C+D+G+N+P+S+T+V)   4   19.048
# Aliphatic     (A+I+L+V)               5   23.810
# Aromatic      (F+H+W+Y)               5   23.810
# Non-polar     (A+C+F+G+I+L+M+P+V+W+Y) 11  52.381
# Polar         (D+E+H+K+N+Q+R+S+T+Z)   9   42.857
# Charged       (B+D+E+H+K+R+Z)         8   38.095
# Basic         (H+K+R)                 8   38.095
# Acidic        (B+D+E+Z)               0   00.000

## AA composition of PDB: 1D9J Cecropin Peptide
aaComp(seq= "KWKLFKKIGIGKFLHSAKKFX")

## Output
#           Number  Mole %
# Tiny           4 19.048
# Small          4 19.048
# Aliphatic      5 23.810
# Aromatic       5 23.810
# NonPolar      11 52.381
# Polar          9 42.857
# Charged        8 38.095
# Basic          8 38.095
# Acidic         0  0.000
</code></pre>

<hr>
<h2 id='AAdata'>Properties, scales and indices for the 20 naturally occurring amino acids from various sources</h2><span id='topic+AAdata'></span>

<h3>Description</h3>

<p>A list with a collection of properties, scales and indices for the 20 naturally occurring amino acids from various sources.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(AAdata)
</code></pre>


<h3>Format</h3>

<p>A list as follows: </p>

<ul>
<li><p>Hydrophobicity The hydrophobicity is an important stabilization force in protein folding; this force changes depending on the solvent in which the protein is found. </p>

<ul>
<li><p>Aboderin: Aboderin, A. A. (1971). An empirical hydrophobicity scale for alpha-amino-acids and some of its applications. International Journal of Biochemistry, 2(11), 537-544.
</p>
</li>
<li><p>AbrahamLeo: Abraham D.J., Leo A.J. Hydrophobicity (delta G1/2 cal). Proteins: Structure, Function and Genetics 2:130-152(1987).
</p>
</li>
<li><p>Argos: Argos, P., Rao, J. K., &amp; Hargrave, P. A. (1982). Structural Prediction of Membrane-Bound Proteins. European Journal of Biochemistry, 128(2-3), 565-575.
</p>
</li>
<li><p>BlackMould: Black S.D., Mould D.R. Hydrophobicity of physiological L-alpha amino acids. Anal. Biochem. 193:72-82(1991).
</p>
</li>
<li><p>BullBreese: Bull H.B., Breese K. Hydrophobicity (free energy of transfer to surface in kcal/mole). Arch. Biochem. Biophys. 161:665-670(1974).
</p>
</li>
<li><p>Casari: Casari, G., &amp; Sippl, M. J. (1992). Structure-derived hydrophobic potential: hydrophobic potential derived from X-ray structures of globular proteins is able to identify native folds. Journal of molecular biology, 224(3), 725-732.
</p>
</li>
<li><p>Chothia: Chothia, C. (1976). The nature of the accessible and buried surfaces in proteins. Journal of molecular biology, 105(1), 1-12.
</p>
</li>
<li><p>Cid: Cid, H., Bunster, M., Canales, M., &amp; Gazitua, F. (1992). Hydrophobicity and structural classes in proteins. Protein engineering, 5(5), 373-375.
</p>
</li>
<li><p>Cowan3.4: Cowan R., Whittaker R.G. Hydrophobicity indices at pH 3.4 determined by HPLC. Peptide Research 3:75-80(1990).
</p>
</li>
<li><p>Cowan7.5: Cowan R., Whittaker R.G. Hydrophobicity indices at pH 7.5 determined by HPLC. Peptide Research 3:75-80(1990).
</p>
</li>
<li><p>Eisenberg: Eisenberg D., Schwarz E., Komarony M., Wall R. Normalized consensus hydrophobicity scale. J. Mol. Biol. 179:125-142(1984).
</p>
</li>
<li><p>Engelman: Engelman, D. M., Steitz, T. A., &amp; Goldman, A. (1986). Identifying nonpolar transbilayer helices in amino acid sequences of membrane proteins. Annual review of biophysics and biophysical chemistry, 15(1), 321-353.
</p>
</li>
<li><p>Fasman: Fasman, G. D. (Ed.). (1989). Prediction of protein structure and the principles of protein conformation. Springer.
</p>
</li>
<li><p>Fauchere: Fauchere J.-L., Pliska V.E. Hydrophobicity scale (pi-r). Eur. J. Med. Chem. 18:369-375(1983).
</p>
</li>
<li><p>Goldsack: Goldsack, D. E., &amp; Chalifoux, R. C. (1973). Contribution of the free energy of mixing of hydrophobic side chains to the stability of the tertiary structure of proteins. Journal of theoretical biology, 39(3), 645-651.
</p>
</li>
<li><p>Guy: Guy H.R. Hydrophobicity scale based on free energy of transfer (kcal/mole). Biophys J. 47:61-70(1985).
</p>
</li>
<li><p>HoppWoods: Hopp T.P., Woods K.R. Hydrophilicity. Proc. Natl. Acad. Sci. U.S.A. 78:3824-3828(1981).
</p>
</li>
<li><p>Janin: Janin J. Free energy of transfer from inside to outside of a globular protein. Nature 277:491-492(1979).
</p>
</li>
<li><p>Jones: Jones, D. D. (1975). Amino acid properties and side-chain orientation in proteins: a cross correlation approach. Journal of theoretical biology, 50(1), 167-183.
</p>
</li>
<li><p>Juretic: Juretic, D., Lucic, B., Zucic, D., &amp; Trinajstic, N. (1998). Protein transmembrane structure: recognition and prediction by using hydrophobicity scales through preference functions. Theoretical and computational chemistry, 5, 405-445.
</p>
</li>
<li><p>Kidera: Kidera, A., Konishi, Y., Oka, M., Ooi, T., &amp; Scheraga, H. A. (1985). Statistical analysis of the physical properties of the 20 naturally occurring amino acids. Journal of Protein Chemistry, 4(1), 23-55.
</p>
</li>
<li><p>Kuhn: Kuhn, L. A., Swanson, C. A., Pique, M. E., Tainer, J. A., &amp; Getzoff, E. D. (1995). Atomic and residue hydrophilicity in the context of folded protein structures. Proteins: Structure, Function, and Bioinformatics, 23(4), 536-547.
</p>
</li>
<li><p>KyteDoolittle: Kyte J., Doolittle R.F. Hydropathicity. J. Mol. Biol. 157:105-132(1982).
</p>
</li>
<li><p>Levitt: Levitt, M. (1976). A simplified representation of protein conformations for rapid simulation of protein folding. Journal of molecular biology, 104(1), 59-107.
</p>
</li>
<li><p>Manavalan: Manavalan P., Ponnuswamy Average surrounding hydrophobicity. P.K. Nature 275:673-674(1978).
</p>
</li>
<li><p>Miyazawa: Miyazawa S., Jernigen R.L. Hydrophobicity scale (contact energy derived from 3D data). Macromolecules 18:534-552(1985).
</p>
</li>
<li><p>Parker: Parker J.M.R., Guo D., Hodges R.S. Hydrophilicity scale derived from HPLC peptide retention times. Biochemistry 25:5425-5431(1986).
</p>
</li>
<li><p>Ponnuswamy: Ponnuswamy, P. K. (1993). Hydrophobic charactesristics of folded proteins. Progress in biophysics and molecular biology, 59(1), 57-103.
</p>
</li>
<li><p>Prabhakaran: Prabhakaran, M. (1990). The distribution of physical, chemical and conformational properties in signal and nascent peptides. Biochem. J, 269, 691-696.
</p>
</li>
<li><p>Rao: Rao M.J.K., Argos P. Membrane buried helix parameter. Biochim. Biophys. Acta 869:197-214(1986).
</p>
</li>
<li><p>Rose: Rose G.D., Geselowitz A.R., Lesser G.J., Lee R.H., Zehfus M.H. Mean fractional area loss (f) [average area buried/standard state area]. Science 229:834-838(1985)
</p>
</li>
<li><p>Roseman: Roseman M.A. Hydrophobicity scale (pi-r). J. Mol. Biol. 200:513-522(1988).
</p>
</li>
<li><p>Sweet: Sweet R.M., Eisenberg D. Optimized matching hydrophobicity (OMH). J. Mol. Biol. 171:479-488(1983).
</p>
</li>
<li><p>Tanford: Tanford C. Hydrophobicity scale (Contribution of hydrophobic interactions to the stability of the globular conformation of proteins). J. Am. Chem. Soc. 84:4240-4274(1962).
</p>
</li>
<li><p>Welling: Welling G.W., Weijer W.J., Van der Zee R., Welling-Wester S. Antigenicity value X 10. FEBS Lett. 188:215-218(1985).
</p>
</li>
<li><p>Wilson: Wilson K.J., Honegger A., Stotzel R.P., Hughes G.J. Hydrophobic constants derived from HPLC peptide retention times. Biochem. J. 199:31-41(1981).
</p>
</li>
<li><p>Wolfenden: Wolfenden R.V., Andersson L., Cullis P.M., Southgate C.C.F. Hydration potential (kcal/mole) at 25C. Biochemistry 20:849-855(1981).
</p>
</li>
<li><p>Zimmerman: Zimmerman, J. M., Eliezer, N., &amp; Simha, R. (1968). The characterization of amino acid sequences in proteins by statistical methods. Journal of theoretical biology, 21(2), 170-201.
</p>
</li>
<li><p>interfaceScale_pH8 White, Stephen (2006-06-29). &quot;Experimentally Determined Hydrophobicity Scales&quot;. University of California, Irvine. Retrieved 2017-05-25
</p>
</li>
<li><p>interfaceScale_pH2 White, Stephen (2006-06-29). &quot;Experimentally Determined Hydrophobicity Scales&quot;. University of California, Irvine. Retrieved 2017-05-25
</p>
</li>
<li><p>octanolScale_pH8 White, Stephen (2006-06-29). &quot;Experimentally Determined Hydrophobicity Scales&quot;. University of California, Irvine. Retrieved 2017-05-25
</p>
</li>
<li><p>octanolScale_pH2 White, Stephen (2006-06-29). &quot;Experimentally Determined Hydrophobicity Scales&quot;. University of California, Irvine. Retrieved 2017-05-25
</p>
</li>
<li><p>oiScale_pH8 White, Stephen (2006-06-29). &quot;Experimentally Determined Hydrophobicity Scales&quot;. University of California, Irvine. Retrieved 2017-05-25
</p>
</li>
<li><p>oiScale_pH2 White, Stephen (2006-06-29). &quot;Experimentally Determined Hydrophobicity Scales&quot;. University of California, Irvine. Retrieved 2017-05-25
</p>
</li></ul>

</li>
<li><p>crucianiProperties: The three Cruciani et. al (2004) properties, are the scaled principal component scores that summarize a broad set of descriptors calculated based on the interaction of each amino acid residue with several chemical groups (or &quot;probes&quot;), such as charged ions, methyl, hydroxyl groups, and so forth. </p>

<ul>
<li><p>PP1: Polarity, 
</p>
</li>
<li><p>PP2: Hydrophobicity, 
</p>
</li>
<li><p>PP3: H-bonding
</p>
</li></ul>

</li>
<li><p>kideraFactors: The Kidera Factors were originally derived by applying multivariate analysis to 188 physical properties of the 20 amino acids and using dimension reduction techniques. A 10-dimensional vector of orthogonal factors was then obtained for each amino acid.The first four factors are essentially pure physical properties; the remaining six factors are superpositions of several physical properties, and are labelled for convenience by the name of the most heavily weighted component </p>

<ul>
<li><p>helix.bend.pref: Helix/bend preference
</p>
</li>
<li><p>side.chain.size: Side-chain size
</p>
</li>
<li><p>extended.str.pref: Extended structure preference
</p>
</li>
<li><p>hydrophobicity: Hydrophobicity
</p>
</li>
<li><p>double.bend.pref: Double-bend preference
</p>
</li>
<li><p>partial.spec.vol: Partial specific volume
</p>
</li>
<li><p>flat.ext.pref: Flat extended preference
</p>
</li>
<li><p>occurrence.alpha.reg: Occurrence in alpha region
</p>
</li>
<li><p>pK.C: pK-C
</p>
</li>
<li><p>surrounding.hydrop: Surrounding hydrophobicity
</p>
</li></ul>

</li>
<li><p>pK</p>

<ul>
<li><p>Bjellqvist: Bjellqvist, B., Hughes, G.J., Pasquali, Ch., Paquet, N., Ravier, F., Sanchez, J.Ch., Frutige,r S., Hochstrasser D. (1993) The focusing positions of polypeptides in immobilized pH gradients can be predicted from their amino acid sequences. Electrophoresis, 14:1023-1031.
</p>
</li>
<li><p>Dawson: Dawson, R. M. C.; Elliot, D. C.; Elliot, W. H.; Jones, K. M. Data for biochemical research. Oxford University Press, 1989; p. 592.
</p>
</li>
<li><p>EMBOSS: EMBOSS data are from http://emboss.sourceforge.net/apps/release/5.0/emboss/apps/iep.html.
</p>
</li>
<li><p>Lehninger: Nelson, D. L.; Cox, M. M. Lehninger Principles of Biochemistry, Fourth Edition; W. H. Freeman, 2004; p. 1100.
</p>
</li>
<li><p>Murray: Murray, R.K., Granner, D.K., Rodwell, V.W. (2006) Harper's illustrated Biochemistry. 27th edition. Published by The McGraw-Hill Companies.
</p>
</li>
<li><p>Rodwell: Rodwell, J. Heterogeneity of component bands in isoelectric focusing patterns. Analytical Biochemistry, 1982, 119 (2), 440-449.
</p>
</li>
<li><p>Sillero: Sillero, A., Maldonado, A. (2006) Isoelectric point determination of proteins and other macromolecules: oscillating method. Comput Biol Med., 36:157-166.
</p>
</li>
<li><p>Solomon: Solomon, T.W.G. (1998) Fundamentals of Organic Chemistry, 5th edition. Published by Wiley.
</p>
</li>
<li><p>Stryer: Stryer L. (1999) Biochemia. czwarta edycja. Wydawnictwo Naukowe PWN.
</p>
</li></ul>

</li>
<li><p>zScales The five Sandberg et al. (1998) Z-scales describe each amino acid with numerical values, descriptors, which represent the physicochemical properties of the amino acids including NMR data and thin-layer chromatography (TLC) data.</p>

<ul>
<li><p>Z1: Lipophilicity
</p>
</li>
<li><p>Z2: Steric properties (Steric bulk/Polarizability)
</p>
</li>
<li><p>Z3: Electronic properties (Polarity / Charge)
</p>
</li>
<li><p>Z4: Related to  electronegativity, heat of formation, electrophilicity and hardness.
</p>
</li>
<li><p>Z5: Related to electronegativity, heat of formation, electrophilicity and hardness.
</p>
</li></ul>

</li>
<li><p>FASGAI Factor Analysis Scale of Generalized Amino Acid Information (FASGAI) proposed by Liang and Li (2007), is a set of amino acid descriptors, that reflects hydrophobicity, alpha and turn propensities, bulky properties, compositional characteristics, local flexibility, and electronic properties, was derived from multi-dimensional properties of 20 naturally occurring amino acids. </p>

<ul>
<li><p>F1: Hydrophobicity index
</p>
</li>
<li><p>F2: Alpha and turn propensities
</p>
</li>
<li><p>F3: Bulky properties
</p>
</li>
<li><p>F4: Compositional characteristic index
</p>
</li>
<li><p>F5: Local flexibility
</p>
</li>
<li><p>F6: Electronic properties
</p>
</li></ul>

</li>
<li><p>VHSE The principal components score Vectors of Hydrophobic, Steric, and Electronic properties, is derived from principal components analysis (PCA) on independent families of 18 hydrophobic properties, 17 steric properties, and 15 electronic properties, respectively, which are included in total 50 physicochemical variables of 20 coded amino acids. </p>

<ul>
<li><p>VHSE1 and VHSE2: Hydrophobic properties
</p>
</li>
<li><p>VHSE3 and VHSE4: Steric properties
</p>
</li>
<li><p>VHSE5 to VHSE8: Electronic properties
</p>
</li></ul>

</li></ul>



<h3>Source</h3>


<ul>
<li><p>Hydrophobicity</p>

<ul>
<li><p> ExPASy-Protscale (http://web.expasy.org/protscale/)
</p>
</li>
<li><p> AAIndex Database (http://www.genome.jp/aaindex/)
</p>
</li></ul>

</li>
<li><p>pK</p>

<ul>
<li><p> Kiraga, J. (2008) Analysis and computer simulations of variability of isoelectric point of proteins in the proteomes. PhD thesis, University of Wroclaw, Poland.
</p>
</li></ul>

</li></ul>



<h3>References</h3>


<ul>
<li><p>Hydrophobicity</p>

<ul>
<li><p> Nakai, K., Kidera, A., and Kanehisa, M.; Cluster analysis of amino acid indices for prediction of protein structure and function. Protein Eng. 2, 93-100 (1988).
</p>
</li>
<li><p> Tomii, K. and Kanehisa, M.; Analysis of amino acid indices and mutation matrices for sequence comparison and structure prediction of proteins. Protein Eng. 9, 27-36 (1996). 
</p>
</li>
<li><p> Kawashima, S., Ogata, H., and Kanehisa, M.; AAindex: amino acid index database. Nucleic Acids Res. 27, 368-369 (1999).
</p>
</li>
<li><p> Kawashima, S. and Kanehisa, M.; AAindex: amino acid index database. Nucleic Acids Res. 28, 374 (2000).
</p>
</li>
<li><p> Kawashima, S., Pokarowski, P., Pokarowska, M., Kolinski, A., Katayama, T., and Kanehisa, M.; AAindex: amino acid index database, progress report 2008. Nucleic Acids Res. 36, D202-D205 (2008). 
</p>
</li></ul>

</li>
<li><p>crucianiProperties: </p>

<ul>
<li><p> Cruciani, G., Baroni, M., Carosati, E., Clementi, M., Valigi, R., and Clementi, S. (2004) Peptide studies by means of principal properties of amino acids derived from MIF descriptors. J. Chemom. 18, 146-155.
</p>
</li></ul>

</li>
<li><p>kideraFactors: </p>

<ul>
<li><p> Kidera, A., Konishi, Y., Oka, M., Ooi, T., &amp; Scheraga, H. A. (1985). Statistical analysis of the physical properties of the 20 naturally occurring amino acids. Journal of Protein Chemistry, 4(1), 23-55.
</p>
</li></ul>

</li>
<li><p>pK: </p>

<ul>
<li><p> Aronson, J. N. The Henderson-Hasselbalch equation revisited. Biochemical Education, 1983, 11 (2), 68.
</p>
</li>
<li><p> Moore, D. S.. Amino acid and peptide net charges: A simple calculational procedure. Biochemical Education, 1986, 13 (1), 10-12.
</p>
</li>
<li><p> Goloborodko, A.A.; Levitsky, L.I.; Ivanov, M.V.; and Gorshkov, M.V. (2013) &quot;Pyteomics - a Python Framework for Exploratory Data Analysis and Rapid Software Prototyping in Proteomics&quot;, Journal of The American Society for Mass Spectrometry, 24(2), 301-304.
</p>
</li>
<li><p> Kiraga, J. (2008) Analysis and computer simulations of variability of isoelectric point of proteins in the proteomes. PhD thesis, University of Wroclaw, Poland.
</p>
</li></ul>

</li>
<li><p>zScales </p>

<ul>
<li><p> Sandberg M, Eriksson L, Jonsson J, Sjostrom M, Wold S: New chemical descriptors relevant for the design of biologically active peptides. A multivariate characterization of 87 amino acids. J Med Chem 1998, 41:2481-2491.
</p>
</li></ul>

</li>
<li><p>FASGAI </p>

<ul>
<li><p> Liang, G., &amp; Li, Z. (2007). Factor analysis scale of generalized amino acid information as the source of a new set of descriptors for elucidating the structure and activity relationships of cationic antimicrobial peptides. Molecular Informatics, 26(6), 754-763.
</p>
</li></ul>

</li>
<li><p>VHSE </p>

<ul>
<li><p> Mei, H. U., Liao, Z. H., Zhou, Y., &amp; Li, S. Z. (2005). A new set of amino acid descriptors and its application in peptide QSARs. Peptide Science, 80(6), 775-786.
</p>
</li></ul>

</li></ul>


<hr>
<h2 id='aaDescriptors'>Compute 66 descriptors for each amino acid of a protein sequence.</h2><span id='topic+aaDescriptors'></span>

<h3>Description</h3>

<p>The function return 66 amino acid descriptors for the 20 natural amino acids. Available descriptors are: </p>

<ul>
<li><p>crucianiProperties: Cruciani, G., Baroni, M., Carosati, E., Clementi, M., Valigi, R., and Clementi, S. (2004) Peptide studies by means of principal properties of amino acids derived from MIF descriptors. J. Chemom. 18, 146-155.,
</p>
</li>
<li><p>kideraFactors: Kidera, A., Konishi, Y., Oka, M., Ooi, T., &amp; Scheraga, H. A. (1985). Statistical analysis of the physical properties of the 20 naturally occurring amino acids. Journal of Protein Chemistry, 4(1), 23-55.,
</p>
</li>
<li><p>zScales: Sandberg M, Eriksson L, Jonsson J, Sjostrom M, Wold S: New chemical descriptors relevant for the design of biologically active peptides. A multivariate characterization of 87 amino acids. J Med Chem 1998, 41:2481-2491.,
</p>
</li>
<li><p>FASGAI: Liang, G., &amp; Li, Z. (2007). Factor analysis scale of generalized amino acid information as the source of a new set of descriptors for elucidating the structure and activity relationships of cationic antimicrobial peptides. Molecular Informatics, 26(6), 754-763.,
</p>
</li>
<li><p>tScales: Tian F, Zhou P, Li Z: T-scale as a novel vector of topological descriptors for amino acids and its application in QSARs of peptides. J Mol Struct. 2007, 830: 106-115. 10.1016/j.molstruc.2006.07.004.,
</p>
</li>
<li><p>VHSE: VHSE-scales (principal components score Vectors of Hydrophobic, Steric, and Electronic properties), is derived from principal components analysis (PCA) on independent families of 18 hydrophobic properties, 17 steric properties, and 15 electronic properties, respectively, which are included in total 50 physicochemical variables of 20 coded amino acids.,
</p>
</li>
<li><p>protFP: van Westen, G. J., Swier, R. F., Wegner, J. K., IJzerman, A. P., van Vlijmen, H. W., &amp; Bender, A. (2013). Benchmarking of protein descriptor sets in proteochemometric modeling (part 1): comparative study of 13 amino acid descriptor sets. Journal of cheminformatics, 5(1), 41.,
</p>
</li>
<li><p>stScales: Yang, L., Shu, M., Ma, K., Mei, H., Jiang, Y., &amp; Li, Z. (2010). ST-scale as a novel amino acid descriptor and its application in QSAM of peptides and analogues. Amino acids, 38(3), 805-816.,
</p>
</li>
<li><p>BLOSUM: Georgiev, A. G. (2009). Interpretable numerical descriptors of amino acid space. Journal of Computational Biology, 16(5), 703-723.,
</p>
</li>
<li><p>MSWHIM: Zaliani, A., &amp; Gancia, E. (1999). MS-WHIM scores for amino acids: a new 3D-description for peptide QSAR and QSPR studies. Journal of chemical information and computer sciences, 39(3), 525-533.
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>aaDescriptors(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aaDescriptors_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence. If multiple sequences are given all of them must have the same length (gap symbols are allowed.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a matrix with 66 amino acid descriptors for each aminoacid in a protein sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>aaDescriptors(seq = "KLKLLLLLKLK")
</code></pre>

<hr>
<h2 id='aaList'>Return a vector with the 20 standard amino acids in upper case</h2><span id='topic+aaList'></span>

<h3>Description</h3>

<p>This function returns a vector with the 20 standard amino acids in upper case.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aaList()
</code></pre>


<h3>Value</h3>

<p>A character vector with the 20 standard amino acids in upper case.
</p>


<h3>Author(s)</h3>

<p>Richel Bilderbeek &lt;richel@richelbilderbeek.nl&gt;
</p>


<h3>References</h3>

<p>Lu, Y., &amp; Freeland, S. (2006). On the evolution of the standard amino-acid alphabet. Genome biology, 7(1), 102.
</p>

<hr>
<h2 id='aaSMILES'>Create Smiles String from aminoacid sequences</h2><span id='topic+aaSMILES'></span>

<h3>Description</h3>

<p>This function converts peptides with aminoacid one-letter abbreviations into
smiles strings to represent the structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aaSMILES(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aaSMILES_+3A_seq">seq</code></td>
<td>
<p>character vector with one-letter aminoacid codes</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output can be stored in a .smi file and converted using openbabel to
drawings of the peptides.
</p>


<h3>Value</h3>

<p>character vector with smiles strings
</p>


<h3>Examples</h3>

<pre><code class='language-R'>aaSMILES("AA")
# [1] "N[C@]([H])(C)C(=O)N[C@]([H])(C)C(=O)O"
aaSMILES(c("AA", "GG"))
# [1] "N[C@]([H])(C)C(=O)N[C@]([H])(C)C(=O)O" "NCC(=O)NCC(=O)O"
</code></pre>

<hr>
<h2 id='aIndex'>Compute the aliphatic index of a protein sequence</h2><span id='topic+aIndex'></span>

<h3>Description</h3>

<p>This function calculates the Ikai (1980) aliphatic index of a protein. The <code>aindex</code> is defined as the relative volume occupied by aliphatic side chains (Alanine, Valine, Isoleucine, and Leucine). It may be regarded as a positive factor for the increase of thermostability of globular proteins.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aIndex(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aIndex_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Aliphatic amino acids (A, I, L and V) are responsible for the thermal stability of proteins. The aliphatic index was proposed by Ikai (1980) and evaluates the thermostability of proteins based on the percentage of each of the aliphatic amino acids that build up proteins.
</p>


<h3>Value</h3>

<p>The computed aliphatic index for a given amino-acids sequence
</p>


<h3>References</h3>

<p>Ikai (1980). Thermostability and aliphatic index of globular proteins. Journal of Biochemistry, 88(6), 1895-1898.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COMPARED TO ExPASy ALIPHATIC INDEX
# http://web.expasy.org/protparam/
# SEQUENCE: SDKEVDEVDAALSDLEITLE
# Aliphatic index: 117.00

aIndex(seq = "SDKEVDEVDAALSDLEITLE")
# [1] 117
</code></pre>

<hr>
<h2 id='autoCorrelation'>Compute the auto-correlation index of a protein sequence</h2><span id='topic+autoCorrelation'></span>

<h3>Description</h3>

<p>This function computes the Cruciani et al (2004) auto-correlation index. The <code>autoCorrelation</code> index is calculated for a lag 'd' using a descriptor 'f' (centred) over a sequence of length 'L'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>autoCorrelation(sequence, lag, property, center = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="autoCorrelation_+3A_sequence">sequence</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
<tr><td><code id="autoCorrelation_+3A_lag">lag</code></td>
<td>
<p>A value for a lag, the max value is equal to the length of shortest peptide minus one.</p>
</td></tr>
<tr><td><code id="autoCorrelation_+3A_property">property</code></td>
<td>
<p>A property to use as value to be correlated.</p>
</td></tr>
<tr><td><code id="autoCorrelation_+3A_center">center</code></td>
<td>
<p>A logical value <code>TRUE</code> or <code>FALSE</code> if the property must be centered.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed auto-correlation index for a given amino-acids sequence
</p>


<h3>References</h3>

<p>Cruciani, G., Baroni, M., Carosati, E., Clementi, M., Valigi, R., and Clementi, S. (2004) Peptide studies by means of principal properties of amino acids derived from MIF descriptors. J. Chemom. 18, 146-155.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Loading a property to evaluate its autocorrelation
data(AAdata)

# Calculate the auto-correlation index for a lag=1
autoCorrelation(
  sequence = "SDKEVDEVDAALSDLEITLE",
  lag = 1,
  property = AAdata$Hydrophobicity$KyteDoolittle,
  center = TRUE
)
# [1] -0.3519908

# Calculate the auto-correlation index for a lag=5
autoCorrelation(
  sequence = "SDKEVDEVDAALSDLEITLE",
  lag = 5,
  property = AAdata$Hydrophobicity$KyteDoolittle,
  center = TRUE
)
# [1] 0.001133553
</code></pre>

<hr>
<h2 id='autoCovariance'>Compute the auto-covariance index of a protein sequence</h2><span id='topic+autoCovariance'></span>

<h3>Description</h3>

<p>This function computes the Cruciani et al (2004) auto-corvariance index. The <code>autoCovariance</code> index is calculated for a lag 'd' using a descriptor 'f' (centred) over a sequence of length 'L'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>autoCovariance(sequence, lag, property, center = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="autoCovariance_+3A_sequence">sequence</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
<tr><td><code id="autoCovariance_+3A_lag">lag</code></td>
<td>
<p>A value for a lag, the max value is equal to the length of the shortest peptide minus one.</p>
</td></tr>
<tr><td><code id="autoCovariance_+3A_property">property</code></td>
<td>
<p>A property to use as value to evaluate the covariance.</p>
</td></tr>
<tr><td><code id="autoCovariance_+3A_center">center</code></td>
<td>
<p>A logical value <code>TRUE</code> or <code>FALSE</code> if the property must be centered.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed auto-covariance index for a given amino-acids sequence
</p>


<h3>References</h3>

<p>Cruciani, G., Baroni, M., Carosati, E., Clementi, M., Valigi, R., and Clementi, S. (2004) Peptide studies by means of principal properties of amino acids derived from MIF descriptors. J. Chemom. 18, 146-155.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Loading a property to evaluate its autocorrelation
data(AAdata)

# Calculate the auto-covariance index for a lag=1
autoCovariance(
  sequence = "SDKEVDEVDAALSDLEITLE",
  lag = 1,
  property = AAdata$Hydrophobicity$KyteDoolittle,
  center = TRUE
)
# [1] -0.4140053

# Calculate the auto-covariance index for a lag=5
autoCovariance(
  sequence = "SDKEVDEVDAALSDLEITLE",
  lag = 5,
  property = AAdata$Hydrophobicity$KyteDoolittle,
  center = TRUE
)
# [1] 0.001000336
</code></pre>

<hr>
<h2 id='blosumIndices'>Compute the BLOSUM62 derived indices of a protein sequence</h2><span id='topic+blosumIndices'></span>

<h3>Description</h3>

<p>BLOSUM indices were derived of physicochemical properties that have been subjected to a VARIMAX analyses and an alignment matrix of the 20 natural AAs using the BLOSUM62 matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blosumIndices(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blosumIndices_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed average of BLOSUM indices of all the amino acids in the corresponding peptide sequence.
</p>


<h3>References</h3>

<p>Georgiev, A. G. (2009). Interpretable numerical descriptors of amino acid space. Journal of Computational Biology, 16(5), 703-723.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>blosumIndices(seq = "KLKLLLLLKLK")
# [[1]]
#   BLOSUM1    BLOSUM2    BLOSUM3    BLOSUM4    BLOSUM5    
# -0.4827273 -0.5618182 -0.8509091 -0.4172727  0.3172727  

#  BLOSUM6   BLOSUM7     BLOSUM8    BLOSUM9   BLOSUM10 
# 0.2527273  0.1463636  0.1427273 -0.2145455 -0.3218182 

</code></pre>

<hr>
<h2 id='boman'>Compute the Boman (Potential Protein Interaction) index</h2><span id='topic+boman'></span>

<h3>Description</h3>

<p>This function computes the potential protein interaction index proposed by Boman (2003) based in the amino acid sequence of a protein. 
The index is equal to the sum of the solubility values for all residues in a sequence, it might give an overall estimate of the potential of a peptide to bind to membranes or other proteins as receptors, to normalize it is divided by the number of residues. 
A protein have high binding potential if the index value is higher than 2.48.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>boman(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="boman_+3A_seq">seq</code></td>
<td>
<p>An amino-acid sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The potential protein interaction index was proposed by Boman (2003) as an easy way to differentiate the action mechanism of hormones (protein-protein) and antimicrobial peptides (protein-membrane) through this index. This function predicts the potential peptide interaction with another protein.
</p>


<h3>Value</h3>

<p>The computed potential protein-protein interaction for a given amino-acids sequence
</p>


<h3>References</h3>

<p>Boman, H. G. (2003). Antibacterial peptides: basic facts and emerging concepts. Journal of Internal Medicine, 254(3), 197-215.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COMPARED TO YADAMP DATABASE
# http://yadamp.unisa.it/showItem.aspx?yadampid=845&amp;x=0,4373912
# SEQUENCE: FLPVLAGLTPSIVPKLVCLLTKKC
# BOMAN INDEX  -1.24  

boman(seq= "FLPVLAGLTPSIVPKLVCLLTKKC")
# [1] -1.235833

</code></pre>

<hr>
<h2 id='charge'>Compute the theoretical net charge of a protein sequence</h2><span id='topic+charge'></span>

<h3>Description</h3>

<p>This function computes the net charge of a protein sequence based on the Henderson-Hasselbalch equation described by Moore, D. S. (1985). The net charge can be calculated at defined pH using one of the 9 pKa scales availables: <code>Bjellqvist</code>, <code>Dawson</code>, <code>EMBOSS</code>, <code>Lehninger</code>, <code>Murray</code>, <code>Rodwell</code>, <code>Sillero</code>, <code>Solomon</code> or <code>Stryer</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>charge(seq, pH = 7, pKscale = "Lehninger")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="charge_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
<tr><td><code id="charge_+3A_ph">pH</code></td>
<td>
<p>A pH value</p>
</td></tr>
<tr><td><code id="charge_+3A_pkscale">pKscale</code></td>
<td>
<p>A character string specifying the pKa scale to be used; must be one of <code>"Bjellqvist"</code>, <code>"Dawson"</code>, <code>"EMBOSS"</code>, <code>"Lehninger"</code>, <code>"Murray"</code>, <code>"Rodwell"</code>, <code>"Sillero"</code>, <code>"Solomon"</code> or <code>"Stryer"</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Original by Daniel Osorio &lt;dcosorioh@tamu.edu&gt;, C++ code optimized by Luis Pedro Coelho &lt;luis@luispedro.org&gt;
</p>


<h3>References</h3>

<p>Kiraga, J. (2008) Analysis and computer simulations of variability of isoelectric point of proteins in the proteomes. PhD thesis, University of Wroclaw, Poland.
</p>
<p>Bjellqvist, B., Hughes, G.J., Pasquali, Ch., Paquet, N., Ravier, F., Sanchez, J.Ch., Frutige,r S., Hochstrasser D. (1993) The focusing positions of polypeptides in immobilized pH gradients can be predicted from their amino acid sequences. Electrophoresis, 14:1023-1031.
</p>
<p>Dawson, R. M. C.; Elliot, D. C.; Elliot, W. H.; Jones, K. M. Data for biochemical research. Oxford University Press, 1989; p. 592.
</p>
<p>EMBOSS data are from http://emboss.sourceforge.net/apps/release/5.0/emboss/apps/iep.html.
</p>
<p>Nelson, D. L.; Cox, M. M. Lehninger Principles of Biochemistry, Fourth Edition; W. H. Freeman, 2004; p. 1100.
</p>
<p>Murray, R.K., Granner, D.K., Rodwell, V.W. (2006) Harper's illustrated Biochemistry. 27th edition. Published by The McGraw-Hill Companies.
</p>
<p>Rodwell, J. Heterogeneity of component bands in isoelectric focusing patterns. Analytical Biochemistry, 1982, 119 (2), 440-449.
</p>
<p>Sillero, A., Maldonado, A. (2006) Isoelectric point determination of proteins and other macromolecules: oscillating method. Comput Biol Med., 36:157-166.
</p>
<p>Solomon, T.W.G. (1998) Fundamentals of Organic Chemistry, 5th edition. Published by Wiley.
</p>
<p>Stryer L. (1999) Biochemia. czwarta edycja. Wydawnictwo Naukowe PWN.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COMPARED TO EMBOSS PEPSTATS
# http://emboss.bioinformatics.nl/cgi-bin/emboss/pepstats
# SEQUENCE: FLPVLAGLTPSIVPKLVCLLTKKC
# Charge   = 3.0

charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 7, pKscale= "Bjellqvist")
# [1] 2.737303
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 7, pKscale= "EMBOSS")
# [1] 2.914112
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 7, pKscale= "Murray")
# [1] 2.907541
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 7, pKscale= "Sillero")
# [1] 2.919812
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 7, pKscale= "Solomon")
# [1] 2.844406
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 7, pKscale= "Stryer")
# [1] 2.876504
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 7, pKscale= "Lehninger")
# [1] 2.87315
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 7, pKscale= "Dawson")
# [1] 2.844406
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 7, pKscale= "Rodwell")
# [1] 2.819755

# COMPARED TO YADAMP
# http://yadamp.unisa.it/showItem.aspx?yadampid=845&amp;x=0,7055475
# SEQUENCE: FLPVLAGLTPSIVPKLVCLLTKKC
# CHARGE pH5: 3.00
# CHARGE pH7: 2.91
# CHARGE pH9: 1.09
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 5, pKscale= "EMBOSS")
# [1] 3.037398
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 7, pKscale= "EMBOSS")
# [1] 2.914112
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= 9, pKscale= "EMBOSS")
# [1] 0.7184524

# JUST ONE COMMAND
charge(seq= "FLPVLAGLTPSIVPKLVCLLTKKC",pH= seq(from = 5,to = 9,by = 2), pKscale= "EMBOSS")
# [1] 3.0373984 2.9141123 0.7184524
</code></pre>

<hr>
<h2 id='crossCovariance'>Compute the cross-covariance index of a protein sequence</h2><span id='topic+crossCovariance'></span>

<h3>Description</h3>

<p>This function computes the Cruciani et al (2004) cross-covariance index. The lagged <code>crossCovariance</code> index is calculated for a lag 'd' using two descriptors 'f1' and 'f2' (centred) over a sequence of length 'L'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crossCovariance(sequence, lag, property1, property2, center = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="crossCovariance_+3A_sequence">sequence</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
<tr><td><code id="crossCovariance_+3A_lag">lag</code></td>
<td>
<p>A value for a lag, the max value is equal to the length of the shortest peptide minus one.</p>
</td></tr>
<tr><td><code id="crossCovariance_+3A_property1">property1</code></td>
<td>
<p>A property to use as value to evaluate the cross-covariance.</p>
</td></tr>
<tr><td><code id="crossCovariance_+3A_property2">property2</code></td>
<td>
<p>A property to use as value to evaluate the cross-covariance.</p>
</td></tr>
<tr><td><code id="crossCovariance_+3A_center">center</code></td>
<td>
<p>A logical value <code>TRUE</code> or <code>FALSE</code> if the property must be centered.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed cross-covariance index for a given amino-acids sequence
</p>


<h3>References</h3>

<p>Cruciani, G., Baroni, M., Carosati, E., Clementi, M., Valigi, R., and Clementi, S. (2004) Peptide studies by means of principal properties of amino acids derived from MIF descriptors. J. Chemom. 18, 146-155.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Loading a property to evaluate its autocorrelation
data(AAdata)

# Calculate the cross-covariance index for a lag=1
crossCovariance(
  sequence = "SDKEVDEVDAALSDLEITLE",
  lag = 1,
  property1 = AAdata$Hydrophobicity$KyteDoolittle,
  property2 = AAdata$Hydrophobicity$Eisenberg,
  center = TRUE
)
# [1] -0.3026609

# Calculate the cross-correlation index for a lag=5
crossCovariance(
  sequence = "SDKEVDEVDAALSDLEITLE",
  lag = 5,
  property1 = AAdata$Hydrophobicity$KyteDoolittle,
  property2 = AAdata$Hydrophobicity$Eisenberg,
  center = TRUE
)
# [1] 0.02598035
</code></pre>

<hr>
<h2 id='crucianiProperties'>Compute the Cruciani properties of a protein sequence</h2><span id='topic+crucianiProperties'></span>

<h3>Description</h3>

<p>This function calculates the Cruciani properties of an amino-acids sequence using the scaled principal component scores that summarize a broad set of descriptors calculated based on the interaction of each amino acid residue with several chemical groups (or &quot;probes&quot;), such as charged ions, methyl, hydroxyl groups, and so forth.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crucianiProperties(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="crucianiProperties_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed average of Cruciani properties of all the amino acids in the corresponding peptide sequence. Each PP represent an amino-acid property as follows: </p>
<ul>
<li><p>PP1: Polarity, </p>
</li>
<li><p>PP2: Hydrophobicity, </p>
</li>
<li><p>PP3: H-bonding</p>
</li></ul>



<h3>References</h3>

<p>Cruciani, G., Baroni, M., Carosati, E., Clementi, M., Valigi, R., and Clementi, S. (2004) Peptide studies by means of principal properties of amino acids derived from MIF descriptors. J. Chemom. 18, 146-155.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>crucianiProperties(seq = "QWGRRCCGWGPGRRYCVRWC")
#     PP1     PP2     PP3
#  -0.1130 -0.0220  0.2735
</code></pre>

<hr>
<h2 id='fasgaiVectors'>Compute the FASGAI vectors of a protein sequence</h2><span id='topic+fasgaiVectors'></span>

<h3>Description</h3>

<p>The FASGAI vectors (Factor Analysis Scales of Generalized Amino Acid Information) is a set of amino acid descriptors, that reflects hydrophobicity, alpha and turn propensities, bulky properties, compositional characteristics, local flexibility, and electronic properties, that can be utilized to represent the sequence structural features of peptides or protein motifs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fasgaiVectors(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fasgaiVectors_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed average of FASGAI factors of all the amino acids in the corresponding peptide sequence. Each factor represent an amino-acid property as follows: </p>
 <ul>
<li><p>F1: Hydrophobicity index, </p>
</li>
<li><p>F2: Alpha and turn propensities, </p>
</li>
<li><p>F3: Bulky properties, </p>
</li>
<li><p>F4: Compositional characteristic index, </p>
</li>
<li><p>F5: Local flexibility, </p>
</li>
<li><p>F6: Electronic properties</p>
</li></ul>



<h3>References</h3>

<p>Liang, G., &amp; Li, Z. (2007). Factor analysis scale of generalized amino acid information as the source of a new set of descriptors for elucidating the structure and activity relationships of cationic antimicrobial peptides. Molecular Informatics, 26(6), 754-763.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fasgaiVectors(seq = "QWGRRCCGWGPGRRYCVRWC")
# [[1]]
#     F1       F2       F3       F4       F5       F6 
# -0.13675 -0.45485 -0.11695 -0.45800 -0.38015  0.52740 
</code></pre>

<hr>
<h2 id='hmoment'>Compute the hydrophobic moment of a protein sequence</h2><span id='topic+hmoment'></span>

<h3>Description</h3>

<p>This function compute the hmoment based on Eisenberg, D., Weiss, R. M., &amp; Terwilliger, T. C. (1984). Hydriphobic moment is a quantitative measure of the amphiphilicity perpendicular to the axis of any periodic peptide structure, such as the a-helix or b-sheet. It can be calculated for an amino acid sequence of N residues and their associated hydrophobicities Hn.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hmoment(seq, angle = 100, window = 11)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hmoment_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
<tr><td><code id="hmoment_+3A_angle">angle</code></td>
<td>
<p>A protein rotational angle (Suggested: a-helix = 100, b-sheet=160)</p>
</td></tr>
<tr><td><code id="hmoment_+3A_window">window</code></td>
<td>
<p>A sequence fraction length</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The hydrophobic moment was proposed by Eisenberg et al. (1982), as a quantitative measure of the amphiphilicity perpendicular to the axis of any periodic peptide structure. It is computed using the standardized Eisenberg (1984) scale, windows (fragment of sequence) of eleven amino acids (by default) and specifying the rotational angle at which it should be calculated.
</p>


<h3>Value</h3>

<p>The computed maximal hydrophobic moment (uH) for a given amino-acids sequence
</p>


<h3>Note</h3>

<p>This function was written by an anonymous reviewer of the RJournal
</p>


<h3>References</h3>

<p>Eisenberg, D., Weiss, R. M., &amp; Terwilliger, T. C. (1984). The hydrophobic moment detects periodicity in protein hydrophobicity. Proceedings of the National Academy of Sciences, 81(1), 140-144.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COMPARED TO EMBOSS:HMOMENT
# http://emboss.bioinformatics.nl/cgi-bin/emboss/hmoment
# SEQUENCE: FLPVLAGLTPSIVPKLVCLLTKKC
# ALPHA-HELIX ANGLE=100 : 0.52
# BETA-SHEET  ANGLE=160 : 0.271

# ALPHA HELIX VALUE
hmoment(seq = "FLPVLAGLTPSIVPKLVCLLTKKC", angle = 100, window = 11)
# [1] 0.5199226

# BETA SHEET VALUE
hmoment(seq = "FLPVLAGLTPSIVPKLVCLLTKKC", angle = 160, window = 11)
# [1] 0.2705906
</code></pre>

<hr>
<h2 id='hydrophobicity'>Compute the hydrophobicity index of a protein sequence</h2><span id='topic+hydrophobicity'></span>

<h3>Description</h3>

<p>This function calculates the GRAVY hydrophobicity index of an amino acids sequence using one of the 38 scales from different sources.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hydrophobicity(seq, scale = "KyteDoolittle")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hydrophobicity_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
<tr><td><code id="hydrophobicity_+3A_scale">scale</code></td>
<td>
<p>A character string specifying the hydophobicity scale to be used; must be one of <code>"Aboderin"</code>, <code>"AbrahamLeo"</code>, <code>"Argos"</code>, <code>"BlackMould"</code>, <code>"BullBreese"</code>, <code>"Casari"</code>, <code>"Chothia"</code>, <code>"Cid"</code>, <code>"Cowan3.4"</code>, <code>"Cowan7.5"</code>, <code>"Eisenberg"</code>, <code>"Engelman"</code>, <code>"Fasman"</code>, <code>"Fauchere"</code>, <code>"Goldsack"</code>, <code>"Guy"</code>, <code>"HoppWoods"</code>, <code>"Janin"</code>, <code>"Jones"</code>, <code>"Juretic"</code>, <code>"Kidera"</code>, <code>"Kuhn"</code>, <code>"KyteDoolittle"</code>, <code>"Levitt"</code>, <code>"Manavalan"</code>, <code>"Miyazawa"</code>, <code>"Parker"</code>, <code>"Ponnuswamy"</code>, <code>"Prabhakaran"</code>, <code>"Rao"</code>, <code>"Rose"</code>, <code>"Roseman"</code>, <code>"Sweet"</code>, <code>"Tanford"</code>, <code>"Welling"</code>, <code>"Wilson"</code>, <code>"Wolfenden"</code>, <code>"Zimmerman"</code>, <code>"interfaceScale_pH8"</code>, <code>"interfaceScale_pH2"</code>, <code>"octanolScale_pH8"</code>, <code>"octanolScale_pH2"</code>, <code>"oiScale_pH8"</code> or <code>"oiScale_pH2"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The hydrophobicity is an important stabilization force in protein folding; this force changes depending on the solvent in which the protein is found. The hydrophobicity index is calculated adding the hydrophobicity of individual amino acids and dividing this value by the length of the sequence.
</p>


<h3>Value</h3>

<p>The computed GRAVY index for a given amino-acid sequence
</p>


<h3>References</h3>

<p>Aboderin, A. A. (1971). An empirical hydrophobicity scale for alpha-amino-acids and some of its applications. International Journal of Biochemistry, 2(11), 537-544.
</p>
<p>Abraham D.J., Leo A.J. Hydrophobicity (delta G1/2 cal). Proteins: Structure, Function and Genetics 2:130-152(1987).
</p>
<p>Argos, P., Rao, J. K., &amp; Hargrave, P. A. (1982). Structural Prediction of Membrane-Bound Proteins. European Journal of Biochemistry, 128(2-3), 565-575.
</p>
<p>Black S.D., Mould D.R. Hydrophobicity of physiological L-alpha amino acids. Anal. Biochem. 193:72-82(1991).
</p>
<p>Bull H.B., Breese K. Hydrophobicity (free energy of transfer to surface in kcal/mole). Arch. Biochem. Biophys. 161:665-670(1974).
</p>
<p>Casari, G., &amp; Sippl, M. J. (1992). Structure-derived hydrophobic potential: hydrophobic potential derived from X-ray structures of globular proteins is able to identify native folds. Journal of molecular biology, 224(3), 725-732.
</p>
<p>Chothia, C. (1976). The nature of the accessible and buried surfaces in proteins. Journal of molecular biology, 105(1), 1-12.
</p>
<p>Cid, H., Bunster, M., Canales, M., &amp; Gazitua, F. (1992). Hydrophobicity and structural classes in proteins. Protein engineering, 5(5), 373-375.
</p>
<p>Cowan R., Whittaker R.G. Hydrophobicity indices at pH 3.4 determined by HPLC. Peptide Research 3:75-80(1990).
</p>
<p>Cowan R., Whittaker R.G. Hydrophobicity indices at pH 7.5 determined by HPLC. Peptide Research 3:75-80(1990).
</p>
<p>Eisenberg D., Schwarz E., Komarony M., Wall R. Normalized consensus hydrophobicity scale. J. Mol. Biol. 179:125-142(1984).
</p>
<p>Engelman, D. M., Steitz, T. A., &amp; Goldman, A. (1986). Identifying nonpolar transbilayer helices in amino acid sequences of membrane proteins. Annual review of biophysics and biophysical chemistry, 15(1), 321-353.
</p>
<p>Fasman, G. D. (Ed.). (1989). Prediction of protein structure and the principles of protein conformation. Springer.
</p>
<p>Fauchere J.-L., Pliska V.E. Hydrophobicity scale (pi-r). Eur. J. Med. Chem. 18:369-375(1983).
</p>
<p>Goldsack, D. E., &amp; Chalifoux, R. C. (1973). Contribution of the free energy of mixing of hydrophobic side chains to the stability of the tertiary structure of proteins. Journal of theoretical biology, 39(3), 645-651.
</p>
<p>Guy H.R. Hydrophobicity scale based on free energy of transfer (kcal/mole). Biophys J. 47:61-70(1985).
</p>
<p>Hopp T.P., Woods K.R. Hydrophilicity. Proc. Natl. Acad. Sci. U.S.A. 78:3824-3828(1981).
</p>
<p>Janin J. Free energy of transfer from inside to outside of a globular protein. Nature 277:491-492(1979).
</p>
<p>Jones, D. D. (1975). Amino acid properties and side-chain orientation in proteins: a cross correlation approach. Journal of theoretical biology, 50(1), 167-183.
</p>
<p>Juretic, D., Lucic, B., Zucic, D., &amp; Trinajstic, N. (1998). Protein transmembrane structure: recognition and prediction by using hydrophobicity scales through preference functions. Theoretical and computational chemistry, 5, 405-445.
</p>
<p>Kidera, A., Konishi, Y., Oka, M., Ooi, T., &amp; Scheraga, H. A. (1985). Statistical analysis of the physical properties of the 20 naturally occurring amino acids. Journal of Protein Chemistry, 4(1), 23-55.
</p>
<p>Kuhn, L. A., Swanson, C. A., Pique, M. E., Tainer, J. A., &amp; Getzoff, E. D. (1995). Atomic and residue hydrophilicity in the context of folded protein structures. Proteins: Structure, Function, and Bioinformatics, 23(4), 536-547.
</p>
<p>Kyte J., Doolittle R.F. Hydropathicity. J. Mol. Biol. 157:105-132(1982).
</p>
<p>Levitt, M. (1976). A simplified representation of protein conformations for rapid simulation of protein folding. Journal of molecular biology, 104(1), 59-107.
</p>
<p>Manavalan P., Ponnuswamy Average surrounding hydrophobicity. P.K. Nature 275:673-674(1978).
</p>
<p>Miyazawa S., Jernigen R.L. Hydrophobicity scale (contact energy derived from 3D data). Macromolecules 18:534-552(1985).
</p>
<p>Parker J.M.R., Guo D., Hodges R.S. Hydrophilicity scale derived from HPLC peptide retention times. Biochemistry 25:5425-5431(1986).
</p>
<p>Ponnuswamy, P. K. (1993). Hydrophobic charactesristics of folded proteins. Progress in biophysics and molecular biology, 59(1), 57-103.
</p>
<p>Prabhakaran, M. (1990). The distribution of physical, chemical and conformational properties in signal and nascent peptides. Biochem. J, 269, 691-696.
</p>
<p>Rao M.J.K., Argos P. Membrane buried helix parameter. Biochim. Biophys. Acta 869:197-214(1986).
</p>
<p>Rose G.D., Geselowitz A.R., Lesser G.J., Lee R.H., Zehfus M.H. Mean fractional area loss (f) [average area buried/standard state area]. Science 229:834-838(1985)
</p>
<p>Roseman M.A. Hydrophobicity scale (pi-r). J. Mol. Biol. 200:513-522(1988).
</p>
<p>Sweet R.M., Eisenberg D. Optimized matching hydrophobicity (OMH). J. Mol. Biol. 171:479-488(1983).
</p>
<p>Tanford C. Hydrophobicity scale (Contribution of hydrophobic interactions to the stability of the globular conformation of proteins). J. Am. Chem. Soc. 84:4240-4274(1962).
</p>
<p>Welling G.W., Weijer W.J., Van der Zee R., Welling-Wester S. Antigenicity value X 10. FEBS Lett. 188:215-218(1985).
</p>
<p>Wilson K.J., Honegger A., Stotzel R.P., Hughes G.J. Hydrophobic constants derived from HPLC peptide retention times. Biochem. J. 199:31-41(1981).
</p>
<p>Wolfenden R.V., Andersson L., Cullis P.M., Southgate C.C.F. Hydration potential (kcal/mole) at 25C. Biochemistry 20:849-855(1981).
</p>
<p>Zimmerman, J. M., Eliezer, N., &amp; Simha, R. (1968). The characterization of amino acid sequences in proteins by statistical methods. Journal of theoretical biology, 21(2), 170-201.
</p>
<p>Nakai, K., Kidera, A., and Kanehisa, M.; Cluster analysis of amino acid indices for prediction of protein structure and function. Protein Eng. 2, 93-100 (1988).
</p>
<p>Tomii, K. and Kanehisa, M.; Analysis of amino acid indices and mutation matrices for sequence comparison and structure prediction of proteins. Protein Eng. 9, 27-36 (1996). 
</p>
<p>Kawashima, S., Ogata, H., and Kanehisa, M.; AAindex: amino acid index database. Nucleic Acids Res. 27, 368-369 (1999).
</p>
<p>Kawashima, S. and Kanehisa, M.; AAindex: amino acid index database. Nucleic Acids Res. 28, 374 (2000).
</p>
<p>Kawashima, S., Pokarowski, P., Pokarowska, M., Kolinski, A., Katayama, T., and Kanehisa, M.; AAindex: amino acid index database, progress report 2008. Nucleic Acids Res. 36, D202-D205 (2008). 
</p>
<p>White, Stephen (2006-06-29). &quot;Experimentally Determined Hydrophobicity Scales&quot;. University of California, Irvine. Retrieved 2017-05-25
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COMPARED TO GRAVY Grand average of hydropathicity (GRAVY) ExPASy
# http://web.expasy.org/cgi-bin/protparam/protparam
# SEQUENCE: QWGRRCCGWGPGRRYCVRWC
# GRAVY: -0.950

hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Aboderin")
#[1] 3.84
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "AbrahamLeo")
#[1] 0.092
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Argos")
#[1] 1.033
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "BlackMould")
#[1] 0.50125
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "BullBreese")
#[1] 0.1575
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Casari")
#[1] 0.38
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Chothia")
#[1] 0.262
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Cid")
#[1] 0.198
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Cowan3.4")
#[1] 0.0845
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Cowan7.5")
#[1] 0.0605
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Eisenberg")
#[1] -0.3265
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Engelman")
#[1] 2.31
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Fasman")
#[1] -1.2905
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Fauchere")
#[1] 0.527
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Goldsack")
#[1] 1.2245
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Guy")
#[1] 0.193
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "HoppWoods")
#[1] -0.14
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Janin")
#[1] -0.105
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Jones")
#[1] 1.4675
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Juretic")
#[1] -1.106
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Kidera")
#[1] -0.0405
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Kuhn")
#[1] 0.9155
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "KyteDoolittle")
#[1] -0.95
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Levitt")
#[1] -0.21
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Manavalan")
#[1] 13.0445
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Miyazawa")
#[1] 5.739
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Parker")
#[1] 1.095
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Ponnuswamy")
#[1] 0.851
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Prabhakaran")
#[1] 9.67
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Rao")
#[1] 0.813
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Rose")
#[1] 0.7575
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Roseman")
#[1] -0.495
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Sweet")
#[1] -0.1135
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Tanford")
#[1] -0.2905
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Welling")
#[1] -0.666
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Wilson")
#[1] 3.16
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Wolfenden")
#[1] -6.307
hydrophobicity(seq = "QWGRRCCGWGPGRRYCVRWC",scale = "Zimmerman")
#[1] 0.943
</code></pre>

<hr>
<h2 id='instaIndex'>Compute the instability index of a protein sequence</h2><span id='topic+instaIndex'></span>

<h3>Description</h3>

<p>This function calculates the instability index proposed by Guruprasad (1990). This index predicts the stability of a protein based on its amino acid composition, a protein whose instability index is smaller than 40 is predicted as stable, a value above 40 predicts that the protein may be unstable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>instaIndex(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="instaIndex_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed instability index for a given amino-acids sequence
</p>


<h3>References</h3>

<p>Guruprasad K, Reddy BV, Pandit MW (1990). &quot;Correlation between stability of a protein and its dipeptide composition: a novel approach for predicting in vivo stability of a protein from its primary sequence&quot;. Protein Eng. 4 (2): 155 - 61. doi:10.1093/protein/4.2.155
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COMPARED TO ExPASy INSTAINDEX
# http://web.expasy.org/protparam/
# SEQUENCE: QWGRRCCGWGPGRRYCVRWC
# The instability index (II) is computed to be 83.68

instaIndex(seq = "QWGRRCCGWGPGRRYCVRWC")
# [1] 83.68

</code></pre>

<hr>
<h2 id='kideraFactors'>Compute the Kidera factors of a protein sequence</h2><span id='topic+kideraFactors'></span>

<h3>Description</h3>

<p>The Kidera Factors were originally derived by applying multivariate analysis to 188 physical properties of the 20 amino acids and using dimension reduction techniques. This function calculates the average of the ten Kidera factors for a protein sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kideraFactors(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kideraFactors_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the average of the ten Kidera factors. 
The first four factors are essentially pure physical properties; the remaining six factors are superpositions of several physical properties, and are labelled for convenience by the name of the most heavily weighted component.
</p>

<ul>
<li><p>KF1: Helix/bend preference,
</p>
</li>
<li><p>KF2: Side-chain size,
</p>
</li>
<li><p>KF3: Extended structure preference,
</p>
</li>
<li><p>KF4: Hydrophobicity,
</p>
</li>
<li><p>KF5: Double-bend preference,
</p>
</li>
<li><p>KF6: Partial specific volume,
</p>
</li>
<li><p>KF7: Flat extended preference,
</p>
</li>
<li><p>KF8: Occurrence in alpha region,
</p>
</li>
<li><p>KF9: pK-C,
</p>
</li>
<li><p>KF10: Surrounding hydrophobicity</p>
</li></ul>



<h3>References</h3>

<p>Kidera, A., Konishi, Y., Oka, M., Ooi, T., &amp; Scheraga, H. A. (1985). Statistical analysis of the physical properties of the 20 naturally occurring amino acids. Journal of Protein Chemistry, 4(1), 23-55.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>kideraFactors(seq = "KLKLLLLLKLK")
# [[1]]
#     KF1         KF2         KF3         KF4         KF5 
# -0.78545455  0.29818182 -0.23636364 -0.08181818  0.21000000 
#     KF6         KF7         KF8         KF9        KF10 
# -1.89363636  1.02909091 -0.51272727  0.11181818  0.81000000 
</code></pre>

<hr>
<h2 id='lengthpep'>Compute the amino acid length of a protein sequence</h2><span id='topic+lengthpep'></span>

<h3>Description</h3>

<p>This function counts the number of amino acids in a protein sequence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lengthpep(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lengthpep_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All proteins are formed by linear chains of small residues known as amino acids attached to each other by peptide bonds. The function <code>lengthpep</code> counts the number of amino acids in a sequence and returns a vector with the count for each peptide used as argument.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COMPARED TO ExPASy ProtParam
# http://web.expasy.org/protparam
# SEQUENCE: QWGRRCCGWGPGRRYCVRWC
# Number of amino acids: 20

lengthpep(seq = "QWGRRCCGWGPGRRYCVRWC")
# [1] 20
</code></pre>

<hr>
<h2 id='massShift'>Calculate the mass difference of modified peptides.</h2><span id='topic+massShift'></span>

<h3>Description</h3>

<p>This function calculates the mass difference of peptides introduced by chemical modifications or heavy isotope labelling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>massShift(seq, label = "none", aaShift = NULL, monoisotopic = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="massShift_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence, in one letter code.</p>
</td></tr>
<tr><td><code id="massShift_+3A_label">label</code></td>
<td>
<p>Set a predefined heavy isotope label. Accepts &quot;none&quot;, &quot;silac_13c&quot;, &quot;silac_13c15n&quot; and &quot;15n&quot;. Overwrites input in <code>aaShift</code>.</p>
</td></tr>
<tr><td><code id="massShift_+3A_aashift">aaShift</code></td>
<td>
<p>Define the mass difference in Dalton of given amino acids as a named vector. 
Use the amino acid one letter code as names and the mass shift in Dalton as values. 
N-terminal and C-terminal modifications can be defined by using &quot;Nterm =&quot; and &quot;Cterm =&quot;, respectively.</p>
</td></tr>
<tr><td><code id="massShift_+3A_monoisotopic">monoisotopic</code></td>
<td>
<p>A logical value <code>'TRUE'</code> or <code>'FALSE'</code> indicating if monoisotopic weights of amino-acids should be used</p>
</td></tr>
</table>


<h3>Source</h3>

<p>For the predefined heavy isotope labels, compare:
</p>

<ul>
<li><p> silac_13c 
<a href="http://www.unimod.org/modifications_view.php?editid1=188">Unimod 188</a>
</p>
</li>
<li><p> silac_13c15n
<a href="http://www.unimod.org/modifications_view.php?editid1=259">Unimod 259</a> and
<a href="http://www.unimod.org/modifications_view.php?editid1=267">Unimod 267</a>
</p>
</li>
<li><p> 15n
<a href="http://www.unimod.org/modifications_view.php?editid1=994">Unimod 994</a>, 
<a href="http://www.unimod.org/modifications_view.php?editid1=995">Unimod 995</a>, 
<a href="http://www.unimod.org/modifications_view.php?editid1=996">Unimod 996</a> and 
<a href="http://www.unimod.org/modifications_view.php?editid1=897">Unimod 897</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>massShift("EGVNDNECEGFFSAR", label = "silac_13c")
massShift("EGVNDNECEGFFSAR", aaShift = c(K = 6.020129, R = 6.020129))

</code></pre>

<hr>
<h2 id='membpos'>Compute theoretically the class of a protein sequence</h2><span id='topic+membpos'></span>

<h3>Description</h3>

<p>This function calculates the theoretical class of a protein sequence based on the relationship between the hydrophobic moment and hydrophobicity scale proposed by Eisenberg (1984).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>membpos(seq, angle = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="membpos_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
<tr><td><code id="membpos_+3A_angle">angle</code></td>
<td>
<p>A protein rotational angle</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Eisenberg et al. (1982) found a correlation between hydrophobicity and hydrophobic moment that defines the protein section as globular, transmembrane or superficial. The function calculates the hydrophobicity (H) and hydrophobic moment (uH) based on the standardized scale of Eisenberg (1984) using windows of 11 amino acids for calculate the theoretical fragment type.
</p>


<h3>Value</h3>

<p>A data frame for each sequence given with the calculated class for each window of eleven amino-acids
</p>


<h3>References</h3>

<p>Eisenberg, David. &quot;Three-dimensional structure of membrane and surface proteins.&quot; Annual review of biochemistry 53.1 (1984): 595-623.
</p>
<p>D. Eisenberg, R. M. Weiss, and T. C. Terwilliger. The helical hydrophobic moment: A measure of the amphiphilicity of a helix. Nature, 299(5881):371-374, 1982. [p7, 8]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>membpos(seq = "ARQQNLFINFCLILIFLLLI",angle = 100)
#       Pep        H     uH       MembPos
# 1 ARQQNLFINFCL 0.083 0.353      Globular
# 2 RQQNLFINFCLI 0.147 0.317      Globular
# 3 QQNLFINFCLIL 0.446 0.274      Globular
# 4 QNLFINFCLILI 0.632 0.274 Transmembrane
# 5 NLFINFCLILIF 0.802 0.253       Surface
# 6 LFINFCLILIFL 0.955 0.113 Transmembrane
# 7 FINFCLILIFLL 0.955 0.113 Transmembrane
# 8 INFCLILIFLLL 0.944 0.108 Transmembrane
# 9 NFCLILIFLLLI 0.944 0.132 Transmembrane

membpos(seq = "ARQQNLFINFCLILIFLLLI",angle = 160)
#       Pep        H     uH    MembPos
# 1 ARQQNLFINFCL 0.083 0.467  Globular
# 2 RQQNLFINFCLI 0.147 0.467  Globular
# 3 QQNLFINFCLIL 0.446 0.285  Globular
# 4 QNLFINFCLILI 0.632 0.358  Surface
# 5 NLFINFCLILIF 0.802 0.358  Surface
# 6 LFINFCLILIFL 0.955 0.269  Surface
# 7 FINFCLILIFLL 0.955 0.269  Surface
# 8 INFCLILIFLLL 0.944 0.257  Surface
# 9 NFCLILIFLLLI 0.944 0.229  Surface
</code></pre>

<hr>
<h2 id='mswhimScores'>Compute the MS-WHIM scores of a protein sequence</h2><span id='topic+mswhimScores'></span>

<h3>Description</h3>

<p>MS-WHIM scores were derived from 36 electrostatic potential properties derived from the three-dimensional structure of the 20 natural amino acids
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mswhimScores(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mswhimScores_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed average of MS-WHIM scores of all the amino acids in the corresponding peptide sequence.
</p>


<h3>References</h3>

<p>Zaliani, A., &amp; Gancia, E. (1999). MS-WHIM scores for amino acids: a new 3D-description for peptide QSAR and QSPR studies. Journal of chemical information and computer sciences, 39(3), 525-533.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mswhimScores(seq = "KLKLLLLLKLK")
# [[1]]
#  MSWHIM1    MSWHIM2    MSWHIM3 
# -0.6563636  0.4872727  0.1163636 
</code></pre>

<hr>
<h2 id='mw'>Compute the molecular weight of a protein sequence</h2><span id='topic+mw'></span>

<h3>Description</h3>

<p>This function calculates the molecular weight of a protein sequence. It is calculated as the sum of the mass of each amino acid using the scale available on Compute pI/Mw tool. It also supports mass calculation of proteins with predefined or custom stable isotope mass labels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mw(
  seq,
  monoisotopic = FALSE,
  avgScale = "expasy",
  label = "none",
  aaShift = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mw_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
<tr><td><code id="mw_+3A_monoisotopic">monoisotopic</code></td>
<td>
<p>A logical value <code>'TRUE'</code> or <code>'FALSE'</code> indicating if monoisotopic weights of amino-acids should be used</p>
</td></tr>
<tr><td><code id="mw_+3A_avgscale">avgScale</code></td>
<td>
<p>Set the mass scale to use for average weight only (if 'monoisotopic == FALSE'). Accepts &quot;expasy&quot; (default) or &quot;mascot&quot;.</p>
</td></tr>
<tr><td><code id="mw_+3A_label">label</code></td>
<td>
<p>Set a predefined heavy isotope label. Accepts &quot;none&quot;, &quot;silac_13c&quot;, &quot;silac_13c15n&quot; and &quot;15n&quot;. Overwrites input in <code>aaShift</code>.</p>
</td></tr>
<tr><td><code id="mw_+3A_aashift">aaShift</code></td>
<td>
<p>Define the mass difference in Dalton of given amino acids as a named vector. Use the amino acid one letter code as names and the mass shift in Dalton as values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The molecular weight is the sum of the masses of each atom constituting a molecule. The molecular weight is directly related to the length of the amino acid sequence and is expressed in units called daltons (Da). In Peptides the function mw computes the molecular weight using the same formulas and weights as ExPASy's &quot;compute pI/mw&quot; tool (Gasteiger et al., 2005). For average weight, the ExPASy tools use the following mass scale: https://web.expasy.org/findmod/findmod_masses.html#AA , while UniMod and Mascot use a slightly different one: http://www.matrixscience.com/help/aa_help.html .
</p>


<h3>Source</h3>

<p>The formula and amino acid scale are the same available on ExPASy Compute pI/Mw tool: http://web.expasy.org/compute_pi/
</p>


<h3>References</h3>

<p>Gasteiger, E., Hoogland, C., Gattiker, A., Wilkins, M. R., Appel, R. D., &amp; Bairoch, A. (2005). Protein identification and analysis tools on the ExPASy server. In The proteomics protocols handbook (pp. 571-607). Humana Press. Chicago
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COMPARED TO ExPASy Compute pI/Mw tool
# http://web.expasy.org/compute_pi/
# SEQUENCE: QWGRRCCGWGPGRRYCVRWC 
# Theoretical pI/Mw: 9.88 / 2485.91 

mw(seq = "QWGRRCCGWGPGRRYCVRWC",monoisotopic = FALSE)
# [1] 2485.911

mw(seq = "QWGRRCCGWGPGRRYCVRWC",monoisotopic = FALSE, avgScale = "mascot")
# [1] 2485.899

mw(seq = "QWGRRCCGWGPGRRYCVRWC",monoisotopic = TRUE)
# [1] 2484.12

</code></pre>

<hr>
<h2 id='mz'>Calculate the m/z for peptides.</h2><span id='topic+mz'></span>

<h3>Description</h3>

<p>This function calculates the (monoisotopic) mass over charge ratio (m/z) for peptides, as measured in mass spectrometry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mz(seq, charge = 2, label = "none", aaShift = NULL, cysteins = 57.021464)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mz_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence, in one letter code.</p>
</td></tr>
<tr><td><code id="mz_+3A_charge">charge</code></td>
<td>
<p>The net charge for which the m/z should be calculated</p>
</td></tr>
<tr><td><code id="mz_+3A_label">label</code></td>
<td>
<p>Set a predefined heavy isotope label. Accepts &quot;none&quot;, &quot;silac_13c&quot;, &quot;silac_13c15n&quot; and &quot;15n&quot;. Overwrites input in <code>aaShift</code>.</p>
</td></tr>
<tr><td><code id="mz_+3A_aashift">aaShift</code></td>
<td>
<p>Define the mass difference in Dalton of given amino acids as a named vector. Use the amino acid one letter code as names and the mass shift in Dalton as values.</p>
</td></tr>
<tr><td><code id="mz_+3A_cysteins">cysteins</code></td>
<td>
<p>Define the mass shift in Dalton of blocked cysteins. Defaults to 57.021464, for cysteins blocked by iodoacetamide.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>mz("EGVNDNECEGFFSAR")
mz("EGVNDNECEGFFSAR", aaShift = c(K = 6.020129, R = 6.020129))
mz("EGVNDNECEGFFSAR", label = "silac_13c", cysteins = 58.005479)

</code></pre>

<hr>
<h2 id='pepdata'>Physicochemical properties and indices from 100 amino acid sequences</h2><span id='topic+pepdata'></span>

<h3>Description</h3>

<p>Physicochemical properties and indices from 100 amino acid sequences (50 antimicrobial and 50 non antimicrobial)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pepdata)
</code></pre>


<h3>Format</h3>

<p>A data frame with 100 observations on the following 23 variables. 
</p>
 
<dl>
<dt><code>sequence</code></dt><dd><p>a character vector with the sequences of 100 peptides (50 antimicrobial and 50 non-antimicrobial)</p>
</dd>
<dt><code>group</code></dt><dd><p>Integrer vector with the group code <code>"0"</code> for non antimicrobial and <code>"1"</code> for antimicrobial</p>
</dd>
<dt><code>length</code></dt><dd><p>a numeric vector with the length of the amino acid sequence</p>
</dd>
<dt><code>mw</code></dt><dd><p>a numeric vector with the molecular weight of the amino acid sequence</p>
</dd>
<dt><code>tinyAA</code></dt><dd><p>A numeric vector with the fraction (as percent) of tiny amino acids that make up the sequence</p>
</dd>
<dt><code>smallAA</code></dt><dd><p>A numeric vector with the fraction (as percent) of small amino acids that make up the sequence</p>
</dd>
<dt><code>aliphaticAA</code></dt><dd><p>A numeric vector with the fraction (as percent) of aliphatic amino acids that make up the sequence</p>
</dd>
<dt><code>aromaticAA</code></dt><dd><p>A numeric vector with the fraction (as percent) of aromatic amino acids that make up the sequence</p>
</dd>
<dt><code>nonpolarAA</code></dt><dd><p>A numeric vector with the fraction (as percent) of non-polar amino acids that make up the sequence</p>
</dd>
<dt><code>polarAA</code></dt><dd><p>A numeric vector with the fraction (as percent) of polar amino acids that make up the sequence</p>
</dd>
<dt><code>chargedAA</code></dt><dd><p>A numeric vector with the fraction (as percent) of charged amino acids that make up the sequence</p>
</dd>
<dt><code>basicAA</code></dt><dd><p>A numeric vector with the fraction (as percent) of basic amino acids that make up the sequence</p>
</dd>
<dt><code>acidicAA</code></dt><dd><p>A numeric vector with the fraction (as percent) of acid amino acids that make up the sequence</p>
</dd>
<dt><code>charge</code></dt><dd><p>a numeric vector with the charge of the amino acid sequence</p>
</dd>
<dt><code>pI</code></dt><dd><p>a numeric vector with the isoelectric point of the amino acid sequence</p>
</dd>
<dt><code>aindex</code></dt><dd><p>a numeric vector with the aliphatic index of the amino acid sequence</p>
</dd>
<dt><code>instaindex</code></dt><dd><p>a numeric vector with the instability index of the amino acid sequence</p>
</dd>
<dt><code>boman</code></dt><dd><p>a numeric vector with the potential peptide-interaction index of the amino acid sequence</p>
</dd>
<dt><code>hydrophobicity</code></dt><dd><p>a numeric vector with the hydrophobicity index of the amino acid sequence</p>
</dd>
<dt><code>hmoment</code></dt><dd><p>a numeric vector with the hydrophobic moment of the amino acid sequence</p>
</dd>
<dt><code>transmembrane</code></dt><dd><p>A numeric vector with the fraction of Transmembrane windows of 11 amino acids that make up the sequence</p>
</dd>
<dt><code>surface</code></dt><dd><p>A numeric vector with the fraction of Surface windows of 11 amino acids that make up the sequence</p>
</dd>
<dt><code>globular</code></dt><dd><p>A numeric vector with the fraction of Globular windows of 11 amino acids that make up the sequence</p>
</dd>
</dl>


<hr>
<h2 id='pI'>Compute the isoelectic point (pI) of a protein sequence</h2><span id='topic+pI'></span>

<h3>Description</h3>

<p>The isoelectric point (pI), is the pH at which a particular molecule or surface carries no net electrical charge.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pI(seq, pKscale = "EMBOSS")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pI_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
<tr><td><code id="pI_+3A_pkscale">pKscale</code></td>
<td>
<p>A character string specifying the pK scale to be used; must be one of <code>"Bjellqvist"</code>, <code>"EMBOSS"</code>, <code>"Murray"</code>, <code>"Sillero"</code>, <code>"Solomon"</code>, <code>"Stryer"</code>, <code>"Lehninger"</code>, <code>"Dawson"</code> or <code>"Rodwell"</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The isoelectric point (pI) is the pH at which the net charge of the protein is equal to 0. It is a variable that affects the solubility of the peptides under certain conditions of pH. When the pH of the solvent is equal to the pI of the protein, it tends to precipitate and lose its biological function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COMPARED TO ExPASy ProtParam
# http://web.expasy.org/cgi-bin/protparam/protparam
# SEQUENCE: QWGRRCCGWGPGRRYCVRWC
# Theoretical pI: 9.88

pI(seq= "QWGRRCCGWGPGRRYCVRWC",pKscale= "Bjellqvist")
# [1] 9.881

# COMPARED TO EMBOSS PEPSTATS
# http://emboss.bioinformatics.nl/cgi-bin/emboss/pepstats
# SEQUENCE: QWGRRCCGWGPGRRYCVRWC
# Isoelectric Point = 9.7158

pI(seq= "QWGRRCCGWGPGRRYCVRWC",pKscale= "EMBOSS")
# [1] 9.716

# OTHER SCALES

pI(seq= "QWGRRCCGWGPGRRYCVRWC",pKscale= "Murray")
# [1] 9.818
pI(seq= "QWGRRCCGWGPGRRYCVRWC",pKscale= "Sillero")
# [1] 9.891
pI(seq= "QWGRRCCGWGPGRRYCVRWC",pKscale= "Solomon")
# [1] 9.582
pI(seq= "QWGRRCCGWGPGRRYCVRWC",pKscale= "Stryer")
# [1] 9.623
pI(seq= "QWGRRCCGWGPGRRYCVRWC",pKscale= "Lehninger")
# [1] 9.931
pI(seq= "QWGRRCCGWGPGRRYCVRWC",pKscale= "Dawson")
# [1] 9.568
pI(seq= "QWGRRCCGWGPGRRYCVRWC",pKscale= "Rodwell")
# [1] 9.718
</code></pre>

<hr>
<h2 id='plotXVG'>Plot time series from GROMACS XVG files</h2><span id='topic+plotXVG'></span>

<h3>Description</h3>

<p>Read and plot output data from a XVG format file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotXVG(XVGfile, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotXVG_+3A_xvgfile">XVGfile</code></td>
<td>
<p>A .XVG output file of the GROMACS molecular dynamics package</p>
</td></tr>
<tr><td><code id="plotXVG_+3A_...">...</code></td>
<td>
<p>Arguments to be passed to methods, such as graphical parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>GROMACS (GROningen MAchine for Chemical Simulations) is a molecular dynamics package designed for simulations of proteins, lipids and nucleic acids. It is free, open source software released under the GNU General Public License. 
The file format used by GROMACS is XVG. This format can be displayed in graphical form through the GRACE program on UNIX/LINUX systems and the GNUPlot program on Windows. XVG files are plain text files containing tabular data separated by tabulators and two types of comments which contain data labels. Although manual editing is possible, this is not a viable option when working with multiple files of this type. 
For ease of reading, information management and data plotting, the functions <code>read.xvg</code> and <code>plot.xvg</code> were incorporated.
</p>


<h3>Author(s)</h3>

<p>Latest: J. Sebastian Paez &lt;jpaezpae@purdue.edu&gt;
</p>
<p>Original: Daniel Osorio &lt;dcosorioh@unal.edu.co&gt;
</p>


<h3>References</h3>

<p>Pronk, S., Pall, S., Schulz, R., Larsson, P., Bjelkmar, P., Apostolov, R., ... &amp; Lindahl, E. (2013). GROMACS 4.5: a high-throughput and highly parallel open source molecular simulation toolkit. Bioinformatics, 29 (7), 845-854.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>XVGfile &lt;- system.file("xvg-files/epot.xvg",package="Peptides")
plotXVG(XVGfile)
</code></pre>

<hr>
<h2 id='protFP'>Compute the protFP descriptors of a protein sequence</h2><span id='topic+protFP'></span>

<h3>Description</h3>

<p>The ProtFP descriptor set was constructed from a large initial selection of indices obtained from the AAindex database for all 20 naturally occurring amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>protFP(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="protFP_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed average of protFP descriptors of all the amino acids in the corresponding peptide sequence.
</p>


<h3>References</h3>

<p>van Westen, G. J., Swier, R. F., Wegner, J. K., IJzerman, A. P., van Vlijmen, H. W., &amp; Bender, A. (2013). Benchmarking of protein descriptor sets in proteochemometric modeling (part 1): comparative study of 13 amino acid descriptor sets. Journal of cheminformatics, 5(1), 41.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>protFP(seq = "QWGRRCCGWGPGRRYCVRWC")
# [[1]]
# ProtFP1 ProtFP2 ProtFP3 ProtFP4 ProtFP5 ProtFP6 ProtFP7 ProtFP8 
# 0.2065 -0.0565  1.9930 -0.2845  0.7315  0.7000  0.1715  0.1135 
</code></pre>

<hr>
<h2 id='readXVG'>Read output data from a XVG format file.</h2><span id='topic+readXVG'></span>

<h3>Description</h3>

<p>XVG is the default format file of the GROMACS molecular dynamics package, contains data formatted to be imported into the Grace 2-D plotting program.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readXVG(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readXVG_+3A_file">file</code></td>
<td>
<p>A .XVG output file of the GROMACS molecular dynamics package</p>
</td></tr>
</table>


<h3>Details</h3>

<p>GROMACS (GROningen MAchine for Chemical Simulations) is a molecular dynamics package designed for simulations of proteins, lipids and nucleic acids. It is free, open source software released under the GNU General Public License. 
The file format used by GROMACS is XVG. This format can be displayed in graphical form through the GRACE program on UNIX/LINUX systems and the GNUPlot program on Windows. XVG files are plain text files containing tabular data separated by tabulators and two types of comments which contain data labels. Although manual editing is possible, this is not a viable option when working with multiple files of this type. 
For ease of reading, information management and data plotting, the functions <code>read.xvg</code> and <code>plot.xvg</code> were incorporated.
</p>


<h3>Author(s)</h3>

<p>Latest: J. Sebastian Paez &lt;jpaezpae@purdue.edu&gt; and hongbo-zhu-cn &lt;@github&gt;
</p>
<p>Original: Daniel Osorio &lt;dcosorioh@unal.edu.co&gt;
</p>


<h3>References</h3>

<p>Pronk, S., Pall, S., Schulz, R., Larsson, P., Bjelkmar, P., Apostolov, R., ... &amp; Lindahl, E. (2013). GROMACS 4.5: a high-throughput and highly parallel open source molecular simulation toolkit. Bioinformatics, 29 (7), 845-854.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># READING FILE
XVGfile &lt;- system.file("xvg-files/epot.xvg",package="Peptides")
readXVG(XVGfile)

#    Time (ps)  Potential
#  1         1 6672471040
#  2         2 6516461568
#  3         3 6351947264
#  4         4 6183133184
#  5         5 6015310336
#  6         6 5854271488
</code></pre>

<hr>
<h2 id='stScales'>Compute the ST-scales of a protein sequence</h2><span id='topic+stScales'></span>

<h3>Description</h3>

<p>ST-scales were proposed by Yang et al, taking 827 properties into account which are mainly constitutional, topological, geometrical, hydrophobic, elec- tronic, and steric properties of a total set of 167 AAs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stScales(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stScales_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed average of ST-scales of all the amino acids in the corresponding peptide sequence.
</p>


<h3>References</h3>

<p>Yang, L., Shu, M., Ma, K., Mei, H., Jiang, Y., &amp; Li, Z. (2010). ST-scale as a novel amino acid descriptor and its application in QSAM of peptides and analogues. Amino acids, 38(3), 805-816.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>stScales(seq = "QWGRRCCGWGPGRRYCVRWC")
# [[1]]
# ST1      ST2      ST3      ST4      ST5      ST6      ST7      ST8
# -0.63760  0.07965  0.05150  0.07135 -0.27905 -0.80995  0.58020  0.54400
</code></pre>

<hr>
<h2 id='tScales'>Compute the T-scales of a protein sequence</h2><span id='topic+tScales'></span>

<h3>Description</h3>

<p>T-scales are based on 67 common topological descriptors of 135 amino acids. These topological descriptors are based on the connectivity table of amino acids alone, and to not explicitly consider 3D properties of each structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tScales(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tScales_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed average of T-scales of all the amino acids in the corresponding peptide sequence.
</p>


<h3>References</h3>

<p>Tian F, Zhou P, Li Z: T-scale as a novel vector of topological descriptors for amino acids and its application in QSARs of peptides. J Mol Struct. 2007, 830: 106-115. 10.1016/j.molstruc.2006.07.004.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tScales(seq = "QWGRRCCGWGPGRRYCVRWC")
# [[1]]
# T1      T2      T3      T4      T5
# -3.2700 -0.0035 -0.3855 -0.1475  0.7585
</code></pre>

<hr>
<h2 id='vhseScales'>Compute the VHSE-scales of a protein sequence</h2><span id='topic+vhseScales'></span>

<h3>Description</h3>

<p>VHSE-scales (principal components score Vectors of Hydrophobic, Steric, and Electronic properties), is derived from principal components analysis (PCA) on independent families of 18 hydrophobic properties, 17 steric properties, and 15 electronic properties, respectively, which are included in total 50 physicochemical variables of 20 coded amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vhseScales(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vhseScales_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed average of VHSE-scales of all the amino acids in the corresponding peptide sequence. Each VSHE-scale represent an amino-acid property as follows: </p>

<ul>
<li><p>VHSE1 and VHSE2: Hydrophobic properties
</p>
</li>
<li><p>VHSE3 and VHSE4: Steric properties
</p>
</li>
<li><p>VHSE5 to VHSE8: Electronic properties
</p>
</li></ul>



<h3>References</h3>

<p>Mei, H. U., Liao, Z. H., Zhou, Y., &amp; Li, S. Z. (2005). A new set of amino acid descriptors and its application in peptide QSARs. Peptide Science, 80(6), 775-786.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vhseScales(seq = "QWGRRCCGWGPGRRYCVRWC")
# [[1]]
#  VHSE1   VHSE2   VHSE3   VHSE4   VHSE5   VHSE6   VHSE7   VHSE8 
#-0.1150  0.0630 -0.0055  0.7955  0.4355  0.2485  0.1740 -0.0960

</code></pre>

<hr>
<h2 id='zScales'>Compute the Z-scales of a protein sequence</h2><span id='topic+zScales'></span>

<h3>Description</h3>

<p>Z-scales are based on physicochemical properties of the AAs including NMR data and thin-layer chromatography (TLC) data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zScales(seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zScales_+3A_seq">seq</code></td>
<td>
<p>An amino-acids sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The computed average of Z-scales of all the amino acids in the corresponding peptide sequence. Each Z scale represent an amino-acid property as follows:
</p>

<ul>
<li><p>Z1: Lipophilicity
</p>
</li>
<li><p>Z2: Steric properties (Steric bulk/Polarizability)
</p>
</li>
<li><p>Z3: Electronic properties (Polarity / Charge) 
</p>
</li>
<li><p>Z4 and Z5: They relate electronegativity, heat of formation, electrophilicity and hardness.
</p>
</li></ul>



<h3>References</h3>

<p>Sandberg M, Eriksson L, Jonsson J, Sjostrom M, Wold S: New chemical descriptors relevant for the design of biologically active peptides. A multivariate characterization of 87 amino acids. J Med Chem 1998, 41:2481-2491.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>zScales(seq = "QWGRRCCGWGPGRRYCVRWC")
# [[1]]
#  Z1      Z2      Z3      Z4      Z5 
# 0.6200  0.0865  0.0665  0.7280 -0.8740 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
