<!DOCTYPE html><html lang="en"><head><title>Help for package PERMANOVA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PERMANOVA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AddClusterToBiplot'>
<p>Add Clusters to a Biplot Object</p></a></li>
<li><a href='#BinaryVectorCheck'>
<p>Checks if a vector is binary</p></a></li>
<li><a href='#BiplotVar'>
<p>Draws a variable on a biplot</p></a></li>
<li><a href='#BootDisMANOVA'>
<p>Multivariate Analysis of Variance based on Distances and Bootstrap</p></a></li>
<li><a href='#BootDistCanonicalAnalysis'><p>Canonical Analysis based on Distances</p></a></li>
<li><a href='#Circle2'>
<p>Draws a circle.</p></a></li>
<li><a href='#ConcEllipse'>
<p>Non-parametric concentration ellipses</p></a></li>
<li><a href='#ConstructContrasts'>
<p>Construction of contrasts for several factors</p></a></li>
<li><a href='#Cumsum'>
<p>Cumulative sums</p></a></li>
<li><a href='#DistBinary'>
<p>Distances for binary data</p></a></li>
<li><a href='#DistContinuous'>
<p>Distances among individuals with continuous data</p></a></li>
<li><a href='#Dlines'>
<p>Connects two sets of points by lines</p></a></li>
<li><a href='#FactorToBinary'>
<p>Converts a Factor into its indicator matrix</p></a></li>
<li><a href='#GetScalesBiplot'>
<p>Calculates the scales for the variables on a linear biplot</p></a></li>
<li><a href='#Ginv'>
<p>G inverse</p></a></li>
<li><a href='#inbox'>
<p>Checks if a point is inside a box</p></a></li>
<li><a href='#IniTransform'>
<p>Initial transformation of a data matrix</p></a></li>
<li><a href='#MANOVA'>
<p>Multivariate Analysis of Variance (MANOVA)</p></a></li>
<li><a href='#MANOVA.Estimation'>
<p>Estimation of the MANOVA parameters.</p></a></li>
<li><a href='#mgc'>
<p>Mixture Gaussian Clustering</p></a></li>
<li><a href='#Ones'>
<p>Matrix of ones</p></a></li>
<li><a href='#PERMANOVA'>
<p>PERMANOVA: MANOVA based on distances</p></a></li>
<li><a href='#PERMANOVA.Estimation'>
<p>Estimation of the PERMANOVA parameters</p></a></li>
<li><a href='#PerMANOVA.Simple'>
<p>PERMANOVA from a matrix of distancies</p></a></li>
<li><a href='#plot.BootCanonAnalysis'>
<p>Plots the principal coordinates of the group centers and the bootstrap confidence regions</p></a></li>
<li><a href='#plot.ellipse'>
<p>Plot a concentration ellipse</p></a></li>
<li><a href='#plot.MANOVA'>
<p>Plots the results of a MANOVA Biplot</p></a></li>
<li><a href='#plot.MANOVA.Biplot'>
<p>Plots the results of a MANOVA Biplot</p></a></li>
<li><a href='#plot.PERMANOVA'>
<p>Plots the results of the PERMANOVA function</p></a></li>
<li><a href='#PlotClustersBiplot'>
<p>Plot clusters on a biplot.</p></a></li>
<li><a href='#PostHocComp'>
<p>Post Hoc pairwise comparisons</p></a></li>
<li><a href='#ProcrustesSimple'>
<p>Simple Procrustes Analysis</p></a></li>
<li><a href='#spiders'>
<p>Hunting Spiders Data</p></a></li>
<li><a href='#spidersb'>
<p>Hunting Spiders Data</p></a></li>
<li><a href='#summary.BootDisMANOVA'>
<p>Summarizes the results of a Bootstrap Manova based on distances</p></a></li>
<li><a href='#TextSmart'>
<p>Labels of a Scatter</p></a></li>
<li><a href='#wine'>
<p>Wine data</p></a></li>
<li><a href='#Zeros'>
<p>Matrix of zeros</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Multivariate Analysis of Variance Based on Distances and
Permutations</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Laura Vicente-Gonzalez &lt;laura20vg@usal.es&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Calculates multivariate analysis of variance based on permutations and some associated pictorial representations. The pictorial representation is based on the principal coordinates of the group means. There are some original results that will be published soon. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0), Matrix, xtable, MASS, scales, deldir, grDevices,
graphics, stats</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-06 09:40:32 UTC; joseluis</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-09-06 10:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='AddClusterToBiplot'>
Add Clusters to a Biplot Object
</h2><span id='topic+AddClusterToBiplot'></span>

<h3>Description</h3>

<p>The function add clusters to a biplot object to be represented on the biplot. The clusters can be defined by a nominal variable provided by the user, obtained from the <code>hclust</code> function of the base package or from the <code>kmeans</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AddClusterToBiplot(Bip, NGroups = 3, ClusterType = "hi", Groups = NULL,
Original = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AddClusterToBiplot_+3A_bip">Bip</code></td>
<td>

<p>A Biplot object obtained from any biplot procedure. It has to be a list containing a field called <code>Bip$RowCoordinates</code> in order to calculate the clusters when necessary.
</p>
</td></tr>
<tr><td><code id="AddClusterToBiplot_+3A_ngroups">NGroups</code></td>
<td>

<p>Number of groups or clusters. Only necessary when hierarchical or k-means procedures are used.
</p>
</td></tr>
<tr><td><code id="AddClusterToBiplot_+3A_clustertype">ClusterType</code></td>
<td>

<p>The type of cluster to add. There are three possibilities &quot;us&quot; (User Defined), &quot;hi&quot; (hierarchical clusters),  &quot;km&quot; (kmeans clustering) or &quot;gm&quot; (gaussian mixture).
</p>
</td></tr>
<tr><td><code id="AddClusterToBiplot_+3A_groups">Groups</code></td>
<td>

<p>A factor defining the groups provided by the user.
</p>
</td></tr>
<tr><td><code id="AddClusterToBiplot_+3A_original">Original</code></td>
<td>

<p>Should the clusters be calculated using the original data rather than the reduced dimensions?
</p>
</td></tr>
<tr><td><code id="AddClusterToBiplot_+3A_...">...</code></td>
<td>

<p>Any other parameter for the <code>hclust</code> and <code>kmeans</code> procedures.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>One of the main shortcomings of cluster analysis is that it is not easy to search for the variables associated to the obtained classification; representing the clusters on the biplot can help to perform that interpretation. If you consider the technique for dimension reduction as a way to separate the signal from the noise, clusters should be constructed using the dimensions retained in the biplot, otherwise the complete original data matrix can be used. The colors used by each cluster should match the color used in the Dendrogram.
User defined clusters can also be plotted, for example, to investigate the relation of the biplot solution to an external nominal variable.
</p>


<h3>Value</h3>

<p>The function returns the biplot object with the information about the clusters added in new fields
</p>
<table role = "presentation">
<tr><td><code>ClusterType</code></td>
<td>
<p>The method of clustering as defined in the argument <code>ClusterType</code>.</p>
</td></tr>
<tr><td><code>Clusters</code></td>
<td>
<p>A factor containing the solution or the user defined clusters</p>
</td></tr>
<tr><td><code>ClusterNames</code></td>
<td>
<p>The names of the clusters</p>
</td></tr>
<tr><td><code>ClusterColors</code></td>
<td>
<p>The colors of the clusters</p>
</td></tr>
<tr><td><code>Dendrogram</code></td>
<td>
<p>The Dendrogram if we have used hirarchical clustering</p>
</td></tr>
<tr><td><code>ClusterObject</code></td>
<td>
<p>The object obtained from <code>hclust</code>, <code>kmeans</code> or <code>MGC</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon, Laura Vicente-Gonzalez
</p>


<h3>References</h3>

<p>Demey, J. R., Vicente-Villardon, J. L., Galindo-Villardon, M. P., &amp; Zambrano, A. Y. (2008). Identifying molecular markers associated with classification of genotypes by External Logistic Biplots. Bioinformatics, 24(24), 2832-2838.
</p>
<p>Gallego-Alvarez, I., &amp; Vicente-Villardon, J. L. (2012). Analysis of environmental indicators in international companies by applying the logistic biplot. Ecological Indicators, 23, 250-261.
</p>
<p>Galindo, P. V., Vaz, T. D. N., &amp; Nijkamp, P. (2011). Institutional capacity to dynamically innovate: an application to the Portuguese case. Technological Forecasting and Social Change, 78(1), 3-12.
</p>
<p>Vazquez-de-Aldana, B. R., Garcia-Criado, B., Vicente-Tavera, S., &amp; Zabalgogeazcoa, I. (2013). Fungal Endophyte (Epichloë festucae) Alters the Nutrient Content of Festuca rubra Regardless of Water Availability. PloS one, 8(12), e84539.
</p>


<h3>See Also</h3>

<p>For clusters not provided by the user the function uses the standard procedures in  <code>hclust</code> and <code>kmeans</code>.
</p>

<hr>
<h2 id='BinaryVectorCheck'>
Checks if a vector is binary
</h2><span id='topic+BinaryVectorCheck'></span>

<h3>Description</h3>

<p>Checks if all the entries of a vector are 0 or 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BinaryVectorCheck(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BinaryVectorCheck_+3A_x">x</code></td>
<td>

<p>The vector to check
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The logical result.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x=c(0, 0, 0, 0,  1, 1, 1, 2)
BinaryVectorCheck(x)
</code></pre>

<hr>
<h2 id='BiplotVar'>
Draws a variable on a biplot
</h2><span id='topic+BiplotVar'></span>

<h3>Description</h3>

<p>Draws a continuous variable on a biplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BiplotVar(bi1, bi2, b0 = 0, xmin = -3, xmax = 3, ymin = -3, ymax = 3,
label = "Point", mode = "a", CexPoint = 0.8, PchPoint = 1, Color = "blue",
ticks = c(-3, -2.5, -2, -1.5, -1, -0.5, 0.5, 1, 1.5, 2, 2.5, 3), ticklabels =
round(ticks, digits = 2), tl = 0.03, ts = "Complete", Position = "Angle",
AddArrow = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BiplotVar_+3A_bi1">bi1</code></td>
<td>

<p>First component of the direction vector.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_bi2">bi2</code></td>
<td>

<p>Second component of the direction vector.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_b0">b0</code></td>
<td>

<p>Constant for the regression adjusted biplots.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_xmin">xmin</code></td>
<td>

<p>Minimum value of the x axis.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_xmax">xmax</code></td>
<td>

<p>Maximum value of the x axis.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_ymin">ymin</code></td>
<td>

<p>Minimum value of the y axis.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_ymax">ymax</code></td>
<td>

<p>Maximum value of the y axis.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_label">label</code></td>
<td>

<p>Label of the variable.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_mode">mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;h&quot;, &quot;ah&quot; and &quot;s&quot;.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_cexpoint">CexPoint</code></td>
<td>

<p>Size for the symbols and labels of the variables.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_pchpoint">PchPoint</code></td>
<td>

<p>Symbols for the variable (when represented as a point).
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_color">Color</code></td>
<td>

<p>Color for the variable.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_ticks">ticks</code></td>
<td>

<p>Ticks when the variable is represented as a graded scale.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_ticklabels">ticklabels</code></td>
<td>

<p>Labels for the ticks when the variable is represented as a graded scale.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_tl">tl</code></td>
<td>

<p>Thick length.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_ts">ts</code></td>
<td>

<p>Size of the mark in the graded  scale.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_position">Position</code></td>
<td>

<p>If the Position is &quot;Angle&quot; the label of the variable is placed using the angle of the vector.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_addarrow">AddArrow</code></td>
<td>

<p>Add an arrow to the representation of other modes of the biplot.
</p>
</td></tr>
<tr><td><code id="BiplotVar_+3A_...">...</code></td>
<td>

<p>Any other graphical parameters
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See plot.PCA.Biplot
</p>


<h3>Value</h3>

<p>No value returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>

<hr>
<h2 id='BootDisMANOVA'>
Multivariate Analysis of Variance based on Distances and Bootstrap
</h2><span id='topic+BootDisMANOVA'></span>

<h3>Description</h3>

<p>Multivariate Analysis of variance based on distances and Bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BootDisMANOVA(Distance, groups, C = NULL, Effects = NULL, nB = 1000, seed = NULL,
CoordPrinc = FALSE, dimens = 2, PCoA = "Standard", ProjectInd = TRUE, tol = 1e-04,
DatosIni = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BootDisMANOVA_+3A_distance">Distance</code></td>
<td>

<p>A matrix of distances.
</p>
</td></tr>
<tr><td><code id="BootDisMANOVA_+3A_groups">groups</code></td>
<td>

<p>A factor containing the groups to compare.
</p>
</td></tr>
<tr><td><code id="BootDisMANOVA_+3A_c">C</code></td>
<td>

<p>A matrix of contrasts (if null the identity is used).
</p>
</td></tr>
<tr><td><code id="BootDisMANOVA_+3A_effects">Effects</code></td>
<td>

<p>A vector of effects.
</p>
</td></tr>
<tr><td><code id="BootDisMANOVA_+3A_nb">nB</code></td>
<td>

<p>Number of Bootstrap replicates.
</p>
</td></tr>
<tr><td><code id="BootDisMANOVA_+3A_seed">seed</code></td>
<td>

<p>Seed for the random numbers.
</p>
</td></tr>
<tr><td><code id="BootDisMANOVA_+3A_coordprinc">CoordPrinc</code></td>
<td>

<p>Should Principal Coordinates be calculated.
</p>
</td></tr>
<tr><td><code id="BootDisMANOVA_+3A_dimens">dimens</code></td>
<td>

<p>Dimension of the solution.
</p>
</td></tr>
<tr><td><code id="BootDisMANOVA_+3A_pcoa">PCoA</code></td>
<td>

<p>Type of Principal Coordinates to calulate.
</p>
</td></tr>
<tr><td><code id="BootDisMANOVA_+3A_projectind">ProjectInd</code></td>
<td>

<p>Should the individuals be projected onto the graph.
</p>
</td></tr>
<tr><td><code id="BootDisMANOVA_+3A_tol">tol</code></td>
<td>

<p>Tolerance for convergence of the algorithms.
</p>
</td></tr>
<tr><td><code id="BootDisMANOVA_+3A_datosini">DatosIni</code></td>
<td>

<p>Should the initial data be included in the results.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Multivariate Analysis of Variance based on distances and Bootstrap.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>Function</p>
</td></tr>
<tr><td><code>Title</code></td>
<td>
<p>Title of the study</p>
</td></tr>
<tr><td><code>Type</code></td>
<td>
<p>BootMANOVA</p>
</td></tr>
<tr><td><code>Distances</code></td>
<td>
<p>A matrix containing the distances between individuals.</p>
</td></tr>
<tr><td><code>C</code></td>
<td>
<p>Contrasts Matrix.</p>
</td></tr>
<tr><td><code>Initial</code></td>
<td>
<p> Containing two matrices:
</p>
<p>* Global -&gt; Global contrast.
</p>
<p>* Contrastes -&gt;Contrasts for groups.
</p>
</td></tr>
<tr><td><code>DistMuestral</code></td>
<td>
<p>Sample distribution of F-exp from permutations.</p>
</td></tr>
<tr><td><code>pvalue</code></td>
<td>
<p>Estimate p-valor for PERMANOVA.</p>
</td></tr>
<tr><td><code>ExplainedVariance</code></td>
<td>
<p>Explained variance by Principal Coordinates selected.</p>
</td></tr>
<tr><td><code>Inertias</code></td>
<td>
<p>Eigenvalue, Explained variance, Cumulative explained variance.</p>
</td></tr>
<tr><td><code>MeanCoordinates</code></td>
<td>
<p>Mean Coordinates by groups for the dimensions obtained in the Principal Coordinates Analysis.</p>
</td></tr>
<tr><td><code>Qualities</code></td>
<td>
<p>Qualities representation by groups for the dimensions of PCoA.</p>
</td></tr>
<tr><td><code>CummulativeQualities</code></td>
<td>
<p>Cummulative qualities representation.</p>
</td></tr>
<tr><td><code>ClusterType</code></td>
<td>
<p>Cluster type selected.</p>
</td></tr>
<tr><td><code>Clusters</code></td>
<td>
<p>Clusters created.</p>
</td></tr>
<tr><td><code>ClusterNames</code></td>
<td>
<p>Names of clusters</p>
</td></tr>
<tr><td><code>ClusterColors</code></td>
<td>
<p>Colors of clusters, color name and HTML code.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Anderson, M. J. (2001). A new method for non-parametric multivariate analysis of variance. Austral ecology, 26(1):32–46.
</p>
<p>Anderson, M. J. (2005). Permanova: a fortran computer program for permutational multivariate analysis of variance. Department of Statistics, University of Auckland, New Zealand, 24.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X = wine[,4:21]
D = DistContinuous (X)
bootwine=BootDisMANOVA(D, wine$Group)
bootwine

</code></pre>

<hr>
<h2 id='BootDistCanonicalAnalysis'>Canonical Analysis based on Distances
</h2><span id='topic+BootDistCanonicalAnalysis'></span>

<h3>Description</h3>

<p>Canonical Analysis based on distances. Confidence Regions for the mean vectors are calculated using bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BootDistCanonicalAnalysis(Distance, groups, dimens = NULL, nB = 100, seed = NULL,
PCoA = "Standard", ProcrustesRot = TRUE, DatosIni = TRUE, tol = 1e-04)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BootDistCanonicalAnalysis_+3A_distance">Distance</code></td>
<td>

<p>A list of three elements containing the data, the distances between individuals and type of distance used.
</p>
</td></tr>
<tr><td><code id="BootDistCanonicalAnalysis_+3A_groups">groups</code></td>
<td>

<p>A factor containing the groups to compare.
</p>
</td></tr>
<tr><td><code id="BootDistCanonicalAnalysis_+3A_dimens">dimens</code></td>
<td>

<p>Number of dimensions to choose . By default is 2.
</p>
</td></tr>
<tr><td><code id="BootDistCanonicalAnalysis_+3A_nb">nB</code></td>
<td>

<p>Number of Bootstrap samples.
</p>
</td></tr>
<tr><td><code id="BootDistCanonicalAnalysis_+3A_seed">seed</code></td>
<td>

<p>Seed for the generation of the random samples (Added for reproducibility)
</p>
</td></tr>
<tr><td><code id="BootDistCanonicalAnalysis_+3A_pcoa">PCoA</code></td>
<td>

<p>The type of Principal Coordinates Analysis.
</p>
<p>There are two possibilities:
</p>
<p>* Standard
</p>
<p>* Weighted
</p>
<p>By default is Standard.
</p>
</td></tr>
<tr><td><code id="BootDistCanonicalAnalysis_+3A_procrustesrot">ProcrustesRot</code></td>
<td>

<p>Should Procrustes rotation be applied to the configurations?
</p>
</td></tr>
<tr><td><code id="BootDistCanonicalAnalysis_+3A_datosini">DatosIni</code></td>
<td>

<p>The input object contains the initial data?. By default is TRUE.
</p>
</td></tr>
<tr><td><code id="BootDistCanonicalAnalysis_+3A_tol">tol</code></td>
<td>

<p>Tolerance
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates a Canonical Analysis based on distance matrices with confidence regions based on bootstrap resampling.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>D</code></td>
<td>
<p>The distance matrix used for calculations</p>
</td></tr>
<tr><td><code>Coefficient</code></td>
<td>
<p>Dissimilarity  coefficient used to calculate the proximities</p>
</td></tr>
<tr><td><code>nB</code></td>
<td>
<p>Number of bootstrap samples</p>
</td></tr>
<tr><td><code>Groups</code></td>
<td>
<p>Factor containing the groups to compare</p>
</td></tr>
<tr><td><code>GroupNames</code></td>
<td>
<p>Names of the groups</p>
</td></tr>
<tr><td><code>Inertia</code></td>
<td>
<p>Variance accounted for ech canonical coordinate</p>
</td></tr>
<tr><td><code>Inertias</code></td>
<td>
<p>able with the eigenvalue, variance accounted and cumulated variance</p>
</td></tr>
<tr><td><code>MeanCoordinates</code></td>
<td>
<p>Coordinates of the groups in the representation</p>
</td></tr>
<tr><td><code>Qualities</code></td>
<td>
<p>Qualities of the representation of the group means</p>
</td></tr>
<tr><td><code>CumulativeQualities</code></td>
<td>
<p>Cumulative Qualities of the representation</p>
</td></tr>
<tr><td><code>CoordBoot</code></td>
<td>
<p>Coordinates of the bootstrap replicates</p>
</td></tr>
<tr><td><code>RowCoordinates</code></td>
<td>
<p>Coordinates of the individuals</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Gower, J. C. and Krzanowski, W. J. (1999). Analysis of distance for structured multivariate
data and extensions to multivariate analysis of variance. Journal of the Royal
Statistical Society: Series C (Applied Statistics), 48(4):505–519.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X=wine[,4:21]
Dist=DistContinuous(X)
canon=BootDistCanonicalAnalysis(Dist, groups=wine$Group, nB=10)
</code></pre>

<hr>
<h2 id='Circle2'>
Draws a circle.
</h2><span id='topic+Circle2'></span>

<h3>Description</h3>

<p>Draws a circle for a given radius at the specified center with the given color.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Circle2(radius = 1, origin = c(0, 0), col = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Circle2_+3A_radius">radius</code></td>
<td>

<p>Radius of the circle.
</p>
</td></tr>
<tr><td><code id="Circle2_+3A_origin">origin</code></td>
<td>

<p>Centre of the circle.
</p>
</td></tr>
<tr><td><code id="Circle2_+3A_col">col</code></td>
<td>

<p>Color of the circle.
</p>
</td></tr>
<tr><td><code id="Circle2_+3A_...">...</code></td>
<td>

<p>Additional graphical parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Draws a circle for a given radius at the specified center with the given color.
</p>


<h3>Value</h3>

<p>No value is returned.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot(0,0)
Circle2(1,c(0,0))
</code></pre>

<hr>
<h2 id='ConcEllipse'>
Non-parametric concentration ellipses
</h2><span id='topic+ConcEllipse'></span>

<h3>Description</h3>

<p>Calculates non-parametric concentration ellipses for a two-dimensional set of points
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConcEllipse(data, center = NULL, confidence = 1, npoints = 100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ConcEllipse_+3A_data">data</code></td>
<td>

<p>A two dimensional set of points.
</p>
</td></tr>
<tr><td><code id="ConcEllipse_+3A_center">center</code></td>
<td>

<p>Center of the data.
</p>
</td></tr>
<tr><td><code id="ConcEllipse_+3A_confidence">confidence</code></td>
<td>

<p>Confidence for the concentration ellipse.
</p>
</td></tr>
<tr><td><code id="ConcEllipse_+3A_npoints">npoints</code></td>
<td>

<p>Number of points to plot.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates non-parametric concentration ellipses for a two-dimensional set of points using distances to the center.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>data</code></td>
<td>
<p>Original data</p>
</td></tr>
<tr><td><code>confidence</code></td>
<td>
<p>Percent of points selected for the ellipse</p>
</td></tr>
<tr><td><code>ellipse</code></td>
<td>
<p>points to plot the ellipse</p>
</td></tr>
<tr><td><code>center</code></td>
<td>
<p>center of the ellipse</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Meulman, J. J., &amp; Heiser, W. J. (1983). The display of bootstrap solutions in multidimensional scaling. Murray Hill, NJ: Bell Laboratories.
</p>
<p>Linting, M., Meulman, J. J., Groenen, P. J., &amp; Van der Kooij, A. J. (2007). Stability of nonlinear principal components analysis: An empirical study using the balanced bootstrap. Psychological Methods, 12(3), 359.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x=runif(30)
y=x+rnorm(30)*0.3
plot(x, y)
el=ConcEllipse(cbind(x,y), confidence=0.95)
plot(el)
</code></pre>

<hr>
<h2 id='ConstructContrasts'>
Construction of contrasts for several factors
</h2><span id='topic+ConstructContrasts'></span>

<h3>Description</h3>

<p>Constructs the contrasts for main effects and interaction for a dataframe of factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConstructContrasts(Factors, MaxOrderIter = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ConstructContrasts_+3A_factors">Factors</code></td>
<td>

<p>A data frame of factors (For general data frames the functions extracts all the columns that are factors).
</p>
</td></tr>
<tr><td><code id="ConstructContrasts_+3A_maxorderiter">MaxOrderIter</code></td>
<td>

<p>Maximum order of the interaction to construct.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Constructs the contrasts for main effects and interaction for a dataframe of factors. For a general data frame the functions extracts all the factors first. The function constructs the contrasts for all the main effects and the interactions up to order MaxOrderIter.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Groups</code></td>
<td>
<p>The groups resulting of combining the levels of all the factors.</p>
</td></tr>
<tr><td><code>Contrasts</code></td>
<td>
<p>A matrix of contrasts containing the contrasts that are associated to the main affects and interaction. The rows contain contrasts and the columns the groups to combine.</p>
</td></tr>
<tr><td><code>Effects</code></td>
<td>
<p>A factor describing which contrasts correspond to each effect.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon, Laura Vicente-Gonzalez
</p>


<h3>References</h3>

<p>Bapat, R. B. (2012). Linear algebra and linear models. Springer Science &amp; Business Media.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ConstructContrasts(wine[,1:2])
</code></pre>

<hr>
<h2 id='Cumsum'>
Cumulative sums
</h2><span id='topic+Cumsum'></span>

<h3>Description</h3>

<p>Cumulative sums.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Cumsum(X, dimens = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Cumsum_+3A_x">X</code></td>
<td>

<p>Data Matrix.
</p>
</td></tr>
<tr><td><code id="Cumsum_+3A_dimens">dimens</code></td>
<td>

<p>Dimension for summing.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Cumulative sums within rows (dimens=1) or columns (dimens=2) of a data matrix.
</p>


<h3>Value</h3>

<p>A matrix of the same size as X with cumulative sums within each row or each column.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X=wine[,4:21]
Cumsum(X,1)
Cumsum(X,2)
</code></pre>

<hr>
<h2 id='DistBinary'>
Distances for binary data
</h2><span id='topic+DistBinary'></span>

<h3>Description</h3>

<p>Calculates distancies among individuals for binary data. It is possible introduce two matrices (x, y) and calculate the distancies between them or introduce only one matrix (x) and calculate the distancies among individuals in the single matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DistBinary(x, y = NULL, coefficient = "Simple_Matching", transformation = "sqrt(1-S)")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DistBinary_+3A_x">x</code></td>
<td>

<p>A matrix containing binary data.
</p>
</td></tr>
<tr><td><code id="DistBinary_+3A_y">y</code></td>
<td>

<p>A matrix containing binary data different than x. By default it is null because if we want calculate the distancies between individuals it's not necessary.
</p>
</td></tr>
<tr><td><code id="DistBinary_+3A_coefficient">coefficient</code></td>
<td>

<p>Similarity coefficients for binary data.By default it is Simple_Matching. See details.
</p>
</td></tr>
<tr><td><code id="DistBinary_+3A_transformation">transformation</code></td>
<td>

<p>Transformations of similarities into distances. By default it is sqrt(1-S). See details.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates coefficients of similarity (among individuals) for binary data and converts them into measures of distance.
</p>
<p>We have 17 different coefficients (numbers instead of names can be used as arguments of the function) :
</p>
<p>* 1.- Kulezynski
</p>
<p>* 2.- Russell_and_Rao
</p>
<p>* 3.- Jaccard
</p>
<p>* 4.- Simple_Matching
</p>
<p>* 5.- Anderberg
</p>
<p>* 6.- Rogers_and_Tanimoto
</p>
<p>* 7.- Sorensen_Dice_and_Czekanowski
</p>
<p>* 8.- Sneath_and_Sokal
</p>
<p>* 9.- Hamman
</p>
<p>* 10.- Kulezynski2
</p>
<p>* 11.- Anderberg2
</p>
<p>* 12.- Ochiai
</p>
<p>* 13.- S13
</p>
<p>* 14.- Pearson_phi
</p>
<p>* 15.- Yule
</p>
<p>* 16.- Sorensen
</p>
<p>* 17.- Dice
</p>
<p>There are nine possible transformations of the initial similarities:
</p>
<p>* 1.- Identity
</p>
<p>* 2.- 1-S
</p>
<p>* 3.- sqrt(1-S)
</p>
<p>* 4.- (-log(S))
</p>
<p>* 5.- (1/S-1)
</p>
<p>* 6.- (sqrt(2(1-S)))
</p>
<p>* 7.- (1-(S+1)/2)
</p>
<p>* 8.- 1-abs(S)
</p>
<p>* 9.- 1/(S+1)
</p>


<h3>Value</h3>

<p>The DistBinary function creates a list that return:
</p>
<table role = "presentation">
<tr><td><code>Data</code></td>
<td>
<p>A matrix with initial data.</p>
</td></tr>
<tr><td><code>D</code></td>
<td>
<p>A matrix containing binary distancies.</p>
</td></tr>
<tr><td><code>Coefficient</code></td>
<td>
<p>Character containing the name of the coefficient used.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Gower, J. C. (2006) Similarity dissimilarity and Distance, measures of. Encyclopedia of Statistical Sciences. 2nd. ed. Volume 12. Wiley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spidersb)
DD=DistBinary(spidersb)
</code></pre>

<hr>
<h2 id='DistContinuous'>
Distances among individuals with continuous data
</h2><span id='topic+DistContinuous'></span>

<h3>Description</h3>

<p>Calculate distances among individuals for continuous data. It is possible introduce two matrices (x, y) and calculate the distances between the two sets of rows or introduce only one matrix (x) and calculate the distances between its rows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DistContinuous(x, y = NULL, coef = "Pythagorean", r = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DistContinuous_+3A_x">x</code></td>
<td>

<p>A matrix containing continuous data.
</p>
</td></tr>
<tr><td><code id="DistContinuous_+3A_y">y</code></td>
<td>

<p>A matrix containing continuous data different from x. By default it is null.
</p>
</td></tr>
<tr><td><code id="DistContinuous_+3A_coef">coef</code></td>
<td>

<p>Coefficient to calculate continuous distances. By default we use Pythagorean distances.
</p>
</td></tr>
<tr><td><code id="DistContinuous_+3A_r">r</code></td>
<td>

<p>For Minkowski distances. By default it is 1.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates distances among individuals for matrices of continuous data using different distance measures. If two matrices are provided, distances among individuals, one from the first matrix and another from the second, are calculated. If only one matrix is provided, interdistances among its rows are calculated.
</p>
<p>The following coefficients are calculated
</p>
<p>1.- Pythagorean =  sqrt(sum((y[i, ] - x[j, ])^2)/p)
</p>
<p>2.- Taxonomic =  sqrt(sum(((y[i,]-x[j,])^2)/r^2)/p)
</p>
<p>3.- City = sum(abs(y[i,]-x[j,])/r)/p
</p>
<p>4.- Minkowski = (sum((abs(y[i,]-x[j,])/r)^t)/p)^(1/t)
</p>
<p>5.- Divergence = sqrt(sum((y[i,]-x[j,])^2/(y[i,]+x[j,])^2)/p)
</p>
<p>6.- dif_sum = sum(abs(y[i,]-x[j,])/abs(y[i,]+x[j,]))/p
</p>
<p>7.- Camberra = sum(abs(y[i,]-x[j,])/(abs(y[i,])+abs(x[j,])))
</p>
<p>8.- Bray_Curtis = sum(abs(y[i,]-x[j,]))/sum(y[i,]+x[j,])
</p>
<p>9.- Soergel = sum(abs(y[i,]-x[j,]))/sum(apply(rbind(y[i,],x[j,]),2,max))
</p>
<p>10.- Ware_hedges = sum(abs(y[i,]-x[j,]))/sum(apply(rbind(y[i,],x[j,]),2,max))
</p>


<h3>Value</h3>

<p>A list with:
</p>
<table role = "presentation">
<tr><td><code>Data</code></td>
<td>
<p>A matrix with the initial data (x matrix).</p>
</td></tr>
<tr><td><code>SupData</code></td>
<td>
<p>A matrix with the supplementary data (y matrix).</p>
</td></tr>
<tr><td><code>D</code></td>
<td>
<p>The matrix of distances.</p>
</td></tr>
<tr><td><code>Coefficient</code></td>
<td>
<p>The coefficient used.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Gower, J. C. (2006) Similarity dissimilarity and Distance, measures of. Encyclopedia of Statistical Sciences. 2nd. ed. Volume 12. Wiley
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X = wine[,4:21]
D=DistContinuous(X)
</code></pre>

<hr>
<h2 id='Dlines'>
Connects two sets of points by lines
</h2><span id='topic+Dlines'></span>

<h3>Description</h3>

<p>Connects two sets of points by lines in a rowwise manner. Adapted from Graffelman(2013).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dlines(SetA, SetB, lin = "dotted", color = "black", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Dlines_+3A_seta">SetA</code></td>
<td>

<p>First set of points.
</p>
</td></tr>
<tr><td><code id="Dlines_+3A_setb">SetB</code></td>
<td>

<p>Second set of points.
</p>
</td></tr>
<tr><td><code id="Dlines_+3A_lin">lin</code></td>
<td>

<p>Line style.
</p>
</td></tr>
<tr><td><code id="Dlines_+3A_color">color</code></td>
<td>

<p>Line color.
</p>
</td></tr>
<tr><td><code id="Dlines_+3A_...">...</code></td>
<td>

<p>Any other graphical parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Connects two sets of points by lines.
</p>


<h3>Value</h3>

<p>NULL
</p>


<h3>Author(s)</h3>

<p>Based on Graffelman (2013)
</p>


<h3>References</h3>

<p>Jan Graffelman (2013). calibrate: Calibration of Scatterplot and Biplot Axes. R package version 1.7.2.
http://CRAN.R-project.org/package=calibrate
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## No examples
</code></pre>

<hr>
<h2 id='FactorToBinary'>
Converts a Factor into its indicator matrix
</h2><span id='topic+FactorToBinary'></span>

<h3>Description</h3>

<p>Converts a factor into a binary matrix with as many columns as categories of the factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FactorToBinary(y, Name = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FactorToBinary_+3A_y">y</code></td>
<td>

<p>A factor.
</p>
</td></tr>
<tr><td><code id="FactorToBinary_+3A_name">Name</code></td>
<td>

<p>Name to use in the final matrix.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An indicator binary matrix.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>y=factor(c(1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 2, 2, 2, 1, 1, 1))
FactorToBinary(y)
</code></pre>

<hr>
<h2 id='GetScalesBiplot'>
Calculates the scales for the variables on a linear biplot
</h2><span id='topic+GetScalesBiplot'></span>

<h3>Description</h3>

<p>Calculates the scales for the variables on a linear prediction biplot.
There are several types of scales and values that can be shown on the graphical representation. See details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetScalesBiplot(Biplot, nticks = 4, TypeScale = "Complete", ValuesScale = "Original")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetScalesBiplot_+3A_biplot">Biplot</code></td>
<td>

<p>Object of class PCA.Biplot.
</p>
</td></tr>
<tr><td><code id="GetScalesBiplot_+3A_nticks">nticks</code></td>
<td>

<p>Number of ticks for the biplot axes.
</p>
</td></tr>
<tr><td><code id="GetScalesBiplot_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scale to use : &quot;Complete&quot;, &quot;StdDev&quot; or &quot;BoxPlot&quot;.
</p>
</td></tr>
<tr><td><code id="GetScalesBiplot_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>Values to show on the scale: &quot;Original&quot; or &quot;Transformed&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function calculates the points on the biplot axes where the scales should be placed.
</p>
<p>There are three types of scales when the transformations of the raw data are made by columns:
</p>
<p>&quot;Complete&quot;: Covers the whole range of the variable using the number of ticks specified in &quot;nticks&quot;. A smaller number of points could be shown if some fall outside the range of the scatter.
</p>
<p>&quot;StdDev&quot;: The mean +/- 1, 2 and 3 times the standard deviation.A smaller number of points could be shown if some fall outside the range of the scatter.
</p>
<p>&quot;BoxPlot&quot;: Median, 25, 75 percentiles maximum and minimum values are shown. The extremes of the interquartile range are connected with a thicker line. A smaller number of points could be shown if some fall outside the range of the scatter.
</p>
<p>There are two kinds of values that can be shown on the biplot axis:
</p>
<p>&quot;Original&quot;: The values before transformation. Only makes sense when the transformations are for each column.
</p>
<p>&quot;Transformed&quot;: The values after transformation, for example, after standardization.
</p>
<p>Although the function is public, the end used will not normally use it.
</p>


<h3>Value</h3>

<p>A list with the following components:
</p>
<table role = "presentation">
<tr><td><code>Ticks</code></td>
<td>
<p>A list containing the ticks for each variable.</p>
</td></tr>
<tr><td><code>Labels</code></td>
<td>
<p>A list containing the labels for each variable.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>

<hr>
<h2 id='Ginv'>
G inverse
</h2><span id='topic+Ginv'></span>

<h3>Description</h3>

<p>Calculates the g-inverse of a squared matrix using the eigen decomposition and removing the eigenvalues smaller than a tolerance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ginv(X, tol = sqrt(.Machine$double.eps))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Ginv_+3A_x">X</code></td>
<td>

<p>Matrix to calculate the g-inverse.
</p>
</td></tr>
<tr><td><code id="Ginv_+3A_tol">tol</code></td>
<td>

<p>Tolerance.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is useful to avoid singularities.
</p>


<h3>Value</h3>

<p>Returns the g-inverse.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
x=as.matrix(iris[,1:4])
S= t(x) 
Ginv(S)
</code></pre>

<hr>
<h2 id='inbox'>
Checks if a point is inside a box
</h2><span id='topic+inbox'></span>

<h3>Description</h3>

<p>Checks if a point is inside a box. The point is specified bi its x and y coordinates and the box with the minimum and maximum values on both coordinate axis: xmin, xmax, ymin, ymax. The vertices of the box are then (xmin, ymin), (xmax, ymin), (xmax, ymax) and (xmin, ymax).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inbox(x, y, xmin, xmax, ymin, ymax)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inbox_+3A_x">x</code></td>
<td>

<p>x coordinate of the point.
</p>
</td></tr>
<tr><td><code id="inbox_+3A_y">y</code></td>
<td>

<p>y coordinate of the point.
</p>
</td></tr>
<tr><td><code id="inbox_+3A_xmin">xmin</code></td>
<td>

<p>Minimum value of X.
</p>
</td></tr>
<tr><td><code id="inbox_+3A_xmax">xmax</code></td>
<td>

<p>Maximum value of X.
</p>
</td></tr>
<tr><td><code id="inbox_+3A_ymin">ymin</code></td>
<td>

<p>Minimum value of Y.
</p>
</td></tr>
<tr><td><code id="inbox_+3A_ymax">ymax</code></td>
<td>

<p>Maximum value of Y.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a logical value: TRUE if the point is inside the box and FALSE otherwise.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>inbox(0, 0, -1, 1, -1, 1)
</code></pre>

<hr>
<h2 id='IniTransform'>
Initial transformation of a data matrix
</h2><span id='topic+IniTransform'></span>

<h3>Description</h3>

<p>Initial transformation of data before the construction of a biplot (or any other technique).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IniTransform(X, InitTransform = "None", transform = "Standardize columns")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="IniTransform_+3A_x">X</code></td>
<td>

<p>Original Raw Data Matrix.
</p>
</td></tr>
<tr><td><code id="IniTransform_+3A_inittransform">InitTransform</code></td>
<td>

<p>Previous transformation (Logarithm or logit).
</p>
</td></tr>
<tr><td><code id="IniTransform_+3A_transform">transform</code></td>
<td>

<p>Transformation to use. See details.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Possible Transformations are:
</p>
<p>1.- &quot;Raw Data&quot;: When no transformation is required.
</p>
<p>2.- &quot;Subtract the global mean&quot;: Eliminate an effect common to all the observations
</p>
<p>3.- &quot;Double centering&quot; : Interaction residuals. When all the elements of the table are comparable. Useful for AMMI models.
</p>
<p>4.- &quot;Column centering&quot;: Remove the column means.
</p>
<p>5.- &quot;Standardize columns&quot;:  Remove the column means and divide by its standard deviation.
</p>
<p>6.- &quot;Row centering&quot;: Remove the row means.
</p>
<p>7.- &quot;Standardize rows&quot;: Divide each row by its standard deviation.
</p>
<p>8.- &quot;Divide by the column means and center&quot;: The resulting dispersion is the coefficient of variation.
</p>
<p>9.- &quot;Normalized residuals from independence&quot; for a contingency table.
</p>
<p>The transformation can be provided to the function by using the string between the quotes or just the associated number.
</p>
<p>The supplementary rows and columns are not used to calculate the parameters (means, standard deviations, etc). Some of the transformations are not compatible with supplementary data.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>X</code></td>
<td>
<p>Transformed data matrix.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>M. J. Baxter (1995) Standardization and Transformation in Principal Component Analysis, with Applications to Archaeometry.
Journal of the Royal Statistical Society. Series C (Applied Statistics). Vol. 44, No. 4 (1995) , pp. 513-527
</p>
<p>Kroonenberg, P. M. (1983). Three-mode principal component analysis: Theory and applications (Vol. 2). DSWO press. (Chapter 6)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
x=as.matrix(iris[,1:4])
x=IniTransform(x, transform=4)
x
</code></pre>

<hr>
<h2 id='MANOVA'>
Multivariate Analysis of Variance (MANOVA)
</h2><span id='topic+MANOVA'></span>

<h3>Description</h3>

<p>Performs a Multivariate Analysis of Variance (MANOVA) based on matrix calculations. Is an extension of the function in the base package of R.
</p>
<p>The function can take care of matrices of contrasts and matrices of linear combinations of variables in order to cope with complex designs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MANOVA(Y, Group, C = NULL, M = NULL, Effects = NULL, InitialTransform = 5,
AddOnes = FALSE, Contrasts = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MANOVA_+3A_y">Y</code></td>
<td>

<p>A matrix containing the response variables.
</p>
</td></tr>
<tr><td><code id="MANOVA_+3A_group">Group</code></td>
<td>

<p>A factor containing the groups to compare (or the treatments).
</p>
</td></tr>
<tr><td><code id="MANOVA_+3A_c">C</code></td>
<td>

<p>A matrix of contrasts on the groups.
</p>
</td></tr>
<tr><td><code id="MANOVA_+3A_m">M</code></td>
<td>

<p>A matrix with the linear combinations of the variables to test.
</p>
</td></tr>
<tr><td><code id="MANOVA_+3A_effects">Effects</code></td>
<td>

<p>A factor with the effects in the rows of C.
</p>
</td></tr>
<tr><td><code id="MANOVA_+3A_initialtransform">InitialTransform</code></td>
<td>

<p>Initial transformation of the response variables.
</p>
</td></tr>
<tr><td><code id="MANOVA_+3A_addones">AddOnes</code></td>
<td>

<p>Add a column of ones to the design matrix. By default is false.
</p>
</td></tr>
<tr><td><code id="MANOVA_+3A_contrasts">Contrasts</code></td>
<td>

<p>Should each contrast be tested separately?. By default is TRUE.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Performs a general MANOVA to compare several groups or treatments. Additional contrasts can be tested using the contrasts matrix <code class="reqn">\bf{C}</code>. This can also be separated into several effects.
</p>
<p>Here we use an approach with matrices rather than te usual multivariate general linear model. Several designs can be obtained using the appropriate matrix of contrasts.
</p>


<h3>Value</h3>

<p>An object of class &quot;MANOVA&quot;  and &quot;Canonical.Biplot&quot;.
</p>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Amaro, I. R., Vicente-Villardon, J. L., &amp; Galindo-Villardon, M. P. (2004). Manova Biplot para arreglos de tratamientos con dos factores basado en modelos lineales generales multivariantes. Interciencia, 29(1), 26-32.
</p>
<p>Gabriel KR (1971) The biplot graphic display of matrices with application to principal component analysis. Biometrika 58(3):453-467.
</p>
<p>Gabriel, K. R. (1995). MANOVA biplots for two-way contingency tables. WJ Krzanowski (Ed.), Recent advances in descriptive multivariate analysis, Oxford University Press, Toronto. 227-268.
</p>
<p>Gower y Hand (1996): Biplots. Chapman &amp; Hall.
</p>
<p>Varas, M. J., Vicente-Tavera, S., Molina, E., &amp; Vicente-Villardon, J. L. (2005). Role of canonical biplot method in the study of building stones: an example from Spanish monumental heritage. Environmetrics, 16(4), 405-419.
</p>
<p>Santana, M. A., Romay, G., Matehus, J., Vcente-Villardon, J. L., &amp; Demey, J. R. (2009). simple and low-cost strategy for micropropagation of cassava (Manihot esculenta Crantz). African Journal of Biotechnology, 8(16).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(wine)
Y=as.matrix(wine[,4:21])
group=wine$Group
manvin=MANOVA(Y, Group=group)
summary(manvin)

</code></pre>

<hr>
<h2 id='MANOVA.Estimation'>
Estimation of the MANOVA parameters.
</h2><span id='topic+MANOVA.Estimation'></span>

<h3>Description</h3>

<p>The function estimates the parameters of the MANOVA for a set of contrasts and a set of linear combinations of the dependent variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MANOVA.Estimation(Y, X, C, M)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MANOVA.Estimation_+3A_y">Y</code></td>
<td>

<p>The matrix of dependent variables.
</p>
</td></tr>
<tr><td><code id="MANOVA.Estimation_+3A_x">X</code></td>
<td>

<p>The matrix of independent variables.
</p>
</td></tr>
<tr><td><code id="MANOVA.Estimation_+3A_c">C</code></td>
<td>

<p>A matrix containing a set of contrasts on the columns of X.
</p>
</td></tr>
<tr><td><code id="MANOVA.Estimation_+3A_m">M</code></td>
<td>

<p>A matrix containing a set of linear combinations of the columns of Y.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function estimates the parameters of the MANOVA for a set of contrasts contained in a matrix C and a set of linear combinations of the dependent variables in a matrix M. The  function is actually used in a more general procedure called MANOVA.
</p>


<h3>Value</h3>

<p>Returns the main results of the MANOVA.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon, Laura Vicente-Gonzalez
</p>


<h3>References</h3>

<p>Seber, G. A. (2009). Multivariate observations (Vol. 252). John Wiley &amp; Sons.
</p>

<hr>
<h2 id='mgc'>
Mixture Gaussian Clustering
</h2><span id='topic+mgc'></span>

<h3>Description</h3>

<p>Model based clustering using mixtures of gaussian distributions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mgc(x, NG = 2, init = "km", RemoveOutliers = FALSE, ConfidOutliers = 0.995,
tolerance = 1e-07, maxiter = 100, show = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mgc_+3A_x">x</code></td>
<td>

<p>The data matrix.
</p>
</td></tr>
<tr><td><code id="mgc_+3A_ng">NG</code></td>
<td>

<p>Number of groups or clusters to obtain.
</p>
</td></tr>
<tr><td><code id="mgc_+3A_init">init</code></td>
<td>

<p>Initial centers can be obtained from k-means (&quot;km&quot;) or at random (&quot;rd&quot;).
</p>
</td></tr>
<tr><td><code id="mgc_+3A_removeoutliers">RemoveOutliers</code></td>
<td>

<p>Should the extreme values be removed to calculate the clusters?
</p>
</td></tr>
<tr><td><code id="mgc_+3A_confidoutliers">ConfidOutliers</code></td>
<td>

<p>Percentage of the points to keep for the calculations when RemoveOutliers is true.
</p>
</td></tr>
<tr><td><code id="mgc_+3A_tolerance">tolerance</code></td>
<td>

<p>Tolerance for convergence.
</p>
</td></tr>
<tr><td><code id="mgc_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iterations.
</p>
</td></tr>
<tr><td><code id="mgc_+3A_show">show</code></td>
<td>

<p>Should the likelihood at each iteration be shown?
</p>
</td></tr>
<tr><td><code id="mgc_+3A_...">...</code></td>
<td>

<p>Any other parameter that can affect k-means if that is the initial configuration.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A basic algorithm for clustering with mixtures of gaussians with no restrictions on the covariance matrices.
</p>


<h3>Value</h3>

<p>Clusters.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>X=as.matrix(iris[,1:4])
mod1=mgc(X,NG=3)
plot(iris[,1:4], col=mod1$Classification)
table(iris[,5],mod1$Classification)
</code></pre>

<hr>
<h2 id='Ones'>
Matrix of ones
</h2><span id='topic+Ones'></span>

<h3>Description</h3>

<p>Square matrix of ones
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ones(n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Ones_+3A_n">n</code></td>
<td>

<p>Order of the matrix.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Square matrix of ones.
</p>


<h3>Value</h3>

<p>A matrix of ones of order n.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Ones(6)
</code></pre>

<hr>
<h2 id='PERMANOVA'>
PERMANOVA: MANOVA based on distances
</h2><span id='topic+PERMANOVA'></span>

<h3>Description</h3>

<p>The correct application of MANOVA needs normal and homocedastic data and the number of variables be much smaller than the number of individuals, but for many applications the conditions do not hold. To extend the application to this data Anderson develops PERMANOVA. This non-parametric test based on distances uses permutation to approximate the sampling distribution of the test statistic.
</p>
<p>Contrasts and Effects can be added to the calculations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PERMANOVA(Distance, group, C = NULL, Effects = NULL, nperm = 1000, seed = NULL,
CoordPrinc = FALSE, dimens = 2, PCoA = "Standard", ProjectInd = TRUE, tol = 1e-04,
DatosIni = TRUE, PostHoc="bonferroni")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PERMANOVA_+3A_distance">Distance</code></td>
<td>

<p>A list of three elements containing the data, the distances between individuals and type of distance used.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_group">group</code></td>
<td>

<p>A factor containing the groups to compare.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_c">C</code></td>
<td>

<p>Contrast matrix. By default it is null and the identity is used.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_effects">Effects</code></td>
<td>

<p>A factor with the effects in the rows of C. By default it is null and each row is considered as an effect.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_nperm">nperm</code></td>
<td>

<p>Number of permutations to perform. By default is 1000.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_seed">seed</code></td>
<td>

<p>Seed to start permutations. By default is null.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_coordprinc">CoordPrinc</code></td>
<td>

<p>Should the principal coordinates be calculated?. By default is FALSE.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_dimens">dimens</code></td>
<td>

<p>Number of dimensions to choose in the PCoA. By default is 2.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_pcoa">PCoA</code></td>
<td>

<p>The type of Principal Coordinates Analysis.
</p>
<p>There are two possibilities:
</p>
<p>* Standard
</p>
<p>* Weighted
</p>
<p>By default is Standard.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_projectind">ProjectInd</code></td>
<td>

<p>Show the row coordinates. By default is TRUE.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_tol">tol</code></td>
<td>

<p>Tolerance.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_datosini">DatosIni</code></td>
<td>

<p>It contains the initial data. By default is TRUE.
</p>
</td></tr>
<tr><td><code id="PERMANOVA_+3A_posthoc">PostHoc</code></td>
<td>

<p>Correction method. There are eight possibilities:
</p>
<p>*holm: Holm
</p>
<p>*hochberg: Hochberg 
</p>
<p>*hommel: Hommel
</p>
<p>*bonferroni: Bonferroni. By default is this method.
</p>
<p>*BH:Benjamini &amp; Hochberg 
</p>
<p>*BY: Benjamini &amp; Yekutieli
</p>
<p>*fdr: Benjamini &amp; Hochberg 
</p>
<p>*none
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function performs a PERMANOVA Analysis.
</p>


<h3>Value</h3>

<p>The PERMANOVA function create a list that return:
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>Function.</p>
</td></tr>
<tr><td><code>Title</code></td>
<td>
<p>Permutation based MANOVA.</p>
</td></tr>
<tr><td><code>Type</code></td>
<td>
<p>PERMANOVA.</p>
</td></tr>
<tr><td><code>Distances</code></td>
<td>
<p>A matrix containing the distances between individuals.</p>
</td></tr>
<tr><td><code>C</code></td>
<td>
<p>Contrasts Matrix.</p>
</td></tr>
<tr><td><code>Initial</code></td>
<td>
<p> Containing two matrices:
</p>
<p>* Global -&gt; Global contrast.
</p>
<p>* Contrastes -&gt;Contrasts for groups.
</p>
</td></tr>
<tr><td><code>DistMuestral</code></td>
<td>
<p>Sample distribution of F-exp from permutations.</p>
</td></tr>
<tr><td><code>pvalue</code></td>
<td>
<p>Estimate p-valor for PERMANOVA.</p>
</td></tr>
<tr><td><code>ExplainedVariance</code></td>
<td>
<p>Explained variance by Principal Coordinates selected.</p>
</td></tr>
<tr><td><code>Inertias</code></td>
<td>
<p>Eigenvalue, Explained variance, Cumulative explained variance.</p>
</td></tr>
<tr><td><code>MeanCoordinates</code></td>
<td>
<p>Mean Coordinates by groups for the dimensions obtained in the Principal Coordinates Analysis.</p>
</td></tr>
<tr><td><code>Qualities</code></td>
<td>
<p>Qualities representation by groups for the dimensions of PCoA.</p>
</td></tr>
<tr><td><code>CummulativeQualities</code></td>
<td>
<p>Cumulative qualities representation.</p>
</td></tr>
<tr><td><code>ClusterType</code></td>
<td>
<p>Cluster type selected.</p>
</td></tr>
<tr><td><code>Clusters</code></td>
<td>
<p>Clusters created.</p>
</td></tr>
<tr><td><code>ClusterNames</code></td>
<td>
<p>Names of clusters.</p>
</td></tr>
<tr><td><code>ClusterColors</code></td>
<td>
<p>Colors of clusters, color name and HTML code.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Anderson, M. J. (2008). A new method for non-parametric multivariate analysis of variance. Austral ecology, 26(1):32–46.
</p>
<p>Anderson, M. J. (2005). Permanova: a fortran computer program for permutational multivariate analysis of variance. Department of Statistics, University of Auckland, New Zealand, 24.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X = wine[,4:21]
X=IniTransform(X)
D = DistContinuous (X)
perwine=PERMANOVA(D, wine$Group)
perwine


C = matrix(c(1, 1, -1, 1, 1, -1, 1, 1, 1, -1, -1, 1), nrow=3, byrow=TRUE)
rownames(C)=c("C1", "C2", "C3")
colnames(C)=levels(wine$Group)

effects=factor(c(1,2,3))
levels(effects)=c("Origin", "Year", "Interaction")
perwine2=PERMANOVA(D, wine$Group, C=C, Effects=effects, CoordPrinc = TRUE)
summary(perwine2)

</code></pre>

<hr>
<h2 id='PERMANOVA.Estimation'>
Estimation of the PERMANOVA parameters
</h2><span id='topic+PERMANOVA.Estimation'></span>

<h3>Description</h3>

<p>Estimation of the PERMANOVA parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PERMANOVA.Estimation(D, X, C, Effects = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PERMANOVA.Estimation_+3A_d">D</code></td>
<td>

<p>A matrix of distances.
</p>
</td></tr>
<tr><td><code id="PERMANOVA.Estimation_+3A_x">X</code></td>
<td>

<p>A matrix of independent variables.
</p>
</td></tr>
<tr><td><code id="PERMANOVA.Estimation_+3A_c">C</code></td>
<td>

<p>Contrast matrix. By default it is null and the identity is used.
</p>
</td></tr>
<tr><td><code id="PERMANOVA.Estimation_+3A_effects">Effects</code></td>
<td>

<p>A factor with the effects in the rows of C. By default it is null and each row is considered as an effect.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Estimation of the PERMANOVA parameters.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>Global</code></td>
<td>
<p>Significance for the global effect</p>
</td></tr>
<tr><td><code>Contrastes</code></td>
<td>
<p>Significance for each contrast</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Anderson, M. J. (2008). A new method for non-parametric multivariate analysis of variance. Austral ecology, 26(1):32–46.
</p>
<p>Anderson, M. J. (2005). Permanova: a fortran computer program for permutational multivariate analysis of variance. Department of Statistics, University of Auckland, New Zealand, 24.
</p>

<hr>
<h2 id='PerMANOVA.Simple'>
PERMANOVA from a matrix of distancies
</h2><span id='topic+PerMANOVA.Simple'></span>

<h3>Description</h3>

<p>This function makes a PERMANOVA from distancies matrix and factor for groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PerMANOVA.Simple(D, grupo, nperm = 999, seed = NULL, C = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PerMANOVA.Simple_+3A_d">D</code></td>
<td>

<p>A matrix containing the distances between individuals.
</p>
</td></tr>
<tr><td><code id="PerMANOVA.Simple_+3A_grupo">grupo</code></td>
<td>

<p>A factor containing the groups to compare.
</p>
</td></tr>
<tr><td><code id="PerMANOVA.Simple_+3A_nperm">nperm</code></td>
<td>

<p>Number of permutation that want to perform. By default it is 999.
</p>
</td></tr>
<tr><td><code id="PerMANOVA.Simple_+3A_seed">seed</code></td>
<td>

<p>Seed to start permutations. By default it is null.
</p>
</td></tr>
<tr><td><code id="PerMANOVA.Simple_+3A_c">C</code></td>
<td>

<p>Contrast matrix. By default it is null.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The PERMANOVA.Simple function create a list that return:
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>Function</p>
</td></tr>
<tr><td><code>nperm</code></td>
<td>
<p>Number of permutation.</p>
</td></tr>
<tr><td><code>Inicial</code></td>
<td>
<p>Containing:
</p>
<p>* TSS -&gt; Total sum of squares.
</p>
<p>* BSS -&gt; Between groups sum of squares.
</p>
<p>* WSS -&gt; Within groups sum of squares.
</p>
<p>* glt -&gt; Total degrees of freedom.
</p>
<p>* glb -&gt; Between groups degrees of freedom.
</p>
<p>* glw -&gt; Within groups degrees of freedom.
</p>
<p>* Fexp -&gt; Experimental F.
</p>
</td></tr>
<tr><td><code>Fvals</code></td>
<td>
<p>F values of the permutation.</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>Estimate p-valor for PERMANOVA.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Anderson, M. J. (2001). A new method for non-parametric multivariate analysis of variance. Austral ecology, 26(1):32–46.
</p>
<p>Anderson, M. J. (2005). Permanova: a fortran computer program for permutational multivariate analysis of variance. Department of Statistics, University of Auckland, New Zealand, 24.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X = wine[,4:21]
Dist = DistContinuous (X)
PerMANOVA.Simple(Dist$D, wine$Group)
  
</code></pre>

<hr>
<h2 id='plot.BootCanonAnalysis'>
Plots the principal coordinates of the group centers and the bootstrap confidence regions
</h2><span id='topic+plot.BootCanonAnalysis'></span>

<h3>Description</h3>

<p>Plots the principal coordinates of the group centers and the bootstrap confidence regions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BootCanonAnalysis'
plot(x, A1 = 1, A2 = 2, centred = FALSE, confidence = 0.9,
PlotReplicates = TRUE, MeanCex = 1.5, MeanPch = 16, 
Title = "Bootstrap Canonical Analysis based on Distances", 
LabelMeans = TRUE, MeanLabels = NULL, MeanColors = NULL, SmartLabels = TRUE, 
BootstrapPlot = "el", PlotIndiv = FALSE, LabelInd = FALSE, IndLabels = NULL, 
IndColors = NULL, CexInd = 0.5, PchInd = 1, ConvexHullsInd = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.BootCanonAnalysis_+3A_x">x</code></td>
<td>

<p>The object to plot.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_a1">A1</code></td>
<td>

<p>Dimension to plot on the X axis.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_a2">A2</code></td>
<td>

<p>Dimension to plot on the Y axis.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_centred">centred</code></td>
<td>

<p>Plot centers of the bootstrap regions.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_confidence">confidence</code></td>
<td>

<p>Confidence for the bootstrap regions.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_plotreplicates">PlotReplicates</code></td>
<td>

<p>Should all the bootstrap replicates be plotted.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_meancex">MeanCex</code></td>
<td>

<p>Size of the points and text representing the centroids of the groups.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_meanpch">MeanPch</code></td>
<td>

<p>Marks of the points and text representing the centroids of the groups.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_title">Title</code></td>
<td>

<p>Main title of the graph.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_labelmeans">LabelMeans</code></td>
<td>

<p>Should the groups be labelled?
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_meanlabels">MeanLabels</code></td>
<td>

<p>Labels for the group means.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_meancolors">MeanColors</code></td>
<td>

<p>Colors for the group means.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_smartlabels">SmartLabels</code></td>
<td>

<p>Should the labels be placed in a smart way?
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_bootstrapplot">BootstrapPlot</code></td>
<td>

<p>Should the bootstrap regions be plotted?
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_plotindiv">PlotIndiv</code></td>
<td>

<p>Should the individuals be plotted?
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_labelind">LabelInd</code></td>
<td>

<p>Should individuals be labelled?
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_indlabels">IndLabels</code></td>
<td>

<p>Labels for the individuals.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_indcolors">IndColors</code></td>
<td>

<p>Colors for the rows (individuals).
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_cexind">CexInd</code></td>
<td>

<p>Sizes for the rows (individuals).
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_pchind">PchInd</code></td>
<td>

<p>Marks for the rows (individuals).
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_convexhullsind">ConvexHullsInd</code></td>
<td>

<p>Convex hulls containing the individuals of each group.
</p>
</td></tr>
<tr><td><code id="plot.BootCanonAnalysis_+3A_...">...</code></td>
<td>

<p>Any other graphical parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the principal coordinates of the means of the groups with the confidence bootstrap region.
</p>


<h3>Value</h3>

<p>The plot.
</p>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>

<hr>
<h2 id='plot.ellipse'>
Plot a concentration ellipse
</h2><span id='topic+plot.ellipse'></span>

<h3>Description</h3>

<p>Plot a concentration ellipse obtained from <code>ConcEllipse</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ellipse'
plot(x, add = TRUE, labeled = FALSE, center = FALSE, centerlabel = "Center",
initial = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.ellipse_+3A_x">x</code></td>
<td>

<p>An object with class <code>ellipse</code> obtained from <code>ConcEllipse</code>.
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_add">add</code></td>
<td>

<p>Should the ellipse be added to the current plot?
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_labeled">labeled</code></td>
<td>

<p>Should the ellipse be labelled with the confidence level?
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_center">center</code></td>
<td>

<p>Should the center be plotted?
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_centerlabel">centerlabel</code></td>
<td>

<p>Label for the center.
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_initial">initial</code></td>
<td>

<p>Should the initial data be plotted?
</p>
</td></tr>
<tr><td><code id="plot.ellipse_+3A_...">...</code></td>
<td>

<p>Any other graphical parameter that can affects the plot (as color, etc ...).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots an ellipse containing a specified percentage of the data.
</p>


<h3>Value</h3>

<p>No value returned.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Meulman, J. J., &amp; Heiser, W. J. (1983). The display of bootstrap solutions in multidimensional scaling. Murray Hill, NJ: Bell Laboratories.
</p>
<p>Linting, M., Meulman, J. J., Groenen, P. J., &amp; Van der Kooij, A. J. (2007). Stability of nonlinear principal components analysis: An empirical study using the balanced bootstrap. Psychological Methods, 12(3), 359.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ConcEllipse">ConcEllipse</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
dat=as.matrix(iris[1:50,1:2])
plot(iris[,1], iris[,2],col=iris[,5], asp=1)
E=ConcEllipse(dat, confidence=0.95)
plot(E, add=TRUE, center=TRUE)
</code></pre>

<hr>
<h2 id='plot.MANOVA'>
Plots the results of a MANOVA Biplot
</h2><span id='topic+plot.MANOVA'></span>

<h3>Description</h3>

<p>Plots the results of a MANOVA Biplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MANOVA'
plot(x, A1 = 1, A2 = 2, ScaleGraph = TRUE, PlotGroups = TRUE,
PlotVars = TRUE, PlotInd = TRUE, WhatInds=NULL,WhatVars=NULL, WhatGroups=NULL,
IndLabels=NULL, VarLabels=NULL, GroupLabels=NULL, AbbreviateLabels=FALSE,
LabelInd=TRUE, LabelVars = TRUE, CexGroup=1, PchGroup=16, margin=0.1,
AddLegend=FALSE, ShowAxes=FALSE, LabelAxes=FALSE, LabelGroups=TRUE,
PlotCircle = TRUE, ConvexHulls = FALSE, TypeCircle = "M", ColorGroups = NULL,
ColorVars = NULL, LegendPos="topright", ColorInd = NULL, voronoi = TRUE,
mode="a", TypeScale = "Complete", ValuesScale = "Original", MinQualityVars = 0,
dpg = 0, dpi=0, dp=0,  PredPoints=0, PlotAxis = FALSE, CexInd = NULL,
CexVar = NULL, PchInd = NULL, PchVar = NULL, ColorVar=NULL, ShowAxis=FALSE,
VoronoiColor="black", ShowBox=FALSE, ShowTitle=TRUE,PlotClus = FALSE,
TypeClus = "ch", ClustConf = 1, ClustCenters = FALSE, UseClusterColors = TRUE,
CexClustCenters=1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.MANOVA_+3A_x">x</code></td>
<td>

<p>An object of class &quot;MANOVA.Biplot&quot;.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_a1">A1</code></td>
<td>

<p>Dimension for the first axis. 1 is the default.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_a2">A2</code></td>
<td>

<p>Dimension for the second axis. 2 is the default.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_scalegraph">ScaleGraph</code></td>
<td>

<p>Rescale the coordinates to optimal matching.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_plotgroups">PlotGroups</code></td>
<td>

<p>Should the group centers be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_plotvars">PlotVars</code></td>
<td>

<p>Should the variables be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_plotind">PlotInd</code></td>
<td>

<p>Should the individuals be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_whatinds">WhatInds</code></td>
<td>

<p>What individuals should be plotted? A vector.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_whatvars">WhatVars</code></td>
<td>

<p>What variables should be plotted? A vector.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_whatgroups">WhatGroups</code></td>
<td>

<p>What groups should be plotted? A vector.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_indlabels">IndLabels</code></td>
<td>

<p>Labels for the individuals.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_varlabels">VarLabels</code></td>
<td>

<p>Labels for the variables.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_grouplabels">GroupLabels</code></td>
<td>

<p>Labels for the groups.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_abbreviatelabels">AbbreviateLabels</code></td>
<td>

<p>Should the labels be Abbreviated for simplicity of the graph?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_labelind">LabelInd</code></td>
<td>

<p>Should the individuals be labelled?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_labelvars">LabelVars</code></td>
<td>

<p>Should the variables be labelled?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_cexgroup">CexGroup</code></td>
<td>

<p>Size of the points for groups.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_pchgroup">PchGroup</code></td>
<td>

<p>Mark of the points for groups.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_margin">margin</code></td>
<td>

<p>Margin for labels (in percentage).
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_addlegend">AddLegend</code></td>
<td>

<p>Should a legend be added?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_showaxes">ShowAxes</code></td>
<td>

<p>Should the axes be shown?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_labelaxes">LabelAxes</code></td>
<td>

<p>Should the axes be labelled?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_labelgroups">LabelGroups</code></td>
<td>

<p>Should the groups be labelled?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_plotcircle">PlotCircle</code></td>
<td>

<p>Should the confidence regions for the groups be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_convexhulls">ConvexHulls</code></td>
<td>

<p>Should the convex hulls containing the individuals for each group be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_typecircle">TypeCircle</code></td>
<td>

<p>Type of confidence region: Univariate (U), Bonferroni(B), Multivariate (M) or Classical (C). By default is &quot;M&quot;.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_colorgroups">ColorGroups</code></td>
<td>

<p>User colors for the groups. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_colorvars">ColorVars</code></td>
<td>

<p>User colors for the variables. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_legendpos">LegendPos</code></td>
<td>

<p>Position of the legend.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_colorind">ColorInd</code></td>
<td>

<p>User colors for the individuals. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_voronoi">voronoi</code></td>
<td>

<p>Should the Voronoi diagram with the prediction regions for each group be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_mode">mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;h&quot;, &quot;ah&quot; and &quot;s&quot;. See details.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scale to use : &quot;Complete&quot;, &quot;StdDev&quot; or &quot;BoxPlot&quot;. See details.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>Values to show on the scale: &quot;Original&quot; or &quot;Transformed&quot;.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_minqualityvars">MinQualityVars</code></td>
<td>

<p>Minimum quality of representation for a variable to be plotted.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_dpg">dpg</code></td>
<td>

<p>A set of indices with the variables that will show projections onto the variables.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_dpi">dpi</code></td>
<td>

<p>A set of indices with the individuals that will show the projections.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_dp">dp</code></td>
<td>

<p>A set of indices with the variables that will show the projections of the individuals.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_predpoints">PredPoints</code></td>
<td>

<p>A vector with integers. The group centers listed in the vector are projected onto all the variables.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_plotaxis">PlotAxis</code></td>
<td>

<p>Not Used
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_cexind">CexInd</code></td>
<td>

<p>Size of the points for individuals.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_cexvar">CexVar</code></td>
<td>

<p>Size of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_pchind">PchInd</code></td>
<td>

<p>Markers of the points for individuals.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_pchvar">PchVar</code></td>
<td>

<p>Markers of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_colorvar">ColorVar</code></td>
<td>

<p>Colors of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Should axis scales be shown?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_voronoicolor">VoronoiColor</code></td>
<td>

<p>Color for the Voronoi diagram.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_showbox">ShowBox</code></td>
<td>

<p>Should a box around the points be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_showtitle">ShowTitle</code></td>
<td>

<p>Should Title be shown?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_plotclus">PlotClus</code></td>
<td>

<p>Should the clusters be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of plot for the clusters. (&quot;ch&quot;- Convex Hull, &quot;el&quot;- Ellipse or &quot;st&quot;- Star)
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_clustconf">ClustConf</code></td>
<td>

<p>Percent of points included in the cluster. Only the ClusConf percent of the points nearest to the center will be used to calculate the cluster.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_clustcenters">ClustCenters</code></td>
<td>

<p>Should the cluster centers be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_useclustercolors">UseClusterColors</code></td>
<td>

<p>Should the cluster colors be used in the plot?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_cexclustcenters">CexClustCenters</code></td>
<td>

<p>Size of the cluster centres.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA_+3A_...">...</code></td>
<td>

<p>Any other graphical parameter.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function plots the results of a MANOVA Biplot. The coordinates for Groups, Individuals and Variables can be shown or not on the plot, each of the three can also be labeled separately.
The are parameters to control the way each different set of coordinates is plotted and labeled.
</p>
<p>There are several modes for plotting the biplot.
</p>
<p>&quot;p&quot;.- Points (Rows and Columns are represented by points)
</p>
<p>&quot;a&quot; .- Arrows (The traditional representation with points for rows and arrows for columns)
</p>
<p>&quot;b&quot; .- The arrows for the columns are extended to both extremes of the plot and labeled outside the plot area.
</p>
<p>&quot;h&quot; .- The arrows for the columns are extended to the positive extreme of the plot and labeled outside the plot area.
</p>
<p>&quot;ah&quot; .- Same as arrows but labeled outside the plot area.
</p>
<p>&quot;s&quot; .- The directions (or biplot axes) have a graded scale for prediction of the original values.
</p>
<p>The <em>TypeScale</em> argument applies only to the <em>&quot;s&quot;</em> mode. There are three types:
</p>
<p>&quot;Complete&quot; .- An equally spaced scale covering the whole range of the data is calculates.
</p>
<p>&quot;StdDev&quot; .- Mean with one, two and three standard deviations
</p>
<p>&quot;BoxPlot&quot; .- Box-Plot like Scale (Median, 25 and 75 percentiles, maximum and minimum values.)
</p>
<p>The <em>ValuesScale</em> argument applies only to the <em>&quot;s&quot;</em> mode and controls if the labels show the  <em>Original</em> ot <em>Transformed</em> values.
</p>
<p>Some of the initial transformations are not compatible with some of the types of biplots and scales. For example, It is not possible to recover by projection the original values when you double centre de data. In that case you have the residuals for interaction and only the transformed values make sense.
</p>


<h3>Value</h3>

<p>A plot is returned.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon, Laura Vicente-Gonzalez
</p>


<h3>References</h3>

<p>Gabriel, K. R. (1972). Analysis of meteorological data by means of canonical decomposition and biplots. Journal of Applied Meteorology, 11(7), 1071-1077.
</p>
<p>Amaro, I. R., Vicente-Villardon, J. L., &amp; Galindo Villardon, M. P. (2004). Manova Biplot para arreglos de tratamientos con dos factores basado en modelos lineales generales multivariantes. Interciencia, 29(1), 26-32.
</p>
<p>Sierra, C., Ruiz-Barzola, O., Menendez, M., Demey, J. R., &amp; Vicente-Villardon, J. L. (2017). Geochemical interactions study in surface river sediments at an artisanal mining area by means of Canonical (MANOVA)-Biplot. Journal of Geochemical Exploration, 175, 72-81.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X=wine[,4:21]
manbip=MANOVA(X, Group=wine$Group)
plot(manbip, TypeCircle="B", Voronoi=FALSE)

</code></pre>

<hr>
<h2 id='plot.MANOVA.Biplot'>
Plots the results of a MANOVA Biplot
</h2><span id='topic+plot.MANOVA.Biplot'></span>

<h3>Description</h3>

<p>Plots the results of a MANOVA Biplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MANOVA.Biplot'
plot(x, A1 = 1, A2 = 2, ScaleGraph = TRUE, PlotGroups = TRUE, 
PlotVars = TRUE, PlotInd = TRUE, WhatInds = NULL, WhatVars = NULL, WhatGroups = NULL, 
IndLabels = NULL, VarLabels = NULL, GroupLabels = NULL, AbbreviateLabels = FALSE, 
LabelInd = TRUE, LabelVars = TRUE, CexGroup = 1, PchGroup = 16, margin = 0.1, 
AddLegend = FALSE, ShowAxes = FALSE, LabelAxes = FALSE, LabelGroups = TRUE, 
PlotCircle = TRUE, ConvexHulls = FALSE, TypeCircle = "M", ColorGroups = NULL, 
ColorVars = NULL, LegendPos = "topright", ColorInd = NULL, voronoi = TRUE, 
mode = "a", TypeScale = "Complete", ValuesScale = "Original", MinQualityVars = 0, 
dpg = 0, dpi = 0, dp = 0, PredPoints = 0, PlotAxis = FALSE, CexInd = NULL, 
CexVar = NULL, PchInd = NULL, PchVar = NULL, ColorVar = NULL, ShowAxis = FALSE, 
VoronoiColor = "black", ShowBox = FALSE, ShowTitle = TRUE, PlotClus = FALSE, 
TypeClus = "ch", ClustConf = 1, ClustCenters = FALSE, UseClusterColors = TRUE, 
CexClustCenters = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.MANOVA.Biplot_+3A_x">x</code></td>
<td>

<p>An object of class &quot;MANOVA.Biplot&quot;
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_a1">A1</code></td>
<td>

<p>Dimension for the first axis. 1 is the default.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_a2">A2</code></td>
<td>

<p>Dimension for the second axis. 2 is the default.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_scalegraph">ScaleGraph</code></td>
<td>

<p>Reescale the coordinates to optimal matching.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_plotgroups">PlotGroups</code></td>
<td>

<p>Shoud the group centers be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_plotvars">PlotVars</code></td>
<td>

<p>Should the variables be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_plotind">PlotInd</code></td>
<td>

<p>Should the individuals be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_whatinds">WhatInds</code></td>
<td>

<p>What individuals should be plooted? A vector.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_whatvars">WhatVars</code></td>
<td>

<p>What variables should be plooted? A vector.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_whatgroups">WhatGroups</code></td>
<td>

<p>What groups should be plooted? A vector.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_indlabels">IndLabels</code></td>
<td>

<p>Labels for the individuals
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_varlabels">VarLabels</code></td>
<td>

<p>Labels for the variables
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_grouplabels">GroupLabels</code></td>
<td>

<p>Labels for the groups
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_abbreviatelabels">AbbreviateLabels</code></td>
<td>

<p>Should the labels be Abbreviated for simplicity of the graph?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_labelind">LabelInd</code></td>
<td>

<p>Should the individuaqls be labelled?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_labelvars">LabelVars</code></td>
<td>

<p>Should the variables be labelled?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_cexgroup">CexGroup</code></td>
<td>

<p>Size of the points for groups
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_pchgroup">PchGroup</code></td>
<td>

<p>Mark of the points for groups
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_margin">margin</code></td>
<td>

<p>Margin for labels (in percentage)
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_addlegend">AddLegend</code></td>
<td>

<p>Should a legend be added?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_showaxes">ShowAxes</code></td>
<td>

<p>Should the axes be shown?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_labelaxes">LabelAxes</code></td>
<td>

<p>Should the axes be labelled?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_labelgroups">LabelGroups</code></td>
<td>

<p>Should the groups be labelled?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_plotcircle">PlotCircle</code></td>
<td>

<p>Should the confidence regions for the groups be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_convexhulls">ConvexHulls</code></td>
<td>

<p>Should the convex hulls containing the individuals for each group be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_typecircle">TypeCircle</code></td>
<td>

<p>Type of confidence region: Univariate (U), Bonferroni(B), Multivariate (M) or Classical (C)
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_colorgroups">ColorGroups</code></td>
<td>

<p>User colors for the groups. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_colorvars">ColorVars</code></td>
<td>

<p>User colors for the variables. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_legendpos">LegendPos</code></td>
<td>

<p>Position of the legend.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_colorind">ColorInd</code></td>
<td>

<p>User colors for the individuals. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_voronoi">voronoi</code></td>
<td>

<p>Should the voronoi diagram with the prediction regións for each group be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_mode">mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;h&quot;, &quot;ah&quot; and &quot;s&quot;.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scale to use : &quot;Complete&quot;, &quot;StdDev&quot; or &quot;BoxPlot&quot;
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>Values to show on the scale: &quot;Original&quot; or &quot;Transformed&quot;
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_minqualityvars">MinQualityVars</code></td>
<td>

<p>Minimum quality of representation for a variable to be plotted
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_dpg">dpg</code></td>
<td>

<p>A set of indices with the variables that will show projections onto the variables 
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_dpi">dpi</code></td>
<td>

<p>A set of indices with the individuals that will show the projections 
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_dp">dp</code></td>
<td>

<p>A set of indices with the variables that will show the projections of the individuals
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_predpoints">PredPoints</code></td>
<td>

<p>A vector with integers. The group centers listed in the vector are projected onto all the variables.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_plotaxis">PlotAxis</code></td>
<td>

<p>Not Used
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_cexind">CexInd</code></td>
<td>

<p>Size of the points for individuals.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_cexvar">CexVar</code></td>
<td>

<p>Size of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_pchind">PchInd</code></td>
<td>

<p>Marhers of the points for individuals.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_pchvar">PchVar</code></td>
<td>

<p>Markers of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_colorvar">ColorVar</code></td>
<td>

<p>Colors of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Should axis scales be shown?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_voronoicolor">VoronoiColor</code></td>
<td>

<p>Color for the Voronoi diagram
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_showbox">ShowBox</code></td>
<td>

<p>Should a box around the poitns be plotted?
</p>
</td></tr> 
<tr><td><code id="plot.MANOVA.Biplot_+3A_showtitle">ShowTitle</code></td>
<td>

<p>Should Title be shown?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_plotclus">PlotClus</code></td>
<td>

<p>Should the clusters be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of cluster to plot (convex hull, ellipse or star)
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_clustconf">ClustConf</code></td>
<td>

<p>Confidence level for the clusters (percentage plotted)
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_clustcenters">ClustCenters</code></td>
<td>

<p>Should the cluster centres be plotted?
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_useclustercolors">UseClusterColors</code></td>
<td>

<p>Use cluster colors for individuals
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_cexclustcenters">CexClustCenters</code></td>
<td>

<p>Size of the cluster centers
</p>
</td></tr>
<tr><td><code id="plot.MANOVA.Biplot_+3A_...">...</code></td>
<td>

<p>Any other graphical parameter
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function plots the results of a MANOVA Biplot. The coordinates for Groups, Individuals and Variables can be shown or not on the plot, each of the three can also be labeled separately.
The are parameters to control the way each different set of coordinates is plotted and labeled.
</p>
<p>There are several modes for plotting the biplot.
</p>
<p>&quot;p&quot;.- Points (Rows and Columns are represented by points)
</p>
<p>&quot;a&quot; .- Arrows (The traditional representation with points for rows and arrows for columns)
</p>
<p>&quot;b&quot; .- The arrows for the columns are extended to both extremes of the plot and labeled outside the plot area.
</p>
<p>&quot;h&quot; .- The arrows for the columns are extended to the positive extreme of the plot and labeled outside the plot area.
</p>
<p>&quot;ah&quot; .- Same as arrows but labeled outside the plot area.
</p>
<p>&quot;s&quot; .- The directions (or biplot axes) have a graded scale for prediction of the original values.
</p>
<p>The <em>TypeScale</em> argument applies only to the <em>&quot;s&quot;</em> mode. There are three types:
</p>
<p>&quot;Complete&quot; .- An equally spaced scale covering the whole range of the data is calculates.
</p>
<p>&quot;StdDev&quot; .- Mean with one, two and three stadard deviations 
</p>
<p>&quot;BoxPlot&quot; .- Box-Plot like Scale (Median, 25 and 75 percentiles, maximum and minimum values.)
</p>
<p>The <em>ValuesScale</em> argument applies only to the <em>&quot;s&quot;</em> mode and controls if the labels show the  <em>Original</em> ot <em>Transformed</em> values.
</p>
<p>Some of the initial transformations are not compatible with some of the types of biplots and scales. For example, It is not possible to recover by projection the original values when you double centre de data. In that case you have the residuals for interaction and only the transformed values make sense.
</p>


<h3>Value</h3>

<p>A plot is returned
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente Villardon, Laura Vicente Gonzalez
</p>


<h3>References</h3>

<p>Gabriel, K. R. (1972). Analysis of meteorological data by means of canonical decomposition and biplots. Journal of Applied Meteorology, 11(7), 1071-1077.
</p>
<p>Amaro, I. R., Vicente-Villardón, J. L., &amp; Galindo Villardón, M. P. (2004). Manova Biplot para arreglos de tratamientos con dos factores basado en modelos lineales generales multivariantes. Interciencia, 29(1), 26-32.
</p>
<p>Sierra, C., Ruíz-Barzola, O., Menéndez, M., Demey, J. R., &amp; Vicente-Villardón, J. L. (2017). Geochemical interactions study in surface river sediments at an artisanal mining area by means of Canonical (MANOVA)-Biplot. Journal of Geochemical Exploration, 175, 72-81.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X=wine[,4:21]
manbip=MANOVA(X, Group=wine$Group)
plot(manbip, TypeCircle="U", Voronoi=FALSE)
</code></pre>

<hr>
<h2 id='plot.PERMANOVA'>
Plots the results of the PERMANOVA function
</h2><span id='topic+plot.PERMANOVA'></span>

<h3>Description</h3>

<p>Plots the principal coordinates of the group centers a the bootstrap confidence regions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'PERMANOVA'
plot(x, A1 = 1, A2 = 2, ScaleGraph = TRUE, ShowAxis = FALSE,
ShowAxes = FALSE, LabelAxes = TRUE, margin = 0.1, ShowBox = TRUE, PlotGroups = TRUE,
LabelGroups = TRUE, CexGroup = 1.5, PchGroup = 16, ColorGroup = NULL, voronoi = TRUE,
VoronoiColor = "black", PlotInd = FALSE, LabelInd = TRUE, CexInd = 0.8, PchInd = 3,
ColorInd = NULL, WhatInds = NULL, IndLabels = NULL, PlotVars = TRUE, LabelVar = TRUE,
CexVar = NULL, PchVar = NULL, ColorVar = NULL, WhatVars = NULL, VarLabels = NULL,
mode = "a", TypeScale = "Complete", ValuesScale = "Original", SmartLabels = TRUE,
AddLegend = TRUE, LegendPos = "topright", PlotCircle = TRUE, ConvexHulls = FALSE,
TypeCircle = "M", MinQualityVars = 0, dpg = 0, dpi = 0, PredPoints = 0,
PlotClus = TRUE, TypeClus = "ch", ClustConf = 1, CexClustCenters=1,ClustCenters = FALSE, 
UseClusterColors = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.PERMANOVA_+3A_x">x</code></td>
<td>

<p>An object of class &quot;PERMANOVA&quot;
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_a1">A1</code></td>
<td>

<p>Dimension for the first axis. 1 is the default.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_a2">A2</code></td>
<td>

<p>Dimension for the second axis. 2 is the default.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_scalegraph">ScaleGraph</code></td>
<td>

<p>Resale the coordinates to optimal matching.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_showaxis">ShowAxis</code></td>
<td>

<p>Should the axis passing trough the origin be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_showaxes">ShowAxes</code></td>
<td>

<p>Should the axes be shown?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_labelaxes">LabelAxes</code></td>
<td>

<p>Should the axes be labelled?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_margin">margin</code></td>
<td>

<p>Margin for labels (in percentage).
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_showbox">ShowBox</code></td>
<td>

<p>Should a box around the points be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_plotgroups">PlotGroups</code></td>
<td>

<p>Should the group centers be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_labelgroups">LabelGroups</code></td>
<td>

<p>Should the groups be labelled?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_cexgroup">CexGroup</code></td>
<td>

<p>Size of the points for groups.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_pchgroup">PchGroup</code></td>
<td>

<p>Mark of the points for groups.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_colorgroup">ColorGroup</code></td>
<td>

<p>User colors for the groups. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_voronoi">voronoi</code></td>
<td>

<p>Should the Voronoi diagram with the prediction regions for each group be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_voronoicolor">VoronoiColor</code></td>
<td>

<p>Color for the Voronoi diagram.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_plotind">PlotInd</code></td>
<td>

<p>Should the individuals be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_labelind">LabelInd</code></td>
<td>

<p>Should the individuals be labelled?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_cexind">CexInd</code></td>
<td>

<p>Size of the points for individuals.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_pchind">PchInd</code></td>
<td>

<p>Markers of the points for individuals.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_colorind">ColorInd</code></td>
<td>

<p>User colors for the individuals. Default colors will be used if NULL.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_whatinds">WhatInds</code></td>
<td>

<p>What individuals should be plotted? A vector.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_indlabels">IndLabels</code></td>
<td>

<p>Labels for the individuals.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_plotvars">PlotVars</code></td>
<td>

<p>Should the variables be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_labelvar">LabelVar</code></td>
<td>

<p>Should the variables be labelled?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_cexvar">CexVar</code></td>
<td>

<p>Size of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_pchvar">PchVar</code></td>
<td>

<p>Markers of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_colorvar">ColorVar</code></td>
<td>

<p>Colors of the points for variables.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_whatvars">WhatVars</code></td>
<td>

<p>What variables should be plotted? A vector.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_varlabels">VarLabels</code></td>
<td>

<p>Labels for the variables.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_mode">mode</code></td>
<td>

<p>Mode of the biplot: &quot;p&quot;, &quot;a&quot;, &quot;b&quot;, &quot;h&quot;, &quot;ah&quot; and &quot;s&quot;.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_typescale">TypeScale</code></td>
<td>

<p>Type of scale to use : &quot;Complete&quot;, &quot;StdDev&quot; or &quot;BoxPlot&quot;.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_valuesscale">ValuesScale</code></td>
<td>

<p>Values to show on the scale: &quot;Original&quot; or &quot;Transformed&quot;.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_smartlabels">SmartLabels</code></td>
<td>

<p>Should the labels be plotted in a smart way?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_addlegend">AddLegend</code></td>
<td>

<p>Should a legend be added?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_legendpos">LegendPos</code></td>
<td>

<p>Position of the legend.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_plotcircle">PlotCircle</code></td>
<td>

<p>Should the confidence regions for the groups be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_convexhulls">ConvexHulls</code></td>
<td>

<p>Should the convex hulls containing the individuals for each group be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_typecircle">TypeCircle</code></td>
<td>

<p>Type of confidence region: Univariate (U), Bonferroni(B), Multivariate (M) or Classical (C). By default is &quot;M&quot;.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_minqualityvars">MinQualityVars</code></td>
<td>

<p>Minimum quality of representation for a variable to be plotted.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_dpg">dpg</code></td>
<td>

<p>A set of indices with the variables that will show projections onto the variables.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_dpi">dpi</code></td>
<td>

<p>A set of indices with the individuals that will show the projections.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_predpoints">PredPoints</code></td>
<td>

<p>A vector with integers. The group centers listed in the vector are projected onto all the variables.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_plotclus">PlotClus</code></td>
<td>

<p>Should the clusters be plotted?
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of plot for the clusters. (&quot;ch&quot;- Convex Hull, &quot;el&quot;- Ellipse or &quot;st&quot;- Star)
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_clustconf">ClustConf</code></td>
<td>

<p>Percent of points included in the cluster. only the ClusConf percent of the points nearest to the center will be used to calculate the cluster.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_cexclustcenters">CexClustCenters</code></td>
<td>

<p>Size of the cluster centers.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_clustcenters">ClustCenters</code></td>
<td>

<p>Should the cluster centers be plotted.
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_useclustercolors">UseClusterColors</code></td>
<td>

<p>Should the cluster colors be used in the plot. 
</p>
</td></tr>
<tr><td><code id="plot.PERMANOVA_+3A_...">...</code></td>
<td>

<p>Any other graphical parameter.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the principal coordinates of the group centers a the bootstrap confidence regions.
</p>


<h3>Value</h3>

<p>The plot.
</p>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(wine)
X = wine[,4:21]
X=IniTransform(X)
D = DistContinuous (X)
perwine=PERMANOVA(D, wine$Group, CoordPrinc=TRUE)
plot(perwine)


</code></pre>

<hr>
<h2 id='PlotClustersBiplot'>
Plot clusters on a biplot.
</h2><span id='topic+PlotClustersBiplot'></span>

<h3>Description</h3>

<p>Highlights several groups or clusters on a biplot representation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotClustersBiplot(A, Groups = Ones(c(nrow(A), 1)), TypeClus = "st", ClusterColors =
NULL, ClusterNames = NULL, centers = TRUE, ClustConf = 1, Legend = FALSE, LegendPos =
"topright", CexClustCenters = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotClustersBiplot_+3A_a">A</code></td>
<td>

<p>Coordinates of the points in the scattergram.
</p>
</td></tr>
<tr><td><code id="PlotClustersBiplot_+3A_groups">Groups</code></td>
<td>

<p>Factor defining the groups to be highlighted.
</p>
</td></tr>
<tr><td><code id="PlotClustersBiplot_+3A_typeclus">TypeClus</code></td>
<td>

<p>Type of representation of the clusters. For the moment just a convex hull but in the future ellipses and stars will be added.
</p>
</td></tr>
<tr><td><code id="PlotClustersBiplot_+3A_clustercolors">ClusterColors</code></td>
<td>

<p>A vector of colors with as many elements as clusters. If <code>NULL</code> the function selects the rainbow colors.
</p>
</td></tr>
<tr><td><code id="PlotClustersBiplot_+3A_clusternames">ClusterNames</code></td>
<td>

<p>A vector of names with as many elements as clusters.
</p>
</td></tr>
<tr><td><code id="PlotClustersBiplot_+3A_centers">centers</code></td>
<td>

<p>Logical variable to control if centres of the clusters are plotted.
</p>
</td></tr>
<tr><td><code id="PlotClustersBiplot_+3A_clustconf">ClustConf</code></td>
<td>

<p>Percent of points included in the cluster. Only the ClusConf percent of the points nearest to the center will be used to calculate the cluster.
</p>
</td></tr>
<tr><td><code id="PlotClustersBiplot_+3A_legend">Legend</code></td>
<td>

<p>Should a legend be plotted?
</p>
</td></tr>
<tr><td><code id="PlotClustersBiplot_+3A_legendpos">LegendPos</code></td>
<td>

<p>Position of the legend.
</p>
</td></tr>
<tr><td><code id="PlotClustersBiplot_+3A_cexclustcenters">CexClustCenters</code></td>
<td>

<p>Size of the centers.
</p>
</td></tr>
<tr><td><code id="PlotClustersBiplot_+3A_...">...</code></td>
<td>

<p>Any other graphical parameters.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The clusters to plot should be added to the biplot object using the function <code>AddClusterToBiplot</code>.
</p>


<h3>Value</h3>

<p>It takes effects on a plot.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>See Also</h3>

<p><code><a href="#topic+AddClusterToBiplot">AddClusterToBiplot</a></code>
</p>

<hr>
<h2 id='PostHocComp'>
Post Hoc pairwise comparisons
</h2><span id='topic+PostHocComp'></span>

<h3>Description</h3>

<p>Creates a contrast C matrix of post hoc comparisons among groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PostHocComp(grupo)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PostHocComp_+3A_grupo">grupo</code></td>
<td>

<p>A factor containing the groups or treatments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a contrast C matrix of post hoc comparisons among groups. The rows of the contrast matrix are not orthogonal.
</p>


<h3>Value</h3>

<p>The PostHocComp function return a matrix to compare the levels of groups introduced.
</p>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez and Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
PostHocComp(wine$Group)
</code></pre>

<hr>
<h2 id='ProcrustesSimple'>
Simple Procrustes Analysis
</h2><span id='topic+ProcrustesSimple'></span>

<h3>Description</h3>

<p>Simple Procrustes Analysis for two matrices
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ProcrustesSimple(X, Y, centre = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ProcrustesSimple_+3A_x">X</code></td>
<td>

<p>Matrix of the first configuration.
</p>
</td></tr>
<tr><td><code id="ProcrustesSimple_+3A_y">Y</code></td>
<td>

<p>Matrix of the second configuration.
</p>
</td></tr>
<tr><td><code id="ProcrustesSimple_+3A_centre">centre</code></td>
<td>

<p>Should the matrices be centered before the calculations?
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Orthogonal Procrustes Analysis for two configurations X and Y.
The first configuration X is used as a reference and the second,
Y, is transformed to match the reference as much as possible.
X = s Y T + 1t +E = Z + E
</p>


<h3>Value</h3>

<p>An object of class <code>Procrustes</code>.This has components:
</p>
<table role = "presentation">
<tr><td><code>X</code></td>
<td>
<p>First Configuration.</p>
</td></tr>
<tr><td><code>Y</code></td>
<td>
<p>Second Configuration.</p>
</td></tr>
<tr><td><code>Yrot</code></td>
<td>
<p>Second Configuration after the transformation.</p>
</td></tr>
<tr><td><code>T</code></td>
<td>
<p>Rotation Matrix.</p>
</td></tr>
<tr><td><code>t</code></td>
<td>
<p>Translation Vector.</p>
</td></tr>
<tr><td><code>s</code></td>
<td>
<p>Scale Factor.</p>
</td></tr>
<tr><td><code>rsss</code></td>
<td>
<p>Residual Sum of Squares.</p>
</td></tr>
<tr><td><code>fit</code></td>
<td>
<p>Goodness of fit as percent of explained variance.</p>
</td></tr>
<tr><td><code>correlations</code></td>
<td>
<p>Correlations among the columns of X and Z.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>References</h3>

<p>Ingwer Borg, I. &amp; Groenen, P. J.F. (2005). Modern Multidimensional Scaling. Theory and Applications. Second Edition. Springer
</p>

<hr>
<h2 id='spiders'>
Hunting Spiders Data
</h2><span id='topic+spiders'></span>

<h3>Description</h3>

<p>Hunting spiders data transformed into Presence/Absence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("spiders")</code></pre>


<h3>Format</h3>

<p>A data frame with 28 observations of presence/absence on the following 12 hunting spider species.
</p>

<dl>
<dt><code>Alopacce</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Alopecosa accentuata.</p>
</dd>
<dt><code>Alopcune</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Alopecosa cuneata.</p>
</dd>
<dt><code>Alopfabr</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Alopecosa fabrilis.</p>
</dd>
<dt><code>Arctlute</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Arctosa lutetiana.</p>
</dd>
<dt><code>Arctperi</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Arctosa perita.</p>
</dd>
<dt><code>Auloalbi</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Aulonia albimana.</p>
</dd>
<dt><code>Pardlugu</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Pardosa lugubris.</p>
</dd>
<dt><code>Pardmont</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Pardosa monticola.</p>
</dd>
<dt><code>Pardnigr</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Pardosa nigriceps.</p>
</dd>
<dt><code>Pardpull</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Pardosa pullata.</p>
</dd>
<dt><code>Trocterr</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Trochosa terricola.</p>
</dd>
<dt><code>Zoraspin</code></dt><dd><p>a factor with levels <code>Absent</code> <code>Present</code> of the species Zora spinimana.</p>
</dd>
</dl>



<h3>Source</h3>

<p>van der Aart, P. J. M., and Smeenk-Enserink, N. (1975) Correlations between distributions of hunting spiders (Lycos- idae, Ctenidae) and environmental characteristics in a dune area. Netherlands Journal of Zoology 25, 1-45.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spiders)
</code></pre>

<hr>
<h2 id='spidersb'>
Hunting Spiders Data
</h2><span id='topic+spidersb'></span>

<h3>Description</h3>

<p>Hunting spiders data transformed into 1/0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("spidersb")</code></pre>


<h3>Format</h3>

<p>A data frame with 28 observations of presence/absence of 12 hunting spider species
</p>

<dl>
<dt>Alopacce</dt><dd><p>Presence/Absence of the species Alopecosa accentuata</p>
</dd>
<dt>Alopcune</dt><dd><p>Presence/Absence of the species Alopecosa cuneata</p>
</dd>
<dt>Alopfabr</dt><dd><p>Presence/Absence of the species Alopecosa fabrilis</p>
</dd>
<dt>Arctlute</dt><dd><p>Presence/Absence of the species Arctosa lutetiana</p>
</dd>
<dt>Arctperi</dt><dd><p>Presence/Absence of the species Arctosa perita</p>
</dd>
<dt>Auloalbi</dt><dd><p>Presence/Absence of the species Aulonia albimana</p>
</dd>
<dt>Pardlugu</dt><dd><p>Presence/Absence of the species Pardosa lugubris</p>
</dd>
<dt>Pardmont</dt><dd><p>Presence/Absence of the species Pardosa monticola</p>
</dd>
<dt>Pardnigr</dt><dd><p>Presence/Absence of the species Pardosa nigriceps</p>
</dd>
<dt>Pardpull</dt><dd><p>Presence/Absence of the species Pardosa pullata</p>
</dd>
<dt>Trocterr</dt><dd><p>Presence/Absence of the species Trochosa terricola</p>
</dd>
<dt>Zoraspin</dt><dd><p>Presence/Absence of the species Zora spinimana</p>
</dd>
</dl>



<h3>Source</h3>

<p>van der Aart, P. J. M., and Smeenk-Enserink, N. (1975) Correlations between distributions of hunting spiders (Lycos- idae, Ctenidae) and environmental characteristics in a dune area. Netherlands Journal of Zoology 25, 1-45.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spidersb)
</code></pre>

<hr>
<h2 id='summary.BootDisMANOVA'>
Summarizes the results of a Bootstrap Manova based on distances
</h2><span id='topic+summary.BootDisMANOVA'></span>

<h3>Description</h3>

<p>Summarizes the results of a Bootstrap Manova based on distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'BootDisMANOVA'
summary(object, Latex = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summary.BootDisMANOVA_+3A_object">object</code></td>
<td>

<p>An object of class &quot;BootDisMANOVA&quot;.
</p>
</td></tr>
<tr><td><code id="summary.BootDisMANOVA_+3A_latex">Latex</code></td>
<td>

<p>Should Latex tables be provided?
</p>
</td></tr>
<tr><td><code id="summary.BootDisMANOVA_+3A_...">...</code></td>
<td>

<p>Any other parameter.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summarizes the results of a Bootstrap Manova based on distances including Latex tables.
</p>


<h3>Value</h3>

<p>Prints the results.
</p>


<h3>Author(s)</h3>

<p>Laura Vicente-Gonzalez, Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
X = wine[,4:21]
D = DistContinuous (X)
bootwine=BootDisMANOVA(D, wine$Group)
summary(bootwine)

</code></pre>

<hr>
<h2 id='TextSmart'>
Labels of a Scatter
</h2><span id='topic+TextSmart'></span>

<h3>Description</h3>

<p>Plots labels of points in a scattergram. labels for points with positive <code>x</code> are placed on the right of
the points, and labels for points with negative values on the left.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TextSmart(A, Labels, CexPoints = 1, ColorPoints = "black", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="TextSmart_+3A_a">A</code></td>
<td>

<p>Coordinates of the points for the scattergram.
</p>
</td></tr>
<tr><td><code id="TextSmart_+3A_labels">Labels</code></td>
<td>

<p>Labels for the points.
</p>
</td></tr>
<tr><td><code id="TextSmart_+3A_cexpoints">CexPoints</code></td>
<td>

<p>Size of the labels.
</p>
</td></tr>
<tr><td><code id="TextSmart_+3A_colorpoints">ColorPoints</code></td>
<td>

<p>Colors of the labels.
</p>
</td></tr>
<tr><td><code id="TextSmart_+3A_...">...</code></td>
<td>

<p>Additional graphical arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is used to improve the readability of the labels in a scattergram.
</p>


<h3>Value</h3>

<p>No value returned.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>

<hr>
<h2 id='wine'>
Wine data
</h2><span id='topic+wine'></span>

<h3>Description</h3>

<p>Comparison of young wines of Ribera de Duero and Toro.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("wine")</code></pre>


<h3>Format</h3>

<p>A data frame with 45 observations on the following 21 variables.
</p>

<dl>
<dt><code>Year</code></dt><dd><p>A factor with levels <code>1986</code> <code>1987</code></p>
</dd>
<dt><code>Origin</code></dt><dd><p>A factor with levels <code>Ribera</code> <code>Toro</code></p>
</dd>
<dt><code>Group</code></dt><dd><p>A factor with levels <code>R86</code> <code>R87</code> <code>T86</code> <code>T87</code></p>
</dd>
<dt><code>A</code></dt><dd><p>Alcoholic content (percentage)</p>
</dd>
<dt><code>VA</code></dt><dd><p>volatil acidity - g acetic acid/l</p>
</dd>
<dt><code>TA</code></dt><dd><p>Total tritable acidity - g tartaric acid/l</p>
</dd>
<dt><code>FA</code></dt><dd><p>Fixed acidity -  g tartaric acid/l</p>
</dd>
<dt><code>pH</code></dt><dd><p>pH</p>
</dd>
<dt><code>TPR</code></dt><dd><p>Total phenolics - g gallic acid /l - Folin</p>
</dd>
<dt><code>TPS</code></dt><dd><p>Total phenolics - Somers</p>
</dd>
<dt><code>V</code></dt><dd><p>Substances reactive to vanilin - mg catechin/l</p>
</dd>
<dt><code>PC</code></dt><dd><p>Procyanidins - mg cyanidin/l</p>
</dd>
<dt><code>ACR</code></dt><dd><p>Total Anthocyanins - mg/l - method 1</p>
</dd>
<dt><code>ACS</code></dt><dd><p>Total Anthocyanins - mg/l - methods 2</p>
</dd>
<dt><code>ACC</code></dt><dd><p>Malvidin - malvidin-3-glucoside mg/l</p>
</dd>
<dt><code>CI</code></dt><dd><p>Color density - </p>
</dd>
<dt><code>CI2</code></dt><dd><p>Color density 2</p>
</dd>
<dt><code>H</code></dt><dd><p>Wine Hue Color</p>
</dd>
<dt><code>I</code></dt><dd><p>Degree of Ionization - Percent</p>
</dd>
<dt><code>CA</code></dt><dd><p>Chemical Age</p>
</dd>
<dt><code>VPC</code></dt><dd><p>ratio V/PC</p>
</dd>
</dl>



<h3>Details</h3>

<p>Comparison of young wines of Ribera de Duero and Toro.
</p>


<h3>Source</h3>

<p>Rivas-Gonzalo, J. C., Gutierrez, Y., Polanco, A. M., Hebrero, E., Vicente-Villardon, J. L., Galindo, P., &amp; Santos-Buelga, C. (1993). Biplot analysis applied to enological parameters in the geographical classification of young red wines. American journal of enology and viticulture, 44(3), 302-308.
</p>


<h3>References</h3>

<p>Rivas-Gonzalo, J. C., Gutierrez, Y., Polanco, A. M., Hebrero, E., Vicente-Villardon, J. L., Galindo, P., &amp; Santos-Buelga, C. (1993). Biplot analysis applied to enological parameters in the geographical classification of young red wines. American journal of enology and viticulture, 44(3), 302-308.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(wine)
</code></pre>

<hr>
<h2 id='Zeros'>
Matrix of zeros
</h2><span id='topic+Zeros'></span>

<h3>Description</h3>

<p>Matrix of zeros as in Matlab.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Zeros(n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Zeros_+3A_n">n</code></td>
<td>

<p>Dimension of the matrix.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of zeros.
</p>


<h3>Author(s)</h3>

<p>Jose Luis Vicente-Villardon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Zeros(6)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
