<!DOCTYPE html><html><head><title>Help for package PGEE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PGEE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#PGEE-package'><p>Penalized Generalized Estimating Equations</p></a></li>
<li><a href='#CVfit'><p>Function to compute cross-validated tuning parameter value</p></a></li>
<li><a href='#MGEE'><p>Function to fit generalized estimating equations</p></a></li>
<li><a href='#PGEE'><p>Function to fit penalized generalized estimating equations</p></a></li>
<li><a href='#yeastG1'><p>Yeast cell-cycle gene expression data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Penalized Generalized Estimating Equations in High-Dimension</td>
</tr>
<tr>
<td>Version:</td>
<td>1.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-04-22</td>
</tr>
<tr>
<td>Author:</td>
<td>Gul Inan (Lecturer, Middle East Technical University), Jianhui Zhou (Associate Professor, University of Virginia) and Lan Wang (Professor, University of Minnesota)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gul Inan &lt;ginan@metu.edu.tr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Fits penalized generalized estimating equations to longitudinal data with high-dimensional covariates.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>MASS, mvtnorm</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-04-22 17:58:14 UTC; gul</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-04-24 21:27:20 UTC</td>
</tr>
</table>
<hr>
<h2 id='PGEE-package'>Penalized Generalized Estimating Equations</h2><span id='topic+PGEE-package'></span>

<h3>Description</h3>

<p>This package fits penalized generalized estimating equations to longitudinal
data with high-dimensional covariates through accommodating SCAD-penalty function into
generalized estimating equations. 
</p>


<h3>Details</h3>

<p>This package consists of three functions. The function
<code><a href="#topic+PGEE">PGEE</a></code> fits penalized generalized estimating equations to the data. But, before that,
the tuning parameter should be estimated through the function <code><a href="#topic+CVfit">CVfit</a></code>. 
On the other hand, the function <code><a href="#topic+MGEE">MGEE</a></code> fits unpenalized generalized estimating 
equations to the data.
</p>


<h3>Author(s)</h3>

<p>Gul Inan, Jianhui Zhou and Lan Wang
</p>
<p>Maintainer: Gul Inan
</p>


<h3>References</h3>

<p>Wang, L., Zhou, J., and Qu, A. (2012). Penalized generalized estimating equations 
for high-dimensional longitudinal data analysis. <em>Biometrics</em>, <b>68</b>, 353&ndash;360.
</p>

<hr>
<h2 id='CVfit'>Function to compute cross-validated tuning parameter value</h2><span id='topic+CVfit'></span><span id='topic+print.CVfit'></span>

<h3>Description</h3>

<p>This function computes cross-validated tuning parameter value for 
longitudinal data with working independence structure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CVfit(formula, id, data, family, scale.fix, scale.value, fold, lambda.vec, pindex,  
eps, maxiter, tol)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CVfit_+3A_formula">formula</code></td>
<td>
<p>A formula expression in the form of <code>response ~ predictors</code>.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_id">id</code></td>
<td>
<p>A vector for identifying subjects/clusters.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_data">data</code></td>
<td>
<p>A data frame which stores the variables in <code>formula</code> with <code>id</code> variable.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_scale.fix">scale.fix</code></td>
<td>
<p>A logical variable; if true, the scale parameter is fixed at the value of <code>scale.value</code>. 
The default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_scale.value">scale.value</code></td>
<td>
<p>If <code>scale.fix = TRUE</code>, this assignes a numeric value to which the scale parameter should be 
fixed. The default value is 1.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_family">family</code></td>
<td>
<p>A <code>family</code> object in <code><a href="#topic+PGEE">PGEE</a></code>.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_fold">fold</code></td>
<td>
<p>The number of folds used in cross-validation.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_lambda.vec">lambda.vec</code></td>
<td>
<p>A vector of tuning parameters that will be used in the cross-validation.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_pindex">pindex</code></td>
<td>
<p>An index vector showing the parameters which are not subject to penalization. The default value
is <code>NULL</code>. However, in case of a model with intercept, the intercept parameter should be never penalized.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_eps">eps</code></td>
<td>
<p>A numerical value for the epsilon used in minorization-maximization algorithm. The default value is 
<code>10^-6</code>.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_maxiter">maxiter</code></td>
<td>
<p>The number of iterations that is used in the estimation algorithm. The default value is <code>25</code>.</p>
</td></tr>
<tr><td><code id="CVfit_+3A_tol">tol</code></td>
<td>
<p>The tolerance level that is used in the estimation algorithm. The default value is <code>10^-3</code>.</p>
</td></tr> 
</table>


<h3>Value</h3>

<p>An object class of <code>CVfit</code>.</p>


<h3>References</h3>

<p>Wang, L., Zhou, J., and Qu, A. (2012). Penalized generalized estimating equations 
for high-dimensional longitudinal data analysis. <em>Biometrics</em>, <b>68</b>, 353&ndash;360.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PGEE">PGEE</a></code>
</p>

<hr>
<h2 id='MGEE'>Function to fit generalized estimating equations</h2><span id='topic+MGEE'></span><span id='topic+MGee'></span><span id='topic+mycor_gee1'></span><span id='topic+S_H_M'></span><span id='topic+print.MGEE'></span><span id='topic+print.summary.MGEE'></span><span id='topic+summary.MGEE'></span><span id='topic+print.MGee'></span><span id='topic+print.summary.MGee'></span><span id='topic+summary.MGee'></span>

<h3>Description</h3>

<p>This function fits a generalized estimating equation model to longitudinal data.</p>


<h3>Usage</h3>

<pre><code class='language-R'>MGEE(formula, id, data, na.action = NULL, family = gaussian(link = "identity"), 
corstr = "independence", Mv = NULL, beta_int = NULL, R = NULL, scale.fix = TRUE, 
scale.value = 1, maxiter = 25, tol = 10^-3, silent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MGEE_+3A_formula">formula</code></td>
<td>
<p>A formula expression in the form of <code>response ~ predictors</code>.</p>
</td></tr>
<tr><td><code id="MGEE_+3A_id">id</code></td>
<td>
<p>A vector for identifying subjects/clusters.</p>
</td></tr>
<tr><td><code id="MGEE_+3A_data">data</code></td>
<td>
<p>A data frame which stores the variables in <code>formula</code> with <code>id</code> variable.</p>
</td></tr>
<tr><td><code id="MGEE_+3A_na.action">na.action</code></td>
<td>
<p>A function to remove missing values from the data. Only <code>na.omit</code> is allowed here.</p>
</td></tr>
<tr><td><code id="MGEE_+3A_family">family</code></td>
<td>
<p>A <code>family</code> object: a list of functions and expressions for defining <code>link</code> and 
<code>variance</code> functions. Families supported in <code>MGEE</code> are <code>binomial</code>, <code>gaussian</code>, <code>gamma</code> and 
<code>poisson</code>. The <code>links</code>, which are not available in <code>gee</code>, is not available here. The default family 
is <code>gaussian</code>.</p>
</td></tr> 
<tr><td><code id="MGEE_+3A_corstr">corstr</code></td>
<td>
<p>A character string, which specifies the type of correlation structure. 
Structures supported in <code>MGEE</code> are <code>"AR-1"</code>,<code>"exchangeable"</code>, <code>"fixed"</code>, <code>"independence"</code>,
<code>"stat_M_dep"</code>,<code>"non_stat_M_dep"</code>, and <code>"unstructured"</code>. The default <code>corstr</code> type is 
<code>"independence"</code>.</p>
</td></tr> 
<tr><td><code id="MGEE_+3A_mv">Mv</code></td>
<td>
<p>If either <code>"stat_M_dep"</code>, or <code>"non_stat_M_dep"</code> is specified in <code>corstr</code>, then this assigns 
a numeric value for <code>Mv</code>. Otherwise, the default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="MGEE_+3A_beta_int">beta_int</code></td>
<td>
<p>User specified initial values for regression parameters. The default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="MGEE_+3A_r">R</code></td>
<td>
<p>If <code>corstr = "fixed"</code> is specified, then <code>R</code> is a square matrix of dimension maximum cluster 
size containing the user specified correlation. Otherwise, the default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="MGEE_+3A_scale.fix">scale.fix</code></td>
<td>
<p>A logical variable; if true, the scale parameter is fixed at the value of <code>scale.value</code>. 
The default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="MGEE_+3A_scale.value">scale.value</code></td>
<td>
<p>If <code>scale.fix = TRUE</code>, this assignes a numeric value to which the scale parameter should be fixed. 
The default value is 1.</p>
</td></tr>
<tr><td><code id="MGEE_+3A_maxiter">maxiter</code></td>
<td>
<p>The number of iterations that is used in the estimation algorithm. The default value is <code>25</code>.</p>
</td></tr>
<tr><td><code id="MGEE_+3A_tol">tol</code></td>
<td>
<p>The tolerance level that is used in the estimation algorithm. The default value is <code>10^-3</code>.</p>
</td></tr>
<tr><td><code id="MGEE_+3A_silent">silent</code></td>
<td>
<p>A logical variable; if false, the regression parameter estimates at each iteration are
printed. The default value is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object class of <code>MGEE</code> representing the fit.</p>


<h3>Note</h3>

<p>The structures <code>"non_stat_M_dep"</code> and <code>"unstructured"</code> are valid only when the data is balanced.
</p>


<h3>References</h3>

<p>Liang, K.Y. and Zeger, S.L. (1986).
Longitudinal data analysis using generalized linear models.
<em>Biometrika</em>, <b>73</b>, 13&ndash;22. 
</p>
<p>Zeger, S.L. and Liang, K.Y. (1986)
.  Longitudinal data analysis for discrete and continuous outcomes.
<em>Biometrics</em>, <b>42</b>, 121&ndash;130.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PGEE">PGEE</a></code>
</p>

<hr>
<h2 id='PGEE'>Function to fit penalized generalized estimating equations</h2><span id='topic+PGEE'></span><span id='topic+PGee'></span><span id='topic+mycor_gee2'></span><span id='topic+q_scad'></span><span id='topic+S_H_E_M'></span><span id='topic+print.PGEE'></span><span id='topic+print.summary.PGEE'></span><span id='topic+summary.PGEE'></span><span id='topic+print.PGee'></span><span id='topic+print.summary.PGee'></span><span id='topic+summary.PGee'></span>

<h3>Description</h3>

<p>This function fits a penalized generalized estimating equation model to longitudinal data.</p>


<h3>Usage</h3>

<pre><code class='language-R'>PGEE(formula, id, data, na.action = NULL, family = gaussian(link = "identity"), 
corstr = "independence", Mv = NULL, beta_int = NULL, R = NULL, scale.fix = TRUE, 
scale.value = 1, lambda, pindex = NULL, eps = 10^-6, maxiter = 30, tol = 10^-3, 
silent = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PGEE_+3A_formula">formula</code></td>
<td>
<p>A formula expression in the form of <code>response ~ predictors</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_id">id</code></td>
<td>
<p>A vector for identifying subjects/clusters.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_data">data</code></td>
<td>
<p>A data frame which stores the variables in <code>formula</code> with <code>id</code> variable.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_na.action">na.action</code></td>
<td>
<p>A function to remove missing values from the data. Only <code>na.omit</code> is allowed here.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_family">family</code></td>
<td>
<p>A <code>family</code> object: a list of functions and expressions for defining <code>link</code> and 
<code>variance</code> functions. Families supported in <code>PGEE</code> are <code>binomial</code>, <code>gaussian</code>, <code>gamma</code> and 
<code>poisson</code>. The <code>links</code>, which are not available in <code>gee</code>, is not available here. The default family 
is <code>gaussian</code>.</p>
</td></tr> 
<tr><td><code id="PGEE_+3A_corstr">corstr</code></td>
<td>
<p>A character string, which specifies the type of correlation structure. 
Structures supported in <code>PGEE</code> are <code>"AR-1"</code>,<code>"exchangeable"</code>, <code>"fixed"</code>, <code>"independence"</code>,
<code>"stat_M_dep"</code>,<code>"non_stat_M_dep"</code>, and <code>"unstructured"</code>. The default <code>corstr</code> type is 
<code>"independence"</code>.</p>
</td></tr> 
<tr><td><code id="PGEE_+3A_mv">Mv</code></td>
<td>
<p>If either <code>"stat_M_dep"</code>, or <code>"non_stat_M_dep"</code> is specified in <code>corstr</code>, then this 
assigns a numeric value for <code>Mv</code>. Otherwise, the default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_beta_int">beta_int</code></td>
<td>
<p>User specified initial values for regression parameters. The default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_r">R</code></td>
<td>
<p>If <code>corstr = "fixed"</code> is specified, then <code>R</code> is a square matrix of dimension maximum cluster 
size containing the user specified correlation. Otherwise, the default value is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_scale.fix">scale.fix</code></td>
<td>
<p>A logical variable; if true, the scale parameter is fixed at the value of <code>scale.value</code>. 
The default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_scale.value">scale.value</code></td>
<td>
<p>If <code>scale.fix = TRUE</code>, this assignes a numeric value to which the scale parameter should be 
fixed. The default value is 1.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_lambda">lambda</code></td>
<td>
<p>A numerical value for the penalization parameter of the scad function, which is estimated via 
cross-validation.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_pindex">pindex</code></td>
<td>
<p>An index vector showing the parameters which are not subject to penalization. The default value
is <code>NULL</code>. However, in case of a model with intercept, the intercept parameter should be never penalized.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_eps">eps</code></td>
<td>
<p>A numerical value for the epsilon used in minorization-maximization algorithm. The default value is 
<code>10^-6</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_maxiter">maxiter</code></td>
<td>
<p>The number of iterations that is used in the estimation algorithm. The default value is <code>25</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_tol">tol</code></td>
<td>
<p>The tolerance level that is used in the estimation algorithm. The default value is <code>10^-3</code>.</p>
</td></tr>
<tr><td><code id="PGEE_+3A_silent">silent</code></td>
<td>
<p>A logical variable; if false, the regression parameter estimates at each iteration are
printed. The default value is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object class of <code>PGEE</code> representing the fit.</p>


<h3>References</h3>

<p>Wang, L., Zhou, J., and Qu, A. (2012). Penalized generalized estimating equations 
for high-dimensional longitudinal data analysis. <em>Biometrics</em>, <b>68</b>, 353&ndash;360.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CVfit">CVfit</a></code>, <code><a href="#topic+MGEE">MGEE</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Consider an example similar to example 1 
# in Wang et al. (2012).

# required R package 
library(mvtnorm)
# number of subjects
n &lt;- 200
# number of covariates 
pn &lt;- 10
# number of time points
m &lt;- 4

# vector if subject ids
id.vect &lt;- rep(1:n, each = m) 

# covariance matrix of (pn-1) number of continuous covariates 
X.sigma &lt;- matrix(0,(pn-1),(pn-1))
{
for (i in 1:(pn-1))
X.sigma[i,] &lt;- 0.5^(abs((1:(pn-1))-i))  
}

# generate matrix of covariates    
x.mat &lt;- as.matrix(rmvnorm(n*m, mean = rep(0,(pn-1)), X.sigma))
x.mat &lt;- cbind(rbinom(n*m,1, 0.5), x.mat)

# true values
beta.true &lt;- c(2,3,1.5,2,rep(0,6))
sigma2 &lt;- 1
rho &lt;- 0.5
R &lt;- matrix(rho,m,m)+diag(rep(1-rho,m))

# covariance matrix of error
SIGMA &lt;- sigma2*R
error &lt;- rmvnorm(n, mean = rep(0,m),SIGMA)

# generate longitudinal data with continuous outcomes
y.temp &lt;- x.mat%*%beta.true
y.vect &lt;- y.temp+as.vector(t(error))

mydata &lt;- data.frame(id.vect,y.vect,x.mat) 
colnames(mydata) &lt;- c("id","y",paste("x",1:length(beta.true),sep = ""))

###Input Arguments for CVfit fitting###
library(PGEE)
formula &lt;- "y ~.-id-1"
data &lt;- mydata
family &lt;- gaussian(link = "identity")
lambda.vec &lt;- seq(0.1,1,0.1)

## Not run: 
cv &lt;- CVfit(formula = formula, id = id, data = data, family = family,
fold = 4, lambda.vec = lambda.vec, pindex = NULL, eps = 10^-6, maxiter = 30, 
tol = 10^-3)

names(cv)
cv$lam.opt

## End(Not run)

lambda &lt;- 0.1 #this value obtained through CVfit

# analyze the data through penalized generalized estimating equations

myfit1 &lt;- PGEE(formula = formula, id = id, data = data, na.action = NULL, 
family = family, corstr = "exchangeable", Mv = NULL, 
beta_int = c(rep(0,length(beta.true))), R = NULL, scale.fix = TRUE, 
scale.value = 1, lambda = lambda, pindex = NULL, eps = 10^-6, maxiter = 30, 
tol = 10^-3, silent = TRUE)

summary(myfit1)

# analyze the data through unpenalized generalized estimating equations

myfit2 &lt;- MGEE(formula = formula, id = id, data = data, na.action = NULL, 
family = family, corstr = "exchangeable", Mv = NULL, 
beta_int = c(rep(0,length(beta.true))), R = NULL, scale.fix = TRUE, 
scale.value = 1, maxiter = 30, tol = 10^-3, silent = TRUE)

summary(myfit2)
</code></pre>

<hr>
<h2 id='yeastG1'>Yeast cell-cycle gene expression data</h2><span id='topic+yeastG1'></span>

<h3>Description</h3>

<p>A yeast cell-cycle gene expression data set collected in the CDC15 experiment of 
Spellman et al. (1998) where genome-wide mRNA levels of 6178 yeast 
open reading frames (ORFs) in a two cell-cycle period were measured at M/G1-G1-S-G2-M stages.
However, to better understand the phenomenon underlying cell-cycle process, it is important to identify transcription 
factors (TFs) that regulate the gene expression levels of cell cycle-regulated genes. In this study, 
we presented a subset of 283 cell-cycled-regularized genes observed over 4 time points at G1 stage and 
the standardized binding probabilities of a total of 96 TFs obtained from a mixture model approach of 
Wang et al. (2007) based on the ChIP data of Lee et al. (2002).</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("yeastG1")</code></pre>


<h3>Details</h3>

<p>A data frame with 1132 observations (283 cell-cycled-regularized genes observed over 4 time points) 
with 99 variables (e.g., id, y, time, and 96 TFs).</p>


<h3>References</h3>

<p>Lee, T.I., Rinaldi, N.J., Robert, F., Odom, D.T., Bar-Joseph, Z., Gerber, G.K., Hannett, N.M., 
Harbison, C.T., Thompson, C.M., Simon, I., et al. (2002). Transcriptional regulatory networks in
Saccharomyces cerevisiae. <em>Science</em>, <b>298</b>, 799&ndash;804.
</p>
<p>Spellman, P.T., Sherlock, G., Zhang, M.Q., Iyer, V.R., Anders, K., Eisen, M.B., Brown, P.O., 
Botstein, D., and Futcher, B. (1998). Comprehensive identification of cell cycle regulated genes of the yeast
Saccharomyces cerevisiae by microarray hybridization. <em>Molecular Biology of Cell</em>, <b>9</b>,
3273&ndash;3297.
</p>
<p>Wang, L., Chen, G., and Li, H. (2007). Group SCAD regression analysis for microarray time course
gene expression data. <em>Bioinformatics</em>, <b>23</b>, 1486&ndash;1494.
</p>
<p>Wang, L., Zhou, J., and Qu, A. (2012). Penalized generalized estimating equations 
for high-dimensional longitudinal data anaysis. <em>Biometrics</em>, <b>68</b>, 353&ndash;360.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(PGEE)
# load data
data(yeastG1)
data &lt;- yeastG1
# get the column names
colnames(data)[1:9]
# see some portion of yeast G1 data
head(data,5)[1:9]

# define the input arguments
formula &lt;- "y ~.-id"
family &lt;- gaussian(link = "identity")
lambda.vec &lt;- seq(0.01,0.2,0.01)
# find the optimum lambda
cv &lt;- CVfit(formula = formula, id = id, data = data, family = family, scale.fix = TRUE,
scale.value = 1, fold = 4, lambda.vec = lambda.vec, pindex = c(1,2), eps = 10^-6,
maxiter = 30, tol = 10^-6)
# print the results
print(cv)

# see the returned values by CVfit
names(cv)
# get the optimum lambda
cv$lam.opt

#fit the PGEE model
myfit1 &lt;- PGEE(formula = formula, id = id, data = data, na.action = NULL,
family = family, corstr = "independence", Mv = NULL,
beta_int = c(rep(0,dim(data)[2]-1)), R = NULL, scale.fix = TRUE,
scale.value = 1, lambda = cv$lam.opt, pindex = c(1,2), eps = 10^-6, 
maxiter = 30, tol = 10^-6, silent = TRUE)

# get the values returned by myfit object
names(myfit1)
# get the values returned by summary(myfit) object
names(summary(myfit1))
# see a portion of the results returned by coef(summary(myfit1))
head(coef(summary(myfit1)),7)

# see the variables which have non-zero coefficients
index1 &lt;- which(abs(coef(summary(myfit1))[,"Estimate"]) &gt; 10^-3)
names(abs(coef(summary(myfit1))[index1,"Estimate"]))

# see the PGEE summary statistics of these non-zero variables
coef(summary(myfit1))[index1,]

# fit the GEE model
myfit2 &lt;- MGEE(formula = formula, id = id, data = data, na.action = NULL,
family = family, corstr = "independence", Mv = NULL,
beta_int = c(rep(0,dim(data)[2]-1)), R = NULL, scale.fix = TRUE,
scale.value = 1, maxiter = 30, tol = 10^-6, silent = TRUE)

# get the GEE summary statistics of the variables that turned out to be
# non-zero in PGEE analysis
coef(summary(myfit2))[index1,]

# see the significantly associated TFs in PGEE analysis
names(which(abs(coef(summary(myfit1))[index1,"Robust z"]) &gt; 1.96))

# see the significantly associated TFs in GEE analysis
names(which(abs(coef(summary(myfit2))[,"Robust z"]) &gt; 1.96))


## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
