<!DOCTYPE html><html lang="en"><head><title>Help for package pGRN</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {pGRN}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#data_transform'><p>Pseudotime based Expression Data Transformation</p></a></li>
<li><a href='#get_dtw_dist_bidirectional'><p>Bidirectional DTW Distance</p></a></li>
<li><a href='#get_dtw_dist_mat'><p>DTW distance matrix for all genes</p></a></li>
<li><a href='#get_networks'><p>Get the list of sub-networks</p></a></li>
<li><a href='#matrix2adj'><p>Convert distance matrix to adjacency dataframe</p></a></li>
<li><a href='#module_networks'><p>Get module level networks</p></a></li>
<li><a href='#pGRN'><p>pGRN: creates gene regulatory network based on single cell pseudotime information</p></a></li>
<li><a href='#pGRNDB'><p>pGRN example data</p></a></li>
<li><a href='#plot_network'><p>Plot stationary network</p></a></li>
<li><a href='#plot_network_i'><p>Plot interactive network</p></a></li>
<li><a href='#run_dtw'><p>Get network adjacency dataframe based on DTW method</p></a></li>
<li><a href='#run_granger_test'><p>Use Granger-causality Test to get gene-gene regulatory relationship</p></a></li>
<li><a href='#slideWindows'><p>Sliding Window Average</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Single-Cell RNA Sequencing Pseudo-Time Based Gene Regulatory
Network Inference</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.5</td>
</tr>
<tr>
<td>Author:</td>
<td>Gangcai Xie</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gangcai Xie &lt;gcxiester@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Inference and visualize gene regulatory network based on single-cell RNA sequencing pseudo-time information.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10),</td>
</tr>
<tr>
<td>Imports:</td>
<td>bigmemory, doParallel, dtw, foreach, ggplot2, ggraph, igraph,
lmtest, proxy, tidygraph, visNetwork, future</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, webshot, rmarkdown,</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-01-16 14:36:55 UTC; gangcai</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-01-17 17:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='data_transform'>Pseudotime based Expression Data Transformation</h2><span id='topic+data_transform'></span>

<h3>Description</h3>

<p>Based on single-cell pseudotime information, get the sliding window average expression,
and then standard normlize the expression for each gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_transform(data, pseudotime, slide_window_size = 100, slide_step_size = 50)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="data_transform_+3A_data">data</code></td>
<td>
<p>expression matrix data</p>
</td></tr>
<tr><td><code id="data_transform_+3A_pseudotime">pseudotime</code></td>
<td>
<p>list of pseudotime</p>
</td></tr>
<tr><td><code id="data_transform_+3A_slide_window_size">slide_window_size</code></td>
<td>
<p>sliding window size</p>
</td></tr>
<tr><td><code id="data_transform_+3A_slide_step_size">slide_step_size</code></td>
<td>
<p>sliding window step size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Transformed new matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- matrix(1,100,1000)
ptime &lt;- seq(1:1000)
data_transform(data,
               ptime,
               slide_window_size=100,
               slide_step_size=50)

</code></pre>

<hr>
<h2 id='get_dtw_dist_bidirectional'>Bidirectional DTW Distance</h2><span id='topic+get_dtw_dist_bidirectional'></span>

<h3>Description</h3>

<p>Get bidirectional DTW distance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_dtw_dist_bidirectional(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_dtw_dist_bidirectional_+3A_x">x</code></td>
<td>
<p>list of x input</p>
</td></tr>
<tr><td><code id="get_dtw_dist_bidirectional_+3A_y">y</code></td>
<td>
<p>list of y input</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_dtw_dist_bidirectional(c(1:1000),c(1:1000))

</code></pre>

<hr>
<h2 id='get_dtw_dist_mat'>DTW distance matrix for all genes</h2><span id='topic+get_dtw_dist_mat'></span>

<h3>Description</h3>

<p>Get DTW distance matrix for all genes using pseudotime based sliding window transfromation,
parallel computing allowed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_dtw_dist_mat(
  data,
  ptime,
  slide_window_size = 50,
  slide_step_size = 25,
  cores = 2
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_dtw_dist_mat_+3A_data">data</code></td>
<td>
<p>gene expression matrix (Gene * Cells)</p>
</td></tr>
<tr><td><code id="get_dtw_dist_mat_+3A_ptime">ptime</code></td>
<td>
<p>pseudotime matched with the column cells of the gene expression matrix</p>
</td></tr>
<tr><td><code id="get_dtw_dist_mat_+3A_slide_window_size">slide_window_size</code></td>
<td>
<p>sliding window size</p>
</td></tr>
<tr><td><code id="get_dtw_dist_mat_+3A_slide_step_size">slide_step_size</code></td>
<td>
<p>sliding window step size</p>
</td></tr>
<tr><td><code id="get_dtw_dist_mat_+3A_cores">cores</code></td>
<td>
<p>number of cores for parallel computing</p>
</td></tr>
</table>


<h3>Value</h3>

<p>bidirectional DTW distance matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_data &lt;- pGRNDB
expression_matrix &lt;- example_data[["expression"]]
pseudotime_list &lt;- example_data[["ptime"]]$PseudoTime
dtw_dist_matrix &lt;- get_dtw_dist_mat(expression_matrix,
                                    pseudotime_list,
                                    cores=1)

</code></pre>

<hr>
<h2 id='get_networks'>Get the list of sub-networks</h2><span id='topic+get_networks'></span>

<h3>Description</h3>

<p>Get sub-networks based on given adjacency data.frame input
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_networks(
  data,
  centrality_degree_mod = "out",
  components_mod = "weak",
  network_min_genes = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_networks_+3A_data">data</code></td>
<td>
<p>adjacency data.frame</p>
</td></tr>
<tr><td><code id="get_networks_+3A_centrality_degree_mod">centrality_degree_mod</code></td>
<td>
<p>mode of centrality degree for popularity calculation</p>
</td></tr>
<tr><td><code id="get_networks_+3A_components_mod">components_mod</code></td>
<td>
<p>mode of sub-network extraction methods</p>
</td></tr>
<tr><td><code id="get_networks_+3A_network_min_genes">network_min_genes</code></td>
<td>
<p>minimal number of gene elements required for extracted sub-networks</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of tabl_graph objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_data &lt;- pGRNDB
expression_matrix &lt;- example_data[["expression"]]
pseudotime_list &lt;- example_data[["ptime"]]$PseudoTime
dtw_dist_matrix &lt;- get_dtw_dist_mat(expression_matrix,
                                    pseudotime_list,
                                    cores=1)
adj_df &lt;- matrix2adj(dtw_dist_matrix)
get_networks(adj_df,network_min_genes=5)
</code></pre>

<hr>
<h2 id='matrix2adj'>Convert distance matrix to adjacency dataframe</h2><span id='topic+matrix2adj'></span>

<h3>Description</h3>

<p>Convert distance matrix to adjacency dataframe for network construction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matrix2adj(data, quantile_cutoff = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="matrix2adj_+3A_data">data</code></td>
<td>
<p>distance matrix</p>
</td></tr>
<tr><td><code id="matrix2adj_+3A_quantile_cutoff">quantile_cutoff</code></td>
<td>
<p>an integer value (1-99) for quantile cutoff</p>
</td></tr>
</table>


<h3>Value</h3>

<p>adjacency dataframe (with columns &quot;from, to, distance,direction, similarity&quot;)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_data &lt;- pGRNDB
expression_matrix &lt;- example_data[["expression"]]
pseudotime_list &lt;- example_data[["ptime"]]$PseudoTime
dtw_dist_matrix &lt;- get_dtw_dist_mat(expression_matrix,
                                    pseudotime_list,
                                    cores=1)
adj_df &lt;- matrix2adj(dtw_dist_matrix)

</code></pre>

<hr>
<h2 id='module_networks'>Get module level networks</h2><span id='topic+module_networks'></span>

<h3>Description</h3>

<p>Given a distance matrix, calculate gene modules based on hierarchical clustering method and then get module level networks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>module_networks(
  data,
  k = 10,
  quantile_cutoff = 10,
  centrality_degree_mod = "out",
  components_mod = "weak",
  network_min_genes = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="module_networks_+3A_data">data</code></td>
<td>
<p>distance matrix</p>
</td></tr>
<tr><td><code id="module_networks_+3A_k">k</code></td>
<td>
<p>number of gene clusters for module inference</p>
</td></tr>
<tr><td><code id="module_networks_+3A_quantile_cutoff">quantile_cutoff</code></td>
<td>
<p>distance cutoff based on quantile(1-99) for edge identification</p>
</td></tr>
<tr><td><code id="module_networks_+3A_centrality_degree_mod">centrality_degree_mod</code></td>
<td>
<p>&quot;in&quot; or &quot;out&quot; for nodes popularity calculation</p>
</td></tr>
<tr><td><code id="module_networks_+3A_components_mod">components_mod</code></td>
<td>
<p>&quot;weak&quot; or &quot;strong&quot; for sub-network components inference</p>
</td></tr>
<tr><td><code id="module_networks_+3A_network_min_genes">network_min_genes</code></td>
<td>
<p>minial number of genes required for a network</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list networks for each module
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_data &lt;- pGRNDB
expression_matrix &lt;- example_data[["expression"]]
pseudotime_list &lt;- example_data[["ptime"]]$PseudoTime
dtw_dist_matrix &lt;- get_dtw_dist_mat(expression_matrix,
                                    pseudotime_list,
                                    cores=1)
nets &lt;- module_networks(dtw_dist_matrix,k=1,quantile_cutoff=50)
plot_network(nets[["module1"]])
</code></pre>

<hr>
<h2 id='pGRN'>pGRN: creates gene regulatory network based on single cell pseudotime information</h2><span id='topic+pGRN'></span>

<h3>Description</h3>

<p>Given single cell matrix and pseudotime, construct gene regulatory network (GRN)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pGRN(
  expression_matrix,
  pseudotime_list,
  method = "DTW",
  slide_window_size = 20,
  slide_step_size = 10,
  centrality_degree_mod = "out",
  components_mod = "weak",
  network_min_genes = 10,
  quantile_cutoff = 5,
  order = 1,
  cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pGRN_+3A_expression_matrix">expression_matrix</code></td>
<td>
<p>expression matrix data</p>
</td></tr>
<tr><td><code id="pGRN_+3A_pseudotime_list">pseudotime_list</code></td>
<td>
<p>list of pseudotime</p>
</td></tr>
<tr><td><code id="pGRN_+3A_method">method</code></td>
<td>
<p>method for GRN construction: DTW, granger</p>
</td></tr>
<tr><td><code id="pGRN_+3A_slide_window_size">slide_window_size</code></td>
<td>
<p>sliding window size</p>
</td></tr>
<tr><td><code id="pGRN_+3A_slide_step_size">slide_step_size</code></td>
<td>
<p>sliding window step size</p>
</td></tr>
<tr><td><code id="pGRN_+3A_centrality_degree_mod">centrality_degree_mod</code></td>
<td>
<p>(for DTW method) mode of centrality degree for popularity calculation</p>
</td></tr>
<tr><td><code id="pGRN_+3A_components_mod">components_mod</code></td>
<td>
<p>(for DTW method) mode of sub-network extraction methods (weak or strong)</p>
</td></tr>
<tr><td><code id="pGRN_+3A_network_min_genes">network_min_genes</code></td>
<td>
<p>minimal number of gene elements required for extracted sub-networks</p>
</td></tr>
<tr><td><code id="pGRN_+3A_quantile_cutoff">quantile_cutoff</code></td>
<td>
<p>an integer value (1-99) for quantile cutoff</p>
</td></tr>
<tr><td><code id="pGRN_+3A_order">order</code></td>
<td>
<p>(for granger method) integer specifying the order of lags to include in the auxiliary regression</p>
</td></tr>
<tr><td><code id="pGRN_+3A_cores">cores</code></td>
<td>
<p>number of cores for parallel computing</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of tabl_graph objects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_data &lt;- pGRNDB
expression_matrix &lt;- example_data[["expression"]]
pseudotime_list &lt;- example_data[["ptime"]]$PseudoTime

# try DTW method
nets &lt;- pGRN(expression_matrix,
             pseudotime_list, 
             method= "DTW",
             quantile_cutoff=50,
             cores=1)
plot_network(nets[[1]])

# plot the network interactively
plot_network_i(nets[[1]])

</code></pre>

<hr>
<h2 id='pGRNDB'>pGRN example data</h2><span id='topic+pGRNDB'></span>

<h3>Description</h3>

<p>A list with expression dataframe and pseudotime dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pGRNDB
</code></pre>


<h3>Format</h3>



<h4><code>pGRNDB</code></h4>

<p>A list with items expression and ptime
</p>

<dl>
<dt>expression</dt><dd><p>data frame of single cell expression</p>
</dd>
<dt>ptime</dt><dd><p>pseudotime of the single cells</p>
</dd>
</dl>
<p>...

</p>



<h3>Source</h3>

<p>pGRN
</p>

<hr>
<h2 id='plot_network'>Plot stationary network</h2><span id='topic+plot_network'></span>

<h3>Description</h3>

<p>Plot stationary network through ggraph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_network(graph, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_network_+3A_graph">graph</code></td>
<td>
<p>a tbl_graph object</p>
</td></tr>
<tr><td><code id="plot_network_+3A_...">...</code></td>
<td>
<p>other parameters for ggraph</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggraph
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_data &lt;- pGRNDB
expression_matrix &lt;- example_data[["expression"]]
pseudotime_list &lt;- example_data[["ptime"]]$PseudoTime
dtw_dist_matrix &lt;- get_dtw_dist_mat(expression_matrix,
                                    pseudotime_list,
                                    cores=1)
nets &lt;- module_networks(dtw_dist_matrix,k=1,quantile_cutoff=50)
plot_network(nets[["module1"]])
</code></pre>

<hr>
<h2 id='plot_network_i'>Plot interactive network</h2><span id='topic+plot_network_i'></span>

<h3>Description</h3>

<p>Plot interactive network based on igraph layout input
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_network_i(graph, save_file = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_network_i_+3A_graph">graph</code></td>
<td>
<p>igraph layout object</p>
</td></tr>
<tr><td><code id="plot_network_i_+3A_save_file">save_file</code></td>
<td>
<p>file name of the saved file, not save if NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>visNetwork htmlwidget
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_data &lt;- pGRNDB
expression_matrix &lt;- example_data[["expression"]]
pseudotime_list &lt;- example_data[["ptime"]]$PseudoTime
dtw_dist_matrix &lt;- get_dtw_dist_mat(expression_matrix,
                                    pseudotime_list,
                                    cores=1)
nets &lt;- module_networks(dtw_dist_matrix,k=1,quantile_cutoff=50)
plot_network_i(nets[["module1"]])
</code></pre>

<hr>
<h2 id='run_dtw'>Get network adjacency dataframe based on DTW method</h2><span id='topic+run_dtw'></span>

<h3>Description</h3>

<p>Use DTW to calcuate gene-gene distance based on their expression and pseudotime
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_dtw(
  expression_matrix,
  pseudotime_list,
  slide_window_size = 50,
  slide_step_size = 25,
  quantile_cutoff = 5,
  cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_dtw_+3A_expression_matrix">expression_matrix</code></td>
<td>
<p>expression matrix data</p>
</td></tr>
<tr><td><code id="run_dtw_+3A_pseudotime_list">pseudotime_list</code></td>
<td>
<p>list of pseudotime</p>
</td></tr>
<tr><td><code id="run_dtw_+3A_slide_window_size">slide_window_size</code></td>
<td>
<p>sliding window size</p>
</td></tr>
<tr><td><code id="run_dtw_+3A_slide_step_size">slide_step_size</code></td>
<td>
<p>sliding window step size</p>
</td></tr>
<tr><td><code id="run_dtw_+3A_quantile_cutoff">quantile_cutoff</code></td>
<td>
<p>an integer value (1-99) for quantile cutoff</p>
</td></tr>
<tr><td><code id="run_dtw_+3A_cores">cores</code></td>
<td>
<p>number of cores for parallel computing</p>
</td></tr>
</table>


<h3>Value</h3>

<p>adjacency dataframe (with columns &quot;from, to, distance,direction, similarity&quot;)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>example_data &lt;- pGRNDB
expression_matrix &lt;- example_data[["expression"]]
pseudotime_list &lt;- example_data[["ptime"]]$PseudoTime
adj_df &lt;- run_dtw(expression_matrix,
                  pseudotime_list,
                  quantile_cutoff=50,
                  cores=1)
                  
</code></pre>

<hr>
<h2 id='run_granger_test'>Use Granger-causality Test to get gene-gene regulatory relationship</h2><span id='topic+run_granger_test'></span>

<h3>Description</h3>

<p>Based on single-cell gene expression matrix and pseudotime, calculate Granger-causality Test
based gene-gene regulatory relationship
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_granger_test(
  data,
  ptime,
  slide_window_size = 20,
  slide_step_size = 10,
  pvalue_cutoff = 0.01,
  order = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="run_granger_test_+3A_data">data</code></td>
<td>
<p>gene expression matrix (Gene * Cells)</p>
</td></tr>
<tr><td><code id="run_granger_test_+3A_ptime">ptime</code></td>
<td>
<p>pseudotime matched with the column cells of the gene expression matrix</p>
</td></tr>
<tr><td><code id="run_granger_test_+3A_slide_window_size">slide_window_size</code></td>
<td>
<p>sliding window size</p>
</td></tr>
<tr><td><code id="run_granger_test_+3A_slide_step_size">slide_step_size</code></td>
<td>
<p>sliding window step size</p>
</td></tr>
<tr><td><code id="run_granger_test_+3A_pvalue_cutoff">pvalue_cutoff</code></td>
<td>
<p>cutoff for the pvalue from transfer entropy test</p>
</td></tr>
<tr><td><code id="run_granger_test_+3A_order">order</code></td>
<td>
<p>integer specifying the order of lags to include in the auxiliary regression</p>
</td></tr>
<tr><td><code id="run_granger_test_+3A_...">...</code></td>
<td>
<p>other parameters for grangertest function in lmtest</p>
</td></tr>
</table>


<h3>Value</h3>

<p>adjacency data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
example_data &lt;- pGRNDB
expression_matrix &lt;- example_data[["expression"]]
pseudotime_list &lt;- example_data[["ptime"]]$PseudoTime
gt_adj_df &lt;- run_granger_test(expression_matrix, pseudotime_list)

</code></pre>

<hr>
<h2 id='slideWindows'>Sliding Window Average</h2><span id='topic+slideWindows'></span>

<h3>Description</h3>

<p>Get sliding windows average values for given vector/list
</p>


<h3>Usage</h3>

<pre><code class='language-R'>slideWindows(data, window = 2, step = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="slideWindows_+3A_data">data</code></td>
<td>
<p>list of expression</p>
</td></tr>
<tr><td><code id="slideWindows_+3A_window">window</code></td>
<td>
<p>sliding window size</p>
</td></tr>
<tr><td><code id="slideWindows_+3A_step">step</code></td>
<td>
<p>sliding window step size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list/vector of sliding windows with average expression value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>slideWindows(c(1:1000),window=200,step=100)
slideWindows(c(1:1000),window=100,step=50)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
