<!DOCTYPE html><html><head><title>Help for package PhenotypeSimulator</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {PhenotypeSimulator}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#addNonNulls'><p>Add all non-NULL elements of list.</p></a></li>
<li><a href='#commaList2vector'><p>Comma-separated string to numeric vector.</p></a></li>
<li><a href='#correlatedBgEffects'><p>Simulate correlated background effects.</p></a></li>
<li><a href='#expGen2probGen'><p>Rewrite expected genotypes into genotype probabilities.</p></a></li>
<li><a href='#geneticBgEffects'><p>Simulate infinitesimal genetic effects (reflecting sample kinship).</p></a></li>
<li><a href='#geneticFixedEffects'><p>Simulate genetic variant effects.</p></a></li>
<li><a href='#getAlleleFrequencies'><p>Compute allele frequencies from genotype data.</p></a></li>
<li><a href='#getCausalSNPs'><p>Draw random SNPs from genotypes.</p></a></li>
<li><a href='#getKinship'><p>Get genetic kinship.</p></a></li>
<li><a href='#noiseBgEffects'><p>Simulate observational noise effects.</p></a></li>
<li><a href='#noiseFixedEffects'><p>Simulate noise fixed effects.</p></a></li>
<li><a href='#PhenotypeSimulator'><p>PhenotypeSimulator: A package for simulating phenotypes from different</p>
genetic and noise components</a></li>
<li><a href='#probGen2expGen'><p>Compute expected genotypes from genotype probabilities.</p></a></li>
<li><a href='#read_lines'><p>Scan file for specific line numbers</p></a></li>
<li><a href='#readStandardGenotypes'><p>Read genotypes from file.</p></a></li>
<li><a href='#rescaleVariance'><p>Scale phenotype component.</p></a></li>
<li><a href='#runSimulation'><p>Run phenotype simulation.</p></a></li>
<li><a href='#savePheno'><p>Save final phenotype and phenotype components.</p></a></li>
<li><a href='#setModel'><p>Set simulation model.</p></a></li>
<li><a href='#simulateDist'><p>Data simulation for different distributions.</p></a></li>
<li><a href='#simulateGenotypes'><p>Simulate bi-allelic genotypes.</p></a></li>
<li><a href='#simulatePhenotypes'><p>Command line execution for PhenotypeSimulator.</p></a></li>
<li><a href='#standardiseGenotypes'><p>Standardise genotypes.</p></a></li>
<li><a href='#testNumerics'><p>Test lists for different properties of numerics.</p></a></li>
<li><a href='#transformNonlinear'><p>Phenotype transformation.</p></a></li>
<li><a href='#vmessage'><p>Print userinfo.</p></a></li>
<li><a href='#writeStandardOutput'><p>Write simulated data into formats used by standard GWAS software.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Flexible Phenotype Simulation from Different Genetic and Noise
Models</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.4</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/HannahVMeyer/PhenotypeSimulator">https://github.com/HannahVMeyer/PhenotypeSimulator</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/HannahVMeyer/PhenotypeSimulator/issues">https://github.com/HannahVMeyer/PhenotypeSimulator/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Simulation is a critical part of method development and assessment
    in quantitative genetics. 'PhenotypeSimulator' allows for the flexible 
    simulation of phenotypes under different models, including genetic variant 
    and  infinitesimal genetic effects (reflecting population structure) as well 
    as non-genetic covariate effects, observational noise and additional 
    correlation effects. The different phenotype components are combined into a 
    final phenotype while controlling for the proportion of variance explained 
    by each of the components. For each effect component, the number of 
    variables, their distribution and the design of their effect across traits 
    can be customised. For the simulation of the genetic effects, external 
    genotype data from a number of standard software ('plink', 'hapgen2'/
    'impute2', 'genome', 'bimbam', simple text files) can be imported. The final 
    simulated phenotypes and its components can be automatically saved into .rds 
    or .csv files. In addition, they can be saved in formats compatible with 
    commonly used genetic association software ('gemma', 'bimbam', 'plink', 
    'snptest', 'LiMMBo'). </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, optparse, Hmisc, R.utils, mvtnorm, snpStats, zoo,
data.table (&ge; 1.11.0), Rcpp (&ge; 0.12.11), cowplot, ggplot2,
reshape2, dplyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, formatR, rmarkdown</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-07-16 13:09:07 UTC; hannah</td>
</tr>
<tr>
<td>Author:</td>
<td>Hannah Meyer <a href="https://orcid.org/0000-0003-4564-0899"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Konrad Rudolph <a href="https://orcid.org/0000-0002-9866-7051"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hannah Meyer &lt;hannah.v.meyer@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-07-16 13:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='addNonNulls'>Add all non-NULL elements of list.</h2><span id='topic+addNonNulls'></span>

<h3>Description</h3>

<p>Add all non-NULL elements of list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addNonNulls(compList)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addNonNulls_+3A_complist">compList</code></td>
<td>
<p>List of numeric matrices or data.frames of the equal 
dimensions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix or data.frame containing sum of all list elements where 
<code>is.null</code> is FALSE.
</p>

<hr>
<h2 id='commaList2vector'>Comma-separated string to numeric vector.</h2><span id='topic+commaList2vector'></span>

<h3>Description</h3>

<p>Split input of comma-separated string into vector of numeric, logical or 
character values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>commaList2vector(commastring = NULL, type = "numeric")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="commaList2vector_+3A_commastring">commastring</code></td>
<td>
<p>Input [character] vector containing numbers separated by 
commas.</p>
</td></tr>
<tr><td><code id="commaList2vector_+3A_type">type</code></td>
<td>
<p>Name [string] of the type of input variables; one of numeric, 
logical or character</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of values extracted from commastring.
</p>

<hr>
<h2 id='correlatedBgEffects'>Simulate correlated background effects.</h2><span id='topic+correlatedBgEffects'></span>

<h3>Description</h3>

<p>correlatedBgEffects computes a background effect that simulates structured 
correlation between the phenotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correlatedBgEffects(
  N,
  P,
  pcorr = NULL,
  corr_mat = NULL,
  sampleID = "ID_",
  phenoID = "Trait_",
  id_samples = NULL,
  id_phenos = NULL,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="correlatedBgEffects_+3A_n">N</code></td>
<td>
<p>Number [integer] of samples to simulate.</p>
</td></tr>
<tr><td><code id="correlatedBgEffects_+3A_p">P</code></td>
<td>
<p>Number [integer] of phenotypes to simulate.</p>
</td></tr>
<tr><td><code id="correlatedBgEffects_+3A_pcorr">pcorr</code></td>
<td>
<p>Initial strength of correlation [double] between neighbouring 
traits. Decreases by pcorr^(distance); distance from 0 to P-1. See details.</p>
</td></tr>
<tr><td><code id="correlatedBgEffects_+3A_corr_mat">corr_mat</code></td>
<td>
<p>[P x P] correlation matrix [double] as covariance 
component for the multivariate normal distribution. If not provided, pcorr is
used to construct the correlation matrix.</p>
</td></tr>
<tr><td><code id="correlatedBgEffects_+3A_sampleid">sampleID</code></td>
<td>
<p>Prefix [string] for naming samples.</p>
</td></tr>
<tr><td><code id="correlatedBgEffects_+3A_phenoid">phenoID</code></td>
<td>
<p>Prefix [string] for naming traits.</p>
</td></tr>
<tr><td><code id="correlatedBgEffects_+3A_id_samples">id_samples</code></td>
<td>
<p>Vector of [NrSamples] sample IDs [string]; if not provided
constructed by paste(sampleID, 1:N, sep=&quot;&quot;).</p>
</td></tr>
<tr><td><code id="correlatedBgEffects_+3A_id_phenos">id_phenos</code></td>
<td>
<p>Vector of [NrTraits] phenotype IDs [string]; if not provided
constructed by paste(phenoID, 1:P, sep=&quot;&quot;).</p>
</td></tr>
<tr><td><code id="correlatedBgEffects_+3A_verbose">verbose</code></td>
<td>
<p>[boolean] If TRUE, progress info is printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>correlatedBgEffects can be used to simulate phenotypes with a 
defined level of correlation between traits. If the corr_mat is not provided,
a simple correlation structure based on the distance of the traits will be 
constructed. Traits of distance d=1 (adjacent columns) will have correlation 
cor=<code class="reqn">pcorr^1</code>, traits with d=2 have cor=<code class="reqn">pcorr^2</code> 
up to traits with d=(P-1) cor=<code class="reqn">pcorr^{(P-1)}</code> and 
0 &lt; pcorr &lt; 1. The correlated background effect correlated is simulated based 
on this correlation structure C: 
<code class="reqn">correlated ~ N_{NP}(0,C)</code>.
</p>


<h3>Value</h3>

<p>Named list with [N x P] matrix of correlated background effects (
correlatedBg) and the correlation matrix (cov_correlated). If corr_mat 
provided corr_mat == cov_correlated.
</p>


<h3>See Also</h3>

<p><code><a href="mvtnorm.html#topic+rmvnorm">rmvnorm</a></code> which is used to simulate the 
multivariate normal distribution
</p>


<h3>Examples</h3>

<pre><code class='language-R'>correlatedBg &lt;- correlatedBgEffects(N=100, P=20, pcorr=0.4)
</code></pre>

<hr>
<h2 id='expGen2probGen'>Rewrite expected genotypes into genotype probabilities.</h2><span id='topic+expGen2probGen'></span>

<h3>Description</h3>

<p>Convert genotype frequencies to genotypes encoded as triplets of probablities 
(p(AA), p(Aa), p(aa)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>expGen2probGen(geno)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="expGen2probGen_+3A_geno">geno</code></td>
<td>
<p>Vector [numeric] with genotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of length [length(geno)*3] with the genotype encoded 
as probabbilities (p(AA), p(Aa), p(aa)).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>nrSamples &lt;- 10
# Simulate binomial SNP with 0.2 allele frequency
geno &lt;- rbinom(nrSamples, 2, p=0.2)
geno_prob&lt;- expGen2probGen(geno)
</code></pre>

<hr>
<h2 id='geneticBgEffects'>Simulate infinitesimal genetic effects (reflecting sample kinship).</h2><span id='topic+geneticBgEffects'></span>

<h3>Description</h3>

<p>geneticBgEffects simulates an infinitesimal genetic effects with a proportion 
of the effect shared across samples and a proportion independent across 
samples; they are based on the kinship estimates of the (simulated) samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneticBgEffects(
  P,
  N,
  kinship,
  phenoID = "Trait_",
  id_samples = colnames(kinship),
  shared = TRUE,
  independent = TRUE,
  id_phenos = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geneticBgEffects_+3A_p">P</code></td>
<td>
<p>Number [integer] of phenotypes to simulate .</p>
</td></tr>
<tr><td><code id="geneticBgEffects_+3A_n">N</code></td>
<td>
<p>Number [integer] of samples to simulate; has to be provided as a 
dimnesionality check for kinship and downstream analyses; nrow(kinship) has 
to be equal to N.</p>
</td></tr>
<tr><td><code id="geneticBgEffects_+3A_kinship">kinship</code></td>
<td>
<p>[N x N] Matrix of kinship estimates [double].</p>
</td></tr>
<tr><td><code id="geneticBgEffects_+3A_phenoid">phenoID</code></td>
<td>
<p>Prefix [string] for naming traits.</p>
</td></tr>
<tr><td><code id="geneticBgEffects_+3A_id_samples">id_samples</code></td>
<td>
<p>Vector of [NrSamples] sample IDs [string]; if not provided
colnames(kinship) are used.</p>
</td></tr>
<tr><td><code id="geneticBgEffects_+3A_shared">shared</code></td>
<td>
<p>[bool] shared effect simulated if set to TRUE; at least one of 
shared or independent has to be set to TRUE.</p>
</td></tr>
<tr><td><code id="geneticBgEffects_+3A_independent">independent</code></td>
<td>
<p>[bool] independent effect simulated if set to TRUE.</p>
</td></tr>
<tr><td><code id="geneticBgEffects_+3A_id_phenos">id_phenos</code></td>
<td>
<p>Vector of [NrTraits] phenotype IDs [string]; if not provided
constructed by paste(phenoID, 1:P, sep=&quot;&quot;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the simulation of the infinitesimal genetic effects, three 
matrix components are used: i) the kinship matrix K [N x N] which is treated 
as the sample design matrix, ii) matrix B [N x P] with vec(B) drawn from a 
normal distribution and iii) the trait design matrix A [P x P]. For the
independent effect, A is a diagonal matrix with normally distributed values.  
A for the shared effect is a matrix of rowrank one, with normally distributed 
entries in row 1 and zeros elsewhere. To construct the final effects, the 
three matrices are multiplied as: E = cholesky(K)BA^T.
</p>


<h3>Value</h3>

<p>Named list of shared infinitesimal genetic effects (shared: [N x P] 
matrix) and independent infinitesimal genetic effects (independent: [N x P] 
matrix), the covariance term of the shared effect (cov_shared: [P x P] 
matrix), the covariance term of the independent effect (cov_independent: 
[P x P] matrix), the eigenvectors (eigenvec_kinship: [N x N]) and eigenvalues
(eigenval_kinship: [N]) of the kinship matrix.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotypes &lt;- simulateGenotypes(N=100, NrSNP=400, verbose=FALSE)
kinship &lt;- getKinship(N=100, X=genotypes$genotypes, standardise=TRUE, 
verbose=FALSE)
geneticBg &lt;- geneticBgEffects(N=100, P=10, kinship=kinship)
</code></pre>

<hr>
<h2 id='geneticFixedEffects'>Simulate genetic variant effects.</h2><span id='topic+geneticFixedEffects'></span>

<h3>Description</h3>

<p>geneticFixedEffects takes genetic variants which should be added as genetic 
variant effects to the phenotype. These variants can have the same effects 
across all traits (shared) or can be independent across traits (independent); 
in addition, only a certain proportion of traits can be affected by the 
genetic variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneticFixedEffects(
  X_causal,
  P,
  N,
  phenoID = "Trait_",
  id_samples = rownames(X_causal),
  id_phenos = NULL,
  pTraitsAffected = 1,
  pIndependentGenetic = 0.4,
  pTraitIndependentGenetic = 0.2,
  keepSameIndependent = FALSE,
  distBeta = "norm",
  mBeta = 0,
  sdBeta = 1,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geneticFixedEffects_+3A_x_causal">X_causal</code></td>
<td>
<p>[N x NrCausalSNPs] Matrix of [NrCausalSNPs] SNPs from [N] 
samples.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_p">P</code></td>
<td>
<p>Number [integer] of phenotypes to simulate.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_n">N</code></td>
<td>
<p>Number [integer] of samples to simulate; has to be provided as a 
dimnesionality check for X_causal and downstream analyses; nrow(X_causal) has 
to be equal to N.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_phenoid">phenoID</code></td>
<td>
<p>Prefix [string] for naming traits.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_id_samples">id_samples</code></td>
<td>
<p>Vector of [NrSamples] sample IDs [string]; if not provided
colnames(X_causal) used.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_id_phenos">id_phenos</code></td>
<td>
<p>Vector of [NrTraits] phenotype IDs [string]; if not provided
constructed by paste(phenoID, 1:P, sep=&quot;&quot;).</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_ptraitsaffected">pTraitsAffected</code></td>
<td>
<p>Proportion [double] of traits affected by the genetic
effect. For non-integer results of pTraitsAffected*P, the ceiling of the 
result is used. Allows to simulate for instance different levels of 
pleiotropy.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_pindependentgenetic">pIndependentGenetic</code></td>
<td>
<p>Proportion [double] of genetic effects (SNPs) to
have an independent fixed effect.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_ptraitindependentgenetic">pTraitIndependentGenetic</code></td>
<td>
<p>Proportion [double] of traits influenced by 
independent fixed genetic effects.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_keepsameindependent">keepSameIndependent</code></td>
<td>
<p>[boolean] If set to TRUE, the independent genetic 
effects always influence the same subset of traits.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_distbeta">distBeta</code></td>
<td>
<p>Vector of name(s) [string] of distribution to use to simulate 
effect sizes of SNPs; one of &quot;unif&quot; or &quot;norm&quot;.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_mbeta">mBeta</code></td>
<td>
<p>Vector of mean/midpoint(s) [double] of normal/uniform 
distribution for effect sizes of SNPs.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_sdbeta">sdBeta</code></td>
<td>
<p>Vector of standard deviation/distance from midpoint [double] 
of normal/uniform distribution for effect sizes of SNPs.</p>
</td></tr>
<tr><td><code id="geneticFixedEffects_+3A_verbose">verbose</code></td>
<td>
<p>[boolean] If TRUE, progress info is printed to standard out</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named list of shared fixed genetic effects (shared: [N x P] matrix), 
independent fixed genetic effects (independent: [N x P] matrix), 
the causal SNPs labeled as shared or independent effect 
(cov: [NrCausalSNPs x N] matrix) and the simulated effect sizes of the causal 
SNPs (cov_effect: [P x NrCausalSNPs] dataframe).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genotypes &lt;- simulateGenotypes(N=100, NrSNP=20, verbose=FALSE)
causalSNPs &lt;- getCausalSNPs(N=100, genotypes=genotypes$genotypes)
geneticFixed &lt;- geneticFixedEffects(N=100, X_causal=causalSNPs, 
P=10)
</code></pre>

<hr>
<h2 id='getAlleleFrequencies'>Compute allele frequencies from genotype data.</h2><span id='topic+getAlleleFrequencies'></span>

<h3>Description</h3>

<p>Compute allele frequencies from genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getAlleleFrequencies(snp)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getAlleleFrequencies_+3A_snp">snp</code></td>
<td>
<p>[N x 1] Vector of length [N] samples with genotypes of a single
bi-allelic genetic variant/SNP encoded as 0,1 and 2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector with ref (0-encoded) and alt (1-encoded) allele frequencies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create snp vector with minor allele frequency 0.3
snp &lt;- rbinom(200, 2, 0.3)
allelefreq &lt;- getAlleleFrequencies(snp)
</code></pre>

<hr>
<h2 id='getCausalSNPs'>Draw random SNPs from genotypes.</h2><span id='topic+getCausalSNPs'></span>

<h3>Description</h3>

<p>Draw random SNPs from genotypes provided or external genotype files.
When drawing from external genotype files, only lines of randomly
chosen SNPs are read, which is recommended for large genotype files. See
details for more information. The latter option currently supports file in
simple delim-formats (with specified delimiter and optional number of fields
to skip) and the bimbam and the oxgen format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCausalSNPs(
  N,
  NrCausalSNPs = 20,
  genotypes = NULL,
  chr = NULL,
  NrSNPsOnChromosome = NULL,
  NrChrCausal = NULL,
  genoFilePrefix = NULL,
  genoFileSuffix = NULL,
  format = "delim",
  delimiter = ",",
  header = FALSE,
  skipFields = NULL,
  probabilities = FALSE,
  sampleID = "ID_",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCausalSNPs_+3A_n">N</code></td>
<td>
<p>Number [integer] of samples to simulate.</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_nrcausalsnps">NrCausalSNPs</code></td>
<td>
<p>Number [integer] of SNPs to chose at random.</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_genotypes">genotypes</code></td>
<td>
<p>[NrSamples x totalNrSNPs] Matrix of genotypes [integer]/
[double].</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_chr">chr</code></td>
<td>
<p>Vector of chromosome(s) [integer] to chose NrCausalSNPs from; only
used when external genotype data is provided i.e. !is.null(genoFilePrefix).</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_nrsnpsonchromosome">NrSNPsOnChromosome</code></td>
<td>
<p>Vector of number(s) of SNPs [integer] per entry in
chr (see above); has to be the same length as chr. If not provided, number of
SNPS in file will be determined from line count (which can be slow for large
files); (optional) header lines will be ignored, so accurate number of SNPs
not lines in file should be specified.</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_nrchrcausal">NrChrCausal</code></td>
<td>
<p>Number [integer] of causal chromosomes to sample
NrCausalSNPs from (as opposed to the actual chromosomes to chose from via chr
); only used when external genotype data is provided i.e.
!is.null(genoFilePrefix).</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_genofileprefix">genoFilePrefix</code></td>
<td>
<p>full path/to/chromosome-wise-genotype-file-ending-
before-&quot;chrChromosomeNumber&quot; (no '~' expansion!) [string].</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_genofilesuffix">genoFileSuffix</code></td>
<td>
<p>[string] Following chromosome number including
.fileformat (e.g. &quot;.csv&quot;); File described by genoFilePrefix-genoFileSuffix
has to be a text format i.e. comma/tab/space separated.</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_format">format</code></td>
<td>
<p>Name [string] of genotype file format. Options are:
&quot;oxgen&quot;, &quot;bimbam&quot; or &quot;delim&quot;. See <a href="#topic+readStandardGenotypes">readStandardGenotypes</a> for details.</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_delimiter">delimiter</code></td>
<td>
<p>Field separator [string] of genotypefile or
genoFilePrefix-genoFileSuffix file if format == 'delim'.</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_header">header</code></td>
<td>
<p>[logical] Can be set to indicate if
genoFilePrefix-genoFileSuffix file has a header for format == 'delim'. 
See details.</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_skipfields">skipFields</code></td>
<td>
<p>Number [integer] of fields (columns) to skip in
genoFilePrefix-genoFileSuffix file if format == 'delim'. See details.</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_probabilities">probabilities</code></td>
<td>
<p>[boolean]. If set to TRUE, the genotypes in the files
described by genoFilePrefix-genoFileSuffix are provided as triplets of
probabilities (p(AA), p(Aa), p(aa)) and are converted into their expected
genotype frequencies by 0*p(AA) + p(Aa) + 2p(aa) via <a href="#topic+probGen2expGen">probGen2expGen</a>.</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_sampleid">sampleID</code></td>
<td>
<p>Prefix [string] for naming samples (will be followed by
sample number from 1 to N when constructing id_samples)</p>
</td></tr>
<tr><td><code id="getCausalSNPs_+3A_verbose">verbose</code></td>
<td>
<p>[boolean] If TRUE, progress info is printed to standard out</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In order to chose SNPs from external genotype files without reading
them into memory, genotypes for each chromosome need to be accesible as
[SNPs x samples] in a separate file, containing &quot;chrChromosomenumber&quot; (e.g
chr22) in the file name (e.g. /path/to/dir/related_nopopstructure_chr22.csv).
All genotype files need to be saved in the same directory. genoFilePrefix
(/path/to/dir/related_nopopstructure_) and genoFileSuffix (.csv) specify the
strings leading and following the &quot;chrChromosomenumber&quot;. If format== delim,
the first column in each file needs to be the SNP_ID, the first row can
either contain sample IDs or the first row of genotypes (specified with
header). Subsequent columns containing additional SNP information can be
skipped by setting skipFields. If format==oxgen or bimbam, files need to be
in the oxgen or bimbam format (see <a href="#topic+readStandardGenotypes">readStandardGenotypes</a> for details)
and no additional information about delim, header or skipFields will be
considered.
getCausalSNPs generates a vector of chromosomes from which to sample the
SNPs. For each of the chromosomes, it counts the number of SNPs in the
chromosome file and creates vectors of random numbers ranging
from 1:NrSNPSinFile. Only the lines corresponding to these numbers are then
read into R. The example data provided for chromosome 22 contains genotypes
(50 samples) of the first 500 SNPs on chromosome 22 with a minor allele
frequency of greater than 2
Genomes project.
</p>


<h3>Value</h3>

<p>[N x NrCausalSNPs] Matrix of randomly drawn genotypes [integer]/
[double]
</p>


<h3>See Also</h3>

<p><code><a href="#topic+standardiseGenotypes">standardiseGenotypes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># get causal SNPs from genotypes simulated within PhenotypeSimulator
geno &lt;- simulateGenotypes(N=10, NrSNP=10)
causalSNPsFromSimulatedGenoStandardised &lt;- getCausalSNPs(N=10,
NrCausalSNPs=10, genotypes=geno$genotypes)

# Get causal SNPs by sampling lines from large SNP files
genotypeFile &lt;- system.file("extdata/genotypes/",
"genotypes_chr22.csv",
package = "PhenotypeSimulator")
genoFilePrefix &lt;- gsub("chr.*", "", genotypeFile)
genoFileSuffix &lt;- ".csv"
causalSNPsFromLines &lt;- getCausalSNPs(N=50, NrCausalSNPs=10, chr=22,
genoFilePrefix=genoFilePrefix,
genoFileSuffix=genoFileSuffix)
</code></pre>

<hr>
<h2 id='getKinship'>Get genetic kinship.</h2><span id='topic+getKinship'></span>

<h3>Description</h3>

<p>Estimate kinship from standardised genotypes or read pre-computed kinship
file. Standardised genotypes can be obtained via
<code><a href="#topic+standardiseGenotypes">standardiseGenotypes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKinship(
  N,
  sampleID = "ID_",
  X = NULL,
  kinshipfile = NULL,
  id_samples = NULL,
  standardise = FALSE,
  sep = ",",
  header = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKinship_+3A_n">N</code></td>
<td>
<p>Number [integer] of samples to simulate.</p>
</td></tr>
<tr><td><code id="getKinship_+3A_sampleid">sampleID</code></td>
<td>
<p>Prefix [string] for naming samples (will be followed by
sample number from 1 to N when constructing id_samples).</p>
</td></tr>
<tr><td><code id="getKinship_+3A_x">X</code></td>
<td>
<p>[NrSamples x totalNrSNPs] Matrix of (standardised) genotypes.</p>
</td></tr>
<tr><td><code id="getKinship_+3A_kinshipfile">kinshipfile</code></td>
<td>
<p>path/to/kinshipfile [string] to be read; either X or
kinshipfile must be provided.</p>
</td></tr>
<tr><td><code id="getKinship_+3A_id_samples">id_samples</code></td>
<td>
<p>Vector of [NrSamples] sample IDs [string]; if not provided
constructed by paste(sampleID, 1:N, sep=&quot;&quot;).</p>
</td></tr>
<tr><td><code id="getKinship_+3A_standardise">standardise</code></td>
<td>
<p>[boolean] If TRUE genotypes will be standardised before
kinship estimation.</p>
</td></tr>
<tr><td><code id="getKinship_+3A_sep">sep</code></td>
<td>
<p>Field separator [string] of kinship file.</p>
</td></tr>
<tr><td><code id="getKinship_+3A_header">header</code></td>
<td>
<p>[boolean], If TRUE kinship file has header information.</p>
</td></tr>
<tr><td><code id="getKinship_+3A_verbose">verbose</code></td>
<td>
<p>[boolean]; If TRUE, progress info is printed to standard out</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The kinship is estimated as <code class="reqn">K = XX_T</code>, with X the standardised
genotypes of the samples. When estimating the kinship from the provided
genotypes, the kinship is normalised by the mean of its diagonal
elements and 1e-4 added to the diagonal for numerical stability.
</p>


<h3>Value</h3>

<p>[NrSamples x NrSamples] Matrix of kinship estimate.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geno &lt;- simulateGenotypes(N=10, NrSNP=50)
K_fromGenotypesNormalised &lt;- getKinship(N=10, X=geno$genotypes,
standardise=TRUE)

kinshipfile &lt;- system.file("extdata/kinship",
"kinship.csv",
package = "PhenotypeSimulator")
K_fromFile &lt;- getKinship(N=50, kinshipfile=kinshipfile)
</code></pre>

<hr>
<h2 id='noiseBgEffects'>Simulate observational noise effects.</h2><span id='topic+noiseBgEffects'></span>

<h3>Description</h3>

<p>noiseBgEffects simulates observational noise with a proportion of the effect 
shared across samples and a proportion independent across samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>noiseBgEffects(
  N,
  P,
  mean = 0,
  sd = 1,
  sampleID = "ID_",
  phenoID = "Trait_",
  shared = TRUE,
  independent = TRUE,
  id_samples = NULL,
  id_phenos = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="noiseBgEffects_+3A_n">N</code></td>
<td>
<p>Number [integer] of samples to simulate.</p>
</td></tr>
<tr><td><code id="noiseBgEffects_+3A_p">P</code></td>
<td>
<p>Number [integer] of phenotypes to simulate.</p>
</td></tr>
<tr><td><code id="noiseBgEffects_+3A_mean">mean</code></td>
<td>
<p>Mean [double] of the normal distribution.</p>
</td></tr>
<tr><td><code id="noiseBgEffects_+3A_sd">sd</code></td>
<td>
<p>Standard deviation [double] of the normal distribution.</p>
</td></tr>
<tr><td><code id="noiseBgEffects_+3A_sampleid">sampleID</code></td>
<td>
<p>Prefix [string] for naming samples.</p>
</td></tr>
<tr><td><code id="noiseBgEffects_+3A_phenoid">phenoID</code></td>
<td>
<p>Prefix [string] for naming traits.</p>
</td></tr>
<tr><td><code id="noiseBgEffects_+3A_shared">shared</code></td>
<td>
<p>[bool] shared effect simulated if set to TRUE; at least one of 
shared or independent has to be set to TRUE.</p>
</td></tr>
<tr><td><code id="noiseBgEffects_+3A_independent">independent</code></td>
<td>
<p>[bool] independent effect simulated if set to TRUE.</p>
</td></tr>
<tr><td><code id="noiseBgEffects_+3A_id_samples">id_samples</code></td>
<td>
<p>Vector of [NrSamples] sample IDs [string]; if not provided
constructed by paste(sampleID, 1:N, sep=&quot;&quot;).</p>
</td></tr>
<tr><td><code id="noiseBgEffects_+3A_id_phenos">id_phenos</code></td>
<td>
<p>Vector of [NrTraits] phenotype IDs [string]; if not provided
constructed by paste(phenoID, 1:P, sep=&quot;&quot;).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For the simulation of the observational noise effects, two 
components are used: i) matrix B [N x P] with vec(B) drawn from a normal 
distribution with mean=mean and sd=sd and ii) the trait design matrix A 
[P x P]. For the independent effect, A is a diagonal matrix with normally 
distributed values.  A for the shared effect is a matrix of rowrank one, with 
normally distributed entries in row 1 and zeros elsewhere.  To construct the 
final effects, the two matrices are multiplied as: E = BA^T.
</p>


<h3>Value</h3>

<p>Named list of shared noise effects (shared: [N x P] matrix) and 
independent noise effects (independent: [N x P] matrix), the covariance term 
of the shared effect (cov_shared: [P x P] matrix) and the covariance term of 
the independent effect (cov_independent: [P x P] matrix).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>noiseBG &lt;- noiseBgEffects(N=100, P=20, mean=2)
</code></pre>

<hr>
<h2 id='noiseFixedEffects'>Simulate noise fixed effects.</h2><span id='topic+noiseFixedEffects'></span>

<h3>Description</h3>

<p>noiseFixedEffects simulates a number of non-genetic covariate effects 
(confounders). Confounders can have effects across all traits (shared) 
or to a number of traits only (independent); in addition, only a certain 
proportion of traits can be affected by the confounders.
Confounders can be simulated as categorical variables or following a binomial 
, uniform or normal distribution. Effect sizes for the noise effects can be 
simulated from a uniform or normal distribution. Multiple confounder sets 
drawn from different distributions/different parameters of the same 
distribution can be simulated by specifying NrFixedEffects and supplying the 
respective distribution parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>noiseFixedEffects(
  N,
  P,
  NrConfounders = 10,
  sampleID = "ID_",
  phenoID = "Trait_",
  id_samples = NULL,
  id_phenos = NULL,
  pTraitsAffected = 1,
  NrFixedEffects = 1,
  pIndependentConfounders = 0.4,
  pTraitIndependentConfounders = 0.2,
  keepSameIndependent = FALSE,
  distConfounders = "norm",
  mConfounders = 0,
  sdConfounders = 1,
  catConfounders = NULL,
  probConfounders = NULL,
  distBeta = "norm",
  mBeta = 0,
  sdBeta = 1,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="noiseFixedEffects_+3A_n">N</code></td>
<td>
<p>Number [integer] of samples to simulate.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_p">P</code></td>
<td>
<p>Number [integer] of phenotypes to simulate.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_nrconfounders">NrConfounders</code></td>
<td>
<p>Vector of number(s) [integer] of confounders from a 
specified distribution to simulate.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_sampleid">sampleID</code></td>
<td>
<p>Prefix [string] for naming samples.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_phenoid">phenoID</code></td>
<td>
<p>Prefix [string] for naming traits.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_id_samples">id_samples</code></td>
<td>
<p>Vector of [NrSamples] sample IDs [string]; if not provided
constructed by paste(sampleID, 1:N, sep=&quot;&quot;).</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_id_phenos">id_phenos</code></td>
<td>
<p>Vector of [NrTraits] phenotype IDs [string]; if not provided
constructed by paste(phenoID, 1:P, sep=&quot;&quot;).</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_ptraitsaffected">pTraitsAffected</code></td>
<td>
<p>Vector of proportion(s) [double] of traits affected by 
the confounders. For non-integer results of pTraitsAffected*P, the ceiling of 
the result is used.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_nrfixedeffects">NrFixedEffects</code></td>
<td>
<p>Number [integer] of different confounder effects to 
simulate; allows to simulate fixed effects from different distributions or 
with different parameters; if only one type of confounder distribution is 
wanted, set NrFixedEffects=1 and choose the number of confounders with eg 
NrConfounders=10.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_pindependentconfounders">pIndependentConfounders</code></td>
<td>
<p>Vector of proportion(s) [double] of 
confounders to have a trait-independent effect.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_ptraitindependentconfounders">pTraitIndependentConfounders</code></td>
<td>
<p>Vector of proportion(s) [double] of 
traits influenced by independent confounder effects.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_keepsameindependent">keepSameIndependent</code></td>
<td>
<p>[boolean] If set to TRUE, the independent genetic 
effects always influence the same subset of traits.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_distconfounders">distConfounders</code></td>
<td>
<p>Vector of name(s) [string] of distribution to use to 
simulate confounders; one of &quot;unif&quot;, &quot;norm&quot;, &quot;bin&quot;, &quot;cat_norm&quot;, &quot;cat_unif&quot;.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_mconfounders">mConfounders</code></td>
<td>
<p>Vector of mean/midpoint(s) [double] of normal/uniform 
distribution for confounders.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_sdconfounders">sdConfounders</code></td>
<td>
<p>Vector of standard deviation(s)/distance from 
midpoint(s) [double] of normal/uniform distribution for confounders.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_catconfounders">catConfounders</code></td>
<td>
<p>Vector of number(s) of confounder categories [integer]; 
required if distConfounders &quot;cat_norm&quot; or &quot;cat_unif&quot;.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_probconfounders">probConfounders</code></td>
<td>
<p>Vector of probability(s) [double] of binomial 
confounders (0/1); required if distConfounders &quot;bin&quot;.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_distbeta">distBeta</code></td>
<td>
<p>Vector of name(s) [string] of distribution to use to simulate 
effect sizes of confounders; one of &quot;unif&quot; or &quot;norm&quot;.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_mbeta">mBeta</code></td>
<td>
<p>Vector of mean/midpoint [double] of normal/uniform distribution 
for effect sizes of confounders.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_sdbeta">sdBeta</code></td>
<td>
<p>Vector of standard deviation/distance from midpoint [double] 
of normal/uniform distribution for effect sizes of confounders.</p>
</td></tr>
<tr><td><code id="noiseFixedEffects_+3A_verbose">verbose</code></td>
<td>
<p>[boolean] If TRUE, progress info is printed to standard out</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named list of shared confounder effects (shared: [N x P] matrix), 
independent confoudner effects (independent: [N x P] matrix), 
the confounders labeled as shared or independent effect 
(cov: [NrConfounders x N] matrix) and the simulated effect sizes of the 
confounders (cov_effect: [P x NrConfounders] dataframe).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+simulateDist">simulateDist</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># fixed noise effect with default setting
noiseFE &lt;- noiseFixedEffects(P=5, N=20)

# 1 categorical fixed noise effect with uniform distribution of the 
# categories
noiseFE_catUnif &lt;- noiseFixedEffects(P=10, N=20, NrConfounders=1, 
distConfounders="cat_unif", catConfounders=3)

# 10 fixed noise effect with uniform distribution between 1 and 5 (3 +/- 2) 
# categories
noiseFE_uniformConfounders_normBetas &lt;- noiseFixedEffects(P=10, N=20, 
NrConfounders=10, distConfounders="unif", mConfounders=3, sdConfounders=2, 
distBeta="norm",  sdBeta=2)

 # 4 fixed noise effect with binomial distribution with p=0.2 
noiseFE_binomialConfounders_uniformBetas &lt;- noiseFixedEffects(P=10, N=20, 
NrConfounders=4, distConfounders="bin", probConfounders=0.2, distBeta="norm", 
sdBeta=2)

 # 2 fixed noise effect with 1 binomial confounders and 1 normally 
 # distributed confounder; the latter only affects 2 traits 
 noiseFE_binomialandNormalConfounders &lt;- noiseFixedEffects(P=10, N=20, 
 NrFixedEffects=2, pTraitsAffected =c (1,0.2), NrConfounders=c(2,2), 
 distConfounders=c("bin", "norm"),  probConfounders=0.2)
</code></pre>

<hr>
<h2 id='PhenotypeSimulator'>PhenotypeSimulator: A package for simulating phenotypes from different
genetic and noise components</h2><span id='topic+PhenotypeSimulator'></span>

<h3>Description</h3>

<p>Type, Name and Dependencies
</p>

<hr>
<h2 id='probGen2expGen'>Compute expected genotypes from genotype probabilities.</h2><span id='topic+probGen2expGen'></span>

<h3>Description</h3>

<p>Convert genotypes encoded as triplets of probablities (p(AA), p(Aa), p(aa))
into their expected genotype frequencies by 0*p(AA) + p(Aa) + 2p(aa).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>probGen2expGen(probGeno)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="probGen2expGen_+3A_probgeno">probGeno</code></td>
<td>
<p>Vector [numeric] with genotype probabilites; has to be a
multiple of 3.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of length [length(probGeno)/3] with the expected 
genotype value per individual.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>nrSamples &lt;- 10
# Construct genotype probability vector (usually from external input)
# First, assign zero probabilty of AA, Aa and aa for all samples
genotype_prob &lt;- rep(0, 3*nrSamples)
# Second, for each sample draw one of 0,1,2 (corresponding to AA, Aa and aa)
genotype_prob[seq(1, nrSamples*3, 3) + sample(0:2, 10, replace=TRUE)] &lt;- 1
genotype_exp &lt;- probGen2expGen(genotype_prob)
</code></pre>

<hr>
<h2 id='read_lines'>Scan file for specific line numbers</h2><span id='topic+read_lines'></span>

<h3>Description</h3>

<p>Scan file for specific line numbers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_lines(filename, lines, sep = "\n")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_lines_+3A_filename">filename</code></td>
<td>
<p>/path/to/chromosomefile [string]</p>
</td></tr>
<tr><td><code id="read_lines_+3A_lines">lines</code></td>
<td>
<p>vector of line numbers [integer] to be read</p>
</td></tr>
<tr><td><code id="read_lines_+3A_sep">sep</code></td>
<td>
<p>[string] end-of-line delimiter</p>
</td></tr>
</table>

<hr>
<h2 id='readStandardGenotypes'>Read genotypes from file.</h2><span id='topic+readStandardGenotypes'></span>

<h3>Description</h3>

<p>readStandardGenotypes can read genotypes from a number of input
formats for standard GWAS (binary plink, snptest, bimbam, gemma) or
simulation software (binary plink, hapgen2, genome). Alternatively, simple
text files (with specified delimiter) can be read. For more information on
the different file formats see <em>External genotype software and formats</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readStandardGenotypes(
  N,
  filename,
  format = NULL,
  verbose = TRUE,
  sampleID = "ID_",
  snpID = "SNP_",
  delimiter = ","
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readStandardGenotypes_+3A_n">N</code></td>
<td>
<p>Number [integer] of samples to simulate.</p>
</td></tr>
<tr><td><code id="readStandardGenotypes_+3A_filename">filename</code></td>
<td>
<p>path/to/genotypefile [string] in plink, oxgen
(impute2/snptest/hapgen2), genome, bimbam or [delimiter]-delimited format (
for format information see <em>External genotype software and formats</em>).</p>
</td></tr>
<tr><td><code id="readStandardGenotypes_+3A_format">format</code></td>
<td>
<p>Name [string] of genotype file format. Options are: &quot;plink&quot;,
&quot;oxgen&quot;, &quot;genome&quot;, &quot;bimbam&quot; or &quot;delim&quot;.</p>
</td></tr>
<tr><td><code id="readStandardGenotypes_+3A_verbose">verbose</code></td>
<td>
<p>[boolean] If TRUE, progress info is printed to standard out.</p>
</td></tr>
<tr><td><code id="readStandardGenotypes_+3A_sampleid">sampleID</code></td>
<td>
<p>Prefix [string] for naming samples (will be followed by
sample number from 1 to N when constructing id_samples).</p>
</td></tr>
<tr><td><code id="readStandardGenotypes_+3A_snpid">snpID</code></td>
<td>
<p>Prefix [string] for naming SNPs (will be followed by SNP number
from 1 to NrSNP when constructing id_snps).</p>
</td></tr>
<tr><td><code id="readStandardGenotypes_+3A_delimiter">delimiter</code></td>
<td>
<p>Field separator [string] of genotype file when
format == &quot;delim&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The file formats and software formates supported are described
below. For large external genotypes, consider the option to only read
randomly selected, causal SNPs into memory via <a href="#topic+getCausalSNPs">getCausalSNPs</a>.
</p>


<h3>Value</h3>

<p>Named list of [NrSamples X NrSNPs] genotypes (genotypes), their
[NrSNPs] SNP IDs (id_snps), their [NrSamples] sample IDs (id_samples) and
format-specific additional files (such as format-specific genotypes encoding
or sample information; might be used for output writing).
</p>


<h3>External genotype software and formats</h3>



<h4>Formats:</h4>


<ul>
<li><p> PLINK: consists of three files: .bed, .bim and .fam. When specifying
the filepath, only the core of the name without the ending should be
specified (i.e. for geno.bed, geno.bim and geno.fam, geno should be
specified). When reading data from plink files, the absolute path to the
plink-format file has to be provided, tilde expansion not provided.
From <a href="https://www.cog-genomics.org/plink/1.9/formats">https://www.cog-genomics.org/plink/1.9/formats</a>:  The .bed
files contain the  primary representation of genotype calls at biallelic
variants in a binary format. The .bim is a text file with no header
line, and one line  per variant with the following six fields: i) Chromosome
code (either an integer, or 'X'/'Y'/'XY'/'MT'; '0' indicates unknown) or
name, ii) Variant identifier, iii) Position in morgans or centimorgans (safe
to use dummy value of '0'), iv) Base-pair coordinate (normally 1-based, but
0 ok; limited to 231-2), v) Allele 1 (corresponding to clear bits in .bed;
usually minor), vi) Allele 2 (corresponding to set bits in .bed; usually
major). The .fam file is a text file with no header line, and one line per
sample with the following six fields: i) Family ID ('FID'), ii), Within-
family ID ('IID'; cannot be '0'), iii) Within-family ID of father ('0' if
father isn't in dataset, iv) within-family ID of mother ('0' if mother isn't
in dataset), v) sex code ('1' = male, '2' = female, '0' = unknown), vi)
Phenotype value ('1' = control, '2' = case, '-9'/'0'/non-numeric = missing
data if case/control)
</p>
</li>
<li><p> oxgen: consists of two files: the space-separated genotype file ending
in .gen and the space-separated sample file ending in .sample. When
specifying the filepath, only the core of the name without the ending should
be specified (i.e. for geno.gen and geno.sample, geno should be specified).
From
<a href="https://www.well.ox.ac.uk/~gav/snptest/#input_file_formats">https://www.well.ox.ac.uk/~gav/snptest/#input_file_formats</a>:
The genotype file stores data on a one-line-per-SNP format. The first five
entries of each line should be the SNP ID, RS ID of the SNP, base-pair
position of the SNP, the allele coded A and the allele coded B. The SNP ID
can be used to denote the chromosome number of each SNP. The next three
numbers on the line should be the probabilities of the three genotypes AA,
AB and BB at the SNP for the first individual in the cohort. The next three
numbers should be the genotype probabilities for the second individual in the
cohort. The next three numbers are for the third individual and so on. The
order of individuals in the genotype file should match the order of the
individuals in the sample file. The sample file has three parts (a) a header
line detailing the names of the columns in the file, (b) a line detailing the
types of variables stored in each column, and (c) a line for each individual
detailing the information for that individual. For more information on the
sample file visit the above url or see <a href="#topic+writeStandardOutput">writeStandardOutput</a>.
</p>
</li>
<li><p> genome: The entire output of genome can be saved via 'genome -options &gt;
outputfile'. The /path/to/outputfile should be provided and this function
extracts the relevant genotype information from this output file.
<a href="http://csg.sph.umich.edu/liang/genome/">http://csg.sph.umich.edu/liang/genome/</a>
</p>
</li>
<li><p> bimbam: Mean genotype file format of bimbam which is a single, comma-
separated file, without information on individuals. From the documentation
for bimbam at <a href="http://www.haplotype.org/software.html">http://www.haplotype.org/software.html</a>: the first column
of the mean genotype files is the SNP ID, the second and third columns are
allele types with minor allele first. The remaining columns are the mean
genotypes of different individuals – numbers between 0 and 2 that represents
the (posterior) mean genotype, or dosage of the minor allele. </p>
</li>
<li><p> delim: a
[delimter]-delimited file of [(NrSNPs+1) x (NrSamples+1)] genotypes with the
snpIDs in the first column and the sampleIDs in the first row and genotypes
encoded as numbers between 0 and 2 representing the (posterior) mean
genotype, or dosage of the minor allele. Can be user-genotypes or genotypes
simulated with foward-time algorithms such as simupop
(<a href="http://simupop.sourceforge.net/Main/HomePage">http://simupop.sourceforge.net/Main/HomePage</a>)
that allow for user-specified output formats.
</p>
</li></ul>



<h4>Genotype simulation characteristics:</h4>


<ul>
<li><p> PLINK: simple, bi-allelelic genotypes without LD structure, details can
be found at <a href="https://www.cog-genomics.org/plink/1.9/input#simulate">https://www.cog-genomics.org/plink/1.9/input#simulate</a>.
</p>
</li>
<li><p> Hapgen2: resampling-based genotype simulation, details can be found at
<a href="http://mathgen.stats.ox.ac.uk/genetics_software/hapgen/hapgen2.html">http://mathgen.stats.ox.ac.uk/genetics_software/hapgen/hapgen2.html</a>.
</p>
</li>
<li><p> Genome: coalescent-based genotype simulation, details can be found at
<a href="http://csg.sph.umich.edu/liang/genome/GENOME-manual.pdf">http://csg.sph.umich.edu/liang/genome/GENOME-manual.pdf</a>.
</p>
</li></ul>

<p>Examples on how to call these genotype simulation tools can be found in the
vignette <em>sample-scripts-external-genotype-simulation</em>.</p>



<h3>Examples</h3>

<pre><code class='language-R'># Genome format
filename_genome  &lt;- system.file("extdata/genotypes/genome/",
"genotypes_genome.txt",
package = "PhenotypeSimulator")
data_genome &lt;- readStandardGenotypes(N=100, filename_genome, format ="genome")

filename_hapgen  &lt;- system.file("extdata/genotypes/hapgen/",
"genotypes_hapgen.controls.gen",
package = "PhenotypeSimulator")
filename_hapgen &lt;- gsub("\\.gen", "", filename_hapgen)
data_hapgen &lt;- readStandardGenotypes(N=100, filename_hapgen, format='oxgen')

filename_plink  &lt;- system.file("extdata/genotypes/plink/",
"genotypes_plink.bed",
package = "PhenotypeSimulator")
filename_plink &lt;- gsub("\\.bed", "", filename_plink)
data_plink &lt;- readStandardGenotypes(N=100, filename=filename_plink,
format="plink")

filename_delim  &lt;- system.file("extdata/genotypes/",
"genotypes_chr22.csv",
package = "PhenotypeSimulator")
data_delim &lt;- readStandardGenotypes(N=50, filename=filename_delim,
format="delim")
</code></pre>

<hr>
<h2 id='rescaleVariance'>Scale phenotype component.</h2><span id='topic+rescaleVariance'></span>

<h3>Description</h3>

<p>The function scales the specified component such that the average column 
variance is equal to the user-specified proportion of variance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rescaleVariance(component, propvar)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rescaleVariance_+3A_component">component</code></td>
<td>
<p>[N x P] Phenotype matrix [double] where [N] are the number 
of samples and P the number of phenotypes</p>
</td></tr>
<tr><td><code id="rescaleVariance_+3A_propvar">propvar</code></td>
<td>
<p>Number [double] specifying the proportion of variance that 
should be explained by this phenotype component</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If propvar != 0, a named list with the [N x P] matrix of the scaled 
component (component) and its scale factor [double] (scale_factor) else 
returns NULL
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- matrix(rnorm(100), nc=10)
x_scaled &lt;- rescaleVariance(x, propvar=0.4)
</code></pre>

<hr>
<h2 id='runSimulation'>Run phenotype simulation.</h2><span id='topic+runSimulation'></span>

<h3>Description</h3>

<p>runSimulation wraps around setModel, the phenotype component functions 
(genFixedEffects, genBgEffects, noiseBgEffects, noiseFixedEffects and 
correlatedBgEffects), rescales each component and combines them into the 
final phenotype. For details to all parameters, see the respective functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runSimulation(
  N,
  P,
  genVar = NULL,
  h2s = NULL,
  theta = 0.8,
  h2bg = NULL,
  eta = 0.8,
  noiseVar = NULL,
  rho = NULL,
  delta = NULL,
  gamma = 0.8,
  phi = NULL,
  alpha = 0.8,
  tNrSNP = 5000,
  cNrSNP = 20,
  SNPfrequencies = c(0.1, 0.2, 0.4),
  genotypefile = NULL,
  format = "delim",
  genoFilePrefix = NULL,
  genoFileSuffix = NULL,
  genoDelimiter = ",",
  skipFields = NULL,
  header = FALSE,
  probabilities = FALSE,
  chr = NULL,
  NrSNPsOnChromosome = NULL,
  NrChrCausal = NULL,
  kinshipfile = NULL,
  kinshipHeader = FALSE,
  kinshipDelimiter = ",",
  standardise = TRUE,
  distBetaGenetic = "norm",
  mBetaGenetic = 0,
  sdBetaGenetic = 1,
  pTraitsAffectedGenetics = 1,
  pIndependentGenetic = 0.4,
  pTraitIndependentGenetic = 0.2,
  keepSameIndependentSNPs = FALSE,
  NrFixedEffects = 1,
  NrConfounders = 10,
  distConfounders = "norm",
  mConfounders = 0,
  sdConfounders = 1,
  catConfounders = NULL,
  probConfounders = NULL,
  distBetaConfounders = "norm",
  mBetaConfounders = 0,
  sdBetaConfounders = 1,
  pTraitsAffectedConfounders = 1,
  pIndependentConfounders = 0.4,
  pTraitIndependentConfounders = 0.2,
  keepSameIndependentConfounders = FALSE,
  pcorr = 0.8,
  corrmatfile = NULL,
  meanNoiseBg = 0,
  sdNoiseBg = 1,
  nonlinear = NULL,
  logbase = 10,
  expbase = NULL,
  power = NULL,
  customTransform = NULL,
  transformNeg = "abs",
  proportionNonlinear = 0,
  sampleID = "ID_",
  phenoID = "Trait_",
  snpID = "SNP_",
  seed = 219453,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runSimulation_+3A_n">N</code></td>
<td>
<p>Number [integer] of samples to simulate.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_p">P</code></td>
<td>
<p>Number [integer] of phenotypes to simulate.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_genvar">genVar</code></td>
<td>
<p>Proportion [double] of total genetic variance.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_h2s">h2s</code></td>
<td>
<p>Proportion [double] of genetic variance of genetic variant effects.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_theta">theta</code></td>
<td>
<p>Proportion [double] of variance of shared genetic variant 
effects.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_h2bg">h2bg</code></td>
<td>
<p>Proportion [double] of genetic variance of infinitesimal genetic 
effects; either h2s or h2bg have to be specified and h2s + h2bg = 1.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_eta">eta</code></td>
<td>
<p>Proportion [double] of variance of shared infinitesimal genetic 
effects.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_noisevar">noiseVar</code></td>
<td>
<p>Proportion [double] of total noise variance.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_rho">rho</code></td>
<td>
<p>Proportion [double] of noise variance of correlated effects; sum 
of rho, delta and phi has to be equal 1.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_delta">delta</code></td>
<td>
<p>Proportion [double] of noise variance of non-genetic covariate 
effects; sum of rho, delta and phi  has to be equal 1.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_gamma">gamma</code></td>
<td>
<p>Proportion [double] of variance of shared non-genetic covariate 
effects.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_phi">phi</code></td>
<td>
<p>Proportion [double] of noise variance of observational noise 
effects; sum of rho, delta and phi has to be equal 1.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_alpha">alpha</code></td>
<td>
<p>Variance [double] of shared observational noise effect.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_tnrsnp">tNrSNP</code></td>
<td>
<p>Total number [integer] of SNPs to simulate; these SNPs are used
for kinship estimation.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_cnrsnp">cNrSNP</code></td>
<td>
<p>Number [integer] of causal SNPs; used as genetic variant 
effects.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_snpfrequencies">SNPfrequencies</code></td>
<td>
<p>Vector of allele frequencies [double] from which to 
sample.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_genotypefile">genotypefile</code></td>
<td>
<p>Needed when reading external genotypes (into memory), 
path/to/genotype file [string] in format specified by <a href="base.html#topic+format">format</a>.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_format">format</code></td>
<td>
<p>Needed when reading external genotypes, specifies 
the format of the genotype data; has to be one of plink, oxgen, genome, 
bimbam and delim when reading files into memory, or one of oxgen, bimbam or
delim if sampling genetic variants from file; for details see
<a href="#topic+readStandardGenotypes">readStandardGenotypes</a> and <a href="#topic+getCausalSNPs">getCausalSNPs</a>.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_genofileprefix">genoFilePrefix</code></td>
<td>
<p>Needed when sampling cuasal SNPs from file, full 
path/to/chromosome-wise-genotype-file-ending-before-&quot;chrChromosomeNumber&quot; 
(no '~' expansion!) [string]</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_genofilesuffix">genoFileSuffix</code></td>
<td>
<p>Needed when sampling causal SNPs from file, 
following chromosome number including fileformat (e.g. &quot;.csv&quot;) [string]</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_genodelimiter">genoDelimiter</code></td>
<td>
<p>Field separator [string] of genotypefile or genoFile if
format == delim.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_skipfields">skipFields</code></td>
<td>
<p>Number [integer] of fields (columns) in to skip in 
genoFilePrefix-genoFileSuffix-file. See details in <a href="#topic+getCausalSNPs">getCausalSNPs</a> if
format == delim.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_header">header</code></td>
<td>
<p>[logical] Can be set to indicate if
genoFilePrefix-genoFileSuffix file has a header for format == 'delim'. 
See details in <a href="#topic+getCausalSNPs">getCausalSNPs</a>.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_probabilities">probabilities</code></td>
<td>
<p>[bool]. If set to TRUE, the genotypes in the files 
described by genoFilePrefix and genoFileSuffix are provided as triplets of 
probablities (p(AA), p(Aa), p(aa)) and are converted into their expected 
genotype frequencies by 0*p(AA) + p(Aa) + 2p(aa) via <a href="#topic+probGen2expGen">probGen2expGen</a>.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_chr">chr</code></td>
<td>
<p>Numeric vector of chromosomes [integer] to chose NrCausalSNPs 
from; only used when external genotype data is sampled i.e. 
!is.null(genoFilePrefix)</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_nrsnpsonchromosome">NrSNPsOnChromosome</code></td>
<td>
<p>Specifies the number of SNPs [integer] per entry in 
chr (see above); has to be the same length as chr. If not provided, lines in 
genoFilePrefix-genoFileSuffix file will be counted (which can be slow for 
large files).</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_nrchrcausal">NrChrCausal</code></td>
<td>
<p>Number [integer] of causal chromosomes to chose 
NrCausalSNPs from (as opposed to the actual chromosomes to chose from via chr
);  only used when external genotype data is sampled i.e. 
!is.null(genoFilePrefix).</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_kinshipfile">kinshipfile</code></td>
<td>
<p>path/to/kinshipfile [string]; if provided, 
kinship for simulation of genetic backgound effect will be read from file.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_kinshipheader">kinshipHeader</code></td>
<td>
<p>[boolean] If TRUE kinship file has header information.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_kinshipdelimiter">kinshipDelimiter</code></td>
<td>
<p>Field separator [string] of kinship file.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_standardise">standardise</code></td>
<td>
<p>[boolean] If TRUE genotypes will be standardised for 
kinship estimation (recommended).</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_distbetagenetic">distBetaGenetic</code></td>
<td>
<p>Name [string] of distribution to use to simulate 
effect sizes of genetic variants; one of &quot;unif&quot; or &quot;norm&quot;.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_mbetagenetic">mBetaGenetic</code></td>
<td>
<p>Mean/midpoint [double] of normal/uniform distribution 
for effect sizes of genetic variants.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_sdbetagenetic">sdBetaGenetic</code></td>
<td>
<p>Standard deviation/extension from midpoint [double] 
of normal/uniform distribution for effect sizes of genetic variants.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_ptraitsaffectedgenetics">pTraitsAffectedGenetics</code></td>
<td>
<p>Proportion [double] of traits affected by the 
genetic variant effect. For non-integer results of pTraitsAffected*P, the 
ceiling of the result is used. Allows to simulate for instance different 
levels of pleiotropy.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_pindependentgenetic">pIndependentGenetic</code></td>
<td>
<p>Proportion [double] of genetic variant effects to 
have a trait-independent fixed effect.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_ptraitindependentgenetic">pTraitIndependentGenetic</code></td>
<td>
<p>Proportion [double] of traits influenced by 
independent genetic variant effects.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_keepsameindependentsnps">keepSameIndependentSNPs</code></td>
<td>
<p>[boolean] If set to TRUE, the 
independent SNPs effects always influence the same subset of traits.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_nrfixedeffects">NrFixedEffects</code></td>
<td>
<p>Number [integer] of different non-genetic covariate 
effects to simulate; allows to simulate non-genetic covariate effects from 
different distributions or with different parameters.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_nrconfounders">NrConfounders</code></td>
<td>
<p>Number [integer] of non-genetic covariates; used as 
non-genetic covariate effects.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_distconfounders">distConfounders</code></td>
<td>
<p>Vector of name(s) [string] of distributions to use to 
simulate confounders; one of &quot;unif&quot;, &quot;norm&quot;, &quot;bin&quot;, &quot;cat_norm&quot;, &quot;cat_unif&quot;.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_mconfounders">mConfounders</code></td>
<td>
<p>Vector of mean(s)/midpoint(s) [double] of 
normal/uniform distribution for confounders.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_sdconfounders">sdConfounders</code></td>
<td>
<p>Vector of standard deviation(s)/extension from 
midpoint(s) [double] of normal/uniform distribution for confounders.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_catconfounders">catConfounders</code></td>
<td>
<p>Vector of confounder categories [factor]; required if 
distConfounders &quot;cat_norm&quot; or &quot;cat_unif&quot;.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_probconfounders">probConfounders</code></td>
<td>
<p>Vector of probability(ies) [double] of binomial 
confounders (0/1); required if distConfounders &quot;bin&quot;.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_distbetaconfounders">distBetaConfounders</code></td>
<td>
<p>Vector of name(s) [string] of distribution to use 
to simulate effect sizes of confounders; one of &quot;unif&quot; or &quot;norm&quot;.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_mbetaconfounders">mBetaConfounders</code></td>
<td>
<p>Vector of mean(s)/midpoint(s) [double] of 
normal/uniform distribution for effect sizes of confounders.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_sdbetaconfounders">sdBetaConfounders</code></td>
<td>
<p>Vector of standard deviation(s)/extension from 
midpoint(s) [double] of normal/uniform distribution for effect sizes of 
confounders.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_ptraitsaffectedconfounders">pTraitsAffectedConfounders</code></td>
<td>
<p>Proportion(s) [double] of traits 
affected by the non-genetic covariates. For non-integer results of 
pTraitsAffected*P, the ceiling of the result is used.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_pindependentconfounders">pIndependentConfounders</code></td>
<td>
<p>Vector of proportion(s) [double] of 
non-genetic covariate effects to have a trait-independent effect.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_ptraitindependentconfounders">pTraitIndependentConfounders</code></td>
<td>
<p>Vector of proportion(s) [double] of 
traits influenced by independent non-genetic covariate effects.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_keepsameindependentconfounders">keepSameIndependentConfounders</code></td>
<td>
<p>[boolean] If set to TRUE, the 
independent confounder effects always influence the same subset of traits.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_pcorr">pcorr</code></td>
<td>
<p>Correlation [double] between phenotypes.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_corrmatfile">corrmatfile</code></td>
<td>
<p>path/to/corrmatfile.csv [string] with comma-separated 
[P x P] numeric [double] correlation matrix; if provided,  correlation matrix 
for simulation of correlated backgound effect will be read from file; 
file should NOT contain an index or header column.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_meannoisebg">meanNoiseBg</code></td>
<td>
<p>Mean [double] of the normal distributions for the 
simulation observational noise effects.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_sdnoisebg">sdNoiseBg</code></td>
<td>
<p>Standard deviation [double] of the normal distributions for 
the simulations of the observational noise effects.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_nonlinear">nonlinear</code></td>
<td>
<p>nonlinear transformation method [string]; one exp 
(exponential), log (logarithm), poly (polynomial), sqrt (squareroot) or 
custom (user-supplied function); if log or exp, base can be specified; if 
poly, power can be specified; if custom, a custom function (see for details). 
Non-linear transformation is optional, default is NULL ie no transformation
(see details).</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_logbase">logbase</code></td>
<td>
<p>[int] base of logarithm for non-linear phenotype 
transformation (see details).</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_expbase">expbase</code></td>
<td>
<p>[int] base of exponential function for non-linear phenotype 
transformation (see details).</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_power">power</code></td>
<td>
<p>[double] power of polynomial function for non-linear phenotype 
transformation.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_customtransform">customTransform</code></td>
<td>
<p>[function] custom transformation function accepting 
a single argument.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_transformneg">transformNeg</code></td>
<td>
<p>[string] transformation method for negative values in non
linear phenotype transformation. One of abs (absolute value) or set0 (set all 
negative values to zero). If nonlinear==log and transformNeg==set0, negative
values set to 1e-5</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_proportionnonlinear">proportionNonlinear</code></td>
<td>
<p>[double] proportion of the phenotype to be non-
linear (see details)</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_sampleid">sampleID</code></td>
<td>
<p>Prefix [string] for naming samples (will be followed by 
sample number from 1 to N when constructing sample IDs); only used if 
genotypes/kinship are simulated/do not have sample IDs.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_phenoid">phenoID</code></td>
<td>
<p>Prefix [string] for naming traits (will be followed by 
phenotypes number from 1 to P when constructing phenotype IDs).</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_snpid">snpID</code></td>
<td>
<p>Prefix [string] for naming SNPs (will be followed by 
SNP number from 1 to NrSNP when constructing SNP IDs).</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_seed">seed</code></td>
<td>
<p>Seed [integer] to initiate random number generation.</p>
</td></tr>
<tr><td><code id="runSimulation_+3A_verbose">verbose</code></td>
<td>
<p>[boolean]; If TRUE, progress info is printed to standard out</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Phenotypes are modeled under a linear additive model where
Y = WA + BX + G + C + Phi, with WA the non-genetic covariates, BX the genetic
variant effects, G the infinitesimal genetic effects, C the correlated 
background effects and the Phi the observational noise. For more information
on these components look at the respective function descriptions (see also)
Optionally the phenotypes can be non-linearly transformed via:
Y_trans = (1-alpha) x Y + alpha x f(Y). Alpha is the proportion of non-
linearity of the phenotype and f is a non-linear transformation, and one of
exp, log or sqrt.
</p>


<h3>Value</h3>

<p>Named list of i) dataframe of proportion of variance 
explained for each component (varComponents), 
ii) a named list with the final simulated phenotype components 
(phenoComponentsFinal), iii) a named list with the intermediate simulated 
phenotype components (phenoComponentsIntermediate), iv) a named list of 
parameters describing the model setup (setup) and v) a named list of raw 
components (rawComponents) used for genetic effect simulation (genotypes 
and/or kinship, eigenvalues and eigenvectors of kinship)
</p>


<h3>See Also</h3>

<p><a href="#topic+setModel">setModel</a>, <a href="#topic+geneticFixedEffects">geneticFixedEffects</a>,
<a href="#topic+geneticBgEffects">geneticBgEffects</a>, <a href="#topic+noiseBgEffects">noiseBgEffects</a>, <a href="#topic+noiseFixedEffects">noiseFixedEffects</a>,
<a href="#topic+correlatedBgEffects">correlatedBgEffects</a> and <a href="#topic+rescaleVariance">rescaleVariance</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simulate phenotype of 100 samples, 10 traits from genetic and noise 
# background effects, with variance explained of 0.2 and 0.8 respectively
genVar = 0.2
simulatedPhenotype &lt;- runSimulation(N=100, P=5, cNrSNP=10,
genVar=genVar, h2s=1, phi=1)
</code></pre>

<hr>
<h2 id='savePheno'>Save final phenotype and phenotype components.</h2><span id='topic+savePheno'></span>

<h3>Description</h3>

<p>savePheno saves simulated phenotypes and their components, model setup 
parameters and variance components to the specified directories. Requires a 
simulatedData list which is the output of <a href="#topic+runSimulation">runSimulation</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>savePheno(
  simulatedData,
  directory,
  format = ".csv",
  outstring = "",
  saveIntermediate = TRUE,
  intercept_gemma = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="savePheno_+3A_simulateddata">simulatedData</code></td>
<td>
<p>Named list of i) dataframe of proportion of variance 
explained for each component (varComponents), 
ii) a named list with the final simulated phenotype components 
(phenoComponentsFinal), iii) a named list with the intermediate simulated 
phenotype components (phenoComponentsIntermediate), iv) a named list of 
parameters describing the model setup (setup) and v) a named list of raw 
components (rawComponents) used for genetic effect simulation (genotypes 
and/or kinship); obtained from <a href="#topic+runSimulation">runSimulation</a></p>
</td></tr>
<tr><td><code id="savePheno_+3A_directory">directory</code></td>
<td>
<p>Absolute path (no tilde expansion) to parent directory 
[string] where simulated data should be saved [needs user writing 
permission]</p>
</td></tr>
<tr><td><code id="savePheno_+3A_format">format</code></td>
<td>
<p>Vector of format name(s) [string] specifying the output format;
multiple output formats can be requested. Options are: plink, bimbam, 
snptest, gemma, limmbo, csv or rds. For information on format see details. In
orde to save intermediate phenotype components, at least one of csv or rds
need to be specified. plink/bimbam/snptest will only save final
phenotype/genotype, kinship and covariate data.</p>
</td></tr>
<tr><td><code id="savePheno_+3A_outstring">outstring</code></td>
<td>
<p>Optional name [string] of subdirectory (in relation to 
directory) to save set-up dependent simulation results; if
set to NULL, subdirectory named by NrSamples, NrSNPs, genetic Model and 
noise Model and genVar is created.</p>
</td></tr>
<tr><td><code id="savePheno_+3A_saveintermediate">saveIntermediate</code></td>
<td>
<p>[bool] If TRUE, intermediate phenotype components
such as shared and independent effect components are saved.</p>
</td></tr>
<tr><td><code id="savePheno_+3A_intercept_gemma">intercept_gemma</code></td>
<td>
<p>[boolean] When modeling an intercept term in gemma, a 
column of 1's have to be appended to the covariate files. Set intercept_gemma
to TRUE to include a column of 1's in the output; only used when 
&quot;gemma&quot; %in% format</p>
</td></tr>
<tr><td><code id="savePheno_+3A_verbose">verbose</code></td>
<td>
<p>[boolean]; If TRUE, progress info is printed to standard out</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Path [string] to final output directory. If outstring is 
NULL, this directory will be a subdirectory of the input directory.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>simulatedPhenotype &lt;- runSimulation(N=100, P=5, cNrSNP=10,
genVar=0.2, h2s=0.2, phi=1)
## Not run: 
outputdir &lt;- savePheno(simulatedPhenotype, directory=tempdir(),  
outstring="Data_simulation", format=c("csv", "plink"))
## End(Not run)
</code></pre>

<hr>
<h2 id='setModel'>Set simulation model.</h2><span id='topic+setModel'></span>

<h3>Description</h3>

<p>Based on parameters provided, this function sets the name for the phenotype 
simulation. It carries out compatibiltiy checks of the specifie parameters 
and checks for any missing information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setModel(
  genVar = NULL,
  h2s = NULL,
  theta = 0.8,
  h2bg = NULL,
  eta = 0.8,
  noiseVar = NULL,
  delta = NULL,
  gamma = 0.8,
  rho = NULL,
  phi = NULL,
  alpha = 0.8,
  pcorr = 0.6,
  pIndependentConfounders = 0.4,
  pTraitIndependentConfounders = 0.2,
  pIndependentGenetic = 0.4,
  pTraitIndependentGenetic = 0.2,
  proportionNonlinear = 0,
  cNrSNP = NULL,
  NrConfounders = 10,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="setModel_+3A_genvar">genVar</code></td>
<td>
<p>Total genetic variance [double].</p>
</td></tr>
<tr><td><code id="setModel_+3A_h2s">h2s</code></td>
<td>
<p>Proportion [double] of variance of genetic variant effects.</p>
</td></tr>
<tr><td><code id="setModel_+3A_theta">theta</code></td>
<td>
<p>Proportion [double] of variance of shared genetic variant 
effects.</p>
</td></tr>
<tr><td><code id="setModel_+3A_h2bg">h2bg</code></td>
<td>
<p>Proportion [double] of variance of infinitesimal genetic effects
i.e. correlation introduced by sample kinship).</p>
</td></tr>
<tr><td><code id="setModel_+3A_eta">eta</code></td>
<td>
<p>Proportion [double] of variance of shared infinitesimal genetic 
effects.</p>
</td></tr>
<tr><td><code id="setModel_+3A_noisevar">noiseVar</code></td>
<td>
<p>Total noise variance [double].</p>
</td></tr>
<tr><td><code id="setModel_+3A_delta">delta</code></td>
<td>
<p>Proportion [double] of variance of non-genetic covariate effect.</p>
</td></tr>
<tr><td><code id="setModel_+3A_gamma">gamma</code></td>
<td>
<p>Proportion [double] of variance of shared non-genetic covariate 
effects.</p>
</td></tr>
<tr><td><code id="setModel_+3A_rho">rho</code></td>
<td>
<p>Proportion [double] of variance of correlated noise effects.</p>
</td></tr>
<tr><td><code id="setModel_+3A_phi">phi</code></td>
<td>
<p>Proportion [double] of variance of observational noise effects.</p>
</td></tr>
<tr><td><code id="setModel_+3A_alpha">alpha</code></td>
<td>
<p>Proportion [double] of variance of shared observational noise 
effect.</p>
</td></tr>
<tr><td><code id="setModel_+3A_pcorr">pcorr</code></td>
<td>
<p>Correlation [double] between phenotypes.</p>
</td></tr>
<tr><td><code id="setModel_+3A_pindependentconfounders">pIndependentConfounders</code></td>
<td>
<p>Proportion [double] of non-genetic covariate 
to have a trait-independent effect.</p>
</td></tr>
<tr><td><code id="setModel_+3A_ptraitindependentconfounders">pTraitIndependentConfounders</code></td>
<td>
<p>Proportion [double] of traits influenced
by independent non-genetic covariate effects.</p>
</td></tr>
<tr><td><code id="setModel_+3A_pindependentgenetic">pIndependentGenetic</code></td>
<td>
<p>Proportion [double] of genetic variant effects to 
have a trait-independent fixed effect.</p>
</td></tr>
<tr><td><code id="setModel_+3A_ptraitindependentgenetic">pTraitIndependentGenetic</code></td>
<td>
<p>Proportion [double] of traits influenced by 
independent genetic variant effects.</p>
</td></tr>
<tr><td><code id="setModel_+3A_proportionnonlinear">proportionNonlinear</code></td>
<td>
<p>[double] proportion of the phenotype to be non-
linear</p>
</td></tr>
<tr><td><code id="setModel_+3A_cnrsnp">cNrSNP</code></td>
<td>
<p>Number [integer] of causal SNPs; used as genetic variant 
effects.</p>
</td></tr>
<tr><td><code id="setModel_+3A_nrconfounders">NrConfounders</code></td>
<td>
<p>Number [integer] of non-genetic covariates; used as 
non-genetic covariate effects.</p>
</td></tr>
<tr><td><code id="setModel_+3A_verbose">verbose</code></td>
<td>
<p>[boolean]; If TRUE, progress info is printed to standard out.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named list containing the genetic model (modelGenetic), the noise 
model (modelNoise) and the input parameters (h2s, h2bg, noiseVar, rho, delta, 
phi, gamma, theta, eta, alpha, pcorr, proportionNonlinear). Model options 
are: modelNoise: &quot;noNoise&quot;, &quot;noiseFixedOnly&quot;, &quot;noiseBgOnly&quot;, 
&quot;noiseCorrelatedOnly&quot;, &quot;noiseFixedAndBg&quot;,&quot;noiseCorrelatedAndBg&quot;, 
&quot;noiseFixedAndCorrelated&quot;, &quot;noiseFixedAndBgAndCorrelated&quot;
modelGenetic: &quot;noGenetic&quot;,&quot;geneticBgOnly&quot;, &quot;geneticFixedOnly&quot;,
&quot;geneticFixedAndBg&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#genetic fixed effects only
model &lt;- setModel(genVar=1, h2s=1)

#genetic fixed and bg effects
model &lt;- setModel(genVar=1, h2s=0.01)

#genetic and noise fixed effects only
model &lt;- setModel(genVar=0.4, h2s=1, delta=1)
</code></pre>

<hr>
<h2 id='simulateDist'>Data simulation for different distributions.</h2><span id='topic+simulateDist'></span>

<h3>Description</h3>

<p>Wrapper function to simulate data from different distribution with different 
parameter settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateDist(
  x,
  dist = c("unif", "norm", "bin", "cat_norm", "cat_unif"),
  m = NULL,
  std = 1,
  categories = NULL,
  prob = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulateDist_+3A_x">x</code></td>
<td>
<p>The number [integer] of observations to simulate.</p>
</td></tr>
<tr><td><code id="simulateDist_+3A_dist">dist</code></td>
<td>
<p>Name of distribution [string] from which the observations are 
drawn. 'norm' is the normal distribution, 'unif' the uniform distribution 
'bin' the binomial distribution, &quot;cat_norm&quot; samples categorical variables 
according to a normal distribution and &quot;cat_unif&quot; 
according to a uniform distribution. For &quot;cat_norm&quot;, length(category)/2 is 
used mean for the normal distribution unless
specified otherwise.</p>
</td></tr>
<tr><td><code id="simulateDist_+3A_m">m</code></td>
<td>
<p>Mean of the normal distribution [double]/the mean between min 
and max for the uniform distribution [double]/ 
the rank of the category to be used as mean for &quot;cat_norm&quot; [integer].</p>
</td></tr>
<tr><td><code id="simulateDist_+3A_std">std</code></td>
<td>
<p>Standard deviation of the normal distribution or the distance 
of min/max from the mean for the uniform distribution [double].</p>
</td></tr>
<tr><td><code id="simulateDist_+3A_categories">categories</code></td>
<td>
<p>Number of categories [integer] for simulating categorical 
variables (for distr=&quot;cat_norm&quot; or &quot;cat_unif&quot;).</p>
</td></tr>
<tr><td><code id="simulateDist_+3A_prob">prob</code></td>
<td>
<p>Probability [double] of success for each trial 
(for distr=&quot;bin&quot;).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric vector of length [x] with the sampled values
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+runif">runif</a></code>, <code><a href="stats.html#topic+rnorm">rnorm</a></code>, <code><a href="stats.html#topic+rbinom">rbinom</a></code> for 
documentation of the underlying distributions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>normal &lt;- simulateDist(x=10, dist="norm", m=2, std=4)
cat_normal &lt;- simulateDist(x=10, dist="cat_norm", categories=5)
cat_uniform &lt;- simulateDist(x=10, dist="cat_unif", categories=5)
uniform &lt;- simulateDist(x=10, dist="unif", m=4, std=1)
binomial &lt;- simulateDist(x=10, dist="bin", prob=0.4)
</code></pre>

<hr>
<h2 id='simulateGenotypes'>Simulate bi-allelic genotypes.</h2><span id='topic+simulateGenotypes'></span>

<h3>Description</h3>

<p>Simulate bi-allelic genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateGenotypes(
  N,
  NrSNP = 5000,
  frequencies = c(0.1, 0.2, 0.4),
  sampleID = "ID_",
  snpID = "SNP_",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulateGenotypes_+3A_n">N</code></td>
<td>
<p>Number of samples for which to simulate bi-allelic genotypes.</p>
</td></tr>
<tr><td><code id="simulateGenotypes_+3A_nrsnp">NrSNP</code></td>
<td>
<p>Number of SNPs to simulate.</p>
</td></tr>
<tr><td><code id="simulateGenotypes_+3A_frequencies">frequencies</code></td>
<td>
<p>Vector of allele frequencies [double] from which to
sample.</p>
</td></tr>
<tr><td><code id="simulateGenotypes_+3A_sampleid">sampleID</code></td>
<td>
<p>Prefix [string] for naming samples (will be followed by
sample number from 1 to N when constructing id_samples).</p>
</td></tr>
<tr><td><code id="simulateGenotypes_+3A_snpid">snpID</code></td>
<td>
<p>Prefix [string] for naming SNPs (will be followed by SNP number
from 1 to NrSNP when constructing id_snps).</p>
</td></tr>
<tr><td><code id="simulateGenotypes_+3A_verbose">verbose</code></td>
<td>
<p>[boolean] If TRUE, progress info is printed to standard out.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Named list with [N x NrSNP] matrix of simulated genotypes
(genotypes), their SNP frequencies (freq), a vector of sample IDs
(id_samples) and a vector of SNP IDs (id_snps).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+standardiseGenotypes">standardiseGenotypes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>N10NrSNP10 &lt;- simulateGenotypes(N=10, NrSNP=10)
N10NrSNP10 &lt;- simulateGenotypes(N=10, NrSNP=10,
frequencies=c(0.2,0.3,0.4))
</code></pre>

<hr>
<h2 id='simulatePhenotypes'>Command line execution for PhenotypeSimulator.</h2><span id='topic+simulatePhenotypes'></span>

<h3>Description</h3>

<p>simulatePhenotypes runs without arguments. Upon call, it reads command-line
parameters and supplies these to <code><a href="#topic+runSimulation">runSimulation</a></code> and 
<code><a href="#topic+savePheno">savePheno</a></code>. For details on input to <code><a href="#topic+runSimulation">runSimulation</a></code> 
and <code><a href="#topic+savePheno">savePheno</a></code>, please refer to their help pages. For help on 
the command line arguments that can be passed, see examples below. From the 
command line, the help function can be called via 
'Rscript -e &quot;PhenotypeSimulator::simulatePhenotypes()&quot; &ndash;args &ndash;help
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulatePhenotypes()
</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'># (not run)
# Simulate simple phenotype of genetic and noise background effects only:
# (not run)
# Rscript -e "PhenotypeSimulator::simulatePhenotypes()" \
#--args \ 
#--NrSamples=100 --NrPhenotypes=15 \
#--tNrSNPs=10000 --cNrSNPs=30 \
#--SNPfrequencies=0.05,0.1,0.3,0.4 \
#--genVar=0.4 --h2s=0.025 --phi=0.6 --delta=0.3 --gamma=1 \
#--pcorr=0.8 \
#--NrFixedEffects=4 --NrConfounders=1,2,1,2 \
#--pIndependentConfounders=0,1,1,0.5 \
#--distConfounders=bin,cat_norm,cat_unif,norm \
#--probConfounders=0.2 \
#--catConfounders=0,3,4,0 \
#--directory=/tmp \
#--showProgress \
</code></pre>

<hr>
<h2 id='standardiseGenotypes'>Standardise genotypes.</h2><span id='topic+standardiseGenotypes'></span>

<h3>Description</h3>

<p>Genotypes are standardised as described in Yang et al:
snp_standardised = (snp - 2 * ref_allele_freq)/
sqrt(2 * ref_allele_freq * alt_allele_freq).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>standardiseGenotypes(geno, impute = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="standardiseGenotypes_+3A_geno">geno</code></td>
<td>
<p>[N x NrSNP] Matrix/dataframe of genotypes [integer]/[double].</p>
</td></tr>
<tr><td><code id="standardiseGenotypes_+3A_impute">impute</code></td>
<td>
<p>[logical] Indicating if missing genotypes should be imputed; if
set FALSE and data contains missing values,  <code>standardiseGenotypes</code> will
return an error.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Missing genotypes can be mean-imputed and rounded to nearest integer
before standardisation. If genotypes contain missing values and impute is set
to FALSE, <code>standardiseGenotypes</code> will return an error.
</p>


<h3>Value</h3>

<p>[N x NrSNP] Matrix of standardised genotypes [double].
</p>


<h3>References</h3>

<p>Yang, J., Lee, S.H., Goddard, M.E., Visscher, P.M. (2011) GCTA:
a tool for genome-wide complex trait analysis, AJHG: 88
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getAlleleFrequencies">getAlleleFrequencies</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>geno &lt;- cbind(rbinom(2000, 2, 0.3), rbinom(2000, 2, 0.4),rbinom(2000, 2, 0.5))
geno_sd &lt;- standardiseGenotypes(geno)
</code></pre>

<hr>
<h2 id='testNumerics'>Test lists for different properties of numerics.</h2><span id='topic+testNumerics'></span>

<h3>Description</h3>

<p>Test all elements of a list if they are numeric, positive numbers, integers
or proportions (range 0-1)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testNumerics(numbers, positives = NULL, integers = NULL, proportions = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="testNumerics_+3A_numbers">numbers</code></td>
<td>
<p>List whose elements are tested for being numeric</p>
</td></tr>
<tr><td><code id="testNumerics_+3A_positives">positives</code></td>
<td>
<p>List whose elements are tested for being positive numbers</p>
</td></tr>
<tr><td><code id="testNumerics_+3A_integers">integers</code></td>
<td>
<p>List whose elements are tested for being integers</p>
</td></tr>
<tr><td><code id="testNumerics_+3A_proportions">proportions</code></td>
<td>
<p>List whose elements are tested for being proportions 
between 0 and 1</p>
</td></tr>
</table>

<hr>
<h2 id='transformNonlinear'>Phenotype transformation.</h2><span id='topic+transformNonlinear'></span>

<h3>Description</h3>

<p>Transformation of phenotype component by applying a user-specified 
non-linear transformation to the phenotype component.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transformNonlinear(
  component,
  alpha,
  method,
  logbase = 10,
  power = 2,
  expbase = NULL,
  transformNeg = "abs",
  f = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transformNonlinear_+3A_component">component</code></td>
<td>
<p>[N x P] Phenotype matrix [double] where [N] are the number 
of samples and P the number of phenotypes</p>
</td></tr>
<tr><td><code id="transformNonlinear_+3A_alpha">alpha</code></td>
<td>
<p>[double] weighting scalar for non-linearity: alpha==0 fully
linear phenotype, alpha==1 fully non-linear phenotype. See @details.</p>
</td></tr>
<tr><td><code id="transformNonlinear_+3A_method">method</code></td>
<td>
<p>[string] one of exp (exponential), log (logarithm), poly
(polynomial), sqrt (squareroot) or custom (user-supplied function)</p>
</td></tr>
<tr><td><code id="transformNonlinear_+3A_logbase">logbase</code></td>
<td>
<p>[int] when method==log, sets the log base for transformation</p>
</td></tr>
<tr><td><code id="transformNonlinear_+3A_power">power</code></td>
<td>
<p>[double] when method==poly, sets the power to raise to.</p>
</td></tr>
<tr><td><code id="transformNonlinear_+3A_expbase">expbase</code></td>
<td>
<p>[double] when method==exp, sets the exp base for
transformation.</p>
</td></tr>
<tr><td><code id="transformNonlinear_+3A_transformneg">transformNeg</code></td>
<td>
<p>[string] one of abs (absolute value) or set0 (set all 
negative values to zero). If method==log and transformNeg==set0, negative
values set to 1e-5</p>
</td></tr>
<tr><td><code id="transformNonlinear_+3A_f">f</code></td>
<td>
<p>[function] function accepting component as a single argument.</p>
</td></tr>
<tr><td><code id="transformNonlinear_+3A_verbose">verbose</code></td>
<td>
<p>[boolean]; If TRUE, progress info is printed to standard out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>transformNonlinear takes a phenotype component as input and 
transforms it according to the specified transformation method. The user can 
choose how strongly non-linear the resulting phenotype component should be, 
by specifying the weighting parameter alpha:
component_transformed = (1 - alpha) \* component + 
alpha \* transformfunction(component)
</p>


<h3>Value</h3>

<p>[N x P] transformed phenotype matrix [double]
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Simulate non-genetic covariate effects 
cov_effects &lt;- noiseFixedEffects(N=100, P=5)
# Transform logarithmically
covs_log &lt;- transformNonlinear(cov_effects$shared, alpha=0.5, method="log",
transformNeg="abs")
# Transform custom
f_custom &lt;- function(x) {x^2 + 3*x}
covs_custom &lt;- transformNonlinear(cov_effects$shared, alpha=0.5, 
method="custom", f=f_custom)
</code></pre>

<hr>
<h2 id='vmessage'>Print userinfo.</h2><span id='topic+vmessage'></span>

<h3>Description</h3>

<p>Wrapper function around <code><a href="base.html#topic+message">message</a></code> that allows to turn the 
printing of messages to standard out.
on or off
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vmessage(userinfo, verbose = TRUE, sep = " ")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vmessage_+3A_userinfo">userinfo</code></td>
<td>
<p>Vector of [string] element(s) and variables</p>
</td></tr>
<tr><td><code id="vmessage_+3A_verbose">verbose</code></td>
<td>
<p>[boolean] If TRUE message is displayed on standard out, if 
FALSE, message is suppressed.</p>
</td></tr>
<tr><td><code id="vmessage_+3A_sep">sep</code></td>
<td>
<p>Delimiter [string] to separate message elements when userinfo 
given as vector.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="base.html#topic+message">message</a></code> which this function wraps
</p>

<hr>
<h2 id='writeStandardOutput'>Write simulated data into formats used by standard GWAS software.</h2><span id='topic+writeStandardOutput'></span>

<h3>Description</h3>

<p>writeStandardOutput can write genotypes and phenotypes as well as possible
covariates and kinship matrices into a number of formats for standard GWAS 
software: plink, snptest, bimbam, gemma, limmbo. For more information on the 
different file formats see <em>External formats</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeStandardOutput(
  directory,
  phenotypes = NULL,
  genotypes = NULL,
  additionalPhenotypes = NULL,
  covariates = NULL,
  kinship = NULL,
  eval_kinship = NULL,
  evec_kinship = NULL,
  id_samples,
  id_snps,
  id_phenos,
  outstring = NULL,
  standardInput_samples = NULL,
  standardInput_genotypes = NULL,
  format = NULL,
  intercept_gemma = FALSE,
  nameAdditional = "_nonLinear",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeStandardOutput_+3A_directory">directory</code></td>
<td>
<p>Absolute path (no tilde expansion) to parent directory 
[string] where the data should be saved [needs user writing permission]</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_phenotypes">phenotypes</code></td>
<td>
<p>[NrSamples x NrTrait] Data.frame/matrix of phenotypes 
[doubles].</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_genotypes">genotypes</code></td>
<td>
<p>[NrSamples x NrSNP] Data.frame/matrix of genotypes 
[integers]/[doubles].</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_additionalphenotypes">additionalPhenotypes</code></td>
<td>
<p>[NrSamples x NrTrait] Data.frame/matrix of 
additional phenotypes (for instance non-linearly tranformed orginal</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_covariates">covariates</code></td>
<td>
<p>[NrSamples x NrCovariates] Data.frame/matrix of covariates
[integers]/[doubles].</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_kinship">kinship</code></td>
<td>
<p>[NrSamples x NrSamples] Data.frame/matrix of kinship estimates
[doubles].</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_eval_kinship">eval_kinship</code></td>
<td>
<p>[NrSamples] vector with eigenvalues of kinship matrix
[doubles].</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_evec_kinship">evec_kinship</code></td>
<td>
<p>[NrSamples x NrSamples] Data.frame/matrix with
eigenvectors of kinship matrix [doubles].</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_id_samples">id_samples</code></td>
<td>
<p>Vector of [NrSamples] sample IDs [string] of simulated 
phenotypes, genotypes and covariates.</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_id_snps">id_snps</code></td>
<td>
<p>Vector of [NrSNPs] SNP IDs [string] of (simulated) 
genotypes.</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_id_phenos">id_phenos</code></td>
<td>
<p>Vector of [NrTraits] phenotype IDs [string] of 
simulated phenotypes.</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_outstring">outstring</code></td>
<td>
<p>(optional) Name [string] of subdirectory (in relation to 
directory) to save set-up independent simulation results.</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_standardinput_samples">standardInput_samples</code></td>
<td>
<p>(optional) Data.frame of sample information 
obtained when genotypes were read from plink, oxgen or genome file.</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_standardinput_genotypes">standardInput_genotypes</code></td>
<td>
<p>(optional) Data.frame of genotypes obtained 
when reading genotypes from plink, oxgen, or genome file.</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_format">format</code></td>
<td>
<p>Vector of name(s) [string] of file formats, options are: 
&quot;plink&quot;, &quot;snptest&quot;, &quot;gemma&quot;, &quot;bimbam&quot;, &quot;delim&quot;. For details on the file 
formats see <em>External formats</em>.</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_intercept_gemma">intercept_gemma</code></td>
<td>
<p>[boolean] When modeling an intercept term in gemma, a 
column of 1's have to be appended to the covariate files. Set intercept_gemma
to TRUE to include a column of 1's in the output.</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_nameadditional">nameAdditional</code></td>
<td>
<p>name [string] of additonal phenotypes to be appended to 
filename.</p>
</td></tr>
<tr><td><code id="writeStandardOutput_+3A_verbose">verbose</code></td>
<td>
<p>[boolean]; If TRUE, progress info is printed to standard out</p>
</td></tr>
</table>


<h3>External formats</h3>


<ul>
<li><p> plink format: consists of three files, .bed, .bim and .fam. 
From <a href="https://www.cog-genomics.org/plink/1.9/formats">https://www.cog-genomics.org/plink/1.9/formats</a>:  The .bed 
files contain the  primary representation of genotype calls at biallelic 
variants in a binary format. The .bim is a text file with no header 
line, and one line  per variant with the following six fields: i) Chromosome 
code (either an integer, or 'X'/'Y'/'XY'/'MT'; '0' indicates unknown) or 
name, ii) Variant identifier, iii) Position in morgans or centimorgans (safe 
to use dummy value of '0'), iv) Base-pair coordinate (normally 1-based, but 
0 ok; limited to 231-2), v) Allele 1 (corresponding to clear bits in .bed; 
usually minor), vi) Allele 2 (corresponding to set bits in .bed; usually 
major). The .fam file is a text file with no header line, and one line per 
sample with the following six fields: i) Family ID ('FID'), ii), Within-
family ID ('IID'; cannot be '0'), iii) Within-family ID of father ('0' if 
father isn't in dataset, iv) within-family ID of mother ('0' if mother isn't 
in dataset), v) sex code ('1' = male, '2' = female, '0' = unknown), vi) 
Phenotype value ('1' = control, '2' = case, '-9'/'0'/non-numeric = missing 
data if case/control)
</p>
</li>
<li><p> snptest format: consists of two files, the genotype file ending in .gen 
(genotypes_snptest.gen) and the sample file ending in .sample 
(Ysim_snptest.sample). From
<a href="https://www.well.ox.ac.uk/~gav/snptest/#input_file_formats">https://www.well.ox.ac.uk/~gav/snptest/#input_file_formats</a>:
The genotype file stores data on a one-line-per-SNP format. The first 5 
entries of each line should be the SNP ID, RS ID of the SNP, base-pair 
position of the SNP, the allele coded A and the allele coded B. The SNP ID 
can be used to denote the chromosome number of each SNP. The next three 
numbers on the line should be the probabilities of the three genotypes AA, 
AB and BB at the SNP for the first individual in the cohort. The next three 
numbers should be the genotype probabilities for the second individual in the 
cohort. The next three numbers are for the third individual and so on. The 
order of individuals in the genotype file should match the order of the 
individuals in the sample file. The sample file has three parts (a) a header 
line detailing the names of the columns in the file, (b) a line detailing the 
types of variables stored in each column, and (c) a line for each individual 
detailing the information for that individual. 
a) The header line needs a minimum of three entries. The first three entries 
should always be ID_1, ID_2 and missing. They denote that the first three 
columns contain the first ID, second ID and missing data proportion of each 
individual. Additional entries on this line should be the names of covariates 
or phenotypes that are included in the file. In the above example, there are 
4 covariates named cov_1, cov_2, cov_3, cov_4, a continuous phenotype named 
pheno1 and a binary phenotype named bin1. All phenotypes should appear after 
the covariates in this file. b) The second line (the variable type line) 
details the type of variables included in each column. The first three 
entries of this line should be set to 0. Subsequent entries in this line for 
covariates and phenotypes should be specified by the following rules: D for
Discrete covariates (coded using positive integers), C for Continuous 
covariates, P for Continuous Phenotype, B for Binary Phenotype (0 = Controls, 
1 = Cases). c) Individual information: one line for each individual 
containing the information specified by the entries of the header line.
Entries of the sample file are separated by spaces.
</p>
</li>
<li><p> bimbam format: consists of a) a simple, tab-separated phenotype file
without sample or phenotype header/index (Ysim_bimbam.txt) and b) the mean
genotype file format which is a single file, without information on
individuals: (genotypes.bimbam). From the bimbam documentation at
<a href="http://www.haplotype.org/software.html">http://www.haplotype.org/software.html</a>: The first column of the mean
genotype files is the SNP ID, the second and third columns are allele types
with minor allele first. The rest columns are the mean genotypes of different
individuals – numbers between 0 and 2 that represents the (posterior) mean
genotype, or dosage of the minor allele. </p>
</li>
<li><p> gemma format: consists of a) a
simple, tab-separated phenotype file without sample or phenotype header/index
(Ysim_gemma.txt) and b) the mean genotype file format which is a single file,
without information on individuals(genotypes.gemma); a) and b) both the same
as above for bimbam format). In addition and if applicable, c) a kinship file
(kinship_gemma.txt) and d) covariate file (Covs_gemma.txt). From
<a href="http://www.xzlab.org/software/GEMMAmanual.pdf">http://www.xzlab.org/software/GEMMAmanual.pdf</a>: The kinship file
contains a NrSample × NrSample matrix, where each row and each column
corresponds to individuals in the same order as in the mean genotype file,
and ith row and jth column is a number indicating the relatedness value
between ith and jth individuals. The covariates file has the same format as
the phenotype file dsecribed above and must contain a column of 1’s if one
wants to include an intercept term (set parameter intercept_gemma=TRUE).
</p>
</li>
<li><p> limmbo format: consists of a) a comma-separated phenotype file without
sample IDs as index and phenotype IDs as header (Ysim_limmbo.csv), b) the
mean genotype file format with one genetic variant per line. The first column
contains the variant ID, column 2-N+1 contain the genotype code (numbers
between 0 and 2 that represent the (posterior) mean genotype/dosage of the
minor allele) for N samples, c)  a kinship file (kinship_limmbo.csv) and d)
covariate file (covs_limmbo.csv). From
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+readStandardGenotypes">readStandardGenotypes</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>simulation &lt;- runSimulation(N=10, P=2, genVar=0.4, h2s=0.2, phi=1)
genotypes &lt;- simulation$rawComponents$genotypes
kinship &lt;-  simulation$rawComponents$kinship
phenotypes &lt;- simulation$phenoComponents$Y

## Not run: 
# Save in plink format (.bed, .bim, .fam, Y_sim_plink.txt)
writeStandardOutput(directory=tempdir(), 
genotypes=genotypes$genotypes, phenotypes=phenotypes, 
id_samples = genotypes$id_samples, id_snps = genotypes$id_snps, 
id_phenos = colnames(phenotypes), format="plink")

# Save in gemma and snptest format
writeStandardOutput(directory=tempdir(), 
genotypes=genotypes$genotypes, phenotypes=phenotypes, 
id_samples = genotypes$id_samples, id_snps = genotypes$id_snps, 
id_phenos = colnames(phenotypes), kinship=kinship, 
format=c("snptest", "gemma"))

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
