<!DOCTYPE html><html lang="en"><head><title>Help for package phers</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {phers}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#demoSample'><p>Sample table of demographic information</p></a></li>
<li><a href='#diseaseDxIcdMap'><p>Mapping of diseases and diagnostic ICD codes</p></a></li>
<li><a href='#diseaseHpoMap'><p>Mapping of Mendelian diseases and their clinical features</p></a></li>
<li><a href='#getDxStatus'><p>Identify cases and controls for Mendelian diseases</p></a></li>
<li><a href='#getGeneticAssociations'><p>Perform association tests between phenotype risk scores and genotypes</p></a></li>
<li><a href='#getPhecodeOccurrences'><p>Map ICD code occurrences to phecode occurrences</p></a></li>
<li><a href='#getResidualScores'><p>Calculate residual phenotype risk scores</p></a></li>
<li><a href='#getScores'><p>Calculate phenotype risk scores</p></a></li>
<li><a href='#getWeights'><p>Calculate phecode-specific weights for phenotype risk scores</p></a></li>
<li><a href='#hpoPhecodeMap'><p>Mapping of HPO terms and phecodes</p></a></li>
<li><a href='#icdPhecodeMap'><p>Mapping of ICD codes and phecodes</p></a></li>
<li><a href='#icdSample'><p>Sample table of ICD occurrences</p></a></li>
<li><a href='#mapDiseaseToPhecode'><p>Map diseases to phecodes via HPO terms</p></a></li>
<li><a href='#preCalcWeights'><p>Pre-calculated weights for calculating phenotype risk scores</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Calculate Phenotype Risk Scores</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Use phenotype risk scores based on linked clinical and genetic data
  to study Mendelian disease and rare genetic variants. See Bastarache et al.
  2018 &lt;<a href="https://doi.org/10.1126%2Fscience.aal4043">doi:10.1126/science.aal4043</a>&gt;.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://phers.hugheylab.org">https://phers.hugheylab.org</a>, <a href="https://github.com/hugheylab/phers">https://github.com/hugheylab/phers</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>BEDMatrix (&ge; 2.0.3), checkmate (&ge; 2.0.0), data.table (&ge;
1.5.0), foreach (&ge; 1.5.2), iterators (&ge; 1.0.14), survival (&ge;
3.3.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>doParallel (&ge; 1.0.17), knitr, rmarkdown, testthat (&ge;
3.1.0), qs (&ge; 0.25.2)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-26 13:56:04 UTC; jakehughey</td>
</tr>
<tr>
<td>Author:</td>
<td>Jake Hughey [aut, cre],
  Layla Aref [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jake Hughey &lt;jakejhughey@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-26 14:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='demoSample'>Sample table of demographic information</h2><span id='topic+demoSample'></span>

<h3>Description</h3>

<p>The data are artificial and do not correspond to real patients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demoSample
</code></pre>


<h3>Format</h3>

<p>A data table with the following columns:
</p>

<ul>
<li> <p><code>person_id</code>: Character vector of the identifier for each person in the
cohort
</p>
</li>
<li> <p><code>sex</code>: Character vector indicating biological sex
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+getWeights">getWeights()</a></code>, <code><a href="#topic+getScores">getScores()</a></code>
</p>

<hr>
<h2 id='diseaseDxIcdMap'>Mapping of diseases and diagnostic ICD codes</h2><span id='topic+diseaseDxIcdMap'></span>

<h3>Description</h3>

<p>This table provides a mapping between 27 Mendelian diseases and the
corresponding ICD-9 and ICD-10 codes that indicate a genetic diagnosis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diseaseDxIcdMap
</code></pre>


<h3>Format</h3>

<p>A data.table with the following columns:
</p>

<ul>
<li> <p><code>disease_id</code>: Numeric vector of OMIM disease identifiers
</p>
</li>
<li> <p><code>disease_name</code>: Character vector of disease names
</p>
</li>
<li> <p><code>icd</code>: Character vector of ICD codes indicating a genetic diagnosis
</p>
</li>
<li> <p><code>flag</code>: Integer vector of the vocabulary of the ICD code
(<strong>9</strong>: ICD-9-CM,  <strong>10</strong>: ICD-10-CM)
</p>
</li>
<li> <p><code>icd_name</code>: Character vector containing the description of each ICD code
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+getPhecodeOccurrences">getPhecodeOccurrences()</a></code>, <code><a href="#topic+getDxStatus">getDxStatus()</a></code>
</p>

<hr>
<h2 id='diseaseHpoMap'>Mapping of Mendelian diseases and their clinical features</h2><span id='topic+diseaseHpoMap'></span>

<h3>Description</h3>

<p>This table provides a mapping between Mendelian diseases and their clinical
features, represented as Human Phenotype Ontology (HPO) terms. The mapping is
based on annotations from Online Mendelian Inheritance in Man (OMIM).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diseaseHpoMap
</code></pre>


<h3>Format</h3>

<p>A data.table with the following columns:
</p>

<ul>
<li> <p><code>disease_id</code>: Numeric vector of OMIM disease identifiers
</p>
</li>
<li> <p><code>disease_name</code>: Character vector of disease names
</p>
</li>
<li> <p><code>hpo_term_id</code>: Character vector of HPO identifiers corresponding to each
disease's clinical features
</p>
</li>
<li> <p><code>hpo_term_name</code>: Character vector of HPO term descriptions
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://hpo.jax.org/app/download/annotation">https://hpo.jax.org/app/download/annotation</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mapDiseaseToPhecode">mapDiseaseToPhecode()</a></code>
</p>

<hr>
<h2 id='getDxStatus'>Identify cases and controls for Mendelian diseases</h2><span id='topic+getDxStatus'></span>

<h3>Description</h3>

<p>This function is useful for verifying that raw or residual phenotype risk
scores of diagnosed individuals (cases) tend to be higher than scores of
undiagnosed individuals (controls).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDxStatus(
  demos,
  icdOccurrences,
  minUniqueAges = 2L,
  diseaseDxIcdMap = phers::diseaseDxIcdMap
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getDxStatus_+3A_demos">demos</code></td>
<td>
<p>A data.table having one row per person in the cohort. Must have
a column <code>person_id</code>.</p>
</td></tr>
<tr><td><code id="getDxStatus_+3A_icdoccurrences">icdOccurrences</code></td>
<td>
<p>A data.table of occurrences of ICD codes for each
person in the cohort. Must have columns <code>person_id</code>, <code>icd</code>, <code>flag</code>, and
<code>occurrence_age</code>.</p>
</td></tr>
<tr><td><code id="getDxStatus_+3A_minuniqueages">minUniqueAges</code></td>
<td>
<p>Integer indicating the minimum number of unique
ICD code entry ages required to classify a person as a case. Persons with
at least one, but fewer than <code>minUniqueAges</code> entry ages, are assigned as
neither cases nor controls.</p>
</td></tr>
<tr><td><code id="getDxStatus_+3A_diseasedxicdmap">diseaseDxIcdMap</code></td>
<td>
<p>A data.table of the mapping between diseases and
the corresponding ICD codes that indicate a diagnosis. Must have columns
<code>disease_id</code>, <code>icd</code>, and <code>flag</code>. Default is <a href="#topic+diseaseDxIcdMap">diseaseDxIcdMap</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table with columns <code>person_id</code>, <code>disease_id</code>, and <code>dx_status</code>
(1 indicates a case, 0 indicates a control, -1 indicates neither).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')


icdSample1 = merge(icdSample, demoSample[, .(person_id, dob)], by = 'person_id')
icdSample1[, occurrence_age := as.numeric((entry_date - dob)/365.25)]
icdSample1[, `:=`(entry_date = NULL, dob = NULL)]

dxStatus = getDxStatus(demoSample, icdSample1)
</code></pre>

<hr>
<h2 id='getGeneticAssociations'>Perform association tests between phenotype risk scores and genotypes</h2><span id='topic+getGeneticAssociations'></span>

<h3>Description</h3>

<p>The association test for each disease-variant pair is based on a linear
model, with the phenotype risk score as the dependent variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getGeneticAssociations(
  scores,
  genotypes,
  demos,
  diseaseVariantMap,
  lmFormula,
  modelType = c("genotypic", "additive", "dominant", "recessive"),
  level = 0.95,
  dopar = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getGeneticAssociations_+3A_scores">scores</code></td>
<td>
<p>A data.table of phenotype risk scores. Must have columns
<code>person_id</code>, <code>disease_id</code>, <code>score</code>.</p>
</td></tr>
<tr><td><code id="getGeneticAssociations_+3A_genotypes">genotypes</code></td>
<td>
<p>A matrix or 'BEDMatrix' object containing genetic data, with
rownames corresponding to <code>person_id</code>s in <code>demos</code> and <code>scores</code>, and
colnames corresponding to <code>variant_id</code>s in <code>diseaseVariantMap</code>.</p>
</td></tr>
<tr><td><code id="getGeneticAssociations_+3A_demos">demos</code></td>
<td>
<p>A data.table of characteristics for each person in the cohort.
Must have column <code>person_id</code>.</p>
</td></tr>
<tr><td><code id="getGeneticAssociations_+3A_diseasevariantmap">diseaseVariantMap</code></td>
<td>
<p>A data.table indicating which genetic variants to
test for association with phenotype risk scores for which diseases. Must
have columns <code>disease_id</code> and <code>variant_id</code>.</p>
</td></tr>
<tr><td><code id="getGeneticAssociations_+3A_lmformula">lmFormula</code></td>
<td>
<p>A formula representing the linear model (excluding the term
for genotype) to use for the association tests. All terms in the formula
must correspond to columns in <code>demos</code>.</p>
</td></tr>
<tr><td><code id="getGeneticAssociations_+3A_modeltype">modelType</code></td>
<td>
<p>A string indicating how to encode genotype in the model.</p>
</td></tr>
<tr><td><code id="getGeneticAssociations_+3A_level">level</code></td>
<td>
<p>A number indicating the level of the confidence interval.
Default is 0.95.</p>
</td></tr>
<tr><td><code id="getGeneticAssociations_+3A_dopar">dopar</code></td>
<td>
<p>Logical indicating whether to run calculations in parallel if
a parallel backend is already set up, e.g., using
<code><a href="doParallel.html#topic+registerDoParallel">doParallel::registerDoParallel()</a></code>. Recommended to minimize runtime.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table of statistics for the association tests (if a model
fails to converge, NAs will be reported):
</p>

<ul>
<li> <p><code>disease_id</code>: Disease identifier
</p>
</li>
<li> <p><code>variant_id</code>: Variant identifier
</p>
</li>
<li> <p><code>n_total</code>: Number of persons with non-missing genotype data for the
given variant.
</p>
</li>
<li> <p><code>n_wt</code>: Number of persons homozygous for the wild-type allele.
</p>
</li>
<li> <p><code>n_het</code>: Number of persons having one copy of the alternate allele.
</p>
</li>
<li> <p><code>n_hom</code>: Number of persons homozygous for the alternate allele.
</p>
</li>
<li> <p><code>beta</code>: Coefficient for the association of genotype with score
</p>
</li>
<li> <p><code>se</code>: Standard error for <code>beta</code>
</p>
</li>
<li> <p><code>pval</code>: P-value for <code>beta</code> being non-zero
</p>
</li>
<li> <p><code>ci_lower</code>: Lower bound of the confidence interval for <code>beta</code>
</p>
</li>
<li> <p><code>ci_upper</code>: Upper bound of the confidence interval for <code>beta</code>
</p>
</li></ul>

<p>If <code>modelType</code> is &quot;genotypic&quot;, the data.table will include separate
statistics for heterozygous and homozygous genotypes.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+lm">stats::lm()</a></code>, <code><a href="stats.html#topic+confint">stats::confint()</a></code>, <code><a href="#topic+getScores">getScores()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')
library('BEDMatrix')

# map ICD codes to phecodes
phecodeOccurrences = getPhecodeOccurrences(icdSample)

# calculate weights
weights = getWeights(demoSample, phecodeOccurrences)

# OMIM disease IDs for which to calculate phenotype risk scores
diseaseId = 154700

# map diseases to phecodes
diseasePhecodeMap = mapDiseaseToPhecode()

# calculate scores
scores = getScores(weights, diseasePhecodeMap[disease_id == diseaseId])

# map diseases to genetic variants
nvar = 10
diseaseVariantMap = data.table(disease_id = diseaseId, variant_id = paste0('snp', 1:nvar))

# load sample genetic data
npop = 50
genoSample = BEDMatrix(system.file('extdata', 'geno_sample.bed', package = 'phers'))
colnames(genoSample) = paste0('snp', 1:nvar)
rownames(genoSample) = 1:npop

# run genetic association tests
genoStats = getGeneticAssociations(
  scores, genoSample, demoSample, diseaseVariantMap, lmFormula = ~ sex,
  modelType = 'additive')
</code></pre>

<hr>
<h2 id='getPhecodeOccurrences'>Map ICD code occurrences to phecode occurrences</h2><span id='topic+getPhecodeOccurrences'></span>

<h3>Description</h3>

<p>This is typically the first step of an analysis using phenotype risk scores,
the next is <code><a href="#topic+getWeights">getWeights()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPhecodeOccurrences(
  icdOccurrences,
  icdPhecodeMap = phers::icdPhecodeMap,
  dxIcd = phers::diseaseDxIcdMap
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getPhecodeOccurrences_+3A_icdoccurrences">icdOccurrences</code></td>
<td>
<p>A data.table of occurrences of ICD codes for each
person in the cohort. Must have columns <code>person_id</code>, <code>icd</code>, and <code>flag</code>.</p>
</td></tr>
<tr><td><code id="getPhecodeOccurrences_+3A_icdphecodemap">icdPhecodeMap</code></td>
<td>
<p>A data.table of the mapping between ICD codes and
phecodes. Must have columns <code>icd</code>, <code>phecode</code>, and <code>flag</code>. Default is the
map included in this package.</p>
</td></tr>
<tr><td><code id="getPhecodeOccurrences_+3A_dxicd">dxIcd</code></td>
<td>
<p>A data.table of ICD codes to exclude from mapping to phecodes.
Must have columns <code>icd</code> and <code>flag</code>. Default is the table of Mendelian
diseases and the corresponding ICD codes that indicate a genetic diagnosis.
If <code>NULL</code>, no ICD codes will be excluded.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table of phecode occurrences for each person.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getWeights">getWeights()</a></code>, <code><a href="#topic+getScores">getScores()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

# map ICD codes to phecodes
phecodeOccurrences = getPhecodeOccurrences(icdSample)

# calculate weights (using the prevalence method)
weights = getWeights(demoSample, phecodeOccurrences)

# OMIM disease IDs for which to calculate phenotype risk scores
diseaseId = 154700

# map diseases to phecodes
diseasePhecodeMap = mapDiseaseToPhecode()

# calculate scores
scores = getScores(weights, diseasePhecodeMap[disease_id == diseaseId])

# calculate residual scores
rscores = getResidualScores(demoSample, scores, lmFormula = ~ sex)
</code></pre>

<hr>
<h2 id='getResidualScores'>Calculate residual phenotype risk scores</h2><span id='topic+getResidualScores'></span>

<h3>Description</h3>

<p>The residual score indicates to what extent a person's phenotype risk score
for a given disease deviates from the expected score, after adjusting for
the person's characteristics in a linear model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getResidualScores(demos, scores, lmFormula)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getResidualScores_+3A_demos">demos</code></td>
<td>
<p>A data.table of characteristics for each person in the cohort.
Must have column <code>person_id</code>.</p>
</td></tr>
<tr><td><code id="getResidualScores_+3A_scores">scores</code></td>
<td>
<p>A data.table containing the phenotype risk score for each
person for each disease. Must have columns <code>person_id</code>, <code>disease_id</code>, and
<code>score</code>.</p>
</td></tr>
<tr><td><code id="getResidualScores_+3A_lmformula">lmFormula</code></td>
<td>
<p>A formula representing the linear model to use for
calculating residual scores. All terms in the formula must correspond to
columns in <code>demos</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table, based on <code>scores</code>, with an additional column
<code>resid_score</code>. Residual scores for each disease are standardized to have
unit variance.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+influence.measures">stats::rstandard()</a></code>, <code><a href="#topic+getScores">getScores()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

# map ICD codes to phecodes
phecodeOccurrences = getPhecodeOccurrences(icdSample)

# calculate weights (using the prevalence method)
weights = getWeights(demoSample, phecodeOccurrences)

# OMIM disease IDs for which to calculate phenotype risk scores
diseaseId = 154700

# map diseases to phecodes
diseasePhecodeMap = mapDiseaseToPhecode()

# calculate scores
scores = getScores(weights, diseasePhecodeMap[disease_id == diseaseId])

# calculate residual scores
rscores = getResidualScores(demoSample, scores, lmFormula = ~ sex)
</code></pre>

<hr>
<h2 id='getScores'>Calculate phenotype risk scores</h2><span id='topic+getScores'></span>

<h3>Description</h3>

<p>A person's phenotype risk score for a given disease corresponds to the
sum of the weights of the disease-relevant phecodes that the person has
received.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getScores(weights, diseasePhecodeMap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getScores_+3A_weights">weights</code></td>
<td>
<p>A data.table of phecodes and their corresponding weights.
Must have columns <code>person_id</code>, <code>phecode</code> and <code>w</code>.</p>
</td></tr>
<tr><td><code id="getScores_+3A_diseasephecodemap">diseasePhecodeMap</code></td>
<td>
<p>A data.table of the mapping between diseases and
phecodes. Must have columns <code>disease_id</code> and <code>phecode</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table containing the phenotype risk score for each person for
each disease.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mapDiseaseToPhecode">mapDiseaseToPhecode()</a></code>, <code><a href="#topic+getPhecodeOccurrences">getPhecodeOccurrences()</a></code>, <code><a href="#topic+getWeights">getWeights()</a></code>,
<code><a href="#topic+getResidualScores">getResidualScores()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

# map ICD codes to phecodes
phecodeOccurrences = getPhecodeOccurrences(icdSample)

# calculate weights (using the prevalence method)
weights = getWeights(demoSample, phecodeOccurrences)

# OMIM disease IDs for which to calculate phenotype risk scores
diseaseId = 154700

# map diseases to phecodes
diseasePhecodeMap = mapDiseaseToPhecode()

# calculate scores
scores = getScores(weights, diseasePhecodeMap[disease_id == diseaseId])

# calculate residual scores
rscores = getResidualScores(demoSample, scores, lmFormula = ~ sex)
</code></pre>

<hr>
<h2 id='getWeights'>Calculate phecode-specific weights for phenotype risk scores</h2><span id='topic+getWeights'></span>

<h3>Description</h3>

<p>This is typically the second step of an analysis using phenotype risk scores,
the next is <code><a href="#topic+getScores">getScores()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getWeights(
  demos,
  phecodeOccurrences,
  method = c("prevalence", "logistic", "cox", "loglinear", "prevalence_precalc"),
  methodFormula = NULL,
  negativeWeights = FALSE,
  dopar = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getWeights_+3A_demos">demos</code></td>
<td>
<p>A data.table having one row per person in the cohort. Must have
a column <code>person_id</code>. When the <code>cox</code> method is used, <code>demos</code>
must have columns <code>first_age</code> and <code>last_age</code> corresponding to first and
last age of visit (in years).</p>
</td></tr>
<tr><td><code id="getWeights_+3A_phecodeoccurrences">phecodeOccurrences</code></td>
<td>
<p>A data.table of phecode occurrences for each person
in the cohort. Must have columns <code>person_id</code> and <code>phecode</code> under the
&quot;prevalence&quot; or &quot;logistic&quot; methods, columns <code>person_id</code>, <code>phecode</code>, and
<code>num_occurrences</code> under the &quot;loglinear&quot; method, and columns <code>person_id</code>,
<code>phecode</code>, and <code>occurrence_age</code> under the &quot;cox&quot; method. <code>num_occurrences</code>
refers to the number of unique dates a phecode was recorded for a person.
<code>occurrence_age</code> refers to the first age (in years) a person acquired a
phecode.</p>
</td></tr>
<tr><td><code id="getWeights_+3A_method">method</code></td>
<td>
<p>A string indicating the statistical model for calculating
weights.</p>
</td></tr>
<tr><td><code id="getWeights_+3A_methodformula">methodFormula</code></td>
<td>
<p>A formula representing the right-hand side of the model
corresponding to <code>method</code>. All terms in the formula must correspond to
columns in <code>demos</code>. A method formula is not required for the &quot;prevalence&quot;
and &quot;prevalence_precalc&quot; methods. Do not use age-related covariates with
the &quot;cox&quot; method.</p>
</td></tr>
<tr><td><code id="getWeights_+3A_negativeweights">negativeWeights</code></td>
<td>
<p>Logical indicating whether to allow negative weights
for individuals with no occurrences of a phecode. This option is not
required for the &quot;loglinear&quot; method since under this method, individuals
with a nonzero phecode occurrence can also have negative weights.</p>
</td></tr>
<tr><td><code id="getWeights_+3A_dopar">dopar</code></td>
<td>
<p>Logical indicating whether to run calculations in parallel if
a parallel backend is already set up, e.g., using
<code><a href="doParallel.html#topic+registerDoParallel">doParallel::registerDoParallel()</a></code>. Recommended to minimize runtime.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table with columns <code>person_id</code>, <code>phecode</code>, <code>pred</code>, and <code>w</code>.
The column <code>pred</code> represents a different quantity depending on <code>method</code>.
Under the &quot;prevalence&quot; <code>method</code>, it is fraction of the cohort that has
at least one occurrence of the given phecode. The &quot;prevalence_precalc&quot;
<code>method</code> is similar to the &quot;prevalence&quot; <code>method</code> but <code>pred</code> is calculated
based on EHR data from the Vanderbilt University Medical Center.
Under &quot;logistic&quot; or &quot;cox&quot; <code>method</code>, it is the predicted probability of
given individual having a given phecode based on <code>methodFormula</code>.
Under the &quot;loglinear&quot; <code>method</code>, it is the predicted
<code>log2(num_occurrences + 1)</code> of a given phecode for a given individual
based on <code>methodFormula</code>. For the &quot;prevalence&quot;, &quot;prevalence_precalc&quot;,
&quot;cox&quot;, and &quot;logistic&quot; <code>method</code>s, weight is calculated as <code>-log10(pred)</code>
when an individual has non-zero phecode occurrence and <code>log10(1 - pred)</code>
when an individual has zero phecode occurrence. For the &quot;loglinear&quot; <code>method</code>
weight is calculated as the difference between the observed
<code>log2(num_occurrences + 1)</code> and <code>pred</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getPhecodeOccurrences">getPhecodeOccurrences()</a></code>, <code><a href="#topic+getScores">getScores()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')
library('survival')

# map ICD codes to phecodes
phecodeOccurrences = getPhecodeOccurrences(icdSample)

# calculate weights using the prevalence method
weightsPrev = getWeights(demoSample, phecodeOccurrences)

# calculate weights using the prevalence method
# (assign negative weights to those with zero phecode occurrence)
weightsPrevNeg = getWeights(
  demoSample, phecodeOccurrences, negativeWeights = TRUE)

# calculate weights using the logistic method
weightsLogistic = getWeights(
  demoSample, phecodeOccurrences, method = 'logistic', methodFormula = ~ sex)

# calculate weights using the loglinear method
phecodeOccurrences2 = phecodeOccurrences[, .(
  num_occurrences = uniqueN(entry_date)), by = .(person_id, phecode)]
weightsLoglinear = getWeights(
  demoSample, phecodeOccurrences2, method = 'loglinear', methodFormula = ~ sex)

# calculate weights using the cox method
phecodeOccurrences3 = phecodeOccurrences[, .(
  first_occurrence_date = min(entry_date)) , by = .(person_id, phecode)]
phecodeOccurrences3 = merge(
  phecodeOccurrences3, demoSample[, .(person_id, dob)], by = 'person_id')
phecodeOccurrences3[,
  occurrence_age := as.numeric((first_occurrence_date - dob)/365.25)]
phecodeOccurrences3[, `:=`(first_occurrence_date = NULL, dob = NULL)]
demoSample3 = demoSample[, .(
  person_id, sex,
  first_age = as.numeric((first_visit_date - dob)/365.25),
  last_age = as.numeric((last_visit_date - dob)/365.25))]
weightsCox = getWeights(
  demoSample3, phecodeOccurrences3, method = 'cox', methodFormula = ~ sex)

# calculate weights using pre-calculated weights based on data from
# Vanderbilt University Medical Center
weightsPreCalc = getWeights(
  demoSample, phecodeOccurrences, method = 'prevalence_precalc')
</code></pre>

<hr>
<h2 id='hpoPhecodeMap'>Mapping of HPO terms and phecodes</h2><span id='topic+hpoPhecodeMap'></span>

<h3>Description</h3>

<p>This table provides a mapping between Human Phenotype Ontology (HPO)
terms and phecodes, and is useful for using phecodes to represent the
clinical features of Mendelian diseases (version 1.2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hpoPhecodeMap
</code></pre>


<h3>Format</h3>

<p>A data.table with the following columns:
</p>

<ul>
<li> <p><code>hpo_term_id</code>: Character vector of HPO term identifiers
</p>
</li>
<li> <p><code>hpo_term_name</code>: Character vector of HPO term descriptions
</p>
</li>
<li> <p><code>phecode</code>: Character vector of phecodes
</p>
</li>
<li> <p><code>phecode_name</code>: Character vector of phecode descriptions
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+mapDiseaseToPhecode">mapDiseaseToPhecode()</a></code>
</p>

<hr>
<h2 id='icdPhecodeMap'>Mapping of ICD codes and phecodes</h2><span id='topic+icdPhecodeMap'></span>

<h3>Description</h3>

<p>This table provides a mapping between International Classification of
Diseases 9th and 10th revisions (ICD-9-CM and ICD-10-CM) and phecodes
(version 1.2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>icdPhecodeMap
</code></pre>


<h3>Format</h3>

<p>A data.table with the following columns:
</p>

<ul>
<li> <p><code>icd</code>: Character vector of ICD codes
</p>
</li>
<li> <p><code>flag</code>: Integer vector of the vocabulary of the ICD code
(<strong>9</strong>: ICD-9-CM,  <strong>10</strong>: ICD-10-CM)
</p>
</li>
<li> <p><code>icd_name</code>: Character vector of ICD code descriptions
</p>
</li>
<li> <p><code>phecode</code>: Character vector of phecodes
</p>
</li>
<li> <p><code>phecode_name</code>: Character vector of phecode descriptions
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://phewascatalog.org/phecodes">https://phewascatalog.org/phecodes</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getPhecodeOccurrences">getPhecodeOccurrences()</a></code>
</p>

<hr>
<h2 id='icdSample'>Sample table of ICD occurrences</h2><span id='topic+icdSample'></span>

<h3>Description</h3>

<p>The data are artificial and do not correspond to real patients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>icdSample
</code></pre>


<h3>Format</h3>

<p>A data.table with the following columns:
</p>

<ul>
<li> <p><code>person_id</code>: Character vector of the identifier for each person
</p>
</li>
<li> <p><code>icd</code>: Character vector of the ICD codes recorded for each person
</p>
</li>
<li> <p><code>flag</code>: Integer vector of the vocabulary of the ICD code
(<strong>9</strong>: ICD-9-CM, <strong>10</strong>: ICD-10-CM)
</p>
</li>
<li> <p><code>entry_date</code>: Vector of type <code>Date</code> indicating the date each ICD code was
recorded.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+getPhecodeOccurrences">getPhecodeOccurrences()</a></code>, <code><a href="#topic+getWeights">getWeights()</a></code>, <code><a href="#topic+getScores">getScores()</a></code>
</p>

<hr>
<h2 id='mapDiseaseToPhecode'>Map diseases to phecodes via HPO terms</h2><span id='topic+mapDiseaseToPhecode'></span>

<h3>Description</h3>

<p>A mapping of diseases to their clinical features, represented as phecodes,
is required for calculating phenotype risk scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapDiseaseToPhecode(
  diseaseHpoMap = phers::diseaseHpoMap,
  hpoPhecodeMap = phers::hpoPhecodeMap
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mapDiseaseToPhecode_+3A_diseasehpomap">diseaseHpoMap</code></td>
<td>
<p>A data.table containing the mapping between diseases and
HPO terms. Must have columns <code>disease_id</code> and <code>term_id</code>. Default is the map
included in this package.</p>
</td></tr>
<tr><td><code id="mapDiseaseToPhecode_+3A_hpophecodemap">hpoPhecodeMap</code></td>
<td>
<p>A data.table containing the mapping between HPO terms
and phecodes. Must have columns <code>term_id</code> and <code>phecode</code>. Default is the map
included in this package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.table with columns <code>disease_id</code> and <code>phecode</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getScores">getScores()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')
library('survival')

# map ICD codes to phecodes
phecodeOccurrences = getPhecodeOccurrences(icdSample)

# calculate weights using the prevalence method
weightsPrev = getWeights(demoSample, phecodeOccurrences)

# calculate weights using the prevalence method
# (assign negative weights to those with zero phecode occurrence)
weightsPrevNeg = getWeights(
  demoSample, phecodeOccurrences, negativeWeights = TRUE)

# calculate weights using the logistic method
weightsLogistic = getWeights(
  demoSample, phecodeOccurrences, method = 'logistic', methodFormula = ~ sex)

# calculate weights using the loglinear method
phecodeOccurrences2 = phecodeOccurrences[, .(
  num_occurrences = uniqueN(entry_date)), by = .(person_id, phecode)]
weightsLoglinear = getWeights(
  demoSample, phecodeOccurrences2, method = 'loglinear', methodFormula = ~ sex)

# calculate weights using the cox method
phecodeOccurrences3 = phecodeOccurrences[, .(
  first_occurrence_date = min(entry_date)) , by = .(person_id, phecode)]
phecodeOccurrences3 = merge(
  phecodeOccurrences3, demoSample[, .(person_id, dob)], by = 'person_id')
phecodeOccurrences3[,
  occurrence_age := as.numeric((first_occurrence_date - dob)/365.25)]
phecodeOccurrences3[, `:=`(first_occurrence_date = NULL, dob = NULL)]
demoSample3 = demoSample[, .(
  person_id, sex,
  first_age = as.numeric((first_visit_date - dob)/365.25),
  last_age = as.numeric((last_visit_date - dob)/365.25))]
weightsCox = getWeights(
  demoSample3, phecodeOccurrences3, method = 'cox', methodFormula = ~ sex)

# calculate weights using pre-calculated weights based on data from
# Vanderbilt University Medical Center
weightsPreCalc = getWeights(
  demoSample, phecodeOccurrences, method = 'prevalence_precalc')
</code></pre>

<hr>
<h2 id='preCalcWeights'>Pre-calculated weights for calculating phenotype risk scores</h2><span id='topic+preCalcWeights'></span>

<h3>Description</h3>

<p>The weights are based on EHR data from the Vanderbilt University Medical
Center Synthetic Derivative (SD) and ICD-phecode map version 1.2 and are
calculated using the &quot;prevalence&quot; method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preCalcWeights
</code></pre>


<h3>Format</h3>

<p>A data.table with the following columns:
</p>

<ul>
<li> <p><code>phecode</code>: Character vector of phecodes
</p>
</li>
<li> <p><code>prev</code>: Numeric vector of prevalences, i.e., fraction of subjects in the SD
that have at least one occurrence of the given phecode
</p>
</li>
<li> <p><code>w</code>: Numeric vector of weights, calculated as <code>-log10(prev)</code>
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+getWeights">getWeights()</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
