<!DOCTYPE html><html><head><title>Help for package phoenics</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {phoenics}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#from_ASICS_to_PHOENICS'><p>Prepare quantification data from ASICS outputs</p></a></li>
<li><a href='#MTBLS422'><p>Dataset &quot;MTBLS422&quot;</p></a></li>
<li><a href='#overlap_coefficient'><p>Calculate overlap coefficient between pathways</p></a></li>
<li><a href='#pathway_search'><p>Query KEGG pathways for a given set of metabolites</p></a></li>
<li><a href='#pathwayRes'><p>Class pathwayRes</p></a></li>
<li><a href='#test_pathway'><p>Pathway differential analysis based on longitudinal metabolomics data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Pathways Longitudinal and Differential Analysis in Metabolomics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-06-26</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>lme4, blme, tibble, FactoMineR, factoextra, tidyr, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>KEGGREST</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Camille Guilmineau &lt;camille.guilmineau@inrae.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform a differential analysis at pathway level based on metabolite quantifications and information on pathway metabolite composition.
  The method is based on a Principal Component Analysis step and on a linear mixed model.
  Automatic query of metabolic pathways is also implemented. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://forgemia.inra.fr/panoramics/phoenics/-/issues">https://forgemia.inra.fr/panoramics/phoenics/-/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://forgemia.inra.fr/panoramics/phoenics">https://forgemia.inra.fr/panoramics/phoenics</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-27 07:09:04 UTC; cguilmineau</td>
</tr>
<tr>
<td>Author:</td>
<td>Camille Guilmineau [aut, cre],
  Remi Servien <a href="https://orcid.org/0000-0003-1270-1843"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Nathalie Vialaneix
    <a href="https://orcid.org/0000-0003-1156-0639"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-27 08:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='from_ASICS_to_PHOENICS'>Prepare quantification data from ASICS outputs</h2><span id='topic+from_ASICS_to_PHOENICS'></span>

<h3>Description</h3>

<p>Prepare quantification data from <code>ASICS</code> outputs for
<code>test_pathway</code>. In short, it replaces metabolite names by metabolites
KEGG codes and transposes the matrix to have samples in rows and metabolites
in columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>from_ASICS_to_PHOENICS(quantif)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="from_ASICS_to_PHOENICS_+3A_quantif">quantif</code></td>
<td>
<p>output matrix of <code>ASICS</code> quantification</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of quantification with samples in rows and metabolites in
columns, properly formatted for <code>test_pathway</code>
</p>


<h3>Author(s)</h3>

<p>Camille Guilmineau &lt;camille.guilmineau@inrae.fr&gt;<br />
Remi Servien &lt;remi.servien@inrae.fr&gt;<br />
Nathalie Vialaneix &lt;nathalie.vialaneix@inrae.fr&gt;
</p>


<h3>References</h3>

<p>Lefort G., Liaubet L., Canlet C., Tardivel P., P\'ere M.C., Quesnel H., 
Paris A., Iannuccelli N., Vialaneix N. Servien R. (2019). ASICS: an R
package for a whole analysis workflow of 1D 1H NMR spectra.
<em>Bioinformatics</em>, <strong>35</strong>(21): 4356&ndash;4363.
<a href="https://doi.org/10.1093/bioinformatics/btz248">doi:10.1093/bioinformatics/btz248</a>
</p>
<p>Tardivel P., Canlet C., Lefort G., Tremblay-Franco M., Debrauwer L.,
Concordet D., Servien R. (2017). ASICS: an automatic method for
identification and quantification of metabolites in complex 1D 1H NMR
spectra. <em>Metabolomics</em>, <strong>13</strong>(10): 109.
<a href="https://doi.org/10.1007/s11306-017-1244-5">doi:10.1007/s11306-017-1244-5</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("MTBLS422")
quantif &lt;- from_ASICS_to_PHOENICS(quantif)

</code></pre>

<hr>
<h2 id='MTBLS422'>Dataset &quot;MTBLS422&quot;</h2><span id='topic+MTBLS422'></span><span id='topic+quantif'></span><span id='topic+pathways'></span><span id='topic+design'></span>

<h3>Description</h3>

<p>Metabolites quantifications, associated metabolic pathways and
experimental design to characterize the effects of two clinically important
antibiotic treatments, ciprofloxacin and vancomycin-imipenem on mice.
</p>


<h3>Format</h3>

<p>3 datasets are provided: </p>

<ul>
<li> <p><code>quantif</code>: a data frame with 10 rows (metabolites name) and 11
columns (samples id), which corresponds to the metabolites quantifications
in the samples.
</p>
</li>
<li> <p><code>pathways</code>: a data frame with 11 rows and 4 columns, which
contains informations about pathways and their metabolites.
</p>
</li>
<li> <p><code>design</code>: a data frame with 11 rows (samples id) and 4 columns
(id and effects to be added in the model).
</p>
</li></ul>



<h3>Details</h3>

<p>The raw dataset have been made available on MetaboLights (with the
id MTBLS422 <a href="https://www.ebi.ac.uk/metabolights/editor/MTBLS422">https://www.ebi.ac.uk/metabolights/editor/MTBLS422</a>) by
[Choo <em>et al.</em>, 2017]. Metabolites quantifications were obtained
based on the raw signal using <code>ASICS</code> package. Pathways were obtained
using <code>KEGGREST</code> package. The datasets provided for the example
are a subset of the original dataset.
</p>


<h3>References</h3>

<p>Choo J. M., Kanno T., Zain N. M. M., Leong L. E. X.,
Abell G. C. J., Keeble J. E., Bruce K. D., Mason A. J., Rogers G. B. (2017).
Divergent relationships between fecal microbiota and metabolome following
distinct antibiotic-induced disruptions. <em>mSphere</em>, <strong>2</strong>(1)
<a href="https://doi.org/10.1128/msphere.00005-17">doi:10.1128/msphere.00005-17</a>
</p>
<p>Lefort G., Liaubet L., Canlet C., Tardivel P., P\'ere M.C., Quesnel H.,
Paris A., Iannuccelli N., Vialaneix N. Servien R. (2019). ASICS: an R
package for a whole analysis workflow of 1D 1H NMR spectra.
<em>Bioinformatics</em>, <strong>35</strong>(21): 4356&ndash;4363.
<a href="https://doi.org/10.1093/bioinformatics/btz248">doi:10.1093/bioinformatics/btz248</a>
</p>
<p>Tardivel P., Canlet C., Lefort G., Tremblay-Franco M., Debrauwer L.,
Concordet D., Servien R. (2017). ASICS: an automatic method for
identification and quantification of metabolites in complex 1D 1H NMR
spectra. <em>Metabolomics</em>, <strong>13</strong>(10): 109.
<a href="https://doi.org/10.1007/s11306-017-1244-5">doi:10.1007/s11306-017-1244-5</a>
</p>
<p>Tenenbaum D., Maintainer B. (2022). KEGGREST: Client-side REST access to the
Kyoto Encyclopedia of Genes and Genomes (KEGG). R package version 1.38.0.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MTBLS422)

design[1:5, ]
pathways[1:5, ]
quantif[1:5, 1:5]


</code></pre>

<hr>
<h2 id='overlap_coefficient'>Calculate overlap coefficient between pathways</h2><span id='topic+overlap_coefficient'></span>

<h3>Description</h3>

<p>Calculate overlap coefficient between pathways
</p>


<h3>Usage</h3>

<pre><code class='language-R'>overlap_coefficient(pathwayA, pathwayB, pathways = NULL, organism = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="overlap_coefficient_+3A_pathwaya">pathwayA</code></td>
<td>
<p>a character string of pathway name or pathway code or a
character vector of metabolite names or metabolite codes</p>
</td></tr>
<tr><td><code id="overlap_coefficient_+3A_pathwayb">pathwayB</code></td>
<td>
<p>a character string of pathway name or pathway code or a
character vector of metabolite names or metabolite codes</p>
</td></tr>
<tr><td><code id="overlap_coefficient_+3A_pathways">pathways</code></td>
<td>
<p>data.frame or matrix with metabolites in rows and the
following information in columns: </p>

<ul>
<li> <p><code>metabolite_code</code> metabolite code
</p>
</li>
<li> <p><code>metabolite_name</code> metabolite name
</p>
</li>
<li> <p><code>pathway_code</code> pathway code (identifier)
</p>
</li>
<li> <p><code>pathway_name</code> name of the pathway
</p>
</li></ul>

<p>Used if <code>pathwayA</code> and <code>pathwayB</code> are pathway names or pathway
codes.</p>
</td></tr>
<tr><td><code id="overlap_coefficient_+3A_organism">organism</code></td>
<td>
<p>organism code in KEGG database. Required if
<code>pathways = NULL</code> and <code>pathwayA</code> and <code>pathwayB</code> are pathway 
names and ignored otherwise.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A value between 0 and 1 calculated with the formula:
</p>
<p style="text-align: center;"><code class="reqn">\text{OC}(A,B) = \frac{\vert A \cap B \vert}{\min(\vert A \vert,
\vert B \vert)}</code>
</p>

<p>An overlap coefficient of 1 means that one pathway is included in the other.
An overlap coefficient of 0 means that there is no overlap between the
pathways.
</p>


<h3>References</h3>

<p>Wieder C., Lai R.P.J., Ebbels T.M.D. (2022). Single sample pathway analysis
in metabolomics: performance evaluation and application.
<em>BMC Bioinformatics</em>, <strong>23</strong>, 481.
<a href="https://doi.org/10.1186/s12859-022-05005-1">doi:10.1186/s12859-022-05005-1</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("MTBLS422")
pathwayA &lt;- "Galactose metabolism"
pathwayB &lt;- "Vitamin digestion and absorption"
overlap_coefficient(pathwayA, pathwayB, pathways)

if (requireNamespace("KEGGREST", quietly = TRUE)) {
  pathwayA &lt;- "Galactose metabolism"
  pathwayB &lt;- "Vitamin digestion and absorption"
  
  overlap_coefficient(pathwayA, pathwayB, organism = "mmu")
  
}

pathwayA &lt;- "mmu00052"
pathwayB &lt;- "mmu00562"
overlap_coefficient(pathwayA, pathwayB, pathways)

pathwayA &lt;- c("C00029", "C00116", "C00137", "C00794", "C00984", "C01697")
pathwayB &lt;- c("C00191", "C00092", "C00137")
overlap_coefficient(pathwayA, pathwayB)

</code></pre>

<hr>
<h2 id='pathway_search'>Query KEGG pathways for a given set of metabolites</h2><span id='topic+pathway_search'></span>

<h3>Description</h3>

<p>Query KEGG pathways for a given set of metabolites
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway_search(metab, organism, min_size = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway_search_+3A_metab">metab</code></td>
<td>
<p>vector of metabolite KEGG codes</p>
</td></tr>
<tr><td><code id="pathway_search_+3A_organism">organism</code></td>
<td>
<p>organism code in KEGG database</p>
</td></tr>
<tr><td><code id="pathway_search_+3A_min_size">min_size</code></td>
<td>
<p>minimal number of metabolites required for a pathway to be
returned</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame with metabolites in rows and the following information
in columns: </p>

<ul>
<li> <p><code>metabolite_code</code> metabolite code
</p>
</li>
<li> <p><code>metabolite_name</code> metabolite name
</p>
</li>
<li> <p><code>pathway_code</code> pathway code (identifier)
</p>
</li>
<li> <p><code>pathway_name</code> name of the pathway
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Camille Guilmineau &lt;camille.guilmineau@inrae.fr&gt;<br />
Remi Servien &lt;remi.servien@inrae.fr&gt;<br />
Nathalie Vialaneix &lt;nathalie.vialaneix@inrae.fr&gt;
</p>


<h3>References</h3>

<p>Kanehisa M., Goto S. (2000). KEGG: Kyoto Encyclopedia of Genes and Genomes,
<em>Nucleic Acids Research</em>, Volume 28, Issue 1, Pages 27&ndash;30,
<a href="https://doi.org/10.1093/nar/28.1.27">doi:10.1093/nar/28.1.27</a>
</p>
<p>Tenenbaum D., Maintainer B. (2022). KEGGREST: Client-side REST access to the
Kyoto Encyclopedia of Genes and Genomes (KEGG). R package version 1.38.0.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("KEGGREST", quietly = TRUE)) {
  data("MTBLS422")
  quantif &lt;- from_ASICS_to_PHOENICS(quantif)
  
  pathways &lt;- pathway_search(metab = colnames(quantif), organism = "mmu")
  
}

</code></pre>

<hr>
<h2 id='pathwayRes'>Class pathwayRes</h2><span id='topic+pathwayRes'></span><span id='topic+summary.pathwayRes'></span><span id='topic+print.pathwayRes'></span><span id='topic+plot.pathwayRes'></span><span id='topic+head'></span><span id='topic+head.pathwayRes'></span><span id='topic+extract'></span><span id='topic+extract.pathwayRes'></span><span id='topic+adjust_pval'></span><span id='topic+adjust_pval.pathwayRes'></span>

<h3>Description</h3>

<p>S3 class for pathway differential analysis results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pathwayRes'
summary(object, ...)

## S3 method for class 'pathwayRes'
print(x, ...)

## S3 method for class 'pathwayRes'
plot(
  x,
  ...,
  pathway_id = NULL,
  plot = c("eig", "var", "ind"),
  habillage = "none"
)

head(object)

extract(object, pathway_id)

adjust_pval(object, method = p.adjust.methods, n = length(object))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathwayRes_+3A_object">object</code>, <code id="pathwayRes_+3A_x">x</code></td>
<td>
<p>object of class <code>pathwayRes</code></p>
</td></tr>
<tr><td><code id="pathwayRes_+3A_...">...</code></td>
<td>
<p>not used</p>
</td></tr>
<tr><td><code id="pathwayRes_+3A_pathway_id">pathway_id</code></td>
<td>
<p>a character string or vector of pathway codes or names</p>
</td></tr>
<tr><td><code id="pathwayRes_+3A_plot">plot</code></td>
<td>
<p>a character string indicating the type of plot to return. Default
to <code>"eig"</code> (the screegraph of the PCA is displayed)</p>
</td></tr>
<tr><td><code id="pathwayRes_+3A_habillage">habillage</code></td>
<td>
<p>a character string indicating the column of the design used 
to color the individuals. Only used when <code>plot = "ind"</code>. Default to 
<code>"none"</code> (no color)</p>
</td></tr>
<tr><td><code id="pathwayRes_+3A_method">method</code></td>
<td>
<p>a character string indicating the correction method to be used
for multiple testing correction (authorized values are those of 
<code><a href="stats.html#topic+p.adjust.methods">p.adjust.methods</a></code>)</p>
</td></tr>
<tr><td><code id="pathwayRes_+3A_n">n</code></td>
<td>
<p>number of comparisons for multiple testing correction</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Methods for the class pathwayRes
</p>


<h3>Value</h3>

<p>The function <code>extract</code> returns an object of class
<code>pathwayRes</code> which is a list of pathway results, containing only the
pathways in <code>pathway_id</code>.
</p>
<p>The function <code>adjust_pval</code> returns a data.frame with pathways
in rows and the following information in columns:
</p>
<table>
<tr><td><code>pathway_name</code></td>
<td>
<p>name of the pathway</p>
</td></tr>
<tr><td><code>pathway_code</code></td>
<td>
<p>pathway code (identifier)</p>
</td></tr>
<tr><td><code>Fixed_effect</code></td>
<td>
<p>tested effect</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>raw p-value of the pathway</p>
</td></tr>
<tr><td><code>adjusted_pval</code></td>
<td>
<p>adjusted p-value of the pathway</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data("MTBLS422")
quantif &lt;- from_ASICS_to_PHOENICS(quantif)
out_test &lt;- test_pathway(quantif, design, pathways, 
                         fixed = c("Age", "Treatment"), random = "Mouse", 
                         npc = 2, model = "blmer")
summary(out_test)
print(out_test)
plot(out_test)
plot(out_test, "mmu00052", plot = "var")
plot(out_test, "mmu00052", plot = "ind", habillage = "Age")
head(out_test)
extract(out_test, "mmu00562")
adj_pval &lt;- adjust_pval(out_test)
</code></pre>

<hr>
<h2 id='test_pathway'>Pathway differential analysis based on longitudinal metabolomics data</h2><span id='topic+test_pathway'></span>

<h3>Description</h3>

<p>Perform a differential analysis at pathway level based on 
metabolite quantifications and information on pathway metabolite composition.
The method relies on a PCA step.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_pathway(
  quantif,
  design,
  pathways = "auto",
  fixed,
  random,
  npc = 1L,
  model = c("lmer", "blmer"),
  organism = NULL,
  min_size = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="test_pathway_+3A_quantif">quantif</code></td>
<td>
<p>data.frame or matrix of the metabolite quantification with 
samples in rows (sample identifiers must be row names) and metabolites in 
columns (metabolite codes must be column names)</p>
</td></tr>
<tr><td><code id="test_pathway_+3A_design">design</code></td>
<td>
<p>data.frame or matrix with samples in rows (sample identifiers
must be row names) and the different effects to be included in the model
in columns. Column names must be provided and are used as arguments for
<code>fixed</code> and <code>random</code></p>
</td></tr>
<tr><td><code id="test_pathway_+3A_pathways">pathways</code></td>
<td>
<p>data.frame or matrix with metabolites in rows (all 
metabolites in columns of <code>quantif</code> must have a row in this input) and the
following information in columns: </p>

<ul>
<li> <p><code>metabolite_code</code> metabolite code
</p>
</li>
<li> <p><code>metabolite_name</code> metabolite name
</p>
</li>
<li> <p><code>pathway_code</code> pathway code (identifier)
</p>
</li>
<li> <p><code>pathway_name</code> name of the pathway
</p>
</li></ul>
</td></tr>
<tr><td><code id="test_pathway_+3A_fixed">fixed</code></td>
<td>
<p>character vector of fixed effects to be included in the model.
They must correspond to column names of <code>design</code></p>
</td></tr>
<tr><td><code id="test_pathway_+3A_random">random</code></td>
<td>
<p>character vector of random effects to be included in the model.
They must correspond to column names of <code>design</code></p>
</td></tr>
<tr><td><code id="test_pathway_+3A_npc">npc</code></td>
<td>
<p>number of PCs for the PCA step</p>
</td></tr>
<tr><td><code id="test_pathway_+3A_model">model</code></td>
<td>
<p>a character string indicating if the model has to be fitted
using <a href="lme4.html#topic+lmer">lmer</a> or <a href="blme.html#topic+blmer">blmer</a>. Default to <code>"lmer"</code></p>
</td></tr>
<tr><td><code id="test_pathway_+3A_organism">organism</code></td>
<td>
<p>organism code in KEGG database. Required if
<code>pathways = "auto"</code> and ignored otherwise</p>
</td></tr>
<tr><td><code id="test_pathway_+3A_min_size">min_size</code></td>
<td>
<p>minimal number of metabolites in a pathway. Required if
<code>pathways = "auto"</code> and ignored otherwise. Default to <code>2</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>If <code>pathways = "auto"</code>, information on pathways in which metabolites are
present is automatically obtained by the function <a href="#topic+pathway_search">pathway_search</a> using
<code>KEGGREST</code> that queries KEGG database. Results are specific to a given
organism (passed in <code>organism</code>). Pathways containing less than
<code>min_size</code> metabolites are filtered out.
</p>


<h3>Value</h3>

<p>an object of class <code>pathwayRes</code> which is a list of pathway 
results. Each element of the list contain the following entries:
</p>
<table>
<tr><td><code>pathway_name</code></td>
<td>
<p>a character corresponding to the pathway name</p>
</td></tr>
<tr><td><code>pathway_code</code></td>
<td>
<p>a character corresponding to the pathway code</p>
</td></tr>
<tr><td><code>metabolites</code></td>
<td>
<p>a data.frame with the names and codes of the quantified
metabolites in the pathway</p>
</td></tr>
<tr><td><code>PCA</code></td>
<td>
<p>the result of the pathway PCA (a <code>PCA</code> object as obtained 
from <a href="FactoMineR.html#topic+PCA">PCA</a>)</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>the output of the mixed model fit</p>
</td></tr>
<tr><td><code>test_pathway</code></td>
<td>
<p>a data.frame with the p-values for each tested fixed
effect</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Camille Guilmineau &lt;camille.guilmineau@inrae.fr&gt;<br />
Remi Servien &lt;remi.servien@inrae.fr&gt;<br />
Nathalie Vialaneix &lt;nathalie.vialaneix@inrae.fr&gt;
</p>


<h3>See Also</h3>

<p><a href="#topic+pathway_search">pathway_search</a> for the automatic annotation of metabolites in KEGG 
pathways.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("MTBLS422")
quantif &lt;- from_ASICS_to_PHOENICS(quantif)
out_test &lt;- test_pathway(quantif, design, pathways, 
                         fixed = c("Age", "Treatment"), random = "Mouse", 
                         npc = 2, model = "blmer")
out_test

if (requireNamespace("KEGGREST", quietly = TRUE)) { 

  out_test2 &lt;- test_pathway(quantif, design, pathways = "auto", 
                            fixed = c("Age", "Treatment"), random = "Mouse", 
                            npc = 2, model = "blmer", organism = "mmu")
  out_test2
 
}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
