<!DOCTYPE html><html><head><title>Help for package phylosamp</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {phylosamp}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#phylosamp-package'><p>phylosamp: Sample Size Calculations for Molecular and Phylogenetic Studies</p></a></li>
<li><a href='#exp_links'><p>Calculate expected number of links in a sample</p></a></li>
<li><a href='#falsediscoveryrate'><p>Calculate false discovery rate of a sample</p></a></li>
<li><a href='#gen_dists'><p>Calculate genetic distance distribution</p></a></li>
<li><a href='#gendist_distribution'><p>Calculate genetic distance distribution</p></a></li>
<li><a href='#gendist_roc_format'><p>Make ROC curve from sensitivity and specificity</p></a></li>
<li><a href='#gendist_sensspec_cutoff'><p>Calculate sensitivity and specificity of a genetic distance cutoff</p></a></li>
<li><a href='#genDistSim'><p>Simulations of the genetic distance distribution</p></a></li>
<li><a href='#get_optim_roc'><p>Find optimal ROC threshold</p></a></li>
<li><a href='#obs_pairs_mtml'><p>Expected number of observed pairs assuming multiple-transmission and multiple-linkage</p></a></li>
<li><a href='#obs_pairs_mtsl'><p>Expected number of observed pairs assuming multiple-transmission and single-linkage</p></a></li>
<li><a href='#obs_pairs_stsl'><p>Expected number of observed pairs assuming single-transmission and single-linkage</p></a></li>
<li><a href='#optim_roc_threshold'><p>Find optimal ROC threshold</p></a></li>
<li><a href='#prob_trans_mtml'><p>Probability of transmission assuming multiple-transmission and multiple-linkage</p></a></li>
<li><a href='#prob_trans_mtsl'><p>Probability of transmission assuming multiple-transmission and single-linkage</p></a></li>
<li><a href='#prob_trans_stsl'><p>Probability of transmission assuming single-transmission and single-linkage</p></a></li>
<li><a href='#relR_power'><p>Calculate power for detecting differential transmission given a sample size</p></a></li>
<li><a href='#relR_power_simulated'><p>Simulate power for detecting differential transmission</p></a></li>
<li><a href='#relR_samplesize'><p>Calculate sample size needed to detect differential transmission</p></a></li>
<li><a href='#relR_samplesize_basic'><p>Calculate simple derived sample size for detecting differential transmission</p></a></li>
<li><a href='#relR_samplesize_ci'><p>Calculate sample size for detecting differential transmission with uncertainty bounds</p></a></li>
<li><a href='#relR_samplesize_linkerr'><p>Calculate sample size for detecting differential transmission correcting for sensitivity and specificity</p></a></li>
<li><a href='#relR_samplesize_opterr'><p>Function to calculate the error in estimated sample size for use in optimize function</p></a></li>
<li><a href='#relR_samplesize_simsolve'><p>Calculate optimized sample size for detecting differential transmission</p></a></li>
<li><a href='#relR_samplesize_solve'><p>Calculate optimal sample size for detecting differential transmission with imperfect specificity</p></a></li>
<li><a href='#samplesize'><p>Calculate sample size</p></a></li>
<li><a href='#sens_spec_calc'><p>Calculate sensitivity and specificity</p></a></li>
<li><a href='#sens_spec_roc'><p>Make ROC from sensitivity and specificity</p></a></li>
<li><a href='#translink_expected_links_obs'><p>Calculate expected number of transmission links in a sample</p></a></li>
<li><a href='#translink_expected_links_obs_mtml'><p>Calculate expected number of observed pairs assuming multiple-transmission and multiple-linkage</p></a></li>
<li><a href='#translink_expected_links_obs_mtsl'><p>Calculate expected number of observed pairs assuming multiple-transmission and single-linkage</p></a></li>
<li><a href='#translink_expected_links_obs_stsl'><p>Calculate expected number of observed pairs assuming single-transmission and single-linkage</p></a></li>
<li><a href='#translink_expected_links_true'><p>Calculate expected number of true transmission pairs</p></a></li>
<li><a href='#translink_expected_links_true_mtml'><p>Calculate expected number of true transmission pairs assuming multiple-transmission and multiple-linkage</p></a></li>
<li><a href='#translink_expected_links_true_mtsl'><p>Calculate expected number of true transmission pairs assuming multiple-transmission and single-linkage</p></a></li>
<li><a href='#translink_expected_links_true_stsl'><p>Calculate expected number of true transmission pairs assuming single-transmission and single-linkage</p></a></li>
<li><a href='#translink_fdr'><p>Calculate false discovery rate of identifying transmission pairs in a sample</p></a></li>
<li><a href='#translink_prob_transmit'><p>Calculate probability of transmission</p></a></li>
<li><a href='#translink_prob_transmit_mtml'><p>Calculate probability of transmission assuming multiple-transmission and multiple-linkage</p></a></li>
<li><a href='#translink_prob_transmit_mtsl'><p>Calculate probability of transmission assuming multiple-transmission and single-linkage</p></a></li>
<li><a href='#translink_prob_transmit_stsl'><p>Calculate probability of transmission assuming single-transmission and single-linkage</p></a></li>
<li><a href='#translink_samplesize'><p>Calculate sample size needed to identify true transmission links</p></a></li>
<li><a href='#translink_tdr'><p>Calculate true discovery rate of identifying transmission pairs</p></a></li>
<li><a href='#true_pairs'><p>Calculate expected number of true transmission pairs</p></a></li>
<li><a href='#true_pairs_mtml'><p>Expected number of true transmission pairs assuming multiple-transmission and multiple-linkage</p></a></li>
<li><a href='#true_pairs_mtsl'><p>Expected number of true transmission pairs assuming multiple-transmission and single-linkage</p></a></li>
<li><a href='#true_pairs_stsl'><p>Expected number of true transmission pairs assuming single-transmission and single-linkage</p></a></li>
<li><a href='#truediscoveryrate'><p>Calculate true discovery rate of a sample</p></a></li>
<li><a href='#varfreq_cdf_logistic'><p>Calculate cumulative observed variant prevalence at time t given logistic growth</p></a></li>
<li><a href='#varfreq_expected_mbias'><p>Calculate multiplicative bias (observed / actual) in variant prevalence</p></a></li>
<li><a href='#varfreq_freq_logistic'><p>Calculate observed variant prevalence at time t given logistic growth</p></a></li>
<li><a href='#varfreq_obs_freq'><p>Calculate observed variant prevalence</p></a></li>
<li><a href='#vartrack_cod_ratio'><p>Calculate the coefficient of detection ratio for two variants</p></a></li>
<li><a href='#vartrack_prob_detect'><p>Calculate the probability of detecting a variant given a sample size</p></a></li>
<li><a href='#vartrack_prob_detect_cont'><p>Calculate probability of detecting a variant given a per-timestep sample size assuming periodic sampling</p></a></li>
<li><a href='#vartrack_prob_detect_xsect'><p>Calculate probability of detecting a variant assuming cross-sectional sampling</p></a></li>
<li><a href='#vartrack_prob_prev'><p>Calculate confidence in a variant estimate given a sample size</p></a></li>
<li><a href='#vartrack_prob_prev_xsect'><p>Calculate confidence in a variant estimate assuming cross-sectional sampling</p></a></li>
<li><a href='#vartrack_samplesize_detect'><p>Calculate sample size needed for variant detection given a desired probability of detection</p></a></li>
<li><a href='#vartrack_samplesize_detect_cont'><p>Calculate sample size needed for variant detection assuming periodic sampling</p></a></li>
<li><a href='#vartrack_samplesize_detect_xsect'><p>Calculate sample size needed for variant detection assuming cross-sectional sampling</p></a></li>
<li><a href='#vartrack_samplesize_prev'><p>Calculate sample size needed for estimating variant prevalence given a desired confidence</p></a></li>
<li><a href='#vartrack_samplesize_prev_xsect'><p>Calculate sample size needed for variant prevalence estimation under cross-sectional sampling</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Sample Size Calculations for Molecular and Phylogenetic Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-04-25</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Justin Lessler &lt;jlessler@unc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements novel tools for estimating sample sizes needed for phylogenetic studies, including
    studies focused on estimating the probability of true pathogen transmission between two cases given phylogenetic linkage and studies focused on tracking pathogen variants at a population level. Methods described in Wohl, Giles, and Lessler (2021) and in Wohl, Lee, DiPrete, and Lessler (2023).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/HopkinsIDD/phylosamp">https://github.com/HopkinsIDD/phylosamp</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/HopkinsIDD/phylosamp/issues">https://github.com/HopkinsIDD/phylosamp/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), stats</td>
</tr>
<tr>
<td>Imports:</td>
<td>cli, lifecycle, rlang</td>
</tr>
<tr>
<td>Suggests:</td>
<td>cowplot, ggplot2, knitr, purrr, RColorBrewer, reshape2,
rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-23 21:59:22 UTC; swohl</td>
</tr>
<tr>
<td>Author:</td>
<td>Shirlee Wohl [aut, ctb],
  Elizabeth C Lee [aut, ctb],
  Lucy D'Agostino McGowan
    <a href="https://orcid.org/0000-0002-6983-2759"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, ctb],
  John R Giles [aut, ctb],
  Justin Lessler [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-23 23:12:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='phylosamp-package'>phylosamp: Sample Size Calculations for Molecular and Phylogenetic Studies</h2><span id='topic+phylosamp'></span><span id='topic+phylosamp-package'></span>

<h3>Description</h3>

<p>Implements novel tools for estimating sample sizes needed for phylogenetic studies, including studies focused on estimating the probability of true pathogen transmission between two cases given phylogenetic linkage and studies focused on tracking pathogen variants at a population level. Methods described in Wohl, Giles, and Lessler (2021) and in Wohl, Lee, DiPrete, and Lessler (2023).
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Justin Lessler <a href="mailto:jlessler@unc.edu">jlessler@unc.edu</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Shirlee Wohl <a href="mailto:swohl@scripps.edu">swohl@scripps.edu</a> [contributor]
</p>
</li>
<li><p> Elizabeth C Lee <a href="mailto:elizabeth.c.lee@jhu.edu">elizabeth.c.lee@jhu.edu</a> [contributor]
</p>
</li>
<li><p> Lucy D'Agostino McGowan <a href="mailto:lucydagostino@gmail.com">lucydagostino@gmail.com</a> (<a href="https://orcid.org/0000-0002-6983-2759">ORCID</a>) [contributor]
</p>
</li>
<li><p> John R Giles <a href="mailto:jrgiles@uw.edu">jrgiles@uw.edu</a> [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/HopkinsIDD/phylosamp">https://github.com/HopkinsIDD/phylosamp</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/HopkinsIDD/phylosamp/issues">https://github.com/HopkinsIDD/phylosamp/issues</a>
</p>
</li></ul>


<hr>
<h2 id='exp_links'>Calculate expected number of links in a sample</h2><span id='topic+exp_links'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the expected number of observed pairs in the sample that are linked by the linkage criteria. The function requires the sensitivity <code class="reqn">\eta</code>
and specificity <code class="reqn">\chi</code> of the linkage criteria, and sample size <code class="reqn">M</code>. Assumptions about transmission and linkage (single or multiple)
can be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>exp_links(eta, chi, rho, M, R = NULL, assumption = "mtml")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="exp_links_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="exp_links_+3A_chi">chi</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="exp_links_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="exp_links_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="exp_links_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen (default=NULL)</p>
</td></tr>
<tr><td><code id="exp_links_+3A_assumption">assumption</code></td>
<td>
<p>a character vector indicating which assumptions about transmission and linkage criteria. Default = <code>'mtml'</code>. Accepted arguments are:
</p>

<ol>
<li> <p><code>'stsl'</code> for the single-transmission single-linkage assumption (<code><a href="#topic+prob_trans_stsl">prob_trans_stsl()</a></code>).
</p>
</li>
<li> <p><code>'mtsl'</code> for the multiple-transmission single-linkage assumption (<code><a href="#topic+prob_trans_mtsl">prob_trans_mtsl()</a></code>).
</p>
</li>
<li> <p><code>'mtml'</code> for the multiple-transmission multiple-linkage assumption (<code><a href="#topic+prob_trans_mtml">prob_trans_mtml()</a></code>).
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of observed links in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other obs_pairs: 
<code><a href="#topic+obs_pairs_mtml">obs_pairs_mtml</a>()</code>,
<code><a href="#topic+obs_pairs_mtsl">obs_pairs_mtsl</a>()</code>,
<code><a href="#topic+obs_pairs_stsl">obs_pairs_stsl</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The simplest case: single-transmission, single-linkage, and perfect sensitivity
exp_links(eta=1, chi=0.9, rho=0.5, M=100, assumption='stsl')

# Multiple-transmission and imperfect sensitivity
exp_links(eta=0.99, chi=0.9, rho=1, M=50, R=1, assumption='mtsl')

# Small outbreak, larger sampling proportion
exp_links(eta=0.99, chi=0.95, rho=1, M=50, R=1, assumption='mtml')

# Large outbreak, small sampling proportion
exp_links(eta=0.99, chi=0.95, rho=0.05, M=1000, R=1, assumption='mtml')

</code></pre>

<hr>
<h2 id='falsediscoveryrate'>Calculate false discovery rate of a sample</h2><span id='topic+falsediscoveryrate'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the false discovery rate (proportion of linked pairs that are false positives) in a sample given the sensitivity <code class="reqn">\eta</code>
and specificity <code class="reqn">\chi</code> of the linkage criteria, and sample size <code class="reqn">M</code>. Assumptions about transmission and linkage (single or multiple)
can be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>falsediscoveryrate(eta, chi, rho, M, R = NULL, assumption = "mtml")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="falsediscoveryrate_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="falsediscoveryrate_+3A_chi">chi</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="falsediscoveryrate_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="falsediscoveryrate_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="falsediscoveryrate_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen (default=NULL)</p>
</td></tr>
<tr><td><code id="falsediscoveryrate_+3A_assumption">assumption</code></td>
<td>
<p>a character vector indicating which assumptions about transmission and linkage criteria. Default = <code>'mtml'</code>. Accepted arguments are:
</p>

<ol>
<li> <p><code>'stsl'</code> for the single-transmission single-linkage assumption (<code><a href="#topic+prob_trans_stsl">prob_trans_stsl()</a></code>).
</p>
</li>
<li> <p><code>'mtsl'</code> for the multiple-transmission single-linkage assumption (<code><a href="#topic+prob_trans_mtsl">prob_trans_mtsl()</a></code>).
</p>
</li>
<li> <p><code>'mtml'</code> for the multiple-transmission multiple-linkage assumption (<code><a href="#topic+prob_trans_mtml">prob_trans_mtml()</a></code>).
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the true discovery rate
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other discovery_rate: 
<code><a href="#topic+truediscoveryrate">truediscoveryrate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The simplest case: single-transmission, single-linkage, and perfect sensitivity
falsediscoveryrate(eta=1, chi=0.9, rho=0.5, M=100, assumption='stsl')

# Multiple-transmission and imperfect sensitivity
falsediscoveryrate(eta=0.99, chi=0.9, rho=1, M=50, R=1, assumption='mtsl')

# Small outbreak, larger sampling proportion
falsediscoveryrate(eta=0.99, chi=0.95, rho=1, M=50, R=1, assumption='mtml')

# Large outbreak, small sampling proportion
falsediscoveryrate(eta=0.99, chi=0.95, rho=0.5, M=1000, R=1, assumption='mtml')

</code></pre>

<hr>
<h2 id='gen_dists'>Calculate genetic distance distribution</h2><span id='topic+gen_dists'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
Function calculates the distribution of genetic distances in a population of viruses
with the given parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gen_dists(
  mut_rate,
  mean_gens_pdf,
  max_link_gens = 1,
  max_gens = NULL,
  max_dist = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gen_dists_+3A_mut_rate">mut_rate</code></td>
<td>
<p>mean number of mutations per generation, assumed to be Poisson distributed</p>
</td></tr>
<tr><td><code id="gen_dists_+3A_mean_gens_pdf">mean_gens_pdf</code></td>
<td>
<p>the density distribution of the mean number of generations between cases;
the index of this vector is assumed to be the discrete distance between cases</p>
</td></tr>
<tr><td><code id="gen_dists_+3A_max_link_gens">max_link_gens</code></td>
<td>
<p>the maximum generations of separation for linked pairs</p>
</td></tr>
<tr><td><code id="gen_dists_+3A_max_gens">max_gens</code></td>
<td>
<p>the maximum number of generations to consider, if <code>NULL</code> (default) value is set to the highest
number of generations in mean_gens_pdf with a non-zero probability</p>
</td></tr>
<tr><td><code id="gen_dists_+3A_max_dist">max_dist</code></td>
<td>
<p>the maximum distance to calculate, if <code>NULL</code> (default) value is set to max_gens * 99.9th percentile
of mut_rate Poisson distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with distances and probabilities
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other mutrate_functions: 
<code><a href="#topic+get_optim_roc">get_optim_roc</a>()</code>,
<code><a href="#topic+sens_spec_calc">sens_spec_calc</a>()</code>,
<code><a href="#topic+sens_spec_roc">sens_spec_roc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># ebola-like pathogen
R &lt;- 1.5
mut_rate &lt;- 1

# use simulated generation distributions from the provided 'genDistSim' data object
data('genDistSim')
mean_gens_pdf &lt;- as.numeric(genDistSim[genDistSim$R == R, -(1:2)])

# get theoretical genetic distance dist based on mutation rate and generation parameters
gen_dists(mut_rate = mut_rate,
          mean_gens_pdf = mean_gens_pdf,
          max_link_gens = 1)

</code></pre>

<hr>
<h2 id='gendist_distribution'>Calculate genetic distance distribution</h2><span id='topic+gendist_distribution'></span>

<h3>Description</h3>

<p>Function calculates the distribution of genetic distances in a population of viruses
with the given parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gendist_distribution(
  mut_rate,
  mean_gens_pdf,
  max_link_gens = 1,
  max_gens = NULL,
  max_dist = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gendist_distribution_+3A_mut_rate">mut_rate</code></td>
<td>
<p>mean number of mutations per generation, assumed to be Poisson distributed</p>
</td></tr>
<tr><td><code id="gendist_distribution_+3A_mean_gens_pdf">mean_gens_pdf</code></td>
<td>
<p>the density distribution of the mean number of generations between cases;
the index of this vector is assumed to be the discrete distance between cases</p>
</td></tr>
<tr><td><code id="gendist_distribution_+3A_max_link_gens">max_link_gens</code></td>
<td>
<p>the maximum generations of separation for linked pairs</p>
</td></tr>
<tr><td><code id="gendist_distribution_+3A_max_gens">max_gens</code></td>
<td>
<p>the maximum number of generations to consider, if <code>NULL</code> (default) value is set to the highest
number of generations in mean_gens_pdf with a non-zero probability</p>
</td></tr>
<tr><td><code id="gendist_distribution_+3A_max_dist">max_dist</code></td>
<td>
<p>the maximum distance to calculate, if <code>NULL</code> (default) value is set to max_gens * 99.9th percentile
of mut_rate Poisson distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with distances and probabilities
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other genetic distance functions: 
<code><a href="#topic+gendist_roc_format">gendist_roc_format</a>()</code>,
<code><a href="#topic+gendist_sensspec_cutoff">gendist_sensspec_cutoff</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># ebola-like pathogen
R &lt;- 1.5
mut_rate &lt;- 1

# use simulated generation distributions from the provided 'genDistSim' data object
data('genDistSim')
mean_gens_pdf &lt;- as.numeric(genDistSim[genDistSim$R == R, -(1:2)])

# get theoretical genetic distance dist based on mutation rate and generation parameters
gendist_distribution(mut_rate = mut_rate,
                     mean_gens_pdf = mean_gens_pdf,
                     max_link_gens = 1)

</code></pre>

<hr>
<h2 id='gendist_roc_format'>Make ROC curve from sensitivity and specificity</h2><span id='topic+gendist_roc_format'></span>

<h3>Description</h3>

<p>This is a wrapper function that takes output from the <code>gendist_sensspec_cutoff()</code> function and constructs values for the
Receiver Operating Characteristic (ROC) curve
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gendist_roc_format(
  cutoff,
  mut_rate,
  mean_gens_pdf,
  max_link_gens = 1,
  max_gens = NULL,
  max_dist = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gendist_roc_format_+3A_cutoff">cutoff</code></td>
<td>
<p>the maximum genetic distance at which to consider cases linked</p>
</td></tr>
<tr><td><code id="gendist_roc_format_+3A_mut_rate">mut_rate</code></td>
<td>
<p>mean number of mutations per generation, assumed to be Poisson distributed</p>
</td></tr>
<tr><td><code id="gendist_roc_format_+3A_mean_gens_pdf">mean_gens_pdf</code></td>
<td>
<p>the density distribution of the mean number of generations between cases;
the index of this vector is assumed to be the discrete distance between cases</p>
</td></tr>
<tr><td><code id="gendist_roc_format_+3A_max_link_gens">max_link_gens</code></td>
<td>
<p>the maximum generations of separation for linked pairs</p>
</td></tr>
<tr><td><code id="gendist_roc_format_+3A_max_gens">max_gens</code></td>
<td>
<p>the maximum number of generations to consider, if <code>NULL</code> (default) value set to the highest
number of generations in mean_gens_pdf with a non-zero probability</p>
</td></tr>
<tr><td><code id="gendist_roc_format_+3A_max_dist">max_dist</code></td>
<td>
<p>the maximum distance to calculate, if <code>NULL</code> (default) value set to max_gens * 99.9th percentile
of mut_rate Poisson distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with cutoff, sensitivity, and 1-specificity
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other genetic distance functions: 
<code><a href="#topic+gendist_distribution">gendist_distribution</a>()</code>,
<code><a href="#topic+gendist_sensspec_cutoff">gendist_sensspec_cutoff</a>()</code>
</p>
<p>Other ROC functions: 
<code><a href="#topic+optim_roc_threshold">optim_roc_threshold</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># ebola-like pathogen
R &lt;- 1.5
mut_rate &lt;- 1

# use simulated generation distributions
data('genDistSim')
mean_gens_pdf &lt;- as.numeric(genDistSim[genDistSim$R == R, -(1:2)])

# get theoretical genetic distance dist based on mutation rate and generation parameters
dists &lt;- as.data.frame(gendist_distribution(mut_rate = mut_rate,
                       mean_gens_pdf = mean_gens_pdf,
                       max_link_gens = 1))

dists &lt;- reshape2::melt(dists,
                        id.vars = 'dist',
                        variable.name = 'status',
                        value.name = 'prob')

# get sensitivity and specificity using the same paramters
roc_calc &lt;- gendist_roc_format(cutoff = 1:(max(dists$dist)-1),
                               mut_rate = mut_rate,
                               mean_gens_pdf = mean_gens_pdf)

</code></pre>

<hr>
<h2 id='gendist_sensspec_cutoff'>Calculate sensitivity and specificity of a genetic distance cutoff</h2><span id='topic+gendist_sensspec_cutoff'></span>

<h3>Description</h3>

<p>Function to calculate the sensitivity and specificity of a genetic distance cutoff
given an underlying mutation rate and mean number of generations between cases
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gendist_sensspec_cutoff(
  cutoff,
  mut_rate,
  mean_gens_pdf,
  max_link_gens = 1,
  max_gens = NULL,
  max_dist = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gendist_sensspec_cutoff_+3A_cutoff">cutoff</code></td>
<td>
<p>the maximum genetic distance at which to consider cases linked</p>
</td></tr>
<tr><td><code id="gendist_sensspec_cutoff_+3A_mut_rate">mut_rate</code></td>
<td>
<p>mean number of mutations per generation, assumed to be Poisson distributed</p>
</td></tr>
<tr><td><code id="gendist_sensspec_cutoff_+3A_mean_gens_pdf">mean_gens_pdf</code></td>
<td>
<p>the density distribution of the mean number of generations between cases;
the index of this vector is assumed to be the discrete distance between cases</p>
</td></tr>
<tr><td><code id="gendist_sensspec_cutoff_+3A_max_link_gens">max_link_gens</code></td>
<td>
<p>the maximum generations of separation for linked pairs</p>
</td></tr>
<tr><td><code id="gendist_sensspec_cutoff_+3A_max_gens">max_gens</code></td>
<td>
<p>the maximum number of generations to consider, if <code>NULL</code> (default) value set to the highest
number of generations in mean_gens_pdf with a non-zero probability</p>
</td></tr>
<tr><td><code id="gendist_sensspec_cutoff_+3A_max_dist">max_dist</code></td>
<td>
<p>the maximum distance to calculate, if <code>NULL</code> (default) value set to max_gens * 99.9th percentile
of mut_rate Poisson distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with the sensitivity and specificity for a particular genetic distance cutoff
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other genetic distance functions: 
<code><a href="#topic+gendist_distribution">gendist_distribution</a>()</code>,
<code><a href="#topic+gendist_roc_format">gendist_roc_format</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># calculate the sensitivity and specificity for a specific genetic distance threshold of 2 mutations
gendist_sensspec_cutoff(cutoff=2,
                        mut_rate=1,
                        mean_gens_pdf=c(0.02,0.08,0.15,0.75),
                        max_link_gens=1)

# calculate the sensitivity and specificity for a a range of genetic distance thresholds
gendist_sensspec_cutoff(cutoff=1:10,
                        mut_rate=1,
                        mean_gens_pdf=c(0.02,0.08,0.15,0.75),
                        max_link_gens=1)

</code></pre>

<hr>
<h2 id='genDistSim'>Simulations of the genetic distance distribution</h2><span id='topic+genDistSim'></span>

<h3>Description</h3>

<p>This data object contains the genetic distance distributions
for 168 values of <code class="reqn">R</code> between 1.3 and 18. The distributions represent the
the average of 1000 simulations for each value, which can be used as a
reasonable proxy for the generation distribution for large outbreaks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genDistSim
</code></pre>


<h3>Format</h3>

<p>dataframe
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, John Giles, and Justin Lessler
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genDistSim)

</code></pre>

<hr>
<h2 id='get_optim_roc'>Find optimal ROC threshold</h2><span id='topic+get_optim_roc'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function takes the dataframe output of the <code>sens_spec_roc()</code> function and finds the optimal threshold
of sensitivity and specificity by minimizing the distance to the top left corner of the Receiver Operating Characteristic (ROC) curve
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_optim_roc(roc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_optim_roc_+3A_roc">roc</code></td>
<td>
<p>a dataframe produced by the <code>sens_spec_roc()</code> function containing the Receiver Operating Characteristic (ROC) curve</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector containing optimal thresholds of sensitivity and specificity
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, John Giles, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other mutrate_functions: 
<code><a href="#topic+gen_dists">gen_dists</a>()</code>,
<code><a href="#topic+sens_spec_calc">sens_spec_calc</a>()</code>,
<code><a href="#topic+sens_spec_roc">sens_spec_roc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># ebola-like pathogen
R &lt;- 1.5
mut_rate &lt;- 1

# use simulated generation distributions
data(genDistSim)
mean_gens_pdf &lt;- as.numeric(genDistSim[genDistSim$R == R, -(1:2)])

# get theoretical genetic distance dist based on mutation rate and generation parameters
dists &lt;- as.data.frame(gen_dists(mut_rate = mut_rate,
                                 mean_gens_pdf = mean_gens_pdf,
                                 max_link_gens = 1))

# reshape dataframe for plotting
dists &lt;- reshape2::melt(dists,
                        id.vars = 'dist',
                        variable.name = 'status',
                        value.name = 'prob')

# get sensitivity and specificity using the same paramters
roc_calc &lt;- sens_spec_roc(cutoff = 1:(max(dists$dist)-1),
                          mut_rate = mut_rate,
                          mean_gens_pdf = mean_gens_pdf)

# get the optimal value for the ROC plot
optim_point &lt;- get_optim_roc(roc_calc)

</code></pre>

<hr>
<h2 id='obs_pairs_mtml'>Expected number of observed pairs assuming multiple-transmission and multiple-linkage</h2><span id='topic+obs_pairs_mtml'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the expected number of pairs observed in a sample of size <code>M</code>.
The multiple-transmission and multiple-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to multiple cases <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li>
<li><p> Linkage events are independent of one another (i.e, linkage of case <code class="reqn">i</code> to case <code class="reqn">j</code> has no bearing on linkage of case <code class="reqn">i</code> to any other sample).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>obs_pairs_mtml(chi, eta, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="obs_pairs_mtml_+3A_chi">chi</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="obs_pairs_mtml_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="obs_pairs_mtml_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="obs_pairs_mtml_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="obs_pairs_mtml_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of linked pairs observed in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other obs_pairs: 
<code><a href="#topic+exp_links">exp_links</a>()</code>,
<code><a href="#topic+obs_pairs_mtsl">obs_pairs_mtsl</a>()</code>,
<code><a href="#topic+obs_pairs_stsl">obs_pairs_stsl</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Perfect sensitivity and specificity
obs_pairs_mtml(eta=1, chi=1, rho=0.5, M=100, R=1)

obs_pairs_mtml(eta=0.99, chi=0.9, rho=1, M=50, R=1)

obs_pairs_mtml(eta=0.99, chi=0.9, rho=0.5, M=100, R=1)

</code></pre>

<hr>
<h2 id='obs_pairs_mtsl'>Expected number of observed pairs assuming multiple-transmission and single-linkage</h2><span id='topic+obs_pairs_mtsl'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the expected number of pairs observed in a sample of size <code>M</code>.
The multiple-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>obs_pairs_mtsl(chi, eta, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="obs_pairs_mtsl_+3A_chi">chi</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="obs_pairs_mtsl_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="obs_pairs_mtsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="obs_pairs_mtsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="obs_pairs_mtsl_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of linked pairs observed in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other obs_pairs: 
<code><a href="#topic+exp_links">exp_links</a>()</code>,
<code><a href="#topic+obs_pairs_mtml">obs_pairs_mtml</a>()</code>,
<code><a href="#topic+obs_pairs_stsl">obs_pairs_stsl</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Perfect sensitivity and specificity
obs_pairs_mtsl(eta=1, chi=1, rho=0.5, M=100, R=1)

obs_pairs_mtsl(eta=0.99, chi=0.9, rho=1, M=50, R=1)

obs_pairs_mtsl(eta=0.99, chi=0.9, rho=0.5, M=100, R=1)

</code></pre>

<hr>
<h2 id='obs_pairs_stsl'>Expected number of observed pairs assuming single-transmission and single-linkage</h2><span id='topic+obs_pairs_stsl'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the expected number of link pairs observed in a sample of size <code>M</code>.
The single-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is linked by transmission to only one other case <code class="reqn">j</code> in the population (<code class="reqn">N</code>).
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>obs_pairs_stsl(eta, chi, rho, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="obs_pairs_stsl_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="obs_pairs_stsl_+3A_chi">chi</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="obs_pairs_stsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="obs_pairs_stsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of linked pairs observed in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other obs_pairs: 
<code><a href="#topic+exp_links">exp_links</a>()</code>,
<code><a href="#topic+obs_pairs_mtml">obs_pairs_mtml</a>()</code>,
<code><a href="#topic+obs_pairs_mtsl">obs_pairs_mtsl</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># perfect sensitivity and specificity
obs_pairs_stsl(eta=1, chi=1, rho=0.5, M=100)

obs_pairs_stsl(eta=0.99, chi=0.9, rho=1, M=50)

obs_pairs_stsl(eta=0.99, chi=0.9, rho=0.5, M=100)

</code></pre>

<hr>
<h2 id='optim_roc_threshold'>Find optimal ROC threshold</h2><span id='topic+optim_roc_threshold'></span>

<h3>Description</h3>

<p>This function takes the dataframe output of the <code>gendist_roc_format()</code> function and finds the optimal threshold
of sensitivity and specificity by minimizing the distance to the top left corner of the Receiver Operating Characteristic (ROC) curve
</p>


<h3>Usage</h3>

<pre><code class='language-R'>optim_roc_threshold(roc)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="optim_roc_threshold_+3A_roc">roc</code></td>
<td>
<p>a dataframe produced by the <code>gendist_roc_format()</code> function containing the Receiver Operating Characteristic (ROC) curve</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector containing optimal thresholds of sensitivity and specificity
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, John Giles, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other ROC functions: 
<code><a href="#topic+gendist_roc_format">gendist_roc_format</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># ebola-like pathogen
R &lt;- 1.5
mut_rate &lt;- 1

# use simulated generation distributions
data("genDistSim")
mean_gens_pdf &lt;- as.numeric(genDistSim[genDistSim$R == R, -(1:2)])

# get theoretical genetic distance dist based on mutation rate and generation parameters
dists &lt;- as.data.frame(gendist_distribution(mut_rate = mut_rate,
                       mean_gens_pdf = mean_gens_pdf,
                       max_link_gens = 1))

# reshape dataframe for plotting
dists &lt;- reshape2::melt(dists,
                        id.vars = "dist",
                        variable.name = "status",
                        value.name = "prob")

# get sensitivity and specificity using the same paramters
roc_calc &lt;- gendist_roc_format(cutoff = 1:(max(dists$dist)-1),
                          mut_rate = mut_rate,
                          mean_gens_pdf = mean_gens_pdf)

# get the optimal value for the ROC plot
optim_point &lt;- optim_roc_threshold(roc_calc)

</code></pre>

<hr>
<h2 id='prob_trans_mtml'>Probability of transmission assuming multiple-transmission and multiple-linkage</h2><span id='topic+prob_trans_mtml'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the probability that two cases are linked by direct transmission
given that they have been linked by phylogenetic criteria. The multiple-transmission and multiple-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to multiple cases <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li>
<li><p> Linkage events are independent of one another (i.e, linkage of case <code class="reqn">i</code> to case <code class="reqn">j</code> has no bearing on linkage of case <code class="reqn">i</code> to any other sample).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>prob_trans_mtml(eta, chi, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prob_trans_mtml_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="prob_trans_mtml_+3A_chi">chi</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="prob_trans_mtml_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="prob_trans_mtml_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="prob_trans_mtml_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the probability of transmission between two cases given linkage by phylogenetic criteria
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other prob_trans: 
<code><a href="#topic+prob_trans_mtsl">prob_trans_mtsl</a>()</code>,
<code><a href="#topic+prob_trans_stsl">prob_trans_stsl</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Perfect sensitivity and specificity
prob_trans_mtml(eta=1, chi=1, rho=0.5, M=100, R=1)

prob_trans_mtml(eta=0.99, chi=0.9, rho=1, M=50, R=1)

prob_trans_mtml(eta=0.99, chi=0.9, rho=0.5, M=100, R=1)

</code></pre>

<hr>
<h2 id='prob_trans_mtsl'>Probability of transmission assuming multiple-transmission and single-linkage</h2><span id='topic+prob_trans_mtsl'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the probability that two cases are linked by direct transmission
given that they have been linked by phylogenetic criteria. The multiple-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>prob_trans_mtsl(chi, eta, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prob_trans_mtsl_+3A_chi">chi</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="prob_trans_mtsl_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="prob_trans_mtsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="prob_trans_mtsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="prob_trans_mtsl_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the probability of transmission between two cases given linkage by phylogenetic criteria
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other prob_trans: 
<code><a href="#topic+prob_trans_mtml">prob_trans_mtml</a>()</code>,
<code><a href="#topic+prob_trans_stsl">prob_trans_stsl</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Perfect sensitivity and specificity
prob_trans_mtsl(eta=1, chi=1, rho=0.5, M=100, R=1)

prob_trans_mtsl(eta=0.99, chi=0.9, rho=1, M=50, R=1)

prob_trans_mtsl(eta=0.99, chi=0.9, rho=0.5, M=100, R=1)

</code></pre>

<hr>
<h2 id='prob_trans_stsl'>Probability of transmission assuming single-transmission and single-linkage</h2><span id='topic+prob_trans_stsl'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the probability that two cases are linked by direct transmission
given that they have been linked by phylogenetic criteria. The single-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is linked by transmission to only one other case <code class="reqn">j</code> in the population (<code class="reqn">N</code>).
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>

<p>For perfect sensitivity, set <code>eta = 1</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prob_trans_stsl(eta, chi, rho, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prob_trans_stsl_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="prob_trans_stsl_+3A_chi">chi</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="prob_trans_stsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="prob_trans_stsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the probability of transmission between two cases given linkage by phylogenetic criteria
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other prob_trans: 
<code><a href="#topic+prob_trans_mtml">prob_trans_mtml</a>()</code>,
<code><a href="#topic+prob_trans_mtsl">prob_trans_mtsl</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># perfect sensitivity and specificity
prob_trans_stsl(eta=1, chi=1, rho=0.2, M=100)

# perfect sensitivity only
prob_trans_stsl(eta=1, chi=0.95, rho=0.2, M=100)

prob_trans_stsl(eta=0.99, chi=0.95, rho=0.9, M=50)

prob_trans_stsl(eta=0.99, chi=0.95, rho=0.05, M=100)

</code></pre>

<hr>
<h2 id='relR_power'>Calculate power for detecting differential transmission given a sample size</h2><span id='topic+relR_power'></span>

<h3>Description</h3>

<p>Function to calculate the power given a sample size. This is the top
level function to be called to calculate power given a sample size m
and a proportion sampled.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relR_power(
  m,
  R_a,
  R_b,
  p_a,
  N = NULL,
  rho = NULL,
  alpha = 0.05,
  alternative = c("two_sided", "less", "greater"),
  sensitivity = 1,
  specificity = 1,
  overdispersion = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relR_power_+3A_m">m</code></td>
<td>
<p>the sample size.</p>
</td></tr>
<tr><td><code id="relR_power_+3A_r_a">R_a</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
denominator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_power_+3A_r_b">R_b</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
numerator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_power_+3A_p_a">p_a</code></td>
<td>
<p>Numeric. The proportion of the population in group <code>a</code>. Must be
between 0 and 1.</p>
</td></tr>
<tr><td><code id="relR_power_+3A_n">N</code></td>
<td>
<p>Numeric (Positive). The size of the infected pool. Only one of
<code>rho</code> or <code>N</code> should be specified.</p>
</td></tr>
<tr><td><code id="relR_power_+3A_rho">rho</code></td>
<td>
<p>Numeric. The proportion of the infected pool sampled.
Only one of <code>rho</code> or <code>N</code> should be specified. Values should be between 0
and 1.</p>
</td></tr>
<tr><td><code id="relR_power_+3A_alpha">alpha</code></td>
<td>
<p>Numeric. The desired alpha level. Default: 0.05</p>
</td></tr>
<tr><td><code id="relR_power_+3A_alternative">alternative</code></td>
<td>
<p>Character. Specifies the alternative hypothesis.
Must be: <code>two_sided</code> (Default), <code>less</code>, or <code>greater</code></p>
</td></tr>
<tr><td><code id="relR_power_+3A_sensitivity">sensitivity</code></td>
<td>
<p>Numeric. The sensitivity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_power_+3A_specificity">specificity</code></td>
<td>
<p>Numeric. The specificity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_power_+3A_overdispersion">overdispersion</code></td>
<td>
<p>Numeric (Positive). An overdispersion parameter, set
if the assumed distribution of the number of edges is negative binomial.
If <code>NULL</code> the assumed distribution is Poisson (equivalent to an
overdispersion parameter of infinity) Default: <code>NULL</code> Note that this is
equivalent to setting the overdispersion parameter to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The power given <code>m</code>
</p>

<hr>
<h2 id='relR_power_simulated'>Simulate power for detecting differential transmission</h2><span id='topic+relR_power_simulated'></span>

<h3>Description</h3>

<p>Simulate power for detecting differential transmission
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relR_power_simulated(
  m,
  R_a,
  R_b,
  p_a,
  N,
  alpha = 0.05,
  alternative = c("two_sided", "less", "greater"),
  sensitivity = 1,
  specificity = 1,
  overdispersion = NULL,
  nsims = 1e+05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relR_power_simulated_+3A_m">m</code></td>
<td>
<p>the sample size.</p>
</td></tr>
<tr><td><code id="relR_power_simulated_+3A_r_a">R_a</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
denominator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_power_simulated_+3A_r_b">R_b</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
numerator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_power_simulated_+3A_p_a">p_a</code></td>
<td>
<p>Numeric. The proportion of the population in group <code>a</code>. Must be
between 0 and 1.</p>
</td></tr>
<tr><td><code id="relR_power_simulated_+3A_n">N</code></td>
<td>
<p>Numeric (Positive). The size of the infected pool. Only one of
<code>rho</code> or <code>N</code> should be specified.</p>
</td></tr>
<tr><td><code id="relR_power_simulated_+3A_alpha">alpha</code></td>
<td>
<p>Numeric. The desired alpha level. Default: 0.05</p>
</td></tr>
<tr><td><code id="relR_power_simulated_+3A_alternative">alternative</code></td>
<td>
<p>Character. Specifies the alternative hypothesis.
Must be: <code>two_sided</code> (Default), <code>less</code>, or <code>greater</code></p>
</td></tr>
<tr><td><code id="relR_power_simulated_+3A_sensitivity">sensitivity</code></td>
<td>
<p>Numeric. The sensitivity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_power_simulated_+3A_specificity">specificity</code></td>
<td>
<p>Numeric. The specificity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_power_simulated_+3A_overdispersion">overdispersion</code></td>
<td>
<p>Numeric (Positive). An overdispersion parameter, set
if the assumed distribution of the number of edges is negative binomial.
If <code>NULL</code> the assumed distribution is Poisson (equivalent to an
overdispersion parameter of infinity) Default: <code>NULL</code> Note that this is
equivalent to setting the overdispersion parameter to <code>Inf</code>.</p>
</td></tr>
<tr><td><code id="relR_power_simulated_+3A_nsims">nsims</code></td>
<td>
<p>Numeric. The number of simulations. Default: 100000</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Simulated power
</p>

<hr>
<h2 id='relR_samplesize'>Calculate sample size needed to detect differential transmission</h2><span id='topic+relR_samplesize'></span>

<h3>Description</h3>

<p>Function for calculating sample size given a set of assumptions. This is the
high level wrapper function that users should call directly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relR_samplesize(
  R_a,
  R_b,
  p_a,
  N,
  alpha = 0.05,
  alternative = c("two_sided", "less", "greater"),
  power = 0.8,
  sensitivity = 1,
  specificity = 1,
  overdispersion = NULL,
  correct_for_imbalance = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relR_samplesize_+3A_r_a">R_a</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
denominator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_+3A_r_b">R_b</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
numerator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_+3A_p_a">p_a</code></td>
<td>
<p>Numeric. The proportion of the population in group <code>a</code>. Must be
between 0 and 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_+3A_n">N</code></td>
<td>
<p>Numeric (Positive). The size of the infected pool. Only one of
<code>rho</code> or <code>N</code> should be specified.</p>
</td></tr>
<tr><td><code id="relR_samplesize_+3A_alpha">alpha</code></td>
<td>
<p>Numeric. The desired alpha level. Default: 0.05</p>
</td></tr>
<tr><td><code id="relR_samplesize_+3A_alternative">alternative</code></td>
<td>
<p>Character. Specifies the alternative hypothesis.
Must be: <code>two_sided</code> (Default), <code>less</code>, or <code>greater</code></p>
</td></tr>
<tr><td><code id="relR_samplesize_+3A_power">power</code></td>
<td>
<p>Numeric. The desired power. Must be a value between 0 and 1.
Default: 0.8.</p>
</td></tr>
<tr><td><code id="relR_samplesize_+3A_sensitivity">sensitivity</code></td>
<td>
<p>Numeric. The sensitivity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_+3A_specificity">specificity</code></td>
<td>
<p>Numeric. The specificity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_+3A_overdispersion">overdispersion</code></td>
<td>
<p>Numeric (Positive). An overdispersion parameter, set
if the assumed distribution of the number of edges is negative binomial.
If <code>NULL</code> the assumed distribution is Poisson (equivalent to an
overdispersion parameter of infinity) Default: <code>NULL</code> Note that this is
equivalent to setting the overdispersion parameter to <code>Inf</code>.</p>
</td></tr>
<tr><td><code id="relR_samplesize_+3A_correct_for_imbalance">correct_for_imbalance</code></td>
<td>
<p>Logical. Should we use simulation to correct
for being over/under powered due to large differences in group sizes?
Default: <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Sample size needed achieve desired type I and II error rates
under assumptions. Will return <code>NA</code> and throw a warning if impossible.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Calculate sample size needed to detect a difference between groups where 
## group A has a reproductive value of 2, group B has a reproductive 
## value of 2.5, the groups are balanced, and the total outbreak size is 
## 1,000

relR_samplesize(R_a = 2, 
                R_b = 2.5, 
                p_a = 0.5,
                N = 1000)

## Update the above calculation to account for imperfect sensitivity = 0.7
relR_samplesize(R_a = 2, 
                R_b = 2.5, 
                p_a = 0.5,
                N = 1000,
                sensitivity = 0.7)

## Update the above calculation to allow for overdispersion
relR_samplesize(R_a = 2, 
                R_b = 2.5, 
                p_a = 0.5,
                N = 1000,
                sensitivity = 0.7,
                overdispersion = 2000)

</code></pre>

<hr>
<h2 id='relR_samplesize_basic'>Calculate simple derived sample size for detecting differential transmission</h2><span id='topic+relR_samplesize_basic'></span>

<h3>Description</h3>

<p>Function that does the simple derived sample size calculation with
no corrections. I.e., directly applies the math as if sensitivity
and specificity are perfect.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relR_samplesize_basic(
  R_a,
  R_b,
  p_a,
  N,
  alpha = 0.05,
  alternative = c("two_sided", "less", "greater"),
  power = 0.8,
  overdispersion = NULL,
  allow_impossible_m = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relR_samplesize_basic_+3A_r_a">R_a</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
denominator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_basic_+3A_r_b">R_b</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
numerator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_basic_+3A_p_a">p_a</code></td>
<td>
<p>Numeric. The proportion of the population in group <code>a</code>. Must be
between 0 and 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_basic_+3A_n">N</code></td>
<td>
<p>Numeric (Positive). The size of the infected pool. Only one of
<code>rho</code> or <code>N</code> should be specified.</p>
</td></tr>
<tr><td><code id="relR_samplesize_basic_+3A_alpha">alpha</code></td>
<td>
<p>Numeric. The desired alpha level. Default: 0.05</p>
</td></tr>
<tr><td><code id="relR_samplesize_basic_+3A_alternative">alternative</code></td>
<td>
<p>Character. Specifies the alternative hypothesis.
Must be: <code>two_sided</code> (Default), <code>less</code>, or <code>greater</code></p>
</td></tr>
<tr><td><code id="relR_samplesize_basic_+3A_power">power</code></td>
<td>
<p>Numeric. The desired power. Must be a value between 0 and 1.
Default: 0.8.</p>
</td></tr>
<tr><td><code id="relR_samplesize_basic_+3A_overdispersion">overdispersion</code></td>
<td>
<p>Numeric (Positive). An overdispersion parameter, set
if the assumed distribution of the number of edges is negative binomial.
If <code>NULL</code> the assumed distribution is Poisson (equivalent to an
overdispersion parameter of infinity) Default: <code>NULL</code> Note that this is
equivalent to setting the overdispersion parameter to <code>Inf</code>.</p>
</td></tr>
<tr><td><code id="relR_samplesize_basic_+3A_allow_impossible_m">allow_impossible_m</code></td>
<td>
<p>Logical. Indicates whether a value for <code>m</code> can be
returned that is greater than the input <code>N</code>. Default: <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The required sample size. <code>NA</code> if larger than <code>N</code>.
</p>

<hr>
<h2 id='relR_samplesize_ci'>Calculate sample size for detecting differential transmission with uncertainty bounds</h2><span id='topic+relR_samplesize_ci'></span>

<h3>Description</h3>

<p>This function assumes you
want to correct for imbalance, if not there is a closed form solution
for the estimated sample size that does not include uncertainty bounds.
(see <code><a href="#topic+relR_samplesize">relR_samplesize</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relR_samplesize_ci(
  R_a,
  R_b,
  p_a,
  N,
  alpha = 0.05,
  alternative = c("two_sided", "less", "greater"),
  power = 0.8,
  sensitivity = 1,
  specificity = 1,
  overdispersion = NULL,
  nsims = 1000,
  uncertainty_percent = 0.95,
  B = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relR_samplesize_ci_+3A_r_a">R_a</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
denominator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_r_b">R_b</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
numerator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_p_a">p_a</code></td>
<td>
<p>Numeric. The proportion of the population in group <code>a</code>. Must be
between 0 and 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_n">N</code></td>
<td>
<p>Numeric (Positive). The size of the infected pool. Only one of
<code>rho</code> or <code>N</code> should be specified.</p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_alpha">alpha</code></td>
<td>
<p>Numeric. The desired alpha level. Default: 0.05</p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_alternative">alternative</code></td>
<td>
<p>Character. Specifies the alternative hypothesis.
Must be: <code>two_sided</code> (Default), <code>less</code>, or <code>greater</code></p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_power">power</code></td>
<td>
<p>Numeric. The desired power. Must be a value between 0 and 1.
Default: 0.8.</p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_sensitivity">sensitivity</code></td>
<td>
<p>Numeric. The sensitivity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_specificity">specificity</code></td>
<td>
<p>Numeric. The specificity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_overdispersion">overdispersion</code></td>
<td>
<p>Numeric (Positive). An overdispersion parameter, set
if the assumed distribution of the number of edges is negative binomial.
If <code>NULL</code> the assumed distribution is Poisson (equivalent to an
overdispersion parameter of infinity) Default: <code>NULL</code> Note that this is
equivalent to setting the overdispersion parameter to <code>Inf</code>.</p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_nsims">nsims</code></td>
<td>
<p>The number of inner simulations run per estimate.
Default: 10000</p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_uncertainty_percent">uncertainty_percent</code></td>
<td>
<p>The percent of the uncertainty interval.
Default: .95</p>
</td></tr>
<tr><td><code id="relR_samplesize_ci_+3A_b">B</code></td>
<td>
<p>The number of outer simulations run to estimate the uncertainty.
Default: 1000</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with three quantities:
</p>

<ul>
<li><p> sample size: Sample size needed achieve desired type I and II error rates
under assumptions. Will return NA and throw a warning if impossible.
</p>
</li>
<li><p> lower bound: The lower bound of an uncertainty interval
</p>
</li>
<li><p> upper bound: The upper bound of an uncertainty interval
</p>
</li></ul>


<hr>
<h2 id='relR_samplesize_linkerr'>Calculate sample size for detecting differential transmission correcting for sensitivity and specificity</h2><span id='topic+relR_samplesize_linkerr'></span>

<h3>Description</h3>

<p>Function to run the sample size calculation correcting for imperfect
sensitivity and specificity, but not doing any simulation based corrections.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relR_samplesize_linkerr(
  R_a,
  R_b,
  p_a,
  N,
  alpha = 0.05,
  alternative = c("two_sided", "less", "greater"),
  power = 0.8,
  sensitivity = 1,
  specificity = 1,
  overdispersion = NULL,
  allow_impossible_m = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relR_samplesize_linkerr_+3A_r_a">R_a</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
denominator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_linkerr_+3A_r_b">R_b</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
numerator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_linkerr_+3A_p_a">p_a</code></td>
<td>
<p>Numeric. The proportion of the population in group <code>a</code>. Must be
between 0 and 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_linkerr_+3A_n">N</code></td>
<td>
<p>Numeric (Positive). The size of the infected pool. Only one of
<code>rho</code> or <code>N</code> should be specified.</p>
</td></tr>
<tr><td><code id="relR_samplesize_linkerr_+3A_alpha">alpha</code></td>
<td>
<p>Numeric. The desired alpha level. Default: 0.05</p>
</td></tr>
<tr><td><code id="relR_samplesize_linkerr_+3A_alternative">alternative</code></td>
<td>
<p>Character. Specifies the alternative hypothesis.
Must be: <code>two_sided</code> (Default), <code>less</code>, or <code>greater</code></p>
</td></tr>
<tr><td><code id="relR_samplesize_linkerr_+3A_power">power</code></td>
<td>
<p>Numeric. The desired power. Must be a value between 0 and 1.
Default: 0.8.</p>
</td></tr>
<tr><td><code id="relR_samplesize_linkerr_+3A_sensitivity">sensitivity</code></td>
<td>
<p>Numeric. The sensitivity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_linkerr_+3A_specificity">specificity</code></td>
<td>
<p>Numeric. The specificity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_linkerr_+3A_overdispersion">overdispersion</code></td>
<td>
<p>Numeric (Positive). An overdispersion parameter, set
if the assumed distribution of the number of edges is negative binomial.
If <code>NULL</code> the assumed distribution is Poisson (equivalent to an
overdispersion parameter of infinity) Default: <code>NULL</code> Note that this is
equivalent to setting the overdispersion parameter to <code>Inf</code>.</p>
</td></tr>
<tr><td><code id="relR_samplesize_linkerr_+3A_allow_impossible_m">allow_impossible_m</code></td>
<td>
<p>Logical. Indicates whether a value for <code>m</code> can be
returned that is greater than the input <code>N</code>. Default: <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Sample size needed achieve desired type I and II error rates
under assumptions. Will return NA and throw a warning if impossible.
</p>

<hr>
<h2 id='relR_samplesize_opterr'>Function to calculate the error in estimated sample size for use in optimize function</h2><span id='topic+relR_samplesize_opterr'></span>

<h3>Description</h3>

<p>Function to calculate the error in estimated sample size for use in optimize function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relR_samplesize_opterr(
  m,
  R_a,
  R_b,
  p_a,
  N,
  alpha,
  alternative,
  power,
  sensitivity,
  specificity,
  overdispersion
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relR_samplesize_opterr_+3A_m">m</code></td>
<td>
<p>the sample size.</p>
</td></tr>
<tr><td><code id="relR_samplesize_opterr_+3A_r_a">R_a</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
denominator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_opterr_+3A_r_b">R_b</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
numerator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_opterr_+3A_p_a">p_a</code></td>
<td>
<p>Numeric. The proportion of the population in group <code>a</code>. Must be
between 0 and 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_opterr_+3A_n">N</code></td>
<td>
<p>Numeric (Positive). The size of the infected pool. Only one of
<code>rho</code> or <code>N</code> should be specified.</p>
</td></tr>
<tr><td><code id="relR_samplesize_opterr_+3A_alpha">alpha</code></td>
<td>
<p>Numeric. The desired alpha level. Default: 0.05</p>
</td></tr>
<tr><td><code id="relR_samplesize_opterr_+3A_alternative">alternative</code></td>
<td>
<p>Character. Specifies the alternative hypothesis.
Must be: <code>two_sided</code> (Default), <code>less</code>, or <code>greater</code></p>
</td></tr>
<tr><td><code id="relR_samplesize_opterr_+3A_power">power</code></td>
<td>
<p>Numeric. The desired power. Must be a value between 0 and 1.
Default: 0.8.</p>
</td></tr>
<tr><td><code id="relR_samplesize_opterr_+3A_sensitivity">sensitivity</code></td>
<td>
<p>Numeric. The sensitivity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_opterr_+3A_specificity">specificity</code></td>
<td>
<p>Numeric. The specificity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_opterr_+3A_overdispersion">overdispersion</code></td>
<td>
<p>Numeric (Positive). An overdispersion parameter, set
if the assumed distribution of the number of edges is negative binomial.
If <code>NULL</code> the assumed distribution is Poisson (equivalent to an
overdispersion parameter of infinity) Default: <code>NULL</code> Note that this is
equivalent to setting the overdispersion parameter to <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Squared error between the input sample size and estimated sample size
</p>

<hr>
<h2 id='relR_samplesize_simsolve'>Calculate optimized sample size for detecting differential transmission</h2><span id='topic+relR_samplesize_simsolve'></span>

<h3>Description</h3>

<p>Function to calculate optimized sample size by solving the
transcendental equation that occurs when you replace the R values with ones
that account for sensitivity and specificity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relR_samplesize_simsolve(
  R_a,
  R_b,
  p_a,
  N,
  alpha = 0.05,
  alternative = c("two_sided", "less", "greater"),
  power = 0.8,
  sensitivity = 1,
  specificity = 1,
  overdispersion = NULL,
  epsilon = 0.01,
  nsims = 1e+05,
  tolerance = 10
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relR_samplesize_simsolve_+3A_r_a">R_a</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
denominator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_r_b">R_b</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
numerator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_p_a">p_a</code></td>
<td>
<p>Numeric. The proportion of the population in group <code>a</code>. Must be
between 0 and 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_n">N</code></td>
<td>
<p>Numeric (Positive). The size of the infected pool. Only one of
<code>rho</code> or <code>N</code> should be specified.</p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_alpha">alpha</code></td>
<td>
<p>Numeric. The desired alpha level. Default: 0.05</p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_alternative">alternative</code></td>
<td>
<p>Character. Specifies the alternative hypothesis.
Must be: <code>two_sided</code> (Default), <code>less</code>, or <code>greater</code></p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_power">power</code></td>
<td>
<p>Numeric. The desired power. Must be a value between 0 and 1.
Default: 0.8.</p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_sensitivity">sensitivity</code></td>
<td>
<p>Numeric. The sensitivity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_specificity">specificity</code></td>
<td>
<p>Numeric. The specificity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_overdispersion">overdispersion</code></td>
<td>
<p>Numeric (Positive). An overdispersion parameter, set
if the assumed distribution of the number of edges is negative binomial.
If <code>NULL</code> the assumed distribution is Poisson (equivalent to an
overdispersion parameter of infinity) Default: <code>NULL</code> Note that this is
equivalent to setting the overdispersion parameter to <code>Inf</code>.</p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_epsilon">epsilon</code></td>
<td>
<p>Numeric. Dictates the minimum value for <code>R_b = R_a + epsilon</code>
attempted in the simulation. Default: 0.01.</p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_nsims">nsims</code></td>
<td>
<p>Dictates the number of simulations for each power simulation.
Default: 100000</p>
</td></tr>
<tr><td><code id="relR_samplesize_simsolve_+3A_tolerance">tolerance</code></td>
<td>
<p>Dictates the tolerance for the binary search. Default: 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Simulated sample size needed achieve desired type I and II error rates
under assumptions. Will return NA and throw a warning if impossible.
</p>

<hr>
<h2 id='relR_samplesize_solve'>Calculate optimal sample size for detecting differential transmission with imperfect specificity</h2><span id='topic+relR_samplesize_solve'></span>

<h3>Description</h3>

<p>Function to solve for optimal sample size when the specificity isn't 1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relR_samplesize_solve(
  R_a,
  R_b,
  p_a,
  N,
  alpha = 0.05,
  alternative = c("two_sided", "less", "greater"),
  power = 0.8,
  sensitivity = 1,
  specificity = 1,
  overdispersion = NULL,
  allow_impossible_m = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relR_samplesize_solve_+3A_r_a">R_a</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
denominator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_solve_+3A_r_b">R_b</code></td>
<td>
<p>Numeric (Positive). The assumed R among the group in the
numerator of the ratio. Input value must be greater than 0.</p>
</td></tr>
<tr><td><code id="relR_samplesize_solve_+3A_p_a">p_a</code></td>
<td>
<p>Numeric. The proportion of the population in group <code>a</code>. Must be
between 0 and 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_solve_+3A_n">N</code></td>
<td>
<p>Numeric (Positive). The size of the infected pool. Only one of
<code>rho</code> or <code>N</code> should be specified.</p>
</td></tr>
<tr><td><code id="relR_samplesize_solve_+3A_alpha">alpha</code></td>
<td>
<p>Numeric. The desired alpha level. Default: 0.05</p>
</td></tr>
<tr><td><code id="relR_samplesize_solve_+3A_alternative">alternative</code></td>
<td>
<p>Character. Specifies the alternative hypothesis.
Must be: <code>two_sided</code> (Default), <code>less</code>, or <code>greater</code></p>
</td></tr>
<tr><td><code id="relR_samplesize_solve_+3A_power">power</code></td>
<td>
<p>Numeric. The desired power. Must be a value between 0 and 1.
Default: 0.8.</p>
</td></tr>
<tr><td><code id="relR_samplesize_solve_+3A_sensitivity">sensitivity</code></td>
<td>
<p>Numeric. The sensitivity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_solve_+3A_specificity">specificity</code></td>
<td>
<p>Numeric. The specificity of the linkage criteria.
Must be between 0 and 1. Default: 1.</p>
</td></tr>
<tr><td><code id="relR_samplesize_solve_+3A_overdispersion">overdispersion</code></td>
<td>
<p>Numeric (Positive). An overdispersion parameter, set
if the assumed distribution of the number of edges is negative binomial.
If <code>NULL</code> the assumed distribution is Poisson (equivalent to an
overdispersion parameter of infinity) Default: <code>NULL</code> Note that this is
equivalent to setting the overdispersion parameter to <code>Inf</code>.</p>
</td></tr>
<tr><td><code id="relR_samplesize_solve_+3A_allow_impossible_m">allow_impossible_m</code></td>
<td>
<p>Logical. Indicates whether a value for <code>m</code> can be
returned that is greater than the input <code>N</code>. Default: <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The sample size
</p>

<hr>
<h2 id='samplesize'>Calculate sample size</h2><span id='topic+samplesize'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the sample size needed to obtain at least a defined false discovery rate given
a final outbreak size <code class="reqn">N</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>samplesize(eta, chi, N, R = NULL, phi, min_pairs = 1, assumption = "mtml")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="samplesize_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="samplesize_+3A_chi">chi</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="samplesize_+3A_n">N</code></td>
<td>
<p>scalar or vector giving the final outbreak size</p>
</td></tr>
<tr><td><code id="samplesize_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
<tr><td><code id="samplesize_+3A_phi">phi</code></td>
<td>
<p>scalar or vector giving the desired true discovery rate (1-false discovery rate)</p>
</td></tr>
<tr><td><code id="samplesize_+3A_min_pairs">min_pairs</code></td>
<td>
<p>minimum number of linked pairs observed in the sample, defaults to 1 pair (2 samples); this is to ensure reasonable results are obtained</p>
</td></tr>
<tr><td><code id="samplesize_+3A_assumption">assumption</code></td>
<td>
<p>a character vector indicating which assumptions about transmission and linkage criteria. Default = <code>'mtml'</code>. Accepted arguments are:
</p>

<ol>
<li> <p><code>'stsl'</code> for the single-transmission single-linkage assumption (<code><a href="#topic+prob_trans_stsl">prob_trans_stsl()</a></code>).
</p>
</li>
<li> <p><code>'mtsl'</code> for the multiple-transmission single-linkage assumption (<code><a href="#topic+prob_trans_mtsl">prob_trans_mtsl()</a></code>).
</p>
</li>
<li> <p><code>'mtml'</code> for the multiple-transmission multiple-linkage assumption (<code><a href="#topic+prob_trans_mtml">prob_trans_mtml()</a></code>).
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the sample size needed to meet the given conditions
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>Examples</h3>

<pre><code class='language-R'>samplesize(eta=0.99, chi=0.995, N=100, R=1, phi=0.75)

</code></pre>

<hr>
<h2 id='sens_spec_calc'>Calculate sensitivity and specificity</h2><span id='topic+sens_spec_calc'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
Function to calculate the sensitivity and specificity of a genetic distance cutoff
given an underlying mutation rate and mean number of generations between cases
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sens_spec_calc(
  cutoff,
  mut_rate,
  mean_gens_pdf,
  max_link_gens = 1,
  max_gens = NULL,
  max_dist = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sens_spec_calc_+3A_cutoff">cutoff</code></td>
<td>
<p>the maximum genetic distance at which to consider cases linked</p>
</td></tr>
<tr><td><code id="sens_spec_calc_+3A_mut_rate">mut_rate</code></td>
<td>
<p>mean number of mutations per generation, assumed to be Poisson distributed</p>
</td></tr>
<tr><td><code id="sens_spec_calc_+3A_mean_gens_pdf">mean_gens_pdf</code></td>
<td>
<p>the density distribution of the mean number of generations between cases;
the index of this vector is assumed to be the discrete distance between cases</p>
</td></tr>
<tr><td><code id="sens_spec_calc_+3A_max_link_gens">max_link_gens</code></td>
<td>
<p>the maximum generations of separation for linked pairs</p>
</td></tr>
<tr><td><code id="sens_spec_calc_+3A_max_gens">max_gens</code></td>
<td>
<p>the maximum number of generations to consider, if <code>NULL</code> (default) value set to the highest
number of generations in mean_gens_pdf with a non-zero probability</p>
</td></tr>
<tr><td><code id="sens_spec_calc_+3A_max_dist">max_dist</code></td>
<td>
<p>the maximum distance to calculate, if <code>NULL</code> (default) value set to max_gens * 99.9th percentile
of mut_rate Poisson distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with the sensitivity and specificity for a particular genetic distance cutoff
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other mutrate_functions: 
<code><a href="#topic+gen_dists">gen_dists</a>()</code>,
<code><a href="#topic+get_optim_roc">get_optim_roc</a>()</code>,
<code><a href="#topic+sens_spec_roc">sens_spec_roc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># calculate the sensitivity and specificity for a specific genetic distance threshold of 2 mutations
sens_spec_calc(cutoff=2,
               mut_rate=1,
               mean_gens_pdf=c(0.02,0.08,0.15,0.75),
               max_link_gens=1)

# calculate the sensitivity and specificity for a a range of genetic distance thresholds
sens_spec_calc(cutoff=1:10,
               mut_rate=1,
               mean_gens_pdf=c(0.02,0.08,0.15,0.75),
               max_link_gens=1)

</code></pre>

<hr>
<h2 id='sens_spec_roc'>Make ROC from sensitivity and specificity</h2><span id='topic+sens_spec_roc'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This is a wrapper function that takes output from the <code>sens_spec_calc()</code> function and constructs values for the
Receiver Operating Characteristic (ROC) curve
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sens_spec_roc(
  cutoff,
  mut_rate,
  mean_gens_pdf,
  max_link_gens = 1,
  max_gens = NULL,
  max_dist = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sens_spec_roc_+3A_cutoff">cutoff</code></td>
<td>
<p>the maximum genetic distance at which to consider cases linked</p>
</td></tr>
<tr><td><code id="sens_spec_roc_+3A_mut_rate">mut_rate</code></td>
<td>
<p>mean number of mutations per generation, assumed to be Poisson distributed</p>
</td></tr>
<tr><td><code id="sens_spec_roc_+3A_mean_gens_pdf">mean_gens_pdf</code></td>
<td>
<p>the density distribution of the mean number of generations between cases;
the index of this vector is assumed to be the discrete distance between cases</p>
</td></tr>
<tr><td><code id="sens_spec_roc_+3A_max_link_gens">max_link_gens</code></td>
<td>
<p>the maximum generations of separation for linked pairs</p>
</td></tr>
<tr><td><code id="sens_spec_roc_+3A_max_gens">max_gens</code></td>
<td>
<p>the maximum number of generations to consider, if <code>NULL</code> (default) value set to the highest
number of generations in mean_gens_pdf with a non-zero probability</p>
</td></tr>
<tr><td><code id="sens_spec_roc_+3A_max_dist">max_dist</code></td>
<td>
<p>the maximum distance to calculate, if <code>NULL</code> (default) value set to max_gens * 99.9th percentile
of mut_rate Poisson distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with cutoff, sensitivity, and 1-specificity
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other mutrate_functions: 
<code><a href="#topic+gen_dists">gen_dists</a>()</code>,
<code><a href="#topic+get_optim_roc">get_optim_roc</a>()</code>,
<code><a href="#topic+sens_spec_calc">sens_spec_calc</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># ebola-like pathogen
R &lt;- 1.5
mut_rate &lt;- 1

# use simulated generation distributions
data('genDistSim')
mean_gens_pdf &lt;- as.numeric(genDistSim[genDistSim$R == R, -(1:2)])

# get theoretical genetic distance dist based on mutation rate and generation parameters
dists &lt;- as.data.frame(gen_dists(mut_rate = mut_rate,
                                 mean_gens_pdf = mean_gens_pdf,
                                 max_link_gens = 1))

dists &lt;- reshape2::melt(dists,
                        id.vars = 'dist',
                        variable.name = 'status',
                        value.name = 'prob')

# get sensitivity and specificity using the same paramters
roc_calc &lt;- sens_spec_roc(cutoff = 1:(max(dists$dist)-1),
                          mut_rate = mut_rate,
                          mean_gens_pdf = mean_gens_pdf)

</code></pre>

<hr>
<h2 id='translink_expected_links_obs'>Calculate expected number of transmission links in a sample</h2><span id='topic+translink_expected_links_obs'></span>

<h3>Description</h3>

<p>This function calculates the expected number of observed pairs in the sample that are linked by the linkage criteria. The function requires the sensitivity
and specificity of the linkage criteria, and sample size <code class="reqn">M</code>. Assumptions about transmission and linkage (single or multiple)
can be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translink_expected_links_obs(
  sensitivity,
  specificity,
  rho,
  M,
  R = NULL,
  assumption = "mtml"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_expected_links_obs_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_+3A_specificity">specificity</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen (default=NULL)</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_+3A_assumption">assumption</code></td>
<td>
<p>a character vector indicating which assumptions about transmission and linkage criteria. Default = <code>'mtml'</code>. Accepted arguments are:
</p>

<ol>
<li> <p><code>'stsl'</code> for the single-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtsl'</code> for the multiple-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtml'</code> for the multiple-transmission multiple-linkage assumption.
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of observed links in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The simplest case: single-transmission, single-linkage, and perfect sensitivity
translink_expected_links_obs(sensitivity=1, specificity=0.9, rho=0.5, M=100, assumption='stsl')

# Multiple-transmission and imperfect sensitivity
translink_expected_links_obs(sensitivity=0.99, specificity=0.9, rho=1, M=50, R=1, assumption='mtsl')

# Small outbreak, larger sampling proportion
translink_expected_links_obs(sensitivity=0.99, specificity=0.95, rho=1, M=50, 
R=1, assumption='mtml')

# Large outbreak, small sampling proportion
translink_expected_links_obs(sensitivity=0.99, specificity=0.95, 
rho=0.05, M=1000, R=1, assumption='mtml')

</code></pre>

<hr>
<h2 id='translink_expected_links_obs_mtml'>Calculate expected number of observed pairs assuming multiple-transmission and multiple-linkage</h2><span id='topic+translink_expected_links_obs_mtml'></span>

<h3>Description</h3>

<p>This function calculates the expected number of pairs observed in a sample of size <code>M</code>.
The multiple-transmission and multiple-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to multiple cases <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li>
<li><p> Linkage events are independent of one another (i.e, linkage of case <code class="reqn">i</code> to case <code class="reqn">j</code> has no bearing on linkage of case <code class="reqn">i</code> to any other sample).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>translink_expected_links_obs_mtml(specificity, sensitivity, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_expected_links_obs_mtml_+3A_specificity">specificity</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_mtml_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_mtml_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_mtml_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_mtml_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of linked pairs observed in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Perfect sensitivity and specificity
translink_expected_links_obs_mtml(sensitivity=1, specificity=1, rho=0.5, M=100, R=1)

translink_expected_links_obs_mtml(sensitivity=0.99, specificity=0.9, rho=1, M=50, R=1)

translink_expected_links_obs_mtml(sensitivity=0.99, specificity=0.9, rho=0.5, M=100, R=1)

</code></pre>

<hr>
<h2 id='translink_expected_links_obs_mtsl'>Calculate expected number of observed pairs assuming multiple-transmission and single-linkage</h2><span id='topic+translink_expected_links_obs_mtsl'></span>

<h3>Description</h3>

<p>This function calculates the expected number of pairs observed in a sample of size <code>M</code>.
The multiple-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>translink_expected_links_obs_mtsl(specificity, sensitivity, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_expected_links_obs_mtsl_+3A_specificity">specificity</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_mtsl_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_mtsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_mtsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_mtsl_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of linked pairs observed in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Perfect sensitivity and specificity
translink_expected_links_obs_mtsl(sensitivity=1, specificity=1, rho=0.5, M=100, R=1)

translink_expected_links_obs_mtsl(sensitivity=0.99, specificity=0.9, rho=1, M=50, R=1)

translink_expected_links_obs_mtsl(sensitivity=0.99, specificity=0.9, rho=0.5, M=100, R=1)

</code></pre>

<hr>
<h2 id='translink_expected_links_obs_stsl'>Calculate expected number of observed pairs assuming single-transmission and single-linkage</h2><span id='topic+translink_expected_links_obs_stsl'></span>

<h3>Description</h3>

<p>This function calculates the expected number of link pairs observed in a sample of size <code>M</code>.
The single-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is linked by transmission to only one other case <code class="reqn">j</code> in the population (<code class="reqn">N</code>).
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>translink_expected_links_obs_stsl(sensitivity, specificity, rho, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_expected_links_obs_stsl_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_stsl_+3A_specificity">specificity</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_stsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_obs_stsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of linked pairs observed in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># perfect sensitivity and specificity
translink_expected_links_obs_stsl(sensitivity=1, specificity=1, rho=0.5, M=100)

translink_expected_links_obs_stsl(sensitivity=0.99, specificity=0.9, rho=1, M=50)

translink_expected_links_obs_stsl(sensitivity=0.99, specificity=0.9, rho=0.5, M=100)

</code></pre>

<hr>
<h2 id='translink_expected_links_true'>Calculate expected number of true transmission pairs</h2><span id='topic+translink_expected_links_true'></span>

<h3>Description</h3>

<p>This function calculates the expected number true transmission pairs in a sample of size <code>M</code>.
Assumptions about transmission and linkage (single or multiple) can be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translink_expected_links_true(
  sensitivity,
  rho,
  M,
  R = NULL,
  assumption = "mtml"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_expected_links_true_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen (default=NULL)</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_+3A_assumption">assumption</code></td>
<td>
<p>a character vector indicating which assumptions about transmission and linkage criteria. Default = <code>'mtml'</code>. Accepted arguments are:
</p>

<ol>
<li> <p><code>'stsl'</code> for the single-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtsl'</code> for the multiple-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtml'</code> for the multiple-transmission multiple-linkage assumption.
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of true transmission pairs in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>translink_expected_links_true(sensitivity=0.99, rho=0.75, M=100, R=1)

</code></pre>

<hr>
<h2 id='translink_expected_links_true_mtml'>Calculate expected number of true transmission pairs assuming multiple-transmission and multiple-linkage</h2><span id='topic+translink_expected_links_true_mtml'></span>

<h3>Description</h3>

<p>This function calculates the expected number of true transmission pairs in a sample of size <code>M</code>.
The multiple-transmission and multiple-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to multiple cases <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li>
<li><p> Linkage events are independent of one another (i.e, linkage of case <code class="reqn">i</code> to case <code class="reqn">j</code> has no bearing on linkage of case <code class="reqn">i</code> to any other sample).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>translink_expected_links_true_mtml(sensitivity, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_expected_links_true_mtml_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_mtml_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_mtml_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_mtml_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of true transmission pairs in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>translink_expected_links_true_mtml(sensitivity=0.95, rho=0.2, M=1000, R=1)

</code></pre>

<hr>
<h2 id='translink_expected_links_true_mtsl'>Calculate expected number of true transmission pairs assuming multiple-transmission and single-linkage</h2><span id='topic+translink_expected_links_true_mtsl'></span>

<h3>Description</h3>

<p>This function calculates the expected number true transmission pairs in a sample of size <code>M</code>.
The multiple-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>translink_expected_links_true_mtsl(sensitivity, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_expected_links_true_mtsl_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_mtsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_mtsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_mtsl_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of true transmission pairs in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>translink_expected_links_true_mtsl(sensitivity=0.95, rho=0.2, M=200, R=1)

</code></pre>

<hr>
<h2 id='translink_expected_links_true_stsl'>Calculate expected number of true transmission pairs assuming single-transmission and single-linkage</h2><span id='topic+translink_expected_links_true_stsl'></span>

<h3>Description</h3>

<p>This function calculates the expected number of true transmission pairs in a sample of size <code>M</code>.
The single-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is linked by transmission to only one other case <code class="reqn">j</code> in the population (<code class="reqn">N</code>).
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>translink_expected_links_true_stsl(sensitivity, rho, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_expected_links_true_stsl_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_stsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_expected_links_true_stsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of true transmission pairs in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>translink_expected_links_true_stsl(sensitivity=0.95, rho=0.2, M=200)

</code></pre>

<hr>
<h2 id='translink_fdr'>Calculate false discovery rate of identifying transmission pairs in a sample</h2><span id='topic+translink_fdr'></span>

<h3>Description</h3>

<p>This function calculates the false discovery rate (proportion of linked pairs that are false positives) in a sample given the sensitivity
and specificity of the linkage criteria, and sample size <code class="reqn">M</code>. Assumptions about transmission and linkage (single or multiple)
can be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translink_fdr(sensitivity, specificity, rho, M, R = NULL, assumption = "mtml")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_fdr_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_fdr_+3A_specificity">specificity</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_fdr_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_fdr_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="translink_fdr_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen (default=NULL)</p>
</td></tr>
<tr><td><code id="translink_fdr_+3A_assumption">assumption</code></td>
<td>
<p>a character vector indicating which assumptions about transmission and linkage criteria. Default = <code>'mtml'</code>. Accepted arguments are:
</p>

<ol>
<li> <p><code>'stsl'</code> for the single-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtsl'</code> for the multiple-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtml'</code> for the multiple-transmission multiple-linkage assumption.
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the true discovery rate
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The simplest case: single-transmission, single-linkage, and perfect sensitivity
translink_fdr(sensitivity=1, specificity=0.9, rho=0.5, M=100, assumption='stsl')

# Multiple-transmission and imperfect sensitivity
translink_fdr(sensitivity=0.99, specificity=0.9, rho=1, M=50, R=1, assumption='mtsl')

# Small outbreak, larger sampling proportion
translink_fdr(sensitivity=0.99, specificity=0.95, rho=1, M=50, R=1, assumption='mtml')

# Large outbreak, small sampling proportion
translink_fdr(sensitivity=0.99, specificity=0.95, rho=0.5, M=1000, R=1, assumption='mtml')

</code></pre>

<hr>
<h2 id='translink_prob_transmit'>Calculate probability of transmission</h2><span id='topic+translink_prob_transmit'></span>

<h3>Description</h3>

<p>This function calculates the probability that two cases are linked by direct transmission
given that they have been linked by phylogenetic criteria.
Assumptions about transmission and linkage (single or multiple) can be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translink_prob_transmit(
  sensitivity,
  specificity,
  rho,
  M,
  R,
  assumption = "mtml"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_prob_transmit_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_+3A_specificity">specificity</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_+3A_assumption">assumption</code></td>
<td>
<p>a character vector indicating which assumptions about transmission and linkage criteria. Default = <code>'mtml'</code>. Accepted arguments are:
</p>

<ol>
<li> <p><code>'stsl'</code> for the single-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtsl'</code> for the multiple-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtml'</code> for the multiple-transmission multiple-linkage assumption.
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the probability of transmission between two cases given linkage by phylogenetic criteria
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>translink_prob_transmit(sensitivity=0.99, specificity=0.9, rho=0.5, M=100, R=1)

</code></pre>

<hr>
<h2 id='translink_prob_transmit_mtml'>Calculate probability of transmission assuming multiple-transmission and multiple-linkage</h2><span id='topic+translink_prob_transmit_mtml'></span>

<h3>Description</h3>

<p>This function calculates the probability that two cases are linked by direct transmission
given that they have been linked by phylogenetic criteria. The multiple-transmission and multiple-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to multiple cases <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li>
<li><p> Linkage events are independent of one another (i.e, linkage of case <code class="reqn">i</code> to case <code class="reqn">j</code> has no bearing on linkage of case <code class="reqn">i</code> to any other sample).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>translink_prob_transmit_mtml(sensitivity, specificity, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_prob_transmit_mtml_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_mtml_+3A_specificity">specificity</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_mtml_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_mtml_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_mtml_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the probability of transmission between two cases given linkage by phylogenetic criteria
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Perfect sensitivity and specificity
translink_prob_transmit_mtml(sensitivity=1, specificity=1, rho=0.5, M=100, R=1)

translink_prob_transmit_mtml(sensitivity=0.99, specificity=0.9, rho=1, M=50, R=1)

translink_prob_transmit_mtml(sensitivity=0.99, specificity=0.9, rho=0.5, M=100, R=1)

</code></pre>

<hr>
<h2 id='translink_prob_transmit_mtsl'>Calculate probability of transmission assuming multiple-transmission and single-linkage</h2><span id='topic+translink_prob_transmit_mtsl'></span>

<h3>Description</h3>

<p>This function calculates the probability that two cases are linked by direct transmission
given that they have been linked by phylogenetic criteria. The multiple-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>translink_prob_transmit_mtsl(specificity, sensitivity, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_prob_transmit_mtsl_+3A_specificity">specificity</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_mtsl_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_mtsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_mtsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_mtsl_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the probability of transmission between two cases given linkage by phylogenetic criteria
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Perfect sensitivity and specificity
translink_prob_transmit_mtsl(sensitivity=1, specificity=1, rho=0.5, M=100, R=1)

translink_prob_transmit_mtsl(sensitivity=0.99, specificity=0.9, rho=1, M=50, R=1)

translink_prob_transmit_mtsl(sensitivity=0.99, specificity=0.9, rho=0.5, M=100, R=1)

</code></pre>

<hr>
<h2 id='translink_prob_transmit_stsl'>Calculate probability of transmission assuming single-transmission and single-linkage</h2><span id='topic+translink_prob_transmit_stsl'></span>

<h3>Description</h3>

<p>This function calculates the probability that two cases are linked by direct transmission
given that they have been linked by phylogenetic criteria. The single-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is linked by transmission to only one other case <code class="reqn">j</code> in the population (<code class="reqn">N</code>).
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>translink_prob_transmit_stsl(sensitivity, specificity, rho, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_prob_transmit_stsl_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_stsl_+3A_specificity">specificity</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_stsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_prob_transmit_stsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For perfect sensitivity, set <code>sensitivity = 1</code>.
</p>


<h3>Value</h3>

<p>scalar or vector giving the probability of transmission between two cases given linkage by phylogenetic criteria
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># perfect sensitivity and specificity
translink_prob_transmit_stsl(sensitivity=1, specificity=1, rho=0.2, M=100)

# perfect sensitivity only
translink_prob_transmit_stsl(sensitivity=1, specificity=0.95, rho=0.2, M=100)

translink_prob_transmit_stsl(sensitivity=0.99, specificity=0.95, rho=0.9, M=50)

translink_prob_transmit_stsl(sensitivity=0.99, specificity=0.95, rho=0.05, M=100)

</code></pre>

<hr>
<h2 id='translink_samplesize'>Calculate sample size needed to identify true transmission links</h2><span id='topic+translink_samplesize'></span>

<h3>Description</h3>

<p>This function calculates the sample size needed to identify transmission links at
a predefined false discovery rate, given a final outbreak size <code class="reqn">N</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translink_samplesize(
  sensitivity,
  specificity,
  N,
  R = NULL,
  tdr,
  min_pairs = 1,
  assumption = "mtml"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_samplesize_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_samplesize_+3A_specificity">specificity</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_samplesize_+3A_n">N</code></td>
<td>
<p>scalar or vector giving the final outbreak size</p>
</td></tr>
<tr><td><code id="translink_samplesize_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
<tr><td><code id="translink_samplesize_+3A_tdr">tdr</code></td>
<td>
<p>scalar or vector giving the desired true discovery rate (1-false discovery rate)</p>
</td></tr>
<tr><td><code id="translink_samplesize_+3A_min_pairs">min_pairs</code></td>
<td>
<p>minimum number of linked pairs observed in the sample, defaults to 1 pair (2 samples); this is to ensure reasonable results are obtained</p>
</td></tr>
<tr><td><code id="translink_samplesize_+3A_assumption">assumption</code></td>
<td>
<p>a character vector indicating which assumptions about transmission and linkage criteria. Default = <code>'mtml'</code>. Accepted arguments are:
</p>

<ol>
<li> <p><code>'stsl'</code> for the single-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtsl'</code> for the multiple-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtml'</code> for the multiple-transmission multiple-linkage assumption.
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the sample size needed to meet the given conditions
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_tdr">translink_tdr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>translink_samplesize(sensitivity=0.99, specificity=0.995, N=100, R=1, tdr=0.75)

</code></pre>

<hr>
<h2 id='translink_tdr'>Calculate true discovery rate of identifying transmission pairs</h2><span id='topic+translink_tdr'></span>

<h3>Description</h3>

<p>This function calculates the true discovery rate (proportion of true transmission pairs) in a sample given the sensitivity
and specificity of the linkage criteria, and sample size <code class="reqn">M</code>. Assumptions about transmission and linkage (single or multiple)
can be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>translink_tdr(sensitivity, specificity, rho, M, R = NULL, assumption = "mtml")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="translink_tdr_+3A_sensitivity">sensitivity</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_tdr_+3A_specificity">specificity</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="translink_tdr_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="translink_tdr_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="translink_tdr_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen (default=NULL)</p>
</td></tr>
<tr><td><code id="translink_tdr_+3A_assumption">assumption</code></td>
<td>
<p>a character vector indicating which assumptions about transmission and linkage criteria. Default = <code>'mtml'</code>. Accepted arguments are:
</p>

<ol>
<li> <p><code>'stsl'</code> for the single-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtsl'</code> for the multiple-transmission single-linkage assumption.
</p>
</li>
<li> <p><code>'mtml'</code> for the multiple-transmission multiple-linkage assumption.
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the true discovery rate
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other transmission linkage functions: 
<code><a href="#topic+translink_expected_links_obs_mtml">translink_expected_links_obs_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_mtsl">translink_expected_links_obs_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs_stsl">translink_expected_links_obs_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_obs">translink_expected_links_obs</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtml">translink_expected_links_true_mtml</a>()</code>,
<code><a href="#topic+translink_expected_links_true_mtsl">translink_expected_links_true_mtsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true_stsl">translink_expected_links_true_stsl</a>()</code>,
<code><a href="#topic+translink_expected_links_true">translink_expected_links_true</a>()</code>,
<code><a href="#topic+translink_fdr">translink_fdr</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtml">translink_prob_transmit_mtml</a>()</code>,
<code><a href="#topic+translink_prob_transmit_mtsl">translink_prob_transmit_mtsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit_stsl">translink_prob_transmit_stsl</a>()</code>,
<code><a href="#topic+translink_prob_transmit">translink_prob_transmit</a>()</code>,
<code><a href="#topic+translink_samplesize">translink_samplesize</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The simplest case: single-transmission, single-linkage, and perfect sensitivity
translink_tdr(sensitivity=1, specificity=0.9, rho=0.5, M=100, assumption='stsl')

# Multiple-transmission and imperfect sensitivity
translink_tdr(sensitivity=0.99, specificity=0.9, rho=1, M=50, R=1, assumption='mtsl')

# Small outbreak, larger sampling proportion
translink_tdr(sensitivity=0.99, specificity=0.95, rho=1, M=50, R=1, assumption='mtml')

# Large outbreak, small sampling proportion
translink_tdr(sensitivity=0.99, specificity=0.95, rho=0.5, M=1000, R=1, assumption='mtml')

</code></pre>

<hr>
<h2 id='true_pairs'>Calculate expected number of true transmission pairs</h2><span id='topic+true_pairs'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the expected number true transmission pairs in a sample of size <code>M</code>.
Assumptions about transmission and linkage (single or multiple) can be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>true_pairs(eta, rho, M, R = NULL, assumption = "mtml")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="true_pairs_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="true_pairs_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="true_pairs_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="true_pairs_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen (default=NULL)</p>
</td></tr>
<tr><td><code id="true_pairs_+3A_assumption">assumption</code></td>
<td>
<p>a character vector indicating which assumptions about transmission and linkage criteria. Default = <code>'mtml'</code>. Accepted arguments are:
</p>

<ol>
<li> <p><code>'stsl'</code> for the single-transmission single-linkage assumption (<code><a href="#topic+prob_trans_stsl">prob_trans_stsl()</a></code>).
</p>
</li>
<li> <p><code>'mtsl'</code> for the multiple-transmission single-linkage assumption (<code><a href="#topic+prob_trans_mtsl">prob_trans_mtsl()</a></code>).
</p>
</li>
<li> <p><code>'mtml'</code> for the multiple-transmission multiple-linkage assumption (<code><a href="#topic+prob_trans_mtml">prob_trans_mtml()</a></code>).
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of true transmission pairs in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other true_pairs: 
<code><a href="#topic+true_pairs_mtml">true_pairs_mtml</a>()</code>,
<code><a href="#topic+true_pairs_mtsl">true_pairs_mtsl</a>()</code>,
<code><a href="#topic+true_pairs_stsl">true_pairs_stsl</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>true_pairs(eta=0.99, rho=0.75, M=100, R=1)

</code></pre>

<hr>
<h2 id='true_pairs_mtml'>Expected number of true transmission pairs assuming multiple-transmission and multiple-linkage</h2><span id='topic+true_pairs_mtml'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the expected number of true transmission pairs in a sample of size <code>M</code>.
The multiple-transmission and multiple-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to multiple cases <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li>
<li><p> Linkage events are independent of one another (i.e, linkage of case <code class="reqn">i</code> to case <code class="reqn">j</code> has no bearing on linkage of case <code class="reqn">i</code> to any other sample).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>true_pairs_mtml(eta, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="true_pairs_mtml_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="true_pairs_mtml_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="true_pairs_mtml_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="true_pairs_mtml_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of true transmission pairs in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other true_pairs: 
<code><a href="#topic+true_pairs_mtsl">true_pairs_mtsl</a>()</code>,
<code><a href="#topic+true_pairs_stsl">true_pairs_stsl</a>()</code>,
<code><a href="#topic+true_pairs">true_pairs</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>true_pairs_mtml(eta=0.95, rho=0.2, M=1000, R=1)

</code></pre>

<hr>
<h2 id='true_pairs_mtsl'>Expected number of true transmission pairs assuming multiple-transmission and single-linkage</h2><span id='topic+true_pairs_mtsl'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the expected number true transmission pairs in a sample of size <code>M</code>.
The multiple-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is, on average, the infector of <code>R</code> cases in the population (<code class="reqn">N</code>)
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is allowed to be linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>true_pairs_mtsl(eta, rho, M, R)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="true_pairs_mtsl_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="true_pairs_mtsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="true_pairs_mtsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="true_pairs_mtsl_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of true transmission pairs in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other true_pairs: 
<code><a href="#topic+true_pairs_mtml">true_pairs_mtml</a>()</code>,
<code><a href="#topic+true_pairs_stsl">true_pairs_stsl</a>()</code>,
<code><a href="#topic+true_pairs">true_pairs</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>true_pairs_mtsl(eta=0.95, rho=0.2, M=200, R=1)

</code></pre>

<hr>
<h2 id='true_pairs_stsl'>Expected number of true transmission pairs assuming single-transmission and single-linkage</h2><span id='topic+true_pairs_stsl'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the expected number of true transmission pairs in a sample of size <code>M</code>.
The single-transmission and single-linkage method assumes the following:
</p>

<ol>
<li><p> Each case <code class="reqn">i</code> is linked by transmission to only one other case <code class="reqn">j</code> in the population (<code class="reqn">N</code>).
</p>
</li>
<li><p> Each case <code class="reqn">i</code> is linked by the linkage criteria to only one other case <code class="reqn">j</code> in the sampled population (<code class="reqn">M</code>).
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>true_pairs_stsl(eta, rho, M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="true_pairs_stsl_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="true_pairs_stsl_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="true_pairs_stsl_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the expected number of true transmission pairs in the sample
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other true_pairs: 
<code><a href="#topic+true_pairs_mtml">true_pairs_mtml</a>()</code>,
<code><a href="#topic+true_pairs_mtsl">true_pairs_mtsl</a>()</code>,
<code><a href="#topic+true_pairs">true_pairs</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>true_pairs_stsl(eta=0.95, rho=0.2, M=200)

</code></pre>

<hr>
<h2 id='truediscoveryrate'>Calculate true discovery rate of a sample</h2><span id='topic+truediscoveryrate'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a>
This function calculates the true discovery rate (proportion of true transmission pairs) in a sample given the sensitivity <code class="reqn">\eta</code>
and specificity <code class="reqn">\chi</code> of the linkage criteria, and sample size <code class="reqn">M</code>. Assumptions about transmission and linkage (single or multiple)
can be specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>truediscoveryrate(eta, chi, rho, M, R = NULL, assumption = "mtml")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="truediscoveryrate_+3A_eta">eta</code></td>
<td>
<p>scalar or vector giving the sensitivity of the linkage criteria</p>
</td></tr>
<tr><td><code id="truediscoveryrate_+3A_chi">chi</code></td>
<td>
<p>scalar or vector giving the specificity of the linkage criteria</p>
</td></tr>
<tr><td><code id="truediscoveryrate_+3A_rho">rho</code></td>
<td>
<p>scalar or vector giving the proportion of the final outbreak size that is sampled</p>
</td></tr>
<tr><td><code id="truediscoveryrate_+3A_m">M</code></td>
<td>
<p>scalar or vector giving the number of cases sampled</p>
</td></tr>
<tr><td><code id="truediscoveryrate_+3A_r">R</code></td>
<td>
<p>scalar or vector giving the effective reproductive number of the pathogen (default=NULL)</p>
</td></tr>
<tr><td><code id="truediscoveryrate_+3A_assumption">assumption</code></td>
<td>
<p>a character vector indicating which assumptions about transmission and linkage criteria. Default = <code>'mtml'</code>. Accepted arguments are:
</p>

<ol>
<li> <p><code>'stsl'</code> for the single-transmission single-linkage assumption (<code><a href="#topic+prob_trans_stsl">prob_trans_stsl()</a></code>).
</p>
</li>
<li> <p><code>'mtsl'</code> for the multiple-transmission single-linkage assumption (<code><a href="#topic+prob_trans_mtsl">prob_trans_mtsl()</a></code>).
</p>
</li>
<li> <p><code>'mtml'</code> for the multiple-transmission multiple-linkage assumption (<code><a href="#topic+prob_trans_mtml">prob_trans_mtml()</a></code>).
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar or vector giving the true discovery rate
</p>


<h3>Author(s)</h3>

<p>John Giles, Shirlee Wohl, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other discovery_rate: 
<code><a href="#topic+falsediscoveryrate">falsediscoveryrate</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The simplest case: single-transmission, single-linkage, and perfect sensitivity
truediscoveryrate(eta=1, chi=0.9, rho=0.5, M=100, assumption='stsl')

# Multiple-transmission and imperfect sensitivity
truediscoveryrate(eta=0.99, chi=0.9, rho=1, M=50, R=1, assumption='mtsl')

# Small outbreak, larger sampling proportion
truediscoveryrate(eta=0.99, chi=0.95, rho=1, M=50, R=1, assumption='mtml')

# Large outbreak, small sampling proportion
truediscoveryrate(eta=0.99, chi=0.95, rho=0.5, M=1000, R=1, assumption='mtml')

</code></pre>

<hr>
<h2 id='varfreq_cdf_logistic'>Calculate cumulative observed variant prevalence at time t given logistic growth</h2><span id='topic+varfreq_cdf_logistic'></span>

<h3>Description</h3>

<p>This function calculates the cumulative observed variant prevalence after t time steps (e.g., days)
given a logistic growth rate and initial variant prevalence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varfreq_cdf_logistic(t, p0_v1, r_v1, c_ratio = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varfreq_cdf_logistic_+3A_t">t</code></td>
<td>
<p>time step number (e.g., days) at which to calculate prevalence</p>
</td></tr>
<tr><td><code id="varfreq_cdf_logistic_+3A_p0_v1">p0_v1</code></td>
<td>
<p>initial variant prevalence (# introductions / infected population size)</p>
</td></tr>
<tr><td><code id="varfreq_cdf_logistic_+3A_r_v1">r_v1</code></td>
<td>
<p>logistic growth rate</p>
</td></tr>
<tr><td><code id="varfreq_cdf_logistic_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2. Default = 1 (no bias)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar giving the cdf of variant prevalence at time t
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other logistic growth functions: 
<code><a href="#topic+varfreq_freq_logistic">varfreq_freq_logistic</a>()</code>
</p>
<p>Other variant frequency functions: 
<code><a href="#topic+varfreq_expected_mbias">varfreq_expected_mbias</a>()</code>,
<code><a href="#topic+varfreq_freq_logistic">varfreq_freq_logistic</a>()</code>,
<code><a href="#topic+varfreq_obs_freq">varfreq_obs_freq</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>varfreq_cdf_logistic(t = 30, p0_v1 = 1/10000, r_v1 = 0.1, c_ratio = 1)

</code></pre>

<hr>
<h2 id='varfreq_expected_mbias'>Calculate multiplicative bias (observed / actual) in variant prevalence</h2><span id='topic+varfreq_expected_mbias'></span>

<h3>Description</h3>

<p>This function calculates the multiplicative bias of the observed variant proportion
relative to the actual variant proportion. This function assumes that variant 1 is the variant of concern.
This function is specific to the two-variant system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varfreq_expected_mbias(p_v1, c_ratio)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varfreq_expected_mbias_+3A_p_v1">p_v1</code></td>
<td>
<p>actual variant prevalence (proportion)</p>
</td></tr>
<tr><td><code id="varfreq_expected_mbias_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar giving the multiplicative bias of variant 1
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant frequency functions: 
<code><a href="#topic+varfreq_cdf_logistic">varfreq_cdf_logistic</a>()</code>,
<code><a href="#topic+varfreq_freq_logistic">varfreq_freq_logistic</a>()</code>,
<code><a href="#topic+varfreq_obs_freq">varfreq_obs_freq</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>varfreq_expected_mbias(p_v1 = 0.1, c_ratio = 1.1)

</code></pre>

<hr>
<h2 id='varfreq_freq_logistic'>Calculate observed variant prevalence at time t given logistic growth</h2><span id='topic+varfreq_freq_logistic'></span>

<h3>Description</h3>

<p>This function calculates the observed variant prevalence after t time steps (e.g., days)
given a logistic growth rate and initial variant prevalence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varfreq_freq_logistic(t, p0_v1, r_v1, c_ratio = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varfreq_freq_logistic_+3A_t">t</code></td>
<td>
<p>time step number (e.g., days) at which to calculate prevalence</p>
</td></tr>
<tr><td><code id="varfreq_freq_logistic_+3A_p0_v1">p0_v1</code></td>
<td>
<p>initial variant prevalence (# introductions / infected population size)</p>
</td></tr>
<tr><td><code id="varfreq_freq_logistic_+3A_r_v1">r_v1</code></td>
<td>
<p>logistic growth rate</p>
</td></tr>
<tr><td><code id="varfreq_freq_logistic_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2; default = 1 (no bias)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar giving the variant prevalence at time t
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other logistic growth functions: 
<code><a href="#topic+varfreq_cdf_logistic">varfreq_cdf_logistic</a>()</code>
</p>
<p>Other variant frequency functions: 
<code><a href="#topic+varfreq_cdf_logistic">varfreq_cdf_logistic</a>()</code>,
<code><a href="#topic+varfreq_expected_mbias">varfreq_expected_mbias</a>()</code>,
<code><a href="#topic+varfreq_obs_freq">varfreq_obs_freq</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>varfreq_freq_logistic(t = 30, p0_v1 = 1/10000, r_v1 = 0.1, c_ratio = 1)

</code></pre>

<hr>
<h2 id='varfreq_obs_freq'>Calculate observed variant prevalence</h2><span id='topic+varfreq_obs_freq'></span>

<h3>Description</h3>

<p>This function calculates the observed variant prevalence from the coefficient of detection ratio
and the actual variant prevalence. This function assumes that variant 1 is the variant of concern.
This function is specific to the two-variant system.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varfreq_obs_freq(p_v1, c_ratio)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="varfreq_obs_freq_+3A_p_v1">p_v1</code></td>
<td>
<p>actual variant prevalence (proportion)</p>
</td></tr>
<tr><td><code id="varfreq_obs_freq_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar of observed prevalence of variant 1
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant frequency functions: 
<code><a href="#topic+varfreq_cdf_logistic">varfreq_cdf_logistic</a>()</code>,
<code><a href="#topic+varfreq_expected_mbias">varfreq_expected_mbias</a>()</code>,
<code><a href="#topic+varfreq_freq_logistic">varfreq_freq_logistic</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>varfreq_obs_freq(p_v1 = 0.1, c_ratio = 1.1)

</code></pre>

<hr>
<h2 id='vartrack_cod_ratio'>Calculate the coefficient of detection ratio for two variants</h2><span id='topic+vartrack_cod_ratio'></span>

<h3>Description</h3>

<p>This function calculates the coefficient of detection ratio <code class="reqn">C_{V_1}/C_{V_2}</code> for two variants.
This function assumes that variant 1 is the variant of concern.
This function is specific to the two-variant system.
Parameters not provided are assumed to be equivalent between the two variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vartrack_cod_ratio(
  phi_v1 = 1,
  phi_v2 = 1,
  gamma_v1 = 1,
  gamma_v2 = 1,
  psi_v1 = 1,
  psi_v2 = 1,
  tau_a = 1,
  tau_s = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vartrack_cod_ratio_+3A_phi_v1">phi_v1</code></td>
<td>
<p>probability that a tested infection caused by variant 1 results in a positive test (sensitivity)</p>
</td></tr>
<tr><td><code id="vartrack_cod_ratio_+3A_phi_v2">phi_v2</code></td>
<td>
<p>probability that a tested infection caused by variant 2 results in a positive test (sensitivity)</p>
</td></tr>
<tr><td><code id="vartrack_cod_ratio_+3A_gamma_v1">gamma_v1</code></td>
<td>
<p>probability that a detected infection caused by variant 1 meets some quality threshold</p>
</td></tr>
<tr><td><code id="vartrack_cod_ratio_+3A_gamma_v2">gamma_v2</code></td>
<td>
<p>probability that a detected infection caused by variant 2 meets some quality threshold</p>
</td></tr>
<tr><td><code id="vartrack_cod_ratio_+3A_psi_v1">psi_v1</code></td>
<td>
<p>probability that an infection caused by variant 1 is asymptomatic</p>
</td></tr>
<tr><td><code id="vartrack_cod_ratio_+3A_psi_v2">psi_v2</code></td>
<td>
<p>probability that an infection caused by variant 2 is asymptomatic</p>
</td></tr>
<tr><td><code id="vartrack_cod_ratio_+3A_tau_a">tau_a</code></td>
<td>
<p>probability of testing an asymptomatic infection (any variant); note that this parameter is not required if psi_v1==psi_v2</p>
</td></tr>
<tr><td><code id="vartrack_cod_ratio_+3A_tau_s">tau_s</code></td>
<td>
<p>probability of testing a symptomatic infection (any variant); note that this parameter is not required if psi_v1==psi_v2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar giving the multiplicative bias of variant 1
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant tracking functions: 
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vartrack_cod_ratio(phi_v1=0.975, phi_v2=0.95, gamma_v1=0.8, gamma_v2=0.6)

</code></pre>

<hr>
<h2 id='vartrack_prob_detect'>Calculate the probability of detecting a variant given a sample size</h2><span id='topic+vartrack_prob_detect'></span>

<h3>Description</h3>

<p>This function calculates the probability of detecting the presence of a variant
given a sample size and sampling strategy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vartrack_prob_detect(
  n,
  t = NA,
  p_v1 = NA,
  omega,
  p0_v1 = NA,
  r_v1 = NA,
  c_ratio = 1,
  sampling_freq
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vartrack_prob_detect_+3A_n">n</code></td>
<td>
<p>sample size (either of cross-section or per timestep)</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_+3A_t">t</code></td>
<td>
<p>time step number (e.g., days) at which variant should be detected by. Default = NA (either <code>'t'</code> or <code>'p_v1'</code> should be provided, not both)</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_+3A_p_v1">p_v1</code></td>
<td>
<p>the desired prevalence to detect a variant by. Default = NA (either <code>'t'</code> or <code>'p_v1'</code> should be provided, not both)</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_+3A_omega">omega</code></td>
<td>
<p>probability of sequencing (or other characterization) success</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_+3A_p0_v1">p0_v1</code></td>
<td>
<p>initial variant prevalence (# introductions / infected population size)</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_+3A_r_v1">r_v1</code></td>
<td>
<p>logistic growth rate</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2. Default = 1 (no bias)</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_+3A_sampling_freq">sampling_freq</code></td>
<td>
<p>the sampling frequency (must be either 'xsect' or 'cont')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar of detection probability
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant detection functions: 
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>
</p>
<p>Other variant tracking functions: 
<code><a href="#topic+vartrack_cod_ratio">vartrack_cod_ratio</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Cross-sectional sampling
vartrack_prob_detect(p_v1 = 0.02, n = 100, omega = 0.8, c_ratio = 1, sampling_freq = 'xsect')

# Periodic sampling
vartrack_prob_detect(n = 158, t = 30, omega = 0.8, p0_v1 = 1/10000, 
r_v1 = 0.1, c_ratio = 1, sampling_freq = 'cont')

</code></pre>

<hr>
<h2 id='vartrack_prob_detect_cont'>Calculate probability of detecting a variant given a per-timestep sample size assuming periodic sampling</h2><span id='topic+vartrack_prob_detect_cont'></span>

<h3>Description</h3>

<p>This function calculates the probability of detecting the presence of a variant
given a sample size and either a desired maximum time until detection
or a desired prevalence by which to detect the variant by.
It assumes a periodic sampling strategy, where samples are collected at regular intervals (time steps).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vartrack_prob_detect_cont(
  n,
  t = NA,
  p_v1 = NA,
  omega,
  p0_v1,
  r_v1,
  c_ratio = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vartrack_prob_detect_cont_+3A_n">n</code></td>
<td>
<p>per-timestep (e.g., per day) sample size</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_cont_+3A_t">t</code></td>
<td>
<p>time step number (e.g., days) at which variant should be detected by. Default = NA (either <code>'t'</code> or <code>'p_v1'</code> should be provided, not both)</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_cont_+3A_p_v1">p_v1</code></td>
<td>
<p>the desired prevalence to detect a variant by. Default = NA (either <code>'t'</code> or <code>'p_v1'</code> should be provided, not both)</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_cont_+3A_omega">omega</code></td>
<td>
<p>probability of sequencing (or other characterization) success</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_cont_+3A_p0_v1">p0_v1</code></td>
<td>
<p>initial variant prevalence (# introductions / infected population size)</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_cont_+3A_r_v1">r_v1</code></td>
<td>
<p>logistic growth rate</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_cont_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2. Default = 1 (no bias)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar of detection probability
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant detection functions: 
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>
</p>
<p>Other variant tracking functions: 
<code><a href="#topic+vartrack_cod_ratio">vartrack_cod_ratio</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vartrack_prob_detect_cont(n = 158, t = 30, omega = 0.8, p0_v1 = 1/10000, r_v1 = 0.1, c_ratio = 1)

</code></pre>

<hr>
<h2 id='vartrack_prob_detect_xsect'>Calculate probability of detecting a variant assuming cross-sectional sampling</h2><span id='topic+vartrack_prob_detect_xsect'></span>

<h3>Description</h3>

<p>This function calculates the probability of detecting the presence of a variant
given a sample size and assuming a single, cross-sectional sample of detected infections.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vartrack_prob_detect_xsect(p_v1, n, omega, c_ratio = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vartrack_prob_detect_xsect_+3A_p_v1">p_v1</code></td>
<td>
<p>variant prevalence (proportion)</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_xsect_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_xsect_+3A_omega">omega</code></td>
<td>
<p>probability of sequencing (or other characterization) success</p>
</td></tr>
<tr><td><code id="vartrack_prob_detect_xsect_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2. Default = 1 (no bias)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar of expected sample size
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant detection functions: 
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>
</p>
<p>Other variant tracking functions: 
<code><a href="#topic+vartrack_cod_ratio">vartrack_cod_ratio</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vartrack_prob_detect_xsect(p_v1 = 0.02, n = 100, omega = 0.8, c_ratio = 1)

</code></pre>

<hr>
<h2 id='vartrack_prob_prev'>Calculate confidence in a variant estimate given a sample size</h2><span id='topic+vartrack_prob_prev'></span>

<h3>Description</h3>

<p>This function calculates the probability of accurately estimating variant prevalence
given a sample size and desired precision in the variant prevalence estimate.
Currently, only cross-sectional sampling is supported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vartrack_prob_prev(p_v1, n, omega, precision, c_ratio = 1, sampling_freq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vartrack_prob_prev_+3A_p_v1">p_v1</code></td>
<td>
<p>variant prevalence (proportion)</p>
</td></tr>
<tr><td><code id="vartrack_prob_prev_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="vartrack_prob_prev_+3A_omega">omega</code></td>
<td>
<p>probability of sequencing (or other characterization) success</p>
</td></tr>
<tr><td><code id="vartrack_prob_prev_+3A_precision">precision</code></td>
<td>
<p>desired precision in variant prevalence estimate</p>
</td></tr>
<tr><td><code id="vartrack_prob_prev_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2. Default = 1 (no bias)</p>
</td></tr>
<tr><td><code id="vartrack_prob_prev_+3A_sampling_freq">sampling_freq</code></td>
<td>
<p>the sampling frequency (must be either 'xsect' in current implementation)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar of expected sample size
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant prevalence estimation functions: 
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>
<p>Other variant tracking functions: 
<code><a href="#topic+vartrack_cod_ratio">vartrack_cod_ratio</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vartrack_prob_prev(p_v1 = 0.1, n = 200, omega = 0.8, precision = 0.1, 
c_ratio = 1, sampling_freq = 'xsect')

</code></pre>

<hr>
<h2 id='vartrack_prob_prev_xsect'>Calculate confidence in a variant estimate assuming cross-sectional sampling</h2><span id='topic+vartrack_prob_prev_xsect'></span>

<h3>Description</h3>

<p>This function calculates the probability of accurately estimating variant prevalence
given a given a sample size and desired precision in the variant prevalence estimate,
and assuming a single, cross-sectional sample of detected infections.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vartrack_prob_prev_xsect(p_v1, n, omega, precision, c_ratio = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vartrack_prob_prev_xsect_+3A_p_v1">p_v1</code></td>
<td>
<p>variant prevalence (proportion)</p>
</td></tr>
<tr><td><code id="vartrack_prob_prev_xsect_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="vartrack_prob_prev_xsect_+3A_omega">omega</code></td>
<td>
<p>probability of sequencing (or other characterization) success</p>
</td></tr>
<tr><td><code id="vartrack_prob_prev_xsect_+3A_precision">precision</code></td>
<td>
<p>desired precision in variant prevalence estimate</p>
</td></tr>
<tr><td><code id="vartrack_prob_prev_xsect_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2. Default = 1 (no bias)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar of expected sample size
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant prevalence estimation functions: 
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>
<p>Other variant tracking functions: 
<code><a href="#topic+vartrack_cod_ratio">vartrack_cod_ratio</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vartrack_prob_prev_xsect(p_v1 = 0.1, n = 200, precision = 0.1, omega = 0.8, c_ratio = 1)

</code></pre>

<hr>
<h2 id='vartrack_samplesize_detect'>Calculate sample size needed for variant detection given a desired probability of detection</h2><span id='topic+vartrack_samplesize_detect'></span>

<h3>Description</h3>

<p>This function calculates the sample size needed for detecting the presence of a variant
given a desired probability of detection and sampling strategy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vartrack_samplesize_detect(
  prob,
  t = NA,
  p_v1 = NA,
  omega,
  p0_v1 = NA,
  r_v1 = NA,
  c_ratio = 1,
  sampling_freq
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vartrack_samplesize_detect_+3A_prob">prob</code></td>
<td>
<p>desired probability of detection</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_+3A_t">t</code></td>
<td>
<p>time step number (e.g., days) at which variant should be detected by. Default = NA (either <code>'t'</code> or <code>'p_v1'</code> should be provided, not both)</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_+3A_p_v1">p_v1</code></td>
<td>
<p>the desired prevalence to detect a variant by. Default = NA (either <code>'t'</code> or <code>'p_v1'</code> should be provided, not both)</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_+3A_omega">omega</code></td>
<td>
<p>probability of sequencing (or other characterization) success</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_+3A_p0_v1">p0_v1</code></td>
<td>
<p>initial variant prevalence (# introductions / infected population size)</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_+3A_r_v1">r_v1</code></td>
<td>
<p>logistic growth rate</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2. Default = 1 (no bias)</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_+3A_sampling_freq">sampling_freq</code></td>
<td>
<p>the sampling frequency (must be either 'xsect' or 'cont')</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar of expected sample size
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant detection functions: 
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>
</p>
<p>Other variant tracking functions: 
<code><a href="#topic+vartrack_cod_ratio">vartrack_cod_ratio</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Cross-sectional sampling
vartrack_samplesize_detect(p_v1 = 0.1, prob = 0.95, omega = 0.8,
                           c_ratio = 1, sampling_freq = 'xsect')

# Periodic sampling
vartrack_samplesize_detect(prob = 0.95, t = 30, omega = 0.8, p0_v1 = 1/10000,
                           r_v1 = 0.1, c_ratio = 1, sampling_freq = 'cont')

</code></pre>

<hr>
<h2 id='vartrack_samplesize_detect_cont'>Calculate sample size needed for variant detection assuming periodic sampling</h2><span id='topic+vartrack_samplesize_detect_cont'></span>

<h3>Description</h3>

<p>This function calculates the sample size needed for detecting the presence of a variant
given a desired probability of detection and either a desired maximum time until detection
or a desired prevalence by which to detect the variant by.
It assumes a periodic sampling strategy, where samples are collected at regular intervals (time steps).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vartrack_samplesize_detect_cont(
  prob,
  t = NA,
  p_v1 = NA,
  omega,
  p0_v1,
  r_v1,
  c_ratio = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vartrack_samplesize_detect_cont_+3A_prob">prob</code></td>
<td>
<p>desired probability of detection</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_cont_+3A_t">t</code></td>
<td>
<p>time step number (e.g., days) at which variant should be detected by. Default = NA (either <code>'t'</code> or <code>'p_v1'</code> should be provided, not both)</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_cont_+3A_p_v1">p_v1</code></td>
<td>
<p>the desired prevalence to detect a variant by. Default = NA (either <code>'t'</code> or <code>'p_v1'</code> should be provided, not both)</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_cont_+3A_omega">omega</code></td>
<td>
<p>probability of sequencing (or other characterization) success</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_cont_+3A_p0_v1">p0_v1</code></td>
<td>
<p>initial variant prevalence (# introductions / infected population size)</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_cont_+3A_r_v1">r_v1</code></td>
<td>
<p>logistic growth rate</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_cont_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2. Default = 1 (no bias)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar of expected sample size
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant detection functions: 
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>
</p>
<p>Other variant tracking functions: 
<code><a href="#topic+vartrack_cod_ratio">vartrack_cod_ratio</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vartrack_samplesize_detect_cont(prob = 0.95, t = 30, omega = 0.8, 
p0_v1 = 1/10000, r_v1 = 0.1, c_ratio = 1)

</code></pre>

<hr>
<h2 id='vartrack_samplesize_detect_xsect'>Calculate sample size needed for variant detection assuming cross-sectional sampling</h2><span id='topic+vartrack_samplesize_detect_xsect'></span>

<h3>Description</h3>

<p>This function calculates the sample size needed for detecting the presence of a variant
given a desired probability of detection and assuming a single, cross-sectional sample of detected infections.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vartrack_samplesize_detect_xsect(p_v1, prob, omega, c_ratio = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vartrack_samplesize_detect_xsect_+3A_p_v1">p_v1</code></td>
<td>
<p>variant prevalence (proportion)</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_xsect_+3A_prob">prob</code></td>
<td>
<p>desired probability of detection</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_xsect_+3A_omega">omega</code></td>
<td>
<p>probability of sequencing (or other characterization) success</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_detect_xsect_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2. Default = 1 (no bias)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar of expected sample size
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant detection functions: 
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>
</p>
<p>Other variant tracking functions: 
<code><a href="#topic+vartrack_cod_ratio">vartrack_cod_ratio</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vartrack_samplesize_detect_xsect(p_v1 = 0.1, prob = 0.95, omega = 0.8, c_ratio = 1)

</code></pre>

<hr>
<h2 id='vartrack_samplesize_prev'>Calculate sample size needed for estimating variant prevalence given a desired confidence</h2><span id='topic+vartrack_samplesize_prev'></span>

<h3>Description</h3>

<p>This function calculates the sample size needed for estimating variant prevalence
given a desired confidence and desired precision in the variant prevalence estimate.
Currently, only cross-sectional sampling is supported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vartrack_samplesize_prev(
  p_v1,
  prob,
  precision,
  omega,
  c_ratio = 1,
  sampling_freq
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vartrack_samplesize_prev_+3A_p_v1">p_v1</code></td>
<td>
<p>variant prevalence (proportion)</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_prev_+3A_prob">prob</code></td>
<td>
<p>desired confidence in variant prevalence estimate</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_prev_+3A_precision">precision</code></td>
<td>
<p>desired precision in variant prevalence estimate</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_prev_+3A_omega">omega</code></td>
<td>
<p>probability of sequencing (or other characterization) success</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_prev_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2. Default = 1 (no bias)</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_prev_+3A_sampling_freq">sampling_freq</code></td>
<td>
<p>the sampling frequency (must be 'xsect' in current implementation)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar of sample size
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant prevalence estimation functions: 
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>
</p>
<p>Other variant tracking functions: 
<code><a href="#topic+vartrack_cod_ratio">vartrack_cod_ratio</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev_xsect">vartrack_samplesize_prev_xsect</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vartrack_samplesize_prev(p_v1 = 0.1, prob = 0.95, precision = 0.25, 
omega = 0.8, c_ratio = 1, sampling_freq = 'xsect')

</code></pre>

<hr>
<h2 id='vartrack_samplesize_prev_xsect'>Calculate sample size needed for variant prevalence estimation under cross-sectional sampling</h2><span id='topic+vartrack_samplesize_prev_xsect'></span>

<h3>Description</h3>

<p>This function calculates the sample size needed for estimating variant prevalence
given a desired confidence and desired precision in the variant prevalence estimate
and assuming a single, cross-sectional sample of detected infections.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vartrack_samplesize_prev_xsect(p_v1, prob, precision, omega, c_ratio = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vartrack_samplesize_prev_xsect_+3A_p_v1">p_v1</code></td>
<td>
<p>variant prevalence (proportion)</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_prev_xsect_+3A_prob">prob</code></td>
<td>
<p>desired confidence in variant prevalence estimate</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_prev_xsect_+3A_precision">precision</code></td>
<td>
<p>desired precision in variant prevalence estimate</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_prev_xsect_+3A_omega">omega</code></td>
<td>
<p>probability of sequencing (or other characterization) success</p>
</td></tr>
<tr><td><code id="vartrack_samplesize_prev_xsect_+3A_c_ratio">c_ratio</code></td>
<td>
<p>coefficient of detection ratio, calculated as the ratio of the coefficients of variant 1 to variant 2. Default = 1 (no bias)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scalar of sample size
</p>


<h3>Author(s)</h3>

<p>Shirlee Wohl, Elizabeth C. Lee, Bethany L. DiPrete, and Justin Lessler
</p>


<h3>See Also</h3>

<p>Other variant prevalence estimation functions: 
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>
<p>Other variant tracking functions: 
<code><a href="#topic+vartrack_cod_ratio">vartrack_cod_ratio</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_cont">vartrack_prob_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_prob_detect_xsect">vartrack_prob_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_detect">vartrack_prob_detect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev_xsect">vartrack_prob_prev_xsect</a>()</code>,
<code><a href="#topic+vartrack_prob_prev">vartrack_prob_prev</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_cont">vartrack_samplesize_detect_cont</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect_xsect">vartrack_samplesize_detect_xsect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_detect">vartrack_samplesize_detect</a>()</code>,
<code><a href="#topic+vartrack_samplesize_prev">vartrack_samplesize_prev</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vartrack_samplesize_prev_xsect(p_v1 = 0.1, prob = 0.95, precision = 0.25, omega = 0.8, c_ratio = 1)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
