<!DOCTYPE html><html><head><title>Help for package phylotools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {phylotools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#clean.fasta.name'>
<p>Clean the name of a fasta file</p></a></li>
<li><a href='#dat2fasta'>
<p>Convert and Save sequence data frame to fasta file</p></a></li>
<li><a href='#dat2phylip'>
<p>Conver the data frame to sequential PHYLIP format file</p></a></li>
<li><a href='#get.fasta.name'>
<p>get the names of all the sequences of fasta file</p></a></li>
<li><a href='#get.phylip.name'>
<p>get the names of sequences from a PHYLIP file</p></a></li>
<li><a href='#phylotools-package'>
<p>Phylogenetic tools for building PHYLIP supermatrix and more</p></a></li>
<li><a href='#read.fasta'>
<p>Read FASTA file</p></a></li>
<li><a href='#read.phylip'>
<p>read phylip file</p></a></li>
<li><a href='#rename.fasta'>
<p>Rename the sequences for a fasta file</p></a></li>
<li><a href='#rm.sequence.fasta'>
<p>Delete sequences from fasta file</p></a></li>
<li><a href='#split_dat'>
<p>grouping the data frame containing sequences and names and generate fasta file</p></a></li>
<li><a href='#sub.taxa.label'>
<p>Substitute the tip labels of a phylogenetic tree</p></a></li>
<li><a href='#supermat'>
<p>Build PHYLIP supermatrix and RAxML partition file using aligned FASTA or PHYLIP files.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Phylogenetic Tools for Eco-Phylogenetics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-12-08</td>
</tr>
<tr>
<td>Author:</td>
<td>Jinlong Zhang [aut, cre],
  Nancai Pei [ctb],
  Xiangcheng Mi [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of tools for building RAxML supermatrix using
             PHYLIP or aligned FASTA files. These functions will be
             useful for building large phylogenies using multiple markers.</td>
</tr>
<tr>
<td>Depends:</td>
<td>ape</td>
</tr>
<tr>
<td>Suggests:</td>
<td>vegan</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/helixcn/phylotools">https://github.com/helixcn/phylotools</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-12-09 02:19:12 UTC; jinlong</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-12-10 12:34:27 UTC</td>
</tr>
</table>
<hr>
<h2 id='clean.fasta.name'>
Clean the name of a fasta file
</h2><span id='topic+clean.fasta.name'></span>

<h3>Description</h3>

<p>Cleaning the names of sequences for a fasta file. The punctuation characters and the white space will be replaced with &quot;_&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean.fasta.name(infile = NULL, outfile = "name_cleaned.fasta")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clean.fasta.name_+3A_infile">infile</code></td>
<td>

<p>character string representing the name of the fasta file.
</p>
</td></tr>
<tr><td><code id="clean.fasta.name_+3A_outfile">outfile</code></td>
<td>

<p>Character string representing the file name to be generated.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Punctuation characters and white space will be replaced by &quot;_&quot;. More information can be found at <code><a href="base.html#topic+regex">regex</a></code>.
</p>


<h3>Value</h3>

<p>This is a subroutine without a return value. A fasta file with all the names of sequences renamed will be saved to the working directory.
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>References</h3>

<p><a href="http://www.genomatix.de/online_help/help/sequence_formats.html">http://www.genomatix.de/online_help/help/sequence_formats.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fasta">read.fasta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  cat(
    "&gt;seq_1*66",  "--TTACAAATTGACTTATTATA",
    "&gt;seq_2()r",  "GATTACAAATTGACTTATTATA",
    "&gt;seq_3:test",  "GATTACAAATTGACTTATTATA",
    "&gt;seq_588",  "GATTACAAATTGACTTATTATA",
    "&gt;seq_8$$yu",  "GATTACAAATTGACTTATTATA",
    "&gt;seq_10", "---TACAAATTGAATTATTATA",
    file = "matk.fasta", sep = "\n")

  clean.fasta.name(infile = "matk.fasta")
  get.fasta.name("name_cleaned.fasta")

  # Delete file
  unlink("matk.fasta")
  unlink("name_cleaned.fasta")
</code></pre>

<hr>
<h2 id='dat2fasta'>
Convert and Save sequence data frame to fasta file
</h2><span id='topic+dat2fasta'></span>

<h3>Description</h3>

<p>Convert and Save sequence data frame to fasta file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat2fasta(dat, outfile = "out.fasta")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dat2fasta_+3A_dat">dat</code></td>
<td>

<p>data frame by <code><a href="#topic+read.phylip">read.phylip</a></code> or <code><a href="#topic+read.fasta">read.fasta</a></code>
</p>
</td></tr>
<tr><td><code id="dat2fasta_+3A_outfile">outfile</code></td>
<td>

<p>a character string, representing the name of the fasta file to be generated
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The column of the data frame must be: 1. seq.name, 2. seq.text, represent the name of the sequences, the content of the sequence, eg. ATCGGGAAC.
</p>


<h3>Value</h3>

<p>This is a routine without return value.
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>References</h3>

<p><a href="http://www.genomatix.de/online_help/help/sequence_formats.html">http://www.genomatix.de/online_help/help/sequence_formats.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fasta">read.fasta</a></code>,<code><a href="#topic+read.phylip">read.phylip</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
cat(
"&gt;seq_2", "GTCTTATAAGAAAGAATAAGAAAG--AAATACAAA-------AAAAAAGA",
"&gt;seq_3", "GTCTTATAAGAAAGAAATAGAAAAGTAAAAAAAAA-------AAAAAAAG",
"&gt;seq_5", "GACATAAGACATAAAATAGAATACTCAATCAGAAACCAACCCATAAAAAC",
"&gt;seq_8", "ATTCCAAAATAAAATACAAAAAGAAAAAACTAGAAAGTTTTTTTTCTTTG",
"&gt;seq_9", "ATTCTTTGTTCTTTTTTTTCTTTAATCTTTAAATAAACCTTTTTTTTTTA",
file = "trn1.fasta", sep = "\n")

res &lt;- read.fasta("trn1.fasta")
dat2fasta(res)
unlink("trn1.fasta")
unlink("out.fasta")
</code></pre>

<hr>
<h2 id='dat2phylip'>
Conver the data frame to sequential PHYLIP format file
</h2><span id='topic+dat2phylip'></span>

<h3>Description</h3>

<p>Convert and save a data frame to sequential PHYLIP file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat2phylip(dat, outfile = "out.phy")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dat2phylip_+3A_dat">dat</code></td>
<td>

<p>the data frame returned by  <code><a href="#topic+read.phylip">read.phylip</a></code>,  <code><a href="#topic+read.fasta">read.fasta</a></code>.
</p>
</td></tr>
<tr><td><code id="dat2phylip_+3A_outfile">outfile</code></td>
<td>

<p>character string represents the phylip file to be generated.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output will be in sequential PHYLIP format.
</p>


<h3>Value</h3>

<p>This is a subroutine, there is no return value.
</p>


<h3>Note</h3>

<p>The names of the sequences should not contain white space or Punctuation characters. See <code><a href="base.html#topic+regex">regex</a></code> for more details.
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>References</h3>

<p><a href="http://www.genomatix.de/online_help/help/sequence_formats.html">http://www.genomatix.de/online_help/help/sequence_formats.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dat2fasta">dat2fasta</a></code>, <code><a href="#topic+read.fasta">read.fasta</a></code>, <code><a href="#topic+read.phylip">read.phylip</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  cat(
  "&gt;seq_2", "GTCTTATAAGAAAGAATAAGAAAG--AAATACAAA-------AAAAAAGA",
  "&gt;seq_3", "GTCTTATAAGAAAGAAATAGAAAAGTAAAAAAAAA-------AAAAAAAG",
  "&gt;seq_5", "GACATAAGACATAAAATAGAATACTCAATCAGAAACCAACCCATAAAAAC",
  "&gt;seq_8", "ATTCCAAAATAAAATACAAAAAGAAAAAACTAGAAAGTTTTTTTTCTTTG",
  "&gt;seq_9", "ATTCTTTGTTCTTTTTTTTCTTTAATCTTTAAATAAACCTTTTTTTTTTA",
  file = "trn1.fasta", sep = "\n")

res &lt;- read.fasta("trn1.fasta")
dat2phylip(res)
unlink("trn1.fasta")
unlink("out.phy")
</code></pre>

<hr>
<h2 id='get.fasta.name'>
get the names of all the sequences of fasta file
</h2><span id='topic+get.fasta.name'></span>

<h3>Description</h3>

<p>get the names of all the sequences of a fasta file, and perform cleaning of the names of the sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.fasta.name(infile, clean_name = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.fasta.name_+3A_infile">infile</code></td>
<td>

<p>character string representing the name of the fasta file.
</p>
</td></tr>
<tr><td><code id="get.fasta.name_+3A_clean_name">clean_name</code></td>
<td>

<p>logical, representing cleaning of the names will be performed.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector containing the names of the sequences
</p>


<h3>Note</h3>

<p>Punctuation characters and white space be replaced by &quot;_&quot;.  Definition of Punctuation characters can be found at <code><a href="base.html#topic+regex">regex</a></code>.
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>References</h3>

<p><a href="http://www.genomatix.de/online_help/help/sequence_formats.html">http://www.genomatix.de/online_help/help/sequence_formats.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fasta">read.fasta</a></code>, <code><a href="base.html#topic+regex">regex</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  cat(
  "&gt;seq_2", "GTCTTATAAGAAAGAATAAGAAAG--AAATACAAA-------AAAAAAGA",
  "&gt;seq_3", "GTCTTATAAGAAAGAAATAGAAAAGTAAAAAAAAA-------AAAAAAAG",
  "&gt;seq_5", "GACATAAGACATAAAATAGAATACTCAATCAGAAACCAACCCATAAAAAC",
  "&gt;seq_8", "ATTCCAAAATAAAATACAAAAAGAAAAAACTAGAAAGTTTTTTTTCTTTG",
  "&gt;seq_9", "ATTCTTTGTTCTTTTTTTTCTTTAATCTTTAAATAAACCTTTTTTTTTTA",
  file = "trn1.fasta", sep = "\n")
get.fasta.name("trn1.fasta")
unlink("trn1.fasta")
</code></pre>

<hr>
<h2 id='get.phylip.name'>
get the names of sequences from a PHYLIP file
</h2><span id='topic+get.phylip.name'></span>

<h3>Description</h3>

<p>get the names of sequences from a PHYLIP file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.phylip.name(infile, clean_name = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.phylip.name_+3A_infile">infile</code></td>
<td>

<p>character representing the name or path of the phylip file.
</p>
</td></tr>
<tr><td><code id="get.phylip.name_+3A_clean_name">clean_name</code></td>
<td>

<p>logical, representing cleaning of the names will be performed.  </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Punctuation characters and white space be replaced by &quot;_&quot;. Definition of Punctuation characters can be found at <code><a href="base.html#topic+regex">regex</a></code>.
</p>


<h3>Value</h3>

<p>a character vector of the names of the sequences
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.phylip">read.phylip</a></code>, <code><a href="base.html#topic+regex">regex</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  cat("6 22",
  "seq_1    --TTACAAATTGACTTATTATA",
  "seq_2    GATTACAAATTGACTTATTATA",
  "seq_3    GATTACAAATTGACTTATTATA",
  "seq_5    GATTACAAATTGACTTATTATA",
  "seq_8    GATTACAAATTGACTTATTATA",
  "seq_10   ---TACAAATTGAATTATTATA",
  file = "matk.phy", sep = "\n")
get.phylip.name("matk.phy")
unlink("matk.phy")
</code></pre>

<hr>
<h2 id='phylotools-package'>
Phylogenetic tools for building PHYLIP supermatrix and more
</h2><span id='topic+phylotools-package'></span><span id='topic+phylotools'></span>

<h3>Description</h3>

<p>A collection of a few functions for handling DNA-barcoding sequences,
building PHYLIP supermatrix for RAxML etc.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> phylotools</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.2.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2017-12-09</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GLP-2</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Jinlong Zhang
</p>
<p>Maintainer: Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>

<hr>
<h2 id='read.fasta'>
Read FASTA file
</h2><span id='topic+read.fasta'></span>

<h3>Description</h3>

<p>Read and convert the fasta file to data frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.fasta(file = NULL, clean_name = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.fasta_+3A_file">file</code></td>
<td>

<p>character string representing the name of the fasta file.
</p>
</td></tr>
<tr><td><code id="read.fasta_+3A_clean_name">clean_name</code></td>
<td>

<p>logical, representing cleaning of the names will be performed. Punctuation characters and white space be replaced by &quot;_&quot; . See <code><a href="base.html#topic+regex">regex</a></code> for more details.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In this function, names of the sequences are identified by &quot;&gt;&quot;, and all the lines before next &quot;&gt;&quot; will be concatenated.
</p>


<h3>Value</h3>

<p>a data frame with two columns:
(1) seq.name, the names for all the sequences.
(2) seq.text, the raw sequence data.
</p>


<h3>Note</h3>

<p>Punctuation characters and white space in the names of the sequences will be replaced by &quot;_&quot;.
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>References</h3>

<p><a href="http://www.genomatix.de/online_help/help/sequence_formats.html">http://www.genomatix.de/online_help/help/sequence_formats.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.phylip">read.phylip</a></code>,<code><a href="#topic+dat2fasta">dat2fasta</a></code>,<code><a href="#topic+dat2phylip">dat2phylip</a></code>,<code><a href="#topic+split_dat">split_dat</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
cat(
"&gt;seq_2", "GTCTTATAAGAAAGAATAAGAAAG--AAATACAAA-------AAAAAAGA",
"&gt;seq_3", "GTCTTATAAGAAAGAAATAGAAAAGTAAAAAAAAA-------AAAAAAAG",
"&gt;seq_5", "GACATAAGACATAAAATAGAATACTCAATCAGAAACCAACCCATAAAAAC",
"&gt;seq_8", "ATTCCAAAATAAAATACAAAAAGAAAAAACTAGAAAGTTTTTTTTCTTTG",
"&gt;seq_9", "ATTCTTTGTTCTTTTTTTTCTTTAATCTTTAAATAAACCTTTTTTTTTTA",
file = "trn1.fasta", sep = "\n")

res &lt;- read.fasta("trn1.fasta")
unlink("trn1.fasta")
</code></pre>

<hr>
<h2 id='read.phylip'>
read phylip file
</h2><span id='topic+read.phylip'></span>

<h3>Description</h3>

<p>read the phylip file, and store the sequences and their names in data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.phylip(infile, clean_name = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.phylip_+3A_infile">infile</code></td>
<td>

<p>character string for the name of the phylip file.
</p>
</td></tr>
<tr><td><code id="read.phylip_+3A_clean_name">clean_name</code></td>
<td>

<p>logical, representing cleaning of the names will be performed. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>read.phylip accepts both interleaved and sequential phylip, the number of sequences is identified by parsing the first line of the file. Sequences and their names will be stored in a data frame.
</p>
<p>If clean_name is TRUE, punctuation characters and white space be replaced by &quot;_&quot;. Definition of punctuation characters can be found at <code><a href="base.html#topic+regex">regex</a></code>.
</p>


<h3>Value</h3>

<p>a data frame with two columns:
(1) seq.name, the names for all the sequences;
(2) seq.text, the raw sequence data.
</p>


<h3>Note</h3>

<p>the Punctuation characters and white space in the names of the sequences will be replaced by &quot;_&quot;.
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fasta">read.fasta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  cat("6 22",
  "seq_1    --TTACAAATTGACTTATTATA",
  "seq_2    GATTACAAATTGACTTATTATA",
  "seq_3    GATTACAAATTGACTTATTATA",
  "seq_5    GATTACAAATTGACTTATTATA",
  "seq_8    GATTACAAATTGACTTATTATA",
  "seq_10   ---TACAAATTGAATTATTATA",
  file = "matk.phy", sep = "\n")

res &lt;- read.phylip(infile = "matk.phy")
unlink("matk.phy")
</code></pre>

<hr>
<h2 id='rename.fasta'>
Rename the sequences for a fasta file
</h2><span id='topic+rename.fasta'></span>

<h3>Description</h3>

<p>Rename the sequences within a fasta file according to a data frame supplied.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rename.fasta(infile = NULL, ref_table, outfile = "renamed.fasta")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rename.fasta_+3A_infile">infile</code></td>
<td>

<p>character string containing the name of the fasta file.
</p>
</td></tr>
<tr><td><code id="rename.fasta_+3A_ref_table">ref_table</code></td>
<td>

<p>a data frame with first column for original name, second column for the new name of the sequence.
</p>
</td></tr>
<tr><td><code id="rename.fasta_+3A_outfile">outfile</code></td>
<td>

<p>The name of the fasta file with sequences renamed.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the orginal name was not found in the ref_table, the name for the sequence will be changed into &quot;old_name_&quot; + orginal name.
</p>


<h3>Value</h3>

<p>This is a subroutine without return value.
</p>


<h3>Note</h3>

<p>Since whitespace and punctuation characters will be replaced with &quot;_&quot;,  name of a sequence might change. It is suggest to obtain the name of the sequences by calling read.fasta first, and save the data.frame to a csv file to obtain the &quot;original&quot; name for the sequences.
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>References</h3>

<p><a href="http://www.genomatix.de/online_help/help/sequence_formats.html">http://www.genomatix.de/online_help/help/sequence_formats.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fasta">read.fasta</a></code>, <code><a href="#topic+split_dat">split_dat</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
cat(
    "&gt;seq_1",  "--TTACAAATTGACTTATTATA",
    "&gt;seq_2",  "GATTACAAATTGACTTATTATA",
    "&gt;seq_3",  "GATTACAAATTGACTTATTATA",
    "&gt;seq_5",  "GATTACAAATTGACTTATTATA",
    "&gt;seq_8",  "GATTACAAATTGACTTATTATA",
    "&gt;seq_10", "---TACAAATTGAATTATTATA",
    file = "matk.fasta", sep = "\n")
old_name &lt;- get.fasta.name("matk.fasta")
new_name &lt;- c("Magnolia", "Ranunculus", "Carex", "Morus", "Ulmus", "Salix")
ref2 &lt;- data.frame(old_name, new_name)
rename.fasta(infile = "matk.fasta", ref_table = ref2, outfile = "renamed.fasta")
unlink("matk.fasta")
unlink("renamed.fasta")
</code></pre>

<hr>
<h2 id='rm.sequence.fasta'>
Delete sequences from fasta file
</h2><span id='topic+rm.sequence.fasta'></span>

<h3>Description</h3>

<p>Delete sequences from fasta file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rm.sequence.fasta(infile, outfile = "sequence.removed.fasta", to.rm = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rm.sequence.fasta_+3A_infile">infile</code></td>
<td>

<p>Character string representing the name of the fasta file.
</p>
</td></tr>
<tr><td><code id="rm.sequence.fasta_+3A_outfile">outfile</code></td>
<td>

<p>Character string representing the name of the output fasta file.
</p>
</td></tr>
<tr><td><code id="rm.sequence.fasta_+3A_to.rm">to.rm</code></td>
<td>

<p>Vector of character string containing the names of sequences to be deleted.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Delete sequences from a fasta file.
</p>


<h3>Value</h3>

<p>This is a subroutine without return value.
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>References</h3>

<p><a href="http://www.genomatix.de/online_help/help/sequence_formats.html">http://www.genomatix.de/online_help/help/sequence_formats.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fasta">read.fasta</a></code>, <code><a href="#topic+dat2fasta">dat2fasta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
cat(
"&gt;seq_1",  "---TCCGCCCCCCTACTCTA",
"&gt;seq_3",  "CTCTCCGCCCCTCTACTCTA",
"&gt;seq_5",  "---TCCGCCC-TTTACTCTA",
"&gt;seq_6",  "---TCCGCCCCTCTACTCTA",
"&gt;seq_9",  "---TCCGCCC-TCTACTCTA",
"&gt;seq_12", "CTCTCCGCCC-TCTACTCTA",
file = "trn2.fasta", sep = "\n")

rm.sequence.fasta(infile = "trn2.fasta", to.rm = c("seq_1","seq_12"))

unlink("trn2.fasta")
unlink("sequence.removed.fasta")
</code></pre>

<hr>
<h2 id='split_dat'>
grouping the data frame containing sequences and names and generate fasta file
</h2><span id='topic+split_dat'></span>

<h3>Description</h3>

<p>Splite the data frame of sequences based on the reference table of grouping.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>split_dat(dat, ref_table)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="split_dat_+3A_dat">dat</code></td>
<td>

<p>data frame generated by <code><a href="#topic+read.phylip">read.phylip</a></code> or <code><a href="#topic+read.fasta">read.fasta</a></code>
</p>
</td></tr>
<tr><td><code id="split_dat_+3A_ref_table">ref_table</code></td>
<td>

<p>data frame with first column for the name of the sequence, second column for the group the sequence belongs to.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each group of sequences will be saved to a fasta file. Sequences not included in the ref_table will be saved in &quot;Ungrouped.fasta&quot;
</p>


<h3>Value</h3>

<p>This is a subroutine, there is no return value.
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>References</h3>

<p><a href="http://www.genomatix.de/online_help/help/sequence_formats.html">http://www.genomatix.de/online_help/help/sequence_formats.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rename.fasta">rename.fasta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  cat(
  "&gt;seq_1",   "--TTACAAATTGACTTATTATA",
  "&gt;seq_2",   "GATTACAAATTGACTTATTATA",
  "&gt;seq_3",   "GATTACAAATTGACTTATTATA",
  "&gt;seq_5",   "GATTACAAATTGACTTATTATA",
  "&gt;seq_8",   "GATTACAAATTGACTTATTATA",
  "&gt;seq_10",  "---TACAAATTGAATTATTATA",
  "&gt;seq_11",  "--TTACAAATTGACTTATTATA",
  "&gt;seq_12",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_13",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_15",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_16",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_17",  "---TACAAATTGAATTATTATA",
  file = "trnh.fasta", sep = "\n")

sequence_name &lt;- get.fasta.name("trnh.fasta")
sequence_group &lt;- c("group1","group1","group1","group1","group1",
"group2","group2","group2","group3","group3","group3","group3")
group &lt;- data.frame(sequence_name, sequence_group)

fasta &lt;- read.fasta("trnh.fasta")
split_dat(fasta, group)

unlink("trnh.fasta")
unlink("ungrouped.fasta")
unlink("group1.fasta")
unlink("group2.fasta")
unlink("group3.fasta")

</code></pre>

<hr>
<h2 id='sub.taxa.label'>
Substitute the tip labels of a phylogenetic tree
</h2><span id='topic+sub.taxa.label'></span>

<h3>Description</h3>

<p>Substitute the tip labels of a phylogenetic tree according to a reference data table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sub.taxa.label(tree, dat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sub.taxa.label_+3A_tree">tree</code></td>
<td>

<p>Phylogenetic tree
</p>
</td></tr>
<tr><td><code id="sub.taxa.label_+3A_dat">dat</code></td>
<td>

<p>A dataframe with the first column the tip labels and the second column the new names.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Phylogenetic tree with the tip labels substituted
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang <a href="mailto:jinlongzhang01@gmail.com">jinlongzhang01@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+read.tree">read.tree</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(ape)
data(bird.families)
tips &lt;- bird.families$tip.label
abr &lt;- paste("fam",1:length(tips), sep = "")
dat &lt;- data.frame(tips, abr)
ntree &lt;- sub.taxa.label(bird.families, dat)

</code></pre>

<hr>
<h2 id='supermat'>
Build PHYLIP supermatrix and RAxML partition file using aligned FASTA or PHYLIP files.
</h2><span id='topic+supermat'></span>

<h3>Description</h3>

<p>Build PHYLIP supermatrix and create RAxML partition file using aligned fasta or phylip files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>supermat(infiles, outfile = "supermat.out.phy",
         partition.file = "gene_partition.txt")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="supermat_+3A_infiles">infiles</code></td>
<td>

<p>a character string vector for phylip or aligned fasta file.
</p>
</td></tr>
<tr><td><code id="supermat_+3A_outfile">outfile</code></td>
<td>

<p>the name of the PHYLIP supermatrix
</p>
</td></tr>
<tr><td><code id="supermat_+3A_partition.file">partition.file</code></td>
<td>

<p>partition data summary describing the genes.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Supermatrix here means a phylip file with combined aligned sequences. The missing sequences should be replaced with either &quot;?&quot; or &quot;-&quot;.
</p>


<h3>Value</h3>

<p>A list containing:
(1)supermat.dat:a list containing all the data frames read by read.phylip or read.fasta
(2)res.super.dat: a data frame containing the sequences and the names
(3)partition.dat: summary for all the fasta or phylip files
(4)partition.dat.vector: character string vector for the partition file for RAxML
</p>


<h3>Note</h3>

<p>Punctuation characters and white space in the names of the sequences will be replaced by &quot;_&quot;. More information can be found at <code><a href="base.html#topic+regex">regex</a></code>.
Type of the sequence in the RAxML partition file should be changed manually according to the manual of RAxML.
</p>


<h3>Author(s)</h3>

<p>Jinlong Zhang &lt;jinlongzhang01@gmail.com&gt;
</p>


<h3>References</h3>

<p>Kress, W. J., Erickson, D. L., Jones, F. A., Swenson, N. G., Perez, R., Sanjur, O., &amp; Bermingham, E. (2009). Plant DNA barcodes and a community phylogeny of a tropical forest dynamics plot in Panama. Proceedings of the National Academy of Sciences, 106(44), 18621-18626.
</p>
<p>de Queiroz, A.and Gatesy, J. (2007). The supermatrix approach to systematics. Trends in Ecology &amp; Evolution, 22(1), 34-41.
</p>
<p><a href="https://github.com/stamatak/standard-RAxML">https://github.com/stamatak/standard-RAxML</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fasta">read.fasta</a></code>,<code><a href="#topic+read.phylip">read.phylip</a></code>,<code><a href="#topic+dat2phylip">dat2phylip</a></code>,
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  cat("6 22",
  "seq_1    --TTACAAATTGACTTATTATA",
  "seq_2    GATTACAAATTGACTTATTATA",
  "seq_3    GATTACAAATTGACTTATTATA",
  "seq_5    GATTACAAATTGACTTATTATA",
  "seq_8    GATTACAAATTGACTTATTATA",
  "seq_10   ---TACAAATTGAATTATTATA",
  file = "matk.phy", sep = "\n")

  cat("5 15",
  "seq_1     GATTACAAATTGACT",
  "seq_3     GATTACAAATTGACT",
  "seq_4     GATTACAAATTGACT",
  "seq_5     GATTACAAATTGACT",
  "seq_8     GATTACAAATTGACT",
  file = "rbcla.phy", sep = "\n")

  cat("5 50",
  "seq_2          GTCTTATAAGAAAGAATAAGAAAG--AAATACAAA-------AAAAAAGA",
  "seq_3          GTCTTATAAGAAAGAAATAGAAAAGTAAAAAAAAA-------AAAAAAAG",
  "seq_5          GACATAAGACATAAAATAGAATACTCAATCAGAAACCAACCCATAAAAAC",
  "seq_8          ATTCCAAAATAAAATACAAAAAGAAAAAACTAGAAAGTTTTTTTTCTTTG",
  "seq_9          ATTCTTTGTTCTTTTTTTTCTTTAATCTTTAAATAAACCTTTTTTTTTTA",
  file = "trn1.phy", sep = "\n")

supermat(infiles = c("matk.phy", "rbcla.phy", "trn1.phy"))
unlink(c("matk.phy", "rbcla.phy", "trn1.phy"))
unlink(c("supermat.out.phy","gene_partition.txt"))

cat(
  "&gt;seq_1",  "--TTACAAATTGACTTATTATA",
  "&gt;seq_2",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_3",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_5",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_8",  "GATTACAAATTGACTTATTATA",
  "&gt;seq_10", "---TACAAATTGAATTATTATA",
  file = "matk.fasta", sep = "\n")

cat(
  "&gt;seq_1", "GATTACAAATTGACT",
  "&gt;seq_3", "GATTACAAATTGACT",
  "&gt;seq_4", "GATTACAAATTGACT",
  "&gt;seq_5", "GATTACAAATTGACT",
  "&gt;seq_8", "GATTACAAATTGACT",
  file = "rbcla.fasta", sep = "\n")

cat(
  "&gt;seq_2", "GTCTTATAAGAAAGAATAAGAAAG--AAATACAAA-------AAAAAAGA",
  "&gt;seq_3", "GTCTTATAAGAAAGAAATAGAAAAGTAAAAAAAAA-------AAAAAAAG",
  "&gt;seq_5", "GACATAAGACATAAAATAGAATACTCAATCAGAAACCAACCCATAAAAAC",
  "&gt;seq_8", "ATTCCAAAATAAAATACAAAAAGAAAAAACTAGAAAGTTTTTTTTCTTTG",
  "&gt;seq_9", "ATTCTTTGTTCTTTTTTTTCTTTAATCTTTAAATAAACCTTTTTTTTTTA",
  file = "trn1.fasta", sep = "\n")

supermat(infiles = c("matk.fasta", "rbcla.fasta", "trn1.fasta"))
unlink(c("matk.fasta", "rbcla.fasta", "trn1.fasta"))

unlink(c("supermat.out.phy","gene_partition.txt"))

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
