<!DOCTYPE html><html><head><title>Help for package phylosignal</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {phylosignal}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.layouterize'><p>Set layout for plots</p></a></li>
<li><a href='#.layouterizeRatio'><p>Set layout for plots</p></a></li>
<li><a href='#.orderGrArg'><p>Reordering vector or matrix of settings.</p></a></li>
<li><a href='#barplot.phylo4d'><p>Barplot of Traits Values along a Phylogeny</p></a></li>
<li><a href='#descentsNames'><p>Names of descents</p></a></li>
<li><a href='#directDescents'><p>ID of direct descents</p></a></li>
<li><a href='#distEq'><p>Pairwise Distance from Regularly Distributed Points</p></a></li>
<li><a href='#dotplot'><p>Dotplot</p></a></li>
<li><a href='#dotplot.phylo4d'><p>Dotplot of Traits Values along a Phylogeny</p></a></li>
<li><a href='#evenColors'><p>Palette of evenly distributed colors</p></a></li>
<li><a href='#focusTraits'><p>Focus on sub parts of a plot</p></a></li>
<li><a href='#graphClust'><p>Phylogenetically constrained  clustering</p></a></li>
<li><a href='#gridplot'><p>Gridplot</p></a></li>
<li><a href='#gridplot.phylo4d'><p>Gridplot of Traits Values along a Phylogeny</p></a></li>
<li><a href='#kStarTest'><p>Computes permutation test for Blomberg's K Star</p></a></li>
<li><a href='#kTest'><p>Computes permutation test for Blomberg's K</p></a></li>
<li><a href='#lambdaTest'><p>Test Pagel's Lambda</p>
Optimize Pagel's Lambda and do a likelihood ratio test.</a></li>
<li><a href='#lipaMoran'><p>Local Indicator of Phylogenetic Association</p></a></li>
<li><a href='#mantelStat'><p>Mantel statistic</p></a></li>
<li><a href='#matchTipsAndTraits'><p>Match matrix row/col names with phylo4d tips/traits</p></a></li>
<li><a href='#moranTest'><p>Computes permutation test for Moran's I</p></a></li>
<li><a href='#multiplot.phylo4d'><p>Plots of Traits Values along a Phylogeny</p></a></li>
<li><a href='#navic'><p>Phylogeny and pollution sensitivity of diatoms</p></a></li>
<li><a href='#pagelLogLik'><p>Computes log-likelihood for data and a given value of Pagel's Lambda</p></a></li>
<li><a href='#phyloCorrelogram'><p>Phylogenetic correlogram</p></a></li>
<li><a href='#phylosignal'><p>phylosignal</p></a></li>
<li><a href='#phyloSignal'><p>Computes phylogenetic signal with different methods</p></a></li>
<li><a href='#phyloSignalBS'><p>Computes phylogenetic signal for bootstrapped replicates of a phylogeny.</p></a></li>
<li><a href='#phyloSignalINT'><p>Computes phylogenetic signal at each internal node of a phylogeny</p></a></li>
<li><a href='#phyloSim'><p>Simulate the behaviour of phylogenetic signal statistics with a given phylogeny</p></a></li>
<li><a href='#phyloSimSignal'><p>Phylogenetic signal estimation as a fraction of a Brownian Motion process.</p></a></li>
<li><a href='#phyloWeights'><p>Phylogenetic weights matrix</p></a></li>
<li><a href='#plot.graphclust'><p>Plot phylogenetically constrained  clustering</p></a></li>
<li><a href='#plot.phylocorrelogram'><p>Plot a phylogenetic correlogram</p></a></li>
<li><a href='#plot.phylosim'><p>Plot <code>phylosim</code> object</p></a></li>
<li><a href='#plot.phylosimsignal'><p>Plot signal estimation as a fraction of a Brownian Motion process.</p></a></li>
<li><a href='#print.graphclust'><p>#' Print results of phylogenetically constrained  clustering</p></a></li>
<li><a href='#print.phylosimsignal'><p>Print signal estimation as a fraction of a Brownian Motion process.</p></a></li>
<li><a href='#read.p4d'><p>Read a phylo4d object from files</p></a></li>
<li><a href='#rTraitContWeight'><p>Traits Values Simulation</p></a></li>
<li><a href='#subsetPhyloSimSignal'><p>phyloSimSignal subsetting</p></a></li>
<li><a href='#white2red'><p>Color Palette</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Exploring the Phylogenetic Signal in Continuous Traits</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-10-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Francois Keck &lt;francois.keck@gmail.com&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Francois Keck &lt;francois.keck@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of tools to explore the phylogenetic signal in univariate and multivariate data. The package provides functions to plot traits data against a phylogenetic tree, different measures and tests for the phylogenetic signal, methods to describe where the signal is located and a phylogenetic clustering method.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11.0), adephylo, igraph, ape, phylobase, boot, DBI,
graphics, grDevices, methods, stats, utils</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-10-12 07:04:12 UTC; ecoadmin</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-10-12 10:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='.layouterize'>Set layout for plots</h2><span id='topic+.layouterize'></span>

<h3>Description</h3>

<p>Set layout for plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.layouterize(n.traits, show.tip)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".layouterize_+3A_n.traits">n.traits</code></td>
<td>
<p>the number of traits in the layout</p>
</td></tr>
<tr><td><code id=".layouterize_+3A_show.tip">show.tip</code></td>
<td>
<p>a logical indicating whether tip names are included in the layout</p>
</td></tr>
</table>

<hr>
<h2 id='.layouterizeRatio'>Set layout for plots</h2><span id='topic+.layouterizeRatio'></span>

<h3>Description</h3>

<p>Set layout for plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.layouterizeRatio(tree.ratio, n.traits, show.tip)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".layouterizeRatio_+3A_tree.ratio">tree.ratio</code></td>
<td>
<p>the ratio of phylogenetic tree included in the layout</p>
</td></tr>
<tr><td><code id=".layouterizeRatio_+3A_n.traits">n.traits</code></td>
<td>
<p>the number of traits in the layout</p>
</td></tr>
<tr><td><code id=".layouterizeRatio_+3A_show.tip">show.tip</code></td>
<td>
<p>a logical indicating whether tip names are included in the layout</p>
</td></tr>
</table>

<hr>
<h2 id='.orderGrArg'>Reordering vector or matrix of settings.</h2><span id='topic+.orderGrArg'></span>

<h3>Description</h3>

<p>Internal function of multiplot.phylo4d.
Reordering vector or matrix of settings and check for names consistencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.orderGrArg(x, n.tips, n.traits, new.order, tips, default)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".orderGrArg_+3A_x">x</code></td>
<td>
<p>a vector or a matrix to order.</p>
</td></tr>
<tr><td><code id=".orderGrArg_+3A_n.tips">n.tips</code></td>
<td>
<p>number of tips.</p>
</td></tr>
<tr><td><code id=".orderGrArg_+3A_n.traits">n.traits</code></td>
<td>
<p>number of traits</p>
</td></tr>
<tr><td><code id=".orderGrArg_+3A_new.order">new.order</code></td>
<td>
<p>a numeric vector giving the new order.</p>
</td></tr>
<tr><td><code id=".orderGrArg_+3A_tips">tips</code></td>
<td>
<p>a character vector giving the tips labels (with the new order)</p>
</td></tr>
<tr><td><code id=".orderGrArg_+3A_default">default</code></td>
<td>
<p>the default value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An ordered vector or matrix.
An error if problem of consistency.
</p>

<hr>
<h2 id='barplot.phylo4d'>Barplot of Traits Values along a Phylogeny</h2><span id='topic+barplot.phylo4d'></span>

<h3>Description</h3>

<p>Barplot of Traits Values along a Phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylo4d'
barplot(
  height,
  trait = names(tdata(height)),
  center = TRUE,
  scale = TRUE,
  tree.ladderize = FALSE,
  tree.type = "phylogram",
  tree.ratio = NULL,
  tree.xlim = NULL,
  tree.open.angle = 0,
  tree.open.crown = TRUE,
  show.tip = TRUE,
  tip.labels = NULL,
  tip.col = "black",
  tip.cex = 1,
  tip.font = 3,
  tip.adj = 0,
  data.xlim = NULL,
  bar.lwd = 10,
  bar.col = "grey35",
  show.data.axis = TRUE,
  show.trait = TRUE,
  trait.labels = NULL,
  trait.col = "black",
  trait.cex = 1,
  trait.font = 1,
  trait.bg.col = "grey90",
  error.bar.sup = NULL,
  error.bar.inf = NULL,
  error.bar.col = 1,
  show.box = FALSE,
  grid.vertical = TRUE,
  grid.horizontal = FALSE,
  grid.col = "grey25",
  grid.lty = "dashed",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="barplot.phylo4d_+3A_height">height</code></td>
<td>
<p>a <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_trait">trait</code></td>
<td>
<p>the traits in the <code>phylo4d</code> object to include in the plot.
Can be a character vector giving the name of the traits or numbers giving the column index
in the table of the data slot of the p4d object. Can be used to reorder the traits in the plot.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_center">center</code></td>
<td>
<p>a logical indicating whether traits values should be centered.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_scale">scale</code></td>
<td>
<p>a logical indicating whether traits values should be scaled.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_tree.ladderize">tree.ladderize</code></td>
<td>
<p>a logical indicating whether the tree should be (right) ladderized.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_tree.type">tree.type</code></td>
<td>
<p>a character string specifying the type of phylogeny to be drawn.
Can be &quot;<code>phylogram</code>&quot;, &quot;<code>cladogram</code>&quot; or &quot;<code>fan</code>&quot;.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_tree.ratio">tree.ratio</code></td>
<td>
<p>a numeric value in [0, 1] giving the proportion of width of the figure for the tree.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_tree.xlim">tree.xlim</code></td>
<td>
<p>a numeric vector of length 2 giving the limits of the x-axis for the tree. If <code>NULL</code>,
it is determined automatically.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_tree.open.angle">tree.open.angle</code></td>
<td>
<p>a numeric value giving the angle in degrees left blank if <code>tree.type = "fan"</code>.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_tree.open.crown">tree.open.crown</code></td>
<td>
<p>a logical indicating whether the crowns should be drawn following the value
of <code>tree.open.angle</code> (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_show.tip">show.tip</code></td>
<td>
<p>logical indicating whether tips labels should be drawn.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_tip.labels">tip.labels</code></td>
<td>
<p>character vector to label the tips.
If <code>NULL</code> the tips labels of the <code>phylo4d</code> object are used</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_tip.col">tip.col</code></td>
<td>
<p>a vector of R colors to use for the tips labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_tip.cex">tip.cex</code></td>
<td>
<p>a numeric vector to control character size of the tips labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_tip.font">tip.font</code></td>
<td>
<p>an integer vector specifying the type of font for the tips labels:
1 (plain text), 2 (bold), 3 (italic), or 4 (bold italic). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_tip.adj">tip.adj</code></td>
<td>
<p>a vector of numeric in [0, 1] to control tips labels justification:
0 (left-justification), 0.5 (centering), or 1 (right-justification). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_data.xlim">data.xlim</code></td>
<td>
<p>numeric vector of length 2 or matrix giving the x coordinates range for
the barplots/dotplots (see Details).</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_bar.lwd">bar.lwd</code></td>
<td>
<p>a vector of numeric giving bar widths of the barplot(s).
Recycled along the tips, reapeated for each trait.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_bar.col">bar.col</code></td>
<td>
<p>a vector of R colors to use for the bars.
Recycled along the tips, reapeated for each trait.
The user can also provide a matrix for a finer tuning (see Details)</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_show.data.axis">show.data.axis</code></td>
<td>
<p>logical indicating whether barplots/dotplots axes should be drawn.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_show.trait">show.trait</code></td>
<td>
<p>logical indicating whether traits labels should be drawn.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_trait.labels">trait.labels</code></td>
<td>
<p>character vector to label the traits.
If <code>NULL</code> the traits labels of the <code>phylo4d</code> object are used.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_trait.col">trait.col</code></td>
<td>
<p>a vector of R colors to use for the traits labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_trait.cex">trait.cex</code></td>
<td>
<p>a numeric vector to control character size of the trait labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_trait.font">trait.font</code></td>
<td>
<p>an integer vector specifying the type of font for the traits labels:
1 (plain text), 2 (bold), 3 (italic), or 4 (bold italic). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_trait.bg.col">trait.bg.col</code></td>
<td>
<p>a vector of R colors to use for the background of the barplots.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_error.bar.sup">error.bar.sup</code></td>
<td>
<p>a matrix giving the superior limit for error bars.
Columns and rows names must match with traits and tips labels, respectively.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_error.bar.inf">error.bar.inf</code></td>
<td>
<p>a matrix giving the inferior limit for error bars.
Columns and rows names must match with traits and tips labels, respectively.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_error.bar.col">error.bar.col</code></td>
<td>
<p>a vector of R colors to use for the bars.
Recycled along the tips, reapeated for each trait.
The user can also provide a matrix for a finer tuning (see Details)</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_show.box">show.box</code></td>
<td>
<p>a logical indicating whether a box should be drawn around the plots.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_grid.vertical">grid.vertical</code></td>
<td>
<p>a logical incating whether vertical lines of the grid should be drawn.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_grid.horizontal">grid.horizontal</code></td>
<td>
<p>a logical incating whether horizontal lines of the grid should be drawn.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_grid.col">grid.col</code></td>
<td>
<p>a vector of R colors to use for the lines of the grid.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_grid.lty">grid.lty</code></td>
<td>
<p>the lines type of the grid. Possibly a vector.</p>
</td></tr>
<tr><td><code id="barplot.phylo4d_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code>plot.phylo</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(navic)
barplot(navic)

</code></pre>

<hr>
<h2 id='descentsNames'>Names of descents</h2><span id='topic+descentsNames'></span>

<h3>Description</h3>

<p>This function return all the names of the descents (tips) of a given node.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>descentsNames(phy, node)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="descentsNames_+3A_phy">phy</code></td>
<td>
<p>an object of class &quot;<code>phylo</code>.</p>
</td></tr>
<tr><td><code id="descentsNames_+3A_node">node</code></td>
<td>
<p>the ID of the node.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector tip labels.
</p>

<hr>
<h2 id='directDescents'>ID of direct descents</h2><span id='topic+directDescents'></span>

<h3>Description</h3>

<p>This function return the node number (ID) of the direct descents of a given node
</p>


<h3>Usage</h3>

<pre><code class='language-R'>directDescents(phy, node)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="directDescents_+3A_phy">phy</code></td>
<td>
<p>an object of class &quot;<code>phylo</code>.</p>
</td></tr>
<tr><td><code id="directDescents_+3A_node">node</code></td>
<td>
<p>the ID of the node.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of node numbers.
</p>

<hr>
<h2 id='distEq'>Pairwise Distance from Regularly Distributed Points</h2><span id='topic+distEq'></span>

<h3>Description</h3>

<p>This function return a distance (or proximity) matrix between n points regularly distributed in 1 dimension.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distEq(n, prox = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distEq_+3A_n">n</code></td>
<td>
<p>the number of points</p>
</td></tr>
<tr><td><code id="distEq_+3A_prox">prox</code></td>
<td>
<p>a logical indicating whether to return a matrix of proximity.
Default to <code>FALSE</code> so the function returns the matrix of distance.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A squared matrix. A matrix of class <code>dist</code> if <code>prox</code> is set to <code>FALSE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- distEq(5)
</code></pre>

<hr>
<h2 id='dotplot'>Dotplot</h2><span id='topic+dotplot'></span>

<h3>Description</h3>

<p>Dotplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dotplot(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dotplot_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>

<hr>
<h2 id='dotplot.phylo4d'>Dotplot of Traits Values along a Phylogeny</h2><span id='topic+dotplot.phylo4d'></span>

<h3>Description</h3>

<p>Dotplot of Traits Values along a Phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylo4d'
dotplot(
  p4d,
  trait = names(tdata(p4d)),
  center = TRUE,
  scale = TRUE,
  tree.ladderize = FALSE,
  tree.type = "phylogram",
  tree.ratio = NULL,
  tree.xlim = NULL,
  tree.open.angle = 0,
  tree.open.crown = TRUE,
  show.tip = TRUE,
  tip.labels = NULL,
  tip.col = "black",
  tip.cex = 1,
  tip.font = 3,
  tip.adj = 0,
  data.xlim = NULL,
  show.data.axis = TRUE,
  dot.col = "black",
  dot.pch = 20,
  dot.cex = 2,
  show.trait = TRUE,
  trait.labels = NULL,
  trait.col = "black",
  trait.cex = 1,
  trait.font = 1,
  trait.bg.col = "grey90",
  error.bar.sup = NULL,
  error.bar.inf = NULL,
  error.bar.col = 1,
  show.box = FALSE,
  grid.vertical = FALSE,
  grid.horizontal = TRUE,
  grid.col = "grey25",
  grid.lty = "dashed",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dotplot.phylo4d_+3A_p4d">p4d</code></td>
<td>
<p>a <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_trait">trait</code></td>
<td>
<p>the traits in the <code>phylo4d</code> object to include in the plot.
Can be a character vector giving the name of the traits or numbers giving the column index
in the table of the data slot of the p4d object. Can be used to reorder the traits in the plot.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_center">center</code></td>
<td>
<p>a logical indicating whether traits values should be centered.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_scale">scale</code></td>
<td>
<p>a logical indicating whether traits values should be scaled.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_tree.ladderize">tree.ladderize</code></td>
<td>
<p>a logical indicating whether the tree should be (right) ladderized.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_tree.type">tree.type</code></td>
<td>
<p>a character string specifying the type of phylogeny to be drawn.
Can be &quot;<code>phylogram</code>&quot;, &quot;<code>cladogram</code>&quot; or &quot;<code>fan</code>&quot;.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_tree.ratio">tree.ratio</code></td>
<td>
<p>a numeric value in [0, 1] giving the proportion of width of the figure for the tree.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_tree.xlim">tree.xlim</code></td>
<td>
<p>a numeric vector of length 2 giving the limits of the x-axis for the tree. If <code>NULL</code>,
it is determined automatically.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_tree.open.angle">tree.open.angle</code></td>
<td>
<p>a numeric value giving the angle in degrees left blank if <code>tree.type = "fan"</code>.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_tree.open.crown">tree.open.crown</code></td>
<td>
<p>a logical indicating whether the crowns should be drawn following the value
of <code>tree.open.angle</code> (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_show.tip">show.tip</code></td>
<td>
<p>logical indicating whether tips labels should be drawn.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_tip.labels">tip.labels</code></td>
<td>
<p>character vector to label the tips.
If <code>NULL</code> the tips labels of the <code>phylo4d</code> object are used</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_tip.col">tip.col</code></td>
<td>
<p>a vector of R colors to use for the tips labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_tip.cex">tip.cex</code></td>
<td>
<p>a numeric vector to control character size of the tips labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_tip.font">tip.font</code></td>
<td>
<p>an integer vector specifying the type of font for the tips labels:
1 (plain text), 2 (bold), 3 (italic), or 4 (bold italic). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_tip.adj">tip.adj</code></td>
<td>
<p>a vector of numeric in [0, 1] to control tips labels justification:
0 (left-justification), 0.5 (centering), or 1 (right-justification). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_data.xlim">data.xlim</code></td>
<td>
<p>numeric vector of length 2 or matrix giving the x coordinates range for
the barplots/dotplots (see Details).</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_show.data.axis">show.data.axis</code></td>
<td>
<p>logical indicating whether barplots/dotplots axes should be drawn.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_dot.col">dot.col</code></td>
<td>
<p>a vector of R colors to use for the points.
Recycled along the tips, reapeated for each trait.
The user can also provide a matrix for a finer tuning (see Details)</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_dot.pch">dot.pch</code></td>
<td>
<p>a numerical vector of symbol to use for the points.
Recycled along the tips, reapeated for each trait.
The user can also provide a matrix for a finer tuning (see Details)</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_dot.cex">dot.cex</code></td>
<td>
<p>a numerical vector. Character (or symbol) expansion for the points.
Recycled along the tips, reapeated for each trait.
The user can also provide a matrix for a finer tuning (see Details)</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_show.trait">show.trait</code></td>
<td>
<p>logical indicating whether traits labels should be drawn.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_trait.labels">trait.labels</code></td>
<td>
<p>character vector to label the traits.
If <code>NULL</code> the traits labels of the <code>phylo4d</code> object are used.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_trait.col">trait.col</code></td>
<td>
<p>a vector of R colors to use for the traits labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_trait.cex">trait.cex</code></td>
<td>
<p>a numeric vector to control character size of the trait labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_trait.font">trait.font</code></td>
<td>
<p>an integer vector specifying the type of font for the traits labels:
1 (plain text), 2 (bold), 3 (italic), or 4 (bold italic). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_trait.bg.col">trait.bg.col</code></td>
<td>
<p>a vector of R colors to use for the background of the barplots.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_error.bar.sup">error.bar.sup</code></td>
<td>
<p>a matrix giving the superior limit for error bars.
Columns and rows names must match with traits and tips labels, respectively.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_error.bar.inf">error.bar.inf</code></td>
<td>
<p>a matrix giving the inferior limit for error bars.
Columns and rows names must match with traits and tips labels, respectively.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_error.bar.col">error.bar.col</code></td>
<td>
<p>a vector of R colors to use for the bars.
Recycled along the tips, reapeated for each trait.
The user can also provide a matrix for a finer tuning (see Details)</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_show.box">show.box</code></td>
<td>
<p>a logical indicating whether a box should be drawn around the plots.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_grid.vertical">grid.vertical</code></td>
<td>
<p>a logical incating whether vertical lines of the grid should be drawn.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_grid.horizontal">grid.horizontal</code></td>
<td>
<p>a logical incating whether horizontal lines of the grid should be drawn.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_grid.col">grid.col</code></td>
<td>
<p>a vector of R colors to use for the lines of the grid.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_grid.lty">grid.lty</code></td>
<td>
<p>the lines type of the grid. Possibly a vector.</p>
</td></tr>
<tr><td><code id="dotplot.phylo4d_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code>plot.phylo</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(navic)
dotplot(navic)

</code></pre>

<hr>
<h2 id='evenColors'>Palette of evenly distributed colors</h2><span id='topic+evenColors'></span>

<h3>Description</h3>

<p>This function generates a vector of n colors evenly distributed in the RGB space.
Usefull to create palettes of distinct colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evenColors(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evenColors_+3A_n">n</code></td>
<td>
<p>the number of colors to be in the palette.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of hexadecimal colors.
</p>

<hr>
<h2 id='focusTraits'>Focus on sub parts of a plot</h2><span id='topic+focusTraits'></span><span id='topic+focusTree'></span><span id='topic+focusTips'></span><span id='topic+focusStop'></span>

<h3>Description</h3>

<p>These functions can be used after <code><a href="#topic+barplot.phylo4d">barplot.phylo4d</a></code>, <code><a href="#topic+dotplot.phylo4d">dotplot.phylo4d</a></code>
and <code><a href="#topic+gridplot.phylo4d">gridplot.phylo4d</a></code> when
<code>tree.type</code> is <code>"phylogram"</code> or <code>"cladogram"</code> to focus on
the different part of the plot and add graphical elements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>focusTraits(x)

focusTree()

focusTips()

focusStop()
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="focusTraits_+3A_x">x</code></td>
<td>
<p>the trait to focus on.
Can be a character string giving the name of the trait or an integer giving
the number of the trait in order of appearance in the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>#' Use <code>focusTree</code> to focus on the phylogenetic tree, <code>focusTraits</code>
to focus on a given trait and <code>focusTips</code> to focus on the tips labels.
Use <code>focusStop</code> to close the editing and restore graphical settings.
For each part of the plot, the coordinate system is restored, making edition easier.
For the phylogeny, post-editing functions of the package <code>ape</code> like <code>nodelabels</code> can be used.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(ape)
require(phylobase)
data(navic)
dat &lt;- tdata(navic)
neidium.sp &lt;- c("Neidium bisulcatum",
                "Neidium affine",
                "Neidium productum")
stauroneis.sp &lt;- c("Stauroneis kriegeri",
                   "Stauroneis acuta",
                   "Stauroneis gracilior",
                   "Stauroneis phoenicenteron")
neidium.mean &lt;- mean(dat[neidium.sp,])
stauroneis.mean &lt;- mean(dat[stauroneis.sp, ])

dotplot(navic, center = FALSE, scale = FALSE, data.xlim= c(0, 6))

focusTree()
nodelabels(node=c(22, 32), pch = 20, cex = 3, col = c(2, 3))

focusTraits()
segments(x0 = neidium.mean, y0 = 14.5,
         x1 = neidium.mean, y1 = 17.5,
         col = 3, lty = "dashed", lwd = 2)
segments(x0 = stauroneis.mean, y0 = 2.5,
         x1 = stauroneis.mean, y1 = 7.5,
         col = 2, lty = "dashed", lwd = 2)

focusTips()
rect(xleft = 0, ybottom = 2.5,
     xright = 0.9, ytop = 7.5,
     col = "#FF000020", border = NA)
rect(xleft = 0, ybottom = 14.5,
     xright = 0.9, ytop = 17.5,
     col = "#00FF0020", border = NA)

focusStop()

</code></pre>

<hr>
<h2 id='graphClust'>Phylogenetically constrained  clustering</h2><span id='topic+graphClust'></span>

<h3>Description</h3>

<p>This function extracts clusters of species based on traits values and phylogenetic proximities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graphClust(
  p4d,
  trait = names(tdata(p4d)),
  lim.phylo = 0.2,
  lim.trait = 0.2,
  select.method = "ellipse",
  dist.phylo = "patristic",
  dist.trait = "euclidean",
  scale.lim = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="graphClust_+3A_p4d">p4d</code></td>
<td>
<p>a <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="graphClust_+3A_trait">trait</code></td>
<td>
<p>the traits in the <code>phylo4d</code> object to use for clustering.
Can be a character vector giving the name of the traits or numbers giving the column index
in the table of the data slot of the <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="graphClust_+3A_lim.phylo">lim.phylo</code></td>
<td>
<p>the maximum phylogenetic distance for edges selection.</p>
</td></tr>
<tr><td><code id="graphClust_+3A_lim.trait">lim.trait</code></td>
<td>
<p>the maximum trait-based distance for edges selection.</p>
</td></tr>
<tr><td><code id="graphClust_+3A_select.method">select.method</code></td>
<td>
<p>a character string specifying the method used to select edges.
This must be one of &quot;<code>line</code>&quot;, &quot;<code>rectangle</code>&quot; or &quot;<code>ellipse</code>&quot;.</p>
</td></tr>
<tr><td><code id="graphClust_+3A_dist.phylo">dist.phylo</code></td>
<td>
<p>a matrix of phylogenetic distances or a character string specifying a method to compute it.
See Details.</p>
</td></tr>
<tr><td><code id="graphClust_+3A_dist.trait">dist.trait</code></td>
<td>
<p>a character string specifying the method used to compute traits distances.
See Details.</p>
</td></tr>
<tr><td><code id="graphClust_+3A_scale.lim">scale.lim</code></td>
<td>
<p>logical (default <code>TRUE</code>) indicating if <code>lim.phylo</code> and <code>lim.trait</code> are scaled
(divided by their max value).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If &quot;<code>dist.phylo</code>&quot; is a character string,
the phylogenetic distance matrix is computed internally
using the function <code><a href="adephylo.html#topic+distTips">distTips</a></code> from the package <span class="pkg">adephylo</span>.
All the methods supported by <code><a href="adephylo.html#topic+distTips">distTips</a></code> are available:
&quot;<code>patristic</code>&quot;,&quot;<code>nNodes</code>&quot;,&quot;<code>Abouheif</code>&quot; and &quot;<code>sumDD</code>&quot;.
See <code><a href="adephylo.html#topic+distTips">distTips</a></code> for details about the methods.
</p>
<p>If &quot;<code>dist.trait</code>&quot; is a character string,
the traits distance matrix is computed with the <code><a href="stats.html#topic+dist">dist</a></code> function.
All the methods supported by <code><a href="stats.html#topic+dist">dist</a></code> are available:
&quot;<code>euclidean</code>&quot;,&quot;<code>maximum</code>&quot;, &quot;<code>manhattan</code>&quot;,
&quot;<code>canberra</code>&quot;, &quot;<code>binary</code>&quot; and &quot;<code>minkowski</code>&quot;.
See <code><a href="stats.html#topic+dist">dist</a></code> for details about the methods.
</p>


<h3>Value</h3>

<p>An object of class <code>graphclust</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(navic)
gC &lt;- graphClust(navic, lim.phylo = 1, lim.trait = 2, scale.lim = FALSE)
gC
plot.graphclust(gC, which = "selection", ask = FALSE)
plot.graphclust(gC, which = "graph", ask = FALSE)
plot.graphclust(gC, which = "tree", ask = FALSE)

</code></pre>

<hr>
<h2 id='gridplot'>Gridplot</h2><span id='topic+gridplot'></span>

<h3>Description</h3>

<p>Gridplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gridplot(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gridplot_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods.</p>
</td></tr>
</table>

<hr>
<h2 id='gridplot.phylo4d'>Gridplot of Traits Values along a Phylogeny</h2><span id='topic+gridplot.phylo4d'></span>

<h3>Description</h3>

<p>Gridplot of Traits Values along a Phylogeny
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylo4d'
gridplot(
  p4d,
  trait = names(tdata(p4d)),
  center = TRUE,
  scale = TRUE,
  tree.ladderize = FALSE,
  tree.type = "phylogram",
  tree.ratio = NULL,
  tree.xlim = NULL,
  tree.open.angle = 0,
  tree.open.crown = TRUE,
  show.tip = TRUE,
  tip.labels = NULL,
  tip.col = "black",
  tip.cex = 1,
  tip.font = 3,
  tip.adj = 0,
  cell.col = white2red(100),
  show.color.scale = TRUE,
  show.trait = TRUE,
  trait.labels = NULL,
  trait.col = "black",
  trait.cex = 0.7,
  trait.font = 1,
  trait.bg.col = "grey90",
  show.box = FALSE,
  grid.vertical = FALSE,
  grid.horizontal = FALSE,
  grid.col = "grey25",
  grid.lty = "dashed",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gridplot.phylo4d_+3A_p4d">p4d</code></td>
<td>
<p>a <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_trait">trait</code></td>
<td>
<p>the traits in the <code>phylo4d</code> object to include in the plot.
Can be a character vector giving the name of the traits or numbers giving the column index
in the table of the data slot of the p4d object. Can be used to reorder the traits in the plot.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_center">center</code></td>
<td>
<p>a logical indicating whether traits values should be centered.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_scale">scale</code></td>
<td>
<p>a logical indicating whether traits values should be scaled.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_tree.ladderize">tree.ladderize</code></td>
<td>
<p>a logical indicating whether the tree should be (right) ladderized.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_tree.type">tree.type</code></td>
<td>
<p>a character string specifying the type of phylogeny to be drawn.
Can be &quot;<code>phylogram</code>&quot;, &quot;<code>cladogram</code>&quot; or &quot;<code>fan</code>&quot;.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_tree.ratio">tree.ratio</code></td>
<td>
<p>a numeric value in [0, 1] giving the proportion of width of the figure for the tree.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_tree.xlim">tree.xlim</code></td>
<td>
<p>a numeric vector of length 2 giving the limits of the x-axis for the tree. If <code>NULL</code>,
it is determined automatically.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_tree.open.angle">tree.open.angle</code></td>
<td>
<p>a numeric value giving the angle in degrees left blank if <code>tree.type = "fan"</code>.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_tree.open.crown">tree.open.crown</code></td>
<td>
<p>a logical indicating whether the crowns should be drawn following the value
of <code>tree.open.angle</code> (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_show.tip">show.tip</code></td>
<td>
<p>logical indicating whether tips labels should be drawn.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_tip.labels">tip.labels</code></td>
<td>
<p>character vector to label the tips.
If <code>NULL</code> the tips labels of the <code>phylo4d</code> object are used</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_tip.col">tip.col</code></td>
<td>
<p>a vector of R colors to use for the tips labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_tip.cex">tip.cex</code></td>
<td>
<p>a numeric vector to control character size of the tips labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_tip.font">tip.font</code></td>
<td>
<p>an integer vector specifying the type of font for the tips labels:
1 (plain text), 2 (bold), 3 (italic), or 4 (bold italic). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_tip.adj">tip.adj</code></td>
<td>
<p>a vector of numeric in [0, 1] to control tips labels justification:
0 (left-justification), 0.5 (centering), or 1 (right-justification). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_cell.col">cell.col</code></td>
<td>
<p>a vector of colors for <code>gridplot</code> cells.
Easily generated by <code><a href="grDevices.html#topic+heat.colors">heat.colors</a></code>,
<code><a href="grDevices.html#topic+topo.colors">topo.colors</a></code>, <code><a href="grDevices.html#topic+terrain.colors">terrain.colors</a></code>
or other functions created with <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_show.color.scale">show.color.scale</code></td>
<td>
<p>logical indicating whether color scale should be drawn.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_show.trait">show.trait</code></td>
<td>
<p>logical indicating whether traits labels should be drawn.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_trait.labels">trait.labels</code></td>
<td>
<p>character vector to label the traits.
If <code>NULL</code> the traits labels of the <code>phylo4d</code> object are used.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_trait.col">trait.col</code></td>
<td>
<p>a vector of R colors to use for the traits labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_trait.cex">trait.cex</code></td>
<td>
<p>a numeric vector to control character size of the trait labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_trait.font">trait.font</code></td>
<td>
<p>an integer vector specifying the type of font for the traits labels:
1 (plain text), 2 (bold), 3 (italic), or 4 (bold italic). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_trait.bg.col">trait.bg.col</code></td>
<td>
<p>a vector of R colors to use for the background of the barplots.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_show.box">show.box</code></td>
<td>
<p>a logical indicating whether a box should be drawn around the plots.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_grid.vertical">grid.vertical</code></td>
<td>
<p>a logical incating whether vertical lines of the grid should be drawn.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_grid.horizontal">grid.horizontal</code></td>
<td>
<p>a logical incating whether horizontal lines of the grid should be drawn.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_grid.col">grid.col</code></td>
<td>
<p>a vector of R colors to use for the lines of the grid.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_grid.lty">grid.lty</code></td>
<td>
<p>the lines type of the grid. Possibly a vector.</p>
</td></tr>
<tr><td><code id="gridplot.phylo4d_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code>plot.phylo</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(navic)
gridplot(navic)

# Multivariate data
require(phylobase)
tipData(navic) &lt;- matrix(rnorm(170), nrow = 17)
gridplot(navic)

</code></pre>

<hr>
<h2 id='kStarTest'>Computes permutation test for Blomberg's K Star</h2><span id='topic+kStarTest'></span>

<h3>Description</h3>

<p>Computes permutation test for Blomberg's K Star
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kStarTest(xr, vcvr, reps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kStarTest_+3A_xr">xr</code></td>
<td>
<p>a vector of data</p>
</td></tr>
<tr><td><code id="kStarTest_+3A_vcvr">vcvr</code></td>
<td>
<p>phylogenetic variance-covariance matrix</p>
</td></tr>
<tr><td><code id="kStarTest_+3A_reps">reps</code></td>
<td>
<p>the number of permutations</p>
</td></tr>
</table>

<hr>
<h2 id='kTest'>Computes permutation test for Blomberg's K</h2><span id='topic+kTest'></span>

<h3>Description</h3>

<p>Computes permutation test for Blomberg's K
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kTest(xr, vcvr, reps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kTest_+3A_xr">xr</code></td>
<td>
<p>a vector of data</p>
</td></tr>
<tr><td><code id="kTest_+3A_vcvr">vcvr</code></td>
<td>
<p>phylogenetic variance-covariance matrix</p>
</td></tr>
<tr><td><code id="kTest_+3A_reps">reps</code></td>
<td>
<p>the number of permutations</p>
</td></tr>
</table>

<hr>
<h2 id='lambdaTest'>Test Pagel's Lambda
Optimize Pagel's Lambda and do a likelihood ratio test.</h2><span id='topic+lambdaTest'></span>

<h3>Description</h3>

<p>Test Pagel's Lambda
Optimize Pagel's Lambda and do a likelihood ratio test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lambdaTest(x, vcv)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lambdaTest_+3A_x">x</code></td>
<td>
<p>a vector of numeric data.</p>
</td></tr>
<tr><td><code id="lambdaTest_+3A_vcv">vcv</code></td>
<td>
<p>the phylogenetic variance-covariance matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The optimization process is currently performed in R.
</p>

<hr>
<h2 id='lipaMoran'>Local Indicator of Phylogenetic Association</h2><span id='topic+lipaMoran'></span>

<h3>Description</h3>

<p>This function computes Local Indicator of Phylogenetic Association (local Moran's I)
for each tip of a tree.
Tests are based on permutations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lipaMoran(
  p4d,
  trait = names(tdata(p4d)),
  reps = 999,
  alternative = "greater",
  prox.phylo = "patristic",
  as.p4d = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lipaMoran_+3A_p4d">p4d</code></td>
<td>
<p>a <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="lipaMoran_+3A_trait">trait</code></td>
<td>
<p>the traits in the <code>phylo4d</code> object for which to compute LIPA.
Can be a character vector giving the name of the traits
or numbers giving the column in the table of the data slot of the <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="lipaMoran_+3A_reps">reps</code></td>
<td>
<p>a numeric value. Number of repetitions for the estimation of p.values with randomization.</p>
</td></tr>
<tr><td><code id="lipaMoran_+3A_alternative">alternative</code></td>
<td>
<p>a character string specifying the alternative hypothesis for the tests.
Must be one of <code>greater</code> (default), <code>two-sided</code> or <code>less</code>.</p>
</td></tr>
<tr><td><code id="lipaMoran_+3A_prox.phylo">prox.phylo</code></td>
<td>
<p>a matrix of phylogenetic proximities or
a character string specifying a method to compute it.
See Details.</p>
</td></tr>
<tr><td><code id="lipaMoran_+3A_as.p4d">as.p4d</code></td>
<td>
<p>logical. Should the results returned as a <code>phylo4d</code> object?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If &quot;<code>prox.phylo</code>&quot; is a character string,
the phylogenetic proximity matrix is computed internally
using the function <code><a href="adephylo.html#topic+proxTips">proxTips</a></code> from the package <span class="pkg">adephylo</span>.
Different methods are available:
&quot;<code>patristic</code>&quot;,&quot;<code>nNodes</code>&quot;,&quot;<code>Abouheif</code>&quot; and &quot;<code>sumDD</code>&quot;.
See <code><a href="adephylo.html#topic+proxTips">proxTips</a></code> for details about the methods.
</p>


<h3>Value</h3>

<p>If <code>as.p4d</code> is <code>FALSE</code> (default), the function returns a list:
</p>

<dl>
<dt>lipa</dt><dd><p>A matrix of LIPA indices computed for each tip of the tree and each trait.</p>
</dd>
<dt>p.value</dt><dd><p>A matrix of p-values (tests of LIPA indices)</p>
</dd>
<dt>reps</dt><dd><p>Number of permutations for the tests</p>
</dd>
<dt>alternative</dt><dd><p>Alternative hypothesis for the tests</p>
</dd>
</dl>

<p>If <code>as.p4d</code> is <code>TRUE</code>, the function returns a <code>phylo4d</code> object
with LIPA values as tips associated data.
</p>


<h3>References</h3>

<p>Anselin L. (1995) Local Indicators of Spatial Association-LISA. Geographical Analysis 27, 93-115.
</p>

<hr>
<h2 id='mantelStat'>Mantel statistic</h2><span id='topic+mantelStat'></span>

<h3>Description</h3>

<p>Mantel statistic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mantelStat(xr, Wr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mantelStat_+3A_xr">xr</code></td>
<td>
<p>a matrix of traits.</p>
</td></tr>
<tr><td><code id="mantelStat_+3A_wr">Wr</code></td>
<td>
<p>a matrix of weights.</p>
</td></tr>
</table>

<hr>
<h2 id='matchTipsAndTraits'>Match matrix row/col names with phylo4d tips/traits</h2><span id='topic+matchTipsAndTraits'></span>

<h3>Description</h3>

<p>Check for constitency of rows and columns names between a matrix of traits values and phylo4d object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>matchTipsAndTraits(
  x,
  p4d = NULL,
  p4d.tips = NULL,
  p4d.traits = NULL,
  subset = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="matchTipsAndTraits_+3A_x">x</code></td>
<td>
<p>a matrix of data.</p>
</td></tr>
<tr><td><code id="matchTipsAndTraits_+3A_p4d">p4d</code></td>
<td>
<p>a phylo4d object.</p>
</td></tr>
<tr><td><code id="matchTipsAndTraits_+3A_p4d.tips">p4d.tips</code></td>
<td>
<p>tips labels (relevant if p4d is <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="matchTipsAndTraits_+3A_p4d.traits">p4d.traits</code></td>
<td>
<p>traits labels (relevant if p4d is <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="matchTipsAndTraits_+3A_subset">subset</code></td>
<td>
<p>a logical. Should the data matrix be subsetted using tips and traits labels.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The data matrix (eventually subsetted).
An error if no consistency between the data and the tree.
</p>

<hr>
<h2 id='moranTest'>Computes permutation test for Moran's I</h2><span id='topic+moranTest'></span>

<h3>Description</h3>

<p>Computes permutation test for Moran's I
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moranTest(xr, Wr, reps)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="moranTest_+3A_xr">xr</code></td>
<td>
<p>a vector of data</p>
</td></tr>
<tr><td><code id="moranTest_+3A_wr">Wr</code></td>
<td>
<p>a weighting matrix</p>
</td></tr>
<tr><td><code id="moranTest_+3A_reps">reps</code></td>
<td>
<p>the number of permutations</p>
</td></tr>
</table>

<hr>
<h2 id='multiplot.phylo4d'>Plots of Traits Values along a Phylogeny</h2><span id='topic+multiplot.phylo4d'></span>

<h3>Description</h3>

<p>This function provides a general interface to plot <code>phylo4d</code> object
(i.e. phylogenetic tree and data).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiplot.phylo4d(
  p4d,
  trait = names(tdata(p4d)),
  center = TRUE,
  scale = TRUE,
  plot.type = "barplot",
  tree.ladderize = FALSE,
  tree.type = "phylogram",
  tree.ratio = NULL,
  tree.xlim = NULL,
  tree.open.angle = 0,
  tree.open.crown = TRUE,
  show.tip = TRUE,
  tip.labels = NULL,
  tip.col = "black",
  tip.cex = 1,
  tip.font = 3,
  tip.adj = 0,
  data.xlim = NULL,
  bar.lwd = 10,
  bar.col = "grey35",
  show.data.axis = TRUE,
  dot.col = "black",
  dot.pch = 20,
  dot.cex = 2,
  cell.col = white2red(100),
  show.color.scale = TRUE,
  show.trait = TRUE,
  trait.labels = NULL,
  trait.col = "black",
  trait.cex = 1,
  trait.font = 1,
  trait.bg.col = "grey90",
  error.bar.sup = NULL,
  error.bar.inf = NULL,
  error.bar.col = 1,
  show.box = FALSE,
  grid.vertical = TRUE,
  grid.horizontal = FALSE,
  grid.col = "grey25",
  grid.lty = "dashed",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiplot.phylo4d_+3A_p4d">p4d</code></td>
<td>
<p>a <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_trait">trait</code></td>
<td>
<p>the traits in the <code>phylo4d</code> object to include in the plot.
Can be a character vector giving the name of the traits or numbers giving the column index
in the table of the data slot of the p4d object. Can be used to reorder the traits in the plot.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_center">center</code></td>
<td>
<p>a logical indicating whether traits values should be centered.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_scale">scale</code></td>
<td>
<p>a logical indicating whether traits values should be scaled.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_plot.type">plot.type</code></td>
<td>
<p>a character string specifying the type of plot for traits data.
Can be &quot;<code>barplot</code>&quot;, &quot;<code>dotplot</code>&quot; or &quot;<code>gridplot</code>&quot;.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_tree.ladderize">tree.ladderize</code></td>
<td>
<p>a logical indicating whether the tree should be (right) ladderized.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_tree.type">tree.type</code></td>
<td>
<p>a character string specifying the type of phylogeny to be drawn.
Can be &quot;<code>phylogram</code>&quot;, &quot;<code>cladogram</code>&quot; or &quot;<code>fan</code>&quot;.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_tree.ratio">tree.ratio</code></td>
<td>
<p>a numeric value in [0, 1] giving the proportion of width of the figure for the tree.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_tree.xlim">tree.xlim</code></td>
<td>
<p>a numeric vector of length 2 giving the limits of the x-axis for the tree. If <code>NULL</code>,
it is determined automatically.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_tree.open.angle">tree.open.angle</code></td>
<td>
<p>a numeric value giving the angle in degrees left blank if <code>tree.type = "fan"</code>.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_tree.open.crown">tree.open.crown</code></td>
<td>
<p>a logical indicating whether the crowns should be drawn following the value
of <code>tree.open.angle</code> (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_show.tip">show.tip</code></td>
<td>
<p>logical indicating whether tips labels should be drawn.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_tip.labels">tip.labels</code></td>
<td>
<p>character vector to label the tips.
If <code>NULL</code> the tips labels of the <code>phylo4d</code> object are used</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_tip.col">tip.col</code></td>
<td>
<p>a vector of R colors to use for the tips labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_tip.cex">tip.cex</code></td>
<td>
<p>a numeric vector to control character size of the tips labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_tip.font">tip.font</code></td>
<td>
<p>an integer vector specifying the type of font for the tips labels:
1 (plain text), 2 (bold), 3 (italic), or 4 (bold italic). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_tip.adj">tip.adj</code></td>
<td>
<p>a vector of numeric in [0, 1] to control tips labels justification:
0 (left-justification), 0.5 (centering), or 1 (right-justification). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_data.xlim">data.xlim</code></td>
<td>
<p>numeric vector of length 2 or matrix giving the x coordinates range for
the barplots/dotplots (see Details).</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_bar.lwd">bar.lwd</code></td>
<td>
<p>a vector of numeric giving bar widths of the barplot(s).
Recycled along the tips, reapeated for each trait.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_bar.col">bar.col</code></td>
<td>
<p>a vector of R colors to use for the bars.
Recycled along the tips, reapeated for each trait.
The user can also provide a matrix for a finer tuning (see Details)</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_show.data.axis">show.data.axis</code></td>
<td>
<p>logical indicating whether barplots/dotplots axes should be drawn.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_dot.col">dot.col</code></td>
<td>
<p>a vector of R colors to use for the points.
Recycled along the tips, reapeated for each trait.
The user can also provide a matrix for a finer tuning (see Details)</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_dot.pch">dot.pch</code></td>
<td>
<p>a numerical vector of symbol to use for the points.
Recycled along the tips, reapeated for each trait.
The user can also provide a matrix for a finer tuning (see Details)</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_dot.cex">dot.cex</code></td>
<td>
<p>a numerical vector. Character (or symbol) expansion for the points.
Recycled along the tips, reapeated for each trait.
The user can also provide a matrix for a finer tuning (see Details)</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_cell.col">cell.col</code></td>
<td>
<p>a vector of colors for <code>gridplot</code> cells.
Easily generated by <code><a href="grDevices.html#topic+heat.colors">heat.colors</a></code>,
<code><a href="grDevices.html#topic+topo.colors">topo.colors</a></code>, <code><a href="grDevices.html#topic+terrain.colors">terrain.colors</a></code>
or other functions created with <code><a href="grDevices.html#topic+colorRampPalette">colorRampPalette</a></code>.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_show.color.scale">show.color.scale</code></td>
<td>
<p>logical indicating whether color scale should be drawn.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_show.trait">show.trait</code></td>
<td>
<p>logical indicating whether traits labels should be drawn.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_trait.labels">trait.labels</code></td>
<td>
<p>character vector to label the traits.
If <code>NULL</code> the traits labels of the <code>phylo4d</code> object are used.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_trait.col">trait.col</code></td>
<td>
<p>a vector of R colors to use for the traits labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_trait.cex">trait.cex</code></td>
<td>
<p>a numeric vector to control character size of the trait labels.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_trait.font">trait.font</code></td>
<td>
<p>an integer vector specifying the type of font for the traits labels:
1 (plain text), 2 (bold), 3 (italic), or 4 (bold italic). Recycled if necessary.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_trait.bg.col">trait.bg.col</code></td>
<td>
<p>a vector of R colors to use for the background of the barplots.
Recycled if necessary.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_error.bar.sup">error.bar.sup</code></td>
<td>
<p>a matrix giving the superior limit for error bars.
Columns and rows names must match with traits and tips labels, respectively.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_error.bar.inf">error.bar.inf</code></td>
<td>
<p>a matrix giving the inferior limit for error bars.
Columns and rows names must match with traits and tips labels, respectively.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_error.bar.col">error.bar.col</code></td>
<td>
<p>a vector of R colors to use for the bars.
Recycled along the tips, reapeated for each trait.
The user can also provide a matrix for a finer tuning (see Details)</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_show.box">show.box</code></td>
<td>
<p>a logical indicating whether a box should be drawn around the plots.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_grid.vertical">grid.vertical</code></td>
<td>
<p>a logical incating whether vertical lines of the grid should be drawn.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_grid.horizontal">grid.horizontal</code></td>
<td>
<p>a logical incating whether horizontal lines of the grid should be drawn.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_grid.col">grid.col</code></td>
<td>
<p>a vector of R colors to use for the lines of the grid.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_grid.lty">grid.lty</code></td>
<td>
<p>the lines type of the grid. Possibly a vector.</p>
</td></tr>
<tr><td><code id="multiplot.phylo4d_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code>plot.phylo</code>.</p>
</td></tr>
</table>

<hr>
<h2 id='navic'>Phylogeny and pollution sensitivity of diatoms</h2><span id='topic+navic'></span>

<h3>Description</h3>

<p>A phylogenetic tree and the pollution sensitivity of 17 diatoms species from the order Naviculales.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(navic)
</code></pre>


<h3>Format</h3>

<p>a <code>phylo4d</code> object.
</p>


<h3>Source</h3>

<p>Keck F., Rimet F., Franc A. &amp; Bouchez A. (In press) Phylogenetic signal in diatom ecology: perspectives for aquatic ecosystems biomonitoring. Ecological Applications.
</p>

<hr>
<h2 id='pagelLogLik'>Computes log-likelihood for data and a given value of Pagel's Lambda</h2><span id='topic+pagelLogLik'></span>

<h3>Description</h3>

<p>Computes log-likelihood for data and a given value of Pagel's Lambda
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pagelLogLik(lambda, xr, vcvr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pagelLogLik_+3A_lambda">lambda</code></td>
<td>
<p>the value of Pagel's Lambda</p>
</td></tr>
<tr><td><code id="pagelLogLik_+3A_xr">xr</code></td>
<td>
<p>a vector of data</p>
</td></tr>
<tr><td><code id="pagelLogLik_+3A_vcvr">vcvr</code></td>
<td>
<p>phylogenetic variance-covariance matrix</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Adapted from Liam Revell's R function 'phylosig' in phytools.
</p>

<hr>
<h2 id='phyloCorrelogram'>Phylogenetic correlogram</h2><span id='topic+phyloCorrelogram'></span>

<h3>Description</h3>

<p>This function computes a phylogenetic correlogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyloCorrelogram(
  p4d,
  trait = names(tdata(p4d)),
  dist.phylo = "patristic",
  sigma = NULL,
  n.points = 100,
  ci.bs = 1000,
  ci.conf = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyloCorrelogram_+3A_p4d">p4d</code></td>
<td>
<p>a <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="phyloCorrelogram_+3A_trait">trait</code></td>
<td>
<p>the traits in the <code>phylo4d</code> object to use for the correlogram.
Can be a character vector giving the name of the traits or numbers giving the column index
in the table of the data slot of the <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="phyloCorrelogram_+3A_dist.phylo">dist.phylo</code></td>
<td>
<p>a matrix of phylogenetic distances or a character string specifying a method to compute it.
See Details.</p>
</td></tr>
<tr><td><code id="phyloCorrelogram_+3A_sigma">sigma</code></td>
<td>
<p>a numeric value giving the standard deviation of the normal distribution used
to compute the matrix of phylogenetic weights. If <code>NULL</code> (default),
the function computes a value from the phylogeny.</p>
</td></tr>
<tr><td><code id="phyloCorrelogram_+3A_n.points">n.points</code></td>
<td>
<p>an integer giving the number of points at which to compute the correlogram's statistics.</p>
</td></tr>
<tr><td><code id="phyloCorrelogram_+3A_ci.bs">ci.bs</code></td>
<td>
<p>an integer giving the number of bootstrap replicates for confidence interval estimation.</p>
</td></tr>
<tr><td><code id="phyloCorrelogram_+3A_ci.conf">ci.conf</code></td>
<td>
<p>a value between 0 and 1 giving the confidence level of the confidence interval.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes a correlogram on a continuous scale of phylogenetic distance.
This is achieved by using a collection of specific phylogenetic weights matrices generated
with the &quot;<code>lag-norm</code>&quot; method of <code><a href="#topic+phyloWeights">phyloWeights</a></code> and different values of &quot;<code>mu</code>&quot;.
</p>
<p>The confidence envelope is computed by bootstrapping.
At each iteration, the autocorrelation is re-estimated after re-standardization of the
matrix of phylogenetic weights.
The nonparametric confidence intervals are computed at each lag by first order normal approximation.
Intervals are constrained between 0 and 1.
</p>
<p>If there is one trait, the function computes Moran's I. If there is more than one trait,
the function computes the Mantel's statistic (Oden and Sokal 1986).
</p>
<p>If &quot;<code>dist.phylo</code>&quot; is a character string,
the phylogenetic distance matrix is computed internally
using the function <code><a href="adephylo.html#topic+distTips">distTips</a></code> from the package <span class="pkg">adephylo</span>.
See <code><a href="adephylo.html#topic+distTips">distTips</a></code> for details about the methods.
</p>


<h3>Value</h3>

<p>An object of class &quot;<code>phylocorrelogram</code>&quot;.
</p>


<h3>References</h3>

<p>Oden N.L. &amp; Sokal R.R. (1986) Directional Autocorrelation: An Extension of Spatial Correlograms to Two Dimensions. Systematic Zoology 35, 608-617.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.phylocorrelogram">plot.phylocorrelogram</a></code>,
<code><a href="ape.html#topic+correlogram.formula">correlogram.formula</a></code> in <span class="pkg">ape</span> for correlograms based on taxonomic levels.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(navic)
pc &lt;- phyloCorrelogram(navic)
plot(pc)

## End(Not run)
</code></pre>

<hr>
<h2 id='phylosignal'>phylosignal</h2><span id='topic+phylosignal'></span>

<h3>Description</h3>

<p>A collection of tools to explore the phylogenetic signal in univariate and multivariate data.
The package provides functions to plot traits data against a phylogenetic tree, different
measures and tests for the phylogenetic signal, methods to describe where the signal is
located and a phylogenetic clustering method.
</p>

<hr>
<h2 id='phyloSignal'>Computes phylogenetic signal with different methods</h2><span id='topic+phyloSignal'></span>

<h3>Description</h3>

<p>This function computes phylogenetic signal statistics
(Blomberg's K and K*, Abouheif's Cmean, Moran's I, and Pagel's Lambda) for traits in <code>phylo4d</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyloSignal(
  p4d,
  methods = c("all", "I", "Cmean", "Lambda", "K", "K.star"),
  reps = 999,
  W = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyloSignal_+3A_p4d">p4d</code></td>
<td>
<p>a <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="phyloSignal_+3A_methods">methods</code></td>
<td>
<p>a character vector giving the methods to compute phylogenetic signal (see Details).</p>
</td></tr>
<tr><td><code id="phyloSignal_+3A_reps">reps</code></td>
<td>
<p>an integer. The number of repetitions for the estimation of p.values with randomization.</p>
</td></tr>
<tr><td><code id="phyloSignal_+3A_w">W</code></td>
<td>
<p>an optional matrix of phylogenetic weights to compute Moran's I. By default the matrix
is computed with the function <code><a href="adephylo.html#topic+proxTips">proxTips</a></code> with patristic distances.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>p4d must be a <code>phylo4d</code> object as defined in <span class="pkg">phylobase</span> package.
By default, the <code>methods</code> argument is set to &quot;<code>all</code>&quot; and all the available methods are used.
The user can specify which method(s) to use. Possible methods are
&quot;<code>I</code>&quot; (Gittleman &amp; Kot 1990), &quot;<code>Cmean</code>&quot; (Abouheif 1999)&quot;, &quot;<code>Lambda</code>&quot; (Pagel 1999),
&quot;<code>K</code>&quot; and &quot;<code>K.star</code>&quot; (Blomberg et al. 2003).
</p>


<h3>Value</h3>

<p>A list of two dataframes with the values of statistics and associated
p.values for each tested trait and method.
</p>


<h3>Author(s)</h3>

<p>This function is a general wrapper for C++ subroutines.
C++ code is adapted from R functions in Pavoine and Ricotta (2013)
and Revell (2012).
</p>


<h3>References</h3>

<p>Abouheif E. (1999) A method for testing the assumption of phylogenetic independence in comparative data. Evolutionary Ecology Research 1, 895-909.
Blomberg S.P., Garland Jr T. &amp; Ives A.R. (2003) Testing for phylogenetic signal in comparative data: behavioral traits are more labile. Evolution 57, 717-745.
Gittleman J.L. &amp; Kot M. (1990) Adaptation: Statistics and a null model for estimating phylogenetic effects. Systematic Biology 39, 227-241.
Pagel M. (1999) Inferring the historical patterns of biological evolution. Nature 401, 877-884.
Revell L.J. (2012) phytools: an R package for phylogenetic comparative biology (and other things). Methods in Ecology and Evolution 3, 217-223.
Pavoine S. &amp; Ricotta C. (2013) Testing for Phylogenetic Signal in Biological Traits: The Ubiquity of Cross-Product Statistics. Evolution 67, 828-840.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phyloSimSignal">phyloSimSignal</a></code> .
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(ape)
require(phylobase)
data(navic)
tipData(navic)$rand &lt;- rnorm(17)
tipData(navic)$BM &lt;- rTraitCont(as(navic, "phylo"))
phyloSignal(navic)

</code></pre>

<hr>
<h2 id='phyloSignalBS'>Computes phylogenetic signal for bootstrapped replicates of a phylogeny.</h2><span id='topic+phyloSignalBS'></span>

<h3>Description</h3>

<p>This function computes phylogenetic signal statistics and p-values
for bootsrapped replicates a phylogenetic tree and produce boxplots to represent the results.
This can be useful to check the impact of phylogenetic reconstruction uncertainty on phylogenetic signal.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyloSignalBS(
  p4d,
  multiphylo,
  methods = c("all", "I", "Cmean", "Lambda", "K", "K.star"),
  reps = 999,
  W = NULL,
  pb = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyloSignalBS_+3A_p4d">p4d</code></td>
<td>
<p>a <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="phyloSignalBS_+3A_multiphylo">multiphylo</code></td>
<td>
<p>a <code>multiphylo</code> object containing bootstrapped trees of <code>p4d</code>.</p>
</td></tr>
<tr><td><code id="phyloSignalBS_+3A_methods">methods</code></td>
<td>
<p>a character vector giving the methods to compute phylogenetic signal (see <code><a href="#topic+phyloSignal">phyloSignal</a></code>).</p>
</td></tr>
<tr><td><code id="phyloSignalBS_+3A_reps">reps</code></td>
<td>
<p>an integer. The number of repetitions for the estimation of p.values with randomization.</p>
</td></tr>
<tr><td><code id="phyloSignalBS_+3A_w">W</code></td>
<td>
<p>an optional matrix of phylogenetic weights to compute Moran's I. By default the matrix
is computed with the function <code><a href="adephylo.html#topic+proxTips">proxTips</a></code> with patristic distances.</p>
</td></tr>
<tr><td><code id="phyloSignalBS_+3A_pb">pb</code></td>
<td>
<p>a logical. Should a progress bar be printed? (default <code>TRUE</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Time consumption can be important if there are many bootraped trees and tested traits.
</p>


<h3>Value</h3>

<p>The data generated are returned invisibly as a list.
</p>

<hr>
<h2 id='phyloSignalINT'>Computes phylogenetic signal at each internal node of a phylogeny</h2><span id='topic+phyloSignalINT'></span>

<h3>Description</h3>

<p>This function computes phylogenetic signal statistics and p-values for a given trait
and a given method at each internal node of a phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyloSignalINT(
  p4d,
  trait = names(tipData(p4d))[1],
  method = "Cmean",
  reps = 999,
  W = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyloSignalINT_+3A_p4d">p4d</code></td>
<td>
<p>a <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="phyloSignalINT_+3A_trait">trait</code></td>
<td>
<p>a character string giving the trait to use to compute the signal. By default the first trait is taken from <code>p4d</code>.</p>
</td></tr>
<tr><td><code id="phyloSignalINT_+3A_method">method</code></td>
<td>
<p>a character vector giving the method to use to compute phylogenetic signal
(default is &quot;<code>Cmean</code>&quot;; see <code><a href="#topic+phyloSignal">phyloSignal</a></code>).</p>
</td></tr>
<tr><td><code id="phyloSignalINT_+3A_reps">reps</code></td>
<td>
<p>an integer. The number of repetitions for the estimation of p.values with randomization.</p>
</td></tr>
<tr><td><code id="phyloSignalINT_+3A_w">W</code></td>
<td>
<p>an optional matrix of phylogenetic weights to compute Moran's I. By default the matrix
is computed with the function <code><a href="adephylo.html#topic+proxTips">proxTips</a></code> with patristic distances.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>phylo4d</code> object with phylogenetic signal statistics and p-values as nodes associated data.
</p>

<hr>
<h2 id='phyloSim'>Simulate the behaviour of phylogenetic signal statistics with a given phylogeny</h2><span id='topic+phyloSim'></span>

<h3>Description</h3>

<p>This function simulates different phylogenetic signal statistics
for a given phylogenetic tree along a gradient of Brownian Motion influence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyloSim(
  tree,
  methods = c("all", "I", "Cmean", "Lambda", "K", "K.star"),
  nsim = 99,
  reps = 999,
  W = NULL,
  model = "BM",
  pb = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyloSim_+3A_tree">tree</code></td>
<td>
<p>a <code>phylo</code>, <code>phylo4</code> or <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="phyloSim_+3A_methods">methods</code></td>
<td>
<p>a character vector giving the methods to compute phylogenetic signal (see Details).</p>
</td></tr>
<tr><td><code id="phyloSim_+3A_nsim">nsim</code></td>
<td>
<p>a numeric value. Number of simulated traits at each step in the gradient.</p>
</td></tr>
<tr><td><code id="phyloSim_+3A_reps">reps</code></td>
<td>
<p>a numeric value. Number of repetitions for the estimation of p.values with randomization.</p>
</td></tr>
<tr><td><code id="phyloSim_+3A_w">W</code></td>
<td>
<p>an optional matrix of phylogenetic weights to compute Moran's I. By default the matrix
is computed with the function <code><a href="adephylo.html#topic+proxTips">proxTips</a></code> with patristic distances.</p>
</td></tr>
<tr><td><code id="phyloSim_+3A_model">model</code></td>
<td>
<p>the model to use for traits simulation (only &quot;<code>BM</code>&quot;, default, is available).</p>
</td></tr>
<tr><td><code id="phyloSim_+3A_pb">pb</code></td>
<td>
<p>a logical. Should a progress bar be printed? (default <code>TRUE</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, the <code>methods</code> argument is set to &quot;<code>all</code>&quot; and all the available methods are used.
The user can specify which method(s) to use. Possible values are
&quot;<code>I</code>&quot;, &quot;<code>Cmean</code>&quot;, &quot;<code>Lambda</code>&quot;, &quot;<code>K</code>&quot; and &quot;<code>K.star</code>&quot;,
see <code><a href="#topic+phyloSignal">phyloSignal</a></code> for further details.
</p>


<h3>Value</h3>

<p>An object of class <code>phylosim</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phyloSimSignal">phyloSimSignal</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(navic)
psim &lt;- phyloSim(navic)
plot(psim)
plot.phylosim(psim, what = "pval", stacked.methods = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='phyloSimSignal'>Phylogenetic signal estimation as a fraction of a Brownian Motion process.</h2><span id='topic+phyloSimSignal'></span>

<h3>Description</h3>

<p>This function crosses traits data with simulations to estimate the phylogenetic signal as a
fraction of a Brownian Motion process. This is experimental.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyloSimSignal(p4d, phylosim, quantiles = c(0.05, 0.95))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyloSimSignal_+3A_p4d">p4d</code></td>
<td>
<p>a <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="phyloSimSignal_+3A_phylosim">phylosim</code></td>
<td>
<p>a <code>phylosim</code> object.</p>
</td></tr>
<tr><td><code id="phyloSimSignal_+3A_quantiles">quantiles</code></td>
<td>
<p>a vector of two numeric values between 0 and 1 giving the minimum
and the maximum quantiles to estimate statistics fluctuations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>phylosimsignal</code> with a <code>print</code> and a <code>plot</code> method.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+phyloSignal">phyloSignal</a></code>, <code><a href="#topic+phyloSim">phyloSim</a></code>.
</p>

<hr>
<h2 id='phyloWeights'>Phylogenetic weights matrix</h2><span id='topic+phyloWeights'></span>

<h3>Description</h3>

<p>This function can be used to compute a phylogenetic weights matrix
with different methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyloWeights(
  tree,
  dist.phylo = "patristic",
  method = "lag-norm",
  mu = 0,
  sigma = 5,
  dmax = 10,
  alpha = 1,
  beta = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyloWeights_+3A_tree">tree</code></td>
<td>
<p>a <code>phylo</code>, <code>phylo4</code> or <code>phylo4d</code> object.</p>
</td></tr>
<tr><td><code id="phyloWeights_+3A_dist.phylo">dist.phylo</code></td>
<td>
<p>a character string specifying the method used to compute phylogenetic distances.
Available distances are &quot;<code>patristic</code>&quot;,&quot;<code>nNodes</code>&quot;,&quot;<code>Abouheif</code>&quot; and &quot;<code>sumDD</code>&quot;.
See Details.</p>
</td></tr>
<tr><td><code id="phyloWeights_+3A_method">method</code></td>
<td>
<p>a method to compute phylogenetic weights from phylogenetic distances.
Available methods are &quot;<code>lag-norm</code>&quot;, &quot;<code>clade</code>&quot;, &quot;<code>inverse</code>&quot; and &quot;<code>exponential</code>&quot;.
See Details.</p>
</td></tr>
<tr><td><code id="phyloWeights_+3A_mu">mu</code></td>
<td>
<p>a numeric value giving the mean of the distribution if <code>method</code> is <code>lag-norm</code>.
This is a phylogenetic distance.</p>
</td></tr>
<tr><td><code id="phyloWeights_+3A_sigma">sigma</code></td>
<td>
<p>a numeric value giving the standard deviation
of the distribution if <code>method</code> is <code>lag-norm</code>.</p>
</td></tr>
<tr><td><code id="phyloWeights_+3A_dmax">dmax</code></td>
<td>
<p>the maximum phylogenetic distance to use to delineate clades.</p>
</td></tr>
<tr><td><code id="phyloWeights_+3A_alpha">alpha</code></td>
<td>
<p>a numeric value giving the exponent to use if <code>method</code> is <code>inverse</code>.</p>
</td></tr>
<tr><td><code id="phyloWeights_+3A_beta">beta</code></td>
<td>
<p>a numeric value giving the factor to use if <code>method</code> is <code>exponential</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Method &quot;<code>inverse</code>&quot;: </p>
<p style="text-align: center;"><code class="reqn">\frac{1}{d^{\alpha}}</code>
</p>

<p>The phylogenetic distance matrix is computed internally
using the function <code><a href="adephylo.html#topic+distTips">distTips</a></code> from the package <span class="pkg">adephylo</span>.
See <code><a href="adephylo.html#topic+distTips">distTips</a></code> for details about the methods.
</p>


<h3>Value</h3>

<p>A square matrix of phylogenetic weights whose sums of rows is 1.
</p>


<h3>See Also</h3>

<p><code><a href="adephylo.html#topic+proxTips">proxTips</a></code> in <span class="pkg">adephylo</span>.
</p>

<hr>
<h2 id='plot.graphclust'>Plot phylogenetically constrained  clustering</h2><span id='topic+plot.graphclust'></span>

<h3>Description</h3>

<p>This function produces three plots (selectable by <code>which</code>):
a plot of edges selection based on phylogenetic against trait distances of taxa pairs,
a plot of the graph produced with the selected edges
and a plot of the clustered phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'graphclust'
plot(
  x,
  which = c("selection", "graph", "tree"),
  ask = TRUE,
  colored = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.graphclust_+3A_x">x</code></td>
<td>
<p>a <code>graphclust</code> object as produced by <code><a href="#topic+graphClust">graphClust</a></code>.</p>
</td></tr>
<tr><td><code id="plot.graphclust_+3A_which">which</code></td>
<td>
<p>a character vector to select plots.
Must be one or more of &quot;<code>selection</code>&quot;, &quot;<code>graph</code>&quot;, &quot;<code>tree</code>&quot;.</p>
</td></tr>
<tr><td><code id="plot.graphclust_+3A_ask">ask</code></td>
<td>
<p>logical if <code>TRUE</code> (default), the user is asked before each plot.</p>
</td></tr>
<tr><td><code id="plot.graphclust_+3A_colored">colored</code></td>
<td>
<p>logical indicating if plots must include colors.</p>
</td></tr>
<tr><td><code id="plot.graphclust_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to or from other methods. They are currently ignored in this function.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(navic)
gC &lt;- graphClust(navic, lim.phylo = 1, lim.trait = 2, scale.lim = FALSE)
plot.graphclust(gC, which = "selection", ask = FALSE)
plot.graphclust(gC, which = "graph", ask = FALSE)
plot.graphclust(gC, which = "tree", ask = FALSE)

</code></pre>

<hr>
<h2 id='plot.phylocorrelogram'>Plot a phylogenetic correlogram</h2><span id='topic+plot.phylocorrelogram'></span>

<h3>Description</h3>

<p>This function plots phylogenetic correlograms produced by <code><a href="#topic+phyloCorrelogram">phyloCorrelogram</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylocorrelogram'
plot(
  x,
  show.ci = TRUE,
  show.h0 = TRUE,
  show.test = TRUE,
  xlab = "Phylogenetic distance",
  ylab = "Correlation",
  main = "Phylogenetic correlogram",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.phylocorrelogram_+3A_x">x</code></td>
<td>
<p>a <code>phylocorrelogram</code> object.</p>
</td></tr>
<tr><td><code id="plot.phylocorrelogram_+3A_show.ci">show.ci</code></td>
<td>
<p>a logical indicating whether to plot the confidence interval envelop (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="plot.phylocorrelogram_+3A_show.h0">show.h0</code></td>
<td>
<p>a logical indicating whether to plot the the line showing the expected
value of Moran's I under the null hypothesis of no phylogenetic autocorrelation (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="plot.phylocorrelogram_+3A_show.test">show.test</code></td>
<td>
<p>a logical indicating whether to plot indicator of significance (default <code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="plot.phylocorrelogram_+3A_xlab">xlab</code></td>
<td>
<p>a label for the x axis.</p>
</td></tr>
<tr><td><code id="plot.phylocorrelogram_+3A_ylab">ylab</code></td>
<td>
<p>a label for the y axis.</p>
</td></tr>
<tr><td><code id="plot.phylocorrelogram_+3A_main">main</code></td>
<td>
<p>a main title for the plot</p>
</td></tr>
<tr><td><code id="plot.phylocorrelogram_+3A_...">...</code></td>
<td>
<p>other graphical parameters passed to the <code>plot</code> function.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(navic)
pc &lt;- phyloCorrelogram(navic)
plot(pc)

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.phylosim'>Plot <code>phylosim</code> object</h2><span id='topic+plot.phylosim'></span>

<h3>Description</h3>

<p>This function plots a <code>phylosim</code> object to visualize the behaviour of
phylogenetic signal statistics for a given phylogenetic tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylosim'
plot(
  x,
  what = c("stat", "pval"),
  stacked.methods = FALSE,
  quantiles = c(0.05, 0.95),
  col = 1:5,
  legend = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.phylosim_+3A_x">x</code></td>
<td>
<p>a <code>phylosim</code> object.</p>
</td></tr>
<tr><td><code id="plot.phylosim_+3A_what">what</code></td>
<td>
<p>what to represent on the plot.
Can be the statistics used to measure the signal (&quot;<code>stat</code>&quot;) or the p-values (&quot;<code>pval</code>&quot;).</p>
</td></tr>
<tr><td><code id="plot.phylosim_+3A_stacked.methods">stacked.methods</code></td>
<td>
<p>a logical. If different methods have been used, should they
be plotted on the same graphic (<code>TRUE</code>) or not (<code>FALSE</code>, default).</p>
</td></tr>
<tr><td><code id="plot.phylosim_+3A_quantiles">quantiles</code></td>
<td>
<p>a vector of two numeric values between 0 and 1
giving the minimum and the maximum quantiles to plot.
Set to <code>NULL</code> to not plot quantiles.</p>
</td></tr>
<tr><td><code id="plot.phylosim_+3A_col">col</code></td>
<td>
<p>a vector of colors for the different methods.</p>
</td></tr>
<tr><td><code id="plot.phylosim_+3A_legend">legend</code></td>
<td>
<p>a logical. If <code>stacked.methods</code> is set to <code>TRUE</code>,
should a legend be printed to differentiate the different methods?</p>
</td></tr>
<tr><td><code id="plot.phylosim_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to or from other methods.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+phyloSim">phyloSim</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(navic)
psim &lt;- phyloSim(navic)
plot(psim)
plot.phylosim(psim, what = "pval", stacked.methods = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='plot.phylosimsignal'>Plot signal estimation as a fraction of a Brownian Motion process.</h2><span id='topic+plot.phylosimsignal'></span>

<h3>Description</h3>

<p>Plot signal estimation as a fraction of a Brownian Motion process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylosimsignal'
plot(
  x,
  methods = NULL,
  traits = NULL,
  stacked.methods = FALSE,
  stacked.traits = FALSE,
  print.quantiles = TRUE,
  col = 1:5,
  legend = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.phylosimsignal_+3A_x">x</code></td>
<td>
<p>an object of class <code>phylosimsignal</code>.</p>
</td></tr>
<tr><td><code id="plot.phylosimsignal_+3A_methods">methods</code></td>
<td>
<p>a character vector giving the methods
(included in the <code>phylosimsignal</code> object) to plot.</p>
</td></tr>
<tr><td><code id="plot.phylosimsignal_+3A_traits">traits</code></td>
<td>
<p>a character vector giving the traits
(included in the <code>phylosimsignal</code> object) to plot.</p>
</td></tr>
<tr><td><code id="plot.phylosimsignal_+3A_stacked.methods">stacked.methods</code></td>
<td>
<p>If different methods have been used, should they
be plotted on the same graphic (<code>TRUE</code>) or not (<code>FALSE</code>, default).</p>
</td></tr>
<tr><td><code id="plot.phylosimsignal_+3A_stacked.traits">stacked.traits</code></td>
<td>
<p>If different traits have been used, should they
be plotted on the same graphic (<code>TRUE</code>) or not (<code>FALSE</code>, default).</p>
</td></tr>
<tr><td><code id="plot.phylosimsignal_+3A_print.quantiles">print.quantiles</code></td>
<td>
<p>logical stating whether quantiles should be plotted.</p>
</td></tr>
<tr><td><code id="plot.phylosimsignal_+3A_col">col</code></td>
<td>
<p>a vector of colors for the different methods.</p>
</td></tr>
<tr><td><code id="plot.phylosimsignal_+3A_legend">legend</code></td>
<td>
<p>a logical. If <code>stacked.methods</code> is set to <code>TRUE</code>,
should a legend be printed to differentiate the different methods?</p>
</td></tr>
<tr><td><code id="plot.phylosimsignal_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to or from other methods.</p>
</td></tr>
</table>

<hr>
<h2 id='print.graphclust'>#' Print results of phylogenetically constrained  clustering</h2><span id='topic+print.graphclust'></span>

<h3>Description</h3>

<p>#' Print results of phylogenetically constrained  clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'graphclust'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.graphclust_+3A_x">x</code></td>
<td>
<p>a <code>graphclust</code> object as produced by <code><a href="#topic+graphClust">graphClust</a></code>.</p>
</td></tr>
<tr><td><code id="print.graphclust_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to or from other methods.</p>
</td></tr>
</table>

<hr>
<h2 id='print.phylosimsignal'>Print signal estimation as a fraction of a Brownian Motion process.</h2><span id='topic+print.phylosimsignal'></span>

<h3>Description</h3>

<p>Print signal estimation as a fraction of a Brownian Motion process.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylosimsignal'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.phylosimsignal_+3A_x">x</code></td>
<td>
<p>an object of class <code>phylosimsignal</code>.</p>
</td></tr>
<tr><td><code id="print.phylosimsignal_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to or from other methods.</p>
</td></tr>
</table>

<hr>
<h2 id='read.p4d'>Read a phylo4d object from files</h2><span id='topic+read.p4d'></span>

<h3>Description</h3>

<p>This function creates an object of class <code>phylo4d</code> by combining
a phylogenetic tree and its associated tips data stored in two distinct files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.p4d(phylo.file, data.file, phylo.format = "newick", data.format = "table")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.p4d_+3A_phylo.file">phylo.file</code></td>
<td>
<p>the name of the file which the phylogenetic tree is to be read from.</p>
</td></tr>
<tr><td><code id="read.p4d_+3A_data.file">data.file</code></td>
<td>
<p>the name of the file which the tips data are to be read from.</p>
</td></tr>
<tr><td><code id="read.p4d_+3A_phylo.format">phylo.format</code></td>
<td>
<p>the format of the phylogenetic tree provided.
Possible formats are &quot;<code>newick</code>&quot; and &quot;<code>nexus</code>&quot;.</p>
</td></tr>
<tr><td><code id="read.p4d_+3A_data.format">data.format</code></td>
<td>
<p>the format of the table with tips data.
Possible formats are &quot;<code>table</code>&quot;, &quot;<code>csv</code>&quot;, &quot;<code>csv2</code>&quot;, &quot;<code>delim</code>&quot; and &quot;<code>delim2</code>&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>phylo.file</code> and <code>data.file</code> can be provided as objects of mode character
or double-quoted strings.
</p>
<p>The phylogenetic tree can be imported in two formats.
</p>

<ul>
<li><p> The <code>newick</code> format refers
to the simple parenthetic format known as the Newick or New Hampshire format. The tree is
read by calling the function <code><a href="ape.html#topic+read.tree">read.tree</a></code> of the <span class="pkg">ape</span> package.
</p>
</li>
<li><p> The <code>nexus</code> format refers to NEXUS format. The tree is
read by calling the function <code><a href="ape.html#topic+read.nexus">read.nexus</a></code> of the <span class="pkg">ape</span> package.
</p>
</li></ul>

<p>Tips data are imported from a table formatted file. The different formats allow to use
different separator and decimal characters.
They correspond to the variants of <code><a href="utils.html#topic+read.table">read.table</a></code>:
</p>

<ul>
<li> <p><code>table</code> use <code><a href="utils.html#topic+read.table">read.table</a></code> with default settings.
</p>
</li>
<li> <p><code>csv</code> use <code><a href="utils.html#topic+read.csv">read.csv</a></code> with default settings.
</p>
</li>
<li> <p><code>csv2</code> use <code><a href="utils.html#topic+read.csv2">read.csv2</a></code> with default settings.
</p>
</li>
<li> <p><code>delim</code> use <code><a href="utils.html#topic+read.delim">read.delim</a></code> with default settings.
</p>
</li>
<li> <p><code>delim2</code> use <code><a href="utils.html#topic+read.delim2">read.delim2</a></code> with default settings.
</p>
</li></ul>



<h3>Value</h3>

<p>An object of class <code>phylo4d</code>.
</p>


<h3>See Also</h3>

<p><code><a href="phylobase.html#topic+phylo4d">phylo4d</a></code> to create a <code>phylo4d</code> object.
</p>

<hr>
<h2 id='rTraitContWeight'>Traits Values Simulation</h2><span id='topic+rTraitContWeight'></span>

<h3>Description</h3>

<p>This function simulates the evolution of continuous characters along a phylogeny.
Traits values can be more or less influenced by the provided model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rTraitContWeight(tree, model = "BM", weight = 1, as.p4d = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rTraitContWeight_+3A_tree">tree</code></td>
<td>
<p>an object of class &quot;<code>phylo</code>.</p>
</td></tr>
<tr><td><code id="rTraitContWeight_+3A_model">model</code></td>
<td>
<p>the model to use (&quot;<code>BM</code>&quot; or &quot;<code>OU</code>&quot;).</p>
</td></tr>
<tr><td><code id="rTraitContWeight_+3A_weight">weight</code></td>
<td>
<p>a numeric vector with values ranging between 0 and 1
giving the balance between pure random evolution and the selected model.</p>
</td></tr>
<tr><td><code id="rTraitContWeight_+3A_as.p4d">as.p4d</code></td>
<td>
<p>logical. Should the phylogenetic tree and the simulated data coerced to a <code>phylo4d</code> object?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class phyloSimSignal.
</p>

<hr>
<h2 id='subsetPhyloSimSignal'>phyloSimSignal subsetting</h2><span id='topic+subsetPhyloSimSignal'></span>

<h3>Description</h3>

<p>This function subset all the components of a phyloSimSignal object following the methods and traits provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsetPhyloSimSignal(x, methods, traits)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsetPhyloSimSignal_+3A_x">x</code></td>
<td>
<p>an object of class <code>phyloSimSignal</code> to subset.</p>
</td></tr>
<tr><td><code id="subsetPhyloSimSignal_+3A_methods">methods</code></td>
<td>
<p>a character vector giving the methods to keep.</p>
</td></tr>
<tr><td><code id="subsetPhyloSimSignal_+3A_traits">traits</code></td>
<td>
<p>a character vector giving the traits to keep.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>phyloSimSignal</code> object
</p>

<hr>
<h2 id='white2red'>Color Palette</h2><span id='topic+white2red'></span>

<h3>Description</h3>

<p>A simple color palette for gridplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>white2red(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="white2red_+3A_n">n</code></td>
<td>
<p>the number of colors to be in the palette.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
