<!DOCTYPE html><html><head><title>Help for package phylogram</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {phylogram}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#as.cladogram'><p>Apply unweighted branch lengths.</p></a></li>
<li><a href='#as.dendrogram.phylo'><p>Convert a &quot;phylo&quot; object to a dendrogram.</p></a></li>
<li><a href='#as.phylo.dendrogram'><p>Convert a dendrogram to a &quot;phylo&quot; object.</p></a></li>
<li><a href='#ladder'><p>Reorder tree branches in ladderized pattern.</p></a></li>
<li><a href='#phylogram'><p>Dendrograms for evolutionary analysis.</p></a></li>
<li><a href='#prune'><p>Remove tree nodes by regular expression pattern matching.</p></a></li>
<li><a href='#read.dendrogram'><p>Read a dendrogram from parenthetic text.</p></a></li>
<li><a href='#remidpoint'><p>Set dendrogram attributes for a nested list.</p></a></li>
<li><a href='#reposition'><p>Reset dendrogram height attributes.</p></a></li>
<li><a href='#write.dendrogram'><p>Write a dendrogram object to parenthetic text.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Dendrograms for Evolutionary Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Shaun Wilkinson &lt;shaunpwilkinson@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains functions for developing phylogenetic trees as
    deeply-nested lists ("dendrogram" objects).
    Enables bi-directional conversion between dendrogram and
    "phylo" objects
    (see Paradis et al (2004) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtg412">doi:10.1093/bioinformatics/btg412</a>&gt;),
    and features several tools for command-line tree
    manipulation and import/export via Newick parenthetic text.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://github.com/ropensci/phylogram">http://github.com/ropensci/phylogram</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="http://github.com/ropensci/phylogram/issues">http://github.com/ropensci/phylogram/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape (&ge; 4.0), methods, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>dendextend, knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-06-25 11:46:26 UTC; shaun</td>
</tr>
<tr>
<td>Author:</td>
<td>Shaun Wilkinson <a href="https://orcid.org/0000-0002-7332-7931"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Simon Davy <a href="https://orcid.org/0000-0003-3584-5356"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-06-25 15:24:56 UTC</td>
</tr>
</table>
<hr>
<h2 id='as.cladogram'>Apply unweighted branch lengths.</h2><span id='topic+as.cladogram'></span>

<h3>Description</h3>

<p>This function sets the 'height' attributes of all leaf nodes to zero and
progressively resets the heights of the inner nodes by single incremental
units in a bottom-up fashion.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.cladogram(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.cladogram_+3A_x">x</code></td>
<td>
<p>an object of class <code>"dendrogram"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code>"dendrogram"</code>.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  x &lt;- read.dendrogram(text = "(A:0.1,B:0.2,(C:0.3,D:0.4):0.5);")
  plot(x, horiz = TRUE)
  x &lt;- as.cladogram(x)
  plot(x, horiz = TRUE)
</code></pre>

<hr>
<h2 id='as.dendrogram.phylo'>Convert a &quot;phylo&quot; object to a dendrogram.</h2><span id='topic+as.dendrogram.phylo'></span>

<h3>Description</h3>

<p>This function converts a &quot;phylo&quot; object (Paradis et al 2004)
to a dendrogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylo'
as.dendrogram(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.dendrogram.phylo_+3A_object">object</code></td>
<td>
<p>an object of class &quot;phylo&quot;.</p>
</td></tr>
<tr><td><code id="as.dendrogram.phylo_+3A_...">...</code></td>
<td>
<p>further arguments to be passed between methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class &quot;dendrogram&quot;.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>References</h3>

<p>Paradis E, Claude J, Strimmer K, (2004) APE: analyses of phylogenetics
and evolution in R language. <em>Bioinformatics</em> <strong>20</strong>, 289-290.
</p>
<p>Paradis E (2008) Definition of Formats for Coding Phylogenetic Trees in R.
<a href="http://ape-package.ird.fr/misc/FormatTreeR_24Oct2012.pdf">http://ape-package.ird.fr/misc/FormatTreeR_24Oct2012.pdf</a>
</p>
<p>Paradis E (2012) Analysis of Phylogenetics and Evolution with R
(Second Edition). Springer, New York.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  newick &lt;- "(A:0.1,B:0.2,(C:0.3,D:0.4):0.5);"
  x &lt;- read.dendrogram(text = newick)
  y &lt;- as.phylo(x)
  z &lt;- as.dendrogram(y)
  identical(x, z)
</code></pre>

<hr>
<h2 id='as.phylo.dendrogram'>Convert a dendrogram to a &quot;phylo&quot; object.</h2><span id='topic+as.phylo.dendrogram'></span>

<h3>Description</h3>

<p>This function converts a dendrogram into an object of class &quot;phylo&quot;
(see Paradis et al 2004).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'dendrogram'
as.phylo(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.phylo.dendrogram_+3A_x">x</code></td>
<td>
<p>a dendrogram.</p>
</td></tr>
<tr><td><code id="as.phylo.dendrogram_+3A_...">...</code></td>
<td>
<p>further arguments to be passed between methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class &quot;phylo&quot;.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>References</h3>

<p>Paradis E, Claude J, Strimmer K, (2004) APE: analyses of phylogenetics
and evolution in R language. <em>Bioinformatics</em> <strong>20</strong>, 289-290.
</p>
<p>Paradis E (2008) Definition of Formats for Coding Phylogenetic Trees in R.
<a href="http://ape-package.ird.fr/misc/FormatTreeR_24Oct2012.pdf">http://ape-package.ird.fr/misc/FormatTreeR_24Oct2012.pdf</a>
</p>
<p>Paradis E (2012) Analysis of Phylogenetics and Evolution with R
(Second Edition). Springer, New York.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  newick &lt;- "(A:0.1,B:0.2,(C:0.3,D:0.4):0.5);"
  x &lt;- read.dendrogram(text = newick)
  y &lt;- as.phylo(x)
  z &lt;- as.dendrogram(y)
</code></pre>

<hr>
<h2 id='ladder'>Reorder tree branches in ladderized pattern.</h2><span id='topic+ladder'></span>

<h3>Description</h3>

<p>This function ladderizes the branches of a <code>dendrogram</code> object
to aid in visual interpretation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ladder(x, decreasing = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ladder_+3A_x">x</code></td>
<td>
<p>an object of class <code>"dendrogram"</code>.</p>
</td></tr>
<tr><td><code id="ladder_+3A_decreasing">decreasing</code></td>
<td>
<p>logical indicating whether the tree should be
ladderized upwards or downwards. Defaults to FALSE (downwards).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is the <code>dendrogram</code> analogue of the
<code><a href="ape.html#topic+ladderize">ladderize</a></code> function in the <code><a href="ape.html#topic+ape">ape</a></code>
package (Paradis et al 2004, 2012).
</p>


<h3>Value</h3>

<p>Returns an object of class <code>dendrogram</code>.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>References</h3>

<p>Paradis E, Claude J, Strimmer K, (2004) APE: analyses of phylogenetics
and evolution in R language. <em>Bioinformatics</em> <strong>20</strong>, 289-290.
</p>
<p>Paradis E (2012) Analysis of Phylogenetics and Evolution with R
(Second Edition). Springer, New York.
</p>


<h3>See Also</h3>

<p>The <code><a href="ape.html#topic+ladderize">ladderize</a></code> function in the
<code><a href="ape.html#topic+ape">ape</a></code> package performs a similar operation for objects
of class <code>"phylo"</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  x &lt;- read.dendrogram(text = "(A:0.1,B:0.2,(C:0.3,D:0.4):0.5);")
  plot(x, horiz = TRUE)
  x &lt;- ladder(x, decreasing = TRUE)
  plot(x, horiz = TRUE)
</code></pre>

<hr>
<h2 id='phylogram'>Dendrograms for evolutionary analysis.</h2><span id='topic+phylogram'></span><span id='topic+phylogram-package'></span>

<h3>Description</h3>

<p>The <strong>phylogram</strong> R package is a tool for for developing
phylogenetic trees as deeply-nested lists known as &quot;dendrogram&quot; objects.
It provides functions for conversion between &quot;dendrogram&quot; and
&quot;phylo&quot; class objects, as well as several tools for command-line tree
manipulation and import/export via Newick parenthetic text.
This improves accessibility to the comprehensive range of object-specific
analytical and tree-visualization functions found across a wide array of
bioinformatic R packages.
</p>


<h3>Functions</h3>

<p>A brief description of the primary <span class="pkg">phylogram</span> functions are
provided with links to their help pages below.
</p>


<h3>File import/export</h3>


<ul>
<li> <p><code><a href="#topic+read.dendrogram">read.dendrogram</a></code> reads a Newick
parenthetic text string from a file or text connection
and creates an object of class <code>"dendrogram"</code>
</p>
</li>
<li> <p><code><a href="#topic+write.dendrogram">write.dendrogram</a></code> outputs an object of class
<code>"dendrogram"</code> to a text string or file in Newick
format
</p>
</li></ul>



<h3>Object conversion</h3>


<ul>
<li> <p><code><a href="#topic+as.phylo.dendrogram">as.phylo.dendrogram</a></code> converts a dendrogram to
an object of class &quot;phylo&quot;
<code>"dendrogram"</code>
</p>
</li>
<li> <p><code><a href="#topic+as.dendrogram.phylo">as.dendrogram.phylo</a></code> converts a &quot;phylo&quot; object
to a dendrogram
</p>
</li></ul>



<h3>Tree editing and manipulation</h3>


<ul>
<li> <p><code><a href="#topic+prune">prune</a></code> remove branches from a <code>dendrogram</code> object
based on regular expression pattern matching
</p>
</li>
<li> <p><code><a href="#topic+ladder">ladder</a></code> reorders the branches of a <code>dendrogram</code>
object to aid visualization
</p>
</li>
<li> <p><code><a href="#topic+remidpoint">remidpoint</a></code> recursively sets &quot;midpoint&quot; and &quot;members&quot;
attributes for a nested list/<code>dendrogram</code> object
</p>
</li>
<li> <p><code><a href="#topic+reposition">reposition</a></code> shifts a <code>dendrogram</code> object up or
down (or sideways if plotted horizontally)
</p>
</li>
<li> <p><code><a href="#topic+as.cladogram">as.cladogram</a></code> modifies the &quot;height&quot; attributes of the
nodes such that all leaves terminate at zero
</p>
</li></ul>


<hr>
<h2 id='prune'>Remove tree nodes by regular expression pattern matching.</h2><span id='topic+prune'></span>

<h3>Description</h3>

<p><code>"prune"</code> takes an object of class <code>"dendrogram"</code> and
removes all branches whose branch labels match a given regular
expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prune(tree, pattern, keep = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prune_+3A_tree">tree</code></td>
<td>
<p>an object of class <code>"dendrogram"</code>.</p>
</td></tr>
<tr><td><code id="prune_+3A_pattern">pattern</code></td>
<td>
<p>a regular expression.</p>
</td></tr>
<tr><td><code id="prune_+3A_keep">keep</code></td>
<td>
<p>logical indicating whether the nodes whose labels match
the regular expression provided in &quot;pattern&quot; should be
kept and the remainder discarded. Defaults to FALSE.
Note that nodes without &quot;label&quot; attributes are ignored.</p>
</td></tr>
<tr><td><code id="prune_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code>grepl</code> and <code>gsub</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function recursively tests the &quot;label&quot; attribute of each
dendrogram node (including non-leaf inner nodes if applicable) for
the specified pattern, removing those that register a positive hit.
Note that positive matching inner nodes are removed along with all of
their sub-nodes, regardless of whether the &quot;label&quot; attributes of the
sub-nodes match the pattern.
</p>


<h3>Value</h3>

<p>Returns an object of class <code>"dendrogram"</code>.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>See Also</h3>

<p>The <code><a href="ape.html#topic+drop.tip">drop.tip</a></code> function in the
<code><a href="ape.html#topic+ape">ape</a></code> package performs a similar operation for objects
of class <code>"phylo"</code>. See <code><a href="base.html#topic+regex">regex</a></code> for help with
compiling regular expressions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  x &lt;- read.dendrogram(text = "(A:0.1,B:0.2,(C:0.3,D:0.4):0.5);")
  plot(x, horiz = TRUE)
  x &lt;- prune(x, pattern = "^A$")
  plot(x, horiz = TRUE)
</code></pre>

<hr>
<h2 id='read.dendrogram'>Read a dendrogram from parenthetic text.</h2><span id='topic+read.dendrogram'></span>

<h3>Description</h3>

<p>This function wraps the <code><a href="ape.html#topic+read.tree">read.tree</a></code> parser from the
<code><a href="ape.html#topic+ape">ape</a></code> package to read a phylogenetic tree from
parenthetic text in the Newick/New Hampshire format, and
converts it to object of class &quot;dendrogram&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.dendrogram(file = "", text = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.dendrogram_+3A_file">file</code></td>
<td>
<p>character string giving a valid path to the file from
which to read the data.</p>
</td></tr>
<tr><td><code id="read.dendrogram_+3A_text">text</code></td>
<td>
<p>optional character string in lieu of a &quot;file&quot; argument.
If a text argument is provided instead of a file path, the data
are read via a text connection.</p>
</td></tr>
<tr><td><code id="read.dendrogram_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to
<code><a href="ape.html#topic+read.tree">read.tree</a></code> (which may then be passed on to
<code>scan</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an object of class <code>"dendrogram"</code>.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>References</h3>

<p>Felsenstein J (1986) The Newick tree format.
<a href="http://evolution.genetics.washington.edu/phylip/newicktree.html">http://evolution.genetics.washington.edu/phylip/newicktree.html</a>
</p>
<p>Olsen G (1990) Interpretation of the &quot;Newick's 8:45&quot; tree format standard.
<a href="http://evolution.genetics.washington.edu/phylip/newick_doc.html">http://evolution.genetics.washington.edu/phylip/newick_doc.html</a>
</p>
<p>Paradis E, Claude J, Strimmer K, (2004) APE: analyses of phylogenetics
and evolution in R language. <em>Bioinformatics</em> <strong>20</strong>, 289-290.
</p>
<p>Paradis E (2008) Definition of Formats for Coding Phylogenetic Trees in R.
<a href="http://ape-package.ird.fr/misc/FormatTreeR_24Oct2012.pdf">http://ape-package.ird.fr/misc/FormatTreeR_24Oct2012.pdf</a>
</p>
<p>Paradis E (2012) Analysis of Phylogenetics and Evolution with R
(Second Edition). Springer, New York.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+write.dendrogram">write.dendrogram</a></code> writes an object of
class <code>"dendrogram"</code> to a Newick text string.
The <code><a href="ape.html#topic+read.tree">read.tree</a></code> function in the
<code><a href="ape.html#topic+ape">ape</a></code> package parses objects
of class <code>"phylo"</code> and <code>"multiPhylo"</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  x &lt;- read.dendrogram(text = "(A:0.1,B:0.2,(C:0.3,D:0.4):0.5);")
  plot(x, horiz = TRUE)
</code></pre>

<hr>
<h2 id='remidpoint'>Set dendrogram attributes for a nested list.</h2><span id='topic+remidpoint'></span>

<h3>Description</h3>

<p><code>remidpoint</code> is a helper function used for manually creating
<code>"dendrogram"</code> objects from nested lists. The function
recursively assigns the necessary 'midpoint', 'members', and
'leaf' attributes at each node.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remidpoint(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remidpoint_+3A_x">x</code></td>
<td>
<p>a nested list, possibly of class <code>"dendrogram"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a nested list, or an object of class <code>"dendrogram"</code>
depending on the class of the input object.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## manually create a small dendrogram with three members, A, B and C
  x &lt;- list("A", list("B", "C"))
  attr(x[[1]], "leaf") &lt;- TRUE
  attr(x[[2]][[1]], "leaf") &lt;- TRUE
  attr(x[[2]][[2]], "leaf") &lt;- TRUE
  attr(x[[1]], "label") &lt;- "A"
  attr(x[[2]][[1]], "label") &lt;- "B"
  attr(x[[2]][[2]], "label") &lt;- "C"
  attr(x, "height") &lt;- 1
  attr(x[[1]], "height") &lt;- 0
  attr(x[[2]], "height") &lt;- 0.5
  attr(x[[2]][[1]], "height") &lt;- 0
  attr(x[[2]][[2]], "height") &lt;- 0
  x &lt;- remidpoint(x)
  class(x) &lt;- "dendrogram"
  plot(x, horiz = TRUE)
</code></pre>

<hr>
<h2 id='reposition'>Reset dendrogram height attributes.</h2><span id='topic+reposition'></span>

<h3>Description</h3>

<p><code>reposition</code> is a helper function used for manually creating
<code>"dendrogram"</code> objects from nested lists. The function
recursively reassigns the 'height' attributes at each node by
a given constant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reposition(x, shift = "reset")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reposition_+3A_x">x</code></td>
<td>
<p>an object of class <code>"dendrogram"</code>.</p>
</td></tr>
<tr><td><code id="reposition_+3A_shift">shift</code></td>
<td>
<p>either the character string &quot;reset&quot; (shift the graph so that
the height of the farthest leaf from the root is zero), or a numeric value
giving the amount to shift the graph along the primary axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>"dendrogram"</code>.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  x &lt;- read.dendrogram(text = "(A:0.1,B:0.2,(C:0.3,D:0.4):0.5);")
  plot(x, horiz = TRUE)
  x &lt;- reposition(x)
  plot(x, horiz = TRUE)
</code></pre>

<hr>
<h2 id='write.dendrogram'>Write a dendrogram object to parenthetic text.</h2><span id='topic+write.dendrogram'></span>

<h3>Description</h3>

<p>This function exports a dendrogram object as a Newick/New Hampshire
text string.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.dendrogram(x, file = "", append = FALSE, edges = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.dendrogram_+3A_x">x</code></td>
<td>
<p>an object of class <code>"dendrogram"</code>.</p>
</td></tr>
<tr><td><code id="write.dendrogram_+3A_file">file</code></td>
<td>
<p>a character string naming a file or connection to write the
output to. If no file path is specified or <code>file = ""</code> the result
is printed to the console.</p>
</td></tr>
<tr><td><code id="write.dendrogram_+3A_append">append</code></td>
<td>
<p>logical indicating whether the output should be
appended to the file. If <code>append = FALSE</code> the contents of the
file will be overwritten (the default setting).</p>
</td></tr>
<tr><td><code id="write.dendrogram_+3A_edges">edges</code></td>
<td>
<p>logical indicating whether edge weights should be
included in the output string.</p>
</td></tr>
<tr><td><code id="write.dendrogram_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code>format</code>. Used to
specify the numbering style of the edge weights (if edges = TRUE).</p>
</td></tr>
</table>


<h3>References</h3>

<p>Felsenstein J (1986) The Newick tree format.
<a href="http://evolution.genetics.washington.edu/phylip/newicktree.html">http://evolution.genetics.washington.edu/phylip/newicktree.html</a>
</p>
<p>Olsen G (1990) Interpretation of the &quot;Newick's 8:45&quot; tree format standard.
<a href="http://evolution.genetics.washington.edu/phylip/newick_doc.html">http://evolution.genetics.washington.edu/phylip/newick_doc.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.dendrogram">read.dendrogram</a></code> to parse a <code>"dendrogram"</code>
object from a text file.
The <code><a href="ape.html#topic+write.tree">write.tree</a></code> function in the <code><a href="ape.html#topic+ape">ape</a></code>
package performs a similar operation for <code>"phylo"</code>
and <code>"multiPhylo"</code> objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  newick &lt;- "(A:0.1,B:0.2,(C:0.3,D:0.4):0.5);"
  x &lt;- read.dendrogram(text = newick)
  write.dendrogram(x, edges = TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
