<!DOCTYPE html><html lang="en"><head><title>Help for package phylotypr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {phylotypr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#build_kmer_database'><p>Build kmer database</p></a></li>
<li><a href='#classify_sequence'><p>Classify 16S rRNA gene sequence fragment</p></a></li>
<li><a href='#filter_taxonomy'><p>Filter taxonomy</p></a></li>
<li><a href='#phylotypr_example'><p>Get path to phylotypr example</p></a></li>
<li><a href='#print_taxonomy'><p>Print taxonomy for an unknown sequence</p></a></li>
<li><a href='#read_fasta'><p>Read in a FASTA-formatted file containing DNA sequences</p></a></li>
<li><a href='#read_taxonomy'><p>Read in taxonomy files</p></a></li>
<li><a href='#trainset9_rdp'><p>RDP training set v9</p></a></li>
<li><a href='#write_fasta'><p>Write to a FASTA-formatted file</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Classifying DNA Sequences to Taxonomic Groupings</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Classification based analysis of DNA sequences to taxonomic groupings. This package primarily implements Naive Bayesian Classifier from the Ribosomal Database Project. This approach has traditionally been used to classify 16S rRNA gene sequences to bacterial taxonomic outlines; however, it can be used for any type of gene sequence. The method was originally described by Wang, Garrity, Tiedje, and Cole in Applied and Environmental Microbiology 73(16):5261-7 &lt;<a href="https://doi.org/10.1128%2FAEM.00062-07">doi:10.1128/AEM.00062-07</a>&gt;. The package also provides functions to read in 'FASTA'-formatted sequence data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mothur/phylotypr">https://github.com/mothur/phylotypr</a>, <a href="https://mothur.org/phylotypr/">https://mothur.org/phylotypr/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mothur/phylotypr/issues">https://github.com/mothur/phylotypr/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, purrr, dplyr, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, readr (&ge; 2.1.0), Rfast (&ge; 2.1.0), stats (&ge; 4.0.0),
stringi (&ge; 1.8.0)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-21 14:28:08 UTC; pschloss</td>
</tr>
<tr>
<td>Author:</td>
<td>Pat Schloss <a href="https://orcid.org/0000-0002-6935-4275"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Pat Schloss &lt;pschloss@umich.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-21 14:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='build_kmer_database'>Build kmer database</h2><span id='topic+build_kmer_database'></span>

<h3>Description</h3>

<p>Build kmer database for classifying 16S rRNA and other gene sequences to
a genus when a kmer size is provided.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_kmer_database(sequences, genera, kmer_size = 8)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="build_kmer_database_+3A_sequences">sequences</code></td>
<td>
<p>A vector of reference sequences for which we have
genus-level taxonomic information in the same order as the
value for genera.</p>
</td></tr>
<tr><td><code id="build_kmer_database_+3A_genera">genera</code></td>
<td>
<p>A character vector of genus-level taxonomic information for
reference sequences in the same order as the value for
sequences. Ideally, taxonomic information will be provided
back to the domain level with each level separated by
semicolons and no spaces.</p>
</td></tr>
<tr><td><code id="build_kmer_database_+3A_kmer_size">kmer_size</code></td>
<td>
<p>An integer indicating the length of the nucleotide word to
base our classification on (default = 8)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object containing the genus level conditional probability
(<code>conditional_prob</code>) of seeing each kmer in a given genus as well as
the genus names (<code>genera</code>)
</p>


<h3>References</h3>

<p>Wang Q, Garrity GM, Tiedje JM, Cole JR. Naive Bayesian classifier for rapid
assignment of rRNA sequences into the new bacterial taxonomy. Appl Environ
Microbiol. 2007 Aug;73(16):5261-7.
doi:<a href="https://pubmed.ncbi.nlm.nih.gov/17586664/">10.1128/AEM.00062-07</a>
PMID: 17586664; PMCID: PMC1950982.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>kmer_size &lt;- 3
sequences &lt;- c("ATGCGCTA", "ATGCGCTC", "ATGCGCTC")
genera &lt;- c("A", "B", "B")

build_kmer_database(sequences, genera, kmer_size)

</code></pre>

<hr>
<h2 id='classify_sequence'>Classify 16S rRNA gene sequence fragment</h2><span id='topic+classify_sequence'></span>

<h3>Description</h3>

<p>The <code>classify_sequence()</code> function implements the Wang et al. naive Bayesian
classification algorithm for 16S rRNA gene sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>classify_sequence(
  unknown_sequence,
  database,
  kmer_size = 8,
  num_bootstraps = 100
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="classify_sequence_+3A_unknown_sequence">unknown_sequence</code></td>
<td>
<p>A character object representing a DNA sequence that
needs to be classified</p>
</td></tr>
<tr><td><code id="classify_sequence_+3A_database">database</code></td>
<td>
<p>A kmer database generated using <code><a href="#topic+build_kmer_database">build_kmer_database</a></code></p>
</td></tr>
<tr><td><code id="classify_sequence_+3A_kmer_size">kmer_size</code></td>
<td>
<p>An integer value (default of 8) indicating the size of kmers
to use for classifying sequences. Higher values use more
RAM with potentially more specificity Lower values use
less RAM with potentially less specificity. Benchmarking
has found that the default of 8 provides the best
specificity with the lowest possible memory requirement and
fastest execution time.</p>
</td></tr>
<tr><td><code id="classify_sequence_+3A_num_bootstraps">num_bootstraps</code></td>
<td>
<p>An integer value (default of 100). The value of
<code>num_bootstraps</code> is the number of randomizations to perform
where <code>1/kmer_size</code> of all kmers are sampled (without
replacement) from <code>unknown_sequence</code>. Higher values will
provide greater precision on the confidence score.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object of two vectors. One vector (<code>taxonomy</code>) is the
taxonomic assignment for each level. The second vector
(<code>confidence</code>) is the percentage of <code>num_bootstraps</code> that the
classifier gave the same classification at that level
</p>


<h3>References</h3>

<p>Wang Q, Garrity GM, Tiedje JM, Cole JR. Naive Bayesian classifier for rapid
assignment of rRNA sequences into the new bacterial taxonomy. Appl Environ
Microbiol. 2007 Aug;73(16):5261-7.
doi:<a href="https://pubmed.ncbi.nlm.nih.gov/17586664/">10.1128/AEM.00062-07</a>
PMID: 17586664; PMCID: PMC1950982.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>kmer_size &lt;- 3
sequences &lt;- c("ATGCGCTA", "ATGCGCTC", "ATGCGCTC")
genera &lt;- c("A", "B", "B")

db &lt;- build_kmer_database(sequences, genera, kmer_size)
unknown_sequence &lt;- "ATGCGCTC"

classify_sequence(
  unknown_sequence = unknown_sequence,
  database = db,
  kmer_size = kmer_size
)
</code></pre>

<hr>
<h2 id='filter_taxonomy'>Filter taxonomy</h2><span id='topic+filter_taxonomy'></span>

<h3>Description</h3>

<p>The <code>filter_taxonomy()</code> function will filter a consensus taxonomy to remove
any taxonomic levels where the confidence score is below a <code>min_confidence</code>
level
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_taxonomy(consensus, min_confidence = 80)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_taxonomy_+3A_consensus">consensus</code></td>
<td>
<p>A list object that contains two slots each with an equal
sized vector. The <code>taxonomy</code> vector contains the
classification at each taxonomic level and the <code>confidence</code>
vector contains the percentage of bootstraps that had the
specified classification</p>
</td></tr>
<tr><td><code id="filter_taxonomy_+3A_min_confidence">min_confidence</code></td>
<td>
<p>A double value between 0 and 100 (default = 80). The
minimum percentage of bootstrap replicates that had
the same classification. Any confidence score below
this value will have the corresponding taxonomy
removed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object containing two equally sized vectors that are filtered
to remove low confidence taxonomies. One vector, <code>taxonomy</code>,
contains the taxonomy at each taxonomic level and the other vector,
<code>confidence</code> contains the confidence score for that taxonomy. There
will be no taxonomies or confidence scores below <code>min_confidence</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>oscillospiraceae &lt;- list(
  taxonomy = c(
    "Bacteria", "Bacillota", "Clostridia",
    "Eubacteriales", "Oscillospiraceae",
    "Flintibacter"
  ),
  confidence = c(100, 100, 99, 99, 98, 58)
)

filter_taxonomy(oscillospiraceae, min_confidence = 80)
</code></pre>

<hr>
<h2 id='phylotypr_example'>Get path to phylotypr example</h2><span id='topic+phylotypr_example'></span>

<h3>Description</h3>

<p>phylotypr comes bundled with some example files in its <code>inst/extdata</code>
directory. This function make them easy to access.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylotypr_example(path = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="phylotypr_example_+3A_path">path</code></td>
<td>
<p>Name of file. If <code>NULL</code>, the example files will be listed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string indicating path to the file listed in <code>path</code>. If <code>NULL</code> is
given then the return value is a vector of file names in the
<code style="white-space: pre;">&#8288;extdata/&#8288;</code> directory
</p>


<h3>Examples</h3>

<pre><code class='language-R'>phylotypr_example()
phylotypr_example("miseq_sop.fasta.gz")
</code></pre>

<hr>
<h2 id='print_taxonomy'>Print taxonomy for an unknown sequence</h2><span id='topic+print_taxonomy'></span>

<h3>Description</h3>

<p>The <code>print_taxonomy()</code> will output the consensus taxonomy for an unknown
sequence with confidence scores for each taxonomic level and each taxonomic
level separated by semi-colons
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_taxonomy(consensus, n_levels = 6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print_taxonomy_+3A_consensus">consensus</code></td>
<td>
<p>A list object that contains two slots each with an equal
sized vector. The <code>taxonomy</code> vector contains the
classification at each taxonomic level and the <code>confidence</code>
vector contains the percentage of bootstraps that had the
specified classification</p>
</td></tr>
<tr><td><code id="print_taxonomy_+3A_n_levels">n_levels</code></td>
<td>
<p>An integer indicating the number of taxonomic levels to
expect. If the number of observed levels is less than this
value, then missing levels will have &quot;_unclassified&quot; to the
end of the last named classification</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string indicating the classification at each taxonomic
level with the corresponding confidence in parentheses. Each taxonomic
level is separated by a semi-colon
</p>


<h3>Examples</h3>

<pre><code class='language-R'>oscillospiraceae &lt;- list(
  taxonomy = c(
    "Bacteria", "Bacillota", "Clostridia",
    "Eubacteriales", "Oscillospiraceae"
  ),
  confidence = c(100, 100, 99, 99, 98)
)

print_taxonomy(oscillospiraceae, n_levels = 6)
</code></pre>

<hr>
<h2 id='read_fasta'>Read in a FASTA-formatted file containing DNA sequences</h2><span id='topic+read_fasta'></span>

<h3>Description</h3>

<p>Given a
<a href="https://en.wikipedia.org/wiki/FASTA_format">standard FASTA-formatted file</a>,
<code>read_fasta</code> will read in the contents of the file and create a three column
data frame with columns for the sequence id, the sequence itself, and any
comments found in the header line for each sequence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_fasta(file, degap = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_fasta_+3A_file">file</code></td>
<td>
<p>Either a path to a file, a connection, or literal data (either a single
string or a raw vector) containing DNA sequences in the standard FASTA
format. There are no checks to determine whether the data are DNA or amino
acid sequences.
</p>
<p>Files ending in .gz, .bz2, .xz, or .zip will be automatically uncompressed.
Files starting with <code style="white-space: pre;">&#8288;http://&#8288;</code>, <code style="white-space: pre;">&#8288;https://&#8288;</code>, <code style="white-space: pre;">&#8288;ftp://&#8288;</code>, or <code style="white-space: pre;">&#8288;ftps://&#8288;</code> will be
automatically downloaded. Remote gz files can also be autom downloaded and
decompressed.</p>
</td></tr>
<tr><td><code id="read_fasta_+3A_degap">degap</code></td>
<td>
<p>Logical value (default = TRUE) Removes gap characters from sequences
indicated by &quot;.&quot; or &quot;-&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame object with three columns. The <code>id</code> column will contain
the non-space characters following the <code>&gt;</code> in the header line of each
sequence; the <code>sequence</code> column will contain the sequence; and the
<code>comment</code> column will contain any text found after the first whitespace
character on the header line.
</p>


<h3>Note</h3>

<p>The sequences in the FASTA file can have line breaks within them and
<code>read_fasta()</code> will put those separate lines into the same sequence
</p>


<h3>Examples</h3>

<pre><code class='language-R'>temp &lt;- tempfile()
write("&gt;seqA\nATGCATGC\n&gt;seqB\nTACGTACG", file = temp)
write("&gt;seqC\nTCCGATGC", file = temp, append = TRUE)
write("&gt;seqD B.ceresus UW85\nTCCGATGC", file = temp, append = TRUE)
write("&gt;seq4\tE. coli K12\tBacteria;Proteobacteria;\nTCCGATGC",
  file = temp,
  append = TRUE
)
write("&gt;seq_4\tSalmonella LT2\tBacteria;Proteobacteria;\nTCCGATGC",
  file = temp, append = TRUE
)
write("&gt;seqE B.ceresus UW123\nTCCGATGC\nTCCGATGC",
  file = temp,
  append = TRUE
)

sequence_df &lt;- read_fasta(temp)

</code></pre>

<hr>
<h2 id='read_taxonomy'>Read in taxonomy files</h2><span id='topic+read_taxonomy'></span>

<h3>Description</h3>

<p>Read a
<a href="https://mothur.org/wiki/taxonomy_file/">mothur-formatted taxonomy file</a>
into R as a data frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_taxonomy(file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_taxonomy_+3A_file">file</code></td>
<td>
<p>Either a path to a file, a connection, or literal data
(either a single string or a raw vector) containing the
sequence id and the taxonomy information for each sequence.
</p>
<p>Files ending in .gz, .bz2, .xz, or .zip will be automatically uncompressed.
Files starting with <code style="white-space: pre;">&#8288;http://&#8288;</code>, <code style="white-space: pre;">&#8288;https://&#8288;</code>, <code style="white-space: pre;">&#8288;ftp://&#8288;</code>, or <code style="white-space: pre;">&#8288;ftps://&#8288;</code> will be
automatically downloaded. Remote gz files can also be autom downloaded and
decompressed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with two columns. The <code>id</code> column contains a name for each
sequence and the <code>taxonomy</code> column, which contains the taxonomy for each
sequence. The string in the <code>taxonomy</code> column is a series of taxonomic names
separated by semi-colons. The string does not have a semi-colon at the end of
the sequence
</p>


<h3>Note</h3>

<p>There are no checks to insure that each sequence has a unique id value. It is
also assumed that each sequence has the same number of taxonomic levels
represented in the second column of the input file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>temp &lt;- tempfile()
write("seqA\tA;B;C;", file = temp)
write("seqB\tA;B; C;", file = temp, append = TRUE)
write("seqC\tA; B;C;", file = temp, append = TRUE)
write("seqD\tA;B;C", file = temp, append = TRUE)
write("seqE\tA;B; C", file = temp, append = TRUE)
write("seqF\tA; B;C", file = temp, append = TRUE)
write("seq G\tA;B;C;", file = temp, append = TRUE)

read_taxonomy(temp)
</code></pre>

<hr>
<h2 id='trainset9_rdp'>RDP training set v9</h2><span id='topic+trainset9_rdp'></span><span id='topic+trainset9_pds'></span>

<h3>Description</h3>

<p>The sequence and taxonomy data for the 10,049 sequences found in the
Ribosomal Database Project's trainset9_032012 training set for use with the
naive Bayesian classifier as implemented in the <code>{phylyotypr}</code> R package.
Originally released by the RDP in September 2012. The <code>rdp</code> version contains
the same sequences as provided by the official RDP version (9,665 bacterial
and 384 archaeal). The <code>pds</code> version contains extra eukaryotic sequences
including 119 chloroplasts and mitochondria (10,168 total sequences). See the
mothur reference file page in &quot;Sources&quot; for more information. Be sure to see
the mothur GitHub project where you can find the phylotyprrefdata package
(https://github.com/mothur/phylotyprrefdata) for access to other taxonomic
reference data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trainset9_rdp

trainset9_pds
</code></pre>


<h3>Format</h3>

<p>A data frame with 3 columns. Each row represents a different
sequence:
</p>

<dl>
<dt>id</dt><dd><p>Sequence accession identifier</p>
</dd>
<dt>sequence</dt><dd><p>DNA sequence string</p>
</dd>
<dt>taxonomy</dt><dd><p>Taxonomic string with each level separated with a <code style="white-space: pre;">&#8288;;&#8288;</code></p>
</dd>
</dl>

<p>An object of class <code>data.frame</code> with 10169 rows and 3 columns.
</p>


<h3>Source</h3>


<ul>
<li> <p><a href="https://mothur.org/wiki/rdp_reference_files/">mothur-formatted files</a>
</p>
</li>
<li> <p><a href="https://sourceforge.net/projects/rdp-classifier/files/RDP_Classifier_TrainingData/RDPClassifier_16S_trainsetNo9_rawtrainingdata.zip/download">RDP sourceforge page</a> # nolint: line_length_linter
</p>
</li></ul>


<hr>
<h2 id='write_fasta'>Write to a FASTA-formatted file</h2><span id='topic+write_fasta'></span>

<h3>Description</h3>

<p>Writes a data frame containing id, sequence, and comment columns,
<code>write_fasta</code> will write the data frame out to a
<a href="https://en.wikipedia.org/wiki/FASTA_format">standard FASTA-formatted file</a>.
The header will have a tab character between the sequence id and any
comments. There won't be a tab if there's no comment for the sequence. All
sequence data will be on a single line
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_fasta(data_frame, file = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_fasta_+3A_data_frame">data_frame</code></td>
<td>
<p>A data frame object with three columns. The <code>id</code> column will contain
the non-space characters following the <code>&gt;</code> in the header line of each
sequence; the <code>sequence</code> column will contain the sequence; and the
<code>comment</code> column will contain any text found after the first whitespace
character on the header line. The <code>comment</code> column is optional.</p>
</td></tr>
<tr><td><code id="write_fasta_+3A_file">file</code></td>
<td>
<p>Either a path to a file, a connection, or literal data (either a single
string or a raw vector) to write to a standard FASTA formatted file. There
are no checks to determine whether the data are DNA or amino acid sequences.
</p>
<p>Files ending in .gz, .bz2, .xz, or .zip will be automatically compressed.
Files starting with <code style="white-space: pre;">&#8288;http://&#8288;</code>, <code style="white-space: pre;">&#8288;https://&#8288;</code>, <code style="white-space: pre;">&#8288;ftp://&#8288;</code>, or <code style="white-space: pre;">&#8288;ftps://&#8288;</code> will be
automatically downloaded. Remote gz files can also be autom downloaded and
decompressed.
</p>
<p>If the value of <code>file</code> is <code>NULL</code> (default), the string will be written out
to the screen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Sequence data is either written out to the screen (<code>file = NULL</code>) or to a
file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df_d &lt;- data.frame(
  id = c("seqA", "seqB", "seqC"),
  sequence = c("ATGCATGC", "ATGCATGA", "ATGCATGT"),
  comment = c("comment 1", "", "comment 3")
)

string_d &lt;- write_fasta(df_d)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
