<!DOCTYPE html><html><head><title>Help for package phylepic</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {phylepic}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#phylepic-package'><p>phylepic: Combined Visualisation of Phylogenetic and Epidemiological Data</p></a></li>
<li><a href='#coord_tree'><p>Cartesian coordinates with specialised grid for trees</p></a></li>
<li><a href='#create_tree_layout'><p>Create a graph layout for plotting</p></a></li>
<li><a href='#drop.clade'><p>Drop a clade from a phylogentic tree</p></a></li>
<li><a href='#geom_node_text_filled'><p>Annotate nodes with text and a background</p></a></li>
<li><a href='#GeomCalendar'><p>Specialised tile geometry for calendar plots</p></a></li>
<li><a href='#oob_infinite'><p>Out of bounds handling</p></a></li>
<li><a href='#phylepic'><p>Combine metadata (a line list) with a phylogenetic tree</p></a></li>
<li><a href='#plot_bars'><p>Plot metadata bars panel</p></a></li>
<li><a href='#plot_calendar'><p>Plot calendar panel</p></a></li>
<li><a href='#plot_epicurve'><p>Plot epidemic curve panel</p></a></li>
<li><a href='#plot_tree'><p>Plot phylogenetic tree panel</p></a></li>
<li><a href='#plot.phylepic'><p>Plot &quot;phylepic&quot; objects</p></a></li>
<li><a href='#scale_x_week'><p>Date scale with breaks specified by week</p></a></li>
<li><a href='#stat_week'><p>Calculate week bins from dates</p></a></li>
<li><a href='#stat_week_2d'><p>Calculate week bins with additional binning in the y axis</p></a></li>
<li><a href='#week_breaks'><p>Breaks for week-binning date axes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Combined Visualisation of Phylogenetic and Epidemiological Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of utilities and 'ggplot2' extensions to assist with
  visualisations in genomic epidemiology. This includes the 'phylepic' chart,
  a visual combination of a phylogenetic tree and a matched epidemic curve.
  The included 'ggplot2' extensions such as date axes binned by week are
  relevant for other applications in epidemiology and beyond. The approach is
  described in Suster et al. (2024) &lt;<a href="https://doi.org/10.1101%2F2024.04.02.24305229">doi:10.1101/2024.04.02.24305229</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, cli, cowplot, dplyr, forcats, ggnewscale, ggplot2 (&ge;
3.5.0), ggraph, igraph, rlang, scales, tidygraph, vctrs</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/cidm-ph/phylepic">https://github.com/cidm-ph/phylepic</a>,
<a href="https://cidm-ph.github.io/phylepic/">https://cidm-ph.github.io/phylepic/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/cidm-ph/phylepic/issues">https://github.com/cidm-ph/phylepic/issues</a></td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-30 23:54:45 UTC; carl</td>
</tr>
<tr>
<td>Author:</td>
<td>Carl Suster <a href="https://orcid.org/0000-0001-7021-9380"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Western Sydney Local Health District, NSW Health [cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Carl Suster &lt;Carl.Suster@health.nsw.gov.au&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-31 19:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='phylepic-package'>phylepic: Combined Visualisation of Phylogenetic and Epidemiological Data</h2><span id='topic+phylepic-package'></span>

<h3>Description</h3>

<p>A collection of utilities and 'ggplot2' extensions to assist with visualisations in genomic epidemiology. This includes the 'phylepic' chart, a visual combination of a phylogenetic tree and a matched epidemic curve. The included 'ggplot2' extensions such as date axes binned by week are relevant for other applications in epidemiology and beyond. The approach is described in Suster et al. (2024) <a href="https://doi.org/10.1101/2024.04.02.24305229">doi:10.1101/2024.04.02.24305229</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Carl Suster <a href="mailto:Carl.Suster@health.nsw.gov.au">Carl.Suster@health.nsw.gov.au</a> (<a href="https://orcid.org/0000-0001-7021-9380">ORCID</a>)
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Western Sydney Local Health District, NSW Health [copyright holder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/cidm-ph/phylepic">https://github.com/cidm-ph/phylepic</a>
</p>
</li>
<li> <p><a href="https://cidm-ph.github.io/phylepic/">https://cidm-ph.github.io/phylepic/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/cidm-ph/phylepic/issues">https://github.com/cidm-ph/phylepic/issues</a>
</p>
</li></ul>


<hr>
<h2 id='coord_tree'>Cartesian coordinates with specialised grid for trees</h2><span id='topic+coord_tree'></span><span id='topic+CoordTree'></span>

<h3>Description</h3>

<p>This coord is based on the default Cartesian coordinates, but draws the a
filled background in addition to the normal grid lines. The grid is forced
to appear on every integer value within the scale's range.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coord_tree(
  xlim = NULL,
  ylim = NULL,
  expand = TRUE,
  default = FALSE,
  clip = "on"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coord_tree_+3A_xlim">xlim</code>, <code id="coord_tree_+3A_ylim">ylim</code>, <code id="coord_tree_+3A_expand">expand</code>, <code id="coord_tree_+3A_default">default</code>, <code id="coord_tree_+3A_clip">clip</code></td>
<td>
<p>See <code>ggplot2::coord_cartesian()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The appearance of the grid can be controlled with theme elements:
</p>

<dl>
<dt><code>phylepic.grid.bar</code></dt><dd><p>filled grid (<code>element_rect()</code>).</p>
</dd>
<dt><code>phylepic.grid.line</code></dt><dd><p>grid line (<code>element_line()</code>).</p>
</dd>
<dt><code>phylepic.grid.every</code></dt><dd><p>grid frequency (<code>integer</code>). Default for both
<code>phylepic.grid.every.bar</code> and <code>phylepic.grid.every.stripe</code></p>
</dd>
<dt><code>phylepic.grid.every.bar</code></dt><dd><p>grid bar frequency (<code>integer</code>).
Defaults to 2 to give an alternative striped background</p>
</dd>
<dt><code>phylepic.grid.every.stripe</code></dt><dd><p>grid bar frequency (<code>integer</code>).
Defaults to 1 so that every tip on a tree has its own line</p>
</dd>
</dl>



<h3>Value</h3>

<p>coord suitable for adding to a plot
</p>

<hr>
<h2 id='create_tree_layout'>Create a graph layout for plotting</h2><span id='topic+create_tree_layout'></span>

<h3>Description</h3>

<p>This lays out a graph using <code>ggraph::create_layout()</code> with the <code>"dendrogram"</code>
layout, takes edge lengths from the tree, and flips the layout coordinates.
The plotting functions associated with <code><a href="#topic+phylepic">phylepic()</a></code> expect the graph to
be laid out using these settings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_tree_layout(tree, tip_data = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_tree_layout_+3A_tree">tree</code></td>
<td>
<p>A tree-like graph or a <code>phylepic</code> object.</p>
</td></tr>
<tr><td><code id="create_tree_layout_+3A_tip_data">tip_data</code></td>
<td>
<p>A data frame with tip metadata. There must be a column called
<code>.phylepic.name</code> with values that correspond to the names of leaf nodes in
the tree. If <code>NULL</code>, no tip data is joined onto the tree.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A &quot;layout_ggraph&quot; object suitable for plotting with <a href="ggplot2.html#topic+ggplot">ggplot2::ggplot</a>'.
</p>

<hr>
<h2 id='drop.clade'>Drop a clade from a phylogentic tree</h2><span id='topic+drop.clade'></span>

<h3>Description</h3>

<p><code>drop.clade</code> invokes <code><a href="ape.html#topic+drop.tip">ape::drop.tip()</a></code> on all tips descendent from the
specified node. This is convenient when used alongside <code><a href="ape.html#topic+mrca">ape::getMRCA()</a></code> to
drop a clade defined by the most recent common ancestor of a set of tips,
rather than exhaustively specifying all of its tips.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drop.clade(phy, node, root.edge = 0, collapse.singles = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drop.clade_+3A_phy">phy</code></td>
<td>
<p>an object of class &quot;phylo&quot;.</p>
</td></tr>
<tr><td><code id="drop.clade_+3A_node">node</code></td>
<td>
<p>number specifying the parent node of the clade to delete.</p>
</td></tr>
<tr><td><code id="drop.clade_+3A_root.edge">root.edge</code>, <code id="drop.clade_+3A_collapse.singles">collapse.singles</code></td>
<td>
<p>passed to <code><a href="ape.html#topic+drop.tip">ape::drop.tip()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>New phylo object with the chosen clade removed
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("ape")
data(bird.orders)
plot(bird.orders)

# find the common ancestor of some tips
mrca &lt;- ape::getMRCA(bird.orders, c("Passeriformes", "Coliiformes"))

# drop the clade descending from that ancestor
plot(drop.clade(bird.orders, mrca))
</code></pre>

<hr>
<h2 id='geom_node_text_filled'>Annotate nodes with text and a background</h2><span id='topic+geom_node_text_filled'></span><span id='topic+GeomTextFilled'></span>

<h3>Description</h3>

<p>This geom behaves like <code>ggraph::geom_node_text()</code> except that it also inserts
a white background behind the text extending to the left margin. This will
only make sense for a horizontal dendrogram graph layout with the root node
on the left.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_node_text_filled(
  mapping = NULL,
  data = NULL,
  position = "identity",
  parse = FALSE,
  check_overlap = FALSE,
  show.legend = NA,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geom_node_text_filled_+3A_mapping">mapping</code>, <code id="geom_node_text_filled_+3A_data">data</code>, <code id="geom_node_text_filled_+3A_position">position</code>, <code id="geom_node_text_filled_+3A_parse">parse</code>, <code id="geom_node_text_filled_+3A_check_overlap">check_overlap</code>, <code id="geom_node_text_filled_+3A_show.legend">show.legend</code>, <code id="geom_node_text_filled_+3A_...">...</code></td>
<td>
<p>Arguments passed to the geom that powers <code>ggraph::geom_node_text()</code>.
Note that the additional arguments of that function such as <code>repel</code> are
not supported here.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This background covers up part of the grid rendered by the coord layer.
The reason that this is done as part of the text instead of as a separate
layer is so that we have access to the rendered dimensions of the text grobs.
</p>


<h3>Value</h3>

<p>Layer that draws text and background grobs
</p>

<hr>
<h2 id='GeomCalendar'>Specialised tile geometry for calendar plots</h2><span id='topic+GeomCalendar'></span><span id='topic+geom_calendar'></span>

<h3>Description</h3>

<p>This geom behaves mostly the same as <code><a href="ggplot2.html#topic+geom_tile">ggplot2::geom_tile()</a></code> with a few
additions. Firstly, the <code>label</code> aesthetic is supported to draw text on top of
the tiles. Secondly, out of bounds values can be drawn as arrows at the edge
of the scale (see details below).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geom_calendar(
  mapping = NULL,
  data = NULL,
  stat = "identity",
  position = "identity",
  ...,
  linejoin = "mitre",
  label_params = list(colour = "grey30"),
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GeomCalendar_+3A_mapping">mapping</code>, <code id="GeomCalendar_+3A_data">data</code>, <code id="GeomCalendar_+3A_stat">stat</code>, <code id="GeomCalendar_+3A_position">position</code>, <code id="GeomCalendar_+3A_linejoin">linejoin</code>, <code id="GeomCalendar_+3A_na.rm">na.rm</code>, <code id="GeomCalendar_+3A_show.legend">show.legend</code>, <code id="GeomCalendar_+3A_inherit.aes">inherit.aes</code>, <code id="GeomCalendar_+3A_...">...</code></td>
<td>
<p>see <code><a href="ggplot2.html#topic+geom_tile">ggplot2::geom_tile()</a></code>.</p>
</td></tr>
<tr><td><code id="GeomCalendar_+3A_label_params">label_params</code></td>
<td>
<p>additional parameters for text labels if present
(see <code><a href="ggplot2.html#topic+geom_text">ggplot2::geom_text()</a></code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Any <code>x</code> values that are infinite (i.e. <code>-Inf</code> or <code>Inf</code>) would normally be
dropped by ggplot's layers. If any such values survive the stat processing,
they will be drawn by <code>geom_calendar()</code> as triangles at the respective edges
of the scale. This is intended to work with a scale configured to use
<code><a href="#topic+oob_infinite">oob_infinite()</a></code> for out of bounds handling.
The triangles are drawn with their base (vertical edge) sitting on the scale
limit, and their width equal to half of the median bin width.
</p>
<p>Note that the <code>label</code> aesthetic will be dropped if the data are not grouped
in the expected way. In general this means that all rows contributing to a
given bin must have the same value for the <code>label</code> aesthetic.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)

set.seed(1)
events &lt;- rep(as.Date("2024-01-31") - 0:30, rpois(31, 6))
values &lt;- round(rgamma(length(events), 1, 0.01))
df &lt;- data.frame(date = events, value = values)

ggplot(df) +
    geom_calendar(
        aes(date, value, label = after_stat(count)),
        colour = "white",
        stat = "week_2d",
        week_start = "Monday",
        bins.y = 10
    ) +
    scale_x_week(
        limits = as.Date(c("2024-01-08", NA)),
        expand = expansion(add = 3.5)
    )
</code></pre>

<hr>
<h2 id='oob_infinite'>Out of bounds handling</h2><span id='topic+oob_infinite'></span>

<h3>Description</h3>

<p>This helper works the same way as <code><a href="scales.html#topic+oob">scales::oob_censor()</a></code> and similar. Out of
bounds values are pushed to positive or negative infinity. This is not useful
for builtin ggplot layers which will display a warning and drop rows with
infinite values in required aesthetics. <code><a href="#topic+geom_calendar">geom_calendar()</a></code> however uses the
infinite values to indicate out of bounds values explicitly on the plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oob_infinite(x, range = c(0, 1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="oob_infinite_+3A_x">x</code></td>
<td>
<p>A numeric vector of values to modify.</p>
</td></tr>
<tr><td><code id="oob_infinite_+3A_range">range</code></td>
<td>
<p>A numeric vector of length two giving the minimum and maximum
limit of the desired output range respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numerical vector of the same length as <code>x</code> where out of bound
values have been replaced by <code>Inf</code> or <code>-Inf</code> accordingly.
</p>

<hr>
<h2 id='phylepic'>Combine metadata (a line list) with a phylogenetic tree</h2><span id='topic+phylepic'></span>

<h3>Description</h3>

<p>Some checks are performed to catch issues where the metadata and tree tips
don't match up. Any columns in <code>metadata</code> that are factors have all levels
that do not appear in the data dropped.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylepic(
  tree,
  metadata,
  name,
  date,
  unmatched_tips = c("error", "drop", "keep")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylepic_+3A_tree">tree</code></td>
<td>
<p>An object convertible to a <code>tbl_graph</code>. This will usually be a
&quot;phylo&quot; object, but see <a href="tidygraph.html#topic+tbl_graph">tidygraph::tbl_graph</a> for more details.</p>
</td></tr>
<tr><td><code id="phylepic_+3A_metadata">metadata</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="phylepic_+3A_name">name</code></td>
<td>
<p>Column in <code>metadata</code> that corresponds to the tree's tip labels
(tidy-eval).</p>
</td></tr>
<tr><td><code id="phylepic_+3A_date">date</code></td>
<td>
<p>Column in <code>metadata</code> that contains the date data (class &quot;Date&quot;)
for the tips (tidy-eval).</p>
</td></tr>
<tr><td><code id="phylepic_+3A_unmatched_tips">unmatched_tips</code></td>
<td>
<p>Action to take when <code>tree</code> contains tip labels that
do not appear in <code>name</code>. <code>"error"</code> aborts with an error message, <code>"drop"</code>
drops unmatched tips from <code>tree</code>, <code>"keep"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To reduce surprises when matching <code>metadata</code> and <code>tree</code>, by default an error
occurs when there are tree tips that do not have associated metadata. On the
other hand, it it expected that <code>metadata</code> might contain rows that do not
correspond to the tips in <code>tree</code>.
</p>
<p>This often means that <code>factor</code> columns from <code>metadata</code> will contain levels
that do not appear at all in the tree. For plotting,
<code>ggplot2::discrete_scale</code> normally solves this with <code>drop = TRUE</code>, however
this can lead to inconsistencies when sharing the same scale across multiple
phylepic panels. <code>phylepic()</code> drops unused levels in all factors so that
scales can use <code>drop = FALSE</code> for consistency.
</p>


<h3>Value</h3>

<p>An object of class &quot;phylepic&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ape)

tree &lt;- read.tree(system.file("enteric.newick", package = "phylepic"))
metadata &lt;- read.csv(
  system.file("enteric_metadata.csv", package = "phylepic")
)
phylepic(tree, metadata, name, as.Date(collection_date))
</code></pre>

<hr>
<h2 id='plot_bars'>Plot metadata bars panel</h2><span id='topic+plot_bars'></span>

<h3>Description</h3>

<p>This uses <code><a href="ggplot2.html#topic+geom_tile">ggplot2::geom_tile()</a></code> to produce a grid with a row aligned with
each tip on the tree, and a column for each type of data specified. If no
scales are specified, one is created for each factor column in the metadata
table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_bars(phylepic, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_bars_+3A_phylepic">phylepic</code></td>
<td>
<p>object of class &quot;phylepic&quot;.</p>
</td></tr>
<tr><td><code id="plot_bars_+3A_...">...</code></td>
<td>
<p>scale specifications.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>phylepic</code> is specified returns a ggplot, otherwise a function
that when passed a &quot;phylepic&quot; object produces a ggplot for use with
<code><a href="#topic+plot.phylepic">plot.phylepic()</a></code>.
</p>


<h3>See Also</h3>

<p>Other phylepic plots: 
<code><a href="#topic+plot.phylepic">plot.phylepic</a>()</code>,
<code><a href="#topic+plot_calendar">plot_calendar</a>()</code>,
<code><a href="#topic+plot_epicurve">plot_epicurve</a>()</code>,
<code><a href="#topic+plot_tree">plot_tree</a>()</code>
</p>

<hr>
<h2 id='plot_calendar'>Plot calendar panel</h2><span id='topic+plot_calendar'></span>

<h3>Description</h3>

<p>Plot calendar panel
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_calendar(
  phylepic,
  fill = NULL,
  weeks = TRUE,
  week_start = getOption("phylepic.week_start"),
  labels = NULL,
  labels.params = list(size = 3, fontface = "bold", colour = "white")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_calendar_+3A_phylepic">phylepic</code></td>
<td>
<p>Object of class &quot;phylepic&quot;.</p>
</td></tr>
<tr><td><code id="plot_calendar_+3A_fill">fill</code></td>
<td>
<p>Variable in metadata table to use for the fill aesthetic (tidy-eval).</p>
</td></tr>
<tr><td><code id="plot_calendar_+3A_weeks">weeks</code></td>
<td>
<p>When <code>TRUE</code>, bin the date axis by weeks.</p>
</td></tr>
<tr><td><code id="plot_calendar_+3A_week_start">week_start</code></td>
<td>
<p>Day the week begins (defaults to Monday).
Can be specified as a case-insensitive English weekday name such as &quot;Monday&quot;
or an integer. Since you generally won't want to mix definitions, it is
more convenient to control this globally with the <code>"phylepic.week_start"</code>
option, e.g. <code>options(phylepic.week_start = "Monday")</code>.</p>
</td></tr>
<tr><td><code id="plot_calendar_+3A_labels">labels</code></td>
<td>
<p>Controls the format of date labels on calendar tiles.
If <code>NULL</code>, no labels are drawn.
If a character scalar, controls the date format (see <code><a href="base.html#topic+strptime">strptime()</a></code>).</p>
</td></tr>
<tr><td><code id="plot_calendar_+3A_labels.params">labels.params</code></td>
<td>
<p>Passed to <code><a href="ggplot2.html#topic+geom_text">ggplot2::geom_text()</a></code> if <code>labels</code> are drawn.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>phylepic</code> is specified returns a ggplot, otherwise a function
that when passed a &quot;phylepic&quot; object produces a ggplot for use with
<code><a href="#topic+plot.phylepic">plot.phylepic()</a></code>.
</p>


<h3>See Also</h3>

<p>Other phylepic plots: 
<code><a href="#topic+plot.phylepic">plot.phylepic</a>()</code>,
<code><a href="#topic+plot_bars">plot_bars</a>()</code>,
<code><a href="#topic+plot_epicurve">plot_epicurve</a>()</code>,
<code><a href="#topic+plot_tree">plot_tree</a>()</code>
</p>

<hr>
<h2 id='plot_epicurve'>Plot epidemic curve panel</h2><span id='topic+plot_epicurve'></span>

<h3>Description</h3>

<p>Plot epidemic curve panel
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_epicurve(
  phylepic,
  fill = NULL,
  weeks = TRUE,
  week_start = getOption("phylepic.week_start")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_epicurve_+3A_phylepic">phylepic</code></td>
<td>
<p>Object of class &quot;phylepic&quot;.</p>
</td></tr>
<tr><td><code id="plot_epicurve_+3A_fill">fill</code></td>
<td>
<p>Variable in metadata table to use for the fill aesthetic (tidy-eval).</p>
</td></tr>
<tr><td><code id="plot_epicurve_+3A_weeks">weeks</code></td>
<td>
<p>When <code>TRUE</code>, bin the date axis by weeks.</p>
</td></tr>
<tr><td><code id="plot_epicurve_+3A_week_start">week_start</code></td>
<td>
<p>Day the week begins (defaults to Monday).
Can be specified as a case-insensitive English weekday name such as &quot;Monday&quot;
or an integer. Since you generally won't want to mix definitions, it is
more convenient to control this globally with the <code>"phylepic.week_start"</code>
option, e.g. <code>options(phylepic.week_start = "Monday")</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>phylepic</code> is specified returns a ggplot, otherwise a function
that when passed a &quot;phylepic&quot; object produces a ggplot for use with
<code><a href="#topic+plot.phylepic">plot.phylepic()</a></code>.
</p>


<h3>See Also</h3>

<p>Other phylepic plots: 
<code><a href="#topic+plot.phylepic">plot.phylepic</a>()</code>,
<code><a href="#topic+plot_bars">plot_bars</a>()</code>,
<code><a href="#topic+plot_calendar">plot_calendar</a>()</code>,
<code><a href="#topic+plot_tree">plot_tree</a>()</code>
</p>

<hr>
<h2 id='plot_tree'>Plot phylogenetic tree panel</h2><span id='topic+plot_tree'></span>

<h3>Description</h3>

<p>The tree is drawn using <code>ggraph</code> with its dendrogram layout. When
customising it, you may wish to add layers such as
<code><a href="ggraph.html#topic+geom_node_point">ggraph::geom_node_point()</a></code>.
The metadata table is joined onto the tree, so all its column names are
available for use in the various <code>ggraph</code> geoms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_tree(phylepic, label = .data$name, bootstrap = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_tree_+3A_phylepic">phylepic</code></td>
<td>
<p>object of class &quot;phylepic&quot;.</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_label">label</code></td>
<td>
<p>variable in metadata table corresponding to the tip labels (tidy-eval).</p>
</td></tr>
<tr><td><code id="plot_tree_+3A_bootstrap">bootstrap</code></td>
<td>
<p>when <code>TRUE</code>, draw bootstrap vaues on the tree. These are
only drawn if they are detected from the node labels having the form &quot;a/b&quot;
where both &quot;a&quot; and &quot;b&quot; are numbers. Currently, the bootstrap values are
displayed as a percentage, suppressing zero values and values for very
short branches. To customise the appearance or details instead use
<code>bootstrap = FALSE</code> and add your own layer with <a href="ggraph.html#topic+geom_edge_elbow">ggraph::geom_edge_elbow</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>phylepic</code> is specified returns a ggplot, otherwise a function
that when passed a &quot;phylepic&quot; object produces a ggplot for use with
<code><a href="#topic+plot.phylepic">plot.phylepic()</a></code>.
</p>


<h3>See Also</h3>

<p>Other phylepic plots: 
<code><a href="#topic+plot.phylepic">plot.phylepic</a>()</code>,
<code><a href="#topic+plot_bars">plot_bars</a>()</code>,
<code><a href="#topic+plot_calendar">plot_calendar</a>()</code>,
<code><a href="#topic+plot_epicurve">plot_epicurve</a>()</code>
</p>

<hr>
<h2 id='plot.phylepic'>Plot &quot;phylepic&quot; objects</h2><span id='topic+plot.phylepic'></span><span id='topic+autoplot.phylepic'></span>

<h3>Description</h3>

<p>The <code>autoplot()</code> and <code>plot()</code> methods for &quot;phylepic&quot; objects assemble various
panels into the final plot. To facilitate customisations, the plots from
each panel can be overwritten. Some effort is made to ensure that the
specified plots will look reasonable when assembled.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'phylepic'
plot(
  x,
  ...,
  plot.tree = plot_tree(),
  plot.bars = plot_bars(),
  plot.calendar = plot_calendar(),
  plot.epicurve = plot_epicurve(),
  scale.date = NULL,
  scale.fill = NULL,
  width.tree = 10,
  width.bars = 1,
  width.date = 5,
  width.legend = 2,
  height.tree = 2
)

## S3 method for class 'phylepic'
autoplot(
  object,
  ...,
  plot.tree = plot_tree(),
  plot.bars = plot_bars(),
  plot.calendar = plot_calendar(),
  plot.epicurve = plot_epicurve(),
  scale.date = NULL,
  scale.fill = NULL,
  width.tree = 10,
  width.bars = 1,
  width.date = 5,
  width.legend = 2,
  height.tree = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.phylepic_+3A_...">...</code></td>
<td>
<p>Ignored.</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_plot.tree">plot.tree</code></td>
<td>
<p>ggplot for the tree panel (see <a href="#topic+plot_tree">plot_tree</a>).</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_plot.bars">plot.bars</code></td>
<td>
<p>ggplot for the metadata bars panel (see <a href="#topic+plot_bars">plot_bars</a>).</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_plot.calendar">plot.calendar</code></td>
<td>
<p>ggplot for the calendar panel (see <a href="#topic+plot_calendar">plot_calendar</a>).</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_plot.epicurve">plot.epicurve</code></td>
<td>
<p>ggplot for the epidemic curve panel (see <a href="#topic+plot_epicurve">plot_epicurve</a>).</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_scale.date">scale.date</code></td>
<td>
<p>A date scale passed to both the calendar and epicurve panels
(see <a href="ggplot2.html#topic+scale_date">ggplot2::scale_x_date</a>).</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_scale.fill">scale.fill</code></td>
<td>
<p>A fill scale passed to both the calendar and epicurve panels
(see <a href="ggplot2.html#topic+scale_date">ggplot2::scale_x_date</a>).</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_width.tree">width.tree</code></td>
<td>
<p>Relative width of the tree panel.</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_width.bars">width.bars</code></td>
<td>
<p>Relative width of the metadata bars panel.</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_width.date">width.date</code></td>
<td>
<p>Relative width of the calendar panel.</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_width.legend">width.legend</code></td>
<td>
<p>Relative width of the legend, if present.</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_height.tree">height.tree</code></td>
<td>
<p>Relative height of the tree panel.</p>
</td></tr>
<tr><td><code id="plot.phylepic_+3A_object">object</code>, <code id="plot.phylepic_+3A_x">x</code></td>
<td>
<p>Object of class &quot;phylepic&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In general, if you wish to suppress a panel from the plot, set the
corresponding <code style="white-space: pre;">&#8288;plot.*&#8288;</code> argument to <code>NULL</code>. To customise it, use the
corresponding <code style="white-space: pre;">&#8288;plot_*()&#8288;</code> function, which returns a ggplot plot. You can then
add new layers or themes to that plot. See <code>vignette("phylepic")</code> for
examples.
</p>
<p>Legends from all panels are collected and de-duplicated. They are drawn on
the right edge of the overall plot.
</p>


<h3>Value</h3>

<p><code>plot()</code> is usually called to display the plot, whereas <code>autoplot()</code>
returns a &quot;ggplot&quot; object that can later be displayed with <code>print()</code>.
</p>


<h3>See Also</h3>

<p>Other phylepic plots: 
<code><a href="#topic+plot_bars">plot_bars</a>()</code>,
<code><a href="#topic+plot_calendar">plot_calendar</a>()</code>,
<code><a href="#topic+plot_epicurve">plot_epicurve</a>()</code>,
<code><a href="#topic+plot_tree">plot_tree</a>()</code>
</p>

<hr>
<h2 id='scale_x_week'>Date scale with breaks specified by week</h2><span id='topic+scale_x_week'></span>

<h3>Description</h3>

<p>This produces a scale that is measured in days as with <a href="ggplot2.html#topic+scale_date">ggplot2::scale_x_date</a>,
however it will snap breaks and limits to week boundaries so that things
work as intended when binning by week.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_x_week(
  name = waiver(),
  week_breaks = waiver(),
  labels = waiver(),
  date_labels = waiver(),
  week_minor_breaks = waiver(),
  oob = oob_infinite,
  limits = NULL,
  ...,
  week_start = getOption("phylepic.week_start")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scale_x_week_+3A_name">name</code>, <code id="scale_x_week_+3A_labels">labels</code>, <code id="scale_x_week_+3A_date_labels">date_labels</code>, <code id="scale_x_week_+3A_oob">oob</code>, <code id="scale_x_week_+3A_limits">limits</code>, <code id="scale_x_week_+3A_...">...</code></td>
<td>
<p>see <code><a href="ggplot2.html#topic+scale_date">ggplot2::scale_x_date()</a></code>.</p>
</td></tr>
<tr><td><code id="scale_x_week_+3A_week_breaks">week_breaks</code>, <code id="scale_x_week_+3A_week_minor_breaks">week_minor_breaks</code></td>
<td>
<p>frequency of breaks in number of weeks (e.g. <code>2</code> for fortnightly breaks).</p>
</td></tr>
<tr><td><code id="scale_x_week_+3A_week_start">week_start</code></td>
<td>
<p>Day the week begins (defaults to Monday).
Can be specified as a case-insensitive English weekday name such as &quot;Monday&quot;
or an integer. Since you generally won't want to mix definitions, it is
more convenient to control this globally with the <code>"phylepic.week_start"</code>
option, e.g. <code>options(phylepic.week_start = "Monday")</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Any <code>limits</code> specified are converted to the nearest week boundary that
includes the specified dates, i.e. the lower limit will be rounded down and
the upper limit rounded up so that the limits are week boundaries.
</p>


<h3>Value</h3>

<p>a ggplot scale object.
</p>

<hr>
<h2 id='stat_week'>Calculate week bins from dates</h2><span id='topic+stat_week'></span><span id='topic+StatWeek'></span><span id='topic+StatWeek2d'></span>

<h3>Description</h3>

<p>Computes weeks for date data. This is mostly equivalent to
<code><a href="ggplot2.html#topic+geom_histogram">ggplot2::stat_bin()</a></code> with the bins fixed to weeks starting on a particular
day.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_week(
  mapping = NULL,
  data = NULL,
  geom = "bar",
  position = "stack",
  ...,
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE,
  week_start = getOption("phylepic.week_start"),
  pad = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stat_week_+3A_mapping">mapping</code>, <code id="stat_week_+3A_data">data</code>, <code id="stat_week_+3A_geom">geom</code>, <code id="stat_week_+3A_position">position</code>, <code id="stat_week_+3A_na.rm">na.rm</code>, <code id="stat_week_+3A_show.legend">show.legend</code>, <code id="stat_week_+3A_inherit.aes">inherit.aes</code>, <code id="stat_week_+3A_pad">pad</code>, <code id="stat_week_+3A_...">...</code></td>
<td>
<p>See <code><a href="ggplot2.html#topic+geom_histogram">ggplot2::stat_bin()</a></code>.</p>
</td></tr>
<tr><td><code id="stat_week_+3A_week_start">week_start</code></td>
<td>
<p>Day the week begins (defaults to Monday).
Can be specified as a case-insensitive English weekday name such as &quot;Monday&quot;
or an integer. Since you generally won't want to mix definitions, it is
more convenient to control this globally with the <code>"phylepic.week_start"</code>
option, e.g. <code>options(phylepic.week_start = "Monday")</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 stat layer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)

set.seed(1)
events &lt;- rep(as.Date("2024-01-31") - 0:30, rpois(31, 2))
df &lt;- data.frame(date = events)

ggplot(df) + stat_week(aes(date), week_start = "Monday")

# or equivalently:
# ggplot(df) + geom_bar(aes(date), stat = "week", week_start = "Monday")
</code></pre>

<hr>
<h2 id='stat_week_2d'>Calculate week bins with additional binning in the y axis</h2><span id='topic+stat_week_2d'></span>

<h3>Description</h3>

<p>Computes week bins for date data in the x aesthetic, and allows
the binning to be specified for the y aesthetic. This is mostly equivalent to
<code><a href="ggplot2.html#topic+geom_bin_2d">ggplot2::stat_bin_2d()</a></code> with the x aesthetic handling fixed to weeks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stat_week_2d(
  mapping = NULL,
  data = NULL,
  geom = "tile",
  position = "identity",
  ...,
  bins.y = NULL,
  binwidth.y = NULL,
  breaks.y = NULL,
  center.y = NULL,
  boundary.y = NULL,
  closed.y = c("left", "right"),
  drop = TRUE,
  week_start = getOption("phylepic.week_start"),
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stat_week_2d_+3A_mapping">mapping</code>, <code id="stat_week_2d_+3A_data">data</code>, <code id="stat_week_2d_+3A_geom">geom</code>, <code id="stat_week_2d_+3A_position">position</code>, <code id="stat_week_2d_+3A_na.rm">na.rm</code>, <code id="stat_week_2d_+3A_show.legend">show.legend</code>, <code id="stat_week_2d_+3A_inherit.aes">inherit.aes</code>, <code id="stat_week_2d_+3A_...">...</code></td>
<td>
<p>See <a href="ggplot2.html#topic+geom_bin_2d">ggplot2::stat_bin_2d</a>.</p>
</td></tr>
<tr><td><code id="stat_week_2d_+3A_bins.y">bins.y</code>, <code id="stat_week_2d_+3A_binwidth.y">binwidth.y</code>, <code id="stat_week_2d_+3A_breaks.y">breaks.y</code>, <code id="stat_week_2d_+3A_center.y">center.y</code>, <code id="stat_week_2d_+3A_boundary.y">boundary.y</code>, <code id="stat_week_2d_+3A_closed.y">closed.y</code></td>
<td>
<p>See the analogous parameters in <a href="ggplot2.html#topic+geom_bin_2d">ggplot2::stat_bin_2d</a>.</p>
</td></tr>
<tr><td><code id="stat_week_2d_+3A_drop">drop</code></td>
<td>
<p>drop bins with zero count.</p>
</td></tr>
<tr><td><code id="stat_week_2d_+3A_week_start">week_start</code></td>
<td>
<p>Day the week begins (defaults to Monday).
Can be specified as a case-insensitive English weekday name such as &quot;Monday&quot;
or an integer. Since you generally won't want to mix definitions, it is
more convenient to control this globally with the <code>"phylepic.week_start"</code>
option, e.g. <code>options(phylepic.week_start = "Monday")</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The computed aesthetics are similar to those of <code>stat_bin_2d()</code>, including
<code>after_stat(count)</code>, <code>after_stat(density)</code>, and the bin positions and sizes:
<code>after_stat(xmin)</code>, <code>after_stat(height)</code>, and so on.
</p>


<h3>Value</h3>

<p>ggplot2 stat layer.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(ggplot2)

set.seed(1)
events &lt;- rep(as.Date("2024-01-31") - 0:30, rpois(31, 6))
values &lt;- round(rgamma(length(events), 1, 0.01))
df &lt;- data.frame(date = events, value = values)

ggplot(df) + stat_week_2d(aes(date, value), week_start = "Monday")
</code></pre>

<hr>
<h2 id='week_breaks'>Breaks for week-binning date axes</h2><span id='topic+week_breaks'></span>

<h3>Description</h3>

<p>Breaks for week-binning date axes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>week_breaks(width = 1L, week_start = getOption("phylepic.week_start"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="week_breaks_+3A_width">width</code></td>
<td>
<p>Number of weeks between breaks (e.g. <code>2</code> will give a break every
fortnight).</p>
</td></tr>
<tr><td><code id="week_breaks_+3A_week_start">week_start</code></td>
<td>
<p>Day the week begins (defaults to Monday).
Can be specified as a case-insensitive English weekday name such as &quot;Monday&quot;
or an integer. Since you generally won't want to mix definitions, it is
more convenient to control this globally with the <code>"phylepic.week_start"</code>
option, e.g. <code>options(phylepic.week_start = "Monday")</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A break function suitable for use in <code><a href="ggplot2.html#topic+scale_date">ggplot2::scale_x_date()</a></code> et al.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
